<!DOCTYPE html><html><head><title>Help for package abc.data</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {abc.data}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#human'>
<p>A set of R objects containing observed data from three human</p>
populations, and simulated data under three different demographic
models. The data set is used to illustrate model selection and parameter
inference in an ABC framework (see the vignette of the <code>abc</code>
package for more details).</a></li>
<li><a href='#musigma2'>
<p>A set of objects used to estimate the population mean and variance in a</p>
Gaussian model with ABC (see the vignette of the <code>abc</code>
package for more details).</a></li>
<li><a href='#ppc'>
<p>Data to illustrate the posterior predictive checks for the data</p>
<code>human</code>. <code>ppc</code> and <code>human</code> are used to
illustrate model selection and parameter inference in an ABC framework
(see the vignette of the <code>abc</code> package for more details).</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Data Only: Tools for Approximate Bayesian Computation (ABC)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Description:</td>
<td>Contains data which are used by functions of the 'abc' package.</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-24 10:04:54 UTC; hornik</td>
</tr>
<tr>
<td>Author:</td>
<td>Csillery Katalin [aut],
  Lemaire Louisiane [aut],
  Francois Olivier [aut],
  Blum Michael [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Blum Michael &lt;michael.blum.temp@gmail.com&gt;</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-24 10:15:14 UTC</td>
</tr>
</table>
<hr>
<h2 id='human'>
A set of R objects containing observed data from three human
populations, and simulated data under three different demographic
models. The data set is used to illustrate model selection and parameter
inference in an ABC framework (see the vignette of the <code>abc</code>
package for more details).
</h2><span id='topic+human'></span><span id='topic+stat.voight'></span><span id='topic+par.italy.sim'></span><span id='topic+stat.3pops.sim'></span><span id='topic+models'></span>

<h3>Description</h3>

<p><code>data(human)</code> loads in four R objects: <code>stat.voight</code> is a
data frame with 3 rows and 3 columns and contains the observed summary
statistics for three human populations, <code>stat.3pops.sim</code> is also a
data frame with 150,000 rows and 3 columns and contains the simulated
summary statistics, <code>models</code> is a vector of character strings of
length 150,000 and contains the model indices, <code>par.italy.sim</code> is a
data frame with 50,000 rows and 4 columns and contains the parameter
values that were used to simulate data under a population bottleneck
model. The corresponding summary statistics can be subsetted from the
<code>stat.3pops.sim</code> object as <code>subset(stat.3pops.sim,
  subset=models=="bott")</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(human)</code></pre>


<h3>Format</h3>

<p>The <code>stat.voight</code> data frame contains the following columns:
</p>

<dl>
<dt><code>pi</code></dt><dd>
<p>The mean nucleotide diversity over 50 loci in 3 human
populations, Hausa, Italian, and Chinese.
</p>
</dd>
<dt><code>TajD.m</code></dt><dd>
<p>The mean of Tajima's D statistic over 50 loci in 3 human
populations, Hausa, Italian, and Chinese.
</p>
</dd>
<dt><code>TajD.v</code></dt><dd>
<p>The variance of Tajima's D statistic over 50 loci in 3 human
populations, Hausa, Italian, and Chinese.
</p>
</dd>
</dl>

<p>Each row represents a simulation. Under each model 50,000 simulations
were performed. Row names indicate the type of demographic model.
</p>
<p>The <code>stat.3pops.sim</code> data frame contains the following columns:
</p>

<dl>
<dt><code>pi</code></dt><dd>
<p>The mean of nucleotide diversity over 50 simulated loci under 3
demographic scenarios: constant size population, population
bottleneck, and population expansion.
</p>
</dd>
<dt><code>TajD.m</code></dt><dd>
<p>The mean of Tajima's D statistic over 50 simulated loci under 3
demographic scenarios: constant size population, population
bottleneck, and population expansion.
</p>
</dd>
<dt><code>TajD.v</code></dt><dd>
<p>The variance of Tajima's D statistic over 50 simulated loci under
3 demographic scenarios: constant size population, population
bottleneck, and population expansion.
</p>
</dd>
</dl>

<p>Each row represents a simulation. Under each model 50,000 simulations
were performed. Row names indicate the type of demographic model.
</p>
<p>The <code>par.italy.sim</code> data frame contains the following columns:
</p>

<dl>
<dt><code>Ne</code></dt><dd>
<p>The effective population size.
</p>
</dd>
<dt><code>a</code></dt><dd>
<p>The intensity of the bottleneck (i.e. the ratio of the population sizes
before and during the bottleneck).
</p>
</dd>
<dt><code>duration</code></dt><dd>
<p>The duration of the bottleneck.
</p>
</dd>
<dt><code>start</code></dt><dd>
<p>The start of the bottleneck.
</p>
</dd>
</dl>

<p>Each row represents a simulation. 
</p>
<p><code>models</code> contains the names of the demographic models.  
</p>


<h3>Details</h3>

<p>Data is provided to estimate the posterior probabilities of classical
demographic scenarios in three human populations: Hausa, Italian, and
Chinese. These three populations represent the three continents:
Africa, Europe, Asia, respectively. <code>par.italy.sim</code> may then used
to estimate the ancestral population size of the European population
assuming a bottleneck model.
</p>
<p>It is generally believed that African human populations are expanding,
while human populations from outside of Africa have gone through a
population bottleneck. Tajima's D statistic has been classically used
to detect changes in historical population size. A negative Tajima's D
signifies an excess of low frequency polymorphisms, indicating
population size expansion. While a positive Tajima's D indicates low
levels of both low and high frequency polymorphisms, thus a sign of a
population bottleneck. In constant size populations, Tajima's D is
expected to be zero.
</p>
<p>With the help of the <code>human</code> data one can reach these expected
conclusions for the three human population samples, in accordance with
the conclusions of Voight et al. (2005) (where the observed statistics
was taken from), but using ABC.
</p>


<h3>Source</h3>

<p>The observed statistics were taken from Voight et al. 2005 (Table
1.). Also, the same input parameters were used as in Voight et al. 2005 to
simulate data under the three demographic models. Simulations were
performed using the software ms and the summary statistics were
calculated using sample_stats (Hudson 1983).
</p>


<h3>References</h3>

<p>B. F. Voight, A. M. Adams, L. A. Frisse, Y. Qian, R. R. Hudson and
A. Di Rienzo (2005) Interrogating multiple aspects of variation in a
full resequencing data set to infer human population size
changes. PNAS <b>102</b>, 18508-18513.
</p>
<p>Hudson, R. R. (2002) Generating samples under a Wright-Fisher neutral
model of genetic variation. Bioinformatics <b>18</b> 337-338.
</p>

<hr>
<h2 id='musigma2'>
A set of objects used to estimate the population mean and variance in a
Gaussian model with ABC (see the vignette of the <code>abc</code>
package for more details).
</h2><span id='topic+musigma2'></span><span id='topic+par.sim'></span><span id='topic+stat.sim'></span><span id='topic+stat.obs'></span><span id='topic+post.mu'></span><span id='topic+post.sigma2'></span>

<h3>Description</h3>

<p><code>musigma2</code> loads in five R objects: <code>par.sim</code> is a data
frame and contains the parameter values of the simulated data sets,
<code>stat</code> is a data frame and contains the simulated summary
statistics, <code>stat.obs</code> is a data frame and contains the observed
summary statistics, <code>post.mu</code> and <code>post.sigma2</code> are data
frames and contain the true posterior distributions for the two
parameters of interest, <code class="reqn">\mu</code> and <code class="reqn">\sigma^2</code>, respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(musigma2)</code></pre>


<h3>Format</h3>

<p>The <code>par.sim</code> data frame contains the following columns:
</p>

<dl>
<dt><code>mu</code></dt><dd>
<p>The population mean.
</p>
</dd>
<dt><code>sigma2</code></dt><dd>
<p>The population variance.
</p>
</dd>
</dl>

<p>The <code>stat.sim</code> and <code>stat.obs</code> data frames contain the
following columns:
</p>

<dl>
<dt><code>mean</code></dt><dd>
<p>The sample mean.
</p>
</dd>
<dt><code>var</code></dt><dd>
<p>The logarithm of the sample variance.
</p>
</dd>
</dl>

<p>The <code>post.mu</code> and <code>post.sigma2</code> data frames contain the
following columns:
</p>

<dl>
<dt><code>x</code></dt><dd>
<p>the coordinates of the points where the density is estimated.
</p>
</dd>
<dt><code>y</code></dt><dd>
<p>the posterior density values.
</p>
</dd>
</dl>



<h3>Details</h3>

<p>The prior of <code class="reqn">\sigma^2</code> is an inverse <code class="reqn">\chi^2</code> distribution
with one degree of freedom. The prior of <code class="reqn">\mu</code> is a normal
distribution with variance of <code class="reqn">\sigma^2</code>. For this simple example,
the closed form of the posterior distribution is available.
</p>


<h3>Source</h3>

<p>The observed statistics are the mean and variance of the sepal of
<em>Iris setosa</em>, estimated from part of the <code>iris</code> data.
</p>
<p>The data were collected by Anderson, Edgar.
</p>


<h3>References</h3>

<p>Anderson, E. (1935). The irises of the Gaspe Peninsula, <em>Bulletin
of the American Iris Society</em>, <b>59</b>, 2-5.
</p>

<hr>
<h2 id='ppc'>
Data to illustrate the posterior predictive checks for the data
<code><a href="#topic+human">human</a></code>. <code>ppc</code> and <code><a href="#topic+human">human</a></code> are used to
illustrate model selection and parameter inference in an ABC framework
(see the vignette of the <code>abc</code> package for more details).
</h2><span id='topic+ppc'></span><span id='topic+post.bott'></span>

<h3>Description</h3>

<p><code>data(ppc)</code> loads in the data frame <code>post.bott</code>, which
contains the summary statistics calculated from data simulated a
posteriori under the bottleneck model (see <code>data(human)</code> and the
package's vignette for more details).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ppc)</code></pre>


<h3>Format</h3>

<p>The <code>post.bott</code> data frame contains the following columns:
</p>

<dl>
<dt><code>pi</code></dt><dd>
<p>The mean nucleotide diversity over 50 loci.
</p>
</dd>
<dt><code>TajD.m</code></dt><dd>
<p>The mean of Tajima's D statistic over 50 loci.
</p>
</dd>
<dt><code>TajD.v</code></dt><dd>
<p>The variance of Tajima's D statistic over 50 loci.
</p>
</dd>
</dl>

<p>Each row represents a simulation. 1000 simulations
were performed under the bottleneck model.
</p>

</main>


</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
