<!DOCTYPE html><html><head><title>Help for package ABC.RAP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ABC.RAP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#annotate_data'><p>Annotating the filtered probes</p></a></li>
<li><a href='#annotation_file'><p>annotation file for the 450k probes</p></a></li>
<li><a href='#CpG_hits'><p>Identifying genes for which multiple CpG sites show significant methylation difference</p></a></li>
<li><a href='#delta_beta_data'><p>Applying delta beta analysis to calculate the difference between cases and controls</p></a></li>
<li><a href='#filter_data'><p>Filtering DNA methylation 450k non_specific probes</p></a></li>
<li><a href='#nonspecific_probes'><p>450k DNA methylation non specific probes</p></a></li>
<li><a href='#overlap_data'><p>Overlapping Student's t-test and delta beta results</p></a></li>
<li><a href='#plot_candidate_genes'><p>Plotting highly different and significant probes annotated by their corresponding gene names</p></a></li>
<li><a href='#plot_data'><p>Overview description of the DNA methylation pattern for cases and controls</p></a></li>
<li><a href='#plot_gene'><p>Plotting and exporting methylation profile for candidate genes</p></a></li>
<li><a href='#process.ABC.RAP'><p>An automated analysis applying all ABC.RAP functions in one script</p></a></li>
<li><a href='#test_data'><p>test dataset of 450k DNA methylation</p></a></li>
<li><a href='#ttest_data'><p>applying t-test analysis</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Array Based CpG Region Analysis Pipeline</td>
</tr>
<tr>
<td>Version:</td>
<td>0.9.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Abdulmonem Alsaleh [cre, aut], Robert Weeks [aut], Ian Morison [aut], RStudio [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Abdulmonem Alsaleh &lt;a.alsaleh@hotmail.co.nz&gt;</td>
</tr>
<tr>
<td>Contact:</td>
<td>Ian Morison &lt;ian.morison@otago.ac.nz&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>It aims to identify candidate genes that are “differentially
    methylated” between cases and controls. It applies Student’s t-test and delta beta analysis to
    identify candidate genes containing multiple “CpG sites”.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>graphics, stats, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>GNU make</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1.9000</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-10-20 01:50:08 UTC; abdul</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-10-20 10:52:16</td>
</tr>
</table>
<hr>
<h2 id='annotate_data'>Annotating the filtered probes</h2><span id='topic+annotate_data'></span>

<h3>Description</h3>

<p>This function annotates each filtered probe with gene name, chromosome number, probe location, distance from transcription start site (TSS), and relation to CpG islands. The annotation file is based on &quot;UCSC platform&quot; annotation format and was obtained from Illumina GPL13534_HumanMethylation450_15017482_v1.1 file (BS0010894-AQP_content.bpm).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annotate_data(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annotate_data_+3A_x">x</code></td>
<td>
<p>the filtered probes from filter_data</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(test_data)
data(nonspecific_probes)
data(annotation_file)
test_data_filtered &lt;- filter_data(test_data)
test_data_annotated &lt;- annotate_data(test_data_filtered)

</code></pre>

<hr>
<h2 id='annotation_file'>annotation file for the 450k probes</h2><span id='topic+annotation_file'></span>

<h3>Description</h3>

<p>UCSC annotation for the 450k DNA methylation probes. The annotation was obtained from &quot;Illumina GPL13534_HumanMethylation450_15017482_v1.1&quot; file with few amendments on the gene names</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("annotation_file")</code></pre>


<h3>Format</h3>

<p>A data frame</p>

<hr>
<h2 id='CpG_hits'>Identifying genes for which multiple CpG sites show significant methylation difference</h2><span id='topic+CpG_hits'></span>

<h3>Description</h3>

<p>This function calculates the number of significantly different CpG sites between cases and controls for each gene and produces a frequency table with genes that have more than one CpG site.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CpG_hits(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CpG_hits_+3A_x">x</code></td>
<td>
<p>Results from the overlap_data function</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
data(test_data)
data(nonspecific_probes)
data(annotation_file)
test_data_filtered &lt;- filter_data(test_data)
test_data_ttest &lt;- ttest_data(test_data_filtered, 1, 2, 3, 4, 1e-3)
test_data_delta_beta &lt;- delta_beta_data(test_data_filtered, 1, 2, 3, 4, 0.5, -0.5, 0.94, 0.06)
test_overlapped_data &lt;- overlap_data(test_data_ttest, test_data_delta_beta)
test_CpG_hits &lt;- CpG_hits(test_overlapped_data)

</code></pre>

<hr>
<h2 id='delta_beta_data'>Applying delta beta analysis to calculate the difference between cases and controls</h2><span id='topic+delta_beta_data'></span>

<h3>Description</h3>

<p>This function calculates the delta beta value for the filtered probes. It calculates the difference in mean DNA methylation between cases and controls for each probe. Also, it selects probes with DNA methylation differences that are higher in cases than controls by a user specified meth_cutoff value and differences that are lower in cases than controls by the unmeth_cutoff value. In addition, the function provides the option to specify probes where the average beta value of the cases or controls is greater than a high_meth cutoff value or less than a low_meth cutoff value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>delta_beta_data(x, cases_column_1, cases_column_n, controls_column_1,
  controls_column_n, meth_cutoff, unmeth_cutoff, high_meth, low_meth)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="delta_beta_data_+3A_x">x</code></td>
<td>
<p>the filtered 450k probes from filter_data function</p>
</td></tr>
<tr><td><code id="delta_beta_data_+3A_cases_column_1">cases_column_1</code></td>
<td>
<p>The first column (column number) for cases in the filtered dataset</p>
</td></tr>
<tr><td><code id="delta_beta_data_+3A_cases_column_n">cases_column_n</code></td>
<td>
<p>The last column (column number) for cases in the filtered dataset</p>
</td></tr>
<tr><td><code id="delta_beta_data_+3A_controls_column_1">controls_column_1</code></td>
<td>
<p>The first column (column number) for controls in the filtered dataset</p>
</td></tr>
<tr><td><code id="delta_beta_data_+3A_controls_column_n">controls_column_n</code></td>
<td>
<p>The last column (column number) for controls in the filtered dataset</p>
</td></tr>
<tr><td><code id="delta_beta_data_+3A_meth_cutoff">meth_cutoff</code></td>
<td>
<p>The cutoff level for the methylation difference between cases and controls (cases minus controls)</p>
</td></tr>
<tr><td><code id="delta_beta_data_+3A_unmeth_cutoff">unmeth_cutoff</code></td>
<td>
<p>The cutoff level for the methylation difference between controls and cases (cases minus controls). Consequently, it requires a negative value.</p>
</td></tr>
<tr><td><code id="delta_beta_data_+3A_high_meth">high_meth</code></td>
<td>
<p>The upper margin for the highly methylated probes</p>
</td></tr>
<tr><td><code id="delta_beta_data_+3A_low_meth">low_meth</code></td>
<td>
<p>The lower margin for the low methylation</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(test_data)
data(nonspecific_probes)
test_data_filtered &lt;- filter_data(test_data)
test_data_delta_beta &lt;- delta_beta_data(test_data_filtered, 1, 2, 3, 4, 0.5, -0.5, 0.94, 0.06)

</code></pre>

<hr>
<h2 id='filter_data'>Filtering DNA methylation 450k non_specific probes</h2><span id='topic+filter_data'></span>

<h3>Description</h3>

<p>This function filters the reported nonspecific probes, and also filters probes that interrogate SNPs of minor allele frequency (MAF) &gt; 0.1. A list of nonspecific probes was obtained from Chen et al (2013) supplementary files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_data(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_data_+3A_x">x</code></td>
<td>
<p>The normalised beta values in a data matrix format, where conditions are arranged in columns and cg probes are arranged in rows.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Chen YA, Lemire M, Choufani S, et al. Discovery of cross-reactive probes and polymorphic CpGs in the Illumina Infinium HumanMethylation450 microarray. Epigenetics 2013;8:203-9.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(test_data)
data(nonspecific_probes)
test_data_filtered &lt;- filter_data(test_data)

</code></pre>

<hr>
<h2 id='nonspecific_probes'>450k DNA methylation non specific probes</h2><span id='topic+nonspecific_probes'></span>

<h3>Description</h3>

<p>data frame of the non specific probes that need to be filtered out from 450k array datasets</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("nonspecific_probes")</code></pre>


<h3>Format</h3>

<p>A data frame</p>


<h3>Details</h3>

<p>These non specific probes interrogates SNPs with mean allelic frequency (MAF) &gt; 0.1, and also those that don't align uniquely on the genome. The list of nonspecific probes was obtained from Chen et al (2013) supplementary files</p>


<h3>References</h3>

<p>Chen YA, Lemire M, Choufani S, et al. Discovery of cross-reactive probes and polymorphic CpGs in the Illumina Infinium HumanMethylation450 microarray. Epigenetics 2013;8;203-9</p>

<hr>
<h2 id='overlap_data'>Overlapping Student's t-test and delta beta results</h2><span id='topic+overlap_data'></span>

<h3>Description</h3>

<p>This function overlaps the results from both Student’s t-test and delta beta analyses to identify probes (CpG sites) that are highly and significantly different between cases and controls.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>overlap_data(x, y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="overlap_data_+3A_x">x</code></td>
<td>
<p>Results from t-test or delta beta analyses</p>
</td></tr>
<tr><td><code id="overlap_data_+3A_y">y</code></td>
<td>
<p>Results from t-test or delta beta analyses</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
data(test_data)
data(nonspecific_probes)
data(annotation_file)
test_data_filtered &lt;- filter_data(test_data)
test_data_ttest &lt;- ttest_data(test_data_filtered, 1, 2, 3, 4, 1e-3)
test_data_delta_beta &lt;- delta_beta_data(test_data_filtered, 1, 2, 3, 4, 0.5, -0.5, 0.94, 0.06)
test_overlapped_data &lt;- overlap_data(test_data_ttest, test_data_delta_beta)

</code></pre>

<hr>
<h2 id='plot_candidate_genes'>Plotting highly different and significant probes annotated by their corresponding gene names</h2><span id='topic+plot_candidate_genes'></span>

<h3>Description</h3>

<p>This function plots the potential candidate genes for which multiple CpG sites show significant difference.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_candidate_genes(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_candidate_genes_+3A_x">x</code></td>
<td>
<p>Results from the overlap_data function</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
data(test_data)
data(nonspecific_probes)
data(annotation_file)
test_data_filtered &lt;- filter_data(test_data)
test_data_ttest &lt;- ttest_data(test_data_filtered, 1, 2, 3, 4, 1e-3)
test_data_delta_beta &lt;- delta_beta_data(test_data_filtered, 1, 2, 3, 4, 0.5, -0.5, 0.94, 0.06)
test_overlapped_data &lt;- overlap_data(test_data_ttest, test_data_delta_beta)
plot_candidate_genes(test_overlapped_data)

</code></pre>

<hr>
<h2 id='plot_data'>Overview description of the DNA methylation pattern for cases and controls</h2><span id='topic+plot_data'></span>

<h3>Description</h3>

<p>This function produces four distribution plots that summarise the DNA methylation patterns for cases (top left) and controls (top right). The top two histograms show the pattern of mean DNA methylation levels for cases and controls. The bottom two plots show the difference in DNA methylation between cases and controls (a boxplot comparing methylation profile for cases and controls, and a delta beta plot describing the methylation difference between cases and controls). The function also provides summary statistics for the delta beta analysis that can be used to select cutoff values for the delta_beta_data function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_data(x, cases_column_1, cases_column_n, controls_column_1,
  controls_column_n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_data_+3A_x">x</code></td>
<td>
<p>The filtered 450k probes from filter_data() function</p>
</td></tr>
<tr><td><code id="plot_data_+3A_cases_column_1">cases_column_1</code></td>
<td>
<p>The first column (column number) for cases in the filtered dataset</p>
</td></tr>
<tr><td><code id="plot_data_+3A_cases_column_n">cases_column_n</code></td>
<td>
<p>The last column (column number) for cases in the filtered dataset</p>
</td></tr>
<tr><td><code id="plot_data_+3A_controls_column_1">controls_column_1</code></td>
<td>
<p>The first column (column number) for controls in the filtered dataset</p>
</td></tr>
<tr><td><code id="plot_data_+3A_controls_column_n">controls_column_n</code></td>
<td>
<p>The last column (column number) for controls in the filtered dataset</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(test_data)
data(nonspecific_probes)
test_data_filtered &lt;- filter_data(test_data)
plot_data(test_data_filtered, 1, 2, 3, 4)

</code></pre>

<hr>
<h2 id='plot_gene'>Plotting and exporting methylation profile for candidate genes</h2><span id='topic+plot_gene'></span>

<h3>Description</h3>

<p>This function explores the DNA methylation profile for any gene. The function generates four plots: the top plots show the difference in DNA methylation between cases and controls (a bar chart of the delta beta values for all probes arranged from 5’ to 3’ positions and a plot showing the difference in mean DNA methylation between cases and controls). The bottom plots show the distribution of DNA methylation for each probe that interrogates a CpG site in the investigated gene, for cases (left) and controls (right), respectively. Also, an annotation table for the arranged probes is generated with the following columns: probe names, gene name, distance from TSS, mean methylation for cases, mean methylation for controls, delta beta values (cases minus controls), and t-test p.values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_gene(x, b, cases_column_1, cases_column_n, controls_column_1,
  controls_column_n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_gene_+3A_x">x</code></td>
<td>
<p>The filtered and annotated 450k probes</p>
</td></tr>
<tr><td><code id="plot_gene_+3A_b">b</code></td>
<td>
<p>Gene name between quotation marks</p>
</td></tr>
<tr><td><code id="plot_gene_+3A_cases_column_1">cases_column_1</code></td>
<td>
<p>The first column (column number) for cases in the filtered dataset</p>
</td></tr>
<tr><td><code id="plot_gene_+3A_cases_column_n">cases_column_n</code></td>
<td>
<p>The last column (column number) for cases in the filtered dataset</p>
</td></tr>
<tr><td><code id="plot_gene_+3A_controls_column_1">controls_column_1</code></td>
<td>
<p>The first column (column number) for controls in the filtered dataset</p>
</td></tr>
<tr><td><code id="plot_gene_+3A_controls_column_n">controls_column_n</code></td>
<td>
<p>The last column (column number) for controls in the filtered dataset</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'> 
data(test_data)
data(nonspecific_probes)
data(annotation_file)
test_data_filtered &lt;- filter_data(test_data)
test_data_annotated &lt;- annotate_data(test_data_filtered)
KLHL34 &lt;- plot_gene(test_data_annotated, 'KLHL34', 1, 2, 3, 4)

</code></pre>

<hr>
<h2 id='process.ABC.RAP'>An automated analysis applying all ABC.RAP functions in one script</h2><span id='topic+process.ABC.RAP'></span>

<h3>Description</h3>

<p>This function processes the ABC.RAP workflow automatically
</p>


<h3>Usage</h3>

<pre><code class='language-R'>process.ABC.RAP(x, cases_column_1, cases_column_n, controls_column_1,
  controls_column_n, ttest_cutoff, meth_cutoff, unmeth_cutoff, high_meth,
  low_meth)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.ABC.RAP_+3A_x">x</code></td>
<td>
<p>The normalised beta values in a data matrix format, where conditions are arranged in columns and cg probes are arranged in rows.</p>
</td></tr>
<tr><td><code id="process.ABC.RAP_+3A_cases_column_1">cases_column_1</code></td>
<td>
<p>The first column (column number) for cases in the filtered dataset</p>
</td></tr>
<tr><td><code id="process.ABC.RAP_+3A_cases_column_n">cases_column_n</code></td>
<td>
<p>The last column (column number) for cases in the filtered dataset</p>
</td></tr>
<tr><td><code id="process.ABC.RAP_+3A_controls_column_1">controls_column_1</code></td>
<td>
<p>The first column (column number) for controls in the filtered dataset</p>
</td></tr>
<tr><td><code id="process.ABC.RAP_+3A_controls_column_n">controls_column_n</code></td>
<td>
<p>The last column (column number) for controls in the filtered dataset</p>
</td></tr>
<tr><td><code id="process.ABC.RAP_+3A_ttest_cutoff">ttest_cutoff</code></td>
<td>
<p>The cutoff level to filter insignificant p-values</p>
</td></tr>
<tr><td><code id="process.ABC.RAP_+3A_meth_cutoff">meth_cutoff</code></td>
<td>
<p>The cutoff level for the methylation difference between cases and controls (cases minus controls)</p>
</td></tr>
<tr><td><code id="process.ABC.RAP_+3A_unmeth_cutoff">unmeth_cutoff</code></td>
<td>
<p>The cutoff level for the methylation difference between controls and cases (controls minus cases). Consequently, it requires a negative value.</p>
</td></tr>
<tr><td><code id="process.ABC.RAP_+3A_high_meth">high_meth</code></td>
<td>
<p>The upper margin for the highly methylated probes</p>
</td></tr>
<tr><td><code id="process.ABC.RAP_+3A_low_meth">low_meth</code></td>
<td>
<p>The lower margin for the low methylation</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
data(test_data)
data(nonspecific_probes)
data(annotation_file)
process.ABC.RAP(test_data, 1, 2, 3, 4, 1e-3,  0.5, -0.5, 0.94, 0.06)

</code></pre>

<hr>
<h2 id='test_data'>test dataset of 450k DNA methylation</h2><span id='topic+test_data'></span>

<h3>Description</h3>

<p>This is a small dataset of 450k DNA methylation array with 10000 probes. The dataset has four columns; columns 1 and 2 contain normalised beta values for paediatric B ALL cases, and columns 3 and 4 contain beta values for controls (remission cases)</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("test_data")</code></pre>


<h3>Format</h3>

<p>A data frame</p>


<h3>Details</h3>

<p>a small test dataset</p>


<h3>References</h3>

<p>Busche S, Ge B, Vidal R, etc. Integration of high-resolution methylome and transcriptome analyses to dissect epigenomic changes in childhood acute lymphoblastic leukaemia. Cancer Research 2013; 73(14); 4323-4336</p>

<hr>
<h2 id='ttest_data'>applying t-test analysis</h2><span id='topic+ttest_data'></span>

<h3>Description</h3>

<p>This function applies &quot;two.sided&quot;, unequal variance Student's t-test analysis for each probe comparing cases and controls. A cutoff for p-values can be entered to minimise multiple testing bias to filter insignificant p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ttest_data(x, cases_column_1, cases_column_n, controls_column_1,
  controls_column_n, ttest_cutoff)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ttest_data_+3A_x">x</code></td>
<td>
<p>The filtered 450k probes from filter_data() function</p>
</td></tr>
<tr><td><code id="ttest_data_+3A_cases_column_1">cases_column_1</code></td>
<td>
<p>The first column (column number) for cases in the filtered dataset</p>
</td></tr>
<tr><td><code id="ttest_data_+3A_cases_column_n">cases_column_n</code></td>
<td>
<p>The last column (column number) for cases in the filtered dataset</p>
</td></tr>
<tr><td><code id="ttest_data_+3A_controls_column_1">controls_column_1</code></td>
<td>
<p>The first column (column number) for controls in the filtered dataset</p>
</td></tr>
<tr><td><code id="ttest_data_+3A_controls_column_n">controls_column_n</code></td>
<td>
<p>The last column (column number) for controls in the filtered dataset</p>
</td></tr>
<tr><td><code id="ttest_data_+3A_ttest_cutoff">ttest_cutoff</code></td>
<td>
<p>The cutoff level to filter insignificant p-values</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(test_data)
data(nonspecific_probes)
test_data_filtered &lt;- filter_data(test_data)
test_data_ttest &lt;- ttest_data(test_data_filtered, 1, 2, 3, 4, 1e-3)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
