<!DOCTYPE html><html lang="en"><head><title>Help for package abcrf</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {abcrf}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#abcrf'><p>Create an ABC-RF object: a classification random forest from a reference table towards performing an ABC model choice</p></a></li>
<li><a href='#covRegAbcrf'>
<p>Predict posterior covariance between two parameters for new data using two reg-ABC-RF objects</p></a></li>
<li><a href='#densityPlot'><p>Plot the posterior density given a new summary statistic</p></a></li>
<li><a href='#err.abcrf'><p>Calculate and plot for different numbers of tree, the out-of-bag errors associated with an ABC-RF object</p></a></li>
<li><a href='#err.regAbcrf'><p>Calculate and plot for different numbers of tree, the out-of-bag mean squared errors associated with a REG-ABC-RF object</p></a></li>
<li><a href='#plot.abcrf'><p>Plot of an ABC-RF object</p></a></li>
<li><a href='#plot.regAbcrf'><p>Plot of a reg-ABC-RF object</p></a></li>
<li><a href='#predict.abcrf'><p>Predict and evaluate the posterior probability of the MAP</p>
for new data using an ABC-RF object</a></li>
<li><a href='#predict.regAbcrf'>
<p>Predict posterior expectation, median, variance and quantiles</p>
given a new dataset using a reg-ABC-RF object</a></li>
<li><a href='#predictOOB'>
<p>Predict out-of-bag posterior expectation, median, variance, quantiles and error measures using a reg-ABC-RF object</p></a></li>
<li><a href='#readRefTable'><p>Read a reference table simulated from DIYABC</p></a></li>
<li><a href='#regAbcrf'>
<p>Create a reg-ABC-RF object: a regression random forest from a reference table aimed out predicting posterior expectation, variance and quantiles for a parameter</p></a></li>
<li><a href='#snp'><p>A simulated example in population genetics</p></a></li>
<li><a href='#variableImpPlot'><p>Variable importance plot from a random forest</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Approximate Bayesian Computation via Random Forests</td>
</tr>
<tr>
<td>Version:</td>
<td>1.9</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-08-09</td>
</tr>
<tr>
<td>Description:</td>
<td>Performs Approximate Bayesian Computation (ABC) model choice and parameter inference via random forests.
  Pudlo P., Marin J.-M., Estoup A., Cornuet J.-M., Gautier M. and Robert C. P. (2016) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtv684">doi:10.1093/bioinformatics/btv684</a>&gt;.
  Estoup A., Raynal L., Verdu P. and Marin J.-M. <a href="http://journal-sfds.fr/article/view/709">http://journal-sfds.fr/article/view/709</a>.
  Raynal L., Marin J.-M., Pudlo P., Ribatet M., Robert C. P. and Estoup A. (2019) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbty867">doi:10.1093/bioinformatics/bty867</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Imports:</td>
<td>readr, MASS, matrixStats, ranger, doParallel, parallel,
foreach, stringr, Rcpp (&ge; 0.11.2)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.1)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jean-Michel Marin &lt;jean-michel.marin@umontpellier.fr&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Author:</td>
<td>Jean-Michel Marin [aut, cre],
  Louis Raynal [aut],
  Pierre Pudlo [aut],
  Christian P. Robert [ctb],
  Arnaud Estoup [ctb]</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-08-09 09:16:29 UTC; jmm</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-08-09 15:30:19 UTC</td>
</tr>
</table>
<hr>
<h2 id='abcrf'>Create an ABC-RF object: a classification random forest from a reference table towards performing an ABC model choice</h2><span id='topic+abcrf'></span><span id='topic+abcrf.formula'></span>

<h3>Description</h3>

<p><code>abcrf</code> constructs a random forest from a reference table towards performing
an ABC model choice. Basically, the reference table (i.e. the dataset that will
be treated with the present package) includes a column with the index
of the models to be compared and additional columns corresponding
to the values of the simulated summary statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'formula'
abcrf(formula, data, group=list(), lda=TRUE, ntree=500, sampsize=min(1e5, nrow(data)),
paral=FALSE, ncores= if(paral) max(detectCores()-1,1) else 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="abcrf_+3A_formula">formula</code></td>
<td>
<p>a formula: left of ~, variable representing
the model index; right of ~, summary statistics of the reference table.</p>
</td></tr>
<tr><td><code id="abcrf_+3A_data">data</code></td>
<td>
<p>a data frame containing the reference table.</p>
</td></tr>
<tr><td><code id="abcrf_+3A_group">group</code></td>
<td>
<p>a list containing groups (at least 2) of model(s)
on which the model choice will be performed. This is not necessarily a partition,
one or more models can be excluded from the elements of the list and
by default no grouping is done.</p>
</td></tr>
<tr><td><code id="abcrf_+3A_lda">lda</code></td>
<td>
<p>should LDA scores be added to the list of summary statistics?</p>
</td></tr>
<tr><td><code id="abcrf_+3A_ntree">ntree</code></td>
<td>
<p>number of trees to grow in the forest, by default 500 trees.</p>
</td></tr>
<tr><td><code id="abcrf_+3A_sampsize">sampsize</code></td>
<td>
<p>size of the sample from the reference table to grow a tree
of the classification forest, by default the minimum between the number of elements of the reference table and 100,000.</p>
</td></tr>
<tr><td><code id="abcrf_+3A_paral">paral</code></td>
<td>
<p>a boolean that indicates if the calculations of the classification random forest
(forest used to assign a model to the observed dataset) should be parallelized.</p>
</td></tr>
<tr><td><code id="abcrf_+3A_ncores">ncores</code></td>
<td>
<p>the number of CPU cores to use. If paral=TRUE, it is used the number of CPU cores minus 1. If ncores is not specified and <code><a href="parallel.html#topic+detectCores">detectCores</a></code> does not detect the number of CPU cores with success then 1 core is used.</p>
</td></tr>
<tr><td><code id="abcrf_+3A_...">...</code></td>
<td>
<p>additional arguments to be passed on to <code><a href="ranger.html#topic+ranger">ranger</a></code> used to construct the classification random forest that preditcs the selected model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>abcrf</code>, which is a list with the
following components:
</p>
<table role = "presentation">
<tr><td><code>call</code></td>
<td>
<p>the original call to <code>abcrf</code>,</p>
</td></tr>
<tr><td><code>lda</code></td>
<td>
<p>a boolean indicating if LDA scores have been added to the list of summary statistics,</p>
</td></tr>
<tr><td><code>formula</code></td>
<td>
<p>the formula used to construct the classification random forest,</p>
</td></tr>
<tr><td><code>group</code></td>
<td>
<p>a list contining the groups of model(s) used. This list is empty if no grouping has been performed,</p>
</td></tr>
<tr><td><code>model.rf</code></td>
<td>
<p>an object of class <code>randomForest</code> containing the trained forest with the reference table,</p>
</td></tr>
<tr><td><code>model.lda</code></td>
<td>
<p>an object of class <code>lda</code> containing the Linear Discriminant Analysis based on the reference table,</p>
</td></tr>
<tr><td><code>prior.err</code></td>
<td>
<p>prior error rates of model selection on the reference table, estimated with the &quot;out-of-bag&quot; error of the forest.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Pudlo P., Marin J.-M., Estoup A., Cornuet J.-M., Gautier M. and Robert, C. P. (2016)
<em>Reliable ABC model choice via random forests</em> Bioinformatics
<a href="https://doi.org/10.1093/bioinformatics/btv684">doi:10.1093/bioinformatics/btv684</a>
</p>
<p>Estoup A., Raynal L., Verdu P. and Marin J.-M. (2018)
<em>Model choice using Approximate Bayesian Computation and Random Forests: analyses based on model grouping to make inferences about the genetic history of Pygmy human populations</em> Jounal de la Société Française de Statistique
<a href="http://journal-sfds.fr/article/view/709">http://journal-sfds.fr/article/view/709</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.abcrf">plot.abcrf</a></code>,
<code><a href="#topic+predict.abcrf">predict.abcrf</a></code>,
<code><a href="#topic+err.abcrf">err.abcrf</a></code>,
<code><a href="ranger.html#topic+ranger">ranger</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp)
modindex &lt;- snp$modindex[1:500]
sumsta &lt;- snp$sumsta[1:500,]
data1 &lt;- data.frame(modindex, sumsta)
model.rf1 &lt;- abcrf(modindex~., data = data1, ntree=100)
model.rf1
model.rf2 &lt;- abcrf(modindex~., data = data1, group = list(c("1","2"),"3"), ntree=100)
model.rf2
</code></pre>

<hr>
<h2 id='covRegAbcrf'>
Predict posterior covariance between two parameters for new data using two reg-ABC-RF objects
</h2><span id='topic+covRegAbcrf'></span><span id='topic+covRegAbcrf.regAbcrf'></span>

<h3>Description</h3>

<p>Using two reg-ABC-RF objects constructed on the same reference table for two different response variables, this function predicts the posterior covariance between those two response variables, given a new dataset of summaries.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'regAbcrf'
covRegAbcrf(regForest1, regForest2, obs, training1, training2,
ntree=500, mtry=max(floor((dim(training1)[2]-1)/3), 1),
sampsize=min(1e5, dim(training1)[1]), paral = FALSE, 
ncores = if(paral) max(detectCores()-1,1) else 1, paral.predict = FALSE,
ncores.predict = if(paral.predict) max(detectCores()-1,1) else 1, ... )
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="covRegAbcrf_+3A_regforest1">regForest1</code>, <code id="covRegAbcrf_+3A_regforest2">regForest2</code></td>
<td>
<p><code><a href="#topic+regAbcrf">regAbcrf</a></code> objects.</p>
</td></tr>
<tr><td><code id="covRegAbcrf_+3A_obs">obs</code></td>
<td>
<p>a data frame containing the summary statistics of the observed data sets.</p>
</td></tr>
<tr><td><code id="covRegAbcrf_+3A_training1">training1</code>, <code id="covRegAbcrf_+3A_training2">training2</code></td>
<td>
<p>data frames containing the reference table respectively used to train the <code><a href="#topic+regAbcrf">regAbcrf</a></code> objects regForest1 and regForest2.</p>
</td></tr>
<tr><td><code id="covRegAbcrf_+3A_ntree">ntree</code></td>
<td>
<p>number of trees to grow in the forest, by default equal to 500 trees.</p>
</td></tr>
<tr><td><code id="covRegAbcrf_+3A_mtry">mtry</code></td>
<td>
<p>Number of variables to possibly split at in each node for the regression random forest. Default is the minimum between 1 and the number of variables divided by 3.</p>
</td></tr>
<tr><td><code id="covRegAbcrf_+3A_sampsize">sampsize</code></td>
<td>
<p>size of the sample from the reference table used to grow a tree of the regression forest, by default the minimum between the number of elements of the reference table and 100,000.</p>
</td></tr>
<tr><td><code id="covRegAbcrf_+3A_paral">paral</code></td>
<td>
<p>a boolean that indicates whether or not the calculations of the regression random forest (forest used to predict a response from the observed dataset) should be parallelized.</p>
</td></tr>
<tr><td><code id="covRegAbcrf_+3A_ncores">ncores</code></td>
<td>
<p>the number of CPU cores to use. If paral=TRUE, it is used the number of CPU cores minus 1. If ncores is not specified and <code><a href="parallel.html#topic+detectCores">detectCores</a></code> does not detect the number of CPU cores with success then 1 core is used.</p>
</td></tr>
<tr><td><code id="covRegAbcrf_+3A_paral.predict">paral.predict</code></td>
<td>
<p>a boolean that indicates if random forests predictions should be parallelized.</p>
</td></tr>
<tr><td><code id="covRegAbcrf_+3A_ncores.predict">ncores.predict</code></td>
<td>
<p>the number of CPU cores to use for the regression random forest predictions. If paral.predict=TRUE, it is used the number of CPU cores minus 1. If ncores.predict is not specified and <code><a href="parallel.html#topic+detectCores">detectCores</a></code> does not detect the number of CPU cores with success then 1 core is used.</p>
</td></tr>
<tr><td><code id="covRegAbcrf_+3A_...">...</code></td>
<td>
<p>additional arguments to be passed on to <code><a href="ranger.html#topic+ranger">ranger</a></code> used to construct the regression random forest that predicts posterior covariance.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>covRegAbcrf</code> returns predicted posterior covariances between response variables of two reg-ABC-RF objects, for a new data set.
</p>


<h3>References</h3>

<p>Raynal L., Marin J.-M. Pudlo P., Ribatet M., Robert C. P. and Estoup, A. (2019)
<em>ABC random forests for Bayesian parameter inference</em> Bioinformatics
<a href="https://doi.org/10.1093/bioinformatics/bty867">doi:10.1093/bioinformatics/bty867</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+regAbcrf">regAbcrf</a></code>,
<code><a href="#topic+predict.regAbcrf">predict.regAbcrf</a></code>,
<code><a href="#topic+err.regAbcrf">err.regAbcrf</a></code>,
<code><a href="#topic+plot.regAbcrf">plot.regAbcrf</a></code>,
<code><a href="ranger.html#topic+ranger">ranger</a></code>,
<code><a href="#topic+densityPlot">densityPlot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp)
modindex &lt;- snp$modindex
sumsta &lt;- snp$sumsta[modindex == "3",]
r &lt;- snp$param$r[modindex == "3"]
r &lt;- r[1:500]
sumsta &lt;- sumsta[1:500,]
data2 &lt;- data.frame(r, sumsta)
model.rf.r &lt;- regAbcrf(r~., data2, ntree=100)
N1 &lt;- snp$param$N1[modindex == "3"]
N1 &lt;- N1[1:500]
data3 &lt;- data.frame(N1, sumsta) 
model.rf.N1 &lt;- regAbcrf(N1~., data3, ntree=100)
data(snp.obs)
covRegAbcrf(model.rf.r, model.rf.N1, snp.obs, 
            data2, data3, ntree=100)
</code></pre>

<hr>
<h2 id='densityPlot'>Plot the posterior density given a new summary statistic</h2><span id='topic+densityPlot'></span><span id='topic+densityPlot.regAbcrf'></span>

<h3>Description</h3>

<p>Given a reg-ABC-RF object and a new value of the summary statistics, 
<code>densityPlot</code> gives the corresponding posterior density plot of the parameter, as well as the prior (in grey).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'regAbcrf'
densityPlot(object, obs, training,
add=TRUE, main="Posterior density", log="", xlim=NULL,
ylim=NULL, xlab=NULL, ylab=NULL, paral=FALSE,
ncores= if(paral) max(detectCores()-1,1) else 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="densityPlot_+3A_object">object</code></td>
<td>
<p>a <code><a href="#topic+regAbcrf">regAbcrf</a></code> object.</p>
</td></tr>
<tr><td><code id="densityPlot_+3A_obs">obs</code></td>
<td>
<p>a data frame containing the summary statistics of the observed data sets.</p>
</td></tr>
<tr><td><code id="densityPlot_+3A_training">training</code></td>
<td>
<p>the data frame containing the reference table used to train the <code><a href="#topic+regAbcrf">regAbcrf</a></code> object.</p>
</td></tr>
<tr><td><code id="densityPlot_+3A_add">add</code></td>
<td>
<p>a boolean that indicates if the posterior distributions should be ploted on the same graph or not, when more than one observed summary statistics is given.</p>
</td></tr>
<tr><td><code id="densityPlot_+3A_main">main</code></td>
<td>
<p>main title to be used for the posterior density plot.</p>
</td></tr>
<tr><td><code id="densityPlot_+3A_log">log</code></td>
<td>
<p>a character string which contains &quot;x&quot; if the x axis is to be logarithmic, &quot;y&quot; if the y axis is to be logarithmic and &quot;xy&quot; or &quot;yx&quot; if both axes are to be logarithmic. The default value &quot;&quot; implies no logarithmic transfomation.</p>
</td></tr>
<tr><td><code id="densityPlot_+3A_xlim">xlim</code></td>
<td>
<p>range of the abscissa.</p>
</td></tr>
<tr><td><code id="densityPlot_+3A_ylim">ylim</code></td>
<td>
<p>range of the ordinate.</p>
</td></tr>
<tr><td><code id="densityPlot_+3A_xlab">xlab</code></td>
<td>
<p>label of the abscissa.</p>
</td></tr>
<tr><td><code id="densityPlot_+3A_ylab">ylab</code></td>
<td>
<p>label of the ordinate.</p>
</td></tr>
<tr><td><code id="densityPlot_+3A_paral">paral</code></td>
<td>
<p>a boolean that indicates if random forests predictions should be parallelized.</p>
</td></tr>
<tr><td><code id="densityPlot_+3A_ncores">ncores</code></td>
<td>
<p>the number of CPU cores to use for the regression random forest predictions. If paral=TRUE, it is used the number of CPU cores minus 1. If ncores is not specified and <code><a href="parallel.html#topic+detectCores">detectCores</a></code> does not detect the number of CPU cores with success then 1 core is used.</p>
</td></tr>
<tr><td><code id="densityPlot_+3A_...">...</code></td>
<td>
<p>additional arguments to be passed on to <code><a href="stats.html#topic+density">density</a></code>, as for example the smoothing bandwidth <code>bw</code> to be used.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Raynal L., Marin J.-M. Pudlo P., Ribatet M., Robert C. P. and Estoup, A. (2019)
<em>ABC random forests for Bayesian parameter inference</em> Bioinformatics
<a href="https://doi.org/10.1093/bioinformatics/bty867">doi:10.1093/bioinformatics/bty867</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+regAbcrf">regAbcrf</a></code>,
<code><a href="#topic+predict.regAbcrf">predict.regAbcrf</a></code>,
<code><a href="#topic+err.regAbcrf">err.regAbcrf</a></code>,
<code><a href="#topic+covRegAbcrf">covRegAbcrf</a></code>,
<code><a href="ranger.html#topic+ranger">ranger</a></code>,
<code><a href="#topic+plot.regAbcrf">plot.regAbcrf</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp)
modindex &lt;- snp$modindex
sumsta &lt;- snp$sumsta[modindex == "3",]
r &lt;- snp$param$r[modindex == "3"]
r &lt;- r[1:500]
sumsta &lt;- sumsta[1:500,]
data2 &lt;- data.frame(r, sumsta)
model.rf.r &lt;- regAbcrf(r~., data2, ntree=100)
data(snp.obs)
densityPlot(model.rf.r, snp.obs, data2, ylab="density", main = "Posterior density of r")
</code></pre>

<hr>
<h2 id='err.abcrf'>Calculate and plot for different numbers of tree, the out-of-bag errors associated with an ABC-RF object</h2><span id='topic+err.abcrf'></span>

<h3>Description</h3>

<p><code>err.abcrf</code> returns out-of-bag errors and plot them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>err.abcrf(object, training, paral=FALSE,
ncores= if(paral) max(detectCores()-1,1) else 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="err.abcrf_+3A_object">object</code></td>
<td>
<p>an <code>abcrf</code> object.</p>
</td></tr>
<tr><td><code id="err.abcrf_+3A_training">training</code></td>
<td>
<p>the data frame containing the reference table used to train the <code><a href="#topic+abcrf">abcrf</a></code> object.</p>
</td></tr>
<tr><td><code id="err.abcrf_+3A_paral">paral</code></td>
<td>
<p>a boolean that indicates if random forests predictions should be parallelized.</p>
</td></tr>
<tr><td><code id="err.abcrf_+3A_ncores">ncores</code></td>
<td>
<p>the number of CPU cores to use for the random forest predictions. If paral=TRUE, it is used the number of CPU cores minus 1. If ncores is not specified and <code><a href="parallel.html#topic+detectCores">detectCores</a></code> does not detect the number of CPU cores with success then 1 core is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with 2 columns: the number of trees and the out-of-bag errors. Errors are computed from 40 trees to the total number.
</p>


<h3>References</h3>

<p>Pudlo P., Marin J.-M., Estoup A., Cornuet J.-M., Gautier M. and Robert, C. P. (2016)
<em>Reliable ABC model choice via random forests</em> Bioinformatics
<a href="https://doi.org/10.1093/bioinformatics/btv684">doi:10.1093/bioinformatics/btv684</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+abcrf">abcrf</a></code>,
<code><a href="#topic+predict.abcrf">predict.abcrf</a></code>,
<code><a href="#topic+plot.abcrf">plot.abcrf</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp)
modindex &lt;- snp$modindex[1:500]
sumsta &lt;- snp$sumsta[1:500,]
data1 &lt;- data.frame(modindex, sumsta)
model.rf &lt;- abcrf(modindex~., data1, ntree=100)
err.rf &lt;- err.abcrf(model.rf, data1)
</code></pre>

<hr>
<h2 id='err.regAbcrf'>Calculate and plot for different numbers of tree, the out-of-bag mean squared errors associated with a REG-ABC-RF object</h2><span id='topic+err.regAbcrf'></span>

<h3>Description</h3>

<p><code>err.regAbcrf</code> returns out-of-bag mean squared errors and plot them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>err.regAbcrf(object, training, paral=FALSE, 
ncores= if(paral) max(detectCores()-1,1) else 1, what="mean")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="err.regAbcrf_+3A_object">object</code></td>
<td>
<p>a <code>regAbcrf</code> object.</p>
</td></tr>
<tr><td><code id="err.regAbcrf_+3A_training">training</code></td>
<td>
<p>the data frame containing the reference table used to train the <code><a href="#topic+regAbcrf">regAbcrf</a></code> object.</p>
</td></tr>
<tr><td><code id="err.regAbcrf_+3A_paral">paral</code></td>
<td>
<p>a boolean that indicates if random forests predictions should be parallelized.</p>
</td></tr>
<tr><td><code id="err.regAbcrf_+3A_ncores">ncores</code></td>
<td>
<p>the number of CPU cores to use for the random forest predictions. If paral=TRUE, it is used the number of CPU cores minus 1. If ncores is not specified and <code><a href="parallel.html#topic+detectCores">detectCores</a></code> does not detect the number of CPU cores with success then 1 core is used.</p>
</td></tr>
<tr><td><code id="err.regAbcrf_+3A_what">what</code></td>
<td>
<p>a string caracter &quot;mean&quot; or &quot;median&quot; indicating if the predictions are computed with mean or median of the response variable.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with 2 columns: the number of trees and the out-of-bag mean squared errors. NAs might be returned if the number of trees is too low. Errors are computed from 40 trees to the total number.
</p>


<h3>References</h3>

<p>Raynal L., Marin J.-M. Pudlo P., Ribatet M., Robert C. P. and Estoup, A. (2019)
<em>ABC random forests for Bayesian parameter inference</em> Bioinformatics
<a href="https://doi.org/10.1093/bioinformatics/bty867">doi:10.1093/bioinformatics/bty867</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+regAbcrf">regAbcrf</a></code>,
<code><a href="#topic+predict.regAbcrf">predict.regAbcrf</a></code>,
<code><a href="#topic+plot.regAbcrf">plot.regAbcrf</a></code>,
<code><a href="#topic+densityPlot">densityPlot</a></code>,
<code><a href="#topic+covRegAbcrf">covRegAbcrf</a></code>,
<code><a href="ranger.html#topic+ranger">ranger</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(snp)
  modindex &lt;- snp$modindex
  sumsta &lt;- snp$sumsta[modindex == "3",]
  r &lt;- snp$param$r[modindex == "3"]
  r &lt;- r[1:500]
  sumsta &lt;- sumsta[1:500,]
  data2 &lt;- data.frame(r, sumsta)
  model.rf.r &lt;- regAbcrf(r~., data2, ntree=100)
  err.regAbcrf(model.rf.r, data2)
</code></pre>

<hr>
<h2 id='plot.abcrf'>Plot of an ABC-RF object</h2><span id='topic+plot.abcrf'></span>

<h3>Description</h3>

<p><code>plot.abcrf</code> provides both a variable importance plot of a model choice ABC-RF object
and the projection of the reference table on the LDA axes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'abcrf'
plot(x, training, obs=NULL, n.var=20, pdf=FALSE, xlim=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.abcrf_+3A_x">x</code></td>
<td>
<p>an <code>abcrf</code> object.</p>
</td></tr>
<tr><td><code id="plot.abcrf_+3A_training">training</code></td>
<td>
<p>the data frame containing the reference table used to train the <code><a href="#topic+abcrf">abcrf</a></code> object.</p>
</td></tr>
<tr><td><code id="plot.abcrf_+3A_obs">obs</code></td>
<td>
<p>a vector containing the summary statistics of an observed dataset that
will be added to the graph of the projected reference table (black star or vertical line).</p>
</td></tr>
<tr><td><code id="plot.abcrf_+3A_n.var">n.var</code></td>
<td>
<p>number of variables in the variable importance representation.</p>
</td></tr>
<tr><td><code id="plot.abcrf_+3A_pdf">pdf</code></td>
<td>
<p>a boolean that indicates if a pdf version of the graph(s) should be saved in the current directory.</p>
</td></tr>
<tr><td><code id="plot.abcrf_+3A_xlim">xlim</code></td>
<td>
<p>range of the abscissa for the variable importance plot.</p>
</td></tr>
<tr><td><code id="plot.abcrf_+3A_...">...</code></td>
<td>
<p>not used.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The graph of the reference table projected
on the LD axes is shown only if LD axes has
been added to the set of summary statistics
in the call of <code>abcrf</code>.
</p>


<h3>References</h3>

<p>Pudlo P., Marin J.-M., Estoup A., Cornuet J.-M., Gautier M. and Robert, C. P. (2016)
<em>Reliable ABC model choice via random forests</em> Bioinformatics
<a href="https://doi.org/10.1093/bioinformatics/btv684">doi:10.1093/bioinformatics/btv684</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+abcrf">abcrf</a></code>,
<code><a href="#topic+predict.abcrf">predict.abcrf</a></code>,
<code><a href="#topic+err.abcrf">err.abcrf</a></code>,
<code><a href="#topic+variableImpPlot">variableImpPlot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp)
modindex &lt;- snp$modindex[1:500]
sumsta &lt;- snp$sumsta[1:500,]
data1 &lt;- data.frame(modindex, sumsta)
model.rf &lt;- abcrf(modindex~., data1, ntree=100)
plot(model.rf, data1)
data(snp.obs)
plot(model.rf, data1, obs=snp.obs[1,])
</code></pre>

<hr>
<h2 id='plot.regAbcrf'>Plot of a reg-ABC-RF object</h2><span id='topic+plot.regAbcrf'></span>

<h3>Description</h3>

<p><code>plot.regAbcrf</code> provides a variable importance plot used to construct the reg-ABC-RF object, as measured by <code><a href="ranger.html#topic+ranger">ranger</a></code> with the argument importance='impurity'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'regAbcrf'
plot(x, n.var=min(30, length(x$model.rf$variable.importance)), xlim=NULL, main=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.regAbcrf_+3A_x">x</code></td>
<td>
<p>a <code><a href="#topic+regAbcrf">regAbcrf</a></code> object.</p>
</td></tr>
<tr><td><code id="plot.regAbcrf_+3A_n.var">n.var</code></td>
<td>
<p>number of variables in the variable importance representation. The default value is equal to the minimum between 30 and the number of summary statistics.</p>
</td></tr>
<tr><td><code id="plot.regAbcrf_+3A_xlim">xlim</code></td>
<td>
<p>range of the abscissa for the variable importance plot.</p>
</td></tr>
<tr><td><code id="plot.regAbcrf_+3A_main">main</code></td>
<td>
<p>an overall title for the variable importance plot.</p>
</td></tr>
<tr><td><code id="plot.regAbcrf_+3A_...">...</code></td>
<td>
<p>not used.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Raynal L., Marin J.-M. Pudlo P., Ribatet M., Robert C. P. and Estoup, A. (2019)
<em>ABC random forests for Bayesian parameter inference</em> Bioinformatics
<a href="https://doi.org/bioinformatics/bty867">doi:bioinformatics/bty867</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+regAbcrf">regAbcrf</a></code>,
<code><a href="#topic+predict.regAbcrf">predict.regAbcrf</a></code>,
<code><a href="#topic+err.regAbcrf">err.regAbcrf</a></code>,
<code><a href="#topic+covRegAbcrf">covRegAbcrf</a></code>,
<code><a href="ranger.html#topic+ranger">ranger</a></code>,
<code><a href="#topic+densityPlot">densityPlot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp)
modindex &lt;- snp$modindex
sumsta &lt;- snp$sumsta[modindex == "3",]
r &lt;- snp$param$r[modindex == "3"]
r &lt;- r[1:500]
sumsta &lt;- sumsta[1:500,]
data2 &lt;- data.frame(r, sumsta)
model.rf.r &lt;- regAbcrf(r~., data2, ntree=100)
plot(model.rf.r)
</code></pre>

<hr>
<h2 id='predict.abcrf'>Predict and evaluate the posterior probability of the MAP
for new data using an ABC-RF object</h2><span id='topic+predict.abcrf'></span>

<h3>Description</h3>

<p>Based on an ABC-RF object this function predicts the best model for
new data and evaluate the posterior probability of the MAP.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'abcrf'
predict(object, obs, training, ntree = 1000, 
sampsize = min(1e5,  object$model.rf$num.samples ), paral = FALSE,
ncores = if(paral) max(detectCores()-1,1) else 1, paral.predict = FALSE, 
ncores.predict = if(paral.predict) max(detectCores()-1,1) else 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predict.abcrf_+3A_object">object</code></td>
<td>
<p>an <code><a href="#topic+abcrf">abcrf</a></code> object.</p>
</td></tr>
<tr><td><code id="predict.abcrf_+3A_obs">obs</code></td>
<td>
<p>a data frame containing the summary statistics of the observed data sets.</p>
</td></tr>
<tr><td><code id="predict.abcrf_+3A_training">training</code></td>
<td>
<p>the data frame containing the reference table used to train the <code><a href="#topic+abcrf">abcrf</a></code> object.</p>
</td></tr>
<tr><td><code id="predict.abcrf_+3A_ntree">ntree</code></td>
<td>
<p>number of trees to grow in the regression forest, by default 1,000 trees.</p>
</td></tr>
<tr><td><code id="predict.abcrf_+3A_sampsize">sampsize</code></td>
<td>
<p>size of the sample from the reference table used to grow a tree
of the forest, by default the minimum between the number of elements of the reference table and 100,000.</p>
</td></tr>
<tr><td><code id="predict.abcrf_+3A_paral">paral</code></td>
<td>
<p>a boolean that indicates if the calculations of the regression random forest
(forest that returns the posterior probability of the selected model) should be parallelized.</p>
</td></tr>
<tr><td><code id="predict.abcrf_+3A_ncores">ncores</code></td>
<td>
<p>the number of CPU cores to use for the regression random forest construction. If paral=TRUE, it is used the number of CPU cores minus 1. If ncores is not specified and <code><a href="parallel.html#topic+detectCores">detectCores</a></code> does not detect the number of CPU cores with success then 1 core is used.</p>
</td></tr>
<tr><td><code id="predict.abcrf_+3A_paral.predict">paral.predict</code></td>
<td>
<p>a boolean that indicates if random forests predictions should be parallelized.</p>
</td></tr>
<tr><td><code id="predict.abcrf_+3A_ncores.predict">ncores.predict</code></td>
<td>
<p>the number of CPU cores to use for random forest predictions (classification and regression). If paral.predict=TRUE, it is used the number of CPU cores minus 1. If ncores.predict is not specified and <code><a href="parallel.html#topic+detectCores">detectCores</a></code> does not detect the number of CPU cores with success then 1 core is used.</p>
</td></tr>
<tr><td><code id="predict.abcrf_+3A_...">...</code></td>
<td>
<p>additional arguments to be passed on to <code><a href="ranger.html#topic+ranger">ranger</a></code> used to construct
the regression random forest that estimates the posterior probability of the selected model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>abcrfpredict</code>, which is a list with the
following components:
</p>
<table role = "presentation">
<tr><td><code>allocation</code></td>
<td>
<p>indices of the selected models for each observed data set,</p>
</td></tr>
<tr><td><code>vote</code></td>
<td>
<p>votes for each observed dataset,</p>
</td></tr>
<tr><td><code>post.prob</code></td>
<td>
<p>ABC-RF approximations of the posterior probability of the selected model for each observed dataset.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Pudlo P., Marin J.-M., Estoup A., Cornuet J.-M., Gautier M. and Robert, C. P. (2016)
<em>Reliable ABC model choice via random forests</em> Bioinformatics
<a href="https://doi.org/10.1093/bioinformatics/btv684">doi:10.1093/bioinformatics/btv684</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+abcrf">abcrf</a></code>,
<code><a href="#topic+plot.abcrf">plot.abcrf</a></code>,
<code><a href="#topic+err.abcrf">err.abcrf</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp)
modindex &lt;- snp$modindex[1:500]
sumsta &lt;- snp$sumsta[1:500,]
data1 &lt;- data.frame(modindex, sumsta)
model.rf &lt;- abcrf(modindex~., data1, ntree=100)
data(snp.obs)
predict(model.rf, snp.obs, data1, ntree=100)
</code></pre>

<hr>
<h2 id='predict.regAbcrf'>
Predict posterior expectation, median, variance and quantiles
given a new dataset using a reg-ABC-RF object
</h2><span id='topic+predict.regAbcrf'></span>

<h3>Description</h3>

<p>Based on a reg-ABC-RF object this function predicts the posterior expectation, median, variance, quantiles for the corresponding parameter given new dataset. Somes posterior errors can be 
computed at an higher computational price.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'regAbcrf'
predict(object, obs, training, quantiles=c(0.025,0.975),
paral = FALSE, ncores = if(paral) max(detectCores()-1,1) else 1, rf.weights = FALSE,
post.err.med = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predict.regAbcrf_+3A_object">object</code></td>
<td>
<p>a <code><a href="#topic+regAbcrf">regAbcrf</a></code> object.</p>
</td></tr>
<tr><td><code id="predict.regAbcrf_+3A_obs">obs</code></td>
<td>
<p>a data frame containing the summary statistics of the observed data sets.</p>
</td></tr>
<tr><td><code id="predict.regAbcrf_+3A_training">training</code></td>
<td>
<p>the data frame containing the reference table used to train the <code><a href="#topic+regAbcrf">regAbcrf</a></code> object.</p>
</td></tr>
<tr><td><code id="predict.regAbcrf_+3A_quantiles">quantiles</code></td>
<td>
<p>numeric vector of probabilities with values in [0,1]. The default value is equal to <code>c(0.025, 0.975)</code>.</p>
</td></tr>
<tr><td><code id="predict.regAbcrf_+3A_paral">paral</code></td>
<td>
<p>a boolean that indicates if random forests predictions should be parallelized.</p>
</td></tr>
<tr><td><code id="predict.regAbcrf_+3A_ncores">ncores</code></td>
<td>
<p>the number of CPU cores to use for the regression random forest predictions. If paral=TRUE, it is used the number of CPU cores minus 1. If ncores is not specified and <code><a href="parallel.html#topic+detectCores">detectCores</a></code> does not detect the number of CPU cores with success then 1 core is used.</p>
</td></tr>
<tr><td><code id="predict.regAbcrf_+3A_rf.weights">rf.weights</code></td>
<td>
<p>a boolean that indicates if the random forest weights used to predict quantities of interest should we returned. The default value is FALSE.</p>
</td></tr>
<tr><td><code id="predict.regAbcrf_+3A_post.err.med">post.err.med</code></td>
<td>
<p>a boolean that indicates if posterior errors based on posterior medians should be computed. The default value is FALSE. If computed, this function might take a much more time.</p>
</td></tr>
<tr><td><code id="predict.regAbcrf_+3A_...">...</code></td>
<td>
<p>optional arguments to be passed on to the function <code><a href="ranger.html#topic+predict.ranger">predict.ranger</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>regAbcrfpredict</code>, which is a list with the following components:
</p>
<table role = "presentation">
<tr><td><code>expectation</code></td>
<td>
<p>predicted posterior expectation for each oberved data set,</p>
</td></tr>
<tr><td><code>med</code></td>
<td>
<p>predicted posterior median for each oberved data set,</p>
</td></tr>
<tr><td><code>variance</code></td>
<td>
<p>predicted posterior variance for each observed data set, computed by reusing weights, this quantity is also the posterior mean squared error,</p>
</td></tr>
<tr><td><code>variance.cdf</code></td>
<td>
<p>predicted posterior variance for each observed data set, computed by approximation of the cumulative distribution function,</p>
</td></tr>
<tr><td><code>quantiles</code></td>
<td>
<p>predicted posterior quantiles for each observed data set,</p>
</td></tr>
<tr><td><code>weights</code></td>
<td>
<p>a matrix composed of the weights used to predict quantities of interest. Returned if <code>rf.weights</code> is TRUE,</p>
</td></tr>
<tr><td><code>post.NMAE.mean</code></td>
<td>
<p>posterior normalized mean absolute error obtained using the out-of-bag posterior expectation (mean) and previously computed random forest weights, for each observed data set,</p>
</td></tr>
<tr><td><code>post.MSE.med</code></td>
<td>
<p>posterior mean squared error obtained using the out-of-bag posterior median and previously computed random forest weights, for each observed data set,</p>
</td></tr>
<tr><td><code>post.NMAE.med</code></td>
<td>
<p>posterior normalized mean absolute error obtained using the out-of-bag posterior expectation (mean) and previously computed random forest weights, for each observed data set,</p>
</td></tr>
<tr><td><code>prior.MSE</code></td>
<td>
<p>prior mean squared error computed with training out-of-bag prediction based on mean of response variable,</p>
</td></tr>
<tr><td><code>prior.NMAE</code></td>
<td>
<p>prior normalized mean absolute error computed with training out-of-bag predictions based on mean of response variable,</p>
</td></tr>
<tr><td><code>prior.MSE.med</code></td>
<td>
<p>prior mean squared error computed with training out-of-bag predictions based on median of response variable,</p>
</td></tr>
<tr><td><code>prior.NMAE.med</code></td>
<td>
<p>prior normalized mean absolute error with training out-of-bag predictions based on median of response variable,</p>
</td></tr>
<tr><td><code>prior.coverage</code></td>
<td>
<p>prior credible inteval coverage computed for training instances, if only two quantiles are of interest, NULL otherwise.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Raynal L., Marin J.-M. Pudlo P., Ribatet M., Robert C. P. and Estoup, A. (2019)
<em>ABC random forests for Bayesian parameter inference</em> Bioinformatics
<a href="https://doi.org/10.1093/bioinformatics/bty867">doi:10.1093/bioinformatics/bty867</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+regAbcrf">regAbcrf</a></code>,
<code><a href="#topic+predictOOB">predictOOB</a></code>,
<code><a href="#topic+plot.regAbcrf">plot.regAbcrf</a></code>,
<code><a href="#topic+err.regAbcrf">err.regAbcrf</a></code>,
<code><a href="#topic+covRegAbcrf">covRegAbcrf</a></code>,
<code><a href="ranger.html#topic+ranger">ranger</a></code>,
<code><a href="#topic+densityPlot">densityPlot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp)
modindex &lt;- snp$modindex
sumsta &lt;- snp$sumsta[modindex == "3",]
r &lt;- snp$param$r[modindex == "3"]
r &lt;- r[1:500]
sumsta &lt;- sumsta[1:500,]
data2 &lt;- data.frame(r, sumsta)
model.rf.r &lt;- regAbcrf(r~., data2, ntree=100)
data(snp.obs)
predict(model.rf.r, snp.obs, data2)
</code></pre>

<hr>
<h2 id='predictOOB'>
Predict out-of-bag posterior expectation, median, variance, quantiles and error measures using a reg-ABC-RF object
</h2><span id='topic+predictOOB'></span><span id='topic+predictOOB.regAbcrf'></span>

<h3>Description</h3>

<p>Based on a reg-ABC-RF object this function predicts the out-of-bag posterior expectation, median, variance, quantiles, mean squared errors, normalized mean absolute errors, credible interval and coverage, for the corresponding parameter using the out-of-bag observations of the training data set.
</p>
<p>Mean squared errors and normalized mean absolute errors are computed both with mean and median of the response variable.
</p>
<p>Memory allocation issues might be encountered when the size of the training data set is large.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'regAbcrf'
predictOOB(object, training, quantiles=c(0.025,0.975), paral = FALSE,
ncores = if(paral) max(detectCores()-1,1) else 1,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predictOOB_+3A_object">object</code></td>
<td>
<p>a <code><a href="#topic+regAbcrf">regAbcrf</a></code> object.</p>
</td></tr>
<tr><td><code id="predictOOB_+3A_training">training</code></td>
<td>
<p>the data frame containing the reference table used to train the <code><a href="#topic+regAbcrf">regAbcrf</a></code> object.</p>
</td></tr>
<tr><td><code id="predictOOB_+3A_quantiles">quantiles</code></td>
<td>
<p>numeric vector of probabilities with values in [0,1]. The default value is equal to <code>c(0.025, 0.975)</code>.</p>
</td></tr>
<tr><td><code id="predictOOB_+3A_paral">paral</code></td>
<td>
<p>a boolean that indicates if training data predictions should be parallelized or not.</p>
</td></tr>
<tr><td><code id="predictOOB_+3A_ncores">ncores</code></td>
<td>
<p>the number of CPU cores to use for the regression random forest predictions. If paral=TRUE, it is used the number of CPU cores minus 1. If ncores is not specified and <code><a href="parallel.html#topic+detectCores">detectCores</a></code> does not detect the number of CPU cores with success then 1 core is used.</p>
</td></tr>
<tr><td><code id="predictOOB_+3A_...">...</code></td>
<td>
<p>optional arguments to be passed on to the function <code><a href="ranger.html#topic+predict.ranger">predict.ranger</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>regAbcrfOOBpredict</code>, which is a list with the following components:
</p>
<table role = "presentation">
<tr><td><code>expectation</code></td>
<td>
<p>predicted posterior expectation for each oberved data set,</p>
</td></tr>
<tr><td><code>med</code></td>
<td>
<p>predicted posterior median for each oberved data set,</p>
</td></tr>
<tr><td><code>variance</code></td>
<td>
<p>predicted posterior variance for each observed data set, computed by reusing weights,</p>
</td></tr>
<tr><td><code>variance.cdf</code></td>
<td>
<p>predicted posterior variance for each observed data set, computed by approximation of the cumulative distribution function,</p>
</td></tr>
<tr><td><code>quantiles</code></td>
<td>
<p>predicted posterior quantiles for each observed data set,</p>
</td></tr>
<tr><td><code>MSE</code></td>
<td>
<p>mean squared error computed with prediction based on mean of response variable,</p>
</td></tr>
<tr><td><code>NMAE</code></td>
<td>
<p>normalized mean absolute error computed with predictions based on mean of response variable,</p>
</td></tr>
<tr><td><code>MSE.med</code></td>
<td>
<p>mean squared error computed with predictions based on median of response variable,</p>
</td></tr>
<tr><td><code>NMAE.med</code></td>
<td>
<p>normalized mean absolute error with predictions based on median of response variable,</p>
</td></tr>
<tr><td><code>coverage</code></td>
<td>
<p>credible inteval coverage if only two quantiles are of interest, NULL otherwise.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Raynal L., Marin J.-M. Pudlo P., Ribatet M., Robert C. P. and Estoup, A. (2019)
<em>ABC random forests for Bayesian parameter inference</em> Bioinformatics
<a href="https://doi.org/10.1093/bioinformatics/bty867">doi:10.1093/bioinformatics/bty867</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+regAbcrf">regAbcrf</a></code>,
<code><a href="#topic+predict.regAbcrf">predict.regAbcrf</a></code>,
<code><a href="#topic+plot.regAbcrf">plot.regAbcrf</a></code>,
<code><a href="#topic+err.regAbcrf">err.regAbcrf</a></code>,
<code><a href="#topic+covRegAbcrf">covRegAbcrf</a></code>,
<code><a href="ranger.html#topic+ranger">ranger</a></code>,
<code><a href="#topic+densityPlot">densityPlot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp)
modindex &lt;- snp$modindex
sumsta &lt;- snp$sumsta[modindex == "3",]
r &lt;- snp$param$r[modindex == "3"]
r &lt;- r[1:500]
sumsta &lt;- sumsta[1:500,]
data2 &lt;- data.frame(r, sumsta)
model.rf.r &lt;- regAbcrf(r~., data2, ntree=100)
res &lt;- predictOOB(model.rf.r, data2) 
</code></pre>

<hr>
<h2 id='readRefTable'>Read a reference table simulated from DIYABC</h2><span id='topic+readRefTable'></span>

<h3>Description</h3>

<p><code>readRefTable</code> reads a reference table simulated from DIYABC thanks to a .bin and a .txt file, respectively containing the reference table and its header.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readRefTable(filename = "reftable.bin", header = "header.txt", N = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readRefTable_+3A_filename">filename</code></td>
<td>
<p>a .bin file from DIYABC containing the reference table.</p>
</td></tr>
<tr><td><code id="readRefTable_+3A_header">header</code></td>
<td>
<p>a .txt file containing the header of the reference table in <code>filename</code>.</p>
</td></tr>
<tr><td><code id="readRefTable_+3A_n">N</code></td>
<td>
<p>an integer indicating the number of observations to extract from the reference table. The default is 0 indicating that the whole reference table is used. Warning: if N is specified, nrecscen returns is obsolete.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table role = "presentation">
<tr><td><code>nrec</code></td>
<td>
<p>number of individuals of the reference table,</p>
</td></tr>
<tr><td><code>nscen</code></td>
<td>
<p>number of scenarios in the reference table,</p>
</td></tr>
<tr><td><code>nrecscen</code></td>
<td>
<p>number of individuals by scenario,</p>
</td></tr>
<tr><td><code>nparam</code></td>
<td>
<p>number of parameters by scenario,</p>
</td></tr>
<tr><td><code>scenarios</code></td>
<td>
<p>a vector of factor containing the scenario indices,</p>
</td></tr>
<tr><td><code>params</code></td>
<td>
<p>a matrix with the parameters,</p>
</td></tr>
<tr><td><code>stats</code></td>
<td>
<p>a matrix with the summary statistics.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Cornuet J.-M., Pudlo P., Veyssier J., Dehne-Garcia A., Gautier M., Leblois R., Marin J.-M. and Estoup A. (2014)
<em>DIYABC v2.0: a software to make Approximate Bayesian Computation inferences about population history using Single Nucleotide Polymorphism, DNA sequence and microsatellite data</em> Bioinformatics
<a href="https://doi.org/10.1093/bioinformatics/btn514">doi:10.1093/bioinformatics/btn514</a>
</p>

<hr>
<h2 id='regAbcrf'>
Create a reg-ABC-RF object: a regression random forest from a reference table aimed out predicting posterior expectation, variance and quantiles for a parameter</h2><span id='topic+regAbcrf'></span><span id='topic+regAbcrf.formula'></span>

<h3>Description</h3>

<p><code>regAbcrf</code> constructs a regression random forest from a reference table towards predicting posterior expectations, variances and quantiles of a parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'formula'
regAbcrf(formula, data, ntree=500,
mtry=max(floor((dim(data)[2]-1)/3), 1), sampsize=min(1e5, nrow(data)),
paral=FALSE, ncores=if(paral) max(detectCores()-1,1) else 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="regAbcrf_+3A_formula">formula</code></td>
<td>
<p>a formula: left of ~, variable representing
the response variable; right of ~, summary statistics of the reference table.</p>
</td></tr>
<tr><td><code id="regAbcrf_+3A_data">data</code></td>
<td>
<p>a data frame containing the reference table, composed of response variable (parameter) and summary statistics.</p>
</td></tr>
<tr><td><code id="regAbcrf_+3A_ntree">ntree</code></td>
<td>
<p>number of trees to grow in the forest, by default 500 trees.</p>
</td></tr>
<tr><td><code id="regAbcrf_+3A_mtry">mtry</code></td>
<td>
<p>Number of variables to possibly split at in each node. Default is the minimum between 1 and the number of variables divided by 3.</p>
</td></tr>
<tr><td><code id="regAbcrf_+3A_sampsize">sampsize</code></td>
<td>
<p>size of the sample from the reference table used to grow a tree of the regression forest, by default the minimum between the number of elements of the reference table and 100,000.</p>
</td></tr>
<tr><td><code id="regAbcrf_+3A_paral">paral</code></td>
<td>
<p>a boolean that indicates if the calculations of the regression random forest should be parallelized.</p>
</td></tr>
<tr><td><code id="regAbcrf_+3A_ncores">ncores</code></td>
<td>
<p>the number of CPU cores to use. If paral=TRUE, it is used the number of CPU cores minus 1. If ncores is not specified and <code><a href="parallel.html#topic+detectCores">detectCores</a></code> does not detect the number of CPU cores with success then 1 core is used.</p>
</td></tr>
<tr><td><code id="regAbcrf_+3A_...">...</code></td>
<td>
<p>additional arguments to be passed on to <code><a href="ranger.html#topic+ranger">ranger</a></code> used to construct the regression random forest that predicts the response variable.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>regAbcrf</code>, which is a list with the
following components:
</p>
<table role = "presentation">
<tr><td><code>call</code></td>
<td>
<p>the original call to <code>regAbcrf</code>,</p>
</td></tr>
<tr><td><code>formula</code></td>
<td>
<p>the formula used to construct the regression random forest,</p>
</td></tr>
<tr><td><code>model.rf</code></td>
<td>
<p>an object of class <code>ranger</code> containing the trained forest with the reference table.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Raynal L., Marin J.-M. Pudlo P., Ribatet M., Robert C. P. and Estoup, A. (2019)
<em>ABC random forests for Bayesian parameter inference</em> Bioinformatics
<a href="https://doi.org/10.1093/bioinformatics/bty867">doi:10.1093/bioinformatics/bty867</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.regAbcrf">plot.regAbcrf</a></code>,
<code><a href="#topic+err.regAbcrf">err.regAbcrf</a></code>,
<code><a href="#topic+predict.regAbcrf">predict.regAbcrf</a></code>,
<code><a href="#topic+covRegAbcrf">covRegAbcrf</a></code>,
<code><a href="ranger.html#topic+ranger">ranger</a></code>,
<code><a href="#topic+densityPlot">densityPlot</a></code>,
<code><a href="#topic+predictOOB">predictOOB</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp)
modindex &lt;- snp$modindex
sumsta &lt;- snp$sumsta[modindex == "3",]
r &lt;- snp$param$r[modindex == "3"]
r &lt;- r[1:500]
sumsta &lt;- sumsta[1:500,]
data2 &lt;- data.frame(r, sumsta)
model.rf.r &lt;- regAbcrf(r~., data2, ntree=100)
model.rf.r
</code></pre>

<hr>
<h2 id='snp'>A simulated example in population genetics</h2><span id='topic+snp'></span><span id='topic+snp.obs'></span>

<h3>Description</h3>

<p>The simulated example of population genetics with SNP loci used in Pudlo et al.
(2016): <code>snp</code> contains the reference table on which to perform ABC model choice, it also contains the simulated parameters to perform regression random forest.
<code>snp.obs</code> contains two pseudo-observed data sets. The first one
(<code>favorable</code>) should be easily allocated to a model, while that is not the case for the second one (<code>unfavorable</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(snp)
data(snp.obs)
</code></pre>


<h3>Format</h3>

<p><code>snp</code> is a list containing an ABC reference table of 10,000
simulations from a Bayesian prior predictive model (see Pudlo et al., 2016, for a description of the model choice issue). The first element, named <code>modindex</code> is a <code>factor</code> containing the model indices, the second element, <code>param</code>, is a data frame with seven simulated parameters. The last element of this list, named <code>sumsta</code>, contains the reference table on which to perform ABC model choice and parameter estimation.
</p>
<p><code>snp.obs</code> is a data frame containing the summary statistics of two pseudo-observed data sets.
</p>


<h3>Source</h3>

<p>Pudlo, P., Marin, J.-M., Estoup, A., Cornuet, J.-M., Gautier, M. and Robert, C.P. (2016)
<em>Reliable ABC model choice via random forests</em> Bioinformatics
<a href="https://doi.org/10.1093/bioinformatics/btv684">doi:10.1093/bioinformatics/btv684</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp)
data(snp.obs)
</code></pre>

<hr>
<h2 id='variableImpPlot'>Variable importance plot from a random forest</h2><span id='topic+variableImpPlot'></span>

<h3>Description</h3>

<p><code>variableImpPlot</code> provides a dotchart of variable importance as measured by <code><a href="ranger.html#topic+ranger">ranger</a></code> with the argument importance='impurity'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>variableImpPlot(object, 
n.var=min(30, length(object$model.rf$variable.importance)),
xlim=NULL, main=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="variableImpPlot_+3A_object">object</code></td>
<td>
<p>an <code>abcrf</code> or <code>regAbcrf</code> object.</p>
</td></tr>
<tr><td><code id="variableImpPlot_+3A_n.var">n.var</code></td>
<td>
<p>number of variables in the variable importance representation.</p>
</td></tr>
<tr><td><code id="variableImpPlot_+3A_xlim">xlim</code></td>
<td>
<p>range of the abscissa.</p>
</td></tr>
<tr><td><code id="variableImpPlot_+3A_main">main</code></td>
<td>
<p>an overall title for the variable importance plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly, the importance of the variables that were plotted.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+abcrf">abcrf</a></code>,
<code><a href="#topic+plot.abcrf">plot.abcrf</a></code>,
<code><a href="#topic+plot.regAbcrf">plot.regAbcrf</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp)
modindex &lt;- snp$modindex[1:500]
sumsta &lt;- snp$sumsta[1:500,]
data1 &lt;- data.frame(modindex, sumsta)
model.rf &lt;- abcrf(modindex~., data1, ntree=100)
variableImpPlot(model.rf)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
