<!DOCTYPE html><html><head><title>Help for package abctools</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {abctools}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#abctools-package'><p>Tools for ABC analyses</p></a></li>
<li><a href='#AS.select'>
<p>Summary statistics selection using approximate sufficiency.</p></a></li>
<li><a href='#AS.test'>
<p>Test for relative approximate sufficiency between two posterior samples.</p></a></li>
<li><a href='#coal'><p>Examples of coalescent data</p></a></li>
<li><a href='#combmat'><p>table of combinations</p></a></li>
<li><a href='#cov.pi'>
<p>Coverage property diagnostics</p></a></li>
<li><a href='#mc.ci'>
<p>Diagnostic plots for model choice coverage output</p></a></li>
<li><a href='#mincrit'>
<p>Summary statistics selection by minimizing a posterior sample measure.</p></a></li>
<li><a href='#nn.ent'><p>Works out entropy of a sample.</p></a></li>
<li><a href='#recalibrationABC'>
<p>ABC inference with a recalibration adjustment</p></a></li>
<li><a href='#rsse'><p>Simulation error measures.</p></a></li>
<li><a href='#saABC'>
<p>Summary statistic construction by semi-automatic ABC</p></a></li>
<li><a href='#selectsumm'>
<p>Generic function for selecting summary statistics in ABC inference.</p></a></li>
<li><a href='#semiauto.abc'><p>Performs semi-automatic ABC based on summary statistics regression.</p></a></li>
<li><a href='#stage2'><p>stage2</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Tools for ABC Analyses</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-09-18</td>
</tr>
<tr>
<td>Description:</td>
<td>Tools for approximate Bayesian computation including summary statistic selection and assessing coverage.
    See Nunes and Prangle (2015) &lt;<a href="https://doi.org/10.32614%2FRJ-2015-030">doi:10.32614/RJ-2015-030</a>&gt; and Rodrigues, Prangle and Sisson (2018) &lt;<a href="https://doi.org/10.1016%2Fj.csda.2018.04.004">doi:10.1016/j.csda.2018.04.004</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), abc, abind, parallel, plyr, Hmisc</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ggplot2, abc.data</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/dennisprangle/abctools">https://github.com/dennisprangle/abctools</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/dennisprangle/abctools/issues">https://github.com/dennisprangle/abctools/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Author:</td>
<td>Matt Nunes [aut, cre],
  Dennis Prangle [aut],
  Guilherme Rodrigues [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Matt Nunes &lt;nunesrpackages@gmail.com&gt;</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-18 10:17:20 UTC; matt</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-18 10:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='abctools-package'>Tools for ABC analyses</h2><span id='topic+abctools-package'></span><span id='topic+abctools'></span>

<h3>Description</h3>

<p>Tools for approximate Bayesian computation including summary statistic selection and assessing coverage.</p>


<h3>Author(s)</h3>

<p>Matt Nunes and Dennis Prangle
</p>
<p>Maintainer: Matthew Nunes &lt;nunesrpackages@gmail.com&gt;
</p>


<h3>References</h3>

<p>This package is described in:
</p>
<p>Nunes, M. A. and Prangle, D. (2016) abctools: an R package for tuning
approximate Bayesian computation analyses. <em>The R Journal</em>
<b>7</b>, Issue 2, 189&ndash;205.
</p>
<p>For details on methods for summary statistics selection, see:
</p>
<p>Blum, M. G. B, Nunes, M. A., Prangle, D. and Sisson, S. A. (2013) A
comparative review of dimension reduction methods in approximate
Bayesian computation. <em>Stat. Sci.</em> <b>28</b>, Issue 2, 189&ndash;208.<br /><br />
Fearnhead, P. and Prangle, D. (2012) Constructing summary statistics for approximate Bayesian computation: semi-automatic approximate Bayesian computation. <em>J. R. Stat. Soc. B</em> <b>74</b>, Part 3, 1&ndash;28.<br /><br />
Joyce, P. and P. Marjoram (2008) Approximately sufficient statistics and Bayesian computation. <em>Stat. Appl. Gen. Mol. Biol.</em> <b>7</b> Article 26.<br /><br />
Nunes, M. A. and Balding, D. J. (2010) On Optimal Selection of Summary Statistics for Approximate Bayesian Computation.  <em>Stat. Appl. Gen. Mol. Biol.</em> <b>9</b>, Iss. 1, Art. 34. <br /><br />
</p>
<p>Details of ABC coverage methods appear in:
</p>
<p>Prangle D., Blum M. G. B., Popovic G., Sisson S. A. (2014) Diagnostic
tools of approximate Bayesian computation using the coverage
property. <em>Australian and New Zealand Journal of Statistics</em>
<b>56</b>, Issue 4, 309&ndash;329.
</p>


<h3>See Also</h3>

<p><code><a href="abc.html#topic+abc">abc</a></code>
</p>

<hr>
<h2 id='AS.select'>
Summary statistics selection using approximate sufficiency.
</h2><span id='topic+AS.select'></span>

<h3>Description</h3>

<p>This function uses approximate sufficiency to assess subsets of summary statistics for ABC inference.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AS.select(obs, param, sumstats, obspar=NULL, abcmethod=abc, 
grid=10, inturn=TRUE, limit=ncol(sumstats), allow.none=FALSE, 
do.err=FALSE, final.dens=FALSE, errfn=rsse, trace=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AS.select_+3A_obs">obs</code></td>
<td>
<p>(matrix of) observed summary statistics.</p>
</td></tr>
<tr><td><code id="AS.select_+3A_param">param</code></td>
<td>
<p>matrix of simulated model parameter values.</p>
</td></tr>
<tr><td><code id="AS.select_+3A_sumstats">sumstats</code></td>
<td>
<p>matrix of simulated summary statistics.</p>
</td></tr>
<tr><td><code id="AS.select_+3A_obspar">obspar</code></td>
<td>
<p>optional observed parameters (for use to assess simulation performance).</p>
</td></tr>
<tr><td><code id="AS.select_+3A_abcmethod">abcmethod</code></td>
<td>
<p>a function to perform ABC inference, e.g. the <code>abc</code> function from package <em>abc</em>.</p>
</td></tr>
<tr><td><code id="AS.select_+3A_grid">grid</code></td>
<td>
<p>the number of bins into which to divide the posterior sample for the approximate sufficiency calculation.</p>
</td></tr>
<tr><td><code id="AS.select_+3A_inturn">inturn</code></td>
<td>
<p>a boolean value indicating whether &quot;bad&quot; statistics should be dropped and tested sequentially (<code>inturn=TRUE</code>) or all at the end (<code>inturn=FALSE</code>).</p>
</td></tr>
<tr><td><code id="AS.select_+3A_limit">limit</code></td>
<td>
<p>an optional integer indicating whether to limit summary selection to subsets of a maximum size.</p>
</td></tr>
<tr><td><code id="AS.select_+3A_allow.none">allow.none</code></td>
<td>
<p>a boolean values indicating whether an empty subset of statistics is considered in the selection procedure.</p>
</td></tr>
<tr><td><code id="AS.select_+3A_do.err">do.err</code></td>
<td>
<p>a boolean value indicating whether the simulation error should be returned.  Note: if <code>do.err=TRUE</code>, <code>obspar</code> must be supplied.</p>
</td></tr>

<tr><td><code id="AS.select_+3A_final.dens">final.dens</code></td>
<td>
<p>a boolean value indicating whether the posterior sample should be returned.</p>
</td></tr>
<tr><td><code id="AS.select_+3A_errfn">errfn</code></td>
<td>
<p>an error function to assess ABC inference performance.</p>
</td></tr>
<tr><td><code id="AS.select_+3A_trace">trace</code></td>
<td>
<p>whether to show progress messages.</p>
</td></tr>
<tr><td><code id="AS.select_+3A_...">...</code></td>
<td>
<p>any other optional arguments to the ABC inference procedure (e.g. arguments to the <code>abc</code> function).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The summary selection procedure works by sequentially testing randomly chosen statistics for inclusion, using the ratio of ABC posterior samples to determine whether a statistic is added.  Since adding a statistic may result in a suboptimal subset of summaries, the included statistics are then individually dropped and retested, to determine whether a smaller subset of statistics is equally / more informative than the accepted set of statistics.</p>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>best</code></td>
<td>
<p>the final subset of included statistics.</p>
</td></tr>
<tr><td><code>err</code></td>
<td>
<p>simulation error (if <code>obspar</code> is supplied and <code>do.err=TRUE</code>).</p>
</td></tr>
<tr><td><code>post.sample</code></td>
<td>
<p>an array of dimension <code>nacc x npar x ndatasets</code> giving the posterior sample for each observed dataset.  Not returned if <code>final.dens=FALSE</code>.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The approximate sufficiency techniques described here are only suitable for single parameters only.</p>


<h3>Author(s)</h3>

<p>Matt Nunes</p>


<h3>References</h3>

<p>Blum, M. G. B, Nunes, M. A., Prangle, D. and Sisson, S. A. (2013) A
comparative review of dimension reduction methods in approximate
Bayesian computation. <em>Stat. Sci.</em> <b>28</b>, Issue 2, 189&ndash;208.<br /><br />
Joyce, P. and P. Marjoram (2008) Approximately sufficient statistics and
Bayesian computation. <em>Stat. Appl. Gen. Mol. Biol.</em> <b>7</b>
Article 26.<br /><br />
Nunes, M. A. and Prangle, D. (2016) abctools: an R package for tuning
approximate Bayesian computation analyses. <em>The R Journal</em>
<b>7</b>, Issue 2, 189&ndash;205.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+AS.test">AS.test</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# load example data:

data(coal)
data(coalobs)

param&lt;-coal[,2]
simstats&lt;-coal[,4:6]

# use matrix below just in case to preserve dimensions.

obsstats&lt;-matrix(coalobs[1,4:6],nrow=1)

# example of AS.select:

## Not run: 
tmp &lt;-AS.select(obsstats, param, simstats, tol=.1, method="neuralnet",
nument=5, allow.none=FALSE, inturn=TRUE)

tmp$best

## End(Not run)

</code></pre>

<hr>
<h2 id='AS.test'>
Test for relative approximate sufficiency between two posterior samples.
</h2><span id='topic+AS.test'></span>

<h3>Description</h3>

<p>The function tests to determine adding a (set of) statistics is informative in ABC inference.</p>


<h3>Usage</h3>

<pre><code class='language-R'>AS.test(grid = 10, x1, x2, supp=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AS.test_+3A_grid">grid</code></td>
<td>
<p>the number of bins into which to divide the posterior sample for the approximate sufficiency calculation.</p>
</td></tr>
<tr><td><code id="AS.test_+3A_x1">x1</code></td>
<td>
<p>the posterior sample using the first set of summary statistics.</p>
</td></tr>
<tr><td><code id="AS.test_+3A_x2">x2</code></td>
<td>
<p>the posterior sample using the second (alternative) set of summary statistics.</p>
</td></tr>

<tr><td><code id="AS.test_+3A_supp">supp</code></td>
<td>
<p>the &quot;support&quot; of the prior (e.g. uniform bounds).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>After dividing each posterior sample into a number of bins (specified by <code>grid</code>), the function computes the ratio of the posterior densities.  This is seen as a measure of information added (sufficiency) by using the alternative posterior sample instead of the first posterior sample.  If the ratio exceeds a particular threshold (a number of standard deviations away from the expected counts in each bin), then the alternative set of summaries is seen as being more informative.</p>


<h3>Value</h3>

<table>
<tr><td><code>extreme</code></td>
<td>
<p>a boolean value indicating whether the alternative posterior sample is more informative than the first (i.e. the extra summary statistics add information.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Matt Nunes</p>


<h3>References</h3>

<p>Blum, M. G. B, Nunes, M. A., Prangle, D. and Sisson, S. A. (2013) A
comparative review of dimension reduction methods in approximate
Bayesian computation. <em>Stat. Sci.</em> <b>28</b>, Issue 2, 189&ndash;208.<br /><br />
Joyce, P. and P. Marjoram (2008) Approximately sufficient statistics and
Bayesian computation. <em>Stat. Appl. Gen. Mol. Biol.</em> <b>7</b>
Article 26.<br /><br />
Nunes, M. A. and Prangle, D. (2016) abctools: an R package for tuning
approximate Bayesian computation analyses. <em>The R Journal</em>
<b>7</b>, Issue 2, 189&ndash;205.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+AS.select">AS.select</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
#create two fake posterior samples:

x1&lt;-runif(10000)
x2&lt;-rnorm(10000)

AS.test(x1=x1,x2=x2,supp=range(x2))

</code></pre>

<hr>
<h2 id='coal'>Examples of coalescent data</h2><span id='topic+coal'></span><span id='topic+coalobs'></span>

<h3>Description</h3>

<p>Data generated from a coalescent model for genetic variation.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(coal)</code></pre>


<h3>Format</h3>

<p>Matrices of parameters (first 2 columns) and summary statistics
(remaining columns) from a coalescent model.</p>


<h3>Details</h3>

<p>&quot;coal&quot; contains 100,000 simulated datasets to use for an ABC
analysis. &quot;coalobs&quot; contains 100 which can each be used as observed
datasets. This allows methods to be tested on many different
observations.
</p>
<p>The parameters are the scaled mutation rate (theta) and scaled
recombination rate (rho).
</p>
<p>The summary statistics are the number of segregating sites (segsites),
independent Uniform[0,25] random noise (unif),
the pairwise mean number of nucleotidic differences (meandiff),
the mean R^2 across pairs separated by &lt; 10% of the simulated genomic regions (R2),
the number of distinct haplotypes (nhap),
the frequency of the most common haplotype (fhap),
and the number of singleton haplotypes (shap).
</p>
<p>To simulate a dataset 5,001 basepair DNA sequences for 50 individuals
are generated from the coalescent model, with recombination, under the
infinite-sites mutation model, using the software ms (Hudson 2002)
</p>
<p>Data of this form has been analysed in several ABC papers.  See the
references for more details.
</p>


<h3>References</h3>

<p>Blum, M. G. B., M. A. Nunes, D. Prangle and S. A. Sisson 'A comparative
review of dimension reduction methods in approximate Bayesian
computation' Statistical Applications in Genetics and
Molecular Biology 13 (2014)
</p>
<p>Hudson, R. R. 'Generating samples under a Wright-Fisher neutral
model of genetic variation' Bioinformatics 18 (2002)
</p>
<p>Joyce, P. and P. Marjoram 'Approximately sufficient statistics
and Bayesian computation' Statistical Applications in Genetics and
Molecular Biology 7 (2008)
</p>
<p>Nunes, M. A. and D. J. Balding 'On optimal selection of summary
statistics for approximate Bayesian computation' Statistical
Applications in Genetics and Molecular Biology 9 (2010)
</p>
<p>Nunes, M. A. and Prangle, D. abctools: an R package for tuning
approximate Bayesian computation analyses. The R Journal 7 (2016)
</p>

<hr>
<h2 id='combmat'>table of combinations</h2><span id='topic+combmat'></span><span id='topic+fillcomb'></span><span id='topic+combtable'></span>

<h3>Description</h3>

<p>This function creates a table of binary masks representing combinations of statistics.</p>


<h3>Usage</h3>

<pre><code class='language-R'>combmat(n,limit = NULL)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combmat_+3A_n">n</code></td>
<td>
<p>number of statistics</p>
</td></tr>

<tr><td><code id="combmat_+3A_limit">limit</code></td>
<td>
<p>an optional (integer) value indicating whether to limit the table to subsets up to a certain size.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>m</code></td>
<td>
<p>The matrix of binary masks.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Matt Nunes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# Find all binary masks of a set of statistics {C1,C2,C3,C4},
# listing all singlets, pairs, triples and then the whole set:

combmat(4,TRUE)

</code></pre>

<hr>
<h2 id='cov.pi'>
Coverage property diagnostics
</h2><span id='topic+cov.pi'></span><span id='topic+cov.mc'></span><span id='topic+covstats.pi'></span><span id='topic+covstats.mc'></span>

<h3>Description</h3>

<p>These functions produce diagnostic statistics for an ABC analysis to
judge when the tolerance level is small enough to produce roughly
no approximation error.  This is done by running analyses for many
pseudo-observed test data sets and assessing whether the results satisfy the &quot;coverage
property&quot; (roughly speaking: credible intervals have the claimed
coverage levels).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cov.pi(param, sumstat, testsets, tol, eps, diagnostics = c(), 
multicore = FALSE, cores, method = "rejection", nacc.min=20, ...)

cov.mc(index, sumstat, testsets, tol, eps, diagnostics = c(), 
multicore = FALSE, cores, method = "rejection", nacc.min=20, ...)

covstats.pi(raw, diagnostics = c("KS", "CGR"), nacc.min = 20)

covstats.mc(raw, index, diagnostics = c("freq", "loglik.binary", 
"loglik.multi"), nacc.min = 20) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cov.pi_+3A_param">param</code></td>
<td>

<p>A data frame of parameter values.  It must have the same number of
rows as <code>sumstat</code> and contain numeric values only.
</p>
</td></tr>
<tr><td><code id="cov.pi_+3A_index">index</code></td>
<td>

<p>A vector of model indices. Any value which can be converted to
factor is ok (e.g. character or numeric entries).  It must have the
same length as <code>nrow(sumstat)</code>.
</p>
</td></tr>
<tr><td><code id="cov.pi_+3A_sumstat">sumstat</code></td>
<td>

<p>A data frame of summary statistic values whose the ith row has been
simulated using <code>param[i,]</code> or <code>index[i]</code>.
</p>
</td></tr>
<tr><td><code id="cov.pi_+3A_testsets">testsets</code></td>
<td>

<p>A numeric vector giving the rows of <code>sumstat</code> to be used as
pseudo-observed data to test the coverage property.
</p>
</td></tr>
<tr><td><code id="cov.pi_+3A_tol">tol</code></td>
<td>

<p>A vector of proportions of ABC acceptances which will be
investigated.
</p>
</td></tr>
<tr><td><code id="cov.pi_+3A_eps">eps</code></td>
<td>

<p>A vector of ABC thresholds which will be investigated.  These are
used when <code>tol</code> is missing.  One of <code>eps</code> and <code>tol</code>
must be supplied.
</p>
</td></tr>
<tr><td><code id="cov.pi_+3A_diagnostics">diagnostics</code></td>
<td>

<p>A character vector containing diagnostics to be calculated.
Allowable values for parameter inference are &quot;KL&quot; (Kullback-Leibler
based test) or &quot;CGR&quot; (Cook, Gelman and Rubin test). Allowable values
for model choice are &quot;freq&quot; (a separate frequency-based test for
each model), &quot;loglik.binary&quot; (a separate <br /> log-likelihood based test
for each model) or &quot;loglik.multi&quot; (single log-likelihood based test).  If
<code>diagnostics</code> is empty only raw results will be returned.
</p>
</td></tr>
<tr><td><code id="cov.pi_+3A_multicore">multicore</code></td>
<td>

<p>Whether to use the <code><a href="lattice.html#topic+parallel">parallel</a></code> package to perform
analyses of test datasets in parallel.
</p>
</td></tr>
<tr><td><code id="cov.pi_+3A_cores">cores</code></td>
<td>

<p>Number of cores to use when <code>multicore==TRUE</code>.
</p>
</td></tr>
<tr><td><code id="cov.pi_+3A_method">method</code></td>
<td>

<p>Method used for ABC analysis.  The default is &quot;rejection&quot;.  For
alternatives see <code><a href="abc.html#topic+abc">abc</a></code> (parameter inference) or
<code><a href="abc.html#topic+postpr">postpr</a></code> (model choice).
</p>
</td></tr>
<tr><td><code id="cov.pi_+3A_nacc.min">nacc.min</code></td>
<td>

<p>Minimum number of ABC acceptances required to compute diagnostics.
See Values for details of how this is used.
</p>
</td></tr>
<tr><td><code id="cov.pi_+3A_...">...</code></td>
<td>

<p>Extra arguments to be supplied to the function performing abc
analysis i.e. <code><a href="abc.html#topic+abc">abc</a></code> (parameter inference) or
<code><a href="abc.html#topic+postpr">postpr</a></code> (model choice).
</p>
</td></tr>
<tr><td><code id="cov.pi_+3A_raw">raw</code></td>
<td>

<p>Raw output component from <code>cov.pi</code> or <code>cov.mc</code> for which
diagnostics are to be calculated.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions are intended to be applied as follows (i) random
models/parameters are generated, data sets simulated for each and
summary statistics calculated (ii) these are input to <code>cov.pi</code>
(parameter inference) or <code>cov.mc</code> (model choice) which outputs raw
results and diagnostics (see below) (iii) the output can be passed to
<code>covstats.pi</code> or <code>covstats.mc</code> if further diagnostics are
required later (or to find diagnostics for a subset of the pseudo-observed
data).
</p>
<p>The <code>cov.pi</code> and <code>cov.mc</code> functions operate by performing
many ABC analyses.  The user specifies which datasets amongst those
simulated will used as pseudo-observed &quot;test&quot; data to be analysed.
The results of each analysis are compared to the known
model/parameters which produced the data to see whether they are
consistent in a particular sense (i.e. if the coverage property is
satisfied). Various diagnostics are provided to judge this easily, and
determine what happens as the ABC threshold is varied.  Raw results
are also returned which can be investigated in more detail.
</p>
<p>All ABC analyses use a rejection sampling algorithm implemented by the
<code><a href="abc.html#topic+abc">abc</a></code> package.  The user may specify regression
post-processing as part of this analysis.
</p>


<h3>Value</h3>

<p>Output of <code>cov.pi</code> or <code>cov.mc</code> is a list of two data frames,
<code>raw</code> and <code>diag</code>.  The <code>covstats.pi</code> and
<code>covstats.mc</code> functions just return the latter data frame.
</p>
<p>For parameter inference, <code>raw</code> contains estimated cdfs (referred
to as p0 estimates in Prangle et al 2013) of the true parameter values
for each input configuration (i.e. for every tol/eps value at every
test dataset).  <code>diag</code> is a data frame of tol/eps value,
parameter name, diagnostic name and p-value.  Here the p-value relates
to the test statistic used as a diagnostic.  It is NA if any analyses
had fewer than <code>nacc.min</code> acceptances (Diagnostics based on a
small number of acceptances can be misleading.)
</p>
<p>For model choice, <code>raw</code> contains estimated model weights for each input
configuration, and <code>diag</code> is a data frame of tol/eps value, model,
diagnostic name and p-value (NA under the same conditions as before.)
</p>
<p>In both cases, <code>raw</code> also reports the number of acceptances.  Note that
<code>raw</code> contains p0 estimates/weights of NA if regression correction is
requested but there are too few acceptances to compute it.
</p>


<h3>Author(s)</h3>

<p>Dennis Prangle
</p>


<h3>References</h3>

<p>Nunes, M. A. and Prangle, D. (2016) abctools: an R package for tuning
approximate Bayesian computation analyses. <em>The R Journal</em>
<b>7</b>, Issue 2, 189&ndash;205.
</p>
<p>Prangle D., Blum M. G. B., Popovic G., Sisson S. A. (2014) Diagnostic
tools of approximate Bayesian computation using the coverage
property. <em>Australian and New Zealand Journal of Statistics</em>
<b>56</b>, Issue 4, 309&ndash;329.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mc.ci">mc.ci</a></code> for a diagnostic plot of raw model choice results
</p>
<p><code><a href="abc.html#topic+abc">abc</a></code> and <code><a href="abc.html#topic+postpr">postpr</a></code> to perform ABC for a given dataset
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ##The examples below are chosen to run relatively quickly (&lt;5 mins)
  ##and do not represent recommended tuning choices.
  ## Not run: 
  data(musigma2)
  library(ggplot2)
  ##Parameter inference example
  parameters &lt;- data.frame(par.sim)
  sumstats &lt;- data.frame(stat.sim)
  covdiag &lt;- cov.pi(param=parameters, sumstat=sumstats, testsets=1:100, 
  tol=seq(0.1,1,by=0.1), diagnostics=c("KS"))

  #Plot of diagnostic results
  qplot(x=tol, y=pvalue, facets=.~parameter, data=covdiag$diag) 
  #Plot of raw results for tol=0.5
  qplot(x=mu, data=subset(covdiag$raw, tol==0.5)) 
  #Plot of raw results for tol=0.5
  qplot(x=sigma2, data=subset(covdiag$raw, tol==0.5)) 

  #Compute CGR statistic and plot
  cgrout &lt;- covstats.pi(covdiag$raw, diagnostics="CGR") 
  qplot(x=tol, y=pvalue, facets=.~parameter, data=cgrout) 

  ##Model choice example, based on simple simulated data
  index &lt;- sample(1:2, 1E4, replace=TRUE)
  sumstat &lt;- ifelse(index==1, rnorm(1E4,0,1), rnorm(1E4,0,rexp(1E4,1)))
  sumstat &lt;- data.frame(ss=sumstat)
  covdiag &lt;- cov.mc(index=index, sumstat=sumstat, testsets=1:100, 
  tol=seq(0.1,1,by=0.1), diagnostics=c("freq"))
  qplot(x=tol, y=pvalue, data=covdiag$diag)
  llout &lt;- covstats.mc(covdiag$raw, index=index, 
  diagnostics="loglik.binary")
  qplot(x=tol, y=pvalue, data=llout)
  mc.ci(covdiag$raw, tol=0.5, modname=1, modtrue=index[1:200])
  
## End(Not run)
</code></pre>

<hr>
<h2 id='mc.ci'>
Diagnostic plots for model choice coverage output
</h2><span id='topic+mc.ci'></span>

<h3>Description</h3>

<p>Plots credible interval estimates for raw model choice output from
<code><a href="#topic+cov.mc">cov.mc</a></code>.  This is used to investigate whether the coverage
property holds and validate whether diagnostic statistics are acting
as intended.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  mc.ci(raw, tol, eps, modname, modtrue, nbins=5, 
bintype=c("interval", "quantile"), bw=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mc.ci_+3A_raw">raw</code></td>
<td>

<p>The <code>raw</code> item from the list output by <code>cov.mc</code>.
</p>
</td></tr>
<tr><td><code id="mc.ci_+3A_tol">tol</code></td>
<td>

<p>The value of <code>tol</code> to test.
</p>
</td></tr>
<tr><td><code id="mc.ci_+3A_eps">eps</code></td>
<td>

<p>The value of <code>eps</code> to test.  This is used when <code>tol</code> is
missing.  One of <code>eps</code> and <code>tol</code> must be supplied.
</p>
</td></tr>
<tr><td><code id="mc.ci_+3A_modname">modname</code></td>
<td>

<p>The name of the model to test.
</p>
</td></tr>
<tr><td><code id="mc.ci_+3A_modtrue">modtrue</code></td>
<td>

<p>Vector containing the true models generating the pseudo-observed
test data.  i.e. <code>modtrue[i]</code> is the model generating dataset i.    
</p>
</td></tr>
<tr><td><code id="mc.ci_+3A_nbins">nbins</code></td>
<td>

<p>Number of bins to display.
</p>
</td></tr>
<tr><td><code id="mc.ci_+3A_bintype">bintype</code></td>
<td>

<p>How to choose the bins (see Details).
</p>
</td></tr>
<tr><td><code id="mc.ci_+3A_bw">bw</code></td>
<td>

<p>Whether to produce a black and white image.  Default is FALSE.
Colour is used to make different bins stand out.
</p>
</td></tr>
<tr><td><code id="mc.ci_+3A_...">...</code></td>
<td>

<p>Additional plotting arguments - anything that can be used by <code>plot</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function provides a plot which can be used as an informal test of
the model choice coverage hypothesis for a particular value of
<code>eps</code> or <code>tol</code> and choice of model.  The plot is more
flexible than the diagnostics, but not suitable as the basis of a
formal test.
</p>
<p>For each pseudo-observed data set, the ABC probability that the model
is <code>modname</code> is taken from <code>raw</code>, and the true model is
taken from <code>modtrue</code>.  The probabilities are binned into
<code>nbins</code> intervals, either of equal length or based on <code>nbins+1</code>
equally spaced empirical quantiles.  The function estimates the
observed probability of <code>modname</code> within each bin using Bayesian
inference for a binomial proportion under a uniform prior.  The plot
shows the mean and 95% credible interval plotted against predicted
probabilities.  Informally, the coverage property should be rejected
if predicted values are too unlikely given the observed values.
</p>


<h3>Author(s)</h3>

<p>Dennis Prangle
</p>


<h3>References</h3>

<p>Nunes, M. A. and Prangle, D. (2016) abctools: an R package for tuning
approximate Bayesian computation analyses. <em>The R Journal</em>
<b>7</b>, Issue 2, 189&ndash;205.
</p>
<p>Prangle D., Blum M. G. B., Popovic G., Sisson S. A. (2014) Diagnostic
tools of approximate Bayesian computation using the coverage
property. <em>Australian and New Zealand Journal of Statistics</em>
<b>56</b>, Issue 4, 309&ndash;329.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cov.mc">cov.mc</a></code> to produce the input for this function
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ##The examples below are chosen to run relatively quickly (&lt;5 mins)
  ##and do not represent recommended tuning choices.
  ## Not run: 
  index &lt;- sample(1:2, 1E4, replace=TRUE)
  sumstat &lt;- ifelse(index==1, rnorm(1E4,0,1), rnorm(1E4,0,rexp(1E4,1)))
  sumstat &lt;- data.frame(ss=sumstat)
  covdiag &lt;- cov.mc(index=index, sumstat=sumstat, testsets=1:100, 
  tol=seq(0.1,1,by=0.1), diagnostics=c("freq"))
  mc.ci(covdiag$raw, tol=0.5, modname=1, modtrue=index[1:100])
  
## End(Not run)
</code></pre>

<hr>
<h2 id='mincrit'>
Summary statistics selection by minimizing a posterior sample measure. 
</h2><span id='topic+mincrit'></span>

<h3>Description</h3>

<p>The function cycles through all possible subsets of summary statistics and computes a criterion from the posterior sample.  The subset which achieves the minimum is chosen as the most informative subset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mincrit(obs, param, sumstats, obspar = NULL, abcmethod = abc,
crit = nn.ent, sumsubs = 1:ncol(sumstats), limit=length(sumsubs), 
do.only = NULL, verbose = TRUE, do.crit = TRUE, do.err = FALSE, 
final.dens = FALSE, errfn = rsse, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mincrit_+3A_obs">obs</code></td>
<td>
<p>(matrix of) observed summary statistics.</p>
</td></tr>
<tr><td><code id="mincrit_+3A_param">param</code></td>
<td>
<p>matrix of simulated model parameter values.</p>
</td></tr>
<tr><td><code id="mincrit_+3A_sumstats">sumstats</code></td>
<td>
<p>matrix of simulated summary statistics.</p>
</td></tr>
<tr><td><code id="mincrit_+3A_obspar">obspar</code></td>
<td>
<p>optional observed parameters (for use to assess simulation performance).</p>
</td></tr>
<tr><td><code id="mincrit_+3A_abcmethod">abcmethod</code></td>
<td>
<p>a function to perform ABC inference, e.g. the <code>abc</code> function from package <em>abc</em>.</p>
</td></tr>
<tr><td><code id="mincrit_+3A_crit">crit</code></td>
<td>
<p>a function to minimize to measure information from a posterior sample, e.g. <code>nn.ent</code>.</p>
</td></tr>
<tr><td><code id="mincrit_+3A_sumsubs">sumsubs</code></td>
<td>
<p>an optional index into the summary statistics to limit summary selection to a specific subset of summaries.</p>
</td></tr>
<tr><td><code id="mincrit_+3A_limit">limit</code></td>
<td>
<p>an optional integer indicating whether to limit summary selection to subsets of a maximum size.</p>
</td></tr>
<tr><td><code id="mincrit_+3A_do.only">do.only</code></td>
<td>
<p>an optional index into the summary statistics combination table.  Can be used to limit entropy calculations to certain summary statistics subsets only.</p>
</td></tr>
<tr><td><code id="mincrit_+3A_verbose">verbose</code></td>
<td>
<p>a boolean value indicating whether informative statements should be printed to screen.</p>
</td></tr>

<tr><td><code id="mincrit_+3A_do.crit">do.crit</code></td>
<td>
<p>a boolean value indicating whether the measure on the posterior sample should be returned.</p>
</td></tr>
<tr><td><code id="mincrit_+3A_do.err">do.err</code></td>
<td>
<p>a boolean value indicating whether the simulation error should be returned.  Note: if <code>do.err=TRUE</code>, <code>obspar</code> must be supplied.</p>
</td></tr>
<tr><td><code id="mincrit_+3A_final.dens">final.dens</code></td>
<td>
<p>a boolean value indicating whether the posterior sample should be returned.</p>
</td></tr>
<tr><td><code id="mincrit_+3A_errfn">errfn</code></td>
<td>
<p>an error function to assess ABC inference performance.</p>
</td></tr>
<tr><td><code id="mincrit_+3A_...">...</code></td>
<td>
<p>any other optional arguments to the ABC inference procedure (e.g. arguments to the <code>abc</code> function).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function uses a criterion (e.g.sample entropy) as a proxy for information in a posterior sample.  The criterion for each possible subset of statistics is computed, and the best subset is judged as the one which minimises this vector of values.</p>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>best</code></td>
<td>
<p>the best subset(s) of statistics.</p>
</td></tr>
<tr><td><code>critvals</code></td>
<td>
<p>the calculated criterion values (if <code>do.crit=TRUE</code>).</p>
</td></tr>
<tr><td><code>err</code></td>
<td>
<p>simulation error (if <code>obspar</code> is supplied and <code>do.err=TRUE</code>).</p>
</td></tr>
<tr><td><code>order</code></td>
<td>
<p>the subsets considered during the algorithm (same as the input <code>do.only</code>.</p>
</td></tr>
<tr><td><code>post.sample</code></td>
<td>
<p>an array of dimension <code>nacc x npar x ndatasets</code> giving the posterior sample for each observed dataset.  Not returned if <code>final.dens=FALSE</code>.</p>
</td></tr>
<tr><td><code>sumsubs</code></td>
<td>
<p>an index into the subsets considered during the algorithm.</p>
</td></tr>
</table>


<h3>Warning</h3>

<p>These functions are computationally intensive due to the cyclic ABC inference procedure.</p>


<h3>Author(s)</h3>

<p>Matt Nunes</p>


<h3>References</h3>

<p>Nunes, M. A. and Balding, D. J. (2010) On Optimal Selection of Summary
Statistics for Approximate Bayesian Computation.
<em>Stat. Appl. Gen. Mol. Biol.</em> <b>9</b>, Iss. 1, Art. 34.
</p>
<p>Nunes, M. A. and Prangle, D. (2016) abctools: an R package for tuning
approximate Bayesian computation analyses. <em>The R Journal</em>
<b>7</b>, Issue 2, 189&ndash;205.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+nn.ent">nn.ent</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# load example data:

data(coal)
data(coalobs)

param&lt;-coal[1:50000,2]
simstats&lt;-coal[1:50000,4:6]

# use matrix below just in case to preserve dimensions.

obsstats&lt;-matrix(coalobs[1,4:6],nrow=1)
obsparam&lt;-matrix(coalobs[1,1])

# example of entropy minimization algorithm:

tmp &lt;-mincrit(obsstats, param, simstats, tol=.01, method="rejection", 
do.crit=TRUE)

tmp$critvals

</code></pre>

<hr>
<h2 id='nn.ent'>Works out entropy of a sample.</h2><span id='topic+nn.ent'></span>

<h3>Description</h3>

<p>The function computes the k nearest neighbour sample entropy.</p>


<h3>Usage</h3>

<pre><code class='language-R'>nn.ent(th, k=4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nn.ent_+3A_th">th</code></td>
<td>
<p>The sample from which to compute the entropy.</p>
</td></tr>
<tr><td><code id="nn.ent_+3A_k">k</code></td>
<td>
<p>The order (number of neighbours) of the sample entropy calculation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The sample entropy gives a measure of information in a (posterior) sample, or lack of it.</p>


<h3>Value</h3>

<p> The k nearest neighbour entropy from the sample.</p>


<h3>Warning</h3>

<p>For high-dimensional posterior samples, the <code>nn.ent</code> calculation is quite computationally intensive.
</p>


<h3>Author(s)</h3>

<p>Matt Nunes</p>


<h3>References</h3>

<p>Nunes, M. A. and Prangle, D. (2016) abctools: an R package for tuning
approximate Bayesian computation analyses. <em>The R Journal</em>
<b>7</b>, Issue 2, 189&ndash;205.<br /><br />  
Singh, H. et al. (2003) Nearest neighbor estimates of entropy. <em>Am. J. Math. Man. Sci.</em>,<b>23</b>, 301&ndash;321.<br /><br />
Shannon, C. E. and Weaver, W. (1948) A mathematical theory of communication. <em>Bell Syst. Tech. J.</em>, <b>27</b>, 379&ndash;423.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mincrit">mincrit</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# create a dummy sample to calculate an entropy measure:

theta&lt;-rnorm(10000)

nn.ent(theta)

</code></pre>

<hr>
<h2 id='recalibrationABC'>
ABC inference with a recalibration adjustment
</h2><span id='topic+recalibrationABC'></span>

<h3>Description</h3>

<p>This function post-processes ABC output with the aim of calibrating
its credible intervals to have the correct probabilities. The approach
can be thought of as extending coverage tests to correct deviations
from the desired posterior. See the reference for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  recalibrationABC(target, param, sumstat, eps, tol, method="rejection",
  multicore=FALSE, cores=NULL, abc.p.options=list(method="loclinear"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="recalibrationABC_+3A_target">target</code></td>
<td>

<p>A vector of the observed summary statistics.
</p>
</td></tr>
<tr><td><code id="recalibrationABC_+3A_param">param</code></td>
<td>

<p>A data frame of the simulated parameter values.
</p>
</td></tr>
<tr><td><code id="recalibrationABC_+3A_sumstat">sumstat</code></td>
<td>

<p>A data frame of the simulated summary statistics.
</p>
</td></tr>
<tr><td><code id="recalibrationABC_+3A_eps">eps</code></td>
<td>

<p>The ABC acceptance threshold i.e. max acceptable distance. This or
<code>tol</code> must be specified (but not both).
</p>
</td></tr>
<tr><td><code id="recalibrationABC_+3A_tol">tol</code></td>
<td>

<p>The ABC acceptance tolerance i.e. proportion of simulations to
accept. This or <code>eps</code> must be specified (but not both).    
</p>
</td></tr>
<tr><td><code id="recalibrationABC_+3A_method">method</code></td>
<td>

<p>A character string indicating the type of ABC algorithm to be
applied. Possible values are &quot;rejection&quot;, &quot;loclinear&quot;,
&quot;neuralnet&quot; and &quot;ridge&quot;.
</p>
</td></tr>
<tr><td><code id="recalibrationABC_+3A_multicore">multicore</code></td>
<td>

<p>Whether to use the <code><a href="lattice.html#topic+parallel">parallel</a></code> package to perform
analyses of test datasets in parallel.
</p>
</td></tr>
<tr><td><code id="recalibrationABC_+3A_cores">cores</code></td>
<td>

<p>Number of cores to use when <code>multicore==TRUE</code>.
</p>
</td></tr>
<tr><td><code id="recalibrationABC_+3A_abc.p.options">abc.p.options</code></td>
<td>

<p>A list of further arguments to be supplied to the <code>abc</code> command
when applying regression correction to the p-values based on the
summary statistics.
</p>
</td></tr>
<tr><td><code id="recalibrationABC_+3A_...">...</code></td>
<td>

<p>Further arguments to be supplied to the <code>abc</code> command when
performing inference for each simulated data set. Typically these will
control regression correction. If omitted no correction is performed.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components is returned.
</p>
<p><code>sample.abc</code> is the ordinary ABC output sample (with any
regression correction requested). The rows represent accepted samples
and the columns represent the parameters.
</p>
<p><code>sample.recal</code> is the ABC output sample following
recalibration. The rows are Monte Carlo draws from an approximation to
the posterior and the columns represent the parameters.
</p>
<p><code>sample.regrecal</code> is the ABC output sample following coverage
correction including regression correction of p-values. It has a
similar form to <code>sample.recal</code>.
</p>
<p><code>weights</code> are weights for the ABC output. These apply to all
types of ABC.
</p>
<p>Each row of <code>pvalues</code> corresponds to a particular accepted
dataset. It gives the p-values of the true parameters within the ABC
sample generated from this data.
</p>
<p>Each row of <code>pvalues.reg</code> corresponds to <code>pvalues</code> after
a regression-adjustment has been performed on them.
</p>
<p><code>svalues</code> is a subset of the rows of <code>sumstat</code> corresponding
to accepted datasets. These can be used in conjunction with
<code>pvalues</code> to perform a recalibration correction manually.
</p>


<h3>Author(s)</h3>

<p>Dennis Prangle and Guilherme Rodrigues
</p>


<h3>References</h3>

<p>G. S. Rodrigues, D. Prangle and S. A. Sisson (2017)
Recalibration: A post-processing method for approximate Bayesian
computation. <em>In submission</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## Not run: 
  data(musigma2)
  P &lt;- data.frame(par.sim)
  S &lt;- data.frame(stat.sim)
  out &lt;- recalibrationABC(target=stat.obs, param=P, sumstat=S, tol=0.3)
  plot(rbind(out$sample.plain[1:500,], out$sample.recal[1:500,]),
  col=c(rep("red",500), rep("blue", 500)))
  ##Red shows plain ABC sample, blue shows recalibrated output

## End(Not run)
</code></pre>

<hr>
<h2 id='rsse'>Simulation error measures.</h2><span id='topic+rsse'></span><span id='topic+sse'></span>

<h3>Description</h3>

<p>Computes error measures between two sets of data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rsse(a, b, v = 1)
sse(a, b, v = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rsse_+3A_a">a</code></td>
<td>
<p>Dataset 1, of dimension (n x p)</p>
</td></tr>
<tr><td><code id="rsse_+3A_b">b</code></td>
<td>
<p>Dataset 2: of dimension (1 x p) or a vector.</p>
</td></tr>
<tr><td><code id="rsse_+3A_v">v</code></td>
<td>
<p>an optional factor to normalise the data before computation of the RSSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The RSSE between dataset <code>a</code> and <code>b</code>.</p>


<h3>Author(s)</h3>

<p>Matt Nunes</p>


<h3>Examples</h3>

<pre><code class='language-R'>
a&lt;-matrix(rnorm(1000),ncol=2)
b&lt;-runif(2)

rsse(a,b)

</code></pre>

<hr>
<h2 id='saABC'>
Summary statistic construction by semi-automatic ABC
</h2><span id='topic+saABC'></span>

<h3>Description</h3>

<p><code>saABC</code> fits parameter estimators based on simulated data to be
used as summary statistics within ABC.  Fitting is by linear
regression.  Some simple diagnostics are provided for assistance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>saABC(theta, X, plot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="saABC_+3A_theta">theta</code></td>
<td>

<p>A n x d matrix or data frame of simulated parameter values.
<code>theta[i,j]</code> is the ith simulated value of parameter j.
</p>
</td></tr>
<tr><td><code id="saABC_+3A_x">X</code></td>
<td>

<p>A n x p matrix or data frame of simulated data and/or associated
transformations.  <code>X[i,]</code> is a vector of the data for parameter
values <code>theta[i,]</code>.  A constant term should not be included.
</p>
</td></tr>
<tr><td><code id="saABC_+3A_plot">plot</code></td>
<td>

<p>When <code>plot==TRUE</code>, a plot of parameter values against fitted
values is produced for each parameter as a side-effect.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The semi-automatic ABC method of Fearnhead and Prangle (2012) is as
follows:
</p>
<p>1) Simulate parameter vectors <code class="reqn">\theta_i</code> and corresponding data sets <code class="reqn">x_i</code> for i=1,2,...,N.
</p>
<p>2) Use the simulations to fit an estimator of each parameter as a
linear combination of f(x), where f(x) is a vector of
transformations of x (including a constant term).
</p>
<p>3) Run ABC using these simulations.
</p>
<p>The <code>saABC</code> function automates step 2 of this process.  The user
must supply simulated parameter values <code>theta</code> and corresponding
f(x) values <code>x</code> (n.b. excluding the constant term).  The function
returns weights for the linear combinations which can easily be used
for step 3.  In particular, fitted weights are returned as a matrix
of weights for the columns of <code>x</code> and a vector of constants.  The
vector can usually be discarded, as it is not needed to find
differences between summary statistics.
</p>
<p>The function also returns BIC values for each parameter so that the
user can judge the quality of the fits, and compare different choices
of f(x).  Diagnostic plots of supplied parameter values against fitted
values are also optionally provided.  These are useful for exploratory
purposes when there are a small number of parameters, but provide less
protection from overfitting than BIC values.
</p>


<h3>Value</h3>

<table>
<tr><td><code>B0</code></td>
<td>
<p>Vector of constant terms from fitted regressions.</p>
</td></tr>
<tr><td><code>B</code></td>
<td>
<p>Matrix of weights from fitted regressions.</p>
</td></tr>
<tr><td><code>BICs</code></td>
<td>
<p>Vector of BIC values for each fitted regression.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dennis Prangle
</p>


<h3>References</h3>

<p>Blum, M. G. B, Nunes, M. A., Prangle, D. and Sisson, S. A. (2013) A
comparative review of dimension reduction methods in approximate
Bayesian computation. <em>Stat. Sci.</em> <b>28</b>, Issue 2, 189&ndash;208.<br /><br />
Fearnhead, P. and Prangle, D. (2012) Constructing summary statistics for approximate Bayesian computation: 
semi-automatic approximate Bayesian
computation. <em>J. R. Stat. Soc. B</em> <b>74</b>, Part 3, 1&ndash;28.<br /><br />
Nunes, M. A. and Prangle, D. (2016) abctools: an R package for tuning
approximate Bayesian computation analyses. <em>The R Journal</em>
<b>7</b>, Issue 2, 189&ndash;205.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
theta &lt;- matrix(runif(2E3),ncol=2)
colnames(theta) &lt;- c("Mean", "Variance")

X &lt;- replicate(5, rnorm(1E3, theta[,1], theta[,2]))

saABC(theta, X)$BICs
saABC(theta, cbind(X, X^2))$BICs ##Variance parameter estimated better 
</code></pre>

<hr>
<h2 id='selectsumm'>
Generic function for selecting summary statistics in ABC inference.
</h2><span id='topic+selectsumm'></span>

<h3>Description</h3>

<p>The function implements functions which implement summary statistics selection methods.</p>


<h3>Usage</h3>

<pre><code class='language-R'>selectsumm(obs, param, sumstats, obspar=NULL, ssmethod = mincrit, 
verbose = TRUE, final.dens = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="selectsumm_+3A_obs">obs</code></td>
<td>
<p>(matrix of) observed summary statistics.</p>
</td></tr>
<tr><td><code id="selectsumm_+3A_param">param</code></td>
<td>
<p>matrix of simulated model parameter values.</p>
</td></tr>
<tr><td><code id="selectsumm_+3A_sumstats">sumstats</code></td>
<td>
<p>matrix of simulated summary statistics.</p>
</td></tr>
<tr><td><code id="selectsumm_+3A_obspar">obspar</code></td>
<td>
<p>optional observed parameters (for use to assess simulation performance).</p>
</td></tr>
<tr><td><code id="selectsumm_+3A_ssmethod">ssmethod</code></td>
<td>
<p>a function to perform summary statistics selection.  Current methods are <br />
<code>AS.select</code> and <code>mincrit</code>.</p>
</td></tr>
<tr><td><code id="selectsumm_+3A_verbose">verbose</code></td>
<td>
<p>a boolean value indicating whether informative statements should be printed to screen.</p>
</td></tr>
<tr><td><code id="selectsumm_+3A_final.dens">final.dens</code></td>
<td>
<p>a boolean value indicating whether the posterior sample should be returned.</p>
</td></tr>
<tr><td><code id="selectsumm_+3A_...">...</code></td>
<td>
<p>any other optional arguments to the summary selection procedure.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is essentially a wrapper for more specific summary selection methods, and is designed to be flexible for future additions and minimization criteria.  See the help files for each summary selection method for more details.
</p>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>best</code></td>
<td>
<p>the best subset(s) of statistics.</p>
</td></tr>
<tr><td><code>critvals</code></td>
<td>
<p>the calculated criterion values (if <code>do.crit=TRUE</code>).</p>
</td></tr>
<tr><td><code>err</code></td>
<td>
<p>simulation error (if <code>obspar</code> is supplied and <code>do.err=TRUE</code>).</p>
</td></tr>
<tr><td><code>order</code></td>
<td>
<p>the subsets considered during the algorithm (same as the input <code>do.only</code>.</p>
</td></tr>
<tr><td><code>post.sample</code></td>
<td>
<p>an array of dimension <code>nacc x npar x ndatasets</code> giving the posterior sample for each observed dataset.  Not returned if <code>final.dens=FALSE</code>.</p>
</td></tr>
<tr><td><code>sumsubs</code></td>
<td>
<p>an index into the subsets considered during the algorithm.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Matt Nunes</p>


<h3>References</h3>

<p>Blum, M. G. B, Nunes, M. A., Prangle, D. and Sisson, S. A. (2013) A
comparative review of dimension reduction methods in approximate
Bayesian computation. <em>Stat. Sci.</em> <b>28</b>, Issue 2, 189&ndash;208.<br /><br />
Fearnhead, P. and Prangle, D. (2012) Constructing summary statistics for approximate Bayesian computation: semi-automatic approximate Bayesian computation. <em>J. R. Stat. Soc. B</em> <b>74</b>, Part 3, 1&ndash;28.<br /><br />
Joyce, P. and P. Marjoram (2008) Approximately sufficient statistics and Bayesian computation. <em>Stat. Appl. Gen. Mol. Biol.</em> <b>7</b> Article 26.<br /><br />
Nunes, M. A. and Balding, D. J. (2010) On Optimal Selection of Summary
Statistics for Approximate Bayesian Computation.
<em>Stat. Appl. Gen. Mol. Biol.</em> <b>9</b>, Iss. 1, Art. 34. <br /><br />
Nunes, M. A. and Prangle, D. (2016) abctools: an R package for tuning
approximate Bayesian computation analyses. <em>The R Journal</em>
<b>7</b>, Issue 2, 189&ndash;205.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mincrit">mincrit</a></code>, <code><a href="#topic+AS.select">AS.select</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# load example data:

data(coal)
data(coalobs)

param&lt;-coal[,2]
simstats&lt;-coal[,4:6]

# use matrix below just in case to preserve dimensions.

obsstats&lt;-matrix(coalobs[1,4:6],nrow=1)

tmp&lt;-selectsumm(obsstats, param, simstats, ssmethod =AS.select, tol =.1,
method = "rejection", allow.none = FALSE, inturn = TRUE, hcorr = TRUE)

tmp$best


</code></pre>

<hr>
<h2 id='semiauto.abc'>Performs semi-automatic ABC based on summary statistics regression.</h2><span id='topic+semiauto.abc'></span>

<h3>Description</h3>

<p>Performs semi-automatic ABC based on summary statistics regression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>semiauto.abc(obs, param, sumstats, obspar=NULL, abcmethod = abc, 
saprop = 0.5, abcprop = 0.5, overlap = FALSE, satr = list(), 
plot = FALSE, verbose = TRUE, do.err = FALSE, final.dens = FALSE, 
errfn = rsse, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="semiauto.abc_+3A_obs">obs</code></td>
<td>
<p>(matrix of) observed summary statistics.</p>
</td></tr>
<tr><td><code id="semiauto.abc_+3A_param">param</code></td>
<td>
<p>matrix of simulated model parameter values.</p>
</td></tr>
<tr><td><code id="semiauto.abc_+3A_sumstats">sumstats</code></td>
<td>
<p>matrix of simulated summary statistics.</p>
</td></tr>
<tr><td><code id="semiauto.abc_+3A_obspar">obspar</code></td>
<td>
<p>optional observed parameters (for use to assess simulation performance).</p>
</td></tr>
<tr><td><code id="semiauto.abc_+3A_abcmethod">abcmethod</code></td>
<td>
<p>a function to perform ABC inference, e.g. the <code>abc</code> function from package <em>abc</em>.</p>
</td></tr>
<tr><td><code id="semiauto.abc_+3A_saprop">saprop</code></td>
<td>
<p>a proportion, denoting the proportion of simulated datasets with which to perform semi-automatic ABC regression.</p>
</td></tr>
<tr><td><code id="semiauto.abc_+3A_abcprop">abcprop</code></td>
<td>
<p>a proportion, denoting the proportion of simulated datasets with which to perform ABC using <code>abcmethod</code>.</p>
</td></tr>
<tr><td><code id="semiauto.abc_+3A_overlap">overlap</code></td>
<td>
<p>a boolean value indicating whether the simulated datasets specified by <code>saprop</code> and <code>abcprop</code> are disjoint (<code>overlap=FALSE</code>) or not.</p>
</td></tr>
<tr><td><code id="semiauto.abc_+3A_satr">satr</code></td>
<td>
<p>a list of functions indicating transformations of the summary statistics <code>sumstats</code>.  These must be *suitable* functions, and must each return a vector, matrix or array with the number of elements being a multiple of the rows of <code>sumstats</code>.  See details and examples sections for more information</p>
</td></tr>
<tr><td><code id="semiauto.abc_+3A_plot">plot</code></td>
<td>
<p>When plot==TRUE, a plot of parameter values against fitted values is produced for each parameter as a side-effect.  This is most useful when the number of parameters is reasonably small.</p>
</td></tr>
<tr><td><code id="semiauto.abc_+3A_verbose">verbose</code></td>
<td>
<p>a boolean value indicating whether informative statements should be printed to screen.</p>
</td></tr>
<tr><td><code id="semiauto.abc_+3A_do.err">do.err</code></td>
<td>
<p>a boolean value indicating whether the simulation error should be returned.  Note: if <code>do.err=TRUE</code>, <code>obspar</code> must be supplied.</p>
</td></tr>
<tr><td><code id="semiauto.abc_+3A_final.dens">final.dens</code></td>
<td>
<p>a boolean value indicating whether the posterior sample should be returned.</p>
</td></tr>
<tr><td><code id="semiauto.abc_+3A_errfn">errfn</code></td>
<td>
<p>an error function to assess ABC inference performance.</p>
</td></tr>
<tr><td><code id="semiauto.abc_+3A_...">...</code></td>
<td>
<p>any other optional arguments to the ABC inference procedure (e.g. arguments to the <code>abc</code> function).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is essentially a wrapper for <code>saABC</code>.  See the details section of <code>saABC</code> for more details on the implementation.  The argument <code>satr</code> can be almost anything sensible in <code>function</code> form, see Examples section for example specifications.</p>


<h3>Value</h3>

<p> A list with the following components:
</p>
<table>
<tr><td><code>err</code></td>
<td>
<p>simulation error (if <code>obspar</code> is supplied and <code>do.err=TRUE</code>).</p>
</td></tr>
<tr><td><code>post.sample</code></td>
<td>
<p>an array of dimension <code>nacc x npar x ndatasets</code> giving the posterior sample for each observed dataset.  Not returned if <code>final.dens=FALSE</code>.</p>
</td></tr>
<tr><td><code>sainfo</code></td>
<td>
<p>A list with the following information about the semi-automatic ABC run:<br /> <code>saprop</code>, <code>abcprop</code> ,<code>overlap</code>, <code>satr</code>.  See arguments for more details.</p>
</td></tr>
</table>


<h3>Warning</h3>

<p>The argument <code>satr</code> must be supplied with valid functions.  Whilst there are checks, these are minimal, since doing sophisticated checks is quite difficult.
</p>


<h3>Author(s)</h3>

<p>Matt Nunes and Dennis Prangle</p>


<h3>References</h3>

<p>Blum, M. G. B, Nunes, M. A., Prangle, D. and Sisson, S. A. (2013) A
comparative review of dimension reduction methods in approximate
Bayesian computation. <em>Stat. Sci.</em> <b>28</b>, Issue 2, 189&ndash;208.<br /><br />
Fearnhead, P. and Prangle, D. (2012) Constructing summary statistics for approximate Bayesian computation:
semi-automatic approximate Bayesian
computation. <em>J. R. Stat. Soc. B</em> <b>74</b>, Part 3, 1&ndash;28.<br /><br />
Nunes, M. A. and Prangle, D. (2016) abctools: an R package for tuning
approximate Bayesian computation analyses. <em>The R Journal</em>
<b>7</b>, Issue 2, 189&ndash;205.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+saABC">saABC</a></code>, <code><a href="#topic+selectsumm">selectsumm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
data(coal)
data(coalobs)

param&lt;-coal[,2]
simstats&lt;-coal[,4:6]

# use matrix below just in case to preserve dimensions.

obsstats&lt;-matrix(coalobs[1,4:6],nrow=1)
obsparam&lt;-matrix(coalobs[1,1])

# perform semi-automatic ABC with summary statistics defined by 
# X, X^2,X^3,X^4:
# other alternative specifications for this could be:
# list(function(x){ cbind(x,x^2,x^3,x^4) })
# list(as.function(alist(x=,cbind(x,x^2,x^3)))) etc

tmp&lt;-semiauto.abc(obsstats, param, simstats,tol=.01,method="rejection",
satr=list(function(x){outer(x,Y=1:4,"^")}))

tmp$sa.info

# both these functions may be problematic: 

 tmp&lt;-semiauto.abc(obsstats, param, simstats,tol=.01,method="rejection",
satr=list(unique,sum))

## End(Not run)

</code></pre>

<hr>
<h2 id='stage2'>stage2</h2><span id='topic+stage2'></span>

<h3>Description</h3>

<p>Summary statistics selection for ABC inference using estimated posterior error. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stage2(obs, param, sumstats, obspar = NULL, init.best, dsets = 100, 
sumsubs = 1:ncol(sumstats), limit = length(sumsubs), do.only=NULL, 
do.err = FALSE, final.dens = FALSE,  ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stage2_+3A_obs">obs</code></td>
<td>
<p>observed summary statistics.</p>
</td></tr>
<tr><td><code id="stage2_+3A_param">param</code></td>
<td>
<p>matrix of simulated model parameter values.</p>
</td></tr>
<tr><td><code id="stage2_+3A_sumstats">sumstats</code></td>
<td>
<p>matrix of simulated summary statistics.</p>
</td></tr>
<tr><td><code id="stage2_+3A_obspar">obspar</code></td>
<td>
<p>optional observed parameters (for use to assess simulation performance).</p>
</td></tr>
<tr><td><code id="stage2_+3A_init.best">init.best</code></td>
<td>
<p>an initial estimate of the best summary statistics subset.  Can be either an index into the summaries combination table (see <code><a href="#topic+combmat">combmat</a></code>) or a vector of indices into <code>1:nstats</code>.  See details.</p>
</td></tr>
<tr><td><code id="stage2_+3A_dsets">dsets</code></td>
<td>
<p>the number of simulated datasets to treat as observed when estimating the posterior error.  See details.</p>
</td></tr>
<tr><td><code id="stage2_+3A_sumsubs">sumsubs</code></td>
<td>
<p>an optional index into the summary statistics to limit summary selection to a specific subset of summaries.</p>
</td></tr>
<tr><td><code id="stage2_+3A_limit">limit</code></td>
<td>
<p>an optional integer indicating whether to limit summary selection to subsets of a maximum size.</p>
</td></tr>
<tr><td><code id="stage2_+3A_do.only">do.only</code></td>
<td>
<p>an optional index into the summary statistics combination table.  Can be used to limit entropy calculations to certain summary statistics subsets only.</p>
</td></tr>
<tr><td><code id="stage2_+3A_do.err">do.err</code></td>
<td>
<p>a boolean value indicating whether the simulation error should be returned.  Note: if <code>do.err=TRUE</code>, <code>obspar</code> must be supplied.</p>
</td></tr>
<tr><td><code id="stage2_+3A_final.dens">final.dens</code></td>
<td>
<p>a boolean value indicating whether the posterior sample should be returned.</p>
</td></tr>

<tr><td><code id="stage2_+3A_...">...</code></td>
<td>
<p>any other optional arguments to the ABC inference procedure (e.g. arguments to the <code>abc</code> function).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function uses the <code>init.best</code> set of summaries to determine the <code>dsets</code> simulated datasets which are closest (in Euclidean norm) to the observed dataset.  Since the model parameters generating the summary statistics are known for these simulated datasets, for each candidate subset of summary statistics, we can compute the error under ABC inference for each of these datasets.  The best subset of summary statistics is that which minimizes this (average) error over all <code>dsets</code> datasets.</p>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>best</code></td>
<td>
<p>the best subset of statistics.</p>
</td></tr>
<tr><td><code>closest</code></td>
<td>
<p>the indices of the <code>dsets</code> simulated datasets closest to the oberved dataset as measured by the <code>init.best</code> subset of summaries.</p>
</td></tr>
<tr><td><code>err</code></td>
<td>
<p>simulation error (if <code>obspar</code> is supplied and <code>do.err=TRUE</code>).</p>
</td></tr>
<tr><td><code>order</code></td>
<td>
<p>the subsets considered during the algorithm (same as the input <code>do.only</code>.</p>
</td></tr>
<tr><td><code>post.sample</code></td>
<td>
<p>an array of dimension <code>nacc x npar x ndatasets</code> giving the posterior sample for each observed dataset.  Not returned if <code>final.dens=FALSE</code>.</p>
</td></tr>
<tr><td><code>sumsubs</code></td>
<td>
<p>an index into the subsets considered during the algorithm.</p>
</td></tr>
</table>


<h3>Warning</h3>

<p>This function is computationally intensive due to its cyclic ABC inference procedure.</p>


<h3>Author(s)</h3>

<p>Matt Nunes</p>


<h3>References</h3>

<p>Blum, M. G. B, Nunes, M. A., Prangle, D. and Sisson, S. A. (2013) A
comparative review of dimension reduction methods in approximate
Bayesian computation. <em>Stat. Sci.</em> <b>28</b>, Issue 2, 189&ndash;208.<br /><br />
Nunes, M. A. and Balding, D. J. (2010) On Optimal Selection of Summary
Statistics for Approximate Bayesian Computation.
<em>Stat. Appl. Gen. Mol. Biol.</em> <b>9</b>, Iss. 1, Art. 34.<br /><br />
Nunes, M. A. and Prangle, D. (2016) abctools: an R package for tuning
approximate Bayesian computation analyses. <em>The R Journal</em>
<b>7</b>, Issue 2, 189&ndash;205.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# load example data:

data(coal)
data(coalobs)

param&lt;-coal[,2]
simstats&lt;-coal[,5:8]

# use matrix below just in case to preserve dimensions.

obsstats&lt;-matrix(coalobs[1,5:8],nrow=1)
obsparam&lt;-matrix(coalobs[1,1])

## Not run: 
tmp&lt;-stage2(obsstats, param, simstats, init.bes=c(1,3), dsets = 10) 
tmp$best

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
