<!DOCTYPE html><html><head><title>Help for package abd</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {abd}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#abd-package'><p>Data sets from The Analysis of Biological Data</p></a></li>
<li><a href='#abdData'><p>Find data in <em>Analysis of Biological Data</em></p></a></li>
<li><a href='#AlgaeCO2'><p>Carbon Dioxide and Growth Rate in Algae</p></a></li>
<li><a href='#Antilles'><p>Antilles Bird Immigration Dates</p></a></li>
<li><a href='#Aspirin'><p>Effects of Aspirin on Cancer Rates</p></a></li>
<li><a href='#BeeGenes'><p>Foraging Gene Expression</p></a></li>
<li><a href='#BeeLifespans'><p>Bee Lifespans</p></a></li>
<li><a href='#Beetles'><p>Beetle Wings and Horns</p></a></li>
<li><a href='#BirdSexRatio'><p>Sex Ratios in Birds</p></a></li>
<li><a href='#Blackbirds'><p>Testosterone Levels in Blackbirds</p></a></li>
<li><a href='#BodyFatHeatLoss'><p>Heat Loss and Body Fat</p></a></li>
<li><a href='#BrainExpression'><p>Proteolipid Protein 1 Gene Expression</p></a></li>
<li><a href='#BrookTrout'><p>Salmon Survival in the Presence of Brook Trout</p></a></li>
<li><a href='#Cavalry'><p>Deaths from Horse Kicks</p></a></li>
<li><a href='#Chickadees'><p>Alarm Calls in Chickadees</p></a></li>
<li><a href='#ChimpBrains'><p>Brodmann's Area 44 in Chimps</p></a></li>
<li><a href='#Cichlids'><p>Cichlid Mating Preference</p></a></li>
<li><a href='#CichlidsGnRH'><p>GnRH Levels in Cichlids</p></a></li>
<li><a href='#Clearcuts'><p>Biomass Change in Rainforests near Clearcuts</p></a></li>
<li><a href='#CocaineDopamine'><p>Effects of Cocaine on Dopamine Receptors</p></a></li>
<li><a href='#col.abd'><p>Lattice theme for Analysis of Biological Data</p></a></li>
<li><a href='#Convictions'><p>Frequency of Convictions for a Cohort of English Boys</p></a></li>
<li><a href='#ConvictionsAndIncome'><p>Convictions and Income Level in a Cohort of English Boys</p></a></li>
<li><a href='#Crickets'><p>Immunity and Sperm Viability in Crickets</p></a></li>
<li><a href='#DaphniaLongevity'><p>Daphnia Longevity</p></a></li>
<li><a href='#DaphniaResistance'><p>Daphnia Resistance to Cyanobacteria</p></a></li>
<li><a href='#dataInfo'><p><code>abd</code> Data Sets</p></a></li>
<li><a href='#DayOfBirth'><p>Day of Birth</p></a></li>
<li><a href='#DEET'><p>DEET and Mosquito Bites</p></a></li>
<li><a href='#DesertBirds'><p>Desert Bird Census Data</p></a></li>
<li><a href='#Dioecy'><p>Dioecy vs. Monomorphism in Plants</p></a></li>
<li><a href='#Dolphins'><p>Dolphin Swimming Behavior</p></a></li>
<li><a href='#DungBeetles'><p>Heritability of Body Condition in Dung Beetles</p></a></li>
<li><a href='#Earthworms'><p>Earthworm Diversity and Soil Nitrogen Levels</p></a></li>
<li><a href='#Earwigs'><p>Earwig Density and Forceps</p></a></li>
<li><a href='#Eelgrass'><p>Eelgrass Genotypes</p></a></li>
<li><a href='#ElectricFish'><p>Electric Fish</p></a></li>
<li><a href='#ElVerde'><p>Diet Breadth in a Rainforest Community</p></a></li>
<li><a href='#EndangeredSpecies'><p>Endangered and Threatened Species</p></a></li>
<li><a href='#FingerRatio'><p>2D:4D Finger Ratio</p></a></li>
<li><a href='#Fireflies'><p>Spermatophore Mass in Fireflies</p></a></li>
<li><a href='#FireflyFlash'><p>Firefly Flash Duration</p></a></li>
<li><a href='#FlycatcherPatch'><p>Forehead Patch Size in Collared Flycatachers</p></a></li>
<li><a href='#FlyTestes'><p>Testes Size in Flies</p></a></li>
<li><a href='#GeneRegulation'><p>Gene Regulation in Saccharomyces</p></a></li>
<li><a href='#GlidingSnakes'><p>GlidingSnakes</p></a></li>
<li><a href='#GodwitArrival'><p>Godwit Arrival Dates</p></a></li>
<li><a href='#Grassland'><p>Grassland Diversity</p></a></li>
<li><a href='#GreatTitMalaria'><p>Malaria in Populations of Great Tit</p></a></li>
<li><a href='#Greenspace'><p>Diversity in Urban Green Space</p></a></li>
<li><a href='#Guppies'><p>Ornamentation and Attractiveness in Guppies</p></a></li>
<li><a href='#Hemoglobin'><p>Hemoglobin Levels in High Altitude Populations</p></a></li>
<li><a href='#HippocampusLesions'><p>Memory and the Hippocampus</p></a></li>
<li><a href='#histochart'><p>Histogram from tabulated data</p></a></li>
<li><a href='#HornedLizards'><p>Horn Length and Predation Status of Horned Lizards</p></a></li>
<li><a href='#HumanBodyTemp'><p>Human Body Temperature</p></a></li>
<li><a href='#HumanGeneLengths'><p>Human Gene Lengths</p></a></li>
<li><a href='#Hurricanes'><p>Intense Hurricanes</p></a></li>
<li><a href='#Iguanas'><p>Iguana Body Length Changes</p></a></li>
<li><a href='#IntertidalAlgae'><p>Intertidal Algae</p></a></li>
<li><a href='#JetLagKnees'><p>Circadian Rhythm Phase Shift</p></a></li>
<li><a href='#KenyaFinches'><p>Body Mass and Beak Length in Three Species of Finches in Kenya</p></a></li>
<li><a href='#LanguageBrains'><p>Brain Structure in Bilingual Humans</p></a></li>
<li><a href='#LarvalFish'><p>Exploited Larval Fish</p></a></li>
<li><a href='#Lefthanded'><p>Left-handedness and Rates of Violence</p></a></li>
<li><a href='#LionCubs'><p>Time to Reproduction in Female Lions</p></a></li>
<li><a href='#LionNoses'><p>Lion Age and Nose Coloration</p></a></li>
<li><a href='#LiverPreparation'><p>Liver Preparation</p></a></li>
<li><a href='#LizardBite'><p>Bite Force in Collard Lizards</p></a></li>
<li><a href='#LizardSprint'><p>Sprint Speeds in Canyon Lizards</p></a></li>
<li><a href='#Lobsters'><p>Lobster Orientation</p></a></li>
<li><a href='#LodgepolePines'><p>Lodgepole Pine Cone Masses</p></a></li>
<li><a href='#LupusMice'><p>Autoimmune Reactivity in Lupus-prone Mice</p></a></li>
<li><a href='#Lynx'><p>Population Cycles of Lynx in Canada 1752-1819</p></a></li>
<li><a href='#MarineReserve'><p>Marine Reserve Biomass</p></a></li>
<li><a href='#MassExtinctions'><p>Mass Extinction Frequency</p></a></li>
<li><a href='#MoleRats'><p>Energy Expenditure in Mole Rats</p></a></li>
<li><a href='#Mosquitoes'><p>Body Size in Anopheles Mosquitoes</p></a></li>
<li><a href='#MouseEmpathy'><p>Mouse Empathy</p></a></li>
<li><a href='#NeanderthalBrains'><p>Cranial Capacity in Neanderthals and Modern Humans</p></a></li>
<li><a href='#NematodeLifespan'><p>Effects of Trimethadione on Lifespan in Nematodes</p></a></li>
<li><a href='#NeotropicalTrees'><p>Photosynthesis in Neotropical Trees</p></a></li>
<li><a href='#Newts'><p>Tetrodotoxin Resistance in Garter Snakes</p></a></li>
<li><a href='#NorthSeaCod'><p>Atlantic Cod Recruits</p></a></li>
<li><a href='#NoSmokingDay'><p>No Smoking Day</p></a></li>
<li><a href='#OstrichTemp'><p>Ostrich Body and Brain Temperatures</p></a></li>
<li><a href='#Penguins'><p>Penguin Heart Rate</p></a></li>
<li><a href='#PlantPersistence'><p>Population Persistence Times</p></a></li>
<li><a href='#Pollen'><p>Sterility in Hybrid Pollens</p></a></li>
<li><a href='#Powerball'><p>Powerball Tickets Sold</p></a></li>
<li><a href='#PrimateMetabolism'><p>Primate Metabolic Rates</p></a></li>
<li><a href='#PrimateWBC'><p>Primate White Blood Cell Counts and Promiscuity</p></a></li>
<li><a href='#ProgesteroneExercise'><p>Progesterone and Exercise</p></a></li>
<li><a href='#Pseudoscorpions'><p>Multiple Mating in Pseudoscorpions</p></a></li>
<li><a href='#Pufferfish'><p>Pufferfish Mimicry</p></a></li>
<li><a href='#Rattlesnakes'><p>Temperature Change and Meal Size in Rattlesnakes</p></a></li>
<li><a href='#Rigormortis'><p>Rigormortis and Time of Death</p></a></li>
<li><a href='#RopeTrick'><p>Indian Rope Trick</p></a></li>
<li><a href='#SagebrushCrickets'><p>Sagebrush Cricket Mating Times</p></a></li>
<li><a href='#SalmonColor'><p>Pacific Salmon Color</p></a></li>
<li><a href='#Seedlings'><p>Number of Seedlings Per Quadrat</p></a></li>
<li><a href='#Selection'><p>Data for Meta-analysis</p></a></li>
<li><a href='#SexualSelection'><p>Sexual Conflict</p></a></li>
<li><a href='#ShadParasites'><p>Shad Parasites</p></a></li>
<li><a href='#ShrinkingSeals'><p>Seal Body Lengths and Age</p></a></li>
<li><a href='#ShuttleDisaster'><p>Ambient Temperature and O-Ring Failures</p></a></li>
<li><a href='#Silversword'><p>Rate of Speciation in Silverswords</p></a></li>
<li><a href='#SleepAndPerformance'><p>Sleep and Learning</p></a></li>
<li><a href='#SockeyeFemales'><p>Body Masses of Female Sockeye Salmon</p></a></li>
<li><a href='#Sparrows'><p>Lifetime Reproductive Success in House Sparrows</p></a></li>
<li><a href='#SpiderColonies'><p>Social Spiders</p></a></li>
<li><a href='#SpiderSpeed'><p>Spider Running Speeds after Amputation</p></a></li>
<li><a href='#Stalkies1'><p>Eye Widths in Stalk-Eyed Flies</p></a></li>
<li><a href='#Stalkies2'><p>Stalk-eyed Fly Eyespan</p></a></li>
<li><a href='#SticklebackPlates'><p>Number of Lateral Plates in Sticklebacks</p></a></li>
<li><a href='#SticklebackPreference'><p>Mating Preferences in Sticklebacks</p></a></li>
<li><a href='#Sumo'><p>Sumo Wrestling Wins</p></a></li>
<li><a href='#SyrupSwimming'><p>Syrup Swimming</p></a></li>
<li><a href='#TeenDeaths'><p>Causes of Teenage Deaths</p></a></li>
<li><a href='#Telomeres'><p>Telomere Shortening</p></a></li>
<li><a href='#TimeOfDeath'><p>Hypoxanthine and Time Since Death</p></a></li>
<li><a href='#Toads'><p>Right-handed Toads</p></a></li>
<li><a href='#Tobacco'><p>Flower Length in Tobacco Plants</p></a></li>
<li><a href='#Tobacco2'><p>Flower Length in Tobacco Plants</p></a></li>
<li><a href='#ToothAge'><p>Radioactive Teeth</p></a></li>
<li><a href='#TreeSeedlings'><p>Tree Seedlings and Sunflecks</p></a></li>
<li><a href='#Trematodes'><p>Frequencies of Fish Eaten by Trematode Infection Level</p></a></li>
<li><a href='#Trillium'><p>Trillium Recruitment near Clearcuts</p></a></li>
<li><a href='#Truffles'><p>Truffle Distribution</p></a></li>
<li><a href='#TsetseLearning'><p>Dietary Learning in Tsetse Flies</p></a></li>
<li><a href='#TwoKids'><p>Number of Boys in Two-Child Families</p></a></li>
<li><a href='#VampireBites'><p>Vampire Bat Bites</p></a></li>
<li><a href='#VasopressinVoles'><p>Vasopressin Manipulation in the Meadow Vole</p></a></li>
<li><a href='#Vines'><p>Climbing Vines</p></a></li>
<li><a href='#VoleDispersal'><p>Home Range Size in Field Voles</p></a></li>
<li><a href='#WalkingStickFemurs'><p>Walking Stick Femur Length</p></a></li>
<li><a href='#WalkingStickHeads'><p>Walking Stick Head Width</p></a></li>
<li><a href='#WeddellSeals'><p>Energetic Cost of Diving</p></a></li>
<li><a href='#WillsDebates'><p>Presidential &quot;Wills&quot;</p></a></li>
<li><a href='#WillsPresidents'><p>Presidential &quot;Wills&quot;</p></a></li>
<li><a href='#WolfTeeth'><p>Wolf Tooth Measurements</p></a></li>
<li><a href='#Wolves'><p>Inbreeding in Wolves</p></a></li>
<li><a href='#WorldCup'><p>World Cup Goals</p></a></li>
<li><a href='#WrasseSexes'><p>Distribution of Wrasses</p></a></li>
<li><a href='#YeastGenes'><p>Yeast Regulatory Genes</p></a></li>
<li><a href='#ZebraFinchBeaks'><p>Mate Preference in Zebra Finches</p></a></li>
<li><a href='#ZebraFinches'><p>Zebra Finch Carotenoids</p></a></li>
<li><a href='#ZooMortality'><p>Home Range Size and Mortality</p></a></li>
<li><a href='#Zooplankton'><p>Zooplankton Depredation</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>The Analysis of Biological Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2-8</td>
</tr>
<tr>
<td>Date:</td>
<td>2015-07-02</td>
</tr>
<tr>
<td>Author:</td>
<td>Kevin M. Middleton &lt;middletonk@missouri.edu&gt;, Randall Pruim
    &lt;rpruim@calvin.edu&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kevin M. Middleton &lt;middletonk@missouri.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0), nlme, lattice, grid, mosaic</td>
</tr>
<tr>
<td>Suggests:</td>
<td>boot, car, ggplot2, plyr, HH, ICC, vcd, Hmisc</td>
</tr>
<tr>
<td>Description:</td>
<td>The abd package contains data sets and sample code for The
    Analysis of Biological Data by Michael Whitlock and Dolph Schluter (2009;
    Roberts &amp; Company Publishers).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Collate:</td>
<td>'abdData.R' 'histochart.R' 'datasets.R' 'themes.R'</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2015-07-03 01:35:03 UTC; kmm</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2015-07-03 05:44:45</td>
</tr>
</table>
<hr>
<h2 id='abd-package'>Data sets from The Analysis of Biological Data</h2><span id='topic+abd'></span><span id='topic+abd-package'></span>

<h3>Description</h3>

<p>The <code>abd</code> package contains data sets and sample code for the book,
<em>The Analysis of Biological Data</em> by Michael C. Whitlock and Dolph
Schluter (2009; Roberts and Company Publishers).
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;"> Package: </td><td style="text-align: left;"> abd</td>
</tr>
<tr>
 <td style="text-align: left;">
              Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
              Version: </td><td style="text-align: left;"> 0.2-8</td>
</tr>
<tr>
 <td style="text-align: left;">
              Date: </td><td style="text-align: left;"> 2015-07-02</td>
</tr>
<tr>
 <td style="text-align: left;">
              License: </td><td style="text-align: left;"> GPL</td>
</tr>
<tr>
 <td style="text-align: left;">
              LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
              LazyData: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Kevin M. Middleton (<a href="middletonk@missouri.edu">middletonk@missouri.edu</a>); Randall Pruim
(<a href="rpruim@calvin.edu">rpruim@calvin.edu</a>)
</p>


<h3>References</h3>

<p>Whitlock, M.C. and D. Schluter. 2009. <em>The Analysis of
Biological Data</em>. Roberts and Company Publishers. ISBN: 0981519407.
<a href="http://www.roberts-publishers.com/biology/the-analysis-of-biological-data.html">http://www.roberts-publishers.com/biology/the-analysis-of-biological-data.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>trellis.par.set(theme=col.abd())  # set color theme
show.settings()
abdData(3)                        # look for data sets in chapter 3
abdData('Finch')                  # look for data sets with 'finch' in name
</code></pre>

<hr>
<h2 id='abdData'>Find data in <em>Analysis of Biological Data</em></h2><span id='topic+abdData'></span>

<h3>Description</h3>

<p>A utility function to assist users to locate data sets in <em>Analysis of
Biological Data</em> within the <code>abd</code> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>abdData(..., chapters = 1:21, types = c("Example", "Problem"),
  numbers = 1:100, pattern = "*", ignore.case = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="abdData_+3A_chapters">chapters</code></td>
<td>
<p>a numeric vector of chapters to search within</p>
</td></tr>
<tr><td><code id="abdData_+3A_types">types</code></td>
<td>
<p>a sub-vector of c('Example','Problem')</p>
</td></tr>
<tr><td><code id="abdData_+3A_numbers">numbers</code></td>
<td>
<p>a numeric vector of problem numbers</p>
</td></tr>
<tr><td><code id="abdData_+3A_pattern">pattern</code></td>
<td>
<p>a pattern to use for regular expression matching against the
name of the data frame.</p>
</td></tr>
<tr><td><code id="abdData_+3A_ignore.case">ignore.case</code></td>
<td>
<p>should case be ignored when matching pattern?</p>
</td></tr>
<tr><td><code id="abdData_+3A_...">...</code></td>
<td>
<p>values for any of <code>chapters</code>, <code>types</code>, or
<code>pattern</code>.  Which is meant will be inferred from the type of object
supplied.  This allows users to specify these values in any order and
without naming.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame describing data sets from <code>abd</code> that match the
search criteria, or NULL if there are no matches.
</p>


<h3>Author(s)</h3>

<p>Randall Pruim (<a href="mailto:rpruim@calvin.edu">rpruim@calvin.edu</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># find all data from examples in chapters 3 and 4
abdData(3:4, 'Example')

# order doesn't matter
abdData('Example', 3:4)

# look for data sets with Example in their name.
abdData(pattern='Example')

# look for data sets with Exercise in their name.
abdData('Exercise')
</code></pre>

<hr>
<h2 id='AlgaeCO2'>Carbon Dioxide and Growth Rate in Algae</h2><span id='topic+AlgaeCO2'></span>

<h3>Description</h3>

<p>Growth rates of the unicellular alga <em>Chlamydomonas</em> after 1,000
generations of selection under <code>High</code> and <code>Normal</code> levels of
carbon dioxide.
</p>


<h3>Format</h3>

<p>A data frame with 14 observations on the following 2 variables.
</p>
 <dl>
<dt>treatment</dt><dd><p>a factor with levels <code>Normal</code> and
<code>High</code></p>
</dd> <dt>growthrate</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Collins, S. and G. Bell. 2004. Phenotypic consequences of 1,000
generations of selection at elevated CO<code class="reqn">_{2}</code> in a green alga.
<em>Nature</em> 431: 566-569.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>AlgaeCO2
xyplot(growthrate ~ treatment, AlgaeCO2, type = c('p', 'a'))
</code></pre>

<hr>
<h2 id='Antilles'>Antilles Bird Immigration Dates</h2><span id='topic+Antilles'></span>

<h3>Description</h3>

<p>Approximate dates of immigration for 37 species of birds in the Lesser
Antilles.
</p>


<h3>Format</h3>

<p>A data frame with 37 observations of one variable. </p>

<dl>
<dt>immigration.date</dt><dd><p>approximate immigration date (in millions of
years)</p>
</dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> Ricklefs, R.E. and E. Bermingham. 2001.
Nonequilibrium diversity dynamics of the Lesser Antillean avifauna.
<em>Science</em> 294: 1522-1524.
</p>


<h3>References</h3>

<p><a href="http://www.sciencemag.org/cgi/content/abstract/sci;294/5546/1522">http://www.sciencemag.org/cgi/content/abstract/sci;294/5546/1522</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>histogram(~immigration.date, Antilles,n=15)
densityplot(~immigration.date, Antilles)
</code></pre>

<hr>
<h2 id='Aspirin'>Effects of Aspirin on Cancer Rates</h2><span id='topic+Aspirin'></span>

<h3>Description</h3>

<p>Frequency of cancer in 39,876 women taking and not taking aspirin.
</p>


<h3>Format</h3>

<p>A data frame with 39876 observations on the following </p>

<dl>
<dt>treatment</dt><dd><p>a factor with levels <code>Aspirin</code> and
<code>Placebo</code></p>
</dd> <dt>cancer</dt><dd><p>a factor with levels <code>no</code> and
<code>yes</code></p>
</dd> </dl>


<h3>Source</h3>

<p>Cook, N.R., I. Lee, J.M. Gaziano, D. Gordon, P.M. Ridker, J.E.
Manson, C.H. Hennekens, and J.E. Buring. 2005. Low-dose aspirin in the
primary prevention of cancer. <em>Journal of the American Medical
Association</em> 294: 47-55.
</p>


<h3>References</h3>

<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/15998890">http://www.ncbi.nlm.nih.gov/pubmed/15998890</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>demo(sec9.2)
</code></pre>

<hr>
<h2 id='BeeGenes'>Foraging Gene Expression</h2><span id='topic+BeeGenes'></span>

<h3>Description</h3>

<p>Levels of expression of the foraging gene (<em>for</em>; <code>Expression</code>) in
two worker types (<code>type</code>) in three bee colonies (<code>colony</code>). Note
that <code>colony</code> is not coded as a factor.
</p>


<h3>Format</h3>

<p>A data frame with 6 observations on the following 3 variables.
</p>
 <dl>
<dt>type</dt><dd><p>a factor with levels <code>forager</code>
<code>nurse</code></p>
</dd> <dt>colony</dt><dd><p>a numeric identifier</p>
</dd>
<dt>expression</dt><dd><p>expression level of the <em>for</em> gene</p>
</dd> </dl>


<h3>Source</h3>

<p>Ben-Shahar, Y., A. Robichon, M.B. Sokolowski, and G.E. Robinson.
2002. Influence of gene action across different time scales on behavior.
<em>Science</em> 296: 741-744.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(BeeGenes)
BeeGenes
xtabs( expression ~ type + colony, BeeGenes )
</code></pre>

<hr>
<h2 id='BeeLifespans'>Bee Lifespans</h2><span id='topic+BeeLifespans'></span>

<h3>Description</h3>

<p>Lifespan of 33 foraging honey bees.
</p>


<h3>Format</h3>

<p>A data frame with 33 observations on the following variable.
</p>
 <dl>
<dt>hours</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> Visscher, P.K. and R. Dukas. 1997. Survivorship
of foraging honey bees. <em>Insectes Sociaux</em> 44: 1-5.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>histogram(~hours, BeeLifespans, n=10)
densityplot(~hours, BeeLifespans)
</code></pre>

<hr>
<h2 id='Beetles'>Beetle Wings and Horns</h2><span id='topic+Beetles'></span>

<h3>Description</h3>

<p>Relative size of the horns and wings in 19 female <em>Onthophagus
sagittarius</em> beetles.
</p>


<h3>Format</h3>

<p>A data frame with 19 observations on the following 2 variables.
</p>
 <dl>
<dt>horn.size</dt><dd><p>a numeric vector</p>
</dd>
<dt>wing.mass</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Emlen, D.J. 2001. Costs and the diversification of exaggerated
animal structures. <em>Science</em> 291: 1534-1536.
</p>


<h3>References</h3>

<p><a href="http://www.scienceonline.org/cgi/content/abstract/291/5508/1534">http://www.scienceonline.org/cgi/content/abstract/291/5508/1534</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(Beetles)
xyplot(wing.mass ~ horn.size, Beetles)
</code></pre>

<hr>
<h2 id='BirdSexRatio'>Sex Ratios in Birds</h2><span id='topic+BirdSexRatio'></span>

<h3>Description</h3>

<p>Correlation coefficient of sex ratio in bird offspring.
</p>


<h3>Format</h3>

<p>A data frame with 15 observations of one variable </p>

<dl>
<dt>corr.coeff</dt><dd><p>correlation coefficient of sex ratio in bird
offspring</p>
</dd> </dl>


<h3>Source</h3>

<p>West, S.A. and B.C. Sheldon. 2002. Constraints in the evolution of
sex ratio adjustment. <em>Science</em> 295: 1695-1688.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>histogram(~corr.coeff, BirdSexRatio, n = 10,
  xlab = "Correlation Coefficient")
</code></pre>

<hr>
<h2 id='Blackbirds'>Testosterone Levels in Blackbirds</h2><span id='topic+Blackbirds'></span>

<h3>Description</h3>

<p>Experimental manipulation of testosterone levels in male Red-winged
Blackbirds (<em>Agelaius phoeniceus</em>) and resulting changes in antibody
levels
</p>


<h3>Format</h3>

<p>A data frame with 13 observations on the following 6 variables.
</p>
 <dl>
<dt>before</dt><dd><p>a numeric vector</p>
</dd> <dt>after</dt><dd><p>a
numeric vector</p>
</dd> <dt>log.before</dt><dd><p>a numeric vector</p>
</dd>
<dt>log.after</dt><dd><p>a numeric vector</p>
</dd> <dt>diff.in.logs</dt><dd><p>a
numeric vector</p>
</dd> <dt>diff</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Hasselquist, D., J.A. Marsh, P.W. Sherman, and J.C. Wingfield. 1999.
Is avian immunocompetence suppressed by testosterone? <em>Behavioral
Ecology and Sociobiology</em> 45: 167-175.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Blackbirds
xyplot(log.after ~ log.before, data = Blackbirds,
  ylab = "log Antibody production after implant",
  xlab = "log Antibody production before implant"
)
</code></pre>

<hr>
<h2 id='BodyFatHeatLoss'>Heat Loss and Body Fat</h2><span id='topic+BodyFatHeatLoss'></span>

<h3>Description</h3>

<p>Heat loss during exercise and relative body fat in 12 boys.
</p>


<h3>Format</h3>

<p>A data frame with 12 observations on the following 2 variables.
</p>
 <dl>
<dt>leanness</dt><dd><p>a numeric vector</p>
</dd>
<dt>lossrate</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Sloan, R.E.G. and W.R. Keatinge. 1973. Cooling rates of young people
swimming in cold water. <em>Journal of Applied Physiology</em> 35: 371-375.
</p>


<h3>References</h3>

<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/4732330">http://www.ncbi.nlm.nih.gov/pubmed/4732330</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(lossrate ~ leanness, BodyFatHeatLoss)
</code></pre>

<hr>
<h2 id='BrainExpression'>Proteolipid Protein 1 Gene Expression</h2><span id='topic+BrainExpression'></span>

<h3>Description</h3>

<p>Expression levels of the proteolipid protein 1 gene (PLP1;
<code>PLP1.expression</code>) in 45 individuals in one of three <code>group</code>s.
</p>


<h3>Format</h3>

<p>A data frame with 45 observations on the following 2 variables.
</p>
 <dl>
<dt>group</dt><dd><p>a factor with levels: <code>bipolar</code>,
<code>control</code>, and <code>schizo</code></p>
</dd> <dt>PLP1.expression</dt><dd><p>a numeric
vector</p>
</dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> Tkachev, D., M.L. Mimmack, M.M. Ryan, M.
Wayland, T. Freeman, P.B. Jones, M. Starkey, M.J. Webster, R.H. Yolken, S.
Bahn. 2003. Oligodendrocyte dysfunction in schizophrenia and bipolar
disorder. <em>Lancet</em> 362(9386): 798-805.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bwplot(PLP1.expression ~ group, BrainExpression)
</code></pre>

<hr>
<h2 id='BrookTrout'>Salmon Survival in the Presence of Brook Trout</h2><span id='topic+BrookTrout'></span><span id='topic+BrookTrout2'></span>

<h3>Description</h3>

<p>Total numbers of salmon released (<code>salmon.released</code>) and surviving
(<code>salmon.surviving</code>) in 12 streams, 6 with brook trout <code>present</code>
and 6 with brook trout <code>absent</code>. The proportion of salmon surviving
(<code>proportion.surviving</code>) is given for each stream.
</p>


<h3>Format</h3>

<p><code>BrookTrout</code> is a data frame with 12 observations on the
following 4 variables.  <code>BrookTrout2</code> is a different summary of the
same study and gives survival rates for chinook in different years.
</p>
 <dl>
<dt>trout</dt><dd><p>a factor with levels <code>absent</code> and
<code>present</code> indicating whether brook trout are absent or present in the
stream</p>
</dd> <dt>salmon.released</dt><dd><p>a numeric vector of the total number
of salmon released</p>
</dd> <dt>salmon.surviving</dt><dd><p>a numeric vector of the
number of salmon surviving</p>
</dd> <dt>proportion.surviving</dt><dd><p>a numeric
vector of the proportion of salmon surviving</p>
</dd> </dl>


<h3>Source</h3>

<p>Levin, P.S., S. Achord, B.E. Fiest, and R.W. Zabel. 2002.
Non-indigenous brook trout and the demise of Pacific salmon: a forgotten
threat? <em>Proceedings of the Royal Society of London, Series B,
Biological Sciences</em> 269: 1663-1670.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(BrookTrout)
str(BrookTrout2)

bwplot(proportion.surviving ~ trout, BrookTrout)

aggregate(proportion.surviving ~ trout, BrookTrout, FUN = favstats)
summary(proportion.surviving ~ trout, BrookTrout, fun = favstats)
</code></pre>

<hr>
<h2 id='Cavalry'>Deaths from Horse Kicks</h2><span id='topic+Cavalry'></span>

<h3>Description</h3>

<p>Numbers of deaths resulting from horse kicks per regiment-years for the
Prussian army.
</p>


<h3>Format</h3>

<p>A data frame with 5 observations on the following 2 variables.
</p>
 <dl>
<dt>deaths</dt><dd><p>a numeric vector</p>
</dd> <dt>count</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Bortkiewicz, L. 1898. Das Gesetz der Kleinen Zahlen (Teubner,
Leipzig), <em>as cited in</em> Larson, R.J. and M.L. Marx. 1981. <em>An
Introduction to Mathematical Statistics and its Applications</em>.
Prentice-Hall: Englewood Cliffs, NJ.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Cavalry
xyplot(count ~ deaths, Cavalry, type='h', lwd=4)
barchart(count ~ deaths, Cavalry, horizontal = FALSE,
         box.ratio = 1000, origin=0)
</code></pre>

<hr>
<h2 id='Chickadees'>Alarm Calls in Chickadees</h2><span id='topic+Chickadees'></span>

<h3>Description</h3>

<p>Number of &quot;dee&quot; notes per call in Black-capped Chickadees (<em>Poecile
atricapilla</em>) for 13 predator species with differing body masses.
</p>


<h3>Format</h3>

<p>A data frame with 13 observations on the following 3 variables.
</p>
 <dl>
<dt>species</dt><dd><p>a character vector</p>
</dd> <dt>mass</dt><dd><p>a
numeric vector</p>
</dd> <dt>dees</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Templeton, C.N., E. Greene, and K. Davis. 2005. Allometry of alarm
calls: Black-capped Chickadees encode information about predator size.
<em>Science</em> 308: 1934-1937.
</p>


<h3>References</h3>

<p><a href="http://www.sciencemag.org/cgi/content/short/308/5730/1934">http://www.sciencemag.org/cgi/content/short/308/5730/1934</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(Chickadees)
Chickadees

xyplot(dees ~ mass, data = Chickadees,
   xlab = "Predator body mass (kg)",
   ylab = "'Dees' per call", type=c('p','r')
)
</code></pre>

<hr>
<h2 id='ChimpBrains'>Brodmann's Area 44 in Chimps</h2><span id='topic+ChimpBrains'></span>

<h3>Description</h3>

<p>Asymmetry of Brodmann's area 44 in 20 chimpanzees.
</p>


<h3>Format</h3>

<p>A data frame with 20 observations on the following 3 variables.
</p>
 <dl>
<dt>name</dt><dd><p>name of chimp</p>
</dd> <dt>sex</dt><dd><p>a factor
with levels <code>F</code> and <code>M</code></p>
</dd> <dt>asymmetry</dt><dd><p>asymmetry score</p>
</dd>
</dl>


<h3>Source</h3>

<p>Cantalupo, C. and W.D. Hopkins. 2001. Asymmetric Broca's area in
great apes. <em>Nature</em> 414: 505.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(asymmetry ~ sex, ChimpBrains)
aggregate(asymmetry ~ sex, ChimpBrains, FUN = favstats)
summary(asymmetry ~ sex, ChimpBrains, fun = favstats)
</code></pre>

<hr>
<h2 id='Cichlids'>Cichlid Mating Preference</h2><span id='topic+Cichlids'></span>

<h3>Description</h3>

<p>Preference index in F1 and F2 crosses of two species of cichlids from Lake
Victoria, <em>Pundamilia pundamilia</em> and <em>P. nyererei</em>.
</p>


<h3>Format</h3>

<p>A data frame with 53 observations on the following 2 variables.
</p>
 <dl>
<dt>genotype</dt><dd><p>a factor with levels <code>F1</code> and
<code>F2</code></p>
</dd> <dt>preference</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Haeslery, M.P. and O. Seehausen. 2005. Inheritance of female mating
preference in a sympatric sibling species pair of Lake Victoria cichlids:
implications for speciation. <em>Proceedings of the Royal Society of
London, Series B, Biological Sciences</em> 272: 237-245.
</p>


<h3>References</h3>

<p><a href="http://rspb.royalsocietypublishing.org/content/272/1560/237.full.pdf">http://rspb.royalsocietypublishing.org/content/272/1560/237.full.pdf</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(Cichlids)

summary(preference ~ genotype, Cichlids, fun = favstats)

if (require(plyr)) {
ddply(Cichlids, .(genotype),
  function(df)c(mean = mean(df$preference),
                standard.deviation = sd(df$preference),
                n = length(df$preference)))
}
</code></pre>

<hr>
<h2 id='CichlidsGnRH'>GnRH Levels in Cichlids</h2><span id='topic+CichlidsGnRH'></span>

<h3>Description</h3>

<p>Levels of mRNA for gonadotropin-releasing hormone in cichlids
(<em>Haplochromis burtoni</em>) that are (<em>n</em> = 5) and are not (<em>n</em>
= 6) territorial.
</p>


<h3>Format</h3>

<p>A data frame with 11 observations on the following 2 variables.
</p>
 <dl>
<dt>territorial</dt><dd><p>a factor with levels <code>No</code> and
<code>Yes</code></p>
</dd> <dt>GnRH.mRNA</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>White, S.A., T. Nguyen, and R.D. Fernald. 2002. Social regulation of
gonadotropin-releasing hormone. <em>Journal of Experimental Biology</em> 205:
2567-2581.
</p>


<h3>References</h3>

<p><a href="http://jeb.biologists.org/cgi/content/abstract/205/17/2567">http://jeb.biologists.org/cgi/content/abstract/205/17/2567</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(GnRH.mRNA ~ territorial, CichlidsGnRH, type=c('p','a'))
</code></pre>

<hr>
<h2 id='Clearcuts'>Biomass Change in Rainforests near Clearcuts</h2><span id='topic+Clearcuts'></span>

<h3>Description</h3>

<p>Biomass change in 36 Amazonian rainforests following clearcuts ranging from
50 m to several kilometers.
</p>


<h3>Format</h3>

<p>A data frame with 36 observations of one variable. </p>

<dl>
<dt>biomass.change</dt><dd></dd> </dl>


<h3>Source</h3>

<p>Laurance, W.F., S.G. Laurance, L.V. Ferreira, J.M. Rankin-de Merona,
C. Gascon, T.E. Lovejoy. 1997. Biomass collapse in Amazonian forest
fragments. <em>Science</em> 278: 1117-1118.
</p>


<h3>References</h3>

<p><a href="http://www.sciencemag.org/cgi/content/abstract/278/5340/1117">http://www.sciencemag.org/cgi/content/abstract/278/5340/1117</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(Clearcuts)
histogram(~biomass.change, Clearcuts)
</code></pre>

<hr>
<h2 id='CocaineDopamine'>Effects of Cocaine on Dopamine Receptors</h2><span id='topic+CocaineDopamine'></span>

<h3>Description</h3>

<p>Percent of dopamine receptors blocked (<code>percent.bocked</code>) and the
perceived level of high as determined by PET scans (<code>high</code>) in 34
humans.
</p>


<h3>Format</h3>

<p>A data frame with 34 observations on the following 2 variables.
</p>
 <dl>
<dt>percent.blocked</dt><dd><p>a numeric vector</p>
</dd>
<dt>high</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Volkow, N.D., G.-J. Wang, R.W. Foltin, J.S. Fowler, N.N. Abumrad, S.
Vitkun, J. Logan, S.J. Gatley, N. Pappas, R. Hitzemann, and C.E. Shea. 1997.
Relationship between subjective effects of cocaine and dopamine transporter
occupancy. <em>Nature</em> 386: 827-830.
</p>


<h3>References</h3>

<p><a href="http://www.nature.com/nature/journal/v386/n6627/abs/386827a0.html">http://www.nature.com/nature/journal/v386/n6627/abs/386827a0.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(CocaineDopamine)
xyplot(high ~ percent.blocked, CocaineDopamine)
</code></pre>

<hr>
<h2 id='col.abd'>Lattice theme for Analysis of Biological Data</h2><span id='topic+col.abd'></span><span id='topic+col.abd+2Ctheme.abd'></span><span id='topic+theme.abd'></span>

<h3>Description</h3>

<p>This theme will help produce plots with color scheme similar to the one used
in <em>Analysis of Biological Data</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>col.abd(bw = FALSE, lty = 1:7)

theme.abd(bw = FALSE, lty = 1:7)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="col.abd_+3A_bw">bw</code></td>
<td>
<p>a logical. Use a grayscale theme instead of color?</p>
</td></tr>
<tr><td><code id="col.abd_+3A_lty">lty</code></td>
<td>
<p>line types used for <code>panel.superpose</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>theme.abd</code> and <code>col.abd</code> are the same function with
two names.
</p>


<h3>Value</h3>

<p>a list that can be used as a <code>lattice</code> theme.
</p>


<h3>Author(s)</h3>

<p>Randall Pruim (<a href="mailto:rpruim@calvin.edu">rpruim@calvin.edu</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>trellis.par.set(theme=col.abd(bw=TRUE))
show.settings()
trellis.par.set(theme=theme.abd(lty=1))
show.settings()
</code></pre>

<hr>
<h2 id='Convictions'>Frequency of Convictions for a Cohort of English Boys</h2><span id='topic+Convictions'></span>

<h3>Description</h3>

<p>Data on frequency of convictions for a cohort of 395 boys.
</p>


<h3>Format</h3>

<p>A data frame with 15 observations on the following 2 variables.
</p>
 <dl>
<dt>convictions</dt><dd><p>number of convictions</p>
</dd>
<dt>boys</dt><dd><p>number of boys with given number of convictions</p>
</dd> </dl>


<h3>Source</h3>

<p>Farrington, D.P. 1994. <em>Cambridge Study in Delinquent
Development</em> [Great Britain], 1961-1981. 2nd ICPSR ed. Inter-university
Consortium for Political and Social Research, Ann Arbor, MI.
</p>


<h3>References</h3>

<p><a href="http://www.icpsr.umich.edu/icpsrweb/NACJD/archive.jsp">http://www.icpsr.umich.edu/icpsrweb/NACJD/archive.jsp</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(Convictions)
barchart(boys ~ as.factor(convictions), Convictions, horizontal = FALSE, origin=0)
xyplot( boys ~ convictions, Convictions, type = "h", lwd = 20)
</code></pre>

<hr>
<h2 id='ConvictionsAndIncome'>Convictions and Income Level in a Cohort of English Boys</h2><span id='topic+ConvictionsAndIncome'></span>

<h3>Description</h3>

<p>Data reporting the number of individual with and without convictions per
income level.
</p>


<h3>Format</h3>

<p>A data frame with 395 observations on the following 2 variables.
</p>
 <dl>
<dt>convicted</dt><dd><p>a factor with levels <code>no</code> and
<code>yes</code></p>
</dd> <dt>income</dt><dd><p>a factor with levels <code>adequate</code>,
<code>comfortable</code>, and <code>inadequate</code></p>
</dd> </dl>


<h3>Source</h3>

<p>Farrington, D.P. 1994. <em>Cambridge Study in Delinquent
Development</em> [Great Britain], 1961-1981. 2nd ICPSR ed. Inter-university
Consortium for Political and Social Research, Ann Arbor, MI.
</p>


<h3>References</h3>

<p><a href="http://www.icpsr.umich.edu/icpsrweb/NACJD/archive.jsp">http://www.icpsr.umich.edu/icpsrweb/NACJD/archive.jsp</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(ConvictionsAndIncome)
ConvictionsAndIncome

xtabs(~ convicted + income, data = ConvictionsAndIncome)
</code></pre>

<hr>
<h2 id='Crickets'>Immunity and Sperm Viability in Crickets</h2><span id='topic+Crickets'></span>

<h3>Description</h3>

<p>Sperm viability and immune function, measured by lysozyme activity in
crickets. Each observation is a mean for a single family of males.
</p>


<h3>Format</h3>

<p>A data frame with 41 observations on the following 2 variables.
</p>
 <dl>
<dt>sperm.viability</dt><dd><p>a numeric vector</p>
</dd>
<dt>lysozyme</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Simmons, L.W. and B. Roberts. 2005. Bacterial immunity traded for
sperm viability in male crickets. <em>Science</em> 309: 2031.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Crickets
xyplot(lysozyme ~ sperm.viability, Crickets)
</code></pre>

<hr>
<h2 id='DaphniaLongevity'>Daphnia Longevity</h2><span id='topic+DaphniaLongevity'></span>

<h3>Description</h3>

<p>Number of spores and host longevity in the crustacean <em>Daphnia magna</em>.
</p>


<h3>Format</h3>

<p>A data frame with 32 observations on the following 2 variables.
</p>
 <dl>
<dt>longevity</dt><dd><p>a numeric vector</p>
</dd>
<dt>sqrt.spores</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Jensen, K.H., T.J. Little, A. Skorping, and D. Ebert. 2006.
Empirical support for optimal virulence in a castrating parasite. <em>PLoS
Biology</em> 4(7): e197
</p>


<h3>References</h3>

<p><a href="http://www.plosbiology.org/article/info:doi/10.1371/journal.pbio.0040197">http://www.plosbiology.org/article/info:doi/10.1371/journal.pbio.0040197</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(DaphniaLongevity)
xyplot(sqrt.spores ~ longevity, DaphniaLongevity)
</code></pre>

<hr>
<h2 id='DaphniaResistance'>Daphnia Resistance to Cyanobacteria</h2><span id='topic+DaphniaResistance'></span>

<h3>Description</h3>

<p>Resistance of <em>Daphnia</em> eggs to different levels of cyanobacteria
(<code>cyandensity</code>) from 1962-1997.
</p>


<h3>Format</h3>

<p>A data frame with 32 observations on the following 2 variables.
</p>
 <dl>
<dt>density</dt><dd><p>a factor with levels: <code>high</code>,
<code>low</code>, and <code>med</code></p>
</dd> <dt>resistance</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> Hairston, N.G., Jr., W. Lampert, C.E. CÃ¡ceres,
C.L. Holtmeier, L.J. Weider, U. Gaedke, J.M. Fischer, J.A. Fox, and D.M.
Post. 1999. Dormant eggs record rapid evolution. <em>Nature</em> 401: 446.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(DaphniaResistance)


bwplot(resistance ~ density, DaphniaResistance)
# with such a small data set, we can display all the data
# rather than a summary
xyplot(resistance ~ density, DaphniaResistance)
histogram( ~ resistance| density, DaphniaResistance,
	strip=FALSE, strip.left = TRUE,
	layout=c(1,3)
	)
</code></pre>

<hr>
<h2 id='dataInfo'><code>abd</code> Data Sets</h2><span id='topic+dataInfo'></span>

<h3>Description</h3>

<p>Information about the location of data sets in <em>Analysis of Biological
Data</em>
</p>


<h3>Format</h3>

<p>A data frame with 143 observations on the following 5 variables.
</p>
 <dl>
<dt>name</dt><dd><p>name of data set</p>
</dd>
<dt>chapter</dt><dd><p>chapter in which data set appears</p>
</dd>
<dt>type</dt><dd><p>used in an <code>Example</code> or a <code>Problem</code></p>
</dd>
<dt>number</dt><dd><p>example or problem number</p>
</dd>
<dt>sub</dt><dd><p>sub-problem:  <code>a</code> <code>b</code> <code>c</code></p>
</dd> </dl>


<h3>See Also</h3>

<p><code><a href="#topic+abdData">abdData</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(dataInfo)
</code></pre>

<hr>
<h2 id='DayOfBirth'>Day of Birth</h2><span id='topic+DayOfBirth'></span>

<h3>Description</h3>

<p>Day of the week for 350 U.S. births in 1999.
</p>


<h3>Format</h3>

<p>A data frame with 7 observations on the following 2 variables.
</p>
 <dl>
<dt>day</dt><dd><p>a character vector</p>
</dd> <dt>births</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Ventura, S.J., J.A. Martin, S.C. Curtin, F. Menacker, and B.E.
Hamilton. 2001. Births: final data for 1999. <em>National Vital Statistics
Reports</em> Vol. 49, No. 1.
</p>


<h3>References</h3>

<p><a href="http://cdc.gov/NCHS/products/nvsr.htm">http://cdc.gov/NCHS/products/nvsr.htm</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>DayOfBirth
barchart( day ~ births, DayOfBirth, origin=0)

# Fix bad ordering of days
DayOfBirth$oday &lt;- with(DayOfBirth, ordered(day, levels = day))
barchart( oday ~ births, DayOfBirth, origin=0)
barchart( births ~ oday, DayOfBirth, horizontal = FALSE, origin=0)
barchart( births ~ oday, DayOfBirth, horizontal = FALSE, origin=0,
 scales = list(x=list(rot=45)))

barplot(DayOfBirth$births,
  ylim = c(0, 70),
  names.arg = DayOfBirth$day,
  las = 2,
  mgp = c(3, 0.75, 0))
</code></pre>

<hr>
<h2 id='DEET'>DEET and Mosquito Bites</h2><span id='topic+DEET'></span>

<h3>Description</h3>

<p>Administered dose of DEET and number of mosquito bites for 52 women.
</p>


<h3>Format</h3>

<p>A data frame with 52 observations on the following 2 variables.
</p>
 <dl>
<dt>dose</dt><dd><p>a numeric vector</p>
</dd> <dt>bites</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Golenda, C.F., V.B. Solberg, R. Burge, J.M. Gambel, and R.A. Wirtz.
1999. Gender-related efficacy difference to an extended duration formulation
of topical N,N-diethyl-<em>m</em>-toluamide (DEET). <em>American Journal of
Tropical Medicine and Hygiene</em> 60: 654-657.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(DEET)
xyplot(bites ~ dose, DEET)
</code></pre>

<hr>
<h2 id='DesertBirds'>Desert Bird Census Data</h2><span id='topic+DesertBirds'></span>

<h3>Description</h3>

<p>Census data for desert birds.
</p>


<h3>Format</h3>

<p>A data frame with 43 observations on the following 2 variables.
</p>
 <dl>
<dt>species</dt><dd><p>a character vector</p>
</dd> <dt>count</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Sauer, J.R., J.E. Hines, and J. Fallon. 2003. The North American
breeding bird survey, results and analysis 1966-2002. Version 2003.1. USGS
Patuxent Wildlife Research Center, Laurel, MD.
</p>


<h3>References</h3>

<p><a href="http://www.mbr-pwrc.usgs.gov/bbs/">http://www.mbr-pwrc.usgs.gov/bbs/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(DesertBirds)
histogram(~ count, DesertBirds,
  xlab = "Abundance"
  )
</code></pre>

<hr>
<h2 id='Dioecy'>Dioecy vs. Monomorphism in Plants</h2><span id='topic+Dioecy'></span>

<h3>Description</h3>

<p>Number of <code>dioecious</code> and <code>monomorphic</code> taxa among pairs of
closely related plants.
</p>


<h3>Format</h3>

<p>A data frame with 28 observations on the following 3 variables.
</p>
 <dl>
<dt>dioecious</dt><dd><p>a numeric vector</p>
</dd>
<dt>monomorphic</dt><dd><p>a numeric vector</p>
</dd>
<dt>taxon.pair</dt><dd><p>identifier for pair</p>
</dd> </dl>


<h3>Source</h3>

<p>Heilbuth, J.C. 2000. Lower species richness in dioecious clades.
<em>The American Naturalist</em> 156: 221-241.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(dioecious ~ monomorphic, Dioecy, alpha = 0.65, pch = 16)
</code></pre>

<hr>
<h2 id='Dolphins'>Dolphin Swimming Behavior</h2><span id='topic+Dolphins'></span>

<h3>Description</h3>

<p>Percentage of time 8 sleeping dolphins from the Southern Hemisphere spent
swimming clockwise.
</p>


<h3>Format</h3>

<p>A data frame with 8 observations on one variable. </p>

<dl>
<dt>percent.clockwise</dt><dd><p>percent of time spent swimming clockwise
while sleeping.</p>
</dd> </dl>


<h3>Source</h3>

<p>Stafne, G.M. and P.R. Manger. 2004. Predominance of clockwise
swimming during rest in Southern Hemisphere dolphins. <em>Physiology and
Behavior</em> 82: 919-926.
</p>


<h3>References</h3>

<p><a href="http://faculty.washington.edu/chudler/dolp.html">http://faculty.washington.edu/chudler/dolp.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Dolphins
hist(Dolphins$percent.clockwise)
histogram(~ percent.clockwise, Dolphins)
</code></pre>

<hr>
<h2 id='DungBeetles'>Heritability of Body Condition in Dung Beetles</h2><span id='topic+DungBeetles'></span>

<h3>Description</h3>

<p>Body condition (<code>offspring.condition</code>) in 36 dung beetles
(<em>Onthophagus taurus</em>) from 12 <code>male</code>s each mated to 3 different
virgin females.
</p>


<h3>Format</h3>

<p>A data frame with 36 observations on the following 2 variables.
</p>
 <dl>
<dt>id</dt><dd><p>a numeric vector</p>
</dd>
<dt>offspring.condition</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> Kotiaho, J.S., L.W. Simmons, and J.L. Tomkins.
2001. Towards a resolution of the lek paradox. <em>Nature</em> 410: 684-686.
</p>


<h3>References</h3>

<p><a href="http://en.wikipedia.org/wiki/Dung_beetle">http://en.wikipedia.org/wiki/Dung_beetle</a>
</p>
<p><a href="http://www.nature.com/nature/journal/v410/n6829/abs/410684a0.html">http://www.nature.com/nature/journal/v410/n6829/abs/410684a0.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(DungBeetles)
xyplot(offspring.condition ~ factor(id), DungBeetles,
  xlab='Dung Beetle',
  ylab='offspring condition')
</code></pre>

<hr>
<h2 id='Earthworms'>Earthworm Diversity and Soil Nitrogen Levels</h2><span id='topic+Earthworms'></span>

<h3>Description</h3>

<p>Number of earthworm species and total nitrogen content in the soil in 39
hardwood forest plots.
</p>


<h3>Format</h3>

<p>A data frame with 39 observations on the following 2 variables.
</p>
 <dl>
<dt>worm.species</dt><dd><p>a numeric vector</p>
</dd>
<dt>nitrogen</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Gundale, M.J., W.M. Jolly, and T.H. Deluca. 2005. Susceptibility of
a northern hardwood forest to exotic earthworm invasion. <em>Conservation
Biology</em> 19: 1075-1083.
</p>


<h3>References</h3>

<p><a href="http://www3.interscience.wiley.com/journal/118701215/abstract">http://www3.interscience.wiley.com/journal/118701215/abstract</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(Earthworms)
xyplot(nitrogen ~ worm.species, Earthworms)
</code></pre>

<hr>
<h2 id='Earwigs'>Earwig Density and Forceps</h2><span id='topic+Earwigs'></span>

<h3>Description</h3>

<p>Earwig (<em>Forficula auricularia</em>) density and the proportion of trapped
earwigs with abdominal forceps (used for fighting and courtship).
</p>


<h3>Format</h3>

<p>A data frame with 7 observations on the following 2 variables.
</p>
 <dl>
<dt>density</dt><dd><p>a numeric vector</p>
</dd>
<dt>proportion.forceps</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Tomkins, J.L.  and G.S. Brown. 2004. Population density drives the
local evolution of a threshold dimorphism. <em>Nature</em> 431: 1099-1103.
</p>


<h3>References</h3>

<p><a href="http://en.wikipedia.org/wiki/Forficula_auricularia">http://en.wikipedia.org/wiki/Forficula_auricularia</a>
</p>
<p><a href="http://www.arkive.org/common-european-earwig/forficula-auricularia/">http://www.arkive.org/common-european-earwig/forficula-auricularia/</a>
</p>
<p><a href="http://eol.org/pages/473785">http://eol.org/pages/473785</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(proportion.forceps ~ density, data=Earwigs, type='h', lwd=6)
</code></pre>

<hr>
<h2 id='Eelgrass'>Eelgrass Genotypes</h2><span id='topic+Eelgrass'></span>

<h3>Description</h3>

<p>Number of shoots (<code>shoots</code>) surviving in each of 32 experimental plots
planted with 1, 3, or 6 different genotypes of eelgrass
(<code>treatment.genotypes</code>).
</p>


<h3>Format</h3>

<p>A data frame with 32 observations on the following 2 variables.
</p>
 <dl>
<dt>genotypes</dt><dd><p>a numeric vector of the number of
genotypes planted in each plot</p>
</dd> <dt>shoots</dt><dd><p>a numeric vector of
the total number of shoots in each plot</p>
</dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> Reusch, T.B.H., A. Ehlers, A. HÃ¤mmerli, and B.
Worm. 2005. Ecosystem recovery after climatic extremes enhanced by genotypic
diversity. <em>Proceedings of the National Academy of Sciences (USA)</em> 102:
2826-2831.
</p>


<h3>References</h3>

<p><a href="http://www.pnas.org/content/102/8/2826.abstract">http://www.pnas.org/content/102/8/2826.abstract</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Eelgrass

# Convert treatment.genotypes to a factor
Eelgrass$genotypesF &lt;-
  factor(Eelgrass$genotypes)
str(Eelgrass)
xyplot(shoots ~ genotypes, Eelgrass)
xyplot(shoots ~ genotypesF, Eelgrass)
</code></pre>

<hr>
<h2 id='ElectricFish'>Electric Fish</h2><span id='topic+ElectricFish'></span>

<h3>Description</h3>

<p>Species abundance of electric fish upstream and downstream of the entrance
of a tributary in the Amazon basin.
</p>


<h3>Format</h3>

<p>A data frame with 12 observations on the following 3 variables.
</p>
 <dl>
<dt>tributary</dt><dd><p>a character vector</p>
</dd>
<dt>species.upstream</dt><dd><p>a numeric vector of the number of species of
electric fish present upstream of the tributary</p>
</dd>
<dt>species.downstream</dt><dd><p>a numeric vector of the number of species
of electric fish present downstream of the tributary</p>
</dd> </dl>


<h3>Source</h3>

<p>Fernandes, C.C., J. Podos, and J.G. Lundberg. 2004. Amazonian
ecology: tributaries enhance the diversity of electric fishes.
<em>Science</em> 305: 1960-1962.
</p>


<h3>References</h3>

<p><a href="http://www.sciencemag.org/cgi/content/abstract/305/5692/1960">http://www.sciencemag.org/cgi/content/abstract/305/5692/1960</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ElectricFish
require(grid)
xyplot(species.upstream ~ species.downstream, data = ElectricFish,
  panel=function(x, y, ...){
    grid.text(ElectricFish$tributary, x=x, y=y,
      rot = 45,
      gp = gpar(cex=.6),
      default.units = 'native')
    }
  )
</code></pre>

<hr>
<h2 id='ElVerde'>Diet Breadth in a Rainforest Community</h2><span id='topic+ElVerde'></span>

<h3>Description</h3>

<p>Number of different species (<code>breadth</code>) in 127 species
(<code>no.species</code>) in the rainforest community at El Verde, Puerto Rico
</p>


<h3>Format</h3>

<p>A data frame with 38 observations on the following 2 variables.
</p>
 <dl>
<dt>breadth</dt><dd><p>a numeric vector</p>
</dd>
<dt>num.species</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Waide R.B. and W.B. Reagan, eds. 1996. <em>The Food Web of a
Tropical Rainforest</em>. University of Chicago Press, Chicago.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ElVerde
xyplot(num.species ~ breadth, ElVerde, type='h',lwd=3)
</code></pre>

<hr>
<h2 id='EndangeredSpecies'>Endangered and Threatened Species</h2><span id='topic+EndangeredSpecies'></span>

<h3>Description</h3>

<p>Frequency of taxon groups on the U.S. Fish and Wildlife Service list of
endangered and threatened species (2002).
</p>


<h3>Format</h3>

<p>A data frame with 11 observations on the following 2 variables.
</p>
 <dl>
<dt>taxon</dt><dd><p>a character vector</p>
</dd>
<dt>num.species</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>U.S. Fish and Wildlife Service. 2001. Number of U.S. listed species
per calendar year.
</p>


<h3>References</h3>

<p><a href="http://www.fws.gov/endangered/">http://www.fws.gov/endangered/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(EndangeredSpecies)
EndangeredSpecies
</code></pre>

<hr>
<h2 id='FingerRatio'>2D:4D Finger Ratio</h2><span id='topic+FingerRatio'></span>

<h3>Description</h3>

<p>The ratio of the lengths of the index finger to the ring finger in 46 males
and the number of CAG repeats for each.
</p>


<h3>Format</h3>

<p>A data frame with 46 observations on the following 2 variables.
</p>
 <dl>
<dt>CAGrepeats</dt><dd><p>a numeric vector of the number of CAG
repeats</p>
</dd> <dt>finger.ratio</dt><dd><p>a numeric vector of the ratio of digit
2 to digit 4</p>
</dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> Manning, J.T., P.E. Bundred, D.J. Newton, and
B.F. Flanagan. 2003. The second to fourth digit ratio and variation in the
androgen receptor gene. <em>Evolution and Human Behavior</em> 24: 399-405.
</p>


<h3>References</h3>

<p><a href="http://en.wikipedia.org/wiki/Digit_ratio">http://en.wikipedia.org/wiki/Digit_ratio</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(FingerRatio)
xyplot(finger.ratio ~ CAGrepeats, FingerRatio,
  xlab = "Number of CAG Repeats",
  ylab = "2D:4D Ratio"
)
</code></pre>

<hr>
<h2 id='Fireflies'>Spermatophore Mass in Fireflies</h2><span id='topic+Fireflies'></span>

<h3>Description</h3>

<p>Measurements of spermatophore mass (milligrams) in 35 fireflies
(<em>Photinus ignitus</em>).
</p>


<h3>Format</h3>

<p>A data frame with 35 observations of one variable. </p>

<dl>
<dt>sp.mass</dt><dd></dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> Cratsley, C.K. and S.M. Lewis. 2003. Female
preference for male courtship flashes in <em>Photinus ignitus</em> fireflies.
<em>Behavioral Ecology</em> 14: 135-140.
</p>


<h3>References</h3>

<p><a href="http://beheco.oxfordjournals.org/cgi/content/abstract/14/1/135">http://beheco.oxfordjournals.org/cgi/content/abstract/14/1/135</a>
</p>
<p><a href="http://en.wikipedia.org/wiki/Firefly">http://en.wikipedia.org/wiki/Firefly</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(Fireflies)
histogram(~sp.mass, Fireflies, n=12)
</code></pre>

<hr>
<h2 id='FireflyFlash'>Firefly Flash Duration</h2><span id='topic+FireflyFlash'></span>

<h3>Description</h3>

<p>Flash duration (measured in milliseconds) of a sample of male fireflies
(<em>Photinus ignitus</em>; <em>n</em> = 35).
</p>


<h3>Format</h3>

<p>A data frame with 35 observations of one variable. </p>

<dl>
<dt>flash</dt><dd><p>duration of flash (milliseconds)</p>
</dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> Cratsley, C.K. and S.M. Lewis. 2003. Female
preference for male courtship flashes in <em>Photinus ignitus</em> fireflies.
<em>Behavioral Ecology</em> 14: 135-140.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(FireflyFlash)
histogram(~flash, FireflyFlash)
</code></pre>

<hr>
<h2 id='FlycatcherPatch'>Forehead Patch Size in Collared Flycatachers</h2><span id='topic+FlycatcherPatch'></span>

<h3>Description</h3>

<p>Forehead patch size in 30 male Collared Flycatachers measured in two
consecutive years.
</p>


<h3>Format</h3>

<p>A data frame with 30 observations on the following 2 variables.
</p>
 <dl>
<dt>patch98</dt><dd><p>a numeric vector</p>
</dd> <dt>patch99</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Griffith, S.C. and B.C. Sheldon. 2001. Phenotypic plasticity in the
expression of a sexually selected trait: neglected components of variation.
<em>Animal Behaviour</em> 61: 987-993.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(FlycatcherPatch)
xyplot(patch99 ~ patch98, FlycatcherPatch)
</code></pre>

<hr>
<h2 id='FlyTestes'>Testes Size in Flies</h2><span id='topic+FlyTestes'></span>

<h3>Description</h3>

<p>Testes size (square mm; <code>Testes.area</code>) in 8 populations of common
yellow dung flies (<em>Scathophaga stercoraria</em>) with different mating
systems (<code>Mating.system</code>).
</p>


<h3>Format</h3>

<p>A data frame with 8 observations on the following 2 variables.
</p>
 <dl>
<dt>mating</dt><dd><p>a factor with levels <code>Monogamous</code>
<code>Polyandrous</code></p>
</dd> <dt>testes.area</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Hosken, D.J. and P.I. Ward. 2001.  Experimental evidence for testis
size evolution via sperm competition. <em>Ecology Letters</em> 4: 10-13.
</p>


<h3>References</h3>

<p><a href="http://en.wikipedia.org/wiki/Scathophaga_stercoraria">http://en.wikipedia.org/wiki/Scathophaga_stercoraria</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(FlyTestes)
FlyTestes
</code></pre>

<hr>
<h2 id='GeneRegulation'>Gene Regulation in Saccharomyces</h2><span id='topic+GeneRegulation'></span>

<h3>Description</h3>

<p>Number of genes regulated by 109 regulatory genes of <em>Saccharomyces
cerevisiae</em>.
</p>


<h3>Format</h3>

<p>A data frame with 26 observations on the following 2 variables.
</p>
 <dl>
<dt>genes.regulated</dt><dd><p>a numeric vector</p>
</dd>
<dt>count</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Guelzim, N., S. Bottani, P. Bourgine and F. KÃ©pÃ¨s. 2002. Topological
and causal structure of the yeast transcriptional regulatory network.
<em>Nature Genetics</em> 31: 60-63.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(GeneRegulation)
xyplot(count ~ genes.regulated, GeneRegulation, type='h', lwd=3)
</code></pre>

<hr>
<h2 id='GlidingSnakes'>GlidingSnakes</h2><span id='topic+GlidingSnakes'></span>

<h3>Description</h3>

<p>Undulation rate (<em>Hz</em>) of 8 paradise tree snakes (<em>Chrysopelea
paradisi</em>).
</p>


<h3>Format</h3>

<p>A data frame with eight observations of one variable. </p>

<dl>
<dt>undulation.rate</dt><dd><p>undulation rate</p>
</dd> </dl>


<h3>Source</h3>

<p>Socha, J.J. 2002. Gliding flight in the paradise tree snake.
<em>Nature</em> 418: 603-604.
</p>


<h3>References</h3>

<p><a href="http://www.nature.com/nature/journal/v418/n6898/abs/418603a.html">http://www.nature.com/nature/journal/v418/n6898/abs/418603a.html</a>
</p>
<p><a href="http://www.flyingsnake.org/">http://www.flyingsnake.org/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>histogram(~undulation.rate , data=GlidingSnakes, n=7,
  xlab = "Undulation rate (Hz)",
  type='count')
</code></pre>

<hr>
<h2 id='GodwitArrival'>Godwit Arrival Dates</h2><span id='topic+GodwitArrival'></span>

<h3>Description</h3>

<p>Arrival dates for males and females in 10 pairs of Black-tailed Godwits
(<em>Limosa limosa</em>)
</p>


<h3>Format</h3>

<p>A data frame with 10 observations on the following 2 variables.
</p>
 <dl>
<dt>female</dt><dd><p>a numeric vector</p>
</dd> <dt>male</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Gunnarsson, T.G., J.A. Gill, T. SigurbjÃ¶rnsson, and W.J. Sutherland.
2004. Pair bonds: arrival synchrony in migratory birds. <em>Nature</em> 431:
646.
</p>


<h3>References</h3>

<p><a href="http://en.wikipedia.org/wiki/Black-tailed_godwit">http://en.wikipedia.org/wiki/Black-tailed_godwit</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(male~female, GodwitArrival, main='Arrival of Godwit pairs')
</code></pre>

<hr>
<h2 id='Grassland'>Grassland Diversity</h2><span id='topic+Grassland'></span>

<h3>Description</h3>

<p>Species diversity in 10 experimental plots in the Park Grass Experiment at
Rothamsted Experimental Station to which varying numbers of nutrients have
been added.
</p>


<h3>Format</h3>

<p>A data frame with 10 observations on the following 2 variables.
</p>
 <dl>
<dt>nutrients</dt><dd><p>a numeric vector</p>
</dd>
<dt>num.species</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Harpole, W. S. and D. Tilman. 2007. Grassland species loss due to
reduced niche dimension. <em>Nature</em> 446: 791-793.
</p>


<h3>References</h3>

<p><a href="http://www.rothamsted.ac.uk/">http://www.rothamsted.ac.uk/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(num.species ~ jitter(nutrients, amount=0.1), Grassland, pch=16)
</code></pre>

<hr>
<h2 id='GreatTitMalaria'>Malaria in Populations of Great Tit</h2><span id='topic+GreatTitMalaria'></span>

<h3>Description</h3>

<p>Two-by-two contingency table of malaria (<em>Plasmodium</em>) infection status
in control and egg-removal populations of Great Tit (<em>Parus major</em>).
</p>


<h3>Format</h3>

<p>A data frame with 65 observations on the following 2 variables.
</p>
 <dl>
<dt>treatment</dt><dd><p>a factor with levels <code>Control</code> and
<code>Egg removal</code></p>
</dd> <dt>response</dt><dd><p>a ordered factor with levels
<code>No Malaria</code> and <code>Malaria</code></p>
</dd> </dl>


<h3>Source</h3>

<p>Oppliger, A., P. Christe, and H. Richner. 1996. Clutch size and
malaria resistance. <em>Nature</em> 381: 565.
</p>


<h3>References</h3>

<p><a href="http://www.nature.com/nature/journal/v381/n6583/abs/381565a0.html">http://www.nature.com/nature/journal/v381/n6583/abs/381565a0.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(GreatTitMalaria)

table(GreatTitMalaria)

if(require(vcd)) {
  mosaic(~treatment + response, GreatTitMalaria)
}
</code></pre>

<hr>
<h2 id='Greenspace'>Diversity in Urban Green Space</h2><span id='topic+Greenspace'></span>

<h3>Description</h3>

<p>Measures of biodiversity in 15 urban green spaces in Sheffield, England.
</p>


<h3>Format</h3>

<p>A data frame with 15 observations on the following 6 variables.
</p>
 <dl>
<dt>site</dt><dd><p>a factor with levels <code>A</code> - <code>O</code></p>
</dd>
<dt>attachment</dt><dd><p>a numeric vector</p>
</dd> <dt>area</dt><dd><p>a numeric
vector</p>
</dd> <dt>butterfly</dt><dd><p>a numeric vector</p>
</dd> <dt>bird</dt><dd><p>a
numeric vector</p>
</dd> <dt>ln.plant</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Fuller, R.A., K.N. Irvine, P. Devine-Wright, P.H. Warren, and K.J.
Gaston. 2007. Psychological benefits of greenspace increase with
biodiversity. <em>Biology Letters</em> 3: 390-394.
</p>


<h3>References</h3>

<p><a href="http://rsbl.royalsocietypublishing.org/content/3/4/390.abstract">http://rsbl.royalsocietypublishing.org/content/3/4/390.abstract</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(Greenspace)
splom(Greenspace[,2:6])
</code></pre>

<hr>
<h2 id='Guppies'>Ornamentation and Attractiveness in Guppies</h2><span id='topic+Guppies'></span>

<h3>Description</h3>

<p>The father's ornamentation (composite score of color and brightness) and
son's attractiveness (relative rates of visits by females) in male guppies
(<em>Poecilia reticulata</em>).
</p>


<h3>Format</h3>

<p>A data frame with 36 observations on the following 2 variables.
</p>
 <dl>
<dt>father.ornament</dt><dd><p>a numeric vector</p>
</dd>
<dt>son.attract</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> Brooks, R. 2000. Negative genetic correlation
between male sexual attractiveness and survival. <em>Nature</em> 406: 67-70.
</p>


<h3>References</h3>

<p><a href="http://www.nature.com/nature/journal/v406/n6791/abs/406067a0.html">http://www.nature.com/nature/journal/v406/n6791/abs/406067a0.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(Guppies)
xyplot(son.attract ~ father.ornament,
  Guppies,
  xlab = "Father's ornamentation",
  ylab = "Son's attractiveness"
  )
</code></pre>

<hr>
<h2 id='Hemoglobin'>Hemoglobin Levels in High Altitude Populations</h2><span id='topic+Hemoglobin'></span>

<h3>Description</h3>

<p>Relative rates of hemoglobin concentration in four populations of humans
living at different altitudes.
</p>


<h3>Format</h3>

<p>A data frame with 40 observations on the following 3 variables.
</p>
 <dl>
<dt>hemoglobin</dt><dd><p>a numeric vector</p>
</dd>
<dt>group</dt><dd><p>a factor with levels: <code>Andes</code>, <code>Ethiopia</code>,
<code>Tibet</code>, and <code>USA</code></p>
</dd> <dt>relative.frequency</dt><dd><p>a numeric
vector</p>
</dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> Beall, C.M., M.J. Decker, G.M. Bittenham, I.
Kushner, A. Gebremedhin, K.P. Strohl. 2002. An Ethiopian pattern of human
adaptation to high-altitude hypoxia. <em>Proceeding of the National
Academy of Sciences (USA)</em> 99(26): 17215-17218.
</p>


<h3>References</h3>

<p><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC139295/">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC139295/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(Hemoglobin)

xyplot(relative.frequency ~ hemoglobin | group, Hemoglobin,
  type ='h', lwd=4, layout=c(1,4))
</code></pre>

<hr>
<h2 id='HippocampusLesions'>Memory and the Hippocampus</h2><span id='topic+HippocampusLesions'></span>

<h3>Description</h3>

<p>Spatial memory score (<code>memory</code>) and percent lesion of the hippocampus
(<code>lesion</code>).
</p>


<h3>Format</h3>

<p>A data frame with 57 observations on the following 2 variables.
</p>
 <dl>
<dt>lesion</dt><dd><p>a numeric vector</p>
</dd> <dt>memory</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Broadbent, N.J., L.R. Squire, and R.E. Clark. 2004. Spatial memory,
recognition memory, and the hippocampus. <em>Proceedings of the National
Academy of Sciences (USA)</em> 101: 14515-14520.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>HippocampusLesions

xyplot(memory ~ lesion, data = HippocampusLesions,
  pch = 16, col = "red")

plot(memory ~ lesion, data = HippocampusLesions,
  pch = 16, col = "red")
</code></pre>

<hr>
<h2 id='histochart'>Histogram from tabulated data</h2><span id='topic+histochart'></span>

<h3>Description</h3>

<p>Uses lattice::barchart() to build a histogram from tabulated data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>histochart(x, data = NULL, box.ratio = 100, origin = 0,
  horizontal = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="histochart_+3A_x">x</code></td>
<td>
<p>formula of form <code>frequency ~ value</code></p>
</td></tr>
<tr><td><code id="histochart_+3A_data">data</code></td>
<td>
<p>data frame in which the formula <code>x</code> is interpreted</p>
</td></tr>
<tr><td><code id="histochart_+3A_box.ratio">box.ratio</code></td>
<td>
<p>ratio of bar widths to gaps between bars</p>
</td></tr>
<tr><td><code id="histochart_+3A_origin">origin</code></td>
<td>
<p>where do bars begin?</p>
</td></tr>
<tr><td><code id="histochart_+3A_horizontal">horizontal</code></td>
<td>
<p>Should bars go horizontal?</p>
</td></tr>
<tr><td><code id="histochart_+3A_...">...</code></td>
<td>
<p>other arguments passed to lattice::barchart()</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is just a convenience wrapper around lattice::barchart().
</p>


<h3>Author(s)</h3>

<p>Randall Pruim (<a href="mailto:rpruim@calvin.edu">rpruim@calvin.edu</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>histochart( dbinom(0:30, 30, 0.35) ~ 0:30 )
</code></pre>

<hr>
<h2 id='HornedLizards'>Horn Length and Predation Status of Horned Lizards</h2><span id='topic+HornedLizards'></span>

<h3>Description</h3>

<p>Squamosal horn length (mm; <code>horn.length</code>) and predation status
(<code>group</code>; living or killed) for 184 horned lizards (<em>Phrynosoma
mcalli</em>).
</p>


<h3>Format</h3>

<p>A data frame with 184 observations on the following 2 variables.
</p>
 <dl>
<dt>horn.length</dt><dd><p>a numeric vector</p>
</dd>
<dt>group</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Young, K.V., E.D. Brodie, Jr., and E.D. Brodie, III. 2004. How the
horned lizard got its horns. <em>Science</em> 304: 65.
</p>


<h3>References</h3>

<p><a href="http://www.sciencemag.org/cgi/pdf_extract/304/5667/65">http://www.sciencemag.org/cgi/pdf_extract/304/5667/65</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(HornedLizards)

histogram(~horn.length | group, HornedLizards,
  layout=c(1,2),
  xlab="Horn Length (mm)")
</code></pre>

<hr>
<h2 id='HumanBodyTemp'>Human Body Temperature</h2><span id='topic+HumanBodyTemp'></span>

<h3>Description</h3>

<p>Body temperature for 25 randomly chosen health people
</p>


<h3>Format</h3>

<p>A data frame with 25 observations of one variable. </p>

<dl>
<dt>temp</dt><dd><p>body temperature (degrees F)</p>
</dd> </dl>


<h3>Source</h3>

<p>Shoemaker, A. L. 1996. What's normal? &ndash; Temperature, gender, and
heart rate. <em>Journal of Statistics Education</em> 4(2).
</p>


<h3>References</h3>

<p><a href="http://www.amstat.org/publications/jse/v4n2/datasets.shoemaker.html">http://www.amstat.org/publications/jse/v4n2/datasets.shoemaker.html</a>
</p>
<p>Mackowiak, P.A., Wasserman, S.S., and Levine, M.M. 1992. A critical
appraisal of 98.6 degrees F, the upper limit of the normal body temperature,
and other legacies of Carl Reinhold August Wunderlich. <em>Journal of the
American Medical Association</em> 268: 1578-1580.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>histogram(~temp, HumanBodyTemp)
stem(HumanBodyTemp$temp, scale = 2)
favstats(HumanBodyTemp$temp)
</code></pre>

<hr>
<h2 id='HumanGeneLengths'>Human Gene Lengths</h2><span id='topic+HumanGeneLengths'></span>

<h3>Description</h3>

<p>Lengths in number of nucleotides (<code>gene.length</code>) for 20,290 human genes
</p>


<h3>Format</h3>

<p>A data frame with 20,290 observations on the following variable.
</p>
 <dl>
<dt>gene.length</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Hubbard, T., D. Andrews, M. Caccamo, G. Cameron, Y. Chen, M. Clamp,
L. Clarke, G. Coates, T. Cox, F. Cunningham, V. Curwen, T. Cutts, T. Down,
R. Durbin, X. M. Fernandez-Suarez, J. Gilbert, M. Hammond, J. Herrero, H.
Hotz, K. Howe, V. Iyer, K. Jekosch, A. Kahari, A. Kasprzyk, D. Keefe, S.
Keenan, F. Kokocinsci, D. London, I. Longden, G. McVicker, C. Melsopp, P.
Meidl, S. Potter, G. Proctor, M. Rae, D. Rios, M. Schuster, S. Searle, J.
Severin, G. Slater, D. Smedley, J. Smith, W. Spooner, A. Stabenau, J.
Stalker, R. Storey, S. Trevanion, A. Ureta-Vidal, J. Vogel, S. White, C.
Woodwark, and E. Birne. 2005. Ensembl 2005. <em>Nucleic Acids Research</em>
33: D447-D453.
</p>


<h3>References</h3>

<p><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC540092/">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC540092/</a>
</p>
<p><a href="http://www.ensembl.org/">http://www.ensembl.org/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(HumanGeneLengths)
histogram(~ gene.length, HumanGeneLengths,
          subset = gene.length &lt; 15000)
</code></pre>

<hr>
<h2 id='Hurricanes'>Intense Hurricanes</h2><span id='topic+Hurricanes'></span>

<h3>Description</h3>

<p>Number of hurricanes greater than or equal to category 3 on the
Safir-Simpson scale during the 20th century.
</p>


<h3>Format</h3>

<p>A data frame with 4 observations on the following 2 variables.
</p>
 <dl>
<dt>hurricanes</dt><dd><p>a numeric vector</p>
</dd>
<dt>count</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Blake, E.S., E.N. Rappaport, J.D. Jarrell, and C.W. Landsea. 2005.
The deadliest, costliest, and most intense United States tropical cyclones
from 1851 to 2006 (and other frequently requested hurricane facts).
<em>NOAA Technical Memorandum NWS TPC-4</em>.
</p>


<h3>References</h3>

<p><a href="http://www.aoml.noaa.gov/hrd/Landsea/Blakeetal_noaamemoApr2007.pdf">http://www.aoml.noaa.gov/hrd/Landsea/Blakeetal_noaamemoApr2007.pdf</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Hurricanes
</code></pre>

<hr>
<h2 id='Iguanas'>Iguana Body Length Changes</h2><span id='topic+Iguanas'></span>

<h3>Description</h3>

<p>Body size change in 64 GalÃ¡pagos marine iguanas (<em>Amblyrhynchus
cristatus</em>) that survived the 1992-1993 El NiÃ±o event.
</p>


<h3>Format</h3>

<p>A data frame with 64 observations of one variable. </p>

<dl>
<dt>change.in.length</dt><dd></dd> </dl>


<h3>Source</h3>

<p>Wikelski, M. and C. Thom. 2000. Marine iguanas shrink to survive El
NiÃ±o. <em>Nature</em> 403: 37-38.
</p>


<h3>References</h3>

<p><a href="http://en.wikipedia.org/wiki/Marine_iguana">http://en.wikipedia.org/wiki/Marine_iguana</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(Iguanas)
histogram(~ change.in.length, Iguanas, n = 10)
</code></pre>

<hr>
<h2 id='IntertidalAlgae'>Intertidal Algae</h2><span id='topic+IntertidalAlgae'></span>

<h3>Description</h3>

<p>Area coverage of red algae (<em>Mazzaella parksii</em>) in two herbivore
treatments (<code>herbivores</code>) at two tide levels (<code>height</code>).
</p>


<h3>Format</h3>

<p>A data frame with 64 observations on the following 3 variables.
</p>
 <dl>
<dt>height</dt><dd><p>a factor with levels <code>low</code> and
<code>mid</code></p>
</dd> <dt>herbivores</dt><dd><p>a factor with levels <code>minus</code> and
<code>plus</code></p>
</dd> <dt>sqrt.area</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Harley, C.D.G. 2003. Individualistic vertical responses of
interacting species determine range limits across a horizontal gradient.
<em>Ecology</em> 84: 1477-1488.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(IntertidalAlgae)

# Using * includes the main effects and the interaction
aov.fit &lt;- aov(sqrt.area ~ herbivores * height, data = IntertidalAlgae)
summary(aov.fit)
lm.fit &lt;- lm(sqrt.area ~ herbivores * height, data = IntertidalAlgae)
anova(lm.fit)
</code></pre>

<hr>
<h2 id='JetLagKnees'>Circadian Rhythm Phase Shift</h2><span id='topic+JetLagKnees'></span>

<h3>Description</h3>

<p>Shift in circadian rhythm (hours; <code>shift</code>) in three light treatments
(<code>treatment</code>).
</p>


<h3>Format</h3>

<p>A data frame with 22 observations on the following 2 variables.
</p>
 <dl>
<dt>treatment</dt><dd><p>a factor with levels <code>control</code>,
<code>eyes</code>, and <code>knee</code></p>
</dd> <dt>shift</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> Wright, K.P., Jr. and C.A. Czeisler 2002.
Absence of circadian phase resetting in response to bright light behind the
knees. <em>Science</em> 297: 571.
</p>


<h3>References</h3>

<p><a href="http://www.sciencemag.org/cgi/content/full/297/5581/571">http://www.sciencemag.org/cgi/content/full/297/5581/571</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>demo(sec15.1)
</code></pre>

<hr>
<h2 id='KenyaFinches'>Body Mass and Beak Length in Three Species of Finches in Kenya</h2><span id='topic+KenyaFinches'></span>

<h3>Description</h3>

<p>Data on body mass and beak length in three species of finches:
Crimson-rumped waxbill (<code>CRU.WAXB</code>), Cutthroat finch (<code>CUTTHROA</code>),
and White-browed sparrow weaver (<code>WB.SPARW</code>).
</p>


<h3>Format</h3>

<p>A data frame with 45 observations on the following 3 variables.
</p>
 <dl>
<dt>species</dt><dd><p>a factor with levels: <code>CRU.WAXB</code>,
<code>CUTTHROA</code>, and <code>WB.SPARW</code></p>
</dd> <dt>mass</dt><dd><p>mass (g)</p>
</dd>
<dt>beak.length</dt><dd><p>beak length (mm)</p>
</dd> </dl>


<h3>Source</h3>

<p>Schluter, D. 1988. The evolution of finch communities on islands and
continents: Kenya vs. Galapagos. <em>Ecological Monographs</em> 58: 229-249.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>table(KenyaFinches$species)
xyplot(beak.length ~ species, KenyaFinches)
bwplot(beak.length ~ species, KenyaFinches)
</code></pre>

<hr>
<h2 id='LanguageBrains'>Brain Structure in Bilingual Humans</h2><span id='topic+LanguageBrains'></span>

<h3>Description</h3>

<p>Proficiency score (summary of reading, writing, and speech) in subjects'
second language and density of gray matter in the left inferior parietal
region.
</p>


<h3>Format</h3>

<p>A data frame with 22 observations on the following 2 variables.
</p>
 <dl>
<dt>proficiency</dt><dd><p>a numeric vector</p>
</dd>
<dt>greymatter</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Mechelli, A., J.T. Crinion, U. Noppeney, J. O'Doherty, J. Ashburner,
R.S. Frackowiak, and C.J. Price. 2004. Structural plasticity in the
bilingual brain. <em>Nature</em> 431: 757.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(LanguageBrains)
xyplot(proficiency ~ greymatter, LanguageBrains)
</code></pre>

<hr>
<h2 id='LarvalFish'>Exploited Larval Fish</h2><span id='topic+LarvalFish'></span>

<h3>Description</h3>

<p>Age (<code>age</code>) and coefficient of variation (<code>cv</code>) in larval fish
from exploited and unexploited species (<code>exploited</code>).
</p>


<h3>Format</h3>

<p>A data frame with 28 observations on the following 3 variables.
</p>
 <dl>
<dt>age</dt><dd><p>a numeric vector</p>
</dd> <dt>cv</dt><dd><p>a numeric
vector</p>
</dd> <dt>exploited</dt><dd><p>a factor with levels <code>no</code> and
<code>yes</code></p>
</dd> </dl>


<h3>Source</h3>

<p>Hsieh, C.H., C.S. Reiss, J.R. Hunter, J.R. Beddington, R.M. May, and
G. Sugihara. 2006. Fishing elevates variability in the abundance of
exploited species. <em>Nature</em> 443: 859-862.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(LarvalFish)
xyplot(cv ~ age | exploited, LarvalFish)
xyplot(cv ~ age, groups=exploited, LarvalFish)
</code></pre>

<hr>
<h2 id='Lefthanded'>Left-handedness and Rates of Violence</h2><span id='topic+Lefthanded'></span>

<h3>Description</h3>

<p>Prevalence of left-handedness (<code>percent.left</code>) and homicide rates
(<code>murder</code>) for 8 societies.
</p>


<h3>Format</h3>

<p>A data frame with 8 observations on the following 2 variables.
</p>
 <dl>
<dt>percent.left</dt><dd><p>a numeric vector</p>
</dd>
<dt>murder.rate</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Faurie, C. and M. Raymond. 2005. Handedness, homicide and negative
frequency-dependent selection. <em>Proceedings of the Royal Society of
London B</em> 272: 25-28.
</p>


<h3>References</h3>

<p><a href="http://rspb.royalsocietypublishing.org/content/272/1558/25.abstract">http://rspb.royalsocietypublishing.org/content/272/1558/25.abstract</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(Lefthanded)
xyplot(murder.rate ~ percent.left, Lefthanded)
</code></pre>

<hr>
<h2 id='LionCubs'>Time to Reproduction in Female Lions</h2><span id='topic+LionCubs'></span>

<h3>Description</h3>

<p>Time to reproduction (<code>Days</code>) based on whether death of previous cubs
was due to infanticide (<code>New</code>) or accidental (<code>Same</code>).
</p>


<h3>Format</h3>

<p>A data frame with 14 observations on the following 2 variables.
</p>
 <dl>
<dt>cause.of.death</dt><dd><p>a factor with <code>accident</code> and
<code>infanticide</code></p>
</dd> <dt>days.to.next.cub</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Packer, C. and A.E. Pusey. 1983. Adaptations of female lions to
infanticide by incoming males. <em>The American Naturalist</em> 121: 716-728.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(days.to.next.cub ~ cause.of.death, LionCubs)
</code></pre>

<hr>
<h2 id='LionNoses'>Lion Age and Nose Coloration</h2><span id='topic+LionNoses'></span>

<h3>Description</h3>

<p>Ages (in years; <code>age</code>) of 32 male lions and relative coloration of
their noses (<code>proportion.black</code>).
</p>


<h3>Format</h3>

<p>A data frame with 32 observations on the following 2 variables.
</p>
 <dl>
<dt>age</dt><dd><p>a numeric vector</p>
</dd>
<dt>proportion.black</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Whitman, K., A.M. Starfield, H.S. Quadling and C. Packer. 2004.
Sustainable trophy hunting of African lions. <em>Nature</em> 428: 175-178.
</p>


<h3>References</h3>

<p><a href="http://www.nature.com/nature/journal/v428/n6979/abs/nature02395.html">http://www.nature.com/nature/journal/v428/n6979/abs/nature02395.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(age ~ proportion.black, LionNoses)
</code></pre>

<hr>
<h2 id='LiverPreparation'>Liver Preparation</h2><span id='topic+LiverPreparation'></span>

<h3>Description</h3>

<p>The unbound fraction of taurocholate for each of five concentrations of
administered taurocholate.
</p>


<h3>Format</h3>

<p>A data frame with 5 observations on the following 2 variables.
</p>
 <dl>
<dt>concentration</dt><dd><p>a numeric vector</p>
</dd>
<dt>unbound.fraction</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Smallwood, R.H., D.J. Morgan, G.W. Mihaly, and R.A. Smallwood. 1998.
Effect of a protein binding change on unbound and total plasma
concentrations for drugs of intermediate hepatic extraction. <em>Journal
of Pharmacokinetics and Pharmacodynamics</em> 16:397-411.
</p>


<h3>References</h3>

<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/3199317">http://www.ncbi.nlm.nih.gov/pubmed/3199317</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(LiverPreparation)
xyplot(unbound.fraction ~ concentration, LiverPreparation)
</code></pre>

<hr>
<h2 id='LizardBite'>Bite Force in Collard Lizards</h2><span id='topic+LizardBite'></span>

<h3>Description</h3>

<p>Bite force (N) and territory area in 11 male collared lizards
(<em>Crotaphytus collaris</em>).
</p>


<h3>Format</h3>

<p>A data frame with 11 observations on the following 2 variables.
</p>
 <dl>
<dt>bite</dt><dd><p>force of bite (N)</p>
</dd>
<dt>territory</dt><dd><p>area of territory</p>
</dd> </dl>


<h3>Note</h3>

<p>In the original publication (Lappin and Husak,
2005; Figure 3A), the data are presented in log-10 units. The
data in <code>LizardBite</code> and in chapter 17, question 9 was
back-transformed using <em>e</em> (i.e., <code>exp()</code>). To recover the data
from the original publication, use <code>10^(log(LizardBite$territory))</code>
and <code>10^(log(LizardBite$bite))</code>.
</p>


<h3>Source</h3>

<p>Lappin, A. K., and J. F. Husak. 2005. Weapon performance, not size,
determines mating success and potential reproductive output in the collared
lizard (<em>Crotaphytus collaris</em>). <em>The American Naturalist</em> 166:
426-436.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(LizardBite)
xyplot(territory ~ bite, LizardBite)
</code></pre>

<hr>
<h2 id='LizardSprint'>Sprint Speeds in Canyon Lizards</h2><span id='topic+LizardSprint'></span>

<h3>Description</h3>

<p>Sprint speeds (<code>speed</code>) in 34 canyon lizards
(<em>Sceloporous merriami</em>) measured in
successive years in Big Bend National Park. Note
that <code>lizard</code> is not coded as a factor.
</p>


<h3>Format</h3>

<p>A data frame with 68 observations on the following 2 variables.
</p>
 <dl>
<dt>lizard</dt><dd><p>a numeric vector</p>
</dd> <dt>speed</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> Huey, R.B. and A.E. Dunham. 1987. The
repeatability of locomotor performance in natural populations of the lizard
<em>Sceloporus merriami</em>. <em>Evolution</em> 42: 1116-1120.
</p>


<h3>References</h3>

<p><a href="http://en.wikipedia.org/wiki/Sceloporus_merriami">http://en.wikipedia.org/wiki/Sceloporus_merriami</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>histogram(~ speed, LizardSprint)
Lizard2 &lt;- aggregate(speed ~ lizard, LizardSprint, mean)
histogram(~ speed, Lizard2)
</code></pre>

<hr>
<h2 id='Lobsters'>Lobster Orientation</h2><span id='topic+Lobsters'></span>

<h3>Description</h3>

<p>Orientation of 15 lobsters relative to initial position.
</p>


<h3>Format</h3>

<p>A data frame with 15 observations of one variable. </p>

<dl>
<dt>orientation</dt><dd></dd> </dl>


<h3>Source</h3>

<p>Boles, L.C. and K.J. Lohmann. 2003. True navigation and magnetic
maps in spiny lobsters. <em>Nature</em> 421: 60-63.
</p>


<h3>References</h3>

<p><a href="http://www.unc.edu/depts/geomag/PDFGeomag/BolesandLohmann2003.pdf">http://www.unc.edu/depts/geomag/PDFGeomag/BolesandLohmann2003.pdf</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>histogram(~ orientation, Lobsters)
dotplot(~ orientation, Lobsters)
</code></pre>

<hr>
<h2 id='LodgepolePines'>Lodgepole Pine Cone Masses</h2><span id='topic+LodgepolePines'></span>

<h3>Description</h3>

<p>Masses of cones of lodgepole pines (<code>conemass</code>) from 16 different
habitat types (<code>habitat</code>) in western North America.
</p>


<h3>Format</h3>

<p>A data frame with 16 observations on the following 4 variables.
</p>
 <dl>
<dt>habitat</dt><dd><p>a factor with levels: <code>island
absent</code>, <code>island present</code>, and <code>mainland present</code></p>
</dd>
<dt>conemass</dt><dd><p>mass of cone</p>
</dd> <dt>location</dt><dd><p><code>island</code>
or <code>mainland</code></p>
</dd> <dt>squirrels</dt><dd><p><code>absent</code> or
<code>present</code></p>
</dd> </dl>


<h3>Source</h3>

<p>Edelaar, P. and C.W. Benkman. 2006. Replicated population divergence
caused by localised coevolution? A test of three hypotheses in the Red
Crossbill-lodgepole pine system. <em>Journal of Evolutionary Biology</em> 19:
1651-1659.
</p>


<h3>References</h3>

<p><a href="http://en.wikipedia.org/wiki/Lodgepole_pine">http://en.wikipedia.org/wiki/Lodgepole_pine</a>
</p>
<p><a href="http://en.wikipedia.org/wiki/Red_crossbill">http://en.wikipedia.org/wiki/Red_crossbill</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>LodgepolePines
str(LodgepolePines)
xyplot(conemass ~ habitat, LodgepolePines)
</code></pre>

<hr>
<h2 id='LupusMice'>Autoimmune Reactivity in Lupus-prone Mice</h2><span id='topic+LupusMice'></span>

<h3>Description</h3>

<p>Autoimmune reactivity (<code>dilution</code> at which reactivity could be
detected) in three <code>treatment</code>s of lupus-prone mice.
</p>


<h3>Format</h3>

<p>A data frame with 20 observations on the following 2 variables.
</p>
 <dl>
<dt>treatment</dt><dd><p>a factor with levels: <code>enhanced</code>,
<code>sham</code>, and <code>untreated</code></p>
</dd> <dt>dilution</dt><dd><p>a numeric vector
of the dilution level at which reactivity could be detected</p>
</dd> </dl>


<h3>Source</h3>

<p>McGaha, T.L., B. Sorrentino, and J.V. Ravetch. 2005. Restoration of
tolerance in lupus by targeted inhibitory receptor expression.
<em>Science</em> 307: 590-593.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(LupusMice)
</code></pre>

<hr>
<h2 id='Lynx'>Population Cycles of Lynx in Canada 1752-1819</h2><span id='topic+Lynx'></span>

<h3>Description</h3>

<p>Number of lynx pelts (<code>pelts</code>) reported in Canada per year from 1752 to
1819.
</p>


<h3>Format</h3>

<p>A data frame with 68 observations on the following 2 variables.
</p>
 <dl>
<dt>year</dt><dd><p>a numeric vector</p>
</dd> <dt>pelts</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Elton, C. and M. Nicholson. 1942. The ten-year cycle in numbers of
the lynx in Canada. <em>Journal of Animal Ecology</em> 11: 215-244.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(pelts ~ year, Lynx, type=c('p','l'))
</code></pre>

<hr>
<h2 id='MarineReserve'>Marine Reserve Biomass</h2><span id='topic+MarineReserve'></span>

<h3>Description</h3>

<p>Relative biomass in 32 marine reserves.
</p>


<h3>Format</h3>

<p>A data frame with 32 observations of one variable.  </p>

<dl>
<dt>biomass.ratio</dt><dd></dd> </dl>


<h3>Source</h3>

<p>Halpern, B.S. 2003. The impact of marine reserves: do reserves work
and does reserve size matter? <em>Ecological Applications</em> 13: S117-S137.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(MarineReserve)
histogram(~ biomass.ratio, MarineReserve)
</code></pre>

<hr>
<h2 id='MassExtinctions'>Mass Extinction Frequency</h2><span id='topic+MassExtinctions'></span>

<h3>Description</h3>

<p>The frequency of mass extinctions in the fossil record.
</p>


<h3>Format</h3>

<p>A data frame with 21 observations on the following 2 variables.
</p>
 <dl>
<dt>num.extinctions</dt><dd><p>a numeric vector</p>
</dd>
<dt>count</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Raup, D.M. and J.J. Sepkoski, Jr. 1982. Mass extinctions in the
marine fossil record. <em>Science</em> 215: 1501-1503.
</p>


<h3>References</h3>

<p><a href="http://www.sciencemag.org/cgi/content/abstract/sci;215/4539/1501">http://www.sciencemag.org/cgi/content/abstract/sci;215/4539/1501</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>MassExtinctions
</code></pre>

<hr>
<h2 id='MoleRats'>Energy Expenditure in Mole Rats</h2><span id='topic+MoleRats'></span>

<h3>Description</h3>

<p>Energy expenditure (<code>ln.energy</code>) in two castes (<code>caste</code>) of
Damaraland mole rats (<em>Cryptomys damarensis</em>) with body mass
(<code>ln.mass</code>) as a covariate.
</p>


<h3>Format</h3>

<p>A data frame with 35 observations on the following 3 variables.
</p>
 <dl>
<dt>caste</dt><dd><p>a factor with levels <code>lazy</code> and
<code>worker</code></p>
</dd> <dt>ln.mass</dt><dd><p>a numeric vector</p>
</dd>
<dt>ln.energy</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> Scantlebury, M., J.R. Speakman, M.K.
Oosthuizen, T.J. Roper and N.C. Bennett. 2006. Energetics reveals
physiologically distinct castes in a eusocial mammal. <em>Nature</em> 440:
795-797.
</p>


<h3>References</h3>

<p><a href="http://www.nature.com/nature/journal/v440/n7085/abs/nature04578.html">http://www.nature.com/nature/journal/v440/n7085/abs/nature04578.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>MoleRats
</code></pre>

<hr>
<h2 id='Mosquitoes'>Body Size in Anopheles Mosquitoes</h2><span id='topic+Mosquitoes'></span>

<h3>Description</h3>

<p>Weights of female and male mosquitos (<em>Anopheles darlingi</em>)
</p>


<h3>Format</h3>

<p>A data frame with 20 observations on the following 2 variables.
</p>
 <dl>
<dt>weight</dt><dd><p>a numeric vector</p>
</dd> <dt>sex</dt><dd><p>a
factor with levels <code>female</code> and <code>male</code></p>
</dd> </dl>


<h3>Source</h3>

<p>Lounibos, L.P., N. Nishimura, J. Conn, and R. Lourenco-de-Oliveira.
1995. Life history correlates of adult size in the malaria vector
<em>Anopheles darlingi</em>. <em>MemÃ³rias do Instituto Oswaldo Cruz</em> 90:
769-774.
</p>


<h3>References</h3>

<p><a href="http://www.bioline.org.br/request?oc95154">http://www.bioline.org.br/request?oc95154</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(weight ~ sex, Mosquitoes)
</code></pre>

<hr>
<h2 id='MouseEmpathy'>Mouse Empathy</h2><span id='topic+MouseEmpathy'></span>

<h3>Description</h3>

<p>Percentage of time spent stretching in three treatments of mice. Both
<code>condition</code> and <code>treatment</code> code for the same variable.
</p>


<h3>Format</h3>

<p>A data frame with 42 observations on the following 3 variables.
</p>
 <dl>
<dt>treatment</dt><dd><p>a factor with levels <code>Both
Writhing</code>, <code>Isolated</code>, and <code>One Writhing</code></p>
</dd>
<dt>percent.stretching</dt><dd><p>a numeric vector</p>
</dd> <dt>trt</dt><dd><p>a
factor with levels <code>bw</code>, <code>isolated</code>, and <code>ow</code></p>
</dd> </dl>


<h3>Source</h3>

<p>Langford, D.J., S.E. Crager, Z. Shehzah, S.B. Smith, S.G. Sotocinal,
J.S. Levenstadt, M.L. Chande, D.J. Levitin, J.S. Mogill. 2006. Social
modulation of pain as evidence for empathy in mice. <em>Science</em> 312:
1967-1970.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(MouseEmpathy)

aov.fit &lt;- aov(percent.stretching ~ treatment, data = MouseEmpathy)
summary(aov.fit)
lm.fit &lt;- lm(percent.stretching ~ treatment, data = MouseEmpathy)
anova(lm.fit)
</code></pre>

<hr>
<h2 id='NeanderthalBrains'>Cranial Capacity in Neanderthals and Modern Humans</h2><span id='topic+NeanderthalBrains'></span>

<h3>Description</h3>

<p>Brain size (<code>lnbrain</code>) and body mass (<code>lnmass</code>) in Neanderthals
and early modern humans (<code>species</code>).
</p>


<h3>Format</h3>

<p>A data frame with 39 observations on the following 3 variables.
</p>
 <dl>
<dt>ln.mass</dt><dd><p>log of body mass (kg)</p>
</dd>
<dt>ln.brain</dt><dd><p>log of brain size</p>
</dd> <dt>species</dt><dd><p>a factor
with levels <code>neanderthal</code> <code>recent</code></p>
</dd> </dl>


<h3>Source</h3>

<p>Ruff, C.B., E. Trinkaus, and T.W. Holliday. 1997. Body mass and
encephalization in Pleistocene <em>Homo</em>. <em>Nature</em> 387: 173-176.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(ln.brain ~ ln.mass, data=NeanderthalBrains, groups=species)
</code></pre>

<hr>
<h2 id='NematodeLifespan'>Effects of Trimethadione on Lifespan in Nematodes</h2><span id='topic+NematodeLifespan'></span>

<h3>Description</h3>

<p><code>lifespan</code> of the nematode <em>Caenorhabditis elegans</em> in control and
three experimental <code>treatment</code>s of the anticonvulsant drug
trimethadione.
</p>


<h3>Format</h3>

<p>A data frame with 200 observations on the following 2 variables.
</p>
 <dl>
<dt>treatment</dt><dd><p>a factor with levels: <code>adult</code>,
<code>larva</code>, <code>larva+adult</code>, and <code>water</code></p>
</dd>
<dt>lifespan</dt><dd><p>a numeric vector of lifespan</p>
</dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> Evason, K., C. Huang, I. Yamben, D.F. Covey,
and K. Kornfeld. 2005. Anticonvulsant medications extend worm life-span.
<em>Science</em> 307: 258-262.
</p>


<h3>References</h3>

<p><a href="http://www.sciencemag.org/cgi/content/abstract/307/5707/258">http://www.sciencemag.org/cgi/content/abstract/307/5707/258</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(NematodeLifespan)
</code></pre>

<hr>
<h2 id='NeotropicalTrees'>Photosynthesis in Neotropical Trees</h2><span id='topic+NeotropicalTrees'></span>

<h3>Description</h3>

<p>Photosynthetic capacity (<code>photosynthetic.capacity</code>) and number of
fruits produced in the previous season (<code>previous.fruits</code>) of 9 females
of <em>Ocotea tenera</em>.
</p>


<h3>Format</h3>

<p>A data frame with 9 observations on the following 2 variables.
</p>
 <dl>
<dt>previous.fruits</dt><dd><p>a numeric vector</p>
</dd>
<dt>photosynthetic.capacity</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> Wheelwright, N.T. and B.A. Logan. 2004.
Previous-year reproduction reduces photosynthetic capacity and slows
lifetime growth in females of a neotropical tree. <em>Proceedings of the
National Academy of Sciences (USA)</em> 101: 8051-8055.
</p>


<h3>References</h3>

<p><a href="http://www.pnas.org/content/101/21/8051.long">http://www.pnas.org/content/101/21/8051.long</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(NeotropicalTrees)
NeotropicalTrees
</code></pre>

<hr>
<h2 id='Newts'>Tetrodotoxin Resistance in Garter Snakes</h2><span id='topic+Newts'></span>

<h3>Description</h3>

<p>Percent reduction in crawl speed (<code>resistance</code>) in the garter snake
after injection of the neurotoxin tetrodotoxin from the rough-skinned newt
(<em>Taricha granulosa</em>).
</p>


<h3>Format</h3>

<p>A data frame with 12 observations on the following 2 variables.
</p>
 <dl>
<dt>locality</dt><dd><p>a factor with levels: <code>Benton</code> and
<code>Warrenton</code></p>
</dd> <dt>resistance</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Geffeney, S., E.D. Brodie, Jr., P.C. Ruben, and E.D. Brodie III.
2002. Mechanisms of adaptation in a predator-prey arms race: TTX-resistant
sodium channels. <em>Science</em> 297: 1336-1339.
</p>


<h3>References</h3>

<p><a href="http://www.sciencemag.org/cgi/content/abstract/297/5585/1336">http://www.sciencemag.org/cgi/content/abstract/297/5585/1336</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Newts
</code></pre>

<hr>
<h2 id='NorthSeaCod'>Atlantic Cod Recruits</h2><span id='topic+NorthSeaCod'></span>

<h3>Description</h3>

<p>Number (<code class="reqn">\log_{10}</code> transformed) of Atlantic cod (<em>Gadus
morhua</em>) that recruited (grew to catchable size) in the North Sea over a 39
years span.
</p>


<h3>Format</h3>

<p>A data frame with 39 observations of one variable. </p>

<dl>
<dt>log10.recruits</dt><dd></dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> Beaugrand, G., K.M. Brander, J.A. Lindley, S.
Souissi, and P.C. Reid. 2003. Plankton effect on cod recruitment in the
North Sea. <em>Nature</em> 426: 661-664.
</p>


<h3>References</h3>

<p><a href="http://www.nature.com/nature/journal/v426/n6967/abs/nature02164.html">http://www.nature.com/nature/journal/v426/n6967/abs/nature02164.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>favstats(NorthSeaCod$log10.recruits)
</code></pre>

<hr>
<h2 id='NoSmokingDay'>No Smoking Day</h2><span id='topic+NoSmokingDay'></span>

<h3>Description</h3>

<p>Number of workplace injuries on No Smoking Day (<code>Injuries.on.NSD</code>)
compared to the same Wednesday in the previous year
(<code>Injuries.before.NSD</code>) for 1987-1996.
</p>


<h3>Format</h3>

<p>A data frame with 10 observations on the following 3 variables.
</p>
 <dl>
<dt>year</dt><dd><p>a numeric vector</p>
</dd>
<dt>injuries.before.NSD</dt><dd><p>a numeric vector</p>
</dd>
<dt>injuries.on.NSD</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Waters, A.J., M.J. Jarvis, and S.R. Sutton. 1998. Nicotine
withdrawal and accident rates. <em>Nature</em> 394: 137.
</p>


<h3>References</h3>

<p><a href="http://www.nosmokingday.org.uk/">http://www.nosmokingday.org.uk/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>NoSmokingDay
</code></pre>

<hr>
<h2 id='OstrichTemp'>Ostrich Body and Brain Temperatures</h2><span id='topic+OstrichTemp'></span>

<h3>Description</h3>

<p>Body and brain temperatures (<code class="reqn">^{\circ}</code>C) in free-ranging
ostriches (<em>Struthio camelus</em>) at the the Lichtenburg Game Breeding
Centre, Lichtenburg, South Africa.
</p>


<h3>Format</h3>

<p>A data frame with 6 observations on the following 3 variables.
</p>
 <dl>
<dt>ostrich</dt><dd><p>a numeric vector identifying ostrich
number</p>
</dd> <dt>body.temp</dt><dd><p>a numeric vector of body temperature in
<code class="reqn">^{\circ}</code>C</p>
</dd> <dt>brain.temp</dt><dd><p>a numeric vector of
brain temperature in <code class="reqn">^{\circ}</code>C</p>
</dd> </dl>


<h3>Source</h3>

<p>Fuller, A., P.R. Kamerman, S.K. Maloney, G. Mitchell, and D.
Mitchell. 2003. Variability in brain and arterial blood temperatures in
free-ranging ostriches in their natural habitat. <em>Journal of
Experimental Biology</em> 206: 1171-1181.
</p>


<h3>References</h3>

<p><a href="http://jeb.biologists.org/cgi/content/abstract/206/7/1171">http://jeb.biologists.org/cgi/content/abstract/206/7/1171</a>
</p>
<p><a href="http://www.sa-venues.com/game-reserves/nwp_lichtenburg.htm">http://www.sa-venues.com/game-reserves/nwp_lichtenburg.htm</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(brain.temp ~ body.temp, OstrichTemp)
</code></pre>

<hr>
<h2 id='Penguins'>Penguin Heart Rate</h2><span id='topic+Penguins'></span>

<h3>Description</h3>

<p>Slope of regressions of mass-specific metabolic rate on heart rate for three
groups of Macaroni Penguins.
</p>


<h3>Format</h3>

<p>A data frame with 24 observations on the following 2 variables.
</p>
 <dl>
<dt>group</dt><dd><p>a factor with levels <code>BF</code>, <code>BM</code>,
and <code>MF</code></p>
</dd> <dt>slope</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Green, J. A., P. J. Butler, A. J. Woakes, I. L. Boyd and R. L.
Holder. 2001. Heart rate and rate of oxygen consumption of exercising
macaroni penguins. <em>Journal of Experimental Biology</em> 204: 673-684.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(Penguins)
dotplot(slope ~ group, Penguins)
</code></pre>

<hr>
<h2 id='PlantPersistence'>Population Persistence Times</h2><span id='topic+PlantPersistence'></span>

<h3>Description</h3>

<p>Persistence times (<code>generations</code>) in the annual plant <em>Cardamine
pensylvanica</em> in four experimental populations (<code>treatment</code>).
</p>


<h3>Format</h3>

<p>A data frame with 16 observations on the following 2 variables.
</p>
 <dl>
<dt>generations</dt><dd><p>a numeric vector</p>
</dd>
<dt>treatment</dt><dd><p>a factor with levels: <code>Isolated</code>,
<code>Medium</code>, <code>Long</code>, and <code>Continuous</code></p>
</dd> </dl>


<h3>Source</h3>

<p>Molofsky, J. and J.-B. Ferdy. 2005. Extinction dynamics in
experimental metapopulations. <em>Proceedings of the National Academy of
Sciences (USA)</em> 102: 3726-3731.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(generations~treatment, PlantPersistence)
</code></pre>

<hr>
<h2 id='Pollen'>Sterility in Hybrid Pollens</h2><span id='topic+Pollen'></span>

<h3>Description</h3>

<p>Genetic distance between pairs of species of the genus <em>Silene</em> and
proportion of their hybrid offspring that are sterile.
</p>


<h3>Format</h3>

<p>A data frame with 23 observations on the following 2 variables.
</p>
 <dl>
<dt>genetic.distance</dt><dd><p>a numeric vector</p>
</dd>
<dt>proportion.sterile</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Moyle, L.C., M.S. Olson, and P. Tiffin. 2004. Patterns of
reproductive isolation in three angiosperm genera. <em>Evolution</em> 58:
1195-1208.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(Pollen)
xyplot(proportion.sterile ~ genetic.distance, Pollen)
</code></pre>

<hr>
<h2 id='Powerball'>Powerball Tickets Sold</h2><span id='topic+Powerball'></span>

<h3>Description</h3>

<p>The number of Powerball tickets sold per day of the week for three years.
</p>


<h3>Format</h3>

<p>A data frame with 7 observations on the following 2 variables.
</p>
 <dl>
<dt>day</dt><dd><p>a character vector</p>
</dd>
<dt>millions.of.tickets.sold</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Oster, E. 2004. Dreaming big: Why do people play Powerball?
<em>Chance News</em> 13.02.
</p>


<h3>References</h3>

<p><a href="http://www.dartmouth.edu/~chance/chance_news/recent_news/chance_news_13.02.html">http://www.dartmouth.edu/~chance/chance_news/recent_news/chance_news_13.02.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Powerball
xyplot(millions.of.tickets.sold ~ day, Powerball)
</code></pre>

<hr>
<h2 id='PrimateMetabolism'>Primate Metabolic Rates</h2><span id='topic+PrimateMetabolism'></span>

<h3>Description</h3>

<p>Body mass (g) and metabolic rate (watts) for 17 species of primates.
</p>


<h3>Format</h3>

<p>A data frame with 17 observations on the following 2 variables.
</p>
 <dl>
<dt>mass</dt><dd><p>mass (g) </p>
</dd> <dt>bmr</dt><dd><p>metabolic rate
(watts)</p>
</dd> </dl>


<h3>Source</h3>

<p>Heusner, A.A. 1991. Size and power in mammals. <em>Journal of
Experimental Biology</em> 160: 25-54.
</p>


<h3>References</h3>

<p><a href="http://jeb.biologists.org/cgi/content/abstract/160/1/25">http://jeb.biologists.org/cgi/content/abstract/160/1/25</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(PrimateMetabolism)
xyplot(bmr ~ mass, PrimateMetabolism)
xyplot(bmr ~ mass, PrimateMetabolism, scales=list(log=TRUE))
</code></pre>

<hr>
<h2 id='PrimateWBC'>Primate White Blood Cell Counts and Promiscuity</h2><span id='topic+PrimateWBC'></span>

<h3>Description</h3>

<p>White blood cell (WBC) counts in pairs of closely related primate species
</p>


<h3>Format</h3>

<p>A data frame with 9 observations on the following 2 variables.
</p>
 <dl>
<dt>WBC.less</dt><dd><p>a numeric vector</p>
</dd>
<dt>WBC.more</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Nunn, C.L., J.L. Gittleman, and J. Antonovics. 2000. Promiscuity and
the primate immune system. <em>Science</em> 290: 1168-1170.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(WBC.more ~ WBC.less, PrimateWBC)
</code></pre>

<hr>
<h2 id='ProgesteroneExercise'>Progesterone and Exercise</h2><span id='topic+ProgesteroneExercise'></span>

<h3>Description</h3>

<p>Progesterone levels and rates of ventilation during submaximal exercise in
30 women.
</p>


<h3>Format</h3>

<p>A data frame with 30 observations on the following 2 variables.
</p>
 <dl>
<dt>progesterone</dt><dd><p>a numeric vector</p>
</dd>
<dt>ventilation</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Brutsaert, T.D., H. Spielvogel, E. Caceres, M. Araoz, R.T.
Chatterton, V.J. Vitzthum. 2002. Effect of menstrual cycle phase on exercise
performance of high-altitude native women at 3600 m. <em>Journal of
Experimental Biology</em> 205: 233-239
</p>


<h3>References</h3>

<p><a href="http://jeb.biologists.org/cgi/content/abstract/205/2/233">http://jeb.biologists.org/cgi/content/abstract/205/2/233</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(ProgesteroneExercise)
xyplot(ventilation ~ progesterone, ProgesteroneExercise)
</code></pre>

<hr>
<h2 id='Pseudoscorpions'>Multiple Mating in Pseudoscorpions</h2><span id='topic+Pseudoscorpions'></span>

<h3>Description</h3>

<p>Successful numbers of broods (<code>Number.of.successful.broods</code>) in two
groups of female pseudoscrpions (<em>Cordylochernes scorpioides</em>), one
mated to the same male twice and one to two different males.
</p>


<h3>Format</h3>

<p>A data frame with 36 observations on the following 2 variables.
</p>
 <dl>
<dt>treatment</dt><dd><p>a factor with levels <code>DM</code>
<code>SM</code></p>
</dd> <dt>successful.broods</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Newcomer, S.D., J.A. Zeh, and D.W. Zeh. 1999. Genetic benefits
enhance the reproductive success of polyandrous females. <em>Proceedings
of the National Academy of Sciences (USA)</em> 96: 10236-10241.
</p>


<h3>References</h3>

<p><a href="http://www.pnas.org/content/96/18/10236.long">http://www.pnas.org/content/96/18/10236.long</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(Pseudoscorpions)
bwplot(successful.broods ~ treatment, Pseudoscorpions)
aggregate(successful.broods ~ treatment, Pseudoscorpions, favstats)
</code></pre>

<hr>
<h2 id='Pufferfish'>Pufferfish Mimicry</h2><span id='topic+Pufferfish'></span>

<h3>Description</h3>

<p>Number of <code>predators</code> approaching models painted to resemble pufferfish
(<em>Canthigaster valentini</em>) across a range of similarities
(<code>resemblance</code>)
</p>


<h3>Format</h3>

<p>A data frame with 20 observations on the following 2 variables.
</p>
 <dl>
<dt>resemblance</dt><dd><p>a numeric vector</p>
</dd>
<dt>predators</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Caley, M.J. and D. Schluter. 2003. Predators favour mimicry in a
tropical reef fish. <em>Proceedings of the Royal Society of London Series
B, Biological Sciences</em> 270: 667-672.
</p>


<h3>References</h3>

<p><a href="http://rspb.royalsocietypublishing.org/content/270/1516/667.full.pdf">http://rspb.royalsocietypublishing.org/content/270/1516/667.full.pdf</a>
</p>
<p><a href="http://en.wikipedia.org/wiki/Canthigaster_valentini">http://en.wikipedia.org/wiki/Canthigaster_valentini</a>
</p>
<p><a href="http://www.fishbase.org/Summary/SpeciesSummary.php?id=6544">http://www.fishbase.org/Summary/SpeciesSummary.php?id=6544</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(Pufferfish)
xyplot(predators ~ jitter(resemblance, amount = 0.1), Pufferfish)
Pufferfish
</code></pre>

<hr>
<h2 id='Rattlesnakes'>Temperature Change and Meal Size in Rattlesnakes</h2><span id='topic+Rattlesnakes'></span>

<h3>Description</h3>

<p>Temperature change after a meal (
rattlesnakes (<em>Crotalus durissus</em>).
</p>


<h3>Format</h3>

<p>A data frame with 17 observations on the following 2 variables.
</p>
 <dl>
<dt>meal.size</dt><dd><p>a numeric vector</p>
</dd>
<dt>temp.change</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Tattersall, G.J., W.K. Milsom, A.S. Abe, S.P. Brito, and D.V.
Andrade. 2004. The thermogenesis of digestion in rattlesnakes. <em>Journal
of Experimental Biology</em> 207: 579-585.
</p>


<h3>References</h3>

<p><a href="http://jeb.biologists.org/cgi/content/abstract/207/4/579">http://jeb.biologists.org/cgi/content/abstract/207/4/579</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(Rattlesnakes)
xyplot(meal.size ~ temp.change, Rattlesnakes)
</code></pre>

<hr>
<h2 id='Rigormortis'>Rigormortis and Time of Death</h2><span id='topic+Rigormortis'></span>

<h3>Description</h3>

<p>Number of bodies reaching rigormortis in each hour after death.
</p>


<h3>Format</h3>

<p>A data frame with 12 observations on the following 2 variables.
</p>
 <dl>
<dt>hours</dt><dd><p>a numeric vector</p>
</dd> <dt>count</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Pounder, D.J. 1995. Postmortem changes and time of death. University
of Dundee.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(count ~ hours, Rigormortis, type='h', lwd=3)
barchart(count ~ hours, Rigormortis, horizontal=FALSE, origin=0)
</code></pre>

<hr>
<h2 id='RopeTrick'>Indian Rope Trick</h2><span id='topic+RopeTrick'></span>

<h3>Description</h3>

<p>Perceived impressiveness (<code>impressiveness</code>) of a written account of the
Indian Rope Trick and the corresponding number of <code>year</code>s since it was
witnessed.
</p>


<h3>Format</h3>

<p>A data frame with 21 observations on the following 2 variables.
</p>
 <dl>
<dt>years</dt><dd><p>a numeric vector</p>
</dd>
<dt>impressiveness</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Wiseman, R. and P. Lamont. 1996. Unravelling the Indian rope-trick.
<em>Nature</em> 383: 212-213.
</p>


<h3>References</h3>

<p><a href="http://www.richardwiseman.com/resources/ropeJSPR.pdf">http://www.richardwiseman.com/resources/ropeJSPR.pdf</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(impressiveness ~ years, RopeTrick)
</code></pre>

<hr>
<h2 id='SagebrushCrickets'>Sagebrush Cricket Mating Times</h2><span id='topic+SagebrushCrickets'></span>

<h3>Description</h3>

<p>Time to mating (<code>time.to.mating</code>) in fed and unfed (<code>treatment</code>)
sagebrush crickets (<em>Cyphoderris strepitans</em>).
</p>


<h3>Format</h3>

<p>A data frame with 24 observations on the following 2 variables.
</p>
 <dl>
<dt>treatment</dt><dd><p>a factor with levels: <code>fed</code> and
<code>starved</code></p>
</dd> <dt>time.to.mating</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Chadwick Johnson, J., T.M. Ivy, and S.K. Sakaluk. 1999. Female
remating propensity contingent on sexual cannibalism in sagebrush crickets,
<em>Cyphoderris strepitans</em>: a mechanism of cryptic female choice.
<em>Behavioral Ecology</em> 10: 227-233.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SagebrushCrickets
str(SagebrushCrickets)
</code></pre>

<hr>
<h2 id='SalmonColor'>Pacific Salmon Color</h2><span id='topic+SalmonColor'></span>

<h3>Description</h3>

<p>Skin color sockeye and kokanee morphs of the Pacific salmon
(<em>Oncorhynchus nerka</em>) raised in a low carotenoid environment.
</p>


<h3>Format</h3>

<p>A data frame with 35 observations on the following 2 variables.
</p>
 <dl>
<dt>species</dt><dd><p>a factor with levels <code>kokanee</code> and
<code>sockeye</code></p>
</dd> <dt>skin.color</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Craig, J.K. and C. Foote. 2001. Countergradient variation and
secondary sexual color: phenotypic convergence promotes genetic divergence
in carotenoid use between sympatric anadromous and nonanadromous morphs of
sockeye salmon (<em>Oncorhynchus nerka</em>). <em>Evolution</em> 55: 380-391.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SalmonColor
histogram(~ skin.color | species, SalmonColor)
bwplot(skin.color ~ species, SalmonColor)
</code></pre>

<hr>
<h2 id='Seedlings'>Number of Seedlings Per Quadrat</h2><span id='topic+Seedlings'></span>

<h3>Description</h3>

<p>Data on frequency of seeding per quadrat for 80 hypothetical quadrats.
</p>


<h3>Format</h3>

<p>A data frame with 8 observations on the following 2 variables.
</p>
 <dl>
<dt>seedlings</dt><dd><p>a numeric vector</p>
</dd> <dt>count</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>Examples</h3>

<pre><code class='language-R'>Seedlings
</code></pre>

<hr>
<h2 id='Selection'>Data for Meta-analysis</h2><span id='topic+Selection'></span>

<h3>Description</h3>

<p>Data for meta-analysis on the relationship between testosterone and
aggression.
</p>


<h3>Format</h3>

<p>A data frame with 814 observations on the following 8 variables.
</p>
 <dl>
<dt>species</dt><dd><p>species investigated</p>
</dd>
<dt>traitname</dt><dd><p>trait investigated</p>
</dd>
<dt>strength.of.selection</dt><dd><p>strength of selection</p>
</dd>
<dt>sample.size</dt><dd><p>size of sample</p>
</dd> <dt>authors</dt><dd><p>authors of
publication</p>
</dd> <dt>year</dt><dd><p>year of publication</p>
</dd>
<dt>journal</dt><dd><p>journal of publication</p>
</dd>
<dt>volume.pages</dt><dd><p>volume and pages</p>
</dd> </dl>


<h3>Source</h3>

<p>Kingsolver, J.G., H.E. Hoekstra, J.M. Hoekstra, D. Berrigan, S.N.
Vignieri, C.E. Hill, A. Hoang, P. Gibert, and P. Beerli. 2001. The strength
of phenotypic selection in natural populations. <em>The American
Naturalist</em> 157: 245-261.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>histogram(~ strength.of.selection, Selection,n=40)
table(Selection$species) -&gt; s
table(s)
s[s&gt;10] # most common species
table(Selection$traitname) -&gt; t
table(t)
t[t&gt;10] # most common traits
</code></pre>

<hr>
<h2 id='SexualSelection'>Sexual Conflict</h2><span id='topic+SexualSelection'></span>

<h3>Description</h3>

<p>Number of species in each of two taxa in closely related taxon pairings and
the difference between the two groups. One taxon has multiple matings
(<code>polyandrous.species</code>) and one has only single matings
(<code>monandrous.species</code>).
</p>


<h3>Format</h3>

<p>A data frame with 25 observations on the following 4 variables.
</p>
 <dl>
<dt>polyandrous.species</dt><dd><p>a numeric vector</p>
</dd>
<dt>monandrous.species</dt><dd><p>a numeric vector</p>
</dd>
<dt>difference</dt><dd><p>a numeric vector</p>
</dd>
<dt>taxon.pair</dt><dd><p>identifier</p>
</dd> </dl>


<h3>Source</h3>

<p>Arnqvist, G., M. Edvardsson, U. Friberg, and T. Nilsson. 2000.
Sexual conflict promotes speciation in insects. <em>Proceedings of the
National Academy of Sciences (USA)</em> 97: 10460-10464.
</p>


<h3>References</h3>

<p><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC27046/">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC27046/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SexualSelection

histogram(~ difference, SexualSelection, n = 20)

hist(SexualSelection$difference, breaks = 20)

# Calculate the number of tests and the number of negative tests
(n &lt;- length(SexualSelection$difference))
(n.neg &lt;- sum(SexualSelection$difference &lt; 0))

2 * pbinom(q = n.neg, size = n, prob = 0.5)

# With a binomial test
binom.test(n.neg, n, p = 0.5)
</code></pre>

<hr>
<h2 id='ShadParasites'>Shad Parasites</h2><span id='topic+ShadParasites'></span>

<h3>Description</h3>

<p>Frequency of the nematode <em>Camallanus oxycephalus</em> per fish.
</p>


<h3>Format</h3>

<p>A data frame with 7 observations on the following 2 variables.
</p>
 <dl>
<dt>parasites</dt><dd><p>a numeric vector</p>
</dd> <dt>count</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Shaw, D.J., B.T. Grenfell, and A.P. Dobson. 1998. Patterns of
macroparasite aggregation in wildlife host populations. <em>Parasitology</em>
117: 597-610.
</p>


<h3>References</h3>

<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/9881385">http://www.ncbi.nlm.nih.gov/pubmed/9881385</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ShadParasites
</code></pre>

<hr>
<h2 id='ShrinkingSeals'>Seal Body Lengths and Age</h2><span id='topic+ShrinkingSeals'></span>

<h3>Description</h3>

<p>Body length (cm) and age (days) for 9,665 female Northern fur seals
(<em>Callorhinus ursinus</em>).
</p>


<h3>Format</h3>

<p>A data frame with 9,665 observations on the following 2 variables.
</p>
 <dl>
<dt>age</dt><dd><p>age (days)</p>
</dd> <dt>length</dt><dd><p>body length
(cm) </p>
</dd> </dl>


<h3>Source</h3>

<p>Trites, A.W. and M.A. Bigg. 1996. Physical growth of northern fur
seals: seasonal fluctuations and migratory influences. <em>Journal of
Zoology (London)</em> 238: 459-482.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(ShrinkingSeals)

plot(ShrinkingSeals, pch = 16, cex = 0.5)
xyplot(length ~ age, ShrinkingSeals, pch=16, alpha=0.65, cex=0.6)
</code></pre>

<hr>
<h2 id='ShuttleDisaster'>Ambient Temperature and O-Ring Failures</h2><span id='topic+ShuttleDisaster'></span>

<h3>Description</h3>

<p>Data on <code>temperature</code> and number of O-ring <code>failures</code> for 23 space
shuttle launches.
</p>


<h3>Format</h3>

<p>A data frame with 23 observations on the following 2 variables.
</p>
 <dl>
<dt>temperature</dt><dd><p>a numeric vector</p>
</dd>
<dt>failures</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Dalal, S.R., E.B. Fowlkes, and B. Hoadley. 1989. Risk analysis of
the Space Shuttle: Pre-Challenger prediction of failure. <em>Journal of
the American Statistical Association</em> 408: 945-957.
</p>


<h3>References</h3>

<p>Tufte, E.R. 1997. <em>Visual Explanations: Images and
Quantities, Evidence and Narrative</em>. Graphics Press.
</p>
<p><a href="http://www.edwardtufte.com/tufte/">http://www.edwardtufte.com/tufte/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(ShuttleDisaster)
xyplot( jitter(failures, amount=0.1) ~ temperature, ShuttleDisaster,
  ylab='number of failures'
  )
</code></pre>

<hr>
<h2 id='Silversword'>Rate of Speciation in Silverswords</h2><span id='topic+Silversword'></span>

<h3>Description</h3>

<p>Speciation &quot;waiting times&quot; in Hawaiian silverswords (<em>Dubautia</em>).
</p>


<h3>Format</h3>

<p>A data frame with 21 observations on the following variable.
</p>
 <dl>
<dt>waiting.time</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> Baldwin, B. G. and M. J. Sanderson 1998. Age
and rate of diversification of the Hawaiian silversword alliance
(Compositae). <em>Proceedings of the National Academy of Sciences (USA)</em>
95: 9402-9406.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Silversword
</code></pre>

<hr>
<h2 id='SleepAndPerformance'>Sleep and Learning</h2><span id='topic+SleepAndPerformance'></span>

<h3>Description</h3>

<p>The increase in &quot;slow-wave&quot; sleep and improvements in spatial learning tasks
in 10 humans.
</p>


<h3>Format</h3>

<p>A data frame with 10 observations on the following 2 variables.
</p>
 <dl>
<dt>sleep</dt><dd><p>a numeric vector</p>
</dd>
<dt>improvement</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Huber, R., M.F. Ghilardi, M. Massimini, and G. Tononi. 2004. Local
sleep and learning. <em>Nature</em> 430: 78-81.
</p>


<h3>References</h3>

<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/15184907">http://www.ncbi.nlm.nih.gov/pubmed/15184907</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(SleepAndPerformance)
xyplot(improvement ~ sleep, SleepAndPerformance)
</code></pre>

<hr>
<h2 id='SockeyeFemales'>Body Masses of Female Sockeye Salmon</h2><span id='topic+SockeyeFemales'></span>

<h3>Description</h3>

<p>Body Masses of 228 female Sockeye Salmon (<em>Oncorhynchus nerka</em>;
<a href="http://www.nmfs.noaa.gov/pr/species/fish/sockeyesalmon.htm">http://www.nmfs.noaa.gov/pr/species/fish/sockeyesalmon.htm</a>)
</p>


<h3>Format</h3>

<p>A data frame with 228 observations of a single variable. </p>

<dl>
<dt>mass</dt><dd><p>body mass (kg)</p>
</dd> </dl>


<h3>Source</h3>

<p>Hendry, A.P., O.K. Berg, and T.P. Quinn. 1999. Condition dependence
and adaptation-by-time: Breeding date, life history, and energy allocation
within a population of salmon. <em>Oikos</em> 85: 499-514.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(SockeyeFemales)
summary(SockeyeFemales)
</code></pre>

<hr>
<h2 id='Sparrows'>Lifetime Reproductive Success in House Sparrows</h2><span id='topic+Sparrows'></span>

<h3>Description</h3>

<p>A cross table of lifetime reproductive success (<code>LifetimeRS</code>) in female
and male house sparrows <em>Passer domesticus</em> in Norway.
</p>


<h3>Format</h3>

<p>A data frame with 9 observations on the following 3 variables.
</p>
 <dl>
<dt>lifetimeRS</dt><dd><p>a numeric vector</p>
</dd>
<dt>females</dt><dd><p>a numeric vector</p>
</dd> <dt>males</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Jensen, H., B.-E. Saether, T.H. Ringsby, J. Tufto, S.C.
Griffith, and H. Ellegren. 2004. Lifetime reproductive success in relation
to morphology in the House Sparrow <em>Passer domesticus</em>. <em>Journal
of Animal Ecology</em> 73: 599-611.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Sparrows
</code></pre>

<hr>
<h2 id='SpiderColonies'>Social Spiders</h2><span id='topic+SpiderColonies'></span>

<h3>Description</h3>

<p>Web height above ground (cm) and colony size for 17 colonies of the spider
<em>Cryptophora citricola</em> in Gabon.
</p>


<h3>Format</h3>

<p>A data frame with 17 observations on the following 3 variables.
</p>
 <dl>
<dt>colony</dt><dd><p>identifier</p>
</dd> <dt>height</dt><dd><p>height of
web above ground (cm)</p>
</dd> <dt>spiders</dt><dd><p>number of spiders in colony</p>
</dd> </dl>


<h3>Source</h3>

<p>Rypstra, A. L. 1979. Foraging folks of spiders, a study of aggregate
behavior in <em>Cryptophora citricola</em> Forskal (Araneae: Araneidae) in
West Africa. <em>Behavioral Ecology and Sociobiology</em> 5: 291-300.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(SpiderColonies)
SpiderColonies
</code></pre>

<hr>
<h2 id='SpiderSpeed'>Spider Running Speeds after Amputation</h2><span id='topic+SpiderSpeed'></span>

<h3>Description</h3>

<p>Data on speed before and after amputation of a pedipalp in the spider genus
<em>Tidarren</em>.
</p>


<h3>Format</h3>

<p>A data frame with 32 observations on the following 2 variables.
</p>
 <dl>
<dt>speed.before</dt><dd><p>speed (cm/s) before amputation </p>
</dd>
<dt>speed.after</dt><dd><p>speed (cm/s) after amputation </p>
</dd> </dl>


<h3>Source</h3>

<p>Ramos, M., D.J. Irschick, and T.E. Christenson. 2004. Overcoming an
evolutionary conflict: Removal of a reproductive organ greatly increases
locomotor performance. <em>Proceedings of the National Academy of Sciences
(USA)</em> 101: 4883-4887.
</p>


<h3>References</h3>

<p><a href="http://en.wikipedia.org/wiki/Pedipalp">http://en.wikipedia.org/wiki/Pedipalp</a>, <a href="http://en.wikipedia.org/wiki/Tidarren">http://en.wikipedia.org/wiki/Tidarren</a>, <a href="http://www.pnas.org/content/101/14/4883">http://www.pnas.org/content/101/14/4883</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(speed.after ~ speed.before, SpiderSpeed)
favstats(SpiderSpeed$speed.before)
favstats(SpiderSpeed$speed.after)
favstats(SpiderSpeed$speed.after - SpiderSpeed$speed.before)
</code></pre>

<hr>
<h2 id='Stalkies1'>Eye Widths in Stalk-Eyed Flies</h2><span id='topic+Stalkies1'></span>

<h3>Description</h3>

<p>Eye width in 9 male stalk-eyed flies (<em>Cyrtodiopsis dalmanni</em>).
</p>


<h3>Format</h3>

<p>a data frame with 9 observations of 1 variable </p>

<dl>
<dt>eye.span</dt><dd><p>eye span (mm)</p>
</dd> </dl>


<h3>Source</h3>

<p>Data provided by Kevin Fowler, University College, London.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Stalkies1
</code></pre>

<hr>
<h2 id='Stalkies2'>Stalk-eyed Fly Eyespan</h2><span id='topic+Stalkies2'></span>

<h3>Description</h3>

<p>Eyespan width (mm; <code>Eye.span</code>) in 45 stalk-eyed flies
(<em>Cyrtodiopsis dalmanni</em>) fed a corn or cotton diet (<code>Food</code>).
</p>


<h3>Format</h3>

<p>A data frame with 45 observations on the following 2 variables.
</p>
 <dl>
<dt>food</dt><dd><p>a factor with levels <code>Corn</code>
<code>Cotton</code></p>
</dd> <dt>eye.span</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>David, P., T. Bjorksten, K. Fowler, and A. Pomiankowski. 2000.
Condition-dependent signalling of genetic variation in stalk-eyed flies.
<em>Nature</em> 406: 186-188.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(Stalkies2)
xyplot(eye.span ~ food, Stalkies2)
aggregate(eye.span ~ food, Stalkies2, FUN = favstats)
</code></pre>

<hr>
<h2 id='SticklebackPlates'>Number of Lateral Plates in Sticklebacks</h2><span id='topic+SticklebackPlates'></span>

<h3>Description</h3>

<p>Number of lateral plates (<code>plates</code>) in threespine sticklebacks
(<em>Gasterosteus aculeatus</em>) with three different <em>Ectodysplasin</em>
genotypes (<code>mm</code>, <code>Mm</code>, and <code>MM</code>).
</p>


<h3>Format</h3>

<p>A data frame with 344 observations on the following 2 variables.
</p>
 <dl>
<dt>genotype</dt><dd><p>a factor with levels <code>mm</code>,
<code>Mm</code>, and <code>MM</code></p>
</dd> <dt>plates</dt><dd><p>number of plates</p>
</dd> </dl>


<h3>Source</h3>

<p>Colosimo, P.F., C.L. Peichel, K. Nereng, B.K. Blackman, M.D.
Shapiro, D. Schluter, and D.M. Kingsley. 2004. The genetic architecture of
parallel armor plate reduction in threespine sticklebacks. <em>PLoS
Biology</em> 2: 635-641.
<a href="http://www.plosbiology.org/article/info:doi/10.1371/journal.pbio.0020109">http://www.plosbiology.org/article/info:doi/10.1371/journal.pbio.0020109</a>
</p>


<h3>References</h3>

<p>Colosimo P.F., K.E. Hosemann, S. Balabhadra, G. Villarreal, M.
Dickson, J. Grimwood, J Schmutz, R.M. Myers, D. Schluter, D.M. Kingsley.
2005. Widespread parallel evolution in sticklebacks by repeated fixation of
ectodysplasin alleles. <em>Science </em>307: 1928-33.
<a href="http://www.sciencemag.org/cgi/content/full/307/5717/1928">http://www.sciencemag.org/cgi/content/full/307/5717/1928</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>aggregate(plates ~ genotype, SticklebackPlates, FUN = favstats)

histogram( ~ plates | genotype, SticklebackPlates,
  layout = c(1,3),
  n = 15,
  xlab = "Number of Lateral Body Plates"
  )

densityplot( ~ plates | genotype, SticklebackPlates,
  xlab = "Number of Lateral Body Plates",
  layout = c(1,3)
  )
</code></pre>

<hr>
<h2 id='SticklebackPreference'>Mating Preferences in Sticklebacks</h2><span id='topic+SticklebackPreference'></span>

<h3>Description</h3>

<p>Mating preference in 9 populations of three-spined sticklebacks.
</p>


<h3>Format</h3>

<p>A data frame with 9 observations of one variable. </p>

<dl>
<dt>preference.index</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>McKinnon, J. S., S. Mori, B.K. Blackman, L. David, D.M. Kingsley, L.
Jamieson, J. Chou, and D. Schluter. 2004. Evidence for ecology's role in
speciation. <em>Nature</em> 429: 294-298.
</p>


<h3>References</h3>

<p><a href="http://www.nature.com/nature/journal/v429/n6989/abs/nature02556.html">http://www.nature.com/nature/journal/v429/n6989/abs/nature02556.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SticklebackPreference
histogram(~ preference.index, SticklebackPreference)
dotplot(~ preference.index, SticklebackPreference)
</code></pre>

<hr>
<h2 id='Sumo'>Sumo Wrestling Wins</h2><span id='topic+Sumo'></span>

<h3>Description</h3>

<p>Counts of number of wins for sumo wrestlers.
</p>


<h3>Format</h3>

<p>A data frame with 16 observations on the following 2 variables.
</p>
 <dl>
<dt>wins</dt><dd><p>a numeric vector</p>
</dd> <dt>count</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Duggan, M. and S.D. Leavitt. 2002. Winning isn't everything:
Corruption in sumo wrestling. <em>The American Economic Review</em> 92:
1594-1605.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(count ~ wins, Sumo, type='h', lwd=4)
</code></pre>

<hr>
<h2 id='SyrupSwimming'>Syrup Swimming</h2><span id='topic+SyrupSwimming'></span>

<h3>Description</h3>

<p>Relative swimming speed (speed in syrup / speed in water) for 18 swimmers.
</p>


<h3>Format</h3>

<p>A data frame with 18 observations of one variable. </p>

<dl>
<dt>relative.speed</dt><dd><p>ratio of speed in syrup to speed in water</p>
</dd> </dl>


<h3>Source</h3>

<p>Gettelfinger, B. and E. L. Cussler. 2004. Will Humans Swim Faster or
Slower in Syrup? <em>AIChE Journal</em> 50: 2646-2647.
</p>


<h3>References</h3>

<p><a href="http://www3.interscience.wiley.com/journal/109665380/issue">http://www3.interscience.wiley.com/journal/109665380/issue</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SyrupSwimming
histogram(~ relative.speed, SyrupSwimming)
dotplot(~ relative.speed, SyrupSwimming)
</code></pre>

<hr>
<h2 id='TeenDeaths'>Causes of Teenage Deaths</h2><span id='topic+TeenDeaths'></span>

<h3>Description</h3>

<p>Data from Table 1 (p. 14) on causes of death for all races, both sexes, ages
15-19.
</p>


<h3>Format</h3>

<p>A data frame with 11 observations on the following 2 variables.
</p>
 <dl>
<dt>cause</dt><dd><p>a character vector</p>
</dd> <dt>deaths</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Anderson, R.N. 2001. Deaths: Leading causes for 1999. <em>National
vital statistics reports</em> 49(11):1-88. National Center for Health
Statistics; Hyattsville, Maryland.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(TeenDeaths)
TeenDeaths

barchart(deaths ~ cause, TeenDeaths,
  horizontal = FALSE,
  ylab = "Number of Deaths",
  xlab = "Cause of Death", origin=0,
  scales = list(x = list(rot=45)))

barchart(deaths~ordered(cause, levels=cause), TeenDeaths,
  horizontal = FALSE,
  ylab = "Number of Deaths",
  xlab = "Cause of Death", origin=0,
  scales=list(x=list(rot=45))
  )
</code></pre>

<hr>
<h2 id='Telomeres'>Telomere Shortening</h2><span id='topic+Telomeres'></span>

<h3>Description</h3>

<p>Telomere length (ratio) and years since their child's diagnosis with chronic
illness.
</p>


<h3>Format</h3>

<p>A data frame with 39 observations on the following 2 variables.
</p>
 <dl>
<dt>years</dt><dd><p>a numeric vector</p>
</dd>
<dt>telomere.length</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Epel, E.S., E.H. Blackburn, J. Lin, F.S. Dhabhar, N.E. Adler, J.D.
Morrow, and R.M. Cawthon. 2004. Accelerated telomere shortening in response
to life stress. <em>Proceedings of the National Academy of Sciences (USA)</em>
101: 17312-17315.
</p>


<h3>References</h3>

<p><a href="http://www.pnas.org/content/101/49/17312">http://www.pnas.org/content/101/49/17312</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(years ~ telomere.length, Telomeres,
  xlab = "Time since diagnosis (years)",
  ylab = "Telomere length (ratio)"
)
</code></pre>

<hr>
<h2 id='TimeOfDeath'>Hypoxanthine and Time Since Death</h2><span id='topic+TimeOfDeath'></span>

<h3>Description</h3>

<p>Hypoxanthine levels in the vitreous humour of the eye and time since death
(hours) for 48 subjects.
</p>


<h3>Format</h3>

<p>A data frame with 48 observations on the following 2 variables.
</p>
 <dl>
<dt>hours</dt><dd><p>a numeric vector</p>
</dd>
<dt>hypoxanthine</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>James, R.A., P.A. Hoadley, and B.G. Sampson. 1997. Determination of
postmortem interval by sampling vitreous humor. <em>American Journal of
Forensic Medicine and Pathology</em> 18: 158-162.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(hypoxanthine ~ hours, TimeOfDeath, type=c('p','r'))
</code></pre>

<hr>
<h2 id='Toads'>Right-handed Toads</h2><span id='topic+Toads'></span>

<h3>Description</h3>

<p>Hypothetical probability of a toad being right-handed
</p>


<h3>Format</h3>

<p>A data frame with 19 observations on the following 2 variables.
</p>
 <dl>
<dt>n.toads</dt><dd><p>a numeric vector</p>
</dd> <dt>prob</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>Examples</h3>

<pre><code class='language-R'>Toads
# generate this data manually
cbind(0:18, dbinom(0:18, 18, 0.5))
xyplot(prob ~ n.toads, Toads, type = 'h', lwd = 4)
barchart(prob ~ n.toads, Toads, origin=0, horizontal=FALSE)
plotDist('binom', params = list(18,0.5), kind = 'hist')
</code></pre>

<hr>
<h2 id='Tobacco'>Flower Length in Tobacco Plants</h2><span id='topic+Tobacco'></span>

<h3>Description</h3>

<p>Distribution of flow lengths in F1 and F2 populations of <em>Nicotiana</em>.
</p>


<h3>Format</h3>

<p>A data frame with 13 observations on the following 3 variables.
</p>
 <dl>
<dt>flower.length</dt><dd><p>a numeric vector of flower length in
mm</p>
</dd> <dt>f1.count</dt><dd><p>a numeric vector of the number of F1 plants with
flower lengths in this size range</p>
</dd> <dt>f2.count</dt><dd><p>a numeric vector
of the number of F2 plants with flower lengths in this size range</p>
</dd> </dl>


<h3>Source</h3>

<p>East, E.M. 1916. Studies on size inheritance in <em>Nicotiana</em>.
<em>Genetics</em> 1: 164-176.
</p>


<h3>References</h3>

<p><a href="http://www.genetics.org/content/vol1/issue2/">http://www.genetics.org/content/vol1/issue2/</a>
</p>
<p><a href="http://en.wikipedia.org/wiki/Nicotiana">http://en.wikipedia.org/wiki/Nicotiana</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Tobacco2">Tobacco2</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Tobacco
</code></pre>

<hr>
<h2 id='Tobacco2'>Flower Length in Tobacco Plants</h2><span id='topic+Tobacco2'></span>

<h3>Description</h3>

<p>Distribution of flow lengths in F1 and F2 populations of <em>Nicotiana</em>.
</p>


<h3>Format</h3>

<p>A data frame with 617 observations on the following 2 variables.
</p>
 <dl>
<dt>flower.length</dt><dd><p>a numeric vector</p>
</dd>
<dt>generation</dt><dd><p>a factor with levels <code>F1</code> <code>F2</code></p>
</dd> </dl>


<h3>Source</h3>

<p>East, E.M. 1916. Studies on size inheritance in <em>Nicotiana</em>.
<em>Genetics</em> 1: 164-176.
</p>


<h3>References</h3>

<p><a href="http://www.genetics.org/content/vol1/issue2/">http://www.genetics.org/content/vol1/issue2/</a>
</p>
<p><a href="http://en.wikipedia.org/wiki/Nicotiana">http://en.wikipedia.org/wiki/Nicotiana</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Tobacco">Tobacco</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xtabs(~ flower.length + generation, Tobacco2)
bwplot(flower.length ~ generation, Tobacco2)
</code></pre>

<hr>
<h2 id='ToothAge'>Radioactive Teeth</h2><span id='topic+ToothAge'></span>

<h3>Description</h3>

<p>Actual birth year and birth year estimated from relative radioactivity of
the enamel for 20 samples.
</p>


<h3>Format</h3>

<p>A data frame with 20 observations on the following 2 variables.
</p>
 <dl>
<dt>actual</dt><dd><p>a numeric vector</p>
</dd>
<dt>estimated</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Spalding, K.L., B.A. Buchholz, L.-E. Bergman, H. Druid, and J.
FrisÃ©n. 2005. Age written in teeth by nuclear tests. <em>Nature</em> 437:
333-334.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(ToothAge)
xyplot(actual ~ estimated, ToothAge)
</code></pre>

<hr>
<h2 id='TreeSeedlings'>Tree Seedlings and Sunflecks</h2><span id='topic+TreeSeedlings'></span>

<h3>Description</h3>

<p>Fleck duration (min) and relative seedling growth rate (mm/mm/week) for 21
seedlings of <em>Shorea leprosula</em>.
</p>


<h3>Format</h3>

<p>A data frame with 21 observations on the following 2 variables.
</p>
 <dl>
<dt>fleck.duration</dt><dd><p>a numeric vector</p>
</dd>
<dt>growth</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Leakey, A.D.B., J.D. Scholes, and M.C. Press. 2005. Physiological
and ecological significance of sunflecks for dipterocarp seedlings.
<em>Journal of Experimental Botany</em> 56: 469-482.
</p>


<h3>References</h3>

<p><a href="http://jxb.oxfordjournals.org/cgi/content/short/56/411/469">http://jxb.oxfordjournals.org/cgi/content/short/56/411/469</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(TreeSeedlings)
splom(TreeSeedlings)
</code></pre>

<hr>
<h2 id='Trematodes'>Frequencies of Fish Eaten by Trematode Infection Level</h2><span id='topic+Trematodes'></span>

<h3>Description</h3>

<p>Frequencies of killifish (<em>Fundulus parvipinnis</em>) eaten by birds
depending on level of infection by the trematode <em>Euhaplorchis
californiensis</em>.
</p>


<h3>Format</h3>

<p>A data frame with 141 observations on the following 2 variables.
</p>
 <dl>
<dt>infection.status</dt><dd><p>a factor with levels:
<code>high</code>, <code>light</code>, and <code>uninfected</code></p>
</dd> <dt>eaten</dt><dd><p>a
factor with levels: <code>no</code> and <code>yes</code></p>
</dd> </dl>


<h3>Source</h3>

<p>Lafferty, K.D. and A.K. Morris. 1996. Altered behavior of
parasitized killifish increases susceptibility to predation by bird final
hosts. <em>Ecology</em> 77: 1390-1397.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>demo(sec9.3)
</code></pre>

<hr>
<h2 id='Trillium'>Trillium Recruitment near Clearcuts</h2><span id='topic+Trillium'></span>

<h3>Description</h3>

<p>Recruitment of <em>Trillium</em> and distance to nearest clearcut in eight
populations in southwestern Oregon.
</p>


<h3>Format</h3>

<p>A data frame with 8 observations on the following 3 variables.
</p>
 <dl>
<dt>population</dt><dd><p>a numeric vector</p>
</dd>
<dt>edge.dist</dt><dd><p>a numeric vector</p>
</dd> <dt>recruitment</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Jules, E.S. and B.J. Rathcke. 1999.  Mechanisms of reduced trillium
recruitment along edges of old-growth forest fragments. <em>Conservation
Biology</em> 13: 784-793
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(Trillium)
splom(Trillium)
</code></pre>

<hr>
<h2 id='Truffles'>Truffle Distribution</h2><span id='topic+Truffles'></span>

<h3>Description</h3>

<p>Number of truffles per plot for 288 plots in an old growth forest in
northeastern California.
</p>


<h3>Format</h3>

<p>A data frame with 5 observations on the following 2 variables.
</p>
 <dl>
<dt>truffles</dt><dd><p>a numeric vector</p>
</dd> <dt>count</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Waters, J.R., K.S. McKelvey, D.L. Luoam, and C.J. Zabel. 1997.
Truffle production in old-growth and mature fir stands in northeastern
California. <em>Forest Ecology and Management</em> 96: 155-166.
</p>


<h3>References</h3>

<p><a href="http://www.fs.fed.us/psw/publications/watersj/waters2.PDF">http://www.fs.fed.us/psw/publications/watersj/waters2.PDF</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Truffles
xyplot(count ~ truffles, Truffles, type='h', lwd=4)
barchart(count ~ truffles, Truffles, origin=0, horizontal=FALSE)
</code></pre>

<hr>
<h2 id='TsetseLearning'>Dietary Learning in Tsetse Flies</h2><span id='topic+TsetseLearning'></span>

<h3>Description</h3>

<p>Dietary conditioning <code>treatment</code> and subsequent proportion of tsetse
flies (<em>Glossina palpalis</em>) feeding on cow blood in each of 13 cohorts.
</p>


<h3>Format</h3>

<p>A data frame with 13 observations on the following 2 variables.
</p>
 <dl>
<dt>treatment</dt><dd><p>a factor with levels <code>cow</code> and
<code>lizard</code></p>
</dd> <dt>proportion.cow</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> Bouyer, J., M. Pruvot, Z. Bengaly, P.M. Guerin,
and R. Lancelot. 2007. Learning influences host choice in tsetse.
<em>Biology Letters</em> 3: 113-116.
</p>


<h3>References</h3>

<p><a href="http://rsbl.royalsocietypublishing.org/content/3/2/113.full">http://rsbl.royalsocietypublishing.org/content/3/2/113.full</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(proportion.cow ~ treatment, TsetseLearning)
</code></pre>

<hr>
<h2 id='TwoKids'>Number of Boys in Two-Child Families</h2><span id='topic+TwoKids'></span>

<h3>Description</h3>

<p>The number of boys in a sample of 2,444 two-child families.
</p>


<h3>Format</h3>

<p>A data frame with 3 observations on the following 2 variables.
</p>
 <dl>
<dt>num.boys</dt><dd><p>a numeric vector</p>
</dd> <dt>count</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Rodgers, J.L. and D. Doughty. 2001. Does having boys or girls run in
the family? <em>Chance Magazine</em> Fall 2001: 8-13.
</p>


<h3>References</h3>

<p><a href="http://www.dartmouth.edu/~chance/chance_news/recent_news/chance_news_10.11.html#item13">http://www.dartmouth.edu/~chance/chance_news/recent_news/chance_news_10.11.html#item13</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>TwoKids
observed &lt;- TwoKids$count
expected &lt;- c(585.3, 1221.4, 637.3)
chisq.test(observed, p = expected, rescale.p = TRUE)

# Alternate calculation, using Pr[male] = 0.512
# and rbinom. See Figure 5.7-1
n &lt;- sum(observed)
pr.m &lt;- 0.512
pr.f &lt;- 0.488

# Calculate the probabilities of 0, 1, and 2 males
(pr.0 &lt;- pr.f^2)
(pr.1 &lt;- pr.m * pr.f + pr.f * pr.m)
(pr.2 &lt;- pr.m^2)

set.seed(1)
(expected2 &lt;- c(rbinom(1, n, pr.0),
                rbinom(1, n, pr.1),
                rbinom(1, n, pr.2)))
chisq.test(observed, p = expected2, rescale.p = TRUE)
</code></pre>

<hr>
<h2 id='VampireBites'>Vampire Bat Bites</h2><span id='topic+VampireBites'></span>

<h3>Description</h3>

<p>Numbers of cattle bitten by the cow's estrous cycle.
</p>


<h3>Format</h3>

<p>A data frame with 4 observations on the following 3 variables.
</p>
 <dl>
<dt>estrous</dt><dd><p>a factor with levels: <code>no</code> and
<code>yes</code></p>
</dd> <dt>bitten</dt><dd><p>a factor with levels: <code>no</code> and
<code>yes</code></p>
</dd> <dt>count</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Turner, D.C. 1975. <em>The Vampire Bat: a Field Study in Behavior
and Ecology</em>. Johns Hopkins Press: Baltimore, MD.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>demo(sec9.4)
</code></pre>

<hr>
<h2 id='VasopressinVoles'>Vasopressin Manipulation in the Meadow Vole</h2><span id='topic+VasopressinVoles'></span>

<h3>Description</h3>

<p>Time spent with a female (<code>percent</code>) in control and
vasopressin-enhanced groups (<code>treatment</code>) of meadow voles
(<em>Microtus pennsylvanicus</em>).
</p>


<h3>Format</h3>

<p>A data frame with 31 observations on the following 2 variables.
</p>
 <dl>
<dt>treatment</dt><dd><p>a factor with levels <code>control</code> and
<code>enhanced</code></p>
</dd> <dt>percent</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> Lim, M.M., Z. Wang, D.E.
Olazabal, X. Ren, E.F. Terwilliger, and L.J. Young. 2004.
Enhanced partner preference in a promiscuous species by manipulating the
expression of a single gene. <em>Nature</em> 429: 754-757.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(percent ~ treatment, VasopressinVoles, type=c('p','a'))
bwplot(percent ~ treatment, VasopressinVoles)
</code></pre>

<hr>
<h2 id='Vines'>Climbing Vines</h2><span id='topic+Vines'></span>

<h3>Description</h3>

<p>Number of <code>climbing</code> and <code>nonclimbing</code> species within closely
related general of plants.
</p>


<h3>Format</h3>

<p>A data frame with 48 observations on the following 2 variables.
</p>
 <dl>
<dt>climbing</dt><dd><p>a numeric vector</p>
</dd>
<dt>nonclimbing</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Gianoli, E. 2004. Evolution of a climbing habit promotes
diversification in flowering plants. <em>Proceedings of the Royal Society
of London, Series B, Biological Sciences</em> 271: 2011-2015.
</p>


<h3>References</h3>

<p><a href="http://rspb.royalsocietypublishing.org/content/271/1552/2011.full.pdf">http://rspb.royalsocietypublishing.org/content/271/1552/2011.full.pdf</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(nonclimbing ~ climbing, Vines, scales=list(log=TRUE))
</code></pre>

<hr>
<h2 id='VoleDispersal'>Home Range Size in Field Voles</h2><span id='topic+VoleDispersal'></span>

<h3>Description</h3>

<p>Home range size size in field voles (<em>Microtus agrestis</em>).
</p>


<h3>Format</h3>

<p>A data frame with 5 observations on the following 3 variables.
</p>
 <dl>
<dt>homeranges</dt><dd><p>a numeric vector</p>
</dd>
<dt>count</dt><dd><p>a numeric vector</p>
</dd> <dt>sex</dt><dd><p>a factor with
levels <code>female</code> and <code>male</code></p>
</dd> </dl>


<h3>Source</h3>

<p>Sandell, M., J. Agrell, S. Erlinge, and J. Nelson. 1991. Adult
philopatry and dispersal in the field vole <em>Microtus agrestis</em>.
<em>Oecologia</em> 86: 153-158.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xtabs(count~sex+homeranges,VoleDispersal)
barchart( xtabs(count~sex+homeranges,VoleDispersal), origin=0, auto.key=TRUE)
barchart(count~sex+homeranges,VoleDispersal, origin=0)
barchart(count~sex,groups=homeranges,VoleDispersal, origin=0)
barchart(count~sex,groups=homeranges,VoleDispersal, origin=0,stack=TRUE)
</code></pre>

<hr>
<h2 id='WalkingStickFemurs'>Walking Stick Femur Length</h2><span id='topic+WalkingStickFemurs'></span>

<h3>Description</h3>

<p>Two measures of femur length <code>femur.length</code> for each of 25
walking sticks (<em>Timema cristinae</em>). Note that <code>specimen</code>
is not coded as a factor.
</p>


<h3>Format</h3>

<p>A data frame with 50 observations on the following 2 variables.
</p>
 <dl>
<dt>specimen</dt><dd><p>a integer denoting specimen number.</p>
</dd>
<dt>femur.length</dt><dd><p>a numeric vector of femur length</p>
</dd> </dl>


<h3>Source</h3>

<p>Nosil, P. and B.J. Crespi. 2006. Experimental evidence that
predation promotes divergence in adaptive radiation. <em>Proceedings of
the National Academy of Sciences (USA)</em> 103: 9090-9095.
</p>


<h3>References</h3>

<p><a href="http://www.sfu.ca/biology/faculty/crespi/pdfs/96-Nosil&amp;Crespi2006PNAS.pdf">http://www.sfu.ca/biology/faculty/crespi/pdfs/96-Nosil&amp;Crespi2006PNAS.pdf</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>demo(sec15.6)
</code></pre>

<hr>
<h2 id='WalkingStickHeads'>Walking Stick Head Width</h2><span id='topic+WalkingStickHeads'></span>

<h3>Description</h3>

<p>Two measures of head width (<code>head.width</code>) for each of 25 walking sticks
(<em>Timema cristinae</em>).
</p>


<h3>Format</h3>

<p>A data frame with 50 observations on the following 2 variables.
</p>
 <dl>
<dt>specimen</dt><dd><p>a factor with levels <code>1-25</code></p>
</dd>
<dt>head.width</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Nosil, P. and B.J. Crespi. 2006. Experimental evidence that
predation promotes divergence in adaptive radiation. <em>Proceedings of
the National Academy of Sciences (USA)</em> 103: 9090-9095.
</p>


<h3>References</h3>

<p><a href="http://www.sfu.ca/biology/faculty/crespi/pdfs/96-Nosil&amp;Crespi2006PNAS.pdf">http://www.sfu.ca/biology/faculty/crespi/pdfs/96-Nosil&amp;Crespi2006PNAS.pdf</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>aggregate(head.width ~ specimen, data=WalkingStickHeads, mean) -&gt; WS
histogram(~ head.width, WS)
</code></pre>

<hr>
<h2 id='WeddellSeals'>Energetic Cost of Diving</h2><span id='topic+WeddellSeals'></span>

<h3>Description</h3>

<p>Comparison of oxygen consumption in feeding vs. non-feeding dives of the
same length in the Weddell seal (<em>Leptonychotes weddellii</em>).
</p>


<h3>Format</h3>

<p>A data frame with 10 observations on the following 3 variables.
</p>
 <dl>
<dt>individual</dt><dd><p>a numeric vector</p>
</dd>
<dt>oxygen.use.nonfeeding</dt><dd><p>a numeric vector</p>
</dd>
<dt>oxygen.use.feeding</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Williams, T.M., L.A. Fuiman, M. Horning, and R.W. Davis. 2004. The
cost of foraging by a marine predator, the Weddell seal <em>Leptonychotes
weddellii</em>: pricing by the stroke. <em>Journal of Experimental Biology</em>
207: 973-982.
</p>


<h3>References</h3>

<p><a href="http://jeb.biologists.org/cgi/content/full/207/6/973">http://jeb.biologists.org/cgi/content/full/207/6/973</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(oxygen.use.nonfeeding ~ oxygen.use.feeding, WeddellSeals)
</code></pre>

<hr>
<h2 id='WillsDebates'>Presidential &quot;Wills&quot;</h2><span id='topic+WillsDebates'></span>

<h3>Description</h3>

<p>Number of times a presidential candidate said &quot;will,&quot; &quot;shall,&quot; or &quot;going to&quot;
in presidential debates from 1960-2004 (years incomplete).
</p>


<h3>Format</h3>

<p>A data frame with 8 observations on the following 6 variables.
</p>
 <dl>
<dt>year</dt><dd><p>year of presidential debate(s)</p>
</dd>
<dt>winner</dt><dd><p>winner of the popular vote (may not be winner of
election)</p>
</dd> <dt>loser</dt><dd><p>loser of popular vote (may not be loser of
election)</p>
</dd> <dt>winner.wills</dt><dd><p>number of times will/shall used by
winner during debates</p>
</dd> <dt>loser.wills</dt><dd><p>number of times will/shall
used by loser during debates</p>
</dd> <dt>diff.wills</dt><dd><p>difference between
number of times will/shall used by two candidates</p>
</dd> </dl>


<h3>Examples</h3>

<pre><code class='language-R'>WillsDebates
</code></pre>

<hr>
<h2 id='WillsPresidents'>Presidential &quot;Wills&quot;</h2><span id='topic+WillsPresidents'></span>

<h3>Description</h3>

<p>Number of times a presidential candidate said &quot;will,&quot; &quot;shall,&quot; or &quot;going to&quot;
in presidential debates from 1960-2004 (years incomplete).
</p>


<h3>Format</h3>

<p>A data frame with 16 observations on the following 3 variables.
</p>
 <dl>
<dt>candidate</dt><dd><p>a character vector with the candidate's
name</p>
</dd> <dt>winner</dt><dd><p>a factor with levels <code>n</code> <code>y</code>
indicating whether the candidate won the election <code>y</code> or not.</p>
</dd>
<dt>wills</dt><dd><p>a numeric vector</p>
</dd> <dt>loser.wills</dt><dd><p>a numeric
vector</p>
</dd> <dt>difference</dt><dd><p>a numeric vector</p>
</dd> <dt>year</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>See Also</h3>

<p><code><a href="#topic+WillsDebates">WillsDebates</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>WillsPresidents
</code></pre>

<hr>
<h2 id='WolfTeeth'>Wolf Tooth Measurements</h2><span id='topic+WolfTeeth'></span>

<h3>Description</h3>

<p>Measurement (cm) of the distance between the canine and last molar teeth in
35 wolves.
</p>


<h3>Format</h3>

<p>A data frame with 35 observations of one variable.  </p>

<dl>
<dt>length</dt><dd><p>distance from canine to last molar teach (cm)</p>
</dd> </dl>


<h3>Source</h3>

<p>Whitlock, M. 1996. The heritability of fluctuating asymmetry and the
genetic control of developmental stability. <em>Proceedings of the Royal
Society, Series B</em> 263: 849-853.
</p>


<h3>References</h3>

<p><a href="http://rspb.royalsocietypublishing.org/content/263/1372/849.abstract">http://rspb.royalsocietypublishing.org/content/263/1372/849.abstract</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>histogram(~ length, WolfTeeth)
</code></pre>

<hr>
<h2 id='Wolves'>Inbreeding in Wolves</h2><span id='topic+Wolves'></span>

<h3>Description</h3>

<p>Inbreeding coefficient and the number of pups produced in 24 mated pairs of
wolves (<em>Canis lupus</em>) from 1983-2002.
</p>


<h3>Format</h3>

<p>A data frame with 24 observations on the following 2 variables.
</p>
 <dl>
<dt>inbreeding.coefficient</dt><dd><p>a numeric vector</p>
</dd>
<dt>pups</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Liberg, O.H., H. AndrÃ©n, H.-C. Pedersen, H. Sand, D. Sejberg, P.
Wabakken, M. Ãkesson, and S. Bensch. 2005. Severe inbreeding depression in a
wild wolf (<em>Canis lupus</em>) population. <em>Biology Letters</em> 1: 17-20.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Wolves
xyplot(inbreeding.coefficient ~ jitter(pups, amount=0.15), Wolves)
</code></pre>

<hr>
<h2 id='WorldCup'>World Cup Goals</h2><span id='topic+WorldCup'></span>

<h3>Description</h3>

<p>Number of goals per team during the 2002 World Cup.
</p>


<h3>Format</h3>

<p>A data frame with 7 observations on the following 2 variables.
</p>
 <dl>
<dt>score</dt><dd><p>a numeric vector</p>
</dd> <dt>count</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>Examples</h3>

<pre><code class='language-R'>xyplot(count ~ score, WorldCup, type='h', lwd=4)
barchart(count ~ score, WorldCup, origin=0, horizontal=FALSE)
</code></pre>

<hr>
<h2 id='WrasseSexes'>Distribution of Wrasses</h2><span id='topic+WrasseSexes'></span>

<h3>Description</h3>

<p>Number and sex of adult wrasses in a section of the Great Barrier Reef.
</p>


<h3>Format</h3>

<p>A data frame with 3 observations on the following 3 variables.
</p>
 <dl>
<dt>males</dt><dd><p>a numeric vector</p>
</dd> <dt>females</dt><dd><p>a
numeric vector</p>
</dd> <dt>count</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Examples</h3>

<pre><code class='language-R'>xtabs(count ~ males + females, WrasseSexes)
</code></pre>

<hr>
<h2 id='YeastGenes'>Yeast Regulatory Genes</h2><span id='topic+YeastGenes'></span>

<h3>Description</h3>

<p>Number of genes regulated by 109 yeast regulatory genes.
</p>


<h3>Format</h3>

<p>A data frame with 6 observations on the following 2 variables.
</p>
 <dl>
<dt>genes.controlled</dt><dd><p>a numeric vector</p>
</dd>
<dt>count</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Guelzim, N., S. Bottani, P. Bourgine and F. KÃ©pÃ¨s. 2002. Topological
and causal structure of the yeast transcriptional regulatory network.
<em>Nature Genetics</em> 31: 60-63.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(YeastGenes)
barchart(count ~ genes.controlled , origin=0, YeastGenes, horizontal=FALSE)
</code></pre>

<hr>
<h2 id='ZebraFinchBeaks'>Mate Preference in Zebra Finches</h2><span id='topic+ZebraFinchBeaks'></span>

<h3>Description</h3>

<p>Percentage of time that a female spent next to a carotenoid-supplemented
male Zebra Finch compared to his non-supplemented brother.
</p>


<h3>Format</h3>

<p>A numeric vector with 10 observations.</p>


<h3>Source</h3>

<p>Blount, J.D., N.B. Metcalfe, T.R. Birkhead, P.F. Surai. 2003.
Carotenoid modulation of immune function and sexual attractiveness in Zebra
Finches. <em>Science</em> 300: 125-127.
</p>


<h3>References</h3>

<p><a href="http://www.sciencemag.org/cgi/content/abstract/300/5616/125">http://www.sciencemag.org/cgi/content/abstract/300/5616/125</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ZebraFinchBeaks
</code></pre>

<hr>
<h2 id='ZebraFinches'>Zebra Finch Carotenoids</h2><span id='topic+ZebraFinches'></span>

<h3>Description</h3>

<p>Data on cell-mediated immunocompetence (<code>PHA</code>) and humoral immunity
(<code>SRBC</code>) in Zebra Finches that received supplemental carotenoids
(<code>CAROT</code>) and those that did not (<code>NO</code>).
</p>


<h3>Format</h3>

<p>A data frame with 20 observations on the following 3 variables.
</p>
 <dl>
<dt>treatment</dt><dd><p>a factor with levels: <code>CAROT</code> and
<code>NO</code></p>
</dd> <dt>PHA</dt><dd><p>a numeric vector</p>
</dd> <dt>SRBC</dt><dd><p>a
numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>McGraw, K.J. and D.R. Ardia. 2003. Carotenoids, immunocompetence,
and the information content of sexual colors: an experimental test.
<em>The American Naturalist</em> 162: 704-712.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ZebraFinches
</code></pre>

<hr>
<h2 id='ZooMortality'>Home Range Size and Mortality</h2><span id='topic+ZooMortality'></span>

<h3>Description</h3>

<p>Home range size (<code class="reqn">\log_{10}</code>) and captive infant mortality (
for 20 species of carnivores.
</p>


<h3>Format</h3>

<p>A data frame with 20 observations on the following 2 variables.
</p>
 <dl>
<dt>log.homerange</dt><dd><p>a numeric vector</p>
</dd>
<dt>mortality</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p>Clubb, R. and G. Mason. 2003. Captivity effects on wide ranging
carnivores. <em>Nature</em> 425: 473-474.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(ZooMortality)
</code></pre>

<hr>
<h2 id='Zooplankton'>Zooplankton Depredation</h2><span id='topic+Zooplankton'></span>

<h3>Description</h3>

<p>Diversity of zooplankton (<code>zooplankton</code>) prey in each of 5 replicate
blocks (<code>block</code>) of three treatment levels (<code>treatment</code>). By
default, <code>block</code> is not coded as a factor.
</p>


<h3>Format</h3>

<p>A data frame with 15 observations on the following 3 variables.
</p>
 <dl>
<dt>treatment</dt><dd><p>a factor with levels <code>control</code>,
<code>high</code>, and <code>low</code></p>
</dd> <dt>zooplankton</dt><dd><p>a numeric vector</p>
</dd>
<dt>block</dt><dd><p>a numeric vector</p>
</dd> </dl>


<h3>Source</h3>

<p><em>inferred from</em> SvanbÃ¤ck, R. and D.I. Bolnick. 2007.
Intraspecific competition drives increased resource use diversity within a
natural population. <em>Proceedings of the Royal Society of London Series
B, Biological Sciences</em> 274: 839-844.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Zooplankton

Zooplankton$block &lt;- factor(Zooplankton$block)
str(Zooplankton)

aov.fit &lt;- aov(zooplankton ~ block + treatment,
  data = Zooplankton)
summary(aov.fit)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
