<!DOCTYPE html><html><head><title>Help for package ABHgenotypeR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ABHgenotypeR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ABHgenotypeR'><p>ABHgenotypeR: A package for easy visualization and manipulating of ABH genotypes.</p></a></li>
<li><a href='#correctStretches'><p>Correct short miscalled stretches based on flanking alleles.</p></a></li>
<li><a href='#correctUndercalledHets'><p>Correct undercalled heterozygous sites based on flanking alleles.</p></a></li>
<li><a href='#imputeByFlanks'><p>Impute missing genotypes based on flanking alleles</p></a></li>
<li><a href='#plotAlleleFreq'><p>Plot the  parental allele frequencies along the chromosomes.</p></a></li>
<li><a href='#plotCompareGenos'><p>Compare to genotype matrices</p></a></li>
<li><a href='#plotGenos'><p>Plot graphical genotypes.</p></a></li>
<li><a href='#plotMarkerDensity'><p>Plot the marker density along the chromosomes.</p></a></li>
<li><a href='#readABHgenotypes'><p>Read in the output of the genosToABH plugin.</p></a></li>
<li><a href='#reportGenos'><p>Report the total and relative number of each allele in a genotype object.</p></a></li>
<li><a href='#writeABHgenotypes'><p>Export a genotype list to a .csv file.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Easy Visualization of ABH Genotypes</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-02-04</td>
</tr>
<tr>
<td>Description:</td>
<td>Easy to use functions to visualize marker data
    from biparental populations. Useful for both analyzing and
    presenting genotypes in the ABH format.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://github.com/StefanReuscher/ABHgenotypeR">http://github.com/StefanReuscher/ABHgenotypeR</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="http://github.com/StefanReuscher/ABHgenotypeR/issues">http://github.com/StefanReuscher/ABHgenotypeR/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, reshape2, utils</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-02-04 05:07:28 UTC; Stefan</td>
</tr>
<tr>
<td>Author:</td>
<td>Stefan Reuscher [aut, cre],
  Tomoyuki Furuta [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Stefan Reuscher &lt;reuscher.stefan@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-02-04 11:27:29</td>
</tr>
</table>
<hr>
<h2 id='ABHgenotypeR'>ABHgenotypeR: A package for easy visualization and manipulating of ABH genotypes.</h2><span id='topic+ABHgenotypeR'></span><span id='topic+ABHgenotypeR-package'></span>

<h3>Description</h3>

<p>The ABHgenotypeR packages is meant as a companion packages in between the TASSEL
GBS pipeline and R/qtl. It allows easy visualization of ABH-encoded genotypes
in a .csv format as ouptut by the TASSEL ABHGenosPlugin which is also the
format necessary for R/qtl input.
</p>

<hr>
<h2 id='correctStretches'>Correct short miscalled stretches based on flanking alleles.</h2><span id='topic+correctStretches'></span>

<h3>Description</h3>

<p>Correct short miscalled stretches based on flanking alleles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>correctStretches(inputGenos = "genotypes", maxHapLength = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="correctStretches_+3A_inputgenos">inputGenos</code></td>
<td>
<p>A genotypes list object.</p>
</td></tr>
<tr><td><code id="correctStretches_+3A_maxhaplength">maxHapLength</code></td>
<td>
<p>The maximum length of stretches flanked
by non-heterzygous sites that are changed. If set to 1
(default) only AXA or BXB will be corrected. If set to 2, both AXA and AXYA
(or BXB and BXYB) will be corrected.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genotype object in which short miscalled stretches are
corrected if both flanking alleles match.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: corrStretchGenos &lt;- correctStretches(genotypes, maxHapLength = 3)
</code></pre>

<hr>
<h2 id='correctUndercalledHets'>Correct undercalled heterozygous sites based on flanking alleles.</h2><span id='topic+correctUndercalledHets'></span>

<h3>Description</h3>

<p>Correct undercalled heterozygous sites based on flanking alleles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>correctUndercalledHets(inputGenos = "genotypes", maxHapLength = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="correctUndercalledHets_+3A_inputgenos">inputGenos</code></td>
<td>
<p>A genotypes list object.</p>
</td></tr>
<tr><td><code id="correctUndercalledHets_+3A_maxhaplength">maxHapLength</code></td>
<td>
<p>The maximum length of not heterozygous stretches flanked
by heterzygous sites that are changed to heterozygous. If set to 1
(default) only HAH or HBH will be corrected. If set to 2, both HAH and HAAH
(or HBH and HBBH) will be corrected.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genotype object in which undercalled heterozygous sites are
corrected if both flanking alleles match.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: corrUndHetsGenos &lt;- correctUndercalledHets(genotypes, maxHapLength = 3)
</code></pre>

<hr>
<h2 id='imputeByFlanks'>Impute missing genotypes based on flanking alleles</h2><span id='topic+imputeByFlanks'></span>

<h3>Description</h3>

<p>Impute missing genotypes based on flanking alleles
</p>


<h3>Usage</h3>

<pre><code class='language-R'>imputeByFlanks(inputGenos = "genotypes")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="imputeByFlanks_+3A_inputgenos">inputGenos</code></td>
<td>
<p>A genotypes list object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genotype object in which missing data is imputed based on flanking
alleles. Any number of N is replaced by either A, B or N if the alleles which flank the N match
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: imputedGenos &lt;- imputeByFlanks(genotypes)
</code></pre>

<hr>
<h2 id='plotAlleleFreq'>Plot the  parental allele frequencies along the chromosomes.</h2><span id='topic+plotAlleleFreq'></span>

<h3>Description</h3>

<p>Plot the  parental allele frequencies along the chromosomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotAlleleFreq(genos = "genotypes")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotAlleleFreq_+3A_genos">genos</code></td>
<td>
<p>The output of readABHgenotypes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot of parental allele frequencies along the chromosomes. If the
output is assigned a name a ggplot2 object is returned for further
manipulation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: plotAlleleFreq(genotypes)
## Not run: p &lt;- plotAlleleFreq(genotypes)
</code></pre>

<hr>
<h2 id='plotCompareGenos'>Compare to genotype matrices</h2><span id='topic+plotCompareGenos'></span>

<h3>Description</h3>

<p>Compare to genotype matrices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotCompareGenos(genos_1 = "genotypes_1", genos_2 = "genotypes_2",
  markerToPlot = "all", individualsToPlot = "all", chromToPlot = "all",
  CompColors = c("#000000", "#E69F00"), textSize = 12,
  showMarkerNames = FALSE, showIndividualNames = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotCompareGenos_+3A_genos_1">genos_1</code></td>
<td>
<p>Output of readABHgenotypes</p>
</td></tr>
<tr><td><code id="plotCompareGenos_+3A_genos_2">genos_2</code></td>
<td>
<p>Output of readABHgenotypes. Note that both genos object need to
have identical numbers of marker x individuals.</p>
</td></tr>
<tr><td><code id="plotCompareGenos_+3A_markertoplot">markerToPlot</code></td>
<td>
<p>A character vector of marker names which appear in the
plot. Defaults to all.</p>
</td></tr>
<tr><td><code id="plotCompareGenos_+3A_individualstoplot">individualsToPlot</code></td>
<td>
<p>A character vector of individual names which appear
in the plot. Defaults to all.</p>
</td></tr>
<tr><td><code id="plotCompareGenos_+3A_chromtoplot">chromToPlot</code></td>
<td>
<p>A character vector of chromosome names which appear in the
plot. Defaults to all.</p>
</td></tr>
<tr><td><code id="plotCompareGenos_+3A_compcolors">CompColors</code></td>
<td>
<p>A character vector of length 2 giving the color names or
values to use for differnt and identical markers.
Defaults to black and orange.</p>
</td></tr>
<tr><td><code id="plotCompareGenos_+3A_textsize">textSize</code></td>
<td>
<p>The size of all text elements in the plot. Useful for making a
nice plot. Defaults to 12.</p>
</td></tr>
<tr><td><code id="plotCompareGenos_+3A_showmarkernames">showMarkerNames</code></td>
<td>
<p>Show the marker names along the x axis. This and
showIndividualnames are useful when you display only a few markers and
want them labeled. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="plotCompareGenos_+3A_showindividualnames">showIndividualNames</code></td>
<td>
<p>Show individual names along the y axis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A graphical comparison of genotypes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: plotCompareGenos(preImpGenotypes,postImpGenotypes)
## Not run: #for more examples see plotGenos()
</code></pre>

<hr>
<h2 id='plotGenos'>Plot graphical genotypes.</h2><span id='topic+plotGenos'></span>

<h3>Description</h3>

<p>Plot graphical genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotGenos(genos = "genotypes", markerToPlot = "all",
  individualsToPlot = "all", chromToPlot = "all",
  alleleColors = c("#56B4E9", "#E69F00", "#009E73", "#000000"),
  textSize = 12, showMarkerNames = FALSE, showIndividualNames = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotGenos_+3A_genos">genos</code></td>
<td>
<p>The output of readABHgenotypes</p>
</td></tr>
<tr><td><code id="plotGenos_+3A_markertoplot">markerToPlot</code></td>
<td>
<p>A character vector of marker names which appear in the
plot. Defaults to all.</p>
</td></tr>
<tr><td><code id="plotGenos_+3A_individualstoplot">individualsToPlot</code></td>
<td>
<p>A character vector of individual names which appear
in the plot. Defaults to all.</p>
</td></tr>
<tr><td><code id="plotGenos_+3A_chromtoplot">chromToPlot</code></td>
<td>
<p>A character vector of chromosome names which appear in the
plot. Defaults to all.</p>
</td></tr>
<tr><td><code id="plotGenos_+3A_allelecolors">alleleColors</code></td>
<td>
<p>A character vector of length 4 giving the color names or
values to use for the A,B,H and n.d genotypes. Defaults to orange, blue,
green and black.</p>
</td></tr>
<tr><td><code id="plotGenos_+3A_textsize">textSize</code></td>
<td>
<p>The size of all text elements in the plot. Useful for making a
nice plot. Defaults to 12.</p>
</td></tr>
<tr><td><code id="plotGenos_+3A_showmarkernames">showMarkerNames</code></td>
<td>
<p>Show the marker names along the x axis. This and
showIndividualNames are useful when you display only a few markers and
want them labeled. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="plotGenos_+3A_showindividualnames">showIndividualNames</code></td>
<td>
<p>Show individual names along the y axis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Graphical genotypes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: plotGenos(genotypes)
markerNames &lt;- c("marker1", "marker2", "marker3")
individualNames &lt;- c("F2_100", "F2_101", "F2_102", "F2_103")
someColors &lt;- c("black", "red", "gold", "white")
## Not run: plotgenos(genotypes, markerNames, individualNames, 1:3, someColors)

## Not run: p &lt;- plotGenos(genotypes)
</code></pre>

<hr>
<h2 id='plotMarkerDensity'>Plot the marker density along the chromosomes.</h2><span id='topic+plotMarkerDensity'></span>

<h3>Description</h3>

<p>Plot the marker density along the chromosomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotMarkerDensity(genos = "genotypes")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotMarkerDensity_+3A_genos">genos</code></td>
<td>
<p>The output of readABHgenotypes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot of marker densities along the chromosomes. If the output is
assigned a name a ggplot2 object is returned for further manipulation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: plotMarkerDensity(genotypes)
## Not run: p &lt;- plotMarkerDensity(genotypes)
</code></pre>

<hr>
<h2 id='readABHgenotypes'>Read in the output of the genosToABH plugin.</h2><span id='topic+readABHgenotypes'></span>

<h3>Description</h3>

<p>Read in the output of the genosToABH plugin.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readABHgenotypes(pathToABH, nameA = "A", nameB = "B", readPos = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readABHgenotypes_+3A_pathtoabh">pathToABH</code></td>
<td>
<p>The path and filename of the input file.</p>
</td></tr>
<tr><td><code id="readABHgenotypes_+3A_namea">nameA</code></td>
<td>
<p>Name of the parent represented by &quot;A&quot; in the input file.</p>
</td></tr>
<tr><td><code id="readABHgenotypes_+3A_nameb">nameB</code></td>
<td>
<p>Name of the parent represented by &quot;B&quot; in the input file.</p>
</td></tr>
<tr><td><code id="readABHgenotypes_+3A_readpos">readPos</code></td>
<td>
<p>Should the function attempt read the physical position of
markers from the input ?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input files should be a .csv file holding genotypes as specified by
the qtl package and its &quot;csvs&quot; format.
All characters in the genotype matrix which are not either A,B or H
will be set to N.
If readPos = TRUE (default) marker names must conform to S1_123456 meaning 123456 bp
on chromosome 1. If FALSE, pos is set to NULL and needs to be manually constructed
as shown in the examples. Note that this might throw off some plotting function.
</p>


<h3>Value</h3>

<p>A genotype list object which holds the information from the input file.
This list is the fundamental datastructure used by the other functions in this
package. See the vignette for what each item in the list is.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: genotypes &lt;- readABHgenotypes("./genotypes.csv", "NB", "OL")

 ## Not run: otherGenotypes &lt;- readABHgenotypes("./otherGenotypes.csv", readPos = FALSE)
 #arbitrary position to keep marker order intact
 ## Not run: therGenotypes$pos &lt;- 1:length(otherGenotypes$marker_names)
</code></pre>

<hr>
<h2 id='reportGenos'>Report the total and relative number of each allele in a genotype object.</h2><span id='topic+reportGenos'></span>

<h3>Description</h3>

<p>Report the total and relative number of each allele in a genotype object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reportGenos(genos1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reportGenos_+3A_genos1">genos1</code></td>
<td>
<p>A genotypes list object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Console output of the total an relative number of each allele.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: reportGenos(preImputation)
</code></pre>

<hr>
<h2 id='writeABHgenotypes'>Export a genotype list to a .csv file.</h2><span id='topic+writeABHgenotypes'></span>

<h3>Description</h3>

<p>Export a genotype list to a .csv file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeABHgenotypes(genos = "genotypes", outfile = "./outfile.csv")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writeABHgenotypes_+3A_genos">genos</code></td>
<td>
<p>The output of readABHgenotypes or one of the impuation/error
correction functions.</p>
</td></tr>
<tr><td><code id="writeABHgenotypes_+3A_outfile">outfile</code></td>
<td>
<p>The path and filename of the output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A file which can be used in R/qtl or elsewhere.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: writeABHgenotypes(genotypes, outfile = "./outfile_name.csv")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
