<!DOCTYPE html><html><head><title>Help for package Ac3net</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Ac3net}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Ac3net'>
<p>Implement Ac3net algorithm with a single function.</p></a></li>
<li><a href='#Ac3net.commonlinks'>
<p>Find the common links between the network lists.</p></a></li>
<li><a href='#Ac3net.cutoff'>
<p>Calculates a quick cutoff value from an adjacency matrix</p></a></li>
<li><a href='#Ac3net.filtersames'>
<p>The correlations of the same variables are set to zero.</p></a></li>
<li><a href='#Ac3net.getDifferentLinks'>
<p>Find the different links between the network lists.</p></a></li>
<li><a href='#Ac3net.getDirectedOrDualLinks'>
<p>Find the unique directed or dual links in the network.</p></a></li>
<li><a href='#Ac3net.maxmim'>
<p>Find the maximum valued partner of each variable.</p></a></li>
<li><a href='#Ac3net.MTC'>
<p>Return adjusted p-values of the adjancency matrix.</p></a></li>
<li><a href='#Ac3net.performance'>
<p>Find the performance of the first network based on the second.</p></a></li>
<li><a href='#expdata'>
<p>Real E.coli Gene Expression Dataset</p></a></li>
<li><a href='#truenet'>
<p>The known intreactions of E.coli from RegulonDB database.</p></a></li>
<li><a href='#zeroindx'>
<p>The indices to force to zero correlations.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Inferring Directional Conservative Causal Core Gene Networks</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-02-05</td>
</tr>
<tr>
<td>Author:</td>
<td>Gokmen Altay</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Gokmen Altay &lt;altaylabs@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Infers directional conservative causal core (gene) networks. It is an advanced version of the algorithm C3NET by providing directional network. Gokmen Altay (2018) &lt;<a href="https://doi.org/10.1101%2F271031">doi:10.1101/271031</a>&gt;, bioRxiv.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.0), data.table</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-02-26 20:46:39 UTC; altay</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-02-26 22:08:20 UTC</td>
</tr>
</table>
<hr>
<h2 id='Ac3net'>
Implement Ac3net algorithm with a single function.
</h2><span id='topic+Ac3net'></span>

<h3>Description</h3>

<p><code>Ac3net</code> A quick way to implement Ac3net algorithm with a single function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ac3net(DataOrMim, processed=FALSE, ratio_ = 0.002, PCmincutoff=0.6,
PCmaxcutoff=0.96, cutoff=0, estmethod='pearson', pval=1, 
iterations=10, MTC=FALSE, MTCmethod="BH" )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Ac3net_+3A_dataormim">DataOrMim</code></td>
<td>
<p>It can be an adjacency matrix or data matrix. If adjacency matrix then the element at row i and column j corresponds to the correlation or mutual information between variables row i and column j. Rows and columns must have the variable names. Also the 'processed' parameter must be set to TRUE. If it is data matrix then rows are variables and columns are the samples. Rows must have the variable names. Also the 'processed' parameter is already set to FALSE as the default usage case is data matrix. </p>
</td></tr>
<tr><td><code id="Ac3net_+3A_processed">processed</code></td>
<td>
<p>Set to FALSE if the input is a data matrix, set to TRUE if the input is adjacency matrix.</p>
</td></tr>  
<tr><td><code id="Ac3net_+3A_ratio_">ratio_</code></td>
<td>
<p>The ratio of all the elements in the input adjacency matrix (even if the matrix is symmetric), which the user considers as significant portion.</p>
</td></tr>
<tr><td><code id="Ac3net_+3A_pcmincutoff">PCmincutoff</code></td>
<td>
<p>The minimum absolute Pearson correlation value ( by default but can be different) where the below of is not considered as significant.</p>
</td></tr>
<tr><td><code id="Ac3net_+3A_pcmaxcutoff">PCmaxcutoff</code></td>
<td>
<p>The maximum absolute Pearson correlation value ( by default but can be different) where all the above of is considered as significant.</p>
</td></tr>
<tr><td><code id="Ac3net_+3A_cutoff">cutoff</code></td>
<td>
<p>Default is 0 means inactive. If set, it must be betwen 0 and 1, and the input ise used directly as cutoff.</p>
</td></tr>
<tr><td><code id="Ac3net_+3A_estmethod">estmethod</code></td>
<td>
<p>The default method is 'pearson' assuming that the data was normalized (at least with Log-2). If the data is not normalized, its recomended setting is 'spearman'.</p>
</td></tr>  
<tr><td><code id="Ac3net_+3A_pval">pval</code></td>
<td>
<p>Default is 1, means do not use p-values cutoff. If &lt; 1, then the corresponding cutoff is calculated and the adjacency matrix is filtered based on this cutoff calculated by the input pval p-value.</p>
</td></tr>  
<tr><td><code id="Ac3net_+3A_iterations">iterations</code></td>
<td>
<p>Number of iterations to get null distribuiton from multiple shuffling of the data.</p>
</td></tr>  
<tr><td><code id="Ac3net_+3A_mtc">MTC</code></td>
<td>
<p>If TRUE,as the default status, the the returned p-value matrix has the adjusted p-values</p>
</td></tr>
<tr><td><code id="Ac3net_+3A_mtcmethod">MTCmethod</code></td>
<td>
<p>The Multiple Testing Correction method. BH is default. See R p.adjust function for options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Ac3net</code> The input can be an adjacency matrix or data matrix. If adjacency matrix then the element at row i and column j corresponds to the correlation or mutual information between variables row i and column j. Rows and columns must have the variable names. Also the 'processed' parameter must be set to TRUE. If it is data matrix then rows are variables and columns are the samples. Rows must have the variable names. Also the 'processed' parameter is already set to FALSE as the default usage case is data matrix.
If MTC is set to TRUE then it sliminates non-significants based on adjusted p-values but this process takes too long.
If MTC is set to FALSE as default then cutoff parameter is used.
If the cutoff parameter is set to a value other than 0, then it is used as cutoff directly.
If the cutoff wanted to be calculated very quicly, for comparison purposes for example, then first it is set to 0 as default and then it is calculated between minimum and maximum input cutoffs as the value that correspond to the input ratio_ parameter from top to down.
</p>


<h3>Value</h3>

<p><code>Ac3net</code> returns a network matrix where the first column is the variable names of the input adjacency matrix and second columns have the variables names that correspond to the absolute maximum value for the variable in the first column.  
</p>


<h3>Author(s)</h3>

<p>Gokmen Altay
</p>


<h3>References</h3>

<p>G. Altay,&quot;Directed Conservative Causal Core Gene Networks&quot;, bioRxiv, 2018.
G. Altay, F. Emmert-Streib, &quot;Inferring the conservative causal core of gene regulatory networks&quot;, BMC Systems Biology (2010) 4:132. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Ac3net.maxmim">Ac3net.maxmim</a></code>, <code><a href="#topic+Ac3net.commonlinks">Ac3net.commonlinks</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'># data(expdata)
# ac3net.net &lt;- Ac3net(DataOrMim=expdata, processed=FALSE, ratio_ = 0.002, PCmincutoff=0.6,
#              PCmaxcutoff=0.96, cutoff=0, estmethod='pearson', pval=1, 
# iterations=10, MTC=FALSE, MTCmethod="BH" )
</code></pre>

<hr>
<h2 id='Ac3net.commonlinks'>
Find the common links between the network lists.
</h2><span id='topic+Ac3net.commonlinks'></span>

<h3>Description</h3>

<p><code>Ac3net.commonlinks</code> takes two network list matrices and returns directional or undirectional common pairs between the two regarding the first two columns of the networks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ac3net.commonlinks(net1,net2, directed=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Ac3net.commonlinks_+3A_net1">net1</code></td>
<td>
<p>A matrix that represents the links between two variables at each row. Namely, the element at row i and column 1 and the element at row i and column 2 are assumed to be paired or linked. There is no connection assumed among rows.</p>
</td></tr>
<tr><td><code id="Ac3net.commonlinks_+3A_net2">net2</code></td>
<td>
<p>A matrix that represents the links between two variables at each row. Namely, the element at row i and column 1 and the element at row i and column 2 are assumed to be paired or linked. There is no connection assumed among rows.</p>
</td></tr>
<tr><td><code id="Ac3net.commonlinks_+3A_directed">directed</code></td>
<td>
<p> It is TRUE by default, which means there is  directionality assumed from column 1 to column 2 at each matrices. Therefore common links are the directionale common links. If it is FALSE, then the pairs assumed to be undirectional and all the common pairs are taken as common. E.g. If there is A&ndash;B in net1 and B&ndash;A in net2 then they are considered as common.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Ac3net.commonlinks</code> takes two network list matrices and based on the first two columns it makes comparison. If the argument directed is equal to TRUE, as in the default case, the it returns only directionally common pairs from column 1 to column 2 of the matrices. E.g. For a pair of A&ndash;&gt;B in the first network, if there is also A&ndash;&gt;B in the second network then it is taken as common but the opposite direction B&ndash;&gt;A is not counted. If the argument directed is equal to FALSE, then B&ndash;&gt;A is also counted, namely all the common pairs, with no directional considerations, is taken as common between the two network. 
</p>


<h3>Value</h3>

<p><code>Ac3net.commonlinks</code>  returns a network matrix that correspond to the common pairs between net1 and net2. 
</p>


<h3>Author(s)</h3>

<p>Gokmen Altay
</p>


<h3>References</h3>

<p>G. Altay,&quot;Directed Conservative Causal Core Gene Networks&quot;, bioRxiv, 2018.
G. Altay, F. Emmert-Streib, &quot;Inferring the conservative causal core of gene regulatory networks&quot;, BMC Systems Biology (2010) 4:132. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Ac3net.maxmim">Ac3net.maxmim</a></code>, <code><a href="#topic+Ac3net.cutoff">Ac3net.cutoff</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'># net1
# net2
# net &lt;- Ac3net.commonlinks(net1,net2, directed=TRUE)
</code></pre>

<hr>
<h2 id='Ac3net.cutoff'>
Calculates a quick cutoff value from an adjacency matrix
</h2><span id='topic+Ac3net.cutoff'></span>

<h3>Description</h3>

<p><code>Ac3net.cutoff</code> A quick way to get a very rough estimate of the cutoff value for the significants of the input adjacency matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ac3net.cutoff(mim, ratio_ = 0.002, PCmincutoff=0.6, PCmaxcutoff=0.96)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Ac3net.cutoff_+3A_mim">mim</code></td>
<td>
<p>An adjacency matrix, where the element at row i and column j corresponds to the correlation or mutual information between variables row i and column j. Row and columns of the matrix must have the variable names.</p>
</td></tr>
<tr><td><code id="Ac3net.cutoff_+3A_ratio_">ratio_</code></td>
<td>
<p>The ratio of all the elements in the input adjacency matrix (even if the matrix is symmetric), which the user considers as significant portion.</p>
</td></tr>
<tr><td><code id="Ac3net.cutoff_+3A_pcmincutoff">PCmincutoff</code></td>
<td>
<p>The minimum absolute Pearson correlation value ( by default but can be different) where the below of is not considered as significant.</p>
</td></tr>
<tr><td><code id="Ac3net.cutoff_+3A_pcmaxcutoff">PCmaxcutoff</code></td>
<td>
<p>The maximum absolute Pearson correlation value ( by default but can be different) where all the above of is considered as significant.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Ac3net.cutoff</code> takes an adjacency matrix, ratio and minimum absolute Pearson correlation information. Then return a cutoff value that either correspond to the input ratio or the minimum.
</p>


<h3>Value</h3>

<p><code>Ac3net.cutoff</code> returns a scaler as a very rough estimate of the cutoff value for the significants of the input adjacency matrix. 
</p>


<h3>Author(s)</h3>

<p>Gokmen Altay
</p>


<h3>References</h3>

<p>G. Altay,&quot;Directed Conservative Causal Core Gene Networks&quot;, bioRxiv, 2018.
G. Altay, F. Emmert-Streib, &quot;Inferring the conservative causal core of gene regulatory networks&quot;, BMC Systems Biology (2010) 4:132. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Ac3net.maxmim">Ac3net.maxmim</a></code>, <code><a href="#topic+Ac3net.commonlinks">Ac3net.commonlinks</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'> data(expdata)
 mim &lt;- cor(t(expdata), method = 'pearson')
 cutoff &lt;- Ac3net.cutoff(mim=mim, ratio_ = 0.002,
              PCmincutoff=0.6, PCmaxcutoff=0.96)
</code></pre>

<hr>
<h2 id='Ac3net.filtersames'>
The correlations of the same variables are set to zero.
</h2><span id='topic+Ac3net.filtersames'></span>

<h3>Description</h3>

<p><code>Ac3net.filtersames</code> The correlations of the same variables are set to zero as they are likly to have the maximum correlations already.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ac3net.filtersames(mim)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Ac3net.filtersames_+3A_mim">mim</code></td>
<td>
<p>An adjacency matrix, where the element at row i and column j corresponds to the correlation or mutual information between variables row i and column j. Row and columns of the matrix must have the variable names. Row names must be available.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Ac3net.filtersames</code> takes an adjacency matrix with variable names are on the rows. Then it sets 0 for the self correlations of the variables by checking their names.
</p>


<h3>Value</h3>

<p><code>Ac3net.filtersames</code> returns an adjacency matrix with no self dependency. 
</p>


<h3>Author(s)</h3>

<p>Gokmen Altay
</p>


<h3>References</h3>

<p>G. Altay,&quot;Directed Conservative Causal Core Gene Networks&quot;, bioRxiv, 2018.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Ac3net.maxmim">Ac3net.maxmim</a></code>, <code><a href="#topic+Ac3net.commonlinks">Ac3net.commonlinks</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'># data(expdata)
# mim &lt;- cor(t(expdata), method = 'pearson')
# mim &lt;- Ac3net.filtersames(mim)
</code></pre>

<hr>
<h2 id='Ac3net.getDifferentLinks'>
Find the different links between the network lists.
</h2><span id='topic+Ac3net.getDifferentLinks'></span>

<h3>Description</h3>

<p><code>Ac3net.getDifferentLinks</code> takes two network list matrices and returns directional or undirectional different pairs between the two regarding the first two columns of the networks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ac3net.getDifferentLinks(net1,net2, directed=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Ac3net.getDifferentLinks_+3A_net1">net1</code></td>
<td>
<p>A matrix that represents the links between two variables at each row. Namely, the element at row i and column 1 and the element at row i and column 2 are assumed to be paired or linked. There is no connection assumed among rows.</p>
</td></tr>
<tr><td><code id="Ac3net.getDifferentLinks_+3A_net2">net2</code></td>
<td>
<p>A matrix that represents the links between two variables at each row. Namely, the element at row i and column 1 and the element at row i and column 2 are assumed to be paired or linked. There is no connection assumed among rows.</p>
</td></tr>
<tr><td><code id="Ac3net.getDifferentLinks_+3A_directed">directed</code></td>
<td>
<p> It is TRUE by default, which means there is  directionality assumed from column 1 to column 2 at each matrices. if TRUE, then selects only the different links from A&ndash;&gt;B with respect to net2. If it is FALSE, then the pairs assumed to be undirectional and all the dual pairs are considered as common and filtered. E.g. If there is A&ndash;B in net1 and B&ndash;A in net2 then they are considered as common and filtered out.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Ac3net.getDifferentLinks</code> takes two network list matrices and based on the first two columns it makes comparison. If the argument directed is equal to TRUE, as in the default case, the it returns only directionally different pairs from column 1 to column 2 of the matrices. E.g. For a pair of A&ndash;&gt;B in the first network, if there is also A&ndash;&gt;B in the second network then it is taken as common and filtered out but the opposite direction B&ndash;&gt;A is not counted. If the argument directed is equal to FALSE, then B&ndash;&gt;A is also counted and filtered out as well for net1 with compared to net2. 
</p>


<h3>Value</h3>

<p><code>Ac3net.getDifferentLinks</code> returns a network matrix that correspond to the different pairs of net1 with compared to net2. 
</p>


<h3>Author(s)</h3>

<p>Gokmen Altay
</p>


<h3>References</h3>

<p>G. Altay,&quot;Directed Conservative Causal Core Gene Networks&quot;, bioRxiv, 2018.
G. Altay, F. Emmert-Streib, &quot;Inferring the conservative causal core of gene regulatory networks&quot;, BMC Systems Biology (2010) 4:132. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Ac3net.maxmim">Ac3net.maxmim</a></code>, <code><a href="#topic+Ac3net.cutoff">Ac3net.cutoff</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'># net1
# net2
# diffnet &lt;- Ac3net.getDifferentLinks(net1,net2, directed=TRUE)
</code></pre>

<hr>
<h2 id='Ac3net.getDirectedOrDualLinks'>
Find the unique directed or dual links in the network.
</h2><span id='topic+Ac3net.getDirectedOrDualLinks'></span>

<h3>Description</h3>

<p><code>Ac3net.getDirectedOrDualLinks</code> takes a network list matrix and returns a unique network; either directionals or the duals. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ac3net.getDirectedOrDualLinks(net1, dual_=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Ac3net.getDirectedOrDualLinks_+3A_net1">net1</code></td>
<td>
<p>A matrix that represents the links between two variables at each row. Namely, the element at row i and column 1 and the element at row i and column 2 are assumed to be paired or linked. There is no connection assumed among rows.</p>
</td></tr>
<tr><td><code id="Ac3net.getDirectedOrDualLinks_+3A_dual_">dual_</code></td>
<td>
<p> It is FALSE by default, which means there is  directionality assumed from column 1 to column 2 at each matrices. A&ndash;B and B&ndash;A is cases filtered out. If TRUE, then only the dual pairs selected.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Ac3net.getDirectedOrDualLinks</code> takes a network list matrix and returns a unique network; either directionals or the duals. If directed=TRUE then it is the unique pairs from column 1 to column 2. If directed=FALSE then it is the unique dual pairs (selects if and only if A&ndash;B and B&ndash;A exist). 
directional or undirectional common pairs between the two regarding the first two columns of the networks. 
</p>


<h3>Value</h3>

<p><code>Ac3net.getDirectedOrDualLinks</code> returns a unique network matrix.
</p>


<h3>Author(s)</h3>

<p>Gokmen Altay
</p>


<h3>References</h3>

<p>G. Altay,&quot;Directed Conservative Causal Core Gene Networks&quot;, bioRxiv, 2018.
G. Altay, F. Emmert-Streib, &quot;Inferring the conservative causal core of gene regulatory networks&quot;, BMC Systems Biology (2010) 4:132. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Ac3net.maxmim">Ac3net.maxmim</a></code>, <code><a href="#topic+Ac3net.cutoff">Ac3net.cutoff</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'># net1
# net &lt;-  Ac3net.getDirectedOrDualLinks(net1, dual_=FALSE)
</code></pre>

<hr>
<h2 id='Ac3net.maxmim'>
Find the maximum valued partner of each variable.
</h2><span id='topic+Ac3net.maxmim'></span>

<h3>Description</h3>

<p><code>Ac3net.maxmim</code> takes an adjacency matrix and returns the pairs with the absolute maximum correlated partner of each variable on the rows.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ac3net.maxmim(mim_, net_= TRUE, cutoff_=0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Ac3net.maxmim_+3A_mim_">mim_</code></td>
<td>
<p>An adjacency matrix, where the element at row i and column j corresponds to the correlation or mutual information between variables row i and column. Row and columns of the matrix must have the variable names.</p>
</td></tr>
<tr><td><code id="Ac3net.maxmim_+3A_net_">net_</code></td>
<td>
<p>If equal to TRUE, then it returns a sorted list of interaction of Ac3net algorithm. The significance cutoff values needs to be set in this case. In this case the resulting network directions is shown by Source and Target columns to make it look intuitive and avoid confusion in general. If equal to FALSE, then it only returns the maximum valued partner for each gene by the rows of the matrix, not sorted and no filtered by the cutoff. The gene of interest is in the second column and its maximum valued partner is in the first column, similar to above.</p>
</td></tr>
<tr><td><code id="Ac3net.maxmim_+3A_cutoff_">cutoff_</code></td>
<td>
<p>A scaler cutoff magnitude above which is considered as significant dependecny score.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Ac3net.maxmim</code> takes an adjacency matrix and returns the pairs with the absolute maximum correlated partner.
</p>


<h3>Value</h3>

<p><code>Ac3net.maxmim</code> returns a network matrix where the second column is the variable names of the row names of the input adjacency matrix and second columns have the variables names that correspond to the absolute maximum value for the variables in the second column. If net_ is TRUE, then this list of gene pairs are further filtered out based on the cutoff and sorted, which results the inferred network of Ac3net algorithm. If net_ is FALSE, the it only returns the maximum valued partner of each variable of the input matrix.
</p>


<h3>Author(s)</h3>

<p>Gokmen Altay
</p>


<h3>References</h3>

<p>G. Altay,&quot;Directed Conservative Causal Core Gene Networks&quot;, bioRxiv, 2018.
G. Altay, F. Emmert-Streib, &quot;Inferring the conservative causal core of gene regulatory networks&quot;, BMC Systems Biology (2010) 4:132. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Ac3net.commonlinks">Ac3net.commonlinks</a></code>, <code><a href="#topic+Ac3net.cutoff">Ac3net.cutoff</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'># data(expdata)
# mim &lt;- cor(t(expdata), method = 'pearson')
# mim &lt;- Ac3net.filtersames(mim)
# cutoff &lt;- Ac3net.cutoff(mim=mim, ratio_ = 0.002, PCmincutoff=0.6, PCmaxcutoff=0.96)
# Ac3net.net &lt;-  Ac3net.maxmim(mim_= mim, net_=TRUE, cutoff_= cutoff)
</code></pre>

<hr>
<h2 id='Ac3net.MTC'>
Return adjusted p-values of the adjancency matrix.
</h2><span id='topic+Ac3net.MTC'></span>

<h3>Description</h3>

<p><code>Ac3net.MTC</code> Return adjusted p-values of the adjancency matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ac3net.MTC(data, iterations=10, MTC=TRUE, MTCmethod="BH",
estmethod='pearson')
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Ac3net.MTC_+3A_data">data</code></td>
<td>
<p>Data matrix where rows are variables and columns are the samples. Rows must have the variable names</p>
</td></tr>
<tr><td><code id="Ac3net.MTC_+3A_iterations">iterations</code></td>
<td>
<p>Number of iterations to get null distribuiton from multiple shuffling of the data.</p>
</td></tr>  
<tr><td><code id="Ac3net.MTC_+3A_mtc">MTC</code></td>
<td>
<p>If TRUE,as the default status, the the returned p-value matrix has the adjusted p-values</p>
</td></tr>
<tr><td><code id="Ac3net.MTC_+3A_mtcmethod">MTCmethod</code></td>
<td>
<p>The Multiple Testing Correction method. BH is default. See R p.adjust function for options.</p>
</td></tr>
<tr><td><code id="Ac3net.MTC_+3A_estmethod">estmethod</code></td>
<td>
<p>The default method is 'pearson' assuming that the data was normalized (at least with Log-2). If the data is not normalized, its recomended setting is 'spearman'.</p>
</td></tr>  
</table>


<h3>Details</h3>

<p><code>Ac3net.MTC</code> takes a data matrix and returns the adjusted p-values matrix that correspond to the adjacency matrix of the input data. This way varios p-values can be tried to get different cutoff values without re-calculating the p-values.
</p>


<h3>Value</h3>

<p><code>Ac3net.MTC</code> returns a default adjusted p-values matrix that corresponds to the input adjacency matrix. If MTC is set to FALSE then it returns raw p-values.  
</p>


<h3>Author(s)</h3>

<p>Gokmen Altay
</p>


<h3>References</h3>

<p>G. Altay,&quot;Directed Conservative Causal Core Gene Networks&quot;, bioRxiv, 2018.
G. Altay, F. Emmert-Streib, &quot;Inferring the conservative causal core of gene regulatory networks&quot;, BMC Systems Biology (2010) 4:132. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Ac3net.maxmim">Ac3net.maxmim</a></code>, <code><a href="#topic+Ac3net.commonlinks">Ac3net.commonlinks</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#data(expdata)
#mim.pvals &lt;- Ac3net.MTC(data = expdata, iterations=10,
#            MTC=TRUE, MTCmethod="BH", estmethod='pearson')
</code></pre>

<hr>
<h2 id='Ac3net.performance'>
Find the performance of the first network based on the second.
</h2><span id='topic+Ac3net.performance'></span>

<h3>Description</h3>

<p><code>Ac3net.performance</code> takes two network list matrices. First is the predicted network and second is the reference true network. Returns performance scores.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ac3net.performance(predictNet, referenceNet, data_, directed=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Ac3net.performance_+3A_predictnet">predictNet</code></td>
<td>
<p>The predicted network. A matrix that represents the links between two variables at each row. Namely, the element at row i and column 1 and the element at row i and column 2 are assumed to be paired or linked. There is no connection assumed among rows.</p>
</td></tr>
<tr><td><code id="Ac3net.performance_+3A_referencenet">referenceNet</code></td>
<td>
<p>The reference network over which the performance will ve evaluated. A matrix that represents the links between two variables at each row. Namely, the element at row i and column 1 and the element at row i and column 2 are assumed to be paired or linked. There is no connection assumed among rows.</p>
</td></tr>
<tr><td><code id="Ac3net.performance_+3A_data_">data_</code></td>
<td>
<p>The data matrix where the variables are available as row names of it. Only row names will be used to get all the variable names for the analaysis.</p>
</td></tr>
<tr><td><code id="Ac3net.performance_+3A_directed">directed</code></td>
<td>
<p> It is TRUE by default, which means there is  directionality assumed from column 1 to column 2 at each matrices. Therefore common links are the directionale common links. If it is FALSE, then the pairs assumed to be undirectional and all the common pairs are taken as common. E.g. If there is A&ndash;B in net1 and B&ndash;A in net2 then they are considered as common.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Ac3net.performance</code> takes two network list matrices and based on the first two columns it makes comparison betwee the predicted network and the reference network. If the argument directed is equal to TRUE, as in the default case, then it assumes directionally from column 1 to column 2 of the matrices and computes the performance based on it. E.g. For a pair of A&ndash;&gt;B in the predicted network, if there is also A&ndash;&gt;B in the reference network then it is taken as True Positive (TP) but the opposite direction B&ndash;&gt;A is not accepted as TP. If the argument directed is equal to FALSE, then eitherA&ndash;&gt;B or B&ndash;&gt;A is acepted as TP, with no directional considerations. 
</p>


<h3>Value</h3>

<p><code>Ac3net.performance</code>  returns an object with all the performance scores. 
</p>


<h3>Author(s)</h3>

<p>Gokmen Altay
</p>


<h3>References</h3>

<p>G. Altay,&quot;Directed Conservative Causal Core Gene Networks&quot;, bioRxiv, 2018.
G. Altay, F. Emmert-Streib, &quot;Inferring the conservative causal core of gene regulatory networks&quot;, BMC Systems Biology (2010) 4:132. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Ac3net.maxmim">Ac3net.maxmim</a></code>, <code><a href="#topic+Ac3net.cutoff">Ac3net.cutoff</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'> # data(expdata)
 # data(truenet)
 # data(zeroindx) # zero indices for the example dataset, not in general.
 ## step by step to get Ac3net network
 # mim &lt;- cor(t(expdata), method = "pearson")
 # mim[zeroindx, zeroindx] &lt;- 0 #for the example dataset, not in general.
 # mim &lt;- Ac3net.filtersames(mim=mim) #in case if there are same variables.
 #diag(mim) &lt;-0
 #cutoff &lt;- Ac3net.cutoff(mim, ratio_ = 0.002, PCmincutoff=0.6, PCmaxcutoff=0.96)
 #mim[abs(mim) &lt; cutoff] &lt;- 0
 #Ac3net.net &lt;- Ac3net.maxmim(mim=mim, net_=TRUE, cutoff_= cutoff)
 ### one command could do the same as below if zeroindx was not available. ###
 ## ac3net.net &lt;- Ac3net(DataOrMim=expdata, processed=FALSE, ratio_ = 0.002, PCmincutoff=0.6,
 #          PCmaxcutoff=0.96, cutoff=0, estmethod='pearson', pval=1, 
 #          iterations=10, MTC=FALSE, MTCmethod="BH" )
 
# performance &lt;- Ac3net.performance(predictNet=ac3net.net, 
#                referenceNet = truenet, data_, directed=TRUE)
# performance
</code></pre>

<hr>
<h2 id='expdata'>
Real E.coli Gene Expression Dataset
</h2><span id='topic+expdata'></span>

<h3>Description</h3>

<p>Normalized real microarray gene expression E.coli data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("expdata")</code></pre>


<h3>Details</h3>

<p>This dataset compiled in (Faith, 2007) . The example data set is also given to provide a sample data set so that one can take it as reference while formatting the new data set as entry to  As in this example datset, the input data sets must be complete and neither empty or non-numerical characters are allowed. 
</p>


<h3>Source</h3>

<p>This data was downloaded from m3d.mssm.edu/network_inference.html
</p>


<h3>References</h3>

<p>Faith JJ, Hayete B, Thaden JT, Mogno I, Wierzbowski J, Cottarel G, Kasif S, Collins JJ, Gardner TS. Large-scale mapping and validation of Escherichia coli transcriptional regulation from a compendium of expression profiles. PLoS Biol. 2007 Jan;5(1):e8
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(expdata) #returns data matrix with the name expdata.

</code></pre>

<hr>
<h2 id='truenet'>
The known intreactions of E.coli from RegulonDB database.
</h2><span id='topic+truenet'></span>

<h3>Description</h3>

<p>This true interactions of E.coli from RegulonDB database. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("truenet")</code></pre>


<h3>Details</h3>

<p>These true interactions are mainly from  RegulonDB database.
We compiled the one from (Faith, 2007), and the RegulonDB database (TF-TF and TF-gene).We removed the links where the gene names is not evailable in our example   E.coli dataset. See the references for the usage information of the data.
</p>


<h3>Source</h3>

<p>This data was downloaded from http://regulondb.ccg.unam.mx/menu/download/datasets/index.jsp and  m3d.mssm.edu/network_inference.html .
</p>


<h3>References</h3>

<p>Gama-Castro S, et.al., RegulonDB version 9.0: high-level integration of gene regulation, coexpression, motif clustering and beyond., Nucleic Acids Res. 2016 Jan 4;44(D1):D133-43.
</p>
<p>Faith JJ, Hayete B, Thaden JT, Mogno I, Wierzbowski J, Cottarel G, Kasif S, Collins JJ, Gardner TS. Large-scale mapping and validation of Escherichia coli transcriptional regulation from a compendium of expression profiles. PLoS Biol. 2007 Jan;5(1):e8
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(truenet)

</code></pre>

<hr>
<h2 id='zeroindx'>
The indices to force to zero correlations.
</h2><span id='topic+zeroindx'></span>

<h3>Description</h3>

<p>The indices to force to zero correlations of E.coli expresseion data. Because REgulonDB database provided only TF-TF or TF-gene interactions. We need to remove gene-gene interactions before the performance analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("zeroindx")</code></pre>


<h3>Details</h3>

<p>The indices to force to zero correlations of E.coli expresseion data. Because REgulonDB database provided only TF-TF or TF-gene interactions. We need to remove gene-gene interactions before the performance analysis.
These indice are also from (Faith, 2007),  available along with the expression data.
</p>


<h3>Source</h3>

<p>This data was downloaded from m3d.mssm.edu/network_inference.html .
</p>


<h3>References</h3>

<p>Faith JJ, Hayete B, Thaden JT, Mogno I, Wierzbowski J, Cottarel G, Kasif S, Collins JJ, Gardner TS. Large-scale mapping and validation of Escherichia coli transcriptional regulation from a compendium of expression profiles. PLoS Biol. 2007 Jan;5(1):e8
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#data(zeroindx)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
