<!DOCTYPE html><html><head><title>Help for package ace2fastq</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ace2fastq}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ace2fastq'><p>ace2fastq</p></a></li>
<li><a href='#ace_to_fastq'><p>ace_to_fastq</p></a></li>
<li><a href='#print.ace2fastq'><p>print.ace2fastq</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>ACE File to FASTQ Converter</td>
</tr>
<tr>
<td>Version:</td>
<td>0.6.0</td>
</tr>
<tr>
<td>Description:</td>
<td>The ACE file format is used in genomics to store contigs from sequencing machines.
  This tools converts it into FASTQ format. Both formats contain the 
  sequence characters and their
  corresponding quality information. Unlike the FASTQ file, the ace file stores the
  quality values numerically.
  The conversion algorithm uses the standard Sanger formula. The package facilitates insertion
  into pipelines, and content inspection.</td>
</tr>
<tr>
<td>Imports:</td>
<td>stringr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown, covr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/c5sire/ace2fastq">https://github.com/c5sire/ace2fastq</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/c5sire/ace2fastq/issues">https://github.com/c5sire/ace2fastq/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-06-19 23:38:38 UTC; Reinhard</td>
</tr>
<tr>
<td>Author:</td>
<td>Reinhard Simon <a href="https://orcid.org/0000-0002-4608-9077"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Reinhard Simon &lt;rsimon64@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-06-20 14:50:18 UTC</td>
</tr>
</table>
<hr>
<h2 id='ace2fastq'>ace2fastq</h2><span id='topic+ace2fastq'></span><span id='topic+ace2fastq-package'></span>

<h3>Description</h3>

<p>The package provides a function that converts &quot;.ace&quot; files (ABI Sanger capillary sequence assembly files) to standard &quot;.fastq&quot; files. The file format is currently used in genomics to store contigs. To the best of our knowledge, no R function is available to convert this format into the more popular fastq file format. The development was motivated in the context of the analysis of 16S metagenomic data by the need to convert the .ace files for further analysis.
</p>


<h3>ace2fastq functionality</h3>

<p>See: <code><a href="#topic+ace_to_fastq">ace_to_fastq</a></code>
</p>

<hr>
<h2 id='ace_to_fastq'>ace_to_fastq</h2><span id='topic+ace_to_fastq'></span>

<h3>Description</h3>

<p>Converts one or more contig sequences in .ace file format to .fastq format.
The parameter target_dir has a special value 'stdout' which will just return
the contigs as a list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ace_to_fastq(filename, target_dir = dirname(filename), name2id = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ace_to_fastq_+3A_filename">filename</code></td>
<td>
<p>.ace file</p>
</td></tr>
<tr><td><code id="ace_to_fastq_+3A_target_dir">target_dir</code></td>
<td>
<p>target directory or stdout</p>
</td></tr>
<tr><td><code id="ace_to_fastq_+3A_name2id">name2id</code></td>
<td>
<p>use the file name as primary id or not. Default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>


<h3>Author(s)</h3>

<p>Reinhard Simon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  library(ace2fastq)
  filename &lt;- system.file("sampledat/1.seq.ace", package = "ace2fastq")
 
  fileout &lt;- ace_to_fastq(filename, target_dir = tempdir())

</code></pre>

<hr>
<h2 id='print.ace2fastq'>print.ace2fastq</h2><span id='topic+print.ace2fastq'></span>

<h3>Description</h3>

<p>print method for an object of class ace2fastq
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ace2fastq'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.ace2fastq_+3A_x">x</code></td>
<td>
<p>an ace2fastq object</p>
</td></tr>
<tr><td><code id="print.ace2fastq_+3A_...">...</code></td>
<td>
<p>other print parameters</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Reinhard Simon
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
