<!DOCTYPE html><html><head><title>Help for package acnr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {acnr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#acnr-package'><p>Annotated Copy-Number Regions</p></a></li>
<li><a href='#getMinorMajorCopyNumbers'><p>Get minor and major copy number labels from region annotation labels</p></a></li>
<li><a href='#GSE11976_CRL2324'><p>Annotated copy-number regions from the GEO GSE11976 data set.</p></a></li>
<li><a href='#GSE13372_HCC1143'><p>Annotated copy-number regions from the GEO GSE13372 data set.</p></a></li>
<li><a href='#GSE29172_H1395'><p>Annotated copy-number regions from the GEO GSE29172 (and GSE26302) data sets.</p></a></li>
<li><a href='#listDataSets'><p>List available data sets</p></a></li>
<li><a href='#listTumorFractions'><p>List of available tumor fractions for a data set</p></a></li>
<li><a href='#loadCnRegionData'><p>loadCnRegionData</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Annotated Copy-Number Regions</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-04-15</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides SNP array data from different types of
    copy-number regions. These regions were identified manually by the authors
    of the package and may be used to generate realistic data sets with known
    truth.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-2.1">LGPL-2.1</a> | <a href="https://www.r-project.org/Licenses/LGPL-3">LGPL-3</a> [expanded from: LGPL (&ge; 2.1)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10),</td>
</tr>
<tr>
<td>Suggests:</td>
<td>R.utils, knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/mpierrejean/acnr">https://github.com/mpierrejean/acnr</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/mpierrejean/acnr/issues">https://github.com/mpierrejean/acnr/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-04-18 08:34:55 UTC; mpierre-jean</td>
</tr>
<tr>
<td>Author:</td>
<td>Morgane Pierre-Jean [aut, cre],
  Pierre Neuvial [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Morgane Pierre-Jean &lt;morgane.pierrejean@genopole.cnrs.fr&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-04-18 09:58:15 UTC</td>
</tr>
</table>
<hr>
<h2 id='acnr-package'>Annotated Copy-Number Regions</h2><span id='topic+acnr-package'></span><span id='topic+acnr'></span>

<h3>Description</h3>

<p>This data package contains SNP array data from different types of copy-number regions.  These regions were identified manually by the authors of the package and may be used to generate realistic data sets with known truth.</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">Package: </td><td style="text-align: left;"> acnr</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Title: </td><td style="text-align: left;"> Annotated Copy-Number Regions</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 0.2.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2014-09-08</td>
</tr>
<tr>
 <td style="text-align: left;">
Author: </td><td style="text-align: left;"> Morgane Pierre-Jean and Pierre Neuvial</td>
</tr>
<tr>
 <td style="text-align: left;">
Maintainer: </td><td style="text-align: left;"> Morgane Pierre-Jean &lt;morgane.pierrejean@genopole.cnrs.fr&gt;</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> LGPL (&gt;= 2.1)</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> R (&gt;= 2.10), R.utils</td>
</tr>
<tr>
 <td style="text-align: left;">
Suggests: </td><td style="text-align: left;"> RUnit, BiocGenerics</td>
</tr>
<tr>
 <td style="text-align: left;">
biocViews: </td><td style="text-align: left;"> ExperimentData</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Morgane Pierre-Jean and Pierre Neuvial</p>

<hr>
<h2 id='getMinorMajorCopyNumbers'>Get minor and major copy number labels from region annotation labels</h2><span id='topic+getMinorMajorCopyNumbers'></span>

<h3>Description</h3>

<p>Get minor and major copy number labels from region annotation labels
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMinorMajorCopyNumbers(region)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getMinorMajorCopyNumbers_+3A_region">region</code></td>
<td>
<p>A character value, the annotation label for a copy number
region.  Should be encoded as <code>"(C1,C2)"</code>,  where </p>
 
<dl>
<dt><code>C1</code></dt><dd><p>denotes the minor copy number, that is, the smallest of
the two parent-specific copy numbers</p>
</dd> <dt><code>C2</code></dt><dd><p>denotes the minor
copy number, that is, the smallest of the two parent-specific copy
numbers</p>
</dd></dl>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>matrix</code> with <code>length(region)</code> rows and two columns:
<code>C1</code> and <code>C2</code>, as described above.
</p>


<h3>References</h3>

<p>Bengtsson H., Neuvial, P. and Speed, T. P. (2010) TumorBoost: 
normalization of allele-specific tumor copy numbers from a single pair of 
tumor-normal genotyping microarrays. BMC bioinformatics 11 (2010), p. 245.
</p>
<p>Neuvial, P., Bengtsson H., and Speed, T. P. (2011) Statistical
analysis of Single Nucleotide Polymorphism microarrays in cancer studies.
Chapter 11 in *Handbook of Statistical Bioinformatics*,  Springer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dat &lt;- loadCnRegionData(dataSet="GSE29172_H1395", tumorFraction=1)
regions &lt;- unique(dat$region)
getMinorMajorCopyNumbers(regions)

</code></pre>

<hr>
<h2 id='GSE11976_CRL2324'>Annotated copy-number regions from the GEO GSE11976 data set.</h2><span id='topic+GSE11976_CRL2324'></span>

<h3>Description</h3>

<p>The GEO GSE11976 data set is a dilution series from the Illumina 
HumanCNV370v1 chip type (Staaf et al, 2008).
</p>


<h3>Format</h3>

<p>A data frame with 770668 observations of 7 variables: 
</p>
 
<dl>
<dt>c</dt><dd><p>total copy number (not log-scaled)</p>
</dd> 
<dt>b</dt><dd><p>allelic ratios in the
diluted tumor sample (after TumorBoost)</p>
</dd> 
<dt>genotype</dt><dd><p>germline
genotypes</p>
</dd> 
<dt>region</dt><dd><p>a character value, annotation label for the region. Should be
encoded as <code>"(C1,C2)"</code>, where <code>C1</code> denotes the minor copy number
and <code>C2</code> denotes the major copy number.  For example, 
</p>
 
<dl>
<dt>(1,1)</dt><dd><p>Normal</p>
</dd> <dt>(0,1)</dt><dd><p>Hemizygous deletion</p>
</dd> 
<dt>(0,0)</dt><dd><p>Homozygous deletion</p>
</dd> <dt>(1,2)</dt><dd><p>Single copy gain</p>
</dd> 
<dt>(0,2)</dt><dd><p>Copy-neutral LOH</p>
</dd> <dt>(2,2)</dt><dd><p>Balanced two-copy gain</p>
</dd> 
<dt>(1,3)</dt><dd><p>Unbalanced two-copy gain</p>
</dd> <dt>(0,3)</dt><dd><p>Single-copy gain with
LOH</p>
</dd> </dl>
 </dd>
<dt>cellularity</dt><dd><p>A numeric value between 0 and 1, the percentage of tumor cells in the  sample.</p>
</dd>
</dl>

<p>@source http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE11976
@references Staaf, J., Lindgren, D., Vallon-Christersson, J., Isaksson, A., 
Goransson, H., Juliusson, G., ... &amp; Ringn\'er, M. (2008). 
Segmentation-based detection of allelic imbalance and 
loss-of-heterozygosity in cancer cells using whole genome SNP arrays. 
Genome Biol, 9(9), R136.</p>


<h3>Details</h3>

<p>These data have been processed from the files available at 
http://cbbp.thep.lu.se/~markus/software/BAFsegmentation/ using scripts that 
are included in the 'inst/preprocessing/GSE11976' directory of this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat &lt;- loadCnRegionData("GSE11976_CRL2324")
unique(dat$region)
</code></pre>

<hr>
<h2 id='GSE13372_HCC1143'>Annotated copy-number regions from the GEO GSE13372 data set.</h2><span id='topic+GSE13372_HCC1143'></span>

<h3>Description</h3>

<p>The GEO GSE13372 data set is from the Affymetrix GenomeWideSNP_6 chip type. 
We have extracted one tumor/normal pair corresponding to the breast cancer 
cell line HCC1143. For consistency with the other data sets in the package 
the tumor and normal samples are labeled according to their tumor 
cellularity, that is, 100
</p>


<h3>Format</h3>

<p>A data frame with 205842 observations of 7 variables: </p>
 
<dl>
<dt>c</dt><dd><p>total copy number (not log-scaled)</p>
</dd> <dt>b</dt><dd><p>allelic ratios in the
diluted tumor sample (after TumorBoost)</p>
</dd> <dt>genotype</dt><dd><p>germline 
genotypes</p>
</dd> <dt>bT</dt><dd><p>allelic ratios in the diluted tumor sample (before 
TumorBoost)</p>
</dd> <dt>bN</dt><dd><p>allelic ratios in the matched normal sample</p>
</dd> 
<dt>region</dt><dd><p>a character value, annotation label for the region. Should be
encoded as <code>"(C1,C2)"</code>, where <code>C1</code> denotes the minor copy number 
and <code>C2</code> denotes the major copy number.  For example, </p>
 
<dl>
<dt>(1,1)</dt><dd><p>Normal</p>
</dd> <dt>(0,1)</dt><dd><p>Hemizygous deletion</p>
</dd> 
<dt>(0,0)</dt><dd><p>Homozygous deletion</p>
</dd> <dt>(1,2)</dt><dd><p>Single copy gain</p>
</dd> 
<dt>(0,2)</dt><dd><p>Copy-neutral LOH</p>
</dd> <dt>(2,2)</dt><dd><p>Balanced two-copy gain</p>
</dd> 
<dt>(1,3)</dt><dd><p>Unbalanced two-copy gain</p>
</dd> <dt>(0,3)</dt><dd><p>Single-copy gain with 
LOH</p>
</dd> </dl>
 </dd> <dt>genotype</dt><dd><p>the (germline) genotype of SNPs. By definition, 
rows with missing genotypes are interpreted as non-polymorphic loci (a.k.a.
copy number probes).</p>
</dd> <dt>cellularity</dt><dd><p>A numeric value between 0 and 1, 
the percentage of tumor cells in the sample.</p>
</dd> </dl>


<h3>Details</h3>

<p>These data have been processed from the files available from GEO using 
scripts that are included in the 'inst/preprocessing/GSE13372' directory of 
this package. This processing includes normalization of the raw CEL files 
using the CRMAv2 method implemented in the aroma.affymetrix package.
</p>


<h3>Source</h3>

<p><a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE13372">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE13372</a> 
<a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE13372">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE13372</a>
</p>


<h3>References</h3>

<p>Chiang DY, Getz G, Jaffe DB, O'Kelly MJ et al. High-resolution
mapping of copy-number alterations with massively parallel sequencing. Nat
Methods 2009 Jan;6(1):99-103. PMID: 19043412
</p>
<p>Bengtsson, H., Wirapati , P. &amp; Speed, T.P. (2009). A single-array
preprocessing method for estimating full-resolution raw copy numbers from 
all Affymetrix genotyping arrays including GenomeWideSNP 5 &amp; 6, 
Bioinformatics 25(17), pp. 2149-56.
</p>
<p>Bengtsson H., Neuvial, P. and Speed, T. P. (2010) TumorBoost: 
normalization of allele-specific tumor copy numbers from a single pair of 
tumor-normal genotyping microarrays. BMC bioinformatics 11 (2010), p. 245.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat &lt;- loadCnRegionData("GSE13372_HCC1143")
unique(dat$region)
</code></pre>

<hr>
<h2 id='GSE29172_H1395'>Annotated copy-number regions from the GEO GSE29172 (and GSE26302) data sets.</h2><span id='topic+GSE29172_H1395'></span>

<h3>Description</h3>

<p>The GEO GSE29172 data set is a dilution series from the Affymetrix 
GenomeWideSNP_6 chip type. The GEO GSE26302 data set contains the experiment 
corresponding to the matched normal (i.e. 0% dilution).
</p>


<h3>Format</h3>

<p>A data frame with 770668 observations of 7 variables: </p>
 
<dl>
<dt>c</dt><dd><p>total copy number (not log-scaled)</p>
</dd> <dt>b</dt><dd><p>allelic ratios in the
diluted tumor sample (after TumorBoost)</p>
</dd> <dt>genotype</dt><dd><p>germline 
genotypes</p>
</dd> <dt>bT</dt><dd><p>allelic ratios in the diluted tumor sample (before 
TumorBoost)</p>
</dd> <dt>bN</dt><dd><p>allelic ratios in the matched normal sample</p>
</dd> 
<dt>region</dt><dd><p>a character value, annotation label for the region. Should be
encoded as <code>"(C1,C2)"</code>, where <code>C1</code> denotes the minor copy number 
and <code>C2</code> denotes the major copy number.  For example, </p>
 
<dl>
<dt>(1,1)</dt><dd><p>Normal</p>
</dd> <dt>(0,1)</dt><dd><p>Hemizygous deletion</p>
</dd> 
<dt>(0,0)</dt><dd><p>Homozygous deletion</p>
</dd> <dt>(1,2)</dt><dd><p>Single copy gain</p>
</dd> 
<dt>(0,2)</dt><dd><p>Copy-neutral LOH</p>
</dd> <dt>(2,2)</dt><dd><p>Balanced two-copy gain</p>
</dd> 
<dt>(1,3)</dt><dd><p>Unbalanced two-copy gain</p>
</dd> <dt>(0,3)</dt><dd><p>Single-copy gain with 
LOH</p>
</dd> </dl>
 </dd> <dt>genotype</dt><dd><p>the (germline) genotype of SNPs. By definition, rows 
with missing genotypes are interpreted as non-polymorphic loci (a.k.a. copy
number probes).</p>
</dd> <dt>cellularity</dt><dd><p>A numeric value between 0 and 1, the
percentage of tumor cells in the  sample.</p>
</dd> </dl>


<h3>Details</h3>

<p>These data have been processed from the files available from GEO using 
scripts that are included in the 'inst/preprocessing/GSE29172' directory of
this package. This processing includes normalization of the raw CEL files
using the CRMAv2 method implemented in the aroma.affymetrix package.
</p>


<h3>Source</h3>

<p><a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE29172">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE29172</a> 
<a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE26302">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE26302</a>
</p>


<h3>References</h3>

<p>Rasmussen, M., Sundstr\&quot;om, M., Kultima, H. G., Botling, J., 
Micke, P., Birgisson, H., Glimelius, B. &amp; Isaksson, A. (2011). 
Allele-specific copy number analysis of tumor samples with aneuploidy and 
tumor heterogeneity. Genome Biology, 12(10), R108.#'
</p>
<p>Bengtsson, H., Wirapati , P. &amp; Speed, T.P. (2009). A single-array
preprocessing method for estimating full-resolution raw copy numbers from 
all Affymetrix genotyping arrays including GenomeWideSNP 5 &amp; 6, 
Bioinformatics 25(17), pp. 2149-56.
</p>
<p>Bengtsson H., Neuvial, P. and Speed, T. P. (2010) TumorBoost:
normalization of allele-specific tumor copy numbers from a single pair of
tumor-normal genotyping microarrays. BMC bioinformatics 11 (2010), p. 245.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat &lt;- loadCnRegionData("GSE29172_H1395")
unique(dat$region)
</code></pre>

<hr>
<h2 id='listDataSets'>List available data sets</h2><span id='topic+listDataSets'></span>

<h3>Description</h3>

<p>List available data sets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>listDataSets()
</code></pre>


<h3>Value</h3>

<p>name of one of the data sets of the package, see
<a href="#topic+listDataSets">listDataSets</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>listDataSets()

</code></pre>

<hr>
<h2 id='listTumorFractions'>List of available tumor fractions for a data set</h2><span id='topic+listTumorFractions'></span>

<h3>Description</h3>

<p>List of available tumor fractions for a data set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>listTumorFractions(dataSet)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="listTumorFractions_+3A_dataset">dataSet</code></td>
<td>
<p>The name of a data set from the package, see <a href="#topic+listDataSets">listDataSets</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector, the available tumor fractions for a data set
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dataSets &lt;- listDataSets()
fracs &lt;- listTumorFractions(dataSets[1])

</code></pre>

<hr>
<h2 id='loadCnRegionData'>loadCnRegionData</h2><span id='topic+loadCnRegionData'></span>

<h3>Description</h3>

<p>Load real, annotated copy number data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadCnRegionData(dataSet, tumorFraction = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="loadCnRegionData_+3A_dataset">dataSet</code></td>
<td>
<p>name of one of the data sets of the package, see
<code><a href="#topic+listDataSets">listDataSets</a></code></p>
</td></tr>
<tr><td><code id="loadCnRegionData_+3A_tumorfraction">tumorFraction</code></td>
<td>
<p>proportion of tumor cells in the &quot;tumor&quot; sample (a.k.a.
tumor cellularity). See <code><a href="#topic+listTumorFractions">listTumorFractions</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a wrapper to load real genotyping array data taken from
</p>
<p>* a dilution series from the Affymetrix GenomeWideSNP_6 chip type (Rasmussen
et al, 2011), see <code><a href="#topic+GSE29172_H1395">GSE29172_H1395</a></code> * a dilution series from the
Illumina HumanCNV370v1 chip type (Staaf et al, 2008), see
<code><a href="#topic+GSE11976_CRL2324">GSE11976_CRL2324</a></code> * a tumor/normal pair from the Affymetrix
GenomeWideSNP_6 chip type (Chiang et al, 2008), see
<code><a href="#topic+GSE13372_HCC1143">GSE13372_HCC1143</a></code>
</p>


<h3>Value</h3>

<p>a data.frame containing copy number data for different types of copy
number regions.  Columns:</p>
 <dl>
<dt>c</dt><dd><p>Total copy number</p>
</dd>
<dt>b</dt><dd><p>Allele B fraction (a.k.a. BAF)</p>
</dd> <dt>region</dt><dd><p>a character value,
annotation label for the region.  Should be encoded as <code>"(C1,C2)"</code>,
where <code>C1</code> denotes the minor copy number and <code>C2</code> denotes the
major copy number.  For example, </p>
 <dl>
<dt>(1,1)</dt><dd><p>Normal</p>
</dd>
<dt>(0,1)</dt><dd><p>Hemizygous deletion</p>
</dd> <dt>(0,0)</dt><dd><p>Homozygous deletion</p>
</dd>
<dt>(1,2)</dt><dd><p>Single copy gain</p>
</dd> <dt>(0,2)</dt><dd><p>Copy-neutral LOH</p>
</dd>
<dt>(2,2)</dt><dd><p>Balanced two-copy gain</p>
</dd> <dt>(1,3)</dt><dd><p>Unbalanced two-copy gain</p>
</dd>
<dt>(0,3)</dt><dd><p>Single-copy gain with LOH</p>
</dd> </dl>
</dd> <dt>muN</dt><dd><p>the (germline)
genotype of SNPs. By definition, rows with missing genotypes are
interpreted as non-polymorphic loci (a.k.a. copy number probes).</p>
</dd></dl>



<h3>Author(s)</h3>

<p>Morgane Pierre-Jean and Pierre Neuvial
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
affyDat &lt;- loadCnRegionData(dataSet="GSE29172_H1395", tumorFraction=1)
str(affyDat)

illuDat &lt;- loadCnRegionData(dataSet="GSE11976_CRL2324", tumorFraction=.79)
str(illuDat)

affyDat2 &lt;- loadCnRegionData(dataSet="GSE13372_HCC1143", tumorFraction=1)
str(affyDat2)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
