<!DOCTYPE html><html><head><title>Help for package ActiveDriver</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ActiveDriver}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ActiveDriver'><p>Identification of active protein sites (post-translational modification sites, signalling domains, etc) with</p>
specific and significant mutations.</a></li>
<li><a href='#kinase_domains'><p>Example kinase domains for ActiveDriver</p></a></li>
<li><a href='#mutations'><p>Example mutations for ActiveDriver</p></a></li>
<li><a href='#phosphosites'><p>Example phosphosites for ActiveDriver</p></a></li>
<li><a href='#read_fasta'><p>Read FASTA file as character vector.</p></a></li>
<li><a href='#sequence_disorder'><p>Example protein disorder for ActiveDriver</p></a></li>
<li><a href='#sequences'><p>Example protein sequences for ActiveDriver</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Description:</td>
<td>A mutation analysis tool that discovers cancer driver genes with frequent mutations in protein signalling sites such as post-translational modifications (phosphorylation, ubiquitination, etc). The Poisson generalised linear regression model identifies genes where cancer mutations in signalling sites are more frequent than expected from the sequence of the entire gene. Integration of mutations with signalling information helps find new driver genes and propose candidate mechanisms to known drivers. Reference: Systematic analysis of somatic mutations in phosphorylation signaling predicts novel cancer drivers. Juri Reimand and Gary D Bader. Molecular Systems Biology (2013) 9:637 &lt;<a href="https://doi.org/10.1038%2Fmsb.2012.68">doi:10.1038/msb.2012.68</a>&gt;.</td>
</tr>
<tr>
<td>Title:</td>
<td>Finding Cancer Driver Proteins with Enriched Mutations in
Post-Translational Modification Sites</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, parallel, MASS</td>
</tr>
<tr>
<td>Collate:</td>
<td>'ActiveDriver.R'</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1.9000</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-08-23 18:26:21 UTC; dthompson</td>
</tr>
<tr>
<td>Author:</td>
<td>Juri Reimand [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Juri Reimand &lt;juri.reimand@utoronto.ca&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-08-23 20:55:51 UTC</td>
</tr>
</table>
<hr>
<h2 id='ActiveDriver'>Identification of active protein sites (post-translational modification sites, signalling domains, etc) with
specific and significant mutations.</h2><span id='topic+ActiveDriver'></span>

<h3>Description</h3>

<p>Identification of active protein sites (post-translational modification sites, signalling domains, etc) with
specific and significant mutations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ActiveDriver(sequences, seq_disorder, mutations, active_sites, flank = 7,
  mid_flank = 2, mc.cores = 1, simplified = FALSE,
  return_records = FALSE, skip_mismatch = TRUE,
  regression_type = "poisson", enriched_only = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ActiveDriver_+3A_sequences">sequences</code></td>
<td>
<p>character vector of protein sequences, names are protein IDs.</p>
</td></tr>
<tr><td><code id="ActiveDriver_+3A_seq_disorder">seq_disorder</code></td>
<td>
<p>character vector of disorder in protein sequences, names are protein IDs and values are strings
1/0 for disordered/ordered protein residues.</p>
</td></tr>
<tr><td><code id="ActiveDriver_+3A_mutations">mutations</code></td>
<td>
<p>data frame of mutations, with [gene, sample_id, position, wt_residue, mut_residue] as columns.</p>
</td></tr>
<tr><td><code id="ActiveDriver_+3A_active_sites">active_sites</code></td>
<td>
<p>data frame of active sites, with [gene, position, residue, kinase] as columns. Kinase field may
be blank and is shown for informative purposes.</p>
</td></tr>
<tr><td><code id="ActiveDriver_+3A_flank">flank</code></td>
<td>
<p>numeric for selecting region size around active sites considered important for site activity. Default
value is 7. Ignored in case of simplified analysis.</p>
</td></tr>
<tr><td><code id="ActiveDriver_+3A_mid_flank">mid_flank</code></td>
<td>
<p>numeric for splitting flanking region size into proximal (&lt;=X) and distal (&gt;X). Default value is
2. Ignored in case of simplified analysis.</p>
</td></tr>
<tr><td><code id="ActiveDriver_+3A_mc.cores">mc.cores</code></td>
<td>
<p>numeric for indicating number of computing cores dedicated to computation. Default value is 1.</p>
</td></tr>
<tr><td><code id="ActiveDriver_+3A_simplified">simplified</code></td>
<td>
<p>true/false for selecting simplified analysis. Default value is FALSE. If TRUE, no flanking regions
are considered and only indicated sites are tested for mutations.</p>
</td></tr>
<tr><td><code id="ActiveDriver_+3A_return_records">return_records</code></td>
<td>
<p>true/false for returning a collection of gene records with more data regarding sites and
mutations. Default value is FALSE.</p>
</td></tr>
<tr><td><code id="ActiveDriver_+3A_skip_mismatch">skip_mismatch</code></td>
<td>
<p>true/false for skipping mutations whose reference protein residue does not match expected
residue from FASTA sequence file.</p>
</td></tr>
<tr><td><code id="ActiveDriver_+3A_regression_type">regression_type</code></td>
<td>
<p>'nb' for negative binomial, 'poisson' for poisson GLM. The latter is default.</p>
</td></tr>
<tr><td><code id="ActiveDriver_+3A_enriched_only">enriched_only</code></td>
<td>
<p>true/false to indicate whether only sites with enriched active site mutations will be included
in the final p-value estimation (TRUE is default). If FALSE, sites with less than expected mutations will be also
included.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with the following components: 
@return all_active_mutations - table with mutations that hit or flank an active site. Additional columns of
interest include Status (DI - direct active mutation; N1 - proximal flanking mutation; N2 - distal flanking
mutation) and Active_region (region ID of active sites in that protein).
</p>
<p>all_active_sites -
</p>
<p>all_region_based_pval - p-values for regions of sites, statistics on observed mutations (obs) and expected
mutations (exp, low, high based on mean and s.d. from Poisson sampling). The field Region identifies region in
all_active_sites.
</p>


<h3>Author(s)</h3>

<p>Juri Reimand &lt;juri.reimand@utoronto.ca&gt;
</p>


<h3>References</h3>

<p>Systematic analysis of somatic mutations in phosphorylation signaling predicts novel cancer drivers
(2013, Molecular Systems Biology) by Juri Reimand and Gary Bader.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ActiveDriver_data)

phos_results = ActiveDriver(sequences, sequence_disorder, mutations, phosphosites)
ovarian_mutations = mutations[grep("ovarian", mutations$sample_id),]
phos_results_ovarian = ActiveDriver(sequences, sequence_disorder, ovarian_mutations, phosphosites)
GBM_muts = mutations[grep("glioblastoma", mutations$sample_id),]
kin_rslt_GBM = ActiveDriver(sequences, sequence_disorder, GBM_muts, kinase_domains, simplified=TRUE)

kin_results = ActiveDriver(sequences, sequence_disorder, mutations, kinase_domains, simplified=TRUE)
</code></pre>

<hr>
<h2 id='kinase_domains'>Example kinase domains for ActiveDriver</h2><span id='topic+kinase_domains'></span>

<h3>Description</h3>

<p>A dataset describing kinase domains. The variables are as follows:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ActiveDriver_data)
</code></pre>


<h3>Format</h3>

<p>A data frame with 1 observation of 4 variables</p>


<h3>Details</h3>


<ul>
<li><p> gene. the gene symbol of the gene where the kinase domain occurs
</p>
</li>
<li><p> position. the position in the protein sequence where the kinase domain begins
</p>
</li>
<li><p> phos. TRUE
</p>
</li>
<li><p> residue. the kinase domain residues
</p>
</li></ul>


<hr>
<h2 id='mutations'>Example mutations for ActiveDriver</h2><span id='topic+mutations'></span>

<h3>Description</h3>

<p>A dataset describing mis-sense mutations (i.e., substitutions in proteins). The variables are as follows:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ActiveDriver_data)
</code></pre>


<h3>Format</h3>

<p>A data frame with 408 observations of 5 variables</p>


<h3>Details</h3>


<ul>
<li><p> gene. the mutated gene
</p>
</li>
<li><p> sample_id. the sample where the mutation originates
</p>
</li>
<li><p> position. the position in the protein sequence where the mutation occurs
</p>
</li>
<li><p> wt_residue. the wild-type residue
</p>
</li>
<li><p> mut_residue. the mutant residue
</p>
</li></ul>


<hr>
<h2 id='phosphosites'>Example phosphosites for ActiveDriver</h2><span id='topic+phosphosites'></span>

<h3>Description</h3>

<p>A dataset describing protein phosphorylation sites. The variables are as follows:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ActiveDriver_data)
</code></pre>


<h3>Format</h3>

<p>A data frame with 131 observations of 4 variables</p>


<h3>Details</h3>


<ul>
<li><p> gene. the gene symbol the phosphosite occurs in
</p>
</li>
<li><p> position. the position in the protein sequence where the phosphosite occurs
</p>
</li>
<li><p> residue. the phosphosite residue
</p>
</li>
<li><p> kinase. the kinase that phosphorylates this site
</p>
</li></ul>


<hr>
<h2 id='read_fasta'>Read FASTA file as character vector.</h2><span id='topic+read_fasta'></span>

<h3>Description</h3>

<p>Read FASTA file as character vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_fasta(fname)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_fasta_+3A_fname">fname</code></td>
<td>
<p>name of file to be read.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector with names corresponding to annotations from FASTA.
</p>

<hr>
<h2 id='sequence_disorder'>Example protein disorder for ActiveDriver</h2><span id='topic+sequence_disorder'></span>

<h3>Description</h3>

<p>A dataset containing the disorder of four proteins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ActiveDriver_data)
</code></pre>


<h3>Format</h3>

<p>A named character vector with 4 elements</p>

<hr>
<h2 id='sequences'>Example protein sequences for ActiveDriver</h2><span id='topic+sequences'></span>

<h3>Description</h3>

<p>A dataset containing the sequences of four proteins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ActiveDriver_data)
</code></pre>


<h3>Format</h3>

<p>A named character vector with 4 elements</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
