<!DOCTYPE html><html><head><title>Help for package adephylo</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {adephylo}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#.tipToRoot'><p>Low-level auxiliary functions for adephylo</p></a></li>
<li><a href='#abouheif.moran'><p>Abouheif's test based on Moran's I</p></a></li>
<li><a href='#adephylo-package'><p>The adephylo package</p></a></li>
<li><a href='#bullseye'><p>Fan-like phylogeny with possible representation of traits on tips</p></a></li>
<li><a href='#carni19'><p>Phylogeny and quantative trait of carnivora</p></a></li>
<li><a href='#carni70'><p>Phylogeny and quantitative traits of carnivora</p></a></li>
<li><a href='#distRoot'><p>Compute the distance of tips to the root</p></a></li>
<li><a href='#distTips'><p>Compute some phylogenetic distance between tips</p></a></li>
<li><a href='#listDD'><p>List direct descendants for all nodes of a tree</p></a></li>
<li><a href='#listTips'><p>List tips descendings from all nodes of a tree</p></a></li>
<li><a href='#lizards'><p>Phylogeny and quantitative traits of lizards</p></a></li>
<li><a href='#maples'><p>Phylogeny and quantitative traits of flowers</p></a></li>
<li><a href='#mjrochet'><p>Phylogeny and quantitative traits of teleos fishes</p></a></li>
<li><a href='#moran.idx'><p>Computes Moran's index for a variable</p></a></li>
<li><a href='#orthobasis.phylo'><p>Computes Moran's eigenvectors from a tree or a phylogenetic proximity matrix</p></a></li>
<li><a href='#orthogram'><p>Orthonormal decomposition of variance</p></a></li>
<li><a href='#palm'><p>Phylogenetic and quantitative traits of amazonian palm trees</p></a></li>
<li><a href='#ppca'><p>Phylogenetic principal component analysis</p></a></li>
<li><a href='#procella'><p>Phylogeny and quantitative traits of birds</p></a></li>
<li><a href='#proxTips'><p>Compute some phylogenetic proximities between tips</p></a></li>
<li><a href='#sp.tips'><p>Find the shortest path between tips of a tree</p></a></li>
<li><a href='#table.phylo4d'><p>Graphical display of phylogeny and traits</p></a></li>
<li><a href='#tithonia'><p>Phylogeny and quantitative traits of flowers</p></a></li>
<li><a href='#treePart'><p>Define partitions of tips according from a tree</p></a></li>
<li><a href='#ungulates'><p>Phylogeny and quantitative traits of ungulates.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.1-16</td>
</tr>
<tr>
<td>Title:</td>
<td>Exploratory Analyses for the Phylogenetic Comparative Method</td>
</tr>
<tr>
<td>Depends:</td>
<td>methods, ade4 (&ge; 1.7-10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>phylobase, ape, adegenet</td>
</tr>
<tr>
<td>Description:</td>
<td>Multivariate tools to analyze comparative data, i.e. a phylogeny and some traits measured for each taxa. The package contains functions to represent comparative data, compute phylogenetic proximities, perform multivariate analysis with phylogenetic constraints and test for the presence of phylogenetic autocorrelation. The package is described in Jombart et al (2010) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtq292">doi:10.1093/bioinformatics/btq292</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-05 08:51:41 UTC; siberchicot</td>
</tr>
<tr>
<td>Author:</td>
<td>Stéphane Dray <a href="https://orcid.org/0000-0003-0153-1105"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Thibaut Jombart [aut],
  Anders Ellern Bilgrau [ctb],
  Aurélie Siberchicot
    <a href="https://orcid.org/0000-0002-7638-8318"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Aurélie Siberchicot &lt;aurelie.siberchicot@univ-lyon1.fr&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-06 16:20:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='.tipToRoot'>Low-level auxiliary functions for adephylo</h2><span id='topic+.tipToRoot'></span>

<h3>Description</h3>

<p>These hidden functions are utils for adephylo, used by other functions.
Regular users can use them as well, but no validity checks are performed for
the arguments: speed is here favored over safety.  Most of these functions
handle trees inheriting phylo4 class.<br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.tipToRoot(x, tip, root, include.root = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".tipToRoot_+3A_x">x</code></td>
<td>
<p>A valid tree of class phylo4.</p>
</td></tr>
<tr><td><code id=".tipToRoot_+3A_tip">tip</code></td>
<td>
<p>An integer identifying a tip by its numbers.</p>
</td></tr>
<tr><td><code id=".tipToRoot_+3A_root">root</code></td>
<td>
<p>An integer identifying the root of the tree by its number.</p>
</td></tr>
<tr><td><code id=".tipToRoot_+3A_include.root">include.root</code></td>
<td>
<p>a logical stating whether the root must be included as a
node of the path from tip to root (TRUE), or not (FALSE, default).</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>.tipToRoot</code> finds the set of nodes between a tip and the root of a
tree.<br />
</p>


<h3>Value</h3>

<p><code>.tipToRoot</code>: a vector of named integers identifying nodes.<br />
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:tjombart@imperial.ac.uk">tjombart@imperial.ac.uk</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(require(ape) &amp; require(phylobase)){
## make a tree
x &lt;- as(rtree(20),"phylo4")
plot(x,show.node=TRUE)

## .tipToRoot
root &lt;- rootNode(x)
.tipToRoot(x, 1, root)
lapply(1:nTips(x), function(i) .tipToRoot(x, i, root))
}

</code></pre>

<hr>
<h2 id='abouheif.moran'>Abouheif's test based on Moran's I</h2><span id='topic+abouheif.moran'></span>

<h3>Description</h3>

<p>The test of Abouheif (1999) is designed to detect phylogenetic
autocorrelation in a quantitative trait. Pavoine <em>et al.</em> (2008) have
shown that this tests is in fact a Moran's I test using a particular
phylogenetic proximity between tips (see details). The function
<code>abouheif.moran</code> performs basically Abouheif's test for several traits
at a time, but it can incorporate other phylogenetic proximities as well.<br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>abouheif.moran(
  x,
  W = NULL,
  method = c("oriAbouheif", "patristic", "nNodes", "Abouheif", "sumDD"),
  f = function(x) {
     1/x
 },
  nrepet = 999,
  alter = c("greater", "less", "two-sided")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="abouheif.moran_+3A_x">x</code></td>
<td>
<p>a data frame with continuous variables, or a phylo4d
object (i.e. containing both a tree, and tip data). In the latter case,
<code>method</code> argument is used to determine which proximity should be used.</p>
</td></tr>
<tr><td><code id="abouheif.moran_+3A_w">W</code></td>
<td>
<p>a <em>n</em> by <em>n</em> matrix (<em>n</em> being the number rows in x)
of phylogenetic proximities, as produced by <code><a href="#topic+proxTips">proxTips</a></code>.</p>
</td></tr>
<tr><td><code id="abouheif.moran_+3A_method">method</code></td>
<td>
<p>a character string (full or unambiguously abbreviated)
specifying the type of proximity to be used. By default, the proximity used
is that of the original Abouheif's test. See details in
<code><a href="#topic+proxTips">proxTips</a></code> for information about other methods.</p>
</td></tr>
<tr><td><code id="abouheif.moran_+3A_f">f</code></td>
<td>
<p>a function to turn a distance into a proximity (see
<code><a href="#topic+proxTips">proxTips</a></code>).</p>
</td></tr>
<tr><td><code id="abouheif.moran_+3A_nrepet">nrepet</code></td>
<td>
<p>number of random permutations of data for the randomization
test</p>
</td></tr>
<tr><td><code id="abouheif.moran_+3A_alter">alter</code></td>
<td>
<p>a character string specifying the alternative hypothesis, must
be one of &quot;greater&quot; (default), &quot;less&quot; or &quot;two-sided&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that the original Abouheif's proximity (Abouheif, 1999; Pavoine
<em>et al.</em> 2008) unifies Moran's I and Geary'c tests (Thioulouse <em>et
al.</em> 1995).<br />
</p>
<p><code>abouheif.moran</code> can be used in two ways:<br /> - providing a data.frame
of traits (<code>x</code>) and a matrix of phylogenetic proximities (<code>W</code>)<br />
- providing a phylo4d object (<code>x</code>) and specifying the
type of proximity to be used (<code>method</code>).
</p>
<p><code>W</code> is a squared symmetric matrix whose terms are all positive or
null.<br />
</p>
<p><code>W</code> is firstly transformed in frequency matrix A by dividing it by the
total sum of data matrix : </p>
<p style="text-align: center;"><code class="reqn">a_{ij} =
\frac{W_{ij}}{\sum_{i=1}^{n}\sum_{j=1}^{n}W_{ij}}</code>
</p>
<p> The neighbouring weights is defined by the matrix <code class="reqn">D =
diag(d_1,d_2, \ldots)</code> where <code class="reqn">d_i = \sum_{j=1}^{n}W_{ij}</code>. For each vector x of the data frame x, the test is based on the Moran
statistic <code class="reqn">x^{t}Ax</code> where x is D-centred.
</p>


<h3>Value</h3>

<p>Returns an object of class <code>krandtest</code> (randomization tests
from ade4), containing one Monte Carlo test for each trait.
</p>


<h3>Author(s)</h3>

<p>Original code from ade4 (gearymoran function) by Sebastien Ollier<br />
Adapted and maintained by Thibaut Jombart &lt;tjombart@imperial.ac.uk&gt;.
</p>


<h3>References</h3>

<p>Thioulouse, J., Chessel, D. and Champely, S. (1995) Multivariate analysis of
spatial patterns: a unified approach to local and global structures.
<em>Environmental and Ecological Statistics</em>, <b>2</b>, 1&ndash;14.
</p>


<h3>See Also</h3>

<p>- <code><a href="ade4.html#topic+gearymoran">gearymoran</a></code> from the ade4 package<br /> -
<code><a href="ape.html#topic+Moran.I">Moran.I</a></code> from the ape package for the classical Moran's I
test. <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if(require(ade4)&amp;&amp; require(ape) &amp;&amp; require(phylobase)){
## load data
data(ungulates)
tre &lt;- read.tree(text=ungulates$tre)
x &lt;- phylo4d(tre, ungulates$tab)

## Abouheif's tests for each trait
myTests &lt;- abouheif.moran(x)
myTests
plot(myTests)

## a variant using another proximity
plot(abouheif.moran(x, method="nNodes") )

## Another example

data(maples)
tre &lt;- read.tree(text=maples$tre)
dom &lt;- maples$tab$Dom

## Abouheif's tests for each trait (equivalent to Cmean)
W1 &lt;- proxTips(tre,method="oriAbouheif")
abouheif.moran(dom,W1)

## Equivalence with moran.idx

W2 &lt;- proxTips(tre,method="Abouheif")
abouheif.moran(dom,W2)
moran.idx(dom,W2) 
}

</code></pre>

<hr>
<h2 id='adephylo-package'>The adephylo package</h2><span id='topic+adephylo-package'></span><span id='topic+adephylo'></span>

<h3>Description</h3>

<p>This package is devoted to exploratory analysis of phylogenetic comparative
data. It re-implements and extends phylogenetic procedures from the
<code>ade4</code> package (which are now deprecated).<br />
</p>


<h3>Details</h3>

<p>Comparative data (phylogeny+traits) are handled as phylo4d
objects, a canonical class implemented by the <code>phylobase</code> package.
Trees are handled as <code><a href="ape.html#topic+read.tree">phylo</a></code> objects (from the
<code>ape</code> package) or as phylo4 objects (<code>phylobase</code>'s
extension of <code>phylo</code> objects).<br />
</p>
<p>Main functionalities of <code>adephylo</code> are summarized below.<br />
</p>
<p>=== TOPOLOGICAL INFORMATION ===<br /> Several functions allow one to retrieve
topological information from a tree; such information can be used, for
instance, as a basis to compute distances or proximities between tips.<br />
</p>
<p>- <code><a href="#topic+listDD">listDD</a></code>: lists the direct descendants from each node of a
tree.<br />
</p>
<p>- <code><a href="#topic+listTips">listTips</a></code>: lists the tips descending from each node of a
tree.<br />
</p>
<p>- <code><a href="#topic+.tipToRoot">.tipToRoot</a></code>: finds the set of nodes between a tip and the
root of a tree.<br />
</p>
<p>- <code><a href="#topic+sp.tips">sp.tips</a></code>: finds the shortest path between tips of a tree.<br />
</p>
<p>- <code><a href="#topic+treePart">treePart</a></code>: defines partitions of tips reflecting the topology
of a tree. This function can output non-independent dummy vectors, or
alternatively an orthonormal basis used by the orthogram procedure.<br />
</p>
<p>=== PHYLOGENETIC PROXIMITIES/DISTANCES ===<br /> Several phylogenetic
proximities and distances are implemented. Auxiliary function easing the
computation of other distances/proximities are also provided:<br />
</p>
<p>- <code><a href="#topic+distRoot">distRoot</a></code>: computes different distances of a set of tips to
the root.<br />
</p>
<p>- <code><a href="#topic+distTips">distTips</a></code>: computes different pairwise distances in a set of
tips.<br />
</p>
<p>- <code><a href="#topic+proxTips">proxTips</a></code>: computes different proximities between a set of
tips.<br />
</p>
<p>=== MEASURES/TESTS OF PHYLOGENETIC AUTOCORRELATION ===<br /> Several procedures
allow one to measure, and/or test phylogenetic signal in biological
traits:<br />
</p>
<p>- <code><a href="#topic+abouheif.moran">abouheif.moran</a></code>: performs Abouheif's test, designed to detect
phylogenetic autocorrelation in a quantitative trait. This implementation is
not based on original heuristic procedure, but on the exact formulation
proposed by Pavoine et al. (2008), showing that the test is in fact a
Moran's index test. This implementation further extends the procedure by
allowing any measure of phylogenetic proximity (5 are proposed).<br />
</p>
<p>- <code><a href="#topic+orthogram">orthogram</a></code>: performs the orthonormal decomposition of
variance of a quantitative variable on an orthonormal basis as in Ollier et
al. (2005). It also returns the results of five non parametric tests
associated to the variance decomposition.<br />
</p>
<p>- <code><a href="#topic+moran.idx">moran.idx</a></code>: computes Moran's index of autocorrelation given a
variable and a matrix of proximities among observations (no test).<br />
</p>
<p>=== MODELLING/INVESTIGATION OF PHYLOGENETIC SIGNAL ===<br /> Rather than
testing or measuring phylogenetic autocorrelation, these procedures can be
used for further investigation of phylogenetic signal. Some, like
<code><a href="#topic+me.phylo">me.phylo</a></code>, can be used to remove phylogenetic autocorrelation.
Others can be used to understand the nature of this autocorrelation (i.e.,
to ascertain which traits and tips are concerned by phylogenetic
non-independence).<br />
</p>
<p>- <code><a href="#topic+me.phylo">me.phylo</a></code>/<code><a href="#topic+orthobasis.phylo">orthobasis.phylo</a></code>: these synonymous
functions compute Moran's eigenvectors (ME) associated to a tree. These
vectors model different observable phylogenetic signals. They can be used as
covariables to remove phylogenetic autocorrelation from data.<br />
</p>
<p>- <code><a href="#topic+orthogram">orthogram</a></code>: the orthogram mentioned above also provides a
description of how biological variability is structured on a phylogeny.<br />
</p>
<p>- <code><a href="#topic+ppca">ppca</a></code>: performs a phylogenetic Principal Component Analysis
(pPCA, Jombart et al. 2010). This multivariate method investigates
phylogenetic patterns in a set of quantitative traits.<br />
</p>
<p>=== GRAPHICS ===<br /> Some plotting functions are proposed, most of them being
devoted to representing phylogeny and a quantitative information at the same
time.<br />
</p>
<p>- <code><a href="#topic+table.phylo4d">table.phylo4d</a></code>: fairly customisable way of representing
traits onto the tips of a phylogeny. Several traits can be plotted in a
single graphic.<br />
</p>
<p>- <code><a href="#topic+bullseye">bullseye</a></code>: an alternative to <code><a href="#topic+table.phylo4d">table.phylo4d</a></code>
based on fan-like representation, better for large trees.<br />
</p>
<p>- <code><a href="#topic+scatter.ppca">scatter.ppca</a></code>, <code><a href="#topic+screeplot.ppca">screeplot.ppca</a></code>,
<code><a href="#topic+plot.ppca">plot.ppca</a></code>: several plots associated to a phylogenetic
principal component analysis (see <code><a href="#topic+ppca">ppca</a></code>).<br />
</p>
<p>=== DATASETS ===<br /> Several datasets are also proposed. Some of these
datasets replace former version from <code>ade4</code>, which are now deprecated.
Here is a list of available datasets: <code><a href="#topic+carni19">carni19</a></code>,
<code><a href="#topic+carni70">carni70</a></code>, <code><a href="#topic+lizards">lizards</a></code>, <code><a href="#topic+maples">maples</a></code>,
<code><a href="#topic+mjrochet">mjrochet</a></code>, <code><a href="#topic+palm">palm</a></code>, <code><a href="#topic+procella">procella</a></code>,
<code><a href="#topic+tithonia">tithonia</a></code>, and <code><a href="#topic+ungulates">ungulates</a></code>.<br />
</p>
<p>To cite adephylo, please use the reference given by
<code>citation("adephylo")</code>.
</p>

<table>
<tr>
 <td style="text-align: left;"> Package: </td><td style="text-align: left;"> adephylo</td>
</tr>
<tr>
 <td style="text-align: left;"> Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;"> License: </td><td style="text-align: left;"> GPL (&gt;=2) </td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Thibaut Jombart &lt;tjombart@imperial.ac.uk&gt;<br /> with contributions
Stephane Dray &lt;stephane.dray@univ-lyon1.fr&gt; and Anders Ellern Bilgrau
&lt;abilgrau@math.aau.dk&gt;. <br /> Parts of former code from <code>ade4</code> by Daniel
Chessel and Sebastien Ollier.
</p>


<h3>See Also</h3>

<p>The <code>ade4</code> package for multivariate analysis.
</p>

<hr>
<h2 id='bullseye'>Fan-like phylogeny with possible representation of traits on tips</h2><span id='topic+bullseye'></span>

<h3>Description</h3>

<p>This function represents a phylogeny as a fan, using circles to provide a
legend for distances and optionally colored symbols to represent traits
associated to the tips of the tree. This function uses and is compatible
with ape's <code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bullseye(
  phy,
  traits = NULL,
  col.tips.by = NULL,
  col.pal = spectral,
  circ.n = 6,
  circ.bg = transp("royalblue", 0.1),
  circ.unit = NULL,
  legend = TRUE,
  leg.posi = "bottomleft",
  leg.title = "",
  leg.bg = "white",
  traits.inset = 1.1,
  traits.space = 0.05,
  traits.pch = 19,
  traits.cex = 1,
  alpha = 1,
  axis = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bullseye_+3A_phy">phy</code></td>
<td>
<p>a tree in <code>phylo</code>, phylo4 or
phylo4d format.</p>
</td></tr>
<tr><td><code id="bullseye_+3A_traits">traits</code></td>
<td>
<p>an optional data.frame of traits.</p>
</td></tr>
<tr><td><code id="bullseye_+3A_col.tips.by">col.tips.by</code></td>
<td>
<p>an optional vector used to define colors for tip labels;
if unamed, must be ordered in the same order as <code>phy$tip.label</code>.</p>
</td></tr>
<tr><td><code id="bullseye_+3A_col.pal">col.pal</code></td>
<td>
<p>a function generating colors according to a given palette;
several palettes can be provided as a list, in the case of several traits;
the first palette is always reserved for the tip colors; this argument is
recycled.</p>
</td></tr>
<tr><td><code id="bullseye_+3A_circ.n">circ.n</code></td>
<td>
<p>the number of circles for the distance annotations.</p>
</td></tr>
<tr><td><code id="bullseye_+3A_circ.bg">circ.bg</code></td>
<td>
<p>the color of the circles.</p>
</td></tr>
<tr><td><code id="bullseye_+3A_circ.unit">circ.unit</code></td>
<td>
<p>the unit of the circles; if NULL, determined automatically
from the data.</p>
</td></tr>
<tr><td><code id="bullseye_+3A_legend">legend</code></td>
<td>
<p>a logical specifying whether a legend should be plotted; only
one legend is displayed, with priority to tip colors first, and then to the
first trait.</p>
</td></tr>
<tr><td><code id="bullseye_+3A_leg.posi">leg.posi</code>, <code id="bullseye_+3A_leg.title">leg.title</code>, <code id="bullseye_+3A_leg.bg">leg.bg</code></td>
<td>
<p>position, title and background for the
legend.</p>
</td></tr>
<tr><td><code id="bullseye_+3A_traits.inset">traits.inset</code></td>
<td>
<p>inset for positioning the traits; 1 corresponds to the
circle crossing the furthest tip, 0 to the center of the plot.</p>
</td></tr>
<tr><td><code id="bullseye_+3A_traits.space">traits.space</code></td>
<td>
<p>a coefficient indicating the spacing between traits.</p>
</td></tr>
<tr><td><code id="bullseye_+3A_traits.pch">traits.pch</code>, <code id="bullseye_+3A_traits.cex">traits.cex</code></td>
<td>
<p>type and size of the symbols used for the
traits; recycled if needed.</p>
</td></tr>
<tr><td><code id="bullseye_+3A_alpha">alpha</code></td>
<td>
<p>alpha value to be used for the color transparency, between 0
(invisible) and 1 (plain).</p>
</td></tr>
<tr><td><code id="bullseye_+3A_axis">axis</code></td>
<td>
<p>a logical indicating whether an axis should be displayed.</p>
</td></tr>
<tr><td><code id="bullseye_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to plot methods from <code>ape</code>.
See <code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, function produces only a plot. 
</p>
<p>The phylo4d class for storing <code>phylogeny+data</code>.<br />
</p>
<p><code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code> from the <code>ape</code> package.<br />
</p>
<p><code><a href="ade4.html#topic+dotchart.phylog">dotchart.phylog</a></code>.
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:tjombart@imperial.ac.uk">tjombart@imperial.ac.uk</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+table.phylo4d">table.phylo4d</a></code> for non-radial plots.<br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(require(ape) &amp;&amp; require(phylobase) &amp;&amp; require(adegenet)){

data(lizards)
tre &lt;- read.tree(text=lizards$hprA) # make a tree

## basic plots
bullseye(tre)
bullseye(tre, lizards$traits)

## customized
oldpar &lt;- par(mar=c(6,6,6,6))
bullseye(tre, lizards$traits, traits.cex=sqrt(1:7), alpha=.7,
         legend=FALSE, circ.unit=10, circ.bg=transp("black",.1),
         edge.width=2)
par(oldpar)
}

</code></pre>

<hr>
<h2 id='carni19'>Phylogeny and quantative trait of carnivora</h2><span id='topic+carni19'></span>

<h3>Description</h3>

<p>This data set describes the phylogeny of carnivora as reported by
Diniz-Filho et al. (1998). It also gives the body mass of these 19 species.
</p>


<h3>Format</h3>

<p><code>carni19</code> is a list containing the 2 following objects :
</p>
 <dl>
<dt>tre</dt><dd><p>is a character string giving the phylogenetic tree in
Newick format.</p>
</dd> <dt>bm</dt><dd><p>is a numeric vector which values correspond to the
body mass of the 19 species (log scale).</p>
</dd> </dl>



<h3>Note</h3>

<p>This dataset replaces the former version in ade4.
</p>


<h3>Source</h3>

<p>Diniz-Filho, J. A. F., de Sant'Ana, C.E.R. and Bini, L.M. (1998) An
eigenvector method for estimating phylogenetic inertia. <em>Evolution</em>,
<b>52</b>, 1247&ndash;1262.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if(require(ape) &amp;&amp; require(phylobase)){

data(carni19)
tre &lt;- read.tree(text=carni19$tre)
x &lt;- phylo4d(tre, data.frame(carni19$bm))
table.phylo4d(x, ratio=.5, center=FALSE)
}


</code></pre>

<hr>
<h2 id='carni70'>Phylogeny and quantitative traits of carnivora</h2><span id='topic+carni70'></span>

<h3>Description</h3>

<p>This data set describes the phylogeny of 70 carnivora as reported by
Diniz-Filho and Torres (2002). It also gives the geographic range size and
body size corresponding to these 70 species.
</p>


<h3>Format</h3>

<p><code>carni70</code> is a list containing the 2 following objects:
</p>
 <dl>
<dt>tre</dt><dd><p>is a character string giving the phylogenetic tree in
Newick format.  Branch lengths are expressed as divergence times (millions
of years)</p>
</dd> <dt>tab</dt><dd><p>is a data frame with 70 species and two traits: size
(body size (kg)) ; range (geographic range size (km)).</p>
</dd> </dl>



<h3>Note</h3>

<p>This dataset replaces the former version in ade4.
</p>


<h3>Source</h3>

<p>Diniz-Filho, J. A. F., and N. M. Torres. (2002) Phylogenetic
comparative methods and the geographic range size-body size relationship in
new world terrestrial carnivora. <em>Evolutionary Ecology</em>, <b>16</b>,
351&ndash;367.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if(require(ape) &amp;&amp; require(phylobase)){

data(carni70)
rownames(carni70$tab) &lt;- gsub("_", ".", rownames(carni70$tab))
tre &lt;- read.tree(text=carni70$tre)
x &lt;- phylo4d(tre, carni70$tab)
table.phylo4d(x)

oldpar &lt;- par(mar=rep(.1,4))
table.phylo4d(x,cex.lab=.5, show.n=FALSE, ratio=.5)


## transform size in log and test for a phylogenetic signal
size &lt;- log(carni70$tab)[,1]
names(size) &lt;- row.names(carni70$tab)
orthogram(size, tre)

## transform range and test for a phylogenetic signal
yrange &lt;- scale(carni70$tab)[,2]
names(yrange) &lt;- row.names(carni70$tab)
orthogram(yrange, tre)

par(oldpar)
}


</code></pre>

<hr>
<h2 id='distRoot'>Compute the distance of tips to the root</h2><span id='topic+distRoot'></span>

<h3>Description</h3>

<p>The function <code>distRoot</code> computes the distance of a set of tips to the
root. Several distances can be used, defaulting to the sum of branch
lengths.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distRoot(
  x,
  tips = "all",
  method = c("patristic", "nNodes", "Abouheif", "sumDD")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="distRoot_+3A_x">x</code></td>
<td>
<p>a tree of class <code><a href="ape.html#topic+read.tree">phylo</a></code>,
phylo4 or phylo4d.</p>
</td></tr>
<tr><td><code id="distRoot_+3A_tips">tips</code></td>
<td>
<p>A vector of integers identifying tips by their numbers, or a
vector of characters identifying tips by their names.</p>
</td></tr>
<tr><td><code id="distRoot_+3A_method">method</code></td>
<td>
<p>a character string (full or abbreviated without ambiguity)
specifying the method used to compute distances ; possible values are:<br /> -
<code>patristic</code>: patristic distance, i.e. sum of branch lengths <br /> -
<code>nNodes</code>: number of nodes on the path between the nodes <br /> -
<code>Abouheif</code>: Abouheif's distance (see details) <br /> - <code>sumDD</code>: sum
of direct descendants of all nodes on the path (see details) <br /></p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Abouheif</code> distance refers to the phylogenetic distance underlying the
test of Abouheif (see references). Let P be the set of all the nodes in the
path going from <code>node1</code> to <code>node2</code>. Let DDP be the number of
direct descendants from each node in P. Then, the so-called 'Abouheif'
distance is the product of all terms in DDP.<br />
</p>
<p><code>sumDD</code> refers to a phylogenetic distance quite similar to that of
Abouheif. We consider the same sets P and DDP. But instead of computing the
product of all terms in DDP, this distance computes the sum of all terms in
DDP.
</p>


<h3>Value</h3>

<p>A numeric vector containing one distance value for each tip.
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:tjombart@imperial.ac.uk">tjombart@imperial.ac.uk</a>
</p>


<h3>References</h3>

<p>Pavoine, S.; Ollier, S.; Pontier, D. &amp; Chessel, D. (2008)
Testing for phylogenetic signal in life history variable: Abouheif's test
revisited. <em>Theoretical Population Biology</em>: <b>73</b>, 79-91.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+distTips">distTips</a></code> which computes the same phylogenetic
distances, but between tips.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(require(ape) &amp; require(phylobase)){
## make a tree
x &lt;- as(rtree(50),"phylo4")
## compute 4 different distances
met &lt;- c("patristic","nNodes","Abouheif","sumDD")
D &lt;- lapply(met, function(e) distRoot(x, method=e) )
names(D) &lt;- met
D &lt;- as.data.frame(D)

## plot these distances along with the tree
temp &lt;- phylo4d(x, D)
table.phylo4d(temp, show.node=FALSE, cex.lab=.6)
}

</code></pre>

<hr>
<h2 id='distTips'>Compute some phylogenetic distance between tips</h2><span id='topic+distTips'></span>

<h3>Description</h3>

<p>The function <code>distTips</code> computes a given distance between a set of tips
of a phylogeny. A vector of tips is supplied: distances between all possible
pairs of these tips are computed.  The distances are computed from the
shortest path between the tips. Several distances can be used, defaulting to
the sum of branch lengths (see argument <code>method</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distTips(
  x,
  tips = "all",
  method = c("patristic", "nNodes", "Abouheif", "sumDD"),
  useC = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="distTips_+3A_x">x</code></td>
<td>
<p>a tree of class <code><a href="ape.html#topic+read.tree">phylo</a></code>,
phylo4 or phylo4d.</p>
</td></tr>
<tr><td><code id="distTips_+3A_tips">tips</code></td>
<td>
<p>A vector of integers identifying tips by their numbers, or a
vector of characters identifying tips by their names. Distances will be
computed between all possible pairs of tips.</p>
</td></tr>
<tr><td><code id="distTips_+3A_method">method</code></td>
<td>
<p>a character string (full or abbreviated without ambiguity)
specifying the method used to compute distances ; possible values are:<br /> -
<code>patristic</code>: patristic distance, i.e. sum of branch lengths <br /> -
<code>nNodes</code>: number of nodes on the path between the nodes <br /> -
<code>Abouheif</code>: Abouheif's distance (see details) <br /> - <code>sumDD</code>: sum
of direct descendants of all nodes on the path (see details) <br /></p>
</td></tr>
<tr><td><code id="distTips_+3A_usec">useC</code></td>
<td>
<p>a logical indicating whether computations should be performed
using compiled C code (TRUE, default), or using a pure R version (FALSE). C
version is several orders of magnitude faster, and R version is kept for
backward compatibility.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>An option (enabled by default) allows computations to be run using compiled
C code, which is much faster than pure R code. In this case, a matrix of all
pairwise distances is returned (i.e., <code>tips</code> argument is ignored).
</p>
<p><code>Abouheif</code> distance refers to the phylogenetic distance underlying the
test of Abouheif (see references). Let P be the set of all the nodes in the
path going from <code>node1</code> to <code>node2</code>. Let DDP be the number of
direct descendants from each node in P. Then, the so-called 'Abouheif'
distance is the product of all terms in DDP.<br />
</p>
<p><code>sumDD</code> refers to a phylogenetic distance quite similar to that of
Abouheif. We consider the same sets P and DDP. But instead of computing the
product of all terms in DDP, this distance computes the sum of all terms in
DDP.
</p>


<h3>Value</h3>

<p>An object of class <code>dist</code>, containing phylogenetic distances.
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:tjombart@imperial.ac.uk">tjombart@imperial.ac.uk</a>
</p>


<h3>References</h3>

<p>Pavoine, S.; Ollier, S.; Pontier, D. &amp; Chessel, D. (2008)
Testing for phylogenetic signal in life history variable: Abouheif's test
revisited. <em>Theoretical Population Biology</em>: <b>73</b>, 79-91.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+distTips">distTips</a></code> which computes several phylogenetic
distances between tips.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(require(ape) &amp; require(phylobase)){
## make a tree
x &lt;- as(rtree(10),"phylo4")
plot(x, show.node=TRUE)
axisPhylo()
## compute different distances
distTips(x, 1:3)
distTips(x, 1:3, "nNodes")
distTips(x, 1:3, "Abouheif")
distTips(x, 1:3, "sumDD")

## compare C and pure R code outputs
x &lt;- rtree(10)
all.equal(as.matrix(distTips(x)), as.matrix(distTips(x, useC=FALSE)))
all.equal(as.matrix(distTips(x, meth="nNode")),
   as.matrix(distTips(x, meth="nNode", useC=FALSE)))
all.equal(as.matrix(distTips(x, meth="Abou")),
   as.matrix(distTips(x, meth="Abou", useC=FALSE)))
all.equal(as.matrix(distTips(x, meth="sumDD")),
   as.matrix(distTips(x, meth="sumDD", useC=FALSE)))

## compare speed
x &lt;- rtree(50)
tim1 &lt;- system.time(distTips(x, useC=FALSE)) # old pure R version
tim2 &lt;- system.time(distTips(x)) # new version using C
tim1[c(1,3)]/tim2[c(1,3)] # C is about a thousand time faster in this case
}

</code></pre>

<hr>
<h2 id='listDD'>List direct descendants for all nodes of a tree</h2><span id='topic+listDD'></span>

<h3>Description</h3>

<p>The function <code>listDD</code> lists the direct descendants from each node of a
tree. The tree can be of class <code><a href="ape.html#topic+read.tree">phylo</a></code>,
phylo4 or phylo4d.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>listDD(x, nameBy = c("label", "number"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="listDD_+3A_x">x</code></td>
<td>
<p>A tree of class <code><a href="ape.html#topic+read.tree">phylo</a></code>,
phylo4 or phylo4d.</p>
</td></tr>
<tr><td><code id="listDD_+3A_nameby">nameBy</code></td>
<td>
<p>a character string indicating whether the returned list must
be named by node labels (&quot;label&quot;) or by node numbers (&quot;number&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list whose components are vectors of named nodes (or tips) for a
given internal node.
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:tjombart@imperial.ac.uk">tjombart@imperial.ac.uk</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+listTips">listTips</a></code> which lists the tips descending from each
node. <br />
</p>
<p><code><a href="#topic+treePart">treePart</a></code> which defines partitions of tips according to the
tree topology.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(require(ape) &amp; require(phylobase)){
## make a tree
x &lt;- as(rtree(20),"phylo4")
plot(x,show.node=TRUE)
listDD(x)
}

</code></pre>

<hr>
<h2 id='listTips'>List tips descendings from all nodes of a tree</h2><span id='topic+listTips'></span>

<h3>Description</h3>

<p>The function <code>listTips</code> lists the tips descending from each node of a
tree. The tree can be of class <code><a href="ape.html#topic+read.tree">phylo</a></code>,
phylo4 or phylo4d.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>listTips(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="listTips_+3A_x">x</code></td>
<td>
<p>A tree of class <code><a href="ape.html#topic+read.tree">phylo</a></code>,
phylo4 or phylo4d.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list whose components are vectors of named tips for a given node.
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:tjombart@imperial.ac.uk">tjombart@imperial.ac.uk</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+listDD">listDD</a></code> which lists the direct descendants for each
node. <br />
</p>
<p><code><a href="#topic+treePart">treePart</a></code> which defines partitions of tips according to the
tree topology.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(require(ape) &amp; require(phylobase)){
## make a tree
x &lt;- as(rtree(20),"phylo4")
plot(x,show.node=TRUE)
listTips(x)
}

</code></pre>

<hr>
<h2 id='lizards'>Phylogeny and quantitative traits of lizards</h2><span id='topic+lizards'></span>

<h3>Description</h3>

<p>This data set describes the phylogeny of 18 lizards as reported by Bauwens
and D\'iaz-Uriarte (1997). It also gives life-history traits corresponding
to these 18 species.
</p>


<h3>Format</h3>

<p><code>lizards</code> is a list containing the 3 following objects :
</p>
 <dl>
<dt>traits</dt><dd><p>is a data frame with 18 species and 8 traits.</p>
</dd>
<dt>hprA</dt><dd><p>is a character string giving the phylogenetic tree (hypothesized
phylogenetic relationships based on immunological distances) in Newick
format.</p>
</dd> <dt>hprB</dt><dd><p>is a character string giving the phylogenetic tree
(hypothesized phylogenetic relationships based on morphological
characteristics) in Newick format.</p>
</dd> </dl>



<h3>Details</h3>

<p>Variables of <code>lizards$traits</code> are the following ones : mean.L (mean
length (mm)), matur.L (length at maturity (mm)), max.L (maximum length
(mm)), hatch.L (hatchling length (mm)), hatch.m (hatchling mass (g)),
clutch.S (Clutch size), age.mat (age at maturity (number of months of
activity)), clutch.F (clutch frequency).
</p>


<h3>Note</h3>

<p>This dataset replaces the former version in ade4.
</p>


<h3>References</h3>

<p>Bauwens, D., and D\'iaz-Uriarte, R. (1997) Covariation of
life-history traits in lacertid lizards: a comparative study.
<em>American Naturalist</em>, <b>149</b>, 91&ndash;111.
</p>
<p>See a data description at <a href="http://pbil.univ-lyon1.fr/R/pdf/pps063.pdf">http://pbil.univ-lyon1.fr/R/pdf/pps063.pdf</a>
(in French).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if(require(ape) &amp;&amp; require(phylobase)){

## see data
data(lizards)
liz.tr &lt;- read.tree(tex=lizards$hprA) # make a tree
liz &lt;- phylo4d(liz.tr, lizards$traits) # make a phylo4d object
table.phylo4d(liz)

## compute and plot principal components
if(require(ade4)){
liz.pca1 &lt;- dudi.pca(lizards$traits, cent=TRUE,
   scale=TRUE, scannf=FALSE, nf=2) # PCA of traits
myPC &lt;- phylo4d(liz.tr, liz.pca1$li) # store PC in a phylo4d object
varlab &lt;- paste("Principal \ncomponent", 1:2) # make labels for PCs
table.phylo4d(myPC, ratio=.8, var.lab=varlab) # plot the PCs
add.scatter.eig(liz.pca1$eig,2,1,2,posi="topleft", inset=c(0,.15))
title("Phylogeny and the principal components")

## compute a pPCA ##
## remove size effect
temp &lt;- lapply(liz.pca1$tab, function(e) residuals(lm(e~-1+liz.pca1$li[,1])) )
temp &lt;- data.frame(temp)
row.names(temp) &lt;- tipLabels(liz)

## build corresponding phylo4d object
liz.noSize &lt;- phylo4d(liz.tr, temp)
ppca1 &lt;- ppca(liz.noSize, method="Abouheif", scale=FALSE, scannf=FALSE)
plot(ppca1)

}
}


</code></pre>

<hr>
<h2 id='maples'>Phylogeny and quantitative traits of flowers</h2><span id='topic+maples'></span>

<h3>Description</h3>

<p>This data set describes the phylogeny of 17 flowers as reported by Ackerly
and Donoghue (1998). It also gives 31 traits corresponding to these 17
species.
</p>


<h3>Format</h3>

<p><code>tithonia</code> is a list containing the 2 following objects : -
tre: a character string giving the phylogenetic tree in Newick format.<br /> -
tab: a data frame with 17 species and 31 traits.<br />
</p>


<h3>Note</h3>

<p>This dataset replaces the former version in ade4.
</p>


<h3>References</h3>

<p>Ackerly, D. D. and Donoghue, M.J. (1998) Leaf size, sappling
allometry, and Corner's rules: phylogeny and correlated evolution in Maples
(Acer). <em>American Naturalist</em>, <b>152</b>, 767&ndash;791.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if(require(ape) &amp;&amp; require(phylobase)){

data(maples)

## see the tree
tre &lt;- read.tree(text=maples$tre)
plot(tre)
axisPhylo()

## look at two variables
dom &lt;- maples$tab$Dom
bif &lt;- maples$tab$Bif
plot(bif,dom,pch = 20)
abline(lm(dom~bif)) # a strong negative correlation ?
summary(lm(dom~bif))
cor.test(bif,dom)

## look at the two variables onto the phylogeny
temp &lt;- phylo4d(tre, data.frame(dom,bif, row.names=tre$tip.label))
table.phylo4d(temp) # correlation is strongly linked to phylogeny

## use ape's PIC (phylogenetic independent contrasts)
pic.bif &lt;- pic(bif, tre)
pic.dom &lt;- pic(dom, tre)
cor.test(pic.bif, pic.dom) # correlation is no longer significant
}


</code></pre>

<hr>
<h2 id='mjrochet'>Phylogeny and quantitative traits of teleos fishes</h2><span id='topic+mjrochet'></span>

<h3>Description</h3>

<p>This data set describes the phylogeny of 49 teleos fishes as reported by
Rochet et al. (2000). It also gives life-history traits corresponding to
these 49 species.
</p>


<h3>Format</h3>

<p><code>mjrochet</code> is a list containing the 2 following objects :
</p>
 <dl>
<dt>tre</dt><dd><p>is a character string giving the phylogenetic tree in
Newick format.</p>
</dd> <dt>tab</dt><dd><p>is a data frame with 49 rows and 7 traits.</p>
</dd> </dl>



<h3>Details</h3>

<p>Variables of <code>mjrochet$tab</code> are the following ones : tm (age at
maturity (years)), lm (length at maturity (cm)), l05 (length at 5 per cent
survival (cm)), t05 (time to 5 per cent survival (years)), fb (slope of the
log-log fecundity-length relationship), fm (fecundity the year of maturity),
egg (volume of eggs (<code class="reqn">mm^{3}</code>)).
</p>


<h3>Note</h3>

<p>This dataset replaces the former version in ade4.
</p>


<h3>References</h3>

<p>Rochet, M. J., Cornillon, P-A., Sabatier, R. and Pontier, D.
(2000) Comparative analysis of phylogenic and fishing effects in life
history patterns of teleos fishes.  <em>Oikos</em>, <b>91</b>, 255&ndash;270.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if(require(ape) &amp;&amp; require(phylobase)){

data(mjrochet)
tre &lt;- read.tree(text=mjrochet$tre) # make a tree
traits &lt;- log((mjrochet$tab))

## build a phylo4d
mjr &lt;- phylo4d(tre, traits)

## see data
table.phylo4d(mjr,cex.lab=.5,show.node=FALSE,symb="square")

## perform Abouheif's test for each trait
mjr.tests &lt;- abouheif.moran(mjr, nrep=499)
mjr.tests


}

</code></pre>

<hr>
<h2 id='moran.idx'>Computes Moran's index for a variable</h2><span id='topic+moran.idx'></span>

<h3>Description</h3>

<p>This simple function computes Moran's index of autocorrelation given a
variable and a matrix of proximities among observations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moran.idx(x, prox, addInfo = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="moran.idx_+3A_x">x</code></td>
<td>
<p>a numeric vector whose autocorrelation is computed.</p>
</td></tr>
<tr><td><code id="moran.idx_+3A_prox">prox</code></td>
<td>
<p>a matrix of proximities between observations, as computed by the
<code><a href="#topic+proxTips">proxTips</a></code>. Off-diagonal terms must be positive or null, while
diagonal terms must all equal zero.</p>
</td></tr>
<tr><td><code id="moran.idx_+3A_addinfo">addInfo</code></td>
<td>
<p>a logical indicating whether supplementary info (null value,
minimum and maximum values) should be returned (TRUE) or not (FALSE,
default); if computed, these quantities are returned as attributes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The numeric value of Moran's index.
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:tjombart@imperial.ac.uk">tjombart@imperial.ac.uk</a>
</p>


<h3>References</h3>

<p>Moran, P.A.P. (1948) The interpretation of statistical maps.
<em>Journal of the Royal Statistical Society, B</em> <b>10</b>, 243&ndash;251.
</p>
<p>Moran, P.A.P. (1950) Notes on continuous stochastic phenomena.
<em>Biometrika</em>, <b>37</b>, 17&ndash;23.
</p>
<p>de Jong, P. and Sprenger, C. and van Veen, F. (1984) On extreme values of
Moran's I and Geary's c. <em>Geographical Analysis</em>, <b>16</b>, 17&ndash;24.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+proxTips">proxTips</a></code> which computes phylogenetic proximities
between tips of a phylogeny.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

## use maples dataset
if(require(ape) &amp;&amp; require(phylobase)){
data(maples)
tre &lt;- ape::read.tree(text=maples$tre)
dom &lt;- maples$tab$Dom
bif &lt;- maples$tab$Bif


## get a proximity matrix between tips 
W &lt;- proxTips(tre, met="Abouheif")

## compute Moran's I for two traits (dom and bif)
moran.idx(dom, W)
moran.idx(bif, W)
moran.idx(rnorm(nTips(tre)), W)

## build a simple permutation test for 'bif'
sim &lt;- replicate(499, moran.idx(sample(bif), W)) # permutations
sim &lt;- c(moran.idx(bif, W), sim)

pval &lt;- mean(sim&gt;=sim[1]) # right-tail p-value
pval

hist(sim, col="grey", main="Moran's I Monte Carlo test for 'bif'") # plot
mtext("Histogram of permutations and observation (in red)")
abline(v=sim[1], col="red", lwd=3)
}

</code></pre>

<hr>
<h2 id='orthobasis.phylo'>Computes Moran's eigenvectors from a tree or a phylogenetic proximity matrix</h2><span id='topic+orthobasis.phylo'></span><span id='topic+me.phylo'></span>

<h3>Description</h3>

<p>The function <code>orthobasis.phylo</code> (also nicknamed <code>me.phylo</code>)
computes Moran's eigenvectors (ME) associated to a tree. If the tree has 'n'
tips, (n-1) vectors will be produced. These vectors form an orthonormal
basis: they are centred to mean zero, have unit variance, and are
uncorrelated. Each vector models a different pattern of phylogenetic
autocorrelation. The first vectors are those with maximum positive
autocorrelation, while the last vectors are those with maximum negative
autocorrelation. ME can be used, for instance, as regressors to remove
phylogenetic autocorrelation from data (see references).<br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>orthobasis.phylo(
  x = NULL,
  prox = NULL,
  method = c("patristic", "nNodes", "oriAbouheif", "Abouheif", "sumDD"),
  f = function(x) {
     1/x
 }
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="orthobasis.phylo_+3A_x">x</code></td>
<td>
<p>A tree of class <code><a href="ape.html#topic+read.tree">phylo</a></code>,
phylo4 or phylo4d.</p>
</td></tr>
<tr><td><code id="orthobasis.phylo_+3A_prox">prox</code></td>
<td>
<p>a matrix of phylogenetic proximities as returned by
<code><a href="#topic+proxTips">proxTips</a></code>.</p>
</td></tr>
<tr><td><code id="orthobasis.phylo_+3A_method">method</code></td>
<td>
<p>a character string (full or abbreviated without ambiguity)
specifying the method used to compute proximities; possible values are:<br /> -
<code>patristic</code>: (inversed sum of) branch lengths <br /> - <code>nNodes</code>:
(inversed) number of nodes on the path between the nodes <br /> -
<code>oriAbouheif</code>: original Abouheif's proximity, with diagonal (see
details in <code><a href="#topic+proxTips">proxTips</a></code>) <br /> - <code>Abouheif</code>: Abouheif's
proximity (see details in <code><a href="#topic+proxTips">proxTips</a></code>) <br /> - <code>sumDD</code>:
(inversed) sum of direct descendants of all nodes on the path (see details
in <code><a href="#topic+proxTips">proxTips</a></code>).</p>
</td></tr>
<tr><td><code id="orthobasis.phylo_+3A_f">f</code></td>
<td>
<p>a function to change a distance into a proximity.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>ME can be obtained from a tree, specifying the phylogenetic proximity to be
used. Alternatively, they can be obtained directly from a matrix of
phylogenetic proximities as constructed by <code><a href="#topic+proxTips">proxTips</a></code>.
</p>


<h3>Value</h3>

<p>An object of class <code>orthobasis</code>. This is a data.frame with
Moran's eigenvectors in column, with special attributes:<br /> -
attr(...,&quot;values&quot;): Moran's index for each vector - attr(...,&quot;weights&quot;):
weights of tips; current implementation uses only uniform weights
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:tjombart@imperial.ac.uk">tjombart@imperial.ac.uk</a>
</p>


<h3>References</h3>

<p>Peres-Neto, P. (2006) A unified strategy for estimating and
controlling spatial, temporal and phylogenetic autocorrelation in ecological
models <em>Oecologica Brasiliensis</em> <b>10</b>: 105-119.<br />
</p>
<p>Dray, S.; Legendre, P. and Peres-Neto, P. (2006) Spatial modelling: a
comprehensive framework for principal coordinate analysis of neighbours
matrices (PCNM) <em>Ecological Modelling</em> <b>196</b>: 483-493.<br />
</p>
<p>Griffith, D. and Peres-Neto, P. (2006) Spatial modeling in ecology: the
flexibility of eigenfunction spatial analyses <em>Ecology</em> <b>87</b>:
2603-2613.<br />
</p>


<h3>See Also</h3>

<p>- <code><a href="#topic+proxTips">proxTips</a></code> which computes phylogenetic proximities
between tips.<br />
</p>
<p>- <code><a href="#topic+treePart">treePart</a></code> which can compute an orthobasis based on the
topology of a phylogeny.<br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(require(ape) &amp;&amp; require(phylobase)){

## SIMPLE EXAMPLE ##
## make a tree
x &lt;- rtree(50)

## compute Moran's eigenvectors
ME &lt;- me.phylo(x, met="Abouheif")
ME

## plot the 10 first vectors
obj &lt;- phylo4d(x, as.data.frame(ME[,1:10]))
table.phylo4d(obj, cex.sym=.7, cex.lab=.7)



## REMOVING PHYLOGENETIC AUTOCORRELATION IN A MODEL ##
## use example in ungulates dataset
data(ungulates)
tre &lt;- read.tree(text=ungulates$tre)
plot(tre)

## look at two traits
afbw &lt;- log(ungulates$tab[,1])
neonatw &lt;- log((ungulates$tab[,2]+ungulates$tab[,3])/2)
names(afbw) &lt;- tre$tip.label
names(neonatw) &lt;- tre$tip.label
plot(afbw, neonatw) # relationship between traits
lm1 &lt;- lm(neonatw~afbw)
abline(lm1)

lm1
resid1 &lt;- residuals(lm1)
orthogram(resid1, tre) # residuals are autocorrelated

## compute Moran's eigenvectors (ME)
myME &lt;- me.phylo(tre, method="Abou")
lm2 &lt;- lm(neonatw ~ myME[,1] + afbw) # use for ME as covariable
resid2 &lt;- residuals(lm2)
orthogram(resid2, tre) # there is no longer phylogenetic autocorrelation

## see the difference
table.phylo4d(phylo4d(tre, cbind.data.frame(resid1, resid2)))
}


</code></pre>

<hr>
<h2 id='orthogram'>Orthonormal decomposition of variance</h2><span id='topic+orthogram'></span>

<h3>Description</h3>

<p>This function performs the orthonormal decomposition of variance of a 
quantitative variable on an orthonormal basis. It also returns the results of
five non parametric tests associated to the variance decomposition.  It thus
provides tools (graphical displays and test) for analysing phylogenetic,
pattern in one quantitative trait. This implementation replace the
(deprecated) version from the <code>ade4</code> package.<br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>orthogram(
  x,
  tre = NULL,
  orthobas = NULL,
  prox = NULL,
  nrepet = 999,
  posinega = 0,
  tol = 1e-07,
  cdot = 1.5,
  cfont.main = 1.5,
  lwd = 2,
  nclass,
  high.scores = 0,
  alter = c("greater", "less", "two-sided")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="orthogram_+3A_x">x</code></td>
<td>
<p>a numeric vector corresponding to the quantitative variable</p>
</td></tr>
<tr><td><code id="orthogram_+3A_tre">tre</code></td>
<td>
<p>a tree of class <code><a href="ape.html#topic+read.tree">phylo</a></code>, 
phylo4 or phylo4d.</p>
</td></tr>
<tr><td><code id="orthogram_+3A_orthobas">orthobas</code></td>
<td>
<p>an object of class <code>'orthobasis'</code></p>
</td></tr>
<tr><td><code id="orthogram_+3A_prox">prox</code></td>
<td>
<p>a matrix of phylogenetic proximities as returned by 
<code><a href="#topic+proxTips">proxTips</a></code>.</p>
</td></tr>
<tr><td><code id="orthogram_+3A_nrepet">nrepet</code></td>
<td>
<p>an integer giving the number of permutations</p>
</td></tr>
<tr><td><code id="orthogram_+3A_posinega">posinega</code></td>
<td>
<p>a parameter for the ratio test. If posinega &gt; 0, the function
computes the ratio test.</p>
</td></tr>
<tr><td><code id="orthogram_+3A_tol">tol</code></td>
<td>
<p>a tolerance threshold for orthonormality condition</p>
</td></tr>
<tr><td><code id="orthogram_+3A_cdot">cdot</code></td>
<td>
<p>a character size for points on the cumulative decomposition 
display</p>
</td></tr>
<tr><td><code id="orthogram_+3A_cfont.main">cfont.main</code></td>
<td>
<p>a character size for titles</p>
</td></tr>
<tr><td><code id="orthogram_+3A_lwd">lwd</code></td>
<td>
<p>a character size for dash lines</p>
</td></tr>
<tr><td><code id="orthogram_+3A_nclass">nclass</code></td>
<td>
<p>a single number giving the number of cells for the histogram</p>
</td></tr>
<tr><td><code id="orthogram_+3A_high.scores">high.scores</code></td>
<td>
<p>a single number giving the number of vectors to return. If
&gt; 0, the function returns labels of vectors that explains the larger part 
of variance.</p>
</td></tr>
<tr><td><code id="orthogram_+3A_alter">alter</code></td>
<td>
<p>a character string specifying the alternative hypothesis, must 
be one of &quot;greater&quot; (default), &quot;less&quot; or &quot;two-sided&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Several orthonormal bases can be used. By default, basis is constructed from 
a partition of tips according to tree topology (as returned by 
<code><a href="#topic+treePart">treePart</a></code>); for this, the argument <code>tre</code> must be provided. 
Alternatively, one can provide an orthonormal basis as returned by 
<code><a href="#topic+orthobasis.phylo">orthobasis.phylo</a></code>/<code><a href="#topic+me.phylo">me.phylo</a></code> (argument 
<code>orthobas</code>), or provide a proximity matrix from which an orthobasis 
based on Moran's eigenvectors will be constructed (argument <code>prox</code>).
</p>
<p>The function computes the variance decomposition of a quantitative vector x 
on an orthonormal basis B. The variable is normalized given the uniform 
weight to eliminate problem of scales.  It plots the squared correlations 
<code class="reqn">R^{2}</code> between x and vectors of B (variance decomposition) and the 
cumulated squared correlations <code class="reqn">SR^{2}</code> (cumulative decomposition).
The function also provides five non parametric tests to test the existence of
autocorrelation. The tests derive from the five following statistics :
</p>
<p>- R2Max=<code class="reqn">\max(R^{2})</code>. It takes high value when a high part of 
the variability is explained by one score.<br /> - 
SkR2k=<code class="reqn">\sum_{i=1}^{n-1}(iR^{2}_i)</code>. It compares 
the part of variance explained by internal nodes to the one explained by end 
nodes.<br /> - Dmax=<code class="reqn">\max_{m=1,...,n-1}(\sum_{j=1}^{m}R^{2}_j - 
</code><code class="reqn"> 
\frac{m}{n-1})</code>. It examines
the accumulation of variance for a sequence of scores.<br /> - 
SCE=<code class="reqn">\sum_{m=1}^{n-1} (\sum_{j=1}^{m}R^{2}_j - 
</code><code class="reqn"> 
\frac{m}{n-1})^{2}</code>. It 
examines also the accumulation of variance for a sequence of scores.<br /> - 
ratio: depends of the parameter posinega. If posinega &gt; 0, the statistic 
ratio exists and equals <code class="reqn">\sum_{i=1}^{posinega}R^{2}_i</code>. It compares the part of variance explained by internal
nodes to the one explained by end nodes when we can define how many vectors
correspond to internal nodes.
</p>


<h3>Value</h3>

<p>If (high.scores = 0), returns an object of class <code>'krandtest'</code> 
(randomization tests) corresponding to the five non parametric tests. <br /> 
<br /> If (high.scores &gt; 0), returns a list containg : </p>
<table>
<tr><td><code>w</code></td>
<td>
<p>: an object of
class <code>'krandtest'</code> (randomization tests)</p>
</td></tr> <tr><td><code>scores.order</code></td>
<td>
<p>: a 
vector which terms give labels of vectors that explain the larger part of 
variance</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function replaces the former version from the ade4 package, which 
is deprecated. Note that if ade4 is not loaded BEFORE adephylo, then the 
version from ade4 will erase that of adephylo, which will still be
available from adephylo::orthogram. In practice, though, this should never
happen, since ade4 is loaded as a dependence by adephylo.
</p>


<h3>Author(s)</h3>

<p>Original code: Sebastien Ollier and Daniel Chessel.<br />
</p>
<p>Current maintainer: Stephane Dray &lt;stephane.dray@univ-lyon1.fr&gt;
</p>


<h3>References</h3>

<p>Ollier, S., Chessel, D. and Couteron, P. (2005) Orthonormal 
Transform to Decompose the Variance of a Life-History Trait across a 
Phylogenetic Tree. <em>Biometrics</em>, <b>62</b>, 471&ndash;477.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+orthobasis.phylo">orthobasis.phylo</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if(require(ape) &amp;&amp; require(phylobase)){

## a phylogenetic example
data(ungulates)
tre &lt;- read.tree(text=ungulates$tre)
plot(tre)

## look at two traits
afbw &lt;- log(ungulates$tab[,1])
neonatw &lt;- log((ungulates$tab[,2]+ungulates$tab[,3])/2)
names(afbw) &lt;- tre$tip.label
names(neonatw) &lt;- tre$tip.label
plot(afbw, neonatw) # relationship between traits
lm1 &lt;- lm(neonatw~afbw)
resid &lt;- residuals(lm1)
abline(lm1)

## plot the two traits and the residuals of lm1
x &lt;- phylo4d(tre, cbind.data.frame(afbw, neonatw, residuals=resid))
table.phylo4d(x) # residuals are surely not independant

## default orthogram for residuals of lm1
orthogram(resid, tre)

## using another orthonormal basis (derived from Abouheif's proximity)
myOrthoBasis &lt;- orthobasis.phylo(tre, method="oriAbouheif") # Abouheif's proximities
orthogram(resid, ortho=myOrthoBasis) # significant phylog. signal

## Abouheif's test
W &lt;- proxTips(tre, method="oriAbouheif") # proximity matrix
abouheif.moran(resid, W)
}


</code></pre>

<hr>
<h2 id='palm'>Phylogenetic and quantitative traits of amazonian palm trees</h2><span id='topic+palm'></span>

<h3>Description</h3>

<p>This data set describes the phylogeny of 66 amazonian palm trees. It also
gives 7 traits corresponding to these 66 species.
</p>


<h3>Format</h3>

<p><code>palm</code> is a list containing the 2 following objects: </p>

<dl>
<dt>tre</dt><dd><p>is a character string giving the phylogenetic tree in Newick
format.</p>
</dd> <dt>traits</dt><dd><p>is a data frame with 66 species (rows) and 7 traits
(columns).</p>
</dd> </dl>



<h3>Details</h3>

<p>Variables of <code>palm$traits</code> are the following ones: <br /> - rord: specific
richness with five ordered levels<br /> - h: height in meter (squared
transform)<br /> - dqual: diameter at breast height in centimeter with five
levels <code>sout : subterranean</code>, <code> d1(0, 5 cm)</code>, <code> d2(5, 15
cm)</code>, <code> d3(15, 30 cm)</code> and <code> d4(30, 100 cm)</code><br /> - vfruit: fruit
volume in <code class="reqn">mm^{3}</code> (logged transform)<br /> - vgrain: seed volume in
<code class="reqn">mm^{3}</code> (logged transform)<br /> - aire: spatial distribution area
(<code class="reqn">km^{2}</code>)<br /> - alti: maximum altitude in meter (logged
transform)<br />
</p>


<h3>Note</h3>

<p>This dataset replaces the former version in ade4.
</p>


<h3>Source</h3>

<p>This data set was obtained by Clementine Gimaret-Carpentier.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if(require(ape) &amp;&amp; require(phylobase)){

## load data, make a tree and a phylo4d object
data(palm)
tre &lt;- read.tree(text=palm$tre)
rord &lt;- as.integer(palm$traits$rord) # just use this for plotting purpose
traits &lt;- data.frame(rord, palm$traits[,-1])
x &lt;- phylo4d(tre, traits)

## plot data
oldpar &lt;- par(mar=rep(.1,4))
table.phylo4d(x, cex.lab=.6)

## test phylogenetic autocorrelation
if(require(ade4)){
prox &lt;- proxTips(x, method="sumDD")
phylAutoTests &lt;- gearymoran(prox, traits[,-3], nrep=499)
plot(phylAutoTests)
}
par(oldpar)
}


</code></pre>

<hr>
<h2 id='ppca'>Phylogenetic principal component analysis</h2><span id='topic+ppca'></span><span id='topic+print.ppca'></span><span id='topic+summary.ppca'></span><span id='topic+scatter.ppca'></span><span id='topic+screeplot.ppca'></span><span id='topic+plot.ppca'></span>

<h3>Description</h3>

<p>These functions are designed to perform a phylogenetic principal component 
analysis (pPCA, Jombart et al. 2010) and to display the results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ppca(
  x,
  prox = NULL,
  method = c("patristic", "nNodes", "oriAbouheif", "Abouheif", "sumDD"),
  f = function(x) {
     1/x
 },
  center = TRUE,
  scale = TRUE,
  scannf = TRUE,
  nfposi = 1,
  nfnega = 0
)

## S3 method for class 'ppca'
scatter(x, axes = 1:ncol(x$li), useLag = FALSE, ...)

## S3 method for class 'ppca'
print(x, ...)

## S3 method for class 'ppca'
summary(object, ..., printres = TRUE)

## S3 method for class 'ppca'
screeplot(x, ..., main = NULL)

## S3 method for class 'ppca'
plot(x, axes = 1:ncol(x$li), useLag = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ppca_+3A_x">x</code></td>
<td>
<p>a phylo4d object (for <code>ppca</code>) or a ppca object 
(for other methods).</p>
</td></tr>
<tr><td><code id="ppca_+3A_prox">prox</code></td>
<td>
<p>a marix of phylogenetic proximities as returned by 
<code><a href="#topic+proxTips">proxTips</a></code>. If not provided, this matrix will be constructed 
using the arguments <code>method</code> and <code>a</code>.</p>
</td></tr>
<tr><td><code id="ppca_+3A_method">method</code></td>
<td>
<p>a character string (full or abbreviated without ambiguity) 
specifying the method used to compute proximities; possible values are:<br />
- <code>patristic</code>: (inversed sum of) branch lengths <br /> - <code>nNodes</code>: 
(inversed) number of nodes on the path between the nodes <br /> - 
<code>oriAbouheif</code>: original Abouheif's proximity, with diagonal (see 
details in <code><a href="#topic+proxTips">proxTips</a></code>) <br /> - <code>Abouheif</code>: Abouheif's 
proximity (see details in <code><a href="#topic+proxTips">proxTips</a></code>) <br /> - <code>sumDD</code>: 
(inversed) sum of direct descendants of all nodes on the path (see details 
in <code><a href="#topic+proxTips">proxTips</a></code>).</p>
</td></tr>
<tr><td><code id="ppca_+3A_f">f</code></td>
<td>
<p>a function to change a distance into a proximity.</p>
</td></tr>
<tr><td><code id="ppca_+3A_center">center</code></td>
<td>
<p>a logical indicating whether traits should be centred to mean 
zero (TRUE, default) or not (FALSE).</p>
</td></tr>
<tr><td><code id="ppca_+3A_scale">scale</code></td>
<td>
<p>a logical indicating whether traits should be scaled to unit 
variance (TRUE, default) or not (FALSE).</p>
</td></tr>
<tr><td><code id="ppca_+3A_scannf">scannf</code></td>
<td>
<p>a logical stating whether eigenvalues should be chosen 
interactively (TRUE, default) or not (FALSE).</p>
</td></tr>
<tr><td><code id="ppca_+3A_nfposi">nfposi</code></td>
<td>
<p>an integer giving the number of positive eigenvalues retained 
('global structures').</p>
</td></tr>
<tr><td><code id="ppca_+3A_nfnega">nfnega</code></td>
<td>
<p>an integer giving the number of negative eigenvalues retained 
('local structures').</p>
</td></tr>
<tr><td><code id="ppca_+3A_axes">axes</code></td>
<td>
<p>the index of the principal components to be represented.</p>
</td></tr>
<tr><td><code id="ppca_+3A_uselag">useLag</code></td>
<td>
<p>a logical stating whether the lagged components (<code>x\$ls</code>) 
should be used instead of the components (<code>x\$li</code>).</p>
</td></tr>
<tr><td><code id="ppca_+3A_...">...</code></td>
<td>
<p>further arguments passed to other methods. Can be used to 
provide arguments to <code><a href="#topic+table.phylo4d">table.phylo4d</a></code> in <code>plot</code> method.</p>
</td></tr>
<tr><td><code id="ppca_+3A_object">object</code></td>
<td>
<p>a <code>ppca</code> object.</p>
</td></tr>
<tr><td><code id="ppca_+3A_printres">printres</code></td>
<td>
<p>a logical stating whether results should be printed on the 
screen (TRUE, default) or not (FALSE).</p>
</td></tr>
<tr><td><code id="ppca_+3A_main">main</code></td>
<td>
<p>a title for the screeplot; if NULL, a default one is used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>ppca</code> performs the phylogenetic component analysis. Other functions 
are:<br />
</p>
<p>- <code>print.ppca</code>: prints the ppca content<br />
</p>
<p>- <code>summary.ppca</code>: provides useful information about a ppca object, 
including the decomposition of eigenvalues of all axes<br />
</p>
<p>- <code>scatter.ppca</code>: plot principal components using 
<code><a href="#topic+table.phylo4d">table.phylo4d</a></code><br />
</p>
<p>- <code>screeplot.ppca</code>: graphical display of the decomposition of pPCA 
eigenvalues<br />
</p>
<p>- <code>plot.ppca</code>: several graphics describing a ppca object<br />
</p>
<p>The phylogenetic Principal Component Analysis (pPCA, Jombart et al., 2010) is
derived from the spatial Principal Component Analysis (spca, Jombart et al.
2008), implemented in the adegenet package (see 
<code><a href="adegenet.html#topic+spca">spca</a></code>).<br />
</p>
<p>pPCA is designed to investigate phylogenetic patterns a set of quantitative 
traits. The analysis returns principal components maximizing the product of 
variance of the scores and their phylogenetic autocorrelation (Moran's I), 
therefore reflecting life histories that are phylogenetically structured. 
Large positive and large negative eigenvalues correspond to global and local 
structures.<br />
</p>


<h3>Value</h3>

<p>The class <code>ppca</code> are given to lists with the following 
components:<br /> </p>
<table>
<tr><td><code>eig</code></td>
<td>
<p>a numeric vector of eigenvalues.</p>
</td></tr>
<tr><td><code>nfposi</code></td>
<td>
<p>an integer giving the number of global structures retained.</p>
</td></tr>
<tr><td><code>nfnega</code></td>
<td>
<p>an integer giving the number of local structures retained.</p>
</td></tr>
<tr><td><code>c1</code></td>
<td>
<p>a data.frame of loadings of traits for each axis.</p>
</td></tr> <tr><td><code>li</code></td>
<td>
<p>a
data.frame of coordinates of taxa onto the ppca axes (i.e., principal
components).</p>
</td></tr> <tr><td><code>ls</code></td>
<td>
<p>a data.frame of lagged prinpal components; useful
to represent of global scores.</p>
</td></tr> <tr><td><code>as</code></td>
<td>
<p>a data.frame giving the
coordinates of the axes of an 'ordinary' PCA onto the ppca axes.</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>the matched call.</p>
</td></tr> <tr><td><code>tre</code></td>
<td>
<p>a phylogenetic tre with class
phylo4.</p>
</td></tr> <tr><td><code>prox</code></td>
<td>
<p>a matrix of phylogenetic proximities.</p>
</td></tr>
</table>
<p>Other functions have different outputs:<br />
</p>
<p>- <code>scatter.ppca</code> returns the matched call.<br />
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:tjombart@imperial.ac.uk">tjombart@imperial.ac.uk</a>
</p>


<h3>References</h3>

<p>Jombart, T.; Pavoine, S.; Dufour, A. &amp; Pontier, D. (2010, in 
press) Exploring phylogeny as a source of ecological variation: a 
methodological approach. doi:10.1016/j.jtbi.2010.03.038
</p>
<p>Jombart, T., Devillard, S., Dufour, A.-B. and Pontier, D. (2008) Revealing 
cryptic phylogenetic patterns in genetic variability by a new multivariate 
method. <em>Heredity</em>, <b>101</b>, 92&ndash;103.
</p>


<h3>See Also</h3>

<p>The implementation of <code><a href="adegenet.html#topic+spca">spca</a></code> in the adegenet 
package (<code><a href="adegenet.html#topic+adegenet">adegenet</a></code>) <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(lizards)

if(require(ape) &amp;&amp; require(phylobase)){

#### ORIGINAL EXAMPLE FROM JOMBART ET AL 2010 ####


## BUILD A TREE AND A PHYLO4D OBJECT
liz.tre &lt;- read.tree(tex=lizards$hprA)
liz.4d &lt;- phylo4d(liz.tre, lizards$traits)
oldpar &lt;- par(mar=rep(.1,4))
table.phylo4d(liz.4d,var.lab=c(names(lizards$traits),
   "ACP 1\n(\"size effect\")"),show.node=FALSE, cex.lab=1.2)


## REMOVE DUPLICATED POPULATIONS
liz.4d &lt;- prune(liz.4d, c(7,14))
table.phylo4d(liz.4d)


## CORRECT LABELS
lab &lt;- c("Pa", "Ph", "Ll", "Lmca", "Lmcy", "Phha", "Pha",
   "Pb", "Pm", "Ae", "Tt", "Ts", "Lviv", "La", "Ls", "Lvir")
tipLabels(liz.4d) &lt;- lab


## REMOVE SIZE EFFECT
dat &lt;- tdata(liz.4d, type="tip")
dat &lt;- log(dat)
newdat &lt;- data.frame(lapply(dat, function(v) residuals(lm(v~dat$mean.L))))
rownames(newdat) &lt;- rownames(dat)
tdata(liz.4d, type="tip") &lt;- newdat[,-1] # replace data in the phylo4d object


## pPCA
liz.ppca &lt;- ppca(liz.4d,scale=FALSE,scannf=FALSE,nfposi=1,nfnega=1, method="Abouheif")
liz.ppca
tempcol &lt;- rep("grey",7)
tempcol[c(1,7)] &lt;- "black"
barplot(liz.ppca$eig,main='pPCA eigenvalues',cex.main=1.8,col=tempcol)

par(mar=rep(.1,4))
plot(liz.ppca,ratio.tree=.7)


## CONTRIBUTIONS TO PC (LOADINGS) (viewed as dotcharts)
dotchart(liz.ppca$c1[,1],lab=rownames(liz.ppca$c1),main="Global principal
component 1")
abline(v=0,lty=2)

dotchart(liz.ppca$c1[,2],lab=rownames(liz.ppca$c1),main="Local principal
component 1")
abline(v=0,lty=2)


## REPRODUCE FIGURES FROM THE PAPER
obj.ppca &lt;- liz.4d
tdata(obj.ppca, type="tip") &lt;- liz.ppca$li
myLab &lt;- paste(" ",rownames(liz.ppca$li), sep="")

## FIGURE 1
par(mar=c(.1,2.4,2.1,1))
table.phylo4d(obj.ppca, ratio=.7, var.lab=c("1st global PC", "1st local
   PC"), tip.label=myLab,box=FALSE,cex.lab=1.4, cex.sym=1.2, show.node.label=TRUE)
add.scatter.eig(liz.ppca$eig,1,1,1,csub=1.2, posi="topleft", ratio=.23)


## FIGURE 2
s.arrow(liz.ppca$c1,xlim=c(-1,1),clab=1.3,cgrid=1.3)



#### ANOTHER EXAMPLE - INCLUDING NA REPLACEMENT ####
## LOAD THE DATA
data(maples)
tre &lt;- read.tree(text=maples$tre)
x &lt;- phylo4d(tre, maples$tab)
omar &lt;- par("mar")
par(mar=rep(.1,4))
table.phylo4d(x, cex.lab=.5, cex.sym=.6, ratio=.1) # note NAs in last trait ('x')

## FUNCTION TO REPLACE NAS
f1 &lt;- function(vec){
if(any(is.na(vec))){
m &lt;- mean(vec, na.rm=TRUE)
vec[is.na(vec)] &lt;- m
}
return(vec)
}


## PERFORM THE PPCA
dat &lt;- apply(maples$tab,2,f1) # replace NAs
x.noNA &lt;- phylo4d(tre, as.data.frame(dat))
map.ppca &lt;- ppca(x.noNA, scannf=FALSE, method="Abouheif")
map.ppca


## SOME GRAPHICS
screeplot(map.ppca)
scatter(map.ppca, useLag=TRUE)
plot(map.ppca, useLag=TRUE)


## MOST STRUCTURED TRAITS
a &lt;- map.ppca$c1[,1] # loadings on PC 1
names(a) &lt;- row.names(map.ppca$c1)
highContrib &lt;- a[a&lt; quantile(a,0.1) | a&gt;quantile(a,0.9)]
datSel &lt;- cbind.data.frame(dat[, names(highContrib)], map.ppca$li)
temp &lt;- phylo4d(tre, datSel)
table.phylo4d(temp) # plot of most structured traits


## PHYLOGENETIC AUTOCORRELATION TESTS FOR THESE TRAITS
prox &lt;- proxTips(tre, method="Abouheif")
abouheif.moran(dat[, names(highContrib)], prox)
par(oldpar)

}

</code></pre>

<hr>
<h2 id='procella'>Phylogeny and quantitative traits of birds</h2><span id='topic+procella'></span>

<h3>Description</h3>

<p>This data set describes the phylogeny of 19 birds as reported by Bried et
al. (2002). It also gives 6 traits corresponding to these 19 species.
</p>


<h3>Format</h3>

<p><code>procella</code> is a list containing the 2 following objects:
</p>
 <dl>
<dt>tre</dt><dd><p>is a character string giving the phylogenetic tree in
Newick format.</p>
</dd> <dt>traits</dt><dd><p>is a data frame with 19 species and 6 traits</p>
</dd>
</dl>



<h3>Details</h3>

<p>Variables of <code>procella$traits</code> are the following ones: <br /> - site.fid:
a numeric vector that describes the percentage of site fidelity<br /> -
mate.fid: a numeric vector that describes the percentage of mate fidelity<br />
- mass: an integer vector that describes the adult body weight (g)<br /> - ALE:
a numeric vector that describes the adult life expectancy (years)<br /> - BF: a
numeric vector that describes the breeding frequencies<br /> - col.size: an
integer vector that describes the colony size (no nests monitored)
</p>


<h3>Note</h3>

<p>This dataset replaces the former version in ade4.
</p>


<h3>References</h3>

<p>Bried, J., Pontier, D. and Jouventin, P. (2002) Mate fidelity in
monogamus birds: a re-examination of the Procellariiformes. <em>Animal
Behaviour</em>, <b>65</b>, 235&ndash;246.
</p>
<p>See a data description at <a href="http://pbil.univ-lyon1.fr/R/pdf/pps037.pdf">http://pbil.univ-lyon1.fr/R/pdf/pps037.pdf</a>
(in French).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if(require(ape) &amp;&amp; require(phylobase)){

## load data, make tree and phylo4d object
data(procella)
tre &lt;- read.tree(text=procella$tre)
x &lt;- phylo4d(tre, procella$traits)
oldpar &lt;- par(mar=rep(.1,4))
table.phylo4d(x,cex.lab=.7)
par(oldpar)
}


</code></pre>

<hr>
<h2 id='proxTips'>Compute some phylogenetic proximities between tips</h2><span id='topic+proxTips'></span>

<h3>Description</h3>

<p>The function <code>proxTips</code> computes a given proximity between a set of
tips of a phylogeny. A vector of tips is supplied: proximities between all
possible pairs of these tips are computed.  The proximities are computed
from the shortest path between the tips. <br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>proxTips(
  x,
  tips = "all",
  method = c("patristic", "nNodes", "oriAbouheif", "Abouheif", "sumDD"),
  f = function(x) {
     1/x
 },
  normalize = c("row", "col", "none"),
  symmetric = TRUE,
  useC = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="proxTips_+3A_x">x</code></td>
<td>
<p>a tree of class <code><a href="ape.html#topic+read.tree">phylo</a></code>,
phylo4 or phylo4d.</p>
</td></tr>
<tr><td><code id="proxTips_+3A_tips">tips</code></td>
<td>
<p>A vector of integers identifying tips by their numbers, or a
vector of characters identifying tips by their names. Distances will be
computed between all possible pairs of tips.</p>
</td></tr>
<tr><td><code id="proxTips_+3A_method">method</code></td>
<td>
<p>a character string (full or abbreviated without ambiguity)
specifying the method used to compute proximities; possible values are:<br /> -
<code>patristic</code>: (inversed sum of) branch length <br /> - <code>nNodes</code>:
(inversed) number of nodes on the path between the nodes <br /> -
<code>oriAbouheif</code>: original Abouheif's proximity, with diagonal (see
details) <br /> - <code>Abouheif</code>: Abouheif's proximity without diagonal (see
details) <br /> - <code>sumDD</code>: (inversed) sum of direct descendants of all
nodes on the path (see details) <br /></p>
</td></tr>
<tr><td><code id="proxTips_+3A_f">f</code></td>
<td>
<p>a function to change a distance into a proximity.</p>
</td></tr>
<tr><td><code id="proxTips_+3A_normalize">normalize</code></td>
<td>
<p>a character string specifying whether the matrix must be
normalized by row (<code>row</code>), column (<code>col</code>), or not (<code>none</code>).
Normalization amounts to dividing each row (or column) so that the marginal
sum is 1. Hence, default is matrix with each row summing to 1.</p>
</td></tr>
<tr><td><code id="proxTips_+3A_symmetric">symmetric</code></td>
<td>
<p>a logical stating whether M must be coerced to be symmetric
(TRUE, default) or not. This is achieved by taking (denoting N the matrix of
proximities before re-symmetrization): </p>
<p style="text-align: center;"><code class="reqn">M = 0.5 * (N + N^{T})</code>
</p>
<p> Note
that <code class="reqn">x^{T}Ny = x^{T}My</code>, but the latter has the advantage of using a
bilinear symmetric form (more appropriate for optimization purposes).</p>
</td></tr>
<tr><td><code id="proxTips_+3A_usec">useC</code></td>
<td>
<p>a logical indicating whether computations of distances (before
transformation into proximities) should be performed using compiled C code
(TRUE, default), or using a pure R version (FALSE). C version is several
orders of magnitude faster, and R version is kept for backward
compatibility.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Proximities are computed as the inverse (to the power <code>a</code>) of a
phylogenetic distance (computed by <code><a href="#topic+distTips">distTips</a></code>. Denoting
<code class="reqn">D=[d_{ij}]</code> a matrix of phylogenetic distances, the proximity matrix
<code class="reqn">M=[m_{ij}]</code> is computed as: </p>
<p style="text-align: center;"><code class="reqn">m_{ij} = \frac{1}{d_{ij}^a} \forall i
\neq j</code>
</p>
<p> and </p>
<p style="text-align: center;"><code class="reqn">m_{ii}
= 0</code>
</p>

<p>Several distances can be used, defaulting to the sum of branch lengths (see
argument <code>method</code>).  Proximities are not true similarity measures,
since the proximity of a tip with itself is always set to zero.<br />
</p>
<p>The obtained matrix of phylogenetic proximities (M) defines a bilinear
symmetric form when M is symmetric (default):<br /> </p>
<p style="text-align: center;"><code class="reqn">f(x,y) = x^{T}My</code>
</p>

<p>In general, M is not a metric because it is not positive-definite.  Such a
matrice can be used to measure phylogenetic autocorrelation (using Moran's
index): </p>
<p style="text-align: center;"><code class="reqn">I(x) = \frac{x^TMx}{var(x)}</code>
</p>

<p>or to compute lag vectors (Mx) used in autoregressive models, like: </p>
<p style="text-align: center;"><code class="reqn">x
= Mx + ... + e</code>
</p>
<p> where '...' is the non-autoregressive part of the model, and
'e' are residuals.
</p>
<p><code>Abouheif</code> proximity refers to the phylogenetic proximity underlying
the test of Abouheif (see references). Let P be the set of all the nodes in
the path going from <code>node1</code> to <code>node2</code>. Let DDP be the number of
direct descendants from each node in P. Then, the so-called 'Abouheif'
distance is the inverse of the product of all terms in DDP.
<code>oriAbouheif</code> returns a matrix with non-null diagonal elements, as
formulated in Pavoine <em>et al.</em> (2008). This matrix is bistochastic (all
marginal sums equal 1), but this bilinear symmetric form does not give rise
to a Moran's index, since it requires a null diagonal. <code>Abouheif</code>
contains Abouheif's proximities but has a null diagonal, giving rise to a
Moran's index.<br />
</p>
<p><code>sumDD</code> refers to a phylogenetic proximity quite similar to that of
Abouheif. We consider the same sets P and DDP. But instead of taking the
inverse of the product of all terms in DDP, this proximity computes the
inverse of the sum of all terms in DDP. This matrix was denoted 'M' in
Pavoine <em>et al.</em> (2008), who reported that it is related to May's index
(May, 1990).
</p>


<h3>Value</h3>

<p>A matrix of phylogenetic proximities.
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:tjombart@imperial.ac.uk">tjombart@imperial.ac.uk</a>
</p>


<h3>References</h3>

<p>== About Moran's index with various proximities == <br /> Pavoine,
S.; Ollier, S.; Pontier, D.; Chessel, D. (2008) Testing for phylogenetic
signal in life history variable: Abouheif's test revisited.
<em>Theoretical Population Biology</em>: <b>73</b>, 79-91.<br />
</p>
<p>== About regression on phylogenetic lag vector == <br /> Cheverud, J. M.; Dow,
M. M.; Leutenegger, W. (1985) The quantitative assessment of phylogentic
constaints in comparative analyses: sexual dimorphism in body weights among
primates. <em>Evolution</em> <b>39</b>, 1335-1351.<br />
</p>
<p>Cheverud, J. M.; Dow, M. M. (1985) An autocorrelation analysis of genetic
variation due to lineal fission in social groups of Rhesus macaques.
<em>American Journal of Phyisical Anthropology</em> <b>67</b>, 113-121.<br />
</p>
<p>== Abouheif's original paper ==<br /> Abouheif, E. (1999) A method for testing
the assumption of phylogenetic independence in comparative data.
<em>Evolutionary Ecology Research</em>, <b>1</b>, 895-909.<br />
</p>
<p>== May's index ==<br /> May, R.M. (1990) Taxonomy as destiny. <em>Nature</em>
<b>347</b>, 129-130.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+distTips">distTips</a></code> which computes several phylogenetic
distances between tips.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(require(ape) &amp; require(phylobase)){
## make a tree
x &lt;- as(rtree(10),"phylo4")
plot(x, show.node=TRUE)
axisPhylo()
## compute different distances
proxTips(x, 1:5)
proxTips(x, 1:5, "nNodes")
proxTips(x, 1:5, "Abouheif")
proxTips(x, , "sumDD")

## see what one proximity looks like
M &lt;- proxTips(x)
obj &lt;- phylo4d(x,as.data.frame(M))
table.phylo4d(obj,symbol="sq")
}

</code></pre>

<hr>
<h2 id='sp.tips'>Find the shortest path between tips of a tree</h2><span id='topic+sp.tips'></span>

<h3>Description</h3>

<p>The function <code>sp.tips</code> finds the shortest path between tips of a tree,
identified as <code>tip1</code> and <code>tip2</code>.  This function applies to trees
with the class <code><a href="ape.html#topic+read.tree">phylo</a></code>, phylo4 or
phylo4d. Several tips can be provided at a time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sp.tips(x, tip1, tip2, useTipNames = FALSE, quiet = FALSE, include.mrca = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sp.tips_+3A_x">x</code></td>
<td>
<p>A tree of class <code><a href="ape.html#topic+read.tree">phylo</a></code>,
phylo4 or phylo4d.</p>
</td></tr>
<tr><td><code id="sp.tips_+3A_tip1">tip1</code></td>
<td>
<p>A vector of integers identifying tips by their numbers, or a
vector of characters identifying tips by their names. Recycled if needed.</p>
</td></tr>
<tr><td><code id="sp.tips_+3A_tip2">tip2</code></td>
<td>
<p>A vector of integers identifying tips by their numbers, or a
vector of characters identifying tips by their names. Recycled if needed.</p>
</td></tr>
<tr><td><code id="sp.tips_+3A_usetipnames">useTipNames</code></td>
<td>
<p>a logical stating whether the output must be named using
tip names in all cases (TRUE), or not (FALSE). If not, names of <code>tip1</code>
and <code>tip2</code> will be used.</p>
</td></tr>
<tr><td><code id="sp.tips_+3A_quiet">quiet</code></td>
<td>
<p>a logical stating whether a warning must be issued when
tip1==tip2, or not (see details).</p>
</td></tr>
<tr><td><code id="sp.tips_+3A_include.mrca">include.mrca</code></td>
<td>
<p>a logical stating whether the most recent common
ancestor shall be included in the returned path (TRUE, default) or not
(FALSE).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function checks if there are cases where tip1 and tip2 are the same.
These cases are deleted when detected, issuing a warning (unless
<code>quiet</code> is set to TRUE).
</p>


<h3>Value</h3>

<p>A list whose components are vectors of named nodes forming the
shortest path between a couple of tips.
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:tjombart@imperial.ac.uk">tjombart@imperial.ac.uk</a>
</p>


<h3>See Also</h3>

<p><code><a href="phylobase.html#topic+shortestPath">shortestPath</a></code> which does the same thing as
<code>sp.tips</code>, for any node (internal or tip), but much more slowly. <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if(require(ape) &amp; require(phylobase)){
## make a tree
x &lt;- as(rtree(20),"phylo4")
plot(x,show.node=TRUE)
## get shortest path between tip 1 and all other tips.
sp.tips(x, "t1", "t2")
sp.tips(x, 1, 2:20, TRUE)

}

</code></pre>

<hr>
<h2 id='table.phylo4d'>Graphical display of phylogeny and traits</h2><span id='topic+table.phylo4d'></span>

<h3>Description</h3>

<p>This function represents traits onto the tips of a phylogeny. Plotted objects
must be valid phylo4d objects (implemented by the 
<code>phylobase</code> package). Current version allows plotting of a tree and one 
or more quantitative traits (possibly containing missing data, represented by
an 'x').<br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>table.phylo4d(
  x,
  treetype = c("phylogram", "cladogram"),
  symbol = c("circles", "squares", "colors"),
  repVar = 1:ncol(tdata(x, type = "tip")),
  center = TRUE,
  scale = TRUE,
  legend = TRUE,
  grid = TRUE,
  box = TRUE,
  show.tip.label = TRUE,
  show.node.label = TRUE,
  show.var.label = TRUE,
  ratio.tree = 1/3,
  font = 3,
  tip.label = tipLabels(x),
  var.label = colnames(tdata(x, type = "tip")),
  cex.symbol = 1,
  cex.label = 1,
  cex.legend = 1,
  pch = 20,
  col = heat.colors(100),
  coord.legend = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="table.phylo4d_+3A_x">x</code></td>
<td>
<p>a phylo4d object</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_treetype">treetype</code></td>
<td>
<p>the type of tree to be plotted (&quot;phylogram&quot; or &quot;cladogram&quot;)</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_symbol">symbol</code></td>
<td>
<p>the type of symbol used to represent data (&quot;circles&quot;, 
&quot;squares&quot;, or &quot;colors&quot;)</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_repvar">repVar</code></td>
<td>
<p>the numerical index of variables to be plotted</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_center">center</code></td>
<td>
<p>a logical stating whether variables should be centred (TRUE, 
default) or not (FALSE)</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_scale">scale</code></td>
<td>
<p>a logical stating whether variables should be scaled (TRUE, 
default) or not (FALSE)</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_legend">legend</code></td>
<td>
<p>a logical stating whether a legend should be added to the plot 
(TRUE) or not (FALSE, default)</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_grid">grid</code></td>
<td>
<p>a logical stating whether a grid should be added to the plot 
(TRUE, default) or not (FALSE)</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_box">box</code></td>
<td>
<p>a logical stating whether a box should be added around the plot 
(TRUE, default) or not (FALSE)</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_show.tip.label">show.tip.label</code></td>
<td>
<p>a logical stating whether tip labels should be printed 
(TRUE, default) or not (FALSE)</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_show.node.label">show.node.label</code></td>
<td>
<p>a logical stating whether node labels should be 
printed (TRUE, default) or not (FALSE)</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_show.var.label">show.var.label</code></td>
<td>
<p>a logical stating whether labels of variables should be
printed (TRUE, default) or not (FALSE)</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_ratio.tree">ratio.tree</code></td>
<td>
<p>the proportion of width of the figure occupied by the tree</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_font">font</code></td>
<td>
<p>an integer specifying the type of font for the labels: 1 (plain 
text), 2 (bold), 3 (italic, default), or 4 (bold italic).</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_tip.label">tip.label</code></td>
<td>
<p>a character vector giving the tip labels</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_var.label">var.label</code></td>
<td>
<p>a character vector giving the labels of variables</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_cex.symbol">cex.symbol</code></td>
<td>
<p>a numeric giving the factor scaling the symbols</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_cex.label">cex.label</code></td>
<td>
<p>a numeric giving the factor scaling all labels</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_cex.legend">cex.legend</code></td>
<td>
<p>a numeric giving the factor scaling the legend</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_pch">pch</code></td>
<td>
<p>is <code>symbol</code> is set to 'colors', a number indicating the type 
of point to be plotted (see ?points)</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_col">col</code></td>
<td>
<p>is <code>symbol</code> is set to 'colors', a vector of colors to be used
to represent the data</p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_coord.legend">coord.legend</code></td>
<td>
<p>an optional list with two components 'x' and 'y' 
indicating the lower-left position of the legend. Can be set to 
<code>locator(1) to position the legend interactively.</code></p>
</td></tr>
<tr><td><code id="table.phylo4d_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to plot methods from <code>ape</code>. 
See <code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot of phylogenies is performed by a call to 
<code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code> from the <code>ape</code> package. Hence, many of the
arguments of <code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code> can be passed to 
<code>table.phylo4d</code>, through the ... argument, but their names must be 
complete.
</p>
<p>For large trees, consider using <code><a href="#topic+bullseye">bullseye</a></code>.
</p>
<p>The function <code>table.phylo4d</code> is based on former plot method for 
phylo4d objects from the <code>phylobase</code> package.  It replaces
the deprecated <code>ade4</code> functions <code><a href="ade4.html#topic+symbols.phylog">symbols.phylog</a></code> and
<code><a href="ade4.html#topic+table.phylog">table.phylog</a></code>.
</p>


<h3>Value</h3>

<p>No return value, function produces only a plot.
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:tjombart@imperial.ac.uk">tjombart@imperial.ac.uk</a>
</p>


<h3>See Also</h3>

<p>The phylo4d class for storing 
<code>phylogeny+data</code>.<br />
</p>
<p>For large trees, consider using <code><a href="#topic+bullseye">bullseye</a></code>.
</p>
<p><code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code> from the <code>ape</code> package.<br />
</p>
<p>An alternative (deprecated) representation is available from 
<code><a href="ade4.html#topic+dotchart.phylog">dotchart.phylog</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(require(ape) &amp; require(phylobase) &amp; require(ade4)){

## simulated data
tr &lt;- rtree(20)
dat &lt;- data.frame(a = rnorm(20), b = scale(1:20), c=runif(20,-2,2) )
dat[3:6, 2] &lt;- NA # introduce some NAs
obj &lt;- phylo4d(tr, dat) # build a phylo4d object
table.phylo4d(obj) # default scatterplot
table.phylo4d(obj,cex.leg=.6, use.edge.length=FALSE) # customized
table.phylo4d(obj,treetype="clad", show.node=FALSE, cex.leg=.6,
use.edge.length=FALSE, edge.color="blue", edge.width=3) # more customized


## teleost fishes data
data(mjrochet)
temp &lt;- read.tree(text=mjrochet$tre) # make a tree
mjr &lt;- phylo4d(x=temp,tip.data=mjrochet$tab) # male a phylo4d object
table.phylo4d(mjr,cex.lab=.5,show.node=FALSE,symb="square")


## lizards data
data(lizards)
liz.tr &lt;- read.tree(tex=lizards$hprA) # make a tree
liz &lt;- phylo4d(liz.tr, lizards$traits) # make a phylo4d object
table.phylo4d(liz)


## plotting principal components
liz.pca1 &lt;- dudi.pca(lizards$traits, scannf=FALSE, nf=2) # PCA of traits
myPC &lt;- phylo4d(liz.tr, liz.pca1$li) # store PC in a phylo4d object
varlab &lt;- paste("Principal \ncomponent", 1:2) # make labels for PCs
table.phylo4d(myPC, ratio=.8, var.lab=varlab) # plot the PCs
add.scatter.eig(liz.pca1$eig,2,1,2,posi="topleft", inset=c(0,.15))
title("Phylogeny and the principal components")

}

</code></pre>

<hr>
<h2 id='tithonia'>Phylogeny and quantitative traits of flowers</h2><span id='topic+tithonia'></span>

<h3>Description</h3>

<p>This data set describes the phylogeny of 11 flowers as reported by Morales
(2000). It also gives morphologic and demographic traits corresponding to
these 11 species.
</p>


<h3>Format</h3>

<p><code>tithonia</code> is a list containing the 2 following objects :
</p>
 <dl>
<dt>tre</dt><dd><p>is a character string giving the phylogenetic tree in
Newick format.</p>
</dd> <dt>tab</dt><dd><p>is a data frame with 11 species and 14 traits (6
morphologic traits and 8 demographic).</p>
</dd> </dl>



<h3>Details</h3>

<p>Variables of <code>tithonia$tab</code> are the following ones : <br /> morho1: is a
numeric vector that describes the seed size (mm)<br /> morho2: is a numeric
vector that describes the flower size (mm)<br /> morho3: is a numeric vector
that describes the female leaf size (cm)<br /> morho4: is a numeric vector that
describes the head size (mm)<br /> morho5: is a integer vector that describes
the number of flowers per head <br /> morho6: is a integer vector that
describes the number of seeds per head <br /> demo7: is a numeric vector that
describes the seedling height (cm)<br /> demo8: is a numeric vector that
describes the growth rate (cm/day)<br /> demo9: is a numeric vector that
describes the germination time<br /> demo10: is a numeric vector that describes
the establishment (per cent)<br /> demo11: is a numeric vector that describes
the viability (per cent)<br /> demo12: is a numeric vector that describes the
germination (per cent)<br /> demo13: is a integer vector that describes the
resource allocation<br /> demo14: is a numeric vector that describes the adult
height (m)<br />
</p>


<h3>Note</h3>

<p>This dataset replaces the former version in ade4.
</p>


<h3>Source</h3>

<p>Data were obtained from Morales, E. (2000) Estimating phylogenetic
inertia in Tithonia (Asteraceae) : a comparative approach. <em>Evolution</em>,
<b>54</b>, 2, 475&ndash;484.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if(require(ape) &amp;&amp; require(phylobase)){

data(tithonia)
tre &lt;- read.tree(text=tithonia$tre)
traits &lt;- log(tithonia$tab + 1)
rownames(traits) &lt;- gsub("_", ".", rownames(traits))

## build a phylo4d object
x &lt;- phylo4d(tre, traits)
oldpar &lt;- par(mar=rep(.1,4))
table.phylo4d(x)
par(oldpar)
}


</code></pre>

<hr>
<h2 id='treePart'>Define partitions of tips according from a tree</h2><span id='topic+treePart'></span>

<h3>Description</h3>

<p>The function <code>treePart</code> defines partitions of tips reflecting the
topology of a tree. There are two possible outputs (handled by the argument
<code>result</code>):<br /> - <code>basis</code> mode: each node but the root is translated
into a dummy vector having one value for each tip: this value is '1' if the
tip descends from this node, and '0' otherwise.<br /> - <code>orthobasis</code>: in
this mode, an orthonormal basis is derived from the basis previously
mentionned. This orthobasis was proposed in the orthogram (Ollier <em>et
al.</em> 2006).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>treePart(x, result = c("dummy", "orthobasis"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="treePart_+3A_x">x</code></td>
<td>
<p>a tree of class <code><a href="ape.html#topic+read.tree">phylo</a></code>,
phylo4 or phylo4d.</p>
</td></tr>
<tr><td><code id="treePart_+3A_result">result</code></td>
<td>
<p>a character string specifying the type of result: either a
basis of dummy vectors (<code>dummy</code>), or an orthobasis derived from these
dummy vectors (<code>orthobasis</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Orthobasis produced by this function are identical to those stored in the
Bscores component of deprecated <a href="ade4.html#topic+phylog">phylog</a> objects, from the ade4
package.
</p>


<h3>Value</h3>

<p>A matrix of numeric vectors (in columns) having one value for each
tip (rows).
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:tjombart@imperial.ac.uk">tjombart@imperial.ac.uk</a>
</p>


<h3>References</h3>

<p>Ollier, S., Chessel, D. and Couteron, P. (2005) Orthonormal
Transform to Decompose the Variance of a Life-History Trait across a
Phylogenetic Tree. <em>Biometrics</em>, <b>62</b>, 471&ndash;477.
</p>


<h3>See Also</h3>

<p>- <code><a href="#topic+listDD">listDD</a></code> which is called by <code>treePart</code>.<br /> -
<code><a href="#topic+orthogram">orthogram</a></code>, which uses by default the orthobasis produced by
<code>treePart</code>.<br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


if(require(ape) &amp; require(phylobase)){
## make a tree
x &lt;- as(rtree(10),"phylo4")
partition &lt;- treePart(x)
partition

## plot the dummy vectors with the tree
temp &lt;- phylo4d(x, partition)
table.phylo4d(temp, cent=FALSE, scale=FALSE)
}


</code></pre>

<hr>
<h2 id='ungulates'>Phylogeny and quantitative traits of ungulates.</h2><span id='topic+ungulates'></span>

<h3>Description</h3>

<p>This data set describes the phylogeny of 18 ungulates as reported by
Pelabon et al. (1995). It also gives 4 traits corresponding to these 18
species.
</p>


<h3>Format</h3>

<p><code>fission</code> is a list containing the 2 following objects :
</p>
 <dl>
<dt>tre</dt><dd><p>is a character string giving the phylogenetic tree in
Newick format.</p>
</dd> <dt>tab</dt><dd><p>is a data frame with 18 species and 4 traits</p>
</dd> </dl>



<h3>Details</h3>

<p>Variables of <code>ungulates$tab</code> are the following ones : <br />
</p>
<p>- afbw: is a numeric vector that describes the adult female body weight (g)
<br /> - mnw: is a numeric vector that describes the male neonatal weight (g)
<br /> - fnw: is a numeric vector that describes the female neonatal weight (g)
<br /> - ls: is a numeric vector that describes the litter size <br />
</p>


<h3>Note</h3>

<p>This dataset replaces the former version in ade4.
</p>


<h3>Source</h3>

<p>Data were obtained from Pelabon, C., Gaillard, J.M., Loison, A. and
Portier, A. (1995) Is sex-biased maternal care limited by total maternal
expenditure in polygynous ungulates?  <em>Behavioral Ecology and
Sociobiology</em>, <b>37</b>, 311&ndash;319.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if(require(ape) &amp;&amp; require(phylobase)){
## load data
data(ungulates)
tre &lt;- read.tree(text=ungulates$tre)
plot(tre)

## look at two traits
afbw &lt;- log(ungulates$tab[,1])
neonatw &lt;- log((ungulates$tab[,2]+ungulates$tab[,3])/2)
names(afbw) &lt;- tre$tip.label
names(neonatw) &lt;- tre$tip.label
plot(afbw, neonatw) # relationship between traits
lm1 &lt;- lm(neonatw~afbw)
abline(lm1)
x &lt;- phylo4d(tre, cbind.data.frame(afbw, neonatw)) # traits on the phylogeny

## test phylogenetic inertia in residuals
orthogram(residuals(lm1), x) 
}


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
