<!DOCTYPE html><html lang="en-US"><head><title>Help for package adjclust</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {adjclust}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#adjClust'><p>Adjacency-constrained Clustering</p></a></li>
<li><a href='#chac'><p>Class chac</p></a></li>
<li><a href='#hicClust'><p>Adjacency-constrained Clustering of Hi-C contact maps</p></a></li>
<li><a href='#plotSim'><p>Plot (dis)similarity matrix</p></a></li>
<li><a href='#select'><p>Clustering selection</p></a></li>
<li><a href='#snpClust'><p>Adjacency-constrained Clustering of Single Nucleotide Polymorphisms</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Maintainer:</td>
<td>Pierre Neuvial &lt;pierre.neuvial@math.univ-toulouse.fr&gt;</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-10-07</td>
</tr>
<tr>
<td>Version:</td>
<td>0.6.10</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Title:</td>
<td>Adjacency-Constrained Clustering of a Block-Diagonal Similarity
Matrix</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements a constrained version of hierarchical agglomerative 
    clustering, in which each observation is associated to a position, and only 
    adjacent clusters can be merged. Typical application fields in 
    bioinformatics include Genome-Wide Association Studies or Hi-C data 
    analysis, where the similarity between items is a decreasing function of 
    their genomic distance. Taking advantage of this feature, the implemented 
    algorithm is time and memory efficient. This algorithm is described in 
    Ambroise et al (2019) &lt;<a href="https://doi.org/10.1186%2Fs13015-019-0157-4">doi:10.1186/s13015-019-0157-4</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, graphics, grDevices, Rcpp (&ge; 1.0.6), Matrix,
sparseMatrixStats, methods, utils, capushe, ggplot2,
dendextend, rlang</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, testthat, rmarkdown, rioja, HiTC, snpStats,
BiocGenerics</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Clustering, FeatureExtraction</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://pneuvial.github.io/adjclust/">https://pneuvial.github.io/adjclust/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/pneuvial/adjclust/issues">https://github.com/pneuvial/adjclust/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-10-07 13:21:31 UTC; nathalie</td>
</tr>
<tr>
<td>Author:</td>
<td>Christophe Ambroise [aut],
  Shubham Chaturvedi [aut],
  Alia Dehman [aut],
  Pierre Neuvial <a href="https://orcid.org/0000-0003-3584-9998"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Guillem Rigaill [aut],
  Nathalie Vialaneix
    <a href="https://orcid.org/0000-0003-1156-0639"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Gabriel Hoffman [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-10-08 08:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='adjClust'>Adjacency-constrained Clustering</h2><span id='topic+adjClust'></span>

<h3>Description</h3>

<p>Adjacency-constrained hierarchical agglomerative clustering
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjClust(
  mat,
  type = c("similarity", "dissimilarity"),
  h = ncol(mat) - 1,
  strictCheck = TRUE,
  nthreads = 1L
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="adjClust_+3A_mat">mat</code></td>
<td>
<p>A similarity matrix or a dist object. Most sparse formats from
<code><a href="Matrix.html#topic+sparseMatrix">sparseMatrix</a></code> are allowed</p>
</td></tr>
<tr><td><code id="adjClust_+3A_type">type</code></td>
<td>
<p>Type of matrix : similarity or dissimilarity. Defaults to
<code>"similarity"</code></p>
</td></tr>
<tr><td><code id="adjClust_+3A_h">h</code></td>
<td>
<p>band width. It is assumed that the similarity between two items is 0
when these items are at a distance of more than band width h. Default value
is <code>ncol(mat)-1</code></p>
</td></tr>
<tr><td><code id="adjClust_+3A_strictcheck">strictCheck</code></td>
<td>
<p>Logical (default to <code>TRUE</code>) to systematically check
default of positivity in input similarities. Can be disabled to avoid
computationally expensive checks when the number of features is large.</p>
</td></tr>
<tr><td><code id="adjClust_+3A_nthreads">nthreads</code></td>
<td>
<p>Integer (default to <code>1L</code>). Number of threads use for
matrix precomputations.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Adjacency-constrained hierarchical agglomerative clustering (HAC) is HAC in
which each observation is associated to a position, and the clustering is
constrained so as only adjacent clusters are merged. These methods are useful
in various application fields, including ecology (Quaternary data) and
bioinformatics (e.g., in Genome-Wide Association Studies (GWAS)).
</p>
<p>This function is a fast implementation of the method that takes advantage of
sparse similarity matrices (i.e., that have 0 entries outside of a diagonal
band of width <code>h</code>). The method is fully described in (Dehman, 2015) and
based on a kernel version of the algorithm. The different options for the
implementation are available in the package vignette entitled 
<a href="https://pneuvial.github.io/adjclust/articles/notesCHAC.html">&quot;Notes on CHAC implementation in adjclust&quot;</a>.
</p>


<h3>Value</h3>

<p>An object of class <code><a href="#topic+chac">chac</a></code> which describes the tree
produced by the clustering process. The object is a list with the same
elements as an object of class <code><a href="stats.html#topic+hclust">hclust</a></code> (<code>merge</code>,
<code>height</code>, <code>order</code>, <code>labels</code>, <code>call</code>, <code>method</code>,
<code>dist.method</code>), and two extra elements: 
</p>
<table role = "presentation">
<tr><td><code>mat</code></td>
<td>
<p>: (the data on which the clustering has been performed, 
possibly after the pre-transformations described in the vignette entitled
<a href="https://pneuvial.github.io/adjclust/articles/notesCHAC.html#notes-on-relations-between-similarity-and-dissimilarity-implementation">&quot;Notes on CHAC implementation in adjclust&quot;</a></p>
</td></tr></table>
<p>.
</p>
<table role = "presentation">
<tr><td><code>correction</code></td>
<td>
<p>: the value of the correction for non positive
definite similarity matrices (also described in the same vignette). If
<code>correction == 0</code>, it means that the initial data were not 
pre-transformed.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>When performed on a distance matrix <code class="reqn">d</code> with the option 
<code>type = "dissimilarity"</code>, <code>adjclust</code> is identical to using the 
option <code>"ward.D"</code> on <code class="reqn">d^2</code> in the function 
<code><a href="stats.html#topic+hclust">hclust</a></code> when the ordering of the (unconstrained) 
clustering (in <code><a href="stats.html#topic+hclust">hclust</a></code>) is compatible with the natural 
ordering of objects used as a constraint. It is also equivalent (under the
same assumption or orderings) to the option <code>"ward.D2"</code> performed on the
distance matrix <code class="reqn">d</code> itself, except for the final heights of the merges 
that are equal to the square of the heights obtained with <code>"ward.D2"</code> in
<code><a href="stats.html#topic+hclust">hclust</a></code>. See the 
<a href="https://pneuvial.github.io/adjclust/articles/notesCHAC.html#relations-with-hclust-and-rioja">vignette on implementation</a> 
and (Murtagh and Legendre, 2014) for further details.
</p>


<h3>References</h3>

<p>Murtagh F., and Legendre P. (2014). Ward's hierarchical
agglomerative clustering method: which algorithms implement Ward's 
criterion? <em>Journal of Classification</em>, <strong>31</strong>, 274-295. 
DOI: <a href="https://doi.org/10.1007/s00357-014-9161-z">doi:10.1007/s00357-014-9161-z</a>.
</p>
<p>Dehman A. (2015). <em>Spatial Clustering of Linkage
Disequilibrium Blocks for Genome-Wide Association Studies</em>, PhD thesis,
Universite Paris Saclay, France.
</p>
<p>Ambroise C., Dehman A., Neuvial P., Rigaill G., and Vialaneix N
(2019). Adjacency-constrained hierarchical clustering of a band similarity 
matrix with application to genomics. <em>Algorithms for Molecular
Biology</em>, <strong>14</strong>(22).
DOI: <a href="https://doi.org/10.1007/s11222-018-9806-6">doi:10.1007/s11222-018-9806-6</a>.
</p>
<p>Randriamihamison N., Vialaneix N., and Neuvial P. (2020).
Applicability and interpretability of Ward's hierarchical agglomerative
clustering with or without contiguity constraints. <em>Journal of 
Classification</em>, <strong>38</strong>, 1-27.
DOI: <a href="https://doi.org/10.1007/s00357-020-09377-y">doi:10.1007/s00357-020-09377-y</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+snpClust">snpClust</a></code> to cluster SNPs based on linkage
disequilibrium
</p>
<p><code><a href="#topic+hicClust">hicClust</a></code> to cluster Hi-C data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- matrix(
  c(1.0, 0.1, 0.2, 0.3,
    0.1, 1.0 ,0.4 ,0.5,
    0.2, 0.4, 1.0, 0.6,
    0.3, 0.5, 0.6, 1.0), 
 nrow = 4)

## similarity, full width
fit1 &lt;- adjClust(sim, "similarity")
plot(fit1)

## similarity, h &lt; p-1
fit2 &lt;- adjClust(sim, "similarity", h = 2)
plot(fit2)

## dissimilarity
dist &lt;- as.dist(sqrt(2-(2*sim)))

## dissimilarity, full width
fit3 &lt;- adjClust(dist, "dissimilarity")
plot(fit3)

## dissimilarity, h &lt; p-1
fit4 &lt;- adjClust(dist, "dissimilarity", h = 2)
plot(fit4)

</code></pre>

<hr>
<h2 id='chac'>Class chac</h2><span id='topic+chac'></span><span id='topic+as.hclust.chac'></span><span id='topic+print.chac'></span><span id='topic+head.chac'></span><span id='topic+summary.chac'></span><span id='topic+plot.chac'></span><span id='topic+diagnose'></span><span id='topic+diagnose.chac'></span><span id='topic+correct'></span><span id='topic+correct.chac'></span><span id='topic+cutree_chac'></span><span id='topic+cuttree_chac'></span>

<h3>Description</h3>

<p>S3 class for Constrained Hierarchical Agglomerative Clustering results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'chac'
as.hclust(x, ...)

## S3 method for class 'chac'
print(x, ...)

## S3 method for class 'chac'
head(x, ...)

## S3 method for class 'chac'
summary(object, ...)

## S3 method for class 'chac'
plot(
  x,
  y,
  ...,
  mode = c("standard", "corrected", "total-disp", "within-disp", "average-disp"),
  nodeLabel = FALSE
)

diagnose(x, graph = TRUE, verbose = TRUE)

correct(x)

cutree_chac(tree, k = NULL, h = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="chac_+3A_x">x</code>, <code id="chac_+3A_object">object</code>, <code id="chac_+3A_tree">tree</code></td>
<td>
<p>an object of class 'chac'</p>
</td></tr>
<tr><td><code id="chac_+3A_...">...</code></td>
<td>
<p>for <code><a href="base.html#topic+plot">plot</a></code>, arguments passed to the function 
<code><a href="stats.html#topic+plot.dendrogram">plot.dendrogram</a></code>. Default values for <code>type</code> and
<code>leaflab</code> are respectively set to <code>"triangle"</code> and <code>"none"</code></p>
</td></tr>
<tr><td><code id="chac_+3A_y">y</code></td>
<td>
<p>not used</p>
</td></tr>
<tr><td><code id="chac_+3A_mode">mode</code></td>
<td>
<p>type of dendrogram to plot (see Details). Default to 
<code>"standard"</code></p>
</td></tr>
<tr><td><code id="chac_+3A_nodelabel">nodeLabel</code></td>
<td>
<p>(logical) whether the order of merging has to be displayed
or not. <code>nodeLabel=TRUE</code> prints orders of fusion at corresponding
nodes. Default to <code>FALSE</code></p>
</td></tr>
<tr><td><code id="chac_+3A_graph">graph</code></td>
<td>
<p>(logical) whether the diagnostic plot has to be displayed or 
not. Default to <code>TRUE</code></p>
</td></tr>
<tr><td><code id="chac_+3A_verbose">verbose</code></td>
<td>
<p>(logical) whether to print a summary of the result or not.
Default to <code>TRUE</code></p>
</td></tr>
<tr><td><code id="chac_+3A_k">k</code></td>
<td>
<p>an integer scalar or vector with the desired number of groups</p>
</td></tr>
<tr><td><code id="chac_+3A_h">h</code></td>
<td>
<p>numeric scalar or vector with heights where the tree should be cut.
Only available when the heights are increasing</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Methods for class 'chac'
</p>
<p>When <code><a href="#topic+plot.chac">plot.chac</a></code> is called with 
<code>mode = "standard"</code>, the standard dendrogram is plotted, even though,
due to contingency constrains, some branches are reversed (decreasing
merges). When <code><a href="#topic+plot.chac">plot.chac</a></code> is called with 
<code>mode = "corrected"</code>, a correction is applied to original heights so as
to have only non decreasing merges). It does not change the result of the 
clustering, only the look of the dendrogram for easier interpretation.<br /><br />
Other modes are provided that correspond to different alternatives
described in Grimm (1987): </p>

<ul>
<li><p> in <code>mode = "within-disp"</code>, heights correspond to within-cluster
dispersion, <em>i.e.</em>, for a corresponding cluster, its height is 
</p>
<p style="text-align: center;"><code class="reqn">I(C) = \sum_{i \in C} d(i,g_C)</code>
</p>
<p> where <code class="reqn">d</code> is the dissimilarity 
used to cluster objects and <code class="reqn">g_C</code> is the center of gravity of cluster
<code class="reqn">C</code>. In this case, heights are always non decreasing;
</p>
</li>
<li><p> in <code>mode = "total-disp"</code>, heights correspond to the total
within-cluster dispersion. It is obtained from <code>mode = "standard"</code> by
the cumulative sum of its heights. In this case, heights are always
non decreasing;
</p>
</li>
<li><p> in <code>mode = "average-disp"</code>, heights correspond to the 
within-cluster dispersion divided by the cluster size. In this case, there 
is no guaranty that the heights are non decreasing. When reversals are 
detected, a warning is printed to advice the user to change the mode of the
representation.</p>
</li></ul>

<p>Grimm (1987) indicates that heights as provided by 
<code>mode = "within-disp"</code> are highly dependent on cluster sizes and that 
the most advisable representation is the one provided by 
<code>mode = "total-disp"</code>. Further details are provided in the vignette 
&quot;Notes on CHAC implementation in adjclust&quot;.
</p>


<h3>Value</h3>

<p>The function <code>plot.chac</code> displays the dendrogram and 
additionally invisibly returns an object of class 
<code><a href="stats.html#topic+dendrogram">dendrogram</a></code> with heights as specified by the user through
the option <code>mode</code>.
</p>
<p><code><a href="#topic+diagnose">diagnose</a></code> invisibly exports a data frame with the 
numbers of decreasing merges described by the labels of the clusters being
merged at this step and at the previous one, as well as the corresponding
merge heights.
</p>
<p>The function <code><a href="#topic+correct">correct</a></code> returns a <code>chac</code> objects with
modified heights so as they are increasing. The new heights are calculated in
an way identical to the option <code>mode = "corrected"</code> of the function
<code>plot.chac</code> (see Details). In addition, the <code>chac</code> object has its
field <code>method</code> modified from <code>adjClust</code> to 
<code>adjClust-modified</code>.
</p>
<p>The function <code><a href="#topic+cutree_chac">cutree_chac</a></code> returns the clustering with 
<code>k</code> groups or with the groups obtained by cutting the tree at height
<code>h</code>. If the heights are not increasing, the cutting of the tree is based
on the corrected heights as provided by the function <code>correct</code>.
</p>


<h3>References</h3>

<p> Grimm, E.C. (1987) CONISS: a fortran 77 program for
stratigraphically constrained analysis by the method of incremental sum of
squares. <em>Computer &amp; Geosciences</em>, <strong>13</strong>(1), 13-35. 
</p>

<hr>
<h2 id='hicClust'>Adjacency-constrained Clustering of Hi-C contact maps</h2><span id='topic+hicClust'></span>

<h3>Description</h3>

<p>Adjacency-constrained hierarchical agglomerative clustering of Hi-C contact
maps
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hicClust(x, h = NULL, log = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hicClust_+3A_x">x</code></td>
<td>
<p>either: 1. A pxp contact sparse or dense matrix (classes matrix,
Matrix, dscMatrix, dgTMatrix, dgCMatrix, dgeMatrix). Its entries are the 
number of counts of physical interactions observed between all pairs of loci. 
2. An object of class HiTC::HTCexp. The corresponding Hi-C data is stored as 
a Matrix::dsCMatrix object in the intdata slot. 3. A text file path with one 
line per pair of loci for which an interaction has been observed (in the 
format: locus1&lt;tab&gt;locus2&lt;tab&gt;signal) or a matrix or data frame with similar
data (3 columns).</p>
</td></tr>
<tr><td><code id="hicClust_+3A_h">h</code></td>
<td>
<p>band width. If not provided, <code>h</code> is set to default value 'p-1'.</p>
</td></tr>
<tr><td><code id="hicClust_+3A_log">log</code></td>
<td>
<p>logical. Whether to log-transform the count data. Default to 
<code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="hicClust_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to <code><a href="utils.html#topic+read.table">read.table</a></code> 
function when <code>x</code> is a text file name. If not provided, the text file 
is supposed to be separated by tabulations, with no header.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Adjacency-constrained hierarchical agglomerative clustering (HAC) is HAC in
which each observation is associated to a position, and the clustering is 
constrained so as only adjacent clusters are merged. Genomic regions (loci)
are clustered according to information provided by high-throughput
conformation capture data (Hi-C).
</p>


<h3>Value</h3>

<p>An object of class <code><a href="#topic+chac">chac</a></code>.
</p>


<h3>References</h3>

<p>Ambroise C., Dehman A., Neuvial P., Rigaill G., and Vialaneix N
(2019). <em>Adjacency-constrained hierarchical clustering of a band
similarity matrix with application to genomics</em>, Algorithms for Molecular
Biology 14(22)&quot;
</p>
<p>Servant N. <em>et al</em> (2012). <em>HiTC : Exploration of 
High-Throughput 'C' experiments. Bioinformatics</em>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+adjClust">adjClust</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># input as HiTC::HTCexp object
## Not run: 
if (require("HiTC", quietly = TRUE)) {
  load(system.file("extdata", "hic_imr90_40_XX.rda", package = "adjclust"))
  res1 &lt;- hicClust(hic_imr90_40_XX)
}

## End(Not run)

# input as Matrix::dsCMatrix contact map
## Not run: 
mat &lt;- HiTC::intdata(hic_imr90_40_XX) 
res2 &lt;- hicClust(mat)

## End(Not run)

# input as text file
res3 &lt;- hicClust(system.file("extdata", "sample.txt", package = "adjclust"))

</code></pre>

<hr>
<h2 id='plotSim'>Plot (dis)similarity matrix</h2><span id='topic+plotSim'></span>

<h3>Description</h3>

<p>Heatmap of the (dis)similarity matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSim(
  mat,
  type = c("similarity", "dissimilarity"),
  clustering = NULL,
  dendro = NULL,
  k = NULL,
  log = TRUE,
  legendName = "intensity",
  main = NULL,
  priorCount = 0.5,
  stats = c("R.squared", "D.prime"),
  h = NULL,
  axis = FALSE,
  naxis = min(10, nrow(mat)),
  axistext = NULL,
  xlab = "objects",
  cluster_col = "darkred",
  mode = c("standard", "corrected", "total-disp", "within-disp", "average-disp")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotSim_+3A_mat">mat</code></td>
<td>
<p>matrix to plot. It can be of class <code>'matrix'</code>, 
<code>'dgCMatrix'</code>, <code>'dsCMatrix'</code>, <code>'dist'</code>, <code>'HTCexp'</code>,
<code>'snpMatrix'</code>.</p>
</td></tr>
<tr><td><code id="plotSim_+3A_type">type</code></td>
<td>
<p>input matrix type. Can be either <code>"similarity"</code> or 
<code>"dissimilarity"</code> (kernels are supposed to be of type 
<code>"similarity"</code>).</p>
</td></tr>
<tr><td><code id="plotSim_+3A_clustering">clustering</code></td>
<td>
<p>vector of clusters to display on the matrix (if not 
<code>NULL</code>). If <code>clustering</code> is provided, it must be a numeric vector
of length identical to the matrix size with clusters identified as 
consecutive integers 1, 2, 3, ...</p>
</td></tr>
<tr><td><code id="plotSim_+3A_dendro">dendro</code></td>
<td>
<p>dendrogram provided as an <code>hclust</code> object, or as another
type of object that can be converted to <code>hclust</code> with <code>as.hclust</code>.</p>
</td></tr>
<tr><td><code id="plotSim_+3A_k">k</code></td>
<td>
<p>number of clusters to display. Used only when <code>dendro</code> is not
<code>NULL</code> and <code>clustering</code> is <code>NULL</code>. The clustering is then 
deduced from the dendrogram by a standard cut.</p>
</td></tr>
<tr><td><code id="plotSim_+3A_log">log</code></td>
<td>
<p>logical. Should the breaks be based on log-scaled values of the
matrix entries. Default to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plotSim_+3A_legendname">legendName</code></td>
<td>
<p>character. Title of the legend. Default to 
<code>"intensity"</code>.</p>
</td></tr>
<tr><td><code id="plotSim_+3A_main">main</code></td>
<td>
<p>character. Title of the plot. Default to <code>NULL</code> (no title).</p>
</td></tr>
<tr><td><code id="plotSim_+3A_priorcount">priorCount</code></td>
<td>
<p>numeric. Average count to be added to each entry of the
matrix to avoid taking log of zero. Used only if <code>log == TRUE</code> and 
default to 0.5.</p>
</td></tr>
<tr><td><code id="plotSim_+3A_stats">stats</code></td>
<td>
<p>input SNP correlation type. Used when <code>mat</code> is of type 
<code>SnpMatrix</code>.</p>
</td></tr>
<tr><td><code id="plotSim_+3A_h">h</code></td>
<td>
<p>positive integer. Threshold distance for SNP correlation 
computation. Used when <code>mat</code> is of type <code>SnpMatrix</code>.</p>
</td></tr>
<tr><td><code id="plotSim_+3A_axis">axis</code></td>
<td>
<p>logical. Should x-axis be displayed on the plot? Default to 
<code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plotSim_+3A_naxis">naxis</code></td>
<td>
<p>integer. If <code>axis == TRUE</code>, number of ticks to display on
the x-axis.</p>
</td></tr>
<tr><td><code id="plotSim_+3A_axistext">axistext</code></td>
<td>
<p>character vector. If <code>axis == TRUE</code>, labels to display
of the x-axis (its length has to be equal to <code>naxis</code>).</p>
</td></tr>
<tr><td><code id="plotSim_+3A_xlab">xlab</code></td>
<td>
<p>character. If <code>axis == TRUE</code>, x-axis title.</p>
</td></tr>
<tr><td><code id="plotSim_+3A_cluster_col">cluster_col</code></td>
<td>
<p>colour for the cluster line if <code>clustering</code> is not
<code>NULL</code>.</p>
</td></tr>
<tr><td><code id="plotSim_+3A_mode">mode</code></td>
<td>
<p>type of dendrogram to plot (see <code><a href="#topic+plot.chac">plot.chac</a></code>). Default
to <code>"standard"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function produces a heatmap for the used (dis)similarity matrix that 
can be used as a diagnostic plot to check the consistency between the 
obtained clustering and the original (dis)similarity
</p>


<h3>See Also</h3>

<p><code><a href="#topic+select">select</a></code>, <code><a href="#topic+adjClust">adjClust</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
clustering &lt;- rep(1:3, each = 50)
dist_data &lt;- as.matrix(dist(iris[, 1:4]))
dendro_iris &lt;- adjClust(dist_data, type = "dissimilarity")
plotSim(dist_data, type = "dissimilarity", dendro = dendro_iris, axis = TRUE)
plotSim(dist_data, type = "dissimilarity", dendro = dendro_iris,
        clustering = clustering)
plotSim(dist_data, type = "dissimilarity", dendro = dendro_iris, axis = TRUE,
        k = 3)
plotSim(dist_data, type = "dissimilarity", legendName = "IF", axis = TRUE, 
        clustering = clustering)
p &lt;- plotSim(dist(iris[, 1:4]), type = "dissimilarity", log = FALSE, 
             clustering = clustering, cluster_col = "blue")
# custom palette
p + scale_fill_gradient(low = "yellow", high = "red")
# dsCMatrix
m &lt;- Matrix(c(0, 0, 2, 0, 3, 0, 2, 0, 0), ncol = 3)
res &lt;- adjClust(m)
plotSim(m, axis = TRUE)
plotSim(m, dendro = res)
# dgCMatrix
m &lt;- as(m, "generalMatrix")
plotSim(m)
m &lt;- as.dist(m)
if (require("HiTC", quietly = TRUE)) {
  load(system.file("extdata", "hic_imr90_40_XX.rda", package = "adjclust"))
  res &lt;- hicClust(hic_imr90_40_XX, log = TRUE)
  plotSim(hic_imr90_40_XX, axis = TRUE)
}
if (requireNamespace("snpStats", quietly = TRUE)) {
  data(testdata, package = "snpStats")
  plotSim(Autosomes[1:200, 1:5], h = 3, stats = "R.squared", axis = TRUE,
          axistext = c("A", "B", "C", "D", "E"))
}

## End(Not run)
</code></pre>

<hr>
<h2 id='select'>Clustering selection</h2><span id='topic+select'></span><span id='topic+select.chac'></span>

<h3>Description</h3>

<p>Clustering selection from a chac object with the slope heuristic
or the broken stick heuristic
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select(
  x,
  type = c("capushe", "bstick"),
  k.max = NULL,
  graph = FALSE,
  pct = 0.15
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="select_+3A_x">x</code></td>
<td>
<p>an object of class 'chac'</p>
</td></tr>
<tr><td><code id="select_+3A_type">type</code></td>
<td>
<p>model selection approach between slope heuristic 
(<code>"capushe"</code>) and broken stick approach (<code>"bstick"</code>)</p>
</td></tr>
<tr><td><code id="select_+3A_k.max">k.max</code></td>
<td>
<p>maximum number of clusters that can be selected. Default to 
<code>NULL</code>, in which case it is set to 
<code class="reqn">\min(\max(100, \frac{n}{\log(n)}), \frac{n}{2})</code> where <code class="reqn">n</code> is the
number of objects to be clustered for capushe and to <code class="reqn">n</code> for the broken
stick model</p>
</td></tr>
<tr><td><code id="select_+3A_graph">graph</code></td>
<td>
<p>logical. Whether the diagnostic plot for the capushe selection
is displayed or not. Default to <code>FALSE</code></p>
</td></tr>
<tr><td><code id="select_+3A_pct">pct</code></td>
<td>
<p>minimum percentage of points for the plateau selection in 
capushe selection. See <code><a href="capushe.html#topic+DDSE">DDSE</a></code> for further details</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns the clustering selected by the slope heuristic,
as implemented in the R package <code>capushe</code>.
</p>


<h3>References</h3>

<p>Baudry J.P., Maugis C., and Michel B. (2012). Slope heuristics: 
overview and implementation. <em>Statistics and Computing</em>, <strong>22</strong>(2),
355-470.
MacArthur, R.H. (1957) On the relative abundance of bird species. 
<em>Proceedings of the National Academy of Sciences</em>, <strong>43</strong>, 293-295.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: if (require("HiTC", quietly = TRUE)) {
  load(system.file("extdata", "hic_imr90_40_XX.rda", package = "adjclust"))
  res &lt;- hicClust(hic_imr90_40_XX, log = TRUE)
  selected.capushe &lt;- select(res)
  table(selected.capushe)
  selected.bs &lt;- select(res, type = "bstick")
  table(selected.bs)
}
## End(Not run)

res &lt;- adjClust(dist(iris[, 1:4]))
select.clust &lt;- select(res, "bs")
table(select.clust)

</code></pre>

<hr>
<h2 id='snpClust'>Adjacency-constrained Clustering of Single Nucleotide Polymorphisms</h2><span id='topic+snpClust'></span>

<h3>Description</h3>

<p>Adjacency-constrained hierarchical agglomerative clustering of Single
Nucleotide Polymorphisms based on Linkage Disequilibrium
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snpClust(x, h = ncol(x) - 1, stats = c("R.squared", "D.prime"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="snpClust_+3A_x">x</code></td>
<td>
<p>either a genotype matrix of class
<code><a href="snpStats.html#topic+SnpMatrix-class">SnpMatrix</a></code>/<code><a href="base.html#topic+matrix">matrix</a></code> or a
linkage disequilibrium matrix of class
<code><a href="Matrix.html#topic+dgCMatrix-class">dgCMatrix</a></code>. In the latter case the LD
values are expected to be in [0,1]</p>
</td></tr>
<tr><td><code id="snpClust_+3A_h">h</code></td>
<td>
<p>band width. If not provided, <code>h</code> is set to default value 'p-1'
where 'p' is the number of columns of <code>x</code></p>
</td></tr>
<tr><td><code id="snpClust_+3A_stats">stats</code></td>
<td>
<p>a character vector specifying the linkage disequilibrium
measures to be calculated (using the <code><a href="snpStats.html#topic+ld">ld</a></code>
function) when <code>x</code> is a genotype matrix. Only &quot;R.squared&quot; and
&quot;D.prime&quot; are allowed, see Details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Adjacency-constrained hierarchical agglomerative clustering (HAC) is HAC in
which each observation is associated to a position, and the clustering is
constrained so as only adjacent clusters are merged. SNPs are clustered based
on their similarity as measured by the linkage disequilibrium.
</p>
<p>In the special case where genotypes are given as input and the corresponding
LD matrix has missing entries, the clustering cannot be performed. This can
typically happen when there is insufficient variability in the sample
genotypes. In this special case, the indices of the SNP pairs which yield
missing values are returned.
</p>
<p>If <code>x</code> is of class
<code><a href="snpStats.html#topic+SnpMatrix-class">SnpMatrix</a></code> or <code><a href="base.html#topic+matrix">matrix</a></code>,
it is assumed to be a <code class="reqn">n \times p</code> matrix of <code class="reqn">p</code> genotypes for
<code class="reqn">n</code> individuals. This input is converted to a LD similarity matrix
using the <code>snpStats::ld</code>. If <code>x</code> is of class
<code><a href="Matrix.html#topic+dgCMatrix-class">dgCMatrix</a></code>, it is assumed to be a
(squared) LD matrix.
</p>
<p>Clustering on a LD similarity other than &quot;R.squared&quot; or &quot;D.prime&quot; can be
performed by providing the LD values directly as argument <code>x</code>. These
values are expected to be in [0,1], otherwise they are truncated to [0,1].
</p>


<h3>Value</h3>

<p>An object of class <code><a href="#topic+chac">chac</a></code> (when no LD value is missing)
</p>


<h3>References</h3>

<p>Dehman A. (2015) <em>Spatial Clustering of Linkage
Disequilibrium Blocks for Genome-Wide Association Studies</em>, PhD thesis,
Universite Paris Saclay.
</p>
<p>Dehman, A. Ambroise, C. and Neuvial, P. (2015). Performance of a
blockwise approach in variable selection using linkage disequilibrium
information. *BMC Bioinformatics* 16:148.
</p>
<p>Ambroise C., Dehman A., Neuvial P., Rigaill G., and Vialaneix N
(2019). <em>Adjacency-constrained hierarchical clustering of a band
similarity matrix with application to genomics</em>, Algorithms for Molecular
Biology 14(22)&quot;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+adjClust">adjClust</a></code> <code><a href="snpStats.html#topic+ld">ld</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## a very small example
if (requireNamespace("snpStats", quietly = TRUE)) {
  data(testdata, package = "snpStats")

  # input as snpStats::SnpMatrix
  fit1 &lt;- snpClust(Autosomes[1:200, 1:5], h = 3, stats = "R.squared")

  # input as base::matrix
  fit2 &lt;- snpClust(as.matrix(Autosomes[1:200, 1:5]), h = 3, stats = "R.squared")

  # input as Matrix::dgCMatrix
  ldres &lt;- snpStats::ld(Autosomes[1:200, 1:5], depth = 3, stats = "R.squared", symmetric = TRUE)
  fit3 &lt;- snpClust(ldres, 3)
}

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
