<!DOCTYPE html><html><head><title>Help for package AEenrich</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {AEenrich}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AEenrich-package'><p>AEenrich: Adverse Event Enrichment Tests</p></a></li>
<li><a href='#count_cases'><p>Convert data on the report level to aggregated data.</p></a></li>
<li><a href='#covid1'><p>Covid Vaccine Adverse Event Data</p></a></li>
<li><a href='#covid2'><p>Covid Vaccine Adverse Event Data</p></a></li>
<li><a href='#enrich'><p>Perform Adverse Event Enrichment Tests</p></a></li>
<li><a href='#group'><p>Group Structure Data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Title:</td>
<td>Adverse Event Enrichment Tests</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Description:</td>
<td>We extend existing gene enrichment tests to perform adverse
    event enrichment analysis. Unlike the continuous gene expression data,
    adverse event data are counts. Therefore, adverse event data has many zeros
    and ties. We propose two enrichment tests. One is a modified Fisher's exact
    test based on pre-selected significant adverse events, while the other is
    based on a modified Kolmogorov-Smirnov statistic. We add Covariate
    adjustment to improve the analysis."Adverse event enrichment tests using
    VAERS" Shuoran Li, Lili Zhao (2020) &lt;<a href="https://arxiv.org/abs/2007.02266">arXiv:2007.02266</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Biarch:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, magrittr, qvalue, doParallel, tidyr, modelr, foreach,
rlang, utils</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/umich-biostatistics/AEenrich">https://github.com/umich-biostatistics/AEenrich</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/umich-biostatistics/AEenrich/issues">https://github.com/umich-biostatistics/AEenrich/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-10-02 19:00:57 UTC; 86151</td>
</tr>
<tr>
<td>Author:</td>
<td>Shuoran Li [aut],
  Hongfan Chen [aut],
  Lili Zhao [aut],
  Michael Kleinsasser [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Michael Kleinsasser &lt;mkleinsa@umich.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-11-01 15:20:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='AEenrich-package'>AEenrich: Adverse Event Enrichment Tests</h2><span id='topic+AEenrich'></span><span id='topic+AEenrich-package'></span>

<h3>Description</h3>

<p>The count_cases function is used to convert data on the report
level to aggregated data, grouping by specified covariates.
</p>
<p>Use the function <code>count_cases</code> to convert report level data into aggregated data.
</p>
<p>See our <a href="https://github.com/umich-biostatistics/AEenrich">Github home page</a>
or run ?count_cases for examples.
</p>
<p>Perform Adverse Event Enrichment Tests
The enrich function is used to perform Adverse event (AE) enrichment analysis.
Unlike the continuous gene expression data, AE data are counts. Therefore,
AE data has many zeros and ties. We propose two enrichment tests. AEFisher is
a modified Fisher's exact test based on pre-selected significant AEs, while
AEKS is based on a modified Kolmogorov-Smirnov statistic.
</p>
<p>Use the function <code>enrich</code> to fit models and inspect results.
</p>
<p>See our <a href="https://github.com/umich-biostatistics/AEenrich">Github home page</a>
or run ?enrich for examples.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Michael Kleinsasser <a href="mailto:mkleinsa@umich.edu">mkleinsa@umich.edu</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Shuoran Li <a href="mailto:shuoranl@umich.edu">shuoranl@umich.edu</a>
</p>
</li>
<li><p> Hongfan Chen <a href="mailto:chenhf@umich.edu">chenhf@umich.edu</a>
</p>
</li>
<li><p> Lili Zhao <a href="mailto:zhaolili@med.umich.edu">zhaolili@med.umich.edu</a>
</p>
</li></ul>

<p><strong>Maintainer</strong>: Michael Kleinsasser <a href="mailto:mkleinsa@umich.edu">mkleinsa@umich.edu</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Shuoran Li <a href="mailto:shuoranl@umich.edu">shuoranl@umich.edu</a>
</p>
</li>
<li><p> Hongfan Chen <a href="mailto:chenhf@umich.edu">chenhf@umich.edu</a>
</p>
</li>
<li><p> Lili Zhao <a href="mailto:zhaolili@med.umich.edu">zhaolili@med.umich.edu</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/umich-biostatistics/AEenrich">https://github.com/umich-biostatistics/AEenrich</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/umich-biostatistics/AEenrich/issues">https://github.com/umich-biostatistics/AEenrich/issues</a>
</p>
</li></ul>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/umich-biostatistics/AEenrich">https://github.com/umich-biostatistics/AEenrich</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/umich-biostatistics/AEenrich/issues">https://github.com/umich-biostatistics/AEenrich/issues</a>
</p>
</li></ul>


<hr>
<h2 id='count_cases'>Convert data on the report level to aggregated data.</h2><span id='topic+count_cases'></span>

<h3>Description</h3>

<p>The count_cases function is used to convert data on the report
level to aggregated data, grouping by specified covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count_cases(
  data,
  drug.case = drug.case,
  drug.control = NULL,
  covar_disc = NULL,
  covar_cont = NULL,
  breaks = NULL,
  cores = detectCores(),
  min_AE = 10
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="count_cases_+3A_data">data</code></td>
<td>
<p>a data.frame with at least 3 columns, consisting data on the report
level, having ID, Drug type and AE name as the first 3 columns with
covariates(optional) followed. The order of columns is not interchangeable.</p>
</td></tr>
<tr><td><code id="count_cases_+3A_drug.case">drug.case</code></td>
<td>
<p>a character string for the target drug of interest.</p>
</td></tr>
<tr><td><code id="count_cases_+3A_drug.control">drug.control</code></td>
<td>
<p>a character string for the reference drug. If NULL(default),
all other drugs combined are the reference.</p>
</td></tr>
<tr><td><code id="count_cases_+3A_covar_disc">covar_disc</code></td>
<td>
<p>a character vector of categorical covariates.</p>
</td></tr>
<tr><td><code id="count_cases_+3A_covar_cont">covar_cont</code></td>
<td>
<p>a character vector of continuous covariates.</p>
</td></tr>
<tr><td><code id="count_cases_+3A_breaks">breaks</code></td>
<td>
<p>a list consists of vectors used for creating specific bins to
transform continuous covariates into categorical. Breaks Should have the same
length as covar_cont. Given a vector of non-decreasing breakpoints in <code>breaks[i]</code>,
find the interval containing each element of <code>covar_cont[i]</code>; i.e., for each index
j in <code>breaks[i]</code>, value j is assigned to <code>covar_cont[i]</code> if and only if <code>breaks[i][j]</code>
<code style="white-space: pre;">&#8288;&lt;= covar_cont[i] &lt; breaks[i][j+1]&#8288;</code>.</p>
</td></tr>
<tr><td><code id="count_cases_+3A_cores">cores</code></td>
<td>
<p>the number of cores to use for parallel execution.</p>
</td></tr>
<tr><td><code id="count_cases_+3A_min_ae">min_AE</code></td>
<td>
<p>the minimum number of cases required to start counting
for a specific AE. Default 10.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <strong>data.frame</strong> consists of aggregated data.
</p>
<p>The returned data.frame contains the following columns:
</p>

<ul>
<li><p>DRUG_TYPE: type of the drug, DrugYes for target drug and DrugNo for referenced drug
</p>
</li>
<li><p>AE_NAME: the name of the adverse event
</p>
</li>
<li><p>AEyes: number of observations that have this AE
</p>
</li>
<li><p>AEno: number of observations that do not have this AE
</p>
</li>
<li><p>covariates: covariates specifed by user
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
# count_cases(data = covid1, drug.case = "COVID19", drug.control = "OTHER",
#             covar_cont = c("AGE"), covar_disc = c("SEX"),
#             breaks = list(c(16,30,50,65,120)))
            
</code></pre>

<hr>
<h2 id='covid1'>Covid Vaccine Adverse Event Data</h2><span id='topic+covid1'></span>

<h3>Description</h3>

<p>Adverse event data in the long format. Each row is a single adverse
event, along with covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covid1
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 12500 rows and 5 columns.
</p>


<h3>Details</h3>


<ul>
<li><p> VAERS_ID Event ID
</p>
</li>
<li><p> VAX_LABEL Vaccine type
</p>
</li>
<li><p> AE_NAME Adverse event name
</p>
</li>
<li><p> AGE covariate
</p>
</li>
<li><p> SEX covariate
</p>
</li></ul>


<hr>
<h2 id='covid2'>Covid Vaccine Adverse Event Data</h2><span id='topic+covid2'></span>

<h3>Description</h3>

<p>Adverse event data in the short format. Each row is a count of adverse
events with the given name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covid2
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 2656 rows and 6 columns.
</p>


<h3>Details</h3>


<ul>
<li><p> DRUG_TYPE Vaccine type
</p>
</li>
<li><p> AE_NAME Adverse event name
</p>
</li>
<li><p> AEYes Number of observations that have this AE
</p>
</li>
<li><p> AENo Number of observations that do not have this AE
</p>
</li>
<li><p> AGE covariate
</p>
</li>
<li><p> SEX covariate
</p>
</li></ul>


<hr>
<h2 id='enrich'>Perform Adverse Event Enrichment Tests</h2><span id='topic+enrich'></span>

<h3>Description</h3>

<p>The enrich function is used to perform Adverse event (AE) enrichment analysis.
Unlike the continuous gene expression data, AE data are counts. Therefore,
AE data has many zeros and ties. We propose two enrichment tests. AEFisher is
a modified Fisher's exact test based on pre-selected significant AEs, while
AEKS is based on a modified Kolmogorov-Smirnov statistic.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enrich(
  data,
  dd.group,
  drug.case,
  drug.control = NULL,
  method = "aeks",
  n_perms = 1000,
  covar = NULL,
  p = 0,
  q.cut = 0.1,
  or.cut = 1.5,
  zero = FALSE,
  min_size = 5,
  min_AE = 10,
  cores = detectCores()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="enrich_+3A_data">data</code></td>
<td>
<p>a data.frame. Two data types are allowed. Type I data consisting
data on the report level, having ID, Drug type and AE name as the first 3
columns with covariates(optional) followed. Type II data have drug type and
AE name as the first two columns, with the 3rd and 4th Columns giving the
numbers of successes(have AE) and failures(Do not have AE) respectively, then
followed by covariates. See example data for details.</p>
</td></tr>
<tr><td><code id="enrich_+3A_dd.group">dd.group</code></td>
<td>
<p>a data.frame with AE name and Group name. This data.frame have
the group information for each individual AE.</p>
</td></tr>
<tr><td><code id="enrich_+3A_drug.case">drug.case</code></td>
<td>
<p>a character string for the target drug of interest.</p>
</td></tr>
<tr><td><code id="enrich_+3A_drug.control">drug.control</code></td>
<td>
<p>a character string for the reference drug. If NULL(default),
all other drugs combined are the reference.</p>
</td></tr>
<tr><td><code id="enrich_+3A_method">method</code></td>
<td>
<p>a character string specifying the method for the enrichment test.
It must take &quot;aeks&quot; (default) or &quot;aefisher&quot;; &quot;aeks&quot; is the rank-based
enrichment test, and &quot;aefisher&quot; is the Fisher enrichment test. See details
described in the paper (see reference section of this document).</p>
</td></tr>
<tr><td><code id="enrich_+3A_n_perms">n_perms</code></td>
<td>
<p>an integer value specifying the number of permutations in
permutation test.</p>
</td></tr>
<tr><td><code id="enrich_+3A_covar">covar</code></td>
<td>
<p>a character vector specifying the columns of covariates, default
NULL.</p>
</td></tr>
<tr><td><code id="enrich_+3A_p">p</code></td>
<td>
<p>a numerical value to control the weight of the step, can take any
value between 0 and 1. If 0(default), reduces to the standard Kolmogorov-Smirnov
statistics.</p>
</td></tr>
<tr><td><code id="enrich_+3A_q.cut">q.cut</code></td>
<td>
<p>a numerical value specifying the significance cut for q value
of AEs in aefisher.</p>
</td></tr>
<tr><td><code id="enrich_+3A_or.cut">or.cut</code></td>
<td>
<p>a numerical value specifying the significance cut for odds ratio
of AEs in aefisher.</p>
</td></tr>
<tr><td><code id="enrich_+3A_zero">zero</code></td>
<td>
<p>logical, default FALSE.If TRUE, add zero indicator to enrichment score.</p>
</td></tr>
<tr><td><code id="enrich_+3A_min_size">min_size</code></td>
<td>
<p>the minimum size of group required for enrichment analysis.</p>
</td></tr>
<tr><td><code id="enrich_+3A_min_ae">min_AE</code></td>
<td>
<p>the minimum number of cases required to start counting
for a specific AE.</p>
</td></tr>
<tr><td><code id="enrich_+3A_cores">cores</code></td>
<td>
<p>the number of cores to use for parallel execution.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing 2 data.frames named <strong>Final_result</strong> and <strong>AE_info</strong>.
</p>
<p>The <strong>Final_result</strong> data.frame contains the following columns:
</p>

<ul>
<li><p>GROUP_NAME: AE group names
</p>
</li>
<li><p>ES: enrichment score
</p>
</li>
<li><p>p_value: p value of the enrichment test
</p>
</li>
<li><p>GROUP_SIZE: number of AEs per group
</p>
</li></ul>

<p>The <strong>AE_info</strong> contains the following columns:
</p>

<ul>
<li><p>AE_NAME: AE names
</p>
</li>
<li><p>OR: odds ratio for each individual AE
</p>
</li>
<li><p>p_value: p value for AE-drug association
</p>
</li>
<li><p>95Lower: lower bound of 95 percent confidence interval of odds ratio
</p>
</li>
<li><p>95Lower: upper bound of 95 percent confidence interval of odds ratio
</p>
</li>
<li><p>se(logOR): standard error of log odds ratio
</p>
</li></ul>



<h3>References</h3>

<p>Li, S. and Zhao, L. (2020). Adverse event enrichment tests using
VAERS. <a href="https://arxiv.org/abs/2007.02266">arXiv:2007.02266</a>.
</p>
<p>Subramanian, A.e.a. (2005). Gene set enrichment analysis: a knowledge-based
approach for interpreting genome-wide expression profiles. Proc Natl Acad
Sci U S A. Proceedings of the National Academy of Sciences. 102. 15545-15550.
</p>
<p>Tian, Lu &amp; Greenberg, Steven &amp; Kong, Sek Won &amp; Altschuler, Josiah &amp; Kohane, Isaac &amp; Park,
Peter. (2005). Discovering statistically significant pathways in expression profiling studies.
Proceedings of the National Academy of Sciences of the United States of America.
102. 13544-9. 10.1073/pnas.0506577102.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# AEKS

### Type I data: data on report level
# enrich(data = covid1, covar = c("SEX", "AGE"), p = 0, method = "aeks",
#        n_perms = 1000, drug.case = "COVID19", dd.group = group, cores = 2,
#        drug.control = "OTHER", min_size = 5, min_AE = 10, zero = FALSE)
      
## Type II data: aggregated data
# enrich(data = covid2, covar = c("SEX", "AGE"), p = 0, method = "aeks",
#        n_perms = 1000, drug.case = "DrugYes", dd.group = group, cores = 2,
#        drug.control = "DrugNo", min_size = 5, min_AE = 10)
      
# AEFISHER
## Type I data: data on report level
# enrich(data = covid1, covar = c("SEX", "AGE"), p = 0, method = "aefisher",
#        n_perms = 1000, drug.case = "COVID19", dd.group = group,
#        drug.control = "OTHER", min_size = 5, min_AE = 10, q.cut = 0.05, 
#        or.cut = 1.5, cores = 2)
      
## Type II data: aggregated data
# enrich(data = covid2, covar = c("SEX", "AGE"), p = 0, method = "aefisher",
#        n_perms = 1000, drug.case = "DrugYes", dd.group = group,
#        drug.control = "DrugNo", min_size = 5, min_AE = 10, cores = 2)
      
</code></pre>

<hr>
<h2 id='group'>Group Structure Data</h2><span id='topic+group'></span>

<h3>Description</h3>

<p>Identifies which group each set of adverse events belongs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>group
</code></pre>


<h3>Format</h3>

<p>An object of class <code>NULL</code> of length 0.
</p>


<h3>Details</h3>


<ul>
<li><p> AE_NAME Adverse event name
</p>
</li>
<li><p> GROUP_NAME Group name
</p>
</li></ul>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
