<!DOCTYPE html><html lang="en"><head><title>Help for package AGHmatrix</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {AGHmatrix}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Amatrix'><p>Construction of Relationship Matrix A</p></a></li>
<li><a href='#AmatrixPolyCross'><p>Construction of pedigree-based relationship matrix with parental guessing possibility</p></a></li>
<li><a href='#datatreat'><p>Organizes pedigree data in a chronological way</p></a></li>
<li><a href='#expandAmatrix'><p>Add new crosses to a current A matrix</p></a></li>
<li><a href='#filterpedigree'><p>Filter the pedigree to keep only the genealogy of a subset of individuals</p></a></li>
<li><a href='#formatmatrix'><p>Transform a matrix in 3 columns</p></a></li>
<li><a href='#Gmatrix'><p>Construction of Relationship Matrix G</p></a></li>
<li><a href='#Hmatrix'><p>Construction of Combined Relationship Matrix H</p></a></li>
<li><a href='#Mcheck'><p>Check and filter markers</p></a></li>
<li><a href='#missingdata'><p>Survying on missing data</p></a></li>
<li><a href='#ped.mrode'><p>Pedigree Data</p></a></li>
<li><a href='#ped.sol'><p>Pedigree data for autopolyploid examples</p></a></li>
<li><a href='#snp.pine'><p>Molecular data for diploid examples</p></a></li>
<li><a href='#snp.sol'><p>Molecular data for autopolyploid examples</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Relationship Matrices for Diploid and Autopolyploid Species</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.4</td>
</tr>
<tr>
<td>Description:</td>
<td>Computation of A (pedigree), G (genomic-base), and H (A corrected
    by G) relationship matrices for diploid and autopolyploid species. Several methods
    are implemented considering additive and non-additive models.</td>
</tr>
<tr>
<td>Author:</td>
<td>Rodrigo Amadeu [aut, cre],
  Luis Ferrao [aut, ctb],
  Catherine Cellon [ctb],
  Leticia Lara [ctb],
  Marcio Resende [ctb],
  Ivone Oliveira [ctb],
  Patricio Munoz [ctb],
  Augusto Garcia [ctb]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Matrix (&ge; 1.2-7.1), zoo (&ge; 1.8.6)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, MASS, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/rramadeu/AGHmatrix">https://github.com/rramadeu/AGHmatrix</a></td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Rodrigo Amadeu &lt;rramadeu@gmail.com&gt;</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-03 15:25:06 UTC; GNIEZ</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-03 18:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='Amatrix'>Construction of Relationship Matrix A</h2><span id='topic+Amatrix'></span>

<h3>Description</h3>

<p>Creates an additive relationship matrix A from a pedigree data in a 3-column way format based on ploidy level (an even number) and, if ploidy equals 4, based on proportion of parental gametes that are IBD (Identical by Descent) due to double reduction. Returns a dominance relationship matrix if dominance true (ploidy 2 only). Autopolyploid matrices based on Kerr (2012), used when 'ploidy' argument is higher than '2' and 'dominance=FALSE'.
Diploid additive numerator relationship matrix built as in Henderson (1976), used when 'ploidy=2' and 'dominance=FALSE'. Diploid dominance numerator relationship matrix built as in Cockerham (1954), used when 'ploidy=2' and 'dominance=FALSE'. For details of recursive method see Mrode (2005).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Amatrix(
  data = NULL,
  ploidy = 2,
  w = 0,
  verify = TRUE,
  dominance = FALSE,
  slater = FALSE,
  ASV = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Amatrix_+3A_data">data</code></td>
<td>
<p>pedigree data name (3-column way format). Unknown value should be equal 0.</p>
</td></tr>
<tr><td><code id="Amatrix_+3A_ploidy">ploidy</code></td>
<td>
<p>an even number (default=2).</p>
</td></tr>
<tr><td><code id="Amatrix_+3A_w">w</code></td>
<td>
<p>proportion of parental gametas IBD due to double reduction (default=0), only if ploidy=4.</p>
</td></tr>
<tr><td><code id="Amatrix_+3A_verify">verify</code></td>
<td>
<p>verifies pedigree file for conflictuos entries (default=TRUE).</p>
</td></tr>
<tr><td><code id="Amatrix_+3A_dominance">dominance</code></td>
<td>
<p>if true, returns the dominance relationship matrix</p>
</td></tr>
<tr><td><code id="Amatrix_+3A_slater">slater</code></td>
<td>
<p>if true, returns the additive autotetraploid relationship matrix as Slater (2013)</p>
</td></tr>
<tr><td><code id="Amatrix_+3A_asv">ASV</code></td>
<td>
<p>if TRUE, transform matrix into average semivariance (ASV) equivalent (K = K / (trace(K) / (nrow(K)-1))). Details formula 2 of Fieldmann et al. (2022). Default = FALSE.</p>
</td></tr>
<tr><td><code id="Amatrix_+3A_...">...</code></td>
<td>
<p>arguments to be passed to datatreat()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix with the Relationship between the individuals.
</p>


<h3>Author(s)</h3>

<p>Rodrigo R Amadeu, <a href="mailto:rramadeu@gmail.com">rramadeu@gmail.com</a>
</p>


<h3>References</h3>

<p><em>Cockerham, CC. 1954. An extension of the concept of partitioning hereditary variance for analysis of covariances among relatives when epistasis is present. Genetics 39, 859–882</em>
</p>
<p><em>Feldmann MJ, et al. 2022. Average semivariance directly yields accurate estimates of the genomic variance in complex trait analyses. G3 (Bethesda), 12(6).</em>
</p>
<p><em>Henderson, CR. 1976. A simple method for computing the inverse of a numerator relationship matrix used in prediction of breeding values. Biometrics 32, 69-83</em>
</p>
<p><em>Kerr, RJ, et al. 2012. Use of the numerator relationship matrix in genetic analysis of autopolyploid species. Theoretical and Applied Genetics 124 1271-1282</em>
</p>
<p><em>Mrode, RA. 2014. Chapter 2: Genetic Covariance Between Relatives and Chapter 9: Non-additive Animal Models in Mrode, RA. 2014. Linear models for the prediction of animal breeding values. Cabi, 3rd edition.</em>
</p>
<p><em>Slater, AT, et al. 2013. Improving the analysis of low heritability complex traits for enhanced genetic gain in potato. Theoretical and Applied Genetics 127, 809-820</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ped.mrode)
#Computing additive relationship matrix considering diploidy (Henderson 1976):
Amatrix(ped.mrode, ploidy=2)
#Computing non-additive relationship matrix considering diploidy (Cockerham 1954):
Amatrix(ped.mrode, ploidy=2, dominance=TRUE)
#Computing additive relationship matrix considering autotetraploidy (Kerr 2012):
Amatrix(ped.mrode, ploidy=4)
#Computing additive relationship matrix considering autooctaploidy (Kerr 2012):
Amatrix(ped.mrode, ploidy=8)
#Computing additive relationship matrix considering autotetraploidy and double-
#reduction of 0.1 (Kerr 2012):
Amatrix(ped.mrode, ploidy=4, w=0.1)
#Computing additive relationship matrix considering 
#autotetraploidy and double-reduction of 0.1 (Slater 2014):
Amatrix(ped.mrode, ploidy=4, w=0.1, slater = TRUE)
#Computing additive relationship matrix considering autohexaploidy and double-
#reduction of 0.1 (Kerr 2012):
Amatrix(ped.mrode, ploidy=6, w=0.1)

</code></pre>

<hr>
<h2 id='AmatrixPolyCross'>Construction of pedigree-based relationship matrix with parental guessing possibility</h2><span id='topic+AmatrixPolyCross'></span>

<h3>Description</h3>

<p>Creates an additive relationship matrix A based on a non-deterministic pedigree with 4+ columns where each column represents a possible parent. This function was built with the following designs in mind. 
1) A mating design where you have equally possible parents. For example, a generation of insects derived from the mating of three insects in a cage. All the insects in this generation will have the same expected relatedness with all the possible parents (1/3). If there are only two parents in the cage, the function assumes no-inbreeding and the pedigree is deterministic (the individual is offspring of the cross between the two parents). Another example, a population of 10 open-pollinated plants where you harvest the seeds without tracking the mother. 
2) When fixedParent is TRUE: a mating design where you know one parent and might know the other possible parents. For example, a polycross design where you have seeds harvested from a mother plant and possible polen donors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AmatrixPolyCross(data = NULL, fixedParent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AmatrixPolyCross_+3A_data">data</code></td>
<td>
<p>pedigree data name. Unknown value should be equal 0. See example for construction.</p>
</td></tr>
<tr><td><code id="AmatrixPolyCross_+3A_fixedparent">fixedParent</code></td>
<td>
<p>if false, assumes that all the parents are equally possible parents. If true, assumes that the first parental is known and the others are equally possible parents. Default = FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix with the relationship between the individuals.
</p>


<h3>Author(s)</h3>

<p>Rodrigo R Amadeu, <a href="mailto:rramadeu@gmail.com">rramadeu@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#the following pedigree has the id of the individual followed by possible parents
#if 0 is unknown
#the possible parents are filled from left to right
#in the pedigree data frame examples:
#id 1,2,3,4 have unknown parents and are assumed unrelated
#id 5 has three possible parents (1,2,3)
#id 6 has three possible parents (2,3,4)
#id 7 has two parents (deterministic case here, the parents are 3 and 4)
#id 8 has four possible parents (5,6,7,1)

pedigree = data.frame(id=1:8,
                      parent1 = c(0,0,0,0,1,2,3,5),
                      parent2 = c(0,0,0,0,2,3,4,6),
                      parent3 = c(0,0,0,0,3,4,0,7),
                      parent4 = c(0,0,0,0,0,0,0,1),
                      parent5 = 0)

print(pedigree)

AmatrixPolyCross(pedigree)

#when polyCross is set to be true:
#id 5 is offspring of parent 1 in a deterministic way and two other possible parents (2,3)
#id 6 is offspring of parent 2 in a deterministic way and two other possible parents (3,4)
#id 7 has two parents (deterministic case here, the parents are 3 and 4); as before
#id 8 is offspring of parent 5 in a deterministic way and has three other possible parents (6,7,1)

AmatrixPolyCross(pedigree,fixedParent=TRUE)

</code></pre>

<hr>
<h2 id='datatreat'>Organizes pedigree data in a chronological way</h2><span id='topic+datatreat'></span>

<h3>Description</h3>

<p>This function organizes pedigree data in a chronological way and return 3 lists: i) parental 1 values (numeric); ii) parental 2 values (numeric); iii) real names of the individuals. Also save a .txt file with new pedigree file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>datatreat(data = NULL, n.max = 50, unk = 0, save = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="datatreat_+3A_data">data</code></td>
<td>
<p>name of the pedigree data frame. Default=NULL.</p>
</td></tr>
<tr><td><code id="datatreat_+3A_n.max">n.max</code></td>
<td>
<p>max number of iteractions to get the chronological order. Default = 50</p>
</td></tr>
<tr><td><code id="datatreat_+3A_unk">unk</code></td>
<td>
<p>the code of the data missing. Default=0.</p>
</td></tr>
<tr><td><code id="datatreat_+3A_save">save</code></td>
<td>
<p>if TRUE, save the genealogy in a .txt file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with parental 1, parental 2, and real names of the individuals (key) also saves a txt file with the new chronological pedigree.
</p>


<h3>Author(s)</h3>

<p>Rodrigo R Amadeu, <a href="mailto:rramadeu@gmail.com">rramadeu@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ped.mrode)
datatreat(ped.mrode)

</code></pre>

<hr>
<h2 id='expandAmatrix'>Add new crosses to a current A matrix</h2><span id='topic+expandAmatrix'></span>

<h3>Description</h3>

<p>Expand a current A matrix with a new pedigree. The parents in the new pedigree should also be in the A matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expandAmatrix(newPedigree = NULL, A = NULL, returnAll = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="expandAmatrix_+3A_newpedigree">newPedigree</code></td>
<td>
<p>pedigree data name (3-column way format). Unknown value should be equal 0.</p>
</td></tr>
<tr><td><code id="expandAmatrix_+3A_a">A</code></td>
<td>
<p>numerator relationship matrix output from Amatrix function.</p>
</td></tr>
<tr><td><code id="expandAmatrix_+3A_returnall">returnAll</code></td>
<td>
<p>if TRUE returns old A with new A, if FALSE returns only new A</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix with the Relationship between the individuals.
</p>


<h3>Author(s)</h3>

<p>Rodrigo R Amadeu, <a href="mailto:rramadeu@gmail.com">rramadeu@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ped.sol)
ped.initial = ped.sol[1:1120,]
ped.new = ped.sol[-c(1:1120),]
#Computing additive relationship matrix:
A = Amatrix(ped.initial, ploidy=2)
Anew = expandAmatrix(ped.new, A)

#Comparing with one-step building..
Afull = Amatrix(ped.sol, ploidy=2)
test = Anew-Afull
which(test!=0)

</code></pre>

<hr>
<h2 id='filterpedigree'>Filter the pedigree to keep only the genealogy of a subset of individuals</h2><span id='topic+filterpedigree'></span>

<h3>Description</h3>

<p>Filter the pedigree to keep only the genealogy of a subset of individuals
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterpedigree(inds, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filterpedigree_+3A_inds">inds</code></td>
<td>
<p>vector with strings of individuals to keep their genealogy in the matrix</p>
</td></tr>
<tr><td><code id="filterpedigree_+3A_data">data</code></td>
<td>
<p>name of the pedigree data frame. Default=NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame with pedigree containing the genealogy of the selected individuals
</p>


<h3>Author(s)</h3>

<p>Rodrigo R Amadeu, <a href="mailto:rramadeu@gmail.com">rramadeu@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ped.sol)
new.ped.sol = filterpedigree(inds = c("MSW168-2","W14090-3","W14090-4"),data=ped.sol)

</code></pre>

<hr>
<h2 id='formatmatrix'>Transform a matrix in 3 columns</h2><span id='topic+formatmatrix'></span>

<h3>Description</h3>

<p>Given any square matrix transform it in a 3 columns way (row, column, value) mainly to be used in outsourcing data processing (as ASREML-standalone)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>formatmatrix(
  data = NULL,
  save = TRUE,
  return = FALSE,
  name = deparse(substitute(data)),
  round.by = 12,
  exclude.0 = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="formatmatrix_+3A_data">data</code></td>
<td>
<p>matrix (nxn).</p>
</td></tr>
<tr><td><code id="formatmatrix_+3A_save">save</code></td>
<td>
<p>if TRUE save the output in a file. Default=TRUE.</p>
</td></tr>
<tr><td><code id="formatmatrix_+3A_return">return</code></td>
<td>
<p>if TRUE return the output in a object. Default=FALSE.</p>
</td></tr>
<tr><td><code id="formatmatrix_+3A_name">name</code></td>
<td>
<p>name of the csv file to be saved. Default=data name.</p>
</td></tr>
<tr><td><code id="formatmatrix_+3A_round.by">round.by</code></td>
<td>
<p>select the number of digits after 0 you want in your data. Default = 12</p>
</td></tr>
<tr><td><code id="formatmatrix_+3A_exclude.0">exclude.0</code></td>
<td>
<p>if TRUE, remove all lines equal to zero (ASREML option). Default = TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a object or a csv file with a table with 3 columns representing the matrix.
</p>


<h3>Author(s)</h3>

<p>Rodrigo R Amadeu, <a href="mailto:rramadeu@gmail.com">rramadeu@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Example with random matrix
data&lt;-matrix(c(1,0.1,0,0.1,1,0,0,0,1.1),3)
formatmatrix(data=data,save=FALSE,return=TRUE,exclude.0=TRUE)

#Example with pedigree matrix
#Reading the example data
data(ped.mrode)
#Making Relationship Matrix
Amrode&lt;-Amatrix(ped.mrode)
#Inverting the Matrix
Amrode.inv&lt;-solve(Amrode)
#Making the 3 columns format
Amrode.inv.ASREML&lt;-formatmatrix(Amrode,save=FALSE,return=TRUE,exclude.0=TRUE)
#Printing it
Amrode.inv.ASREML 

</code></pre>

<hr>
<h2 id='Gmatrix'>Construction of Relationship Matrix G</h2><span id='topic+Gmatrix'></span>

<h3>Description</h3>

<p>Given a matrix (individual x markers), a method, a missing value, and a maf threshold, return a additive or non-additive relationship matrix. For diploids, the methods &quot;Yang&quot; and &quot;VanRaden&quot; for additive relationship matrices, and &quot;Su&quot; and &quot;Vitezica&quot; for non-additive relationship matrices are implemented. For autopolyploids, the method &quot;VanRaden&quot; for additive relationship, method &quot;Slater&quot; for full-autopolyploid model including non-additive effects, and pseudo-diploid parametrization are implemented. Weights are implemented for &quot;VanRaden&quot; method as described in Liu (2020).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Gmatrix(
  SNPmatrix = NULL,
  method = "VanRaden",
  missingValue = -9,
  maf = 0,
  thresh.missing = 0.5,
  verify.posdef = FALSE,
  ploidy = 2,
  pseudo.diploid = FALSE,
  integer = TRUE,
  ratio = FALSE,
  impute.method = "mean",
  rmv.mono = FALSE,
  thresh.htzy = 0,
  ratio.check = TRUE,
  weights = NULL,
  ploidy.correction = FALSE,
  ASV = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Gmatrix_+3A_snpmatrix">SNPmatrix</code></td>
<td>
<p>matrix (n x m), where n is is individual names and m is marker names (coded inside the matrix as 0, 1, 2, ..., ploidy, and, missingValue).</p>
</td></tr>
<tr><td><code id="Gmatrix_+3A_method">method</code></td>
<td>
<p>&quot;Yang&quot; or &quot;VanRaden&quot; for marker-based additive relationship matrix. &quot;Su&quot; or &quot;Vitezica&quot; for marker-based dominance relationship matrix. &quot;Slater&quot; for full-autopolyploid model including non-additive effects. &quot;Endelman&quot; for autotetraploid dominant (digentic) relationship matrix. &quot;MarkersMatrix&quot; for a matrix with the amount of shared markers between individuals (3). Default is &quot;VanRaden&quot;, for autopolyploids will be computed a scaled product (similar to Covarrubias-Pazaran, 2006).</p>
</td></tr>
<tr><td><code id="Gmatrix_+3A_missingvalue">missingValue</code></td>
<td>
<p>missing value in data. Default=-9.</p>
</td></tr>
<tr><td><code id="Gmatrix_+3A_maf">maf</code></td>
<td>
<p>minimum allele frequency accepted to each marker. Default=0.</p>
</td></tr>
<tr><td><code id="Gmatrix_+3A_thresh.missing">thresh.missing</code></td>
<td>
<p>threshold on missing data, SNPs below of this frequency value will be maintained, if equal to 1, no threshold and imputation is considered. Default = 0.50.</p>
</td></tr>
<tr><td><code id="Gmatrix_+3A_verify.posdef">verify.posdef</code></td>
<td>
<p>verify if the resulting matrix is positive-definite. Default=FALSE.</p>
</td></tr>
<tr><td><code id="Gmatrix_+3A_ploidy">ploidy</code></td>
<td>
<p>data ploidy (an even number between 2 and 20). Default=2.</p>
</td></tr>
<tr><td><code id="Gmatrix_+3A_pseudo.diploid">pseudo.diploid</code></td>
<td>
<p>if TRUE, uses pseudodiploid parametrization of Slater (2016).</p>
</td></tr>
<tr><td><code id="Gmatrix_+3A_integer">integer</code></td>
<td>
<p>if FALSE, not check for integer numbers. Default=TRUE.</p>
</td></tr>
<tr><td><code id="Gmatrix_+3A_ratio">ratio</code></td>
<td>
<p>if TRUE, molecular data are considered ratios and its computed the scaled product of the matrix (as in &quot;VanRaden&quot; method).</p>
</td></tr>
<tr><td><code id="Gmatrix_+3A_impute.method">impute.method</code></td>
<td>
<p>&quot;mean&quot; to impute the missing data by the mean per marker, &quot;mode&quot; to impute the missing data by the mode per marker, &quot;global.mean&quot; to impute the missing data by the mean across all markers, &quot;global.mode&quot; to impute the missing data my the mode across all marker. Default = &quot;mean&quot;.</p>
</td></tr>
<tr><td><code id="Gmatrix_+3A_rmv.mono">rmv.mono</code></td>
<td>
<p>if monomorphic markers should be removed. Default=FALSE.</p>
</td></tr>
<tr><td><code id="Gmatrix_+3A_thresh.htzy">thresh.htzy</code></td>
<td>
<p>threshold heterozigosity, remove SNPs below this threshold. Default=0.</p>
</td></tr>
<tr><td><code id="Gmatrix_+3A_ratio.check">ratio.check</code></td>
<td>
<p>if TRUE, run Mcheck with ratio data.</p>
</td></tr>
<tr><td><code id="Gmatrix_+3A_weights">weights</code></td>
<td>
<p>vector with weights for each marker. Only works if method=&quot;VanRaden&quot;. Default is a vector of 1's (equal weight).</p>
</td></tr>
<tr><td><code id="Gmatrix_+3A_ploidy.correction">ploidy.correction</code></td>
<td>
<p>It sets the denominator (correction) of the crossprod. Used only when ploidy &gt; 2 for &quot;VanRaden&quot; and ratio models. If TRUE, it uses the sum of &quot;Ploidy&quot; times &quot;Frequency&quot; times &quot;(1-Frequency)&quot; of each marker as method 1 in VanRaden 2008 and Endelman (2018). When ratio=TRUE, it uses &quot;1/Ploidy&quot; times &quot;Frequency&quot; times &quot;(1-Frequency)&quot;. If FALSE, it uses the sum of the sampling variance of each marker. Default = FALSE.</p>
</td></tr>
<tr><td><code id="Gmatrix_+3A_asv">ASV</code></td>
<td>
<p>if TRUE, transform matrix into average semivariance (ASV) equivalent (K = K / (trace(K) / (nrow(K)-1))). Details formula 2 of Fieldmann et al. (2022). Default = FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix with the marker-bases relationships between the individuals
</p>


<h3>Author(s)</h3>

<p>Rodrigo R Amadeu <a href="mailto:rramadeu@gmail.com">rramadeu@gmail.com</a>, Marcio Resende Jr, Letícia AC Lara, Ivone Oliveira, and Felipe V Ferrao
</p>


<h3>References</h3>

<p><em>Covarrubias-Pazaran, G. 2016. Genome assisted prediction of quantitative traits using the R package sommer. PLoS ONE 11(6):1-15.</em>
</p>
<p><em>Endelman, JB, et al., 2018. Genetic variance partitioning and genome-wide prediction with allele dosage information in autotetraploid potato. Genetics, 209(1) pp. 77-87.</em>
</p>
<p><em>Feldmann MJ, et al. 2022. Average semivariance directly yields accurate estimates of the genomic variance in complex trait analyses. G3 (Bethesda), 12(6).</em>
</p>
<p><em>Liu, A, et al. 2020. Weighted single-step genomic best linear unbiased prediction integrating variants selected from sequencing data by association and bioinformatics analyses. Genet Sel Evol 52, 48.</em>
</p>
<p><em>Slater, AT, et al. 2016. Improving genetic gain with genomic selection in autotetraploid potato. The Plant Genome 9(3), pp.1-15.</em>
</p>
<p><em>Su, G, et al. 2012. Estimating additive and non-additive genetic variances and predicting genetic merits using genome-wide dense single nucleotide polymorphism markers. PloS one, 7(9), p.e45293.</em>
</p>
<p><em>VanRaden, PM, 2008. Efficient methods to compute genomic predictions. Journal of dairy science, 91(11), pp.4414-4423.</em>
</p>
<p><em>Vitezica, ZG, et al. 2013. On the additive and dominant variance and covariance of individuals within the genomic selection scope. Genetics, 195(4), pp.1223-1230.</em>
</p>
<p><em>Yang, J, et al. 2010. Common SNPs explain a large proportion of the heritability for human height. Nature genetics, 42(7), pp.565-569.</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## Diploid Example
data(snp.pine)
#Verifying if data is coded as 0,1,2 and missing value.
str(snp.pine)
#Build G matrices
Gmatrix.Yang &lt;- Gmatrix(snp.pine, method="Yang", missingValue=-9, maf=0.05)
Gmatrix.VanRaden &lt;- Gmatrix(snp.pine, method="VanRaden", missingValue=-9, maf=0.05)
Gmatrix.Su &lt;- Gmatrix(snp.pine, method="Su", missingValue=-9, maf=0.05)
Gmatrix.Vitezica &lt;- Gmatrix(snp.pine, method="Vitezica", missingValue=-9, maf=0.05)

## Autetraploid example
data(snp.sol)
#Build G matrices
Gmatrix.VanRaden &lt;- Gmatrix(snp.sol, method="VanRaden", ploidy=4)
Gmatrix.Endelman &lt;- Gmatrix(snp.sol, method="Endelman", ploidy=4) 
Gmatrix.Slater &lt;- Gmatrix(snp.sol, method="Slater", ploidy=4)
Gmatrix.Pseudodiploid &lt;- Gmatrix(snp.sol, method="VanRaden", ploidy=4, pseudo.diploid=TRUE) 

#Build G matrix with weights
Gmatrix.weighted &lt;- Gmatrix(snp.sol, method="VanRaden", weights = runif(3895,0.001,0.1), ploidy=4)

## End(Not run)

</code></pre>

<hr>
<h2 id='Hmatrix'>Construction of Combined Relationship Matrix H</h2><span id='topic+Hmatrix'></span>

<h3>Description</h3>

<p>Given a matrix A and a matrix G returns a H matrix. H matrix is the relationship matrix using combined information from the pedigree and genomic relationship matrices. First, you need to compute the matrices separated and then use them as input to build the combined H matrix. 
Two methods are implemented: 'Munoz' shrinks the G matrix towards the A matrix scaling the molecular relatadness by each relationship classes; 
'Martini' is a modified version from Legarra et al. (2009) where combines A and G matrix using scaling factors. When method is equal 'Martini' and 'tau=1' and 'omega=1' you have the same H matrix as in Legarra et al. (2009).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Hmatrix(
  A = NULL,
  G = NULL,
  markers = NULL,
  c = 0,
  method = "Martini",
  tau = 1,
  omega = 1,
  missingValue = -9,
  maf = 0,
  ploidy = 2,
  roundVar = 3,
  ASV = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Hmatrix_+3A_a">A</code></td>
<td>
<p>A matrix from function Amatrix</p>
</td></tr>
<tr><td><code id="Hmatrix_+3A_g">G</code></td>
<td>
<p>G matrix from function Gmatrix</p>
</td></tr>
<tr><td><code id="Hmatrix_+3A_markers">markers</code></td>
<td>
<p>matrix marker which generated the Gmatrix</p>
</td></tr>
<tr><td><code id="Hmatrix_+3A_c">c</code></td>
<td>
<p>constant value of H computation, default: c=0</p>
</td></tr>
<tr><td><code id="Hmatrix_+3A_method">method</code></td>
<td>
<p>&quot;Martini&quot; or &quot;Munoz&quot;, default=&quot;Martini&quot;</p>
</td></tr>
<tr><td><code id="Hmatrix_+3A_tau">tau</code></td>
<td>
<p>to be used for Martini's method, default=1.</p>
</td></tr>
<tr><td><code id="Hmatrix_+3A_omega">omega</code></td>
<td>
<p>to be used of Martini's method, default=1.</p>
</td></tr>
<tr><td><code id="Hmatrix_+3A_missingvalue">missingValue</code></td>
<td>
<p>missing value in data, default=-9.</p>
</td></tr>
<tr><td><code id="Hmatrix_+3A_maf">maf</code></td>
<td>
<p>max of missing data accepted to each markerm default=0.05.</p>
</td></tr>
<tr><td><code id="Hmatrix_+3A_ploidy">ploidy</code></td>
<td>
<p>data ploidy (an even number between 2 and 20), default=2.</p>
</td></tr>
<tr><td><code id="Hmatrix_+3A_roundvar">roundVar</code></td>
<td>
<p>only used for Munoz's method, how many digits to consider the relationship be of same class, default=2.</p>
</td></tr>
<tr><td><code id="Hmatrix_+3A_asv">ASV</code></td>
<td>
<p>if TRUE, transform matrix into average semivariance (ASV) equivalent (K = K / (trace(K) / (nrow(K)-1))). Details formula 2 of Fieldmann et al. (2022). Default = FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>H Matrix with the relationship between the individuals based on pedigree and corrected by molecular information
</p>


<h3>Author(s)</h3>

<p>Rodrigo R Amadeu, <a href="mailto:rramadeu@gmail.com">rramadeu@gmail.com</a>
</p>


<h3>References</h3>

<p><em>Feldmann MJ, et al. 2022. Average semivariance directly yields accurate estimates of the genomic variance in complex trait analyses. G3 (Bethesda), 12(6).</em>
</p>
<p><em>Munoz, PR. 2014 Unraveling additive from nonadditive effects using genomic relationship matrices. Genetics 198, 1759-1768</em>
</p>
<p><em>Martini, JW, et al. 2018 The effect of the H-1 scaling factors tau and omega on the structure of H in the single-step procedure. Genetics Selection Evolution 50(1), 16</em>
</p>
<p><em>Legarra, A, et al. 2009 A relationship matrix including full pedigree and genomic information. Journal of Dairy Science 92, 4656–4663</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(ped.sol)
data(snp.sol)
#Computing the numerator relationship matrix 10% of double-reduction
Amat &lt;- Amatrix(ped.sol, ploidy=4, w = 0.1)
#Computing the additive relationship matrix based on VanRaden (modified)
Gmat &lt;- Gmatrix(snp.sol, ploidy=4, 
                maf=0.05, method="VanRaden")
Gmat &lt;- round(Gmat,3) #to be easy to invert

#Computing H matrix (Martini)
Hmat_Martini &lt;- Hmatrix(A=Amat, G=Gmat, method="Martini", 
                     ploidy=4, 
                     maf=0.05)
                     
#Computing H matrix (Munoz)
Hmat_Munoz &lt;- Hmatrix(A=Amat, G=Gmat, markers = snp.sol, 
                      ploidy=4, method="Munoz",
                      roundVar=2,
                      maf=0.05)

## End(Not run)

</code></pre>

<hr>
<h2 id='Mcheck'>Check and filter markers</h2><span id='topic+Mcheck'></span>

<h3>Description</h3>

<p>This function does different filtering on the marker matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Mcheck(
  SNPmatrix = NULL,
  ploidy = 2,
  missingValue = -9,
  thresh.maf = 0.05,
  thresh.missing = 0.9,
  thresh.htzy = 0,
  impute.method = "mean",
  rmv.mono = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Mcheck_+3A_snpmatrix">SNPmatrix</code></td>
<td>
<p>matrix (n x m), where n is is individual names and m is marker names (coded inside the matrix as 0, 1, 2, ..., ploidy, and, missingValue).</p>
</td></tr>
<tr><td><code id="Mcheck_+3A_ploidy">ploidy</code></td>
<td>
<p>data ploidy (an even number between 2 and 20). Default=2.</p>
</td></tr>
<tr><td><code id="Mcheck_+3A_missingvalue">missingValue</code></td>
<td>
<p>missing value in data. Default=-9.</p>
</td></tr>
<tr><td><code id="Mcheck_+3A_thresh.maf">thresh.maf</code></td>
<td>
<p>minimum allele frequency accepted to each marker. Default=0.05.</p>
</td></tr>
<tr><td><code id="Mcheck_+3A_thresh.missing">thresh.missing</code></td>
<td>
<p>threshold on missing data, SNPs below of this frequency value will be maintained, if equal to 1, no threshold and imputation is considered. Default = 0.50.</p>
</td></tr>
<tr><td><code id="Mcheck_+3A_thresh.htzy">thresh.htzy</code></td>
<td>
<p>threshold heterozigosity, remove SNPs below this threshold. Default=0.</p>
</td></tr>
<tr><td><code id="Mcheck_+3A_impute.method">impute.method</code></td>
<td>
<p>&quot;mean&quot; to impute the missing data by the mean per marker, &quot;mode&quot; to impute the missing data by the mode per marker, &quot;global.mean&quot; to impute the missing data by the mean across all markers, &quot;global.mode&quot; to impute the missing data my the mode across all marker. Default = &quot;mean&quot;.</p>
</td></tr>
<tr><td><code id="Mcheck_+3A_rmv.mono">rmv.mono</code></td>
<td>
<p>if monomorphic markers should be removed. Default=TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SNPmatrix after filtering steps.
</p>


<h3>Author(s)</h3>

<p>Luis F V Ferrao and Rodrigo Amadeu, <a href="mailto:rramadeu@gmail.com">rramadeu@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp.pine)
M = Mcheck(snp.pine)

</code></pre>

<hr>
<h2 id='missingdata'>Survying on missing data</h2><span id='topic+missingdata'></span>

<h3>Description</h3>

<p>This function verify which rows in a pedigree data has missing parental or conflictuos data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>missingdata(data, unk = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="missingdata_+3A_data">data</code></td>
<td>
<p>data name from a pedigree list</p>
</td></tr>
<tr><td><code id="missingdata_+3A_unk">unk</code></td>
<td>
<p>unknown value of your data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with $conflict: rows of the data which are at least one parental name equal to the individual. $missing.sire: rows of the data which arie missing data sire (Parental 1) information. $missing.dire: same as above for dire (Parental 2). $summary.missing: summary of the missing data. 2 columns, 1st for the name of the parental listed, 2nd for the how many times appeared in the data.
</p>


<h3>Author(s)</h3>

<p>Rodrigo R Amadeu, <a href="mailto:rramadeu@gmail.com">rramadeu@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ped.mrode)
missingdata(ped.mrode)

</code></pre>

<hr>
<h2 id='ped.mrode'>Pedigree Data</h2><span id='topic+ped.mrode'></span>

<h3>Description</h3>

<p>Data from pedigree example proposed by Mrode 2005
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ped.mrode)
</code></pre>


<h3>Format</h3>

<p>table
</p>


<h3>References</h3>

<p>R. A. Mrode, R. Thompson. Linear Models for the Prediction of Animal Breeding Values. CABI, 2005.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ped.mrode)

</code></pre>

<hr>
<h2 id='ped.sol'>Pedigree data for autopolyploid examples</h2><span id='topic+ped.sol'></span>

<h3>Description</h3>

<p>Dataset extract from  supplementary material from Endelman et al. (2018). 
Pedigree data frame of Potato population, missing data as 0.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ped.sol)
</code></pre>


<h3>Format</h3>

<p>data.frame
</p>


<h3>References</h3>

<p>Endelman, JB, et al., 2018 Genetic variance partitioning and genome-wide prediction with allele dosage information in autotetraploid potato. Genetics, 209(1) pp. 77-87.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ped.sol)

</code></pre>

<hr>
<h2 id='snp.pine'>Molecular data for diploid examples</h2><span id='topic+snp.pine'></span>

<h3>Description</h3>

<p>Dataset extract from  supplementary material from Resende et al. (2012). 
SNP marker matrix from Pine tree coded as 0,1, and 2, and missing value as -9.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(snp.pine)
</code></pre>


<h3>Format</h3>

<p>matrix
</p>


<h3>References</h3>

<p>Resende, MF, et al., 2012 Accuracy of genomic selection methods in a standard data set of loblolly pine (Pinus taeda l.). Genetics 190: 1503–1510.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp.pine)

</code></pre>

<hr>
<h2 id='snp.sol'>Molecular data for autopolyploid examples</h2><span id='topic+snp.sol'></span>

<h3>Description</h3>

<p>Dataset extract from  supplementary material from Endelman et al. (2018). 
SNP marker matrix from Pine tree coded as 0,1,2,3,4 and missing value as -9.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(snp.sol)
</code></pre>


<h3>Format</h3>

<p>data.frame
</p>


<h3>References</h3>

<p>Endelman, JB, et al., 2018 Genetic variance partitioning and genome-wide prediction with allele dosage information in autotetraploid potato. Genetics, 209(1) pp. 77-87.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp.sol)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
