<!DOCTYPE html><html lang="en"><head><title>Help for package agricolae</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {agricolae}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#agricolae-package'>
<p>Statistical Procedures for Agricultural Research</p></a></li>
<li><a href='#AMMI'><p> AMMI Analysis</p></a></li>
<li><a href='#AMMI.contour'><p> AMMI contour</p></a></li>
<li><a href='#audpc'><p> Calculating the absolute or relative value of the AUDPC</p></a></li>
<li><a href='#audps'><p> The Area Under the Disease Progress Stairs</p></a></li>
<li><a href='#bar.err'><p> Plotting the standard error or standard deviance of a multiple</p>
comparison of means</a></li>
<li><a href='#bar.group'><p> Plotting the multiple comparison of means</p></a></li>
<li><a href='#BIB.test'><p> Finding the Variance Analysis of the Balanced Incomplete Block Design</p></a></li>
<li><a href='#carolina'><p> North Carolina Designs I, II and III</p></a></li>
<li><a href='#Chz2006'><p> Data amendment Carhuaz 2006</p></a></li>
<li><a href='#CIC'><p> Data for late blight of potatoes</p></a></li>
<li><a href='#clay'><p> Data of Ralstonia population in clay soil</p></a></li>
<li><a href='#ComasOxapampa'><p> Data AUDPC Comas - Oxapampa</p></a></li>
<li><a href='#consensus'><p> consensus of clusters</p></a></li>
<li><a href='#corn'><p> Data of corn</p></a></li>
<li><a href='#correl'><p> Correlation Coefficient</p></a></li>
<li><a href='#correlation'><p> Correlation analysis. Methods of Pearson, Spearman, Kendall and Lin</p></a></li>
<li><a href='#cotton'><p> Data of cotton</p></a></li>
<li><a href='#cv.model'><p> Coefficient of the experiment variation</p></a></li>
<li><a href='#cv.similarity'><p> Coefficient of the similarity matrix variation</p></a></li>
<li><a href='#DAU.test'><p> Finding the Variance Analysis of the Augmented block Design</p></a></li>
<li><a href='#DC'><p> Data for the analysis of carolina genetic design</p></a></li>
<li><a href='#delete.na'><p> Omitting the rows or columns with missing observations of a matrix (NA)</p></a></li>
<li><a href='#design.ab'><p> Design of experiments for a factorial</p></a></li>
<li><a href='#design.alpha'><p> Alpha design type (0,1)</p></a></li>
<li><a href='#design.bib'><p> Randomized Balanced Incomplete Block Designs. BIB</p></a></li>
<li><a href='#design.crd'><p> Completely Randomized Design</p></a></li>
<li><a href='#design.cyclic'><p> Cyclic designs</p></a></li>
<li><a href='#design.dau'><p> Augmented block design</p></a></li>
<li><a href='#design.graeco'><p> Graeco - latin square design</p></a></li>
<li><a href='#design.lattice'><p> Lattice designs</p></a></li>
<li><a href='#design.lsd'><p> Latin Square Design</p></a></li>
<li><a href='#design.mat'><p> Experimental design matrix</p></a></li>
<li><a href='#design.rcbd'><p> Randomized Complete Block Design</p></a></li>
<li><a href='#design.split'><p> Split Plot Design</p></a></li>
<li><a href='#design.strip'><p> Strip Plot Design</p></a></li>
<li><a href='#design.youden'><p> Incomplete Latin Square Design</p></a></li>
<li><a href='#diffograph'><p> Plotting the multiple comparison of means</p></a></li>
<li><a href='#disease'><p> Data evaluation of the disease overtime</p></a></li>
<li><a href='#duncan.test'><p> Duncan's new multiple range test</p></a></li>
<li><a href='#durbin.test'><p> Durbin test and multiple comparison of treatments</p></a></li>
<li><a href='#friedman'><p> Friedman test and multiple comparison of treatments</p></a></li>
<li><a href='#frijol'><p> Data of frijol</p></a></li>
<li><a href='#genxenv'><p> Data of potato yield in a different environment</p></a></li>
<li><a href='#Glycoalkaloids'><p> Data Glycoalkaloids</p></a></li>
<li><a href='#graph.freq'><p> Histogram</p></a></li>
<li><a href='#grass'><p> Data for Friedman test</p></a></li>
<li><a href='#greenhouse'><p> Data in greenhouse</p></a></li>
<li><a href='#growth'><p> Data growth of trees</p></a></li>
<li><a href='#haynes'><p> Data of AUDPC for nonparametrical stability analysis</p></a></li>
<li><a href='#Hco2006'><p> Data amendment Huanuco 2006</p></a></li>
<li><a href='#hcut'><p> Cut tree of consensus</p></a></li>
<li><a href='#heterosis'><p> Data of potato, Heterosis</p></a></li>
<li><a href='#hgroups'><p> groups of hclust</p></a></li>
<li><a href='#HSD.test'><p> Multiple comparisons: Tukey</p></a></li>
<li><a href='#huasahuasi'><p> Data: Rainfall thresholds as support for timing fungicide applications in the control of potato late blight in Peru</p></a></li>
<li><a href='#index.AMMI'><p> AMMI index and yield stability</p></a></li>
<li><a href='#index.bio'><p> Biodiversity Index</p></a></li>
<li><a href='#index.smith'><p> Uniformity soil. Smith's Index of Soil Heterogeneity</p></a></li>
<li><a href='#inter.freq'><p> Class intervals</p></a></li>
<li><a href='#join.freq'><p> Join class for histogram</p></a></li>
<li><a href='#kendall'><p> Correlation of Kendall</p></a></li>
<li><a href='#kruskal'><p> Kruskal Wallis test and multiple comparison of treatments.</p></a></li>
<li><a href='#kurtosis'><p> Finding the Kurtosis coefficient</p></a></li>
<li><a href='#lastC'><p> Setting the last character of a chain</p></a></li>
<li><a href='#lateblight'>
<p>LATEBLIGHT - Simulator for potato late blight Version LB2004</p></a></li>
<li><a href='#lineXtester'><p> Line x Tester Analysis</p></a></li>
<li><a href='#LSD.test'><p> Multiple comparisons, &quot;Least significant difference&quot; and Adjust P-values</p></a></li>
<li><a href='#LxT'><p> Data Line by tester</p></a></li>
<li><a href='#markers'><p> Data of molecular markers</p></a></li>
<li><a href='#Median.test'><p> Median test. Multiple comparisons</p></a></li>
<li><a href='#melon'><p> Data of yield of melon in a Latin square experiment</p></a></li>
<li><a href='#montecarlo'><p> Random generation by Montecarlo</p></a></li>
<li><a href='#natives'><p> Data of native potato</p></a></li>
<li><a href='#nonadditivity'><p> Nonadditivity model test</p></a></li>
<li><a href='#normal.freq'><p> Normal curve on the histogram</p></a></li>
<li><a href='#ogive.freq'><p> Plotting the ogive from a histogram</p></a></li>
<li><a href='#order.group'><p> Ordering the treatments according to the multiple comparison</p></a></li>
<li><a href='#orderPvalue'><p> Grouping the treatments averages in a comparison with a minimum value</p></a></li>
<li><a href='#pamCIP'><p> Data Potato Wild</p></a></li>
<li><a href='#paracsho'><p> Data of Paracsho biodiversity</p></a></li>
<li><a href='#path.analysis'><p> Path Analysis</p></a></li>
<li><a href='#PBIB.test'><p> Analysis of the Partially Balanced Incomplete Block Design</p></a></li>
<li><a href='#plot.AMMI'><p> PLOT AMMI</p></a></li>
<li><a href='#plot.graph.freq'><p> Histogram</p></a></li>
<li><a href='#plot.group'><p> Plotting the multiple comparison of means</p></a></li>
<li><a href='#plots'><p> Data for an analysis in split-plot</p></a></li>
<li><a href='#plrv'><p> Data clones from the PLRV population</p></a></li>
<li><a href='#polygon.freq'><p> The polygon of frequency on the histogram</p></a></li>
<li><a href='#potato'><p> Data of cutting</p></a></li>
<li><a href='#ralstonia'><p> Data of assessment of the population in the soil R.solanacearum</p></a></li>
<li><a href='#reg.homog'><p> Homologation of regressions</p></a></li>
<li><a href='#REGW.test'><p> Ryan, Einot and Gabriel and Welsch multiple range test</p></a></li>
<li><a href='#resampling.cv'><p> Resampling to find the optimal number of markers</p></a></li>
<li><a href='#resampling.model'><p> Resampling for linear models</p></a></li>
<li><a href='#rice'><p> Data of Grain yield of rice variety IR8</p></a></li>
<li><a href='#RioChillon'><p> Data and analysis Mother and baby trials</p></a></li>
<li><a href='#scheffe.test'><p> Multiple comparisons, scheffe</p></a></li>
<li><a href='#similarity'><p> Matrix of similarity in binary data</p></a></li>
<li><a href='#simulation.model'><p> Simulation of the linear model under normality</p></a></li>
<li><a href='#sinRepAmmi'><p> Data for AMMI without repetition</p></a></li>
<li><a href='#skewness'><p> Finding the skewness coefficient</p></a></li>
<li><a href='#SNK.test'><p> Student-Newman-Keuls (SNK)</p></a></li>
<li><a href='#soil'><p> Data of soil analysis for 13 localities</p></a></li>
<li><a href='#sp.plot'><p> Splip-Plot analysis</p></a></li>
<li><a href='#ssp.plot'><p> Split-split-Plot analysis</p></a></li>
<li><a href='#stability.nonpar'><p> Nonparametric stability analysis</p></a></li>
<li><a href='#stability.par'><p> Stability analysis. SHUKLA'S STABILITY VARIANCE AND KANG'S</p></a></li>
<li><a href='#stat.freq'><p> Descriptive measures of grouped data</p></a></li>
<li><a href='#strip.plot'><p> Strip-Plot analysis</p></a></li>
<li><a href='#sturges.freq'><p> Class intervals for a histogram, the rule of Sturges</p></a></li>
<li><a href='#summary.graph.freq'><p> frequency Table of a Histogram</p></a></li>
<li><a href='#sweetpotato'><p> Data of sweetpotato yield</p></a></li>
<li><a href='#table.freq'><p> frequency Table of a Histogram</p></a></li>
<li><a href='#tapply.stat'><p> Statistics of data grouped by factors</p></a></li>
<li><a href='#vark'><p> Variance K, ties, Kendall</p></a></li>
<li><a href='#waerden.test'><p> Multiple comparisons. The van der Waerden (Normal Scores)</p></a></li>
<li><a href='#waller'><p> Computations of Bayesian t-values for multiple comparisons</p></a></li>
<li><a href='#waller.test'><p> Multiple comparisons, Waller-Duncan</p></a></li>
<li><a href='#weatherSeverity'>
<p>Weather and Severity</p></a></li>
<li><a href='#wilt'><p> Data of Bacterial Wilt (AUDPC) and soil</p></a></li>
<li><a href='#yacon'><p> Data Yacon</p></a></li>
<li><a href='#zigzag'><p> order plot in serpentine</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Statistical Procedures for Agricultural Research</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3-7</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-10-22</td>
</tr>
<tr>
<td>Author:</td>
<td>Felipe de Mendiburu</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Felipe de Mendiburu &lt;fmendiburu@lamolina.edu.pe&gt;</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS, nlme, cluster, AlgDesign, graphics</td>
</tr>
<tr>
<td>Description:</td>
<td>Original idea was presented in the thesis "A statistical analysis tool for agricultural research" to obtain the degree of Master on science, National Engineering University (UNI), Lima-Peru. Some experimental data for the examples come from the CIP and others research. Agricolae offers extensive functionality on experimental design especially for agricultural and plant breeding experiments, which can also be useful for other purposes. It supports planning of lattice, Alpha, Cyclic, Complete Block, Latin Square, Graeco-Latin Squares, augmented block, factorial, split and strip plot designs. There are also various analysis facilities for experimental data, e.g. treatment comparison procedures and several non-parametric tests comparison, biodiversity indexes and consensus cluster.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-22 20:25:25 UTC; felipe</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-22 21:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='agricolae-package'>
Statistical Procedures for Agricultural Research
</h2><span id='topic+agricolae-package'></span><span id='topic+agricolae'></span>

<h3>Description</h3>

<p>This package contains functionality for the Statistical Analysis of experimental
designs applied specially for field experiments in agriculture and plant breeding.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> agricolae</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.3-7</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2023-10-22</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Planning of field experiments: lattice, factorial, RCBD, CRD, Latin Square,
Youden, Graeco, BIB, Alpha design, Cyclic, augmented block, split and strip plot
Designs. Comparison of multi-location trials: AMMI, Index AMMI Stability
(biplot, triplot), comparison between treatments: LSD, Bonferroni and
other p-adjust, HSD, Waller, Student Newman Keuls SNK, Duncan, REGW, Scheffe;
Non parametric tests: Kruskal, Friedman, Durbin, Van Der Waerden, Median.
Analysis of genetic experiments: North Carolina designs, LinexTester, 
Balanced Incomplete Block, Strip plot, Split-Plot, Partially Balanced Incomplete 
Block, analysis Mother and baby trials (see data RioChillon).
Resampling and simulation: resampling.model, simulation.model, montecarlo,
lateblight Simulator for potato. Ecology: Biodiversity Index, Path Analysis.
Soil Uniformity: Smith's Index. Cluster Analysis: Consensus Cluster.
Descriptive statistics utilities: *.freq
</p>


<h3>Author(s)</h3>

<p>Felipe de Mendiburu
Statistical Engineer
Master in Systems Engineering
Professor of Applied Statistics
</p>
<p>Maintainer: Felipe de Mendiburu &lt;fmendiburu@lamolina.edu.pe&gt;
</p>


<h3>References</h3>

<p>De Mendiburu, Felipe (2009). Una herramienta de analisis estadistico para la
investigacion agricola. Tesis. Universidad Nacional de Ingenieria (UNI-PERU).
</p>
<p>Universidad Nacional Agraria La Molina, Lima-PERU.
Facultad de Economia y Planificacion
Departamento Academico de Estadistica e Informatica
</p>

<hr>
<h2 id='AMMI'> AMMI Analysis</h2><span id='topic+AMMI'></span>

<h3>Description</h3>

<p>Additive Main Effects and Multiplicative Interaction Models (AMMI) are widely used 
to analyze main effects and genotype by environment (GEN, ENV) interactions in 
multilocation variety trials. Furthermore, this function generates data to biplot, triplot 
graphs and analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AMMI(ENV, GEN, REP, Y, MSE = 0,console=FALSE,PC=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AMMI_+3A_env">ENV</code></td>
<td>
<p> Environment </p>
</td></tr>
<tr><td><code id="AMMI_+3A_gen">GEN</code></td>
<td>
<p> Genotype </p>
</td></tr>
<tr><td><code id="AMMI_+3A_rep">REP</code></td>
<td>
<p> Replication </p>
</td></tr>
<tr><td><code id="AMMI_+3A_y">Y</code></td>
<td>
<p> Response </p>
</td></tr>
<tr><td><code id="AMMI_+3A_mse">MSE</code></td>
<td>
<p> Mean Square Error </p>
</td></tr>
<tr><td><code id="AMMI_+3A_console">console</code></td>
<td>
<p> ouput TRUE or FALSE </p>
</td></tr>
<tr><td><code id="AMMI_+3A_pc">PC</code></td>
<td>
<p> Principal components ouput TRUE or FALSE </p>
</td></tr>
</table>


<h3>Details</h3>

<p>additional graphics see help(plot.AMMI).
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>ANOVA</code></td>
<td>
<p>analysis of variance general</p>
</td></tr>
<tr><td><code>genXenv</code></td>
<td>
<p>class by, genopyte and environment</p>
</td></tr>
<tr><td><code>analysis</code></td>
<td>
<p>analysis of variance principal components</p>
</td></tr>
<tr><td><code>means</code></td>
<td>
<p>average genotype and environment</p>
</td></tr>
<tr><td><code>biplot</code></td>
<td>
<p>data to produce graphics</p>
</td></tr>
<tr><td><code>PC</code></td>
<td>
<p> class princomp </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> F. de Mendiburu </p>


<h3>References</h3>

 
<p>Crossa, J. 1990. Statistical analysis of multilocation trials.
Advances in Agronomy 44:55-85
</p>


<h3>See Also</h3>

<p><code><a href="#topic+lineXtester">lineXtester</a></code>,<code><a href="#topic+plot.AMMI">plot.AMMI</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'># Full replications
library(agricolae)
# Example 1
data(plrv)
model&lt;- with(plrv,AMMI(Locality, Genotype, Rep, Yield, console=FALSE))
model$ANOVA
# see help(plot.AMMI)
# biplot
plot(model)
# biplot PC1 vs Yield 
plot(model, first=0,second=1, number=TRUE)
# Example 2
data(CIC)
data1&lt;-CIC$comas[,c(1,6,7,17,18)]
data2&lt;-CIC$oxapampa[,c(1,6,7,19,20)]
cic &lt;- rbind(data1,data2)
model&lt;-with(cic,AMMI(Locality, Genotype, Rep, relative))
model$ANOVA
plot(model,0,1,angle=20,ecol="brown")
# Example 3
# Only means. Mean square error is well-known.
data(sinRepAmmi)
REP &lt;- 3
MSerror &lt;- 93.24224
#startgraph
model&lt;-with(sinRepAmmi,AMMI(ENV, GEN, REP, YLD, MSerror,PC=TRUE))
# print anova
print(model$ANOVA,na.print = "")
# Biplot with the one restored observed.
plot(model,0,1)
# with principal components model$PC is class "princomp" 
pc&lt;- model$PC
pc$loadings
summary(pc)
biplot(pc)
# Principal components by means of the covariance similar AMMI
# It is to compare results with AMMI
cova&lt;-cov(model$genXenv)
values&lt;-eigen(cova)
total&lt;-sum(values$values)
round(values$values*100/total,2)
# AMMI: 64.81 18.58 13.50  3.11  0.00
</code></pre>

<hr>
<h2 id='AMMI.contour'> AMMI contour </h2><span id='topic+AMMI.contour'></span>

<h3>Description</h3>

<p>Draws a polygon or a circumference around the center of the Biplot with a 
proportional radio at the longest distance of the genotype.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AMMI.contour(model, distance, shape, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AMMI.contour_+3A_model">model</code></td>
<td>
<p> Object </p>
</td></tr>
<tr><td><code id="AMMI.contour_+3A_distance">distance</code></td>
<td>
<p> Circumference radius &gt;0 and &lt;=1 </p>
</td></tr>
<tr><td><code id="AMMI.contour_+3A_shape">shape</code></td>
<td>
<p> Numerical, relating to the shape of the polygon outline. </p>
</td></tr>
<tr><td><code id="AMMI.contour_+3A_...">...</code></td>
<td>
<p> Parameters corresponding to the R lines function </p>
</td></tr>
</table>


<h3>Details</h3>

<p>First, it is necessary to execute the AMMI function.
It is only valid for the BIPLOT function but not for the TRIPLOT one.
</p>


<h3>Value</h3>

<p>Genotypes within and outside the area.
</p>
<table role = "presentation">
<tr><td><code>distance</code></td>
<td>
<p>Distance from genotype to origin (0,0) </p>
</td></tr>
</table>


<h3>Note</h3>

 
<p>Complement graphics AMMI 
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+AMMI">AMMI</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
# see AMMI.
data(sinRepAmmi)
Environment &lt;- sinRepAmmi$ENV
Genotype &lt;- sinRepAmmi$GEN
Yield &lt;- sinRepAmmi$YLD
REP &lt;- 3
MSerror &lt;- 93.24224
model&lt;-AMMI(Environment, Genotype, REP, Yield, MSerror)
plot(model)
AMMI.contour(model,distance=0.7,shape=8,col="red",lwd=2,lty=5)
</code></pre>

<hr>
<h2 id='audpc'> Calculating the absolute or relative value of the AUDPC </h2><span id='topic+audpc'></span>

<h3>Description</h3>

<p>Area Under Disease Progress Curve. The AUDPC measures the disease throughout a 
period. The AUDPC is the area that is determined by the sum of trapezes under 
the curve.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>audpc(evaluation, dates, type = "absolute")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="audpc_+3A_evaluation">evaluation</code></td>
<td>
<p> Table of data of the evaluations: Data frame </p>
</td></tr>
<tr><td><code id="audpc_+3A_dates">dates</code></td>
<td>
<p> Vector of dates corresponding to each evaluation </p>
</td></tr>
<tr><td><code id="audpc_+3A_type">type</code></td>
<td>
<p> relative, absolute </p>
</td></tr>
</table>


<h3>Details</h3>

<p>AUDPC. For the illustration one considers three evaluations
(14, 21 and 28 days) and percentage of damage in the plant 40, 80 and 90
(interval between dates of evaluation 7 days).  AUDPC = 1045.
The evaluations can be at different interval.
</p>


<h3>Value</h3>

<p>Vector with relative or absolute audpc.
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Campbell, C. L., L. V. Madden. (1990): Introduction to Plant Disease Epidemiology. John Wiley &amp; Sons, New York City. </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
dates&lt;-c(14,21,28) # days
# example 1: evaluation - vector
evaluation&lt;-c(40,80,90)
audpc(evaluation,dates)
# example 2: evaluation: dataframe nrow=1
evaluation&lt;-data.frame(E1=40,E2=80,E3=90) # percentages
plot(dates,evaluation,type="h",ylim=c(0,100),col="red",axes=FALSE)
title(cex.main=0.8,main="Absolute or Relative AUDPC\nTotal area = 100*(28-14)=1400")
lines(dates,evaluation,col="red")
text(dates,evaluation+5,evaluation)
text(18,20,"A = (21-14)*(80+40)/2")
text(25,60,"B = (28-21)*(90+80)/2")
text(25,40,"audpc = A+B = 1015")
text(24.5,33,"relative = audpc/area = 0.725")
abline(h=0)
axis(1,dates)
axis(2,seq(0,100,5),las=2)
lines(rbind(c(14,40),c(14,100)),lty=8,col="green")
lines(rbind(c(14,100),c(28,100)),lty=8,col="green")
lines(rbind(c(28,90),c(28,100)),lty=8,col="green")
# It calculates audpc absolute
absolute&lt;-audpc(evaluation,dates,type="absolute")
print(absolute)
rm(evaluation, dates, absolute)
# example 3: evaluation dataframe nrow&gt;1
data(disease)
dates&lt;-c(1,2,3) # week
evaluation&lt;-disease[,c(4,5,6)]
# It calculates audpc relative
index &lt;-audpc(evaluation, dates, type = "relative")
# Correlation between the yield and audpc
correlation(disease$yield, index, method="kendall")
# example 4: days infile
data(CIC)
comas &lt;- CIC$comas
oxapampa &lt;- CIC$oxapampa
dcomas &lt;- names(comas)[9:16]
days&lt;- as.numeric(substr(dcomas,2,3))
AUDPC&lt;- audpc(comas[,9:16],days)
relative&lt;-audpc(comas[,9:16],days,type = "relative")
h1&lt;-graph.freq(AUDPC,border="red",density=4,col="blue")
table.freq(h1)
h2&lt;-graph.freq(relative,border="red",density=4,col="blue",
frequency=2, ylab="relative frequency")
</code></pre>

<hr>
<h2 id='audps'> The Area Under the Disease Progress Stairs </h2><span id='topic+audps'></span>

<h3>Description</h3>

<p>A better estimate of disease progress is the area under the disease progress 
stairs (AUDPS). The AUDPS approach improves the estimation of disease progress 
by giving a weight closer to optimal to the first and last observations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>audps(evaluation, dates, type = "absolute")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="audps_+3A_evaluation">evaluation</code></td>
<td>
<p> Table of data of the evaluations: Data frame </p>
</td></tr>
<tr><td><code id="audps_+3A_dates">dates</code></td>
<td>
<p> Vector of dates corresponding to each evaluation </p>
</td></tr>
<tr><td><code id="audps_+3A_type">type</code></td>
<td>
<p> relative, absolute </p>
</td></tr>
</table>


<h3>Details</h3>

<p>AUDPS. For the illustration one considers three evaluations
(14, 21 and 28 days) and percentage of damage in the plant 40, 80 and 90
(interval between dates of evaluation 7 days).  AUDPS = 1470.
The evaluations can be at different interval.
AUDPS= sum( rectangle area by interval in times evaluation ) see example.
</p>


<h3>Value</h3>

<p>Vector with relative or absolute audps.
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Ivan Simko, and Hans-Peter Piepho, (2012). The area under the disease 
progress stairs: Calculation, advantage, and application. Phytopathology 102:381- 389.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
dates&lt;-c(14,21,28) # days
# example 1: evaluation - vector
evaluation&lt;-c(40,80,90)
audps(evaluation,dates)
audps(evaluation,dates,"relative")
x&lt;-seq(10.5,31.5,7)
y&lt;-c(40,80,90,90)
plot(x,y,"s",ylim=c(0,100),xlim=c(10,32),axes=FALSE,col="red" ,ylab="",xlab="")
title(cex.main=0.8,main="Absolute or Relative AUDPS\nTotal area=(31.5-10.5)*100=2100",
ylab="evaluation",xlab="dates" )
points(x,y,type="h")
z&lt;-c(14,21,28)
points(z,y[-3],col="blue",lty=2,pch=19)
points(z,y[-3],col="blue",lty=2,pch=19)
axis(1,x,pos=0)
axis(2,c(0,40,80,90,100),las=2)
text(dates,evaluation+5,dates,col="blue")
text(14,20,"A = (17.5-10.5)*40",cex=0.8)
text(21,40,"B = (24.5-17.5)*80",cex=0.8)
text(28,60,"C = (31.5-24.5)*90",cex=0.8)
text(14,95,"audps = A+B+C = 1470")
text(14,90,"relative = audps/area = 0.7")
# It calculates audpc absolute
absolute&lt;-audps(evaluation,dates,type="absolute")
print(absolute)
rm(evaluation, dates, absolute)
</code></pre>

<hr>
<h2 id='bar.err'> Plotting the standard error or standard deviance of a multiple 
comparison of means</h2><span id='topic+bar.err'></span>

<h3>Description</h3>

<p>It plots bars of the averages of treatments and standard error or standard 
deviance. It uses the objects generated by a procedure of comparison like LSD,
HSD, Kruskal and Waller-Duncan. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bar.err(x,variation=c("SE","SD","range","IQR"),horiz=FALSE, bar=TRUE,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bar.err_+3A_x">x</code></td>
<td>
<p> object means of the comparisons the LSD.test, HSD.test,...,etc </p>
</td></tr>
<tr><td><code id="bar.err_+3A_variation">variation</code></td>
<td>
<p> SE=standard error, range=Max-Min or IQR=interquartil range</p>
</td></tr>
<tr><td><code id="bar.err_+3A_horiz">horiz</code></td>
<td>
<p> Horizontal or vertical bars </p>
</td></tr>
<tr><td><code id="bar.err_+3A_bar">bar</code></td>
<td>
<p>paint bar</p>
</td></tr>
<tr><td><code id="bar.err_+3A_...">...</code></td>
<td>
<p> Parameters of the function barplot()</p>
</td></tr>
</table>


<h3>Details</h3>

<p>x: data frame formed by 5 columns: name of the bars, height, level
out: LSD.test, HSD, waller.test, scheffe.test, duncan.test, SNK.test, 
friedman, kruskal, waerden.test and Median.test. 
</p>


<h3>Value</h3>

<p>A list with numeric vectors giving the coordinates of all the bar midpoints drawn.
</p>
<table role = "presentation">
<tr><td><code>x</code></td>
<td>
<p> eje-1 coordinate </p>
</td></tr>
<tr><td><code>height</code></td>
<td>
<p> eje-2 coordinate by group</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+LSD.test">LSD.test</a></code>, <code><a href="#topic+HSD.test">HSD.test</a></code>, 
<code><a href="#topic+waller.test">waller.test</a></code>, <code><a href="#topic+kruskal">kruskal</a></code>, <code><a href="#topic+bar.group">bar.group</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(sweetpotato)
model&lt;-aov(yield~virus,data=sweetpotato)
out &lt;- waller.test(model,"virus", console=TRUE,
main="Yield of sweetpotato\ndealt with different virus")
oldpar&lt;-par(mfrow=c(2,2),cex=1)
bar.err(out$means,variation="range",horiz=TRUE,xlim=c(0,45),angle=125,density=6,
 main="range")
bar.err(out$means,variation="SD",ylim=c(0,45),col=colors()[30],
 main="Standard deviation",density=8)
bar.err(out$means,variation="SE",horiz=TRUE,xlim=c(0,45),density=8,
 col="brown",main="Standard error")
bar.err(out$means,variation="range",ylim=c(0,45),bar=FALSE,col="green",
 main="range")
par(oldpar)
oldpar&lt;-par(mfrow=c(1,2),cex=1)
bar.err(out$means,variation="range",ylim=c(0,45),bar=FALSE,col=0)
abline(h=0)
# horiz = TRUE
bar.err(out$means,variation="SE",horiz=TRUE,xlim=c(0,45),bar=FALSE,col=0)
#startgraph
par(oldpar)
#endgraph
</code></pre>

<hr>
<h2 id='bar.group'> Plotting the multiple comparison of means </h2><span id='topic+bar.group'></span>

<h3>Description</h3>

<p>It plots bars of the averages of treatments to compare.
It uses the objects generated by a procedure of comparison like LSD, HSD,
Kruskall, Waller-Duncan, Friedman or Durbin. It can also display the 'average' 
value over each bar in a bar chart.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bar.group(x,horiz=FALSE, decreasing=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bar.group_+3A_x">x</code></td>
<td>
<p> Object created by a test of comparison </p>
</td></tr>
<tr><td><code id="bar.group_+3A_horiz">horiz</code></td>
<td>
<p> Horizontal or vertical bars </p>
</td></tr>
<tr><td><code id="bar.group_+3A_decreasing">decreasing</code></td>
<td>
<p>Logical, decreasing order of the mean</p>
</td></tr>
<tr><td><code id="bar.group_+3A_...">...</code></td>
<td>
<p> Parameters of the function barplot() </p>
</td></tr> 
</table>


<h3>Details</h3>

<p>x: data frame formed by 5 columns: name of the bars, height and level of the bar.
</p>


<h3>Value</h3>

<p>A list with numeric vectors giving the coordinates of all the bar midpoints drawn.
</p>
<table role = "presentation">
<tr><td><code>x</code></td>
<td>
<p> eje-1 coordinate </p>
</td></tr>
<tr><td><code>height</code></td>
<td>
<p> eje-2 coordinate by group</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Meniburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+LSD.test">LSD.test</a></code>, <code><a href="#topic+HSD.test">HSD.test</a></code>, <code><a href="#topic+kruskal">kruskal</a></code>
, <code><a href="#topic+friedman">friedman</a></code>, <code><a href="#topic+durbin.test">durbin.test</a></code>, <code><a href="#topic+waller.test">waller.test</a></code> 
, <code><a href="#topic+plot.group">plot.group</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1
library(agricolae)
data(sweetpotato)
model&lt;-aov(yield~virus,data=sweetpotato)
comparison&lt;- LSD.test(model,"virus",alpha=0.01,group=TRUE)
print(comparison$groups)
oldpar&lt;-par(cex=1.5)
bar.group(comparison$groups,horiz=TRUE,density=8,col="blue",border="red", xlim=c(0,50),las=1)
title(cex.main=0.8,main="Comparison between\ntreatment means",xlab="Yield",ylab="Virus")
# Example 2
library(agricolae)
x &lt;- 1:4
y &lt;- c(0.29, 0.44, 0.09, 0.49)
xy &lt;- data.frame(x,y,y)
par(oldpar)
oldpar&lt;-par(cex=1.5)
bar.group(xy,density=30,angle=90,col="brown",border=FALSE,ylim=c(0,0.6),lwd=2,las=1)
par(oldpar)
</code></pre>

<hr>
<h2 id='BIB.test'> Finding the Variance Analysis of the Balanced Incomplete Block Design  </h2><span id='topic+BIB.test'></span>

<h3>Description</h3>

<p>Analysis of variance BIB and comparison mean adjusted. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BIB.test(block, trt, y, test = c("lsd","tukey","duncan","waller","snk"), 
alpha = 0.05, group = TRUE,console=FALSE)

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BIB.test_+3A_block">block</code></td>
<td>
<p> blocks </p>
</td></tr>
<tr><td><code id="BIB.test_+3A_trt">trt</code></td>
<td>
<p> Treatment </p>
</td></tr>
<tr><td><code id="BIB.test_+3A_y">y</code></td>
<td>
<p> Response  </p>
</td></tr>
<tr><td><code id="BIB.test_+3A_test">test</code></td>
<td>
<p> Comparison treatments </p>
</td></tr>
<tr><td><code id="BIB.test_+3A_alpha">alpha</code></td>
<td>
<p> Significant test </p>
</td></tr>
<tr><td><code id="BIB.test_+3A_group">group</code></td>
<td>
<p>logical </p>
</td></tr>
<tr><td><code id="BIB.test_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Test of comparison treatment. 
lsd: Least significant difference.
tukey: Honestly significant differente.
duncan: Duncan's new multiple range test
waller: Waller-Duncan test.
snk: Student-Newman-Keuls (SNK)
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>statistics</code></td>
<td>
<p>Statistics of the model</p>
</td></tr>
<tr><td><code>comparison</code></td>
<td>
<p>Comparison between treatments</p>
</td></tr>
<tr><td><code>means</code></td>
<td>
<p>Adjusted mean and statistics summary</p>
</td></tr>
<tr><td><code>groups</code></td>
<td>
<p>Grouping of treatments</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> F. de Mendiburu </p>


<h3>References</h3>

<p> Design of Experiments. Robert O. Kuehl. 2nd ed., Duxbury, 2000 
Linear Estimation and Design of Experiments. D.D. Joshi. WILEY EASTERN LIMITED
1987, New Delhi, India. Introduction to experimental statistics. Ching Chun Li
McGraw - Hill Book Company, Inc. New York. 1964 </p>


<h3>See Also</h3>

<p><code><a href="#topic+DAU.test">DAU.test</a></code>, <code><a href="#topic+duncan.test">duncan.test</a></code>, <code><a href="#topic+durbin.test">durbin.test</a></code>, 
<code><a href="#topic+friedman">friedman</a></code>, <code><a href="#topic+HSD.test">HSD.test</a></code>, <code><a href="#topic+kruskal">kruskal</a></code>, 
<code><a href="#topic+LSD.test">LSD.test</a></code>, <code><a href="#topic+Median.test">Median.test</a></code>, <code><a href="#topic+PBIB.test">PBIB.test</a></code>, 
<code><a href="#topic+REGW.test">REGW.test</a></code>, <code><a href="#topic+scheffe.test">scheffe.test</a></code>, <code><a href="#topic+SNK.test">SNK.test</a></code>,
<code><a href="#topic+waerden.test">waerden.test</a></code>, <code><a href="#topic+waller.test">waller.test</a></code>, <code><a href="#topic+plot.group">plot.group</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
# Example Design of Experiments. Robert O. Kuehl. 2da. Edicion. 2001
run&lt;-gl(10,3)
psi&lt;-c(250,325,475,250,475,550,325,400,550,400,475,550,325,475,550,
250,400,475,250,325,400,250,400,550,250,325,550,325,400,475)
monovinyl&lt;-c(16,18,32,19,46,45,26,39,61,21,35,55,19,47,48,20,33,31,13,13,34,21,
 30,52,24,10,50,24,31,37)
out&lt;-BIB.test(run,psi,monovinyl,test="waller",group=FALSE)
print(out)
bar.err(out$means,variation="range",ylim=c(0,60),bar=FALSE,col=0)
out&lt;-BIB.test(run,psi,monovinyl,test="waller",group=TRUE)
out&lt;-BIB.test(run,psi,monovinyl,test="tukey",group=TRUE,console=TRUE)
out&lt;-BIB.test(run,psi,monovinyl,test="tukey",group=FALSE,console=TRUE)
rm(run,psi,monovinyl,out)
# Example linear estimation and design of experiments. D.D. Joshi. 1987
# Professor of Statistics, Institute of Social Sciences Agra, India
# 6 varieties of wheat crop in a BIB whit 10 blocks of 3 plots each.
y &lt;-c(69,77,72,63,70,54,65,65,57,59,50,45,68,75,59,38,60,60,62,
 55,54,65,62,65,61,39,54,67,63,56)
varieties&lt;-gl(6,5)
block &lt;- c(1,2,3,4,5,1,2,6,7,8,1,3,6,9,10,2,4,7,9,10,3,5,7,8,9,4,5,6,8,10)
BIB.test(block, varieties, y)
# Example Introduction to experimental statistics. Ching Chun Li. 1964
# pag. 395 table. 27.2
# 7 trt, k=3 and b=7.
y &lt;-c(10,15,11,4,12,15,5,14,10,14,19,19,8,10,17,6,11,12,5,14,21)
block&lt;-gl(7,3)
trt &lt;- c(1,2,4,2,3,5,3,4,6,4,5,7,1,5,6,2,6,7,1,3,7)
out&lt;-BIB.test(block, trt, y, test="duncan")
bar.group(out$groups,col="blue",density=4,ylim=c(0,max(y)))
rm(y,block,trt,out)
</code></pre>

<hr>
<h2 id='carolina'> North Carolina Designs I, II and III </h2><span id='topic+carolina'></span>

<h3>Description</h3>

<p>Statistic analysis of the Carolina I, II and III genetic designs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carolina(model,data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="carolina_+3A_model">model</code></td>
<td>
<p> Constant </p>
</td></tr>
<tr><td><code id="carolina_+3A_data">data</code></td>
<td>
<p> Data frame </p>
</td></tr>
</table>


<h3>Details</h3>

<p>model = 1,2 and 3 is I, II and III see carolina1,2 and 3.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>model</code></td>
<td>
<p> model analysis (I, II or III) of caroline design </p>
</td></tr>
</table>
<p>and variance and additive variance of male, female and male.female interaction.
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Biometrical Methods in Quantitative Genetic Analysis,
Singh, Chaudhary. 1979 </p>


<h3>See Also</h3>

<p><code><a href="#topic+DC">DC</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(agricolae)
data(DC)
carolina1 &lt;- DC$carolina1
# str(carolina1)
output&lt;-carolina(model=1,carolina1)
output[][-1]

carolina2 &lt;- DC$carolina2
# str(carolina2)
majes&lt;-subset(carolina2,carolina2[,1]==1)
majes&lt;-majes[,c(2,5,4,3,6:8)]
output&lt;-carolina(model=2,majes[,c(1:4,6)])
output[][-1]

carolina3 &lt;- DC$carolina3
# str(carolina3)
output&lt;-carolina(model=3,carolina3)
output[][-1]
</code></pre>

<hr>
<h2 id='Chz2006'> Data amendment Carhuaz 2006 </h2><span id='topic+Chz2006'></span>

<h3>Description</h3>

<p>Incidents and performance of healthy tubers and rotten potato field infested 
with naturally Ralstonia solanacearum Race 3/Bv 2A, after application of inorganic 
amendments and a rotation crop in Carhuaz Peru, 2006.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Chz2006)</code></pre>


<h3>Format</h3>

<p>The format is:
List of 2
</p>

<dl>
<dt><code>amendment</code></dt><dd><p>a factor</p>
</dd>
<dt><code>crop</code></dt><dd><p>a factor</p>
</dd>
<dt><code>block</code></dt><dd><p>a numeric vector, replications</p>
</dd>
<dt><code>plant</code></dt><dd><p>a numeric vector, number plant</p>
</dd>  
<dt><code>wilt_percent</code></dt><dd><p>a numeric vector, wilt percentage at 60 days</p>
</dd>  
<dt><code>health</code></dt><dd><p>a numeric vector, kg/8m2</p>
</dd> 
<dt><code>rot</code></dt><dd><p>a numeric vector, kg/8m2</p>
</dd> 
</dl>



<h3>Details</h3>

<p>Application of inorganic amendment and crop rotation to control bacterial wilt 
of the potato (MBP).
</p>


<h3>Source</h3>

<p>Experimental field, 2006. Data Kindly provided by Pedro Aley.
</p>


<h3>References</h3>

<p>International Potato Center. CIP - Lima Peru.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(Chz2006)
str(Chz2006)
wilt&lt;-Chz2006$wilt
yield&lt;-Chz2006$yield
means &lt;- tapply.stat(wilt[,5],wilt[,1:3],function(x) mean(x,na.rm=TRUE))
names(means)[4]&lt;-"wilt_percent"
model &lt;- aov(wilt_percent ~ block + crop, means)
anova(model)
cv.model(model)
yield&lt;-yield[order(paste(yield[,1],yield[,2],yield[,3])),]
correlation(means[,4],yield[,4],method="spearman")
</code></pre>

<hr>
<h2 id='CIC'> Data for late blight of potatoes </h2><span id='topic+CIC'></span>

<h3>Description</h3>

<p>A study of Phytophthora infestans in the potato plant in the localities of Comas and
Oxapampa in Peru, 2005.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(CIC)</code></pre>


<h3>Format</h3>

<p>The format is:
List of 2 (comas, oxapampa)
</p>

<dl>
<dt><code>Locality</code></dt><dd><p>a factor with levels <code>Comas</code> <code>Oxapampa</code> </p>
</dd>
<dt><code>Genotype</code></dt><dd><p>a factor</p>
</dd>
<dt><code>Rep</code></dt><dd><p>a numeric vector, replications</p>
</dd>
<dt><code>E9</code></dt><dd><p>a numeric vector, infestans percentaje to 9 days</p>
</dd>
<dt><code>AUDPC</code></dt><dd><p>a numeric vector: the area under the disease-progress curve</p>
</dd>  
<dt><code>Relative</code></dt><dd><p>a numeric vector, relative area</p>
</dd>  
</dl>



<h3>Details</h3>

<p>comas: temperature=59.9 Fahrenheit, relative humidity=83.3
oxapampa: temperature=64.8 Fahrenheit, relative humidity=86.2
AUDPC and relative see function audpc(). help(audpc)
Exx: Evaluation in percentaje, xx is days. 
ORD1, ORD2, SBLK and row are references location of the plot in the field.
</p>


<h3>Source</h3>

<p>Experimental field, 2004-2005. Data Kindly provided by Matilde Orrillo.
</p>


<h3>References</h3>

<p>International Potato Center. CIP - Lima Peru. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(CIC)
CIC$comas
CIC$oxapampa
</code></pre>

<hr>
<h2 id='clay'> Data of Ralstonia population in clay soil</h2><span id='topic+clay'></span>

<h3>Description</h3>

<p>An evaluation over a time period.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(clay)</code></pre>


<h3>Format</h3>

<p>A data frame with 69 observations on the following 3 variables.
</p>

<dl>
<dt><code>per.clay</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>days</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ralstonia</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Experimental field.
</p>


<h3>References</h3>

<p>International Potato Center. CIP - Lima Peru.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(clay)
str(clay)
</code></pre>

<hr>
<h2 id='ComasOxapampa'> Data AUDPC Comas - Oxapampa</h2><span id='topic+ComasOxapampa'></span>

<h3>Description</h3>

<p>Fifty-three potato varieties developed by the breeding program of the 
International Potato Center and released in different countries around 
the world were evaluated for their resistance to late blight in two locations 
in Peru.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ComasOxapampa)</code></pre>


<h3>Format</h3>

<p>A data frame with 168 observations on the following 4 variables.
</p>

<dl>
<dt><code>cultivar</code></dt><dd><p>a factor with 56 levels </p>
</dd>
<dt><code>replication</code></dt><dd><p>a factor with 3 levels </p>
</dd>
<dt><code>comas</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>oxapampa</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>The experimental design was a randomized complete block design with 3 replications 
of 15 apical stem cuttings in Oxapampa and 10 tubers in Mariscal Castilla.
Plots were 11.9 x 18.5 m in size with 30 cm in-row and 0.9 m between-row spacings.
Spreader rows around plots were used at each site. Mancozeb was applied weekly
until 30 days after transplanting or planting, after which the plants were left
to natural infection.
Due to climatic conditions not conductive to the disease
in Oxapampa, inoculum was enhanced with local isolate (POX 067, with virulence
R1, 2, 3, 4, 5, 6, 7, 10, 11) at a concentration of 5000-sporangia/ ml at 49 days
after planting. Percentage of foliar infection was estimated visually every 3 days
for 8 times in Oxapampa and every 7 days for 12 times in Comas, then values were
converted to the relative area under the diseases progress curve (rAUPDC).
rAUDPC rankings were analyzed for phenotypic stability with nonparametric measures.
</p>


<h3>Source</h3>

<p>Experimental field, 2002. Data Kindly provided by Wilmer Perez.
</p>


<h3>References</h3>

<p>International Potato Center. CIP - Lima Peru.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(ComasOxapampa)
# Oxapampa (10 35 31 S latitude, 75 23 0 E longitude, 1813 m.a.s.l )
# Comas, Mariscal Castilla (11  42  54  S latitude, 75 04 45 E longitude, 2800 m.a.s.l,)
# cultivars LBr-40 (resistant), Cruza 148 (moderately resistant) and Pimpernell (susceptible)
str(ComasOxapampa)
means &lt;- tapply.stat(ComasOxapampa[,3:4],ComasOxapampa$cultivar,mean)
correlation(means$comas,means$oxapampa, method="kendall")
</code></pre>

<hr>
<h2 id='consensus'> consensus of clusters </h2><span id='topic+consensus'></span>

<h3>Description</h3>

<p>The criterion of the consensus is to produce many trees by means of
boostrap and to such calculate the relative frequency with members of the clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>consensus(data,distance=c("binary","euclidean","maximum","manhattan",
"canberra", "minkowski", "gower","chisq"),method=c("complete","ward","single","average",
"mcquitty","median", "centroid"),nboot=500,duplicate=TRUE,cex.text=1, 
col.text="red", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="consensus_+3A_data">data</code></td>
<td>
<p> data frame </p>
</td></tr>
<tr><td><code id="consensus_+3A_distance">distance</code></td>
<td>
<p> method distance, see dist() </p>
</td></tr>
<tr><td><code id="consensus_+3A_method">method</code></td>
<td>
<p> method cluster, see hclust() </p>
</td></tr>
<tr><td><code id="consensus_+3A_nboot">nboot</code></td>
<td>
<p> The number of bootstrap samples desired. </p>
</td></tr>
<tr><td><code id="consensus_+3A_duplicate">duplicate</code></td>
<td>
<p> control is TRUE other case is FALSE </p>
</td></tr>
<tr><td><code id="consensus_+3A_cex.text">cex.text</code></td>
<td>
<p> size text on percentage consensus </p>
</td></tr>
<tr><td><code id="consensus_+3A_col.text">col.text</code></td>
<td>
<p> color text on percentage consensus </p>
</td></tr>
<tr><td><code id="consensus_+3A_...">...</code></td>
<td>
<p> parameters of the plot dendrogram </p>
</td></tr>
</table>


<h3>Details</h3>

<p>distance: &quot;euclidean&quot;, &quot;maximum&quot;, &quot;manhattan&quot;, &quot;canberra&quot;, &quot;binary&quot;, &quot;minkowski&quot;, &quot;gower&quot;, &quot;chisq&quot;.
Method: &quot;ward&quot;, &quot;single&quot;, &quot;complete&quot;, &quot;average&quot;, &quot;mcquitty&quot;, &quot;median&quot;, &quot;centroid&quot;.
see functions: dist(), hclust() and daisy() of cluster.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>table.dend</code></td>
<td>
<p>The groups and consensus percentage</p>
</td></tr>
<tr><td><code>dendrogram</code></td>
<td>
<p>The class object is hclust, dendrogram plot</p>
</td></tr>
<tr><td><code>duplicate</code></td>
<td>
<p>Homonymous elements</p>
</td></tr>  
</table>


<h3>Author(s)</h3>

<p> F. de Mendiburu </p>


<h3>References</h3>

<p> An Introduction to the Boostrap. Bradley Efron
and Robert J. Tibshirani. 1993. Chapman and Hall/CRC </p>


<h3>See Also</h3>

 <p><code><a href="stats.html#topic+hclust">hclust</a></code>, <code><a href="#topic+hgroups">hgroups</a></code>, <code><a href="#topic+hcut">hcut</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(pamCIP)
# only code
rownames(pamCIP)&lt;-substr(rownames(pamCIP),1,6)
output&lt;-consensus( pamCIP,distance="binary", method="complete",nboot=5)
# Order consensus
Groups&lt;-output$table.dend[,c(6,5)]
Groups&lt;-Groups[order(Groups[,2],decreasing=TRUE),]
print(Groups)
## Identification of the codes with the numbers.
cbind(output$dendrogram$labels)
## To reproduce dendrogram
dend&lt;-output$dendrogram
data&lt;-output$table.dend
plot(dend)
text(data[,3],data[,4],data[,5])
# Other examples
# classical dendrogram
dend&lt;-as.dendrogram(output$dendrogram)
plot(dend,type="r",edgePar = list(lty=1:2, col=2:1))
text(data[,3],data[,4],data[,5],col="blue",cex=1)
plot(dend,type="t",edgePar = list(lty=1:2, col=2:1))
text(data[,3],data[,4],data[,5],col="blue",cex=1)
## Without the control of duplicates
output&lt;-consensus( pamCIP,duplicate=FALSE,nboot=5)
## using distance gower, require cluster package.
# output&lt;-consensus( pamCIP,distance="gower", method="complete",nboot=5)
</code></pre>

<hr>
<h2 id='corn'> Data of corn </h2><span id='topic+corn'></span>

<h3>Description</h3>

<p>Data from a completely randomized design where four different methods of growing
corn resulted in various yields per acre on various plots of ground where the 
four methods were tried. Ordinarily, only one statistical analysis is used, 
but here we will use the kuskal-wallis test so that a rough comparison may be made
with the mediasn test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(corn)</code></pre>


<h3>Format</h3>

<p>A data frame with 34 observations on the following 3 variables.
</p>

<dl>
<dt><code>method</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>observation</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rx</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>The observations are ranked from the smallest, 77, of rank 1 to the largest 101, 
of rank N=34. Ties values receive the averarge rank.
</p>


<h3>Source</h3>

<p>Book: Practical Nonparametric Statistics.
</p>


<h3>References</h3>

<p>Practical Nonparametrics Statistics. W.J. Conover. Third Edition, 1999.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(corn)
str(corn)
</code></pre>

<hr>
<h2 id='correl'> Correlation Coefficient </h2><span id='topic+correl'></span>

<h3>Description</h3>

<p>An exact correlation for ties or without ties.
Methods of Kendall, Spearman and Pearson.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>correl(x, y, method = "pearson",alternative="two.sided")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="correl_+3A_x">x</code></td>
<td>
<p> Vector </p>
</td></tr>
<tr><td><code id="correl_+3A_y">y</code></td>
<td>
<p> Vector </p>
</td></tr>
<tr><td><code id="correl_+3A_method">method</code></td>
<td>
<p> &quot;pearson&quot;, &quot;kendall&quot;, &quot;spearman&quot; </p>
</td></tr>
<tr><td><code id="correl_+3A_alternative">alternative</code></td>
<td>
<p> &quot;two.sided&quot;, &quot;less&quot;, &quot;greater&quot; </p>
</td></tr>
</table>


<h3>Value</h3>

<p>The correlation of x,y vector with the statistical value and its probability
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p>Numerical Recipes in C. Second Edition. </p>


<h3>See Also</h3>

<p><code><a href="#topic+correlation">correlation</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(soil)
with(soil,correl(pH,clay,method="kendall"))
with(soil,correl(pH,clay,method="spearman"))
with(soil,correl(pH,clay,method="pearson"))
</code></pre>

<hr>
<h2 id='correlation'> Correlation analysis. Methods of Pearson, Spearman, Kendall and Lin</h2><span id='topic+correlation'></span>

<h3>Description</h3>

<p>It obtains the coefficients of correlation and p-value between all the
variables of a data table. The methods to apply are Pearson, Spearman
, Kendall and lin's concordance index. In case of not specifying the method, the Pearson method 
will be used. The results are similar to SAS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>correlation(x,y=NULL, method = c("pearson", "kendall", "spearman", "lin")
,alternative="two.sided")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="correlation_+3A_x">x</code></td>
<td>
<p> table, matrix or vector </p>
</td></tr>
<tr><td><code id="correlation_+3A_y">y</code></td>
<td>
<p> table, matrix or vector </p>
</td></tr>
<tr><td><code id="correlation_+3A_method">method</code></td>
<td>
<p> &quot;pearson&quot;, &quot;kendall&quot;, &quot;spearman&quot;, &quot;lin&quot; </p>
</td></tr>
<tr><td><code id="correlation_+3A_alternative">alternative</code></td>
<td>
<p> &quot;two.sided&quot;, &quot;less&quot;, &quot;greater&quot; </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Parameters equal to function cor()
</p>


<h3>Value</h3>

<p>The correlation matrix with its probability
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p>Lin LI. A concordance correlation coefficient to evaluate reproducibility. 
Biometrics. 1989; 45, 255-268.</p>


<h3>See Also</h3>

<p><code><a href="#topic+correl">correl</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(soil)
# example 1
analysis&lt;-correlation(soil[,2:8],method="pearson")
analysis
# Example 2: correlation between pH, variable 2 and other elements from soil.
analysis&lt;-with(soil,correlation(pH,soil[,3:8],method="pearson",alternative="less"))
analysis
# Example 3: correlation between pH and clay method kendall.
with(soil,correlation(pH,clay,method="kendall", alternative="two.sided"))
</code></pre>

<hr>
<h2 id='cotton'> Data of cotton </h2><span id='topic+cotton'></span>

<h3>Description</h3>

<p>Data of cotton collected in experiments of two localities in Lima and Pisco, Peru.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cotton)</code></pre>


<h3>Format</h3>

<p>A data frame with 96 observations on the following 5 variables.
</p>

<dl>
<dt><code>site</code></dt><dd><p>a factor with levels <code>Lima</code> <code>Pisco</code></p>
</dd>
<dt><code>block</code></dt><dd><p>a factor with levels <code>I</code> <code>II</code> <code>III</code> <code>IV</code> <code>V</code> <code>VI</code></p>
</dd>
<dt><code>lineage</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>epoca</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>yield</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Book spanish: Metodos estadisticos para la investigacion.
Autor: Calzada Benza
Universidad Nacional Agraria - La Molina - Peru..
</p>


<h3>References</h3>

<p>Book spanish: Metodos estadisticos para la investigacion.
Autor: Calzada Benza
Universidad Nacional Agraria - La Molina - Peru.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(cotton)
str(cotton)
</code></pre>

<hr>
<h2 id='cv.model'> Coefficient of the experiment variation </h2><span id='topic+cv.model'></span>

<h3>Description</h3>

<p>It obtains the coefficient of variation of the experiment obtained by 
models lm() or aov()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cv.model(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cv.model_+3A_x">x</code></td>
<td>
<p> object of model lm() or AOV() </p>
</td></tr>
</table>


<h3>Details</h3>

<p>sqrt(MSerror)*100/mean(x)
</p>


<h3>Value</h3>

<p>Returns the coefficient of variation of the experiment according 
to the applied statistical model
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+LSD.test">LSD.test</a></code>, <code><a href="#topic+HSD.test">HSD.test</a></code>, 
<code><a href="#topic+waller.test">waller.test</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># see examples from LSD , Waller-Duncan or HSD and complete with it:
library(agricolae)
# not run
# cv&lt;-cv.model(model)

</code></pre>

<hr>
<h2 id='cv.similarity'> Coefficient of the similarity matrix variation </h2><span id='topic+cv.similarity'></span>

<h3>Description</h3>

<p>This process consists of finding the coefficient of the distances of 
similarity of binary tables (1 and 0) as used for scoring molecular
marker data for presence and absence of PCR amplification products.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cv.similarity(A)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cv.similarity_+3A_a">A</code></td>
<td>
<p> matrix of binary data </p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the coefficient of variation of the similarity model
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+similarity">similarity</a></code>, <code><a href="#topic+resampling.cv">resampling.cv</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'># molecular markers.
library(agricolae)
data(markers)
cv&lt;-cv.similarity(markers)

</code></pre>

<hr>
<h2 id='DAU.test'> Finding the Variance Analysis of the Augmented block Design  </h2><span id='topic+DAU.test'></span>

<h3>Description</h3>

<p>Analysis of variance Augmented block and comparison mean adjusted. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DAU.test(block, trt, y, method = c("lsd","tukey"),alpha=0.05,group=TRUE,console=FALSE)

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DAU.test_+3A_block">block</code></td>
<td>
<p> blocks </p>
</td></tr>
<tr><td><code id="DAU.test_+3A_trt">trt</code></td>
<td>
<p> Treatment </p>
</td></tr>
<tr><td><code id="DAU.test_+3A_y">y</code></td>
<td>
<p> Response  </p>
</td></tr>
<tr><td><code id="DAU.test_+3A_method">method</code></td>
<td>
<p> Comparison treatments </p>
</td></tr>
<tr><td><code id="DAU.test_+3A_alpha">alpha</code></td>
<td>
<p> Significant test </p>
</td></tr>
<tr><td><code id="DAU.test_+3A_group">group</code></td>
<td>
<p> TRUE or FALSE </p>
</td></tr>
<tr><td><code id="DAU.test_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Method of comparison treatment. 
lsd: Least significant difference.
tukey: Honestly significant differente.
The controls can have different repetitions, at least two, do not use missing data.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>means</code></td>
<td>
<p>Statistical summary of the study variable</p>
</td></tr>
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>statistics</code></td>
<td>
<p>Statistics of the model</p>
</td></tr>
<tr><td><code>comparison</code></td>
<td>
<p>Comparison between treatments</p>
</td></tr>
<tr><td><code>groups</code></td>
<td>
<p>Formation of treatment groups</p>
</td></tr>
<tr><td><code>SE.difference</code></td>
<td>
<p> Standard error of:<br />
Two Control Treatments<br />
Two Augmented Treatments<br />
Two Augmented Treatments(Different Blocks)<br />
A Augmented Treatment and A Control Treatment<br /> 
</p>
</td></tr>
<tr><td><code>vartau</code></td>
<td>
<p>Variance-covariance matrix of the difference in treatments</p>
</td></tr>   
</table>


<h3>Author(s)</h3>

<p> F. de Mendiburu </p>


<h3>References</h3>

<p> Federer, W. T. (1956). Augmented (or hoonuiaku) designs. Hawaiian Planters, Record LV(2):191-208. </p>


<h3>See Also</h3>

<p><code><a href="#topic+BIB.test">BIB.test</a></code>, <code><a href="#topic+duncan.test">duncan.test</a></code>, <code><a href="#topic+durbin.test">durbin.test</a></code>, 
<code><a href="#topic+friedman">friedman</a></code>, <code><a href="#topic+HSD.test">HSD.test</a></code>, <code><a href="#topic+kruskal">kruskal</a></code>, 
<code><a href="#topic+LSD.test">LSD.test</a></code>, <code><a href="#topic+Median.test">Median.test</a></code>, <code><a href="#topic+PBIB.test">PBIB.test</a></code>, 
<code><a href="#topic+REGW.test">REGW.test</a></code>, <code><a href="#topic+scheffe.test">scheffe.test</a></code>, <code><a href="#topic+SNK.test">SNK.test</a></code>,
<code><a href="#topic+waerden.test">waerden.test</a></code>, <code><a href="#topic+waller.test">waller.test</a></code>, <code><a href="#topic+plot.group">plot.group</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
block&lt;-c(rep("I",7),rep("II",6),rep("III",7))
trt&lt;-c("A","B","C","D","g","k","l","A","B","C","D","e","i","A","B","C","D","f","h","j")
yield&lt;-c(83,77,78,78,70,75,74,79,81,81,91,79,78,92,79,87,81,89,96,82)
out&lt;- DAU.test(block,trt,yield,method="lsd", group=TRUE)
print(out$groups)
plot(out)
</code></pre>

<hr>
<h2 id='DC'> Data for the analysis of carolina genetic design</h2><span id='topic+DC'></span>

<h3>Description</h3>

<p>Data for the analysis of carolina I, II and III genetic design
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(DC)</code></pre>


<h3>Details</h3>

<p>DC is list, 3 data.frame: carolina1(72 obs, 6 var), carolina2(300 obs, 9 var) and  carolina3(64 obs, 5 var).
</p>
<p>Carolina1: Data for the analysis of Carolina I Genetic design. In this design F2 or any advanced generation maintained by random mating, 
produced from cross between two pure-lines, is taken as base population.
From the population an individual is randomly selected and used as a male.
A set of 4 randomly selected plans are used as females and are mated to the
above male. Thus a set of 4 full-sib families are produced. This is denoted
as a male group. Similarly, a large number of male groups are produced. 
No female is used for any second mating. four male groups (16 female groups)
from a set.
</p>
<p>Carolina2: Data for the analysis of Carolina II Genetic design.
Both paternal and maternal half-sibs are produced in this design.
From an F2 population, n1 males and n2 females are randomly selected
and each male is crossed to each of the females. Thus n1 x n2 progenies
are produced whitch are analysed in a suitably laid experiment.
</p>
<p>Carolina3: Data for the analysis of Carolina III genetic design. The F2 
population is produced by crossing two inbreds, say L1 and L2. 
The material for estimation of genetic parameters is produced 
by back crossing randomly selected F2 individuals (using as males)
to each of the inbreds (used as females).
</p>


<h3>Source</h3>

<p>Biometrical Methods in Quantitative Genetic Analysis, Singh, Chaudhary. 1979.
</p>


<h3>References</h3>

<p>Biometrical Methods in Quantitative Genetic Analysis, Singh, Chaudhary. 1979.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(DC)
names(DC)
str(DC$carolina1)
str(DC$carolina2)
str(DC$carolina3)
</code></pre>

<hr>
<h2 id='delete.na'> Omitting the rows or columns with missing observations of a matrix (NA)  </h2><span id='topic+delete.na'></span>

<h3>Description</h3>

<p>In many situations it is required to omit the rows or columns less or greater
with NA of the matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>delete.na(x, alternative=c("less", "greater") )
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="delete.na_+3A_x">x</code></td>
<td>
<p> matrix with NA </p>
</td></tr>
<tr><td><code id="delete.na_+3A_alternative">alternative</code></td>
<td>
<p> &quot;less&quot; or &quot;greater&quot; </p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>x</code></td>
<td>
<p>matrix</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
x&lt;-c(2,5,3,7,5,NA,8,0,4,3,NA,NA)
dim(x)&lt;-c(4,3)
x
#     [,1] [,2] [,3]
#[1,]    2    5    4
#[2,]    5   NA    3
#[3,]    3    8   NA
#[4,]    7    0   NA

delete.na(x,"less")
#     [,1]
#[1,]    2
#[2,]    5
#[3,]    3
#[4,]    7

delete.na(x,"greater")
#     [,1] [,2] [,3]
#[1,]    2    5    4

</code></pre>

<hr>
<h2 id='design.ab'> Design of experiments for a factorial </h2><span id='topic+design.ab'></span>

<h3>Description</h3>

<p>It generates a design of blocks, randomize and latin square for combined n.
factors uses the methods of number generation in R. The seed is by
set.seed(seed, kinds).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>design.ab(trt, r, serie = 2, design=c("rcbd","crd","lsd"),
seed = 0, kinds = "Super-Duper",first=TRUE,randomization=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="design.ab_+3A_trt">trt</code></td>
<td>
<p> n levels factors </p>
</td></tr>
<tr><td><code id="design.ab_+3A_r">r</code></td>
<td>
<p> Replications or Blocks </p>
</td></tr>
<tr><td><code id="design.ab_+3A_serie">serie</code></td>
<td>
<p> number plot, 1: 11,12; 2: 101,102; 3: 1001,1002 </p>
</td></tr>
<tr><td><code id="design.ab_+3A_design">design</code></td>
<td>
<p> type </p>
</td></tr>
<tr><td><code id="design.ab_+3A_seed">seed</code></td>
<td>
<p> Seed </p>
</td></tr>
<tr><td><code id="design.ab_+3A_kinds">kinds</code></td>
<td>
<p> Method for to randomize </p>
</td></tr>
<tr><td><code id="design.ab_+3A_first">first</code></td>
<td>
<p> TRUE or FALSE - randomize rep 1</p>
</td></tr>
<tr><td><code id="design.ab_+3A_randomization">randomization</code></td>
<td>
<p> TRUE or FALSE - randomize</p>
</td></tr>
</table>


<h3>Details</h3>

<p>kinds &lt;- c(&quot;Wichmann-Hill&quot;, &quot;Marsaglia-Multicarry&quot;, &quot;Super-Duper&quot;,
&quot;Mersenne-Twister&quot;, &quot;Knuth-TAOCP&quot;, &quot;user-supplied&quot;,  &quot;Knuth-TAOCP-2002&quot;,
&quot;default&quot; )
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>book</code></td>
<td>
<p>Fieldbook</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Introduction to Experimental Statistics. 
Ching Chun Li. McGraw-Hill Book Company, INC, New. York, 1964 </p>


<h3>See Also</h3>

<p><code><a href="#topic+design.split">design.split</a></code>, <code><a href="#topic+design.alpha">design.alpha</a></code>,<code><a href="#topic+design.bib">design.bib</a></code>, 
<code><a href="#topic+design.crd">design.crd</a> </code>, <code><a href="#topic+design.cyclic">design.cyclic</a> </code>, <code><a href="#topic+design.dau">design.dau</a> </code>,
<code><a href="#topic+design.graeco">design.graeco</a></code>, <code><a href="#topic+design.lattice">design.lattice</a></code>, <code><a href="#topic+design.lsd">design.lsd</a></code>,
<code><a href="#topic+design.rcbd">design.rcbd</a></code>, <code><a href="#topic+design.strip">design.strip</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'># factorial 3 x 2 with 3 blocks
library(agricolae)
trt&lt;-c(3,2) # factorial 3x2
outdesign &lt;-design.ab(trt, r=3, serie=2)
book&lt;-outdesign$book
head(book,10) # print of the field book
# factorial 2 x 2 x 2 with 5 replications in completely randomized design.
trt&lt;-c(2,2,2)
outdesign&lt;-design.ab(trt, r=5, serie=2,design="crd")
book&lt;-outdesign$book
print(book)
# factorial 3 x 3 in latin square design.
trt &lt;-c(3,3)
outdesign&lt;-design.ab(trt, serie=2, design="lsd")
book&lt;-outdesign$book
print(book)
</code></pre>

<hr>
<h2 id='design.alpha'> Alpha design type (0,1) </h2><span id='topic+design.alpha'></span>

<h3>Description</h3>

<p>Generates an alpha designs starting from the alpha design fixing under 
the series formulated by Patterson and Williams. These designs are 
generated by the alpha arrangements. They are similar to the lattice designs, 
but the tables are rectangular s by k (with s blocks and k&lt;s columns. 
The number of  treatments should be equal to s*k and all the experimental 
units r*s*k (r replications).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>design.alpha(trt, k, r, serie = 2, seed = 0, kinds = "Super-Duper",randomization=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="design.alpha_+3A_trt">trt</code></td>
<td>
<p> Treatments</p>
</td></tr>
<tr><td><code id="design.alpha_+3A_k">k</code></td>
<td>
<p> size block </p>
</td></tr>
<tr><td><code id="design.alpha_+3A_r">r</code></td>
<td>
<p> Replications </p>
</td></tr>
<tr><td><code id="design.alpha_+3A_serie">serie</code></td>
<td>
<p> number plot, 1: 11,12; 2: 101,102; 3: 1001,1002 </p>
</td></tr>
<tr><td><code id="design.alpha_+3A_seed">seed</code></td>
<td>
<p> seed </p>
</td></tr>
<tr><td><code id="design.alpha_+3A_kinds">kinds</code></td>
<td>
<p> method for to randomize </p>
</td></tr>
<tr><td><code id="design.alpha_+3A_randomization">randomization</code></td>
<td>
<p> TRUE or FALSE - randomize</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Parameters for the alpha design:
I. 	r=2, k &lt;= s;
II.	r=3, s odd, k &lt;= s;
III.r=3, s even, k &lt;= s-1;
IV.	r=4, s odd but not a multiple of 3, k&lt;=s
</p>
<p>r= replications
s=number of blocks
k=size of block
Number of treatment is equal to k*s
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>statistics</code></td>
<td>
<p>Design statistics</p>
</td></tr>
<tr><td><code>sketch</code></td>
<td>
<p>Design sketch</p>
</td></tr>
<tr><td><code>book</code></td>
<td>
<p>Fieldbook</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

 
<p>H.D. Patterson and E.R. Williams. Biometrika (1976) 
A new class of resolvable incomplete block designs. printed in Great Britain. 
Online: http://biomet.oxfordjournals.org/cgi/content/abstract/63/1/83 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+design.ab">design.ab</a></code>, <code><a href="#topic+design.split">design.split</a></code>,<code><a href="#topic+design.bib">design.bib</a></code>, 
<code><a href="#topic+design.crd">design.crd</a> </code>, <code><a href="#topic+design.cyclic">design.cyclic</a> </code>, <code><a href="#topic+design.dau">design.dau</a> </code>,
<code><a href="#topic+design.graeco">design.graeco</a></code>, <code><a href="#topic+design.lattice">design.lattice</a></code>, <code><a href="#topic+design.lsd">design.lsd</a></code>,
<code><a href="#topic+design.rcbd">design.rcbd</a></code>, <code><a href="#topic+design.strip">design.strip</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
#Example one
trt&lt;-1:30
t &lt;- length(trt)
# size block k
k&lt;-3
# Blocks s
s&lt;-t/k
# replications r
r &lt;- 2
outdesign&lt;- design.alpha(trt,k,r,serie=2)
book&lt;-outdesign$book
plots&lt;-book[,1]
dim(plots)&lt;-c(k,s,r)
for (i in 1:r) print(t(plots[,,i]))
outdesign$sketch
# Example two 
trt&lt;-letters[1:12] 
t &lt;- length(trt)
k&lt;-3
r&lt;-3
s&lt;-t/k
outdesign&lt;- design.alpha(trt,k,r,serie=2)
book&lt;-outdesign$book
plots&lt;-book[,1]
dim(plots)&lt;-c(k,s,r)
for (i in 1:r) print(t(plots[,,i]))
outdesign$sketch
</code></pre>

<hr>
<h2 id='design.bib'> Randomized Balanced Incomplete Block Designs. BIB</h2><span id='topic+design.bib'></span>

<h3>Description</h3>

<p>Creates Randomized Balanced Incomplete Block Design.
&quot;Random&quot; uses the methods of number generation in R.
The seed is by set.seed(seed, kinds).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>design.bib(trt, k, r=NULL, serie = 2, seed = 0, kinds = "Super-Duper",
maxRep=20,randomization=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="design.bib_+3A_trt">trt</code></td>
<td>
<p> Treatments</p>
</td></tr>
<tr><td><code id="design.bib_+3A_k">k</code></td>
<td>
<p> size block </p>
</td></tr>
<tr><td><code id="design.bib_+3A_r">r</code></td>
<td>
<p> Replications </p>
</td></tr>
<tr><td><code id="design.bib_+3A_serie">serie</code></td>
<td>
<p> number plot, 1: 11,12; 2: 101,102; 3: 1001,1002 </p>
</td></tr>
<tr><td><code id="design.bib_+3A_seed">seed</code></td>
<td>
<p> seed </p>
</td></tr>
<tr><td><code id="design.bib_+3A_kinds">kinds</code></td>
<td>
<p> method for to randomize </p>
</td></tr>
<tr><td><code id="design.bib_+3A_maxrep">maxRep</code></td>
<td>
<p> repetition maximum </p>
</td></tr>
<tr><td><code id="design.bib_+3A_randomization">randomization</code></td>
<td>
<p> TRUE or FALSE - randomize</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The package AlgDesign is necessary.
</p>
<p>if r = NULL, then it calculates the value of r smaller for k defined.
In the case of r = value, then the possible values for &quot;r&quot; is calculated
</p>
<p>K is the smallest integer number of treatments and both values are
consistent in design.
</p>
<p>kinds &lt;- c(&quot;Wichmann-Hill&quot;, &quot;Marsaglia-Multicarry&quot;, &quot;Super-Duper&quot;,
&quot;Mersenne-Twister&quot;, &quot;Knuth-TAOCP&quot;, &quot;user-supplied&quot;,  &quot;Knuth-TAOCP-2002&quot;,
&quot;default&quot; )
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>statistics</code></td>
<td>
<p>Design statistics</p>
</td></tr>
<tr><td><code>sketch</code></td>
<td>
<p>Design sketch</p>
</td></tr>
<tr><td><code>book</code></td>
<td>
<p>Fieldbook</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p>1. Experimental design. Cochran and Cox. Second edition.
Wiley Classics Library Edition published 1992
</p>
<p>2. Optimal Experimental Design with R.
Dieter Rasch, Jurgen Pilz, Rob Verdooren and Albrecht Gebhardt.
2011 by Taylor and Francis Group, LLC CRC Press is an imprint of Taylor
and Francis Group, an Informa business.
</p>
<p>3. Design of Experiments.
Robert O. Kuehl. 2nd ed., Duxbury, 2000.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+design.ab">design.ab</a></code>, <code><a href="#topic+design.alpha">design.alpha</a></code>,<code><a href="#topic+design.split">design.split</a></code>, 
<code><a href="#topic+design.crd">design.crd</a> </code>, <code><a href="#topic+design.cyclic">design.cyclic</a> </code>, <code><a href="#topic+design.dau">design.dau</a> </code>,
<code><a href="#topic+design.graeco">design.graeco</a></code>, <code><a href="#topic+design.lattice">design.lattice</a></code>, <code><a href="#topic+design.lsd">design.lsd</a></code>,
<code><a href="#topic+design.rcbd">design.rcbd</a></code>, <code><a href="#topic+design.strip">design.strip</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
# 4 treatments and k=3 size block
trt&lt;-c("A","B","C","D")
k&lt;-3
outdesign&lt;-design.bib(trt,k,serie=2,seed =41,kinds ="Super-Duper") # seed = 41
print(outdesign$parameters)
book&lt;-outdesign$book
plots &lt;-as.numeric(book[,1])
matrix(plots,byrow=TRUE,ncol=k)
print(outdesign$sketch)
# write in hard disk
# write.csv(book,"book.csv", row.names=FALSE)
# file.show("book.csv")

</code></pre>

<hr>
<h2 id='design.crd'> Completely Randomized Design </h2><span id='topic+design.crd'></span>

<h3>Description</h3>

<p>It generates completely a randomized design with equal or different repetition.
&quot;Random&quot; uses the methods of number generation in R.
The seed is by set.seed(seed, kinds).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>design.crd(trt, r, serie = 2, seed = 0, kinds = "Super-Duper",randomization=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="design.crd_+3A_trt">trt</code></td>
<td>
<p> Treatments</p>
</td></tr>
<tr><td><code id="design.crd_+3A_r">r</code></td>
<td>
<p> Replications </p>
</td></tr>
<tr><td><code id="design.crd_+3A_serie">serie</code></td>
<td>
<p> number plot, 1: 11,12; 2: 101,102; 3: 1001,1002 </p>
</td></tr>
<tr><td><code id="design.crd_+3A_seed">seed</code></td>
<td>
<p> seed </p>
</td></tr>
<tr><td><code id="design.crd_+3A_kinds">kinds</code></td>
<td>
<p> method for to randomize </p>
</td></tr>
<tr><td><code id="design.crd_+3A_randomization">randomization</code></td>
<td>
<p> TRUE or FALSE - randomize</p>
</td></tr>
</table>


<h3>Details</h3>

<p>kinds &lt;- c(&quot;Wichmann-Hill&quot;, &quot;Marsaglia-Multicarry&quot;, &quot;Super-Duper&quot;,
&quot;Mersenne-Twister&quot;, &quot;Knuth-TAOCP&quot;, &quot;user-supplied&quot;,  &quot;Knuth-TAOCP-2002&quot;,
&quot;default&quot; )
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>book</code></td>
<td>
<p>Fieldbook</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Introduction to Experimental Statistics. 
Ching Chun Li. McGraw-Hill Book Company, INC, New. York, 1964 </p>


<h3>See Also</h3>

<p><code><a href="#topic+design.ab">design.ab</a></code>, <code><a href="#topic+design.alpha">design.alpha</a></code>,<code><a href="#topic+design.bib">design.bib</a></code>, 
<code><a href="#topic+design.split">design.split</a> </code>, <code><a href="#topic+design.cyclic">design.cyclic</a> </code>, <code><a href="#topic+design.dau">design.dau</a> </code>,
<code><a href="#topic+design.graeco">design.graeco</a></code>, <code><a href="#topic+design.lattice">design.lattice</a></code>, <code><a href="#topic+design.lsd">design.lsd</a></code>,
<code><a href="#topic+design.rcbd">design.rcbd</a></code>, <code><a href="#topic+design.strip">design.strip</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
trt &lt;-c("CIP-101","CIP-201","CIP-301","CIP-401","CIP-501")
r &lt;-c(4,3,5,4,3)
# seed = 12543
outdesign1 &lt;-design.crd(trt,r,serie=2,2543,"Mersenne-Twister")
book1&lt;-outdesign1
# no seed
outdesign2 &lt;-design.crd(trt,r,serie=3)
print(outdesign2$parameters)
book2&lt;-outdesign2
# write to hard disk
# write.table(book1,"crd.txt", row.names=FALSE, sep="\t")
# file.show("crd.txt")
</code></pre>

<hr>
<h2 id='design.cyclic'> Cyclic designs </h2><span id='topic+design.cyclic'></span>

<h3>Description</h3>

<p>The cyclic design is a incomplete blocks designs, it is generated from 
a incomplete block initial of the size k, the plan is generated and 
randomized. The efficient and robust cyclic designs for 6 to 30 treatments, 
replications &lt;= 10. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>design.cyclic(trt, k, r, serie = 2, rowcol = FALSE, seed = 0, kinds = "Super-Duper"
,randomization=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="design.cyclic_+3A_trt">trt</code></td>
<td>
<p> vector treatments </p>
</td></tr>
<tr><td><code id="design.cyclic_+3A_k">k</code></td>
<td>
<p> block size</p>
</td></tr>
<tr><td><code id="design.cyclic_+3A_r">r</code></td>
<td>
<p> Replications </p>
</td></tr>
<tr><td><code id="design.cyclic_+3A_serie">serie</code></td>
<td>
<p> number plot, 1: 11,12; 2: 101,102; 3: 1001,1002 </p>
</td></tr>
<tr><td><code id="design.cyclic_+3A_rowcol">rowcol</code></td>
<td>
<p> TRUE: row-column design </p>
</td></tr>
<tr><td><code id="design.cyclic_+3A_seed">seed</code></td>
<td>
<p> init seed random </p>
</td></tr>
<tr><td><code id="design.cyclic_+3A_kinds">kinds</code></td>
<td>
<p> random method </p>
</td></tr>
<tr><td><code id="design.cyclic_+3A_randomization">randomization</code></td>
<td>
<p> TRUE or FALSE - randomize</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Number o treatment 6 to 30.
(r) Replication 2 to 10.
(k) size of block 2 to 10.
replication = i*k, &quot;i&quot; is value integer.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>sketch</code></td>
<td>
<p>Design sketch</p>
</td></tr>
<tr><td><code>book</code></td>
<td>
<p>Fieldbook</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Kuehl, Robert(2000), Design of Experiments. 2nd ed., Duxbury. 
John, J.A. (1981) Efficient Cyclic Design. J. R. Statist. Soc. B, 43, No. 1, pp, 76-80.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+design.ab">design.ab</a></code>, <code><a href="#topic+design.alpha">design.alpha</a></code>,<code><a href="#topic+design.bib">design.bib</a></code>, 
<code><a href="#topic+design.crd">design.crd</a> </code>, <code><a href="#topic+design.split">design.split</a> </code>, <code><a href="#topic+design.dau">design.dau</a> </code>,
<code><a href="#topic+design.graeco">design.graeco</a></code>, <code><a href="#topic+design.lattice">design.lattice</a></code>, <code><a href="#topic+design.lsd">design.lsd</a></code>,
<code><a href="#topic+design.rcbd">design.rcbd</a></code>, <code><a href="#topic+design.strip">design.strip</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
trt&lt;-letters[1:8]
# block size = 2, replication = 6
outdesign1 &lt;- design.cyclic(trt,k=2, r=6,serie=2)
names(outdesign1)
# groups 1,2,3
outdesign1$sketch[[1]]
outdesign1$sketch[[2]]
outdesign1$sketch[[3]]
outdesign1$book
# row-column design
outdesign2&lt;- design.cyclic(trt,k=2, r=6, serie=2, rowcol=TRUE)
outdesign2$sketch
</code></pre>

<hr>
<h2 id='design.dau'> Augmented block design </h2><span id='topic+design.dau'></span>

<h3>Description</h3>

<p>These are designs for two types of treatments: the control treatments (common) 
and the increased treatments. The common treatments are applied in complete 
randomized blocks, and the increased treatments, at random. Each treatment 
should be applied in any block once only. It is understood that the common 
treatments are of a greater interest; the standard error of the difference 
is much smaller than when between two increased ones in different blocks. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>design.dau(trt1, trt2, r, serie = 2, seed = 0, kinds = "Super-Duper", name="trt"
,randomization=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="design.dau_+3A_trt1">trt1</code></td>
<td>
<p> checks </p>
</td></tr>
<tr><td><code id="design.dau_+3A_trt2">trt2</code></td>
<td>
<p> new </p>
</td></tr>
<tr><td><code id="design.dau_+3A_r">r</code></td>
<td>
<p> Replications or blocks </p>
</td></tr>
<tr><td><code id="design.dau_+3A_serie">serie</code></td>
<td>
<p> number plot, 1: 11,12; 2: 101,102; 3: 1001,1002 </p>
</td></tr>
<tr><td><code id="design.dau_+3A_seed">seed</code></td>
<td>
<p> seed </p>
</td></tr>
<tr><td><code id="design.dau_+3A_kinds">kinds</code></td>
<td>
<p> method for to randomize </p>
</td></tr>
<tr><td><code id="design.dau_+3A_name">name</code></td>
<td>
<p> name of treatments</p>
</td></tr>
<tr><td><code id="design.dau_+3A_randomization">randomization</code></td>
<td>
<p> TRUE or FALSE - randomize</p>
</td></tr>
</table>


<h3>Details</h3>

<p>kinds &lt;- c(&quot;Wichmann-Hill&quot;, &quot;Marsaglia-Multicarry&quot;, &quot;Super-Duper&quot;,
&quot;Mersenne-Twister&quot;, &quot;Knuth-TAOCP&quot;, &quot;user-supplied&quot;,  &quot;Knuth-TAOCP-2002&quot;,
&quot;default&quot; )
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>book</code></td>
<td>
<p>Fieldbook</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p>1. Augmented (or Hoonuiaku) Design. Federer, W.T. (1956), Hawaii Plr. rec., 55: 191-208.
2. In Augmented Designs. Federer, W.T and Raghavarao, D. (1975). Bometrics, vol. 31, No. 1 (mar.., 1975), pp. 29-35
</p>


<h3>See Also</h3>

<p><code><a href="#topic+design.ab">design.ab</a></code>, <code><a href="#topic+design.alpha">design.alpha</a></code>,<code><a href="#topic+design.bib">design.bib</a></code>, 
<code><a href="#topic+design.crd">design.crd</a> </code>, <code><a href="#topic+design.cyclic">design.cyclic</a> </code>, <code><a href="#topic+design.split">design.split</a> </code>,
<code><a href="#topic+design.graeco">design.graeco</a></code>, <code><a href="#topic+design.lattice">design.lattice</a></code>, <code><a href="#topic+design.lsd">design.lsd</a></code>,
<code><a href="#topic+design.rcbd">design.rcbd</a></code>, <code><a href="#topic+design.strip">design.strip</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
# 4 treatments and 5 blocks
T1&lt;-c("A","B","C","D")
T2&lt;-letters[20:26]
outdesign &lt;-design.dau(T1,T2, r=5,serie=2)
# field book
book&lt;-outdesign$book
by(book,book[2],function(x) paste(x[,1],"-",as.character(x[,3])))
# write in hard disk
# write.table(book,"dau.txt", row.names=FALSE, sep="\t")
# file.show("dau.txt")
# Augmented designs in Completely Randomized Design
trt&lt;-c(T1,T2)
r&lt;-c(4,4,4,4,1,1,1,1,1,1,1)
outdesign &lt;- design.crd(trt,r)
outdesign$book
</code></pre>

<hr>
<h2 id='design.graeco'> Graeco - latin square design</h2><span id='topic+design.graeco'></span>

<h3>Description</h3>

<p>A graeco - latin square is a KxK pattern that permits the study of k 
treatments simultaneously with three different blocking variables, each 
at k levels.
</p>
<p>The function is only for squares of the odd numbers and even numbers (4, 8, 10 and 12)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>design.graeco(trt1, trt2, serie = 2, seed = 0, kinds = "Super-Duper",randomization=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="design.graeco_+3A_trt1">trt1</code></td>
<td>
<p> Treatments </p>
</td></tr>
<tr><td><code id="design.graeco_+3A_trt2">trt2</code></td>
<td>
<p> Treatments </p>
</td></tr>
<tr><td><code id="design.graeco_+3A_serie">serie</code></td>
<td>
<p> number plot, 1: 11,12; 2: 101,102; 3: 1001,1002 </p>
</td></tr>
<tr><td><code id="design.graeco_+3A_seed">seed</code></td>
<td>
<p> seed </p>
</td></tr>
<tr><td><code id="design.graeco_+3A_kinds">kinds</code></td>
<td>
<p> method for to randomize </p>
</td></tr>
<tr><td><code id="design.graeco_+3A_randomization">randomization</code></td>
<td>
<p> TRUE or FALSE - randomize</p>
</td></tr>
</table>


<h3>Details</h3>

<p>kinds &lt;- c(&quot;Wichmann-Hill&quot;, &quot;Marsaglia-Multicarry&quot;, &quot;Super-Duper&quot;,
&quot;Mersenne-Twister&quot;, &quot;Knuth-TAOCP&quot;, &quot;user-supplied&quot;,  &quot;Knuth-TAOCP-2002&quot;,
&quot;default&quot; )
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>book</code></td>
<td>
<p>Fieldbook</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

 
<p>1. Statistics for Experimenters Design, Innovation, and Discovery
Second Edition. George E. P. Box. Wiley-Interscience. 2005.
</p>
<p>2. Experimental design. Cochran and Cox. Second edition. 
Wiley Classics Library Edition published 1992.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+design.ab">design.ab</a></code>, <code><a href="#topic+design.alpha">design.alpha</a></code>,<code><a href="#topic+design.bib">design.bib</a></code>, 
<code><a href="#topic+design.crd">design.crd</a> </code>, <code><a href="#topic+design.cyclic">design.cyclic</a> </code>, <code><a href="#topic+design.dau">design.dau</a> </code>,
<code><a href="#topic+design.split">design.split</a></code>, <code><a href="#topic+design.lattice">design.lattice</a></code>, <code><a href="#topic+design.lsd">design.lsd</a></code>,
<code><a href="#topic+design.rcbd">design.rcbd</a></code>, <code><a href="#topic+design.strip">design.strip</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
T1&lt;-c("a","b","c","d")
T2&lt;-c("v","w","x","y")
outdesign &lt;- design.graeco(T1,T2,serie=1)
graeco&lt;-outdesign$book
plots &lt;-as.numeric(graeco[,1])
print(outdesign$sketch)
print(matrix(plots,byrow=TRUE,ncol=4))
# 10 x 10
T1 &lt;- letters[1:10]
T2 &lt;- 1:10
outdesign &lt;-  design.graeco(T1,T2,serie=2)
print(outdesign$sketch)
</code></pre>

<hr>
<h2 id='design.lattice'> Lattice designs </h2><span id='topic+design.lattice'></span>

<h3>Description</h3>

<p>SIMPLE and TRIPLE lattice designs. It randomizes treatments in k x k lattice.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>design.lattice(trt, r=3, serie = 2, seed = 0, kinds = "Super-Duper",randomization=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="design.lattice_+3A_trt">trt</code></td>
<td>
<p> treatments </p>
</td></tr>
<tr><td><code id="design.lattice_+3A_r">r</code></td>
<td>
<p>r=2(simple) or r=3(triple) lattice</p>
</td></tr>
<tr><td><code id="design.lattice_+3A_serie">serie</code></td>
<td>
<p> number plot, 1: 11,12; 2: 101,102; 3: 1001,1002 </p>
</td></tr>
<tr><td><code id="design.lattice_+3A_seed">seed</code></td>
<td>
<p> seed </p>
</td></tr>
<tr><td><code id="design.lattice_+3A_kinds">kinds</code></td>
<td>
<p> method for to randomize </p>
</td></tr>
<tr><td><code id="design.lattice_+3A_randomization">randomization</code></td>
<td>
<p> TRUE or FALSE - randomize</p>
</td></tr>
</table>


<h3>Details</h3>

<p>kinds &lt;- c(&quot;Wichmann-Hill&quot;, &quot;Marsaglia-Multicarry&quot;, &quot;Super-Duper&quot;,
&quot;Mersenne-Twister&quot;, &quot;Knuth-TAOCP&quot;, &quot;user-supplied&quot;,  &quot;Knuth-TAOCP-2002&quot;,
&quot;default&quot; )
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>statistics</code></td>
<td>
<p>Design statistics</p>
</td></tr>
<tr><td><code>sketch</code></td>
<td>
<p>Design sketch</p>
</td></tr>
<tr><td><code>book</code></td>
<td>
<p>Fieldbook</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> FIELD PLOT TECHNIQUE. Erwin L. LeCLERG. 2nd ed., 1962, 
Burgess Publishing Company, Minnesota </p>


<h3>See Also</h3>

<p><code><a href="#topic+design.ab">design.ab</a></code>, <code><a href="#topic+design.alpha">design.alpha</a></code>,<code><a href="#topic+design.bib">design.bib</a></code>, 
<code><a href="#topic+design.crd">design.crd</a> </code>, <code><a href="#topic+design.cyclic">design.cyclic</a> </code>, <code><a href="#topic+design.dau">design.dau</a> </code>,
<code><a href="#topic+design.graeco">design.graeco</a></code>, <code><a href="#topic+design.split">design.split</a></code>, <code><a href="#topic+design.lsd">design.lsd</a></code>,
<code><a href="#topic+design.rcbd">design.rcbd</a></code>, <code><a href="#topic+design.strip">design.strip</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
# triple lattice
trt&lt;-LETTERS[1:9]
outdesign&lt;-design.lattice(trt,r=3,serie=2) # triple lattice design ( 9 trt)
# simple lattice
trt&lt;-1:100
outdesign&lt;-design.lattice(trt,r=2,serie=3) # simple lattice design, 10x10 
</code></pre>

<hr>
<h2 id='design.lsd'> Latin Square Design </h2><span id='topic+design.lsd'></span>

<h3>Description</h3>

<p>It generates Latin Square Design.
&quot;Random&quot; uses the methods of number generation in R.
The seed is by set.seed(seed, kinds).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>design.lsd(trt, serie = 2, seed = 0, kinds = "Super-Duper",first=TRUE,randomization=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="design.lsd_+3A_trt">trt</code></td>
<td>
<p> Treatments </p>
</td></tr>
<tr><td><code id="design.lsd_+3A_serie">serie</code></td>
<td>
<p> number plot, 1: 11,12; 2: 101,102; 3: 1001,1002 </p>
</td></tr>
<tr><td><code id="design.lsd_+3A_seed">seed</code></td>
<td>
<p> seed </p>
</td></tr>
<tr><td><code id="design.lsd_+3A_kinds">kinds</code></td>
<td>
<p> method for to randomize </p>
</td></tr>
<tr><td><code id="design.lsd_+3A_first">first</code></td>
<td>
<p> TRUE or FALSE - randomize rep 1</p>
</td></tr>
<tr><td><code id="design.lsd_+3A_randomization">randomization</code></td>
<td>
<p> TRUE or FALSE - randomize</p>
</td></tr>
</table>


<h3>Details</h3>

<p>kinds &lt;- c(&quot;Wichmann-Hill&quot;, &quot;Marsaglia-Multicarry&quot;, &quot;Super-Duper&quot;,
&quot;Mersenne-Twister&quot;, &quot;Knuth-TAOCP&quot;, &quot;user-supplied&quot;,  &quot;Knuth-TAOCP-2002&quot;,
&quot;default&quot; )
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>book</code></td>
<td>
<p>Fieldbook</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Introduction to Experimental Statistics. 
Ching Chun Li. McGraw-Hill Book Company, INC, New. York, 1969 </p>


<h3>See Also</h3>

<p><code><a href="#topic+design.ab">design.ab</a></code>, <code><a href="#topic+design.alpha">design.alpha</a></code>,<code><a href="#topic+design.bib">design.bib</a></code>, 
<code><a href="#topic+design.crd">design.crd</a> </code>, <code><a href="#topic+design.cyclic">design.cyclic</a> </code>, <code><a href="#topic+design.dau">design.dau</a> </code>,
<code><a href="#topic+design.graeco">design.graeco</a></code>, <code><a href="#topic+design.lattice">design.lattice</a></code>, <code><a href="#topic+design.split">design.split</a></code>,
<code><a href="#topic+design.rcbd">design.rcbd</a></code>, <code><a href="#topic+design.strip">design.strip</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
varieties&lt;-c("perricholi","yungay","maria bonita","tomasa")
outdesign &lt;-design.lsd(varieties,serie=2,seed=23)
lsd &lt;- outdesign$book 
print(outdesign$sketch)
print(lsd) # field book.
plots &lt;-as.numeric(lsd[,1])
print(matrix(plots,byrow = TRUE, ncol = 4))
# Write on hard disk.
# write.table(lsd,"lsd.txt", row.names=FALSE, sep="\t")
# file.show("lsd.txt")
</code></pre>

<hr>
<h2 id='design.mat'> Experimental design matrix </h2><span id='topic+design.mat'></span>

<h3>Description</h3>

<p>Generate the design matrix from the fieldbook generated by an experimental plan 
or a dataframe for analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>design.mat(book, locations)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="design.mat_+3A_book">book</code></td>
<td>
<p> data frame or matrix, field book </p>
</td></tr>
<tr><td><code id="design.mat_+3A_locations">locations</code></td>
<td>
<p> numeric, column position of the field book </p>
</td></tr>
</table>


<h3>Value</h3>

<p>X is matrix design.
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+design.ab">design.ab</a></code>, <code><a href="#topic+design.alpha">design.alpha</a></code>,<code><a href="#topic+design.bib">design.bib</a></code>, 
<code><a href="#topic+design.crd">design.crd</a> </code>, <code><a href="#topic+design.cyclic">design.cyclic</a> </code>, <code><a href="#topic+design.split">design.split</a> </code>,
<code><a href="#topic+design.graeco">design.graeco</a></code>, <code><a href="#topic+design.lattice">design.lattice</a></code>, <code><a href="#topic+design.lsd">design.lsd</a></code>,
<code><a href="#topic+design.rcbd">design.rcbd</a></code>, <code><a href="#topic+design.strip">design.strip</a></code>, <code><a href="#topic+design.dau">design.dau</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'># dataframe: data analysis
library(agricolae)
data(sweetpotato)
X&lt;-design.mat(sweetpotato,1)
print(X)
# fieldbook: RCBD design
trt &lt;- LETTERS[1:4]
r&lt;-3
plan&lt;-design.rcbd(trt,r,seed=11)
X&lt;-design.mat(plan$book,2:3)
print(X)
</code></pre>

<hr>
<h2 id='design.rcbd'> Randomized Complete Block Design </h2><span id='topic+design.rcbd'></span>

<h3>Description</h3>

<p>It generates Randomized Complete Block Design.
&quot;Random&quot; uses the methods of number generation in R.
The seed is by set.seed(seed, kinds).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>design.rcbd(trt, r, serie = 2, seed = 0, kinds = "Super-Duper", first=TRUE,
continue=FALSE,randomization=TRUE )
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="design.rcbd_+3A_trt">trt</code></td>
<td>
<p> Treatments</p>
</td></tr>
<tr><td><code id="design.rcbd_+3A_r">r</code></td>
<td>
<p> Replications or blocks </p>
</td></tr>
<tr><td><code id="design.rcbd_+3A_serie">serie</code></td>
<td>
<p> number plot, 1: 11,12; 2: 101,102; 3: 1001,1002 </p>
</td></tr>
<tr><td><code id="design.rcbd_+3A_seed">seed</code></td>
<td>
<p> seed </p>
</td></tr>
<tr><td><code id="design.rcbd_+3A_kinds">kinds</code></td>
<td>
<p> method for to randomize </p>
</td></tr>
<tr><td><code id="design.rcbd_+3A_first">first</code></td>
<td>
<p> TRUE or FALSE - randomize rep 1</p>
</td></tr>
<tr><td><code id="design.rcbd_+3A_continue">continue</code></td>
<td>
<p> TRUE or FALSE, continuous numbering of plot</p>
</td></tr>
<tr><td><code id="design.rcbd_+3A_randomization">randomization</code></td>
<td>
<p> TRUE or FALSE - randomize</p>
</td></tr>
</table>


<h3>Details</h3>

<p>kinds &lt;- c(&quot;Wichmann-Hill&quot;, &quot;Marsaglia-Multicarry&quot;, &quot;Super-Duper&quot;,
&quot;Mersenne-Twister&quot;, &quot;Knuth-TAOCP&quot;, &quot;user-supplied&quot;,  &quot;Knuth-TAOCP-2002&quot;,
&quot;default&quot; )
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>sketch</code></td>
<td>
<p>Design sketch</p>
</td></tr>
<tr><td><code>book</code></td>
<td>
<p>Fieldbook</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Introduction to Experimental Statistics. 
Ching Chun Li. McGraw-Hill Book Company, INC, New. York, 1964 </p>


<h3>See Also</h3>

<p><code><a href="#topic+design.ab">design.ab</a></code>, <code><a href="#topic+design.alpha">design.alpha</a></code>,<code><a href="#topic+design.bib">design.bib</a></code>, 
<code><a href="#topic+design.crd">design.crd</a> </code>, <code><a href="#topic+design.cyclic">design.cyclic</a> </code>, <code><a href="#topic+design.dau">design.dau</a> </code>,
<code><a href="#topic+design.graeco">design.graeco</a></code>, <code><a href="#topic+design.lattice">design.lattice</a></code>, <code><a href="#topic+design.lsd">design.lsd</a></code>,
<code><a href="#topic+design.split">design.split</a></code>, <code><a href="#topic+design.strip">design.strip</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
# 5 treatments and 6 blocks
trt&lt;-c("A","B","C","D","E")
outdesign &lt;-design.rcbd(trt,6,serie=2,986,"Wichmann-Hill") # seed = 986
book &lt;-outdesign$book # field book
# write in hard disk
# write.table(book,"rcbd.txt", row.names=FALSE, sep="\t")
# file.show("rcbd.txt")
# Plots in field model ZIGZAG
fieldbook &lt;- zigzag(outdesign)
print(outdesign$sketch)
print(matrix(fieldbook[,1],byrow=TRUE,ncol=5))
# continuous numbering of plot
outdesign &lt;-design.rcbd(trt,6,serie=0,continue=TRUE)
head(outdesign$book)
</code></pre>

<hr>
<h2 id='design.split'> Split Plot Design </h2><span id='topic+design.split'></span>

<h3>Description</h3>

<p>It generates split plot design.
&quot;Random&quot; uses the methods of number generation in R.
The seed is by set.seed(seed, kinds).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>design.split(trt1, trt2,r=NULL, design=c("rcbd","crd","lsd"),serie = 2,
seed = 0, kinds = "Super-Duper", first=TRUE,randomization=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="design.split_+3A_trt1">trt1</code></td>
<td>
<p> Treatments in Plots</p>
</td></tr>
<tr><td><code id="design.split_+3A_trt2">trt2</code></td>
<td>
<p> Treatments in Subplots</p>
</td></tr>
<tr><td><code id="design.split_+3A_r">r</code></td>
<td>
<p> Replications or blocks </p>
</td></tr>
<tr><td><code id="design.split_+3A_design">design</code></td>
<td>
<p> Experimental design </p>
</td></tr>
<tr><td><code id="design.split_+3A_serie">serie</code></td>
<td>
<p> number plot, 1: 11,12; 2: 101,102; 3: 1001,1002 </p>
</td></tr>
<tr><td><code id="design.split_+3A_seed">seed</code></td>
<td>
<p> seed </p>
</td></tr>
<tr><td><code id="design.split_+3A_kinds">kinds</code></td>
<td>
<p> method for to randomize </p>
</td></tr>
<tr><td><code id="design.split_+3A_first">first</code></td>
<td>
<p> TRUE or FALSE - randomize rep 1</p>
</td></tr>
<tr><td><code id="design.split_+3A_randomization">randomization</code></td>
<td>
<p> TRUE or FALSE - randomize</p>
</td></tr>
</table>


<h3>Details</h3>

<p>kinds &lt;- c(&quot;Wichmann-Hill&quot;, &quot;Marsaglia-Multicarry&quot;, &quot;Super-Duper&quot;,
&quot;Mersenne-Twister&quot;, &quot;Knuth-TAOCP&quot;, &quot;user-supplied&quot;,  &quot;Knuth-TAOCP-2002&quot;,
&quot;default&quot; )
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>book</code></td>
<td>
<p>Fieldbook</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Statistical Procedures for Agricultural Research. 
Kwanchai A. Gomez, Arturo A. Gomez. John Wiley &amp; Sons, new York, 1984 </p>


<h3>See Also</h3>

<p><code><a href="#topic+design.ab">design.ab</a></code>, <code><a href="#topic+design.alpha">design.alpha</a></code>,<code><a href="#topic+design.bib">design.bib</a></code>, 
<code><a href="#topic+design.crd">design.crd</a> </code>, <code><a href="#topic+design.cyclic">design.cyclic</a> </code>, <code><a href="#topic+design.dau">design.dau</a> </code>,
<code><a href="#topic+design.graeco">design.graeco</a></code>, <code><a href="#topic+design.lattice">design.lattice</a></code>, <code><a href="#topic+design.lsd">design.lsd</a></code>,
<code><a href="#topic+design.rcbd">design.rcbd</a></code>, <code><a href="#topic+design.strip">design.strip</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
# 4 treatments and 5 blocks in split-plot
t1&lt;-c("A","B","C","D")
t2&lt;-c(1,2,3)
outdesign &lt;-design.split(t1,t2,r=3,serie=2,seed=45,kinds ="Super-Duper")#seed=45
book&lt;-outdesign$book# field book
# write in hard disk
# write.table(book,"book.txt", row.names=FALSE, sep="\t")
# file.show("book.txt")

</code></pre>

<hr>
<h2 id='design.strip'> Strip Plot Design </h2><span id='topic+design.strip'></span>

<h3>Description</h3>

<p>It generates strip plot design.
&quot;Random&quot; uses the methods of number generation in R.
The seed is by set.seed(seed, kinds).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>design.strip(trt1, trt2,r, serie = 2, seed = 0, kinds = "Super-Duper",randomization=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="design.strip_+3A_trt1">trt1</code></td>
<td>
<p> Row treatments</p>
</td></tr>
<tr><td><code id="design.strip_+3A_trt2">trt2</code></td>
<td>
<p> column treatments</p>
</td></tr>
<tr><td><code id="design.strip_+3A_r">r</code></td>
<td>
<p> Replications </p>
</td></tr>
<tr><td><code id="design.strip_+3A_serie">serie</code></td>
<td>
<p> number plot, 1: 11,12; 2: 101,102; 3: 1001,1002 </p>
</td></tr>
<tr><td><code id="design.strip_+3A_seed">seed</code></td>
<td>
<p> seed </p>
</td></tr>
<tr><td><code id="design.strip_+3A_kinds">kinds</code></td>
<td>
<p> method for to randomize </p>
</td></tr>
<tr><td><code id="design.strip_+3A_randomization">randomization</code></td>
<td>
<p> TRUE or FALSE - randomize</p>
</td></tr>
</table>


<h3>Details</h3>

<p>kinds &lt;- c(&quot;Wichmann-Hill&quot;, &quot;Marsaglia-Multicarry&quot;, &quot;Super-Duper&quot;,
&quot;Mersenne-Twister&quot;, &quot;Knuth-TAOCP&quot;, &quot;user-supplied&quot;,  &quot;Knuth-TAOCP-2002&quot;,
&quot;default&quot; )
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>book</code></td>
<td>
<p>Fieldbook</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Statistical Procedures for Agricultural Research. 
Kwanchai A. Gomez, Arturo A. Gomez. John Wiley &amp; Sons, new York, 1984 </p>


<h3>See Also</h3>

<p><code><a href="#topic+design.ab">design.ab</a></code>, <code><a href="#topic+design.alpha">design.alpha</a></code>,<code><a href="#topic+design.bib">design.bib</a></code>, 
<code><a href="#topic+design.crd">design.crd</a> </code>, <code><a href="#topic+design.cyclic">design.cyclic</a> </code>, <code><a href="#topic+design.dau">design.dau</a> </code>,
<code><a href="#topic+design.graeco">design.graeco</a></code>, <code><a href="#topic+design.lattice">design.lattice</a></code>, <code><a href="#topic+design.lsd">design.lsd</a></code>,
<code><a href="#topic+design.rcbd">design.rcbd</a></code>, <code><a href="#topic+design.split">design.split</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
# 4 and 3 treatments and 3 blocks in strip-plot
t1&lt;-c("A","B","C","D")
t2&lt;-c(1,2,3)
r&lt;-3
outdesign &lt;-design.strip(t1,t2,r, serie=2,seed=45,kinds ="Super-Duper") # seed = 45
book &lt;-outdesign$book # field book
# write in hard disk
# write.table(book,"book.txt", row.names=FALSE, sep="\t")
# file.show("book.txt")

</code></pre>

<hr>
<h2 id='design.youden'> Incomplete Latin Square Design </h2><span id='topic+design.youden'></span>

<h3>Description</h3>

<p>Such designs are referred to as Youden squares since they were introduced
by Youden (1937) after Yates (1936) considered the special case of column
equal to number treatment minus 1. &quot;Random&quot; uses the methods of number
generation in R. The seed is by set.seed(seed, kinds).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>design.youden(trt, r, serie = 2, seed = 0, kinds = "Super-Duper",first=TRUE
,randomization=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="design.youden_+3A_trt">trt</code></td>
<td>
<p> Treatments </p>
</td></tr>
<tr><td><code id="design.youden_+3A_r">r</code></td>
<td>
<p> Replications or number of columns </p>
</td></tr>
<tr><td><code id="design.youden_+3A_serie">serie</code></td>
<td>
<p> number plot, 1: 11,12; 2: 101,102; 3: 1001,1002 </p>
</td></tr>
<tr><td><code id="design.youden_+3A_seed">seed</code></td>
<td>
<p> seed </p>
</td></tr>
<tr><td><code id="design.youden_+3A_kinds">kinds</code></td>
<td>
<p> method for to randomize </p>
</td></tr>
<tr><td><code id="design.youden_+3A_first">first</code></td>
<td>
<p> TRUE or FALSE - randomize rep 1</p>
</td></tr>
<tr><td><code id="design.youden_+3A_randomization">randomization</code></td>
<td>
<p> TRUE or FALSE - randomize</p>
</td></tr>
</table>


<h3>Details</h3>

<p>kinds &lt;- c(&quot;Wichmann-Hill&quot;, &quot;Marsaglia-Multicarry&quot;, &quot;Super-Duper&quot;,
&quot;Mersenne-Twister&quot;, &quot;Knuth-TAOCP&quot;, &quot;user-supplied&quot;,  &quot;Knuth-TAOCP-2002&quot;,
&quot;default&quot; )
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>sketch</code></td>
<td>
<p>Design sketch</p>
</td></tr>
<tr><td><code>book</code></td>
<td>
<p>Fieldbook</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Design and Analysis of experiment.
Hinkelmann, Klaus and Kempthorne, Oscar. Wiley-Interscience.
Copyright (2008) by John Wiley and Sons. Inc., Hoboken, new Yersy </p>


<h3>See Also</h3>

<p><code><a href="#topic+design.ab">design.ab</a></code>, <code><a href="#topic+design.alpha">design.alpha</a></code>,<code><a href="#topic+design.bib">design.bib</a></code>, 
<code><a href="#topic+design.crd">design.crd</a> </code>, <code><a href="#topic+design.cyclic">design.cyclic</a> </code>, <code><a href="#topic+design.dau">design.dau</a> </code>,
<code><a href="#topic+design.graeco">design.graeco</a></code>, <code><a href="#topic+design.lattice">design.lattice</a></code>, <code><a href="#topic+design.split">design.split</a></code>,
<code><a href="#topic+design.rcbd">design.rcbd</a></code>, <code><a href="#topic+design.strip">design.strip</a></code>, <code><a href="#topic+design.lsd">design.lsd</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
varieties&lt;-c("perricholi","yungay","maria bonita","tomasa")
r&lt;-3
outdesign &lt;-design.youden(varieties,r,serie=2,seed=23)
youden &lt;- outdesign$book
print(outdesign$sketch)
plots &lt;-as.numeric(youden[,1])
print(matrix(plots,byrow=TRUE,ncol=r))
print(youden) # field book.
# Write on hard disk.
# write.table(youden,"youden.txt", row.names=FALSE, sep="\t")
# file.show("youden.txt")
</code></pre>

<hr>
<h2 id='diffograph'> Plotting the multiple comparison of means </h2><span id='topic+diffograph'></span>

<h3>Description</h3>

<p>It plots bars of the averages of treatments to compare.  It uses the objects generated by a procedure 
of comparison like LSD (Fisher), duncan, Tukey (HSD), Student Newman Keul (SNK), Scheffe, Ryan, Einot 
and Gabriel and Welsch (REGW), Kruskal Wallis, Friedman and Waerden.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diffograph(x, main=NULL,color1="red",color2="blue",color3="black",
cex.axis=0.8,las=1,pch=20,bty="l",cex=0.8,lwd=1,xlab="",ylab="",...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="diffograph_+3A_x">x</code></td>
<td>
<p> Object created by a test of comparison, group=FALSE </p>
</td></tr>
<tr><td><code id="diffograph_+3A_main">main</code></td>
<td>
<p> The main title (on top) </p>
</td></tr>
<tr><td><code id="diffograph_+3A_color1">color1</code></td>
<td>
<p> non significant color </p>
</td></tr>
<tr><td><code id="diffograph_+3A_color2">color2</code></td>
<td>
<p> significant color </p>
</td></tr>
<tr><td><code id="diffograph_+3A_color3">color3</code></td>
<td>
<p> center line color </p>
</td></tr> 
<tr><td><code id="diffograph_+3A_cex.axis">cex.axis</code></td>
<td>
<p> parameters of the plot() </p>
</td></tr> 
<tr><td><code id="diffograph_+3A_las">las</code></td>
<td>
<p> parameters of the plot() </p>
</td></tr> 
<tr><td><code id="diffograph_+3A_pch">pch</code></td>
<td>
<p> parameters of the plot() </p>
</td></tr>
<tr><td><code id="diffograph_+3A_bty">bty</code></td>
<td>
<p> parameters of the plot() </p>
</td></tr> 
<tr><td><code id="diffograph_+3A_cex">cex</code></td>
<td>
<p> parameters of the plot() </p>
</td></tr> 
<tr><td><code id="diffograph_+3A_lwd">lwd</code></td>
<td>
<p> parameters of the plot() </p>
</td></tr>
<tr><td><code id="diffograph_+3A_xlab">xlab</code></td>
<td>
<p> parameters of the plot() </p>
</td></tr>
<tr><td><code id="diffograph_+3A_ylab">ylab</code></td>
<td>
<p> parameters of the plot() </p>
</td></tr>
<tr><td><code id="diffograph_+3A_...">...</code></td>
<td>
<p> Other parameters of the function plot() </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The graph.diff function should be used for functions: LSD, duncan, SNK, scheffe, 
REGW, HSD, kruskal, friedman and waerden test.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>x</code></td>
<td>
<p>list, object comparison test</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Multiple comparisons theory and methods. Departament of statistics 
the Ohio State University. USA, 1996. Jason C. Hsu. Chapman Hall/CRC </p>


<h3>See Also</h3>

<p><code><a href="#topic+LSD.test">LSD.test</a></code>, <code><a href="#topic+HSD.test">HSD.test</a></code>, <code><a href="#topic+duncan.test">duncan.test</a></code>, <code><a href="#topic+SNK.test">SNK.test</a></code>,
<code><a href="#topic+scheffe.test">scheffe.test</a></code>, <code><a href="#topic+REGW.test">REGW.test</a></code>,  <code><a href="#topic+kruskal">kruskal</a></code>,<code><a href="#topic+friedman">friedman</a></code>, 
<code><a href="#topic+waerden.test">waerden.test</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1
library(agricolae)
data(sweetpotato)
model&lt;-aov(yield~virus,data=sweetpotato)
x&lt;- LSD.test(model,"virus",alpha=0.01,group=FALSE)
diffograph(x,cex.axis=0.8,xlab="Yield",ylab="")
# Example 2
x&lt;- REGW.test(model,"virus",alpha=0.01,group=FALSE)
diffograph(x,cex.axis=0.6,xlab="Yield",ylab="",color1="brown",color2="green")
</code></pre>

<hr>
<h2 id='disease'> Data evaluation of the disease overtime</h2><span id='topic+disease'></span>

<h3>Description</h3>

<p>Three evaluations over time and the potato yield when applying several 
treatments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(disease)</code></pre>


<h3>Format</h3>

<p>A data frame with 21 observations on the following 7 variables.
</p>

<dl>
<dt><code>plots</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rep</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>trt</code></dt><dd><p>a factor with levels <code>T0</code> <code>T1</code> <code>T2</code> <code>T3</code> <code>T4</code> <code>T5</code> <code>T6</code></p>
</dd>
<dt><code>E2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>E5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>E7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>yield</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Experimental data.
</p>


<h3>References</h3>

<p>International Potato Center. CIP - Lima Peru.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(disease)
str(disease)
</code></pre>

<hr>
<h2 id='duncan.test'> Duncan's new multiple range test </h2><span id='topic+duncan.test'></span>

<h3>Description</h3>

<p>This test is adapted from the Newman-Keuls method. Duncan's test does not control family wise error rate at the specified alpha level. It has more power than the other post tests, but only because it doesn't control the error rate properly. The Experimentwise Error Rate at: 1-(1-alpha)^(a-1); where &quot;a&quot; is the number of means and is the Per-Comparison Error Rate. Duncan's procedure is only very slightly more conservative than LSD. The level by alpha default is 0.05.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>duncan.test(y, trt, DFerror, MSerror, alpha = 0.05, group=TRUE, main = NULL,console=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="duncan.test_+3A_y">y</code></td>
<td>
<p> model(aov or lm) or answer of the experimental unit  </p>
</td></tr>
<tr><td><code id="duncan.test_+3A_trt">trt</code></td>
<td>
<p> Constant( only y=model) or vector treatment applied to each experimental unit  </p>
</td></tr>
<tr><td><code id="duncan.test_+3A_dferror">DFerror</code></td>
<td>
<p> Degree free </p>
</td></tr>
<tr><td><code id="duncan.test_+3A_mserror">MSerror</code></td>
<td>
<p> Mean Square Error </p>
</td></tr>
<tr><td><code id="duncan.test_+3A_alpha">alpha</code></td>
<td>
<p> Significant level </p>
</td></tr>
<tr><td><code id="duncan.test_+3A_group">group</code></td>
<td>
<p> TRUE or FALSE </p>
</td></tr>
<tr><td><code id="duncan.test_+3A_main">main</code></td>
<td>
<p> Title </p>
</td></tr>
<tr><td><code id="duncan.test_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is necessary first makes a analysis of variance.<br />
</p>
<p>if y = model, then to apply the instruction:<br /> 
duncan.test(model, &quot;trt&quot;, alpha = 0.05, group = TRUE, main = NULL, console = FALSE)<br />
where the model class is aov or lm.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>statistics</code></td>
<td>
<p>Statistics of the model</p>
</td></tr>
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>duncan</code></td>
<td>
<p>Critical Range Table</p>
</td></tr>
<tr><td><code>means</code></td>
<td>
<p>Statistical summary of the study variable</p>
</td></tr>
<tr><td><code>comparison</code></td>
<td>
<p>Comparison between treatments</p>
</td></tr>
<tr><td><code>groups</code></td>
<td>
<p>Formation of treatment groups</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

  
<p>1. Principles and procedures of statistics a biometrical approach
Steel &amp; Torry &amp; Dickey. Third Edition 1997<br />
2. Multiple comparisons theory and methods. Departament of statistics 
the Ohio State University. USA, 1996. Jason C. Hsu. Chapman Hall/CRC.<br />
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BIB.test">BIB.test</a></code>, <code><a href="#topic+DAU.test">DAU.test</a></code>, <code><a href="#topic+durbin.test">durbin.test</a></code>, 
<code><a href="#topic+friedman">friedman</a></code>, <code><a href="#topic+HSD.test">HSD.test</a></code>, <code><a href="#topic+kruskal">kruskal</a></code>, 
<code><a href="#topic+LSD.test">LSD.test</a></code>, <code><a href="#topic+Median.test">Median.test</a></code>, <code><a href="#topic+PBIB.test">PBIB.test</a></code>, 
<code><a href="#topic+REGW.test">REGW.test</a></code>, <code><a href="#topic+scheffe.test">scheffe.test</a></code>, <code><a href="#topic+SNK.test">SNK.test</a></code>,
<code><a href="#topic+waerden.test">waerden.test</a></code>, <code><a href="#topic+waller.test">waller.test</a></code>, <code><a href="#topic+plot.group">plot.group</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(sweetpotato)
model&lt;-aov(yield~virus,data=sweetpotato)
out &lt;- duncan.test(model,"virus", 
main="Yield of sweetpotato. Dealt with different virus")
plot(out,variation="IQR")
duncan.test(model,"virus",alpha=0.01,console=TRUE)
# version old duncan.test()
df&lt;-df.residual(model)
MSerror&lt;-deviance(model)/df
out &lt;- with(sweetpotato,duncan.test(yield,virus,df,MSerror, group=TRUE))
plot(out,horiz=TRUE,las=1)
print(out$groups)
</code></pre>

<hr>
<h2 id='durbin.test'> Durbin test and multiple comparison of treatments </h2><span id='topic+durbin.test'></span>

<h3>Description</h3>

<p>A multiple comparison of the Durbin test for the balanced incomplete 
blocks for sensorial or categorical evaluation. It forms groups according
to the demanded ones for level of significance (alpha); by default, 0.05.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>durbin.test(judge, trt, evaluation, alpha = 0.05, group =TRUE, 
main = NULL, console=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="durbin.test_+3A_judge">judge</code></td>
<td>
<p> Identification of the judge in the evaluation </p>
</td></tr>
<tr><td><code id="durbin.test_+3A_trt">trt</code></td>
<td>
<p> Treatments </p>
</td></tr>
<tr><td><code id="durbin.test_+3A_evaluation">evaluation</code></td>
<td>
<p> variable </p>
</td></tr>
<tr><td><code id="durbin.test_+3A_alpha">alpha</code></td>
<td>
<p> level of significant </p>
</td></tr>
<tr><td><code id="durbin.test_+3A_group">group</code></td>
<td>
<p> TRUE or FALSE </p>
</td></tr>
<tr><td><code id="durbin.test_+3A_main">main</code></td>
<td>
<p> Title </p>
</td></tr>
<tr><td><code id="durbin.test_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The post hoc test is using the criterium Fisher's least significant difference.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>statistics</code></td>
<td>
<p>Statistics of the model</p>
</td></tr>
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>means</code></td>
<td>
<p>Statistical summary of the study variable</p>
</td></tr>
<tr><td><code>rank</code></td>
<td>
<p>rank table of the study variable</p>
</td></tr>
<tr><td><code>comparison</code></td>
<td>
<p>Comparison between treatments</p>
</td></tr>
<tr><td><code>groups</code></td>
<td>
<p>Formation of treatment groups</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Practical Nonparametrics Statistics. W.J. Conover, 1999
Nonparametric Statistical Methods. Myles Hollander and Douglas A. Wofe, 1999</p>


<h3>See Also</h3>

<p><code><a href="#topic+BIB.test">BIB.test</a></code>, <code><a href="#topic+DAU.test">DAU.test</a></code>, <code><a href="#topic+duncan.test">duncan.test</a></code>,
<code><a href="#topic+friedman">friedman</a></code>, <code><a href="#topic+HSD.test">HSD.test</a></code>, <code><a href="#topic+kruskal">kruskal</a></code>, 
<code><a href="#topic+LSD.test">LSD.test</a></code>, <code><a href="#topic+Median.test">Median.test</a></code>, <code><a href="#topic+PBIB.test">PBIB.test</a></code>, 
<code><a href="#topic+REGW.test">REGW.test</a></code>, <code><a href="#topic+scheffe.test">scheffe.test</a></code>, <code><a href="#topic+SNK.test">SNK.test</a></code>,
<code><a href="#topic+waerden.test">waerden.test</a></code>, <code><a href="#topic+waller.test">waller.test</a></code>, <code><a href="#topic+plot.group">plot.group</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
# Example 1. Conover, pag 391
person&lt;-gl(7,3)
variety&lt;-c(1,2,4,2,3,5,3,4,6,4,5,7,1,5,6,2,6,7,1,3,7)
preference&lt;-c(2,3,1,3,1,2,2,1,3,1,2,3,3,1,2,3,1,2,3,1,2)
out&lt;-durbin.test(person,variety,preference,group=TRUE,console=TRUE,
main="Seven varieties of ice cream manufacturer")
#startgraph
bar.group(out$groups,horiz=TRUE,xlim=c(0,10),density=4,las=1)
#endgraph
# Example 2. Myles Hollander, pag 311
# Source: W. Moore and C.I. Bliss. 1942
day&lt;-gl(7,3)
chemical&lt;-c("A","B","D","A","C","E","C","D","G","A","F","G","B","C","F",
 "B","E","G","D","E","F")
toxic&lt;-c(0.465,0.343,0.396,0.602,0.873,0.634,0.875,0.325,0.330,0.423,0.987,
0.426,0.652,1.142,0.989,0.536,0.409,0.309,0.609,0.417,0.931)
out&lt;-durbin.test(day,chemical,toxic,group=TRUE,console=TRUE,
main="Logarithm of Toxic Dosages")
plot(out)
</code></pre>

<hr>
<h2 id='friedman'> Friedman test and multiple comparison of treatments </h2><span id='topic+friedman'></span>

<h3>Description</h3>

<p>The data consist of b-blocks mutually independent k-variate random variables 
Xij, i=1,..,b; j=1,..k. The random variable X is in block i and is associated 
with treatment j. It makes the multiple comparison of the Friedman test with 
or without ties. A first result is obtained by friedman.test of R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>friedman(judge,trt,evaluation,alpha=0.05,group=TRUE,main=NULL,console=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="friedman_+3A_judge">judge</code></td>
<td>
<p> Identification of the judge in the evaluation </p>
</td></tr>
<tr><td><code id="friedman_+3A_trt">trt</code></td>
<td>
<p> Treatment </p>
</td></tr>
<tr><td><code id="friedman_+3A_evaluation">evaluation</code></td>
<td>
<p> Variable </p>
</td></tr>
<tr><td><code id="friedman_+3A_alpha">alpha</code></td>
<td>
<p> Significant test </p>
</td></tr>
<tr><td><code id="friedman_+3A_group">group</code></td>
<td>
<p> TRUE or FALSE </p>
</td></tr>
<tr><td><code id="friedman_+3A_main">main</code></td>
<td>
<p> Title </p>
</td></tr>
<tr><td><code id="friedman_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The post hoc friedman test is using the criterium Fisher's least 
significant difference (LSD)
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>statistics</code></td>
<td>
<p>Statistics of the model</p>
</td></tr>
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>means</code></td>
<td>
<p>Statistical summary of the study variable</p>
</td></tr>
<tr><td><code>comparison</code></td>
<td>
<p>Comparison between treatments</p>
</td></tr>
<tr><td><code>groups</code></td>
<td>
<p>Formation of treatment groups</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Practical Nonparametrics Statistics. W.J. Conover, 1999 </p>


<h3>See Also</h3>

<p><code><a href="#topic+BIB.test">BIB.test</a></code>, <code><a href="#topic+DAU.test">DAU.test</a></code>, <code><a href="#topic+duncan.test">duncan.test</a></code>,
<code><a href="#topic+durbin.test">durbin.test</a></code>, <code><a href="#topic+HSD.test">HSD.test</a></code>, <code><a href="#topic+kruskal">kruskal</a></code>, 
<code><a href="#topic+LSD.test">LSD.test</a></code>, <code><a href="#topic+Median.test">Median.test</a></code>, <code><a href="#topic+PBIB.test">PBIB.test</a></code>, 
<code><a href="#topic+REGW.test">REGW.test</a></code>, <code><a href="#topic+scheffe.test">scheffe.test</a></code>, <code><a href="#topic+SNK.test">SNK.test</a></code>,
<code><a href="#topic+waerden.test">waerden.test</a></code>, <code><a href="#topic+waller.test">waller.test</a></code>, <code><a href="#topic+plot.group">plot.group</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(grass)
out&lt;-with(grass,friedman(judge,trt, evaluation,alpha=0.05, group=TRUE,console=TRUE,
main="Data of the book of Conover"))
#startgraph
plot(out,variation="IQR")
#endgraph
</code></pre>

<hr>
<h2 id='frijol'> Data of frijol </h2><span id='topic+frijol'></span>

<h3>Description</h3>

<p>Data of frijol under 4 technologies for the homogeneity of regression study.
Yield of Frijol in kg/ha in clean and dry grain.
</p>
<p>Tecnnologies:
20-40-20 kg/ha.  N. P2O5 and K2O + 2 t/ha of gallinaza.
40-80-40 kg/ha.  N. P2O5 and K2O + 2 t/ha of gallinaza.
60-120-60 kg/ha. N. P2O5 and K2O + 2 t/ha of gallinaza.
40-80-40 kg/ha.  N. P2O5 and K2O + 4 t/ha of gallinaza.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(frijol)</code></pre>


<h3>Format</h3>

<p>A data frame with 84 observations on the following 3 variables.
</p>

<dl>
<dt><code>technology</code></dt><dd><p>a factor with levels <code>a</code> <code>b</code> <code>c</code> <code>d</code></p>
</dd>
<dt><code>production</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>index</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>References</h3>

<p>Oriente antioqueno (1972) (ICA.- Orlando Martinez W.)
Colombia.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(frijol)
str(frijol)
</code></pre>

<hr>
<h2 id='genxenv'> Data of potato yield in a different environment </h2><span id='topic+genxenv'></span>

<h3>Description</h3>

<p>50 genotypes and 5 environments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(genxenv)</code></pre>


<h3>Format</h3>

<p>A data frame with 250 observations on the following 3 variables.
</p>

<dl>
<dt><code>ENV</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>GEN</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>YLD</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>International Potato Center. CIP - Lima Peru.
</p>


<h3>References</h3>

<p>International Potato Center. CIP - Lima Peru.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(genxenv)
str(genxenv)
</code></pre>

<hr>
<h2 id='Glycoalkaloids'> Data Glycoalkaloids</h2><span id='topic+Glycoalkaloids'></span>

<h3>Description</h3>

<p>A measurement of the Glycoalkaloids by two methods: HPLC and spectrophotometer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Glycoalkaloids)</code></pre>


<h3>Format</h3>

<p>A data frame with 25 observations on the following 2 variables.
</p>

<dl>
<dt><code>HPLC</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>spectrophotometer</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>International Potato Center. CIP - Lima Peru.
</p>


<h3>References</h3>

<p>International Potato Center. CIP - Lima Peru.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(Glycoalkaloids)
str(Glycoalkaloids)
</code></pre>

<hr>
<h2 id='graph.freq'> Histogram </h2><span id='topic+graph.freq'></span>

<h3>Description</h3>

<p>In many situations it has intervals of class defined with its respective
frequencies. By means of this function, the graphic of frequency is obtained 
and it is possible to superpose the normal distribution, polygon of frequency, 
Ojiva and to construct the table of complete frequency.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>graph.freq(x, breaks=NULL,counts=NULL,frequency=1, plot=TRUE, nclass=NULL,
xlab="",ylab="",axes = "",las=1,...)

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="graph.freq_+3A_x">x</code></td>
<td>
<p> a vector of values, a object hist(), graph.freq() </p>
</td></tr>
<tr><td><code id="graph.freq_+3A_counts">counts</code></td>
<td>
<p> frequency and x is class intervals</p>
</td></tr>
<tr><td><code id="graph.freq_+3A_breaks">breaks</code></td>
<td>
<p> a vector giving the breakpoints between histogram cells </p>
</td></tr>
<tr><td><code id="graph.freq_+3A_frequency">frequency</code></td>
<td>
<p> 1=counts, 2=relative, 3=density </p>
</td></tr>
<tr><td><code id="graph.freq_+3A_plot">plot</code></td>
<td>
<p> logic </p>
</td></tr>
<tr><td><code id="graph.freq_+3A_nclass">nclass</code></td>
<td>
<p> number of classes </p>
</td></tr>
<tr><td><code id="graph.freq_+3A_xlab">xlab</code></td>
<td>
<p> x labels</p>
</td></tr>
<tr><td><code id="graph.freq_+3A_ylab">ylab</code></td>
<td>
<p> y labels</p>
</td></tr>
<tr><td><code id="graph.freq_+3A_las">las</code></td>
<td>
<p> values 0,1,2 and 3 are the axis styles. see plot()</p>
</td></tr> 
<tr><td><code id="graph.freq_+3A_axes">axes</code></td>
<td>
<p> TRUE or FALSE</p>
</td></tr>    
<tr><td><code id="graph.freq_+3A_...">...</code></td>
<td>
<p> other parameters of plot</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>breaks</code></td>
<td>
<p> a vector giving the breakpoints between histogram cells </p>
</td></tr>
<tr><td><code>counts</code></td>
<td>
<p> frequency and x is class intervals</p>
</td></tr>
<tr><td><code>mids</code></td>
<td>
<p> center point in class </p>
</td></tr>
<tr><td><code>relative</code></td>
<td>
<p> Relative frequency, height </p>
</td></tr>
<tr><td><code>density</code></td>
<td>
<p> Density frequency, height  </p>
</td></tr>  
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+polygon.freq">polygon.freq</a></code>, <code><a href="#topic+table.freq">table.freq</a></code>, 
<code><a href="#topic+stat.freq">stat.freq</a></code>,<code><a href="#topic+inter.freq">inter.freq</a></code>,<code><a href="#topic+sturges.freq">sturges.freq</a></code>,
<code><a href="#topic+join.freq">join.freq</a></code>,<code><a href="#topic+ogive.freq">ogive.freq</a></code>, <code><a href="#topic+normal.freq">normal.freq</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(agricolae)
data(genxenv)
yield &lt;- subset(genxenv$YLD,genxenv$ENV==2)
yield &lt;- round(yield,1)
h&lt;- graph.freq(yield,axes=FALSE, frequency=1, ylab="frequency",col="yellow")
axis(1,h$breaks)
axis(2,seq(0,20,0.1))
# To reproduce histogram.
h1 &lt;- graph.freq(h, col="blue", frequency=2,border="red", density=8,axes=FALSE,
xlab="YIELD",ylab="relative")
axis(1,h$breaks)
axis(2,seq(0,.4,0.1))
# summary, only frecuency
limits &lt;-seq(10,40,5)
frequencies &lt;-c(2,6,8,7,3,4)
#startgraph
h&lt;-graph.freq(limits,counts=frequencies,col="bisque",xlab="Classes")
polygon.freq(h,col="red")
title( main="Histogram and polygon of frequency",
ylab="frequency")
#endgraph
# Statistics
measures&lt;-stat.freq(h)
print(measures)
# frequency table full
round(table.freq(h),2)
#startgraph
# ogive
ogive.freq(h,col="red",type="b",ylab="Accumulated relative frequency",
xlab="Variable")
# only .frequency polygon
h&lt;-graph.freq(limits,counts=frequencies,border=FALSE,col=NULL,xlab="  ",ylab="")
title( main="Polygon of frequency",
xlab="Variable", ylab="Frecuency")
polygon.freq(h,col="blue")
grid(col="brown")
#endgraph
# Draw curve for Histogram
h&lt;- graph.freq(yield,axes=FALSE, frequency=3, ylab="f(yield)",col="yellow")
axis(1,h$breaks)
axis(2,seq(0,0.18,0.03),las=2)
lines(density(yield), col = "red", lwd = 2)
title("Draw curve for Histogram")
</code></pre>

<hr>
<h2 id='grass'> Data for Friedman test</h2><span id='topic+grass'></span>

<h3>Description</h3>

<p>Twelve homeowners are selected randomly to participate in an experiment with
a plant nursery. Each homeowner is asked to select four fairly identical
areas in his yard and to plant four different types of grasses, one in each area.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(grass)</code></pre>


<h3>Format</h3>

<p>A data frame with 48 observations on the following 3 variables.
</p>

<dl>
<dt><code>judge</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>trt</code></dt><dd><p>a factor with levels <code>t1</code> <code>t2</code> <code>t3</code> <code>t4</code></p>
</dd>
<dt><code>evaluation</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>Each of the 12 blocks consists of four fairly identical plots of land,
each receiving care of approximately the same degree of skill because the
four plots are presumably cared for by the same homeowern.
</p>


<h3>Source</h3>

<p>Book: Practical Nonparametrics Statistics, pag 372.
</p>


<h3>References</h3>

<p>Practical Nonparametrics Statistics. W.J. Conover, 1999
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(grass)
str(grass)
</code></pre>

<hr>
<h2 id='greenhouse'> Data in greenhouse</h2><span id='topic+greenhouse'></span>

<h3>Description</h3>

<p>Potato minituber production in greenhouse, three sets of data in potato varieties
with different methods: hydroponics, Aeroponic, Pots and Plant beds, 
the unit is in grams and the number of tubers in units, 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(greenhouse)</code></pre>


<h3>Details</h3>

<p>greenhouse is list, three tables: greenhouse1(480 obs, 5 var), yield for plant, unit is grams.
greenhouse2(48 obs, 5 var), Yields of 10 plants by experimental unit(grams). planting date(April 24, 2004) and harvest date(July 16, 2004)   and 
greenhouse3(480 obs, 5 var), Tubers by plants.
</p>


<h3>Source</h3>

<p>International Potato Center(CIP). Lima-Peru. Data Kindly provided by Carlos Chuquillanqui.
</p>


<h3>References</h3>

<p>- Produccion de semila de papa por hidroponia tecnica de flujo continuo de una pelicula de solucion nutritiva (nft)
Carlos Chuquillanqui(CIP), Jorge Tenorio(CIP) and L. F. Salazar(Agdia Inc). AGROENFOQUE Lima-Peru (2004)
- Potato Minituber Production Using Aeroponics: Effect of Plant Density and Harvesting Intervals
American Journal of Potato Research,  Jan/Feb 2006  by Farran, Imma,  Mingo-Castel, Angel M
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(greenhouse)
greenhouse1 &lt;- greenhouse$greenhouse1
greenhouse2 &lt;- greenhouse$greenhouse2
greenhouse3 &lt;- greenhouse$greenhouse3
</code></pre>

<hr>
<h2 id='growth'> Data growth of trees</h2><span id='topic+growth'></span>

<h3>Description</h3>

<p>Data growth of pijuayo trees in several localities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(growth)</code></pre>


<h3>Format</h3>

<p>A data frame with 30 observations on the following 3 variables.
</p>

<dl>
<dt><code>place</code></dt><dd><p>a factor with levels <code>L1</code> <code>L2</code></p>
</dd>
<dt><code>slime</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>height</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Experimental data (Pucallpa - Peru)
</p>


<h3>References</h3>

<p>ICRAF lima Peru.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(growth)
str(growth)
</code></pre>

<hr>
<h2 id='haynes'> Data of AUDPC for nonparametrical stability analysis </h2><span id='topic+haynes'></span>

<h3>Description</h3>

<p>Published data. Haynes. Mean area under the disease progress curve (AUDPC) for each of 16 potato clones evaluated at eight sites across the United States in 1996
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(haynes)</code></pre>


<h3>Format</h3>

<p>A data frame with 16 observations on the following 9 variables.
</p>

<dl>
<dt><code>clone</code></dt><dd><p>a factor with levels <code>A84118-3</code> <code>AO80432-1</code> <code>AO84275-3</code> <code>AWN86514-2</code> <code>B0692-4</code> <code>B0718-3</code> <code>B0749-2F</code> <code>B0767-2</code> <code>Bertita</code> <code>Bzura</code> <code>C0083008-1</code> <code>Elba</code> <code>Greta</code> <code>Krantz</code> <code>Libertas</code> <code>Stobrawa</code></p>
</dd>
<dt><code>FL</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>MI</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ME</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>MN</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ND</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>NY</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PA</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>WI</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>References</h3>

<p>Haynes K G, Lambert D H, Christ B J, Weingartner D P, Douches D S,
Backlund J E, Fry W and Stevenson W. 1998. Phenotypic stability of resistance
to late blight in potato clones evaluated at eight sites in the United States
American Journal Potato Research 75, pag 211-217.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(haynes)
str(haynes)
</code></pre>

<hr>
<h2 id='Hco2006'> Data amendment Huanuco 2006 </h2><span id='topic+Hco2006'></span>

<h3>Description</h3>

<p>Incidents and performance of healthy tubers and rotten potato field infested 
with naturally Ralstonia solanacearum Race 3/Bv 2A, after application of inorganic 
amendments and a rotation crop in Huanuco Peru, 2006.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Hco2006)</code></pre>


<h3>Format</h3>

<p>The format is:
List of 2
</p>

<dl>
<dt><code>amendment</code></dt><dd><p>a factor</p>
</dd>
<dt><code>crop</code></dt><dd><p>a factor</p>
</dd>
<dt><code>block</code></dt><dd><p>a numeric vector, replications</p>
</dd>
<dt><code>plant</code></dt><dd><p>a numeric vector, number plant</p>
</dd>  
<dt><code>wilt_percent</code></dt><dd><p>a numeric vector, wilt percentage at 60 days</p>
</dd>  
<dt><code>health</code></dt><dd><p>a numeric vector, kg/8m2, 20 plants</p>
</dd> 
<dt><code>rot</code></dt><dd><p>a numeric vector, kg/8m2, 20 plants</p>
</dd> 
</dl>



<h3>Details</h3>

<p>Application of inorganic amendment and crop rotation to control bacterial wilt 
of the potato (MBP).
</p>


<h3>Source</h3>

<p>Experimental field, 2006. Data Kindly provided by Pedro Aley.
</p>


<h3>References</h3>

<p>International Potato Center. CIP - Lima Peru.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(Hco2006)
str(Hco2006)
wilt&lt;-Hco2006$wilt
yield&lt;-Hco2006$yield
means &lt;- tapply.stat(wilt[,5],wilt[,1:3],function(x) mean(x,na.rm=TRUE))
names(means)[4]&lt;-"wilt_percent"
model &lt;- aov(wilt_percent ~ block + crop, means)
anova(model)
cv.model(model)
yield&lt;-yield[order(paste(yield[,1],yield[,2],yield[,3])),]
correlation(means[,4],yield[,4],method="spearman")
</code></pre>

<hr>
<h2 id='hcut'> Cut tree of consensus </h2><span id='topic+hcut'></span>

<h3>Description</h3>

<p>It shows dendrogram of a consensus of a tree generated by hclust.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hcut(consensus,h,group,col.text="blue",cex.text=1,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hcut_+3A_consensus">consensus</code></td>
<td>
<p> object consensus </p>
</td></tr>
<tr><td><code id="hcut_+3A_h">h</code></td>
<td>
<p> numeric scalar or vector with heights where the tree should be cut.</p>
</td></tr>
<tr><td><code id="hcut_+3A_group">group</code></td>
<td>
<p> an integer scalar with the desired number of group </p>
</td></tr>
<tr><td><code id="hcut_+3A_col.text">col.text</code></td>
<td>
<p> color of number consensus </p>
</td></tr>
<tr><td><code id="hcut_+3A_cex.text">cex.text</code></td>
<td>
<p> size of number consensus </p>
</td></tr>
<tr><td><code id="hcut_+3A_...">...</code></td>
<td>
<p> Other parameters of the function plot() in cut() </p>
</td></tr> 
</table>


<h3>Value</h3>

<p>hcut Returns a data frame with group memberships and consensus tree.
</p>


<h3>Author(s)</h3>

<p> F. de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+hclust">hclust</a></code>, <code><a href="#topic+consensus">consensus</a></code>, <code><a href="#topic+hgroups">hgroups</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(pamCIP)
# only code
rownames(pamCIP)&lt;-substr(rownames(pamCIP),1,6)
# groups of clusters
# output&lt;-consensus(pamCIP,nboot=100)
# hcut(output,h=0.4,group=5,main="Group 5")
# 
# hcut(output,h=0.4,group=8,type="t",edgePar=list(lty=1:2,col=2:1),main="group 8"
# ,col.text="blue",cex.text=1)
</code></pre>

<hr>
<h2 id='heterosis'> Data of potato, Heterosis</h2><span id='topic+heterosis'></span>

<h3>Description</h3>

<p>Determination of heterosis, general combining ability (GCA) and 
specific combining ability in tuber dry matter, reducing sugars and 
agronomic characteristics in TPS families.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(heterosis)</code></pre>


<h3>Format</h3>

<p>A data frame with 216 observations on the following 11 variables.
</p>

<dl>
<dt><code>Place</code></dt><dd><p>1: La Molina, 2=Huancayo</p>
</dd>
<dt><code>Replication</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Treatment</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Factor</code></dt><dd><p>a factor with levels <code>Control</code> <code>progenie</code> <code>progenitor</code> <code>testigo</code></p>
</dd>
<dt><code>Female</code></dt><dd><p>a factor with levels <code>Achirana</code> <code>LT-8</code> <code>MF-I</code> <code>MF-II</code> <code>Serrana</code> <code>TPS-2</code> <code>TPS-25</code> <code>TPS-7</code></p>
</dd>
<dt><code>Male</code></dt><dd><p>a factor with levels <code>TPS-13</code> <code>TPS-67</code> <code>TS-15</code></p>
</dd>
<dt><code>v1</code></dt><dd><p>Yield (Kg/plant)</p>
</dd>
<dt><code>v2</code></dt><dd><p>Reducing sugars (scale):1 low and 5=High</p>
</dd>
<dt><code>v3</code></dt><dd><p>Tuber dry matter (percentage)</p>
</dd>
<dt><code>v4</code></dt><dd><p>Tuber number/plant</p>
</dd>
<dt><code>v5</code></dt><dd><p>Average tuber weight (g)</p>
</dd>
</dl>



<h3>Details</h3>

<p>The study was conducted in 3 environments, La Molina-PERU to 240 masl. during 
autumn-winter and spring, and in Huancayo-PERU 3180 masl., during summer. 
The experimental material consisted of 24 families half brother in the form 
of tubers derived from TPS, obtained crossing between 8 female and 3 male parents. 
Design used was randomized complete block with three repetitions. 
The experimental unit was 30 plants in two rows at a distance of 30cm 
between plants and 90 cm between rows. Variables evaluated were Yield, Tubers number, 
Dry matter and content and reducing sugars. The analysis was conducted 
line x tester. The control variety was Desiree.
</p>


<h3>Source</h3>

<p>International Potato Center(CIP). Lima-Peru. Data Kindly provided by of Rolando Cabello. 
</p>


<h3>References</h3>

<p>Tesis &quot;Heterosis, habilidad combinatoria general y especifica para materia seca, 
azucares reductores y caracteres agronomicos en familias de tuberculos provenientes 
de semilla sexual de papa. Magister Scientiae Rodolfo Valdivia Lorente. Universidad 
Nacional Agraria La molina-Lima Peru, Escuela de Post Grado, Mejoramiento genetico
de plantas, 2004&quot;. Poster: Congreso de la Sociedad Peruana de Genetica - Peru, 2008.  
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(heterosis)
str(heterosis)
site1&lt;-subset(heterosis,heterosis[,1]==1)
site2&lt;-subset(heterosis,heterosis[,1]==2)
site3&lt;-subset(heterosis,heterosis[,1]==3)
model1&lt;-with(site1,lineXtester(Replication, Female, Male, v1))
DFe &lt;- df.residual(model1)
CMe &lt;- deviance(model1)/DFe
test1 &lt;- with(site1,HSD.test(v1, Factor,DFe,CMe))
test2 &lt;- with(site1,HSD.test(v1, Treatment,DFe,CMe))
model22&lt;-with(site2,lineXtester(Replication, Female, Male, v3))
model3&lt;-with(site3,lineXtester(Replication, Female, Male, v4))
</code></pre>

<hr>
<h2 id='hgroups'> groups of hclust </h2><span id='topic+hgroups'></span>

<h3>Description</h3>

<p>Returns a vector with group memberships. This function is used 
by the function consensus of clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hgroups(hmerge)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hgroups_+3A_hmerge">hmerge</code></td>
<td>
<p> The object is components of the hclust</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The merge clusters is printed.
</p>


<h3>Author(s)</h3>

<p> F. de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+hclust">hclust</a></code>, <code><a href="#topic+hcut">hcut</a></code>, <code><a href="#topic+consensus">consensus</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(pamCIP)
# only code
rownames(pamCIP)&lt;-substr(rownames(pamCIP),1,6)
distance &lt;- dist(pamCIP,method="binary")
clusters&lt;- hclust( distance, method="complete")
# groups of clusters
hgroups(clusters$merge)
</code></pre>

<hr>
<h2 id='HSD.test'> Multiple comparisons: Tukey  </h2><span id='topic+HSD.test'></span>

<h3>Description</h3>

<p>It makes multiple comparisons of treatments by means of Tukey.
The level by alpha default is 0.05.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HSD.test(y, trt, DFerror, MSerror, alpha = 0.05, group=TRUE, main = NULL, 
unbalanced=FALSE,console=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HSD.test_+3A_y">y</code></td>
<td>
<p> model(aov or lm) or answer of the experimental unit  </p>
</td></tr>
<tr><td><code id="HSD.test_+3A_trt">trt</code></td>
<td>
<p> Constant( only y=model) or vector treatment applied to each experimental unit  </p>
</td></tr>
<tr><td><code id="HSD.test_+3A_dferror">DFerror</code></td>
<td>
<p> Degree free </p>
</td></tr>
<tr><td><code id="HSD.test_+3A_mserror">MSerror</code></td>
<td>
<p> Mean Square Error </p>
</td></tr>
<tr><td><code id="HSD.test_+3A_alpha">alpha</code></td>
<td>
<p> Significant level </p>
</td></tr>
<tr><td><code id="HSD.test_+3A_group">group</code></td>
<td>
<p> TRUE or FALSE </p>
</td></tr>
<tr><td><code id="HSD.test_+3A_main">main</code></td>
<td>
<p> Title </p>
</td></tr>
<tr><td><code id="HSD.test_+3A_unbalanced">unbalanced</code></td>
<td>
<p> TRUE or FALSE. not equal replication</p>
</td></tr>  
<tr><td><code id="HSD.test_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is necessary first makes a analysis of variance.<br />
</p>
<p>if y = model, then to apply the instruction:<br /> 
HSD.test (model, &quot;trt&quot;, alpha = 0.05, group = TRUE, main = NULL, unbalanced=FALSE, console=FALSE)<br />
where the model class is aov or lm.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>statistics</code></td>
<td>
<p>Statistics of the model</p>
</td></tr>
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>means</code></td>
<td>
<p>Statistical summary of the study variable</p>
</td></tr>
<tr><td><code>comparison</code></td>
<td>
<p>Comparison between treatments</p>
</td></tr>
<tr><td><code>groups</code></td>
<td>
<p>Formation of treatment groups</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

  
<p>1. Principles and procedures of statistics a biometrical approach
Steel &amp; Torry &amp; Dickey. Third Edition 1997<br />
2. Multiple comparisons theory and methods. Departament of statistics 
the Ohio State University. USA, 1996. Jason C. Hsu. Chapman Hall/CRC.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BIB.test">BIB.test</a></code>, <code><a href="#topic+DAU.test">DAU.test</a></code>, <code><a href="#topic+duncan.test">duncan.test</a></code>,
<code><a href="#topic+durbin.test">durbin.test</a></code>, <code><a href="#topic+friedman">friedman</a></code>, <code><a href="#topic+kruskal">kruskal</a></code>, 
<code><a href="#topic+LSD.test">LSD.test</a></code>, <code><a href="#topic+Median.test">Median.test</a></code>, <code><a href="#topic+PBIB.test">PBIB.test</a></code>, 
<code><a href="#topic+REGW.test">REGW.test</a></code>, <code><a href="#topic+scheffe.test">scheffe.test</a></code>, <code><a href="#topic+SNK.test">SNK.test</a></code>,
<code><a href="#topic+waerden.test">waerden.test</a></code>, <code><a href="#topic+waller.test">waller.test</a></code>, <code><a href="#topic+plot.group">plot.group</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(sweetpotato)
model&lt;-aov(yield~virus, data=sweetpotato)
out &lt;- HSD.test(model,"virus", group=TRUE,console=TRUE,
main="Yield of sweetpotato\nDealt with different virus")
#stargraph
# Variation range: max and min
plot(out)
#endgraph
out&lt;-HSD.test(model,"virus", group=FALSE)
print(out$comparison)
# Old version HSD.test()
df&lt;-df.residual(model)
MSerror&lt;-deviance(model)/df
with(sweetpotato,HSD.test(yield,virus,df,MSerror, group=TRUE,console=TRUE,
main="Yield of sweetpotato. Dealt with different virus"))
</code></pre>

<hr>
<h2 id='huasahuasi'> Data: Rainfall thresholds as support for timing fungicide applications in the control of potato late blight in Peru </h2><span id='topic+huasahuasi'></span>

<h3>Description</h3>

<p>Timing fungicide sprays based on accumulated rainfall thresholds can be a successful 
component of integrated management packages that include cultivars with moderate 
or high levels of resistance to late blight. The simplicity of measuring accumulated rainfall means that the technology 
can potentially be used by resource-poor farmers in developing countries. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(huasahuasi)</code></pre>


<h3>Format</h3>

<p>The format is:
List of 2 ( AUDPC, YIELD )
</p>

<dl>
<dt><code>block</code></dt><dd><p>a factor with levels <code>I</code> <code>II</code> <code>III</code></p>
</dd>
<dt><code>trt</code></dt><dd><p>a factor with levels <code>40mm</code> <code>7-days</code> <code>Non-application</code></p>
</dd>
<dt><code>clon</code></dt><dd><p>a factor with levels <code>C386209.10</code> <code>C387164.4</code> <code>Cruza148</code> <code>Musuq</code> <code>Yungay</code></p>
</dd>
<dt><code>y1da</code></dt><dd><p>a numeric vector, Kgr./plot</p>
</dd>
<dt><code>y2da</code></dt><dd><p>a numeric vector, Kgr./plot</p>
</dd>
<dt><code>y3ra</code></dt><dd><p>a numeric vector, Kgr./plot</p>
</dd>
<dt><code>d44</code></dt><dd><p>a numeric vector, 44 days</p>
</dd>
<dt><code>d51</code></dt><dd><p>a numeric vector, 51 days</p>
</dd>    
<dt><code>d100</code></dt><dd><p>a numeric vector, 100 days</p>
</dd>    
</dl>



<h3>Details</h3>

<p>The experimental unit was formed by 4 furrows of 1.8 m of length, with distance 
between furrows from 0.90 m and between plants of 0.30 m. In each furrow was 
installed 5 plants. The experiment had 3 repetitions. 
From the beginning of the experiment were fulfilled the following treatments
Thresholds 40 mm: Apply the fungicide when 40 precipitation mm accumulates. 
The minimum interval between applications will be of 7 days. 
Schedule 7 days: The applications should be carried out every 7 days calendar. 
Without application: No fungicide application will be made. 
The evaluation of the severity of the late blight in each treatment started to 
emergency 80 percentage and then evaluations were made every 7 days until being observed 
a physiological maturation of the crop. 
</p>


<h3>Source</h3>

<p>Experimental field, 2003. Data Kindly provided by Wilmer Perez.
</p>


<h3>References</h3>

<p>International Potato Center. CIP - Lima Peru.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(huasahuasi)
names(huasahuasi)
str(huasahuasi$AUDPC)
str(huasahuasi$YIELD)

</code></pre>

<hr>
<h2 id='index.AMMI'> AMMI index and yield stability</h2><span id='topic+index.AMMI'></span>

<h3>Description</h3>

<p>calculate AMMI stability value (ASV) and Yield stability index (YSI).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>index.AMMI(model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="index.AMMI_+3A_model">model</code></td>
<td>
<p> object AMMI </p>
</td></tr>
</table>


<h3>Details</h3>

<p>AMMI stability value (ASV) was calculated using the following formula,
as suggested by Purchase (1997)
</p>
<p>ASV = sqrt((SSpc1/SSpc2 * PC1i)^2+(PC2i)^2)
</p>
<p>YSI = RASV + RY
</p>
<p>RASV = rank(ASV) and RY = rank(Y across by environment)
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>ASV</code></td>
<td>
<p>AMMI stability value</p>
</td></tr>
<tr><td><code>YSI</code></td>
<td>
<p>Yield stability index</p>
</td></tr>
<tr><td><code>rASV</code></td>
<td>
<p>Rank of AMMI stability value</p>
</td></tr>
<tr><td><code>rYSI</code></td>
<td>
<p>Rank of yield stability index</p>
</td></tr>
<tr><td><code>means</code></td>
<td>
<p>average genotype by environment</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> F. de Mendiburu </p>


<h3>References</h3>

 
<p>The use of an AMMI model and its parameters to analyse yield stability
in multienvironment trials. N. SABAGHNIA, S.H. SABAGHPOUR
AND H. DEHGHANI. Journal of Agricultural Science (2008), 146, 571-581.
f 2008 Cambridge University Press 571 doi:10.1017/S0021859608007831
Printed in the United Kingdom
</p>
<p>Parametric analysis to describe genotype  x environment interaction and yield
stability in winter wheat. PURCHASE, J. L. (1997).
Ph.D. Thesis, Department of Agronomy, Faculty of Agriculture of the
University of the Free State, Bloemfontein, South Africa.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+AMMI">AMMI</a></code>,<code><a href="#topic+plot.AMMI">plot.AMMI</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
# Index AMMI
data(plrv)
model&lt;- with(plrv,AMMI(Locality, Genotype, Rep, Yield, console=FALSE))
Idx&lt;-index.AMMI(model)
names(Idx)
# Crops with improved stability according AMMI.
print(Idx[order(Idx[,3]),])
# Crops with better response and improved stability according AMMI.
print(Idx[order(Idx[,4]),])
</code></pre>

<hr>
<h2 id='index.bio'> Biodiversity Index </h2><span id='topic+index.bio'></span>

<h3>Description</h3>

<p>Scientists use a formula called the biodiversity index to describe the amount of
species diversity in a given area.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>index.bio(data, method = c("Margalef", "Simpson.Dom", "Simpson.Div", 
"Berger.Parker", "McIntosh", "Shannon"), level=95, nboot=100, console=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="index.bio_+3A_data">data</code></td>
<td>
<p> number of specimens </p>
</td></tr>
<tr><td><code id="index.bio_+3A_method">method</code></td>
<td>
<p> Describe method bio-diversity </p>
</td></tr>
<tr><td><code id="index.bio_+3A_level">level</code></td>
<td>
<p> Significant level </p>
</td></tr>
<tr><td><code id="index.bio_+3A_nboot">nboot</code></td>
<td>
<p> size bootstrap </p>
</td></tr>
<tr><td><code id="index.bio_+3A_console">console</code></td>
<td>
<p> output console TRUE</p>
</td></tr>
</table>


<h3>Details</h3>

<p>method bio-diversity.
&quot;Margalef&quot;
&quot;Simpson.Dom&quot;
&quot;Simpson.Div&quot;
&quot;Berger.Parker&quot;
&quot;McIntosh&quot;
&quot;Shannon&quot;
</p>


<h3>Value</h3>

<p>Index and confidence intervals.
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p>Magurran, A.E. (1988) Ecological diversity and its measurement. 
Princeton University Press Efron, B., Tibshirani, R. (1993) An Introduction to 
the Boostrap. Chapman and Hall/CRC 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(paracsho)
# date 22-06-05 and treatment CON = application with insecticide
specimens &lt;- paracsho[1:10,6]
output1 &lt;- index.bio(specimens,method="Simpson.Div",level=95,nboot=100)
output2 &lt;- index.bio(specimens,method="Shannon",level=95,nboot=100)
rbind(output1, output2)
</code></pre>

<hr>
<h2 id='index.smith'> Uniformity soil. Smith's Index of Soil Heterogeneity </h2><span id='topic+index.smith'></span>

<h3>Description</h3>

<p>Smith's index of soil heterogeneity is used primarily to derive optimum
plot size. The index gives a single value as a quantitative measure of 
soil heterogeneity in an area.
Graph CV for plot size and shape.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>index.smith(data, PLOT=TRUE,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="index.smith_+3A_data">data</code></td>
<td>
<p> dataframe or matrix </p>
</td></tr>
<tr><td><code id="index.smith_+3A_plot">PLOT</code></td>
<td>
<p> graphics, TRUE or FALSE </p>
</td></tr>
<tr><td><code id="index.smith_+3A_...">...</code></td>
<td>
<p> Parameters of the plot() </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Vx=V(x)/x b
</p>
<p>V(x) is the between-plot variance,
Vx is the variance per unit area for plot size of x basic unit, 
and b is the Smith' index of soil heterogeneity.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>model</code></td>
<td>
<p>function pattern of uniformity</p>
</td></tr>
<tr><td><code>uniformity</code></td>
<td>
<p>Table of the soil uniformity</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Statistical Procedures for Agriculture Research. Second Edition.
Kwanchai A. Gomez and Arturo A. Gomez. 1976. USA
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(rice)
#startgraph
table&lt;-index.smith(rice,
 main="Relationship between CV per unit area and plot size",col="red")
#endgraph
uniformity &lt;- data.frame(table$uniformity)
uniformity
# regression variance per unit area an plot size.
model &lt;- lm(Vx ~ I(log(Size)),uniformity)
coeff &lt;- coef(model)
x&lt;-1:max(uniformity$Size)
Vx&lt;- coeff[1]+coeff[2]*log(x)
#startgraph
plot(x,Vx, type="l", col="blue",
 main="Relationship between variance per unit area and plot size")
points(uniformity$Size,uniformity$Vx)
#endgraph
</code></pre>

<hr>
<h2 id='inter.freq'> Class intervals </h2><span id='topic+inter.freq'></span>

<h3>Description</h3>

<p>List class intervals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inter.freq(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="inter.freq_+3A_x">x</code></td>
<td>
<p> class graph.freq, histogram or numeric </p>
</td></tr>
</table>


<h3>Value</h3>

<p>It show interval classes.
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+polygon.freq">polygon.freq</a></code>, <code><a href="#topic+table.freq">table.freq</a></code>, <code><a href="#topic+stat.freq">stat.freq</a></code>,
<code><a href="#topic+graph.freq">graph.freq</a></code>, <code><a href="#topic+sturges.freq">sturges.freq</a></code>, <code><a href="#topic+join.freq">join.freq</a></code>, 
<code><a href="#topic+ogive.freq">ogive.freq</a></code>, <code><a href="#topic+normal.freq">normal.freq</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
# example 1
data(growth)
h&lt;-hist(growth$height,plot=FALSE)
inter.freq(h)
# example 2
x&lt;-seq(10,40,5)
y&lt;-c(2,6,8,7,3,4)
inter.freq(x)
histogram &lt;- graph.freq(x,counts=y)
</code></pre>

<hr>
<h2 id='join.freq'> Join class for histogram </h2><span id='topic+join.freq'></span>

<h3>Description</h3>

<p>In many situations it is required to join classes because of the low .frequency
in the intervals. In this process, it is required to join the intervals and
ad the .frequencies of them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>join.freq(histogram, join)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="join.freq_+3A_histogram">histogram</code></td>
<td>
<p> Class graph.freq </p>
</td></tr>
<tr><td><code id="join.freq_+3A_join">join</code></td>
<td>
<p> vector </p>
</td></tr>
</table>


<h3>Value</h3>

<p>New histogram with union of classes.
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+polygon.freq">polygon.freq</a></code>, <code><a href="#topic+table.freq">table.freq</a></code>, <code><a href="#topic+stat.freq">stat.freq</a></code>,
<code><a href="#topic+inter.freq">inter.freq</a></code>, <code><a href="#topic+sturges.freq">sturges.freq</a></code>, <code><a href="#topic+graph.freq">graph.freq</a></code>,
<code><a href="#topic+ogive.freq">ogive.freq</a></code>, <code><a href="#topic+normal.freq">normal.freq</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(natives)
# histogram
h1&lt;-graph.freq(natives$size,plot=FALSE)
round(table.freq(h1),4)
# Join classes  9, 10,11 and 12 with little frequency.
h2&lt;-join.freq(h1,9:12)
# new table
plot(h2,col="bisque",xlab="Size")
round(summary(h2),4)
</code></pre>

<hr>
<h2 id='kendall'> Correlation of Kendall </h2><span id='topic+kendall'></span>

<h3>Description</h3>

<p>Correlation of Kendall two set. Compute exact p-value with ties.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kendall(data1, data2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="kendall_+3A_data1">data1</code></td>
<td>
<p> vector </p>
</td></tr>
<tr><td><code id="kendall_+3A_data2">data2</code></td>
<td>
<p> vector </p>
</td></tr>
</table>


<h3>Value</h3>

<p>The correlation of data1, data2 vector with the statistical value and its probability
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Numerical Recipes in C. Second Edition. Pag 634</p>


<h3>See Also</h3>

<p><code><a href="#topic+correlation">correlation</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
x &lt;-c(1,1,1,4,2,2,3,1,3,2,1,1,2,3,2,1,1,2,1,2)
y &lt;-c(1,1,2,3,4,4,2,1,2,3,1,1,3,4,2,1,1,3,1,2)
kendall(x,y)
</code></pre>

<hr>
<h2 id='kruskal'> Kruskal Wallis test and multiple comparison of treatments. </h2><span id='topic+kruskal'></span>

<h3>Description</h3>

<p>It makes the multiple comparison with Kruskal-Wallis. The alpha parameter by default is 0.05. Post hoc test is using the criterium Fisher's least significant difference. The adjustment methods include the Bonferroni correction  and others.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kruskal(y, trt, alpha = 0.05, p.adj=c("none","holm","hommel", 
"hochberg", "bonferroni", "BH", "BY", "fdr"), group=TRUE, main = NULL,console=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="kruskal_+3A_y">y</code></td>
<td>
<p> response </p>
</td></tr>
<tr><td><code id="kruskal_+3A_trt">trt</code></td>
<td>
<p> treatment </p>
</td></tr>
<tr><td><code id="kruskal_+3A_alpha">alpha</code></td>
<td>
<p> level signification </p>
</td></tr>
<tr><td><code id="kruskal_+3A_p.adj">p.adj</code></td>
<td>
<p> Method for adjusting p values (see p.adjust)  </p>
</td></tr>
<tr><td><code id="kruskal_+3A_group">group</code></td>
<td>
<p> TRUE or FALSE </p>
</td></tr>
<tr><td><code id="kruskal_+3A_main">main</code></td>
<td>
<p> Title </p>
</td></tr>
<tr><td><code id="kruskal_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
</table>


<h3>Details</h3>

<p>For equal or different repetition.<br />
For the adjustment methods, see the function p.adjusted.<br />
p-adj = &quot;none&quot; is t-student.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>statistics</code></td>
<td>
<p>Statistics of the model</p>
</td></tr>
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>means</code></td>
<td>
<p>Statistical summary of the study variable</p>
</td></tr>
<tr><td><code>comparison</code></td>
<td>
<p>Comparison between treatments</p>
</td></tr>
<tr><td><code>groups</code></td>
<td>
<p>Formation of treatment groups</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Practical Nonparametrics Statistics. W.J. Conover, 1999 </p>


<h3>See Also</h3>

<p><code><a href="#topic+BIB.test">BIB.test</a></code>, <code><a href="#topic+DAU.test">DAU.test</a></code>, <code><a href="#topic+duncan.test">duncan.test</a></code>,
<code><a href="#topic+durbin.test">durbin.test</a></code>, <code><a href="#topic+friedman">friedman</a></code>, <code><a href="#topic+HSD.test">HSD.test</a></code>,
<code><a href="#topic+LSD.test">LSD.test</a></code>, <code><a href="#topic+Median.test">Median.test</a></code>, <code><a href="#topic+PBIB.test">PBIB.test</a></code>, 
<code><a href="#topic+REGW.test">REGW.test</a></code>, <code><a href="#topic+scheffe.test">scheffe.test</a></code>, <code><a href="#topic+SNK.test">SNK.test</a></code>,
<code><a href="#topic+waerden.test">waerden.test</a></code>, <code><a href="#topic+waller.test">waller.test</a></code>, <code><a href="#topic+plot.group">plot.group</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(corn)
str(corn)
comparison&lt;-with(corn,kruskal(observation,method,group=TRUE, main="corn"))
comparison&lt;-with(corn,kruskal(observation,method,p.adj="bon",group=FALSE, main="corn"))
</code></pre>

<hr>
<h2 id='kurtosis'> Finding the Kurtosis coefficient </h2><span id='topic+kurtosis'></span>

<h3>Description</h3>

<p>It obtains the value of the kurtosis for a normally distributed variable.
The result is similar to SAS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kurtosis(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="kurtosis_+3A_x">x</code></td>
<td>
<p> a numeric vector </p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>x</code></td>
<td>
<p> The kurtosis of x </p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+skewness">skewness</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
x&lt;-c(3,4,5,2,3,4,5,6,4,NA,7)
kurtosis(x)
# value is -0.1517996
</code></pre>

<hr>
<h2 id='lastC'> Setting the last character of a chain  </h2><span id='topic+lastC'></span>

<h3>Description</h3>

<p>A special function for the group of treatments in the multiple comparison tests. 
Use plot.group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lastC(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lastC_+3A_x">x</code></td>
<td>
<p> letters </p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>x</code></td>
<td>
<p>Returns the last character of a string</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.group">plot.group</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
x&lt;-c("a","ab","b","c","cd")
lastC(x)
# "a" "b" "b" "c" "d"

</code></pre>

<hr>
<h2 id='lateblight'>
LATEBLIGHT - Simulator for potato late blight Version LB2004
</h2><span id='topic+lateblight'></span>

<h3>Description</h3>

<p>LATEBLIGHT is a mathematical model that simulates the effect of weather,
host growth and resistance, and fungicide use on asexual development and
growth of Phytophthora infestans on potato foliage.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lateblight(WS, Cultivar, ApplSys,InocDate, LGR, IniSpor, SR, IE, LP, InMicCol,
MatTime=c('EARLYSEASON','MIDSEASON','LATESEASON'),...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lateblight_+3A_ws">WS</code></td>
<td>
<p>object weather-severity</p>
</td></tr>
<tr><td><code id="lateblight_+3A_cultivar">Cultivar</code></td>
<td>
<p>chr </p>
</td></tr>
<tr><td><code id="lateblight_+3A_applsys">ApplSys</code></td>
<td>
<p>chr </p>
</td></tr>
<tr><td><code id="lateblight_+3A_inocdate">InocDate</code></td>
<td>
<p>days </p>
</td></tr>
<tr><td><code id="lateblight_+3A_lgr">LGR</code></td>
<td>
<p>num, see example </p>
</td></tr>
<tr><td><code id="lateblight_+3A_inispor">IniSpor</code></td>
<td>
<p>num </p>
</td></tr>
<tr><td><code id="lateblight_+3A_sr">SR</code></td>
<td>
<p>num, see example </p>
</td></tr>
<tr><td><code id="lateblight_+3A_ie">IE</code></td>
<td>
<p>num, Initialization infection </p>
</td></tr>
<tr><td><code id="lateblight_+3A_lp">LP</code></td>
<td>
<p>num, latent period </p>
</td></tr>
<tr><td><code id="lateblight_+3A_inmiccol">InMicCol</code></td>
<td>
<p>num </p>
</td></tr>
<tr><td><code id="lateblight_+3A_mattime">MatTime</code></td>
<td>
<p> chr </p>
</td></tr>
<tr><td><code id="lateblight_+3A_...">...</code></td>
<td>
<p> plot graphics parameters </p>
</td></tr>
</table>


<h3>Details</h3>

<p>LATEBLIGHT Version LB2004 was created in October 2004 (Andrade-Piedra et al.,
2005a, b and c), based on the C-version written by B.E. Ticknor ('BET 21191
modification of cbm8d29.c'), reported by Doster et al. (1990) and described
in detail by Fry et al. (1991) (This version is referred as LB1990 by
Andrade-Piedra et al. [2005a]). The first version of LATEBLIGHT was developed
by Bruhn and Fry (1981) and described  in detail by Bruhn et al. (1980).
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Ofile</code></td>
<td>
<p>&quot;Date&quot;,&quot;nday&quot;,&quot;MicCol&quot;,&quot;SimSeverity&quot;,...</p>
</td></tr>
<tr><td><code>Gfile</code></td>
<td>
<p>&quot;dates&quot;,&quot;nday&quot;,&quot;MeanSeverity&quot;,&quot;StDevSeverity&quot;</p>
</td></tr>
</table>


<h3>Note</h3>

<p>All format data for date is yyyy-mm,dd, for example &quot;2000-04-22&quot;.
change with function as.Date()
</p>


<h3>Author(s)</h3>

<p>Jorge L. Andrade-Piedra (1) (j.andrade@cgar.org),
Gregory A. Forbes (1) (g.forbes@cgiar.org),
Robert J. Hijmans (2) (rhijmans@ucdavis.edu),
William E. Fry (3) (wef1@cornell.edu)
Translation from C language into SAS language:  G.A. Forbes
Modifications: J.L. Andrade-Piedra  and R.J. Hijmans
Translation from SAS into R: Felipe de Mendiburu (1)
(1)	International Potato Center, P.O. Box 1558, Lima 12, Peru
(2)	University of California, One Shields Avenue, Davis, California 95616, USA
(3)	Cornell University, 351 Plant Science, Ithaca, NY 14853, USA
</p>


<h3>References</h3>

<p>Andrade-Piedra, J. L., Hijmans, R. J., Forbes, G. A., Fry, W. E.,
and Nelson, R. J. 2005a. Simulation of potato late blight in the Andes.
I: Modification and parameterization of the LATEBLIGHT model. Phytopathology 95:1191-1199.<br />
Andrade-Piedra, J. L., Hijmans, R. J., Juarez, H. S., Forbes,
G. A., Shtienberg, D., and Fry, W. E. 2005b. Simulation of potato late blight
in the Andes. II: Validation of the LATEBLIGHT model. Phytopathology 95:1200-1208.<br />
Andrade-Piedra, J. L., Forbes, G. A., Shtienberg, D., Grunwald, N. J.,
Chacon, M. G., Taipe, M. V., Hijmans, R. J., and Fry, W. E. 2005c.
Qualification of a plant disease simulation model: Performance of the LATEBLIGHT
model across a broad range of environments. Phytopathology 95:1412-1422.<br />
Bruhn, J.A., Bruck, R.I., Fry, W.E., Arneson, P.A., and Keokosky, E.V. 1980.
User's manual for LATEBLIGHT: a plant disease management game. Cornell University,
Department of Plant Pathology, Ithaca, NY, USA. Mimeo 80-1.<br />
Bruhn, J.A., and Fry, W.E.  1981.  Analysis of potato late blight epidemiology
by simulation modeling.  Phytopathology 71:612-616.<br />
Doster, M. A., Milgroom, M. G., and Fry, W. E. 1990. Quantification of factors
influencing potato late blight suppression and selection for metalaxyl resistance
in Phytophthora infestans - A simulation approach. Phytopathology 80:1190-1198.<br />
Fry, W.E., Milgroom, M.G., Doster, M.A., Bruhn, J.A., and Bruck, R.I.  1991.
LATEBLIGHT:  a plant disease management game - User Manual.  Version 3.1.
Microsoft Windows Adaptation by B. E. Ticknor, and P. A. Arneson.  Ithaca,
Cornell University, Department of Plant Pathology, Ithaca, NY, USA.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+weatherSeverity">weatherSeverity</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
f &lt;- system.file("external/weather.csv", package="agricolae")
weather &lt;- read.csv(f,header=FALSE)
f &lt;- system.file("external/severity.csv", package="agricolae")
severity &lt;- read.csv(f)
weather[,1]&lt;-as.Date(weather[,1],format = "%m/%d/%Y")
# Parameters dates
dates&lt;-c("2000-03-25","2000-04-09","2000-04-12","2000-04-16","2000-04-22")
dates&lt;-as.Date(dates)
EmergDate &lt;- as.Date('2000/01/19')
EndEpidDate &lt;- as.Date("2000-04-22")
dates&lt;-as.Date(dates)
NoReadingsH&lt;- 1
RHthreshold &lt;- 90
WS&lt;-weatherSeverity(weather,severity,dates,EmergDate,EndEpidDate,
NoReadingsH,RHthreshold)
# Parameters Lateblight
InocDate&lt;-"2000-03-18"
LGR &lt;- 0.00410
IniSpor &lt;- 0
SR &lt;- 292000000
IE &lt;- 1.0
LP &lt;- 2.82
InMicCol &lt;- 9
Cultivar &lt;- 'NICOLA'
ApplSys &lt;- "NOFUNGICIDE"
main&lt;-"Cultivar: NICOLA"
#--------------------------
model&lt;-lateblight(WS, Cultivar,ApplSys, InocDate, LGR,IniSpor,SR,IE, LP,
MatTime='LATESEASON',InMicCol,main=main,type="l",xlim=c(65,95),lwd=1.5,
xlab="Time (days after emergence)", ylab="Severity (Percentage)")
# reproduce graph
x&lt;- model$Ofile$nday
y&lt;- model$Ofile$SimSeverity
w&lt;- model$Gfile$nday
z&lt;- model$Gfile$MeanSeverity
Min&lt;-model$Gfile$MinObs
Max&lt;-model$Gfile$MaxObs
plot(x,y,type="l",xlim=c(65,95),lwd=1.5,xlab="Time (days after emergence)",
ylab="Severity (Percentage)")
points(w,z,col="blue",cex=1,pch=19)
npoints &lt;- length(w)
for ( i in 1:npoints){
segments(w[i],Min[i],w[i],Max[i],lwd=1.5,col="blue")
}
legend("topleft",c("Disease progress curves","Weather-Severity"),
title="Description",lty=1,pch=c(3,19),col=c("black","blue"))

</code></pre>

<hr>
<h2 id='lineXtester'> Line x Tester Analysis </h2><span id='topic+lineXtester'></span>

<h3>Description</h3>

<p>It makes the Line x Tester Genetic Analysis. It also estimates the general and 
specific combinatory ability effects and the line and tester genetic contribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lineXtester(replications, lines, testers, y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lineXtester_+3A_replications">replications</code></td>
<td>
<p> Replications</p>
</td></tr>
<tr><td><code id="lineXtester_+3A_lines">lines</code></td>
<td>
<p> Lines </p>
</td></tr>
<tr><td><code id="lineXtester_+3A_testers">testers</code></td>
<td>
<p> Testers </p>
</td></tr>
<tr><td><code id="lineXtester_+3A_y">y</code></td>
<td>
<p> Variable, response </p>
</td></tr>
</table>


<h3>Details</h3>

<p>ANOVA with parents and crosses <br />
ANOVA for line X tester analysis <br />
ANOVA for line X tester analysis including parents <br />
GCA Effects: Lines Effects, Testers Effects and SCA Effects. <br />
Standard Errors for Combining Ability Effects. <br />
Genetic Components. <br />  
...<br />
Proportional contribution of lines, testers and their interactions to total variance 
</p>


<h3>Value</h3>

<p>return anova(formula = Y ~ Replications + Treatments).<br />
where the Treatments contains parents, crosses and crosses vs Parents.<br />
The crosses contains Lines, Testers and its interaction .
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

 
<p>Biometrical Methods in Quantitative Genetic Analysis, Singh, Chaudhary. 1979.
Hierarchial and factorial mating designs for quantitative genetic analysis in tetrasomic potato. R. Ortis; A.Golmirzaie. Theor Appl Genet (2002) 104:675-679 </p>


<h3>See Also</h3>

<p><code><a href="#topic+AMMI">AMMI</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'># see structure line by testers
library(agricolae)
# example 1
data(heterosis)
site1&lt;-subset(heterosis,heterosis[,1]==1)
output1&lt;-with(site1,lineXtester(Replication, Female, Male, v2))
# example 2
data(LxT)
str(LxT)
output2&lt;-with(LxT,lineXtester(replication, line, tester, yield))
</code></pre>

<hr>
<h2 id='LSD.test'> Multiple comparisons, &quot;Least significant difference&quot; and Adjust P-values </h2><span id='topic+LSD.test'></span>

<h3>Description</h3>

<p>Multiple comparisons of treatments by means of LSD and a grouping of treatments. 
The level by alpha default is 0.05. Returns p-values adjusted using one of several methods
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LSD.test(y, trt, DFerror, MSerror, alpha = 0.05, p.adj=c("none","holm","hommel", 
"hochberg", "bonferroni", "BH", "BY", "fdr"), group=TRUE, main = NULL,console=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="LSD.test_+3A_y">y</code></td>
<td>
<p> model(aov or lm) or answer of the experimental unit  </p>
</td></tr>
<tr><td><code id="LSD.test_+3A_trt">trt</code></td>
<td>
<p> Constant( only y=model) or vector treatment applied to each experimental unit  </p>
</td></tr>
<tr><td><code id="LSD.test_+3A_dferror">DFerror</code></td>
<td>
<p> Degrees of freedom of the experimental error </p>
</td></tr>
<tr><td><code id="LSD.test_+3A_mserror">MSerror</code></td>
<td>
<p> Means square error of the experimental  </p>
</td></tr>
<tr><td><code id="LSD.test_+3A_alpha">alpha</code></td>
<td>
<p> Level of risk for the test  </p>
</td></tr>
<tr><td><code id="LSD.test_+3A_p.adj">p.adj</code></td>
<td>
<p> Method for adjusting p values (see p.adjust)  </p>
</td></tr>  
<tr><td><code id="LSD.test_+3A_group">group</code></td>
<td>
<p> TRUE or FALSE </p>
</td></tr>
<tr><td><code id="LSD.test_+3A_main">main</code></td>
<td>
<p> title of the study  </p>
</td></tr>
<tr><td><code id="LSD.test_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
</table>


<h3>Details</h3>

<p>For equal or different repetition.<br />
For the adjustment methods, see the function p.adjusted.<br />
p-adj =&quot;none&quot; is t-student.<br />
</p>
<p>It is necessary first makes a analysis of variance.<br />
if model=y, then to apply the instruction:<br /> 
LSD.test(model, &quot;trt&quot;, alpha = 0.05, p.adj=c(&quot;none&quot;,&quot;holm&quot;,&quot;hommel&quot;, 
&quot;hochberg&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;), group=TRUE, main = NULL,console=FALSE)<br />
where the model class is aov or lm.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>statistics</code></td>
<td>
<p>Statistics of the model</p>
</td></tr>
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>means</code></td>
<td>
<p>Statistical summary of the study variable</p>
</td></tr>
<tr><td><code>comparison</code></td>
<td>
<p>Comparison between treatments</p>
</td></tr>
<tr><td><code>groups</code></td>
<td>
<p>Formation of treatment groups</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p>Steel, R.; Torri,J; Dickey, D.(1997)
Principles and Procedures of Statistics
A Biometrical Approach. pp178.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BIB.test">BIB.test</a></code>, <code><a href="#topic+DAU.test">DAU.test</a></code>, <code><a href="#topic+duncan.test">duncan.test</a></code>,
<code><a href="#topic+durbin.test">durbin.test</a></code>, <code><a href="#topic+friedman">friedman</a></code>, <code><a href="#topic+HSD.test">HSD.test</a></code>,
<code><a href="#topic+kruskal">kruskal</a></code>, <code><a href="#topic+Median.test">Median.test</a></code>, <code><a href="#topic+PBIB.test">PBIB.test</a></code>, 
<code><a href="#topic+REGW.test">REGW.test</a></code>, <code><a href="#topic+scheffe.test">scheffe.test</a></code>, <code><a href="#topic+SNK.test">SNK.test</a></code>,
<code><a href="#topic+waerden.test">waerden.test</a></code>, <code><a href="#topic+waller.test">waller.test</a></code>, <code><a href="#topic+plot.group">plot.group</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(sweetpotato)
model&lt;-aov(yield~virus, data=sweetpotato)
out &lt;- LSD.test(model,"virus", p.adj="bonferroni")
#stargraph
# Variation range: max and min
plot(out)
#endgraph
# Old version LSD.test()
df&lt;-df.residual(model)
MSerror&lt;-deviance(model)/df
out &lt;- with(sweetpotato,LSD.test(yield,virus,df,MSerror))
#stargraph
# Variation interquartil range: Q75 and Q25
plot(out,variation="IQR")
#endgraph
out&lt;-LSD.test(model,"virus",p.adj="hommel",console=TRUE)
plot(out,variation="SD") # variation standard deviation
</code></pre>

<hr>
<h2 id='LxT'> Data Line by tester</h2><span id='topic+LxT'></span>

<h3>Description</h3>

<p>Data frame with yield by line x tester.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(LxT)</code></pre>


<h3>Format</h3>

<p>A data frame with 92 observations on the following 4 variables.
</p>

<dl>
<dt>replication</dt><dd><p>a numeric vector</p>
</dd>
<dt>line</dt><dd><p>a numeric vector</p>
</dd>
<dt>tester</dt><dd><p>a numeric vector</p>
</dd>
<dt>yield</dt><dd><p>a numeric vector</p>
</dd></dl>



<h3>Source</h3>

<p>Biometrical Methods in Quantitative Genetic Analysis, Singh, Chaudhary. 1979
</p>

<hr>
<h2 id='markers'> Data of molecular markers</h2><span id='topic+markers'></span>

<h3>Description</h3>

<p>A partial study on 27 molecular markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(markers)</code></pre>


<h3>Format</h3>

<p>A data frame with 23 observations on the following 27 variables.
</p>

<dl>
<dt><code>marker1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker13</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker14</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker15</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker16</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker17</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker18</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker19</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker20</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker21</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker22</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker23</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker24</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker25</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker26</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>marker27</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>International Potato Center Lima-Peru.
</p>


<h3>References</h3>

<p>International Potato Center Lima-Peru.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(markers)
str(markers)
</code></pre>

<hr>
<h2 id='Median.test'> Median test. Multiple comparisons </h2><span id='topic+Median.test'></span>

<h3>Description</h3>

<p>A nonparametric test for several independent samples. The median test is designed
to examine whether several samples came from populations having the same median.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Median.test(y,trt,alpha=0.05,correct=TRUE,simulate.p.value = FALSE, group = TRUE, 
main = NULL,console=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Median.test_+3A_y">y</code></td>
<td>
<p> Variable response </p>
</td></tr>
<tr><td><code id="Median.test_+3A_trt">trt</code></td>
<td>
<p> Treatments </p>
</td></tr>
<tr><td><code id="Median.test_+3A_alpha">alpha</code></td>
<td>
<p> error type I </p>
</td></tr>
<tr><td><code id="Median.test_+3A_correct">correct</code></td>
<td>
<p> a logical indicating whether to apply continuity correction
when computing the test statistic for 2 groups. The correction will not be bigger
than the differences themselves. No correction is done if simulate.p.value = TRUE.</p>
</td></tr>
<tr><td><code id="Median.test_+3A_simulate.p.value">simulate.p.value</code></td>
<td>
<p>a logical indicating whether to compute p-values by
Monte Carlo simulation</p>
</td></tr>
<tr><td><code id="Median.test_+3A_group">group</code></td>
<td>
<p> TRUE or FALSE </p>
</td></tr>
<tr><td><code id="Median.test_+3A_main">main</code></td>
<td>
<p> Title </p>
</td></tr>
<tr><td><code id="Median.test_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The data consist of k samples of possibly unequal sample size.<br />
Greater: is the number of values that exceed the median of all data and <br />
LessEqual: is the number less than or equal to the median of all data.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>statistics</code></td>
<td>
<p>Statistics of the model</p>
</td></tr>
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>medians</code></td>
<td>
<p>Statistical summary of the study variable</p>
</td></tr>
<tr><td><code>comparison</code></td>
<td>
<p>Comparison between treatments</p>
</td></tr>
<tr><td><code>groups</code></td>
<td>
<p>Formation of treatment groups</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Practical Nonparametrics Statistics. W.J. Conover, 1999 </p>


<h3>See Also</h3>

<p><code><a href="#topic+BIB.test">BIB.test</a></code>, <code><a href="#topic+DAU.test">DAU.test</a></code>, <code><a href="#topic+duncan.test">duncan.test</a></code>,
<code><a href="#topic+durbin.test">durbin.test</a></code>, <code><a href="#topic+friedman">friedman</a></code>, <code><a href="#topic+HSD.test">HSD.test</a></code>,
<code><a href="#topic+kruskal">kruskal</a></code>, <code><a href="#topic+LSD.test">LSD.test</a></code>, <code><a href="#topic+PBIB.test">PBIB.test</a></code>, 
<code><a href="#topic+REGW.test">REGW.test</a></code>, <code><a href="#topic+scheffe.test">scheffe.test</a></code>, <code><a href="#topic+SNK.test">SNK.test</a></code>,
<code><a href="#topic+waerden.test">waerden.test</a></code>, <code><a href="#topic+waller.test">waller.test</a></code>, <code><a href="#topic+plot.group">plot.group</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
# example 1
data(corn)
out&lt;-with(corn,Median.test(observation,method,console=FALSE))
z&lt;-bar.err(out$medians,variation = "range",ylim=c(0,120),
           space=2,border=4,col=3,density=10,angle=45)
# example 2
out&lt;-with(corn,Median.test(observation,method,console=FALSE,group=FALSE))
print(out$comparison)
</code></pre>

<hr>
<h2 id='melon'> Data of yield of melon in a Latin square experiment </h2><span id='topic+melon'></span>

<h3>Description</h3>

<p>An irrigation system evaluation by exudation using four varieties of 
melon, under modality of sowing, SIMPLE ROW.
The goal is to analyze the behavior of three hybrid melon varieties 
and one standard.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(melon)</code></pre>


<h3>Format</h3>

<p>A data frame with 16 observations on the following 4 variables.
</p>

<dl>
<dt><code>row</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>col</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>variety</code></dt><dd><p>a factor with levels <code>V1</code> <code>V2</code> <code>V3</code> <code>V4</code></p>
</dd>
<dt><code>yield</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>Varieties:	
Hibrido Mission (V1); Hibrido Mark (V2); Hibrido Topfligth (V3); Hibrido Hales Best Jumbo (V4).
</p>


<h3>Source</h3>

<p>Tesis. &quot;Evaluacion del sistema de riego por exudacion utilizando cuatro variedades
de melon, bajo modalidad de siembra, SIMPLE HILERA&quot;. Alberto Angeles L. 
Universidad Agraria la Molina - Lima Peru.
</p>


<h3>References</h3>

<p>Universidad Nacional Agraria la molina.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(melon)
str(melon)
</code></pre>

<hr>
<h2 id='montecarlo'> Random generation by Montecarlo </h2><span id='topic+montecarlo'></span>

<h3>Description</h3>

<p>Random generation form data, use function density and parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>montecarlo(data, k, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="montecarlo_+3A_data">data</code></td>
<td>
<p> vector  or object(hist, graph.freq) </p>
</td></tr>
<tr><td><code id="montecarlo_+3A_k">k</code></td>
<td>
<p> number of simulations </p>
</td></tr>
<tr><td><code id="montecarlo_+3A_...">...</code></td>
<td>
<p> Other parameters of the function density, only if data is vector </p>
</td></tr>  
</table>


<h3>Value</h3>

<p>Generate random numbers with empirical distribution.
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+density">density</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
r&lt;-rnorm(50, 10,2)
montecarlo(r, k=100, kernel="epanechnikov")
# other example
h&lt;-hist(r,plot=FALSE)
montecarlo(h, k=100)
# other example
breaks&lt;-c(0, 150, 200, 250, 300)
counts&lt;-c(10, 20, 40, 30)
op&lt;-par(mfrow=c(1,2),cex=0.8,mar=c(2,3,0,0))
h1&lt;-graph.freq(x=breaks,counts=counts,plot=FALSE)
r&lt;-montecarlo(h, k=1000)
plot(h1,frequency = 3,ylim=c(0,0.008))
text(90,0.006,"Population\n100 obs.")
h2&lt;-graph.freq(r,breaks,frequency = 3,ylim=c(0,0.008))
lines(density(r),col="blue")
text(90,0.006,"Montecarlo\n1000 obs.")
par(op)
</code></pre>

<hr>
<h2 id='natives'> Data of native potato </h2><span id='topic+natives'></span>

<h3>Description</h3>

<p>An evaluation of the number, weight and size of 24 native potatoes varieties.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(natives)</code></pre>


<h3>Format</h3>

<p>A data frame with 876 observations on the following 4 variables.
</p>

<dl>
<dt><code>variety</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>number</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>weight</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>size</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>International Potato Center. CIP - Lima Peru.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(natives)
str(natives)
</code></pre>

<hr>
<h2 id='nonadditivity'> Nonadditivity model test </h2><span id='topic+nonadditivity'></span>

<h3>Description</h3>

<p>The resistance for the transformable nonadditivity, due to J. W. Tukey, 
is based on the detection of a curvilinear relation between y-est(y)
and est(y). A freedom degree for the transformable nonadditivity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nonadditivity(y, factor1, factor2, df, MSerror)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nonadditivity_+3A_y">y</code></td>
<td>
<p> Answer of the experimental unit  </p>
</td></tr>
<tr><td><code id="nonadditivity_+3A_factor1">factor1</code></td>
<td>
<p> Firts treatment applied to each experimental unit </p>
</td></tr>
<tr><td><code id="nonadditivity_+3A_factor2">factor2</code></td>
<td>
<p> Second treatment applied to each experimental unit </p>
</td></tr>
<tr><td><code id="nonadditivity_+3A_df">df</code></td>
<td>
<p> Degrees of freedom of the experimental error </p>
</td></tr>
<tr><td><code id="nonadditivity_+3A_mserror">MSerror</code></td>
<td>
<p> Means square error of the experimental </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Only two factor: Block and treatment or factor 1 and factor 2.
</p>


<h3>Value</h3>

<p>P, Q and non-additivity analysis of variance 
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

 
<p>1. Steel, R.; Torri,J; Dickey, D.(1997)
Principles and Procedures of Statistics
A Biometrical Approach
</p>
<p>2. George E.P. Box; J. Stuart Hunter and William G. Hunter. 
Statistics for experimenters. 
Wile Series in probability and statistics </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(potato )
potato[,1]&lt;-as.factor(potato[,1])
model&lt;-lm(cutting ~ date + variety,potato)
df&lt;-df.residual(model)
MSerror&lt;-deviance(model)/df
analysis&lt;-with(potato,nonadditivity(cutting, date, variety, df, MSerror))
</code></pre>

<hr>
<h2 id='normal.freq'> Normal curve on the histogram </h2><span id='topic+normal.freq'></span>

<h3>Description</h3>

<p>A normal distribution graph elaborated from the histogram previously
constructed. The average and variance are obtained from the data grouped
in the histogram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normal.freq(histogram, frequency=1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="normal.freq_+3A_histogram">histogram</code></td>
<td>
<p> object constructed by the function hist</p>
</td></tr>
<tr><td><code id="normal.freq_+3A_frequency">frequency</code></td>
<td>
<p> 1=counts, 2=relative, 3=density </p>
</td></tr>
<tr><td><code id="normal.freq_+3A_...">...</code></td>
<td>
<p> Other parameters of the function hist </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+polygon.freq">polygon.freq</a></code>, <code><a href="#topic+table.freq">table.freq</a></code>, <code><a href="#topic+stat.freq">stat.freq</a></code>,
<code><a href="#topic+inter.freq">inter.freq</a></code>, <code><a href="#topic+sturges.freq">sturges.freq</a></code>, <code><a href="#topic+join.freq">join.freq</a></code>, 
<code><a href="#topic+ogive.freq">ogive.freq</a></code>, <code><a href="#topic+graph.freq">graph.freq</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(growth)
#startgraph
h1&lt;-with(growth,hist(height,col="green",xlim=c(6,14)))
normal.freq(h1,col="blue")
#endgraph
#startgraph
h2&lt;-with(growth,graph.freq(height,col="yellow",xlim=c(6,14),frequency=2))
normal.freq(h2,frequency=2)
#endgraph
</code></pre>

<hr>
<h2 id='ogive.freq'> Plotting the ogive from a histogram </h2><span id='topic+ogive.freq'></span>

<h3>Description</h3>

<p>It plots the cumulative relative .frequencies with the intervals of classes
defined in the histogram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ogive.freq(histogram,type="",xlab="",ylab="",axes="",las=1,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ogive.freq_+3A_histogram">histogram</code></td>
<td>
<p> object created by the function hist() or graph.freq()</p>
</td></tr>
<tr><td><code id="ogive.freq_+3A_type">type</code></td>
<td>
<p> what type of plot should be drawn. See plot()</p>
</td></tr>
<tr><td><code id="ogive.freq_+3A_xlab">xlab</code></td>
<td>
<p> x labels</p>
</td></tr>
<tr><td><code id="ogive.freq_+3A_ylab">ylab</code></td>
<td>
<p> y labels</p>
</td></tr>  
<tr><td><code id="ogive.freq_+3A_axes">axes</code></td>
<td>
<p> TRUE or FALSE</p>
</td></tr>
<tr><td><code id="ogive.freq_+3A_las">las</code></td>
<td>
<p> values 0,1,2 and 3 are the axis styles. see plot()</p>
</td></tr> 
<tr><td><code id="ogive.freq_+3A_...">...</code></td>
<td>
<p> Parameters of the plot() </p>
</td></tr>
</table>


<h3>Value</h3>

<p>Ogive points.
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+polygon.freq">polygon.freq</a></code>, <code><a href="#topic+table.freq">table.freq</a></code>, <code><a href="#topic+stat.freq">stat.freq</a></code>,
<code><a href="#topic+inter.freq">inter.freq</a></code>, <code><a href="#topic+sturges.freq">sturges.freq</a></code>, <code><a href="#topic+join.freq">join.freq</a></code>, 
<code><a href="#topic+graph.freq">graph.freq</a></code>, <code><a href="#topic+normal.freq">normal.freq</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(growth)
h&lt;-graph.freq(growth$height,plot=FALSE)
points&lt;-ogive.freq(h,col="red",frame=FALSE,
xlab="Height", ylab="Accumulated relative frequency", main="ogive")
plot(points,type="b",pch=16,las=1,bty="l")
</code></pre>

<hr>
<h2 id='order.group'> Ordering the treatments according to the multiple comparison </h2><span id='topic+order.group'></span>

<h3>Description</h3>

<p>This function allows us to compare the treatments averages or the adding of 
their ranges with the minimal significant difference which can vary from one
comparison to another one.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>order.group(trt, means, N, MSerror, Tprob, std.err, parameter=1, snk=0, 
DFerror=NULL,alpha=NULL,sdtdif=NULL,vartau=NULL,console)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="order.group_+3A_trt">trt</code></td>
<td>
<p> Treatments </p>
</td></tr>
<tr><td><code id="order.group_+3A_means">means</code></td>
<td>
<p> Means of treatment </p>
</td></tr>
<tr><td><code id="order.group_+3A_n">N</code></td>
<td>
<p> Replications </p>
</td></tr>
<tr><td><code id="order.group_+3A_mserror">MSerror</code></td>
<td>
<p> Mean square error </p>
</td></tr>
<tr><td><code id="order.group_+3A_tprob">Tprob</code></td>
<td>
<p> minimum value for the comparison </p>
</td></tr>
<tr><td><code id="order.group_+3A_std.err">std.err</code></td>
<td>
<p> standard error </p>
</td></tr>
<tr><td><code id="order.group_+3A_parameter">parameter</code></td>
<td>
<p> Constante 1 (Sd), 0.5 (Sx) </p>
</td></tr>
<tr><td><code id="order.group_+3A_snk">snk</code></td>
<td>
<p> Constante = 1 (Student Newman Keuls) </p>
</td></tr>
<tr><td><code id="order.group_+3A_dferror">DFerror</code></td>
<td>
<p> Degrees of freedom of the experimental error </p>
</td></tr>
<tr><td><code id="order.group_+3A_alpha">alpha</code></td>
<td>
<p> Level of risk for the test  </p>
</td></tr> 
<tr><td><code id="order.group_+3A_sdtdif">sdtdif</code></td>
<td>
<p> standard deviation of difference in BIB  </p>
</td></tr>
<tr><td><code id="order.group_+3A_vartau">vartau</code></td>
<td>
<p> matrix var-cov in PBIB  </p>
</td></tr> 
<tr><td><code id="order.group_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function was changed by orderPvalue function that use agricolae.
Now the grouping in agricolae is with the probability of the treatments 
differences and alpha level.
</p>


<h3>Value</h3>

<p>The output is data frame.
</p>
<table role = "presentation">
<tr><td><code>trt</code></td>
<td>
<p> Treatment Levels, Factor </p>
</td></tr>
<tr><td><code>means</code></td>
<td>
<p> height, Numeric </p>
</td></tr>
<tr><td><code>M</code></td>
<td>
<p> groups levels, Factor </p>
</td></tr>
<tr><td><code>N</code></td>
<td>
<p> replications, Numeric </p>
</td></tr>
<tr><td><code>std.err</code></td>
<td>
<p> Standard error, Numeric </p>
</td></tr>
</table>


<h3>Note</h3>

<p>It is considered 81 labels as maximum for the formation of groups, 
greater number will not have label.
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+orderPvalue">orderPvalue</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
treatments &lt;- c("A","B","C","D","E","F")
means&lt;-c(20,40,35,72,49,58)
std.err&lt;-c(1.2, 2, 1.5, 2.4, 1, 3.1)
replications &lt;- c(4,4,3,4,3,3)
MSerror &lt;- 55.8
value.t &lt;- 2.1314
groups&lt;-order.group(treatments,means,replications,MSerror,value.t,std.err,console=FALSE)
print(groups)
</code></pre>

<hr>
<h2 id='orderPvalue'> Grouping the treatments averages in a comparison with a minimum value  </h2><span id='topic+orderPvalue'></span>

<h3>Description</h3>

<p>When there are treatments and their respective values, these can be 
compared with a minimal difference of meaning.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>orderPvalue(treatment, means, alpha, pvalue, console)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="orderPvalue_+3A_treatment">treatment</code></td>
<td>
<p> treatment </p>
</td></tr>
<tr><td><code id="orderPvalue_+3A_means">means</code></td>
<td>
<p> means of treatment </p>
</td></tr>
<tr><td><code id="orderPvalue_+3A_alpha">alpha</code></td>
<td>
<p> Alpha value, significante value to comparison </p>
</td></tr>
<tr><td><code id="orderPvalue_+3A_pvalue">pvalue</code></td>
<td>
<p> Matrix of probabilities to comparison </p>
</td></tr>
<tr><td><code id="orderPvalue_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
</table>


<h3>Value</h3>

<p>The means and groups for treatments.
</p>


<h3>Note</h3>

<p>It is considered 81 labels as maximum for the formation of groups, 
greater number will not have label.
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
treatments &lt;- c("A","B","C")
means&lt;-c(2,5,3)
alpha &lt;- 0.05
pvalue&lt;-matrix(1,nrow=3,ncol=3)
pvalue[1,2]&lt;-pvalue[2,1]&lt;-0.03
pvalue[1,3]&lt;-pvalue[3,1]&lt;-0.10
pvalue[2,3]&lt;-pvalue[3,2]&lt;-0.06
out&lt;-orderPvalue(treatments,means,alpha,pvalue,console=TRUE)
barplot(out[,1],names.arg = row.names(out),col=colors()[84:87])
legend("topright",as.character(out$groups),pch=15,col=colors()[84:87],box.col=0)
</code></pre>

<hr>
<h2 id='pamCIP'> Data Potato Wild</h2><span id='topic+pamCIP'></span>

<h3>Description</h3>

<p>Potato Wild
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pamCIP)</code></pre>


<h3>Format</h3>

<p>A data frame with 43 observations on the following 107 variables.
Rownames: code and genotype's name. 
column data: molecular markers.
</p>


<h3>Details</h3>

<p>To study the molecular markers in Wild.
</p>


<h3>Source</h3>

<p>Laboratory data.
</p>


<h3>References</h3>

<p>International Potato Center Lima-Peru (CIP)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(pamCIP)
str(pamCIP)
</code></pre>

<hr>
<h2 id='paracsho'> Data of Paracsho biodiversity </h2><span id='topic+paracsho'></span>

<h3>Description</h3>

<p>A locality in Peru. A biodiversity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(paracsho)</code></pre>


<h3>Format</h3>

<p>A data frame with 110 observations on the following 6 variables.
</p>

<dl>
<dt><code>date</code></dt><dd><p>a factor with levels <code>15-12-05</code> <code>17-11-05</code> <code>18-10-05</code> <code>20-09-05</code> <code>22-06-05</code> <code>23-08-05</code> <code>28-07-05</code></p>
</dd>
<dt><code>plot</code></dt><dd><p>a factor with levels <code>PARACSHO</code></p>
</dd>
<dt><code>Treatment</code></dt><dd><p>a factor with levels <code>CON</code> <code>SIN</code></p>
</dd>
<dt><code>Orden</code></dt><dd><p>a factor with levels <code>COLEOPTERA</code> <code>DIPTERA</code> <code>HEMIPTERA</code> <code>HYMENOPTERA</code> <code>LEPIDOPTERA</code> <code>NEUROPTERA</code> <code>NEUROPTERO</code> <code>NOCTUIDAE</code></p>
</dd>
<dt><code>Family</code></dt><dd><p>a factor with levels <code>AGROMYZIDAE</code> <code>ANTHOCORIDAE</code> <code>ANTHOMYIIDAE</code> <code>ANTHOMYLIDAE</code> <code>BLEPHAROCERIDAE</code> <code>BRACONIDAE</code> <code>BROCONIDAE</code> <code>CALUPHORIDAE</code> <code>CECIDOMYIDAE</code> <code>CHENEUMONIDAE</code> <code>CHNEUMONIDAE</code> <code>CHRYOMELIDAE</code> <code>CICADELLIDAE</code> <code>CULICIDAE</code> <code>ERIOCPAMIDAE</code> <code>HEMEROBIIDAE</code> <code>ICHNEUMONIDAE</code> <code>LOUCHAPIDAE</code> <code>MIRIDAE</code> <code>MUSCIDAE</code> <code>MUSICADAE</code> <code>MUSLIDAE</code> <code>MYCETOPHILIDAE</code> <code>MYCETOPHILIIDAE</code> <code>NENPHALIDAE</code> <code>NOCLUIDAE</code> <code>NOCTERIDAE</code> <code>NOCTUIDAE</code> <code>PERALIDAE</code> <code>PIPUNCULIDAE</code> <code>PROCTOTRUPIDAE</code> <code>PSYLLIDAE</code> <code>PYRALIDAE</code> <code>SARCOPHAGIDAE</code> <code>SARCOPILAGIDAE</code> <code>SCATOPHAGIDAE</code> <code>SCATOPHOGIDAE</code> <code>SCIARIDAE</code> <code>SERSIDAE</code> <code>SYRPHIDAE</code> <code>TACHINIDAE</code> <code>TIPULIDAE</code></p>
</dd>
<dt><code>Number.of.specimens</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>Country Peru, Deparment Junin, province Tarma, locality Huasahuasi.
</p>


<h3>Source</h3>

<p>Entomology dataset.
</p>


<h3>References</h3>

<p>International Potato Center.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(paracsho)
str(paracsho)
</code></pre>

<hr>
<h2 id='path.analysis'> Path Analysis </h2><span id='topic+path.analysis'></span>

<h3>Description</h3>

<p>If the cause and effect relationship is well defined, it is possible
to represent the whole system of variables in a diagram form known as 
path-analysis. The function calculates the direct and indirect effects
and uses the variables correlation or covariance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>path.analysis(corr.x, corr.y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="path.analysis_+3A_corr.x">corr.x</code></td>
<td>
<p> Matrix of correlations of the independent variables </p>
</td></tr>
<tr><td><code id="path.analysis_+3A_corr.y">corr.y</code></td>
<td>
<p> vector of dependent correlations with each one of
the independent ones </p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is necessary first to calculate the correlations.
</p>


<h3>Value</h3>

<p>Direct and indirect effects and residual Effect^2.
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Biometrical Methods in Quantitative Genetic Analysis, Singh, Chaudhary. 1979 </p>


<h3>See Also</h3>

<p><code><a href="#topic+correlation">correlation</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'># Path analysis. Multivarial Analysis. Anderson. Prentice Hall, pag 616
library(agricolae)
# Example 1
corr.x&lt;- matrix(c(1,0.5,0.5,1),c(2,2))
corr.y&lt;- rbind(0.6,0.7)
names&lt;-c("X1","X2")
dimnames(corr.x)&lt;-list(names,names)
dimnames(corr.y)&lt;-list(names,"Y")
path.analysis(corr.x,corr.y)
# Example 2
# data of the progress of the disease related bacterial wilt to the ground
# for the component CE Ca K2 Cu
data(wilt)
data(soil)
x&lt;-soil[,c(3,12,14,20)]
y&lt;-wilt[,14]
cor.y&lt;-correlation(y,x)$correlation
cor.x&lt;-correlation(x)$correlation
path.analysis(cor.x,cor.y)

</code></pre>

<hr>
<h2 id='PBIB.test'> Analysis of the Partially Balanced Incomplete Block Design  </h2><span id='topic+PBIB.test'></span>

<h3>Description</h3>

<p>Analysis of variance PBIB and comparison mean adjusted.
Applied to resoluble designs: Lattices and alpha design.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PBIB.test(block,trt,replication,y,k, method=c("REML","ML","VC"), 
test = c("lsd","tukey"), alpha=0.05, console=FALSE, group=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PBIB.test_+3A_block">block</code></td>
<td>
<p> blocks </p>
</td></tr>
<tr><td><code id="PBIB.test_+3A_trt">trt</code></td>
<td>
<p> Treatment </p>
</td></tr>
<tr><td><code id="PBIB.test_+3A_replication">replication</code></td>
<td>
<p> Replication </p>
</td></tr>  
<tr><td><code id="PBIB.test_+3A_y">y</code></td>
<td>
<p> Response  </p>
</td></tr>
<tr><td><code id="PBIB.test_+3A_k">k</code></td>
<td>
<p> Block size  </p>
</td></tr>  
<tr><td><code id="PBIB.test_+3A_method">method</code></td>
<td>
<p> Estimation method: REML, ML and VC </p>
</td></tr>
<tr><td><code id="PBIB.test_+3A_test">test</code></td>
<td>
<p> Comparison treatments </p>
</td></tr>
<tr><td><code id="PBIB.test_+3A_alpha">alpha</code></td>
<td>
<p> Significant test </p>
</td></tr>
<tr><td><code id="PBIB.test_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
<tr><td><code id="PBIB.test_+3A_group">group</code></td>
<td>
<p>logical, groups </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Method of comparison treatment. 
lsd: least significant difference.<br />
tukey: Honestly significant difference.<br />
Estimate: specifies the estimation method for the covariance parameters.<br />
The REML is the default method. The REML specification performs residual (restricted) maximum likelihood, and 
The ML specification performs maximum likelihood, and 
the VC specifications apply only to variance component models.<br />
The PBIB.test() function can be called inside a function (improvement by Nelson Nazzicari, Ph.D. Bioinformatician)
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>ANOVA</code></td>
<td>
<p>Analysis of variance </p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p>Estimation method: REML, ML and VC</p>
</td></tr>
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>  
<tr><td><code>statistics</code></td>
<td>
<p>Statistics of the model</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>Object: estimation model </p>
</td></tr>    
<tr><td><code>Fstat</code></td>
<td>
<p>Criterion AIC and BIC</p>
</td></tr>
<tr><td><code>comparison</code></td>
<td>
<p>Comparison between treatments</p>
</td></tr>
<tr><td><code>means</code></td>
<td>
<p>Statistical summary of the study variable</p>
</td></tr>
<tr><td><code>groups</code></td>
<td>
<p>Formation of treatment groups</p>
</td></tr> 
<tr><td><code>vartau</code></td>
<td>
<p>Variance-Covariance Matrix</p>
</td></tr>   
</table>


<h3>Author(s)</h3>

<p> F. de Mendiburu </p>


<h3>References</h3>

<p>1. Iterative Analysis of Generalizad Lattice Designs.
E.R. Williams (1977) Austral J. Statistics 19(1) 39-42. 
</p>
<p>2. Experimental design. Cochran and Cox. Second edition. 
Wiley Classics Library Edition published 1992
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BIB.test">BIB.test</a></code>, <code><a href="#topic+DAU.test">DAU.test</a></code>, <code><a href="#topic+duncan.test">duncan.test</a></code>,
<code><a href="#topic+durbin.test">durbin.test</a></code>, <code><a href="#topic+friedman">friedman</a></code>, <code><a href="#topic+HSD.test">HSD.test</a></code>,
<code><a href="#topic+kruskal">kruskal</a></code>, <code><a href="#topic+LSD.test">LSD.test</a></code>, <code><a href="#topic+Median.test">Median.test</a></code>, 
<code><a href="#topic+REGW.test">REGW.test</a></code>, <code><a href="#topic+scheffe.test">scheffe.test</a></code>, <code><a href="#topic+SNK.test">SNK.test</a></code>,
<code><a href="#topic+waerden.test">waerden.test</a></code>, <code><a href="#topic+waller.test">waller.test</a></code>, <code><a href="#topic+plot.group">plot.group</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(agricolae)
# alpha design 
Genotype&lt;-c(paste("gen0",1:9,sep=""),paste("gen",10:30,sep=""))
ntr&lt;-length(Genotype)
r&lt;-2
k&lt;-3
s&lt;-10
obs&lt;-ntr*r
b &lt;- s*r
book&lt;-design.alpha(Genotype,k,r,seed=5)
book$book[,3]&lt;- gl(20,3)
dbook&lt;-book$book
# dataset
yield&lt;-c(5,2,7,6,4,9,7,6,7,9,6,2,1,1,3,2,4,6,7,9,8,7,6,4,3,2,2,1,1,2,
     1,1,2,4,5,6,7,8,6,5,4,3,1,1,2,5,4,2,7,6,6,5,6,4,5,7,6,5,5,4)
rm(Genotype)
# not run
# analysis
# require(nlme) # method = REML or LM in PBIB.test and require(MASS) method=VC
model &lt;- with(dbook,PBIB.test(block, Genotype, replication, yield, k=3, method="VC"))
# model$ANOVA
# plot(model,las=2)
</code></pre>

<hr>
<h2 id='plot.AMMI'> PLOT AMMI </h2><span id='topic+plot.AMMI'></span>

<h3>Description</h3>

<p>Biplot AMMI.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AMMI'
plot(x,first=1,second=2,third=3,number=FALSE,gcol=NULL,ecol=NULL,
angle=25,lwd=1.8,length=0.1,xlab=NULL,ylab=NULL,xlim=NULL,ylim=NULL,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.AMMI_+3A_x">x</code></td>
<td>
<p> object AMMI </p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_first">first</code></td>
<td>
<p> position axis x, 0=Y-dependent, 1=PC1, 2=PC2, 3=PC3</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_second">second</code></td>
<td>
<p> position axis y,0=Y-dependent, 1=PC1, 2=PC2, 3=PC3</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_third">third</code></td>
<td>
<p> position axis z,0=Y-dependent, 1=PC1, 2=PC2, 3=PC3</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_number">number</code></td>
<td>
<p> TRUE or FALSE names or number genotypes</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_gcol">gcol</code></td>
<td>
<p> genotype color</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_ecol">ecol</code></td>
<td>
<p> environment color </p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_angle">angle</code></td>
<td>
<p> angle from the shaft of the arrow to the edge of the arrow head </p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_lwd">lwd</code></td>
<td>
<p> parameter line width in function arrow</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_length">length</code></td>
<td>
<p> parameter length in function arrow</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_xlab">xlab</code></td>
<td>
<p> x labels</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_ylab">ylab</code></td>
<td>
<p> y labels</p>
</td></tr> 
<tr><td><code id="plot.AMMI_+3A_xlim">xlim</code></td>
<td>
<p> x limites</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_ylim">ylim</code></td>
<td>
<p> y limites</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_...">...</code></td>
<td>
<p> other parameters of plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Produce graphs biplot.
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+AMMI">AMMI</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(plrv)
model&lt;- with(plrv,AMMI(Locality, Genotype, Rep, Yield))
# biplot PC2 vs PC1
plot(model)
## plot PC1 vs Yield
plot(model,0,1,gcol="blue",ecol="green")
</code></pre>

<hr>
<h2 id='plot.graph.freq'> Histogram </h2><span id='topic+plot.graph.freq'></span>

<h3>Description</h3>

<p>In many situations it has intervals of class defined with its respective
frequencies. By means of this function, the graphic of frequency is obtained 
and it is possible to superpose the normal distribution, polygon of frequency, 
Ojiva and to construct the table of complete frequency.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'graph.freq'
plot(x, breaks=NULL,counts=NULL,frequency=1,plot=TRUE,
nclass=NULL,xlab="",ylab="",axes = "",las=1,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.graph.freq_+3A_x">x</code></td>
<td>
<p> a vector of values, a object hist(), graphFreq() </p>
</td></tr>
<tr><td><code id="plot.graph.freq_+3A_counts">counts</code></td>
<td>
<p> frequency and x is class intervals</p>
</td></tr>
<tr><td><code id="plot.graph.freq_+3A_breaks">breaks</code></td>
<td>
<p> a vector giving the breakpoints between histogram cells </p>
</td></tr>
<tr><td><code id="plot.graph.freq_+3A_frequency">frequency</code></td>
<td>
<p> 1=counts, 2=relative, 3=density </p>
</td></tr>
<tr><td><code id="plot.graph.freq_+3A_plot">plot</code></td>
<td>
<p> logic </p>
</td></tr>
<tr><td><code id="plot.graph.freq_+3A_nclass">nclass</code></td>
<td>
<p> number of classes </p>
</td></tr>
<tr><td><code id="plot.graph.freq_+3A_xlab">xlab</code></td>
<td>
<p> x labels</p>
</td></tr>
<tr><td><code id="plot.graph.freq_+3A_ylab">ylab</code></td>
<td>
<p> y labels</p>
</td></tr> 
<tr><td><code id="plot.graph.freq_+3A_axes">axes</code></td>
<td>
<p> TRUE or FALSE</p>
</td></tr>
<tr><td><code id="plot.graph.freq_+3A_las">las</code></td>
<td>
<p> values 0,1,2 and 3 are the axis styles. see plot()</p>
</td></tr>   
<tr><td><code id="plot.graph.freq_+3A_...">...</code></td>
<td>
<p> other parameters of plot</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>breaks</code></td>
<td>
<p> a vector giving the breakpoints between histogram cells </p>
</td></tr>
<tr><td><code>counts</code></td>
<td>
<p> frequency and x is class intervals</p>
</td></tr>
<tr><td><code>mids</code></td>
<td>
<p> center point in class </p>
</td></tr>
<tr><td><code>relative</code></td>
<td>
<p> Relative frequency, height </p>
</td></tr>
<tr><td><code>density</code></td>
<td>
<p> Density frequency, height  </p>
</td></tr> 
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+polygon.freq">polygon.freq</a></code>, <code><a href="#topic+table.freq">table.freq</a></code>, 
<code><a href="#topic+stat.freq">stat.freq</a></code>,<code><a href="#topic+inter.freq">inter.freq</a></code>,<code><a href="#topic+sturges.freq">sturges.freq</a></code>,
<code><a href="#topic+join.freq">join.freq</a></code>,<code><a href="#topic+ogive.freq">ogive.freq</a></code>, <code><a href="#topic+normal.freq">normal.freq</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(agricolae)
data(genxenv)
yield &lt;- subset(genxenv$YLD,genxenv$ENV==2)
yield &lt;- round(yield,1)
h&lt;- graph.freq(yield,axes=FALSE, frequency=1, ylab="frequency",col="yellow")
axis(1,h$breaks)
axis(2,seq(0,20,0.1))
# To reproduce histogram.
h1 &lt;- plot(h, col="blue", frequency=2,border="red", density=8,axes=FALSE, 
xlab="YIELD",ylab="relative")
axis(1,h$breaks)
axis(2,seq(0,.4,0.1))
# summary, only frecuency
limits &lt;-seq(10,40,5)
frequencies &lt;-c(2,6,8,7,3,4)
#startgraph
h&lt;-graph.freq(limits,counts=frequencies,col="bisque",xlab="Classes")
polygon.freq(h,col="red")
title( main="Histogram and polygon of frequency",
ylab=".frequency")
#endgraph
# Statistics
measures&lt;-stat.freq(h)
print(measures)
# frequency table full
round(table.freq(h),2)
#startgraph
# ogive
ogive.freq(h,col="red",type="b",ylab="Accumulated relative frequency",
xlab="Variable")
# only frequency polygon
h&lt;-graph.freq(limits,counts=frequencies,border=FALSE,col=NULL,xlab="  ",ylab="")
title( main="Polygon of frequency",
xlab="Variable", ylab="Frecuency")
polygon.freq(h,col="blue")
grid(col="brown")
#endgraph
# Draw curve for Histogram
h&lt;- graph.freq(yield,axes=FALSE, frequency=3, ylab="f(yield)",col="yellow")
axis(1,h$breaks)
axis(2,seq(0,0.18,0.03),las=2)
lines(density(yield), col = "red", lwd = 2)
title("Draw curve for Histogram")
</code></pre>

<hr>
<h2 id='plot.group'> Plotting the multiple comparison of means </h2><span id='topic+plot.group'></span>

<h3>Description</h3>

<p>It plots bars of the averages of treatments to compare.
It uses the objects generated by a procedure of comparison like LSD, HSD,
Kruskall, Waller-Duncan, Friedman or Durbin. It can also display the 'average' 
value over each bar in a bar chart.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'group'
plot(x,variation=c("range","IQR","SE","SD"), decreasing = TRUE,
              horiz=FALSE,col=NULL,xlim=NULL,ylim=NULL,main=NULL,cex=NULL,hy=0,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.group_+3A_x">x</code></td>
<td>
<p> Object created by a test of comparison </p>
</td></tr>
<tr><td><code id="plot.group_+3A_variation">variation</code></td>
<td>
<p> in lines by range, IQR, standard deviation or error </p>
</td></tr>
<tr><td><code id="plot.group_+3A_decreasing">decreasing</code></td>
<td>
<p> Logical, decreasing order of the mean </p>
</td></tr>
<tr><td><code id="plot.group_+3A_horiz">horiz</code></td>
<td>
<p> Horizontal or vertical image </p>
</td></tr>  
<tr><td><code id="plot.group_+3A_col">col</code></td>
<td>
<p>line colors</p>
</td></tr>
<tr><td><code id="plot.group_+3A_xlim">xlim</code></td>
<td>
<p>optional, axis x limits</p>
</td></tr>
<tr><td><code id="plot.group_+3A_ylim">ylim</code></td>
<td>
<p>optional, axis y limits</p>
</td></tr>
<tr><td><code id="plot.group_+3A_main">main</code></td>
<td>
<p>optional, main title</p>
</td></tr>
<tr><td><code id="plot.group_+3A_cex">cex</code></td>
<td>
<p>optional, group label size</p>
</td></tr>
<tr><td><code id="plot.group_+3A_hy">hy</code></td>
<td>
<p>optional, default =0, sum group label position</p>
</td></tr>
<tr><td><code id="plot.group_+3A_...">...</code></td>
<td>
<p> Parameters of the function barplot() </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The output is a vector that indicates the position of the treatments on the coordinate axes. 
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+BIB.test">BIB.test</a></code>, <code><a href="#topic+DAU.test">DAU.test</a></code>, <code><a href="#topic+duncan.test">duncan.test</a></code>,
<code><a href="#topic+durbin.test">durbin.test</a></code>, <code><a href="#topic+friedman">friedman</a></code>, <code><a href="#topic+HSD.test">HSD.test</a></code>,
<code><a href="#topic+kruskal">kruskal</a></code>, <code><a href="#topic+LSD.test">LSD.test</a></code>, <code><a href="#topic+Median.test">Median.test</a></code>, 
<code><a href="#topic+PBIB.test">PBIB.test</a></code>, <code><a href="#topic+REGW.test">REGW.test</a></code>, <code><a href="#topic+scheffe.test">scheffe.test</a></code>, 
<code><a href="#topic+SNK.test">SNK.test</a></code>, <code><a href="#topic+waerden.test">waerden.test</a></code>, <code><a href="#topic+waller.test">waller.test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(sweetpotato)
model&lt;-aov(yield~virus,data=sweetpotato)
comparison&lt;- LSD.test(model,"virus",alpha=0.01,group=TRUE)
#startgraph
op&lt;-par(cex=1.5)
plot(comparison,horiz=TRUE,xlim=c(0,50),las=1)
title(cex.main=0.8,main="Comparison between\ntreatment means",xlab="Yield",ylab="Virus")
#endgraph
par(op)
</code></pre>

<hr>
<h2 id='plots'> Data for an analysis in split-plot</h2><span id='topic+plots'></span>

<h3>Description</h3>

<p>Experimental data in blocks, factor A in plots and factor B in sub-plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(plots)</code></pre>


<h3>Format</h3>

<p>A data frame with 18 observations on the following 5 variables.
</p>

<dl>
<dt><code>block</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>plot</code></dt><dd><p>a factor with levels <code>p1</code> <code>p2</code> <code>p3</code> <code>p4</code> <code>p5</code> <code>p6</code></p>
</dd>
<dt><code>A</code></dt><dd><p>a factor with levels <code>a1</code> <code>a2</code></p>
</dd>
<dt><code>B</code></dt><dd><p>a factor with levels <code>b1</code> <code>b2</code> <code>b3</code></p>
</dd>
<dt><code>yield</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>International Potato Center. CIP
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(plots)
str(plots)
plots[,1] &lt;-as.factor(plots[,1])
# split-plot analysis
model &lt;- aov(yield ~ block + A + Error(plot)+ B + A:B, data=plots)
summary(model)
b&lt;-nlevels(plots$B)
a&lt;-nlevels(plots$A)
r&lt;-nlevels(plots$block)
dfa &lt;- df.residual(model$plot)
Ea &lt;-deviance(model$plot)/dfa
dfb &lt;- df.residual(model$Within)
Eb &lt;-deviance(model$Within)/dfb
Eab &lt;- (Ea +(b-1)*Eb)/(b*r)
# Satterthwaite
dfab&lt;-(Ea +(b-1)*Eb)^2/(Ea^2/dfa +((b-1)*Eb)^2/dfb)
# Comparison A, A(b1), A(b2), A(b3)
comparison1 &lt;-with(plots,LSD.test(yield,A,dfa,Ea))
comparison2 &lt;-with(plots,LSD.test(yield[B=="b1"],A[B=="b1"],dfab,Eab))
comparison3 &lt;-with(plots,LSD.test(yield[B=="b2"],A[B=="b2"],dfab,Eab))
comparison4 &lt;-with(plots,LSD.test(yield[B=="b3"],A[B=="b3"],dfab,Eab))
# Comparison B, B(a1), B(a2)
comparison5 &lt;-with(plots,LSD.test(yield,B,dfb,Eb))
comparison6 &lt;-with(plots,LSD.test(yield[A=="a1"],B[A=="a1"],dfb,Eb))
comparison7 &lt;-with(plots,LSD.test(yield[A=="a2"],B[A=="a2"],dfb,Eb))
</code></pre>

<hr>
<h2 id='plrv'> Data clones from the PLRV population </h2><span id='topic+plrv'></span>

<h3>Description</h3>

<p>Six environments: Ayacucho, La Molina 02, San Ramon 02, Huancayo,
La Molina 03, San Ramon 03.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(plrv)</code></pre>


<h3>Format</h3>

<p>A data frame with 504 observations on the following 6 variables.
</p>

<dl>
<dt><code>Genotype</code></dt><dd><p>a factor with levels <code>102.18</code> <code>104.22</code> <code>121.31</code> <code>141.28</code> <code>157.26</code> <code>163.9</code> <code>221.19</code> <code>233.11</code> <code>235.6</code> <code>241.2</code> <code>255.7</code> <code>314.12</code> <code>317.6</code> <code>319.20</code> <code>320.16</code> <code>342.15</code> <code>346.2</code> <code>351.26</code> <code>364.21</code> <code>402.7</code> <code>405.2</code> <code>406.12</code> <code>427.7</code> <code>450.3</code> <code>506.2</code> <code>Canchan</code> <code>Desiree</code> <code>Unica</code></p>
</dd>
<dt><code>Locality</code></dt><dd><p>a factor with levels <code>Ayac</code> <code>Hyo-02</code> <code>LM-02</code> <code>LM-03</code> <code>SR-02</code> <code>SR-03</code></p>
</dd>
<dt><code>Rep</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>WeightPlant</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>WeightPlot</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Yield</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>International Potato Center Lima-Peru
</p>


<h3>References</h3>

<p>International Potato Center Lima-Peru
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(plrv)
str(plrv)
</code></pre>

<hr>
<h2 id='polygon.freq'> The polygon of frequency on the histogram </h2><span id='topic+polygon.freq'></span>

<h3>Description</h3>

<p>The polygon is constructed single or on a histogram.
It is necessary to execute the function previously hist.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>polygon.freq(histogram, frequency=1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="polygon.freq_+3A_histogram">histogram</code></td>
<td>
<p> Object constructed by the function hist </p>
</td></tr>
<tr><td><code id="polygon.freq_+3A_frequency">frequency</code></td>
<td>
<p> numeric, counts(1), relative(2) and density(3) </p>
</td></tr>
<tr><td><code id="polygon.freq_+3A_...">...</code></td>
<td>
<p> Other parameters of the function hist </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu Delgado </p>


<h3>See Also</h3>

<p><code><a href="#topic+polygon.freq">polygon.freq</a></code>, <code><a href="#topic+table.freq">table.freq</a></code>, <code><a href="#topic+stat.freq">stat.freq</a></code>,
<code><a href="#topic+inter.freq">inter.freq</a></code>, <code><a href="#topic+sturges.freq">sturges.freq</a></code>, <code><a href="#topic+join.freq">join.freq</a></code>,
<code><a href="#topic+graph.freq">graph.freq</a></code>, <code><a href="#topic+normal.freq">normal.freq</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(growth)
#startgraph
h1&lt;-with(growth,hist(height,border=FALSE,xlim=c(6,14)))
polygon.freq(h1,frequency=1,col="red")
#endgraph
#startgraph
h2&lt;-with(growth,graph.freq(height,frequency=2,col="yellow",xlim=c(6,14)))
polygon.freq(h2,frequency=2,col="red")
#endgraph
</code></pre>

<hr>
<h2 id='potato'> Data of cutting</h2><span id='topic+potato'></span>

<h3>Description</h3>

<p>A study on the yield of two potato varieties performed at the CIP
experimental station.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(potato)</code></pre>


<h3>Format</h3>

<p>A data frame with 18 observations on the following 4 variables.
</p>

<dl>
<dt><code>date</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>variety</code></dt><dd><p>a factor with levels <code>Canchan</code> <code>Unica</code></p>
</dd>
<dt><code>harvest</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>cutting</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Experimental data.
</p>


<h3>References</h3>

<p>International Potato Center
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(potato)
str(potato)
</code></pre>

<hr>
<h2 id='ralstonia'> Data of assessment of the population in the soil R.solanacearum</h2><span id='topic+ralstonia'></span>

<h3>Description</h3>

<p>The assessment of the population of R.solanacearum on the floor took place 
after 48 hours of infestation, during days 15, 29, 43, 58, and 133 days 
after the infestation soil. More information on soil data(soil).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ralstonia)</code></pre>


<h3>Format</h3>

<p>A data frame with 13 observations on the following 8 variables.
</p>

<dl>
<dt><code>place</code></dt><dd><p>a factor with levels <code>Chmar</code> <code>Chz</code> <code>Cnt1</code> <code>Cnt2</code> <code>Cnt3</code> <code>Hco1</code> <code>Hco2</code> <code>Hco3</code> <code>Hyo1</code> <code>Hyo2</code> <code>Namora</code> <code>SR1</code> <code>SR2</code></p>
</dd>
<dt><code>Day2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Day15</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Day29</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Day43</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Day58</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Day73</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Day133</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>Logarithm average counts of colonies on plates containing half of M-SMSA 
3 repetitions (3 plates by repetition) incubated at 30 degrees centigrade 
for 48 hours. log(1+UFC/g soil) </p>


<h3>Source</h3>

<p>Experimental field, 2004. Data Kindly provided by Dr. Sylvie Priou, Liliam Gutarra and Pedro Aley.
</p>


<h3>References</h3>

<p>International Potato Center. CIP - Lima Peru.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(ralstonia)
str(ralstonia)
</code></pre>

<hr>
<h2 id='reg.homog'> Homologation of regressions </h2><span id='topic+reg.homog'></span>

<h3>Description</h3>

<p>It makes the regressions homogeneity test for a group of treatments where
each observation presents a linearly dependent reply from another one. 
There is a linear function in every treatment.
The objective is to find out if the linear models of each treatment
come from the same population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reg.homog(trt, x, y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reg.homog_+3A_trt">trt</code></td>
<td>
<p> treatment </p>
</td></tr>
<tr><td><code id="reg.homog_+3A_x">x</code></td>
<td>
<p> independent variable </p>
</td></tr>
<tr><td><code id="reg.homog_+3A_y">y</code></td>
<td>
<p> dependent variable </p>
</td></tr>
</table>


<h3>Value</h3>

<p>list objects:<br />
Number regressions.<br />
Residual.<br />
Difference of regression.<br />
DF.homgeneity (homogenity degree free).<br />
DF.Residual (degree free error).<br />
F.value. Test statitics. <br />
P.value. P Value (Significant<br />
Criterion. conclusion<br />
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Book in Spanish: Metodos estadisticos para la investigacion.
Calzada Benza 1960 </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(frijol)
evaluation&lt;-with(frijol,reg.homog(technology,index,production))
# Example 2. Applied Regression Analysis a Research tools
# 1988. John O.Rawlings. Wadsworth &amp; brooks/cole Advanced Books
# &amp; Software. Pacific Grove. Califonia.
# Statistics/probability. Series
LineNumber&lt;-c(rep("39","30"),rep("52","30"))
PlantingDate&lt;-rep(c("16","20","21"),20)
HeadWt &lt;- c(2.5,3.0,2.2,2.2,2.8,1.8,3.1,2.8,1.6,4.3,2.7,2.1,2.5,2.6,3.3,4.3,
2.8,3.8,3.8,2.6,3.2,4.3,2.6,3.6,1.7,2.6,4.2,3.1,3.5,1.6,2.0,4.0,1.5,2.4,2.8,
1.4,1.9,3.1,1.7,2.8,4.2,1.3,1.7,3.7,1.7,3.2,3.0,1.6,2.0,2.2,1.4,2.2,2.3,1.0,
2.2,3.8,1.5,2.2,2.0,1.6)
Ascorbic &lt;-c(51,65,54,55,52,59,45,41,66,42,51,54,53,41,45,50,45,49,50,51,49,
52,45,55,56,61,49,49,42,68,58,52,78,55,70,75,67,57,70,61,58,84,67,47,71,68,
56,72,58,72,62,63,63,68,56,54,66,72,60,72)
trt&lt;-paste(LineNumber,PlantingDate,sep="-")
output&lt;-reg.homog(trt,HeadWt,Ascorbic)
</code></pre>

<hr>
<h2 id='REGW.test'> Ryan, Einot and Gabriel and Welsch multiple range test </h2><span id='topic+REGW.test'></span>

<h3>Description</h3>

<p>Multiple range tests for all pairwise comparisons, to obtain a confident inequalities multiple range tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>REGW.test(y, trt, DFerror, MSerror, alpha = 0.05, group=TRUE, main = NULL,console=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="REGW.test_+3A_y">y</code></td>
<td>
<p> model(aov or lm) or answer of the experimental unit  </p>
</td></tr>
<tr><td><code id="REGW.test_+3A_trt">trt</code></td>
<td>
<p> Constant( only y=model) or vector treatment applied to each experimental unit  </p>
</td></tr>
<tr><td><code id="REGW.test_+3A_dferror">DFerror</code></td>
<td>
<p> Degree free </p>
</td></tr>
<tr><td><code id="REGW.test_+3A_mserror">MSerror</code></td>
<td>
<p> Mean Square Error </p>
</td></tr>
<tr><td><code id="REGW.test_+3A_alpha">alpha</code></td>
<td>
<p> Significant level </p>
</td></tr>
<tr><td><code id="REGW.test_+3A_group">group</code></td>
<td>
<p> TRUE or FALSE </p>
</td></tr>
<tr><td><code id="REGW.test_+3A_main">main</code></td>
<td>
<p> Title </p>
</td></tr>
<tr><td><code id="REGW.test_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is necessary first makes a analysis of variance.<br />
</p>
<p>if y = model, then to apply the instruction:<br /> 
REGW.test (model, &quot;trt&quot;, alpha = 0.05, group = TRUE, main = NULL, console = FALSE)<br />
where the model class is aov or lm.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>statistics</code></td>
<td>
<p>Statistics of the model</p>
</td></tr>
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>regw</code></td>
<td>
<p>Critical Range Table</p>
</td></tr>
<tr><td><code>means</code></td>
<td>
<p>Statistical summary of the study variable</p>
</td></tr>
<tr><td><code>comparison</code></td>
<td>
<p>Comparison between treatments</p>
</td></tr>
<tr><td><code>groups</code></td>
<td>
<p>Formation of treatment groups</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Multiple comparisons theory and methods. Departament of statistics 
the Ohio State University. USA, 1996. Jason C. Hsu. Chapman Hall/CRC </p>


<h3>See Also</h3>

<p><code><a href="#topic+BIB.test">BIB.test</a></code>, <code><a href="#topic+DAU.test">DAU.test</a></code>, <code><a href="#topic+duncan.test">duncan.test</a></code>,
<code><a href="#topic+durbin.test">durbin.test</a></code>, <code><a href="#topic+friedman">friedman</a></code>, <code><a href="#topic+HSD.test">HSD.test</a></code>,
<code><a href="#topic+kruskal">kruskal</a></code>, <code><a href="#topic+LSD.test">LSD.test</a></code>, <code><a href="#topic+Median.test">Median.test</a></code>, 
<code><a href="#topic+PBIB.test">PBIB.test</a></code>, <code><a href="#topic+scheffe.test">scheffe.test</a></code>, <code><a href="#topic+SNK.test">SNK.test</a></code>,
<code><a href="#topic+waerden.test">waerden.test</a></code>, <code><a href="#topic+waller.test">waller.test</a></code>, <code><a href="#topic+plot.group">plot.group</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(sweetpotato)
model&lt;-aov(yield~virus,data=sweetpotato)
out&lt;- REGW.test(model,"virus", 
main="Yield of sweetpotato. Dealt with different virus")
print(out)
REGW.test(model,"virus",alpha=0.05,console=TRUE,group=FALSE)
</code></pre>

<hr>
<h2 id='resampling.cv'> Resampling to find the optimal number of markers </h2><span id='topic+resampling.cv'></span>

<h3>Description</h3>

<p>This process finds the curve of CV for a different number of markers which
allows us to determine the number of optimal markers for a given relative 
variability.
A method of the curvature.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>resampling.cv(A, size, npoints)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="resampling.cv_+3A_a">A</code></td>
<td>
<p> data frame or matrix of binary data </p>
</td></tr>
<tr><td><code id="resampling.cv_+3A_size">size</code></td>
<td>
<p> number of re-samplings </p>
</td></tr>
<tr><td><code id="resampling.cv_+3A_npoints">npoints</code></td>
<td>
<p> Number of points to consider the model </p>
</td></tr>
</table>


<h3>Value</h3>

<p>lm(formula = CV ~ I(1/marker))<br />
Table with variation coefficient by number of markers
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p>Efron, B., Tibshirani, R. (1993) An Introduction to the Boostrap. Chapman and Hall/CRC </p>


<h3>See Also</h3>

<p><code><a href="#topic+cv.similarity">cv.similarity</a></code>, <code><a href="#topic+similarity">similarity</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
#example table of molecular markers
data(markers)
study&lt;-resampling.cv(markers,size=1,npoints=15)
#
# Results of the model
summary(study$model)
coef&lt;-coef(study$model)
py&lt;-predict(study$model)
Rsq&lt;-summary(study$model)$"r.squared"
table.cv &lt;- data.frame(study$table.cv,estimate=py)
print(table.cv)

# Plot CV
#startgraph
limy&lt;-max(table.cv[,2])+10
plot(table.cv[,c(1,2)],col="red",frame=FALSE,xlab="number of markers",
ylim=c(10,limy), ylab="CV",cex.main=0.8,main="Estimation of the number of markers")
ty&lt;-quantile(table.cv[,2],1)
tx&lt;-median(table.cv[,1])
tz&lt;-quantile(table.cv[,2],0.95)
text(tx,ty, cex=0.8,as.expression(substitute(CV == a + frac(b,markers),
list(a=round(coef[1],2),b=round(coef[2],2)))) )
text(tx,tz,cex=0.8,as.expression(substitute(R^2==r,list(r=round(Rsq,3)))))

# Plot CV = a + b/n.markers
fy&lt;-function(x,a,b) a+b/x
x&lt;-seq(2,max(table.cv[,1]),length=50)
y &lt;- coef[1] + coef[2]/x
lines(x,y,col="blue")
#grid(col="brown")
rug(table.cv[,1])
#endgraph
</code></pre>

<hr>
<h2 id='resampling.model'> Resampling for linear models </h2><span id='topic+resampling.model'></span>

<h3>Description</h3>

<p>This process consists of finding the values of P-value by means of a 
re-sampling (permutation) process along with the values obtained by variance analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>resampling.model(model,data,k,console=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="resampling.model_+3A_model">model</code></td>
<td>
<p> model in R </p>
</td></tr>
<tr><td><code id="resampling.model_+3A_data">data</code></td>
<td>
<p> data for the study of the model </p>
</td></tr>
<tr><td><code id="resampling.model_+3A_k">k</code></td>
<td>
<p> number of re-samplings </p>
</td></tr>
<tr><td><code id="resampling.model_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
</table>


<h3>Value</h3>

<p>Model solution with resampling.
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p>Efron, B., Tibshirani, R. (1993) An Introduction to the Boostrap. Chapman and Hall/CRC 
Phillip I. Good, (2001) Resampling Methods. Birkhauser. Boston . Basel . Berlin  
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simulation.model">simulation.model</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>#example 1 Simple linear regression
library(agricolae)
data(clay)
model&lt;-"ralstonia ~ days"
analysis&lt;-resampling.model(model,clay,k=2,console=TRUE)

#example 2 Analysis of variance: RCD
data(sweetpotato)
model&lt;-"yield~virus"
analysis&lt;-resampling.model(model,sweetpotato,k=2,console=TRUE)

#example 3 Simple linear regression
data(Glycoalkaloids)
model&lt;-"HPLC ~ spectrophotometer"
analysis&lt;-resampling.model(model,Glycoalkaloids,k=2,console=TRUE)

#example 4 Factorial in RCD

data(potato)
potato[,1]&lt;-as.factor(potato[,1])
potato[,2]&lt;-as.factor(potato[,2])
model&lt;-"cutting~variety + date + variety:date"
analysis&lt;-resampling.model(model,potato,k=2,console=TRUE)

</code></pre>

<hr>
<h2 id='rice'> Data of Grain yield of rice variety IR8</h2><span id='topic+rice'></span>

<h3>Description</h3>

<p>The data correspond to the yield of rice variety IR8 (g/m2) for land uniformity 
studies. The growing area is 18x36 meters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(rice)</code></pre>


<h3>Format</h3>

<p>A data frame with 36 observations on the following 18 variables.
</p>

<dl>
<dt><code>V1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V13</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V14</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V15</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V16</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V17</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V18</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>Table 12.1 Measuring Soil Heterogeneity
</p>


<h3>Source</h3>

<p>Statistical Procedures for Agriculture Research. Second Edition.
Kwanchai A. Gomez and Arturo A. Gomez. 1976. USA
Pag. 481.
</p>


<h3>References</h3>

<p>Statistical Procedures for Agriculture Research. Second Edition.
Kwanchai A. Gomez and Arturo A. Gomez. 1976. USA
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(rice)
str(rice)
</code></pre>

<hr>
<h2 id='RioChillon'> Data and analysis Mother and baby trials</h2><span id='topic+RioChillon'></span>

<h3>Description</h3>

<p>Mother/Baby Trials allow farmers and researchers to test best-bet
technologies or new cultivars. Evaluation of advanced Clones of potato
in the Valley of Rio Chillon - PERU (2004)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(RioChillon)</code></pre>


<h3>Format</h3>

<p>The format is list of 2:<br />
1. mother: data.frame:	30 obs. of  3 variables:<br />
-  block (3 levels)<br />
-  clon (10 levels)<br />
-  yield (kg.)<br />
2. babies: data.frame:	90 obs. of  3 variables:<br />
-  farmer (9 levels)<br />
-  clon (10 levels)<br />
-  yield (kg.)
</p>


<h3>Details</h3>

<p>1. Replicated researcher-managed &quot;mother trials&quot; with typically 10 treatments
evaluated in small plots.<br />
2. Unreplicated &quot;baby trials&quot; with 10 treatments evaluated in large plots.<br />
3. The &quot;baby trials&quot; with a subset of the treatments in the mother trial.
</p>


<h3>Source</h3>

<p>Experimental field.
</p>


<h3>References</h3>

<p>International Potato Center. CIP - Lima Peru.  
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Analisys the Mother/Baby Trial Design
library(agricolae)
data(RioChillon)
# First analysis the Mother Trial Design.
model&lt;-aov(yield ~ block + clon, RioChillon$mother)
anova(model)
cv.model(model)
comparison&lt;-with(RioChillon$mother,LSD.test(yield,clon, 18, 4.922, group=TRUE))
# Second analysis the babies Trial.
comparison&lt;-with(RioChillon$babies,friedman(farmer,clon, yield, group=TRUE))
# Third
# The researcher makes use of data from both mother and baby trials and thereby obtains
# information on suitability of new technologies or cultivars
# for different agro-ecologies and acceptability to farmers.
 
</code></pre>

<hr>
<h2 id='scheffe.test'> Multiple comparisons, scheffe </h2><span id='topic+scheffe.test'></span>

<h3>Description</h3>

<p>Scheffe 1959, method is very general in that all possible contrasts 
can be tested for significance and confidence intervals can be 
constructed for the corresponding linear. The test is conservative. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scheffe.test(y, trt, DFerror, MSerror, Fc, alpha = 0.05, group=TRUE, main = NULL,
console=FALSE )
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scheffe.test_+3A_y">y</code></td>
<td>
<p> model(aov or lm) or answer of the experimental unit  </p>
</td></tr>
<tr><td><code id="scheffe.test_+3A_trt">trt</code></td>
<td>
<p> Constant( only y=model) or vector treatment applied to each experimental unit  </p>
</td></tr>
<tr><td><code id="scheffe.test_+3A_dferror">DFerror</code></td>
<td>
<p> Degrees of freedom </p>
</td></tr>
<tr><td><code id="scheffe.test_+3A_mserror">MSerror</code></td>
<td>
<p> Mean Square Error </p>
</td></tr>
<tr><td><code id="scheffe.test_+3A_fc">Fc</code></td>
<td>
<p> F Value </p>
</td></tr>
<tr><td><code id="scheffe.test_+3A_alpha">alpha</code></td>
<td>
<p> Significant level </p>
</td></tr>
<tr><td><code id="scheffe.test_+3A_group">group</code></td>
<td>
<p> TRUE or FALSE </p>
</td></tr>
<tr><td><code id="scheffe.test_+3A_main">main</code></td>
<td>
<p> Title </p>
</td></tr>
<tr><td><code id="scheffe.test_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is necessary first makes a analysis of variance.<br />
</p>
<p>if y = model, then to apply the instruction:<br /> 
scheffe.test (model, &quot;trt&quot;, alpha = 0.05, group = TRUE, main = NULL, console = FALSE)<br />
where the model class is aov or lm.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>statistics</code></td>
<td>
<p>Statistics of the model</p>
</td></tr>
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>means</code></td>
<td>
<p>Statistical summary of the study variable</p>
</td></tr>
<tr><td><code>comparison</code></td>
<td>
<p>Comparison between treatments</p>
</td></tr>
<tr><td><code>groups</code></td>
<td>
<p>Formation of treatment groups</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu</p>


<h3>References</h3>

 
<p>Robert O. Kuehl. 2nd ed. Design of experiments. Duxbury, copyright 2000.<br />
Steel, R.; Torri,J; Dickey, D.(1997) Principles and Procedures of Statistics
A Biometrical Approach. pp189
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BIB.test">BIB.test</a></code>, <code><a href="#topic+DAU.test">DAU.test</a></code>, <code><a href="#topic+duncan.test">duncan.test</a></code>,
<code><a href="#topic+durbin.test">durbin.test</a></code>, <code><a href="#topic+friedman">friedman</a></code>, <code><a href="#topic+HSD.test">HSD.test</a></code>,
<code><a href="#topic+kruskal">kruskal</a></code>, <code><a href="#topic+LSD.test">LSD.test</a></code>, <code><a href="#topic+Median.test">Median.test</a></code>, 
<code><a href="#topic+PBIB.test">PBIB.test</a></code>, <code><a href="#topic+REGW.test">REGW.test</a></code>, <code><a href="#topic+SNK.test">SNK.test</a></code>,
<code><a href="#topic+waerden.test">waerden.test</a></code>, <code><a href="#topic+waller.test">waller.test</a></code>, <code><a href="#topic+plot.group">plot.group</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(sweetpotato)
model&lt;-aov(yield~virus, data=sweetpotato)
comparison &lt;- scheffe.test(model,"virus", group=TRUE,console=TRUE,
main="Yield of sweetpotato\nDealt with different virus")
# Old version scheffe.test()
df&lt;-df.residual(model)
MSerror&lt;-deviance(model)/df
Fc&lt;-anova(model)["virus",4]
out &lt;- with(sweetpotato,scheffe.test(yield, virus, df, MSerror, Fc))
print(out)
</code></pre>

<hr>
<h2 id='similarity'> Matrix of similarity in binary data </h2><span id='topic+similarity'></span>

<h3>Description</h3>

<p>It finds the similarity matrix of binary tables (1 and 0).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>similarity(A)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="similarity_+3A_a">A</code></td>
<td>
<p> Matrix, data binary </p>
</td></tr>
</table>


<h3>Value</h3>

<p>Distance matrix. Class = dist.
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+cv.similarity">cv.similarity</a></code>, <code><a href="#topic+resampling.cv">resampling.cv</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>#example table of molecular markers
library(agricolae)
data(markers)
distance&lt;-similarity(markers)
#startgraph
tree&lt;-hclust(distance,method="mcquitty")
plot(tree,col="blue")
#endgraph
</code></pre>

<hr>
<h2 id='simulation.model'> Simulation of the linear model under normality </h2><span id='topic+simulation.model'></span>

<h3>Description</h3>

<p>This process consists of validating the variance analysis results 
using a simulation process of the experiment.
The validation consists of comparing the calculated values of each source
of variation of the simulated data with respect to the calculated values
of the original data. If in more than 50 percent of the cases they are higher
than the real one, then it is considered favorable and the probability
reported by the ANOVA is accepted, since the P-Value is the probability of
(F &gt; F.value).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulation.model(model,file, categorical = NULL,k,console=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simulation.model_+3A_model">model</code></td>
<td>
<p> Model in R</p>
</td></tr>
<tr><td><code id="simulation.model_+3A_file">file</code></td>
<td>
<p> Data for the study of the model</p>
</td></tr>
<tr><td><code id="simulation.model_+3A_categorical">categorical</code></td>
<td>
<p> position of the columns of the data that correspond
to categorical variables </p>
</td></tr>
<tr><td><code id="simulation.model_+3A_k">k</code></td>
<td>
<p> Number of simulations </p>
</td></tr>
<tr><td><code id="simulation.model_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>model</code></td>
<td>
<p> ouput linear model, lm</p>
</td></tr>
<tr><td><code>simulation</code></td>
<td>
<p> anova simulation</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+resampling.model">resampling.model</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
#example 1
data(clay)
model&lt;-"ralstonia ~ days"
simulation.model(model,clay,k=15,console=TRUE)
#example 2
data(sweetpotato)
model&lt;-"yield~virus"
simulation.model(model,sweetpotato,categorical=1,k=15,console=TRUE)
#example 3
data(Glycoalkaloids)
model&lt;-"HPLC ~ spectrophotometer"
simulation.model(model,Glycoalkaloids,k=15,console=TRUE)
#example 4
data(potato)
model&lt;-"cutting~date+variety"
simulation.model(model,potato,categorical=c(1,2,3),k=15,console=TRUE)

</code></pre>

<hr>
<h2 id='sinRepAmmi'> Data for AMMI without repetition</h2><span id='topic+sinRepAmmi'></span>

<h3>Description</h3>

<p>Data frame for AMMI analysis with 50 genotypes in 5 environments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sinRepAmmi)</code></pre>


<h3>Format</h3>

<p>A data frame with 250 observations on the following 3 variables.
</p>

<dl>
<dt><code>ENV</code></dt><dd><p>a factor with levels <code>A1</code> <code>A2</code> <code>A3</code> <code>A4</code> <code>A5</code></p>
</dd>
<dt><code>GEN</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>YLD</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Experimental data.
</p>


<h3>References</h3>

<p>International Potato Center - Lima Peru.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(sinRepAmmi)
str(sinRepAmmi)
</code></pre>

<hr>
<h2 id='skewness'> Finding the skewness coefficient </h2><span id='topic+skewness'></span>

<h3>Description</h3>

<p>It returns the skewness of a distribution. It is similar to SAS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>skewness(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="skewness_+3A_x">x</code></td>
<td>
<p>a numeric vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The skewness of x.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+kurtosis">kurtosis</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
x&lt;-c(3,4,5,2,3,4,NA,5,6,4,7)
skewness(x)
# value is 0,3595431, is slightly asimetrica (positive) to the right
</code></pre>

<hr>
<h2 id='SNK.test'> Student-Newman-Keuls (SNK) </h2><span id='topic+SNK.test'></span>

<h3>Description</h3>

<p>SNK is derived from Tukey, but it is less conservative (finds more differences).
Tukey controls the error for all comparisons, where SNK only controls for 
comparisons under consideration. The level by alpha default is 0.05.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SNK.test(y, trt, DFerror, MSerror, alpha = 0.05, group=TRUE, main = NULL,console=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SNK.test_+3A_y">y</code></td>
<td>
<p> model(aov or lm) or answer of the experimental unit  </p>
</td></tr>
<tr><td><code id="SNK.test_+3A_trt">trt</code></td>
<td>
<p> Constant( only y=model) or vector treatment applied to each experimental unit  </p>
</td></tr>
<tr><td><code id="SNK.test_+3A_dferror">DFerror</code></td>
<td>
<p> Degree free </p>
</td></tr>
<tr><td><code id="SNK.test_+3A_mserror">MSerror</code></td>
<td>
<p> Mean Square Error </p>
</td></tr>
<tr><td><code id="SNK.test_+3A_alpha">alpha</code></td>
<td>
<p> Significant level </p>
</td></tr>
<tr><td><code id="SNK.test_+3A_group">group</code></td>
<td>
<p> TRUE or FALSE </p>
</td></tr>
<tr><td><code id="SNK.test_+3A_main">main</code></td>
<td>
<p> Title </p>
</td></tr>
<tr><td><code id="SNK.test_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is necessary first makes a analysis of variance.<br />
</p>
<p>if y = model, then to apply the instruction:<br /> 
SNK.test (model, &quot;trt&quot;, alpha = 0.05, group = TRUE, main = NULL, console = FALSE)<br />
where the model class is aov or lm.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>statistics</code></td>
<td>
<p>Statistics of the model</p>
</td></tr>
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>snk</code></td>
<td>
<p>Critical Range Table</p>
</td></tr>
<tr><td><code>means</code></td>
<td>
<p>Statistical summary of the study variable</p>
</td></tr>
<tr><td><code>comparison</code></td>
<td>
<p>Comparison between treatments</p>
</td></tr>
<tr><td><code>groups</code></td>
<td>
<p>Formation of treatment groups</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

  
<p>1. Principles and procedures of statistics a biometrical approach
Steel &amp; Torry &amp; Dickey. Third Edition 1997<br />
2. Multiple comparisons theory and methods. Departament of statistics 
the Ohio State University. USA, 1996. Jason C. Hsu. Chapman Hall/CRC.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BIB.test">BIB.test</a></code>, <code><a href="#topic+DAU.test">DAU.test</a></code>, <code><a href="#topic+duncan.test">duncan.test</a></code>,
<code><a href="#topic+durbin.test">durbin.test</a></code>, <code><a href="#topic+friedman">friedman</a></code>, <code><a href="#topic+HSD.test">HSD.test</a></code>,
<code><a href="#topic+kruskal">kruskal</a></code>, <code><a href="#topic+LSD.test">LSD.test</a></code>, <code><a href="#topic+Median.test">Median.test</a></code>, 
<code><a href="#topic+PBIB.test">PBIB.test</a></code>, <code><a href="#topic+REGW.test">REGW.test</a></code>, <code><a href="#topic+scheffe.test">scheffe.test</a></code>,
<code><a href="#topic+waerden.test">waerden.test</a></code>, <code><a href="#topic+waller.test">waller.test</a></code>, <code><a href="#topic+plot.group">plot.group</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(sweetpotato)
model&lt;-aov(yield~virus,data=sweetpotato)
out &lt;- SNK.test(model,"virus", console=TRUE, 
main="Yield of sweetpotato. Dealt with different virus")
print(SNK.test(model,"virus", group=FALSE))
# version old SNK.test()
df&lt;-df.residual(model)
MSerror&lt;-deviance(model)/df
out &lt;- with(sweetpotato,SNK.test(yield,virus,df,MSerror, group=TRUE))
print(out$groups)
</code></pre>

<hr>
<h2 id='soil'> Data of soil analysis for 13 localities</h2><span id='topic+soil'></span>

<h3>Description</h3>

<p>We analyzed the physical and chemical properties of different soils, as full 
characterization of soil and special analysis of micro-elements. These analyses 
were conducted in the laboratory analysis of soils, plants, water and fertilizers 
in the La Molina National Agrarian University (UNALM). To which the different soil 
samples were dried to the environment, screened (mesh 0.5xo, 5 mm) and sterilized by 
steam 4 to 5 hours with a Lindinger Steam aerator SA150 and SA700, with the possible 
aim of eliminating bacteria saprophytic or antagonists to prevent the growth of bacteria 
(R.solanacearum).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(soil)</code></pre>


<h3>Format</h3>

<p>A data frame with 13 observations on the following 23 variables.
</p>

<dl>
<dt><code>place</code></dt><dd><p>a factor with levels <code>Chmar</code> <code>Chz</code> <code>Cnt1</code> <code>Cnt2</code> <code>Cnt3</code> <code>Hco1</code> <code>Hco2</code> <code>Hco3</code> <code>Hyo1</code> <code>Hyo2</code> <code>Namora</code> <code>SR1</code> <code>SR2</code></p>
</dd>
<dt><code>pH</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EC</code></dt><dd><p>a numeric vector, electrical conductivity</p>
</dd>
<dt><code>CaCO3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>MO</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CIC</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>P</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>K</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>sand</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>slime</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>clay</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ca</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Mg</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>K2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Na</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Al_H</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>K_Mg</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ca_Mg</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cu</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Fe</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Mn</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Zn</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>Cnt1= Canete, Cnt2=Valle Dulce(Canete), Cnt3=Valle Grande(Canete), Chz=Obraje-Carhuaz(Ancash),
Chmar=Chucmar-Chota(Huanuco, Hco1= Mayobamba-Chinchao(Huanuco), Hco2=Nueva Independencia-Chinchao(Huanuco), 
Hco3=San Marcos-Umari(Huanuco), Hyo1=La Victoria-Huancayo(Junin), Hyo1=El Tambo-Huancayo(Junin), 
Namora=Namora(Cajamarca), SR1= El Milagro-San Ramon(Junin), Sr2=La Chinchana-San Ramon(Junin).
</p>


<h3>Source</h3>

<p>Experimental field, 2004. Data Kindly provided by Dr. Sylvie Priou, Liliam Gutarra and Pedro Aley.
</p>


<h3>References</h3>

<p>International Potato Center - Lima, PERU.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(soil)
str(soil)
</code></pre>

<hr>
<h2 id='sp.plot'> Splip-Plot analysis </h2><span id='topic+sp.plot'></span>

<h3>Description</h3>

<p>The variance analysis of a split plot design is divided into two parts: 
the plot-factor analysis and the sub-plot factor analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sp.plot(block, pplot, splot, Y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sp.plot_+3A_block">block</code></td>
<td>
<p> replications </p>
</td></tr>
<tr><td><code id="sp.plot_+3A_pplot">pplot</code></td>
<td>
<p> main-plot Factor</p>
</td></tr>
<tr><td><code id="sp.plot_+3A_splot">splot</code></td>
<td>
<p> sub-plot Factor </p>
</td></tr>
<tr><td><code id="sp.plot_+3A_y">Y</code></td>
<td>
<p> Variable, response </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The split-plot design is specifically suited for a two-factor experiment on of the factors is assigned to main plot (main-plot factor), the second factor, called the subplot factor, is assigned into subplots.  The model is mixed, the blocks are random and the study factors are fixed applied according to the design.
</p>


<h3>Value</h3>

<p>ANOVA: Splip plot analysis
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu  </p>


<h3>References</h3>

<p> Statistical procedures for agricultural research.
Kwanchai A. Gomez, Arturo A. Gomez. Second Edition. 1984.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ssp.plot">ssp.plot</a></code>, <code><a href="#topic+strip.plot">strip.plot</a></code>, <code><a href="#topic+design.split">design.split</a></code>, 
<code><a href="#topic+design.strip">design.strip</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(plots)
model&lt;-with(plots,sp.plot(block,A,B,yield))
# with aov
plots[,1]&lt;-as.factor(plots[,1])
AOV &lt;- aov(yield ~ block + A*B + Error(block/A),data=plots)
summary(AOV)
</code></pre>

<hr>
<h2 id='ssp.plot'> Split-split-Plot analysis </h2><span id='topic+ssp.plot'></span>

<h3>Description</h3>

<p>The variance analysis of a split-split plot design is divided into three parts: 
the main-plot, subplot and sub-subplot analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ssp.plot(block, pplot, splot, ssplot, Y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ssp.plot_+3A_block">block</code></td>
<td>
<p> replications </p>
</td></tr>
<tr><td><code id="ssp.plot_+3A_pplot">pplot</code></td>
<td>
<p> Factor main plot </p>
</td></tr>
<tr><td><code id="ssp.plot_+3A_splot">splot</code></td>
<td>
<p> Factor subplot </p>
</td></tr>
<tr><td><code id="ssp.plot_+3A_ssplot">ssplot</code></td>
<td>
<p> Factor sub-subplot</p>
</td></tr>
<tr><td><code id="ssp.plot_+3A_y">Y</code></td>
<td>
<p> Variable, response </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The split-split-plot design is an extension of the split-plot design to accommodate a third factor: one factor in main-plot, other in subplot and the third factor in sub-subplot. The model is mixed, the blocks are random and the study factors are fixed applied according to the design. 
</p>


<h3>Value</h3>

<p>ANOVA: Splip Split plot analysis
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu  </p>


<h3>References</h3>

<p> Statistical procedures for agricultural research.
Kwanchai A. Gomez, Arturo A. Gomez. Second Edition. 1984.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sp.plot">sp.plot</a></code>, <code><a href="#topic+strip.plot">strip.plot</a></code>, <code><a href="#topic+design.split">design.split</a></code>, 
<code><a href="#topic+design.strip">design.strip</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'># Statistical procedures for agricultural research, pag 143
# Grain Yields of Three Rice Varieties Grown under 
#Three Management practices and Five Nitrogen levels; in a
#split-split-plot design with nitrogen as main-plot, 
#management practice as subplot, and variety as sub-subplot 
#factores, with three replications.
library(agricolae)
f &lt;- system.file("external/ssp.csv", package="agricolae")
ssp&lt;-read.csv(f)
model&lt;-with(ssp,ssp.plot(block,nitrogen,management,variety,yield))
gla&lt;-model$gl.a; glb&lt;-model$gl.b; glc&lt;-model$gl.c
Ea&lt;-model$Ea; Eb&lt;-model$Eb; Ec&lt;-model$Ec
op&lt;-par(mfrow=c(1,3),cex=0.6)
out1&lt;-with(ssp,LSD.test(yield,nitrogen,gla,Ea,console=TRUE))
out2&lt;-with(ssp,LSD.test(yield,management,glb,Eb,console=TRUE))
out3&lt;-with(ssp,LSD.test(yield,variety,glc,Ec,console=TRUE))
plot(out1,xlab="Nitrogen",las=1,variation="IQR")
plot(out2,xlab="Management",variation="IQR")
plot(out3,xlab="Variety",variation="IQR")
# with aov
ssp$block&lt;-factor(ssp$block)
ssp$nitrogen&lt;-factor(ssp$nitrogen)
ssp$management&lt;-factor(ssp$management)
ssp$variety&lt;-factor(ssp$variety)
AOV&lt;-aov(yield ~  block + nitrogen*management*variety + Error(block/nitrogen/management),data=ssp)
summary(AOV)
par(op)
</code></pre>

<hr>
<h2 id='stability.nonpar'> Nonparametric stability analysis </h2><span id='topic+stability.nonpar'></span>

<h3>Description</h3>

<p>A method based on the statistical ranges of the study variable per 
environment for the stability analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stability.nonpar(data, variable = NULL, ranking = FALSE, console=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stability.nonpar_+3A_data">data</code></td>
<td>
<p> First column the genotypes following environment  </p>
</td></tr>
<tr><td><code id="stability.nonpar_+3A_variable">variable</code></td>
<td>
<p> Name of variable </p>
</td></tr>
<tr><td><code id="stability.nonpar_+3A_ranking">ranking</code></td>
<td>
<p>logical, print ranking </p>
</td></tr>
<tr><td><code id="stability.nonpar_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>ranking</code></td>
<td>
<p> data frame </p>
</td></tr>
<tr><td><code>statistics</code></td>
<td>
<p> Statistical analysis chi square test </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu</p>


<h3>References</h3>

<p>Haynes K G, Lambert D H, Christ B J, Weingartner D P, Douches D S,
Backlund J E, Fry W and Stevenson W. 1998. Phenotypic stability of resistance
to late blight in potato clones evaluated at eight sites in the United States
American Journal Potato Research 75, pag 211-217.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+stability.par">stability.par</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(haynes)
stability.nonpar(haynes,"AUDPC",ranking=TRUE,console=TRUE)
# Example 2
data(CIC)
data1&lt;-CIC$comas[,c(1,6,7,17,18)]
data2&lt;-CIC$oxapampa[,c(1,6,7,19,20)]
cic &lt;- rbind(data1,data2)

means &lt;- by(cic[,5], cic[,c(2,1)], function(x) mean(x,na.rm=TRUE))
means &lt;-as.data.frame(means[,])
cic.mean&lt;-data.frame(genotype=row.names(means),means)
cic.mean&lt;-delete.na(cic.mean,"greater")
out&lt;-stability.nonpar(cic.mean)
out$ranking
out$statistics
</code></pre>

<hr>
<h2 id='stability.par'> Stability analysis. SHUKLA'S STABILITY VARIANCE AND KANG'S </h2><span id='topic+stability.par'></span>

<h3>Description</h3>

<p>This procedure calculates the stability variations as well as the statistics 
of selection for the yield and the stability.
The averages of the genotype through the different environment repetitions 
are required for the calculations. The mean square error must be calculated
from the joint variance analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stability.par(data,rep,MSerror,alpha=0.1,main=NULL,cova = FALSE,name.cov=NULL,
file.cov=0,console=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stability.par_+3A_data">data</code></td>
<td>
<p> matrix of averages, by rows the genotypes and columns the environment </p>
</td></tr>
<tr><td><code id="stability.par_+3A_rep">rep</code></td>
<td>
<p> Number of repetitions </p>
</td></tr>
<tr><td><code id="stability.par_+3A_mserror">MSerror</code></td>
<td>
<p> Mean Square Error </p>
</td></tr>
<tr><td><code id="stability.par_+3A_alpha">alpha</code></td>
<td>
<p> Label significant </p>
</td></tr>
<tr><td><code id="stability.par_+3A_main">main</code></td>
<td>
<p> Title </p>
</td></tr>
<tr><td><code id="stability.par_+3A_cova">cova</code></td>
<td>
<p> Covariable </p>
</td></tr>
<tr><td><code id="stability.par_+3A_name.cov">name.cov</code></td>
<td>
<p> Name covariable </p>
</td></tr>
<tr><td><code id="stability.par_+3A_file.cov">file.cov</code></td>
<td>
<p> Data covariable </p>
</td></tr>
<tr><td><code id="stability.par_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Stable 
(i) determines the contribution of each genotype to GE interaction by calculating var(i);
(ii) assigns ranks to genotypes from highest to lowest yield receiving the rank of 1;
(iii) calculates protected LSD for mean yield comparisons;
(iv) adjusts yield rank according to LSD (the adjusted rank labeled Y);
(v) determines significance of var(i) usign an aproximate F-test;
(vi) assigns stability rating (S) as follows: -8, -4 and -2 for var(i) significant at the
0.01, 0.05 and 0.10 probability levels, and 0 for nonsignificant var(i) 
( the higher the var(i), the less stable the genotype);
(vii) sums adjusted yield rank, Y, and stability rating, S, for each genotype to determine YS(i) statistic; and 
(viii) calculates mean YS(i) and identifies genotypes (selection) with YS(i) &gt; mean YS(i).
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>analysis</code></td>
<td>
<p> Analysis of variance </p>
</td></tr>
<tr><td><code>statistics</code></td>
<td>
<p> Statistics of the model </p>
</td></tr>
<tr><td><code>stability</code></td>
<td>
<p> summary stability analysis </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Kang, M. S. 1993. Simultaneous selection for yield and
stability: Consequences for growers. Agron. J. 85:754-757. 
Manjit S. Kang and Robert Mangari. 1995. Stable: A basic program for 
calculating stability and yield-stability statistics. Agron. J. 87:276-277</p>


<h3>See Also</h3>

<p><code><a href="#topic+stability.nonpar">stability.nonpar</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
# example 1
# Experimental data,
# replication rep= 4
# Mean square error, MSerror = 1.8
# 12 environment
# 17 genotype  = 1,2,3,.., 17
# yield averages of 13 genotypes in localities
f &lt;- system.file("external/dataStb.csv", package="agricolae")
dataStb&lt;-read.csv(f)
stability.par(dataStb, rep=4, MSerror=1.8, alpha=0.1, main="Genotype",console=TRUE)

#example 2 covariable. precipitation
precipitation&lt;- c(1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000,2100)
stability.par(dataStb, rep=4, MSerror=1.8, alpha=0.1, main="Genotype",
 cova=TRUE, name.cov="Precipitation", file.cov=precipitation,console=TRUE)
</code></pre>

<hr>
<h2 id='stat.freq'> Descriptive measures of grouped data </h2><span id='topic+stat.freq'></span>

<h3>Description</h3>

<p>By this process the variance and central measures ar found: average, medium 
and mode of grouped data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stat.freq(histogram)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stat.freq_+3A_histogram">histogram</code></td>
<td>
<p> Object create by function hist() </p>
</td></tr>
</table>


<h3>Value</h3>

<p>Statistics of grouped data.
</p>


<h3>Author(s)</h3>

<p> Felipe de mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+polygon.freq">polygon.freq</a></code>, <code><a href="#topic+table.freq">table.freq</a></code>, <code><a href="#topic+graph.freq">graph.freq</a></code>,
<code><a href="#topic+inter.freq">inter.freq</a></code>, <code><a href="#topic+sturges.freq">sturges.freq</a></code>, <code><a href="#topic+join.freq">join.freq</a></code>,
<code><a href="#topic+ogive.freq">ogive.freq</a></code>, <code><a href="#topic+normal.freq">normal.freq</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(growth)
grouped&lt;-with(growth,hist(height,plot=FALSE))
measures&lt;-stat.freq(grouped)
print(measures)

</code></pre>

<hr>
<h2 id='strip.plot'> Strip-Plot analysis </h2><span id='topic+strip.plot'></span>

<h3>Description</h3>

<p>The variance analysis of a strip-plot design is divided into three parts: 
the horizontal-factor analysis, the vertical-factor analysis, and the 
interaction analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>strip.plot(BLOCK, COL, ROW, Y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="strip.plot_+3A_block">BLOCK</code></td>
<td>
<p> replications </p>
</td></tr>
<tr><td><code id="strip.plot_+3A_col">COL</code></td>
<td>
<p> Factor column </p>
</td></tr>
<tr><td><code id="strip.plot_+3A_row">ROW</code></td>
<td>
<p> Factor row </p>
</td></tr>
<tr><td><code id="strip.plot_+3A_y">Y</code></td>
<td>
<p> Variable, response </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The strip-plot design is specifically suited for a two-factor experiment
in which the desired precision for measuring the interaction effects
between the two factors is higher than that for measuring the main efect
two factors
</p>


<h3>Value</h3>

<p>Data and analysis of the variance of the strip plot design. 
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu  </p>


<h3>References</h3>

<p> Statistical procedures for agricultural research.
Kwanchai A. Gomez, Arturo A. Gomez. Second Edition. 1984.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ssp.plot">ssp.plot</a></code>, <code><a href="#topic+sp.plot">sp.plot</a></code>, <code><a href="#topic+design.split">design.split</a></code>, 
<code><a href="#topic+design.strip">design.strip</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'># Yield
library(agricolae)
data(huasahuasi)
YIELD&lt;-huasahuasi$YIELD
market &lt;- YIELD$y1da + YIELD$y2da
non_market &lt;- YIELD$y3da
yield &lt;- market + non_market
model&lt;-with(YIELD,strip.plot(block, clon, trt, yield))
out1&lt;-with(YIELD,LSD.test(yield,clon,model$gl.a,model$Ea))
oldpar&lt;-par(mar=c(3,8,1,1),cex=0.8)
plot(out1,xlim=c(0,80),horiz=TRUE,las=1)
out2&lt;-with(YIELD,LSD.test(yield,trt,model$gl.b,model$Eb))
plot(out2,xlim=c(0,80),horiz=TRUE,las=1)
par(oldpar)
</code></pre>

<hr>
<h2 id='sturges.freq'> Class intervals for a histogram, the rule of Sturges </h2><span id='topic+sturges.freq'></span>

<h3>Description</h3>

<p>if k=0 then classes: k = 1 + log(n,2).
if k &gt; 0, fixed nclass.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sturges.freq(x,k=0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sturges.freq_+3A_x">x</code></td>
<td>
<p> vector </p>
</td></tr>
<tr><td><code id="sturges.freq_+3A_k">k</code></td>
<td>
<p> constant </p>
</td></tr>
</table>


<h3>Value</h3>

<p>Statistics of sturges for a histogram.
</p>


<h3>Author(s)</h3>

<p> Felipe de mendiburu </p>


<h3>References</h3>

<p>Reza A. Hoshmand. 1988. Statistical Methods for Agricultural Sciences,
Timber Press, Incorporated, pag 18-21.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+polygon.freq">polygon.freq</a></code>, <code><a href="#topic+table.freq">table.freq</a></code>, <code><a href="#topic+stat.freq">stat.freq</a></code>,
<code><a href="#topic+inter.freq">inter.freq</a></code>, <code><a href="#topic+graph.freq">graph.freq</a></code>, <code><a href="#topic+join.freq">join.freq</a></code>,
<code><a href="#topic+ogive.freq">ogive.freq</a></code>, <code><a href="#topic+normal.freq">normal.freq</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(natives)
classes&lt;-with(natives,sturges.freq(size))
# information of the classes
breaks &lt;- classes$breaks
breaks
#startgraph
# Histogram with the established classes
h&lt;-with(natives,graph.freq(size,breaks,frequency=1, col="yellow",axes=FALSE,
    xlim=c(0,0.12),main="",xlab="",ylab=""))
axis(1,breaks,las=2)
axis(2,seq(0,400,50),las=2)
title(main="Histogram of frequency\nSize of the tubercule of the Oca",
xlab="Size of the oca", ylab="Frequency")
#endgraph
</code></pre>

<hr>
<h2 id='summary.graph.freq'> frequency Table of a Histogram  </h2><span id='topic+summary.graph.freq'></span>

<h3>Description</h3>

<p>It finds the absolute, relative and accumulated frequencies with the class 
intervals defined from a previously calculated histogram by the &quot;hist&quot; of R
function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'graph.freq'
summary(object,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.graph.freq_+3A_object">object</code></td>
<td>
<p> Object by function graph.freq() </p>
</td></tr>
<tr><td><code id="summary.graph.freq_+3A_...">...</code></td>
<td>
<p> other parameters of graphic</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Frequency table.<br />
</p>
<table role = "presentation">
<tr><td><code>Lower</code></td>
<td>
<p>Lower limit class</p>
</td></tr>
<tr><td><code>Upper</code></td>
<td>
<p>Upper limit class</p>
</td></tr>
<tr><td><code>Main</code></td>
<td>
<p>class point</p>
</td></tr>
<tr><td><code>Frequency</code></td>
<td>
<p>Frequency</p>
</td></tr>
<tr><td><code>Percentage</code></td>
<td>
<p>Percentage frequency</p>
</td></tr>
<tr><td><code>CF</code></td>
<td>
<p>Cumulative frequency</p>
</td></tr>
<tr><td><code>CPF</code></td>
<td>
<p>Cumulative Percentage frequency</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+polygon.freq">polygon.freq</a></code>, <code><a href="#topic+stat.freq">stat.freq</a></code>, <code><a href="#topic+graph.freq">graph.freq</a></code>,
<code><a href="#topic+inter.freq">inter.freq</a></code>, <code><a href="#topic+sturges.freq">sturges.freq</a></code>, <code><a href="#topic+join.freq">join.freq</a></code>,
<code><a href="#topic+ogive.freq">ogive.freq</a></code>, <code><a href="#topic+normal.freq">normal.freq</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(growth)
h2&lt;-with(growth,graph.freq(height,plot=FALSE))
print(summary(h2),row.names=FALSE)
</code></pre>

<hr>
<h2 id='sweetpotato'> Data of sweetpotato yield </h2><span id='topic+sweetpotato'></span>

<h3>Description</h3>

<p>The data correspond to an experiment with costanero sweetpotato made at the locality 
of the Tacna department, southern Peru. The effect of two viruses (Spfmv and Spcsv) 
was studied. The treatments were the following: CC (Spcsv) = Sweetpotato chlorotic 
dwarf, FF (Spfmv) = Feathery mottle, FC (Spfmv y Spcsv) = Viral complex and OO 
(witness) healthy plants.
In each plot, 50 sweetpotato plants were sown and 12 plots were employed. 
Each treatment was made with 3 repetitions and at the end of the experiment the total 
weight in kilograms was evaluated. The virus transmission was made in the cuttings 
and these were sown in the field.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sweetpotato)</code></pre>


<h3>Format</h3>

<p>A data frame with 12 observations on the following 2 variables.
</p>

<dl>
<dt><code>virus</code></dt><dd><p>a factor with levels <code>cc</code> <code>fc</code> <code>ff</code> <code>oo</code></p>
</dd>
<dt><code>yield</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Experimental field.
</p>


<h3>References</h3>

<p>International Potato Center. CIP - Lima Peru
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(sweetpotato)
str(sweetpotato)
</code></pre>

<hr>
<h2 id='table.freq'> frequency Table of a Histogram  </h2><span id='topic+table.freq'></span>

<h3>Description</h3>

<p>It finds the absolute, relative and accumulated frequencies with the class 
intervals defined from a previously calculated histogram by the &quot;hist&quot; of R
function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>table.freq(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="table.freq_+3A_object">object</code></td>
<td>
<p> Object by function graph.freq() </p>
</td></tr>
</table>


<h3>Value</h3>

<p>Frequency table.<br />
</p>
<table role = "presentation">
<tr><td><code>Lower</code></td>
<td>
<p>Lower limit class</p>
</td></tr>
<tr><td><code>Upper</code></td>
<td>
<p>Upper limit class</p>
</td></tr>
<tr><td><code>Main</code></td>
<td>
<p>class point</p>
</td></tr>
<tr><td><code>Frequency</code></td>
<td>
<p>Frequency</p>
</td></tr>
<tr><td><code>Percentage</code></td>
<td>
<p>Percentage frequency</p>
</td></tr>
<tr><td><code>CF</code></td>
<td>
<p>Cumulative frequency</p>
</td></tr>
<tr><td><code>CPF</code></td>
<td>
<p>Cumulative Percentage frequency</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+polygon.freq">polygon.freq</a></code>, <code><a href="#topic+stat.freq">stat.freq</a></code>, <code><a href="#topic+graph.freq">graph.freq</a></code>,
<code><a href="#topic+inter.freq">inter.freq</a></code>, <code><a href="#topic+sturges.freq">sturges.freq</a></code>, <code><a href="#topic+join.freq">join.freq</a></code>,
<code><a href="#topic+ogive.freq">ogive.freq</a></code>, <code><a href="#topic+normal.freq">normal.freq</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(growth)
h2&lt;-with(growth,graph.freq(height,plot=FALSE))
print(table.freq(h2),row.names=FALSE)
</code></pre>

<hr>
<h2 id='tapply.stat'> Statistics of data grouped by factors </h2><span id='topic+tapply.stat'></span>

<h3>Description</h3>

<p>This process lies in finding statistics which consist of more than one variable,
grouped or crossed by factors. The table must be organized by columns between 
variables and factors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tapply.stat(y, x, stat = "mean")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tapply.stat_+3A_y">y</code></td>
<td>
<p>data.frame variables </p>
</td></tr>
<tr><td><code id="tapply.stat_+3A_x">x</code></td>
<td>
<p>data.frame factors  </p>
</td></tr>  
<tr><td><code id="tapply.stat_+3A_stat">stat</code></td>
<td>
<p> Method </p>
</td></tr>
</table>


<h3>Value</h3>

<p>Statistics of quantitative variables by categorical variables.
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
# case of 1 single factor
data(sweetpotato)
tapply.stat(sweetpotato[,2],sweetpotato[,1],mean)
with(sweetpotato,tapply.stat(yield,virus,sd))
with(sweetpotato,tapply.stat(yield,virus,function(x) max(x)-min(x)))
with(sweetpotato,tapply.stat(yield,virus,
function(x) quantile(x,0.75,6)-quantile(x,0.25,6)))
# other case
data(cotton)
with(cotton,tapply.stat(yield,cotton[,c(1,3,4)],mean))
with(cotton,tapply.stat(yield,cotton[,c(1,4)],max))
# Height of pijuayo
data(growth)
with(growth,tapply.stat(height, growth[,2:1], function(x) mean(x,na.rm=TRUE)))
</code></pre>

<hr>
<h2 id='vark'> Variance K, ties, Kendall</h2><span id='topic+vark'></span>

<h3>Description</h3>

<p>The Kendall method in order to find the K variance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vark(x, y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vark_+3A_x">x</code></td>
<td>
<p> Vector </p>
</td></tr>
<tr><td><code id="vark_+3A_y">y</code></td>
<td>
<p> vector </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Script in C to R.
</p>


<h3>Value</h3>

<p>variance of K for Kendall's tau
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p>Numerical Recipes in C. Second Edition. </p>


<h3>See Also</h3>

<p> cor.matrix, cor.vector, cor.mv </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
x &lt;-c(1,1,1,4,2,2,3,1,3,2,1,1,2,3,2,1,1,2,1,2)
y &lt;-c(1,1,2,3,4,4,2,1,2,3,1,1,3,4,2,1,1,3,1,2)
vark(x,y)
</code></pre>

<hr>
<h2 id='waerden.test'> Multiple comparisons. The van der Waerden (Normal Scores) </h2><span id='topic+waerden.test'></span>

<h3>Description</h3>

<p>A nonparametric test for several independent samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>waerden.test(y, trt, alpha=0.05, group=TRUE, main=NULL,console=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="waerden.test_+3A_y">y</code></td>
<td>
<p> Variable response </p>
</td></tr>
<tr><td><code id="waerden.test_+3A_trt">trt</code></td>
<td>
<p> Treatments </p>
</td></tr>
<tr><td><code id="waerden.test_+3A_alpha">alpha</code></td>
<td>
<p> Significant level </p>
</td></tr>
<tr><td><code id="waerden.test_+3A_group">group</code></td>
<td>
<p> TRUE or FALSE </p>
</td></tr>
<tr><td><code id="waerden.test_+3A_main">main</code></td>
<td>
<p> Title </p>
</td></tr>
<tr><td><code id="waerden.test_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The data consist of k samples of possibly unequal sample size.<br />
The post hoc test is using the criterium Fisher's least <br />
significant difference (LSD).
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>statistics</code></td>
<td>
<p>Statistics of the model</p>
</td></tr>
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>means</code></td>
<td>
<p>Statistical summary of the study variable</p>
</td></tr>
<tr><td><code>comparison</code></td>
<td>
<p>Comparison between treatments</p>
</td></tr>
<tr><td><code>groups</code></td>
<td>
<p>Formation of treatment groups</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

<p> Practical Nonparametrics Statistics. W.J. Conover, 1999 </p>


<h3>See Also</h3>

<p><code><a href="#topic+BIB.test">BIB.test</a></code>, <code><a href="#topic+DAU.test">DAU.test</a></code>, <code><a href="#topic+duncan.test">duncan.test</a></code>,
<code><a href="#topic+durbin.test">durbin.test</a></code>, <code><a href="#topic+friedman">friedman</a></code>, <code><a href="#topic+HSD.test">HSD.test</a></code>,
<code><a href="#topic+kruskal">kruskal</a></code>, <code><a href="#topic+LSD.test">LSD.test</a></code>, <code><a href="#topic+Median.test">Median.test</a></code>, 
<code><a href="#topic+PBIB.test">PBIB.test</a></code>, <code><a href="#topic+REGW.test">REGW.test</a></code>, <code><a href="#topic+scheffe.test">scheffe.test</a></code>, 
<code><a href="#topic+SNK.test">SNK.test</a></code>, <code><a href="#topic+waller.test">waller.test</a></code>, <code><a href="#topic+plot.group">plot.group</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
# example 1
data(corn)
out1&lt;-with(corn,waerden.test(observation,method,group=TRUE))
print(out1$groups)
plot(out1)
out2&lt;-with(corn,waerden.test(observation,method,group=FALSE))
print(out2$comparison)
# example 2
data(sweetpotato)
out&lt;-with(sweetpotato,waerden.test(yield,virus,alpha=0.01,group=TRUE))
print(out)
</code></pre>

<hr>
<h2 id='waller'> Computations of Bayesian t-values for multiple comparisons </h2><span id='topic+waller'></span>

<h3>Description</h3>

<p>A Bayes rule for the symmetric multiple comparisons problem.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>waller(K, q, f, Fc)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="waller_+3A_k">K</code></td>
<td>
<p> Is the loss ratio between type I and type II error </p>
</td></tr>
<tr><td><code id="waller_+3A_q">q</code></td>
<td>
<p> Numerator Degrees of freedom </p>
</td></tr>
<tr><td><code id="waller_+3A_f">f</code></td>
<td>
<p> Denominator Degrees of freedom </p>
</td></tr>
<tr><td><code id="waller_+3A_fc">Fc</code></td>
<td>
<p> F ratio from an analysis of variance </p>
</td></tr>
</table>


<h3>Details</h3>

<p>K-RATIO (K): value specifies the Type 1/Type 2 error seriousness ratio for 
the Waller-Duncan test. Reasonable values for KRATIO are 50, 100, and 500,
which roughly correspond for the two-level case to ALPHA levels of 0.1, 0.05,
and 0.01. By default, the procedure uses the default value of 100. 
</p>


<h3>Value</h3>

<p>Waller value for the Waller and Duncan test.
</p>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>References</h3>

 
<p>Waller, R. A. and Duncan, D. B. (1969).
A Bayes Rule for the Symmetric Multiple Comparison Problem,
Journal of the American Statistical Association 64, pages 1484-1504. 
</p>
<p>Waller, R. A. and Kemp, K. E. (1976)
Computations of Bayesian t-Values for Multiple Comparisons,
Journal of Statistical Computation and Simulation, 75, pages 169-172.
</p>
<p>Principles and procedures of statistics a biometrical approach
Steel &amp; Torry &amp; Dickey. Third Edition 1997.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+waller.test">waller.test</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># Table Duncan-Waller K=100, F=1.2 pag 649 Steel &amp; Torry
library(agricolae)
K&lt;-100
Fc&lt;-1.2
q&lt;-c(8,10,12,14,16,20,40,100)
f&lt;-c(seq(4,20,2),24,30,40,60,120)
n&lt;-length(q)
m&lt;-length(f)
W.D &lt;-rep(0,n*m)
dim(W.D)&lt;-c(n,m)
for (i in 1:n) {
for (j in 1:m) {
W.D[i,j]&lt;-waller(K, q[i], f[j], Fc)
}}
W.D&lt;-round(W.D,2)
dimnames(W.D)&lt;-list(q,f)
print(W.D)

</code></pre>

<hr>
<h2 id='waller.test'> Multiple comparisons, Waller-Duncan </h2><span id='topic+waller.test'></span>

<h3>Description</h3>

<p>The Waller-Duncan k-ratio t test is performed on all main effect means
in the MEANS statement. See the K-RATIO option for information on 
controlling details of the test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>waller.test(y, trt, DFerror, MSerror, Fc, K = 100, group=TRUE, main = NULL, 
console=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="waller.test_+3A_y">y</code></td>
<td>
<p> model(aov or lm) or answer of the experimental unit  </p>
</td></tr>
<tr><td><code id="waller.test_+3A_trt">trt</code></td>
<td>
<p> Constant( only y=model) or vector treatment applied to each unit</p>
</td></tr>
<tr><td><code id="waller.test_+3A_dferror">DFerror</code></td>
<td>
<p> Degrees of freedom </p>
</td></tr>
<tr><td><code id="waller.test_+3A_mserror">MSerror</code></td>
<td>
<p> Mean Square Error </p>
</td></tr>
<tr><td><code id="waller.test_+3A_fc">Fc</code></td>
<td>
<p> F Value </p>
</td></tr>
<tr><td><code id="waller.test_+3A_k">K</code></td>
<td>
<p> K-RATIO </p>
</td></tr>
<tr><td><code id="waller.test_+3A_group">group</code></td>
<td>
<p> TRUE or FALSE </p>
</td></tr>
<tr><td><code id="waller.test_+3A_main">main</code></td>
<td>
<p> Title </p>
</td></tr>
<tr><td><code id="waller.test_+3A_console">console</code></td>
<td>
<p>logical, print output </p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is necessary first makes a analysis of variance.<br />
</p>
<p>K-RATIO (K): value specifies the Type 1/Type 2 error seriousness ratio for 
the Waller-Duncan test. Reasonable values for KRATIO are 50, 100, and 500,
which roughly correspond for the two-level case to ALPHA levels of 0.1, 0.05,
and 0.01. By default, the procedure uses the default value of 100.<br />
</p>
<p>if y = model, then to apply the instruction:<br /> 
waller.test (model, &quot;trt&quot;, alpha = 0.05, group = TRUE, main = NULL, console = FALSE)<br />
where the model class is aov or lm.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>statistics</code></td>
<td>
<p>Statistics of the model</p>
</td></tr>
<tr><td><code>parameters</code></td>
<td>
<p>Design parameters</p>
</td></tr>
<tr><td><code>means</code></td>
<td>
<p>Statistical summary of the study variable</p>
</td></tr>
<tr><td><code>comparison</code></td>
<td>
<p>Comparison between treatments</p>
</td></tr>
<tr><td><code>groups</code></td>
<td>
<p>Formation of treatment groups</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu</p>


<h3>References</h3>

 
<p>Waller, R. A. and Duncan, D. B. (1969).
A Bayes Rule for the Symmetric Multiple Comparison Problem,
Journal of the American Statistical Association 64, pages 1484-1504.<br />
</p>
<p>Waller, R. A. and Kemp, K. E. (1976)
Computations of Bayesian t-Values for Multiple Comparisons,
Journal of Statistical Computation and Simulation, 75, pages 169-172.<br />
</p>
<p>Steel &amp; Torry &amp; Dickey. Third Edition 1997
Principles and procedures of statistics a biometrical approach
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BIB.test">BIB.test</a></code>, <code><a href="#topic+DAU.test">DAU.test</a></code>, <code><a href="#topic+duncan.test">duncan.test</a></code>,
<code><a href="#topic+durbin.test">durbin.test</a></code>, <code><a href="#topic+friedman">friedman</a></code>, <code><a href="#topic+HSD.test">HSD.test</a></code>,
<code><a href="#topic+kruskal">kruskal</a></code>, <code><a href="#topic+LSD.test">LSD.test</a></code>, <code><a href="#topic+Median.test">Median.test</a></code>, 
<code><a href="#topic+PBIB.test">PBIB.test</a></code>, <code><a href="#topic+REGW.test">REGW.test</a></code>, <code><a href="#topic+scheffe.test">scheffe.test</a></code>, 
<code><a href="#topic+SNK.test">SNK.test</a></code>, <code><a href="#topic+waerden.test">waerden.test</a></code>, <code><a href="#topic+plot.group">plot.group</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(sweetpotato)
model&lt;-aov(yield~virus, data=sweetpotato)
out &lt;- waller.test(model,"virus", group=TRUE)
#startgraph
oldpar&lt;-par(mfrow=c(2,2))
# variation: SE is error standard
# variation: range is Max - Min
bar.err(out$means,variation="SD",horiz=TRUE,xlim=c(0,45),bar=FALSE,
col=colors()[25],space=2, main="Standard deviation",las=1)
bar.err(out$means,variation="SE",horiz=FALSE,ylim=c(0,45),bar=FALSE,
col=colors()[15],space=2,main="SE",las=1)
bar.err(out$means,variation="range",ylim=c(0,45),bar=FALSE,col="green",
space=3,main="Range = Max - Min",las=1)
bar.group(out$groups,horiz=FALSE,ylim=c(0,45),density=8,col="red", 
main="Groups",las=1)
#endgraph
# Old version HSD.test()
df&lt;-df.residual(model)
MSerror&lt;-deviance(model)/df
Fc&lt;-anova(model)["virus",4]
out &lt;- with(sweetpotato,waller.test(yield, virus, df, MSerror, Fc, group=TRUE))
print(out)
par(oldpar)
</code></pre>

<hr>
<h2 id='weatherSeverity'>
Weather and Severity
</h2><span id='topic+weatherSeverity'></span>

<h3>Description</h3>

<p>Weather and Severity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>weatherSeverity(weather,severity,dates,EmergDate,EndEpidDate,NoReadingsH,
RHthreshold)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="weatherSeverity_+3A_weather">weather</code></td>
<td>
<p>object, see example</p>
</td></tr>
<tr><td><code id="weatherSeverity_+3A_severity">severity</code></td>
<td>
<p>object, see example </p>
</td></tr>
<tr><td><code id="weatherSeverity_+3A_dates">dates</code></td>
<td>
<p>vector dates</p>
</td></tr>
<tr><td><code id="weatherSeverity_+3A_emergdate">EmergDate</code></td>
<td>
<p>date</p>
</td></tr>
<tr><td><code id="weatherSeverity_+3A_endepiddate">EndEpidDate</code></td>
<td>
<p>date</p>
</td></tr>
<tr><td><code id="weatherSeverity_+3A_noreadingsh">NoReadingsH</code></td>
<td>
<p>num, 1</p>
</td></tr>
<tr><td><code id="weatherSeverity_+3A_rhthreshold">RHthreshold</code></td>
<td>
<p>num, percentage</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Weather and severity
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Wfile</code></td>
<td>
<p>&quot;Date&quot;,&quot;Rainfall&quot;,&quot;Tmp&quot;,&quot;HumidHrs&quot;,&quot;humidtmp&quot;</p>
</td></tr>
<tr><td><code>Sfile</code></td>
<td>
<p>&quot;Cultivar&quot;,&quot;ApplSys&quot;,&quot;dates&quot;,&quot;nday&quot;,&quot;MeanSeverity&quot;,&quot;StDevSeverity&quot;</p>
</td></tr>
<tr><td><code>EmergDate</code></td>
<td>
<p>date</p>
</td></tr>
<tr><td><code>EndEpidDate</code></td>
<td>
<p>date</p>
</td></tr>
</table>


<h3>Note</h3>

<p>All format data for date is yyyy-mm,dd, for example &quot;2000-04-22&quot;.
change with function as.Date()
</p>


<h3>See Also</h3>

<p><code><a href="#topic+lateblight">lateblight</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
f &lt;- system.file("external/weather.csv", package="agricolae")
weather &lt;- read.csv(f,header=FALSE)
f &lt;- system.file("external/severity.csv", package="agricolae")
severity &lt;- read.csv(f)
weather[,1]&lt;-as.Date(weather[,1],format = "%m/%d/%Y")
# Parameters dates and threshold
dates&lt;-c("2000-03-25","2000-04-09","2000-04-12","2000-04-16","2000-04-22")
dates&lt;-as.Date(dates)
EmergDate &lt;- as.Date('2000/01/19')
EndEpidDate &lt;- as.Date("2000-04-22")
dates&lt;-as.Date(dates)
NoReadingsH&lt;- 1
RHthreshold &lt;- 90
#--------------------------
WS&lt;-weatherSeverity(weather,severity,dates,EmergDate,EndEpidDate,
NoReadingsH,RHthreshold)
</code></pre>

<hr>
<h2 id='wilt'> Data of Bacterial Wilt (AUDPC) and soil</h2><span id='topic+wilt'></span>

<h3>Description</h3>

<p>Percentage of bacterial wilt and area under the curve of disease progression 
(AUDPC) relative tomato plants transplanted in different soil types artificially 
infested with R.solanacearum 133 days before. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(wilt)</code></pre>


<h3>Format</h3>

<p>A data frame with 13 observations on the following 15 variables.
</p>

<dl>
<dt><code>place</code></dt><dd><p>a factor with levels <code>Chmar</code> <code>Chz</code> <code>Cnt1</code> <code>Cnt2</code> <code>Cnt3</code> <code>Hco1</code> <code>Hco2</code> <code>Hco3</code> <code>Hyo1</code> <code>Hyo2</code> <code>Namora</code> <code>SR1</code> <code>SR2</code></p>
</dd>
<dt><code>Day7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Day11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Day15</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Day19</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Day23</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Day27</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Day31</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Day35</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Day39</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Day43</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Day47</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Day51</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>AUDPC</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>relative</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>Percentajes bacterial wilt. Day7 = evaluated to 7 days, Days11 = evaluated to 11 days.
see data(soil) and data(ralstonia) </p>


<h3>Source</h3>

<p>Experimental field, 2004. Data Kindly provided by Dr. Sylvie Priou, Liliam Gutarra and Pedro Aley.
</p>


<h3>References</h3>

<p>International Potato Center. CIP - Lima Peru.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(wilt)
days&lt;-c(7,11,15,19,23,27,31,35,39,43,47,51)
AUDPC&lt;-audpc(wilt[,-1],days)
relative&lt;-audpc(wilt[,-1],days,type="relative")

</code></pre>

<hr>
<h2 id='yacon'> Data Yacon</h2><span id='topic+yacon'></span>

<h3>Description</h3>

<p>The yacon (Smallanthus sonchifolius) is a plant native to the Andes, considered a traditional crop in Peru and
natural source of FOS, which is a type of carbohydrate that can not be digested by the
and the human body that have joined several beneficial properties in health, such as
improve the absorption of calcium, reducing the level of triglycerides and cholesterol
and stimulate better gastrointestinal function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(yacon)</code></pre>


<h3>Format</h3>

<p>A data frame with 432 observations on the following 19 variables.
</p>

<dl>
<dt><code>locality</code></dt><dd><p>a factor with levels, Cajamarca, Lima, Oxapampa in PERU</p>
</dd>
<dt><code>site</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>dose</code></dt><dd><p>a factor with levels <code>F0</code> <code>F150</code> <code>F80</code></p>
</dd>
<dt><code>entry</code></dt><dd><p>a factor with levels <code>AKW5075 </code> <code>AMM5136</code> <code>AMM5150</code> <code>AMM5163</code> <code>ARB5125 </code> <code>CLLUNC118</code> <code>P1385</code> <code>SAL136</code></p>
</dd>
<dt><code>replication</code></dt><dd><p>a numeric vector, replications</p>
</dd>
<dt><code>height</code></dt><dd><p>a numeric vector, plant height, centimeters</p>
</dd>
<dt><code>stalks</code></dt><dd><p>a numeric vector, number of stalks</p>
</dd>
<dt><code>wfr</code></dt><dd><p>a numeric vector, weight of fresh roots, grams</p>
</dd>
<dt><code>wff</code></dt><dd><p>a numeric vector, weight of fresh foliage, grams</p>
</dd>
<dt><code>wfk</code></dt><dd><p>a numeric vector, weight fresh kroner, grams</p>
</dd>
<dt><code>roots</code></dt><dd><p>a numeric vector, matter of dried roots, grams</p>
</dd>
<dt><code>FOS</code></dt><dd><p>a numeric vector, fructo-oligosaccharides, percentaje</p>
</dd>
<dt><code>glucose</code></dt><dd><p>a numeric vector, percentaje</p>
</dd>
<dt><code>fructose</code></dt><dd><p>a numeric vector, percentaje</p>
</dd>
<dt><code>sucrose</code></dt><dd><p>a numeric vector, percentaje</p>
</dd>
<dt><code>brix</code></dt><dd><p>a numeric vector, degrees Brix</p>
</dd>
<dt><code>foliage</code></dt><dd><p>a numeric vector, matter dry foliage, grams</p>
</dd>
<dt><code>dry</code></dt><dd><p>a numeric vector, dry matter kroner, grams</p>
</dd>
<dt><code>IH</code></dt><dd><p>a numeric vector, Index harvest, 0 to 1</p>
</dd>
</dl>



<h3>Details</h3>

<p>Proportion or fraction of the plant that is used (seeds, fruit, root) on dry basis.
Part usable in a proportion of total mass dissected. Plant of frijol, weight = 100g
and frijol = 50g then, IH = 50/100 = 0.5 or 50 percentaje. Degrees Brix is 
a measurement of the mass ratio of dissolved sugar to water in a liquid.
</p>


<h3>Source</h3>

<p>CIP. Experimental field, 2003, Data Kindly provided by Ivan Manrique and Carolina Tasso.
</p>


<h3>References</h3>

<p>International Potato Center. CIP - Lima Peru.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
data(yacon)
str(yacon) 
</code></pre>

<hr>
<h2 id='zigzag'> order plot in serpentine </h2><span id='topic+zigzag'></span>

<h3>Description</h3>

<p>applied to designs: complete block, latin square, graeco, split plot, strip plot, lattice, alpha lattice,
Augmented block, cyclic, Balanced Incomplete Block and factorial.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zigzag(outdesign)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="zigzag_+3A_outdesign">outdesign</code></td>
<td>
<p> output design</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>fieldbook</code></td>
<td>
<p>Remuneration of serpentine plots.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Felipe de Mendiburu </p>


<h3>See Also</h3>

<p><code><a href="#topic+design.ab">design.ab</a></code>, <code><a href="#topic+design.alpha">design.alpha</a></code>,<code><a href="#topic+design.bib">design.bib</a></code>, 
<code><a href="#topic+design.split">design.split</a> </code>, <code><a href="#topic+design.cyclic">design.cyclic</a> </code>, <code><a href="#topic+design.dau">design.dau</a> </code>,
<code><a href="#topic+design.graeco">design.graeco</a></code>, <code><a href="#topic+design.lattice">design.lattice</a></code>, <code><a href="#topic+design.lsd">design.lsd</a></code>,
<code><a href="#topic+design.rcbd">design.rcbd</a></code>, <code><a href="#topic+design.strip">design.strip</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(agricolae)
trt&lt;-letters[1:5]
r&lt;-4
outdesign &lt;- design.rcbd(trt,r,seed=9)
fieldbook &lt;- zigzag(outdesign)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
