<!DOCTYPE html><html><head><title>Help for package agvgd</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {agvgd}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#agvgd-package'><p>agvgd: An R Implementation of the 'Align-GVGD' Method</p></a></li>
<li><a href='#%&gt;%'><p>Pipe operator</p></a></li>
<li><a href='#agvgd'><p>Align-GVGD (A-GVGD)</p></a></li>
<li><a href='#alignment_file'><p>Pre-bundled alignments</p></a></li>
<li><a href='#amino_acids'><p>The 20 standard amino acids</p></a></li>
<li><a href='#cpv_ranges'><p>Determine CPV ranges</p></a></li>
<li><a href='#dev'><p>Deviation function</p></a></li>
<li><a href='#gd'><p>Grantham deviation</p></a></li>
<li><a href='#gv'><p>Grantham variation</p></a></li>
<li><a href='#poi_to_res'><p>Convert an alignment position to residue position</p></a></li>
<li><a href='#profile_to_alignment'><p>Converts a sequence profile to an alignment</p></a></li>
<li><a href='#read_agvgdweb_results'><p>Read in AGVGD Web Results</p></a></li>
<li><a href='#read_alignment'><p>Read a protein sequence multiple alignment</p></a></li>
<li><a href='#read_substitutions'><p>Read a file with amino acid substitutions</p></a></li>
<li><a href='#res_to_poi'><p>Convert a residue position to an alignment position</p></a></li>
<li><a href='#str_chop'><p>Chop a string into sub-strings of fixed width</p></a></li>
<li><a href='#write_alignment'><p>Export an alignment to FASTA</p></a></li>
<li><a href='#write_substitutions'><p>Generate and export a list of substitutions</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>An R Implementation of the 'Align-GVGD' Method</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.2</td>
</tr>
<tr>
<td>Description:</td>
<td>'Align-GVGD' ('A-GVGD') is a method to predict the impact of
    'missense' substitutions based on the properties of amino acid side
    chains and protein multiple sequence alignments
    &lt;<a href="https://doi.org/10.1136%2Fjmg.2005.033878">doi:10.1136/jmg.2005.033878</a>&gt;. 'A-GVGD' is an extension of the original
    'Grantham' distance to multiple sequence alignments. This package
    provides an alternative R implementation to the web version found on
    <a href="http://agvgd.hci.utah.edu/">http://agvgd.hci.utah.edu/</a>.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>crayon, dplyr, glue, grantham, magrittr, purrr, rlang, seqinr,
stringr, tibble, tidyr, vctrs</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, covr, testthat (&ge; 3.0.0),</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://maialab.org/agvgd/">https://maialab.org/agvgd/</a>, <a href="https://github.com/maialab/agvgd">https://github.com/maialab/agvgd</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/maialab/agvgd/issues">https://github.com/maialab/agvgd/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-09-10 19:30:44 UTC; rmagno</td>
</tr>
<tr>
<td>Author:</td>
<td>Ramiro Magno <a href="https://orcid.org/0000-0001-5226-3441"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Isabel Duarte <a href="https://orcid.org/0000-0003-0060-2936"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Ana-Teresa Maia <a href="https://orcid.org/0000-0002-0454-9207"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  CINTESIS [cph, fnd]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ramiro Magno &lt;ramiro.magno@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-09-10 19:42:54 UTC</td>
</tr>
</table>
<hr>
<h2 id='agvgd-package'>agvgd: An R Implementation of the 'Align-GVGD' Method</h2><span id='topic+agvgd-package'></span><span id='topic+_PACKAGE'></span>

<h3>Description</h3>

<p>'Align-GVGD' ('A-GVGD') is a method to predict the impact of 'missense' substitutions based on the properties of amino acid side chains and protein multiple sequence alignments <a href="https://doi.org/10.1136/jmg.2005.033878">doi:10.1136/jmg.2005.033878</a>. 'A-GVGD' is an extension of the original 'Grantham' distance to multiple sequence alignments. This package provides an alternative R implementation to the web version found on <a href="http://agvgd.hci.utah.edu/">http://agvgd.hci.utah.edu/</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Ramiro Magno <a href="mailto:ramiro.magno@gmail.com">ramiro.magno@gmail.com</a> (<a href="https://orcid.org/0000-0001-5226-3441">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Isabel Duarte <a href="mailto:iduarte.scientist@gmail.com">iduarte.scientist@gmail.com</a> (<a href="https://orcid.org/0000-0003-0060-2936">ORCID</a>)
</p>
</li>
<li><p> Ana-Teresa Maia <a href="mailto:maia.anateresa@gmail.com">maia.anateresa@gmail.com</a> (<a href="https://orcid.org/0000-0002-0454-9207">ORCID</a>)
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> CINTESIS [copyright holder, funder]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://maialab.org/agvgd/">https://maialab.org/agvgd/</a>
</p>
</li>
<li> <p><a href="https://github.com/maialab/agvgd">https://github.com/maialab/agvgd</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/maialab/agvgd/issues">https://github.com/maialab/agvgd/issues</a>
</p>
</li></ul>


<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling 'rhs(lhs)'.
</p>

<hr>
<h2 id='agvgd'>Align-GVGD (A-GVGD)</h2><span id='topic+agvgd'></span>

<h3>Description</h3>

<p>This function implements the Align-GVGD (A-GVGD) method described in
Tavtigian <em>et al.</em> (2006).
</p>
<p>A-GVGD combines multiple sequence alignment of orthologous sequences with the
Grantham distance to classify missense variants, i.e. to distinguish human
disease susceptibility missense changes from changes of little clinical
significance.
</p>
<p>The biochemical variation at each alignment position is converted to a
Grantham Variation score (GV) and the difference between these properties and
those of the variant amino acid being assessed are calculated and a Grantham
Difference score generated (GD).  The predicted effect is classed as C0, C15,
C25, C35, C45, C55, or C65, with C65 most likely to interfere with function
and C0 least likely.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>agvgd(
  alignment,
  poi,
  sub,
  mode = c("recycle", "expand_grid"),
  sort = FALSE,
  keep_self = TRUE,
  digits = 2L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="agvgd_+3A_alignment">alignment</code></td>
<td>
<p>A character matrix or an alignment object obtained with
<code><a href="#topic+read_alignment">read_alignment()</a></code>. Rows are expected to be sequences of single characters
(protein residues), and columns the alignment positions. The first row must
be the reference sequence, i.e. the sequence whose substitutions will be
evaluated against.</p>
</td></tr>
<tr><td><code id="agvgd_+3A_poi">poi</code></td>
<td>
<p>A whole number indicating the position of interest (POI).</p>
</td></tr>
<tr><td><code id="agvgd_+3A_sub">sub</code></td>
<td>
<p>A character vector of protein residue substitutions to be
classified. The amino acids must be provided as one-letter symbols.</p>
</td></tr>
<tr><td><code id="agvgd_+3A_mode">mode</code></td>
<td>
<p>If both <code>poi</code> and <code>sub</code> contain more than one element, <code>mode</code>
specifies how these two inputs are combined. If <code>mode = 'recycle'</code> the
shortest vector is recycled to match the length of the longest. If <code>mode = 'expand_grid'</code>, all combinations between elements of <code>poi</code> and <code>sub</code> are
combined.</p>
</td></tr>
<tr><td><code id="agvgd_+3A_sort">sort</code></td>
<td>
<p>Whether to sort the output by <code>gd</code>, or not. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="agvgd_+3A_keep_self">keep_self</code></td>
<td>
<p>Whether to keep those results in the output that correspond
to residues being the same in <code>ref</code> and <code>sub</code>. Default is <code>TRUE</code>. But it
will be useful to change it to <code>FALSE</code> if want to compare the results with
those provided by <a href="http://agvgd.hci.utah.edu/">http://agvgd.hci.utah.edu/</a> that filters them out.</p>
</td></tr>
<tr><td><code id="agvgd_+3A_digits">digits</code></td>
<td>
<p>Integer indicating the number of decimal places to be used in
rounding <code>gv</code> and <code>gd</code> values. Default is <code>2</code>. Note that the calculation of
the <code>prediction</code> variable won't be affected by rounding of <code>gv</code> and <code>gd</code>,
as it is calculated prior to the rounding.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="tibble.html#topic+tibble-package">tibble</a> whose observations refer to the
combination alignment position and amino acid substitution; consists of
seven variables:
</p>

<dl>
<dt>res</dt><dd><p>Position of the amino acid residue in the reference protein
(first sequence in the alignment). This position corresponds to <code>poi</code> minus
the gaps in the alignment.</p>
</dd>
<dt>poi</dt><dd><p>Position of interest, i.e. the alignment position at which the
amino acid substitution is being assessed.</p>
</dd>
<dt>ref</dt><dd><p>Reference amino acid, i.e. the amino acid in the first sequence
of the alignment, at the position of interest.</p>
</dd>
<dt>sub</dt><dd><p>Amino acid substitution being assessed.</p>
</dd>
<dt>gv</dt><dd><p>Grantham variation score.</p>
</dd>
<dt>gd</dt><dd><p>Grantham difference score.</p>
</dd>
<dt>prediction</dt><dd><p>Predicted effect of the amino acid substitution. This is
classed as C0, C15, C25, C35, C45, C55, or C65, with C65 most likely to
interfere with function and C0 least likely.</p>
</dd>
</dl>



<h3>References</h3>


<ul>
<li><p> Tavtigian, S.V., Deffenbaugh, A. M., Yin, L., Judkins, T., Scholl, T.,
Samollow, P.B., de Silva, D., Zharkikh, A., Thomas, A. <em>Comprehensive
statistical study of 452 BRCA1 missense substitutions with classification of
eight recurrent substitutions as neutral</em>. Journal of Medical Genetics 43,
295&ndash;305 (2006). <a href="https://doi.org/10.1136/jmg.2005.033878">doi:10.1136/jmg.2005.033878</a>.
</p>
</li>
<li><p> Mathe, E., Olivier, M., Kato, S., Ishioka, C., Hainaut, P., Tavtigian, S.V.
<em>Computational approaches for predicting the biological effect of p53
missense mutations: a comparison of three sequence analysis based methods</em>.
Nucleic Acids Research 34, 1317&ndash;1325 (2006). <a href="https://doi.org/10.1093/nar/gkj518">doi:10.1093/nar/gkj518</a>.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Read an alignment into R, e.g. the alignment for gene ATM.
alignment_ATM &lt;- read_alignment(gene = 'ATM')

# Predict the impact of changing the first residue (Met) to a Serine (S).
agvgd(alignment = alignment_ATM, poi = 1, sub = 'S')

# `poi` can be a vector of positions, e.g., 3 thru 10, allow for prediction
# of multiple positions at once.
agvgd(alignment = alignment_ATM, poi = 3:10, sub = 'S')

# `poi` expects a position in the frame of reference of the alignment, i.e.
# an alignment position (a column index). However, if you know instead
# the residue position in the reference sequence (first sequence in the
# alignment), then you may use the function `res_to_poi()`
# to convert from residue position to alignment position.
#
# Example: The second residue in the reference sequence of the ATM alignment
# is a Serine, after a Methionine. In the alignment, there is a gap between
# the two residues, so the alignment is 3 but the residue position on the
# protein is 2.
(poi2 &lt;- res_to_poi(alignment_ATM, 2))
agvgd(alignment = alignment_ATM, poi = poi2, sub = 'A')

# Because changes are context-dependent, i.e. they depend on the residue
# variation observed at a given alignment position, the same reference
# residue when replaced with the same substitution will in general have
# a different predicted impact.
agvgd(alignment = alignment_ATM, poi = 9:10, sub = 'S')

# Use the ancillary function `amino_acids()` to get a vector of one-letter
# residue substitutions if you want to quickly assess the impact of all
# possible substitutions.
agvgd(alignment = alignment_ATM, poi = 1, sub = amino_acids())

# Parameter `mode` gives you flexibility on how to combine `poi` and `sub`.
agvgd(alignment = alignment_ATM, poi = 3:4, sub = c('A', 'V'))

# Use 'expand_grid' for all combinations.
agvgd(alignment = alignment_ATM, poi = 3:4, sub = c('A', 'V'), mode = 'expand_grid')

</code></pre>

<hr>
<h2 id='alignment_file'>Pre-bundled alignments</h2><span id='topic+alignment_file'></span>

<h3>Description</h3>

<p>This function returns either a data frame of the pre-bundled alignments if
parameter <code>gene</code> is missing (default behaviour), or the file name of the
alignment of a supplied gene name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alignment_file(gene)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alignment_file_+3A_gene">gene</code></td>
<td>
<p>The gene name of one of the pre-bundled alignments. Run
<code><a href="#topic+alignment_file">alignment_file()</a></code> to list all genes available.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Either a data frame of the pre-bundled alignments if parameter <code>gene</code>
is missing (default behaviour), or the file name of the alignment of a
supplied gene name.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># List pre-bundled alignment file names and associated genes
alignment_file()

# Retrieve the file name of an alignment
alignment_file("BRCA1")

# You may get the full path to an alignment file with `system.file()`
system.file("extdata", alignment_file("BRCA1"), package = "agvgd")

</code></pre>

<hr>
<h2 id='amino_acids'>The 20 standard amino acids</h2><span id='topic+amino_acids'></span>

<h3>Description</h3>

<p>The 20 amino acids that are encoded directly by the codons of the universal
genetic code.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>amino_acids(code = c("one_letter", "three_letter"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="amino_acids_+3A_code">code</code></td>
<td>
<p>The type of amino acid symbol to be returned, one-letter
('one_letter') or three-letter ('three_letter') codes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of the 20 standard amino acids.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># By default `amino_acids` returns one-letter symbols
amino_acids()

# Use code = 'three_letter' instead for three-letter symbols
amino_acids(code = 'three_letter')

</code></pre>

<hr>
<h2 id='cpv_ranges'>Determine CPV ranges</h2><span id='topic+cpv_ranges'></span>

<h3>Description</h3>

<p>This function determines the range (minimum and maximum) values for the three
amino acid side chain property values &mdash; composition, polarity and molecular
volume &mdash; from the amino acids at the alignment position of interest.
</p>
<p>The alignment passed in <code>alignment</code> must be an already focused alignment of
three columns whose second column is the position of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cpv_ranges(alignment, exclude = c("-", "X", NA_character_))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cpv_ranges_+3A_alignment">alignment</code></td>
<td>
<p>A character matrix or an alignment object obtained with
<code><a href="#topic+read_alignment">read_alignment()</a></code>. Rows are expected to be sequences of single characters
(protein residues), and columns the alignment positions. The first row must
be the reference sequence, i.e. the sequence whose substitutions will be
evaluated against.</p>
</td></tr>
<tr><td><code id="cpv_ranges_+3A_exclude">exclude</code></td>
<td>
<p>A vector of character values to be ignored when collecting the
amino acids at the position of interest.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="tibble.html#topic+tibble-package">tibble</a> with one single row, of six
variables, i.e., the minimum and maximum values for composition (<code>c_min</code>
and <code>c_max</code>), polarity (<code>p_min</code> and <code>p_max</code>) and molecular volume (<code>v_min</code>
and <code>v_max</code>).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gv">gv()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># You need to first focus the alignment around the position of interest. The
# position of interest is position 4 in the example below. After subsetting
# the alignment, it becomes position 2.
alignment &lt;- read_alignment('ATM')

alignment[, 3:5]

cpv_ranges(alignment[, 3:5])

# If at the position of interest there are symbols other than amino acid
# symbols, e.g. gaps ("-"), then these are ignored and the calculated ranges
# are based only on the observed amino acids.
alignment[, 270:272]

cpv_ranges(alignment[, 270:272])

</code></pre>

<hr>
<h2 id='dev'>Deviation function</h2><span id='topic+dev'></span>

<h3>Description</h3>

<p>This function calculates the deviation in the sense of the Grantham deviation
as introduced by Tavtigian et al. (2006). Essentially, if <code>x</code> lies within the
range [<code>min</code>, <code>max</code>], then <code>dev()</code> returns 0. If <code>x</code> is either below <code>min</code>, or
above <code>max</code>, then <code>dev()</code> returns the absolute difference between <code>x</code> and
<code>min</code> or <code>max</code>, respectively.
</p>
<p><img src="../help/figures/dev-def.svg" alt="dev definition" />

</p>
<p>Inputs are recycled in the sense of <code><a href="vctrs.html#topic+vec_recycle">vctrs::vec_recycle()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dev(x, min, max)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dev_+3A_x">x</code></td>
<td>
<p>A numeric vector.</p>
</td></tr>
<tr><td><code id="dev_+3A_min">min</code></td>
<td>
<p>A numeric vector.</p>
</td></tr>
<tr><td><code id="dev_+3A_max">max</code></td>
<td>
<p>A numeric vector.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Here's a plot showcasing <code>dev()</code> with <code>min = -4</code> and <code>max = 3</code>:
</p>
<div class="sourceCode r"><pre>x &lt;- -10:10; min &lt;- -4; max &lt;- 3
plot(x, y = dev(x, min, max), type = 'l', xlab = 'x', ylab = 'deviation')
</pre></div>
<p><img src="../help/figures/dev-plot-1.png" alt="dev-plot-1.png" />
</p>


<h3>Value</h3>

<p>A numeric vector of deviations.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gd">gd()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># `dev()` returns absolute differences from either min or max (whichever is
# closest).
dev(10, min = -4, max = 4)
dev(-10, min = -4, max = 4)

# `x` can be a vector
dev(-10:10, min = -4, max = 4)

# `min` and `max` can also be vectors, they will be recycled
dev(-10:10, min = -4:16, max = 4:24)

# If `x` contains `NA` values, then `dev()` will return `NA` for
# those cases
dev(c(10, NA), min = -4, max = 4)

# For each calculation of deviation, only either `min` or `max` is used. If
# the unused parameter is `NA` it won't affect the calculation:
dev(c(10, 3), min = c(NA, -4), max = 4)
dev(c(10, -5), min = -4, max = c(4, NA))

</code></pre>

<hr>
<h2 id='gd'>Grantham deviation</h2><span id='topic+gd'></span>

<h3>Description</h3>

<p>This function calculates the Grantham deviation (<code class="reqn">\mathrm{gd}</code>):
</p>
<p style="text-align: center;"><code class="reqn">\mathrm{gd} = \rho \left((\alpha\ \mathrm{dev}^2(c_x, c_{min}, c_{max}) + \beta\ \mathrm{dev}^2(p_x, p_{min}, p_{max}) + \gamma\ \mathrm{dev}^2(v_x, v_{min}, v_{max})\right)^\frac{1}{2}</code>
</p>

<p>where <code class="reqn">c_x</code> is the value for composition <code class="reqn">c</code> of amino acid <code class="reqn">x</code>,
i.e. the atomic weight ratio of hetero (noncarbon) elements in end groups or
rings to carbons in the side chain; <code class="reqn">p_x</code> is the value for polarity
<code class="reqn">p</code> of amino acid <code class="reqn">x</code>; and, <code class="reqn">v_x</code> is the value for molecular
volume <code class="reqn">v</code> of amino acid <code class="reqn">x</code>.
</p>
<p><code class="reqn">c_x</code>, <code class="reqn">p_x</code> and <code class="reqn">v_x</code> are looked up in
<a href="grantham.html#topic+amino_acids_properties">grantham::amino_acids_properties</a> based on the amino acid identities passed
in <code>x</code>. The function <code class="reqn">\mathrm{dev}</code> is implemented in <code><a href="#topic+dev">dev()</a></code>. Remaining
variables in the equation are arguments to <code>gd()</code> and hence are explained
below in the Arguments section.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gd(
  x,
  c_min,
  c_max,
  p_min,
  p_max,
  v_min,
  v_max,
  alpha = 1.833,
  beta = 0.1018,
  gamma = 0.000399,
  rho = 50.723
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gd_+3A_x">x</code></td>
<td>
<p>A character vector of one-letter amino acid codes, indicating
missense substitutions.</p>
</td></tr>
<tr><td><code id="gd_+3A_c_min">c_min</code></td>
<td>
<p>Amino acid composition, minimum value.</p>
</td></tr>
<tr><td><code id="gd_+3A_c_max">c_max</code></td>
<td>
<p>Amino acid, composition, maximum value.</p>
</td></tr>
<tr><td><code id="gd_+3A_p_min">p_min</code></td>
<td>
<p>Amino acid polarity, minimum value.</p>
</td></tr>
<tr><td><code id="gd_+3A_p_max">p_max</code></td>
<td>
<p>Amino acid polarity, maximum value.</p>
</td></tr>
<tr><td><code id="gd_+3A_v_min">v_min</code></td>
<td>
<p>Amino acid molecular volume, maximum value.</p>
</td></tr>
<tr><td><code id="gd_+3A_v_max">v_max</code></td>
<td>
<p>Amino acid molecular volume, maximum value.</p>
</td></tr>
<tr><td><code id="gd_+3A_alpha">alpha</code></td>
<td>
<p>The constant <code class="reqn">\alpha</code> in Grantham's equation. It is the
square inverse of the mean of the composition property.</p>
</td></tr>
<tr><td><code id="gd_+3A_beta">beta</code></td>
<td>
<p>The constant <code class="reqn">\beta</code> in Grantham's equation. It is the
square inverse of the mean of the polarity property.</p>
</td></tr>
<tr><td><code id="gd_+3A_gamma">gamma</code></td>
<td>
<p>The constant <code class="reqn">\gamma</code> in Grantham's equation. It is the
square inverse of the mean of the molecular volume property.</p>
</td></tr>
<tr><td><code id="gd_+3A_rho">rho</code></td>
<td>
<p>Grantham's distances reported in Table 2, Science (1974).
185(4154): 862&ndash;4 by R. Grantham, are scaled by a factor (here named
<code class="reqn">\rho</code>) such that the mean value of all distances are 100. The <code>rho</code>
parameter allows this factor <code class="reqn">\rho</code> to be changed. By default
<code class="reqn">\rho=50.723</code>, the same value used by Grantham. This value is
originally mentioned in the caption of Table 2 of the aforementioned paper.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector of Grantham deviations. Each deviation corresponds
to one of the amino acids indicated in <code>x</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gv">gv()</a></code>, <code><a href="#topic+dev">dev()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gd('S', c_min = 0.39, c_max = 0.74, p_min =4.9, p_max =8.6, v_min = 3, v_max = 32.5)

</code></pre>

<hr>
<h2 id='gv'>Grantham variation</h2><span id='topic+gv'></span>

<h3>Description</h3>

<p>This function calculates the Grantham variation (<code class="reqn">\mathrm{gv}</code>):
</p>
<p style="text-align: center;"><code class="reqn">\mathrm{gv} = \rho \left((\alpha (c_{max}-c_{min})^2 + \beta (p_{max}-p_{min})^2 + \gamma (v_{max}-v_{min})^2\right)^\frac{1}{2}</code>
</p>

<p>The minimum and maximum values are those observed for a set of amino acid
residues at the alignment position of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gv(
  c_min,
  c_max,
  p_min,
  p_max,
  v_min,
  v_max,
  alpha = 1.833,
  beta = 0.1018,
  gamma = 0.000399,
  rho = 50.723
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gv_+3A_c_min">c_min</code></td>
<td>
<p>Amino acid composition, minimum value.</p>
</td></tr>
<tr><td><code id="gv_+3A_c_max">c_max</code></td>
<td>
<p>Amino acid, composition, maximum value.</p>
</td></tr>
<tr><td><code id="gv_+3A_p_min">p_min</code></td>
<td>
<p>Amino acid polarity, minimum value.</p>
</td></tr>
<tr><td><code id="gv_+3A_p_max">p_max</code></td>
<td>
<p>Amino acid polarity, maximum value.</p>
</td></tr>
<tr><td><code id="gv_+3A_v_min">v_min</code></td>
<td>
<p>Amino acid molecular volume, maximum value.</p>
</td></tr>
<tr><td><code id="gv_+3A_v_max">v_max</code></td>
<td>
<p>Amino acid molecular volume, maximum value.</p>
</td></tr>
<tr><td><code id="gv_+3A_alpha">alpha</code></td>
<td>
<p>The constant <code class="reqn">\alpha</code> in Grantham's equation. It is the
square inverse of the mean of the composition property.</p>
</td></tr>
<tr><td><code id="gv_+3A_beta">beta</code></td>
<td>
<p>The constant <code class="reqn">\beta</code> in Grantham's equation. It is the
square inverse of the mean of the polarity property.</p>
</td></tr>
<tr><td><code id="gv_+3A_gamma">gamma</code></td>
<td>
<p>The constant <code class="reqn">\gamma</code> in Grantham's equation. It is the
square inverse of the mean of the molecular volume property.</p>
</td></tr>
<tr><td><code id="gv_+3A_rho">rho</code></td>
<td>
<p>Grantham's distances reported in Table 2, Science (1974).
185(4154): 862&ndash;4 by R. Grantham, are scaled by a factor (here named
<code class="reqn">\rho</code>) such that the mean value of all distances are 100. The <code>rho</code>
parameter allows this factor <code class="reqn">\rho</code> to be changed. By default
<code class="reqn">\rho=50.723</code>, the same value used by Grantham. This value is
originally mentioned in the caption of Table 2 of the aforementioned paper.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector of grantham variation values.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gd">gd()</a></code>, <code><a href="#topic+cpv_ranges">cpv_ranges()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example based on values from Figure 1C of Tavtigian et al. (2006),
# https://doi.org/10.1136/jmg.2005.033878.
gv(c_min = 0, c_max = 0, p_min = 5.7, p_max = 4.9, v_min = 132, v_max = 105)

</code></pre>

<hr>
<h2 id='poi_to_res'>Convert an alignment position to residue position</h2><span id='topic+poi_to_res'></span>

<h3>Description</h3>

<p>This function converts an alignment position to a position in the frame of
the reference protein sequence, i.e., to the positions of the amino acids in
the first sequence of the alignment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>poi_to_res(alignment, poi)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="poi_to_res_+3A_alignment">alignment</code></td>
<td>
<p>An alignment.</p>
</td></tr>
<tr><td><code id="poi_to_res_+3A_poi">poi</code></td>
<td>
<p>An alignment position.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer vector of positions of the amino acid residues in the
reference sequence.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>align_ATM &lt;- read_alignment('ATM')
align_ATM[, 1:5]

# Convert the positions of the first five alignment positions to residue positions
poi_to_res(align_ATM, 1:5)

</code></pre>

<hr>
<h2 id='profile_to_alignment'>Converts a sequence profile to an alignment</h2><span id='topic+profile_to_alignment'></span>

<h3>Description</h3>

<p>This function converts a sequence profile as provided in the format of the
package <code>{protean}</code> to an <code>{agvgd}</code> alignment &mdash; an alignment in this sense
is simply a character matrix whose elements are protein residues in
one-letter notation, rows are sequences and columns correspond to alignment
positions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>profile_to_alignment(profile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="profile_to_alignment_+3A_profile">profile</code></td>
<td>
<p>A sequence profile object as returned by
<code>protean::read_profile()</code> or <code>protean::get_profile()</code>. See the <code>{protean}</code>
package at <a href="https://github.com/maialab/protean">https://github.com/maialab/protean</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An alignment object, i.e. a character matrix whose elements are
protein residues in one-letter notation. Rows are sequences and columns are
alignment positions.
</p>

<hr>
<h2 id='read_agvgdweb_results'>Read in AGVGD Web Results</h2><span id='topic+read_agvgdweb_results'></span>

<h3>Description</h3>

<p>This function imports into R the results generated by the AGVGD Web
application <a href="http://agvgd.hci.utah.edu/">http://agvgd.hci.utah.edu/</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_agvgdweb_results(file = stop("`file` is missing"), alignment = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_agvgdweb_results_+3A_file">file</code></td>
<td>
<p>A file path to the results output by the AGVGD Web app.</p>
</td></tr>
<tr><td><code id="read_agvgdweb_results_+3A_alignment">alignment</code></td>
<td>
<p>A character matrix or an alignment object obtained with
<code><a href="#topic+read_alignment">read_alignment()</a></code>. Rows are expected to be sequences of single characters
(protein residues), and columns the alignment positions. The first row must
be the reference sequence, i.e. the sequence whose substitutions will be
evaluated against. This parameter can be left <code>NULL</code>. If supplied the
column <code>poi</code> in the output will be filled in (default is to be <code>NA</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="tibble.html#topic+tibble-package">tibble</a> of seven columns:
</p>

<dl>
<dt>res</dt><dd><p>Position of the amino acid residue in the reference protein
(first sequence in the alignment). This position corresponds to <code>poi</code> minus
the gaps in the alignment.</p>
</dd>
<dt>poi</dt><dd><p>Position of interest, i.e. the alignment position at which the
amino acid substitution is being assessed. Because this information is not
provided by AGVGD Web app this column is always <code>NA</code>; we keep it though for
coherence with the output of <code>agvgd()</code>.</p>
</dd>
<dt>ref</dt><dd><p>Reference amino acid, i.e. the amino acid in the first sequence
of the alignment, at the position of interest.</p>
</dd>
<dt>sub</dt><dd><p>Amino acid substitution being assessed.</p>
</dd>
<dt>gv</dt><dd><p>Grantham variation score.</p>
</dd>
<dt>gd</dt><dd><p>Grantham difference score.</p>
</dd>
<dt>prediction</dt><dd><p>Predicted effect of the amino acid substitution. This is
classed as C0, C15, C25, C35, C45, C55, or C65, with C65 most likely to
interfere with function and C0 least likely.</p>
</dd>
</dl>


<hr>
<h2 id='read_alignment'>Read a protein sequence multiple alignment</h2><span id='topic+read_alignment'></span>

<h3>Description</h3>

<p>Reads a protein sequence multiple alignment (PSMA) from either a set of
pre-bundled alignments, by gene name, or from a Multi-FASTA file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_alignment(
  gene = c("ATM", "BRCA1", "BRCA2", "CHEK2", "MRE11", "MSH6", "NBN", "PALB2", "PMS2",
    "RAD50", "RAD51", "XRCC2"),
  file = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_alignment_+3A_gene">gene</code></td>
<td>
<p>The gene name for which an alignment is provided with this
package. Use the function <code><a href="#topic+alignment_file">alignment_file()</a></code> to list the pre-bundled
alignments.</p>
</td></tr>
<tr><td><code id="read_alignment_+3A_file">file</code></td>
<td>
<p>The path to a Multi-FASTA file. If this argument is given, it
takes precedence over the <code>gene</code> parameter.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An alignment object; essentially, a character matrix, whose elements
are protein residues in one-letter notation. Rows are sequences and columns
are alignment positions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Read in the alignment for the gene XRCC2
read_alignment('XRCC2')

# Also read in the alignment for the gene XRCC2, but now by specifying
# directly the path to the file.
path &lt;- system.file("extdata", alignment_file("XRCC2"), package = "agvgd")
read_alignment(file = path)

</code></pre>

<hr>
<h2 id='read_substitutions'>Read a file with amino acid substitutions</h2><span id='topic+read_substitutions'></span>

<h3>Description</h3>

<p>This function reads a file with amino acid substitutions. The format of
should be the same one as requested by the web version of
<a href="http://agvgd.hci.utah.edu/help.php">Align-GVGD</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_substitutions(
  file = stop("`file` must be specified"),
  amino_acid_code = c("one_letter", "three_letter")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_substitutions_+3A_file">file</code></td>
<td>
<p>The path to a file with amino acid substitutions.</p>
</td></tr>
<tr><td><code id="read_substitutions_+3A_amino_acid_code">amino_acid_code</code></td>
<td>
<p>The type of symbol used for amino acids in the
returned output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="tibble.html#topic+tibble-package">tibble</a> listing the amino acids
substitutions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># "sub.txt" is an example file containing missense substitutions formatted
# according to the requirements indicated in http://agvgd.hci.utah.edu/help.php.
my_file &lt;- system.file("extdata", "sub.txt", package = "agvgd")
cat(readLines(my_file), sep = "\n")

read_substitutions(file = my_file)

# lee2010_sub.txt is a file containing the missense variants studied by
# Lee et al. (2010): https://doi.org/10.1158/0008-5472.CAN-09-4563.
read_substitutions(file = system.file("extdata", "lee2010_sub.txt", package = "agvgd"))

</code></pre>

<hr>
<h2 id='res_to_poi'>Convert a residue position to an alignment position</h2><span id='topic+res_to_poi'></span>

<h3>Description</h3>

<p>This function converts an residue position to a position in the frame of
the alignment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>res_to_poi(alignment, res)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="res_to_poi_+3A_alignment">alignment</code></td>
<td>
<p>An alignment.</p>
</td></tr>
<tr><td><code id="res_to_poi_+3A_res">res</code></td>
<td>
<p>A residue position.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer vector of alignment positions corresponding to residue position in the reference
sequence.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>align_ATM &lt;- read_alignment('ATM')
align_ATM[, 1:6]

# Convert the positions of the first five residues to alignment positions
res_to_poi(align_ATM, 1:5)

</code></pre>

<hr>
<h2 id='str_chop'>Chop a string into sub-strings of fixed width</h2><span id='topic+str_chop'></span>

<h3>Description</h3>

<p>Chop a string into sub-strings of fixed width
</p>


<h3>Usage</h3>

<pre><code class='language-R'>str_chop(x, width = 50)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="str_chop_+3A_x">x</code></td>
<td>
<p>A single string. This function is not vectorised, so <code>x</code> must be of
length 1.</p>
</td></tr>
<tr><td><code id="str_chop_+3A_width">width</code></td>
<td>
<p>Length of the chopped pieces.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of sub-strings of length <code>width</code>.
</p>

<hr>
<h2 id='write_alignment'>Export an alignment to FASTA</h2><span id='topic+write_alignment'></span>

<h3>Description</h3>

<p>This function takes an alignment and exports it to a FASTA file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_alignment(alignment, file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write_alignment_+3A_alignment">alignment</code></td>
<td>
<p>An alignment. It may be a simple matrix or an object
obtained with <code>read_alignment()</code>.</p>
</td></tr>
<tr><td><code id="write_alignment_+3A_file">file</code></td>
<td>
<p>A file path.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function is run for its side effect of writing a file. But it
returns the file path passed in <code>file</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
alignment &lt;- matrix(
  c('P', 'M', 'I',
    'P', 'I', 'I',
    'P', 'L', 'I'),
  nrow = 3,
  byrow = TRUE
)

# Export an alignment based on a matrix
write_alignment(alignment, "my_alignment.fasta")
cat(readLines("my_alignment.fasta"), sep = "\n")

# Export one of the bundled alignments
write_alignment(read_alignment(gene = 'BRCA1'), "BRCA1.fasta")
cat(readLines("BRCA1.fasta")[1:10], sep = "\n")


</code></pre>

<hr>
<h2 id='write_substitutions'>Generate and export a list of substitutions</h2><span id='topic+write_substitutions'></span>

<h3>Description</h3>

<p>This function exports to a file a list of residue substitutions. The format
used will be  the same one as requested by the web version of
<a href="http://agvgd.hci.utah.edu/help.php">Align-GVGD</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_substitutions(
  file,
  alignment,
  poi,
  sub,
  mode = c("recycle", "expand_grid")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write_substitutions_+3A_file">file</code></td>
<td>
<p>A file path.</p>
</td></tr>
<tr><td><code id="write_substitutions_+3A_alignment">alignment</code></td>
<td>
<p>A character matrix or an alignment object obtained with
<code><a href="#topic+read_alignment">read_alignment()</a></code>. Rows are expected to be sequences of single characters
(protein residues), and columns the alignment positions. The first row must
be the reference sequence, i.e. the sequence whose substitutions will be
evaluated against.</p>
</td></tr>
<tr><td><code id="write_substitutions_+3A_poi">poi</code></td>
<td>
<p>A whole number indicating the position of interest (POI).</p>
</td></tr>
<tr><td><code id="write_substitutions_+3A_sub">sub</code></td>
<td>
<p>A character vector of protein residue substitutions to be
classified. The amino acids must be provided as one-letter symbols.</p>
</td></tr>
<tr><td><code id="write_substitutions_+3A_mode">mode</code></td>
<td>
<p>If both <code>poi</code> and <code>sub</code> contain more than one element, <code>mode</code>
specifies how these two inputs are combined. If <code>mode = 'recycle'</code> the
shortest vector is recycled to match the length of the longest. If <code>mode = 'expand_grid'</code>, all combinations between elements of <code>poi</code> and <code>sub</code> are
combined.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function is run for its side effect of writing a file. But it
returns the file path passed in <code>file</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
write_substitutions(file = "ex01.csv",
                    alignment = read_alignment("ATM"),
                    poi = 20:25,
                    sub = amino_acids())
cat(readLines("ex01.csv"), sep = "\n")

write_substitutions(file = "ex02.csv",
                    alignment = read_alignment("ATM"),
                    poi = 20:21,
                    sub = amino_acids(),
                    mode = 'expand_grid')
cat(readLines("ex02.csv"), sep = "\n")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
