<!DOCTYPE html><html><head><title>Help for package aIc</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {aIc}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#aIc.coherent'><p>Calculate the subcompositional coherence of samples in</p>
a dataset for a given correction.</a></li>
<li><a href='#aIc.dominant'><p><code>aIc.dominant</code> calculates the subcompositional dominance of a sample in</p>
a dataset for a given correction. This compares the distances of samples
of the full dataset and a subset of the dataset.
This is expected to be true if the transform is behaving rationally in
compositional datasets.</a></li>
<li><a href='#aIc.perturb'><p><code>aIc.perturb</code> calculates the perturbation invariance of distance for</p>
samples with a given correction. This compares the distances of samples
of the full dataset and a the perturbed dataset.
This is expected to be true if the transform is behaving rationally in
compositional datasets.</a></li>
<li><a href='#aIc.plot'><p><code>aIc.plot</code> plots the result of the distance tests.</p></a></li>
<li><a href='#aIc.runExample'><p><code>aIc.runExample</code> loads the associated shiny app</p>
This will load the selex example dataset with the default group sizes,
the user can upload their own local dataset and adjust groups accordingly.</a></li>
<li><a href='#aIc.scale'><p><code>aIc.scale</code> calculates the scaling invariance of a sample in</p>
a dataset for a given correction. This compares the distances of samples
of the full dataset and a scaled version of the dataset.
This is expected to be true if the transform is behaving rationally in
compositional datasets.</a></li>
<li><a href='#aIc.singular'><p><code>aIc.singular</code> tests for singular data.</p>
This is expected to be true if the transform is behaving rationally in
compositional datasets and also true in the case of datasets with more
features than samples.</a></li>
<li><a href='#meta16S'><p>16S rRNA tag-sequencing data</p></a></li>
<li><a href='#metaTscome'><p>meta-transcriptome data</p></a></li>
<li><a href='#selex'><p>Selection-based differential sequence variant abundance dataset</p></a></li>
<li><a href='#singleCell'><p>single cell transcriptome data</p></a></li>
<li><a href='#transcriptome'><p>Saccharomyces cerevisiae transcriptome</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Testing for Compositional Pathologies in Datasets</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-09-27</td>
</tr>
<tr>
<td>Author:</td>
<td>Greg Gloor</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Greg Gloor &lt;ggloor@uwo.ca&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A set of tests for compositional pathologies. Tests for coherence of correlations with aIc.coherent() as suggested by (Erb et al. (2020) &lt;<a href="https://doi.org/10.1016%2Fj.acags.2020.100026">doi:10.1016/j.acags.2020.100026</a>&gt;),  compositional dominance of distance with aIc.dominant(), compositional perturbation invariance with aIc.perturb() as suggested by (Aitchison (1992) &lt;<a href="https://doi.org/10.1007%2FBF00891269">doi:10.1007/BF00891269</a>&gt;) and singularity of the covariation matrix with aIc.singular(). Currently tests five data transformations: prop, clr, TMM, TMMwsp, and RLE from the R packages 'ALDEx2', 'edgeR' and 'DESeq2' (Fernandes et al (2014) &lt;<a href="https://doi.org/10.1186%2F2049-2618-2-15">doi:10.1186/2049-2618-2-15</a>&gt;, Anders et al. (2013)&lt;<a href="https://doi.org/10.1038%2Fnprot.2013.099">doi:10.1038/nprot.2013.099</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ggloor/aIc">https://github.com/ggloor/aIc</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ggloor/aIc/issues">https://github.com/ggloor/aIc/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>matrixcalc, zCompositions, shiny, edgeR, ALDEx2, vegan</td>
</tr>
<tr>
<td>Suggests:</td>
<td>BiocStyle, knitr, rmarkdown</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0),</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-10-04 19:55:13 UTC; ggloor</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-10-04 23:40:04 UTC</td>
</tr>
</table>
<hr>
<h2 id='aIc.coherent'>Calculate the subcompositional coherence of samples in
a dataset for a given correction.</h2><span id='topic+aIc.coherent'></span>

<h3>Description</h3>

<p>'aIc.coherent' compares the correlation coefficients
of features in common of the full dataset and a subset of the dataset.
This is expected to be false for all compositional datasets and transforms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aIc.coherent(
  data,
  norm.method = "prop",
  zero.remove = 0.95,
  zero.method = "prior",
  log = FALSE,
  group = NULL,
  cor.test = "spearman"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aIc.coherent_+3A_data">data</code></td>
<td>
<p>can be any dataframe or matrix with samples by column</p>
</td></tr>
<tr><td><code id="aIc.coherent_+3A_norm.method">norm.method</code></td>
<td>
<p>can be prop, clr, RLE, TMM, TMMwsp, lvha, iqlr</p>
</td></tr>
<tr><td><code id="aIc.coherent_+3A_zero.remove">zero.remove</code></td>
<td>
<p>is a value. Filter data to remove features that are 0 
across at least that proportion of samples: default 0.95</p>
</td></tr>
<tr><td><code id="aIc.coherent_+3A_zero.method">zero.method</code></td>
<td>
<p>can be any of NULL, prior, GBM or CZM. NULL will not 
impute or change 0 values, GBM and CZM are from the 
zCompositions R package, and prior will simply add 0.5 to all counts.</p>
</td></tr>
<tr><td><code id="aIc.coherent_+3A_log">log</code></td>
<td>
<p>is a logical. log transform the prop, RLE or TMM outputs, default=FALSE</p>
</td></tr>
<tr><td><code id="aIc.coherent_+3A_group">group</code></td>
<td>
<p>is a vector containing group information. Required for clr, RLE,</p>
</td></tr>
<tr><td><code id="aIc.coherent_+3A_cor.test">cor.test</code></td>
<td>
<p>is either the pearson or spearman method (default)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list with the correlation in <code>cor</code>, a yes/no binary 
decision in <code>is.coherent</code>,  the x and y values for a scatterplot
of the correlations in the full and subcompositions, and the plot and axis
labels in <code>main</code> <code>xlab</code> and <code>ylab</code>.
</p>


<h3>Author(s)</h3>

<p>Greg Gloor
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(selex)
group = c(rep('N', 7), rep('S', 7))
x &lt;- aIc.coherent(selex, group=group, norm.method='clr', zero.method='prior')
plot(x$plot[,1], x$plot[,2], main=x$main, ylab=x$ylab, xlab=x$xlab)

</code></pre>

<hr>
<h2 id='aIc.dominant'><code>aIc.dominant</code> calculates the subcompositional dominance of a sample in
a dataset for a given correction. This compares the distances of samples
of the full dataset and a subset of the dataset.
This is expected to be true if the transform is behaving rationally in
compositional datasets.</h2><span id='topic+aIc.dominant'></span>

<h3>Description</h3>

<p><code>aIc.dominant</code> calculates the subcompositional dominance of a sample in
a dataset for a given correction. This compares the distances of samples
of the full dataset and a subset of the dataset.
This is expected to be true if the transform is behaving rationally in
compositional datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aIc.dominant(
  data,
  norm.method = "prop",
  zero.remove = 0.95,
  zero.method = "prior",
  log = FALSE,
  distance = "euclidian",
  group = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aIc.dominant_+3A_data">data</code></td>
<td>
<p>can be any dataframe or matrix with samples by column</p>
</td></tr>
<tr><td><code id="aIc.dominant_+3A_norm.method">norm.method</code></td>
<td>
<p>can be prop, clr, RLE, TMM, TMMwsp</p>
</td></tr>
<tr><td><code id="aIc.dominant_+3A_zero.remove">zero.remove</code></td>
<td>
<p>is a value. Filter data to remove features that are 0 
across at least that proportion of samples: default 0.95</p>
</td></tr>
<tr><td><code id="aIc.dominant_+3A_zero.method">zero.method</code></td>
<td>
<p>can be any of NULL, prior, GBM or CZM. NULL will not 
impute or change 0 values, GBM (preferred) and CZM are from the 
zCompositions R package, and prior will simply add 0.5 to all counts.</p>
</td></tr>
<tr><td><code id="aIc.dominant_+3A_log">log</code></td>
<td>
<p>is a logical. log transform the RLE or TMM outputs, default=FALSE</p>
</td></tr>
<tr><td><code id="aIc.dominant_+3A_distance">distance</code></td>
<td>
<p>can be euclidian, bray, or jaccard. euclidian on log-ratio
transformed data is the same as the Aitchison distance. default=euclidian</p>
</td></tr>
<tr><td><code id="aIc.dominant_+3A_group">group</code></td>
<td>
<p>is a vector containing group information. Required for clr, RLE, 
TMM, lvha, and iqlr based normalizations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list with the overlap between distances in the full and 
subcompositon in <code>ol</code> (expect 0), a yes/no binary decision in 
<code>is.dominant</code> and the table of distances for the whole and subcomposition
in <code>dist.all</code> and <code>dist.sub</code>, a plot showing a histogram of the resulting
overlap in distances in <code>plot</code>, and the plot and axis
labels in <code>main</code> <code>xlab</code> and <code>ylab</code>
</p>


<h3>Author(s)</h3>

<p>Greg Gloor
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(selex)
group = c(rep('N', 7), rep('S', 7))
x &lt;- aIc.dominant(selex, group=group, norm.method='clr', distance='euclidian', zero.method='prior')
plot(x$plot, main=x$main, ylab=x$ylab, xlab=x$xlab)
</code></pre>

<hr>
<h2 id='aIc.perturb'><code>aIc.perturb</code> calculates the perturbation invariance of distance for
samples with a given correction. This compares the distances of samples
of the full dataset and a the perturbed dataset.
This is expected to be true if the transform is behaving rationally in
compositional datasets.</h2><span id='topic+aIc.perturb'></span>

<h3>Description</h3>

<p><code>aIc.perturb</code> calculates the perturbation invariance of distance for
samples with a given correction. This compares the distances of samples
of the full dataset and a the perturbed dataset.
This is expected to be true if the transform is behaving rationally in
compositional datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aIc.perturb(
  data,
  norm.method = "prop",
  zero.remove = 0.95,
  zero.method = "prior",
  distance = "euclidian",
  log = FALSE,
  group = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aIc.perturb_+3A_data">data</code></td>
<td>
<p>can be any dataframe or matrix with samples by column</p>
</td></tr>
<tr><td><code id="aIc.perturb_+3A_norm.method">norm.method</code></td>
<td>
<p>can be prop, clr, RLE, TMM, TMMwsp</p>
</td></tr>
<tr><td><code id="aIc.perturb_+3A_zero.remove">zero.remove</code></td>
<td>
<p>is a value. Filter data to remove features that are 0 
across at least that proportion of samples: default 0.95</p>
</td></tr>
<tr><td><code id="aIc.perturb_+3A_zero.method">zero.method</code></td>
<td>
<p>can be any of NULL, prior, GBM or CZM. NULL will not 
impute or change 0 values, GBM (preferred) and CZM are from the 
zCompositions R package, and prior will simply add 0.5 to all counts.</p>
</td></tr>
<tr><td><code id="aIc.perturb_+3A_distance">distance</code></td>
<td>
<p>can be euclidian, bray, or jaccard. euclidian on log-ratio
transformed data is the same as the Aitchison distance. default=euclidian</p>
</td></tr>
<tr><td><code id="aIc.perturb_+3A_log">log</code></td>
<td>
<p>is a logical. log transform the RLE or TMM outputs, default=FALSE</p>
</td></tr>
<tr><td><code id="aIc.perturb_+3A_group">group</code></td>
<td>
<p>is a vector containing group information. Required for clr, RLE, 
TMM, lvha, and iqlr based normalizations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list with the maximum proportional perturbation in <code>ol</code> 
(expect 0, but values up to 1
<code>is.perturb</code>, the table of distances for the whole and perturbaton
in <code>dist.all</code> and <code>dist.perturb</code>, the histogram of the
perturbations in <code>plot</code>, and the plot and axis
labels in <code>main</code> <code>xlab</code> and <code>ylab</code>. .
</p>


<h3>Author(s)</h3>

<p>Greg Gloor
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(selex)
group = c(rep('N', 7), rep('S', 7))
x &lt;- aIc.perturb(selex, group=group, norm.method='clr', distance='euclidian', zero.method='prior')
plot(x$plot, main=x$main, ylab=x$ylab, xlab=x$xlab)
</code></pre>

<hr>
<h2 id='aIc.plot'><code>aIc.plot</code> plots the result of the distance tests.</h2><span id='topic+aIc.plot'></span>

<h3>Description</h3>

<p><code>aIc.plot</code> plots the result of the distance tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aIc.plot(test.out)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aIc.plot_+3A_test.out">test.out</code></td>
<td>
<p>is the output from either aIc.dominant, aIc.scale, aIc.perturb</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a plot of the density of the distance test results.
test result.
</p>


<h3>Author(s)</h3>

<p>Greg Gloor
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(selex)
group = c(rep('N', 7), rep('S', 7))
test.out &lt;- aIc.dominant(selex, norm.method='prop', group=group)
aIc.plot(test.out)
</code></pre>

<hr>
<h2 id='aIc.runExample'><code>aIc.runExample</code> loads the associated shiny app
This will load the selex example dataset with the default group sizes,
the user can upload their own local dataset and adjust groups accordingly.</h2><span id='topic+aIc.runExample'></span>

<h3>Description</h3>

<p><code>aIc.runExample</code> loads the associated shiny app
This will load the selex example dataset with the default group sizes,
the user can upload their own local dataset and adjust groups accordingly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aIc.runExample()
</code></pre>


<h3>Value</h3>

<p>No return value, but instead opens a shiny connection to your 
default web browser with the selex dataset as an example.
</p>


<h3>Author(s)</h3>

<p>Greg Gloor
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(aIc)
aIc.runExample()

</code></pre>

<hr>
<h2 id='aIc.scale'><code>aIc.scale</code> calculates the scaling invariance of a sample in
a dataset for a given correction. This compares the distances of samples
of the full dataset and a scaled version of the dataset.
This is expected to be true if the transform is behaving rationally in
compositional datasets.</h2><span id='topic+aIc.scale'></span>

<h3>Description</h3>

<p><code>aIc.scale</code> calculates the scaling invariance of a sample in
a dataset for a given correction. This compares the distances of samples
of the full dataset and a scaled version of the dataset.
This is expected to be true if the transform is behaving rationally in
compositional datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aIc.scale(
  data,
  norm.method = "prop",
  zero.remove = 0.95,
  zero.method = "prior",
  distance = "euclidian",
  log = FALSE,
  group = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aIc.scale_+3A_data">data</code></td>
<td>
<p>can be any dataframe or matrix with samples by column</p>
</td></tr>
<tr><td><code id="aIc.scale_+3A_norm.method">norm.method</code></td>
<td>
<p>can be prop, clr, iqlr, lvha, RLE, TMM, TMMwsp</p>
</td></tr>
<tr><td><code id="aIc.scale_+3A_zero.remove">zero.remove</code></td>
<td>
<p>is a value. Filter data to remove features that are 0 
across at least that proportion of samples: default 0.95</p>
</td></tr>
<tr><td><code id="aIc.scale_+3A_zero.method">zero.method</code></td>
<td>
<p>can be any of NULL, prior, GBM or CZM. NULL will not 
impute or change 0 values, GBM (preferred) and CZM are from the 
zCompositions R package, and prior will simply add 0.5 to all counts.</p>
</td></tr>
<tr><td><code id="aIc.scale_+3A_distance">distance</code></td>
<td>
<p>can be euclidian, bray, or jaccard. euclidian on log-ratio
transformed data is the same as the Aitchison distance. default=euclidian</p>
</td></tr>
<tr><td><code id="aIc.scale_+3A_log">log</code></td>
<td>
<p>is a logical. log transform the RLE or TMM outputs, default=FALSE</p>
</td></tr>
<tr><td><code id="aIc.scale_+3A_group">group</code></td>
<td>
<p>is a vector containing group information. Required for clr, RLE, 
TMM, lvha, and iqlr based normalizations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list with the overlap between distances in the full and 
scaled composition in <code>ol</code> (expect 0), a yes/no binary decision in 
<code>is.scale</code> and the table of distances for the whole and scaled composition
in <code>dist.all</code> and <code>dist.scale</code>, a plot showing a histogram of the resulting
overlap in distances in <code>plot</code>, and the plot and axis
labels in <code>main</code> <code>xlab</code> and <code>ylab</code>
</p>


<h3>Author(s)</h3>

<p>Greg Gloor
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(selex)
group = c(rep('N', 7), rep('S', 7))
x &lt;- aIc.scale(selex, group=group, norm.method='clr', zero.method='prior')
plot(x$plot, main=x$main, ylab=x$ylab, xlab=x$xlab)
</code></pre>

<hr>
<h2 id='aIc.singular'><code>aIc.singular</code> tests for singular data. 
This is expected to be true if the transform is behaving rationally in
compositional datasets and also true in the case of datasets with more 
features than samples.</h2><span id='topic+aIc.singular'></span>

<h3>Description</h3>

<p><code>aIc.singular</code> tests for singular data. 
This is expected to be true if the transform is behaving rationally in
compositional datasets and also true in the case of datasets with more 
features than samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aIc.singular(
  data,
  norm.method = "prop",
  zero.remove = 0.95,
  zero.method = "prior",
  log = FALSE,
  group = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aIc.singular_+3A_data">data</code></td>
<td>
<p>can be any dataframe or matrix with samples by column</p>
</td></tr>
<tr><td><code id="aIc.singular_+3A_norm.method">norm.method</code></td>
<td>
<p>can be prop, clr, RLE, TMM, TMMwsp</p>
</td></tr>
<tr><td><code id="aIc.singular_+3A_zero.remove">zero.remove</code></td>
<td>
<p>is a value. Filter data to remove features that are 0 
across at least that proportion of samples: default 0.95</p>
</td></tr>
<tr><td><code id="aIc.singular_+3A_zero.method">zero.method</code></td>
<td>
<p>can be any of NULL, prior, GBM or CZM. NULL will not 
impute or change 0 values, GBM (preferred) and CZM are from the 
zCompositions R package, and prior will simply add 0.5 to all counts.</p>
</td></tr>
<tr><td><code id="aIc.singular_+3A_log">log</code></td>
<td>
<p>is a logical. log transform the RLE or TMM outputs, default=FALSE</p>
</td></tr>
<tr><td><code id="aIc.singular_+3A_group">group</code></td>
<td>
<p>is a vector containing group information. Required for clr, RLE, 
TMM, lvha, and iqlr based normalizations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list with a yes/no binary decision in 
<code>is.singular</code> and the covariance matrix in <code>cov.matrix</code>
</p>


<h3>Author(s)</h3>

<p>Greg Gloor
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(selex)
group = c(rep('N', 7), rep('S', 7))
x &lt;- aIc.singular(selex, group=group, norm.method='clr', zero.method='prior')
</code></pre>

<hr>
<h2 id='meta16S'>16S rRNA tag-sequencing data</h2><span id='topic+meta16S'></span>

<h3>Description</h3>

<p>A count table of a 16S rRNA amplicon data
Two groups, pupils and centenarians are represented with 198
and 161 samples per group respectively. samples are by column  
and OTU ids are by row.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(meta16S)
</code></pre>


<h3>Format</h3>

<p>A data frame with 359 columns and 860 rows
</p>


<h3>Source</h3>

<p>doi: 10.1128/mSphere.00327-17
</p>

<hr>
<h2 id='metaTscome'>meta-transcriptome data</h2><span id='topic+metaTscome'></span>

<h3>Description</h3>

<p>A count table of a mixed population or metatranscriptome 
experiment. Two groups, H and BV are represented with 7 and 10
samples per group respectively. samples are by column and 
functions are by row.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(metaTscome)
</code></pre>


<h3>Format</h3>

<p>A data frame with 17 columns and 3647 rows
</p>


<h3>Source</h3>

<p>doi:10.1007/978-3-030-71175-7_17 and doi:10.1007/978-1-4939-8728-3_13
</p>

<hr>
<h2 id='selex'>Selection-based differential sequence variant abundance dataset</h2><span id='topic+selex'></span>

<h3>Description</h3>

<p>This data set gives the differential abundance of 1600 enzyme variants
grown under selective (NS) and selective (S) conditions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(selex)
</code></pre>


<h3>Format</h3>

<p>A data frame with 14 columns and 1600 rows
</p>


<h3>Source</h3>

<p>DOI:10.1073/pnas.1322352111
</p>

<hr>
<h2 id='singleCell'>single cell transcriptome data</h2><span id='topic+singleCell'></span>

<h3>Description</h3>

<p>A count table of a single cell transcriptome data subset from 
the count table from  doi:10.1038/s41592-019-0372-4. Two groups
memory T cells, and cytotoxic T cells, 1000 cells per group.
samples are by column and genes are by row.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(singleCell)
</code></pre>


<h3>Format</h3>

<p>A data frame with 2000 columns and 1508 rows
</p>


<h3>Source</h3>

<p>https://www.nature.com/articles/s41592-019-0372-4
</p>

<hr>
<h2 id='transcriptome'>Saccharomyces cerevisiae transcriptome</h2><span id='topic+transcriptome'></span>

<h3>Description</h3>

<p>A count table of a highly replicated RNA-seq experiment
with samples by column and genes by row. Two groups composed
of SNF2 knockout and WT, 48 samples in each.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(transcriptome)
</code></pre>


<h3>Format</h3>

<p>A data frame with 96 columns and 5892 rows
</p>


<h3>Source</h3>

<p>DOI: 10.1261/rna.053959.115 and PRJEB5348
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
