<!DOCTYPE html><html><head><title>Help for package aLFQ</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {aLFQ}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AbsoluteQuantification'><p>Absolute label-free quantification of mass spectrometry proteomics experiments</p></a></li>
<li><a href='#ALF'><p>Generate ALF report</p></a></li>
<li><a href='#aLFQ-package'><p>aLFQ</p></a></li>
<li><a href='#APEX'><p>Training, testing and validation of APEX peptide observability models</p></a></li>
<li><a href='#apexFeatures'><p>Calculation of physicochemical amino acid properties for APEX</p></a></li>
<li><a href='#APEXMS'><p>Calculating absolute and relative protein abundance from mass spectrometry-based protein expression data.</p></a></li>
<li><a href='#import'><p>import of mass spectrometry proteomics data analysis software reports.</p></a></li>
<li><a href='#LUDWIGMS'><p>Estimation of Absolute Protein Quantities of Unlabeled Samples by Selected Reaction Monitoring Mass Spectrometry..</p></a></li>
<li><a href='#PeptideInference'><p>Peptide inference for aLFQ import data frame</p></a></li>
<li><a href='#ProteinInference'><p>Protein inference for aLFQ import data frame</p></a></li>
<li><a href='#proteotypic'><p>Prediction of the flyability of proteotypic peptides</p></a></li>
<li><a href='#UPS2MS'><p>Calculating absolute and relative protein abundance from mass spectrometry-based protein expression data.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Estimating Absolute Protein Quantities from Label-Free LC-MS/MS
Proteomics Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.6</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-01-07</td>
</tr>
<tr>
<td>Author:</td>
<td>George Rosenberger, Hannes Roest, Christina Ludwig, Ruedi Aebersold, Lars Malmstroem</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>George Rosenberger &lt;gr2578@cumc.columbia.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.15.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, plyr, caret, seqinr, lattice, randomForest, ROCR,
reshape2, bio3d</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>Description:</td>
<td>Determination of absolute protein quantities is necessary for multiple applications, such as mechanistic modeling of biological systems. Quantitative liquid chromatography tandem mass spectrometry (LC-MS/MS) proteomics can measure relative protein abundance on a system-wide scale. To estimate absolute quantitative information using these relative abundance measurements requires additional information such as heavy-labeled references of known concentration. Multiple methods have been using different references and strategies; some are easily available whereas others require more effort on the users end. Hence, we believe the field might benefit from making some of these methods available under an automated framework, which also facilitates validation of the chosen strategy. We have implemented the most commonly used absolute label-free protein abundance estimation methods for LC-MS/MS modes quantifying on either MS1-, MS2-levels or spectral counts together with validation algorithms to enable automated data analysis and error estimation. Specifically, we used Monte-carlo cross-validation and bootstrapping for model selection and imputation of proteome-wide absolute protein quantity estimation. Our open-source software is written in the statistical programming language R and validated and demonstrated on a synthetic sample. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/aLFQ">https://github.com/aLFQ</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-01-07 18:02:59 UTC; root</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-01-08 11:10:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='AbsoluteQuantification'>Absolute label-free quantification of mass spectrometry proteomics experiments</h2><span id='topic+AbsoluteQuantification'></span><span id='topic+AbsoluteQuantification.default'></span><span id='topic+predict.AbsoluteQuantification'></span><span id='topic+cval'></span><span id='topic+cval.default'></span><span id='topic+cval.AbsoluteQuantification'></span><span id='topic+print.AbsoluteQuantification'></span><span id='topic+plot.AbsoluteQuantification'></span><span id='topic+hist.AbsoluteQuantification'></span><span id='topic+pivot'></span><span id='topic+pivot.default'></span><span id='topic+pivot.AbsoluteQuantification'></span><span id='topic+export'></span><span id='topic+export.default'></span><span id='topic+export.AbsoluteQuantification'></span>

<h3>Description</h3>

<p>Absolute label-free quantification of mass spectrometry proteomics experiments.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
AbsoluteQuantification(data, total_protein_concentration = 1, ...)
## S3 method for class 'AbsoluteQuantification'
cval(object, cval_method = "mc", mcx = 1000, ...)
## S3 method for class 'AbsoluteQuantification'
print(x, ...)
## S3 method for class 'AbsoluteQuantification'
plot(x, ...)
## S3 method for class 'AbsoluteQuantification'
hist(x, ...)
## S3 method for class 'AbsoluteQuantification'
pivot(x, ...)
## S3 method for class 'AbsoluteQuantification'
export(x, file, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AbsoluteQuantification_+3A_data">data</code></td>
<td>
<p>a mandatory data frame containing the columns <code>"run_id"</code>, <code>"protein_id"</code>, <code>"response"</code>, and <code>"concentration"</code> as generated by ProteinInference. The id column can be defined in any format, while the <code>"response"</code> and <code>"concentration"</code> columns need to be numeric and in non-log form. The data may contain calibration data (with numeric <code>"concentration"</code> and test data (with <code>"concentration"</code> = &quot;?&quot;))</p>
</td></tr>
<tr><td><code id="AbsoluteQuantification_+3A_total_protein_concentration">total_protein_concentration</code></td>
<td>
<p>the total protein concentration in the sample in any unit. This will be used for the normalized protein and concentration columns.</p>
</td></tr>
<tr><td><code id="AbsoluteQuantification_+3A_object">object</code></td>
<td>
<p>an <code>AbsoluteQuantification</code> object.</p>
</td></tr>
<tr><td><code id="AbsoluteQuantification_+3A_cval_method">cval_method</code></td>
<td>
<p>a method for doing crossvalidation: <code>"boot"</code> (bootstrapping), <code>"mc"</code> (monte carlo cross-validation), <code>"loo"</code> (leaving-one-out).</p>
</td></tr>
<tr><td><code id="AbsoluteQuantification_+3A_mcx">mcx</code></td>
<td>
<p>a positive integer value of the number of folds for cross-validation.</p>
</td></tr>
<tr><td><code id="AbsoluteQuantification_+3A_file">file</code></td>
<td>
<p>the location of the output csv file.</p>
</td></tr>
<tr><td><code id="AbsoluteQuantification_+3A_x">x</code></td>
<td>
<p>an <code>AbsoluteQuantification</code> object.</p>
</td></tr>
<tr><td><code id="AbsoluteQuantification_+3A_...">...</code></td>
<td>
<p>future extensions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If absolute quantity estimation based on anchor peptides or proteins is demanded, the calibration peptide or protein abundance must be provided. Both estimated calibration protein intensities and separately determined calibration protein concentrations are log transformed and a first order linear least-squares regression of this log-log data is calculated. The abundance of the target proteins is predicted based on this regression. The error of the regression arises from biological and technical variation as well from the protein and peptide intensity estimators. To perform model selection and to estimate the error of the predicted protein concentrations, bootstrapping and Monte Carlo cross-validation as suggested (Malmstrom et al., 2009; Ludwig et al., 2012) were implemented. For both methods, the objective function is the minimization of the mean fold-error. 
</p>
<p>If, on the other hand, the total protein concentration per cell is supplied in proteome-wide experiments, the absolute protein concentrations are estimated by normalization of the MS intensities or spectral counts to this number (Lu et al., 2006).
</p>


<h3>Value</h3>

<p>An object of class <code>AbsoluteQuantification</code>.
</p>


<h3>Author(s)</h3>

<p>George Rosenberger <a href="mailto:gr2578@cumc.columbia.edu">gr2578@cumc.columbia.edu</a></p>


<h3>References</h3>

<p>Malmstrom, J. et al. <em>Proteome-wide cellular protein concentrations of the human pathogen Leptospira interrogans.</em> Nature 460, 762-765 (2009).
</p>
<p>Ludwig, C., Claassen, M., Schmidt, A. &amp; Aebersold, R. <em>Estimation of Absolute Protein Quantities of Unlabeled Samples by Selected Reaction Monitoring Mass Spectrometry.</em> Molecular &amp; Cellular Proteomics 11, M111.013987-M111.013987 (2012).
</p>
<p>Lu, P., Vogel, C., Wang, R., Yao, X. &amp; Marcotte, E. M. <em>Absolute protein expression profiling estimates the relative contributions of transcriptional and translational regulation.</em> Nat Biotech 25, 117-124 (2006).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+import">import</a></code>, <code><a href="#topic+ProteinInference">ProteinInference</a></code>, <code><a href="#topic+ALF">ALF</a></code>, <code><a href="#topic+APEX">APEX</a></code>, <code><a href="#topic+apexFeatures">apexFeatures</a></code>, <code><a href="#topic+proteotypic">proteotypic</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(UPS2MS)

UPS2_SRM&lt;-head(UPS2_SRM,100) # Remove this line for real applications
data_PI &lt;- ProteinInference(UPS2_SRM)
data_AQ &lt;- predict(cval(AbsoluteQuantification(data_PI),mcx=2))
print(data_AQ)
plot(data_AQ)
hist(data_AQ)
pivot(data_AQ)
</code></pre>

<hr>
<h2 id='ALF'>Generate ALF report</h2><span id='topic+ALF'></span><span id='topic+ALF.default'></span>

<h3>Description</h3>

<p>Estimation of Absolute Protein Quantities of Unlabeled Samples by Targeted Mass Spectrometry.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
ALF(data, report_filename="ALF_report.pdf", 
prediction_filename="ALF_prediction.csv", peptide_methods = c("top"), 
peptide_topx = c(1,2,3), peptide_strictness = "loose", 
peptide_summary = "mean", transition_topx = c(1,2,3), 
transition_strictness = "loose", transition_summary = "sum", fasta = NA, 
apex_model = NA, combine_precursors = FALSE, combine_peptide_sequences = FALSE, 
consensus_proteins = TRUE, consensus_peptides = TRUE, consensus_transitions = TRUE,
cval_method = "boot", cval_mcx = 1000, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ALF_+3A_data">data</code></td>
<td>
<p>a mandatory data frame containing the columns <code>"run_id"</code>, <code>"protein_id"</code>, <code>"peptide_id"</code>, <code>"peptide_sequence"</code>, <code>"precursor_charge"</code>, <code>"peptide_intensity"</code> and <code>"concentration"</code> are required. For quantification on the transition level, the columns <code>"protein_id"</code>, <code>"peptide_id"</code>, <code>"transition_id"</code>, <code>"peptide_sequence"</code>, <code>"precursor_charge"</code>, <code>"transition_intensity"</code> and <code>"concentration"</code> are required. The id columns can be defined in any format, while the <code>"_intensity"</code> and <code>"concentration"</code> columns need to be numeric and in non-log form. The data may contain calibration data (with numeric <code>"concentration"</code> and test data (with <code>"concentration"</code> = &quot;?&quot;))</p>
</td></tr>
<tr><td><code id="ALF_+3A_report_filename">report_filename</code></td>
<td>
<p>the path and filename of the PDF report.</p>
</td></tr>
<tr><td><code id="ALF_+3A_prediction_filename">prediction_filename</code></td>
<td>
<p>the path and filename of the predictions of the optimal model.</p>
</td></tr>
<tr><td><code id="ALF_+3A_peptide_methods">peptide_methods</code></td>
<td>
<p>a vecter containing a combination of <code>"top"</code>, <code>"all"</code>, <code>"iBAQ"</code>, <code>"APEX"</code> or <code>"NSAF"</code> peptide to protein intensity estimation methods.</p>
</td></tr>
<tr><td><code id="ALF_+3A_peptide_topx">peptide_topx</code></td>
<td>
<p>(<code>"top"</code> only:) a positive integer value of the top x peptides to consider for <code>"top"</code> methods.</p>
</td></tr>
<tr><td><code id="ALF_+3A_peptide_strictness">peptide_strictness</code></td>
<td>
<p>(<code>"top"</code> only:) whether <code>peptide_topx</code> should only consider proteins with the minimal peptide number (<code>"strict"</code>) or all (<code>"loose"</code>).</p>
</td></tr>
<tr><td><code id="ALF_+3A_peptide_summary">peptide_summary</code></td>
<td>
<p>(<code>"top"</code> and <code>"all"</code> only:) how to summarize the peptide intensities: <code>"mean"</code>, <code>"median"</code>, <code>"sum"</code>.</p>
</td></tr>
<tr><td><code id="ALF_+3A_transition_topx">transition_topx</code></td>
<td>
<p>a positive integer value of the top x transitions to consider for transition to peptide intensity estimation methods.</p>
</td></tr>
<tr><td><code id="ALF_+3A_transition_strictness">transition_strictness</code></td>
<td>
<p>whether <code>transition_topx</code> should only consider peptides with the minimal transition number (<code>"strict"</code>) or all (<code>"loose"</code>).</p>
</td></tr>
<tr><td><code id="ALF_+3A_transition_summary">transition_summary</code></td>
<td>
<p>how to summarize the transition intensities: <code>"mean"</code>, <code>"median"</code>, <code>"sum"</code>.</p>
</td></tr>
<tr><td><code id="ALF_+3A_fasta">fasta</code></td>
<td>
<p>(<code>"iBAQ"</code>, <code>"APEX"</code> and <code>"NSAF"</code> only:) the path and filename to an amino acid fasta file containing the proteins of interest.</p>
</td></tr>
<tr><td><code id="ALF_+3A_apex_model">apex_model</code></td>
<td>
<p>(<code>"APEX"</code> only:) The <code>"APEX"</code> model to use (see APEX).</p>
</td></tr>
<tr><td><code id="ALF_+3A_combine_precursors">combine_precursors</code></td>
<td>
<p>whether to sum all precursors of the same peptide.</p>
</td></tr>
<tr><td><code id="ALF_+3A_combine_peptide_sequences">combine_peptide_sequences</code></td>
<td>
<p>whether to sum all variants (modifications) of the same peptide sequence.</p>
</td></tr>
<tr><td><code id="ALF_+3A_consensus_proteins">consensus_proteins</code></td>
<td>
<p>if multiple runs are provided, select identical proteins among all runs.</p>
</td></tr>
<tr><td><code id="ALF_+3A_consensus_peptides">consensus_peptides</code></td>
<td>
<p>if multiple runs are provided, select identical peptides among all runs.</p>
</td></tr>
<tr><td><code id="ALF_+3A_consensus_transitions">consensus_transitions</code></td>
<td>
<p>if multiple runs are provided, select identical transitions among all runs.</p>
</td></tr>
<tr><td><code id="ALF_+3A_cval_method">cval_method</code></td>
<td>
<p>a method for doing crossvalidation: <code>"boot"</code> (bootstrapping), <code>"mc"</code> (monte carlo cross-validation), <code>"loo"</code> (leaving-one-out).</p>
</td></tr>
<tr><td><code id="ALF_+3A_cval_mcx">cval_mcx</code></td>
<td>
<p>a positive integer value of the number of folds for cross-validation.</p>
</td></tr>
<tr><td><code id="ALF_+3A_...">...</code></td>
<td>
<p>future extensions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The ALF module enables model selection for TopN transitions and peptides for protein quantification (Ludwig et al., 2012). The workflow is completely automated and a report and prediction (using the best model) is generated.
</p>


<h3>Value</h3>

<p>The reports specified in the function call.
</p>


<h3>Author(s)</h3>

<p>George Rosenberger <a href="mailto:gr2578@cumc.columbia.edu">gr2578@cumc.columbia.edu</a></p>


<h3>References</h3>

<p>Ludwig, C., Claassen, M., Schmidt, A. \&amp; Aebersold, R. <em>Estimation of Absolute Protein Quantities of Unlabeled Samples by Selected Reaction Monitoring Mass Spectrometry.</em> Molecular \&amp; Cellular Proteomics 11, M111.013987-M111.013987 (2012).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+import">import</a></code>, <code><a href="#topic+ProteinInference">ProteinInference</a></code>, <code><a href="#topic+AbsoluteQuantification">AbsoluteQuantification</a></code>, <code><a href="#topic+APEX">APEX</a></code>, <code><a href="#topic+apexFeatures">apexFeatures</a></code>, <code><a href="#topic+proteotypic">proteotypic</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(UPS2MS)

## Not run: ALF(UPS2_SRM)

## Not run: data(LUDWIGMS)

## Not run: ALF(LUDWIG_SRM)
</code></pre>

<hr>
<h2 id='aLFQ-package'>aLFQ</h2><span id='topic+aLFQ-package'></span><span id='topic+aLFQ'></span>

<h3>Description</h3>

<p>Estimating Absolute Protein Quantities from Label-Free LC-MS/MS Proteomics Data</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> aLFQ</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.3.5</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2019-04-30</td>
</tr>
<tr>
 <td style="text-align: left;">
Author: </td><td style="text-align: left;"> George Rosenberger, Hannes Roest, Christina Ludwig, Ruedi Aebersold, Lars Malmstroem</td>
</tr>
<tr>
 <td style="text-align: left;">
Maintainer: </td><td style="text-align: left;"> George Rosenberger &lt;gr2578@cumc.columbia.edu&gt;</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> R (&gt;= 2.15.0)</td>
</tr>
<tr>
 <td style="text-align: left;">
Imports: </td><td style="text-align: left;"> data.table, plyr, caret, seqinr, lattice, randomForest, ROCR, reshape2, protiq, bio3d</td>
</tr>
<tr>
 <td style="text-align: left;">
Suggests: </td><td style="text-align: left;"> testthat</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL version 3 or newer</td>
</tr>
<tr>
 <td style="text-align: left;">
URL: </td><td style="text-align: left;"> <a href="https://github.com/aLFQ">https://github.com/aLFQ</a></td>
</tr>
<tr>
 <td style="text-align: left;">
NeedsCompilation: </td><td style="text-align: left;"> no</td>
</tr>
<tr>
 <td style="text-align: left;">
Repository: </td><td style="text-align: left;"> CRAN</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Determination of absolute protein quantities is necessary for multiple applications, such as mechanistic modeling of biological systems. Quantitative liquid chromatography tandem mass spectrometry (LC-MS/MS) proteomics can measure relative protein abundance on a system-wide scale. To estimate absolute quantitative information using these relative abundance measurements requires additional information such as heavy-labeled references of known concentration. Multiple methods have been using different references and strategies; some are easily available whereas others require more effort on the users end. Hence, we believe the field might benefit from making some of these methods available under an automated framework, which also facilitates validation of the chosen strategy. We have implemented the most commonly used absolute label-free protein abundance estimation methods for LC-MS/MS modes quantifying on either MS1-, MS2-levels or spectral counts together with validation algorithms to enable automated data analysis and error estimation. Specifically, we used Monte-carlo cross-validation and bootstrapping for model selection and imputation of proteome-wide absolute protein quantity estimation. Our open-source software is written in the statistical programming language R and validated and demonstrated on a synthetic sample. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+import">import</a></code>, <code><a href="#topic+ProteinInference">ProteinInference</a></code>, <code><a href="#topic+AbsoluteQuantification">AbsoluteQuantification</a></code>, <code><a href="#topic+ALF">ALF</a></code>, <code><a href="#topic+APEX">APEX</a></code>, <code><a href="#topic+apexFeatures">apexFeatures</a></code>, <code><a href="#topic+proteotypic">proteotypic</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: help(package="aLFQ")
</code></pre>

<hr>
<h2 id='APEX'>Training, testing and validation of APEX peptide observability models</h2><span id='topic+APEX'></span><span id='topic+APEX.default'></span><span id='topic+predict.APEX'></span><span id='topic+cval.APEX'></span><span id='topic+print.APEX'></span><span id='topic+plot.APEX'></span>

<h3>Description</h3>

<p>Calculating absolute and relative protein abundance from mass spectrometry-based protein expression data.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
APEX(data, ...)
## S3 method for class 'APEX'
predict(object, newdata=NULL, ...)
## S3 method for class 'APEX'
cval(object, folds=10, ...)
## S3 method for class 'APEX'
print(x, ...)
## S3 method for class 'APEX'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="APEX_+3A_data">data</code></td>
<td>
<p>an R object of type <code>"apexFeatures"</code>.</p>
</td></tr>
<tr><td><code id="APEX_+3A_object">object</code></td>
<td>
<p>an <code>APEX</code> object.</p>
</td></tr>
<tr><td><code id="APEX_+3A_newdata">newdata</code></td>
<td>
<p>an R object of type <code>"apexFeatures"</code>.</p>
</td></tr>
<tr><td><code id="APEX_+3A_folds">folds</code></td>
<td>
<p>a positive integer value of the number of folds for cross-validation.</p>
</td></tr>
<tr><td><code id="APEX_+3A_x">x</code></td>
<td>
<p>an <code>APEX</code> object.</p>
</td></tr>
<tr><td><code id="APEX_+3A_...">...</code></td>
<td>
<p>future extensions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The APEX module is a reimplementation of the original algorithm (Lu et al., 2006; Vogel et al., 2008) using the randomForest package. It requires <code><a href="#topic+apexFeatures">apexFeatures</a></code> input objects and reports the results in an <code>APEX</code> object, which can be used by the <code><a href="#topic+ProteinInference">ProteinInference</a></code> module for protein quantification.
</p>


<h3>Value</h3>

<p>An object of class <code>APEX</code>.
</p>


<h3>Author(s)</h3>

<p>George Rosenberger <a href="mailto:gr2578@cumc.columbia.edu">gr2578@cumc.columbia.edu</a></p>


<h3>References</h3>

<p>Lu, P., Vogel, C., Wang, R., Yao, X. &amp; Marcotte, E. M. <em>Absolute protein expression profiling estimates the relative contributions of transcriptional and translational regulation.</em> Nat Biotech 25, 117-124 (2006).
</p>
<p>Vogel, C. &amp; Marcotte, E. M. <em>Calculating absolute and relative protein abundance from mass spectrometry-based protein expression data.</em> Nat Protoc 3, 1444-1451 (2008).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+import">import</a></code>, <code><a href="#topic+ProteinInference">ProteinInference</a></code>, <code><a href="#topic+AbsoluteQuantification">AbsoluteQuantification</a></code>, <code><a href="#topic+ALF">ALF</a></code>, <code><a href="#topic+apexFeatures">apexFeatures</a></code>, <code><a href="#topic+proteotypic">proteotypic</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(131)

data(APEXMS)

APEX_ORBI&lt;-head(APEX_ORBI,50) # Remove this line for real applications
APEX_ORBI.af &lt;- apexFeatures(APEX_ORBI)
APEX_ORBI.apex &lt;- APEX(data=APEX_ORBI.af)
print(APEX_ORBI.apex)

APEX_ORBI_cval.apex &lt;- cval(APEX_ORBI.apex, folds=2)
plot(APEX_ORBI_cval.apex)
</code></pre>

<hr>
<h2 id='apexFeatures'>Calculation of physicochemical amino acid properties for APEX</h2><span id='topic+apexFeatures'></span><span id='topic+apexFeatures.default'></span><span id='topic+print.apexFeatures'></span>

<h3>Description</h3>

<p>Calculation of physicochemical amino acid properties for APEX.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
apexFeatures(x, ...)
## S3 method for class 'apexFeatures'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="apexFeatures_+3A_x">x</code></td>
<td>
<p>a mandatory data frame containing the variables in the model.
The data frame requires the columns <code>"peptide_sequence"</code>, <code>"apex"</code>. The data may 
contain training data (with boolean <code>"apex"</code> and test data (with <code>"apex"</code>=<code>NA</code>))</p>
</td></tr>
<tr><td><code id="apexFeatures_+3A_...">...</code></td>
<td>
<p>future extensions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The apexFeatures function computes the APEX or PeptideSieve features (Mallick et al., 2006; Vogel et al., 2008) based on AAindex (Kawashima et al., 2008) and returns them in an apexFeatures object for further usage in the <code><a href="#topic+APEX">APEX</a></code> module.
</p>


<h3>Value</h3>

<p>An object of class <code>apexFeatures</code>.
</p>


<h3>Author(s)</h3>

<p>George Rosenberger <a href="mailto:gr2578@cumc.columbia.edu">gr2578@cumc.columbia.edu</a></p>


<h3>References</h3>

<p>Kawashima, S. et al. <em>AAindex: amino acid index database, progress report 2008.</em> Nucleic Acids Research 36, D202-5 (2008).
</p>
<p>Mallick, P. et al. <em>Computational prediction of proteotypic peptides for quantitative proteomics.</em> Nat Biotech 25, 125-131 (2006).
</p>
<p>Vogel, C. &amp; Marcotte, E. M. <em>Calculating absolute and relative protein abundance from mass spectrometry-based protein expression data.</em> Nat Protoc 3, 1444-1451 (2008).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+import">import</a></code>, <code><a href="#topic+ProteinInference">ProteinInference</a></code>, <code><a href="#topic+AbsoluteQuantification">AbsoluteQuantification</a></code>, <code><a href="#topic+ALF">ALF</a></code>, <code><a href="#topic+APEX">APEX</a></code>, <code><a href="#topic+proteotypic">proteotypic</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(APEXMS)

# APEX_ORBI
APEX_ORBI&lt;-head(APEX_ORBI,20) # Remove this line for real applications
APEX_ORBI.af &lt;- apexFeatures(APEX_ORBI)
print(APEX_ORBI.af)

# APEX_LCQ
APEX_LCQ&lt;-head(APEX_LCQ,20) # Remove this line for real applications
APEX_LCQ.af &lt;- apexFeatures(APEX_LCQ)
print(APEX_LCQ.af)
</code></pre>

<hr>
<h2 id='APEXMS'>Calculating absolute and relative protein abundance from mass spectrometry-based protein expression data.</h2><span id='topic+APEXMS'></span><span id='topic+APEX_ORBI'></span><span id='topic+APEX_LCQ'></span>

<h3>Description</h3>

<p>This dataset contains training ThermoFinnigan LTQ-OrbiTrap (ORBI) and ThermoFinnigan Surveyor/DecaXP+ iontrap (LCQ) data for training of an <code><a href="#topic+APEX">APEX</a></code> classifier. The dataset was generated by Christine Vogel and Edward M. Marcotte (see references).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(APEXMS)</code></pre>


<h3>Format</h3>

<p>A data.frame with the following components:
</p>

<ol>
<li><p>peptide_sequence character vector: Peptide sequence. 
</p>
</li>
<li><p>apex character vector: observed in experiment: 0 = no; 1 = yes. 
</p>
</li></ol>



<h3>Source</h3>

<p>The dataset was obtained from:<br />
<a href="http://marcottelab.org/APEX_Protocol/">http://marcottelab.org/APEX_Protocol/</a>.
</p>


<h3>References</h3>

<p>Vogel, C. &amp; Marcotte, E. M. <em>Calculating absolute and relative protein abundance from mass spectrometry-based protein expression data.</em> Nat Protoc 3, 1444-1451 (2008).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+import">import</a></code>, <code><a href="#topic+ProteinInference">ProteinInference</a></code>, <code><a href="#topic+AbsoluteQuantification">AbsoluteQuantification</a></code>, <code><a href="#topic+ALF">ALF</a></code>, <code><a href="#topic+APEX">APEX</a></code>, <code><a href="#topic+apexFeatures">apexFeatures</a></code>, <code><a href="#topic+proteotypic">proteotypic</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(APEXMS)
</code></pre>

<hr>
<h2 id='import'>import of mass spectrometry proteomics data analysis software reports.</h2><span id='topic+import'></span><span id='topic+import.default'></span>

<h3>Description</h3>

<p>import of mass spectrometry proteomics data analysis software reports.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
import(ms_filenames, ms_filetype, concentration_filename=NA,
averageruns=FALSE, sumruns=FALSE, mprophet_cutoff=0.01,
openswath_superimpose_identifications=FALSE, openswath_replace_run_id=FALSE,
openswath_filtertop=FALSE, openswath_removedecoys=TRUE,
peptideprophet_cutoff=0.95, abacus_column="ADJNSAF", pepxml2csv_runsplit="~",
...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_+3A_ms_filenames">ms_filenames</code></td>
<td>
<p>the paths and filenames of files to import in a character or array class.</p>
</td></tr>
<tr><td><code id="import_+3A_ms_filetype">ms_filetype</code></td>
<td>
<p>one of <code>"openswath"</code>, <code>"mprophet"</code>, <code>"openmslfq"</code>, <code>"skyline"</code>, <code>"abacus"</code> or
<code>"pepxml2csv"</code> filetypes. Multiple files of the same type can be supplied in a vector.</p>
</td></tr>
<tr><td><code id="import_+3A_concentration_filename">concentration_filename</code></td>
<td>
<p>the filename of a csv with concentrations (in any unit). Needs to have 
the columns <code>"peptide_id"</code> (or <code>"protein_id"</code>) and <code>"concentration"</code>.</p>
</td></tr>
<tr><td><code id="import_+3A_averageruns">averageruns</code></td>
<td>
<p>whether different MS runs should be averaged.</p>
</td></tr>
<tr><td><code id="import_+3A_sumruns">sumruns</code></td>
<td>
<p>whether different MS runs should be summed.</p>
</td></tr>
<tr><td><code id="import_+3A_mprophet_cutoff">mprophet_cutoff</code></td>
<td>
<p>(openswath and mprophet only:) the FDR cutoff (m_score) for OpenSWATH and mProphet reports.</p>
</td></tr>
<tr><td><code id="import_+3A_openswath_superimpose_identifications">openswath_superimpose_identifications</code></td>
<td>
<p>(openswath only:) enables propagation of identification among several runs if feature alignment or requantification was conducted.</p>
</td></tr>
<tr><td><code id="import_+3A_openswath_replace_run_id">openswath_replace_run_id</code></td>
<td>
<p>whether the run_id of the MS data should be replaced by the filename.</p>
</td></tr>
<tr><td><code id="import_+3A_openswath_filtertop">openswath_filtertop</code></td>
<td>
<p>(openswath only:) whether only the top peakgroup should be considered.</p>
</td></tr>
<tr><td><code id="import_+3A_openswath_removedecoys">openswath_removedecoys</code></td>
<td>
<p>(openswath only:) whether decoys should be removed.</p>
</td></tr>
<tr><td><code id="import_+3A_peptideprophet_cutoff">peptideprophet_cutoff</code></td>
<td>
<p>(abacus and pepxml2csv only:) the PeptideProphet probability cutoff for Abacus reports.</p>
</td></tr>
<tr><td><code id="import_+3A_abacus_column">abacus_column</code></td>
<td>
<p>(abacus only:) target score: one of <code>"NUMSPECSTOT"</code>,<code>"TOTNSAF"</code>,<code>"NUMSPECSUNIQ"</code>,
<code>"UNIQNSAF"</code>,<code>"NUMSPECSADJ"</code> or <code>"ADJNSAF"</code>.</p>
</td></tr>
<tr><td><code id="import_+3A_pepxml2csv_runsplit">pepxml2csv_runsplit</code></td>
<td>
<p>(pepxml2csv only:) the separator of the run_id and spectrum_id column.</p>
</td></tr>
<tr><td><code id="import_+3A_...">...</code></td>
<td>
<p>future extensions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The import function provides unified access to the results of various standard proteomic quantification applications like OpenSWATH (Roest et al., 2014), mProphet (Reiter et al., 2011), OpenMS (Sturm et al., 2008; Weisser et al., 2013),Skyline (MacLean et al., 2010) and Abacus (Fermin et al., 2011). This enables generic application of all further steps using the same data structure and enables extension to support other data formats. If multiple runs, i.e. replicates, are supplied, the averaged or summed values can be used to summarize the experimental data. In addition to the input from the analysis software, an input table with the anchor peptides or proteins and sample specific absolute abundance, or an estimate of the total protein concentration in the sample is required. The endpoint of this step is a unified input data structure.
</p>


<h3>Value</h3>

<p>A standard aLFQ import data frame, either on transition, peptide (precursor) or protein level.
</p>


<h3>Author(s)</h3>

<p>George Rosenberger <a href="mailto:gr2578@cumc.columbia.edu">gr2578@cumc.columbia.edu</a></p>


<h3>References</h3>

<p>Roest H. L. et al. <em>A tool for the automated, targeted analysis of data-independent acquisition (DIA) MS-data: OpenSWATH.</em> Nat Biotech, in press.
</p>
<p>Reiter, L. et al. <em>mProphet: automated data processing and statistical validation for large-scale SRM experiments.</em> Nat Meth 8, 430-435 (2011).
</p>
<p>Sturm, M. et al.  <em>OpenMS - An open-source software framework for mass spectrometry.</em> BMC Bioinformatics 9, 163 (2008).
</p>
<p>Weisser, H. et al. <em>An automated pipeline for high-throughput label-free quantitative proteomics.</em> J. Proteome Res. 130208071745007 (2013). doi:10.1021/pr300992u
</p>
<p>MacLean, B. et al. <em>Skyline: an open source document editor for creating and analyzing targeted proteomics experiments.</em> Bioinformatics 26, 966-968 (2010).
</p>
<p>Fermin, D., Basrur, V., Yocum, A. K. &amp; Nesvizhskii, A. I. <em>Abacus: A computational tool for extracting and pre-processing spectral count data for label-free quantitative proteomic analysis.</em> PROTEOMICS 11, 1340-1345 (2011).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ProteinInference">ProteinInference</a></code>, <code><a href="#topic+AbsoluteQuantification">AbsoluteQuantification</a></code>, <code><a href="#topic+ALF">ALF</a></code>, <code><a href="#topic+APEX">APEX</a></code>, <code><a href="#topic+apexFeatures">apexFeatures</a></code>, <code><a href="#topic+proteotypic">proteotypic</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>import(ms_filenames = system.file("extdata","example_openswath.txt",package="aLFQ"),
ms_filetype = "openswath", concentration_filename=NA,
averageruns=FALSE, sumruns=FALSE, mprophet_cutoff=0.01,
openswath_superimpose_identifications=FALSE, openswath_replace_run_id=FALSE,
openswath_filtertop=FALSE, openswath_removedecoys=TRUE)

import(ms_filenames = system.file("extdata","example_mprophet.txt",package="aLFQ"),
ms_filetype = "mprophet",
concentration_filename = system.file("extdata","example_concentration_peptide.csv",
package="aLFQ"), averageruns=FALSE, sumruns=FALSE, mprophet_cutoff=0.01)

import(ms_filenames = system.file("extdata","example_openmslfq.csv",package="aLFQ"),
ms_filetype = "openmslfq", concentration_filename=NA, averageruns=FALSE, sumruns=FALSE)

import(ms_filenames = system.file("extdata","example_skyline.csv",package="aLFQ"),
ms_filetype = "skyline",
concentration_filename =
system.file("extdata","example_concentration_protein.csv",package="aLFQ"),
averageruns=FALSE, sumruns=FALSE)

import(ms_filenames = system.file("extdata","example_abacus_protein.txt",package="aLFQ"),
ms_filetype = "abacus", concentration_filename =
system.file("extdata","example_concentration_protein.csv",
package="aLFQ"), averageruns=FALSE, sumruns=FALSE,
peptideprophet_cutoff=0.95, abacus_column="ADJNSAF")
</code></pre>

<hr>
<h2 id='LUDWIGMS'>Estimation of Absolute Protein Quantities of Unlabeled Samples by Selected Reaction Monitoring Mass Spectrometry..</h2><span id='topic+LUDWIGMS'></span><span id='topic+LUDWIG_SRM'></span>

<h3>Description</h3>

<p>This dataset contains the Leptospira interrogans MS data from Ludwig C., et al. (2012).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(LUDWIGMS)</code></pre>


<h3>Format</h3>

<p>The data structure for LUDWIG_SRM represents a data.frame containing the following column header: <code>"run_id"</code> (freetext), <code>"protein_id"</code> (freetext), <code>"peptide_id"</code> (freetext), <code>"transition_id"</code> (freetext), <code>"peptide_sequence"</code> (unmodified, natural amino acid sequence in 1-letter nomenclature), <code>"precursor_charge"</code> (positive integer value), <code>"transition_intensity"</code> (positive non-logarithm floating value) and <code>"concentration"</code> (calibration: positive non-logarithm floating value, prediction: &quot;?&quot;).
</p>


<h3>References</h3>

<p>Ludwig, C., Claassen, M., Schmidt, A. &amp; Aebersold, R. <em>Estimation of Absolute Protein Quantities of Unlabeled Samples by Selected Reaction Monitoring Mass Spectrometry.</em> Molecular &amp; Cellular Proteomics 11, M111.013987-M111.013987 (2012).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+import">import</a></code>, <code><a href="#topic+ProteinInference">ProteinInference</a></code>, <code><a href="#topic+AbsoluteQuantification">AbsoluteQuantification</a></code>, <code><a href="#topic+ALF">ALF</a></code>, <code><a href="#topic+APEX">APEX</a></code>, <code><a href="#topic+apexFeatures">apexFeatures</a></code>, <code><a href="#topic+proteotypic">proteotypic</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(LUDWIGMS)
</code></pre>

<hr>
<h2 id='PeptideInference'>Peptide inference for aLFQ import data frame</h2><span id='topic+PeptideInference'></span><span id='topic+PeptideInference.default'></span>

<h3>Description</h3>

<p>Peptide inference for aLFQ import data frame.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
PeptideInference(data, transition_topx = 3,
transition_strictness = "strict",transition_summary = "sum",
consensus_proteins = TRUE, consensus_transitions = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PeptideInference_+3A_data">data</code></td>
<td>
<p>a mandatory data frame containing the <code>"protein_id"</code>, <code>"peptide_id"</code>, <code>"transition_id"</code>, <code>"peptide_sequence"</code>, <code>"precursor_charge"</code>, <code>"transition_intensity"</code> and <code>"concentration"</code> are required. The id columns can be defined in any format, while the <code>"_intensity"</code> and <code>"concentration"</code> columns need to be numeric and in non-log form. The data may contain calibration data (with numeric <code>"concentration"</code> and test data (with <code>"concentration"</code> = &quot;?&quot;))</p>
</td></tr>
<tr><td><code id="PeptideInference_+3A_transition_topx">transition_topx</code></td>
<td>
<p>a positive integer value of the top x transitions to consider for transition to peptide intensity estimation methods.</p>
</td></tr>
<tr><td><code id="PeptideInference_+3A_transition_strictness">transition_strictness</code></td>
<td>
<p>whether <code>transition_topx</code> should only consider peptides with the minimal transition number (<code>"strict"</code>) or all (<code>"loose"</code>).</p>
</td></tr>
<tr><td><code id="PeptideInference_+3A_transition_summary">transition_summary</code></td>
<td>
<p>how to summarize the transition intensities: <code>"mean"</code>, <code>"median"</code>, <code>"sum"</code>.</p>
</td></tr>
<tr><td><code id="PeptideInference_+3A_consensus_proteins">consensus_proteins</code></td>
<td>
<p>if multiple runs are provided, select identical proteins among all runs.</p>
</td></tr>
<tr><td><code id="PeptideInference_+3A_consensus_transitions">consensus_transitions</code></td>
<td>
<p>if multiple runs are provided, select identical transitions among all runs.</p>
</td></tr>
<tr><td><code id="PeptideInference_+3A_...">...</code></td>
<td>
<p>future extensions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The PeptideInference module provides functionality to infer peptide / precursor quantities from the measured precursor or fragment intensities or peptide spectral counts.
</p>


<h3>Value</h3>

<p>A standard aLFQ import data frame on peptide / precursor level.
</p>


<h3>Author(s)</h3>

<p>George Rosenberger <a href="mailto:gr2578@cumc.columbia.edu">gr2578@cumc.columbia.edu</a></p>


<h3>References</h3>

<p>Ludwig, C., Claassen, M., Schmidt, A. &amp; Aebersold, R. <em>Estimation of Absolute Protein Quantities of Unlabeled Samples by Selected Reaction Monitoring Mass Spectrometry.</em> Molecular &amp; Cellular Proteomics 11, M111.013987-M111.013987 (2012).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+import">import</a></code>, <code><a href="#topic+AbsoluteQuantification">AbsoluteQuantification</a></code>, <code><a href="#topic+ALF">ALF</a></code>, <code><a href="#topic+APEX">APEX</a></code>, <code><a href="#topic+apexFeatures">apexFeatures</a></code>, <code><a href="#topic+proteotypic">proteotypic</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(UPS2MS)

data_PI &lt;- PeptideInference(UPS2_SRM)
print(data_PI)
</code></pre>

<hr>
<h2 id='ProteinInference'>Protein inference for aLFQ import data frame</h2><span id='topic+ProteinInference'></span><span id='topic+ProteinInference.default'></span>

<h3>Description</h3>

<p>Protein inference for aLFQ import data frame.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
ProteinInference(data, peptide_method = "top", peptide_topx = 2, 
peptide_strictness = "strict",peptide_summary = "mean", transition_topx = 3, 
transition_strictness = "strict",transition_summary = "sum", fasta = NA, 
apex_model = NA, combine_precursors = FALSE, combine_peptide_sequences = FALSE, 
consensus_proteins = TRUE, consensus_peptides = TRUE, 
consensus_transitions = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ProteinInference_+3A_data">data</code></td>
<td>
<p>a mandatory data frame containing the columns <code>"run_id"</code>, <code>"protein_id"</code>, <code>"protein_intensity"</code>, and <code>"concentration"</code> for quantification on the protein level. For quantification on the peptide level, the columns <code>"run_id"</code>, <code>"protein_id"</code>, <code>"peptide_id"</code>, <code>"peptide_sequence"</code>, <code>"precursor_charge"</code>, <code>"peptide_intensity"</code> and <code>"concentration"</code> are required. For quantification on the transition level, the columns <code>"protein_id"</code>, <code>"peptide_id"</code>, <code>"transition_id"</code>, <code>"peptide_sequence"</code>, <code>"precursor_charge"</code>, <code>"transition_intensity"</code> and <code>"concentration"</code> are required. The id columns can be defined in any format, while the <code>"_intensity"</code> and <code>"concentration"</code> columns need to be numeric and in non-log form. The data may contain calibration data (with numeric <code>"concentration"</code> and test data (with <code>"concentration"</code> = &quot;?&quot;))</p>
</td></tr>
<tr><td><code id="ProteinInference_+3A_peptide_method">peptide_method</code></td>
<td>
<p>one of <code>"top"</code>, <code>"all"</code>, <code>"iBAQ"</code>, <code>"APEX"</code> or <code>"NSAF"</code> peptide to protein intensity estimation methods.</p>
</td></tr>
<tr><td><code id="ProteinInference_+3A_peptide_topx">peptide_topx</code></td>
<td>
<p>(<code>"top"</code> only:) a positive integer value of the top x peptides to consider for <code>"top"</code> methods.</p>
</td></tr>
<tr><td><code id="ProteinInference_+3A_peptide_strictness">peptide_strictness</code></td>
<td>
<p>(<code>"top"</code> only:) whether <code>peptide_topx</code> should only consider proteins with the minimal peptide number (<code>"strict"</code>) or all (<code>"loose"</code>).</p>
</td></tr>
<tr><td><code id="ProteinInference_+3A_peptide_summary">peptide_summary</code></td>
<td>
<p>(<code>"top"</code> and <code>"all"</code> only:) how to summarize the peptide intensities: <code>"mean"</code>, <code>"median"</code>, <code>"sum"</code>.</p>
</td></tr>
<tr><td><code id="ProteinInference_+3A_transition_topx">transition_topx</code></td>
<td>
<p>a positive integer value of the top x transitions to consider for transition to peptide intensity estimation methods.</p>
</td></tr>
<tr><td><code id="ProteinInference_+3A_transition_strictness">transition_strictness</code></td>
<td>
<p>whether <code>transition_topx</code> should only consider peptides with the minimal transition number (<code>"strict"</code>) or all (<code>"loose"</code>).</p>
</td></tr>
<tr><td><code id="ProteinInference_+3A_transition_summary">transition_summary</code></td>
<td>
<p>how to summarize the transition intensities: <code>"mean"</code>, <code>"median"</code>, <code>"sum"</code>.</p>
</td></tr>
<tr><td><code id="ProteinInference_+3A_fasta">fasta</code></td>
<td>
<p>(<code>"iBAQ"</code>, <code>"APEX"</code> and <code>"NSAF"</code>:) the path and filename to an amino acid fasta file containing the proteins of interest.</p>
</td></tr>
<tr><td><code id="ProteinInference_+3A_apex_model">apex_model</code></td>
<td>
<p>(<code>"APEX"</code> only:) The <code>"APEX"</code> model to use (see APEX).</p>
</td></tr>
<tr><td><code id="ProteinInference_+3A_combine_precursors">combine_precursors</code></td>
<td>
<p>whether to sum all precursors of the same peptide.</p>
</td></tr>
<tr><td><code id="ProteinInference_+3A_combine_peptide_sequences">combine_peptide_sequences</code></td>
<td>
<p>whether to sum all variants (modifications) of the same peptide sequence.</p>
</td></tr>
<tr><td><code id="ProteinInference_+3A_consensus_proteins">consensus_proteins</code></td>
<td>
<p>if multiple runs are provided, select identical proteins among all runs.</p>
</td></tr>
<tr><td><code id="ProteinInference_+3A_consensus_peptides">consensus_peptides</code></td>
<td>
<p>if multiple runs are provided, select identical peptides among all runs.</p>
</td></tr>
<tr><td><code id="ProteinInference_+3A_consensus_transitions">consensus_transitions</code></td>
<td>
<p>if multiple runs are provided, select identical transitions among all runs.</p>
</td></tr>
<tr><td><code id="ProteinInference_+3A_...">...</code></td>
<td>
<p>future extensions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The ProteinInference module provides functionality to infer protein quantities from the measured precursor or fragment intensities or peptide spectral counts. If the dataset contains targeted MS2-level data, the paired precursor and fragment ion signals, the transitions, are first summarized to the precursor level. Different methods for aggregation can be specified, e.g. sum, mean or median and a limit for the selection of the most intense transitions can be provided. It is further possible to exclude precursors, which do not have sufficient transitions to fulfill this boundary. To summarize precursor intensities or spectral counts to theoretical protein intensities, the mean, TopN (Silva et al., 2006; Malmstrom et al., 2009; Schmidt et al., 2011; Ludwig et al., 2012), APEX (Lu et al., 2006), iBAQ (Schwanhausser et al., 2011) and NSAF (Zybailov et al., 2006) protein intensity estimators are provided. For APEX, iBAQ and NSAF, the protein database in FASTA format needs to be supplied. In terms of targeted data acquisition, for both APEX and iBAQ methods all peptides of a protein must be targeted. The results are reported in the same unified data structure as from the previous step
</p>


<h3>Value</h3>

<p>A standard aLFQ import data frame on protein level.
</p>


<h3>Author(s)</h3>

<p>George Rosenberger <a href="mailto:gr2578@cumc.columbia.edu">gr2578@cumc.columbia.edu</a></p>


<h3>References</h3>

<p>Silva, J. C., Gorenstein, M. V., Li, G.-Z., Vissers, Johannes P. C. &amp; Geromanos, S. J. <em>Absolute quantification of proteins by LCMSE: a virtue of parallel MS acquisition.</em> Mol. Cell Proteomics 5, 144-156 (2006).
</p>
<p>Malmstrom, J. et al. <em>Proteome-wide cellular protein concentrations of the human pathogen Leptospira interrogans.</em> Nature 460, 762-765 (2009).
</p>
<p>Schmidt, A. et al. <em>Absolute quantification of microbial proteomes at different states by directed mass spectrometry.</em> Molecular Systems Biology 7, 1-16 (2011).
</p>
<p>Ludwig, C., Claassen, M., Schmidt, A. &amp; Aebersold, R. <em>Estimation of Absolute Protein Quantities of Unlabeled Samples by Selected Reaction Monitoring Mass Spectrometry.</em> Molecular &amp; Cellular Proteomics 11, M111.013987-M111.013987 (2012).
</p>
<p>Lu, P., Vogel, C., Wang, R., Yao, X. &amp; Marcotte, E. M. <em>Absolute protein expression profiling estimates the relative contributions of transcriptional and translational regulation.</em> Nat Biotech 25, 117-124 (2006).
</p>
<p>Schwanhausser, B. et al. <em>Global quantification of mammalian gene expression control.</em> Nature 473, 337-342 (2011).
</p>
<p>Zybailov, B. et al. <em>Statistical Analysis of Membrane Proteome Expression Changes in Saccharomyces c erevisiae.</em> J. Proteome Res. 5, 2339-2347 (2006).
</p>
<p>Gerster S., Kwon T., Ludwig C., Matondo M., Vogel C., Marcotte E. M., Aebersold R., Buhlmann P. <em>Statistical approach to protein quantification.</em> Molecular &amp; Cellular Proteomics 13, M112.02445 (2014).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+import">import</a></code>, <code><a href="#topic+AbsoluteQuantification">AbsoluteQuantification</a></code>, <code><a href="#topic+ALF">ALF</a></code>, <code><a href="#topic+APEX">APEX</a></code>, <code><a href="#topic+apexFeatures">apexFeatures</a></code>, <code><a href="#topic+proteotypic">proteotypic</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(UPS2MS)

data_ProteinInference &lt;- ProteinInference(UPS2_SRM)
print(data_ProteinInference)
</code></pre>

<hr>
<h2 id='proteotypic'>Prediction of the flyability of proteotypic peptides</h2><span id='topic+proteotypic'></span><span id='topic+proteotypic.default'></span>

<h3>Description</h3>

<p>Prediction of the flyability of proteotypic peptides.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
proteotypic(fasta, apex_model, min_aa=4 , max_aa=20, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="proteotypic_+3A_fasta">fasta</code></td>
<td>
<p>a amino acid FASTA file.</p>
</td></tr>
<tr><td><code id="proteotypic_+3A_apex_model">apex_model</code></td>
<td>
<p>an <code>APEX</code> object.</p>
</td></tr>
<tr><td><code id="proteotypic_+3A_min_aa">min_aa</code></td>
<td>
<p>the minimum number of amino acids for proteotypic peptides.</p>
</td></tr>
<tr><td><code id="proteotypic_+3A_max_aa">max_aa</code></td>
<td>
<p>the maximum number of amino acids for proteotypic peptides.</p>
</td></tr>
<tr><td><code id="proteotypic_+3A_...">...</code></td>
<td>
<p>future extensions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function provides prediction of the &quot;flyability&quot; of proteotypic peptides using the APEX method (Lu et al., 2006; Vogel et al., 2008). The APEX scores are probabilities that indicate detectability of the peptide amino acid sequence in LC-MS/MS experiments.
</p>


<h3>Value</h3>

<p>A data.frame containing peptide sequences and associated APEX scores.
</p>


<h3>Author(s)</h3>

<p>George Rosenberger <a href="mailto:gr2578@cumc.columbia.edu">gr2578@cumc.columbia.edu</a></p>


<h3>References</h3>

<p>Lu, P., Vogel, C., Wang, R., Yao, X. &amp; Marcotte, E. M. <em>Absolute protein expression profiling estimates the relative contributions of transcriptional and translational regulation.</em> Nat Biotech 25, 117-124 (2006).
</p>
<p>Vogel, C. &amp; Marcotte, E. M. <em>Calculating absolute and relative protein abundance from mass spectrometry-based protein expression data.</em> Nat Protoc 3, 1444-1451 (2008).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+import">import</a></code>, <code><a href="#topic+ProteinInference">ProteinInference</a></code>, <code><a href="#topic+AbsoluteQuantification">AbsoluteQuantification</a></code>, <code><a href="#topic+ALF">ALF</a></code>, <code><a href="#topic+APEX">APEX</a></code>, <code><a href="#topic+apexFeatures">apexFeatures</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(131)

data(APEXMS)

APEX_ORBI&lt;-head(APEX_ORBI,20) # Remove this line for real applications
APEX_ORBI.af &lt;- apexFeatures(APEX_ORBI)
APEX_ORBI.apex &lt;- APEX(data=APEX_ORBI.af)

peptides &lt;- proteotypic(fasta=system.file("extdata","example.fasta",package="aLFQ"),
apex_model=APEX_ORBI.apex, min_aa=4 , max_aa=20)
## Not run: print(peptides)
</code></pre>

<hr>
<h2 id='UPS2MS'>Calculating absolute and relative protein abundance from mass spectrometry-based protein expression data.</h2><span id='topic+UPS2MS'></span><span id='topic+UPS2'></span><span id='topic+UPS2_SC'></span><span id='topic+UPS2_LFQ'></span><span id='topic+UPS2_SRM'></span>

<h3>Description</h3>

<p>We assessed the performance of aLFQ and the different quantification estimation methods it supports by investigating a commercially available synthetic sample. The Universal Proteomic Standard 2 (UPS2) consists of 48 proteins spanning a dynamic range of five orders of magnitude in bins of eight proteins. The sample was measured in a complex background consisting of Mycobacterium bovis BCG total cell lysate in shotgun and targeted MS modes. Three datasets are available: UPS2_SC (spectral counts), UPS2_LFQ (MS1 intensity), UPS2_SRM (MS2 intensity).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(UPS2MS)</code></pre>


<h3>Format</h3>

<p>The data structure for UPS2_SRM represents a data.frame containing the following column header: <code>"run_id"</code> (freetext), <code>"protein_id"</code> (freetext), <code>"peptide_id"</code> (freetext), <code>"transition_id"</code> (freetext), <code>"peptide_sequence"</code> (unmodified, natural amino acid sequence in 1-letter nomenclature), <code>"precursor_charge"</code> (positive integer value), <code>"transition_intensity"</code> (positive non-logarithm floating value) and <code>"concentration"</code> (calibration: positive non-logarithm floating value, prediction: &quot;?&quot;).
</p>
<p>The data structure for UPS2_LFQ (MS1-level intensity) / UPS2_SC (spectral counts) represents a data.frame containing the columns <code>"run_id"</code> (freetext), <code>"protein_id"</code> (freetext), <code>"peptide_id"</code> (freetext), <code>"peptide_sequence"</code> (unmodified, natural amino acid sequence in 1-letter nomenclature), <code>"precursor_charge"</code> (positive integer value), <code>"peptide_intensity"</code> (positive non-logarithm floating value) and <code>"concentration"</code> (calibration: positive non-logarithm floating value, prediction: &quot;?&quot;). It should be noted, that the spectral count value is also represented by <code>"peptide_intensity"</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+import">import</a></code>, <code><a href="#topic+ProteinInference">ProteinInference</a></code>, <code><a href="#topic+AbsoluteQuantification">AbsoluteQuantification</a></code>, <code><a href="#topic+ALF">ALF</a></code>, <code><a href="#topic+APEX">APEX</a></code>, <code><a href="#topic+apexFeatures">apexFeatures</a></code>, <code><a href="#topic+proteotypic">proteotypic</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(UPS2MS)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
