<!DOCTYPE html><html><head><title>Help for package AlleleShift</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {AlleleShift}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#amova.rda'><p>Analysis of Molecular Variance (AMOVA) via Redundancy Analysis</p></a></li>
<li><a href='#count.model'><p>Calibration and Prediction Functions for Changes in Allele Frequencies</p></a></li>
<li><a href='#Poptri.genind'><p>Example data sets for Populus trichocarpa.</p></a></li>
<li><a href='#population.shift'><p>Shifts of Populations in Environmental Space as Response to Climate Change</p></a></li>
<li><a href='#shift.dot.ggplot'><p>Shifts of Allele Frequencies as Response to Climate Change</p></a></li>
<li><a href='#shift.moon.ggplot'><p>Shifts of Allele Frequencies as Response to Climate Change</p></a></li>
<li><a href='#shift.pie.ggplot'><p>Shifts of Allele Frequencies as Response to Climate Change</p></a></li>
<li><a href='#shift.surf.ggplot'><p>Shifts of Allele Frequencies as Response to Climate Change</p></a></li>
<li><a href='#shift.waffle.ggplot'><p>Shifts of Allele Frequencies as Response to Climate Change</p></a></li>
<li><a href='#VIF.subset'><p>Reduce the Climate Data based on Variance Inflation Factor Analysis.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Predict and Visualize Population-Level Changes in Allele
Frequencies in Response to Climate Change</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1-2</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-10-28</td>
</tr>
<tr>
<td>Author:</td>
<td>Roeland Kindt <a href="https://orcid.org/0000-0002-7672-0712"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [cre, aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Roeland Kindt &lt;RoelandCEKindt@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Methods (&lt;<a href="https://doi.org/10.7717%2Fpeerj.11534">doi:10.7717/peerj.11534</a>&gt;) are provided of calibrating and predicting shifts in allele frequencies through redundancy analysis ('vegan::rda()') and generalized additive models ('mgcv::gam()'). Visualization functions for predicted changes in allele frequencies include 'shift.dot.ggplot()', 'shift.pie.ggplot()', 'shift.moon.ggplot()', 'shift.waffle.ggplot()' and 'shift.surf.ggplot()' that are made with input data sets that are prepared by helper functions for each visualization method. Examples in the documentation show how to prepare animated climate change graphics through a time series with the 'gganimate' package. Function 'amova.rda()' shows how Analysis of Molecular Variance can be directly conducted with the results from redundancy analysis.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>vegan (&ge; 2.6-4), BiodiversityR (&ge; 2.15-4), adegenet</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>poppr, mgcv, dplyr, ggplot2, GGally, ggforce, ggrepel, ggsci,
gggibbous, gganimate, patchwork</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-28 16:17:49 UTC; RKINDT</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-28 21:50:12 UTC</td>
</tr>
</table>
<hr>
<h2 id='amova.rda'>Analysis of Molecular Variance (AMOVA) via Redundancy Analysis</h2><span id='topic+amova.rda'></span>

<h3>Description</h3>

<p>The main objective of the function is to illustrate how information on 'mean squares' can be extracted from redundancy analysis results. For balanced data sets, the final 'Phi' statistics are expected to be the same as those obtained with <code><a href="poppr.html#topic+poppr.amova">poppr.amova</a></code>. The function can only handle 1-level (population) and 2-level (eg, region/population or population/subpopulation) hierarchies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
amova.rda(x, x.data)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="amova.rda_+3A_x">x</code></td>
<td>
<p> Result of redundancy analysis fitted via <code><a href="vegan.html#topic+rda">rda</a></code>. </p>
</td></tr>
<tr><td><code id="amova.rda_+3A_x.data">x.data</code></td>
<td>
<p> Data used to fit the redundancy analysis via <code><a href="BiodiversityR.html#topic+ensemble.VIF.dataframe">ensemble.VIF.dataframe</a></code>. </p>
</td></tr> 
</table>


<h3>Details</h3>

<p>A similar analysis of the relationship between AMOVA and RDA is given by Kindt (2020), also exploring relationships with the Multivariate Analysis of Variance Using Distance Matrices methods that are available via <code><a href="vegan.html#topic+adonis2">adonis2</a></code>. For discussions on AMOVA and its relationship with the matrix of squared Euclidean distances between individuals, see Michalakis and Excoffier (1996), Peakall et al. (1995) or Meirmans and Liu (2018) (the last article also shows how AMOVA can be expanded beyond haploid and diploid organisms). These authors provide the coefficients (equations 3, 4a and 4b) that need to be used with unbalanced numbers of individuals in populations and/or groups, which could then be used to obtain the exact AMOVA statistics from the estimated Mean Squares (there is no practical point in doing this as AMOVA is available via <code><a href="poppr.html#topic+poppr.amova">poppr.amova</a></code>).
</p>
<p>I have also cross-checked the results with GenAlEx 6.5 (Peakall and Smouse 2012), using the export function of <code><a href="poppr.html#topic+genind2genalex">genind2genalex</a></code> to cross-check that RDA obtains the correct Sums-of-Squares and Mean-Squares. Important to note here is that the values in GenAlEx are obtained from only the upper part of the distance matrix, hence these are exactly 50 percent of the Sums-of-Squares from RDA.
</p>
<p>Note that for diploid organisms, the option here is not to include the hierarchical level within individuals.
</p>


<h3>Value</h3>

<p>The function proceeds with Analysis of Molecular Variance from data generated by redundancy analysis (RDA).
</p>


<h3>Author(s)</h3>

<p>Roeland Kindt (World Agroforestry, CIFOR-ICRAF)</p>


<h3>References</h3>

<p>Kindt R. 2020. Analysis of Molecular Variance (AMOVA) with vegan and BiodiversityR, including a graphical method to identify potential migrants. <a href="https://rpubs.com/Roeland-KINDT">https://rpubs.com/Roeland-KINDT</a>
</p>
<p>Meirmans PG and Liu S. 2018. Analysis of Molecular Variance (AMOVA) for Autopolyploids. Front. Ecol. Evol., 23 May 2018. <a href="https://doi.org/10.3389/fevo.2018.00066">doi:10.3389/fevo.2018.00066</a>
</p>
<p>Michalakis Y and Excoffier L. 1996. A Generic Estimation of Population Subdivision Using Distances Between Alleles With Special Reference for Microsatellite Loci. Genetics 142: 1061-1064.
</p>
<p>Peakall R, Smouse PE and Huff DR. 1995. Evolutionary implications of allozyme and RAPD variation in diploid populations of dioecious buffalograss (Buchloe dactyloides (Nutt.) Engelm.). <a href="https://doi.org/10.1111/j.1365-294X.1995.tb00203.x">doi:10.1111/j.1365-294X.1995.tb00203.x</a>.
</p>
<p>Peakall R and Smouse PE. 2012. GenAlEx 6.5: genetic analysis in Excel. Population genetic software for teaching and research an update. <a href="https://doi.org/10.1093/bioinformatics/bts460">doi:10.1093/bioinformatics/bts460</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 

library(poppr)
library(BiodiversityR) # also loads vegan

# Example with 1 level
data(nancycats)
names(alleles(nancycats)) &lt;- locNames(nancycats) # small bug in this data set
nancycats2 &lt;- missingno(nancycats, type = "loci", cutoff=0)
nancy.dist &lt;- vegdist(nancycats2@tab, method="euc")

# Same method as in GenAlEx via the squared Euclidean distance
nancy.dist.amova &lt;- poppr.amova(nancycats2, ~Population, within=FALSE,
                                dist=nancy.dist, squared=FALSE)
nancy.dist.amova

# Via vegan::rda
library(vegan)
nancycats.rda.data &lt;- data.frame(Population=nancycats2@pop)

nancycats.rda.result &lt;- rda(nancycats2@tab ~ Population, data=nancycats.rda.data)  
nancycats.rda.result
amova.rda(nancycats.rda.result, nancycats.rda.data) 

# Example with 2 levels
# Same example as for poppr::poppr.amova
data(Aeut)
strata(Aeut) &lt;- other(Aeut)$population_hierarchy[-1]
agc &lt;- as.genclone(Aeut)
agc
amova.result &lt;- poppr.amova(agc, ~Pop/Subpop, within=FALSE)
amova.result

agc.rda.data &lt;- data.frame(as.matrix(agc$other$population_hierarchy))
agc.rda.data[, 1] &lt;- as.factor(agc.rda.data[, 1])
agc.rda.data[, 2] &lt;- as.factor(agc.rda.data[, 2])
agc.rda.data[, 3] &lt;- as.factor(agc.rda.data[, 3])
str(agc.rda.data)

agc.rda.result &lt;- rda(agc@tab ~ Pop + Pop_Subpop, data=agc.rda.data)  
agc.rda.result
amova.rda(agc.rda.result, agc.rda.data) 

# Example with a balanced data set
# library(BiodiversityR)
data(warcom)
data(warenv)

warburgia.genind &lt;- genind(warcom)
warburgia.genind@strata &lt;- warenv
poppr.amova(warburgia.genind, ~ population, within=FALSE)

warburgia.rda &lt;- rda(warcom ~ population, data=warenv)
warburgia.rda
amova.rda(warburgia.rda, warenv)


## End(Not run)
</code></pre>

<hr>
<h2 id='count.model'>Calibration and Prediction Functions for Changes in Allele Frequencies</h2><span id='topic+count.model'></span><span id='topic+count.pred'></span><span id='topic+freq.model'></span><span id='topic+freq.pred'></span><span id='topic+freq.ggplot'></span>

<h3>Description</h3>

<p>The functions implement a two-step calibration and prediction process for allele frequencies, whereby the second calibration step uses the predictions of the first step.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
count.model(genpop.data, env.data, permutations = 99,
    ordistep = FALSE, cca.model = FALSE)

count.pred(count.modeled, env.data)

freq.model(count.predicted)

freq.pred(freq.modeled, count.predicted)

freq.ggplot(freq.predicted, plot.best = TRUE, threshold = 0.50,
	colour.Pop = TRUE, manual.colour.values = NULL,
	xlim = c(0.0, 1.0), ylim = c(0.0, 1.0))

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="count.model_+3A_genpop.data">genpop.data</code></td>
<td>
<p> Data in the format of <code><a href="adegenet.html#topic+genpop">genpop</a></code>. </p>
</td></tr>
<tr><td><code id="count.model_+3A_env.data">env.data</code></td>
<td>
<p> Descriptors of (bio-)climatic conditions for the populations, either for the baseline climate (to check calibrations) or future/past climates. It is expected that these are in the same sequence as the populations in the <code>genpop.data</code>.  </p>
</td></tr>
<tr><td><code id="count.model_+3A_permutations">permutations</code></td>
<td>
<p> Number of permutations as in <code><a href="vegan.html#topic+anova.cca">anova.cca</a></code>. </p>
</td></tr>
<tr><td><code id="count.model_+3A_ordistep">ordistep</code></td>
<td>
<p> Check the results of  <a href="vegan.html#topic+ordistep">ordistep</a>. </p>
</td></tr>
<tr><td><code id="count.model_+3A_cca.model">cca.model</code></td>
<td>
<p> Fit a CCA model with the minor allele frequency as in Blumstein et al. 2020. </p>
</td></tr>
<tr><td><code id="count.model_+3A_count.modeled">count.modeled</code></td>
<td>
<p> Model fitted by <code>count.model</code>. </p>
</td></tr>
<tr><td><code id="count.model_+3A_count.predicted">count.predicted</code></td>
<td>
<p> Predictions made by <code>count.pred</code>, expected to be predicted for the baseline climate as a second step in the calibration process. </p>
</td></tr>
<tr><td><code id="count.model_+3A_freq.modeled">freq.modeled</code></td>
<td>
<p> Model fitted by <code>freq.model</code>. </p>
</td></tr>
<tr><td><code id="count.model_+3A_freq.predicted">freq.predicted</code></td>
<td>
<p> Allele frequencies predicted by <code>freq.pred</code>. </p>
</td></tr>
<tr><td><code id="count.model_+3A_plot.best">plot.best</code></td>
<td>
<p> Plot the models with highest R2 (<code>TRUE</code>) or lowest R2 values (<code>FALSE</code>). </p>
</td></tr>
<tr><td><code id="count.model_+3A_threshold">threshold</code></td>
<td>
<p> Threshold used to distinguish between the best and worst model. </p>
</td></tr>
<tr><td><code id="count.model_+3A_colour.pop">colour.Pop</code></td>
<td>
<p> Colour populations differently (<code>TRUE</code>), or alleles differently (<code>FALSE</code>) </p>
</td></tr>
<tr><td><code id="count.model_+3A_manual.colour.values">manual.colour.values</code></td>
<td>
<p> Manual specifications for colour values </p>
</td></tr>
<tr><td><code id="count.model_+3A_xlim">xlim</code></td>
<td>
<p> limits of the x-axis </p>
</td></tr>
<tr><td><code id="count.model_+3A_ylim">ylim</code></td>
<td>
<p> limits of the y-axis </p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions allow for an almost completely alternative workflow of predicting shifts in allele frequencies under climate change than the protocol developed by Blumstein et al. 2020. The methodology available here calibrates and predicts changes in allele frequencies via redundancy analysis (Blumstein et al. use canonical correspondence analysis) that were calibrated from allele counts (Blumstein et al. use allele frequencies) for all alleles (Blumstein et al. only calibrate the minor alleles) in a two-step calibration process (the second step via freq.model and freq.predict to ensure that predicted alleles are in the range of 0-1 is not included in protocol developed by Blumstein et al.). Other key differences in the methodology are that explanatory variables are expected to be bioclimatic variables (and not principal components as in Blumstein et al.) and that the input data is expected to be in the <code><a href="adegenet.html#topic+genpop">genpop</a></code> format. Although a method to reduce the number of explanatory variables via <code><a href="vegan.html#topic+ordistep">ordistep</a></code> is shown, I advise against reducing the explanatory variables as this likely will reduce the explanatory power of the models, whereas explanatory power is the major objective of calibrating these functions. Motivations for the differences in methodologies of Blumstein et al. and the ones available in AlleleShift are explained by Kindt 2021.
</p>
<p>Confidence intervals are calculated via <code><a href="stats.html#topic+qt">qt</a></code> with p=0.95 and df=np (number of populations), although the GAM is fitted only once for all the alleles and all the populations (that was my choice to reduce overfitting the baseline data).
</p>
<p>The 'darkolivegreen4'-coloured reference lines shown via <code>freq.ggplot</code> correspond to a 1:1 relationship (full line) and 1:0.95 and 1:1.05 relationships (dashed lines).
</p>


<h3>Value</h3>

<p>The functions enable calibration and prediction of allele frequencies.
</p>


<h3>Author(s)</h3>

<p>Roeland Kindt (World Agroforestry, CIFOR-ICRAF)</p>


<h3>References</h3>

<p>Blumstein et al. 2020. Protocol for Projecting Allele Frequency Change under Future Climate Change at Adaptive-Associated Loci. <a href="https://doi.org/10.1016/j.xpro.2020.100061">doi:10.1016/j.xpro.2020.100061</a>
</p>
<p>Kindt R. 2021. AlleleShift: An R package to predict and visualize population-level changes in allele frequencies in response to climate change. PeerJ 9:e11534. <a href="https://doi.org/10.7717/peerj.11534">doi:10.7717/peerj.11534</a>
</p>
<p>Brauer CJ., Sandoval-Castillo J, Gates K et al. 2023, Natural hybridization reduces vulnerability to climate change. Nat. Clim. Chang. 13: 282-289, <a href="https://doi.org/10.1038/s41558-022-01585-1">doi:10.1038/s41558-022-01585-1</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+VIF.subset">VIF.subset</a></code>, <code><a href="#topic+population.shift">population.shift</a></code>, <code><a href="#topic+amova.rda">amova.rda</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


# A typical work flow would consist of:

# 1. Reduce the number of explanatory variables

data(Poptri.baseline.env)
data(Poptri.future.env)

VIF.select &lt;- VIF.subset(Poptri.baseline.env,
                         keep=c("MAT", "CMI"),
                         cor.plot=TRUE)

VIF.select$VIF$vars.included

baseline.env &lt;- Poptri.baseline.env[, VIF.select$VIF$vars.included]
summary(baseline.env)

future.env &lt;- Poptri.future.env[, VIF.select$VIF$vars.included]


# 2. Create the genpop object

data(Poptri.genind)
Poptri.genpop &lt;- adegenet::genind2genpop(Poptri.genind)

# Get to know the populations and the alleles
poppr::poppr(Poptri.genind)
adegenet::makefreq(Poptri.genpop)

# 3. Calibrate the models

# Note that the ordistep procedure is not needed
# CCA model only to compare results with those by Blumstein et al. 2020
Poptri.count.model &lt;- count.model(Poptri.genpop,
                                  env.data=baseline.env,
				  ordistep=TRUE,
                                  cca.model=TRUE)

Poptri.pred.baseline &lt;- count.pred(Poptri.count.model, env.data=baseline.env)
head(Poptri.pred.baseline)

Poptri.freq.model &lt;- freq.model(Poptri.pred.baseline)
Poptri.freq.baseline &lt;- freq.pred(Poptri.freq.model,
                                  count.predicted=Poptri.pred.baseline)

head(Poptri.freq.baseline)

# 4. Check how well the models predict baseline allele frequencies

# Populations are split in those with R2 &gt; 0.50 and those with R2 &lt; 0.50
# Better populations
plotA1 &lt;- freq.ggplot(Poptri.freq.baseline,
            plot.best=TRUE,
            ylim=c(0.0, 0.8))
plotA1

# Populations with low R2
manual.colour.values1 &lt;- ggsci::pal_npg()(5)

plotB1 &lt;- freq.ggplot(Poptri.freq.baseline,
            plot.best=FALSE,
            manual.colour.values=manual.colour.values1,
            xlim=c(0, 0.5),
            ylim=c(0, 0.25))
plotB1

# Colouring by alleles

plotA2 &lt;- freq.ggplot(Poptri.freq.baseline,
            colour.Pop=FALSE,
            plot.best=TRUE,
            ylim=c(0.0, 0.8),
            manual.colour.values=manual.colour.values1)

plotA2

plotB2 &lt;- freq.ggplot(Poptri.freq.baseline,
            colour.Pop=FALSE,
            plot.best=FALSE,
            manual.colour.values=manual.colour.values1,
            xlim=c(0, 0.5),
            ylim=c(0, 0.25))

plotB2

# 5. Predict future allele frequencies

Poptri.pred.future &lt;- count.pred(Poptri.count.model, env.data=future.env)
head(Poptri.pred.future)

Poptri.freq.future &lt;- freq.pred(Poptri.freq.model,
                                count.predicted=Poptri.pred.future)

# The key results are variables 'Allele.freq' representing the baseline allele frequencies
# and variables 'Freq.e2', the predicted frequency for the future/ past climate.
# Variable 'Freq.e1' is the predicted allele frequency in step 1

head(Poptri.freq.future)

# 5. Visualize the changes
# See functions shift.dot.ggplot, shift.pie.ggplot, shift.moon.gpplot,
# shift.waffle.ggplot and shift.surf.ggplot


</code></pre>

<hr>
<h2 id='Poptri.genind'>Example data sets for Populus trichocarpa.</h2><span id='topic+Poptri.genind'></span><span id='topic+Poptri.baseline.env'></span><span id='topic+Poptri.future.env'></span><span id='topic+Poptri.loc'></span><span id='topic+Poptri.freq.baseline'></span><span id='topic+Poptri.freq.future'></span><span id='topic+Poptri.1985to2085'></span>

<h3>Description</h3>

<p>These data sets include the same data that were used by Blumstein et al. 2020 to document their protocol. Genomic data has been processed further into the <code><a href="adegenet.html#topic+genind">genind</a></code> format. Data sets Poptri.freq.baseline, Poptri.freq.future and Poptri.1985to2085 were obtained with the calibration and prediction methods of this package. They are included to make the documentation of the various plotting methods shorter, but also show the types of results someone can obtain.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Poptri.genind)
data(Poptri.baseline.env)
data(Poptri.future.env)
data(Poptri.loc)
data(Poptri.freq.baseline)
data(Poptri.freq.future)
data(Poptri.1985to2085)
</code></pre>


<h3>References</h3>

<p>Blumstein et al. 2020. Protocol for Projecting Allele Frequency Change under Future Climate Change at Adaptive-Associated Loci. <a href="https://doi.org/10.1016/j.xpro.2020.100061">doi:10.1016/j.xpro.2020.100061</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Poptri.genind)
</code></pre>

<hr>
<h2 id='population.shift'>Shifts of Populations in Environmental Space as Response to Climate Change</h2><span id='topic+population.shift'></span><span id='topic+environmental.novel'></span>

<h3>Description</h3>

<p>The function plots the locations of each population in baseline and future climates. Arrows indicate the shifts in positions of the populations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
population.shift(baseline.env.data, future.env.data,
	option=c("PCA", "RDA"), vector.multiply=1)

environmental.novel(baseline.env.data, future.env.data)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="population.shift_+3A_baseline.env.data">baseline.env.data</code></td>
<td>
<p> Baseline (bio-)climatic conditions for the populations. </p>
</td></tr>
<tr><td><code id="population.shift_+3A_future.env.data">future.env.data</code></td>
<td>
<p> Changed (bio-)climatic conditions in future/past for the populations. </p>
</td></tr>
<tr><td><code id="population.shift_+3A_option">option</code></td>
<td>
<p> Should an explanatory variable corresponding to the climate period be used by <code><a href="vegan.html#topic+rda">rda</a></code>. </p>
</td></tr>
<tr><td><code id="population.shift_+3A_vector.multiply">vector.multiply</code></td>
<td>
<p> Multiplier for vector scores in the ordination diagrams. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>See Kindt (2020) for alternative methods of generating ordination diagrams via vegan, BiodiversityR and ggplot2.
</p>
<p>Function <code>environmental.novel</code> identifies populations with future (or past) environmental conditions that are outside the baseline range. The function further calculates the probability of observing the future condition via <code><a href="stats.html#topic+pnorm">pnorm</a></code> with the mean and standard deviation from the baseline conditions. Where one or several variables are outside the baseline range, data are provided for the variable with the smallest probability.
</p>


<h3>Value</h3>

<p>The main function generates an ordination diagram that depicts changes between baseline and future/past conditions for the populations.
</p>


<h3>Author(s)</h3>

<p>Roeland Kindt (World Agroforestry, CIFOR-ICRAF)</p>


<h3>References</h3>

<p>Kindt R. 2020. Ordination graphs with vegan, BiodiversityR and ggplot2. <a href="https://rpubs.com/Roeland-KINDT">https://rpubs.com/Roeland-KINDT</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


data(Poptri.baseline.env)
data(Poptri.future.env)

environmental.novel(Poptri.baseline.env, Poptri.future.env)
# as if for past climates
environmental.novel(Poptri.future.env, Poptri.baseline.env)

VIF.select &lt;- VIF.subset(Poptri.baseline.env,
                         keep=c("MAT", "CMI"),
                         cor.plot=FALSE)

VIF.select$vars.included

baseline.env &lt;- Poptri.baseline.env[, VIF.select$vars.included]
future.env &lt;- Poptri.future.env[, VIF.select$vars.included]

environmental.novel(baseline.env, future.env)

plotA &lt;- population.shift(baseline.env,
                 future.env,
                 option="PCA")
plotA

plotB &lt;- population.shift(baseline.env,
                 future.env,
                 option="RDA")
plotB


</code></pre>

<hr>
<h2 id='shift.dot.ggplot'>Shifts of Allele Frequencies as Response to Climate Change</h2><span id='topic+shift.dot.ggplot'></span>

<h3>Description</h3>

<p>The function shows changes in allele frequencies between the baseline and future/past climate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
shift.dot.ggplot(freq.future,
	mean.change = FALSE, change.FUN = stats::median,
	baseline.colour = "black",
	future.colour = "dodgerblue3",
	manual.colour.values=c("firebrick3", "chartreuse4"))

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="shift.dot.ggplot_+3A_freq.future">freq.future</code></td>
<td>
<p> Result from <code>AlleleShift::predFreq</code>.. </p>
</td></tr>
<tr><td><code id="shift.dot.ggplot_+3A_mean.change">mean.change</code></td>
<td>
<p> Aggregate changes among alleles. </p>
</td></tr>
<tr><td><code id="shift.dot.ggplot_+3A_change.fun">change.FUN</code></td>
<td>
<p> Function used the aggregate changes. </p>
</td></tr>
<tr><td><code id="shift.dot.ggplot_+3A_baseline.colour">baseline.colour</code>, <code id="shift.dot.ggplot_+3A_future.colour">future.colour</code>, <code id="shift.dot.ggplot_+3A_manual.colour.values">manual.colour.values</code></td>
<td>
<p> Colours to be used in the plots. </p>
</td></tr> 
</table>


<h3>Value</h3>

<p>The function generates a ggplot that depicts changes between baseline and future/past allele frequencies of the populations.
</p>


<h3>Author(s)</h3>

<p>Roeland Kindt (World Agroforestry, CIFOR-ICRAF)</p>


<h3>See Also</h3>

<p><code><a href="#topic+shift.pie.ggplot">shift.pie.ggplot</a></code>, <code><a href="#topic+shift.moon.ggplot">shift.moon.ggplot</a></code>, <code><a href="#topic+shift.waffle.ggplot">shift.waffle.ggplot</a></code>, <code><a href="#topic+shift.surf.ggplot">shift.surf.ggplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# The data can be obtained via the count.model and freq.model calibrations.
# These procedures are not repeated here.
data(Poptri.freq.future)

ggdot1 &lt;- shift.dot.ggplot(Poptri.freq.future)
ggdot1

# create an animation

## Not run: 

library(ggplot2)
library(gganimate)
library(gifski)

# The data is an interpolation and extrapolation between the baseline and future climate.
# For actual application, interpolate between climate data from available sources

data(Poptri.1985to2085)

ggdot.all &lt;- ggplot(data=Poptri.1985to2085, group=Decade) +
  scale_y_continuous(limits=c(-0.1, 1.1),
                     breaks=c(0.0, 0.25, 0.5, 0.75, 1.0)) +
  geom_errorbar(aes(x=Pop, ymin=LCL, ymax=UCL),
                colour="grey30", width=0.8, show.legend=FALSE) +
  geom_segment(aes(x=Pop, y=Allele.freq, xend=Pop, yend=Freq.e2, colour=increasing),
               size=1.2) +  
  geom_point(aes(x=Pop, y=Allele.freq),
             colour="black", size=10, alpha=0.7) +                    
  geom_point(aes(x=Pop, y=Freq.e2),
             colour="dodgerblue3", size=10, alpha=0.7) +  
  coord_flip() +
  xlab(element_blank()) +
  ylab("Allele frequencies") +
  theme(panel.grid.minor = element_blank()) +
  labs(colour="Future change in allele frequencies") +
  scale_colour_manual(values=c("firebrick3", "chartreuse4"), 
                      labels=c("decreasing", "increasing")) +
  theme(axis.text.x=element_text(angle=90, vjust=0.5, size=10)) +
  theme(legend.position="top") +
  facet_grid( ~ Allele, scales="free")

ggdot.all

ggdot.anim &lt;- ggdot.all + 
  transition_states(as.factor(Decade), transition_length = 10, state_length = 100) +
  labs(title = "Decade: {closest_state}s")

ggdot.anim2 &lt;- animate(ggdot.anim, fps=5, width=1280, height=720)

getwd()
anim_save(filename="Allele shift animation.gif", animation=ggdot.anim2)


## End(Not run)

</code></pre>

<hr>
<h2 id='shift.moon.ggplot'>Shifts of Allele Frequencies as Response to Climate Change</h2><span id='topic+shift.moon.ggplot'></span><span id='topic+moon.waxer'></span>

<h3>Description</h3>

<p>The function shows changes in allele frequencies between the baseline and future/past climate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
shift.moon.ggplot(baseline.moon, future.moon,
	manual.colour.values = c("white", "grey", "firebrick3", "chartreuse4"),
	manual.colour.codes = c("A baseline ", "B", "A decreasing", "A increasing"))

moon.waxer(freq.in, sort.index= "Pop.index",
	mean.change = FALSE, change.FUN = stats::median,
	freq.focus = "Allele.freq", 
	ypos = 0, right = TRUE)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="shift.moon.ggplot_+3A_baseline.moon">baseline.moon</code>, <code id="shift.moon.ggplot_+3A_future.moon">future.moon</code></td>
<td>
<p> Result from <code>AlleleShift::moon.waxer</code>. </p>
</td></tr>
<tr><td><code id="shift.moon.ggplot_+3A_manual.colour.values">manual.colour.values</code></td>
<td>
<p> Colours to be used in the plot. </p>
</td></tr>
<tr><td><code id="shift.moon.ggplot_+3A_manual.colour.codes">manual.colour.codes</code></td>
<td>
<p> Sequence for the manual colour values. </p>
</td></tr>
<tr><td><code id="shift.moon.ggplot_+3A_freq.in">freq.in</code></td>
<td>
<p> Result from <code>AlleleShift::predFreq</code>. </p>
</td></tr>
<tr><td><code id="shift.moon.ggplot_+3A_sort.index">sort.index</code></td>
<td>
<p> Sequence of the populations in the plot.</p>
</td></tr>
<tr><td><code id="shift.moon.ggplot_+3A_mean.change">mean.change</code></td>
<td>
<p> Aggregate changes among alleles. </p>
</td></tr>
<tr><td><code id="shift.moon.ggplot_+3A_change.fun">change.FUN</code></td>
<td>
<p> Function used the aggregate changes. </p>
</td></tr>
<tr><td><code id="shift.moon.ggplot_+3A_freq.focus">freq.focus</code></td>
<td>
<p> Allele frequency for which to calculate statistics, either 'Allele.freq' or 'Freq.e2'. </p>
</td></tr>
<tr><td><code id="shift.moon.ggplot_+3A_ypos">ypos</code>, <code id="shift.moon.ggplot_+3A_right">right</code></td>
<td>
<p> Arguments used for plotting, mainly as in <code><a href="gggibbous.html#topic+geom_moon">geom_moon</a></code> </p>
</td></tr> 
</table>


<h3>Value</h3>

<p>The function generates a ggplot that depicts changes between baseline and future/past allele frequencies of the populations.
</p>


<h3>Author(s)</h3>

<p>Roeland Kindt (World Agroforestry, CIFOR-ICRAF)</p>


<h3>See Also</h3>

<p><code><a href="#topic+shift.dot.ggplot">shift.dot.ggplot</a></code>, <code><a href="#topic+shift.pie.ggplot">shift.pie.ggplot</a></code>, <code><a href="#topic+shift.waffle.ggplot">shift.waffle.ggplot</a></code>, <code><a href="#topic+shift.surf.ggplot">shift.surf.ggplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# The data can be obtained via the count.model and freq.model calibrations.
# These procedures are not repeated here.
data(Poptri.freq.baseline)
data(Poptri.freq.future)

Poptri.baseline.moon &lt;- moon.waxer(Poptri.freq.baseline,
                                   sort.index="Latitude.index")

Poptri.future.moon &lt;- moon.waxer(Poptri.freq.future, 
                                 sort.index="Latitude.index",
                                 freq.focus="Freq.e2",
                                ypos=1)

ggmoon1 &lt;- shift.moon.ggplot(Poptri.baseline.moon, 
                             Poptri.future.moon)
ggmoon1

</code></pre>

<hr>
<h2 id='shift.pie.ggplot'>Shifts of Allele Frequencies as Response to Climate Change</h2><span id='topic+shift.pie.ggplot'></span><span id='topic+pie.baker'></span>

<h3>Description</h3>

<p>The function shows changes in allele frequencies between the baseline and future/past climate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
shift.pie.ggplot(baseline.pie, future.pie,
	manual.colour.values = c("black", "grey", "firebrick3", "chartreuse4"),
	manual.colour.codes = c("A baseline ", "B", "A decreasing", "A increasing"))

pie.baker(freq.in, sort.index= "Pop.index",
	mean.change = FALSE, change.FUN = stats::median,
	freq.focus = "Allele.freq", ypos = 0,
	r0 = 0.1, r = 0.5, focus = 0.2 )

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="shift.pie.ggplot_+3A_baseline.pie">baseline.pie</code>, <code id="shift.pie.ggplot_+3A_future.pie">future.pie</code></td>
<td>
<p> Result from <code>AlleleShift::pie.baker</code>. </p>
</td></tr>
<tr><td><code id="shift.pie.ggplot_+3A_manual.colour.values">manual.colour.values</code></td>
<td>
<p> Colours to be used in the plot. </p>
</td></tr>
<tr><td><code id="shift.pie.ggplot_+3A_manual.colour.codes">manual.colour.codes</code></td>
<td>
<p> Sequence for the manual colour values. </p>
</td></tr>
<tr><td><code id="shift.pie.ggplot_+3A_freq.in">freq.in</code></td>
<td>
<p> Result from <code>AlleleShift::predFreq</code>. </p>
</td></tr>
<tr><td><code id="shift.pie.ggplot_+3A_sort.index">sort.index</code></td>
<td>
<p> Sequence of the populations in the plot.</p>
</td></tr>
<tr><td><code id="shift.pie.ggplot_+3A_mean.change">mean.change</code></td>
<td>
<p> Aggregate changes among alleles. </p>
</td></tr>
<tr><td><code id="shift.pie.ggplot_+3A_change.fun">change.FUN</code></td>
<td>
<p> Function used the aggregate changes. </p>
</td></tr>
<tr><td><code id="shift.pie.ggplot_+3A_freq.focus">freq.focus</code></td>
<td>
<p> Allele frequency for which to calculate statistics, either 'Allele.freq' or 'Freq.e2'. </p>
</td></tr>
<tr><td><code id="shift.pie.ggplot_+3A_ypos">ypos</code>, <code id="shift.pie.ggplot_+3A_r0">r0</code>, <code id="shift.pie.ggplot_+3A_r">r</code>, <code id="shift.pie.ggplot_+3A_focus">focus</code></td>
<td>
<p> Arguments used for plotting, mainly as in <code><a href="ggforce.html#topic+geom_arc_bar">geom_arc_bar</a></code> </p>
</td></tr> 
</table>


<h3>Value</h3>

<p>The function generates a ggplot that depicts changes between baseline and future/past allele frequencies of the populations.
</p>


<h3>Author(s)</h3>

<p>Roeland Kindt (World Agroforestry, CIFOR-ICRAF)</p>


<h3>See Also</h3>

<p><code><a href="#topic+shift.dot.ggplot">shift.dot.ggplot</a></code>, <code><a href="#topic+shift.moon.ggplot">shift.moon.ggplot</a></code>, <code><a href="#topic+shift.waffle.ggplot">shift.waffle.ggplot</a></code>, <code><a href="#topic+shift.surf.ggplot">shift.surf.ggplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# The data can be obtained via the count.model and freq.model calibrations.
# These procedures are not repeated here.
data(Poptri.freq.baseline)
data(Poptri.freq.future)

Poptri.baseline.pie &lt;- pie.baker(Poptri.freq.baseline, r0=0.1,
                                 sort.index="Latitude.index")

Poptri.future.pie &lt;- pie.baker(Poptri.freq.future, r0=0.1, 
                        freq.focus="Freq.e2",
                        sort.index="Latitude.index",
                        ypos=1)

ggpie1 &lt;- shift.pie.ggplot(Poptri.baseline.pie, 
                           Poptri.future.pie)
ggpie1


# create an animation

## Not run: 

library(ggplot2)
library(ggforce)
library(gganimate)
library(gifski)
library(transformr)

# The data is an interpolation and extrapolation between the baseline and future climate.
# For actual application, interpolate between climate data from available sources

data(Poptri.1985to2085)

decades &lt;- sort(unique(Poptri.1985to2085$Decade))

for (d in 1:length(decades)) {
  decade.focal &lt;- decades[d]
  decade.data &lt;- Poptri.1985to2085[Poptri.1985to2085$Decade == decade.focal, ]
  decade.pie &lt;- pie.baker(decade.data, r0=0.1, 
                        freq.focus="Freq.e2",
                        sort.index="Latitude.index",
                        ypos=1)
  decade.pie &lt;- cbind(Decade=rep(decade.focal, nrow(decade.pie)), decade.pie)
  
  if (d == 1) {
    future.pies &lt;- decade.pie
  }else{
    future.pies &lt;- rbind(future.pies, decade.pie)
  }
}

np &lt;- length(unique(Poptri.baseline.pie$Pop))
manual.colour.values &lt;- c("black", "grey", "firebrick3", "chartreuse4")
  
ggpie.all &lt;- ggplot(data=future.pies, group=Decade) +
  scale_x_continuous(limits=c(0.5, np+0.5),
                     breaks=seq(from=1, to=np, by=1),
                     labels=levels(Poptri.baseline.pie$Pop)) +
  geom_arc_bar(data=Poptri.baseline.pie,
               aes(x0=sort.index, y0=ypos, r0=r0, r=0.4, 
                   start=start, end=end, fill=colour),
               size=0.04, alpha=1, colour="snow1") +                    
  geom_arc_bar(data=future.pies,
               aes(x0=sort.index, y0=ypos, r0=r0, r=0.4, 
                   start=start, end=end, fill=change.colour),
               size=0.04, alpha=1, colour="snow1") + 
  geom_point(data=subset(future.pies, increasing==TRUE),
            aes(x=sort.index, y=ypos),
            size=5, shape=21, fill=manual.colour.values[4], stroke=0.03, show.legend=FALSE) +
  geom_point(data=subset(future.pies, increasing==FALSE),
            aes(x=sort.index, y=ypos),
            size=5, shape=21, fill=manual.colour.values[3], stroke=0.03, show.legend=FALSE) +
  coord_flip() +
  xlab(element_blank()) +
  ylab(element_blank()) +
  labs(fill=" ") +
  scale_fill_manual(values=manual.colour.values, 
                    labels=c("A baseline ", "B", "A decreasing", "A increasing")) +
  theme(panel.grid = element_blank()) +
  theme(axis.text.x=element_blank()) +
  theme(axis.ticks.x = element_blank()) +
  theme(legend.position="top") +
  facet_grid( ~ Allele, scales="free")

ggpie.all
  
# note this will take quite a while!
ggpie.anim &lt;- ggpie.all + 
  transition_states(as.factor(Decade), transition_length = 10, state_length = 100) +
  labs(title = "Decade: {closest_state}s")

ggpie.anim2 &lt;- animate(ggpie.anim, fps=5, width=1280, height=720)

getwd()
anim_save(filename="Allele shift pie animation.gif", animation=ggpie.anim2)


## End(Not run)

</code></pre>

<hr>
<h2 id='shift.surf.ggplot'>Shifts of Allele Frequencies as Response to Climate Change</h2><span id='topic+shift.surf.ggplot'></span>

<h3>Description</h3>

<p>The function shows changes in allele frequencies between the baseline and future/past climate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
shift.surf.ggplot(freq.future,
	Allele.focus=unique(freq.future$Allele)[1],
	freq.focus="Allele.freq",
	xcoord="LON", ycoord="LAT",
	mean.change = FALSE, change.FUN = stats::median,
	manual.colour.values = c("firebrick3", "chartreuse4"),
	...)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="shift.surf.ggplot_+3A_freq.future">freq.future</code></td>
<td>
<p> Result from <code>AlleleShift::predFreq</code>. </p>
</td></tr>
<tr><td><code id="shift.surf.ggplot_+3A_freq.focus">freq.focus</code></td>
<td>
<p> Selection of the Allele.</p>
</td></tr>
<tr><td><code id="shift.surf.ggplot_+3A_allele.focus">Allele.focus</code></td>
<td>
<p> Selection of the frequency.</p>
</td></tr>
<tr><td><code id="shift.surf.ggplot_+3A_xcoord">xcoord</code>, <code id="shift.surf.ggplot_+3A_ycoord">ycoord</code></td>
<td>
<p> Geographical coordinates of the populations.</p>
</td></tr>
<tr><td><code id="shift.surf.ggplot_+3A_mean.change">mean.change</code></td>
<td>
<p> Aggregate changes among alleles. </p>
</td></tr>
<tr><td><code id="shift.surf.ggplot_+3A_change.fun">change.FUN</code></td>
<td>
<p> Function used the aggregate changes. </p>
</td></tr>
<tr><td><code id="shift.surf.ggplot_+3A_manual.colour.values">manual.colour.values</code></td>
<td>
<p>Colours to be used in the plot. </p>
</td></tr>
<tr><td><code id="shift.surf.ggplot_+3A_...">...</code></td>
<td>
<p> Options for <code><a href="vegan.html#topic+ordisurf">ordisurf</a></code>. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Populations are plotted in geographical space via ordination plotting methods, which is suitable as fixed coordinate systems are recommended both in ordination diagrams and maps. See Kindt (2020) for alternative methods of generating ordination diagrams via vegan, BiodiversityR and ggplot2.
</p>
<p>Kindt (2021) shows how a STAMEN baseline map can be used to produce high resolution images via the ggmap package.
</p>


<h3>Value</h3>

<p>The function generates a ggplot that depicts changes between baseline and future/past allele frequencies of the populations.
</p>


<h3>Author(s)</h3>

<p>Roeland Kindt (World Agroforestry, CIFOR-ICRAF)</p>


<h3>References</h3>

<p>Kindt R. 2020. Ordination graphs with vegan, BiodiversityR and ggplot2. <a href="https://rpubs.com/Roeland-KINDT">https://rpubs.com/Roeland-KINDT</a>
</p>
<p>Kindt, R. 2021. Plotting smoothed surface diagrams of allele frequencies obtained
from AlleleShift on a baseline map via ggmap. <a href="https://rpubs.com/Roeland-KINDT">https://rpubs.com/Roeland-KINDT</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+shift.dot.ggplot">shift.dot.ggplot</a></code>, <code><a href="#topic+shift.pie.ggplot">shift.pie.ggplot</a></code>, <code><a href="#topic+shift.moon.ggplot">shift.moon.ggplot</a></code>, <code><a href="#topic+shift.waffle.ggplot">shift.waffle.ggplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# The data can be obtained via the count.model and freq.model calibrations.
# These procedures are not repeated here.
data(Poptri.freq.baseline)
data(Poptri.freq.future)

# Plots for the first allele
# Symbols and colours indicate future change (green, ^ = future increase)
# Symbol size reflects the frequency in the climate shown

# Baseline climate
plotA &lt;- shift.surf.ggplot(Poptri.freq.future,
                  xcoord="Long", ycoord="Lat",
                  Allele.focus=unique(Poptri.freq.future$Allele)[1],
                  freq.focus="Allele.freq")
plotA

# Future/past climate
plotB &lt;- shift.surf.ggplot(Poptri.freq.future,
                  xcoord="Long", ycoord="Lat",
                  Allele.focus=unique(Poptri.freq.future$Allele)[1],
                  freq.focus="Freq.e2")
plotB

# Plots for the fifth allele

# Baseline climate
plotC &lt;- shift.surf.ggplot(Poptri.freq.future,
                  xcoord="Long", ycoord="Lat",
                  Allele.focus=unique(Poptri.freq.future$Allele)[5],
                  freq.focus="Allele.freq")
plotC

# Future climate
plotD &lt;- shift.surf.ggplot(Poptri.freq.future,
                  xcoord="Long", ycoord="Lat",
                  Allele.focus=unique(Poptri.freq.future$Allele)[5],
                  freq.focus="Freq.e2")
plotD

# create an animation

## Not run: 

library(ggplot2)
library(ggforce)
library(gganimate)
library(gifski)
library(transformr)

# The data is an interpolation and extrapolation between the baseline and future climate.
# For actual application, interpolate between climate data from available sources

data(Poptri.1985to2085)

Poptri.1985to2085$xcoord &lt;- Poptri.1985to2085$Long
Poptri.1985to2085$ycoord &lt;- Poptri.1985to2085$Lat

alleles &lt;- sort(unique(as.character(Poptri.1985to2085$Allele)))
future.data &lt;- Poptri.1985to2085[Poptri.1985to2085$Allele == alleles[1], ]

decades &lt;- sort(unique(future.data$Decade))

for (d in 1:length(decades)) {
  decade.focal &lt;- decades[d]
  decade.data &lt;- future.data[future.data$Decade == decade.focal, ]
  plotLONLAT &lt;- vegan::ordiplot(decade.data[, c("xcoord", "ycoord")])
  surfAllele &lt;- BiodiversityR::ordisurfgrid.long(vegan::ordisurf(plotLONLAT, y=decade.data$Freq.e2))
  decade.surf &lt;- cbind(Decade=rep(decade.focal, nrow(surfAllele)), surfAllele)
  if (d == 1) {
    future.surfs &lt;- decade.surf
  }else{
    future.surfs &lt;- rbind(future.surfs, decade.surf)
  }
}

# The function above will not be able to predict far into the future.
# The results obtained (future.surfs) can still be used for plotting.

ggsurf.all &lt;- ggplot(data=future.surfs, group=Decade) +
  geom_contour_filled(aes(x=x, y=y, z=z),
                      breaks=seq(from=0.0, to=1.05, by=0.05)) +
  geom_point(data=subset(future.data, Decade==Decade),
             aes(x=xcoord, y=ycoord, size=Freq.e2, shape=increasing),
             colour="red", alpha=0.8, stroke=1.5, show.legend=FALSE) +
  xlab(element_blank()) +
  ylab(element_blank()) +
  labs(fill=alleles[1]) +
  scale_fill_viridis_d() +
  scale_colour_manual(values=c("firebrick3", "chartreuse4"),
                      guide=FALSE) +
  scale_size_area(max_size=6) +
  scale_shape_manual(values=c(6, 2)) +
  theme(panel.grid = element_blank()) +
  theme(axis.text= element_blank()) +
  theme(axis.ticks = element_blank()) +
  theme(legend.title = element_text(size=9)) +
  theme(legend.text = element_text(size=8)) +
  coord_fixed()

ggsurf.all

ggsurf.anim &lt;- ggsurf.all +
  transition_states(as.factor(Decade), transition_length = 10, state_length = 100) +
  labs(title = "Decade: {closest_state}s")

ggsurf.anim2 &lt;- animate(ggsurf.anim, fps=5, width=1280, height=720)

getwd()
anim_save(filename="Allele shift surf animation.gif", animation=ggsurf.anim2)


## End(Not run)

</code></pre>

<hr>
<h2 id='shift.waffle.ggplot'>Shifts of Allele Frequencies as Response to Climate Change</h2><span id='topic+shift.waffle.ggplot'></span><span id='topic+waffle.baker'></span>

<h3>Description</h3>

<p>The function shows changes in allele frequencies between the baseline and future/past climate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
shift.waffle.ggplot(future.waffle,
	manual.colour.values = c("black", "grey", "firebrick3", "chartreuse4"),
	manual.colour.codes = c("A baseline ", "B", "A decreasing", "A increasing"))

waffle.baker(freq.in, sort.index = "Pop.index",
	mean.change = FALSE, change.FUN = stats::median)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="shift.waffle.ggplot_+3A_future.waffle">future.waffle</code></td>
<td>
<p> Result from <code>AlleleShift::waffle.baker</code>. </p>
</td></tr>
<tr><td><code id="shift.waffle.ggplot_+3A_manual.colour.values">manual.colour.values</code></td>
<td>
<p> Colours to be used in the plot. </p>
</td></tr>
<tr><td><code id="shift.waffle.ggplot_+3A_manual.colour.codes">manual.colour.codes</code></td>
<td>
<p> Sequence for the manual colour values. </p>
</td></tr>
<tr><td><code id="shift.waffle.ggplot_+3A_freq.in">freq.in</code></td>
<td>
<p> Result from <code>AlleleShift::predFreq</code>. </p>
</td></tr>
<tr><td><code id="shift.waffle.ggplot_+3A_sort.index">sort.index</code></td>
<td>
<p> Sequence of the populations in the plot. </p>
</td></tr>
<tr><td><code id="shift.waffle.ggplot_+3A_mean.change">mean.change</code></td>
<td>
<p> Aggregate changes among alleles. </p>
</td></tr>
<tr><td><code id="shift.waffle.ggplot_+3A_change.fun">change.FUN</code></td>
<td>
<p> Function used the aggregate changes. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Althought a package <code>ggwaffle</code> exists, I opted to bake my own waffles (possibly the Belgian in me.). As a separate row is created for each square/rectangle of the waffle, the resulting data is quite large. Hence trying to animate this is probably a bad idea (unless you want to make some real waffles while your computer is busy).
</p>


<h3>Value</h3>

<p>The function generates a ggplot that depicts changes between baseline and future/past allele frequencies of the populations.
</p>


<h3>Author(s)</h3>

<p>Roeland Kindt (World Agroforestry, CIFOR-ICRAF)</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# The data can be obtained via the count.model and freq.model calibrations.
# These procedures are not repeated here.
data(Poptri.freq.baseline)
data(Poptri.freq.future)

Poptri.future.waffle &lt;- waffle.baker(Poptri.freq.future,
                           sort.index="Latitude.index")

ggwaffle1 &lt;- shift.waffle.ggplot(Poptri.future.waffle)
ggwaffle1

</code></pre>

<hr>
<h2 id='VIF.subset'>Reduce the Climate Data based on Variance Inflation Factor Analysis.</h2><span id='topic+VIF.subset'></span>

<h3>Description</h3>

<p>Through Variance Inflation Factor (VIF) analysis, a subset of variables is indentified where all variables have VIF below a predefined threshold.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
VIF.subset(data, VIF.max, keep=NULL, silent=FALSE, cor.plot=TRUE)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VIF.subset_+3A_data">data</code></td>
<td>
<p> (Bio)-Climatic or environmental descriptors of the populations </p>
</td></tr>
<tr><td><code id="VIF.subset_+3A_vif.max">VIF.max</code></td>
<td>
<p> Maximum Variance Inflation Factor as in <code><a href="BiodiversityR.html#topic+ensemble.VIF.dataframe">ensemble.VIF.dataframe</a></code> that is used internally. </p>
</td></tr>
<tr><td><code id="VIF.subset_+3A_keep">keep</code></td>
<td>
<p> Variables to keep as in as in <code><a href="BiodiversityR.html#topic+ensemble.VIF.dataframe">ensemble.VIF.dataframe</a></code>. </p>
</td></tr> 
<tr><td><code id="VIF.subset_+3A_silent">silent</code></td>
<td>
<p> Limit the verbose output as in <code><a href="BiodiversityR.html#topic+ensemble.VIF.dataframe">ensemble.VIF.dataframe</a></code>. </p>
</td></tr> 
<tr><td><code id="VIF.subset_+3A_cor.plot">cor.plot</code></td>
<td>
<p> Generate a correlation matrix for the final subset via <code><a href="GGally.html#topic+ggcorr">ggcorr</a></code>. </p>
</td></tr> 
</table>


<h3>Value</h3>

<p>The function returns information on a subset of variables where VIF is below a pre-defined threshold.
</p>


<h3>Author(s)</h3>

<p>Roeland Kindt (World Agroforestry, CIFOR-ICRAF)</p>


<h3>Examples</h3>

<pre><code class='language-R'>


data(Poptri.baseline.env)

# error as many variables are highly correlated
# VIF.subset(Poptri.baseline.env)

VIF.subset(Poptri.baseline.env, keep=c("MAT", "CMI"), cor.plot=TRUE)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
