<!DOCTYPE html><html><head><title>Help for package alleHap</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {alleHap}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#alleHap-package'>
<p>Allele Imputation and Haplotype Reconstruction from Pedigree Databases</p></a></li>
<li><a href='#alleHaplotyper'><p>Haplotyping of a dataset composed by several families.</p></a></li>
<li><a href='#alleImputer'><p>Imputation of missing alleles from a dataset composed by families.</p></a></li>
<li><a href='#alleLoader'><p>Data loading of nuclear families (in .ped format)</p></a></li>
<li><a href='#alleSimulator'><p>Simulation of genetic data (alleles) and non-genetic data (family identifiers)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Allele Imputation and Haplotype Reconstruction from Pedigree
Databases</td>
</tr>
<tr>
<td>Version:</td>
<td>0.9.9</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-08-19</td>
</tr>
<tr>
<td>Author:</td>
<td>Nathan Medina-Rodriguez and Angelo Santana</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Nathan Medina-Rodriguez &lt;nathan.medina@ulpgc.es&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Tools to simulate alphanumeric alleles, impute genetic missing data and reconstruct non-recombinant haplotypes from pedigree databases in a deterministic way. Allelic simulations can be implemented taking into account many factors (such as number of families, markers, alleles per marker,
    probability and proportion of missing genotypes, recombination rate, etc).
    Genotype imputation can be used with simulated datasets or real databases (previously loaded in .ped format). Haplotype reconstruction can be carried
    out even with missing data, since the program firstly imputes each family genotype (without a reference panel), to later reconstruct the corresponding
    haplotypes for each family member. All this considering that each individual (due to meiosis) should unequivocally have two alleles per marker (one inherited
    from each parent) and thus imputation and reconstruction results can be deterministically calculated.</td>
</tr>
<tr>
<td>Imports:</td>
<td>abind, tools, stats, utils</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-08-19 10:14:17 UTC; nathan</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-08-19 10:33:43 UTC</td>
</tr>
</table>
<hr>
<h2 id='alleHap-package'>
Allele Imputation and Haplotype Reconstruction from Pedigree Databases
</h2><span id='topic+alleHap-package'></span><span id='topic+alleHap'></span>

<h3>Description</h3>

<p>Tools to simulate alphanumeric alleles, impute genetic missing data and reconstruct non-recombinant haplotypes from pedigree databases in a deterministic way. Allelic simulations can be implemented taking into account many factors (such as number of families, markers, alleles per marker, probability and proportion of missing genotypes, recombination rate, etc). Genotype imputation can be used with simulated datasets or real databases (previously loaded in .ped format). Haplotype reconstruction can be carried out even with missing data, since the program firstly imputes each family genotype (without a reference panel), to later reconstruct the corresponding haplotypes for each family member. All this considering that each individual (due to meiosis) should unequivocally have two alleles per marker (one inherited from each parent), and thus imputation and reconstruction results can be deterministically calculated.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> alleHap</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 0.9.9</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2017-08-19</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> abind, stats, tools, utils</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;=2)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Nathan Medina-Rodriguez and Angelo Santana
</p>
<p>Maintainer: Nathan Medina-Rodriguez &lt;nathan.medina@ulpgc.es&gt;
</p>


<h3>References</h3>

<p>Medina-Rodriguez, N. Santana A. et al. (2014) alleHap: an efficient algorithm to reconstruct zero-recombinant haplotypes from parent-offspring pedigrees. BMC Bioinformatics, 15, A6 (S-3).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Generation of 10 simulated families with 2 children per family and 20 markers
dataset &lt;- alleSimulator(10,2,20)  # List with simulated alleles and haplotypes
datasetAlls &lt;- dataset[[1]]        # Dataset containing alleles
datasetHaps &lt;- dataset[[2]]        # Dataset containing haplotypes

## Loading of a dataset in .ped format with alphabetical alleles (A,C,G,T)
example1 &lt;- file.path(find.package("alleHap"), "examples", "example1.ped")
datasetAlls1 &lt;- alleLoader(example1)

## Loading of a dataset in .ped format with numerical alleles
example2 &lt;- file.path(find.package("alleHap"), "examples", "example2.ped")
datasetAlls2 &lt;- alleLoader(example2)

## Allele imputation of families with parental missing data
datasetAlls &lt;- alleSimulator(10,4,6,missParProb=0.2)[[1]]
famsImputed &lt;- alleImputer(datasetAlls)

## Allele imputation of families with offspring missing data
datasetAlls &lt;- alleSimulator(10,4,6,missOffProb=0.2)[[1]]
famsImputed &lt;- alleImputer(datasetAlls)

## Haplotype reconstruction for 3 families without missing data.
simulatedFams &lt;- alleSimulator(3,3,6)  
(famsAlls &lt;- simulatedFams[[1]])      # Original data 
famsList &lt;- alleHaplotyper(famsAlls)  # List containing families' alleles and haplotypes
famsList$reImputedAlls                # Re-imputed alleles
famsList$haplotypes                   # Reconstructed haplotypes

## Haplotype reconstruction from a PED file
pedFamPath &lt;- file.path(find.package("alleHap"), "examples", "example3.ped") # PED file path
pedFamAlls &lt;- alleLoader(pedFamPath,dataSummary=FALSE) 
pedFamList &lt;- alleHaplotyper(pedFamAlls)
pedFamAlls                # Original data 
pedFamList$reImputedAlls  # Re-imputed alleles 
pedFamList$haplotypes     # Reconstructed haplotypes
</code></pre>

<hr>
<h2 id='alleHaplotyper'>Haplotyping of a dataset composed by several families.</h2><span id='topic+alleHaplotyper'></span>

<h3>Description</h3>

<p>By analyzing all possible combinations of a parent-offspring pedigree in which parents may be missing (missParProb&gt;0), as long as one child was genotyped, it is possible an unequivocal reconstruction of many parental haplotypes. When neither parent was genotyped (missParProb==1), also it is possible to reconstruct at least two parental haplotypes in certain cases. Regarding offspring haplotypes, if both parents are completely genotyped (missParProb==0), in majority of cases partial offspring haplotypes may be successfully obtained (missOffProb&gt;0).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alleHaplotyper(data, NAsymbol = "?", alleSep = "", invisibleOutput = TRUE,
  dataSummary = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alleHaplotyper_+3A_data">data</code></td>
<td>
<p>Data containing non-genetic and genetic information of families (or PED file path).</p>
</td></tr>
<tr><td><code id="alleHaplotyper_+3A_nasymbol">NAsymbol</code></td>
<td>
<p>Icon which will be placed in the NA values of the haplotypes.</p>
</td></tr>
<tr><td><code id="alleHaplotyper_+3A_allesep">alleSep</code></td>
<td>
<p>Icon which will be used as separator of the haplotype alleles.</p>
</td></tr>
<tr><td><code id="alleHaplotyper_+3A_invisibleoutput">invisibleOutput</code></td>
<td>
<p>Data are not shown by default.</p>
</td></tr>
<tr><td><code id="alleHaplotyper_+3A_datasummary">dataSummary</code></td>
<td>
<p>A summary of the data is shown by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Re-imputed alleles and haplotypes for each loaded family.
</p>


<h3>References</h3>

<p>Medina-Rodriguez, N. Santana A. et al. (2014) alleHap: an efficient algorithm to reconstruct zero-recombinant haplotypes from parent-offspring pedigrees. BMC Bioinformatics, 15, A6 (S-3).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Haplotype reconstruction for 3 families without missing data.
simulatedFams &lt;- alleSimulator(3,3,6)  
(famsAlls &lt;- simulatedFams[[1]])      # Original data 
famsList &lt;- alleHaplotyper(famsAlls)  # List containing families' alleles and haplotypes
famsList$reImputedAlls                # Re-imputed alleles
famsList$haplotypes                   # Reconstructed haplotypes

## Haplotype reconstruction of a family containing missing data in a parent. 
infoFam &lt;- data.frame(famID="FAM002",indID=1:6,patID=c(0,0,1,1,1,1),
                     matID=c(0,0,2,2,2,2),sex=c(1,2,1,2,1,2),phenot=c(2,1,1,2,1,2))
Mkrs &lt;- rbind(c(1,4,2,5,3,6),rep(NA,6),c(1,7,2,3,3,2),
              c(4,7,5,3,6,2),c(1,1,2,2,3,3),c(1,4,2,5,3,6))
colnames(Mkrs) &lt;- c("Mk1_1","Mk1_2","Mk2_1","Mk2_2","Mk3_1","Mk3_2")
(family &lt;- cbind(infoFam,Mkrs))    # Original data 
famList &lt;- alleHaplotyper(family)  # List containing family's alleles and haplotypes
famList$reImputedAlls              # Re-imputed alleles
famList$haplotypes                 # Reconstructed haplotypes

## Haplotype reconstruction from a PED file
pedFamPath &lt;- file.path(find.package("alleHap"), "examples", "example3.ped") # PED file path
pedFamAlls &lt;- alleLoader(pedFamPath,dataSummary=FALSE) 
pedFamList &lt;- alleHaplotyper(pedFamAlls)
pedFamAlls                # Original data 
pedFamList$reImputedAlls  # Re-imputed alleles 
pedFamList$haplotypes     # Reconstructed haplotypes

</code></pre>

<hr>
<h2 id='alleImputer'>Imputation of missing alleles from a dataset composed by families.</h2><span id='topic+alleImputer'></span>

<h3>Description</h3>

<p>By analyzing all possible combinations of a parent-offspring pedigree in which parental and/or offspring genotypes may be missing; as long as one child was genotyped, in certain cases it is possible an unequivocal imputation of the missing genotypes both in parents and children.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alleImputer(data, invisibleOutput = TRUE, dataSummary = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alleImputer_+3A_data">data</code></td>
<td>
<p>Data containing the families' identifiers and the corresponding genetic data (or the path of the PED file).</p>
</td></tr>
<tr><td><code id="alleImputer_+3A_invisibleoutput">invisibleOutput</code></td>
<td>
<p>Data are not shown by default.</p>
</td></tr>
<tr><td><code id="alleImputer_+3A_datasummary">dataSummary</code></td>
<td>
<p>A summary of the data is shown by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Imputed markers, Homozygosity (HMZ) matrix, marker messages and number of unique alleles per marker.
</p>


<h3>References</h3>

<p>Medina-Rodriguez, N. Santana A. et al. (2014) alleHap: an efficient algorithm to reconstruct zero-recombinant haplotypes from parent-offspring pedigrees. BMC Bioinformatics, 15, A6 (S-3).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Imputation of families containing parental missing data
simulatedFams &lt;- alleSimulator(10,4,6,missParProb=0.2) 
famsAlls &lt;- simulatedFams[[1]]       # Original data 
alleImputer(famsAlls)                # Imputed alleles (genotypes)

## Imputation of families containing offspring missing data
datasetAlls &lt;- alleSimulator(10,4,6,missOffProb=0.2)
famsAlls &lt;- simulatedFams[[1]]       # Original data 
alleImputer(famsAlls)                # Imputed alleles (genotypes)

## Imputation of a family marker containing missing values in one parent and one child
infoFam &lt;- data.frame(famID="FAM03",indID=1:5,patID=c(0,0,1,1,1),
                      matID=c(0,0,2,2,2),sex=c(1,2,1,2,1),phenot=0)
mkr &lt;- rbind(father=c(NA,NA),mother=c(1,3),child1=c(1,1),child2=c(2,3),child3=c(NA,NA))
colnames(mkr) &lt;- c("Mkr1_1","Mkr1_2")
famMkr &lt;- cbind(infoFam,mkr)         # Original data 
alleImputer(famMkr)                  # Imputed alleles (genotypes)

</code></pre>

<hr>
<h2 id='alleLoader'>Data loading of nuclear families (in .ped format)</h2><span id='topic+alleLoader'></span>

<h3>Description</h3>

<p>The data to be loaded must be structured in .ped format and families must comprise by parent-offspring pedigrees.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alleLoader(data, invisibleOutput = TRUE, dataSummary = TRUE,
  missingValues = c(-9, -99))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alleLoader_+3A_data">data</code></td>
<td>
<p>Data to be loaded.</p>
</td></tr>
<tr><td><code id="alleLoader_+3A_invisibleoutput">invisibleOutput</code></td>
<td>
<p>Data are not shown by default.</p>
</td></tr>
<tr><td><code id="alleLoader_+3A_datasummary">dataSummary</code></td>
<td>
<p>A summary of the data is shown by default.</p>
</td></tr>
<tr><td><code id="alleLoader_+3A_missingvalues">missingValues</code></td>
<td>
<p>Specification of the character/numerical values which may be missing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Loaded dataset.
</p>


<h3>References</h3>

<p>Medina-Rodriguez, N. Santana A. et al. (2014) alleHap: an efficient algorithm to reconstruct zero-recombinant haplotypes from parent-offspring pedigrees. BMC Bioinformatics, 15, A6 (S-3).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Loading of a dataset in .ped format with alphabetical alleles (A,C,G,T)
example1 &lt;- file.path(find.package("alleHap"), "examples", "example1.ped")
example1Alls &lt;- alleLoader(example1)
head(example1Alls)

## Loading of a dataset in .ped format with numerical alleles
example2 &lt;- file.path(find.package("alleHap"), "examples", "example2.ped")
example2Alls &lt;- alleLoader(example2)
head(example2Alls)

</code></pre>

<hr>
<h2 id='alleSimulator'>Simulation of genetic data (alleles) and non-genetic data (family identifiers)</h2><span id='topic+alleSimulator'></span>

<h3>Description</h3>

<p>Data simulation can be performed taking into account many different factors such as number of families to generate, number of markers (allele pairs), number of different alleles per marker, type of alleles (numeric or character), number of different haplotypes in the population, probability of parent/offspring missing genotypes, proportion of missing genotypes per individual, probability of being affected by disease and recombination rate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alleSimulator(nFams = 2, nChildren = NULL, nMarkers = 3,
  numAllperMrk = NULL, chrAlleles = TRUE, nHaplos = 1200,
  missParProb = 0, missOffProb = 0, ungenotPars = 0, ungenotOffs = 0,
  phenProb = 0.2, recombRate = 0, invisibleOutput = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alleSimulator_+3A_nfams">nFams</code></td>
<td>
<p>Number of families to generate (integer: 1..1000+)</p>
</td></tr>
<tr><td><code id="alleSimulator_+3A_nchildren">nChildren</code></td>
<td>
<p>Number of children of each family (integer: 1..7 or NULL)</p>
</td></tr>
<tr><td><code id="alleSimulator_+3A_nmarkers">nMarkers</code></td>
<td>
<p>Number of markers or allele pairs to generate (integer: 1..1000+)</p>
</td></tr>
<tr><td><code id="alleSimulator_+3A_numallpermrk">numAllperMrk</code></td>
<td>
<p>Number of different alleles per marker (vector or NULL)</p>
</td></tr>
<tr><td><code id="alleSimulator_+3A_chralleles">chrAlleles</code></td>
<td>
<p>Should alleles be expressed as characters A,C,G,T ? (boolean: FALSE, TRUE)</p>
</td></tr>
<tr><td><code id="alleSimulator_+3A_nhaplos">nHaplos</code></td>
<td>
<p>Number of different haplotypes in the population (numeric)</p>
</td></tr>
<tr><td><code id="alleSimulator_+3A_missparprob">missParProb</code></td>
<td>
<p>Probability of parents' missing genotype (numeric: 0..1)</p>
</td></tr>
<tr><td><code id="alleSimulator_+3A_missoffprob">missOffProb</code></td>
<td>
<p>Probability of offspring' missing genotype (numeric: 0..1)</p>
</td></tr>
<tr><td><code id="alleSimulator_+3A_ungenotpars">ungenotPars</code></td>
<td>
<p>Proportion of ungenotyped parents (numeric: 0..1)</p>
</td></tr>
<tr><td><code id="alleSimulator_+3A_ungenotoffs">ungenotOffs</code></td>
<td>
<p>Proportion of ungenotyped offspring (numeric: 0..1)</p>
</td></tr>
<tr><td><code id="alleSimulator_+3A_phenprob">phenProb</code></td>
<td>
<p>Phenotype probability, e.g. being affected by disease (numeric: 0..1)</p>
</td></tr>
<tr><td><code id="alleSimulator_+3A_recombrate">recombRate</code></td>
<td>
<p>Recombination rate (numeric: 0..1)</p>
</td></tr>
<tr><td><code id="alleSimulator_+3A_invisibleoutput">invisibleOutput</code></td>
<td>
<p>Data are not shown by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Families' genotypes and haplotypes.
</p>


<h3>References</h3>

<p>Medina-Rodriguez, N. Santana A. et al. (2014) alleHap: an efficient algorithm to reconstruct zero-recombinant haplotypes from parent-offspring pedigrees. BMC Bioinformatics, 15, A6 (S-3).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Generation of 5 simulated families with 2 children per family and 10 markers
simulatedFams &lt;- alleSimulator(5,2,10)   # List with simulated alleles and haplotypes
simulatedFams[[1]]                       # Alleles (genotypes) of the simulated families
simulatedFams[[2]]                       # Haplotypes of the simulated families

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
