<!DOCTYPE html><html><head><title>Help for package AllelicSeries</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {AllelicSeries}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AllelicSeries-package'><p>AllelicSeries: Allelic Series Test</p></a></li>
<li><a href='#Aggregator'><p>Aggregator</p></a></li>
<li><a href='#ASBT'><p>Allelic Series Burden Test</p></a></li>
<li><a href='#ASKAT'><p>Allelic Series SKAT Test</p></a></li>
<li><a href='#CalcRegParam'><p>Calculate Regression Parameters</p></a></li>
<li><a href='#CheckInputs'><p>Check Inputs</p></a></li>
<li><a href='#COAST'><p>COding-variant Allelic Series Test</p></a></li>
<li><a href='#Comparator'><p>Comparator Test</p></a></li>
<li><a href='#DGP'><p>Data Generating Process</p></a></li>
<li><a href='#FilterGenos'><p>Filter Noncausal Variants</p></a></li>
<li><a href='#GenAnno'><p>Generate Genotype Annotations</p></a></li>
<li><a href='#GenCovar'><p>Generate Covariates</p></a></li>
<li><a href='#GenGeno'><p>Generate Genotypes</p></a></li>
<li><a href='#GenGenoMat'><p>Generate Genotype Matrix</p></a></li>
<li><a href='#GenPheno'><p>Generate Phenotypes</p></a></li>
<li><a href='#OLS'><p>Ordinary Least Squares</p></a></li>
<li><a href='#ResidVar'><p>Calculate Residual Variance</p></a></li>
<li><a href='#Score'><p>Calculate Score Statistic</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Allelic Series Test</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.4.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Implementation of gene-level rare variant association tests targeting allelic series: genes where increasingly deleterious mutations have increasingly large phenotypic effects. The COding-variant Allelic Series Test (COAST) operates on the benign missense variants (BMVs), deleterious missense variants (DMVs), and protein truncating variants (PTVs) within a gene. COAST uses a set of adjustable weights that tailor the test towards rejecting the null hypothesis for genes where the average magnitude of effect increases monotonically from BMVs to DMVs to PTVs. See McCaw ZR, O’Dushlaine C, Somineni H, Bereket M, Klein C, Karaletsos T, Casale FP, Koller D, Soare TW. (2022) "An allelic series rare variant association test for candidate gene discovery" &lt;<a href="https://doi.org/10.1101%2F2022.12.23.521658">doi:10.1101/2022.12.23.521658</a>&gt;. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/BSD-3-Clause">BSD_3_clause</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, RNOmni, SKAT</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.0</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0), withr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-04 23:45:50 UTC; zmccaw</td>
</tr>
<tr>
<td>Author:</td>
<td>Zachary McCaw <a href="https://orcid.org/0000-0002-2006-9828"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Christoph Klein <a href="https://orcid.org/0000-0002-1748-625X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  insitro [cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zachary McCaw &lt;zmccaw@insitro.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-06 06:40:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='AllelicSeries-package'>AllelicSeries: Allelic Series Test</h2><span id='topic+AllelicSeries'></span><span id='topic+AllelicSeries-package'></span>

<h3>Description</h3>

<p>Implementation of gene-level rare variant association tests targeting allelic series: genes where increasingly deleterious mutations have increasingly large phenotypic effects. The COding-variant Allelic Series Test (COAST) operates on the benign missense variants (BMVs), deleterious missense variants (DMVs), and protein truncating variants (PTVs) within a gene. COAST uses a set of adjustable weights that tailor the test towards rejecting the null hypothesis for genes where the average magnitude of effect increases monotonically from BMVs to DMVs to PTVs. See McCaw ZR, O’Dushlaine C, Somineni H, Bereket M, Klein C, Karaletsos T, Casale FP, Koller D, Soare TW. (2022) &quot;An allelic series rare variant association test for candidate gene discovery&quot; <a href="https://doi.org/10.1101/2022.12.23.521658">doi:10.1101/2022.12.23.521658</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Zachary McCaw <a href="mailto:zmccaw@insitro.com">zmccaw@insitro.com</a> (<a href="https://orcid.org/0000-0002-2006-9828">ORCID</a>)
</p>
<p>Other contributors:
</p>

<ul>
<li><p> Christoph Klein <a href="mailto:cklein@insitro.com">cklein@insitro.com</a> (<a href="https://orcid.org/0000-0002-1748-625X">ORCID</a>) [contributor]
</p>
</li>
<li><p> insitro [copyright holder]
</p>
</li></ul>


<hr>
<h2 id='Aggregator'>Aggregator</h2><span id='topic+Aggregator'></span>

<h3>Description</h3>

<p>Aggregates genotypes within annotation categories.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Aggregator(
  anno,
  geno,
  drop_empty = TRUE,
  indicator = FALSE,
  method = "none",
  weights = DEFAULT_WEIGHTS
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Aggregator_+3A_anno">anno</code></td>
<td>
<p>(snps x 1) annotation vector with values in c(0, 1, 2).</p>
</td></tr>
<tr><td><code id="Aggregator_+3A_geno">geno</code></td>
<td>
<p>(n x snps) genotype matrix.</p>
</td></tr>
<tr><td><code id="Aggregator_+3A_drop_empty">drop_empty</code></td>
<td>
<p>Drop empty columns? Default: TRUE.</p>
</td></tr>
<tr><td><code id="Aggregator_+3A_indicator">indicator</code></td>
<td>
<p>Convert raw counts to indicators? Default: FALSE.</p>
</td></tr>
<tr><td><code id="Aggregator_+3A_method">method</code></td>
<td>
<p>Method for aggregating across categories:
&quot;none&quot;, &quot;max&quot;, &quot;sum&quot;. Default: &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="Aggregator_+3A_weights">weights</code></td>
<td>
<p>Annotation category weights.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>(n x 3) Numeric matrix without weighting, (n x 1) numeric matrix
with weighting.
</p>

<hr>
<h2 id='ASBT'>Allelic Series Burden Test</h2><span id='topic+ASBT'></span>

<h3>Description</h3>

<p>Burden test with allelic series weights.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ASBT(
  anno,
  geno,
  pheno,
  apply_int = TRUE,
  covar = NULL,
  indicator = FALSE,
  is_pheno_binary = FALSE,
  method = "none",
  score_test = FALSE,
  weights = DEFAULT_WEIGHTS
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ASBT_+3A_anno">anno</code></td>
<td>
<p>(snps x 1) annotation vector with values in c(0, 1, 2).</p>
</td></tr>
<tr><td><code id="ASBT_+3A_geno">geno</code></td>
<td>
<p>(n x snps) genotype matrix.</p>
</td></tr>
<tr><td><code id="ASBT_+3A_pheno">pheno</code></td>
<td>
<p>(n x 1) phenotype vector.</p>
</td></tr>
<tr><td><code id="ASBT_+3A_apply_int">apply_int</code></td>
<td>
<p>Apply rank-based inverse normal transform to the phenotype?
Default: TRUE. Ignored if phenotype is binary.</p>
</td></tr>
<tr><td><code id="ASBT_+3A_covar">covar</code></td>
<td>
<p>(n x p) covariate matrix. Defaults to an (n x 1) intercept.</p>
</td></tr>
<tr><td><code id="ASBT_+3A_indicator">indicator</code></td>
<td>
<p>Convert raw counts to indicators?</p>
</td></tr>
<tr><td><code id="ASBT_+3A_is_pheno_binary">is_pheno_binary</code></td>
<td>
<p>Is the phenotype binary? Default: FALSE.</p>
</td></tr>
<tr><td><code id="ASBT_+3A_method">method</code></td>
<td>
<p>Method for aggregating across categories: &quot;none&quot;, &quot;max&quot;,
&quot;sum&quot;. Default: &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="ASBT_+3A_score_test">score_test</code></td>
<td>
<p>Run a score test? If FALSE, performs a Wald test.</p>
</td></tr>
<tr><td><code id="ASBT_+3A_weights">weights</code></td>
<td>
<p>(3 x 1) annotation category weights.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric p-value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate data.
data &lt;- DGP(n = 1e3, snps = 1e2)

# Run the Allelic Series Burden Test.
# Note: the output is a scalar p-value.
results &lt;- ASBT(
  anno = data$anno,
  geno = data$geno,
  pheno = data$pheno,
  covar = data$covar
)
</code></pre>

<hr>
<h2 id='ASKAT'>Allelic Series SKAT Test</h2><span id='topic+ASKAT'></span>

<h3>Description</h3>

<p>Sequence kernel association test (SKAT) with allelic series weights.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ASKAT(
  anno,
  geno,
  pheno,
  apply_int = TRUE,
  covar = NULL,
  is_pheno_binary = FALSE,
  return_null_model = FALSE,
  weights = DEFAULT_WEIGHTS
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ASKAT_+3A_anno">anno</code></td>
<td>
<p>(snps x 1) annotation vector with values in c(0, 1, 2).</p>
</td></tr>
<tr><td><code id="ASKAT_+3A_geno">geno</code></td>
<td>
<p>(n x snps) genotype matrix.</p>
</td></tr>
<tr><td><code id="ASKAT_+3A_pheno">pheno</code></td>
<td>
<p>(n x 1) phenotype vector.</p>
</td></tr>
<tr><td><code id="ASKAT_+3A_apply_int">apply_int</code></td>
<td>
<p>Apply rank-based inverse normal transform to the phenotype?
Default: TRUE. Ignored if phenotype is binary.</p>
</td></tr>
<tr><td><code id="ASKAT_+3A_covar">covar</code></td>
<td>
<p>(n x p) covariate matrix. Defaults to an (n x 1) intercept.</p>
</td></tr>
<tr><td><code id="ASKAT_+3A_is_pheno_binary">is_pheno_binary</code></td>
<td>
<p>Is the phenotype binary? Default: FALSE.</p>
</td></tr>
<tr><td><code id="ASKAT_+3A_return_null_model">return_null_model</code></td>
<td>
<p>Return the null model in addition to the p-value?
Useful if running additional SKAT tests. Default: FALSE.</p>
</td></tr>
<tr><td><code id="ASKAT_+3A_weights">weights</code></td>
<td>
<p>(3 x 1) annotation category weights.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>return_null_model</code>, a list containing the p-value and the
SKAT null model. Otherwise, a numeric p-value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate data.
data &lt;- DGP(n = 1e3, snps = 1e2)

# Run the Allelic Series SKAT Test.
# Note: the output is a scalar p-value.
results &lt;- ASKAT(
  anno = data$anno,
  geno = data$geno,
  pheno = data$pheno,
  covar = data$covar
)
</code></pre>

<hr>
<h2 id='CalcRegParam'>Calculate Regression Parameters</h2><span id='topic+CalcRegParam'></span>

<h3>Description</h3>

<p>Calculate phenotypic regression coefficients and the residual variation
based on proportion of variation explained (PVE) by each factor. Note that
the proportion of variation explained by genotype is required, but genetic
effects are not generated here.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CalcRegParam(pve_age = 0.1, pve_pcs = 0.2, pve_sex = 0.1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CalcRegParam_+3A_pve_age">pve_age</code></td>
<td>
<p>PVE by age.</p>
</td></tr>
<tr><td><code id="CalcRegParam_+3A_pve_pcs">pve_pcs</code></td>
<td>
<p>PVE by PCs (collectively).</p>
</td></tr>
<tr><td><code id="CalcRegParam_+3A_pve_sex">pve_sex</code></td>
<td>
<p>PVE by sex.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List containing the (5 x 1) regression coefficient vector &quot;coef&quot; and
the residual standard deviation &quot;sd&quot;.
</p>

<hr>
<h2 id='CheckInputs'>Check Inputs</h2><span id='topic+CheckInputs'></span>

<h3>Description</h3>

<p>Check Inputs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckInputs(anno, covar, geno, is_pheno_binary, pheno, weights)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CheckInputs_+3A_anno">anno</code></td>
<td>
<p>(snps x 1) annotation vector.</p>
</td></tr>
<tr><td><code id="CheckInputs_+3A_covar">covar</code></td>
<td>
<p>(n x p) covariate matrix.</p>
</td></tr>
<tr><td><code id="CheckInputs_+3A_geno">geno</code></td>
<td>
<p>(n x snps) genotype matrix.</p>
</td></tr>
<tr><td><code id="CheckInputs_+3A_is_pheno_binary">is_pheno_binary</code></td>
<td>
<p>Is the phenotype binary?</p>
</td></tr>
<tr><td><code id="CheckInputs_+3A_pheno">pheno</code></td>
<td>
<p>(n x 1) phenotype vector.</p>
</td></tr>
<tr><td><code id="CheckInputs_+3A_weights">weights</code></td>
<td>
<p>(3 x 1) annotation category weights.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None.
</p>

<hr>
<h2 id='COAST'>COding-variant Allelic Series Test</h2><span id='topic+COAST'></span>

<h3>Description</h3>

<p>Main allelic series test. Performs both Burden and SKAT type tests, then
combines the results to calculate an omnibus p-value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>COAST(
  anno,
  geno,
  pheno,
  apply_int = TRUE,
  covar = NULL,
  include_orig_skato_all = FALSE,
  include_orig_skato_ptv = FALSE,
  is_pheno_binary = FALSE,
  return_omni_only = FALSE,
  score_test = FALSE,
  weights = DEFAULT_WEIGHTS
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="COAST_+3A_anno">anno</code></td>
<td>
<p>(snps x 1) annotation vector with values in c(0, 1, 2).</p>
</td></tr>
<tr><td><code id="COAST_+3A_geno">geno</code></td>
<td>
<p>(n x snps) genotype matrix.</p>
</td></tr>
<tr><td><code id="COAST_+3A_pheno">pheno</code></td>
<td>
<p>(n x 1) phenotype vector.</p>
</td></tr>
<tr><td><code id="COAST_+3A_apply_int">apply_int</code></td>
<td>
<p>Apply rank-based inverse normal transform to the phenotype?
Default: TRUE. Ignored if phenotype is binary.</p>
</td></tr>
<tr><td><code id="COAST_+3A_covar">covar</code></td>
<td>
<p>(n x p) covariate matrix. Defaults to an (n x 1) intercept.</p>
</td></tr>
<tr><td><code id="COAST_+3A_include_orig_skato_all">include_orig_skato_all</code></td>
<td>
<p>Include the original version of SKAT-O applied
to all variants in the omnibus test? Default: FALSE.</p>
</td></tr>
<tr><td><code id="COAST_+3A_include_orig_skato_ptv">include_orig_skato_ptv</code></td>
<td>
<p>Include the original version of SKAT-O applied
to PTV variants only in the omnibus test? Default: FALSE.</p>
</td></tr>
<tr><td><code id="COAST_+3A_is_pheno_binary">is_pheno_binary</code></td>
<td>
<p>Is the phenotype binary? Default: FALSE.</p>
</td></tr>
<tr><td><code id="COAST_+3A_return_omni_only">return_omni_only</code></td>
<td>
<p>Return only the omnibus p-value? Default: FALSE.</p>
</td></tr>
<tr><td><code id="COAST_+3A_score_test">score_test</code></td>
<td>
<p>Use a score test for burden analysis? If FALSE, uses a
Wald test.</p>
</td></tr>
<tr><td><code id="COAST_+3A_weights">weights</code></td>
<td>
<p>(3 x 1) annotation category weights.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric p-value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate data.
data &lt;- DGP(n = 1e3, snps = 1e2)

# Run the COding-variant Allelic Series Test.
results &lt;- COAST(
  anno = data$anno,
  geno = data$geno,
  pheno = data$pheno,
  covar = data$covar
)
show(results)

</code></pre>

<hr>
<h2 id='Comparator'>Comparator Test</h2><span id='topic+Comparator'></span>

<h3>Description</h3>

<p>Runs burden, SKAT, and SKAT-O, using default settings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Comparator(covar, geno, pheno, apply_int = TRUE, is_pheno_binary = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Comparator_+3A_covar">covar</code></td>
<td>
<p>(n x p) covariate matrix.</p>
</td></tr>
<tr><td><code id="Comparator_+3A_geno">geno</code></td>
<td>
<p>(n x snps) genotype matrix.</p>
</td></tr>
<tr><td><code id="Comparator_+3A_pheno">pheno</code></td>
<td>
<p>(n x 1) phenotype vector.</p>
</td></tr>
<tr><td><code id="Comparator_+3A_apply_int">apply_int</code></td>
<td>
<p>Apply rank-based inverse normal transform to the phenotype?
Default: TRUE. Ignored if phenotype is binary.</p>
</td></tr>
<tr><td><code id="Comparator_+3A_is_pheno_binary">is_pheno_binary</code></td>
<td>
<p>Is the phenotype binary? Default: FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric vector of p-values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate data.
data &lt;- DGP(n = 1e3, snps = 1e2)

# Run the comparators.
results &lt;- Comparator(
  geno = data$geno,
  pheno = data$pheno,
  covar = data$covar
)
</code></pre>

<hr>
<h2 id='DGP'>Data Generating Process</h2><span id='topic+DGP'></span>

<h3>Description</h3>

<p>Generate a data set consisting of: </p>

<ul>
<li><p>&quot;anno&quot;A SNP-length annotation vector.
</p>
</li>
<li><p>&quot;covar&quot;A subject by 6 covariate matrix.
</p>
</li>
<li><p>&quot;geno&quot;A subject by SNP genotype matrix.
</p>
</li>
<li><p>&quot;pheno&quot;A subject-length phenotype vector.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>DGP(
  anno = NULL,
  beta = c(0, 1, 2),
  binary = FALSE,
  geno = NULL,
  include_residual = TRUE,
  indicator = FALSE,
  maf_range = c(0.005, 0.01),
  method = "none",
  n = 100,
  p_dmv = 0.4,
  p_ptv = 0.1,
  prop_causal = 1,
  random_signs = FALSE,
  random_var = 0,
  snps = 100,
  weights = c(1, 2, 3)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DGP_+3A_anno">anno</code></td>
<td>
<p>Annotation vector, if providing genotypes. Should match the
number of columns in geno.</p>
</td></tr>
<tr><td><code id="DGP_+3A_beta">beta</code></td>
<td>
<p>If method = &quot;none&quot;, a (3 x 1) coefficient vector for bmvs, dmvs,
and ptvs respectively. If method != &quot;none&quot;, a scalar effect size.</p>
</td></tr>
<tr><td><code id="DGP_+3A_binary">binary</code></td>
<td>
<p>Generate binary phenotype? Default: FALSE.</p>
</td></tr>
<tr><td><code id="DGP_+3A_geno">geno</code></td>
<td>
<p>Genotype matrix, if providing genotypes.</p>
</td></tr>
<tr><td><code id="DGP_+3A_include_residual">include_residual</code></td>
<td>
<p>Include residual? If FALSE, returns the expected
value. Intended for testing.</p>
</td></tr>
<tr><td><code id="DGP_+3A_indicator">indicator</code></td>
<td>
<p>Convert raw counts to indicators? Default: FALSE.</p>
</td></tr>
<tr><td><code id="DGP_+3A_maf_range">maf_range</code></td>
<td>
<p>Range of minor allele frequencies: c(MIN, MAX).</p>
</td></tr>
<tr><td><code id="DGP_+3A_method">method</code></td>
<td>
<p>Genotype aggregation method. Default: &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="DGP_+3A_n">n</code></td>
<td>
<p>Sample size.</p>
</td></tr>
<tr><td><code id="DGP_+3A_p_dmv">p_dmv</code></td>
<td>
<p>Frequency of deleterious missense variants. Default of 40% is
based on the frequency of DMVs among rare coding variants in the UK
Biobank.</p>
</td></tr>
<tr><td><code id="DGP_+3A_p_ptv">p_ptv</code></td>
<td>
<p>Frequency of protein truncating variants. Default of 10% is
based on the frequency of PTVs among rare coding variants in the UK
Biobank.</p>
</td></tr>
<tr><td><code id="DGP_+3A_prop_causal">prop_causal</code></td>
<td>
<p>Proportion of variants which are causal. Default: 1.0.</p>
</td></tr>
<tr><td><code id="DGP_+3A_random_signs">random_signs</code></td>
<td>
<p>Randomize signs? FALSE for burden-type genetic
architecture, TRUE for SKAT-type.</p>
</td></tr>
<tr><td><code id="DGP_+3A_random_var">random_var</code></td>
<td>
<p>Frailty variance in the case of random signs. Default: 0.</p>
</td></tr>
<tr><td><code id="DGP_+3A_snps">snps</code></td>
<td>
<p>Number of SNP in the gene. Default: 100.</p>
</td></tr>
<tr><td><code id="DGP_+3A_weights">weights</code></td>
<td>
<p>Aggregation weights.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List containing: genotypes, annotations, covariates, phenotypes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate data.
data &lt;- DGP(n = 100)

# View components.
table(data$anno)
head(data$covar)
head(data$geno[, 1:5])
hist(data$pheno)
</code></pre>

<hr>
<h2 id='FilterGenos'>Filter Noncausal Variants</h2><span id='topic+FilterGenos'></span>

<h3>Description</h3>

<p>Remove a random fraction of variants, which are designated non-causal.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FilterGenos(anno, geno, prop_causal = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FilterGenos_+3A_anno">anno</code></td>
<td>
<p>(snps x 1) annotation vector.</p>
</td></tr>
<tr><td><code id="FilterGenos_+3A_geno">geno</code></td>
<td>
<p>(n x snps) genotype matrix.</p>
</td></tr>
<tr><td><code id="FilterGenos_+3A_prop_causal">prop_causal</code></td>
<td>
<p>Proportion of variants which are causal.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List containing the (n x snps) genotype matrix &quot;geno&quot; and the
(snps x 1) annotation vector &quot;anno&quot;.
</p>

<hr>
<h2 id='GenAnno'>Generate Genotype Annotations</h2><span id='topic+GenAnno'></span>

<h3>Description</h3>

<p>Returns a vector of length = the number of columns (SNPs) in the genotype
matrix. Each SNP is classified as a benign missense variant (0), a
deleterious missense variant (1), or a protein truncating variant (2).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GenAnno(snps, p_dmv = 0.33, p_ptv = 0.33)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GenAnno_+3A_snps">snps</code></td>
<td>
<p>Number of SNPs in the gene.</p>
</td></tr>
<tr><td><code id="GenAnno_+3A_p_dmv">p_dmv</code></td>
<td>
<p>Frequency of deleterious missense variants.</p>
</td></tr>
<tr><td><code id="GenAnno_+3A_p_ptv">p_ptv</code></td>
<td>
<p>Frequency of protein truncating variants.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>(snps x 1) integer vector.
</p>

<hr>
<h2 id='GenCovar'>Generate Covariates</h2><span id='topic+GenCovar'></span>

<h3>Description</h3>

<p>Generate an (n x 6) covariate matrix with columns representing an intercept,
age, sex, and 3 genetic PCs. Because these simulations address rare variant
analysis, correlation between genotypes and the genetic PCs (based on
common variants) is unnecessary.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GenCovar(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GenCovar_+3A_n">n</code></td>
<td>
<p>Sample size.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>(n x 6) numeric matrix.
</p>

<hr>
<h2 id='GenGeno'>Generate Genotypes</h2><span id='topic+GenGeno'></span>

<h3>Description</h3>

<p>Generate Genotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GenGeno(n, snps, maf_range = c(0.005, 0.01), p_dmv = 0.33, p_ptv = 0.33)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GenGeno_+3A_n">n</code></td>
<td>
<p>Sample size.</p>
</td></tr>
<tr><td><code id="GenGeno_+3A_snps">snps</code></td>
<td>
<p>Number of SNP in the gene.</p>
</td></tr>
<tr><td><code id="GenGeno_+3A_maf_range">maf_range</code></td>
<td>
<p>Range of minor allele frequencies: c(MIN, MAX).</p>
</td></tr>
<tr><td><code id="GenGeno_+3A_p_dmv">p_dmv</code></td>
<td>
<p>Frequency of deleterious missense variants.</p>
</td></tr>
<tr><td><code id="GenGeno_+3A_p_ptv">p_ptv</code></td>
<td>
<p>Frequency of protein truncating variants.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List containing the (n x snps) genotype matrix &quot;geno&quot; and the
(snps x 1) annotation vector &quot;anno&quot;.
</p>

<hr>
<h2 id='GenGenoMat'>Generate Genotype Matrix</h2><span id='topic+GenGenoMat'></span>

<h3>Description</h3>

<p>Generate Genotype Matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GenGenoMat(n, snps, maf_range = c(0.005, 0.01))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GenGenoMat_+3A_n">n</code></td>
<td>
<p>Sample size.</p>
</td></tr>
<tr><td><code id="GenGenoMat_+3A_snps">snps</code></td>
<td>
<p>Number of SNP in the gene.</p>
</td></tr>
<tr><td><code id="GenGenoMat_+3A_maf_range">maf_range</code></td>
<td>
<p>Range of minor allele frequencies: c(MIN, MAX).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>(n x snps) numeric matrix.
</p>

<hr>
<h2 id='GenPheno'>Generate Phenotypes</h2><span id='topic+GenPheno'></span>

<h3>Description</h3>

<p>Generate Phenotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GenPheno(
  anno,
  beta,
  covar,
  geno,
  reg_param,
  binary = FALSE,
  include_residual = TRUE,
  indicator = FALSE,
  method = "none",
  prop_causal = 1,
  random_signs = FALSE,
  random_var = 0,
  weights = c(0, 1, 2)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GenPheno_+3A_anno">anno</code></td>
<td>
<p>(snps x 1) annotation vector.</p>
</td></tr>
<tr><td><code id="GenPheno_+3A_beta">beta</code></td>
<td>
<p>(3 x 1) coefficient vector for bmvs, dmvs, and ptvs respectively.</p>
</td></tr>
<tr><td><code id="GenPheno_+3A_covar">covar</code></td>
<td>
<p>Covariate matrix.</p>
</td></tr>
<tr><td><code id="GenPheno_+3A_geno">geno</code></td>
<td>
<p>(n x snps) genotype matrix.</p>
</td></tr>
<tr><td><code id="GenPheno_+3A_reg_param">reg_param</code></td>
<td>
<p>Regression parameters.</p>
</td></tr>
<tr><td><code id="GenPheno_+3A_binary">binary</code></td>
<td>
<p>Generate binary phenotype? Default: FALSE.</p>
</td></tr>
<tr><td><code id="GenPheno_+3A_include_residual">include_residual</code></td>
<td>
<p>Include residual? If FALSE, returns the expected
value. Intended for testing.</p>
</td></tr>
<tr><td><code id="GenPheno_+3A_indicator">indicator</code></td>
<td>
<p>Convert raw counts to indicators? Default: FALSE.</p>
</td></tr>
<tr><td><code id="GenPheno_+3A_method">method</code></td>
<td>
<p>Genotype aggregation method. Default: &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="GenPheno_+3A_prop_causal">prop_causal</code></td>
<td>
<p>Proportion of variants which are causal.</p>
</td></tr>
<tr><td><code id="GenPheno_+3A_random_signs">random_signs</code></td>
<td>
<p>Randomize signs? FALSE for burden-type genetic
architecture, TRUE for SKAT-type.</p>
</td></tr>
<tr><td><code id="GenPheno_+3A_random_var">random_var</code></td>
<td>
<p>Frailty variance in the case of random signs. Default: 0.</p>
</td></tr>
<tr><td><code id="GenPheno_+3A_weights">weights</code></td>
<td>
<p>Aggregation weights.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>(n x 1) numeric vector.
</p>

<hr>
<h2 id='OLS'>Ordinary Least Squares</h2><span id='topic+OLS'></span>

<h3>Description</h3>

<p>Fits the standard OLS model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OLS(y, X)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OLS_+3A_y">y</code></td>
<td>
<p>(n x 1) Numeric vector.</p>
</td></tr>
<tr><td><code id="OLS_+3A_x">X</code></td>
<td>
<p>(n x p) Numeric matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List containing the following:
</p>

<ul>
<li><p>BetaRegression coefficient.
</p>
</li>
<li><p>VOutcome variance.
</p>
</li>
<li><p>SEStandard errors.
</p>
</li>
<li><p>ZZ-scores.
</p>
</li></ul>


<hr>
<h2 id='ResidVar'>Calculate Residual Variance</h2><span id='topic+ResidVar'></span>

<h3>Description</h3>

<p>Calculate Residual Variance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ResidVar(y, X)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ResidVar_+3A_y">y</code></td>
<td>
<p>(n x 1) Numeric phenotype vector.</p>
</td></tr>
<tr><td><code id="ResidVar_+3A_x">X</code></td>
<td>
<p>(n x q) Numeric covariate matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Scalar residual variance.
</p>

<hr>
<h2 id='Score'>Calculate Score Statistic</h2><span id='topic+Score'></span>

<h3>Description</h3>

<p>Calculate Score Statistic
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Score(y, G, X, v)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Score_+3A_y">y</code></td>
<td>
<p>(n x 1) Numeric phenotype vector.</p>
</td></tr>
<tr><td><code id="Score_+3A_g">G</code></td>
<td>
<p>(n x p) Numeric genotype matrix.</p>
</td></tr>
<tr><td><code id="Score_+3A_x">X</code></td>
<td>
<p>(n x q) Numeric covariate matrix.</p>
</td></tr>
<tr><td><code id="Score_+3A_v">v</code></td>
<td>
<p>Scalar residual variance.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Scalar score statistic.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
