<!DOCTYPE html><html><head><title>Help for package allelematch</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {allelematch}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#allelematch-package'>
<p>Identification of unique multilocus genotypes</p></a></li>
<li><a href='#amAlleleFreq'>
<p>Determine allele frequencies</p></a></li>
<li><a href='#amCluster'>
<p>Clustering of multilocus genotypes</p></a></li>
<li><a href='#amCSSForHTML'>
<p>Produce cascading style sheet (CSS) for HTML</p></a></li>
<li><a href='#amDataset'>
<p>Prepare a dataset for use with allelematch</p></a></li>
<li><a href='#amExampleData'>
<p>Data sets to support the tutorials in the supplementary documentation and examples in the manual</p></a></li>
<li><a href='#amMatrix'>
<p>Produce a dissimilarity matrix for pairs of multilocus genotypes</p></a></li>
<li><a href='#amPairwise'>
<p>Pairwise matching of multilocus genotypes</p></a></li>
<li><a href='#amUnique'>
<p>Identification of unique genotypes</p></a></li>
<li><a href='#amUniqueProfile'>
<p>Determine optimal parameter values for the identification of unique genotypes</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Identifying Unique Multilocus Genotypes where Genotyping Error
and Missing Data may be Present</td>
</tr>
<tr>
<td>Version:</td>
<td>2.5.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-08-17</td>
</tr>
<tr>
<td>Author:</td>
<td>Paul Galpern &lt;pgalpern@gmail.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Todd Cross &lt;todd.cross@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Tools for the identification of unique of multilocus genotypes when both genotyping error and missing data may be present; targeted for use with large datasets and databases containing multiple samples of each individual (a common situation in conservation genetics, particularly in non-invasive wildlife sampling applications). Functions explicitly incorporate missing data and can tolerate allele mismatches created by genotyping error. If you use this package, please cite the original publication in Molecular Ecology Resources (Galpern et al., 2012), the details for which can be generated using citation('allelematch'). For a complete vignette, please access via the Data S1 Supplementary documentation and tutorials (PDF) located at &lt;<a href="https://doi.org/10.1111%2Fj.1755-0998.2012.03137.x">doi:10.1111/j.1755-0998.2012.03137.x</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>dynamicTreeCut</td>
</tr>
<tr>
<td>URL:</td>
<td>&lt;doi:10.1111%2Fj.1755-0998.2012.03137.x&gt;</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-24 14:17:35 UTC; toddbcross</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-24 14:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='allelematch-package'>
Identification of unique multilocus genotypes
</h2><span id='topic+allelematch-package'></span><span id='topic+allelematch-tutorial'></span><span id='topic+allelematch'></span>

<h3>Description</h3>

<p>Tools for the identification of unique of multilocus genotypes when both genotyping error and 
missing data may be present; targeted for use with large datasets and databases containing
multiple samples of each individual (a common situation in conservation genetics, particularly
in non-invasive wildlife sampling applications). Functions explicitly incorporate missing data
and can tolerate allele mismatches created by genotyping error.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
		Package: </td><td style="text-align: left;"> allelematch</td>
</tr>
<tr>
 <td style="text-align: left;">
		Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
		Version: </td><td style="text-align: left;"> 2.5.2</td>
</tr>
<tr>
 <td style="text-align: left;">
		Date: </td><td style="text-align: left;"> 2023-05-18</td>
</tr>
<tr>
 <td style="text-align: left;">
		License: </td><td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
 <td style="text-align: left;">
		Requires: </td><td style="text-align: left;"> dynamicTreeCut</td>
</tr>
<tr>
 <td style="text-align: left;">
		LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
	</td>
</tr>

</table>

<p>Supplementary documentation describing the operation of the software in detail and illustrating the
use of the software using tutorials is available as a vignette.
It is installed with the package and linked from the package index help page. An online version is
also available via the Data S1 Supplementary documentation and tutorials (PDF) located at
<a href="https://doi.org/10.1111/j.1755-0998.2012.03137.x">doi:10.1111/j.1755-0998.2012.03137.x</a>.<br />
</p>
<p>Simulations examining the performance of these tools have also been performed, and results are
available in the publication associated with this package. Please refer to the publication:<br />
</p>
<p>Galpern P, Manseau, M, Hettinga P, Smith K, and Wilson P. (2012) allelematch: an R package for
identifying unique multilocus genotypes where genotype error and missing data may be present.
<em>Molecular Ecology Resources</em> 12:771-778.<br />
</p>
<p>Use <code>citation("allelematch")</code> for the full citation. Please also use this publication when
citing the package.<br />
</p>
<p>An important core element of the package is dynamic tree cutting, and this is made possible via the
cutreeHybrid function within the dynamicTreeCut package for R (Langfelder et al., 2008). 
</p>


<h3>Author(s)</h3>

<p>Paul Galpern (<a href="mailto:pgalpern@gmail.com">pgalpern@gmail.com</a>)
</p>


<h3>References</h3>

<p>Langfelder P, Zhang B, Horvath S. (2008) Defining clusters from a hierarchical cluster tree: the
Dynamic Tree Cut package for R. <em>Bioinformatics</em>, 24, 719
</p>

<hr>
<h2 id='amAlleleFreq'>
Determine allele frequencies
</h2><span id='topic+amAlleleFreq'></span><span id='topic+print.amAlleleFreq'></span>

<h3>Description</h3>

<p>Determines the allele frequencies for each locus in a multilocus genetic dataset by 
first removing missing observations. It requires an <code>amDataset</code> object and a map 
vector relating each column of the dataset to a genetic locus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	amAlleleFreq(
		amDatasetFocal, 
		multilocusMap = NULL
		)
		
	## S3 method for class 'amAlleleFreq'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="amAlleleFreq_+3A_amdatasetfocal">amDatasetFocal</code></td>
<td>

<p>An <code>amDataset</code> object
</p>
</td></tr>
<tr><td><code id="amAlleleFreq_+3A_multilocusmap">multilocusMap</code></td>
<td>

<p>Optionally, a vector of integers or strings giving the mappings onto loci for all 
genotype columns in <code>amDatasetFocal</code>.<br />
When omitted, columns are assumed to be paired (i.e., diploid loci with alleles in 
adjacent columns).<br />
See details.
</p>
</td></tr>
<tr><td><code id="amAlleleFreq_+3A_x">x</code></td>
<td>

<p>An <code>amDataset</code> object.
</p>
</td></tr>
<tr><td><code id="amAlleleFreq_+3A_...">...</code></td>
<td>

<p>Additional arguments to summary
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is called by <code>amUnique</code>.<br />
</p>
<p><code>multilocusMap</code> is often not required, as <code>amDataset</code> objects will typically 
consist of paired columns of genotypes, where each pair is a separate locus. In cases 
where this is not the case (e.g., gender is given in only one column), a map vector 
must be specified.<br />
</p>
<p>Example: <code>amDataset</code> consists of gender followed by 4 diploid loci in paired 
columns<br />
<code>multilocusMap = c(1, 2, 2, 3, 3, 4, 4, 5, 5)</code><br />
or equally<br />
<code>multilocusMap=c("GENDER", "LOC1", "LOC1", "LOC2", "LOC2", "LOC3", "LOC4", 
	"LOC4")</code>
</p>


<h3>Value</h3>

<p>An <code>amAlleleFreq</code> object
</p>


<h3>Author(s)</h3>

<p>Paul Galpern (<a href="mailto:pgalpern@gmail.com">pgalpern@gmail.com</a>)
</p>


<h3>References</h3>

<p>For more information and for a complete vignette, please access via the Data S1	
Supplementary documentation and tutorials (PDF) located at
&lt;doi:10.1111/j.1755-0998.2012.03137.x&gt;.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+amUnique">amUnique</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	## Not run: 
	data("amExample5")
	
	## Produce amDataset object
	myDataset1 &lt;- 
		amDataset(
			amExample5, 
			missingCode = "-99", 
			indexColumn = 1, 
			metaDataColumn = 2
			)
	
	## Usage
	myAlleleFreq &lt;- 
		amAlleleFreq(
			myDataset1, 
			multilocusMap = c(1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7, 7, 8, 8, 9, 9, 10, 10, 
			11, 11)
			)
	
	## Produce amDataset object, but remove gender column
	myDataset2 &lt;- 
		amDataset(
			amExample5, 
			missingCode = "-99", 
			indexColumn = 1,
			metaDataColumn = 2, 
			ignoreColumn = "gender"
			)
	
	## Because all columns are paired, usage is simpler
	myAlleleFreq &lt;- amAlleleFreq(myDataset2)
	
## End(Not run)
</code></pre>

<hr>
<h2 id='amCluster'>
Clustering of multilocus genotypes
</h2><span id='topic+amCluster'></span><span id='topic+summary.amCluster'></span><span id='topic+amHTML.amCluster'></span><span id='topic+amCSV.amCluster'></span>

<h3>Description</h3>

<p>Performs clustering of multilocus genotypes to identify unique consensus and singleton genotypes
and generates analysis output in formatted text, HTML, or CSV. These functions are usually
called by <code><a href="#topic+amUnique">amUnique</a></code>. This interface remains to enable a better understanding of how
<code><a href="#topic+amUnique">amUnique</a></code> operates. For more information see example.<br />
</p>
<p>There are three steps to this analysis: (1) identify the dissimilarity between pairs of
genotypes using a metric which takes missing data into account, (2) cluster this dissimilarity
matrix using a standard hierarchical agglomerative clustering approach, and (3) use a dynamic
tree cutting approach to identify clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	amCluster(
		amDatasetFocal, 
		runUntilSingletons = TRUE, 
		cutHeight = 0.3, 
		missingMethod = 2, 
		consensusMethod = 1, 
		clusterMethod = "complete"
		)

	amHTML.amCluster(
		x, 
		htmlFile = NULL, 
		htmlCSS = amCSSForHTML()
		)

	amCSV.amCluster(
		x, 
		csvFile
		)

	## S3 method for class 'amCluster'
summary(object, html = NULL, csv = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="amCluster_+3A_amdatasetfocal">amDatasetFocal</code></td>
<td>

<p>An <code>amDataset</code> object containing genotypes to cluster.
</p>
</td></tr>
<tr><td><code id="amCluster_+3A_rununtilsingletons">runUntilSingletons</code></td>
<td>

<p>When <code>runUntilSingletons = TRUE</code>, the analysis runs recursively with the unique
individuals determined in one analysis feeding into the next until no more clusters are
formed; applicable when the goal is to thin a dataset to unique genotypes.<br />
For more manual control over the process, use <code>runUntilSingletons = FALSE</code>.<br />
See details and examples.
</p>
</td></tr>
<tr><td><code id="amCluster_+3A_cutheight">cutHeight</code></td>
<td>

<p>Sets the tree cutting height using the hybrid method in the <code>dynamicTreeCut</code>
package.<br />
See details and <code>cutreeHybrid</code> for more information.
</p>
</td></tr>
<tr><td><code id="amCluster_+3A_missingmethod">missingMethod</code></td>
<td>

<p>The method used to determine the similarity of multilocus genotypes when data is missing.<br />
The default, (<code>missingMethod = 2</code>), is preferable in all cases.<br />
See <code><a href="#topic+amMatrix">amMatrix</a></code>.
</p>
</td></tr>
<tr><td><code id="amCluster_+3A_consensusmethod">consensusMethod</code></td>
<td>

<p>The method (an integer) used to determine the consensus multilocus genotype from a cluster
of multilocus genotypes.<br />
See details.
</p>
</td></tr>
<tr><td><code id="amCluster_+3A_clustermethod">clusterMethod</code></td>
<td>

<p>The method used by <code><a href="stats.html#topic+hclust">hclust</a></code> for clustering.<br />
Only the default <code>clusterMethod = "complete"</code> performs acceptably in simulations.<br />
This option remains for experimental reasons.
</p>
</td></tr>
<tr><td><code id="amCluster_+3A_object">object</code>, <code id="amCluster_+3A_x">x</code></td>
<td>

<p>An <code>amPairwise</code> object.
</p>
</td></tr>
<tr><td><code id="amCluster_+3A_htmlfile">htmlFile</code></td>
<td>

<p>HTML filepath to create.<br />
If <code>htmlFile = NULL</code>, a file is created in the operating system temporary directory and
is then opened in the default browser.
</p>
</td></tr>
<tr><td><code id="amCluster_+3A_htmlcss">htmlCSS</code></td>
<td>

<p>String containing a valid cascading style sheet.<br />
A default style sheet is provided in <code><a href="#topic+amCSSForHTML">amCSSForHTML</a></code>.<br />
See <code><a href="#topic+amCSSForHTML">amCSSForHTML</a></code> for details of how to tweak this CSS.
</p>
</td></tr>
<tr><td><code id="amCluster_+3A_html">html</code></td>
<td>

<p>If <code>html = NULL</code> or <code>html=FALSE</code>, formatted textual output is displayed on the
console.<br />
If <code>html = TRUE</code>, the <code>summary</code> method produces and loads an HTML file in the
default browser.<br />
<code>html</code> can also contain a path to a file where HTML output will be written.
</p>
</td></tr>
<tr><td><code id="amCluster_+3A_csvfile">csvFile</code>, <code id="amCluster_+3A_csv">csv</code></td>
<td>

<p>CSV filepath to create containing only the unique genotypes determined in the clustering.
</p>
</td></tr>
<tr><td><code id="amCluster_+3A_...">...</code></td>
<td>

<p>Additional arguments to <code>summary</code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Selecting an appropriate <code>cutHeight</code> parameter (also known as the d-hat criterion) is
essential. Typically this function is called from <code>amUnique</code>, and the conversion between
alleleMismatch (m-hat) and cutHeight (d-hat) will be done automatically. Selecting an
appropriate value for alleleMismatch (m-hat) can be done using <code>amUniqueProfile</code>. See the
supplementary documentation for an explanation of how these parameters are related.<br />
</p>
<p><code>runUntilSingletons=TRUE</code> provides an efficient and reliable way to determine the unique
individuals in a dataset if the dataset meets certain criteria. To understand how the clustering
is thinning the dataset run this recursion manually using <code>runUntilSingletons=FALSE</code>. An
example is provided below.<br />
</p>
<p><code>cutHeight</code> in practice gives the amount of dissimilarity (using the metric described in
<code><a href="#topic+amMatrix">amMatrix</a></code>) required for two multilocus genotypes to be declared different (also
known as d-hat). The default setting for <code>consensusMethod</code> performs well.
</p>

<table>
<tr>
 <td style="text-align: left;">
	<code>consensusMethod</code></td>
</tr>
<tr>
 <td style="text-align: left;">
		<code>1</code> </td><td style="text-align: left;"> Genotype with max similarity to others in the cluster is consensus
		(DEFAULT)</td>
</tr>
<tr>
 <td style="text-align: left;">
		<code>2</code> </td><td style="text-align: left;"> Genotype with max similarity to others in the cluster is consensus then
		interpolate missing alleles using mode non-missing allele in each column</td>
</tr>
<tr>
 <td style="text-align: left;">
		<code>3</code> </td><td style="text-align: left;"> Genotype with min missing data is consensus</td>
</tr>
<tr>
 <td style="text-align: left;">
		<code>4</code> </td><td style="text-align: left;"> Genotype with min missing data is consensus then interpolate missing alleles
		using mode non-missing allele in each column</td>
</tr>
<tr>
 <td style="text-align: left;">
	</td>
</tr>

</table>



<h3>Value</h3>

<p><code>amCluster</code> object or side effects: analysis summary written to an HTML file or to the
console, or written to a CSV file.
</p>


<h3>Note</h3>

<p>There is an additional side effect of <code>html = TRUE</code> (or of <code>htmlFile = NULL</code>). If
required, there is a clean up of the operating system temporary directory where AlleleMatch
temporary HTML files are stored. Files that match the pattern am*.html and are older than 24
hours are deleted from this temporary directory.
</p>


<h3>Author(s)</h3>

<p>Paul Galpern (<a href="mailto:pgalpern@gmail.com">pgalpern@gmail.com</a>)
</p>


<h3>References</h3>

<p>For a complete vignette, please access via the Data S1 Supplementary documentation and tutorials
(PDF) located at &lt;doi:10.1111/j.1755-0998.2012.03137.x&gt;.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+amDataset">amDataset</a></code>, <code><a href="#topic+amMatrix">amMatrix</a></code>, <code><a href="#topic+amPairwise">amPairwise</a></code>,
<code><a href="#topic+amUnique">amUnique</a></code>, <code><a href="#topic+amUniqueProfile">amUniqueProfile</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	## Not run: 
	data("amExample5")

	## Produce amDataset object
	myDataset &lt;- 
		amDataset(
			amExample5, 
			missingCode = "-99", 
			indexColumn = 1, 
			metaDataColumn = 2, 
			ignoreColumn = "gender"
			)

	## Usage
	myCluster &lt;- 
		amCluster(
			myDataset, 
			cutHeight = 0.2
			)

	## Display analysis as HTML in default browser
	summary(
		myCluster, 
		html = TRUE
		)

	## Save analysis to HTML file
	summary(
		myCluster, 
		html = "myCluster.htm"
		)

	## Display analysis as formatted text on the console
	summary(myCluster)

	## Save unique genotypes only to a CSV file
	summary(
		myCluster, 
		csv = "myCluster.csv"
		)

	## Demonstration of how amCluster operates
	## Manual control over the recursion in amCluster()
	summary(
		myCluster1 &lt;- 
			amCluster(
				myDataset, 
				runUntilSingletons = FALSE, 
				cutHeight = 0.2
				), 
			html = TRUE
			)
	summary(
		myCluster2 &lt;- 
			amCluster(
				myCluster1$unique, 
				runUntilSingletons = FALSE, 
				cutHeight = 0.2
				),
			html = TRUE
			)
	summary(
		myCluster3 &lt;- 
			amCluster(
				myCluster2$unique, 
				runUntilSingletons = FALSE, 
				cutHeight = 0.2
				), 
			html = TRUE
			)
	summary(
		myCluster4 &lt;- 
			amCluster(
				myCluster3$unique, 
				runUntilSingletons = FALSE, 
				cutHeight = 0.2
				), 
			html = TRUE
			)
	## No more clusters, therefore stop.
	
## End(Not run)
</code></pre>

<hr>
<h2 id='amCSSForHTML'>
Produce cascading style sheet (CSS) for HTML
</h2><span id='topic+amCSSForHTML'></span>

<h3>Description</h3>

<p>Returns a string containing the CSS code for embedding in HTML output by
<code><a href="#topic+amHTML.amPairwise">amHTML.amPairwise</a></code> and <code><a href="#topic+amHTML.amUnique">amHTML.amUnique</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	amCSSForHTML()
</code></pre>


<h3>Details</h3>

<p>This function is used internally. It can also be used as a basis to tweak the CSS
code if different output formatting and colour-coding are desired. See examples.
</p>


<h3>Value</h3>

<p>A string containing a complete cascading style sheet.
</p>


<h3>Author(s)</h3>

<p>Paul Galpern (<a href="mailto:pgalpern@gmail.com">pgalpern@gmail.com</a>)
</p>


<h3>References</h3>

<p>For more information and for a complete vignette, please access via the Data S1	Supplementary
documentation and tutorials (PDF) located at &lt;doi:10.1111/j.1755-0998.2012.03137.x&gt;.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+amHTML.amPairwise">amHTML.amPairwise</a></code>, <code><a href="#topic+amHTML.amUnique">amHTML.amUnique</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	## Not run: 
	data("amExample5")
	
	## Produce CSS file for editing
	cat(amCSSForHTML(), file = "myCSS.css")
	
	## Edit myCSS.css manually using text editor, then reimport as follows
	myCSS &lt;- 
		paste(
			readLines("myCSS.css"), 
			collapse = "\n"
			)
	
	## Perform an allelematch unique analysis
	myUnique &lt;- 
		amUnique(
			amDataset(
				amExample5, 
				missingCode = "-99", 
				indexColumn = 1, 
				metaDataColumn = 2, 
				ignoreColumn = "gender"
				),
			alleleMismatch = 3
			)

	## Produce HTML output using tweaked CSS
	amHTML.amUnique(
		myUnique, 
		html = "myUnique.htm", 
		htmlCSS = myCSS
		)
	
## End(Not run)
</code></pre>

<hr>
<h2 id='amDataset'>
Prepare a dataset for use with allelematch
</h2><span id='topic+amDataset'></span><span id='topic+print.amDataset'></span>

<h3>Description</h3>

<p>Given an input matrix or <code>data.frame</code> produce a amDataset object suitable for use with
other allelematch functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	amDataset(
		multilocusDataset, 
		missingCode = "-99", 
		indexColumn = NULL, 
		metaDataColumn = NULL, 
		ignoreColumn = NULL
		)

	## S3 method for class 'amDataset'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="amDataset_+3A_multilocusdataset">multilocusDataset</code></td>
<td>

<p>A codematrix or <code>data.frame</code> containing samples in rows and alleles in columns.<br />
Sampling IDs and meta-data may be specified in up to two additional columns.
</p>
</td></tr>
<tr><td><code id="amDataset_+3A_missingcode">missingCode</code></td>
<td>

<p>A character string giving the code used for missing data.<br />
Missing data may also be represented as <code>NA</code>.
</p>
</td></tr>
<tr><td><code id="amDataset_+3A_indexcolumn">indexColumn</code></td>
<td>

<p>Optional.<br />
A character string giving the column name, or an integer giving the column number containing
the sampling ID or index information.<br />
If an index is not supplied, then the function creates an alphabetical index.
</p>
</td></tr>
<tr><td><code id="amDataset_+3A_metadatacolumn">metaDataColumn</code></td>
<td>

<p>Optional.<br />
A character string giving the column name, or an integer giving the column number containing
the meta-data.
</p>
</td></tr>
<tr><td><code id="amDataset_+3A_ignorecolumn">ignoreColumn</code></td>
<td>

<p>Optional.<br />
A vector of character string(s) giving the column name(s) or integer(s) giving the column
number(s) that should be removed from the input dataset (i.e., those that matching and
clustering should not consider).
</p>
</td></tr>
<tr><td><code id="amDataset_+3A_x">x</code></td>
<td>

<p>An amDataset object.
</p>
</td></tr>
<tr><td><code id="amDataset_+3A_...">...</code></td>
<td>

<p>Additional arguments to summary.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Examine <code><a href="#topic+amExampleData">amExampleData</a></code> for an example of a typical input dataset in the diploid
case. (Typically these files will be the CSV output from allele calling software). Sample index
or ID information and sample meta-data may be specified in two additional columns. Columns can
optionally be given names, and these are carried through analyses. If column names are not
given, appropriate names are produced.<br />
</p>
<p>Each datum is treated as a character string in <code>allelematch</code> functions, enabling the mixing
of numeric and alphanumeric data.<br />
</p>
<p>The multilocus dataset can contain any number of diploid or haploid markers, and these can be in
any order. Thus in the diploid case there should be two columns for each locus (named, say,
locus1a and locus1b). Please note that <code>AlleleMatch</code> functions pay no attention to
genetics. In other words, each column is considered a comparable state. Thus matching and
clustering of multilocus genotypes is done on the basis of superficial similarity of the data
matrix rows, rather than on any appreciation of the allelic states at each locus. See
<code><a href="#topic+amPairwise">amPairwise</a></code> for more discussion.<br />
</p>
<p>For this reason it is important when working with diploid data to ensure that identical
individuals will have identical alleles in each column. This can be achieved by sorting each
locus so that in each case the lower length allele appears in, say, a column &quot;locus1a&quot; and the
higher in column &quot;locus1b.&quot; This pattern is likely the default in allele calling software and
sorting will typically not be required unless data are derived from an unusual source.<br />
</p>
<p>Only one meta-data column is possible with <code>allelematch</code>. If multiple columns must be
associated with a given sample for downstream analyses, try pasting them together into one
string with an appropriate separator, and separating them later when allelematch analyses are
concluded.
</p>


<h3>Value</h3>

<p>An <code>amDataset</code> object.
</p>


<h3>Author(s)</h3>

<p>Paul Galpern (<a href="mailto:pgalpern@gmail.com">pgalpern@gmail.com</a>)
</p>


<h3>References</h3>

<p>For a complete vignette, please access via the Data S1 Supplementary documentation and tutorials
(PDF) located at &lt;doi:10.1111/j.1755-0998.2012.03137.x&gt;.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+amPairwise">amPairwise</a></code>, <code><a href="#topic+amUnique">amUnique</a></code>, <code><a href="#topic+amExampleData">amExampleData</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	## Not run: 
	data("amExample5")
	
	## Typical usage
	myDataset &lt;- 
		amDataset(
			amExample5, 
			missingCode = "-99", 
			indexColumn = 1, 
			metaDataColumn = 2, 
			ignoreColumn = "gender"
			)
	
	## Access elements of amDataset object
	myMetaData &lt;- myDataset$metaData
	mySamplingID &lt;- myDataset$index
	myAlleles &lt;- myDataset$multilocus
	
	## View the structure of amDataset object
	unclass(myDataset)
	
## End(Not run)
</code></pre>

<hr>
<h2 id='amExampleData'>
Data sets to support the tutorials in the supplementary documentation and examples in the manual
</h2><span id='topic+amExampleData'></span><span id='topic+amExample1'></span><span id='topic+amExample2'></span><span id='topic+amExample3'></span><span id='topic+amExample4'></span><span id='topic+amExample5'></span>

<h3>Description</h3>

<p><code>amExample1</code> High quality data set<br />
<code>amExample2</code> Good quality data set<br />
<code>amExample3</code> Marginal quality data set<br />
<code>amExample4</code> Low quality data set<br />
<code>amExample5</code> Wildlife data set<br />
</p>
<p>Data sets 1 to 4 are simulated. Because the data are simulated, the individual from which the
samples are derived is known. This is given in the column <code>knownIndividual</code>, and permits an
assessment of how effective the software has been.
</p>
<p>Data set 5 is a real dataset from a wildlife population that has been non-invasively
sampled. The meta-data has been fabricated. It represents the output from allele calling software.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	data(amExample1)
	data(amExample2)
	data(amExample3)
	data(amExample4)
	data(amExample5)
</code></pre>


<h3>Format</h3>

<p>Data frames with differing numbers of samples in rows, and alleles in columns. Missing data is
represented as &quot;-99&quot;.
</p>


<h3>Details</h3>

<p>Note how in <code>amExample5</code> a single marker &quot;Gender&quot; has been combined with diploid loci.<br />
Also note how in all data sets in diploid loci the lower length allele always comes first.<br />
This pattern is typical in allele calling software.
</p>

<hr>
<h2 id='amMatrix'>
Produce a dissimilarity matrix for pairs of multilocus genotypes
</h2><span id='topic+amMatrix'></span>

<h3>Description</h3>

<p>Given an amDataset object find the dissimilarities between pairs of multilocus genotypes, taking
missing data into account.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	amMatrix(
		amDatasetFocal, 
		missingMethod = 2
		)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="amMatrix_+3A_amdatasetfocal">amDatasetFocal</code></td>
<td>

<p>An <code>amDataset</code> object. See <code><a href="#topic+amDataset">amDataset</a></code>.
</p>
</td></tr>
<tr><td><code id="amMatrix_+3A_missingmethod">missingMethod</code></td>
<td>

<p>The method used to determine the similarity of multilocus genotypes when data is missing.<br />
The default, (<code>missingMethod = 2</code>), is preferable in all cases.<br />
See <code><a href="#topic+amMatrix">amMatrix</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is the behind-the-scenes workhorse of AlleleMatch, and typically will not be
called by the user.<br />
</p>
<p><code>missingMethod=2</code> is the recommended value, and the default, as it has performed
better in simulations. In this method, missing data matches perfectly with missing data,
while missing data matches partially with non-missing data.<br />
</p>
<p><code>missingMethod = 1</code> is retained for experimental purposes. Here, missing data
matches partially with missing and non-missing data. 
</p>


<h3>Value</h3>

<p>A distance/dissimilarity matrix of S3 class <code>amMatrix</code>.
</p>


<h3>Author(s)</h3>

<p>Paul Galpern (<a href="mailto:pgalpern@gmail.com">pgalpern@gmail.com</a>)
</p>


<h3>References</h3>

<p>For a complete vignette, please access via the Data S1 Supplementary documentation and tutorials
(PDF) located at &lt;doi:10.1111/j.1755-0998.2012.03137.x&gt;.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+amPairwise">amPairwise</a></code>, <code><a href="#topic+amUnique">amUnique</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	## Not run: 
	data("amExample1")
	
	## Produce amDataset object
	myDataset &lt;- 
		amDataset(
			amExample1, 
			missingCode = "-99", 
			indexColumn = 1, 
			metaDataColumn = 2
			)
	
	## Produce dissimilarity matrix
	dissimMatrix &lt;- amMatrix(myDataset)
	
## End(Not run)
</code></pre>

<hr>
<h2 id='amPairwise'>
Pairwise matching of multilocus genotypes
</h2><span id='topic+amPairwise'></span><span id='topic+summary.amPairwise'></span><span id='topic+amHTML.amPairwise'></span><span id='topic+amCSV.amPairwise'></span>

<h3>Description</h3>

<p>Functions to perform a pairwise matching analysis of a multilocus genotype dataset, and review
the output in formatted text or HTML. For each genotype in the focal dataset all genotypes in
the comparison genotype are returned that match at or above a threshold matching score. The
matching score is also known as the s-hat criterion (see the supplementary documentation). This
is determined using <code><a href="#topic+amMatrix">amMatrix</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	amPairwise(
		amDatasetFocal, 
		amDatasetComparison = amDatasetFocal, 
		alleleMismatch = NULL, 
		matchThreshold = NULL, 
		missingMethod = 2
		)

	amHTML.amPairwise(
		x, 
		htmlFile = NULL, 
		htmlCSS = amCSSForHTML()
		)

	amCSV.amPairwise(
		x, 
		csvFile
		)

	## S3 method for class 'amPairwise'
summary(object, html = NULL, csv = NULL, ...)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="amPairwise_+3A_amdatasetfocal">amDatasetFocal</code></td>
<td>

<p>An <code>amDataset</code> object containing focal genotypes.
</p>
</td></tr>
<tr><td><code id="amPairwise_+3A_amdatasetcomparison">amDatasetComparison</code></td>
<td>

<p>Optional.<br />
An <code>amDataset</code> object containing comparison genotypes.<br />
If not supplied, the focal dataset is also the comparison dataset (i.e., all focal dataset
members are compared against one another).
</p>
</td></tr>
<tr><td><code id="amPairwise_+3A_allelemismatch">alleleMismatch</code></td>
<td>

<p>Maximum number of mismatching alleles which will be tolerated when identifying individuals;
also known as m-hat parameter.<br />
If specified, then <code>matchThreshold</code> should be omitted.
</p>
</td></tr>
<tr><td><code id="amPairwise_+3A_matchthreshold">matchThreshold</code></td>
<td>

<p>Return comparison genotypes that match with the focal genotype at or above this score or
similarity; also known as s-hat parameter.
</p>
</td></tr>
<tr><td><code id="amPairwise_+3A_missingmethod">missingMethod</code></td>
<td>

<p>Method used to determine the similarity of multilocus genotypes when data is missing.<br />
The default, (<code>missingMethod = 2</code>), is preferable in all cases.<br />
See <code><a href="#topic+amMatrix">amMatrix</a></code>.
</p>
</td></tr>
<tr><td><code id="amPairwise_+3A_object">object</code>, <code id="amPairwise_+3A_x">x</code></td>
<td>

<p>An <code>amPairwise</code> object.
</p>
</td></tr>
<tr><td><code id="amPairwise_+3A_htmlfile">htmlFile</code></td>
<td>

<p>HTML filepath to create.<br />
If <code>htmlFile = NULL</code>, a file is created in the operating system temporary directory and
is then opened in the default browser.
</p>
</td></tr>
<tr><td><code id="amPairwise_+3A_htmlcss">htmlCSS</code></td>
<td>

<p>A string containing a valid cascading style sheet.<br />
A default style sheet is provided in <code><a href="#topic+amCSSForHTML">amCSSForHTML</a></code>.<br />
See <code><a href="#topic+amCSSForHTML">amCSSForHTML</a></code> for details of how to tweak this CSS.
</p>
</td></tr>
<tr><td><code id="amPairwise_+3A_html">html</code></td>
<td>

<p>If <code>html = NULL</code> or <code>html=FALSE</code>, formatted textual output is displayed on the
console.<br />
If <code>html = TRUE</code>, the <code>summary</code> method produces and loads an HTML file in the
default browser.<br />
<code>html</code> can also contain a path to a file where HTML output will be written.
</p>
</td></tr>
<tr><td><code id="amPairwise_+3A_csvfile">csvFile</code>, <code id="amPairwise_+3A_csv">csv</code></td>
<td>

<p>CSV filepath to create containing giving a data frame representation of the pairwise
matching results.
</p>
</td></tr>
<tr><td><code id="amPairwise_+3A_...">...</code></td>
<td>

<p>Additional arguments to <code>summary</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Pairwise matching of genotypes is a useful means to assess data quality and inspect for
genotyping errors.<br />
</p>
<p><code>matchThreshold</code> represents the similarity between two multilocus genotypes and can be
thought of as a percentage similarity (or a Hamming's distance between two vectors) that has
been corrected where missing data is present, such that missing data represents neither a match
nor a mismatch but a &quot;partial&quot; match. See <code><a href="#topic+amMatrix">amMatrix</a></code> for more discussion of this
metric.
</p>


<h3>Value</h3>

<p><code>amPairwise</code> object or side effects: analysis summary written to an HTML file or to the
console, or written to a CSV file.
</p>


<h3>Note</h3>

<p>As <code>matchThreshold</code> is lowered, the size of the output increases rapidly. Typically
analyses will not be very useful or manageable with thresholds below 0.7.<br />
</p>
<p>There is an additional side effect of <code>html = TRUE</code> (or of <code>htmlFile = NULL</code>). If
required, there is a clean up of the operating system temporary directory where AlleleMatch
temporary HTML files are stored. Files that match the pattern am*.html and are older 24 hours
are deleted from this temporary directory.
</p>


<h3>Author(s)</h3>

<p>Paul Galpern (<a href="mailto:pgalpern@gmail.com">pgalpern@gmail.com</a>)
</p>


<h3>References</h3>

<p>For a complete vignette, please access via the Data S1 Supplementary documentation and tutorials
(PDF) located at &lt;doi:10.1111/j.1755-0998.2012.03137.x&gt;.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+amDataset">amDataset</a></code>, <code><a href="#topic+amMatrix">amMatrix</a></code>, <code><a href="#topic+amUnique">amUnique</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	## Not run: 
	data("amExample5")

	## Produce amDataset object
	myDataset &lt;- 
		amDataset(
			amExample5, 
			missingCode = "-99", 
			indexColumn = 1, 
			metaDataColumn = 2, 
			ignoreColumn = "gender"
			)

	## Typical usage
	myPairwise &lt;- 
		amPairwise(
			myDataset, 
			alleleMismatch = 2
			)

	## Display analysis as HTML in default browser
	summary(
		myPairwise, 
		html = TRUE
		)

	## Save analysis to HTML file
	summary(
		myPairwise, 
		html = "myPairwise.htm"
		)

	## Save analysis to CSV file
	summary(
		myPairwise, 
		csv = "myPairwise.csv"
		)

	## Display analysis as formatted text on the console
	summary(myPairwise)

	## Compare one dataset against a second
	## Both must have same number of allele columns
	## Here we create two datasets artificially from one for illustration purposes
	myDatasetA &lt;- 
		amDataset(
			amExample5[sample(nrow(amExample5))[1:25], ], 
			missingCode = "-99", 
			indexColumn = 1, 
			ignoreColumn = 2
			)
	myDatasetB &lt;- 
		amDataset(
			amExample5[sample(nrow(amExample5))[1:100], ], 
			missingCode = "-99", 
			indexColumn = 1, 
			ignoreColumn = 2
			)
	myPairwise2 &lt;- 
		amPairwise(
			myDatasetA, 
			myDatasetB, 
			alleleMismatch = 3
			)
	summary(
		myPairwise2, 
		html = TRUE
		)
	
## End(Not run)
</code></pre>

<hr>
<h2 id='amUnique'>
Identification of unique genotypes
</h2><span id='topic+amUnique'></span><span id='topic+summary.amUnique'></span><span id='topic+amHTML.amUnique'></span><span id='topic+amCSV.amUnique'></span>

<h3>Description</h3>

<p>Identifies unique genotypes and generates analysis output in formatted text, HTML, or
CSV. Samples are clustered and matched based on their dissimilarity score (see
<code><a href="#topic+amMatrix">amMatrix</a></code>). Also calculated is the match probability, Psib, which is the
probability that a sample is a sibling of a unique genotype (and therefore not a
replicate sample) given the allele frequencies in a population consisting of only the
unique genotypes (Wilberg &amp; Dreher, 2004).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	amUnique(
		amDatasetFocal, 
		multilocusMap = NULL, 
		alleleMismatch = NULL, 
		matchThreshold = NULL, 
		cutHeight = NULL, 
		doPsib = "missing", 
		consensusMethod = 1, 
		verbose = FALSE
		)
	
	amHTML.amUnique(
		x, 
		htmlFile = NULL, 
		htmlCSS = amCSSForHTML()
		)

	amCSV.amUnique(
		x, 
		csvFile, 
		uniqueOnly = FALSE
		)
		
		## S3 method for class 'amUnique'
summary(object, html = NULL, csv = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="amUnique_+3A_amdatasetfocal">amDatasetFocal</code></td>
<td>

<p>An <code>amDataset</code> object containing genotypes in which an unknown number of
individuals are sampled multiple times
</p>
</td></tr>
<tr><td><code id="amUnique_+3A_multilocusmap">multilocusMap</code></td>
<td>

<p>Optional.<br />
A vector of integers or strings giving the mappings onto loci for all genotype
columns in amDatasetFocal.<br />
When omitted, columns are assumed to be paired (i.e., diploid loci with alleles in
adjacent columns).<br />
See details.
</p>
</td></tr>
<tr><td><code id="amUnique_+3A_allelemismatch">alleleMismatch</code></td>
<td>

<p>Optional.<br />
Maximum number of mismatching alleles which will be tolerated when identifying
individuals; also known as m-hat parameter.<br />
If specified, then <code>matchThreshold</code> and <code>cutHeight</code> should be
omitted.<br />
All three parameters are related.<br />
See details.
</p>
</td></tr>
<tr><td><code id="amUnique_+3A_matchthreshold">matchThreshold</code></td>
<td>

<p>Optional.<br />
Minimum dissimilarity score which constitutes a match when identifying
individuals; also known as s-hat parameter.<br />
If specified, then <code>alleleMismatch</code> and <code>cutHeight</code> should be omitted;
all three parameters are related.<br />
See details.
</p>
</td></tr>
<tr><td><code id="amUnique_+3A_cutheight">cutHeight</code></td>
<td>

<p>Optional.<br />
The <code>cutHeight</code> parameter used in dynamic tree cutting by <code>amCluster</code>;
also known as d-hat parameter.<br />
If specified, then <code>alleleMismatch</code> and <code>matchThreshold</code> should be
omitted.<br />
All three parameters are related.<br />
See details.
</p>
</td></tr>
<tr><td><code id="amUnique_+3A_dopsib">doPsib</code></td>
<td>

<p>String specifying how match probability should be calculated.<br />
See details.
</p>
</td></tr>
<tr><td><code id="amUnique_+3A_consensusmethod">consensusMethod</code></td>
<td>

<p>The method (an integer) used to determine the consensus multilocus genotype from a
cluster of multilocus genotypes.<br />
See <code><a href="#topic+amCluster">amCluster</a></code> for details.<br />
Typically the default is adequate.
</p>
</td></tr>
<tr><td><code id="amUnique_+3A_verbose">verbose</code></td>
<td>

<p>If <code>verbose = TRUE</code>, report the progress of the analysis to the console.<br />
Useful with datasets consisting of thousands of samples where progress may be
slow.
</p>
</td></tr>
<tr><td><code id="amUnique_+3A_object">object</code>, <code id="amUnique_+3A_x">x</code></td>
<td>

<p>An <code>amUnique</code> object.
</p>
</td></tr>
<tr><td><code id="amUnique_+3A_htmlfile">htmlFile</code></td>
<td>

<p>HTML filepath to create.<br />
If <code>htmlFile = NULL</code>, a file is created in the operating system temporary
directory and is then opened in the default browser.
</p>
</td></tr>
<tr><td><code id="amUnique_+3A_htmlcss">htmlCSS</code></td>
<td>

<p>A string containing a valid cascading style sheet.<br />
A default style sheet is provided in <code><a href="#topic+amCSSForHTML">amCSSForHTML</a></code>.<br />
See <code><a href="#topic+amCSSForHTML">amCSSForHTML</a></code> for details of how to tweak this CSS.
</p>
</td></tr>
<tr><td><code id="amUnique_+3A_html">html</code></td>
<td>

<p>If <code>html = TRUE</code>, the <code>summary</code> method produces and loads an HTML file in the
default browser.<br />
<code>html</code> can also contain a path to a file where HTML output will be
written.<br />
Note that <code>summary</code> does not produce formatted output for the
console.
</p>
</td></tr>
<tr><td><code id="amUnique_+3A_csvfile">csvFile</code>, <code id="amUnique_+3A_csv">csv</code></td>
<td>

<p>CSV filepath to create containing a representation of the <code>amUnique</code>
analysis.
</p>
</td></tr>
<tr><td><code id="amUnique_+3A_uniqueonly">uniqueOnly</code></td>
<td>

<p>If <code>uniqueOnly = TRUE</code>, only the unique genotypes will be saved to a CSV,
with no additional information associated with the analysis.
</p>
</td></tr>
<tr><td><code id="amUnique_+3A_...">...</code></td>
<td>

<p>Additional arguments to <code>summary</code>
</p>
</td></tr> 
</table>


<h3>Details</h3>

<p>Only one of <code>alleleMismatch</code>, <code>cutHeight</code>, <code>matchThreshold</code> can be
specified, as the three parameters are related.<br />
</p>
<p><code>alleleMismatch</code> is the most intuitive way to understand how the identification
of unique genotypes proceeds. For example, a setting of <code>alleleMismatch = 4</code>
implies that up to four alleles may be different for multiple samples to be
representatives of the same individual. In practice, however, this value is only an
approximation of the amount of mismatch that may be tolerated. This is because the
clustering process used to identify unique genotypes, and the subsequent matching
which identifies samples that match these unique genotypes is based on a dissimilarity
metric or score (see <code><a href="#topic+amMatrix">amMatrix</a></code>) that incorporates both allele mismatches
and missing data. <code>alleleMismatch</code> is not used in analyses and is converted to
this dissimilarity metric in the following manner: <code>cutHeight</code> which is parameter
for <code><a href="#topic+amCluster">amCluster</a></code> and called from this function is <code>cutHeight =
	alleleMismatch/(number of allele columns)</code> and <code>matchThreshold</code> which is a
parameter for <code><a href="#topic+amPairwise">amPairwise</a></code> and also called from this function is
<code>matchThreshold = 1 - cutHeight</code>.<br />
</p>
<p>Selecting the appropriate value for <code>alleleMismatch</code>, <code>cutHeight</code>, or
<code>matchThreshold</code> is an important task. Use <code><a href="#topic+amUniqueProfile">amUniqueProfile</a></code> to
assist in this process. Seethe Data S1 Supplementary documentation and tutorials (PDF)
located at &lt;doi:10.1111/j.1755-0998.2012.03137.x&gt;<br />
</p>
<p><code>doPsib = "missing"</code> is the default and specifies that match probability Psib 
should be calculated for samples that match unique genotypes and have no allele
mismatches, but may differ by having missing data. <code>doPsib = "all"</code> specifies
that Psib should be calculated for all samples that match unique genotypes. In this
case, if allele mismatches occur, alleles are assumed to be missing at the mismatching
loci.<br />
</p>
<p><code>multilocusMap</code> is often not required, as amDataset objects will typically
consist of paired columns of genotypes, where each pair is a separate locus. In cases
where this is not the case (e.g., gender is in only one column), a map vector must be
specified.<br />
</p>
<p>Example: <code>amDataset</code> consists of gender followed by 4 diploid loci in paired
columns<br />
<code>multilocusMap = c(1, 2, 2, 3, 3, 4, 4, 5, 5)</code><br />
or equally<br />
<code>multilocusMap=c("GENDER", "LOC1", "LOC1", "LOC2", "LOC2", "LOC3", "LOC4",
	"LOC4")</code>
</p>
<p>For more information on selecting <code>consensusMethod</code> see <code><a href="#topic+amCluster">amCluster</a></code>.
The default <code>consensusMethod = 1</code> is typically adequate.
</p>


<h3>Value</h3>

<p><code>amUnique</code> object or side effects: analysis summary written to an HTML file or to
the console, or written to a CSV file.
</p>


<h3>Note</h3>

<p>There is an additional side effect of <code>html = TRUE</code> (or of <code>htmlFile =
	NULL</code>). If required, there is a clean up of the operating system temporary directory
where AlleleMatch temporary HTML files are stored. Files that match the pattern
am*.html and are older 24 hours are deleted from this temporary directory.
</p>


<h3>Author(s)</h3>

<p>Paul Galpern (<a href="mailto:pgalpern@gmail.com">pgalpern@gmail.com</a>)
</p>


<h3>References</h3>

<p>For a complete vignette, please access via the Data S1 Supplementary documentation and
tutorials (PDF) located at &lt;doi:10.1111/j.1755-0998.2012.03137.x&gt;.<br />
</p>
<p>Wilberg MJ, Dreher BP (2004) GENECAP: a program for analysis of multilocus genotype
data for non-invasive sampling and capture-recapture population estimation.
<em>Molecular Ecology Notes</em>, 4, 783-785.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+amDataset">amDataset</a></code>, <code><a href="#topic+amMatrix">amMatrix</a></code>, <code><a href="#topic+amPairwise">amPairwise</a></code>,
<code><a href="#topic+amCluster">amCluster</a></code>, <code><a href="#topic+amUniqueProfile">amUniqueProfile</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	## Not run: 
	data("amExample2")
	
	## Produce amDataset object
	myDataset &lt;- 
		amDataset(
			amExample2, 
			missingCode = "-99", 
			indexColumn = 1, 
			ignoreColumn = 2
			)

	## Usage
	## Optimal alleleMismatch parameter previously found using amUniqueProfile()
	myUnique &lt;- 
		amUnique(
		myDataset, 
		alleleMismatch = 3
		)

	## Display analysis as HTML in default browser
	summary(
		myUnique, 
		html = TRUE
		)

	## Save analysis to HTML file
	summary(
		myUnique,
		html = "myUnique.htm"
		)

	## Save analysis to a CSV file
	summary(
		myUnique, 
		csv = "myUnique.csv"
		)

	## Save unique genotypes only to a CSV file
	summary(
		myUnique,
		csv = "myUnique.csv",
		uniqueOnly = TRUE
		)

	## Data set with gender information
	data("amExample5")

	## Produce amDataset object
	myDataset2 &lt;- 
		amDataset(
			amExample5, 
			missingCode = "-99", 
			indexColumn = 1,
			metaDataColumn = 2
			)

	## Usage
	## Optimal alleleMismatch parameter previously found using amUniqueProfile()
	myUniqueProfile &lt;- 
		amUnique(
			myDataset2,
			multilocusMap = c(1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7, 7, 8, 8, 9, 9, 10, 10,
			11, 11), 
			alleleMismatch = 3
			)

	
## End(Not run)

</code></pre>

<hr>
<h2 id='amUniqueProfile'>
Determine optimal parameter values for the identification of unique genotypes
</h2><span id='topic+amUniqueProfile'></span>

<h3>Description</h3>

<p>Function to automatically run <code>amUnique</code> at a sequence of parameter values to
determine an optimal setting, and optionally plot the result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	amUniqueProfile(
		amDatasetFocal, 
		multilocusMap = NULL, 
		alleleMismatch = NULL, 
		matchThreshold = NULL, 
		cutHeight = NULL, 
		guessOptimum = TRUE, 
		doPlot = TRUE, 
		consensusMethod = 1, 
		verbose = TRUE
		)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="amUniqueProfile_+3A_amdatasetfocal">amDatasetFocal</code></td>
<td>

<p>An <code>amDataset</code> object containing genotypes in which an unknown number of
individuals are sampled multiple times.
</p>
</td></tr>
<tr><td><code id="amUniqueProfile_+3A_multilocusmap">multilocusMap</code></td>
<td>

<p>Optionally a vector of integers or strings giving the mappings onto loci for all
genotype columns in amDatasetFocal.<br />
When omitted, columns are assumed to be paired (i.e., diploid loci with alleles in
adjacent columns).<br />
See details.
</p>
</td></tr>
<tr><td><code id="amUniqueProfile_+3A_allelemismatch">alleleMismatch</code></td>
<td>

<p>A vector giving a sequence, where elements give the maximum number of mismatching
alleles which will be tolerated when identifying individuals; also known as the
m-hat parameter.<br />
If specified, then <code>matchThreshold</code> and <code>cutHeight</code> should be omitted;
all three parameters are related.<br />
See <code><a href="#topic+amUnique">amUnique</a></code> for details.
</p>
</td></tr>
<tr><td><code id="amUniqueProfile_+3A_matchthreshold">matchThreshold</code></td>
<td>

<p>A vector giving a sequence, where elements give the minimum dissimilarity score
which constitutes a match when identifying individuals; also known as the s-hat
parameter.<br />
If specified, then <code>alleleMismatch</code> and <code>cutHeight</code> should be omitted;
all three parameters are related.<br />
See <code><a href="#topic+amUnique">amUnique</a></code> for details.
</p>
</td></tr>
<tr><td><code id="amUniqueProfile_+3A_cutheight">cutHeight</code></td>
<td>

<p>A vector giving a sequence, where elements give the <code>cutHeight</code> parameter
used in dynamic tree cutting by <code>amCluster</code>; also known as the d-hat
parameter.<br />
If specified, then <code>alleleMismatch</code> and <code>matchThreshold</code> should be
omitted; all three parameters are related.<br />
See details.
</p>
</td></tr>
<tr><td><code id="amUniqueProfile_+3A_doplot">doPlot</code></td>
<td>

<p>If <code>doPlot = TRUE</code>, then a plot showing summary data from multiple runs of
<code>amUnique</code> is produced.
</p>
</td></tr>
<tr><td><code id="amUniqueProfile_+3A_guessoptimum">guessOptimum</code></td>
<td>

<p>If <code>guessOptimum = TRUE</code>, then will estimate the optimal value of the
parameter being profiled by examining the profile for the first minimum associated
with a drop in multiple matches as sensitivity to differences among samples
decreases.
</p>
</td></tr>
<tr><td><code id="amUniqueProfile_+3A_consensusmethod">consensusMethod</code></td>
<td>

<p>The method (an integer) used to determine the consensus multilocus genotype from a
cluster of multilocus genotypes.<br />
See <code><a href="#topic+amCluster">amCluster</a></code> for details.
</p>
</td></tr>
<tr><td><code id="amUniqueProfile_+3A_verbose">verbose</code></td>
<td>

<p>If <code>verbose = TRUE</code>, then the progress of the profiling will be reported to
the console.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Selecting the appropriate value for <code>alleleMismatch</code>, <code>cutHeight</code>, or
<code>matchThreshold</code> is an important task. Use this function to assist in this
process. Typically the optimal value of any of these parameters is found where the
number of multiple matches is minimized (the majority of samples are similar to only
one unique genotype). Usually there is a minimum when these parameters are set to be
very sensitive to differences among samples (i.e., <code>alleleMismatch</code> or
<code>cutHeight</code> are 0, <code>matchThreshold</code> is 1). Simulations suggest that the next
most sensitive minimum in multiple matches is the optimal value. This minimum will
often be associated with a drop in multiple matches as sensitivity drops. For more
discussion of this important step, see the Data S1 Supplementary documentation and
tutorials (PDF) located at &lt;doi:10.1111/j.1755-0998.2012.03137.x&gt;.<br />
</p>
<p>Using <code>guessOptimum = TRUE</code> will attempt to estimate the location of this minimum
and add it to the profile plot. Manual assessment of this estimate using the plot is
strongly recommended.<br />
</p>
<p>If none of <code>alleleMismatch</code>, <code>cutHeight</code>, or <code>matchThreshold</code> is given,
the function runs a sequence of values for <code>alleleMismatch</code> as follows:
<code>seq(from = 0, to = floor(ncol(amDatasetFocal$multilocus) * 0.4), by = 1)</code><br />
</p>
<p><code>multilocusMap</code> is often not required, as <code>amDataset</code> objects will typically
consist of paired columns of genotypes, where each pair is a separate locus. In cases
where this is not the case (e.g., gender is given in only one column), a map vector
must be specified.<br />
</p>
<p>Example: <code>amDataset</code> consists of gender followed by 4 diploid loci in paired
columns<br />
<code>multilocusMap = c(1, 2, 2, 3, 3, 4, 4, 5, 5)</code><br />
or equally<br />
<code>multilocusMap=c("GENDER", "LOC1", "LOC1", "LOC2", "LOC2", "LOC3", "LOC4",
	"LOC4")</code>
</p>
<p>For more information on selecting <code>consensusMethod</code> see <code><a href="#topic+amCluster">amCluster</a></code>.
The default <code>consensusMethod = 1</code> is typically adequate.
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> containing summary data from multiple runs of <code>amUnique</code>
</p>


<h3>Author(s)</h3>

<p>Paul Galpern (<a href="mailto:pgalpern@gmail.com">pgalpern@gmail.com</a>)
</p>


<h3>References</h3>

<p>For a complete vignette, please access via the Data S1 Supplementary documentation and
tutorials (PDF) located at &lt;doi:10.1111/j.1755-0998.2012.03137.x&gt;.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+amUnique">amUnique</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	## Not run: 
	data("amExample2")
	
	## Produce amDataset object
	myDataset &lt;- 
		amDataset(
			amExample2, 
			missingCode = "-99", 
			indexColumn = 1, 
			metaDataColumn = 2
			)

	## Usage (uncomment)
	myUniqueProfile &lt;- 
		amUniqueProfile(myDataset)
	
	## Data set with gender information
	data("amExample5")
	
	## Produce amDataset object
	myDataset2 &lt;- 
		amDataset(
			amExample5, 
			missingCode = "-99", 
			indexColumn = 1, 
			metaDataColumn = 2
			)
	
	## Usage
	myUniqueProfile &lt;- 
		amUniqueProfile(
			myDataset2,
			multilocusMap = c(1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7, 7, 8, 8, 9, 9, 10, 10,
			11, 11))
	
## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
