<!DOCTYPE html><html lang="en"><head><title>Help for package ALLSPICER</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ALLSPICER}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ALLSPICE'><p>ALLSPICE</p></a></li>
<li><a href='#ALLSPICE_simulation'><p>ALLSPICE_simulation</p></a></li>
<li><a href='#format_ALLSPICE_data'><p>format_ALLSPICE_data</p></a></li>
<li><a href='#get_ac_mat'><p>get_ac_mat</p></a></li>
<li><a href='#get_af_mat'><p>get_af_mat</p></a></li>
<li><a href='#get_beta_hat'><p>get_beta_hat</p></a></li>
<li><a href='#get_c_hat'><p>get_c_hat</p></a></li>
<li><a href='#get_geno_mat'><p>get_geno_mat</p></a></li>
<li><a href='#get_likelihood_test_stats'><p>get_likelihood_test_stats</p></a></li>
<li><a href='#get_mle_beta'><p>get_mle_beta</p></a></li>
<li><a href='#get_pheno_pair'><p>get_pheno_pair</p></a></li>
<li><a href='#get_single_geno'><p>get_single_geno</p></a></li>
<li><a href='#get_true_beta'><p>get_true_beta</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>ALLelic Spectrum of Pleiotropy Informed Correlated Effects</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.9</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Wenhan Lu &lt;wlu@broadinstitute.org&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides statistical tools to analyze heterogeneous effects of rare variants within genes that are associated with multiple traits. The package implements methods for assessing pleiotropic effects and identifying allelic heterogeneity, which can be useful in large-scale genetic studies. Methods include likelihood-based statistical tests to assess these effects. For more details, see Lu et al. (2024) &lt;<a href="https://doi.org/10.1101%2F2024.10.01.614806">doi:10.1101/2024.10.01.614806</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, magrittr, readr, mvtnorm, stats</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-10-14 11:16:19 UTC; wlu</td>
</tr>
<tr>
<td>Author:</td>
<td>Wenhan Lu [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-10-16 17:50:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='ALLSPICE'>ALLSPICE</h2><span id='topic+ALLSPICE'></span>

<h3>Description</h3>

<p>ALLSPICE (ALLelic Spectrum of Pleiotropy Informed Correlated Effects)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ALLSPICE(
  data,
  pheno_corr,
  n_ind,
  gene = "GENENAME",
  pheno1 = "PHENO1",
  pheno2 = "PHENO2",
  beta1_field = "BETA1",
  beta2_field = "BETA2",
  af_field = "AF"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ALLSPICE_+3A_data">data</code></td>
<td>
<p>Input data with number of rows indicating number of variants, three columns are required:
1) effect sizes of variants for phenotype 1, 2) effect sizes of variants for phenotype 2, 3) allele frequency of variants
Note: this should include variants from ONE gene that is associated with the two phenotypes,
preferably of the SAME functional category after being filtered to variants with allele frequency below a certain threshold (e.g. 1e-4)</p>
</td></tr>
<tr><td><code id="ALLSPICE_+3A_pheno_corr">pheno_corr</code></td>
<td>
<p>phenotypic correlation between the two phenotypes being tested</p>
</td></tr>
<tr><td><code id="ALLSPICE_+3A_n_ind">n_ind</code></td>
<td>
<p>total number of individuals</p>
</td></tr>
<tr><td><code id="ALLSPICE_+3A_gene">gene</code></td>
<td>
<p>name of the gene being tested, default 'GENENAME'</p>
</td></tr>
<tr><td><code id="ALLSPICE_+3A_pheno1">pheno1</code></td>
<td>
<p>descriptive name of phenotype 1, default 'PHENO1'</p>
</td></tr>
<tr><td><code id="ALLSPICE_+3A_pheno2">pheno2</code></td>
<td>
<p>descriptive name of phenotype 2, default 'PHENO2'</p>
</td></tr>
<tr><td><code id="ALLSPICE_+3A_beta1_field">beta1_field</code></td>
<td>
<p>field name for effect sizes of variants on phenotype 1, default 'BETA1'</p>
</td></tr>
<tr><td><code id="ALLSPICE_+3A_beta2_field">beta2_field</code></td>
<td>
<p>field name for effect sizes of variants on phenotype 2, default 'BETA2'</p>
</td></tr>
<tr><td><code id="ALLSPICE_+3A_af_field">af_field</code></td>
<td>
<p>field name for allele frequencies of variants, default 'AF'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of summary statistics from ALLSPICE test
including phenotype names, gene names, MLE of slope c, ALLSPICE test statistic - lambda, pvalue from a chi-square distribution, total number of variants being tested
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- data.frame(x = rnorm(10), y = rnorm(10), z = runif(10, 0,1))
ALLSPICE(data,pheno_corr=0.5,n_ind=10000,beta1_field='x',beta2_field='y',af_field='z')
</code></pre>

<hr>
<h2 id='ALLSPICE_simulation'>ALLSPICE_simulation</h2><span id='topic+ALLSPICE_simulation'></span>

<h3>Description</h3>

<p>Simulate data and run ALLSPICE
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ALLSPICE_simulation(n_ind, n_var, c, r, pi, sigma, mle = TRUE, null = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ALLSPICE_simulation_+3A_n_ind">n_ind</code></td>
<td>
<p>total number of individuals</p>
</td></tr>
<tr><td><code id="ALLSPICE_simulation_+3A_n_var">n_var</code></td>
<td>
<p>total number of variants</p>
</td></tr>
<tr><td><code id="ALLSPICE_simulation_+3A_c">c</code></td>
<td>
<p>slope between the two sets of variant effect sizes, only applicable when 'null' == TRUE</p>
</td></tr>
<tr><td><code id="ALLSPICE_simulation_+3A_r">r</code></td>
<td>
<p>phenotypic correlation between the two phenotypes</p>
</td></tr>
<tr><td><code id="ALLSPICE_simulation_+3A_pi">pi</code></td>
<td>
<p>probability of variant of having no effect on the phenotype</p>
</td></tr>
<tr><td><code id="ALLSPICE_simulation_+3A_sigma">sigma</code></td>
<td>
<p>variance of the two sets of effect sizes</p>
</td></tr>
<tr><td><code id="ALLSPICE_simulation_+3A_mle">mle</code></td>
<td>
<p>whether to use MLE of c to compute the test statistic, use true c value if FALSE</p>
</td></tr>
<tr><td><code id="ALLSPICE_simulation_+3A_null">null</code></td>
<td>
<p>whether to simulate data under the null hypothesis (no linear relationship) or the alternative hypothesis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of two pieces of results:
1) ALLSPICE test results
2) effect size table: true effect size simulated, effect size estimate from linear model, effect size estimated from MLE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ALLSPICE_simulation(n_ind=10000, n_var=100, c=0.6, r=0.5, pi=0.5, sigma=1, mle = TRUE, null=TRUE)

</code></pre>

<hr>
<h2 id='format_ALLSPICE_data'>format_ALLSPICE_data</h2><span id='topic+format_ALLSPICE_data'></span>

<h3>Description</h3>

<p>data formatting function: format raw data to be loaded into ALLSPICE
</p>


<h3>Usage</h3>

<pre><code class='language-R'>format_ALLSPICE_data(data, beta1_field, beta2_field, af_field)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="format_ALLSPICE_data_+3A_data">data</code></td>
<td>
<p>raw input data</p>
</td></tr>
<tr><td><code id="format_ALLSPICE_data_+3A_beta1_field">beta1_field</code></td>
<td>
<p>field name of effect size for the first phenotype</p>
</td></tr>
<tr><td><code id="format_ALLSPICE_data_+3A_beta2_field">beta2_field</code></td>
<td>
<p>field name of effect size for the second phenotype</p>
</td></tr>
<tr><td><code id="format_ALLSPICE_data_+3A_af_field">af_field</code></td>
<td>
<p>field name of allele frequency information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame containing effect sizes of variants on two phenotypes and their allele frequency information
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- data.frame(x = rnorm(10), y = rnorm(10), z = runif(10, 0,1))
data &lt;- format_ALLSPICE_data(data=data, beta1_field = 'x', beta2_field = 'y', af_field = 'z')
</code></pre>

<hr>
<h2 id='get_ac_mat'>get_ac_mat</h2><span id='topic+get_ac_mat'></span>

<h3>Description</h3>

<p>simulation function: simulate allele count information for 'n_var' variants, with a maximum allele count 'max_cnt'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_ac_mat(n_var, max_cnt = 100)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_ac_mat_+3A_n_var">n_var</code></td>
<td>
<p>total number of variants</p>
</td></tr>
<tr><td><code id="get_ac_mat_+3A_max_cnt">max_cnt</code></td>
<td>
<p>maximum allele count, default 100</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'n_var'x'n_var' diagnal matrix of allele count information for 'n_var' variants
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ac_mat &lt;- get_ac_mat(n_var=100, max_cnt = 100)
</code></pre>

<hr>
<h2 id='get_af_mat'>get_af_mat</h2><span id='topic+get_af_mat'></span>

<h3>Description</h3>

<p>simulation function: compute allele frequency information variants with allele counts stored in diagonal matrix 'AC' from a population of sample size 'n_ind'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_af_mat(AC, n_ind)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_af_mat_+3A_ac">AC</code></td>
<td>
<p>a diagonal matrix of allele count information for all variants</p>
</td></tr>
<tr><td><code id="get_af_mat_+3A_n_ind">n_ind</code></td>
<td>
<p>total number of individuals in the population</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'n_var'x'n_var' diagnal matrix of allele frequency information for 'n_var' (dimension of 'AC') variants
</p>


<h3>Examples</h3>

<pre><code class='language-R'>af_mat &lt;- get_af_mat(AC = c(20, 50, 10, 1, 5), n_ind = 10000)
</code></pre>

<hr>
<h2 id='get_beta_hat'>get_beta_hat</h2><span id='topic+get_beta_hat'></span>

<h3>Description</h3>

<p>simulation function: compute effect sizes estimated form linear regression model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_beta_hat(Y, X, A, n_ind)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_beta_hat_+3A_y">Y</code></td>
<td>
<p>phenotype information</p>
</td></tr>
<tr><td><code id="get_beta_hat_+3A_x">X</code></td>
<td>
<p>genotype information</p>
</td></tr>
<tr><td><code id="get_beta_hat_+3A_a">A</code></td>
<td>
<p>Allele frequency information</p>
</td></tr>
<tr><td><code id="get_beta_hat_+3A_n_ind">n_ind</code></td>
<td>
<p>total number of individuals</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 2x'n_var' matrix of estimated effect size information (first row corresponds to the first phenotype, second row corresponds to the second phenotype)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>AC &lt;- get_ac_mat(n_var=100)
A &lt;- get_af_mat(AC=AC, n_ind=10000)
X &lt;- get_geno_mat(AC, n_ind=10000)
b &lt;- get_true_beta(n_var=100, c=0.6, pi=0.5, sigma=1, null=TRUE)
Y &lt;- get_pheno_pair(b=b, X=X, r=0.5)
b_hat &lt;- get_beta_hat(Y=Y, X=X, A=A, n_ind=10000)
</code></pre>

<hr>
<h2 id='get_c_hat'>get_c_hat</h2><span id='topic+get_c_hat'></span>

<h3>Description</h3>

<p>ALLSPICE function: compute the slope 'c' that maximize the likelihood (maximum likelihood estimate - MLE)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_c_hat(b1_hat, b2_hat, A, r)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_c_hat_+3A_b1_hat">b1_hat</code></td>
<td>
<p>estimated effect size of the first phenotype across all variants</p>
</td></tr>
<tr><td><code id="get_c_hat_+3A_b2_hat">b2_hat</code></td>
<td>
<p>estimated effect size of the second phenotype across all variants</p>
</td></tr>
<tr><td><code id="get_c_hat_+3A_a">A</code></td>
<td>
<p>Allele frequency information</p>
</td></tr>
<tr><td><code id="get_c_hat_+3A_r">r</code></td>
<td>
<p>phenotypic correlation between the two phenotypes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the MLE of slope between two sets of effect sizes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>AC &lt;- get_ac_mat(n_var=100)
A &lt;- get_af_mat(AC=AC, n_ind=10000)
X &lt;- get_geno_mat(AC, n_ind=10000)
b &lt;- get_true_beta(n_var=100, c=0.6, pi=0.5, sigma=1, null=TRUE)
Y &lt;- get_pheno_pair(b=b, X=X, r=0.5)
b_hat &lt;- get_beta_hat(Y=Y, X=X, A=A, n_ind=10000)
b1_hat &lt;- matrix(b_hat[1, ], nrow = 1)
b2_hat &lt;- matrix(b_hat[2, ], nrow = 1)
c_hat &lt;- get_c_hat(b1_hat=b1_hat, b2_hat=b2_hat, A=A, r=0.5)
</code></pre>

<hr>
<h2 id='get_geno_mat'>get_geno_mat</h2><span id='topic+get_geno_mat'></span>

<h3>Description</h3>

<p>simulation function: simulate genotype information for a set of loci with allele counts 'AC'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_geno_mat(AC, n_ind)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_geno_mat_+3A_ac">AC</code></td>
<td>
<p>allele counts of loci (length 'm')</p>
</td></tr>
<tr><td><code id="get_geno_mat_+3A_n_ind">n_ind</code></td>
<td>
<p>total number of indicitions</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An 'n_ind'x'm' matrix of genotype information of 'n_ind' individuals and 'm' variants
</p>


<h3>Examples</h3>

<pre><code class='language-R'>geno_mat &lt;- get_geno_mat(AC = c(20, 50, 10, 1, 5), n_ind = 10000)
</code></pre>

<hr>
<h2 id='get_likelihood_test_stats'>get_likelihood_test_stats</h2><span id='topic+get_likelihood_test_stats'></span>

<h3>Description</h3>

<p>ALLSPICE function: compute the maximum likelihood ratio of the ALLSPICE test statistic
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_likelihood_test_stats(n_ind, r, b1_hat, b2_hat, c, A)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_likelihood_test_stats_+3A_n_ind">n_ind</code></td>
<td>
<p>total number of individuals</p>
</td></tr>
<tr><td><code id="get_likelihood_test_stats_+3A_r">r</code></td>
<td>
<p>phenotypic correlation between the two phenotypes</p>
</td></tr>
<tr><td><code id="get_likelihood_test_stats_+3A_b1_hat">b1_hat</code></td>
<td>
<p>estimated effect size of the first phenotype across all variants</p>
</td></tr>
<tr><td><code id="get_likelihood_test_stats_+3A_b2_hat">b2_hat</code></td>
<td>
<p>estimated effect size of the second phenotype across all variants</p>
</td></tr>
<tr><td><code id="get_likelihood_test_stats_+3A_c">c</code></td>
<td>
<p>MLE of the slope between the two sets of variant effect sizes</p>
</td></tr>
<tr><td><code id="get_likelihood_test_stats_+3A_a">A</code></td>
<td>
<p>Allele frequency information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single numeric value representing the test statistic of ALLSPICE (maximum likelihood ratio)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>AC &lt;- get_ac_mat(n_var=100)
A &lt;- get_af_mat(AC=AC, n_ind=10000)
X &lt;- get_geno_mat(AC, n_ind=10000)
b &lt;- get_true_beta(n_var=100, c=0.6, pi=0.5, sigma=1, null=TRUE)
Y &lt;- get_pheno_pair(b=b, X=X, r=0.5)
b_hat &lt;- get_beta_hat(Y=Y, X=X, A=A, n_ind=10000)
b1_hat &lt;- matrix(b_hat[1, ], nrow = 1)
b2_hat &lt;- matrix(b_hat[2, ], nrow = 1)
c_hat &lt;- get_c_hat(b1_hat=b1_hat, b2_hat=b2_hat, A=A, r=0.5)
lambda &lt;- get_likelihood_test_stats(n_ind=10000, r=0.5, b1_hat=b1_hat, b2_hat=b2_hat, c=c_hat, A=A)
</code></pre>

<hr>
<h2 id='get_mle_beta'>get_mle_beta</h2><span id='topic+get_mle_beta'></span>

<h3>Description</h3>

<p>ALLSPICE function: compute the effect size estimates that maximize the likelihood (maximum likelihood estimate - MLE) conditioning on c
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mle_beta(b1_hat, b2_hat, c, r, null = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_mle_beta_+3A_b1_hat">b1_hat</code></td>
<td>
<p>estimated effect size of the first phenotype across all variants</p>
</td></tr>
<tr><td><code id="get_mle_beta_+3A_b2_hat">b2_hat</code></td>
<td>
<p>estimated effect size of the second phenotype across all variants</p>
</td></tr>
<tr><td><code id="get_mle_beta_+3A_c">c</code></td>
<td>
<p>slope between the two sets of variant effect sizes, only applicable when 'null' == TRUE</p>
</td></tr>
<tr><td><code id="get_mle_beta_+3A_r">r</code></td>
<td>
<p>phenotypic correlation between the two phenotypes</p>
</td></tr>
<tr><td><code id="get_mle_beta_+3A_null">null</code></td>
<td>
<p>whether to simulate data under the null hypothesis (no linear relationship) or the alternative hypothesis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 2x'n_var' matrix of MLE estimated effect size information (first row corresponds to the first phenotype, second row corresponds to the second phenotype)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>AC &lt;- get_ac_mat(n_var=100)
A &lt;- get_af_mat(AC=AC, n_ind=10000)
X &lt;- get_geno_mat(AC, n_ind=10000)
b &lt;- get_true_beta(n_var=100, c=0.6, pi=0.5, sigma=1, null=TRUE)
Y &lt;- get_pheno_pair(b=b, X=X, r=0.5)
b_hat &lt;- get_beta_hat(Y=Y, X=X, A=A, n_ind=10000)
b1_hat &lt;- matrix(b_hat[1, ], nrow = 1)
b2_hat &lt;- matrix(b_hat[2, ], nrow = 1)
b_mle &lt;- get_mle_beta(b1_hat=b1_hat, b2_hat=b2_hat, c=0.6, r=0.5, null=TRUE)
</code></pre>

<hr>
<h2 id='get_pheno_pair'>get_pheno_pair</h2><span id='topic+get_pheno_pair'></span>

<h3>Description</h3>

<p>simulation function: simulate true phenotype values of a pair of phenotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pheno_pair(b, X, r)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_pheno_pair_+3A_b">b</code></td>
<td>
<p>true effect size matrix of variants on the two phenotypes</p>
</td></tr>
<tr><td><code id="get_pheno_pair_+3A_x">X</code></td>
<td>
<p>genotype matrix</p>
</td></tr>
<tr><td><code id="get_pheno_pair_+3A_r">r</code></td>
<td>
<p>phenotypic correlation between the two phenotypes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 2x'n_ind' matrix of phenotype information (first row corresponds to the first phenotype, second row corresponds to the second phenotype)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>AC &lt;- get_ac_mat(n_var=100)
X &lt;- get_geno_mat(AC, n_ind=10000)
b &lt;- get_true_beta(n_var=100, c=0.6, pi=0.5, sigma=1, null=TRUE)
Y &lt;- get_pheno_pair(b=b, X=X, r=0.5)
</code></pre>

<hr>
<h2 id='get_single_geno'>get_single_geno</h2><span id='topic+get_single_geno'></span>

<h3>Description</h3>

<p>simulation function: simulate genotype information for one locus, where 'cnt' samples out of 'n_ind' has the mutation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_single_geno(cnt, n_ind)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_single_geno_+3A_cnt">cnt</code></td>
<td>
<p>number of individuals with the mutation</p>
</td></tr>
<tr><td><code id="get_single_geno_+3A_n_ind">n_ind</code></td>
<td>
<p>total number of individuals</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A binary vector representing the genotype information of 'n_ind' individuals for a particular locus, where 'cnt' entries has value 1.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>geno &lt;- get_single_geno(cnt = 100, n_ind = 10000)
</code></pre>

<hr>
<h2 id='get_true_beta'>get_true_beta</h2><span id='topic+get_true_beta'></span>

<h3>Description</h3>

<p>simulation function: simulate true effect size information of 'n_var' variants for two phenotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_true_beta(n_var, c, pi, sigma, null = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_true_beta_+3A_n_var">n_var</code></td>
<td>
<p>total number of variants</p>
</td></tr>
<tr><td><code id="get_true_beta_+3A_c">c</code></td>
<td>
<p>slope between the two sets of variant effect sizes, only applicable when 'null' == TRUE</p>
</td></tr>
<tr><td><code id="get_true_beta_+3A_pi">pi</code></td>
<td>
<p>probability of variant of having no effect on the phenotype</p>
</td></tr>
<tr><td><code id="get_true_beta_+3A_sigma">sigma</code></td>
<td>
<p>variance of the two sets of effect sizes</p>
</td></tr>
<tr><td><code id="get_true_beta_+3A_null">null</code></td>
<td>
<p>whether to simulate data under the null hypothesis (no linear relationship) or the alternative hypothesis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 2x'n_var' matrix of effect size information for 'n_var' variants (first row corresponds to the first phenotype, second row corresponds to the second phenotype)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>true_beta &lt;- get_true_beta(n_var=100, c=0.6, pi=0.5, sigma=1, null=TRUE)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
