<!DOCTYPE html><html><head><title>Help for package Allspice</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Allspice}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#assemble&lt;-,Asset-method'>
<p>Finish Asset contents</p></a></li>
<li><a href='#asset'>
<p>Constructor for Asset</p></a></li>
<li><a href='#bcellALL'>
<p>Simulated data</p></a></li>
<li><a href='#classifier'>
<p>Constructor for Classifier</p></a></li>
<li><a href='#classify,Asset-method'>
<p>Data classification</p></a></li>
<li><a href='#configuration&lt;-,Asset-method'>
<p>Asset configuration</p></a></li>
<li><a href='#covariates&lt;-,Classifier-method'>
<p>Sample covariate data</p></a></li>
<li><a href='#export,Asset-method'>
<p>Data classification</p></a></li>
<li><a href='#information,Classifier-method'>
<p>Classifier contents</p></a></li>
<li><a href='#nomenclature&lt;-,Asset-method'>
<p>Variable names</p></a></li>
<li><a href='#normalize,Asset-method'>
<p>Sample normalization</p></a></li>
<li><a href='#predictions,Classifier-method'>
<p>Category predictions</p></a></li>
<li><a href='#profiles&lt;-,Classifier-method'>
<p>Sample data profiles</p></a></li>
<li><a href='#report,Classifier-method'>
<p>Visualization of results</p></a></li>
<li><a href='#scores,Classifier-method'>
<p>Category scores</p></a></li>
<li><a href='#standardize,Asset-method'>
<p>Data standardization</p></a></li>
<li><a href='#visuals&lt;-,Asset-method'>
<p>Visual attributes</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>RNA-Seq Profile Classifier</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-01-20</td>
</tr>
<tr>
<td>Author:</td>
<td>Ville-Petteri Makinen [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ville-Petteri Makinen &lt;vpmakine@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>We developed a lightweight machine learning tool for RNA profiling of acute lymphoblastic leukemia (ALL), however, it can be used for any problem where multiple classes need to be identified from multi-dimensional data. The methodology is described in Makinen V-P, Rehn J, Breen J, Yeung D, White DL (2022) Multi-cohort transcriptomic subtyping of B-cell acute lymphoblastic leukemia, International Journal of Molecular Sciences 23:4574, &lt;<a href="https://doi.org/10.3390%2Fijms23094574">doi:10.3390/ijms23094574</a>&gt;. The classifier contains optimized mean profiles of the classes (centroids) as observed in the training data, and new samples are matched to these centroids using the shortest Euclidean distance. Centroids derived from a dataset of 1,598 ALL patients are included, but users can train the models with their own data as well. The output includes both numerical and visual presentations of the classification results. Samples with mixed features from multiple classes or atypical values are also identified.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-01-20 03:10:57 UTC; vipmak</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-01-20 03:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='assemble+26lt+3B-+2CAsset-method'>
Finish Asset contents
</h2><span id='topic+assemble+3C-+2CAsset-method'></span><span id='topic+assemble+2CAsset-method'></span><span id='topic+assemble+3C-'></span><span id='topic+assemble'></span>

<h3>Description</h3>

<p>Trains a new classification model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>assemble(obj) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="assemble+2B26lt+2B3B-+2B2CAsset-method_+3A_obj">obj</code></td>
<td>

<p>An object of the class Asset.
</p>
</td></tr>
<tr><td><code id="assemble+2B26lt+2B3B-+2B2CAsset-method_+3A_value">value</code></td>
<td>

<p>A list that containts training data, see details.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The value argument must contain three named elements: <code>title</code>,
<code>dat</code> and <code>bits</code>. Optional <code>predictors</code> and
<code>covariates</code> elements can also be included.
</p>
<p>The title is a descriptive identifier for the asset that will be
displayed by the Classifier object in <code><a href="#topic+report">report</a>()</code>.
</p>
<p>The <code>dat</code> element is a matrix that contains the training samples.
The variables are organized into named rows and the samples into
named columns. Non-finite values are not allowed.
</p>
<p>The <code>predictors</code> element contains the names of the input variables
that should be used for training the model. If empty, all inputs are
used for automatic feature selection and subsequent training steps.
</p>
<p>The <code>covariates</code> element contains additional information for constructing
the final classification models. Unlike the data matrix, variables
are organized into named columns and the samples are stored as rows.
Non-finite values are not allowed.
</p>
<p>The <code>bits</code> element contains labels for category memberships. Three
formats are supported: 1) a character vector of named elements that
contains non-empty strings, 2) a matrix or a data frame with row names
and a single column of non-empty values, and 3) a matrix or a data frame
with multiple columns that contain binary values where 1s indicate
category membership (the name of the column is the name of the
category). Overlapping categories are allowed.
</p>
<p>The final asset is assembled in six steps. First, the training data are
standardized and normalized. Second, input variables are sorted according
to their univariate classification performance. Third, redundant features
are excluded by testing the sorted variables for mutual correlations; this
produces an optimized listing of non-redundant features that are the most
predictive of the category labels. Fourth, mean centroids are calculated for
each category. Fifth, training samples are matched to their nearest
centroids and the distances collected as preliminary predictor scores.
Lastly, logistic regression models are fitted to the preliminary scores,
covariates and category labels to enable the calculation of standardized
predictor scores for new data.
</p>


<h3>Value</h3>

<p>Updates the Asset object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Prepare training data.
simu &lt;- bcellALL(200)
materials &lt;- list(title="Simutypes")
materials$dat &lt;- simu$counts
materials$covariates &lt;- simu$metadata[,c("MALE","AGE")]
materials$bits &lt;- simu$metadata[,"SUBTYPE",drop=FALSE]

# Assemble classification asset.
bALL &lt;- asset()
assemble(bALL) &lt;- materials

# Export asset into a new folder.
tpath &lt;- tempfile()
export(bALL, folder = tpath)

# Create a classifier.
cls &lt;- classifier(tpath, verbose = FALSE)

# Classify new samples.
simu &lt;- bcellALL(5)
covariates(cls) &lt;- simu$metadata
profiles(cls) &lt;- simu$counts
primary &lt;- predictions(cls)[[1]]
print(primary[,c("LABEL","PROX","EXCL")])
</code></pre>

<hr>
<h2 id='asset'>
Constructor for Asset
</h2><span id='topic+asset'></span><span id='topic+Asset-class'></span>

<h3>Description</h3>

<p>Creates a new Asset object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>asset(folder = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="asset_+3A_folder">folder</code></td>
<td>

<p>Path to a folder that contains the necessary
files for a classification asset.
</p>
</td></tr>
<tr><td><code id="asset_+3A_verbose">verbose</code></td>
<td>

<p>If TRUE, accessed asset items (file names) are printed on screen.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an Asset object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Set up an ALL subtyping asset.
folder &lt;- system.file("subtypes", package="Allspice")
a &lt;- asset(folder)
</code></pre>

<hr>
<h2 id='bcellALL'>
Simulated data
</h2><span id='topic+bcellALL'></span>

<h3>Description</h3>

<p>Simulated data of B-cell acute lymphoblastic leukemia.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bcellALL(n = 200, contamination = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bcellALL_+3A_n">n</code></td>
<td>

<p>Number of samples.
</p>
</td></tr>
<tr><td><code id="bcellALL_+3A_contamination">contamination</code></td>
<td>

<p>Proportion of samples with randomly shuffled values.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list with two elements: <code>counts</code> contains gene RNA read
counts and <code>metadata</code> contains age and sex information and the
generating subtype label.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate B-cell ALL samples.
simu &lt;- bcellALL(5)
print(head(simu$counts))
print(simu$metadata)
</code></pre>

<hr>
<h2 id='classifier'>
Constructor for Classifier
</h2><span id='topic+classifier'></span><span id='topic+Classifier-class'></span>

<h3>Description</h3>

<p>Creates a new Classifier object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>classifier(..., verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="classifier_+3A_...">...</code></td>
<td>

<p>Any number of paths to folders that contain assets
(see <code><a href="#topic+asset">asset</a>()</code>).
</p>
</td></tr>
<tr><td><code id="classifier_+3A_verbose">verbose</code></td>
<td>

<p>If TRUE, information about the imported assets is printed on screen.
</p>
</td></tr>  
</table>


<h3>Details</h3>

<p>The first input argument will set the primary asset, and the others
will be considered secondary assets.
</p>


<h3>Value</h3>

<p>Returns a Classifier object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Set up an ALL classifier object.
cls &lt;- classifier()
</code></pre>

<hr>
<h2 id='classify+2CAsset-method'>
Data classification
</h2><span id='topic+classify+2CAsset-method'></span><span id='topic+classify'></span>

<h3>Description</h3>

<p>Assigns category labels to new data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>classify(obj, dat, covariates)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="classify+2B2CAsset-method_+3A_obj">obj</code></td>
<td>

<p>An object of the class Asset.
</p>
</td></tr>
<tr><td><code id="classify+2B2CAsset-method_+3A_dat">dat</code></td>
<td>

<p>A matrix that containts variables as rows and samples as columns.
</p>
</td></tr>
<tr><td><code id="classify+2B2CAsset-method_+3A_covariates">covariates</code></td>
<td>

<p>A data.frame or matrix that containts samples as rows and
covariates as columns.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input data will be automatically normalized and standardized using
the internal asset parameters, see <code><a href="#topic+configuration">configuration</a>()</code> for
details.
</p>


<h3>Value</h3>

<p>Returns a data frame that contains predicted category labels and performance
indicators. The column 'CATEG' contains the final predictions, including
&quot;Unclassified&quot; or &quot;Ambiguous&quot; for samples that could not be reliably
classified. The columns 'MATCH.1st' and 'MATCH.2nd' contain the first and
second best matching categories, respectively.
</p>
<p>The column 'BIOMRK' contains a standardized biomarker score that indicates
how similar the sample is with respect to the best-matching category. The
column 'PROX' tells the likelihood of how likely it is that the observed
biomarker score would have been generated by a training sample from the
best-matching category (balanced group sizes). The column 'EXCL' tells the
likelihood that the sample does not share characteristic features with any
other category.
</p>
<p>The returned data frame also has the attribute &quot;biomarkers&quot; that
contains biomarker scores for all categories.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Import ALL subtyping asset.
base &lt;- system.file(package = "Allspice")
folder &lt;- file.path(base, "subtypes")
a &lt;- asset(folder)

# Simulated data.
simu &lt;- bcellALL(5)

# Predict categories.
res &lt;- classify(a, dat = simu$counts, covariates = simu$metadata)
print(res[,c("LABEL","PROX","EXCL")])
</code></pre>

<hr>
<h2 id='configuration+26lt+3B-+2CAsset-method'>
Asset configuration
</h2><span id='topic+configuration+3C-+2CAsset-method'></span><span id='topic+configuration+2CAsset-method'></span><span id='topic+configuration+3C-'></span><span id='topic+configuration'></span>

<h3>Description</h3>

<p>Set internal parameters for an Asset object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>configuration(obj) &lt;- value
configuration(obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="configuration+2B26lt+2B3B-+2B2CAsset-method_+3A_obj">obj</code></td>
<td>

<p>An object of the class Asset.
</p>
</td></tr>
<tr><td><code id="configuration+2B26lt+2B3B-+2B2CAsset-method_+3A_value">value</code></td>
<td>

<p>A numeric vector with named elements.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Element names from the input are compared with the internal list
of parameters. Those that match will be updated.
</p>
<p>Normalization parameters include 'norm' (if set to 0, normalization
is not performed), 'nonzero.min' (the minimum data value considered larger
than zero) and 'nonzero.ratio' (minimum ratio of non-zero values to include
a variable in the output). See <code><a href="#topic+normalize">normalize</a>()</code> for additional
details.
</p>
<p>Standardization parameters include 'standard' (if set to 0, standardization
is not performed) and 'logarithm' (if set to 0, data values are used without
taking the logarithm). See <code><a href="#topic+standardize">standardize</a>()</code> for additional
details.
</p>
<p>Feature selection parameters include 'ninput.max' (the maximum number of
features to be used for classification) and 'rrinput.max' (the maximum
correlation r-squared to be allowed between features). See
<code><a href="#topic+assemble">assemble</a>()</code> for details on selecting non-redundant inputs.
</p>
<p>Classification parameters include 'probability.min' (minimum probability
rating considered for reliable classification) and 'exclusivity.min'
(minimum exclusivity for non-ambiguous classification).
See <code><a href="#topic+classify">classify</a>()</code> for additional details.
</p>


<h3>Value</h3>

<p>Updates the Asset object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Change asset configuration.
a &lt;- asset()
print(configuration(a))
configuration(a) &lt;- c(nonzero.min=0, nonzero.ratio=0)
print(configuration(a))
</code></pre>

<hr>
<h2 id='covariates+26lt+3B-+2CClassifier-method'>
Sample covariate data
</h2><span id='topic+covariates+3C-+2CClassifier-method'></span><span id='topic+covariates+2CClassifier-method'></span><span id='topic+covariates+3C-'></span><span id='topic+covariates'></span>

<h3>Description</h3>

<p>Add covariate data into the classifier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covariates(obj) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="covariates+2B26lt+2B3B-+2B2CClassifier-method_+3A_obj">obj</code></td>
<td>

<p>An object of the class Classifier.
</p>
</td></tr>
<tr><td><code id="covariates+2B26lt+2B3B-+2B2CClassifier-method_+3A_value">value</code></td>
<td>

<p>A numeric vector or a matrix.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the input is a vector, the elements must be named and these
names will be used to identify variables.
</p>
<p>If the input is a matrix, it must have named rows and named columns that
will be matched with sample identities in <code><a href="#topic+profiles">profiles</a>()</code>.
</p>


<h3>Value</h3>

<p>Updates the Classifier object. Any previous data are discarded.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulated data.
simu &lt;- bcellALL(5)

# Predict subtypes without covariates.
cls &lt;- classifier(verbose = FALSE)
profiles(cls) &lt;- simu$counts
primary &lt;- predictions(cls)[[1]]
print(primary[,c("LABEL","PROX","EXCL")])

# Predict subtypes with covariates.
cls &lt;- classifier(verbose = FALSE)
covariates(cls) &lt;- simu$metadata
profiles(cls) &lt;- simu$counts
primary &lt;- predictions(cls)[[1]]
print(primary[,c("LABEL","PROX","EXCL")])
</code></pre>

<hr>
<h2 id='export+2CAsset-method'>
Data classification
</h2><span id='topic+export+2CAsset-method'></span><span id='topic+export'></span>

<h3>Description</h3>

<p>Assigns category labels to new data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export(obj, folder)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="export+2B2CAsset-method_+3A_obj">obj</code></td>
<td>

<p>An object of the class Asset.
</p>
</td></tr>
<tr><td><code id="export+2B2CAsset-method_+3A_folder">folder</code></td>
<td>

<p>Path to a folder that will contain the asset files.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the names of the files that were saved in the folder.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Import ALL subtyping asset.
base &lt;- system.file(package = "Allspice")
folder &lt;- file.path(base, "subtypes")
a &lt;- asset(folder)

# Export asset into a new folder.
tpath &lt;- tempfile()
fnames &lt;- export(a, folder = tpath)
print(dir(tpath))
</code></pre>

<hr>
<h2 id='information+2CClassifier-method'>
Classifier contents
</h2><span id='topic+information+2CClassifier-method'></span><span id='topic+information'></span>

<h3>Description</h3>

<p>Information about the assets of a classifier
</p>


<h3>Usage</h3>

<pre><code class='language-R'>information(obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="information+2B2CClassifier-method_+3A_obj">obj</code></td>
<td>

<p>An object of the class Classifier.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with three elements: <code>covariates</code> is a data frame that contains
the variable names that are included in the classification assets,
<code>configuration</code> contains the analysis settings for each asset and
<code>categories</code> contains the labels and visual attributes for the assets.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Show the contents of the b-cell ALL classifier.
cls &lt;- classifier(verbose=FALSE)
info &lt;- information(cls)
print(info$covariates)
print(info$configuration)
print(head(info$categories))
print(tail(info$categories))
</code></pre>

<hr>
<h2 id='nomenclature+26lt+3B-+2CAsset-method'>
Variable names
</h2><span id='topic+nomenclature+3C-+2CAsset-method'></span><span id='topic+nomenclature+2CAsset-method'></span><span id='topic+nomenclature+3C-'></span><span id='topic+nomenclature'></span>

<h3>Description</h3>

<p>Conversion table between variable naming schemes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nomenclature(obj) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nomenclature+2B26lt+2B3B-+2B2CAsset-method_+3A_obj">obj</code></td>
<td>

<p>An object of the class Asset.
</p>
</td></tr>
<tr><td><code id="nomenclature+2B26lt+2B3B-+2B2CAsset-method_+3A_value">value</code></td>
<td>

<p>A data frame, see details.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The data frame should contain character columns of variable names
with each column representing a spefic naming scheme from which
the variables are translated into the row names of the data frame.
</p>
<p>For example, to convert gene symbols into Ensemble codes, a data frame
with the gene symbols as row names and one column of ensemble codes
is needed.
</p>


<h3>Value</h3>

<p>Updates the Asset object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Import nomenclature from a system file.
base &lt;- system.file(package = "Allspice")
fname &lt;- file.path(base, "subtypes", "nomenclature.txt")
info &lt;- read.delim(fname, stringsAsFactors = FALSE)

# Set ENSEMBLE identities as row names.
rownames(info) &lt;- info$ENSEMBL
info$ENSEMBL &lt;- NULL
print(head(info))

# Create a new asset and set nomenclature.
a &lt;- asset()
nomenclature(a) &lt;- info

# Prepare training data.
simu &lt;- bcellALL(200)
materials &lt;- list(title="Simutypes")
materials$dat &lt;- simu$counts
materials$covariates &lt;- simu$metadata[,c("MALE","AGE")]
materials$bits &lt;- simu$metadata[,"SUBTYPE",drop=FALSE]

# Assemble classification asset.
assemble(a) &lt;- materials

# Check that nomenclature was set.
simu &lt;- bcellALL(5)
expres &lt;- normalize(a, dat = simu$counts)
print(head(simu$counts))
print(head(expres))
</code></pre>

<hr>
<h2 id='normalize+2CAsset-method'>
Sample normalization
</h2><span id='topic+normalize+2CAsset-method'></span><span id='topic+normalize'></span>

<h3>Description</h3>

<p>Adjust scale differences between samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalize(obj, dat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalize+2B2CAsset-method_+3A_obj">obj</code></td>
<td>

<p>An object of the class Asset.
</p>
</td></tr>
<tr><td><code id="normalize+2B2CAsset-method_+3A_dat">dat</code></td>
<td>

<p>A matrix that containts variables as rows and samples as columns.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The normalization pipeline comprises three steps. First, variable names
are checked against the internal nomenclature and converted to the
internal naming scheme where necessary (see <code><a href="#topic+nomenclature">nomenclature</a>()</code>).
Second, variables that are present in the internal normalization reference
are imputed with reference values if not available in the data. Third, the
data are normalized according to the DESeq2 algortihm (Love MI, Huber W &amp;
Anders S, Moderated estimation of fold change and dispersion for RNA-seq
data with DESeq2, Genome Biol 15, 550, 2014).
</p>


<h3>Value</h3>

<p>Returns a matrix in the same format as the input.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Import ALL subtyping asset.
base &lt;- system.file(package = "Allspice")
folder &lt;- file.path(base, "subtypes")
a &lt;- asset(folder)

# Simulated data.
simu &lt;- bcellALL(5)

# Normalize RNA read counts.
expres &lt;- normalize(a, dat = simu$counts)
print(head(simu$counts))
print(head(expres))
</code></pre>

<hr>
<h2 id='predictions+2CClassifier-method'>
Category predictions
</h2><span id='topic+predictions+2CClassifier-method'></span><span id='topic+predictions'></span>

<h3>Description</h3>

<p>Classifies samples based on their profiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predictions(obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predictions+2B2CClassifier-method_+3A_obj">obj</code></td>
<td>

<p>An object of the class Classifier.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Use the functions <code><a href="#topic+covariates">covariates</a>()</code> and
<code><a href="#topic+profiles">profiles</a>()</code> to import data into the classifier.
</p>


<h3>Value</h3>

<p>Returns a list of data frames that contain the output from each
classification asset within the classifier. See <code><a href="#topic+classify">classify</a>()</code>
for details on the result items.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulated data.
simu &lt;- bcellALL(5)

# Predict subtypes.
cls &lt;- classifier(verbose = FALSE)
covariates(cls) &lt;- simu$metadata
profiles(cls) &lt;- simu$counts
pred &lt;- predictions(cls)
print(pred[[1]][,c("LABEL","PROX","EXCL")])
print(pred[[2]][,c("LABEL","PROX","EXCL")])
print(pred[[3]][,c("LABEL","PROX","EXCL")])
</code></pre>

<hr>
<h2 id='profiles+26lt+3B-+2CClassifier-method'>
Sample data profiles
</h2><span id='topic+profiles+3C-+2CClassifier-method'></span><span id='topic+profiles+3C-'></span><span id='topic+profiles'></span>

<h3>Description</h3>

<p>Analyse new data using classification assets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>profiles(obj) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="profiles+2B26lt+2B3B-+2B2CClassifier-method_+3A_obj">obj</code></td>
<td>

<p>An object of the class Classifier.
</p>
</td></tr>
<tr><td><code id="profiles+2B26lt+2B3B-+2B2CClassifier-method_+3A_value">value</code></td>
<td>

<p>A numeric vector or a matrix where samples are organized into
columns and variables into rows.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the input is a vector, the elements must be named and these
names will be used to identify variables.
</p>
<p>If the input is a matrix, it must have sample identities as column
names and variables identified by row names.
</p>


<h3>Value</h3>

<p>Updates the Classifier object. Any previous data are discarded.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulated data.
simu &lt;- bcellALL(5)

# Predict subtypes.
cls &lt;- classifier(verbose = FALSE)
covariates(cls) &lt;- simu$metadata
profiles(cls) &lt;- simu$counts
primary &lt;- predictions(cls)[[1]]
print(primary[,c("LABEL","PROX","EXCL")])
</code></pre>

<hr>
<h2 id='report+2CClassifier-method'>
Visualization of results
</h2><span id='topic+report+2CClassifier-method'></span><span id='topic+report'></span>

<h3>Description</h3>

<p>Creates a visual report of the classification results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>report(obj, name, file = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="report+2B2CClassifier-method_+3A_obj">obj</code></td>
<td>

<p>An object of the class Classifier.
</p>
</td></tr>
<tr><td><code id="report+2B2CClassifier-method_+3A_name">name</code></td>
<td>

<p>Name of the sample to be shown.
</p>
</td></tr>
<tr><td><code id="report+2B2CClassifier-method_+3A_file">file</code></td>
<td>

<p>Name of the output file.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function generates a Scalable Vector Graphics figure that shows the
results from each classification asset within the Classifier. The report
will highlight the predicted category label and quality metrics for the
primary asset and bar charts for the fits to categories in all assests.
</p>
<p>If no file name is provided, the report is plotted on the current device,
however, note that best visual outcomes are achieved by plotting in a file,
especially with classifiers with three or more assets.
</p>


<h3>Value</h3>

<p>Returns the name of the output file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulated data.
simu &lt;- bcellALL(5)
keys &lt;- colnames(simu$counts)

# Predict subtypes.
cls &lt;- classifier(verbose = FALSE)
covariates(cls) &lt;- simu$metadata
profiles(cls) &lt;- simu$counts

# Show visual report by name.
dev.new()
report(cls, name = keys[3])

# Show visual report by sample index.
dev.new()
report(cls, name = 3)
</code></pre>

<hr>
<h2 id='scores+2CClassifier-method'>
Category scores
</h2><span id='topic+scores+2CClassifier-method'></span><span id='topic+scores'></span>

<h3>Description</h3>

<p>Classification scores for samples based on their profiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scores(obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scores+2B2CClassifier-method_+3A_obj">obj</code></td>
<td>

<p>An object of the class Classifier.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Use the functions <code><a href="#topic+covariates">covariates</a>()</code> and
<code><a href="#topic+profiles">profiles</a>()</code> to import data into the classifier.
</p>


<h3>Value</h3>

<p>Returns a list of data frames that contain the output from each
classification asset within the classifier.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulated data.
simu &lt;- bcellALL(5)

# Predict subtypes.
cls &lt;- classifier(verbose = FALSE)
covariates(cls) &lt;- simu$metadata
profiles(cls) &lt;- simu$counts
z &lt;- scores(cls)
print(z[[1]][,1:5])
print(z[[2]][,1:5])
print(z[[3]][,1:5])
</code></pre>

<hr>
<h2 id='standardize+2CAsset-method'>
Data standardization
</h2><span id='topic+standardize+2CAsset-method'></span><span id='topic+standardize'></span>

<h3>Description</h3>

<p>Standardize scale and location of variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>standardize(obj, dat, trim = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="standardize+2B2CAsset-method_+3A_obj">obj</code></td>
<td>

<p>An object of the class Asset.
</p>
</td></tr>
<tr><td><code id="standardize+2B2CAsset-method_+3A_dat">dat</code></td>
<td>

<p>A matrix that containts variables as rows and samples as columns.
</p>
</td></tr>
<tr><td><code id="standardize+2B2CAsset-method_+3A_trim">trim</code></td>
<td>

<p>If true, returns only variables used as input features for classification.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the asset is so configured, the data are first transformed by log(x + 1).
Values are processed with the mean and standard deviation that were
calculated from the training data when the asset was assembled.
The mean is subtracted and the values divided by SD. To control for
outliers, extreme values are compressed by the t-distribution with 50
degrees of freedom.
</p>


<h3>Value</h3>

<p>Returns a matrix in the same format as the input.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Import ALL subtyping asset.
base &lt;- system.file(package = "Allspice")
folder &lt;- file.path(base, "subtypes")
a &lt;- asset(folder)

# Simulated data.
simu &lt;- bcellALL(5)

# Standardize RNA read counts.
expres &lt;- normalize(a, dat = simu$counts)
zscores &lt;- standardize(a, dat = expres)
print(head(simu$counts))
print(head(expres))
print(head(zscores))
</code></pre>

<hr>
<h2 id='visuals+26lt+3B-+2CAsset-method'>
Visual attributes
</h2><span id='topic+visuals+3C-+2CAsset-method'></span><span id='topic+visuals+2CAsset-method'></span><span id='topic+visuals+3C-'></span><span id='topic+visuals'></span>

<h3>Description</h3>

<p>Attach text and color attributes to categories.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visuals(obj) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visuals+2B26lt+2B3B-+2B2CAsset-method_+3A_obj">obj</code></td>
<td>

<p>An object of the class Asset.
</p>
</td></tr>
<tr><td><code id="visuals+2B26lt+2B3B-+2B2CAsset-method_+3A_value">value</code></td>
<td>

<p>A character vector or a data frame, see details.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the input value is a character vector, the elements are stored as the
category text and names of the elements are stored as category names.
</p>
<p>If the input value is a data frame, the column 'LABEL' is used as the text
and the row names as the names of the categories. Additional columns may
include 'COLOR', 'COLOR.dark' and 'COLOR.light' that must contain strings
of color names or hexadecimal codes as produced by <code><a href="grDevices.html#topic+rgb">rgb</a>()</code>.
In absence of color data, the function assigns automatic colors.
</p>


<h3>Value</h3>

<p>Updates the Asset object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create a new asset and set nomenclature.
a &lt;- asset()

# Set category labels with automatic colors.
labels &lt;- paste("Category", 1:8)
names(labels) &lt;- paste0("cat", 1:8)
visuals(a) &lt;- labels
print(a@categories)

# Add color information.
info &lt;- data.frame(stringsAsFactors = FALSE,
    LABEL = labels, COLOR = "red")
rownames(info) &lt;- names(labels)
visuals(a) &lt;- info
print(a@categories)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
