<!DOCTYPE html><html lang="en-US"><head><title>Help for package AmpGram</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {AmpGram}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AmpGram-package'><p>Prediction of antimicrobial peptides</p></a></li>
<li><a href='#AmpGram_gui'><p>AmpGram Graphical User Interface</p></a></li>
<li><a href='#AmpGram_predictions'><p>Prediction of antimicrobial peptides</p></a></li>
<li><a href='#get_AMPs'><p>Get putative antimicrobial peptides</p></a></li>
<li><a href='#install_AmpGramModel'><p>Install AmpGramModel package containing model for AMP prediction</p></a></li>
<li><a href='#pred2df'><p>Convert predictions to data.frame</p>
Return predictions as data.frame</a></li>
<li><a href='#predict.ampgram_model'><p>Predict antimicrobial peptides</p></a></li>
<li><a href='#read_txt'><p>Read sequences from .txt file</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Prediction of Antimicrobial Peptides</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-05-19</td>
</tr>
<tr>
<td>Description:</td>
<td>Predicts antimicrobial peptides using random forests trained on the
    n-gram encoded peptides. The implemented algorithm can be accessed from
    both the command line and shiny-based GUI. The AmpGram model is too large 
    for CRAN and it has to be downloaded separately from the repository:
    <a href="https://github.com/michbur/AmpGramModel">https://github.com/michbur/AmpGramModel</a>.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/michbur/AmpGram">https://github.com/michbur/AmpGram</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/michbur/AmpGram/issues">https://github.com/michbur/AmpGram/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>biogram, devtools, pbapply, ranger, shiny, stringi</td>
</tr>
<tr>
<td>Suggests:</td>
<td>DT, ggplot2, pander, rmarkdown, shinythemes, spelling</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.0</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-05-22 11:38:49 UTC; michal</td>
</tr>
<tr>
<td>Author:</td>
<td>Michal Burdukiewicz
    <a href="https://orcid.org/0000-0001-8926-582X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [cre, aut],
  Katarzyna Sidorczuk [ctb],
  Filip Pietluch [ctb],
  Dominik Rafacz [aut],
  Stefan Roediger <a href="https://orcid.org/0000-0002-1441-6512"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Jaroslaw Chilimoniuk
    <a href="https://orcid.org/0000-0001-5467-018X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Michal Burdukiewicz &lt;michalburdukiewicz@gmail.com&gt;</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-05-31 10:10:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='AmpGram-package'>Prediction of antimicrobial peptides</h2><span id='topic+AmpGram-package'></span><span id='topic+AmpGram'></span>

<h3>Description</h3>

<p>Antimicrobial peptides (AMPs) are ancient and evolutionarily conserved 
molecules widespread in all living organisms that participate in host 
defence and/or microbial competition. Due to their positive charge, 
hydrophobicity and amphipathicity, they preferentially disrupt 
negatively-charged bacterial membranes. AMPs are considered an important 
alternative to traditional antibiotics, especially in times when the latter 
are drastically losing their effectiveness. Therefore, efficient computational 
tools for AMP prediction are essential to identify the best AMP candidates without 
undertaking expensive experimental studies. AmpGram is our novel tool for predicting 
AMPs based on the stacked random forests and n-gram analysis, able to
successfully predict antimicrobial peptides in proteomes.
</p>


<h3>Details</h3>

<p>AmpGram is available as R function (<code><a href="#topic+predict.ampgram_model">predict.ampgram_model</a></code>) or
shiny GUI (<code><a href="#topic+AmpGram_gui">AmpGram_gui</a></code>).
</p>
<p>AmpGram requires the external package, AmpGramModel, which 
contains models necessary to perform the prediction. The model 
can be installed using <code><a href="#topic+install_AmpGramModel">install_AmpGramModel</a></code>
</p>


<h3>Author(s)</h3>

<p>Maintainer: Michal Burdukiewicz &lt;michalburdukiewicz@gmail.com&gt;
</p>


<h3>References</h3>

<p>Burdukiewicz M, Sidorczuk K, Rafacz D, Pietluch F, Chilimoniuk J,
Roediger S, Gagat P. (2020) <em>AmpGram: a proteome screening tool for 
prediction and design of antimicrobial peptides</em>. (submitted)
</p>

<hr>
<h2 id='AmpGram_gui'>AmpGram Graphical User Interface</h2><span id='topic+AmpGram_gui'></span>

<h3>Description</h3>

<p>Launches graphical user interface that predicts presence of 
antimicrobial peptides.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AmpGram_gui()
</code></pre>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>Warning </h3>

<p>Any ad-blocking software may cause malfunctions.
</p>


<h3>See Also</h3>

<p><code><a href="shiny.html#topic+runApp">runApp</a></code>
</p>

<hr>
<h2 id='AmpGram_predictions'>Prediction of antimicrobial peptides</h2><span id='topic+AmpGram_predictions'></span>

<h3>Description</h3>

<p>Predictions made with the AmpGram methods.
</p>


<h3>Format</h3>

<p>A list of length three: random forest for 10-mer analysis, 
random forest for predictions of AMPs, and a vector of important n-grams.
</p>

<hr>
<h2 id='get_AMPs'>Get putative antimicrobial peptides</h2><span id='topic+get_AMPs'></span>

<h3>Description</h3>

<p>Function gets sequences recognized as antimicrobial peptides and returns as data.frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_AMPs(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_AMPs_+3A_x">x</code></td>
<td>
<p>AmpGram predictions for a single protein</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame with sequences recognized as antimicrobial peptides (AMPs). 
It consists of two columns:
</p>

<dl>
<dt>putative_AMP</dt><dd><p>amino acid sequence of a 10-mer (subsequence of an analyzed peptide)
predicted as AMP.</p>
</dd>
<dt>prob</dt><dd><p>Probability with which a 10-mer is recognized as AMP.</p>
</dd></dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(AmpGram_predictions)
get_AMPs(AmpGram_predictions[[2]])
</code></pre>

<hr>
<h2 id='install_AmpGramModel'>Install AmpGramModel package containing model for AMP prediction</h2><span id='topic+install_AmpGramModel'></span>

<h3>Description</h3>

<p>Installs AmpGramModel package containing model required for prediction
of antimicrobial peptides. Due to large size of our model and file size 
limit on CRAN, it needs to be stored in the external repository.
See readme for more information or in case of installation problems.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>install_AmpGramModel()
</code></pre>

<hr>
<h2 id='pred2df'>Convert predictions to data.frame
Return predictions as data.frame</h2><span id='topic+pred2df'></span>

<h3>Description</h3>

<p>Convert predictions to data.frame
Return predictions as data.frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pred2df(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pred2df_+3A_x">x</code></td>
<td>
<p>results of prediction as produced by <code><a href="#topic+predict.ampgram_model">predict.ampgram_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame with two columns and number of rows corresponding to the
number of peptides/proteins in the results of prediction. Columns contain following
information:
</p>

<dl>
<dt>seq_name</dt><dd><p>Name of an analyzed sequence</p>
</dd>
<dt>probability</dt><dd><p>Probability that a protein/peptide possesses antimicrobial
activity. It assumes values from 0 (non-AMP) to 1 (AMP).</p>
</dd></dl>

<p>Row names contain sequence name and decision if a peptide/protein is classified
as AMP (<code>TRUE</code>) or non-AMP (<code>FALSE</code>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AmpGram_predictions)
pred2df(AmpGram_predictions)
</code></pre>

<hr>
<h2 id='predict.ampgram_model'>Predict antimicrobial peptides</h2><span id='topic+predict.ampgram_model'></span>

<h3>Description</h3>

<p>Recognizes antimicrobial peptides using the AmpGram algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ampgram_model'
predict(object, newdata, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predict.ampgram_model_+3A_object">object</code></td>
<td>
<p><code>ampgram_model</code> object.</p>
</td></tr>
<tr><td><code id="predict.ampgram_model_+3A_newdata">newdata</code></td>
<td>
<p><code>list</code> of sequences (for example as given by
<code><a href="biogram.html#topic+read_fasta">read_fasta</a></code> or <code><a href="#topic+read_txt">read_txt</a></code>).</p>
</td></tr>
<tr><td><code id="predict.ampgram_model_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>AmpGram requires the external package, AmpGramModel, which 
contains models necessary to perform the prediction. The model 
can be installed using <code><a href="#topic+install_AmpGramModel">install_AmpGramModel</a></code>.
</p>
<p>Predictions for each protein are stored in objects of class 
<code>single_ampgram_pred</code>. It consists of three elements:
</p>

<dl>
<dt>seq</dt><dd><p>Character vector of amino acid sequence of an analyzed peptide/protein</p>
</dd>
<dt>all_mers_pred</dt><dd><p>Numeric vector of predictions for each 10-mer (subsequence
of 10 amino acids) of a sequence. Prediction value indicates probability that
a 10-mer possesses antimicrobial activity and ranges from 0 (non-AMP) to 1 
(AMP).</p>
</dd>
<dt>single_prot_pred</dt><dd><p>Named numeric vector of a single prediction value for
a whole peptide/protein. Its value corresponds to the probability that a
peptide/protein exhibits antimicrobial activity. It assumes name <code>TRUE</code> 
if probability is equal or greater than 0.5, i.e. peptide/protein is classified
as antimicrobial (AMP), and <code>FALSE</code> if probability is less that 0.5,
i.e. peptide/protein is classified as non-antimicrobial (non-AMP).</p>
</dd>
</dl>



<h3>Value</h3>

<p><code>list</code> of objects of class <code>single_ampgram_pred</code>. Each object 
of this class contains analyzed sequence, values of predictions for 10-mers and 
result of the prediction for the whole peptide/protein.
</p>

<hr>
<h2 id='read_txt'>Read sequences from .txt file</h2><span id='topic+read_txt'></span>

<h3>Description</h3>

<p>Read sequence data saved in text file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_txt(connection)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_txt_+3A_connection">connection</code></td>
<td>
<p>a <code><a href="base.html#topic+connection">connection</a></code> to the text (.txt) file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input file should contain one or more amino acid sequences separated by 
empty line(s).
</p>


<h3>Value</h3>

<p>a list of sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>(sequences &lt;- read_txt(system.file("AmpGram/prots.txt", package = "AmpGram")))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
