<!DOCTYPE html><html><head><title>Help for package ampir</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ampir}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#aaseq_is_valid'><p>Check protein sequences for non-standard amino acids</p></a></li>
<li><a href='#calc_amphiphilicity'><p>Calculate amphiphilicity (or hydrophobic moment)</p></a></li>
<li><a href='#calc_hydrophobicity'><p>Calculate the hydrophobicity</p></a></li>
<li><a href='#calc_mw'><p>Calculate the molecular weight</p></a></li>
<li><a href='#calc_net_charge'><p>Calculate the net charge</p></a></li>
<li><a href='#calc_pI'><p>Calculate the isoelectric point (pI)</p></a></li>
<li><a href='#calc_pseudo_comp'><p>Calculate the pseudo amino acid composition</p></a></li>
<li><a href='#calculate_features'><p>Calculate a set of numerical features from protein sequences</p></a></li>
<li><a href='#chunk_rows'><p>Determine row breakpoints for dividing a dataset into chunks for parallel processing</p></a></li>
<li><a href='#df_to_faa'><p>Save a dataframe in FASTA format</p></a></li>
<li><a href='#predict_amps'><p>Predict the antimicrobial peptide probability of a protein</p></a></li>
<li><a href='#read_faa'><p>Read FASTA amino acids file into a dataframe</p></a></li>
<li><a href='#remove_nonstandard_aa'><p>Remove non standard amino acids from protein sequences</p></a></li>
<li><a href='#remove_stop_codon'><p>Remove stop codon at end of sequence</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Predict Antimicrobial Peptides</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-06-29</td>
</tr>
<tr>
<td>Description:</td>
<td>A toolkit to predict antimicrobial peptides from protein sequences on a genome-wide scale.
    It incorporates two support vector machine models ("precursor" and "mature") trained on publicly available antimicrobial peptide data using calculated
    physico-chemical and compositional sequence properties described in Meher et al. (2017) &lt;<a href="https://doi.org/10.1038%2Fsrep42362">doi:10.1038/srep42362</a>&gt;.
    In order to support genome-wide analyses, these models are designed to accept any type of protein as input
    and calculation of compositional properties has been optimised for high-throughput use. For best results it is important to select the model that accurately 
    represents your sequence type: for full length proteins, it is recommended to use the default "precursor" model. The alternative, "mature", model is best suited
    for mature peptide sequences that represent the final antimicrobial peptide sequence after post-translational processing. For details see Fingerhut et al. (2020) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtaa653">doi:10.1093/bioinformatics/btaa653</a>&gt;.
    The 'ampir' package is also available via a Shiny based GUI at <a href="https://ampir.marine-omics.net/">https://ampir.marine-omics.net/</a>.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/Legana/ampir">https://github.com/Legana/ampir</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Peptides, caret (&ge; 6.0.0), kernlab, Rcpp, parallel</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), knitr, rmarkdown, e1071</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-06-29 04:35:58 UTC; leganafingerhut</td>
</tr>
<tr>
<td>Author:</td>
<td>Legana Fingerhut <a href="https://orcid.org/0000-0002-2482-5336"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Ira Cooke <a href="https://orcid.org/0000-0001-6520-1397"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Jinlong Zhang [ctb] (R/read_faa.R),
  Nan Xiao [ctb] (R/calc_pseudo_comp.R)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Legana Fingerhut &lt;legana.fingerhut@my.jcu.edu.au&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-06-29 07:10:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='aaseq_is_valid'>Check protein sequences for non-standard amino acids</h2><span id='topic+aaseq_is_valid'></span>

<h3>Description</h3>

<p>Any proteins that contains an amino acid that is not one of the 20 standard amino acids is flagged as invalid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aaseq_is_valid(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aaseq_is_valid_+3A_seq">seq</code></td>
<td>
<p>A vector of protein sequences</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical vector where TRUE indicates a valid protein sequence and FALSE indicates a sequence with invalid amino acids
</p>

<hr>
<h2 id='calc_amphiphilicity'>Calculate amphiphilicity (or hydrophobic moment)</h2><span id='topic+calc_amphiphilicity'></span>

<h3>Description</h3>

<p>Calculate amphiphilicity (or hydrophobic moment)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_amphiphilicity(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_amphiphilicity_+3A_seq">seq</code></td>
<td>
<p>A protein sequence</p>
</td></tr>
</table>


<h3>References</h3>

<p>Osorio, D., Rondon-Villarreal, P. &amp; Torres, R. Peptides: A package for data mining of antimicrobial peptides. The R Journal. 7(1), 4–14 (2015).
The imported function originates from the Peptides package (https://github.com/dosorio/Peptides/).
</p>

<hr>
<h2 id='calc_hydrophobicity'>Calculate the hydrophobicity</h2><span id='topic+calc_hydrophobicity'></span>

<h3>Description</h3>

<p>Calculate the hydrophobicity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_hydrophobicity(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_hydrophobicity_+3A_seq">seq</code></td>
<td>
<p>A protein sequence</p>
</td></tr>
</table>


<h3>References</h3>

<p>Osorio, D., Rondon-Villarreal, P. &amp; Torres, R. Peptides: A package for data mining of antimicrobial peptides. The R Journal. 7(1), 4–14 (2015).
The imported function originates from the Peptides package (https://github.com/dosorio/Peptides/).
</p>

<hr>
<h2 id='calc_mw'>Calculate the molecular weight</h2><span id='topic+calc_mw'></span>

<h3>Description</h3>

<p>Calculate the molecular weight
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_mw(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_mw_+3A_seq">seq</code></td>
<td>
<p>A protein sequence</p>
</td></tr>
</table>


<h3>References</h3>

<p>Osorio, D., Rondon-Villarreal, P. &amp; Torres, R. Peptides: A package for data mining of antimicrobial peptides. The R Journal. 7(1), 4–14 (2015).
The imported function originates from the Peptides package (https://github.com/dosorio/Peptides/).
</p>

<hr>
<h2 id='calc_net_charge'>Calculate the net charge</h2><span id='topic+calc_net_charge'></span>

<h3>Description</h3>

<p>Calculate the net charge
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_net_charge(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_net_charge_+3A_seq">seq</code></td>
<td>
<p>A protein sequence</p>
</td></tr>
</table>


<h3>References</h3>

<p>Osorio, D., Rondon-Villarreal, P. &amp; Torres, R. Peptides: A package for data mining of antimicrobial peptides. The R Journal. 7(1), 4–14 (2015).
The imported function originates from the Peptides package (https://github.com/dosorio/Peptides/).
</p>

<hr>
<h2 id='calc_pI'>Calculate the isoelectric point (pI)</h2><span id='topic+calc_pI'></span>

<h3>Description</h3>

<p>Calculate the isoelectric point (pI)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_pI(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_pI_+3A_seq">seq</code></td>
<td>
<p>pI</p>
</td></tr>
</table>


<h3>References</h3>

<p>Osorio, D., Rondon-Villarreal, P. &amp; Torres, R. Peptides: A package for data mining of antimicrobial peptides. The R Journal. 7(1), 4–14 (2015).
The imported function originates from the Peptides package (https://github.com/dosorio/Peptides/).
</p>

<hr>
<h2 id='calc_pseudo_comp'>Calculate the pseudo amino acid composition</h2><span id='topic+calc_pseudo_comp'></span>

<h3>Description</h3>

<p>This function is adapted from the extractPAAC function from the protr package (https://github.com/nanxstats/protr)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_pseudo_comp(seq, lambda_min = 4, lambda_max = 19)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_pseudo_comp_+3A_seq">seq</code></td>
<td>
<p>A vector of protein sequences as character strings</p>
</td></tr>
<tr><td><code id="calc_pseudo_comp_+3A_lambda_min">lambda_min</code></td>
<td>
<p>Minimum allowable lambda. It is an error to provide a protein sequence shorter than lambda_min+1</p>
</td></tr>
<tr><td><code id="calc_pseudo_comp_+3A_lambda_max">lambda_max</code></td>
<td>
<p>For each sequence lambda will be set to one less than the sequence length or lambda_max, whichever is smaller</p>
</td></tr>
</table>


<h3>References</h3>

<p>Nan Xiao, Dong-Sheng Cao, Min-Feng Zhu, and Qing-Song Xu. (2015). protr/ProtrWeb: R package and web server for generating various numerical representation schemes of protein sequences. Bioinformatics 31 (11), 1857-1859.
</p>

<hr>
<h2 id='calculate_features'>Calculate a set of numerical features from protein sequences</h2><span id='topic+calculate_features'></span>

<h3>Description</h3>

<p>This function calculates set physicochemical and compositional features from protein sequences in preparation for supervised model learning
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_features(df, min_len = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_features_+3A_df">df</code></td>
<td>
<p>A dataframe which contains protein sequence names as the first column and amino acid sequence as the second column</p>
</td></tr>
<tr><td><code id="calculate_features_+3A_min_len">min_len</code></td>
<td>
<p>Minimum length sequence for which features can be calculated. It is an error to provide sequences with length shorter than this</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe containing numerical values related to the protein features of each given protein
</p>


<h3>Note</h3>

<p>This function depends on the Peptides package
</p>


<h3>References</h3>

<p>Osorio, D., Rondon-Villarreal, P. &amp; Torres, R. Peptides: A package for data mining of antimicrobial peptides. The R Journal. 7(1), 4–14 (2015).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
my_protein_df &lt;- read_faa(system.file("extdata/bat_protein.fasta", package = "ampir"))

calculate_features(my_protein_df)
## Output (showing the first six output columns)
#      seq_name     Amphiphilicity  Hydrophobicity     pI          Mw       Charge    ....
# [1] G1P6H5_MYOLU	   0.4145847       0.4373494     8.501312     9013.757   4.53015   ....
</code></pre>

<hr>
<h2 id='chunk_rows'>Determine row breakpoints for dividing a dataset into chunks for parallel processing</h2><span id='topic+chunk_rows'></span>

<h3>Description</h3>

<p>Determine row breakpoints for dividing a dataset into chunks for parallel processing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chunk_rows(nrows, n_cores)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chunk_rows_+3A_nrows">nrows</code></td>
<td>
<p>The number of rows in the dataset to be chunked</p>
</td></tr>
<tr><td><code id="chunk_rows_+3A_n_cores">n_cores</code></td>
<td>
<p>The number of cores that will be used for parallel processing</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of integer vectors consisting of the rows in each chunk
</p>

<hr>
<h2 id='df_to_faa'>Save a dataframe in FASTA format</h2><span id='topic+df_to_faa'></span>

<h3>Description</h3>

<p>This function writes a dataframe out as a FASTA format file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>df_to_faa(df, file = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="df_to_faa_+3A_df">df</code></td>
<td>
<p>a dataframe containing two columns: the sequence name and amino acid sequence itself</p>
</td></tr>
<tr><td><code id="df_to_faa_+3A_file">file</code></td>
<td>
<p>file path to save the named file to</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A FASTA file where protein sequences are represented in two lines: The protein name preceded by a greater than symbol,
and a new second line that contains the protein sequence
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
my_protein &lt;- read_faa(system.file("extdata/bat_protein.fasta", package = "ampir"))

# Write a dataframe to a FASTA file
df_to_faa(my_protein, tempfile("my_protein.fasta", tempdir()))


</code></pre>

<hr>
<h2 id='predict_amps'>Predict the antimicrobial peptide probability of a protein</h2><span id='topic+predict_amps'></span>

<h3>Description</h3>

<p>This function predicts the probability of a protein to be an antimicrobial peptide
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predict_amps(faa_df, min_len = 5, n_cores = 1, model = "precursor")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict_amps_+3A_faa_df">faa_df</code></td>
<td>
<p>A dataframe obtained from <code>read_faa</code> containing two columns: the sequence name (seq_name) and amino acid sequence (seq_aa)</p>
</td></tr>
<tr><td><code id="predict_amps_+3A_min_len">min_len</code></td>
<td>
<p>The minimum protein length for which predictions will be generated</p>
</td></tr>
<tr><td><code id="predict_amps_+3A_n_cores">n_cores</code></td>
<td>
<p>On multicore machines split the task across this many processors. This option does not work on Windows</p>
</td></tr>
<tr><td><code id="predict_amps_+3A_model">model</code></td>
<td>
<p>Either a string with the name of a built-in model (mature, precursor), OR, A train object suitable for passing to the predict.train function in the caret package. If omitted the default model will be used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The original input data.frame with a new column added called <code>prob_AMP</code> with the probability of that sequence to be an antimicrobial peptide. Any sequences that are too short or which contain invalid amin acids will have NA in this column
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
my_bat_faa_df &lt;- read_faa(system.file("extdata/bat_protein.fasta", package = "ampir"))

predict_amps(my_bat_faa_df)
#       seq_name    prob_AMP
# [1] G1P6H5_MYOLU  0.9723796
</code></pre>

<hr>
<h2 id='read_faa'>Read FASTA amino acids file into a dataframe</h2><span id='topic+read_faa'></span>

<h3>Description</h3>

<p>This function reads a FASTA amino acids file into a dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_faa(file = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_faa_+3A_file">file</code></td>
<td>
<p>file path to the FASTA format file containing the protein sequences</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe containing the sequence name (seq_name) and sequence (seq_aa) columns
</p>


<h3>Note</h3>

<p>This function was adapted from 'read.fasta.R' by Jinlong Zhang (jinlongzhang01@gmail.com) for the phylotools package (http://github.com/helixcn/phylotools)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
read_faa(system.file("extdata/bat_protein.fasta", package = "ampir"))

## Output
#         seq_name              seq_aa
# [1] G1P6H5_MYOLU  MALTVRIQAACLLLLLLASLTSYSL....
</code></pre>

<hr>
<h2 id='remove_nonstandard_aa'>Remove non standard amino acids from protein sequences</h2><span id='topic+remove_nonstandard_aa'></span>

<h3>Description</h3>

<p>This function removes anything that is not one of the 20 standard amino acids in protein sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_nonstandard_aa(df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="remove_nonstandard_aa_+3A_df">df</code></td>
<td>
<p>A dataframe which contains protein sequence names as the first column and amino acid sequence as the second column</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe like the input dataframe but with removed proteins that contained non standard amino acids
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
non_standard_df &lt;- readRDS(system.file("extdata/non_standard_df.rds", package = "ampir"))

# non_standard_df
#       seq_name            seq_aa
# [1] G1P6H5_MYOLU    MALTVRIQAACLLLLLLASLTSYSLLLSQTTQLADLQTQ....
# [2] fake_sequence   MKVTHEUSYR$GXMBIJIDG*M80-%

remove_nonstandard_aa(non_standard_df)
#       seq_name        seq_aa
# [1] G1P6H5_MYOLU    MALTVRIQAACLLLLLLASLTSYSLLLSQTTQLADLQTQ....
</code></pre>

<hr>
<h2 id='remove_stop_codon'>Remove stop codon at end of sequence</h2><span id='topic+remove_stop_codon'></span>

<h3>Description</h3>

<p>Stop codons at the end of the amino acid sequences are removed
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_stop_codon(faa_df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="remove_stop_codon_+3A_faa_df">faa_df</code></td>
<td>
<p>A dataframe containing two columns: the sequence name and amino acid sequence</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The input dataframe without the stop codons at the end of sequences
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
