<!DOCTYPE html><html><head><title>Help for package AMR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {AMR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ab_from_text'><p>Retrieve Antimicrobial Drug Names and Doses from Clinical Text</p></a></li>
<li><a href='#ab_property'><p>Get Properties of an Antibiotic</p></a></li>
<li><a href='#add_custom_antimicrobials'><p>Add Custom Antimicrobials</p></a></li>
<li><a href='#add_custom_microorganisms'><p>Add Custom Microorganisms</p></a></li>
<li><a href='#age'><p>Age in Years of Individuals</p></a></li>
<li><a href='#age_groups'><p>Split Ages into Age Groups</p></a></li>
<li><a href='#AMR'><p>The <code>AMR</code> Package</p></a></li>
<li><a href='#AMR-deprecated'><p>Deprecated Functions</p></a></li>
<li><a href='#AMR-options'><p>Options for the AMR package</p></a></li>
<li><a href='#antibiogram'><p>Generate Antibiogram: Traditional, Combined, Syndromic, or Weighted-Incidence Syndromic Combination (WISCA)</p></a></li>
<li><a href='#antibiotic_class_selectors'><p>Antibiotic Selectors</p></a></li>
<li><a href='#antibiotics'><p>Data Sets with 603 Antimicrobial Drugs</p></a></li>
<li><a href='#as.ab'><p>Transform Input to an Antibiotic ID</p></a></li>
<li><a href='#as.av'><p>Transform Input to an Antiviral Drug ID</p></a></li>
<li><a href='#as.disk'><p>Transform Input to Disk Diffusion Diameters</p></a></li>
<li><a href='#as.mic'><p>Transform Input to Minimum Inhibitory Concentrations (MIC)</p></a></li>
<li><a href='#as.mo'><p>Transform Arbitrary Input to Valid Microbial Taxonomy</p></a></li>
<li><a href='#as.sir'><p>Translate MIC and Disk Diffusion to SIR, or Clean Existing SIR Data</p></a></li>
<li><a href='#atc_online_property'><p>Get ATC Properties from WHOCC Website</p></a></li>
<li><a href='#av_from_text'><p>Retrieve Antiviral Drug Names and Doses from Clinical Text</p></a></li>
<li><a href='#av_property'><p>Get Properties of an Antiviral Drug</p></a></li>
<li><a href='#availability'><p>Check Availability of Columns</p></a></li>
<li><a href='#bug_drug_combinations'><p>Determine Bug-Drug Combinations</p></a></li>
<li><a href='#clinical_breakpoints'><p>Data Set with Clinical Breakpoints for SIR Interpretation</p></a></li>
<li><a href='#count'><p>Count Available Isolates</p></a></li>
<li><a href='#custom_eucast_rules'><p>Define Custom EUCAST Rules</p></a></li>
<li><a href='#dosage'><p>Data Set with Treatment Dosages as Defined by EUCAST</p></a></li>
<li><a href='#eucast_rules'><p>Apply EUCAST Rules</p></a></li>
<li><a href='#example_isolates'><p>Data Set with 2 000 Example Isolates</p></a></li>
<li><a href='#example_isolates_unclean'><p>Data Set with Unclean Data</p></a></li>
<li><a href='#first_isolate'><p>Determine First Isolates</p></a></li>
<li><a href='#g.test'><p><em>G</em>-test for Count Data</p></a></li>
<li><a href='#get_episode'><p>Determine Clinical or Epidemic Episodes</p></a></li>
<li><a href='#ggplot_pca'><p>PCA Biplot with <code>ggplot2</code></p></a></li>
<li><a href='#ggplot_sir'><p>AMR Plots with <code>ggplot2</code></p></a></li>
<li><a href='#guess_ab_col'><p>Guess Antibiotic Column</p></a></li>
<li><a href='#intrinsic_resistant'><p>Data Set with Bacterial Intrinsic Resistance</p></a></li>
<li><a href='#italicise_taxonomy'><p>Italicise Taxonomic Families, Genera, Species, Subspecies</p></a></li>
<li><a href='#join'><p>Join microorganisms to a Data Set</p></a></li>
<li><a href='#key_antimicrobials'><p>(Key) Antimicrobials for First Weighted Isolates</p></a></li>
<li><a href='#kurtosis'><p>Kurtosis of the Sample</p></a></li>
<li><a href='#like'><p>Vectorised Pattern Matching with Keyboard Shortcut</p></a></li>
<li><a href='#mdro'><p>Determine Multidrug-Resistant Organisms (MDRO)</p></a></li>
<li><a href='#mean_amr_distance'><p>Calculate the Mean AMR Distance</p></a></li>
<li><a href='#microorganisms'><p>Data Set with 52 171 Microorganisms</p></a></li>
<li><a href='#microorganisms.codes'><p>Data Set with 4 957 Common Microorganism Codes</p></a></li>
<li><a href='#microorganisms.groups'><p>Data Set with 521 Microorganisms In Species Groups</p></a></li>
<li><a href='#mo_matching_score'><p>Calculate the Matching Score for Microorganisms</p></a></li>
<li><a href='#mo_property'><p>Get Properties of a Microorganism</p></a></li>
<li><a href='#mo_source'><p>User-Defined Reference Data Set for Microorganisms</p></a></li>
<li><a href='#pca'><p>Principal Component Analysis (for AMR)</p></a></li>
<li><a href='#plot'><p>Plotting for Classes <code>sir</code>, <code>mic</code> and <code>disk</code></p></a></li>
<li><a href='#proportion'><p>Calculate Antimicrobial Resistance</p></a></li>
<li><a href='#random'><p>Random MIC Values/Disk Zones/SIR Generation</p></a></li>
<li><a href='#resistance_predict'><p>Predict Antimicrobial Resistance</p></a></li>
<li><a href='#skewness'><p>Skewness of the Sample</p></a></li>
<li><a href='#translate'><p>Translate Strings from the AMR Package</p></a></li>
<li><a href='#WHOCC'><p>WHOCC: WHO Collaborating Centre for Drug Statistics Methodology</p></a></li>
<li><a href='#WHONET'><p>Data Set with 500 Isolates - WHONET Example</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>2.1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-10-20</td>
</tr>
<tr>
<td>Title:</td>
<td>Antimicrobial Resistance Data Analysis</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions to simplify and standardise antimicrobial resistance (AMR)
  data analysis and to work with microbial and antimicrobial properties by
  using evidence-based methods, as described in &lt;<a href="https://doi.org/10.18637%2Fjss.v104.i03">doi:10.18637/jss.v104.i03</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Enhances:</td>
<td>cleaner, ggplot2, janitor, skimr, tibble, tidyselect, tsibble</td>
</tr>
<tr>
<td>Suggests:</td>
<td>cli, curl, data.table, dplyr, knitr, progress, readxl,
rmarkdown, rvest, tinytest, xml2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr,rmarkdown</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://msberends.github.io/AMR/">https://msberends.github.io/AMR/</a>, <a href="https://github.com/msberends/AMR">https://github.com/msberends/AMR</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/msberends/AMR/issues">https://github.com/msberends/AMR/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-20 16:04:38 UTC; msberends</td>
</tr>
<tr>
<td>Author:</td>
<td>Matthijs S. Berends
    <a href="https://orcid.org/0000-0001-7620-1800"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Christian F. Luz <a href="https://orcid.org/0000-0001-5809-5995"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, ctb],
  Dennis Souverein <a href="https://orcid.org/0000-0003-0455-0336"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, ctb],
  Erwin E. A. Hassing [aut, ctb],
  Casper J. Albers <a href="https://orcid.org/0000-0002-9213-6743"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ths],
  Peter Dutey-Magni <a href="https://orcid.org/0000-0002-8942-9836"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Judith M. Fonville [ctb],
  Alex W. Friedrich <a href="https://orcid.org/0000-0003-4881-038X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ths],
  Corinna Glasner <a href="https://orcid.org/0000-0003-1241-1328"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ths],
  Eric H. L. C. M. Hazenberg [ctb],
  Gwen Knight <a href="https://orcid.org/0000-0002-7263-9896"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Annick Lenglet <a href="https://orcid.org/0000-0003-2013-8405"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Bart C. Meijer [ctb],
  Dmytro Mykhailenko [ctb],
  Anton Mymrikov [ctb],
  Andrew P. Norgan <a href="https://orcid.org/0000-0002-2955-2066"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Sofia Ny <a href="https://orcid.org/0000-0002-2017-1363"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Jonas Salm [ctb],
  Rogier P. Schade [ctb],
  Bhanu N. M. Sinha <a href="https://orcid.org/0000-0003-1634-0010"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ths],
  Anthony Underwood <a href="https://orcid.org/0000-0002-8547-4277"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Anita Williams <a href="https://orcid.org/0000-0002-5295-8451"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Matthijs S. Berends &lt;m.s.berends@umcg.nl&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-21 23:10:16 UTC</td>
</tr>
</table>
<hr>
<h2 id='ab_from_text'>Retrieve Antimicrobial Drug Names and Doses from Clinical Text</h2><span id='topic+ab_from_text'></span>

<h3>Description</h3>

<p>Use this function on e.g. clinical texts from health care records. It returns a <a href="base.html#topic+list">list</a> with all antimicrobial drugs, doses and forms of administration found in the texts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ab_from_text(
  text,
  type = c("drug", "dose", "administration"),
  collapse = NULL,
  translate_ab = FALSE,
  thorough_search = NULL,
  info = interactive(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ab_from_text_+3A_text">text</code></td>
<td>
<p>text to analyse</p>
</td></tr>
<tr><td><code id="ab_from_text_+3A_type">type</code></td>
<td>
<p>type of property to search for, either <code>"drug"</code>, <code>"dose"</code> or <code>"administration"</code>, see <em>Examples</em></p>
</td></tr>
<tr><td><code id="ab_from_text_+3A_collapse">collapse</code></td>
<td>
<p>a <a href="base.html#topic+character">character</a> to pass on to <code>paste(, collapse = ...)</code> to only return one <a href="base.html#topic+character">character</a> per element of <code>text</code>, see <em>Examples</em></p>
</td></tr>
<tr><td><code id="ab_from_text_+3A_translate_ab">translate_ab</code></td>
<td>
<p>if <code>type = "drug"</code>: a column name of the <a href="#topic+antibiotics">antibiotics</a> data set to translate the antibiotic abbreviations to, using <code><a href="#topic+ab_property">ab_property()</a></code>. The default is <code>FALSE</code>. Using <code>TRUE</code> is equal to using &quot;name&quot;.</p>
</td></tr>
<tr><td><code id="ab_from_text_+3A_thorough_search">thorough_search</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether the input must be extensively searched for misspelling and other faulty input values. Setting this to <code>TRUE</code> will take considerably more time than when using <code>FALSE</code>. At default, it will turn <code>TRUE</code> when all input elements contain a maximum of three words.</p>
</td></tr>
<tr><td><code id="ab_from_text_+3A_info">info</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether a progress bar should be printed - the default is <code>TRUE</code> only in interactive mode</p>
</td></tr>
<tr><td><code id="ab_from_text_+3A_...">...</code></td>
<td>
<p>arguments passed on to <code><a href="#topic+as.ab">as.ab()</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is also internally used by <code><a href="#topic+as.ab">as.ab()</a></code>, although it then only searches for the first drug name and will throw a note if more drug names could have been returned. Note: the <code><a href="#topic+as.ab">as.ab()</a></code> function may use very long regular expression to match brand names of antimicrobial drugs. This may fail on some systems.
</p>


<h4>Argument <code>type</code></h4>

<p>At default, the function will search for antimicrobial drug names. All text elements will be searched for official names, ATC codes and brand names. As it uses <code><a href="#topic+as.ab">as.ab()</a></code> internally, it will correct for misspelling.
</p>
<p>With <code>type = "dose"</code> (or similar, like &quot;dosing&quot;, &quot;doses&quot;), all text elements will be searched for <a href="base.html#topic+numeric">numeric</a> values that are higher than 100 and do not resemble years. The output will be <a href="base.html#topic+numeric">numeric</a>. It supports any unit (g, mg, IE, etc.) and multiple values in one clinical text, see <em>Examples</em>.
</p>
<p>With <code>type = "administration"</code> (or abbreviations, like &quot;admin&quot;, &quot;adm&quot;), all text elements will be searched for a form of drug administration. It supports the following forms (including common abbreviations): buccal, implant, inhalation, instillation, intravenous, nasal, oral, parenteral, rectal, sublingual, transdermal and vaginal. Abbreviations for oral (such as 'po', 'per os') will become &quot;oral&quot;, all values for intravenous (such as 'iv', 'intraven') will become &quot;iv&quot;. It supports multiple values in one clinical text, see <em>Examples</em>.
</p>



<h4>Argument <code>collapse</code></h4>

<p>Without using <code>collapse</code>, this function will return a <a href="base.html#topic+list">list</a>. This can be convenient to use e.g. inside a <code>mutate()</code>):<br />
<code>df %&gt;% mutate(abx = ab_from_text(clinical_text))</code>
</p>
<p>The returned AB codes can be transformed to official names, groups, etc. with all <code><a href="#topic+ab_property">ab_*</a></code> functions such as <code><a href="#topic+ab_name">ab_name()</a></code> and <code><a href="#topic+ab_group">ab_group()</a></code>, or by using the <code>translate_ab</code> argument.
</p>
<p>With using <code>collapse</code>, this function will return a <a href="base.html#topic+character">character</a>:<br />
<code>df %&gt;% mutate(abx = ab_from_text(clinical_text, collapse = "|"))</code>
</p>



<h3>Value</h3>

<p>A <a href="base.html#topic+list">list</a>, or a  <a href="base.html#topic+character">character</a> if <code>collapse</code> is not <code>NULL</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># mind the bad spelling of amoxicillin in this line,
# straight from a true health care record:
ab_from_text("28/03/2020 regular amoxicilliin 500mg po tid")

ab_from_text("500 mg amoxi po and 400mg cipro iv")
ab_from_text("500 mg amoxi po and 400mg cipro iv", type = "dose")
ab_from_text("500 mg amoxi po and 400mg cipro iv", type = "admin")

ab_from_text("500 mg amoxi po and 400mg cipro iv", collapse = ", ")

# if you want to know which antibiotic groups were administered, do e.g.:
abx &lt;- ab_from_text("500 mg amoxi po and 400mg cipro iv")
ab_group(abx[[1]])

if (require("dplyr")) {
  tibble(clinical_text = c(
    "given 400mg cipro and 500 mg amox",
    "started on doxy iv today"
  )) %&gt;%
    mutate(
      abx_codes = ab_from_text(clinical_text),
      abx_doses = ab_from_text(clinical_text, type = "doses"),
      abx_admin = ab_from_text(clinical_text, type = "admin"),
      abx_coll = ab_from_text(clinical_text, collapse = "|"),
      abx_coll_names = ab_from_text(clinical_text,
        collapse = "|",
        translate_ab = "name"
      ),
      abx_coll_doses = ab_from_text(clinical_text,
        type = "doses",
        collapse = "|"
      ),
      abx_coll_admin = ab_from_text(clinical_text,
        type = "admin",
        collapse = "|"
      )
    )
}

</code></pre>

<hr>
<h2 id='ab_property'>Get Properties of an Antibiotic</h2><span id='topic+ab_property'></span><span id='topic+ab_name'></span><span id='topic+ab_cid'></span><span id='topic+ab_synonyms'></span><span id='topic+ab_tradenames'></span><span id='topic+ab_group'></span><span id='topic+ab_atc'></span><span id='topic+ATC'></span><span id='topic+ab_atc_group1'></span><span id='topic+ab_atc_group2'></span><span id='topic+ab_loinc'></span><span id='topic+ab_ddd'></span><span id='topic+ab_ddd_units'></span><span id='topic+ab_info'></span><span id='topic+ab_url'></span><span id='topic+set_ab_names'></span>

<h3>Description</h3>

<p>Use these functions to return a specific property of an antibiotic from the <a href="#topic+antibiotics">antibiotics</a> data set. All input values will be evaluated internally with <code><a href="#topic+as.ab">as.ab()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ab_name(x, language = get_AMR_locale(), tolower = FALSE, ...)

ab_cid(x, ...)

ab_synonyms(x, ...)

ab_tradenames(x, ...)

ab_group(x, language = get_AMR_locale(), ...)

ab_atc(x, only_first = FALSE, ...)

ab_atc_group1(x, language = get_AMR_locale(), ...)

ab_atc_group2(x, language = get_AMR_locale(), ...)

ab_loinc(x, ...)

ab_ddd(x, administration = "oral", ...)

ab_ddd_units(x, administration = "oral", ...)

ab_info(x, language = get_AMR_locale(), ...)

ab_url(x, open = FALSE, ...)

ab_property(x, property = "name", language = get_AMR_locale(), ...)

set_ab_names(
  data,
  ...,
  property = "name",
  language = get_AMR_locale(),
  snake_case = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ab_property_+3A_x">x</code></td>
<td>
<p>any (vector of) text that can be coerced to a valid antibiotic drug code with <code><a href="#topic+as.ab">as.ab()</a></code></p>
</td></tr>
<tr><td><code id="ab_property_+3A_language">language</code></td>
<td>
<p>language of the returned text - the default is the current system language (see <code><a href="#topic+get_AMR_locale">get_AMR_locale()</a></code>) and can also be set with the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_locale</a></code>. Use <code>language = NULL</code> or <code>language = ""</code> to prevent translation.</p>
</td></tr>
<tr><td><code id="ab_property_+3A_tolower">tolower</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether the first <a href="base.html#topic+character">character</a> of every output should be transformed to a lower case <a href="base.html#topic+character">character</a>. This will lead to e.g. &quot;polymyxin B&quot; and not &quot;polymyxin b&quot;.</p>
</td></tr>
<tr><td><code id="ab_property_+3A_...">...</code></td>
<td>
<p>in case of <code><a href="#topic+set_ab_names">set_ab_names()</a></code> and <code>data</code> is a <a href="base.html#topic+data.frame">data.frame</a>: columns to select (supports tidy selection such as <code>column1:column4</code>), otherwise other arguments passed on to <code><a href="#topic+as.ab">as.ab()</a></code></p>
</td></tr>
<tr><td><code id="ab_property_+3A_only_first">only_first</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether only the first ATC code must be returned, with giving preference to J0-codes (i.e., the antimicrobial drug group)</p>
</td></tr>
<tr><td><code id="ab_property_+3A_administration">administration</code></td>
<td>
<p>way of administration, either <code>"oral"</code> or <code>"iv"</code></p>
</td></tr>
<tr><td><code id="ab_property_+3A_open">open</code></td>
<td>
<p>browse the URL using <code><a href="utils.html#topic+browseURL">utils::browseURL()</a></code></p>
</td></tr>
<tr><td><code id="ab_property_+3A_property">property</code></td>
<td>
<p>one of the column names of one of the <a href="#topic+antibiotics">antibiotics</a> data set: <code>vector_or(colnames(antibiotics), sort = FALSE)</code>.</p>
</td></tr>
<tr><td><code id="ab_property_+3A_data">data</code></td>
<td>
<p>a <a href="base.html#topic+data.frame">data.frame</a> of which the columns need to be renamed, or a <a href="base.html#topic+character">character</a> vector of column names</p>
</td></tr>
<tr><td><code id="ab_property_+3A_snake_case">snake_case</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether the names should be in so-called <a href="https://en.wikipedia.org/wiki/Snake_case">snake case</a>: in lower case and all spaces/slashes replaced with an underscore (<code style="white-space: pre;">&#8288;_&#8288;</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All output <a href="#topic+translate">will be translated</a> where possible.
</p>
<p>The function <code><a href="#topic+ab_url">ab_url()</a></code> will return the direct URL to the official WHO website. A warning will be returned if the required ATC code is not available.
</p>
<p>The function <code><a href="#topic+set_ab_names">set_ab_names()</a></code> is a special column renaming function for <a href="base.html#topic+data.frame">data.frame</a>s. It renames columns names that resemble antimicrobial drugs. It always makes sure that the new column names are unique. If <code>property = "atc"</code> is set, preference is given to ATC codes from the J-group.
</p>


<h3>Value</h3>


<ul>
<li><p> An <a href="base.html#topic+integer">integer</a> in case of <code><a href="#topic+ab_cid">ab_cid()</a></code>
</p>
</li>
<li><p> A named <a href="base.html#topic+list">list</a> in case of <code><a href="#topic+ab_info">ab_info()</a></code> and multiple <code><a href="#topic+ab_atc">ab_atc()</a></code>/<code><a href="#topic+ab_synonyms">ab_synonyms()</a></code>/<code><a href="#topic+ab_tradenames">ab_tradenames()</a></code>
</p>
</li>
<li><p> A <a href="base.html#topic+double">double</a> in case of <code><a href="#topic+ab_ddd">ab_ddd()</a></code>
</p>
</li>
<li><p> A <a href="base.html#topic+data.frame">data.frame</a> in case of <code><a href="#topic+set_ab_names">set_ab_names()</a></code>
</p>
</li>
<li><p> A <a href="base.html#topic+character">character</a> in all other cases
</p>
</li></ul>



<h3>Source</h3>

<p>World Health Organization (WHO) Collaborating Centre for Drug Statistics Methodology: <a href="https://www.whocc.no/atc_ddd_index/">https://www.whocc.no/atc_ddd_index/</a>
</p>
<p>European Commission Public Health PHARMACEUTICALS - COMMUNITY REGISTER: <a href="https://ec.europa.eu/health/documents/community-register/html/reg_hum_atc.htm">https://ec.europa.eu/health/documents/community-register/html/reg_hum_atc.htm</a>
</p>


<h3>Reference Data Publicly Available</h3>

<p>All data sets in this <code>AMR</code> package (about microorganisms, antibiotics, SIR interpretation, EUCAST rules, etc.) are publicly and freely available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. We also provide tab-separated plain text files that are machine-readable and suitable for input in any software program, such as laboratory information systems. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>.
</p>


<h3>See Also</h3>

<p><a href="#topic+antibiotics">antibiotics</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># all properties:
ab_name("AMX")
ab_atc("AMX")
ab_cid("AMX")
ab_synonyms("AMX")
ab_tradenames("AMX")
ab_group("AMX")
ab_atc_group1("AMX")
ab_atc_group2("AMX")
ab_url("AMX")

# smart lowercase transformation
ab_name(x = c("AMC", "PLB"))
ab_name(x = c("AMC", "PLB"), tolower = TRUE)

# defined daily doses (DDD)
ab_ddd("AMX", "oral")
ab_ddd_units("AMX", "oral")
ab_ddd("AMX", "iv")
ab_ddd_units("AMX", "iv")

ab_info("AMX") # all properties as a list

# all ab_* functions use as.ab() internally, so you can go from 'any' to 'any':
ab_atc("AMP")
ab_group("J01CA01")
ab_loinc("ampicillin")
ab_name("21066-6")
ab_name(6249)
ab_name("J01CA01")

# spelling from different languages and dyslexia are no problem
ab_atc("ceftriaxon")
ab_atc("cephtriaxone")
ab_atc("cephthriaxone")
ab_atc("seephthriaaksone")

# use set_ab_names() for renaming columns
colnames(example_isolates)
colnames(set_ab_names(example_isolates))
colnames(set_ab_names(example_isolates, NIT:VAN))

if (require("dplyr")) {
  example_isolates %&gt;%
    set_ab_names()

  # this does the same:
  example_isolates %&gt;%
    rename_with(set_ab_names)

  # set_ab_names() works with any AB property:
  example_isolates %&gt;%
    set_ab_names(property = "atc")

  example_isolates %&gt;%
    set_ab_names(where(is.sir)) %&gt;%
    colnames()

  example_isolates %&gt;%
    set_ab_names(NIT:VAN) %&gt;%
    colnames()
}

</code></pre>

<hr>
<h2 id='add_custom_antimicrobials'>Add Custom Antimicrobials</h2><span id='topic+add_custom_antimicrobials'></span><span id='topic+clear_custom_antimicrobials'></span>

<h3>Description</h3>

<p>With <code><a href="#topic+add_custom_antimicrobials">add_custom_antimicrobials()</a></code> you can add your own custom antimicrobial drug names and codes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_custom_antimicrobials(x)

clear_custom_antimicrobials()
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_custom_antimicrobials_+3A_x">x</code></td>
<td>
<p>a <a href="base.html#topic+data.frame">data.frame</a> resembling the <a href="#topic+antibiotics">antibiotics</a> data set, at least containing columns &quot;ab&quot; and &quot;name&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Important:</strong> Due to how <span class="rlang"><b>R</b></span> works, the <code><a href="#topic+add_custom_antimicrobials">add_custom_antimicrobials()</a></code> function has to be run in every <span class="rlang"><b>R</b></span> session - added antimicrobials are not stored between sessions and are thus lost when <span class="rlang"><b>R</b></span> is exited.
</p>
<p>There are two ways to circumvent this and automate the process of adding antimicrobials:
</p>
<p><strong>Method 1:</strong> Using the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_custom_ab</a></code>, which is the preferred method. To use this method:
</p>

<ol>
<li><p> Create a data set in the structure of the <a href="#topic+antibiotics">antibiotics</a> data set (containing at the very least columns &quot;ab&quot; and &quot;name&quot;) and save it with <code><a href="base.html#topic+saveRDS">saveRDS()</a></code> to a location of choice, e.g. <code>"~/my_custom_ab.rds"</code>, or any remote location.
</p>
</li>
<li><p> Set the file location to the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_custom_ab</a></code>: <code>options(AMR_custom_ab = "~/my_custom_ab.rds")</code>. This can even be a remote file location, such as an https URL. Since options are not saved between <span class="rlang"><b>R</b></span> sessions, it is best to save this option to the <code>.Rprofile</code> file so that it will be loaded on start-up of <span class="rlang"><b>R</b></span>. To do this, open the <code>.Rprofile</code> file using e.g. <code>utils::file.edit("~/.Rprofile")</code>, add this text and save the file:
</p>
<div class="sourceCode r"><pre># Add custom antimicrobial codes:
options(AMR_custom_ab = "~/my_custom_ab.rds")
</pre></div>
<p>Upon package load, this file will be loaded and run through the <code><a href="#topic+add_custom_antimicrobials">add_custom_antimicrobials()</a></code> function.
</p>
</li></ol>

<p><strong>Method 2:</strong> Loading the antimicrobial additions directly from your <code>.Rprofile</code> file. Note that the definitions will be stored in a user-specific <span class="rlang"><b>R</b></span> file, which is a suboptimal workflow. To use this method:
</p>

<ol>
<li><p> Edit the <code>.Rprofile</code> file using e.g. <code>utils::file.edit("~/.Rprofile")</code>.
</p>
</li>
<li><p> Add a text like below and save the file:
</p>
<div class="sourceCode r"><pre> # Add custom antibiotic drug codes:
 AMR::add_custom_antimicrobials(
   data.frame(ab = "TESTAB",
              name = "Test Antibiotic",
              group = "Test Group")
 )
</pre></div>
</li></ol>

<p>Use <code><a href="#topic+clear_custom_antimicrobials">clear_custom_antimicrobials()</a></code> to clear the previously added antimicrobials.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+add_custom_microorganisms">add_custom_microorganisms()</a></code> to add custom microorganisms.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# returns NA and throws a warning (which is suppressed here):
suppressWarnings(
  as.ab("testab")
)

# now add a custom entry - it will be considered by as.ab() and
# all ab_*() functions
add_custom_antimicrobials(
  data.frame(
    ab = "TESTAB",
    name = "Test Antibiotic",
    # you can add any property present in the
    # 'antibiotics' data set, such as 'group':
    group = "Test Group"
  )
)

# "testab" is now a new antibiotic:
as.ab("testab")
ab_name("testab")
ab_group("testab")

ab_info("testab")


# Add Co-fluampicil, which is one of the many J01CR50 codes, see
# https://www.whocc.no/ddd/list_of_ddds_combined_products/
add_custom_antimicrobials(
  data.frame(
    ab = "COFLU",
    name = "Co-fluampicil",
    atc = "J01CR50",
    group = "Beta-lactams/penicillins"
  )
)
ab_atc("Co-fluampicil")
ab_name("J01CR50")

# even antibiotic selectors work
x &lt;- data.frame(
  random_column = "some value",
  coflu = as.sir("S"),
  ampicillin = as.sir("R")
)
x
x[, betalactams()]

</code></pre>

<hr>
<h2 id='add_custom_microorganisms'>Add Custom Microorganisms</h2><span id='topic+add_custom_microorganisms'></span><span id='topic+clear_custom_microorganisms'></span>

<h3>Description</h3>

<p>With <code><a href="#topic+add_custom_microorganisms">add_custom_microorganisms()</a></code> you can add your own custom microorganisms, such the non-taxonomic outcome of laboratory analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_custom_microorganisms(x)

clear_custom_microorganisms()
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_custom_microorganisms_+3A_x">x</code></td>
<td>
<p>a <a href="base.html#topic+data.frame">data.frame</a> resembling the <a href="#topic+microorganisms">microorganisms</a> data set, at least containing column &quot;genus&quot; (case-insensitive)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function will fill in missing taxonomy for you, if specific taxonomic columns are missing, see <em>Examples</em>.
</p>
<p><strong>Important:</strong> Due to how <span class="rlang"><b>R</b></span> works, the <code><a href="#topic+add_custom_microorganisms">add_custom_microorganisms()</a></code> function has to be run in every <span class="rlang"><b>R</b></span> session - added microorganisms are not stored between sessions and are thus lost when <span class="rlang"><b>R</b></span> is exited.
</p>
<p>There are two ways to circumvent this and automate the process of adding microorganisms:
</p>
<p><strong>Method 1:</strong> Using the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_custom_mo</a></code>, which is the preferred method. To use this method:
</p>

<ol>
<li><p> Create a data set in the structure of the <a href="#topic+microorganisms">microorganisms</a> data set (containing at the very least column &quot;genus&quot;) and save it with <code><a href="base.html#topic+saveRDS">saveRDS()</a></code> to a location of choice, e.g. <code>"~/my_custom_mo.rds"</code>, or any remote location.
</p>
</li>
<li><p> Set the file location to the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_custom_mo</a></code>: <code>options(AMR_custom_mo = "~/my_custom_mo.rds")</code>. This can even be a remote file location, such as an https URL. Since options are not saved between <span class="rlang"><b>R</b></span> sessions, it is best to save this option to the <code>.Rprofile</code> file so that it will be loaded on start-up of <span class="rlang"><b>R</b></span>. To do this, open the <code>.Rprofile</code> file using e.g. <code>utils::file.edit("~/.Rprofile")</code>, add this text and save the file:
</p>
<div class="sourceCode r"><pre># Add custom microorganism codes:
options(AMR_custom_mo = "~/my_custom_mo.rds")
</pre></div>
<p>Upon package load, this file will be loaded and run through the <code><a href="#topic+add_custom_microorganisms">add_custom_microorganisms()</a></code> function.
</p>
</li></ol>

<p><strong>Method 2:</strong> Loading the microorganism directly from your <code>.Rprofile</code> file. Note that the definitions will be stored in a user-specific <span class="rlang"><b>R</b></span> file, which is a suboptimal workflow. To use this method:
</p>

<ol>
<li><p> Edit the <code>.Rprofile</code> file using e.g. <code>utils::file.edit("~/.Rprofile")</code>.
</p>
</li>
<li><p> Add a text like below and save the file:
</p>
<div class="sourceCode r"><pre> # Add custom antibiotic drug codes:
 AMR::add_custom_microorganisms(
   data.frame(genus = "Enterobacter",
              species = "asburiae/cloacae")
 )
</pre></div>
</li></ol>

<p>Use <code><a href="#topic+clear_custom_microorganisms">clear_custom_microorganisms()</a></code> to clear the previously added microorganisms.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+add_custom_antimicrobials">add_custom_antimicrobials()</a></code> to add custom antimicrobials.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# a combination of species is not formal taxonomy, so
# this will result in "Enterobacter cloacae cloacae",
# since it resembles the input best:
mo_name("Enterobacter asburiae/cloacae")

# now add a custom entry - it will be considered by as.mo() and
# all mo_*() functions
add_custom_microorganisms(
  data.frame(
    genus = "Enterobacter",
    species = "asburiae/cloacae"
  )
)

# E. asburiae/cloacae is now a new microorganism:
mo_name("Enterobacter asburiae/cloacae")

# its code:
as.mo("Enterobacter asburiae/cloacae")

# all internal algorithms will work as well:
mo_name("Ent asburia cloacae")

# and even the taxonomy was added based on the genus!
mo_family("E. asburiae/cloacae")
mo_gramstain("Enterobacter asburiae/cloacae")

mo_info("Enterobacter asburiae/cloacae")


# the function tries to be forgiving:
add_custom_microorganisms(
  data.frame(
    GENUS = "BACTEROIDES / PARABACTEROIDES SLASHLINE",
    SPECIES = "SPECIES"
  )
)
mo_name("BACTEROIDES / PARABACTEROIDES")
mo_rank("BACTEROIDES / PARABACTEROIDES")

# taxonomy still works, even though a slashline genus was given as input:
mo_family("Bacteroides/Parabacteroides")


# for groups and complexes, set them as species or subspecies:
add_custom_microorganisms(
  data.frame(
    genus = "Citrobacter",
    species = c("freundii", "braakii complex"),
    subspecies = c("complex", "")
  )
)
mo_name(c("C. freundii complex", "C. braakii complex"))
mo_species(c("C. freundii complex", "C. braakii complex"))
mo_gramstain(c("C. freundii complex", "C. braakii complex"))

</code></pre>

<hr>
<h2 id='age'>Age in Years of Individuals</h2><span id='topic+age'></span>

<h3>Description</h3>

<p>Calculates age in years based on a reference date, which is the system date at default.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>age(x, reference = Sys.Date(), exact = FALSE, na.rm = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="age_+3A_x">x</code></td>
<td>
<p>date(s), <a href="base.html#topic+character">character</a> (vectors) will be coerced with <code><a href="base.html#topic+as.POSIXlt">as.POSIXlt()</a></code></p>
</td></tr>
<tr><td><code id="age_+3A_reference">reference</code></td>
<td>
<p>reference date(s) (default is today), <a href="base.html#topic+character">character</a> (vectors) will be coerced with <code><a href="base.html#topic+as.POSIXlt">as.POSIXlt()</a></code></p>
</td></tr>
<tr><td><code id="age_+3A_exact">exact</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether age calculation should be exact, i.e. with decimals. It divides the number of days of <a href="https://en.wikipedia.org/wiki/Year-to-date">year-to-date</a> (YTD) of <code>x</code> by the number of days in the year of <code>reference</code> (either 365 or 366).</p>
</td></tr>
<tr><td><code id="age_+3A_na.rm">na.rm</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether missing values should be removed</p>
</td></tr>
<tr><td><code id="age_+3A_...">...</code></td>
<td>
<p>arguments passed on to <code><a href="base.html#topic+as.POSIXlt">as.POSIXlt()</a></code>, such as <code>origin</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Ages below 0 will be returned as <code>NA</code> with a warning. Ages above 120 will only give a warning.
</p>
<p>This function vectorises over both <code>x</code> and <code>reference</code>, meaning that either can have a length of 1 while the other argument has a larger length.
</p>


<h3>Value</h3>

<p>An <a href="base.html#topic+integer">integer</a> (no decimals) if <code>exact = FALSE</code>, a <a href="base.html#topic+double">double</a> (with decimals) otherwise
</p>


<h3>See Also</h3>

<p>To split ages into groups, use the <code><a href="#topic+age_groups">age_groups()</a></code> function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 10 random pre-Y2K birth dates
df &lt;- data.frame(birth_date = as.Date("2000-01-01") - runif(10) * 25000)

# add ages
df$age &lt;- age(df$birth_date)

# add exact ages
df$age_exact &lt;- age(df$birth_date, exact = TRUE)

# add age at millenium switch
df$age_at_y2k &lt;- age(df$birth_date, "2000-01-01")

df
</code></pre>

<hr>
<h2 id='age_groups'>Split Ages into Age Groups</h2><span id='topic+age_groups'></span>

<h3>Description</h3>

<p>Split ages into age groups defined by the <code>split</code> argument. This allows for easier demographic (antimicrobial resistance) analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>age_groups(x, split_at = c(12, 25, 55, 75), na.rm = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="age_groups_+3A_x">x</code></td>
<td>
<p>age, e.g. calculated with <code><a href="#topic+age">age()</a></code></p>
</td></tr>
<tr><td><code id="age_groups_+3A_split_at">split_at</code></td>
<td>
<p>values to split <code>x</code> at - the default is age groups 0-11, 12-24, 25-54, 55-74 and 75+. See <em>Details</em>.</p>
</td></tr>
<tr><td><code id="age_groups_+3A_na.rm">na.rm</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether missing values should be removed</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To split ages, the input for the <code>split_at</code> argument can be:
</p>

<ul>
<li><p> A <a href="base.html#topic+numeric">numeric</a> vector. A value of e.g. <code>c(10, 20)</code> will split <code>x</code> on 0-9, 10-19 and 20+. A value of only <code>50</code> will split <code>x</code> on 0-49 and 50+.
The default is to split on young children (0-11), youth (12-24), young adults (25-54), middle-aged adults (55-74) and elderly (75+).
</p>
</li>
<li><p> A character:
</p>

<ul>
<li> <p><code>"children"</code> or <code>"kids"</code>, equivalent of: <code>c(0, 1, 2, 4, 6, 13, 18)</code>. This will split on 0, 1, 2-3, 4-5, 6-12, 13-17 and 18+.
</p>
</li>
<li> <p><code>"elderly"</code> or <code>"seniors"</code>, equivalent of: <code>c(65, 75, 85)</code>. This will split on 0-64, 65-74, 75-84, 85+.
</p>
</li>
<li> <p><code>"fives"</code>, equivalent of: <code>1:20 * 5</code>. This will split on 0-4, 5-9, ..., 95-99, 100+.
</p>
</li>
<li> <p><code>"tens"</code>, equivalent of: <code>1:10 * 10</code>. This will split on 0-9, 10-19, ..., 90-99, 100+.
</p>
</li></ul>

</li></ul>



<h3>Value</h3>

<p>Ordered <a href="base.html#topic+factor">factor</a>
</p>


<h3>See Also</h3>

<p>To determine ages, based on one or more reference dates, use the <code><a href="#topic+age">age()</a></code> function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ages &lt;- c(3, 8, 16, 54, 31, 76, 101, 43, 21)

# split into 0-49 and 50+
age_groups(ages, 50)

# split into 0-19, 20-49 and 50+
age_groups(ages, c(20, 50))

# split into groups of ten years
age_groups(ages, 1:10 * 10)
age_groups(ages, split_at = "tens")

# split into groups of five years
age_groups(ages, 1:20 * 5)
age_groups(ages, split_at = "fives")

# split specifically for children
age_groups(ages, c(1, 2, 4, 6, 13, 18))
age_groups(ages, "children")


# resistance of ciprofloxacin per age group
if (require("dplyr") &amp;&amp; require("ggplot2")) {
  example_isolates %&gt;%
    filter_first_isolate() %&gt;%
    filter(mo == as.mo("Escherichia coli")) %&gt;%
    group_by(age_group = age_groups(age)) %&gt;%
    select(age_group, CIP) %&gt;%
    ggplot_sir(
      x = "age_group",
      minimum = 0,
      x.title = "Age Group",
      title = "Ciprofloxacin resistance per age group"
    )
}

</code></pre>

<hr>
<h2 id='AMR'>The <code>AMR</code> Package</h2><span id='topic+AMR'></span><span id='topic+AMR-package'></span>

<h3>Description</h3>

<p>Welcome to the <code>AMR</code> package.
</p>
<p>The <code>AMR</code> package is a <a href="https://msberends.github.io/AMR/#copyright">free and open-source</a> R package with <a href="https://en.wikipedia.org/wiki/Dependency_hell">zero dependencies</a> to simplify the analysis and prediction of Antimicrobial Resistance (AMR) and to work with microbial and antimicrobial data and properties, by using evidence-based methods. <strong>Our aim is to provide a standard</strong> for clean and reproducible AMR data analysis, that can therefore empower epidemiological analyses to continuously enable surveillance and treatment evaluation in any setting. <a href="https://msberends.github.io/AMR/authors.html">Many different researchers</a> from around the globe are continually helping us to make this a successful and durable project!
</p>
<p>This work was published in the Journal of Statistical Software (Volume 104(3); <a href="https://doi.org/10.18637/jss.v104.i03">doi:10.18637/jss.v104.i03</a>) and formed the basis of two PhD theses (<a href="https://doi.org/10.33612/diss.177417131">doi:10.33612/diss.177417131</a> and <a href="https://doi.org/10.33612/diss.192486375">doi:10.33612/diss.192486375</a>).
</p>
<p>After installing this package, R knows <a href="https://msberends.github.io/AMR/reference/microorganisms.html"><strong>~52 000 microorganisms</strong></a> (updated December 2022) and all <a href="https://msberends.github.io/AMR/reference/antibiotics.html"><strong>~600 antibiotic, antimycotic and antiviral drugs</strong></a> by name and code (including ATC, EARS-Net, ASIARS-Net, PubChem, LOINC and SNOMED CT), and knows all about valid SIR and MIC values. The integral clinical breakpoint guidelines from CLSI and EUCAST are included, even with epidemiological cut-off (ECOFF) values. It supports and can read any data format, including WHONET data. This package works on Windows, macOS and Linux with all versions of R since R-3.0 (April 2013). <strong>It was designed to work in any setting, including those with very limited resources</strong>. It was created for both routine data analysis and academic research at the Faculty of Medical Sciences of the public <a href="https://www.rug.nl">University of Groningen</a>, in collaboration with non-profit organisations <a href="https://www.certe.nl">Certe Medical Diagnostics and Advice Foundation</a> and <a href="https://www.umcg.nl">University Medical Center Groningen</a>.
</p>
<p>The <code>AMR</code> package is available in English, Chinese, Czech, Danish, Dutch, Finnish, French, German, Greek, Italian, Japanese, Norwegian, Polish, Portuguese, Romanian, Russian, Spanish, Swedish, Turkish, and Ukrainian. Antimicrobial drug (group) names and colloquial microorganism names are provided in these languages.
</p>


<h3>Reference Data Publicly Available</h3>

<p>All data sets in this <code>AMR</code> package (about microorganisms, antibiotics, SIR interpretation, EUCAST rules, etc.) are publicly and freely available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. We also provide tab-separated plain text files that are machine-readable and suitable for input in any software program, such as laboratory information systems. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Matthijs S. Berends <a href="mailto:m.s.berends@umcg.nl">m.s.berends@umcg.nl</a> (<a href="https://orcid.org/0000-0001-7620-1800">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Christian F. Luz (<a href="https://orcid.org/0000-0001-5809-5995">ORCID</a>) [contributor]
</p>
</li>
<li><p> Dennis Souverein (<a href="https://orcid.org/0000-0003-0455-0336">ORCID</a>) [contributor]
</p>
</li>
<li><p> Erwin E. A. Hassing [contributor]
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Casper J. Albers (<a href="https://orcid.org/0000-0002-9213-6743">ORCID</a>) [thesis advisor]
</p>
</li>
<li><p> Peter Dutey-Magni (<a href="https://orcid.org/0000-0002-8942-9836">ORCID</a>) [contributor]
</p>
</li>
<li><p> Judith M. Fonville [contributor]
</p>
</li>
<li><p> Alex W. Friedrich (<a href="https://orcid.org/0000-0003-4881-038X">ORCID</a>) [thesis advisor]
</p>
</li>
<li><p> Corinna Glasner (<a href="https://orcid.org/0000-0003-1241-1328">ORCID</a>) [thesis advisor]
</p>
</li>
<li><p> Eric H. L. C. M. Hazenberg [contributor]
</p>
</li>
<li><p> Gwen Knight (<a href="https://orcid.org/0000-0002-7263-9896">ORCID</a>) [contributor]
</p>
</li>
<li><p> Annick Lenglet (<a href="https://orcid.org/0000-0003-2013-8405">ORCID</a>) [contributor]
</p>
</li>
<li><p> Bart C. Meijer [contributor]
</p>
</li>
<li><p> Dmytro Mykhailenko [contributor]
</p>
</li>
<li><p> Anton Mymrikov [contributor]
</p>
</li>
<li><p> Andrew P. Norgan (<a href="https://orcid.org/0000-0002-2955-2066">ORCID</a>) [contributor]
</p>
</li>
<li><p> Sofia Ny (<a href="https://orcid.org/0000-0002-2017-1363">ORCID</a>) [contributor]
</p>
</li>
<li><p> Jonas Salm [contributor]
</p>
</li>
<li><p> Rogier P. Schade [contributor]
</p>
</li>
<li><p> Bhanu N. M. Sinha (<a href="https://orcid.org/0000-0003-1634-0010">ORCID</a>) [thesis advisor]
</p>
</li>
<li><p> Anthony Underwood (<a href="https://orcid.org/0000-0002-8547-4277">ORCID</a>) [contributor]
</p>
</li>
<li><p> Anita Williams (<a href="https://orcid.org/0000-0002-5295-8451">ORCID</a>) [contributor]
</p>
</li></ul>



<h3>Source</h3>

<p>To cite AMR in publications use:
</p>
<p>Berends MS, Luz CF, Friedrich AW, Sinha BNM, Albers CJ, Glasner C (2022). &quot;AMR: An R Package for Working with Antimicrobial Resistance Data.&quot; <em>Journal of Statistical Software</em>, <em>104</em>(3), 1-31. <a href="https://doi.org/10.18637/jss.v104.i03">doi:10.18637/jss.v104.i03</a>
</p>
<p>A BibTeX entry for LaTeX users is:
</p>
<pre>
@Article{,
  title = {{AMR}: An {R} Package for Working with Antimicrobial Resistance Data},
  author = {Matthijs S. Berends and Christian F. Luz and Alexander W. Friedrich and Bhanu N. M. Sinha and Casper J. Albers and Corinna Glasner},
  journal = {Journal of Statistical Software},
  year = {2022},
  volume = {104},
  number = {3},
  pages = {1--31},
  doi = {10.18637/jss.v104.i03},
}
</pre>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://msberends.github.io/AMR/">https://msberends.github.io/AMR/</a>
</p>
</li>
<li> <p><a href="https://github.com/msberends/AMR">https://github.com/msberends/AMR</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/msberends/AMR/issues">https://github.com/msberends/AMR/issues</a>
</p>
</li></ul>


<hr>
<h2 id='AMR-deprecated'>Deprecated Functions</h2><span id='topic+AMR-deprecated'></span><span id='topic+NA_rsi_'></span><span id='topic+as.rsi'></span><span id='topic+facet_rsi'></span><span id='topic+geom_rsi'></span><span id='topic+ggplot_rsi'></span><span id='topic+ggplot_rsi_predict'></span><span id='topic+is.rsi'></span><span id='topic+is.rsi.eligible'></span><span id='topic+labels_rsi_count'></span><span id='topic+n_rsi'></span><span id='topic+random_rsi'></span><span id='topic+rsi_df'></span><span id='topic+rsi_predict'></span><span id='topic+scale_rsi_colours'></span><span id='topic+theme_rsi'></span>

<h3>Description</h3>

<p>These functions are so-called '<a href="base.html#topic+Deprecated">Deprecated</a>'. <strong>They will be removed in a future release.</strong> Using the functions will give a warning with the name of the function it has been replaced by (if there is one).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NA_rsi_

as.rsi(x, ...)

facet_rsi(...)

geom_rsi(...)

ggplot_rsi(...)

ggplot_rsi_predict(...)

is.rsi(...)

is.rsi.eligible(...)

labels_rsi_count(...)

n_rsi(...)

random_rsi(...)

rsi_df(...)

rsi_predict(...)

scale_rsi_colours(...)

theme_rsi(...)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>rsi</code> (inherits from <code>ordered</code>, <code>factor</code>) of length 1.
</p>

<hr>
<h2 id='AMR-options'>Options for the AMR package</h2><span id='topic+AMR-options'></span>

<h3>Description</h3>

<p>This is an overview of all the package-specific <code><a href="base.html#topic+options">options()</a></code> you can set in the <code>AMR</code> package.
</p>


<h3>Options</h3>


<ul>
<li> <p><code>AMR_custom_ab</code> <br /> Allows to use custom antimicrobial drugs with this package. This is explained in <code><a href="#topic+add_custom_antimicrobials">add_custom_antimicrobials()</a></code>.
</p>
</li>
<li> <p><code>AMR_custom_mo</code> <br /> Allows to use custom microorganisms with this package. This is explained in <code><a href="#topic+add_custom_microorganisms">add_custom_microorganisms()</a></code>.
</p>
</li>
<li> <p><code>AMR_eucastrules</code> <br /> Used for setting the default types of rules for <code><a href="#topic+eucast_rules">eucast_rules()</a></code> function, must be one or more of: <code>"breakpoints"</code>, <code>"expert"</code>, <code>"other"</code>, <code>"custom"</code>, <code>"all"</code>, and defaults to <code>c("breakpoints", "expert")</code>.
</p>
</li>
<li> <p><code>AMR_guideline</code> <br /> Used for setting the default guideline for interpreting MIC values and disk diffusion diameters with <code><a href="#topic+as.sir">as.sir()</a></code>. Can be only the guideline name (e.g., <code>"CLSI"</code>) or the name with a year (e.g. <code>"CLSI 2019"</code>). The default to the latest implemented EUCAST guideline, currently <code>"EUCAST 2023"</code>. Supported guideline are currently EUCAST (2011-2023) and CLSI (2011-2023).
</p>
</li>
<li> <p><code>AMR_ignore_pattern</code> <br /> A <a href="base.html#topic+regex">regular expression</a> to ignore (i.e., make <code>NA</code>) any match given in <code><a href="#topic+as.mo">as.mo()</a></code> and all <code><a href="#topic+mo_property">mo_*</a></code> functions.
</p>
</li>
<li> <p><code>AMR_include_PKPD</code> <br /> A <a href="base.html#topic+logical">logical</a> to use in <code><a href="#topic+as.sir">as.sir()</a></code>, to indicate that PK/PD clinical breakpoints must be applied as a last resort - the default is <code>TRUE</code>.
</p>
</li>
<li> <p><code>AMR_ecoff</code> <br /> A <a href="base.html#topic+logical">logical</a> use in <code><a href="#topic+as.sir">as.sir()</a></code>, to indicate that ECOFF (Epidemiological Cut-Off) values must be used - the default is <code>FALSE</code>.
</p>
</li>
<li> <p><code>AMR_include_screening</code> <br /> A <a href="base.html#topic+logical">logical</a> to use in <code><a href="#topic+as.sir">as.sir()</a></code>, to indicate that clinical breakpoints for screening are allowed - the default is <code>FALSE</code>.
</p>
</li>
<li> <p><code>AMR_keep_synonyms</code> <br /> A <a href="base.html#topic+logical">logical</a> to use in <code><a href="#topic+as.mo">as.mo()</a></code> and all <code><a href="#topic+mo_property">mo_*</a></code> functions, to indicate if old, previously valid taxonomic names must be preserved and not be corrected to currently accepted names. The default is <code>FALSE</code>.
</p>
</li>
<li> <p><code>AMR_cleaning_regex</code> <br /> A <a href="base.html#topic+regex">regular expression</a> (case-insensitive) to use in <code><a href="#topic+as.mo">as.mo()</a></code> and all <code><a href="#topic+mo_property">mo_*</a></code> functions, to clean the user input. The default is the outcome of <code><a href="#topic+mo_cleaning_regex">mo_cleaning_regex()</a></code>, which removes texts between brackets and texts such as &quot;species&quot; and &quot;serovar&quot;.
</p>
</li>
<li> <p><code>AMR_locale</code> <br /> A language to use for the <code>AMR</code> package, can be one of these supported language names or ISO-639-1 codes: English (en), Chinese (zh), Czech (cs), Danish (da), Dutch (nl), Finnish (fi), French (fr), German (de), Greek (el), Italian (it), Japanese (ja), Norwegian (no), Polish (pl), Portuguese (pt), Romanian (ro), Russian (ru), Spanish (es), Swedish (sv), Turkish (tr), or Ukrainian (uk). The default is the current system language (if supported).
</p>
</li>
<li> <p><code>AMR_mo_source</code> <br /> A file location for a manual code list to be used in <code><a href="#topic+as.mo">as.mo()</a></code> and all <code><a href="#topic+mo_property">mo_*</a></code> functions. This is explained in <code><a href="#topic+set_mo_source">set_mo_source()</a></code>.
</p>
</li></ul>



<h3>Saving Settings Between Sessions</h3>

<p>Settings in <span class="rlang"><b>R</b></span> are not saved globally and are thus lost when <span class="rlang"><b>R</b></span> is exited. You can save your options to your own <code>.Rprofile</code> file, which is a user-specific file. You can edit it using:
</p>
<div class="sourceCode r"><pre>  utils::file.edit("~/.Rprofile")
</pre></div>
<p>In this file, you can set options such as:
</p>
<div class="sourceCode r"><pre> options(AMR_locale = "pt")
 options(AMR_include_PKPD = TRUE)
</pre></div>
<p>to add Portuguese language support of antibiotics, and allow PK/PD rules when interpreting MIC values with <code><a href="#topic+as.sir">as.sir()</a></code>.
</p>


<h4>Share Options Within Team</h4>

<p>For a more global approach, e.g. within a data team, save an options file to a remote file location, such as a shared network drive. This would work in this way:
</p>

<ol>
<li><p> Save a plain text file to e.g. &quot;X:/team_folder/R_options.R&quot; and fill it with preferred settings.
</p>
</li>
<li><p> For each user, open the <code>.Rprofile</code> file using <code>utils::file.edit("~/.Rprofile")</code> and put in there:
</p>
<div class="sourceCode r"><pre>  source("X:/team_folder/R_options.R")
</pre></div>
</li>
<li><p> Reload R/RStudio and check the settings with <code><a href="base.html#topic+getOption">getOption()</a></code>, e.g. <code>getOption("AMR_locale")</code> if you have set that value.
</p>
</li></ol>

<p>Now the team settings are configured in only one place, and can be maintained there.
</p>


<hr>
<h2 id='antibiogram'>Generate Antibiogram: Traditional, Combined, Syndromic, or Weighted-Incidence Syndromic Combination (WISCA)</h2><span id='topic+antibiogram'></span><span id='topic+plot.antibiogram'></span><span id='topic+autoplot.antibiogram'></span><span id='topic+knit_print.antibiogram'></span>

<h3>Description</h3>

<p>Generate an antibiogram, and communicate the results in plots or tables. These functions follow the logic of Klinker <em>et al.</em> and Barbieri <em>et al.</em> (see <em>Source</em>), and allow reporting in e.g. R Markdown and Quarto as well.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>antibiogram(
  x,
  antibiotics = where(is.sir),
  mo_transform = "shortname",
  ab_transform = NULL,
  syndromic_group = NULL,
  add_total_n = TRUE,
  only_all_tested = FALSE,
  digits = 0,
  col_mo = NULL,
  language = get_AMR_locale(),
  minimum = 30,
  combine_SI = TRUE,
  sep = " + ",
  info = interactive()
)

## S3 method for class 'antibiogram'
plot(x, ...)

## S3 method for class 'antibiogram'
autoplot(object, ...)

## S3 method for class 'antibiogram'
knit_print(
  x,
  italicise = TRUE,
  na = getOption("knitr.kable.NA", default = ""),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="antibiogram_+3A_x">x</code></td>
<td>
<p>a <a href="base.html#topic+data.frame">data.frame</a> containing at least a column with microorganisms and columns with antibiotic results (class 'sir', see <code><a href="#topic+as.sir">as.sir()</a></code>)</p>
</td></tr>
<tr><td><code id="antibiogram_+3A_antibiotics">antibiotics</code></td>
<td>
<p>vector of any antibiotic name or code (will be evaluated with <code><a href="#topic+as.ab">as.ab()</a></code>, column name of <code>x</code>, or (any combinations of) <a href="#topic+antibiotic_class_selectors">antibiotic selectors</a> such as <code><a href="#topic+aminoglycosides">aminoglycosides()</a></code> or <code><a href="#topic+carbapenems">carbapenems()</a></code>. For combination antibiograms, this can also be set to values separated with <code>"+"</code>, such as &quot;TZP+TOB&quot; or &quot;cipro + genta&quot;, given that columns resembling such antibiotics exist in <code>x</code>. See <em>Examples</em>.</p>
</td></tr>
<tr><td><code id="antibiogram_+3A_mo_transform">mo_transform</code></td>
<td>
<p>a character to transform microorganism input - must be &quot;name&quot;, &quot;shortname&quot;, &quot;gramstain&quot;, or one of the column names of the <a href="#topic+microorganisms">microorganisms</a> data set: &quot;mo&quot;, &quot;fullname&quot;, &quot;status&quot;, &quot;kingdom&quot;, &quot;phylum&quot;, &quot;class&quot;, &quot;order&quot;, &quot;family&quot;, &quot;genus&quot;, &quot;species&quot;, &quot;subspecies&quot;, &quot;rank&quot;, &quot;ref&quot;, &quot;oxygen_tolerance&quot;, &quot;source&quot;, &quot;lpsn&quot;, &quot;lpsn_parent&quot;, &quot;lpsn_renamed_to&quot;, &quot;gbif&quot;, &quot;gbif_parent&quot;, &quot;gbif_renamed_to&quot;, &quot;prevalence&quot;, or &quot;snomed&quot;. Can also be <code>NULL</code> to not transform the input.</p>
</td></tr>
<tr><td><code id="antibiogram_+3A_ab_transform">ab_transform</code></td>
<td>
<p>a character to transform antibiotic input - must be one of the column names of the <a href="#topic+antibiotics">antibiotics</a> data set: &quot;ab&quot;, &quot;cid&quot;, &quot;name&quot;, &quot;group&quot;, &quot;atc&quot;, &quot;atc_group1&quot;, &quot;atc_group2&quot;, &quot;abbreviations&quot;, &quot;synonyms&quot;, &quot;oral_ddd&quot;, &quot;oral_units&quot;, &quot;iv_ddd&quot;, &quot;iv_units&quot;, or &quot;loinc&quot;. Can also be <code>NULL</code> to not transform the input.</p>
</td></tr>
<tr><td><code id="antibiogram_+3A_syndromic_group">syndromic_group</code></td>
<td>
<p>a column name of <code>x</code>, or values calculated to split rows of <code>x</code>, e.g. by using <code><a href="base.html#topic+ifelse">ifelse()</a></code> or <code><a href="dplyr.html#topic+case_when">case_when()</a></code>. See <em>Examples</em>.</p>
</td></tr>
<tr><td><code id="antibiogram_+3A_add_total_n">add_total_n</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether total available numbers per pathogen should be added to the table (default is <code>TRUE</code>). This will add the lowest and highest number of available isolate per antibiotic (e.g, if for <em>E. coli</em> 200 isolates are available for ciprofloxacin and 150 for amoxicillin, the returned number will be &quot;150-200&quot;).</p>
</td></tr>
<tr><td><code id="antibiogram_+3A_only_all_tested">only_all_tested</code></td>
<td>
<p>(for combination antibiograms): a <a href="base.html#topic+logical">logical</a> to indicate that isolates must be tested for all antibiotics, see <em>Details</em></p>
</td></tr>
<tr><td><code id="antibiogram_+3A_digits">digits</code></td>
<td>
<p>number of digits to use for rounding</p>
</td></tr>
<tr><td><code id="antibiogram_+3A_col_mo">col_mo</code></td>
<td>
<p>column name of the names or codes of the microorganisms (see <code><a href="#topic+as.mo">as.mo()</a></code>) - the default is the first column of class <code><a href="#topic+mo">mo</a></code>. Values will be coerced using <code><a href="#topic+as.mo">as.mo()</a></code>.</p>
</td></tr>
<tr><td><code id="antibiogram_+3A_language">language</code></td>
<td>
<p>language to translate text, which defaults to the system language (see <code><a href="#topic+get_AMR_locale">get_AMR_locale()</a></code>)</p>
</td></tr>
<tr><td><code id="antibiogram_+3A_minimum">minimum</code></td>
<td>
<p>the minimum allowed number of available (tested) isolates. Any isolate count lower than <code>minimum</code> will return <code>NA</code> with a warning. The default number of <code>30</code> isolates is advised by the Clinical and Laboratory Standards Institute (CLSI) as best practice, see <em>Source</em>.</p>
</td></tr>
<tr><td><code id="antibiogram_+3A_combine_si">combine_SI</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether all susceptibility should be determined by results of either S or I, instead of only S (default is <code>TRUE</code>)</p>
</td></tr>
<tr><td><code id="antibiogram_+3A_sep">sep</code></td>
<td>
<p>a separating character for antibiotic columns in combination antibiograms</p>
</td></tr>
<tr><td><code id="antibiogram_+3A_info">info</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate info should be printed - the default is <code>TRUE</code> only in interactive mode</p>
</td></tr>
<tr><td><code id="antibiogram_+3A_...">...</code></td>
<td>
<p>when used in <a href="knitr.html#topic+kable">R Markdown or Quarto</a>: arguments passed on to <code><a href="knitr.html#topic+kable">knitr::kable()</a></code> (otherwise, has no use)</p>
</td></tr>
<tr><td><code id="antibiogram_+3A_object">object</code></td>
<td>
<p>an <code><a href="#topic+antibiogram">antibiogram()</a></code> object</p>
</td></tr>
<tr><td><code id="antibiogram_+3A_italicise">italicise</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether the microorganism names in the <a href="knitr.html#topic+kable">knitr</a> table should be made italic, using <code><a href="#topic+italicise_taxonomy">italicise_taxonomy()</a></code>.</p>
</td></tr>
<tr><td><code id="antibiogram_+3A_na">na</code></td>
<td>
<p>character to use for showing <code>NA</code> values</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function returns a table with values between 0 and 100 for <em>susceptibility</em>, not resistance.
</p>
<p><strong>Remember that you should filter your data to let it contain only first isolates!</strong> This is needed to exclude duplicates and to reduce selection bias. Use <code><a href="#topic+first_isolate">first_isolate()</a></code> to determine them in your data set with one of the four available algorithms.
</p>
<p>All types of antibiograms as listed below can be plotted (using <code><a href="ggplot2.html#topic+autoplot">ggplot2::autoplot()</a></code> or base <span class="rlang"><b>R</b></span> <code><a href="#topic+plot">plot()</a></code>/<code><a href="graphics.html#topic+barplot">barplot()</a></code>). The <code>antibiogram</code> object can also be used directly in R Markdown / Quarto (i.e., <code>knitr</code>) for reports. In this case, <code><a href="knitr.html#topic+kable">knitr::kable()</a></code> will be applied automatically and microorganism names will even be printed in italics at default (see argument <code>italicise</code>). You can also use functions from specific 'table reporting' packages to transform the output of <code><a href="#topic+antibiogram">antibiogram()</a></code> to your needs, e.g. with <code>flextable::as_flextable()</code> or <code>gt::gt()</code>.
</p>


<h4>Antibiogram Types</h4>

<p>There are four antibiogram types, as proposed by Klinker <em>et al.</em> (2021, <a href="https://doi.org/10.1177/20499361211011373">doi:10.1177/20499361211011373</a>), and they are all supported by <code><a href="#topic+antibiogram">antibiogram()</a></code>:
</p>

<ol>
<li> <p><strong>Traditional Antibiogram</strong>
</p>
<p>Case example: Susceptibility of <em>Pseudomonas aeruginosa</em> to piperacillin/tazobactam (TZP)
</p>
<p>Code example:
</p>
<div class="sourceCode r"><pre>antibiogram(your_data,
            antibiotics = "TZP")
</pre></div>
</li>
<li> <p><strong>Combination Antibiogram</strong>
</p>
<p>Case example: Additional susceptibility of <em>Pseudomonas aeruginosa</em> to TZP + tobramycin versus TZP alone
</p>
<p>Code example:
</p>
<div class="sourceCode r"><pre>antibiogram(your_data,
            antibiotics = c("TZP", "TZP+TOB", "TZP+GEN"))
</pre></div>
</li>
<li> <p><strong>Syndromic Antibiogram</strong>
</p>
<p>Case example: Susceptibility of <em>Pseudomonas aeruginosa</em> to TZP among respiratory specimens (obtained among ICU patients only)
</p>
<p>Code example:
</p>
<div class="sourceCode r"><pre>antibiogram(your_data,
            antibiotics = penicillins(),
            syndromic_group = "ward")
</pre></div>
</li>
<li> <p><strong>Weighted-Incidence Syndromic Combination Antibiogram (WISCA)</strong>
</p>
<p>Case example: Susceptibility of <em>Pseudomonas aeruginosa</em> to TZP among respiratory specimens (obtained among ICU patients only) for male patients age &gt;=65 years with heart failure
</p>
<p>Code example:
</p>
<div class="sourceCode r"><pre>library(dplyr)
your_data %&gt;%
  filter(ward == "ICU" &amp; specimen_type == "Respiratory") %&gt;%
  antibiogram(antibiotics = c("TZP", "TZP+TOB", "TZP+GEN"),
              syndromic_group = ifelse(.$age &gt;= 65 &amp;
                                         .$gender == "Male" &amp;
                                         .$condition == "Heart Disease",
                                       "Study Group", "Control Group"))
</pre></div>
</li></ol>

<p>Note that for combination antibiograms, it is important to realise that susceptibility can be calculated in two ways, which can be set with the <code>only_all_tested</code> argument (default is <code>FALSE</code>). See this example for two antibiotics, Drug A and Drug B, about how <code><a href="#topic+antibiogram">antibiogram()</a></code> works to calculate the %SI:
</p>
<div class="sourceCode"><pre>--------------------------------------------------------------------
                    only_all_tested = FALSE  only_all_tested = TRUE
                    -----------------------  -----------------------
 Drug A    Drug B   include as  include as   include as  include as
                    numerator   denominator  numerator   denominator
--------  --------  ----------  -----------  ----------  -----------
 S or I    S or I       X            X            X            X
   R       S or I       X            X            X            X
  &lt;NA&gt;     S or I       X            X            -            -
 S or I      R          X            X            X            X
   R         R          -            X            -            X
  &lt;NA&gt;       R          -            -            -            -
 S or I     &lt;NA&gt;        X            X            -            -
   R        &lt;NA&gt;        -            -            -            -
  &lt;NA&gt;      &lt;NA&gt;        -            -            -            -
--------------------------------------------------------------------
</pre></div>



<h3>Source</h3>


<ul>
<li><p> Klinker KP <em>et al.</em> (2021). <strong>Antimicrobial stewardship and antibiograms: importance of moving beyond traditional antibiograms</strong>. <em>Therapeutic Advances in Infectious Disease</em>, May 5;8:20499361211011373; <a href="https://doi.org/10.1177/20499361211011373">doi:10.1177/20499361211011373</a>
</p>
</li>
<li><p> Barbieri E <em>et al.</em> (2021). <strong>Development of a Weighted-Incidence Syndromic Combination Antibiogram (WISCA) to guide the choice of the empiric antibiotic treatment for urinary tract infection in paediatric patients: a Bayesian approach</strong> <em>Antimicrobial Resistance &amp; Infection Control</em> May 1;10(1):74; <a href="https://doi.org/10.1186/s13756-021-00939-2">doi:10.1186/s13756-021-00939-2</a>
</p>
</li>
<li> <p><strong>M39 Analysis and Presentation of Cumulative Antimicrobial Susceptibility Test Data, 5th Edition</strong>, 2022, <em>Clinical and Laboratory Standards Institute (CLSI)</em>. <a href="https://clsi.org/standards/products/microbiology/documents/m39/">https://clsi.org/standards/products/microbiology/documents/m39/</a>.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># example_isolates is a data set available in the AMR package.
# run ?example_isolates for more info.
example_isolates


# Traditional antibiogram ----------------------------------------------

antibiogram(example_isolates,
  antibiotics = c(aminoglycosides(), carbapenems())
)

antibiogram(example_isolates,
  antibiotics = aminoglycosides(),
  ab_transform = "atc",
  mo_transform = "gramstain"
)

antibiogram(example_isolates,
  antibiotics = carbapenems(),
  ab_transform = "name",
  mo_transform = "name"
)


# Combined antibiogram -------------------------------------------------

# combined antibiotics yield higher empiric coverage
antibiogram(example_isolates,
  antibiotics = c("TZP", "TZP+TOB", "TZP+GEN"),
  mo_transform = "gramstain"
)

# names of antibiotics do not need to resemble columns exactly:
antibiogram(example_isolates,
  antibiotics = c("Cipro", "cipro + genta"),
  mo_transform = "gramstain",
  ab_transform = "name",
  sep = " &amp; "
)


# Syndromic antibiogram ------------------------------------------------

# the data set could contain a filter for e.g. respiratory specimens
antibiogram(example_isolates,
  antibiotics = c(aminoglycosides(), carbapenems()),
  syndromic_group = "ward"
)

# now define a data set with only E. coli
ex1 &lt;- example_isolates[which(mo_genus() == "Escherichia"), ]

# with a custom language, though this will be determined automatically
# (i.e., this table will be in Spanish on Spanish systems)
antibiogram(ex1,
  antibiotics = aminoglycosides(),
  ab_transform = "name",
  syndromic_group = ifelse(ex1$ward == "ICU",
    "UCI", "No UCI"
  ),
  language = "es"
)


# Weighted-incidence syndromic combination antibiogram (WISCA) ---------

# the data set could contain a filter for e.g. respiratory specimens/ICU
antibiogram(example_isolates,
  antibiotics = c("AMC", "AMC+CIP", "TZP", "TZP+TOB"),
  mo_transform = "gramstain",
  minimum = 10, # this should be &gt;=30, but now just as example
  syndromic_group = ifelse(example_isolates$age &gt;= 65 &amp;
    example_isolates$gender == "M",
  "WISCA Group 1", "WISCA Group 2"
  )
)


# Print the output for R Markdown / Quarto -----------------------------

ureido &lt;- antibiogram(example_isolates,
  antibiotics = ureidopenicillins(),
  ab_transform = "name"
)

# in an Rmd file, you would just need to return `ureido` in a chunk,
# but to be explicit here:
if (requireNamespace("knitr")) {
  cat(knitr::knit_print(ureido))
}


# Generate plots with ggplot2 or base R --------------------------------

ab1 &lt;- antibiogram(example_isolates,
  antibiotics = c("AMC", "CIP", "TZP", "TZP+TOB"),
  mo_transform = "gramstain"
)
ab2 &lt;- antibiogram(example_isolates,
  antibiotics = c("AMC", "CIP", "TZP", "TZP+TOB"),
  mo_transform = "gramstain",
  syndromic_group = "ward"
)

if (requireNamespace("ggplot2")) {
  ggplot2::autoplot(ab1)
}
if (requireNamespace("ggplot2")) {
  ggplot2::autoplot(ab2)
}

plot(ab1)
plot(ab2)

</code></pre>

<hr>
<h2 id='antibiotic_class_selectors'>Antibiotic Selectors</h2><span id='topic+antibiotic_class_selectors'></span><span id='topic+ab_class'></span><span id='topic+ab_selector'></span><span id='topic+aminoglycosides'></span><span id='topic+aminopenicillins'></span><span id='topic+antifungals'></span><span id='topic+antimycobacterials'></span><span id='topic+betalactams'></span><span id='topic+carbapenems'></span><span id='topic+cephalosporins'></span><span id='topic+cephalosporins_1st'></span><span id='topic+cephalosporins_2nd'></span><span id='topic+cephalosporins_3rd'></span><span id='topic+cephalosporins_4th'></span><span id='topic+cephalosporins_5th'></span><span id='topic+fluoroquinolones'></span><span id='topic+glycopeptides'></span><span id='topic+lincosamides'></span><span id='topic+lipoglycopeptides'></span><span id='topic+macrolides'></span><span id='topic+oxazolidinones'></span><span id='topic+penicillins'></span><span id='topic+polymyxins'></span><span id='topic+streptogramins'></span><span id='topic+quinolones'></span><span id='topic+tetracyclines'></span><span id='topic+trimethoprims'></span><span id='topic+ureidopenicillins'></span><span id='topic+administrable_per_os'></span><span id='topic+administrable_iv'></span><span id='topic+not_intrinsic_resistant'></span>

<h3>Description</h3>

<p>These functions allow for filtering rows and selecting columns based on antibiotic test results that are of a specific antibiotic class or group (according to the <a href="#topic+antibiotics">antibiotics</a> data set), without the need to define the columns or antibiotic abbreviations.
</p>
<p>In short, if you have a column name that resembles an antimicrobial drug, it will be picked up by any of these functions that matches its pharmaceutical class: &quot;cefazolin&quot;, &quot;kefzol&quot;, &quot;CZO&quot; and &quot;J01DB04&quot; will all be picked up by <code><a href="#topic+cephalosporins">cephalosporins()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ab_class(ab_class, only_sir_columns = FALSE, only_treatable = TRUE, ...)

ab_selector(filter, only_sir_columns = FALSE, only_treatable = TRUE, ...)

aminoglycosides(only_sir_columns = FALSE, only_treatable = TRUE, ...)

aminopenicillins(only_sir_columns = FALSE, ...)

antifungals(only_sir_columns = FALSE, ...)

antimycobacterials(only_sir_columns = FALSE, ...)

betalactams(only_sir_columns = FALSE, only_treatable = TRUE, ...)

carbapenems(only_sir_columns = FALSE, only_treatable = TRUE, ...)

cephalosporins(only_sir_columns = FALSE, ...)

cephalosporins_1st(only_sir_columns = FALSE, ...)

cephalosporins_2nd(only_sir_columns = FALSE, ...)

cephalosporins_3rd(only_sir_columns = FALSE, ...)

cephalosporins_4th(only_sir_columns = FALSE, ...)

cephalosporins_5th(only_sir_columns = FALSE, ...)

fluoroquinolones(only_sir_columns = FALSE, ...)

glycopeptides(only_sir_columns = FALSE, ...)

lincosamides(only_sir_columns = FALSE, ...)

lipoglycopeptides(only_sir_columns = FALSE, ...)

macrolides(only_sir_columns = FALSE, ...)

oxazolidinones(only_sir_columns = FALSE, ...)

penicillins(only_sir_columns = FALSE, ...)

polymyxins(only_sir_columns = FALSE, only_treatable = TRUE, ...)

streptogramins(only_sir_columns = FALSE, ...)

quinolones(only_sir_columns = FALSE, ...)

tetracyclines(only_sir_columns = FALSE, ...)

trimethoprims(only_sir_columns = FALSE, ...)

ureidopenicillins(only_sir_columns = FALSE, ...)

administrable_per_os(only_sir_columns = FALSE, ...)

administrable_iv(only_sir_columns = FALSE, ...)

not_intrinsic_resistant(
  only_sir_columns = FALSE,
  col_mo = NULL,
  version_expertrules = 3.3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="antibiotic_class_selectors_+3A_ab_class">ab_class</code></td>
<td>
<p>an antimicrobial class or a part of it, such as <code>"carba"</code> and <code>"carbapenems"</code>. The columns <code>group</code>, <code>atc_group1</code> and <code>atc_group2</code> of the <a href="#topic+antibiotics">antibiotics</a> data set will be searched (case-insensitive) for this value.</p>
</td></tr>
<tr><td><code id="antibiotic_class_selectors_+3A_only_sir_columns">only_sir_columns</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether only columns of class <code>sir</code> must be selected (default is <code>FALSE</code>), see <code><a href="#topic+as.sir">as.sir()</a></code></p>
</td></tr>
<tr><td><code id="antibiotic_class_selectors_+3A_only_treatable">only_treatable</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether antimicrobial drugs should be excluded that are only for laboratory tests (default is <code>TRUE</code>), such as gentamicin-high (<code>GEH</code>) and imipenem/EDTA (<code>IPE</code>)</p>
</td></tr>
<tr><td><code id="antibiotic_class_selectors_+3A_...">...</code></td>
<td>
<p>ignored, only in place to allow future extensions</p>
</td></tr>
<tr><td><code id="antibiotic_class_selectors_+3A_filter">filter</code></td>
<td>
<p>an <a href="base.html#topic+expression">expression</a> to be evaluated in the <a href="#topic+antibiotics">antibiotics</a> data set, such as <code>name %like% "trim"</code></p>
</td></tr>
<tr><td><code id="antibiotic_class_selectors_+3A_col_mo">col_mo</code></td>
<td>
<p>column name of the names or codes of the microorganisms (see <code><a href="#topic+as.mo">as.mo()</a></code>) - the default is the first column of class <code><a href="#topic+mo">mo</a></code>. Values will be coerced using <code><a href="#topic+as.mo">as.mo()</a></code>.</p>
</td></tr>
<tr><td><code id="antibiotic_class_selectors_+3A_version_expertrules">version_expertrules</code></td>
<td>
<p>the version number to use for the EUCAST Expert Rules and Intrinsic Resistance guideline. Can be &quot;3.3&quot;, &quot;3.2&quot;, or &quot;3.1&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions can be used in data set calls for selecting columns and filtering rows. They work with base <span class="rlang"><b>R</b></span>, the Tidyverse, and <code>data.table</code>. They are heavily inspired by the <a href="tidyselect.html#topic+language">Tidyverse selection helpers</a> such as <code><a href="tidyselect.html#topic+everything">everything()</a></code>, but are not limited to <code>dplyr</code> verbs. Nonetheless, they are very convenient to use with <code>dplyr</code> functions such as <code><a href="dplyr.html#topic+select">select()</a></code>, <code><a href="dplyr.html#topic+filter">filter()</a></code> and <code><a href="dplyr.html#topic+summarise">summarise()</a></code>, see <em>Examples</em>.
</p>
<p>All columns in the data in which these functions are called will be searched for known antibiotic names, abbreviations, brand names, and codes (ATC, EARS-Net, WHO, etc.) according to the <a href="#topic+antibiotics">antibiotics</a> data set. This means that a selector such as <code><a href="#topic+aminoglycosides">aminoglycosides()</a></code> will pick up column names like 'gen', 'genta', 'J01GB03', 'tobra', 'Tobracin', etc.
</p>
<p>The <code><a href="#topic+ab_class">ab_class()</a></code> function can be used to filter/select on a manually defined antibiotic class. It searches for results in the <a href="#topic+antibiotics">antibiotics</a> data set within the columns <code>group</code>, <code>atc_group1</code> and <code>atc_group2</code>.
</p>
<p>The <code><a href="#topic+ab_selector">ab_selector()</a></code> function can be used to internally filter the <a href="#topic+antibiotics">antibiotics</a> data set on any results, see <em>Examples</em>. It allows for filtering on a (part of) a certain name, and/or a group name or even a minimum of DDDs for oral treatment. This function yields the highest flexibility, but is also the least user-friendly, since it requires a hard-coded filter to set.
</p>
<p>The <code><a href="#topic+administrable_per_os">administrable_per_os()</a></code> and <code><a href="#topic+administrable_iv">administrable_iv()</a></code> functions also rely on the <a href="#topic+antibiotics">antibiotics</a> data set - antibiotic columns will be matched where a DDD (defined daily dose) for resp. oral and IV treatment is available in the <a href="#topic+antibiotics">antibiotics</a> data set.
</p>
<p>The <code><a href="#topic+not_intrinsic_resistant">not_intrinsic_resistant()</a></code> function can be used to only select antibiotic columns that pose no intrinsic resistance for the microorganisms in the data set. For example, if a data set contains only microorganism codes or names of <em>E. coli</em> and <em>K. pneumoniae</em> and contains a column &quot;vancomycin&quot;, this column will be removed (or rather, unselected) using this function. It currently applies <a href="https://www.eucast.org/expert_rules_and_expected_phenotypes">'EUCAST Expert Rules' and 'EUCAST Intrinsic Resistance and Unusual Phenotypes' v3.3</a> (2021) to determine intrinsic resistance, using the <code><a href="#topic+eucast_rules">eucast_rules()</a></code> function internally. Because of this determination, this function is quite slow in terms of performance.
</p>


<h3>Value</h3>

<p>(internally) a <a href="base.html#topic+character">character</a> vector of column names, with additional class <code>"ab_selector"</code>
</p>


<h3>Full list of supported (antibiotic) classes</h3>


<ul>
<li> <p><code><a href="#topic+aminoglycosides">aminoglycosides()</a></code> can select: <br /> amikacin (AMK), amikacin/fosfomycin (AKF), amphotericin B-high (AMH), apramycin (APR), arbekacin (ARB), astromicin (AST), bekanamycin (BEK), dibekacin (DKB), framycetin (FRM), gentamicin (GEN), gentamicin-high (GEH), habekacin (HAB), hygromycin (HYG), isepamicin (ISE), kanamycin (KAN), kanamycin-high (KAH), kanamycin/cephalexin (KAC), micronomicin (MCR), neomycin (NEO), netilmicin (NET), pentisomicin (PIM), plazomicin (PLZ), propikacin (PKA), ribostamycin (RST), sisomicin (SIS), streptoduocin (STR), streptomycin (STR1), streptomycin-high (STH), tobramycin (TOB), and tobramycin-high (TOH)
</p>
</li>
<li> <p><code><a href="#topic+aminopenicillins">aminopenicillins()</a></code> can select: <br /> amoxicillin (AMX) and ampicillin (AMP)
</p>
</li>
<li> <p><code><a href="#topic+antifungals">antifungals()</a></code> can select: <br /> amphotericin B (AMB), anidulafungin (ANI), butoconazole (BUT), caspofungin (CAS), ciclopirox (CIX), clotrimazole (CTR), econazole (ECO), fluconazole (FLU), flucytosine (FCT), fosfluconazole (FFL), griseofulvin (GRI), hachimycin (HCH), ibrexafungerp (IBX), isavuconazole (ISV), isoconazole (ISO), itraconazole (ITR), ketoconazole (KET), manogepix (MGX), micafungin (MIF), miconazole (MCZ), nystatin (NYS), oteseconazole (OTE), pimaricin (PMR), posaconazole (POS), rezafungin (RZF), ribociclib (RBC), sulconazole (SUC), terbinafine (TRB), terconazole (TRC), and voriconazole (VOR)
</p>
</li>
<li> <p><code><a href="#topic+antimycobacterials">antimycobacterials()</a></code> can select: <br /> 4-aminosalicylic acid (AMA), calcium aminosalicylate (CLA), capreomycin (CAP), clofazimine (CLF), delamanid (DLM), enviomycin (ENV), ethambutol (ETH), ethambutol/isoniazid (ETI), ethionamide (ETI1), isoniazid (INH), isoniazid/sulfamethoxazole/trimethoprim/pyridoxine (IST), morinamide (MRN), p-aminosalicylic acid (PAS), pretomanid (PMD), protionamide (PTH), pyrazinamide (PZA), rifabutin (RIB), rifampicin (RIF), rifampicin/ethambutol/isoniazid (REI), rifampicin/isoniazid (RFI), rifampicin/pyrazinamide/ethambutol/isoniazid (RPEI), rifampicin/pyrazinamide/isoniazid (RPI), rifamycin (RFM), rifapentine (RFP), simvastatin/fenofibrate (SMF), sodium aminosalicylate (SDA), streptomycin/isoniazid (STI), terizidone (TRZ), thioacetazone (TAT), thioacetazone/isoniazid (THI1), tiocarlide (TCR), and viomycin (VIO)
</p>
</li>
<li> <p><code><a href="#topic+betalactams">betalactams()</a></code> can select: <br /> amoxicillin (AMX), amoxicillin/clavulanic acid (AMC), amoxicillin/sulbactam (AXS), ampicillin (AMP), ampicillin/sulbactam (SAM), apalcillin (APL), aspoxicillin (APX), avibactam (AVB), azidocillin (AZD), azlocillin (AZL), aztreonam (ATM), aztreonam/avibactam (AZA), aztreonam/nacubactam (ANC), bacampicillin (BAM), benzathine benzylpenicillin (BNB), benzathine phenoxymethylpenicillin (BNP), benzylpenicillin (PEN), biapenem (BIA), carbenicillin (CRB), carindacillin (CRN), cefacetrile (CAC), cefaclor (CEC), cefadroxil (CFR), cefalexin (LEX), cefaloridine (RID), cefalotin (CEP), cefamandole (MAN), cefapirin (HAP), cefatrizine (CTZ), cefazedone (CZD), cefazolin (CZO), cefcapene (CCP), cefcapene pivoxil (CCX), cefdinir (CDR), cefditoren (DIT), cefditoren pivoxil (DIX), cefepime (FEP), cefepime/clavulanic acid (CPC), cefepime/nacubactam (FNC), cefepime/tazobactam (FPT), cefetamet (CAT), cefetamet pivoxil (CPI), cefetecol (CCL), cefetrizole (CZL), cefixime (CFM), cefmenoxime (CMX), cefmetazole (CMZ), cefodizime (DIZ), cefonicid (CID), cefoperazone (CFP), cefoperazone/sulbactam (CSL), ceforanide (CND), cefoselis (CSE), cefotaxime (CTX), cefotaxime/clavulanic acid (CTC), cefotaxime/sulbactam (CTS), cefotetan (CTT), cefotiam (CTF), cefotiam hexetil (CHE), cefovecin (FOV), cefoxitin (FOX), cefoxitin screening (FOX1), cefozopran (ZOP), cefpimizole (CFZ), cefpiramide (CPM), cefpirome (CPO), cefpodoxime (CPD), cefpodoxime proxetil (CPX), cefpodoxime/clavulanic acid (CDC), cefprozil (CPR), cefquinome (CEQ), cefroxadine (CRD), cefsulodin (CFS), cefsumide (CSU), ceftaroline (CPT), ceftaroline/avibactam (CPA), ceftazidime (CAZ), ceftazidime/avibactam (CZA), ceftazidime/clavulanic acid (CCV), cefteram (CEM), cefteram pivoxil (CPL), ceftezole (CTL), ceftibuten (CTB), ceftiofur (TIO), ceftizoxime (CZX), ceftizoxime alapivoxil (CZP), ceftobiprole (BPR), ceftobiprole medocaril (CFM1), ceftolozane/tazobactam (CZT), ceftriaxone (CRO), ceftriaxone/beta-lactamase inhibitor (CEB), cefuroxime (CXM), cefuroxime axetil (CXA), cephradine (CED), ciclacillin (CIC), clometocillin (CLM), cloxacillin (CLO), dicloxacillin (DIC), doripenem (DOR), epicillin (EPC), ertapenem (ETP), flucloxacillin (FLC), hetacillin (HET), imipenem (IPM), imipenem/EDTA (IPE), imipenem/relebactam (IMR), latamoxef (LTM), lenampicillin (LEN), loracarbef (LOR), mecillinam (MEC), meropenem (MEM), meropenem/nacubactam (MNC), meropenem/vaborbactam (MEV), metampicillin (MTM), meticillin (MET), mezlocillin (MEZ), mezlocillin/sulbactam (MSU), nacubactam (NAC), nafcillin (NAF), oxacillin (OXA), panipenem (PAN), penamecillin (PNM), penicillin/novobiocin (PNO), penicillin/sulbactam (PSU), pheneticillin (PHE), phenoxymethylpenicillin (PHN), piperacillin (PIP), piperacillin/sulbactam (PIS), piperacillin/tazobactam (TZP), piridicillin (PRC), pivampicillin (PVM), pivmecillinam (PME), procaine benzylpenicillin (PRB), propicillin (PRP), razupenem (RZM), ritipenem (RIT), ritipenem acoxil (RIA), sarmoxicillin (SRX), sulbactam (SUL), sulbenicillin (SBC), sultamicillin (SLT6), talampicillin (TAL), tazobactam (TAZ), tebipenem (TBP), temocillin (TEM), ticarcillin (TIC), and ticarcillin/clavulanic acid (TCC)
</p>
</li>
<li> <p><code><a href="#topic+carbapenems">carbapenems()</a></code> can select: <br /> biapenem (BIA), doripenem (DOR), ertapenem (ETP), imipenem (IPM), imipenem/EDTA (IPE), imipenem/relebactam (IMR), meropenem (MEM), meropenem/nacubactam (MNC), meropenem/vaborbactam (MEV), panipenem (PAN), razupenem (RZM), ritipenem (RIT), ritipenem acoxil (RIA), and tebipenem (TBP)
</p>
</li>
<li> <p><code><a href="#topic+cephalosporins">cephalosporins()</a></code> can select: <br /> cefacetrile (CAC), cefaclor (CEC), cefadroxil (CFR), cefalexin (LEX), cefaloridine (RID), cefalotin (CEP), cefamandole (MAN), cefapirin (HAP), cefatrizine (CTZ), cefazedone (CZD), cefazolin (CZO), cefcapene (CCP), cefcapene pivoxil (CCX), cefdinir (CDR), cefditoren (DIT), cefditoren pivoxil (DIX), cefepime (FEP), cefepime/clavulanic acid (CPC), cefepime/tazobactam (FPT), cefetamet (CAT), cefetamet pivoxil (CPI), cefetecol (CCL), cefetrizole (CZL), cefixime (CFM), cefmenoxime (CMX), cefmetazole (CMZ), cefodizime (DIZ), cefonicid (CID), cefoperazone (CFP), cefoperazone/sulbactam (CSL), ceforanide (CND), cefoselis (CSE), cefotaxime (CTX), cefotaxime/clavulanic acid (CTC), cefotaxime/sulbactam (CTS), cefotetan (CTT), cefotiam (CTF), cefotiam hexetil (CHE), cefovecin (FOV), cefoxitin (FOX), cefoxitin screening (FOX1), cefozopran (ZOP), cefpimizole (CFZ), cefpiramide (CPM), cefpirome (CPO), cefpodoxime (CPD), cefpodoxime proxetil (CPX), cefpodoxime/clavulanic acid (CDC), cefprozil (CPR), cefquinome (CEQ), cefroxadine (CRD), cefsulodin (CFS), cefsumide (CSU), ceftaroline (CPT), ceftaroline/avibactam (CPA), ceftazidime (CAZ), ceftazidime/avibactam (CZA), ceftazidime/clavulanic acid (CCV), cefteram (CEM), cefteram pivoxil (CPL), ceftezole (CTL), ceftibuten (CTB), ceftiofur (TIO), ceftizoxime (CZX), ceftizoxime alapivoxil (CZP), ceftobiprole (BPR), ceftobiprole medocaril (CFM1), ceftolozane/tazobactam (CZT), ceftriaxone (CRO), ceftriaxone/beta-lactamase inhibitor (CEB), cefuroxime (CXM), cefuroxime axetil (CXA), cephradine (CED), latamoxef (LTM), and loracarbef (LOR)
</p>
</li>
<li> <p><code><a href="#topic+cephalosporins_1st">cephalosporins_1st()</a></code> can select: <br /> cefacetrile (CAC), cefadroxil (CFR), cefalexin (LEX), cefaloridine (RID), cefalotin (CEP), cefapirin (HAP), cefatrizine (CTZ), cefazedone (CZD), cefazolin (CZO), cefroxadine (CRD), ceftezole (CTL), and cephradine (CED)
</p>
</li>
<li> <p><code><a href="#topic+cephalosporins_2nd">cephalosporins_2nd()</a></code> can select: <br /> cefaclor (CEC), cefamandole (MAN), cefmetazole (CMZ), cefonicid (CID), ceforanide (CND), cefotetan (CTT), cefotiam (CTF), cefoxitin (FOX), cefoxitin screening (FOX1), cefprozil (CPR), cefuroxime (CXM), cefuroxime axetil (CXA), and loracarbef (LOR)
</p>
</li>
<li> <p><code><a href="#topic+cephalosporins_3rd">cephalosporins_3rd()</a></code> can select: <br /> cefcapene (CCP), cefcapene pivoxil (CCX), cefdinir (CDR), cefditoren (DIT), cefditoren pivoxil (DIX), cefetamet (CAT), cefetamet pivoxil (CPI), cefixime (CFM), cefmenoxime (CMX), cefodizime (DIZ), cefoperazone (CFP), cefoperazone/sulbactam (CSL), cefotaxime (CTX), cefotaxime/clavulanic acid (CTC), cefotaxime/sulbactam (CTS), cefotiam hexetil (CHE), cefovecin (FOV), cefpimizole (CFZ), cefpiramide (CPM), cefpodoxime (CPD), cefpodoxime proxetil (CPX), cefpodoxime/clavulanic acid (CDC), cefsulodin (CFS), ceftazidime (CAZ), ceftazidime/avibactam (CZA), ceftazidime/clavulanic acid (CCV), cefteram (CEM), cefteram pivoxil (CPL), ceftibuten (CTB), ceftiofur (TIO), ceftizoxime (CZX), ceftizoxime alapivoxil (CZP), ceftriaxone (CRO), ceftriaxone/beta-lactamase inhibitor (CEB), and latamoxef (LTM)
</p>
</li>
<li> <p><code><a href="#topic+cephalosporins_4th">cephalosporins_4th()</a></code> can select: <br /> cefepime (FEP), cefepime/clavulanic acid (CPC), cefepime/tazobactam (FPT), cefetecol (CCL), cefoselis (CSE), cefozopran (ZOP), cefpirome (CPO), and cefquinome (CEQ)
</p>
</li>
<li> <p><code><a href="#topic+cephalosporins_5th">cephalosporins_5th()</a></code> can select: <br /> ceftaroline (CPT), ceftaroline/avibactam (CPA), ceftobiprole (BPR), ceftobiprole medocaril (CFM1), and ceftolozane/tazobactam (CZT)
</p>
</li>
<li> <p><code><a href="#topic+fluoroquinolones">fluoroquinolones()</a></code> can select: <br /> besifloxacin (BES), ciprofloxacin (CIP), clinafloxacin (CLX), danofloxacin (DAN), delafloxacin (DFX), difloxacin (DIF), enoxacin (ENX), enrofloxacin (ENR), finafloxacin (FIN), fleroxacin (FLE), garenoxacin (GRN), gatifloxacin (GAT), gemifloxacin (GEM), grepafloxacin (GRX), lascufloxacin (LSC), levofloxacin (LVX), levonadifloxacin (LND), lomefloxacin (LOM), marbofloxacin (MAR), metioxate (MXT), miloxacin (MIL), moxifloxacin (MFX), nadifloxacin (NAD), nifuroquine (NIF), norfloxacin (NOR), ofloxacin (OFX), orbifloxacin (ORB), pazufloxacin (PAZ), pefloxacin (PEF), pradofloxacin (PRA), premafloxacin (PRX), prulifloxacin (PRU), rufloxacin (RFL), sarafloxacin (SAR), sitafloxacin (SIT), sparfloxacin (SPX), temafloxacin (TMX), tilbroquinol (TBQ), tioxacin (TXC), tosufloxacin (TFX), and trovafloxacin (TVA)
</p>
</li>
<li> <p><code><a href="#topic+glycopeptides">glycopeptides()</a></code> can select: <br /> avoparcin (AVO), dalbavancin (DAL), norvancomycin (NVA), oritavancin (ORI), ramoplanin (RAM), teicoplanin (TEC), teicoplanin-macromethod (TCM), telavancin (TLV), vancomycin (VAN), and vancomycin-macromethod (VAM)
</p>
</li>
<li> <p><code><a href="#topic+lincosamides">lincosamides()</a></code> can select: <br /> acetylmidecamycin (ACM), acetylspiramycin (ASP), clindamycin (CLI), gamithromycin (GAM), kitasamycin (KIT), lincomycin (LIN), meleumycin (MEL), nafithromycin (ZWK), pirlimycin (PRL), primycin (PRM), solithromycin (SOL), tildipirosin (TIP), tilmicosin (TIL), tulathromycin (TUL), tylosin (TYL), and tylvalosin (TYL1)
</p>
</li>
<li> <p><code><a href="#topic+lipoglycopeptides">lipoglycopeptides()</a></code> can select: <br /> dalbavancin (DAL), oritavancin (ORI), and telavancin (TLV)
</p>
</li>
<li> <p><code><a href="#topic+macrolides">macrolides()</a></code> can select: <br /> acetylmidecamycin (ACM), acetylspiramycin (ASP), azithromycin (AZM), clarithromycin (CLR), dirithromycin (DIR), erythromycin (ERY), flurithromycin (FLR1), gamithromycin (GAM), josamycin (JOS), kitasamycin (KIT), meleumycin (MEL), midecamycin (MID), miocamycin (MCM), nafithromycin (ZWK), oleandomycin (OLE), pirlimycin (PRL), primycin (PRM), rokitamycin (ROK), roxithromycin (RXT), solithromycin (SOL), spiramycin (SPI), telithromycin (TLT), tildipirosin (TIP), tilmicosin (TIL), troleandomycin (TRL), tulathromycin (TUL), tylosin (TYL), and tylvalosin (TYL1)
</p>
</li>
<li> <p><code><a href="#topic+oxazolidinones">oxazolidinones()</a></code> can select: <br /> cadazolid (CDZ), cycloserine (CYC), linezolid (LNZ), tedizolid (TZD), and thiacetazone (THA)
</p>
</li>
<li> <p><code><a href="#topic+penicillins">penicillins()</a></code> can select: <br /> amoxicillin (AMX), amoxicillin/clavulanic acid (AMC), amoxicillin/sulbactam (AXS), ampicillin (AMP), ampicillin/sulbactam (SAM), apalcillin (APL), aspoxicillin (APX), avibactam (AVB), azidocillin (AZD), azlocillin (AZL), aztreonam (ATM), aztreonam/avibactam (AZA), aztreonam/nacubactam (ANC), bacampicillin (BAM), benzathine benzylpenicillin (BNB), benzathine phenoxymethylpenicillin (BNP), benzylpenicillin (PEN), carbenicillin (CRB), carindacillin (CRN), cefepime/nacubactam (FNC), ciclacillin (CIC), clometocillin (CLM), cloxacillin (CLO), dicloxacillin (DIC), epicillin (EPC), flucloxacillin (FLC), hetacillin (HET), lenampicillin (LEN), mecillinam (MEC), metampicillin (MTM), meticillin (MET), mezlocillin (MEZ), mezlocillin/sulbactam (MSU), nacubactam (NAC), nafcillin (NAF), oxacillin (OXA), penamecillin (PNM), penicillin/novobiocin (PNO), penicillin/sulbactam (PSU), pheneticillin (PHE), phenoxymethylpenicillin (PHN), piperacillin (PIP), piperacillin/sulbactam (PIS), piperacillin/tazobactam (TZP), piridicillin (PRC), pivampicillin (PVM), pivmecillinam (PME), procaine benzylpenicillin (PRB), propicillin (PRP), sarmoxicillin (SRX), sulbactam (SUL), sulbenicillin (SBC), sultamicillin (SLT6), talampicillin (TAL), tazobactam (TAZ), temocillin (TEM), ticarcillin (TIC), and ticarcillin/clavulanic acid (TCC)
</p>
</li>
<li> <p><code><a href="#topic+polymyxins">polymyxins()</a></code> can select: <br /> colistin (COL), polymyxin B (PLB), and polymyxin B/polysorbate 80 (POP)
</p>
</li>
<li> <p><code><a href="#topic+quinolones">quinolones()</a></code> can select: <br /> besifloxacin (BES), cinoxacin (CIN), ciprofloxacin (CIP), clinafloxacin (CLX), danofloxacin (DAN), delafloxacin (DFX), difloxacin (DIF), enoxacin (ENX), enrofloxacin (ENR), finafloxacin (FIN), fleroxacin (FLE), flumequine (FLM), garenoxacin (GRN), gatifloxacin (GAT), gemifloxacin (GEM), grepafloxacin (GRX), lascufloxacin (LSC), levofloxacin (LVX), levonadifloxacin (LND), lomefloxacin (LOM), marbofloxacin (MAR), metioxate (MXT), miloxacin (MIL), moxifloxacin (MFX), nadifloxacin (NAD), nalidixic acid (NAL), nemonoxacin (NEM), nifuroquine (NIF), nitroxoline (NTR), norfloxacin (NOR), ofloxacin (OFX), orbifloxacin (ORB), oxolinic acid (OXO), pazufloxacin (PAZ), pefloxacin (PEF), pipemidic acid (PPA), piromidic acid (PIR), pradofloxacin (PRA), premafloxacin (PRX), prulifloxacin (PRU), rosoxacin (ROS), rufloxacin (RFL), sarafloxacin (SAR), sitafloxacin (SIT), sparfloxacin (SPX), temafloxacin (TMX), tilbroquinol (TBQ), tioxacin (TXC), tosufloxacin (TFX), and trovafloxacin (TVA)
</p>
</li>
<li> <p><code><a href="#topic+streptogramins">streptogramins()</a></code> can select: <br /> pristinamycin (PRI) and quinupristin/dalfopristin (QDA)
</p>
</li>
<li> <p><code><a href="#topic+tetracyclines">tetracyclines()</a></code> can select: <br /> cetocycline (CTO), chlortetracycline (CTE), clomocycline (CLM1), demeclocycline (DEM), doxycycline (DOX), eravacycline (ERV), lymecycline (LYM), metacycline (MTC), minocycline (MNO), omadacycline (OMC), oxytetracycline (OXY), penimepicycline (PNM1), rolitetracycline (RLT), sarecycline (SRC), tetracycline (TCY), and tigecycline (TGC)
</p>
</li>
<li> <p><code><a href="#topic+trimethoprims">trimethoprims()</a></code> can select: <br /> brodimoprim (BDP), sulfadiazine (SDI), sulfadiazine/tetroxoprim (SLT), sulfadiazine/trimethoprim (SLT1), sulfadimethoxine (SUD), sulfadimidine (SDM), sulfadimidine/trimethoprim (SLT2), sulfafurazole (SLF), sulfaisodimidine (SLF1), sulfalene (SLF2), sulfamazone (SZO), sulfamerazine (SLF3), sulfamerazine/trimethoprim (SLT3), sulfamethizole (SLF4), sulfamethoxazole (SMX), sulfamethoxypyridazine (SLF5), sulfametomidine (SLF6), sulfametoxydiazine (SLF7), sulfametrole/trimethoprim (SLT4), sulfamoxole (SLF8), sulfamoxole/trimethoprim (SLT5), sulfanilamide (SLF9), sulfaperin (SLF10), sulfaphenazole (SLF11), sulfapyridine (SLF12), sulfathiazole (SUT), sulfathiourea (SLF13), trimethoprim (TMP), and trimethoprim/sulfamethoxazole (SXT)
</p>
</li>
<li> <p><code><a href="#topic+ureidopenicillins">ureidopenicillins()</a></code> can select: <br /> azlocillin (AZL), mezlocillin (MEZ), piperacillin (PIP), and piperacillin/tazobactam (TZP)
</p>
</li></ul>



<h3>Reference Data Publicly Available</h3>

<p>All data sets in this <code>AMR</code> package (about microorganisms, antibiotics, SIR interpretation, EUCAST rules, etc.) are publicly and freely available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. We also provide tab-separated plain text files that are machine-readable and suitable for input in any software program, such as laboratory information systems. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># `example_isolates` is a data set available in the AMR package.
# See ?example_isolates.
example_isolates


# Examples sections below are split into 'base R', 'dplyr', and 'data.table':


# base R ------------------------------------------------------------------

# select columns 'IPM' (imipenem) and 'MEM' (meropenem)
example_isolates[, carbapenems()]

# select columns 'mo', 'AMK', 'GEN', 'KAN' and 'TOB'
example_isolates[, c("mo", aminoglycosides())]

# select only antibiotic columns with DDDs for oral treatment
example_isolates[, administrable_per_os()]

# filter using any() or all()
example_isolates[any(carbapenems() == "R"), ]
subset(example_isolates, any(carbapenems() == "R"))

# filter on any or all results in the carbapenem columns (i.e., IPM, MEM):
example_isolates[any(carbapenems()), ]
example_isolates[all(carbapenems()), ]

# filter with multiple antibiotic selectors using c()
example_isolates[all(c(carbapenems(), aminoglycosides()) == "R"), ]

# filter + select in one go: get penicillins in carbapenem-resistant strains
example_isolates[any(carbapenems() == "R"), penicillins()]

# You can combine selectors with '&amp;' to be more specific. For example,
# penicillins() would select benzylpenicillin ('peni G') and
# administrable_per_os() would select erythromycin. Yet, when combined these
# drugs are both omitted since benzylpenicillin is not administrable per os
# and erythromycin is not a penicillin:
example_isolates[, penicillins() &amp; administrable_per_os()]

# ab_selector() applies a filter in the `antibiotics` data set and is thus
# very flexible. For instance, to select antibiotic columns with an oral DDD
# of at least 1 gram:
example_isolates[, ab_selector(oral_ddd &gt; 1 &amp; oral_units == "g")]


# dplyr -------------------------------------------------------------------

if (require("dplyr")) {
  tibble(kefzol = random_sir(5)) %&gt;%
    select(cephalosporins())
}

if (require("dplyr")) {
  # get AMR for all aminoglycosides e.g., per ward:
  example_isolates %&gt;%
    group_by(ward) %&gt;%
    summarise(across(aminoglycosides(), resistance))
}
if (require("dplyr")) {
  # You can combine selectors with '&amp;' to be more specific:
  example_isolates %&gt;%
    select(penicillins() &amp; administrable_per_os())
}
if (require("dplyr")) {
  # get AMR for only drugs that matter - no intrinsic resistance:
  example_isolates %&gt;%
    filter(mo_genus() %in% c("Escherichia", "Klebsiella")) %&gt;%
    group_by(ward) %&gt;%
    summarise(across(not_intrinsic_resistant(), resistance))
}
if (require("dplyr")) {
  # get susceptibility for antibiotics whose name contains "trim":
  example_isolates %&gt;%
    filter(first_isolate()) %&gt;%
    group_by(ward) %&gt;%
    summarise(across(ab_selector(name %like% "trim"), susceptibility))
}
if (require("dplyr")) {
  # this will select columns 'IPM' (imipenem) and 'MEM' (meropenem):
  example_isolates %&gt;%
    select(carbapenems())
}
if (require("dplyr")) {
  # this will select columns 'mo', 'AMK', 'GEN', 'KAN' and 'TOB':
  example_isolates %&gt;%
    select(mo, aminoglycosides())
}
if (require("dplyr")) {
  # any() and all() work in dplyr's filter() too:
  example_isolates %&gt;%
    filter(
      any(aminoglycosides() == "R"),
      all(cephalosporins_2nd() == "R")
    )
}
if (require("dplyr")) {
  # also works with c():
  example_isolates %&gt;%
    filter(any(c(carbapenems(), aminoglycosides()) == "R"))
}
if (require("dplyr")) {
  # not setting any/all will automatically apply all():
  example_isolates %&gt;%
    filter(aminoglycosides() == "R")
}
if (require("dplyr")) {
  # this will select columns 'mo' and all antimycobacterial drugs ('RIF'):
  example_isolates %&gt;%
    select(mo, ab_class("mycobact"))
}
if (require("dplyr")) {
  # get bug/drug combinations for only glycopeptides in Gram-positives:
  example_isolates %&gt;%
    filter(mo_is_gram_positive()) %&gt;%
    select(mo, glycopeptides()) %&gt;%
    bug_drug_combinations() %&gt;%
    format()
}
if (require("dplyr")) {
  data.frame(
    some_column = "some_value",
    J01CA01 = "S"
  ) %&gt;% # ATC code of ampicillin
    select(penicillins()) # only the 'J01CA01' column will be selected
}
if (require("dplyr")) {
  # with recent versions of dplyr, this is all equal:
  x &lt;- example_isolates[carbapenems() == "R", ]
  y &lt;- example_isolates %&gt;% filter(carbapenems() == "R")
  z &lt;- example_isolates %&gt;% filter(if_all(carbapenems(), ~ .x == "R"))
  identical(x, y) &amp;&amp; identical(y, z)
}


# data.table --------------------------------------------------------------

# data.table is supported as well, just use it in the same way as with
# base R, but add `with = FALSE` if using a single AB selector.

if (require("data.table")) {
  dt &lt;- as.data.table(example_isolates)

  # this does not work, it returns column *names*
  dt[, carbapenems()]
}
if (require("data.table")) {
  # so `with = FALSE` is required
  dt[, carbapenems(), with = FALSE]
}

# for multiple selections or AB selectors, `with = FALSE` is not needed:
if (require("data.table")) {
  dt[, c("mo", aminoglycosides())]
}
if (require("data.table")) {
  dt[, c(carbapenems(), aminoglycosides())]
}

# row filters are also supported:
if (require("data.table")) {
  dt[any(carbapenems() == "S"), ]
}
if (require("data.table")) {
  dt[any(carbapenems() == "S"), penicillins(), with = FALSE]
}

</code></pre>

<hr>
<h2 id='antibiotics'>Data Sets with 603 Antimicrobial Drugs</h2><span id='topic+antibiotics'></span><span id='topic+antivirals'></span>

<h3>Description</h3>

<p>Two data sets containing all antibiotics/antimycotics and antivirals. Use <code><a href="#topic+as.ab">as.ab()</a></code> or one of the <code><a href="#topic+ab_property">ab_*</a></code> functions to retrieve values from the <a href="#topic+antibiotics">antibiotics</a> data set. Three identifiers are included in this data set: an antibiotic ID (<code>ab</code>, primarily used in this package) as defined by WHONET/EARS-Net, an ATC code (<code>atc</code>) as defined by the WHO, and a Compound ID (<code>cid</code>) as found in PubChem. Other properties in this data set are derived from one or more of these codes. Note that some drugs have multiple ATC codes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>antibiotics

antivirals
</code></pre>


<h3>Format</h3>



<h4>For the <a href="#topic+antibiotics">antibiotics</a> data set: a <a href="tibble.html#topic+tibble">tibble</a> with 483 observations and 14 variables:</h4>


<ul>
<li> <p><code>ab</code><br /> Antibiotic ID as used in this package (such as <code>AMC</code>), using the official EARS-Net (European Antimicrobial Resistance Surveillance Network) codes where available. <em>This is a unique identifier.</em>
</p>
</li>
<li> <p><code>cid</code><br /> Compound ID as found in PubChem. <em>This is a unique identifier.</em>
</p>
</li>
<li> <p><code>name</code><br /> Official name as used by WHONET/EARS-Net or the WHO. <em>This is a unique identifier.</em>
</p>
</li>
<li> <p><code>group</code><br /> A short and concise group name, based on WHONET and WHOCC definitions
</p>
</li>
<li> <p><code>atc</code><br /> ATC codes (Anatomical Therapeutic Chemical) as defined by the WHOCC, like <code>J01CR02</code>
</p>
</li>
<li> <p><code>atc_group1</code><br /> Official pharmacological subgroup (3rd level ATC code) as defined by the WHOCC, like <code>"Macrolides, lincosamides and streptogramins"</code>
</p>
</li>
<li> <p><code>atc_group2</code><br /> Official chemical subgroup (4th level ATC code) as defined by the WHOCC, like <code>"Macrolides"</code>
</p>
</li>
<li> <p><code>abbr</code><br /> List of abbreviations as used in many countries, also for antibiotic susceptibility testing (AST)
</p>
</li>
<li> <p><code>synonyms</code><br /> Synonyms (often trade names) of a drug, as found in PubChem based on their compound ID
</p>
</li>
<li> <p><code>oral_ddd</code><br /> Defined Daily Dose (DDD), oral treatment, currently available for 174 drugs
</p>
</li>
<li> <p><code>oral_units</code><br /> Units of <code>oral_ddd</code>
</p>
</li>
<li> <p><code>iv_ddd</code><br /> Defined Daily Dose (DDD), parenteral (intravenous) treatment, currently available for 146 drugs
</p>
</li>
<li> <p><code>iv_units</code><br /> Units of <code>iv_ddd</code>
</p>
</li>
<li> <p><code>loinc</code><br /> All codes associated with the name of the antimicrobial drug from Logical Observation Identifiers Names and Codes (LOINC), Version 2.76 (18 September, 2023). Use <code><a href="#topic+ab_loinc">ab_loinc()</a></code> to retrieve them quickly, see <code><a href="#topic+ab_property">ab_property()</a></code>.
</p>
</li></ul>




<h4>For the <a href="#topic+antivirals">antivirals</a> data set: a <a href="tibble.html#topic+tibble">tibble</a> with 120 observations and 11 variables:</h4>


<ul>
<li> <p><code>av</code><br /> Antiviral ID as used in this package (such as <code>ACI</code>), using the official EARS-Net (European Antimicrobial Resistance Surveillance Network) codes where available. <em>This is a unique identifier.</em> Combinations are codes that contain a <code>+</code> to indicate this, such as <code>ATA+COBI</code> for atazanavir/cobicistat.
</p>
</li>
<li> <p><code>name</code><br /> Official name as used by WHONET/EARS-Net or the WHO. <em>This is a unique identifier.</em>
</p>
</li>
<li> <p><code>atc</code><br /> ATC codes (Anatomical Therapeutic Chemical) as defined by the WHOCC
</p>
</li>
<li> <p><code>cid</code><br /> Compound ID as found in PubChem. <em>This is a unique identifier.</em>
</p>
</li>
<li> <p><code>atc_group</code><br /> Official pharmacological subgroup (3rd level ATC code) as defined by the WHOCC
</p>
</li>
<li> <p><code>synonyms</code><br /> Synonyms (often trade names) of a drug, as found in PubChem based on their compound ID
</p>
</li>
<li> <p><code>oral_ddd</code><br /> Defined Daily Dose (DDD), oral treatment
</p>
</li>
<li> <p><code>oral_units</code><br /> Units of <code>oral_ddd</code>
</p>
</li>
<li> <p><code>iv_ddd</code><br /> Defined Daily Dose (DDD), parenteral treatment
</p>
</li>
<li> <p><code>iv_units</code><br /> Units of <code>iv_ddd</code>
</p>
</li>
<li> <p><code>loinc</code><br /> All codes associated with the name of the antiviral drug from Logical Observation Identifiers Names and Codes (LOINC), Version 2.76 (18 September, 2023). Use <code><a href="#topic+av_loinc">av_loinc()</a></code> to retrieve them quickly, see <code><a href="#topic+av_property">av_property()</a></code>.
</p>
</li></ul>


<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 120 rows and 11 columns.
</p>


<h3>Details</h3>

<p>Properties that are based on an ATC code are only available when an ATC is available. These properties are: <code>atc_group1</code>, <code>atc_group2</code>, <code>oral_ddd</code>, <code>oral_units</code>, <code>iv_ddd</code> and <code>iv_units</code>.
</p>
<p>Synonyms (i.e. trade names) were derived from the PubChem Compound ID (column <code>cid</code>) and consequently only available where a CID is available.
</p>


<h4>Direct download</h4>

<p>Like all data sets in this package, these data sets are publicly available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>.
</p>



<h3>WHOCC</h3>

<p>This package contains <strong>all ~550 antibiotic, antimycotic and antiviral drugs</strong> and their Anatomical Therapeutic Chemical (ATC) codes, ATC groups and Defined Daily Dose (DDD) from the World Health Organization Collaborating Centre for Drug Statistics Methodology (WHOCC, <a href="https://www.whocc.no">https://www.whocc.no</a>) and the Pharmaceuticals Community Register of the European Commission (<a href="https://ec.europa.eu/health/documents/community-register/html/reg_hum_atc.htm">https://ec.europa.eu/health/documents/community-register/html/reg_hum_atc.htm</a>).
</p>
<p>These have become the gold standard for international drug utilisation monitoring and research.
</p>
<p>The WHOCC is located in Oslo at the Norwegian Institute of Public Health and funded by the Norwegian government. The European Commission is the executive of the European Union and promotes its general interest.
</p>
<p><strong>NOTE: The WHOCC copyright does not allow use for commercial purposes, unlike any other info from this package.</strong> See <a href="https://www.whocc.no/copyright_disclaimer/.">https://www.whocc.no/copyright_disclaimer/.</a>
</p>


<h3>Source</h3>


<ul>
<li><p> World Health Organization (WHO) Collaborating Centre for Drug Statistics Methodology (WHOCC): <a href="https://www.whocc.no/atc_ddd_index/">https://www.whocc.no/atc_ddd_index/</a>
</p>
</li>
<li><p> Logical Observation Identifiers Names and Codes (LOINC), Version 2.76 (18 September, 2023). Accessed from <a href="https://loinc.org">https://loinc.org</a> on October 19th, 2023.
</p>
</li>
<li><p> European Commission Public Health PHARMACEUTICALS - COMMUNITY REGISTER: <a href="https://ec.europa.eu/health/documents/community-register/html/reg_hum_atc.htm">https://ec.europa.eu/health/documents/community-register/html/reg_hum_atc.htm</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="#topic+microorganisms">microorganisms</a>, <a href="#topic+intrinsic_resistant">intrinsic_resistant</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>antibiotics
antivirals
</code></pre>

<hr>
<h2 id='as.ab'>Transform Input to an Antibiotic ID</h2><span id='topic+as.ab'></span><span id='topic+ab'></span><span id='topic+is.ab'></span>

<h3>Description</h3>

<p>Use this function to determine the antibiotic drug code of one or more antibiotics. The data set <a href="#topic+antibiotics">antibiotics</a> will be searched for abbreviations, official names and synonyms (brand names).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.ab(x, flag_multiple_results = TRUE, info = interactive(), ...)

is.ab(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.ab_+3A_x">x</code></td>
<td>
<p>a <a href="base.html#topic+character">character</a> vector to determine to antibiotic ID</p>
</td></tr>
<tr><td><code id="as.ab_+3A_flag_multiple_results">flag_multiple_results</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether a note should be printed to the console that probably more than one antibiotic drug code or name can be retrieved from a single input value.</p>
</td></tr>
<tr><td><code id="as.ab_+3A_info">info</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether a progress bar should be printed - the default is <code>TRUE</code> only in interactive mode</p>
</td></tr>
<tr><td><code id="as.ab_+3A_...">...</code></td>
<td>
<p>arguments passed on to internal functions</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All entries in the <a href="#topic+antibiotics">antibiotics</a> data set have three different identifiers: a human readable EARS-Net code (column <code>ab</code>, used by ECDC and WHONET), an ATC code (column <code>atc</code>, used by WHO), and a CID code (column <code>cid</code>, Compound ID, used by PubChem). The data set contains more than 5,000 official brand names from many different countries, as found in PubChem. Not that some drugs contain multiple ATC codes.
</p>
<p>All these properties will be searched for the user input. The <code><a href="#topic+as.ab">as.ab()</a></code> can correct for different forms of misspelling:
</p>

<ul>
<li><p> Wrong spelling of drug names (such as &quot;tobramicin&quot; or &quot;gentamycin&quot;), which corrects for most audible similarities such as f/ph, x/ks, c/z/s, t/th, etc.
</p>
</li>
<li><p> Too few or too many vowels or consonants
</p>
</li>
<li><p> Switching two characters (such as &quot;mreopenem&quot;, often the case in clinical data, when doctors typed too fast)
</p>
</li>
<li><p> Digitalised paper records, leaving artefacts like 0/o/O (zero and O's), B/8, n/r, etc.
</p>
</li></ul>

<p>Use the <code><a href="#topic+ab_property">ab_*</a></code> functions to get properties based on the returned antibiotic ID, see <em>Examples</em>.
</p>
<p>Note: the <code><a href="#topic+as.ab">as.ab()</a></code> and <code><a href="#topic+ab_property">ab_*</a></code> functions may use very long regular expression to match brand names of antimicrobial drugs. This may fail on some systems.
</p>
<p>You can add your own manual codes to be considered by <code><a href="#topic+as.ab">as.ab()</a></code> and all <code><a href="#topic+ab_property">ab_*</a></code> functions, see <code><a href="#topic+add_custom_antimicrobials">add_custom_antimicrobials()</a></code>.
</p>


<h3>Value</h3>

<p>A <a href="base.html#topic+character">character</a> <a href="base.html#topic+vector">vector</a> with additional class <code><a href="#topic+ab">ab</a></code>
</p>


<h3>Source</h3>

<p>World Health Organization (WHO) Collaborating Centre for Drug Statistics Methodology: <a href="https://www.whocc.no/atc_ddd_index/">https://www.whocc.no/atc_ddd_index/</a>
</p>
<p>European Commission Public Health PHARMACEUTICALS - COMMUNITY REGISTER: <a href="https://ec.europa.eu/health/documents/community-register/html/reg_hum_atc.htm">https://ec.europa.eu/health/documents/community-register/html/reg_hum_atc.htm</a>
</p>


<h3>WHOCC</h3>

<p>This package contains <strong>all ~550 antibiotic, antimycotic and antiviral drugs</strong> and their Anatomical Therapeutic Chemical (ATC) codes, ATC groups and Defined Daily Dose (DDD) from the World Health Organization Collaborating Centre for Drug Statistics Methodology (WHOCC, <a href="https://www.whocc.no">https://www.whocc.no</a>) and the Pharmaceuticals Community Register of the European Commission (<a href="https://ec.europa.eu/health/documents/community-register/html/reg_hum_atc.htm">https://ec.europa.eu/health/documents/community-register/html/reg_hum_atc.htm</a>).
</p>
<p>These have become the gold standard for international drug utilisation monitoring and research.
</p>
<p>The WHOCC is located in Oslo at the Norwegian Institute of Public Health and funded by the Norwegian government. The European Commission is the executive of the European Union and promotes its general interest.
</p>
<p><strong>NOTE: The WHOCC copyright does not allow use for commercial purposes, unlike any other info from this package.</strong> See <a href="https://www.whocc.no/copyright_disclaimer/.">https://www.whocc.no/copyright_disclaimer/.</a>
</p>


<h3>Reference Data Publicly Available</h3>

<p>All data sets in this <code>AMR</code> package (about microorganisms, antibiotics, SIR interpretation, EUCAST rules, etc.) are publicly and freely available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. We also provide tab-separated plain text files that are machine-readable and suitable for input in any software program, such as laboratory information systems. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>.
</p>


<h3>See Also</h3>


<ul>
<li> <p><a href="#topic+antibiotics">antibiotics</a> for the <a href="base.html#topic+data.frame">data.frame</a> that is being used to determine ATCs
</p>
</li>
<li> <p><code><a href="#topic+ab_from_text">ab_from_text()</a></code> for a function to retrieve antimicrobial drugs from clinical text (from health care records)
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># these examples all return "ERY", the ID of erythromycin:
as.ab("J01FA01")
as.ab("J 01 FA 01")
as.ab("Erythromycin")
as.ab("eryt")
as.ab("   eryt 123")
as.ab("ERYT")
as.ab("ERY")
as.ab("eritromicine") # spelled wrong, yet works
as.ab("Erythrocin") # trade name
as.ab("Romycin") # trade name

# spelling from different languages and dyslexia are no problem
ab_atc("ceftriaxon")
ab_atc("cephtriaxone") # small spelling error
ab_atc("cephthriaxone") # or a bit more severe
ab_atc("seephthriaaksone") # and even this works

# use ab_* functions to get a specific properties (see ?ab_property);
# they use as.ab() internally:
ab_name("J01FA01")
ab_name("eryt")


if (require("dplyr")) {
  # you can quickly rename 'sir' columns using set_ab_names() with dplyr:
  example_isolates %&gt;%
    set_ab_names(where(is.sir), property = "atc")
}

</code></pre>

<hr>
<h2 id='as.av'>Transform Input to an Antiviral Drug ID</h2><span id='topic+as.av'></span><span id='topic+av'></span><span id='topic+is.av'></span>

<h3>Description</h3>

<p>Use this function to determine the antiviral drug code of one or more antiviral drugs. The data set <a href="#topic+antivirals">antivirals</a> will be searched for abbreviations, official names and synonyms (brand names).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.av(x, flag_multiple_results = TRUE, info = interactive(), ...)

is.av(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.av_+3A_x">x</code></td>
<td>
<p>a <a href="base.html#topic+character">character</a> vector to determine to antiviral drug ID</p>
</td></tr>
<tr><td><code id="as.av_+3A_flag_multiple_results">flag_multiple_results</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether a note should be printed to the console that probably more than one antiviral drug code or name can be retrieved from a single input value.</p>
</td></tr>
<tr><td><code id="as.av_+3A_info">info</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether a progress bar should be printed - the default is <code>TRUE</code> only in interactive mode</p>
</td></tr>
<tr><td><code id="as.av_+3A_...">...</code></td>
<td>
<p>arguments passed on to internal functions</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All entries in the <a href="#topic+antivirals">antivirals</a> data set have three different identifiers: a human readable EARS-Net code (column <code>ab</code>, used by ECDC and WHONET), an ATC code (column <code>atc</code>, used by WHO), and a CID code (column <code>cid</code>, Compound ID, used by PubChem). The data set contains more than 5,000 official brand names from many different countries, as found in PubChem. Not that some drugs contain multiple ATC codes.
</p>
<p>All these properties will be searched for the user input. The <code><a href="#topic+as.av">as.av()</a></code> can correct for different forms of misspelling:
</p>

<ul>
<li><p> Wrong spelling of drug names (such as &quot;acyclovir&quot;), which corrects for most audible similarities such as f/ph, x/ks, c/z/s, t/th, etc.
</p>
</li>
<li><p> Too few or too many vowels or consonants
</p>
</li>
<li><p> Switching two characters (such as &quot;aycclovir&quot;, often the case in clinical data, when doctors typed too fast)
</p>
</li>
<li><p> Digitalised paper records, leaving artefacts like 0/o/O (zero and O's), B/8, n/r, etc.
</p>
</li></ul>

<p>Use the <code><a href="#topic+av_property">av_*</a></code> functions to get properties based on the returned antiviral drug ID, see <em>Examples</em>.
</p>
<p>Note: the <code><a href="#topic+as.av">as.av()</a></code> and <code><a href="#topic+av_property">av_*</a></code> functions may use very long regular expression to match brand names of antimicrobial drugs. This may fail on some systems.
</p>


<h3>Value</h3>

<p>A <a href="base.html#topic+character">character</a> <a href="base.html#topic+vector">vector</a> with additional class <code><a href="#topic+ab">ab</a></code>
</p>


<h3>Source</h3>

<p>World Health Organization (WHO) Collaborating Centre for Drug Statistics Methodology: <a href="https://www.whocc.no/atc_ddd_index/">https://www.whocc.no/atc_ddd_index/</a>
</p>
<p>European Commission Public Health PHARMACEUTICALS - COMMUNITY REGISTER: <a href="https://ec.europa.eu/health/documents/community-register/html/reg_hum_atc.htm">https://ec.europa.eu/health/documents/community-register/html/reg_hum_atc.htm</a>
</p>


<h3>WHOCC</h3>

<p>This package contains <strong>all ~550 antibiotic, antimycotic and antiviral drugs</strong> and their Anatomical Therapeutic Chemical (ATC) codes, ATC groups and Defined Daily Dose (DDD) from the World Health Organization Collaborating Centre for Drug Statistics Methodology (WHOCC, <a href="https://www.whocc.no">https://www.whocc.no</a>) and the Pharmaceuticals Community Register of the European Commission (<a href="https://ec.europa.eu/health/documents/community-register/html/reg_hum_atc.htm">https://ec.europa.eu/health/documents/community-register/html/reg_hum_atc.htm</a>).
</p>
<p>These have become the gold standard for international drug utilisation monitoring and research.
</p>
<p>The WHOCC is located in Oslo at the Norwegian Institute of Public Health and funded by the Norwegian government. The European Commission is the executive of the European Union and promotes its general interest.
</p>
<p><strong>NOTE: The WHOCC copyright does not allow use for commercial purposes, unlike any other info from this package.</strong> See <a href="https://www.whocc.no/copyright_disclaimer/.">https://www.whocc.no/copyright_disclaimer/.</a>
</p>


<h3>Reference Data Publicly Available</h3>

<p>All data sets in this <code>AMR</code> package (about microorganisms, antibiotics, SIR interpretation, EUCAST rules, etc.) are publicly and freely available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. We also provide tab-separated plain text files that are machine-readable and suitable for input in any software program, such as laboratory information systems. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>.
</p>


<h3>See Also</h3>


<ul>
<li> <p><a href="#topic+antivirals">antivirals</a> for the <a href="base.html#topic+data.frame">data.frame</a> that is being used to determine ATCs
</p>
</li>
<li> <p><code><a href="#topic+av_from_text">av_from_text()</a></code> for a function to retrieve antimicrobial drugs from clinical text (from health care records)
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># these examples all return "ACI", the ID of aciclovir:
as.av("J05AB01")
as.av("J 05 AB 01")
as.av("Aciclovir")
as.av("aciclo")
as.av("   aciclo 123")
as.av("ACICL")
as.av("ACI")
as.av("Virorax") # trade name
as.av("Zovirax") # trade name

as.av("acyklofir") # severe spelling error, yet works

# use av_* functions to get a specific properties (see ?av_property);
# they use as.av() internally:
av_name("J05AB01")
av_name("acicl")
</code></pre>

<hr>
<h2 id='as.disk'>Transform Input to Disk Diffusion Diameters</h2><span id='topic+as.disk'></span><span id='topic+disk'></span><span id='topic+NA_disk_'></span><span id='topic+is.disk'></span>

<h3>Description</h3>

<p>This transforms a vector to a new class <code><a href="#topic+disk">disk</a></code>, which is a disk diffusion growth zone size (around an antibiotic disk) in millimetres between 6 and 50.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.disk(x, na.rm = FALSE)

NA_disk_

is.disk(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.disk_+3A_x">x</code></td>
<td>
<p>vector</p>
</td></tr>
<tr><td><code id="as.disk_+3A_na.rm">na.rm</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> indicating whether missing values should be removed</p>
</td></tr>
</table>


<h3>Format</h3>

<p>An object of class <code>disk</code> (inherits from <code>integer</code>) of length 1.
</p>


<h3>Details</h3>

<p>Interpret disk values as SIR values with <code><a href="#topic+as.sir">as.sir()</a></code>. It supports guidelines from EUCAST and CLSI.
</p>
<p>Disk diffusion growth zone sizes must be between 6 and 50 millimetres. Values higher than 50 but lower than 100 will be maximised to 50. All others input values outside the 6-50 range will return <code>NA</code>.
</p>
<p><code>NA_disk_</code> is a missing value of the new <code>disk</code> class.
</p>


<h3>Value</h3>

<p>An <a href="base.html#topic+integer">integer</a> with additional class <code><a href="#topic+disk">disk</a></code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+as.sir">as.sir()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># transform existing disk zones to the `disk` class (using base R)
df &lt;- data.frame(
  microorganism = "Escherichia coli",
  AMP = 20,
  CIP = 14,
  GEN = 18,
  TOB = 16
)
df[, 2:5] &lt;- lapply(df[, 2:5], as.disk)
str(df)


# transforming is easier with dplyr:
if (require("dplyr")) {
  df %&gt;% mutate(across(AMP:TOB, as.disk))
}


# interpret disk values, see ?as.sir
as.sir(
  x = as.disk(18),
  mo = "Strep pneu", # `mo` will be coerced with as.mo()
  ab = "ampicillin", # and `ab` with as.ab()
  guideline = "EUCAST"
)

# interpret whole data set, pretend to be all from urinary tract infections:
as.sir(df, uti = TRUE)
</code></pre>

<hr>
<h2 id='as.mic'>Transform Input to Minimum Inhibitory Concentrations (MIC)</h2><span id='topic+as.mic'></span><span id='topic+mic'></span><span id='topic+NA_mic_'></span><span id='topic+is.mic'></span><span id='topic+droplevels.mic'></span>

<h3>Description</h3>

<p>This transforms vectors to a new class <code><a href="#topic+mic">mic</a></code>, which treats the input as decimal numbers, while maintaining operators (such as &quot;&gt;=&quot;) and only allowing valid MIC values known to the field of (medical) microbiology.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.mic(x, na.rm = FALSE)

NA_mic_

is.mic(x)

## S3 method for class 'mic'
droplevels(x, as.mic = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.mic_+3A_x">x</code></td>
<td>
<p>a <a href="base.html#topic+character">character</a> or <a href="base.html#topic+numeric">numeric</a> vector</p>
</td></tr>
<tr><td><code id="as.mic_+3A_na.rm">na.rm</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> indicating whether missing values should be removed</p>
</td></tr>
<tr><td><code id="as.mic_+3A_as.mic">as.mic</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether the <code>mic</code> class should be kept - the default is <code>FALSE</code></p>
</td></tr>
<tr><td><code id="as.mic_+3A_...">...</code></td>
<td>
<p>arguments passed on to methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To interpret MIC values as SIR values, use <code><a href="#topic+as.sir">as.sir()</a></code> on MIC values. It supports guidelines from EUCAST (2011-2023) and CLSI (2011-2023).
</p>
<p>This class for MIC values is a quite a special data type: formally it is an ordered <a href="base.html#topic+factor">factor</a> with valid MIC values as <a href="base.html#topic+factor">factor</a> levels (to make sure only valid MIC values are retained), but for any mathematical operation it acts as decimal numbers:
</p>
<div class="sourceCode"><pre>x &lt;- random_mic(10)
x
#&gt; Class 'mic'
#&gt;  [1] 16     1      8      8      64     &gt;=128  0.0625 32     32     16

is.factor(x)
#&gt; [1] TRUE

x[1] * 2
#&gt; [1] 32

median(x)
#&gt; [1] 26
</pre></div>
<p>This makes it possible to maintain operators that often come with MIC values, such &quot;&gt;=&quot; and &quot;&lt;=&quot;, even when filtering using <a href="base.html#topic+numeric">numeric</a> values in data analysis, e.g.:
</p>
<div class="sourceCode"><pre>x[x &gt; 4]
#&gt; Class 'mic'
#&gt; [1] 16    8     8     64    &gt;=128 32    32    16

df &lt;- data.frame(x, hospital = "A")
subset(df, x &gt; 4) # or with dplyr: df %&gt;% filter(x &gt; 4)
#&gt;        x hospital
#&gt; 1     16        A
#&gt; 5     64        A
#&gt; 6  &gt;=128        A
#&gt; 8     32        A
#&gt; 9     32        A
#&gt; 10    16        A
</pre></div>
<p>The following <a href="base.html#topic+groupGeneric">generic functions</a> are implemented for the MIC class: <code>!</code>, <code>!=</code>, <code>%%</code>, <code>%/%</code>, <code>&amp;</code>, <code>*</code>, <code>+</code>, <code>-</code>, <code>/</code>, <code>&lt;</code>, <code>&lt;=</code>, <code>==</code>, <code>&gt;</code>, <code>&gt;=</code>, <code>^</code>, <code>|</code>, <code><a href="base.html#topic+abs">abs()</a></code>, <code><a href="base.html#topic+acos">acos()</a></code>, <code><a href="base.html#topic+acosh">acosh()</a></code>, <code><a href="base.html#topic+all">all()</a></code>, <code><a href="base.html#topic+any">any()</a></code>, <code><a href="base.html#topic+asin">asin()</a></code>, <code><a href="base.html#topic+asinh">asinh()</a></code>, <code><a href="base.html#topic+atan">atan()</a></code>, <code><a href="base.html#topic+atanh">atanh()</a></code>, <code><a href="base.html#topic+ceiling">ceiling()</a></code>, <code><a href="base.html#topic+cos">cos()</a></code>, <code><a href="base.html#topic+cosh">cosh()</a></code>, <code><a href="base.html#topic+cospi">cospi()</a></code>, <code><a href="base.html#topic+cummax">cummax()</a></code>, <code><a href="base.html#topic+cummin">cummin()</a></code>, <code><a href="base.html#topic+cumprod">cumprod()</a></code>, <code><a href="base.html#topic+cumsum">cumsum()</a></code>, <code><a href="base.html#topic+digamma">digamma()</a></code>, <code><a href="base.html#topic+exp">exp()</a></code>, <code><a href="base.html#topic+expm1">expm1()</a></code>, <code><a href="base.html#topic+floor">floor()</a></code>, <code><a href="base.html#topic+gamma">gamma()</a></code>, <code><a href="base.html#topic+lgamma">lgamma()</a></code>, <code><a href="base.html#topic+log">log()</a></code>, <code><a href="base.html#topic+log1p">log1p()</a></code>, <code><a href="base.html#topic+log2">log2()</a></code>, <code><a href="base.html#topic+log10">log10()</a></code>, <code><a href="base.html#topic+max">max()</a></code>, <code><a href="base.html#topic+mean">mean()</a></code>, <code><a href="base.html#topic+min">min()</a></code>, <code><a href="base.html#topic+prod">prod()</a></code>, <code><a href="base.html#topic+range">range()</a></code>, <code><a href="base.html#topic+round">round()</a></code>, <code><a href="base.html#topic+sign">sign()</a></code>, <code><a href="base.html#topic+signif">signif()</a></code>, <code><a href="base.html#topic+sin">sin()</a></code>, <code><a href="base.html#topic+sinh">sinh()</a></code>, <code><a href="base.html#topic+sinpi">sinpi()</a></code>, <code><a href="base.html#topic+sqrt">sqrt()</a></code>, <code><a href="base.html#topic+sum">sum()</a></code>, <code><a href="base.html#topic+tan">tan()</a></code>, <code><a href="base.html#topic+tanh">tanh()</a></code>, <code><a href="base.html#topic+tanpi">tanpi()</a></code>, <code><a href="base.html#topic+trigamma">trigamma()</a></code> and <code><a href="base.html#topic+trunc">trunc()</a></code>. Some functions of the <code>stats</code> package are also implemented: <code><a href="stats.html#topic+median">median()</a></code>, <code><a href="stats.html#topic+quantile">quantile()</a></code>, <code><a href="stats.html#topic+mad">mad()</a></code>, <code><a href="stats.html#topic+IQR">IQR()</a></code>, <code><a href="stats.html#topic+fivenum">fivenum()</a></code>. Also, <code><a href="grDevices.html#topic+boxplot.stats">boxplot.stats()</a></code> is supported. Since <code><a href="stats.html#topic+sd">sd()</a></code> and <code><a href="stats.html#topic+var">var()</a></code> are non-generic functions, these could not be extended. Use <code><a href="stats.html#topic+mad">mad()</a></code> as an alternative, or use e.g. <code>sd(as.numeric(x))</code> where <code>x</code> is your vector of MIC values.
</p>
<p>Using <code><a href="base.html#topic+as.double">as.double()</a></code> or <code><a href="base.html#topic+as.numeric">as.numeric()</a></code> on MIC values will remove the operators and return a numeric vector. Do <strong>not</strong> use <code><a href="base.html#topic+as.integer">as.integer()</a></code> on MIC values as by the <span class="rlang"><b>R</b></span> convention on <a href="base.html#topic+factor">factor</a>s, it will return the index of the factor levels (which is often useless for regular users).
</p>
<p>Use <code><a href="base.html#topic+droplevels">droplevels()</a></code> to drop unused levels. At default, it will return a plain factor. Use <code>droplevels(..., as.mic = TRUE)</code> to maintain the <code>mic</code> class.
</p>
<p><code>NA_mic_</code> is a missing value of the new <code>mic</code> class, analogous to e.g. base <span class="rlang"><b>R</b></span>'s <code><a href="base.html#topic+NA">NA_character_</a></code>.
</p>


<h3>Value</h3>

<p>Ordered <a href="base.html#topic+factor">factor</a> with additional class <code><a href="#topic+mic">mic</a></code>, that in mathematical operations acts as decimal numbers. Bare in mind that the outcome of any mathematical operation on MICs will return a <a href="base.html#topic+numeric">numeric</a> value.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+as.sir">as.sir()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mic_data &lt;- as.mic(c("&gt;=32", "1.0", "1", "1.00", 8, "&lt;=0.128", "8", "16", "16"))
mic_data
is.mic(mic_data)

# this can also coerce combined MIC/SIR values:
as.mic("&lt;=0.002; S")

# mathematical processing treats MICs as numeric values
fivenum(mic_data)
quantile(mic_data)
all(mic_data &lt; 512)

# interpret MIC values
as.sir(
  x = as.mic(2),
  mo = as.mo("Streptococcus pneumoniae"),
  ab = "AMX",
  guideline = "EUCAST"
)
as.sir(
  x = as.mic(c(0.01, 2, 4, 8)),
  mo = as.mo("Streptococcus pneumoniae"),
  ab = "AMX",
  guideline = "EUCAST"
)

# plot MIC values, see ?plot
plot(mic_data)
plot(mic_data, mo = "E. coli", ab = "cipro")

if (require("ggplot2")) {
  autoplot(mic_data, mo = "E. coli", ab = "cipro")
}
if (require("ggplot2")) {
  autoplot(mic_data, mo = "E. coli", ab = "cipro", language = "nl") # Dutch
}
</code></pre>

<hr>
<h2 id='as.mo'>Transform Arbitrary Input to Valid Microbial Taxonomy</h2><span id='topic+as.mo'></span><span id='topic+mo'></span><span id='topic+is.mo'></span><span id='topic+mo_uncertainties'></span><span id='topic+mo_renamed'></span><span id='topic+mo_failures'></span><span id='topic+mo_reset_session'></span><span id='topic+mo_cleaning_regex'></span>

<h3>Description</h3>

<p>Use this function to get a valid microorganism code (<code><a href="#topic+mo">mo</a></code>) based on arbitrary user input. Determination is done using intelligent rules and the complete taxonomic tree of the kingdoms Animalia, Archaea, Bacteria, and Protozoa, and most microbial species from the kingdom Fungi (see <em>Source</em>). The input can be almost anything: a full name (like <code>"Staphylococcus aureus"</code>), an abbreviated name (such as <code>"S. aureus"</code>), an abbreviation known in the field (such as <code>"MRSA"</code>), or just a genus. See <em>Examples</em>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.mo(
  x,
  Becker = FALSE,
  Lancefield = FALSE,
  minimum_matching_score = NULL,
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  reference_df = get_mo_source(),
  ignore_pattern = getOption("AMR_ignore_pattern", NULL),
  cleaning_regex = getOption("AMR_cleaning_regex", mo_cleaning_regex()),
  language = get_AMR_locale(),
  info = interactive(),
  ...
)

is.mo(x)

mo_uncertainties()

mo_renamed()

mo_failures()

mo_reset_session()

mo_cleaning_regex()
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.mo_+3A_x">x</code></td>
<td>
<p>a <a href="base.html#topic+character">character</a> vector or a <a href="base.html#topic+data.frame">data.frame</a> with one or two columns</p>
</td></tr>
<tr><td><code id="as.mo_+3A_becker">Becker</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether staphylococci should be categorised into coagulase-negative staphylococci (&quot;CoNS&quot;) and coagulase-positive staphylococci (&quot;CoPS&quot;) instead of their own species, according to Karsten Becker <em>et al.</em> (see <em>Source</em>). Please see <em>Details</em> for a full list of staphylococcal species that will be converted.
</p>
<p>This excludes <em>Staphylococcus aureus</em> at default, use <code>Becker = "all"</code> to also categorise <em>S. aureus</em> as &quot;CoPS&quot;.</p>
</td></tr>
<tr><td><code id="as.mo_+3A_lancefield">Lancefield</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether a beta-haemolytic <em>Streptococcus</em> should be categorised into Lancefield groups instead of their own species, according to Rebecca C. Lancefield (see <em>Source</em>). These streptococci will be categorised in their first group, e.g. <em>Streptococcus dysgalactiae</em> will be group C, although officially it was also categorised into groups G and L. . Please see <em>Details</em> for a full list of streptococcal species that will be converted.
</p>
<p>This excludes enterococci at default (who are in group D), use <code>Lancefield = "all"</code> to also categorise all enterococci as group D.</p>
</td></tr>
<tr><td><code id="as.mo_+3A_minimum_matching_score">minimum_matching_score</code></td>
<td>
<p>a numeric value to set as the lower limit for the <a href="#topic+mo_matching_score">MO matching score</a>. When left blank, this will be determined automatically based on the character length of <code>x</code>, its <a href="#topic+microorganisms">taxonomic kingdom</a> and <a href="#topic+mo_matching_score">human pathogenicity</a>.</p>
</td></tr>
<tr><td><code id="as.mo_+3A_keep_synonyms">keep_synonyms</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate if old, previously valid taxonomic names must be preserved and not be corrected to currently accepted names. The default is <code>FALSE</code>, which will return a note if old taxonomic names were processed. The default can be set with the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_keep_synonyms</a></code>, i.e. <code>options(AMR_keep_synonyms = TRUE)</code> or <code>options(AMR_keep_synonyms = FALSE)</code>.</p>
</td></tr>
<tr><td><code id="as.mo_+3A_reference_df">reference_df</code></td>
<td>
<p>a <a href="base.html#topic+data.frame">data.frame</a> to be used for extra reference when translating <code>x</code> to a valid <code><a href="#topic+mo">mo</a></code>. See <code><a href="#topic+set_mo_source">set_mo_source()</a></code> and <code><a href="#topic+get_mo_source">get_mo_source()</a></code> to automate the usage of your own codes (e.g. used in your analysis or organisation).</p>
</td></tr>
<tr><td><code id="as.mo_+3A_ignore_pattern">ignore_pattern</code></td>
<td>
<p>a Perl-compatible <a href="base.html#topic+regex">regular expression</a> (case-insensitive) of which all matches in <code>x</code> must return <code>NA</code>. This can be convenient to exclude known non-relevant input and can also be set with the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_ignore_pattern</a></code>, e.g. <code>options(AMR_ignore_pattern = "(not reported|contaminated flora)")</code>.</p>
</td></tr>
<tr><td><code id="as.mo_+3A_cleaning_regex">cleaning_regex</code></td>
<td>
<p>a Perl-compatible <a href="base.html#topic+regex">regular expression</a> (case-insensitive) to clean the input of <code>x</code>. Every matched part in <code>x</code> will be removed. At default, this is the outcome of <code><a href="#topic+mo_cleaning_regex">mo_cleaning_regex()</a></code>, which removes texts between brackets and texts such as &quot;species&quot; and &quot;serovar&quot;. The default can be set with the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_cleaning_regex</a></code>.</p>
</td></tr>
<tr><td><code id="as.mo_+3A_language">language</code></td>
<td>
<p>language to translate text like &quot;no growth&quot;, which defaults to the system language (see <code><a href="#topic+get_AMR_locale">get_AMR_locale()</a></code>)</p>
</td></tr>
<tr><td><code id="as.mo_+3A_info">info</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate if a progress bar should be printed if more than 25 items are to be coerced - the default is <code>TRUE</code> only in interactive mode</p>
</td></tr>
<tr><td><code id="as.mo_+3A_...">...</code></td>
<td>
<p>other arguments passed on to functions</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A microorganism (MO) code from this package (class: <code><a href="#topic+mo">mo</a></code>) is human readable and typically looks like these examples:
</p>
<div class="sourceCode"><pre>  Code               Full name
  ---------------    --------------------------------------
  B_KLBSL            Klebsiella
  B_KLBSL_PNMN       Klebsiella pneumoniae
  B_KLBSL_PNMN_RHNS  Klebsiella pneumoniae rhinoscleromatis
  |   |    |    |
  |   |    |    |
  |   |    |    \---&gt; subspecies, a 3-5 letter acronym
  |   |    \----&gt; species, a 3-6 letter acronym
  |   \----&gt; genus, a 4-8 letter acronym
  \----&gt; taxonomic kingdom: A (Archaea), AN (Animalia), B (Bacteria),
                            F (Fungi), PL (Plantae), P (Protozoa)
</pre></div>
<p>Values that cannot be coerced will be considered 'unknown' and will be returned as the MO code <code>UNKNOWN</code> with a warning.
</p>
<p>Use the <code><a href="#topic+mo_property">mo_*</a></code> functions to get properties based on the returned code, see <em>Examples</em>.
</p>
<p>The <code><a href="#topic+as.mo">as.mo()</a></code> function uses a novel <a href="#topic+mo_matching_score">matching score algorithm</a> (see <em>Matching Score for Microorganisms</em> below) to match input against the <a href="#topic+microorganisms">available microbial taxonomy</a> in this package. This will lead to the effect that e.g. <code>"E. coli"</code> (a microorganism highly prevalent in humans) will return the microbial ID of <em>Escherichia coli</em> and not <em>Entamoeba coli</em> (a microorganism less prevalent in humans), although the latter would alphabetically come first.
</p>
<p>With <code>Becker = TRUE</code>, the following 85 staphylococci will be converted to the <strong>coagulase-negative group</strong>: <em>S. argensis</em>, <em>S. arlettae</em>, <em>S. auricularis</em>, <em>S. borealis</em>, <em>S. caeli</em>, <em>S. caledonicus</em>, <em>S. canis</em>, <em>S. capitis</em>, <em>S. capitis capitis</em>, <em>S. capitis urealyticus</em>, <em>S. capitis ureolyticus</em>, <em>S. caprae</em>, <em>S. carnosus</em>, <em>S. carnosus carnosus</em>, <em>S. carnosus utilis</em>, <em>S. casei</em>, <em>S. caseolyticus</em>, <em>S. chromogenes</em>, <em>S. cohnii</em>, <em>S. cohnii cohnii</em>, <em>S. cohnii urealyticum</em>, <em>S. cohnii urealyticus</em>, <em>S. condimenti</em>, <em>S. croceilyticus</em>, <em>S. debuckii</em>, <em>S. devriesei</em>, <em>S. durrellii</em>, <em>S. edaphicus</em>, <em>S. epidermidis</em>, <em>S. equorum</em>, <em>S. equorum equorum</em>, <em>S. equorum linens</em>, <em>S. felis</em>, <em>S. fleurettii</em>, <em>S. gallinarum</em>, <em>S. haemolyticus</em>, <em>S. hominis</em>, <em>S. hominis hominis</em>, <em>S. hominis novobiosepticus</em>, <em>S. jettensis</em>, <em>S. kloosii</em>, <em>S. lentus</em>, <em>S. lloydii</em>, <em>S. lugdunensis</em>, <em>S. massiliensis</em>, <em>S. microti</em>, <em>S. muscae</em>, <em>S. nepalensis</em>, <em>S. pasteuri</em>, <em>S. petrasii</em>, <em>S. petrasii croceilyticus</em>, <em>S. petrasii jettensis</em>, <em>S. petrasii petrasii</em>, <em>S. petrasii pragensis</em>, <em>S. pettenkoferi</em>, <em>S. piscifermentans</em>, <em>S. pragensis</em>, <em>S. pseudoxylosus</em>, <em>S. pulvereri</em>, <em>S. ratti</em>, <em>S. rostri</em>, <em>S. saccharolyticus</em>, <em>S. saprophyticus</em>, <em>S. saprophyticus bovis</em>, <em>S. saprophyticus saprophyticus</em>, <em>S. schleiferi</em>, <em>S. schleiferi schleiferi</em>, <em>S. sciuri</em>, <em>S. sciuri carnaticus</em>, <em>S. sciuri lentus</em>, <em>S. sciuri rodentium</em>, <em>S. sciuri sciuri</em>, <em>S. simulans</em>, <em>S. stepanovicii</em>, <em>S. succinus</em>, <em>S. succinus casei</em>, <em>S. succinus succinus</em>, <em>S. taiwanensis</em>, <em>S. urealyticus</em>, <em>S. ureilyticus</em>, <em>S. veratri</em>, <em>S. vitulinus</em>, <em>S. vitulus</em>, <em>S. warneri</em>, and <em>S. xylosus</em>.<br /> The following 16 staphylococci will be converted to the <strong>coagulase-positive group</strong>: <em>S. agnetis</em>, <em>S. argenteus</em>, <em>S. coagulans</em>, <em>S. cornubiensis</em>, <em>S. delphini</em>, <em>S. hyicus</em>, <em>S. hyicus chromogenes</em>, <em>S. hyicus hyicus</em>, <em>S. intermedius</em>, <em>S. lutrae</em>, <em>S. pseudintermedius</em>, <em>S. roterodami</em>, <em>S. schleiferi coagulans</em>, <em>S. schweitzeri</em>, <em>S. simiae</em>, and <em>S. singaporensis</em>.
</p>
<p>With <code>Lancefield = TRUE</code>, the following streptococci will be converted to their corresponding Lancefield group: <em>S. agalactiae</em> (Group B), <em>S. anginosus anginosus</em> (Group F), <em>S. anginosus whileyi</em> (Group F), <em>S. anginosus</em> (Group F), <em>S. canis</em> (Group G), <em>S. dysgalactiae dysgalactiae</em> (Group C), <em>S. dysgalactiae equisimilis</em> (Group C), <em>S. dysgalactiae</em> (Group C), <em>S. equi equi</em> (Group C), <em>S. equi ruminatorum</em> (Group C), <em>S. equi zooepidemicus</em> (Group C), <em>S. equi</em> (Group C), <em>S. pyogenes</em> (Group A), <em>S. salivarius salivarius</em> (Group K), <em>S. salivarius thermophilus</em> (Group K), <em>S. salivarius</em> (Group K), and <em>S. sanguinis</em> (Group H).
</p>


<h4>Coping with Uncertain Results</h4>

<p>Results of non-exact taxonomic input are based on their <a href="#topic+mo_matching_score">matching score</a>. The lowest allowed score can be set with the <code>minimum_matching_score</code> argument. At default this will be determined based on the character length of the input, and the <a href="#topic+microorganisms">taxonomic kingdom</a> and <a href="#topic+mo_matching_score">human pathogenicity</a> of the taxonomic outcome. If values are matched with uncertainty, a message will be shown to suggest the user to evaluate the results with <code><a href="#topic+mo_uncertainties">mo_uncertainties()</a></code>, which returns a <a href="base.html#topic+data.frame">data.frame</a> with all specifications.
</p>
<p>To increase the quality of matching, the <code>cleaning_regex</code> argument can be used to clean the input (i.e., <code>x</code>). This must be a <a href="base.html#topic+regex">regular expression</a> that matches parts of the input that should be removed before the input is matched against the <a href="#topic+microorganisms">available microbial taxonomy</a>. It will be matched Perl-compatible and case-insensitive. The default value of <code>cleaning_regex</code> is the outcome of the helper function <code><a href="#topic+mo_cleaning_regex">mo_cleaning_regex()</a></code>.
</p>
<p>There are three helper functions that can be run after using the <code><a href="#topic+as.mo">as.mo()</a></code> function:
</p>

<ul>
<li><p> Use <code><a href="#topic+mo_uncertainties">mo_uncertainties()</a></code> to get a <a href="base.html#topic+data.frame">data.frame</a> that prints in a pretty format with all taxonomic names that were guessed. The output contains the matching score for all matches (see <em>Matching Score for Microorganisms</em> below).
</p>
</li>
<li><p> Use <code><a href="#topic+mo_failures">mo_failures()</a></code> to get a <a href="base.html#topic+character">character</a> <a href="base.html#topic+vector">vector</a> with all values that could not be coerced to a valid value.
</p>
</li>
<li><p> Use <code><a href="#topic+mo_renamed">mo_renamed()</a></code> to get a <a href="base.html#topic+data.frame">data.frame</a> with all values that could be coerced based on old, previously accepted taxonomic names.
</p>
</li></ul>




<h4>Microbial Prevalence of Pathogens in Humans</h4>

<p>The coercion rules consider the prevalence of microorganisms in humans, which is available as the <code>prevalence</code> column in the <a href="#topic+microorganisms">microorganisms</a> data set. The grouping into human pathogenic prevalence is explained in the section <em>Matching Score for Microorganisms</em> below.
</p>



<h3>Value</h3>

<p>A <a href="base.html#topic+character">character</a> <a href="base.html#topic+vector">vector</a> with additional class <code><a href="#topic+mo">mo</a></code>
</p>


<h3>Source</h3>


<ol>
<li><p> Berends MS <em>et al.</em> (2022). <strong>AMR: An R Package for Working with Antimicrobial Resistance Data</strong>. <em>Journal of Statistical Software</em>, 104(3), 1-31; <a href="https://doi.org/10.18637/jss.v104.i03">doi:10.18637/jss.v104.i03</a>
</p>
</li>
<li><p> Becker K <em>et al.</em> (2014). <strong>Coagulase-Negative Staphylococci.</strong> <em>Clin Microbiol Rev.</em> 27(4): 870-926; <a href="https://doi.org/10.1128/CMR.00109-13">doi:10.1128/CMR.00109-13</a>
</p>
</li>
<li><p> Becker K <em>et al.</em> (2019). <strong>Implications of identifying the recently defined members of the <em>S. aureus</em> complex, <em>S. argenteus</em> and <em>S. schweitzeri</em>: A position paper of members of the ESCMID Study Group for staphylococci and Staphylococcal Diseases (ESGS).</strong> <em>Clin Microbiol Infect</em>; <a href="https://doi.org/10.1016/j.cmi.2019.02.028">doi:10.1016/j.cmi.2019.02.028</a>
</p>
</li>
<li><p> Becker K <em>et al.</em> (2020). <strong>Emergence of coagulase-negative staphylococci.</strong> <em>Expert Rev Anti Infect Ther.</em> 18(4):349-366; <a href="https://doi.org/10.1080/14787210.2020.1730813">doi:10.1080/14787210.2020.1730813</a>
</p>
</li>
<li><p> Lancefield RC (1933). <strong>A serological differentiation of human and other groups of hemolytic streptococci.</strong> <em>J Exp Med.</em> 57(4): 571-95; <a href="https://doi.org/10.1084/jem.57.4.571">doi:10.1084/jem.57.4.571</a>
</p>
</li>
<li><p> Berends MS <em>et al.</em> (2022). <strong>Trends in Occurrence and Phenotypic Resistance of Coagulase-Negative Staphylococci (CoNS) Found in Human Blood in the Northern Netherlands between 2013 and 2019/</strong> <em>Micro.rganisms</em> 10(9), 1801; <a href="https://doi.org/10.3390/microorganisms10091801">doi:10.3390/microorganisms10091801</a>
</p>
</li>
<li><p> Parte, AC <em>et al.</em> (2020). <strong>List of Prokaryotic names with Standing in Nomenclature (LPSN) moves to the DSMZ.</strong> International Journal of Systematic and Evolutionary Microbiology, 70, 5607-5612; <a href="https://doi.org/10.1099/ijsem.0.004332">doi:10.1099/ijsem.0.004332</a>. Accessed from <a href="https://lpsn.dsmz.de">https://lpsn.dsmz.de</a> on December 11th, 2022.
</p>
</li>
<li><p> GBIF Secretariat (2022). GBIF Backbone Taxonomy. Checklist dataset <a href="https://doi.org/10.15468/39omei">doi:10.15468/39omei</a>. Accessed from <a href="https://www.gbif.org">https://www.gbif.org</a> on December 11th, 2022.
</p>
</li>
<li><p> Reimer, LC <em>et al.</em> (2022). <strong><em>BacDive</em> in 2022: the knowledge base for standardized bacterial and archaeal data.</strong> Nucleic Acids Res., 50(D1):D741-D74; <a href="https://doi.org/10.1093/nar/gkab961">doi:10.1093/nar/gkab961</a>. Accessed from <a href="https://bacdive.dsmz.de">https://bacdive.dsmz.de</a> on May 12th, 2023.
</p>
</li>
<li><p> Public Health Information Network Vocabulary Access and Distribution System (PHIN VADS). US Edition of SNOMED CT from 1 September 2020. Value Set Name 'Microorganism', OID 2.16.840.1.114222.4.11.1009 (v12). URL: <a href="https://phinvads.cdc.gov">https://phinvads.cdc.gov</a>
</p>
</li>
<li><p> Bartlett A <em>et al.</em> (2022). <strong>A comprehensive list of bacterial pathogens infecting humans</strong> <em>Microbiology</em> 168:001269; <a href="https://doi.org/10.1099/mic.0.001269">doi:10.1099/mic.0.001269</a>
</p>
</li></ol>



<h3>Matching Score for Microorganisms</h3>

<p>With ambiguous user input in <code><a href="#topic+as.mo">as.mo()</a></code> and all the <code><a href="#topic+mo_property">mo_*</a></code> functions, the returned results are chosen based on their matching score using <code><a href="#topic+mo_matching_score">mo_matching_score()</a></code>. This matching score <code class="reqn">m</code>, is calculated as:
</p>

<p><img src="../help/figures/mo_matching_score.png" width="300" alt="mo matching score" />
</p>
<p>where:
</p>

<ul>
<li> <p><code class="reqn">x</code> is the user input;
</p>
</li>
<li> <p><code class="reqn">n</code> is a taxonomic name (genus, species, and subspecies);
</p>
</li>
<li> <p><code class="reqn">l_n</code> is the length of <code class="reqn">n</code>;
</p>
</li>
<li> <p><code class="reqn">lev</code> is the <a href="https://en.wikipedia.org/wiki/Levenshtein_distance">Levenshtein distance function</a> (counting any insertion as 1, and any deletion or substitution as 2) that is needed to change <code class="reqn">x</code> into <code class="reqn">n</code>;
</p>
</li>
<li> <p><code class="reqn">p_n</code> is the human pathogenic prevalence group of <code class="reqn">n</code>, as described below;
</p>
</li>
<li> <p><code class="reqn">k_n</code> is the taxonomic kingdom of <code class="reqn">n</code>, set as Bacteria = 1, Fungi = 1.25, Protozoa = 1.5, Archaea = 2, others = 3.
</p>
</li></ul>

<p>The grouping into human pathogenic prevalence <code class="reqn">p</code> is based on recent work from Bartlett <em>et al.</em> (2022, <a href="https://doi.org/10.1099/mic.0.001269">doi:10.1099/mic.0.001269</a>) who extensively studied medical-scientific literature to categorise all bacterial species into these groups:
</p>

<ul>
<li> <p><strong>Established</strong>, if a taxonomic species has infected at least three persons in three or more references. These records have <code>prevalence = 1.0</code> in the <a href="#topic+microorganisms">microorganisms</a> data set;
</p>
</li>
<li> <p><strong>Putative</strong>, if a taxonomic species has fewer than three known cases. These records have <code>prevalence = 1.25</code> in the <a href="#topic+microorganisms">microorganisms</a> data set.
</p>
</li></ul>

<p>Furthermore,
</p>

<ul>
<li><p> Any genus present in the <strong>established</strong> list also has <code>prevalence = 1.0</code> in the <a href="#topic+microorganisms">microorganisms</a> data set;
</p>
</li>
<li><p> Any other genus present in the <strong>putative</strong> list has <code>prevalence = 1.25</code> in the <a href="#topic+microorganisms">microorganisms</a> data set;
</p>
</li>
<li><p> Any other species or subspecies of which the genus is present in the two aforementioned groups, has <code>prevalence = 1.5</code> in the <a href="#topic+microorganisms">microorganisms</a> data set;
</p>
</li>
<li><p> Any <em>non-bacterial</em> genus, species or subspecies of which the genus is present in the following list, has <code>prevalence = 1.25</code> in the <a href="#topic+microorganisms">microorganisms</a> data set: <em>Absidia</em>, <em>Acanthamoeba</em>, <em>Acremonium</em>, <em>Aedes</em>, <em>Alternaria</em>, <em>Amoeba</em>, <em>Ancylostoma</em>, <em>Angiostrongylus</em>, <em>Anisakis</em>, <em>Anopheles</em>, <em>Apophysomyces</em>, <em>Aspergillus</em>, <em>Aureobasidium</em>, <em>Basidiobolus</em>, <em>Beauveria</em>, <em>Blastocystis</em>, <em>Blastomyces</em>, <em>Candida</em>, <em>Capillaria</em>, <em>Chaetomium</em>, <em>Chrysonilia</em>, <em>Cladophialophora</em>, <em>Cladosporium</em>, <em>Conidiobolus</em>, <em>Contracaecum</em>, <em>Cordylobia</em>, <em>Cryptococcus</em>, <em>Curvularia</em>, <em>Demodex</em>, <em>Dermatobia</em>, <em>Dientamoeba</em>, <em>Diphyllobothrium</em>, <em>Dirofilaria</em>, <em>Echinostoma</em>, <em>Entamoeba</em>, <em>Enterobius</em>, <em>Exophiala</em>, <em>Exserohilum</em>, <em>Fasciola</em>, <em>Fonsecaea</em>, <em>Fusarium</em>, <em>Giardia</em>, <em>Haloarcula</em>, <em>Halobacterium</em>, <em>Halococcus</em>, <em>Hendersonula</em>, <em>Heterophyes</em>, <em>Histomonas</em>, <em>Histoplasma</em>, <em>Hymenolepis</em>, <em>Hypomyces</em>, <em>Hysterothylacium</em>, <em>Leishmania</em>, <em>Malassezia</em>, <em>Malbranchea</em>, <em>Metagonimus</em>, <em>Meyerozyma</em>, <em>Microsporidium</em>, <em>Microsporum</em>, <em>Mortierella</em>, <em>Mucor</em>, <em>Mycocentrospora</em>, <em>Necator</em>, <em>Nectria</em>, <em>Ochroconis</em>, <em>Oesophagostomum</em>, <em>Oidiodendron</em>, <em>Opisthorchis</em>, <em>Pediculus</em>, <em>Penicillium</em>, <em>Phlebotomus</em>, <em>Phoma</em>, <em>Pichia</em>, <em>Piedraia</em>, <em>Pithomyces</em>, <em>Pityrosporum</em>, <em>Pneumocystis</em>, <em>Pseudallescheria</em>, <em>Pseudoterranova</em>, <em>Pulex</em>, <em>Rhizomucor</em>, <em>Rhizopus</em>, <em>Rhodotorula</em>, <em>Saccharomyces</em>, <em>Sarcoptes</em>, <em>Scolecobasidium</em>, <em>Scopulariopsis</em>, <em>Scytalidium</em>, <em>Spirometra</em>, <em>Sporobolomyces</em>, <em>Stachybotrys</em>, <em>Strongyloides</em>, <em>Syngamus</em>, <em>Taenia</em>, <em>Talaromyces</em>, <em>Toxocara</em>, <em>Trichinella</em>, <em>Trichobilharzia</em>, <em>Trichoderma</em>, <em>Trichomonas</em>, <em>Trichophyton</em>, <em>Trichosporon</em>, <em>Trichostrongylus</em>, <em>Trichuris</em>, <em>Tritirachium</em>, <em>Trombicula</em>, <em>Trypanosoma</em>, <em>Tunga</em>, or <em>Wuchereria</em>;
</p>
</li>
<li><p> All other records have <code>prevalence = 2.0</code> in the <a href="#topic+microorganisms">microorganisms</a> data set.
</p>
</li></ul>

<p>When calculating the matching score, all characters in <code class="reqn">x</code> and <code class="reqn">n</code> are ignored that are other than A-Z, a-z, 0-9, spaces and parentheses.
</p>
<p>All matches are sorted descending on their matching score and for all user input values, the top match will be returned. This will lead to the effect that e.g., <code>"E. coli"</code> will return the microbial ID of <em>Escherichia coli</em> (<code class="reqn">m = 0.688</code>, a highly prevalent microorganism found in humans) and not <em>Entamoeba coli</em> (<code class="reqn">m = 0.381</code>, a less prevalent microorganism in humans), although the latter would alphabetically come first.
</p>


<h3>Reference Data Publicly Available</h3>

<p>All data sets in this <code>AMR</code> package (about microorganisms, antibiotics, SIR interpretation, EUCAST rules, etc.) are publicly and freely available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. We also provide tab-separated plain text files that are machine-readable and suitable for input in any software program, such as laboratory information systems. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>.
</p>


<h3>See Also</h3>

<p><a href="#topic+microorganisms">microorganisms</a> for the <a href="base.html#topic+data.frame">data.frame</a> that is being used to determine ID's.
</p>
<p>The <code><a href="#topic+mo_property">mo_*</a></code> functions (such as <code><a href="#topic+mo_genus">mo_genus()</a></code>, <code><a href="#topic+mo_gramstain">mo_gramstain()</a></code>) to get properties based on the returned code.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# These examples all return "B_STPHY_AURS", the ID of S. aureus:
as.mo(c(
  "sau", # WHONET code
  "stau",
  "STAU",
  "staaur",
  "S. aureus",
  "S aureus",
  "Sthafilokkockus aureus", # handles incorrect spelling
  "Staphylococcus aureus (MRSA)",
  "MRSA", # Methicillin Resistant S. aureus
  "VISA", # Vancomycin Intermediate S. aureus
  "VRSA", # Vancomycin Resistant S. aureus
  115329001 # SNOMED CT code
))

# Dyslexia is no problem - these all work:
as.mo(c(
  "Ureaplasma urealyticum",
  "Ureaplasma urealyticus",
  "Ureaplasmium urealytica",
  "Ureaplazma urealitycium"
))

# input will get cleaned up with the input given in the `cleaning_regex` argument,
# which defaults to `mo_cleaning_regex()`:
cat(mo_cleaning_regex(), "\n")

as.mo("Streptococcus group A")

as.mo("S. epidermidis") # will remain species: B_STPHY_EPDR
as.mo("S. epidermidis", Becker = TRUE) # will not remain species: B_STPHY_CONS

as.mo("S. pyogenes") # will remain species: B_STRPT_PYGN
as.mo("S. pyogenes", Lancefield = TRUE) # will not remain species: B_STRPT_GRPA

# All mo_* functions use as.mo() internally too (see ?mo_property):
mo_genus("E. coli")
mo_gramstain("ESCO")
mo_is_intrinsic_resistant("ESCCOL", ab = "vanco")

</code></pre>

<hr>
<h2 id='as.sir'>Translate MIC and Disk Diffusion to SIR, or Clean Existing SIR Data</h2><span id='topic+as.sir'></span><span id='topic+sir'></span><span id='topic+NA_sir_'></span><span id='topic+is.sir'></span><span id='topic+is_sir_eligible'></span><span id='topic+as.sir.mic'></span><span id='topic+as.sir.disk'></span><span id='topic+as.sir.data.frame'></span><span id='topic+sir_interpretation_history'></span>

<h3>Description</h3>

<p>Interpret minimum inhibitory concentration (MIC) values and disk diffusion diameters according to EUCAST or CLSI, or clean up existing SIR values. This transforms the input to a new class <code><a href="#topic+sir">sir</a></code>, which is an ordered <a href="base.html#topic+factor">factor</a> with levels <code style="white-space: pre;">&#8288;S &lt; I &lt; R&#8288;</code>.
</p>
<p>Currently available <strong>breakpoint guidelines</strong> are EUCAST 2011-2023 and CLSI 2011-2023, and available <strong>breakpoint types</strong> are &quot;ECOFF&quot;, &quot;animal&quot;, and &quot;human&quot;.
</p>
<p>All breakpoints used for interpretation are publicly available in the <a href="#topic+clinical_breakpoints">clinical_breakpoints</a> data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.sir(x, ...)

NA_sir_

is.sir(x)

is_sir_eligible(x, threshold = 0.05)

## S3 method for class 'mic'
as.sir(
  x,
  mo = NULL,
  ab = deparse(substitute(x)),
  guideline = getOption("AMR_guideline", "EUCAST"),
  uti = NULL,
  conserve_capped_values = FALSE,
  add_intrinsic_resistance = FALSE,
  reference_data = AMR::clinical_breakpoints,
  include_screening = getOption("AMR_include_screening", FALSE),
  include_PKPD = getOption("AMR_include_PKPD", TRUE),
  breakpoint_type = getOption("AMR_breakpoint_type", "human"),
  ...
)

## S3 method for class 'disk'
as.sir(
  x,
  mo = NULL,
  ab = deparse(substitute(x)),
  guideline = getOption("AMR_guideline", "EUCAST"),
  uti = NULL,
  add_intrinsic_resistance = FALSE,
  reference_data = AMR::clinical_breakpoints,
  include_screening = getOption("AMR_include_screening", FALSE),
  include_PKPD = getOption("AMR_include_PKPD", TRUE),
  breakpoint_type = getOption("AMR_breakpoint_type", "human"),
  ...
)

## S3 method for class 'data.frame'
as.sir(
  x,
  ...,
  col_mo = NULL,
  guideline = getOption("AMR_guideline", "EUCAST"),
  uti = NULL,
  conserve_capped_values = FALSE,
  add_intrinsic_resistance = FALSE,
  reference_data = AMR::clinical_breakpoints,
  include_screening = getOption("AMR_include_screening", FALSE),
  include_PKPD = getOption("AMR_include_PKPD", TRUE),
  breakpoint_type = getOption("AMR_breakpoint_type", "human")
)

sir_interpretation_history(clean = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.sir_+3A_x">x</code></td>
<td>
<p>vector of values (for class <code><a href="#topic+mic">mic</a></code>: MIC values in mg/L, for class <code><a href="#topic+disk">disk</a></code>: a disk diffusion radius in millimetres)</p>
</td></tr>
<tr><td><code id="as.sir_+3A_...">...</code></td>
<td>
<p>for using on a <a href="base.html#topic+data.frame">data.frame</a>: names of columns to apply <code><a href="#topic+as.sir">as.sir()</a></code> on (supports tidy selection such as <code>column1:column4</code>). Otherwise: arguments passed on to methods.</p>
</td></tr>
<tr><td><code id="as.sir_+3A_threshold">threshold</code></td>
<td>
<p>maximum fraction of invalid antimicrobial interpretations of <code>x</code>, see <em>Examples</em></p>
</td></tr>
<tr><td><code id="as.sir_+3A_mo">mo</code></td>
<td>
<p>any (vector of) text that can be coerced to valid microorganism codes with <code><a href="#topic+as.mo">as.mo()</a></code>, can be left empty to determine it automatically</p>
</td></tr>
<tr><td><code id="as.sir_+3A_ab">ab</code></td>
<td>
<p>any (vector of) text that can be coerced to a valid antimicrobial drug code with <code><a href="#topic+as.ab">as.ab()</a></code></p>
</td></tr>
<tr><td><code id="as.sir_+3A_guideline">guideline</code></td>
<td>
<p>defaults to EUCAST 2023 (the latest implemented EUCAST guideline in the <a href="#topic+clinical_breakpoints">clinical_breakpoints</a> data set), but can be set with the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_guideline</a></code>. Currently supports EUCAST (2011-2023) and CLSI (2011-2023), see <em>Details</em>.</p>
</td></tr>
<tr><td><code id="as.sir_+3A_uti">uti</code></td>
<td>
<p>(Urinary Tract Infection) A vector with <a href="base.html#topic+logical">logical</a>s (<code>TRUE</code> or <code>FALSE</code>) to specify whether a UTI specific interpretation from the guideline should be chosen. For using <code><a href="#topic+as.sir">as.sir()</a></code> on a <a href="base.html#topic+data.frame">data.frame</a>, this can also be a column containing <a href="base.html#topic+logical">logical</a>s or when left blank, the data set will be searched for a column 'specimen', and rows within this column containing 'urin' (such as 'urine', 'urina') will be regarded isolates from a UTI. See <em>Examples</em>.</p>
</td></tr>
<tr><td><code id="as.sir_+3A_conserve_capped_values">conserve_capped_values</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate that MIC values starting with <code>"&gt;"</code> (but not <code>"&gt;="</code>) must always return &quot;R&quot; , and that MIC values starting with <code>"&lt;"</code> (but not <code>"&lt;="</code>) must always return &quot;S&quot;</p>
</td></tr>
<tr><td><code id="as.sir_+3A_add_intrinsic_resistance">add_intrinsic_resistance</code></td>
<td>
<p><em>(only useful when using a EUCAST guideline)</em> a <a href="base.html#topic+logical">logical</a> to indicate whether intrinsic antibiotic resistance must also be considered for applicable bug-drug combinations, meaning that e.g. ampicillin will always return &quot;R&quot; in <em>Klebsiella</em> species. Determination is based on the <a href="#topic+intrinsic_resistant">intrinsic_resistant</a> data set, that itself is based on <a href="https://www.eucast.org/expert_rules_and_expected_phenotypes">'EUCAST Expert Rules' and 'EUCAST Intrinsic Resistance and Unusual Phenotypes' v3.3</a> (2021).</p>
</td></tr>
<tr><td><code id="as.sir_+3A_reference_data">reference_data</code></td>
<td>
<p>a <a href="base.html#topic+data.frame">data.frame</a> to be used for interpretation, which defaults to the <a href="#topic+clinical_breakpoints">clinical_breakpoints</a> data set. Changing this argument allows for using own interpretation guidelines. This argument must contain a data set that is equal in structure to the <a href="#topic+clinical_breakpoints">clinical_breakpoints</a> data set (same column names and column types). Please note that the <code>guideline</code> argument will be ignored when <code>reference_data</code> is manually set.</p>
</td></tr>
<tr><td><code id="as.sir_+3A_include_screening">include_screening</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate that clinical breakpoints for screening are allowed - the default is <code>FALSE</code>. Can also be set with the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_include_screening</a></code>.</p>
</td></tr>
<tr><td><code id="as.sir_+3A_include_pkpd">include_PKPD</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate that PK/PD clinical breakpoints must be applied as a last resort - the default is <code>TRUE</code>. Can also be set with the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_include_PKPD</a></code>.</p>
</td></tr>
<tr><td><code id="as.sir_+3A_breakpoint_type">breakpoint_type</code></td>
<td>
<p>the type of breakpoints to use, either &quot;ECOFF&quot;, &quot;animal&quot;, or &quot;human&quot;. ECOFF stands for Epidemiological Cut-Off values. The default is <code>"human"</code>, which can also be set with the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_breakpoint_type</a></code>.</p>
</td></tr>
<tr><td><code id="as.sir_+3A_col_mo">col_mo</code></td>
<td>
<p>column name of the names or codes of the microorganisms (see <code><a href="#topic+as.mo">as.mo()</a></code>) - the default is the first column of class <code><a href="#topic+mo">mo</a></code>. Values will be coerced using <code><a href="#topic+as.mo">as.mo()</a></code>.</p>
</td></tr>
<tr><td><code id="as.sir_+3A_clean">clean</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether previously stored results should be forgotten after returning the 'logbook' with results</p>
</td></tr>
</table>


<h3>Format</h3>

<p>An object of class <code>sir</code> (inherits from <code>ordered</code>, <code>factor</code>) of length 1.
</p>


<h3>Details</h3>

<p><em>Note: The clinical breakpoints in this package were validated through and imported from <a href="https://whonet.org">WHONET</a> and the public use of this <code>AMR</code> package has been endorsed by CLSI and EUCAST, please see <a href="#topic+clinical_breakpoints">clinical_breakpoints</a> for more information.</em>
</p>


<h4>How it Works</h4>

<p>The <code><a href="#topic+as.sir">as.sir()</a></code> function works in four ways:
</p>

<ol>
<li><p> For <strong>cleaning raw / untransformed data</strong>. The data will be cleaned to only contain values S, I and R and will try its best to determine this with some intelligence. For example, mixed values with SIR interpretations and MIC values such as <code>"&lt;0.25; S"</code> will be coerced to <code>"S"</code>. Combined interpretations for multiple test methods (as seen in laboratory records) such as <code>"S; S"</code> will be coerced to <code>"S"</code>, but a value like <code>"S; I"</code> will return <code>NA</code> with a warning that the input is unclear.
</p>
</li>
<li><p> For <strong>interpreting minimum inhibitory concentration (MIC) values</strong> according to EUCAST or CLSI. You must clean your MIC values first using <code><a href="#topic+as.mic">as.mic()</a></code>, that also gives your columns the new data class <code><a href="#topic+mic">mic</a></code>. Also, be sure to have a column with microorganism names or codes. It will be found automatically, but can be set manually using the <code>mo</code> argument.
</p>

<ul>
<li><p> Using <code>dplyr</code>, SIR interpretation can be done very easily with either:
</p>
<div class="sourceCode"><pre>your_data %&gt;% mutate_if(is.mic, as.sir)
your_data %&gt;% mutate(across(where(is.mic), as.sir))
</pre></div>
</li>
<li><p> Operators like &quot;&lt;=&quot; will be stripped before interpretation. When using <code>conserve_capped_values = TRUE</code>, an MIC value of e.g. &quot;&gt;2&quot; will always return &quot;R&quot;, even if the breakpoint according to the chosen guideline is &quot;&gt;=4&quot;. This is to prevent that capped values from raw laboratory data would not be treated conservatively. The default behaviour (<code>conserve_capped_values = FALSE</code>) considers &quot;&gt;2&quot; to be lower than &quot;&gt;=4&quot; and might in this case return &quot;S&quot; or &quot;I&quot;.
</p>
</li></ul>

</li>
<li><p> For <strong>interpreting disk diffusion diameters</strong> according to EUCAST or CLSI. You must clean your disk zones first using <code><a href="#topic+as.disk">as.disk()</a></code>, that also gives your columns the new data class <code><a href="#topic+disk">disk</a></code>. Also, be sure to have a column with microorganism names or codes. It will be found automatically, but can be set manually using the <code>mo</code> argument.
</p>

<ul>
<li><p> Using <code>dplyr</code>, SIR interpretation can be done very easily with either:
</p>
<div class="sourceCode"><pre>your_data %&gt;% mutate_if(is.disk, as.sir)
your_data %&gt;% mutate(across(where(is.disk), as.sir))
</pre></div>
</li></ul>

</li>
<li><p> For <strong>interpreting a complete data set</strong>, with automatic determination of MIC values, disk diffusion diameters, microorganism names or codes, and antimicrobial test results. This is done very simply by running <code>as.sir(your_data)</code>.
</p>
</li></ol>

<p><strong>For points 2, 3 and 4: Use <code><a href="#topic+sir_interpretation_history">sir_interpretation_history()</a></code></strong> to retrieve a <a href="base.html#topic+data.frame">data.frame</a> (or <a href="tibble.html#topic+tibble">tibble</a> if the <code>tibble</code> package is installed) with all results of the last <code><a href="#topic+as.sir">as.sir()</a></code> call.
</p>



<h4>Supported Guidelines</h4>

<p>For interpreting MIC values as well as disk diffusion diameters, currently implemented guidelines are EUCAST (2011-2023) and CLSI (2011-2023).
</p>
<p>Thus, the <code>guideline</code> argument must be set to e.g., <code>"EUCAST 2023"</code> or <code>"CLSI 2023"</code>. By simply using <code>"EUCAST"</code> (the default) or <code>"CLSI"</code> as input, the latest included version of that guideline will automatically be selected. You can set your own data set using the <code>reference_data</code> argument. The <code>guideline</code> argument will then be ignored.
</p>
<p>You can set the default guideline with the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_guideline</a></code> (e.g. in your <code>.Rprofile</code> file), such as:
</p>
<div class="sourceCode"><pre>  options(AMR_guideline = "CLSI")
  options(AMR_guideline = "CLSI 2018")
  options(AMR_guideline = "EUCAST 2020")
  # or to reset:
  options(AMR_guideline = NULL)
</pre></div>



<h4>After Interpretation</h4>

<p>After using <code><a href="#topic+as.sir">as.sir()</a></code>, you can use the <code><a href="#topic+eucast_rules">eucast_rules()</a></code> defined by EUCAST to (1) apply inferred susceptibility and resistance based on results of other antimicrobials and (2) apply intrinsic resistance based on taxonomic properties of a microorganism.
</p>



<h4>Machine-Readable Clinical Breakpoints</h4>

<p>The repository of this package <a href="https://github.com/msberends/AMR/blob/main/data-raw/clinical_breakpoints.txt">contains a machine-readable version</a> of all guidelines. This is a CSV file consisting of 29 747 rows and 12 columns. This file is machine-readable, since it contains one row for every unique combination of the test method (MIC or disk diffusion), the antimicrobial drug and the microorganism. <strong>This allows for easy implementation of these rules in laboratory information systems (LIS)</strong>. Note that it only contains interpretation guidelines for humans - interpretation guidelines from CLSI for animals were removed.
</p>



<h4>Other</h4>

<p>The function <code><a href="#topic+is.sir">is.sir()</a></code> detects if the input contains class <code>sir</code>. If the input is a <a href="base.html#topic+data.frame">data.frame</a>, it iterates over all columns and returns a <a href="base.html#topic+logical">logical</a> vector.
</p>
<p>The function <code><a href="#topic+is_sir_eligible">is_sir_eligible()</a></code> returns <code>TRUE</code> when a columns contains at most 5% invalid antimicrobial interpretations (not S and/or I and/or R), and <code>FALSE</code> otherwise. The threshold of 5% can be set with the <code>threshold</code> argument. If the input is a <a href="base.html#topic+data.frame">data.frame</a>, it iterates over all columns and returns a <a href="base.html#topic+logical">logical</a> vector.
</p>

<p><code>NA_sir_</code> is a missing value of the new <code>sir</code> class, analogous to e.g. base <span class="rlang"><b>R</b></span>'s <code><a href="base.html#topic+NA">NA_character_</a></code>.
</p>


<h3>Value</h3>

<p>Ordered <a href="base.html#topic+factor">factor</a> with new class <code>sir</code>
</p>


<h3>Interpretation of SIR</h3>

<p>In 2019, the European Committee on Antimicrobial Susceptibility Testing (EUCAST) has decided to change the definitions of susceptibility testing categories S, I, and R as shown below (<a href="https://www.eucast.org/newsiandr">https://www.eucast.org/newsiandr</a>):
</p>

<ul>
<li> <p><strong>S - Susceptible, standard dosing regimen</strong><br />
A microorganism is categorised as &quot;Susceptible, standard dosing regimen&quot;, when there is a high likelihood of therapeutic success using a standard dosing regimen of the agent.
</p>
</li>
<li> <p><strong>I - Susceptible, increased exposure</strong> <em><br />
A microorganism is categorised as &quot;Susceptible, Increased exposure</em>&quot; when there is a high likelihood of therapeutic success because exposure to the agent is increased by adjusting the dosing regimen or by its concentration at the site of infection.
</p>
</li>
<li> <p><strong>R = Resistant</strong><br />
A microorganism is categorised as &quot;Resistant&quot; when there is a high likelihood of therapeutic failure even when there is increased exposure.
</p>

<ul>
<li> <p><em>Exposure</em> is a function of how the mode of administration, dose, dosing interval, infusion time, as well as distribution and excretion of the antimicrobial agent will influence the infecting organism at the site of infection.
</p>
</li></ul>

</li></ul>

<p>This AMR package honours this insight. Use <code><a href="#topic+susceptibility">susceptibility()</a></code> (equal to <code><a href="#topic+proportion_SI">proportion_SI()</a></code>) to determine antimicrobial susceptibility and <code><a href="#topic+count_susceptible">count_susceptible()</a></code> (equal to <code><a href="#topic+count_SI">count_SI()</a></code>) to count susceptible isolates.
</p>


<h3>Reference Data Publicly Available</h3>

<p>All data sets in this <code>AMR</code> package (about microorganisms, antibiotics, SIR interpretation, EUCAST rules, etc.) are publicly and freely available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. We also provide tab-separated plain text files that are machine-readable and suitable for input in any software program, such as laboratory information systems. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>.
</p>


<h3>Source</h3>

<p>For interpretations of minimum inhibitory concentration (MIC) values and disk diffusion diameters:
</p>

<ul>
<li> <p><strong>M39 Analysis and Presentation of Cumulative Antimicrobial Susceptibility Test Data</strong>, 2011-2023, <em>Clinical and Laboratory Standards Institute</em> (CLSI). <a href="https://clsi.org/standards/products/microbiology/documents/m39/">https://clsi.org/standards/products/microbiology/documents/m39/</a>.
</p>
</li>
<li> <p><strong>M100 Performance Standard for Antimicrobial Susceptibility Testing</strong>, 2011-2023, <em>Clinical and Laboratory Standards Institute</em> (CLSI). <a href="https://clsi.org/standards/products/microbiology/documents/m100/">https://clsi.org/standards/products/microbiology/documents/m100/</a>.
</p>
</li>
<li> <p><strong>Breakpoint tables for interpretation of MICs and zone diameters</strong>, 2011-2023, <em>European Committee on Antimicrobial Susceptibility Testing</em> (EUCAST). <a href="https://www.eucast.org/clinical_breakpoints">https://www.eucast.org/clinical_breakpoints</a>.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+as.mic">as.mic()</a></code>, <code><a href="#topic+as.disk">as.disk()</a></code>, <code><a href="#topic+as.mo">as.mo()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example_isolates
summary(example_isolates) # see all SIR results at a glance

# For INTERPRETING disk diffusion and MIC values -----------------------

# a whole data set, even with combined MIC values and disk zones
df &lt;- data.frame(
  microorganism = "Escherichia coli",
  AMP = as.mic(8),
  CIP = as.mic(0.256),
  GEN = as.disk(18),
  TOB = as.disk(16),
  ERY = "R"
)
as.sir(df)

# return a 'logbook' about the results:
sir_interpretation_history()

# for single values
as.sir(
  x = as.mic(2),
  mo = as.mo("S. pneumoniae"),
  ab = "AMP",
  guideline = "EUCAST"
)

as.sir(
  x = as.disk(18),
  mo = "Strep pneu", # `mo` will be coerced with as.mo()
  ab = "ampicillin", # and `ab` with as.ab()
  guideline = "EUCAST"
)


# the dplyr way
if (require("dplyr")) {
  df %&gt;% mutate_if(is.mic, as.sir)
  df %&gt;% mutate_if(function(x) is.mic(x) | is.disk(x), as.sir)
  df %&gt;% mutate(across(where(is.mic), as.sir))
  df %&gt;% mutate_at(vars(AMP:TOB), as.sir)
  df %&gt;% mutate(across(AMP:TOB, as.sir))

  df %&gt;%
    mutate_at(vars(AMP:TOB), as.sir, mo = .$microorganism)

  # to include information about urinary tract infections (UTI)
  data.frame(
    mo = "E. coli",
    NIT = c("&lt;= 2", 32),
    from_the_bladder = c(TRUE, FALSE)
  ) %&gt;%
    as.sir(uti = "from_the_bladder")

  data.frame(
    mo = "E. coli",
    NIT = c("&lt;= 2", 32),
    specimen = c("urine", "blood")
  ) %&gt;%
    as.sir() # automatically determines urine isolates

  df %&gt;%
    mutate_at(vars(AMP:TOB), as.sir, mo = "E. coli", uti = TRUE)
}

# For CLEANING existing SIR values ------------------------------------

as.sir(c("S", "I", "R", "A", "B", "C"))
as.sir("&lt;= 0.002; S") # will return "S"
sir_data &lt;- as.sir(c(rep("S", 474), rep("I", 36), rep("R", 370)))
is.sir(sir_data)
plot(sir_data) # for percentages
barplot(sir_data) # for frequencies

# the dplyr way
if (require("dplyr")) {
  example_isolates %&gt;%
    mutate_at(vars(PEN:RIF), as.sir)
  # same:
  example_isolates %&gt;%
    as.sir(PEN:RIF)

  # fastest way to transform all columns with already valid AMR results to class `sir`:
  example_isolates %&gt;%
    mutate_if(is_sir_eligible, as.sir)

  # since dplyr 1.0.0, this can also be:
  # example_isolates %&gt;%
  #   mutate(across(where(is_sir_eligible), as.sir))
}

</code></pre>

<hr>
<h2 id='atc_online_property'>Get ATC Properties from WHOCC Website</h2><span id='topic+atc_online_property'></span><span id='topic+atc_online_groups'></span><span id='topic+atc_online_ddd'></span><span id='topic+atc_online_ddd_units'></span>

<h3>Description</h3>

<p>Gets data from the WHOCC website to determine properties of an Anatomical Therapeutic Chemical (ATC) (e.g. an antibiotic), such as the name, defined daily dose (DDD) or standard unit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>atc_online_property(
  atc_code,
  property,
  administration = "O",
  url = "https://www.whocc.no/atc_ddd_index/?code=%s&amp;showdescription=no",
  url_vet = "https://www.whocc.no/atcvet/atcvet_index/?code=%s&amp;showdescription=no"
)

atc_online_groups(atc_code, ...)

atc_online_ddd(atc_code, ...)

atc_online_ddd_units(atc_code, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="atc_online_property_+3A_atc_code">atc_code</code></td>
<td>
<p>a <a href="base.html#topic+character">character</a> (vector) with ATC code(s) of antibiotics, will be coerced with <code><a href="#topic+as.ab">as.ab()</a></code> and <code><a href="#topic+ab_atc">ab_atc()</a></code> internally if not a valid ATC code</p>
</td></tr>
<tr><td><code id="atc_online_property_+3A_property">property</code></td>
<td>
<p>property of an ATC code. Valid values are <code>"ATC"</code>, <code>"Name"</code>, <code>"DDD"</code>, <code>"U"</code> (<code>"unit"</code>), <code>"Adm.R"</code>, <code>"Note"</code> and <code>groups</code>. For this last option, all hierarchical groups of an ATC code will be returned, see <em>Examples</em>.</p>
</td></tr>
<tr><td><code id="atc_online_property_+3A_administration">administration</code></td>
<td>
<p>type of administration when using <code>property = "Adm.R"</code>, see <em>Details</em></p>
</td></tr>
<tr><td><code id="atc_online_property_+3A_url">url</code></td>
<td>
<p>url of website of the WHOCC. The sign <code style="white-space: pre;">&#8288;%s&#8288;</code> can be used as a placeholder for ATC codes.</p>
</td></tr>
<tr><td><code id="atc_online_property_+3A_url_vet">url_vet</code></td>
<td>
<p>url of website of the WHOCC for veterinary medicine. The sign <code style="white-space: pre;">&#8288;%s&#8288;</code> can be used as a placeholder for ATC_vet codes (that all start with &quot;Q&quot;).</p>
</td></tr>
<tr><td><code id="atc_online_property_+3A_...">...</code></td>
<td>
<p>arguments to pass on to <code>atc_property</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Options for argument <code>administration</code>:
</p>

<ul>
<li> <p><code>"Implant"</code> = Implant
</p>
</li>
<li> <p><code>"Inhal"</code> = Inhalation
</p>
</li>
<li> <p><code>"Instill"</code> = Instillation
</p>
</li>
<li> <p><code>"N"</code> = nasal
</p>
</li>
<li> <p><code>"O"</code> = oral
</p>
</li>
<li> <p><code>"P"</code> = parenteral
</p>
</li>
<li> <p><code>"R"</code> = rectal
</p>
</li>
<li> <p><code>"SL"</code> = sublingual/buccal
</p>
</li>
<li> <p><code>"TD"</code> = transdermal
</p>
</li>
<li> <p><code>"V"</code> = vaginal
</p>
</li></ul>

<p>Abbreviations of return values when using <code>property = "U"</code> (unit):
</p>

<ul>
<li> <p><code>"g"</code> = gram
</p>
</li>
<li> <p><code>"mg"</code> = milligram
</p>
</li>
<li> <p><code>"mcg"</code> = microgram
</p>
</li>
<li> <p><code>"U"</code> = unit
</p>
</li>
<li> <p><code>"TU"</code> = thousand units
</p>
</li>
<li> <p><code>"MU"</code> = million units
</p>
</li>
<li> <p><code>"mmol"</code> = millimole
</p>
</li>
<li> <p><code>"ml"</code> = millilitre (e.g. eyedrops)
</p>
</li></ul>

<p><strong>N.B. This function requires an internet connection and only works if the following packages are installed: <code>curl</code>, <code>rvest</code>, <code>xml2</code>.</strong>
</p>


<h3>Source</h3>

<p><a href="https://www.whocc.no/atc_ddd_alterations__cumulative/ddd_alterations/abbrevations/">https://www.whocc.no/atc_ddd_alterations__cumulative/ddd_alterations/abbrevations/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("curl") &amp;&amp; requireNamespace("rvest") &amp;&amp; requireNamespace("xml2")) {
  # oral DDD (Defined Daily Dose) of amoxicillin
  atc_online_property("J01CA04", "DDD", "O")
  atc_online_ddd(ab_atc("amox"))

  # parenteral DDD (Defined Daily Dose) of amoxicillin
  atc_online_property("J01CA04", "DDD", "P")

  atc_online_property("J01CA04", property = "groups") # search hierarchical groups of amoxicillin
}

</code></pre>

<hr>
<h2 id='av_from_text'>Retrieve Antiviral Drug Names and Doses from Clinical Text</h2><span id='topic+av_from_text'></span>

<h3>Description</h3>

<p>Use this function on e.g. clinical texts from health care records. It returns a <a href="base.html#topic+list">list</a> with all antiviral drugs, doses and forms of administration found in the texts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>av_from_text(
  text,
  type = c("drug", "dose", "administration"),
  collapse = NULL,
  translate_av = FALSE,
  thorough_search = NULL,
  info = interactive(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="av_from_text_+3A_text">text</code></td>
<td>
<p>text to analyse</p>
</td></tr>
<tr><td><code id="av_from_text_+3A_type">type</code></td>
<td>
<p>type of property to search for, either <code>"drug"</code>, <code>"dose"</code> or <code>"administration"</code>, see <em>Examples</em></p>
</td></tr>
<tr><td><code id="av_from_text_+3A_collapse">collapse</code></td>
<td>
<p>a <a href="base.html#topic+character">character</a> to pass on to <code>paste(, collapse = ...)</code> to only return one <a href="base.html#topic+character">character</a> per element of <code>text</code>, see <em>Examples</em></p>
</td></tr>
<tr><td><code id="av_from_text_+3A_translate_av">translate_av</code></td>
<td>
<p>if <code>type = "drug"</code>: a column name of the <a href="#topic+antivirals">antivirals</a> data set to translate the antibiotic abbreviations to, using <code><a href="#topic+av_property">av_property()</a></code>. The default is <code>FALSE</code>. Using <code>TRUE</code> is equal to using &quot;name&quot;.</p>
</td></tr>
<tr><td><code id="av_from_text_+3A_thorough_search">thorough_search</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether the input must be extensively searched for misspelling and other faulty input values. Setting this to <code>TRUE</code> will take considerably more time than when using <code>FALSE</code>. At default, it will turn <code>TRUE</code> when all input elements contain a maximum of three words.</p>
</td></tr>
<tr><td><code id="av_from_text_+3A_info">info</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether a progress bar should be printed - the default is <code>TRUE</code> only in interactive mode</p>
</td></tr>
<tr><td><code id="av_from_text_+3A_...">...</code></td>
<td>
<p>arguments passed on to <code><a href="#topic+as.av">as.av()</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is also internally used by <code><a href="#topic+as.av">as.av()</a></code>, although it then only searches for the first drug name and will throw a note if more drug names could have been returned. Note: the <code><a href="#topic+as.av">as.av()</a></code> function may use very long regular expression to match brand names of antiviral drugs. This may fail on some systems.
</p>


<h4>Argument <code>type</code></h4>

<p>At default, the function will search for antiviral drug names. All text elements will be searched for official names, ATC codes and brand names. As it uses <code><a href="#topic+as.av">as.av()</a></code> internally, it will correct for misspelling.
</p>
<p>With <code>type = "dose"</code> (or similar, like &quot;dosing&quot;, &quot;doses&quot;), all text elements will be searched for <a href="base.html#topic+numeric">numeric</a> values that are higher than 100 and do not resemble years. The output will be <a href="base.html#topic+numeric">numeric</a>. It supports any unit (g, mg, IE, etc.) and multiple values in one clinical text, see <em>Examples</em>.
</p>
<p>With <code>type = "administration"</code> (or abbreviations, like &quot;admin&quot;, &quot;adm&quot;), all text elements will be searched for a form of drug administration. It supports the following forms (including common abbreviations): buccal, implant, inhalation, instillation, intravenous, nasal, oral, parenteral, rectal, sublingual, transdermal and vaginal. Abbreviations for oral (such as 'po', 'per os') will become &quot;oral&quot;, all values for intravenous (such as 'iv', 'intraven') will become &quot;iv&quot;. It supports multiple values in one clinical text, see <em>Examples</em>.
</p>



<h4>Argument <code>collapse</code></h4>

<p>Without using <code>collapse</code>, this function will return a <a href="base.html#topic+list">list</a>. This can be convenient to use e.g. inside a <code>mutate()</code>):<br />
<code>df %&gt;% mutate(avx = av_from_text(clinical_text))</code>
</p>
<p>The returned AV codes can be transformed to official names, groups, etc. with all <code><a href="#topic+av_property">av_*</a></code> functions such as <code><a href="#topic+av_name">av_name()</a></code> and <code><a href="#topic+av_group">av_group()</a></code>, or by using the <code>translate_av</code> argument.
</p>
<p>With using <code>collapse</code>, this function will return a <a href="base.html#topic+character">character</a>:<br />
<code>df %&gt;% mutate(avx = av_from_text(clinical_text, collapse = "|"))</code>
</p>



<h3>Value</h3>

<p>A <a href="base.html#topic+list">list</a>, or a  <a href="base.html#topic+character">character</a> if <code>collapse</code> is not <code>NULL</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>av_from_text("28/03/2020 valaciclovir po tid")
av_from_text("28/03/2020 valaciclovir po tid", type = "admin")
</code></pre>

<hr>
<h2 id='av_property'>Get Properties of an Antiviral Drug</h2><span id='topic+av_property'></span><span id='topic+av_name'></span><span id='topic+av_cid'></span><span id='topic+av_synonyms'></span><span id='topic+av_tradenames'></span><span id='topic+av_group'></span><span id='topic+av_atc'></span><span id='topic+av_loinc'></span><span id='topic+av_ddd'></span><span id='topic+av_ddd_units'></span><span id='topic+av_info'></span><span id='topic+av_url'></span>

<h3>Description</h3>

<p>Use these functions to return a specific property of an antiviral drug from the <a href="#topic+antivirals">antivirals</a> data set. All input values will be evaluated internally with <code><a href="#topic+as.av">as.av()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>av_name(x, language = get_AMR_locale(), tolower = FALSE, ...)

av_cid(x, ...)

av_synonyms(x, ...)

av_tradenames(x, ...)

av_group(x, language = get_AMR_locale(), ...)

av_atc(x, ...)

av_loinc(x, ...)

av_ddd(x, administration = "oral", ...)

av_ddd_units(x, administration = "oral", ...)

av_info(x, language = get_AMR_locale(), ...)

av_url(x, open = FALSE, ...)

av_property(x, property = "name", language = get_AMR_locale(), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="av_property_+3A_x">x</code></td>
<td>
<p>any (vector of) text that can be coerced to a valid antiviral drug code with <code><a href="#topic+as.av">as.av()</a></code></p>
</td></tr>
<tr><td><code id="av_property_+3A_language">language</code></td>
<td>
<p>language of the returned text - the default is system language (see <code><a href="#topic+get_AMR_locale">get_AMR_locale()</a></code>) and can also be set with the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_locale</a></code>. Use <code>language = NULL</code> or <code>language = ""</code> to prevent translation.</p>
</td></tr>
<tr><td><code id="av_property_+3A_tolower">tolower</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether the first <a href="base.html#topic+character">character</a> of every output should be transformed to a lower case <a href="base.html#topic+character">character</a>.</p>
</td></tr>
<tr><td><code id="av_property_+3A_...">...</code></td>
<td>
<p>other arguments passed on to <code><a href="#topic+as.av">as.av()</a></code></p>
</td></tr>
<tr><td><code id="av_property_+3A_administration">administration</code></td>
<td>
<p>way of administration, either <code>"oral"</code> or <code>"iv"</code></p>
</td></tr>
<tr><td><code id="av_property_+3A_open">open</code></td>
<td>
<p>browse the URL using <code><a href="utils.html#topic+browseURL">utils::browseURL()</a></code></p>
</td></tr>
<tr><td><code id="av_property_+3A_property">property</code></td>
<td>
<p>one of the column names of one of the <a href="#topic+antivirals">antivirals</a> data set: <code>vector_or(colnames(antivirals), sort = FALSE)</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All output <a href="#topic+translate">will be translated</a> where possible.
</p>
<p>The function <code><a href="#topic+av_url">av_url()</a></code> will return the direct URL to the official WHO website. A warning will be returned if the required ATC code is not available.
</p>


<h3>Value</h3>


<ul>
<li><p> An <a href="base.html#topic+integer">integer</a> in case of <code><a href="#topic+av_cid">av_cid()</a></code>
</p>
</li>
<li><p> A named <a href="base.html#topic+list">list</a> in case of <code><a href="#topic+av_info">av_info()</a></code> and multiple <code><a href="#topic+av_atc">av_atc()</a></code>/<code><a href="#topic+av_synonyms">av_synonyms()</a></code>/<code><a href="#topic+av_tradenames">av_tradenames()</a></code>
</p>
</li>
<li><p> A <a href="base.html#topic+double">double</a> in case of <code><a href="#topic+av_ddd">av_ddd()</a></code>
</p>
</li>
<li><p> A <a href="base.html#topic+character">character</a> in all other cases
</p>
</li></ul>



<h3>Source</h3>

<p>World Health Organization (WHO) Collaborating Centre for Drug Statistics Methodology: <a href="https://www.whocc.no/atc_ddd_index/">https://www.whocc.no/atc_ddd_index/</a>
</p>
<p>European Commission Public Health PHARMACEUTICALS - COMMUNITY REGISTER: <a href="https://ec.europa.eu/health/documents/community-register/html/reg_hum_atc.htm">https://ec.europa.eu/health/documents/community-register/html/reg_hum_atc.htm</a>
</p>


<h3>Reference Data Publicly Available</h3>

<p>All data sets in this <code>AMR</code> package (about microorganisms, antibiotics, SIR interpretation, EUCAST rules, etc.) are publicly and freely available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. We also provide tab-separated plain text files that are machine-readable and suitable for input in any software program, such as laboratory information systems. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>.
</p>


<h3>See Also</h3>

<p><a href="#topic+antivirals">antivirals</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># all properties:
av_name("ACI")
av_atc("ACI")
av_cid("ACI")
av_synonyms("ACI")
av_tradenames("ACI")
av_group("ACI")
av_url("ACI")

# lowercase transformation
av_name(x = c("ACI", "VALA"))
av_name(x = c("ACI", "VALA"), tolower = TRUE)

# defined daily doses (DDD)
av_ddd("ACI", "oral")
av_ddd_units("ACI", "oral")
av_ddd("ACI", "iv")
av_ddd_units("ACI", "iv")

av_info("ACI") # all properties as a list

# all av_* functions use as.av() internally, so you can go from 'any' to 'any':
av_atc("ACI")
av_group("J05AB01")
av_loinc("abacavir")
av_name("29113-8")
av_name(135398513)
av_name("J05AB01")
</code></pre>

<hr>
<h2 id='availability'>Check Availability of Columns</h2><span id='topic+availability'></span>

<h3>Description</h3>

<p>Easy check for data availability of all columns in a data set. This makes it easy to get an idea of which antimicrobial combinations can be used for calculation with e.g. <code><a href="#topic+susceptibility">susceptibility()</a></code> and <code><a href="#topic+resistance">resistance()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>availability(tbl, width = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="availability_+3A_tbl">tbl</code></td>
<td>
<p>a <a href="base.html#topic+data.frame">data.frame</a> or <a href="base.html#topic+list">list</a></p>
</td></tr>
<tr><td><code id="availability_+3A_width">width</code></td>
<td>
<p>number of characters to present the visual availability - the default is filling the width of the console</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function returns a <a href="base.html#topic+data.frame">data.frame</a> with columns <code>"resistant"</code> and <code>"visual_resistance"</code>. The values in that columns are calculated with <code><a href="#topic+resistance">resistance()</a></code>.
</p>


<h3>Value</h3>

<p><a href="base.html#topic+data.frame">data.frame</a> with column names of <code>tbl</code> as row names
</p>


<h3>Examples</h3>

<pre><code class='language-R'>availability(example_isolates)

if (require("dplyr")) {
  example_isolates %&gt;%
    filter(mo == as.mo("Escherichia coli")) %&gt;%
    select_if(is.sir) %&gt;%
    availability()
}

</code></pre>

<hr>
<h2 id='bug_drug_combinations'>Determine Bug-Drug Combinations</h2><span id='topic+bug_drug_combinations'></span><span id='topic+format.bug_drug_combinations'></span>

<h3>Description</h3>

<p>Determine antimicrobial resistance (AMR) of all bug-drug combinations in your data set where at least 30 (default) isolates are available per species. Use <code><a href="base.html#topic+format">format()</a></code> on the result to prettify it to a publishable/printable format, see <em>Examples</em>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bug_drug_combinations(x, col_mo = NULL, FUN = mo_shortname, ...)

## S3 method for class 'bug_drug_combinations'
format(
  x,
  translate_ab = "name (ab, atc)",
  language = get_AMR_locale(),
  minimum = 30,
  combine_SI = TRUE,
  add_ab_group = TRUE,
  remove_intrinsic_resistant = FALSE,
  decimal.mark = getOption("OutDec"),
  big.mark = ifelse(decimal.mark == ",", ".", ","),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bug_drug_combinations_+3A_x">x</code></td>
<td>
<p>a data set with antibiotic columns, such as <code>amox</code>, <code>AMX</code> and <code>AMC</code></p>
</td></tr>
<tr><td><code id="bug_drug_combinations_+3A_col_mo">col_mo</code></td>
<td>
<p>column name of the names or codes of the microorganisms (see <code><a href="#topic+as.mo">as.mo()</a></code>) - the default is the first column of class <code><a href="#topic+mo">mo</a></code>. Values will be coerced using <code><a href="#topic+as.mo">as.mo()</a></code>.</p>
</td></tr>
<tr><td><code id="bug_drug_combinations_+3A_fun">FUN</code></td>
<td>
<p>the function to call on the <code>mo</code> column to transform the microorganism codes - the default is <code><a href="#topic+mo_shortname">mo_shortname()</a></code></p>
</td></tr>
<tr><td><code id="bug_drug_combinations_+3A_...">...</code></td>
<td>
<p>arguments passed on to <code>FUN</code></p>
</td></tr>
<tr><td><code id="bug_drug_combinations_+3A_translate_ab">translate_ab</code></td>
<td>
<p>a <a href="base.html#topic+character">character</a> of length 1 containing column names of the <a href="#topic+antibiotics">antibiotics</a> data set</p>
</td></tr>
<tr><td><code id="bug_drug_combinations_+3A_language">language</code></td>
<td>
<p>language of the returned text - the default is the current system language (see <code><a href="#topic+get_AMR_locale">get_AMR_locale()</a></code>) and can also be set with the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_locale</a></code>. Use <code>language = NULL</code> or <code>language = ""</code> to prevent translation.</p>
</td></tr>
<tr><td><code id="bug_drug_combinations_+3A_minimum">minimum</code></td>
<td>
<p>the minimum allowed number of available (tested) isolates. Any isolate count lower than <code>minimum</code> will return <code>NA</code> with a warning. The default number of <code>30</code> isolates is advised by the Clinical and Laboratory Standards Institute (CLSI) as best practice, see <em>Source</em>.</p>
</td></tr>
<tr><td><code id="bug_drug_combinations_+3A_combine_si">combine_SI</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether values S and I should be summed, so resistance will be based on only R - the default is <code>TRUE</code></p>
</td></tr>
<tr><td><code id="bug_drug_combinations_+3A_add_ab_group">add_ab_group</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate where the group of the antimicrobials must be included as a first column</p>
</td></tr>
<tr><td><code id="bug_drug_combinations_+3A_remove_intrinsic_resistant">remove_intrinsic_resistant</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> to indicate that rows and columns with 100% resistance for all tested antimicrobials must be removed from the table</p>
</td></tr>
<tr><td><code id="bug_drug_combinations_+3A_decimal.mark">decimal.mark</code></td>
<td>
<p>the character to be used to indicate the numeric
decimal point.</p>
</td></tr>
<tr><td><code id="bug_drug_combinations_+3A_big.mark">big.mark</code></td>
<td>
<p>character; if not empty used as mark between every
<code>big.interval</code> decimals <em>before</em> (hence <code>big</code>) the
decimal point.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code><a href="base.html#topic+format">format()</a></code> calculates the resistance per bug-drug combination and returns a table ready for reporting/publishing. Use <code>combine_SI = TRUE</code> (default) to test R vs. S+I and <code>combine_SI = FALSE</code> to test R+I vs. S. This table can also directly be used in R Markdown / Quarto without the need for e.g. <code><a href="knitr.html#topic+kable">knitr::kable()</a></code>.
</p>


<h3>Value</h3>

<p>The function <code><a href="#topic+bug_drug_combinations">bug_drug_combinations()</a></code> returns a <a href="base.html#topic+data.frame">data.frame</a> with columns &quot;mo&quot;, &quot;ab&quot;, &quot;S&quot;, &quot;I&quot;, &quot;R&quot; and &quot;total&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># example_isolates is a data set available in the AMR package.
# run ?example_isolates for more info.
example_isolates


x &lt;- bug_drug_combinations(example_isolates)
head(x)
format(x, translate_ab = "name (atc)")

# Use FUN to change to transformation of microorganism codes
bug_drug_combinations(example_isolates,
  FUN = mo_gramstain
)

bug_drug_combinations(example_isolates,
  FUN = function(x) {
    ifelse(x == as.mo("Escherichia coli"),
      "E. coli",
      "Others"
    )
  }
)

</code></pre>

<hr>
<h2 id='clinical_breakpoints'>Data Set with Clinical Breakpoints for SIR Interpretation</h2><span id='topic+clinical_breakpoints'></span>

<h3>Description</h3>

<p>Data set containing clinical breakpoints to interpret MIC and disk diffusion to SIR values, according to international guidelines. Currently implemented guidelines are EUCAST (2011-2023) and CLSI (2011-2023). Use <code><a href="#topic+as.sir">as.sir()</a></code> to transform MICs or disks measurements to SIR values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clinical_breakpoints
</code></pre>


<h3>Format</h3>

<p>A <a href="tibble.html#topic+tibble">tibble</a> with 29 747 observations and 12 variables:
</p>

<ul>
<li> <p><code>guideline</code><br /> Name of the guideline
</p>
</li>
<li> <p><code>type</code><br /> Breakpoint type, either &quot;ECOFF&quot;, &quot;animal&quot;, or &quot;human&quot;
</p>
</li>
<li> <p><code>method</code><br /> Testing method, either &quot;DISK&quot; or &quot;MIC&quot;
</p>
</li>
<li> <p><code>site</code><br /> Body site for which the breakpoint must be applied, e.g. &quot;Oral&quot; or &quot;Respiratory&quot;
</p>
</li>
<li> <p><code>mo</code><br /> Microbial ID, see <code><a href="#topic+as.mo">as.mo()</a></code>
</p>
</li>
<li> <p><code>rank_index</code><br /> Taxonomic rank index of <code>mo</code> from 1 (subspecies/infraspecies) to 5 (unknown microorganism)
</p>
</li>
<li> <p><code>ab</code><br /> Antibiotic code as used by this package, EARS-Net and WHONET, see <code><a href="#topic+as.ab">as.ab()</a></code>
</p>
</li>
<li> <p><code>ref_tbl</code><br /> Info about where the guideline rule can be found
</p>
</li>
<li> <p><code>disk_dose</code><br /> Dose of the used disk diffusion method
</p>
</li>
<li> <p><code>breakpoint_S</code><br /> Lowest MIC value or highest number of millimetres that leads to &quot;S&quot;
</p>
</li>
<li> <p><code>breakpoint_R</code><br /> Highest MIC value or lowest number of millimetres that leads to &quot;R&quot;
</p>
</li>
<li> <p><code>uti</code><br /> A <a href="base.html#topic+logical">logical</a> value (<code>TRUE</code>/<code>FALSE</code>) to indicate whether the rule applies to a urinary tract infection (UTI)
</p>
</li></ul>



<h3>Details</h3>



<h4>Different types of breakpoints</h4>

<p>Supported types of breakpoints are ECOFF, animal, and human. ECOFF (Epidemiological cut-off) values are used in antimicrobial susceptibility testing to differentiate between wild-type and non-wild-type strains of bacteria or fungi.
</p>
<p>The default is <code>"human"</code>, which can also be set with the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_breakpoint_type</a></code>. Use <code><a href="#topic+as.sir">as.sir(..., breakpoint_type = ...)</a></code> to interpret raw data using a specific breakpoint type, e.g. <code>as.sir(..., breakpoint_type = "ECOFF")</code> to use ECOFFs.
</p>



<h4>Imported from WHONET</h4>

<p>Clinical breakpoints in this package were validated through and imported from <a href="https://whonet.org">WHONET</a>, a free desktop Windows application developed and supported by the WHO Collaborating Centre for Surveillance of Antimicrobial Resistance. More can be read on <a href="https://whonet.org">their website</a>. The developers of WHONET and this <code>AMR</code> package have been in contact about sharing their work. We highly appreciate their development on the WHONET software.
</p>



<h4>Response from CLSI and EUCAST</h4>

<p>The CEO of CLSI and the chairman of EUCAST have endorsed the work and public use of this <code>AMR</code> package (and consequently the use of their breakpoints) in June 2023, when future development of distributing clinical breakpoints was discussed in a meeting between CLSI, EUCAST, the WHO, and developers of WHONET and the <code>AMR</code> package.
</p>



<h4>Download</h4>

<p>Like all data sets in this package, this data set is publicly available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>. They allow for machine reading EUCAST and CLSI guidelines, which is almost impossible with the MS Excel and PDF files distributed by EUCAST and CLSI, though initiatives have started to overcome these burdens.
</p>
<p><strong>NOTE:</strong> this <code>AMR</code> package (and the WHONET software as well) contains internal methods to apply the guidelines, which is rather complex. For example, some breakpoints must be applied on certain species groups (which are in case of this package available through the <a href="#topic+microorganisms.groups">microorganisms.groups</a> data set). It is important that this is considered when using the breakpoints for own use.
</p>



<h3>See Also</h3>

<p><a href="#topic+intrinsic_resistant">intrinsic_resistant</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>clinical_breakpoints
</code></pre>

<hr>
<h2 id='count'>Count Available Isolates</h2><span id='topic+count'></span><span id='topic+count_resistant'></span><span id='topic+count_susceptible'></span><span id='topic+count_R'></span><span id='topic+count_IR'></span><span id='topic+count_I'></span><span id='topic+count_SI'></span><span id='topic+count_S'></span><span id='topic+count_all'></span><span id='topic+n_sir'></span><span id='topic+count_df'></span>

<h3>Description</h3>

<p>These functions can be used to count resistant/susceptible microbial isolates. All functions support quasiquotation with pipes, can be used in <code>summarise()</code> from the <code>dplyr</code> package and also support grouped variables, see <em>Examples</em>.
</p>
<p><code><a href="#topic+count_resistant">count_resistant()</a></code> should be used to count resistant isolates, <code><a href="#topic+count_susceptible">count_susceptible()</a></code> should be used to count susceptible isolates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count_resistant(..., only_all_tested = FALSE)

count_susceptible(..., only_all_tested = FALSE)

count_R(..., only_all_tested = FALSE)

count_IR(..., only_all_tested = FALSE)

count_I(..., only_all_tested = FALSE)

count_SI(..., only_all_tested = FALSE)

count_S(..., only_all_tested = FALSE)

count_all(..., only_all_tested = FALSE)

n_sir(..., only_all_tested = FALSE)

count_df(
  data,
  translate_ab = "name",
  language = get_AMR_locale(),
  combine_SI = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="count_+3A_...">...</code></td>
<td>
<p>one or more vectors (or columns) with antibiotic interpretations. They will be transformed internally with <code><a href="#topic+as.sir">as.sir()</a></code> if needed.</p>
</td></tr>
<tr><td><code id="count_+3A_only_all_tested">only_all_tested</code></td>
<td>
<p>(for combination therapies, i.e. using more than one variable for <code>...</code>): a <a href="base.html#topic+logical">logical</a> to indicate that isolates must be tested for all antibiotics, see section <em>Combination Therapy</em> below</p>
</td></tr>
<tr><td><code id="count_+3A_data">data</code></td>
<td>
<p>a <a href="base.html#topic+data.frame">data.frame</a> containing columns with class <code><a href="#topic+sir">sir</a></code> (see <code><a href="#topic+as.sir">as.sir()</a></code>)</p>
</td></tr>
<tr><td><code id="count_+3A_translate_ab">translate_ab</code></td>
<td>
<p>a column name of the <a href="#topic+antibiotics">antibiotics</a> data set to translate the antibiotic abbreviations to, using <code><a href="#topic+ab_property">ab_property()</a></code></p>
</td></tr>
<tr><td><code id="count_+3A_language">language</code></td>
<td>
<p>language of the returned text - the default is the current system language (see <code><a href="#topic+get_AMR_locale">get_AMR_locale()</a></code>) and can also be set with the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_locale</a></code>. Use <code>language = NULL</code> or <code>language = ""</code> to prevent translation.</p>
</td></tr>
<tr><td><code id="count_+3A_combine_si">combine_SI</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether all values of S and I must be merged into one, so the output only consists of S+I vs. R (susceptible vs. resistant) - the default is <code>TRUE</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions are meant to count isolates. Use the <code><a href="#topic+resistance">resistance()</a></code>/<code><a href="#topic+susceptibility">susceptibility()</a></code> functions to calculate microbial resistance/susceptibility.
</p>
<p>The function <code><a href="#topic+count_resistant">count_resistant()</a></code> is equal to the function <code><a href="#topic+count_R">count_R()</a></code>. The function <code><a href="#topic+count_susceptible">count_susceptible()</a></code> is equal to the function <code><a href="#topic+count_SI">count_SI()</a></code>.
</p>
<p>The function <code><a href="#topic+n_sir">n_sir()</a></code> is an alias of <code><a href="#topic+count_all">count_all()</a></code>. They can be used to count all available isolates, i.e. where all input antibiotics have an available result (S, I or R). Their use is equal to <code>n_distinct()</code>. Their function is equal to <code>count_susceptible(...) + count_resistant(...)</code>.
</p>
<p>The function <code><a href="#topic+count_df">count_df()</a></code> takes any variable from <code>data</code> that has an <code><a href="#topic+sir">sir</a></code> class (created with <code><a href="#topic+as.sir">as.sir()</a></code>) and counts the number of S's, I's and R's. It also supports grouped variables. The function <code><a href="#topic+sir_df">sir_df()</a></code> works exactly like <code><a href="#topic+count_df">count_df()</a></code>, but adds the percentage of S, I and R.
</p>


<h3>Value</h3>

<p>An <a href="base.html#topic+integer">integer</a>
</p>


<h3>Interpretation of SIR</h3>

<p>In 2019, the European Committee on Antimicrobial Susceptibility Testing (EUCAST) has decided to change the definitions of susceptibility testing categories S, I, and R as shown below (<a href="https://www.eucast.org/newsiandr">https://www.eucast.org/newsiandr</a>):
</p>

<ul>
<li> <p><strong>S - Susceptible, standard dosing regimen</strong><br />
A microorganism is categorised as &quot;Susceptible, standard dosing regimen&quot;, when there is a high likelihood of therapeutic success using a standard dosing regimen of the agent.
</p>
</li>
<li> <p><strong>I - Susceptible, increased exposure</strong> <em><br />
A microorganism is categorised as &quot;Susceptible, Increased exposure</em>&quot; when there is a high likelihood of therapeutic success because exposure to the agent is increased by adjusting the dosing regimen or by its concentration at the site of infection.
</p>
</li>
<li> <p><strong>R = Resistant</strong><br />
A microorganism is categorised as &quot;Resistant&quot; when there is a high likelihood of therapeutic failure even when there is increased exposure.
</p>

<ul>
<li> <p><em>Exposure</em> is a function of how the mode of administration, dose, dosing interval, infusion time, as well as distribution and excretion of the antimicrobial agent will influence the infecting organism at the site of infection.
</p>
</li></ul>

</li></ul>

<p>This AMR package honours this insight. Use <code><a href="#topic+susceptibility">susceptibility()</a></code> (equal to <code><a href="#topic+proportion_SI">proportion_SI()</a></code>) to determine antimicrobial susceptibility and <code><a href="#topic+count_susceptible">count_susceptible()</a></code> (equal to <code><a href="#topic+count_SI">count_SI()</a></code>) to count susceptible isolates.
</p>


<h3>Combination Therapy</h3>

<p>When using more than one variable for <code>...</code> (= combination therapy), use <code>only_all_tested</code> to only count isolates that are tested for all antibiotics/variables that you test them for. See this example for two antibiotics, Drug A and Drug B, about how <code><a href="#topic+susceptibility">susceptibility()</a></code> works to calculate the %SI:
</p>
<div class="sourceCode"><pre>--------------------------------------------------------------------
                    only_all_tested = FALSE  only_all_tested = TRUE
                    -----------------------  -----------------------
 Drug A    Drug B   include as  include as   include as  include as
                    numerator   denominator  numerator   denominator
--------  --------  ----------  -----------  ----------  -----------
 S or I    S or I       X            X            X            X
   R       S or I       X            X            X            X
  &lt;NA&gt;     S or I       X            X            -            -
 S or I      R          X            X            X            X
   R         R          -            X            -            X
  &lt;NA&gt;       R          -            -            -            -
 S or I     &lt;NA&gt;        X            X            -            -
   R        &lt;NA&gt;        -            -            -            -
  &lt;NA&gt;      &lt;NA&gt;        -            -            -            -
--------------------------------------------------------------------
</pre></div>
<p>Please note that, in combination therapies, for <code>only_all_tested = TRUE</code> applies that:
</p>
<div class="sourceCode"><pre>    count_S()    +   count_I()    +   count_R()    = count_all()
  proportion_S() + proportion_I() + proportion_R() = 1
</pre></div>
<p>and that, in combination therapies, for <code>only_all_tested = FALSE</code> applies that:
</p>
<div class="sourceCode"><pre>    count_S()    +   count_I()    +   count_R()    &gt;= count_all()
  proportion_S() + proportion_I() + proportion_R() &gt;= 1
</pre></div>
<p>Using <code>only_all_tested</code> has no impact when only using one antibiotic as input.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+proportion">proportion_*</a></code> to calculate microbial resistance and susceptibility.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># example_isolates is a data set available in the AMR package.
# run ?example_isolates for more info.

# base R ------------------------------------------------------------
count_resistant(example_isolates$AMX) # counts "R"
count_susceptible(example_isolates$AMX) # counts "S" and "I"
count_all(example_isolates$AMX) # counts "S", "I" and "R"

# be more specific
count_S(example_isolates$AMX)
count_SI(example_isolates$AMX)
count_I(example_isolates$AMX)
count_IR(example_isolates$AMX)
count_R(example_isolates$AMX)

# Count all available isolates
count_all(example_isolates$AMX)
n_sir(example_isolates$AMX)

# n_sir() is an alias of count_all().
# Since it counts all available isolates, you can
# calculate back to count e.g. susceptible isolates.
# These results are the same:
count_susceptible(example_isolates$AMX)
susceptibility(example_isolates$AMX) * n_sir(example_isolates$AMX)

# dplyr -------------------------------------------------------------

if (require("dplyr")) {
  example_isolates %&gt;%
    group_by(ward) %&gt;%
    summarise(
      R = count_R(CIP),
      I = count_I(CIP),
      S = count_S(CIP),
      n1 = count_all(CIP), # the actual total; sum of all three
      n2 = n_sir(CIP), # same - analogous to n_distinct
      total = n()
    ) # NOT the number of tested isolates!

  # Number of available isolates for a whole antibiotic class
  # (i.e., in this data set columns GEN, TOB, AMK, KAN)
  example_isolates %&gt;%
    group_by(ward) %&gt;%
    summarise(across(aminoglycosides(), n_sir))

  # Count co-resistance between amoxicillin/clav acid and gentamicin,
  # so we can see that combination therapy does a lot more than mono therapy.
  # Please mind that `susceptibility()` calculates percentages right away instead.
  example_isolates %&gt;% count_susceptible(AMC) # 1433
  example_isolates %&gt;% count_all(AMC) # 1879

  example_isolates %&gt;% count_susceptible(GEN) # 1399
  example_isolates %&gt;% count_all(GEN) # 1855

  example_isolates %&gt;% count_susceptible(AMC, GEN) # 1764
  example_isolates %&gt;% count_all(AMC, GEN) # 1936

  # Get number of S+I vs. R immediately of selected columns
  example_isolates %&gt;%
    select(AMX, CIP) %&gt;%
    count_df(translate = FALSE)

  # It also supports grouping variables
  example_isolates %&gt;%
    select(ward, AMX, CIP) %&gt;%
    group_by(ward) %&gt;%
    count_df(translate = FALSE)
}

</code></pre>

<hr>
<h2 id='custom_eucast_rules'>Define Custom EUCAST Rules</h2><span id='topic+custom_eucast_rules'></span>

<h3>Description</h3>

<p>Define custom EUCAST rules for your organisation or specific analysis and use the output of this function in <code><a href="#topic+eucast_rules">eucast_rules()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>custom_eucast_rules(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="custom_eucast_rules_+3A_...">...</code></td>
<td>
<p>rules in <a href="base.html#topic+tilde">formula</a> notation, see <em>Examples</em></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Some organisations have their own adoption of EUCAST rules. This function can be used to define custom EUCAST rules to be used in the <code><a href="#topic+eucast_rules">eucast_rules()</a></code> function.
</p>


<h3>Value</h3>

<p>A <a href="base.html#topic+list">list</a> containing the custom rules
</p>


<h3>How it works</h3>



<h4>Basics</h4>

<p>If you are familiar with the <code><a href="dplyr.html#topic+case_when">case_when()</a></code> function of the <code>dplyr</code> package, you will recognise the input method to set your own rules. Rules must be set using what <span class="rlang"><b>R</b></span> considers to be the 'formula notation'. The rule itself is written <em>before</em> the tilde (<code>~</code>) and the consequence of the rule is written <em>after</em> the tilde:
</p>
<div class="sourceCode r"><pre>x &lt;- custom_eucast_rules(TZP == "S" ~ aminopenicillins == "S",
                         TZP == "R" ~ aminopenicillins == "R")
</pre></div>
<p>These are two custom EUCAST rules: if TZP (piperacillin/tazobactam) is &quot;S&quot;, all aminopenicillins (ampicillin and amoxicillin) must be made &quot;S&quot;, and if TZP is &quot;R&quot;, aminopenicillins must be made &quot;R&quot;. These rules can also be printed to the console, so it is immediately clear how they work:
</p>
<div class="sourceCode r"><pre>x
#&gt; A set of custom EUCAST rules:
#&gt;
#&gt;   1. If TZP is "S" then set to  S :
#&gt;      amoxicillin (AMX), ampicillin (AMP)
#&gt;
#&gt;   2. If TZP is "R" then set to  R :
#&gt;      amoxicillin (AMX), ampicillin (AMP)
</pre></div>
<p>The rules (the part <em>before</em> the tilde, in above example <code>TZP == "S"</code> and <code>TZP == "R"</code>) must be evaluable in your data set: it should be able to run as a filter in your data set without errors. This means for the above example that the column <code>TZP</code> must exist. We will create a sample data set and test the rules set:
</p>
<div class="sourceCode r"><pre>df &lt;- data.frame(mo = c("Escherichia coli", "Klebsiella pneumoniae"),
                 TZP = as.sir("R"),
                 ampi = as.sir("S"),
                 cipro = as.sir("S"))
df
#&gt;                      mo TZP ampi cipro
#&gt; 1      Escherichia coli   R    S     S
#&gt; 2 Klebsiella pneumoniae   R    S     S

eucast_rules(df, rules = "custom", custom_rules = x, info = FALSE)
#&gt;                      mo TZP ampi cipro
#&gt; 1      Escherichia coli   R    R     S
#&gt; 2 Klebsiella pneumoniae   R    R     S
</pre></div>



<h4>Using taxonomic properties in rules</h4>

<p>There is one exception in columns used for the rules: all column names of the <a href="#topic+microorganisms">microorganisms</a> data set can also be used, but do not have to exist in the data set. These column names are: &quot;mo&quot;, &quot;fullname&quot;, &quot;status&quot;, &quot;kingdom&quot;, &quot;phylum&quot;, &quot;class&quot;, &quot;order&quot;, &quot;family&quot;, &quot;genus&quot;, &quot;species&quot;, &quot;subspecies&quot;, &quot;rank&quot;, &quot;ref&quot;, &quot;oxygen_tolerance&quot;, &quot;source&quot;, &quot;lpsn&quot;, &quot;lpsn_parent&quot;, &quot;lpsn_renamed_to&quot;, &quot;gbif&quot;, &quot;gbif_parent&quot;, &quot;gbif_renamed_to&quot;, &quot;prevalence&quot;, and &quot;snomed&quot;. Thus, this next example will work as well, despite the fact that the <code>df</code> data set does not contain a column <code>genus</code>:
</p>
<div class="sourceCode r"><pre>y &lt;- custom_eucast_rules(TZP == "S" &amp; genus == "Klebsiella" ~ aminopenicillins == "S",
                         TZP == "R" &amp; genus == "Klebsiella" ~ aminopenicillins == "R")

eucast_rules(df, rules = "custom", custom_rules = y, info = FALSE)
#&gt;                      mo TZP ampi cipro
#&gt; 1      Escherichia coli   R    S     S
#&gt; 2 Klebsiella pneumoniae   R    R     S
</pre></div>



<h4>Usage of antibiotic group names</h4>

<p>It is possible to define antibiotic groups instead of single antibiotics for the rule consequence, the part <em>after</em> the tilde. In above examples, the antibiotic group <code>aminopenicillins</code> is used to include ampicillin and amoxicillin. The following groups are allowed (case-insensitive). Within parentheses are the drugs that will be matched when running the rule.
</p>

<ul>
<li><p> &quot;aminoglycosides&quot;<br />(amikacin, amikacin/fosfomycin, amphotericin B-high, apramycin, arbekacin, astromicin, bekanamycin, dibekacin, framycetin, gentamicin, gentamicin-high, habekacin, hygromycin, isepamicin, kanamycin, kanamycin-high, kanamycin/cephalexin, micronomicin, neomycin, netilmicin, pentisomicin, plazomicin, propikacin, ribostamycin, sisomicin, streptoduocin, streptomycin, streptomycin-high, tobramycin, and tobramycin-high)
</p>
</li>
<li><p> &quot;aminopenicillins&quot;<br />(amoxicillin and ampicillin)
</p>
</li>
<li><p> &quot;antifungals&quot;<br />(amphotericin B, anidulafungin, butoconazole, caspofungin, ciclopirox, clotrimazole, econazole, fluconazole, flucytosine, fosfluconazole, griseofulvin, hachimycin, ibrexafungerp, isavuconazole, isoconazole, itraconazole, ketoconazole, manogepix, micafungin, miconazole, nystatin, oteseconazole, pimaricin, posaconazole, rezafungin, ribociclib, sulconazole, terbinafine, terconazole, and voriconazole)
</p>
</li>
<li><p> &quot;antimycobacterials&quot;<br />(4-aminosalicylic acid, calcium aminosalicylate, capreomycin, clofazimine, delamanid, enviomycin, ethambutol, ethambutol/isoniazid, ethionamide, isoniazid, isoniazid/sulfamethoxazole/trimethoprim/pyridoxine, morinamide, p-aminosalicylic acid, pretomanid, protionamide, pyrazinamide, rifabutin, rifampicin, rifampicin/ethambutol/isoniazid, rifampicin/isoniazid, rifampicin/pyrazinamide/ethambutol/isoniazid, rifampicin/pyrazinamide/isoniazid, rifamycin, rifapentine, simvastatin/fenofibrate, sodium aminosalicylate, streptomycin/isoniazid, terizidone, thioacetazone, thioacetazone/isoniazid, tiocarlide, and viomycin)
</p>
</li>
<li><p> &quot;betalactams&quot;<br />(amoxicillin, amoxicillin/clavulanic acid, amoxicillin/sulbactam, ampicillin, ampicillin/sulbactam, apalcillin, aspoxicillin, avibactam, azidocillin, azlocillin, aztreonam, aztreonam/avibactam, aztreonam/nacubactam, bacampicillin, benzathine benzylpenicillin, benzathine phenoxymethylpenicillin, benzylpenicillin, biapenem, carbenicillin, carindacillin, cefacetrile, cefaclor, cefadroxil, cefalexin, cefaloridine, cefalotin, cefamandole, cefapirin, cefatrizine, cefazedone, cefazolin, cefcapene, cefcapene pivoxil, cefdinir, cefditoren, cefditoren pivoxil, cefepime, cefepime/clavulanic acid, cefepime/nacubactam, cefepime/tazobactam, cefetamet, cefetamet pivoxil, cefetecol, cefetrizole, cefixime, cefmenoxime, cefmetazole, cefodizime, cefonicid, cefoperazone, cefoperazone/sulbactam, ceforanide, cefoselis, cefotaxime, cefotaxime/clavulanic acid, cefotaxime/sulbactam, cefotetan, cefotiam, cefotiam hexetil, cefovecin, cefoxitin, cefoxitin screening, cefozopran, cefpimizole, cefpiramide, cefpirome, cefpodoxime, cefpodoxime proxetil, cefpodoxime/clavulanic acid, cefprozil, cefquinome, cefroxadine, cefsulodin, cefsumide, ceftaroline, ceftaroline/avibactam, ceftazidime, ceftazidime/avibactam, ceftazidime/clavulanic acid, cefteram, cefteram pivoxil, ceftezole, ceftibuten, ceftiofur, ceftizoxime, ceftizoxime alapivoxil, ceftobiprole, ceftobiprole medocaril, ceftolozane/tazobactam, ceftriaxone, ceftriaxone/beta-lactamase inhibitor, cefuroxime, cefuroxime axetil, cephradine, ciclacillin, clometocillin, cloxacillin, dicloxacillin, doripenem, epicillin, ertapenem, flucloxacillin, hetacillin, imipenem, imipenem/EDTA, imipenem/relebactam, latamoxef, lenampicillin, loracarbef, mecillinam, meropenem, meropenem/nacubactam, meropenem/vaborbactam, metampicillin, meticillin, mezlocillin, mezlocillin/sulbactam, nacubactam, nafcillin, oxacillin, panipenem, penamecillin, penicillin/novobiocin, penicillin/sulbactam, pheneticillin, phenoxymethylpenicillin, piperacillin, piperacillin/sulbactam, piperacillin/tazobactam, piridicillin, pivampicillin, pivmecillinam, procaine benzylpenicillin, propicillin, razupenem, ritipenem, ritipenem acoxil, sarmoxicillin, sulbactam, sulbenicillin, sultamicillin, talampicillin, tazobactam, tebipenem, temocillin, ticarcillin, and ticarcillin/clavulanic acid)
</p>
</li>
<li><p> &quot;carbapenems&quot;<br />(biapenem, doripenem, ertapenem, imipenem, imipenem/EDTA, imipenem/relebactam, meropenem, meropenem/nacubactam, meropenem/vaborbactam, panipenem, razupenem, ritipenem, ritipenem acoxil, and tebipenem)
</p>
</li>
<li><p> &quot;cephalosporins&quot;<br />(cefacetrile, cefaclor, cefadroxil, cefalexin, cefaloridine, cefalotin, cefamandole, cefapirin, cefatrizine, cefazedone, cefazolin, cefcapene, cefcapene pivoxil, cefdinir, cefditoren, cefditoren pivoxil, cefepime, cefepime/clavulanic acid, cefepime/tazobactam, cefetamet, cefetamet pivoxil, cefetecol, cefetrizole, cefixime, cefmenoxime, cefmetazole, cefodizime, cefonicid, cefoperazone, cefoperazone/sulbactam, ceforanide, cefoselis, cefotaxime, cefotaxime/clavulanic acid, cefotaxime/sulbactam, cefotetan, cefotiam, cefotiam hexetil, cefovecin, cefoxitin, cefoxitin screening, cefozopran, cefpimizole, cefpiramide, cefpirome, cefpodoxime, cefpodoxime proxetil, cefpodoxime/clavulanic acid, cefprozil, cefquinome, cefroxadine, cefsulodin, cefsumide, ceftaroline, ceftaroline/avibactam, ceftazidime, ceftazidime/avibactam, ceftazidime/clavulanic acid, cefteram, cefteram pivoxil, ceftezole, ceftibuten, ceftiofur, ceftizoxime, ceftizoxime alapivoxil, ceftobiprole, ceftobiprole medocaril, ceftolozane/tazobactam, ceftriaxone, ceftriaxone/beta-lactamase inhibitor, cefuroxime, cefuroxime axetil, cephradine, latamoxef, and loracarbef)
</p>
</li>
<li><p> &quot;cephalosporins_1st&quot;<br />(cefacetrile, cefadroxil, cefalexin, cefaloridine, cefalotin, cefapirin, cefatrizine, cefazedone, cefazolin, cefroxadine, ceftezole, and cephradine)
</p>
</li>
<li><p> &quot;cephalosporins_2nd&quot;<br />(cefaclor, cefamandole, cefmetazole, cefonicid, ceforanide, cefotetan, cefotiam, cefoxitin, cefoxitin screening, cefprozil, cefuroxime, cefuroxime axetil, and loracarbef)
</p>
</li>
<li><p> &quot;cephalosporins_3rd&quot;<br />(cefcapene, cefcapene pivoxil, cefdinir, cefditoren, cefditoren pivoxil, cefetamet, cefetamet pivoxil, cefixime, cefmenoxime, cefodizime, cefoperazone, cefoperazone/sulbactam, cefotaxime, cefotaxime/clavulanic acid, cefotaxime/sulbactam, cefotiam hexetil, cefovecin, cefpimizole, cefpiramide, cefpodoxime, cefpodoxime proxetil, cefpodoxime/clavulanic acid, cefsulodin, ceftazidime, ceftazidime/avibactam, ceftazidime/clavulanic acid, cefteram, cefteram pivoxil, ceftibuten, ceftiofur, ceftizoxime, ceftizoxime alapivoxil, ceftriaxone, ceftriaxone/beta-lactamase inhibitor, and latamoxef)
</p>
</li>
<li><p> &quot;cephalosporins_4th&quot;<br />(cefepime, cefepime/clavulanic acid, cefepime/tazobactam, cefetecol, cefoselis, cefozopran, cefpirome, and cefquinome)
</p>
</li>
<li><p> &quot;cephalosporins_5th&quot;<br />(ceftaroline, ceftaroline/avibactam, ceftobiprole, ceftobiprole medocaril, and ceftolozane/tazobactam)
</p>
</li>
<li><p> &quot;cephalosporins_except_caz&quot;<br />(cefacetrile, cefaclor, cefadroxil, cefalexin, cefaloridine, cefalotin, cefamandole, cefapirin, cefatrizine, cefazedone, cefazolin, cefcapene, cefcapene pivoxil, cefdinir, cefditoren, cefditoren pivoxil, cefepime, cefepime/clavulanic acid, cefepime/tazobactam, cefetamet, cefetamet pivoxil, cefetecol, cefetrizole, cefixime, cefmenoxime, cefmetazole, cefodizime, cefonicid, cefoperazone, cefoperazone/sulbactam, ceforanide, cefoselis, cefotaxime, cefotaxime/clavulanic acid, cefotaxime/sulbactam, cefotetan, cefotiam, cefotiam hexetil, cefovecin, cefoxitin, cefoxitin screening, cefozopran, cefpimizole, cefpiramide, cefpirome, cefpodoxime, cefpodoxime proxetil, cefpodoxime/clavulanic acid, cefprozil, cefquinome, cefroxadine, cefsulodin, cefsumide, ceftaroline, ceftaroline/avibactam, ceftazidime/avibactam, ceftazidime/clavulanic acid, cefteram, cefteram pivoxil, ceftezole, ceftibuten, ceftiofur, ceftizoxime, ceftizoxime alapivoxil, ceftobiprole, ceftobiprole medocaril, ceftolozane/tazobactam, ceftriaxone, ceftriaxone/beta-lactamase inhibitor, cefuroxime, cefuroxime axetil, cephradine, latamoxef, and loracarbef)
</p>
</li>
<li><p> &quot;fluoroquinolones&quot;<br />(besifloxacin, ciprofloxacin, clinafloxacin, danofloxacin, delafloxacin, difloxacin, enoxacin, enrofloxacin, finafloxacin, fleroxacin, garenoxacin, gatifloxacin, gemifloxacin, grepafloxacin, lascufloxacin, levofloxacin, levonadifloxacin, lomefloxacin, marbofloxacin, metioxate, miloxacin, moxifloxacin, nadifloxacin, nifuroquine, norfloxacin, ofloxacin, orbifloxacin, pazufloxacin, pefloxacin, pradofloxacin, premafloxacin, prulifloxacin, rufloxacin, sarafloxacin, sitafloxacin, sparfloxacin, temafloxacin, tilbroquinol, tioxacin, tosufloxacin, and trovafloxacin)
</p>
</li>
<li><p> &quot;glycopeptides&quot;<br />(avoparcin, dalbavancin, norvancomycin, oritavancin, ramoplanin, teicoplanin, teicoplanin-macromethod, telavancin, vancomycin, and vancomycin-macromethod)
</p>
</li>
<li><p> &quot;glycopeptides_except_lipo&quot;<br />(avoparcin, norvancomycin, ramoplanin, teicoplanin, teicoplanin-macromethod, vancomycin, and vancomycin-macromethod)
</p>
</li>
<li><p> &quot;lincosamides&quot;<br />(acetylmidecamycin, acetylspiramycin, clindamycin, gamithromycin, kitasamycin, lincomycin, meleumycin, nafithromycin, pirlimycin, primycin, solithromycin, tildipirosin, tilmicosin, tulathromycin, tylosin, and tylvalosin)
</p>
</li>
<li><p> &quot;lipoglycopeptides&quot;<br />(dalbavancin, oritavancin, and telavancin)
</p>
</li>
<li><p> &quot;macrolides&quot;<br />(acetylmidecamycin, acetylspiramycin, azithromycin, clarithromycin, dirithromycin, erythromycin, flurithromycin, gamithromycin, josamycin, kitasamycin, meleumycin, midecamycin, miocamycin, nafithromycin, oleandomycin, pirlimycin, primycin, rokitamycin, roxithromycin, solithromycin, spiramycin, telithromycin, tildipirosin, tilmicosin, troleandomycin, tulathromycin, tylosin, and tylvalosin)
</p>
</li>
<li><p> &quot;oxazolidinones&quot;<br />(cadazolid, cycloserine, linezolid, tedizolid, and thiacetazone)
</p>
</li>
<li><p> &quot;penicillins&quot;<br />(amoxicillin, amoxicillin/clavulanic acid, amoxicillin/sulbactam, ampicillin, ampicillin/sulbactam, apalcillin, aspoxicillin, avibactam, azidocillin, azlocillin, aztreonam, aztreonam/avibactam, aztreonam/nacubactam, bacampicillin, benzathine benzylpenicillin, benzathine phenoxymethylpenicillin, benzylpenicillin, carbenicillin, carindacillin, cefepime/nacubactam, ciclacillin, clometocillin, cloxacillin, dicloxacillin, epicillin, flucloxacillin, hetacillin, lenampicillin, mecillinam, metampicillin, meticillin, mezlocillin, mezlocillin/sulbactam, nacubactam, nafcillin, oxacillin, penamecillin, penicillin/novobiocin, penicillin/sulbactam, pheneticillin, phenoxymethylpenicillin, piperacillin, piperacillin/sulbactam, piperacillin/tazobactam, piridicillin, pivampicillin, pivmecillinam, procaine benzylpenicillin, propicillin, sarmoxicillin, sulbactam, sulbenicillin, sultamicillin, talampicillin, tazobactam, temocillin, ticarcillin, and ticarcillin/clavulanic acid)
</p>
</li>
<li><p> &quot;polymyxins&quot;<br />(colistin, polymyxin B, and polymyxin B/polysorbate 80)
</p>
</li>
<li><p> &quot;quinolones&quot;<br />(besifloxacin, cinoxacin, ciprofloxacin, clinafloxacin, danofloxacin, delafloxacin, difloxacin, enoxacin, enrofloxacin, finafloxacin, fleroxacin, flumequine, garenoxacin, gatifloxacin, gemifloxacin, grepafloxacin, lascufloxacin, levofloxacin, levonadifloxacin, lomefloxacin, marbofloxacin, metioxate, miloxacin, moxifloxacin, nadifloxacin, nalidixic acid, nemonoxacin, nifuroquine, nitroxoline, norfloxacin, ofloxacin, orbifloxacin, oxolinic acid, pazufloxacin, pefloxacin, pipemidic acid, piromidic acid, pradofloxacin, premafloxacin, prulifloxacin, rosoxacin, rufloxacin, sarafloxacin, sitafloxacin, sparfloxacin, temafloxacin, tilbroquinol, tioxacin, tosufloxacin, and trovafloxacin)
</p>
</li>
<li><p> &quot;streptogramins&quot;<br />(pristinamycin and quinupristin/dalfopristin)
</p>
</li>
<li><p> &quot;tetracyclines&quot;<br />(cetocycline, chlortetracycline, clomocycline, demeclocycline, doxycycline, eravacycline, lymecycline, metacycline, minocycline, omadacycline, oxytetracycline, penimepicycline, rolitetracycline, sarecycline, tetracycline, and tigecycline)
</p>
</li>
<li><p> &quot;tetracyclines_except_tgc&quot;<br />(cetocycline, chlortetracycline, clomocycline, demeclocycline, doxycycline, eravacycline, lymecycline, metacycline, minocycline, omadacycline, oxytetracycline, penimepicycline, rolitetracycline, sarecycline, and tetracycline)
</p>
</li>
<li><p> &quot;trimethoprims&quot;<br />(brodimoprim, sulfadiazine, sulfadiazine/tetroxoprim, sulfadiazine/trimethoprim, sulfadimethoxine, sulfadimidine, sulfadimidine/trimethoprim, sulfafurazole, sulfaisodimidine, sulfalene, sulfamazone, sulfamerazine, sulfamerazine/trimethoprim, sulfamethizole, sulfamethoxazole, sulfamethoxypyridazine, sulfametomidine, sulfametoxydiazine, sulfametrole/trimethoprim, sulfamoxole, sulfamoxole/trimethoprim, sulfanilamide, sulfaperin, sulfaphenazole, sulfapyridine, sulfathiazole, sulfathiourea, trimethoprim, and trimethoprim/sulfamethoxazole)
</p>
</li>
<li><p> &quot;ureidopenicillins&quot;<br />(azlocillin, mezlocillin, piperacillin, and piperacillin/tazobactam)
</p>
</li></ul>




<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- custom_eucast_rules(
  AMC == "R" &amp; genus == "Klebsiella" ~ aminopenicillins == "R",
  AMC == "I" &amp; genus == "Klebsiella" ~ aminopenicillins == "I"
)
x

# run the custom rule set (verbose = TRUE will return a logbook instead of the data set):
eucast_rules(example_isolates,
  rules = "custom",
  custom_rules = x,
  info = FALSE,
  verbose = TRUE
)

# combine rule sets
x2 &lt;- c(
  x,
  custom_eucast_rules(TZP == "R" ~ carbapenems == "R")
)
x2
</code></pre>

<hr>
<h2 id='dosage'>Data Set with Treatment Dosages as Defined by EUCAST</h2><span id='topic+dosage'></span>

<h3>Description</h3>

<p>EUCAST breakpoints used in this package are based on the dosages in this data set. They can be retrieved with <code><a href="#topic+eucast_dosage">eucast_dosage()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dosage
</code></pre>


<h3>Format</h3>

<p>A <a href="tibble.html#topic+tibble">tibble</a> with 503 observations and 9 variables:
</p>

<ul>
<li> <p><code>ab</code><br /> Antibiotic ID as used in this package (such as <code>AMC</code>), using the official EARS-Net (European Antimicrobial Resistance Surveillance Network) codes where available
</p>
</li>
<li> <p><code>name</code><br /> Official name of the antimicrobial drug as used by WHONET/EARS-Net or the WHO
</p>
</li>
<li> <p><code>type</code><br /> Type of the dosage, either &quot;high_dosage&quot;, &quot;standard_dosage&quot;, or &quot;uncomplicated_uti&quot;
</p>
</li>
<li> <p><code>dose</code><br /> Dose, such as &quot;2 g&quot; or &quot;25 mg/kg&quot;
</p>
</li>
<li> <p><code>dose_times</code><br /> Number of times a dose must be administered
</p>
</li>
<li> <p><code>administration</code><br /> Route of administration, either &quot;im&quot;, &quot;iv&quot;, or &quot;oral&quot;
</p>
</li>
<li> <p><code>notes</code><br /> Additional dosage notes
</p>
</li>
<li> <p><code>original_txt</code><br /> Original text in the PDF file of EUCAST
</p>
</li>
<li> <p><code>eucast_version</code><br /> Version number of the EUCAST Clinical Breakpoints guideline to which these dosages apply, either 13, 12, or 11
</p>
</li></ul>



<h3>Details</h3>

<p>Like all data sets in this package, this data set is publicly available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dosage
</code></pre>

<hr>
<h2 id='eucast_rules'>Apply EUCAST Rules</h2><span id='topic+eucast_rules'></span><span id='topic+EUCAST'></span><span id='topic+eucast_dosage'></span>

<h3>Description</h3>

<p>Apply rules for clinical breakpoints and intrinsic resistance as defined by the European Committee on Antimicrobial Susceptibility Testing (EUCAST, <a href="https://www.eucast.org">https://www.eucast.org</a>), see <em>Source</em>. Use <code><a href="#topic+eucast_dosage">eucast_dosage()</a></code> to get a <a href="base.html#topic+data.frame">data.frame</a> with advised dosages of a certain bug-drug combination, which is based on the <a href="#topic+dosage">dosage</a> data set.
</p>
<p>To improve the interpretation of the antibiogram before EUCAST rules are applied, some non-EUCAST rules can applied at default, see <em>Details</em>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eucast_rules(
  x,
  col_mo = NULL,
  info = interactive(),
  rules = getOption("AMR_eucastrules", default = c("breakpoints", "expert")),
  verbose = FALSE,
  version_breakpoints = 12,
  version_expertrules = 3.3,
  ampc_cephalosporin_resistance = NA,
  only_sir_columns = FALSE,
  custom_rules = NULL,
  ...
)

eucast_dosage(ab, administration = "iv", version_breakpoints = 12)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="eucast_rules_+3A_x">x</code></td>
<td>
<p>a data set with antibiotic columns, such as <code>amox</code>, <code>AMX</code> and <code>AMC</code></p>
</td></tr>
<tr><td><code id="eucast_rules_+3A_col_mo">col_mo</code></td>
<td>
<p>column name of the names or codes of the microorganisms (see <code><a href="#topic+as.mo">as.mo()</a></code>) - the default is the first column of class <code><a href="#topic+mo">mo</a></code>. Values will be coerced using <code><a href="#topic+as.mo">as.mo()</a></code>.</p>
</td></tr>
<tr><td><code id="eucast_rules_+3A_info">info</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether progress should be printed to the console - the default is only print while in interactive sessions</p>
</td></tr>
<tr><td><code id="eucast_rules_+3A_rules">rules</code></td>
<td>
<p>a <a href="base.html#topic+character">character</a> vector that specifies which rules should be applied. Must be one or more of <code>"breakpoints"</code>, <code>"expert"</code>, <code>"other"</code>, <code>"custom"</code>, <code>"all"</code>, and defaults to <code>c("breakpoints", "expert")</code>. The default value can be set to another value using the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_eucastrules</a></code>: <code>options(AMR_eucastrules = "all")</code>. If using <code>"custom"</code>, be sure to fill in argument <code>custom_rules</code> too. Custom rules can be created with <code><a href="#topic+custom_eucast_rules">custom_eucast_rules()</a></code>.</p>
</td></tr>
<tr><td><code id="eucast_rules_+3A_verbose">verbose</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to turn Verbose mode on and off (default is off). In Verbose mode, the function does not apply rules to the data, but instead returns a data set in logbook form with extensive info about which rows and columns would be effected and in which way. Using Verbose mode takes a lot more time.</p>
</td></tr>
<tr><td><code id="eucast_rules_+3A_version_breakpoints">version_breakpoints</code></td>
<td>
<p>the version number to use for the EUCAST Clinical Breakpoints guideline. Can be &quot;12.0&quot;, &quot;11.0&quot;, or &quot;10.0&quot;.</p>
</td></tr>
<tr><td><code id="eucast_rules_+3A_version_expertrules">version_expertrules</code></td>
<td>
<p>the version number to use for the EUCAST Expert Rules and Intrinsic Resistance guideline. Can be &quot;3.3&quot;, &quot;3.2&quot;, or &quot;3.1&quot;.</p>
</td></tr>
<tr><td><code id="eucast_rules_+3A_ampc_cephalosporin_resistance">ampc_cephalosporin_resistance</code></td>
<td>
<p>a <a href="base.html#topic+character">character</a> value that should be applied to cefotaxime, ceftriaxone and ceftazidime for AmpC de-repressed cephalosporin-resistant mutants - the default is <code>NA</code>. Currently only works when <code>version_expertrules</code> is <code>3.2</code> and higher; these version of '<em>EUCAST Expert Rules on Enterobacterales</em>' state that results of cefotaxime, ceftriaxone and ceftazidime should be reported with a note, or results should be suppressed (emptied) for these three drugs. A value of <code>NA</code> (the default) for this argument will remove results for these three drugs, while e.g. a value of <code>"R"</code> will make the results for these drugs resistant. Use <code>NULL</code> or <code>FALSE</code> to not alter results for these three drugs of AmpC de-repressed cephalosporin-resistant mutants. Using <code>TRUE</code> is equal to using <code>"R"</code>. <br /> For <em>EUCAST Expert Rules</em> v3.2, this rule applies to: <em>Citrobacter braakii</em>, <em>Citrobacter freundii</em>, <em>Citrobacter gillenii</em>, <em>Citrobacter murliniae</em>, <em>Citrobacter rodenticum</em>, <em>Citrobacter sedlakii</em>, <em>Citrobacter werkmanii</em>, <em>Citrobacter youngae</em>, <em>Enterobacter</em>, <em>Hafnia alvei</em>, <em>Klebsiella aerogenes</em>, <em>Morganella morganii</em>, <em>Providencia</em>, and <em>Serratia</em>.</p>
</td></tr>
<tr><td><code id="eucast_rules_+3A_only_sir_columns">only_sir_columns</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether only antibiotic columns must be detected that were transformed to class <code>sir</code> (see <code><a href="#topic+as.sir">as.sir()</a></code>) on beforehand (default is <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="eucast_rules_+3A_custom_rules">custom_rules</code></td>
<td>
<p>custom rules to apply, created with <code><a href="#topic+custom_eucast_rules">custom_eucast_rules()</a></code></p>
</td></tr>
<tr><td><code id="eucast_rules_+3A_...">...</code></td>
<td>
<p>column name of an antibiotic, see section <em>Antibiotics</em> below</p>
</td></tr>
<tr><td><code id="eucast_rules_+3A_ab">ab</code></td>
<td>
<p>any (vector of) text that can be coerced to a valid antibiotic drug code with <code><a href="#topic+as.ab">as.ab()</a></code></p>
</td></tr>
<tr><td><code id="eucast_rules_+3A_administration">administration</code></td>
<td>
<p>route of administration, either &quot;im&quot;, &quot;iv&quot;, or &quot;oral&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Note:</strong> This function does not translate MIC values to SIR values. Use <code><a href="#topic+as.sir">as.sir()</a></code> for that. <br />
<strong>Note:</strong> When ampicillin (AMP, J01CA01) is not available but amoxicillin (AMX, J01CA04) is, the latter will be used for all rules where there is a dependency on ampicillin. These drugs are interchangeable when it comes to expression of antimicrobial resistance. <br />
</p>
<p>The file containing all EUCAST rules is located here: <a href="https://github.com/msberends/AMR/blob/main/data-raw/eucast_rules.tsv">https://github.com/msberends/AMR/blob/main/data-raw/eucast_rules.tsv</a>.  <strong>Note:</strong> Old taxonomic names are replaced with the current taxonomy where applicable. For example, <em>Ochrobactrum anthropi</em> was renamed to <em>Brucella anthropi</em> in 2020; the original EUCAST rules v3.1 and v3.2 did not yet contain this new taxonomic name. The <code>AMR</code> package contains the full microbial taxonomy updated until December 11th, 2022, see <a href="#topic+microorganisms">microorganisms</a>.
</p>


<h4>Custom Rules</h4>

<p>Custom rules can be created using <code><a href="#topic+custom_eucast_rules">custom_eucast_rules()</a></code>, e.g.:
</p>
<div class="sourceCode r"><pre>x &lt;- custom_eucast_rules(AMC == "R" &amp; genus == "Klebsiella" ~ aminopenicillins == "R",
                         AMC == "I" &amp; genus == "Klebsiella" ~ aminopenicillins == "I")

eucast_rules(example_isolates, rules = "custom", custom_rules = x)
</pre></div>



<h4>'Other' Rules</h4>

<p>Before further processing, two non-EUCAST rules about drug combinations can be applied to improve the efficacy of the EUCAST rules, and the reliability of your data (analysis). These rules are:
</p>

<ol>
<li><p> A drug <strong>with</strong> enzyme inhibitor will be set to S if the same drug <strong>without</strong> enzyme inhibitor is S
</p>
</li>
<li><p> A drug <strong>without</strong> enzyme inhibitor will be set to R if the same drug <strong>with</strong> enzyme inhibitor is R
</p>
</li></ol>

<p>Important examples include amoxicillin and amoxicillin/clavulanic acid, and trimethoprim and trimethoprim/sulfamethoxazole. Needless to say, for these rules to work, both drugs must be available in the data set.
</p>
<p>Since these rules are not officially approved by EUCAST, they are not applied at default. To use these rules, include <code>"other"</code> to the <code>rules</code> argument, or use <code>eucast_rules(..., rules = "all")</code>. You can also set the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_eucastrules</a></code>, i.e. run <code>options(AMR_eucastrules = "all")</code>.
</p>



<h3>Value</h3>

<p>The input of <code>x</code>, possibly with edited values of antibiotics. Or, if <code>verbose = TRUE</code>, a <a href="base.html#topic+data.frame">data.frame</a> with all original and new values of the affected bug-drug combinations.
</p>


<h3>Antibiotics</h3>

<p>To define antibiotics column names, leave as it is to determine it automatically with <code><a href="#topic+guess_ab_col">guess_ab_col()</a></code> or input a text (case-insensitive), or use <code>NULL</code> to skip a column (e.g. <code>TIC = NULL</code> to skip ticarcillin). Manually defined but non-existing columns will be skipped with a warning.
</p>
<p>The following antibiotics are eligible for the functions <code><a href="#topic+eucast_rules">eucast_rules()</a></code> and <code><a href="#topic+mdro">mdro()</a></code>. These are shown below in the format 'name (<code style="white-space: pre;">&#8288;antimicrobial ID&#8288;</code>, <a href="https://www.whocc.no/atc/structure_and_principles/">ATC code</a>)', sorted alphabetically:
</p>
<p>Amikacin (<code>AMK</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB06&amp;showdescription=no">J01GB06</a>), amoxicillin (<code>AMX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA04&amp;showdescription=no">J01CA04</a>), amoxicillin/clavulanic acid (<code>AMC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CR02&amp;showdescription=no">J01CR02</a>), ampicillin (<code>AMP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA01&amp;showdescription=no">J01CA01</a>), ampicillin/sulbactam (<code>SAM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CR01&amp;showdescription=no">J01CR01</a>), arbekacin (<code>ARB</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB12&amp;showdescription=no">J01GB12</a>), aspoxicillin (<code>APX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA19&amp;showdescription=no">J01CA19</a>), azidocillin (<code>AZD</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CE04&amp;showdescription=no">J01CE04</a>), azithromycin (<code>AZM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA10&amp;showdescription=no">J01FA10</a>), azlocillin (<code>AZL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA09&amp;showdescription=no">J01CA09</a>), aztreonam (<code>ATM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DF01&amp;showdescription=no">J01DF01</a>), bacampicillin (<code>BAM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA06&amp;showdescription=no">J01CA06</a>), bekanamycin (<code>BEK</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB13&amp;showdescription=no">J01GB13</a>), benzathine benzylpenicillin (<code>BNB</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CE08&amp;showdescription=no">J01CE08</a>), benzathine phenoxymethylpenicillin (<code>BNP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CE10&amp;showdescription=no">J01CE10</a>), benzylpenicillin (<code>PEN</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CE01&amp;showdescription=no">J01CE01</a>), besifloxacin (<code>BES</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=S01AE08&amp;showdescription=no">S01AE08</a>), biapenem (<code>BIA</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DH05&amp;showdescription=no">J01DH05</a>), carbenicillin (<code>CRB</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA03&amp;showdescription=no">J01CA03</a>), carindacillin (<code>CRN</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA05&amp;showdescription=no">J01CA05</a>), cefacetrile (<code>CAC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB10&amp;showdescription=no">J01DB10</a>), cefaclor (<code>CEC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC04&amp;showdescription=no">J01DC04</a>), cefadroxil (<code>CFR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB05&amp;showdescription=no">J01DB05</a>), cefalexin (<code>LEX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB01&amp;showdescription=no">J01DB01</a>), cefaloridine (<code>RID</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB02&amp;showdescription=no">J01DB02</a>), cefalotin (<code>CEP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB03&amp;showdescription=no">J01DB03</a>), cefamandole (<code>MAN</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC03&amp;showdescription=no">J01DC03</a>), cefapirin (<code>HAP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB08&amp;showdescription=no">J01DB08</a>), cefatrizine (<code>CTZ</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB07&amp;showdescription=no">J01DB07</a>), cefazedone (<code>CZD</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB06&amp;showdescription=no">J01DB06</a>), cefazolin (<code>CZO</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB04&amp;showdescription=no">J01DB04</a>), cefcapene (<code>CCP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD17&amp;showdescription=no">J01DD17</a>), cefdinir (<code>CDR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD15&amp;showdescription=no">J01DD15</a>), cefditoren (<code>DIT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD16&amp;showdescription=no">J01DD16</a>), cefepime (<code>FEP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DE01&amp;showdescription=no">J01DE01</a>), cefetamet (<code>CAT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD10&amp;showdescription=no">J01DD10</a>), cefixime (<code>CFM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD08&amp;showdescription=no">J01DD08</a>), cefmenoxime (<code>CMX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD05&amp;showdescription=no">J01DD05</a>), cefmetazole (<code>CMZ</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC09&amp;showdescription=no">J01DC09</a>), cefodizime (<code>DIZ</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD09&amp;showdescription=no">J01DD09</a>), cefonicid (<code>CID</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC06&amp;showdescription=no">J01DC06</a>), cefoperazone (<code>CFP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD12&amp;showdescription=no">J01DD12</a>), cefoperazone/sulbactam (<code>CSL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD62&amp;showdescription=no">J01DD62</a>), ceforanide (<code>CND</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC11&amp;showdescription=no">J01DC11</a>), cefotaxime (<code>CTX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD01&amp;showdescription=no">J01DD01</a>), cefotaxime/clavulanic acid (<code>CTC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD51&amp;showdescription=no">J01DD51</a>), cefotetan (<code>CTT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC05&amp;showdescription=no">J01DC05</a>), cefotiam (<code>CTF</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC07&amp;showdescription=no">J01DC07</a>), cefoxitin (<code>FOX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC01&amp;showdescription=no">J01DC01</a>), cefozopran (<code>ZOP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DE03&amp;showdescription=no">J01DE03</a>), cefpiramide (<code>CPM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD11&amp;showdescription=no">J01DD11</a>), cefpirome (<code>CPO</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DE02&amp;showdescription=no">J01DE02</a>), cefpodoxime (<code>CPD</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD13&amp;showdescription=no">J01DD13</a>), cefprozil (<code>CPR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC10&amp;showdescription=no">J01DC10</a>), cefroxadine (<code>CRD</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB11&amp;showdescription=no">J01DB11</a>), cefsulodin (<code>CFS</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD03&amp;showdescription=no">J01DD03</a>), ceftaroline (<code>CPT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DI02&amp;showdescription=no">J01DI02</a>), ceftazidime (<code>CAZ</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD02&amp;showdescription=no">J01DD02</a>), ceftazidime/clavulanic acid (<code>CCV</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD52&amp;showdescription=no">J01DD52</a>), cefteram (<code>CEM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD18&amp;showdescription=no">J01DD18</a>), ceftezole (<code>CTL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB12&amp;showdescription=no">J01DB12</a>), ceftibuten (<code>CTB</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD14&amp;showdescription=no">J01DD14</a>), ceftizoxime (<code>CZX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD07&amp;showdescription=no">J01DD07</a>), ceftobiprole medocaril (<code>CFM1</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DI01&amp;showdescription=no">J01DI01</a>), ceftolozane/tazobactam (<code>CZT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DI54&amp;showdescription=no">J01DI54</a>), ceftriaxone (<code>CRO</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD04&amp;showdescription=no">J01DD04</a>), ceftriaxone/beta-lactamase inhibitor (<code>CEB</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD63&amp;showdescription=no">J01DD63</a>), cefuroxime (<code>CXM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC02&amp;showdescription=no">J01DC02</a>), cephradine (<code>CED</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB09&amp;showdescription=no">J01DB09</a>), chloramphenicol (<code>CHL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01BA01&amp;showdescription=no">J01BA01</a>), ciprofloxacin (<code>CIP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA02&amp;showdescription=no">J01MA02</a>), clarithromycin (<code>CLR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA09&amp;showdescription=no">J01FA09</a>), clindamycin (<code>CLI</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FF01&amp;showdescription=no">J01FF01</a>), clometocillin (<code>CLM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CE07&amp;showdescription=no">J01CE07</a>), cloxacillin (<code>CLO</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CF02&amp;showdescription=no">J01CF02</a>), colistin (<code>COL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XB01&amp;showdescription=no">J01XB01</a>), cycloserine (<code>CYC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J04AB01&amp;showdescription=no">J04AB01</a>), dalbavancin (<code>DAL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XA04&amp;showdescription=no">J01XA04</a>), daptomycin (<code>DAP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XX09&amp;showdescription=no">J01XX09</a>), delafloxacin (<code>DFX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA23&amp;showdescription=no">J01MA23</a>), dibekacin (<code>DKB</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB09&amp;showdescription=no">J01GB09</a>), dicloxacillin (<code>DIC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CF01&amp;showdescription=no">J01CF01</a>), dirithromycin (<code>DIR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA13&amp;showdescription=no">J01FA13</a>), doripenem (<code>DOR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DH04&amp;showdescription=no">J01DH04</a>), doxycycline (<code>DOX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01AA02&amp;showdescription=no">J01AA02</a>), enoxacin (<code>ENX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA04&amp;showdescription=no">J01MA04</a>), epicillin (<code>EPC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA07&amp;showdescription=no">J01CA07</a>), ertapenem (<code>ETP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DH03&amp;showdescription=no">J01DH03</a>), erythromycin (<code>ERY</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA01&amp;showdescription=no">J01FA01</a>), fleroxacin (<code>FLE</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA08&amp;showdescription=no">J01MA08</a>), flucloxacillin (<code>FLC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CF05&amp;showdescription=no">J01CF05</a>), flurithromycin (<code>FLR1</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA14&amp;showdescription=no">J01FA14</a>), fosfomycin (<code>FOS</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XX01&amp;showdescription=no">J01XX01</a>), framycetin (<code>FRM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=D09AA01&amp;showdescription=no">D09AA01</a>), fusidic acid (<code>FUS</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XC01&amp;showdescription=no">J01XC01</a>), garenoxacin (<code>GRN</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA19&amp;showdescription=no">J01MA19</a>), gatifloxacin (<code>GAT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA16&amp;showdescription=no">J01MA16</a>), gemifloxacin (<code>GEM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA15&amp;showdescription=no">J01MA15</a>), gentamicin (<code>GEN</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB03&amp;showdescription=no">J01GB03</a>), grepafloxacin (<code>GRX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA11&amp;showdescription=no">J01MA11</a>), hetacillin (<code>HET</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA18&amp;showdescription=no">J01CA18</a>), imipenem (<code>IPM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DH51&amp;showdescription=no">J01DH51</a>), imipenem/relebactam (<code>IMR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DH56&amp;showdescription=no">J01DH56</a>), isepamicin (<code>ISE</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB11&amp;showdescription=no">J01GB11</a>), josamycin (<code>JOS</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA07&amp;showdescription=no">J01FA07</a>), kanamycin (<code>KAN</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB04&amp;showdescription=no">J01GB04</a>), lascufloxacin (<code>LSC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA25&amp;showdescription=no">J01MA25</a>), latamoxef (<code>LTM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD06&amp;showdescription=no">J01DD06</a>), levofloxacin (<code>LVX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA12&amp;showdescription=no">J01MA12</a>), levonadifloxacin (<code>LND</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA24&amp;showdescription=no">J01MA24</a>), lincomycin (<code>LIN</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FF02&amp;showdescription=no">J01FF02</a>), linezolid (<code>LNZ</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XX08&amp;showdescription=no">J01XX08</a>), lomefloxacin (<code>LOM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA07&amp;showdescription=no">J01MA07</a>), loracarbef (<code>LOR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC08&amp;showdescription=no">J01DC08</a>), mecillinam (<code>MEC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA11&amp;showdescription=no">J01CA11</a>), meropenem (<code>MEM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DH02&amp;showdescription=no">J01DH02</a>), meropenem/vaborbactam (<code>MEV</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DH52&amp;showdescription=no">J01DH52</a>), metampicillin (<code>MTM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA14&amp;showdescription=no">J01CA14</a>), meticillin (<code>MET</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CF03&amp;showdescription=no">J01CF03</a>), mezlocillin (<code>MEZ</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA10&amp;showdescription=no">J01CA10</a>), micronomicin (<code>MCR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=S01AA22&amp;showdescription=no">S01AA22</a>), midecamycin (<code>MID</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA03&amp;showdescription=no">J01FA03</a>), minocycline (<code>MNO</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01AA08&amp;showdescription=no">J01AA08</a>), miocamycin (<code>MCM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA11&amp;showdescription=no">J01FA11</a>), moxifloxacin (<code>MFX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA14&amp;showdescription=no">J01MA14</a>), nadifloxacin (<code>NAD</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=D10AF05&amp;showdescription=no">D10AF05</a>), nafcillin (<code>NAF</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CF06&amp;showdescription=no">J01CF06</a>), nalidixic acid (<code>NAL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MB02&amp;showdescription=no">J01MB02</a>), neomycin (<code>NEO</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB05&amp;showdescription=no">J01GB05</a>), netilmicin (<code>NET</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB07&amp;showdescription=no">J01GB07</a>), nitrofurantoin (<code>NIT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XE01&amp;showdescription=no">J01XE01</a>), norfloxacin (<code>NOR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA06&amp;showdescription=no">J01MA06</a>), ofloxacin (<code>OFX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA01&amp;showdescription=no">J01MA01</a>), oleandomycin (<code>OLE</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA05&amp;showdescription=no">J01FA05</a>), oritavancin (<code>ORI</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XA05&amp;showdescription=no">J01XA05</a>), oxacillin (<code>OXA</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CF04&amp;showdescription=no">J01CF04</a>), panipenem (<code>PAN</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DH55&amp;showdescription=no">J01DH55</a>), pazufloxacin (<code>PAZ</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA18&amp;showdescription=no">J01MA18</a>), pefloxacin (<code>PEF</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA03&amp;showdescription=no">J01MA03</a>), penamecillin (<code>PNM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CE06&amp;showdescription=no">J01CE06</a>), pheneticillin (<code>PHE</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CE05&amp;showdescription=no">J01CE05</a>), phenoxymethylpenicillin (<code>PHN</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CE02&amp;showdescription=no">J01CE02</a>), piperacillin (<code>PIP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA12&amp;showdescription=no">J01CA12</a>), piperacillin/tazobactam (<code>TZP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CR05&amp;showdescription=no">J01CR05</a>), pivampicillin (<code>PVM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA02&amp;showdescription=no">J01CA02</a>), pivmecillinam (<code>PME</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA08&amp;showdescription=no">J01CA08</a>), plazomicin (<code>PLZ</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB14&amp;showdescription=no">J01GB14</a>), polymyxin B (<code>PLB</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XB02&amp;showdescription=no">J01XB02</a>), pristinamycin (<code>PRI</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FG01&amp;showdescription=no">J01FG01</a>), procaine benzylpenicillin (<code>PRB</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CE09&amp;showdescription=no">J01CE09</a>), propicillin (<code>PRP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CE03&amp;showdescription=no">J01CE03</a>), prulifloxacin (<code>PRU</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA17&amp;showdescription=no">J01MA17</a>), quinupristin/dalfopristin (<code>QDA</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FG02&amp;showdescription=no">J01FG02</a>), ribostamycin (<code>RST</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB10&amp;showdescription=no">J01GB10</a>), rifampicin (<code>RIF</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J04AB02&amp;showdescription=no">J04AB02</a>), rokitamycin (<code>ROK</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA12&amp;showdescription=no">J01FA12</a>), roxithromycin (<code>RXT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA06&amp;showdescription=no">J01FA06</a>), rufloxacin (<code>RFL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA10&amp;showdescription=no">J01MA10</a>), sisomicin (<code>SIS</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB08&amp;showdescription=no">J01GB08</a>), sitafloxacin (<code>SIT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA21&amp;showdescription=no">J01MA21</a>), solithromycin (<code>SOL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA16&amp;showdescription=no">J01FA16</a>), sparfloxacin (<code>SPX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA09&amp;showdescription=no">J01MA09</a>), spiramycin (<code>SPI</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA02&amp;showdescription=no">J01FA02</a>), streptoduocin (<code>STR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GA02&amp;showdescription=no">J01GA02</a>), streptomycin (<code>STR1</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GA01&amp;showdescription=no">J01GA01</a>), sulbactam (<code>SUL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CG01&amp;showdescription=no">J01CG01</a>), sulbenicillin (<code>SBC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA16&amp;showdescription=no">J01CA16</a>), sulfadiazine (<code>SDI</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EC02&amp;showdescription=no">J01EC02</a>), sulfadiazine/trimethoprim (<code>SLT1</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EE02&amp;showdescription=no">J01EE02</a>), sulfadimethoxine (<code>SUD</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01ED01&amp;showdescription=no">J01ED01</a>), sulfadimidine (<code>SDM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EB03&amp;showdescription=no">J01EB03</a>), sulfadimidine/trimethoprim (<code>SLT2</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EE05&amp;showdescription=no">J01EE05</a>), sulfafurazole (<code>SLF</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EB05&amp;showdescription=no">J01EB05</a>), sulfaisodimidine (<code>SLF1</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EB01&amp;showdescription=no">J01EB01</a>), sulfalene (<code>SLF2</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01ED02&amp;showdescription=no">J01ED02</a>), sulfamazone (<code>SZO</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01ED09&amp;showdescription=no">J01ED09</a>), sulfamerazine (<code>SLF3</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01ED07&amp;showdescription=no">J01ED07</a>), sulfamerazine/trimethoprim (<code>SLT3</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EE07&amp;showdescription=no">J01EE07</a>), sulfamethizole (<code>SLF4</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EB02&amp;showdescription=no">J01EB02</a>), sulfamethoxazole (<code>SMX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EC01&amp;showdescription=no">J01EC01</a>), sulfamethoxypyridazine (<code>SLF5</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01ED05&amp;showdescription=no">J01ED05</a>), sulfametomidine (<code>SLF6</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01ED03&amp;showdescription=no">J01ED03</a>), sulfametoxydiazine (<code>SLF7</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01ED04&amp;showdescription=no">J01ED04</a>), sulfametrole/trimethoprim (<code>SLT4</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EE03&amp;showdescription=no">J01EE03</a>), sulfamoxole (<code>SLF8</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EC03&amp;showdescription=no">J01EC03</a>), sulfamoxole/trimethoprim (<code>SLT5</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EE04&amp;showdescription=no">J01EE04</a>), sulfanilamide (<code>SLF9</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EB06&amp;showdescription=no">J01EB06</a>), sulfaperin (<code>SLF10</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01ED06&amp;showdescription=no">J01ED06</a>), sulfaphenazole (<code>SLF11</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01ED08&amp;showdescription=no">J01ED08</a>), sulfapyridine (<code>SLF12</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EB04&amp;showdescription=no">J01EB04</a>), sulfathiazole (<code>SUT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EB07&amp;showdescription=no">J01EB07</a>), sulfathiourea (<code>SLF13</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EB08&amp;showdescription=no">J01EB08</a>), sultamicillin (<code>SLT6</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CR04&amp;showdescription=no">J01CR04</a>), talampicillin (<code>TAL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA15&amp;showdescription=no">J01CA15</a>), tazobactam (<code>TAZ</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CG02&amp;showdescription=no">J01CG02</a>), tebipenem (<code>TBP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DH06&amp;showdescription=no">J01DH06</a>), tedizolid (<code>TZD</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XX11&amp;showdescription=no">J01XX11</a>), teicoplanin (<code>TEC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XA02&amp;showdescription=no">J01XA02</a>), telavancin (<code>TLV</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XA03&amp;showdescription=no">J01XA03</a>), telithromycin (<code>TLT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA15&amp;showdescription=no">J01FA15</a>), temafloxacin (<code>TMX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA05&amp;showdescription=no">J01MA05</a>), temocillin (<code>TEM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA17&amp;showdescription=no">J01CA17</a>), tetracycline (<code>TCY</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01AA07&amp;showdescription=no">J01AA07</a>), ticarcillin (<code>TIC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA13&amp;showdescription=no">J01CA13</a>), ticarcillin/clavulanic acid (<code>TCC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CR03&amp;showdescription=no">J01CR03</a>), tigecycline (<code>TGC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01AA12&amp;showdescription=no">J01AA12</a>), tilbroquinol (<code>TBQ</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=P01AA05&amp;showdescription=no">P01AA05</a>), tobramycin (<code>TOB</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB01&amp;showdescription=no">J01GB01</a>), tosufloxacin (<code>TFX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA22&amp;showdescription=no">J01MA22</a>), trimethoprim (<code>TMP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EA01&amp;showdescription=no">J01EA01</a>), trimethoprim/sulfamethoxazole (<code>SXT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EE01&amp;showdescription=no">J01EE01</a>), troleandomycin (<code>TRL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA08&amp;showdescription=no">J01FA08</a>), trovafloxacin (<code>TVA</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA13&amp;showdescription=no">J01MA13</a>), vancomycin (<code>VAN</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XA01&amp;showdescription=no">J01XA01</a>)
</p>


<h3>Reference Data Publicly Available</h3>

<p>All data sets in this <code>AMR</code> package (about microorganisms, antibiotics, SIR interpretation, EUCAST rules, etc.) are publicly and freely available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. We also provide tab-separated plain text files that are machine-readable and suitable for input in any software program, such as laboratory information systems. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>.
</p>


<h3>Source</h3>


<ul>
<li><p> EUCAST Expert Rules. Version 2.0, 2012.<br />
Leclercq et al. <strong>EUCAST expert rules in antimicrobial susceptibility testing.</strong> <em>Clin Microbiol Infect.</em> 2013;19(2):141-60; <a href="https://doi.org/10.1111/j.1469-0691.2011.03703.x">doi:10.1111/j.1469-0691.2011.03703.x</a>
</p>
</li>
<li><p> EUCAST Expert Rules, Intrinsic Resistance and Exceptional Phenotypes Tables. Version 3.1, 2016. <a href="https://www.eucast.org/fileadmin/src/media/PDFs/EUCAST_files/Expert_Rules/Expert_rules_intrinsic_exceptional_V3.1.pdf">(link)</a>
</p>
</li>
<li><p> EUCAST Intrinsic Resistance and Unusual Phenotypes. Version 3.2, 2020. <a href="https://www.eucast.org/fileadmin/src/media/PDFs/EUCAST_files/Expert_Rules/2020/Intrinsic_Resistance_and_Unusual_Phenotypes_Tables_v3.2_20200225.pdf">(link)</a>
</p>
</li>
<li><p> EUCAST Intrinsic Resistance and Unusual Phenotypes. Version 3.3, 2021. <a href="https://www.eucast.org/fileadmin/src/media/PDFs/EUCAST_files/Expert_Rules/2021/Intrinsic_Resistance_and_Unusual_Phenotypes_Tables_v3.3_20211018.pdf">(link)</a>
</p>
</li>
<li><p> EUCAST Breakpoint tables for interpretation of MICs and zone diameters. Version 9.0, 2019. <a href="https://www.eucast.org/fileadmin/src/media/PDFs/EUCAST_files/Breakpoint_tables/v_9.0_Breakpoint_Tables.xlsx">(link)</a>
</p>
</li>
<li><p> EUCAST Breakpoint tables for interpretation of MICs and zone diameters. Version 10.0, 2020. <a href="https://www.eucast.org/fileadmin/src/media/PDFs/EUCAST_files/Breakpoint_tables/v_10.0_Breakpoint_Tables.xlsx">(link)</a>
</p>
</li>
<li><p> EUCAST Breakpoint tables for interpretation of MICs and zone diameters. Version 11.0, 2021. <a href="https://www.eucast.org/fileadmin/src/media/PDFs/EUCAST_files/Breakpoint_tables/v_11.0_Breakpoint_Tables.xlsx">(link)</a>
</p>
</li>
<li><p> EUCAST Breakpoint tables for interpretation of MICs and zone diameters. Version 12.0, 2022. <a href="https://www.eucast.org/fileadmin/src/media/PDFs/EUCAST_files/Breakpoint_tables/v_12.0_Breakpoint_Tables.xlsx">(link)</a>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
a &lt;- data.frame(
  mo = c(
    "Staphylococcus aureus",
    "Enterococcus faecalis",
    "Escherichia coli",
    "Klebsiella pneumoniae",
    "Pseudomonas aeruginosa"
  ),
  VAN = "-", # Vancomycin
  AMX = "-", # Amoxicillin
  COL = "-", # Colistin
  CAZ = "-", # Ceftazidime
  CXM = "-", # Cefuroxime
  PEN = "S", # Benzylpenicillin
  FOX = "S", # Cefoxitin
  stringsAsFactors = FALSE
)

head(a)


# apply EUCAST rules: some results wil be changed
b &lt;- eucast_rules(a)

head(b)


# do not apply EUCAST rules, but rather get a data.frame
# containing all details about the transformations:
c &lt;- eucast_rules(a, verbose = TRUE)
head(c)


# Dosage guidelines:

eucast_dosage(c("tobra", "genta", "cipro"), "iv")

eucast_dosage(c("tobra", "genta", "cipro"), "iv", version_breakpoints = 10)
</code></pre>

<hr>
<h2 id='example_isolates'>Data Set with 2 000 Example Isolates</h2><span id='topic+example_isolates'></span>

<h3>Description</h3>

<p>A data set containing 2 000 microbial isolates with their full antibiograms. This data set contains randomised fictitious data, but reflects reality and can be used to practise AMR data analysis. For examples, please read <a href="https://msberends.github.io/AMR/articles/AMR.html">the tutorial on our website</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_isolates
</code></pre>


<h3>Format</h3>

<p>A <a href="tibble.html#topic+tibble">tibble</a> with 2 000 observations and 46 variables:
</p>

<ul>
<li> <p><code>date</code><br /> Date of receipt at the laboratory
</p>
</li>
<li> <p><code>patient</code><br /> ID of the patient
</p>
</li>
<li> <p><code>age</code><br /> Age of the patient
</p>
</li>
<li> <p><code>gender</code><br /> Gender of the patient, either &quot;F&quot; or &quot;M&quot;
</p>
</li>
<li> <p><code>ward</code><br /> Ward type where the patient was admitted, either &quot;Clinical&quot;, &quot;ICU&quot;, or &quot;Outpatient&quot;
</p>
</li>
<li> <p><code>mo</code><br /> ID of microorganism created with <code><a href="#topic+as.mo">as.mo()</a></code>, see also the <a href="#topic+microorganisms">microorganisms</a> data set
</p>
</li>
<li> <p><code>PEN:RIF</code><br /> 40 different antibiotics with class <code><a href="#topic+sir">sir</a></code> (see <code><a href="#topic+as.sir">as.sir()</a></code>); these column names occur in the <a href="#topic+antibiotics">antibiotics</a> data set and can be translated with <code><a href="#topic+set_ab_names">set_ab_names()</a></code> or <code><a href="#topic+ab_name">ab_name()</a></code>
</p>
</li></ul>



<h3>Details</h3>

<p>Like all data sets in this package, this data set is publicly available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example_isolates
</code></pre>

<hr>
<h2 id='example_isolates_unclean'>Data Set with Unclean Data</h2><span id='topic+example_isolates_unclean'></span>

<h3>Description</h3>

<p>A data set containing 3 000 microbial isolates that are not cleaned up and consequently not ready for AMR data analysis. This data set can be used for practice.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_isolates_unclean
</code></pre>


<h3>Format</h3>

<p>A <a href="tibble.html#topic+tibble">tibble</a> with 3 000 observations and 8 variables:
</p>

<ul>
<li> <p><code>patient_id</code><br /> ID of the patient
</p>
</li>
<li> <p><code>date</code><br /> date of receipt at the laboratory
</p>
</li>
<li> <p><code>hospital</code><br /> ID of the hospital, from A to C
</p>
</li>
<li> <p><code>bacteria</code><br /> info about microorganism that can be transformed with <code><a href="#topic+as.mo">as.mo()</a></code>, see also <a href="#topic+microorganisms">microorganisms</a>
</p>
</li>
<li> <p><code>AMX:GEN</code><br /> 4 different antibiotics that have to be transformed with <code><a href="#topic+as.sir">as.sir()</a></code>
</p>
</li></ul>



<h3>Details</h3>

<p>Like all data sets in this package, this data set is publicly available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example_isolates_unclean
</code></pre>

<hr>
<h2 id='first_isolate'>Determine First Isolates</h2><span id='topic+first_isolate'></span><span id='topic+filter_first_isolate'></span>

<h3>Description</h3>

<p>Determine first isolates of all microorganisms of every patient per episode and (if needed) per specimen type. These functions support all four methods as summarised by Hindler <em>et al.</em> in 2007 (<a href="https://doi.org/10.1086/511864">doi:10.1086/511864</a>). To determine patient episodes not necessarily based on microorganisms, use <code><a href="#topic+is_new_episode">is_new_episode()</a></code> that also supports grouping with the <code>dplyr</code> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>first_isolate(
  x = NULL,
  col_date = NULL,
  col_patient_id = NULL,
  col_mo = NULL,
  col_testcode = NULL,
  col_specimen = NULL,
  col_icu = NULL,
  col_keyantimicrobials = NULL,
  episode_days = 365,
  testcodes_exclude = NULL,
  icu_exclude = FALSE,
  specimen_group = NULL,
  type = "points",
  method = c("phenotype-based", "episode-based", "patient-based", "isolate-based"),
  ignore_I = TRUE,
  points_threshold = 2,
  info = interactive(),
  include_unknown = FALSE,
  include_untested_sir = TRUE,
  ...
)

filter_first_isolate(
  x = NULL,
  col_date = NULL,
  col_patient_id = NULL,
  col_mo = NULL,
  episode_days = 365,
  method = c("phenotype-based", "episode-based", "patient-based", "isolate-based"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="first_isolate_+3A_x">x</code></td>
<td>
<p>a <a href="base.html#topic+data.frame">data.frame</a> containing isolates. Can be left blank for automatic determination, see <em>Examples</em>.</p>
</td></tr>
<tr><td><code id="first_isolate_+3A_col_date">col_date</code></td>
<td>
<p>column name of the result date (or date that is was received on the lab) - the default is the first column with a date class</p>
</td></tr>
<tr><td><code id="first_isolate_+3A_col_patient_id">col_patient_id</code></td>
<td>
<p>column name of the unique IDs of the patients - the default is the first column that starts with 'patient' or 'patid' (case insensitive)</p>
</td></tr>
<tr><td><code id="first_isolate_+3A_col_mo">col_mo</code></td>
<td>
<p>column name of the names or codes of the microorganisms (see <code><a href="#topic+as.mo">as.mo()</a></code>) - the default is the first column of class <code><a href="#topic+mo">mo</a></code>. Values will be coerced using <code><a href="#topic+as.mo">as.mo()</a></code>.</p>
</td></tr>
<tr><td><code id="first_isolate_+3A_col_testcode">col_testcode</code></td>
<td>
<p>column name of the test codes. Use <code>col_testcode = NULL</code> to <strong>not</strong> exclude certain test codes (such as test codes for screening). In that case <code>testcodes_exclude</code> will be ignored.</p>
</td></tr>
<tr><td><code id="first_isolate_+3A_col_specimen">col_specimen</code></td>
<td>
<p>column name of the specimen type or group</p>
</td></tr>
<tr><td><code id="first_isolate_+3A_col_icu">col_icu</code></td>
<td>
<p>column name of the logicals (<code>TRUE</code>/<code>FALSE</code>) whether a ward or department is an Intensive Care Unit (ICU). This can also be a <a href="base.html#topic+logical">logical</a> vector with the same length as rows in <code>x</code>.</p>
</td></tr>
<tr><td><code id="first_isolate_+3A_col_keyantimicrobials">col_keyantimicrobials</code></td>
<td>
<p>(only useful when <code>method = "phenotype-based"</code>) column name of the key antimicrobials to determine first isolates, see <code><a href="#topic+key_antimicrobials">key_antimicrobials()</a></code>. The default is the first column that starts with 'key' followed by 'ab' or 'antibiotics' or 'antimicrobials' (case insensitive). Use <code>col_keyantimicrobials = FALSE</code> to prevent this. Can also be the output of <code><a href="#topic+key_antimicrobials">key_antimicrobials()</a></code>.</p>
</td></tr>
<tr><td><code id="first_isolate_+3A_episode_days">episode_days</code></td>
<td>
<p>episode in days after which a genus/species combination will be determined as 'first isolate' again. The default of 365 days is based on the guideline by CLSI, see <em>Source</em>.</p>
</td></tr>
<tr><td><code id="first_isolate_+3A_testcodes_exclude">testcodes_exclude</code></td>
<td>
<p>a <a href="base.html#topic+character">character</a> vector with test codes that should be excluded (case-insensitive)</p>
</td></tr>
<tr><td><code id="first_isolate_+3A_icu_exclude">icu_exclude</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether ICU isolates should be excluded (rows with value <code>TRUE</code> in the column set with <code>col_icu</code>)</p>
</td></tr>
<tr><td><code id="first_isolate_+3A_specimen_group">specimen_group</code></td>
<td>
<p>value in the column set with <code>col_specimen</code> to filter on</p>
</td></tr>
<tr><td><code id="first_isolate_+3A_type">type</code></td>
<td>
<p>type to determine weighed isolates; can be <code>"keyantimicrobials"</code> or <code>"points"</code>, see <em>Details</em></p>
</td></tr>
<tr><td><code id="first_isolate_+3A_method">method</code></td>
<td>
<p>the method to apply, either <code>"phenotype-based"</code>, <code>"episode-based"</code>, <code>"patient-based"</code> or <code>"isolate-based"</code> (can be abbreviated), see <em>Details</em>. The default is <code>"phenotype-based"</code> if antimicrobial test results are present in the data, and <code>"episode-based"</code> otherwise.</p>
</td></tr>
<tr><td><code id="first_isolate_+3A_ignore_i">ignore_I</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> to indicate whether antibiotic interpretations with <code>"I"</code> will be ignored when <code>type = "keyantimicrobials"</code>, see <em>Details</em></p>
</td></tr>
<tr><td><code id="first_isolate_+3A_points_threshold">points_threshold</code></td>
<td>
<p>minimum number of points to require before differences in the antibiogram will lead to inclusion of an isolate when <code>type = "points"</code>, see <em>Details</em></p>
</td></tr>
<tr><td><code id="first_isolate_+3A_info">info</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate info should be printed - the default is <code>TRUE</code> only in interactive mode</p>
</td></tr>
<tr><td><code id="first_isolate_+3A_include_unknown">include_unknown</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether 'unknown' microorganisms should be included too, i.e. microbial code <code>"UNKNOWN"</code>, which defaults to <code>FALSE</code>. For WHONET users, this means that all records with organism code <code>"con"</code> (<em>contamination</em>) will be excluded at default. Isolates with a microbial ID of <code>NA</code> will always be excluded as first isolate.</p>
</td></tr>
<tr><td><code id="first_isolate_+3A_include_untested_sir">include_untested_sir</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether also rows without antibiotic results are still eligible for becoming a first isolate. Use <code>include_untested_sir = FALSE</code> to always return <code>FALSE</code> for such rows. This checks the data set for columns of class <code>sir</code> and consequently requires transforming columns with antibiotic results using <code><a href="#topic+as.sir">as.sir()</a></code> first.</p>
</td></tr>
<tr><td><code id="first_isolate_+3A_...">...</code></td>
<td>
<p>arguments passed on to <code><a href="#topic+first_isolate">first_isolate()</a></code> when using <code><a href="#topic+filter_first_isolate">filter_first_isolate()</a></code>, otherwise arguments passed on to <code><a href="#topic+key_antimicrobials">key_antimicrobials()</a></code> (such as <code>universal</code>, <code>gram_negative</code>, <code>gram_positive</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To conduct epidemiological analyses on antimicrobial resistance data, only so-called first isolates should be included to prevent overestimation and underestimation of antimicrobial resistance. Different methods can be used to do so, see below.
</p>
<p>These functions are context-aware. This means that the <code>x</code> argument can be left blank if used inside a <a href="base.html#topic+data.frame">data.frame</a> call, see <em>Examples</em>.
</p>
<p>The <code><a href="#topic+first_isolate">first_isolate()</a></code> function is a wrapper around the <code><a href="#topic+is_new_episode">is_new_episode()</a></code> function, but more efficient for data sets containing microorganism codes or names.
</p>
<p>All isolates with a microbial ID of <code>NA</code> will be excluded as first isolate.
</p>


<h4>Different methods</h4>

<p>According to Hindler <em>et al.</em> (2007, <a href="https://doi.org/10.1086/511864">doi:10.1086/511864</a>), there are different methods (algorithms) to select first isolates with increasing reliability: isolate-based, patient-based, episode-based and phenotype-based. All methods select on a combination of the taxonomic genus and species (not subspecies).
</p>
<p>All mentioned methods are covered in the <code><a href="#topic+first_isolate">first_isolate()</a></code> function:</p>

<table>
<tr>
 <td style="text-align: left;">
   <strong>Method</strong> </td><td style="text-align: left;"> <strong>Function to apply</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
   <strong>Isolate-based</strong> </td><td style="text-align: left;"> <code>first_isolate(x, method = "isolate-based")</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   <em>(= all isolates)</em> </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
    </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
    </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
   <strong>Patient-based</strong> </td><td style="text-align: left;"> <code>first_isolate(x, method = "patient-based")</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   <em>(= first isolate per patient)</em> </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
    </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
    </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
   <strong>Episode-based</strong> </td><td style="text-align: left;"> <code>first_isolate(x, method = "episode-based")</code>, or: </td>
</tr>
<tr>
 <td style="text-align: left;">
   <em>(= first isolate per episode)</em> </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
   - 7-Day interval from initial isolate </td><td style="text-align: left;"> - <code>first_isolate(x, method = "e", episode_days = 7)</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   - 30-Day interval from initial isolate </td><td style="text-align: left;"> - <code>first_isolate(x, method = "e", episode_days = 30)</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
    </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
    </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
   <strong>Phenotype-based</strong> </td><td style="text-align: left;"> <code>first_isolate(x, method = "phenotype-based")</code>, or: </td>
</tr>
<tr>
 <td style="text-align: left;">
   <em>(= first isolate per phenotype)</em> </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
   - Major difference in any antimicrobial result </td><td style="text-align: left;"> - <code>first_isolate(x, type = "points")</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   - Any difference in key antimicrobial results </td><td style="text-align: left;"> - <code>first_isolate(x, type = "keyantimicrobials")</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>




<h4>Isolate-based</h4>

<p>This method does not require any selection, as all isolates should be included. It does, however, respect all arguments set in the <code><a href="#topic+first_isolate">first_isolate()</a></code> function. For example, the default setting for <code>include_unknown</code> (<code>FALSE</code>) will omit selection of rows without a microbial ID.
</p>



<h4>Patient-based</h4>

<p>To include every genus-species combination per patient once, set the <code>episode_days</code> to <code>Inf</code>. Although often inappropriate, this method makes sure that no duplicate isolates are selected from the same patient. In a large longitudinal data set, this could mean that isolates are <em>excluded</em> that were found years after the initial isolate.
</p>



<h4>Episode-based</h4>

<p>To include every genus-species combination per patient episode once, set the <code>episode_days</code> to a sensible number of days. Depending on the type of analysis, this could be 14, 30, 60 or 365. Short episodes are common for analysing specific hospital or ward data, long episodes are common for analysing regional and national data.
</p>
<p>This is the most common method to correct for duplicate isolates. Patients are categorised into episodes based on their ID and dates (e.g., the date of specimen receipt or laboratory result). While this is a common method, it does not take into account antimicrobial test results. This means that e.g. a methicillin-resistant <em>Staphylococcus aureus</em> (MRSA) isolate cannot be differentiated from a wildtype <em>Staphylococcus aureus</em> isolate.
</p>



<h4>Phenotype-based</h4>

<p>This is a more reliable method, since it also <em>weighs</em> the antibiogram (antimicrobial test results) yielding so-called 'first weighted isolates'. There are two different methods to weigh the antibiogram:
</p>

<ol>
<li><p> Using <code>type = "points"</code> and argument <code>points_threshold</code> (default)
</p>
<p>This method weighs <em>all</em> antimicrobial drugs available in the data set. Any difference from I to S or R (or vice versa) counts as <code>0.5</code> points, a difference from S to R (or vice versa) counts as <code>1</code> point. When the sum of points exceeds <code>points_threshold</code>, which defaults to <code>2</code>, an isolate will be selected as a first weighted isolate.
</p>
<p>All antimicrobials are internally selected using the <code><a href="#topic+all_antimicrobials">all_antimicrobials()</a></code> function. The output of this function does not need to be passed to the <code><a href="#topic+first_isolate">first_isolate()</a></code> function.
</p>
</li>
<li><p> Using <code>type = "keyantimicrobials"</code> and argument <code>ignore_I</code>
</p>
<p>This method only weighs specific antimicrobial drugs, called <em>key antimicrobials</em>. Any difference from S to R (or vice versa) in these key antimicrobials will select an isolate as a first weighted isolate. With <code>ignore_I = FALSE</code>, also differences from I to S or R (or vice versa) will lead to this.
</p>
<p>Key antimicrobials are internally selected using the <code><a href="#topic+key_antimicrobials">key_antimicrobials()</a></code> function, but can also be added manually as a variable to the data and set in the <code>col_keyantimicrobials</code> argument. Another option is to pass the output of the <code><a href="#topic+key_antimicrobials">key_antimicrobials()</a></code> function directly to the <code>col_keyantimicrobials</code> argument.
</p>
</li></ol>

<p>The default method is phenotype-based (using <code>type = "points"</code>) and episode-based (using <code>episode_days = 365</code>). This makes sure that every genus-species combination is selected per patient once per year, while taking into account all antimicrobial test results. If no antimicrobial test results are available in the data set, only the episode-based method is applied at default.
</p>



<h3>Value</h3>

<p>A <a href="base.html#topic+logical">logical</a> vector
</p>


<h3>Source</h3>

<p>Methodology of this function is strictly based on:
</p>

<ul>
<li> <p><strong>M39 Analysis and Presentation of Cumulative Antimicrobial Susceptibility Test Data, 5th Edition</strong>, 2022, <em>Clinical and Laboratory Standards Institute (CLSI)</em>. <a href="https://clsi.org/standards/products/microbiology/documents/m39/">https://clsi.org/standards/products/microbiology/documents/m39/</a>.
</p>
</li>
<li><p> Hindler JF and Stelling J (2007). <strong>Analysis and Presentation of Cumulative Antibiograms: A New Consensus Guideline from the Clinical and Laboratory Standards Institute.</strong> Clinical Infectious Diseases, 44(6), 867-873. <a href="https://doi.org/10.1086/511864">doi:10.1086/511864</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+key_antimicrobials">key_antimicrobials()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># `example_isolates` is a data set available in the AMR package.
# See ?example_isolates.

example_isolates[first_isolate(info = TRUE), ]

# get all first Gram-negatives
example_isolates[which(first_isolate(info = FALSE) &amp; mo_is_gram_negative()), ]

if (require("dplyr")) {
  # filter on first isolates using dplyr:
  example_isolates %&gt;%
    filter(first_isolate(info = TRUE))
}
if (require("dplyr")) {
  # short-hand version:
  example_isolates %&gt;%
    filter_first_isolate(info = FALSE)
}
if (require("dplyr")) {
  # flag the first isolates per group:
  example_isolates %&gt;%
    group_by(ward) %&gt;%
    mutate(first = first_isolate(info = TRUE)) %&gt;%
    select(ward, date, patient, mo, first)
}

</code></pre>

<hr>
<h2 id='g.test'><em>G</em>-test for Count Data</h2><span id='topic+g.test'></span>

<h3>Description</h3>

<p><code><a href="#topic+g.test">g.test()</a></code> performs chi-squared contingency table tests and goodness-of-fit tests, just like <code><a href="stats.html#topic+chisq.test">chisq.test()</a></code> but is more reliable (1). A <em>G</em>-test can be used to see whether the number of observations in each category fits a theoretical expectation (called a <strong><em>G</em>-test of goodness-of-fit</strong>), or to see whether the proportions of one variable are different for different values of the other variable (called a <strong><em>G</em>-test of independence</strong>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>g.test(x, y = NULL, p = rep(1/length(x), length(x)), rescale.p = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="g.test_+3A_x">x</code></td>
<td>
<p>a numeric vector or matrix. <code>x</code> and <code>y</code> can also
both be factors.</p>
</td></tr>
<tr><td><code id="g.test_+3A_y">y</code></td>
<td>
<p>a numeric vector; ignored if <code>x</code> is a matrix.  If
<code>x</code> is a factor, <code>y</code> should be a factor of the same length.</p>
</td></tr>
<tr><td><code id="g.test_+3A_p">p</code></td>
<td>
<p>a vector of probabilities of the same length as <code>x</code>.
An error is given if any entry of <code>p</code> is negative.</p>
</td></tr>
<tr><td><code id="g.test_+3A_rescale.p">rescale.p</code></td>
<td>
<p>a logical scalar; if TRUE then <code>p</code> is rescaled
(if necessary) to sum to 1.  If <code>rescale.p</code> is FALSE, and
<code>p</code> does not sum to 1, an error is given.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>x</code> is a <a href="base.html#topic+matrix">matrix</a> with one row or column, or if <code>x</code> is a vector and <code>y</code> is not given, then a <em>goodness-of-fit test</em> is performed (<code>x</code> is treated as a one-dimensional contingency table). The entries of <code>x</code> must be non-negative integers. In this case, the hypothesis tested is whether the population probabilities equal those in <code>p</code>, or are all equal if <code>p</code> is not given.
</p>
<p>If <code>x</code> is a <a href="base.html#topic+matrix">matrix</a> with at least two rows and columns, it is taken as a two-dimensional contingency table: the entries of <code>x</code> must be non-negative integers.  Otherwise, <code>x</code> and <code>y</code> must be vectors or factors of the same length; cases with missing values are removed, the objects are coerced to factors, and the contingency table is computed from these.  Then Pearson's chi-squared test is performed of the null hypothesis that the joint distribution of the cell counts in a 2-dimensional contingency table is the product of the row and column marginals.
</p>
<p>The p-value is computed from the asymptotic chi-squared distribution of the test statistic.
</p>
<p>In the contingency table case simulation is done by random sampling from the set of all contingency tables with given marginals, and works only if the marginals are strictly positive. Note that this is not the usual sampling situation assumed for a chi-squared test (such as the <em>G</em>-test) but rather that for Fisher's exact test.
</p>
<p>In the goodness-of-fit case simulation is done by random sampling from the discrete distribution specified by <code>p</code>, each sample being of size <code>n = sum(x)</code>. This simulation is done in <span class="rlang"><b>R</b></span> and may be slow.
</p>


<h4><em>G</em>-test Of Goodness-of-Fit (Likelihood Ratio Test)</h4>

<p>Use the <em>G</em>-test of goodness-of-fit when you have one nominal variable with two or more values (such as male and female, or red, pink and white flowers). You compare the observed counts of numbers of observations in each category with the expected counts, which you calculate using some kind of theoretical expectation (such as a 1:1 sex ratio or a 1:2:1 ratio in a genetic cross).
</p>
<p>If the expected number of observations in any category is too small, the <em>G</em>-test may give inaccurate results, and you should use an exact test instead (<code><a href="stats.html#topic+fisher.test">fisher.test()</a></code>).
</p>
<p>The <em>G</em>-test of goodness-of-fit is an alternative to the chi-square test of goodness-of-fit (<code><a href="stats.html#topic+chisq.test">chisq.test()</a></code>); each of these tests has some advantages and some disadvantages, and the results of the two tests are usually very similar.
</p>



<h4><em>G</em>-test of Independence</h4>

<p>Use the <em>G</em>-test of independence when you have two nominal variables, each with two or more possible values. You want to know whether the proportions for one variable are different among values of the other variable.
</p>
<p>It is also possible to do a <em>G</em>-test of independence with more than two nominal variables. For example, Jackson et al. (2013) also had data for children under 3, so you could do an analysis of old vs. young, thigh vs. arm, and reaction vs. no reaction, all analyzed together.
</p>
<p>Fisher's exact test (<code><a href="stats.html#topic+fisher.test">fisher.test()</a></code>) is an <strong>exact</strong> test, where the <em>G</em>-test is still only an <strong>approximation</strong>. For any 2x2 table, Fisher's Exact test may be slower but will still run in seconds, even if the sum of your observations is multiple millions.
</p>
<p>The <em>G</em>-test of independence is an alternative to the chi-square test of independence (<code><a href="stats.html#topic+chisq.test">chisq.test()</a></code>), and they will give approximately the same results.
</p>



<h4>How the Test Works</h4>

<p>Unlike the exact test of goodness-of-fit (<code><a href="stats.html#topic+fisher.test">fisher.test()</a></code>), the <em>G</em>-test does not directly calculate the probability of obtaining the observed results or something more extreme. Instead, like almost all statistical tests, the <em>G</em>-test has an intermediate step; it uses the data to calculate a test statistic that measures how far the observed data are from the null expectation. You then use a mathematical relationship, in this case the chi-square distribution, to estimate the probability of obtaining that value of the test statistic.
</p>
<p>The <em>G</em>-test uses the log of the ratio of two likelihoods as the test statistic, which is why it is also called a likelihood ratio test or log-likelihood ratio test. The formula to calculate a <em>G</em>-statistic is:
</p>
<p><code class="reqn">G = 2 * sum(x * log(x / E))</code>
</p>
<p>where <code>E</code> are the expected values. Since this is chi-square distributed, the p value can be calculated in <span class="rlang"><b>R</b></span> with:
</p>
<div class="sourceCode"><pre>p &lt;- stats::pchisq(G, df, lower.tail = FALSE)
</pre></div>
<p>where <code>df</code> are the degrees of freedom.
</p>
<p>If there are more than two categories and you want to find out which ones are significantly different from their null expectation, you can use the same method of testing each category vs. the sum of all categories, with the Bonferroni correction. You use <em>G</em>-tests for each category, of course.
</p>



<h3>Value</h3>

<p>A list with class <code>"htest"</code> containing the following
components:
</p>
<table>
<tr><td><code>statistic</code></td>
<td>
<p>the value the chi-squared test statistic.</p>
</td></tr>
<tr><td><code>parameter</code></td>
<td>
<p>the degrees of freedom of the approximate
chi-squared distribution of the test statistic, <code>NA</code> if the
p-value is computed by Monte Carlo simulation.</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>the p-value for the test.</p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p>a character string indicating the type of test
performed, and whether Monte Carlo simulation or continuity
correction was used.</p>
</td></tr>
<tr><td><code>data.name</code></td>
<td>
<p>a character string giving the name(s) of the data.</p>
</td></tr>
<tr><td><code>observed</code></td>
<td>
<p>the observed counts.</p>
</td></tr>
<tr><td><code>expected</code></td>
<td>
<p>the expected counts under the null hypothesis.</p>
</td></tr>
<tr><td><code>residuals</code></td>
<td>
<p>the Pearson residuals,
<code>(observed - expected) / sqrt(expected)</code>.</p>
</td></tr>
<tr><td><code>stdres</code></td>
<td>
<p>standardized residuals,
<code>(observed - expected) / sqrt(V)</code>, where <code>V</code> is the residual cell variance (Agresti, 2007,
section 2.4.5 for the case where <code>x</code> is a matrix, <code>n * p * (1 - p)</code> otherwise).</p>
</td></tr>
</table>


<h3>Source</h3>

<p>The code for this function is identical to that of <code><a href="stats.html#topic+chisq.test">chisq.test()</a></code>, except that:
</p>

<ul>
<li><p> The calculation of the statistic was changed to <code class="reqn">2 * sum(x * log(x / E))</code>
</p>
</li>
<li><p> Yates' continuity correction was removed as it does not apply to a <em>G</em>-test
</p>
</li>
<li><p> The possibility to simulate p values with <code>simulate.p.value</code> was removed
</p>
</li></ul>



<h3>References</h3>


<ol>
<li><p> McDonald, J.H. 2014. <strong>Handbook of Biological Statistics (3rd ed.)</strong>. Sparky House Publishing, Baltimore, Maryland. <a href="http://www.biostathandbook.com/gtestgof.html">http://www.biostathandbook.com/gtestgof.html</a>.
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="stats.html#topic+chisq.test">chisq.test()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># = EXAMPLE 1 =
# Shivrain et al. (2006) crossed clearfield rice (which are resistant
# to the herbicide imazethapyr) with red rice (which are susceptible to
# imazethapyr). They then crossed the hybrid offspring and examined the
# F2 generation, where they found 772 resistant plants, 1611 moderately
# resistant plants, and 737 susceptible plants. If resistance is controlled
# by a single gene with two co-dominant alleles, you would expect a 1:2:1
# ratio.

x &lt;- c(772, 1611, 737)
g.test(x, p = c(1, 2, 1) / 4)

# There is no significant difference from a 1:2:1 ratio.
# Meaning: resistance controlled by a single gene with two co-dominant
# alleles, is plausible.


# = EXAMPLE 2 =
# Red crossbills (Loxia curvirostra) have the tip of the upper bill either
# right or left of the lower bill, which helps them extract seeds from pine
# cones. Some have hypothesized that frequency-dependent selection would
# keep the number of right and left-billed birds at a 1:1 ratio. Groth (1992)
# observed 1752 right-billed and 1895 left-billed crossbills.

x &lt;- c(1752, 1895)
g.test(x)

# There is a significant difference from a 1:1 ratio.
# Meaning: there are significantly more left-billed birds.
</code></pre>

<hr>
<h2 id='get_episode'>Determine Clinical or Epidemic Episodes</h2><span id='topic+get_episode'></span><span id='topic+is_new_episode'></span>

<h3>Description</h3>

<p>These functions determine which items in a vector can be considered (the start of) a new episode. This can be used to determine clinical episodes for any epidemiological analysis. The <code><a href="#topic+get_episode">get_episode()</a></code> function returns the index number of the episode per group, while the <code><a href="#topic+is_new_episode">is_new_episode()</a></code> function returns <code>TRUE</code> for every new <code><a href="#topic+get_episode">get_episode()</a></code> index. Both absolute and relative episode determination are supported.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_episode(x, episode_days = NULL, case_free_days = NULL, ...)

is_new_episode(x, episode_days = NULL, case_free_days = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_episode_+3A_x">x</code></td>
<td>
<p>vector of dates (class <code>Date</code> or <code>POSIXt</code>), will be sorted internally to determine episodes</p>
</td></tr>
<tr><td><code id="get_episode_+3A_episode_days">episode_days</code></td>
<td>
<p>episode length in days to specify the time period after which a new episode begins, can also be less than a day or <code>Inf</code>, see <em>Details</em></p>
</td></tr>
<tr><td><code id="get_episode_+3A_case_free_days">case_free_days</code></td>
<td>
<p>(inter-epidemic) interval length in days after which a new episode will start, can also be less than a day or <code>Inf</code>, see <em>Details</em></p>
</td></tr>
<tr><td><code id="get_episode_+3A_...">...</code></td>
<td>
<p>ignored, only in place to allow future extensions</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Episodes can be determined in two ways: absolute and relative.
</p>

<ol>
<li><p> Absolute
</p>
<p>This method uses <code>episode_days</code> to define an episode length in days, after which a new episode will start. A common use case in AMR data analysis is microbial epidemiology: episodes of <em>S. aureus</em> bacteraemia in ICU patients for example. The episode length could then be 30 days, so that new <em>S. aureus</em> isolates after an ICU episode of 30 days will be considered a different (or new) episode.
</p>
<p>Thus, this method counts <strong>since the start of the previous episode</strong>.
</p>
</li>
<li><p> Relative
</p>
<p>This method uses <code>case_free_days</code> to quantify the duration of case-free days (the inter-epidemic interval), after which a new episode will start. A common use case is infectious disease epidemiology: episodes of norovirus outbreaks in a hospital for example. The case-free period could then be 14 days, so that new norovirus cases after that time will be considered a different (or new) episode.
</p>
<p>Thus, this methods counts <strong>since the last case in the previous episode</strong>.
</p>
</li></ol>

<p>In a table:</p>

<table>
<tr>
 <td style="text-align: center;">
   Date </td><td style="text-align: center;"> Using <code>episode_days = 7</code> </td><td style="text-align: center;"> Using <code>case_free_days = 7</code> </td>
</tr>
<tr>
 <td style="text-align: center;">
   2023-01-01 </td><td style="text-align: center;"> 1 </td><td style="text-align: center;"> 1 </td>
</tr>
<tr>
 <td style="text-align: center;">
   2023-01-02 </td><td style="text-align: center;"> 1 </td><td style="text-align: center;"> 1 </td>
</tr>
<tr>
 <td style="text-align: center;">
   2023-01-05 </td><td style="text-align: center;"> 1 </td><td style="text-align: center;"> 1 </td>
</tr>
<tr>
 <td style="text-align: center;">
   2023-01-08 </td><td style="text-align: center;"> 2** </td><td style="text-align: center;"> 1 </td>
</tr>
<tr>
 <td style="text-align: center;">
   2023-02-21 </td><td style="text-align: center;"> 3 </td><td style="text-align: center;"> 2*** </td>
</tr>
<tr>
 <td style="text-align: center;">
   2023-02-22 </td><td style="text-align: center;"> 3 </td><td style="text-align: center;"> 2 </td>
</tr>
<tr>
 <td style="text-align: center;">
   2023-02-23 </td><td style="text-align: center;"> 3 </td><td style="text-align: center;"> 2 </td>
</tr>
<tr>
 <td style="text-align: center;">
   2023-02-24 </td><td style="text-align: center;"> 3 </td><td style="text-align: center;"> 2 </td>
</tr>
<tr>
 <td style="text-align: center;">
   2023-03-01 </td><td style="text-align: center;"> 4 </td><td style="text-align: center;"> 2 </td>
</tr>
<tr>
 <td style="text-align: center;">
</td>
</tr>

</table>

<p>** This marks the start of a new episode, because 8 January 2023 is more than 7 days since the start of the previous episode (1 January 2023). <br />
*** This marks the start of a new episode, because 21 January 2023 is more than 7 days since the last case in the previous episode (8 January 2023).
</p>
<p>Either <code>episode_days</code> or <code>case_free_days</code> must be provided in the function.
</p>


<h4>Difference between <code>get_episode()</code> and <code>is_new_episode()</code></h4>

<p>The <code><a href="#topic+get_episode">get_episode()</a></code> function returns the index number of the episode, so all cases/patients/isolates in the first episode will have the number 1, all cases/patients/isolates in the second episode will have the number 2, etc.
</p>
<p>The <code><a href="#topic+is_new_episode">is_new_episode()</a></code> function on the other hand, returns <code>TRUE</code> for every new <code><a href="#topic+get_episode">get_episode()</a></code> index.
</p>
<p>To specify, when setting <code>episode_days = 365</code> (using method 1 as explained above), this is how the two functions differ:</p>

<table>
<tr>
 <td style="text-align: center;">
   patient </td><td style="text-align: center;"> date </td><td style="text-align: center;"> <code>get_episode()</code> </td><td style="text-align: center;"> <code>is_new_episode()</code> </td>
</tr>
<tr>
 <td style="text-align: center;">
   A </td><td style="text-align: center;"> 2019-01-01 </td><td style="text-align: center;"> 1 </td><td style="text-align: center;"> TRUE </td>
</tr>
<tr>
 <td style="text-align: center;">
   A </td><td style="text-align: center;"> 2019-03-01 </td><td style="text-align: center;"> 1 </td><td style="text-align: center;"> FALSE </td>
</tr>
<tr>
 <td style="text-align: center;">
   A </td><td style="text-align: center;"> 2021-01-01 </td><td style="text-align: center;"> 2 </td><td style="text-align: center;"> TRUE </td>
</tr>
<tr>
 <td style="text-align: center;">
   B </td><td style="text-align: center;"> 2008-01-01 </td><td style="text-align: center;"> 1 </td><td style="text-align: center;"> TRUE </td>
</tr>
<tr>
 <td style="text-align: center;">
   B </td><td style="text-align: center;"> 2008-01-01 </td><td style="text-align: center;"> 1 </td><td style="text-align: center;"> FALSE </td>
</tr>
<tr>
 <td style="text-align: center;">
   C </td><td style="text-align: center;"> 2020-01-01 </td><td style="text-align: center;"> 1 </td><td style="text-align: center;"> TRUE </td>
</tr>
<tr>
 <td style="text-align: center;">
</td>
</tr>

</table>




<h4>Other</h4>

<p>The <code><a href="#topic+first_isolate">first_isolate()</a></code> function is a wrapper around the <code><a href="#topic+is_new_episode">is_new_episode()</a></code> function, but is more efficient for data sets containing microorganism codes or names and allows for different isolate selection methods.
</p>
<p>The <code>dplyr</code> package is not required for these functions to work, but these episode functions do support <a href="dplyr.html#topic+group_by">variable grouping</a> and work conveniently inside <code>dplyr</code> verbs such as <code><a href="dplyr.html#topic+filter">filter()</a></code>, <code><a href="dplyr.html#topic+mutate">mutate()</a></code> and <code><a href="dplyr.html#topic+summarise">summarise()</a></code>.
</p>



<h3>Value</h3>


<ul>
<li> <p><code><a href="#topic+get_episode">get_episode()</a></code>: an <a href="base.html#topic+integer">integer</a> vector
</p>
</li>
<li> <p><code><a href="#topic+is_new_episode">is_new_episode()</a></code>: a <a href="base.html#topic+logical">logical</a> vector
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+first_isolate">first_isolate()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># difference between absolute and relative determination of episodes:
x &lt;- data.frame(dates = as.Date(c(
  "2021-01-01",
  "2021-01-02",
  "2021-01-05",
  "2021-01-08",
  "2021-02-21",
  "2021-02-22",
  "2021-02-23",
  "2021-02-24",
  "2021-03-01",
  "2021-03-01"
)))
x$absolute &lt;- get_episode(x$dates, episode_days = 7)
x$relative &lt;- get_episode(x$dates, case_free_days = 7)
x


# `example_isolates` is a data set available in the AMR package.
# See ?example_isolates
df &lt;- example_isolates[sample(seq_len(2000), size = 100), ]

get_episode(df$date, episode_days = 60) # indices
is_new_episode(df$date, episode_days = 60) # TRUE/FALSE

# filter on results from the third 60-day episode only, using base R
df[which(get_episode(df$date, 60) == 3), ]

# the functions also work for less than a day, e.g. to include one per hour:
get_episode(
  c(
    Sys.time(),
    Sys.time() + 60 * 60
  ),
  episode_days = 1 / 24
)


if (require("dplyr")) {
  # is_new_episode() can also be used in dplyr verbs to determine patient
  # episodes based on any (combination of) grouping variables:
  df %&gt;%
    mutate(condition = sample(
      x = c("A", "B", "C"),
      size = 100,
      replace = TRUE
    )) %&gt;%
    group_by(patient, condition) %&gt;%
    mutate(new_episode = is_new_episode(date, 365)) %&gt;%
    select(patient, date, condition, new_episode) %&gt;%
    arrange(patient, condition, date)
}

if (require("dplyr")) {
  df %&gt;%
    group_by(ward, patient) %&gt;%
    transmute(date,
      patient,
      new_index = get_episode(date, 60),
      new_logical = is_new_episode(date, 60)
    ) %&gt;%
    arrange(patient, ward, date)
}

if (require("dplyr")) {
  df %&gt;%
    group_by(ward) %&gt;%
    summarise(
      n_patients = n_distinct(patient),
      n_episodes_365 = sum(is_new_episode(date, episode_days = 365)),
      n_episodes_60 = sum(is_new_episode(date, episode_days = 60)),
      n_episodes_30 = sum(is_new_episode(date, episode_days = 30))
    )
}

# grouping on patients and microorganisms leads to the same
# results as first_isolate() when using 'episode-based':
if (require("dplyr")) {
  x &lt;- df %&gt;%
    filter_first_isolate(
      include_unknown = TRUE,
      method = "episode-based"
    )

  y &lt;- df %&gt;%
    group_by(patient, mo) %&gt;%
    filter(is_new_episode(date, 365)) %&gt;%
    ungroup()

  identical(x, y)
}

# but is_new_episode() has a lot more flexibility than first_isolate(),
# since you can now group on anything that seems relevant:
if (require("dplyr")) {
  df %&gt;%
    group_by(patient, mo, ward) %&gt;%
    mutate(flag_episode = is_new_episode(date, 365)) %&gt;%
    select(group_vars(.), flag_episode)
}

</code></pre>

<hr>
<h2 id='ggplot_pca'>PCA Biplot with <code>ggplot2</code></h2><span id='topic+ggplot_pca'></span>

<h3>Description</h3>

<p>Produces a <code>ggplot2</code> variant of a so-called <a href="https://en.wikipedia.org/wiki/Biplot">biplot</a> for PCA (principal component analysis), but is more flexible and more appealing than the base <span class="rlang"><b>R</b></span> <code><a href="stats.html#topic+biplot">biplot()</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_pca(
  x,
  choices = 1:2,
  scale = 1,
  pc.biplot = TRUE,
  labels = NULL,
  labels_textsize = 3,
  labels_text_placement = 1.5,
  groups = NULL,
  ellipse = TRUE,
  ellipse_prob = 0.68,
  ellipse_size = 0.5,
  ellipse_alpha = 0.5,
  points_size = 2,
  points_alpha = 0.25,
  arrows = TRUE,
  arrows_colour = "darkblue",
  arrows_size = 0.5,
  arrows_textsize = 3,
  arrows_textangled = TRUE,
  arrows_alpha = 0.75,
  base_textsize = 10,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggplot_pca_+3A_x">x</code></td>
<td>
<p>an object returned by <code><a href="#topic+pca">pca()</a></code>, <code><a href="stats.html#topic+prcomp">prcomp()</a></code> or <code><a href="stats.html#topic+princomp">princomp()</a></code></p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_choices">choices</code></td>
<td>

<p>length 2 vector specifying the components to plot. Only the default
is a biplot in the strict sense.
</p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_scale">scale</code></td>
<td>

<p>The variables are scaled by <code>lambda ^ scale</code> and the
observations are scaled by <code>lambda ^ (1-scale)</code> where
<code>lambda</code> are the singular values as computed by
<code><a href="stats.html#topic+princomp">princomp</a></code>. Normally <code>0 &lt;= scale &lt;= 1</code>, and a warning
will be issued if the specified <code>scale</code> is outside this range.
</p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_pc.biplot">pc.biplot</code></td>
<td>

<p>If true, use what Gabriel (1971) refers to as a &quot;principal component
biplot&quot;, with <code>lambda = 1</code> and observations scaled up by sqrt(n) and
variables scaled down by sqrt(n).  Then inner products between
variables approximate covariances and distances between observations
approximate Mahalanobis distance.
</p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_labels">labels</code></td>
<td>
<p>an optional vector of labels for the observations. If set, the labels will be placed below their respective points. When using the <code><a href="#topic+pca">pca()</a></code> function as input for <code>x</code>, this will be determined automatically based on the attribute <code>non_numeric_cols</code>, see <code><a href="#topic+pca">pca()</a></code>.</p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_labels_textsize">labels_textsize</code></td>
<td>
<p>the size of the text used for the labels</p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_labels_text_placement">labels_text_placement</code></td>
<td>
<p>adjustment factor the placement of the variable names (<code style="white-space: pre;">&#8288;&gt;=1&#8288;</code> means further away from the arrow head)</p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_groups">groups</code></td>
<td>
<p>an optional vector of groups for the labels, with the same length as <code>labels</code>. If set, the points and labels will be coloured according to these groups. When using the <code><a href="#topic+pca">pca()</a></code> function as input for <code>x</code>, this will be determined automatically based on the attribute <code>non_numeric_cols</code>, see <code><a href="#topic+pca">pca()</a></code>.</p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_ellipse">ellipse</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether a normal data ellipse should be drawn for each group (set with <code>groups</code>)</p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_ellipse_prob">ellipse_prob</code></td>
<td>
<p>statistical size of the ellipse in normal probability</p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_ellipse_size">ellipse_size</code></td>
<td>
<p>the size of the ellipse line</p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_ellipse_alpha">ellipse_alpha</code></td>
<td>
<p>the alpha (transparency) of the ellipse line</p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_points_size">points_size</code></td>
<td>
<p>the size of the points</p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_points_alpha">points_alpha</code></td>
<td>
<p>the alpha (transparency) of the points</p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_arrows">arrows</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether arrows should be drawn</p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_arrows_colour">arrows_colour</code></td>
<td>
<p>the colour of the arrow and their text</p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_arrows_size">arrows_size</code></td>
<td>
<p>the size (thickness) of the arrow lines</p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_arrows_textsize">arrows_textsize</code></td>
<td>
<p>the size of the text at the end of the arrows</p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_arrows_textangled">arrows_textangled</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> whether the text at the end of the arrows should be angled</p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_arrows_alpha">arrows_alpha</code></td>
<td>
<p>the alpha (transparency) of the arrows and their text</p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_base_textsize">base_textsize</code></td>
<td>
<p>the text size for all plot elements except the labels and arrows</p>
</td></tr>
<tr><td><code id="ggplot_pca_+3A_...">...</code></td>
<td>
<p>arguments passed on to functions</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The colours for labels and points can be changed by adding another scale layer for colour, such as <code>scale_colour_viridis_d()</code> and <code>scale_colour_brewer()</code>.
</p>


<h3>Source</h3>

<p>The <code><a href="#topic+ggplot_pca">ggplot_pca()</a></code> function is based on the <code>ggbiplot()</code> function from the <code>ggbiplot</code> package by Vince Vu, as found on GitHub: <a href="https://github.com/vqv/ggbiplot">https://github.com/vqv/ggbiplot</a> (retrieved: 2 March 2020, their latest commit: <a href="https://github.com/vqv/ggbiplot/commit/7325e880485bea4c07465a0304c470608fffb5d9"><code>7325e88</code></a>; 12 February 2015).
</p>
<p>As per their GPL-2 licence that demands documentation of code changes, the changes made based on the source code were:
</p>

<ol>
<li><p> Rewritten code to remove the dependency on packages <code>plyr</code>, <code>scales</code> and <code>grid</code>
</p>
</li>
<li><p> Parametrised more options, like arrow and ellipse settings
</p>
</li>
<li><p> Hardened all input possibilities by defining the exact type of user input for every argument
</p>
</li>
<li><p> Added total amount of explained variance as a caption in the plot
</p>
</li>
<li><p> Cleaned all syntax based on the <code>lintr</code> package, fixed grammatical errors and added integrity checks
</p>
</li>
<li><p> Updated documentation
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'># `example_isolates` is a data set available in the AMR package.
# See ?example_isolates.


if (require("dplyr")) {
  # calculate the resistance per group first
  resistance_data &lt;- example_isolates %&gt;%
    group_by(
      order = mo_order(mo), # group on anything, like order
      genus = mo_genus(mo)
    ) %&gt;% #   and genus as we do here;
    filter(n() &gt;= 30) %&gt;% # filter on only 30 results per group
    summarise_if(is.sir, resistance) # then get resistance of all drugs

  # now conduct PCA for certain antimicrobial drugs
  pca_result &lt;- resistance_data %&gt;%
    pca(AMC, CXM, CTX, CAZ, GEN, TOB, TMP, SXT)

  summary(pca_result)

  # old base R plotting method:
  biplot(pca_result)

  # new ggplot2 plotting method using this package:
  if (require("ggplot2")) {
    ggplot_pca(pca_result)

    # still extendible with any ggplot2 function
    ggplot_pca(pca_result) +
      scale_colour_viridis_d() +
      labs(title = "Title here")
  }
}

</code></pre>

<hr>
<h2 id='ggplot_sir'>AMR Plots with <code>ggplot2</code></h2><span id='topic+ggplot_sir'></span><span id='topic+geom_sir'></span><span id='topic+facet_sir'></span><span id='topic+scale_y_percent'></span><span id='topic+scale_sir_colours'></span><span id='topic+theme_sir'></span><span id='topic+labels_sir_count'></span>

<h3>Description</h3>

<p>Use these functions to create bar plots for AMR data analysis. All functions rely on <a href="ggplot2.html#topic+ggplot">ggplot2</a> functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_sir(
  data,
  position = NULL,
  x = "antibiotic",
  fill = "interpretation",
  facet = NULL,
  breaks = seq(0, 1, 0.1),
  limits = NULL,
  translate_ab = "name",
  combine_SI = TRUE,
  minimum = 30,
  language = get_AMR_locale(),
  nrow = NULL,
  colours = c(S = "#3CAEA3", SI = "#3CAEA3", I = "#F6D55C", IR = "#ED553B", R =
    "#ED553B"),
  datalabels = TRUE,
  datalabels.size = 2.5,
  datalabels.colour = "grey15",
  title = NULL,
  subtitle = NULL,
  caption = NULL,
  x.title = "Antimicrobial",
  y.title = "Proportion",
  ...
)

geom_sir(
  position = NULL,
  x = c("antibiotic", "interpretation"),
  fill = "interpretation",
  translate_ab = "name",
  minimum = 30,
  language = get_AMR_locale(),
  combine_SI = TRUE,
  ...
)

facet_sir(facet = c("interpretation", "antibiotic"), nrow = NULL)

scale_y_percent(breaks = seq(0, 1, 0.1), limits = NULL)

scale_sir_colours(..., aesthetics = "fill")

theme_sir()

labels_sir_count(
  position = NULL,
  x = "antibiotic",
  translate_ab = "name",
  minimum = 30,
  language = get_AMR_locale(),
  combine_SI = TRUE,
  datalabels.size = 3,
  datalabels.colour = "grey15"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggplot_sir_+3A_data">data</code></td>
<td>
<p>a <a href="base.html#topic+data.frame">data.frame</a> with column(s) of class <code><a href="#topic+sir">sir</a></code> (see <code><a href="#topic+as.sir">as.sir()</a></code>)</p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_position">position</code></td>
<td>
<p>position adjustment of bars, either <code>"fill"</code>, <code>"stack"</code> or <code>"dodge"</code></p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_x">x</code></td>
<td>
<p>variable to show on x axis, either <code>"antibiotic"</code> (default) or <code>"interpretation"</code> or a grouping variable</p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_fill">fill</code></td>
<td>
<p>variable to categorise using the plots legend, either <code>"antibiotic"</code> (default) or <code>"interpretation"</code> or a grouping variable</p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_facet">facet</code></td>
<td>
<p>variable to split plots by, either <code>"interpretation"</code> (default) or <code>"antibiotic"</code> or a grouping variable</p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_breaks">breaks</code></td>
<td>
<p>a <a href="base.html#topic+numeric">numeric</a> vector of positions</p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_limits">limits</code></td>
<td>
<p>a <a href="base.html#topic+numeric">numeric</a> vector of length two providing limits of the scale, use <code>NA</code> to refer to the existing minimum or maximum</p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_translate_ab">translate_ab</code></td>
<td>
<p>a column name of the <a href="#topic+antibiotics">antibiotics</a> data set to translate the antibiotic abbreviations to, using <code><a href="#topic+ab_property">ab_property()</a></code></p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_combine_si">combine_SI</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether all values of S and I must be merged into one, so the output only consists of S+I vs. R (susceptible vs. resistant) - the default is <code>TRUE</code></p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_minimum">minimum</code></td>
<td>
<p>the minimum allowed number of available (tested) isolates. Any isolate count lower than <code>minimum</code> will return <code>NA</code> with a warning. The default number of <code>30</code> isolates is advised by the Clinical and Laboratory Standards Institute (CLSI) as best practice, see <em>Source</em>.</p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_language">language</code></td>
<td>
<p>language of the returned text - the default is the current system language (see <code><a href="#topic+get_AMR_locale">get_AMR_locale()</a></code>) and can also be set with the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_locale</a></code>. Use <code>language = NULL</code> or <code>language = ""</code> to prevent translation.</p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_nrow">nrow</code></td>
<td>
<p>(when using <code>facet</code>) number of rows</p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_colours">colours</code></td>
<td>
<p>a named vactor with colour to be used for filling. The default colours are colour-blind friendly.</p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_datalabels">datalabels</code></td>
<td>
<p>show datalabels using <code><a href="#topic+labels_sir_count">labels_sir_count()</a></code></p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_datalabels.size">datalabels.size</code></td>
<td>
<p>size of the datalabels</p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_datalabels.colour">datalabels.colour</code></td>
<td>
<p>colour of the datalabels</p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_title">title</code></td>
<td>
<p>text to show as title of the plot</p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_subtitle">subtitle</code></td>
<td>
<p>text to show as subtitle of the plot</p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_caption">caption</code></td>
<td>
<p>text to show as caption of the plot</p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_x.title">x.title</code></td>
<td>
<p>text to show as x axis description</p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_y.title">y.title</code></td>
<td>
<p>text to show as y axis description</p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_...">...</code></td>
<td>
<p>other arguments passed on to <code><a href="#topic+geom_sir">geom_sir()</a></code> or, in case of <code><a href="#topic+scale_sir_colours">scale_sir_colours()</a></code>, named values to set colours. The default colours are colour-blind friendly, while maintaining the convention that e.g. 'susceptible' should be green and 'resistant' should be red. See <em>Examples</em>.</p>
</td></tr>
<tr><td><code id="ggplot_sir_+3A_aesthetics">aesthetics</code></td>
<td>
<p>aesthetics to apply the colours to - the default is &quot;fill&quot; but can also be (a combination of) &quot;alpha&quot;, &quot;colour&quot;, &quot;fill&quot;, &quot;linetype&quot;, &quot;shape&quot; or &quot;size&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>At default, the names of antibiotics will be shown on the plots using <code><a href="#topic+ab_name">ab_name()</a></code>. This can be set with the <code>translate_ab</code> argument. See <code><a href="#topic+count_df">count_df()</a></code>.
</p>


<h4>The Functions</h4>

<p><code><a href="#topic+geom_sir">geom_sir()</a></code> will take any variable from the data that has an <code><a href="#topic+sir">sir</a></code> class (created with <code><a href="#topic+as.sir">as.sir()</a></code>) using <code><a href="#topic+sir_df">sir_df()</a></code> and will plot bars with the percentage S, I, and R. The default behaviour is to have the bars stacked and to have the different antibiotics on the x axis.
</p>
<p><code><a href="#topic+facet_sir">facet_sir()</a></code> creates 2d plots (at default based on S/I/R) using <code><a href="ggplot2.html#topic+facet_wrap">ggplot2::facet_wrap()</a></code>.
</p>
<p><code><a href="#topic+scale_y_percent">scale_y_percent()</a></code> transforms the y axis to a 0 to 100% range using <code><a href="ggplot2.html#topic+scale_continuous">ggplot2::scale_y_continuous()</a></code>.
</p>
<p><code><a href="#topic+scale_sir_colours">scale_sir_colours()</a></code> sets colours to the bars (green for S, yellow for I, and red for R). with multilingual support. The default colours are colour-blind friendly, while maintaining the convention that e.g. 'susceptible' should be green and 'resistant' should be red.
</p>
<p><code><a href="#topic+theme_sir">theme_sir()</a></code> is a [ggplot2 theme][<code><a href="ggplot2.html#topic+theme">ggplot2::theme()</a></code> with minimal distraction.
</p>
<p><code><a href="#topic+labels_sir_count">labels_sir_count()</a></code> print datalabels on the bars with percentage and amount of isolates using <code><a href="ggplot2.html#topic+geom_text">ggplot2::geom_text()</a></code>.
</p>
<p><code><a href="#topic+ggplot_sir">ggplot_sir()</a></code> is a wrapper around all above functions that uses data as first input. This makes it possible to use this function after a pipe (<code style="white-space: pre;">&#8288;%&gt;%&#8288;</code>). See <em>Examples</em>.
</p>



<h3>Examples</h3>

<pre><code class='language-R'>
if (require("ggplot2") &amp;&amp; require("dplyr")) {
  # get antimicrobial results for drugs against a UTI:
  ggplot(example_isolates %&gt;% select(AMX, NIT, FOS, TMP, CIP)) +
    geom_sir()
}
if (require("ggplot2") &amp;&amp; require("dplyr")) {
  # prettify the plot using some additional functions:
  df &lt;- example_isolates %&gt;% select(AMX, NIT, FOS, TMP, CIP)
  ggplot(df) +
    geom_sir() +
    scale_y_percent() +
    scale_sir_colours() +
    labels_sir_count() +
    theme_sir()
}
if (require("ggplot2") &amp;&amp; require("dplyr")) {
  # or better yet, simplify this using the wrapper function - a single command:
  example_isolates %&gt;%
    select(AMX, NIT, FOS, TMP, CIP) %&gt;%
    ggplot_sir()
}
if (require("ggplot2") &amp;&amp; require("dplyr")) {
  # get only proportions and no counts:
  example_isolates %&gt;%
    select(AMX, NIT, FOS, TMP, CIP) %&gt;%
    ggplot_sir(datalabels = FALSE)
}
if (require("ggplot2") &amp;&amp; require("dplyr")) {
  # add other ggplot2 arguments as you like:
  example_isolates %&gt;%
    select(AMX, NIT, FOS, TMP, CIP) %&gt;%
    ggplot_sir(
      width = 0.5,
      colour = "black",
      size = 1,
      linetype = 2,
      alpha = 0.25
    )
}
if (require("ggplot2") &amp;&amp; require("dplyr")) {
  # you can alter the colours with colour names:
  example_isolates %&gt;%
    select(AMX) %&gt;%
    ggplot_sir(colours = c(SI = "yellow"))
}
if (require("ggplot2") &amp;&amp; require("dplyr")) {
  # but you can also use the built-in colour-blind friendly colours for
  # your plots, where "S" is green, "I" is yellow and "R" is red:
  data.frame(
    x = c("Value1", "Value2", "Value3"),
    y = c(1, 2, 3),
    z = c("Value4", "Value5", "Value6")
  ) %&gt;%
    ggplot() +
    geom_col(aes(x = x, y = y, fill = z)) +
    scale_sir_colours(Value4 = "S", Value5 = "I", Value6 = "R")
}
if (require("ggplot2") &amp;&amp; require("dplyr")) {
  # resistance of ciprofloxacine per age group
  example_isolates %&gt;%
    mutate(first_isolate = first_isolate()) %&gt;%
    filter(
      first_isolate == TRUE,
      mo == as.mo("Escherichia coli")
    ) %&gt;%
    # age_groups() is also a function in this AMR package:
    group_by(age_group = age_groups(age)) %&gt;%
    select(age_group, CIP) %&gt;%
    ggplot_sir(x = "age_group")
}
if (require("ggplot2") &amp;&amp; require("dplyr")) {
  # a shorter version which also adjusts data label colours:
  example_isolates %&gt;%
    select(AMX, NIT, FOS, TMP, CIP) %&gt;%
    ggplot_sir(colours = FALSE)
}
if (require("ggplot2") &amp;&amp; require("dplyr")) {
  # it also supports groups (don't forget to use the group var on `x` or `facet`):
  example_isolates %&gt;%
    filter(mo_is_gram_negative(), ward != "Outpatient") %&gt;%
    # select only UTI-specific drugs
    select(ward, AMX, NIT, FOS, TMP, CIP) %&gt;%
    group_by(ward) %&gt;%
    ggplot_sir(
      x = "ward",
      facet = "antibiotic",
      nrow = 1,
      title = "AMR of Anti-UTI Drugs Per Ward",
      x.title = "Ward",
      datalabels = FALSE
    )
}

</code></pre>

<hr>
<h2 id='guess_ab_col'>Guess Antibiotic Column</h2><span id='topic+guess_ab_col'></span>

<h3>Description</h3>

<p>This tries to find a column name in a data set based on information from the <a href="#topic+antibiotics">antibiotics</a> data set. Also supports WHONET abbreviations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>guess_ab_col(
  x = NULL,
  search_string = NULL,
  verbose = FALSE,
  only_sir_columns = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="guess_ab_col_+3A_x">x</code></td>
<td>
<p>a <a href="base.html#topic+data.frame">data.frame</a></p>
</td></tr>
<tr><td><code id="guess_ab_col_+3A_search_string">search_string</code></td>
<td>
<p>a text to search <code>x</code> for, will be checked with <code><a href="#topic+as.ab">as.ab()</a></code> if this value is not a column in <code>x</code></p>
</td></tr>
<tr><td><code id="guess_ab_col_+3A_verbose">verbose</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether additional info should be printed</p>
</td></tr>
<tr><td><code id="guess_ab_col_+3A_only_sir_columns">only_sir_columns</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether only antibiotic columns must be detected that were transformed to class <code>sir</code> (see <code><a href="#topic+as.sir">as.sir()</a></code>) on beforehand (default is <code>FALSE</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>You can look for an antibiotic (trade) name or abbreviation and it will search <code>x</code> and the <a href="#topic+antibiotics">antibiotics</a> data set for any column containing a name or code of that antibiotic.
</p>


<h3>Value</h3>

<p>A column name of <code>x</code>, or <code>NULL</code> when no result is found.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df &lt;- data.frame(
  amox = "S",
  tetr = "R"
)

guess_ab_col(df, "amoxicillin")
guess_ab_col(df, "J01AA07") # ATC code of tetracycline

guess_ab_col(df, "J01AA07", verbose = TRUE)
# NOTE: Using column 'tetr' as input for J01AA07 (tetracycline).

# WHONET codes
df &lt;- data.frame(
  AMP_ND10 = "R",
  AMC_ED20 = "S"
)
guess_ab_col(df, "ampicillin")
guess_ab_col(df, "J01CR02")
guess_ab_col(df, as.ab("augmentin"))
</code></pre>

<hr>
<h2 id='intrinsic_resistant'>Data Set with Bacterial Intrinsic Resistance</h2><span id='topic+intrinsic_resistant'></span>

<h3>Description</h3>

<p>Data set containing defined intrinsic resistance by EUCAST of all bug-drug combinations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intrinsic_resistant
</code></pre>


<h3>Format</h3>

<p>A <a href="tibble.html#topic+tibble">tibble</a> with 134 634 observations and 2 variables:
</p>

<ul>
<li> <p><code>mo</code><br /> Microorganism ID
</p>
</li>
<li> <p><code>ab</code><br /> Antibiotic ID
</p>
</li></ul>



<h3>Details</h3>

<p>This data set is based on <a href="https://www.eucast.org/expert_rules_and_expected_phenotypes">'EUCAST Expert Rules' and 'EUCAST Intrinsic Resistance and Unusual Phenotypes' v3.3</a> (2021).
</p>


<h4>Direct download</h4>

<p>Like all data sets in this package, this data set is publicly available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>.
</p>
<p>They <strong>allow for machine reading EUCAST and CLSI guidelines</strong>, which is almost impossible with the MS Excel and PDF files distributed by EUCAST and CLSI.
</p>



<h3>Examples</h3>

<pre><code class='language-R'>intrinsic_resistant
</code></pre>

<hr>
<h2 id='italicise_taxonomy'>Italicise Taxonomic Families, Genera, Species, Subspecies</h2><span id='topic+italicise_taxonomy'></span><span id='topic+italicize_taxonomy'></span>

<h3>Description</h3>

<p>According to the binomial nomenclature, the lowest four taxonomic levels (family, genus, species, subspecies) should be printed in italics. This function finds taxonomic names within strings and makes them italic.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>italicise_taxonomy(string, type = c("markdown", "ansi"))

italicize_taxonomy(string, type = c("markdown", "ansi"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="italicise_taxonomy_+3A_string">string</code></td>
<td>
<p>a <a href="base.html#topic+character">character</a> (vector)</p>
</td></tr>
<tr><td><code id="italicise_taxonomy_+3A_type">type</code></td>
<td>
<p>type of conversion of the taxonomic names, either &quot;markdown&quot; or &quot;ansi&quot;, see <em>Details</em></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function finds the taxonomic names and makes them italic based on the <a href="#topic+microorganisms">microorganisms</a> data set.
</p>
<p>The taxonomic names can be italicised using markdown (the default) by adding <code>*</code> before and after the taxonomic names, or using ANSI colours by adding <code style="white-space: pre;">&#8288;\033[3m&#8288;</code> before and <code style="white-space: pre;">&#8288;\033[23m&#8288;</code> after the taxonomic names. If multiple ANSI colours are not available, no conversion will occur.
</p>
<p>This function also supports abbreviation of the genus if it is followed by a species, such as &quot;E. coli&quot; and &quot;K. pneumoniae ozaenae&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>italicise_taxonomy("An overview of Staphylococcus aureus isolates")
italicise_taxonomy("An overview of S. aureus isolates")

cat(italicise_taxonomy("An overview of S. aureus isolates", type = "ansi"))
</code></pre>

<hr>
<h2 id='join'>Join <a href="#topic+microorganisms">microorganisms</a> to a Data Set</h2><span id='topic+join'></span><span id='topic+inner_join_microorganisms'></span><span id='topic+inner_join'></span><span id='topic+left_join_microorganisms'></span><span id='topic+right_join_microorganisms'></span><span id='topic+full_join_microorganisms'></span><span id='topic+semi_join_microorganisms'></span><span id='topic+anti_join_microorganisms'></span>

<h3>Description</h3>

<p>Join the data set <a href="#topic+microorganisms">microorganisms</a> easily to an existing data set or to a <a href="base.html#topic+character">character</a> vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inner_join_microorganisms(x, by = NULL, suffix = c("2", ""), ...)

left_join_microorganisms(x, by = NULL, suffix = c("2", ""), ...)

right_join_microorganisms(x, by = NULL, suffix = c("2", ""), ...)

full_join_microorganisms(x, by = NULL, suffix = c("2", ""), ...)

semi_join_microorganisms(x, by = NULL, ...)

anti_join_microorganisms(x, by = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="join_+3A_x">x</code></td>
<td>
<p>existing data set to join, or <a href="base.html#topic+character">character</a> vector. In case of a <a href="base.html#topic+character">character</a> vector, the resulting <a href="base.html#topic+data.frame">data.frame</a> will contain a column 'x' with these values.</p>
</td></tr>
<tr><td><code id="join_+3A_by">by</code></td>
<td>
<p>a variable to join by - if left empty will search for a column with class <code><a href="#topic+mo">mo</a></code> (created with <code><a href="#topic+as.mo">as.mo()</a></code>) or will be <code>"mo"</code> if that column name exists in <code>x</code>, could otherwise be a column name of <code>x</code> with values that exist in <code>microorganisms$mo</code> (such as <code>by = "bacteria_id"</code>), or another column in <a href="#topic+microorganisms">microorganisms</a> (but then it should be named, like <code>by = c("bacteria_id" = "fullname")</code>)</p>
</td></tr>
<tr><td><code id="join_+3A_suffix">suffix</code></td>
<td>
<p>if there are non-joined duplicate variables in <code>x</code> and <code>y</code>, these suffixes will be added to the output to disambiguate them. Should be a <a href="base.html#topic+character">character</a> vector of length 2.</p>
</td></tr>
<tr><td><code id="join_+3A_...">...</code></td>
<td>
<p>ignored, only in place to allow future extensions</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Note:</strong> As opposed to the <code>join()</code> functions of <code>dplyr</code>, <a href="base.html#topic+character">character</a> vectors are supported and at default existing columns will get a suffix <code>"2"</code> and the newly joined columns will not get a suffix.
</p>
<p>If the <code>dplyr</code> package is installed, their join functions will be used. Otherwise, the much slower <code><a href="base.html#topic+merge">merge()</a></code> and <code><a href="base.html#topic+interaction">interaction()</a></code> functions from base <span class="rlang"><b>R</b></span> will be used.
</p>


<h3>Value</h3>

<p>a <a href="base.html#topic+data.frame">data.frame</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>left_join_microorganisms(as.mo("K. pneumoniae"))
left_join_microorganisms("B_KLBSL_PNMN")

df &lt;- data.frame(
  date = seq(
    from = as.Date("2018-01-01"),
    to = as.Date("2018-01-07"),
    by = 1
  ),
  bacteria = as.mo(c(
    "S. aureus", "MRSA", "MSSA", "STAAUR",
    "E. coli", "E. coli", "E. coli"
  )),
  stringsAsFactors = FALSE
)
colnames(df)

df_joined &lt;- left_join_microorganisms(df, "bacteria")
colnames(df_joined)


if (require("dplyr")) {
  example_isolates %&gt;%
    left_join_microorganisms() %&gt;%
    colnames()
}

</code></pre>

<hr>
<h2 id='key_antimicrobials'>(Key) Antimicrobials for First Weighted Isolates</h2><span id='topic+key_antimicrobials'></span><span id='topic+all_antimicrobials'></span><span id='topic+antimicrobials_equal'></span>

<h3>Description</h3>

<p>These functions can be used to determine first weighted isolates by considering the phenotype for isolate selection (see <code><a href="#topic+first_isolate">first_isolate()</a></code>). Using a phenotype-based method to determine first isolates is more reliable than methods that disregard phenotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>key_antimicrobials(
  x = NULL,
  col_mo = NULL,
  universal = c("ampicillin", "amoxicillin/clavulanic acid", "cefuroxime",
    "piperacillin/tazobactam", "ciprofloxacin", "trimethoprim/sulfamethoxazole"),
  gram_negative = c("gentamicin", "tobramycin", "colistin", "cefotaxime", "ceftazidime",
    "meropenem"),
  gram_positive = c("vancomycin", "teicoplanin", "tetracycline", "erythromycin",
    "oxacillin", "rifampin"),
  antifungal = c("anidulafungin", "caspofungin", "fluconazole", "miconazole", "nystatin",
    "voriconazole"),
  only_sir_columns = FALSE,
  ...
)

all_antimicrobials(x = NULL, only_sir_columns = FALSE, ...)

antimicrobials_equal(
  y,
  z,
  type = c("points", "keyantimicrobials"),
  ignore_I = TRUE,
  points_threshold = 2,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="key_antimicrobials_+3A_x">x</code></td>
<td>
<p>a <a href="base.html#topic+data.frame">data.frame</a> with antibiotics columns, like <code>AMX</code> or <code>amox</code>. Can be left blank to determine automatically</p>
</td></tr>
<tr><td><code id="key_antimicrobials_+3A_col_mo">col_mo</code></td>
<td>
<p>column name of the names or codes of the microorganisms (see <code><a href="#topic+as.mo">as.mo()</a></code>) - the default is the first column of class <code><a href="#topic+mo">mo</a></code>. Values will be coerced using <code><a href="#topic+as.mo">as.mo()</a></code>.</p>
</td></tr>
<tr><td><code id="key_antimicrobials_+3A_universal">universal</code></td>
<td>
<p>names of <strong>broad-spectrum</strong> antimicrobial drugs, case-insensitive. Set to <code>NULL</code> to ignore. See <em>Details</em> for the default antimicrobial drugs</p>
</td></tr>
<tr><td><code id="key_antimicrobials_+3A_gram_negative">gram_negative</code></td>
<td>
<p>names of antibiotic drugs for <strong>Gram-positives</strong>, case-insensitive. Set to <code>NULL</code> to ignore. See <em>Details</em> for the default antibiotic drugs</p>
</td></tr>
<tr><td><code id="key_antimicrobials_+3A_gram_positive">gram_positive</code></td>
<td>
<p>names of antibiotic drugs for <strong>Gram-negatives</strong>, case-insensitive. Set to <code>NULL</code> to ignore. See <em>Details</em> for the default antibiotic drugs</p>
</td></tr>
<tr><td><code id="key_antimicrobials_+3A_antifungal">antifungal</code></td>
<td>
<p>names of antifungal drugs for <strong>fungi</strong>, case-insensitive. Set to <code>NULL</code> to ignore. See <em>Details</em> for the default antifungal drugs</p>
</td></tr>
<tr><td><code id="key_antimicrobials_+3A_only_sir_columns">only_sir_columns</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether only columns must be included that were transformed to class <code>sir</code> (see <code><a href="#topic+as.sir">as.sir()</a></code>) on beforehand (default is <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="key_antimicrobials_+3A_...">...</code></td>
<td>
<p>ignored, only in place to allow future extensions</p>
</td></tr>
<tr><td><code id="key_antimicrobials_+3A_y">y</code>, <code id="key_antimicrobials_+3A_z">z</code></td>
<td>
<p><a href="base.html#topic+character">character</a> vectors to compare</p>
</td></tr>
<tr><td><code id="key_antimicrobials_+3A_type">type</code></td>
<td>
<p>type to determine weighed isolates; can be <code>"keyantimicrobials"</code> or <code>"points"</code>, see <em>Details</em></p>
</td></tr>
<tr><td><code id="key_antimicrobials_+3A_ignore_i">ignore_I</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> to indicate whether antibiotic interpretations with <code>"I"</code> will be ignored when <code>type = "keyantimicrobials"</code>, see <em>Details</em></p>
</td></tr>
<tr><td><code id="key_antimicrobials_+3A_points_threshold">points_threshold</code></td>
<td>
<p>minimum number of points to require before differences in the antibiogram will lead to inclusion of an isolate when <code>type = "points"</code>, see <em>Details</em></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code><a href="#topic+key_antimicrobials">key_antimicrobials()</a></code> and <code><a href="#topic+all_antimicrobials">all_antimicrobials()</a></code> functions are context-aware. This means that the <code>x</code> argument can be left blank if used inside a <a href="base.html#topic+data.frame">data.frame</a> call, see <em>Examples</em>.
</p>
<p>The function <code><a href="#topic+key_antimicrobials">key_antimicrobials()</a></code> returns a <a href="base.html#topic+character">character</a> vector with 12 antimicrobial results for every isolate. The function <code><a href="#topic+all_antimicrobials">all_antimicrobials()</a></code> returns a <a href="base.html#topic+character">character</a> vector with all antimicrobial drug results for every isolate. These vectors can then be compared using <code><a href="#topic+antimicrobials_equal">antimicrobials_equal()</a></code>, to check if two isolates have generally the same antibiogram. Missing and invalid values are replaced with a dot (<code>"."</code>) by <code><a href="#topic+key_antimicrobials">key_antimicrobials()</a></code> and ignored by <code><a href="#topic+antimicrobials_equal">antimicrobials_equal()</a></code>.
</p>
<p>Please see the <code><a href="#topic+first_isolate">first_isolate()</a></code> function how these important functions enable the 'phenotype-based' method for determination of first isolates.
</p>
<p>The default antimicrobial drugs used for <strong>all rows</strong> (set in <code>universal</code>) are:
</p>

<ul>
<li><p> Ampicillin
</p>
</li>
<li><p> Amoxicillin/clavulanic acid
</p>
</li>
<li><p> Cefuroxime
</p>
</li>
<li><p> Ciprofloxacin
</p>
</li>
<li><p> Piperacillin/tazobactam
</p>
</li>
<li><p> Trimethoprim/sulfamethoxazole
</p>
</li></ul>

<p>The default antimicrobial drugs used for <strong>Gram-negative bacteria</strong> (set in <code>gram_negative</code>) are:
</p>

<ul>
<li><p> Cefotaxime
</p>
</li>
<li><p> Ceftazidime
</p>
</li>
<li><p> Colistin
</p>
</li>
<li><p> Gentamicin
</p>
</li>
<li><p> Meropenem
</p>
</li>
<li><p> Tobramycin
</p>
</li></ul>

<p>The default antimicrobial drugs used for <strong>Gram-positive bacteria</strong> (set in <code>gram_positive</code>) are:
</p>

<ul>
<li><p> Erythromycin
</p>
</li>
<li><p> Oxacillin
</p>
</li>
<li><p> Rifampin
</p>
</li>
<li><p> Teicoplanin
</p>
</li>
<li><p> Tetracycline
</p>
</li>
<li><p> Vancomycin
</p>
</li></ul>

<p>The default antimicrobial drugs used for <strong>fungi</strong> (set in <code>antifungal</code>) are:
</p>

<ul>
<li><p> Anidulafungin
</p>
</li>
<li><p> Caspofungin
</p>
</li>
<li><p> Fluconazole
</p>
</li>
<li><p> Miconazole
</p>
</li>
<li><p> Nystatin
</p>
</li>
<li><p> Voriconazole
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+first_isolate">first_isolate()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># `example_isolates` is a data set available in the AMR package.
# See ?example_isolates.

# output of the `key_antimicrobials()` function could be like this:
strainA &lt;- "SSSRR.S.R..S"
strainB &lt;- "SSSIRSSSRSSS"

# those strings can be compared with:
antimicrobials_equal(strainA, strainB, type = "keyantimicrobials")
# TRUE, because I is ignored (as well as missing values)

antimicrobials_equal(strainA, strainB, type = "keyantimicrobials", ignore_I = FALSE)
# FALSE, because I is not ignored and so the 4th [character] differs


if (require("dplyr")) {
  # set key antibiotics to a new variable
  my_patients &lt;- example_isolates %&gt;%
    mutate(keyab = key_antimicrobials(antifungal = NULL)) %&gt;% # no need to define `x`
    mutate(
      # now calculate first isolates
      first_regular = first_isolate(col_keyantimicrobials = FALSE),
      # and first WEIGHTED isolates
      first_weighted = first_isolate(col_keyantimicrobials = "keyab")
    )

  # Check the difference in this data set, 'weighted' results in more isolates:
  sum(my_patients$first_regular, na.rm = TRUE)
  sum(my_patients$first_weighted, na.rm = TRUE)
}

</code></pre>

<hr>
<h2 id='kurtosis'>Kurtosis of the Sample</h2><span id='topic+kurtosis'></span><span id='topic+kurtosis.default'></span><span id='topic+kurtosis.matrix'></span><span id='topic+kurtosis.data.frame'></span>

<h3>Description</h3>

<p>Kurtosis is a measure of the &quot;tailedness&quot; of the probability distribution of a real-valued random variable. A normal distribution has a kurtosis of 3 and a excess kurtosis of 0.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kurtosis(x, na.rm = FALSE, excess = FALSE)

## Default S3 method:
kurtosis(x, na.rm = FALSE, excess = FALSE)

## S3 method for class 'matrix'
kurtosis(x, na.rm = FALSE, excess = FALSE)

## S3 method for class 'data.frame'
kurtosis(x, na.rm = FALSE, excess = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kurtosis_+3A_x">x</code></td>
<td>
<p>a vector of values, a <a href="base.html#topic+matrix">matrix</a> or a <a href="base.html#topic+data.frame">data.frame</a></p>
</td></tr>
<tr><td><code id="kurtosis_+3A_na.rm">na.rm</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether <code>NA</code> values should be stripped before the computation proceeds</p>
</td></tr>
<tr><td><code id="kurtosis_+3A_excess">excess</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether the <em>excess kurtosis</em> should be returned, defined as the kurtosis minus 3.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+skewness">skewness()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>kurtosis(rnorm(10000))
kurtosis(rnorm(10000), excess = TRUE)
</code></pre>

<hr>
<h2 id='like'>Vectorised Pattern Matching with Keyboard Shortcut</h2><span id='topic+like'></span><span id='topic++25like+25'></span><span id='topic++25unlike+25'></span><span id='topic++25like_case+25'></span><span id='topic++25unlike_case+25'></span>

<h3>Description</h3>

<p>Convenient wrapper around <code><a href="base.html#topic+grepl">grepl()</a></code> to match a pattern: <code>x %like% pattern</code>. It always returns a <code><a href="base.html#topic+logical">logical</a></code> vector and is always case-insensitive (use <code>x %like_case% pattern</code> for case-sensitive matching). Also, <code>pattern</code> can be as long as <code>x</code> to compare items of each index in both vectors, or they both can have the same length to iterate over all cases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>like(x, pattern, ignore.case = TRUE)

x %like% pattern

x %unlike% pattern

x %like_case% pattern

x %unlike_case% pattern
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="like_+3A_x">x</code></td>
<td>
<p>a <a href="base.html#topic+character">character</a> vector where matches are sought, or an object which can be coerced by <code><a href="base.html#topic+as.character">as.character()</a></code> to a <a href="base.html#topic+character">character</a> vector.</p>
</td></tr>
<tr><td><code id="like_+3A_pattern">pattern</code></td>
<td>
<p>a <a href="base.html#topic+character">character</a> vector containing regular expressions (or a <a href="base.html#topic+character">character</a> string for <code>fixed = TRUE</code>) to be matched in the given <a href="base.html#topic+character">character</a> vector. Coerced by <code><a href="base.html#topic+as.character">as.character()</a></code> to a <a href="base.html#topic+character">character</a> string if possible.</p>
</td></tr>
<tr><td><code id="like_+3A_ignore.case">ignore.case</code></td>
<td>
<p>if <code>FALSE</code>, the pattern matching is <em>case sensitive</em> and if <code>TRUE</code>, case is ignored during matching.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These <code><a href="#topic+like">like()</a></code> and <code style="white-space: pre;">&#8288;%like%&#8288;</code>/<code style="white-space: pre;">&#8288;%unlike%&#8288;</code> functions:
</p>

<ul>
<li><p> Are case-insensitive (use <code style="white-space: pre;">&#8288;%like_case%&#8288;</code>/<code style="white-space: pre;">&#8288;%unlike_case%&#8288;</code> for case-sensitive matching)
</p>
</li>
<li><p> Support multiple patterns
</p>
</li>
<li><p> Check if <code>pattern</code> is a valid regular expression and sets <code>fixed = TRUE</code> if not, to greatly improve speed (vectorised over <code>pattern</code>)
</p>
</li>
<li><p> Always use compatibility with Perl unless <code>fixed = TRUE</code>, to greatly improve speed
</p>
</li></ul>

<p>Using RStudio? The <code style="white-space: pre;">&#8288;%like%&#8288;</code>/<code style="white-space: pre;">&#8288;%unlike%&#8288;</code> functions can also be directly inserted in your code from the Addins menu and can have its own keyboard shortcut like <code>Shift+Ctrl+L</code> or <code>Shift+Cmd+L</code> (see menu <code>Tools</code> &gt; <code style="white-space: pre;">&#8288;Modify Keyboard Shortcuts...&#8288;</code>). If you keep pressing your shortcut, the inserted text will be iterated over <code style="white-space: pre;">&#8288;%like%&#8288;</code> -&gt; <code style="white-space: pre;">&#8288;%unlike%&#8288;</code> -&gt; <code style="white-space: pre;">&#8288;%like_case%&#8288;</code> -&gt; <code style="white-space: pre;">&#8288;%unlike_case%&#8288;</code>.
</p>


<h3>Value</h3>

<p>A <a href="base.html#topic+logical">logical</a> vector
</p>


<h3>Source</h3>

<p>Idea from the <a href="https://github.com/Rdatatable/data.table/blob/ec1259af1bf13fc0c96a1d3f9e84d55d8106a9a4/R/like.R"><code>like</code> function from the <code>data.table</code> package</a>, although altered as explained in <em>Details</em>.
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+grepl">grepl()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># data.table has a more limited version of %like%, so unload it:
try(detach("package:data.table", unload = TRUE), silent = TRUE)

a &lt;- "This is a test"
b &lt;- "TEST"
a %like% b
b %like% a

# also supports multiple patterns
a &lt;- c("Test case", "Something different", "Yet another thing")
b &lt;- c("case", "diff", "yet")
a %like% b
a %unlike% b

a[1] %like% b
a %like% b[1]


# get isolates whose name start with 'Entero' (case-insensitive)
example_isolates[which(mo_name() %like% "^entero"), ]

if (require("dplyr")) {
  example_isolates %&gt;%
    filter(mo_name() %like% "^ent")
}

</code></pre>

<hr>
<h2 id='mdro'>Determine Multidrug-Resistant Organisms (MDRO)</h2><span id='topic+mdro'></span><span id='topic+MDR'></span><span id='topic+XDR'></span><span id='topic+PDR'></span><span id='topic+BRMO'></span><span id='topic+3MRGN'></span><span id='topic+4MRGN'></span><span id='topic+custom_mdro_guideline'></span><span id='topic+brmo'></span><span id='topic+mrgn'></span><span id='topic+mdr_tb'></span><span id='topic+mdr_cmi2012'></span><span id='topic+eucast_exceptional_phenotypes'></span>

<h3>Description</h3>

<p>Determine which isolates are multidrug-resistant organisms (MDRO) according to international, national and custom guidelines.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mdro(
  x = NULL,
  guideline = "CMI2012",
  col_mo = NULL,
  info = interactive(),
  pct_required_classes = 0.5,
  combine_SI = TRUE,
  verbose = FALSE,
  only_sir_columns = FALSE,
  ...
)

custom_mdro_guideline(..., as_factor = TRUE)

brmo(x = NULL, only_sir_columns = FALSE, ...)

mrgn(x = NULL, only_sir_columns = FALSE, ...)

mdr_tb(x = NULL, only_sir_columns = FALSE, ...)

mdr_cmi2012(x = NULL, only_sir_columns = FALSE, ...)

eucast_exceptional_phenotypes(x = NULL, only_sir_columns = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mdro_+3A_x">x</code></td>
<td>
<p>a <a href="base.html#topic+data.frame">data.frame</a> with antibiotics columns, like <code>AMX</code> or <code>amox</code>. Can be left blank for automatic determination.</p>
</td></tr>
<tr><td><code id="mdro_+3A_guideline">guideline</code></td>
<td>
<p>a specific guideline to follow, see sections <em>Supported international / national guidelines</em> and <em>Using Custom Guidelines</em> below. When left empty, the publication by Magiorakos <em>et al.</em> (see below) will be followed.</p>
</td></tr>
<tr><td><code id="mdro_+3A_col_mo">col_mo</code></td>
<td>
<p>column name of the names or codes of the microorganisms (see <code><a href="#topic+as.mo">as.mo()</a></code>) - the default is the first column of class <code><a href="#topic+mo">mo</a></code>. Values will be coerced using <code><a href="#topic+as.mo">as.mo()</a></code>.</p>
</td></tr>
<tr><td><code id="mdro_+3A_info">info</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether progress should be printed to the console - the default is only print while in interactive sessions</p>
</td></tr>
<tr><td><code id="mdro_+3A_pct_required_classes">pct_required_classes</code></td>
<td>
<p>minimal required percentage of antimicrobial classes that must be available per isolate, rounded down. For example, with the default guideline, 17 antimicrobial classes must be available for <em>S. aureus</em>. Setting this <code>pct_required_classes</code> argument to <code>0.5</code> (default) means that for every <em>S. aureus</em> isolate at least 8 different classes must be available. Any lower number of available classes will return <code>NA</code> for that isolate.</p>
</td></tr>
<tr><td><code id="mdro_+3A_combine_si">combine_SI</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether all values of S and I must be merged into one, so resistance is only considered when isolates are R, not I. As this is the default behaviour of the <code><a href="#topic+mdro">mdro()</a></code> function, it follows the redefinition by EUCAST about the interpretation of I (increased exposure) in 2019, see section 'Interpretation of S, I and R' below. When using <code>combine_SI = FALSE</code>, resistance is considered when isolates are R or I.</p>
</td></tr>
<tr><td><code id="mdro_+3A_verbose">verbose</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to turn Verbose mode on and off (default is off). In Verbose mode, the function does not return the MDRO results, but instead returns a data set in logbook form with extensive info about which isolates would be MDRO-positive, or why they are not.</p>
</td></tr>
<tr><td><code id="mdro_+3A_only_sir_columns">only_sir_columns</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether only antibiotic columns must be detected that were transformed to class <code>sir</code> (see <code><a href="#topic+as.sir">as.sir()</a></code>) on beforehand (default is <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="mdro_+3A_...">...</code></td>
<td>
<p>in case of <code><a href="#topic+custom_mdro_guideline">custom_mdro_guideline()</a></code>: a set of rules, see section <em>Using Custom Guidelines</em> below. Otherwise: column name of an antibiotic, see section <em>Antibiotics</em> below.</p>
</td></tr>
<tr><td><code id="mdro_+3A_as_factor">as_factor</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether the returned value should be an ordered <a href="base.html#topic+factor">factor</a> (<code>TRUE</code>, default), or otherwise a <a href="base.html#topic+character">character</a> vector</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions are context-aware. This means that the <code>x</code> argument can be left blank if used inside a <a href="base.html#topic+data.frame">data.frame</a> call, see <em>Examples</em>.
</p>
<p>For the <code>pct_required_classes</code> argument, values above 1 will be divided by 100. This is to support both fractions (<code>0.75</code> or <code>3/4</code>) and percentages (<code>75</code>).
</p>
<p><strong>Note:</strong> Every test that involves the Enterobacteriaceae family, will internally be performed using its newly named <em>order</em> Enterobacterales, since the Enterobacteriaceae family has been taxonomically reclassified by Adeolu <em>et al.</em> in 2016. Before that, Enterobacteriaceae was the only family under the Enterobacteriales (with an i) order. All species under the old Enterobacteriaceae family are still under the new Enterobacterales (without an i) order, but divided into multiple families. The way tests are performed now by this <code><a href="#topic+mdro">mdro()</a></code> function makes sure that results from before 2016 and after 2016 are identical.
</p>


<h3>Value</h3>


<ul>
<li><p> CMI 2012 paper - function <code><a href="#topic+mdr_cmi2012">mdr_cmi2012()</a></code> or <code><a href="#topic+mdro">mdro()</a></code>:<br />
Ordered <a href="base.html#topic+factor">factor</a> with levels <code>Negative</code> &lt; <code>Multi-drug-resistant (MDR)</code> &lt; <code style="white-space: pre;">&#8288;Extensively drug-resistant (XDR)&#8288;</code> &lt; <code>Pandrug-resistant (PDR)</code>
</p>
</li>
<li><p> TB guideline - function <code><a href="#topic+mdr_tb">mdr_tb()</a></code> or <code><a href="#topic+mdro">mdro(..., guideline = &quot;TB&quot;)</a></code>:<br />
Ordered <a href="base.html#topic+factor">factor</a> with levels <code>Negative</code> &lt; <code>Mono-resistant</code> &lt; <code>Poly-resistant</code> &lt; <code>Multi-drug-resistant</code> &lt; <code style="white-space: pre;">&#8288;Extensively drug-resistant&#8288;</code>
</p>
</li>
<li><p> German guideline - function <code><a href="#topic+mrgn">mrgn()</a></code> or <code><a href="#topic+mdro">mdro(..., guideline = &quot;MRGN&quot;)</a></code>:<br />
Ordered <a href="base.html#topic+factor">factor</a> with levels <code>Negative</code> &lt; <code style="white-space: pre;">&#8288;3MRGN&#8288;</code> &lt; <code style="white-space: pre;">&#8288;4MRGN&#8288;</code>
</p>
</li>
<li><p> Everything else, except for custom guidelines:<br />
Ordered <a href="base.html#topic+factor">factor</a> with levels <code>Negative</code> &lt; <code style="white-space: pre;">&#8288;Positive, unconfirmed&#8288;</code> &lt; <code>Positive</code>. The value <code>"Positive, unconfirmed"</code> means that, according to the guideline, it is not entirely sure if the isolate is multi-drug resistant and this should be confirmed with additional (e.g. molecular) tests
</p>
</li></ul>



<h3>Supported International / National Guidelines</h3>

<p>Currently supported guidelines are (case-insensitive):
</p>

<ul>
<li> <p><code>guideline = "CMI2012"</code> (default)
</p>
<p>Magiorakos AP, Srinivasan A <em>et al.</em> &quot;Multidrug-resistant, extensively drug-resistant and pandrug-resistant bacteria: an international expert proposal for interim standard definitions for acquired resistance.&quot; Clinical Microbiology and Infection (2012) (<a href="https://doi.org/10.1111/j.1469-0691.2011.03570.x">doi:10.1111/j.1469-0691.2011.03570.x</a>)
</p>
</li>
<li> <p><code>guideline = "EUCAST3.3"</code> (or simply <code>guideline = "EUCAST"</code>)
</p>
<p>The European international guideline - EUCAST Expert Rules Version 3.3 &quot;Intrinsic Resistance and Unusual Phenotypes&quot; (<a href="https://www.eucast.org/fileadmin/src/media/PDFs/EUCAST_files/Expert_Rules/2021/Intrinsic_Resistance_and_Unusual_Phenotypes_Tables_v3.3_20211018.pdf">link</a>)
</p>
</li>
<li> <p><code>guideline = "EUCAST3.2"</code>
</p>
<p>The European international guideline - EUCAST Expert Rules Version 3.2 &quot;Intrinsic Resistance and Unusual Phenotypes&quot; (<a href="https://www.eucast.org/fileadmin/src/media/PDFs/EUCAST_files/Expert_Rules/2020/Intrinsic_Resistance_and_Unusual_Phenotypes_Tables_v3.2_20200225.pdf">link</a>)
</p>
</li>
<li> <p><code>guideline = "EUCAST3.1"</code>
</p>
<p>The European international guideline - EUCAST Expert Rules Version 3.1 &quot;Intrinsic Resistance and Exceptional Phenotypes Tables&quot; (<a href="https://www.eucast.org/fileadmin/src/media/PDFs/EUCAST_files/Expert_Rules/Expert_rules_intrinsic_exceptional_V3.1.pdf">link</a>)
</p>
</li>
<li> <p><code>guideline = "TB"</code>
</p>
<p>The international guideline for multi-drug resistant tuberculosis - World Health Organization &quot;Companion handbook to the WHO guidelines for the programmatic management of drug-resistant tuberculosis&quot; (<a href="https://www.who.int/publications/i/item/9789241548809">link</a>)
</p>
</li>
<li> <p><code>guideline = "MRGN"</code>
</p>
<p>The German national guideline - Mueller et al. (2015) Antimicrobial Resistance and Infection Control 4:7; <a href="https://doi.org/10.1186/s13756-015-0047-6">doi:10.1186/s13756-015-0047-6</a>
</p>
</li>
<li> <p><code>guideline = "BRMO"</code>
</p>
<p>The Dutch national guideline - Rijksinstituut voor Volksgezondheid en Milieu &quot;WIP-richtlijn BRMO (Bijzonder Resistente Micro-Organismen) (ZKH)&quot; (<a href="https://www.rivm.nl/wip-richtlijn-brmo-bijzonder-resistente-micro-organismen-zkh">link</a>)
</p>
</li></ul>

<p>Please suggest your own (country-specific) guidelines by letting us know: <a href="https://github.com/msberends/AMR/issues/new">https://github.com/msberends/AMR/issues/new</a>.
</p>


<h3>Using Custom Guidelines</h3>

<p>Custom guidelines can be set with the <code><a href="#topic+custom_mdro_guideline">custom_mdro_guideline()</a></code> function. This is of great importance if you have custom rules to determine MDROs in your hospital, e.g., rules that are dependent on ward, state of contact isolation or other variables in your data.
</p>
<p>If you are familiar with the <code><a href="dplyr.html#topic+case_when">case_when()</a></code> function of the <code>dplyr</code> package, you will recognise the input method to set your own rules. Rules must be set using what <span class="rlang"><b>R</b></span> considers to be the 'formula notation'. The rule is written <em>before</em> the tilde (<code>~</code>) and the consequence of the rule is written <em>after</em> the tilde:
</p>
<div class="sourceCode"><pre>custom &lt;- custom_mdro_guideline(CIP == "R" &amp; age &gt; 60 ~ "Elderly Type A",
                                ERY == "R" &amp; age &gt; 60 ~ "Elderly Type B")
</pre></div>
<p>If a row/an isolate matches the first rule, the value after the first <code>~</code> (in this case <em>'Elderly Type A'</em>) will be set as MDRO value. Otherwise, the second rule will be tried and so on. The number of rules is unlimited.
</p>
<p>You can print the rules set in the console for an overview. Colours will help reading it if your console supports colours.
</p>
<div class="sourceCode"><pre>custom
#&gt; A set of custom MDRO rules:
#&gt;   1. CIP is "R" and age is higher than 60 -&gt; Elderly Type A
#&gt;   2. ERY is "R" and age is higher than 60 -&gt; Elderly Type B
#&gt;   3. Otherwise -&gt; Negative
#&gt;
#&gt; Unmatched rows will return NA.
</pre></div>
<p>The outcome of the function can be used for the <code>guideline</code> argument in the <code><a href="#topic+mdro">mdro()</a></code> function:
</p>
<div class="sourceCode"><pre>x &lt;- mdro(example_isolates,
          guideline = custom)
table(x)
#&gt;       Negative Elderly Type A Elderly Type B
#&gt;           1070            198            732
</pre></div>
<p>Rules can also be combined with other custom rules by using <code><a href="base.html#topic+c">c()</a></code>:
</p>
<div class="sourceCode"><pre>x &lt;- mdro(example_isolates,
          guideline = c(custom,
                        custom_mdro_guideline(ERY == "R" &amp; age &gt; 50 ~ "Elderly Type C")))
table(x)
#&gt;       Negative Elderly Type A Elderly Type B Elderly Type C
#&gt;            961            198            732            109
</pre></div>
<p>The rules set (the <code>custom</code> object in this case) could be exported to a shared file location using <code><a href="base.html#topic+saveRDS">saveRDS()</a></code> if you collaborate with multiple users. The custom rules set could then be imported using <code><a href="base.html#topic+readRDS">readRDS()</a></code>.
</p>


<h3>Antibiotics</h3>

<p>To define antibiotics column names, leave as it is to determine it automatically with <code><a href="#topic+guess_ab_col">guess_ab_col()</a></code> or input a text (case-insensitive), or use <code>NULL</code> to skip a column (e.g. <code>TIC = NULL</code> to skip ticarcillin). Manually defined but non-existing columns will be skipped with a warning.
</p>
<p>The following antibiotics are eligible for the functions <code><a href="#topic+eucast_rules">eucast_rules()</a></code> and <code><a href="#topic+mdro">mdro()</a></code>. These are shown below in the format 'name (<code style="white-space: pre;">&#8288;antimicrobial ID&#8288;</code>, <a href="https://www.whocc.no/atc/structure_and_principles/">ATC code</a>)', sorted alphabetically:
</p>
<p>Amikacin (<code>AMK</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB06&amp;showdescription=no">J01GB06</a>), amoxicillin (<code>AMX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA04&amp;showdescription=no">J01CA04</a>), amoxicillin/clavulanic acid (<code>AMC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CR02&amp;showdescription=no">J01CR02</a>), ampicillin (<code>AMP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA01&amp;showdescription=no">J01CA01</a>), ampicillin/sulbactam (<code>SAM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CR01&amp;showdescription=no">J01CR01</a>), arbekacin (<code>ARB</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB12&amp;showdescription=no">J01GB12</a>), aspoxicillin (<code>APX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA19&amp;showdescription=no">J01CA19</a>), azidocillin (<code>AZD</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CE04&amp;showdescription=no">J01CE04</a>), azithromycin (<code>AZM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA10&amp;showdescription=no">J01FA10</a>), azlocillin (<code>AZL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA09&amp;showdescription=no">J01CA09</a>), aztreonam (<code>ATM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DF01&amp;showdescription=no">J01DF01</a>), bacampicillin (<code>BAM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA06&amp;showdescription=no">J01CA06</a>), bekanamycin (<code>BEK</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB13&amp;showdescription=no">J01GB13</a>), benzathine benzylpenicillin (<code>BNB</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CE08&amp;showdescription=no">J01CE08</a>), benzathine phenoxymethylpenicillin (<code>BNP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CE10&amp;showdescription=no">J01CE10</a>), benzylpenicillin (<code>PEN</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CE01&amp;showdescription=no">J01CE01</a>), besifloxacin (<code>BES</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=S01AE08&amp;showdescription=no">S01AE08</a>), biapenem (<code>BIA</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DH05&amp;showdescription=no">J01DH05</a>), carbenicillin (<code>CRB</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA03&amp;showdescription=no">J01CA03</a>), carindacillin (<code>CRN</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA05&amp;showdescription=no">J01CA05</a>), cefacetrile (<code>CAC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB10&amp;showdescription=no">J01DB10</a>), cefaclor (<code>CEC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC04&amp;showdescription=no">J01DC04</a>), cefadroxil (<code>CFR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB05&amp;showdescription=no">J01DB05</a>), cefalexin (<code>LEX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB01&amp;showdescription=no">J01DB01</a>), cefaloridine (<code>RID</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB02&amp;showdescription=no">J01DB02</a>), cefalotin (<code>CEP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB03&amp;showdescription=no">J01DB03</a>), cefamandole (<code>MAN</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC03&amp;showdescription=no">J01DC03</a>), cefapirin (<code>HAP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB08&amp;showdescription=no">J01DB08</a>), cefatrizine (<code>CTZ</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB07&amp;showdescription=no">J01DB07</a>), cefazedone (<code>CZD</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB06&amp;showdescription=no">J01DB06</a>), cefazolin (<code>CZO</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB04&amp;showdescription=no">J01DB04</a>), cefcapene (<code>CCP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD17&amp;showdescription=no">J01DD17</a>), cefdinir (<code>CDR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD15&amp;showdescription=no">J01DD15</a>), cefditoren (<code>DIT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD16&amp;showdescription=no">J01DD16</a>), cefepime (<code>FEP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DE01&amp;showdescription=no">J01DE01</a>), cefetamet (<code>CAT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD10&amp;showdescription=no">J01DD10</a>), cefixime (<code>CFM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD08&amp;showdescription=no">J01DD08</a>), cefmenoxime (<code>CMX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD05&amp;showdescription=no">J01DD05</a>), cefmetazole (<code>CMZ</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC09&amp;showdescription=no">J01DC09</a>), cefodizime (<code>DIZ</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD09&amp;showdescription=no">J01DD09</a>), cefonicid (<code>CID</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC06&amp;showdescription=no">J01DC06</a>), cefoperazone (<code>CFP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD12&amp;showdescription=no">J01DD12</a>), cefoperazone/sulbactam (<code>CSL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD62&amp;showdescription=no">J01DD62</a>), ceforanide (<code>CND</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC11&amp;showdescription=no">J01DC11</a>), cefotaxime (<code>CTX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD01&amp;showdescription=no">J01DD01</a>), cefotaxime/clavulanic acid (<code>CTC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD51&amp;showdescription=no">J01DD51</a>), cefotetan (<code>CTT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC05&amp;showdescription=no">J01DC05</a>), cefotiam (<code>CTF</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC07&amp;showdescription=no">J01DC07</a>), cefoxitin (<code>FOX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC01&amp;showdescription=no">J01DC01</a>), cefozopran (<code>ZOP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DE03&amp;showdescription=no">J01DE03</a>), cefpiramide (<code>CPM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD11&amp;showdescription=no">J01DD11</a>), cefpirome (<code>CPO</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DE02&amp;showdescription=no">J01DE02</a>), cefpodoxime (<code>CPD</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD13&amp;showdescription=no">J01DD13</a>), cefprozil (<code>CPR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC10&amp;showdescription=no">J01DC10</a>), cefroxadine (<code>CRD</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB11&amp;showdescription=no">J01DB11</a>), cefsulodin (<code>CFS</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD03&amp;showdescription=no">J01DD03</a>), ceftaroline (<code>CPT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DI02&amp;showdescription=no">J01DI02</a>), ceftazidime (<code>CAZ</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD02&amp;showdescription=no">J01DD02</a>), ceftazidime/clavulanic acid (<code>CCV</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD52&amp;showdescription=no">J01DD52</a>), cefteram (<code>CEM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD18&amp;showdescription=no">J01DD18</a>), ceftezole (<code>CTL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB12&amp;showdescription=no">J01DB12</a>), ceftibuten (<code>CTB</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD14&amp;showdescription=no">J01DD14</a>), ceftizoxime (<code>CZX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD07&amp;showdescription=no">J01DD07</a>), ceftobiprole medocaril (<code>CFM1</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DI01&amp;showdescription=no">J01DI01</a>), ceftolozane/tazobactam (<code>CZT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DI54&amp;showdescription=no">J01DI54</a>), ceftriaxone (<code>CRO</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD04&amp;showdescription=no">J01DD04</a>), ceftriaxone/beta-lactamase inhibitor (<code>CEB</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD63&amp;showdescription=no">J01DD63</a>), cefuroxime (<code>CXM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC02&amp;showdescription=no">J01DC02</a>), cephradine (<code>CED</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DB09&amp;showdescription=no">J01DB09</a>), chloramphenicol (<code>CHL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01BA01&amp;showdescription=no">J01BA01</a>), ciprofloxacin (<code>CIP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA02&amp;showdescription=no">J01MA02</a>), clarithromycin (<code>CLR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA09&amp;showdescription=no">J01FA09</a>), clindamycin (<code>CLI</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FF01&amp;showdescription=no">J01FF01</a>), clometocillin (<code>CLM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CE07&amp;showdescription=no">J01CE07</a>), cloxacillin (<code>CLO</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CF02&amp;showdescription=no">J01CF02</a>), colistin (<code>COL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XB01&amp;showdescription=no">J01XB01</a>), cycloserine (<code>CYC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J04AB01&amp;showdescription=no">J04AB01</a>), dalbavancin (<code>DAL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XA04&amp;showdescription=no">J01XA04</a>), daptomycin (<code>DAP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XX09&amp;showdescription=no">J01XX09</a>), delafloxacin (<code>DFX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA23&amp;showdescription=no">J01MA23</a>), dibekacin (<code>DKB</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB09&amp;showdescription=no">J01GB09</a>), dicloxacillin (<code>DIC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CF01&amp;showdescription=no">J01CF01</a>), dirithromycin (<code>DIR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA13&amp;showdescription=no">J01FA13</a>), doripenem (<code>DOR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DH04&amp;showdescription=no">J01DH04</a>), doxycycline (<code>DOX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01AA02&amp;showdescription=no">J01AA02</a>), enoxacin (<code>ENX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA04&amp;showdescription=no">J01MA04</a>), epicillin (<code>EPC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA07&amp;showdescription=no">J01CA07</a>), ertapenem (<code>ETP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DH03&amp;showdescription=no">J01DH03</a>), erythromycin (<code>ERY</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA01&amp;showdescription=no">J01FA01</a>), fleroxacin (<code>FLE</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA08&amp;showdescription=no">J01MA08</a>), flucloxacillin (<code>FLC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CF05&amp;showdescription=no">J01CF05</a>), flurithromycin (<code>FLR1</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA14&amp;showdescription=no">J01FA14</a>), fosfomycin (<code>FOS</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XX01&amp;showdescription=no">J01XX01</a>), framycetin (<code>FRM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=D09AA01&amp;showdescription=no">D09AA01</a>), fusidic acid (<code>FUS</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XC01&amp;showdescription=no">J01XC01</a>), garenoxacin (<code>GRN</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA19&amp;showdescription=no">J01MA19</a>), gatifloxacin (<code>GAT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA16&amp;showdescription=no">J01MA16</a>), gemifloxacin (<code>GEM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA15&amp;showdescription=no">J01MA15</a>), gentamicin (<code>GEN</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB03&amp;showdescription=no">J01GB03</a>), grepafloxacin (<code>GRX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA11&amp;showdescription=no">J01MA11</a>), hetacillin (<code>HET</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA18&amp;showdescription=no">J01CA18</a>), imipenem (<code>IPM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DH51&amp;showdescription=no">J01DH51</a>), imipenem/relebactam (<code>IMR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DH56&amp;showdescription=no">J01DH56</a>), isepamicin (<code>ISE</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB11&amp;showdescription=no">J01GB11</a>), josamycin (<code>JOS</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA07&amp;showdescription=no">J01FA07</a>), kanamycin (<code>KAN</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB04&amp;showdescription=no">J01GB04</a>), lascufloxacin (<code>LSC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA25&amp;showdescription=no">J01MA25</a>), latamoxef (<code>LTM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DD06&amp;showdescription=no">J01DD06</a>), levofloxacin (<code>LVX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA12&amp;showdescription=no">J01MA12</a>), levonadifloxacin (<code>LND</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA24&amp;showdescription=no">J01MA24</a>), lincomycin (<code>LIN</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FF02&amp;showdescription=no">J01FF02</a>), linezolid (<code>LNZ</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XX08&amp;showdescription=no">J01XX08</a>), lomefloxacin (<code>LOM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA07&amp;showdescription=no">J01MA07</a>), loracarbef (<code>LOR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DC08&amp;showdescription=no">J01DC08</a>), mecillinam (<code>MEC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA11&amp;showdescription=no">J01CA11</a>), meropenem (<code>MEM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DH02&amp;showdescription=no">J01DH02</a>), meropenem/vaborbactam (<code>MEV</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DH52&amp;showdescription=no">J01DH52</a>), metampicillin (<code>MTM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA14&amp;showdescription=no">J01CA14</a>), meticillin (<code>MET</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CF03&amp;showdescription=no">J01CF03</a>), mezlocillin (<code>MEZ</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA10&amp;showdescription=no">J01CA10</a>), micronomicin (<code>MCR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=S01AA22&amp;showdescription=no">S01AA22</a>), midecamycin (<code>MID</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA03&amp;showdescription=no">J01FA03</a>), minocycline (<code>MNO</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01AA08&amp;showdescription=no">J01AA08</a>), miocamycin (<code>MCM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA11&amp;showdescription=no">J01FA11</a>), moxifloxacin (<code>MFX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA14&amp;showdescription=no">J01MA14</a>), nadifloxacin (<code>NAD</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=D10AF05&amp;showdescription=no">D10AF05</a>), nafcillin (<code>NAF</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CF06&amp;showdescription=no">J01CF06</a>), nalidixic acid (<code>NAL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MB02&amp;showdescription=no">J01MB02</a>), neomycin (<code>NEO</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB05&amp;showdescription=no">J01GB05</a>), netilmicin (<code>NET</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB07&amp;showdescription=no">J01GB07</a>), nitrofurantoin (<code>NIT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XE01&amp;showdescription=no">J01XE01</a>), norfloxacin (<code>NOR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA06&amp;showdescription=no">J01MA06</a>), ofloxacin (<code>OFX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA01&amp;showdescription=no">J01MA01</a>), oleandomycin (<code>OLE</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA05&amp;showdescription=no">J01FA05</a>), oritavancin (<code>ORI</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XA05&amp;showdescription=no">J01XA05</a>), oxacillin (<code>OXA</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CF04&amp;showdescription=no">J01CF04</a>), panipenem (<code>PAN</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DH55&amp;showdescription=no">J01DH55</a>), pazufloxacin (<code>PAZ</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA18&amp;showdescription=no">J01MA18</a>), pefloxacin (<code>PEF</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA03&amp;showdescription=no">J01MA03</a>), penamecillin (<code>PNM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CE06&amp;showdescription=no">J01CE06</a>), pheneticillin (<code>PHE</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CE05&amp;showdescription=no">J01CE05</a>), phenoxymethylpenicillin (<code>PHN</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CE02&amp;showdescription=no">J01CE02</a>), piperacillin (<code>PIP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA12&amp;showdescription=no">J01CA12</a>), piperacillin/tazobactam (<code>TZP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CR05&amp;showdescription=no">J01CR05</a>), pivampicillin (<code>PVM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA02&amp;showdescription=no">J01CA02</a>), pivmecillinam (<code>PME</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA08&amp;showdescription=no">J01CA08</a>), plazomicin (<code>PLZ</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB14&amp;showdescription=no">J01GB14</a>), polymyxin B (<code>PLB</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XB02&amp;showdescription=no">J01XB02</a>), pristinamycin (<code>PRI</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FG01&amp;showdescription=no">J01FG01</a>), procaine benzylpenicillin (<code>PRB</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CE09&amp;showdescription=no">J01CE09</a>), propicillin (<code>PRP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CE03&amp;showdescription=no">J01CE03</a>), prulifloxacin (<code>PRU</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA17&amp;showdescription=no">J01MA17</a>), quinupristin/dalfopristin (<code>QDA</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FG02&amp;showdescription=no">J01FG02</a>), ribostamycin (<code>RST</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB10&amp;showdescription=no">J01GB10</a>), rifampicin (<code>RIF</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J04AB02&amp;showdescription=no">J04AB02</a>), rokitamycin (<code>ROK</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA12&amp;showdescription=no">J01FA12</a>), roxithromycin (<code>RXT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA06&amp;showdescription=no">J01FA06</a>), rufloxacin (<code>RFL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA10&amp;showdescription=no">J01MA10</a>), sisomicin (<code>SIS</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB08&amp;showdescription=no">J01GB08</a>), sitafloxacin (<code>SIT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA21&amp;showdescription=no">J01MA21</a>), solithromycin (<code>SOL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA16&amp;showdescription=no">J01FA16</a>), sparfloxacin (<code>SPX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA09&amp;showdescription=no">J01MA09</a>), spiramycin (<code>SPI</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA02&amp;showdescription=no">J01FA02</a>), streptoduocin (<code>STR</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GA02&amp;showdescription=no">J01GA02</a>), streptomycin (<code>STR1</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GA01&amp;showdescription=no">J01GA01</a>), sulbactam (<code>SUL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CG01&amp;showdescription=no">J01CG01</a>), sulbenicillin (<code>SBC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA16&amp;showdescription=no">J01CA16</a>), sulfadiazine (<code>SDI</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EC02&amp;showdescription=no">J01EC02</a>), sulfadiazine/trimethoprim (<code>SLT1</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EE02&amp;showdescription=no">J01EE02</a>), sulfadimethoxine (<code>SUD</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01ED01&amp;showdescription=no">J01ED01</a>), sulfadimidine (<code>SDM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EB03&amp;showdescription=no">J01EB03</a>), sulfadimidine/trimethoprim (<code>SLT2</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EE05&amp;showdescription=no">J01EE05</a>), sulfafurazole (<code>SLF</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EB05&amp;showdescription=no">J01EB05</a>), sulfaisodimidine (<code>SLF1</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EB01&amp;showdescription=no">J01EB01</a>), sulfalene (<code>SLF2</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01ED02&amp;showdescription=no">J01ED02</a>), sulfamazone (<code>SZO</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01ED09&amp;showdescription=no">J01ED09</a>), sulfamerazine (<code>SLF3</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01ED07&amp;showdescription=no">J01ED07</a>), sulfamerazine/trimethoprim (<code>SLT3</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EE07&amp;showdescription=no">J01EE07</a>), sulfamethizole (<code>SLF4</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EB02&amp;showdescription=no">J01EB02</a>), sulfamethoxazole (<code>SMX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EC01&amp;showdescription=no">J01EC01</a>), sulfamethoxypyridazine (<code>SLF5</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01ED05&amp;showdescription=no">J01ED05</a>), sulfametomidine (<code>SLF6</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01ED03&amp;showdescription=no">J01ED03</a>), sulfametoxydiazine (<code>SLF7</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01ED04&amp;showdescription=no">J01ED04</a>), sulfametrole/trimethoprim (<code>SLT4</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EE03&amp;showdescription=no">J01EE03</a>), sulfamoxole (<code>SLF8</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EC03&amp;showdescription=no">J01EC03</a>), sulfamoxole/trimethoprim (<code>SLT5</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EE04&amp;showdescription=no">J01EE04</a>), sulfanilamide (<code>SLF9</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EB06&amp;showdescription=no">J01EB06</a>), sulfaperin (<code>SLF10</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01ED06&amp;showdescription=no">J01ED06</a>), sulfaphenazole (<code>SLF11</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01ED08&amp;showdescription=no">J01ED08</a>), sulfapyridine (<code>SLF12</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EB04&amp;showdescription=no">J01EB04</a>), sulfathiazole (<code>SUT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EB07&amp;showdescription=no">J01EB07</a>), sulfathiourea (<code>SLF13</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EB08&amp;showdescription=no">J01EB08</a>), sultamicillin (<code>SLT6</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CR04&amp;showdescription=no">J01CR04</a>), talampicillin (<code>TAL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA15&amp;showdescription=no">J01CA15</a>), tazobactam (<code>TAZ</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CG02&amp;showdescription=no">J01CG02</a>), tebipenem (<code>TBP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01DH06&amp;showdescription=no">J01DH06</a>), tedizolid (<code>TZD</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XX11&amp;showdescription=no">J01XX11</a>), teicoplanin (<code>TEC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XA02&amp;showdescription=no">J01XA02</a>), telavancin (<code>TLV</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XA03&amp;showdescription=no">J01XA03</a>), telithromycin (<code>TLT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA15&amp;showdescription=no">J01FA15</a>), temafloxacin (<code>TMX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA05&amp;showdescription=no">J01MA05</a>), temocillin (<code>TEM</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA17&amp;showdescription=no">J01CA17</a>), tetracycline (<code>TCY</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01AA07&amp;showdescription=no">J01AA07</a>), ticarcillin (<code>TIC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CA13&amp;showdescription=no">J01CA13</a>), ticarcillin/clavulanic acid (<code>TCC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01CR03&amp;showdescription=no">J01CR03</a>), tigecycline (<code>TGC</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01AA12&amp;showdescription=no">J01AA12</a>), tilbroquinol (<code>TBQ</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=P01AA05&amp;showdescription=no">P01AA05</a>), tobramycin (<code>TOB</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01GB01&amp;showdescription=no">J01GB01</a>), tosufloxacin (<code>TFX</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA22&amp;showdescription=no">J01MA22</a>), trimethoprim (<code>TMP</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EA01&amp;showdescription=no">J01EA01</a>), trimethoprim/sulfamethoxazole (<code>SXT</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01EE01&amp;showdescription=no">J01EE01</a>), troleandomycin (<code>TRL</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01FA08&amp;showdescription=no">J01FA08</a>), trovafloxacin (<code>TVA</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01MA13&amp;showdescription=no">J01MA13</a>), vancomycin (<code>VAN</code>, <a href="https://www.whocc.no/atc_ddd_index/?code=J01XA01&amp;showdescription=no">J01XA01</a>)
</p>


<h3>Interpretation of SIR</h3>

<p>In 2019, the European Committee on Antimicrobial Susceptibility Testing (EUCAST) has decided to change the definitions of susceptibility testing categories S, I, and R as shown below (<a href="https://www.eucast.org/newsiandr">https://www.eucast.org/newsiandr</a>):
</p>

<ul>
<li> <p><strong>S - Susceptible, standard dosing regimen</strong><br />
A microorganism is categorised as &quot;Susceptible, standard dosing regimen&quot;, when there is a high likelihood of therapeutic success using a standard dosing regimen of the agent.
</p>
</li>
<li> <p><strong>I - Susceptible, increased exposure</strong> <em><br />
A microorganism is categorised as &quot;Susceptible, Increased exposure</em>&quot; when there is a high likelihood of therapeutic success because exposure to the agent is increased by adjusting the dosing regimen or by its concentration at the site of infection.
</p>
</li>
<li> <p><strong>R = Resistant</strong><br />
A microorganism is categorised as &quot;Resistant&quot; when there is a high likelihood of therapeutic failure even when there is increased exposure.
</p>

<ul>
<li> <p><em>Exposure</em> is a function of how the mode of administration, dose, dosing interval, infusion time, as well as distribution and excretion of the antimicrobial agent will influence the infecting organism at the site of infection.
</p>
</li></ul>

</li></ul>

<p>This AMR package honours this insight. Use <code><a href="#topic+susceptibility">susceptibility()</a></code> (equal to <code><a href="#topic+proportion_SI">proportion_SI()</a></code>) to determine antimicrobial susceptibility and <code><a href="#topic+count_susceptible">count_susceptible()</a></code> (equal to <code><a href="#topic+count_SI">count_SI()</a></code>) to count susceptible isolates.
</p>


<h3>Source</h3>

<p>See the supported guidelines above for the <a href="base.html#topic+list">list</a> of publications used for this function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- mdro(example_isolates, guideline = "EUCAST")
str(out)
table(out)

out &lt;- mdro(example_isolates,
  guideline = custom_mdro_guideline(
    AMX == "R" ~ "Custom MDRO 1",
    VAN == "R" ~ "Custom MDRO 2"
  )
)
table(out)


if (require("dplyr")) {
  example_isolates %&gt;%
    mdro() %&gt;%
    table()

  # no need to define `x` when used inside dplyr verbs:
  example_isolates %&gt;%
    mutate(MDRO = mdro()) %&gt;%
    pull(MDRO) %&gt;%
    table()
}

</code></pre>

<hr>
<h2 id='mean_amr_distance'>Calculate the Mean AMR Distance</h2><span id='topic+mean_amr_distance'></span><span id='topic+mean_amr_distance.sir'></span><span id='topic+mean_amr_distance.data.frame'></span><span id='topic+amr_distance_from_row'></span>

<h3>Description</h3>

<p>Calculates a normalised mean for antimicrobial resistance between multiple observations, to help to identify similar isolates without comparing antibiograms by hand.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mean_amr_distance(x, ...)

## S3 method for class 'sir'
mean_amr_distance(x, ..., combine_SI = TRUE)

## S3 method for class 'data.frame'
mean_amr_distance(x, ..., combine_SI = TRUE)

amr_distance_from_row(amr_distance, row)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mean_amr_distance_+3A_x">x</code></td>
<td>
<p>a vector of class <a href="#topic+as.sir">sir</a>, <a href="#topic+as.mic">mic</a> or <a href="#topic+as.disk">disk</a>, or a <a href="base.html#topic+data.frame">data.frame</a> containing columns of any of these classes</p>
</td></tr>
<tr><td><code id="mean_amr_distance_+3A_...">...</code></td>
<td>
<p>variables to select (supports <a href="tidyselect.html#topic+language">tidyselect language</a> such as <code>column1:column4</code> and <code>where(is.mic)</code>, and can thus also be <a href="#topic+ab_selector">antibiotic selectors</a></p>
</td></tr>
<tr><td><code id="mean_amr_distance_+3A_combine_si">combine_SI</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether all values of S and I must be merged into one, so the input only consists of S+I vs. R (susceptible vs. resistant) - the default is <code>TRUE</code></p>
</td></tr>
<tr><td><code id="mean_amr_distance_+3A_amr_distance">amr_distance</code></td>
<td>
<p>the outcome of <code><a href="#topic+mean_amr_distance">mean_amr_distance()</a></code></p>
</td></tr>
<tr><td><code id="mean_amr_distance_+3A_row">row</code></td>
<td>
<p>an index, such as a row number</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The mean AMR distance is effectively <a href="https://en.wikipedia.org/wiki/Standard_score">the Z-score</a>; a normalised numeric value to compare AMR test results which can help to identify similar isolates, without comparing antibiograms by hand.
</p>
<p>MIC values (see <code><a href="#topic+as.mic">as.mic()</a></code>) are transformed with <code><a href="base.html#topic+log2">log2()</a></code> first; their distance is thus calculated as <code>(log2(x) - mean(log2(x))) / sd(log2(x))</code>.
</p>
<p>SIR values (see <code><a href="#topic+as.sir">as.sir()</a></code>) are transformed using <code>"S"</code> = 1, <code>"I"</code> = 2, and <code>"R"</code> = 3. If <code>combine_SI</code> is <code>TRUE</code> (default), the <code>"I"</code> will be considered to be 1.
</p>
<p>For data sets, the mean AMR distance will be calculated per column, after which the mean per row will be returned, see <em>Examples</em>.
</p>
<p>Use <code><a href="#topic+amr_distance_from_row">amr_distance_from_row()</a></code> to subtract distances from the distance of one row, see <em>Examples</em>.
</p>


<h3>Interpretation</h3>

<p>Isolates with distances less than 0.01 difference from each other should be considered similar. Differences lower than 0.025 should be considered suspicious.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sir &lt;- random_sir(10)
sir
mean_amr_distance(sir)

mic &lt;- random_mic(10)
mic
mean_amr_distance(mic)
# equal to the Z-score of their log2:
(log2(mic) - mean(log2(mic))) / sd(log2(mic))

disk &lt;- random_disk(10)
disk
mean_amr_distance(disk)

y &lt;- data.frame(
  id = LETTERS[1:10],
  amox = random_sir(10, ab = "amox", mo = "Escherichia coli"),
  cipr = random_disk(10, ab = "cipr", mo = "Escherichia coli"),
  gent = random_mic(10, ab = "gent", mo = "Escherichia coli"),
  tobr = random_mic(10, ab = "tobr", mo = "Escherichia coli")
)
y
mean_amr_distance(y)
y$amr_distance &lt;- mean_amr_distance(y, where(is.mic))
y[order(y$amr_distance), ]

if (require("dplyr")) {
  y %&gt;%
    mutate(
      amr_distance = mean_amr_distance(y),
      check_id_C = amr_distance_from_row(amr_distance, id == "C")
    ) %&gt;%
    arrange(check_id_C)
}
if (require("dplyr")) {
  # support for groups
  example_isolates %&gt;%
    filter(mo_genus() == "Enterococcus" &amp; mo_species() != "") %&gt;%
    select(mo, TCY, carbapenems()) %&gt;%
    group_by(mo) %&gt;%
    mutate(dist = mean_amr_distance(.)) %&gt;%
    arrange(mo, dist)
}
</code></pre>

<hr>
<h2 id='microorganisms'>Data Set with 52 171 Microorganisms</h2><span id='topic+microorganisms'></span>

<h3>Description</h3>

<p>A data set containing the full microbial taxonomy (<strong>last updated: December 11th, 2022</strong>) of five kingdoms from the List of Prokaryotic names with Standing in Nomenclature (LPSN) and the Global Biodiversity Information Facility (GBIF). This data set is the backbone of this <code>AMR</code> package. MO codes can be looked up using <code><a href="#topic+as.mo">as.mo()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>microorganisms
</code></pre>


<h3>Format</h3>

<p>A <a href="tibble.html#topic+tibble">tibble</a> with 52 171 observations and 23 variables:
</p>

<ul>
<li> <p><code>mo</code><br /> ID of microorganism as used by this package. <em>This is a unique identifier.</em>
</p>
</li>
<li> <p><code>fullname</code><br /> Full name, like <code>"Escherichia coli"</code>. For the taxonomic ranks genus, species and subspecies, this is the 'pasted' text of genus, species, and subspecies. For all taxonomic ranks higher than genus, this is the name of the taxon. <em>This is a unique identifier.</em>
</p>
</li>
<li> <p><code>status</code> <br /> Status of the taxon, either &quot;accepted&quot; or &quot;synonym&quot;
</p>
</li>
<li> <p><code>kingdom</code>, <code>phylum</code>, <code>class</code>, <code>order</code>, <code>family</code>, <code>genus</code>, <code>species</code>, <code>subspecies</code><br /> Taxonomic rank of the microorganism
</p>
</li>
<li> <p><code>rank</code><br /> Text of the taxonomic rank of the microorganism, such as <code>"species"</code> or <code>"genus"</code>
</p>
</li>
<li> <p><code>ref</code><br /> Author(s) and year of related scientific publication. This contains only the <em>first surname</em> and year of the <em>latest</em> authors, e.g. &quot;Wallis <em>et al.</em> 2006 <em>emend.</em> Smith and Jones 2018&quot; becomes &quot;Smith <em>et al.</em>, 2018&quot;. This field is directly retrieved from the source specified in the column <code>source</code>. Moreover, accents were removed to comply with CRAN that only allows ASCII characters.
</p>
</li>
<li> <p><code>lpsn</code><br /> Identifier ('Record number') of the List of Prokaryotic names with Standing in Nomenclature (LPSN). This will be the first/highest LPSN identifier to keep one identifier per row. For example, <em>Acetobacter ascendens</em> has LPSN Record number 7864 and 11011. Only the first is available in the <code>microorganisms</code> data set.
</p>
</li>
<li> <p><code>oxygen_tolerance</code> <br /> Oxygen tolerance, either &quot;aerobe&quot;, &quot;anaerobe&quot;, &quot;anaerobe/microaerophile&quot;, &quot;facultative anaerobe&quot;, &quot;likely facultative anaerobe&quot;, or &quot;microaerophile&quot;. These data were retrieved from BacDive (see <em>Source</em>). Items that contain &quot;likely&quot; are missing from BacDive and were extrapolated from other species within the same genus to guess the oxygen tolerance. Currently 73.4% of all ~37 000 bacteria in the data set contain an oxygen tolerance.
</p>
</li>
<li> <p><code>lpsn_parent</code><br /> LPSN identifier of the parent taxon
</p>
</li>
<li> <p><code>lpsn_renamed_to</code><br /> LPSN identifier of the currently valid taxon
</p>
</li>
<li> <p><code>gbif</code><br /> Identifier ('taxonID') of the Global Biodiversity Information Facility (GBIF)
</p>
</li>
<li> <p><code>gbif_parent</code><br /> GBIF identifier of the parent taxon
</p>
</li>
<li> <p><code>gbif_renamed_to</code><br /> GBIF identifier of the currently valid taxon
</p>
</li>
<li> <p><code>source</code><br /> Either &quot;GBIF&quot;, &quot;LPSN&quot;, or &quot;manually added&quot; (see <em>Source</em>)
</p>
</li>
<li> <p><code>prevalence</code><br /> Prevalence of the microorganism according to Bartlett <em>et al.</em> (2022, <a href="https://doi.org/10.1099/mic.0.001269">doi:10.1099/mic.0.001269</a>), see <code><a href="#topic+mo_matching_score">mo_matching_score()</a></code> for the full explanation
</p>
</li>
<li> <p><code>snomed</code><br /> Systematized Nomenclature of Medicine (SNOMED) code of the microorganism, version of July 1st, 2021 (see <em>Source</em>). Use <code><a href="#topic+mo_snomed">mo_snomed()</a></code> to retrieve it quickly, see <code><a href="#topic+mo_property">mo_property()</a></code>.
</p>
</li></ul>



<h3>Details</h3>

<p>Please note that entries are only based on the List of Prokaryotic names with Standing in Nomenclature (LPSN) and the Global Biodiversity Information Facility (GBIF) (see below). Since these sources incorporate entries based on (recent) publications in the International Journal of Systematic and Evolutionary Microbiology (IJSEM), it can happen that the year of publication is sometimes later than one might expect.
</p>
<p>For example, <em>Staphylococcus pettenkoferi</em> was described for the first time in Diagnostic Microbiology and Infectious Disease in 2002 (<a href="https://doi.org/10.1016/s0732-8893%2802%2900399-1">doi:10.1016/s0732-8893(02)00399-1</a>), but it was not before 2007 that a publication in IJSEM followed (<a href="https://doi.org/10.1099/ijs.0.64381-0">doi:10.1099/ijs.0.64381-0</a>). Consequently, the <code>AMR</code> package returns 2007 for <code>mo_year("S. pettenkoferi")</code>.
</p>


<h3>Included Taxa</h3>

<p>Included taxonomic data are:
</p>

<ul>
<li><p> All ~37 000 (sub)species from the kingdoms of Archaea and Bacteria
</p>
</li>
<li><p> ~7 900 (sub)species from the kingdom of Fungi. The kingdom of Fungi is a very large taxon with almost 300,000 different (sub)species, of which most are not microbial (but rather macroscopic, like mushrooms). Because of this, not all fungi fit the scope of this package. Only relevant fungi are covered (such as all species of <em>Aspergillus</em>, <em>Candida</em>, <em>Cryptococcus</em>, <em>Histoplasma</em>, <em>Pneumocystis</em>, <em>Saccharomyces</em> and <em>Trichophyton</em>).
</p>
</li>
<li><p> ~5 100 (sub)species from the kingdom of Protozoa
</p>
</li>
<li><p> ~1 400 (sub)species from 43 other relevant genera from the kingdom of Animalia (such as <em>Strongyloides</em> and <em>Taenia</em>)
</p>
</li>
<li><p> All ~9 800 previously accepted names of all included (sub)species (these were taxonomically renamed)
</p>
</li>
<li><p> The complete taxonomic tree of all included (sub)species: from kingdom to subspecies
</p>
</li>
<li><p> The identifier of the parent taxons
</p>
</li>
<li><p> The year and first author of the related scientific publication
</p>
</li></ul>



<h4>Manual additions</h4>

<p>For convenience, some entries were added manually:
</p>

<ul>
<li><p> ~1 500 entries of <em>Salmonella</em>, such as the city-like serovars and groups A to H
</p>
</li>
<li><p> 36 species groups (such as the beta-haemolytic <em>Streptococcus</em> groups A to K, coagulase-negative <em>Staphylococcus</em> (CoNS), <em>Mycobacterium tuberculosis</em> complex, etc.), of which the group compositions are stored in the <a href="#topic+microorganisms.groups">microorganisms.groups</a> data set
</p>
</li>
<li><p> 1 entry of <em>Blastocystis</em> (<em>B.  hominis</em>), although it officially does not exist (Noel <em>et al.</em> 2005, PMID 15634993)
</p>
</li>
<li><p> 1 entry of <em>Moraxella</em> (<em>M. catarrhalis</em>), which was formally named <em>Branhamella catarrhalis</em> (Catlin, 1970) though this change was never accepted within the field of clinical microbiology
</p>
</li>
<li><p> 8 other 'undefined' entries (unknown, unknown Gram-negatives, unknown Gram-positives, unknown yeast, unknown fungus, and unknown anaerobic Gram-pos/Gram-neg bacteria)
</p>
</li></ul>

<p>The syntax used to transform the original data to a cleansed <span class="rlang"><b>R</b></span> format, can be found here: <a href="https://github.com/msberends/AMR/blob/main/data-raw/reproduction_of_microorganisms.R">https://github.com/msberends/AMR/blob/main/data-raw/reproduction_of_microorganisms.R</a>.
</p>



<h4>Direct download</h4>

<p>Like all data sets in this package, this data set is publicly available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>.
</p>



<h3>About the Records from LPSN (see <em>Source</em>)</h3>

<p>LPSN is the main source for bacteriological taxonomy of this <code>AMR</code> package.
</p>
<p>The List of Prokaryotic names with Standing in Nomenclature (LPSN) provides comprehensive information on the nomenclature of prokaryotes. LPSN is a free to use service founded by Jean P. Euzeby in 1997 and later on maintained by Aidan C. Parte.
</p>


<h3>Source</h3>


<ul>
<li><p> Parte, AC <em>et al.</em> (2020). <strong>List of Prokaryotic names with Standing in Nomenclature (LPSN) moves to the DSMZ.</strong> International Journal of Systematic and Evolutionary Microbiology, 70, 5607-5612; <a href="https://doi.org/10.1099/ijsem.0.004332">doi:10.1099/ijsem.0.004332</a>. Accessed from <a href="https://lpsn.dsmz.de">https://lpsn.dsmz.de</a> on December 11th, 2022.
</p>
</li>
<li><p> GBIF Secretariat (2022). GBIF Backbone Taxonomy. Checklist dataset <a href="https://doi.org/10.15468/39omei">doi:10.15468/39omei</a>. Accessed from <a href="https://www.gbif.org">https://www.gbif.org</a> on December 11th, 2022.
</p>
</li>
<li><p> Reimer, LC <em>et al.</em> (2022). <strong><em>BacDive</em> in 2022: the knowledge base for standardized bacterial and archaeal data.</strong> Nucleic Acids Res., 50(D1):D741-D74; <a href="https://doi.org/10.1093/nar/gkab961">doi:10.1093/nar/gkab961</a>. Accessed from <a href="https://bacdive.dsmz.de">https://bacdive.dsmz.de</a> on May 12th, 2023.
</p>
</li>
<li><p> Public Health Information Network Vocabulary Access and Distribution System (PHIN VADS). US Edition of SNOMED CT from 1 September 2020. Value Set Name 'Microorganism', OID 2.16.840.1.114222.4.11.1009 (v12). URL: <a href="https://phinvads.cdc.gov">https://phinvads.cdc.gov</a>
</p>
</li>
<li><p> Grimont <em>et al.</em> (2007). Antigenic Formulae of the Salmonella Serovars, 9th Edition. WHO Collaborating Centre for Reference and Research on <em>Salmonella</em> (WHOCC-SALM).
</p>
</li>
<li><p> Bartlett <em>et al.</em> (2022). <strong>A comprehensive list of bacterial pathogens infecting humans</strong> <em>Microbiology</em> 168:001269; <a href="https://doi.org/10.1099/mic.0.001269">doi:10.1099/mic.0.001269</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+as.mo">as.mo()</a></code>, <code><a href="#topic+mo_property">mo_property()</a></code>, <a href="#topic+microorganisms.groups">microorganisms.groups</a>, <a href="#topic+microorganisms.codes">microorganisms.codes</a>, <a href="#topic+intrinsic_resistant">intrinsic_resistant</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>microorganisms
</code></pre>

<hr>
<h2 id='microorganisms.codes'>Data Set with 4 957 Common Microorganism Codes</h2><span id='topic+microorganisms.codes'></span>

<h3>Description</h3>

<p>A data set containing commonly used codes for microorganisms, from laboratory systems and <a href="https://whonet.org">WHONET</a>. Define your own with <code><a href="#topic+set_mo_source">set_mo_source()</a></code>. They will all be searched when using <code><a href="#topic+as.mo">as.mo()</a></code> and consequently all the <code><a href="#topic+mo_property">mo_*</a></code> functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>microorganisms.codes
</code></pre>


<h3>Format</h3>

<p>A <a href="tibble.html#topic+tibble">tibble</a> with 4 957 observations and 2 variables:
</p>

<ul>
<li> <p><code>code</code><br /> Commonly used code of a microorganism. <em>This is a unique identifier.</em>
</p>
</li>
<li> <p><code>mo</code><br /> ID of the microorganism in the <a href="#topic+microorganisms">microorganisms</a> data set
</p>
</li></ul>



<h3>Details</h3>

<p>Like all data sets in this package, this data set is publicly available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+as.mo">as.mo()</a></code> <a href="#topic+microorganisms">microorganisms</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>microorganisms.codes

# 'ECO' or 'eco' is the WHONET code for E. coli:
microorganisms.codes[microorganisms.codes$code == "ECO", ]

# and therefore, 'eco' will be understood as E. coli in this package:
mo_info("eco")

# works for all AMR functions:
mo_is_intrinsic_resistant("eco", ab = "vancomycin")
</code></pre>

<hr>
<h2 id='microorganisms.groups'>Data Set with 521 Microorganisms In Species Groups</h2><span id='topic+microorganisms.groups'></span>

<h3>Description</h3>

<p>A data set containing species groups and microbiological complexes, which are used in <a href="#topic+clinical_breakpoints">the clinical breakpoints table</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>microorganisms.groups
</code></pre>


<h3>Format</h3>

<p>A <a href="tibble.html#topic+tibble">tibble</a> with 521 observations and 4 variables:
</p>

<ul>
<li> <p><code>mo_group</code><br /> ID of the species group / microbiological complex
</p>
</li>
<li> <p><code>mo</code><br /> ID of the microorganism belonging in the species group / microbiological complex
</p>
</li>
<li> <p><code>mo_group_name</code><br /> Name of the species group / microbiological complex, as retrieved with <code><a href="#topic+mo_name">mo_name()</a></code>
</p>
</li>
<li> <p><code>mo_name</code><br /> Name of the microorganism belonging in the species group / microbiological complex, as retrieved with <code><a href="#topic+mo_name">mo_name()</a></code>
</p>
</li></ul>



<h3>Details</h3>

<p>Like all data sets in this package, this data set is publicly available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+as.mo">as.mo()</a></code> <a href="#topic+microorganisms">microorganisms</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>microorganisms.groups

# these are all species in the Bacteroides fragilis group, as per WHONET:
microorganisms.groups[microorganisms.groups$mo_group == "B_BCTRD_FRGL-C", ]
</code></pre>

<hr>
<h2 id='mo_matching_score'>Calculate the Matching Score for Microorganisms</h2><span id='topic+mo_matching_score'></span>

<h3>Description</h3>

<p>This algorithm is used by <code><a href="#topic+as.mo">as.mo()</a></code> and all the <code><a href="#topic+mo_property">mo_*</a></code> functions to determine the most probable match of taxonomic records based on user input.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mo_matching_score(x, n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mo_matching_score_+3A_x">x</code></td>
<td>
<p>Any user input value(s)</p>
</td></tr>
<tr><td><code id="mo_matching_score_+3A_n">n</code></td>
<td>
<p>A full taxonomic name, that exists in <code><a href="#topic+microorganisms">microorganisms$fullname</a></code></p>
</td></tr>
</table>


<h3>Matching Score for Microorganisms</h3>

<p>With ambiguous user input in <code><a href="#topic+as.mo">as.mo()</a></code> and all the <code><a href="#topic+mo_property">mo_*</a></code> functions, the returned results are chosen based on their matching score using <code><a href="#topic+mo_matching_score">mo_matching_score()</a></code>. This matching score <code class="reqn">m</code>, is calculated as:
</p>

<p><img src="../help/figures/mo_matching_score.png" width="300" alt="mo matching score" />
</p>
<p>where:
</p>

<ul>
<li> <p><code class="reqn">x</code> is the user input;
</p>
</li>
<li> <p><code class="reqn">n</code> is a taxonomic name (genus, species, and subspecies);
</p>
</li>
<li> <p><code class="reqn">l_n</code> is the length of <code class="reqn">n</code>;
</p>
</li>
<li> <p><code class="reqn">lev</code> is the <a href="https://en.wikipedia.org/wiki/Levenshtein_distance">Levenshtein distance function</a> (counting any insertion as 1, and any deletion or substitution as 2) that is needed to change <code class="reqn">x</code> into <code class="reqn">n</code>;
</p>
</li>
<li> <p><code class="reqn">p_n</code> is the human pathogenic prevalence group of <code class="reqn">n</code>, as described below;
</p>
</li>
<li> <p><code class="reqn">k_n</code> is the taxonomic kingdom of <code class="reqn">n</code>, set as Bacteria = 1, Fungi = 1.25, Protozoa = 1.5, Archaea = 2, others = 3.
</p>
</li></ul>

<p>The grouping into human pathogenic prevalence <code class="reqn">p</code> is based on recent work from Bartlett <em>et al.</em> (2022, <a href="https://doi.org/10.1099/mic.0.001269">doi:10.1099/mic.0.001269</a>) who extensively studied medical-scientific literature to categorise all bacterial species into these groups:
</p>

<ul>
<li> <p><strong>Established</strong>, if a taxonomic species has infected at least three persons in three or more references. These records have <code>prevalence = 1.0</code> in the <a href="#topic+microorganisms">microorganisms</a> data set;
</p>
</li>
<li> <p><strong>Putative</strong>, if a taxonomic species has fewer than three known cases. These records have <code>prevalence = 1.25</code> in the <a href="#topic+microorganisms">microorganisms</a> data set.
</p>
</li></ul>

<p>Furthermore,
</p>

<ul>
<li><p> Any genus present in the <strong>established</strong> list also has <code>prevalence = 1.0</code> in the <a href="#topic+microorganisms">microorganisms</a> data set;
</p>
</li>
<li><p> Any other genus present in the <strong>putative</strong> list has <code>prevalence = 1.25</code> in the <a href="#topic+microorganisms">microorganisms</a> data set;
</p>
</li>
<li><p> Any other species or subspecies of which the genus is present in the two aforementioned groups, has <code>prevalence = 1.5</code> in the <a href="#topic+microorganisms">microorganisms</a> data set;
</p>
</li>
<li><p> Any <em>non-bacterial</em> genus, species or subspecies of which the genus is present in the following list, has <code>prevalence = 1.25</code> in the <a href="#topic+microorganisms">microorganisms</a> data set: <em>Absidia</em>, <em>Acanthamoeba</em>, <em>Acremonium</em>, <em>Aedes</em>, <em>Alternaria</em>, <em>Amoeba</em>, <em>Ancylostoma</em>, <em>Angiostrongylus</em>, <em>Anisakis</em>, <em>Anopheles</em>, <em>Apophysomyces</em>, <em>Aspergillus</em>, <em>Aureobasidium</em>, <em>Basidiobolus</em>, <em>Beauveria</em>, <em>Blastocystis</em>, <em>Blastomyces</em>, <em>Candida</em>, <em>Capillaria</em>, <em>Chaetomium</em>, <em>Chrysonilia</em>, <em>Cladophialophora</em>, <em>Cladosporium</em>, <em>Conidiobolus</em>, <em>Contracaecum</em>, <em>Cordylobia</em>, <em>Cryptococcus</em>, <em>Curvularia</em>, <em>Demodex</em>, <em>Dermatobia</em>, <em>Dientamoeba</em>, <em>Diphyllobothrium</em>, <em>Dirofilaria</em>, <em>Echinostoma</em>, <em>Entamoeba</em>, <em>Enterobius</em>, <em>Exophiala</em>, <em>Exserohilum</em>, <em>Fasciola</em>, <em>Fonsecaea</em>, <em>Fusarium</em>, <em>Giardia</em>, <em>Haloarcula</em>, <em>Halobacterium</em>, <em>Halococcus</em>, <em>Hendersonula</em>, <em>Heterophyes</em>, <em>Histomonas</em>, <em>Histoplasma</em>, <em>Hymenolepis</em>, <em>Hypomyces</em>, <em>Hysterothylacium</em>, <em>Leishmania</em>, <em>Malassezia</em>, <em>Malbranchea</em>, <em>Metagonimus</em>, <em>Meyerozyma</em>, <em>Microsporidium</em>, <em>Microsporum</em>, <em>Mortierella</em>, <em>Mucor</em>, <em>Mycocentrospora</em>, <em>Necator</em>, <em>Nectria</em>, <em>Ochroconis</em>, <em>Oesophagostomum</em>, <em>Oidiodendron</em>, <em>Opisthorchis</em>, <em>Pediculus</em>, <em>Penicillium</em>, <em>Phlebotomus</em>, <em>Phoma</em>, <em>Pichia</em>, <em>Piedraia</em>, <em>Pithomyces</em>, <em>Pityrosporum</em>, <em>Pneumocystis</em>, <em>Pseudallescheria</em>, <em>Pseudoterranova</em>, <em>Pulex</em>, <em>Rhizomucor</em>, <em>Rhizopus</em>, <em>Rhodotorula</em>, <em>Saccharomyces</em>, <em>Sarcoptes</em>, <em>Scolecobasidium</em>, <em>Scopulariopsis</em>, <em>Scytalidium</em>, <em>Spirometra</em>, <em>Sporobolomyces</em>, <em>Stachybotrys</em>, <em>Strongyloides</em>, <em>Syngamus</em>, <em>Taenia</em>, <em>Talaromyces</em>, <em>Toxocara</em>, <em>Trichinella</em>, <em>Trichobilharzia</em>, <em>Trichoderma</em>, <em>Trichomonas</em>, <em>Trichophyton</em>, <em>Trichosporon</em>, <em>Trichostrongylus</em>, <em>Trichuris</em>, <em>Tritirachium</em>, <em>Trombicula</em>, <em>Trypanosoma</em>, <em>Tunga</em>, or <em>Wuchereria</em>;
</p>
</li>
<li><p> All other records have <code>prevalence = 2.0</code> in the <a href="#topic+microorganisms">microorganisms</a> data set.
</p>
</li></ul>

<p>When calculating the matching score, all characters in <code class="reqn">x</code> and <code class="reqn">n</code> are ignored that are other than A-Z, a-z, 0-9, spaces and parentheses.
</p>
<p>All matches are sorted descending on their matching score and for all user input values, the top match will be returned. This will lead to the effect that e.g., <code>"E. coli"</code> will return the microbial ID of <em>Escherichia coli</em> (<code class="reqn">m = 0.688</code>, a highly prevalent microorganism found in humans) and not <em>Entamoeba coli</em> (<code class="reqn">m = 0.381</code>, a less prevalent microorganism in humans), although the latter would alphabetically come first.
</p>


<h3>Reference Data Publicly Available</h3>

<p>All data sets in this <code>AMR</code> package (about microorganisms, antibiotics, SIR interpretation, EUCAST rules, etc.) are publicly and freely available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. We also provide tab-separated plain text files that are machine-readable and suitable for input in any software program, such as laboratory information systems. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>.
</p>


<h3>Note</h3>

<p>This algorithm was originally described in: Berends MS <em>et al.</em> (2022). <strong>AMR: An R Package for Working with Antimicrobial Resistance Data</strong>. <em>Journal of Statistical Software</em>, 104(3), 1-31; <a href="https://doi.org/10.18637/jss.v104.i03">doi:10.18637/jss.v104.i03</a>.
</p>
<p>Later, the work of Bartlett A <em>et al.</em> about bacterial pathogens infecting humans (2022, <a href="https://doi.org/10.1099/mic.0.001269">doi:10.1099/mic.0.001269</a>) was incorporated.
</p>


<h3>Author(s)</h3>

<p>Dr. Matthijs Berends, 2018
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mo_reset_session()

as.mo("E. coli")
mo_uncertainties()

mo_matching_score(
  x = "E. coli",
  n = c("Escherichia coli", "Entamoeba coli")
)
</code></pre>

<hr>
<h2 id='mo_property'>Get Properties of a Microorganism</h2><span id='topic+mo_property'></span><span id='topic+mo_name'></span><span id='topic+mo_fullname'></span><span id='topic+mo_shortname'></span><span id='topic+mo_subspecies'></span><span id='topic+mo_species'></span><span id='topic+mo_genus'></span><span id='topic+mo_family'></span><span id='topic+mo_order'></span><span id='topic+mo_class'></span><span id='topic+mo_phylum'></span><span id='topic+mo_kingdom'></span><span id='topic+mo_domain'></span><span id='topic+mo_type'></span><span id='topic+mo_status'></span><span id='topic+mo_pathogenicity'></span><span id='topic+mo_gramstain'></span><span id='topic+mo_is_gram_negative'></span><span id='topic+mo_is_gram_positive'></span><span id='topic+mo_is_yeast'></span><span id='topic+mo_is_intrinsic_resistant'></span><span id='topic+mo_oxygen_tolerance'></span><span id='topic+mo_is_anaerobic'></span><span id='topic+mo_snomed'></span><span id='topic+mo_ref'></span><span id='topic+mo_authors'></span><span id='topic+mo_year'></span><span id='topic+mo_lpsn'></span><span id='topic+mo_gbif'></span><span id='topic+mo_rank'></span><span id='topic+mo_taxonomy'></span><span id='topic+mo_synonyms'></span><span id='topic+mo_current'></span><span id='topic+mo_info'></span><span id='topic+mo_url'></span>

<h3>Description</h3>

<p>Use these functions to return a specific property of a microorganism based on the latest accepted taxonomy. All input values will be evaluated internally with <code><a href="#topic+as.mo">as.mo()</a></code>, which makes it possible to use microbial abbreviations, codes and names as input. See <em>Examples</em>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mo_name(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_fullname(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_shortname(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_subspecies(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_species(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_genus(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_family(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_order(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_class(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_phylum(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_kingdom(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_domain(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_type(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_status(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_pathogenicity(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_gramstain(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_is_gram_negative(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_is_gram_positive(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_is_yeast(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_is_intrinsic_resistant(
  x,
  ab,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_oxygen_tolerance(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_is_anaerobic(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_snomed(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_ref(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_authors(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_year(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_lpsn(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_gbif(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_rank(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_taxonomy(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_synonyms(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_current(x, language = get_AMR_locale(), ...)

mo_info(
  x,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_url(
  x,
  open = FALSE,
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)

mo_property(
  x,
  property = "fullname",
  language = get_AMR_locale(),
  keep_synonyms = getOption("AMR_keep_synonyms", FALSE),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mo_property_+3A_x">x</code></td>
<td>
<p>any <a href="base.html#topic+character">character</a> (vector) that can be coerced to a valid microorganism code with <code><a href="#topic+as.mo">as.mo()</a></code>. Can be left blank for auto-guessing the column containing microorganism codes if used in a data set, see <em>Examples</em>.</p>
</td></tr>
<tr><td><code id="mo_property_+3A_language">language</code></td>
<td>
<p>language to translate text like &quot;no growth&quot;, which defaults to the system language (see <code><a href="#topic+get_AMR_locale">get_AMR_locale()</a></code>)</p>
</td></tr>
<tr><td><code id="mo_property_+3A_keep_synonyms">keep_synonyms</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate if old, previously valid taxonomic names must be preserved and not be corrected to currently accepted names. The default is <code>FALSE</code>, which will return a note if old taxonomic names were processed. The default can be set with the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_keep_synonyms</a></code>, i.e. <code>options(AMR_keep_synonyms = TRUE)</code> or <code>options(AMR_keep_synonyms = FALSE)</code>.</p>
</td></tr>
<tr><td><code id="mo_property_+3A_...">...</code></td>
<td>
<p>other arguments passed on to <code><a href="#topic+as.mo">as.mo()</a></code>, such as 'minimum_matching_score', 'ignore_pattern', and 'remove_from_input'</p>
</td></tr>
<tr><td><code id="mo_property_+3A_ab">ab</code></td>
<td>
<p>any (vector of) text that can be coerced to a valid antibiotic drug code with <code><a href="#topic+as.ab">as.ab()</a></code></p>
</td></tr>
<tr><td><code id="mo_property_+3A_open">open</code></td>
<td>
<p>browse the URL using <code><a href="utils.html#topic+browseURL">browseURL()</a></code></p>
</td></tr>
<tr><td><code id="mo_property_+3A_property">property</code></td>
<td>
<p>one of the column names of the <a href="#topic+microorganisms">microorganisms</a> data set: &quot;mo&quot;, &quot;fullname&quot;, &quot;status&quot;, &quot;kingdom&quot;, &quot;phylum&quot;, &quot;class&quot;, &quot;order&quot;, &quot;family&quot;, &quot;genus&quot;, &quot;species&quot;, &quot;subspecies&quot;, &quot;rank&quot;, &quot;ref&quot;, &quot;oxygen_tolerance&quot;, &quot;source&quot;, &quot;lpsn&quot;, &quot;lpsn_parent&quot;, &quot;lpsn_renamed_to&quot;, &quot;gbif&quot;, &quot;gbif_parent&quot;, &quot;gbif_renamed_to&quot;, &quot;prevalence&quot;, or &quot;snomed&quot;, or must be <code>"shortname"</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>All functions will, at default, <strong>not</strong> keep old taxonomic properties, as synonyms are automatically replaced with the current taxonomy. Take for example <em>Enterobacter aerogenes</em>, which was initially named in 1960 but renamed to <em>Klebsiella aerogenes</em> in 2017:
</p>

<ul>
<li> <p><code>mo_genus("Enterobacter aerogenes")</code> will return <code>"Klebsiella"</code> (with a note about the renaming)
</p>
</li>
<li> <p><code>mo_genus("Enterobacter aerogenes", keep_synonyms = TRUE)</code> will return <code>"Enterobacter"</code> (with a once-per-session warning that the name is outdated)
</p>
</li>
<li> <p><code>mo_ref("Enterobacter aerogenes")</code> will return <code>"Tindall et al., 2017"</code> (with a note)
</p>
</li>
<li> <p><code>mo_ref("Enterobacter aerogenes", keep_synonyms = TRUE)</code> will return <code>"Hormaeche et al., 1960"</code> (with a warning)
</p>
</li></ul>

<p>The short name (<code><a href="#topic+mo_shortname">mo_shortname()</a></code>) returns the first character of the genus and the full species, such as <code>"E. coli"</code>, for species and subspecies. Exceptions are abbreviations of staphylococci (such as <em>&quot;CoNS&quot;</em>, Coagulase-Negative Staphylococci) and beta-haemolytic streptococci (such as <em>&quot;GBS&quot;</em>, Group B Streptococci). Please bear in mind that e.g. <em>E. coli</em> could mean <em>Escherichia coli</em> (kingdom of Bacteria) as well as <em>Entamoeba coli</em> (kingdom of Protozoa). Returning to the full name will be done using <code><a href="#topic+as.mo">as.mo()</a></code> internally, giving priority to bacteria and human pathogens, i.e. <code>"E. coli"</code> will be considered <em>Escherichia coli</em>. As a result, <code>mo_fullname(mo_shortname("Entamoeba coli"))</code> returns <code>"Escherichia coli"</code>.
</p>
<p>Since the top-level of the taxonomy is sometimes referred to as 'kingdom' and sometimes as 'domain', the functions <code><a href="#topic+mo_kingdom">mo_kingdom()</a></code> and <code><a href="#topic+mo_domain">mo_domain()</a></code> return the exact same results.
</p>
<p>Determination of human pathogenicity (<code><a href="#topic+mo_pathogenicity">mo_pathogenicity()</a></code>) is strongly based on Bartlett <em>et al.</em> (2022, <a href="https://doi.org/10.1099/mic.0.001269">doi:10.1099/mic.0.001269</a>). This function returns a <a href="base.html#topic+factor">factor</a> with the levels <em>Pathogenic</em>, <em>Potentially pathogenic</em>, <em>Non-pathogenic</em>, and <em>Unknown</em>.
</p>
<p>Determination of the Gram stain (<code><a href="#topic+mo_gramstain">mo_gramstain()</a></code>) will be based on the taxonomic kingdom and phylum. Originally, Cavalier-Smith defined the so-called subkingdoms Negibacteria and Posibacteria (2002, <a href="https://pubmed.ncbi.nlm.nih.gov/11837318/">PMID 11837318</a>), and only considered these phyla as Posibacteria: Actinobacteria, Chloroflexi, Firmicutes, and Tenericutes. These phyla were later renamed to Actinomycetota, Chloroflexota, Bacillota, and Mycoplasmatota (2021, <a href="https://pubmed.ncbi.nlm.nih.gov/34694987/">PMID 34694987</a>). Bacteria in these phyla are considered Gram-positive in this <code>AMR</code> package, except for members of the class Negativicutes (within phylum Bacillota) which are Gram-negative. All other bacteria are considered Gram-negative. Species outside the kingdom of Bacteria will return a value <code>NA</code>. Functions <code><a href="#topic+mo_is_gram_negative">mo_is_gram_negative()</a></code> and <code><a href="#topic+mo_is_gram_positive">mo_is_gram_positive()</a></code> always return <code>TRUE</code> or <code>FALSE</code> (or <code>NA</code> when the input is <code>NA</code> or the MO code is <code>UNKNOWN</code>), thus always return <code>FALSE</code> for species outside the taxonomic kingdom of Bacteria.
</p>
<p>Determination of yeasts (<code><a href="#topic+mo_is_yeast">mo_is_yeast()</a></code>) will be based on the taxonomic kingdom and class. <em>Budding yeasts</em> are fungi of the phylum Ascomycota, class Saccharomycetes (also called Hemiascomycetes). <em>True yeasts</em> are aggregated into the underlying order Saccharomycetales. Thus, for all microorganisms that are member of the taxonomic class Saccharomycetes, the function will return <code>TRUE</code>. It returns <code>FALSE</code> otherwise (or <code>NA</code> when the input is <code>NA</code> or the MO code is <code>UNKNOWN</code>).
</p>
<p>Determination of intrinsic resistance (<code><a href="#topic+mo_is_intrinsic_resistant">mo_is_intrinsic_resistant()</a></code>) will be based on the <a href="#topic+intrinsic_resistant">intrinsic_resistant</a> data set, which is based on <a href="https://www.eucast.org/expert_rules_and_expected_phenotypes">'EUCAST Expert Rules' and 'EUCAST Intrinsic Resistance and Unusual Phenotypes' v3.3</a> (2021). The <code><a href="#topic+mo_is_intrinsic_resistant">mo_is_intrinsic_resistant()</a></code> function can be vectorised over both argument <code>x</code> (input for microorganisms) and <code>ab</code> (input for antibiotics).
</p>
<p>Determination of bacterial oxygen tolerance (<code><a href="#topic+mo_oxygen_tolerance">mo_oxygen_tolerance()</a></code>) will be based on BacDive, see <em>Source</em>. The function <code><a href="#topic+mo_is_anaerobic">mo_is_anaerobic()</a></code> only returns <code>TRUE</code> if the oxygen tolerance is <code>"anaerobe"</code>, indicting an obligate anaerobic species or genus. It always returns <code>FALSE</code> for species outside the taxonomic kingdom of Bacteria.
</p>
<p>The function <code><a href="#topic+mo_url">mo_url()</a></code> will return the direct URL to the online database entry, which also shows the scientific reference of the concerned species.
</p>
<p>SNOMED codes (<code><a href="#topic+mo_snomed">mo_snomed()</a></code>) are from the version of July 1st, 2021. See <em>Source</em> and the <a href="#topic+microorganisms">microorganisms</a> data set for more info.
</p>
<p>Old taxonomic names (so-called 'synonyms') can be retrieved with <code><a href="#topic+mo_synonyms">mo_synonyms()</a></code> (which will have the scientific reference as <a href="base.html#topic+names">name</a>), the current taxonomic name can be retrieved with <code><a href="#topic+mo_current">mo_current()</a></code>. Both functions return full names.
</p>
<p>All output <a href="#topic+translate">will be translated</a> where possible.
</p>


<h3>Value</h3>


<ul>
<li><p> An <a href="base.html#topic+integer">integer</a> in case of <code><a href="#topic+mo_year">mo_year()</a></code>
</p>
</li>
<li><p> An <a href="base.html#topic+factor">ordered factor</a> in case of <code><a href="#topic+mo_pathogenicity">mo_pathogenicity()</a></code>
</p>
</li>
<li><p> A <a href="base.html#topic+list">list</a> in case of <code><a href="#topic+mo_taxonomy">mo_taxonomy()</a></code>, <code><a href="#topic+mo_synonyms">mo_synonyms()</a></code>, <code><a href="#topic+mo_snomed">mo_snomed()</a></code> and <code><a href="#topic+mo_info">mo_info()</a></code>
</p>
</li>
<li><p> A named <a href="base.html#topic+character">character</a> in case of <code><a href="#topic+mo_url">mo_url()</a></code>
</p>
</li>
<li><p> A <a href="base.html#topic+character">character</a> in all other cases
</p>
</li></ul>



<h3>Matching Score for Microorganisms</h3>

<p>This function uses <code><a href="#topic+as.mo">as.mo()</a></code> internally, which uses an advanced algorithm to translate arbitrary user input to valid taxonomy using a so-called matching score. You can read about this public algorithm on the <a href="#topic+mo_matching_score">MO matching score page</a>.
</p>


<h3>Source</h3>


<ol>
<li><p> Berends MS <em>et al.</em> (2022). <strong>AMR: An R Package for Working with Antimicrobial Resistance Data</strong>. <em>Journal of Statistical Software</em>, 104(3), 1-31; <a href="https://doi.org/10.18637/jss.v104.i03">doi:10.18637/jss.v104.i03</a>
</p>
</li>
<li><p> Becker K <em>et al.</em> (2014). <strong>Coagulase-Negative Staphylococci.</strong> <em>Clin Microbiol Rev.</em> 27(4): 870-926; <a href="https://doi.org/10.1128/CMR.00109-13">doi:10.1128/CMR.00109-13</a>
</p>
</li>
<li><p> Becker K <em>et al.</em> (2019). <strong>Implications of identifying the recently defined members of the <em>S. aureus</em> complex, <em>S. argenteus</em> and <em>S. schweitzeri</em>: A position paper of members of the ESCMID Study Group for staphylococci and Staphylococcal Diseases (ESGS).</strong> <em>Clin Microbiol Infect</em>; <a href="https://doi.org/10.1016/j.cmi.2019.02.028">doi:10.1016/j.cmi.2019.02.028</a>
</p>
</li>
<li><p> Becker K <em>et al.</em> (2020). <strong>Emergence of coagulase-negative staphylococci.</strong> <em>Expert Rev Anti Infect Ther.</em> 18(4):349-366; <a href="https://doi.org/10.1080/14787210.2020.1730813">doi:10.1080/14787210.2020.1730813</a>
</p>
</li>
<li><p> Lancefield RC (1933). <strong>A serological differentiation of human and other groups of hemolytic streptococci.</strong> <em>J Exp Med.</em> 57(4): 571-95; <a href="https://doi.org/10.1084/jem.57.4.571">doi:10.1084/jem.57.4.571</a>
</p>
</li>
<li><p> Berends MS <em>et al.</em> (2022). <strong>Trends in Occurrence and Phenotypic Resistance of Coagulase-Negative Staphylococci (CoNS) Found in Human Blood in the Northern Netherlands between 2013 and 2019/</strong> <em>Micro.rganisms</em> 10(9), 1801; <a href="https://doi.org/10.3390/microorganisms10091801">doi:10.3390/microorganisms10091801</a>
</p>
</li>
<li><p> Parte, AC <em>et al.</em> (2020). <strong>List of Prokaryotic names with Standing in Nomenclature (LPSN) moves to the DSMZ.</strong> International Journal of Systematic and Evolutionary Microbiology, 70, 5607-5612; <a href="https://doi.org/10.1099/ijsem.0.004332">doi:10.1099/ijsem.0.004332</a>. Accessed from <a href="https://lpsn.dsmz.de">https://lpsn.dsmz.de</a> on December 11th, 2022.
</p>
</li>
<li><p> GBIF Secretariat (2022). GBIF Backbone Taxonomy. Checklist dataset <a href="https://doi.org/10.15468/39omei">doi:10.15468/39omei</a>. Accessed from <a href="https://www.gbif.org">https://www.gbif.org</a> on December 11th, 2022.
</p>
</li>
<li><p> Reimer, LC <em>et al.</em> (2022). <strong><em>BacDive</em> in 2022: the knowledge base for standardized bacterial and archaeal data.</strong> Nucleic Acids Res., 50(D1):D741-D74; <a href="https://doi.org/10.1093/nar/gkab961">doi:10.1093/nar/gkab961</a>. Accessed from <a href="https://bacdive.dsmz.de">https://bacdive.dsmz.de</a> on May 12th, 2023.
</p>
</li>
<li><p> Public Health Information Network Vocabulary Access and Distribution System (PHIN VADS). US Edition of SNOMED CT from 1 September 2020. Value Set Name 'Microorganism', OID 2.16.840.1.114222.4.11.1009 (v12). URL: <a href="https://phinvads.cdc.gov">https://phinvads.cdc.gov</a>
</p>
</li>
<li><p> Bartlett A <em>et al.</em> (2022). <strong>A comprehensive list of bacterial pathogens infecting humans</strong> <em>Microbiology</em> 168:001269; <a href="https://doi.org/10.1099/mic.0.001269">doi:10.1099/mic.0.001269</a>
</p>
</li></ol>



<h3>Reference Data Publicly Available</h3>

<p>All data sets in this <code>AMR</code> package (about microorganisms, antibiotics, SIR interpretation, EUCAST rules, etc.) are publicly and freely available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. We also provide tab-separated plain text files that are machine-readable and suitable for input in any software program, such as laboratory information systems. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>.
</p>


<h3>See Also</h3>

<p>Data set <a href="#topic+microorganisms">microorganisms</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># taxonomic tree -----------------------------------------------------------

mo_kingdom("Klebsiella pneumoniae")
mo_phylum("Klebsiella pneumoniae")
mo_class("Klebsiella pneumoniae")
mo_order("Klebsiella pneumoniae")
mo_family("Klebsiella pneumoniae")
mo_genus("Klebsiella pneumoniae")
mo_species("Klebsiella pneumoniae")
mo_subspecies("Klebsiella pneumoniae")


# full names and short names -----------------------------------------------

mo_name("Klebsiella pneumoniae")
mo_fullname("Klebsiella pneumoniae")
mo_shortname("Klebsiella pneumoniae")


# other properties ---------------------------------------------------------

mo_pathogenicity("Klebsiella pneumoniae")
mo_gramstain("Klebsiella pneumoniae")
mo_snomed("Klebsiella pneumoniae")
mo_type("Klebsiella pneumoniae")
mo_rank("Klebsiella pneumoniae")
mo_url("Klebsiella pneumoniae")
mo_is_yeast(c("Candida", "Trichophyton", "Klebsiella"))


# scientific reference -----------------------------------------------------

mo_ref("Klebsiella aerogenes")
mo_authors("Klebsiella aerogenes")
mo_year("Klebsiella aerogenes")
mo_lpsn("Klebsiella aerogenes")
mo_gbif("Klebsiella aerogenes")
mo_synonyms("Klebsiella aerogenes")


# abbreviations known in the field -----------------------------------------

mo_genus("MRSA")
mo_species("MRSA")
mo_shortname("VISA")
mo_gramstain("VISA")

mo_genus("EHEC")
mo_species("EIEC")
mo_name("UPEC")


# known subspecies ---------------------------------------------------------

mo_fullname("K. pneu rh")
mo_shortname("K. pneu rh")


# Becker classification, see ?as.mo ----------------------------------------

mo_fullname("Staph epidermidis")
mo_fullname("Staph epidermidis", Becker = TRUE)
mo_shortname("Staph epidermidis")
mo_shortname("Staph epidermidis", Becker = TRUE)


# Lancefield classification, see ?as.mo ------------------------------------

mo_fullname("Strep agalactiae")
mo_fullname("Strep agalactiae", Lancefield = TRUE)
mo_shortname("Strep agalactiae")
mo_shortname("Strep agalactiae", Lancefield = TRUE)


# language support  --------------------------------------------------------

mo_gramstain("Klebsiella pneumoniae", language = "de") # German
mo_gramstain("Klebsiella pneumoniae", language = "nl") # Dutch
mo_gramstain("Klebsiella pneumoniae", language = "es") # Spanish
mo_gramstain("Klebsiella pneumoniae", language = "el") # Greek
mo_gramstain("Klebsiella pneumoniae", language = "uk") # Ukrainian

# mo_type is equal to mo_kingdom, but mo_kingdom will remain untranslated
mo_kingdom("Klebsiella pneumoniae")
mo_type("Klebsiella pneumoniae")
mo_kingdom("Klebsiella pneumoniae", language = "zh") # Chinese, no effect
mo_type("Klebsiella pneumoniae", language = "zh") # Chinese, translated

mo_fullname("S. pyogenes", Lancefield = TRUE, language = "de")
mo_fullname("S. pyogenes", Lancefield = TRUE, language = "uk")


# other --------------------------------------------------------------------

# gram stains and intrinsic resistance can be used as a filter in dplyr verbs
if (require("dplyr")) {
  example_isolates %&gt;%
    filter(mo_is_gram_positive()) %&gt;%
    count(mo_genus(), sort = TRUE)
}
if (require("dplyr")) {
  example_isolates %&gt;%
    filter(mo_is_intrinsic_resistant(ab = "vanco")) %&gt;%
    count(mo_genus(), sort = TRUE)
}

# get a list with the complete taxonomy (from kingdom to subspecies)
mo_taxonomy("Klebsiella pneumoniae")

# get a list with the taxonomy, the authors, Gram-stain,
# SNOMED codes, and URL to the online database
mo_info("Klebsiella pneumoniae")

</code></pre>

<hr>
<h2 id='mo_source'>User-Defined Reference Data Set for Microorganisms</h2><span id='topic+mo_source'></span><span id='topic+set_mo_source'></span><span id='topic+get_mo_source'></span>

<h3>Description</h3>

<p>These functions can be used to predefine your own reference to be used in <code><a href="#topic+as.mo">as.mo()</a></code> and consequently all <code><a href="#topic+mo_property">mo_*</a></code> functions (such as <code><a href="#topic+mo_genus">mo_genus()</a></code> and <code><a href="#topic+mo_gramstain">mo_gramstain()</a></code>).
</p>
<p>This is <strong>the fastest way</strong> to have your organisation (or analysis) specific codes picked up and translated by this package, since you don't have to bother about it again after setting it up once.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_mo_source(
  path,
  destination = getOption("AMR_mo_source", "~/mo_source.rds")
)

get_mo_source(destination = getOption("AMR_mo_source", "~/mo_source.rds"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mo_source_+3A_path">path</code></td>
<td>
<p>location of your reference file, this can be any text file (comma-, tab- or pipe-separated) or an Excel file (see <em>Details</em>). Can also be <code>""</code>, <code>NULL</code> or <code>FALSE</code> to delete the reference file.</p>
</td></tr>
<tr><td><code id="mo_source_+3A_destination">destination</code></td>
<td>
<p>destination of the compressed data file - the default is the user's home directory.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The reference file can be a text file separated with commas (CSV) or tabs or pipes, an Excel file (either 'xls' or 'xlsx' format) or an <span class="rlang"><b>R</b></span> object file (extension '.rds'). To use an Excel file, you will need to have the <code>readxl</code> package installed.
</p>
<p><code><a href="#topic+set_mo_source">set_mo_source()</a></code> will check the file for validity: it must be a <a href="base.html#topic+data.frame">data.frame</a>, must have a column named <code>"mo"</code> which contains values from <code><a href="#topic+microorganisms">microorganisms$mo</a></code> or <code><a href="#topic+microorganisms">microorganisms$fullname</a></code> and must have a reference column with your own defined values. If all tests pass, <code><a href="#topic+set_mo_source">set_mo_source()</a></code> will read the file into <span class="rlang"><b>R</b></span> and will ask to export it to <code>"~/mo_source.rds"</code>. The CRAN policy disallows packages to write to the file system, although '<em>exceptions may be allowed in interactive sessions if the package obtains confirmation from the user</em>'. For this reason, this function only works in interactive sessions so that the user can <strong>specifically confirm and allow</strong> that this file will be created. The destination of this file can be set with the <code>destination</code> argument and defaults to the user's home directory. It can also be set with the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_mo_source</a></code>, e.g. <code>options(AMR_mo_source = "my/location/file.rds")</code>.
</p>
<p>The created compressed data file <code>"mo_source.rds"</code> will be used at default for MO determination (function <code><a href="#topic+as.mo">as.mo()</a></code> and consequently all <code style="white-space: pre;">&#8288;mo_*&#8288;</code> functions like <code><a href="#topic+mo_genus">mo_genus()</a></code> and <code><a href="#topic+mo_gramstain">mo_gramstain()</a></code>). The location and timestamp of the original file will be saved as an <a href="base.html#topic+attributes">attribute</a> to the compressed data file.
</p>
<p>The function <code><a href="#topic+get_mo_source">get_mo_source()</a></code> will return the data set by reading <code>"mo_source.rds"</code> with <code><a href="base.html#topic+readRDS">readRDS()</a></code>. If the original file has changed (by checking the location and timestamp of the original file), it will call <code><a href="#topic+set_mo_source">set_mo_source()</a></code> to update the data file automatically if used in an interactive session.
</p>
<p>Reading an Excel file (<code>.xlsx</code>) with only one row has a size of 8-9 kB. The compressed file created with <code><a href="#topic+set_mo_source">set_mo_source()</a></code> will then have a size of 0.1 kB and can be read by <code><a href="#topic+get_mo_source">get_mo_source()</a></code> in only a couple of microseconds (millionths of a second).
</p>


<h3>How to Setup</h3>

<p>Imagine this data on a sheet of an Excel file. The first column contains the organisation specific codes, the second column contains valid taxonomic names:
</p>
<div class="sourceCode"><pre>  |         A          |            B          |
--|--------------------|-----------------------|
1 | Organisation XYZ   | mo                    |
2 | lab_mo_ecoli       | Escherichia coli      |
3 | lab_mo_kpneumoniae | Klebsiella pneumoniae |
4 |                    |                       |
</pre></div>
<p>We save it as <code>"home/me/ourcodes.xlsx"</code>. Now we have to set it as a source:
</p>
<div class="sourceCode"><pre>set_mo_source("home/me/ourcodes.xlsx")
#&gt; NOTE: Created mo_source file '/Users/me/mo_source.rds' (0.3 kB) from
#&gt;       '/Users/me/Documents/ourcodes.xlsx' (9 kB), columns
#&gt;       "Organisation XYZ" and "mo"
</pre></div>
<p>It has now created a file <code>"~/mo_source.rds"</code> with the contents of our Excel file. Only the first column with foreign values and the 'mo' column will be kept when creating the RDS file.
</p>
<p>And now we can use it in our functions:
</p>
<div class="sourceCode"><pre>as.mo("lab_mo_ecoli")
#&gt; Class 'mo'
#&gt; [1] B_ESCHR_COLI

mo_genus("lab_mo_kpneumoniae")
#&gt; [1] "Klebsiella"

# other input values still work too
as.mo(c("Escherichia coli", "E. coli", "lab_mo_ecoli"))
#&gt; NOTE: Translation to one microorganism was guessed with uncertainty.
#&gt;       Use mo_uncertainties() to review it.
#&gt; Class 'mo'
#&gt; [1] B_ESCHR_COLI B_ESCHR_COLI B_ESCHR_COLI
</pre></div>
<p>If we edit the Excel file by, let's say, adding row 4 like this:
</p>
<div class="sourceCode"><pre>  |         A          |            B          |
--|--------------------|-----------------------|
1 | Organisation XYZ   | mo                    |
2 | lab_mo_ecoli       | Escherichia coli      |
3 | lab_mo_kpneumoniae | Klebsiella pneumoniae |
4 | lab_Staph_aureus   | Staphylococcus aureus |
5 |                    |                       |
</pre></div>
<p>...any new usage of an MO function in this package will update your data file:
</p>
<div class="sourceCode"><pre>as.mo("lab_mo_ecoli")
#&gt; NOTE: Updated mo_source file '/Users/me/mo_source.rds' (0.3 kB) from
#&gt;       '/Users/me/Documents/ourcodes.xlsx' (9 kB), columns
#&gt;        "Organisation XYZ" and "mo"
#&gt; Class 'mo'
#&gt; [1] B_ESCHR_COLI

mo_genus("lab_Staph_aureus")
#&gt; [1] "Staphylococcus"
</pre></div>
<p>To delete the reference data file, just use <code>""</code>, <code>NULL</code> or <code>FALSE</code> as input for <code><a href="#topic+set_mo_source">set_mo_source()</a></code>:
</p>
<div class="sourceCode"><pre>set_mo_source(NULL)
#&gt; Removed mo_source file '/Users/me/mo_source.rds'
</pre></div>
<p>If the original file (in the previous case an Excel file) is moved or deleted, the <code>mo_source.rds</code> file will be removed upon the next use of <code><a href="#topic+as.mo">as.mo()</a></code> or any <code><a href="#topic+mo_property">mo_*</a></code> function.
</p>

<hr>
<h2 id='pca'>Principal Component Analysis (for AMR)</h2><span id='topic+pca'></span>

<h3>Description</h3>

<p>Performs a principal component analysis (PCA) based on a data set with automatic determination for afterwards plotting the groups and labels, and automatic filtering on only suitable (i.e. non-empty and numeric) variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pca(
  x,
  ...,
  retx = TRUE,
  center = TRUE,
  scale. = TRUE,
  tol = NULL,
  rank. = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pca_+3A_x">x</code></td>
<td>
<p>a <a href="base.html#topic+data.frame">data.frame</a> containing <a href="base.html#topic+numeric">numeric</a> columns</p>
</td></tr>
<tr><td><code id="pca_+3A_...">...</code></td>
<td>
<p>columns of <code>x</code> to be selected for PCA, can be unquoted since it supports quasiquotation.</p>
</td></tr>
<tr><td><code id="pca_+3A_retx">retx</code></td>
<td>
<p>a logical value indicating whether the rotated variables
should be returned.</p>
</td></tr>
<tr><td><code id="pca_+3A_center">center</code></td>
<td>
<p>a logical value indicating whether the variables
should be shifted to be zero centered. Alternately, a vector of
length equal the number of columns of <code>x</code> can be supplied.
The value is passed to <code>scale</code>.</p>
</td></tr>
<tr><td><code id="pca_+3A_scale.">scale.</code></td>
<td>
<p>a logical value indicating whether the variables should
be scaled to have unit variance before the analysis takes
place.  The default is <code>FALSE</code> for consistency with S, but
in general scaling is advisable.  Alternatively, a vector of length
equal the number of columns of <code>x</code> can be supplied.  The
value is passed to <code><a href="base.html#topic+scale">scale</a></code>.</p>
</td></tr>
<tr><td><code id="pca_+3A_tol">tol</code></td>
<td>
<p>a value indicating the magnitude below which components
should be omitted. (Components are omitted if their
standard deviations are less than or equal to <code>tol</code> times the
standard deviation of the first component.)  With the default null
setting, no components are omitted (unless <code>rank.</code> is specified
less than <code>min(dim(x))</code>.).  Other settings for tol could be
<code>tol = 0</code> or <code>tol = sqrt(.Machine$double.eps)</code>, which
would omit essentially constant components.</p>
</td></tr>
<tr><td><code id="pca_+3A_rank.">rank.</code></td>
<td>
<p>optionally, a number specifying the maximal rank, i.e.,
maximal number of principal components to be used.  Can be set as
alternative or in addition to <code>tol</code>, useful notably when the
desired rank is considerably smaller than the dimensions of the matrix.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code><a href="#topic+pca">pca()</a></code> function takes a <a href="base.html#topic+data.frame">data.frame</a> as input and performs the actual PCA with the <span class="rlang"><b>R</b></span> function <code><a href="stats.html#topic+prcomp">prcomp()</a></code>.
</p>
<p>The result of the <code><a href="#topic+pca">pca()</a></code> function is a <a href="stats.html#topic+prcomp">prcomp</a> object, with an additional attribute <code>non_numeric_cols</code> which is a vector with the column names of all columns that do not contain <a href="base.html#topic+numeric">numeric</a> values. These are probably the groups and labels, and will be used by <code><a href="#topic+ggplot_pca">ggplot_pca()</a></code>.
</p>


<h3>Value</h3>

<p>An object of classes <a href="#topic+pca">pca</a> and <a href="stats.html#topic+prcomp">prcomp</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># `example_isolates` is a data set available in the AMR package.
# See ?example_isolates.


if (require("dplyr")) {
  # calculate the resistance per group first
  resistance_data &lt;- example_isolates %&gt;%
    group_by(
      order = mo_order(mo), # group on anything, like order
      genus = mo_genus(mo)
    ) %&gt;% #   and genus as we do here;
    filter(n() &gt;= 30) %&gt;% # filter on only 30 results per group
    summarise_if(is.sir, resistance) # then get resistance of all drugs

  # now conduct PCA for certain antimicrobial drugs
  pca_result &lt;- resistance_data %&gt;%
    pca(AMC, CXM, CTX, CAZ, GEN, TOB, TMP, SXT)

  pca_result
  summary(pca_result)

  # old base R plotting method:
  biplot(pca_result)
  # new ggplot2 plotting method using this package:
  if (require("ggplot2")) {
    ggplot_pca(pca_result)

    ggplot_pca(pca_result) +
      scale_colour_viridis_d() +
      labs(title = "Title here")
  }
}

</code></pre>

<hr>
<h2 id='plot'>Plotting for Classes <code>sir</code>, <code>mic</code> and <code>disk</code></h2><span id='topic+plot'></span><span id='topic+plot.mic'></span><span id='topic+autoplot.mic'></span><span id='topic+fortify.mic'></span><span id='topic+plot.disk'></span><span id='topic+autoplot.disk'></span><span id='topic+fortify.disk'></span><span id='topic+plot.sir'></span><span id='topic+autoplot.sir'></span><span id='topic+fortify.sir'></span>

<h3>Description</h3>

<p>Functions to plot classes <code>sir</code>, <code>mic</code> and <code>disk</code>, with support for base <span class="rlang"><b>R</b></span> and <code>ggplot2</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'mic'
plot(
  x,
  mo = NULL,
  ab = NULL,
  guideline = "EUCAST",
  main = deparse(substitute(x)),
  ylab = translate_AMR("Frequency", language = language),
  xlab = translate_AMR("Minimum Inhibitory Concentration (mg/L)", language = language),
  colours_SIR = c("#3CAEA3", "#F6D55C", "#ED553B"),
  language = get_AMR_locale(),
  expand = TRUE,
  include_PKPD = getOption("AMR_include_PKPD", TRUE),
  breakpoint_type = getOption("AMR_breakpoint_type", "human"),
  ...
)

## S3 method for class 'mic'
autoplot(
  object,
  mo = NULL,
  ab = NULL,
  guideline = "EUCAST",
  title = deparse(substitute(object)),
  ylab = translate_AMR("Frequency", language = language),
  xlab = translate_AMR("Minimum Inhibitory Concentration (mg/L)", language = language),
  colours_SIR = c("#3CAEA3", "#F6D55C", "#ED553B"),
  language = get_AMR_locale(),
  expand = TRUE,
  include_PKPD = getOption("AMR_include_PKPD", TRUE),
  breakpoint_type = getOption("AMR_breakpoint_type", "human"),
  ...
)

## S3 method for class 'mic'
fortify(object, ...)

## S3 method for class 'disk'
plot(
  x,
  main = deparse(substitute(x)),
  ylab = translate_AMR("Frequency", language = language),
  xlab = translate_AMR("Disk diffusion diameter (mm)", language = language),
  mo = NULL,
  ab = NULL,
  guideline = "EUCAST",
  colours_SIR = c("#3CAEA3", "#F6D55C", "#ED553B"),
  language = get_AMR_locale(),
  expand = TRUE,
  include_PKPD = getOption("AMR_include_PKPD", TRUE),
  breakpoint_type = getOption("AMR_breakpoint_type", "human"),
  ...
)

## S3 method for class 'disk'
autoplot(
  object,
  mo = NULL,
  ab = NULL,
  title = deparse(substitute(object)),
  ylab = translate_AMR("Frequency", language = language),
  xlab = translate_AMR("Disk diffusion diameter (mm)", language = language),
  guideline = "EUCAST",
  colours_SIR = c("#3CAEA3", "#F6D55C", "#ED553B"),
  language = get_AMR_locale(),
  expand = TRUE,
  include_PKPD = getOption("AMR_include_PKPD", TRUE),
  breakpoint_type = getOption("AMR_breakpoint_type", "human"),
  ...
)

## S3 method for class 'disk'
fortify(object, ...)

## S3 method for class 'sir'
plot(
  x,
  ylab = translate_AMR("Percentage", language = language),
  xlab = translate_AMR("Antimicrobial Interpretation", language = language),
  main = deparse(substitute(x)),
  language = get_AMR_locale(),
  ...
)

## S3 method for class 'sir'
autoplot(
  object,
  title = deparse(substitute(object)),
  xlab = translate_AMR("Antimicrobial Interpretation", language = language),
  ylab = translate_AMR("Frequency", language = language),
  colours_SIR = c("#3CAEA3", "#F6D55C", "#ED553B"),
  language = get_AMR_locale(),
  ...
)

## S3 method for class 'sir'
fortify(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_+3A_x">x</code>, <code id="plot_+3A_object">object</code></td>
<td>
<p>values created with <code><a href="#topic+as.mic">as.mic()</a></code>, <code><a href="#topic+as.disk">as.disk()</a></code> or <code><a href="#topic+as.sir">as.sir()</a></code> (or their <code style="white-space: pre;">&#8288;random_*&#8288;</code> variants, such as <code><a href="#topic+random_mic">random_mic()</a></code>)</p>
</td></tr>
<tr><td><code id="plot_+3A_mo">mo</code></td>
<td>
<p>any (vector of) text that can be coerced to a valid microorganism code with <code><a href="#topic+as.mo">as.mo()</a></code></p>
</td></tr>
<tr><td><code id="plot_+3A_ab">ab</code></td>
<td>
<p>any (vector of) text that can be coerced to a valid antimicrobial drug code with <code><a href="#topic+as.ab">as.ab()</a></code></p>
</td></tr>
<tr><td><code id="plot_+3A_guideline">guideline</code></td>
<td>
<p>interpretation guideline to use - the default is the latest included EUCAST guideline, see <em>Details</em></p>
</td></tr>
<tr><td><code id="plot_+3A_main">main</code>, <code id="plot_+3A_title">title</code></td>
<td>
<p>title of the plot</p>
</td></tr>
<tr><td><code id="plot_+3A_xlab">xlab</code>, <code id="plot_+3A_ylab">ylab</code></td>
<td>
<p>axis title</p>
</td></tr>
<tr><td><code id="plot_+3A_colours_sir">colours_SIR</code></td>
<td>
<p>colours to use for filling in the bars, must be a vector of three values (in the order S, I and R). The default colours are colour-blind friendly.</p>
</td></tr>
<tr><td><code id="plot_+3A_language">language</code></td>
<td>
<p>language to be used to translate 'Susceptible', 'Increased exposure'/'Intermediate' and 'Resistant' - the default is system language (see <code><a href="#topic+get_AMR_locale">get_AMR_locale()</a></code>) and can be overwritten by setting the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_locale</a></code>, e.g. <code>options(AMR_locale = "de")</code>, see <a href="#topic+translate">translate</a>. Use <code>language = NULL</code> or <code>language = ""</code> to prevent translation.</p>
</td></tr>
<tr><td><code id="plot_+3A_expand">expand</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether the range on the x axis should be expanded between the lowest and highest value. For MIC values, intermediate values will be factors of 2 starting from the highest MIC value. For disk diameters, the whole diameter range will be filled.</p>
</td></tr>
<tr><td><code id="plot_+3A_include_pkpd">include_PKPD</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate that PK/PD clinical breakpoints must be applied as a last resort - the default is <code>TRUE</code>. Can also be set with the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_include_PKPD</a></code>.</p>
</td></tr>
<tr><td><code id="plot_+3A_breakpoint_type">breakpoint_type</code></td>
<td>
<p>the type of breakpoints to use, either &quot;ECOFF&quot;, &quot;animal&quot;, or &quot;human&quot;. ECOFF stands for Epidemiological Cut-Off values. The default is <code>"human"</code>, which can also be set with the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_breakpoint_type</a></code>.</p>
</td></tr>
<tr><td><code id="plot_+3A_...">...</code></td>
<td>
<p>arguments passed on to methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The interpretation of &quot;I&quot; will be named &quot;Increased exposure&quot; for all EUCAST guidelines since 2019, and will be named &quot;Intermediate&quot; in all other cases.
</p>
<p>For interpreting MIC values as well as disk diffusion diameters, supported guidelines to be used as input for the <code>guideline</code> argument are: &quot;EUCAST 2023&quot;, &quot;EUCAST 2022&quot;, &quot;EUCAST 2021&quot;, &quot;EUCAST 2020&quot;, &quot;EUCAST 2019&quot;, &quot;EUCAST 2018&quot;, &quot;EUCAST 2017&quot;, &quot;EUCAST 2016&quot;, &quot;EUCAST 2015&quot;, &quot;EUCAST 2014&quot;, &quot;EUCAST 2013&quot;, &quot;EUCAST 2012&quot;, &quot;EUCAST 2011&quot;, &quot;CLSI 2023&quot;, &quot;CLSI 2022&quot;, &quot;CLSI 2021&quot;, &quot;CLSI 2020&quot;, &quot;CLSI 2019&quot;, &quot;CLSI 2018&quot;, &quot;CLSI 2017&quot;, &quot;CLSI 2016&quot;, &quot;CLSI 2015&quot;, &quot;CLSI 2014&quot;, &quot;CLSI 2013&quot;, &quot;CLSI 2012&quot;, and &quot;CLSI 2011&quot;.
</p>
<p>Simply using <code>"CLSI"</code> or <code>"EUCAST"</code> as input will automatically select the latest version of that guideline.
</p>


<h3>Value</h3>

<p>The <code>autoplot()</code> functions return a <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code> model that is extendible with any <code>ggplot2</code> function.
</p>
<p>The <code>fortify()</code> functions return a <a href="base.html#topic+data.frame">data.frame</a> as an extension for usage in the <code><a href="ggplot2.html#topic+ggplot">ggplot2::ggplot()</a></code> function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>some_mic_values &lt;- random_mic(size = 100)
some_disk_values &lt;- random_disk(size = 100, mo = "Escherichia coli", ab = "cipro")
some_sir_values &lt;- random_sir(50, prob_SIR = c(0.55, 0.05, 0.30))

plot(some_mic_values)
plot(some_disk_values)
plot(some_sir_values)

# when providing the microorganism and antibiotic, colours will show interpretations:
plot(some_mic_values, mo = "S. aureus", ab = "ampicillin")
plot(some_disk_values, mo = "Escherichia coli", ab = "cipro")
plot(some_disk_values, mo = "Escherichia coli", ab = "cipro", language = "nl")


if (require("ggplot2")) {
  autoplot(some_mic_values)
}
if (require("ggplot2")) {
  autoplot(some_disk_values, mo = "Escherichia coli", ab = "cipro")
}
if (require("ggplot2")) {
  autoplot(some_sir_values)
}

</code></pre>

<hr>
<h2 id='proportion'>Calculate Antimicrobial Resistance</h2><span id='topic+proportion'></span><span id='topic+resistance'></span><span id='topic+portion'></span><span id='topic+susceptibility'></span><span id='topic+sir_confidence_interval'></span><span id='topic+proportion_R'></span><span id='topic+proportion_IR'></span><span id='topic+proportion_I'></span><span id='topic+proportion_SI'></span><span id='topic+proportion_S'></span><span id='topic+proportion_df'></span><span id='topic+sir_df'></span>

<h3>Description</h3>

<p>These functions can be used to calculate the (co-)resistance or susceptibility of microbial isolates (i.e. percentage of S, SI, I, IR or R). All functions support quasiquotation with pipes, can be used in <code>summarise()</code> from the <code>dplyr</code> package and also support grouped variables, see <em>Examples</em>.
</p>
<p><code><a href="#topic+resistance">resistance()</a></code> should be used to calculate resistance, <code><a href="#topic+susceptibility">susceptibility()</a></code> should be used to calculate susceptibility.<br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>resistance(..., minimum = 30, as_percent = FALSE, only_all_tested = FALSE)

susceptibility(..., minimum = 30, as_percent = FALSE, only_all_tested = FALSE)

sir_confidence_interval(
  ...,
  ab_result = "R",
  minimum = 30,
  as_percent = FALSE,
  only_all_tested = FALSE,
  confidence_level = 0.95,
  side = "both",
  collapse = FALSE
)

proportion_R(..., minimum = 30, as_percent = FALSE, only_all_tested = FALSE)

proportion_IR(..., minimum = 30, as_percent = FALSE, only_all_tested = FALSE)

proportion_I(..., minimum = 30, as_percent = FALSE, only_all_tested = FALSE)

proportion_SI(..., minimum = 30, as_percent = FALSE, only_all_tested = FALSE)

proportion_S(..., minimum = 30, as_percent = FALSE, only_all_tested = FALSE)

proportion_df(
  data,
  translate_ab = "name",
  language = get_AMR_locale(),
  minimum = 30,
  as_percent = FALSE,
  combine_SI = TRUE,
  confidence_level = 0.95
)

sir_df(
  data,
  translate_ab = "name",
  language = get_AMR_locale(),
  minimum = 30,
  as_percent = FALSE,
  combine_SI = TRUE,
  confidence_level = 0.95
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="proportion_+3A_...">...</code></td>
<td>
<p>one or more vectors (or columns) with antibiotic interpretations. They will be transformed internally with <code><a href="#topic+as.sir">as.sir()</a></code> if needed. Use multiple columns to calculate (the lack of) co-resistance: the probability where one of two drugs have a resistant or susceptible result. See <em>Examples</em>.</p>
</td></tr>
<tr><td><code id="proportion_+3A_minimum">minimum</code></td>
<td>
<p>the minimum allowed number of available (tested) isolates. Any isolate count lower than <code>minimum</code> will return <code>NA</code> with a warning. The default number of <code>30</code> isolates is advised by the Clinical and Laboratory Standards Institute (CLSI) as best practice, see <em>Source</em>.</p>
</td></tr>
<tr><td><code id="proportion_+3A_as_percent">as_percent</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether the output must be returned as a hundred fold with % sign (a character). A value of <code>0.123456</code> will then be returned as <code>"12.3%"</code>.</p>
</td></tr>
<tr><td><code id="proportion_+3A_only_all_tested">only_all_tested</code></td>
<td>
<p>(for combination therapies, i.e. using more than one variable for <code>...</code>): a <a href="base.html#topic+logical">logical</a> to indicate that isolates must be tested for all antibiotics, see section <em>Combination Therapy</em> below</p>
</td></tr>
<tr><td><code id="proportion_+3A_ab_result">ab_result</code></td>
<td>
<p>antibiotic results to test against, must be one or more values of &quot;S&quot;, &quot;I&quot;, or &quot;R&quot;</p>
</td></tr>
<tr><td><code id="proportion_+3A_confidence_level">confidence_level</code></td>
<td>
<p>the confidence level for the returned confidence interval. For the calculation, the number of S or SI isolates, and R isolates are compared with the total number of available isolates with R, S, or I by using <code><a href="stats.html#topic+binom.test">binom.test()</a></code>, i.e., the Clopper-Pearson method.</p>
</td></tr>
<tr><td><code id="proportion_+3A_side">side</code></td>
<td>
<p>the side of the confidence interval to return. The default is <code>"both"</code> for a length 2 vector, but can also be (abbreviated as) <code>"min"</code>/<code>"left"</code>/<code>"lower"</code>/<code>"less"</code> or <code>"max"</code>/<code>"right"</code>/<code>"higher"</code>/<code>"greater"</code>.</p>
</td></tr>
<tr><td><code id="proportion_+3A_collapse">collapse</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether the output values should be 'collapsed', i.e. be merged together into one value, or a character value to use for collapsing</p>
</td></tr>
<tr><td><code id="proportion_+3A_data">data</code></td>
<td>
<p>a <a href="base.html#topic+data.frame">data.frame</a> containing columns with class <code><a href="#topic+sir">sir</a></code> (see <code><a href="#topic+as.sir">as.sir()</a></code>)</p>
</td></tr>
<tr><td><code id="proportion_+3A_translate_ab">translate_ab</code></td>
<td>
<p>a column name of the <a href="#topic+antibiotics">antibiotics</a> data set to translate the antibiotic abbreviations to, using <code><a href="#topic+ab_property">ab_property()</a></code></p>
</td></tr>
<tr><td><code id="proportion_+3A_language">language</code></td>
<td>
<p>language of the returned text - the default is the current system language (see <code><a href="#topic+get_AMR_locale">get_AMR_locale()</a></code>) and can also be set with the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_locale</a></code>. Use <code>language = NULL</code> or <code>language = ""</code> to prevent translation.</p>
</td></tr>
<tr><td><code id="proportion_+3A_combine_si">combine_SI</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether all values of S and I must be merged into one, so the output only consists of S+I vs. R (susceptible vs. resistant) - the default is <code>TRUE</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Remember that you should filter your data to let it contain only first isolates!</strong> This is needed to exclude duplicates and to reduce selection bias. Use <code><a href="#topic+first_isolate">first_isolate()</a></code> to determine them in your data set with one of the four available algorithms.
</p>
<p>The function <code><a href="#topic+resistance">resistance()</a></code> is equal to the function <code><a href="#topic+proportion_R">proportion_R()</a></code>. The function <code><a href="#topic+susceptibility">susceptibility()</a></code> is equal to the function <code><a href="#topic+proportion_SI">proportion_SI()</a></code>.
</p>
<p>Use <code><a href="#topic+sir_confidence_interval">sir_confidence_interval()</a></code> to calculate the confidence interval, which relies on <code><a href="stats.html#topic+binom.test">binom.test()</a></code>, i.e., the Clopper-Pearson method. This function returns a vector of length 2 at default for antimicrobial <em>resistance</em>. Change the <code>side</code> argument to &quot;left&quot;/&quot;min&quot; or &quot;right&quot;/&quot;max&quot; to return a single value, and change the <code>ab_result</code> argument to e.g. <code>c("S", "I")</code> to test for antimicrobial <em>susceptibility</em>, see Examples.
</p>
<p>These functions are not meant to count isolates, but to calculate the proportion of resistance/susceptibility. Use the <code><a href="#topic+count">count_*()</a></code> functions to count isolates. The function <code><a href="#topic+susceptibility">susceptibility()</a></code> is essentially equal to <code><a href="#topic+count_susceptible">count_susceptible()</a></code><code>/</code><code><a href="#topic+count_all">count_all()</a></code>. <em>Low counts can influence the outcome - the <code style="white-space: pre;">&#8288;proportion_*()&#8288;</code> functions may camouflage this, since they only return the proportion (albeit dependent on the <code>minimum</code> argument).</em>
</p>
<p>The function <code><a href="#topic+proportion_df">proportion_df()</a></code> takes any variable from <code>data</code> that has an <code><a href="#topic+sir">sir</a></code> class (created with <code><a href="#topic+as.sir">as.sir()</a></code>) and calculates the proportions S, I, and R. It also supports grouped variables. The function <code><a href="#topic+sir_df">sir_df()</a></code> works exactly like <code><a href="#topic+proportion_df">proportion_df()</a></code>, but adds the number of isolates.
</p>


<h3>Value</h3>

<p>A <a href="base.html#topic+double">double</a> or, when <code>as_percent = TRUE</code>, a <a href="base.html#topic+character">character</a>.
</p>


<h3>Combination Therapy</h3>

<p>When using more than one variable for <code>...</code> (= combination therapy), use <code>only_all_tested</code> to only count isolates that are tested for all antibiotics/variables that you test them for. See this example for two antibiotics, Drug A and Drug B, about how <code><a href="#topic+susceptibility">susceptibility()</a></code> works to calculate the %SI:
</p>
<div class="sourceCode"><pre>--------------------------------------------------------------------
                    only_all_tested = FALSE  only_all_tested = TRUE
                    -----------------------  -----------------------
 Drug A    Drug B   include as  include as   include as  include as
                    numerator   denominator  numerator   denominator
--------  --------  ----------  -----------  ----------  -----------
 S or I    S or I       X            X            X            X
   R       S or I       X            X            X            X
  &lt;NA&gt;     S or I       X            X            -            -
 S or I      R          X            X            X            X
   R         R          -            X            -            X
  &lt;NA&gt;       R          -            -            -            -
 S or I     &lt;NA&gt;        X            X            -            -
   R        &lt;NA&gt;        -            -            -            -
  &lt;NA&gt;      &lt;NA&gt;        -            -            -            -
--------------------------------------------------------------------
</pre></div>
<p>Please note that, in combination therapies, for <code>only_all_tested = TRUE</code> applies that:
</p>
<div class="sourceCode"><pre>    count_S()    +   count_I()    +   count_R()    = count_all()
  proportion_S() + proportion_I() + proportion_R() = 1
</pre></div>
<p>and that, in combination therapies, for <code>only_all_tested = FALSE</code> applies that:
</p>
<div class="sourceCode"><pre>    count_S()    +   count_I()    +   count_R()    &gt;= count_all()
  proportion_S() + proportion_I() + proportion_R() &gt;= 1
</pre></div>
<p>Using <code>only_all_tested</code> has no impact when only using one antibiotic as input.
</p>


<h3>Interpretation of SIR</h3>

<p>In 2019, the European Committee on Antimicrobial Susceptibility Testing (EUCAST) has decided to change the definitions of susceptibility testing categories S, I, and R as shown below (<a href="https://www.eucast.org/newsiandr">https://www.eucast.org/newsiandr</a>):
</p>

<ul>
<li> <p><strong>S - Susceptible, standard dosing regimen</strong><br />
A microorganism is categorised as &quot;Susceptible, standard dosing regimen&quot;, when there is a high likelihood of therapeutic success using a standard dosing regimen of the agent.
</p>
</li>
<li> <p><strong>I - Susceptible, increased exposure</strong> <em><br />
A microorganism is categorised as &quot;Susceptible, Increased exposure</em>&quot; when there is a high likelihood of therapeutic success because exposure to the agent is increased by adjusting the dosing regimen or by its concentration at the site of infection.
</p>
</li>
<li> <p><strong>R = Resistant</strong><br />
A microorganism is categorised as &quot;Resistant&quot; when there is a high likelihood of therapeutic failure even when there is increased exposure.
</p>

<ul>
<li> <p><em>Exposure</em> is a function of how the mode of administration, dose, dosing interval, infusion time, as well as distribution and excretion of the antimicrobial agent will influence the infecting organism at the site of infection.
</p>
</li></ul>

</li></ul>

<p>This AMR package honours this insight. Use <code><a href="#topic+susceptibility">susceptibility()</a></code> (equal to <code><a href="#topic+proportion_SI">proportion_SI()</a></code>) to determine antimicrobial susceptibility and <code><a href="#topic+count_susceptible">count_susceptible()</a></code> (equal to <code><a href="#topic+count_SI">count_SI()</a></code>) to count susceptible isolates.
</p>


<h3>Source</h3>

<p><strong>M39 Analysis and Presentation of Cumulative Antimicrobial Susceptibility Test Data, 5th Edition</strong>, 2022, <em>Clinical and Laboratory Standards Institute (CLSI)</em>. <a href="https://clsi.org/standards/products/microbiology/documents/m39/">https://clsi.org/standards/products/microbiology/documents/m39/</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+count">count()</a></code> to count resistant and susceptible isolates.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># example_isolates is a data set available in the AMR package.
# run ?example_isolates for more info.
example_isolates


# base R ------------------------------------------------------------
# determines %R
resistance(example_isolates$AMX)
sir_confidence_interval(example_isolates$AMX)
sir_confidence_interval(example_isolates$AMX,
  confidence_level = 0.975
)
sir_confidence_interval(example_isolates$AMX,
  confidence_level = 0.975,
  collapse = ", "
)

# determines %S+I:
susceptibility(example_isolates$AMX)
sir_confidence_interval(example_isolates$AMX,
  ab_result = c("S", "I")
)

# be more specific
proportion_S(example_isolates$AMX)
proportion_SI(example_isolates$AMX)
proportion_I(example_isolates$AMX)
proportion_IR(example_isolates$AMX)
proportion_R(example_isolates$AMX)

# dplyr -------------------------------------------------------------

if (require("dplyr")) {
  example_isolates %&gt;%
    group_by(ward) %&gt;%
    summarise(
      r = resistance(CIP),
      n = n_sir(CIP)
    ) # n_sir works like n_distinct in dplyr, see ?n_sir
}
if (require("dplyr")) {
  example_isolates %&gt;%
    group_by(ward) %&gt;%
    summarise(
      cipro_R = resistance(CIP),
      ci_min = sir_confidence_interval(CIP, side = "min"),
      ci_max = sir_confidence_interval(CIP, side = "max"),
    )
}
if (require("dplyr")) {
  # scoped dplyr verbs with antibiotic selectors
  # (you could also use across() of course)
  example_isolates %&gt;%
    group_by(ward) %&gt;%
    summarise_at(
      c(aminoglycosides(), carbapenems()),
      resistance
    )
}
if (require("dplyr")) {
  example_isolates %&gt;%
    group_by(ward) %&gt;%
    summarise(
      R = resistance(CIP, as_percent = TRUE),
      SI = susceptibility(CIP, as_percent = TRUE),
      n1 = count_all(CIP), # the actual total; sum of all three
      n2 = n_sir(CIP), # same - analogous to n_distinct
      total = n()
    ) # NOT the number of tested isolates!

  # Calculate co-resistance between amoxicillin/clav acid and gentamicin,
  # so we can see that combination therapy does a lot more than mono therapy:
  example_isolates %&gt;% susceptibility(AMC) # %SI = 76.3%
  example_isolates %&gt;% count_all(AMC) #   n = 1879

  example_isolates %&gt;% susceptibility(GEN) # %SI = 75.4%
  example_isolates %&gt;% count_all(GEN) #   n = 1855

  example_isolates %&gt;% susceptibility(AMC, GEN) # %SI = 94.1%
  example_isolates %&gt;% count_all(AMC, GEN) #   n = 1939


  # See Details on how `only_all_tested` works. Example:
  example_isolates %&gt;%
    summarise(
      numerator = count_susceptible(AMC, GEN),
      denominator = count_all(AMC, GEN),
      proportion = susceptibility(AMC, GEN)
    )

  example_isolates %&gt;%
    summarise(
      numerator = count_susceptible(AMC, GEN, only_all_tested = TRUE),
      denominator = count_all(AMC, GEN, only_all_tested = TRUE),
      proportion = susceptibility(AMC, GEN, only_all_tested = TRUE)
    )


  example_isolates %&gt;%
    group_by(ward) %&gt;%
    summarise(
      cipro_p = susceptibility(CIP, as_percent = TRUE),
      cipro_n = count_all(CIP),
      genta_p = susceptibility(GEN, as_percent = TRUE),
      genta_n = count_all(GEN),
      combination_p = susceptibility(CIP, GEN, as_percent = TRUE),
      combination_n = count_all(CIP, GEN)
    )

  # Get proportions S/I/R immediately of all sir columns
  example_isolates %&gt;%
    select(AMX, CIP) %&gt;%
    proportion_df(translate = FALSE)

  # It also supports grouping variables
  # (use sir_df to also include the count)
  example_isolates %&gt;%
    select(ward, AMX, CIP) %&gt;%
    group_by(ward) %&gt;%
    sir_df(translate = FALSE)
}

</code></pre>

<hr>
<h2 id='random'>Random MIC Values/Disk Zones/SIR Generation</h2><span id='topic+random'></span><span id='topic+random_mic'></span><span id='topic+random_disk'></span><span id='topic+random_sir'></span>

<h3>Description</h3>

<p>These functions can be used for generating random MIC values and disk diffusion diameters, for AMR data analysis practice. By providing a microorganism and antimicrobial drug, the generated results will reflect reality as much as possible.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>random_mic(size = NULL, mo = NULL, ab = NULL, ...)

random_disk(size = NULL, mo = NULL, ab = NULL, ...)

random_sir(size = NULL, prob_SIR = c(0.33, 0.33, 0.33), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="random_+3A_size">size</code></td>
<td>
<p>desired size of the returned vector. If used in a <a href="base.html#topic+data.frame">data.frame</a> call or <code>dplyr</code> verb, will get the current (group) size if left blank.</p>
</td></tr>
<tr><td><code id="random_+3A_mo">mo</code></td>
<td>
<p>any <a href="base.html#topic+character">character</a> that can be coerced to a valid microorganism code with <code><a href="#topic+as.mo">as.mo()</a></code></p>
</td></tr>
<tr><td><code id="random_+3A_ab">ab</code></td>
<td>
<p>any <a href="base.html#topic+character">character</a> that can be coerced to a valid antimicrobial drug code with <code><a href="#topic+as.ab">as.ab()</a></code></p>
</td></tr>
<tr><td><code id="random_+3A_...">...</code></td>
<td>
<p>ignored, only in place to allow future extensions</p>
</td></tr>
<tr><td><code id="random_+3A_prob_sir">prob_SIR</code></td>
<td>
<p>a vector of length 3: the probabilities for &quot;S&quot; (1st value), &quot;I&quot; (2nd value) and &quot;R&quot; (3rd value)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The base <span class="rlang"><b>R</b></span> function <code><a href="base.html#topic+sample">sample()</a></code> is used for generating values.
</p>
<p>Generated values are based on the EUCAST 2023 guideline as implemented in the <a href="#topic+clinical_breakpoints">clinical_breakpoints</a> data set. To create specific generated values per bug or drug, set the <code>mo</code> and/or <code>ab</code> argument.
</p>


<h3>Value</h3>

<p>class <code>mic</code> for <code><a href="#topic+random_mic">random_mic()</a></code> (see <code><a href="#topic+as.mic">as.mic()</a></code>) and class <code>disk</code> for <code><a href="#topic+random_disk">random_disk()</a></code> (see <code><a href="#topic+as.disk">as.disk()</a></code>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>random_mic(25)
random_disk(25)
random_sir(25)


# make the random generation more realistic by setting a bug and/or drug:
random_mic(25, "Klebsiella pneumoniae") # range 0.0625-64
random_mic(25, "Klebsiella pneumoniae", "meropenem") # range 0.0625-16
random_mic(25, "Streptococcus pneumoniae", "meropenem") # range 0.0625-4

random_disk(25, "Klebsiella pneumoniae") # range 8-50
random_disk(25, "Klebsiella pneumoniae", "ampicillin") # range 11-17
random_disk(25, "Streptococcus pneumoniae", "ampicillin") # range 12-27

</code></pre>

<hr>
<h2 id='resistance_predict'>Predict Antimicrobial Resistance</h2><span id='topic+resistance_predict'></span><span id='topic+sir_predict'></span><span id='topic+plot.resistance_predict'></span><span id='topic+ggplot_sir_predict'></span><span id='topic+autoplot.resistance_predict'></span>

<h3>Description</h3>

<p>Create a prediction model to predict antimicrobial resistance for the next years on statistical solid ground. Standard errors (SE) will be returned as columns <code>se_min</code> and <code>se_max</code>. See <em>Examples</em> for a real live example.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>resistance_predict(
  x,
  col_ab,
  col_date = NULL,
  year_min = NULL,
  year_max = NULL,
  year_every = 1,
  minimum = 30,
  model = NULL,
  I_as_S = TRUE,
  preserve_measurements = TRUE,
  info = interactive(),
  ...
)

sir_predict(
  x,
  col_ab,
  col_date = NULL,
  year_min = NULL,
  year_max = NULL,
  year_every = 1,
  minimum = 30,
  model = NULL,
  I_as_S = TRUE,
  preserve_measurements = TRUE,
  info = interactive(),
  ...
)

## S3 method for class 'resistance_predict'
plot(x, main = paste("Resistance Prediction of", x_name), ...)

ggplot_sir_predict(
  x,
  main = paste("Resistance Prediction of", x_name),
  ribbon = TRUE,
  ...
)

## S3 method for class 'resistance_predict'
autoplot(
  object,
  main = paste("Resistance Prediction of", x_name),
  ribbon = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="resistance_predict_+3A_x">x</code></td>
<td>
<p>a <a href="base.html#topic+data.frame">data.frame</a> containing isolates. Can be left blank for automatic determination, see <em>Examples</em>.</p>
</td></tr>
<tr><td><code id="resistance_predict_+3A_col_ab">col_ab</code></td>
<td>
<p>column name of <code>x</code> containing antimicrobial interpretations (<code>"R"</code>, <code>"I"</code> and <code>"S"</code>)</p>
</td></tr>
<tr><td><code id="resistance_predict_+3A_col_date">col_date</code></td>
<td>
<p>column name of the date, will be used to calculate years if this column doesn't consist of years already - the default is the first column of with a date class</p>
</td></tr>
<tr><td><code id="resistance_predict_+3A_year_min">year_min</code></td>
<td>
<p>lowest year to use in the prediction model, dafaults to the lowest year in <code>col_date</code></p>
</td></tr>
<tr><td><code id="resistance_predict_+3A_year_max">year_max</code></td>
<td>
<p>highest year to use in the prediction model - the default is 10 years after today</p>
</td></tr>
<tr><td><code id="resistance_predict_+3A_year_every">year_every</code></td>
<td>
<p>unit of sequence between lowest year found in the data and <code>year_max</code></p>
</td></tr>
<tr><td><code id="resistance_predict_+3A_minimum">minimum</code></td>
<td>
<p>minimal amount of available isolates per year to include. Years containing less observations will be estimated by the model.</p>
</td></tr>
<tr><td><code id="resistance_predict_+3A_model">model</code></td>
<td>
<p>the statistical model of choice. This could be a generalised linear regression model with binomial distribution (i.e. using <code>glm(..., family = binomial)</code>, assuming that a period of zero resistance was followed by a period of increasing resistance leading slowly to more and more resistance. See <em>Details</em> for all valid options.</p>
</td></tr>
<tr><td><code id="resistance_predict_+3A_i_as_s">I_as_S</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether values <code>"I"</code> should be treated as <code>"S"</code> (will otherwise be treated as <code>"R"</code>). The default, <code>TRUE</code>, follows the redefinition by EUCAST about the interpretation of I (increased exposure) in 2019, see section <em>Interpretation of S, I and R</em> below.</p>
</td></tr>
<tr><td><code id="resistance_predict_+3A_preserve_measurements">preserve_measurements</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether predictions of years that are actually available in the data should be overwritten by the original data. The standard errors of those years will be <code>NA</code>.</p>
</td></tr>
<tr><td><code id="resistance_predict_+3A_info">info</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether textual analysis should be printed with the name and <code><a href="base.html#topic+summary">summary()</a></code> of the statistical model.</p>
</td></tr>
<tr><td><code id="resistance_predict_+3A_...">...</code></td>
<td>
<p>arguments passed on to functions</p>
</td></tr>
<tr><td><code id="resistance_predict_+3A_main">main</code></td>
<td>
<p>title of the plot</p>
</td></tr>
<tr><td><code id="resistance_predict_+3A_ribbon">ribbon</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> to indicate whether a ribbon should be shown (default) or error bars</p>
</td></tr>
<tr><td><code id="resistance_predict_+3A_object">object</code></td>
<td>
<p>model data to be plotted</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Valid options for the statistical model (argument <code>model</code>) are:
</p>

<ul>
<li> <p><code>"binomial"</code> or <code>"binom"</code> or <code>"logit"</code>: a generalised linear regression model with binomial distribution
</p>
</li>
<li> <p><code>"loglin"</code> or <code>"poisson"</code>: a generalised log-linear regression model with poisson distribution
</p>
</li>
<li> <p><code>"lin"</code> or <code>"linear"</code>: a linear regression model
</p>
</li></ul>



<h3>Value</h3>

<p>A <a href="base.html#topic+data.frame">data.frame</a> with extra class <code><a href="#topic+resistance_predict">resistance_predict</a></code> with columns:
</p>

<ul>
<li> <p><code>year</code>
</p>
</li>
<li> <p><code>value</code>, the same as <code>estimated</code> when <code>preserve_measurements = FALSE</code>, and a combination of <code>observed</code> and <code>estimated</code> otherwise
</p>
</li>
<li> <p><code>se_min</code>, the lower bound of the standard error with a minimum of <code>0</code> (so the standard error will never go below 0%)
</p>
</li>
<li> <p><code>se_max</code> the upper bound of the standard error with a maximum of <code>1</code> (so the standard error will never go above 100%)
</p>
</li>
<li> <p><code>observations</code>, the total number of available observations in that year, i.e. <code class="reqn">S + I + R</code>
</p>
</li>
<li> <p><code>observed</code>, the original observed resistant percentages
</p>
</li>
<li> <p><code>estimated</code>, the estimated resistant percentages, calculated by the model
</p>
</li></ul>

<p>Furthermore, the model itself is available as an attribute: <code>attributes(x)$model</code>, see <em>Examples</em>.
</p>


<h3>Interpretation of SIR</h3>

<p>In 2019, the European Committee on Antimicrobial Susceptibility Testing (EUCAST) has decided to change the definitions of susceptibility testing categories S, I, and R as shown below (<a href="https://www.eucast.org/newsiandr">https://www.eucast.org/newsiandr</a>):
</p>

<ul>
<li> <p><strong>S - Susceptible, standard dosing regimen</strong><br />
A microorganism is categorised as &quot;Susceptible, standard dosing regimen&quot;, when there is a high likelihood of therapeutic success using a standard dosing regimen of the agent.
</p>
</li>
<li> <p><strong>I - Susceptible, increased exposure</strong> <em><br />
A microorganism is categorised as &quot;Susceptible, Increased exposure</em>&quot; when there is a high likelihood of therapeutic success because exposure to the agent is increased by adjusting the dosing regimen or by its concentration at the site of infection.
</p>
</li>
<li> <p><strong>R = Resistant</strong><br />
A microorganism is categorised as &quot;Resistant&quot; when there is a high likelihood of therapeutic failure even when there is increased exposure.
</p>

<ul>
<li> <p><em>Exposure</em> is a function of how the mode of administration, dose, dosing interval, infusion time, as well as distribution and excretion of the antimicrobial agent will influence the infecting organism at the site of infection.
</p>
</li></ul>

</li></ul>

<p>This AMR package honours this insight. Use <code><a href="#topic+susceptibility">susceptibility()</a></code> (equal to <code><a href="#topic+proportion_SI">proportion_SI()</a></code>) to determine antimicrobial susceptibility and <code><a href="#topic+count_susceptible">count_susceptible()</a></code> (equal to <code><a href="#topic+count_SI">count_SI()</a></code>) to count susceptible isolates.
</p>


<h3>See Also</h3>

<p>The <code><a href="#topic+proportion">proportion()</a></code> functions to calculate resistance
</p>
<p>Models: <code><a href="stats.html#topic+lm">lm()</a></code> <code><a href="stats.html#topic+glm">glm()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- resistance_predict(example_isolates,
  col_ab = "AMX",
  year_min = 2010,
  model = "binomial"
)
plot(x)

if (require("ggplot2")) {
  ggplot_sir_predict(x)
}

# using dplyr:
if (require("dplyr")) {
  x &lt;- example_isolates %&gt;%
    filter_first_isolate() %&gt;%
    filter(mo_genus(mo) == "Staphylococcus") %&gt;%
    resistance_predict("PEN", model = "binomial")
  print(plot(x))

  # get the model from the object
  mymodel &lt;- attributes(x)$model
  summary(mymodel)
}

# create nice plots with ggplot2 yourself
if (require("dplyr") &amp;&amp; require("ggplot2")) {
  data &lt;- example_isolates %&gt;%
    filter(mo == as.mo("E. coli")) %&gt;%
    resistance_predict(
      col_ab = "AMX",
      col_date = "date",
      model = "binomial",
      info = FALSE,
      minimum = 15
    )
  head(data)
  autoplot(data)
}

</code></pre>

<hr>
<h2 id='skewness'>Skewness of the Sample</h2><span id='topic+skewness'></span><span id='topic+skewness.default'></span><span id='topic+skewness.matrix'></span><span id='topic+skewness.data.frame'></span>

<h3>Description</h3>

<p>Skewness is a measure of the asymmetry of the probability distribution of a real-valued random variable about its mean.
</p>
<p>When negative ('left-skewed'): the left tail is longer; the mass of the distribution is concentrated on the right of a histogram. When positive ('right-skewed'): the right tail is longer; the mass of the distribution is concentrated on the left of a histogram. A normal distribution has a skewness of 0.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>skewness(x, na.rm = FALSE)

## Default S3 method:
skewness(x, na.rm = FALSE)

## S3 method for class 'matrix'
skewness(x, na.rm = FALSE)

## S3 method for class 'data.frame'
skewness(x, na.rm = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="skewness_+3A_x">x</code></td>
<td>
<p>a vector of values, a <a href="base.html#topic+matrix">matrix</a> or a <a href="base.html#topic+data.frame">data.frame</a></p>
</td></tr>
<tr><td><code id="skewness_+3A_na.rm">na.rm</code></td>
<td>
<p>a <a href="base.html#topic+logical">logical</a> value indicating whether <code>NA</code> values should be stripped before the computation proceeds</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+kurtosis">kurtosis()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>skewness(runif(1000))
</code></pre>

<hr>
<h2 id='translate'>Translate Strings from the AMR Package</h2><span id='topic+translate'></span><span id='topic+get_AMR_locale'></span><span id='topic+set_AMR_locale'></span><span id='topic+reset_AMR_locale'></span><span id='topic+translate_AMR'></span>

<h3>Description</h3>

<p>For language-dependent output of <code>AMR</code> functions, such as <code><a href="#topic+mo_name">mo_name()</a></code>, <code><a href="#topic+mo_gramstain">mo_gramstain()</a></code>, <code><a href="#topic+mo_type">mo_type()</a></code> and <code><a href="#topic+ab_name">ab_name()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_AMR_locale()

set_AMR_locale(language)

reset_AMR_locale()

translate_AMR(x, language = get_AMR_locale())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translate_+3A_language">language</code></td>
<td>
<p>language to choose. Use one of these supported language names or ISO-639-1 codes: English (en), Chinese (zh), Czech (cs), Danish (da), Dutch (nl), Finnish (fi), French (fr), German (de), Greek (el), Italian (it), Japanese (ja), Norwegian (no), Polish (pl), Portuguese (pt), Romanian (ro), Russian (ru), Spanish (es), Swedish (sv), Turkish (tr), or Ukrainian (uk).</p>
</td></tr>
<tr><td><code id="translate_+3A_x">x</code></td>
<td>
<p>text to translate</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The currently 20 supported languages are English (en), Chinese (zh), Czech (cs), Danish (da), Dutch (nl), Finnish (fi), French (fr), German (de), Greek (el), Italian (it), Japanese (ja), Norwegian (no), Polish (pl), Portuguese (pt), Romanian (ro), Russian (ru), Spanish (es), Swedish (sv), Turkish (tr), and Ukrainian (uk). All these languages have translations available for all antimicrobial drugs and colloquial microorganism names.
</p>
<p>To permanently silence the once-per-session language note on a non-English operating system, you can set the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_locale</a></code> in your <code>.Rprofile</code> file like this:
</p>
<div class="sourceCode r"><pre># Open .Rprofile file
utils::file.edit("~/.Rprofile")

# Then add e.g. Italian support to that file using:
options(AMR_locale = "Italian")
</pre></div>
<p>And then save the file.
</p>
<p>Please read about adding or updating a language in <a href="https://github.com/msberends/AMR/wiki/">our Wiki</a>.
</p>


<h4>Changing the Default Language</h4>

<p>The system language will be used at default (as returned by <code>Sys.getenv("LANG")</code> or, if <code>LANG</code> is not set, <code><a href="base.html#topic+Sys.getlocale">Sys.getlocale(&quot;LC_COLLATE&quot;)</a></code>), if that language is supported. But the language to be used can be overwritten in two ways and will be checked in this order:
</p>

<ol>
<li><p> Setting the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_locale</a></code>, either by using e.g. <code>set_AMR_locale("German")</code> or by running e.g. <code>options(AMR_locale = "German")</code>.
</p>
<p>Note that setting an <span class="rlang"><b>R</b></span> option only works in the same session. Save the command <code>options(AMR_locale = "(your language)")</code> to your <code>.Rprofile</code> file to apply it for every session. Run <code>utils::file.edit("~/.Rprofile")</code> to edit your <code>.Rprofile</code> file.
</p>
</li>
<li><p> Setting the system variable <code>LANGUAGE</code> or <code>LANG</code>, e.g. by adding <code>LANGUAGE="de_DE.utf8"</code> to your <code>.Renviron</code> file in your home directory.
</p>
</li></ol>

<p>Thus, if the <a href="#topic+AMR-options">package option</a> <code><a href="#topic+AMR-options">AMR_locale</a></code> is set, the system variables <code>LANGUAGE</code> and <code>LANG</code> will be ignored.
</p>



<h3>Examples</h3>

<pre><code class='language-R'># Current settings (based on system language)
ab_name("Ciprofloxacin")
mo_name("Coagulase-negative Staphylococcus (CoNS)")

# setting another language
set_AMR_locale("Dutch")
ab_name("Ciprofloxacin")
mo_name("Coagulase-negative Staphylococcus (CoNS)")

# setting yet another language
set_AMR_locale("German")
ab_name("Ciprofloxacin")
mo_name("Coagulase-negative Staphylococcus (CoNS)")

# set_AMR_locale() understands endonyms, English exonyms, and ISO-639-1:
set_AMR_locale("Deutsch")
set_AMR_locale("German")
set_AMR_locale("de")
ab_name("amox/clav")

# reset to system default
reset_AMR_locale()
ab_name("amox/clav")
</code></pre>

<hr>
<h2 id='WHOCC'>WHOCC: WHO Collaborating Centre for Drug Statistics Methodology</h2><span id='topic+WHOCC'></span>

<h3>Description</h3>

<p>All antimicrobial drugs and their official names, ATC codes, ATC groups and defined daily dose (DDD) are included in this package, using the WHO Collaborating Centre for Drug Statistics Methodology.
</p>


<h3>WHOCC</h3>

<p>This package contains <strong>all ~550 antibiotic, antimycotic and antiviral drugs</strong> and their Anatomical Therapeutic Chemical (ATC) codes, ATC groups and Defined Daily Dose (DDD) from the World Health Organization Collaborating Centre for Drug Statistics Methodology (WHOCC, <a href="https://www.whocc.no">https://www.whocc.no</a>) and the Pharmaceuticals Community Register of the European Commission (<a href="https://ec.europa.eu/health/documents/community-register/html/reg_hum_atc.htm">https://ec.europa.eu/health/documents/community-register/html/reg_hum_atc.htm</a>).
</p>
<p>These have become the gold standard for international drug utilisation monitoring and research.
</p>
<p>The WHOCC is located in Oslo at the Norwegian Institute of Public Health and funded by the Norwegian government. The European Commission is the executive of the European Union and promotes its general interest.
</p>
<p><strong>NOTE: The WHOCC copyright does not allow use for commercial purposes, unlike any other info from this package.</strong> See <a href="https://www.whocc.no/copyright_disclaimer/.">https://www.whocc.no/copyright_disclaimer/.</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>as.ab("meropenem")
ab_name("J01DH02")

ab_tradenames("flucloxacillin")
</code></pre>

<hr>
<h2 id='WHONET'>Data Set with 500 Isolates - WHONET Example</h2><span id='topic+WHONET'></span>

<h3>Description</h3>

<p>This example data set has the exact same structure as an export file from WHONET. Such files can be used with this package, as this example data set shows. The antibiotic results are from our <a href="#topic+example_isolates">example_isolates</a> data set. All patient names were created using online surname generators and are only in place for practice purposes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WHONET
</code></pre>


<h3>Format</h3>

<p>A <a href="tibble.html#topic+tibble">tibble</a> with 500 observations and 53 variables:
</p>

<ul>
<li> <p><code style="white-space: pre;">&#8288;Identification number&#8288;</code><br /> ID of the sample
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;Specimen number&#8288;</code><br /> ID of the specimen
</p>
</li>
<li> <p><code>Organism</code><br /> Name of the microorganism. Before analysis, you should transform this to a valid microbial class, using <code><a href="#topic+as.mo">as.mo()</a></code>.
</p>
</li>
<li> <p><code>Country</code><br /> Country of origin
</p>
</li>
<li> <p><code>Laboratory</code><br /> Name of laboratory
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;Last name&#8288;</code><br /> Fictitious last name of patient
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;First name&#8288;</code><br /> Fictitious initial of patient
</p>
</li>
<li> <p><code>Sex</code><br /> Fictitious gender of patient
</p>
</li>
<li> <p><code>Age</code><br /> Fictitious age of patient
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;Age category&#8288;</code><br /> Age group, can also be looked up using <code><a href="#topic+age_groups">age_groups()</a></code>
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;Date of admission&#8288;</code><br /> <a href="base.html#topic+Date">Date</a> of hospital admission
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;Specimen date&#8288;</code><br /> <a href="base.html#topic+Date">Date</a> when specimen was received at laboratory
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;Specimen type&#8288;</code><br /> Specimen type or group
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;Specimen type (Numeric)&#8288;</code><br /> Translation of <code>"Specimen type"</code>
</p>
</li>
<li> <p><code>Reason</code><br /> Reason of request with Differential Diagnosis
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;Isolate number&#8288;</code><br /> ID of isolate
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;Organism type&#8288;</code><br /> Type of microorganism, can also be looked up using <code><a href="#topic+mo_type">mo_type()</a></code>
</p>
</li>
<li> <p><code>Serotype</code><br /> Serotype of microorganism
</p>
</li>
<li> <p><code>Beta-lactamase</code><br /> Microorganism produces beta-lactamase?
</p>
</li>
<li> <p><code>ESBL</code><br /> Microorganism produces extended spectrum beta-lactamase?
</p>
</li>
<li> <p><code>Carbapenemase</code><br /> Microorganism produces carbapenemase?
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;MRSA screening test&#8288;</code><br /> Microorganism is possible MRSA?
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;Inducible clindamycin resistance&#8288;</code><br /> Clindamycin can be induced?
</p>
</li>
<li> <p><code>Comment</code><br /> Other comments
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;Date of data entry&#8288;</code><br /> <a href="base.html#topic+Date">Date</a> this data was entered in WHONET
</p>
</li>
<li> <p><code>AMP_ND10:CIP_EE</code><br /> 28 different antibiotics. You can lookup the abbreviations in the <a href="#topic+antibiotics">antibiotics</a> data set, or use e.g. <code><a href="#topic+ab_name">ab_name(&quot;AMP&quot;)</a></code> to get the official name immediately. Before analysis, you should transform this to a valid antibiotic class, using <code><a href="#topic+as.sir">as.sir()</a></code>.
</p>
</li></ul>



<h3>Details</h3>

<p>Like all data sets in this package, this data set is publicly available for download in the following formats: R, MS Excel, Apache Feather, Apache Parquet, SPSS, SAS, and Stata. Please visit <a href="https://msberends.github.io/AMR/articles/datasets.html">our website for the download links</a>. The actual files are of course available on <a href="https://github.com/msberends/AMR/tree/main/data-raw">our GitHub repository</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>WHONET
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
