<!DOCTYPE html><html lang="en"><head><title>Help for package AmyloGram</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {AmyloGram}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AmyloGram-package'><p>Prediction of amyloids</p></a></li>
<li><a href='#AmyloGram_gui'><p>AmyloGram Graphical User Interface</p></a></li>
<li><a href='#AmyloGram_model'><p>Random forest model of amyloid proteins</p></a></li>
<li><a href='#is_protein'><p>Protein test</p></a></li>
<li><a href='#pep424'><p>pep424 data set</p></a></li>
<li><a href='#predict.ag_model'><p>Predict amyloids</p></a></li>
<li><a href='#print.ag_model'><p>Print AmyloGram object</p></a></li>
<li><a href='#read_txt'><p>Read sequences from .txt file</p></a></li>
<li><a href='#spec_sens'><p>Specificity/sensitivity balance</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Prediction of Amyloid Proteins</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-10-11</td>
</tr>
<tr>
<td>Description:</td>
<td>Predicts amyloid proteins using random forests trained on the
    n-gram encoded peptides. The implemented algorithm can be accessed from
    both the command line and shiny-based GUI.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/michbur/AmyloGram">https://github.com/michbur/AmyloGram</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/michbur/AmyloGram/issues">https://github.com/michbur/AmyloGram/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>biogram, ranger, seqinr, shiny</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-10-11 14:35:25 UTC; michal</td>
</tr>
<tr>
<td>Author:</td>
<td>Michal Burdukiewicz [cre, aut],
  Piotr Sobczyk [ctb],
  Stefan Roediger [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Michal Burdukiewicz &lt;michalburdukiewicz@gmail.com&gt;</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-10-11 14:46:15 UTC</td>
</tr>
</table>
<hr>
<h2 id='AmyloGram-package'>Prediction of amyloids</h2><span id='topic+AmyloGram-package'></span><span id='topic+AmyloGram'></span>

<h3>Description</h3>

<p>Amyloids are proteins associated with the number of clinical disorders (e.g.,
Alzheimer's, Creutzfeldt-Jakob's and Huntington's diseases). Despite their
diversity, all amyloid proteins can undergo aggregation initiated by 6- to
15-residue segments called hot spots. Henceforth, amyloids form unique,
zipper-like beta-structures, which are often harmful. To find the patterns
defining the hot spots, we developed our novel predictor of amyloidogenicity
AmyloGram, based on random forests.
</p>


<h3>Details</h3>

<p>AmyloGram is available as R function (<code><a href="#topic+predict.ag_model">predict.ag_model</a></code>) or
shiny GUI (<code><a href="#topic+AmyloGram_gui">AmyloGram_gui</a></code>).
</p>
<p>The package is enriched with the benchmark data set <code><a href="#topic+pep424">pep424</a></code>.
</p>


<h3>Author(s)</h3>

<p>Maintainer: Michal Burdukiewicz &lt;michalburdukiewicz@gmail.com&gt;
</p>


<h3>References</h3>

<p>Burdukiewicz MJ, Sobczyk P, Roediger S, Duda-Madej A,
Mackiewicz P, Kotulska M. (2017) <em>Amyloidogenic motifs revealed
by n-gram analysis</em>. Scientific Reports 7
<a href="https://doi.org/10.1038/s41598-017-13210-9">https://doi.org/10.1038/s41598-017-13210-9</a>
</p>

<hr>
<h2 id='AmyloGram_gui'>AmyloGram Graphical User Interface</h2><span id='topic+AmyloGram_gui'></span>

<h3>Description</h3>

<p>Launches graphical user interface that predicts presence of amyloids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AmyloGram_gui()
</code></pre>


<h3>Warning </h3>

<p>Any ad-blocking software may cause malfunctions.
</p>

<hr>
<h2 id='AmyloGram_model'>Random forest model of amyloid proteins</h2><span id='topic+AmyloGram_model'></span>

<h3>Description</h3>

<p>Random forest grown using the <code>ranger</code> package with additional
information.
</p>


<h3>Format</h3>

<p>A list of length three: random forest, a vector of important n-grams
and the best-performing encoding.</p>


<h3>See Also</h3>

<p><code><a href="ranger.html#topic+ranger">ranger</a></code>
</p>

<hr>
<h2 id='is_protein'>Protein test</h2><span id='topic+is_protein'></span>

<h3>Description</h3>

<p>Checks if an object is a protein (contains letters from one-letter amino acid code).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_protein(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is_protein_+3A_object">object</code></td>
<td>
<p><code>character</code> vector where each elemenents represent one amino acid.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>TRUE</code> or <code>FALSE</code>.
</p>

<hr>
<h2 id='pep424'>pep424 data set</h2><span id='topic+pep424'></span>

<h3>Description</h3>

<p>Benchmark dataset for PASTA 2.0. 5 sequences shorter than 6 amino acids
(1% of the original dataset) were removed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pep424
</code></pre>


<h3>Format</h3>

<p>a list of 424 peptides (class <code><a href="seqinr.html#topic+SeqFastaAA">SeqFastaAA</a></code>).</p>


<h3>Source</h3>

<p>Walsh, I., Seno, F., Tosatto, S.C.E., and Trovato, A. (2014).
<em>PASTA 2.0: an improved server for protein aggregation prediction</em>.
Nucleic Acids Research gku399.
</p>

<hr>
<h2 id='predict.ag_model'>Predict amyloids</h2><span id='topic+predict.ag_model'></span>

<h3>Description</h3>

<p>Recognizes amyloids using AmyloGram algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ag_model'
predict(object, newdata, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predict.ag_model_+3A_object">object</code></td>
<td>
<p><code>ag_model</code> object.</p>
</td></tr>
<tr><td><code id="predict.ag_model_+3A_newdata">newdata</code></td>
<td>
<p><code>list</code> of sequences (for example as given by
<code><a href="seqinr.html#topic+read.fasta">read.fasta</a></code>).</p>
</td></tr>
<tr><td><code id="predict.ag_model_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(AmyloGram_model)
data(pep424)
predict(AmyloGram_model, pep424[17])
</code></pre>

<hr>
<h2 id='print.ag_model'>Print AmyloGram object</h2><span id='topic+print.ag_model'></span>

<h3>Description</h3>

<p>Prints <code>ag_model</code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ag_model'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.ag_model_+3A_x">x</code></td>
<td>
<p><code>ag_model</code> object.</p>
</td></tr>
<tr><td><code id="print.ag_model_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(AmyloGram_model)
print(AmyloGram_model)
</code></pre>

<hr>
<h2 id='read_txt'>Read sequences from .txt file</h2><span id='topic+read_txt'></span>

<h3>Description</h3>

<p>Read sequence data saved in text file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_txt(connection)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_txt_+3A_connection">connection</code></td>
<td>
<p>a <code><a href="base.html#topic+connection">connection</a></code> to the text (.txt) file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input file should contain one or more amino acid sequences separated by 
empty line(s).
</p>


<h3>Value</h3>

<p>a list of sequences. Each element has class <code><a href="seqinr.html#topic+SeqFastaAA">SeqFastaAA</a></code>. If
connection contains no characters, function prompts warning and returns <code>NULL</code>.
</p>

<hr>
<h2 id='spec_sens'>Specificity/sensitivity balance</h2><span id='topic+spec_sens'></span>

<h3>Description</h3>

<p>Sensitivity, specificity and Matthew's Correlation Coefficient
of AmyloGram for different cutoffs computed on <code>pep424</code> dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spec_sens
</code></pre>


<h3>Format</h3>

<p>a data frame with four columns and 99 rows.</p>


<h3>Source</h3>

<p>Walsh, I., Seno, F., Tosatto, S.C.E., and Trovato, A. (2014).
<em>PASTA 2.0: an improved server for protein aggregation prediction</em>.
Nucleic Acids Research gku399.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
