<!DOCTYPE html><html><head><title>Help for package AnaCoDa</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {AnaCoDa}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AAToCodon'><p>Amino Acid to codon set</p></a></li>
<li><a href='#acfCSP'><p>Plots ACF for codon specific parameter traces</p></a></li>
<li><a href='#acfMCMC'><p>Autocorrelation function for the likelihood or posterior trace</p></a></li>
<li><a href='#addObservedSynthesisRateSet'><p>Add gene observed synthesis rates</p></a></li>
<li><a href='#aminoAcids'><p>Amino acids</p></a></li>
<li><a href='#calculateMarginalLogLikelihood'><p>Calculates the marginal log-likelihood for a set of parameters</p></a></li>
<li><a href='#calculateSCUO'><p>calculates the synonymous codon usage order (SCUO)</p></a></li>
<li><a href='#codons'><p>Codons</p></a></li>
<li><a href='#codonToAA'><p>translates codon to amino acid</p></a></li>
<li><a href='#convergence.test'><p>Convergence Test</p></a></li>
<li><a href='#findOptimalCodon'><p>Find and return list of optimal codons</p></a></li>
<li><a href='#fixDEta'><p>fixDEta</p></a></li>
<li><a href='#fixDM'><p>fixDM</p></a></li>
<li><a href='#fixSphi'><p>fixSphi</p></a></li>
<li><a href='#geomMean'><p>Take the geometric mean of a vector</p></a></li>
<li><a href='#getAdaptiveWidth'><p>getAdaptiveWidth</p></a></li>
<li><a href='#getCAI'><p>Calculate the Codon Adaptation Index</p></a></li>
<li><a href='#getCAIweights'><p>Calculate the CAI codon weigths for a reference genome</p></a></li>
<li><a href='#getCodonCounts'><p>Get Codon Counts For all Amino Acids</p></a></li>
<li><a href='#getCodonCountsForAA'><p>Get Codon Counts For a specific Amino Acid</p></a></li>
<li><a href='#getCodonSpecificPosteriorMeanForCodon'><p>getCodonSpecificPosteriorMeanForCodon</p></a></li>
<li><a href='#getCodonSpecificPosteriorVarianceForCodon'><p>getCodonSpecificPosteriorVarianceForCodon</p></a></li>
<li><a href='#getCodonSpecificQuantilesForCodon'><p>getCodonSpecificQuantilesForCodon</p></a></li>
<li><a href='#getCSPEstimates'><p>Return Codon Specific Paramters (or write to csv) estimates as data.frame</p></a></li>
<li><a href='#getEstimatedMixtureAssignmentForGene'><p>getEstimatedMixtureAssignmentForGene</p></a></li>
<li><a href='#getEstimatedMixtureAssignmentProbabilitiesForGene'><p>getEstimatedMixtureAssignmentProbabilitiesForGene</p></a></li>
<li><a href='#getExpressionEstimates'><p>Returns the estimated phi posterior for a gene</p></a></li>
<li><a href='#getGroupList'><p>getGroupList</p></a></li>
<li><a href='#getLogLikelihoodTrace'><p>getLogLikelihoodTrace</p></a></li>
<li><a href='#getLogPosteriorMean'><p>getLogPosteriorMean</p></a></li>
<li><a href='#getLogPosteriorTrace'><p>getLogPosteriorTrace</p></a></li>
<li><a href='#getMixtureAssignmentEstimate'><p>Returns mixture assignment estimates for each gene</p></a></li>
<li><a href='#getNames'><p>Gene Names of Genome</p></a></li>
<li><a href='#getNc'><p>Calculate the Effective Number of Codons</p></a></li>
<li><a href='#getNcAA'><p>Calculate the Effective Number of Codons for each Amino Acid</p></a></li>
<li><a href='#getNoiseOffsetPosteriorMean'><p>getNoiseOffsetPosteriorMean</p></a></li>
<li><a href='#getNoiseOffsetVariance'><p>getNoiseOffsetVariance</p></a></li>
<li><a href='#getObservedSynthesisRateSet'><p>Get gene observed synthesis rates</p></a></li>
<li><a href='#getSamples'><p>getSamples</p></a></li>
<li><a href='#getSelectionCoefficients'><p>Calculate Selection coefficients</p></a></li>
<li><a href='#getStdDevSynthesisRatePosteriorMean'><p>getStdDevSynthesisRatePosteriorMean</p></a></li>
<li><a href='#getStdDevSynthesisRateVariance'><p>getStdDevSynthesisRateVariance</p></a></li>
<li><a href='#getStepsToAdapt'><p>getStepsToAdapt</p></a></li>
<li><a href='#getSynthesisRate'><p>getSynthesisRate</p></a></li>
<li><a href='#getSynthesisRatePosteriorMeanForGene'><p>getSynthesisRatePosteriorMeanForGene</p></a></li>
<li><a href='#getSynthesisRatePosteriorVarianceForGene'><p>getSynthesisRatePosteriorVarianceForGene</p></a></li>
<li><a href='#getThinning'><p>getThinning</p></a></li>
<li><a href='#getTrace'><p>extracts an object of traces from a parameter object.</p></a></li>
<li><a href='#getTraceObject'><p>getTraceObject</p></a></li>
<li><a href='#initializeCovarianceMatrices'><p>Initialize Covariance Matrices</p></a></li>
<li><a href='#initializeGenomeObject'><p>Genome Initialization</p></a></li>
<li><a href='#initializeMCMCObject'><p>Initialize MCMC</p></a></li>
<li><a href='#initializeModelObject'><p>Model Initialization</p></a></li>
<li><a href='#initializeParameterObject'><p>Initialize Parameter</p></a></li>
<li><a href='#initializeSynthesisRateByGenome'><p>initializeSynthesisRateByGenome</p></a></li>
<li><a href='#initializeSynthesisRateByList'><p>initializeSynthesisRateByList</p></a></li>
<li><a href='#initializeSynthesisRateByRandom'><p>initializeSynthesisRateByRandom</p></a></li>
<li><a href='#initMutationCategories'><p>initMutationCategories</p></a></li>
<li><a href='#initSelectionCategories'><p>initSelectionCategories</p></a></li>
<li><a href='#length.Rcpp_Genome'><p>Length of Genome</p></a></li>
<li><a href='#loadMCMCObject'><p>Load MCMC Object</p></a></li>
<li><a href='#loadParameterObject'><p>Load Parameter Object</p></a></li>
<li><a href='#plot.Rcpp_FONSEModel'><p>Plot Model Object</p></a></li>
<li><a href='#plot.Rcpp_FONSEParameter'><p>Plot Parameter</p></a></li>
<li><a href='#plot.Rcpp_MCMCAlgorithm'><p>Plot MCMC algorithm</p></a></li>
<li><a href='#plot.Rcpp_ROCModel'><p>Plot Model Object</p></a></li>
<li><a href='#plot.Rcpp_ROCParameter'><p>Plot Parameter</p></a></li>
<li><a href='#plot.Rcpp_Trace'><p>Plot Trace Object</p></a></li>
<li><a href='#plotAcceptanceRatios'><p>Plot Acceptance ratios</p></a></li>
<li><a href='#plotCodonSpecificParameters'><p>Plot Codon Specific Parameter</p></a></li>
<li><a href='#readPhiValue'><p>readPhiValue</p></a></li>
<li><a href='#runMCMC'><p>Run MCMC</p></a></li>
<li><a href='#setAdaptiveWidth'><p>setAdaptiveWidth</p></a></li>
<li><a href='#setGroupList'><p>setGroupList</p></a></li>
<li><a href='#setRestartFileSettings'><p>setRestartFileSettings</p></a></li>
<li><a href='#setRestartSettings'><p>Set Restart Settings</p></a></li>
<li><a href='#setSamples'><p>setSamples</p></a></li>
<li><a href='#setStepsToAdapt'><p>setStepsToAdapt</p></a></li>
<li><a href='#setThinning'><p>setThinning</p></a></li>
<li><a href='#simulateGenome'><p>simulateGenome</p></a></li>
<li><a href='#summary.Rcpp_Genome'><p>Summary of Genome</p></a></li>
<li><a href='#writeMCMCObject'><p>Write MCMC Object</p></a></li>
<li><a href='#writeParameterObject'><p>Write Parameter Object to a File</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Analysis of Codon Data under Stationarity using a Bayesian
Framework</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.4.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-09-11</td>
</tr>
<tr>
<td>Author:</td>
<td>Authors@R</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Cedric Landerer &lt;cedric.landerer@gmail.com&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/clandere/AnaCoDa">https://github.com/clandere/AnaCoDa</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.0), Rcpp (&ge; 0.11.3), VGAM, methods, mvtnorm</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, Hmisc, coda, testthat, lmodel2, markdown</td>
</tr>
<tr>
<td>RcppModules:</td>
<td>Test_mod, Trace_mod, CovarianceMatrix_mod,
MCMCAlgorithm_mod, Model_mod, Parameter_mod, Genome_mod,
Gene_mod, SequenceSummary_mod</td>
</tr>
<tr>
<td>Description:</td>
<td>Is a collection of models to analyze genome scale codon
        data using a Bayesian framework. Provides visualization
        routines and checkpointing for model fittings. Currently
        published models to analyze gene data for selection on codon
        usage based on Ribosome Overhead Cost (ROC) are: ROC (Gilchrist
        et al. (2015) &lt;<a href="https://doi.org/10.1093%2Fgbe%2Fevv087">doi:10.1093/gbe/evv087</a>&gt;), and ROC with phi
        (Wallace &amp; Drummond (2013) &lt;<a href="https://doi.org/10.1093%2Fmolbev%2Fmst051">doi:10.1093/molbev/mst051</a>&gt;). In
        addition 'AnaCoDa' contains three currently unpublished models.
        The FONSE (First order approximation On NonSense Error) model
        analyzes gene data for selection on codon usage against of
        nonsense error rates. The PA (PAusing time) and PANSE (PAusing
        time + NonSense Error) models use ribosome footprinting data to
        analyze estimate ribosome pausing times with and without
        nonsense error rate from ribosome footprinting data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-09-15 09:04:26 UTC; landerer</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-09-15 09:40:19 UTC</td>
</tr>
</table>
<hr>
<h2 id='AAToCodon'>Amino Acid to codon set</h2><span id='topic+AAToCodon'></span>

<h3>Description</h3>

<p>Converts one character amino acid code to the set of codon encoding that amino acid
</p>


<h3>Usage</h3>

<pre><code class='language-R'> AAToCodon(aa, focal = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AAToCodon_+3A_aa">aa</code></td>
<td>
<p>Amino acid in single character notation</p>
</td></tr>
<tr><td><code id="AAToCodon_+3A_focal">focal</code></td>
<td>
<p>logical, Include the alphabetically last (focal) codon</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the names of the codon encoding the give amino acid
</p>


<h3>See Also</h3>

<p><code><a href="#topic+codonToAA">codonToAA</a></code>
</p>

<hr>
<h2 id='acfCSP'>Plots ACF for codon specific parameter traces</h2><span id='topic+acfCSP'></span>

<h3>Description</h3>

<p>The function calculates and by defaults plots the acf and estimates the autocorrelation in the trace
</p>


<h3>Usage</h3>

<pre><code class='language-R'>acfCSP(
  parameter,
  csp = "Mutation",
  numMixtures = 1,
  samples = NULL,
  lag.max = 40,
  plot = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="acfCSP_+3A_parameter">parameter</code></td>
<td>
<p>object of class Parameter</p>
</td></tr>
<tr><td><code id="acfCSP_+3A_csp">csp</code></td>
<td>
<p>indicates which parameter to calculate the autocorrelation. Must be Mutation (the default, ROC, FONSE), Selection (ROC, FONSE), Alpha (PA, PANSE), LambdaPrime (PA, PANSE), NSERate (PA, PANSE)&quot;</p>
</td></tr>
<tr><td><code id="acfCSP_+3A_nummixtures">numMixtures</code></td>
<td>
<p>indicates the number of CSP mixtures used</p>
</td></tr>
<tr><td><code id="acfCSP_+3A_samples">samples</code></td>
<td>
<p>number of samples at the end of the trace used to calculate the acf</p>
</td></tr>
<tr><td><code id="acfCSP_+3A_lag.max">lag.max</code></td>
<td>
<p>Maximum amount of lag to calculate acf. Default is 10*log10(N), where N i the number of observations.</p>
</td></tr>
<tr><td><code id="acfCSP_+3A_plot">plot</code></td>
<td>
<p>logical. If TRUE (default) a plot of the acf is created</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+acfMCMC">acfMCMC</a></code>
</p>

<hr>
<h2 id='acfMCMC'>Autocorrelation function for the likelihood or posterior trace</h2><span id='topic+acfMCMC'></span>

<h3>Description</h3>

<p>The function calculates and by defaults plots the acf and estimates the autocorrelation in the trace.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>acfMCMC(mcmc, type = "LogPosterior", samples = NULL, lag.max = 40, plot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="acfMCMC_+3A_mcmc">mcmc</code></td>
<td>
<p>object of class MCMC</p>
</td></tr>
<tr><td><code id="acfMCMC_+3A_type">type</code></td>
<td>
<p>&quot;LogPosterior&quot; or &quot;LogLikelihood&quot;, defaults to &quot;LogPosterior&quot;</p>
</td></tr>
<tr><td><code id="acfMCMC_+3A_samples">samples</code></td>
<td>
<p>number of samples at the end of the trace used to calculate the acf</p>
</td></tr>
<tr><td><code id="acfMCMC_+3A_lag.max">lag.max</code></td>
<td>
<p>Maximum amount of lag to calculate acf. Default is 10*log10(N), where N i the number of observations.</p>
</td></tr>
<tr><td><code id="acfMCMC_+3A_plot">plot</code></td>
<td>
<p>logical. If TRUE (default) a plot of the acf is created</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+acfCSP">acfCSP</a></code>
</p>

<hr>
<h2 id='addObservedSynthesisRateSet'>Add gene observed synthesis rates</h2><span id='topic+addObservedSynthesisRateSet'></span>

<h3>Description</h3>

<p><code>addObservedSynthesisRateSet</code> returns the observed 
synthesis rates of the genes within the genome specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addObservedSynthesisRateSet(
  genome,
  observed.expression.file,
  match.expression.by.id = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addObservedSynthesisRateSet_+3A_genome">genome</code></td>
<td>
<p>A genome object initialized with 
<code><a href="#topic+initializeGenomeObject">initializeGenomeObject</a></code> to add observed expression data.</p>
</td></tr>
<tr><td><code id="addObservedSynthesisRateSet_+3A_observed.expression.file">observed.expression.file</code></td>
<td>
<p>A string containing 
the location of a file containing empirical expression rates (optional).</p>
</td></tr>
<tr><td><code id="addObservedSynthesisRateSet_+3A_match.expression.by.id">match.expression.by.id</code></td>
<td>
<p>If TRUE (default) observed expression 
values will be assigned by matching sequence identifier.
If FALSE observed expression values will be assigned by order</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the genome after adding the new gene expression values
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")
expression_file &lt;- system.file("extdata", "expression.csv", package = "AnaCoDa") 
## reading genome
genome &lt;- initializeGenomeObject(file = genome_file)


## add expression values after the genome was initiallized, 
## or adding an additional set of expression values
genome &lt;- addObservedSynthesisRateSet(genome = genome, 
                   observed.expression.file = expression_file)

</code></pre>

<hr>
<h2 id='aminoAcids'>Amino acids</h2><span id='topic+aminoAcids'></span>

<h3>Description</h3>

<p>Returns a vector of all amino acids
</p>


<h3>Usage</h3>

<pre><code class='language-R'> aminoAcids()
</code></pre>


<h3>Value</h3>

<p>Returns a vector of all amino acids
</p>


<h3>See Also</h3>

<p><code><a href="#topic+codons">codons</a></code>
</p>

<hr>
<h2 id='calculateMarginalLogLikelihood'>Calculates the marginal log-likelihood for a set of parameters</h2><span id='topic+calculateMarginalLogLikelihood'></span>

<h3>Description</h3>

<p>initializes the model object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateMarginalLogLikelihood(
  parameter,
  mcmc,
  mixture,
  n.samples,
  divisor,
  warnings = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateMarginalLogLikelihood_+3A_parameter">parameter</code></td>
<td>
<p>An object created with <code>initializeParameterObject</code>.</p>
</td></tr>
<tr><td><code id="calculateMarginalLogLikelihood_+3A_mcmc">mcmc</code></td>
<td>
<p>An object created with <code>initializeMCMCObject</code></p>
</td></tr>
<tr><td><code id="calculateMarginalLogLikelihood_+3A_mixture">mixture</code></td>
<td>
<p>determines for which mixture the marginal log-likelihood should be calculated</p>
</td></tr>
<tr><td><code id="calculateMarginalLogLikelihood_+3A_n.samples">n.samples</code></td>
<td>
<p>How many samples should be used for the calculation</p>
</td></tr>
<tr><td><code id="calculateMarginalLogLikelihood_+3A_divisor">divisor</code></td>
<td>
<p>A value &gt; 1 in order to scale down the tails of the importance distribution</p>
</td></tr>
<tr><td><code id="calculateMarginalLogLikelihood_+3A_warnings">warnings</code></td>
<td>
<p>Print warnings such as when the variance of a parameter is 0, which might occur when parameter is fixed</p>
</td></tr>
</table>


<h3>Details</h3>

<p>calculateMarginalLogLikelihood Calculate marginal log-likelihood for
calculation of the Bayes factor using a generalized harmonix mean estimator of the 
marginal likelihood. See Gronau et al. (2017) for details
</p>


<h3>Value</h3>

<p>This function returns the model object created.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Calculate the log-marginal likelihood
parameter &lt;- loadParameterObject("parameter.Rda")
mcmc &lt;- loadMCMCObject("mcmc.Rda")
calculate_marginal_likelihood(parameter, mcmc, mixture = 1, 
samples = 500, scaling = 1.5)

# Calculate the Bayes factor for two models
parameter1 &lt;- loadParameterObject("parameter1.Rda")
parameter2 &lt;- loadParameterObject("parameter2.Rda")
mcmc1 &lt;- loadMCMCObject("mcmc1.Rda")
mcmc2 &lt;- loadMCMCObject("mcmc2.Rda")
mll1 &lt;- calculate_marginal_likelihood(parameter1, mcmc1, mixture = 1, 
samples = 500, scaling = 1.5)
mll2 &lt;- calculate_marginal_likelihood(parameter2, mcmc2, mixture = 1, 
samples = 500, scaling = 1.5)
cat("Bayes factor: ", mll1 - mll2, "\n")

## End(Not run)
 
</code></pre>

<hr>
<h2 id='calculateSCUO'>calculates the synonymous codon usage order (SCUO)</h2><span id='topic+calculateSCUO'></span>

<h3>Description</h3>

<p><code>calculateSCUO</code> calulates the SCUO value for each gene in genome. Note that if a codon is absent, this will be treated as NA and will be skipped in final calculation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateSCUO(genome)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateSCUO_+3A_genome">genome</code></td>
<td>
<p>A genome object initialized with <code><a href="#topic+initializeGenomeObject">initializeGenomeObject</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns the SCUO value for each gene in genome
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")
 
## reading genome
genome &lt;- initializeGenomeObject(file = genome_file)
scuo &lt;- calculateSCUO(genome)

</code></pre>

<hr>
<h2 id='codons'>Codons</h2><span id='topic+codons'></span>

<h3>Description</h3>

<p>Returns a vector of all codons
</p>


<h3>Usage</h3>

<pre><code class='language-R'> codons()
</code></pre>


<h3>Value</h3>

<p>Returns a vector of all codons
</p>


<h3>See Also</h3>

<p><code><a href="#topic+aminoAcids">aminoAcids</a></code>
</p>

<hr>
<h2 id='codonToAA'>translates codon to amino acid</h2><span id='topic+codonToAA'></span>

<h3>Description</h3>

<p>Translates a given codon into the amino acid encoded by it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> codonToAA(codon)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="codonToAA_+3A_codon">codon</code></td>
<td>
<p>character, codon to translate</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the amino acid encoded by the given codon as character
</p>


<h3>See Also</h3>

<p><code><a href="#topic+AAToCodon">AAToCodon</a></code>
</p>

<hr>
<h2 id='convergence.test'>Convergence Test</h2><span id='topic+convergence.test'></span>

<h3>Description</h3>

<p>Convergence Test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convergence.test(
  object,
  samples = 10,
  frac1 = 0.1,
  frac2 = 0.5,
  thin = 1,
  plot = FALSE,
  what = "Mutation",
  mixture = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convergence.test_+3A_object">object</code></td>
<td>
<p>an object of either class Trace or MCMC</p>
</td></tr>
<tr><td><code id="convergence.test_+3A_samples">samples</code></td>
<td>
<p>number of samples at the end of the trace used to determine convergence (&lt; length of trace). 
Will use as starting point of convergence test. If the MCMC trace
is of length x, then starting point for convergence test will be x - samples.</p>
</td></tr>
<tr><td><code id="convergence.test_+3A_frac1">frac1</code></td>
<td>
<p>fraction to use from beginning of samples</p>
</td></tr>
<tr><td><code id="convergence.test_+3A_frac2">frac2</code></td>
<td>
<p>fraction to use from end of samples</p>
</td></tr>
<tr><td><code id="convergence.test_+3A_thin">thin</code></td>
<td>
<p>the thinning interval between consecutive observations, which is used in creating a coda::mcmc object (according to the Coda documentation, users should specify if a MCMC chain has already been thinned using a the thin parameter). This does not further thin the data.</p>
</td></tr>
<tr><td><code id="convergence.test_+3A_plot">plot</code></td>
<td>
<p>(logical) plot result instead of returning an object</p>
</td></tr>
<tr><td><code id="convergence.test_+3A_what">what</code></td>
<td>
<p>(for Trace Object only) which parameter to calculate convergence.test &ndash; current options are Selection, Mutation, MixtureProbability, Sphi, Mphi, ExpectedPhi, and AcceptanceCSP</p>
</td></tr>
<tr><td><code id="convergence.test_+3A_mixture">mixture</code></td>
<td>
<p>(for Trace Object only) mixture for which to calculate convergence.test</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Be aware that convergence.test for Trace objects works primarily for Trace objects from the ROC parameter class. Future updates will adapt this function to work for parameters from other models and expression traces
</p>


<h3>Value</h3>

<p>Geweke score object evaluating whether means of two fractions (frac1 and frac2) differ.  Convergence occurs when they don't differ significantly, i.e. pnorm(abs(convergence.test(mcmcObj)$a, ,lower.tail=FALSE)*2 &gt; 0.05
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## check for convergence after a run:

genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")

genome &lt;- initializeGenomeObject(file = genome_file)
sphi_init &lt;- c(1,1)
numMixtures &lt;- 2
geneAssignment &lt;- c(rep(1,floor(length(genome)/2)),rep(2,ceiling(length(genome)/2)))
parameter &lt;- initializeParameterObject(genome = genome, sphi = sphi_init, 
                                       num.mixtures = numMixtures, 
                                       gene.assignment = geneAssignment, 
                                       mixture.definition = "allUnique")
samples &lt;- 2500
thinning &lt;- 50
adaptiveWidth &lt;- 25
mcmc &lt;- initializeMCMCObject(samples = samples, thinning = thinning, 
                             adaptive.width=adaptiveWidth, est.expression=TRUE, 
                             est.csp=TRUE, est.hyper=TRUE, est.mix = TRUE) 
divergence.iteration &lt;- 10
## Not run: 
runMCMC(mcmc = mcmc, genome = genome, model = model, 
        ncores = 4, divergence.iteration = divergence.iteration)
# check if posterior trace has converged
convergence.test(object = mcmc, samples = 500, plot = TRUE)

trace &lt;- getTrace(parameter)
# check if Mutation trace has converged
convergence.test(object = trace, samples = 500, plot = TRUE, what = "Mutation")
# check if Sphi trace has converged
convergence.test(object = trace, samples = 500, plot = TRUE, what = "Sphi")
# check if ExpectedPhi trace has converged
convergence.test(object = trace, samples = 500, plot = TRUE, what = "ExpectedPhi")

## End(Not run)
</code></pre>

<hr>
<h2 id='findOptimalCodon'>Find and return list of optimal codons</h2><span id='topic+findOptimalCodon'></span>

<h3>Description</h3>

<p><code>findOptimalCodon</code> extracrs the optimal codon for each amino acid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findOptimalCodon(csp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="findOptimalCodon_+3A_csp">csp</code></td>
<td>
<p>a <code>data.frame</code> as returned by <code>getCSPEstimates</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list with with optimal codons for each amino acid.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")

genome &lt;- initializeGenomeObject(file = genome_file)
sphi_init &lt;- 1
numMixtures &lt;- 1
geneAssignment &lt;- rep(1, length(genome))
parameter &lt;- initializeParameterObject(genome = genome, sphi = sphi_init, 
                                       num.mixtures = numMixtures, 
                                       gene.assignment = geneAssignment, 
                                       mixture.definition = "allUnique")
model &lt;- initializeModelObject(parameter = parameter, model = "ROC")
samples &lt;- 2500
thinning &lt;- 50
adaptiveWidth &lt;- 25
mcmc &lt;- initializeMCMCObject(samples = samples, thinning = thinning, 
                             adaptive.width=adaptiveWidth, est.expression=TRUE, 
                             est.csp=TRUE, est.hyper=TRUE, est.mix = TRUE) 
divergence.iteration &lt;- 10
## Not run: 
runMCMC(mcmc = mcmc, genome = genome, model = model, 
        ncores = 4, divergence.iteration = divergence.iteration)

csp_mat &lt;- getCSPEstimates(parameter, CSP="Selection")
opt_codons &lt;- findOptimalCodon(csp_mat)

## End(Not run)
</code></pre>

<hr>
<h2 id='fixDEta'>fixDEta</h2><span id='topic+fixDEta'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Fix the value of selection its current value
</p>

<hr>
<h2 id='fixDM'>fixDM</h2><span id='topic+fixDM'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Fix the value of mutation its current value
</p>

<hr>
<h2 id='fixSphi'>fixSphi</h2><span id='topic+fixSphi'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Fix the value of s_phi (standard deviation of lognormal for synthesis rates) at its current value
</p>

<hr>
<h2 id='geomMean'>Take the geometric mean of a vector</h2><span id='topic+geomMean'></span>

<h3>Description</h3>

<p><code>geomMean</code> will calculate the geometric mean of a list of numerical values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geomMean(x, rm.invalid = TRUE, default = 1e-05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geomMean_+3A_x">x</code></td>
<td>
<p>A vector of numerical .</p>
</td></tr>
<tr><td><code id="geomMean_+3A_rm.invalid">rm.invalid</code></td>
<td>
<p>Boolean value for handling 0, negative, or NA values in the vector. Default is TRUE and will not
include these values in the calculation. If FALSE, these values will be replaced by the value give to <code>default</code> and will
be included in the calculation.</p>
</td></tr>
<tr><td><code id="geomMean_+3A_default">default</code></td>
<td>
<p>Numerical value that serves as the value to replace 0, negative, or NA values in the calculation when rm.invalid is FALSE.
Default is 1e-5.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a special version of the geometric mean specifically for AnaCoda.
Most models in Anacoda assume a log normal distribution for phi values, thus all values in <code>x</code> are expectd to be positive.
geomMean returns the geometric mean of a vector and can handle 0, negative, or NA values.
</p>


<h3>Value</h3>

<p>Returns the geometric mean of a vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(1, 2, 3, 4)
geomMean(x)

y&lt;- c(1, NA, 3, 4, 0, -1)
# Only take the mean of non-Na values greater than 0
geomMean(y)

# Replace values &lt;= 0 or NAs with a default value 0.001 and then take the mean
geomMean(y, rm.invalid = FALSE, default = 0.001)

</code></pre>

<hr>
<h2 id='getAdaptiveWidth'>getAdaptiveWidth</h2><span id='topic+getAdaptiveWidth'></span>

<h3>Description</h3>

<p>Return sample adaptiveWidth value, which is the number of samples (not iterations) between adapting parameter proposal widths
</p>


<h3>Value</h3>

<p>number of sample steps between adapting proposal widths
</p>

<hr>
<h2 id='getCAI'>Calculate the Codon Adaptation Index</h2><span id='topic+getCAI'></span>

<h3>Description</h3>

<p><code>getCAI</code> returns the Codon Adaptation Index for a 
genome based on a provided reference.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCAI(referenceGenome, testGenome, default.weight = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCAI_+3A_referencegenome">referenceGenome</code></td>
<td>
<p>A genome object initialized with <code><a href="#topic+initializeGenomeObject">initializeGenomeObject</a></code>.
Serves as reference set to calculate the necessary codon weights.</p>
</td></tr>
<tr><td><code id="getCAI_+3A_testgenome">testGenome</code></td>
<td>
<p>A genome object initialized with <code><a href="#topic+initializeGenomeObject">initializeGenomeObject</a></code>.
The genome for which the CAI is supposed to be calculated</p>
</td></tr>
<tr><td><code id="getCAI_+3A_default.weight">default.weight</code></td>
<td>
<p>Default weight to use if codon is missing from referenceGenome</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named vector with the CAI for each gene
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
genome_file1 &lt;- system.file("extdata", "more_genes.fasta", package = "AnaCoDa")
genome_file2 &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")

## reading genome
referenceGenome &lt;- initializeGenomeObject(file = genome_file1)
testGenome &lt;- initializeGenomeObject(file = genome_file2)

cai &lt;- getCAI(referenceGenome, testGenome)

</code></pre>

<hr>
<h2 id='getCAIweights'>Calculate the CAI codon weigths for a reference genome</h2><span id='topic+getCAIweights'></span>

<h3>Description</h3>

<p><code>getCAIweights</code> returns the weights for the Codon Adaptation Index
based on a reference genome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCAIweights(referenceGenome, default.weight = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCAIweights_+3A_referencegenome">referenceGenome</code></td>
<td>
<p>A genome object initialized with <code><a href="#topic+initializeGenomeObject">initializeGenomeObject</a></code>.</p>
</td></tr>
<tr><td><code id="getCAIweights_+3A_default.weight">default.weight</code></td>
<td>
<p>Set default weight for any codon not observed in the reference genome</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named vector with the CAI weights for each codon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")

## reading genome
referenceGenome &lt;- initializeGenomeObject(file = genome_file)

wi &lt;- getCAIweights(referenceGenome)

</code></pre>

<hr>
<h2 id='getCodonCounts'>Get Codon Counts For all Amino Acids</h2><span id='topic+getCodonCounts'></span>

<h3>Description</h3>

<p>provides the codon counts for a fiven amino acid across all genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCodonCounts(genome)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCodonCounts_+3A_genome">genome</code></td>
<td>
<p>A genome object from which the counts of each
codon can be obtained.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The returned matrix containes a row for each gene and a column 
for each synonymous codon of <code>aa</code>.
</p>


<h3>Value</h3>

<p>Returns a data.frame storing the codon counts for each amino acid.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")
 
## reading genome
genome &lt;- initializeGenomeObject(file = genome_file)
counts &lt;- getCodonCounts(genome)

</code></pre>

<hr>
<h2 id='getCodonCountsForAA'>Get Codon Counts For a specific Amino Acid</h2><span id='topic+getCodonCountsForAA'></span>

<h3>Description</h3>

<p>provides the codon counts for a fiven amino acid across all genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCodonCountsForAA(aa, genome)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCodonCountsForAA_+3A_aa">aa</code></td>
<td>
<p>One letter code of the amino acid for which the codon counts should be returned</p>
</td></tr>
<tr><td><code id="getCodonCountsForAA_+3A_genome">genome</code></td>
<td>
<p>A genome object from which the counts of each
codon can be obtained.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The returned matrix containes a row for each gene and a coloumn 
for each synonymous codon of <code>aa</code>.
</p>


<h3>Value</h3>

<p>Returns a data.frame storing the codon counts for the specified amino acid.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")
 
## reading genome
genome &lt;- initializeGenomeObject(file = genome_file)
counts &lt;- getCodonCountsForAA("A", genome)

</code></pre>

<hr>
<h2 id='getCodonSpecificPosteriorMeanForCodon'>getCodonSpecificPosteriorMeanForCodon</h2><span id='topic+getCodonSpecificPosteriorMeanForCodon'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Calculate codon-specific parameter (CSP) posterior mean
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCodonSpecificPosteriorMeanForCodon_+3A_mixtureelement">mixtureElement</code></td>
<td>
<p>mixture to calculate CSP posterior mean. Should be between 1 and n, where n is number of mixtures.</p>
</td></tr>
<tr><td><code id="getCodonSpecificPosteriorMeanForCodon_+3A_samples">samples</code></td>
<td>
<p>number of samples to use for calculating posterior mean</p>
</td></tr>
<tr><td><code id="getCodonSpecificPosteriorMeanForCodon_+3A_codon">codon</code></td>
<td>
<p>codon to calculate CSP</p>
</td></tr>
<tr><td><code id="getCodonSpecificPosteriorMeanForCodon_+3A_paramtype">paramType</code></td>
<td>
<p>CSP to calculate posterior mean for. 0: Mutation (ROC,FONSE) or Alpha (PA, PANSE). 1: Selection (ROC,FONSE), Lambda (PANSE), Lambda^prime (PA). 2: NSERate (PANSE)</p>
</td></tr>
<tr><td><code id="getCodonSpecificPosteriorMeanForCodon_+3A_withoutreference">withoutReference</code></td>
<td>
<p>If model uses reference codon, then ignore this codon (fixed at 0). Should be TRUE for ROC and FONSE. Should be FALSE for PA and PANSE.</p>
</td></tr>
<tr><td><code id="getCodonSpecificPosteriorMeanForCodon_+3A_log_scale">log_scale</code></td>
<td>
<p>If true, calculate posterior mean on log scale. Should only be used for PA and PANSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>posterior mean value for CSP
</p>

<hr>
<h2 id='getCodonSpecificPosteriorVarianceForCodon'>getCodonSpecificPosteriorVarianceForCodon</h2><span id='topic+getCodonSpecificPosteriorVarianceForCodon'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Calculate codon-specific parameter (CSP) variance
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCodonSpecificPosteriorVarianceForCodon_+3A_mixtureelement">mixtureElement</code></td>
<td>
<p>mixture to calculate CSP variance. Should be between 1 and n, where n is number of mixtures.</p>
</td></tr>
<tr><td><code id="getCodonSpecificPosteriorVarianceForCodon_+3A_samples">samples</code></td>
<td>
<p>number of samples to use for calculating variance</p>
</td></tr>
<tr><td><code id="getCodonSpecificPosteriorVarianceForCodon_+3A_codon">codon</code></td>
<td>
<p>codon to calculate CSP</p>
</td></tr>
<tr><td><code id="getCodonSpecificPosteriorVarianceForCodon_+3A_paramtype">paramType</code></td>
<td>
<p>CSP to calculate variance for. 0: Mutation (ROC,FONSE) or Alpha (PA, PANSE). 1: Selection (ROC,FONSE), Lambda (PANSE), Lambda^prime (PA). 2: NSERate (PANSE)</p>
</td></tr>
<tr><td><code id="getCodonSpecificPosteriorVarianceForCodon_+3A_unbiased">unbiased</code></td>
<td>
<p>If TRUE, should calculate variance using unbiased (N-1). Otherwise, used biased (N) correction</p>
</td></tr>
<tr><td><code id="getCodonSpecificPosteriorVarianceForCodon_+3A_withoutreference">withoutReference</code></td>
<td>
<p>If model uses reference codon, then ignore this codon (fixed at 0). Should be TRUE for ROC and FONSE. Should be FALSE for PA and PANSE.</p>
</td></tr>
<tr><td><code id="getCodonSpecificPosteriorVarianceForCodon_+3A_log_scale">log_scale</code></td>
<td>
<p>If true, calculate posterior mean on log scale. Should only be used for PA and PANSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>variance over trace for CSP
</p>

<hr>
<h2 id='getCodonSpecificQuantilesForCodon'>getCodonSpecificQuantilesForCodon</h2><span id='topic+getCodonSpecificQuantilesForCodon'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Calculate quantiles of CSP traces
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCodonSpecificQuantilesForCodon_+3A_mixtureelement">mixtureElement</code></td>
<td>
<p>mixture to calculate CSP variance. Should be between 1 and n, where n is number of mixtures.</p>
</td></tr>
<tr><td><code id="getCodonSpecificQuantilesForCodon_+3A_samples">samples</code></td>
<td>
<p>number of samples to use for calculating variance</p>
</td></tr>
<tr><td><code id="getCodonSpecificQuantilesForCodon_+3A_codon">codon</code></td>
<td>
<p>codon to calculate CSP</p>
</td></tr>
<tr><td><code id="getCodonSpecificQuantilesForCodon_+3A_paramtype">paramType</code></td>
<td>
<p>CSP to calculate variance for. 0: Mutation (ROC,FONSE) or Alpha (PA, PANSE). 1: Selection (ROC,FONSE), Lambda (PANSE), Lambda^prime (PA). 2: NSERate (PANSE)</p>
</td></tr>
<tr><td><code id="getCodonSpecificQuantilesForCodon_+3A_probs">probs</code></td>
<td>
<p>vector of two doubles between 0 and 1 indicating range over which to calculate quantiles. &lt;0.0275, 0.975&gt; would give 95% quantiles.</p>
</td></tr>
<tr><td><code id="getCodonSpecificQuantilesForCodon_+3A_withoutreference">withoutReference</code></td>
<td>
<p>If model uses reference codon, then ignore this codon (fixed at 0). Should be TRUE for ROC and FONSE. Should be FALSE for PA and PANSE.</p>
</td></tr>
<tr><td><code id="getCodonSpecificQuantilesForCodon_+3A_log_scale">log_scale</code></td>
<td>
<p>If true, calculate posterior mean on log scale. Should only be used for PA and PANSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector representing lower and upper bound of quantile
</p>

<hr>
<h2 id='getCSPEstimates'>Return Codon Specific Paramters (or write to csv) estimates as data.frame</h2><span id='topic+getCSPEstimates'></span>

<h3>Description</h3>

<p><code>getCSPEstimates</code> returns the codon specific
parameter estimates for a given parameter and mixture or write it to a csv file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCSPEstimates(
  parameter,
  filename = NULL,
  mixture = 1,
  samples = 10,
  relative.to.optimal.codon = T,
  report.original.ref = T,
  log.scale = F
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCSPEstimates_+3A_parameter">parameter</code></td>
<td>
<p>parameter an object created by <code>initializeParameterObject</code>.</p>
</td></tr>
<tr><td><code id="getCSPEstimates_+3A_filename">filename</code></td>
<td>
<p>Posterior estimates will be written to file (format: csv). Filename will be in the format &lt;parameter_name&gt;_&lt;filename&gt;.csv.</p>
</td></tr>
<tr><td><code id="getCSPEstimates_+3A_mixture">mixture</code></td>
<td>
<p>estimates for which mixture should be returned</p>
</td></tr>
<tr><td><code id="getCSPEstimates_+3A_samples">samples</code></td>
<td>
<p>The number of samples used for the posterior estimates.</p>
</td></tr>
<tr><td><code id="getCSPEstimates_+3A_relative.to.optimal.codon">relative.to.optimal.codon</code></td>
<td>
<p>Boolean determining if parameters should be relative to the preferred codon or the alphabetically last codon (Default=TRUE). Only applies to ROC and FONSE models</p>
</td></tr>
<tr><td><code id="getCSPEstimates_+3A_report.original.ref">report.original.ref</code></td>
<td>
<p>Include the original reference codon (Default = TRUE). Note this is only included for the purposes of simulations, which expect the input parameter file to be in a specific format. Later version of AnaCoDa will remove this.</p>
</td></tr>
<tr><td><code id="getCSPEstimates_+3A_log.scale">log.scale</code></td>
<td>
<p>Calculate posterior means, standard deviation, and posterior probability intervals on the natural log scale. Should be used for PA and PANSE models only.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a list data.frame with the posterior estimates of the models 
codon specific parameters or writes it directly to a csv file if <code>filename</code> is specified
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")

genome &lt;- initializeGenomeObject(file = genome_file)
sphi_init &lt;- c(1,1)
numMixtures &lt;- 2
geneAssignment &lt;- c(rep(1,floor(length(genome)/2)),rep(2,ceiling(length(genome)/2)))
parameter &lt;- initializeParameterObject(genome = genome, sphi = sphi_init, 
                                       num.mixtures = numMixtures, 
                                       gene.assignment = geneAssignment, 
                                       mixture.definition = "allUnique")
model &lt;- initializeModelObject(parameter = parameter, model = "ROC")
samples &lt;- 2500
thinning &lt;- 50
adaptiveWidth &lt;- 25
mcmc &lt;- initializeMCMCObject(samples = samples, thinning = thinning, 
                             adaptive.width=adaptiveWidth, est.expression=TRUE, 
                             est.csp=TRUE, est.hyper=TRUE, est.mix = TRUE) 
divergence.iteration &lt;- 10
## Not run: 
runMCMC(mcmc = mcmc, genome = genome, model = model, 
        ncores = 4, divergence.iteration = divergence.iteration)

## return estimates for codon specific parameters
csp_mat &lt;- getCSPEstimates(parameter)

# write the result directly to the filesystem as a csv file. No values are returned
getCSPEstimates(parameter, filename=file.path(tempdir(), "test.csv"))


## End(Not run)
</code></pre>

<hr>
<h2 id='getEstimatedMixtureAssignmentForGene'>getEstimatedMixtureAssignmentForGene</h2><span id='topic+getEstimatedMixtureAssignmentForGene'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Get estimated mixture assignment for gene
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="getEstimatedMixtureAssignmentForGene_+3A_samples">samples</code></td>
<td>
<p>number of samples over which to calculate mixture assignment</p>
</td></tr>
<tr><td><code id="getEstimatedMixtureAssignmentForGene_+3A_geneindex">geneIndex</code></td>
<td>
<p>corresponding index of gene in genome. Should be a number between 1 and length(genome).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns value between 1 and n, where n is number of mixtures
</p>

<hr>
<h2 id='getEstimatedMixtureAssignmentProbabilitiesForGene'>getEstimatedMixtureAssignmentProbabilitiesForGene</h2><span id='topic+getEstimatedMixtureAssignmentProbabilitiesForGene'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Get estimated mixture assignment probabilities for gene
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="getEstimatedMixtureAssignmentProbabilitiesForGene_+3A_samples">samples</code></td>
<td>
<p>number of samples over which to calculate mixture assignment probabilities</p>
</td></tr>
<tr><td><code id="getEstimatedMixtureAssignmentProbabilitiesForGene_+3A_geneindex">geneIndex</code></td>
<td>
<p>corresponding index of gene in genome. Should be a number between 1 and length(genome).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns vector of probabilities representing mixture probabilities for gene
</p>

<hr>
<h2 id='getExpressionEstimates'>Returns the estimated phi posterior for a gene</h2><span id='topic+getExpressionEstimates'></span>

<h3>Description</h3>

<p>Posterior estimates for the phi value of specified genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getExpressionEstimates(
  parameter,
  gene.index,
  samples,
  quantiles = c(0.025, 0.975),
  genome = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getExpressionEstimates_+3A_parameter">parameter</code></td>
<td>
<p>on object created by <code>initializeParameterObject</code>.</p>
</td></tr>
<tr><td><code id="getExpressionEstimates_+3A_gene.index">gene.index</code></td>
<td>
<p>a integer or vector of integers representing the gene(s) of interesst.</p>
</td></tr>
<tr><td><code id="getExpressionEstimates_+3A_samples">samples</code></td>
<td>
<p>number of samples for the posterior estimate</p>
</td></tr>
<tr><td><code id="getExpressionEstimates_+3A_quantiles">quantiles</code></td>
<td>
<p>vector of quantiles, (default: c(0.025, 0.975))</p>
</td></tr>
<tr><td><code id="getExpressionEstimates_+3A_genome">genome</code></td>
<td>
<p>if genome is given, then will include gene ids in output (default is NULL)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The returned vector is unnamed as gene ids are only stored in the <code>genome</code> object, 
but the <code>gene.index</code> vector can be used to match the assignment to the genome.
</p>


<h3>Value</h3>

<p>returns a vector with the mixture assignment of each gene corresbonding to <code>gene.index</code> in the same order as the genome.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")

genome &lt;- initializeGenomeObject(file = genome_file)
sphi_init &lt;- c(1,1)
numMixtures &lt;- 2
geneAssignment &lt;- c(rep(1,floor(length(genome)/2)),rep(2,ceiling(length(genome)/2)))
parameter &lt;- initializeParameterObject(genome = genome, sphi = sphi_init, 
                                       num.mixtures = numMixtures, 
                                       gene.assignment = geneAssignment, 
                                       mixture.definition = "allUnique")
model &lt;- initializeModelObject(parameter = parameter, model = "ROC")
samples &lt;- 2500
thinning &lt;- 50
adaptiveWidth &lt;- 25
mcmc &lt;- initializeMCMCObject(samples = samples, thinning = thinning, 
                             adaptive.width=adaptiveWidth, est.expression=TRUE,
                             est.csp=TRUE, est.hyper=TRUE, est.mix = TRUE) 
divergence.iteration &lt;- 10
## Not run: 
runMCMC(mcmc = mcmc, genome = genome, model = model, 
        ncores = 4, divergence.iteration = divergence.iteration)

# get the estimated expression values for all genes based on the mixture 
# they are assigned to at each step
estimatedExpression &lt;- getExpressionEstimates(parameter, 1:length(genome), 1000)

## End(Not run)

</code></pre>

<hr>
<h2 id='getGroupList'>getGroupList</h2><span id='topic+getGroupList'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Get amino acids (ROC, FONSE) or codons (PA, PANSE) for which parameters will be estimated
</p>


<h3>Value</h3>

<p>returns list of amino acids or codons
</p>

<hr>
<h2 id='getLogLikelihoodTrace'>getLogLikelihoodTrace</h2><span id='topic+getLogLikelihoodTrace'></span>

<h3>Description</h3>

<p>Method of MCMC class (access via mcmc$&lt;function name&gt;, where mcmc is an object initialized by initializeMCMCObject). Returns the logLikelihood trace
</p>


<h3>Value</h3>

<p>vector representing logLikelihood trace
</p>

<hr>
<h2 id='getLogPosteriorMean'>getLogPosteriorMean</h2><span id='topic+getLogPosteriorMean'></span>

<h3>Description</h3>

<p>Method of MCMC class (access via mcmc$&lt;function name&gt;, where mcmc is an object initialized by initializeMCMCObject). Calculate the mean log posterior probability over the last n samples
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="getLogPosteriorMean_+3A_samples">samples</code></td>
<td>
<p>postive value less than total length of the MCMC trace</p>
</td></tr>
</table>


<h3>Value</h3>

<p>mean logPosterior
</p>

<hr>
<h2 id='getLogPosteriorTrace'>getLogPosteriorTrace</h2><span id='topic+getLogPosteriorTrace'></span>

<h3>Description</h3>

<p>Method of MCMC class (access via mcmc$&lt;function name&gt;, where mcmc is an object initialized by initializeMCMCObject). Returns the logPosterior trace
</p>


<h3>Value</h3>

<p>vector representing logPosterior trace
</p>

<hr>
<h2 id='getMixtureAssignmentEstimate'>Returns mixture assignment estimates for each gene</h2><span id='topic+getMixtureAssignmentEstimate'></span>

<h3>Description</h3>

<p>Posterior estimates for the mixture assignment of specified genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMixtureAssignmentEstimate(parameter, gene.index, samples)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getMixtureAssignmentEstimate_+3A_parameter">parameter</code></td>
<td>
<p>on object created by <code>initializeParameterObject</code></p>
</td></tr>
<tr><td><code id="getMixtureAssignmentEstimate_+3A_gene.index">gene.index</code></td>
<td>
<p>a integer or vector of integers representing the gene(s) of interesst.</p>
</td></tr>
<tr><td><code id="getMixtureAssignmentEstimate_+3A_samples">samples</code></td>
<td>
<p>number of samples for the posterior estimate</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The returned vector is unnamed as gene ids are only stored in the <code>genome</code> object, 
but the <code>gene.index</code> vector can be used to match the assignment to the genome.
</p>


<h3>Value</h3>

<p>returns a vector with the mixture assignment of each gene corresbonding to <code>gene.index</code> in the same order as the genome.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")

genome &lt;- initializeGenomeObject(file = genome_file)
sphi_init &lt;- c(1,1)
numMixtures &lt;- 2
geneAssignment &lt;- c(rep(1,floor(length(genome)/2)),rep(2,ceiling(length(genome)/2)))
parameter &lt;- initializeParameterObject(genome = genome, sphi = sphi_init, 
                                       num.mixtures = numMixtures, 
                                       gene.assignment = geneAssignment, 
                                       mixture.definition = "allUnique")
model &lt;- initializeModelObject(parameter = parameter, model = "ROC")
samples &lt;- 2500
thinning &lt;- 50
adaptiveWidth &lt;- 25
mcmc &lt;- initializeMCMCObject(samples = samples, thinning = thinning, adaptive.width=adaptiveWidth, 
                             est.expression=TRUE, est.csp=TRUE, est.hyper=TRUE, est.mix = TRUE) 
divergence.iteration &lt;- 10
## Not run: 
runMCMC(mcmc = mcmc, genome = genome, model = model, 
        ncores = 4, divergence.iteration = divergence.iteration)

# get the mixture assignment for all genes
mixAssign &lt;- getMixtureAssignmentEstimate(parameter = parameter, 
                                          gene.index = 1:length(genome), samples = 1000)

# get the mixture assignment for a subsample
mixAssign &lt;- getMixtureAssignmentEstimate(parameter = parameter, 
                                          gene.index = 5:100, samples = 1000)
# or
mixAssign &lt;- getMixtureAssignmentEstimate(parameter = parameter, 
                                          gene.index = c(10, 30:50, 3, 90), samples = 1000)

## End(Not run)

</code></pre>

<hr>
<h2 id='getNames'>Gene Names of Genome</h2><span id='topic+getNames'></span>

<h3>Description</h3>

<p>returns the identifiers of the genes within the genome specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getNames(genome, simulated = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getNames_+3A_genome">genome</code></td>
<td>
<p>A genome object initialized with <code><a href="#topic+initializeGenomeObject">initializeGenomeObject</a></code>.</p>
</td></tr>
<tr><td><code id="getNames_+3A_simulated">simulated</code></td>
<td>
<p>A logical value denoting if the gene names to be listed are simulated or not.
The default value is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>gene.names Returns the names of the genes as a vector of strings.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")
 
## reading genome
genome &lt;- initializeGenomeObject(file = genome_file)

## return all gene ids for the genome
geneIDs &lt;- getNames(genome, FALSE)

</code></pre>

<hr>
<h2 id='getNc'>Calculate the Effective Number of Codons</h2><span id='topic+getNc'></span>

<h3>Description</h3>

<p><code>getNc</code> returns the Effective Number of Codons for a genome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getNc(genome)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getNc_+3A_genome">genome</code></td>
<td>
<p>A genome object initialized with <code><a href="#topic+initializeGenomeObject">initializeGenomeObject</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named vector with the Effective Number of Codons for each gene
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
genome_file &lt;- system.file("extdata", "more_genes.fasta", package = "AnaCoDa")
## reading genome
genome &lt;- initializeGenomeObject(file = genome_file)

nc &lt;- getNc(genome)

</code></pre>

<hr>
<h2 id='getNcAA'>Calculate the Effective Number of Codons for each Amino Acid</h2><span id='topic+getNcAA'></span>

<h3>Description</h3>

<p><code>getNcAA</code> returns the Effective Number of Codons for each Amino Acid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getNcAA(genome)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getNcAA_+3A_genome">genome</code></td>
<td>
<p>A genome object initialized with <code><a href="#topic+initializeGenomeObject">initializeGenomeObject</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of type <code>data.frame</code> with the Effective Number of Codons
for each amino acid in each gene.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
genome_file &lt;- system.file("extdata", "more_genes.fasta", package = "AnaCoDa")
## reading genome
genome &lt;- initializeGenomeObject(file = genome_file)

nc &lt;- getNcAA(genome)

</code></pre>

<hr>
<h2 id='getNoiseOffsetPosteriorMean'>getNoiseOffsetPosteriorMean</h2><span id='topic+getNoiseOffsetPosteriorMean'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Calculate posterior mean of standard deviation parameter of lognormal describing distribution of synthesis rates
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="getNoiseOffsetPosteriorMean_+3A_index">index</code></td>
<td>
<p>mixture index to use. Should be number between 0 and n-1, where n is number of mixtures</p>
</td></tr>
<tr><td><code id="getNoiseOffsetPosteriorMean_+3A_samples">samples</code></td>
<td>
<p>number of samples over which to calculate posterior mean</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns posterior mean for standard deviation of lognormal distribution of synthesis rates
</p>

<hr>
<h2 id='getNoiseOffsetVariance'>getNoiseOffsetVariance</h2><span id='topic+getNoiseOffsetVariance'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Calculate variance of noise offset parameter used when fitting model with empirical estimates of synthesis rates (ie. withPhi fits)
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="getNoiseOffsetVariance_+3A_index">index</code></td>
<td>
<p>mixture index to use. Should be number between 0 and n-1, where n is number of mixtures</p>
</td></tr>
<tr><td><code id="getNoiseOffsetVariance_+3A_samples">samples</code></td>
<td>
<p>number of samples over which to calculate variance</p>
</td></tr>
<tr><td><code id="getNoiseOffsetVariance_+3A_unbiased">unbiased</code></td>
<td>
<p>If TRUE, should calculate variance using unbiased (N-1). Otherwise, used biased (N) correction</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns variance for noise offset
</p>

<hr>
<h2 id='getObservedSynthesisRateSet'>Get gene observed synthesis rates</h2><span id='topic+getObservedSynthesisRateSet'></span>

<h3>Description</h3>

<p><code>getObservedSynthesisRateSet</code> returns the observed 
synthesis rates of the genes within the genome specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getObservedSynthesisRateSet(genome, simulated = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getObservedSynthesisRateSet_+3A_genome">genome</code></td>
<td>
<p>A genome object initialized with <code><a href="#topic+initializeGenomeObject">initializeGenomeObject</a></code>.</p>
</td></tr>
<tr><td><code id="getObservedSynthesisRateSet_+3A_simulated">simulated</code></td>
<td>
<p>A logical value denoting if the synthesis 
rates to be listed are simulated or not. The default value is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data.frame with the observed expression values in genome
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")
expression_file &lt;- system.file("extdata", "expression.csv", package = "AnaCoDa") 
## reading genome
genome &lt;- initializeGenomeObject(file = genome_file)


## return expression values as a data.frame with gene ids in the first column.
expressionValues &lt;- getObservedSynthesisRateSet(genome = genome)

</code></pre>

<hr>
<h2 id='getSamples'>getSamples</h2><span id='topic+getSamples'></span>

<h3>Description</h3>

<p>Method of MCMC class (access via mcmc$&lt;function name&gt;, where mcmc is an object initialized by initializeMCMCObject). Return number of samples set for MCMCAlgorithm object
</p>


<h3>Value</h3>

<p>number of samples used during MCMC
</p>

<hr>
<h2 id='getSelectionCoefficients'>Calculate Selection coefficients</h2><span id='topic+getSelectionCoefficients'></span>

<h3>Description</h3>

<p><code>getSelectionCoefficients</code> calculates the selection coefficient of each codon in each gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSelectionCoefficients(genome, parameter, samples = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSelectionCoefficients_+3A_genome">genome</code></td>
<td>
<p>A genome object initialized with 
<code><a href="#topic+initializeGenomeObject">initializeGenomeObject</a></code> to add observed expression data.</p>
</td></tr>
<tr><td><code id="getSelectionCoefficients_+3A_parameter">parameter</code></td>
<td>
<p>an object created by <code>initializeParameterObject</code>.</p>
</td></tr>
<tr><td><code id="getSelectionCoefficients_+3A_samples">samples</code></td>
<td>
<p>The number of samples used for the posterior estimates.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with selection coefficients.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")

genome &lt;- initializeGenomeObject(file = genome_file)
sphi_init &lt;- 1
numMixtures &lt;- 1
geneAssignment &lt;- rep(1, length(genome))
parameter &lt;- initializeParameterObject(genome = genome, sphi = sphi_init, 
                                       num.mixtures = numMixtures, 
                                       gene.assignment = geneAssignment, 
                                       mixture.definition = "allUnique")
model &lt;- initializeModelObject(parameter = parameter, model = "ROC")
samples &lt;- 2500
thinning &lt;- 50
adaptiveWidth &lt;- 25
mcmc &lt;- initializeMCMCObject(samples = samples, thinning = thinning, 
                             adaptive.width=adaptiveWidth, est.expression=TRUE, 
                             est.csp=TRUE, est.hyper=TRUE, est.mix = TRUE) 
divergence.iteration &lt;- 10
## Not run: 
runMCMC(mcmc = mcmc, genome = genome, model = model, 
        ncores = 4, divergence.iteration = divergence.iteration)

## return estimates for selection coefficients s for each codon in each gene
selection.coefficients &lt;- getSelectionCoefficients(genome = genome, 
                                                   parameter = parameter, samples = 1000)

## End(Not run)

</code></pre>

<hr>
<h2 id='getStdDevSynthesisRatePosteriorMean'>getStdDevSynthesisRatePosteriorMean</h2><span id='topic+getStdDevSynthesisRatePosteriorMean'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Calculate posterior mean of standard deviation parameter of lognormal describing distribution of synthesis rates
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="getStdDevSynthesisRatePosteriorMean_+3A_samples">samples</code></td>
<td>
<p>number of samples over which to calculate posterior mean</p>
</td></tr>
<tr><td><code id="getStdDevSynthesisRatePosteriorMean_+3A_mixture">mixture</code></td>
<td>
<p>mixture index to use. Should be number between 0 and n-1, where n is number of mixtures</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns posterior mean for standard deviation of lognormal distribution of synthesis rates
</p>

<hr>
<h2 id='getStdDevSynthesisRateVariance'>getStdDevSynthesisRateVariance</h2><span id='topic+getStdDevSynthesisRateVariance'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Calculate variance of standard deviation parameter of lognormal describing distribution of synthesis rates
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="getStdDevSynthesisRateVariance_+3A_samples">samples</code></td>
<td>
<p>number of samples over which to calculate variance</p>
</td></tr>
<tr><td><code id="getStdDevSynthesisRateVariance_+3A_mixture">mixture</code></td>
<td>
<p>mixture index to use. Should be number between 0 and n-1, where n is number of mixtures</p>
</td></tr>
<tr><td><code id="getStdDevSynthesisRateVariance_+3A_unbiased">unbiased</code></td>
<td>
<p>If TRUE, should calculate variance using unbiased (N-1). Otherwise, used biased (N) correction</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns variance for standard deviation of lognormal distribution of synthesis rates
</p>

<hr>
<h2 id='getStepsToAdapt'>getStepsToAdapt</h2><span id='topic+getStepsToAdapt'></span>

<h3>Description</h3>

<p>Method of MCMC class (access via mcmc$&lt;function name&gt;, where mcmc is an object initialized by initializeMCMCObject). Return number of iterations (total iterations = samples * thinning) to allow proposal widths to adapt
</p>


<h3>Value</h3>

<p>number of sample steps to adapt
</p>

<hr>
<h2 id='getSynthesisRate'>getSynthesisRate</h2><span id='topic+getSynthesisRate'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Get current synthesis rates for all genes and all mixtures
</p>


<h3>Value</h3>

<p>2 by 2 vector of numeric values
</p>

<hr>
<h2 id='getSynthesisRatePosteriorMeanForGene'>getSynthesisRatePosteriorMeanForGene</h2><span id='topic+getSynthesisRatePosteriorMeanForGene'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Get posterior mean synthesis rate value for a gene
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSynthesisRatePosteriorMeanForGene_+3A_samples">samples</code></td>
<td>
<p>number of samples over which to calculate mean</p>
</td></tr>
<tr><td><code id="getSynthesisRatePosteriorMeanForGene_+3A_geneindex">geneIndex</code></td>
<td>
<p>corresponding index of gene in genome for which posterior mean synthesis rate will be calculated. Should be a number between 1 and length(genome)</p>
</td></tr>
<tr><td><code id="getSynthesisRatePosteriorMeanForGene_+3A_log_scale">log_scale</code></td>
<td>
<p>Calculate posterior mean on log scale</p>
</td></tr>
</table>


<h3>Value</h3>

<p>posterior mean synthesis rate for gene
</p>

<hr>
<h2 id='getSynthesisRatePosteriorVarianceForGene'>getSynthesisRatePosteriorVarianceForGene</h2><span id='topic+getSynthesisRatePosteriorVarianceForGene'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Get synthesis rate variance for a gene
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSynthesisRatePosteriorVarianceForGene_+3A_samples">samples</code></td>
<td>
<p>number of samples over which to calculate variance</p>
</td></tr>
<tr><td><code id="getSynthesisRatePosteriorVarianceForGene_+3A_geneindex">geneIndex</code></td>
<td>
<p>corresponding index of gene in genome for which synthesis rate variance will be calculated. Should be a number between 1 and length(genome)</p>
</td></tr>
<tr><td><code id="getSynthesisRatePosteriorVarianceForGene_+3A_unbiased">unbiased</code></td>
<td>
<p>Should calculate variance using unbiased (N-1) or biased (N) correction</p>
</td></tr>
<tr><td><code id="getSynthesisRatePosteriorVarianceForGene_+3A_log_scale">log_scale</code></td>
<td>
<p>Calculate variance on log scale</p>
</td></tr>
</table>


<h3>Value</h3>

<p>posterior mean synthesis rate for gene
</p>

<hr>
<h2 id='getThinning'>getThinning</h2><span id='topic+getThinning'></span>

<h3>Description</h3>

<p>Method of MCMC class (access via mcmc$&lt;function name&gt;, where mcmc is an object initialized by initializeMCMCObject). Return thinning value, which is the number of iterations (total iterations = samples * thinning) not being kept
</p>


<h3>Value</h3>

<p>thinning value used during MCMC
</p>

<hr>
<h2 id='getTrace'>extracts an object of traces from a parameter object.</h2><span id='topic+getTrace'></span>

<h3>Description</h3>

<p>extracts an object of traces from a parameter object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getTrace(parameter)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getTrace_+3A_parameter">parameter</code></td>
<td>
<p>A Parameter object that corresponds to one of the model types.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>trace Returns an object of type Trace extracted from the given parameter object
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")

genome &lt;- initializeGenomeObject(file = genome_file)
sphi_init &lt;- c(1,1)
numMixtures &lt;- 2
geneAssignment &lt;- c(rep(1,floor(length(genome)/2)),rep(2,ceiling(length(genome)/2)))
parameter &lt;- initializeParameterObject(genome = genome, sphi = sphi_init, 
                                       num.mixtures = numMixtures, 
                                       gene.assignment = geneAssignment, 
                                       mixture.definition = "allUnique")

trace &lt;- getTrace(parameter) # empty trace object since no MCMC was perfomed

</code></pre>

<hr>
<h2 id='getTraceObject'>getTraceObject</h2><span id='topic+getTraceObject'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Get Trace object stored by a Parameter object. Useful for plotting certain parameter traces.
</p>


<h3>Value</h3>

<p>Trace object
</p>

<hr>
<h2 id='initializeCovarianceMatrices'>Initialize Covariance Matrices</h2><span id='topic+initializeCovarianceMatrices'></span>

<h3>Description</h3>

<p>Initialize Covariance Matrices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initializeCovarianceMatrices(
  parameter,
  genome,
  numMixtures,
  geneAssignment,
  init.csp.variance = 0.0025
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="initializeCovarianceMatrices_+3A_parameter">parameter</code></td>
<td>
<p>A Parameter object that corresponds to one of the model types. 
Valid values are &quot;ROC&quot;, &quot;PA&quot;, and &quot;FONSE&quot;.</p>
</td></tr>
<tr><td><code id="initializeCovarianceMatrices_+3A_genome">genome</code></td>
<td>
<p>An object of type Genome necessary for the initialization of the Parameter object.</p>
</td></tr>
<tr><td><code id="initializeCovarianceMatrices_+3A_nummixtures">numMixtures</code></td>
<td>
<p>The number of mixture elements for the underlying mixture distribution (numMixtures &gt; 0).</p>
</td></tr>
<tr><td><code id="initializeCovarianceMatrices_+3A_geneassignment">geneAssignment</code></td>
<td>
<p>A vector holding the initial mixture assignment for each gene.
The vector length has to equal the number of genes in the genome.
Valid values for the vector range from 1 to numMixtures.
It is possible but not advised to leave a mixture element empty.</p>
</td></tr>
<tr><td><code id="initializeCovarianceMatrices_+3A_init.csp.variance">init.csp.variance</code></td>
<td>
<p>initial proposal variance for codon specific parameter, default is 0.0025.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>parameter Returns the Parameter argument, now modified with initialized mutation, selection, and covariance matrices.
</p>

<hr>
<h2 id='initializeGenomeObject'>Genome Initialization</h2><span id='topic+initializeGenomeObject'></span>

<h3>Description</h3>

<p><code>initializeGenomeObject</code> initializes the Rcpp Genome object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initializeGenomeObject(
  file,
  genome = NULL,
  observed.expression.file = NULL,
  fasta = TRUE,
  positional = FALSE,
  match.expression.by.id = TRUE,
  append = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="initializeGenomeObject_+3A_file">file</code></td>
<td>
<p>A file of coding sequences in fasta or RFPData format</p>
</td></tr>
<tr><td><code id="initializeGenomeObject_+3A_genome">genome</code></td>
<td>
<p>A genome object can be passed in to concatenate the input file to it (optional).</p>
</td></tr>
<tr><td><code id="initializeGenomeObject_+3A_observed.expression.file">observed.expression.file</code></td>
<td>
<p>String containing the location of a file containing
empirical expression rates (optional). Default value is NULL.</p>
</td></tr>
<tr><td><code id="initializeGenomeObject_+3A_fasta">fasta</code></td>
<td>
<p>Boolean value indicating whether <code>file</code> argument is a
fasta file (TRUE) or an RFPData file (FALSE). Default value is TRUE.</p>
</td></tr>
<tr><td><code id="initializeGenomeObject_+3A_positional">positional</code></td>
<td>
<p>Boolean indicating if the positional information in the RFPData file is necessary. Default value is FALSE</p>
</td></tr>
<tr><td><code id="initializeGenomeObject_+3A_match.expression.by.id">match.expression.by.id</code></td>
<td>
<p>If TRUE, observed expression values will be assigned by matching sequence identifier. 
If FALSE, observed expression values will be assigned by order. Default value is TRUE.</p>
</td></tr>
<tr><td><code id="initializeGenomeObject_+3A_append">append</code></td>
<td>
<p>If TRUE, function will read in additional genome data to append to an existing genome.
If FALSE, genome data is cleared before reading in data (no preexisting data). Default value is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns the initialized Genome object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")
genes_file &lt;- system.file("extdata", "more_genes.fasta", package = "AnaCoDa")
expression_file &lt;- system.file("extdata", "expression.csv", package = "AnaCoDa")
 
## reading genome
genome &lt;- initializeGenomeObject(file = genome_file)

## reading genome and observed expression data
genome &lt;- initializeGenomeObject(file = genome_file, observed.expression.file = expression_file)
 
## add aditional genes to existing genome
genome &lt;- initializeGenomeObject(file = genome_file)
genome &lt;- initializeGenomeObject(file = genes_file, genome = genome, append = TRUE)   

</code></pre>

<hr>
<h2 id='initializeMCMCObject'>Initialize MCMC</h2><span id='topic+initializeMCMCObject'></span>

<h3>Description</h3>

<p><code>initializeMCMCObject</code> initializes a MCMC object to 
perform a model fitting for a parameter and model object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initializeMCMCObject(
  samples,
  thinning = 1,
  adaptive.width = 100,
  est.expression = TRUE,
  est.csp = TRUE,
  est.hyper = TRUE,
  est.mix = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="initializeMCMCObject_+3A_samples">samples</code></td>
<td>
<p>Number of samples to be produced when running the 
MCMC algorithm. No default value.</p>
</td></tr>
<tr><td><code id="initializeMCMCObject_+3A_thinning">thinning</code></td>
<td>
<p>The thinning interval between consecutive observations. If set to 
1, every step will be saved as a sample. Default value is 1.</p>
</td></tr>
<tr><td><code id="initializeMCMCObject_+3A_adaptive.width">adaptive.width</code></td>
<td>
<p>Number that determines how often the acceptance/rejection
window should be altered. Default value is 100 samples.
Proportion of MCMC steps where the proposal distribution is adaptive can be set using <code>mcmc$setStepsToAdapt</code>. The default parameter passed in as -1 uses the full iterations.</p>
</td></tr>
<tr><td><code id="initializeMCMCObject_+3A_est.expression">est.expression</code></td>
<td>
<p>Boolean that tells whether or not synthesis rate values
should be estimated in the MCMC algorithm run. Default value is TRUE.</p>
</td></tr>
<tr><td><code id="initializeMCMCObject_+3A_est.csp">est.csp</code></td>
<td>
<p>Boolean that tells whether or not codon specific values
should be estimated in the MCMC algorithm run. Default value is TRUE.</p>
</td></tr>
<tr><td><code id="initializeMCMCObject_+3A_est.hyper">est.hyper</code></td>
<td>
<p>Boolean that tells whether or not hyper parameters
should be estimated in the MCMC algorithm run. Default value is TRUE.
Setting for expression noise parameter sepsilon can be overridden by setting <code>fix.observation.noise</code> in <code>initializeModelObject()</code></p>
</td></tr>
<tr><td><code id="initializeMCMCObject_+3A_est.mix">est.mix</code></td>
<td>
<p>Boolean that tells whether or not the genes' mixture element
should be estimated in the MCMC algorithm run. Default value is TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>initializeMCMCObject</code> sets up the MCMC object 
(monte carlo markov chain) and returns the object so a model fitting can be done.
It is important to note that est.expression and est.hyper will affect one another
negatively if their values differ.
</p>


<h3>Value</h3>

<p>mcmc Returns an intialized MCMC object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## initializing an object of type mcmc

samples &lt;- 2500
thinning &lt;- 50
adaptiveWidth &lt;- 25

## estimate all parameter types
mcmc &lt;- initializeMCMCObject(samples = samples, thinning = thinning, adaptive.width=adaptiveWidth, 
                             est.expression=TRUE, est.csp=TRUE, est.hyper=TRUE, est.mix = TRUE) 
                             
## do not estimate expression values, initial conditions will remain constant
mcmc &lt;- initializeMCMCObject(samples = samples, thinning = thinning, adaptive.width=adaptiveWidth, 
                             est.expression=FALSE, est.csp=TRUE, est.hyper=TRUE, est.mix = TRUE) 
                             
## do not estimate hyper parameters, initial conditions will remain constant
mcmc &lt;- initializeMCMCObject(samples = samples, thinning = thinning, adaptive.width=adaptiveWidth, 
                             est.expression=TRUE, est.csp=TRUE, est.hyper=FALSE, est.mix = TRUE) 

</code></pre>

<hr>
<h2 id='initializeModelObject'>Model Initialization</h2><span id='topic+initializeModelObject'></span>

<h3>Description</h3>

<p>initializes the model object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initializeModelObject(
  parameter,
  model = "ROC",
  with.phi = FALSE,
  fix.observation.noise = FALSE,
  rfp.count.column = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="initializeModelObject_+3A_parameter">parameter</code></td>
<td>
<p>An object created with <code>initializeParameterObject</code>.</p>
</td></tr>
<tr><td><code id="initializeModelObject_+3A_model">model</code></td>
<td>
<p>A string containing the model to run (ROC, FONSE, or PA), has to match parameter object.</p>
</td></tr>
<tr><td><code id="initializeModelObject_+3A_with.phi">with.phi</code></td>
<td>
<p>(ROC only) A boolean that determines whether or not to include empirical
phi values (expression rates) for the calculations. Default value is FALSE</p>
</td></tr>
<tr><td><code id="initializeModelObject_+3A_fix.observation.noise">fix.observation.noise</code></td>
<td>
<p>(ROC only) Allows fixing the noise term sepsilon in the observed expression dataset to its initial condition.  This value should override the est.hyper=TRUE setting in <code>initializeMCMCObject()</code>
The initial condition for the observed expression noise is set in the parameter object. Default value is FALSE.</p>
</td></tr>
<tr><td><code id="initializeModelObject_+3A_rfp.count.column">rfp.count.column</code></td>
<td>
<p>(PA and PANSE only) A number representing the RFP count column to use. Default value is 1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>initializeModelObject initializes a model. The type of model is determined based on the string passed to the <code>model</code> argument.
The Parameter object has to match the model that is initialized. E.g. to initialize a ROC model, 
it is required that a ROC parameter object is passed to the function.
</p>


<h3>Value</h3>

<p>This function returns the model object created.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#initializing a model object

genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")
expression_file &lt;- system.file("extdata", "expression.csv", package = "AnaCoDa")

genome &lt;- initializeGenomeObject(file = genome_file, 
                                 observed.expression.file = expression_file)
sphi_init &lt;- c(1,1)
numMixtures &lt;- 2
geneAssignment &lt;- c(rep(1,floor(length(genome)/2)),rep(2,ceiling(length(genome)/2)))
parameter &lt;- initializeParameterObject(genome = genome, sphi = sphi_init, 
                                       num.mixtures = numMixtures, 
                                       gene.assignment = geneAssignment, 
                                       mixture.definition = "allUnique")

# initializing a model object assuming we have observed expression (phi) 
# values stored in the genome object.
initializeModelObject(parameter = parameter, model = "ROC", with.phi = TRUE)

# initializing a model object ignoring observed expression (phi) 
# values stored in the genome object.
initializeModelObject(parameter = parameter, model = "ROC", with.phi = FALSE)

</code></pre>

<hr>
<h2 id='initializeParameterObject'>Initialize Parameter</h2><span id='topic+initializeParameterObject'></span>

<h3>Description</h3>

<p><code>initializeParameterObject</code> initializes a new parameter object or reconstructs one from a restart file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initializeParameterObject(
  genome = NULL,
  sphi = NULL,
  num.mixtures = 1,
  gene.assignment = NULL,
  initial.expression.values = NULL,
  model = "ROC",
  split.serine = TRUE,
  mixture.definition = "allUnique",
  mixture.definition.matrix = NULL,
  init.with.restart.file = NULL,
  mutation.prior.mean = 0,
  mutation.prior.sd = 0.35,
  propose.by.prior = FALSE,
  init.csp.variance = 0.0025,
  init.sepsilon = 0.1,
  init.w.obs.phi = FALSE,
  init.initiation.cost = 4,
  init.partition.function = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="initializeParameterObject_+3A_genome">genome</code></td>
<td>
<p>An object of type Genome necessary for the initialization of the Parameter object.
The default value is NULL.</p>
</td></tr>
<tr><td><code id="initializeParameterObject_+3A_sphi">sphi</code></td>
<td>
<p>Initial values for sphi. Expected is a vector of length numMixtures.
The default value is NULL.</p>
</td></tr>
<tr><td><code id="initializeParameterObject_+3A_num.mixtures">num.mixtures</code></td>
<td>
<p>The number of mixtures elements for the underlying mixture distribution (numMixtures &gt; 0).
The default value is 1.</p>
</td></tr>
<tr><td><code id="initializeParameterObject_+3A_gene.assignment">gene.assignment</code></td>
<td>
<p>A vector holding the initial mixture assignment for each gene. 
The vector length has to equal the number of genes in the genome.
Valid values for the vector range from 1 to numMixtures. 
It is possible but not advised to leave a mixture element empty.
The default Value is NULL.</p>
</td></tr>
<tr><td><code id="initializeParameterObject_+3A_initial.expression.values">initial.expression.values</code></td>
<td>
<p>(Optional) A vector with intial phi values.
The length of the vector has to equal the number of genes in the Genome object and the order of the genes should match the order of the genes in the Genome.
The default value is NULL.</p>
</td></tr>
<tr><td><code id="initializeParameterObject_+3A_model">model</code></td>
<td>
<p>Specifies the model used. Valid options are &quot;ROC&quot;, &quot;PA&quot;, &quot;PANSE&quot;, or &quot;FONSE&quot;.
The default model is &quot;ROC&quot;.
ROC is described in Gilchrist et al. 2015.
PA, PANSE and FONSE are currently unpublished.</p>
</td></tr>
<tr><td><code id="initializeParameterObject_+3A_split.serine">split.serine</code></td>
<td>
<p>Whether serine should be considered as 
one or two amino acids when running the model.
TRUE and FALSE are the only valid values.
The default value for split.serine is TRUE.</p>
</td></tr>
<tr><td><code id="initializeParameterObject_+3A_mixture.definition">mixture.definition</code></td>
<td>
<p>A string describing how each mixture should
be treated with respect to mutation and selection.
Valid values consist of &quot;allUnique&quot;, &quot;mutationShared&quot;, and &quot;selectionShared&quot;.
The default value for mixture.definition is &quot;allUnique&quot;.
See details for more information.</p>
</td></tr>
<tr><td><code id="initializeParameterObject_+3A_mixture.definition.matrix">mixture.definition.matrix</code></td>
<td>
<p>A matrix representation of how
the mutation and selection categories correspond to the mixtures.
The default value for mixture.definition.matrix is NULL.
If provided, the model will use the matrix to initialize the mutation and selection
categories instead of the definition listed directly above.
See details for more information.</p>
</td></tr>
<tr><td><code id="initializeParameterObject_+3A_init.with.restart.file">init.with.restart.file</code></td>
<td>
<p>File name containing information to reinitialize a 
previous Parameter object.
If given, all other arguments will be ignored.
The default value for init.with.restart.file is NULL.</p>
</td></tr>
<tr><td><code id="initializeParameterObject_+3A_mutation.prior.mean">mutation.prior.mean</code></td>
<td>
<p>Controlling the mean of the normal prior on mutation paramters.
If passed in as single number (default is 0), this will be the mean value for all categories, for all codons. User may also
supply a vector with n * 40 values, where n is the number of mutation categories. Future versions will check the number of rows matches
the number of mutation categories definded by user.</p>
</td></tr>
<tr><td><code id="initializeParameterObject_+3A_mutation.prior.sd">mutation.prior.sd</code></td>
<td>
<p>Controlling the standard deviation of the normal prior on the mutation parameters.
If passed in as single number (default is 0.35), this will be the standard deviation value for all categories, for all codons. User may also
supply a vector with n * 40 values, where n is the number of mutation categories. Future versions will check the number of rows matches
the number of mutation categories definded by user.</p>
</td></tr>
<tr><td><code id="initializeParameterObject_+3A_propose.by.prior">propose.by.prior</code></td>
<td>
<p>Mutation bias parameters will be proposed based on the means and standard deviations set in mutation.prior.mean and mutation.prior.sd</p>
</td></tr>
<tr><td><code id="initializeParameterObject_+3A_init.csp.variance">init.csp.variance</code></td>
<td>
<p>specifies the initial proposal width for codon specific parameter (default is 0.0025). 
The proposal width adapts during the runtime to reach a taget acceptance rate of ~0.25</p>
</td></tr>
<tr><td><code id="initializeParameterObject_+3A_init.sepsilon">init.sepsilon</code></td>
<td>
<p>specifies the initial value for sepsilon. default is 0.1</p>
</td></tr>
<tr><td><code id="initializeParameterObject_+3A_init.w.obs.phi">init.w.obs.phi</code></td>
<td>
<p>If TRUE, initialize phi values with observed phi values 
(data from RNAseq, mass spectrometry, ribosome footprinting) Default is FALSE. 
If multiple observed phi values exist for a gene, the geometric mean of these values is used as initial phi.
When using this function, one should remove any genes with 
missing phi values, as these genes will not have an initial phi value.</p>
</td></tr>
<tr><td><code id="initializeParameterObject_+3A_init.initiation.cost">init.initiation.cost</code></td>
<td>
<p>FOR FONSE ONLY. Initializes the initiation cost a_1 at this value.</p>
</td></tr>
<tr><td><code id="initializeParameterObject_+3A_init.partition.function">init.partition.function</code></td>
<td>
<p>FOR PANSE ONLY. initializes the partition function Z.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>initializeParameterObject</code> checks the values of the arguments 
given to insure the values are valid.
</p>
<p>The mixture definition and mixture definition matrix describes how the mutation
and selection categories are set up with respect to the number of mixtures. For
example, if mixture.definition = &quot;allUnique&quot; and numMixtures = 3, a matrix
representation would be <code>matrix(c(1,2,3,1,2,3), ncol=2)</code>
where each row represents a mixture, the first column represents the mutation
category, and the second column represents the selection category.
Another example would be mixture.definition = &quot;selectionShared&quot; and numMixtures = 4 (
<code>matrix(c(1,2,3,4,1,1,1,1), ncol=2)</code>).
In this case, the selection category is the same for every mixture. If a matrix
is given, and it is valid, then the mutation/selection relationship will be
defined by the given matrix and the keyword will be ignored. A matrix should only
be given in cases where the keywords would not create the desired matrix.
</p>


<h3>Value</h3>

<p>parameter Returns an initialized Parameter object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")
restart_file &lt;- system.file("extdata", "restart_file.rst", package = "AnaCoDa")

genome &lt;- initializeGenomeObject(file = genome_file)

## initialize a new parameter object
sphi_init &lt;- 1
numMixtures &lt;- 1
geneAssignment &lt;- rep(1, length(genome))
parameter &lt;- initializeParameterObject(genome = genome, sphi = sphi_init, 
                                       num.mixtures = numMixtures, 
                                       gene.assignment = geneAssignment, 
                                       mixture.definition = "allUnique")

## re-initialize a parameter object from a restart file. Useful for checkpointing
parameter &lt;- initializeParameterObject(init.with.restart.file = restart_file)

## initialize a parameter object with a custon mixture definition matrix
def.matrix &lt;- matrix(c(1,1,1,2), ncol=2)
geneAssignment &lt;- c(rep(1,floor(length(genome)/2)),rep(2,ceiling(length(genome)/2)))
parameter &lt;- initializeParameterObject(genome = genome, sphi = c(0.5, 2), num.mixtures = 2,
                                       gene.assignment = geneAssignment,
                                       mixture.definition.matrix = def.matrix)

</code></pre>

<hr>
<h2 id='initializeSynthesisRateByGenome'>initializeSynthesisRateByGenome</h2><span id='topic+initializeSynthesisRateByGenome'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Initialize synthesis rates using SCUO values calcuated from the genome
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="initializeSynthesisRateByGenome_+3A_genome">genome</code></td>
<td>
<p>a Genome object</p>
</td></tr>
</table>

<hr>
<h2 id='initializeSynthesisRateByList'>initializeSynthesisRateByList</h2><span id='topic+initializeSynthesisRateByList'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Initialize synthesis rates with values passed in as a list
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="initializeSynthesisRateByList_+3A_expression">expression</code></td>
<td>
<p>a list of values to use as initial synthesis rate values. Should be same size as number of genes in genome.</p>
</td></tr>
</table>

<hr>
<h2 id='initializeSynthesisRateByRandom'>initializeSynthesisRateByRandom</h2><span id='topic+initializeSynthesisRateByRandom'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Initialize synthesis rates by drawing a from a lognormal distribution with mean = -(sd_phi)^2/2 and sd = sd_phi
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="initializeSynthesisRateByRandom_+3A_sd_phi">sd_phi</code></td>
<td>
<p>a positive value which will be the standard deviation of the lognormal distribution</p>
</td></tr>
</table>

<hr>
<h2 id='initMutationCategories'>initMutationCategories</h2><span id='topic+initMutationCategories'></span>

<h3>Description</h3>

<p>Initialize values for mutation CSP. File should be of comma-separated with header. Three columns should be of order Amino_acid,Codon,Value
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="initMutationCategories_+3A_files">files</code></td>
<td>
<p>list of files containing starting values. Number of files should equal the number of categories.</p>
</td></tr>
<tr><td><code id="initMutationCategories_+3A_numcategories">numCategories</code></td>
<td>
<p>number of mutation categories (should be less than or equal to number of mixtures)</p>
</td></tr>
<tr><td><code id="initMutationCategories_+3A_fix">fix</code></td>
<td>
<p>Can use this parameter to fix mutation at current values (won't change over course of MCMC run)</p>
</td></tr>
</table>

<hr>
<h2 id='initSelectionCategories'>initSelectionCategories</h2><span id='topic+initSelectionCategories'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Initialize values for selection CSP. File should be of comma-separated with header. Three columns should be of order Amino_acid,Codon,Value
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="initSelectionCategories_+3A_files">files</code></td>
<td>
<p>list of files containing starting values. Number of files should equal the number of categories.</p>
</td></tr>
<tr><td><code id="initSelectionCategories_+3A_numcategories">numCategories</code></td>
<td>
<p>number of mutation categories (should be less than or equal to number of mixtures)</p>
</td></tr>
<tr><td><code id="initSelectionCategories_+3A_fix">fix</code></td>
<td>
<p>Can use this parameter to fix selection at current values (won't change over course of MCMC run)</p>
</td></tr>
</table>

<hr>
<h2 id='length.Rcpp_Genome'>Length of Genome</h2><span id='topic+length.Rcpp_Genome'></span>

<h3>Description</h3>

<p><code>length</code> gives the length of a genome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Rcpp_Genome'
length(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="length.Rcpp_Genome_+3A_x">x</code></td>
<td>
<p>A genome object initialized with <code><a href="#topic+initializeGenomeObject">initializeGenomeObject</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns the number of genes in a genome
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")
 
## reading genome
genome &lt;- initializeGenomeObject(file = genome_file)
length(genome) # 10

</code></pre>

<hr>
<h2 id='loadMCMCObject'>Load MCMC Object</h2><span id='topic+loadMCMCObject'></span>

<h3>Description</h3>

<p><code>loadMCMCObject</code> creates a new MCMC object and fills it with
the information in the file given.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadMCMCObject(files)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="loadMCMCObject_+3A_files">files</code></td>
<td>
<p>The filenames where the data will be stored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This MCMC object is not intended to be used to do another model fitting, only
to graph the stored results.
</p>


<h3>Value</h3>

<p>This function has no return value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## loading mcmc objects from the filesystem
## Not run: 
# load one mcmc object
mcmc &lt;- loadMCMCObject(files = "mcmc.Rda")

# load and combine multiple mcmc objects. Useful when using checkpointing
mcmc &lt;- loadMCMCObject(files = c("mcmc1.Rda", "mcmc2.Rda"))

## End(Not run)
</code></pre>

<hr>
<h2 id='loadParameterObject'>Load Parameter Object</h2><span id='topic+loadParameterObject'></span>

<h3>Description</h3>

<p><code>loadParameterObject</code> will load a parameter object from the filesystem
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadParameterObject(files)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="loadParameterObject_+3A_files">files</code></td>
<td>
<p>A list of parameter filenames to be loaded. If multiple files are given,
the parameter objects will be concatenated in the order provided</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function loads one or multiple files. In the case of multiple file, e.g. due to the use of check pointing, the files will
be concatenated to one parameter object. See <a href="#topic+writeParameterObject">writeParameterObject</a> for the writing of parameter objects
</p>


<h3>Value</h3>

<p>Returns an initialized Parameter object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# load a single parameter object
parameter &lt;- loadParameterObject("parameter.Rda")

# load and concatenate multiple parameter object
parameter &lt;- loadParameterObject(c("parameter1.Rda", "parameter2.Rda"))

## End(Not run)

</code></pre>

<hr>
<h2 id='plot.Rcpp_FONSEModel'>Plot Model Object</h2><span id='topic+plot.Rcpp_FONSEModel'></span>

<h3>Description</h3>

<p>Plots traces from the model object such as synthesis rates for each gene.
Will work regardless of whether or not expression/synthesis rate levels are being
estimated. If you wish to plot observed/empirical values, these values MUST be set
using the initial.expression.values parameter found in initializeParameterObject.
Otherwise, the expression values plotted will just be SCUO values estimated upon
initialization of the Parameter object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Rcpp_FONSEModel'
plot(
  x,
  genome,
  samples = 100,
  mixture = 1,
  simulated = FALSE,
  codon.window = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.Rcpp_FONSEModel_+3A_x">x</code></td>
<td>
<p>An Rcpp model object initialized with <code>initializeModelObject</code>.</p>
</td></tr>
<tr><td><code id="plot.Rcpp_FONSEModel_+3A_genome">genome</code></td>
<td>
<p>An Rcpp genome object initialized with <code>initializeGenomeObject</code>.</p>
</td></tr>
<tr><td><code id="plot.Rcpp_FONSEModel_+3A_samples">samples</code></td>
<td>
<p>The number of samples in the trace</p>
</td></tr>
<tr><td><code id="plot.Rcpp_FONSEModel_+3A_mixture">mixture</code></td>
<td>
<p>The mixture for which to graph values.</p>
</td></tr>
<tr><td><code id="plot.Rcpp_FONSEModel_+3A_simulated">simulated</code></td>
<td>
<p>A boolean value that determines whether to use the simulated genome.</p>
</td></tr>
<tr><td><code id="plot.Rcpp_FONSEModel_+3A_codon.window">codon.window</code></td>
<td>
<p>A boolean value that determines the codon window to use for calculating codon frequencies. If NULL (the default), use complete sequences.</p>
</td></tr>
<tr><td><code id="plot.Rcpp_FONSEModel_+3A_...">...</code></td>
<td>
<p>Optional, additional arguments.
For this function, a possible title for the plot in the form of a list if set with &quot;main&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function has no return value.
</p>

<hr>
<h2 id='plot.Rcpp_FONSEParameter'>Plot Parameter</h2><span id='topic+plot.Rcpp_FONSEParameter'></span>

<h3>Description</h3>

<p><code>plot</code> graphs the mutation or selection parameter for a ROC or FONSE
parameter object for each mixture element.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Rcpp_FONSEParameter'
plot(
  x,
  what = "Mutation",
  samples = 100,
  mixture.name = NULL,
  with.ci = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.Rcpp_FONSEParameter_+3A_x">x</code></td>
<td>
<p>A parameter object</p>
</td></tr>
<tr><td><code id="plot.Rcpp_FONSEParameter_+3A_what">what</code></td>
<td>
<p>Which aspect of the parameter to plot. Default value is
&quot;Mutation&quot;.</p>
</td></tr>
<tr><td><code id="plot.Rcpp_FONSEParameter_+3A_samples">samples</code></td>
<td>
<p>Number of samples to plot using the posterior mean. Default
value is 100.</p>
</td></tr>
<tr><td><code id="plot.Rcpp_FONSEParameter_+3A_mixture.name">mixture.name</code></td>
<td>
<p>a vector with names/descriptions of the mixture distributions in the parameter object</p>
</td></tr>
<tr><td><code id="plot.Rcpp_FONSEParameter_+3A_with.ci">with.ci</code></td>
<td>
<p>Plot with or without confidence intervals. Default value
is TRUE</p>
</td></tr>
<tr><td><code id="plot.Rcpp_FONSEParameter_+3A_...">...</code></td>
<td>
<p>Arguments to be passed to methods, such as graphical parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Graphs are based off the last # samples for the posterior mean.
</p>


<h3>Value</h3>

<p>This function has no return value.
</p>

<hr>
<h2 id='plot.Rcpp_MCMCAlgorithm'>Plot MCMC algorithm</h2><span id='topic+plot.Rcpp_MCMCAlgorithm'></span>

<h3>Description</h3>

<p>This function will plot the logLikelihood trace, and if the Hmisc package is installed, it will 
plot a subplot of the logLikelihood trace with the first few samples removed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Rcpp_MCMCAlgorithm'
plot(x, what = "LogPosterior", zoom.window = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.Rcpp_MCMCAlgorithm_+3A_x">x</code></td>
<td>
<p>An Rcpp_MCMC object initialized with <code>initializeMCMCObject</code>.</p>
</td></tr>
<tr><td><code id="plot.Rcpp_MCMCAlgorithm_+3A_what">what</code></td>
<td>
<p>character defining if log(Posterior) (Default) or log(Likelihood) 
options are: LogPosterior or logLikelihood</p>
</td></tr>
<tr><td><code id="plot.Rcpp_MCMCAlgorithm_+3A_zoom.window">zoom.window</code></td>
<td>
<p>A vector describing the start and end of the zoom window.</p>
</td></tr>
<tr><td><code id="plot.Rcpp_MCMCAlgorithm_+3A_...">...</code></td>
<td>
<p>Arguments to be passed to methods, such as graphical parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function has no return value.
</p>

<hr>
<h2 id='plot.Rcpp_ROCModel'>Plot Model Object</h2><span id='topic+plot.Rcpp_ROCModel'></span>

<h3>Description</h3>

<p>Plots traces from the model object such as synthesis rates for each gene.
Will work regardless of whether or not expression/synthesis rate levels are being
estimated. If you wish to plot observed/empirical values, these values MUST be set
using the initial.expression.values parameter found in initializeParameterObject.
Otherwise, the expression values plotted will just be SCUO values estimated upon
initialization of the Parameter object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Rcpp_ROCModel'
plot(x, genome = NULL, samples = 100, mixture = 1, simulated = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.Rcpp_ROCModel_+3A_x">x</code></td>
<td>
<p>An Rcpp model object initialized with <code>initializeModelObject</code>.</p>
</td></tr>
<tr><td><code id="plot.Rcpp_ROCModel_+3A_genome">genome</code></td>
<td>
<p>An Rcpp genome object initialized with <code>initializeGenomeObject</code>.</p>
</td></tr>
<tr><td><code id="plot.Rcpp_ROCModel_+3A_samples">samples</code></td>
<td>
<p>The number of samples in the trace</p>
</td></tr>
<tr><td><code id="plot.Rcpp_ROCModel_+3A_mixture">mixture</code></td>
<td>
<p>The mixture for which to graph values.</p>
</td></tr>
<tr><td><code id="plot.Rcpp_ROCModel_+3A_simulated">simulated</code></td>
<td>
<p>A boolean value that determines whether to use the simulated genome.</p>
</td></tr>
<tr><td><code id="plot.Rcpp_ROCModel_+3A_...">...</code></td>
<td>
<p>Optional, additional arguments.
For this function, a possible title for the plot in the form of a list if set with &quot;main&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function has no return value.
</p>

<hr>
<h2 id='plot.Rcpp_ROCParameter'>Plot Parameter</h2><span id='topic+plot.Rcpp_ROCParameter'></span>

<h3>Description</h3>

<p><code>plot</code> graphs the mutation or selection parameter for a ROC or FONSE
parameter object for each mixture element.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Rcpp_ROCParameter'
plot(
  x,
  what = "Mutation",
  samples = 100,
  mixture.name = NULL,
  with.ci = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.Rcpp_ROCParameter_+3A_x">x</code></td>
<td>
<p>A parameter object</p>
</td></tr>
<tr><td><code id="plot.Rcpp_ROCParameter_+3A_what">what</code></td>
<td>
<p>Which aspect of the parameter to plot. Default value is
&quot;Mutation&quot;.</p>
</td></tr>
<tr><td><code id="plot.Rcpp_ROCParameter_+3A_samples">samples</code></td>
<td>
<p>Number of samples to plot using the posterior mean. Default
value is 100.</p>
</td></tr>
<tr><td><code id="plot.Rcpp_ROCParameter_+3A_mixture.name">mixture.name</code></td>
<td>
<p>a vector with names/descriptions of the mixture distributions in the parameter object</p>
</td></tr>
<tr><td><code id="plot.Rcpp_ROCParameter_+3A_with.ci">with.ci</code></td>
<td>
<p>Plot with or without confidence intervals. Default value
is TRUE</p>
</td></tr>
<tr><td><code id="plot.Rcpp_ROCParameter_+3A_...">...</code></td>
<td>
<p>Arguments to be passed to methods, such as graphical parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Graphs are based off the last # samples for the posterior mean.
</p>


<h3>Value</h3>

<p>This function has no return value.
</p>

<hr>
<h2 id='plot.Rcpp_Trace'>Plot Trace Object</h2><span id='topic+plot.Rcpp_Trace'></span>

<h3>Description</h3>

<p>Plots different traces, specified with the <code>what</code> parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Rcpp_Trace'
plot(
  x,
  what = c("Mutation", "Selection", "MixtureProbability", "Sphi", "Mphi", "Aphi",
    "Sepsilon", "ExpectedPhi", "Expression", "NSEProb", "NSERate", "InitiationCost",
    "PartitionFunction"),
  geneIndex = 1,
  mixture = 1,
  log.10.scale = F,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.Rcpp_Trace_+3A_x">x</code></td>
<td>
<p>An Rcpp trace object initialized with <code>initializeTraceObject</code>.</p>
</td></tr>
<tr><td><code id="plot.Rcpp_Trace_+3A_what">what</code></td>
<td>
<p>A string containing one of the following to graph: <code>Mutation, Selection, Alpha, LambdaPrime, MeanWaitingTime, VarWaitingTime
MixtureProbability, Sphi, Mphi, Aphi, Spesilon, ExpectedPhi, Expression</code>.</p>
</td></tr>
<tr><td><code id="plot.Rcpp_Trace_+3A_geneindex">geneIndex</code></td>
<td>
<p>When plotting expression, the index of the gene to be plotted.</p>
</td></tr>
<tr><td><code id="plot.Rcpp_Trace_+3A_mixture">mixture</code></td>
<td>
<p>The mixture for which to plot values.</p>
</td></tr>
<tr><td><code id="plot.Rcpp_Trace_+3A_log.10.scale">log.10.scale</code></td>
<td>
<p>A logical value determining if figures should be plotted on the log.10.scale (default=F). Should not be applied to mutation and selection parameters estimated by ROC/FONSE.</p>
</td></tr>
<tr><td><code id="plot.Rcpp_Trace_+3A_...">...</code></td>
<td>
<p>Optional, additional arguments.
For this function, may be a logical value determining if the trace is ROC-based or not.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function has no return value.
</p>

<hr>
<h2 id='plotAcceptanceRatios'>Plot Acceptance ratios</h2><span id='topic+plotAcceptanceRatios'></span>

<h3>Description</h3>

<p>Plots acceptance ratios for codon-specific parameters. Will be by amino acid for ROC and FONSE models, but will be by codon for PA and PANSE models. Note assumes estimating parameters for all codons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotAcceptanceRatios(trace, main = "CSP Acceptance Ratio Traces")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotAcceptanceRatios_+3A_trace">trace</code></td>
<td>
<p>An Rcpp trace object initialized with <code>initializeTraceObject</code>.</p>
</td></tr>
<tr><td><code id="plotAcceptanceRatios_+3A_main">main</code></td>
<td>
<p>The title of the plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function has no return value.
</p>

<hr>
<h2 id='plotCodonSpecificParameters'>Plot Codon Specific Parameter</h2><span id='topic+plotCodonSpecificParameters'></span>

<h3>Description</h3>

<p>Plots a codon-specific set of traces, specified with the <code>type</code> parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotCodonSpecificParameters(
  trace,
  mixture,
  type = "Mutation",
  main = "Mutation Parameter Traces",
  ROC.or.FONSE = TRUE,
  log.10.scale = F
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotCodonSpecificParameters_+3A_trace">trace</code></td>
<td>
<p>An Rcpp trace object initialized with <code>initializeTraceObject</code>.</p>
</td></tr>
<tr><td><code id="plotCodonSpecificParameters_+3A_mixture">mixture</code></td>
<td>
<p>The mixture for which to plot values.</p>
</td></tr>
<tr><td><code id="plotCodonSpecificParameters_+3A_type">type</code></td>
<td>
<p>A string containing one of the following to graph: <code>Mutation, Selection, Alpha, LambdaPrime, MeanWaitingTime, VarWaitingTime</code>.</p>
</td></tr>
<tr><td><code id="plotCodonSpecificParameters_+3A_main">main</code></td>
<td>
<p>The title of the plot.</p>
</td></tr>
<tr><td><code id="plotCodonSpecificParameters_+3A_roc.or.fonse">ROC.or.FONSE</code></td>
<td>
<p>A logical value determining if the Parameter was ROC/FONSE or not.</p>
</td></tr>
<tr><td><code id="plotCodonSpecificParameters_+3A_log.10.scale">log.10.scale</code></td>
<td>
<p>A logical value determining if figures should be plotted on the log.10.scale (default=F). Should not be applied to mutation and selection parameters estimated by ROC/FONSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function has no return value.
</p>

<hr>
<h2 id='readPhiValue'>readPhiValue</h2><span id='topic+readPhiValue'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Read synthesis rate values from file. File should be two column file &lt;gene_id,phi&gt; and is expected to have a header row
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="readPhiValue_+3A_filename">filename</code></td>
<td>
<p>name of file to be read</p>
</td></tr>
</table>

<hr>
<h2 id='runMCMC'>Run MCMC</h2><span id='topic+runMCMC'></span>

<h3>Description</h3>

<p><code>runMCMC</code> will run a monte carlo markov chain algorithm
for the given mcmc, genome, and model objects to perform a model fitting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runMCMC(mcmc, genome, model, ncores = 1, divergence.iteration = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="runMCMC_+3A_mcmc">mcmc</code></td>
<td>
<p>MCMC object that will run the model fitting algorithm.</p>
</td></tr>
<tr><td><code id="runMCMC_+3A_genome">genome</code></td>
<td>
<p>Genome that the model fitting will run on. Should be 
the same genome associated with the parameter and model objects.</p>
</td></tr>
<tr><td><code id="runMCMC_+3A_model">model</code></td>
<td>
<p>Model to run the fitting on. Should be associated with
the given genome.</p>
</td></tr>
<tr><td><code id="runMCMC_+3A_ncores">ncores</code></td>
<td>
<p>Number of cores to perform the model fitting with. Default
value is 1.</p>
</td></tr>
<tr><td><code id="runMCMC_+3A_divergence.iteration">divergence.iteration</code></td>
<td>
<p>Number of steps that the initial conditions
can diverge from the original conditions given. Default value is 0.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>runMCMC</code> will run for the number of samples times the number
thinning given when the mcmc object is initialized. Updates are provided every 100
steps, and the state of the chain is saved every thinning steps.
</p>


<h3>Value</h3>

<p>This function has no return value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#fitting a model to a genome using the runMCMC function

genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")

genome &lt;- initializeGenomeObject(file = genome_file)
sphi_init &lt;- c(1,1)
numMixtures &lt;- 2
geneAssignment &lt;- c(rep(1,floor(length(genome)/2)),rep(2,ceiling(length(genome)/2)))
parameter &lt;- initializeParameterObject(genome = genome, sphi = sphi_init, 
                                       num.mixtures = numMixtures, 
                                       gene.assignment = geneAssignment, 
                                       mixture.definition = "allUnique")
model &lt;- initializeModelObject(parameter = parameter, model = "ROC")
samples &lt;- 2500
thinning &lt;- 50
adaptiveWidth &lt;- 25
mcmc &lt;- initializeMCMCObject(samples = samples, thinning = thinning, 
                             adaptive.width=adaptiveWidth, est.expression=TRUE, 
                             est.csp=TRUE, est.hyper=TRUE, est.mix = TRUE) 
divergence.iteration &lt;- 10
## Not run: 
runMCMC(mcmc = mcmc, genome = genome, model = model, 
        ncores = 4, divergence.iteration = divergence.iteration)

## End(Not run)

</code></pre>

<hr>
<h2 id='setAdaptiveWidth'>setAdaptiveWidth</h2><span id='topic+setAdaptiveWidth'></span>

<h3>Description</h3>

<p>Method of MCMC class (access via mcmc$&lt;function name&gt;, where mcmc is an object initialized by initializeMCMCObject). Set sample adaptiveWidth value, which is the number of samples (not iterations) between adapting parameter proposal widths
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="setAdaptiveWidth_+3A__adaptivewidth">_adaptiveWidth</code></td>
<td>
<p>postive value</p>
</td></tr>
</table>

<hr>
<h2 id='setGroupList'>setGroupList</h2><span id='topic+setGroupList'></span>

<h3>Description</h3>

<p>Method of Parameter class (access via parameter$&lt;function name&gt;, where parameter is an object initialized by initializeParameterObject). Set amino acids (ROC, FONSE) or codons (PA, PANSE) for which parameters will be estimated. Note that non-default groupLists are still in beta testing and should be used with caution.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="setGroupList_+3A_list">List</code></td>
<td>
<p>of strings epresenting groups for parameters to be estimated. Should be one letter amino acid (ROC, FONSE) or list of sense codons (PA, PANSE).</p>
</td></tr>
</table>

<hr>
<h2 id='setRestartFileSettings'>setRestartFileSettings</h2><span id='topic+setRestartFileSettings'></span>

<h3>Description</h3>

<p>Method of MCMC class (access via mcmc$&lt;function name&gt;, where mcmc is an object initialized by initializeMCMCObject). Set restart file output name and frequency prior to running MCMC
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="setRestartFileSettings_+3A_filename">filename</code></td>
<td>
<p>name of restart file</p>
</td></tr>
<tr><td><code id="setRestartFileSettings_+3A_interval">interval</code></td>
<td>
<p>number of samples (ie. iterations * thinning) between writing new restart file</p>
</td></tr>
<tr><td><code id="setRestartFileSettings_+3A_multiple">multiple</code></td>
<td>
<p>if true, will output a new restart file at each interval (file name will include sample it was written at)</p>
</td></tr>
</table>

<hr>
<h2 id='setRestartSettings'>Set Restart Settings</h2><span id='topic+setRestartSettings'></span>

<h3>Description</h3>

<p><code>setRestartSettings</code> sets the needed information (what the file 
is called, how often the file should be written) to write
information to restart the MCMC algorithm from a given point.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setRestartSettings(mcmc, filename, samples, write.multiple = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setRestartSettings_+3A_mcmc">mcmc</code></td>
<td>
<p>MCMC object that will run the model fitting algorithm.</p>
</td></tr>
<tr><td><code id="setRestartSettings_+3A_filename">filename</code></td>
<td>
<p>Filename for the restart files to be written.</p>
</td></tr>
<tr><td><code id="setRestartSettings_+3A_samples">samples</code></td>
<td>
<p>Number of samples that should occur before a file is written.</p>
</td></tr>
<tr><td><code id="setRestartSettings_+3A_write.multiple">write.multiple</code></td>
<td>
<p>Boolean that determines if multiple restart files
are written. Default value is TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>setRestartSettings</code> writes a restart file every set amount of samples
that occur. Also, if write.multiple is true, instead of overwriting the previous restart
file, the sample number is prepended onto the file name and multiple rerstart files
are generated for a run.
</p>


<h3>Value</h3>

<p>This function has no return value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## set restart settings for checkpointing

samples &lt;- 2500
thinning &lt;- 50
adaptiveWidth &lt;- 25

## estimate all parameter types
mcmc &lt;- initializeMCMCObject(samples = samples, thinning = thinning, 
                             adaptive.width=adaptiveWidth, est.expression=TRUE, 
                             est.csp=TRUE, est.hyper=TRUE, est.mix = TRUE) 
                             
# prompts the mcmc to write a restart file every 100 samples during the run.
setRestartSettings(mcmc = mcmc, filename = "test_restart", samples = 100)

# prompts the mcmc to write a restart file every 100 samples during the run, 
# but will overwrite it each time.
setRestartSettings(mcmc = mcmc, filename = "test_restart", samples = 100, 
                   write.multiple = FALSE)
           
</code></pre>

<hr>
<h2 id='setSamples'>setSamples</h2><span id='topic+setSamples'></span>

<h3>Description</h3>

<p>Method of MCMC class (access via mcmc$&lt;function name&gt;, where mcmc is an object initialized by initializeMCMCObject). Set number of samples set for MCMCAlgorithm object
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="setSamples_+3A__samples">_samples</code></td>
<td>
<p>postive value</p>
</td></tr>
</table>

<hr>
<h2 id='setStepsToAdapt'>setStepsToAdapt</h2><span id='topic+setStepsToAdapt'></span>

<h3>Description</h3>

<p>Method of MCMC class (access via mcmc$&lt;function name&gt;, where mcmc is an object initialized by initializeMCMCObject). Set number of iterations (total iterations = samples * thinning) to allow proposal widths to adapt
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="setStepsToAdapt_+3A_steps">steps</code></td>
<td>
<p>a postive value</p>
</td></tr>
</table>

<hr>
<h2 id='setThinning'>setThinning</h2><span id='topic+setThinning'></span>

<h3>Description</h3>

<p>Set thinning value, which is the number of iterations (total iterations = samples * thinning) not being kept
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="setThinning_+3A__thinning">_thinning</code></td>
<td>
<p>postive value</p>
</td></tr>
</table>

<hr>
<h2 id='simulateGenome'>simulateGenome</h2><span id='topic+simulateGenome'></span>

<h3>Description</h3>

<p>Method of Model class (access via model$&lt;function name&gt;, where model is an object initialized by initializeModelObject). Will simulate a version of the given genome using the current set of parameters stored in the Parameter object. This can be written to a FASTA file using genome$writeFasta(&lt;filename&gt;,simulated = TRUE).
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulateGenome_+3A_genome">genome</code></td>
<td>
<p>a Genome object initialized by initializeGenomeObject</p>
</td></tr>
</table>

<hr>
<h2 id='summary.Rcpp_Genome'>Summary of Genome</h2><span id='topic+summary.Rcpp_Genome'></span>

<h3>Description</h3>

<p><code>summary</code> summarizes the description of a genome, such as number of genes and average gene length.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Rcpp_Genome'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.Rcpp_Genome_+3A_object">object</code></td>
<td>
<p>A genome object initialized with <code><a href="#topic+initializeGenomeObject">initializeGenomeObject</a></code>.</p>
</td></tr>
<tr><td><code id="summary.Rcpp_Genome_+3A_...">...</code></td>
<td>
<p>Optional, additional arguments to be passed to the main summary function 
that affect the summary produced.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns by default an object of class c(&quot;summaryDefault&quot;, table&quot;).
</p>

<hr>
<h2 id='writeMCMCObject'>Write MCMC Object</h2><span id='topic+writeMCMCObject'></span>

<h3>Description</h3>

<p><code>writeMCMCObject</code> stores the MCMC information from the 
model fitting run in a file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeMCMCObject(mcmc, file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writeMCMCObject_+3A_mcmc">mcmc</code></td>
<td>
<p>MCMC object that has run the model fitting algorithm.</p>
</td></tr>
<tr><td><code id="writeMCMCObject_+3A_file">file</code></td>
<td>
<p>A filename where the data will be stored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function has no return value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## saving the MCMC object after model fitting
genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")

genome &lt;- initializeGenomeObject(file = genome_file)
sphi_init &lt;- c(1,1)
numMixtures &lt;- 2
geneAssignment &lt;- c(rep(1,floor(length(genome)/2)),rep(2,ceiling(length(genome)/2)))
parameter &lt;- initializeParameterObject(genome = genome, sphi = sphi_init, 
                                       num.mixtures = numMixtures, 
                                       gene.assignment = geneAssignment, 
                                       mixture.definition = "allUnique")
samples &lt;- 2500
thinning &lt;- 50
adaptiveWidth &lt;- 25
mcmc &lt;- initializeMCMCObject(samples = samples, thinning = thinning, 
                             adaptive.width=adaptiveWidth, est.expression=TRUE, 
                             est.csp=TRUE, est.hyper=TRUE, est.mix = TRUE) 
divergence.iteration &lt;- 10
## Not run: 
runMCMC(mcmc = mcmc, genome = genome, model = model, 
        ncores = 4, divergence.iteration = divergence.iteration)
writeMCMCObject(mcmc = mcmc, file = file.path(tempdir(), "file.Rda"))


## End(Not run)
</code></pre>

<hr>
<h2 id='writeParameterObject'>Write Parameter Object to a File</h2><span id='topic+writeParameterObject'></span>

<h3>Description</h3>

<p><code>writeParameterObject</code> will write the parameter object as binary to the filesystem
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeParameterObject(parameter, file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writeParameterObject_+3A_parameter">parameter</code></td>
<td>
<p>parameter on object created by <code>initializeParameterObject</code>.</p>
</td></tr>
<tr><td><code id="writeParameterObject_+3A_file">file</code></td>
<td>
<p>A filename that where the data will be stored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As Rcpp object are not serializable with the default R <code>save</code> function, 
therefore this custom save function is provided (see <a href="#topic+loadParameterObject">loadParameterObject</a>).
</p>


<h3>Value</h3>

<p>This function has no return value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")

genome &lt;- initializeGenomeObject(file = genome_file)
sphi_init &lt;- c(1,1)
numMixtures &lt;- 2
geneAssignment &lt;- c(rep(1,floor(length(genome)/2)),rep(2,ceiling(length(genome)/2)))
parameter &lt;- initializeParameterObject(genome = genome, sphi = sphi_init, 
                                       num.mixtures = numMixtures, 
                                       gene.assignment = geneAssignment, 
                                       mixture.definition = "allUnique")

## writing an empty parameter object as the runMCMC routine was not called yet
writeParameterObject(parameter = parameter, file = file.path(tempdir(), "file.Rda"))


## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
