<!DOCTYPE html><html><head><title>Help for package Anaconda</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Anaconda}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Bacteria'><p>Bacteria</p></a></li>
<li><a href='#clusteringGOs'><p>clusteringGOs</p></a></li>
<li><a href='#dasva_raw_input'><p>dasva_raw_input</p></a></li>
<li><a href='#database_bacteria_creation'><p>database_bacteria_creation</p></a></li>
<li><a href='#database_fungi_creation'><p>database_fungi_creation</p></a></li>
<li><a href='#database_fungi_creation_RepSeq'><p>database_fungi_creation_RepSeq</p></a></li>
<li><a href='#database_metazoan_creation'><p>database_metazoan_creation</p></a></li>
<li><a href='#fisherTest'><p>fisherTest</p></a></li>
<li><a href='#format_input'><p>format_input</p></a></li>
<li><a href='#Fungi'><p>Fungi</p></a></li>
<li><a href='#funguild_input_targeted'><p>funguild_input_targeted</p></a></li>
<li><a href='#get_bactotraits_targeted'><p>get_bactotraits_targeted</p></a></li>
<li><a href='#get_dasva'><p>get_dasva</p></a></li>
<li><a href='#get_funguilds'><p>get_funguilds</p></a></li>
<li><a href='#get_funguilds_targeted'><p>get_funguilds_targeted</p></a></li>
<li><a href='#get_input_files'><p>get_input_files</p></a></li>
<li><a href='#get_link_guilds'><p>get_link_guilds</p></a></li>
<li><a href='#get_taxon_list_drawer'><p>get_taxon_list_drawer</p></a></li>
<li><a href='#heatmap_condition'><p>heatmap_condition</p></a></li>
<li><a href='#heatmap_data_dasva'><p>heatmap_data_dasva</p></a></li>
<li><a href='#heatmap_samples_hclust'><p>heatmap_samples_hclust</p></a></li>
<li><a href='#heatmap_samples_matrix'><p>heatmap_samples_matrix</p></a></li>
<li><a href='#heatmap_taxo'><p>heatmap_taxo</p></a></li>
<li><a href='#input_global_analysis'><p>input_global_analysis</p></a></li>
<li><a href='#Metazoan'><p>Metazoan</p></a></li>
<li><a href='#move_files'><p>move_files</p></a></li>
<li><a href='#mwuTest'><p>mwuTest</p></a></li>
<li><a href='#PCA_data_dasva'><p>PCA_data_dasva</p></a></li>
<li><a href='#plotDispASVs'><p>plotDispASVs</p></a></li>
<li><a href='#plotMA.dasva'><p>plotMA.dasva</p></a></li>
<li><a href='#plotPCA.san'><p>plotPCA.san</p></a></li>
<li><a href='#plotSparsityASV'><p>plotSparsityASV</p></a></li>
<li><a href='#samplesInfo'><p>samplesInfo</p></a></li>
<li><a href='#target_file'><p>target_file</p></a></li>
<li><a href='#taxon_mwu_list'><p>taxon_mwu_list</p></a></li>
<li><a href='#taxon_mwuPlot'><p>taxon_mwuPlot</p></a></li>
<li><a href='#taxon_mwuPlot_guilds'><p>taxon_mwuPlot_guilds</p></a></li>
<li><a href='#taxon_mwuStats'><p>taxon_mwuStats</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Targeted Differential and Global Enrichment Analysis of
Taxonomic Rank by Shared Asvs</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.5</td>
</tr>
<tr>
<td>Author:</td>
<td>Pierre-Louis Stenger [cre, aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Pierre-Louis Stenger &lt;Pierrelouis.stenger@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Targeted differential and global enrichment analysis of taxonomic rank by shared ASVs (Amplicon Sequence Variant), for high-throughput eDNA sequencing of fungi, bacteria, and metazoan.
    Actually works in two steps: I) Targeted differential analysis from QIIME2 data and II) Global analysis by Taxon Mann-Whitney U test analysis from targeted analysis (I)
    (I) Estimate variance-mean dependence in count/abundance ASVs data from high-throughput sequencing assays and test for differential represented ASVs based on a model using the negative binomial distribution.
    (II) NCBITaxon_MWU uses continuous measure of significance (such as fold-change or -log(p-value)) to identify NCBITaxon that are significantly enriches with either up- or down-represented ASVs. If the measure is binary (0 or 1) the script will perform a typical 'NCBITaxon enrichment' analysis based Fisher's exact test: it will show NCBITaxon over-represented among the ASVs that have 1 as their measure. On the plot, different fonts are used to indicate significance and color indicates enrichment with either up (red) or down (blue) regulated ASVs. No colors are shown for binary measure analysis. The tree on the plot is hierarchical clustering of NCBITaxon based on shared ASVs. Categories with no branch length between them are subsets of each other. The fraction next to the category name indicates the fraction of 'good' ASVs in it; 'good' ASVs are the ones exceeding the arbitrary absValue cutoff (option in taxon_mwuPlot()). For Fisher's based test, specify absValue=0.5. This value does not affect statistics and is used for plotting only. The original idea was for genes differential expression analysis from Wright et al (2015) &lt;<a href="https://doi.org/10.1186%2Fs12864-015-1540-2">doi:10.1186/s12864-015-1540-2</a>&gt;; adapted here for taxonomic analysis.
    The 'Anaconda' package makes it possible to carry out these analyses by automatically creating several graphs and tables and storing them in specially created subfolders. You will need your QIIME2 pipeline output for each kingdom (eg; Fungi and/or Bacteria and/or Metazoan): i) taxonomy.tsv, ii) taxonomy_RepSeq.tsv, iii) ASV.tsv and iv) SampleSheet_comparison.txt (the latter being created by you).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/PLStenger/Anaconda">https://github.com/PLStenger/Anaconda</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/PLStenger/Anaconda/issues">https://github.com/PLStenger/Anaconda/issues</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>ggrepel, pheatmap, lookup, plyr, data.table, rafalib,
RColorBrewer, methods, graphics</td>
</tr>
<tr>
<td>Depends:</td>
<td>ape, DESeq2, ggplot2</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-10-13 08:09:09 UTC; pierre-louisstenger</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-10-14 08:55:29 UTC</td>
</tr>
</table>
<hr>
<h2 id='Bacteria'>Bacteria</h2><span id='topic+Bacteria'></span>

<h3>Description</h3>

<p>This function create a new folder named Bacteria and set your working directory into this folder. Please, run setwd(&quot;Bacteria&quot;) after this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Bacteria(nothing)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Bacteria_+3A_nothing">nothing</code></td>
<td>
<p>It's important not to write anything between the brackets, a new folder named Bacteria will be created and your working directory will be set into this folder, depending of the selected Kingdom.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new folder named Bacteria will be created and your working directory will be set into this folder, depending of the selected Kingdom.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: Bacteria()
# Please, run setwd("Bacteria") after this function.
</code></pre>

<hr>
<h2 id='clusteringGOs'>clusteringGOs</h2><span id='topic+clusteringGOs'></span>

<h3>Description</h3>

<p>clusteringGOs from DESeq2 analysis pipeline
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clusteringGOs(gen2go, div, cutHeight)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clusteringGOs_+3A_gen2go">gen2go</code></td>
<td>
<p>from DESeq2 analysis pipeline</p>
</td></tr>
<tr><td><code id="clusteringGOs_+3A_div">div</code></td>
<td>
<p>div</p>
</td></tr>
<tr><td><code id="clusteringGOs_+3A_cutheight">cutHeight</code></td>
<td>
<p>cutHeight</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a clustering GO
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: clusteringGOs()
</code></pre>

<hr>
<h2 id='dasva_raw_input'>dasva_raw_input</h2><span id='topic+dasva_raw_input'></span>

<h3>Description</h3>

<p>Used in heatmap_samples_hclust(), heatmap_samples_matrix(), PCA_data_dasva() and get_dasva() functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dasva_raw_input(sampleTable, directory = ".", design, ignoreRank = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dasva_raw_input_+3A_sampletable">sampleTable</code></td>
<td>
<p>Depending of the heatmap_samples_hclust(), heatmap_samples_matrix(), PCA_data_dasva() and get_dasva() functions.</p>
</td></tr>
<tr><td><code id="dasva_raw_input_+3A_directory">directory</code></td>
<td>
<p>directory</p>
</td></tr>
<tr><td><code id="dasva_raw_input_+3A_design">design</code></td>
<td>
<p>design</p>
</td></tr>
<tr><td><code id="dasva_raw_input_+3A_ignorerank">ignoreRank</code></td>
<td>
<p>ignoreRank</p>
</td></tr>
<tr><td><code id="dasva_raw_input_+3A_...">...</code></td>
<td>
<p>...</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## Not run: dasva_raw &lt;- dasva_raw_input(sampleTable = sampleTable,
  directory = targeted_analysis_dir,
  design= ~ condition)
## End(Not run)
</code></pre>

<hr>
<h2 id='database_bacteria_creation'>database_bacteria_creation</h2><span id='topic+database_bacteria_creation'></span>

<h3>Description</h3>

<p>Create a Database for Bacteria kingdom for Global analysis by Taxon_MWU analysis from targeted analysis. Please, run setwd(&quot;02_Global_analysis&quot;) after this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>database_bacteria_creation(nothing)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="database_bacteria_creation_+3A_nothing">nothing</code></td>
<td>
<p>It's important not to write anything between the brackets, the database will create itself.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame file named database_bacteria_package_all.tab created from the taxonomy_all_bacteria_QIIME2_and_NCBI_format.txt file and your own taxonomy_RepSeq.tsv file.
database_bacteria_creation()
</p>


<h3>Examples</h3>

<pre><code class='language-R'># It is important not to write anything between the brackets, the database will create itself.
## Not run: database_bacteria_creation()
# Please, run setwd("02_Global_analysis") after this function.
</code></pre>

<hr>
<h2 id='database_fungi_creation'>database_fungi_creation</h2><span id='topic+database_fungi_creation'></span>

<h3>Description</h3>

<p>Create a Database for Fungi kingdom for Global analysis by Taxon_MWU analysis from targeted analysis only from rarefied ASVs. Please, run setwd(&quot;02_Global_analysis&quot;) after this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>database_fungi_creation(nothing)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="database_fungi_creation_+3A_nothing">nothing</code></td>
<td>
<p>It's important not to write anything between the brackets, the database will create itself.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame file named database_fungi_package_all.tab created from the taxonomy_all_bacteria_QIIME2_and_NCBI_format.txt file and your own taxonomy.tsv file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># It is important not to write anything between the brackets, the database will create itself.
## Not run: database_fungi_creation()
# Please, run setwd("02_Global_analysis") after this function.
</code></pre>

<hr>
<h2 id='database_fungi_creation_RepSeq'>database_fungi_creation_RepSeq</h2><span id='topic+database_fungi_creation_RepSeq'></span>

<h3>Description</h3>

<p>Create a Database for Fungi kingdom for Global analysis by Taxon_MWU analysis from targeted analysis. Please, run setwd(&quot;02_Global_analysis&quot;) after this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>database_fungi_creation_RepSeq(nothing)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="database_fungi_creation_RepSeq_+3A_nothing">nothing</code></td>
<td>
<p>It's important not to write anything between the brackets, the database will create itself.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame file named database_fungi_package_all.tab created from the taxonomy_all_bacteria_QIIME2_and_NCBI_format.txt file and your own taxonomy_RepSeq.tsv file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># It is important not to write anything between the brackets, the database will create itself.
## Not run: database_fungi_creation_RepSeq()
# Please, run setwd("02_Global_analysis") after this function.
</code></pre>

<hr>
<h2 id='database_metazoan_creation'>database_metazoan_creation</h2><span id='topic+database_metazoan_creation'></span>

<h3>Description</h3>

<p>Create a Database for metazoan kingdom for Global analysis by Taxon_MWU analysis from targeted analysis. Please, run setwd(&quot;02_Global_analysis&quot;) after this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>database_metazoan_creation(nothing)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="database_metazoan_creation_+3A_nothing">nothing</code></td>
<td>
<p>It's important not to write anything between the brackets, the database will create itself.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame file named database_metazoan_package_all.tab created from the taxonomy_all_metazoan_QIIME2_and_NCBI_format.txt file and your own taxonomy_RepSeq.tsv file.
database_metazoan_creation()
</p>


<h3>Examples</h3>

<pre><code class='language-R'># It is important not to write anything between the brackets, the database will create itself.
## Not run: database_metazoan_creation()
# Please, run setwd("02_Global_analysis") after this function.
</code></pre>

<hr>
<h2 id='fisherTest'>fisherTest</h2><span id='topic+fisherTest'></span>

<h3>Description</h3>

<p>Ficher Test from RBGOA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fisherTest(gotable)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fisherTest_+3A_gotable">gotable</code></td>
<td>
<p>from gomwuStats from RBGOA</p>
</td></tr>
</table>


<h3>Value</h3>

<p>fisherTest
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: fisherTest()
</code></pre>

<hr>
<h2 id='format_input'>format_input</h2><span id='topic+format_input'></span>

<h3>Description</h3>

<p>Apply logP on both positive and negative ASVs FC
</p>


<h3>Usage</h3>

<pre><code class='language-R'>format_input(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="format_input_+3A_x">x</code></td>
<td>
<p>Object from the Differential ASV abundance (DASVA) analysis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an input for the input_global_analysis() function
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: format_input(x)
</code></pre>

<hr>
<h2 id='Fungi'>Fungi</h2><span id='topic+Fungi'></span>

<h3>Description</h3>

<p>This function create a new folder named Fungi and set your working directory into this folder. Please, run setwd(&quot;Fungi&quot;) after this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Fungi(nothing)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Fungi_+3A_nothing">nothing</code></td>
<td>
<p>It's important not to write anything between the brackets, a new folder named Fungi will be created and your working directory will be set into this folder, depending of the selected Kingdom.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new folder named Fungi will be created and your working directory will be set into this folder, depending of the selected Kingdom.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: Fungi()
# plaese, run setwd("Fungi") after this function.
</code></pre>

<hr>
<h2 id='funguild_input_targeted'>funguild_input_targeted</h2><span id='topic+funguild_input_targeted'></span>

<h3>Description</h3>

<p>Prepare Object for Fungi Guilds for Fungi kingdom for targeted analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>funguild_input_targeted(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="funguild_input_targeted_+3A_x">x</code></td>
<td>
<p>Object from the Differential ASV abundance (DASVA) analysis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An Object used for Fungi Guilds informations for Fungi kingdom for targeted analysis from the Differential ASV abundance (DASVA) analysis
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: get_funguilds_targeted(res_forest_vs_long_fallow_guilds)
</code></pre>

<hr>
<h2 id='get_bactotraits_targeted'>get_bactotraits_targeted</h2><span id='topic+get_bactotraits_targeted'></span>

<h3>Description</h3>

<p>Obtain Bacterial Traits for Bacteria kingdom for targeted analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_bactotraits_targeted(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_bactotraits_targeted_+3A_x">x</code></td>
<td>
<p>Object from the Differential ASV abundance (DASVA) analysis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame file with Bacterial Traits informations for Bacteria kingdom for targeted analysis from the Differential ASV abundance (DASVA) analysis
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: get_bactotraits_targeted(res_forest_vs_long_fallow)
</code></pre>

<hr>
<h2 id='get_dasva'>get_dasva</h2><span id='topic+get_dasva'></span>

<h3>Description</h3>

<p>Creates the DASVA object. Fit a Gamma-Poisson Generalized Linear Model, dispersion estimates for Negative Binomial distributed data, &quot;parametric&quot;, local&quot; or &quot;mean&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_dasva(fitType = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_dasva_+3A_fittype">fitType</code></td>
<td>
<p>Fit a Gamma-Poisson Generalized Linear Model, dispersion estimates for Negative Binomial distributed data, &quot;parametric&quot;, local&quot; or &quot;mean&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DASVA object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: dasva &lt;- get_dasva(fitType="parametric")
dasva &lt;- get_dasva(fitType="local")
dasva &lt;- get_dasva(fitType="mean")
## End(Not run)
</code></pre>

<hr>
<h2 id='get_funguilds'>get_funguilds</h2><span id='topic+get_funguilds'></span>

<h3>Description</h3>

<p>get Fungi Guilds from taxon_list_drawer Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_funguilds(taxon_list_drawer)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_funguilds_+3A_taxon_list_drawer">taxon_list_drawer</code></td>
<td>
<p>object from get_taxon_list_drawer() function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>funguilds Object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: funguilds &lt;- get_funguilds(taxon_list_drawer)
</code></pre>

<hr>
<h2 id='get_funguilds_targeted'>get_funguilds_targeted</h2><span id='topic+get_funguilds_targeted'></span>

<h3>Description</h3>

<p>Obtain Fungi Guilds for Fungi kingdom for targeted analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_funguilds_targeted(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_funguilds_targeted_+3A_x">x</code></td>
<td>
<p>Object from the funguild_input_targeted() output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame file with Fungi Guilds informations for Fungi kingdom for targeted analysis from the Differential ASV abundance (DASVA) analysis
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: get_funguilds_targeted(res_forest_vs_long_fallow_guilds)
</code></pre>

<hr>
<h2 id='get_input_files'>get_input_files</h2><span id='topic+get_input_files'></span>

<h3>Description</h3>

<p>Created sub directory &quot;Targeted_analysis&quot; if not already exist. Then, create one file by condition into it, and then upload the taxonomy file. Please, run setwd(&quot;01_Targeted_analysis&quot;) after this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_input_files(nothing)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_input_files_+3A_nothing">nothing</code></td>
<td>
<p>It's important not to write anything between the brackets, all inputs will be adapted automatically.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>taxo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: taxo &lt;- get_input_files()
# please, run setwd("01_Targeted_analysis") after this function.
</code></pre>

<hr>
<h2 id='get_link_guilds'>get_link_guilds</h2><span id='topic+get_link_guilds'></span>

<h3>Description</h3>

<p>get link guilds from taxon_list and funguilds Objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_link_guilds(taxon_list, funguilds)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_link_guilds_+3A_taxon_list">taxon_list</code></td>
<td>
<p>object from taxon_mwu_list() function</p>
</td></tr>
<tr><td><code id="get_link_guilds_+3A_funguilds">funguilds</code></td>
<td>
<p>object from get_funguilds() function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>link_guilds Object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: link_guilds &lt;- get_link_guilds(taxon_list, funguilds)
</code></pre>

<hr>
<h2 id='get_taxon_list_drawer'>get_taxon_list_drawer</h2><span id='topic+get_taxon_list_drawer'></span>

<h3>Description</h3>

<p>get taxonomic list drawer
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_taxon_list_drawer(taxon_list)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_taxon_list_drawer_+3A_taxon_list">taxon_list</code></td>
<td>
<p>object from taxon_mwu_list() function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>taxon_list_drawer Object and &quot;taxon_list_drawer_input.txt&quot; file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: taxon_list_drawer &lt;- get_taxon_list_drawer(taxon_list)
</code></pre>

<hr>
<h2 id='heatmap_condition'>heatmap_condition</h2><span id='topic+heatmap_condition'></span>

<h3>Description</h3>

<p>For Clustering step. Fill directly the annotation_col variable of the pheatmap() function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmap_condition(nothing)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heatmap_condition_+3A_nothing">nothing</code></td>
<td>
<p>It's important not to write anything between the brackets, all inputs will be adapted automatically.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Fill directly the annotation_col variable of the pheatmap() function
</p>

<hr>
<h2 id='heatmap_data_dasva'>heatmap_data_dasva</h2><span id='topic+heatmap_data_dasva'></span>

<h3>Description</h3>

<p>For Clustering step. Create the log2.norm.counts object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmap_data_dasva(nothing)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heatmap_data_dasva_+3A_nothing">nothing</code></td>
<td>
<p>It's important not to write anything between the brackets, all inputs will be adapted automatically.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Create the log2.norm.counts object.
</p>

<hr>
<h2 id='heatmap_samples_hclust'>heatmap_samples_hclust</h2><span id='topic+heatmap_samples_hclust'></span>

<h3>Description</h3>

<p>Adapt hclust for heatmap sample to sample analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmap_samples_hclust(nothing)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heatmap_samples_hclust_+3A_nothing">nothing</code></td>
<td>
<p>It's important not to write anything between the brackets, all inputs will be adapted automatically.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>hclust object for the heatmap.2() function
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: hc &lt;- heatmap_samples_hclust()
</code></pre>

<hr>
<h2 id='heatmap_samples_matrix'>heatmap_samples_matrix</h2><span id='topic+heatmap_samples_matrix'></span>

<h3>Description</h3>

<p>Adapt samples matrix for heatmap sample to sample analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmap_samples_matrix(nothing)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heatmap_samples_matrix_+3A_nothing">nothing</code></td>
<td>
<p>It's important not to write anything between the brackets, all inputs will be adapted automatically.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>samples matrix object for the heatmap.2() function
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: mat &lt;- heatmap_samples_matrix()
</code></pre>

<hr>
<h2 id='heatmap_taxo'>heatmap_taxo</h2><span id='topic+heatmap_taxo'></span>

<h3>Description</h3>

<p>Adding taxonomy in the pheatmap plot, instead of ASVs codes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmap_taxo(nothing)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heatmap_taxo_+3A_nothing">nothing</code></td>
<td>
<p>It's important not to write anything between the brackets, all inputs will be adapted automatically.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>log2.norm.counts_taxo used fro adding taxonomy in the pheatmap plot, instead of ASVs codes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: log2.norm.counts_taxo &lt;- heatmap_taxo()
</code></pre>

<hr>
<h2 id='input_global_analysis'>input_global_analysis</h2><span id='topic+input_global_analysis'></span>

<h3>Description</h3>

<p>Input files creation for each condition for Global analysis by Taxon_MWU analysis from targeted analysis (I)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>input_global_analysis(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="input_global_analysis_+3A_x">x</code></td>
<td>
<p>Object from the Differential ASV abundance (DASVA) analysis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Input Object for Global analysis by Taxon_MWU analysis from targeted analysis (I)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: input_global_analysis(res_forest_vs_long_fallow)
</code></pre>

<hr>
<h2 id='Metazoan'>Metazoan</h2><span id='topic+Metazoan'></span>

<h3>Description</h3>

<p>This function create a new folder named Metazoan and set your working directory into this folder. Please, run setwd(&quot;Metazoan&quot;) after this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Metazoan(nothing)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Metazoan_+3A_nothing">nothing</code></td>
<td>
<p>It's important not to write anything between the brackets, a new folder named Metazoan will be created and your working directory will be set into this folder, depending of the selected Kingdom.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new folder named Metazoan will be created and your working directory will be set into this folder, depending of the selected Kingdom.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: Metazoan()
# plaese, run setwd("Metazoan") after this function.
</code></pre>

<hr>
<h2 id='move_files'>move_files</h2><span id='topic+move_files'></span>

<h3>Description</h3>

<p>Move the file in the good folders. Depending on the previous Kingdom selection (e.g., Fungi 'Fungi()', Bacteria 'Bacteria()', etc.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>move_files(nothing)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="move_files_+3A_nothing">nothing</code></td>
<td>
<p>It's important not to write anything between the brackets, files will move in the good folders, depending of the selected Kingdom before.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Move the file in the good folders.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: move_files()
</code></pre>

<hr>
<h2 id='mwuTest'>mwuTest</h2><span id='topic+mwuTest'></span>

<h3>Description</h3>

<p>Mann-Whitney U Test from RBGOA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mwuTest(gotable, Alternative)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mwuTest_+3A_gotable">gotable</code></td>
<td>
<p>from gomwuStats from RBGOA</p>
</td></tr>
<tr><td><code id="mwuTest_+3A_alternative">Alternative</code></td>
<td>
<p>from gomwuStats from RBGOA</p>
</td></tr>
</table>


<h3>Value</h3>

<p>mwuTest
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: mwuTest()
</code></pre>

<hr>
<h2 id='PCA_data_dasva'>PCA_data_dasva</h2><span id='topic+PCA_data_dasva'></span>

<h3>Description</h3>

<p>Compute the PCA (Pincipal Component Analysis) data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PCA_data_dasva(nothing)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PCA_data_dasva_+3A_nothing">nothing</code></td>
<td>
<p>It's important not to write anything between the brackets, all inputs will be adapted automatically.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data. The PCA (Pincipal Component Analysis) data.
</p>

<hr>
<h2 id='plotDispASVs'>plotDispASVs</h2><span id='topic+plotDispASVs'></span>

<h3>Description</h3>

<p>Create a plot of Dispersion ASV
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDispASVs(
  object,
  ymin,
  CV = FALSE,
  genecol = "black",
  fitcol = "red",
  finalcol = "dodgerblue",
  legend = TRUE,
  xlab,
  ylab,
  log = "xy",
  cex = 0.45,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotDispASVs_+3A_object">object</code></td>
<td>
<p>Corresponding to the DASVA (Differential ASV abundance) object</p>
</td></tr>
<tr><td><code id="plotDispASVs_+3A_ymin">ymin</code></td>
<td>
<p>ymin</p>
</td></tr>
<tr><td><code id="plotDispASVs_+3A_cv">CV</code></td>
<td>
<p>CV</p>
</td></tr>
<tr><td><code id="plotDispASVs_+3A_genecol">genecol</code></td>
<td>
<p>genecol</p>
</td></tr>
<tr><td><code id="plotDispASVs_+3A_fitcol">fitcol</code></td>
<td>
<p>fitcol</p>
</td></tr>
<tr><td><code id="plotDispASVs_+3A_finalcol">finalcol</code></td>
<td>
<p>finalcol</p>
</td></tr>
<tr><td><code id="plotDispASVs_+3A_legend">legend</code></td>
<td>
<p>legend</p>
</td></tr>
<tr><td><code id="plotDispASVs_+3A_xlab">xlab</code></td>
<td>
<p>xlab</p>
</td></tr>
<tr><td><code id="plotDispASVs_+3A_ylab">ylab</code></td>
<td>
<p>ylab</p>
</td></tr>
<tr><td><code id="plotDispASVs_+3A_log">log</code></td>
<td>
<p>log</p>
</td></tr>
<tr><td><code id="plotDispASVs_+3A_cex">cex</code></td>
<td>
<p>cex</p>
</td></tr>
<tr><td><code id="plotDispASVs_+3A_...">...</code></td>
<td>
<p>...</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot of Dispersion ASV
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: plotDispASVs(dasva)
</code></pre>

<hr>
<h2 id='plotMA.dasva'>plotMA.dasva</h2><span id='topic+plotMA.dasva'></span>

<h3>Description</h3>

<p>Custom MA plots for the Differential ASV abundance (DASVA) analysis. defining a new function to plot all ASVs and not only log2FoldChange &gt; 2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotMA.dasva(
  object,
  alpha,
  main = "",
  xlab = "mean of normalized counts",
  ylim,
  MLE = FALSE,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotMA.dasva_+3A_object">object</code></td>
<td>
<p>Object from the Differential ASV abundance (DASVA) analysis</p>
</td></tr>
<tr><td><code id="plotMA.dasva_+3A_alpha">alpha</code></td>
<td>
<p>alpha</p>
</td></tr>
<tr><td><code id="plotMA.dasva_+3A_main">main</code></td>
<td>
<p>main</p>
</td></tr>
<tr><td><code id="plotMA.dasva_+3A_xlab">xlab</code></td>
<td>
<p>xlab</p>
</td></tr>
<tr><td><code id="plotMA.dasva_+3A_ylim">ylim</code></td>
<td>
<p>ylim</p>
</td></tr>
<tr><td><code id="plotMA.dasva_+3A_mle">MLE</code></td>
<td>
<p>MLE</p>
</td></tr>
<tr><td><code id="plotMA.dasva_+3A_verbose">verbose</code></td>
<td>
<p>verbose</p>
</td></tr>
<tr><td><code id="plotMA.dasva_+3A_...">...</code></td>
<td>
<p>...</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A MA plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: plotMA.dasva(rXXX, main="XXX", ylim=c(-20,20))
</code></pre>

<hr>
<h2 id='plotPCA.san'>plotPCA.san</h2><span id='topic+plotPCA.san'></span>

<h3>Description</h3>

<p>Custom plotPCA function to plot PC1 et PC3
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPCA.san(object, intgroup = "condition", ntop = 500, returnData = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotPCA.san_+3A_object">object</code></td>
<td>
<p>An object use for the PCA</p>
</td></tr>
<tr><td><code id="plotPCA.san_+3A_intgroup">intgroup</code></td>
<td>
<p>intgroup</p>
</td></tr>
<tr><td><code id="plotPCA.san_+3A_ntop">ntop</code></td>
<td>
<p>ntop</p>
</td></tr>
<tr><td><code id="plotPCA.san_+3A_returndata">returnData</code></td>
<td>
<p>returnData</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A PCA
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: plotPCA.san(object)
</code></pre>

<hr>
<h2 id='plotSparsityASV'>plotSparsityASV</h2><span id='topic+plotSparsityASV'></span>

<h3>Description</h3>

<p>Create a plot of Sparsity ASV
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSparsityASV(x, normalized = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotSparsityASV_+3A_x">x</code></td>
<td>
<p>Corresponding to the DASVA (Differential ASV abundance) object</p>
</td></tr>
<tr><td><code id="plotSparsityASV_+3A_normalized">normalized</code></td>
<td>
<p>normalized</p>
</td></tr>
<tr><td><code id="plotSparsityASV_+3A_...">...</code></td>
<td>
<p>...</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot of Sparsity ASV
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: plotSparsityASV(dasva)
</code></pre>

<hr>
<h2 id='samplesInfo'>samplesInfo</h2><span id='topic+samplesInfo'></span>

<h3>Description</h3>

<p>Imports conditions information from your SampleSheet_comparison.txt file, with focus on samplesInfo.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>samplesInfo(nothing)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="samplesInfo_+3A_nothing">nothing</code></td>
<td>
<p>It's important not to write anything between the brackets, comparisons will create themselves.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame with conditions information from your SampleSheet_comparison.txt file, with focus on samplesInfo.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: samplesInfo &lt;- samplesInfo()
</code></pre>

<hr>
<h2 id='target_file'>target_file</h2><span id='topic+target_file'></span>

<h3>Description</h3>

<p>Imports conditions information from your SampleSheet_comparison.txt file, with focus on iput files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>target_file(nothing)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="target_file_+3A_nothing">nothing</code></td>
<td>
<p>It's important not to write anything between the brackets, comparisons will create themselves.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame with conditions information from your SampleSheet_comparison.txt file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: target_file &lt;- target_file()
</code></pre>

<hr>
<h2 id='taxon_mwu_list'>taxon_mwu_list</h2><span id='topic+taxon_mwu_list'></span>

<h3>Description</h3>

<p>taxon Mann-Whitney U list for taxonomic analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxon_mwu_list(
  inFile,
  goAnnotations,
  goDivision,
  level1 = 0.1,
  level2 = 0.05,
  level3 = 0.01,
  absValue = -log(0.05, 10),
  adjusted = TRUE,
  txtsize = 1,
  font.family = "sans",
  treeHeight = 0.5,
  colors = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxon_mwu_list_+3A_infile">inFile</code></td>
<td>
<p>inFile - results object from the DASVA analysis</p>
</td></tr>
<tr><td><code id="taxon_mwu_list_+3A_goannotations">goAnnotations</code></td>
<td>
<p>parallel to goAnnotations from gomwuStats from RBGOA. Here, &quot;database_bacteria_package_all.tab&quot; if Bacteria, &quot;database_fungi_package_all.tab&quot; if Fungi</p>
</td></tr>
<tr><td><code id="taxon_mwu_list_+3A_godivision">goDivision</code></td>
<td>
<p>parallel to goAnnotations from gomwuStats from RBGOA. Here, &quot;TR&quot; = taxonomic Rank, don't change this</p>
</td></tr>
<tr><td><code id="taxon_mwu_list_+3A_level1">level1</code></td>
<td>
<p>level1</p>
</td></tr>
<tr><td><code id="taxon_mwu_list_+3A_level2">level2</code></td>
<td>
<p>level2</p>
</td></tr>
<tr><td><code id="taxon_mwu_list_+3A_level3">level3</code></td>
<td>
<p>level3</p>
</td></tr>
<tr><td><code id="taxon_mwu_list_+3A_absvalue">absValue</code></td>
<td>
<p>absValue</p>
</td></tr>
<tr><td><code id="taxon_mwu_list_+3A_adjusted">adjusted</code></td>
<td>
<p>adjusted</p>
</td></tr>
<tr><td><code id="taxon_mwu_list_+3A_txtsize">txtsize</code></td>
<td>
<p>txtsize</p>
</td></tr>
<tr><td><code id="taxon_mwu_list_+3A_font.family">font.family</code></td>
<td>
<p>font.family</p>
</td></tr>
<tr><td><code id="taxon_mwu_list_+3A_treeheight">treeHeight</code></td>
<td>
<p>treeHeight</p>
</td></tr>
<tr><td><code id="taxon_mwu_list_+3A_colors">colors</code></td>
<td>
<p>colors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List for the statistical analysis for taxonomic rank
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: taxon_list &lt;- taxon_mwu_list(input, ...)
</code></pre>

<hr>
<h2 id='taxon_mwuPlot'>taxon_mwuPlot</h2><span id='topic+taxon_mwuPlot'></span>

<h3>Description</h3>

<p>taxon mwuPlot for taxonomic analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxon_mwuPlot(
  inFile,
  goAnnotations,
  goDivision,
  level1 = 0.1,
  level2 = 0.05,
  level3 = 0.01,
  absValue = -log(0.05, 10),
  adjusted = TRUE,
  txtsize = 1,
  font.family = "sans",
  treeHeight = 0.5,
  colors = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxon_mwuPlot_+3A_infile">inFile</code></td>
<td>
<p>inFile - results object from the DASVA analysis</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_+3A_goannotations">goAnnotations</code></td>
<td>
<p>parallel to goAnnotations from gomwuStats from RBGOA. Here, &quot;database_bacteria_package_all.tab&quot; if Bacteria, &quot;database_fungi_package_all.tab&quot; if Fungi</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_+3A_godivision">goDivision</code></td>
<td>
<p>parallel to goAnnotations from gomwuStats from RBGOA. Here, &quot;TR&quot; = taxonomic Rank, don't change this</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_+3A_level1">level1</code></td>
<td>
<p>level1</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_+3A_level2">level2</code></td>
<td>
<p>level2</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_+3A_level3">level3</code></td>
<td>
<p>level3</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_+3A_absvalue">absValue</code></td>
<td>
<p>absValue</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_+3A_adjusted">adjusted</code></td>
<td>
<p>adjusted</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_+3A_txtsize">txtsize</code></td>
<td>
<p>txtsize</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_+3A_font.family">font.family</code></td>
<td>
<p>font.family</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_+3A_treeheight">treeHeight</code></td>
<td>
<p>treeHeight</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_+3A_colors">colors</code></td>
<td>
<p>colors</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_+3A_verbose">verbose</code></td>
<td>
<p>verbose</p>
</td></tr>
</table>


<h3>Value</h3>

<p>taxon mwuPlot and goods &quot;Table_02_taxon_mwuPlot.txt&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: taxon_mwuPlot(input,...)
</code></pre>

<hr>
<h2 id='taxon_mwuPlot_guilds'>taxon_mwuPlot_guilds</h2><span id='topic+taxon_mwuPlot_guilds'></span>

<h3>Description</h3>

<p>taxon Mann-Whitney U Plot with Fungi Guilds added
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxon_mwuPlot_guilds(
  inFile,
  goAnnotations,
  goDivision,
  level1 = 0.1,
  level2 = 0.05,
  level3 = 0.01,
  absValue = -log(0.05, 10),
  adjusted = TRUE,
  txtsize = 1,
  font.family = "sans",
  treeHeight = 0.5,
  colors = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxon_mwuPlot_guilds_+3A_infile">inFile</code></td>
<td>
<p>inFile - results object from the DASVA analysis</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_guilds_+3A_goannotations">goAnnotations</code></td>
<td>
<p>parallel to goAnnotations from gomwuStats from RBGOA. Here, &quot;database_bacteria_package_all.tab&quot; if Bacteria, &quot;database_fungi_package_all.tab&quot; if Fungi</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_guilds_+3A_godivision">goDivision</code></td>
<td>
<p>parallel to goAnnotations from gomwuStats from RBGOA. Here, &quot;TR&quot; = taxonomic Rank, don't change this</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_guilds_+3A_level1">level1</code></td>
<td>
<p>level1</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_guilds_+3A_level2">level2</code></td>
<td>
<p>level2</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_guilds_+3A_level3">level3</code></td>
<td>
<p>level3</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_guilds_+3A_absvalue">absValue</code></td>
<td>
<p>absValue</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_guilds_+3A_adjusted">adjusted</code></td>
<td>
<p>adjusted</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_guilds_+3A_txtsize">txtsize</code></td>
<td>
<p>txtsize</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_guilds_+3A_font.family">font.family</code></td>
<td>
<p>font.family</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_guilds_+3A_treeheight">treeHeight</code></td>
<td>
<p>treeHeight</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_guilds_+3A_colors">colors</code></td>
<td>
<p>colors</p>
</td></tr>
<tr><td><code id="taxon_mwuPlot_guilds_+3A_verbose">verbose</code></td>
<td>
<p>verbose</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List for the statistical analysis for taxonomic rank
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: taxon_mwuPlot_guilds(input, ...)
</code></pre>

<hr>
<h2 id='taxon_mwuStats'>taxon_mwuStats</h2><span id='topic+taxon_mwuStats'></span>

<h3>Description</h3>

<p>mwuStats from RBGOA adapted for taxonomic analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxon_mwuStats(
  input,
  goDatabase,
  goAnnotations,
  goDivision,
  Alternative = "t",
  adjust.multcomp = "BH",
  clusterCutHeight = 0.25,
  largest = 0.1,
  smallest = 5,
  perlPath = "perl",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxon_mwuStats_+3A_input">input</code></td>
<td>
<p>input</p>
</td></tr>
<tr><td><code id="taxon_mwuStats_+3A_godatabase">goDatabase</code></td>
<td>
<p>goDatabase</p>
</td></tr>
<tr><td><code id="taxon_mwuStats_+3A_goannotations">goAnnotations</code></td>
<td>
<p>goAnnotations</p>
</td></tr>
<tr><td><code id="taxon_mwuStats_+3A_godivision">goDivision</code></td>
<td>
<p>goDivision</p>
</td></tr>
<tr><td><code id="taxon_mwuStats_+3A_alternative">Alternative</code></td>
<td>
<p>Alternative</p>
</td></tr>
<tr><td><code id="taxon_mwuStats_+3A_adjust.multcomp">adjust.multcomp</code></td>
<td>
<p>adjust.multcomp</p>
</td></tr>
<tr><td><code id="taxon_mwuStats_+3A_clustercutheight">clusterCutHeight</code></td>
<td>
<p>clusterCutHeight</p>
</td></tr>
<tr><td><code id="taxon_mwuStats_+3A_largest">largest</code></td>
<td>
<p>largest</p>
</td></tr>
<tr><td><code id="taxon_mwuStats_+3A_smallest">smallest</code></td>
<td>
<p>smallest</p>
</td></tr>
<tr><td><code id="taxon_mwuStats_+3A_perlpath">perlPath</code></td>
<td>
<p>perlPath</p>
</td></tr>
<tr><td><code id="taxon_mwuStats_+3A_verbose">verbose</code></td>
<td>
<p>verbose</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Statistical analysis for taxonomic rank
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: taxon_mwuStats(input, ...)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
