<!DOCTYPE html><html><head><title>Help for package AnanseSeurat</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {AnanseSeurat}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#%&gt;%'><p>Pipe operator</p></a></li>
<li><a href='#config_scANANSE'><p>config_scANANSE</p></a></li>
<li><a href='#DEGS_scANANSE'><p>DEGS_scANANSE</p></a></li>
<li><a href='#export_ATAC_maelstrom'><p>export_seurat_Maelstrom</p></a></li>
<li><a href='#export_ATAC_scANANSE'><p>export_ATAC_scANANSE</p></a></li>
<li><a href='#export_CPM_scANANSE'><p>export_CPM_scANANSE</p></a></li>
<li><a href='#Factor_Motif_Plot'><p>Factor_Motif_Plot</p></a></li>
<li><a href='#import_seurat_maelstrom'><p>import_seurat_Maelstrom</p></a></li>
<li><a href='#import_seurat_scANANSE'><p>import_seurat_scANANSE</p></a></li>
<li><a href='#Maelstrom_Motif2TF'><p>Maelstrom_Motif2TF</p></a></li>
<li><a href='#per_cluster_df'><p>per_cluster_df</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Construct ANANSE GRN-Analysis Seurat</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Enables gene regulatory network (GRN) analysis on single cell clusters,
    using the GRN analysis software 'ANANSE', Xu et al.(2021) &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgkab598">doi:10.1093/nar/gkab598</a>&gt;.
    Export data from 'Seurat' objects, for GRN analysis by 'ANANSE' 
    implemented in 'snakemake'. Finally, incorporate results for visualization 
    and interpretation.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License (&ge; 2)</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/JGASmits/AnanseSeurat/issues">https://github.com/JGASmits/AnanseSeurat/issues</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/JGASmits/AnanseSeurat/">https://github.com/JGASmits/AnanseSeurat/</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, ggplot2, ggpubr, magrittr, patchwork, png, purrr,
rlang, Seurat, stringr, utils,</td>
</tr>
<tr>
<td>Suggests:</td>
<td>covr, knitr, rmarkdown, Signac, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.50)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-11 20:53:08 UTC; jossm</td>
</tr>
<tr>
<td>Author:</td>
<td>Jos Smits <a href="https://orcid.org/0000-0001-5858-3905"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre, cph],
  Huiqing Jo Zhou <a href="https://orcid.org/0000-0002-2434-3986"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, fnd, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jos Smits &lt;Jsmits@science.ru.nl&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-11 21:43:17 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling <code>rhs(lhs)</code>.
</p>

<hr>
<h2 id='config_scANANSE'>config_scANANSE</h2><span id='topic+config_scANANSE'></span>

<h3>Description</h3>

<p>This functions generates a sample file and config file for running Anansnake based on the seurat object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>config_scANANSE(
  seurat_object,
  output_dir,
  min_cells = 50,
  cluster_id = "seurat_clusters",
  genome = "./scANANSE/data/hg38",
  additional_contrasts = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="config_scANANSE_+3A_seurat_object">seurat_object</code></td>
<td>
<p>seurat object</p>
</td></tr>
<tr><td><code id="config_scANANSE_+3A_output_dir">output_dir</code></td>
<td>
<p>directory where the files are outputted</p>
</td></tr>
<tr><td><code id="config_scANANSE_+3A_min_cells">min_cells</code></td>
<td>
<p>minimum of cells a cluster needs to be exported</p>
</td></tr>
<tr><td><code id="config_scANANSE_+3A_cluster_id">cluster_id</code></td>
<td>
<p>ID used for finding clusters of cells</p>
</td></tr>
<tr><td><code id="config_scANANSE_+3A_genome">genome</code></td>
<td>
<p>genomepy name or location of the genome fastq file</p>
</td></tr>
<tr><td><code id="config_scANANSE_+3A_additional_contrasts">additional_contrasts</code></td>
<td>
<p>additional contrasts to add between clusters within cluster_ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None, outputs snakemake config file in the output directory
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sce_small &lt;- readRDS(system.file("extdata","sce_small.Rds",package = 'AnanseSeurat'))
config_scANANSE(sce_small, min_cells = 2, output_dir = tempdir())
</code></pre>

<hr>
<h2 id='DEGS_scANANSE'>DEGS_scANANSE</h2><span id='topic+DEGS_scANANSE'></span>

<h3>Description</h3>

<p>Calculate the differential genes needed for ananse influence
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DEGS_scANANSE(
  seurat_object,
  output_dir,
  min_cells = 50,
  cluster_id = "seurat_clusters",
  genome = "./scANANSE/data/hg38",
  RNA_count_assay = "RNA",
  additional_contrasts = "None"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DEGS_scANANSE_+3A_seurat_object">seurat_object</code></td>
<td>
<p>seurat object</p>
</td></tr>
<tr><td><code id="DEGS_scANANSE_+3A_output_dir">output_dir</code></td>
<td>
<p>directory where the files are outputted</p>
</td></tr>
<tr><td><code id="DEGS_scANANSE_+3A_min_cells">min_cells</code></td>
<td>
<p>minimum of cells a cluster needs to be exported</p>
</td></tr>
<tr><td><code id="DEGS_scANANSE_+3A_cluster_id">cluster_id</code></td>
<td>
<p>ID used for finding clusters of cells</p>
</td></tr>
<tr><td><code id="DEGS_scANANSE_+3A_genome">genome</code></td>
<td>
<p>path to the genome folder used for the anansnake config file</p>
</td></tr>
<tr><td><code id="DEGS_scANANSE_+3A_rna_count_assay">RNA_count_assay</code></td>
<td>
<p>assay containing the RNA data</p>
</td></tr>
<tr><td><code id="DEGS_scANANSE_+3A_additional_contrasts">additional_contrasts</code></td>
<td>
<p>additional contrasts to add
between clusters within cluster_ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None, outputs DEG files in the output directory
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sce_small &lt;- readRDS(system.file("extdata","sce_obj_tiny.Rds",package = 'AnanseSeurat'))
DEGS_scANANSE(sce_small, min_cells = 2, output_dir = tempdir())
</code></pre>

<hr>
<h2 id='export_ATAC_maelstrom'>export_seurat_Maelstrom</h2><span id='topic+export_ATAC_maelstrom'></span>

<h3>Description</h3>

<p>normalize and export the peak table of a seurat object based on clusters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_ATAC_maelstrom(
  seurat_object,
  output_dir,
  min_cells = 50,
  ATAC_peak_assay = "peaks",
  cluster_id = "seurat_clusters",
  select_top_rows = TRUE,
  n_top_rows = 1e+05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="export_ATAC_maelstrom_+3A_seurat_object">seurat_object</code></td>
<td>
<p>object</p>
</td></tr>
<tr><td><code id="export_ATAC_maelstrom_+3A_output_dir">output_dir</code></td>
<td>
<p>directory where the files are outputted</p>
</td></tr>
<tr><td><code id="export_ATAC_maelstrom_+3A_min_cells">min_cells</code></td>
<td>
<p>minimum of cells a cluster needs to be exported</p>
</td></tr>
<tr><td><code id="export_ATAC_maelstrom_+3A_atac_peak_assay">ATAC_peak_assay</code></td>
<td>
<p>assay of the seurat object containing the peaks and peakcounts</p>
</td></tr>
<tr><td><code id="export_ATAC_maelstrom_+3A_cluster_id">cluster_id</code></td>
<td>
<p>ID used for finding clusters of cells</p>
</td></tr>
<tr><td><code id="export_ATAC_maelstrom_+3A_select_top_rows">select_top_rows</code></td>
<td>
<p>only output the top variable rows, or all rows if false</p>
</td></tr>
<tr><td><code id="export_ATAC_maelstrom_+3A_n_top_rows">n_top_rows</code></td>
<td>
<p>amount of variable rows to export</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None, outputs maelstrom peak counts table in the output directory
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sce_small &lt;- readRDS(system.file("extdata","sce_small.Rds",package = 'AnanseSeurat'))
config_scANANSE(sce_small, min_cells = 2, output_dir = tempdir())
</code></pre>

<hr>
<h2 id='export_ATAC_scANANSE'>export_ATAC_scANANSE</h2><span id='topic+export_ATAC_scANANSE'></span>

<h3>Description</h3>

<p>This functions exports ATAC values from a seurat object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_ATAC_scANANSE(
  seurat_object,
  output_dir,
  min_cells = 50,
  ATAC_peak_assay = "peaks",
  cluster_id = "seurat_clusters"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="export_ATAC_scANANSE_+3A_seurat_object">seurat_object</code></td>
<td>
<p>object</p>
</td></tr>
<tr><td><code id="export_ATAC_scANANSE_+3A_output_dir">output_dir</code></td>
<td>
<p>directory where the files are outputted</p>
</td></tr>
<tr><td><code id="export_ATAC_scANANSE_+3A_min_cells">min_cells</code></td>
<td>
<p>minimum of cells a cluster needs to be exported</p>
</td></tr>
<tr><td><code id="export_ATAC_scANANSE_+3A_atac_peak_assay">ATAC_peak_assay</code></td>
<td>
<p>assay of the seurat object containing the peaks and peakcounts</p>
</td></tr>
<tr><td><code id="export_ATAC_scANANSE_+3A_cluster_id">cluster_id</code></td>
<td>
<p>ID used for finding clusters of cells</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None, outputs ATAC peak count file in the output directory
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sce_small &lt;- readRDS(system.file("extdata","sce_small.Rds",package = 'AnanseSeurat'))
export_ATAC_scANANSE(sce_small, min_cells = 2, output_dir = tempdir())
</code></pre>

<hr>
<h2 id='export_CPM_scANANSE'>export_CPM_scANANSE</h2><span id='topic+export_CPM_scANANSE'></span>

<h3>Description</h3>

<p>This functions exports CPM values from a seurat object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_CPM_scANANSE(
  seurat_object,
  output_dir,
  min_cells = 50,
  RNA_count_assay = "RNA",
  cluster_id = "seurat_clusters"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="export_CPM_scANANSE_+3A_seurat_object">seurat_object</code></td>
<td>
<p>the seurat object used to export the CPM values from</p>
</td></tr>
<tr><td><code id="export_CPM_scANANSE_+3A_output_dir">output_dir</code></td>
<td>
<p>directory where the files are outputted</p>
</td></tr>
<tr><td><code id="export_CPM_scANANSE_+3A_min_cells">min_cells</code></td>
<td>
<p>minimum of cells a cluster needs to be exported</p>
</td></tr>
<tr><td><code id="export_CPM_scANANSE_+3A_rna_count_assay">RNA_count_assay</code></td>
<td>
<p>assay of the seurat object containing the RNA count data</p>
</td></tr>
<tr><td><code id="export_CPM_scANANSE_+3A_cluster_id">cluster_id</code></td>
<td>
<p>ID used for finding clusters of cells</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None, outputs CPM and counts files in the output directory
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sce_small &lt;- readRDS(system.file("extdata","sce_small.Rds",package = 'AnanseSeurat'))
export_CPM_scANANSE(sce_small, min_cells = 2, output_dir = tempdir())
</code></pre>

<hr>
<h2 id='Factor_Motif_Plot'>Factor_Motif_Plot</h2><span id='topic+Factor_Motif_Plot'></span>

<h3>Description</h3>

<p>plot both expression of a TF, and the motif accessibility of the associated motif. Finally, fetch the motif logo from the Maelstrom directory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Factor_Motif_Plot(
  seurat_object,
  TF_list,
  assay_RNA = "RNA",
  assay_maelstrom = "MotifTFanticor",
  logo_dir = "~/maelstrom/logos",
  col = c("darkred", "white", "darkgrey"),
  dim_reduction = "umap"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Factor_Motif_Plot_+3A_seurat_object">seurat_object</code></td>
<td>
<p>seurat object</p>
</td></tr>
<tr><td><code id="Factor_Motif_Plot_+3A_tf_list">TF_list</code></td>
<td>
<p>list of TFs to plot the expression and linked motif Z-score for</p>
</td></tr>
<tr><td><code id="Factor_Motif_Plot_+3A_assay_rna">assay_RNA</code></td>
<td>
<p>RNA_count_assay assay containing the RNA data</p>
</td></tr>
<tr><td><code id="Factor_Motif_Plot_+3A_assay_maelstrom">assay_maelstrom</code></td>
<td>
<p>maelstrom assay used for zscore vizualization, often either TFcor or TFanticor</p>
</td></tr>
<tr><td><code id="Factor_Motif_Plot_+3A_logo_dir">logo_dir</code></td>
<td>
<p>directory containing motif logos generated by gimme maelstrom</p>
</td></tr>
<tr><td><code id="Factor_Motif_Plot_+3A_col">col</code></td>
<td>
<p>colours used for zscore vizualization</p>
</td></tr>
<tr><td><code id="Factor_Motif_Plot_+3A_dim_reduction">dim_reduction</code></td>
<td>
<p>dimensionality reduction method to use</p>
</td></tr>
</table>


<h3>Value</h3>

<p>patchwork plot containing a expression dimreduction plot, a maelstrom motif score dimreduction plot, and a png image of the motif
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sce_small &lt;- readRDS(system.file("extdata","sce_small.Rds",package = 'AnanseSeurat'))
logos_dir_path &lt;- system.file("extdata","maelstrom","logos",package = 'AnanseSeurat')
sce_small &lt;- Factor_Motif_Plot(sce_small,
  c('gene1', 'gene2'),
  dim_reduction = 'pca',
  logo_dir = logos_dir_path)
</code></pre>

<hr>
<h2 id='import_seurat_maelstrom'>import_seurat_Maelstrom</h2><span id='topic+import_seurat_maelstrom'></span>

<h3>Description</h3>

<p>load Maelstrom enriched motifs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_seurat_maelstrom(
  seurat_object,
  cluster_id = "seurat_clusters",
  maelstrom_file = "~/final.out.txt",
  return_df = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_seurat_maelstrom_+3A_seurat_object">seurat_object</code></td>
<td>
<p>object</p>
</td></tr>
<tr><td><code id="import_seurat_maelstrom_+3A_cluster_id">cluster_id</code></td>
<td>
<p>ID used for finding clusters of cells</p>
</td></tr>
<tr><td><code id="import_seurat_maelstrom_+3A_maelstrom_file">maelstrom_file</code></td>
<td>
<p>maelstrom final.out.txt file</p>
</td></tr>
<tr><td><code id="import_seurat_maelstrom_+3A_return_df">return_df</code></td>
<td>
<p>return both the seurat object and a dataframe with maelstrom scores as a list</p>
</td></tr>
</table>


<h3>Value</h3>

<p>seurat object with the maelstrom motif scores addes as an assay
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sce_small &lt;- readRDS(system.file("extdata","sce_small.Rds",package = 'AnanseSeurat'))
maelstromfile_path &lt;- system.file("extdata","maelstrom","final.out.txt",package = 'AnanseSeurat')
sce_small &lt;- import_seurat_maelstrom(sce_small, maelstrom_file = maelstromfile_path)
</code></pre>

<hr>
<h2 id='import_seurat_scANANSE'>import_seurat_scANANSE</h2><span id='topic+import_seurat_scANANSE'></span>

<h3>Description</h3>

<p>import the influences from a anansnake directory into a seurat object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_seurat_scANANSE(
  seurat_object,
  cluster_id = "seurat_clusters",
  anansnake_inf_dir = "None",
  return_df = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_seurat_scANANSE_+3A_seurat_object">seurat_object</code></td>
<td>
<p>seurat object</p>
</td></tr>
<tr><td><code id="import_seurat_scANANSE_+3A_cluster_id">cluster_id</code></td>
<td>
<p>ID used for finding clusters of cells</p>
</td></tr>
<tr><td><code id="import_seurat_scANANSE_+3A_anansnake_inf_dir">anansnake_inf_dir</code></td>
<td>
<p>influence directory generated by anansnake</p>
</td></tr>
<tr><td><code id="import_seurat_scANANSE_+3A_return_df">return_df</code></td>
<td>
<p>return both the seurat object and a dataframe with influence scores as a list</p>
</td></tr>
</table>


<h3>Value</h3>

<p>seurat object with the influence scores addes as an assay
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sce_small &lt;- readRDS(system.file("extdata","sce_small.Rds",package = 'AnanseSeurat'))
infdir &lt;- system.file("extdata","influence",package = 'AnanseSeurat')
sce_small &lt;- import_seurat_scANANSE(sce_small, anansnake_inf_dir = infdir)
</code></pre>

<hr>
<h2 id='Maelstrom_Motif2TF'>Maelstrom_Motif2TF</h2><span id='topic+Maelstrom_Motif2TF'></span>

<h3>Description</h3>

<p>create motif-factor links &amp; export tables for printing motif score alongside its binding factor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Maelstrom_Motif2TF(
  seurat_object,
  mot_mat = NULL,
  m2f_df = NULL,
  cluster_id = "seurat_clusters",
  maelstrom_dir = "./maelstrom/",
  combine_motifs = "means",
  RNA_expression_assay = "RNA",
  RNA_expression_slot = "data",
  expr_tresh = 10,
  cor_tresh = 0.3,
  curated_motifs = FALSE,
  cor_method = "pearson",
  return_df = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Maelstrom_Motif2TF_+3A_seurat_object">seurat_object</code></td>
<td>
<p>object</p>
</td></tr>
<tr><td><code id="Maelstrom_Motif2TF_+3A_mot_mat">mot_mat</code></td>
<td>
<p>motif_matrix, if not provided extracts one from the single cell object from the maelstrom assay</p>
</td></tr>
<tr><td><code id="Maelstrom_Motif2TF_+3A_m2f_df">m2f_df</code></td>
<td>
<p>motif to factor dataframe, if not provided extracts from the maelstrom directory</p>
</td></tr>
<tr><td><code id="Maelstrom_Motif2TF_+3A_cluster_id">cluster_id</code></td>
<td>
<p>ID used for finding clusters of cells</p>
</td></tr>
<tr><td><code id="Maelstrom_Motif2TF_+3A_maelstrom_dir">maelstrom_dir</code></td>
<td>
<p>directory where the GimmeMotifs m2f table is stored</p>
</td></tr>
<tr><td><code id="Maelstrom_Motif2TF_+3A_combine_motifs">combine_motifs</code></td>
<td>
<p>means (take mean multiple motifscores), max_var (take motif with highest variance), or max_cor (take motif with best correlation to gene expression)</p>
</td></tr>
<tr><td><code id="Maelstrom_Motif2TF_+3A_rna_expression_assay">RNA_expression_assay</code></td>
<td>
<p>Seurat assay containing factor expression info</p>
</td></tr>
<tr><td><code id="Maelstrom_Motif2TF_+3A_rna_expression_slot">RNA_expression_slot</code></td>
<td>
<p>slot within assay used for calculating mean factor expression per cluster</p>
</td></tr>
<tr><td><code id="Maelstrom_Motif2TF_+3A_expr_tresh">expr_tresh</code></td>
<td>
<p>minimum sum of gene counts over all cells in RNA_expression_assay to filter genes by</p>
</td></tr>
<tr><td><code id="Maelstrom_Motif2TF_+3A_cor_tresh">cor_tresh</code></td>
<td>
<p>minimum value of to filter the cor() output by</p>
</td></tr>
<tr><td><code id="Maelstrom_Motif2TF_+3A_curated_motifs">curated_motifs</code></td>
<td>
<p>use only curated motifs (T), or all motifs in the database (F)</p>
</td></tr>
<tr><td><code id="Maelstrom_Motif2TF_+3A_cor_method">cor_method</code></td>
<td>
<p>specify one of the cor() methods</p>
</td></tr>
<tr><td><code id="Maelstrom_Motif2TF_+3A_return_df">return_df</code></td>
<td>
<p>return both the seurat object and two dataframes with maelstrom scores and expression values as a list</p>
</td></tr>
</table>


<h3>Value</h3>

<p>seurat object with two assays added, MotifTFcor for TFs with positive correlation to the linked motif, and MotifTFanticor for TFs with positive correlation to the linked motif
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sce_small &lt;- readRDS(system.file("extdata","sce_small.Rds",package = 'AnanseSeurat'))
maelstrom_dir_path &lt;- system.file("extdata","maelstrom",package = 'AnanseSeurat')
sce_small &lt;- Maelstrom_Motif2TF(sce_small, maelstrom_dir = maelstrom_dir_path)
</code></pre>

<hr>
<h2 id='per_cluster_df'>per_cluster_df</h2><span id='topic+per_cluster_df'></span>

<h3>Description</h3>

<p>generate a table of the assay score averages per cluster identifier cell
</p>


<h3>Usage</h3>

<pre><code class='language-R'>per_cluster_df(
  seurat_object,
  assay = "influence",
  cluster_id = "seurat_clusters"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="per_cluster_df_+3A_seurat_object">seurat_object</code></td>
<td>
<p>seurat object</p>
</td></tr>
<tr><td><code id="per_cluster_df_+3A_assay">assay</code></td>
<td>
<p>assay containing influence or motif scores generated from cluster pseudobulk</p>
</td></tr>
<tr><td><code id="per_cluster_df_+3A_cluster_id">cluster_id</code></td>
<td>
<p>ID used for finding clusters of cells</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe with assay scores, concatinating cells from each per cluster
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sce_small &lt;- readRDS(system.file("extdata","sce_small.Rds",package = 'AnanseSeurat'))
df &lt;- per_cluster_df(sce_small)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
