<!DOCTYPE html><html><head><title>Help for package anapuce</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {anapuce}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#apo'><p>Apo AI data</p></a></li>
<li><a href='#DiffAnalysis'><p>Differential Analysis for paired data</p></a></li>
<li><a href='#DiffAnalysis.unpaired'><p> Differential Analysis for unpaired data</p></a></li>
<li><a href='#est.varmixt'><p>Variance Mixture Estimation</p></a></li>
<li><a href='#fdr.estimate.eta0'><p>Estimate the Proportion of Null p-Values</p></a></li>
<li><a href='#FileChoice'><p> Function to choose the files to analyze</p></a></li>
<li><a href='#filterByDefault'><p> Filter applied on data before normalisation</p></a></li>
<li><a href='#fonctionMobile'><p> fonctionMobile</p></a></li>
<li><a href='#Intensity.Norm'><p> Function to compute normalized intensity for each channel</p></a></li>
<li><a href='#LBI'><p> Label Bias Index</p></a></li>
<li><a href='#lissageMM'><p> Mean Average Smoothing</p></a></li>
<li><a href='#LocalFDR'><p> Local FDR</p></a></li>
<li><a href='#MeanBySpot'><p> Mean By Spot</p></a></li>
<li><a href='#mixgamma'><p> Gamma mixture</p></a></li>
<li><a href='#normalisation'><p> Function to normalize microarray data</p></a></li>
<li><a href='#rowVars'><p> Form row variances</p></a></li>
<li><a href='#spleen'><p>Spleen Data set</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Tools for Microarray Data Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>2.3</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions for normalisation, differentially analysis of microarray data and local False Discovery Rate.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-02-05 14:36:42 UTC; aubert</td>
</tr>
<tr>
<td>Author:</td>
<td>Julie Aubert [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Julie Aubert &lt;julie.aubert@agroparistech.fr&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-02-05 15:13:20 UTC</td>
</tr>
</table>
<hr>
<h2 id='apo'>Apo AI data </h2><span id='topic+apo'></span>

<h3>Description</h3>

<p>An exemple data set for unpaired data analysis.
A dataframe with the normalized Apo data set as used in the original article.
</p>


<h3>Format</h3>

<p>apo is a dataframe. The first column contains gene names. The 8 columns cond1.1 to cond1.8 contain normalized normal mice measurements
and the 8 columns cond2.1 to cond2.8 correspond to normalized KO mice measurements
</p>


<h3>References</h3>

<p>M.J. Callow, S. Dudoit, E.L. Gong, T.P. Speed, and E.M. Rubin. Microarray expression profiling
identifies genes with altered expression in hdl-deficien mice. Genome Res., 10(12) : 2022-9, 2000
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+DiffAnalysis.unpaired">DiffAnalysis.unpaired</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(apo)
# resApo &lt;- DiffAnalysis.unpaired(apo,n=1,ind.array=1:8,varmixt.meth=TRUE)
# Histogramm of the raw-pvalues
# hist(resApo$PValueVM)
</code></pre>

<hr>
<h2 id='DiffAnalysis'>Differential Analysis for paired data </h2><span id='topic+DiffAnalysis'></span>

<h3>Description</h3>

<p>Performs the differential analysis from normalised paired microarray data according to different ways of variance modelling
and computes lists of differentially expressed genes according different multiple testing procedures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DiffAnalysis(fileIN = "resNorm.txt", n = 3, ind.array = 1:2, name.A = "A",
name.M = "M.norm", fileOUT = "ListOfGenes.txt",
fileDelete = "GenesOutOfAnalysis.txt", procs = c("bonferroni", "BH"),
alpha = c(0.05, 0.05), dyeswap = TRUE, indDS = c(2), fileID = NULL,
function.trt = NULL, by.var = "ID", varmixt.meth = TRUE, header = TRUE,
sep = "\t", sep.write = "\t", dec.write = ".", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DiffAnalysis_+3A_filein">fileIN</code></td>
<td>
<p> normalized data set. </p>
</td></tr>
<tr><td><code id="DiffAnalysis_+3A_n">n</code></td>
<td>
<p> number of identificant columns. </p>
</td></tr>
<tr><td><code id="DiffAnalysis_+3A_ind.array">ind.array</code></td>
<td>
<p> the indices of arrays to analyze. </p>
</td></tr>
<tr><td><code id="DiffAnalysis_+3A_name.a">name.A</code></td>
<td>
<p> character string containing a <a href="base.html#topic+regular+20expression">regular expression</a> contained in the columnames corresponding to the A values. </p>
</td></tr>
<tr><td><code id="DiffAnalysis_+3A_name.m">name.M</code></td>
<td>
<p> character string containing a <a href="base.html#topic+regular+20expression">regular expression</a> contained in the columnames corresponding to the M values. </p>
</td></tr>
<tr><td><code id="DiffAnalysis_+3A_fileout">fileOUT</code></td>
<td>
<p> output data file. </p>
</td></tr>
<tr><td><code id="DiffAnalysis_+3A_filedelete">fileDelete</code></td>
<td>
<p> output data file containing the list of withdrawn genes. </p>
</td></tr>
<tr><td><code id="DiffAnalysis_+3A_procs">procs</code></td>
<td>
<p> adjusting p-values procedures to be used. </p>
</td></tr>
<tr><td><code id="DiffAnalysis_+3A_alpha">alpha</code></td>
<td>
<p> values of the first type error to be used in the different procedures. </p>
</td></tr>
<tr><td><code id="DiffAnalysis_+3A_dyeswap">dyeswap</code></td>
<td>
<p> logical indicating whether the design is a dye swap. </p>
</td></tr>
<tr><td><code id="DiffAnalysis_+3A_indds">indDS</code></td>
<td>
<p> index of arrays related to dye swap. </p>
</td></tr>
<tr><td><code id="DiffAnalysis_+3A_fileid">fileID</code></td>
<td>
<p> file giving information about genes. </p>
</td></tr>
<tr><td><code id="DiffAnalysis_+3A_function.trt">function.trt</code></td>
<td>
<p> function to be applied before differential analysis. </p>
</td></tr>
<tr><td><code id="DiffAnalysis_+3A_by.var">by.var</code></td>
<td>
<p> argument passed to function.trt. </p>
</td></tr>
<tr><td><code id="DiffAnalysis_+3A_varmixt.meth">varmixt.meth</code></td>
<td>
<p> logical indicating whether to perform the variance modelling proposed by Delmar et al. (2005). </p>
</td></tr>
<tr><td><code id="DiffAnalysis_+3A_header">header</code></td>
<td>
<p> a logical value indicating whether the file contains the names of the variables as its first line. Used in the <a href="utils.html#topic+read.table">read.table</a> function. </p>
</td></tr>
<tr><td><code id="DiffAnalysis_+3A_sep">sep</code></td>
<td>
<p> the field separator string to use in the <a href="utils.html#topic+read.table">read.table</a> function. </p>
</td></tr>
<tr><td><code id="DiffAnalysis_+3A_sep.write">sep.write</code></td>
<td>
<p> the field separator string to use in the <a href="utils.html#topic+write.table">write.table</a> function. </p>
</td></tr>
<tr><td><code id="DiffAnalysis_+3A_dec.write">dec.write</code></td>
<td>
<p> the string to use for decimal points in the <a href="utils.html#topic+write.table">write.table</a> function. </p>
</td></tr>
<tr><td><code id="DiffAnalysis_+3A_...">...</code></td>
<td>
<p> Further arguments to be passed to read.table. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function performs a differential analysis in the gene-specific and homoscedastic cases.
If <em>varmixt.meth = TRUE</em>, the method used is the method proposed by Delmar et al. (2005), (the same as coded in the vm.analysis.paired function from the varmixt package.)
</p>


<h3>Value</h3>

<p>By default these following files are created
</p>
<p>- one list of differentially expressed genes by procedure
</p>
<p>- The list of withdrawn genes
</p>
<p>- The list of all the genes
</p>
<p>An R object is returned if function return value is assigned.
</p>


<h3>Author(s)</h3>

<p> J. Aubert </p>


<h3>References</h3>

<p> Delmar, P., Robin, S. and Daudin, J.J., (2005), VarMixt: efficient variance modelling for the differential analysis of replicated gene expression data, <em>Bioinformatics</em>, <b>21</b>,(4), 502&ndash;8
</p>
<p>Dudoit, S., Yang, Y. H., Callow, M. J. and Speed, T.P., (2002), Statistical methods for identifying differentially expressed genes in replicated cdna microarray experiments, <em>Statisticia Sinica</em>, <b>12</b>, 111&ndash;139
</p>


<h3>See Also</h3>

 <p><code><a href="stats.html#topic+p.adjust">p.adjust</a></code>,  <code><a href="#topic+MeanBySpot">MeanBySpot</a></code>,  <code><a href="#topic+DiffAnalysis.unpaired">DiffAnalysis.unpaired</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spleen)
# Analysis on the first 100 genes
resSpleen &lt;- DiffAnalysis(spleen[1:100, ], n = 1, ind.array = 1:6, name.A = "A.", name.M = "M.",
varmixt.meth = TRUE, dyeswap = TRUE, indDS = c(2,4,6))
# Histogramm of the raw-pvalues
# hist(resSpleen$PValueVM)
</code></pre>

<hr>
<h2 id='DiffAnalysis.unpaired'> Differential Analysis for unpaired data </h2><span id='topic+DiffAnalysis.unpaired'></span>

<h3>Description</h3>

<p>Performs the differential analysis from normalised unpaired data according to different ways of variance modelling and
computes lists of differentially expressed genes according different multiple testing procedures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DiffAnalysis.unpaired(fileIN = "resNorm.txt", n = 3, cond1 = "cond1.",
cond2 = "cond2.", fileOUT = "ListOfGenes.txt",
fileDelete = "GenesOutOfAnalysis.txt", procs = c("bonferroni", "BH"),
alpha = c(0.05, 0.05), fileID = NULL, function.trt = NULL,
by.var = "ID", varmixt.meth = TRUE, header = TRUE, sep = "\t",
sep.write = "\t", dec.write = ".", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DiffAnalysis.unpaired_+3A_filein">fileIN</code></td>
<td>
<p> normalized data set. </p>
</td></tr>
<tr><td><code id="DiffAnalysis.unpaired_+3A_n">n</code></td>
<td>
<p> number of identificant columns. </p>
</td></tr>
<tr><td><code id="DiffAnalysis.unpaired_+3A_cond1">cond1</code></td>
<td>
<p> a <a href="base.html#topic+regular+20expression">regular expression</a> corresponding to the first condition. </p>
</td></tr>
<tr><td><code id="DiffAnalysis.unpaired_+3A_cond2">cond2</code></td>
<td>
<p> a <a href="base.html#topic+regular+20expression">regular expression</a> corresponding to the second condition. </p>
</td></tr>
<tr><td><code id="DiffAnalysis.unpaired_+3A_fileout">fileOUT</code></td>
<td>
<p> output data file. </p>
</td></tr>
<tr><td><code id="DiffAnalysis.unpaired_+3A_filedelete">fileDelete</code></td>
<td>
<p> output data file containing the list of withdrawn genes. </p>
</td></tr>
<tr><td><code id="DiffAnalysis.unpaired_+3A_procs">procs</code></td>
<td>
<p> adjusting p-values procedures to be used. </p>
</td></tr>
<tr><td><code id="DiffAnalysis.unpaired_+3A_alpha">alpha</code></td>
<td>
<p> values of the first type error to be used in the different procedures. </p>
</td></tr>
<tr><td><code id="DiffAnalysis.unpaired_+3A_fileid">fileID</code></td>
<td>
<p> file giving information about genes. </p>
</td></tr>
<tr><td><code id="DiffAnalysis.unpaired_+3A_function.trt">function.trt</code></td>
<td>
<p> function to be applied before differential analysis. </p>
</td></tr>
<tr><td><code id="DiffAnalysis.unpaired_+3A_by.var">by.var</code></td>
<td>
<p> argument passed to function.trt. </p>
</td></tr>
<tr><td><code id="DiffAnalysis.unpaired_+3A_varmixt.meth">varmixt.meth</code></td>
<td>
<p> logical indicating whether to perform the variance modelling proposed by Delmar et al. (2005). </p>
</td></tr>
<tr><td><code id="DiffAnalysis.unpaired_+3A_header">header</code></td>
<td>
<p> a logical value indicating whether the file contains the names of the variables as its first line. Used in the <a href="utils.html#topic+read.table">read.table</a> function. </p>
</td></tr>
<tr><td><code id="DiffAnalysis.unpaired_+3A_sep">sep</code></td>
<td>
<p> the field separator string to use in the <a href="utils.html#topic+read.table">read.table</a> function. </p>
</td></tr>
<tr><td><code id="DiffAnalysis.unpaired_+3A_sep.write">sep.write</code></td>
<td>
<p> the field separator string to use in the <a href="utils.html#topic+write.table">write.table</a> function. </p>
</td></tr>
<tr><td><code id="DiffAnalysis.unpaired_+3A_dec.write">dec.write</code></td>
<td>
<p> the string to use for decimal points in the <a href="utils.html#topic+write.table">write.table</a> function. </p>
</td></tr>
<tr><td><code id="DiffAnalysis.unpaired_+3A_...">...</code></td>
<td>
<p> Further arguments to be passed to read.table. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function performs a differential analysis in the gene-specific and homoscedastic cases.
</p>
<p>If <em>varmixt.meth = TRUE</em>, the method used is the method proposed by Delmar et al. (2005), (the same as coded in the vm.analysis function from the varmixt package.)
</p>


<h3>Value</h3>

<p>By default these following files are created
</p>
<p>- one list of differentially expressed genes by procedure
</p>
<p>- The list of withdrawn genes
</p>
<p>- The list of all the genes
</p>
<p>An R object is returned if function return value is assigned.
</p>


<h3>Author(s)</h3>

<p> J. Aubert </p>


<h3>References</h3>

<p> Delmar, P., Robin, S. and Daudin, J.J., (2005), VarMixt: efficient variance modelling for the differential analysis of replicated gene expression data, <em>Bioinformatics</em>, <b>21</b>,(4), 502&ndash;8
</p>
<p>Dudoit, S., Yang, Y. H., Callow, M. J. and Speed, T.P., (2002), Statistical methods for identifying differentially expressed genes in replicated cdna microarray experiments, <em>Statisticia Sinica</em>, <b>12</b>, 111&ndash;139
</p>


<h3>See Also</h3>

 <p><code><a href="stats.html#topic+p.adjust">p.adjust</a></code>,  <code><a href="#topic+MeanBySpot">MeanBySpot</a></code>,  <code><a href="#topic+DiffAnalysis">DiffAnalysis</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(apo)
# Analysis on the first 100 genes
resApo &lt;- DiffAnalysis.unpaired(apo[1:100,], n = 1, ind.array = 1:8, varmixt.meth = TRUE)
# Histogramm of the raw-pvalues
# hist(resApo$PValueVM)
</code></pre>

<hr>
<h2 id='est.varmixt'>Variance Mixture Estimation</h2><span id='topic+est.varmixt'></span>

<h3>Description</h3>

<p>Performs variance mixture analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>est.varmixt(VAR, Kmax, dfreedom)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="est.varmixt_+3A_var">VAR</code></td>
<td>
<p> vector of estimated variance. </p>
</td></tr>
<tr><td><code id="est.varmixt_+3A_kmax">Kmax</code></td>
<td>
<p> maximal number of variance components. </p>
</td></tr>
<tr><td><code id="est.varmixt_+3A_dfreedom">dfreedom</code></td>
<td>
<p> degrees of freedom of the estimated variance. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>a LIST with the following components :
</p>
<table>
<tr><td><code>BIC.crit</code></td>
<td>
<p>value of the BIC criterion</p>
</td></tr>
<tr><td><code>p.i</code></td>
<td>
<p>the probability of each variance component</p>
</td></tr>
<tr><td><code>vars</code></td>
<td>
<p>variances</p>
</td></tr>
<tr><td><code>loglike</code></td>
<td>
<p>value of the criterion based on Loglikelihood</p>
</td></tr>
<tr><td><code>nmixt</code></td>
<td>
<p>number of variance components</p>
</td></tr>
<tr><td><code>tau</code></td>
<td>
<p>the matrix of posterior probability that a gene belongs to each variance component. One row per gene, one column per variance component. </p>
</td></tr>
<tr><td><code>VM2</code></td>
<td>
<p>the variance attributed to each gene according to the MAP rule - Delmar et al. (2005) <em>JRSS</em></p>
</td></tr>
<tr><td><code>VM</code></td>
<td>
<p>the variance attributed to each gene taking into account the tau values - Delmar et al. (2005) <em>Bioinformatics</em></p>
</td></tr> 
</table>


<h3>Author(s)</h3>

<p> M-L Martin-Magniette and J. Aubert </p>


<h3>References</h3>

 
<p>Delmar P, Robin S, Le Roux D, Daudin J.J (2005), Mixture model on the variance for the differential analysis of gene expression, <em>JRSS series C</em>, <b>54:1</b>, 31-50.
</p>
<p>Delmar P, Robin S, Daudin J.J (2005), VarMixt: efficient variance modelling for the differential analysis of replicated gene expression data, <em>Bioinformatics</em>, <b>21(4)</b>, 502-8.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+DiffAnalysis">DiffAnalysis</a></code>,<code><a href="#topic+DiffAnalysis.unpaired">DiffAnalysis.unpaired</a></code> </p>

<hr>
<h2 id='fdr.estimate.eta0'>Estimate the Proportion of Null p-Values</h2><span id='topic+fdr.estimate.eta0'></span>

<h3>Description</h3>

<p><code>fdr.estimate.eta0</code> estimates the proportion eta0 of null p-values in a given
vector of p-values. This quantity is an important parameter
when controlling  the false discovery rate (FDR).  A conservative choice is
eta0 = 1 but a choice closer to the true value will increase efficiency
and power
- see   Benjamini and Hochberg (1995, 2000) and Storey (2002) for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fdr.estimate.eta0(p, method=c("conservative", "adaptive", "bootstrap",
 "smoother"), lambda=seq(0,0.95,0.05) )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fdr.estimate.eta0_+3A_p">p</code></td>
<td>
<p>vector of p-values.</p>
</td></tr>
<tr><td><code id="fdr.estimate.eta0_+3A_method">method</code></td>
<td>
<p>algorithm used to estimate the proportion of null p-values.
Available options are
&quot;conservative&quot; (default), &quot;adaptive&quot;, &quot;bootstrap&quot;, and &quot;smoother&quot;.</p>
</td></tr>
<tr><td><code id="fdr.estimate.eta0_+3A_lambda">lambda</code></td>
<td>
<p>optional tuning parameter vector needed for &quot;bootstrap&quot;
and &quot;smoothing&quot; methods (defaults to <code>seq(0,0.95,0.05)</code>)
- see Storey (2002) and Storey and Tibshirani (2003).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>fdr.estimate.eta0</code> provides four algorithms: the &quot;conservative&quot;
method always returns eta0 = 1  (Benjamini and Hochberg, 1995), &quot;adaptive&quot;
uses the approach suggested in Benjamini and Hochberg (2000), &quot;bootstrap&quot;
employs the method from Storey (2002), and &quot;smoother&quot; uses the smoothing spline
approach in Storey and Tibshirani (2003).
</p>


<h3>Value</h3>

<p>The estimated proportion eta0 of null p-values.
</p>


<h3>Author(s)</h3>

<p>Konstantinos Fokianos and Korbinian Strimmer.
</p>
<p>Adapted in part from S-PLUS code by  Y. Benjamini
and R code from J.D. Storey (<a href="http://genomics.princeton.edu/storeylab/">http://genomics.princeton.edu/storeylab/</a>).
</p>


<h3>References</h3>

<p><em>&quot;conservative&quot; procedure:</em> Benjamini, Y., and Y. Hochberg (1995)  Controlling the false
discovery rate: a practical and powerful approach to multiple testing.
<em>J. Roy. Statist. Soc. B</em>, <b>57</b>, 289&ndash;300.
</p>
<p><em>&quot;adaptive&quot; procedure:</em> Benjamini, Y., and Y. Hochberg (2000) The adaptive control
of the false discovery rate in multiple hypotheses testing with independent statistics.
<em>J. Behav. Educ. Statist.</em>, <b>25</b>, 60&ndash;83.
</p>
<p><em>&quot;bootstrap&quot; procedure:</em> Storey, J. D. (2002) A direct approach to false
discovery rates.
<em>J. Roy. Statist. Soc. B.</em>, <b>64</b>, 479&ndash;498.
</p>
<p><em>&quot;bootstrap&quot; procedure:</em> Storey, J. D., and R. Tibshirani (2003)
Statistical significance for genome-wide experiments.
<em>Proc. Nat. Acad. Sci. USA</em>, <b>100</b>, 9440-9445.
</p>

<hr>
<h2 id='FileChoice'> Function to choose the files to analyze </h2><span id='topic+FileChoice'></span>

<h3>Description</h3>

<p>This function sets the working directory to the directory choosen by the user,
and allows the user to choose the files to analyze inside this directory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FileChoice(pattern = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FileChoice_+3A_pattern">pattern</code></td>
<td>
<p> an optional <a href="base.html#topic+regular+20expression">regular expression</a>. Only file names which match the regular expression will be returned. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector containing the names of the files to analyze
</p>


<h3>Author(s)</h3>

<p> J. Aubert </p>


<h3>See Also</h3>

 <p><code><a href="#topic+normalisation">normalisation</a></code> </p>

<hr>
<h2 id='filterByDefault'> Filter applied on data before normalisation </h2><span id='topic+filterByDefault'></span>

<h3>Description</h3>

<p>Apply a filter on input data by deleting spots with values for filter.var are mentionned in <em>filter.fic</em>
by deleting spots with flags equal to <em>flag0</em> (-100 by default).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterByDefault(fileIN, flag0, m, filter.fic = filter.fic, 
filter.var = filter.var, sep = sep, sep.write = sep.write, 
dec.write = dec.write, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filterByDefault_+3A_filein">fileIN</code></td>
<td>
<p> input datafame. </p>
</td></tr>
<tr><td><code id="filterByDefault_+3A_flag0">flag0</code></td>
<td>
<p> value(s) of flags to delete. </p>
</td></tr>
<tr><td><code id="filterByDefault_+3A_m">m</code></td>
<td>
<p> number of identificant columns. </p>
</td></tr>
<tr><td><code id="filterByDefault_+3A_filter.fic">filter.fic</code></td>
<td>
<p> name of the file containing the values of the variable named filter.var to delete. </p>
</td></tr>
<tr><td><code id="filterByDefault_+3A_filter.var">filter.var</code></td>
<td>
<p> name of the variable concerned with deletion. </p>
</td></tr>
<tr><td><code id="filterByDefault_+3A_sep">sep</code></td>
<td>
<p> the field separator string to use in the <a href="utils.html#topic+read.table">read.table</a> function. </p>
</td></tr>
<tr><td><code id="filterByDefault_+3A_sep.write">sep.write</code></td>
<td>
<p> the field separator string to use in the <a href="utils.html#topic+write.table">write.table</a> function. </p>
</td></tr>
<tr><td><code id="filterByDefault_+3A_dec.write">dec.write</code></td>
<td>
<p> the string to use for decimal points in the <a href="utils.html#topic+write.table">write.table</a> function. </p>
</td></tr>
<tr><td><code id="filterByDefault_+3A_...">...</code></td>
<td>
<p> Further arguments to be passed to read.table. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A subset of the input dataframe
</p>


<h3>Author(s)</h3>

<p>J. Aubert</p>


<h3>See Also</h3>

 <p><code><a href="#topic+normalisation">normalisation</a></code></p>

<hr>
<h2 id='fonctionMobile'> fonctionMobile </h2><span id='topic+fonctionMobile'></span>

<h3>Description</h3>

<p>fonctionMobile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fonctionMobile(x, FUN, lag)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fonctionMobile_+3A_x">x</code></td>
<td>
  </td></tr>
<tr><td><code id="fonctionMobile_+3A_fun">FUN</code></td>
<td>
  </td></tr>
<tr><td><code id="fonctionMobile_+3A_lag">lag</code></td>
<td>
  </td></tr>
</table>


<h3>Author(s)</h3>

<p> J. Aubert </p>

<hr>
<h2 id='Intensity.Norm'> Function to compute normalized intensity for each channel </h2><span id='topic+Intensity.Norm'></span>

<h3>Description</h3>

<p>Computes normalized intensity for each channel and each array from the file containing normalized data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Intensity.Norm(fileIN = "resNorm.txt", n = 3, ind.array = NULL, 
name.A = "A", name.M = "M.norm", sep = "\t", center = FALSE, 
log.transf = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Intensity.Norm_+3A_filein">fileIN</code></td>
<td>
<p> input dataframe. </p>
</td></tr>
<tr><td><code id="Intensity.Norm_+3A_n">n</code></td>
<td>
<p> number of identificant columns. </p>
</td></tr>
<tr><td><code id="Intensity.Norm_+3A_ind.array">ind.array</code></td>
<td>
<p> indices of arrays to use. </p>
</td></tr>
<tr><td><code id="Intensity.Norm_+3A_name.a">name.A</code></td>
<td>
<p> character string containing a <a href="base.html#topic+regular+20expression">regular expression</a> contained in the columnames corresponding to the A values. </p>
</td></tr>
<tr><td><code id="Intensity.Norm_+3A_name.m">name.M</code></td>
<td>
<p> character string containing a <a href="base.html#topic+regular+20expression">regular expression</a> contained in the columnames corresponding to the M values. </p>
</td></tr>
<tr><td><code id="Intensity.Norm_+3A_sep">sep</code></td>
<td>
<p> the field separator string to use in the <a href="utils.html#topic+read.table">read.table</a> function.</p>
</td></tr>
<tr><td><code id="Intensity.Norm_+3A_center">center</code></td>
<td>
<p> logical indicating wether data will be centered by array. </p>
</td></tr>
<tr><td><code id="Intensity.Norm_+3A_log.transf">log.transf</code></td>
<td>
<p> logical indicating wether data will stay log-transformed. </p>
</td></tr>
<tr><td><code id="Intensity.Norm_+3A_...">...</code></td>
<td>
<p> Further arguments to be passed to read.table. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A R dataframe containing the n first columns of the input dataframe and the &quot;normalized&quot; intensity 
for the red and green channels of each chosen array.
</p>


<h3>Author(s)</h3>

<p> J. Aubert </p>


<h3>References</h3>

<p>Thorne, N.P. (2004) Single-channel normalisation and analysis of twocolour cDNA microarray data, PhD thesis </p>


<h3>See Also</h3>

 <p><code><a href="#topic+normalisation">normalisation</a></code></p>

<hr>
<h2 id='LBI'> Label Bias Index </h2><span id='topic+LBI'></span>

<h3>Description</h3>

<p>This function calculates the LBI (Label Bias Index) for two self-self hybridization microarrays.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LBI(infile, name.M = "M.norm", ind.array = 1:2, graph = TRUE,
graphout = "FigM1M2")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LBI_+3A_infile">infile</code></td>
<td>
<p> input data file with normalized log-ratio base 2.</p>
</td></tr>
<tr><td><code id="LBI_+3A_name.m">name.M</code></td>
<td>
<p> character string containing a <a href="base.html#topic+regular+20expression">regular expression</a> contained in the columnames corresponding to the M values.</p>
</td></tr>
<tr><td><code id="LBI_+3A_ind.array">ind.array</code></td>
<td>
<p> the indices of arrays to analyze. </p>
</td></tr>
<tr><td><code id="LBI_+3A_graph">graph</code></td>
<td>
<p> logical indicating wether to perform graphics. </p>
</td></tr>
<tr><td><code id="LBI_+3A_graphout">graphout</code></td>
<td>
<p> name of the graphic output. </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> J. Aubert </p>


<h3>References</h3>

<p> Martin-Magniette M.L, Aubert J, Cabannes E, Daudin J.J (2005) Evaluation of the gene-specific dye bias in cDNA microarray experiments, <em>Bioinformatics</em>, <b>21(9)</b>, 1995-2000.  </p>

<hr>
<h2 id='lissageMM'> Mean Average Smoothing </h2><span id='topic+lissageMM'></span>

<h3>Description</h3>

<p> Internal function used in LocalFDR corresponding to mean average smoothing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lissageMM(x, indice, mm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lissageMM_+3A_x">x</code></td>
<td>
 </td></tr>
<tr><td><code id="lissageMM_+3A_indice">indice</code></td>
<td>
 </td></tr>
<tr><td><code id="lissageMM_+3A_mm">mm</code></td>
<td>
 </td></tr>
</table>


<h3>Author(s)</h3>

<p> J. Aubert </p>


<h3>See Also</h3>

 <p><code><a href="#topic+LocalFDR">LocalFDR</a></code>. </p>

<hr>
<h2 id='LocalFDR'> Local FDR </h2><span id='topic+LocalFDR'></span>

<h3>Description</h3>

<p>Estimation of local false discovery rates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LocalFDR(dataf = dataf, graph = TRUE, method = NULL, lambda0 = 0.5, 
smoothing = "1", thres = c(0.01, 0.05, 0.1, 0.2), mm = c(3, 5, 15, NA))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LocalFDR_+3A_dataf">dataf</code></td>
<td>
<p> input data file with two columns (1 = gene name, 2= ordered raw pvalues). </p>
</td></tr>
<tr><td><code id="LocalFDR_+3A_graph">graph</code></td>
<td>
<p> if TRUE a pdf graphic is created.</p>
</td></tr>
<tr><td><code id="LocalFDR_+3A_method">method</code></td>
<td>
<p> method for estimating m0. This must be one of the strings &quot;adaptive&quot;, &quot;conservative&quot;, &quot;bootstap&quot;, &quot;smoother&quot; or NULL. </p>
</td></tr>
<tr><td><code id="LocalFDR_+3A_lambda0">lambda0</code></td>
<td>
<p> value used in calculating m0. </p>
</td></tr>
<tr><td><code id="LocalFDR_+3A_smoothing">smoothing</code></td>
<td>
<p> &quot;1&quot; for the initial published method, &quot;2&quot; for the PAVA method
isotonic (monotonely increasing nonparametric) least squares regression - see P. Broberg (2005). </p>
</td></tr>
<tr><td><code id="LocalFDR_+3A_thres">thres</code></td>
<td>
<p> threshold defining intervals used for the movering average smoothing. </p>
</td></tr>
<tr><td><code id="LocalFDR_+3A_mm">mm</code></td>
<td>
<p> parameter defining intervals used for the movering average smoothing. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>- if graph = TRUE, a file of graphics named LocalFDRGraph.pdf
</p>
<p>- A data file (LocalFDRFile.txt) with 4 columns : gene name, raw pvalues and two columns corresponding to smoothed FDR values.
</p>


<h3>Author(s)</h3>

<p> J. Aubert </p>


<h3>References</h3>

<p> Aubert J, Bar-Hen A, Daudin J.J, Robin S (2004) Determination of the differentially expressed genes in microarrays experiments using local FDR, <em>BMC Bioinformatics</em>, <b>5:125</b>.
</p>
<p>Aubert J , Bar-Hen A, Daudin J.J, Robin S (2005) Correction: Determination of the differentially expressed genes in microarray experiments using local FDR, <em>BMC Bioinformatics</em>, <b>6:42</b>.
</p>
<p>Per Broberg (2005) A comparative review of estimates of the proportion unchanged genes and the false discovery rate, <em>BMC Bioinformatics</em> <b>6:199</b> </p>

<hr>
<h2 id='MeanBySpot'> Mean By Spot </h2><span id='topic+MeanBySpot'></span>

<h3>Description</h3>

<p>Computes the mean on the different values of spots grouping by <em>by.var</em>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MeanBySpot(fileIN, n = 3, name.A = "A", name.M = "M.norm", 
by.var = "ID", na.rm=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MeanBySpot_+3A_filein">fileIN</code></td>
<td>
<p> input dataframe. </p>
</td></tr>
<tr><td><code id="MeanBySpot_+3A_n">n</code></td>
<td>
<p> number of identificant columns. </p>
</td></tr>
<tr><td><code id="MeanBySpot_+3A_name.a">name.A</code></td>
<td>
<p> character string containing a <a href="base.html#topic+regular+20expression">regular expression</a> contained in the columnames corresponding to the A values. </p>
</td></tr>
<tr><td><code id="MeanBySpot_+3A_name.m">name.M</code></td>
<td>
<p> character string containing a <a href="base.html#topic+regular+20expression">regular expression</a> contained in the columnames corresponding to the M values. </p>
</td></tr>
<tr><td><code id="MeanBySpot_+3A_by.var">by.var</code></td>
<td>
<p> name of the grouping variable. </p>
</td></tr>
<tr><td><code id="MeanBySpot_+3A_na.rm">na.rm</code></td>
<td>
<p> a logical value indicating whether NA values should be stripped before the computation proceeds</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A R dataframe
</p>


<h3>Author(s)</h3>

<p> J. Aubert </p>


<h3>See Also</h3>

 <p><code><a href="#topic+DiffAnalysis">DiffAnalysis</a></code>, <code><a href="#topic+DiffAnalysis.unpaired">DiffAnalysis.unpaired</a></code>,<code><a href="base.html#topic+mean">mean</a></code> </p>

<hr>
<h2 id='mixgamma'> Gamma mixture </h2><span id='topic+mixgamma'></span>

<h3>Description</h3>

<p>This function provides an estimation of a mixture of gamma distributions.
The code is inspired from the varmixt package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mixgamma(VAR, dfreedom, var.init, pi.init, nmixt, stop.crit, display = TRUE,
         niter.max = 50000, criterion = criterion)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mixgamma_+3A_var">VAR</code></td>
<td>
<p> vector of estimated variance. </p>
</td></tr>
<tr><td><code id="mixgamma_+3A_dfreedom">dfreedom</code></td>
<td>
<p> degrees of freedom of the estimated variance. </p>
</td></tr>
<tr><td><code id="mixgamma_+3A_var.init">var.init</code></td>
<td>
<p> vector of initial variances for the mixture. </p>
</td></tr>
<tr><td><code id="mixgamma_+3A_pi.init">pi.init</code></td>
<td>
<p> vector of initial prior probability for the mixture. </p>
</td></tr>
<tr><td><code id="mixgamma_+3A_nmixt">nmixt</code></td>
<td>
<p> integer : number of components in the mixture model. </p>
</td></tr>
<tr><td><code id="mixgamma_+3A_stop.crit">stop.crit</code></td>
<td>
<p> the stopping relative precision limit for stopping EM algorithm. </p>
</td></tr>
<tr><td><code id="mixgamma_+3A_display">display</code></td>
<td>
<p> boolean : should the result of model fitting be displayed on the screen ? </p>
</td></tr>
<tr><td><code id="mixgamma_+3A_niter.max">niter.max</code></td>
<td>
<p> integer : maximum number of iterations of the EM algorithm. </p>
</td></tr>
<tr><td><code id="mixgamma_+3A_criterion">criterion</code></td>
<td>
<p> criterion for the convergence of the EM algorithm : either &quot;likelihood&quot;
for criterion base on loglikelihood either &quot;parameter&quot;. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>a LIST with the following components :
</p>
<table>
<tr><td><code>BIC.crit</code></td>
<td>
<p>value of the BIC criterion</p>
</td></tr>
<tr><td><code>p.i</code></td>
<td>
<p>the probability of each variance component</p>
</td></tr>
<tr><td><code>vars</code></td>
<td>
<p>variances</p>
</td></tr>
<tr><td><code>loglike</code></td>
<td>
<p>value of the criterion base on Loglikelihood</p>
</td></tr>
<tr><td><code>nmixt</code></td>
<td>
<p>number of variance components</p>
</td></tr>
<tr><td><code>tau</code></td>
<td>
<p>the matrix of posterior probability that a gene belongs to each variance component. One row per gene, one column per variance component. </p>
</td></tr>
<tr><td><code>VM2</code></td>
<td>
<p>the variance attributed to each gene according to the MAP rule - Delmar et al. (2005) <em>JRSS</em></p>
</td></tr>
<tr><td><code>VM</code></td>
<td>
<p>the variance attributed to each gene taking into account the tau values - Delmar et al. (2005) <em>Bioinformatics</em></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> P. Delmar and M.-L. Martin-Magniette</p>


<h3>See Also</h3>

 <p><code><a href="#topic+est.varmixt">est.varmixt</a></code></p>

<hr>
<h2 id='normalisation'> Function to normalize microarray data </h2><span id='topic+normalisation'></span>

<h3>Description</h3>

<p>Performs the data normalization using a general loess and 
a block effect correction via the subtraction of the median by block if there is a Block column.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalisation(fileIN = NULL, Red = "F635.Median", Green = "F532.Median",
n = 3, flag = -100, graph = TRUE, filter.function = filterByDefault, 
filter.fic = NULL, filter.var = NULL, sep.write = "\t", dec.write = ".",
header = TRUE, sep = "\t", skip = 0, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalisation_+3A_filein">fileIN</code></td>
<td>
<p> on optional regular expression contained in names of the files to analyze</p>
</td></tr>
<tr><td><code id="normalisation_+3A_red">Red</code></td>
<td>
<p> name of the column containing the Cy5 values</p>
</td></tr>
<tr><td><code id="normalisation_+3A_green">Green</code></td>
<td>
<p> name of the column containing the Cy3 values </p>
</td></tr>
<tr><td><code id="normalisation_+3A_n">n</code></td>
<td>
<p> number of identificant columns </p>
</td></tr>
<tr><td><code id="normalisation_+3A_flag">flag</code></td>
<td>
<p> list of flags values to delete </p>
</td></tr>
<tr><td><code id="normalisation_+3A_graph">graph</code></td>
<td>
<p> logical indicating wether to perform graphics </p>
</td></tr>
<tr><td><code id="normalisation_+3A_filter.function">filter.function</code></td>
<td>
<p> function to perform before beginning the normalization </p>
</td></tr>
<tr><td><code id="normalisation_+3A_filter.fic">filter.fic</code></td>
<td>
<p> argument passed to filter.funtion - file containing the values of the variable named <em>filter.var</em> to delete </p>
</td></tr>
<tr><td><code id="normalisation_+3A_filter.var">filter.var</code></td>
<td>
<p> argument passed to filter.funtion </p>
</td></tr>
<tr><td><code id="normalisation_+3A_sep.write">sep.write</code></td>
<td>
<p> the field separator string to use in the <a href="utils.html#topic+write.table">write.table</a> function </p>
</td></tr>
<tr><td><code id="normalisation_+3A_dec.write">dec.write</code></td>
<td>
<p> the string to use for decimal points in the <a href="utils.html#topic+write.table">write.table</a> function </p>
</td></tr>
<tr><td><code id="normalisation_+3A_header">header</code></td>
<td>
<p> a logical value indicating whether the file contains the names of the variables as its first line. Used in the <a href="utils.html#topic+read.table">read.table</a> function </p>
</td></tr>
<tr><td><code id="normalisation_+3A_sep">sep</code></td>
<td>
<p> the field separator string to use in the <a href="utils.html#topic+read.table">read.table</a> function </p>
</td></tr>
<tr><td><code id="normalisation_+3A_skip">skip</code></td>
<td>
<p> integer: the number of lines of the data file to skip before beginning to read data. </p>
</td></tr>
<tr><td><code id="normalisation_+3A_...">...</code></td>
<td>
<p> Further arguments to be passed to read.table. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default <em>fileIN=NULL</em> and the files to analyze have to be choosen among all the files in the working directory. 
If fileIN is not NULL, only file names which match the <a href="base.html#topic+regular+20expression">regular expression</a> fileIN will be proposed to the normalization step.
</p>


<h3>Value</h3>

<p>A file <em>resNorm.txt</em> containing the normalized data and if <em>graph=TRUE</em> a file .pdf per array with graphics
</p>


<h3>Author(s)</h3>

<p> J. Aubert </p>


<h3>References</h3>

<p> Yang, Y., Dudoit, S., Luu, P., Lin, D., Peng, V., Ngai, J. and Speed, T., (2002), Normalization for cDNA microarray data: a robust composite method addressing single and multiple slide systematic variation, <em>Nucleic Acids Research</em> <b>30</b>.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+filterByDefault">filterByDefault</a></code>,<code><a href="utils.html#topic+read.table">read.table</a></code></p>

<hr>
<h2 id='rowVars'> Form row variances</h2><span id='topic+rowVars'></span>

<h3>Description</h3>

<p>Form row variances for numeric arrays. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rowVars(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rowVars_+3A_x">x</code></td>
<td>
<p> an array of two or more dimensions, containing numeric, complex, integer or logical values, or a numeric data frame.
</p>
</td></tr>
<tr><td><code id="rowVars_+3A_...">...</code></td>
<td>
<p> ~~Describe <code>...</code> here~~ </p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is equivalent tu use of apply with FUN = var with MARGIN = 1, but is a lot faster.
</p>


<h3>Value</h3>

<p>A numeric or complex array of suitable size, or a vector if the result is one-dimensional.
</p>


<h3>See Also</h3>

 <p><code><a href="base.html#topic+rowSums">rowSums</a></code>, <code><a href="base.html#topic+rowMeans">rowMeans</a></code>, <code><a href="base.html#topic+apply">apply</a></code>  </p>

<hr>
<h2 id='spleen'>Spleen Data set</h2><span id='topic+spleen'></span>

<h3>Description</h3>

<p>An example data set for paired data analysis.
A dataframe with the normalized Spleen data set as used in the original article.  
</p>


<h3>Format</h3>

<p>spleen is a dataframe. The first column contains gene names. The 6 columns A.1 to A.6 contain normalized mean log-intensity
and the 6 columns M.1 to M.6 correspond to normalized log-ratio
</p>


<h3>References</h3>

<p>P. Delmar, Robin, S., Tronik-Le Roux S. and Daudin J.-J. (2005) Mixture model on the variance for the differential analysis of gene expression data, JRSS series C, 54(1), 31:50</p>


<h3>See Also</h3>

 <p><code><a href="#topic+DiffAnalysis">DiffAnalysis</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spleen)
# resSpleen &lt;- DiffAnalysis(spleen,n=1,ind.array=1:6,name.A="A.",name.M="M.",
# varmixt.meth=TRUE,dyeswap=TRUE,indDS=c(2,4,6))
# Histogramm of the raw-pvalues
# hist(resSpleen$PValueVM)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
