<!DOCTYPE html><html><head><title>Help for package AncestryMapper</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {AncestryMapper}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AncestryMapper'><p>Ancestry Mapper 2.0</p></a></li>
<li><a href='#arithmeticRefsMedoids'><p>Object each reference file is assigned as it is loaded in or read.</p></a></li>
<li><a href='#calculateAMids'><p>Calculate genetic distances.</p></a></li>
<li><a href='#calculateAMidsArith'><p>Calculate genetic distances.</p></a></li>
<li><a href='#createMedoid'><p>Creates Ancestry Mapper Population Reference.</p></a></li>
<li><a href='#MinMaxFreq'><p>Allele Variants for Demo SNPs According to dbSNP.</p></a></li>
<li><a href='#plotAMids'><p>Visualises genetic distances.</p></a></li>
<li><a href='#refAdd'><p>Add individuals to the CorPheno file.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>2.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-09-24</td>
</tr>
<tr>
<td>Title:</td>
<td>Assigning Ancestry Based on Population References</td>
</tr>
<tr>
<td>Author:</td>
<td>Tiago R Magalhaes, Eoghan T O'Halloran</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Eoghan T O'Halloran &lt;eoghan.ohalloran2@mail.dcu.ie&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Assigns genetic ancestry to an individual and
    studies relationships between local and global populations.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-09-24 21:31:34 UTC; eoghan</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>None, but greater than 12 GB of RAM highly
recommended.</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>svd(&ge; 0.3.3-2), parallel(&ge; 3.2.1), grDevices, graphics,
stats, utils</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-09-28 19:28:04</td>
</tr>
</table>
<hr>
<h2 id='AncestryMapper'>Ancestry Mapper 2.0</h2><span id='topic+AncestryMapper'></span>

<h3>Description</h3>

<p>Package computes the genetic distance, defined as the euclidean distance
amongst a sample set of individuals and any number of human population
references. The package allows for the visualisation of the relationship
of sample individuals to the reference populations, thus permitting the
user to assess the relationship of individuals to different geographic
groupings.
</p>
<p>Package comes with pre-loaded with toy data and toy references from
various sources comprising 158 global populations.
</p>
<p>Additional and full  Population References can be downloaded from:
</p>
<p>http://bit.ly/1OUstDP
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> AncestryMapper</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-08-??</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> None</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Eoghan T. O'Halloran ,Tiago R. Magalh\~aes, Darren J. Fitzpatrick
</p>
<p>Maintainer: Eoghan T. O'Halloran  &lt;eoghan.ohalloran2@mail.dcu.ie&gt;
</p>


<h3>References</h3>

<p>Magalh\~aes et al, 2012 PLOS One accepted.
</p>


<h3>See Also</h3>

<p>image dist
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(AncestryMapper)
Refs &lt;- system.file ("data", package = "AncestryMapper")

tpeds &lt;- system.file ("extdata", package = "AncestryMapper")

Corpheno &lt;- system.file ("extdata", "CorPheno", package =
"AncestryMapper")

All00Frq &lt;- system.file ("data", "MinMaxFreq.rda", package = "AncestryMapper")

genetic.distance &lt;- calculateAMidsArith(pathTotpeds = tpeds,
                                         NameOut = "Example",
                                         pathToAriMedoids = Refs,
                                         pathAll00 = All00Frq)

plotAMids(AMids = genetic.distance, phenoFile = Corpheno, columnPlot = "I")
## End(Not run)
</code></pre>

<hr>
<h2 id='arithmeticRefsMedoids'>Object each reference file is assigned as it is loaded in or read.</h2><span id='topic+arithmeticRefsMedoids'></span>

<h3>Description</h3>

<p>arithmeticRefsMedoids is the name of the object each reference is
assigned to after it is loaded or read.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>arithmeticRefsMedoids</code></pre>


<h3>Format</h3>

<p>A list of two character vectors, the first containing the name
of each SNP and the second containing the arithmetic medoid value for each SNP.</p>


<h3>Source</h3>

<p>Ancestry Mapper from various datasets. See: http://bit.ly/1OUstDP</p>


<h3>References</h3>

<p>Various
</p>

<hr>
<h2 id='calculateAMids'>Calculate genetic distances.</h2><span id='topic+calculateAMids'></span>

<h3>Description</h3>

<p>calculates and assigns Ancestry Mapper Ids (AMids) in a more crude, but faster manner than calculateAMids
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateAMids(pedtxtFile, fileReferences)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateAMids_+3A_pedtxtfile">pedtxtFile</code></td>
<td>
<p>Character vector giving path to PED file to be used. The PED file should include all 51 HGDP references and the individuals for which the user wishes to calculate the genetic distance.</p>
</td></tr>
<tr><td><code id="calculateAMids_+3A_filereferences">fileReferences</code></td>
<td>
<p>Character vector giving path to file File detailing the individuals in the ped file that correspond to the references, and the populations they refer to. A file that uses the 51 HGDP reference populations is provided with the package.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(AncestryMapper)

HGDP_References &lt;- system.file('extdata',
                               'HGDP_References.txt',
                                package = 'AncestryMapper')


HGDP_500SNPs &lt;- system.file('extdata',
                            'HGDP_500SNPs.ped',
                             package = 'AncestryMapper')

Corpheno &lt;- system.file('extdata',
                        'CorPheno',
                         package = 'AncestryMapper')

genetic.distance &lt;- calculateAMids(pedtxtFile = HGDP_500SNPs,
                                   fileReferences = HGDP_References)

plotAMids(AMids = genetic.distance, phenoFile = Corpheno, columnPlot = "I")

## End(Not run)
</code></pre>

<hr>
<h2 id='calculateAMidsArith'>Calculate genetic distances.</h2><span id='topic+calculateAMidsArith'></span>

<h3>Description</h3>

<p>Calculates genetic distance between samples and population references.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateAMidsArith(pathTotpeds, pathToAriMedoids, AMmcapply = F, nrcores,
  seqchip = "", noseqdat = F, wd, NameOut = NULL, pathAll00)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateAMidsArith_+3A_pathtotpeds">pathTotpeds</code></td>
<td>
<p>Character vector giving path to folder containing the plink tPED file(s) to be used.</p>
</td></tr>
<tr><td><code id="calculateAMidsArith_+3A_pathtoarimedoids">pathToAriMedoids</code></td>
<td>
<p>Character vector giving path to folder containing the arithmetic references to be used.</p>
</td></tr>
<tr><td><code id="calculateAMidsArith_+3A_ammcapply">AMmcapply</code></td>
<td>
<p>Logical value (TRUE or FALSE), specifying if the multicore funcion mcapply, should be used.
Inappropriate for most HPC cluster systems. Default = FALSE</p>
</td></tr>
<tr><td><code id="calculateAMidsArith_+3A_nrcores">nrcores</code></td>
<td>
<p>Numeric value detailing how many cores should be used if AMmcapply==TRUE.
If left unspecificed the number of cores will be detected and nrcores will be set to that number -2.</p>
</td></tr>
<tr><td><code id="calculateAMidsArith_+3A_seqchip">seqchip</code></td>
<td>
<p>Character vector specifying if only references from one main SNP chip panel are to be used. All references are marked with what chip panel they use at the end of their file names, eg 'Yoruba.HGDP.20000.Illumina.ods'
May be important if your data has few SNPs in common with one panel. All toy references prepared use 'Illumina' panels. Whole Genome sequence data is specified with 'WG'.
Supports custom designations, but will trigger a warning when used.</p>
</td></tr>
<tr><td><code id="calculateAMidsArith_+3A_noseqdat">noseqdat</code></td>
<td>
<p>Logical value (TRUE or FALSE), specifying if sequence data is to be excluded, will use only references that do not have names ending in '.WG.ods/rds/rda'. Default = FALSE</p>
</td></tr>
<tr><td><code id="calculateAMidsArith_+3A_wd">wd</code></td>
<td>
<p>Character vector giving the desired working directory to house the outputs of calculateAMidsArith. If left unspecified will use current working directory.</p>
</td></tr>
<tr><td><code id="calculateAMidsArith_+3A_nameout">NameOut</code></td>
<td>
<p>Character vector giving the desired prefix name for the AMid file. Default is NULL.</p>
</td></tr>
<tr><td><code id="calculateAMidsArith_+3A_pathall00">pathAll00</code></td>
<td>
<p>Character vector giving the path to a file containing the full data table of each dbSNP and both alleles. An example version covering the SNPs used in the example data is included. A full version can be found at: http://bit.ly/1OUstDP</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Refs &lt;- system.file('data', package = 'AncestryMapper')
tpeds &lt;- system.file('extdata', package = 'AncestryMapper')
Corpheno &lt;- system.file('extdata', 'CorPheno', package = 'AncestryMapper')
All00Frq &lt;- system.file ('data', 'MinMaxFreq.rda', package = 'AncestryMapper')

genetic.distance &lt;- calculateAMidsArith(pathTotpeds = tpeds,
                                   NameOut = 'Example',
                                   pathToAriMedoids = Refs,
                                   pathAll00 = All00Frq)

plotAMids(AMids = genetic.distance, phenoFile = Corpheno, columnPlot = "I")

## End(Not run)
</code></pre>

<hr>
<h2 id='createMedoid'>Creates Ancestry Mapper Population Reference.</h2><span id='topic+createMedoid'></span>

<h3>Description</h3>

<p>Generates arithmetic population reference from PLINK tPED files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createMedoid(pathTotpeds, AMmcapply = F, nrcores, wd, pathAll00,
  chipMan = "ChipMan", OutForm = "rda")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="createMedoid_+3A_pathtotpeds">pathTotpeds</code></td>
<td>
<p>Character vector giving path to folder containing tPED file(s) to be used.</p>
</td></tr>
<tr><td><code id="createMedoid_+3A_ammcapply">AMmcapply</code></td>
<td>
<p>Logical value (TRUE or FALSE), specifying if the multicore funcion mcapply, should be used. Inappropriate for most HPC cluster systems.
Default = FALSE</p>
</td></tr>
<tr><td><code id="createMedoid_+3A_nrcores">nrcores</code></td>
<td>
<p>Numeric value detailing how many cores should be used if AMmcapply==TRUE. If left unspecificed the number of cores will be detected and mc.cores will be set to that number -1.</p>
</td></tr>
<tr><td><code id="createMedoid_+3A_wd">wd</code></td>
<td>
<p>Character vector giving the desired working directory to house the outputs of calculateAMids. If left unspecified will use current working directory.</p>
</td></tr>
<tr><td><code id="createMedoid_+3A_pathall00">pathAll00</code></td>
<td>
<p>Character vector giving the path to a file containing the full data table of each dbSNP and both alleles. A toy version covering the SNPs used in the toy data is included. A full version can be found at: http://bit.ly/1OUstDP</p>
</td></tr>
<tr><td><code id="createMedoid_+3A_chipman">chipMan</code></td>
<td>
<p>Character vector giving name of company from which the SNP panel is derived. Eg 'Illumina', 'Affymetrix'. If no value is given will default to 'ChipMan'. If it is whole genome sequencing, please put 'WG'.
The value will appear in the name of the arithmetic reference file. e.g. 'medoidArithmetic_Yoruba_HGDP_1000_Illumina.rda'.</p>
</td></tr>
<tr><td><code id="createMedoid_+3A_outform">OutForm</code></td>
<td>
<p>Character vector giving option for output format for arithmetic medoids. Can be one of three options.
'ods' will generate a raw text file with the default extension of '.ods'. This is the default option as is the most flexible format.
'rds' will save the arithmetic medoid as a .rds file which can be loaded into R faster and is also roughly a third the size of the raw text version.
'rda' will save the arithmetic medoid as a .rda file which can be loaded into R faster and is also roughly a third the size of the raw text version.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
chipManExample &lt;- 'Illumina'
tpeds &lt;- system.file('extdata', package='AncestryMapper')

createMedoid(pathTotpeds = tpeds, chipMan = chipManExample)


## End(Not run)
</code></pre>

<hr>
<h2 id='MinMaxFreq'>Allele Variants for Demo SNPs According to dbSNP.</h2><span id='topic+MinMaxFreq'></span>

<h3>Description</h3>

<p>This data set contains the major and minor alleles for the 1000 SNPs
used in the demo data according to the strand orientation used on dbSNP.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MinMaxFreq</code></pre>


<h3>Format</h3>

<p>A matrix containing 1000 observations of two variables.</p>


<h3>Source</h3>

<p>dbSNP.</p>


<h3>References</h3>

<p>&quot;ftp://ftp.ncbi.nlm.nih.gov/snp/organisms/human_9606/VCF/00-All.vcf.gz&quot;
</p>

<hr>
<h2 id='plotAMids'>Visualises genetic distances.</h2><span id='topic+plotAMids'></span>

<h3>Description</h3>

<p>plotAMids is used to visualise the relationship amongst individuals and references.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotAMids(AMids, phenoFile, columnPlot = "I", quantilePlot = TRUE,
  colorPlot = "BlBrewer", sepLinesPop = TRUE, plotIndNames = FALSE,
  legColor = TRUE, legRef = TRUE, legPheno = TRUE, legAxisPop = TRUE,
  legData = FALSE, bmar, lmar, tmar, rmar, cexref = 0.9, cexind = 0.8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotAMids_+3A_amids">AMids</code></td>
<td>
<p>Dataframe of genetic distances calculated by calculateAMids or calculateAMidsArith.</p>
</td></tr>
<tr><td><code id="plotAMids_+3A_phenofile">phenoFile</code></td>
<td>
<p>Optional file with phenotype, color and order information for individuals and populations. An example file, called CorPheno, is contained in the 'extdata' folder with the package.</p>
</td></tr>
<tr><td><code id="plotAMids_+3A_columnplot">columnPlot</code></td>
<td>
<p>Takes values 'I' or 'C'. 'I' is the default option. 'I' plots the normalised euclidean distances whereas 'C' plots the crude distances.</p>
</td></tr>
<tr><td><code id="plotAMids_+3A_quantileplot">quantilePlot</code></td>
<td>
<p>Logical. Takes values TRUE or FALSE. TRUE is the default option. If columnPlot is 'C', TRUE will plot the quantiles, FALSE will plot the raw values.</p>
</td></tr>
<tr><td><code id="plotAMids_+3A_colorplot">colorPlot</code></td>
<td>
<p>Colors for the AMids. Possible choices are 'RedBl', 'RedBlGr' and 'BLBrewer'. The user can also provide a vector of colors.</p>
</td></tr>
<tr><td><code id="plotAMids_+3A_seplinespop">sepLinesPop</code></td>
<td>
<p>Logical. Takes values TRUE or FALSE. The default is TRUE.
If TRUE, a line demarcating populations is plotted.</p>
</td></tr>
<tr><td><code id="plotAMids_+3A_plotindnames">plotIndNames</code></td>
<td>
<p>Logical. Takes values TRUE or FALSE. The default is FALSE. 
If TRUE, the individual ids are plotted on the left axis.</p>
</td></tr>
<tr><td><code id="plotAMids_+3A_legcolor">legColor</code></td>
<td>
<p>Logical. Takes values TRUE or FALSE. The default is TRUE. 
If TRUE, the legend for the colour gradient will be plotted in the top left.</p>
</td></tr>
<tr><td><code id="plotAMids_+3A_legref">legRef</code></td>
<td>
<p>Logical. Takes values TRUE or FALSE. The default is TRUE. If TRUE, text giving names of references will be plotted along the x axis.</p>
</td></tr>
<tr><td><code id="plotAMids_+3A_legpheno">legPheno</code></td>
<td>
<p>Logical. Takes values TRUE or 'no. The default is TRUE. If TRUE, will plot colour blocks relating to the population, dataset and regional origin of data if sample IDs have been given these in the Corpheno file, if not present, will plot them under 'Unspecified'.</p>
</td></tr>
<tr><td><code id="plotAMids_+3A_legaxispop">legAxisPop</code></td>
<td>
<p>Logical. Takes values TRUE or FALSE. The default is TRUE. 
If TRUE text will be plotted giving name and number of populations for samples used on the right y axis of the plot.</p>
</td></tr>
<tr><td><code id="plotAMids_+3A_legdata">legData</code></td>
<td>
<p>Logical. Takes values TRUE or FALSE. The default is FALSE.
If TRUE, the reference to the dataset used to create the reference is appended to the reference population name on the bottom x axis.</p>
</td></tr>
<tr><td><code id="plotAMids_+3A_bmar">bmar</code></td>
<td>
<p>Takes numeric value. Changes the size of the bottom outer margin of the plot. The default is empty. 
For more see ?par()</p>
</td></tr>
<tr><td><code id="plotAMids_+3A_lmar">lmar</code></td>
<td>
<p>Takes numeric value. Changes the size of the left outer margin of the plot. The default is empty.
For more see ?par()</p>
</td></tr>
<tr><td><code id="plotAMids_+3A_tmar">tmar</code></td>
<td>
<p>Takes numeric value. Changes the size of the top outer margin of the plot. The default is empty.
For more see ?par()</p>
</td></tr>
<tr><td><code id="plotAMids_+3A_rmar">rmar</code></td>
<td>
<p>Takes numeric value. Changes the size of the right outer margin of the plot. The default is empty. 
For more see ?par()</p>
</td></tr>
<tr><td><code id="plotAMids_+3A_cexref">cexref</code></td>
<td>
<p>Takes numeric value. Controls text size of reference names on y axis. Default is 0.9.</p>
</td></tr>
<tr><td><code id="plotAMids_+3A_cexind">cexind</code></td>
<td>
<p>Takes numeric value. Controls text size of sample names on y axis. Default is 0.8.
Individual sample IDs need plotIndNames = &quot;yes&quot; to display, this is set to FALSE by default.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Refs &lt;- system.file('data', package = 'AncestryMapper')
tpeds &lt;- system.file('extdata', package = 'AncestryMapper')
Corpheno &lt;- system.file('extdata', 'CorPheno', package = 'AncestryMapper')
All00Frq &lt;- system.file ('data', 'MinMaxFreq.rda', package = 'AncestryMapper')

genetic.distance &lt;- calculateAMidsArith(pathTotpeds = tpeds,
                                   NameOut = 'Example',
                                   pathToAriMedoids = Refs,
                                   pathAll00 = All00Frq)

plotAMids(AMids = genetic.distance, phenoFile = Corpheno, columnPlot = "I")

## End(Not run)
</code></pre>

<hr>
<h2 id='refAdd'>Add individuals to the CorPheno file.</h2><span id='topic+refAdd'></span>

<h3>Description</h3>

<p>Add individuals to the CorPheno file in order to order and plot them with specific colours. Ids that aren't in the CorPheno will be plotted under 'Unspecified' and given a grey colour by plotAMids()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refAdd(phenoId, phenoValues, ignoreDupes = F, phenoFile, writeCor = T)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refAdd_+3A_phenoid">phenoId</code></td>
<td>
<p>Path to file containing list of invididuals to be added to CorPheno with the three columns 'UNIQID','Fam','Pheno_Pop'.
A full example called 'Example.phenoId' is present in the 'extdata' folder of the AncestryMapper package.</p>
</td></tr>
<tr><td><code id="refAdd_+3A_phenovalues">phenoValues</code></td>
<td>
<p>Path to file containing information on each population to be added, such as continental origin and colours as well as other information.
A full example called 'Example.phenoValues' is present in the 'extdata' folder of the AncestryMapper package.</p>
</td></tr>
<tr><td><code id="refAdd_+3A_ignoredupes">ignoreDupes</code></td>
<td>
<p>Logical value (TRUE or FALSE), specifying if the presence of individual IDs already in CorPheno should be ignored. Useful if the user knows this is the case and just wants the individuals not already included in the directed phenoFile.
Default = FALSE</p>
</td></tr>
<tr><td><code id="refAdd_+3A_phenofile">phenoFile</code></td>
<td>
<p>Main file with phenotype information for each individual. A sample file called CorPheno is included with the package in the ext folder. It contains values for the samples from the HGDP. This function augments this file with any novel individuals.
If no value is given the sample file in the ext folder is used by default.</p>
</td></tr>
<tr><td><code id="refAdd_+3A_writecor">writeCor</code></td>
<td>
<p>Logical value (TRUE or FALSE), specifying if the new CorPheno should overwrite the file the given in 'phenoFile'. A backup of the previous CorPheno file with the same path as given in 'phenoFile' with '_Original' appended to the name will also be produced. You could alternatively write out your new file to your preferred location with write.table, making sure to keep the columns tab spaced. Default = TRUE</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
phenoIdPth &lt;- system.file ("extdata", "Example.phenoId", package = "AncestryMapper")
PhenoValPth &lt;- system.file("extdata", "Example.phenoValues", package = "AncestryMapper")
Corpheno &lt;- system.file("extdata", "CorPheno", package = "AncestryMapper")

refAdd(phenoId = phenoIdPth, phenoValues = PhenoValPth, phenoFile = Corpheno)


## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
