<!DOCTYPE html><html><head><title>Help for package annmatrix</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {annmatrix}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#annmatrix'><p>annmatrix Objects and Basic Functionality</p></a></li>
<li><a href='#autocomplete'><p>Auto Complete Functions for annmatrix Class</p></a></li>
<li><a href='#bind'><p>Bind several annmatrix Objects Together</p></a></li>
<li><a href='#convert'><p>Convert annmatrix Objects to and from Other Types</p></a></li>
<li><a href='#groupgenerics'><p>Group Generic Functions for annmatrix Class</p></a></li>
<li><a href='#matrixgenerics'><p>Matrix Generic Functions for annmatrix Class</p></a></li>
<li><a href='#pca'><p>Principal Component Analysis for annmatrix Objects</p></a></li>
<li><a href='#print'><p>Print annmatrix Object</p></a></li>
<li><a href='#scale'><p>Scaling and Centering of annmatrix Objects</p></a></li>
<li><a href='#stack'><p>Stack an annmatrix object</p></a></li>
<li><a href='#subset'><p>Subset annmatrix Objects</p></a></li>
<li><a href='#transpose'><p>Transposing annmatrix Objects</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Annotated Matrix: Matrices with Persistent Row and Column
Annotations</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.2</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Karolis Koncevičius &lt;karolis.koncevicius@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements persistent row and column annotations for R matrices. The annotations associated with rows and columns are preserved after subsetting, transposition, and various other matrix-specific operations. Intended use case is for storing and manipulating genomic datasets which typically consist of a matrix of measurements (like gene expression values) as well as annotations about rows (i.e. genomic locations) and annotations about columns (i.e. meta-data about collected samples). But 'annmatrix' objects are also expected to be useful in various other contexts.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>utils, methods, stats</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/karoliskoncevicius/annmatrix">https://github.com/karoliskoncevicius/annmatrix</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/karoliskoncevicius/annmatrix/issues">https://github.com/karoliskoncevicius/annmatrix/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-05-14 07:45:16 UTC; karolis</td>
</tr>
<tr>
<td>Author:</td>
<td>Karolis Koncevičius [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-05-14 08:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='annmatrix'>annmatrix Objects and Basic Functionality</h2><span id='topic+annmatrix'></span><span id='topic+rowanns'></span><span id='topic+rowanns+3C-'></span><span id='topic+colanns'></span><span id='topic+colanns+3C-'></span><span id='topic++40.annmatrix'></span><span id='topic++40+3C-.annmatrix'></span><span id='topic++24.annmatrix'></span><span id='topic++24+3C-.annmatrix'></span>

<h3>Description</h3>

<p>Annotated matrix is a regular matrix with additional functionality for
attaching persistent information about row and column entries. Annotations
associated with rows and columns are preserved after subsetting,
transposition, and various other matrix-specific operations.
</p>
<p>Intended 'annmatrix' use case is for storing and manipulating genomic
datasets that typically consist of a matrix of measurements (like gene
expression values) as well as annotations about rows (i.e. genomic
locations) and annotations about columns (i.e. meta-data about collected
samples). But <code>annmatrix</code> objects are also expected be useful in
various other contexts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annmatrix(x, rann, cann)

rowanns(x, names)
rowanns(x, names) &lt;- value

colanns(x, names)
colanns(x, names) &lt;- value

## S3 method for class 'annmatrix'
@(object, name)
## S3 replacement method for class 'annmatrix'
@(object, name) &lt;- value

## S3 method for class 'annmatrix'
x$name
## S3 replacement method for class 'annmatrix'
x$name &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annmatrix_+3A_x">x</code>, <code id="annmatrix_+3A_object">object</code></td>
<td>
<p>an R object.</p>
</td></tr>
<tr><td><code id="annmatrix_+3A_rann">rann</code></td>
<td>
<p>annotation <code>data.frame</code> for rows of the <code>annmatrix</code> object.</p>
</td></tr>
<tr><td><code id="annmatrix_+3A_cann">cann</code></td>
<td>
<p>annotation <code>data.frame</code> for columns of the <code>annmatrix</code> object.</p>
</td></tr>
<tr><td><code id="annmatrix_+3A_names">names</code></td>
<td>
<p>a character vector of existing row/column annotation names.</p>
</td></tr>
<tr><td><code id="annmatrix_+3A_value">value</code></td>
<td>
<p>a value that will be assigned to row/column annotation field.</p>
</td></tr>
<tr><td><code id="annmatrix_+3A_name">name</code></td>
<td>
<p>a name of an existing row/column annotation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>annmatrix()</code> constructs an annmatrix. The function expects <code>x</code> to
be a <code>matrix</code> and <code>rowanns</code> and <code>colanns</code> to be of class
<code>data.frame</code>. If the passed objects are of a different classes they
will be converted via the use of <code>as.matrix</code> and <code>as.data.frame</code>.
</p>
<p><code>rowanns</code> and <code>colanns</code> returns the selected field from column and
row annotations respectively. When the selected field is not specified the
whole annotation <code>data.frame</code> is returned.
</p>
<p><code>@</code> and <code>$</code> are convenience shortcuts for selecting annotations.
<code>X@value</code> selects an existing column from row annotations while
<code>X$value</code> selects a column from column annotations. An empty selection
of <code>X@''</code> and <code>X$''</code> will return the whole annotation data.frame
for rows and columns respectively.
</p>
<p><code>rowanns&lt;-</code> and <code>colanns&lt;-</code> functions can be used to replace the
column and row annotations respectively. When the selected field is not
specified the whole annotation <code>data.frame</code> is replaced.
</p>
<p><code>@&lt;-</code> and <code>$&lt;-</code> are convenience shortcuts for the above (see
Examples). A replacement of an empty value - <code>X@'' &lt;- df</code> and
<code>X$'' &lt;- df</code> will replace the whole annotation data.frame.
</p>


<h3>Value</h3>

<p><code>annmatrix</code> returns an R object of class 'annmatrix'.
</p>


<h3>Author(s)</h3>

<p>Karolis Koncevičius
</p>


<h3>See Also</h3>

<p><code>as.annmatrix</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># construct annmatrix object
x &lt;- matrix(rnorm(20*10), 20, 10)

coldata &lt;- data.frame(group  = rep(c("case", "control"), each = 5),
                      gender = sample(c("M", "F"), 10, replace = TRUE))

rowdata &lt;- data.frame(chr = sample(c("chr1", "chr2"), 20, replace = TRUE),
                      pos = runif(20, 0, 1000000))

X &lt;- annmatrix(x, rowdata, coldata)

is.matrix(x)
is.matrix(X)

is.annmatrix(x)
is.annmatrix(X)

# manipulating annotations without using shortcuts
rowanns(X)
colanns(X)

rowanns(X, "chr")
rowanns(X, "gene") &lt;- letters[1:20]
rowanns(X, c("chr", "gene"))
rowanns(X, "gene") &lt;- NULL
rowanns(X)

colanns(X, "group")
colanns(X, "age") &lt;- 1:10*10
colanns(X, "age")
colanns(X, "age") &lt;- NULL
colanns(X, "age")

# more convenient
X@''
X@chr
X@gene &lt;- letters[1:20]
X@gene
X@gene &lt;- NULL
X@gene

X$''
X$group
X$age &lt;- 1:10*10
X$age
X$age &lt;- NULL
X$age
X$'' &lt;- data.frame(id = 1:10, name = LETTERS[1:10])
X$name

# annotations are preserved after subsetting
Y &lt;- X[X@chr == "chr1", X$name %in% c("A", "B", "C")]
Y@chr
Y$''

Y[, 1]
Y[, 1, drop = FALSE]

</code></pre>

<hr>
<h2 id='autocomplete'>Auto Complete Functions for annmatrix Class</h2><span id='topic+autocomplete'></span><span id='topic+.DollarNames.annmatrix'></span><span id='topic+.AtNames.annmatrix'></span>

<h3>Description</h3>

<p>Function used to select autocomplete options for dollar '$' and at '@' operators.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'annmatrix'
.DollarNames(x, pattern = "")

## S3 method for class 'annmatrix'
.AtNames(x, pattern = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="autocomplete_+3A_x">x</code></td>
<td>
<p>annmatrix object.</p>
</td></tr>
<tr><td><code id="autocomplete_+3A_pattern">pattern</code></td>
<td>
<p>a regular expression used to select possible auto-completion names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A set of possible auto-completion names for row (<code>@</code>) or column (<code>$</code>) annotation fields.
</p>


<h3>Author(s)</h3>

<p>Karolis Koncevičius
</p>

<hr>
<h2 id='bind'>Bind several annmatrix Objects Together</h2><span id='topic+bind'></span><span id='topic+rbind.annmatrix'></span><span id='topic+cbind.annmatrix'></span>

<h3>Description</h3>

<p>Implementation of <code>rbind</code> and <code>cbind</code> methods for annmatrix objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'annmatrix'
rbind(...)

## S3 method for class 'annmatrix'
cbind(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bind_+3A_...">...</code></td>
<td>
<p>(generalized) vector or matrix objects</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All the inputs are bound together in exact same way as if all the annmatrix objects were regular matrices.
Then, the row and column annotations of the supplied annmatrix objects are merged together.
Missing annotations are filled in using 'NA' values.
</p>
<p>For demonstration purposes only <code>rbind</code> will be described with <code>cbind</code> behaving accordingly.
</p>
<p>1) Matrix.
The obtained matrix will be exactly the same as calling <code>rbind</code> with all annmatrix objects replaced by regular matrices.
</p>
<p>2) Column annotations.
When <code>rbind</code> is called the matrices are all assumed to have the same set of columns.
Hence, the column annotations are assumed to be shared between all provided annmatrix objects.
Thus, in order to retain all possible column annotations, they are merged together.
This way any column annotation field present in at least one of the provided annmatrix objects will be present in the final result.
In case of conflicts, when the same annotation field is present in multiple annmatrix objects but contains different values, the first occuring instance is selected and an appropriate warning is displayed.
Non-annmatrix objects are assumed to share the column annotations present in supplied annmatrix objects.
</p>
<p>3) Row annotations.
When <code>rbind</code> is called the matrices are assumed to have a separate unique set of rows.
Hence no conflicts between annotation values are possible for row annotations.
In order to retain all possible row annotations, row annotations are merged together.
Thus, the resulting row annotation data frame will have as many fields as there were unique row annotation fields among all the provided annmatrix objects.
Unlike with column annotations, if a particular annmatrix only had a subset of the final collection of annotation fields, then the missing fields are added and the annotation is filled with NA values.
All the rows associated with non-annmatrix objects will have missing (NA) values for all the annotation fields.
</p>


<h3>Value</h3>

<p>a single annmatrix object resulting from binding all the inputs together
</p>


<h3>Author(s)</h3>

<p>Karolis Koncevičius
</p>


<h3>Examples</h3>

<pre><code class='language-R'># construct annmatrix object
x &lt;- matrix(rnorm(20*10), 20, 10)

coldata &lt;- data.frame(group  = rep(c("case", "control"), each = 5),
                      gender = sample(c("M", "F"), 10, replace = TRUE))

rowdata &lt;- data.frame(chr = sample(c("chr1", "chr2"), 20, replace = TRUE),
                      pos = runif(20, 0, 1000000))

X &lt;- annmatrix(x, rowdata, coldata)

X1 &lt;- X[1:10,]
X2 &lt;- X[11:20,]
all.equal(X, rbind(X1, X2))

X1 &lt;- X[,1:5]
X2 &lt;- X[,6:10]
all.equal(X, cbind(X1, X2))

X11 &lt;- X[1:10, 1:5]
X21 &lt;- X[11:20, 1:5]
X12 &lt;- X[1:10, 6:10]
X22 &lt;- X[11:20, 6:10]
all.equal(X, cbind(rbind(X11, X21), rbind(X12, X22)))

</code></pre>

<hr>
<h2 id='convert'>Convert annmatrix Objects to and from Other Types</h2><span id='topic+convert'></span><span id='topic+as.annmatrix'></span><span id='topic+as.annmatrix.default'></span><span id='topic+as.annmatrix.matrix'></span><span id='topic+as.matrix.annmatrix'></span><span id='topic+is.annmatrix'></span>

<h3>Description</h3>

<p>Methods for turning R objects to class annmatrix and vice versa.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.annmatrix(x)

## Default S3 method:
as.annmatrix(x)

## S3 method for class 'matrix'
as.annmatrix(x)

## S3 method for class 'annmatrix'
as.matrix(x, ...)

is.annmatrix(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert_+3A_x">x</code></td>
<td>
<p>an R object.</p>
</td></tr>
<tr><td><code id="convert_+3A_...">...</code></td>
<td>
<p>additional arguments to be passed to or from methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>as.annmatrix</code> will attempt to convert an object to annmatrix.
</p>
<p><code>as.matrix</code> will turn an <code>annmatrix</code> object into a regular matrix.
</p>
<p><code>is.annmatrix</code> checks if the object is an instance of <code>annmatrix</code>.
</p>


<h3>Value</h3>

<p><code>is.annmatrix</code> returns TRUE if object is of class 'annmatrix' and FALSE otherwise.
<code>as.annmatrix</code> methods return an object of class 'annmatrix'.
<code>as.matrix</code> returns a regular matrix.
</p>


<h3>Author(s)</h3>

<p>Karolis Koncevičius
</p>


<h3>Examples</h3>

<pre><code class='language-R'># construct annmatrix object
x &lt;- matrix(rnorm(20*10), 20, 10)
X &lt;- as.annmatrix(x)

X$group  &lt;- rep(c("case", "control"), each = 5)
X$gender &lt;- sample(c("M", "F"), 10, replace = TRUE)
X@chr    &lt;- sample(c("chr1", "chr2"), 20, replace = TRUE)
X@pos    &lt;- runif(20, 0, 1000000)

is.matrix(x)
is.matrix(X)

is.annmatrix(x)
is.annmatrix(X)

as.matrix(X)

</code></pre>

<hr>
<h2 id='groupgenerics'>Group Generic Functions for annmatrix Class</h2><span id='topic+groupgenerics'></span><span id='topic+Ops.annmatrix'></span><span id='topic+chooseOpsMethod.annmatrix'></span>

<h3>Description</h3>

<p>The functions listed here work under the hood and are almost never called by the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'annmatrix'
Ops(e1, e2)

## S3 method for class 'annmatrix'
chooseOpsMethod(x, y, mx, my, cl, reverse)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="groupgenerics_+3A_e1">e1</code>, <code id="groupgenerics_+3A_e2">e2</code></td>
<td>
<p>annmatrix objects.</p>
</td></tr>
<tr><td><code id="groupgenerics_+3A_x">x</code>, <code id="groupgenerics_+3A_y">y</code></td>
<td>
<p>The objects being dispatched on by the group generic.</p>
</td></tr>
<tr><td><code id="groupgenerics_+3A_mx">mx</code>, <code id="groupgenerics_+3A_my">my</code></td>
<td>
<p>The methods found for objects 'x' and 'y'.</p>
</td></tr>
<tr><td><code id="groupgenerics_+3A_cl">cl</code></td>
<td>
<p>The call to the group generic.</p>
</td></tr>
<tr><td><code id="groupgenerics_+3A_reverse">reverse</code></td>
<td>
<p>A logical value indicating whether 'x' and 'y' are reversed from the way they were supplied to the generic.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'annmatrix'.
</p>


<h3>Author(s)</h3>

<p>Karolis Koncevičius
</p>

<hr>
<h2 id='matrixgenerics'>Matrix Generic Functions for annmatrix Class</h2><span id='topic+matrixgenerics'></span><span id='topic++25+2A+25.annmatrix'></span>

<h3>Description</h3>

<p>Matrix cross-product operator implemented for annmatrix class
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'annmatrix'
x %*% y
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="matrixgenerics_+3A_x">x</code>, <code id="matrixgenerics_+3A_y">y</code></td>
<td>
<p>numeric or complex matrices or vectors.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The resulting matrix will be the same as a product between two regular matrices.
If present annmatrix row annotations will be carried over from the first matrix <code>x</code>
while the annotations for rows will be carried over from the second matrix <code>y</code>.
</p>


<h3>Value</h3>

<p>an object of class 'annmatrix'.
</p>


<h3>Author(s)</h3>

<p>Karolis Koncevičius
</p>


<h3>Examples</h3>

<pre><code class='language-R'># construct annmatrix object
x &lt;- matrix(rnorm(20*10), 20, 10)

coldata &lt;- data.frame(group  = rep(c("case", "control"), each = 5),
                      gender = sample(c("M", "F"), 10, replace = TRUE))

rowdata &lt;- data.frame(chr = sample(c("chr1", "chr2"), 20, replace = TRUE),
                      pos = runif(20, 0, 1000000))

X &lt;- annmatrix(x, rowdata, coldata)

res &lt;- 1:20 %*% X
res$group

res &lt;- X %*% 1:10
res@chr

res &lt;- t(X) %*% X
res@group
res$group

</code></pre>

<hr>
<h2 id='pca'>Principal Component Analysis for annmatrix Objects</h2><span id='topic+pca'></span><span id='topic+prcomp.annmatrix'></span>

<h3>Description</h3>

<p>Performs principal component analysis on annmatrix objects and preserves row and column annotations in scores and loadings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'annmatrix'
prcomp(
  x,
  retx = TRUE,
  center = TRUE,
  scale. = FALSE,
  tol = NULL,
  rank. = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pca_+3A_x">x</code></td>
<td>
<p>annmatrix object.</p>
</td></tr>
<tr><td><code id="pca_+3A_retx">retx</code></td>
<td>
<p>logical indicator whether the rotated variables should be returned (defaults to TRUE).</p>
</td></tr>
<tr><td><code id="pca_+3A_center">center</code></td>
<td>
<p>logical value indicating whether variables should be centered (defaults to TRUE).</p>
</td></tr>
<tr><td><code id="pca_+3A_scale.">scale.</code></td>
<td>
<p>logical value indicating whether variables should be scaled (defaults to FALSE).</p>
</td></tr>
<tr><td><code id="pca_+3A_tol">tol</code></td>
<td>
<p>tolerance value indicating magnitude below which components will be omitted.</p>
</td></tr>
<tr><td><code id="pca_+3A_rank.">rank.</code></td>
<td>
<p>number specifying the maximal rank (max number of principal components to be used).</p>
</td></tr>
<tr><td><code id="pca_+3A_...">...</code></td>
<td>
<p>arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The resulting loadings ('rotation') and scores ('x') are turned into annmatrix objects with additional row and column annotations.
Row annotations are copied from the original input matrix while column annotations contain computed information about each principal component.
The added information contains: 1) principal component number, 2) standard deviation, 3) variance, 4) fraction of variance explained.
</p>


<h3>Value</h3>

<p>prcom object with rotation and x matrices turned into annmatrix
</p>


<h3>Author(s)</h3>

<p>Karolis Koncevičius
</p>


<h3>Examples</h3>

<pre><code class='language-R'># construct annmatrix object
x &lt;- matrix(rnorm(20*10), 20, 10)

coldata &lt;- data.frame(group  = rep(c("case", "control"), each = 5),
                      gender = sample(c("M", "F"), 10, replace = TRUE))

rowdata &lt;- data.frame(chr = sample(c("chr1", "chr2"), 20, replace = TRUE),
                      pos = runif(20, 0, 1000000))

X &lt;- annmatrix(x, rowdata, coldata)

pca &lt;- prcomp(t(X))
pca$rotation
pca$rotation$''

scores &lt;- t(pca$rotation) %*% X
scores@var_explained
scores$gender

</code></pre>

<hr>
<h2 id='print'>Print annmatrix Object</h2><span id='topic+print'></span><span id='topic+print.annmatrix'></span>

<h3>Description</h3>

<p>Functions that print an annmatrix object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'annmatrix'
print(x, nrow = 5, ncol = 5, digits = getOption("digits"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print_+3A_x">x</code></td>
<td>
<p>a matrix.</p>
</td></tr>
<tr><td><code id="print_+3A_nrow">nrow</code></td>
<td>
<p>number of rows to display (default is 5).</p>
</td></tr>
<tr><td><code id="print_+3A_ncol">ncol</code></td>
<td>
<p>number of columns to display (default is 5).</p>
</td></tr>
<tr><td><code id="print_+3A_digits">digits</code></td>
<td>
<p>number of digits to display (default set to getOptions(&quot;digits&quot;)).</p>
</td></tr>
<tr><td><code id="print_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>annmatrix objects are printed in a shortened form (5 rows and 5 columns by default).
In addition the function displays information about available meta-data for rows and columns.
</p>


<h3>Value</h3>

<p>invisibly returns annmatrix object.
</p>


<h3>Author(s)</h3>

<p>Karolis Koncevičius
</p>


<h3>Examples</h3>

<pre><code class='language-R'># construct annmatrix object
x &lt;- matrix(rnorm(20*10), 20, 10)

coldata &lt;- data.frame(group  = rep(c("case", "control"), each = 5),
                      gender = sample(c("M", "F"), 10, replace = TRUE))

rowdata &lt;- data.frame(chr = sample(c("chr1", "chr2"), 20, replace = TRUE),
                      pos = runif(20, 0, 1000000))

X &lt;- annmatrix(x, rowdata, coldata)

print(X)
print(X, 10, 5)
print(X, 2, 2)

# also works with a list-based matrix
x &lt;- matrix(list(mtcars, iris3, USAccDeaths, rivers), ncol=2)
print(x)
X &lt;- annmatrix(x)
print(X)

</code></pre>

<hr>
<h2 id='scale'>Scaling and Centering of annmatrix Objects</h2><span id='topic+scale'></span><span id='topic+scale.annmatrix'></span>

<h3>Description</h3>

<p>Centers and scales the columns of an annmatrix object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'annmatrix'
scale(x, center = TRUE, scale = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scale_+3A_x">x</code></td>
<td>
<p>annmatrix object.</p>
</td></tr>
<tr><td><code id="scale_+3A_center">center</code></td>
<td>
<p>either a logical value or a numeric vector of length equal to the number of columns of 'x' (default is TRUE).</p>
</td></tr>
<tr><td><code id="scale_+3A_scale">scale</code></td>
<td>
<p>either a logical value or a numeric vector of length equal to the number of columns of 'x' (default is TRUE).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Behaves exactly as <code>scale</code> on a regular matrix with the preservation of 'annmatrix' class being the only difference.
</p>


<h3>Value</h3>

<p>The centered and/or scaled annmatrix object with additional attributes &quot;scaled:center&quot; and &quot;scaled:scale&quot; holding the numbers used for centering and scaling of each column.
</p>


<h3>Author(s)</h3>

<p>Karolis Koncevičius
</p>


<h3>See Also</h3>

<p><code>scale.default</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># construct annmatrix object
x &lt;- matrix(rnorm(20*10), 20, 10)

coldata &lt;- data.frame(group  = rep(c("case", "control"), each = 5),
                      gender = sample(c("M", "F"), 10, replace = TRUE))

rowdata &lt;- data.frame(chr = sample(c("chr1", "chr2"), 20, replace = TRUE),
                      pos = runif(20, 0, 1000000))

X &lt;- annmatrix(x, rowdata, coldata)

scale(X)
scale(X, center = colMeans(X))

</code></pre>

<hr>
<h2 id='stack'>Stack an annmatrix object</h2><span id='topic+stack'></span><span id='topic+stack.annmatrix'></span>

<h3>Description</h3>

<p>Turns annmatrix into a data frame by transforming the matrix along with row and column annotations into separate data frame columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'annmatrix'
stack(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stack_+3A_x">x</code></td>
<td>
<p>annmatrix object.</p>
</td></tr>
<tr><td><code id="stack_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>transposed annmatrix object
</p>


<h3>Author(s)</h3>

<p>Karolis Koncevičius
</p>


<h3>Examples</h3>

<pre><code class='language-R'># construct annmatrix object
x &lt;- matrix(rnorm(20*10), 20, 10)

coldata &lt;- data.frame(group  = rep(c("case", "control"), each = 5),
                      gender = sample(c("M", "F"), 10, replace = TRUE))

rowdata &lt;- data.frame(chr = sample(c("chr1", "chr2"), 20, replace = TRUE),
                      pos = runif(20, 0, 1000000))

X &lt;- annmatrix(x, rowdata, coldata)

# stack all information into a long-format data.frame
Y &lt;- stack(X)

</code></pre>

<hr>
<h2 id='subset'>Subset annmatrix Objects</h2><span id='topic+subset'></span><span id='topic++5B.annmatrix'></span>

<h3>Description</h3>

<p>Methods for selecting a set of rows or columns from annmatrix while keeping
the associated annotations intact.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'annmatrix'
x[i, j, ..., drop = TRUE]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset_+3A_x">x</code></td>
<td>
<p>an R object.</p>
</td></tr>
<tr><td><code id="subset_+3A_i">i</code></td>
<td>
<p>subset for rows.</p>
</td></tr>
<tr><td><code id="subset_+3A_j">j</code></td>
<td>
<p>subset for columns.</p>
</td></tr>
<tr><td><code id="subset_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from methods.</p>
</td></tr>
<tr><td><code id="subset_+3A_drop">drop</code></td>
<td>
<p>if TRUE (default) subsetting a single row or column will returned a vector.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>X[i,j]</code> returns a selected subset of annmatrix object. Row and column
annotations are preserved and subsetted where needed. In the special case
when only one column or row is selected in order to be consistent with the
<code>matrix</code> behavior the dimensions of matrix are dropped and a vector is
returned. Just like in the case of matrices the additional argument
<code>drop=FALSE</code> can be provided in order to return a proper matrix
instead.
</p>


<h3>Value</h3>

<p>A selected subset of the provided 'annmatrix' object.
</p>


<h3>Author(s)</h3>

<p>Karolis Koncevičius
</p>


<h3>See Also</h3>

<p><code>as.annmatrix</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># construct annmatrix object
x &lt;- matrix(rnorm(20*10), 20, 10)

coldata &lt;- data.frame(group  = rep(c("case", "control"), each = 5),
                      gender = sample(c("M", "F"), 10, replace = TRUE))

rowdata &lt;- data.frame(chr = sample(c("chr1", "chr2"), 20, replace = TRUE),
                      pos = runif(20, 0, 1000000))

X &lt;- annmatrix(x, rowdata, coldata)

# annotations are preserved after subsetting
Y &lt;- X[X@chr == "chr1", X$group == "case"]
Y@chr
Y$''

Y[, 1]
Y[, 1, drop = FALSE]

</code></pre>

<hr>
<h2 id='transpose'>Transposing annmatrix Objects</h2><span id='topic+transpose'></span><span id='topic+t.annmatrix'></span>

<h3>Description</h3>

<p>Transpose annmatrix along with the associated row and column annotations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'annmatrix'
t(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="transpose_+3A_x">x</code></td>
<td>
<p>annmatrix object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>transposed annmatrix object with appropriately adjusted row and column annotations.
</p>


<h3>Author(s)</h3>

<p>Karolis Koncevičius
</p>


<h3>Examples</h3>

<pre><code class='language-R'># construct annmatrix object
x &lt;- matrix(rnorm(20*10), 20, 10)

coldata &lt;- data.frame(group  = rep(c("case", "control"), each = 5),
                      gender = sample(c("M", "F"), 10, replace = TRUE))

rowdata &lt;- data.frame(chr = sample(c("chr1", "chr2"), 20, replace = TRUE),
                      pos = runif(20, 0, 1000000))

X &lt;- annmatrix(x, rowdata, coldata)

# transposes the main matrix along with row and column annotations
Xt &lt;- t(X)

print(X)
print(Xt)

X@chr
Xt$chr

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
