<!DOCTYPE html><html><head><title>Help for package AnnoProbe</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {AnnoProbe}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#annoGene'><p>Annotate gene IDs according to GTF files in gencode</p></a></li>
<li><a href='#check_diff_genes'><p>Check a list of genes how they show difference.</p></a></li>
<li><a href='#checkGPL'><p>Check whether the input gpl in our platform list or not</p></a></li>
<li><a href='#deg_heatmap'><p>draw a heatmap for DEG result</p></a></li>
<li><a href='#deg_volcano'><p>draw a volcano for DEG result</p></a></li>
<li><a href='#filterEM'><p>Filter expression matrix based on annotation</p></a></li>
<li><a href='#geoChina'><p>Download expression dataset by GSE id</p></a></li>
<li><a href='#getGPLList'><p>Get all GPL list in our package</p>
<code>getGPLList</code> returns all the GPL number checklist stored in package</a></li>
<li><a href='#GSE27533'><p>An example dataset</p></a></li>
<li><a href='#GSE95166'><p>An example dataset</p></a></li>
<li><a href='#idmap'><p>Get Probe Annotation</p></a></li>
<li><a href='#printGPLInfo'><p>Print GPL information</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Annotate the Gene Symbols for Probes in Expression Array</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-11-12</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yonghe Xia &lt;xiayh17@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>We curated 147 of expression array, from 3 species(human,mouse,rat), 
              3 companies('Affymetrix','Illumina','Agilent'), 
              by aligning the 'Fasta' sequences of all probes of each platform to their corresponding reference genome, 
              and then annotate them to genes. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License (&ge; 2)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/jmzeng1314/AnnoProbe">https://github.com/jmzeng1314/AnnoProbe</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, DT, ggpubr, pheatmap, utils, methods, Biobase, stats,
xml2, httr, curl</td>
</tr>
<tr>
<td>biocViews:</td>
<td>DataImport, Microarray, OneChannel, SAGE, Software,
TwoChannel</td>
</tr>
<tr>
<td>Suggests:</td>
<td>limma, GEOquery, knitr, rmarkdown</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-11-12 11:39:35 UTC; xiayh</td>
</tr>
<tr>
<td>Author:</td>
<td>Jianming Zeng [aut],
  Yujia Xiang [aut],
  Yonghe Xia [ctb, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-11-14 08:30:11 UTC</td>
</tr>
</table>
<hr>
<h2 id='annoGene'>Annotate gene IDs according to GTF files in gencode</h2><span id='topic+annoGene'></span>

<h3>Description</h3>

<p>annoGene will return a data.frame of gene information or write them to a file (csv or html format).
The user should set a list of genes to be annotated, with  &quot;ENSEMBL&quot; or &quot;SYMBOL&quot; style.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annoGene(IDs, ID_type, species = "human", out_file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annoGene_+3A_ids">IDs</code></td>
<td>
<p>a list of genes</p>
</td></tr>
<tr><td><code id="annoGene_+3A_id_type">ID_type</code></td>
<td>
<p>the type of input IDs, should be &quot;ENSEMBL&quot; or &quot;SYMBOL&quot;</p>
</td></tr>
<tr><td><code id="annoGene_+3A_species">species</code></td>
<td>
<p>choose human or mouse, or rat, default: human</p>
</td></tr>
<tr><td><code id="annoGene_+3A_out_file">out_file</code></td>
<td>
<p>the filename, should be &quot;.csv&quot; or &quot;.html&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe which columns contain genesymbol, biotypes, ensembl ids and the positions of genes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>IDs &lt;- c("DDX11L1", "MIR6859-1", "OR4G4P", "OR4F5")
ID_type = "SYMBOL"
annoGene(IDs, ID_type)

annoGene(IDs, ID_type,out_file = tempfile(fileext = ".html"))
annoGene(IDs, ID_type,out_file = tempfile(fileext = ".csv"))

</code></pre>

<hr>
<h2 id='check_diff_genes'>Check a list of genes how they show difference.</h2><span id='topic+check_diff_genes'></span>

<h3>Description</h3>

<p>How does a gene or a list of genes show difference between two group.
The boxplot or heatmap will be drawed.
just a wrap function of ggpubr and pheatmap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_diff_genes(gene, genes_expr, group_list)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_diff_genes_+3A_gene">gene</code></td>
<td>
<p>A vector contains all gene ids of interest. Gene ids should
be gene symbol.</p>
</td></tr>
<tr><td><code id="check_diff_genes_+3A_genes_expr">genes_expr</code></td>
<td>
<p>An expression matrix, the rownames should be  gene symbol.</p>
</td></tr>
<tr><td><code id="check_diff_genes_+3A_group_list">group_list</code></td>
<td>
<p>A vector contains the group information of each samples in  expression matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A figure : boxplot or heatmap
</p>


<h3>Examples</h3>

<pre><code class='language-R'>attach(GSE95166)
check_diff_genes('LRCH3',genes_expr,group_list )

x=DEG$logFC
names(x)=rownames(DEG)
cg=c(names(head(sort(x),100)),  names(tail(sort(x),100)))
check_diff_genes(cg,genes_expr,group_list )

</code></pre>

<hr>
<h2 id='checkGPL'>Check whether the input gpl in our platform list or not</h2><span id='topic+checkGPL'></span>

<h3>Description</h3>

<p>Check whether the input gpl in our platform list or not
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkGPL(GPL = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkGPL_+3A_gpl">GPL</code></td>
<td>
<p>GPL(GEO platform) number, eg: GPL570</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a boolean value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>checkGPL('GPL570')
checkGPL('GPL15314')
checkGPL('GPL10558')
</code></pre>

<hr>
<h2 id='deg_heatmap'>draw a heatmap for DEG result</h2><span id='topic+deg_heatmap'></span>

<h3>Description</h3>

<p><code>deg_heatmap</code> will draw a heatmap for you.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deg_heatmap(deg, genes_expr, group_list, topn = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="deg_heatmap_+3A_deg">deg</code></td>
<td>
<p>the result from limma.</p>
</td></tr>
<tr><td><code id="deg_heatmap_+3A_genes_expr">genes_expr</code></td>
<td>
<p>the expression matrix</p>
</td></tr>
<tr><td><code id="deg_heatmap_+3A_group_list">group_list</code></td>
<td>
<p>a vector</p>
</td></tr>
<tr><td><code id="deg_heatmap_+3A_topn">topn</code></td>
<td>
<p>the number of genes in heatmap, default:20</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot2 style figure.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>attach(GSE27533)
deg_heatmap(DEG,genes_expr,group_list)
</code></pre>

<hr>
<h2 id='deg_volcano'>draw a volcano for DEG result</h2><span id='topic+deg_volcano'></span>

<h3>Description</h3>

<p><code>deg_volcano</code> will draw a volcano for you.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deg_volcano(need_deg, style = 1, p_thred = 0.05, logFC_thred = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="deg_volcano_+3A_need_deg">need_deg</code></td>
<td>
<p>should be 3 columns : gene, logFC, p.value(or p.adjust</p>
</td></tr>
<tr><td><code id="deg_volcano_+3A_style">style</code></td>
<td>
<p>you can try 1 or 2, default: 1</p>
</td></tr>
<tr><td><code id="deg_volcano_+3A_p_thred">p_thred</code></td>
<td>
<p>default:0.05</p>
</td></tr>
<tr><td><code id="deg_volcano_+3A_logfc_thred">logFC_thred</code></td>
<td>
<p>default:1</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot2 style figure.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>deg=GSE27533$DEG
need_deg=data.frame(symbols=rownames(deg), logFC=deg$logFC, p=deg$P.Value)
deg_volcano(need_deg,2)

deg_volcano(need_deg,1)

</code></pre>

<hr>
<h2 id='filterEM'>Filter expression matrix based on annotation</h2><span id='topic+filterEM'></span>

<h3>Description</h3>

<p><code>filterEM</code> will annotate the probes in expression matrix and remove the duplicated gene symbols.
because there will be many probes mapped to same genes, we will only keep the max value one.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterEM(probes_expr, probe2gene)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filterEM_+3A_probes_expr">probes_expr</code></td>
<td>
<p>is an expression matrix which rownames are probes of probe2gene and each column is a sample</p>
</td></tr>
<tr><td><code id="filterEM_+3A_probe2gene">probe2gene</code></td>
<td>
<p>the first column is probes and the second column is corresponding gene symbols</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a expression matrix which has been filtered duplicated gene symbols
</p>


<h3>Examples</h3>

<pre><code class='language-R'>attach(GSE95166)
# head(probes_expr)
# head(probe2gene)
genes_expr &lt;- filterEM(probes_expr,probe2gene)
# head(genes_expr)
</code></pre>

<hr>
<h2 id='geoChina'>Download expression dataset by GSE id</h2><span id='topic+geoChina'></span>

<h3>Description</h3>

<p><code>geoChina</code> will download the expression matrix and phenotype data as ExpressionSet format
from cloud in mainland China,
it's a alternative method for getGEO function from GEOquery package.
geoChina('gse1009') is the same as eSet=getGEO('gse1009', getGPL = F)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geoChina(gse = "GSE2546", mirror = "tencent", destdir = getwd())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geoChina_+3A_gse">gse</code></td>
<td>
<p>input GSE id, such as GSE1009, GSE2546, gse1009.</p>
</td></tr>
<tr><td><code id="geoChina_+3A_mirror">mirror</code></td>
<td>
<p>&quot;tencent&quot; only for now.</p>
</td></tr>
<tr><td><code id="geoChina_+3A_destdir">destdir</code></td>
<td>
<p>The destination directory for data downloads.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of ExpressionSet, which contains the  expression matrix and phenotype data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
geoChina('GSE1009',destdir=tempdir())

## End(Not run)
</code></pre>

<hr>
<h2 id='getGPLList'>Get all GPL list in our package
<code>getGPLList</code> returns all the GPL number checklist stored in package</h2><span id='topic+getGPLList'></span>

<h3>Description</h3>

<p>Get all GPL list in our package
<code>getGPLList</code> returns all the GPL number checklist stored in package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGPLList()
</code></pre>


<h3>Value</h3>

<p>a data.frame which contains the gpl and name of array.
</p>

<hr>
<h2 id='GSE27533'>An example dataset</h2><span id='topic+GSE27533'></span>

<h3>Description</h3>

<p>A dataset containing genes_expr, group_list, DEG
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GSE27533
</code></pre>


<h3>Format</h3>

<p>A list with 6 elements:
</p>

<dl>
<dt>genes_expr</dt><dd><p>genes_expr, genes_expr</p>
</dd>
<dt>DEG</dt><dd><p>DEG, DEG</p>
</dd>
</dl>
<p>...

</p>

<hr>
<h2 id='GSE95166'>An example dataset</h2><span id='topic+GSE95166'></span>

<h3>Description</h3>

<p>A dataset containing eSet, probes_expr, probe2gene, genes_expr, group_list, DEG
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GSE95166
</code></pre>


<h3>Format</h3>

<p>A list with 6 elements:
</p>

<dl>
<dt>probes_expr</dt><dd><p>probes_expr, probes_expr</p>
</dd>
<dt>probe2gene</dt><dd><p>probe2gene, probe2gene</p>
</dd>
<dt>genes_expr</dt><dd><p>genes_expr, genes_expr</p>
</dd>
<dt>group_list</dt><dd><p>group_list, group_list</p>
</dd>
<dt>DEG</dt><dd><p>DEG, DEG</p>
</dd>
</dl>
<p>...

</p>

<hr>
<h2 id='idmap'>Get Probe Annotation</h2><span id='topic+idmap'></span>

<h3>Description</h3>

<p><code>idmap</code> returns probe annotations for input gpl
</p>


<h3>Usage</h3>

<pre><code class='language-R'>idmap(gpl = "GPL570", type = "bioc", mirror = "tencent", destdir = getwd())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="idmap_+3A_gpl">gpl</code></td>
<td>
<p>GPL(GEO platform) number, eg: GPL570</p>
</td></tr>
<tr><td><code id="idmap_+3A_type">type</code></td>
<td>
<p>source of probe anntation stored, one of &quot;pipe&quot;, &quot;bioc&quot;, &quot;soft&quot;, default:&quot;pipe&quot;</p>
</td></tr>
<tr><td><code id="idmap_+3A_mirror">mirror</code></td>
<td>
<p>&quot;tencent&quot; only for now</p>
</td></tr>
<tr><td><code id="idmap_+3A_destdir">destdir</code></td>
<td>
<p>The destination directory for data downloads.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>probe annotaions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ids=idmap('GPL570',destdir=tempdir())

ids=idmap('GPL570',type='soft',destdir=tempdir())
ids=idmap('GPL18084',type='pipe',destdir=tempdir())

</code></pre>

<hr>
<h2 id='printGPLInfo'>Print GPL information</h2><span id='topic+printGPLInfo'></span>

<h3>Description</h3>

<p>Print GPL information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>printGPLInfo(GPL = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="printGPLInfo_+3A_gpl">GPL</code></td>
<td>
<p>GPL(GEO platform) number, eg: GPL570</p>
</td></tr>
</table>


<h3>Value</h3>

<p>print detail information of the input GEO platform
</p>


<h3>Examples</h3>

<pre><code class='language-R'>printGPLInfo('GPL93')
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
