<!DOCTYPE html><html lang="en"><head><title>Help for package AnnotationBustR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {AnnotationBustR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AnnotationBust'><p>Breaks up genbank sequences into their annotated components based on a set of search terms and writes each subsequence of interest to a FASTA for each accession number supplied.</p></a></li>
<li><a href='#AnnotationBustR'><p>An R package to extract sub-sequences from GenBank annotations under different synonyms</p></a></li>
<li><a href='#cpDNAterms'><p>Chloroplast DNA (cpDNA) Search Terms</p></a></li>
<li><a href='#FindLongestSeq'><p>Find the longest sequence for each species from a list of GenBank accession numbers.</p></a></li>
<li><a href='#MergeSearchTerms'><p>Merging of two tables containing search terms to expand search term database for the AnnotationBust function.</p></a></li>
<li><a href='#mtDNAterms'><p>Mitochondrial DNA Search Terms for Animals</p></a></li>
<li><a href='#mtDNAtermsPlants'><p>Mitochondrial DNA Search Terms for Plants</p></a></li>
<li><a href='#rDNAterms'><p>Ribosomal DNA (rDNA) Search Terms</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>1.3.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-09-23</td>
</tr>
<tr>
<td>Title:</td>
<td>Extract Subsequences from GenBank Annotations</td>
</tr>
<tr>
<td>Author:</td>
<td>Samuel R. Borstein &lt;sam@borstein.com&gt;, Brian O'Meara &lt;bomeara@utk.edu&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Samuel R. Borstein &lt;sam@borstein.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape (&ge; 4.1), seqinr (&ge; 3.3-6)</td>
</tr>
<tr>
<td>Description:</td>
<td>Extraction of subsequences into FASTA files from GenBank annotations where gene names may vary among accessions. Borstein &amp; O'Meara (2018) &lt;<a href="https://doi.org/10.7717%2Fpeerj.5179">doi:10.7717/peerj.5179</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/sborstein/AnnotationBustR">https://github.com/sborstein/AnnotationBustR</a>,
<a href="https://www.ncbi.nlm.nih.gov/nuccore">https://www.ncbi.nlm.nih.gov/nuccore</a>,
<a href="https://peerj.com/articles/5179/">https://peerj.com/articles/5179/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/sborstein/AnnotationBustR/issues">https://github.com/sborstein/AnnotationBustR/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-09-24 03:08:57 UTC; sam</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-09-24 08:10:13 UTC</td>
</tr>
</table>
<hr>
<h2 id='AnnotationBust'>Breaks up genbank sequences into their annotated components based on a set of search terms and writes each subsequence of interest to a FASTA for each accession number supplied.</h2><span id='topic+AnnotationBust'></span>

<h3>Description</h3>

<p>Breaks up genbank sequences into their annotated components based on a set of search terms and writes each subsequence of interest to a FASTA for each accession number supplied.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AnnotationBust(
  Accessions,
  Terms,
  Duplicates = NULL,
  DuplicateInstances = NULL,
  TranslateSeqs = FALSE,
  TranslateCode = 1,
  DuplicateSpecies = FALSE,
  Prefix = NULL,
  TidyAccessions = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AnnotationBust_+3A_accessions">Accessions</code></td>
<td>
<p>A vector of INSDC GenBank accession numbers. Note that refseq numbers are not supported at the moment (i.e. prefixes like XM_ and NC_) will not work.</p>
</td></tr>
<tr><td><code id="AnnotationBust_+3A_terms">Terms</code></td>
<td>
<p>A data frame of search terms. Search terms for animal mitogenomes, nuclear rRNA, chloroplast genomes, and plant mitogenomes are pre-made and can be loaded using the data()function. Additional terms can be addded using the MergeSearchTerms function.</p>
</td></tr>
<tr><td><code id="AnnotationBust_+3A_duplicates">Duplicates</code></td>
<td>
<p>A vector of loci names that have more than one copy. Default is NULL</p>
</td></tr>
<tr><td><code id="AnnotationBust_+3A_duplicateinstances">DuplicateInstances</code></td>
<td>
<p>A numeric vector the length of Duplicates of the number of duplicates for each duplicated gene you wish to extract.Default is NULL</p>
</td></tr>
<tr><td><code id="AnnotationBust_+3A_translateseqs">TranslateSeqs</code></td>
<td>
<p>Logical as to whether or not the sequence should be translated to the corresponding peptide sequence. Default is FALSE. Note that this only makes sense to list as TRUE for protein coding sequences.</p>
</td></tr>
<tr><td><code id="AnnotationBust_+3A_translatecode">TranslateCode</code></td>
<td>
<p>Numerical representing the GenBank translation code for which sequences should be translated under. Default is 1. For all options see ?seqinr::getTrans. Some possible useful ones are: 2. Vertebrate Mitochondrial, 5. Invertebrate Mitochondrial, and 11. bacterial+plantplastid</p>
</td></tr>
<tr><td><code id="AnnotationBust_+3A_duplicatespecies">DuplicateSpecies</code></td>
<td>
<p>Logical. As to whether there are duplicate individuals per species. If TRUE, adds the accession number to the fasta file</p>
</td></tr>
<tr><td><code id="AnnotationBust_+3A_prefix">Prefix</code></td>
<td>
<p>Character If a prefix is specified, all output FASTA files written will begin with the prefix. Default is NULL.</p>
</td></tr>
<tr><td><code id="AnnotationBust_+3A_tidyaccessions">TidyAccessions</code></td>
<td>
<p>Logical as to whether the accession table should have a single row per species. If numerous accessions for a species occur, they will be seperated by a comma in the accession table. Default=TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The AnnotationBust function takes a vector of accession numbers and a data frame of search terms and extracts subsequences from genomes or concatenated sequences.
This function requires a steady internet connection. It writes files in the FASTA format to the working directory and returns an accession table. Files append, so use different prefixes between runs, otherwise they will be added to the current files in the working directory if the names are the same. AnnoitationBustR comes with pre-made
search terms for metazoan mitogenomes, plant mitogenomes, chloroplast genomes, and rDNA that can be loaded using data(mtDNAterms), data(mtDNAtermsPlants),data(cpDNAterms), and data(rDNAterms) respectively.
Search terms can be completely made by the user as long as they follow a similar format with three columns. The first, Locus, should contain the name of the locus that will also be used to name the files. We recommend following
a similar naming convention to what we currently have in the pre-made data frames to ensure that files are named properly, characters like &quot;-&quot; or &quot;.&quot;, and names starting with numbers should be avoided as it can cause errors with R.
The second column, Type, contains the type of subsequence it is, with options being CDS, rRNA, tRNA, misc_RNA, Intro, Exon, and D_Loop. The last column, Name, consists of a possible
name for the locus of interest as it might appear in an annotation. For numerous synonyms for the same locus, one should have each synonym as its own row.An additional fourth column is needed for extracting introns/exons.
This column, called IntronExonNumber should contain the number of the intron to extract. If extracting both introns/exons and non-intron/exon sequences the fourth column should be NA for non-intron/exon sequence types. See the examples below and the vignette for detailed examples on extracting intron and exons.
It is possible that some subsequences are not fully annotated on ACNUC and, therefore, are not extractable. These will return in the accession table as &quot;type not fully Ann&quot;. It is also possible that the sequence has no annotations at all, for which it will return &quot;No Ann. For&quot;.
If the function returns &quot;Acc. Not Found&quot;, the accession number supplied could not be found on NCBI. If &quot;Not On ACNUC GenBank&quot; is returned, the accession is not available through AcNUC.
This may be due to ACNUC not being fully up to date. To see the last time ACNUC was updated, run seqinr::choosebank(&quot;genbank&quot;, infobank=T).
</p>
<p>For a more detailed walkthrough on using AnnotationBust you can access the vignette with vignette(&quot;AnnotationBustR).
</p>


<h3>Value</h3>

<p>Writes a fasta file(s) to the current working directory selected for each unique subsequence of interest in Terms containing all the accession numbers the subsequence was found in.
</p>
<p>An accesion table of class data.frame.
</p>


<h3>References</h3>

<p>Borstein, Samuel R., and Brian C. O'Meara. &quot;AnnotationBustR: An R package to extract subsequences from GenBank annotations.&quot; PeerJ Preprints 5 (2017): e2920v1.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Create vector of three NCBI accessions of rDNA toget subsequences of and load rDNA terms.
ncbi.accessions&lt;-c("FJ706295","FJ706343","FJ706292")
data(rDNAterms)#load rDNA search terms from AnnotationBustR
#Run AnnotationBustR and write files to working directory
my.sequences&lt;-AnnotationBust(ncbi.accessions, rDNAterms, DuplicateSpecies=TRUE,
Prefix="Example1")
my.sequences#Return the accession table for each species.

###Example With matK CDS and trnK Introns/Exons##
#Subset out matK from cpDNAterms
cds.terms&lt;-subset(cpDNAterms,cpDNAterms$Locus=="matK")
#Create a vecotr of NA so we can merge with the search terms for introns and exons
cds.terms&lt;-cbind(cds.terms,(rep(NA,length(cds.terms$Locus))))
colnames(cds.terms)[4]&lt;-"IntronExonNumber"

#Prepare a search term table for the intron and exons to remove
#We can start with the cpDNAterms for trnK
IntronExon.terms&lt;-subset(cpDNAterms,cpDNAterms$Locus=="trnK")
#As we want to go for two exons, we will want the synonyms repeated as we are doing and intron
#and an exon
IntronExon.terms&lt;-rbind(IntronExon.terms,IntronExon.terms)#duplicate the terms
#rep the sequence type we want to extract
IntronExon.terms$Type&lt;-rep(c("Intron","Intron","Exon","Exon"))
IntronExon.terms$Locus&lt;-rep(c("trnK_Intron","trnK_Exon2"),each=2)
IntronExon.terms&lt;-cbind(IntronExon.terms,rep(c(1,1,2,2)))#Add intron/exon number info
#change column name for number info for IntronExon name
colnames(IntronExon.terms)[4]&lt;-"IntronExonNumber"
#We can then merge everything together with MergeSearchTerms terms
IntronExonExampleTerms&lt;-MergeSearchTerms(IntronExon.terms,cds.terms)

#Run AnnotationBust
IntronExon.example&lt;-AnnotationBust(Accessions=c("KX687911.1", "KX687910.1"),
Terms=IntronExonExampleTerms, Prefix="DemoIntronExon")

## End(Not run)
</code></pre>

<hr>
<h2 id='AnnotationBustR'>An R package to extract sub-sequences from GenBank annotations under different synonyms</h2><span id='topic+AnnotationBustR'></span>

<h3>Description</h3>

<p>An R package to extract sub-sequences from GenBank annotations under different synonyms.
</p>


<h3>Details</h3>

<p>Package: AnnotationBustR
</p>
<p>Type: Package
</p>
<p>Title: An R package to extract sub-sequences from GenBank annotations under different synonyms
</p>
<p>Version: 1.1
</p>
<p>Date: 2017-8-15
</p>
<p>License: GPL (&gt;= 2)
</p>
<p>This package allows users to quickly extract sub-sequences from GenBank accession numbers that may be annotated
under different synonyms. It writes these sub-sequences to FASTA files and creates a corresponding accession
table. The package comes with pre-made search terms with synonyms. A vignette going over the basic functions and
how to use them can be accessed with vignette(&quot;AnnotationBustR-vignette&quot;).
</p>


<h3>Author(s)</h3>

<p>Samuel Borstein, Brian O'Meara.
Maintainer: Samuel Borstein &lt;sborstei@vols.utk.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+AnnotationBust">AnnotationBust</a></code>,<code><a href="#topic+cpDNAterms">cpDNAterms</a></code>,<code><a href="#topic+FindLongestSeq">FindLongestSeq</a></code>,<code><a href="#topic+MergeSearchTerms">MergeSearchTerms</a></code>,<code><a href="#topic+mtDNAterms">mtDNAterms</a></code>,<code><a href="#topic+rDNAterms">rDNAterms</a></code>
</p>

<hr>
<h2 id='cpDNAterms'>Chloroplast DNA (cpDNA) Search Terms</h2><span id='topic+cpDNAterms'></span>

<h3>Description</h3>

<p>A data frame containing search terms for Chloroplast loci. Can be subset for loci of 
interest. Columns are as follows and users should follow the column format if they wish to
add search terms using the MergeSearchTerms function:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cpDNAterms
</code></pre>


<h3>Format</h3>

<p>A data frame of of 364 rows and 3 columns
</p>

<ul>
<li><p> Locus: Locus name, FASTA files will be written with this name
</p>
</li>
<li><p> Type: Type of subsequence, either CDS,tRNA,rRNA, or misc_RNA
</p>
</li>
<li><p> Name:Name of synonym for a locus to search for
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+MergeSearchTerms">MergeSearchTerms</a></code>
</p>

<hr>
<h2 id='FindLongestSeq'>Find the longest sequence for each species from a list of GenBank accession numbers.</h2><span id='topic+FindLongestSeq'></span>

<h3>Description</h3>

<p>Find the longest sequence for each species from a list of GenBank accession numbers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FindLongestSeq(Accessions)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FindLongestSeq_+3A_accessions">Accessions</code></td>
<td>
<p>A vector of GenBank accession numbers.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For a set of GenBank accession numbers, this will return the longest sequence for in the set for species.
</p>


<h3>Value</h3>

<p>A list of genbank accessions numbers for the longest sequence for each taxon in a list of accession numbers.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#a vector of 4 genbank accessions, there are two for each species.
genbank.accessions&lt;-c("KP978059.1","KP978060.1","JX516105.1","JX516111.1")
## Not run: 
#returns the longest sequence respectively for the two species.
long.seq.result&lt;-FindLongestSeq(genbank.accessions)

## End(Not run)
</code></pre>

<hr>
<h2 id='MergeSearchTerms'>Merging of two tables containing search terms to expand search term database for the AnnotationBust function.</h2><span id='topic+MergeSearchTerms'></span>

<h3>Description</h3>

<p>This function merges two data frames with search terms. This allows users to easily add search terms to data frames (either their
own or ones included in this package using data() as GenBank annotations for the same genes may vary in gene name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MergeSearchTerms(..., SortGenes = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MergeSearchTerms_+3A_...">...</code></td>
<td>
<p>the data frames of search terms you want to combine into a single data frame The Data frame(s) should have stringsAsFactors=FALSE listed if you want to sort them.</p>
</td></tr>
<tr><td><code id="MergeSearchTerms_+3A_sortgenes">SortGenes</code></td>
<td>
<p>Should the final data frame be sorted by gene name? Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new merged data frame with all the search terms combined from the lists supplied. If sort.gene=TRUE, genes will be sorted by name.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load the list of search terms for mitochondrial genes
data(mtDNAterms) 

#Make a data.frame of new terms to add.
#This is a dummy example for a non-real annoation of COI, but lets pretend it is real.
add.name&lt;-data.frame("COI","CDS", "CX1")

# make the column names the same for combination.
colnames(add.name)&lt;-colnames(mtDNAterms)

#Run the merge search term function without sorting based on gene name.
new.terms&lt;-MergeSearchTerms(add.name, mtDNAterms, SortGenes=FALSE)

#Run the merge search term function with sorting based on gene name.
new.terms&lt;-MergeSearchTerms(add.name, mtDNAterms, SortGenes=TRUE)

#Merge search terms and create an additional column for introns and/or exons to
#In this example, add the trnK intron to the terms
#create empty IntornExonNumber column for non-intron/exons
cp.terms&lt;-cbind(cpDNAterms,rep(NA,length(cpDNAterms$Name)))
colnames(cp.terms)[4]&lt;-"IntronExonNumber"#Name the column IntronExonNumber
trnK.intron.terms&lt;-subset(cpDNAterms,cpDNAterms$Locus=="trnK")#subset trnK
#Create a vector of 1's the same length as the number of rows for trnK
trnK.terms&lt;-cbind(trnK.intron.terms,rep(1,length(trnK.intron.terms$Name)))
colnames(trnK.terms)[4]&lt;-"IntronExonNumber"#Name the column IntronExonNumber
#Use MergeSearchTerms to merge the modified cpDNAterms and new intron terms
all.terms&lt;-MergeSearchTerms(cp.terms,trnK.terms)
</code></pre>

<hr>
<h2 id='mtDNAterms'>Mitochondrial DNA Search Terms for Animals</h2><span id='topic+mtDNAterms'></span>

<h3>Description</h3>

<p>A data frame containing search terms for animal mitochondrial loci. Can be subset for loci of 
interest. Columns are as follows and users should follow the column format if they wish to
add search terms using the MergeSearchTerms function:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mtDNAterms
</code></pre>


<h3>Format</h3>

<p>A data frame of of 253 rows and 3 columns
</p>

<ul>
<li><p> Locus: Locus name, FASTA files will be written with this name
</p>
</li>
<li><p> Type: Type of subsequence, either CDS,tRNA,rRNA,misc_RNA, or D-loop
</p>
</li>
<li><p> Name:Name of synonym for a locus to search for
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+MergeSearchTerms">MergeSearchTerms</a></code>
</p>

<hr>
<h2 id='mtDNAtermsPlants'>Mitochondrial DNA Search Terms for Plants</h2><span id='topic+mtDNAtermsPlants'></span>

<h3>Description</h3>

<p>A data frame containing search terms for plant mitochondrial loci. Can be subset for loci of 
interest. Columns are as follows and users should follow the column format if they wish to
add search terms using the MergeSearchTerms function:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mtDNAtermsPlants
</code></pre>


<h3>Format</h3>

<p>A data frame of of 248 rows and 3 columns
</p>

<ul>
<li><p> Locus: Locus name, FASTA files will be written with this name
</p>
</li>
<li><p> Type: Type of subsequence, either CDS,tRNA,rRNA,misc_RNA, or D-loop
</p>
</li>
<li><p> Name:Name of synonym for a locus to search for
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+MergeSearchTerms">MergeSearchTerms</a></code>
</p>

<hr>
<h2 id='rDNAterms'>Ribosomal DNA (rDNA) Search Terms</h2><span id='topic+rDNAterms'></span>

<h3>Description</h3>

<p>A data frame containing search terms for ribosomal RNA loci. Can be subset for loci of 
interest. Columns are as follows and users should follow the column format if they wish to
add search terms using the MergeSearchTerms function:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rDNAterms
</code></pre>


<h3>Format</h3>

<p>A data frame of of 7 rows and 3 columns
</p>

<ul>
<li><p> Locus: Locus name, FASTA files will be written with this name
</p>
</li>
<li><p> Type: Type of subsequence, either rRNA or misc_RNA
</p>
</li>
<li><p> Name:Name of synonym for a locus to search for
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+MergeSearchTerms">MergeSearchTerms</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
