<!DOCTYPE html><html><head><title>Help for package AntAngioCOOL</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {AntAngioCOOL}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AntAngioCOOL'><p>AntAngioCOOL</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Anti-Angiogenic Peptide Prediction</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-08-01</td>
</tr>
<tr>
<td>Author:</td>
<td>Babak Khorsand &lt;khorsand@yahoo.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Javad Zahiri &lt;zahiri@modares.ac.ir&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Machine learning based package to predict anti-angiogenic peptides using heterogeneous sequence descriptors. 'AntAngioCOOL' exploits five descriptor types of a peptide of interest to do prediction including: pseudo amino acid composition, k-mer composition, k-mer composition (reduced alphabet), physico-chemical profile and atomic profile. According to the obtained results, 'AntAngioCOOL' reached to a satisfactory performance in anti-angiogenic peptide prediction on a benchmark non-redundant independent test dataset.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>caret,rJava,RWeka,rpart, R(&ge; 2.10.0)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-08-01 09:53:32 UTC; Babak</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-08-01 14:01:06</td>
</tr>
</table>
<hr>
<h2 id='AntAngioCOOL'>AntAngioCOOL</h2><span id='topic+AntAngioCOOL'></span>

<h3>Description</h3>

<p>Machine learning based package to predict anti-angiogenic peptides using heterogeneous sequence descriptors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AntAngioCOOL(Input_Sequence, Classifier = 1, SF = FALSE, AF = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AntAngioCOOL_+3A_input_sequence">Input_Sequence</code></td>
<td>
<p>A peptide sequence</p>
</td></tr>
<tr><td><code id="AntAngioCOOL_+3A_classifier">Classifier</code></td>
<td>
<p>1 if you want to get the prediction from model with maximum Accuracy (according to the independent test reasults), 2 if maximum Sensivity is desired and 3 if maximum Specefity is desired.</p>
</td></tr>
<tr><td><code id="AntAngioCOOL_+3A_sf">SF</code></td>
<td>
<p>if True then all 2343 selected features (out of 175062 features) that had been used for prediction will be returned.</p>
</td></tr>
<tr><td><code id="AntAngioCOOL_+3A_af">AF</code></td>
<td>
<p>if True then all 175062 extracted features will be returned.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>AntAngioCOOL is a machine learning based package to predict anti-angiogenic peptides using heterogeneous sequence descriptors.
</p>
<p>This package consists of three different predictors according to the obtained performances on independent test set: sensitive model, specific model and accurate model. These models have been build using the gold standard dataset that published by Ramaprasad et al. (Ettayapuram Ramaprasad et al., 2015).
</p>
<p>Four different features have been used to encode peptides:
</p>
<p>1- Pseudo Amino Acid Composition  (PseAAC) that has been used effectively in predicting cell penetrating peptides (Chen, Chu, Huang, Kong, &amp; Cai, 2015). Despite the simple amino acid composition, PseAAC considers the sequence-order information of the peptide.
</p>
<p>2- K-mer composition that shows the fraction of all possible subsequences with length k in the given peptide. To compute k-mer composition features, reduced amino acid alphabet that proposed by Zahiri et al (Zahiri et al., 2014) has been exploited: the 20 alphabet of amino acids have been reduced to a new alphabet with size 8 according to 544 physicochemical and biochemical indices that extracted from AAIndex database (Kawashima et al., 2008) (C1=A, E, C2=I, L, F, M, V, C3=N, D, T, S, C4=G, C5=P, C6=R, K, Q, H, C7=Y, W, C8=C).  We have computed k-mer composition for k=2,3,4 for each peptide.
</p>
<p>3- Physico-chemical profile : To compute these features, 544 different physico-chemical indices have been extracted from AAINDEX (Kawashima et al., 2008). To remove redundant indices, a subset of indices with correlation coefficient less than 0.8 and greater than -0.8 has been selected. This feature type for 5 amino acids of N- termini (5-NT) and C-termini (5-CT).
</p>
<p>4- Atomic profile : A 50-dimentional feature vector has been used to encode each peptide according to its atomic properties (frequency of five types of atoms: C, H, N, O, S). For details of atomic composition for each 20 natural amino acid see Kumar et al., 2015.
</p>
<p>References
</p>
<p>1- Chen, L., Chu, C., Huang, T., Kong, X., &amp; Cai, Y.-D. (2015). Prediction and analysis of cell-penetrating peptides using pseudo-amino acid composition and random forest models. Amino Acids, 47(7), 1485-93. http://doi.org/10.1007/s00726-015-1974-5
</p>
<p>2- Ettayapuram Ramaprasad, A. S., Singh, S., Gajendra P. S, R., Venkatesan, S., Brem, S., Cotran, R., . Stephens, R. (2015). AntiAngioPred: A Server for Prediction of Anti-Angiogenic Peptides. PLOS ONE, 10(9), e0136990. http://doi.org/10.1371/journal.pone.0136990
</p>
<p>3- Kawashima, S., Pokarowski, P., Pokarowska, M., Kolinski, A., Katayama, T., &amp; Kanehisa, M. (2008). AAindex: amino acid index database, progress report 2008. Nucleic Acids Research, 36(Database issue), D202-5. http://doi.org/10.1093/nar/gkm998
</p>
<p>3- Kumar, R., Chaudhary, K., Singh Chauhan, J., Nagpal, G., Kumar, R., Sharma, M., &amp; Raghava, G. P. S. (2015). An in silico platform for predicting, screening and designing of antihypertensive peptides. Scientific Reports, 5, 12512. http://doi.org/10.1038/srep12512
</p>
<p>4- Zahiri, J., Mohammad-Noori, M., Ebrahimpour, R., Saadat, S., Bozorgmehr, J. H., Goldberg, T., &amp; Masoudi-Nejad, A. (2014). LocFuse: Human protein-protein interaction prediction via classifier fusion using protein localization information. Genomics, 104(6), 496-503.
</p>


<h3>Value</h3>

<p>If Predicted class (Anti-angiogenic/ Not anti-angiogenic) of the input peptide and a subset of descriptors upon request.
</p>


<h3>Author(s)</h3>

<p>Babak Khorsand
</p>


<h3>Examples</h3>

<pre><code class='language-R'>AntAngioCOOL("AAPFLECQGN",2,SF=TRUE)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
