<!DOCTYPE html><html><head><title>Help for package APackOfTheClones</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {APackOfTheClones}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AdjustAPOTC'><p>Adjust the parameters of the APackOfTheClones reduction in a seurat</p>
object</a></li>
<li><a href='#APackOfTheClones-package'><p>APackOfTheClones: Visualization of Clonal Expansion for Single Cell Immune Profiles</p></a></li>
<li><a href='#APOTCPlot'><p>Various variations of visualizations of clonal expansion post-RunAPOTC</p></a></li>
<li><a href='#clonal_expansion_plot'><p>DEFUNCT: Visualize T cell clonal expansion with a ball-packing plot.</p></a></li>
<li><a href='#combined_pbmc'><p>Example Multi-sampled T-cell seurat object with integrated TCR library</p></a></li>
<li><a href='#containsApotcRun'><p>Check for the existence of an APackOfTheClones run with its run id</p></a></li>
<li><a href='#count_clone_sizes'><p>DEPRECATED: count the number of clonotype sizes per cell cluster in a seurat</p>
object integrated with a TCR library</a></li>
<li><a href='#countCloneSizes'><p>count the number of clonotype sizes per cell cluster in a seurat</p>
object combined with a VDJ library</a></li>
<li><a href='#deleteApotcData'><p>Delete the results of an APackOfTheClones run</p></a></li>
<li><a href='#getApotcDataIds'><p>Get all run ids of previous RunAPOTC runs on a seurat object</p></a></li>
<li><a href='#getLastApotcDataId'><p>Get the object id of the most recent RunAPOTC run on a seurat object</p></a></li>
<li><a href='#getReductionCentroids'><p>Calculate seurat cluster centroids based on a Dimensional reduction</p></a></li>
<li><a href='#integrate_tcr'><p>DEPRECATED: Integrate a single TCR library into Seurat object metadata</p></a></li>
<li><a href='#mini_clonotype_data'><p>Artificially generated T cell receptor library</p></a></li>
<li><a href='#mini_seurat_obj'><p>Artificially generated Seurat object</p></a></li>
<li><a href='#RunAPOTC'><p>Run the APackOfTheClones method on a combined Seurat object for</p>
downstream visualization of clonal expansion</a></li>
<li><a href='#vizAPOTC'><p>Directly visualize clonal expansion of a combined seurat object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Visualization of Clonal Expansion for Single Cell Immune
Profiles</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Qile Yang &lt;qile.yang@berkeley.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Visualize clonal expansion via circle-packing. 'APackOfTheClones' extends 'scRepertoire' to produce a publication-ready visualization of clonal expansion at a single cell resolution, by representing expanded clones as differently sized circles. The method was originally implemented by Murray Christian and Ben Murrell in the following immunology study: Ma et al. (2021) &lt;<a href="https://doi.org/10.1126%2Fsciimmunol.abg6356">doi:10.1126/sciimmunol.abg6356</a>&gt;.</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Qile0317/APackOfTheClones/issues/">https://github.com/Qile0317/APackOfTheClones/issues/</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>LazyData:</td>
<td>false</td>
</tr>
<tr>
<td>Suggests:</td>
<td>cowplot, covr, knitr, rmarkdown, sf, spelling, testthat (&ge;
3.0.0), vdiffr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, dplyr, ggforce, ggplot2, grDevices, hash,
lifecycle, methods, Rcpp, rlang, Seurat</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://qile0317.github.io/APackOfTheClones/">https://qile0317.github.io/APackOfTheClones/</a></td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-10 16:25:25 UTC; qile</td>
</tr>
<tr>
<td>Author:</td>
<td>Qile Yang [cre, aut, cph]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-13 13:20:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='AdjustAPOTC'>Adjust the parameters of the APackOfTheClones reduction in a seurat
object</h2><span id='topic+AdjustAPOTC'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>If the user is unsatisfied with the clonal expansion plot that
was generated from <code>RunAPOTC</code> and <code>APOTCPlot</code>, this function has a range of
arguments to modify the data and/or parameters of the visualization. Note
that some of the arguments may conflict with eachother.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AdjustAPOTC(
  seurat_obj,
  reduction_base = NULL,
  clonecall = NULL,
  ...,
  extra_filter = NULL,
  run_id = NULL,
  new_rad_scale_factor = NULL,
  new_clone_scale_factor = NULL,
  repulse = FALSE,
  repulsion_threshold = 1,
  repulsion_strength = 1,
  max_repulsion_iter = 10L,
  relocate_cluster = NULL,
  relocation_coord = NULL,
  nudge_cluster = NULL,
  nudge_vector = NULL,
  recolor_cluster = NULL,
  new_color = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AdjustAPOTC_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>The seurat object to be adjusted.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_reduction_base">reduction_base</code></td>
<td>
<p>character. The seurat reduction to base the clonal
expansion plotting on. Defaults to <code>'umap'</code> but can be any reduction present
within the reductions slot of the input seurat object, including custom ones.
If ''pca'&ldquo;, the cluster coordinates will be based on PC1 and PC2.
However, generally APackOfTheClones is used for displaying UMAP and
occasionally t-SNE versions to intuitively highlight clonal expansion.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_clonecall">clonecall</code></td>
<td>
<p>character. The column name in the seurat object metadata to
use. See <code>scRepertoire</code> documentation for more information about this
parameter that is central to both packages.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_...">...</code></td>
<td>
<p>additional &quot;subsetting&quot; keyword arguments indicating the rows
corresponding to elements in the seurat object metadata that should be
filtered by. E.g., <code>seurat_clusters = c(1, 9, 10)</code> will filter the cells to
those in the <code>seurat_clusters</code> column with any of the values 1, 9, and 10.
Unfortunately, column names in the seurat object metadata cannot
conflict with the keyword arguments. <em><strong>MAJOR NOTE</strong></em> if any subsetting
keyword arguments are a <em>prefix</em> of any preceding argument names (e.g. a
column named <code>reduction</code> is a prefix of the <code>reduction_base</code> argument)
R will interpret it as the same argument unless <em>both</em> arguments
are named. Additionally, this means any subsequent arguments <em>must</em> be named.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_extra_filter">extra_filter</code></td>
<td>
<p>character. An additional string that should be formatted
<em>exactly</em> like a statement one would pass into <a href="dplyr.html#topic+filter">dplyr::filter</a> that does
<em>additional</em> filtering to cells in the seurat object - on top of the other
keyword arguments - based on the metadata. This means that it will be
logically AND'ed with any keyword argument filters. This is a more flexible
alternative / addition to the filtering keyword arguments. For example, if
one wanted to filter by the length of the amino acid sequence of TCRs, one
could pass in something like <code>extra_filter = "nchar(CTaa) - 1 &gt; 10"</code>. When
involving characters, ensure to enclose with single quotes.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_run_id">run_id</code></td>
<td>
<p>character. This will be the ID associated with the data of a
run, and will be used by other important functions like <a href="#topic+APOTCPlot">APOTCPlot</a> and
<a href="#topic+AdjustAPOTC">AdjustAPOTC</a>. Defaults to <code>NULL</code>, in which case the ID will be generated
in the following format:
</p>
<p><code style="white-space: pre;">&#8288;reduction_base;clonecall;keyword_arguments;extra_filter&#8288;</code>
</p>
<p>where if keyword arguments and extra_filter are underscore characters if
there was no input for the <code>...</code> and <code>extra_filter</code> parameters.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_new_rad_scale_factor">new_rad_scale_factor</code></td>
<td>
<p>a single numeric in (0, 1]. changes the
radius scaling factor of all circles.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_new_clone_scale_factor">new_clone_scale_factor</code></td>
<td>
<p>a single numeric in (0, 1]. changes the
clone_scale_factor</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_repulse">repulse</code></td>
<td>
<p>If <code>TRUE</code>, will attempt to push overlapping clusters away from
each other.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_repulsion_threshold">repulsion_threshold</code></td>
<td>
<p>numeric. The radius that clonal circle clusters
overlap is acceptable when repulsing.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_repulsion_strength">repulsion_strength</code></td>
<td>
<p>numeric. The smaller the value the less the
clusters repulse each other per iteration, and vice versa.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_max_repulsion_iter">max_repulsion_iter</code></td>
<td>
<p>integer. The number of repulsion iterations.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_relocate_cluster">relocate_cluster</code></td>
<td>
<p>numeric of arbitrary length. Indicates which
cluster(s) to relocate to new coordinates</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_relocation_coord">relocation_coord</code></td>
<td>
<p>numeric of length two or a list of numerics of length
two of length of <code>relocate_cluster</code>. If its a list, indicates each coordinate
that the clusters in <code>relocate_cluster</code> should move to. If its just a
numeric, then will relocate all clusters in <code>relocate_cluster</code> to the input,
which is likely not desired behavior, so this should only be convenience
syntax if <code>relocate_cluster</code> has length 1.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_nudge_cluster">nudge_cluster</code></td>
<td>
<p>numeric of arbitrary length. Indicates which
cluster(s) to &quot;nudge&quot;/translate their coordinate(s) by.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_nudge_vector">nudge_vector</code></td>
<td>
<p>numeric of length two or a list of numerics of length
two of length of <code>nudge_cluster</code>. If its a list, indicates each translation
vector (in other words, x-y coordinates) that the clusters in
<code>nudge_cluster</code> should be translate by. If its just a numeric, then will
translate all clusters in <code>nudge_cluster</code> by the input - which mostly is
syntactic sugar for translating a single cluster if the input of
<code>nudge_cluster</code> is of length 1.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_recolor_cluster">recolor_cluster</code></td>
<td>
<p>numeric of arbitrary length. Indicates which
cluster(s) to change their color by.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_new_color">new_color</code></td>
<td>
<p>character of arbitrary length. Indicates the corresponding
new colors that selected clusters in <code>recolor_cluster</code> should be changed to.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_verbose">verbose</code></td>
<td>
<p>logical. Decides if visual cues are displayed to the R console
of the progress.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The adjusted <code>seurat_obj</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># do an APackOfTheClones run
pbmc &lt;- RunAPOTC(get(data("combined_pbmc")), verbose = FALSE)

# adjust the rad_scale_factor, and nudge cluster 1 by (1, 1)
pbmc &lt;- AdjustAPOTC(
    pbmc,
    new_rad_scale_factor = 0.9,
    nudge_cluster = 1,
    nudge_vector = c(1, 1),
    verbose = FALSE
)

# plot the result
APOTCPlot(pbmc)

# perhaps multiple clusters need to be relocated and repulsed
pbmc &lt;- AdjustAPOTC(
    pbmc,
    relocate_cluster = c(5, 10),
    relocation_coord = list(c(2, 3.5), c(0, 5)),
    repulse = TRUE,
    verbose = FALSE
)

# plot again to check results
APOTCPlot(pbmc)

</code></pre>

<hr>
<h2 id='APackOfTheClones-package'>APackOfTheClones: Visualization of Clonal Expansion for Single Cell Immune Profiles</h2><span id='topic+APackOfTheClones'></span><span id='topic+APackOfTheClones-package'></span>

<h3>Description</h3>

<p><img src="../help/figures/logo.png" style='float: right' alt='logo' width='120' />
</p>
<p>Visualize clonal expansion via circle-packing. 'APackOfTheClones' extends 'scRepertoire' to produce a publication-ready visualization of clonal expansion at a single cell resolution, by representing expanded clones as differently sized circles. The method was originally implemented by Murray Christian and Ben Murrell in the following immunology study: Ma et al. (2021) <a href="https://doi.org/10.1126/sciimmunol.abg6356">doi:10.1126/sciimmunol.abg6356</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Qile Yang <a href="mailto:qile.yang@berkeley.edu">qile.yang@berkeley.edu</a> [copyright holder]
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://qile0317.github.io/APackOfTheClones/">https://qile0317.github.io/APackOfTheClones/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/Qile0317/APackOfTheClones/issues/">https://github.com/Qile0317/APackOfTheClones/issues/</a>
</p>
</li></ul>


<hr>
<h2 id='APOTCPlot'>Various variations of visualizations of clonal expansion post-RunAPOTC</h2><span id='topic+APOTCPlot'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>Given a seurat object with an 'apotc' (APackOfTheClones) object
from running <a href="#topic+RunAPOTC">RunAPOTC</a>, this function will read the information and return
a customizable ggplot2 object of the clonal expansion with a circle size
legend. If the user is unhappy about certain aspects of the plot, many
parameters can be adjusted with the <a href="#topic+AdjustAPOTC">AdjustAPOTC</a> function.
</p>
<p>The specific APackOfTheClones run to be plotted can be identified in two
ways: either by inputting the <code>run_id</code> associated with the run that was
either defined / auto-generated during <a href="#topic+RunAPOTC">RunAPOTC</a>, or by inputting the
<code>reduction_base</code>, <code>clonecall</code>, <code>extra_filter</code> and any other keyword arguments
that corresponded to the run. Its heavily recommended to use the <code>run_id</code>.
If none of these parameters are inputted, the function defaults to returning
the plot of the latest run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>APOTCPlot(
  seurat_obj,
  reduction_base = NULL,
  clonecall = NULL,
  ...,
  extra_filter = NULL,
  run_id = NULL,
  res = 360L,
  linetype = "blank",
  use_default_theme = TRUE,
  retain_axis_scales = FALSE,
  show_labels = FALSE,
  label_size = 5,
  add_size_legend = TRUE,
  legend_sizes = "auto",
  legend_position = "auto",
  legend_buffer = 0.2,
  legend_color = "#808080",
  legend_spacing = "auto",
  legend_label = "Clone sizes",
  legend_text_size = 5,
  add_legend_background = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="APOTCPlot_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>A seurat object that has been integrated with clonotype
data and has had a valid run of <a href="#topic+RunAPOTC">RunAPOTC</a>.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_reduction_base">reduction_base</code></td>
<td>
<p>character. The seurat reduction to base the clonal
expansion plotting on. Defaults to <code>'umap'</code> but can be any reduction present
within the reductions slot of the input seurat object, including custom ones.
If ''pca'&ldquo;, the cluster coordinates will be based on PC1 and PC2.
However, generally APackOfTheClones is used for displaying UMAP and
occasionally t-SNE versions to intuitively highlight clonal expansion.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_clonecall">clonecall</code></td>
<td>
<p>character. The column name in the seurat object metadata to
use. See <code>scRepertoire</code> documentation for more information about this
parameter that is central to both packages.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_...">...</code></td>
<td>
<p>additional &quot;subsetting&quot; keyword arguments indicating the rows
corresponding to elements in the seurat object metadata that should be
filtered by. E.g., <code>seurat_clusters = c(1, 9, 10)</code> will filter the cells to
those in the <code>seurat_clusters</code> column with any of the values 1, 9, and 10.
Unfortunately, column names in the seurat object metadata cannot
conflict with the keyword arguments. <em><strong>MAJOR NOTE</strong></em> if any subsetting
keyword arguments are a <em>prefix</em> of any preceding argument names (e.g. a
column named <code>reduction</code> is a prefix of the <code>reduction_base</code> argument)
R will interpret it as the same argument unless <em>both</em> arguments
are named. Additionally, this means any subsequent arguments <em>must</em> be named.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_extra_filter">extra_filter</code></td>
<td>
<p>character. An additional string that should be formatted
<em>exactly</em> like a statement one would pass into <a href="dplyr.html#topic+filter">dplyr::filter</a> that does
<em>additional</em> filtering to cells in the seurat object - on top of the other
keyword arguments - based on the metadata. This means that it will be
logically AND'ed with any keyword argument filters. This is a more flexible
alternative / addition to the filtering keyword arguments. For example, if
one wanted to filter by the length of the amino acid sequence of TCRs, one
could pass in something like <code>extra_filter = "nchar(CTaa) - 1 &gt; 10"</code>. When
involving characters, ensure to enclose with single quotes.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_run_id">run_id</code></td>
<td>
<p>character. This will be the ID associated with the data of a
run, and will be used by other important functions like <a href="#topic+APOTCPlot">APOTCPlot</a> and
<a href="#topic+AdjustAPOTC">AdjustAPOTC</a>. Defaults to <code>NULL</code>, in which case the ID will be generated
in the following format:
</p>
<p><code style="white-space: pre;">&#8288;reduction_base;clonecall;keyword_arguments;extra_filter&#8288;</code>
</p>
<p>where if keyword arguments and extra_filter are underscore characters if
there was no input for the <code>...</code> and <code>extra_filter</code> parameters.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_res">res</code></td>
<td>
<p>The number of points on the generated path per full circle. From
plot viewers, if circles seem slightly too pixelated, it is recommended to
first try to export the plot as an <code>.svg</code> before increasing <code>res</code> due to
increased plotting times from <a href="ggforce.html#topic+geom_circle">ggforce::geom_circle</a>.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_linetype">linetype</code></td>
<td>
<p>The type of outline each circle should have. defaults to
<code style="white-space: pre;">&#8288;"blank&#8288;</code> meaning no outline. More information is in the function
documentation of <code>ggforce::geom_circle</code>.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_use_default_theme">use_default_theme</code></td>
<td>
<p>logical that defaults to <code>TRUE</code>. If <code>TRUE</code>,
the resulting plot will have the same theme as the seurat reference reduction
plot. Else, the plot will simply have a blank background.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_retain_axis_scales">retain_axis_scales</code></td>
<td>
<p>If <code>TRUE</code>, approximately maintains the axis scales
of the original reduction plot. However, it will only attempt to extend the
axes and never shorten. This is recommended to be set to <code>TRUE</code> especially if
working with subsetted versions of the clonal data.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_show_labels">show_labels</code></td>
<td>
<p>If <code>TRUE</code>, will label each circle cluster at the centroid,
defaulting to &quot;C0, C1, ...&quot;.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_label_size">label_size</code></td>
<td>
<p>The text size of labels if shown. Defaults to 5.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_add_size_legend">add_size_legend</code></td>
<td>
<p>If <code>TRUE</code>, adds a legend to the plot visualizing the
relative sizes of clones. Note that it is simply an overlay and not a real
ggplot2 legend.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_legend_sizes">legend_sizes</code></td>
<td>
<p>numeric vector. Indicates the circle sizes to be
displayed on the legend, and will always be sorted from smallest to greatest.
Defaults to <code>"auto"</code> which estimate a reasonable range of sizes to display.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_legend_position">legend_position</code></td>
<td>
<p>character or numeric. Can be set to either
<code>"top_left"</code>, <code>"top_right"</code>, <code>"bottom_left"</code>, <code>"bottom_right"</code> and places the
legend roughly in the corresponding position. Otherwise, can be a numeric
vector of length 2 indicating the x and y position of the <em>topmost (smallest)
circle</em> of the legend.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_legend_buffer">legend_buffer</code></td>
<td>
<p>numeric. Indicates how much to &quot;push&quot; the legend towards
the center of the plot from the selected corner. If negative, will push away</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_legend_color">legend_color</code></td>
<td>
<p>character. Indicates the hex color of the circles
displayed on the legend. Defaults to the hex code for a gray tone</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_legend_spacing">legend_spacing</code></td>
<td>
<p>numeric. Indicates the horizontal distance between each
stacked circle on the size legend. Defaults to <code>"auto"</code> which will use an
estimated value depending on plot size</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_legend_label">legend_label</code></td>
<td>
<p>character. The title of the legend, which defaults to
<code style="white-space: pre;">&#8288;"clone sizes&#8288;</code>.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>numeric. The text size of the letters and numbers on
the legend</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_add_legend_background">add_legend_background</code></td>
<td>
<p>logical. If <code>TRUE</code>, will add a border around the
legend and fill the background to be white, overlaying anything else.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object of the APackOfTheClones clonal expansion plot of the
seurat object
</p>


<h3>See Also</h3>

<p><a href="#topic+AdjustAPOTC">AdjustAPOTC</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("combined_pbmc")

combined_pbmc &lt;- RunAPOTC(
    combined_pbmc, run_id = "run1", verbose = FALSE
)

# plotting with default arguments will plot the latest "run1"
clonal_packing_plot &lt;- APOTCPlot(combined_pbmc)

</code></pre>

<hr>
<h2 id='clonal_expansion_plot'>DEFUNCT: Visualize T cell clonal expansion with a ball-packing plot.</h2><span id='topic+clonal_expansion_plot'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#defunct"><img src="../help/figures/lifecycle-defunct.svg" alt='[Defunct]' /></a>
</p>
<p><em><strong>ALL v0.1.x functions are deprecated</strong></em>, and the workflow has been completely revamped - now depending on the scRepertoire v2 package - which allows for the processing of multi-sampled single cell data. Please read the vignettes with <code>browseVignettes("APackOfTheClones")</code>or visit https://qile0317.github.io/APackOfTheClones/
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clonal_expansion_plot(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clonal_expansion_plot_+3A_...">...</code></td>
<td>
<p>arbitrary arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>error message
</p>

<hr>
<h2 id='combined_pbmc'>Example Multi-sampled T-cell seurat object with integrated TCR library</h2><span id='topic+combined_pbmc'></span>

<h3>Description</h3>

<p>r lifecycle::badge(&quot;experimental&quot;)'
</p>
<p>Generated with <code>scRepertoire::combineExpression</code>, more specifically, with:
</p>
<p>scRepertoire::combineExpression(scRepertoire::combineTCR(get(data(&quot;contig_list&quot;, ,     package = &quot;scRepertoire&quot;)), samples = c(&quot;P17B&quot;, &quot;P17L&quot;, &quot;P18B&quot;, ,     &quot;P18L&quot;, &quot;P19B&quot;, &quot;P19L&quot;, &quot;P20B&quot;, &quot;P20L&quot;), removeNA = FALSE, ,     removeMulti = FALSE, filterMulti = FALSE), get(data(&quot;scRep_example&quot;, ,     package = &quot;scRepertoire&quot;)), cloneCall = &quot;gene&quot;, proportion = TRUE)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("combined_pbmc")
</code></pre>


<h3>Format</h3>

<p>A Seurat object with the following slots filled
</p>

<dl>
<dt>assays</dt><dd>
<p>Currently only contains one assay (&quot;RNA&quot; - scRNA-seq expression data)
</p>
<ul>
<li><p>counts - Raw expression data
</p>
</li>
<li><p>data - Normalized expression data
</p>
</li>
<li><p>scale.data - Scaled expression data
</p>
</li>
<li><p>var.features - names of the current features selected as variable
</p>
</li>
<li><p>meta.features - Assay level metadata such as mean and variance
</p>
</li></ul>
</dd>
<dt>meta.data</dt><dd><p>Cell level metadata with a combined TCR contig list <code style="white-space: pre;">&#8288;from scRepertoire&#8288;</code></p>
</dd>
<dt>active.assay</dt><dd><p>Current default assay</p>
</dd>
<dt>active.ident</dt><dd><p>Current default idents</p>
</dd>
<dt>graphs</dt><dd><p>Neighbor graphs computed, currently stores the SNN</p>
</dd>
<dt>reductions</dt><dd><p>Dimensional reductions: UMAP</p>
</dd>
<dt>version</dt><dd><p>Seurat version used to create the object</p>
</dd>
<dt>commands</dt><dd><p>Command history, including the one used to create this object &quot;combineExpression&quot;</p>
</dd>
</dl>


<hr>
<h2 id='containsApotcRun'>Check for the existence of an APackOfTheClones run with its run id</h2><span id='topic+containsApotcRun'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>A convenience function to check for the existence of an APackOfTheClones
run with its run id, regardless of if any run has been made
</p>


<h3>Usage</h3>

<pre><code class='language-R'>containsApotcRun(seurat_obj, run_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="containsApotcRun_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>a seurat object</p>
</td></tr>
<tr><td><code id="containsApotcRun_+3A_run_id">run_id</code></td>
<td>
<p>character. The id of the associated ApotcRun.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical indicating whether the run exists.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pbmc &lt;- RunAPOTC(
    seurat_obj = get(data("combined_pbmc")),
    reduction_base = "umap",
    clonecall = "strict",
    run_id = "run1",
    verbose = FALSE
)

containsApotcRun(pbmc, "run1")
#&gt; [1] TRUE

containsApotcRun(pbmc, "run2")
#&gt; [1] FALSE

</code></pre>

<hr>
<h2 id='count_clone_sizes'>DEPRECATED: count the number of clonotype sizes per cell cluster in a seurat
object integrated with a TCR library</h2><span id='topic+count_clone_sizes'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
</p>
<p><em><strong>ALL v0.1.x functions are deprecated</strong></em>, and the workflow has been completely revamped - now depending on the scRepertoire v2 package - which allows for the processing of multi-sampled single cell data. Please read the vignettes with <code>browseVignettes("APackOfTheClones")</code>or visit https://qile0317.github.io/APackOfTheClones/
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count_clone_sizes(integrated_seurat_obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="count_clone_sizes_+3A_integrated_seurat_obj">integrated_seurat_obj</code></td>
<td>
<p>Seurat object that has been integrated with a
T-cell receptor library with the deprecated function
<code><a href="#topic+integrate_tcr">integrate_tcr</a></code>. More specifically, in the metadata, there must
at least be the elements <code>seurat_clusters</code> and <code>raw_clonotype_id</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is no longer needed as the <code>scRepertoire</code> workflow takes care
of this step.
</p>


<h3>Value</h3>

<p>A list of <code>table</code> objects, where each element is tabled
clonotype frequencies for the seurat cluster corresponding to the same index
</p>

<ul>
<li> 

<ol>
<li><p> For example, the 5th element is a tabled frequency of counts that
corresponds to the 4th seurat cluster (as seurat clusters are 0-indexed).
If an element is <code>NULL</code>, it indicates that there were no corresponding T-cell
receptor barcode for the cells in the cluster.
</p>
</li></ol>

</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
library(APackOfTheClones)
data("mini_clonotype_data","mini_seurat_obj")

# produce an integrated seurat_object
integrated_seurat_object &lt;- integrate_tcr(
    mini_seurat_obj, mini_clonotype_data, verbose = FALSE
)

clonotype_counts &lt;- count_clone_sizes(integrated_seurat_object)
clonotype_counts

</code></pre>

<hr>
<h2 id='countCloneSizes'>count the number of clonotype sizes per cell cluster in a seurat
object combined with a VDJ library</h2><span id='topic+countCloneSizes'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>Get clonotype frequencies from a seurat object's meta.data slot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>countCloneSizes(seurat_obj, clonecall = "strict", extra_filter = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="countCloneSizes_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>a seurat object combined with a VDJ library with the
<code>scRepertoire</code>.</p>
</td></tr>
<tr><td><code id="countCloneSizes_+3A_clonecall">clonecall</code></td>
<td>
<p>character. The column name in the seurat object metadata to
use. See <code>scRepertoire</code> documentation for more information about this
parameter that is central to both packages.</p>
</td></tr>
<tr><td><code id="countCloneSizes_+3A_extra_filter">extra_filter</code></td>
<td>
<p>character. An additional string that should be formatted
<em>exactly</em> like a statement one would pass into <a href="dplyr.html#topic+filter">dplyr::filter</a> that does
<em>additional</em> filtering to cells in the seurat object - on top of the other
keyword arguments - based on the metadata. This means that it will be
logically AND'ed with any keyword argument filters. This is a more flexible
alternative / addition to the filtering keyword arguments. For example, if
one wanted to filter by the length of the amino acid sequence of TCRs, one
could pass in something like <code>extra_filter = "nchar(CTaa) - 1 &gt; 10"</code>. When
involving characters, ensure to enclose with single quotes.</p>
</td></tr>
<tr><td><code id="countCloneSizes_+3A_...">...</code></td>
<td>
<p>additional &quot;subsetting&quot; keyword arguments indicating the rows
corresponding to elements in the seurat object metadata that should be
filtered by. E.g., <code>seurat_clusters = c(1, 9, 10)</code> will filter the cells to
those in the <code>seurat_clusters</code> column with any of the values 1, 9, and 10.
Unfortunately, column names in the seurat object metadata cannot
conflict with the keyword arguments. <em><strong>MAJOR NOTE</strong></em> if any subsetting
keyword arguments are a <em>prefix</em> of any preceding argument names (e.g. a
column named <code>reduction</code> is a prefix of the <code>reduction_base</code> argument)
R will interpret it as the same argument unless <em>both</em> arguments
are named. Additionally, this means any subsequent arguments <em>must</em> be named.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of table objects, with the table at each index corresponding
to each cluster index. Each table's names are the clonotype name indicated
by <code>clonecall</code> after filtering, while the values are the actual clone sizes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("combined_pbmc")

countCloneSizes(combined_pbmc)
countCloneSizes(combined_pbmc, "aa")
countCloneSizes(combined_pbmc, "nt", orig.ident = c("P17B", "P17L"))

</code></pre>

<hr>
<h2 id='deleteApotcData'>Delete the results of an APackOfTheClones run</h2><span id='topic+deleteApotcData'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>A convenience function to erase all data associated with a particular run,
including the ApotcData and command in seurat_obj@command. The run_id would
be no longer accessible afterwards.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deleteApotcData(seurat_obj, run_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="deleteApotcData_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>a seurat object that has had RunAPOTC ran on it before in
order of the functions being called.</p>
</td></tr>
<tr><td><code id="deleteApotcData_+3A_run_id">run_id</code></td>
<td>
<p>character. The id of the associated ApotcRun.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The modified input seurat object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pbmc &lt;- RunAPOTC(
    seurat_obj = get(data("combined_pbmc")),
    reduction_base = "umap",
    clonecall = "strict",
    run_id = "run1",
    verbose = FALSE
)

getApotcDataIds(pbmc)
#&gt; [1] "run1"

# delete the data
pbmc &lt;- deleteApotcData(pbmc, "run1")

getApotcDataIds(pbmc)
#&gt; NULL

</code></pre>

<hr>
<h2 id='getApotcDataIds'>Get all run ids of previous RunAPOTC runs on a seurat object</h2><span id='topic+getApotcDataIds'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>A convenience function to get all run ids of previous RunAPOTC run IDs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getApotcDataIds(seurat_obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getApotcDataIds_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>a seurat object that has had RunAPOTC ran on it before in
order of the functions being called.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of all run ids of previous RunAPOTC runs, in
the order they were ran in. If there are no runs on the object, it returns
<code>NULL</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pbmc &lt;- RunAPOTC(
    seurat_obj = get(data("combined_pbmc")),
    reduction_base = "umap",
    clonecall = "strict",
    verbose = FALSE
)

getApotcDataIds(pbmc)
#&gt; [1] "umap;CTstrict;_;_"

pbmc &lt;- RunAPOTC(
    seurat_obj = pbmc,
    reduction_base = "umap",
    clonecall = "gene",
    verbose = FALSE
)

getApotcDataIds(pbmc)
#&gt; [1] "umap;CTstrict;_;_" "umap;CTgene;_;_"

</code></pre>

<hr>
<h2 id='getLastApotcDataId'>Get the object id of the most recent RunAPOTC run on a seurat object</h2><span id='topic+getLastApotcDataId'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>A convenience function to get the object id of the most recent valid
<a href="#topic+RunAPOTC">RunAPOTC</a> run, to be used by <a href="#topic+APOTCPlot">APOTCPlot</a> and <a href="#topic+AdjustAPOTC">AdjustAPOTC</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getLastApotcDataId(seurat_obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getLastApotcDataId_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>a seurat object that has had RunAPOTC ran on it before in
order of the functions being called.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character of the object id of the last <a href="#topic+RunAPOTC">RunAPOTC</a> call
</p>


<h3>Examples</h3>

<pre><code class='language-R'># first run
pbmc &lt;- RunAPOTC(
    seurat_obj = get(data("combined_pbmc")),
    reduction_base = "umap",
    clonecall = "strict",
    verbose = FALSE
)

getLastApotcDataId(pbmc)
#&gt; [1] "umap;CTstrict;_;_"

# second run with a different clonecall
pbmc &lt;- RunAPOTC(
    seurat_obj = pbmc,
    reduction_base = "umap",
    clonecall = "gene",
    verbose = FALSE
)

getLastApotcDataId(pbmc)
#&gt; [1] "umap;CTgene;_;_"

</code></pre>

<hr>
<h2 id='getReductionCentroids'>Calculate seurat cluster centroids based on a Dimensional reduction</h2><span id='topic+getReductionCentroids'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>Utility function to calculate the physical xy coordinates of each seurat
cluster based on a dimensional reduction already present in the object.
The results are returned in a list with the length of the number of distinct
seurat clusters based on the seurat_obj <code>meta.data</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getReductionCentroids(seurat_obj, reduction)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getReductionCentroids_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>input seurat object with the dimensional reduction of
choice already present, and seurat clusters computed.</p>
</td></tr>
<tr><td><code id="getReductionCentroids_+3A_reduction">reduction</code></td>
<td>
<p>character. The reduction that the centroid calculation
should be based on.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of the length of the number of distinct clusters in the
seurat object metadata, where each element of the list is a numeric vector
of length 2, with the numbers corresponding to the x and y coordinate
respectively of the seurat cluster with the corresponding index.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("combined_pbmc")
getReductionCentroids(combined_pbmc, reduction = "umap")

</code></pre>

<hr>
<h2 id='integrate_tcr'>DEPRECATED: Integrate a single TCR library into Seurat object metadata</h2><span id='topic+integrate_tcr'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
</p>
<p><em><strong>ALL v0.1.x functions are deprecated</strong></em>, and the workflow has been completely revamped - now depending on the scRepertoire v2 package - which allows for the processing of multi-sampled single cell data. Please read the vignettes with <code>browseVignettes("APackOfTheClones")</code>or visit https://qile0317.github.io/APackOfTheClones/
</p>
<p>Modifies a <code>Seurat</code> object's metadata by taking all the columns
of the <code>all_contig_annotations.csv</code>, and adding new elements to
<code>seurat_obj@meta.data</code>, corresponding to each cells' barcodes and handling
duplicates (since the sequencing of TRA and TRB genes creates multiple data
points for the same cell).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>integrate_tcr(seurat_obj, tcr_file, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="integrate_tcr_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>Seurat object</p>
</td></tr>
<tr><td><code id="integrate_tcr_+3A_tcr_file">tcr_file</code></td>
<td>
<p><code>data.frame</code> of the T cell library generated by Cell Ranger.
It is very important that the row with cell barcodes is strictly named
<code>"barcode"</code>, which is the default name of barcodes in 10X's
<code>all_contig_annotations.csv</code> file.</p>
</td></tr>
<tr><td><code id="integrate_tcr_+3A_verbose">verbose</code></td>
<td>
<p>if <code>TRUE</code>, will display a progress bar to the R console.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Columns from cells (barcodes) that had duplicates in another row are
concatenated into strings, separated by <code style="white-space: pre;">&#8288;__&#8288;</code> in the metadata element.
Barcodes from the TCR library that had no matches to barcodes in the
<code>seurat_obj</code> will add <code>NA</code>s for all elements of the same index.
</p>


<h3>Value</h3>

<p>Returns a new Seurat object with new elements in the metadata
</p>


<h3>References</h3>

<p>atakanekiz (2019) Tutorial:Integrating VDJ sequencing data with Seurat.
<code style="white-space: pre;">&#8288;https://www.biostars.org/p/384640/&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
library(APackOfTheClones)
data("mini_clonotype_data","mini_seurat_obj")

# integrate the TCR data into new seurat object
integrated_seurat_object &lt;- integrate_tcr(
    mini_seurat_obj, mini_clonotype_data, verbose = FALSE
)

integrated_seurat_object

</code></pre>

<hr>
<h2 id='mini_clonotype_data'>Artificially generated T cell receptor library</h2><span id='topic+mini_clonotype_data'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
</p>
<p>A generated dataframe of a T-cell receptor (TCR) library generated from
single cell immune profiling. It is a subset the full dataframe
which would usually have up to 18 columns containing different data,
because the intended purpose of this object is to test various functions
in 'APackOfTheClones'. The dataframe compliments <code>mini_seurat_obj</code> and
can be integrated into it with <code>integrate_tcr</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("mini_clonotype_data")
</code></pre>


<h3>Format</h3>

<p><code>data.frame</code>
A data frame with 80 rows and 2 columns:
</p>

<dl>
<dt>barcode</dt><dd><p>barcodes corresponding to each sequenced cell</p>
</dd>
<dt>raw_clonotype_id</dt><dd><p>clonotype information for each cell</p>
</dd>
</dl>



<h3>Details</h3>

<p>Note that the clonotypes in the <code>raw_clonotype_id</code> column
actually do not contain all of clonotype<code>1</code>...clonotype<code>n</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mini_seurat_obj">mini_seurat_obj()</a></code>
</p>

<hr>
<h2 id='mini_seurat_obj'>Artificially generated Seurat object</h2><span id='topic+mini_seurat_obj'></span>

<h3>Description</h3>

<p>A generated 'SeuratObject' of a small single-sample sc-RNAseq experiment.
Has a corresponding T-cell receptor library generated from
single cell immune profiling, named <code>"mini_clonotype_data"</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("mini_seurat_obj")
</code></pre>


<h3>Format</h3>

<p>A Seurat object with the following slots filled
</p>

<dl>
<dt>assays</dt><dd>
<p>Currently only contains one assay (&quot;RNA&quot; - scRNA-seq expression data)
</p>
<ul>
<li><p>counts - Raw expression data
</p>
</li>
<li><p>data - Normalized expression data
</p>
</li>
<li><p>scale.data - Scaled expression data
</p>
</li>
<li><p>var.features - names of the current features selected as variable
</p>
</li>
<li><p>meta.features - Assay level metadata such as mean and variance
</p>
</li></ul>
</dd>
<dt>meta.data</dt><dd><p>Cell level metadata</p>
</dd>
<dt>active.assay</dt><dd><p>Current default assay</p>
</dd>
<dt>active.ident</dt><dd><p>Current default idents</p>
</dd>
<dt>graphs</dt><dd><p>Neighbor graphs computed, currently stores the SNN</p>
</dd>
<dt>reductions</dt><dd><p>Dimensional reductions: PCA, UMAP, and tSNE</p>
</dd>
<dt>version</dt><dd><p>Seurat version used to create the object</p>
</dd>
<dt>commands</dt><dd><p>Command history</p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+mini_clonotype_data">mini_clonotype_data()</a></code>
</p>

<hr>
<h2 id='RunAPOTC'>Run the APackOfTheClones method on a combined Seurat object for
downstream visualization of clonal expansion</h2><span id='topic+RunAPOTC'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>Computes necessary information for an APackOfTheClones
clonal expansion plot (<a href="#topic+APOTCPlot">APOTCPlot</a>) and stores it in the seurat object.
Gets sizes of unique clones and utilizes a circle-packing algorithm to
pack circles representing individual clones in approximately the same
dimensional reduction (<code>reduction_base</code>) coordinates.
</p>
<p>The parameter <code>extra_filter</code> along with an unlimited number of additional
keyword arguments can be used to filter the cells by certain conditions
in the metadata, and new results will be stored in addition to other runs
the users may have done.
</p>
<p>Each APackOfTheClones run is uniquely identified by the parameters
<code>reduction_base</code>, <code>clonecall</code>, <code>extra_filter</code>, and any additional keywords
passed to filter the metadata. Each distinct run result is stored in the
seurat object and has an associated Id generated from the aforementioned
parameters. To view the id of the latest run, call <a href="#topic+getLastApotcDataId">getLastApotcDataId</a>.
To view all the ids of previous runs, call <a href="#topic+getApotcDataIds">getApotcDataIds</a>. To work further
with a specific run (most importantly, plotting), the user can use this id
in the arguments with is slightly more convenient than passing in the
original RunAPOTC parameters again but both ways work.
</p>
<p>If the user wishes to manually customize/fix the expansion plot
generated, the circular packing information can be modified
with the <a href="#topic+AdjustAPOTC">AdjustAPOTC</a> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RunAPOTC(
  seurat_obj,
  reduction_base = "umap",
  clonecall = "strict",
  ...,
  extra_filter = NULL,
  run_id = NULL,
  clone_scale_factor = "auto",
  rad_scale_factor = 0.95,
  order_clones = TRUE,
  try_place = FALSE,
  repulse = TRUE,
  repulsion_threshold = 1,
  repulsion_strength = 1,
  max_repulsion_iter = 20L,
  override = FALSE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RunAPOTC_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>Seurat object with one or more dimension reductions and
already have been integrated with a TCR/BCR library with
<code>scRepertoire::combineExpression</code>.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_reduction_base">reduction_base</code></td>
<td>
<p>character. The seurat reduction to base the clonal
expansion plotting on. Defaults to <code>'umap'</code> but can be any reduction present
within the reductions slot of the input seurat object, including custom ones.
If ''pca'&ldquo;, the cluster coordinates will be based on PC1 and PC2.
However, generally APackOfTheClones is used for displaying UMAP and
occasionally t-SNE versions to intuitively highlight clonal expansion.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_clonecall">clonecall</code></td>
<td>
<p>character. The column name in the seurat object metadata to
use. See <code>scRepertoire</code> documentation for more information about this
parameter that is central to both packages.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_...">...</code></td>
<td>
<p>additional &quot;subsetting&quot; keyword arguments indicating the rows
corresponding to elements in the seurat object metadata that should be
filtered by. E.g., <code>seurat_clusters = c(1, 9, 10)</code> will filter the cells to
those in the <code>seurat_clusters</code> column with any of the values 1, 9, and 10.
Unfortunately, column names in the seurat object metadata cannot
conflict with the keyword arguments. <em><strong>MAJOR NOTE</strong></em> if any subsetting
keyword arguments are a <em>prefix</em> of any preceding argument names (e.g. a
column named <code>reduction</code> is a prefix of the <code>reduction_base</code> argument)
R will interpret it as the same argument unless <em>both</em> arguments
are named. Additionally, this means any subsequent arguments <em>must</em> be named.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_extra_filter">extra_filter</code></td>
<td>
<p>character. An additional string that should be formatted
<em>exactly</em> like a statement one would pass into <a href="dplyr.html#topic+filter">dplyr::filter</a> that does
<em>additional</em> filtering to cells in the seurat object - on top of the other
keyword arguments - based on the metadata. This means that it will be
logically AND'ed with any keyword argument filters. This is a more flexible
alternative / addition to the filtering keyword arguments. For example, if
one wanted to filter by the length of the amino acid sequence of TCRs, one
could pass in something like <code>extra_filter = "nchar(CTaa) - 1 &gt; 10"</code>. When
involving characters, ensure to enclose with single quotes.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_run_id">run_id</code></td>
<td>
<p>character. This will be the ID associated with the data of a
run, and will be used by other important functions like <a href="#topic+APOTCPlot">APOTCPlot</a> and
<a href="#topic+AdjustAPOTC">AdjustAPOTC</a>. Defaults to <code>NULL</code>, in which case the ID will be generated
in the following format:
</p>
<p><code style="white-space: pre;">&#8288;reduction_base;clonecall;keyword_arguments;extra_filter&#8288;</code>
</p>
<p>where if keyword arguments and extra_filter are underscore characters if
there was no input for the <code>...</code> and <code>extra_filter</code> parameters.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_clone_scale_factor">clone_scale_factor</code></td>
<td>
<p>Dictates how much to scale each circle(between 0,1)
radius when converting from clonotype counts into circles that represent
individual clonotypes. The argument defaults to the character <code>"auto"</code>, and
if so, the most visually pleasing factor will be estimated.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_rad_scale_factor">rad_scale_factor</code></td>
<td>
<p>numeric between 0 and 1. This value decreases the
radius of the smallest clones by this scale factor. And the absolute value
of this decrease will be applied to all packed circles, effectively shrinking
all circles on the spot, and introduce more constant spacing in between.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_order_clones">order_clones</code></td>
<td>
<p>logical. Decides if the largest clone circles should be
near cluster centroids. This is highly recommended to be set to TRUE for
increased intuitiveness of the visualization, as resulting plots tend to
give an improved impression of the proportion of expanded clones. If
<code style="white-space: pre;">&#8288;FALSE,&#8288;</code> will randomly scramble the positions of each circle. For the sake
of being replicable, a random seed is recommended to be set with <a href="base.html#topic+set.seed">set.seed</a>.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_try_place">try_place</code></td>
<td>
<p>If <code>TRUE</code>, always minimizes distance from a newly placed
circle to the origin in the circle packing algorithm.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_repulse">repulse</code></td>
<td>
<p>If <code>TRUE</code>, will attempt to push overlapping clusters away from
each other.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_repulsion_threshold">repulsion_threshold</code></td>
<td>
<p>numeric. The radius that clonal circle clusters
overlap is acceptable when repulsing.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_repulsion_strength">repulsion_strength</code></td>
<td>
<p>numeric. The smaller the value the less the
clusters repulse each other per iteration, and vice versa.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_max_repulsion_iter">max_repulsion_iter</code></td>
<td>
<p>integer. The number of repulsion iterations.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_override">override</code></td>
<td>
<p>logical. If <code>TRUE</code>, will override any existing
APackOfTheClones run data with the same <code>run_id</code>.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_verbose">verbose</code></td>
<td>
<p>logical. Decides if visual cues are displayed to the R console
of the progress.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All APackOfTheClones run data is stored in the Seurat object under
<code>seurat_object@misc$APackOfTheClones</code>, which is a list of S4 objects of the
type &quot;ApotcData&quot;, with each element corresponding to a unique run. The id of
each run is the name of each element in the list. The user
<em><strong>really shouldn't</strong></em> manually modify anything in the list as it may cause
unexpected behavior with many other functions.
</p>
<p>Additionally, it also logs a seurat command associated with the run in the
<code>seurat_object@commands</code> slot as a &quot;SeuratCommand&quot; object (from Seurat),
where the name of the object in the list is formatted as <code>RunAPOTC.run_id</code>.
Again, the user should not modify anything in these objects as they are used
by some related functions, mainly <a href="#topic+AdjustAPOTC">AdjustAPOTC</a>.
</p>


<h3>Value</h3>

<p>A modified version of the input seurat object, which harbors data
necessary for visualizing the clonal expansion of the cells with <a href="#topic+APOTCPlot">APOTCPlot</a>
and has a friendly user interface to modify certain attributes with
<a href="#topic+AdjustAPOTC">AdjustAPOTC</a>.
</p>


<h3>See Also</h3>

<p><a href="#topic+APOTCPlot">APOTCPlot</a>, <a href="#topic+AdjustAPOTC">AdjustAPOTC</a>, <a href="#topic+getApotcDataIds">getApotcDataIds</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("combined_pbmc")

# this is the recommended approach to use a custom run_id with default params
combined_pbmc &lt;- RunAPOTC(combined_pbmc, run_id = "default", verbose = FALSE)

# here's a seperate run with some filters to the meta data, where
# `orig.ident` is a custom column in the example data. Notice that it is not
# a `RunAPOTC` parameter but a user keyword argument
combined_pbmc &lt;- RunAPOTC(
    combined_pbmc, run_id = "sample17", orig.ident = c("P17B", "P17L"),
    verbose = FALSE
)

# the exact same thing can be achieved with the `extra_filter` parameter
combined_pbmc &lt;- RunAPOTC(
    combined_pbmc,
    run_id = "sample17",
    extra_filter = "substr(orig.ident, 2, 3) == '17'",
    override = TRUE,
    verbose = FALSE
)

</code></pre>

<hr>
<h2 id='vizAPOTC'>Directly visualize clonal expansion of a combined seurat object</h2><span id='topic+vizAPOTC'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>This function combines the functionality of both <code>RunAPOTC</code> and <code>APOTCPlot</code>.
Given a Seurat object, it first runs the APackOfTheClones method (<a href="#topic+RunAPOTC">RunAPOTC</a>)
to compute clonal expansion information, and then generates a customizable
ggplot2 object of the clonal expansion plot with a circle size legend
(<a href="#topic+APOTCPlot">APOTCPlot</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vizAPOTC(
  seurat_obj,
  reduction_base = "umap",
  clonecall = "strict",
  ...,
  extra_filter = NULL,
  clone_scale_factor = "auto",
  rad_scale_factor = 0.95,
  order_clones = TRUE,
  try_place = FALSE,
  repulse = TRUE,
  repulsion_threshold = 1,
  repulsion_strength = 1,
  max_repulsion_iter = 20L,
  res = 360L,
  linetype = "blank",
  use_default_theme = TRUE,
  retain_axis_scales = FALSE,
  show_labels = FALSE,
  label_size = 5,
  add_size_legend = TRUE,
  legend_sizes = "auto",
  legend_position = "auto",
  legend_buffer = 0.2,
  legend_color = "#808080",
  legend_spacing = "auto",
  legend_label = "Clone sizes",
  legend_text_size = 5,
  add_legend_background = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vizAPOTC_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>Seurat object with one or more dimension reductions and
already have been integrated with a TCR/BCR library with
<code>scRepertoire::combineExpression</code>.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_reduction_base">reduction_base</code></td>
<td>
<p>character. The seurat reduction to base the clonal
expansion plotting on. Defaults to <code>'umap'</code> but can be any reduction present
within the reductions slot of the input seurat object, including custom ones.
If ''pca'&ldquo;, the cluster coordinates will be based on PC1 and PC2.
However, generally APackOfTheClones is used for displaying UMAP and
occasionally t-SNE versions to intuitively highlight clonal expansion.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_clonecall">clonecall</code></td>
<td>
<p>character. The column name in the seurat object metadata to
use. See <code>scRepertoire</code> documentation for more information about this
parameter that is central to both packages.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_...">...</code></td>
<td>
<p>additional &quot;subsetting&quot; keyword arguments indicating the rows
corresponding to elements in the seurat object metadata that should be
filtered by. E.g., <code>seurat_clusters = c(1, 9, 10)</code> will filter the cells to
those in the <code>seurat_clusters</code> column with any of the values 1, 9, and 10.
Unfortunately, column names in the seurat object metadata cannot
conflict with the keyword arguments. <em><strong>MAJOR NOTE</strong></em> if any subsetting
keyword arguments are a <em>prefix</em> of any preceding argument names (e.g. a
column named <code>reduction</code> is a prefix of the <code>reduction_base</code> argument)
R will interpret it as the same argument unless <em>both</em> arguments
are named. Additionally, this means any subsequent arguments <em>must</em> be named.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_extra_filter">extra_filter</code></td>
<td>
<p>character. An additional string that should be formatted
<em>exactly</em> like a statement one would pass into <a href="dplyr.html#topic+filter">dplyr::filter</a> that does
<em>additional</em> filtering to cells in the seurat object - on top of the other
keyword arguments - based on the metadata. This means that it will be
logically AND'ed with any keyword argument filters. This is a more flexible
alternative / addition to the filtering keyword arguments. For example, if
one wanted to filter by the length of the amino acid sequence of TCRs, one
could pass in something like <code>extra_filter = "nchar(CTaa) - 1 &gt; 10"</code>. When
involving characters, ensure to enclose with single quotes.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_clone_scale_factor">clone_scale_factor</code></td>
<td>
<p>Dictates how much to scale each circle(between 0,1)
radius when converting from clonotype counts into circles that represent
individual clonotypes. The argument defaults to the character <code>"auto"</code>, and
if so, the most visually pleasing factor will be estimated.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_rad_scale_factor">rad_scale_factor</code></td>
<td>
<p>numeric between 0 and 1. This value decreases the
radius of the smallest clones by this scale factor. And the absolute value
of this decrease will be applied to all packed circles, effectively shrinking
all circles on the spot, and introduce more constant spacing in between.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_order_clones">order_clones</code></td>
<td>
<p>logical. Decides if the largest clone circles should be
near cluster centroids. This is highly recommended to be set to TRUE for
increased intuitiveness of the visualization, as resulting plots tend to
give an improved impression of the proportion of expanded clones. If
<code style="white-space: pre;">&#8288;FALSE,&#8288;</code> will randomly scramble the positions of each circle. For the sake
of being replicable, a random seed is recommended to be set with <a href="base.html#topic+set.seed">set.seed</a>.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_try_place">try_place</code></td>
<td>
<p>If <code>TRUE</code>, always minimizes distance from a newly placed
circle to the origin in the circle packing algorithm.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_repulse">repulse</code></td>
<td>
<p>If <code>TRUE</code>, will attempt to push overlapping clusters away from
each other.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_repulsion_threshold">repulsion_threshold</code></td>
<td>
<p>numeric. The radius that clonal circle clusters
overlap is acceptable when repulsing.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_repulsion_strength">repulsion_strength</code></td>
<td>
<p>numeric. The smaller the value the less the
clusters repulse each other per iteration, and vice versa.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_max_repulsion_iter">max_repulsion_iter</code></td>
<td>
<p>integer. The number of repulsion iterations.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_res">res</code></td>
<td>
<p>The number of points on the generated path per full circle. From
plot viewers, if circles seem slightly too pixelated, it is recommended to
first try to export the plot as an <code>.svg</code> before increasing <code>res</code> due to
increased plotting times from <a href="ggforce.html#topic+geom_circle">ggforce::geom_circle</a>.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_linetype">linetype</code></td>
<td>
<p>The type of outline each circle should have. defaults to
<code style="white-space: pre;">&#8288;"blank&#8288;</code> meaning no outline. More information is in the function
documentation of <code>ggforce::geom_circle</code>.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_use_default_theme">use_default_theme</code></td>
<td>
<p>logical that defaults to <code>TRUE</code>. If <code>TRUE</code>,
the resulting plot will have the same theme as the seurat reference reduction
plot. Else, the plot will simply have a blank background.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_retain_axis_scales">retain_axis_scales</code></td>
<td>
<p>If <code>TRUE</code>, approximately maintains the axis scales
of the original reduction plot. However, it will only attempt to extend the
axes and never shorten. This is recommended to be set to <code>TRUE</code> especially if
working with subsetted versions of the clonal data.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_show_labels">show_labels</code></td>
<td>
<p>If <code>TRUE</code>, will label each circle cluster at the centroid,
defaulting to &quot;C0, C1, ...&quot;.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_label_size">label_size</code></td>
<td>
<p>The text size of labels if shown. Defaults to 5.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_add_size_legend">add_size_legend</code></td>
<td>
<p>If <code>TRUE</code>, adds a legend to the plot visualizing the
relative sizes of clones. Note that it is simply an overlay and not a real
ggplot2 legend.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_legend_sizes">legend_sizes</code></td>
<td>
<p>numeric vector. Indicates the circle sizes to be
displayed on the legend, and will always be sorted from smallest to greatest.
Defaults to <code>"auto"</code> which estimate a reasonable range of sizes to display.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_legend_position">legend_position</code></td>
<td>
<p>character or numeric. Can be set to either
<code>"top_left"</code>, <code>"top_right"</code>, <code>"bottom_left"</code>, <code>"bottom_right"</code> and places the
legend roughly in the corresponding position. Otherwise, can be a numeric
vector of length 2 indicating the x and y position of the <em>topmost (smallest)
circle</em> of the legend.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_legend_buffer">legend_buffer</code></td>
<td>
<p>numeric. Indicates how much to &quot;push&quot; the legend towards
the center of the plot from the selected corner. If negative, will push away</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_legend_color">legend_color</code></td>
<td>
<p>character. Indicates the hex color of the circles
displayed on the legend. Defaults to the hex code for a gray tone</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_legend_spacing">legend_spacing</code></td>
<td>
<p>numeric. Indicates the horizontal distance between each
stacked circle on the size legend. Defaults to <code>"auto"</code> which will use an
estimated value depending on plot size</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_legend_label">legend_label</code></td>
<td>
<p>character. The title of the legend, which defaults to
<code style="white-space: pre;">&#8288;"clone sizes&#8288;</code>.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>numeric. The text size of the letters and numbers on
the legend</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_add_legend_background">add_legend_background</code></td>
<td>
<p>logical. If <code>TRUE</code>, will add a border around the
legend and fill the background to be white, overlaying anything else.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_verbose">verbose</code></td>
<td>
<p>logical. Decides if visual cues are displayed to the R console
of the progress.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object of the APackOfTheClones clonal expansion plot of the
seurat object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("combined_pbmc")

# plot with default parameters
vizAPOTC(combined_pbmc, verbose = FALSE)

# use arguments from RunAPOTC and APOTCPlot
vizAPOTC(
    combined_pbmc, try_place = TRUE, show_labels = TRUE, verbose = FALSE
)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
