<!DOCTYPE html><html><head><title>Help for package APackOfTheClones</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {APackOfTheClones}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#APackOfTheClones-package'><p>APackOfTheClones: Visualization of Clonal Expansion for Single Cell Immune Profiles</p></a></li>
<li><a href='#AdjustAPOTC'><p>Adjust the parameters of the APackOfTheClones reduction in a seurat</p>
object</a></li>
<li><a href='#APOTCPlot'><p>Various variations of visualizations of clonal expansion post-RunAPOTC</p></a></li>
<li><a href='#clonal_expansion_plot'><p>DEFUNCT: Visualize T cell clonal expansion with a ball-packing plot.</p></a></li>
<li><a href='#combined_pbmc'><p>Example Multi-sampled T-cell seurat object with integrated TCR library</p></a></li>
<li><a href='#containsApotcRun'><p>Check for the existence of an APackOfTheClones run with its run id</p></a></li>
<li><a href='#count_clone_sizes'><p>DEFUNCT: count the number of clonotype sizes per cell cluster in a seurat</p>
object integrated with a TCR library</a></li>
<li><a href='#countCloneSizes'><p>count the number of clonotype sizes in a seurat object combined with a</p>
VDJ library overall or by cluster</a></li>
<li><a href='#deleteApotcData'><p>Delete the results of an APackOfTheClones run</p></a></li>
<li><a href='#getApotcDataIds'><p>Get all run ids of previous RunAPOTC runs on a seurat object</p></a></li>
<li><a href='#getLastApotcDataId'><p>Get the object id of the most recent RunAPOTC run on a seurat object</p></a></li>
<li><a href='#getReductionCentroids'><p>Calculate seurat cluster centroids based on a Dimensional reduction</p></a></li>
<li><a href='#getSharedClones'><p>Compute a list of clonotypes that are shared between seurat clusters</p></a></li>
<li><a href='#integrate_tcr'><p>DEFUNCT: Integrate a single TCR library into Seurat object metadata</p></a></li>
<li><a href='#mergeCloneSizes'><p>Merge a list of Clustered Clonotype Frequency Tables</p></a></li>
<li><a href='#overlayLegend'><p>overlay a clone size legend on an APackOfTheClones plot</p></a></li>
<li><a href='#removeLegend'><p>Remove current APackOfTheClones legend</p></a></li>
<li><a href='#renameApotcRun'><p>Rename an APackOfTheClones run</p></a></li>
<li><a href='#RunAPOTC'><p>Run the APackOfTheClones method on a combined Seurat object for</p>
downstream visualization of clonal expansion</a></li>
<li><a href='#showCloneHighlight'><p>Highlight specific clones on an APackOfTheClones ggplot</p></a></li>
<li><a href='#vizAPOTC'><p>Directly visualize clonal expansion of a combined seurat object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Visualization of Clonal Expansion for Single Cell Immune
Profiles</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Qile Yang &lt;qile.yang@berkeley.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Visualize clonal expansion via circle-packing. 'APackOfTheClones' extends 'scRepertoire' to produce a publication-ready visualization of clonal expansion at a single cell resolution, by representing expanded clones as differently sized circles. The method was originally implemented by Murray Christian and Ben Murrell in the following immunology study: Ma et al. (2021) &lt;<a href="https://doi.org/10.1126%2Fsciimmunol.abg6356">doi:10.1126/sciimmunol.abg6356</a>&gt;.</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Qile0317/APackOfTheClones/issues/">https://github.com/Qile0317/APackOfTheClones/issues/</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>LazyData:</td>
<td>false</td>
</tr>
<tr>
<td>Suggests:</td>
<td>cowplot, covr, knitr, rmarkdown, sf, spelling, testthat (&ge;
3.0.0), vdiffr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, ggforce, ggplot2, grDevices, hash, lifecycle, magrittr,
methods, Rcpp, rlang, Seurat, SeuratObject</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://qile0317.github.io/APackOfTheClones/">https://qile0317.github.io/APackOfTheClones/</a></td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-16 09:36:49 UTC; qile</td>
</tr>
<tr>
<td>Author:</td>
<td>Qile Yang [cre, aut, cph]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-16 09:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='APackOfTheClones-package'>APackOfTheClones: Visualization of Clonal Expansion for Single Cell Immune Profiles</h2><span id='topic+APackOfTheClones'></span><span id='topic+APackOfTheClones-package'></span>

<h3>Description</h3>

<p><img src="../help/figures/logo.png" style='float: right' alt='logo' width='120' />
</p>
<p>Visualize clonal expansion via circle-packing. 'APackOfTheClones' extends 'scRepertoire' to produce a publication-ready visualization of clonal expansion at a single cell resolution, by representing expanded clones as differently sized circles. The method was originally implemented by Murray Christian and Ben Murrell in the following immunology study: Ma et al. (2021) <a href="https://doi.org/10.1126/sciimmunol.abg6356">doi:10.1126/sciimmunol.abg6356</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Qile Yang <a href="mailto:qile.yang@berkeley.edu">qile.yang@berkeley.edu</a> [copyright holder]
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://qile0317.github.io/APackOfTheClones/">https://qile0317.github.io/APackOfTheClones/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/Qile0317/APackOfTheClones/issues/">https://github.com/Qile0317/APackOfTheClones/issues/</a>
</p>
</li></ul>


<hr>
<h2 id='AdjustAPOTC'>Adjust the parameters of the APackOfTheClones reduction in a seurat
object</h2><span id='topic+AdjustAPOTC'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>If the user is unsatisfied with the clonal expansion plot that
was generated from <code>RunAPOTC</code> and <code>APOTCPlot</code>, this function has a range of
arguments to modify the data and/or parameters of the visualization. Note
that some of the arguments may conflict with eachother.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AdjustAPOTC(
  seurat_obj,
  reduction_base = NULL,
  clonecall = NULL,
  ...,
  extra_filter = NULL,
  run_id = NULL,
  new_rad_scale_factor = NULL,
  new_clone_scale_factor = NULL,
  repulse = FALSE,
  repulsion_threshold = 1,
  repulsion_strength = 1,
  max_repulsion_iter = 10L,
  relocate_cluster = NULL,
  relocation_coord = NULL,
  nudge_cluster = NULL,
  nudge_vector = NULL,
  recolor_cluster = NULL,
  new_color = NULL,
  rename_label = NULL,
  new_label = NULL,
  relocate_label = NULL,
  label_relocation_coord = NULL,
  nudge_label = NULL,
  label_nudge_vector = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AdjustAPOTC_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>The seurat object to be adjusted.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_reduction_base">reduction_base</code></td>
<td>
<p>character. The seurat reduction to base the clonal
expansion plotting on. Defaults to <code>'umap'</code> but can be any reduction present
within the reductions slot of the input seurat object, including custom ones.
If ''pca'&ldquo;, the cluster coordinates will be based on PC1 and PC2.
However, generally APackOfTheClones is used for displaying UMAP and
occasionally t-SNE versions to intuitively highlight clonal expansion.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_clonecall">clonecall</code></td>
<td>
<p>character. The column name in the seurat object metadata to
use. See <code>scRepertoire</code> documentation for more information about this
parameter that is central to both packages.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_...">...</code></td>
<td>
<p>additional &quot;subsetting&quot; keyword arguments indicating the rows
corresponding to elements in the seurat object metadata that should be
filtered by. E.g., <code>seurat_clusters = c(1, 9, 10)</code> will filter the cells to
those in the <code>seurat_clusters</code> column with any of the values 1, 9, and 10.
Unfortunately, column names in the seurat object metadata cannot
conflict with the keyword arguments. <em><strong>MAJOR NOTE</strong></em> if any subsetting
keyword arguments are a <em>prefix</em> of any preceding argument names (e.g. a
column named <code>reduction</code> is a prefix of the <code>reduction_base</code> argument)
R will interpret it as the same argument unless <em>both</em> arguments
are named. Additionally, this means any subsequent arguments <em>must</em> be named.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_extra_filter">extra_filter</code></td>
<td>
<p>character. An additional string that should be formatted
<em>exactly</em> like a statement one would pass into <a href="dplyr.html#topic+filter">dplyr::filter</a> that does
<em>additional</em> filtering to cells in the seurat object - on top of the other
keyword arguments - based on the metadata. This means that it will be
logically AND'ed with any keyword argument filters. This is a more flexible
alternative / addition to the filtering keyword arguments. For example, if
one wanted to filter by the length of the amino acid sequence of TCRs, one
could pass in something like <code>extra_filter = "nchar(CTaa) - 1 &gt; 10"</code>. When
involving characters, ensure to enclose with single quotes.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_run_id">run_id</code></td>
<td>
<p>character. This will be the ID associated with the data of a
run, and will be used by other important functions like <a href="#topic+APOTCPlot">APOTCPlot</a> and
<a href="#topic+AdjustAPOTC">AdjustAPOTC</a>. Defaults to <code>NULL</code>, in which case the ID will be generated
in the following format:
</p>
<p><code style="white-space: pre;">&#8288;reduction_base;clonecall;keyword_arguments;extra_filter&#8288;</code>
</p>
<p>where if keyword arguments and extra_filter are underscore characters if
there was no input for the <code>...</code> and <code>extra_filter</code> parameters.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_new_rad_scale_factor">new_rad_scale_factor</code></td>
<td>
<p>a single numeric in (0, 1]. changes the
radius scaling factor of all circles.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_new_clone_scale_factor">new_clone_scale_factor</code></td>
<td>
<p>a single numeric in (0, 1]. changes the
clone_scale_factor</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_repulse">repulse</code></td>
<td>
<p>If <code>TRUE</code>, will attempt to push overlapping clusters away from
each other.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_repulsion_threshold">repulsion_threshold</code></td>
<td>
<p>numeric. The radius that clonal circle clusters
overlap is acceptable when repulsing.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_repulsion_strength">repulsion_strength</code></td>
<td>
<p>numeric. The smaller the value the less the
clusters repulse each other per iteration, and vice versa.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_max_repulsion_iter">max_repulsion_iter</code></td>
<td>
<p>integer. The number of repulsion iterations.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_relocate_cluster">relocate_cluster</code></td>
<td>
<p>Numeric or Character. Indicates which
cluster(s) based on the index or label to relocate to new coordinates.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_relocation_coord">relocation_coord</code></td>
<td>
<p>numeric of length two or a list of numerics of length
two of length of <code>relocate_cluster</code>. If its a list, indicates each coordinate
that the clusters in <code>relocate_cluster</code> should move to. If its just a
numeric, then will relocate all clusters in <code>relocate_cluster</code> to the input,
which is likely not desired behavior, so this should only be convenience
syntax if <code>relocate_cluster</code> has length 1.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_nudge_cluster">nudge_cluster</code></td>
<td>
<p>Numeric or Character. Indicates which
cluster(s) based on the index or label to &quot;nudge&quot;/translate their
coordinate(s) by.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_nudge_vector">nudge_vector</code></td>
<td>
<p>numeric of length two or a list of numerics of length
two of length of <code>nudge_cluster</code>. If its a list, indicates each translation
vector (in other words, x-y coordinates) that the clusters in
<code>nudge_cluster</code> should be translate by. If its just a numeric, then will
translate all clusters in <code>nudge_cluster</code> by the input - which mostly is
syntactic sugar for translating a single cluster if the input of
<code>nudge_cluster</code> is of length 1.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_recolor_cluster">recolor_cluster</code></td>
<td>
<p>Numeric or Character. Indicates which
cluster(s) based on the index or label to change their color by.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_new_color">new_color</code></td>
<td>
<p>character of arbitrary length. Indicates the corresponding
new colors that selected clusters in <code>recolor_cluster</code> should be changed to.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_rename_label">rename_label</code></td>
<td>
<p>Numeric or character. Indicates the index or name of
label(s) to be renamed.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_new_label">new_label</code></td>
<td>
<p>Character. Indicates the corresponding new label(s) that
selected label(s) in <code>rename_label</code> should be changed to.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_relocate_label">relocate_label</code></td>
<td>
<p>Numeric or character. Indicates the index or name of
label(s) to be relocated.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_label_relocation_coord">label_relocation_coord</code></td>
<td>
<p>Numeric of length two or a list of numerics of
length two of length of <code>relocate_label</code>. If it's a list, indicates each
coordinate that the labels in <code>relocate_label</code> should move to. If it's just
a numeric, then will relocate all labels in <code>relocate_label</code> to the input,
which is likely not desired behavior, so this should only be convenience
syntax if <code>relocate_label</code> has length 1.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_nudge_label">nudge_label</code></td>
<td>
<p>Numeric or character. Indicates the index or name of
label(s) to be &quot;nudged&quot;/translated.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_label_nudge_vector">label_nudge_vector</code></td>
<td>
<p>Numeric of length two or a list of numerics of
length two of length of <code>nudge_label</code>. If it's a list, indicates each
translation vector (in other words, x-y coordinates) that the labels in
<code>nudge_label</code> should be translated by. If it's just a numeric, then will
translate all labels in <code>nudge_label</code> by the input - which mostly is
syntactic sugar for translating a single label if the input of
<code>nudge_label</code> is of length 1.</p>
</td></tr>
<tr><td><code id="AdjustAPOTC_+3A_verbose">verbose</code></td>
<td>
<p>logical. Decides if visual cues are displayed to the R console
of the progress.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The adjusted <code>seurat_obj</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># do an APackOfTheClones run
pbmc &lt;- RunAPOTC(get(data("combined_pbmc")), verbose = FALSE)

# adjust the rad_scale_factor, and nudge cluster 1 by x = 1, y = 1
pbmc &lt;- AdjustAPOTC(
    pbmc,
    new_rad_scale_factor = 0.9,
    nudge_cluster = 1,
    nudge_vector = c(1, 1),
    verbose = FALSE
)

# plot the result
APOTCPlot(pbmc)

# rename some labels
pbmc &lt;- AdjustAPOTC(
    pbmc, rename_label = c(2, 5), new_label = c("Foo", "Bar")
)

# perhaps multiple clusters need to be relocated and repulsed
pbmc &lt;- AdjustAPOTC(
    pbmc,
    relocate_cluster = c("Foo", "C10"), # using labels works too
    relocation_coord = list(c(2, 3.5), c(0, 5)),
    repulse = TRUE,
    verbose = FALSE
)

# plot again to check results
APOTCPlot(pbmc, show_labels = TRUE, verbose = FALSE)

</code></pre>

<hr>
<h2 id='APOTCPlot'>Various variations of visualizations of clonal expansion post-RunAPOTC</h2><span id='topic+APOTCPlot'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>Given a seurat object with an 'apotc' (APackOfTheClones) object
from running <a href="#topic+RunAPOTC">RunAPOTC</a>, this function will read the information and return
a customizable ggplot2 object of the clonal expansion with a circle size
legend. If the user is unhappy about certain aspects of the plot, many
parameters can be adjusted with the <a href="#topic+AdjustAPOTC">AdjustAPOTC</a> function.
</p>
<p>The specific APackOfTheClones run to be plotted can be identified in two
ways: either by inputting the <code>run_id</code> associated with the run that was
either defined / auto-generated during <a href="#topic+RunAPOTC">RunAPOTC</a>, or by inputting the
<code>reduction_base</code>, <code>clonecall</code>, <code>extra_filter</code> and any other keyword arguments
that corresponded to the run. Its heavily recommended to use the <code>run_id</code>.
If none of these parameters are inputted, the function defaults to returning
the plot of the latest run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>APOTCPlot(
  seurat_obj,
  reduction_base = NULL,
  clonecall = NULL,
  ...,
  extra_filter = NULL,
  run_id = NULL,
  show_shared = NULL,
  only_link = NULL,
  clone_link_width = "auto",
  clone_link_color = "black",
  clone_link_alpha = 0.5,
  res = 360L,
  linetype = "blank",
  use_default_theme = TRUE,
  retain_axis_scales = FALSE,
  alpha = 1,
  show_labels = FALSE,
  label_size = 5,
  add_size_legend = TRUE,
  legend_sizes = "auto",
  legend_position = "auto",
  legend_buffer = 0.2,
  legend_color = "#808080",
  legend_spacing = "auto",
  legend_label = "Clone sizes",
  legend_text_size = 5,
  add_legend_background = TRUE,
  add_legend_centerspace = 0,
  detail = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="APOTCPlot_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>A seurat object that has been integrated with clonotype
data and has had a valid run of <a href="#topic+RunAPOTC">RunAPOTC</a>.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_reduction_base">reduction_base</code></td>
<td>
<p>character. The seurat reduction to base the clonal
expansion plotting on. Defaults to <code>'umap'</code> but can be any reduction present
within the reductions slot of the input seurat object, including custom ones.
If ''pca'&ldquo;, the cluster coordinates will be based on PC1 and PC2.
However, generally APackOfTheClones is used for displaying UMAP and
occasionally t-SNE versions to intuitively highlight clonal expansion.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_clonecall">clonecall</code></td>
<td>
<p>character. The column name in the seurat object metadata to
use. See <code>scRepertoire</code> documentation for more information about this
parameter that is central to both packages.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_...">...</code></td>
<td>
<p>additional &quot;subsetting&quot; keyword arguments indicating the rows
corresponding to elements in the seurat object metadata that should be
filtered by. E.g., <code>seurat_clusters = c(1, 9, 10)</code> will filter the cells to
those in the <code>seurat_clusters</code> column with any of the values 1, 9, and 10.
Unfortunately, column names in the seurat object metadata cannot
conflict with the keyword arguments. <em><strong>MAJOR NOTE</strong></em> if any subsetting
keyword arguments are a <em>prefix</em> of any preceding argument names (e.g. a
column named <code>reduction</code> is a prefix of the <code>reduction_base</code> argument)
R will interpret it as the same argument unless <em>both</em> arguments
are named. Additionally, this means any subsequent arguments <em>must</em> be named.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_extra_filter">extra_filter</code></td>
<td>
<p>character. An additional string that should be formatted
<em>exactly</em> like a statement one would pass into <a href="dplyr.html#topic+filter">dplyr::filter</a> that does
<em>additional</em> filtering to cells in the seurat object - on top of the other
keyword arguments - based on the metadata. This means that it will be
logically AND'ed with any keyword argument filters. This is a more flexible
alternative / addition to the filtering keyword arguments. For example, if
one wanted to filter by the length of the amino acid sequence of TCRs, one
could pass in something like <code>extra_filter = "nchar(CTaa) - 1 &gt; 10"</code>. When
involving characters, ensure to enclose with single quotes.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_run_id">run_id</code></td>
<td>
<p>character. This will be the ID associated with the data of a
run, and will be used by other important functions like <a href="#topic+APOTCPlot">APOTCPlot</a> and
<a href="#topic+AdjustAPOTC">AdjustAPOTC</a>. Defaults to <code>NULL</code>, in which case the ID will be generated
in the following format:
</p>
<p><code style="white-space: pre;">&#8288;reduction_base;clonecall;keyword_arguments;extra_filter&#8288;</code>
</p>
<p>where if keyword arguments and extra_filter are underscore characters if
there was no input for the <code>...</code> and <code>extra_filter</code> parameters.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_show_shared">show_shared</code></td>
<td>
<p>The output of <a href="#topic+getSharedClones">getSharedClones</a> can be inputted here,
and the resulting plot will overlay lines between clone circles if that
clonotype is common between clusters. Note that the input <em><strong>must</strong></em> be
generated from data in the correct <code>APackOfTheClones</code> run, and the behavior
is undefined otherwise and will likely error. The next 4 arguments allow for
aesthetic customization of these line links.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_only_link">only_link</code></td>
<td>
<p>Optional integer indicating to only display clone
links originating from this cluster if showing shared clones.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_clone_link_width">clone_link_width</code></td>
<td>
<p>numeric. The width of the lines that connect shared
clones. Defaults to <code>"auto"</code> which will estimate a reasonable value depending
on circle sizes.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_clone_link_color">clone_link_color</code></td>
<td>
<p>character. The color of the lines that connect shared
clones. Defaults to <code>"blend"</code> which will use the average colors of the two
connected clones. Else, any hex color or valid color string input will work,
and the corresponding color will be applied on all links.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_clone_link_alpha">clone_link_alpha</code></td>
<td>
<p>numeric. The alpha of the lines that connect shared
clones.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_res">res</code></td>
<td>
<p>The number of points on the generated path per full circle. From
plot viewers, if circles seem slightly too pixelated, it is recommended to
first try to export the plot as an <code>.svg</code> before increasing <code>res</code> due to
increased plotting times from <a href="ggforce.html#topic+geom_circle">ggforce::geom_circle</a>.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_linetype">linetype</code></td>
<td>
<p>The type of outline each circle should have. defaults to
<code style="white-space: pre;">&#8288;"blank&#8288;</code> meaning no outline. More information is in the function
documentation of <code>ggforce::geom_circle</code>.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_use_default_theme">use_default_theme</code></td>
<td>
<p>logical that defaults to <code>TRUE</code>. If <code>TRUE</code>,
the resulting plot will have the same theme as the seurat reference reduction
plot. Else, the plot will simply have a blank background.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_retain_axis_scales">retain_axis_scales</code></td>
<td>
<p>If <code>TRUE</code>, approximately maintains the axis scales
of the original reduction plot. However, it will only attempt to extend the
axes and never shorten. Users are recommended to set this to <code>TRUE</code>
especially if working with subsetted versions of the clonal data to better
preserve the geometric relation to the original dimensional reduction.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_alpha">alpha</code></td>
<td>
<p>numeric. The alpha of the circles in (0, 1]. Defaults to 1.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_show_labels">show_labels</code></td>
<td>
<p>If <code>TRUE</code>, will label each circle cluster at the centroid,
defaulting to &quot;C0, C1, ...&quot;.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_label_size">label_size</code></td>
<td>
<p>The text size of labels if shown. Defaults to 5.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_add_size_legend">add_size_legend</code></td>
<td>
<p>If <code>TRUE</code>, adds a legend to the plot visualizing the
relative sizes of clones. Note that it is simply an overlay and not a real
ggplot2 legend.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_legend_sizes">legend_sizes</code></td>
<td>
<p>numeric vector. Indicates the circle sizes to be
displayed on the legend, and will always be sorted from smallest to greatest.
Defaults to <code>"auto"</code> which estimate a reasonable range of sizes to display.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_legend_position">legend_position</code></td>
<td>
<p>character or numeric. Can be set to either
<code>"top_left"</code>, <code>"top_right"</code>, <code>"bottom_left"</code>, <code>"bottom_right"</code> and places the
legend roughly in the corresponding position. Otherwise, can be a numeric
vector of length 2 indicating the x and y position of the <em>topmost (smallest)
circle</em> of the legend.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_legend_buffer">legend_buffer</code></td>
<td>
<p>numeric. Indicates how much to &quot;push&quot; the legend towards
the center of the plot from the selected corner. If negative, will push away</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_legend_color">legend_color</code></td>
<td>
<p>character. Indicates the hex color of the circles
displayed on the legend. Defaults to the hex code for a gray tone</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_legend_spacing">legend_spacing</code></td>
<td>
<p>numeric. Indicates the horizontal distance between each
stacked circle on the size legend. Defaults to <code>"auto"</code> which will use an
estimated value depending on plot size</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_legend_label">legend_label</code></td>
<td>
<p>character. The title of the legend, which defaults to
<code style="white-space: pre;">&#8288;"clone sizes&#8288;</code>.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>numeric. The text size of the letters and numbers on
the legend</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_add_legend_background">add_legend_background</code></td>
<td>
<p>logical. If <code>TRUE</code>, will add a border around the
legend and fill the background to be white, overlaying anything else.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_add_legend_centerspace">add_legend_centerspace</code></td>
<td>
<p>numeric. An additional amount of distance
changed between the circle sizes on the left side of the legend and the
numbers on the right. Useful to set to around 0.5 (or more / less) when there
are particularly large clone sizes that may cover the numbers.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_detail">detail</code></td>
<td>
<p>logical. If <code>FALSE</code>, will only plot entire clusters as one
large circle, which may be useful in cases where there are a high number
of clones resulting in a large number of circles on the resulting ggplot,
which has increased plotting times, and certain aspects of the plot needs
to be finely adjusted with <a href="#topic+AdjustAPOTC">AdjustAPOTC</a> or simply inspected. This should
not be set to <code>FALSE</code> for the actual clonal expansion plot.</p>
</td></tr>
<tr><td><code id="APOTCPlot_+3A_verbose">verbose</code></td>
<td>
<p>logical. Decides if visual cues are displayed to the R console
of the progress.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object of the APackOfTheClones clonal expansion plot of the
seurat object. There is an additional 10th element in the object named
<code>"APackOfTheClones"</code> used by other functions in this package and shouldn't
interfere with any other ggplot functionality. (As far as currently known)
</p>


<h3>See Also</h3>

<p><a href="#topic+AdjustAPOTC">AdjustAPOTC</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("combined_pbmc")

combined_pbmc &lt;- RunAPOTC(
    combined_pbmc, run_id = "run1", verbose = FALSE
)

# plotting with default arguments will plot the latest "run1"
clonal_packing_plot &lt;- APOTCPlot(combined_pbmc)

</code></pre>

<hr>
<h2 id='clonal_expansion_plot'>DEFUNCT: Visualize T cell clonal expansion with a ball-packing plot.</h2><span id='topic+clonal_expansion_plot'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#defunct"><img src="../help/figures/lifecycle-defunct.svg" alt='[Defunct]' /></a>
</p>
<p><em><strong>ALL v0.1.x functions are deprecated</strong></em>, and the workflow has been completely revamped - now depending on the scRepertoire v2 package - which allows for the processing of multi-sampled single cell data. Please read the vignettes with <code>browseVignettes("APackOfTheClones")</code>or visit https://qile0317.github.io/APackOfTheClones/
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clonal_expansion_plot(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clonal_expansion_plot_+3A_...">...</code></td>
<td>
<p>arbitrary arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>error message
</p>

<hr>
<h2 id='combined_pbmc'>Example Multi-sampled T-cell seurat object with integrated TCR library</h2><span id='topic+combined_pbmc'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>Generated with <code>scRepertoire::combineExpression</code>. To construct this object
from scratch, try:
</p>
<p>scRepertoire::combineExpression(scRepertoire::combineTCR(get(data(&quot;contig_list&quot;, ,     package = &quot;scRepertoire&quot;)), samples = c(&quot;P17B&quot;, &quot;P17L&quot;, &quot;P18B&quot;, ,     &quot;P18L&quot;, &quot;P19B&quot;, &quot;P19L&quot;, &quot;P20B&quot;, &quot;P20L&quot;), removeNA = FALSE, ,     removeMulti = FALSE, filterMulti = FALSE), get(data(&quot;scRep_example&quot;, ,     package = &quot;scRepertoire&quot;)), cloneCall = &quot;gene&quot;, proportion = TRUE)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("combined_pbmc")
</code></pre>


<h3>Format</h3>

<p>A Seurat object with the following slots filled
</p>

<dl>
<dt>assays</dt><dd>
<p>Currently only contains one assay (&quot;RNA&quot; - scRNA-seq expression data)
</p>
<ul>
<li><p>counts - Raw expression data
</p>
</li>
<li><p>data - Normalized expression data
</p>
</li>
<li><p>scale.data - Scaled expression data
</p>
</li>
<li><p>var.features - names of the current features selected as variable
</p>
</li>
<li><p>meta.features - Assay level metadata such as mean and variance
</p>
</li></ul>
</dd>
<dt>meta.data</dt><dd><p>Cell level metadata with a combined TCR contig list <code style="white-space: pre;">&#8288;from scRepertoire&#8288;</code></p>
</dd>
<dt>active.assay</dt><dd><p>Current default assay</p>
</dd>
<dt>active.ident</dt><dd><p>Current default idents</p>
</dd>
<dt>graphs</dt><dd><p>Neighbor graphs computed, currently stores the SNN</p>
</dd>
<dt>reductions</dt><dd><p>Dimensional reductions: UMAP</p>
</dd>
<dt>version</dt><dd><p>Seurat version used to create the object</p>
</dd>
<dt>commands</dt><dd><p>Command history, including the one used to create this object &quot;combineExpression&quot;</p>
</dd>
</dl>


<hr>
<h2 id='containsApotcRun'>Check for the existence of an APackOfTheClones run with its run id</h2><span id='topic+containsApotcRun'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>A convenience function to check for the existence of an APackOfTheClones
run with its run id, regardless of if any run has been made
</p>


<h3>Usage</h3>

<pre><code class='language-R'>containsApotcRun(seurat_obj, run_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="containsApotcRun_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>a seurat object</p>
</td></tr>
<tr><td><code id="containsApotcRun_+3A_run_id">run_id</code></td>
<td>
<p>character. The id of the associated ApotcRun.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical indicating whether the run exists.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pbmc &lt;- RunAPOTC(
    seurat_obj = get(data("combined_pbmc")),
    reduction_base = "umap",
    clonecall = "strict",
    run_id = "run1",
    verbose = FALSE
)

containsApotcRun(pbmc, "run1")
#&gt; [1] TRUE

containsApotcRun(pbmc, "run2")
#&gt; [1] FALSE

</code></pre>

<hr>
<h2 id='count_clone_sizes'>DEFUNCT: count the number of clonotype sizes per cell cluster in a seurat
object integrated with a TCR library</h2><span id='topic+count_clone_sizes'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#defunct"><img src="../help/figures/lifecycle-defunct.svg" alt='[Defunct]' /></a>
</p>
<p><em><strong>ALL v0.1.x functions are deprecated</strong></em>, and the workflow has been completely revamped - now depending on the scRepertoire v2 package - which allows for the processing of multi-sampled single cell data. Please read the vignettes with <code>browseVignettes("APackOfTheClones")</code>or visit https://qile0317.github.io/APackOfTheClones/
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count_clone_sizes(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="count_clone_sizes_+3A_...">...</code></td>
<td>
<p>arbitrary arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>error message
</p>

<hr>
<h2 id='countCloneSizes'>count the number of clonotype sizes in a seurat object combined with a
VDJ library overall or by cluster</h2><span id='topic+countCloneSizes'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>Get clonotype frequencies from a seurat object's metadata, either as one
whole table, or in a list of tables, based on the current / some custom
ident of each cell. Note that depending on the ident (indicated by the
<code>by_cluster</code> argument) there may be more or less clonotypes counted based
on the number of rows containing NA for that column of that ident if it
isn't the active ident.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>countCloneSizes(
  seurat_obj,
  clonecall = "strict",
  extra_filter = NULL,
  ...,
  by_cluster = TRUE,
  sort_decreasing = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="countCloneSizes_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>a seurat object combined with a VDJ library with the
<code>scRepertoire</code>.</p>
</td></tr>
<tr><td><code id="countCloneSizes_+3A_clonecall">clonecall</code></td>
<td>
<p>character. The column name in the seurat object metadata to
use. See <code>scRepertoire</code> documentation for more information about this
parameter that is central to both packages.</p>
</td></tr>
<tr><td><code id="countCloneSizes_+3A_extra_filter">extra_filter</code></td>
<td>
<p>character. An additional string that should be formatted
<em>exactly</em> like a statement one would pass into <a href="dplyr.html#topic+filter">dplyr::filter</a> that does
<em>additional</em> filtering to cells in the seurat object - on top of the other
keyword arguments - based on the metadata. This means that it will be
logically AND'ed with any keyword argument filters. This is a more flexible
alternative / addition to the filtering keyword arguments. For example, if
one wanted to filter by the length of the amino acid sequence of TCRs, one
could pass in something like <code>extra_filter = "nchar(CTaa) - 1 &gt; 10"</code>. When
involving characters, ensure to enclose with single quotes.</p>
</td></tr>
<tr><td><code id="countCloneSizes_+3A_...">...</code></td>
<td>
<p>additional &quot;subsetting&quot; keyword arguments indicating the rows
corresponding to elements in the seurat object metadata that should be
filtered by. E.g., <code>seurat_clusters = c(1, 9, 10)</code> will filter the cells to
those in the <code>seurat_clusters</code> column with any of the values 1, 9, and 10.
Unfortunately, column names in the seurat object metadata cannot
conflict with the keyword arguments. <em><strong>MAJOR NOTE</strong></em> if any subsetting
keyword arguments are a <em>prefix</em> of any preceding argument names (e.g. a
column named <code>reduction</code> is a prefix of the <code>reduction_base</code> argument)
R will interpret it as the same argument unless <em>both</em> arguments
are named. Additionally, this means any subsequent arguments <em>must</em> be named.</p>
</td></tr>
<tr><td><code id="countCloneSizes_+3A_by_cluster">by_cluster</code></td>
<td>
<p>Logical or Character. If <code>TRUE</code>, will output a list of
table objects, with the table at each index corresponding to level in
Idents(). Each table's names are the clonotype name indicated by <code>clonecall</code>
after filtering, while the values are the actual clone sizes. If <code>FALSE</code>,
outputs just the aggregate clone sizes for all cells. Note that if <code>FALSE</code>,
the output should be identical to that produced by
<code>mergeCloneSizes(countCloneSizes(..., by_cluster = TRUE))</code>. Otherwise, this
argument can also be a character indicating some column in the seurat object
metadata to use a cell identity guiding (e.g. <code>"seurat_clusters"</code>).</p>
</td></tr>
<tr><td><code id="countCloneSizes_+3A_sort_decreasing">sort_decreasing</code></td>
<td>
<p>a logical or NULL. If <code>TRUE</code>/<code>FALSE</code>, sorts each/the
table by clonotype frequency with largest/smallest clones first with a stable
sorting algorithm, and if NULL, no order is guaranteed but the output is
deterministic.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of tables or a single table depending on <code>by_cluster</code>
</p>


<h3>See Also</h3>

<p><a href="#topic+mergeCloneSizes">mergeCloneSizes</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("combined_pbmc")

countCloneSizes(combined_pbmc)
countCloneSizes(combined_pbmc, "aa")
countCloneSizes(combined_pbmc, "nt", orig.ident = c("P17B", "P17L"))

</code></pre>

<hr>
<h2 id='deleteApotcData'>Delete the results of an APackOfTheClones run</h2><span id='topic+deleteApotcData'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>A convenience function to erase all data associated with a particular run,
including the ApotcData and command in seurat_obj@command. The run_id would
be no longer accessible afterwards.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deleteApotcData(seurat_obj, run_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="deleteApotcData_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>a seurat object that has had RunAPOTC ran on it before in
order of the functions being called.</p>
</td></tr>
<tr><td><code id="deleteApotcData_+3A_run_id">run_id</code></td>
<td>
<p>character. The id of the associated ApotcRun.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The modified input seurat object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pbmc &lt;- RunAPOTC(
    seurat_obj = get(data("combined_pbmc")),
    reduction_base = "umap",
    clonecall = "strict",
    run_id = "run1",
    verbose = FALSE
)

getApotcDataIds(pbmc)
#&gt; [1] "run1"

# delete the data
pbmc &lt;- deleteApotcData(pbmc, "run1")

getApotcDataIds(pbmc)
#&gt; NULL

</code></pre>

<hr>
<h2 id='getApotcDataIds'>Get all run ids of previous RunAPOTC runs on a seurat object</h2><span id='topic+getApotcDataIds'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>A convenience function to get all run ids of previous RunAPOTC run IDs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getApotcDataIds(seurat_obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getApotcDataIds_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>a seurat object that has had RunAPOTC ran on it before in
order of the functions being called.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of all run ids of previous RunAPOTC runs, in
the order they were ran in. If there are no runs on the object, it returns
<code>NULL</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pbmc &lt;- RunAPOTC(
    seurat_obj = get(data("combined_pbmc")),
    reduction_base = "umap",
    clonecall = "strict",
    verbose = FALSE
)

getApotcDataIds(pbmc)
#&gt; [1] "umap;CTstrict;_;_"

pbmc &lt;- RunAPOTC(
    seurat_obj = pbmc,
    reduction_base = "umap",
    clonecall = "gene",
    verbose = FALSE
)

getApotcDataIds(pbmc)
#&gt; [1] "umap;CTstrict;_;_" "umap;CTgene;_;_"

</code></pre>

<hr>
<h2 id='getLastApotcDataId'>Get the object id of the most recent RunAPOTC run on a seurat object</h2><span id='topic+getLastApotcDataId'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>A convenience function to get the object id of the most recent valid
<a href="#topic+RunAPOTC">RunAPOTC</a> run, to be used by <a href="#topic+APOTCPlot">APOTCPlot</a> and <a href="#topic+AdjustAPOTC">AdjustAPOTC</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getLastApotcDataId(seurat_obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getLastApotcDataId_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>a seurat object that has had RunAPOTC ran on it before in
order of the functions being called.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character of the object id of the last <a href="#topic+RunAPOTC">RunAPOTC</a> call
</p>


<h3>Examples</h3>

<pre><code class='language-R'># first run
pbmc &lt;- RunAPOTC(
    seurat_obj = get(data("combined_pbmc")),
    reduction_base = "umap",
    clonecall = "strict",
    verbose = FALSE
)

getLastApotcDataId(pbmc)
#&gt; [1] "umap;CTstrict;_;_"

# second run with a different clonecall
pbmc &lt;- RunAPOTC(
    seurat_obj = pbmc,
    reduction_base = "umap",
    clonecall = "gene",
    verbose = FALSE
)

getLastApotcDataId(pbmc)
#&gt; [1] "umap;CTgene;_;_"

</code></pre>

<hr>
<h2 id='getReductionCentroids'>Calculate seurat cluster centroids based on a Dimensional reduction</h2><span id='topic+getReductionCentroids'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>Utility function to calculate the physical xy coordinates of each seurat
cluster based on a dimensional reduction already present in the object.
The results are returned in a list with the length of the number of distinct
seurat clusters based on the seurat_obj <code>meta.data</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getReductionCentroids(seurat_obj, reduction)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getReductionCentroids_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>input seurat object with the dimensional reduction of
choice already present, and seurat clusters computed.</p>
</td></tr>
<tr><td><code id="getReductionCentroids_+3A_reduction">reduction</code></td>
<td>
<p>character. The reduction that the centroid calculation
should be based on.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of the length of the number of distinct clusters in the
seurat object metadata, where each element of the list is a numeric vector
of length 2, with the numbers corresponding to the x and y coordinate
respectively of the seurat cluster with the corresponding index.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("combined_pbmc")
getReductionCentroids(combined_pbmc, reduction = "umap")

</code></pre>

<hr>
<h2 id='getSharedClones'>Compute a list of clonotypes that are shared between seurat clusters</h2><span id='topic+getSharedClones'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>This function allows users to get a list of clonotypes that are shared
between clusters based on the levels of the active cell identities / some
custom identity based on the <code>alt_ident</code>. A list is returned with its
<em><strong>names</strong></em> being the shared clonotypes, and the values are numeric vectors
indicating the index of the clusters that clonotype is found in. The index
corresponds to the index in the default levels of the factored identities.
</p>
<p>If <code>run_id</code> is inputted, then the function will attempt to get the shared
clonotypes from the corresponding APackOfTheClones run generated from
<a href="#topic+RunAPOTC">RunAPOTC</a>. Otherwise, it will use the filtering / subsetting parameters
to generate the shared clones.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSharedClones(
  seurat_obj,
  reduction_base = "umap",
  clonecall = "strict",
  ...,
  extra_filter = NULL,
  alt_ident = NULL,
  run_id = NULL,
  top = NULL,
  top_per_cl = NULL,
  intop = NULL,
  intop_per_cl = NULL,
  publicity = c(2L, Inf)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSharedClones_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>Seurat object with one or more dimension reductions and
already have been integrated with a TCR/BCR library with
<code>scRepertoire::combineExpression</code>.</p>
</td></tr>
<tr><td><code id="getSharedClones_+3A_reduction_base">reduction_base</code></td>
<td>
<p>character. The seurat reduction to base the clonal
expansion plotting on. Defaults to <code>'umap'</code> but can be any reduction present
within the reductions slot of the input seurat object, including custom ones.
If ''pca'&ldquo;, the cluster coordinates will be based on PC1 and PC2.
However, generally APackOfTheClones is used for displaying UMAP and
occasionally t-SNE versions to intuitively highlight clonal expansion.</p>
</td></tr>
<tr><td><code id="getSharedClones_+3A_clonecall">clonecall</code></td>
<td>
<p>character. The column name in the seurat object metadata to
use. See <code>scRepertoire</code> documentation for more information about this
parameter that is central to both packages.</p>
</td></tr>
<tr><td><code id="getSharedClones_+3A_...">...</code></td>
<td>
<p>additional &quot;subsetting&quot; keyword arguments indicating the rows
corresponding to elements in the seurat object metadata that should be
filtered by. E.g., <code>seurat_clusters = c(1, 9, 10)</code> will filter the cells to
those in the <code>seurat_clusters</code> column with any of the values 1, 9, and 10.
Unfortunately, column names in the seurat object metadata cannot
conflict with the keyword arguments. <em><strong>MAJOR NOTE</strong></em> if any subsetting
keyword arguments are a <em>prefix</em> of any preceding argument names (e.g. a
column named <code>reduction</code> is a prefix of the <code>reduction_base</code> argument)
R will interpret it as the same argument unless <em>both</em> arguments
are named. Additionally, this means any subsequent arguments <em>must</em> be named.</p>
</td></tr>
<tr><td><code id="getSharedClones_+3A_extra_filter">extra_filter</code></td>
<td>
<p>character. An additional string that should be formatted
<em>exactly</em> like a statement one would pass into <a href="dplyr.html#topic+filter">dplyr::filter</a> that does
<em>additional</em> filtering to cells in the seurat object - on top of the other
keyword arguments - based on the metadata. This means that it will be
logically AND'ed with any keyword argument filters. This is a more flexible
alternative / addition to the filtering keyword arguments. For example, if
one wanted to filter by the length of the amino acid sequence of TCRs, one
could pass in something like <code>extra_filter = "nchar(CTaa) - 1 &gt; 10"</code>. When
involving characters, ensure to enclose with single quotes.</p>
</td></tr>
<tr><td><code id="getSharedClones_+3A_alt_ident">alt_ident</code></td>
<td>
<p>character. By default, cluster identity is assumed to be
whatever is in <code>Idents(seurat_obj)</code>, and clones will be grouped by the active
ident. However, <code>alt_ident</code> could be set as the name of some column in the
meta data of the seurat object to be grouped by. This column is meant to have
been a product of <code>Seurat::StashIdent</code> or manually added.</p>
</td></tr>
<tr><td><code id="getSharedClones_+3A_run_id">run_id</code></td>
<td>
<p>character. This will be the ID associated with the data of a
run, and will be used by other important functions like <a href="#topic+APOTCPlot">APOTCPlot</a> and
<a href="#topic+AdjustAPOTC">AdjustAPOTC</a>. Defaults to <code>NULL</code>, in which case the ID will be generated
in the following format:
</p>
<p><code style="white-space: pre;">&#8288;reduction_base;clonecall;keyword_arguments;extra_filter&#8288;</code>
</p>
<p>where if keyword arguments and extra_filter are underscore characters if
there was no input for the <code>...</code> and <code>extra_filter</code> parameters.</p>
</td></tr>
<tr><td><code id="getSharedClones_+3A_top">top</code></td>
<td>
<p>integer or numeric in (0, 1) - if not null, filters the output
clones so that only the shared clonotypes with counts the top <code>top</code> count /
proportion (for numeric in (0, 1) input) shared clones are kept. For cases
where several clonotypes tie in size, the clonotype(s) added are not
guaranteed but deterministic given the other arguments are identical.</p>
</td></tr>
<tr><td><code id="getSharedClones_+3A_top_per_cl">top_per_cl</code></td>
<td>
<p>integer or numeric in (0, 1) - if not null, filters the
output clones so that for each seurat cluster, only the clonotypes with the
<code>top_per_cl</code> frequency/count is preserved when aggregating shared clones,
in the same way as the above. Note that if inputted in conjunction with
<code>top</code>, it will get the <em>intersection</em> of the clonotypes filtered each way.
For cases where several clonotypes tie in size, the clonotype(s) added are
not guaranteed but deterministic given the other arguments are identical.</p>
</td></tr>
<tr><td><code id="getSharedClones_+3A_intop">intop</code></td>
<td>
<p>integer or numeric in (0, 1) - if not null, filters the raw
clone sizes before computing the shared clonotypes so that only the
clonotypes that have their overall size in the top <code>intop</code> largest sizes
(if it is integer, else the <code>intop</code> proportion) are kept. To emphasize,
this argument <em><strong>does not necessarily</strong></em> return the <code>top</code> shared clones
and likely a little less, because this filters the raw clone sizes, of
which, its very likely that not all those clones end up being shared.</p>
</td></tr>
<tr><td><code id="getSharedClones_+3A_intop_per_cl">intop_per_cl</code></td>
<td>
<p>integer or numeric in (0, 1) - if not null, filters
the raw <em>clustered</em> clone sizes before computing shared clones, so that
for every clone in a seurat cluster, the top <code>intop_per_cl</code> count /
proportion (for numeric in (0, 1) input) clones are kept.</p>
</td></tr>
<tr><td><code id="getSharedClones_+3A_publicity">publicity</code></td>
<td>
<p>numeric pair. A simple filter range of
<code>c(lowerbound, upperbound)</code> to retain only shared clones with their
&quot;publicity&quot; - number of clusters they are present in - within this
range.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named list where each name is a clonotype, each element is a
numeric indicating which seurat cluster(s) its in, in no particular order.
If no shared clones are present, the output is an empty list.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("combined_pbmc")

getSharedClones(combined_pbmc)

getSharedClones(
    combined_pbmc,
    orig.ident = c("P17B", "P18B"), # a named subsetting parameter
    clonecall = "aa"
)

# extract shared clones from a past RunAPOTC run
combined_pbmc &lt;- RunAPOTC(
    combined_pbmc, run_id = "foo", verbose = FALSE
)

getSharedClones(
    combined_pbmc, run_id = "foo", top = 5
)

# doing a run and then getting the clones works too
combined_pbmc &lt;- RunAPOTC(combined_pbmc, run_id = "run1", verbose = FALSE)
getSharedClones(combined_pbmc, run_id = "run1")

</code></pre>

<hr>
<h2 id='integrate_tcr'>DEFUNCT: Integrate a single TCR library into Seurat object metadata</h2><span id='topic+integrate_tcr'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#defunct"><img src="../help/figures/lifecycle-defunct.svg" alt='[Defunct]' /></a>
</p>
<p><em><strong>ALL v0.1.x functions are deprecated</strong></em>, and the workflow has been completely revamped - now depending on the scRepertoire v2 package - which allows for the processing of multi-sampled single cell data. Please read the vignettes with <code>browseVignettes("APackOfTheClones")</code>or visit https://qile0317.github.io/APackOfTheClones/
</p>


<h3>Usage</h3>

<pre><code class='language-R'>integrate_tcr(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="integrate_tcr_+3A_...">...</code></td>
<td>
<p>arbitrary arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>error message
</p>

<hr>
<h2 id='mergeCloneSizes'>Merge a list of Clustered Clonotype Frequency Tables</h2><span id='topic+mergeCloneSizes'></span>

<h3>Description</h3>

<p>The list of clustered clonotype frequencies from <a href="#topic+countCloneSizes">countCloneSizes</a>
can be merged by this function to a frequency table of all clonotypes
similar to the data that can be seen in the seurat object metadata.
By default, this function sorts the table with largest clonotypes first,
and this may be useful for quickly gauging which clonotypes are the most
expanded overall.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mergeCloneSizes(clustered_clone_sizes, sort_decreasing = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mergeCloneSizes_+3A_clustered_clone_sizes">clustered_clone_sizes</code></td>
<td>
<p>the output of <a href="#topic+countCloneSizes">countCloneSizes</a>.</p>
</td></tr>
<tr><td><code id="mergeCloneSizes_+3A_sort_decreasing">sort_decreasing</code></td>
<td>
<p>a logical or NULL. If <code>TRUE</code>/<code>FALSE</code>, sorts the
table by clonotype frequency with largest/smallest clones first, and if
NULL, no order is guaranteed but the output is deterministic.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a table object
</p>


<h3>See Also</h3>

<p><a href="#topic+countCloneSizes">countCloneSizes</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>clustered_clone_sizes &lt;- countCloneSizes(get(data("combined_pbmc")))
mergeCloneSizes(clustered_clone_sizes)

</code></pre>

<hr>
<h2 id='overlayLegend'>overlay a clone size legend on an APackOfTheClones plot</h2><span id='topic+overlayLegend'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>This function has most of the parameters related to legend in
<a href="#topic+APOTCPlot">APOTCPlot</a>, and can plot a new / override the current legend.
However, it is very important that the input plot to the function
is a plot generated solely by <a href="#topic+APOTCPlot">APOTCPlot</a> or <a href="#topic+vizAPOTC">vizAPOTC</a> due to it
being a custom ggplot object. It will not override or erase any
additional layers that the user/other functions have added. To just
remove the legend, see <a href="#topic+removeLegend">removeLegend</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>overlayLegend(
  apotc_ggplot,
  legend_sizes = "auto",
  legend_position = "auto",
  legend_buffer = 0.2,
  legend_color = "#808080",
  legend_spacing = "auto",
  legend_label = "Clone sizes",
  legend_text_size = 5,
  add_legend_background = TRUE,
  add_legend_centerspace = 0,
  linetype = "blank",
  res = 360L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="overlayLegend_+3A_apotc_ggplot">apotc_ggplot</code></td>
<td>
<p>a ggplot object that is the output of <a href="#topic+APOTCPlot">APOTCPlot</a> or
<a href="#topic+vizAPOTC">vizAPOTC</a></p>
</td></tr>
<tr><td><code id="overlayLegend_+3A_legend_sizes">legend_sizes</code></td>
<td>
<p>numeric vector. Indicates the circle sizes to be
displayed on the legend, and will always be sorted from smallest to greatest.
Defaults to <code>"auto"</code> which estimate a reasonable range of sizes to display.</p>
</td></tr>
<tr><td><code id="overlayLegend_+3A_legend_position">legend_position</code></td>
<td>
<p>character or numeric. Can be set to either
<code>"top_left"</code>, <code>"top_right"</code>, <code>"bottom_left"</code>, <code>"bottom_right"</code> and places the
legend roughly in the corresponding position. Otherwise, can be a numeric
vector of length 2 indicating the x and y position of the <em>topmost (smallest)
circle</em> of the legend.</p>
</td></tr>
<tr><td><code id="overlayLegend_+3A_legend_buffer">legend_buffer</code></td>
<td>
<p>numeric. Indicates how much to &quot;push&quot; the legend towards
the center of the plot from the selected corner. If negative, will push away</p>
</td></tr>
<tr><td><code id="overlayLegend_+3A_legend_color">legend_color</code></td>
<td>
<p>character. Indicates the hex color of the circles
displayed on the legend. Defaults to the hex code for a gray tone</p>
</td></tr>
<tr><td><code id="overlayLegend_+3A_legend_spacing">legend_spacing</code></td>
<td>
<p>numeric. Indicates the horizontal distance between each
stacked circle on the size legend. Defaults to <code>"auto"</code> which will use an
estimated value depending on plot size</p>
</td></tr>
<tr><td><code id="overlayLegend_+3A_legend_label">legend_label</code></td>
<td>
<p>character. The title of the legend, which defaults to
<code style="white-space: pre;">&#8288;"clone sizes&#8288;</code>.</p>
</td></tr>
<tr><td><code id="overlayLegend_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>numeric. The text size of the letters and numbers on
the legend</p>
</td></tr>
<tr><td><code id="overlayLegend_+3A_add_legend_background">add_legend_background</code></td>
<td>
<p>logical. If <code>TRUE</code>, will add a border around the
legend and fill the background to be white, overlaying anything else.</p>
</td></tr>
<tr><td><code id="overlayLegend_+3A_add_legend_centerspace">add_legend_centerspace</code></td>
<td>
<p>numeric. An additional amount of distance
changed between the circle sizes on the left side of the legend and the
numbers on the right. Useful to set to around 0.5 (or more / less) when there
are particularly large clone sizes that may cover the numbers.</p>
</td></tr>
<tr><td><code id="overlayLegend_+3A_linetype">linetype</code></td>
<td>
<p>The type of outline each circle should have. defaults to
<code style="white-space: pre;">&#8288;"blank&#8288;</code> meaning no outline. More information is in the function
documentation of <code>ggforce::geom_circle</code>.</p>
</td></tr>
<tr><td><code id="overlayLegend_+3A_res">res</code></td>
<td>
<p>The number of points on the generated path per full circle. From
plot viewers, if circles seem slightly too pixelated, it is recommended to
first try to export the plot as an <code>.svg</code> before increasing <code>res</code> due to
increased plotting times from <a href="ggforce.html#topic+geom_circle">ggforce::geom_circle</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The size legend on APackOfTheClones plots are simply a collection of
annotation layers of the <code>rect</code>, <code>text</code>, <code>circle</code> geoms. Therefore it
isn't quite a ggplot legend. In the actual ggplot object, the <code style="white-space: pre;">&#8288;$layers&#8288;</code>
element should all be named with an empty character <code>""</code>, and those
that comprise the annotation layers of the legend should be named whatever
<code>APackOfTheClones:::.ApotcLegendLayerName</code> is. Note that this is simply
an implementation detail that the user should not interfere with.
</p>


<h3>Value</h3>

<p>A ggplot object of the APackOfTheClones clonal expansion plot of the
seurat object. There is an additional 10th element in the object named
<code>"APackOfTheClones"</code> used by other functions in this package and shouldn't
interfere with any other ggplot functionality. (As far as currently known)
</p>


<h3>See Also</h3>

<p><a href="#topic+removeLegend">removeLegend</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(magrittr)

# create a plot with a legend
apotc_plot &lt;- vizAPOTC(get(data("combined_pbmc")), verbose = FALSE)

# reposition the legend to top right
overlayLegend(apotc_plot, legend_position = "top right")

# use different sizes and label - may be nice to use the pipe
apotc_plot %&gt;% overlayLegend(
    legend_sizes = c(1, 3, 7, 9),
    legend_label = "odd sizes"
)

</code></pre>

<hr>
<h2 id='removeLegend'>Remove current APackOfTheClones legend</h2><span id='topic+removeLegend'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>Removes the clone size legend on an APackOfTheClones plot, if
one is present. Will preserve any additional ggplot layers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>removeLegend(apotc_ggplot)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="removeLegend_+3A_apotc_ggplot">apotc_ggplot</code></td>
<td>
<p>a ggplot object that is the output of <a href="#topic+APOTCPlot">APOTCPlot</a> or
<a href="#topic+vizAPOTC">vizAPOTC</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object of the APackOfTheClones clonal expansion plot of the
seurat object. There is an additional 10th element in the object named
<code>"APackOfTheClones"</code> used by other functions in this package and shouldn't
interfere with any other ggplot functionality. (As far as currently known)
</p>


<h3>See Also</h3>

<p><a href="#topic+overlayLegend">overlayLegend</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># create an APackOfTheClones plot with a legend
apotc_plot &lt;- vizAPOTC(
    get(data("combined_pbmc")),
    add_size_legend = TRUE,
    verbose = FALSE
)

# remove the legend
apotc_plot &lt;- removeLegend(apotc_plot)
apotc_plot

</code></pre>

<hr>
<h2 id='renameApotcRun'>Rename an APackOfTheClones run</h2><span id='topic+renameApotcRun'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>A function to rename an APackOfTheClones run identified by its run id in
a Seurat object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>renameApotcRun(seurat_obj, old_run_id, new_run_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="renameApotcRun_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>A Seurat object containing APackOfTheClones data - the
output of <a href="#topic+RunAPOTC">RunAPOTC</a></p>
</td></tr>
<tr><td><code id="renameApotcRun_+3A_old_run_id">old_run_id</code></td>
<td>
<p>Character. The current id of the APackOfTheClones run to
be renamed.</p>
</td></tr>
<tr><td><code id="renameApotcRun_+3A_new_run_id">new_run_id</code></td>
<td>
<p>Character. The new id to assign to the APackOfTheClones
run.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Seurat object with the APackOfTheClones run renamed.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pbmc &lt;- RunAPOTC(
    seurat_obj = get(data("combined_pbmc")),
    reduction_base = "umap",
    clonecall = "strict",
    run_id = "run1",
    verbose = FALSE
)

pbmc &lt;- renameApotcRun(pbmc, "run1", "new_run")
# Now "run1" has been renamed to "new_run"

</code></pre>

<hr>
<h2 id='RunAPOTC'>Run the APackOfTheClones method on a combined Seurat object for
downstream visualization of clonal expansion</h2><span id='topic+RunAPOTC'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>Computes necessary information for an APackOfTheClones
clonal expansion plot (<a href="#topic+APOTCPlot">APOTCPlot</a>) and stores it in the seurat object.
Gets sizes of unique clones and utilizes a circle-packing algorithm to
pack circles representing individual clones in approximately the same
dimensional reduction (<code>reduction_base</code>) coordinates based on some cell
ident (defaults to the active ident).
</p>
<p>The parameter <code>extra_filter</code> along with an unlimited number of additional
keyword arguments can be used to filter the cells by certain conditions
in the metadata, and new results will be stored in addition to other runs
the users may have done.
</p>
<p>Each APackOfTheClones run is uniquely identified by the parameters
<code>reduction_base</code>, <code>clonecall</code>, <code>extra_filter</code>, and any additional keywords
passed to filter the metadata. Each distinct run result is stored in the
seurat object and has an associated Id generated from the aforementioned
parameters. To view the id of the latest run, call <a href="#topic+getLastApotcDataId">getLastApotcDataId</a>.
To view all the ids of previous runs, call <a href="#topic+getApotcDataIds">getApotcDataIds</a>. To work further
with a specific run (most importantly, plotting), the user can use this id
in the arguments with is slightly more convenient than passing in the
original RunAPOTC parameters again but both ways work.
</p>
<p>If the user wishes to manually customize/fix the expansion plot
generated, the circular packing information can be modified
with the <a href="#topic+AdjustAPOTC">AdjustAPOTC</a> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RunAPOTC(
  seurat_obj,
  reduction_base = "umap",
  clonecall = "strict",
  ...,
  extra_filter = NULL,
  alt_ident = NULL,
  run_id = NULL,
  clone_scale_factor = "auto",
  rad_scale_factor = 0.95,
  order_clones = TRUE,
  try_place = FALSE,
  repulse = TRUE,
  repulsion_threshold = 1,
  repulsion_strength = 1,
  max_repulsion_iter = 20L,
  override = FALSE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RunAPOTC_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>Seurat object with one or more dimension reductions and
already have been integrated with a TCR/BCR library with
<code>scRepertoire::combineExpression</code>.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_reduction_base">reduction_base</code></td>
<td>
<p>character. The seurat reduction to base the clonal
expansion plotting on. Defaults to <code>'umap'</code> but can be any reduction present
within the reductions slot of the input seurat object, including custom ones.
If ''pca'&ldquo;, the cluster coordinates will be based on PC1 and PC2.
However, generally APackOfTheClones is used for displaying UMAP and
occasionally t-SNE versions to intuitively highlight clonal expansion.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_clonecall">clonecall</code></td>
<td>
<p>character. The column name in the seurat object metadata to
use. See <code>scRepertoire</code> documentation for more information about this
parameter that is central to both packages.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_...">...</code></td>
<td>
<p>additional &quot;subsetting&quot; keyword arguments indicating the rows
corresponding to elements in the seurat object metadata that should be
filtered by. E.g., <code>seurat_clusters = c(1, 9, 10)</code> will filter the cells to
those in the <code>seurat_clusters</code> column with any of the values 1, 9, and 10.
Unfortunately, column names in the seurat object metadata cannot
conflict with the keyword arguments. <em><strong>MAJOR NOTE</strong></em> if any subsetting
keyword arguments are a <em>prefix</em> of any preceding argument names (e.g. a
column named <code>reduction</code> is a prefix of the <code>reduction_base</code> argument)
R will interpret it as the same argument unless <em>both</em> arguments
are named. Additionally, this means any subsequent arguments <em>must</em> be named.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_extra_filter">extra_filter</code></td>
<td>
<p>character. An additional string that should be formatted
<em>exactly</em> like a statement one would pass into <a href="dplyr.html#topic+filter">dplyr::filter</a> that does
<em>additional</em> filtering to cells in the seurat object - on top of the other
keyword arguments - based on the metadata. This means that it will be
logically AND'ed with any keyword argument filters. This is a more flexible
alternative / addition to the filtering keyword arguments. For example, if
one wanted to filter by the length of the amino acid sequence of TCRs, one
could pass in something like <code>extra_filter = "nchar(CTaa) - 1 &gt; 10"</code>. When
involving characters, ensure to enclose with single quotes.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_alt_ident">alt_ident</code></td>
<td>
<p>character. By default, cluster identity is assumed to be
whatever is in <code>Idents(seurat_obj)</code>, and clones will be grouped by the active
ident. However, <code>alt_ident</code> could be set as the name of some column in the
meta data of the seurat object to be grouped by. This column is meant to have
been a product of <code>Seurat::StashIdent</code> or manually added.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_run_id">run_id</code></td>
<td>
<p>character. This will be the ID associated with the data of a
run, and will be used by other important functions like <a href="#topic+APOTCPlot">APOTCPlot</a> and
<a href="#topic+AdjustAPOTC">AdjustAPOTC</a>. Defaults to <code>NULL</code>, in which case the ID will be generated
in the following format:
</p>
<p><code style="white-space: pre;">&#8288;reduction_base;clonecall;keyword_arguments;extra_filter&#8288;</code>
</p>
<p>where if keyword arguments and extra_filter are underscore characters if
there was no input for the <code>...</code> and <code>extra_filter</code> parameters.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_clone_scale_factor">clone_scale_factor</code></td>
<td>
<p>Dictates how much to scale each circle(between 0,1)
radius when converting from clonotype counts into circles that represent
individual clonotypes. The argument defaults to the character <code>"auto"</code>, and
if so, the most visually pleasing factor will be estimated.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_rad_scale_factor">rad_scale_factor</code></td>
<td>
<p>numeric between 0 and 1. This value decreases the
radius of the smallest clones by this scale factor. And the absolute value
of this decrease will be applied to all packed circles, effectively shrinking
all circles on the spot, and introduce more constant spacing in between.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_order_clones">order_clones</code></td>
<td>
<p>logical. Decides if the largest clone circles should be
near cluster centroids. This is highly recommended to be set to TRUE for
increased intuitiveness of the visualization, as resulting plots tend to
give an improved impression of the proportion of expanded clones. If
<code style="white-space: pre;">&#8288;FALSE,&#8288;</code> will randomly scramble the positions of each circle. For the sake
of being replicable, a random seed is recommended to be set with <a href="base.html#topic+set.seed">set.seed</a>.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_try_place">try_place</code></td>
<td>
<p>If <code>TRUE</code>, always minimizes distance from a newly placed
circle to the origin in the circle packing algorithm.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_repulse">repulse</code></td>
<td>
<p>If <code>TRUE</code>, will attempt to push overlapping clusters away from
each other.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_repulsion_threshold">repulsion_threshold</code></td>
<td>
<p>numeric. The radius that clonal circle clusters
overlap is acceptable when repulsing.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_repulsion_strength">repulsion_strength</code></td>
<td>
<p>numeric. The smaller the value the less the
clusters repulse each other per iteration, and vice versa.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_max_repulsion_iter">max_repulsion_iter</code></td>
<td>
<p>integer. The number of repulsion iterations.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_override">override</code></td>
<td>
<p>logical. If <code>TRUE</code>, will override any existing
APackOfTheClones run data with the same <code>run_id</code>.</p>
</td></tr>
<tr><td><code id="RunAPOTC_+3A_verbose">verbose</code></td>
<td>
<p>logical. Decides if visual cues are displayed to the R console
of the progress.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that the subsetting arguments <code>...</code> and <code>extra_filter</code> are only a
quick convenience to subset based on metadata, and the <code>subset</code> S3 method
defined in <code>Seurat</code> is much more mature are has more features. Additionally,
users need to work with data subsets are recommended to and likely already
are working with seurat objects subsetted/split with <code>Seurat::SplitObject</code>.
</p>
<p>All APackOfTheClones run data is stored in the Seurat object under
<code>seurat_object@misc$APackOfTheClones</code>, which is a list of S4 objects of the
type &quot;ApotcData&quot;, with each element corresponding to a unique run. The id of
each run is the name of each element in the list. The user
<em><strong>really shouldn't</strong></em> manually modify anything in the list as it may cause
unexpected behavior with many other functions.
</p>
<p>Additionally, it logs a seurat command associated with the run in the
<code>seurat_object@commands</code> slot as a &quot;SeuratCommand&quot; object (from Seurat),
where the name of the object in the list is formatted as <code>RunAPOTC.run_id</code>.
</p>


<h3>Value</h3>

<p>A modified version of the input seurat object, which harbors data
necessary for visualizing the clonal expansion of the cells with <a href="#topic+APOTCPlot">APOTCPlot</a>
and has a friendly user interface to modify certain attributes with
<a href="#topic+AdjustAPOTC">AdjustAPOTC</a>.
</p>


<h3>Cluster labelling</h3>

<p>For the ident that was used to cluster the clones, labels for each cluster
are inferred and stored in the run so that they can be used by other
functions and optionally overlaid on the plot over clusters. If the levels
of the ident used is a naturally ordered integer sequence, then the labels
generated would be <code style="white-space: pre;">&#8288;"C1", "C2", "C3" ... &#8288;</code>, else they would be the actual
ident levels themselves.
</p>


<h3>See Also</h3>

<p><a href="#topic+APOTCPlot">APOTCPlot</a>, <a href="#topic+AdjustAPOTC">AdjustAPOTC</a>, <a href="#topic+getApotcDataIds">getApotcDataIds</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("combined_pbmc")

# this is the recommended approach to use a custom run_id with default params
combined_pbmc &lt;- RunAPOTC(combined_pbmc, run_id = "default", verbose = FALSE)

# here's a seperate run with some filters to the meta data, where
# `orig.ident` is a custom column in the example data. Notice that it is not
# a `RunAPOTC` parameter but a user keyword argument
combined_pbmc &lt;- RunAPOTC(
    combined_pbmc, run_id = "sample17", orig.ident = c("P17B", "P17L"),
    verbose = FALSE
)

# the exact same thing can be achieved with the `extra_filter` parameter
combined_pbmc &lt;- RunAPOTC(
    combined_pbmc,
    run_id = "sample17",
    extra_filter = "substr(orig.ident, 2, 3) == '17'",
    override = TRUE,
    verbose = FALSE
)

</code></pre>

<hr>
<h2 id='showCloneHighlight'>Highlight specific clones on an APackOfTheClones ggplot</h2><span id='topic+showCloneHighlight'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>This is an analogue for <code>scRepertoire::highlightClones</code> that can highlight
certain clonotypes on an APackOfTheClones clonal expansion plot. For most
combinations of the arguments, there will be a ggplot fill legend on the
right side that correspond to each (existing) clonotype.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>showCloneHighlight(
  apotc_ggplot,
  clonotype,
  color_each = TRUE,
  default_color = "#808080",
  scale_bg = 1,
  fill_legend = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="showCloneHighlight_+3A_apotc_ggplot">apotc_ggplot</code></td>
<td>
<p>A ggplot object that is the output of <a href="#topic+APOTCPlot">APOTCPlot</a> or
<a href="#topic+vizAPOTC">vizAPOTC</a> of an APackOfTheClones plot to be highlighted on.</p>
</td></tr>
<tr><td><code id="showCloneHighlight_+3A_clonotype">clonotype</code></td>
<td>
<p>character vector of the sequence(s) to highlight. Note
that it must be of the clonecall of the code that created the plot. A
warning will be shown if any of the sequences are not present.</p>
</td></tr>
<tr><td><code id="showCloneHighlight_+3A_color_each">color_each</code></td>
<td>
<p>Either a logical of length 1, or a character(s). It is
<code>TRUE</code> by default, which assigns a unique default ggplot color to each
highlighted clone. If <code>FALSE</code>, each highlighted clone will retain its
current color and no legend based on color is shown. A possible application
here is to simply gauge the distribution of any shared clone.
It can also indicate the uniform color of each highlighted clone: if it is
a character of length 1, all highlighted clones will be of that color.
Else it must be a character vector of the same length as
<code>clonotype</code>, with each color corresponding to the clone. Here is a suitable
place to use any palette function from the many other CRAN palette packages
such as <code>"viridis"</code> or <code>"RColorBrewer"</code>. Note that currently, the user must
ensure <code>clonotype</code> contains only unique characters.</p>
</td></tr>
<tr><td><code id="showCloneHighlight_+3A_default_color">default_color</code></td>
<td>
<p>A character of length 1 or <code>NULL</code> indicating the color
of non-highlighted clones. If <code>NULL</code>, all un-highlighted sequences will
retain their original color in <code>sc.data</code>. Else, if it is a character, it
should be a valid color that all un-highlighted clones are. Defaults to the
hexcode for gray.</p>
</td></tr>
<tr><td><code id="showCloneHighlight_+3A_scale_bg">scale_bg</code></td>
<td>
<p>A positive numeric. Scales the brightness value of each color
of the non-highlighted clones by itself as a scaling factor. Defaults to 1
which will not alter the current brightness. Note that if
<code>color_each = FALSE</code> and <code>default_color = NULL</code>, this is equivalent to not
highlighting any clones - in this case, it may be useful to alter <code>scale_bg</code>
slightly so that the non-highlighted clones are darkened/brightened.</p>
</td></tr>
<tr><td><code id="showCloneHighlight_+3A_fill_legend">fill_legend</code></td>
<td>
<p>logical indicating whether a ggplot legend of the &quot;fill&quot;
of each clonotype should be displayed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Under the hood, this function simply mutates the plotting dataframe under
<code style="white-space: pre;">&#8288;$data&#8288;</code> in the ggplot object, and operates on a column named <code>color</code>.
</p>
<p>Note that if <code>color_each = FALSE</code> and <code>default_color = NULL</code>, this is
equivalent to simply not highlighting anything and a warning will be shown.
</p>


<h3>Value</h3>

<p>A ggplot object with the data modified to the highlighted colors
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(magrittr)
data("combined_pbmc")

# piping the plot can be nice to read syntatically -
# By default, assigns unique colors to highlights and everything else is gray
vizAPOTC(combined_pbmc, clonecall = "aa", verbose = FALSE) %&gt;%
    showCloneHighlight("CASLSGSARQLTF_CASSSTVAGEQYF")

# one useful application is to highlight shared clones - beware that the
# clonotype sequences may get extremely long in the legend
shared_aa_clones &lt;- names(getSharedClones(combined_pbmc, clonecall = "aa"))
vizAPOTC(combined_pbmc, clonecall = "aa", verbose = FALSE) %&gt;%
    showCloneHighlight(shared_aa_clones)

</code></pre>

<hr>
<h2 id='vizAPOTC'>Directly visualize clonal expansion of a combined seurat object</h2><span id='topic+vizAPOTC'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>This function combines the functionality of both <a href="#topic+RunAPOTC">RunAPOTC</a> and <a href="#topic+APOTCPlot">APOTCPlot</a>.
Given a Seurat object, it first runs the APackOfTheClones method (<a href="#topic+RunAPOTC">RunAPOTC</a>)
to compute clonal expansion information, and then generates a customizable
ggplot2 object of the clonal expansion plot with a circle size legend
(<a href="#topic+APOTCPlot">APOTCPlot</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vizAPOTC(
  seurat_obj,
  reduction_base = "umap",
  clonecall = "strict",
  ...,
  extra_filter = NULL,
  alt_ident = NULL,
  clone_scale_factor = "auto",
  rad_scale_factor = 0.95,
  order_clones = TRUE,
  try_place = FALSE,
  repulse = TRUE,
  repulsion_threshold = 1,
  repulsion_strength = 1,
  max_repulsion_iter = 20L,
  show_shared = NULL,
  only_link = NULL,
  clone_link_width = "auto",
  clone_link_color = "black",
  clone_link_alpha = 0.5,
  res = 360L,
  linetype = "blank",
  use_default_theme = TRUE,
  retain_axis_scales = FALSE,
  alpha = 1,
  show_labels = FALSE,
  label_size = 5,
  add_size_legend = TRUE,
  legend_sizes = "auto",
  legend_position = "auto",
  legend_buffer = 0.2,
  legend_color = "#808080",
  legend_spacing = "auto",
  legend_label = "Clone sizes",
  legend_text_size = 5,
  add_legend_background = TRUE,
  add_legend_centerspace = 0,
  detail = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vizAPOTC_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>A seurat object that has been integrated with clonotype
data with <code>scRepertoire::combineExpression</code>.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_reduction_base">reduction_base</code></td>
<td>
<p>character. The seurat reduction to base the clonal
expansion plotting on. Defaults to <code>'umap'</code> but can be any reduction present
within the reductions slot of the input seurat object, including custom ones.
If ''pca'&ldquo;, the cluster coordinates will be based on PC1 and PC2.
However, generally APackOfTheClones is used for displaying UMAP and
occasionally t-SNE versions to intuitively highlight clonal expansion.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_clonecall">clonecall</code></td>
<td>
<p>character. The column name in the seurat object metadata to
use. See <code>scRepertoire</code> documentation for more information about this
parameter that is central to both packages.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_...">...</code></td>
<td>
<p>additional &quot;subsetting&quot; keyword arguments indicating the rows
corresponding to elements in the seurat object metadata that should be
filtered by. E.g., <code>seurat_clusters = c(1, 9, 10)</code> will filter the cells to
those in the <code>seurat_clusters</code> column with any of the values 1, 9, and 10.
Unfortunately, column names in the seurat object metadata cannot
conflict with the keyword arguments. <em><strong>MAJOR NOTE</strong></em> if any subsetting
keyword arguments are a <em>prefix</em> of any preceding argument names (e.g. a
column named <code>reduction</code> is a prefix of the <code>reduction_base</code> argument)
R will interpret it as the same argument unless <em>both</em> arguments
are named. Additionally, this means any subsequent arguments <em>must</em> be named.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_extra_filter">extra_filter</code></td>
<td>
<p>character. An additional string that should be formatted
<em>exactly</em> like a statement one would pass into <a href="dplyr.html#topic+filter">dplyr::filter</a> that does
<em>additional</em> filtering to cells in the seurat object - on top of the other
keyword arguments - based on the metadata. This means that it will be
logically AND'ed with any keyword argument filters. This is a more flexible
alternative / addition to the filtering keyword arguments. For example, if
one wanted to filter by the length of the amino acid sequence of TCRs, one
could pass in something like <code>extra_filter = "nchar(CTaa) - 1 &gt; 10"</code>. When
involving characters, ensure to enclose with single quotes.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_alt_ident">alt_ident</code></td>
<td>
<p>character. By default, cluster identity is assumed to be
whatever is in <code>Idents(seurat_obj)</code>, and clones will be grouped by the active
ident. However, <code>alt_ident</code> could be set as the name of some column in the
meta data of the seurat object to be grouped by. This column is meant to have
been a product of <code>Seurat::StashIdent</code> or manually added.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_clone_scale_factor">clone_scale_factor</code></td>
<td>
<p>Dictates how much to scale each circle(between 0,1)
radius when converting from clonotype counts into circles that represent
individual clonotypes. The argument defaults to the character <code>"auto"</code>, and
if so, the most visually pleasing factor will be estimated.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_rad_scale_factor">rad_scale_factor</code></td>
<td>
<p>numeric between 0 and 1. This value decreases the
radius of the smallest clones by this scale factor. And the absolute value
of this decrease will be applied to all packed circles, effectively shrinking
all circles on the spot, and introduce more constant spacing in between.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_order_clones">order_clones</code></td>
<td>
<p>logical. Decides if the largest clone circles should be
near cluster centroids. This is highly recommended to be set to TRUE for
increased intuitiveness of the visualization, as resulting plots tend to
give an improved impression of the proportion of expanded clones. If
<code style="white-space: pre;">&#8288;FALSE,&#8288;</code> will randomly scramble the positions of each circle. For the sake
of being replicable, a random seed is recommended to be set with <a href="base.html#topic+set.seed">set.seed</a>.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_try_place">try_place</code></td>
<td>
<p>If <code>TRUE</code>, always minimizes distance from a newly placed
circle to the origin in the circle packing algorithm.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_repulse">repulse</code></td>
<td>
<p>If <code>TRUE</code>, will attempt to push overlapping clusters away from
each other.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_repulsion_threshold">repulsion_threshold</code></td>
<td>
<p>numeric. The radius that clonal circle clusters
overlap is acceptable when repulsing.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_repulsion_strength">repulsion_strength</code></td>
<td>
<p>numeric. The smaller the value the less the
clusters repulse each other per iteration, and vice versa.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_max_repulsion_iter">max_repulsion_iter</code></td>
<td>
<p>integer. The number of repulsion iterations.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_show_shared">show_shared</code></td>
<td>
<p>The output of <a href="#topic+getSharedClones">getSharedClones</a> can be inputted here,
and the resulting plot will overlay lines between clone circles if that
clonotype is common between clusters. Note that the input <em><strong>must</strong></em> be
generated from data in the correct <code>APackOfTheClones</code> run, and the behavior
is undefined otherwise and will likely error. The next 4 arguments allow for
aesthetic customization of these line links.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_only_link">only_link</code></td>
<td>
<p>Optional integer indicating to only display clone
links originating from this cluster if showing shared clones.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_clone_link_width">clone_link_width</code></td>
<td>
<p>numeric. The width of the lines that connect shared
clones. Defaults to <code>"auto"</code> which will estimate a reasonable value depending
on circle sizes.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_clone_link_color">clone_link_color</code></td>
<td>
<p>character. The color of the lines that connect shared
clones. Defaults to <code>"blend"</code> which will use the average colors of the two
connected clones. Else, any hex color or valid color string input will work,
and the corresponding color will be applied on all links.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_clone_link_alpha">clone_link_alpha</code></td>
<td>
<p>numeric. The alpha of the lines that connect shared
clones.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_res">res</code></td>
<td>
<p>The number of points on the generated path per full circle. From
plot viewers, if circles seem slightly too pixelated, it is recommended to
first try to export the plot as an <code>.svg</code> before increasing <code>res</code> due to
increased plotting times from <a href="ggforce.html#topic+geom_circle">ggforce::geom_circle</a>.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_linetype">linetype</code></td>
<td>
<p>The type of outline each circle should have. defaults to
<code style="white-space: pre;">&#8288;"blank&#8288;</code> meaning no outline. More information is in the function
documentation of <code>ggforce::geom_circle</code>.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_use_default_theme">use_default_theme</code></td>
<td>
<p>logical that defaults to <code>TRUE</code>. If <code>TRUE</code>,
the resulting plot will have the same theme as the seurat reference reduction
plot. Else, the plot will simply have a blank background.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_retain_axis_scales">retain_axis_scales</code></td>
<td>
<p>If <code>TRUE</code>, approximately maintains the axis scales
of the original reduction plot. However, it will only attempt to extend the
axes and never shorten. Users are recommended to set this to <code>TRUE</code>
especially if working with subsetted versions of the clonal data to better
preserve the geometric relation to the original dimensional reduction.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_alpha">alpha</code></td>
<td>
<p>numeric. The alpha of the circles in (0, 1]. Defaults to 1.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_show_labels">show_labels</code></td>
<td>
<p>If <code>TRUE</code>, will label each circle cluster at the centroid,
defaulting to &quot;C0, C1, ...&quot;.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_label_size">label_size</code></td>
<td>
<p>The text size of labels if shown. Defaults to 5.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_add_size_legend">add_size_legend</code></td>
<td>
<p>If <code>TRUE</code>, adds a legend to the plot visualizing the
relative sizes of clones. Note that it is simply an overlay and not a real
ggplot2 legend.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_legend_sizes">legend_sizes</code></td>
<td>
<p>numeric vector. Indicates the circle sizes to be
displayed on the legend, and will always be sorted from smallest to greatest.
Defaults to <code>"auto"</code> which estimate a reasonable range of sizes to display.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_legend_position">legend_position</code></td>
<td>
<p>character or numeric. Can be set to either
<code>"top_left"</code>, <code>"top_right"</code>, <code>"bottom_left"</code>, <code>"bottom_right"</code> and places the
legend roughly in the corresponding position. Otherwise, can be a numeric
vector of length 2 indicating the x and y position of the <em>topmost (smallest)
circle</em> of the legend.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_legend_buffer">legend_buffer</code></td>
<td>
<p>numeric. Indicates how much to &quot;push&quot; the legend towards
the center of the plot from the selected corner. If negative, will push away</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_legend_color">legend_color</code></td>
<td>
<p>character. Indicates the hex color of the circles
displayed on the legend. Defaults to the hex code for a gray tone</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_legend_spacing">legend_spacing</code></td>
<td>
<p>numeric. Indicates the horizontal distance between each
stacked circle on the size legend. Defaults to <code>"auto"</code> which will use an
estimated value depending on plot size</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_legend_label">legend_label</code></td>
<td>
<p>character. The title of the legend, which defaults to
<code style="white-space: pre;">&#8288;"clone sizes&#8288;</code>.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>numeric. The text size of the letters and numbers on
the legend</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_add_legend_background">add_legend_background</code></td>
<td>
<p>logical. If <code>TRUE</code>, will add a border around the
legend and fill the background to be white, overlaying anything else.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_add_legend_centerspace">add_legend_centerspace</code></td>
<td>
<p>numeric. An additional amount of distance
changed between the circle sizes on the left side of the legend and the
numbers on the right. Useful to set to around 0.5 (or more / less) when there
are particularly large clone sizes that may cover the numbers.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_detail">detail</code></td>
<td>
<p>logical. If <code>FALSE</code>, will only plot entire clusters as one
large circle, which may be useful in cases where there are a high number
of clones resulting in a large number of circles on the resulting ggplot,
which has increased plotting times, and certain aspects of the plot needs
to be finely adjusted with <a href="#topic+AdjustAPOTC">AdjustAPOTC</a> or simply inspected. This should
not be set to <code>FALSE</code> for the actual clonal expansion plot.</p>
</td></tr>
<tr><td><code id="vizAPOTC_+3A_verbose">verbose</code></td>
<td>
<p>logical. Decides if visual cues are displayed to the R console
of the progress.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that the subsetting arguments <code>...</code> and <code>extra_filter</code> are only a
quick convenience to subset based on metadata, and the <code>subset</code> S3 method
defined in <code>Seurat</code> is much more mature are has more features. Additionally,
users need to work with data subsets are recommended to and likely already
are working with seurat objects subsetted/split with <code>Seurat::SplitObject</code>.
</p>


<h3>Value</h3>

<p>A ggplot object of the APackOfTheClones clonal expansion plot of the
seurat object. There is an additional 10th element in the object named
<code>"APackOfTheClones"</code> used by other functions in this package and shouldn't
interfere with any other ggplot functionality. (As far as currently known)
</p>


<h3>Cluster labelling</h3>

<p>For the ident that was used to cluster the clones, labels for each cluster
are inferred and stored in the run so that they can be used by other
functions and optionally overlaid on the plot over clusters. If the levels
of the ident used is a naturally ordered integer sequence, then the labels
generated would be <code style="white-space: pre;">&#8288;"C1", "C2", "C3" ... &#8288;</code>, else they would be the actual
ident levels themselves.
</p>


<h3>See Also</h3>

<p><a href="#topic+AdjustAPOTC">AdjustAPOTC</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("combined_pbmc")

# plot with default parameters
vizAPOTC(combined_pbmc, verbose = FALSE)

# use arguments from RunAPOTC and APOTCPlot
vizAPOTC(
    combined_pbmc, try_place = TRUE, show_labels = TRUE, verbose = FALSE
)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
