<!DOCTYPE html><html lang="en"><head><title>Help for package APCalign</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {APCalign}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#APCalign'><p>Standardising Taxonomic Names in Australian Plants</p></a></li>
<li><a href='#align_taxa'><p>Align Australian plant scientific names to the APC or APNI</p></a></li>
<li><a href='#create_species_state_origin_matrix'><p>State level native and introduced origin status</p></a></li>
<li><a href='#create_taxonomic_update_lookup'><p>Create a table with the best-possible scientific name match for</p>
Australian plant names</a></li>
<li><a href='#default_version'><p>Get the default version for stable data</p></a></li>
<li><a href='#gbif_lite'><p>GBIF Australian Plant Data</p></a></li>
<li><a href='#get_apc_genus_family_lookup'><p>Lookup Family by Genus from APC</p></a></li>
<li><a href='#get_versions'><p>Which versions of taxonomic resources are available?</p></a></li>
<li><a href='#load_taxonomic_resources'><p>Load taxonomic reference lists, APC &amp; APNI</p></a></li>
<li><a href='#native_anywhere_in_australia'><p>Native anywhere in Australia</p></a></li>
<li><a href='#reexports'><p>Objects exported from other packages</p></a></li>
<li><a href='#standardise_names'><p>Standardise taxon names</p></a></li>
<li><a href='#standardise_taxon_rank'><p>Standardise taxon ranks</p></a></li>
<li><a href='#state_diversity_counts'><p>State- and territory-level diversity</p></a></li>
<li><a href='#strip_names'><p>Strip taxon names</p></a></li>
<li><a href='#strip_names_extra'><p>Strip taxon names, extra</p></a></li>
<li><a href='#update_taxonomy'><p>Update to currently accepted APC name and add APC/APNI name metadata</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Resolving Plant Taxon Names Using the Australian Plant Census</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.3</td>
</tr>
<tr>
<td>Description:</td>
<td>The process of resolving taxon names is necessary when working with biodiversity data. 'APCalign' uses the Australian Plant Census (APC) and the Australian Plant Name Index (APNI) to align and update plant taxon names to current, accepted standards. 'APCalign' also supplies information about the established status of plant taxa across different states/territories.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0),</td>
</tr>
<tr>
<td>Imports:</td>
<td>readr, purrr, dplyr, stringr, stringi, stringdist, crayon,
httr, jsonlite, curl, arrow, rlang</td>
</tr>
<tr>
<td>Suggests:</td>
<td>janitor, tidyr, covr, knitr, rmarkdown, kableExtra, here,
testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://traitecoevo.github.io/APCalign/">https://traitecoevo.github.io/APCalign/</a>,
<a href="https://github.com/traitecoevo/APCalign">https://github.com/traitecoevo/APCalign</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/traitecoevo/APCalign/issues">https://github.com/traitecoevo/APCalign/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-11 00:01:38 UTC; dfalster</td>
</tr>
<tr>
<td>Author:</td>
<td>Daniel Falster <a href="https://orcid.org/0000-0002-9814-092X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre, cph],
  Elizabeth Wenk <a href="https://orcid.org/0000-0001-5640-5910"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, ctb],
  Will Cornwell <a href="https://orcid.org/0000-0003-4080-4073"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, ctb],
  Fonti Kar <a href="https://orcid.org/0000-0002-2760-3974"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    ctb],
  Carl Boettiger <a href="https://orcid.org/0000-0002-1642-628X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Daniel Falster &lt;daniel.falster@unsw.edu.au&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-11 13:40:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='APCalign'>Standardising Taxonomic Names in Australian Plants</h2><span id='topic+APCalign-package'></span><span id='topic+APCalign'></span>

<h3>Description</h3>

<p>The process of standardising taxon names is necessary when working with
biodiversity data. 'APCalign' uses the Australian Plant Name Index (APNI)
and the Australian Plant Census (APC) to align and update plant taxon names
to current, accepted standards. 'APCalign' can also supply information about
the established status of plant taxa across different states/territories.
</p>


<h3>Functions</h3>

<p><strong>Standarise taxon names</strong>
</p>

<ul>
<li> <p><a href="#topic+load_taxonomic_resources">load_taxonomic_resources</a>
</p>
</li>
<li> <p><a href="#topic+create_taxonomic_update_lookup">create_taxonomic_update_lookup</a>
</p>
</li>
<li> <p><a href="#topic+align_taxa">align_taxa</a>
</p>
</li>
<li> <p><a href="#topic+update_taxonomy">update_taxonomy</a>
</p>
</li></ul>

<p><strong>Established status by region</strong>
</p>

<ul>
<li> <p><a href="#topic+state_diversity_counts">state_diversity_counts</a>
</p>
</li>
<li> <p><a href="#topic+create_species_state_origin_matrix">create_species_state_origin_matrix</a>
</p>
</li>
<li> <p><a href="#topic+native_anywhere_in_australia">native_anywhere_in_australia</a>
</p>
</li></ul>



<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Daniel Falster <a href="mailto:daniel.falster@unsw.edu.au">daniel.falster@unsw.edu.au</a> (<a href="https://orcid.org/0000-0002-9814-092X">ORCID</a>) [copyright holder]
</p>
<p>Authors:
</p>

<ul>
<li><p> Elizabeth Wenk <a href="mailto:e.wenk@unsw.edu.au">e.wenk@unsw.edu.au</a> (<a href="https://orcid.org/0000-0001-5640-5910">ORCID</a>) [contributor]
</p>
</li>
<li><p> Will Cornwell <a href="mailto:w.cornwell@unsw.edu.au">w.cornwell@unsw.edu.au</a> (<a href="https://orcid.org/0000-0003-4080-4073">ORCID</a>) [contributor]
</p>
</li>
<li><p> Fonti Kar <a href="mailto:f.kar@unsw.edu.au">f.kar@unsw.edu.au</a> (<a href="https://orcid.org/0000-0002-2760-3974">ORCID</a>) [contributor]
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Carl Boettiger <a href="mailto:cboettig@gmail.com">cboettig@gmail.com</a> (<a href="https://orcid.org/0000-0002-1642-628X">ORCID</a>) [contributor]
</p>
</li></ul>



<h3>References</h3>

<p>If you have any questions, comments or suggestions, please
submit an issue at our
<a href="https://github.com/traitecoevo/APCalign/issues">GitHub repository</a>
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://traitecoevo.github.io/APCalign/">https://traitecoevo.github.io/APCalign/</a>
</p>
</li>
<li> <p><a href="https://github.com/traitecoevo/APCalign">https://github.com/traitecoevo/APCalign</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/traitecoevo/APCalign/issues">https://github.com/traitecoevo/APCalign/issues</a>
</p>
</li></ul>


<hr>
<h2 id='align_taxa'>Align Australian plant scientific names to the APC or APNI</h2><span id='topic+align_taxa'></span>

<h3>Description</h3>

<p>For a list of Australian plant names, find taxonomic or scientific name
alignments to the APC or APNI through standardizing formatting and fixing
spelling errors.
</p>
<p>Usage case: Users will run this function if they wish to see the details
of the matching algorithms, the many output columns that the matching
function compares to as it seeks the best alignment. They may also select
this function if they want to adjust the “fuzziness” level for fuzzy
matches, options not allowed in create_taxonomic_update_lookup. This
function is the first half of create_taxonomic_update_lookup.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>align_taxa(
  original_name,
  output = NULL,
  full = FALSE,
  resources = load_taxonomic_resources(),
  quiet = FALSE,
  fuzzy_abs_dist = 3,
  fuzzy_rel_dist = 0.2,
  fuzzy_matches = TRUE,
  imprecise_fuzzy_matches = FALSE,
  APNI_matches = TRUE,
  identifier = NA_character_
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="align_taxa_+3A_original_name">original_name</code></td>
<td>
<p>A list of names to query for taxonomic alignments.</p>
</td></tr>
<tr><td><code id="align_taxa_+3A_output">output</code></td>
<td>
<p>(optional) The name of the file to save the results to.</p>
</td></tr>
<tr><td><code id="align_taxa_+3A_full">full</code></td>
<td>
<p>Parameter to determine how many columns are output</p>
</td></tr>
<tr><td><code id="align_taxa_+3A_resources">resources</code></td>
<td>
<p>the taxonomic resources used to align the taxa names.
Loading this can be slow, so call <code><a href="#topic+load_taxonomic_resources">load_taxonomic_resources</a></code>
separately to greatly speed this function up and pass the resources in.</p>
</td></tr>
<tr><td><code id="align_taxa_+3A_quiet">quiet</code></td>
<td>
<p>Logical to indicate whether to display messages while
aligning taxa.</p>
</td></tr>
<tr><td><code id="align_taxa_+3A_fuzzy_abs_dist">fuzzy_abs_dist</code></td>
<td>
<p>The number of characters allowed to be different for a
fuzzy match.</p>
</td></tr>
<tr><td><code id="align_taxa_+3A_fuzzy_rel_dist">fuzzy_rel_dist</code></td>
<td>
<p>The proportion of characters allowed to be different
for a fuzzy match.</p>
</td></tr>
<tr><td><code id="align_taxa_+3A_fuzzy_matches">fuzzy_matches</code></td>
<td>
<p>Fuzzy matches are turned on as a default.
The relative and absolute distances allowed for fuzzy matches to species and
infraspecific taxon names are defined by the parameters <code>fuzzy_abs_dist</code>
and <code>fuzzy_rel_dist</code></p>
</td></tr>
<tr><td><code id="align_taxa_+3A_imprecise_fuzzy_matches">imprecise_fuzzy_matches</code></td>
<td>
<p>Imprecise fuzzy matches uses the
fuzzy matching function with lenient levels set (absolute distance of
5 characters; relative distance = 0.25).
It offers a way to get a wider range of possible names, possibly
corresponding to very distant spelling mistakes.
This is FALSE as default and all outputs should be checked as it often
makes erroneous matches.</p>
</td></tr>
<tr><td><code id="align_taxa_+3A_apni_matches">APNI_matches</code></td>
<td>
<p>Name matches to the APNI (Australian Plant Names Index)
are turned on as a default.</p>
</td></tr>
<tr><td><code id="align_taxa_+3A_identifier">identifier</code></td>
<td>
<p>A dataset, location or other identifier,
which defaults to NA.</p>
</td></tr>
</table>


<h3>Details</h3>


<ul>
<li><p> This function finds taxonomic alignments in APC or scientific name
alignments in APNI.
</p>
</li>
<li><p> It uses the internal function <code>match_taxa</code> to attempt to match input
strings to taxon names in the APC/APNI.
</p>
</li>
<li><p> It sequentially searches for matches against more than 20 different string
patterns, prioritising exact matches (to accepted names as well as
synonyms, orthographic variants) over fuzzy matches.
</p>
</li>
<li><p> It prioritises matches to taxa in the APC over names in the APNI.
</p>
</li>
<li><p> It identifies string patterns in input names that suggest a name can only
be aligned to a genus (hybrids that are not in the APC/APNI; graded species;
taxa not identified to species), and indicates these names only have a
genus-rank match.
</p>
</li></ul>

<p>Notes:
</p>

<ul>
<li><p> If you will be running the function APCalign::create_taxonomic_update_lookup
many times, it is best to load the taxonomic resources separately using
resources &lt;- load_taxonomic_resources(), then add the argument
resources = resources
</p>
</li>
<li><p> The name Banksia cerrata does not align as the fuzzy matching algorithm
does not allow the first letter of the genus and species epithet to change.
</p>
</li>
<li><p> With this function you have the option of changing the fuzzy matching
parameters. The defaults, with fuzzy matches only allowing changes of 3
(or fewer) characters AND 20% (or less) of characters has been carefully
calibrated to catch just about all typos, but very, very rarely mis-align
a name. If you wish to introduce less conservative fuzzy matching it is
recommended you manually check the aligned names.
</p>
</li>
<li><p> It is recommended that you begin with imprecise_fuzzy_matches = FALSE (the
default), as quite a few of the less precise fuzzy matches are likely to be
erroneous. This argument should be turned on only if you plan to check all
alignments manually.
</p>
</li>
<li><p> The argument identifier allows you to add a fix text string to all genus-
and family- level names, such as identifier = &quot;Royal NP&quot; would return &quot;Acacia
sp. [Royal NP]&quot;.
</p>
</li></ul>



<h3>Value</h3>

<p>A tibble with columns that include original_name, aligned_name,
taxonomic_dataset, taxon_rank, aligned_reason, alignment_code.
</p>

<ul>
<li><p> original_name: the original plant name input.
</p>
</li>
<li><p> aligned_name: the original plant name after the function standardise_names
has standardised the syntax of infraspecific taxon designations.
</p>
</li>
<li><p> taxonomic_dataset: the source of the aligned names (APC or APNI).
</p>
</li>
<li><p> taxon_rank: the taxonomic rank of the aligned name.
</p>
</li>
<li><p> aligned_reason: the explanation of a specific taxon name alignment
(from an original name to an aligned name).
</p>
</li>
<li><p> alignment_code: a code that accompanies the aligned_reason, indicating the
relative sequence of the match during the alignment process.
</p>
</li>
<li><p> cleaned_name: original name with punctuation and infraspecific taxon
designation terms standardised by the function standardise_names;
streamlines exact matches.
</p>
</li>
<li><p> stripped_name: cleaned name with punctuation and infraspecific taxon
designation terms removed by the function strip_names;
improves fuzzy matches.
</p>
</li>
<li><p> stripped_name2: cleaned name with punctuation, infraspecific taxon
designation terms, and other filler words removed by
the function <code>strip_names_extra</code>;
required for matches to <code style="white-space: pre;">&#8288;first two word&#8288;</code> and <code style="white-space: pre;">&#8288;first three words&#8288;</code>.
</p>
</li>
<li><p> trinomial: the first three words in <code>stripped_name2</code>, required for matches
that ignore all other text in the original_name;
improves phrase name matches.
</p>
</li>
<li><p> binomial: the first two words in <code>stripped_name2</code>, required for matches
that ignore all other text in the original_name;
improves phrase name matches.
</p>
</li>
<li><p> genus: the first two words in <code>cleaned_name</code>;
required for genus-rank matches and reprocessing of genus-rank names.
</p>
</li>
<li><p> fuzzy_match_genus: fuzzy match of genus column to best match among
APC-accepted names;
required for fuzzy matches of genus-rank names.
</p>
</li>
<li><p> fuzzy_match_genus_synonym: fuzzy match of genus column to best match among
APC-synonymous names, only considering different matches to those documented
under APC-accepted genera; required for fuzzy matches of genus-rank names.
</p>
</li>
<li><p> fuzzy_match_genus_APNI: fuzzy match of genus column to best match among
APNI names, only considering different matches to those documented under
APC-accepted and APC-known genera; required for fuzzy matches of
genus-rank names.
</p>
</li>
<li><p> fuzzy_match_family: fuzzy match of genus column to best match among
APC-accepted family names; required for fuzzy matches of family-rank names.
</p>
</li>
<li><p> fuzzy_match_family_synonym: fuzzy match of genus column to best match
among APC-synonymous family names; required for fuzzy matches of
family-rank names.
</p>
</li>
<li><p> fuzzy_match_cleaned_APC: fuzzy match of stripped_name to APC-accepted
names; created for yet-to-be-aligned names at the match step 05a
in the function <code>match_taxa</code>.
</p>
</li>
<li><p> fuzzy_match_cleaned_APC_synonym: fuzzy match of stripped_name to
APC-synonymous names; created for yet-to-be-aligned names at the
match step 05b in the function <code>match_taxa</code>.
</p>
</li>
<li><p> fuzzy_match_cleaned_APC_imprecise: imprecise fuzzy match of stripped_name
to APC-accepted names; created for yet-to-be-aligned names at the
match step 07a in the function <code>match_taxa</code>.
</p>
</li>
<li><p> fuzzy_match_cleaned_APC_synonym_imprecise: imprecise fuzzy match of
stripped_name to APC-accepted names; created for yet-to-be-aligned names
at the match step 07b in the function <code>match_taxa</code>.
</p>
</li>
<li><p> fuzzy_match_binomial: fuzzy match of binomial column to best match among
APC-accepted names; created for yet-to-be-aligned names at
match step 10c in the function <code>match_taxa</code>.
</p>
</li>
<li><p> fuzzy_match_binomial_APC_synonym: fuzzy match of binomial column to best
match among APC-synonymous names; created for yet-to-be-aligned names at
match step 10d in the function <code>match_taxa</code>.
</p>
</li>
<li><p> fuzzy_match_trinomial: fuzzy match of trinomial column to best match
among APC-accepted names; created for yet-to-be-aligned names at
match step 09c in the function <code>match_taxa</code>.
</p>
</li>
<li><p> fuzzy_match_trinomial_synonym: fuzzy match of trinomial column to best
match among APC-synonymous names; created for yet-to-be-aligned names at
match step 09d in the function <code>match_taxa</code>.
</p>
</li>
<li><p> fuzzy_match_cleaned_APNI: fuzzy match of stripped_name to APNI names;
created for yet-to-be-aligned names at the match step 11a in the
function <code>match_taxa</code>.
</p>
</li>
<li><p> fuzzy_match_cleaned_APNI_imprecise: imprecise fuzzy match of
stripped_name to APNI names; created for yet-to-be-aligned names
at the match step 11b in the function <code>match_taxa</code>.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+load_taxonomic_resources">load_taxonomic_resources</a></code>
</p>
<p>Other taxonomic alignment functions: 
<code><a href="#topic+create_taxonomic_update_lookup">create_taxonomic_update_lookup</a>()</code>,
<code><a href="#topic+update_taxonomy">update_taxonomy</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

resources &lt;- load_taxonomic_resources()

# example 1
align_taxa(c("Poa annua", "Abies alba"), resources=resources)

# example 2
input &lt;- c("Banksia serrata", "Banksia serrate", "Banksia cerrata", 
"Banksia serrrrata", "Dryandra sp.", "Banksia big red flowers")

aligned_taxa &lt;-
  APCalign::align_taxa(
    original_name = input,
    identifier = "APCalign test",
    full = TRUE,
    resources=resources
  ) 
  



</code></pre>

<hr>
<h2 id='create_species_state_origin_matrix'>State level native and introduced origin status</h2><span id='topic+create_species_state_origin_matrix'></span>

<h3>Description</h3>

<p>This function uses the taxon distribution data from the APC to determine
state level native and introduced origin status.
</p>
<p>This function processes the geographic data available in the APC and
returns state level native, introduced and more complicated origins status for all taxa.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_species_state_origin_matrix(resources = load_taxonomic_resources())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_species_state_origin_matrix_+3A_resources">resources</code></td>
<td>
<p>the taxonomic resources required to make the summary statistics.
Loading this can be slow, so call load_taxonomic_resources separately to greatly
speed this function up and pass the resources in.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with columns representing each state and rows representing each
species. The values in each cell represent the origin of the species in that state.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+load_taxonomic_resources">load_taxonomic_resources</a></code>
</p>
<p>Other diversity methods: 
<code><a href="#topic+native_anywhere_in_australia">native_anywhere_in_australia</a>()</code>,
<code><a href="#topic+state_diversity_counts">state_diversity_counts</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>create_species_state_origin_matrix()



</code></pre>

<hr>
<h2 id='create_taxonomic_update_lookup'>Create a table with the best-possible scientific name match for
Australian plant names</h2><span id='topic+create_taxonomic_update_lookup'></span>

<h3>Description</h3>

<p>This function takes a list of Australian plant names that need to be
reconciled with current taxonomy and generates a lookup table of the
best-possible scientific name match for each input name.
</p>
<p>Usage case: This is APCalign’s core function, merging together the alignment
and updating of taxonomy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_taxonomic_update_lookup(
  taxa,
  stable_or_current_data = "stable",
  version = default_version(),
  taxonomic_splits = "most_likely_species",
  full = FALSE,
  fuzzy_abs_dist = 3,
  fuzzy_rel_dist = 0.2,
  fuzzy_matches = TRUE,
  APNI_matches = TRUE,
  imprecise_fuzzy_matches = FALSE,
  identifier = NA_character_,
  resources = load_taxonomic_resources(quiet = quiet),
  quiet = FALSE,
  output = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_taxonomic_update_lookup_+3A_taxa">taxa</code></td>
<td>
<p>A list of Australian plant species that needs to be reconciled
with current taxonomy.</p>
</td></tr>
<tr><td><code id="create_taxonomic_update_lookup_+3A_stable_or_current_data">stable_or_current_data</code></td>
<td>
<p>either &quot;stable&quot; for a consistent version,
or &quot;current&quot; for the leading edge version.</p>
</td></tr>
<tr><td><code id="create_taxonomic_update_lookup_+3A_version">version</code></td>
<td>
<p>The version number of the dataset to use.</p>
</td></tr>
<tr><td><code id="create_taxonomic_update_lookup_+3A_taxonomic_splits">taxonomic_splits</code></td>
<td>
<p>How to handle one_to_many taxonomic matches.
Default is &quot;return_all&quot;.  The other options are &quot;collapse_to_higher_taxon&quot;
and &quot;most_likely_species&quot;. most_likely_species defaults to the original_name
if that name is accepted by the APC; this will be right for certain species
subsets, but make errors in other cases, use with caution.</p>
</td></tr>
<tr><td><code id="create_taxonomic_update_lookup_+3A_full">full</code></td>
<td>
<p>logical for whether the full lookup table is returned or
just key columns</p>
</td></tr>
<tr><td><code id="create_taxonomic_update_lookup_+3A_fuzzy_abs_dist">fuzzy_abs_dist</code></td>
<td>
<p>The number of characters allowed to be different for
a fuzzy match.</p>
</td></tr>
<tr><td><code id="create_taxonomic_update_lookup_+3A_fuzzy_rel_dist">fuzzy_rel_dist</code></td>
<td>
<p>The proportion of characters allowed to be different
for a fuzzy match.</p>
</td></tr>
<tr><td><code id="create_taxonomic_update_lookup_+3A_fuzzy_matches">fuzzy_matches</code></td>
<td>
<p>Fuzzy matches are turned on as a default. The relative
and absolute distances allowed for fuzzy matches to species and
infraspecific taxon names are defined by the parameters <code>fuzzy_abs_dist</code>
and <code>fuzzy_rel_dist</code>.</p>
</td></tr>
<tr><td><code id="create_taxonomic_update_lookup_+3A_apni_matches">APNI_matches</code></td>
<td>
<p>Name matches to the APNI (Australian Plant Names Index)
are turned off as a default.</p>
</td></tr>
<tr><td><code id="create_taxonomic_update_lookup_+3A_imprecise_fuzzy_matches">imprecise_fuzzy_matches</code></td>
<td>
<p>Imprecise fuzzy matches uses the fuzzy
matching function with lenient levels set (absolute distance of
5 characters; relative distance = 0.25).
It offers a way to get a wider range of possible names, possibly
corresponding to very distant spelling mistakes.
This is FALSE as default and all outputs should be checked as it often
makes erroneous matches.</p>
</td></tr>
<tr><td><code id="create_taxonomic_update_lookup_+3A_identifier">identifier</code></td>
<td>
<p>A dataset, location or other identifier,
which defaults to NA.</p>
</td></tr>
<tr><td><code id="create_taxonomic_update_lookup_+3A_resources">resources</code></td>
<td>
<p>These are the taxonomic resources used for cleaning, this
will default to loading them from a local place on your computer. If this is
to be called repeatedly, it's much faster to load the resources using
<code><a href="#topic+load_taxonomic_resources">load_taxonomic_resources</a></code> separately and pass the data in.</p>
</td></tr>
<tr><td><code id="create_taxonomic_update_lookup_+3A_quiet">quiet</code></td>
<td>
<p>Logical to indicate whether to display messages while loading data and
aligning taxa.</p>
</td></tr>
<tr><td><code id="create_taxonomic_update_lookup_+3A_output">output</code></td>
<td>
<p>file path to save the output. If this file already exists,
this function will check if it's a subset of the species passed in and try
to add to this file. This can be useful for large and growing projects.</p>
</td></tr>
</table>


<h3>Details</h3>


<ul>
<li><p> It uses first the function <code>align_taxa</code>, then the function <code>update_taxonomy</code>
to achieve the output. The aligned name is plant name that has been aligned
to a taxon name in the APC or APNI by the align_taxa function.
</p>
</li></ul>

<p>Notes:
</p>

<ul>
<li><p> If you will be running the function APCalign::create_taxonomic_update_lookup
many times, it is best to load the taxonomic resources separately using
<code>resources &lt;- load_taxonomic_resources()</code>, then add the argument
resources = resources
</p>
</li>
<li><p> The name Banksia cerrata does not align as the fuzzy matching algorithm
does not allow the first letter of the genus and species epithet to change.
</p>
</li>
<li><p> The argument taxonomic_splits allows you to choose the outcome for updating
the names of taxa with ambiguous taxonomic histories; this applies to
scientific names that were once attached to a more broadly circumscribed
taxon concept, that was then split into several more narrowly circumscribed
taxon concepts, one of which retains the original name. There are three
options: most_likely_species returns the name that is retained, with
alternative names documented in square brackets; return_all adds additional
rows to the output, one for each possible taxon concept;
collapse_to_higher_taxon returns the genus with possible names in square
brackets.
</p>
</li>
<li><p> The argument identifier allows you to add a fix text string to all genus-
and family- level names, such as identifier = &quot;Royal NP&quot; would return
<code style="white-space: pre;">&#8288;Acacia sp. \[Royal NP]&#8288;</code>.
</p>
</li></ul>



<h3>Value</h3>

<p>A lookup table containing the accepted and suggested names for each
original name input, and additional taxonomic information such as taxon
rank, taxonomic status, taxon IDs and genera.
</p>

<ul>
<li><p> original_name: the original plant name.
</p>
</li>
<li><p> aligned_name: the input plant name that has been aligned to a taxon name in
the APC or APNI by the align_taxa function.
</p>
</li>
<li><p> accepted_name: the APC-accepted plant name, when available.
</p>
</li>
<li><p> suggested_name: the suggested plant name to use. Identical to the
accepted_name, when an accepted_name exists;
otherwise the the suggested_name is the aligned_name.
</p>
</li>
<li><p> genus: the genus of the accepted (or suggested) name;
only APC-accepted genus names are filled in.
</p>
</li>
<li><p> family: the family of the accepted (or suggested) name;
only APC-accepted family names are filled in.
</p>
</li>
<li><p> taxon_rank: the taxonomic rank of the suggested (and accepted) name.
</p>
</li>
<li><p> taxonomic_dataset: the source of the suggested (and accepted) names
(APC or APNI).
</p>
</li>
<li><p> taxonomic_status: the taxonomic status of the suggested (and accepted) name.
</p>
</li>
<li><p> taxonomic_status_aligned: the taxonomic status of the aligned name,
before any taxonomic updates have been applied.
</p>
</li>
<li><p> aligned_reason: the explanation of a specific taxon name alignment
(from an original name to an aligned name).
</p>
</li>
<li><p> update_reason: the explanation of a specific taxon name update
(from an aligned name to an accepted or suggested name).
</p>
</li>
<li><p> subclass: the subclass of the accepted name.
</p>
</li>
<li><p> taxon_distribution: the distribution of the accepted name;
only filled in if an APC accepted_name is available.
</p>
</li>
<li><p> scientific_name_authorship: the authorship information for the accepted
(or synonymous) name; available for both APC and APNI names.
</p>
</li>
<li><p> taxon_ID: the unique taxon concept identifier for the accepted_name;
only filled in if an APC accepted_name is available.
</p>
</li>
<li><p> taxon_ID_genus: an identifier for the genus;
only filled in if an APC-accepted genus name is available.
</p>
</li>
<li><p> scientific_name_ID: an identifier for the nomenclatural (not taxonomic)
details of a scientific name; available for both APC and APNI names.
</p>
</li>
<li><p> row_number: the row number of a specific original_name in the input.
</p>
</li>
<li><p> number_of_collapsed_taxa: when taxonomic_splits == &quot;collapse_to_higher_taxon&quot;,
the number of possible taxon names that have been collapsed.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+load_taxonomic_resources">load_taxonomic_resources</a></code>
</p>
<p>Other taxonomic alignment functions: 
<code><a href="#topic+align_taxa">align_taxa</a>()</code>,
<code><a href="#topic+update_taxonomy">update_taxonomy</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
resources &lt;- load_taxonomic_resources()

# example 1
create_taxonomic_update_lookup(c("Eucalyptus regnans",
                                 "Acacia melanoxylon",
                                 "Banksia integrifolia",
                                 "Not a species"),
                                 resources = resources)
                                 
# example 2
input &lt;- c("Banksia serrata", "Banksia serrate", "Banksia cerrata", 
"Banksea serrata", "Banksia serrrrata", "Dryandra")

create_taxonomic_update_lookup(
    taxa = input,
    identifier = "APCalign test",
    full = TRUE,
    resources = resources
  )

# example 3
taxon_list &lt;-
  readr::read_csv(
  system.file("extdata", "test_taxa.csv", package = "APCalign"),
  show_col_types = FALSE)

create_taxonomic_update_lookup(
    taxa = taxon_list$original_name,
    identifier = taxon_list$notes,
    full = TRUE,
    resources = resources
  )


</code></pre>

<hr>
<h2 id='default_version'>Get the default version for stable data</h2><span id='topic+default_version'></span>

<h3>Description</h3>

<p>This function returns the default version for stable data, which is used when no
version is specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>default_version()
</code></pre>


<h3>Value</h3>

<p>A character string representing the default version for stable data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>default_version()

</code></pre>

<hr>
<h2 id='gbif_lite'>GBIF Australian Plant Data</h2><span id='topic+gbif_lite'></span>

<h3>Description</h3>

<p>A subset of plant data from the Global Biodiversity Information Facility
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gbif_lite
</code></pre>


<h3>Format</h3>

<p><code>gbif_lite</code>
A tibble with 129 rows and 7 columns:
</p>

<dl>
<dt>species</dt><dd><p>The name of the first or species of scientificname</p>
</dd>
<dt>infraspecificepithet</dt><dd><p>	The name of the lowest or terminal infraspecific epithet of the scientificname</p>
</dd>
<dt>taxonrank</dt><dd><p>The taxonomic rank of the most specific name</p>
</dd>
<dt>decimalLongitude</dt><dd><p>Longitude in decimal degrees</p>
</dd>
<dt>decimalLatitude</dt><dd><p>Latitude in decimal degrees</p>
</dd>
<dt>scientificname</dt><dd><p>Scientific Name</p>
</dd>
<dt>verbatimscientificname</dt><dd><p>Scientific name as it appeared in original record</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://www.gbif.org/">https://www.gbif.org/</a>
</p>

<hr>
<h2 id='get_apc_genus_family_lookup'>Lookup Family by Genus from APC</h2><span id='topic+get_apc_genus_family_lookup'></span>

<h3>Description</h3>

<p>Retrieve the family name for a given genus using taxonomic data from the
Australian Plant Census (APC).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_apc_genus_family_lookup(genus, resources = load_taxonomic_resources())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_apc_genus_family_lookup_+3A_genus">genus</code></td>
<td>
<p>A character vector of genus names for which to retrieve the
corresponding family names.</p>
</td></tr>
<tr><td><code id="get_apc_genus_family_lookup_+3A_resources">resources</code></td>
<td>
<p>The taxonomic resources required to make the lookup.
Loading this can be slow, so call <code><a href="#topic+load_taxonomic_resources">load_taxonomic_resources</a></code>
separately to speed up this function and pass the resources in.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with two columns: &quot;genus&quot;, indicating the genus name,
and &quot;family&quot;, indicating the corresponding family name from the APC.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+load_taxonomic_resources">load_taxonomic_resources</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> get_apc_genus_family_lookup(genus = c("Acacia", "Eucalyptus"))
</code></pre>

<hr>
<h2 id='get_versions'>Which versions of taxonomic resources are available?</h2><span id='topic+get_versions'></span>

<h3>Description</h3>

<p>Which versions of taxonomic resources are available?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_versions()
</code></pre>


<h3>Value</h3>

<p>tibble of dates when APC/APNI resources were downloaded as a Github Release
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_versions()
</code></pre>

<hr>
<h2 id='load_taxonomic_resources'>Load taxonomic reference lists, APC &amp; APNI</h2><span id='topic+load_taxonomic_resources'></span>

<h3>Description</h3>

<p>This function loads two taxonomic datasets for Australia's vascular plants,
the APC and APNI, into the global environment. It creates several data frames
by filtering and selecting data from the loaded lists.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_taxonomic_resources(
  stable_or_current_data = "stable",
  version = default_version(),
  quiet = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="load_taxonomic_resources_+3A_stable_or_current_data">stable_or_current_data</code></td>
<td>
<p>Type of dataset to access.
The default is &quot;stable&quot;, which loads the dataset from a github archived file.
If set to &quot;current&quot;, the dataset will be loaded from a URL which is the
cutting edge version, but this may change at any time without notice.</p>
</td></tr>
<tr><td><code id="load_taxonomic_resources_+3A_version">version</code></td>
<td>
<p>The version number of the dataset to use.
Defaults to the default version.</p>
</td></tr>
<tr><td><code id="load_taxonomic_resources_+3A_quiet">quiet</code></td>
<td>
<p>A logical indicating whether to print status of loading to screen.
Defaults to FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>


<ul>
<li><p> It accesses taxonomic data from a dataset using the provided version number
or the default version.
</p>
</li>
<li><p> The output is several dataframes that include subsets of the APC/APNI based
on taxon rank and taxonomic status.
</p>
</li></ul>



<h3>Value</h3>

<p>The taxonomic resources data loaded into the global environment.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
load_taxonomic_resources(stable_or_current_data="stable", 
version="2024-10-11")

</code></pre>

<hr>
<h2 id='native_anywhere_in_australia'>Native anywhere in Australia</h2><span id='topic+native_anywhere_in_australia'></span>

<h3>Description</h3>

<p>This function checks which species from a list is thought to be native anywhere in
Australia according to the APC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>native_anywhere_in_australia(species, resources = load_taxonomic_resources())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="native_anywhere_in_australia_+3A_species">species</code></td>
<td>
<p>A character string typically representing the binomial for the species.</p>
</td></tr>
<tr><td><code id="native_anywhere_in_australia_+3A_resources">resources</code></td>
<td>
<p>An optional list of taxonomic resources to use for the lookup.
If not provided, the function will load default taxonomic resources using the
<code>load_taxonomic_resources()</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Important caveats:
</p>

<ul>
<li><p> This function will not detect within-Australia introductions,
e.g. if a species is from Western Australia and is invasive on the east coast.
</p>
</li>
<li><p> Very recent invasions are unlikely to be documented yet in APC.
</p>
</li>
<li><p> Ideally check spelling and taxonomy updates first via
<a href="#topic+create_taxonomic_update_lookup">create_taxonomic_update_lookup</a>.
</p>
</li>
<li><p> For the complete matrix of species by states that also represents
within-Australia invasions, use <a href="#topic+create_species_state_origin_matrix">create_species_state_origin_matrix</a>.
</p>
</li></ul>



<h3>Value</h3>

<p>A tibble with two columns: <code>species</code>, which is the same as the unique values of
the input <code>species</code>, and <code>native_anywhere_in_aus</code>, a vector indicating whether each
species is native anywhere in Australia, introduced by humans from elsewhere, or
unknown with respect to the APC resource.
</p>


<h3>See Also</h3>

<p>Other diversity methods: 
<code><a href="#topic+create_species_state_origin_matrix">create_species_state_origin_matrix</a>()</code>,
<code><a href="#topic+state_diversity_counts">state_diversity_counts</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>native_anywhere_in_australia(c("Eucalyptus globulus","Pinus radiata","Banksis notaspecies"))
</code></pre>

<hr>
<h2 id='reexports'>Objects exported from other packages</h2><span id='topic+reexports'></span><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>dplyr</dt><dd><p><code><a href="dplyr.html#topic+reexports">%&gt;%</a></code></p>
</dd>
</dl>

<hr>
<h2 id='standardise_names'>Standardise taxon names</h2><span id='topic+standardise_names'></span>

<h3>Description</h3>

<p>Standardises taxon names by performing a series of text substitutions to
remove common inconsistencies in taxonomic nomenclature.
</p>
<p>The function takes a character vector of taxon names as input and
returns a character vector of taxon names using standardised taxonomic syntax
as output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>standardise_names(taxon_names)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="standardise_names_+3A_taxon_names">taxon_names</code></td>
<td>
<p>A character vector of taxon names that need to be standardised.</p>
</td></tr>
</table>


<h3>Details</h3>


<ul>
<li><p> It removes stray punctuation at the start and end of a character string.
</p>
</li>
<li><p> It standardises unusual characters and symbols to ASCII equivalents.
</p>
</li>
<li><p> It standardises taxon rank abbreviations and qualifiers (subsp., var., f.),
as people use many variants of these terms.
</p>
</li>
<li><p> It standardises or removes a few additional filler words used within
taxon names (affinis becomes aff.; s.l. and s.s. are removed).
</p>
</li></ul>



<h3>Value</h3>

<p>A character vector of standardised taxon names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>standardise_names(c("Quercus suber",
                    "Eucalyptus sp.",
                    "Eucalyptus spp.",
                    "Agave americana var. marginata",
                    "Agave americana v marginata",
                    "Notelaea longifolia forma longifolia",
                    "Notelaea longifolia f longifolia"))
</code></pre>

<hr>
<h2 id='standardise_taxon_rank'>Standardise taxon ranks</h2><span id='topic+standardise_taxon_rank'></span>

<h3>Description</h3>

<p>Standardise taxon ranks from Latin into English.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>standardise_taxon_rank(taxon_rank)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="standardise_taxon_rank_+3A_taxon_rank">taxon_rank</code></td>
<td>
<p>A character vector of Latin taxon ranks.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function takes a character vector of Latin taxon ranks as input and
returns a character vector of taxon ranks using standardised English terms.
</p>


<h3>Value</h3>

<p>A character vector of English taxon ranks.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>standardise_taxon_rank(c("regnum", "kingdom", "classis", "class"))
</code></pre>

<hr>
<h2 id='state_diversity_counts'>State- and territory-level diversity</h2><span id='topic+state_diversity_counts'></span>

<h3>Description</h3>

<p>For Australian states and territories, use geographic distribution data from
the APC to calculate state-level diversity for native, introduced,
and more complicated species origins
</p>


<h3>Usage</h3>

<pre><code class='language-R'>state_diversity_counts(state, resources = load_taxonomic_resources())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="state_diversity_counts_+3A_state">state</code></td>
<td>
<p>A character string indicating the Australian state or
territory to calculate the diversity for. Possible values are &quot;NSW&quot;, &quot;NT&quot;,
&quot;Qld&quot;, &quot;WA&quot;, &quot;ChI&quot;, &quot;SA&quot;, &quot;Vic&quot;, &quot;Tas&quot;, &quot;ACT&quot;, &quot;NI&quot;, &quot;LHI&quot;, &quot;MI&quot;, &quot;HI&quot;,
&quot;MDI&quot;, &quot;CoI&quot;, &quot;CSI&quot;, and &quot;AR&quot;.</p>
</td></tr>
<tr><td><code id="state_diversity_counts_+3A_resources">resources</code></td>
<td>
<p>the taxonomic resources required to make the summary
statistics.  loading this can be slow, so call load_taxonomic_resources
separately to greatly speed this function up and pass the resources in.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble of diversity counts for the specified state or territory,
including native, introduced, and more complicated species origins.
The tibble has three columns: &quot;origin&quot; indicating the origin of the
species, &quot;state&quot; indicating the Australian state or territory, and
&quot;num_species&quot; indicating the number of species for that origin and state.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+load_taxonomic_resources">load_taxonomic_resources</a></code>
</p>
<p>Other diversity methods: 
<code><a href="#topic+create_species_state_origin_matrix">create_species_state_origin_matrix</a>()</code>,
<code><a href="#topic+native_anywhere_in_australia">native_anywhere_in_australia</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> state_diversity_counts(state = "NSW")
</code></pre>

<hr>
<h2 id='strip_names'>Strip taxon names</h2><span id='topic+strip_names'></span>

<h3>Description</h3>

<p>Strip taxonomic names of taxon rank abbreviations and qualifiers
and special characters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>strip_names(taxon_names)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="strip_names_+3A_taxon_names">taxon_names</code></td>
<td>
<p>A character vector of taxonomic names to be stripped.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Given a vector of taxonomic names, this function removes:
</p>

<ul>
<li><p> subtaxa designations (&quot;subsp.&quot;, &quot;var.&quot;, &quot;f.&quot;, and &quot;ser&quot;)
</p>
</li>
<li><p> special characters (e.g., &quot;-&quot;, &quot;.&quot;, &quot;(&quot;, &quot;)&quot;, &quot;?&quot;)
</p>
</li>
<li><p> extra whitespace
</p>
</li></ul>

<p>The resulting vector of names is also converted to lowercase.
</p>


<h3>Value</h3>

<p>A character vector of stripped taxonomic names,
with subtaxa designations, special characters, and extra whitespace
removed, and all letters converted to lowercase.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>strip_names(c("Abies lasiocarpa subsp. lasiocarpa",
              "Quercus kelloggii",
              "Pinus contorta var. latifolia"))

</code></pre>

<hr>
<h2 id='strip_names_extra'>Strip taxon names, extra</h2><span id='topic+strip_names_extra'></span>

<h3>Description</h3>

<p>Strip taxonomic names of <code>sp.</code> and hybrid symbols. This function assumes
that a character function has already been run through <code>strip_names</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>strip_names_extra(taxon_names)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="strip_names_extra_+3A_taxon_names">taxon_names</code></td>
<td>
<p>A character vector of taxonomic names to be stripped.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Given a vector of taxonomic names, this function removes additional filler
words (&quot; x &quot; for hybrid taxa, &quot;sp.&quot;) not removed by the function
<code>strip_names</code>
</p>


<h3>Value</h3>

<p>A character vector of stripped taxonomic names,
with <code>sp.</code> and hybrid symbols removed.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>strip_names_extra(c("Abies lasiocarpa subsp. lasiocarpa",
              "Quercus kelloggii",
              "Pinus contorta var. latifolia",
              "Acacia sp.",
              "Lepidium sp. Tanguin Hill (K.R.Newbey 10501)"))

</code></pre>

<hr>
<h2 id='update_taxonomy'>Update to currently accepted APC name and add APC/APNI name metadata</h2><span id='topic+update_taxonomy'></span>

<h3>Description</h3>

<p>For a list of taxon names aligned to the APC, update the name to an accepted
taxon concept per the APC and add scientific name and taxon concept metadata
to names aligned to either the APC or APNI.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>update_taxonomy(
  aligned_data,
  taxonomic_splits = "most_likely_species",
  quiet = TRUE,
  output = NULL,
  resources = load_taxonomic_resources()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="update_taxonomy_+3A_aligned_data">aligned_data</code></td>
<td>
<p>A tibble of plant names to update. This table must
include 5 columns, original_name, aligned_name, taxon_rank,
taxonomic_dataset, and aligned_reason.
These columns are created by the function <code>align_taxa</code>.
The columns <code>original_name</code> and <code>aligned_name</code> must be in the format of the
scientific name, with genus and species,
and may contain additional qualifiers such as subspecies or varieties. The
names are case insensitive.</p>
</td></tr>
<tr><td><code id="update_taxonomy_+3A_taxonomic_splits">taxonomic_splits</code></td>
<td>
<p>Variable that determines what protocol to use to
update taxon names that are ambiguous due to taxonomic splits.
The three options are:
</p>

<ul>
<li> <p><code>most_likely_species</code>, which returns the species name in use before the
split; alternative names are returned in a separate column
</p>
</li>
<li> <p><code>return_all</code>, which returns all possible names
</p>
</li>
<li> <p><code>collapse_to_higher_taxon</code>, which declares that an ambiguous name cannot
be aligned to an accepted species/infraspecific name and the name is
demoted to genus rank
</p>
</li></ul>
</td></tr>
<tr><td><code id="update_taxonomy_+3A_quiet">quiet</code></td>
<td>
<p>Logical to indicate whether to display messages while updating
taxa.</p>
</td></tr>
<tr><td><code id="update_taxonomy_+3A_output">output</code></td>
<td>
<p>(optional) Name of the file where results are saved. The
default is NULL and no file is created. If specified, the output will be
saved in a CSV file with the given name.</p>
</td></tr>
<tr><td><code id="update_taxonomy_+3A_resources">resources</code></td>
<td>
<p>the taxonomic resources required to make the summary
statistics.  Loading this can be slow, so call load_taxonomic_resources
separately to greatly speed this function up and pass the resources in.</p>
</td></tr>
</table>


<h3>Details</h3>


<ul>
<li><p> This function uses the APC to update the taxonomy of names aligned to a
taxon concept listed in the APC to the currently accepted name for the taxon
concept.
</p>
</li>
<li><p> The aligned_data data frame that is input must contain 5 columns,
<code>original_name</code>, <code>aligned_name</code>, <code>taxon_rank</code>, <code>taxonomic_dataset</code>, and
<code>aligned_reason</code>. (These are the columns output by the function <code>align_taxa</code>.)
</p>
</li>
<li><p> The aligned name is a plant name that has been aligned to a taxon name in
the APC or APNI by the align_taxa function.
</p>
</li></ul>

<p>Notes:
</p>

<ul>
<li><p> As the input for this function is a table with 5 columns (output by
align_taxa), this function will only be used when you explicitly want to
separate the alignment and updating components of APCalign. This function is
the second half of create_taxonomic_update_lookup.
</p>
</li></ul>



<h3>Value</h3>

<p>A tibble with updated taxonomy for the specified plant names. The
tibble contains the following columns:
</p>

<ul>
<li><p> original_name: the original plant name.
</p>
</li>
<li><p> aligned_name: the input plant name that has been aligned to a taxon name
in the APC or APNI by the align_taxa function.
</p>
</li>
<li><p> accepted_name: the APC-accepted plant name, when available.
</p>
</li>
<li><p> suggested_name: the suggested plant name to use. Identical to the
accepted_name, when an accepted_name exists; otherwise the the suggested_name
is the aligned_name.
</p>
</li>
<li><p> genus: the genus of the accepted (or suggested) name; only APC-accepted
genus names are filled in.
</p>
</li>
<li><p> family: the family of the accepted (or suggested) name; only APC-accepted
family names are filled in.
</p>
</li>
<li><p> taxon_rank: the taxonomic rank of the suggested (and accepted) name.
</p>
</li>
<li><p> taxonomic_dataset: the source of the suggested (and accepted) names (APC or
APNI).
</p>
</li>
<li><p> taxonomic_status: the taxonomic status of the suggested (and accepted) name.
</p>
</li>
<li><p> taxonomic_status_aligned: the taxonomic status of the aligned name, before
any taxonomic updates have been applied.
</p>
</li>
<li><p> aligned_reason: the explanation of a specific taxon name alignment (from an
original name to an aligned name).
</p>
</li>
<li><p> update_reason: the explanation of a specific taxon name update (from an
aligned name to an accepted or suggested name).
</p>
</li>
<li><p> subclass: the subclass of the accepted name.
</p>
</li>
<li><p> taxon_distribution: the distribution of the accepted name; only filled in
if an APC accepted_name is available.
</p>
</li>
<li><p> scientific_name_authorship: the authorship information for the accepted
(or synonymous) name; available for both APC and APNI names.
</p>
</li>
<li><p> taxon_ID: the unique taxon concept identifier for the accepted_name; only
filled in if an APC accepted_name is available.
</p>
</li>
<li><p> taxon_ID_genus: an identifier for the genus; only filled in if an
APC-accepted genus name is available.
</p>
</li>
<li><p> scientific_name_ID: an identifier for the nomenclatural (not taxonomic)
details of a scientific name; available for both APC and APNI names.
</p>
</li>
<li><p> row_number: the row number of a specific original_name in the input.
</p>
</li>
<li><p> number_of_collapsed_taxa: when taxonomic_splits == &quot;collapse_to_higher_taxon&quot;,
the number of possible taxon names that have been collapsed.
</p>
</li></ul>



<h3>See Also</h3>

<p>load_taxonomic_resources
</p>
<p>Other taxonomic alignment functions: 
<code><a href="#topic+align_taxa">align_taxa</a>()</code>,
<code><a href="#topic+create_taxonomic_update_lookup">create_taxonomic_update_lookup</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Update taxonomy for two plant names and print the result

resources &lt;- load_taxonomic_resources()

update_taxonomy(
 dplyr::tibble(
   original_name = c("Dryandra preissii", "Banksia acuminata"),
   aligned_name = c("Dryandra preissii", "Banksia acuminata"),
   taxon_rank = c("species", "species"),
   taxonomic_dataset = c("APC", "APC"),
   aligned_reason = c(NA_character_,
   NA_character_)
 ),
 resources = resources
)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
