<!DOCTYPE html><html><head><title>Help for package apex</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {apex}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#[,multidna,ANY,ANY,ANY-method'><p>Subset multidna objects</p></a></li>
<li><a href='#accessors'><p>multidna Accessors</p></a></li>
<li><a href='#add.gaps'><p>Add gap-only sequences for missing data</p></a></li>
<li><a href='#apex-package'><p>apex: Phylogenetic Methods for Multiple Gene Data</p></a></li>
<li><a href='#concatenate'><p>Concatenate genes into a single matrix</p></a></li>
<li><a href='#dist.multidna'><p>Pairwise distances for multiple gene data</p></a></li>
<li><a href='#getTree'><p>Build phylogenies from multiple gene data</p></a></li>
<li><a href='#initialize,multidna-method'><p>multidna constructor</p></a></li>
<li><a href='#initialize,multiphyDat-method'><p>multiphyDat constructor</p></a></li>
<li><a href='#multidna-class'><p>multidna: class for multiple gene data</p></a></li>
<li><a href='#multidna2alignment'><p>Convert from multidna into alignment (seqinr)</p></a></li>
<li><a href='#multidna2genind'><p>Convert multidna into genind</p></a></li>
<li><a href='#multidna2multiphyDat'><p>Conversions between multidna and multiphyDat</p></a></li>
<li><a href='#multiphyDat-class'><p>multiphyDat: class for multiple gene data</p></a></li>
<li><a href='#plot,multidna-method'><p>Display multidna objects</p></a></li>
<li><a href='#read.multidna'><p>Read multiple DNA alignments</p></a></li>
<li><a href='#rm.gaps'><p>Remove gap-only sequences for missing data</p></a></li>
<li><a href='#show,multidna-method'><p>Display multidna objects</p></a></li>
<li><a href='#show,multiphyDat-method'><p>Display multiphyDat objects</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Phylogenetic Methods for Multiple Gene Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.6</td>
</tr>
<tr>
<td>Description:</td>
<td>Toolkit for the analysis of multiple gene data (Jombart et al. 2017) &lt;<a href="https://doi.org/10.1111%2F1755-0998.12567">doi:10.1111/1755-0998.12567</a>&gt;. 
    'apex' implements the new S4 classes 'multidna', 'multiphyDat' and associated methods to handle aligned DNA sequences from multiple genes.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.3), methods, ape, phangorn</td>
</tr>
<tr>
<td>Imports:</td>
<td>utils, graphics, stats, adegenet</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/thibautjombart/apex">https://github.com/thibautjombart/apex</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/thibautjombart/apex/issues">https://github.com/thibautjombart/apex/issues</a></td>
</tr>
<tr>
<td>Collate:</td>
<td>doc.R internal.R multidna.class.R multiphyDat.class.R
add.gaps.R rm.gaps.R show.R multidna.constructor.R
multiphyDat.constructor.R accessors.R subset.R concatenate.R
plot.R readfiles.R datasets.R dist.R getTree.R exports.R</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-29 15:32:08 UTC; klaus</td>
</tr>
<tr>
<td>Author:</td>
<td>Klaus Schliep <a href="https://orcid.org/0000-0003-2941-0161"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Thibaut Jombart [aut],
  Zhian Namir Kamvar [aut],
  Eric Archer [aut],
  Rebecca Harris [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Klaus Schliep &lt;klaus.schliep@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-31 19:40:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='+5B+2Cmultidna+2CANY+2CANY+2CANY-method'>Subset multidna objects</h2><span id='topic++5B+2Cmultidna+2CANY+2CANY+2CANY-method'></span><span id='topic++5B+2Cmultidna-method'></span><span id='topic++5B.multidna'></span><span id='topic++5B+2CmultiphyDat-method'></span><span id='topic++5B.multiphyDat'></span><span id='topic++5B+2CmultiphyDat+2CANY+2CANY+2CANY-method'></span>

<h3>Description</h3>

<p>Individuals in a <a href="#topic+multidna-class">multidna</a> or <a href="#topic+multiphyDat-class">multiphyDat</a> object 
can be subsetted like the rows of a matrix, with the form x[i,].
Genes can be subsetted like the columns of a matrix, i.e. with the form x[,j].
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'multidna,ANY,ANY,ANY'
x[i, j, ..., drop = TRUE]

## S4 method for signature 'multiphyDat,ANY,ANY,ANY'
x[i, j, ..., drop = TRUE]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B2Cmultidna+2B2CANY+2B2CANY+2B2CANY-method_+3A_x">x</code></td>
<td>
<p>the <a href="#topic+multidna-class">multidna</a> object to subset.</p>
</td></tr>
<tr><td><code id="+2B5B+2B2Cmultidna+2B2CANY+2B2CANY+2B2CANY-method_+3A_i">i</code></td>
<td>
<p>a vector of logical, integers or characters to subset data by individuals; characters will be matched against individual labels.</p>
</td></tr>
<tr><td><code id="+2B5B+2B2Cmultidna+2B2CANY+2B2CANY+2B2CANY-method_+3A_j">j</code></td>
<td>
<p>a vector of logical, integers or characters to subset data by genes; characters will be matched against gene names labels.</p>
</td></tr>
<tr><td><code id="+2B5B+2B2Cmultidna+2B2CANY+2B2CANY+2B2CANY-method_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to other methods; currently ignored.</p>
</td></tr>
<tr><td><code id="+2B5B+2B2Cmultidna+2B2CANY+2B2CANY+2B2CANY-method_+3A_drop">drop</code></td>
<td>
<p>present for compatibility with the generic; currently not used.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(woodmouse)
genes &lt;- list(gene1=woodmouse[,1:500], gene2=woodmouse[,501:965])
x &lt;- new("multidna", genes)
x
plot(x)

## keep only the first 5 individuals
x[1:5,]
plot(x[1:5,])

## keep individuals 2,4,6 and the second gene
x[c(2,4,6),2]
plot(x[c(2,4,6),2])

</code></pre>

<hr>
<h2 id='accessors'>multidna Accessors</h2><span id='topic+accessors'></span><span id='topic+getNumInd+2Cmultidna-method'></span><span id='topic+getNumInd'></span><span id='topic+getNumInd+2Cmultidna'></span><span id='topic+getNumInd+2CmultiphyDat-method'></span><span id='topic+getNumInd+2CmultiphyDat'></span><span id='topic+getNumLoci'></span><span id='topic+getNumLoci+2Cmultidna-method'></span><span id='topic+getNumLoci+2Cmultidna'></span><span id='topic+getNumLoci+2CmultiphyDat-method'></span><span id='topic+getNumLoci+2CmultiphyDat'></span><span id='topic+getLocusNames'></span><span id='topic+getLocusNames+2Cmultidna-method'></span><span id='topic+getLocusNames+2Cmultidna'></span><span id='topic+getLocusNames+2CmultiphyDat-method'></span><span id='topic+getLocusNames+2CmultiphyDat'></span><span id='topic+setLocusNames+3C-'></span><span id='topic+setLocusNames+3C-+2Cmultidna-method'></span><span id='topic+setLocusNames+3C-+2Cmultidna'></span><span id='topic+setLocusNames+3C-+2CmultiphyDat-method'></span><span id='topic+setLocusNames+3C-+2CmultiphyDat'></span><span id='topic+getNumSequences'></span><span id='topic+getNumSequences+2Cmultidna-method'></span><span id='topic+getNumSequences+2Cmultidna'></span><span id='topic+getNumSequences+2CmultiphyDat-method'></span><span id='topic+getNumSequences+2CmultiphyDat'></span><span id='topic+getSequenceNames'></span><span id='topic+getSequenceNames+2Cmultidna-method'></span><span id='topic+getSequenceNames+2Cmultidna'></span><span id='topic+getSequenceNames+2CmultiphyDat-method'></span><span id='topic+getSequenceNames+2CmultiphyDat'></span><span id='topic+getSequences'></span><span id='topic+getSequences+2Cmultidna-method'></span><span id='topic+getSequences+2Cmultidna'></span><span id='topic+getSequences+2CmultiphyDat-method'></span><span id='topic+getSequences+2CmultiphyDat'></span>

<h3>Description</h3>

<p>Accessors for slots in <a href="#topic+multidna-class">multidna</a> and <a href="#topic+multiphyDat-class">multiphyDat</a> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getNumInd(x, ...)

## S4 method for signature 'multidna'
getNumInd(x, ...)

## S4 method for signature 'multiphyDat'
getNumInd(x, ...)

getNumLoci(x, ...)

## S4 method for signature 'multidna'
getNumLoci(x, ...)

## S4 method for signature 'multiphyDat'
getNumLoci(x, ...)

getLocusNames(x, ...)

## S4 method for signature 'multidna'
getLocusNames(x, ...)

## S4 method for signature 'multiphyDat'
getLocusNames(x, ...)

setLocusNames(x) &lt;- value

## S4 replacement method for signature 'multidna'
setLocusNames(x) &lt;- value

## S4 replacement method for signature 'multiphyDat'
setLocusNames(x) &lt;- value

getNumSequences(x, ...)

## S4 method for signature 'multidna'
getNumSequences(x, exclude.gap.only = TRUE,
  loci = NULL, ...)

## S4 method for signature 'multiphyDat'
getNumSequences(x, exclude.gap.only = TRUE,
  loci = NULL, ...)

getSequenceNames(x, ...)

## S4 method for signature 'multidna'
getSequenceNames(x, exclude.gap.only = TRUE,
  loci = NULL, ...)

## S4 method for signature 'multiphyDat'
getSequenceNames(x, exclude.gap.only = TRUE,
  loci = NULL, ...)

getSequences(x, ...)

## S4 method for signature 'multidna'
getSequences(x, loci = NULL, ids = NULL,
  simplify = TRUE, exclude.gap.only = TRUE, ...)

## S4 method for signature 'multiphyDat'
getSequences(x, loci = NULL, ids = NULL,
  simplify = TRUE, exclude.gap.only = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="accessors_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+multidna-class">multidna</a> or <a href="#topic+multiphyDat-class">multiphyDat</a> object.</p>
</td></tr>
<tr><td><code id="accessors_+3A_...">...</code></td>
<td>
<p>further arguments passed on to other functions.</p>
</td></tr>
<tr><td><code id="accessors_+3A_value">value</code></td>
<td>
<p>a replacement value for the slot.</p>
</td></tr>
<tr><td><code id="accessors_+3A_exclude.gap.only">exclude.gap.only</code></td>
<td>
<p>logical. Remove or ignore sequences containing all gaps?</p>
</td></tr>
<tr><td><code id="accessors_+3A_loci">loci</code></td>
<td>
<p>a character, numeric, or logical vector identifying which
loci to return.</p>
</td></tr>
<tr><td><code id="accessors_+3A_ids">ids</code></td>
<td>
<p>a character, numeric, or logical vector identifying which
sequences to return within a locus.</p>
</td></tr>
<tr><td><code id="accessors_+3A_simplify">simplify</code></td>
<td>
<p>logical. If <code>FALSE</code>, always return a list of
DNAbin sequences. If <code>TRUE</code> and only one locus has been requested,
return a single DNAbin object.</p>
</td></tr>
</table>


<h3>Details</h3>


<dl>
<dt>getNumInd</dt><dd><p>Returns the number of individuals.</p>
</dd>
<dt>getNumLoci</dt><dd><p>Returns the number of loci.</p>
</dd>
<dt>getLocusNames</dt><dd><p>Returns the names of each locus.</p>
</dd>
<dt>setLocusNames</dt><dd><p>Sets the names of each locus.</p>
</dd>
<dt>getNumSequences</dt><dd><p>Returns the number of sequences in each locus.</p>
</dd>
<dt>getSequenceNames</dt><dd><p>Returns the names of individual sequences at each
locus.</p>
</dd>
<dt>getSequences</dt><dd><p>Returns sequences of specified loci and individuals.</p>
</dd>
</dl>



<h3>Value</h3>

<p>returns the information stored in a slot, see details.
</p>

<hr>
<h2 id='add.gaps'>Add gap-only sequences for missing data</h2><span id='topic+add.gaps'></span><span id='topic+add.gaps.generic'></span><span id='topic+add.gaps.multidna'></span><span id='topic+add.gaps.multiphyDat'></span><span id='topic+add.gaps+2Cmultidna-method'></span><span id='topic+add.gaps+2CmultiphyDat-method'></span>

<h3>Description</h3>

<p>In <a href="#topic+multidna-class">multidna</a> and <a href="#topic+multiphyDat-class">multiphyDat</a>, some individuals may not be sequenced for all genes.
The generic function <code>add.gaps</code> has method for both objects; it identifies the missing sequences, and adds gap-only sequences to the alignments wherever needed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add.gaps(x, ...)

## S4 method for signature 'multidna'
add.gaps(x, ...)

## S4 method for signature 'multiphyDat'
add.gaps(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add.gaps_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+multidna-class">multidna</a> or <a href="#topic+multiphyDat-class">multiphyDat</a> object.</p>
</td></tr>
<tr><td><code id="add.gaps_+3A_...">...</code></td>
<td>
<p>further arguments passed to other methods (currently not used).</p>
</td></tr>
</table>

<hr>
<h2 id='apex-package'>apex: Phylogenetic Methods for Multiple Gene Data</h2><span id='topic+apex'></span><span id='topic+apex-package'></span>

<h3>Description</h3>

<p>Toolkit for the analysis of multiple gene data (Jombart et al. 2017) <a href="https://doi.org/10.1111/1755-0998.12567">doi:10.1111/1755-0998.12567</a>. 'apex' implements the new S4 classes 'multidna', 'multiphyDat' and associated methods to handle aligned DNA sequences from multiple genes.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Klaus Schliep <a href="mailto:klaus.schliep@gmail.com">klaus.schliep@gmail.com</a> (<a href="https://orcid.org/0000-0003-2941-0161">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a>
</p>
</li>
<li><p> Zhian Namir Kamvar <a href="mailto:kamvarz@science.oregonstate.edu">kamvarz@science.oregonstate.edu</a>
</p>
</li>
<li><p> Eric Archer <a href="mailto:eric.archer@noaa.gov">eric.archer@noaa.gov</a>
</p>
</li>
<li><p> Rebecca Harris <a href="mailto:rbharris@uw.edu">rbharris@uw.edu</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/thibautjombart/apex">https://github.com/thibautjombart/apex</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/thibautjombart/apex/issues">https://github.com/thibautjombart/apex/issues</a>
</p>
</li></ul>


<hr>
<h2 id='concatenate'>Concatenate genes into a single matrix</h2><span id='topic+concatenate'></span><span id='topic+concatenate.generic'></span><span id='topic+concatenate.multidna'></span><span id='topic+concatenate.multiphyDat'></span><span id='topic+concatenate+2Cmultidna-method'></span><span id='topic+concatenate+2CmultiphyDat-method'></span>

<h3>Description</h3>

<p>These functions concatenate separate DNA alignments into a single alignement matrix.
<code>concatenate</code> is a generic with methods for:
</p>

<ul>
<li> <p><code>multidna</code>: returns a <code>DNAbin</code> matrix
</p>
</li>
<li> <p><code>multiphyDat</code>: returns a <code>phyDat</code> object
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>concatenate(x, ...)

## S4 method for signature 'multidna'
concatenate(x, genes = TRUE, ...)

## S4 method for signature 'multiphyDat'
concatenate(x, genes = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="concatenate_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+multidna-class">multidna</a> or a <a href="#topic+multiphyDat-class">multiphyDat</a> object.</p>
</td></tr>
<tr><td><code id="concatenate_+3A_...">...</code></td>
<td>
<p>further arguments passed to other methods (currently not used).</p>
</td></tr>
<tr><td><code id="concatenate_+3A_genes">genes</code></td>
<td>
<p>an optional vector indicating the genes to retain for the concatenation; any way to subset the list in x@dna is acceptable; by default, all genes are used.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## simple conversion with nicely ordered output
data(woodmouse)
genes &lt;- list(gene1=woodmouse[,1:500], gene2=woodmouse[,501:965])
x &lt;- new("multidna", genes)
x
plot(x)

image(concatenate(x))
</code></pre>

<hr>
<h2 id='dist.multidna'>Pairwise distances for multiple gene data</h2><span id='topic+dist.multidna'></span>

<h3>Description</h3>

<p>This function computes pairwise genetic distances between individuals using genes in a <a href="#topic+multidna-class">multidna</a> object.
By default, one distance matrix (dist object) is created for each each, but a single distance can be derived by pooling all genes (argument <code>pool=TRUE</code>)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dist.multidna(x, pool = FALSE, genes = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dist.multidna_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+multidna-class">multidna</a> object.</p>
</td></tr>
<tr><td><code id="dist.multidna_+3A_pool">pool</code></td>
<td>
<p>a logical indicating if all genes should be pooled (concatenated) to obtain a single distance matrix; defaults to FALSE.</p>
</td></tr>
<tr><td><code id="dist.multidna_+3A_genes">genes</code></td>
<td>
<p>an optional vector indicating the genes to retain for the concatenation; any way to subset the list in x@dna is acceptable; by default, all genes are used.</p>
</td></tr>
<tr><td><code id="dist.multidna_+3A_...">...</code></td>
<td>
<p>further arguments passed to <code><a href="ape.html#topic+dist.dna">dist.dna</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of dist objects (pool=FALSE) or a single dist object (pool=TRUE)
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a>
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+dist.dna">dist.dna</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## simple conversion with nicely ordered output
data(woodmouse)
genes &lt;- list(gene1=woodmouse[,1:500], gene2=woodmouse[,501:965])
x &lt;- new("multidna", genes)
x
plot(x)

## get separate distance matrix and pooled one
lD &lt;- dist.multidna(x)
D &lt;- dist.multidna(x, pool=TRUE)

## get corresponding NJ trees
ltrees &lt;- lapply(lD, nj)
tree &lt;- nj(D)

opar &lt;- par(no.readonly=TRUE)
par(mfrow=c(3,1))
for(i in 1:2) plot(ltrees[[i]], main=names(ltrees)[i])
plot(tree, main="Pooled distances")
par(opar)

</code></pre>

<hr>
<h2 id='getTree'>Build phylogenies from multiple gene data</h2><span id='topic+getTree'></span>

<h3>Description</h3>

<p>This function builds separate phylogenetic trees for each gene in a <a href="#topic+multidna-class">multidna</a> object, specifying a method for computing pairwise distances between individuals, and a method to build the tree from the distance matrix. By default, procedures from ape are used.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getTree(x, pool = FALSE, genes = TRUE, model = "N",
  pairwise.deletion = TRUE, method = nj, ladderize = TRUE,
  negative.branch.length = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getTree_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+multidna-class">multidna</a> object.</p>
</td></tr>
<tr><td><code id="getTree_+3A_pool">pool</code></td>
<td>
<p>a logical indicating if all genes should be pooled (concatenated) to obtain a single tree; defaults to FALSE.</p>
</td></tr>
<tr><td><code id="getTree_+3A_genes">genes</code></td>
<td>
<p>an optional vector indicating the genes to retain for the concatenation; any way to subset the list in x@dna is acceptable; by default, all genes are used.</p>
</td></tr>
<tr><td><code id="getTree_+3A_model">model</code></td>
<td>
<p>a character string passed to <code><a href="ape.html#topic+dist.dna">dist.dna</a></code> describing the model to be used to compute genetic distances; defaults to 'N', the absolute number of mutations separating sequences.</p>
</td></tr>
<tr><td><code id="getTree_+3A_pairwise.deletion">pairwise.deletion</code></td>
<td>
<p>a logical passed to <code><a href="ape.html#topic+dist.dna">dist.dna</a></code> indicating if pairwise deletions should be used; the alternative is to remove all sites for which at least one missing value is present.</p>
</td></tr>
<tr><td><code id="getTree_+3A_method">method</code></td>
<td>
<p>a function building a tree from a matrix of pairwise genetic distances.</p>
</td></tr>
<tr><td><code id="getTree_+3A_ladderize">ladderize</code></td>
<td>
<p>a logical indicating if the tree should be ladderized; defaults to TRUE.</p>
</td></tr>
<tr><td><code id="getTree_+3A_negative.branch.length">negative.branch.length</code></td>
<td>
<p>a logical indicating if negative branch lengths should be allowed (e.g. in the case of Neighbor-Joining reconstruction), or not, in which case they are set to 0 (FALSE, default).</p>
</td></tr>
<tr><td><code id="getTree_+3A_...">...</code></td>
<td>
<p>further arguments passed to the tree reconstruction method defined by 'method'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>multiPhylo</code> object
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dist.multidna">dist.multidna</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## simple conversion with nicely ordered output
data(woodmouse)
genes &lt;- list(gene1=woodmouse[,1:500], gene2=woodmouse[,501:965])
x &lt;- new("multidna", genes)
x
plot(x)

## make trees, default parameters
trees &lt;- getTree(x)
trees
plot(trees, type="unrooted")

## make one single tree based on concatenated genes
tree &lt;- getTree(x, pool=TRUE)
tree
plot(tree, type="unrooted")

</code></pre>

<hr>
<h2 id='initialize+2Cmultidna-method'>multidna constructor</h2><span id='topic+initialize+2Cmultidna-method'></span><span id='topic+initialize+2Cmultidna-methods'></span><span id='topic+new.multidna'></span>

<h3>Description</h3>

<p>New <a href="#topic+multidna-class">multidna</a> objects can be created using <code>new("multidna", ...)</code> where &quot;...&quot; are arguments documented below.
The main input is a list of DNAbin matrices. The constructor ensures that all matrices will be reordered in the same way, and as an option (setting <code>add.gaps=TRUE</code>, gap-only sequences (&quot;...&mdash;&ndash;...&quot;) will be added wherever sequences are missing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'multidna'
initialize(.Object, dna = NULL, ind.info = NULL,
  gene.info = NULL, add.gaps = TRUE, quiet = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="initialize+2B2Cmultidna-method_+3A_.object">.Object</code></td>
<td>
<p>the object skeleton, automatically generated when calling <code>new</code>.</p>
</td></tr>
<tr><td><code id="initialize+2B2Cmultidna-method_+3A_dna">dna</code></td>
<td>
<p>a list of DNAbin matrices (1 per gene); rows should be labelled and indicate individuals, but different individuals and different orders can be used in different matrices.</p>
</td></tr>
<tr><td><code id="initialize+2B2Cmultidna-method_+3A_ind.info">ind.info</code></td>
<td>
<p>an optional data.frame containing information on the individuals, where individuals are in rows.</p>
</td></tr>
<tr><td><code id="initialize+2B2Cmultidna-method_+3A_gene.info">gene.info</code></td>
<td>
<p>an optional data.frame containing information on the genes, where genes are in rows.</p>
</td></tr>
<tr><td><code id="initialize+2B2Cmultidna-method_+3A_add.gaps">add.gaps</code></td>
<td>
<p>a logical indicating if gap-only sequences should be used where sequences are missing; defaults to TRUE.</p>
</td></tr>
<tr><td><code id="initialize+2B2Cmultidna-method_+3A_quiet">quiet</code></td>
<td>
<p>a logical indicating if messages should be shown; defaults to FALSE.</p>
</td></tr>
<tr><td><code id="initialize+2B2Cmultidna-method_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class <a href="#topic+multidna-class">multidna</a> containing alignments.
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a>
</p>


<h3>See Also</h3>


<ul>
<li><p> the <a href="#topic+multidna-class">multidna</a> class
</p>
</li>
<li> <p><code><a href="#topic+read.multidna">read.multidna</a></code> and <code><a href="#topic+read.multidna">read.multidna</a></code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
## empty object
new("multidna")

## simple conversion with nicely ordered output
data(woodmouse)
genes &lt;- list(gene1=woodmouse[,1:500], gene2=woodmouse[,501:965])
x &lt;- new("multidna", genes)
x
image(woodmouse)
image(x@dna[[1]])
image(x@dna[[2]])

## trickier conversion with missing sequences / wrong order
genes &lt;- list(gene1=woodmouse[,1:500], gene2=woodmouse[c(5:1,14:15),501:965])
x &lt;- new("multidna", genes)
x
image(x@dna[[1]])
image(x@dna[[2]])

</code></pre>

<hr>
<h2 id='initialize+2CmultiphyDat-method'>multiphyDat constructor</h2><span id='topic+initialize+2CmultiphyDat-method'></span><span id='topic+initialize+2CmultiphyDat-methods'></span><span id='topic+new.multiphyDat'></span>

<h3>Description</h3>

<p>New <a href="#topic+multiphyDat-class">multiphyDat</a> objects can be created using <code>new("multiphyDat", ...)</code> where &quot;...&quot; are arguments documented below.
The main input is a list of phyDat matrices. The constructor ensures that all matrices will be reordered in the same way, and genes with missing individuals will be filled by sequences of gaps (&quot;-&quot;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'multiphyDat'
initialize(.Object, seq = NULL,
  type = character(0), ind.info = NULL, gene.info = NULL,
  add.gaps = TRUE, quiet = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="initialize+2B2CmultiphyDat-method_+3A_.object">.Object</code></td>
<td>
<p>the object skeleton, automatically generated when calling <code>new</code>.</p>
</td></tr>
<tr><td><code id="initialize+2B2CmultiphyDat-method_+3A_seq">seq</code></td>
<td>
<p>a list of phyDat matrices (1 per gene); rows should be labelled and indicate individuals, but different individuals and different orders can be used in different matrices.</p>
</td></tr>
<tr><td><code id="initialize+2B2CmultiphyDat-method_+3A_type">type</code></td>
<td>
<p>a character string indicating the type of the sequences stored: &quot;DNA&quot; for DNA sequences, &quot;AA&quot; for amino-acids.</p>
</td></tr>
<tr><td><code id="initialize+2B2CmultiphyDat-method_+3A_ind.info">ind.info</code></td>
<td>
<p>an optional data.frame containing information on the individuals, where individuals are in rows.</p>
</td></tr>
<tr><td><code id="initialize+2B2CmultiphyDat-method_+3A_gene.info">gene.info</code></td>
<td>
<p>an optional data.frame containing information on the genes, where genes are in rows.</p>
</td></tr>
<tr><td><code id="initialize+2B2CmultiphyDat-method_+3A_add.gaps">add.gaps</code></td>
<td>
<p>a logical indicating if gap-only sequences should be used where sequences are missing; defaults to TRUE.</p>
</td></tr>
<tr><td><code id="initialize+2B2CmultiphyDat-method_+3A_quiet">quiet</code></td>
<td>
<p>a logical indicating if messages should be shown; defaults to FALSE.</p>
</td></tr>
<tr><td><code id="initialize+2B2CmultiphyDat-method_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class <a href="#topic+multiphyDat-class">multiphyDat</a> containing alignments.
</p>


<h3>Author(s)</h3>

<p>Klaus Schliep <a href="mailto:klaus.schliep@gmail.com">klaus.schliep@gmail.com</a>
</p>
<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a>
</p>


<h3>See Also</h3>


<ul>
<li><p> the <a href="#topic+multiphyDat-class">multiphyDat</a> class
</p>
</li>
<li> <p><code><a href="#topic+read.multiphyDat">read.multiphyDat</a></code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>data(Laurasiatherian)
#' ## empty object
new("multiphyDat")

## simple conversion with nicely ordered output
genes &lt;- list(gene1=Laurasiatherian[, 1:1600],
    gene2=Laurasiatherian[, 1601:3179])
x &lt;- new("multiphyDat", genes)
x

## trickier conversion with missing sequences / wrong order
genes &lt;- list(gene1=Laurasiatherian[1:40,],
    gene2=Laurasiatherian[8:47, ])
x &lt;- new("multiphyDat", genes)
x

</code></pre>

<hr>
<h2 id='multidna-class'>multidna: class for multiple gene data</h2><span id='topic+multidna-class'></span><span id='topic+multidna'></span><span id='topic+listOrNULL'></span><span id='topic+data.frameOrNULL'></span>

<h3>Description</h3>

<p>This formal (S4) class is used to store multiple DNA alignments.
Sequences are stored as a (possibly named) list, with each element of the list being a separate DNA alignment stored as a DNAbin matrix.
The rows of the separate matrices all correspond to the same individuals, ordered identically.
</p>


<h3>Slots</h3>


<dl>
<dt><code>dna</code></dt><dd><p>a list of DNAbin matrices; empty slot should be NULL</p>
</dd>
<dt><code>labels</code></dt><dd><p>a vector of labels of individuals</p>
</dd>
<dt><code>n.ind</code></dt><dd><p>the number of individuals</p>
</dd>
<dt><code>n.seq</code></dt><dd><p>the total number of sequences (pooling all genes), including gap sequences</p>
</dd>
<dt><code>n.seq.miss</code></dt><dd><p>the total number of gap-only sequences</p>
</dd>
<dt><code>ind.info</code></dt><dd><p>a data.frame containing information on the individuals, where individuals are in rows; empty slot should be NULL</p>
</dd>
<dt><code>gene.info</code></dt><dd><p>a data.frame containing information on the genes, where genes are in rows; empty slot should be NULL</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## empty object
new("multidna")

## simple conversion with nicely ordered output
data(woodmouse)
genes &lt;- list(gene1=woodmouse[,1:500], gene2=woodmouse[,501:965])
x &lt;- new("multidna", genes)
x
image(woodmouse)
image(x@dna[[1]])
image(x@dna[[2]])

## trickier conversion with missing sequences / wrong order
genes &lt;- list(gene1=woodmouse[,1:500], gene2=woodmouse[c(5:1,14:15),501:965])
x &lt;- new("multidna", genes)
x
image(x@dna[[1]])
image(x@dna[[2]])

</code></pre>

<hr>
<h2 id='multidna2alignment'>Convert from multidna into alignment (seqinr)</h2><span id='topic+multidna2alignment'></span><span id='topic+multiphyDat2alignment'></span>

<h3>Description</h3>

<p>The functions <code>multidna2alignment</code> and <code>multiphyDat2alignment</code> concatenates separate sequences and return an alignment object of the seqinr package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multidna2alignment(x, genes = TRUE)

multiphyDat2alignment(x, genes = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multidna2alignment_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+multidna-class">multidna</a> or <a href="#topic+multiphyDat-class">multiphyDat</a> object.</p>
</td></tr>
<tr><td><code id="multidna2alignment_+3A_genes">genes</code></td>
<td>
<p>an optional vector indicating the genes to retain for the concatenation; any way to subset the list in x@dna or x@seq is acceptable; by default, all genes are used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a alignment object
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a>, Zhian N. Kamvar, Klaus Schliep
</p>


<h3>See Also</h3>


<ul>
<li><p> concatenate
</p>
</li>
<li> <p><code><a href="ape.html#topic+as.alignment">as.alignment</a></code> to convert single DNAbin objects.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## simple conversion with nicely ordered output
data(woodmouse)
genes &lt;- list(gene1=woodmouse[,1:500], gene2=woodmouse[,501:965])
x &lt;- new("multidna", genes)
x
y &lt;- multidna2alignment(x)
y
x2 &lt;- multidna2multiphyDat(x)
z &lt;- multiphyDat2alignment(x2)
</code></pre>

<hr>
<h2 id='multidna2genind'>Convert multidna into genind</h2><span id='topic+multidna2genind'></span><span id='topic+multiphyDat2genind'></span>

<h3>Description</h3>

<p>The functions <code>multidna2genind</code> and <code>multiphyDat2genind</code> concatenates separate DNA alignments, and then extracts SNPs of the resulting alignment into a <a href="adegenet.html#topic+genind-class">genind</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multidna2genind(x, genes = TRUE, mlst = FALSE, gapIsNA = FALSE)

multiphyDat2genind(x, genes = TRUE, mlst = FALSE, gapIsNA = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multidna2genind_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+multidna-class">multidna</a> or <a href="#topic+multiphyDat-class">multiphyDat</a> object.</p>
</td></tr>
<tr><td><code id="multidna2genind_+3A_genes">genes</code></td>
<td>
<p>an optional vector indicating the genes to retain for the concatenation; any way to subset the list in x@dna or x@seq is acceptable; by default, all genes are used.</p>
</td></tr>
<tr><td><code id="multidna2genind_+3A_mlst">mlst</code></td>
<td>
<p>if <code>TRUE</code>, each gene will result in a single locus in the genind object. (Default to <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="multidna2genind_+3A_gapisna">gapIsNA</code></td>
<td>
<p>if <code>TRUE</code> and <code>mlst = TRUE</code>, sequences that consist entirely of gaps will be considered as NAs. (Default to <code>FALSE</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="adegenet.html#topic+genind-class">genind</a> object
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a>, Zhian N. Kamvar, Klaus Schliep
</p>


<h3>See Also</h3>


<ul>
<li><p> concatenate
</p>
</li>
<li> <p><code><a href="adegenet.html#topic+DNAbin2genind">DNAbin2genind</a></code> to convert single DNAbin objects.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
## simple conversion with nicely ordered output
data(woodmouse)
genes &lt;- list(gene1=woodmouse[,1:500], gene2=woodmouse[,501:965])
x &lt;- new("multidna", genes)
x
y &lt;- multidna2multiphyDat(x)
y
z1 &lt;- multidna2genind(x)
z1
z2 &lt;- multiphyDat2genind(y)
all.equal(z1, z2)

</code></pre>

<hr>
<h2 id='multidna2multiphyDat'>Conversions between multidna and multiphyDat</h2><span id='topic+multidna2multiphyDat'></span><span id='topic+multiphyDat2multidna'></span>

<h3>Description</h3>

<p>The functions <code>multidna2multiphyDat</code> and <code>multiphyDat2multidna</code> are used to convert data between <a href="#topic+multidna-class">multidna</a> and  <a href="#topic+multiphyDat-class">multiphyDat</a> classes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multidna2multiphyDat(x)

multiphyDat2multidna(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multidna2multiphyDat_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+multidna-class">multidna</a> or <a href="#topic+multiphyDat-class">multiphyDat</a> object.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a>, Zhian N. Kamvar, Klaus Schliep
</p>


<h3>See Also</h3>


<ul>
<li><p> concatenate
</p>
</li>
<li> <p><code><a href="adegenet.html#topic+DNAbin2genind">DNAbin2genind</a></code> to convert single DNAbin objects.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
## simple conversion with nicely ordered output
data(woodmouse)
genes &lt;- list(gene1=woodmouse[,1:500], gene2=woodmouse[,501:965])
x &lt;- new("multidna", genes)
x

## conversion multidna -&gt; multiphyDat
y &lt;- multidna2multiphyDat(x)
y

## check round trip
identical(x, multiphyDat2multidna(y))

</code></pre>

<hr>
<h2 id='multiphyDat-class'>multiphyDat: class for multiple gene data</h2><span id='topic+multiphyDat-class'></span><span id='topic+multiphyDat'></span>

<h3>Description</h3>

<p>This formal (S4) class is identical to <a href="#topic+multidna-class">multidna</a>, except that DNA sequences are stored using <code>phyDat</code> objects from the <code>phangorn</code> package.
Sequences are stored as a (possibly named) list, with each element of the list being a separate DNA alignment stored as a <code>phyDat</code> object.
The rows of the separate matrices all correspond to the same individuals, ordered identically.
</p>


<h3>Slots</h3>


<dl>
<dt><code>seq</code></dt><dd><p>a list of phyDat objects; empty slot should be NULL</p>
</dd>
<dt><code>type</code></dt><dd><p>a character string indicating the type of the sequences stored: &quot;DNA&quot; for DNA sequences, &quot;AA&quot; for amino-acids.</p>
</dd>
<dt><code>labels</code></dt><dd><p>a vector of labels of individuals</p>
</dd>
<dt><code>n.ind</code></dt><dd><p>the number of individuals</p>
</dd>
<dt><code>n.seq</code></dt><dd><p>the total number of sequences (pooling all genes), including gap sequences</p>
</dd>
<dt><code>n.seq.miss</code></dt><dd><p>the total number of gap-only sequences</p>
</dd>
<dt><code>ind.info</code></dt><dd><p>a data.frame containing information on the individuals, where individuals are in rows; empty slot should be NULL</p>
</dd>
<dt><code>gene.info</code></dt><dd><p>a data.frame containing information on the genes, where genes are in rows; empty slot should be NULL</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Klaus Schliep <a href="mailto:klaus.schliep@gmail.com">klaus.schliep@gmail.com</a>
</p>
<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Laurasiatherian)

## empty object
new("multiphyDat")

## simple conversion with nicely ordered output
data(Laurasiatherian)
genes &lt;- list(gene1=Laurasiatherian[, 1:1600],
              gene2=Laurasiatherian[, 1601:3179])
x &lt;- new("multiphyDat", genes)
x

## trickier conversion with missing sequences / wrong order
genes &lt;- list(gene1=Laurasiatherian[1:40,],
    gene2=Laurasiatherian[8:47,])
x &lt;- new("multiphyDat", genes)
x

</code></pre>

<hr>
<h2 id='plot+2Cmultidna-method'>Display multidna objects</h2><span id='topic+plot+2Cmultidna-method'></span><span id='topic+plot.multidna'></span>

<h3>Description</h3>

<p>Default printing for multidna objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'multidna'
plot(x, y, rows = TRUE, ask = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Cmultidna-method_+3A_x">x</code></td>
<td>
<p>a multidna object</p>
</td></tr>
<tr><td><code id="plot+2B2Cmultidna-method_+3A_y">y</code></td>
<td>
<p>an integer vector indicating the genes to plot</p>
</td></tr>
<tr><td><code id="plot+2B2Cmultidna-method_+3A_rows">rows</code></td>
<td>
<p>a logical indicating if different genes should be displayed in separate rows</p>
</td></tr>
<tr><td><code id="plot+2B2Cmultidna-method_+3A_ask">ask</code></td>
<td>
<p>a logical indicating if the user should be prompted between graphs</p>
</td></tr>
<tr><td><code id="plot+2B2Cmultidna-method_+3A_...">...</code></td>
<td>
<p>arguments passed to <code><a href="ape.html#topic+image.DNAbin">image.DNAbin</a></code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simple conversion with nicely ordered output
data(woodmouse)
genes &lt;- list(gene1=woodmouse[,1:500], gene2=woodmouse[,501:965])
x &lt;- new("multidna", genes)
x
plot(x)

</code></pre>

<hr>
<h2 id='read.multidna'>Read multiple DNA alignments</h2><span id='topic+read.multidna'></span><span id='topic+read.multiFASTA'></span><span id='topic+read.multiphyDat'></span>

<h3>Description</h3>

<p>These functions read multiple DNA alignments and store the output in a <a href="#topic+multidna-class">multidna</a> object.
They are relying on ape's original functions <code><a href="ape.html#topic+read.dna">read.dna</a></code> and <code><a href="ape.html#topic+read.FASTA">read.FASTA</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.multidna(files, add.gaps = TRUE, ...)

read.multiFASTA(files, add.gaps = TRUE)

read.multiphyDat(files, add.gaps = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.multidna_+3A_files">files</code></td>
<td>
<p>a vector of characters indicating the paths to the files to read from.</p>
</td></tr>
<tr><td><code id="read.multidna_+3A_add.gaps">add.gaps</code></td>
<td>
<p>a logical indicating if gap-only sequences should be added wherever sequences are missing; defaults to TRUE.</p>
</td></tr>
<tr><td><code id="read.multidna_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to the functions <code><a href="ape.html#topic+read.dna">read.dna</a></code> and <code><a href="ape.html#topic+read.FASTA">read.FASTA</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>read.multidna</code> and <code>read.multiFASTA</code> return an object of
class <a href="#topic+multidna-class">multidna</a>, <code>read.multiphyDat</code> returns an object of
class <a href="#topic+multiphyDat-class">multiphyDat</a>.
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a>
</p>
<p>Klaus Schliep <a href="mailto:klaus.schliep@gmail.com">klaus.schliep@gmail.com</a>
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="ape.html#topic+read.dna">read.dna</a></code>
</p>
</li>
<li>  <p><code><a href="ape.html#topic+read.FASTA">read.FASTA</a></code>
</p>
</li>
<li> <p><code><a href="phangorn.html#topic+read.phyDat">read.phyDat</a></code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## get path to the files
files &lt;- dir(system.file(package="apex"),patter="patr", full=TRUE)
files

## read files
x &lt;- read.multiFASTA(files)
x
opar &lt;- par(no.readonly=TRUE)
par(mfrow=c(2,2))
plot(x, row=FALSE)
par(opar)

y &lt;- read.multiphyDat(files, format="fasta")
y

</code></pre>

<hr>
<h2 id='rm.gaps'>Remove gap-only sequences for missing data</h2><span id='topic+rm.gaps'></span><span id='topic+rm.gaps.generic'></span><span id='topic+rm.gaps.multidna'></span><span id='topic+rm.gaps.multiphyDat'></span><span id='topic+rm.gaps+2Cmultidna-method'></span><span id='topic+rm.gaps+2CmultiphyDat-method'></span>

<h3>Description</h3>

<p>In <a href="#topic+multidna-class">multidna</a> and <a href="#topic+multiphyDat-class">multiphyDat</a>, some individuals may not be sequenced for all genes, resulting in gap-only sequences for missing data.
The generic function <code>rm.gaps</code> has method for both objects; it identifies the missing sequences, and removes gap-only sequences from the alignments wherever needed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rm.gaps(x, ...)

## S4 method for signature 'multidna'
rm.gaps(x, ...)

## S4 method for signature 'multiphyDat'
rm.gaps(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rm.gaps_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+multidna-class">multidna</a> or <a href="#topic+multiphyDat-class">multiphyDat</a> object.</p>
</td></tr>
<tr><td><code id="rm.gaps_+3A_...">...</code></td>
<td>
<p>further arguments passed to other methods (currently not used).</p>
</td></tr>
</table>

<hr>
<h2 id='show+2Cmultidna-method'>Display multidna objects</h2><span id='topic+show+2Cmultidna-method'></span><span id='topic+show.multidna'></span>

<h3>Description</h3>

<p>Default printing for multidna objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'multidna'
show(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show+2B2Cmultidna-method_+3A_object">object</code></td>
<td>
<p>a multidna object</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>show</code> returns an invisible NULL, called for side effects.
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a>
</p>

<hr>
<h2 id='show+2CmultiphyDat-method'>Display multiphyDat objects</h2><span id='topic+show+2CmultiphyDat-method'></span><span id='topic+show.multiphyDat'></span>

<h3>Description</h3>

<p>Default printing for multiphyDat objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'multiphyDat'
show(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show+2B2CmultiphyDat-method_+3A_object">object</code></td>
<td>
<p>a multiphyDat object</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>show</code> returns an invisible NULL, called for side effects.
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
