<!DOCTYPE html><html><head><title>Help for package aroma.apd</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {aroma.apd}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#aroma.apd-package'><p>Package aroma.apd</p></a></li>
<li><a href='#cdfToApdMap'><p>Generates an APD read map file from an Affymetrix CDF file</p></a></li>
<li><a href='#celToApd'><p>Generates an APD file from an Affymetrix CEL file</p></a></li>
<li><a href='#createApd'><p>Creates an Affymetrix probe data (APD) file</p></a></li>
<li><a href='#findApdMap'><p>Search for APD map files in multiple directories</p></a></li>
<li><a href='#gtypeCelToPQ'><p>Function to immitate Affymetrix' gtype_cel_to_pq software</p></a></li>
<li><a href='#Non-documented objects'><p>Non-documented objects</p></a></li>
<li><a href='#readApd'><p>Reads an Affymetrix probe data (APD) file</p></a></li>
<li><a href='#readApdHeader'><p>Reads the header of an Affymetrix probe data (APD) file</p></a></li>
<li><a href='#readApdMap'><p>Reads an APD probe map file</p></a></li>
<li><a href='#readApdRectangle'><p>Reads a spatial subset of probe-level data from Affymetrix APD files</p></a></li>
<li><a href='#readApdUnits'><p>Reads Affymetrix probe data (APD) as units (probesets)</p></a></li>
<li><a href='#updateApd'><p>Updates an Affymetrix probe data (APD) file</p></a></li>
<li><a href='#updateApdHeader'><p>Updates the header of an Affymetrix probe data (APD) file</p></a></li>
<li><a href='#updateApdUnits'><p>Updates an Affymetrix probe data (APD) file by units (probesets)</p></a></li>
<li><a href='#writeApd'><p>Writes an APD probe data file</p></a></li>
<li><a href='#writeApdMap'><p>Writes an APD probe map file</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>0.7.0</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>R.methodsS3 (&ge; 1.7.1), R.oo (&ge; 1.23.0), R.utils (&ge; 2.2.0),
R.huge (&ge; 0.9.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>affxparser</td>
</tr>
<tr>
<td>SuggestsNote:</td>
<td>Recommended: affxparser</td>
</tr>
<tr>
<td>Title:</td>
<td>A Probe-Level Data File Format Used by 'aroma.affymetrix'
[deprecated]</td>
</tr>
<tr>
<td>Description:</td>
<td>DEPRECATED. Do not start building new projects based on this package. (The (in-house) APD file format was initially developed to store Affymetrix probe-level data, e.g. normalized CEL intensities.  Chip types can be added to APD file and similar to methods in the affxparser package, this package provides methods to read APDs organized by units (probesets).  In addition, the probe elements can be arranged optimally such that the elements are guaranteed to be read in order when, for instance, data is read unit by unit.  This speeds up the read substantially.  This package is supporting the Aroma framework and should not be used elsewhere.)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-2.1">LGPL-2.1</a> | <a href="https://www.r-project.org/Licenses/LGPL-3">LGPL-3</a> [expanded from: LGPL (&ge; 2.1)]</td>
</tr>
<tr>
<td>URL:</td>
<td>URL: <a href="https://www.aroma-project.org/">https://www.aroma-project.org/</a>,
<a href="https://github.com/HenrikBengtsson/aroma.apd">https://github.com/HenrikBengtsson/aroma.apd</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/HenrikBengtsson/aroma.apd/issues">https://github.com/HenrikBengtsson/aroma.apd/issues</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>TRUE</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Microarray, DataImport</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-06-23 10:29:47 UTC; henrik</td>
</tr>
<tr>
<td>Author:</td>
<td>Henrik Bengtsson [aut, cre, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Henrik Bengtsson &lt;henrikb@braju.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-06-23 11:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='aroma.apd-package'>Package aroma.apd</h2><span id='topic+aroma.apd-package'></span><span id='topic+aroma.apd'></span>

<h3>Description</h3>

<p><em>This package has been deprecated. Do not start building new
projects based on it.</em>
</p>
<p>DEPRECATED. Do not start building new projects based on this package. (The (in-house) APD file format was initially developed to store Affymetrix probe-level data, e.g. normalized CEL intensities.  Chip types can be added to APD file and similar to methods in the affxparser package, this package provides methods to read APDs organized by units (probesets).  In addition, the probe elements can be arranged optimally such that the elements are guaranteed to be read in order when, for instance, data is read unit by unit.  This speeds up the read substantially.  This package is supporting the Aroma framework and should not be used elsewhere.)
</p>


<h3>Requirements</h3>

<p>This package requires the packages <span class="pkg">R.huge</span> and <span class="pkg">affxparser</span>.
</p>


<h3>Installation and updates</h3>

<p>To install this package, see <a href="https://www.braju.com/R/">https://www.braju.com/R/</a>.
</p>


<h3>To get started</h3>

<p>To get started, see:
</p>

<ol>
<li> <p><code><a href="#topic+readApd">readApd</a></code>(), <code><a href="#topic+readApdUnits">readApdUnits</a></code>(), <code><a href="#topic+readApdRectangle">readApdRectangle</a></code>()
- Reads APD files.
</p>
</li>
<li> <p><code><a href="#topic+celToApd">celToApd</a></code>() - creates an APD file from a CEL file.
</p>
</li>
<li> <p><code><a href="#topic+cdfToApdMap">cdfToApdMap</a></code>() - creates an APD read map from a CDF file.
</p>
</li>
<li> <p><code><a href="#topic+findApdMap">findApdMap</a></code>() - finds an APD read map.
</p>
</li>
<li> <p><code><a href="#topic+createApd">createApd</a></code>(), <code><a href="#topic+writeApd">writeApd</a></code>() - creates APD files.
</p>
</li>
<li> <p><code><a href="#topic+updateApd">updateApd</a></code>(), <code><a href="#topic+updateApdUnits">updateApdUnits</a></code>() - updates APD files.
</p>
</li></ol>



<h3>Search paths</h3>

<p>Typically you do not have to specify the pathname of the CDF file
when reading APD files or similar.  Instead, the chip type is
obtained from the APD header and the corresponding APD file is
search for in several predefined locations.  For details how to
specify the search path, see <code><a href="affxparser.html#topic+findCdf">findCdf</a></code>.
</p>
<p>Some APD files uses a corresponding read map in order to read data
faster.  The <code><a href="#topic+findApdMap">findApdMap</a></code>() method is used to find the
corresponding APD map file containing the map vector.  How to
specify search paths for map files, see that method.
</p>


<h3>How to cite this package</h3>

<p>Currently no information.
</p>


<h3>License</h3>

<p>The releases of this package is licensed under
LGPL version 2.1 or newer.
</p>
<p>The development code of the packages is under a private licence
(where applicable) and patches sent to the author fall under the
latter license, but will be, if incorporated, released under the
&quot;release&quot; license above.
</p>


<h3>References</h3>

<p>[1] H. Bengtsson, <em>The R.oo package - Object-Oriented Programming with References Using Standard R Code</em>, In Kurt Hornik, Friedrich Leisch and Achim Zeileis, editors, Proceedings of the 3rd International Workshop on Distributed Statistical Computing (DSC 2003), March 20-22, Vienna, Austria. <a href="https://www.r-project.org/conferences/DSC-2003/Proceedings/">https://www.r-project.org/conferences/DSC-2003/Proceedings/</a>
<br />
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='cdfToApdMap'>Generates an APD read map file from an Affymetrix CDF file</h2><span id='topic+cdfToApdMap.default'></span><span id='topic+cdfToApdMap'></span>

<h3>Description</h3>

<p>Generates an APD read map file from an Affymetrix CDF file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
cdfToApdMap(filename, mapType=NULL, mapFile=NULL, mapPath=NULL, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cdfToApdMap_+3A_filename">filename</code></td>
<td>
<p>The pathname of the CDF file.</p>
</td></tr>
<tr><td><code id="cdfToApdMap_+3A_maptype">mapType</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string naming the type of the map.
If <code><a href="base.html#topic+NULL">NULL</a></code>, the chip type will be used.</p>
</td></tr>
<tr><td><code id="cdfToApdMap_+3A_mapfile">mapFile</code></td>
<td>
<p>The filename of the resulting APD map file. If <code><a href="base.html#topic+NULL">NULL</a></code>,
the map type with filename extension <code>apm</code> will be used.</p>
</td></tr>
<tr><td><code id="cdfToApdMap_+3A_mappath">mapPath</code></td>
<td>
<p>An optional path where to the map file will be stored.</p>
</td></tr>
<tr><td><code id="cdfToApdMap_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to
<code><a href="affxparser.html#topic+readCdfUnitsWriteMap">readCdfUnitsWriteMap</a></code>, e.g. <code>units</code>.</p>
</td></tr>
<tr><td><code id="cdfToApdMap_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns (invisibly) a <code><a href="base.html#topic+list">list</a></code> structure with elements:
</p>
<table>
<tr><td><code>pathname</code></td>
<td>
<p>The pathname of the generated APD map file.</p>
</td></tr>
<tr><td><code>mapType</code></td>
<td>
<p>The map type <code><a href="base.html#topic+character">character</a></code> string.</p>
</td></tr>
<tr><td><code>chipType</code></td>
<td>
<p>The chip type <code><a href="base.html#topic+character">character</a></code> string.</p>
</td></tr>
<tr><td><code>readMap</code></td>
<td>
<p>The generated read map <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>To read an APD map file, see <code><a href="#topic+readApdMap">readApdMap</a></code>().
</p>

<hr>
<h2 id='celToApd'>Generates an APD file from an Affymetrix CEL file</h2><span id='topic+celToApd.default'></span><span id='topic+celToApd'></span>

<h3>Description</h3>

<p>Generates an APD file from an Affymetrix CEL file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
celToApd(filename, apdFile=NULL, mapType="asChipType", writeMap=NULL, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="celToApd_+3A_filename">filename</code></td>
<td>
<p>The pathname of the CEL file.</p>
</td></tr>
<tr><td><code id="celToApd_+3A_apdfile">apdFile</code></td>
<td>
<p>An optional pathname of the APD file, otherwise
it will be the same as the CEL file with extension replaced
with 'apd'.</p>
</td></tr>
<tr><td><code id="celToApd_+3A_maptype">mapType</code></td>
<td>
<p>The type of read map for the generated APD file.
If <code><a href="base.html#topic+NULL">NULL</a></code>, no remapping of the cell indices is done.
If <code>"asChipType"</code>, the map type is the same as the chip type
of the CEL file.
If any other <code><a href="base.html#topic+character">character</a></code> string, it sets the map type to that string.
Note that there must be a APD map file with this type that can
be found by <code><a href="#topic+findApdMap">findApdMap</a></code>().
</p>
</td></tr>
<tr><td><code id="celToApd_+3A_writemap">writeMap</code></td>
<td>
<p>An optional <em>write</em> map <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> used
to remap the cell indices for optimal reading speed.  If <code><a href="base.html#topic+NULL">NULL</a></code>,
the write map may be obtained from the read map file specified by
the <code>mapType</code> argument.</p>
</td></tr>
<tr><td><code id="celToApd_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+writeApd">writeApd</a></code>().</p>
</td></tr>
<tr><td><code id="celToApd_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns (invisibly) the pathname of the written APD file.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>To create an APD map file from a CDF file, see <code><a href="#topic+cdfToApdMap">cdfToApdMap</a></code>().
To read an APD file, see <code><a href="#topic+readApd">readApd</a></code>().
To read an APD map file, see <code><a href="#topic+readApdMap">readApdMap</a></code>().
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library("R.utils")  ## Arguments

# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# 1. Scan for existing CEL files
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# a) Scan for CEL files
files &lt;- list.files(pattern="[.](cel|CEL)$")
files &lt;- files[!file.info(files)$isdir]
if (length(files) &gt; 0 &amp;&amp; require("affxparser")) {
  cat("Create an optimal read map for CEL file:", files[1], "\n")
  cdffile &lt;- findCdf(readCelHeader(files[1])$chiptype)
  res &lt;- cdfToApdMap(cdffile)
 
  cat("Converting CEL file to APD file:", files[1], "\n")
  apdfile &lt;- celToApd(files[1])
  cat("Created APD file:", apdfile, "\n")
  file.remove(apdfile)
 
  cat("Converting CEL file to APD file with an optimized read map:", files[1], "\n")
  apdfile &lt;- celToApd(files[1], mapType=res$mapType)
  cat("Created APD file:", apdfile, "\n")
 
  writeMap &lt;- invertMap(res$readMap)
  for (file in files[-1]) {
    cat("Converting CEL file to APD file with an optimized read map:", file, "\n")
    apdfile &lt;- celToApd(file, mapType=res$mapType, writeMap=writeMap)
    cat("Created APD file:", apdfile, "\n")
  }
} # if (length(files) &gt; 0)
</code></pre>

<hr>
<h2 id='createApd'>Creates an Affymetrix probe data (APD) file</h2><span id='topic+createApd.default'></span><span id='topic+createApd'></span>

<h3>Description</h3>

<p>Creates an Affymetrix probe data (APD) file.
</p>
<p>An Affymetrix probe data (APD) structure can hold a header and
a numeric data vector.  Since the APD structure is kept on file
all the time, the number of elements in the data vector is only
limited by the file system and not the amount of system memory
available.  For more details, see the <code><a href="R.huge.html#topic+FileVector">FileVector</a></code>
class (and its superclass), which is used internally.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
createApd(filename, nbrOfCells, dataType=c("float", "double", "integer", "short",
  "byte"), chipType=NULL, mapType=NULL, ..., verbose=FALSE, .checkArgs=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="createApd_+3A_filename">filename</code></td>
<td>
<p>The filename of the APD file.</p>
</td></tr>
<tr><td><code id="createApd_+3A_nbrofcells">nbrOfCells</code></td>
<td>
<p>The number of cells (probes) data elements the
APD file structure should hold.</p>
</td></tr>
<tr><td><code id="createApd_+3A_datatype">dataType</code></td>
<td>
<p>The data type of the data elements.</p>
</td></tr>
<tr><td><code id="createApd_+3A_chiptype">chipType</code></td>
<td>
<p>An (optional) <code><a href="base.html#topic+character">character</a></code> string specifying the
chip type.</p>
</td></tr>
<tr><td><code id="createApd_+3A_maptype">mapType</code></td>
<td>
<p>An (optional) <code><a href="base.html#topic+character">character</a></code> string specifying the
probe-index map.  Use by <code><a href="#topic+findApdMap">findApdMap</a></code>() to find read map.</p>
</td></tr>
<tr><td><code id="createApd_+3A_...">...</code></td>
<td>
<p>Additional named arguments added to the header of the
APD file structure.</p>
</td></tr>
<tr><td><code id="createApd_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
<tr><td><code id="createApd_+3A_.checkargs">.checkArgs</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, arguments are checked, otherwise not.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns (invisibly) the pathname of the file created.
</p>


<h3>Data type</h3>

<p>Valid data types are: byte (1 byte), short (2 bytes), integer (4 bytes),
float (4 bytes), and double (8 bytes).
</p>
<p>Note that in Affymetrix CEL files, the probe intensities as well as
the standard deviations are stored as floats (4 bytes) and not doubles
(8 bytes).  This is why, the default data type is <code>"float"</code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+updateApd">updateApd</a></code>() and <code><a href="#topic+readApd">readApd</a></code>().
To find a map of a certain type, see <code><a href="#topic+findApdMap">findApdMap</a></code>().
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# Float precision
.Machine$float.eps &lt;- (2^((8-4)*8)*.Machine$double.eps)
tol &lt;- .Machine$float.eps ^ 0.5


# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# 1. Create an Affymetrix Probe Signal (APD) file for a
#    'Mapping50K_Hind240' with 1600-by-1600 probes.
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
chipType &lt;- "Mapping50K_Hind240"
nbrOfCells &lt;- 1600^2

pathname &lt;- paste(tempfile(), "apd", sep=".")
createApd(pathname, nbrOfCells=nbrOfCells, chipType=chipType)

# File size
cat("File name:", pathname, "\n")
cat("File size:", file.info(pathname)$size, "bytes\n")
cat("APD header:\n")
header &lt;- readApdHeader(pathname)
print(header)


# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# 2. Update the signals for a subset of probes
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
cells &lt;- c(1, 1100:1120, 220:201, 998300:999302)
signals &lt;- log(cells+1, base=2)  # Fake signals
updateApd(pathname, indices=cells, data=signals)


# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# 3. Re-read the signals for a subset of probes
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
apd &lt;- readApd(pathname, indices=cells)

# Signals in APD files are stored as floats (since this is
# the precision in CEL files).
stopifnot(all.equal(signals, apd$intensities, tolerance=tol))


# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# 4. Re-read the signals for a subset of probes
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
if (require("affxparser") &amp;&amp; FALSE) {
  cdfFile &lt;- findCdf(chipType)
  if (length(cdfFile) &gt; 0) {
 
    apd &lt;- readApdUnits(pathname, units=100:104)
   
    # Sample new data (with one decimal precision)
    apd2 &lt;- lapply(apd, function(unit) {
      lapply(unit, function(groups) {
        n &lt;- length(groups$intensities)
        values &lt;- as.integer(runif(n, max=655350))/10
        list(intensities=values)
      })
    })
   
    # Update APD file with new data
    updateApdUnits(pathname, units=100:104, data=apd2)
   
    # Re-read data to verify correctness
    apd &lt;- readApdUnits(pathname, units=100:104)
   
    # Signals in APD files are stored as floats (since this is
    # the precision in CEL files).
    stopifnot(all.equal(apd2, apd, tolerance=tol))
  } # if (length(cdfFile) &gt; 0 ...)
} # if (require("affxparser"))


# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# 4. Clean up
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
file.remove(pathname)


</code></pre>

<hr>
<h2 id='findApdMap'>Search for APD map files in multiple directories</h2><span id='topic+findApdMap.default'></span><span id='topic+findApdMap'></span>

<h3>Description</h3>

<p>Search for APD map files in multiple directories.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
findApdMap(mapType=NULL, paths=NULL, pattern="[.](a|A)(p|P)(m|M)$", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="findApdMap_+3A_maptype">mapType</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string of the map type to search for.</p>
</td></tr>
<tr><td><code id="findApdMap_+3A_paths">paths</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> of paths to be searched.
The current directory is always searched at the beginning.
If <code><a href="base.html#topic+NULL">NULL</a></code>, default paths are searched.  For more details, see below.</p>
</td></tr>
<tr><td><code id="findApdMap_+3A_pattern">pattern</code></td>
<td>
<p>A regular expression file name pattern to match.</p>
</td></tr>
<tr><td><code id="findApdMap_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="R.utils.html#topic+findFiles">findFiles</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note, the current directory is always searched at the beginning.
This provides an easy way to override other files in the search path.
If <code>paths</code> is <code><a href="base.html#topic+NULL">NULL</a></code>, then a set of default paths are searched.
The default search path is consituted of:
</p>

<ol>
<li> <p><code>"."</code>
</p>
</li>
<li> <p><code>getOption("AFFX_APD_PATH")</code>
</p>
</li>
<li> <p><code>Sys.getenv("AFFX_APD_PATH")</code>
</p>
</li></ol>

<p>One of the easiest ways to set system variables for <span class="rlang"><b>R</b></span> is to
set them in an <code>.Renviron</code> file, see <code><a href="base.html#topic+Startup">Startup</a></code>
for more details.
</p>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+vector">vector</a></code> of the full pathnames of the files found.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='gtypeCelToPQ'>Function to immitate Affymetrix' gtype_cel_to_pq software</h2><span id='topic+gtypeCelToPQ.default'></span><span id='topic+gtypeCelToPQ'></span>

<h3>Description</h3>

<p>Function to immitate Affymetrix' gtype_cel_to_pq software.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
gtypeCelToPQ(filename, units=NULL, ..., cdf=NULL, nbrOfQuartets=NULL, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gtypeCelToPQ_+3A_filename">filename</code></td>
<td>
<p>The name of a CEL file.</p>
</td></tr>
<tr><td><code id="gtypeCelToPQ_+3A_units">units</code></td>
<td>
<p>Indices of CDF units to be returned.</p>
</td></tr>
<tr><td><code id="gtypeCelToPQ_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="affxparser.html#topic+readCelUnits">readCelUnits</a></code>.</p>
</td></tr>
<tr><td><code id="gtypeCelToPQ_+3A_cdf">cdf</code></td>
<td>
<p>A CDF <code><a href="base.html#topic+list">list</a></code> structure, the pathname of the CDF file, or
<code><a href="base.html#topic+NULL">NULL</a></code>.  If <code><a href="base.html#topic+NULL">NULL</a></code>, the CDF file corresponding to the chip type of
the CEL file is searched for using <code><a href="affxparser.html#topic+findCdf">findCdf</a></code>.</p>
</td></tr>
<tr><td><code id="gtypeCelToPQ_+3A_nbrofquartets">nbrOfQuartets</code></td>
<td>
<p>The number of probe quartets in the returned
<code><a href="base.html#topic+matrix">matrix</a></code>.</p>
</td></tr>
<tr><td><code id="gtypeCelToPQ_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an NxK <code><a href="base.html#topic+matrix">matrix</a></code> where N is the number of probesets (SNPs) and
K=4*Q where Q is the number of probe quartets (PMA,MMA,PMB,MMB).
The rownames corresponds to the probeset names.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] Affymetrix, <em>Genotyping Probe Set Structure</em>,
Developers' Network, White paper, 2005-2015.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gtypeCelToPQ">gtypeCelToPQ</a></code>().
<code><a href="affxparser.html#topic+applyCdfGroups">applyCdfGroups</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Scan for CEL files
files &lt;- list.files(pattern="[.](cel|CEL)$")

# Convert each to RAW file
for (file in files) {
  rawFile &lt;- gsub("[.][^.]*$", ".raw", file)
  file.remove(rawFile)
  cel &lt;- gtypeCelToPQ(file, verbose=TRUE)
  write.table(file=rawFile, cel, sep="\t", quote=FALSE)
}
</code></pre>

<hr>
<h2 id='Non-documented+20objects'>Non-documented objects</h2><span id='topic+Non-documented+20objects'></span>

<h3>Description</h3>

<p>This page contains aliases for all &quot;non-documented&quot; objects that
<code>R CMD check</code> detects in this package.
</p>
<p>Most of the are to be documented.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='readApd'>Reads an Affymetrix probe data (APD) file</h2><span id='topic+readApd.default'></span><span id='topic+readApd'></span>

<h3>Description</h3>

<p>Reads an Affymetrix probe data (APD) file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
readApd(filename, indices=NULL, readMap="byMapType", name=NULL, ..., verbose=FALSE,
  .checkArgs=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readApd_+3A_filename">filename</code></td>
<td>
<p>The filename of the APD file.</p>
</td></tr>
<tr><td><code id="readApd_+3A_indices">indices</code></td>
<td>
<p>An optional <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of cell (probe) indices
specifying what cells to read.  If <code><a href="base.html#topic+NULL">NULL</a></code>, all are read.</p>
</td></tr>
<tr><td><code id="readApd_+3A_readmap">readMap</code></td>
<td>
<p>A <code><a href="base.html#topic+vector">vector</a></code> remapping cell indices to file indices.
If <code>"byMapType"</code>, the read map of type according to APD header
will be search for and read.  It is much faster to specify the
read map explicitly compared with searching for it each time.
If <code><a href="base.html#topic+NULL">NULL</a></code>, no map is used.</p>
</td></tr>
<tr><td><code id="readApd_+3A_name">name</code></td>
<td>
<p>The name of the data field.
If <code><a href="base.html#topic+NULL">NULL</a></code>, the APD header <code>name</code> is used.  If not specified,
it defaults to <code>"intensities"</code>.</p>
</td></tr>
<tr><td><code id="readApd_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="readApd_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
<tr><td><code id="readApd_+3A_.checkargs">.checkArgs</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, arguments are validated, otherwise not.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To read one <em>large</em> contiguous block of elements is faster than
to read individual elements one by one.  For this reason, internally
more elements than requested may be read and therefore allocation more
memory than necessary.  This means, in worst case <code class="reqn">N</code> elements
may read allocation <code class="reqn">N*8</code> bytes of <span class="rlang"><b>R</b></span> memory, although only two
elements are queried.  However, to date even with the largest arrays
from Affymetrix this will still only require tens of megabytes of
<em>temporary</em> memory.  For instance, Affymetrix Mapping 100K arrays
holds 2,560,000 probes requiring 20Mb of temporary memory.
</p>


<h3>Value</h3>

<p>A named <code><a href="base.html#topic+list">list</a></code> with the two elements <code>header</code> and
<code>data</code>.  The header is in turn a <code><a href="base.html#topic+list">list</a></code> structure and
the second is a <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> holding the queried data.
</p>


<h3>Remapping indices</h3>

<p>Argument <code>readMap</code> can be used to remap indices.  For instance,
the indices of the probes can be reorder such that the probes within
a probeset is in a contiguous set of probe indices.  Then, given that
the values are stored in such an order, when reading complete probesets,
data will be access much faster from file than if the values were
scatter all over the file.
</p>
<p>Example of speed improvements.  Reading all 40000 values in units
1001 to 2000 of an Affymetrix Mapping 100K Xba chip is more than
10-30 times faster with mapping compared to without.
</p>


<h3>File format</h3>

<p>The file format of an APD file is identical to the file format of an
<code><a href="R.huge.html#topic+FileVector">FileVector</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+createApd">createApd</a></code>() and <code><a href="#topic+updateApd">updateApd</a></code>().
See also <code><a href="#topic+readApdHeader">readApdHeader</a></code>().
To create a cell-index read map from an CDF file, see
<code><a href="affxparser.html#topic+readCdfUnitsWriteMap">readCdfUnitsWriteMap</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: #See ?createApd for an example.</code></pre>

<hr>
<h2 id='readApdHeader'>Reads the header of an Affymetrix probe data (APD) file</h2><span id='topic+readApdHeader.default'></span><span id='topic+readApdHeader'></span>

<h3>Description</h3>

<p>Reads the header of an Affymetrix probe data (APD) file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
readApdHeader(filename, ..., verbose=FALSE, .checkArgs=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readApdHeader_+3A_filename">filename</code></td>
<td>
<p>The filename of the APD file.</p>
</td></tr>
<tr><td><code id="readApdHeader_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="readApdHeader_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
<tr><td><code id="readApdHeader_+3A_.checkargs">.checkArgs</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, arguments are validated, otherwise not.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The file format of an APD file is identical to the file format of an
<code><a href="R.huge.html#topic+FileVector">FileVector</a></code>.  Most elements of the APD header are stored
in the <code>comment</code> <code><a href="base.html#topic+character">character</a></code> string of the file vector's header.
The APD header <code>nbrOfProbes</code> is identical to the length of the
file vector, and is <em>not</em> stored in the above comment string.
</p>


<h3>Value</h3>

<p>A named <code><a href="base.html#topic+list">list</a></code>.
The <code><a href="base.html#topic+numeric">numeric</a></code> element <code>nbrOfProbes</code> is the number of probe values
available in the APD file.
The optional <code><a href="base.html#topic+character">character</a></code> element <code>name</code> specifies the name of
the APD vector.
The optional <code><a href="base.html#topic+character">character</a></code> element <code>chipType</code> specifies the
chip type, cf. the same field in <code><a href="affxparser.html#topic+readCelHeader">readCelHeader</a></code>.
The optional <code><a href="base.html#topic+character">character</a></code> element <code>maptype</code> specifies the type of
probe-index map for this APD file.  Its value can be used to find
the mapping file, see <code><a href="#topic+findApdMap">findApdMap</a></code>() and <code><a href="#topic+readApdMap">readApdMap</a></code>().
All other fields are optional and <code><a href="base.html#topic+character">character</a></code> values.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+readApd">readApd</a></code>().
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: #See ?createApd for an example.</code></pre>

<hr>
<h2 id='readApdMap'>Reads an APD probe map file</h2><span id='topic+readApdMap.default'></span><span id='topic+readApdMap'></span>

<h3>Description</h3>

<p>Reads an APD probe map file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
readApdMap(filename, path=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readApdMap_+3A_filename">filename</code></td>
<td>
<p>The filename of the APD file.</p>
</td></tr>
<tr><td><code id="readApdMap_+3A_path">path</code></td>
<td>
<p>The path to the APD file.</p>
</td></tr>
<tr><td><code id="readApdMap_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+readApd">readApd</a></code>().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named <code><a href="base.html#topic+list">list</a></code> with the two elements <code>header</code> and
<code>map</code>.  The header is in turn a <code><a href="base.html#topic+list">list</a></code> structure and
the second is a <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> holding the probe map indices.
</p>


<h3>File format</h3>

<p>The file format of an APD map file is identical to the file format
of an APD file, see <code><a href="#topic+readApd">readApd</a></code>().  The APD map file identified by
the name of the data defaults to <code>"map"</code>.  If not, an error
is thrown.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>To search for an APD map file, see <code><a href="#topic+findApdMap">findApdMap</a></code>().
To create a cell index map from an CDF file, see
<code><a href="affxparser.html#topic+readCdfUnitsWriteMap">readCdfUnitsWriteMap</a></code>.
Internally, <code><a href="#topic+readApd">readApd</a></code>() is used.
</p>

<hr>
<h2 id='readApdRectangle'>Reads a spatial subset of probe-level data from Affymetrix APD files</h2><span id='topic+readApdRectangle.default'></span><span id='topic+readApdRectangle'></span>

<h3>Description</h3>

<p>Reads a spatial subset of probe-level data from Affymetrix APD files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
readApdRectangle(filename, xrange=c(0, Inf), yrange=c(0, Inf), ..., asMatrix=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readApdRectangle_+3A_filename">filename</code></td>
<td>
<p>The pathname of the APD file.</p>
</td></tr>
<tr><td><code id="readApdRectangle_+3A_xrange">xrange</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length two giving the left
and right coordinates of the cells to be returned.</p>
</td></tr>
<tr><td><code id="readApdRectangle_+3A_yrange">yrange</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length two giving the top
and bottom coordinates of the cells to be returned.</p>
</td></tr>
<tr><td><code id="readApdRectangle_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+readApd">readApd</a></code>().</p>
</td></tr>
<tr><td><code id="readApdRectangle_+3A_asmatrix">asMatrix</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the APD data fields are returned as
matrices with element (1,1) corresponding to cell
(xrange[1],yrange[1]).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named <code><a href="base.html#topic+list">list</a></code> APD structure similar to what <code><a href="#topic+readApd">readApd</a></code>().
In addition, if <code>asMatrix</code> is <code><a href="base.html#topic+logical">TRUE</a></code>, the APD data fields
are returned as matrices, otherwise not.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>The <code><a href="#topic+readApd">readApd</a></code>() method is used internally.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Local functions
rotate270 &lt;- function(x, ...) {
  x &lt;- t(x)
  nc &lt;- ncol(x)
  if (nc &lt; 2) return(x)
  x[,nc:1,drop=FALSE]
}

# Scan current directory for APD files
files &lt;- list.files(pattern="[.](apd|APD)$")
files &lt;- files[!file.info(files)$isdir]
if (length(files) &gt; 0) {
  apdFile &lt;- files[1]
 
  # Read APD intensities in the upper left corner
  apd &lt;- readApdRectangle(apdFile, xrange=c(0,250), yrange=c(0,250))
  z &lt;- rotate270(apd$intensities)
  sub &lt;- paste("Chip type:", apd$header$chipType)
  image(z, col=gray.colors(256), axes=FALSE, main=apdFile, sub=sub)
  text(x=0, y=1, labels="(0,0)", adj=c(0,-0.7), cex=0.8, xpd=TRUE)
  text(x=1, y=0, labels="(250,250)", adj=c(1,1.2), cex=0.8, xpd=TRUE)
}
</code></pre>

<hr>
<h2 id='readApdUnits'>Reads Affymetrix probe data (APD) as units (probesets)</h2><span id='topic+readApdUnits.default'></span><span id='topic+readApdUnits'></span>

<h3>Description</h3>

<p>Reads Affymetrix probe data (APD) as units (probesets) by using the unit and group definitions in the
corresponding Affymetrix CDF file.
</p>
<p>If more than one APD file is read, all files are assumed to be of
the same chip type, and have the same read map, if any.
It is not possible to read APD files of different types at the
same time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
readApdUnits(filenames, units=NULL, ..., transforms=NULL, cdf=NULL,
  stratifyBy=c("nothing", "pmmm", "pm", "mm"), addDimnames=FALSE, readMap="byMapType",
  dropArrayDim=TRUE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readApdUnits_+3A_filenames">filenames</code></td>
<td>
<p>The filenames of the APD files.  All APD files must
be of the same chip type.</p>
</td></tr>
<tr><td><code id="readApdUnits_+3A_units">units</code></td>
<td>
<p>An <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> of unit indices specifying which units
to be read.  If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are read.</p>
</td></tr>
<tr><td><code id="readApdUnits_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+readApd">readApd</a></code>().</p>
</td></tr>
<tr><td><code id="readApdUnits_+3A_transforms">transforms</code></td>
<td>
<p>A <code><a href="base.html#topic+list">list</a></code> of exactly <code>length(filenames)</code>
<code><a href="base.html#topic+function">function</a></code>s.  If <code><a href="base.html#topic+NULL">NULL</a></code>, no transformation is performed.
Values read are passed through the corresponding transform
function before being returned.</p>
</td></tr>
<tr><td><code id="readApdUnits_+3A_cdf">cdf</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> filename of a CDF file, or a CDF <code><a href="base.html#topic+list">list</a></code>
structure.  If <code><a href="base.html#topic+NULL">NULL</a></code>, the CDF file is searched for by
<code><a href="affxparser.html#topic+findCdf">findCdf</a></code> first starting from the current directory
and then from the directory where the first APD file is.</p>
</td></tr>
<tr><td><code id="readApdUnits_+3A_stratifyby">stratifyBy</code></td>
<td>
<p>Argument passed to low-level method
<code><a href="affxparser.html#topic+readCdfCellIndices">readCdfCellIndices</a></code>.</p>
</td></tr>
<tr><td><code id="readApdUnits_+3A_adddimnames">addDimnames</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, dimension names are added to arrays,
otherwise not.  The size of the returned APD structure in bytes
increases by 30-40% with dimension names.</p>
</td></tr>
<tr><td><code id="readApdUnits_+3A_readmap">readMap</code></td>
<td>
<p>A <code><a href="base.html#topic+vector">vector</a></code> remapping cell indices to file indices.
If <code>"byMapType"</code>, the read map of type according to APD header
will be search for and read.  It is much faster to specify the
read map explicitly compared with searching for it each time.
If <code><a href="base.html#topic+NULL">NULL</a></code>, no map is used.</p>
</td></tr>
<tr><td><code id="readApdUnits_+3A_droparraydim">dropArrayDim</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code> and only one array is read, the elements of
the group field do <em>not</em> have an array dimension.</p>
</td></tr>
<tr><td><code id="readApdUnits_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named <code><a href="base.html#topic+list">list</a></code> where the names corresponds to the names of the units
read.  Each element of the <code><a href="base.html#topic+list">list</a></code> is in turn a <code><a href="base.html#topic+list">list</a></code> structure
with groups (aka blocks).
</p>


<h3>Speed</h3>

<p>Since the cell indices are semi-randomized across the array and
with units (probesets), it is very unlikely that the read will
consist of subsequent cells (which would be faster to read).
However, the speed of this method, which uses <code><a href="R.huge.html#topic+FileVector">FileVector</a></code>
to read data, is comparable to the speed of
<code><a href="affxparser.html#topic+readCelUnits">readCelUnits</a></code>, which uses the Fusion SDK
(<code><a href="affxparser.html#topic+readCel">readCel</a></code>) to read data.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>To read CEL units, <code><a href="affxparser.html#topic+readCelUnits">readCelUnits</a></code>.
Internally, the <code><a href="#topic+readApd">readApd</a></code>() method is used for read probe data,
and <code><a href="#topic+readApdMap">readApdMap</a></code>(), if APD file has a map type specified and
the read map was not given explicitly.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library("R.utils") # Arguments

verbose &lt;- Arguments$getVerbose(TRUE)

# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# 1. Scan for existing CEL files
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# a) Scan current directory for CEL files
files &lt;- list.files(pattern="[.](cel|CEL)$")
files &lt;- files[!file.info(files)$isdir]

if (length(files) &gt; 0 &amp;&amp; require("affxparser")) {
  # b) Corresponding APD filenames
  celNames &lt;- files
  apdNames &lt;- gsub(pattern, ".apd", files)
 
  # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
  # 1. Copy the probe intensities from a CEL to an APD file
  # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
  for (kk in 1) {
    verbose &amp;&amp; enter(verbose, "Reading CEL file #", kk)
    cel &lt;- readCel(celNames[kk])
    verbose &amp;&amp; exit(verbose)
 
    if (!file.exists(apdNames[kk])) {
      verbose &amp;&amp; enter(verbose, "Creating APD file #", kk)
      chipType &lt;- cel$header$chiptype
      writeApd(apdNames[kk], data=cel$intensities, chipType=chipType)
      verbose &amp;&amp; exit(verbose)
    }
 
    verbose &amp;&amp; enter(verbose, "Verifying APD file #", kk)
    apd &lt;- readApd(apdNames[kk])
    verbose &amp;&amp; exit(verbose)
    stopifnot(identical(apd$intensities, cel$intensities))
  }
 
 
  # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
  # 2. Read a subset of the units
  # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
  units &lt;- c(1, 20:205)
  cel &lt;- readCelUnits(celNames[1], units=units)
  apd &lt;- readApdUnits(apdNames[1], units=units)
  stopifnot(identical(apd, cel))
 
 
  # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
  # 3. The same, but stratified on PMs and MMs
  # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
  apd &lt;- readApdUnits(apdNames[1], units=units, stratifyBy="pmmm",
                                                addDimnames=TRUE)
} # if (length(files) &gt; 0)
</code></pre>

<hr>
<h2 id='updateApd'>Updates an Affymetrix probe data (APD) file</h2><span id='topic+updateApd.default'></span><span id='topic+updateApd'></span>

<h3>Description</h3>

<p>Updates an Affymetrix probe data (APD) file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
updateApd(filename, indices=NULL, data, writeMap=NULL, ..., verbose=FALSE,
  .checkArgs=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="updateApd_+3A_filename">filename</code></td>
<td>
<p>The filename of the APD file.</p>
</td></tr>
<tr><td><code id="updateApd_+3A_indices">indices</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of cell (probe) indices specifying
which cells to updated.</p>
</td></tr>
<tr><td><code id="updateApd_+3A_data">data</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of data elements to be assigned.</p>
</td></tr>
<tr><td><code id="updateApd_+3A_writemap">writeMap</code></td>
<td>
<p>A <code><a href="base.html#topic+vector">vector</a></code> of indicies used to change the order how
data elements are <em>written</em> (by default).</p>
</td></tr>
<tr><td><code id="updateApd_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="updateApd_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
<tr><td><code id="updateApd_+3A_.checkargs">.checkArgs</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, arguments are checked, otherwise not.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns (invisibly) the pathname of the file updated.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+createApd">createApd</a></code>() and <code><a href="#topic+updateApd">updateApd</a></code>().
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: #See ?createApd for an example.</code></pre>

<hr>
<h2 id='updateApdHeader'>Updates the header of an Affymetrix probe data (APD) file</h2><span id='topic+updateApdHeader.default'></span><span id='topic+updateApdHeader'></span>

<h3>Description</h3>

<p>Updates the header of an Affymetrix probe data (APD) file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
updateApdHeader(filename, path=NULL, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="updateApdHeader_+3A_filename">filename</code></td>
<td>
<p>The filename of the APD file.</p>
</td></tr>
<tr><td><code id="updateApdHeader_+3A_path">path</code></td>
<td>
<p>The path to the APD file.</p>
</td></tr>
<tr><td><code id="updateApdHeader_+3A_...">...</code></td>
<td>
<p>A set of named header values to be updated/added to the
header.  A value of <code><a href="base.html#topic+NULL">NULL</a></code> will be removed from the current header.</p>
</td></tr>
<tr><td><code id="updateApdHeader_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns (invisibly) the pathname of the file updated.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+createApd">createApd</a></code>() and <code><a href="#topic+updateApd">updateApd</a></code>().
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: #See ?createApd for an example.</code></pre>

<hr>
<h2 id='updateApdUnits'>Updates an Affymetrix probe data (APD) file by units (probesets)</h2><span id='topic+updateApdUnits.default'></span><span id='topic+updateApdUnits'></span>

<h3>Description</h3>

<p>Updates an Affymetrix probe data (APD) file by units (probesets) by using the unit and group definitions in the
corresponding Affymetrix CDF file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
updateApdUnits(filename, units=NULL, data, ..., cdf=NULL, stratifyBy=c("nothing", "pmmm",
  "pm", "mm"), verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="updateApdUnits_+3A_filename">filename</code></td>
<td>
<p>The filename of the APD file.</p>
</td></tr>
<tr><td><code id="updateApdUnits_+3A_units">units</code></td>
<td>
<p>An <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> of unit indices specifying which
units to be read.  If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are updated.</p>
</td></tr>
<tr><td><code id="updateApdUnits_+3A_data">data</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of data elements to be assigned.</p>
</td></tr>
<tr><td><code id="updateApdUnits_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+updateApd">updateApd</a></code>(),
e.g. <code>writeMap</code>.</p>
</td></tr>
<tr><td><code id="updateApdUnits_+3A_cdf">cdf</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> filename of a CDF file, or a CDF <code><a href="base.html#topic+list">list</a></code>
structure.  If <code><a href="base.html#topic+NULL">NULL</a></code>, the CDF file is searched for by
<code><a href="affxparser.html#topic+findCdf">findCdf</a></code>.</p>
</td></tr>
<tr><td><code id="updateApdUnits_+3A_stratifyby">stratifyBy</code></td>
<td>
<p>Argument passed to low-level method
<code><a href="affxparser.html#topic+readCdfCellIndices">readCdfCellIndices</a></code>.</p>
</td></tr>
<tr><td><code id="updateApdUnits_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns nothing.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+readApdUnits">readApdUnits</a></code>() to read unit by units.
<code><a href="#topic+updateApd">updateApd</a></code>() to update cell by cell.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: #See ?createApd for an example.</code></pre>

<hr>
<h2 id='writeApd'>Writes an APD probe data file</h2><span id='topic+writeApd.default'></span><span id='topic+writeApd'></span>

<h3>Description</h3>

<p>Writes an APD probe data file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
writeApd(filename, data, ..., writeMap=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writeApd_+3A_filename">filename</code></td>
<td>
<p>The filename of the APD file.</p>
</td></tr>
<tr><td><code id="writeApd_+3A_data">data</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of elements to be written.</p>
</td></tr>
<tr><td><code id="writeApd_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+createApd">createApd</a></code>(), e.g. <code>chipType</code>,
<code>mapType</code> etc.</p>
</td></tr>
<tr><td><code id="writeApd_+3A_writemap">writeMap</code></td>
<td>
<p>A <code><a href="base.html#topic+vector">vector</a></code> of indicies used to change the order how
data elements are <em>written</em> (by default).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns (invisibly) the pathname to the created file.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>To create an APD map file, see <code><a href="#topic+readApdMap">readApdMap</a></code>().
</p>

<hr>
<h2 id='writeApdMap'>Writes an APD probe map file</h2><span id='topic+writeApdMap.default'></span><span id='topic+writeApdMap'></span>

<h3>Description</h3>

<p>Writes an APD probe map file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
writeApdMap(filename, path=NULL, map, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writeApdMap_+3A_filename">filename</code></td>
<td>
<p>The filename of the APD file.</p>
</td></tr>
<tr><td><code id="writeApdMap_+3A_path">path</code></td>
<td>
<p>The path to the APD file.</p>
</td></tr>
<tr><td><code id="writeApdMap_+3A_map">map</code></td>
<td>
<p>A <code><a href="base.html#topic+vector">vector</a></code> of indicies.</p>
</td></tr>
<tr><td><code id="writeApdMap_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+writeApd">writeApd</a></code>().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns (invisibly) the pathname to the create file.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>To read an APD map file, see <code><a href="#topic+readApdMap">readApdMap</a></code>().
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
