<!DOCTYPE html><html><head><title>Help for package aroma.cn</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {aroma.cn}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#aroma.cn-package'><p>Package aroma.cn</p></a></li>
<li><a href='#AbstractCurveNormalization'><p>The AbstractCurveNormalization class</p></a></li>
<li><a href='#callAllelicBalanceByBAFs.PairedPSCBS'><p>Calls regions that are in allelic balance</p></a></li>
<li><a href='#callCopyNeutralRegions.PairedPSCBS'><p>Calls regions that are copy neutral</p></a></li>
<li><a href='#callPeaks.PeaksAndValleys'><p>Calls the peaks in peaks-and-valley estimates</p></a></li>
<li><a href='#callXXorXY.numeric'><p>Calls XX or XY from ChrX allele B fractions of a normal sample</p></a></li>
<li><a href='#cnr'><p>Simple creation of a CopyNumberRegions object</p></a></li>
<li><a href='#extractTupleOfDataFiles.MultiSourceCopyNumberNormalization'><p>Gets a list of data files for a particular name across several data sets</p></a></li>
<li><a href='#findAtomicAberrations.CopyNumberRegions'><p>Finds all possible atomic regions</p></a></li>
<li><a href='#fitOne.MultiSourceCopyNumberNormalization'><p>Fits the multi-source model for one sample</p></a></li>
<li><a href='#getAllNames.MultiSourceCopyNumberNormalization'><p>Gets the names of all unique samples across all sources</p></a></li>
<li><a href='#getFitAromaUgpFile.MultiSourceCopyNumberNormalization'><p>Gets the UGP file specifying the common set of loci to normalize at</p></a></li>
<li><a href='#getInputDataSets.MultiSourceCopyNumberNormalization'><p>Gets the list of data sets to be normalized</p></a></li>
<li><a href='#getSmoothedDataSets.MultiSourceCopyNumberNormalization'><p>Gets the data sets smoothed toward the UGP file</p></a></li>
<li><a href='#getSubsetToFit.MultiSourceCopyNumberNormalization'><p>Gets subset of (smoothing) units for fitting the model</p></a></li>
<li><a href='#MultiSourceCopyNumberNormalization'><p>The MultiSourceCopyNumberNormalization class</p></a></li>
<li><a href='#Non-documented objects'><p>Non-documented objects</p></a></li>
<li><a href='#normalizeBAFsByRegions.PairedPSCBS'><p>Normalizes allele B fractions (BAFs) based on region-based PSCN estimates</p></a></li>
<li><a href='#normalizeMirroredBAFsByRegions.matrix'><p>Normalizes region-level mirrored allele B fractions (mBAFs)</p></a></li>
<li><a href='#normalizePrincipalCurve.matrix'><p>Normalizes data in K dimensions using principal curves</p></a></li>
<li><a href='#PairedPscbsModel'><p>The PairedPscbsModel class</p></a></li>
<li><a href='#PrincipalCurveNormalization'><p>The PrincipalCurveNormalization class</p></a></li>
<li><a href='#process.MultiSourceCopyNumberNormalization'><p>Normalizes all samples</p></a></li>
<li><a href='#testAllelicBalanceByBAFs'><p>Tests for allelic balance in a genomic region</p></a></li>
<li><a href='#TotalCnBinnedSmoothing'><p>The TotalCnBinnedSmoothing class</p></a></li>
<li><a href='#TotalCnKernelSmoothing'><p>The TotalCnKernelSmoothing class</p></a></li>
<li><a href='#TotalCnSmoothing'><p>The abstract TotalCnSmoothing class</p></a></li>
<li><a href='#TumorBoostNormalization'><p>The TumorBoostNormalization class</p></a></li>
<li><a href='#XYCurveNormalization'><p>The XYCurveNormalization class</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.7.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.2), R.utils (&ge; 2.11.0), aroma.core (&ge; 3.2.2)</td>
</tr>
<tr>
<td>Imports:</td>
<td>R.methodsS3 (&ge; 1.8.1), R.oo (&ge; 1.24.0), R.filesets (&ge;
2.14.0), R.cache (&ge; 0.15.0), matrixStats (&ge; 0.61.0), PSCBS
(&ge; 0.65.0), future.apply</td>
</tr>
<tr>
<td>Suggests:</td>
<td>aroma.light (&ge; 2.2.1), DNAcopy (&ge; 1.40.0), GLAD (&ge; 1.12.0)</td>
</tr>
<tr>
<td>SuggestsNote:</td>
<td>BioC (&gt;= 3.0), Recommended: aroma.light, DNAcopy</td>
</tr>
<tr>
<td>Title:</td>
<td>Copy-Number Analysis of Large Microarray Data Sets</td>
</tr>
<tr>
<td>Description:</td>
<td>Methods for analyzing DNA copy-number data.  Specifically,
  this package implements the multi-source copy-number normalization (MSCN)
  method for normalizing copy-number data obtained on various platforms and
  technologies.  It also implements the TumorBoost method for normalizing
  paired tumor-normal SNP data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-2.1">LGPL-2.1</a> | <a href="https://www.r-project.org/Licenses/LGPL-3">LGPL-3</a> [expanded from: LGPL (&ge; 2.1)]</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>TRUE</td>
</tr>
<tr>
<td>biocViews:</td>
<td>ProprietaryPlatforms, aCGH, CopyNumberVariants, SNP,
Microarray, OneChannel, TwoChannel, DataImport,
DataRepresentation, Preprocessing, QualityControl</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://www.aroma-project.org/">https://www.aroma-project.org/</a>,
<a href="https://github.com/HenrikBengtsson/aroma.cn">https://github.com/HenrikBengtsson/aroma.cn</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/HenrikBengtsson/aroma.cn/issues">https://github.com/HenrikBengtsson/aroma.cn/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-17 19:57:52 UTC; henrik</td>
</tr>
<tr>
<td>Author:</td>
<td>Henrik Bengtsson [aut, cre, cph],
  Pierre Neuvial [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Henrik Bengtsson &lt;henrikb@braju.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-17 20:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='aroma.cn-package'>Package aroma.cn</h2><span id='topic+aroma.cn-package'></span><span id='topic+aroma.cn'></span>

<h3>Description</h3>

<p>Methods for analyzing DNA copy-number data.  Specifically,   this package implements the multi-source copy-number normalization (MSCN)   method for normalizing copy-number data obtained on various platforms and   technologies.  It also implements the TumorBoost method for normalizing   paired tumor-normal SNP data.
</p>
<p>This package should be considered to be in an alpha or beta phase.
You should expect the API to be changing over time.
</p>


<h3>Installation and updates</h3>

<p>To install this package, call <code>install.packages("aroma.cn")</code>.
</p>


<h3>To get started</h3>

<p>To get started, see:
</p>

<ol>
<li><p> ...
</p>
</li></ol>



<h3>License</h3>

<p>The releases of this package is licensed under
LGPL version 2.1 or newer.
</p>
<p>The development code of the packages is under a private licence
(where applicable) and patches sent to the author fall under the
latter license, but will be, if incorporated, released under the
&quot;release&quot; license above.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson, Pierre Neuvial</p>


<h3>References</h3>

<p>Please cite PSCBS using one or more of the following references:
</p>
<p>A.B. Olshen, H. Bengtsson, P. Neuvial, P.T. Spellman, R.A. Olshen,
V.E. Seshan. Parent-specific copy number in paired tumor-normal
studies using circular binary segmentation, Bioinformatics, 2011
</p>
<p>Please cite PSCBS using one or more of the following references:
</p>
<p>H. Bengtsson, P. Neuvial and T.P. Speed. TumorBoost: Normalization of
allele-specific tumor copy numbers from a single pair of tumor-normal
genotyping microarrays, BMC Bioinformatics, 2010
</p>
<p>H. Bengtsson, A. Ray, P. Spellman and T.P. Speed. A single-sample
method for normalizing and combining full-resolutioncopy numbers from
multiple platforms, labs and analysis methods, Bioinformatics, 2009
</p>
<p>H. Bengtsson; K. Simpson; J. Bullard; K. Hansen. aroma.affymetrix: A
generic framework in R for analyzing small to very large Affymetrix
data sets in bounded memory, Tech Report 745, Department of Statistics,
University of California, Berkeley, February 2008
</p>
<p>H. Bengtsson, R. Irizarry, B. Carvalho, &amp; T.P. Speed. Estimation and
assessment of raw copy numbers at the single locus level,
Bioinformatics, 2008
</p>
<p>To see these entries in BibTeX format, use 'print(&lt;citation&gt;,
bibtex=TRUE)', 'toBibtex(.)', or set
'options(citation.bibtex.max=999)'.
</p>

<hr>
<h2 id='AbstractCurveNormalization'>The AbstractCurveNormalization class</h2><span id='topic+AbstractCurveNormalization'></span>

<h3>Description</h3>

<p>Package:  aroma.cn <br />
<b>Class AbstractCurveNormalization</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><em><code>AbstractCurveNormalization</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+PrincipalCurveNormalization">PrincipalCurveNormalization</a>, <a href="#topic+XYCurveNormalization">XYCurveNormalization</a><br />
</p>
<p>public abstract static class <b>AbstractCurveNormalization</b><br />
extends <a href="R.oo.html#topic+Object">Object</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AbstractCurveNormalization(dataSet=NULL, targetSet=NULL, subsetToFit=NULL, tags="*",
  copyTarget=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AbstractCurveNormalization_+3A_dataset">dataSet</code></td>
<td>
<p>An <code><a href="aroma.core.html#topic+AromaUnitTotalCnBinarySet">AromaUnitTotalCnBinarySet</a></code> of
&quot;test&quot; samples to be normalized.</p>
</td></tr>
<tr><td><code id="AbstractCurveNormalization_+3A_targetset">targetSet</code></td>
<td>
<p>An <code><a href="aroma.core.html#topic+AromaUnitTotalCnBinarySet">AromaUnitTotalCnBinarySet</a></code> of
paired target samples.</p>
</td></tr>
<tr><td><code id="AbstractCurveNormalization_+3A_subsettofit">subsetToFit</code></td>
<td>
<p>The subset of loci to be used to fit the
normalization functions.
If <code><a href="base.html#topic+NULL">NULL</a></code>, loci on chromosomes 1-22 are used, but not on ChrX and ChrY.
</p>
</td></tr>
<tr><td><code id="AbstractCurveNormalization_+3A_tags">tags</code></td>
<td>
<p>(Optional) Sets the tags for the output data sets.</p>
</td></tr>
<tr><td><code id="AbstractCurveNormalization_+3A_copytarget">copyTarget</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, target arrays are copied to the output
data set, otherwise not.</p>
</td></tr>
<tr><td><code id="AbstractCurveNormalization_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFullName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getInputDataSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getOutputDataSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getTags</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getTargetDataSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setTags</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='callAllelicBalanceByBAFs.PairedPSCBS'>Calls regions that are in allelic balance</h2><span id='topic+callAllelicBalanceByBAFs.PairedPSCBS'></span><span id='topic+PairedPSCBS.callAllelicBalanceByBAFs'></span><span id='topic+callAllelicBalanceByBAFs+2CPairedPSCBS-method'></span>

<h3>Description</h3>

<p>Calls regions that are in allelic balance from the allele B fractions (BAF).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PairedPSCBS'
callAllelicBalanceByBAFs(fit, maxScore="auto", ..., force=FALSE, cache=FALSE,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="callAllelicBalanceByBAFs.PairedPSCBS_+3A_fit">fit</code></td>
<td>
<p>A <code><a href="PSCBS.html#topic+PairedPSCBS">PairedPSCBS</a></code> fit object as returned by
<code><a href="PSCBS.html#topic+segmentByPairedPSCBS">segmentByPairedPSCBS</a></code>.</p>
</td></tr>
<tr><td><code id="callAllelicBalanceByBAFs.PairedPSCBS_+3A_maxscore">maxScore</code></td>
<td>
<p>A positive <code><a href="base.html#topic+double">double</a></code> threshold.
If <code>"auto"</code>, the threshold is estimated empirically.</p>
</td></tr>
<tr><td><code id="callAllelicBalanceByBAFs.PairedPSCBS_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="callAllelicBalanceByBAFs.PairedPSCBS_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, an already called object is skipped, otherwise not.</p>
</td></tr>
<tr><td><code id="callAllelicBalanceByBAFs.PairedPSCBS_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="PSCBS.html#topic+PairedPSCBS">PairedPSCBS</a></code> fit object
where columns for allelic imbalance scores and
p-values as well as allelic balance calls are added.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson, Pierre Neuvial</p>


<h3>See Also</h3>

<p>Internally, <code><a href="#topic+testAllelicBalanceByBAFs">testAllelicBalanceByBAFs</a></code>() is used.
</p>
<p>Note that this AB caller differs from the default one in the
<span class="pkg">PSCBS</span> package, cf. <code><a href="PSCBS.html#topic+callAB.PairedPSCBS">callAB.PairedPSCBS</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (Sys.getenv("_R_CHECK_FULL_") != "" &amp;&amp; require("PSCBS")) {

# Load example ASCN data
data &lt;- PSCBS::exampleData("paired.chr01");

# PSCBS segmentation
fit &lt;- segmentByPairedPSCBS(data, verbose=-10);

# Normalize
fitN &lt;- normalizeBAFsByRegions(fit, verbose=-10);

fitN2 &lt;- callAllelicBalanceByBAFs(fitN, verbose=-10);
print(fitN2);

} # if (require("PSCBS"))
</code></pre>

<hr>
<h2 id='callCopyNeutralRegions.PairedPSCBS'>Calls regions that are copy neutral</h2><span id='topic+callCopyNeutralRegions.PairedPSCBS'></span><span id='topic+PairedPSCBS.callCopyNeutralRegions'></span><span id='topic+callCopyNeutralRegions+2CPairedPSCBS-method'></span>

<h3>Description</h3>

<p>Calls regions that are copy neutral from the allele B fractions (BAF).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PairedPSCBS'
callCopyNeutralRegions(fit, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="callCopyNeutralRegions.PairedPSCBS_+3A_fit">fit</code></td>
<td>
<p>A <code><a href="PSCBS.html#topic+PairedPSCBS">PairedPSCBS</a></code> fit object as returned by
<code><a href="PSCBS.html#topic+segmentByPairedPSCBS">segmentByPairedPSCBS</a></code>.</p>
</td></tr>
<tr><td><code id="callCopyNeutralRegions.PairedPSCBS_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to
<code><a href="PSCBS.html#topic+callCopyNeutral.PairedPSCBS">callCopyNeutral.PairedPSCBS</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="PSCBS.html#topic+PairedPSCBS">PairedPSCBS</a></code> fit object
where a column with the copy-neutral call.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (Sys.getenv("_R_CHECK_FULL_") != "" &amp;&amp; require("PSCBS")) {

# Load example ASCN data
data &lt;- PSCBS::exampleData("paired.chr01");

# PSCBS segmentation
fit &lt;- segmentByPairedPSCBS(data, verbose=-10);

fit &lt;- bootstrapTCNandDHByRegion(fit, verbose=-10);

fitC &lt;- callCopyNeutralRegions(fit, verbose=-10);
print(fitC);

# Normalize
fitN &lt;- normalizeBAFsByRegions(fitC, verbose=-10);

} # if (require("PSCBS"))
</code></pre>

<hr>
<h2 id='callPeaks.PeaksAndValleys'>Calls the peaks in peaks-and-valley estimates</h2><span id='topic+callPeaks.PeaksAndValleys'></span><span id='topic+PeaksAndValleys.callPeaks'></span><span id='topic+callPeaks+2CPeaksAndValleys-method'></span>

<h3>Description</h3>

<p>Calls the peaks in peaks-and-valley estimates to a set of known state.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PeaksAndValleys'
callPeaks(fit, expected=c(-1/2, -1/4, 0, +1/4, +1/2) * pi, flavor=c("decreasing", "all"),
  verbose=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="callPeaks.PeaksAndValleys_+3A_fit">fit</code></td>
<td>
<p>A KxC <code><a href="base.html#topic+data.frame">data.frame</a></code> of peaks-and-valley estimates.</p>
</td></tr>
<tr><td><code id="callPeaks.PeaksAndValleys_+3A_expected">expected</code></td>
<td>
<p>The expected locations of the peaks to be called.</p>
</td></tr>
<tr><td><code id="callPeaks.PeaksAndValleys_+3A_flavor">flavor</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying what flavor of the
caller to use.</p>
</td></tr>
<tr><td><code id="callPeaks.PeaksAndValleys_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> object.</p>
</td></tr>
<tr><td><code id="callPeaks.PeaksAndValleys_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a Kx(C+2) <code><a href="base.html#topic+data.frame">data.frame</a></code>.
</p>


<h3>Flavors</h3>

<p>If <code>flavor == "all"</code>, each peak is called to the state with the
closest expected value.
If <code>flavor == "decreasing"</code>, the strongest peak is called to the
state with the closest expected value, then the second strongest peak
is called analogously to one of the remaining states, and so on.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson, Pierre Neuvial</p>


<h3>See Also</h3>

<p>To get peaks-and-valley estimates, use
<code><a href="aroma.light.html#topic+findPeaksAndValleys">findPeaksAndValleys</a></code>.
</p>

<hr>
<h2 id='callXXorXY.numeric'>Calls XX or XY from ChrX allele B fractions of a normal sample</h2><span id='topic+callXXorXY.numeric'></span><span id='topic+callXXorXY'></span>

<h3>Description</h3>

<p>Calls XX or XY from ChrX allele B fractions of a normal sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'numeric'
callXXorXY(betaX, betaY=NULL, flavor=c("density"), adjust=1.5, ...,
  censorAt=c(-0.5, +1.5), verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="callXXorXY.numeric_+3A_betax">betaX</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> containing ChrX allele B fractions.</p>
</td></tr>
<tr><td><code id="callXXorXY.numeric_+3A_betay">betaY</code></td>
<td>
<p>A optional <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> containing ChrY allele B fractions.</p>
</td></tr>
<tr><td><code id="callXXorXY.numeric_+3A_flavor">flavor</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying the type of algorithm used.</p>
</td></tr>
<tr><td><code id="callXXorXY.numeric_+3A_adjust">adjust</code></td>
<td>
<p>A postive <code><a href="base.html#topic+double">double</a></code> specifying the amount smoothing for
the empirical density estimator.</p>
</td></tr>
<tr><td><code id="callXXorXY.numeric_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to
<code><a href="aroma.light.html#topic+findPeaksAndValleys">findPeaksAndValleys</a></code>.</p>
</td></tr>
<tr><td><code id="callXXorXY.numeric_+3A_censorat">censorAt</code></td>
<td>
<p>A <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length two specifying the range
for which values are considered finite.  Values below (above) this
range are treated as -<code><a href="base.html#topic+is.finite">Inf</a></code> (+<code><a href="base.html#topic+is.finite">Inf</a></code>).</p>
</td></tr>
<tr><td><code id="callXXorXY.numeric_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a ...
</p>


<h3>Missing and non-finite values</h3>

<p>Missing and non-finite values are dropped before trying to call XX or XY.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson, Pierre Neuvial</p>


<h3>See Also</h3>

<p>Internally <code><a href="aroma.light.html#topic+findPeaksAndValleys">findPeaksAndValleys</a></code> is used to identify
the thresholds.
</p>

<hr>
<h2 id='cnr'>Simple creation of a CopyNumberRegions object</h2><span id='topic+cnr'></span>

<h3>Description</h3>

<p>Simple creation of a CopyNumberRegions object containing a single region.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cnr(start, stop=NULL, mean=1, chromosome=1, xScale=1e+06, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cnr_+3A_start">start</code>, <code id="cnr_+3A_stop">stop</code></td>
<td>
<p>Two <code><a href="base.html#topic+numeric">numeric</a></code>s specifying the start and stop
location of the region.  If <code>stop</code> is <code><a href="base.html#topic+NULL">NULL</a></code>, then
<code>start[2]</code> is used instead.</p>
</td></tr>
<tr><td><code id="cnr_+3A_mean">mean</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> specifying the mean level of the region.</p>
</td></tr>
<tr><td><code id="cnr_+3A_chromosome">chromosome</code></td>
<td>
<p>An <code><a href="base.html#topic+integer">integer</a></code> specifying the chromosome ID.</p>
</td></tr>
<tr><td><code id="cnr_+3A_xscale">xScale</code></td>
<td>
<p>The default scaling parameter used for <code>start</code>
and <code>stop</code>, which is then also used for plotting genomic
locations.</p>
</td></tr>
<tr><td><code id="cnr_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to
<code><a href="aroma.core.html#topic+CopyNumberRegions">CopyNumberRegions</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return an <code><a href="aroma.core.html#topic+CopyNumberRegions">CopyNumberRegions</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>Examples</h3>

<pre><code class='language-R'># - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Build up a tumor CN profile
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
pT &lt;- cnr(1,1000, 2) +
     cnr(400,500) +
     cnr(600,800) +
     cnr(600,700) +
     cnr(100,200) - cnr(850,900);
print(pT);


# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Simulate CN signals from this profile
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
cn &lt;- simulateRawCopyNumbers(pT, n=2000, sd=1/2);
print(cn);


# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Plot
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
plot(cn, col="#aaaaaa", ylim=c(0,5));
drawLevels(pT, col="#ff0000", lwd=3);
</code></pre>

<hr>
<h2 id='extractTupleOfDataFiles.MultiSourceCopyNumberNormalization'>Gets a list of data files for a particular name across several data sets</h2><span id='topic+extractTupleOfDataFiles.MultiSourceCopyNumberNormalization'></span><span id='topic+MultiSourceCopyNumberNormalization.extractTupleOfDataFiles'></span><span id='topic+extractTupleOfDataFiles+2CMultiSourceCopyNumberNormalization-method'></span>

<h3>Description</h3>

<p>Gets a list of data files for a particular name across several data sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MultiSourceCopyNumberNormalization'
extractTupleOfDataFiles(this, dsList, name, ..., na.rm=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractTupleOfDataFiles.MultiSourceCopyNumberNormalization_+3A_name">name</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying the sample name of interest.</p>
</td></tr>
<tr><td><code id="extractTupleOfDataFiles.MultiSourceCopyNumberNormalization_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="extractTupleOfDataFiles.MultiSourceCopyNumberNormalization_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> of K <code><a href="aroma.core.html#topic+AromaUnitTotalCnBinarySet">AromaUnitTotalCnBinarySet</a></code>:s.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+MultiSourceCopyNumberNormalization">MultiSourceCopyNumberNormalization</a></code>.
</p>

<hr>
<h2 id='findAtomicAberrations.CopyNumberRegions'>Finds all possible atomic regions</h2><span id='topic+findAtomicAberrations.CopyNumberRegions'></span><span id='topic+CopyNumberRegions.findAtomicAberrations'></span><span id='topic+findAtomicAberrations+2CCopyNumberRegions-method'></span><span id='topic+findAtomicAberrations'></span>

<h3>Description</h3>

<p>Finds all possible atomic regions of a certain length.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CopyNumberRegions'
findAtomicAberrations(cnr, data, H=1, alpha=0.02, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="findAtomicAberrations.CopyNumberRegions_+3A_cnr">cnr</code></td>
<td>
<p>The segments defining the partitioning of the data.</p>
</td></tr>
<tr><td><code id="findAtomicAberrations.CopyNumberRegions_+3A_data">data</code></td>
<td>
<p>The data  used to test for equality.</p>
</td></tr>
<tr><td><code id="findAtomicAberrations.CopyNumberRegions_+3A_h">H</code></td>
<td>
<p>A positive <code><a href="base.html#topic+integer">integer</a></code> specifying how many segments each
atomic abberation should contain.</p>
</td></tr>
<tr><td><code id="findAtomicAberrations.CopyNumberRegions_+3A_alpha">alpha</code></td>
<td>
<p>A <code><a href="base.html#topic+double">double</a></code> in [0,1] specifying the significance level
for testing the null-hypothesis that the flanking segments
are equal.</p>
</td></tr>
<tr><td><code id="findAtomicAberrations.CopyNumberRegions_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="findAtomicAberrations.CopyNumberRegions_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>An <em>aberration of length H</em> is defined as any H consecutive segments.
Each aberrations has two <em>flanking segments</em> on each side.
Regardless of the content of the aberration, it is possible
to test the null-hypothesis that the two flanking segments are
equal or not.
The two flanking regions are said to be <em>equal</em>, if the
null-hypothesis of being equal is <em>not</em> rejected.
If the two flanking regions are called equal, then the contained
abberation (of length H) is called <em>atomic</em>, otherwise not.
</p>
<p>For consistency one may also define atomic aberrations of length H=0.
Consider that an imaginary aberration of zero length splits a single
segment into to flanking segments.  Then by construction those two
segments are equal.  The case where H=0 is still not implemented.
</p>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+data.frame">data.frame</a></code> with K rows, where K &gt;= 0 is the number
of atomic aberrations found.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("aroma.cn")

# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Local functions
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
xAxisBar &lt;- function(x0, x1=NULL, y=par("usr")[3], lwd=4, col=par("col"),
                     length=0.04, xpd=FALSE, ...) {
  if (is.null(x1)) {
    x1 &lt;- x0[2]
    x0 &lt;- x0[1]
  }
  arrows(x0=x0, x1=x1, y0=y, code=3, angle=90, length=length, lwd=lwd,
         col=col, xpd=xpd, lend=2, ...)
} # xAxisBar()


verbose &lt;- Arguments$getVerbose(-8, timestamp=TRUE)

# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Simulating copy-number data
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Build up CN profile generation by generation
pT &lt;- cnr(1,2000, 2) +
     cnr(1000,1500) +
     cnr(1000,1250) +
     cnr(1650,1800) +
     cnr(200,300) - cnr(650,800)
print(pT)

# Simulate data from the track
cn &lt;- simulateRawCopyNumbers(pT, n=2000, sd=1/2)
print(cn)


# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Find atomic aberrations
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
subplots(4, ncol=1)
par(mar=c(2,4,1,1)+0.1)

# Plot observed and true signals
plot(cn, col="#aaaaaa", ylim=c(0,5))
drawLevels(pT, col="white", lwd=4, lty=1)
drawLevels(pT, col="black", lwd=3, lty=6)
stext(side=3, pos=1, line=-1, "\"H=0\"", cex=1.5)

for (H in 1:3) {
  plot(cn, col="#aaaaaa", ylim=c(0,5))
  drawLevels(pT, col="white", lwd=4, lty=1)
  drawLevels(pT, col="black", lwd=3, lty=6)
  col &lt;- H+1
  stext(side=3, pos=1, line=-1, sprintf("H=%d", H), cex=1.5, col=col)
  par &lt;- par("usr")
  y0 &lt;- par("usr")[3]
  y1 &lt;- par("usr")[4]
  dy &lt;- 0.05*(y1-y0)

  fit &lt;- findAtomicAberrations(cnr=pT, data=cn, H=H, verbose=verbose)
  df &lt;- fit$res

  for (kk in seq_len(nrow(df))) {
    dfKK &lt;- df[kk,]
    segments &lt;- as.integer(dfKK[,c("firstRegion", "lastRegion")])
    segments &lt;- segments[1]:segments[2]
    xRange &lt;- as.double(dfKK[,c("start", "stop")])
    cnrKK &lt;- subset(pT, subset=segments)
    drawLevels(cnrKK, col=col, lwd=3)
    x &lt;- xRange/1e6
    y &lt;- y0 + 0.2*dy
    xAxisBar(x0=x[1], x1=x[2], y=y, col=col)
    box()
  }
} # for (H ...)

</code></pre>

<hr>
<h2 id='fitOne.MultiSourceCopyNumberNormalization'>Fits the multi-source model for one sample</h2><span id='topic+fitOne.MultiSourceCopyNumberNormalization'></span><span id='topic+MultiSourceCopyNumberNormalization.fitOne'></span><span id='topic+fitOne+2CMultiSourceCopyNumberNormalization-method'></span>

<h3>Description</h3>

<p>Fits the multi-source model for one sample.
The model is fitted on the subset of units returned
by <code><a href="#topic+getSubsetToFit.MultiSourceCopyNumberNormalization">*getSubsetToFit</a>()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MultiSourceCopyNumberNormalization'
fitOne(this, dfList, ..., force=FALSE, .retData=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fitOne.MultiSourceCopyNumberNormalization_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="fitOne.MultiSourceCopyNumberNormalization_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">FALSE</a></code>, cached model fits are returned, otherwise not.</p>
</td></tr>
<tr><td><code id="fitOne.MultiSourceCopyNumberNormalization_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> of transforms.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+MultiSourceCopyNumberNormalization">MultiSourceCopyNumberNormalization</a></code>.
</p>

<hr>
<h2 id='getAllNames.MultiSourceCopyNumberNormalization'>Gets the names of all unique samples across all sources</h2><span id='topic+getAllNames.MultiSourceCopyNumberNormalization'></span><span id='topic+MultiSourceCopyNumberNormalization.getAllNames'></span><span id='topic+getAllNames+2CMultiSourceCopyNumberNormalization-method'></span>

<h3>Description</h3>

<p>Gets the names of all unique samples across all sources.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MultiSourceCopyNumberNormalization'
getAllNames(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAllNames.MultiSourceCopyNumberNormalization_+3A_...">...</code></td>
<td>
<p>Passed to <code>getNames(...)</code> of each data set.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+MultiSourceCopyNumberNormalization">MultiSourceCopyNumberNormalization</a></code>.
</p>

<hr>
<h2 id='getFitAromaUgpFile.MultiSourceCopyNumberNormalization'>Gets the UGP file specifying the common set of loci to normalize at</h2><span id='topic+getFitAromaUgpFile.MultiSourceCopyNumberNormalization'></span><span id='topic+MultiSourceCopyNumberNormalization.getFitAromaUgpFile'></span><span id='topic+getFitAromaUgpFile+2CMultiSourceCopyNumberNormalization-method'></span>

<h3>Description</h3>

<p>Gets the UGP file specifying the common set of loci to normalize at.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MultiSourceCopyNumberNormalization'
getFitAromaUgpFile(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getFitAromaUgpFile.MultiSourceCopyNumberNormalization_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="aroma.core.html#topic+AromaUgpFile">AromaUgpFile</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+MultiSourceCopyNumberNormalization">MultiSourceCopyNumberNormalization</a></code>.
</p>

<hr>
<h2 id='getInputDataSets.MultiSourceCopyNumberNormalization'>Gets the list of data sets to be normalized</h2><span id='topic+getInputDataSets.MultiSourceCopyNumberNormalization'></span><span id='topic+MultiSourceCopyNumberNormalization.getInputDataSets'></span><span id='topic+getInputDataSets+2CMultiSourceCopyNumberNormalization-method'></span>

<h3>Description</h3>

<p>Gets the list of data sets to be normalized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MultiSourceCopyNumberNormalization'
getInputDataSets(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getInputDataSets.MultiSourceCopyNumberNormalization_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+MultiSourceCopyNumberNormalization">MultiSourceCopyNumberNormalization</a></code>.
</p>

<hr>
<h2 id='getSmoothedDataSets.MultiSourceCopyNumberNormalization'>Gets the data sets smoothed toward the UGP file</h2><span id='topic+getSmoothedDataSets.MultiSourceCopyNumberNormalization'></span><span id='topic+MultiSourceCopyNumberNormalization.getSmoothedDataSets'></span><span id='topic+getSmoothedDataSets+2CMultiSourceCopyNumberNormalization-method'></span>

<h3>Description</h3>

<p>Gets the data sets smoothed toward the UGP file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MultiSourceCopyNumberNormalization'
getSmoothedDataSets(this, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSmoothedDataSets.MultiSourceCopyNumberNormalization_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="getSmoothedDataSets.MultiSourceCopyNumberNormalization_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This method smooth each data set, each array, and each chromosome
toward the target (smoothing) UGP file independently of everything else.
</p>
<p>The resulting data sets are stored in a separate location where they
will be located automatically in future sessions.
</p>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> of K <code><a href="aroma.core.html#topic+AromaUnitTotalCnBinarySet">AromaUnitTotalCnBinarySet</a></code>:s.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+getFitAromaUgpFile.MultiSourceCopyNumberNormalization">*getFitAromaUgpFile</a>()</code>.
For more information see <code><a href="#topic+MultiSourceCopyNumberNormalization">MultiSourceCopyNumberNormalization</a></code>.
</p>

<hr>
<h2 id='getSubsetToFit.MultiSourceCopyNumberNormalization'>Gets subset of (smoothing) units for fitting the model</h2><span id='topic+getSubsetToFit.MultiSourceCopyNumberNormalization'></span><span id='topic+MultiSourceCopyNumberNormalization.getSubsetToFit'></span><span id='topic+getSubsetToFit+2CMultiSourceCopyNumberNormalization-method'></span>

<h3>Description</h3>

<p>Gets subset of (smoothing) units for fitting the model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MultiSourceCopyNumberNormalization'
getSubsetToFit(this, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSubsetToFit.MultiSourceCopyNumberNormalization_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="getSubsetToFit.MultiSourceCopyNumberNormalization_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> of unit indices.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+MultiSourceCopyNumberNormalization">MultiSourceCopyNumberNormalization</a></code>.
</p>

<hr>
<h2 id='MultiSourceCopyNumberNormalization'>The MultiSourceCopyNumberNormalization class</h2><span id='topic+MultiSourceCopyNumberNormalization'></span>

<h3>Description</h3>

<p>Package:  aroma.cn <br />
<b>Class MultiSourceCopyNumberNormalization</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code>MultiSourceCopyNumberNormalization</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>MultiSourceCopyNumberNormalization</b><br />
extends <a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a><br />
</p>
<p>The multi-source copy-number normalization (MSCN) method [1] is a
normalization method that normalizes copy-number estimates measured
by multiple sites and/or platforms for common samples.  It normalizes the
estimates toward a common scale such that for any copy-number level
the mean level of the normalized data are the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MultiSourceCopyNumberNormalization(dsList=NULL, fitUgp=NULL, subsetToFit=NULL,
  targetDimension=1, align=c("byChromosome", "none"), tags="*", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MultiSourceCopyNumberNormalization_+3A_dslist">dsList</code></td>
<td>
<p>A <code><a href="base.html#topic+list">list</a></code> of K <code><a href="aroma.core.html#topic+AromaUnitTotalCnBinarySet">AromaUnitTotalCnBinarySet</a></code>:s.</p>
</td></tr>
<tr><td><code id="MultiSourceCopyNumberNormalization_+3A_fitugp">fitUgp</code></td>
<td>
<p>An <code><a href="aroma.core.html#topic+AromaUgpFile">AromaUgpFile</a></code> that specifies the
common set of loci used to normalize the data sets at.</p>
</td></tr>
<tr><td><code id="MultiSourceCopyNumberNormalization_+3A_subsettofit">subsetToFit</code></td>
<td>
<p>The subset of loci (as mapped by the <code>fitUgp</code>
object) to be used to fit the normalization functions.
If <code><a href="base.html#topic+NULL">NULL</a></code>, loci on chromosomes 1-22 are used, but not on ChrX and ChrY.
</p>
</td></tr>
<tr><td><code id="MultiSourceCopyNumberNormalization_+3A_targetdimension">targetDimension</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> index specifying the data set in
<code>dsList</code> to which each platform in standardize towards.
If <code><a href="base.html#topic+NULL">NULL</a></code>, the arbitrary scale along the fitted principal curve
is used.  This always starts at zero and increases.</p>
</td></tr>
<tr><td><code id="MultiSourceCopyNumberNormalization_+3A_align">align</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> specifying type of alignment applied, if any.
If <code>"none"</code>, no alignment is done.
If <code>"byChromosome"</code>, the signals are shifted chromosome
by chromosome such the corresponding smoothed signals have the same
median signal across sources.
For more details, see below.
</p>
</td></tr>
<tr><td><code id="MultiSourceCopyNumberNormalization_+3A_tags">tags</code></td>
<td>
<p>(Optional) Sets the tags for the output data sets.</p>
</td></tr>
<tr><td><code id="MultiSourceCopyNumberNormalization_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The multi-source normalization method is by nature a single-sample method,
that is, it normalizes arrays for one sample at the time and independently
of all other samples/arrays.
</p>
<p>However, the current implementation is such that it first generates
smoothed data for <em>all</em> samples/arrays.  Then, it normalizes the
sample one by one.
</p>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAllNames</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAsteriskTags</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getInputDataSets</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getOutputDataSets</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getTags</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfDataSets</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Different preprocessing methods normalize ChrX &amp; ChrY differently</h3>

<p>Some preprocessing methods estimate copy numbers on sex chromosomes
differently from the autosomal chromosomes.  The way this is done may
vary from method to method and we cannot assume anything about what
approach is.  This is the main reason why the estimation of the
normalization  function is by default based on signals from autosomal
chromosomes only; this protects the estimate of the function from
being biased by specially estimated sex-chromosome signals.
Note that the normalization function is still applied to all chromosomes.
</p>
<p>This means that if the transformation applied by a particular
preprocessing method is not the same for the sex chromosomes as the
autosomal chromosomes, the normalization applied on the sex
chromosomes is not optimal one.  This is why multi-source
normalization sometimes fails to bring sex-chromosome signals
to the same scale across sources.  Unfortunately, there is no
automatic way to handle this.
The only way would be to fit a specific normalization function to each
of the sex chromosomes, but that would require that there exist
copy-number abberations on those chromosomes, which could be a too
strong assumption.
</p>
<p>A more conservative approach is to normalize the signals such that
afterward the median of the smoothed copy-number levels are the same
across sources for any particular chromosome.
This is done by setting argument <code>align="byChromosome"</code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] H. Bengtsson, A. Ray, P. Spellman &amp; T.P. Speed,
<em>A single-sample method for normalizing and combining
full-resolution copy numbers from multiple platforms,
labs and analysis methods</em>,
Bioinformatics 2009. <br />
</p>

<hr>
<h2 id='Non-documented+20objects'>Non-documented objects</h2><span id='topic+Non-documented+20objects'></span><span id='topic+as.character.MultiSourceCopyNumberNormalization'></span><span id='topic+extractTupleOfDataFiles'></span><span id='topic+fitOne'></span><span id='topic+getAllNames'></span><span id='topic+getAsteriskTags.TotalCnSmoothing'></span><span id='topic+getFitAromaUgpFile'></span><span id='topic+getInputDataSets'></span><span id='topic+getOutputDataSets'></span><span id='topic+getOutputDataSets.MultiSourceCopyNumberNormalization'></span><span id='topic+getOutputFiles.TotalCnSmoothing'></span><span id='topic+getOutputPaths'></span><span id='topic+getOutputPaths.MultiSourceCopyNumberNormalization'></span><span id='topic+getParameters.MultiSourceCopyNumberNormalization'></span><span id='topic+getParameters.TotalCnSmoothing'></span><span id='topic+getParametersAsString.MultiSourceCopyNumberNormalization'></span><span id='topic+getPath.TotalCnSmoothing'></span><span id='topic+getRootPath.TotalCnSmoothing'></span><span id='topic+getSmoothedDataSets'></span><span id='topic+getSubsetToFit'></span><span id='topic+getTargetPositions'></span><span id='topic+getTargetPositions.TotalCnSmoothing'></span><span id='topic+getTargetUgpFile'></span><span id='topic+getTargetUgpFile.TotalCnSmoothing'></span><span id='topic+makeSmoothSplinePredict'></span><span id='topic+makeSmoothSplinePredict.numeric'></span><span id='topic+nbrOfDataSets'></span><span id='topic+nbrOfDataSets.MultiSourceCopyNumberNormalization'></span><span id='topic+normalizeOne'></span><span id='topic+normalizeOne.MultiSourceCopyNumberNormalization'></span><span id='topic+normalizePrincipalCurve'></span><span id='topic+pairs2'></span><span id='topic+pairs2.principal.curve'></span><span id='topic+process.TotalCnSmoothing'></span><span id='topic+getAsteriskTags.TotalCnKernelSmoothing'></span><span id='topic+getParameters.TotalCnKernelSmoothing'></span><span id='topic+smoothRawCopyNumbers'></span><span id='topic+smoothRawCopyNumbers.TotalCnKernelSmoothing'></span><span id='topic+smoothRawCopyNumbers.TotalCnSmoothing'></span><span id='topic+asCartesianSnpData'></span><span id='topic+asCartesianSnpData.CartesianSnpData'></span><span id='topic+asCartesianSnpData.PolarSnpData'></span><span id='topic+asCartesianSnpData.TotalFracBSnpData'></span><span id='topic+asPolarSnpData'></span><span id='topic+asPolarSnpData.CartesianSnpData'></span><span id='topic+asPolarSnpData.PolarSnpData'></span><span id='topic+asPolarSnpData.TotalFracBSnpData'></span><span id='topic+asTotalFracBSnpData'></span><span id='topic+asTotalFracBSnpData.CartesianSnpData'></span><span id='topic+asTotalFracBSnpData.PolarSnpData'></span><span id='topic+asTotalFracBSnpData.TotalFracBSnpData'></span><span id='topic+callGenotypes'></span><span id='topic+callGenotypes.SnpData'></span><span id='topic+callGenotypes.TotalFracBSnpData'></span><span id='topic+CartesianSnpData'></span><span id='topic+pairedBoost'></span><span id='topic+pairedBoost.CartesianSnpData'></span><span id='topic+pairedBoost.PolarSnpData'></span><span id='topic+pairedBoost.TotalFracBSnpData'></span><span id='topic+plot.CartesianSnpData'></span><span id='topic+plot.PolarSnpData'></span><span id='topic+plot.SnpData'></span><span id='topic+plot.TotalFracBSnpData'></span><span id='topic+PolarSnpData'></span><span id='topic+SnpData'></span><span id='topic+TotalFracBSnpData'></span><span id='topic+as.character.TumorBoostNormalization'></span><span id='topic+getDataSets'></span><span id='topic+getDataSets.TumorBoostNormalization'></span><span id='topic+getInputDataSet.TumorBoostNormalization'></span><span id='topic+getNormalDataSet'></span><span id='topic+getNormalDataSet.TumorBoostNormalization'></span><span id='topic+getNormalGenotypeCallSet'></span><span id='topic+getNormalGenotypeCallSet.TumorBoostNormalization'></span><span id='topic+getOutputDataSet.TumorBoostNormalization'></span><span id='topic+getOutputPath'></span><span id='topic+getOutputPath.TumorBoostNormalization'></span><span id='topic+getRootPath.TumorBoostNormalization'></span><span id='topic+nbrOfFiles.TumorBoostNormalization'></span><span id='topic+process.TumorBoostNormalization'></span><span id='topic+getAsteriskTags.TumorBoostNormalization'></span><span id='topic+getFullName.TumorBoostNormalization'></span><span id='topic+getName.TumorBoostNormalization'></span><span id='topic+getPath.TumorBoostNormalization'></span><span id='topic+getTags.TumorBoostNormalization'></span><span id='topic+setTags.TumorBoostNormalization'></span><span id='topic+getAsteriskTags.MultiSourceCopyNumberNormalization'></span><span id='topic+getTags.MultiSourceCopyNumberNormalization'></span><span id='topic+getFlavor'></span><span id='topic+getFlavor.TumorBoostNormalization'></span><span id='topic+as.character.AbstractCurveNormalization'></span><span id='topic+backtransformOne'></span><span id='topic+backtransformOne.AbstractCurveNormalization'></span><span id='topic+backtransformOne.PrincipalCurveNormalization'></span><span id='topic+backtransformOne.XYCurveNormalization'></span><span id='topic+fitOne.AbstractCurveNormalization'></span><span id='topic+fitOne.PrincipalCurveNormalization'></span><span id='topic+fitOne.XYCurveNormalization'></span><span id='topic+getAsteriskTags.AbstractCurveNormalization'></span><span id='topic+getDataSets.AbstractCurveNormalization'></span><span id='topic+getFullName.AbstractCurveNormalization'></span><span id='topic+getInputDataSet.AbstractCurveNormalization'></span><span id='topic+getName.AbstractCurveNormalization'></span><span id='topic+getOutputDataSet.AbstractCurveNormalization'></span><span id='topic+getPairedDataSet.AbstractCurveNormalization'></span><span id='topic+getPath.AbstractCurveNormalization'></span><span id='topic+getRootPath.AbstractCurveNormalization'></span><span id='topic+getSubsetToFit.AbstractCurveNormalization'></span><span id='topic+getTags.AbstractCurveNormalization'></span><span id='topic+getTargetDataSet.AbstractCurveNormalization'></span><span id='topic+nbrOfFiles.AbstractCurveNormalization'></span><span id='topic+process.AbstractCurveNormalization'></span><span id='topic+setTags.AbstractCurveNormalization'></span><span id='topic+getSubsetToFit.PrincipalCurveNormalization'></span><span id='topic+getPairedDataSet'></span><span id='topic+getTargetDataSet'></span><span id='topic+callGenotypes.RawAlleleBFractions'></span><span id='topic+getCalls'></span><span id='topic+getCalls.RawGenotypeCalls'></span><span id='topic+getColors'></span><span id='topic+getColors.RawGenotypeCalls'></span><span id='topic+isHeterozygous.RawGenotypeCalls'></span><span id='topic+isHomozygous.RawGenotypeCalls'></span><span id='topic+RawGenotypeCalls'></span><span id='topic+normalizeTumorBoost.RawAlleleBFractions'></span><span id='topic+plotSmoothedPairsOne'></span><span id='topic+plotSmoothedPairsOne.MultiSourceCopyNumberNormalization'></span><span id='topic+getBacktransforms'></span><span id='topic+getBacktransforms.PrincipalCurve'></span><span id='topic+getPrincipalCurveEstimator'></span><span id='topic+getPrincipalCurveEstimator.MultiSourceCopyNumberNormalization'></span><span id='topic+plotBacktransforms'></span><span id='topic+plotBacktransforms.PrincipalCurve'></span><span id='topic+getDataSet'></span><span id='topic+getDataSet.NaiveFracBGenotyping'></span><span id='topic+getGenotypeCallSet'></span><span id='topic+getGenotypeCallSet.NaiveFracBGenotyping'></span><span id='topic+NaiveFracBGenotyping'></span><span id='topic+nbrOfLoci.NaiveFracBGenotyping'></span><span id='topic+nbrOfSNPs'></span><span id='topic+nbrOfSNPs.NaiveFracBGenotyping'></span><span id='topic+process.NaiveFracBGenotyping'></span><span id='topic+processOne'></span><span id='topic+processOne.NaiveFracBGenotyping'></span><span id='topic+segmentByPruneCBS'></span><span id='topic+segmentByPruneCBS.RawGenomicSignals'></span><span id='topic+normalizeBAFsByRegions'></span><span id='topic+callAllelicBalanceByBAFs'></span><span id='topic+callCopyNeutralRegions'></span><span id='topic+getXScale.CopyNumberRegions'></span><span id='topic+getYScale.CopyNumberRegions'></span><span id='topic+plot.CopyNumberRegions'></span><span id='topic+segmentByPruneCBSv1'></span><span id='topic+segmentByPruneCBSv1.RawGenomicSignals'></span><span id='topic+orthogonalizeC1C2'></span><span id='topic+ABtoC1C2'></span><span id='topic+bootstrap'></span><span id='topic+bootstrap.PairedPSCBS'></span><span id='topic+C1C2toAB'></span><span id='topic+extractC1C2'></span><span id='topic+extractC1C2.list'></span><span id='topic+fitLoess2D'></span><span id='topic+fitLoess2D.C1C2'></span><span id='topic+fitLoessKD'></span><span id='topic+fitLoessKD.data.frame'></span><span id='topic+fitLoessKD.matrix'></span><span id='topic+normalizeLoess2D'></span><span id='topic+normalizeLoess2D.C1C2'></span><span id='topic+normalizeLoessKD'></span><span id='topic+normalizeLoessKD.data.frame'></span><span id='topic+normalizeLoessKD.matrix'></span><span id='topic+plot.C1C2'></span><span id='topic+points.C1C2'></span><span id='topic+resampleA'></span><span id='topic+resampleA.PairedPSCBS'></span><span id='topic+resampleB'></span><span id='topic+resampleB.PairedPSCBS'></span><span id='topic+resampleC'></span><span id='topic+resampleC.PairedPSCBS'></span><span id='topic+callPeaks'></span><span id='topic+callPeaks.data.frame'></span><span id='topic+estimateC2Bias'></span><span id='topic+estimateC2Bias.PairedPSCBS'></span><span id='topic+fitC1C2Densities'></span><span id='topic+fitC1C2Densities.PairedPSCBS'></span><span id='topic+fitC1C2Peaks'></span><span id='topic+fitC1C2Peaks.PairedPSCBS'></span><span id='topic+plotC1C2Grid'></span><span id='topic+plotC1C2Grid.PairedPSCBS'></span><span id='topic+translateC1C2'></span><span id='topic+translateC1C2.PairedPSCBS'></span><span id='topic+transformC1C2'></span><span id='topic+transformC1C2.PairedPSCBS'></span><span id='topic+PairedPSCBSFile'></span><span id='topic+PairedPSCBSFileSet'></span><span id='topic+byName.PairedPSCBSFileSet'></span><span id='topic+byPath.PairedPSCBSFileSet'></span><span id='topic+findAtomicAberrations.PairedPSCBS'></span><span id='topic+findByName.PairedPSCBSFileSet'></span><span id='topic+loadObject.PairedPSCBSFile'></span><span id='topic+plotTracks'></span><span id='topic+plotTracks.PruneCNA'></span><span id='topic+pruneCNA'></span><span id='topic+pruneCNA.PairedPSCBS'></span><span id='topic+testEqualityC1C2ByChiSquare'></span><span id='topic+testEqualityC1C2ByMean'></span><span id='topic+testEqualityTcnByMean'></span><span id='topic+testEqualityTcnByT'></span><span id='topic+cat'></span><span id='topic+getOption'></span><span id='topic+drawC1C2Density'></span><span id='topic+drawC1C2Density.PairedPSCBS'></span><span id='topic+getAsteriskTags.TotalCnBinnedSmoothing'></span><span id='topic+getParameters.TotalCnBinnedSmoothing'></span><span id='topic+smoothRawCopyNumbers.TotalCnBinnedSmoothing'></span><span id='topic+findAtomicAberrations.CBS'></span><span id='topic+pruneCNA.AbstractCBS'></span><span id='topic++5B.PruneCNA'></span><span id='topic+extractGenerations'></span><span id='topic+extractGenerations.PruneCNA'></span><span id='topic+nbrOfGenerations'></span><span id='topic+nbrOfGenerations.PruneCNA'></span><span id='topic+testEqualityC1C2ByChiSq'></span><span id='topic+testEqualityC1orC2ByT'></span><span id='topic+as.character.PairedPscbsModel'></span><span id='topic+fit.PairedPscbsModel'></span><span id='topic+getAsteriskTags.PairedPscbsModel'></span><span id='topic+getChipType.PairedPscbsModel'></span><span id='topic+getChromosomes.PairedPscbsModel'></span><span id='topic+getDataSets.PairedPscbsModel'></span><span id='topic+getFitFunction.PairedPscbsModel'></span><span id='topic+getFullName.PairedPscbsModel'></span><span id='topic+getName.PairedPscbsModel'></span><span id='topic+getNormalDataSet.PairedPscbsModel'></span><span id='topic+getOptionalArguments.PairedPscbsModel'></span><span id='topic+getParameters.PairedPscbsModel'></span><span id='topic+getParametersAsString.PairedPscbsModel'></span><span id='topic+getPath.PairedPscbsModel'></span><span id='topic+getRandomSeed.PairedPscbsModel'></span><span id='topic+getRootPath.PairedPscbsModel'></span><span id='topic+getTags.PairedPscbsModel'></span><span id='topic+getTumorDataSet'></span><span id='topic+getTumorDataSet.PairedPscbsModel'></span><span id='topic+indexOf.PairedPscbsModel'></span><span id='topic+nbrOfFiles.PairedPscbsModel'></span><span id='topic+setRandomSeed.PairedPscbsModel'></span><span id='topic+setTags.PairedPscbsModel'></span><span id='topic+calculatePairedPSCNByGenotype'></span><span id='topic+calculatePairedPSCNByGenotype.data.frame'></span><span id='topic+calculatePairedPSCNByGenotype.numeric'></span><span id='topic+calculateTumorPSCNByGenotypeUsingCTCN'></span><span id='topic+calculateTumorPSCNByGenotypeUsingCTCN.numeric'></span><span id='topic+PairedPscbsCaller'></span><span id='topic+getAsteriskTags.PairedPscbsCaller'></span><span id='topic+getParameters.PairedPscbsCaller'></span><span id='topic+getPath.PairedPscbsCaller'></span><span id='topic+getPlatform.PairedPscbsCaller'></span><span id='topic+getRootPath.PairedPscbsCaller'></span><span id='topic+process.PairedPscbsCaller'></span><span id='topic+getOutputDataSet.PairedPscbsModel'></span><span id='topic+loadFit.PairedPSCBSFile'></span><span id='topic+getChipType.PairedPSCBSFileSet'></span><span id='topic+getDefaultFullName.PairedPSCBSFileSet'></span><span id='topic+getPlatform.PairedPSCBSFileSet'></span><span id='topic+getOptionalArguments.PairedPscbsCaller'></span><span id='topic+backgroundCorrect'></span><span id='topic+backgroundCorrect.PairedPSCBS'></span><span id='topic+getOutputDataSet0.TotalCnSmoothing'></span><span id='topic+getOutputFileClass'></span><span id='topic+getOutputFileClass.TotalCnSmoothing'></span><span id='topic+getOutputFileExtension'></span><span id='topic+getOutputFileExtension.TotalCnSmoothing'></span><span id='topic+getOutputFileSetClass'></span><span id='topic+getOutputFileSetClass.TotalCnSmoothing'></span><span id='topic+doSegmentByPairedPSCBS'></span><span id='topic+doSegmentByPairedPSCBS.AromaUnitPscnBinarySet'></span><span id='topic+findLargeGaps.AromaUgpFile'></span><span id='topic+findLargeGaps.AromaUnitPscnBinarySet'></span>

<h3>Description</h3>

<p>This page contains aliases for all &quot;non-documented&quot; objects that
<code>R CMD check</code> detects in this package.
</p>
<p>Almost all of them are <em>generic</em> functions that have specific
document for the corresponding method coupled to a specific class.
Other functions are re-defined by <code>setMethodS3()</code> to
<em>default</em> methods. Neither of these two classes are non-documented
in reality.
The rest are deprecated methods.
</p>

<hr>
<h2 id='normalizeBAFsByRegions.PairedPSCBS'>Normalizes allele B fractions (BAFs) based on region-based PSCN estimates</h2><span id='topic+normalizeBAFsByRegions.PairedPSCBS'></span><span id='topic+PairedPSCBS.normalizeBAFsByRegions'></span><span id='topic+normalizeBAFsByRegions+2CPairedPSCBS-method'></span>

<h3>Description</h3>

<p>Normalizes allele B fractions (BAFs) based on region-based PSCN estimates as given by the PSCBS segmentation method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PairedPSCBS'
normalizeBAFsByRegions(fit, by=c("betaTN", "betaT"), ..., force=FALSE, cache=TRUE,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalizeBAFsByRegions.PairedPSCBS_+3A_fit">fit</code></td>
<td>
<p>A PairedPSCBS fit object as returned by
<code><a href="PSCBS.html#topic+segmentByPairedPSCBS">segmentByPairedPSCBS</a></code>.</p>
</td></tr>
<tr><td><code id="normalizeBAFsByRegions.PairedPSCBS_+3A_by">by</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying if the normalization function
should be estimated based on TumorBoost normalized or non-normalized
tumor allele B fractions (BAFs).</p>
</td></tr>
<tr><td><code id="normalizeBAFsByRegions.PairedPSCBS_+3A_...">...</code></td>
<td>
<p>Additional arguments passed
<code><a href="#topic+normalizeMirroredBAFsByRegions">normalizeMirroredBAFsByRegions</a></code>.</p>
</td></tr>
<tr><td><code id="normalizeBAFsByRegions.PairedPSCBS_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that his normalization method depends on the segmentation
results. Hence, it recommended <em>not</em> to resegment the
normalized signals returned by this, because such a segmentation
will be highly dependent on the initial segmentation round.
</p>


<h3>Value</h3>

<p>Returns a PairedPSCBS fit object where the region-level
decrease-in-heterozygosity (DH) means have been normalized,
as well as the locus-specific tumor allele B fractions.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson, Pierre Neuvial</p>


<h3>See Also</h3>

<p>Internally <code><a href="#topic+normalizeMirroredBAFsByRegions">normalizeMirroredBAFsByRegions</a></code> is used.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("aroma.cn")

if (Sys.getenv("_R_CHECK_FULL_") != "" &amp;&amp; require("PSCBS")) {

# Load example ASCN data
data &lt;- PSCBS::exampleData("paired.chr01")
R.oo::attachLocally(data)

# AD HOC: Robustification
CT[CT &lt; 0] &lt;- 0
CT[CT &gt; 30] &lt;- 30

# PSCBS segmentation
fit &lt;- segmentByPairedPSCBS(CT, betaT=betaT, betaN=betaN, x=x, verbose=-10)

# Normalize
fitN &lt;- normalizeBAFsByRegions(fit, verbose=-10)

devSet("tracks")
subplots(10, ncol=2, byrow=FALSE)
par(mar=c(1,3.5,1,0.5)+1)
plot(fit, subplots=TRUE)
plot(fitN, subplots=TRUE)


devSet("C1C2")
Clim &lt;- c(0,4)
subplots(4, ncol=2, byrow=TRUE)
par(mar=c(1,3.5,1,0.5)+1)
plotC1C2(fit, Clim=Clim)
linesC1C2(fit)
title(main="(C1,C2)")

plotC1C2(fitN, Clim=Clim)
linesC1C2(fitN)
title(main="(C1,C2) - adjusted")

plotC1C2(fit, col="gray", Clim=Clim)
linesC1C2(fit, col="gray")
pointsC1C2(fitN)
linesC1C2(fitN)
title(main="(C1,C2) - both")

} # if (require("PSCBS"))
</code></pre>

<hr>
<h2 id='normalizeMirroredBAFsByRegions.matrix'>Normalizes region-level mirrored allele B fractions (mBAFs)</h2><span id='topic+normalizeMirroredBAFsByRegions.matrix'></span><span id='topic+normalizeMirroredBAFsByRegions'></span>

<h3>Description</h3>

<p>Normalizes region-level mirrored allele B fractions (mBAFs) for heterozygous and homozygous SNPs by rescaling both
equally much such that the normalized homozygous mBAFs become exactly
or close to their expected values (which is at zero and one).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'matrix'
normalizeMirroredBAFsByRegions(data, flavor=c("plain", "total"), ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalizeMirroredBAFsByRegions.matrix_+3A_data">data</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> Kx2 or Kx3 <code><a href="base.html#topic+matrix">matrix</a></code>, where K is the number of
segments and the first and the second column contains average
heterozygous and homozygous mBAF estimates, respectively.
The third column, which is optional, contains total copy numbers.</p>
</td></tr>
<tr><td><code id="normalizeMirroredBAFsByRegions.matrix_+3A_flavor">flavor</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying how the normalization
function/scale factors are estimated.</p>
</td></tr>
<tr><td><code id="normalizeMirroredBAFsByRegions.matrix_+3A_...">...</code></td>
<td>
<p>Additional arguments passed <code><a href="aroma.light.html#topic+fitXYCurve">fitXYCurve</a></code>,
which is used if <code>flavor="total"</code>.</p>
</td></tr>
<tr><td><code id="normalizeMirroredBAFsByRegions.matrix_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The mBAFs for heterozygous SNPs are also known as the
Decrease in Heterozygosity signals (DHs).
</p>


<h3>Value</h3>

<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+matrix">matrix</a></code> of the same dimensions as argument <code>data</code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson, Pierre Neuvial</p>

<hr>
<h2 id='normalizePrincipalCurve.matrix'>Normalizes data in K dimensions using principal curves</h2><span id='topic+normalizePrincipalCurve.matrix'></span>

<h3>Description</h3>

<p>Normalizes data in K dimensions using principal curves such that afterward the data cluster (approximately
linearly) along the diagonal (in K dimensions).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'matrix'
normalizePrincipalCurve(x, ..., center=TRUE, returnFit=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalizePrincipalCurve.matrix_+3A_x">x</code></td>
<td>
<p>An NxK <code><a href="base.html#topic+matrix">matrix</a></code> where the columns represent the
(K &gt;= 2) dimensions.</p>
</td></tr>
<tr><td><code id="normalizePrincipalCurve.matrix_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to
<code><a href="aroma.light.html#topic+fitPrincipalCurve">fitPrincipalCurve</a></code> used for fitting the model.</p>
</td></tr>
<tr><td><code id="normalizePrincipalCurve.matrix_+3A_center">center</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, normalized data is centered such that the median
signal in each dimension is at zero.</p>
</td></tr>
<tr><td><code id="normalizePrincipalCurve.matrix_+3A_returnfit">returnFit</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the fitted principal curve parameters are
returned as an attribute.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an NxK <code><a href="base.html#topic+matrix">matrix</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] Hastie, T. and Stuetzle, W, <em>Principal Curves</em>, JASA, 1989.
</p>


<h3>See Also</h3>

<p><code><a href="aroma.light.html#topic+fitPrincipalCurve">fitPrincipalCurve</a></code> and
<code><a href="aroma.light.html#topic+backtransformPrincipalCurve">backtransformPrincipalCurve</a></code>.
</p>

<hr>
<h2 id='PairedPscbsModel'>The PairedPscbsModel class</h2><span id='topic+PairedPscbsModel'></span>

<h3>Description</h3>

<p>Package:  aroma.cn <br />
<b>Class PairedPscbsModel</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code>PairedPscbsModel</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>PairedPscbsModel</b><br />
extends <a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a><br />
</p>
<p>This class represents the Paired PSCBS method [1], which
segments matched tumor-normal parental copy-number data into
piecewise constant segments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PairedPscbsModel(dsT=NULL, dsN=NULL, tags="*", ..., dropTcnOutliers=TRUE,
  gapMinLength=1e+06, seed=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PairedPscbsModel_+3A_dst">dsT</code>, <code id="PairedPscbsModel_+3A_dsn">dsN</code></td>
<td>
<p>The tumor and the normal
<code><a href="aroma.core.html#topic+AromaUnitPscnBinarySet">AromaUnitPscnBinarySet</a></code>.</p>
</td></tr>
<tr><td><code id="PairedPscbsModel_+3A_tags">tags</code></td>
<td>
<p>Tags added to the output data sets.</p>
</td></tr>
<tr><td><code id="PairedPscbsModel_+3A_...">...</code></td>
<td>
<p>(Optional) Additional arguments passed to
<code><a href="PSCBS.html#topic+segmentByPairedPSCBS">segmentByPairedPSCBS</a></code>.</p>
</td></tr>
<tr><td><code id="PairedPscbsModel_+3A_droptcnoutliers">dropTcnOutliers</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, then TCN outliers are dropped using
<code><a href="PSCBS.html#topic+dropSegmentationOutliers">dropSegmentationOutliers</a></code>.</p>
</td></tr>
<tr><td><code id="PairedPscbsModel_+3A_gapminlength">gapMinLength</code></td>
<td>
<p>Genomic regions with no data points that are of
this length and greater are considered to be &quot;gaps&quot; and are ignored
in the segmentation.  If +<code><a href="base.html#topic+is.finite">Inf</a></code>, no gaps are identified.</p>
</td></tr>
<tr><td><code id="PairedPscbsModel_+3A_seed">seed</code></td>
<td>
<p>An optional <code><a href="base.html#topic+integer">integer</a></code> specifying the random seed to be
used in the segmentation.  Seed needs to be set for exact numerical
reproducibility.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>fit</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChromosomes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getDataSets</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFullName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getNormalDataSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getOutputDataSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getTags</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getTumorDataSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>indexOf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfFiles</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setTags</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>References</h3>

<p>[1] ... <br />
</p>


<h3>See Also</h3>

<p>...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  dataSet &lt;- "GSE12702"
tags &lt;- "ASCRMAv2"
chipType &lt;- "Mapping250K_Nsp"
ds &lt;- AromaUnitPscnBinarySet$byName(dataSet, tags=tags, chipType=chipType)
print(ds)

# Extract tumors and normals
idxs &lt;- seq(from=1, to=nbrOfFiles(ds), by=2)
dsT &lt;- extract(ds, idxs);
idxs &lt;- seq(from=2, to=nbrOfFiles(ds), by=2)
dsN &lt;- extract(ds, idxs);

# Setup Paired PSCBS model
seg &lt;- PairedPscbsModel(dsT=dsT, dsN=dsN)
print(seg)

# Segment all tumor-normal pairs
fit(seg, verbose=-10)


## End(Not run)</code></pre>

<hr>
<h2 id='PrincipalCurveNormalization'>The PrincipalCurveNormalization class</h2><span id='topic+PrincipalCurveNormalization'></span>

<h3>Description</h3>

<p>Package:  aroma.cn <br />
<b>Class PrincipalCurveNormalization</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+AbstractCurveNormalization">AbstractCurveNormalization</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code>PrincipalCurveNormalization</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>PrincipalCurveNormalization</b><br />
extends <em><a href="#topic+AbstractCurveNormalization">AbstractCurveNormalization</a></em><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PrincipalCurveNormalization(..., subset=1/20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PrincipalCurveNormalization_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AbstractCurveNormalization">AbstractCurveNormalization</a></code>.</p>
</td></tr>
<tr><td><code id="PrincipalCurveNormalization_+3A_subset">subset</code></td>
<td>
<p>A <code><a href="base.html#topic+double">double</a></code> in (0,1] specifying the fraction of the
<code>subsetToFit</code> to be used for fitting.  Since the fit function
for this class is rather slow, the default is to use a 1/20:th
of the default data points.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from AbstractCurveNormalization</b>:<br />
as.character, backtransformOne, fitOne, getAsteriskTags, getDataSets, getFullName, getInputDataSet, getName, getOutputDataSet, getPairedDataSet, getPath, getRootPath, getSubsetToFit, getTags, getTargetDataSet, nbrOfFiles, process, setTags
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='process.MultiSourceCopyNumberNormalization'>Normalizes all samples</h2><span id='topic+process.MultiSourceCopyNumberNormalization'></span><span id='topic+MultiSourceCopyNumberNormalization.process'></span><span id='topic+process+2CMultiSourceCopyNumberNormalization-method'></span>

<h3>Description</h3>

<p>Normalizes all samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MultiSourceCopyNumberNormalization'
process(this, ..., force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.MultiSourceCopyNumberNormalization_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="process.MultiSourceCopyNumberNormalization_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> of K <code><a href="aroma.core.html#topic+AromaUnitTotalCnBinarySet">AromaUnitTotalCnBinarySet</a></code>:s.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+MultiSourceCopyNumberNormalization">MultiSourceCopyNumberNormalization</a></code>.
</p>

<hr>
<h2 id='testAllelicBalanceByBAFs'>Tests for allelic balance in a genomic region</h2><span id='topic+testAllelicBalanceByBAFs.default'></span><span id='topic+testAllelicBalanceByBAFs'></span>

<h3>Description</h3>

<p>Tests for allelic balance in a genomic region.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
testAllelicBalanceByBAFs(betaT, muN, flavor=c("var", "mean"), ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="testAllelicBalanceByBAFs_+3A_betat">betaT</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of J tumor allele B fractions (BAFs).</p>
</td></tr>
<tr><td><code id="testAllelicBalanceByBAFs_+3A_mun">muN</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of J normal (germline) genotypes.</p>
</td></tr>
<tr><td><code id="testAllelicBalanceByBAFs_+3A_flavor">flavor</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> specifying the type of test to be performed.</p>
</td></tr>
<tr><td><code id="testAllelicBalanceByBAFs_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="testAllelicBalanceByBAFs_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="base.html#topic+list">list</a></code> of class &quot;htest&quot;.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson, Pierre Neuvial</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (Sys.getenv("_R_CHECK_FULL_") != "" &amp;&amp; require("PSCBS")) {

# Load example ASCN data
data &lt;- PSCBS::exampleData("paired.chr01");

# PSCBS segmentation
fit &lt;- segmentByPairedPSCBS(data, verbose=-10);

# Normalize
fitN &lt;- normalizeBAFsByRegions(fit, verbose=-10);

fitN2 &lt;- callAllelicBalanceByBAFs(fitN, verbose=-10);
print(fitN2);

} # if (require("PSCBS"))
</code></pre>

<hr>
<h2 id='TotalCnBinnedSmoothing'>The TotalCnBinnedSmoothing class</h2><span id='topic+TotalCnBinnedSmoothing'></span>

<h3>Description</h3>

<p>Package:  aroma.cn <br />
<b>Class TotalCnBinnedSmoothing</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+TotalCnSmoothing">TotalCnSmoothing</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code>TotalCnBinnedSmoothing</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>TotalCnBinnedSmoothing</b><br />
extends <em><a href="#topic+TotalCnSmoothing">TotalCnSmoothing</a></em><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TotalCnBinnedSmoothing(..., robust=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TotalCnBinnedSmoothing_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+TotalCnSmoothing">TotalCnSmoothing</a></code>.</p>
</td></tr>
<tr><td><code id="TotalCnBinnedSmoothing_+3A_robust">robust</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, a robust smoother is used, otherwise not.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that <code>dsS &lt;- TotalCnBinnedSmoothing(ds, targetUgp=ugp)</code> where
<code>ugp &lt;- getAromaUgpFile(ds)</code> returns a data set with an identical
set of loci as the input data set and identical signals as the
input ones, <em>except</em> for loci with duplicated positions.  If all
loci have unique positions, the the output is identical to the input.
</p>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from TotalCnSmoothing</b>:<br />
getAsteriskTags, getOutputDataSet0, getOutputFileClass, getOutputFileExtension, getOutputFileSetClass, getOutputFiles, getParameters, getPath, getRootPath, getTargetPositions, getTargetUgpFile, process, smoothRawCopyNumbers
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='TotalCnKernelSmoothing'>The TotalCnKernelSmoothing class</h2><span id='topic+TotalCnKernelSmoothing'></span>

<h3>Description</h3>

<p>Package:  aroma.cn <br />
<b>Class TotalCnKernelSmoothing</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+TotalCnSmoothing">TotalCnSmoothing</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code>TotalCnKernelSmoothing</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>TotalCnKernelSmoothing</b><br />
extends <em><a href="#topic+TotalCnSmoothing">TotalCnSmoothing</a></em><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TotalCnKernelSmoothing(..., kernel=c("gaussian", "uniform"), bandwidth=50000, censorH=3,
  robust=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TotalCnKernelSmoothing_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+TotalCnSmoothing">TotalCnSmoothing</a></code>.</p>
</td></tr>
<tr><td><code id="TotalCnKernelSmoothing_+3A_kernel">kernel</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying the type of kernel
to be used.</p>
</td></tr>
<tr><td><code id="TotalCnKernelSmoothing_+3A_bandwidth">bandwidth</code></td>
<td>
<p>A <code><a href="base.html#topic+double">double</a></code> specifying the bandwidth of the smoothing.</p>
</td></tr>
<tr><td><code id="TotalCnKernelSmoothing_+3A_censorh">censorH</code></td>
<td>
<p>A positive <code><a href="base.html#topic+double">double</a></code> specifying the bandwidth threshold
where values outside are ignored (zero weight).</p>
</td></tr>
<tr><td><code id="TotalCnKernelSmoothing_+3A_robust">robust</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, a robust smoother is used, otherwise not.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from TotalCnSmoothing</b>:<br />
getAsteriskTags, getOutputDataSet0, getOutputFileClass, getOutputFileExtension, getOutputFileSetClass, getOutputFiles, getParameters, getPath, getRootPath, getTargetPositions, getTargetUgpFile, process, smoothRawCopyNumbers
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='TotalCnSmoothing'>The abstract TotalCnSmoothing class</h2><span id='topic+TotalCnSmoothing'></span>

<h3>Description</h3>

<p>Package:  aroma.cn <br />
<b>Class TotalCnSmoothing</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><em><code>TotalCnSmoothing</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+TotalCnBinnedSmoothing">TotalCnBinnedSmoothing</a>, <a href="#topic+TotalCnKernelSmoothing">TotalCnKernelSmoothing</a><br />
</p>
<p>public abstract static class <b>TotalCnSmoothing</b><br />
extends <em><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></em><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TotalCnSmoothing(dataSet=NULL, ..., targetUgp=NULL,
  .reqSetClass="AromaUnitTotalCnBinarySet")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TotalCnSmoothing_+3A_dataset">dataSet</code></td>
<td>
<p>An <code><a href="aroma.core.html#topic+AromaUnitTotalCnBinarySet">AromaUnitTotalCnBinarySet</a></code>.</p>
</td></tr>
<tr><td><code id="TotalCnSmoothing_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code>.</p>
</td></tr>
<tr><td><code id="TotalCnSmoothing_+3A_targetugp">targetUgp</code></td>
<td>
<p>An <code><a href="aroma.core.html#topic+AromaUgpFile">AromaUgpFile</a></code> specifying the
target loci for which smoothed copy-number are generated.</p>
</td></tr>
<tr><td><code id="TotalCnSmoothing_+3A_.reqsetclass">.reqSetClass</code></td>
<td>
<p>(internal only)</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getTargetUgpFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='TumorBoostNormalization'>The TumorBoostNormalization class</h2><span id='topic+TumorBoostNormalization'></span>

<h3>Description</h3>

<p>Package:  aroma.cn <br />
<b>Class TumorBoostNormalization</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code>TumorBoostNormalization</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>TumorBoostNormalization</b><br />
extends <a href="R.oo.html#topic+Object">Object</a><br />
</p>
<p>TumorBoost is normalization method that normalizes the allele B fractions
of a tumor sample given the allele B fractions and genotype calls for
a matched normal.
The method is a single-sample (single-pair) method.  It does not require
total copy number estimates.
The normalization is done such that the total copy number is unchanged
afterwards.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TumorBoostNormalization(dsT=NULL, dsN=NULL, gcN=NULL, flavor=c("v4", "v3", "v2", "v1"),
  preserveScale=TRUE, collapseHomozygous=FALSE, tags="*", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TumorBoostNormalization_+3A_dst">dsT</code></td>
<td>
<p>An <code><a href="aroma.core.html#topic+AromaUnitFracBCnBinarySet">AromaUnitFracBCnBinarySet</a></code> of
tumor samples.</p>
</td></tr>
<tr><td><code id="TumorBoostNormalization_+3A_dsn">dsN</code></td>
<td>
<p>An <code><a href="aroma.core.html#topic+AromaUnitFracBCnBinarySet">AromaUnitFracBCnBinarySet</a></code> of
match normal samples.</p>
</td></tr>
<tr><td><code id="TumorBoostNormalization_+3A_gcn">gcN</code></td>
<td>
<p>An <code><a href="aroma.core.html#topic+AromaUnitGenotypeCallSet">AromaUnitGenotypeCallSet</a></code> of
genotypes for the normals.</p>
</td></tr>
<tr><td><code id="TumorBoostNormalization_+3A_flavor">flavor</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying the type of
correction applied.</p>
</td></tr>
<tr><td><code id="TumorBoostNormalization_+3A_preservescale">preserveScale</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, SNPs that are heterozygous in the
matched normal are corrected for signal compression using an estimate
of signal compression based on the amount of correction performed
by TumorBoost on SNPs that are homozygous in the matched normal.</p>
</td></tr>
<tr><td><code id="TumorBoostNormalization_+3A_collapsehomozygous">collapseHomozygous</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, SNPs that are homozygous in the
matched normal are also called homozygous in the tumor, that is,
it's allele B fraction is collapsed to either 0 or 1.
If <code><a href="base.html#topic+logical">FALSE</a></code>, the homozygous values are normalized according the
model. [NOT USED YET]
</p>
</td></tr>
<tr><td><code id="TumorBoostNormalization_+3A_tags">tags</code></td>
<td>
<p>(Optional) Sets the tags for the output data sets.</p>
</td></tr>
<tr><td><code id="TumorBoostNormalization_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFullName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getInputDataSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getNormalDataSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getNormalGenotypeCallSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getOutputDataSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getTags</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfFiles</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setTags</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson, Pierre Neuvial</p>

<hr>
<h2 id='XYCurveNormalization'>The XYCurveNormalization class</h2><span id='topic+XYCurveNormalization'></span>

<h3>Description</h3>

<p>Package:  aroma.cn <br />
<b>Class XYCurveNormalization</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+AbstractCurveNormalization">AbstractCurveNormalization</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code>XYCurveNormalization</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>XYCurveNormalization</b><br />
extends <em><a href="#topic+AbstractCurveNormalization">AbstractCurveNormalization</a></em><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>XYCurveNormalization(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="XYCurveNormalization_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AbstractCurveNormalization">AbstractCurveNormalization</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from AbstractCurveNormalization</b>:<br />
as.character, backtransformOne, fitOne, getAsteriskTags, getDataSets, getFullName, getInputDataSet, getName, getOutputDataSet, getPairedDataSet, getPath, getRootPath, getSubsetToFit, getTags, getTargetDataSet, nbrOfFiles, process, setTags
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
