<!DOCTYPE html><html lang="en"><head><title>Help for package aroma.core</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {aroma.core}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#aroma.core-package'><p>Package aroma.core</p></a></li>
<li><a href='#AbstractCNData'><p>The AbstractCNData class</p></a></li>
<li><a href='#AbstractPSCNData'><p>The AbstractPSCNData class</p></a></li>
<li><a href='#AromaCellCpgFile'><p>A binary file holding local CpG density for each cell (probe/feature)</p></a></li>
<li><a href='#AromaCellPositionFile'><p>A binary file holding chromosome/position for each cell</p></a></li>
<li><a href='#AromaCellTabularBinaryFile'><p>The AromaCellTabularBinaryFile class</p></a></li>
<li><a href='#AromaGenomeTextFile'><p>The AromaGenomeTextFile class</p></a></li>
<li><a href='#AromaMicroarrayDataFile'><p>The abstract AromaMicroarrayDataFile class</p></a></li>
<li><a href='#AromaMicroarrayDataSet'><p>The AromaMicroarrayDataSet class</p></a></li>
<li><a href='#AromaMicroarrayDataSetTuple'><p>The AromaMicroarrayDataSetTuple class</p></a></li>
<li><a href='#AromaMicroarrayTabularBinaryFile'><p>The AromaMicroarrayTabularBinaryFile class</p></a></li>
<li><a href='#AromaPlatform'><p>The AromaPlatform class</p></a></li>
<li><a href='#AromaPlatformInterface'><p>The AromaPlatformInterface class</p></a></li>
<li><a href='#AromaRepository'><p>The AromaRepository class</p></a></li>
<li><a href='#AromaTabularBinaryFile'><p>The AromaTabularBinaryFile class</p></a></li>
<li><a href='#AromaTabularBinaryFile+24allocate'><p>Creates an AromaTabularBinaryFile</p></a></li>
<li><a href='#AromaTabularBinarySet'><p>The AromaTabularBinarySet class</p></a></li>
<li><a href='#AromaTransform'><p>The AromaTransform class</p></a></li>
<li><a href='#AromaUnitCallFile'><p>The AromaUnitCallFile class</p></a></li>
<li><a href='#AromaUnitCallSet'><p>The AromaUnitCallSet class</p></a></li>
<li><a href='#AromaUnitFracBCnBinaryFile'><p>The AromaUnitFracBCnBinaryFile class</p></a></li>
<li><a href='#AromaUnitFracBCnBinarySet'><p>The AromaUnitFracBCnBinarySet class</p></a></li>
<li><a href='#AromaUnitGenotypeCallFile'><p>The AromaUnitGenotypeCallFile class</p></a></li>
<li><a href='#AromaUnitGenotypeCallSet'><p>The AromaUnitGenotypeCallSet class</p></a></li>
<li><a href='#AromaUnitPscnBinaryFile'><p>The AromaUnitPscnBinaryFile class</p></a></li>
<li><a href='#AromaUnitPscnBinarySet'><p>The AromaUnitPscnBinarySet class</p></a></li>
<li><a href='#AromaUnitSignalBinaryFile'><p>The AromaUnitSignalBinaryFile class</p></a></li>
<li><a href='#AromaUnitSignalBinarySet'><p>The AromaUnitSignalBinarySet class</p></a></li>
<li><a href='#AromaUnitTabularBinaryFile'><p>The AromaUnitTabularBinaryFile class</p></a></li>
<li><a href='#AromaUnitTotalCnBinaryFile'><p>The AromaUnitTotalCnBinaryFile class</p></a></li>
<li><a href='#AromaUnitTotalCnBinarySet'><p>The AromaUnitTotalCnBinarySet class</p></a></li>
<li><a href='#AromaUnitTypesFile'><p>The AromaUnitTypesFile class</p></a></li>
<li><a href='#as.GrayscaleImage.matrix'><p>Creates a Grayscale (Color) Image from a matrix file</p></a></li>
<li><a href='#BinnedScatter'><p>The BinnedScatter class</p></a></li>
<li><a href='#CacheKeyInterface'><p>The CacheKeyInterface class interface</p></a></li>
<li><a href='#CbsModel'><p>The CbsModel class</p></a></li>
<li><a href='#ChromosomalModel'><p>The ChromosomalModel class</p></a></li>
<li><a href='#ChromosomeExplorer'><p>The ChromosomeExplorer class</p></a></li>
<li><a href='#colBinnedSmoothing.matrix'><p>Binned smoothing of a matrix column by column</p></a></li>
<li><a href='#colKernelSmoothing.matrix'><p>Kernel smoothing of a matrix column by column</p></a></li>
<li><a href='#CopyNumberChromosomalModel'><p>The CopyNumberChromosomalModel class</p></a></li>
<li><a href='#CopyNumberSegmentationModel'><p>The CopyNumberSegmentationModel class</p></a></li>
<li><a href='#display.Explorer'><p>Displays the explorer in the default browser</p></a></li>
<li><a href='#doCBS'><p>Performs Circular Binary Segmentation (CBS) on a data set</p></a></li>
<li><a href='#downloadChipTypeFile.AromaRepository'><p>Download a particular chip type annotation file</p></a></li>
<li><a href='#downloadFile.AromaRepository'><p>Download a particular file from the repository</p></a></li>
<li><a href='#estimateSds.CopyNumberChromosomalModel'><p>Estimates the standard deviation of the raw copy numbers (log2-ratios) robustly</p></a></li>
<li><a href='#estimateStandardDeviation.RawGenomicSignals'><p>Estimates the standard deviation of the raw Ys</p></a></li>
<li><a href='#Explorer'><p>The Explorer class</p></a></li>
<li><a href='#exportAromaUnitPscnBinarySet'><p>Export total and allele B signal data sets as a unified parent-specific copy number signal data set</p></a></li>
<li><a href='#extractRawCopyNumbers.CopyNumberChromosomalModel'><p>Extracts relative copy numbers</p></a></li>
<li><a href='#FileCacheKeyInterface'><p>The FileCacheKeyInterface class interface</p></a></li>
<li><a href='#findAnnotationData'><p>Locates an annotation data file</p></a></li>
<li><a href='#findAnnotationDataByChipType'><p>Locates an annotation data file by its chip type</p></a></li>
<li><a href='#findFilesTodo.AromaTransform'><p>Finds files in the data set still not processed</p></a></li>
<li><a href='#fit.CopyNumberChromosomalModel'><p>Fits the model</p></a></li>
<li><a href='#fit.CopyNumberSegmentationModel'><p>Fits the model</p></a></li>
<li><a href='#fitGenotypeCone.matrix'><p>Fits an affine transformation to allele A and allele B data</p></a></li>
<li><a href='#fitGenotypeConeBySfit.matrix'><p>Fits an affine transformation to allele A and allele B data</p></a></li>
<li><a href='#fitMultiDimensionalCone.matrix'><p>Fits an affine transformation to multi-dimensional data</p></a></li>
<li><a href='#getAlias.Explorer'><p>Gets the alias of the output set</p></a></li>
<li><a href='#getAromaPlatform.AromaPlatformInterface'><p>Gets the platform</p></a></li>
<li><a href='#getCacheKey.CacheKeyInterface'><p>Gets a list of cache key items</p></a></li>
<li><a href='#getCacheKey.FileCacheKeyInterface'><p>Gets a list of cache key items</p></a></li>
<li><a href='#getChipType.AromaPlatformInterface'><p>Gets the chip type</p></a></li>
<li><a href='#getChipType.ChromosomalModel'><p>Gets a label for all chip types merged</p></a></li>
<li><a href='#getChromosomes.ChromosomalModel'><p>Gets the chromosomes to be processed</p></a></li>
<li><a href='#getChromosomes.ChromosomeExplorer'><p>Gets the chromosomes available</p></a></li>
<li><a href='#getFullName.AromaTransform'><p>Gets the full name of the output data set</p></a></li>
<li><a href='#getInputDataSet.AromaTransform'><p>Gets the input data set</p></a></li>
<li><a href='#getModel.ChromosomeExplorer'><p>Gets the model</p></a></li>
<li><a href='#getName.AromaTransform'><p>Gets the name of the output data set</p></a></li>
<li><a href='#getName.Explorer'><p>Gets the name of the explorer</p></a></li>
<li><a href='#getNames.ChromosomalModel'><p>Gets the names of the arrays</p></a></li>
<li><a href='#getNames.Explorer'><p>Gets the names of the input samples</p></a></li>
<li><a href='#getOutputDataSet.AromaTransform'><p>Gets the transformed data set</p></a></li>
<li><a href='#getParameters.ParametersInterface'><p>Gets a list of parameters</p></a></li>
<li><a href='#getParametersAsString.ParametersInterface'><p>Gets the parameters as character</p></a></li>
<li><a href='#getPath.AromaTransform'><p>Gets the path of the output directory</p></a></li>
<li><a href='#getPath.Explorer'><p>Gets the path of the output directory</p></a></li>
<li><a href='#getPlatform.AromaPlatformInterface'><p>Gets the platform</p></a></li>
<li><a href='#getRootPath.AromaTransform'><p>Gets the root path of the output directory</p></a></li>
<li><a href='#getRootPath.Explorer'><p>Gets the root path of the output directory</p></a></li>
<li><a href='#getTags.AromaTransform'><p>Gets the tags of the output data set</p></a></li>
<li><a href='#getTags.Explorer'><p>Gets the tags of the explorer</p></a></li>
<li><a href='#getUnitAnnotationDataFile.AromaPlatformInterface'><p>Gets a unit annotation data file of a particular class</p></a></li>
<li><a href='#GladModel'><p>The GladModel class</p></a></li>
<li><a href='#HaarSegModel'><p>The HaarSegModel class</p></a></li>
<li><a href='#indexOf.UnitNamesFile'><p>Gets the indices of units by their names</p></a></li>
<li><a href='#isCompatibleWith.AromaPlatformInterface'><p>Checks if a particular unit annotation data file is compatible</p></a></li>
<li><a href='#isDone.AromaTransform'><p>Checks if the data set is processed or not</p></a></li>
<li><a href='#listFiles.AromaRepository'><p>Retrieves the files available on the repository under a particular path</p></a></li>
<li><a href='#matrixBlockPolish.matrix'><p>Applies a polishing function to blocks of rows and columns repeatedly</p></a></li>
<li><a href='#mergeBoxplotStats.list'><p>Merges a list of boxplot.stats() elements</p></a></li>
<li><a href='#nbrOfArrays.ChromosomalModel'><p>Gets the number of arrays</p></a></li>
<li><a href='#nbrOfArrays.Explorer'><p>Gets the total number of arrays</p></a></li>
<li><a href='#nbrOfChipTypes.AromaMicroarrayDataSetTuple'><p>Gets the number of chip types</p></a></li>
<li><a href='#nbrOfChipTypes.ChromosomalModel'><p>Gets the number of chip types</p></a></li>
<li><a href='#Non-documented+20objects'><p>Non-documented objects</p></a></li>
<li><a href='#NonPairedPSCNData'><p>The NonPairedPSCNData class</p></a></li>
<li><a href='#PairedPSCNData'><p>The PairedPSCNData class</p></a></li>
<li><a href='#ParametersInterface'><p>The ParametersInterface class interface</p></a></li>
<li><a href='#plotTracks.PairedPSCNData'><p>Plots parental specific copy numbers along the genome</p></a></li>
<li><a href='#process.AromaTransform'><p>Processes the data set</p></a></li>
<li><a href='#process.ChromosomeExplorer'><p>Generates image files, scripts and dynamic pages for the explorer</p></a></li>
<li><a href='#process.Explorer'><p>Generates image files, scripts and dynamic pages for the explorer</p></a></li>
<li><a href='#processTime'><p>Gets the running time of the R process and its children processes</p></a></li>
<li><a href='#RawAlleleBFractions'><p>The RawAlleleBFractions class</p></a></li>
<li><a href='#RawCopyNumberModel'><p>The RawCopyNumberModel class</p></a></li>
<li><a href='#RawCopyNumbers'><p>The RawCopyNumbers class</p></a></li>
<li><a href='#RawGenomicSignals'><p>The RawGenomicSignals class</p></a></li>
<li><a href='#RawMirroredAlleleBFractions'><p>The RawMirroredAlleleBFractions class</p></a></li>
<li><a href='#RawSequenceReads'><p>The RawSequenceReads class</p></a></li>
<li><a href='#readFooter.AromaTabularBinaryFile'><p>Reads the file footer in XML format into a named nested list</p></a></li>
<li><a href='#segmentByCBS.RawGenomicSignals'><p>Segment copy numbers using the CBS method</p></a></li>
<li><a href='#segmentByGLAD.RawGenomicSignals'><p>Segment copy numbers using the GLAD method</p></a></li>
<li><a href='#segmentByHaarSeg.RawGenomicSignals'><p>Segment copy numbers using the HaarSeg method</p></a></li>
<li><a href='#segmentByMPCBS.RawGenomicSignals'><p>Segment copy numbers using the multi-platform CBS (mpCBS) method</p></a></li>
<li><a href='#SegmentedAlleleBFractions'><p>The SegmentedAlleleBFractions class</p></a></li>
<li><a href='#SegmentedCopyNumbers'><p>The SegmentedCopyNumbers class</p></a></li>
<li><a href='#SegmentedGenomicSignalsInterface'><p>The SegmentedGenomicSignalsInterface class interface</p></a></li>
<li><a href='#setAlias.Explorer'><p>Sets the alias of the output set</p></a></li>
<li><a href='#setArrays.ChromosomeExplorer'><p>Sets the arrays</p></a></li>
<li><a href='#setArrays.Explorer'><p>Sets the arrays</p></a></li>
<li><a href='#UnitAnnotationDataFile'><p>The UnitAnnotationDataFile interface class</p></a></li>
<li><a href='#UnitNamesFile'><p>The UnitNamesFile interface class</p></a></li>
<li><a href='#UnitTypesFile'><p>The UnitTypesFile interface class</p></a></li>
<li><a href='#updateSetupExplorerFile.ChromosomeExplorer'><p>Updates the Javascript file</p></a></li>
<li><a href='#writeDataFrame.AromaUnitSignalBinaryFile'><p>Writes the data file as a tab-delimited text file</p></a></li>
<li><a href='#writeDataFrame.AromaUnitSignalBinarySet'><p>Writes the data set as a tab-delimited text file</p></a></li>
<li><a href='#writeDataFrame.AromaUnitTabularBinaryFile'><p>Writes the data file as a tab-delimited text file</p></a></li>
<li><a href='#writeFooter.AromaTabularBinaryFile'><p>Writes a named nested list to the file footer in XML format</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>3.3.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.1), R.utils (&ge; 2.10.1), R.filesets (&ge; 2.14.0),
R.devices (&ge; 2.16.1)</td>
</tr>
<tr>
<td>DependsNote:</td>
<td>BioC (&gt;= 3.2)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, tools, utils, R.methodsS3 (&ge; 1.8.1), R.oo (&ge; 1.24.0),
R.cache (&ge; 0.14.0), R.rsp (&ge; 0.44.0), matrixStats (&ge;
0.57.0), RColorBrewer (&ge; 1.1-2), PSCBS (&ge; 0.65.0), listenv,
future, BiocManager</td>
</tr>
<tr>
<td>Suggests:</td>
<td>KernSmooth, png (&ge; 0.1-7), Cairo (&ge; 1.5-9), EBImage (&ge;
4.8.3), preprocessCore (&ge; 1.28.0), aroma.light (&ge; 2.2.1),
DNAcopy (&ge; 1.40.0), GLAD (&ge; 2.30.0), sfit (&ge; 0.1.8),
expectile (&ge; 0.2.5), HaarSeg (&ge; 0.0.2), mpcbs (&ge; 1.1.1)</td>
</tr>
<tr>
<td>SuggestsNote:</td>
<td>BioC (&gt;= 3.0), Recommended: aroma.light, DNAcopy, png,
preprocessCore, sfit</td>
</tr>
<tr>
<td>Additional_repositories:</td>
<td><a href="https://henrikbengtsson.r-universe.dev">https://henrikbengtsson.r-universe.dev</a>,
<a href="https://r-forge.r-project.org">https://r-forge.r-project.org</a></td>
</tr>
<tr>
<td>Title:</td>
<td>Core Methods and Classes Used by 'aroma.*' Packages Part of the
Aroma Framework</td>
</tr>
<tr>
<td>Description:</td>
<td>Core methods and classes used by higher-level 'aroma.*' packages
        part of the Aroma Project, e.g. 'aroma.affymetrix' and 'aroma.cn'.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-2.1">LGPL-2.1</a> | <a href="https://www.r-project.org/Licenses/LGPL-3">LGPL-3</a> [expanded from: LGPL (&ge; 2.1)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/HenrikBengtsson/aroma.core">https://github.com/HenrikBengtsson/aroma.core</a>,
<a href="https://www.aroma-project.org/">https://www.aroma-project.org/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/HenrikBengtsson/aroma.core/issues">https://github.com/HenrikBengtsson/aroma.core/issues</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>TRUE</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Microarray, OneChannel, TwoChannel, MultiChannel,
DataImport, DataRepresentation, GUI, Visualization,
Preprocessing, QualityControl, aCGH, CopyNumberVariants</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-19 02:33:17 UTC; henrik</td>
</tr>
<tr>
<td>Author:</td>
<td>Henrik Bengtsson [aut, cre, cph],
  Mark Robinson [ctb],
  Ken Simpson [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Henrik Bengtsson &lt;henrikb@braju.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-19 08:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='aroma.core-package'>Package aroma.core</h2><span id='topic+aroma.core-package'></span><span id='topic+aroma.core'></span>

<h3>Description</h3>

<p>Core methods and classes used by higher-level 'aroma.*' packages         part of the Aroma Project, e.g. 'aroma.affymetrix' and 'aroma.cn'.
</p>
<p>This package is a support package for <span class="pkg">aroma.affymetrix</span>.
This package should be considered to be in an alpha or beta phase.
You should expect the API to be changing over time.
For practical purposes, consider this package a private package.
</p>


<h3>Installation and updates</h3>

<p>This package is available on CRAN, i.e. to install do
<code>install.packages("aroma.core")</code>.
</p>


<h3>License</h3>

<p>The releases of this package is licensed under
LGPL version 2.1 or newer.
</p>
<p>The development code of the packages is under a private licence
(where applicable) and patches sent to the author fall under the
latter license, but will be, if incorporated, released under the
&quot;release&quot; license above.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] H. Bengtsson, <em>The R.oo package - Object-Oriented Programming with References Using Standard R Code</em>, In Kurt Hornik, Friedrich Leisch and Achim Zeileis, editors, Proceedings of the 3rd International Workshop on Distributed Statistical Computing (DSC 2003), March 20-22, Vienna, Austria. <a href="https://www.r-project.org/conferences/DSC-2003/Proceedings/">https://www.r-project.org/conferences/DSC-2003/Proceedings/</a>
<br />
</p>

<hr>
<h2 id='AbstractCNData'>The AbstractCNData class</h2><span id='topic+AbstractCNData'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AbstractCNData</b><br />
</p>
<p><code>data.frame</code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+RichDataFrame">RichDataFrame</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+RawGenomicSignals">RawGenomicSignals</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code>AbstractCNData</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+AbstractPSCNData">AbstractPSCNData</a>, <a href="#topic+NonPairedPSCNData">NonPairedPSCNData</a>, <a href="#topic+PairedPSCNData">PairedPSCNData</a><br />
</p>
<p>public class <b>AbstractCNData</b><br />
extends <a href="#topic+RawGenomicSignals">RawGenomicSignals</a><br />
</p>
<p>An AbstractCNData object holds copy number data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AbstractCNData(chromosome=NULL, x=NULL, y=NULL, ..., name=NULL, .virtuals=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AbstractCNData_+3A_chromosome">chromosome</code></td>
<td>
<p>(Optional) An <code><a href="base.html#topic+integer">integer</a></code> scalar (or a <code><a href="base.html#topic+vector">vector</a></code> of length J),
which can be used to specify which chromosome each locus belongs to
in case multiple chromosomes are segments.
This argument is also used for annotation purposes.</p>
</td></tr>
<tr><td><code id="AbstractCNData_+3A_x">x</code></td>
<td>
<p>Optional <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of J genomic locations.
If <code><a href="base.html#topic+NULL">NULL</a></code>, index locations <code>1:J</code> are used.</p>
</td></tr>
<tr><td><code id="AbstractCNData_+3A_y">y</code></td>
<td>
<p>Optional <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of J genomic locations.</p>
</td></tr>
<tr><td><code id="AbstractCNData_+3A_...">...</code></td>
<td>
<p>Optional named locus-specific signal <code><a href="base.html#topic+vector">vector</a></code>s of length J.</p>
</td></tr>
<tr><td><code id="AbstractCNData_+3A_name">name</code></td>
<td>
<p>Optional <code><a href="base.html#topic+character">character</a></code> string.</p>
</td></tr>
<tr><td><code id="AbstractCNData_+3A_.virtuals">.virtuals</code></td>
<td>
<p>(Internal) a <code><a href="base.html#topic+list">list</a></code> with virtual column name <code><a href="base.html#topic+function">function</a></code>s.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>findLargeGaps</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getLocusData</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPlatform</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setPlatform</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from RawGenomicSignals</b>:<br />
*, +, -, addBy, append, applyBinaryOperator, as.character, as.data.frame, assertOneChromosome, binnedSmoothing, binnedSmoothingByField, clearCache, clone, divideBy, drawDensity, estimateStandardDeviation, extractChromosome, extractChromosomes, extractDataForSegmentation, extractRegion, extractRegions, extractSubset, gaussianSmoothing, getBasicField, getCXY, getChromosome, getChromosomes, getDefaultLocusFields, getLocusFields, getPositions, getSigma, getSignalColumnName, getSignalColumnNames, getSignals, getWeights, getXScale, getXY, getYScale, hasWeights, kernelSmoothing, lines, multiplyBy, nbrOfChromosomes, nbrOfLoci, plot, points, print, segmentByCBS, segmentByGLAD, segmentByHaarSeg, segmentByMPCBS, setBasicField, setSigma, setSignals, setWeights, setXScale, setYScale, signalRange, sort, subtractBy, xMax, xMin, xRange, xSeq, yMax, yMin, yRange
</p>
<p><b>Methods inherited from RichDataFrame</b>:<br />
$, $&lt;-, [, [[, [[&lt;-, as.data.frame, as.list, dim, dropVirtualColumn, getColumnNames, getColumnNamesTranslator, getFullName, getName, getTags, getVirtualColumn, getVirtualColumnFunction, getVirtualColumnNames, hasColumn, hasColumns, hasVirtualColumn, hasVirtualColumns, length, names, newInstance, print, rbind, setAttributes, setColumnNamesMap, setColumnNamesTranslator, setName, setTags, setVirtualColumn, subset, translateColumnNames
</p>
<p><b>Methods inherited from data.frame</b>:<br />
$&lt;-,data.frame-method, $&lt;-, Math, Ops,nonStructure,vector-method, Ops,structure,vector-method, Ops,vector,nonStructure-method, Ops,vector,structure-method, Ops, Summary, [, [&lt;-,data.frame-method, [&lt;-, [[, [[&lt;-,data.frame-method, [[&lt;-, aggregate, anyDuplicated, anyNA, as.NonPairedPSCNData, as.PairedPSCNData, as.data.frame, as.list, as.matrix, as.vector, attachLocally, by, callSegmentationOutliers, cbind, coerce,ANY,list-method, coerce,oldClass,S3-method, dim, dimnames, dimnames&lt;-, dropSegmentationOutliers, droplevels, duplicated, edit, findLargeGaps, format, formula, head, initialize,oldClass-method, is.na, merge, na.exclude, na.omit, plot, print, prompt, rbind, row.names, row.names&lt;-, rowsum, segmentByCBS, segmentByPairedPSCBS, show,oldClass-method, slotsFromS3,data.frame-method, split, split&lt;-, stack, str, subset, summary, t, tail, transform, type.convert, unique, unstack, unwrap, within, wrap, writeDataFrame, xtfrm
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AbstractPSCNData'>The AbstractPSCNData class</h2><span id='topic+AbstractPSCNData'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AbstractPSCNData</b><br />
</p>
<p><code>data.frame</code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+RichDataFrame">RichDataFrame</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+RawGenomicSignals">RawGenomicSignals</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+AbstractCNData">AbstractCNData</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code>AbstractPSCNData</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+NonPairedPSCNData">NonPairedPSCNData</a>, <a href="#topic+PairedPSCNData">PairedPSCNData</a><br />
</p>
<p>public class <b>AbstractPSCNData</b><br />
extends <a href="#topic+AbstractCNData">AbstractCNData</a><br />
</p>
<p>A AbstractPSCNData object holds parent-specific copy number data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AbstractPSCNData(chromosome=NULL, x=NULL, isSNP=NULL, mu=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AbstractPSCNData_+3A_chromosome">chromosome</code></td>
<td>
<p>(Optional) An <code><a href="base.html#topic+integer">integer</a></code> scalar (or a <code><a href="base.html#topic+vector">vector</a></code> of length J),
which can be used to specify which chromosome each locus belongs to
in case multiple chromosomes are segments.
This argument is also used for annotation purposes.</p>
</td></tr>
<tr><td><code id="AbstractPSCNData_+3A_x">x</code></td>
<td>
<p>Optional <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of J genomic locations.
If <code><a href="base.html#topic+NULL">NULL</a></code>, index locations <code>1:J</code> are used.</p>
</td></tr>
<tr><td><code id="AbstractPSCNData_+3A_issnp">isSNP</code></td>
<td>
<p>An optional <code><a href="base.html#topic+logical">logical</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length J specifying
whether each locus is a SNP or not (non-polymorphic loci).</p>
</td></tr>
<tr><td><code id="AbstractPSCNData_+3A_mu">mu</code></td>
<td>
<p>An optional <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of J genotype calls in
{0,1/2,1} for AA, AB, and BB, respectively,
and <code><a href="base.html#topic+NA">NA</a></code> for non-polymorphic loci.</p>
</td></tr>
<tr><td><code id="AbstractPSCNData_+3A_...">...</code></td>
<td>
<p>Optional named locus-specific signal <code><a href="base.html#topic+vector">vector</a></code>s of length J.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from AbstractCNData</b>:<br />
findLargeGaps, getChipType, getLocusData, getPlatform, hasKnownPositions, orderAlongGenome, setChipType, setPlatform
</p>
<p><b>Methods inherited from RawGenomicSignals</b>:<br />
*, +, -, addBy, append, applyBinaryOperator, as.character, as.data.frame, assertOneChromosome, binnedSmoothing, binnedSmoothingByField, clearCache, clone, divideBy, drawDensity, estimateStandardDeviation, extractChromosome, extractChromosomes, extractDataForSegmentation, extractRegion, extractRegions, extractSubset, gaussianSmoothing, getBasicField, getCXY, getChromosome, getChromosomes, getDefaultLocusFields, getLocusFields, getPositions, getSigma, getSignalColumnName, getSignalColumnNames, getSignals, getWeights, getXScale, getXY, getYScale, hasWeights, kernelSmoothing, lines, multiplyBy, nbrOfChromosomes, nbrOfLoci, plot, points, print, segmentByCBS, segmentByGLAD, segmentByHaarSeg, segmentByMPCBS, setBasicField, setSigma, setSignals, setWeights, setXScale, setYScale, signalRange, sort, subtractBy, xMax, xMin, xRange, xSeq, yMax, yMin, yRange
</p>
<p><b>Methods inherited from RichDataFrame</b>:<br />
$, $&lt;-, [, [[, [[&lt;-, as.data.frame, as.list, dim, dropVirtualColumn, getColumnNames, getColumnNamesTranslator, getFullName, getName, getTags, getVirtualColumn, getVirtualColumnFunction, getVirtualColumnNames, hasColumn, hasColumns, hasVirtualColumn, hasVirtualColumns, length, names, newInstance, print, rbind, setAttributes, setColumnNamesMap, setColumnNamesTranslator, setName, setTags, setVirtualColumn, subset, translateColumnNames
</p>
<p><b>Methods inherited from data.frame</b>:<br />
$&lt;-,data.frame-method, $&lt;-, Math, Ops,nonStructure,vector-method, Ops,structure,vector-method, Ops,vector,nonStructure-method, Ops,vector,structure-method, Ops, Summary, [, [&lt;-,data.frame-method, [&lt;-, [[, [[&lt;-,data.frame-method, [[&lt;-, aggregate, anyDuplicated, anyNA, as.NonPairedPSCNData, as.PairedPSCNData, as.data.frame, as.list, as.matrix, as.vector, attachLocally, by, callSegmentationOutliers, cbind, coerce,ANY,list-method, coerce,oldClass,S3-method, dim, dimnames, dimnames&lt;-, dropSegmentationOutliers, droplevels, duplicated, edit, findLargeGaps, format, formula, head, initialize,oldClass-method, is.na, merge, na.exclude, na.omit, plot, print, prompt, rbind, row.names, row.names&lt;-, rowsum, segmentByCBS, segmentByPairedPSCBS, show,oldClass-method, slotsFromS3,data.frame-method, split, split&lt;-, stack, str, subset, summary, t, tail, transform, type.convert, unique, unstack, unwrap, within, wrap, writeDataFrame, xtfrm
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaCellCpgFile'>A binary file holding local CpG density for each cell (probe/feature)</h2><span id='topic+AromaCellCpgFile'></span>

<h3>Description</h3>

<p>A binary file holding local CpG density for each cell (probe/feature).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaCellCpgFile(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaCellCpgFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to constructor of
<code><a href="#topic+AromaCellTabularBinaryFile">AromaCellTabularBinaryFile</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this class does <em>not</em> assume a rectangular chip layout.
In other words, there is no concept of mapping a <em>spatial</em>
location on the array to a cell index and vice versa.
The reason for this to be able to use this class also for
non-rectangular chip types.
</p>


<h3>Author(s)</h3>

<p>Mark Robinson</p>

<hr>
<h2 id='AromaCellPositionFile'>A binary file holding chromosome/position for each cell</h2><span id='topic+AromaCellPositionFile'></span>

<h3>Description</h3>

<p>A binary file holding chromosome/position for each cell.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaCellPositionFile(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaCellPositionFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to constructor of
<code><a href="#topic+AromaCellTabularBinaryFile">AromaCellTabularBinaryFile</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this class does <em>not</em> assume a rectangular chip layout.
In other words, there is no concept of mapping a <em>spatial</em>
location on the array to a cell index and vice versa.
The reason for this to be able to use this class also for
non-rectangular chip types.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaCellTabularBinaryFile'>The AromaCellTabularBinaryFile class</h2><span id='topic+AromaCellTabularBinaryFile'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaCellTabularBinaryFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+ColumnNamesInterface">ColumnNamesInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFile">GenericTabularFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaMicroarrayTabularBinaryFile">AromaMicroarrayTabularBinaryFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AromaCellTabularBinaryFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+AromaCellCpgFile">AromaCellCpgFile</a>, <a href="#topic+AromaCellPositionFile">AromaCellPositionFile</a>, <a href="#topic+AromaCellSequenceFile">AromaCellSequenceFile</a><br />
</p>
<p>public abstract static class <b>AromaCellTabularBinaryFile</b><br />
extends <em><a href="#topic+AromaMicroarrayTabularBinaryFile">AromaMicroarrayTabularBinaryFile</a></em><br />
</p>
<p>An AromaCellTabularBinaryFile is an <code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code> with
the constraint that the rows map one-to-one to the cells (features)
of a microarray.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaCellTabularBinaryFile(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaCellTabularBinaryFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfCells</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AromaMicroarrayTabularBinaryFile</b>:<br />
allocate, as.character, byChipType, findByChipType, getChipType, getFilenameExtension, getPlatform
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaTabularBinaryFile</b>:<br />
[, [&lt;-, [[, allocate, as.character, colApply, colMeans, colStats, colSums, dimnames&lt;-, getBytesPerColumn, getColClasses, getDefaultColumnNames, getRootName, importFrom, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, readFooter, readHeader, readRawFooter, setAttributesByTags, subset, summary, updateData, updateDataColumn, writeFooter, writeRawFooter
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericTabularFile</b>:<br />
[, as.character, dim, extractMatrix, head, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, tail, writeColumnsToFiles
</p>
<p><b>Methods inherited from ColumnNamesInterface</b>:<br />
appendColumnNamesTranslator, appendColumnNamesTranslatorByNULL, appendColumnNamesTranslatorBycharacter, appendColumnNamesTranslatorByfunction, appendColumnNamesTranslatorBylist, clearColumnNamesTranslator, clearListOfColumnNamesTranslators, getColumnNames, getColumnNamesTranslator, getDefaultColumnNames, getListOfColumnNamesTranslators, nbrOfColumns, setColumnNames, setColumnNamesTranslator, setListOfColumnNamesTranslators, updateColumnNames
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+AromaUnitTabularBinaryFile">AromaUnitTabularBinaryFile</a></code>.
</p>

<hr>
<h2 id='AromaGenomeTextFile'>The AromaGenomeTextFile class</h2><span id='topic+AromaGenomeTextFile'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaGenomeTextFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+ColumnNamesInterface">ColumnNamesInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFile">GenericTabularFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+TabularTextFile">TabularTextFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AromaGenomeTextFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AromaUcscGenomeTextFile">AromaUcscGenomeTextFile</a></em><br />
</p>
<p>public abstract static class <b>AromaGenomeTextFile</b><br />
extends <a href="#topic+FileCacheKeyInterface">FileCacheKeyInterface</a><br />
</p>
<p>An AromaGenomeTextFile represents a annotation tabular text file that
specifies the number of bases (nucleotides) per chromosome for a
particular genome/organism.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaGenomeTextFile(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaGenomeTextFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="R.filesets.html#topic+TabularTextFile">TabularTextFile</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>An AromaGenomeTextFile is a tab-delimited text file with a header
containing (at least) column names 'chromosome' and 'nbrOfBases'.
The 'chromosome' column specifies the chromosomes (character strings)
and the 'nbrOfBases' column specifies the lengths (integer) of the
chromosomes in number of bases (nucleotides).
</p>
<p>The filename of an AromaGenomeTextFile should have format
&quot;&lt;genome&gt;,chromosomes(,&lt;tag&gt;)*.txt&quot;,
and be located in annotationData/genomes/&lt;genome&gt;/, e.g.
annotationData/genomes/Human/Human,chromosomes,max,20090503.txt.
</p>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byGenome</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readDataFrame</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from TabularTextFile</b>:<br />
as.character, getCommentChar, getDefaultColumnClassPatterns, getDefaultColumnClasses, getDefaultColumnNames, getHeader, getReadArguments, hasColumnHeader, nbrOfLines, nbrOfRows, readColumnNames, readColumns, readDataFrame, readLines, readRawHeader, setCommentChar
</p>
<p><b>Methods inherited from GenericTabularFile</b>:<br />
[, as.character, dim, extractMatrix, head, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, tail, writeColumnsToFiles
</p>
<p><b>Methods inherited from ColumnNamesInterface</b>:<br />
appendColumnNamesTranslator, appendColumnNamesTranslatorByNULL, appendColumnNamesTranslatorBycharacter, appendColumnNamesTranslatorByfunction, appendColumnNamesTranslatorBylist, clearColumnNamesTranslator, clearListOfColumnNamesTranslators, getColumnNames, getColumnNamesTranslator, getDefaultColumnNames, getListOfColumnNamesTranslators, nbrOfColumns, setColumnNames, setColumnNamesTranslator, setListOfColumnNamesTranslators, updateColumnNames
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>Examples</h3>

<pre><code class='language-R'># Locate a Human,chromosomes(,.*)*.txt file
db &lt;- AromaGenomeTextFile$byGenome("Human")
print(db)

# Read the data
df &lt;- readDataFrame(db)
print(df)
str(df)

# Details on the file format
oopts &lt;- options(width=40)
print(readLines(db))
options(oopts)
</code></pre>

<hr>
<h2 id='AromaMicroarrayDataFile'>The abstract AromaMicroarrayDataFile class</h2><span id='topic+AromaMicroarrayDataFile'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaMicroarrayDataFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AromaMicroarrayDataFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>AromaMicroarrayDataFile</b><br />
extends <a href="#topic+FileCacheKeyInterface">FileCacheKeyInterface</a><br />
</p>
<p>An AromaMicroarrayDataFile object represents a single microarray data
file. Each such file originates from a specific chip type on a given
platform.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaMicroarrayDataFile(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaMicroarrayDataFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPlatform</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>isAverageFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>An object of this class is typically part of an
<code><a href="#topic+AromaMicroarrayDataSet">AromaMicroarrayDataSet</a></code>.
</p>

<hr>
<h2 id='AromaMicroarrayDataSet'>The AromaMicroarrayDataSet class</h2><span id='topic+AromaMicroarrayDataSet'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaMicroarrayDataSet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><em><code>AromaMicroarrayDataSet</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>AromaMicroarrayDataSet</b><br />
extends <a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a><br />
</p>
<p>An AromaMicroarrayDataSet object represents a set of
<code><a href="#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code>s with <em>identical</em> chip types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaMicroarrayDataSet(files=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaMicroarrayDataSet_+3A_files">files</code></td>
<td>
<p>A <code><a href="base.html#topic+list">list</a></code> of <code><a href="#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code>:s.</p>
</td></tr>
<tr><td><code id="AromaMicroarrayDataSet_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as.AromaMicroarrayDataSetList</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as.AromaMicroarrayDataSetTuple</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAverageFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPlatform</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaMicroarrayDataSetTuple'>The AromaMicroarrayDataSetTuple class</h2><span id='topic+AromaMicroarrayDataSetTuple'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaMicroarrayDataSetTuple</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSetList">GenericDataFileSetList</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><em><code>AromaMicroarrayDataSetTuple</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AromaUnitTotalCnBinarySetTuple">AromaUnitTotalCnBinarySetTuple</a></em><br />
</p>
<p>public abstract static class <b>AromaMicroarrayDataSetTuple</b><br />
extends <a href="R.filesets.html#topic+GenericDataFileSetList">GenericDataFileSetList</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaMicroarrayDataSetTuple(..., .setClass="AromaMicroarrayDataSet")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaMicroarrayDataSetTuple_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="R.filesets.html#topic+GenericDataFileSetList">GenericDataFileSetList</a></code>.</p>
</td></tr>
<tr><td><code id="AromaMicroarrayDataSetTuple_+3A_.setclass">.setClass</code></td>
<td>
<p>The name of the class of the input set.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as.AromaMicroarrayDataSetTuple</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipTypes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFullNames</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getSets</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getTags</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+nbrOfChipTypes.AromaMicroarrayDataSetTuple">nbrOfChipTypes</a></code> </td><td style="text-align: left;"> Gets the number of chip types.</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from GenericDataFileSetList</b>:<br />
as, as.GenericDataFileSetList, as.character, as.data.frame, as.list, assertDuplicates, clone, extract, getAsteriskTags, getDefaultFullName, getFileList, getFileListClass, getFullNames, getNames, getSet, getSets, getTags, indexOf, length, nbrOfFiles, nbrOfSets, setTags
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaMicroarrayTabularBinaryFile'>The AromaMicroarrayTabularBinaryFile class</h2><span id='topic+AromaMicroarrayTabularBinaryFile'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaMicroarrayTabularBinaryFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+ColumnNamesInterface">ColumnNamesInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFile">GenericTabularFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AromaMicroarrayTabularBinaryFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+AromaCellCpgFile">AromaCellCpgFile</a>, <a href="#topic+AromaCellPositionFile">AromaCellPositionFile</a>, <a href="#topic+AromaCellSequenceFile">AromaCellSequenceFile</a>, <em><a href="#topic+AromaCellTabularBinaryFile">AromaCellTabularBinaryFile</a></em>, <a href="#topic+AromaUflFile">AromaUflFile</a>, <a href="#topic+AromaUgcFile">AromaUgcFile</a>, <a href="#topic+AromaUgpFile">AromaUgpFile</a>, <em><a href="#topic+AromaUnitChromosomeTabularBinaryFile">AromaUnitChromosomeTabularBinaryFile</a></em>, <a href="#topic+AromaUnitGcContentFile">AromaUnitGcContentFile</a>, <em><a href="#topic+AromaUnitTabularBinaryFile">AromaUnitTabularBinaryFile</a></em><br />
</p>
<p>public abstract static class <b>AromaMicroarrayTabularBinaryFile</b><br />
extends <em><a href="#topic+AromaPlatformInterface">AromaPlatformInterface</a></em><br />
</p>
<p>An AromaMicroarrayTabularBinaryFile is an abstract
<code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaMicroarrayTabularBinaryFile(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaMicroarrayTabularBinaryFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFilenameExtension</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPlatform</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaTabularBinaryFile</b>:<br />
[, [&lt;-, [[, allocate, as.character, colApply, colMeans, colStats, colSums, dimnames&lt;-, getBytesPerColumn, getColClasses, getDefaultColumnNames, getRootName, importFrom, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, readFooter, readHeader, readRawFooter, setAttributesByTags, subset, summary, updateData, updateDataColumn, writeFooter, writeRawFooter
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericTabularFile</b>:<br />
[, as.character, dim, extractMatrix, head, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, tail, writeColumnsToFiles
</p>
<p><b>Methods inherited from ColumnNamesInterface</b>:<br />
appendColumnNamesTranslator, appendColumnNamesTranslatorByNULL, appendColumnNamesTranslatorBycharacter, appendColumnNamesTranslatorByfunction, appendColumnNamesTranslatorBylist, clearColumnNamesTranslator, clearListOfColumnNamesTranslators, getColumnNames, getColumnNamesTranslator, getDefaultColumnNames, getListOfColumnNamesTranslators, nbrOfColumns, setColumnNames, setColumnNamesTranslator, setListOfColumnNamesTranslators, updateColumnNames
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code>.
</p>

<hr>
<h2 id='AromaPlatform'>The AromaPlatform class</h2><span id='topic+AromaPlatform'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaPlatform</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><em><code>AromaPlatform</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>AromaPlatform</b><br />
extends <a href="R.oo.html#topic+Object">Object</a><br />
</p>
<p>An AromaPlatform provides methods for a given platform, e.g.
Affymetrix, Agilent, Illumina.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaPlatform(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaPlatform_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>equals</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAromaUgpFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitNamesFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitTypesFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 library("aroma.affymetrix")
platform &lt;- AromaPlatform$byName("Affymetrix")
print(platform)
stopifnot(getName(platform) == "Affymetrix")


## End(Not run)</code></pre>

<hr>
<h2 id='AromaPlatformInterface'>The AromaPlatformInterface class</h2><span id='topic+AromaPlatformInterface'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaPlatformInterface</b><br />
</p>
<p><code><a href="R.oo.html#topic+Interface">Interface</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><em><code>AromaPlatformInterface</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+AromaCellCpgFile">AromaCellCpgFile</a>, <a href="#topic+AromaCellPositionFile">AromaCellPositionFile</a>, <a href="#topic+AromaCellSequenceFile">AromaCellSequenceFile</a>, <em><a href="#topic+AromaCellTabularBinaryFile">AromaCellTabularBinaryFile</a></em>, <em><a href="#topic+AromaMicroarrayTabularBinaryFile">AromaMicroarrayTabularBinaryFile</a></em>, <a href="#topic+AromaUflFile">AromaUflFile</a>, <a href="#topic+AromaUgcFile">AromaUgcFile</a>, <a href="#topic+AromaUgpFile">AromaUgpFile</a>, <a href="#topic+AromaUnitCallFile">AromaUnitCallFile</a>, <em><a href="#topic+AromaUnitChromosomeTabularBinaryFile">AromaUnitChromosomeTabularBinaryFile</a></em>, <a href="#topic+AromaUnitFracBCnBinaryFile">AromaUnitFracBCnBinaryFile</a>, <a href="#topic+AromaUnitGcContentFile">AromaUnitGcContentFile</a>, <a href="#topic+AromaUnitGenotypeCallFile">AromaUnitGenotypeCallFile</a>, <em><a href="#topic+AromaUnitPscnBinaryFile">AromaUnitPscnBinaryFile</a></em>, <a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a>, <em><a href="#topic+AromaUnitTabularBinaryFile">AromaUnitTabularBinaryFile</a></em>, <em><a href="#topic+AromaUnitTotalCnBinaryFile">AromaUnitTotalCnBinaryFile</a></em>, <a href="#topic+AromaUnitTypesFile">AromaUnitTypesFile</a><br />
</p>
<p>public abstract class <b>AromaPlatformInterface</b><br />
extends <a href="R.oo.html#topic+Interface">Interface</a><br />
</p>
<p>An AromaPlatformInterface provides methods for a given platform, e.g.
Affymetrix, Agilent, Illumina.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaPlatformInterface(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaPlatformInterface_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getAromaPlatform.AromaPlatformInterface">getAromaPlatform</a></code> </td><td style="text-align: left;"> Gets the platform.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAromaUflFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAromaUgpFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getChipType.AromaPlatformInterface">getChipType</a></code> </td><td style="text-align: left;"> Gets the chip type.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getPlatform.AromaPlatformInterface">getPlatform</a></code> </td><td style="text-align: left;"> Gets the platform.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitNamesFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitTypesFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from Interface</b>:<br />
extend, print, uses
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaRepository'>The AromaRepository class</h2><span id='topic+AromaRepository'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaRepository</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code>AromaRepository</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>AromaRepository</b><br />
extends <a href="R.oo.html#topic+Object">Object</a><br />
</p>
<p>An AromaRepository object provides methods for downloading annotation data
from the Aroma repository.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaRepository(urlPath="https://www.aroma-project.org/data", verbose=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaRepository_+3A_urlpath">urlPath</code></td>
<td>
<p>The URL to the Aroma repository.</p>
</td></tr>
<tr><td><code id="AromaRepository_+3A_verbose">verbose</code></td>
<td>
<p>The <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> to be used during processing.</p>
</td></tr>
<tr><td><code id="AromaRepository_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>downloadACC</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>downloadACM</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>downloadACP</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>downloadACS</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>downloadAll</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>downloadCDF</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+downloadChipTypeFile.AromaRepository">downloadChipTypeFile</a></code> </td><td style="text-align: left;"> Download a particular chip type annotation file.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>downloadProbeSeqsTXT</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>downloadTXT</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>downloadUFL</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>downloadUGC</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>downloadUGP</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaTabularBinaryFile'>The AromaTabularBinaryFile class</h2><span id='topic+AromaTabularBinaryFile'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaTabularBinaryFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+ColumnNamesInterface">ColumnNamesInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFile">GenericTabularFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AromaTabularBinaryFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+AromaCellCpgFile">AromaCellCpgFile</a>, <a href="#topic+AromaCellPositionFile">AromaCellPositionFile</a>, <a href="#topic+AromaCellSequenceFile">AromaCellSequenceFile</a>, <em><a href="#topic+AromaCellTabularBinaryFile">AromaCellTabularBinaryFile</a></em>, <em><a href="#topic+AromaMicroarrayTabularBinaryFile">AromaMicroarrayTabularBinaryFile</a></em>, <a href="#topic+AromaUflFile">AromaUflFile</a>, <a href="#topic+AromaUgcFile">AromaUgcFile</a>, <a href="#topic+AromaUgpFile">AromaUgpFile</a>, <a href="#topic+AromaUnitCallFile">AromaUnitCallFile</a>, <em><a href="#topic+AromaUnitChromosomeTabularBinaryFile">AromaUnitChromosomeTabularBinaryFile</a></em>, <a href="#topic+AromaUnitFracBCnBinaryFile">AromaUnitFracBCnBinaryFile</a>, <a href="#topic+AromaUnitGcContentFile">AromaUnitGcContentFile</a>, <a href="#topic+AromaUnitGenotypeCallFile">AromaUnitGenotypeCallFile</a>, <em><a href="#topic+AromaUnitPscnBinaryFile">AromaUnitPscnBinaryFile</a></em>, <a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a>, <em><a href="#topic+AromaUnitTabularBinaryFile">AromaUnitTabularBinaryFile</a></em>, <em><a href="#topic+AromaUnitTotalCnBinaryFile">AromaUnitTotalCnBinaryFile</a></em>, <a href="#topic+AromaUnitTypesFile">AromaUnitTypesFile</a><br />
</p>
<p>public abstract static class <b>AromaTabularBinaryFile</b><br />
extends <a href="#topic+FileCacheKeyInterface">FileCacheKeyInterface</a><br />
</p>
<p>A AromaTabularBinaryFile represents a file with a binary format.
It has a well defined header, a data section, and a footer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaTabularBinaryFile(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaTabularBinaryFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="R.filesets.html#topic+GenericTabularFile">GenericTabularFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>[</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>[&lt;-</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>colMeans</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>colSums</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getBytesPerColumn</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getColClasses</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>importFrom</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfColumns</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfRows</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readColumns</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+readFooter.AromaTabularBinaryFile">readFooter</a></code> </td><td style="text-align: left;"> Reads the file footer in XML format into a named nested list.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>subset</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>summary</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+writeFooter.AromaTabularBinaryFile">writeFooter</a></code> </td><td style="text-align: left;"> Writes a named nested list to the file footer in XML format.</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericTabularFile</b>:<br />
[, as.character, dim, extractMatrix, head, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, tail, writeColumnsToFiles
</p>
<p><b>Methods inherited from ColumnNamesInterface</b>:<br />
appendColumnNamesTranslator, appendColumnNamesTranslatorByNULL, appendColumnNamesTranslatorBycharacter, appendColumnNamesTranslatorByfunction, appendColumnNamesTranslatorBylist, clearColumnNamesTranslator, clearListOfColumnNamesTranslators, getColumnNames, getColumnNamesTranslator, getDefaultColumnNames, getListOfColumnNamesTranslators, nbrOfColumns, setColumnNames, setColumnNamesTranslator, setListOfColumnNamesTranslators, updateColumnNames
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code>.
</p>

<hr>
<h2 id='AromaTabularBinaryFile+24allocate'>Creates an AromaTabularBinaryFile</h2><span id='topic+AromaTabularBinaryFile+24allocate'></span><span id='topic+allocate.AromaTabularBinaryFile'></span><span id='topic+AromaTabularBinaryFile.allocate'></span><span id='topic+allocate+2CAromaTabularBinaryFile-method'></span>

<h3>Description</h3>

<p>Creates an AromaTabularBinaryFile of a certain dimension and data column types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## AromaTabularBinaryFile$allocate(filename, path=NULL, nbrOfRows, types, sizes,
##   signeds=TRUE, defaults=NA, comment=NULL, overwrite=FALSE, skip=FALSE, footer=list(),
##   ..., verbose=FALSE)

## Don't use the below:
## S3 method for class 'AromaTabularBinaryFile'
allocate(static, filename, path=NULL, nbrOfRows, types, sizes, signeds=TRUE, defaults=NA,
  comment=NULL, overwrite=FALSE, skip=FALSE, footer=list(), ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaTabularBinaryFile+2B24allocate_+3A_filename">filename</code></td>
<td>
<p>The filename of the new file.</p>
</td></tr>
<tr><td><code id="AromaTabularBinaryFile+2B24allocate_+3A_path">path</code></td>
<td>
<p>The path where to store the new file.</p>
</td></tr>
<tr><td><code id="AromaTabularBinaryFile+2B24allocate_+3A_nbrofrows">nbrOfRows</code></td>
<td>
<p>An <code><a href="base.html#topic+integer">integer</a></code> specifying the number of rows to allocate.</p>
</td></tr>
<tr><td><code id="AromaTabularBinaryFile+2B24allocate_+3A_types">types</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying the data type of each
column.  The length specifies the number of columns to allocate.</p>
</td></tr>
<tr><td><code id="AromaTabularBinaryFile+2B24allocate_+3A_sizes">sizes</code></td>
<td>
<p>An <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> of values in {1,2,4,8} specifying
the size of each column (data type).</p>
</td></tr>
<tr><td><code id="AromaTabularBinaryFile+2B24allocate_+3A_signeds">signeds</code></td>
<td>
<p>An <code><a href="base.html#topic+logical">logical</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying if the data types in each
column is signed or not.</p>
</td></tr>
<tr><td><code id="AromaTabularBinaryFile+2B24allocate_+3A_defaults">defaults</code></td>
<td>
<p>An optional <code><a href="base.html#topic+list">list</a></code> (or <code><a href="base.html#topic+vector">vector</a></code>) containing default
values for each column.</p>
</td></tr>
<tr><td><code id="AromaTabularBinaryFile+2B24allocate_+3A_comment">comment</code></td>
<td>
<p>An optional <code><a href="base.html#topic+character">character</a></code> string written to the file header.</p>
</td></tr>
<tr><td><code id="AromaTabularBinaryFile+2B24allocate_+3A_overwrite">overwrite</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, an existing file is overwritten, otherwise not.</p>
</td></tr>
<tr><td><code id="AromaTabularBinaryFile+2B24allocate_+3A_skip">skip</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code> and <code>overwrite=TRUE</code>, any existing file is
returned as is.</p>
</td></tr>
<tr><td><code id="AromaTabularBinaryFile+2B24allocate_+3A_footer">footer</code></td>
<td>
<p>An optional <code><a href="base.html#topic+list">list</a></code> of attributes written (as character
strings) to the file footer.</p>
</td></tr>
<tr><td><code id="AromaTabularBinaryFile+2B24allocate_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="AromaTabularBinaryFile+2B24allocate_+3A_verbose">verbose</code></td>
<td>
<p><code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code> object.
</p>


<h3>Data types</h3>

<p>Valid data types are currently &quot;<code><a href="base.html#topic+integer">integer</a></code>&quot; and &quot;<code><a href="base.html#topic+double">double</a></code>&quot;.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code>.
</p>

<hr>
<h2 id='AromaTabularBinarySet'>The AromaTabularBinarySet class</h2><span id='topic+AromaTabularBinarySet'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaTabularBinarySet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFileSet">GenericTabularFileSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code>AromaTabularBinarySet</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+AromaUnitCallSet">AromaUnitCallSet</a>, <a href="#topic+AromaUnitFracBCnBinarySet">AromaUnitFracBCnBinarySet</a>, <a href="#topic+AromaUnitGenotypeCallSet">AromaUnitGenotypeCallSet</a>, <a href="#topic+AromaUnitPscnBinarySet">AromaUnitPscnBinarySet</a>, <a href="#topic+AromaUnitSignalBinarySet">AromaUnitSignalBinarySet</a>, <a href="#topic+AromaUnitTotalCnBinarySet">AromaUnitTotalCnBinarySet</a><br />
</p>
<p>public static class <b>AromaTabularBinarySet</b><br />
extends <a href="R.filesets.html#topic+GenericTabularFileSet">GenericTabularFileSet</a><br />
</p>
<p>An AromaTabularBinarySet object represents a set of
<code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code>s with <em>identical</em> chip types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaTabularBinarySet(files=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaTabularBinarySet_+3A_files">files</code></td>
<td>
<p>A <code><a href="base.html#topic+list">list</a></code> of <code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code>:s.</p>
</td></tr>
<tr><td><code id="AromaTabularBinarySet_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from GenericTabularFileSet</b>:<br />
extractMatrix, calculateAverageColumnAcrossFiles
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaTransform'>The AromaTransform class</h2><span id='topic+AromaTransform'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaTransform</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><em><code>AromaTransform</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>AromaTransform</b><br />
extends <a href="#topic+ParametersInterface">ParametersInterface</a><br />
</p>
<p>This abstract class represents a transform (algorithm/operator) that
transforms data.  A transform has an input data set, which is
transformed into an output data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaTransform(dataSet=NULL, tags="*", ..., .reqSetClass="AromaMicroarrayDataSet")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaTransform_+3A_dataset">dataSet</code></td>
<td>
<p>The input data set as an <code><a href="#topic+AromaMicroarrayDataSet">AromaMicroarrayDataSet</a></code>.</p>
</td></tr>
<tr><td><code id="AromaTransform_+3A_tags">tags</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> of tags to be appended to the tags of
the input data set.</p>
</td></tr>
<tr><td><code id="AromaTransform_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="AromaTransform_+3A_.reqsetclass">.reqSetClass</code></td>
<td>
<p>Internal argument.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Subclasses must implement the <code>process()</code> method.
</p>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getFullName.AromaTransform">getFullName</a></code> </td><td style="text-align: left;"> Gets the full name of the output data set.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getInputDataSet.AromaTransform">getInputDataSet</a></code> </td><td style="text-align: left;"> Gets the input data set.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getName.AromaTransform">getName</a></code> </td><td style="text-align: left;"> Gets the name of the output data set.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getOutputDataSet.AromaTransform">getOutputDataSet</a></code> </td><td style="text-align: left;"> Gets the transformed data set.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getPath.AromaTransform">getPath</a></code> </td><td style="text-align: left;"> Gets the path of the output directory.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getRootPath.AromaTransform">getRootPath</a></code> </td><td style="text-align: left;"> Gets the root path of the output directory.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getTags.AromaTransform">getTags</a></code> </td><td style="text-align: left;"> Gets the tags of the output data set.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+isDone.AromaTransform">isDone</a></code> </td><td style="text-align: left;"> Checks if the data set is processed or not.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+process.AromaTransform">process</a></code> </td><td style="text-align: left;"> Processes the data set.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setTags</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaUnitCallFile'>The AromaUnitCallFile class</h2><span id='topic+AromaUnitCallFile'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaUnitCallFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+ColumnNamesInterface">ColumnNamesInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFile">GenericTabularFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>AromaUnitCallFile</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+AromaUnitGenotypeCallFile">AromaUnitGenotypeCallFile</a><br />
</p>
<p>public static class <b>AromaUnitCallFile</b><br />
extends <a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a><br />
</p>
<p>An AromaUnitCallFile is a <code><a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaUnitCallFile(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaUnitCallFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractCallArray</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractCalls</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractMatrix</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>findUnitsTodo</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AromaUnitSignalBinaryFile</b>:<br />
allocate, allocateFromUnitAnnotationDataFile, allocateFromUnitNamesFile, as.character, extractMatrix, extractRawGenomicSignals, fromFile, getChipType, getExtensionPattern, getFilenameExtension, getNumberOfFilesAveraged, getPlatform, isAverageFile, nbrOfUnits, readDataFrame, writeDataFrame
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaTabularBinaryFile</b>:<br />
[, [&lt;-, [[, allocate, as.character, colApply, colMeans, colStats, colSums, dimnames&lt;-, getBytesPerColumn, getColClasses, getDefaultColumnNames, getRootName, importFrom, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, readFooter, readHeader, readRawFooter, setAttributesByTags, subset, summary, updateData, updateDataColumn, writeFooter, writeRawFooter
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericTabularFile</b>:<br />
[, as.character, dim, extractMatrix, head, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, tail, writeColumnsToFiles
</p>
<p><b>Methods inherited from ColumnNamesInterface</b>:<br />
appendColumnNamesTranslator, appendColumnNamesTranslatorByNULL, appendColumnNamesTranslatorBycharacter, appendColumnNamesTranslatorByfunction, appendColumnNamesTranslatorBylist, clearColumnNamesTranslator, clearListOfColumnNamesTranslators, getColumnNames, getColumnNamesTranslator, getDefaultColumnNames, getListOfColumnNamesTranslators, nbrOfColumns, setColumnNames, setColumnNamesTranslator, setListOfColumnNamesTranslators, updateColumnNames
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaUnitCallSet'>The AromaUnitCallSet class</h2><span id='topic+AromaUnitCallSet'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaUnitCallSet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFileSet">GenericTabularFileSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaTabularBinarySet">AromaTabularBinarySet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaUnitSignalBinarySet">AromaUnitSignalBinarySet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>AromaUnitCallSet</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+AromaUnitGenotypeCallSet">AromaUnitGenotypeCallSet</a><br />
</p>
<p>public static class <b>AromaUnitCallSet</b><br />
extends <a href="#topic+AromaUnitSignalBinarySet">AromaUnitSignalBinarySet</a><br />
</p>
<p>An AromaUnitCallSet object represents a set of
<code><a href="#topic+AromaUnitCallFile">AromaUnitCallFile</a></code>s with <em>identical</em> chip types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaUnitCallSet(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaUnitCallSet_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AromaUnitSignalBinarySet">AromaUnitSignalBinarySet</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractCallArray</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractCalls</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractGenotypeMatrix</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>findUnitsTodo</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AromaUnitSignalBinarySet</b>:<br />
byName, findByName, getAromaFullNameTranslatorSet, getAromaUgpFile, getChipType, getPlatform, validate, writeDataFrame
</p>
<p><b>Methods inherited from AromaTabularBinarySet</b>:<br />
getDefaultFullName, getRootName, setAttributesBy, setAttributesBySampleAnnotationFile, setAttributesBySampleAnnotationSet, setAttributesByTags
</p>
<p><b>Methods inherited from GenericTabularFileSet</b>:<br />
extractMatrix, calculateAverageColumnAcrossFiles
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaUnitFracBCnBinaryFile'>The AromaUnitFracBCnBinaryFile class</h2><span id='topic+AromaUnitFracBCnBinaryFile'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaUnitFracBCnBinaryFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+ColumnNamesInterface">ColumnNamesInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFile">GenericTabularFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>AromaUnitFracBCnBinaryFile</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>AromaUnitFracBCnBinaryFile</b><br />
extends <a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a><br />
</p>
<p>An AromaUnitFracBCnBinaryFile is a <code><a href="#topic+AromaUnitTabularBinaryFile">AromaUnitTabularBinaryFile</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaUnitFracBCnBinaryFile(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaUnitFracBCnBinaryFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AromaUnitTabularBinaryFile">AromaUnitTabularBinaryFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractRawAlleleBFractions</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AromaUnitSignalBinaryFile</b>:<br />
allocate, allocateFromUnitAnnotationDataFile, allocateFromUnitNamesFile, as.character, extractMatrix, extractRawGenomicSignals, fromFile, getChipType, getExtensionPattern, getFilenameExtension, getNumberOfFilesAveraged, getPlatform, isAverageFile, nbrOfUnits, readDataFrame, writeDataFrame
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaTabularBinaryFile</b>:<br />
[, [&lt;-, [[, allocate, as.character, colApply, colMeans, colStats, colSums, dimnames&lt;-, getBytesPerColumn, getColClasses, getDefaultColumnNames, getRootName, importFrom, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, readFooter, readHeader, readRawFooter, setAttributesByTags, subset, summary, updateData, updateDataColumn, writeFooter, writeRawFooter
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericTabularFile</b>:<br />
[, as.character, dim, extractMatrix, head, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, tail, writeColumnsToFiles
</p>
<p><b>Methods inherited from ColumnNamesInterface</b>:<br />
appendColumnNamesTranslator, appendColumnNamesTranslatorByNULL, appendColumnNamesTranslatorBycharacter, appendColumnNamesTranslatorByfunction, appendColumnNamesTranslatorBylist, clearColumnNamesTranslator, clearListOfColumnNamesTranslators, getColumnNames, getColumnNamesTranslator, getDefaultColumnNames, getListOfColumnNamesTranslators, nbrOfColumns, setColumnNames, setColumnNamesTranslator, setListOfColumnNamesTranslators, updateColumnNames
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaUnitFracBCnBinarySet'>The AromaUnitFracBCnBinarySet class</h2><span id='topic+AromaUnitFracBCnBinarySet'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaUnitFracBCnBinarySet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFileSet">GenericTabularFileSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaTabularBinarySet">AromaTabularBinarySet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaUnitSignalBinarySet">AromaUnitSignalBinarySet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>AromaUnitFracBCnBinarySet</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>AromaUnitFracBCnBinarySet</b><br />
extends <a href="#topic+AromaUnitSignalBinarySet">AromaUnitSignalBinarySet</a><br />
</p>
<p>An AromaUnitFracBCnBinarySet object represents a set of
<code><a href="#topic+AromaUnitFracBCnBinaryFile">AromaUnitFracBCnBinaryFile</a></code>s with <em>identical</em> chip types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaUnitFracBCnBinarySet(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaUnitFracBCnBinarySet_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AromaUnitSignalBinarySet">AromaUnitSignalBinarySet</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The term &quot;allele B fraction&quot; is also know as &quot;allele B frequency&quot;, which
was coined by Peiffer et al. (2006).  Note that the term &quot;frequency&quot; is
a bit misleading since it is not a frequency in neither the statistical
nor the population sense, but rather only proportion relative to the
total amount of allele A and allele B signals, which is calculated for
each sample independently.
</p>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>writeDataFrame</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AromaUnitSignalBinarySet</b>:<br />
byName, findByName, getAromaFullNameTranslatorSet, getAromaUgpFile, getChipType, getPlatform, validate, writeDataFrame
</p>
<p><b>Methods inherited from AromaTabularBinarySet</b>:<br />
getDefaultFullName, getRootName, setAttributesBy, setAttributesBySampleAnnotationFile, setAttributesBySampleAnnotationSet, setAttributesByTags
</p>
<p><b>Methods inherited from GenericTabularFileSet</b>:<br />
extractMatrix, calculateAverageColumnAcrossFiles
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] Peiffer et al., <em>High-resolution genomic profiling of
chromosomal aberrations using Infinium whole-genome genotyping</em>,
Genome Res, 2006.<br />
</p>

<hr>
<h2 id='AromaUnitGenotypeCallFile'>The AromaUnitGenotypeCallFile class</h2><span id='topic+AromaUnitGenotypeCallFile'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaUnitGenotypeCallFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+ColumnNamesInterface">ColumnNamesInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFile">GenericTabularFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaUnitCallFile">AromaUnitCallFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>AromaUnitGenotypeCallFile</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>AromaUnitGenotypeCallFile</b><br />
extends <a href="#topic+AromaUnitCallFile">AromaUnitCallFile</a><br />
</p>
<p>An AromaUnitGenotypeCallFile is a <code><a href="#topic+AromaUnitTabularBinaryFile">AromaUnitTabularBinaryFile</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaUnitGenotypeCallFile(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaUnitGenotypeCallFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AromaUnitTabularBinaryFile">AromaUnitTabularBinaryFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractGenotypeMatrix</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractGenotypes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>isHeterozygous</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>isHomozygous</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>updateGenotypes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AromaUnitCallFile</b>:<br />
allocate, extractCallArray, extractCalls, extractMatrix, findUnitsTodo
</p>
<p><b>Methods inherited from AromaUnitSignalBinaryFile</b>:<br />
allocate, allocateFromUnitAnnotationDataFile, allocateFromUnitNamesFile, as.character, extractMatrix, extractRawGenomicSignals, fromFile, getChipType, getExtensionPattern, getFilenameExtension, getNumberOfFilesAveraged, getPlatform, isAverageFile, nbrOfUnits, readDataFrame, writeDataFrame
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaTabularBinaryFile</b>:<br />
[, [&lt;-, [[, allocate, as.character, colApply, colMeans, colStats, colSums, dimnames&lt;-, getBytesPerColumn, getColClasses, getDefaultColumnNames, getRootName, importFrom, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, readFooter, readHeader, readRawFooter, setAttributesByTags, subset, summary, updateData, updateDataColumn, writeFooter, writeRawFooter
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericTabularFile</b>:<br />
[, as.character, dim, extractMatrix, head, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, tail, writeColumnsToFiles
</p>
<p><b>Methods inherited from ColumnNamesInterface</b>:<br />
appendColumnNamesTranslator, appendColumnNamesTranslatorByNULL, appendColumnNamesTranslatorBycharacter, appendColumnNamesTranslatorByfunction, appendColumnNamesTranslatorBylist, clearColumnNamesTranslator, clearListOfColumnNamesTranslators, getColumnNames, getColumnNamesTranslator, getDefaultColumnNames, getListOfColumnNamesTranslators, nbrOfColumns, setColumnNames, setColumnNamesTranslator, setListOfColumnNamesTranslators, updateColumnNames
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaUnitGenotypeCallSet'>The AromaUnitGenotypeCallSet class</h2><span id='topic+AromaUnitGenotypeCallSet'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaUnitGenotypeCallSet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFileSet">GenericTabularFileSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaTabularBinarySet">AromaTabularBinarySet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaUnitSignalBinarySet">AromaUnitSignalBinarySet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaUnitCallSet">AromaUnitCallSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>AromaUnitGenotypeCallSet</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>AromaUnitGenotypeCallSet</b><br />
extends <a href="#topic+AromaUnitCallSet">AromaUnitCallSet</a><br />
</p>
<p>An AromaUnitGenotypeCallSet object represents a set of
<code><a href="#topic+AromaUnitGenotypeCallFile">AromaUnitGenotypeCallFile</a></code>s with <em>identical</em> chip types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaUnitGenotypeCallSet(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaUnitGenotypeCallSet_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AromaUnitCallSet">AromaUnitCallSet</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractGenotypes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AromaUnitCallSet</b>:<br />
byPath, extractCallArray, extractCalls, extractGenotypeMatrix, findByName, findUnitsTodo
</p>
<p><b>Methods inherited from AromaUnitSignalBinarySet</b>:<br />
byName, findByName, getAromaFullNameTranslatorSet, getAromaUgpFile, getChipType, getPlatform, validate, writeDataFrame
</p>
<p><b>Methods inherited from AromaTabularBinarySet</b>:<br />
getDefaultFullName, getRootName, setAttributesBy, setAttributesBySampleAnnotationFile, setAttributesBySampleAnnotationSet, setAttributesByTags
</p>
<p><b>Methods inherited from GenericTabularFileSet</b>:<br />
extractMatrix, calculateAverageColumnAcrossFiles
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaUnitPscnBinaryFile'>The AromaUnitPscnBinaryFile class</h2><span id='topic+AromaUnitPscnBinaryFile'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaUnitPscnBinaryFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+ColumnNamesInterface">ColumnNamesInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFile">GenericTabularFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CopyNumberDataFile">CopyNumberDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AromaUnitPscnBinaryFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>AromaUnitPscnBinaryFile</b><br />
extends <em><a href="#topic+CopyNumberDataFile">CopyNumberDataFile</a></em><br />
</p>
<p>An AromaUnitPscnBinaryFile is a <code><a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a></code>
that holds total copy number signals (TCNs) and allele B fractions (BAFs).
The TCNs can either be on an unknown scale or ratios relative
to a reference.  The signals are always stored on the original scale,
i.e. they are never stored on the logaritmic scale.
The BAFs are always on a [0-eps,1+eps] scale, where eps &gt;= 0.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaUnitPscnBinaryFile(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaUnitPscnBinaryFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractRawCopyNumbers</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>hasAlleleBFractions</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>hasStrandiness</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>hasTotalCopyNumberRatios</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from CopyNumberDataFile</b>:<br />
as, as.CopyNumberDataFile, getNumberOfFilesAveraged, hasAlleleBFractions, hasStrandiness
</p>
<p><b>Methods inherited from AromaUnitSignalBinaryFile</b>:<br />
allocate, allocateFromUnitAnnotationDataFile, allocateFromUnitNamesFile, as.character, extractMatrix, extractRawGenomicSignals, fromFile, getChipType, getExtensionPattern, getFilenameExtension, getNumberOfFilesAveraged, getPlatform, isAverageFile, nbrOfUnits, readDataFrame, writeDataFrame
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaTabularBinaryFile</b>:<br />
[, [&lt;-, [[, allocate, as.character, colApply, colMeans, colStats, colSums, dimnames&lt;-, getBytesPerColumn, getColClasses, getDefaultColumnNames, getRootName, importFrom, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, readFooter, readHeader, readRawFooter, setAttributesByTags, subset, summary, updateData, updateDataColumn, writeFooter, writeRawFooter
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericTabularFile</b>:<br />
[, as.character, dim, extractMatrix, head, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, tail, writeColumnsToFiles
</p>
<p><b>Methods inherited from ColumnNamesInterface</b>:<br />
appendColumnNamesTranslator, appendColumnNamesTranslatorByNULL, appendColumnNamesTranslatorBycharacter, appendColumnNamesTranslatorByfunction, appendColumnNamesTranslatorBylist, clearColumnNamesTranslator, clearListOfColumnNamesTranslators, getColumnNames, getColumnNamesTranslator, getDefaultColumnNames, getListOfColumnNamesTranslators, nbrOfColumns, setColumnNames, setColumnNamesTranslator, setListOfColumnNamesTranslators, updateColumnNames
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaUnitPscnBinarySet'>The AromaUnitPscnBinarySet class</h2><span id='topic+AromaUnitPscnBinarySet'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaUnitPscnBinarySet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFileSet">GenericTabularFileSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaTabularBinarySet">AromaTabularBinarySet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaUnitSignalBinarySet">AromaUnitSignalBinarySet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CopyNumberDataSet">CopyNumberDataSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>AromaUnitPscnBinarySet</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>AromaUnitPscnBinarySet</b><br />
extends <a href="#topic+CopyNumberDataSet">CopyNumberDataSet</a><br />
</p>
<p>An AromaUnitPscnBinarySet object represents a set of
<code><a href="#topic+AromaUnitPscnBinaryFile">AromaUnitPscnBinaryFile</a></code>s with <em>identical</em> chip types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaUnitPscnBinarySet(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaUnitPscnBinarySet_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AromaUnitSignalBinarySet">AromaUnitSignalBinarySet</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAverageFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>writeDataFrame</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from CopyNumberDataSet</b>:<br />
as, as.CopyNumberDataSet, doCBS, hasAlleleBFractions, hasStrandiness
</p>
<p><b>Methods inherited from AromaUnitSignalBinarySet</b>:<br />
byName, findByName, getAromaFullNameTranslatorSet, getAromaUgpFile, getChipType, getPlatform, validate, writeDataFrame
</p>
<p><b>Methods inherited from AromaTabularBinarySet</b>:<br />
getDefaultFullName, getRootName, setAttributesBy, setAttributesBySampleAnnotationFile, setAttributesBySampleAnnotationSet, setAttributesByTags
</p>
<p><b>Methods inherited from GenericTabularFileSet</b>:<br />
extractMatrix, calculateAverageColumnAcrossFiles
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaUnitSignalBinaryFile'>The AromaUnitSignalBinaryFile class</h2><span id='topic+AromaUnitSignalBinaryFile'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaUnitSignalBinaryFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+ColumnNamesInterface">ColumnNamesInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFile">GenericTabularFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>AromaUnitSignalBinaryFile</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+AromaUnitCallFile">AromaUnitCallFile</a>, <a href="#topic+AromaUnitFracBCnBinaryFile">AromaUnitFracBCnBinaryFile</a>, <a href="#topic+AromaUnitGenotypeCallFile">AromaUnitGenotypeCallFile</a>, <em><a href="#topic+AromaUnitPscnBinaryFile">AromaUnitPscnBinaryFile</a></em>, <em><a href="#topic+AromaUnitTotalCnBinaryFile">AromaUnitTotalCnBinaryFile</a></em>, <a href="#topic+AromaUnitTypesFile">AromaUnitTypesFile</a><br />
</p>
<p>public static class <b>AromaUnitSignalBinaryFile</b><br />
extends <em><a href="#topic+AromaPlatformInterface">AromaPlatformInterface</a></em><br />
</p>
<p>An AromaUnitSignalBinaryFile is a <code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaUnitSignalBinaryFile(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaUnitSignalBinaryFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractMatrix</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getNumberOfFilesAveraged</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPlatform</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>isAverageFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readDataFrame</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+writeDataFrame.AromaUnitSignalBinaryFile">writeDataFrame</a></code> </td><td style="text-align: left;"> Writes the data file as a tab-delimited text file.</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaTabularBinaryFile</b>:<br />
[, [&lt;-, [[, allocate, as.character, colApply, colMeans, colStats, colSums, dimnames&lt;-, getBytesPerColumn, getColClasses, getDefaultColumnNames, getRootName, importFrom, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, readFooter, readHeader, readRawFooter, setAttributesByTags, subset, summary, updateData, updateDataColumn, writeFooter, writeRawFooter
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericTabularFile</b>:<br />
[, as.character, dim, extractMatrix, head, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, tail, writeColumnsToFiles
</p>
<p><b>Methods inherited from ColumnNamesInterface</b>:<br />
appendColumnNamesTranslator, appendColumnNamesTranslatorByNULL, appendColumnNamesTranslatorBycharacter, appendColumnNamesTranslatorByfunction, appendColumnNamesTranslatorBylist, clearColumnNamesTranslator, clearListOfColumnNamesTranslators, getColumnNames, getColumnNamesTranslator, getDefaultColumnNames, getListOfColumnNamesTranslators, nbrOfColumns, setColumnNames, setColumnNamesTranslator, setListOfColumnNamesTranslators, updateColumnNames
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code>.
</p>

<hr>
<h2 id='AromaUnitSignalBinarySet'>The AromaUnitSignalBinarySet class</h2><span id='topic+AromaUnitSignalBinarySet'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaUnitSignalBinarySet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFileSet">GenericTabularFileSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaTabularBinarySet">AromaTabularBinarySet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code>AromaUnitSignalBinarySet</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+AromaUnitCallSet">AromaUnitCallSet</a>, <a href="#topic+AromaUnitFracBCnBinarySet">AromaUnitFracBCnBinarySet</a>, <a href="#topic+AromaUnitGenotypeCallSet">AromaUnitGenotypeCallSet</a>, <a href="#topic+AromaUnitPscnBinarySet">AromaUnitPscnBinarySet</a>, <a href="#topic+AromaUnitTotalCnBinarySet">AromaUnitTotalCnBinarySet</a><br />
</p>
<p>public static class <b>AromaUnitSignalBinarySet</b><br />
extends <a href="#topic+AromaTabularBinarySet">AromaTabularBinarySet</a><br />
</p>
<p>An AromaUnitSignalBinarySet object represents a set of
<code><a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a></code>s with <em>identical</em> chip types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaUnitSignalBinarySet(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaUnitSignalBinarySet_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AromaTabularBinarySet">AromaTabularBinarySet</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAromaUgpFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPlatform</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+writeDataFrame.AromaUnitSignalBinarySet">writeDataFrame</a></code> </td><td style="text-align: left;"> Writes the data set as a tab-delimited text file.</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AromaTabularBinarySet</b>:<br />
getDefaultFullName, getRootName, setAttributesBy, setAttributesBySampleAnnotationFile, setAttributesBySampleAnnotationSet, setAttributesByTags
</p>
<p><b>Methods inherited from GenericTabularFileSet</b>:<br />
extractMatrix, calculateAverageColumnAcrossFiles
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaUnitTabularBinaryFile'>The AromaUnitTabularBinaryFile class</h2><span id='topic+AromaUnitTabularBinaryFile'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaUnitTabularBinaryFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+ColumnNamesInterface">ColumnNamesInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFile">GenericTabularFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaMicroarrayTabularBinaryFile">AromaMicroarrayTabularBinaryFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+UnitAnnotationDataFile">UnitAnnotationDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AromaUnitTabularBinaryFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+AromaUflFile">AromaUflFile</a>, <a href="#topic+AromaUgcFile">AromaUgcFile</a>, <a href="#topic+AromaUgpFile">AromaUgpFile</a>, <em><a href="#topic+AromaUnitChromosomeTabularBinaryFile">AromaUnitChromosomeTabularBinaryFile</a></em>, <a href="#topic+AromaUnitGcContentFile">AromaUnitGcContentFile</a><br />
</p>
<p>public abstract static class <b>AromaUnitTabularBinaryFile</b><br />
extends <a href="#topic+UnitAnnotationDataFile">UnitAnnotationDataFile</a><br />
</p>
<p>A AromaUnitTabularBinaryFile is an <code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code> with
the constraint that the rows map one-to-one to, and in the same order as,
the units in a annotation chip type file (e.g. CDF file).
The (full) chip type of the annotation chip type file is given by the
mandatory file footer <code>chipType</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaUnitTabularBinaryFile(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaUnitTabularBinaryFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+writeDataFrame.AromaUnitTabularBinaryFile">writeDataFrame</a></code> </td><td style="text-align: left;"> Writes the data file as a tab-delimited text file.</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from UnitAnnotationDataFile</b>:<br />
byChipType, getAromaUflFile, getAromaUgpFile, getChipType, getDefaultExtension, getPlatform, nbrOfUnits
</p>
<p><b>Methods inherited from AromaMicroarrayTabularBinaryFile</b>:<br />
allocate, as.character, byChipType, findByChipType, getChipType, getFilenameExtension, getPlatform
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaTabularBinaryFile</b>:<br />
[, [&lt;-, [[, allocate, as.character, colApply, colMeans, colStats, colSums, dimnames&lt;-, getBytesPerColumn, getColClasses, getDefaultColumnNames, getRootName, importFrom, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, readFooter, readHeader, readRawFooter, setAttributesByTags, subset, summary, updateData, updateDataColumn, writeFooter, writeRawFooter
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericTabularFile</b>:<br />
[, as.character, dim, extractMatrix, head, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, tail, writeColumnsToFiles
</p>
<p><b>Methods inherited from ColumnNamesInterface</b>:<br />
appendColumnNamesTranslator, appendColumnNamesTranslatorByNULL, appendColumnNamesTranslatorBycharacter, appendColumnNamesTranslatorByfunction, appendColumnNamesTranslatorBylist, clearColumnNamesTranslator, clearListOfColumnNamesTranslators, getColumnNames, getColumnNamesTranslator, getDefaultColumnNames, getListOfColumnNamesTranslators, nbrOfColumns, setColumnNames, setColumnNamesTranslator, setListOfColumnNamesTranslators, updateColumnNames
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaUnitTotalCnBinaryFile'>The AromaUnitTotalCnBinaryFile class</h2><span id='topic+AromaUnitTotalCnBinaryFile'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaUnitTotalCnBinaryFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+ColumnNamesInterface">ColumnNamesInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFile">GenericTabularFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CopyNumberDataFile">CopyNumberDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AromaUnitTotalCnBinaryFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>AromaUnitTotalCnBinaryFile</b><br />
extends <em><a href="#topic+CopyNumberDataFile">CopyNumberDataFile</a></em><br />
</p>
<p>An AromaUnitTotalCnBinaryFile is a <code><a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaUnitTotalCnBinaryFile(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaUnitTotalCnBinaryFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractPSCNArray</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractPSCNMatrix</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractRawCopyNumbers</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAM</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>hasAlleleBFractions</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>hasStrandiness</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from CopyNumberDataFile</b>:<br />
as, as.CopyNumberDataFile, getNumberOfFilesAveraged, hasAlleleBFractions, hasStrandiness
</p>
<p><b>Methods inherited from AromaUnitSignalBinaryFile</b>:<br />
allocate, allocateFromUnitAnnotationDataFile, allocateFromUnitNamesFile, as.character, extractMatrix, extractRawGenomicSignals, fromFile, getChipType, getExtensionPattern, getFilenameExtension, getNumberOfFilesAveraged, getPlatform, isAverageFile, nbrOfUnits, readDataFrame, writeDataFrame
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaTabularBinaryFile</b>:<br />
[, [&lt;-, [[, allocate, as.character, colApply, colMeans, colStats, colSums, dimnames&lt;-, getBytesPerColumn, getColClasses, getDefaultColumnNames, getRootName, importFrom, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, readFooter, readHeader, readRawFooter, setAttributesByTags, subset, summary, updateData, updateDataColumn, writeFooter, writeRawFooter
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericTabularFile</b>:<br />
[, as.character, dim, extractMatrix, head, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, tail, writeColumnsToFiles
</p>
<p><b>Methods inherited from ColumnNamesInterface</b>:<br />
appendColumnNamesTranslator, appendColumnNamesTranslatorByNULL, appendColumnNamesTranslatorBycharacter, appendColumnNamesTranslatorByfunction, appendColumnNamesTranslatorBylist, clearColumnNamesTranslator, clearListOfColumnNamesTranslators, getColumnNames, getColumnNamesTranslator, getDefaultColumnNames, getListOfColumnNamesTranslators, nbrOfColumns, setColumnNames, setColumnNamesTranslator, setListOfColumnNamesTranslators, updateColumnNames
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaUnitTotalCnBinarySet'>The AromaUnitTotalCnBinarySet class</h2><span id='topic+AromaUnitTotalCnBinarySet'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaUnitTotalCnBinarySet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFileSet">GenericTabularFileSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaTabularBinarySet">AromaTabularBinarySet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaUnitSignalBinarySet">AromaUnitSignalBinarySet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CopyNumberDataSet">CopyNumberDataSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>AromaUnitTotalCnBinarySet</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>AromaUnitTotalCnBinarySet</b><br />
extends <a href="#topic+CopyNumberDataSet">CopyNumberDataSet</a><br />
</p>
<p>An AromaUnitTotalCnBinarySet object represents a set of
<code><a href="#topic+AromaUnitTotalCnBinaryFile">AromaUnitTotalCnBinaryFile</a></code>s with <em>identical</em> chip types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaUnitTotalCnBinarySet(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaUnitTotalCnBinarySet_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AromaUnitSignalBinarySet">AromaUnitSignalBinarySet</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as.AromaUnitTotalCnBinarySetTuple</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as.CopyNumberDataSetTuple</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>exportAromaUnitPscnBinarySet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>exportTotalCnRatioSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractPSCNArray</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAverageFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitNamesFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>writeDataFrame</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from CopyNumberDataSet</b>:<br />
as, as.CopyNumberDataSet, doCBS, hasAlleleBFractions, hasStrandiness
</p>
<p><b>Methods inherited from AromaUnitSignalBinarySet</b>:<br />
byName, findByName, getAromaFullNameTranslatorSet, getAromaUgpFile, getChipType, getPlatform, validate, writeDataFrame
</p>
<p><b>Methods inherited from AromaTabularBinarySet</b>:<br />
getDefaultFullName, getRootName, setAttributesBy, setAttributesBySampleAnnotationFile, setAttributesBySampleAnnotationSet, setAttributesByTags
</p>
<p><b>Methods inherited from GenericTabularFileSet</b>:<br />
extractMatrix, calculateAverageColumnAcrossFiles
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaUnitTypesFile'>The AromaUnitTypesFile class</h2><span id='topic+AromaUnitTypesFile'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class AromaUnitTypesFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+ColumnNamesInterface">ColumnNamesInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFile">GenericTabularFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+UnitAnnotationDataFile">UnitAnnotationDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+UnitTypesFile">UnitTypesFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>AromaUnitTypesFile</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>AromaUnitTypesFile</b><br />
extends <em><a href="#topic+UnitTypesFile">UnitTypesFile</a></em><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaUnitTypesFile(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AromaUnitTypesFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AromaUnitTabularBinaryFile">AromaUnitTabularBinaryFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPlatform</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitTypes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>importFromUnitTypesFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from UnitTypesFile</b>:<br />
getUnitTypes, nbrOfUnits
</p>
<p><b>Methods inherited from UnitAnnotationDataFile</b>:<br />
byChipType, getAromaUflFile, getAromaUgpFile, getChipType, getDefaultExtension, getPlatform, nbrOfUnits
</p>
<p><b>Methods inherited from AromaUnitSignalBinaryFile</b>:<br />
allocate, allocateFromUnitAnnotationDataFile, allocateFromUnitNamesFile, as.character, extractMatrix, extractRawGenomicSignals, fromFile, getChipType, getExtensionPattern, getFilenameExtension, getNumberOfFilesAveraged, getPlatform, isAverageFile, nbrOfUnits, readDataFrame, writeDataFrame
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaTabularBinaryFile</b>:<br />
[, [&lt;-, [[, allocate, as.character, colApply, colMeans, colStats, colSums, dimnames&lt;-, getBytesPerColumn, getColClasses, getDefaultColumnNames, getRootName, importFrom, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, readFooter, readHeader, readRawFooter, setAttributesByTags, subset, summary, updateData, updateDataColumn, writeFooter, writeRawFooter
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericTabularFile</b>:<br />
[, as.character, dim, extractMatrix, head, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, tail, writeColumnsToFiles
</p>
<p><b>Methods inherited from ColumnNamesInterface</b>:<br />
appendColumnNamesTranslator, appendColumnNamesTranslatorByNULL, appendColumnNamesTranslatorBycharacter, appendColumnNamesTranslatorByfunction, appendColumnNamesTranslatorBylist, clearColumnNamesTranslator, clearListOfColumnNamesTranslators, getColumnNames, getColumnNamesTranslator, getDefaultColumnNames, getListOfColumnNamesTranslators, nbrOfColumns, setColumnNames, setColumnNamesTranslator, setListOfColumnNamesTranslators, updateColumnNames
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='as.GrayscaleImage.matrix'>Creates a Grayscale (Color) Image from a matrix file</h2><span id='topic+as.GrayscaleImage.matrix'></span>

<h3>Description</h3>

<p>Creates a Grayscale (Color) Image from a matrix file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'matrix'
as.GrayscaleImage(z, transforms=NULL, interleaved=c("none", "h", "v", "auto"), scale=1,
  ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="as.GrayscaleImage.matrix_+3A_z">z</code></td>
<td>
<p>A KxN <code><a href="base.html#topic+matrix">matrix</a></code>.</p>
</td></tr>
<tr><td><code id="as.GrayscaleImage.matrix_+3A_transforms">transforms</code></td>
<td>
<p>A <code><a href="base.html#topic+list">list</a></code> of transform <code><a href="base.html#topic+function">function</a></code>s.</p>
</td></tr>
<tr><td><code id="as.GrayscaleImage.matrix_+3A_interleaved">interleaved</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying how the image data
should be interleaved, if at all.</p>
</td></tr>
<tr><td><code id="as.GrayscaleImage.matrix_+3A_scale">scale</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> scale factor in (0,+Inf) for resizing the
imaging. If <code>1</code>, no resizing is done.</p>
</td></tr>
<tr><td><code id="as.GrayscaleImage.matrix_+3A_...">...</code></td>
<td>
<p>Passed to <code>colorize()</code> for the object created.</p>
</td></tr>
<tr><td><code id="as.GrayscaleImage.matrix_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a bitmap image object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson, Ken Simpson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="base.html#topic+matrix">matrix</a></code>.
</p>

<hr>
<h2 id='BinnedScatter'>The BinnedScatter class</h2><span id='topic+BinnedScatter'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class BinnedScatter</b><br />
</p>
<p><code>list</code><br />
<code>~~|</code><br />
<code>~~+--</code><code>BinnedScatter</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public class <b>BinnedScatter</b><br />
extends list<br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BinnedScatter(data=NULL, density=NULL, map=NULL, params=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BinnedScatter_+3A_data">data</code></td>
<td>
<p>A Nx2 <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+matrix">matrix</a></code>.</p>
</td></tr>
<tr><td><code id="BinnedScatter_+3A_density">density</code></td>
<td>
<p>...</p>
</td></tr>
<tr><td><code id="BinnedScatter_+3A_map">map</code></td>
<td>
<p>...</p>
</td></tr>
<tr><td><code id="BinnedScatter_+3A_params">params</code></td>
<td>
<p>A <code><a href="base.html#topic+list">list</a></code> of parameters.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plot</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>points</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>reorder</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>subsample</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>subset</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from list</b>:<br />
Ops,nonStructure,vector-method, Ops,structure,vector-method, Ops,vector,nonStructure-method, Ops,vector,structure-method, all.equal, as.CopyNumberDataSetTuple, as.data.frame, attachLocally, callHooks, coerce,ANY,list-method, exportAromaUnitPscnBinarySet, listToXml, mergeBoxplotStats, relist, type.convert, within
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>The spatial density is estimated by internal functions of the
<span class="pkg">smoothScatter</span> package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Sample scatter data
n &lt;- 10e3
x &lt;- rnorm(n=n)
y &lt;- rnorm(n=n)
xy &lt;- cbind(x=x, y=sin(x)+y/5)

# Bin data and estimate densities
xyd &lt;- binScatter(xy)

layout(matrix(1:4, nrow=2))
par(mar=c(5,4,2,1))

# Plot data
plot(xyd, pch=1)

# Thin scatter data by subsampling
rhos &lt;- c(1/3, 1/4, 1/6)
for (kk in seq_along(rhos)) {
  xyd2 &lt;- subsample(xyd, size=rhos[kk])
  points(xyd2, pch=1, col=kk+1)
}

for (kk in seq_along(rhos)) {
  xyd2 &lt;- subsample(xyd, size=rhos[kk])
  plot(xyd2, pch=1, col=kk+1)
  mtext(side=3, line=0, sprintf("Density: %.1f%%", 100*rhos[kk]))
}
</code></pre>

<hr>
<h2 id='CacheKeyInterface'>The CacheKeyInterface class interface</h2><span id='topic+CacheKeyInterface'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class CacheKeyInterface</b><br />
</p>
<p><code><a href="R.oo.html#topic+Interface">Interface</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code>CacheKeyInterface</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+AromaCellCpgFile">AromaCellCpgFile</a>, <a href="#topic+AromaCellPositionFile">AromaCellPositionFile</a>, <a href="#topic+AromaCellSequenceFile">AromaCellSequenceFile</a>, <em><a href="#topic+AromaCellTabularBinaryFile">AromaCellTabularBinaryFile</a></em>, <em><a href="#topic+AromaGenomeTextFile">AromaGenomeTextFile</a></em>, <em><a href="#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></em>, <em><a href="#topic+AromaMicroarrayTabularBinaryFile">AromaMicroarrayTabularBinaryFile</a></em>, <em><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></em>, <em><a href="#topic+AromaUcscGenomeTextFile">AromaUcscGenomeTextFile</a></em>, <a href="#topic+AromaUflFile">AromaUflFile</a>, <a href="#topic+AromaUgcFile">AromaUgcFile</a>, <a href="#topic+AromaUgpFile">AromaUgpFile</a>, <a href="#topic+AromaUnitCallFile">AromaUnitCallFile</a>, <em><a href="#topic+AromaUnitChromosomeTabularBinaryFile">AromaUnitChromosomeTabularBinaryFile</a></em>, <a href="#topic+AromaUnitFracBCnBinaryFile">AromaUnitFracBCnBinaryFile</a>, <a href="#topic+AromaUnitGcContentFile">AromaUnitGcContentFile</a>, <a href="#topic+AromaUnitGenotypeCallFile">AromaUnitGenotypeCallFile</a>, <em><a href="#topic+AromaUnitPscnBinaryFile">AromaUnitPscnBinaryFile</a></em>, <a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a>, <em><a href="#topic+AromaUnitTabularBinaryFile">AromaUnitTabularBinaryFile</a></em>, <em><a href="#topic+AromaUnitTotalCnBinaryFile">AromaUnitTotalCnBinaryFile</a></em>, <a href="#topic+AromaUnitTypesFile">AromaUnitTypesFile</a>, <em><a href="#topic+CbsSegmentationDataFile">CbsSegmentationDataFile</a></em>, <a href="#topic+FileCacheKeyInterface">FileCacheKeyInterface</a>, <em><a href="#topic+SampleAnnotationFile">SampleAnnotationFile</a></em>, <em><a href="#topic+SegmentationDataFile">SegmentationDataFile</a></em>, <a href="#topic+TextUnitNamesFile">TextUnitNamesFile</a><br />
</p>
<p>public class <b>CacheKeyInterface</b><br />
extends <a href="R.oo.html#topic+Interface">Interface</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CacheKeyInterface(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CacheKeyInterface_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getCacheKey.CacheKeyInterface">getCacheKey</a></code> </td><td style="text-align: left;"> Gets a list of cache key items.</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from Interface</b>:<br />
extend, print, uses
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='CbsModel'>The CbsModel class</h2><span id='topic+CbsModel'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class CbsModel</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+ChromosomalModel">ChromosomalModel</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+CopyNumberChromosomalModel">CopyNumberChromosomalModel</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+CopyNumberSegmentationModel">CopyNumberSegmentationModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code>CbsModel</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>CbsModel</b><br />
extends <em><a href="#topic+CopyNumberSegmentationModel">CopyNumberSegmentationModel</a></em><br />
</p>
<p>This class represents the Circular Binary Segmentation (CBS) model [1].
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CbsModel(cesTuple=NULL, ..., seed=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CbsModel_+3A_cestuple">cesTuple</code></td>
<td>
<p>A <code><a href="#topic+CopyNumberDataSetTuple">CopyNumberDataSetTuple</a></code>.</p>
</td></tr>
<tr><td><code id="CbsModel_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+CopyNumberSegmentationModel">CopyNumberSegmentationModel</a></code>.</p>
</td></tr>
<tr><td><code id="CbsModel_+3A_seed">seed</code></td>
<td>
<p>An (optional) <code><a href="base.html#topic+integer">integer</a></code> that if specified will (temporarily)
set the random seed each time before calling the segmentation method.
For more information, see <code><a href="#topic+segmentByCBS">segmentByCBS</a></code>().</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from CopyNumberSegmentationModel</b>:<br />
fit, getAsteriskTags, getFitFunction, getFullNames, getRegions, getTags, plot, plotCopyNumberRegionLayers, writeRegions
</p>
<p><b>Methods inherited from CopyNumberChromosomalModel</b>:<br />
as.character, calculateChromosomeStatistics, calculateRatios, estimateSds, extractRawCopyNumbers, fit, getChromosomeLength, getDataFileMatrix, getMaxNAFraction, getNames, getOptionalArguments, getPairedNames, getRefSetTuple, getReference, getReferenceSetTuple, isPaired, newPlot, plotAxesLayers, plotChromosomesLayers, plotCytobandLayers, plotFitLayers, plotGridHorizontalLayers, plotRawCopyNumbers, plotSampleLayers, setReference
</p>
<p><b>Methods inherited from ChromosomalModel</b>:<br />
as.character, fit, getAlias, getAromaGenomeTextFile, getAsteriskTags, getChipType, getChipTypes, getChromosomes, getFullName, getFullNames, getGenome, getGenomeData, getGenomeFile, getListOfAromaUgpFiles, getName, getNames, getParentPath, getPath, getReportPath, getRootPath, getSetTuple, getSets, getTags, indexOf, nbrOfArrays, nbrOfChipTypes, setChromosomes, setGenome
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] Olshen, A. B., Venkatraman, E. S., Lucito, R., Wigler, M.
<em>Circular binary segmentation for the analysis of array-based
DNA copy number data. Biostatistics 5: 557-572, 2004.</em><br />
[2] Venkatraman, E. S. &amp; Olshen, A. B.
<em>A faster circular binary segmentation algorithm for the
analysis of array CGH data</em>. Bioinformatics, 2007.<br />
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CopyNumberSegmentationModel">CopyNumberSegmentationModel</a></code>.
</p>

<hr>
<h2 id='ChromosomalModel'>The ChromosomalModel class</h2><span id='topic+ChromosomalModel'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class ChromosomalModel</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><em><code>ChromosomalModel</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+CbsModel">CbsModel</a>, <em><a href="#topic+CopyNumberChromosomalModel">CopyNumberChromosomalModel</a></em>, <em><a href="#topic+CopyNumberSegmentationModel">CopyNumberSegmentationModel</a></em>, <a href="#topic+GladModel">GladModel</a>, <a href="#topic+HaarSegModel">HaarSegModel</a>, <em><a href="#topic+RawCopyNumberModel">RawCopyNumberModel</a></em><br />
</p>
<p>public abstract static class <b>ChromosomalModel</b><br />
extends <a href="R.oo.html#topic+Object">Object</a><br />
</p>
<p>This <em>abstract</em> class represents a chromosomal model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ChromosomalModel(cesTuple=NULL, tags="*", genome="Human", chromosomes=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ChromosomalModel_+3A_cestuple">cesTuple</code></td>
<td>
<p>A <code><a href="#topic+AromaMicroarrayDataSetTuple">AromaMicroarrayDataSetTuple</a></code>.</p>
</td></tr>
<tr><td><code id="ChromosomalModel_+3A_tags">tags</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> of tags.</p>
</td></tr>
<tr><td><code id="ChromosomalModel_+3A_genome">genome</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying what genome is process.</p>
</td></tr>
<tr><td><code id="ChromosomalModel_+3A_chromosomes">chromosomes</code></td>
<td>
<p>(optional) A <code><a href="base.html#topic+vector">vector</a></code> specifying which chromosomes
to process.</p>
</td></tr>
<tr><td><code id="ChromosomalModel_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>fit</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getChipType.ChromosomalModel">getChipType</a></code> </td><td style="text-align: left;"> Gets a label for all chip types merged.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipTypes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getChromosomes.ChromosomalModel">getChromosomes</a></code> </td><td style="text-align: left;"> Gets the chromosomes to be processed.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFullName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFullNames</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getGenome</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getNames.ChromosomalModel">getNames</a></code> </td><td style="text-align: left;"> Gets the names of the arrays.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPath</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getSets</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getTags</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>indexOf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+nbrOfArrays.ChromosomalModel">nbrOfArrays</a></code> </td><td style="text-align: left;"> Gets the number of arrays.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+nbrOfChipTypes.ChromosomalModel">nbrOfChipTypes</a></code> </td><td style="text-align: left;"> Gets the number of chip types.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setChromosomes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setGenome</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Requirements</h3>

<p>This class requires genome information annotation files for
every chip type.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='ChromosomeExplorer'>The ChromosomeExplorer class</h2><span id='topic+ChromosomeExplorer'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class ChromosomeExplorer</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+Explorer">Explorer</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code>ChromosomeExplorer</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>ChromosomeExplorer</b><br />
extends <em><a href="#topic+Explorer">Explorer</a></em><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ChromosomeExplorer(model=NULL, zooms=2^(0:6), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ChromosomeExplorer_+3A_model">model</code></td>
<td>
<p>A <code><a href="#topic+CopyNumberChromosomalModel">CopyNumberChromosomalModel</a></code> object.</p>
</td></tr>
<tr><td><code id="ChromosomeExplorer_+3A_zooms">zooms</code></td>
<td>
<p>An positive <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying for which zoom
levels the graphics should be generated.</p>
</td></tr>
<tr><td><code id="ChromosomeExplorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>display</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getChromosomes.ChromosomeExplorer">getChromosomes</a></code> </td><td style="text-align: left;"> Gets the chromosomes available.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFullNames</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getModel.ChromosomeExplorer">getModel</a></code> </td><td style="text-align: left;"> Gets the model.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getNames</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPath</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getZooms</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>indexOf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+process.ChromosomeExplorer">process</a></code> </td><td style="text-align: left;"> Generates image files, scripts and dynamic pages for the explorer.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+setArrays.ChromosomeExplorer">setArrays</a></code> </td><td style="text-align: left;"> Sets the arrays.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setCytoband</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setZooms</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from Explorer</b>:<br />
addIncludes, addIndexFile, as.character, display, getAlias, getArraysOfInput, getAsteriskTags, getFullName, getIncludePath, getMainPath, getName, getNameOfInput, getNames, getPath, getReportPathPattern, getRootPath, getSampleLayerPrefix, getSubname, getTags, getTagsOfInput, getTemplatePath, getVersion, nbrOfArrays, process, setAlias, setArrays, setReportPathPattern, setSubname, setup, splitByReportPathPattern, updateSetupExplorerFile
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Generating PNG images</h3>

<p>In order to get better looking graphs, but also to be able to generate
bitmap images on systems without direct bitmap support, which is the case
when running R in batch mode or on Unix without X11 support, images are
created using the <code><a href="R.devices.html#topic+png2">png2</a></code> device (a wrapper for
<code>bitmap()</code> imitating <code>png()</code>).  The <code>png()</code> is only
used if <code>png2()</code>, which requires Ghostscript, does not.
Note, when images are created using <code>png2()</code>, the images does
not appear immediately, although the function call is completed,
so be patient.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+CopyNumberChromosomalModel">CopyNumberChromosomalModel</a></code>.
</p>

<hr>
<h2 id='colBinnedSmoothing.matrix'>Binned smoothing of a matrix column by column</h2><span id='topic+colBinnedSmoothing.matrix'></span><span id='topic+colBinnedSmoothing'></span><span id='topic+binnedSmoothing'></span><span id='topic+binnedSmoothing.numeric'></span>

<h3>Description</h3>

<p>Binned smoothing of a matrix column by column.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'matrix'
colBinnedSmoothing(Y, x=seq_len(nrow(Y)), w=NULL, xOut=NULL, xOutRange=NULL,
  from=min(x, na.rm = TRUE), to=max(x, na.rm = TRUE), by=NULL, length.out=length(x),
  na.rm=TRUE, FUN="median", ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="colBinnedSmoothing.matrix_+3A_y">Y</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> JxI <code><a href="base.html#topic+matrix">matrix</a></code> (or a <code><a href="base.html#topic+vector">vector</a></code> of length J.)</p>
</td></tr>
<tr><td><code id="colBinnedSmoothing.matrix_+3A_x">x</code></td>
<td>
<p>A (optional) <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying the positions of
the J entries. The default is to assume uniformly distributed
positions.</p>
</td></tr>
<tr><td><code id="colBinnedSmoothing.matrix_+3A_w">w</code></td>
<td>
<p>A optional <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of prior weights for each of
the J entries.</p>
</td></tr>
<tr><td><code id="colBinnedSmoothing.matrix_+3A_xout">xOut</code></td>
<td>
<p>Optional <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of K bin center locations.</p>
</td></tr>
<tr><td><code id="colBinnedSmoothing.matrix_+3A_xoutrange">xOutRange</code></td>
<td>
<p>Optional Kx2 <code><a href="base.html#topic+matrix">matrix</a></code> specifying the boundary
locations for K bins, where each row represents a bin <code class="reqn">[x0,x1)</code>.
If not specified, the boundaries are set to be the midpoints
of the bin centers, such that the bins have maximum lengths
without overlapping.
Vice verse, if <code>xOut</code> is not specified, then <code>xOut</code> is set
to be the mid points of the <code>xOutRange</code> boundaries.
</p>
</td></tr>
<tr><td><code id="colBinnedSmoothing.matrix_+3A_from">from</code>, <code id="colBinnedSmoothing.matrix_+3A_to">to</code>, <code id="colBinnedSmoothing.matrix_+3A_by">by</code>, <code id="colBinnedSmoothing.matrix_+3A_length.out">length.out</code></td>
<td>

<p>If neither <code>xOut</code> nor <code>xOutRange</code> is specified,
the <code>xOut</code> is generated uniformly from these arguments, which
specify the center location of the first and the last bin, and the
distance between the center locations, utilizing the
<code><a href="base.html#topic+seq">seq</a></code>() function.
Argument <code>length.out</code> can be used as an alternative to
<code>by</code>, in case it specifies the total number of bins instead.
</p>
</td></tr>
<tr><td><code id="colBinnedSmoothing.matrix_+3A_fun">FUN</code></td>
<td>
<p>A <code><a href="base.html#topic+function">function</a></code>.</p>
</td></tr>
<tr><td><code id="colBinnedSmoothing.matrix_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, missing values are excluded, otherwise not.</p>
</td></tr>
<tr><td><code id="colBinnedSmoothing.matrix_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="colBinnedSmoothing.matrix_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that all zero-length bins <code class="reqn">[x0,x1)</code> will get result
in an <code><a href="base.html#topic+NA">NA</a></code> value, because such bins contain no data points.
This also means that <code>colBinnedSmoothing(Y, x=x, xOut=xOut)</code>
where <code>xOut</code> contains duplicated values, will result in
some zero-length bins and hence <code><a href="base.html#topic+NA">NA</a></code> values.
</p>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+numeric">numeric</a></code> KxI <code><a href="base.html#topic+matrix">matrix</a></code> (or a <code><a href="base.html#topic+vector">vector</a></code> of length K) where
K is the total number of bins.
The following attributes are also returned:
</p>

<dl>
<dt><code>xOut</code></dt><dd><p>The center locations of each bin.</p>
</dd>
<dt><code>xOutRange</code></dt><dd><p>The bin boundaries.</p>
</dd>
<dt><code>count</code></dt><dd><p>The number of data points within each bin
(based solely on argument <code>x</code>).</p>
</dd>
<dt><code>binWidth</code></dt><dd><p>The <em>average</em> bin width.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+colKernelSmoothing.matrix">*colKernelSmoothing</a>()</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Number of tracks
I &lt;- 4

# Number of data points per track
J &lt;- 100

# Simulate data with a gain in track 2 and 3
x &lt;- 1:J
Y &lt;- matrix(rnorm(I*J, sd=1/2), ncol=I)
Y[30:50,2:3] &lt;- Y[30:50,2:3] + 3

# Uniformly distributed equal-sized bins
Ys3 &lt;- colBinnedSmoothing(Y, x=x, from=2, by=3)
Ys5 &lt;- colBinnedSmoothing(Y, x=x, from=3, by=5)

# Custom bins
xOutRange &lt;- t(matrix(c(
  1, 11,
 11, 31,
 31, 41,
 41, 51,
 51, 81,
 81, 91,
 91,101
), nrow=2))
YsC &lt;- colBinnedSmoothing(Y, x=x, xOutRange=xOutRange)

# Custom bins specified by center locations with
# maximized width relative to the neighboring bins.
xOut &lt;- c(6, 21, 36, 46, 66, 86, 96)
YsD &lt;- colBinnedSmoothing(Y, x=x, xOut=xOut)

xlim &lt;- range(x)
ylim &lt;- c(-3,5)
layout(matrix(1:I, ncol=1))
par(mar=c(3,3,1,1)+0.1, pch=19)
for (ii in 1:I) {
  plot(NA, xlim=xlim, ylim=ylim)
  points(x, Y[,ii], col="#999999")

  xOut &lt;- attr(Ys3, "xOut")
  lines(xOut, Ys3[,ii], col=2)
  points(xOut, Ys3[,ii], col=2)

  xOut &lt;- attr(Ys5, "xOut")
  lines(xOut, Ys5[,ii], col=3)
  points(xOut, Ys5[,ii], col=3)

  xOut &lt;- attr(YsC, "xOut")
  lines(xOut, YsC[,ii], col=4)
  points(xOut, YsC[,ii], col=4, pch=15)

  xOut &lt;- attr(YsD, "xOut")
  lines(xOut, YsD[,ii], col=5)
  points(xOut, YsD[,ii], col=5, pch=15)

  if (ii == 1) {
    legend("topright", pch=c(19,19,15,15), col=c(2,3,4,5),
           c("by=3", "by=5", "Custom #1", "Custom #2"), horiz=TRUE, bty="n")
  }
}


# Sanity checks
xOut &lt;- x
YsT &lt;- colBinnedSmoothing(Y, x=x, xOut=xOut)
stopifnot(all(YsT == Y))
stopifnot(all(attr(YsT, "counts") == 1))

xOut &lt;- attr(YsD, "xOut")
YsE &lt;- colBinnedSmoothing(YsD, x=xOut, xOut=xOut)
stopifnot(all(YsE == YsD))
stopifnot(all(attr(YsE, "xOutRange") == attr(YsD, "xOutRange")))
stopifnot(all(attr(YsE, "counts") == 1))

# Scramble ordering of loci
idxs &lt;- sample(x)
x2 &lt;- x[idxs]
Y2 &lt;- Y[idxs,,drop=FALSE]
Y2s &lt;- colBinnedSmoothing(Y2, x=x2, xOut=x2)
stopifnot(all(attr(Y2s, "xOut") == x2))
stopifnot(all(attr(Y2s, "counts") == 1))
stopifnot(all(Y2s == Y2))

xOut &lt;- x[seq(from=2, to=J, by=3)]
YsT &lt;- colBinnedSmoothing(Y, x=x, xOut=xOut)
stopifnot(all(YsT == Ys3))
stopifnot(all(attr(YsT, "counts") == 3))

xOut &lt;- x[seq(from=3, to=J, by=5)]
YsT &lt;- colBinnedSmoothing(Y, x=x, xOut=xOut)
stopifnot(all(YsT == Ys5))
stopifnot(all(attr(YsT, "counts") == 5))

</code></pre>

<hr>
<h2 id='colKernelSmoothing.matrix'>Kernel smoothing of a matrix column by column</h2><span id='topic+colKernelSmoothing.matrix'></span><span id='topic+colKernelSmoothing'></span><span id='topic+kernelSmoothing'></span><span id='topic+kernelSmoothing.numeric'></span>

<h3>Description</h3>

<p>Kernel smoothing of a matrix column by column.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'matrix'
colKernelSmoothing(Y, x=seq_len(nrow(Y)), w=NULL, xOut=x,
  kernel=c("gaussian", "uniform"), h, censorH=3, na.rm=TRUE, robust=FALSE, ...,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="colKernelSmoothing.matrix_+3A_y">Y</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> JxI <code><a href="base.html#topic+matrix">matrix</a></code> (or a <code><a href="base.html#topic+vector">vector</a></code> of length J.)</p>
</td></tr>
<tr><td><code id="colKernelSmoothing.matrix_+3A_x">x</code></td>
<td>
<p>A (optional) <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying the positions of
the J entries. The default is to assume uniformly distributed
positions.</p>
</td></tr>
<tr><td><code id="colKernelSmoothing.matrix_+3A_w">w</code></td>
<td>
<p>A optional <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of prior weights for each of
the J entries.</p>
</td></tr>
<tr><td><code id="colKernelSmoothing.matrix_+3A_xout">xOut</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying K target positions where
the kernel is applied.</p>
</td></tr>
<tr><td><code id="colKernelSmoothing.matrix_+3A_kernel">kernel</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string or a <code><a href="base.html#topic+function">function</a></code> specifying the
kernel used.</p>
</td></tr>
<tr><td><code id="colKernelSmoothing.matrix_+3A_h">h</code></td>
<td>
<p>A single positive <code><a href="base.html#topic+numeric">numeric</a></code> specifying the bandwidth of
the kernel.</p>
</td></tr>
<tr><td><code id="colKernelSmoothing.matrix_+3A_censorh">censorH</code></td>
<td>
<p>A single positive <code><a href="base.html#topic+numeric">numeric</a></code> specifying the where to
truncate the kernel. If <code><a href="base.html#topic+is.finite">Inf</a></code>, no truncation is done.</p>
</td></tr>
<tr><td><code id="colKernelSmoothing.matrix_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, missing values are excluded, otherwise not.</p>
</td></tr>
<tr><td><code id="colKernelSmoothing.matrix_+3A_robust">robust</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, robust estimators are used, otherwise not.</p>
</td></tr>
<tr><td><code id="colKernelSmoothing.matrix_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="colKernelSmoothing.matrix_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+numeric">numeric</a></code> KxI <code><a href="base.html#topic+matrix">matrix</a></code> (or a <code><a href="base.html#topic+vector">vector</a></code> of length K).
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+colBinnedSmoothing.matrix">*colBinnedSmoothing</a>()</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>J &lt;- 100
I &lt;- 4
Y &lt;- matrix(rnorm(I*J, sd=1/2), ncol=I)
# Introduce a gain in column 2 and 3
Y[30:50,2:3] &lt;- Y[30:50,2:3] + 3
x &lt;- 1:J

xOut &lt;- x
Ys1 &lt;- colKernelSmoothing(Y, x=x, xOut=xOut, kernel="gaussian", h=1)
Ys5 &lt;- colKernelSmoothing(Y, x=x, xOut=xOut, kernel="gaussian", h=5)

xlim &lt;- range(c(x,xOut))
ylim &lt;- c(-3,5)
layout(matrix(1:I, ncol=1))
par(mar=c(3,3,1,1)+0.1, pch=19)
for (ii in 1:I) {
  plot(NA, xlim=xlim, ylim=ylim)
  points(x, Y[,ii], col="#999999")
  lines(xOut, Ys1[,ii], col=2)
  points(xOut, Ys1[,ii], col=2)
  lines(xOut, Ys5[,ii], col=3)
  points(xOut, Ys5[,ii], col=3)
}
</code></pre>

<hr>
<h2 id='CopyNumberChromosomalModel'>The CopyNumberChromosomalModel class</h2><span id='topic+CopyNumberChromosomalModel'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class CopyNumberChromosomalModel</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+ChromosomalModel">ChromosomalModel</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><em><code>CopyNumberChromosomalModel</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+CbsModel">CbsModel</a>, <em><a href="#topic+CopyNumberSegmentationModel">CopyNumberSegmentationModel</a></em>, <a href="#topic+GladModel">GladModel</a>, <a href="#topic+HaarSegModel">HaarSegModel</a>, <em><a href="#topic+RawCopyNumberModel">RawCopyNumberModel</a></em><br />
</p>
<p>public abstract static class <b>CopyNumberChromosomalModel</b><br />
extends <em><a href="#topic+ChromosomalModel">ChromosomalModel</a></em><br />
</p>
<p>This <em>abstract</em> class represents a copy-number model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CopyNumberChromosomalModel(cesTuple=NULL, refTuple=NULL, calculateRatios=TRUE, tags="*",
  genome="Human", chromosomes=NULL, maxNAFraction=1/5, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CopyNumberChromosomalModel_+3A_cestuple">cesTuple</code></td>
<td>
<p>A <code><a href="#topic+CopyNumberDataSetTuple">CopyNumberDataSetTuple</a></code>.</p>
</td></tr>
<tr><td><code id="CopyNumberChromosomalModel_+3A_reftuple">refTuple</code></td>
<td>
<p>An optional <code><a href="#topic+CopyNumberDataFile">CopyNumberDataFile</a></code>,
or <code><a href="#topic+CopyNumberDataSet">CopyNumberDataSet</a></code> or <code><a href="#topic+CopyNumberDataSetTuple">CopyNumberDataSetTuple</a></code>
for pairwise comparisons.</p>
</td></tr>
<tr><td><code id="CopyNumberChromosomalModel_+3A_calculateratios">calculateRatios</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> specifying whether ratios should
be calculated relative to the reference.
If <code><a href="base.html#topic+logical">FALSE</a></code>, argument <code>refTuple</code> is ignored.
</p>
</td></tr>
<tr><td><code id="CopyNumberChromosomalModel_+3A_tags">tags</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> of tags.</p>
</td></tr>
<tr><td><code id="CopyNumberChromosomalModel_+3A_genome">genome</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying what genome is process.</p>
</td></tr>
<tr><td><code id="CopyNumberChromosomalModel_+3A_chromosomes">chromosomes</code></td>
<td>
<p>(optional) A <code><a href="base.html#topic+vector">vector</a></code> specifying which chromosomes
to process.</p>
</td></tr>
<tr><td><code id="CopyNumberChromosomalModel_+3A_maxnafraction">maxNAFraction</code></td>
<td>
<p>A <code><a href="base.html#topic+double">double</a></code> in [0,1] indicating how many non-finite
signals are allowed in the sanity checks of the data.</p>
</td></tr>
<tr><td><code id="CopyNumberChromosomalModel_+3A_...">...</code></td>
<td>
<p>Optional arguments that may be used by some of the
subclass models.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+extractRawCopyNumbers.CopyNumberChromosomalModel">extractRawCopyNumbers</a></code> </td><td style="text-align: left;"> Extracts relative copy numbers.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+fit.CopyNumberChromosomalModel">fit</a></code> </td><td style="text-align: left;"> Fits the model.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChromosomeLength</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getNames</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getReference</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>isPaired</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ChromosomalModel</b>:<br />
as.character, fit, getAlias, getAromaGenomeTextFile, getAsteriskTags, getChipType, getChipTypes, getChromosomes, getFullName, getFullNames, getGenome, getGenomeData, getGenomeFile, getListOfAromaUgpFiles, getName, getNames, getParentPath, getPath, getReportPath, getRootPath, getSetTuple, getSets, getTags, indexOf, nbrOfArrays, nbrOfChipTypes, setChromosomes, setGenome
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Requirements</h3>

<p>This class requires genome information annotation files for
every chip type.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='CopyNumberSegmentationModel'>The CopyNumberSegmentationModel class</h2><span id='topic+CopyNumberSegmentationModel'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class CopyNumberSegmentationModel</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+ChromosomalModel">ChromosomalModel</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+CopyNumberChromosomalModel">CopyNumberChromosomalModel</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><em><code>CopyNumberSegmentationModel</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+CbsModel">CbsModel</a>, <a href="#topic+GladModel">GladModel</a>, <a href="#topic+HaarSegModel">HaarSegModel</a><br />
</p>
<p>public abstract static class <b>CopyNumberSegmentationModel</b><br />
extends <em><a href="#topic+CopyNumberChromosomalModel">CopyNumberChromosomalModel</a></em><br />
</p>
<p>This <em>abstract</em> class represents a copy-number segmentation model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CopyNumberSegmentationModel(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CopyNumberSegmentationModel_+3A_...">...</code></td>
<td>
<p>Arguments passed to constructor
<code><a href="#topic+CopyNumberChromosomalModel">CopyNumberChromosomalModel</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+fit.CopyNumberSegmentationModel">fit</a></code> </td><td style="text-align: left;"> Fits the model.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFullNames</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getRegions</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getTags</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plot</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>writeRegions</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from CopyNumberChromosomalModel</b>:<br />
as.character, calculateChromosomeStatistics, calculateRatios, estimateSds, extractRawCopyNumbers, fit, getChromosomeLength, getDataFileMatrix, getMaxNAFraction, getNames, getOptionalArguments, getPairedNames, getRefSetTuple, getReference, getReferenceSetTuple, isPaired, newPlot, plotAxesLayers, plotChromosomesLayers, plotCytobandLayers, plotFitLayers, plotGridHorizontalLayers, plotRawCopyNumbers, plotSampleLayers, setReference
</p>
<p><b>Methods inherited from ChromosomalModel</b>:<br />
as.character, fit, getAlias, getAromaGenomeTextFile, getAsteriskTags, getChipType, getChipTypes, getChromosomes, getFullName, getFullNames, getGenome, getGenomeData, getGenomeFile, getListOfAromaUgpFiles, getName, getNames, getParentPath, getPath, getReportPath, getRootPath, getSetTuple, getSets, getTags, indexOf, nbrOfArrays, nbrOfChipTypes, setChromosomes, setGenome
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='display.Explorer'>Displays the explorer in the default browser</h2><span id='topic+display.Explorer'></span><span id='topic+Explorer.display'></span><span id='topic+display+2CExplorer-method'></span>

<h3>Description</h3>

<p>Displays the explorer in the default browser.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Explorer'
display(this, filename=sprintf("%s.html", class(this)[1]), ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="display.Explorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="display.Explorer_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns nothing.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+Explorer">Explorer</a></code>.
</p>

<hr>
<h2 id='doCBS'>Performs Circular Binary Segmentation (CBS) on a data set</h2><span id='topic+doCBS.default'></span><span id='topic+doCBS'></span><span id='topic+doCBS.CopyNumberDataSetTuple'></span><span id='topic+doCBS.CopyNumberDataSet'></span><span id='topic+doCBS.character'></span>

<h3>Description</h3>

<p>Performs Circular Binary Segmentation (CBS) on a data set for one or more chip types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  ## Default S3 method:
doCBS(dataSet, tags=NULL, chipTypes, arrays=NULL, ..., verbose=FALSE)
  ## S3 method for class 'CopyNumberDataSet'
doCBS(ds, arrays=NULL, ...)
  ## S3 method for class 'CopyNumberDataSetTuple'
doCBS(dsTuple, arrays=NULL, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="doCBS_+3A_ds">ds</code>, <code id="doCBS_+3A_dstuple">dsTuple</code>, <code id="doCBS_+3A_dataset">dataSet</code></td>
<td>
<p>A <code><a href="#topic+CopyNumberDataSet">CopyNumberDataSet</a></code>, a <code><a href="#topic+CopyNumberDataSetTuple">CopyNumberDataSetTuple</a></code> or a <code><a href="base.html#topic+character">character</a></code> string with the name of one of them.</p>
</td></tr>
<tr><td><code id="doCBS_+3A_tags">tags</code></td>
<td>
<p>An optional <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> of data set tags (only when <code>dataSet</code> is specified).</p>
</td></tr>
<tr><td><code id="doCBS_+3A_chiptypes">chipTypes</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying the chip types
for the different data sets (only when <code>dataSet</code> is specified).</p>
</td></tr>
<tr><td><code id="doCBS_+3A_arrays">arrays</code></td>
<td>
<p>An optional <code><a href="base.html#topic+vector">vector</a></code> specifying the subset of arrays to process.</p>
</td></tr>
<tr><td><code id="doCBS_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+CbsModel">CbsModel</a></code> and
its <code>fit()</code> method.</p>
</td></tr>
<tr><td><code id="doCBS_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the output dataset of <code><a href="#topic+CbsModel">CbsModel</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 
dataSet &lt;- "GSE20939"
tags &lt;- "ACC,-XY,BPN,-XY,RMA,FLN,-XY"
chipTypes &lt;- c("Mapping250K_Nsp", "Mapping250K_Sty")
fit &lt;- doCBS(dataSet, tags=tags, chipTypes=chipTypes, verbose=-10)
print(fit)


## End(Not run)</code></pre>

<hr>
<h2 id='downloadChipTypeFile.AromaRepository'>Download a particular chip type annotation file</h2><span id='topic+downloadChipTypeFile.AromaRepository'></span><span id='topic+AromaRepository.downloadChipTypeFile'></span><span id='topic+downloadChipTypeFile+2CAromaRepository-method'></span>

<h3>Description</h3>

<p>Download a particular chip type annotation file by its chip type, tags and suffix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaRepository'
downloadChipTypeFile(this, chipType, tags=NULL, suffix=sprintf(".%s", ext), ext=NULL,
  ..., gunzip=TRUE, skip=TRUE, overwrite=FALSE, mustExist=TRUE, verbose=getVerbose(this))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="downloadChipTypeFile.AromaRepository_+3A_chiptype">chipType</code></td>
<td>
<p>The chip type of the file to be downloaded.</p>
</td></tr>
<tr><td><code id="downloadChipTypeFile.AromaRepository_+3A_tags">tags</code></td>
<td>
<p>Optional tags of the file to be downloaded.</p>
</td></tr>
<tr><td><code id="downloadChipTypeFile.AromaRepository_+3A_suffix">suffix</code></td>
<td>
<p>The filename suffix (including any preceding period) of
the file to be downloaded.</p>
</td></tr>
<tr><td><code id="downloadChipTypeFile.AromaRepository_+3A_ext">ext</code></td>
<td>
<p>The filename extension.</p>
</td></tr>
<tr><td><code id="downloadChipTypeFile.AromaRepository_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+downloadFile.AromaRepository">*downloadFile</a>()</code>.</p>
</td></tr>
<tr><td><code id="downloadChipTypeFile.AromaRepository_+3A_skip">skip</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, an already downloaded file is skipped.</p>
</td></tr>
<tr><td><code id="downloadChipTypeFile.AromaRepository_+3A_overwrite">overwrite</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, an not skipping, an already downloaded file
is overwritten, otherwise an error is thrown.</p>
</td></tr>
<tr><td><code id="downloadChipTypeFile.AromaRepository_+3A_mustexist">mustExist</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, an exception is thrown if no file matching
is available either locally or on the repository.</p>
</td></tr>
<tr><td><code id="downloadChipTypeFile.AromaRepository_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the relative pathname of the uncompressed downloaded file.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AromaRepository">AromaRepository</a></code>.
</p>

<hr>
<h2 id='downloadFile.AromaRepository'>Download a particular file from the repository</h2><span id='topic+downloadFile.AromaRepository'></span><span id='topic+AromaRepository.downloadFile'></span><span id='topic+downloadFile+2CAromaRepository-method'></span>

<h3>Description</h3>

<p>Download a particular file from the repository by its relative pathname.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaRepository'
downloadFile(this, filename, path=NULL, caseSensitive=FALSE, gzipped=TRUE, skip=TRUE,
  overwrite=FALSE, ..., verbose=getVerbose(this))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="downloadFile.AromaRepository_+3A_filename">filename</code>, <code id="downloadFile.AromaRepository_+3A_path">path</code></td>
<td>
<p>The filename and the relative path of the file
to be download.</p>
</td></tr>
<tr><td><code id="downloadFile.AromaRepository_+3A_gzipped">gzipped</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, a gzipped file is downloaded and decompressed.</p>
</td></tr>
<tr><td><code id="downloadFile.AromaRepository_+3A_skip">skip</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, an already downloaded file is skipped.</p>
</td></tr>
<tr><td><code id="downloadFile.AromaRepository_+3A_overwrite">overwrite</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, an not skipping, an already downloaded file
is overwritten, otherwise an error is thrown.</p>
</td></tr>
<tr><td><code id="downloadFile.AromaRepository_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="R.utils.html#topic+downloadFile">downloadFile</a></code>.</p>
</td></tr>
<tr><td><code id="downloadFile.AromaRepository_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the pathname of the uncompressed downloaded file.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AromaRepository">AromaRepository</a></code>.
</p>

<hr>
<h2 id='estimateSds.CopyNumberChromosomalModel'>Estimates the standard deviation of the raw copy numbers (log2-ratios) robustly</h2><span id='topic+estimateSds.CopyNumberChromosomalModel'></span><span id='topic+CopyNumberChromosomalModel.estimateSds'></span><span id='topic+estimateSds+2CCopyNumberChromosomalModel-method'></span>

<h3>Description</h3>

<p>Estimates the standard deviation of the raw copy numbers (log2-ratios) robustly using a first-order difference variance estimator, which is
an estimator that is fairly robust for change points.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CopyNumberChromosomalModel'
estimateSds(this, arrays=seq_len(nbrOfArrays(this)), chromosomes=getChromosomes(this),
  ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="estimateSds.CopyNumberChromosomalModel_+3A_arrays">arrays</code></td>
<td>
<p>The arrays to be queried.</p>
</td></tr>
<tr><td><code id="estimateSds.CopyNumberChromosomalModel_+3A_chromosomes">chromosomes</code></td>
<td>
<p>The chromosomes to be queried.</p>
</td></tr>
<tr><td><code id="estimateSds.CopyNumberChromosomalModel_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to
<code><a href="#topic+extractRawCopyNumbers.CopyNumberChromosomalModel">*extractRawCopyNumbers</a>()</code>.</p>
</td></tr>
<tr><td><code id="estimateSds.CopyNumberChromosomalModel_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a CxK <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+matrix">matrix</a></code> where C is the number of chromosomes,
and K is the number of arrays (arrays are always the last dimension).
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+CopyNumberChromosomalModel">CopyNumberChromosomalModel</a></code>.
</p>

<hr>
<h2 id='estimateStandardDeviation.RawGenomicSignals'>Estimates the standard deviation of the raw Ys</h2><span id='topic+estimateStandardDeviation.RawGenomicSignals'></span><span id='topic+RawGenomicSignals.estimateStandardDeviation'></span><span id='topic+estimateStandardDeviation+2CRawGenomicSignals-method'></span>

<h3>Description</h3>

<p>Estimates the standard deviation of the raw Ys robustly or non-robustly using either a &quot;direct&quot; estimator
or a first-order difference estimator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'RawGenomicSignals'
estimateStandardDeviation(this, field=NULL, method=c("diff", "direct"),
  estimator=c("mad", "sd"), na.rm=TRUE, weights=getWeights(this), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="estimateStandardDeviation.RawGenomicSignals_+3A_field">field</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> specifying the field to estimate.</p>
</td></tr>
<tr><td><code id="estimateStandardDeviation.RawGenomicSignals_+3A_method">method</code></td>
<td>
<p>If <code>"diff"</code>, the estimate is based on the first-order
contiguous differences of raw Ys. If <code>"direct"</code>, it is based
directly on the raw Ys.</p>
</td></tr>
<tr><td><code id="estimateStandardDeviation.RawGenomicSignals_+3A_estimator">estimator</code></td>
<td>
<p>If <code>"mad"</code>, the robust <code><a href="stats.html#topic+mad">mad</a></code> estimator
is used.  If <code>"sd"</code>, the <code><a href="stats.html#topic+sd">sd</a></code> estimator is used.</p>
</td></tr>
<tr><td><code id="estimateStandardDeviation.RawGenomicSignals_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, missing values are excluded first.</p>
</td></tr>
<tr><td><code id="estimateStandardDeviation.RawGenomicSignals_+3A_weights">weights</code></td>
<td>
<p>Locus specific weights.</p>
</td></tr>
<tr><td><code id="estimateStandardDeviation.RawGenomicSignals_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a non-negative <code><a href="base.html#topic+numeric">numeric</a></code> value.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+diff">diff</a></code>(), <code><a href="stats.html#topic+sd">sd</a></code>, and <code><a href="stats.html#topic+mad">mad</a></code>.
For more information see <code><a href="#topic+RawGenomicSignals">RawGenomicSignals</a></code>.
</p>

<hr>
<h2 id='Explorer'>The Explorer class</h2><span id='topic+Explorer'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class Explorer</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><em><code>Explorer</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+ChromosomeExplorer">ChromosomeExplorer</a><br />
</p>
<p>public abstract static class <b>Explorer</b><br />
extends <a href="R.oo.html#topic+Object">Object</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Explorer(tags="*", version="0", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Explorer_+3A_tags">tags</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> of tags to be added to the output path.</p>
</td></tr>
<tr><td><code id="Explorer_+3A_version">version</code></td>
<td>
<p>An optional <code><a href="base.html#topic+character">character</a></code> string.</p>
</td></tr>
<tr><td><code id="Explorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+display.Explorer">display</a></code> </td><td style="text-align: left;"> Displays the explorer in the default browser.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFullName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getName.Explorer">getName</a></code> </td><td style="text-align: left;"> Gets the name of the explorer.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getNames.Explorer">getNames</a></code> </td><td style="text-align: left;"> Gets the names of the input samples.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getPath.Explorer">getPath</a></code> </td><td style="text-align: left;"> Gets the path of the output directory.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getRootPath.Explorer">getRootPath</a></code> </td><td style="text-align: left;"> Gets the root path of the output directory.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getTags.Explorer">getTags</a></code> </td><td style="text-align: left;"> Gets the tags of the explorer.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getVersion</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+nbrOfArrays.Explorer">nbrOfArrays</a></code> </td><td style="text-align: left;"> Gets the total number of arrays.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+process.Explorer">process</a></code> </td><td style="text-align: left;"> Generates image files, scripts and dynamic pages for the explorer.</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+setArrays.Explorer">setArrays</a></code> </td><td style="text-align: left;"> Sets the arrays.</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Output directory structure</h3>

<p>The <em>main directory</em> of an Explorer report is
<code>reports/&lt;name&gt;/&lt;subname&gt;/</code>.
The <code>&lt;name&gt;</code> is typically the same as the name of the input
data set, and the <code>&lt;subname&gt;</code> is typically the tags of ditto.
This main directory is where main HTML document is stored.
</p>
<p>For each chip type, real or &quot;virtual&quot; (combined), there is a
subdirectory with the same name as the chip type, i.e.
<code>reports/&lt;name&gt;/&lt;subname&gt;/&lt;chiptype&gt;/</code>.
</p>
<p>For each chip type directory, there are set of subdirectories each
specifying a so called <em>image layer</em>, e.g. an image layer
showing the raw data, another containing the estimates of a model
fit and so on.  Path format:
<code>reports/&lt;name&gt;/&lt;subname&gt;/&lt;chiptype&gt;/&lt;image layer&gt;/</code>.
In this directory all image files are stored, e.g. PNG files.
</p>
<p>In some cases one do not want to all input tags to become part of the
subname, but instead for instance use those to name the image layer(s).
In such cases one has to override the default names.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='exportAromaUnitPscnBinarySet'>Export total and allele B signal data sets as a unified parent-specific copy number signal data set</h2><span id='topic+exportAromaUnitPscnBinarySet'></span><span id='topic+exportAromaUnitPscnBinarySet.AromaUnitTotalCnBinarySet'></span><span id='topic+exportAromaUnitPscnBinarySet.list'></span>

<h3>Description</h3>

<p>Export total and allele B signal data sets as a unified parent-specific copy number signal data set, where each sample is stored in one data file (contrary to the input data sets where each sample is stored in two separated files).
</p>


<h3>Usage</h3>

<pre><code class='language-R'> ## S3 method for class 'AromaUnitTotalCnBinarySet'
exportAromaUnitPscnBinarySet(dsT, dsB="*", dataSet="*", tags="*", ...,
  rootPath="totalAndFracBData/", overwrite=!skip, skip=TRUE, verbose=FALSE)
 ## S3 method for class 'list'
exportAromaUnitPscnBinarySet(dsList, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="exportAromaUnitPscnBinarySet_+3A_dst">dsT</code>, <code id="exportAromaUnitPscnBinarySet_+3A_dsb">dsB</code></td>
<td>
<p>An <code><a href="#topic+AromaUnitTotalCnBinarySet">AromaUnitTotalCnBinarySet</a></code> and an <code><a href="#topic+AromaUnitFracBCnBinarySet">AromaUnitFracBCnBinarySet</a></code> with coupled sets of samples that match up by name.  If they don't match up, an exception is thrown.  The allele B fraction (BAF) data set <code>dsB</code> is by default inferred from the total CN data set <code>dsT</code>.</p>
</td></tr>
<tr><td><code id="exportAromaUnitPscnBinarySet_+3A_dataset">dataSet</code>, <code id="exportAromaUnitPscnBinarySet_+3A_tags">tags</code></td>
<td>
<p>The name and tags of the output data set.
The default is to infer those from the input <code>dsT</code> data set.</p>
</td></tr>
<tr><td><code id="exportAromaUnitPscnBinarySet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="exportAromaUnitPscnBinarySet_+3A_rootpath">rootPath</code></td>
<td>
<p>The root path of the output data set.</p>
</td></tr>
<tr><td><code id="exportAromaUnitPscnBinarySet_+3A_overwrite">overwrite</code>, <code id="exportAromaUnitPscnBinarySet_+3A_skip">skip</code></td>
<td>
<p>Specifies whether to overwrite and/or skip already exported samples.</p>
</td></tr>
<tr><td><code id="exportAromaUnitPscnBinarySet_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AromaUnitPscnBinarySet">AromaUnitPscnBinarySet</a></code> object.
</p>


<h3>Allele-specific CRMAv2 pipeline</h3>

<p>A common use case is to run allele-specific CRMAv2, e.g.
<code>dsNList &lt;- doASCRMAv2(csR)</code>, which outputs a <code><a href="base.html#topic+list">list</a></code> <code>dsNList</code>
with elements corresponding to <code>dsT</code> and <code>dsB</code>.  This output
can be exported to <code><a href="#topic+AromaUnitPscnBinarySet">AromaUnitPscnBinarySet</a></code> by this method as
<code>dsN &lt;- exportAromaUnitPscnBinarySet(dsNList)</code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='extractRawCopyNumbers.CopyNumberChromosomalModel'>Extracts relative copy numbers</h2><span id='topic+extractRawCopyNumbers.CopyNumberChromosomalModel'></span><span id='topic+CopyNumberChromosomalModel.extractRawCopyNumbers'></span><span id='topic+extractRawCopyNumbers+2CCopyNumberChromosomalModel-method'></span>

<h3>Description</h3>

<p>Extracts relative copy numbers for a particular array and chromosome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CopyNumberChromosomalModel'
extractRawCopyNumbers(this, array, chromosome, ..., logBase=2, cache=FALSE, force=FALSE,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extractRawCopyNumbers.CopyNumberChromosomalModel_+3A_array">array</code></td>
<td>
<p>The index of the array to be extracted.</p>
</td></tr>
<tr><td><code id="extractRawCopyNumbers.CopyNumberChromosomalModel_+3A_chromosome">chromosome</code></td>
<td>
<p>The index of the chromosome to be extracted.</p>
</td></tr>
<tr><td><code id="extractRawCopyNumbers.CopyNumberChromosomalModel_+3A_...">...</code></td>
<td>
<p>See subclasses.</p>
</td></tr>
<tr><td><code id="extractRawCopyNumbers.CopyNumberChromosomalModel_+3A_logbase">logBase</code></td>
<td>
<p>(optional) The base of the logarithm used for the ratios.
If <code><a href="base.html#topic+NULL">NULL</a></code>, the ratios are not logged.</p>
</td></tr>
<tr><td><code id="extractRawCopyNumbers.CopyNumberChromosomalModel_+3A_cache">cache</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, results are cached, otherwise not.</p>
</td></tr>
<tr><td><code id="extractRawCopyNumbers.CopyNumberChromosomalModel_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, cached results are ignored.</p>
</td></tr>
<tr><td><code id="extractRawCopyNumbers.CopyNumberChromosomalModel_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See subclasses.
</p>


<h3>Parallel processing</h3>

<p>Except for an in-object caching (<code>cache=TRUE</code>), this method
access data solely in an read-only fashion.
This method is safe to call with different arrays and/or
chromosomes in parallel.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+CopyNumberChromosomalModel">CopyNumberChromosomalModel</a></code>.
</p>

<hr>
<h2 id='FileCacheKeyInterface'>The FileCacheKeyInterface class interface</h2><span id='topic+FileCacheKeyInterface'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class FileCacheKeyInterface</b><br />
</p>
<p><code><a href="R.oo.html#topic+Interface">Interface</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code>FileCacheKeyInterface</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+AromaCellCpgFile">AromaCellCpgFile</a>, <a href="#topic+AromaCellPositionFile">AromaCellPositionFile</a>, <a href="#topic+AromaCellSequenceFile">AromaCellSequenceFile</a>, <em><a href="#topic+AromaCellTabularBinaryFile">AromaCellTabularBinaryFile</a></em>, <em><a href="#topic+AromaGenomeTextFile">AromaGenomeTextFile</a></em>, <em><a href="#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></em>, <em><a href="#topic+AromaMicroarrayTabularBinaryFile">AromaMicroarrayTabularBinaryFile</a></em>, <em><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></em>, <em><a href="#topic+AromaUcscGenomeTextFile">AromaUcscGenomeTextFile</a></em>, <a href="#topic+AromaUflFile">AromaUflFile</a>, <a href="#topic+AromaUgcFile">AromaUgcFile</a>, <a href="#topic+AromaUgpFile">AromaUgpFile</a>, <a href="#topic+AromaUnitCallFile">AromaUnitCallFile</a>, <em><a href="#topic+AromaUnitChromosomeTabularBinaryFile">AromaUnitChromosomeTabularBinaryFile</a></em>, <a href="#topic+AromaUnitFracBCnBinaryFile">AromaUnitFracBCnBinaryFile</a>, <a href="#topic+AromaUnitGcContentFile">AromaUnitGcContentFile</a>, <a href="#topic+AromaUnitGenotypeCallFile">AromaUnitGenotypeCallFile</a>, <em><a href="#topic+AromaUnitPscnBinaryFile">AromaUnitPscnBinaryFile</a></em>, <a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a>, <em><a href="#topic+AromaUnitTabularBinaryFile">AromaUnitTabularBinaryFile</a></em>, <em><a href="#topic+AromaUnitTotalCnBinaryFile">AromaUnitTotalCnBinaryFile</a></em>, <a href="#topic+AromaUnitTypesFile">AromaUnitTypesFile</a>, <em><a href="#topic+CbsSegmentationDataFile">CbsSegmentationDataFile</a></em>, <em><a href="#topic+SampleAnnotationFile">SampleAnnotationFile</a></em>, <em><a href="#topic+SegmentationDataFile">SegmentationDataFile</a></em>, <a href="#topic+TextUnitNamesFile">TextUnitNamesFile</a><br />
</p>
<p>public class <b>FileCacheKeyInterface</b><br />
extends <a href="#topic+CacheKeyInterface">CacheKeyInterface</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FileCacheKeyInterface(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FileCacheKeyInterface_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code><a href="#topic+getCacheKey.FileCacheKeyInterface">getCacheKey</a></code> </td><td style="text-align: left;"> Gets a list of cache key items.</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from Interface</b>:<br />
extend, print, uses
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='findAnnotationData'>Locates an annotation data file</h2><span id='topic+findAnnotationData.default'></span><span id='topic+findAnnotationData'></span>

<h3>Description</h3>

<p>Locates an annotation data file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
findAnnotationData(name=NULL, tags=NULL, set, pattern=NULL, private=FALSE,
  escapes=c("+"), ..., firstOnly=TRUE, paths=NULL, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="findAnnotationData_+3A_name">name</code></td>
<td>
<p>Optional <code><a href="base.html#topic+character">character</a></code> string.</p>
</td></tr>
<tr><td><code id="findAnnotationData_+3A_tags">tags</code></td>
<td>
<p>Optional <code><a href="base.html#topic+character">character</a></code> string.
Only used if argument <code>pattern</code> is not specified.</p>
</td></tr>
<tr><td><code id="findAnnotationData_+3A_pattern">pattern</code></td>
<td>
<p>A filename pattern to search for.
If <code><a href="base.html#topic+NULL">NULL</a></code>, then defaults to the fullname as defined by
arguments <code>name</code> and <code>tags</code>.</p>
</td></tr>
<tr><td><code id="findAnnotationData_+3A_private">private</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">FALSE</a></code>, files and directories starting
with a periods are ignored.</p>
</td></tr>
<tr><td><code id="findAnnotationData_+3A_escapes">escapes</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> specify symbols to be escaped
in argument <code>pattern</code>.</p>
</td></tr>
<tr><td><code id="findAnnotationData_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="R.utils.html#topic+findFiles">findFiles</a></code>.</p>
</td></tr>
<tr><td><code id="findAnnotationData_+3A_firstonly">firstOnly</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, only the first matching pathname is returned.</p>
</td></tr>
<tr><td><code id="findAnnotationData_+3A_paths">paths</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> of paths to search.
If <code><a href="base.html#topic+NULL">NULL</a></code>, default paths are used.</p>
</td></tr>
<tr><td><code id="findAnnotationData_+3A_set">set</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying what type of annotation
to search for.</p>
</td></tr>
<tr><td><code id="findAnnotationData_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code><a href="base.html#topic+NULL">NULL</a></code>, one or several matching pathnames.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='findAnnotationDataByChipType'>Locates an annotation data file by its chip type</h2><span id='topic+findAnnotationDataByChipType.default'></span><span id='topic+findAnnotationDataByChipType'></span>

<h3>Description</h3>

<p>Locates an annotation data file by its chip type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
findAnnotationDataByChipType(chipType, pattern=chipType, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="findAnnotationDataByChipType_+3A_chiptype">chipType</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string.</p>
</td></tr>
<tr><td><code id="findAnnotationDataByChipType_+3A_pattern">pattern</code></td>
<td>
<p>A filename pattern to search for.</p>
</td></tr>
<tr><td><code id="findAnnotationDataByChipType_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+findAnnotationData">findAnnotationData</a></code>().</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='findFilesTodo.AromaTransform'>Finds files in the data set still not processed</h2><span id='topic+findFilesTodo.AromaTransform'></span><span id='topic+AromaTransform.findFilesTodo'></span><span id='topic+findFilesTodo+2CAromaTransform-method'></span>

<h3>Description</h3>

<p>Finds files in the data set still not processed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaTransform'
findFilesTodo(this, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="findFilesTodo.AromaTransform_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="findFilesTodo.AromaTransform_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying the indices of the files
still not processed.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AromaTransform">AromaTransform</a></code>.
</p>

<hr>
<h2 id='fit.CopyNumberChromosomalModel'>Fits the model</h2><span id='topic+fit.CopyNumberChromosomalModel'></span><span id='topic+CopyNumberChromosomalModel.fit'></span><span id='topic+fit+2CCopyNumberChromosomalModel-method'></span>

<h3>Description</h3>

<p>Fits the model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CopyNumberChromosomalModel'
fit(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit.CopyNumberChromosomalModel_+3A_...">...</code></td>
<td>
<p>See subclasses.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See subclasses.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+CopyNumberChromosomalModel">CopyNumberChromosomalModel</a></code>.
</p>

<hr>
<h2 id='fit.CopyNumberSegmentationModel'>Fits the model</h2><span id='topic+fit.CopyNumberSegmentationModel'></span><span id='topic+CopyNumberSegmentationModel.fit'></span><span id='topic+fit+2CCopyNumberSegmentationModel-method'></span>

<h3>Description</h3>

<p>Fits the model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CopyNumberSegmentationModel'
fit(this, arrays=NULL, chromosomes=getChromosomes(this), force=FALSE, ...,
  .retResults=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit.CopyNumberSegmentationModel_+3A_arrays">arrays</code></td>
<td>
<p>A <code><a href="base.html#topic+vector">vector</a></code> of array indices specifying which arrays to
be considered.  If <code><a href="base.html#topic+NULL">NULL</a></code>, all are processed.</p>
</td></tr>
<tr><td><code id="fit.CopyNumberSegmentationModel_+3A_chromosomes">chromosomes</code></td>
<td>
<p>A <code><a href="base.html#topic+vector">vector</a></code> of chromosomes indices specifying which
chromosomes to be considered.  If <code><a href="base.html#topic+NULL">NULL</a></code>, all are processed.</p>
</td></tr>
<tr><td><code id="fit.CopyNumberSegmentationModel_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">FALSE</a></code>, the model will not be fitted again if it was
already fitted.</p>
</td></tr>
<tr><td><code id="fit.CopyNumberSegmentationModel_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the segmentation method for
the <code><a href="#topic+RawGenomicSignals">RawGenomicSignals</a></code>.</p>
</td></tr>
<tr><td><code id="fit.CopyNumberSegmentationModel_+3A_.retresults">.retResults</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the segmentation fit structures are
returned for each fitted array and chromosome.</p>
</td></tr>
<tr><td><code id="fit.CopyNumberSegmentationModel_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named <code><a href="base.html#topic+list">list</a></code> of named <code><a href="base.html#topic+list">list</a></code>s.
</p>


<h3>Additional arguments to the internal fit function</h3>

<p>Arguments in <code>...</code> are passed down to the internal fit function,
which means that it is possible to fine tune even further.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+CopyNumberSegmentationModel">CopyNumberSegmentationModel</a></code>.
</p>

<hr>
<h2 id='fitGenotypeCone.matrix'>Fits an affine transformation to allele A and allele B data</h2><span id='topic+fitGenotypeCone.matrix'></span>

<h3>Description</h3>

<p>Fits an affine transformation to allele A and allele B data using robust estimators.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'matrix'
fitGenotypeCone(y, flavor=c("sfit", "expectile"), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fitGenotypeCone.matrix_+3A_y">y</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> Nx2 <code><a href="base.html#topic+matrix">matrix</a></code> with one column for each allele and
where N is the number of data points.</p>
</td></tr>
<tr><td><code id="fitGenotypeCone.matrix_+3A_flavor">flavor</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying what model/algorithm
should be used to fit the genotype cone.</p>
</td></tr>
<tr><td><code id="fitGenotypeCone.matrix_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the internal fit <code><a href="base.html#topic+function">function</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named <code><a href="base.html#topic+list">list</a></code> structure.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>To backtransform data fitted using this method,
see <code><a href="#topic+backtransformGenotypeCone">backtransformGenotypeCone</a></code>().
Internally, the <code>cfit()</code> function the <span class="pkg">sfit</span> package is used.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Fit genotype cone based on available methods (==packages)
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
flavors &lt;- c("expectile", "sfit")
## The 'expectile' package does not do proper S3 registration
if (getRversion() &gt;= "3.6.0") flavors &lt;- setdiff(flavors, "expectile")
keep &lt;- sapply(flavors, FUN=require, character.only=TRUE)
flavors &lt;- flavors[keep]
cat("Available fitting flavors:", paste(flavors, collapse=", "), "\n")
hasSfit &lt;- is.element("sfit", flavors)


# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Simulate data (taken from the cfit.matrix() example of 'sfit')
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
#set.seed(0xbeef)

N &lt;- 1000

# Simulate genotypes
g &lt;- sample(c("AA", "AB", "AB", "BB"), size=N, replace=TRUE)

# Simulate concentrations of allele A and allele B
X &lt;- matrix(rexp(N), nrow=N, ncol=2)
colnames(X) &lt;- c("A", "B")
X[g == "AA", "B"] &lt;- 0
X[g == "BB", "A"] &lt;- 0
X[g == "AB",] &lt;- X[g == "AB",] / 2

# Transform noisy X
xi &lt;- matrix(rnorm(2*N, mean=0, sd=0.05), ncol=2)
a0 &lt;- c(0,0)+0.3
A &lt;- matrix(c(0.9, 0.1, 0.1, 0.8), nrow=2, byrow=TRUE)
A &lt;- apply(A, MARGIN=2, FUN=function(u) u / sqrt(sum(u^2)))
Z &lt;- t(a0 + A %*% t(X + xi))

# Add noise to Y
eps &lt;- matrix(rnorm(2*N, mean=0, sd=0.05), ncol=2)
Y &lt;- Z + eps


lim &lt;- c(-1/2,6)
xlab &lt;- "Allele A"
ylab &lt;- "Allele B"
plot(Y, xlab=xlab, ylab=ylab, xlim=lim, ylim=lim)
lines(x=c(0,0,2*lim[2]), y=c(2*lim[2],0,0), col="#aaaaaa", lwd=3, lty=3)

# Different alpha sequences to illustrate the impact
alphas &lt;- c(0.075, 0.05, 0.01, 0.03, 0.002, 0.001)

cols &lt;- seq(from=2, to=length(alphas)+1)
legend("topright", sprintf("%.3f", alphas), col=cols, lwd=4, title="Alphas")

for (kk in seq_along(alphas)) {
  for (flavor in flavors) {
    fit &lt;- fitGenotypeCone(Y, alpha=alphas[kk], flavor=flavor)
    YN &lt;- backtransformGenotypeCone(Y, fit=fit)
    if (hasSfit) {
      radials(fit$M, lwd=3, col=cols[kk], lty=ifelse(flavor == "sfit", 1,2))
      drawApex(fit$M, col=cols[kk], pch=19, cex=2)
    }
    points(YN, col=cols[kk])
  }
}
lines(x=c(0,0,2*lim[2]), y=c(2*lim[2],0,0), col="#aaaaaa", lwd=3, lty=3)
</code></pre>

<hr>
<h2 id='fitGenotypeConeBySfit.matrix'>Fits an affine transformation to allele A and allele B data</h2><span id='topic+fitGenotypeConeBySfit.matrix'></span>

<h3>Description</h3>

<p>Fits an affine transformation to allele A and allele B data using robust estimators.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'matrix'
fitGenotypeConeBySfit(y, alpha=c(0.1, 0.075, 0.05, 0.03, 0.01), q=2, Q=98, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fitGenotypeConeBySfit.matrix_+3A_y">y</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> Nx2 <code><a href="base.html#topic+matrix">matrix</a></code> with one column for each allele and
where N is the number of data points.</p>
</td></tr>
<tr><td><code id="fitGenotypeConeBySfit.matrix_+3A_alpha">alpha</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of decreasing values in (0,1).
This parameter &quot;determines how far we are willing to press the
boundary of the [genotype cone]&quot;.  Lowering <code>alpha</code> expand
the cone.  When <code>alpha</code> goes to zero, all data points will
be on or inside the cone.</p>
</td></tr>
<tr><td><code id="fitGenotypeConeBySfit.matrix_+3A_q">q</code>, <code id="fitGenotypeConeBySfit.matrix_+3A_q">Q</code></td>
<td>
<p>Percentiles in [0,100] for which data points that are
below (above) will be assigned zero weight in the fitting of
the parameters.</p>
</td></tr>
<tr><td><code id="fitGenotypeConeBySfit.matrix_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the <code>cfit()</code> of
the <span class="pkg">sfit</span> package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the parameter estimates as a named <code><a href="base.html#topic+list">list</a></code> with elements:
</p>
<table role = "presentation">
<tr><td><code>M</code></td>
<td>
<p>An estimate of the three vertices defining the genotype
triangle.  These three vertices are describes as an 2x3 <code><a href="base.html#topic+matrix">matrix</a></code>
with column <code>origin</code>, <code>AA</code>, and <code>BB</code>.</p>
</td></tr>
<tr><td><code>Minv</code></td>
<td>
<p>The inverse of <code>M</code>.</p>
</td></tr>
<tr><td><code>origin</code></td>
<td>
<p>The estimate of the shift.</p>
</td></tr>
<tr><td><code>W</code></td>
<td>
<p>The estimate of shear/rotation matrix with columns
<code>AA</code> and <code>BB</code>.</p>
</td></tr>
<tr><td><code>Winv</code></td>
<td>
<p>The inverse of <code>W</code>.</p>
</td></tr>
<tr><td><code>params</code></td>
<td>
<p>The parameters used for the fit, i.e.
<code>alpha</code>, <code>q</code>, <code>Q</code>, and  those passed in <code>...</code>.</p>
</td></tr>
<tr><td><code>dimData</code></td>
<td>
<p>The dimension of the input data.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>To backtransform data fitted using this method,
see <code><a href="#topic+backtransformGenotypeCone">backtransformGenotypeCone</a></code>().
Internally, the <code>cfit()</code> function the <span class="pkg">sfit</span> package is used.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Fit genotype cone based on available methods (==packages)
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
flavors &lt;- c("expectile", "sfit")
## The 'expectile' package does not do proper S3 registration
if (getRversion() &gt;= "3.6.0") flavors &lt;- setdiff(flavors, "expectile")
keep &lt;- sapply(flavors, FUN=require, character.only=TRUE)
flavors &lt;- flavors[keep]
cat("Available fitting flavors:", paste(flavors, collapse=", "), "\n")
hasSfit &lt;- is.element("sfit", flavors)


# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Simulate data (taken from the cfit.matrix() example of 'sfit')
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
#set.seed(0xbeef)

N &lt;- 1000

# Simulate genotypes
g &lt;- sample(c("AA", "AB", "AB", "BB"), size=N, replace=TRUE)

# Simulate concentrations of allele A and allele B
X &lt;- matrix(rexp(N), nrow=N, ncol=2)
colnames(X) &lt;- c("A", "B")
X[g == "AA", "B"] &lt;- 0
X[g == "BB", "A"] &lt;- 0
X[g == "AB",] &lt;- X[g == "AB",] / 2

# Transform noisy X
xi &lt;- matrix(rnorm(2*N, mean=0, sd=0.05), ncol=2)
a0 &lt;- c(0,0)+0.3
A &lt;- matrix(c(0.9, 0.1, 0.1, 0.8), nrow=2, byrow=TRUE)
A &lt;- apply(A, MARGIN=2, FUN=function(u) u / sqrt(sum(u^2)))
Z &lt;- t(a0 + A %*% t(X + xi))

# Add noise to Y
eps &lt;- matrix(rnorm(2*N, mean=0, sd=0.05), ncol=2)
Y &lt;- Z + eps


lim &lt;- c(-1/2,6)
xlab &lt;- "Allele A"
ylab &lt;- "Allele B"
plot(Y, xlab=xlab, ylab=ylab, xlim=lim, ylim=lim)
lines(x=c(0,0,2*lim[2]), y=c(2*lim[2],0,0), col="#aaaaaa", lwd=3, lty=3)

# Different alpha sequences to illustrate the impact
alphas &lt;- c(0.075, 0.05, 0.01, 0.03, 0.002, 0.001)

cols &lt;- seq(from=2, to=length(alphas)+1)
legend("topright", sprintf("%.3f", alphas), col=cols, lwd=4, title="Alphas")

for (kk in seq_along(alphas)) {
  for (flavor in flavors) {
    fit &lt;- fitGenotypeCone(Y, alpha=alphas[kk], flavor=flavor)
    YN &lt;- backtransformGenotypeCone(Y, fit=fit)
    if (hasSfit) {
      radials(fit$M, lwd=3, col=cols[kk], lty=ifelse(flavor == "sfit", 1,2))
      drawApex(fit$M, col=cols[kk], pch=19, cex=2)
    }
    points(YN, col=cols[kk])
  }
}
lines(x=c(0,0,2*lim[2]), y=c(2*lim[2],0,0), col="#aaaaaa", lwd=3, lty=3)
</code></pre>

<hr>
<h2 id='fitMultiDimensionalCone.matrix'>Fits an affine transformation to multi-dimensional data</h2><span id='topic+fitMultiDimensionalCone.matrix'></span>

<h3>Description</h3>

<p>Fits an affine transformation to multi-dimensional data using robust estimators.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'matrix'
fitMultiDimensionalCone(y, alpha=c(0.1, 0.075, 0.05, 0.03, 0.01), q=2, Q=98, ...,
  flavor=c("sfit", "expectile"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fitMultiDimensionalCone.matrix_+3A_y">y</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> NxK <code><a href="base.html#topic+matrix">matrix</a></code> with one column for each dimension and
where N is the number of data points.</p>
</td></tr>
<tr><td><code id="fitMultiDimensionalCone.matrix_+3A_alpha">alpha</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of decreasing values in (0,1).
This parameter &quot;determines how far we are willing to press the
boundary of the [genotype cone]&quot;.  Lowering <code>alpha</code> expand
the cone.  When <code>alpha</code> goes to zero, all data points will
be on or inside the cone.</p>
</td></tr>
<tr><td><code id="fitMultiDimensionalCone.matrix_+3A_q">q</code>, <code id="fitMultiDimensionalCone.matrix_+3A_q">Q</code></td>
<td>
<p>Percentiles in [0,100] for which data points that are
below (above) will be assigned zero weight in the fitting of
the parameters.</p>
</td></tr>
<tr><td><code id="fitMultiDimensionalCone.matrix_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the <code>cfit()</code> function
of the <span class="pkg">sfit</span> package.</p>
</td></tr>
<tr><td><code id="fitMultiDimensionalCone.matrix_+3A_flavor">flavor</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying what model/algorithm
should be used to fit the genotype cone.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the parameter estimates as a named <code><a href="base.html#topic+list">list</a></code> with elements:
</p>
<table role = "presentation">
<tr><td><code>M</code></td>
<td>
<p>An estimate of the three vertices defining the genotype
triangle.  These three vertices are describes as an 2x3 <code><a href="base.html#topic+matrix">matrix</a></code>
with column <code>origin</code>, <code>AA</code>, and <code>BB</code>.</p>
</td></tr>
<tr><td><code>Minv</code></td>
<td>
<p>The inverse of <code>M</code>.</p>
</td></tr>
<tr><td><code>origin</code></td>
<td>
<p>The estimate of the shift.</p>
</td></tr>
<tr><td><code>W</code></td>
<td>
<p>The estimate of shear/rotation matrix with columns
<code>AA</code> and <code>BB</code>.</p>
</td></tr>
<tr><td><code>Winv</code></td>
<td>
<p>The inverse of <code>W</code>.</p>
</td></tr>
<tr><td><code>params</code></td>
<td>
<p>The parameters used for the fit, i.e.
<code>alpha</code>, <code>q</code>, <code>Q</code>, and  those passed in <code>...</code>.</p>
</td></tr>
<tr><td><code>dimData</code></td>
<td>
<p>The dimension of the input data.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>To backtransform data fitted using this method,
see <code><a href="#topic+backtransformMultiDimensionalCone.matrix">*backtransformMultiDimensionalCone</a>()</code>.
Internally, the <code>cfit()</code> function the <span class="pkg">sfit</span> package is used.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (require("sfit")) {
 # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Simulate data (taken from the cfit.matrix() example of 'sfit')
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
N &lt;- 1000

# Simulate sequences
nucleotides &lt;- c("A", "C", "G", "T")
g &lt;- sample(nucleotides, size=N, replace=TRUE)
ndim &lt;- length(nucleotides)

# Simulate concentrations of allele A and allele B
X &lt;- matrix(rexp(N), nrow=N, ncol=ndim)
colnames(X) &lt;- nucleotides
for (nucleotide in nucleotides) {
  cc &lt;- match(nucleotide, nucleotides)
  X[g == nucleotide, -cc] &lt;- 0
}

# The true offset
a0 &lt;- rep(0.3, times=ndim)

# The crosstalk matrix
A &lt;- matrix(c(
  0.9, 0.3, 0.2, 0.1,
  0.1, 0.8, 0.1, 0.1,
  0.3, 0.4, 0.7, 0.1,
  0.1, 0.1, 0.6, 0.9
), nrow=ndim, byrow=TRUE)
A &lt;- apply(A, MARGIN=2, FUN=function(u) u / sqrt(sum(u^2)))

# Simulate random errors on the input
xi &lt;- matrix(rnorm(ndim*N, mean=0, sd=0.05), ncol=ndim)

# Generate the noisy crosstalk affected input data
Z &lt;- t(a0 + A %*% t(X + xi))

# Generate the noisy observations of the latter
eps &lt;- matrix(rnorm(ndim*N, mean=0, sd=0.05), ncol=ndim)
Y &lt;- Z + eps

# Fit crosstalk model and calibrate data accordingly
fit &lt;- fitMultiDimensionalCone(Y, flavor="sfit")
Yc &lt;- backtransformMultiDimensionalCone(Y, fit=fit)

lim &lt;- c(-0.5,6)
layout(matrix(c(1,2,3,0,4,5,0,0,6), nrow=3, ncol=3, byrow=TRUE))
par(mar=c(5,4,1,1)+0.1)
for (ii in 1:(ndim-1)) {
  for (jj in (ii+1):ndim) {
    cc &lt;- c(jj,ii)
    labs &lt;- nucleotides[cc]
    plot(Y[,cc], cex=0.8, xlim=lim, ylim=lim, xlab=labs[1], ylab=labs[2])
    points(Yc[,cc], cex=0.8, col="blue")
    Mcc &lt;- fit$M[c(1,1+cc),cc]
    class(Mcc) &lt;- class(fit$M)
    lines(Mcc, lwd=2, col="red")
  }
}

}
</code></pre>

<hr>
<h2 id='getAlias.Explorer'>Gets the alias of the output set</h2><span id='topic+getAlias.Explorer'></span><span id='topic+Explorer.getAlias'></span><span id='topic+getAlias+2CExplorer-method'></span>

<h3>Description</h3>

<p>Gets the alias of the output set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Explorer'
getAlias(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getAlias.Explorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code>, or <code><a href="base.html#topic+NULL">NULL</a></code> if no alias is set.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+setAlias.Explorer">*setAlias</a>()</code>.
<code><a href="#topic+getName.Explorer">*getName</a>()</code>.
For more information see <code><a href="#topic+Explorer">Explorer</a></code>.
</p>

<hr>
<h2 id='getAromaPlatform.AromaPlatformInterface'>Gets the platform</h2><span id='topic+getAromaPlatform.AromaPlatformInterface'></span><span id='topic+AromaPlatformInterface.getAromaPlatform'></span><span id='topic+getAromaPlatform+2CAromaPlatformInterface-method'></span>

<h3>Description</h3>

<p>Gets the platform object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaPlatformInterface'
getAromaPlatform(this, ..., force=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getAromaPlatform.AromaPlatformInterface_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="getAromaPlatform.AromaPlatformInterface_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, any cached results are ignored, otherwise not.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AromaPlatform">AromaPlatform</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+getPlatform.AromaPlatformInterface">*getPlatform</a>()</code>.
For more information see <code><a href="#topic+AromaPlatformInterface">AromaPlatformInterface</a></code>.
</p>

<hr>
<h2 id='getCacheKey.CacheKeyInterface'>Gets a list of cache key items</h2><span id='topic+getCacheKey.CacheKeyInterface'></span><span id='topic+CacheKeyInterface.getCacheKey'></span><span id='topic+getCacheKey+2CCacheKeyInterface-method'></span>

<h3>Description</h3>

<p>Gets a list of cache key items that will be added to other cache key items used to
generate the cache key.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CacheKeyInterface'
getCacheKey(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getCacheKey.CacheKeyInterface_+3A_...">...</code></td>
<td>
<p>Additional named arguments to be appended to the list
of key items.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The default list of cache key items are:
</p>

<ul>
<li><p> the class name of the object as a <code><a href="base.html#topic+character">character</a></code> string.
</p>
</li></ul>

<p>Classes extending/implementing this <code><a href="R.oo.html#topic+Interface">Interface</a></code> may
override these items.
</p>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> of cache items.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+CacheKeyInterface">CacheKeyInterface</a></code>.
</p>

<hr>
<h2 id='getCacheKey.FileCacheKeyInterface'>Gets a list of cache key items</h2><span id='topic+getCacheKey.FileCacheKeyInterface'></span><span id='topic+FileCacheKeyInterface.getCacheKey'></span><span id='topic+getCacheKey+2CFileCacheKeyInterface-method'></span>

<h3>Description</h3>

<p>Gets a list of cache key items that will be added to other cache key items used to
generate the cache key.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FileCacheKeyInterface'
getCacheKey(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getCacheKey.FileCacheKeyInterface_+3A_...">...</code></td>
<td>
<p>Additional named arguments to be appended to the list
of key items.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The default list of cache key items are:
</p>

<ul>
<li><p> the class name of the object as a <code><a href="base.html#topic+character">character</a></code> string.
</p>
</li>
<li><p> the file name (aka the basename) as a <code><a href="base.html#topic+character">character</a></code> string.
</p>
</li>
<li><p> the file size as a <code><a href="base.html#topic+numeric">numeric</a></code> scalar.
</p>
</li>
<li><p> the file checksum as a <code><a href="base.html#topic+character">character</a></code> string.
</p>
</li></ul>

<p>Classes extending/implementing this <code><a href="R.oo.html#topic+Interface">Interface</a></code> may override
these items.
</p>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> of cache items.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code>.
</p>

<hr>
<h2 id='getChipType.AromaPlatformInterface'>Gets the chip type</h2><span id='topic+getChipType.AromaPlatformInterface'></span><span id='topic+AromaPlatformInterface.getChipType'></span><span id='topic+getChipType+2CAromaPlatformInterface-method'></span>

<h3>Description</h3>

<p>Gets the chip type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaPlatformInterface'
getChipType(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getChipType.AromaPlatformInterface_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AromaPlatformInterface">AromaPlatformInterface</a></code>.
</p>

<hr>
<h2 id='getChipType.ChromosomalModel'>Gets a label for all chip types merged</h2><span id='topic+getChipType.ChromosomalModel'></span><span id='topic+ChromosomalModel.getChipType'></span><span id='topic+getChipType+2CChromosomalModel-method'></span>

<h3>Description</h3>

<p>Gets a label for all chip types merged.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ChromosomalModel'
getChipType(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getChipType.ChromosomalModel_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ChromosomalModel">ChromosomalModel</a></code>.
</p>

<hr>
<h2 id='getChromosomes.ChromosomalModel'>Gets the chromosomes to be processed</h2><span id='topic+getChromosomes.ChromosomalModel'></span><span id='topic+ChromosomalModel.getChromosomes'></span><span id='topic+getChromosomes+2CChromosomalModel-method'></span><span id='topic+setChromosomes.ChromosomalModel'></span>

<h3>Description</h3>

<p>Gets the chromosomes to be processed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ChromosomalModel'
getChromosomes(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getChromosomes.ChromosomalModel_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ChromosomalModel">ChromosomalModel</a></code>.
</p>

<hr>
<h2 id='getChromosomes.ChromosomeExplorer'>Gets the chromosomes available</h2><span id='topic+getChromosomes.ChromosomeExplorer'></span><span id='topic+ChromosomeExplorer.getChromosomes'></span><span id='topic+getChromosomes+2CChromosomeExplorer-method'></span>

<h3>Description</h3>

<p>Gets the chromosomes available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ChromosomeExplorer'
getChromosomes(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getChromosomes.ChromosomeExplorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ChromosomeExplorer">ChromosomeExplorer</a></code>.
</p>

<hr>
<h2 id='getFullName.AromaTransform'>Gets the full name of the output data set</h2><span id='topic+getFullName.AromaTransform'></span><span id='topic+AromaTransform.getFullName'></span><span id='topic+getFullName+2CAromaTransform-method'></span>

<h3>Description</h3>

<p>Gets the full name of the output data set, which is the name with comma separated tags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaTransform'
getFullName(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getFullName.AromaTransform_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AromaTransform">AromaTransform</a></code>.
</p>

<hr>
<h2 id='getInputDataSet.AromaTransform'>Gets the input data set</h2><span id='topic+getInputDataSet.AromaTransform'></span><span id='topic+AromaTransform.getInputDataSet'></span><span id='topic+getInputDataSet+2CAromaTransform-method'></span>

<h3>Description</h3>

<p>Gets the input data set that is to be (or has been) transformed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaTransform'
getInputDataSet(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getInputDataSet.AromaTransform_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AromaMicroarrayDataSet">AromaMicroarrayDataSet</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AromaTransform">AromaTransform</a></code>.
</p>

<hr>
<h2 id='getModel.ChromosomeExplorer'>Gets the model</h2><span id='topic+getModel.ChromosomeExplorer'></span><span id='topic+ChromosomeExplorer.getModel'></span><span id='topic+getModel+2CChromosomeExplorer-method'></span>

<h3>Description</h3>

<p>Gets the model for which the explorer is displaying it results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ChromosomeExplorer'
getModel(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getModel.ChromosomeExplorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="#topic+CopyNumberChromosomalModel">CopyNumberChromosomalModel</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ChromosomeExplorer">ChromosomeExplorer</a></code>.
</p>

<hr>
<h2 id='getName.AromaTransform'>Gets the name of the output data set</h2><span id='topic+getName.AromaTransform'></span><span id='topic+AromaTransform.getName'></span><span id='topic+getName+2CAromaTransform-method'></span>

<h3>Description</h3>

<p>Gets the name of the output data set, which is the same as the input data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaTransform'
getName(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getName.AromaTransform_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AromaTransform">AromaTransform</a></code>.
</p>

<hr>
<h2 id='getName.Explorer'>Gets the name of the explorer</h2><span id='topic+getName.Explorer'></span><span id='topic+Explorer.getName'></span><span id='topic+getName+2CExplorer-method'></span>

<h3>Description</h3>

<p>Gets the name of the explorer, which is the same as the name of the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Explorer'
getName(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getName.Explorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If a name alias has not been set explicitly, the name of the data set will
used.
</p>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+Explorer">Explorer</a></code>.
</p>

<hr>
<h2 id='getNames.ChromosomalModel'>Gets the names of the arrays</h2><span id='topic+getNames.ChromosomalModel'></span><span id='topic+ChromosomalModel.getNames'></span><span id='topic+getNames+2CChromosomalModel-method'></span>

<h3>Description</h3>

<p>Gets the names of the arrays available to the model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ChromosomalModel'
getNames(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getNames.ChromosomalModel_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ChromosomalModel">ChromosomalModel</a></code>.
</p>

<hr>
<h2 id='getNames.Explorer'>Gets the names of the input samples</h2><span id='topic+getNames.Explorer'></span><span id='topic+Explorer.getNames'></span><span id='topic+getNames+2CExplorer-method'></span>

<h3>Description</h3>

<p>Gets the names of the input samples for which the explorer is displaying results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Explorer'
getNames(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getNames.Explorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+Explorer">Explorer</a></code>.
</p>

<hr>
<h2 id='getOutputDataSet.AromaTransform'>Gets the transformed data set</h2><span id='topic+getOutputDataSet.AromaTransform'></span><span id='topic+AromaTransform.getOutputDataSet'></span><span id='topic+getOutputDataSet+2CAromaTransform-method'></span>

<h3>Description</h3>

<p>Gets the transformed data set, if processed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaTransform'
getOutputDataSet(this, onMissing=c("dropall", "drop", "NA", "error"), ...,
  incomplete=FALSE, className=NULL, force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getOutputDataSet.AromaTransform_+3A_...">...</code></td>
<td>
<p>Arguments passed to static method <code>byPath()</code> of
the class of the input <code><a href="#topic+AromaMicroarrayDataSet">AromaMicroarrayDataSet</a></code>.</p>
</td></tr>
<tr><td><code id="getOutputDataSet.AromaTransform_+3A_onmissing">onMissing</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying how non-processed files
should be returned.
If <code>"drop"</code>, they are ignored and not part of the returned
data set.
If <code>"dropall"</code>, <code><a href="base.html#topic+NULL">NULL</a></code> is returned unless all files are processed.
If <code>"NA"</code>, they are represented as a &quot;missing&quot; file.
If <code>"error"</code>, they are not accepted and an exception is thrown.
</p>
</td></tr>
<tr><td><code id="getOutputDataSet.AromaTransform_+3A_incomplete">incomplete</code></td>
<td>
<p>[DEPRECATED] If the output data set is incomplete,
then <code><a href="base.html#topic+NULL">NULL</a></code> is returned unless <code>incomplete</code> is <code><a href="base.html#topic+logical">TRUE</a></code>.</p>
</td></tr>
<tr><td><code id="getOutputDataSet.AromaTransform_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, any in-memory cached results are ignored.</p>
</td></tr>
<tr><td><code id="getOutputDataSet.AromaTransform_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AromaMicroarrayDataSet">AromaMicroarrayDataSet</a></code> or <code><a href="base.html#topic+NULL">NULL</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AromaTransform">AromaTransform</a></code>.
</p>

<hr>
<h2 id='getParameters.ParametersInterface'>Gets a list of parameters</h2><span id='topic+getParameters.ParametersInterface'></span><span id='topic+ParametersInterface.getParameters'></span><span id='topic+getParameters+2CParametersInterface-method'></span>

<h3>Description</h3>

<p>Gets a list of parameters associated with the object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ParametersInterface'
getParameters(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getParameters.ParametersInterface_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named <code><a href="base.html#topic+list">list</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ParametersInterface">ParametersInterface</a></code>.
</p>

<hr>
<h2 id='getParametersAsString.ParametersInterface'>Gets the parameters as character</h2><span id='topic+getParametersAsString.ParametersInterface'></span><span id='topic+ParametersInterface.getParametersAsString'></span><span id='topic+getParametersAsString+2CParametersInterface-method'></span>

<h3>Description</h3>

<p>Gets the parameters as character.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ParametersInterface'
getParametersAsString(this, ..., collapse=c(", ", "; "), drop=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getParametersAsString.ParametersInterface_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+getParameters.ParametersInterface">*getParameters</a>()</code>.</p>
</td></tr>
<tr><td><code id="getParametersAsString.ParametersInterface_+3A_collapse">collapse</code></td>
<td>
<p>(optional) A <code><a href="base.html#topic+character">character</a></code> string used to collapse the
individual parameter strings.</p>
</td></tr>
<tr><td><code id="getParametersAsString.ParametersInterface_+3A_drop">drop</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code> and there is only one set of parameters,
then a single <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> is returned, otherwise a <code><a href="base.html#topic+list">list</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> of <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code>s, or a <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ParametersInterface">ParametersInterface</a></code>.
</p>

<hr>
<h2 id='getPath.AromaTransform'>Gets the path of the output directory</h2><span id='topic+getPath.AromaTransform'></span><span id='topic+AromaTransform.getPath'></span><span id='topic+getPath+2CAromaTransform-method'></span>

<h3>Description</h3>

<p>Gets the path of the output directory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaTransform'
getPath(this, create=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getPath.AromaTransform_+3A_create">create</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the path is created, otherwise not.</p>
</td></tr>
<tr><td><code id="getPath.AromaTransform_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Windows Shortcut links are recognized.
</p>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AromaTransform">AromaTransform</a></code>.
</p>

<hr>
<h2 id='getPath.Explorer'>Gets the path of the output directory</h2><span id='topic+getPath.Explorer'></span><span id='topic+Explorer.getPath'></span><span id='topic+getPath+2CExplorer-method'></span>

<h3>Description</h3>

<p>Gets the path of the output directory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Explorer'
getPath(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getPath.Explorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Windows Shortcut links are recognized.
</p>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+Explorer">Explorer</a></code>.
</p>

<hr>
<h2 id='getPlatform.AromaPlatformInterface'>Gets the platform</h2><span id='topic+getPlatform.AromaPlatformInterface'></span><span id='topic+AromaPlatformInterface.getPlatform'></span><span id='topic+getPlatform+2CAromaPlatformInterface-method'></span>

<h3>Description</h3>

<p>Gets the platform label.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaPlatformInterface'
getPlatform(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getPlatform.AromaPlatformInterface_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+getAromaPlatform.AromaPlatformInterface">*getAromaPlatform</a>()</code>.
For more information see <code><a href="#topic+AromaPlatformInterface">AromaPlatformInterface</a></code>.
</p>

<hr>
<h2 id='getRootPath.AromaTransform'>Gets the root path of the output directory</h2><span id='topic+getRootPath.AromaTransform'></span><span id='topic+AromaTransform.getRootPath'></span><span id='topic+getRootPath+2CAromaTransform-method'></span>

<h3>Description</h3>

<p>Gets the root path of the output directory that is returned by <code><a href="#topic+getPath.AromaTransform">*getPath</a>()</code>.
A root path is a directory in the current working directory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaTransform'
getRootPath(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getRootPath.AromaTransform_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+getPath.AromaTransform">*getPath</a>()</code>.
For more information see <code><a href="#topic+AromaTransform">AromaTransform</a></code>.
</p>

<hr>
<h2 id='getRootPath.Explorer'>Gets the root path of the output directory</h2><span id='topic+getRootPath.Explorer'></span><span id='topic+Explorer.getRootPath'></span><span id='topic+getRootPath+2CExplorer-method'></span>

<h3>Description</h3>

<p>Gets the root path of the output directory that is returned by <code><a href="#topic+getPath.Explorer">*getPath</a>()</code>.
A root path is a directory in the current working directory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Explorer'
getRootPath(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getRootPath.Explorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+getPath.Explorer">*getPath</a>()</code>.
For more information see <code><a href="#topic+Explorer">Explorer</a></code>.
</p>

<hr>
<h2 id='getTags.AromaTransform'>Gets the tags of the output data set</h2><span id='topic+getTags.AromaTransform'></span><span id='topic+AromaTransform.getTags'></span><span id='topic+getTags+2CAromaTransform-method'></span>

<h3>Description</h3>

<p>Gets the tags of the output data set, which equals the tags of the input data set plus the tags
of this transformation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaTransform'
getTags(this, collapse=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getTags.AromaTransform_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AromaTransform">AromaTransform</a></code>.
</p>

<hr>
<h2 id='getTags.Explorer'>Gets the tags of the explorer</h2><span id='topic+getTags.Explorer'></span><span id='topic+Explorer.getTags'></span><span id='topic+getTags+2CExplorer-method'></span>

<h3>Description</h3>

<p>Gets the tags of the explorer, which are the tags of the data set plus additional tags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Explorer'
getTags(this, collapse=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getTags.Explorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+Explorer">Explorer</a></code>.
</p>

<hr>
<h2 id='getUnitAnnotationDataFile.AromaPlatformInterface'>Gets a unit annotation data file of a particular class</h2><span id='topic+getUnitAnnotationDataFile.AromaPlatformInterface'></span><span id='topic+AromaPlatformInterface.getUnitAnnotationDataFile'></span><span id='topic+getUnitAnnotationDataFile+2CAromaPlatformInterface-method'></span><span id='topic+AromaPlatformInterface.getUnitNamesFile'></span><span id='topic+getUnitNamesFile.AromaPlatformInterface'></span><span id='topic+getUnitNamesFile+2CAromaPlatformInterface-method'></span><span id='topic+AromaPlatformInterface.getUnitTypesFile'></span><span id='topic+getUnitTypesFile.AromaPlatformInterface'></span><span id='topic+getUnitTypesFile+2CAromaPlatformInterface-method'></span><span id='topic+AromaPlatformInterface.getAromaUgpFile'></span><span id='topic+getAromaUgpFile.AromaPlatformInterface'></span><span id='topic+getAromaUgpFile+2CAromaPlatformInterface-method'></span><span id='topic+AromaPlatformInterface.getAromaUflFile'></span><span id='topic+getAromaUflFile.AromaPlatformInterface'></span><span id='topic+getAromaUflFile+2CAromaPlatformInterface-method'></span>

<h3>Description</h3>

<p>Gets a unit annotation data file of a particular class for this AromaPlatformInterface.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaPlatformInterface'
getUnitAnnotationDataFile(this, ..., className, force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getUnitAnnotationDataFile.AromaPlatformInterface_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>byChipType()</code> for the
class of interest.</p>
</td></tr>
<tr><td><code id="getUnitAnnotationDataFile.AromaPlatformInterface_+3A_classname">className</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying the class of interest.</p>
</td></tr>
<tr><td><code id="getUnitAnnotationDataFile.AromaPlatformInterface_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, any cached results are ignored, otherwise not.</p>
</td></tr>
<tr><td><code id="getUnitAnnotationDataFile.AromaPlatformInterface_+3A_verbose">verbose</code></td>
<td>
<p>The <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> to be used during processing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code><a href="base.html#topic+logical">TRUE</a></code> if compatible and <code><a href="base.html#topic+logical">FALSE</a></code> otherwise.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+getPlatform.AromaPlatformInterface">*getPlatform</a>()</code>.
For more information see <code><a href="#topic+AromaPlatformInterface">AromaPlatformInterface</a></code>.
</p>

<hr>
<h2 id='GladModel'>The GladModel class</h2><span id='topic+GladModel'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class GladModel</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+ChromosomalModel">ChromosomalModel</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+CopyNumberChromosomalModel">CopyNumberChromosomalModel</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+CopyNumberSegmentationModel">CopyNumberSegmentationModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code>GladModel</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>GladModel</b><br />
extends <em><a href="#topic+CopyNumberSegmentationModel">CopyNumberSegmentationModel</a></em><br />
</p>
<p>This class represents the Gain and Loss Analysis of DNA regions
(GLAD) model [1].
This class can model chip-effect estimates obtained from multiple
chip types, and not all samples have to be available on all chip types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GladModel(cesTuple=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GladModel_+3A_cestuple">cesTuple</code></td>
<td>
<p>A <code><a href="#topic+CopyNumberDataSetTuple">CopyNumberDataSetTuple</a></code>.</p>
</td></tr>
<tr><td><code id="GladModel_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+CopyNumberSegmentationModel">CopyNumberSegmentationModel</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Data from multiple chip types are combined &quot;as is&quot;.  This is based
on the assumption that the relative chip effect estimates are
non-biased (or at the equally biased across chip types).
Note that in GLAD there is no way to down weight certain data points,
which is why we can control for differences in variance across
chip types.
</p>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>writeRegions</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from CopyNumberSegmentationModel</b>:<br />
fit, getAsteriskTags, getFitFunction, getFullNames, getRegions, getTags, plot, plotCopyNumberRegionLayers, writeRegions
</p>
<p><b>Methods inherited from CopyNumberChromosomalModel</b>:<br />
as.character, calculateChromosomeStatistics, calculateRatios, estimateSds, extractRawCopyNumbers, fit, getChromosomeLength, getDataFileMatrix, getMaxNAFraction, getNames, getOptionalArguments, getPairedNames, getRefSetTuple, getReference, getReferenceSetTuple, isPaired, newPlot, plotAxesLayers, plotChromosomesLayers, plotCytobandLayers, plotFitLayers, plotGridHorizontalLayers, plotRawCopyNumbers, plotSampleLayers, setReference
</p>
<p><b>Methods inherited from ChromosomalModel</b>:<br />
as.character, fit, getAlias, getAromaGenomeTextFile, getAsteriskTags, getChipType, getChipTypes, getChromosomes, getFullName, getFullNames, getGenome, getGenomeData, getGenomeFile, getListOfAromaUgpFiles, getName, getNames, getParentPath, getPath, getReportPath, getRootPath, getSetTuple, getSets, getTags, indexOf, nbrOfArrays, nbrOfChipTypes, setChromosomes, setGenome
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Benchmarking</h3>

<p>In high-density copy numbers analysis, the most time consuming step
is fitting the GLAD model.  The complexity of the model grows
more than linearly (squared? exponentially?) with the number of data
points in the chromosome and sample being fitted.  This is why it
take much more than twice the time to fit two chip types together
than separately.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] Hupe P et al. <em>Analysis of array CGH data: from signal ratio to
gain and loss of DNA regions</em>. Bioinformatics, 2004, 20, 3413-3422.<br />
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CopyNumberSegmentationModel">CopyNumberSegmentationModel</a></code>.
</p>

<hr>
<h2 id='HaarSegModel'>The HaarSegModel class</h2><span id='topic+HaarSegModel'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class HaarSegModel</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+ChromosomalModel">ChromosomalModel</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+CopyNumberChromosomalModel">CopyNumberChromosomalModel</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+CopyNumberSegmentationModel">CopyNumberSegmentationModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code>HaarSegModel</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>HaarSegModel</b><br />
extends <em><a href="#topic+CopyNumberSegmentationModel">CopyNumberSegmentationModel</a></em><br />
</p>
<p>This class represents the Haar wavelet-based segmentation (HaarSeg)
model [1].
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HaarSegModel(cesTuple=NULL, ..., breaksFdrQ=1e-04)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HaarSegModel_+3A_cestuple">cesTuple</code></td>
<td>
<p>A <code><a href="#topic+CopyNumberDataSetTuple">CopyNumberDataSetTuple</a></code>.</p>
</td></tr>
<tr><td><code id="HaarSegModel_+3A_breaksfdrq">breaksFdrQ</code></td>
<td>
<p>Default tuning parameters specific to the HaarSeg
algorithm.</p>
</td></tr>
<tr><td><code id="HaarSegModel_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+CopyNumberSegmentationModel">CopyNumberSegmentationModel</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from CopyNumberSegmentationModel</b>:<br />
fit, getAsteriskTags, getFitFunction, getFullNames, getRegions, getTags, plot, plotCopyNumberRegionLayers, writeRegions
</p>
<p><b>Methods inherited from CopyNumberChromosomalModel</b>:<br />
as.character, calculateChromosomeStatistics, calculateRatios, estimateSds, extractRawCopyNumbers, fit, getChromosomeLength, getDataFileMatrix, getMaxNAFraction, getNames, getOptionalArguments, getPairedNames, getRefSetTuple, getReference, getReferenceSetTuple, isPaired, newPlot, plotAxesLayers, plotChromosomesLayers, plotCytobandLayers, plotFitLayers, plotGridHorizontalLayers, plotRawCopyNumbers, plotSampleLayers, setReference
</p>
<p><b>Methods inherited from ChromosomalModel</b>:<br />
as.character, fit, getAlias, getAromaGenomeTextFile, getAsteriskTags, getChipType, getChipTypes, getChromosomes, getFullName, getFullNames, getGenome, getGenomeData, getGenomeFile, getListOfAromaUgpFiles, getName, getNames, getParentPath, getPath, getReportPath, getRootPath, getSetTuple, getSets, getTags, indexOf, nbrOfArrays, nbrOfChipTypes, setChromosomes, setGenome
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] Ben-Yaacov E. and Eldar YC. <em>A fast and flexible method for the segmentation of aCGH data</em>, Bioinformatics, 2008.
<a href="https://www.weizmann.ac.il/math/yonina/software-hardware/software/haarseg-fast-and-flexible-microarray-segmentation">https://www.weizmann.ac.il/math/yonina/software-hardware/software/haarseg-fast-and-flexible-microarray-segmentation</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CopyNumberSegmentationModel">CopyNumberSegmentationModel</a></code>.
</p>

<hr>
<h2 id='indexOf.UnitNamesFile'>Gets the indices of units by their names</h2><span id='topic+indexOf.UnitNamesFile'></span><span id='topic+UnitNamesFile.indexOf'></span><span id='topic+indexOf+2CUnitNamesFile-method'></span>

<h3>Description</h3>

<p>Gets the indices of units by their names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'UnitNamesFile'
indexOf(this, pattern=NULL, names=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="indexOf.UnitNamesFile_+3A_pattern">pattern</code></td>
<td>
<p>A pattern to be used for identifying unit names of
interest.  If <code><a href="base.html#topic+NULL">NULL</a></code>, no regular expression matching is done.</p>
</td></tr>
<tr><td><code id="indexOf.UnitNamesFile_+3A_names">names</code></td>
<td>
<p>Names to be match exactly to the unit names.</p>
</td></tr>
<tr><td><code id="indexOf.UnitNamesFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+vector">vector</a></code> of <code><a href="base.html#topic+integer">integer</a></code>s in [1,N] where N is the number of units
in the underlying annotation chip type file.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+getUnitNames.UnitNamesFile">*getUnitNames</a>()</code>.
For more information see <code><a href="#topic+UnitNamesFile">UnitNamesFile</a></code>.
</p>

<hr>
<h2 id='isCompatibleWith.AromaPlatformInterface'>Checks if a particular unit annotation data file is compatible</h2><span id='topic+isCompatibleWith.AromaPlatformInterface'></span><span id='topic+AromaPlatformInterface.isCompatibleWith'></span><span id='topic+isCompatibleWith+2CAromaPlatformInterface-method'></span>

<h3>Description</h3>

<p>Checks if a particular unit annotation data file is compatible with this AromaPlatformInterface class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaPlatformInterface'
isCompatibleWith(this, udf, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="isCompatibleWith.AromaPlatformInterface_+3A_udf">udf</code></td>
<td>
<p>An unit annotation data file.</p>
</td></tr>
<tr><td><code id="isCompatibleWith.AromaPlatformInterface_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code><a href="base.html#topic+logical">TRUE</a></code> if compatible and <code><a href="base.html#topic+logical">FALSE</a></code> otherwise.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+getPlatform.AromaPlatformInterface">*getPlatform</a>()</code>.
For more information see <code><a href="#topic+AromaPlatformInterface">AromaPlatformInterface</a></code>.
</p>

<hr>
<h2 id='isDone.AromaTransform'>Checks if the data set is processed or not</h2><span id='topic+isDone.AromaTransform'></span><span id='topic+AromaTransform.isDone'></span><span id='topic+isDone+2CAromaTransform-method'></span>

<h3>Description</h3>

<p>Checks if the data set is processed or not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaTransform'
isDone(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="isDone.AromaTransform_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+findFilesTodo.AromaTransform">*findFilesTodo</a>()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code><a href="base.html#topic+logical">TRUE</a></code> if the data set is processed, otherwise <code><a href="base.html#topic+logical">FALSE</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AromaTransform">AromaTransform</a></code>.
</p>

<hr>
<h2 id='listFiles.AromaRepository'>Retrieves the files available on the repository under a particular path</h2><span id='topic+listFiles.AromaRepository'></span><span id='topic+AromaRepository.listFiles'></span><span id='topic+listFiles+2CAromaRepository-method'></span>

<h3>Description</h3>

<p>Retrieves the files available on the repository under a particular path.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaRepository'
listFiles(this, path=NULL, full=TRUE, orderBy=c("name", "time"), ..., force=FALSE,
  verbose=getVerbose(this))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="listFiles.AromaRepository_+3A_path">path</code></td>
<td>
<p>The relative path to be listed.</p>
</td></tr>
<tr><td><code id="listFiles.AromaRepository_+3A_orderby">orderBy</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> specifying how the returns files should be ordered.</p>
</td></tr>
<tr><td><code id="listFiles.AromaRepository_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="R.utils.html#topic+downloadFile">downloadFile</a></code>.</p>
</td></tr>
<tr><td><code id="listFiles.AromaRepository_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, cached results are redownloaded, otherwise not.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this method makes strong assumptions of the format of the
downloaded HTML index file.
</p>


<h3>Value</h3>

<p>Returns the relative pathnames of the files available.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AromaRepository">AromaRepository</a></code>.
</p>

<hr>
<h2 id='matrixBlockPolish.matrix'>Applies a polishing function to blocks of rows and columns repeatedly</h2><span id='topic+matrixBlockPolish.matrix'></span>

<h3>Description</h3>

<p>Applies a polishing function to blocks of rows and columns repeatedly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'matrix'
matrixBlockPolish(z, x=NULL, blockSizes=c(1, 1), FUN, ..., tol=0.01, maxIter=10,
  returnEffects=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="matrixBlockPolish.matrix_+3A_z">z</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> KxN <code><a href="base.html#topic+matrix">matrix</a></code>.</p>
</td></tr>
<tr><td><code id="matrixBlockPolish.matrix_+3A_x">x</code></td>
<td>
<p>A optional KxNx2 <code><a href="base.html#topic+array">array</a></code> (or KxN <code><a href="base.html#topic+matrix">matrix</a></code>).</p>
</td></tr>
<tr><td><code id="matrixBlockPolish.matrix_+3A_blocksizes">blockSizes</code></td>
<td>
<p>A positive <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length two.</p>
</td></tr>
<tr><td><code id="matrixBlockPolish.matrix_+3A_fun">FUN</code></td>
<td>
<p>A <code><a href="base.html#topic+function">function</a></code> taking <code><a href="base.html#topic+numeric">numeric</a></code> arguments <code>z</code> and
<code>x</code> and returns a <code><a href="base.html#topic+numeric">numeric</a></code> object with either a scalar
or the same number of elements as in <code>z</code>.</p>
</td></tr>
<tr><td><code id="matrixBlockPolish.matrix_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the <code>FUN</code> <code><a href="base.html#topic+function">function</a></code>.</p>
</td></tr>
<tr><td><code id="matrixBlockPolish.matrix_+3A_tol">tol</code></td>
<td>
<p>A positive threshold specifying when the algorithm has
converged.</p>
</td></tr>
<tr><td><code id="matrixBlockPolish.matrix_+3A_maxiter">maxIter</code></td>
<td>
<p>The maximum number of iterations.</p>
</td></tr>
<tr><td><code id="matrixBlockPolish.matrix_+3A_returneffects">returnEffects</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the row and column effects are returned,
otherwise not.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named <code><a href="base.html#topic+list">list</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+medpolish">medpolish</a></code>.
<code><a href="aroma.light.html#topic+medianPolish.matrix">medianPolish</a>()</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Example 1: median polish
# From example(medpolish) in the 'stats' package
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

# Deaths from sport parachuting;  from ABC of EDA, p.224:
deaths &lt;- matrix(c(14,15,14, 7,4,7, 8,2,10, 15,9,10, 0,2,0), ncol=3, byrow=TRUE)
rownames(deaths) &lt;- c("1-24", "25-74", "75-199", "200++", "NA")
colnames(deaths) &lt;- 1973:1975
print(deaths)

fit1 &lt;- medpolish(deaths, trace=FALSE)
r1 &lt;- residuals(fit1)
fit2 &lt;- matrixBlockPolish(deaths, FUN=function(y, x, ...) median(y, ...))
r2 &lt;- residuals(fit2)
stopifnot(all.equal(r1,r2))


# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Example 2: smooth spline polish ("spatial smoothing")
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# From example(image) in 'graphics' package
x &lt;- y &lt;- seq(-4*pi, 4*pi, len=27)
r &lt;- sqrt(outer(x^2, y^2, FUN="+"))
z &lt;- cos(r^2) * exp(-r/6)

fit &lt;- matrixBlockPolish(z, FUN=function(z, x, ...) median(z, ...),
                                                      returnEffects=TRUE)
r1 &lt;- residuals(fit)

fit &lt;- matrixBlockPolish(z, FUN=function(z, x, ...) {
  fit &lt;- smooth.spline(x=x, y=z, ...)
  predict(fit, x=x)$y
}, spar=0.25)
r2 &lt;- residuals(fit)
print(range(r2))
image(r2)
</code></pre>

<hr>
<h2 id='mergeBoxplotStats.list'>Merges a list of boxplot.stats() elements</h2><span id='topic+mergeBoxplotStats.list'></span>

<h3>Description</h3>

<p>Merges a list of boxplot.stats() elements.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'list'
mergeBoxplotStats(stats, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mergeBoxplotStats.list_+3A_stats">stats</code></td>
<td>
<p>A <code><a href="base.html#topic+list">list</a></code> of elements each in a format returned by
<code><a href="grDevices.html#topic+boxplot.stats">boxplot.stats</a></code>.</p>
</td></tr>
<tr><td><code id="mergeBoxplotStats.list_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> structure in a format that is returned by
<code><a href="graphics.html#topic+boxplot">boxplot</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("stats")

x &lt;- matrix(rnorm(1000), ncol=5)
x &lt;- as.data.frame(x)

stats0 &lt;- boxplot(x, plot=FALSE)
stats1 &lt;- lapply(x, FUN=boxplot.stats)
stats1b &lt;- mergeBoxplotStats(stats1)
stopifnot(all.equal(stats0, stats1b))

bxp(stats1b)

</code></pre>

<hr>
<h2 id='nbrOfArrays.ChromosomalModel'>Gets the number of arrays</h2><span id='topic+nbrOfArrays.ChromosomalModel'></span><span id='topic+ChromosomalModel.nbrOfArrays'></span><span id='topic+nbrOfArrays+2CChromosomalModel-method'></span>

<h3>Description</h3>

<p>Gets the number of arrays used in the model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ChromosomalModel'
nbrOfArrays(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nbrOfArrays.ChromosomalModel_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="base.html#topic+integer">integer</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ChromosomalModel">ChromosomalModel</a></code>.
</p>

<hr>
<h2 id='nbrOfArrays.Explorer'>Gets the total number of arrays</h2><span id='topic+nbrOfArrays.Explorer'></span><span id='topic+Explorer.nbrOfArrays'></span><span id='topic+nbrOfArrays+2CExplorer-method'></span>

<h3>Description</h3>

<p>Gets the total number of arrays considered by the explorer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Explorer'
nbrOfArrays(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nbrOfArrays.Explorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="base.html#topic+integer">integer</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+Explorer">Explorer</a></code>.
</p>

<hr>
<h2 id='nbrOfChipTypes.AromaMicroarrayDataSetTuple'>Gets the number of chip types</h2><span id='topic+nbrOfChipTypes.AromaMicroarrayDataSetTuple'></span><span id='topic+AromaMicroarrayDataSetTuple.nbrOfChipTypes'></span><span id='topic+nbrOfChipTypes+2CAromaMicroarrayDataSetTuple-method'></span>

<h3>Description</h3>

<p>Gets the number of chip types used in the model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaMicroarrayDataSetTuple'
nbrOfChipTypes(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nbrOfChipTypes.AromaMicroarrayDataSetTuple_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="base.html#topic+integer">integer</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+getChipTypes.AromaMicroarrayDataSetTuple">*getChipTypes</a>()</code>.
For more information see <code><a href="#topic+AromaMicroarrayDataSetTuple">AromaMicroarrayDataSetTuple</a></code>.
</p>

<hr>
<h2 id='nbrOfChipTypes.ChromosomalModel'>Gets the number of chip types</h2><span id='topic+nbrOfChipTypes.ChromosomalModel'></span><span id='topic+ChromosomalModel.nbrOfChipTypes'></span><span id='topic+nbrOfChipTypes+2CChromosomalModel-method'></span>

<h3>Description</h3>

<p>Gets the number of chip types used in the model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ChromosomalModel'
nbrOfChipTypes(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nbrOfChipTypes.ChromosomalModel_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="base.html#topic+integer">integer</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ChromosomalModel">ChromosomalModel</a></code>.
</p>

<hr>
<h2 id='Non-documented+20objects'>Non-documented objects</h2><span id='topic+Non-documented+20objects'></span><span id='topic+.Machine'></span><span id='topic++23'></span><span id='topic+DEFUNCT'></span><span id='topic++2F'></span><span id='topic+DEPRECATED'></span><span id='topic+nbrOfArrays'></span><span id='topic+MISC.'></span><span id='topic+applyTo'></span><span id='topic+applyTo.SampleAnnotationFile'></span><span id='topic++5B.AromaTabularBinaryFile'></span><span id='topic++5B+5B.AromaTabularBinaryFile'></span><span id='topic++5B+3C-.AromaTabularBinaryFile'></span><span id='topic+addBy'></span><span id='topic+addBy.RawGenomicSignals'></span><span id='topic+addIncludes'></span><span id='topic+addIncludes.Explorer'></span><span id='topic+addIndexFile'></span><span id='topic+addIndexFile.ChromosomeExplorer'></span><span id='topic+addIndexFile.Explorer'></span><span id='topic+addLocusFields'></span><span id='topic+addLocusFields.RawGenomicSignals'></span><span id='topic+allocate'></span><span id='topic+allocate.AromaCellPositionFile'></span><span id='topic+allocate.AromaCellSequenceFile'></span><span id='topic+allocate.AromaMicroarrayTabularBinaryFile'></span><span id='topic+allocate.AromaUflFile'></span><span id='topic+allocate.AromaUgpFile'></span><span id='topic+allocate.AromaUnitCallFile'></span><span id='topic+allocate.AromaUnitChromosomeTabularBinaryFile'></span><span id='topic+allocate.AromaUnitGenotypeCallFile'></span><span id='topic+allocate.AromaUnitSignalBinaryFile'></span><span id='topic+allocate.AromaUnitTypesFile'></span><span id='topic+allocateFromUnitAnnotationDataFile'></span><span id='topic+allocateFromUnitAnnotationDataFile.AromaUnitSignalBinaryFile'></span><span id='topic+allocateFromUnitAnnotationDataFile.AromaUnitTabularBinaryFile'></span><span id='topic+allocateFromUnitNamesFile'></span><span id='topic+allocateFromUnitNamesFile.AromaUnitSignalBinaryFile'></span><span id='topic+allocateFromUnitNamesFile.AromaUnitTabularBinaryFile'></span><span id='topic+append.RawGenomicSignals'></span><span id='topic+applyBinaryOperator'></span><span id='topic+applyBinaryOperator.RawGenomicSignals'></span><span id='topic+applyRegions'></span><span id='topic+applyRegions.CopyNumberOutliers'></span><span id='topic+applyRows'></span><span id='topic+applyRows.CopyNumberRegions'></span><span id='topic+AromaCellSequenceFile'></span><span id='topic+AromaCore'></span><span id='topic+AromaPackage'></span><span id='topic+aromaSettings'></span><span id='topic+AromaSettings'></span><span id='topic+AromaUflFile'></span><span id='topic+AromaUgpFile'></span><span id='topic+AromaUnitChromosomeTabularBinaryFile'></span><span id='topic+AromaUnitTotalCnBinaryFileList'></span><span id='topic+AromaUnitTotalCnBinarySetTuple'></span><span id='topic+as.AromaMicroarrayDataSetList'></span><span id='topic+as.AromaMicroarrayDataSetList.AromaMicroarrayDataSet'></span><span id='topic+as.AromaMicroarrayDataSetTuple'></span><span id='topic+as.AromaMicroarrayDataSetTuple.AromaMicroarrayDataSet'></span><span id='topic+as.AromaMicroarrayDataSetTuple.AromaMicroarrayDataSetTuple'></span><span id='topic+as.AromaUnitTotalCnBinarySetTuple'></span><span id='topic+as.AromaUnitTotalCnBinarySetTuple.AromaUnitTotalCnBinarySet'></span><span id='topic+as.AromaUnitTotalCnBinarySetTuple.AromaUnitTotalCnBinarySetTuple'></span><span id='topic+as.character.AromaMicroarrayDataSetTuple'></span><span id='topic+as.character.AromaMicroarrayTabularBinaryFile'></span><span id='topic+as.character.AromaTabularBinaryFile'></span><span id='topic+as.character.AromaTransform'></span><span id='topic+as.character.AromaUnitSignalBinaryFile'></span><span id='topic+as.character.ChromosomalModel'></span><span id='topic+as.character.ChromosomeExplorer'></span><span id='topic+as.character.CopyNumberChromosomalModel'></span><span id='topic+as.character.CopyNumberOutliers'></span><span id='topic+as.character.CopyNumberRegions'></span><span id='topic+as.character.Explorer'></span><span id='topic+as.character.RawGenomicSignals'></span><span id='topic+as.character.TextUnitNamesFile'></span><span id='topic+as.CopyNumberDataFile'></span><span id='topic+as.CopyNumberDataFile.CopyNumberDataFile'></span><span id='topic+as.CopyNumberDataSet'></span><span id='topic+as.CopyNumberDataSet.CopyNumberDataSet'></span><span id='topic+as.CopyNumberDataSetTuple'></span><span id='topic+as.CopyNumberDataSetTuple.AromaUnitTotalCnBinarySet'></span><span id='topic+as.CopyNumberDataSetTuple.CopyNumberDataSetTuple'></span><span id='topic+as.CopyNumberDataSetTuple.list'></span><span id='topic+as.data.frame.CopyNumberOutliers'></span><span id='topic+as.data.frame.CopyNumberRegions'></span><span id='topic+as.data.frame.RawCopyNumbers'></span><span id='topic+as.data.frame.RawGenomicSignals'></span><span id='topic+as.data.frame.SegmentedGenomicSignalsInterface'></span><span id='topic+as.GrayscaleImage'></span><span id='topic+as.TrueColorImage'></span><span id='topic+as.TrueColorImage.Image'></span><span id='topic+as.TrueColorImage.matrix'></span><span id='topic+asMatrixOfFiles'></span><span id='topic+asMatrixOfFiles.AromaMicroarrayDataSetTuple'></span><span id='topic+asThis'></span><span id='topic+asThis.BasicObject'></span><span id='topic+asThis.Object'></span><span id='topic+backtransformGenotypeCone'></span><span id='topic+backtransformGenotypeCone.matrix'></span><span id='topic+backtransformMultiDimensionalCone'></span><span id='topic+backtransformMultiDimensionalCone.matrix'></span><span id='topic+barSequence'></span><span id='topic+barSequence.ProbePositionEffects'></span><span id='topic+binnedSmoothing.RawGenomicSignals'></span><span id='topic+binnedSmoothingByState'></span><span id='topic+binnedSmoothingByState.SegmentedGenomicSignalsInterface'></span><span id='topic+binnedSums'></span><span id='topic+binnedSums.RawSequenceReads'></span><span id='topic+binScatter'></span><span id='topic+binScatter.matrix'></span><span id='topic+byChipType'></span><span id='topic+byChipType.AromaCellTabularBinaryFile'></span><span id='topic+byChipType.AromaMicroarrayTabularBinaryFile'></span><span id='topic+byChipType.AromaUnitTabularBinaryFile'></span><span id='topic+byChipType.UnitAnnotationDataFile'></span><span id='topic+byName.AromaPlatform'></span><span id='topic+byName.AromaUnitFracBCnBinarySet'></span><span id='topic+byName.AromaUnitGenotypeCallSet'></span><span id='topic+byName.AromaUnitSignalBinarySet'></span><span id='topic+byName.AromaUnitTotalCnBinarySet'></span><span id='topic+byPath.AromaMicroarrayDataSetTuple'></span><span id='topic+byPath.AromaUnitCallSet'></span><span id='topic+byPath.AromaUnitGenotypeCallSet'></span><span id='topic+calcMargins'></span><span id='topic+calcMargins.matrix'></span><span id='topic+calculateAverageColumnAcrossFiles'></span><span id='topic+calculateAverageColumnAcrossFiles.GenericTabularFileSet'></span><span id='topic+calculateChromosomeStatistics'></span><span id='topic+calculateChromosomeStatistics.CopyNumberChromosomalModel'></span><span id='topic+calculateRatios'></span><span id='topic+calculateRatios.CopyNumberChromosomalModel'></span><span id='topic+cnRange'></span><span id='topic+cnRange.RawCopyNumbers'></span><span id='topic+colGaussianSmoothing'></span><span id='topic+colGaussianSmoothing.matrix'></span><span id='topic+colMeans'></span><span id='topic+colMeans.AromaTabularBinaryFile'></span><span id='topic+colMeans.default'></span><span id='topic+colorize'></span><span id='topic+colorize.Image'></span><span id='topic+colStats'></span><span id='topic+colStats.AromaTabularBinaryFile'></span><span id='topic+colSums'></span><span id='topic+colSums.AromaTabularBinaryFile'></span><span id='topic+colSums.default'></span><span id='topic+convertTable'></span><span id='topic+convertTable.default'></span><span id='topic+CopyNumberDataFile'></span><span id='topic+CopyNumberDataSet'></span><span id='topic+CopyNumberDataSetTuple'></span><span id='topic+CopyNumberOutliers'></span><span id='topic+CopyNumberRegions'></span><span id='topic+countBases'></span><span id='topic+countBases.AromaCellSequenceFile'></span><span id='topic+countBasesInternal'></span><span id='topic+countBasesInternal.AromaCellSequenceFile'></span><span id='topic+dimnames+3C-.AromaTabularBinaryFile'></span><span id='topic+display'></span><span id='topic+display.ChromosomeExplorer'></span><span id='topic+display.Image'></span><span id='topic+divideBy'></span><span id='topic+divideBy.RawGenomicSignals'></span><span id='topic+drawCnRegions'></span><span id='topic+drawCnRegions.DNAcopy'></span><span id='topic+drawCnRegions.HaarSeg'></span><span id='topic+drawCnRegions.profileCGH'></span><span id='topic+drawCytoband'></span><span id='topic+drawCytoband.ChromosomalModel'></span><span id='topic+drawCytoband.profileCGH'></span><span id='topic+drawCytoband2'></span><span id='topic+drawCytoband2.default'></span><span id='topic+drawExtraAnnotations'></span><span id='topic+drawExtraAnnotations.default'></span><span id='topic+drawExtraAnnotations.profileCGH'></span><span id='topic+drawLevels'></span><span id='topic+drawLevels.CopyNumberOutliers'></span><span id='topic+drawLevels.CopyNumberRegions'></span><span id='topic+estimateSds'></span><span id='topic+estimateStandardDeviation'></span><span id='topic+exportFracBDiffSet'></span><span id='topic+exportFracBDiffSet.AromaUnitFracBCnBinarySet'></span><span id='topic+exportTotalCnRatioSet'></span><span id='topic+exportTotalCnRatioSet.AromaUnitTotalCnBinarySet'></span><span id='topic+extractByChromosome'></span><span id='topic+extractByChromosome.AromaUnitChromosomeTabularBinaryFile'></span><span id='topic+extractCallArray'></span><span id='topic+extractCallArray.AromaUnitCallFile'></span><span id='topic+extractCallArray.AromaUnitCallSet'></span><span id='topic+extractCalls'></span><span id='topic+extractCalls.AromaUnitCallFile'></span><span id='topic+extractCalls.AromaUnitCallSet'></span><span id='topic+extractCNRs'></span><span id='topic+extractCNRs.default'></span><span id='topic+extractCopyNumberOutliers'></span><span id='topic+extractCopyNumberOutliers.DNAcopy'></span><span id='topic+extractCopyNumberOutliers.profileCGH'></span><span id='topic+extractCopyNumberRegions'></span><span id='topic+extractCopyNumberRegions.default'></span><span id='topic+extractCopyNumberRegions.DNAcopy'></span><span id='topic+extractCopyNumberRegions.HaarSeg'></span><span id='topic+extractCopyNumberRegions.profileCGH'></span><span id='topic+extractDataForSegmentation'></span><span id='topic+extractDataForSegmentation.RawGenomicSignals'></span><span id='topic+extractGenotypeMatrix'></span><span id='topic+extractGenotypeMatrix.AromaUnitCallSet'></span><span id='topic+extractGenotypeMatrix.AromaUnitGenotypeCallFile'></span><span id='topic+extractGenotypes'></span><span id='topic+extractGenotypes.AromaUnitGenotypeCallFile'></span><span id='topic+extractGenotypes.AromaUnitGenotypeCallSet'></span><span id='topic+extractIGV'></span><span id='topic+extractIGV.CopyNumberRegions'></span><span id='topic+extractMatrix.AromaUnitCallFile'></span><span id='topic+extractMatrix.AromaUnitSignalBinaryFile'></span><span id='topic+extractMergedRawCopyNumbers'></span><span id='topic+extractMergedRawCopyNumbers.AromaUnitTotalCnBinaryFileList'></span><span id='topic+extractOld'></span><span id='topic+extractOld.AromaMicroarrayDataSetTuple'></span><span id='topic+extractRawAlleleBFractions'></span><span id='topic+extractRawAlleleBFractions.AromaUnitFracBCnBinaryFile'></span><span id='topic+extractRawAlleleBFractions.default'></span><span id='topic+extractRawCNs'></span><span id='topic+extractRawCNs.default'></span><span id='topic+extractRawCopyNumbers'></span><span id='topic+extractRawCopyNumbers.AromaUnitTotalCnBinaryFile'></span><span id='topic+extractRawCopyNumbers.AromaUnitTotalCnBinaryFileList'></span><span id='topic+extractRawCopyNumbers.default'></span><span id='topic+extractRawCopyNumbers.DNAcopy'></span><span id='topic+extractRawCopyNumbers.HaarSeg'></span><span id='topic+extractRawCopyNumbers.profileCGH'></span><span id='topic+extractRawCopyNumbers.RawCopyNumbers'></span><span id='topic+extractRawCopyNumbers.RawSequenceReads'></span><span id='topic+extractRawGenomicSignals'></span><span id='topic+extractRawGenomicSignals.AromaUnitSignalBinaryFile'></span><span id='topic+extractRawGenomicSignals.AromaUnitTotalCnBinaryFileList'></span><span id='topic+extractRawGenomicSignals.default'></span><span id='topic+extractRawMirroredAlleleBFractions'></span><span id='topic+extractRawMirroredAlleleBFractions.default'></span><span id='topic+extractRawMirroredAlleleBFractions.RawAlleleBFractions'></span><span id='topic+extractRegion'></span><span id='topic+extractRegion.RawGenomicSignals'></span><span id='topic+extractSubset'></span><span id='topic+extractSubset.RawGenomicSignals'></span><span id='topic+extractSubsetByState'></span><span id='topic+extractSubsetByState.SegmentedGenomicSignalsInterface'></span><span id='topic+findByChipType'></span><span id='topic+findByChipType.AromaMicroarrayTabularBinaryFile'></span><span id='topic+findByChipType.TextUnitNamesFile'></span><span id='topic+findByName.AromaUnitCallSet'></span><span id='topic+findByName.AromaUnitSignalBinarySet'></span><span id='topic+findPngDevice'></span><span id='topic+findPngDevice.default'></span><span id='topic+findSAFs'></span><span id='topic+findSAFs.SampleAnnotationSet'></span><span id='topic+findUnitNamesFile'></span><span id='topic+findUnitNamesFile.AromaPlatform'></span><span id='topic+findUnitsTodo'></span><span id='topic+findUnitsTodo.AromaUnitCallFile'></span><span id='topic+findUnitsTodo.AromaUnitCallSet'></span><span id='topic+findUnitTypesFile'></span><span id='topic+findUnitTypesFile.AromaPlatform'></span><span id='topic+fit'></span><span id='topic+fit.ChromosomalModel'></span><span id='topic+fit2d'></span><span id='topic+fit2d.matrix'></span><span id='topic+fitGenotypeCone'></span><span id='topic+fitGenotypeConeByExpectile'></span><span id='topic+fitGenotypeConeByExpectile.matrix'></span><span id='topic+fitGenotypeConeBySfit'></span><span id='topic+fitMultiDimensionalCone'></span><span id='topic+fitProbePositionEffects'></span><span id='topic+fitProbePositionEffects.numeric'></span><span id='topic+fitSplineBlockPolish'></span><span id='topic+fitSplineBlockPolish.matrix'></span><span id='topic+fitWHLAPLM'></span><span id='topic+fitWHLAPLM.matrix'></span><span id='topic+fitWHRCModel'></span><span id='topic+fitWHRCModel.matrix'></span><span id='topic+fitWLAPLM'></span><span id='topic+fitWLAPLM.matrix'></span><span id='topic+fitWRCModel'></span><span id='topic+fitWRCModel.matrix'></span><span id='topic+fitWRMA'></span><span id='topic+fitWRMA.matrix'></span><span id='topic+fixSearchPath'></span><span id='topic+fixSearchPath.AromaCore'></span><span id='topic+fixSearchPath.AromaPackage'></span><span id='topic+fixSearchPathInternal'></span><span id='topic+fixSearchPathInternal.AromaPackage'></span><span id='topic+fromFile.AromaUnitSignalBinaryFile'></span><span id='topic+fromPath'></span><span id='topic+fromPath.SampleAnnotationFile'></span><span id='topic+fromPath.SampleAnnotationSet'></span><span id='topic+gaussianSmoothing'></span><span id='topic+gaussianSmoothing.matrix'></span><span id='topic+gaussianSmoothing.numeric'></span><span id='topic+gaussianSmoothing.RawGenomicSignals'></span><span id='topic+gaussKernel'></span><span id='topic+getAlias.ChromosomalModel'></span><span id='topic+getAM'></span><span id='topic+getAM.AromaUnitTotalCnBinaryFile'></span><span id='topic+getAromaPlatform'></span><span id='topic+getAromaUflFile'></span><span id='topic+getAromaUflFile.UnitAnnotationDataFile'></span><span id='topic+getAromaUgpFile'></span><span id='topic+getAromaUgpFile.AromaPlatform'></span><span id='topic+getAromaUgpFile.AromaUnitSignalBinaryFile'></span><span id='topic+getAromaUgpFile.AromaUnitSignalBinarySet'></span><span id='topic+getAromaUgpFile.UnitAnnotationDataFile'></span><span id='topic+getArrays'></span><span id='topic+getArrays.AromaMicroarrayDataSetTuple'></span><span id='topic+getArrays.ChromosomalModel'></span><span id='topic+getArrays.Explorer'></span><span id='topic+getArraysOfInput'></span><span id='topic+getArraysOfInput.ChromosomeExplorer'></span><span id='topic+getArraysOfInput.Explorer'></span><span id='topic+getArrayTuple'></span><span id='topic+getArrayTuple.AromaMicroarrayDataSetTuple'></span><span id='topic+getAsteriskTags.AromaMicroarrayDataSetTuple'></span><span id='topic+getAsteriskTags.AromaTransform'></span><span id='topic+getAsteriskTags.ChromosomalModel'></span><span id='topic+getAsteriskTags.CopyNumberSegmentationModel'></span><span id='topic+getAsteriskTags.Explorer'></span><span id='topic+getAsteriskTags.HaarSegModel'></span><span id='topic+getAsteriskTags.RawCopyNumberModel'></span><span id='topic+getAttributeXY'></span><span id='topic+getAttributeXY.AromaMicroarrayDataFile'></span><span id='topic+getAverageFile'></span><span id='topic+getAverageFile.AromaMicroarrayDataSet'></span><span id='topic+getAverageFile.AromaUnitTotalCnBinarySet'></span><span id='topic+getBytesPerColumn'></span><span id='topic+getBytesPerColumn.AromaTabularBinaryFile'></span><span id='topic+getChipEffectFiles'></span><span id='topic+getChipEffectFiles.ChromosomalModel'></span><span id='topic+getChipType'></span><span id='topic+getChipType.AromaMicroarrayDataFile'></span><span id='topic+getChipType.AromaMicroarrayDataSet'></span><span id='topic+getChipType.AromaMicroarrayTabularBinaryFile'></span><span id='topic+getChipType.AromaUnitSignalBinaryFile'></span><span id='topic+getChipType.AromaUnitSignalBinarySet'></span><span id='topic+getChipType.AromaUnitTypesFile'></span><span id='topic+getChipType.TextUnitNamesFile'></span><span id='topic+getChipType.UnitAnnotationDataFile'></span><span id='topic+getChipTypes'></span><span id='topic+getChipTypes.AromaMicroarrayDataSetTuple'></span><span id='topic+getChipTypes.ChromosomalModel'></span><span id='topic+getChromosome'></span><span id='topic+getChromosome.Arguments'></span><span id='topic+getChromosome.RawGenomicSignals'></span><span id='topic+getChromosomes'></span><span id='topic+getChromosomes.Arguments'></span><span id='topic+getChromosomes.AromaUnitChromosomeTabularBinaryFile'></span><span id='topic+getCn'></span><span id='topic+getCn.RawCopyNumbers'></span><span id='topic+getCNs'></span><span id='topic+getCNs.RawCopyNumbers'></span><span id='topic+getColClasses'></span><span id='topic+getColClasses.AromaTabularBinaryFile'></span><span id='topic+getColumnNames.AromaCellPositionFile'></span><span id='topic+getColumnNames.AromaCellSequenceFile'></span><span id='topic+getColumnNames.AromaTabularBinaryFile'></span><span id='topic+getColumnNames.AromaUflFile'></span><span id='topic+getColumnNames.AromaUgpFile'></span><span id='topic+getColumnNames.AromaUnitChromosomeTabularBinaryFile'></span><span id='topic+getCommonListElements'></span><span id='topic+getDataFileMatrix'></span><span id='topic+getDataFileMatrix.CopyNumberChromosomalModel'></span><span id='topic+getDeviceResolution'></span><span id='topic+getDeviceResolution.default'></span><span id='topic+getEffects'></span><span id='topic+getEffects.ProbePositionEffects'></span><span id='topic+getExpectedOutputFiles'></span><span id='topic+getExpectedOutputFiles.AromaTransform'></span><span id='topic+getExpectedOutputFullnames'></span><span id='topic+getExpectedOutputFullnames.AromaTransform'></span><span id='topic+getExtensionPattern.AromaCellPositionFile'></span><span id='topic+getExtensionPattern.AromaCellSequenceFile'></span><span id='topic+getExtensionPattern.AromaUflFile'></span><span id='topic+getExtensionPattern.AromaUgpFile'></span><span id='topic+getExtensionPattern.AromaUnitSignalBinaryFile'></span><span id='topic+getExtensionPattern.SampleAnnotationFile'></span><span id='topic+getExtensionPattern.TextUnitNamesFile'></span><span id='topic+getFilenameExtension.AromaCellPositionFile'></span><span id='topic+getFilenameExtension.AromaCellSequenceFile'></span><span id='topic+getFilenameExtension.AromaMicroarrayTabularBinaryFile'></span><span id='topic+getFilenameExtension.AromaUflFile'></span><span id='topic+getFilenameExtension.AromaUgpFile'></span><span id='topic+getFilenameExtension.AromaUnitChromosomeTabularBinaryFile'></span><span id='topic+getFilenameExtension.AromaUnitSignalBinaryFile'></span><span id='topic+getFilenameExtension.TextUnitNamesFile'></span><span id='topic+getFitFunction'></span><span id='topic+getFitFunction.CbsModel'></span><span id='topic+getFitFunction.CopyNumberSegmentationModel'></span><span id='topic+getFitFunction.GladModel'></span><span id='topic+getFitFunction.HaarSegModel'></span><span id='topic+getFullName.ChromosomalModel'></span><span id='topic+getFullName.Explorer'></span><span id='topic+getFullNames.AromaMicroarrayDataSetTuple'></span><span id='topic+getFullNames.ChromosomalModel'></span><span id='topic+getFullNames.ChromosomeExplorer'></span><span id='topic+getFullNames.CopyNumberSegmentationModel'></span><span id='topic+getGenome'></span><span id='topic+getGenome.ChromosomalModel'></span><span id='topic+getGenomeData'></span><span id='topic+getGenomeData.ChromosomalModel'></span><span id='topic+getGenomeFile'></span><span id='topic+getGenomeFile.ChromosomalModel'></span><span id='topic+getGenomeVersion'></span><span id='topic+getGenomeVersion.AromaUnitChromosomeTabularBinaryFile'></span><span id='topic+getHeaderParameters'></span><span id='topic+getHeaderParameters.TextUnitNamesFile'></span><span id='topic+getImage'></span><span id='topic+getImage.matrix'></span><span id='topic+getIncludePath'></span><span id='topic+getIncludePath.Explorer'></span><span id='topic+getInputDataSet'></span><span id='topic+getListOfAromaUgpFiles'></span><span id='topic+getListOfAromaUgpFiles.ChromosomalModel'></span><span id='topic+getListOfChipEffectSets'></span><span id='topic+getListOfChipEffectSets.ChromosomalModel'></span><span id='topic+getListOfUnitNamesFiles'></span><span id='topic+getListOfUnitNamesFiles.AromaUnitTotalCnBinarySetTuple'></span><span id='topic+getListOfUnitNamesFiles.ChromosomalModel'></span><span id='topic+getListOfUnitTypesFiles'></span><span id='topic+getListOfUnitTypesFiles.ChromosomalModel'></span><span id='topic+getLociFields'></span><span id='topic+getLociFields.RawGenomicSignals'></span><span id='topic+getLocusFields'></span><span id='topic+getLocusFields.RawGenomicSignals'></span><span id='topic+getLog2Ratios'></span><span id='topic+getLog2Ratios.CopyNumberSegmentationModel'></span><span id='topic+getMainPath'></span><span id='topic+getMainPath.Explorer'></span><span id='topic+getMaxLengthRepeats'></span><span id='topic+getMaxLengthRepeats.AromaCellSequenceFile'></span><span id='topic+getModel'></span><span id='topic+getName.AromaPlatform'></span><span id='topic+getName.ChromosomalModel'></span><span id='topic+getName.RawGenomicSignals'></span><span id='topic+getNameOfInput'></span><span id='topic+getNameOfInput.ChromosomeExplorer'></span><span id='topic+getNameOfInput.Explorer'></span><span id='topic+getNames.ChromosomeExplorer'></span><span id='topic+getNames.CopyNumberChromosomalModel'></span><span id='topic+getNumberOfFilesAveraged'></span><span id='topic+getNumberOfFilesAveraged.CopyNumberDataFile'></span><span id='topic+getOutputDataSet'></span><span id='topic+getOutputDataSet0'></span><span id='topic+getOutputDataSet0.AromaTransform'></span><span id='topic+getOutputFiles'></span><span id='topic+getOutputFiles.AromaTransform'></span><span id='topic+getPairedNames'></span><span id='topic+getPairedNames.CopyNumberChromosomalModel'></span><span id='topic+getParallelSafe'></span><span id='topic+getParallelSafe.Explorer'></span><span id='topic+getParametersAsString'></span><span id='topic+getParentPath'></span><span id='topic+getParentPath.ChromosomalModel'></span><span id='topic+getPath.ChromosomalModel'></span><span id='topic+getPath.ChromosomeExplorer'></span><span id='topic+getPatterns'></span><span id='topic+getPatterns.SampleAnnotationFile'></span><span id='topic+getPhysicalPositions'></span><span id='topic+getPhysicalPositions.RawCopyNumbers'></span><span id='topic+getPlatform'></span><span id='topic+getPlatform.AromaMicroarrayDataFile'></span><span id='topic+getPlatform.AromaMicroarrayDataSet'></span><span id='topic+getPlatform.AromaMicroarrayTabularBinaryFile'></span><span id='topic+getPlatform.AromaUnitSignalBinaryFile'></span><span id='topic+getPlatform.AromaUnitSignalBinarySet'></span><span id='topic+getPlatform.AromaUnitTypesFile'></span><span id='topic+getPlatform.TextUnitNamesFile'></span><span id='topic+getPlatform.UnitAnnotationDataFile'></span><span id='topic+getPloidy'></span><span id='topic+getPloidy.AromaMicroarrayDataFile'></span><span id='topic+getPositions'></span><span id='topic+getPositions.AromaUgpFile'></span><span id='topic+getPositions.RawGenomicSignals'></span><span id='topic+getProbeLength'></span><span id='topic+getProbeLength.AromaCellSequenceFile'></span><span id='topic+getProbePositionEffectDesignMatrix'></span><span id='topic+getProbePositionEffectDesignMatrix.character'></span><span id='topic+getProbePositionEffectDesignMatrix.raw'></span><span id='topic+getRam'></span><span id='topic+getRam.AromaSettings'></span><span id='topic+getRawCnData'></span><span id='topic+getRawCnData.CopyNumberChromosomalModel'></span><span id='topic+getReferenceSetTuple'></span><span id='topic+getReferenceSetTuple.CopyNumberChromosomalModel'></span><span id='topic+getRefSetTuple'></span><span id='topic+getRefSetTuple.CopyNumberChromosomalModel'></span><span id='topic+getRegions'></span><span id='topic+getRegions.CopyNumberSegmentationModel'></span><span id='topic+getReportPath'></span><span id='topic+getReportPath.ChromosomalModel'></span><span id='topic+getReportPathPattern'></span><span id='topic+getReportPathPattern.Explorer'></span><span id='topic+getRootPath'></span><span id='topic+getRootPath.ChromosomalModel'></span><span id='topic+getSampleLabels'></span><span id='topic+getSampleLabels.ChromosomeExplorer'></span><span id='topic+getSampleLayerName'></span><span id='topic+getSampleLayerName.Explorer'></span><span id='topic+getSampleLayerPrefix'></span><span id='topic+getSampleLayerPrefix.Explorer'></span><span id='topic+getSets.AromaMicroarrayDataSetTuple'></span><span id='topic+getSets.ChromosomalModel'></span><span id='topic+getSetTag'></span><span id='topic+getSetTag.ChromosomalModel'></span><span id='topic+getSetTag.RawCopyNumberModel'></span><span id='topic+getSetTuple'></span><span id='topic+getSetTuple.ChromosomalModel'></span><span id='topic+getSigma'></span><span id='topic+getSigma.RawGenomicSignals'></span><span id='topic+getSignals'></span><span id='topic+getSignals.RawGenomicSignals'></span><span id='topic+getSnpNucleotides'></span><span id='topic+getSnpNucleotides.AromaCellSequenceFile'></span><span id='topic+getSnpPositions'></span><span id='topic+getSnpPositions.AromaCellSequenceFile'></span><span id='topic+getSnpShifts'></span><span id='topic+getSnpShifts.AromaCellSequenceFile'></span><span id='topic+getStateColorMap'></span><span id='topic+getStateColorMap.SegmentedGenomicSignalsInterface'></span><span id='topic+getStateColors'></span><span id='topic+getStateColors.SegmentedGenomicSignalsInterface'></span><span id='topic+getStates'></span><span id='topic+getStates.SegmentedGenomicSignalsInterface'></span><span id='topic+getSubname'></span><span id='topic+getSubname.Explorer'></span><span id='topic+getTableOfArrays'></span><span id='topic+getTableOfArrays.AromaMicroarrayDataSetTuple'></span><span id='topic+getTableOfArrays.ChromosomalModel'></span><span id='topic+getTags.AromaMicroarrayDataSetTuple'></span><span id='topic+getTags.ChromosomalModel'></span><span id='topic+getTags.CopyNumberSegmentationModel'></span><span id='topic+getTagsOfInput'></span><span id='topic+getTagsOfInput.ChromosomeExplorer'></span><span id='topic+getTagsOfInput.Explorer'></span><span id='topic+getTemplatePath'></span><span id='topic+getTemplatePath.Explorer'></span><span id='topic+getUniqueStates'></span><span id='topic+getUniqueStates.SegmentedGenomicSignalsInterface'></span><span id='topic+getUnitAnnotationDataFile'></span><span id='topic+getUnitNames.TextUnitNamesFile'></span><span id='topic+getUnitNames.UnitNamesFile'></span><span id='topic+getUnitNamesFile'></span><span id='topic+getUnitNamesFile.AromaPlatform'></span><span id='topic+getUnitNamesFile.AromaUnitTotalCnBinarySet'></span><span id='topic+getUnitsAt'></span><span id='topic+getUnitsAt.AromaUgpFile'></span><span id='topic+getUnitsOnChromosome'></span><span id='topic+getUnitsOnChromosome.AromaUgpFile'></span><span id='topic+getUnitsOnChromosome.AromaUnitChromosomeTabularBinaryFile'></span><span id='topic+getUnitsOnChromosomes'></span><span id='topic+getUnitsOnChromosomes.AromaUnitChromosomeTabularBinaryFile'></span><span id='topic+getUnitTypes'></span><span id='topic+getUnitTypes.AromaUnitTypesFile'></span><span id='topic+getUnitTypes.UnitTypesFile'></span><span id='topic+getUnitTypesFile'></span><span id='topic+getUnitTypesFile.AromaPlatform'></span><span id='topic+getVerbose.AromaSettings'></span><span id='topic+getWeights'></span><span id='topic+getWeights.RawGenomicSignals'></span><span id='topic+getXAM'></span><span id='topic+getXAM.AromaMicroarrayDataFile'></span><span id='topic+getXAM.AromaUnitTotalCnBinaryFile'></span><span id='topic+getXScale'></span><span id='topic+getXScale.RawGenomicSignals'></span><span id='topic+getXY'></span><span id='topic+getXY.RawGenomicSignals'></span><span id='topic+getYScale'></span><span id='topic+getYScale.RawGenomicSignals'></span><span id='topic+getZooms'></span><span id='topic+getZooms.ChromosomeExplorer'></span><span id='topic+groupBySnpNucleotides'></span><span id='topic+groupBySnpNucleotides.AromaCellSequenceFile'></span><span id='topic+hasAlleleBFractions'></span><span id='topic+hasAlleleBFractions.AromaUnitTotalCnBinaryFile'></span><span id='topic+hasAlleleBFractions.CopyNumberDataFile'></span><span id='topic+hasAlleleBFractions.CopyNumberDataSet'></span><span id='topic+hasAlleleBFractions.CopyNumberDataSetTuple'></span><span id='topic+hasAttributeXY'></span><span id='topic+hasAttributeXY.AromaMicroarrayDataFile'></span><span id='topic+hasStrandiness'></span><span id='topic+hasStrandiness.AromaUnitTotalCnBinaryFile'></span><span id='topic+hasStrandiness.CopyNumberDataFile'></span><span id='topic+hasStrandiness.CopyNumberDataSet'></span><span id='topic+hasStrandiness.CopyNumberDataSetTuple'></span><span id='topic+hasWeights'></span><span id='topic+hasWeights.RawGenomicSignals'></span><span id='topic+importFrom'></span><span id='topic+importFrom.AromaTabularBinaryFile'></span><span id='topic+importFromGenericTabularFile'></span><span id='topic+importFromGenericTabularFile.AromaUflFile'></span><span id='topic+importFromGenericTabularFile.AromaUgpFile'></span><span id='topic+importFromTable'></span><span id='topic+importFromTable.FileMatrix'></span><span id='topic+importFromUnitTypesFile'></span><span id='topic+importFromUnitTypesFile.AromaUnitTypesFile'></span><span id='topic+indexOf.AromaMicroarrayDataSetTuple'></span><span id='topic+indexOf.ChromosomalModel'></span><span id='topic+indexOf.ChromosomeExplorer'></span><span id='topic+indexOfColumn'></span><span id='topic+indexOfColumn.AromaUnitChromosomeTabularBinaryFile'></span><span id='topic+indexOfUnits'></span><span id='topic+indexOfUnits.AromaUnitTabularBinaryFile'></span><span id='topic+interleave'></span><span id='topic+interleave.Image'></span><span id='topic+isAverageFile'></span><span id='topic+isAverageFile.AromaMicroarrayDataFile'></span><span id='topic+isAverageFile.AromaUnitSignalBinaryFile'></span><span id='topic+isHeterozygote'></span><span id='topic+isHeterozygote.AromaUnitGenotypeCallFile'></span><span id='topic+isHeterozygous'></span><span id='topic+isHeterozygous.AromaUnitGenotypeCallFile'></span><span id='topic+isHomozygote'></span><span id='topic+isHomozygote.AromaUnitGenotypeCallFile'></span><span id='topic+isHomozygous'></span><span id='topic+isHomozygous.AromaUnitGenotypeCallFile'></span><span id='topic+isMissing'></span><span id='topic+isMissing.AromaCellPositionFile'></span><span id='topic+isMissing.AromaCellSequenceFile'></span><span id='topic+isPaired'></span><span id='topic+isPaired.CopyNumberChromosomalModel'></span><span id='topic+kernelSmoothing.RawGenomicSignals'></span><span id='topic+kernelSmoothingByState'></span><span id='topic+kernelSmoothingByState.SegmentedGenomicSignalsInterface'></span><span id='topic+colApply'></span><span id='topic+colApply.AromaTabularBinaryFile'></span><span id='topic+lapplyInChunks'></span><span id='topic+lapplyInChunks.numeric'></span><span id='topic+library'></span><span id='topic+lines.CopyNumberOutliers'></span><span id='topic+lines.CopyNumberRegions'></span><span id='topic+lines.RawGenomicSignals'></span><span id='topic+listToXml'></span><span id='topic+listToXml.list'></span><span id='topic+locallyUnique'></span><span id='topic+locallyUnique.default'></span><span id='topic+log2abs'></span><span id='topic+log2center'></span><span id='topic+log2neg'></span><span id='topic+log2pos'></span><span id='topic+matchPatterns'></span><span id='topic+matchPatterns.SampleAnnotationFile'></span><span id='topic+matrixBlockPolish'></span><span id='topic+mergeBoxplotStats'></span><span id='topic+multiplyBy'></span><span id='topic+multiplyBy.RawGenomicSignals'></span><span id='topic+nbrOfCells'></span><span id='topic+nbrOfCells.AromaCellTabularBinaryFile'></span><span id='topic+nbrOfChipTypes'></span><span id='topic+nbrOfColumns.AromaTabularBinaryFile'></span><span id='topic+nbrOfEnzymes'></span><span id='topic+nbrOfEnzymes.AromaUflFile'></span><span id='topic+nbrOfLoci'></span><span id='topic+nbrOfLoci.RawGenomicSignals'></span><span id='topic+nbrOfReads'></span><span id='topic+nbrOfReads.RawSequenceReads'></span><span id='topic+nbrOfRegions'></span><span id='topic+nbrOfRegions.CopyNumberOutliers'></span><span id='topic+nbrOfRegions.CopyNumberRegions'></span><span id='topic+nbrOfRows.AromaTabularBinaryFile'></span><span id='topic+nbrOfUnits'></span><span id='topic+nbrOfUnits.AromaUnitSignalBinaryFile'></span><span id='topic+nbrOfUnits.AromaUnitTabularBinaryFile'></span><span id='topic+nbrOfUnits.TextUnitNamesFile'></span><span id='topic+nbrOfUnits.UnitAnnotationDataFile'></span><span id='topic+nbrOfUnits.UnitNamesFile'></span><span id='topic+nbrOfUnits.UnitTypesFile'></span><span id='topic+newPlot'></span><span id='topic+newPlot.CopyNumberChromosomalModel'></span><span id='topic+norm2d'></span><span id='topic+norm2d.matrix'></span><span id='topic+normalizeGenotypeCone'></span><span id='topic+normalizeGenotypeCone.matrix'></span><span id='topic+onFitAddGenotypeCalls'></span><span id='topic+onFitAddGenotypeCalls.default'></span><span id='topic+onFitAddGenotypeCalls.GladModel'></span><span id='topic+plot.BinnedScatter'></span><span id='topic+plot.CopyNumberSegmentationModel'></span><span id='topic+plot.ProbePositionEffects'></span><span id='topic+plot.RawAlleleBFractions'></span><span id='topic+plot.RawCopyNumbers'></span><span id='topic+plot.RawGenomicSignals'></span><span id='topic+plot.RawMirroredAlleleBFractions'></span><span id='topic+plot.RawSequenceReads'></span><span id='topic+plot.SegmentedGenomicSignalsInterface'></span><span id='topic+plotAxesLayers'></span><span id='topic+plotAxesLayers.CopyNumberChromosomalModel'></span><span id='topic+plotChromosomesLayers'></span><span id='topic+plotChromosomesLayers.CopyNumberChromosomalModel'></span><span id='topic+plotCopyNumberRegionLayers'></span><span id='topic+plotCopyNumberRegionLayers.CopyNumberSegmentationModel'></span><span id='topic+plotCytobandLayers'></span><span id='topic+plotCytobandLayers.CopyNumberChromosomalModel'></span><span id='topic+plotFitLayers'></span><span id='topic+plotFitLayers.CopyNumberChromosomalModel'></span><span id='topic+plotGridHorizontalLayers'></span><span id='topic+plotGridHorizontalLayers.CopyNumberChromosomalModel'></span><span id='topic+plotProfile2'></span><span id='topic+plotProfile2.profileCGH'></span><span id='topic+plotRawCNs'></span><span id='topic+plotRawCNs.profileCGH'></span><span id='topic+plotRawCopyNumbers'></span><span id='topic+plotRawCopyNumbers.CopyNumberChromosomalModel'></span><span id='topic+plotSampleLayers'></span><span id='topic+plotSampleLayers.CopyNumberChromosomalModel'></span><span id='topic+points.BinnedScatter'></span><span id='topic+points.RawGenomicSignals'></span><span id='topic+points.SegmentedGenomicSignalsInterface'></span><span id='topic+pointsRawCNs'></span><span id='topic+pointsRawCNs.default'></span><span id='topic+pointsRawCNs.profileCGH'></span><span id='topic+pointsSequence'></span><span id='topic+pointsSequence.ProbePositionEffects'></span><span id='topic+predict.ProbePositionEffects'></span><span id='topic+readColumns.AromaTabularBinaryFile'></span><span id='topic+readDataFrame.AromaTabularBinaryFile'></span><span id='topic+readDataFrame.AromaUflFile'></span><span id='topic+readDataFrame.AromaUnitChromosomeTabularBinaryFile'></span><span id='topic+readDataFrame.AromaUnitSignalBinaryFile'></span><span id='topic+readDataFrame.SampleAnnotationFile'></span><span id='topic+readFooter'></span><span id='topic+readHeader'></span><span id='topic+readHeader.AromaTabularBinaryFile'></span><span id='topic+readNeighborSequenceMatrix'></span><span id='topic+readNeighborSequenceMatrix.AromaCellSequenceFile'></span><span id='topic+readPairSequenceMatrix'></span><span id='topic+readPairSequenceMatrix.AromaCellSequenceFile'></span><span id='topic+readPositions'></span><span id='topic+readPositions.AromaCellPositionFile'></span><span id='topic+readRawFooter'></span><span id='topic+readRawFooter.AromaTabularBinaryFile'></span><span id='topic+readSequenceMatrix'></span><span id='topic+readSequenceMatrix.AromaCellSequenceFile'></span><span id='topic+readSequences'></span><span id='topic+readSequences.AromaCellSequenceFile'></span><span id='topic+readTargetStrands'></span><span id='topic+readTargetStrands.AromaCellSequenceFile'></span><span id='topic+remap'></span><span id='topic+remap.default'></span><span id='topic+reorder.BinnedScatter'></span><span id='topic+require'></span><span id='topic+rescale'></span><span id='topic+rescale.Image'></span><span id='topic+rgbTransform'></span><span id='topic+rgbTransform.Image'></span><span id='topic+SampleAnnotationFile'></span><span id='topic+SampleAnnotationSet'></span><span id='topic+segmentByCBS'></span><span id='topic+segmentByGLAD'></span><span id='topic+segmentByHaarSeg'></span><span id='topic+setAlias.ChromosomalModel'></span><span id='topic+setArrays'></span><span id='topic+setAttributesBy.AromaMicroarrayDataSet'></span><span id='topic+setAttributesBySampleAnnotationFile'></span><span id='topic+setAttributesBySampleAnnotationFile.AromaMicroarrayDataSet'></span><span id='topic+setAttributesBySampleAnnotationSet'></span><span id='topic+setAttributesBySampleAnnotationSet.AromaMicroarrayDataSet'></span><span id='topic+setAttributesByTags.AromaMicroarrayDataFile'></span><span id='topic+setAttributesByTags.AromaTabularBinaryFile'></span><span id='topic+setAttributeXY'></span><span id='topic+setAttributeXY.AromaMicroarrayDataFile'></span><span id='topic+setCytoband'></span><span id='topic+setCytoband.ChromosomeExplorer'></span><span id='topic+setGenome'></span><span id='topic+setGenome.ChromosomalModel'></span><span id='topic+setLocusFields'></span><span id='topic+setLocusFields.RawGenomicSignals'></span><span id='topic+setName.RawGenomicSignals'></span><span id='topic+setParallelSafe'></span><span id='topic+setParallelSafe.Explorer'></span><span id='topic+setRam'></span><span id='topic+setRam.AromaSettings'></span><span id='topic+setReportPathPattern'></span><span id='topic+setReportPathPattern.Explorer'></span><span id='topic+setSigma'></span><span id='topic+setSigma.RawGenomicSignals'></span><span id='topic+setStateColorMap'></span><span id='topic+setStateColorMap.SegmentedGenomicSignalsInterface'></span><span id='topic+setStates'></span><span id='topic+setStates.SegmentedGenomicSignalsInterface'></span><span id='topic+setSubname'></span><span id='topic+setSubname.Explorer'></span><span id='topic+setTags.AromaTransform'></span><span id='topic+setup'></span><span id='topic+setup.ChromosomeExplorer'></span><span id='topic+setup.Explorer'></span><span id='topic+setVerbose'></span><span id='topic+setVerbose.AromaSettings'></span><span id='topic+setWeights'></span><span id='topic+setWeights.RawGenomicSignals'></span><span id='topic+setXScale'></span><span id='topic+setXScale.RawGenomicSignals'></span><span id='topic+setYScale'></span><span id='topic+setYScale.RawGenomicSignals'></span><span id='topic+setZooms'></span><span id='topic+setZooms.ChromosomeExplorer'></span><span id='topic+shakyText'></span><span id='topic+shakyText.default'></span><span id='topic+signalRange'></span><span id='topic+signalRange.RawGenomicSignals'></span><span id='topic+smoothWRMA'></span><span id='topic+smoothWRMA.matrix'></span><span id='topic+smoothWSA'></span><span id='topic+smoothWSA.matrix'></span><span id='topic+sort.RawGenomicSignals'></span><span id='topic+splitByReportPathPattern'></span><span id='topic+splitByReportPathPattern.Explorer'></span><span id='topic+splitInChunks'></span><span id='topic+splitInChunks.numeric'></span><span id='topic+sqrtabs'></span><span id='topic+sqrtcenter'></span><span id='topic+sqrtneg'></span><span id='topic+sqrtpos'></span><span id='topic+sqrtsign'></span><span id='topic+stextChipType'></span><span id='topic+stextChipType.character'></span><span id='topic+stextLabel'></span><span id='topic+stextLabel.AromaMicroarrayDataFile'></span><span id='topic+stextLabels'></span><span id='topic+stextLabels.AromaMicroarrayDataFile'></span><span id='topic+stextSize'></span><span id='topic+stextSize.AromaMicroarrayDataFile'></span><span id='topic+stringTree'></span><span id='topic+stringTree.character'></span><span id='topic+subsample'></span><span id='topic+subsample.BinnedScatter'></span><span id='topic+subset.AromaTabularBinaryFile'></span><span id='topic+subset.BinnedScatter'></span><span id='topic+subtractBy'></span><span id='topic+subtractBy.RawGenomicSignals'></span><span id='topic+summary.AromaTabularBinaryFile'></span><span id='topic+summary.RawGenomicSignals'></span><span id='topic+summaryOfUnits'></span><span id='topic+summaryOfUnits.AromaUflFile'></span><span id='topic+text.ProbePositionEffects'></span><span id='topic+textSequence'></span><span id='topic+textSequence.ProbePositionEffects'></span><span id='topic+TextUnitNamesFile'></span><span id='topic+updateData'></span><span id='topic+updateData.AromaTabularBinaryFile'></span><span id='topic+updateDataColumn'></span><span id='topic+updateDataColumn.AromaTabularBinaryFile'></span><span id='topic+updateGenotypes'></span><span id='topic+updateGenotypes.AromaUnitGenotypeCallFile'></span><span id='topic+updatePositions'></span><span id='topic+updatePositions.AromaCellPositionFile'></span><span id='topic+updateSequenceMatrix'></span><span id='topic+updateSequenceMatrix.AromaCellSequenceFile'></span><span id='topic+updateSequences'></span><span id='topic+updateSequences.AromaCellSequenceFile'></span><span id='topic+updateTargetStrands'></span><span id='topic+updateTargetStrands.AromaCellSequenceFile'></span><span id='topic+validate.AromaMicroarrayDataSet'></span><span id='topic+validate.AromaUnitSignalBinarySet'></span><span id='topic+weightedMean'></span><span id='topic+whatDataType'></span><span id='topic+whatDataType.default'></span><span id='topic+writeAxesLayers'></span><span id='topic+writeAxesLayers.ChromosomeExplorer'></span><span id='topic+writeCopyNumberRegionLayers'></span><span id='topic+writeCopyNumberRegionLayers.ChromosomeExplorer'></span><span id='topic+writeCytobandLayers'></span><span id='topic+writeCytobandLayers.ChromosomeExplorer'></span><span id='topic+writeFooter'></span><span id='topic+writeGraphs'></span><span id='topic+writeGraphs.ChromosomeExplorer'></span><span id='topic+writeGridHorizontalLayers'></span><span id='topic+writeGridHorizontalLayers.ChromosomeExplorer'></span><span id='topic+writeRawCopyNumberLayers'></span><span id='topic+writeRawCopyNumberLayers.ChromosomeExplorer'></span><span id='topic+writeRawFooter'></span><span id='topic+writeRawFooter.AromaTabularBinaryFile'></span><span id='topic+writeRegions'></span><span id='topic+writeRegions.ChromosomeExplorer'></span><span id='topic+writeRegions.CopyNumberSegmentationModel'></span><span id='topic+writeRegions.GladModel'></span><span id='topic+xMax'></span><span id='topic+xMax.RawGenomicSignals'></span><span id='topic+xMin'></span><span id='topic+xMin.RawGenomicSignals'></span><span id='topic+xmlToList'></span><span id='topic+xmlToList.character'></span><span id='topic+xRange'></span><span id='topic+xRange.RawGenomicSignals'></span><span id='topic+xSeq'></span><span id='topic+xSeq.RawGenomicSignals'></span><span id='topic+yellow.colors'></span><span id='topic+yMax'></span><span id='topic+yMax.RawGenomicSignals'></span><span id='topic+yMin'></span><span id='topic+yMin.RawGenomicSignals'></span><span id='topic+yRange'></span><span id='topic+yRange.RawGenomicSignals'></span><span id='topic+getDefaultFullName.AromaMicroarrayDataSet'></span><span id='topic+getDefaultFullName.AromaTabularBinarySet'></span><span id='topic+drawCnRegions.MPCBS'></span><span id='topic+extractCopyNumberRegions.MPCBS'></span><span id='topic+segmentByMPCBS'></span><span id='topic+isCompatibleWith'></span><span id='topic+getTags.Arguments'></span><span id='topic+getOutputSet'></span><span id='topic+getOutputSet.ChromosomalModel'></span><span id='topic+setAttributesBy.AromaTabularBinarySet'></span><span id='topic+setAttributesBySampleAnnotationFile.AromaTabularBinarySet'></span><span id='topic+setAttributesBySampleAnnotationSet.AromaTabularBinarySet'></span><span id='topic+setAttributesByTags.AromaTabularBinarySet'></span><span id='topic+allocate.AromaCellCpgFile'></span><span id='topic+getColumnNames.AromaCellCpgFile'></span><span id='topic+getFilenameExtension.AromaCellCpgFile'></span><span id='topic+isMissing.AromaCellCpgFile'></span><span id='topic+readCpgs'></span><span id='topic+readCpgs.AromaCellCpgFile'></span><span id='topic+updateCpgs'></span><span id='topic+updateCpgs.AromaCellCpgFile'></span><span id='topic+getRandomSeed'></span><span id='topic+getRandomSeed.CbsModel'></span><span id='topic+setRandomSeed'></span><span id='topic+setRandomSeed.CbsModel'></span><span id='topic+equals.CopyNumberRegions'></span><span id='topic+writeDataFrame'></span><span id='topic+getParentName'></span><span id='topic+getParentName.GenericDataFile'></span><span id='topic+getParentName.GenericDataFileSet'></span><span id='topic+getRootName'></span><span id='topic+getRootName.AromaTabularBinaryFile'></span><span id='topic+getRootName.AromaTabularBinarySet'></span><span id='topic+getAromaFullNameTranslatorSet'></span><span id='topic+getAromaFullNameTranslatorSet.character'></span><span id='topic+getFullNameTranslatorSet'></span><span id='topic+getFullNameTranslatorSet.character'></span><span id='topic+getFullNameTranslatorSet.GenericDataFileSet'></span><span id='topic+getAromaFullNameTranslatorSet.AromaMicroarrayDataSet'></span><span id='topic+getAromaFullNameTranslatorSet.AromaUnitSignalBinarySet'></span><span id='topic+extractRegions'></span><span id='topic+extractRegions.RawGenomicSignals'></span><span id='topic+subset.CopyNumberRegions'></span><span id='topic+writeDataFrame.AromaUnitTotalCnBinarySet'></span><span id='topic+writeDataFrame.AromaUnitFracBCnBinarySet'></span><span id='topic+-.CopyNumberRegions'></span><span id='topic++2A.CopyNumberRegions'></span><span id='topic++2B.CopyNumberRegions'></span><span id='topic+getMeanAt'></span><span id='topic+getMeanAt.CopyNumberRegions'></span><span id='topic+getValueAt'></span><span id='topic+getValueAt.CopyNumberRegions'></span><span id='topic+prune'></span><span id='topic+prune.CopyNumberRegions'></span><span id='topic+simulateRawCopyNumbers'></span><span id='topic+simulateRawCopyNumbers.CopyNumberRegions'></span><span id='topic+xRange.CopyNumberRegions'></span><span id='topic+xMax.CopyNumberRegions'></span><span id='topic+xMin.CopyNumberRegions'></span><span id='topic+append.CopyNumberRegions'></span><span id='topic+as.character.SegmentationDataSet'></span><span id='topic+byName.CbsSegmentationDataSet'></span><span id='topic+byPath.CbsSegmentationDataSet'></span><span id='topic+byPath.SegmentationDataSet'></span><span id='topic+CbsSegmentationDataFile'></span><span id='topic+CbsSegmentationDataSet'></span><span id='topic+extractByReferenceName'></span><span id='topic+extractByReferenceName.SegmentationDataSet'></span><span id='topic+extractBySampleName'></span><span id='topic+extractBySampleName.SegmentationDataSet'></span><span id='topic+extractCopyNumberRegions.SegmentationDataFile'></span><span id='topic+extractCopyNumberRegions.SegmentationDataSet'></span><span id='topic+getChipType.SegmentationDataSet'></span><span id='topic+getChromosome.SegmentationDataFile'></span><span id='topic+getChromosomes.CopyNumberRegions'></span><span id='topic+getChromosomes.SegmentationDataSet'></span><span id='topic+getDefaultFullName.SegmentationDataSet'></span><span id='topic+getReferenceName'></span><span id='topic+getReferenceName.SegmentationDataFile'></span><span id='topic+getReferenceNames'></span><span id='topic+getReferenceNames.SegmentationDataSet'></span><span id='topic+getSampleName'></span><span id='topic+getSampleName.SegmentationDataFile'></span><span id='topic+getSampleNames'></span><span id='topic+getSampleNames.SegmentationDataSet'></span><span id='topic+loadFit'></span><span id='topic+loadFit.CbsSegmentationDataFile'></span><span id='topic+loadFit.SegmentationDataFile'></span><span id='topic+nbrOfChromosomes'></span><span id='topic+nbrOfChromosomes.CopyNumberRegions'></span><span id='topic+SegmentationDataFile'></span><span id='topic+SegmentationDataSet'></span><span id='topic+unique.CopyNumberRegions'></span><span id='topic+byGenome'></span><span id='topic+byGenome.AromaGenomeTextFile'></span><span id='topic+findByGenome'></span><span id='topic+findByGenome.AromaGenomeTextFile'></span><span id='topic+readDataFrame.AromaGenomeTextFile'></span><span id='topic+-.RawGenomicSignals'></span><span id='topic++2A.RawGenomicSignals'></span><span id='topic++2B.RawGenomicSignals'></span><span id='topic+drawDensity'></span><span id='topic+drawDensity.CopyNumberRegions'></span><span id='topic+drawDensity.RawGenomicSignals'></span><span id='topic+getDensity'></span><span id='topic+getDensity.CopyNumberRegions'></span><span id='topic+getLength'></span><span id='topic+getLength.CopyNumberRegions'></span><span id='topic+getOptionalArguments'></span><span id='topic+getOptionalArguments.CopyNumberChromosomalModel'></span><span id='topic+getChromosomeLength'></span><span id='topic+getChromosomeLength.CopyNumberChromosomalModel'></span><span id='topic+requireWithMemory'></span><span id='topic+requireWithMemory.default'></span><span id='topic+writeImage'></span><span id='topic+writeImage.Image'></span><span id='topic+createImage'></span><span id='topic+createImage.matrix'></span><span id='topic+getImageData'></span><span id='topic+getImageData.Image'></span><span id='topic+setImageData'></span><span id='topic+setImageData.Image'></span><span id='topic+RasterImage'></span><span id='topic++5B.RasterImage'></span><span id='topic+as.character.RasterImage'></span><span id='topic+display.RasterImage'></span><span id='topic+dropAlpha'></span><span id='topic+dropAlpha.RasterImage'></span><span id='topic+getImageData.RasterImage'></span><span id='topic+read'></span><span id='topic+read.RasterImage'></span><span id='topic+rescale.RasterImage'></span><span id='topic+save.RasterImage'></span><span id='topic+setImageData.RasterImage'></span><span id='topic+str.RasterImage'></span><span id='topic+write.RasterImage'></span><span id='topic+writeImage.RasterImage'></span><span id='topic+colorize.RasterImage'></span><span id='topic+interleave.RasterImage'></span><span id='topic+dropRootPathTags.default'></span><span id='topic+loadAll'></span><span id='topic+loadAll.SampleAnnotationSet'></span><span id='topic+byPathnames'></span><span id='topic+byPathnames.SampleAnnotationSet'></span><span id='topic+getAromaGenomeTextFile'></span><span id='topic+getAromaGenomeTextFile.ChromosomalModel'></span><span id='topic+getChromosomeLabels'></span><span id='topic+getChromosomeLabels.ChromosomeExplorer'></span><span id='topic+cat'></span><span id='topic+getOption'></span><span id='topic+append'></span><span id='topic+lapply'></span><span id='topic+downloadACS'></span><span id='topic+downloadACS.AromaRepository'></span><span id='topic+downloadCDF'></span><span id='topic+downloadCDF.AromaRepository'></span><span id='topic+downloadChipTypeFile'></span><span id='topic+downloadProbeSeqsTXT'></span><span id='topic+downloadProbeSeqsTXT.AromaRepository'></span><span id='topic+downloadTXT'></span><span id='topic+downloadTXT.AromaRepository'></span><span id='topic+downloadUFL'></span><span id='topic+downloadUFL.AromaRepository'></span><span id='topic+downloadUGP'></span><span id='topic+downloadUGP.AromaRepository'></span><span id='topic+getUrlPath'></span><span id='topic+getUrlPath.AromaRepository'></span><span id='topic+getVerbose.AromaRepository'></span><span id='topic+setVerbose.AromaRepository'></span><span id='topic+downloadACC'></span><span id='topic+downloadACC.AromaRepository'></span><span id='topic+downloadACM'></span><span id='topic+downloadACM.AromaRepository'></span><span id='topic+downloadACP'></span><span id='topic+downloadACP.AromaRepository'></span><span id='topic+extractPSCNArray'></span><span id='topic+extractPSCNArray.AromaUnitTotalCnBinaryFile'></span><span id='topic+extractPSCNArray.AromaUnitTotalCnBinarySet'></span><span id='topic+extractPSCNMatrix'></span><span id='topic+extractPSCNMatrix.AromaUnitTotalCnBinaryFile'></span><span id='topic+AromaUcscGenomeTextFile'></span><span id='topic+findByGenome.AromaUcscGenomeTextFile'></span><span id='topic+getDefaultExtension'></span><span id='topic+getDefaultExtension.UnitAnnotationDataFile'></span><span id='topic+readDataFrame.AromaUcscGenomeTextFile'></span><span id='topic+getDefaultExtension.AromaCellCpgFile'></span><span id='topic+getDefaultExtension.AromaCellPositionFile'></span><span id='topic+getDefaultExtension.AromaCellSequenceFile'></span><span id='topic+getDefaultExtension.AromaUflFile'></span><span id='topic+getDefaultExtension.AromaUgpFile'></span><span id='topic+plotCoverageMap'></span><span id='topic+plotCoverageMap.AromaUgpFile'></span><span id='topic+assertOneChromosome'></span><span id='topic+assertOneChromosome.RawGenomicSignals'></span><span id='topic+getCXY'></span><span id='topic+getCXY.RawGenomicSignals'></span><span id='topic+getChromosomes.RawGenomicSignals'></span><span id='topic+nbrOfChromosomes.RawGenomicSignals'></span><span id='topic+extractChromosome'></span><span id='topic+extractChromosome.RawGenomicSignals'></span><span id='topic+extractChromosomes'></span><span id='topic+extractChromosomes.RawGenomicSignals'></span><span id='topic+clearCache.RawGenomicSignals'></span><span id='topic+clone.RawGenomicSignals'></span><span id='topic+getBasicField'></span><span id='topic+getBasicField.RawGenomicSignals'></span><span id='topic+print.RawGenomicSignals'></span><span id='topic+setBasicField'></span><span id='topic+setBasicField.RawGenomicSignals'></span><span id='topic+newInstance.RawGenomicSignals'></span><span id='topic+getDefaultLocusFields'></span><span id='topic+getDefaultLocusFields.RawGenomicSignals'></span><span id='topic+getVirtualLocusFields'></span><span id='topic+getVirtualLocusFields.RawGenomicSignals'></span><span id='topic+getVirtualLocusFields.SegmentedGenomicSignalsInterface'></span><span id='topic+getVersion.Explorer'></span><span id='topic+updateSetupExplorerFile'></span><span id='topic+updateSetupExplorerFile.Explorer'></span><span id='topic++24.RichDataFrame'></span><span id='topic++24+3C-.RichDataFrame'></span><span id='topic+RichDataFrame'></span><span id='topic++5B.RichDataFrame'></span><span id='topic++5B+5B.RichDataFrame'></span><span id='topic++5B+5B+3C-.RichDataFrame'></span><span id='topic+as.data.frame.RichDataFrame'></span><span id='topic+as.list.RichDataFrame'></span><span id='topic+binnedSmoothingByField'></span><span id='topic+binnedSmoothingByField.RawGenomicSignals'></span><span id='topic+dim.RichDataFrame'></span><span id='topic+dropVirtualColumn'></span><span id='topic+dropVirtualColumn.RichDataFrame'></span><span id='topic+getColumnNames.RichDataFrame'></span><span id='topic+getColumnNamesTranslator'></span><span id='topic+getColumnNamesTranslator.RichDataFrame'></span><span id='topic+getFullName.RichDataFrame'></span><span id='topic+getGenericSummary'></span><span id='topic+getGenericSummary.RichDataFrame'></span><span id='topic+getName.RichDataFrame'></span><span id='topic+getSignalColumnName'></span><span id='topic+getSignalColumnName.RawGenomicSignals'></span><span id='topic+getSignalColumnNames'></span><span id='topic+getSignalColumnNames.RawGenomicSignals'></span><span id='topic+getSignals.RawCopyNumbers'></span><span id='topic+getTags.RichDataFrame'></span><span id='topic+getVirtualColumn'></span><span id='topic+getVirtualColumn.RichDataFrame'></span><span id='topic+getVirtualColumnFunction'></span><span id='topic+getVirtualColumnFunction.RichDataFrame'></span><span id='topic+getVirtualColumnFunctions'></span><span id='topic+getVirtualColumnFunctions.RichDataFrame'></span><span id='topic+getVirtualColumnNames'></span><span id='topic+getVirtualColumnNames.RichDataFrame'></span><span id='topic+getVirtualField'></span><span id='topic+getVirtualField.SegmentedGenomicSignalsInterface'></span><span id='topic+hasColumn'></span><span id='topic+hasColumn.RichDataFrame'></span><span id='topic+hasColumns'></span><span id='topic+hasColumns.RichDataFrame'></span><span id='topic+hasVirtualColumn'></span><span id='topic+hasVirtualColumn.RichDataFrame'></span><span id='topic+hasVirtualColumns'></span><span id='topic+hasVirtualColumns.RichDataFrame'></span><span id='topic+length.RichDataFrame'></span><span id='topic+names.RichDataFrame'></span><span id='topic+newInstance.RichDataFrame'></span><span id='topic+print.RichDataFrame'></span><span id='topic+rbind.RichDataFrame'></span><span id='topic+setAttributes.RichDataFrame'></span><span id='topic+setColumnNamesMap'></span><span id='topic+setColumnNamesMap.RichDataFrame'></span><span id='topic+setColumnNamesTranslator'></span><span id='topic+setColumnNamesTranslator.RichDataFrame'></span><span id='topic+setName.RichDataFrame'></span><span id='topic+setSignals'></span><span id='topic+setSignals.RawGenomicSignals'></span><span id='topic+setTags.RichDataFrame'></span><span id='topic+setVirtualColumn'></span><span id='topic+setVirtualColumn.RichDataFrame'></span><span id='topic+setVirtualColumnFunctions'></span><span id='topic+setVirtualColumnFunctions.RichDataFrame'></span><span id='topic+subset.RichDataFrame'></span><span id='topic+translateColumnNames.RichDataFrame'></span><span id='topic+as.NonPairedPSCNData'></span><span id='topic+as.NonPairedPSCNData.NonPairedPSCNData'></span><span id='topic+as.NonPairedPSCNData.data.frame'></span><span id='topic+as.PairedPSCNData'></span><span id='topic+as.PairedPSCNData.NonPairedPSCNData'></span><span id='topic+as.PairedPSCNData.PairedPSCNData'></span><span id='topic+as.PairedPSCNData.data.frame'></span><span id='topic+callNaiveGenotypes.NonPairedPSCNData'></span><span id='topic+callNaiveGenotypes.PairedPSCNData'></span><span id='topic+callSNPs'></span><span id='topic+callSNPs.AbstractPSCNData'></span><span id='topic+callSegmentationOutliers'></span><span id='topic+callSegmentationOutliers.NonPairedPSCNData'></span><span id='topic+callSegmentationOutliers.PairedPSCNData'></span><span id='topic+dropSegmentationOutliers'></span><span id='topic+dropSegmentationOutliers.NonPairedPSCNData'></span><span id='topic+dropSegmentationOutliers.PairedPSCNData'></span><span id='topic+extractNonPairedPSCNData'></span><span id='topic+extractNonPairedPSCNData.PairedPSCNData'></span><span id='topic+findLargeGaps'></span><span id='topic+findLargeGaps.AbstractCNData'></span><span id='topic+getChipType.AbstractCNData'></span><span id='topic+getLocusData'></span><span id='topic+getLocusData.AbstractCNData'></span><span id='topic+getPlatform.AbstractCNData'></span><span id='topic+getSNPFields'></span><span id='topic+getSNPFields.AbstractPSCNData'></span><span id='topic+getSignalColumnNames.NonPairedPSCNData'></span><span id='topic+getSignalColumnNames.PairedPSCNData'></span><span id='topic+getTCNs'></span><span id='topic+getTCNs.PairedPSCNData'></span><span id='topic+getTotalCopyNumbers'></span><span id='topic+getTotalCopyNumbers.PairedPSCNData'></span><span id='topic+hasKnownPositions'></span><span id='topic+hasKnownPositions.AbstractCNData'></span><span id='topic+normalizeTumorBoost.PairedPSCNData'></span><span id='topic+orderAlongGenome'></span><span id='topic+orderAlongGenome.AbstractCNData'></span><span id='topic+segmentByCBS.NonPairedPSCNData'></span><span id='topic+segmentByCBS.PairedPSCNData'></span><span id='topic+segmentByPairedPSCBS'></span><span id='topic+segmentByPairedPSCBS.PairedPSCNData'></span><span id='topic+setChipType'></span><span id='topic+setChipType.AbstractCNData'></span><span id='topic+setPlatform'></span><span id='topic+setPlatform.AbstractCNData'></span><span id='topic+normalizeTumorBoost'></span><span id='topic+callNaiveGenotypes'></span><span id='topic+getNumberOfFilesAveraged.AromaUnitSignalBinaryFile'></span><span id='topic+allocate.AromaUnitPscnBinaryFile'></span><span id='topic+byName.AromaUnitPscnBinarySet'></span><span id='topic+extractRawCopyNumbers.AromaUnitPscnBinaryFile'></span><span id='topic+getAverageFile.AromaUnitPscnBinarySet'></span><span id='topic+getColumnNames.AromaUnitPscnBinaryFile'></span><span id='topic+hasAlleleBFractions.AromaUnitPscnBinaryFile'></span><span id='topic+hasStrandiness.AromaUnitPscnBinaryFile'></span><span id='topic+hasTotalCopyNumberRatios'></span><span id='topic+hasTotalCopyNumberRatios.AromaUnitPscnBinaryFile'></span><span id='topic+writeDataFrame.AromaUnitPscnBinarySet'></span><span id='topic+downloadAll'></span><span id='topic+downloadAll.AromaRepository'></span><span id='topic+listFiles'></span><span id='topic+clearCache.AromaRepository'></span><span id='topic+findAnnotationDataByChipType.AromaRepository'></span><span id='topic+downloadUGC'></span><span id='topic+downloadUGC.AromaRepository'></span><span id='topic+AromaUnitGcContentFile'></span><span id='topic+getColumnNames.AromaUnitGcContentFile'></span><span id='topic+getExtensionPattern.AromaUnitGcContentFile'></span><span id='topic+getFilenameExtension.AromaUnitGcContentFile'></span><span id='topic+AromaUgcFile'></span><span id='topic+getReference'></span><span id='topic+getReference.CopyNumberChromosomalModel'></span><span id='topic+setReference'></span><span id='topic+setReference.CopyNumberChromosomalModel'></span><span id='topic+getMaxNAFraction'></span><span id='topic+getMaxNAFraction.CopyNumberChromosomalModel'></span><span id='topic+getDefaultColumnNames.AromaCellCpgFile'></span><span id='topic+getDefaultColumnNames.AromaCellPositionFile'></span><span id='topic+getDefaultColumnNames.AromaCellSequenceFile'></span><span id='topic+getDefaultColumnNames.AromaTabularBinaryFile'></span><span id='topic+getDefaultColumnNames.AromaUflFile'></span><span id='topic+getDefaultColumnNames.AromaUgpFile'></span><span id='topic+getDefaultColumnNames.AromaUnitChromosomeTabularBinaryFile'></span><span id='topic+getDefaultColumnNames.AromaUnitGcContentFile'></span><span id='topic+getDefaultColumnNames.AromaUnitPscnBinaryFile'></span><span id='topic+getParameterSets'></span><span id='topic+getParameterSets.ParametersInterface'></span><span id='topic+getCacheKey'></span><span id='topic+findFilesTodo'></span><span id='topic+getParameters'></span><span id='topic+write'></span><span id='topic+write.default'></span><span id='topic+setChromosomes'></span><span id='topic+getAlias'></span><span id='topic+setAlias'></span><span id='topic+process'></span><span id='topic+findChangePointsByState'></span><span id='topic+findChangePointsByState.SegmentedGenomicSignalsInterface'></span><span id='topic+as.character.AromaPlatform'></span><span id='topic+equals.AromaPlatform'></span><span id='topic+getUnitNames'></span>

<h3>Description</h3>

<p>This page contains aliases for all &quot;non-documented&quot; objects that
<code>R CMD check</code> detects in this package.
</p>
<p>Almost all of them are <em>generic</em> functions that have specific
document for the corresponding method coupled to a specific class.
Other functions are re-defined by <code>setMethodS3()</code> to
<em>default</em> methods. Neither of these two classes are non-documented
in reality.
The rest are deprecated methods.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='NonPairedPSCNData'>The NonPairedPSCNData class</h2><span id='topic+NonPairedPSCNData'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class NonPairedPSCNData</b><br />
</p>
<p><code>data.frame</code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+RichDataFrame">RichDataFrame</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+RawGenomicSignals">RawGenomicSignals</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+AbstractCNData">AbstractCNData</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AbstractPSCNData">AbstractPSCNData</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code>NonPairedPSCNData</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public class <b>NonPairedPSCNData</b><br />
extends <a href="#topic+AbstractPSCNData">AbstractPSCNData</a><br />
</p>
<p>A NonPairedPSCNData object holds parent-specific copy number data.
Two NonPairedPSCNData objects for a matched tumor-normal pair can
be combined into a <code><a href="#topic+PairedPSCNData">PairedPSCNData</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NonPairedPSCNData(chromosome=NULL, x=NULL, isSNP=NULL, mu=NULL, C=NULL, beta=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="NonPairedPSCNData_+3A_c">C</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of J tumor total copy number (TCN)
ratios in [0,+<code><a href="base.html#topic+is.finite">Inf</a></code>) (due to noise, small negative values are
also allowed).  The TCN ratios are typically scaled such that
copy-neutral diploid loci have a mean of two.</p>
</td></tr>
<tr><td><code id="NonPairedPSCNData_+3A_beta">beta</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of J tumor allele B fractions (BAFs)
in [0,1] (due to noise, values may be slightly outside as well)
or <code><a href="base.html#topic+NA">NA</a></code> for non-polymorphic loci.</p>
</td></tr>
<tr><td><code id="NonPairedPSCNData_+3A_mu">mu</code></td>
<td>
<p>An optional <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of J genotype calls in
{0,1/2,1} for AA, AB, and BB, respectively,
and <code><a href="base.html#topic+NA">NA</a></code> for non-polymorphic loci.
If not given, they are estimated from the normal BAFs using
<code><a href="aroma.light.html#topic+callNaiveGenotypes">callNaiveGenotypes</a></code> as described in [2].</p>
</td></tr>
<tr><td><code id="NonPairedPSCNData_+3A_issnp">isSNP</code></td>
<td>
<p>An optional <code><a href="base.html#topic+logical">logical</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length J specifying
whether each locus is a SNP or not (non-polymorphic loci).</p>
</td></tr>
<tr><td><code id="NonPairedPSCNData_+3A_chromosome">chromosome</code></td>
<td>
<p>(Optional) An <code><a href="base.html#topic+integer">integer</a></code> scalar (or a <code><a href="base.html#topic+vector">vector</a></code> of length J),
which can be used to specify which chromosome each locus belongs to
in case multiple chromosomes are segments.
This argument is also used for annotation purposes.</p>
</td></tr>
<tr><td><code id="NonPairedPSCNData_+3A_x">x</code></td>
<td>
<p>Optional <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of J genomic locations.
If <code><a href="base.html#topic+NULL">NULL</a></code>, index locations <code>1:J</code> are used.</p>
</td></tr>
<tr><td><code id="NonPairedPSCNData_+3A_...">...</code></td>
<td>
<p>Optional named locus-specific signal <code><a href="base.html#topic+vector">vector</a></code>s of length J.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as.NonPairedPSCNData</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as.PairedPSCNData</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>callNaiveGenotypes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>callSegmentationOutliers</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>dropSegmentationOutliers</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getSignalColumnNames</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>segmentByCBS</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AbstractPSCNData</b>:<br />
callSNPs, getSNPFields
</p>
<p><b>Methods inherited from AbstractCNData</b>:<br />
findLargeGaps, getChipType, getLocusData, getPlatform, hasKnownPositions, orderAlongGenome, setChipType, setPlatform
</p>
<p><b>Methods inherited from RawGenomicSignals</b>:<br />
*, +, -, addBy, append, applyBinaryOperator, as.character, as.data.frame, assertOneChromosome, binnedSmoothing, binnedSmoothingByField, clearCache, clone, divideBy, drawDensity, estimateStandardDeviation, extractChromosome, extractChromosomes, extractDataForSegmentation, extractRegion, extractRegions, extractSubset, gaussianSmoothing, getBasicField, getCXY, getChromosome, getChromosomes, getDefaultLocusFields, getLocusFields, getPositions, getSigma, getSignalColumnName, getSignalColumnNames, getSignals, getWeights, getXScale, getXY, getYScale, hasWeights, kernelSmoothing, lines, multiplyBy, nbrOfChromosomes, nbrOfLoci, plot, points, print, segmentByCBS, segmentByGLAD, segmentByHaarSeg, segmentByMPCBS, setBasicField, setSigma, setSignals, setWeights, setXScale, setYScale, signalRange, sort, subtractBy, xMax, xMin, xRange, xSeq, yMax, yMin, yRange
</p>
<p><b>Methods inherited from RichDataFrame</b>:<br />
$, $&lt;-, [, [[, [[&lt;-, as.data.frame, as.list, dim, dropVirtualColumn, getColumnNames, getColumnNamesTranslator, getFullName, getName, getTags, getVirtualColumn, getVirtualColumnFunction, getVirtualColumnNames, hasColumn, hasColumns, hasVirtualColumn, hasVirtualColumns, length, names, newInstance, print, rbind, setAttributes, setColumnNamesMap, setColumnNamesTranslator, setName, setTags, setVirtualColumn, subset, translateColumnNames
</p>
<p><b>Methods inherited from data.frame</b>:<br />
$&lt;-,data.frame-method, $&lt;-, Math, Ops,nonStructure,vector-method, Ops,structure,vector-method, Ops,vector,nonStructure-method, Ops,vector,structure-method, Ops, Summary, [, [&lt;-,data.frame-method, [&lt;-, [[, [[&lt;-,data.frame-method, [[&lt;-, aggregate, anyDuplicated, anyNA, as.NonPairedPSCNData, as.PairedPSCNData, as.data.frame, as.list, as.matrix, as.vector, attachLocally, by, callSegmentationOutliers, cbind, coerce,ANY,list-method, coerce,oldClass,S3-method, dim, dimnames, dimnames&lt;-, dropSegmentationOutliers, droplevels, duplicated, edit, findLargeGaps, format, formula, head, initialize,oldClass-method, is.na, merge, na.exclude, na.omit, plot, plotDensity, print, prompt, rbind, row.names, row.names&lt;-, rowsum, segmentByCBS, segmentByPairedPSCBS, show,oldClass-method, slotsFromS3,data.frame-method, split, split&lt;-, stack, str, subset, summary, t, tail, transform, type.convert, unique, unstack, unwrap, within, wrap, writeDataFrame, xtfrm
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='PairedPSCNData'>The PairedPSCNData class</h2><span id='topic+PairedPSCNData'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class PairedPSCNData</b><br />
</p>
<p><code>data.frame</code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+RichDataFrame">RichDataFrame</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+RawGenomicSignals">RawGenomicSignals</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+AbstractCNData">AbstractCNData</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AbstractPSCNData">AbstractPSCNData</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code>PairedPSCNData</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public class <b>PairedPSCNData</b><br />
extends <a href="#topic+AbstractPSCNData">AbstractPSCNData</a><br />
</p>
<p>A PairedPSCNData object holds paired tumor-normal parent-specific
copy number data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PairedPSCNData(chromosome=NULL, x=NULL, isSNP=NULL, muN=NULL, CT=NULL, betaT=NULL,
  CN=NULL, betaN=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PairedPSCNData_+3A_ct">CT</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of J tumor total copy number (TCN)
ratios in [0,+<code><a href="base.html#topic+is.finite">Inf</a></code>) (due to noise, small negative values are
also allowed).  The TCN ratios are typically scaled such that
copy-neutral diploid loci have a mean of two.</p>
</td></tr>
<tr><td><code id="PairedPSCNData_+3A_cn">CN</code></td>
<td>
<p>An optional <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of J normal TCN ratios.</p>
</td></tr>
<tr><td><code id="PairedPSCNData_+3A_betat">betaT</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of J tumor allele B fractions (BAFs)
in [0,1] (due to noise, values may be slightly outside as well)
or <code><a href="base.html#topic+NA">NA</a></code> for non-polymorphic loci.</p>
</td></tr>
<tr><td><code id="PairedPSCNData_+3A_betan">betaN</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of J matched normal BAFs in [0,1]
(due to noise, values may be slightly outside as well) or <code><a href="base.html#topic+NA">NA</a></code>
for non-polymorphic loci.</p>
</td></tr>
<tr><td><code id="PairedPSCNData_+3A_mun">muN</code></td>
<td>
<p>An optional <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of J genotype calls in
{0,1/2,1} for AA, AB, and BB, respectively,
and <code><a href="base.html#topic+NA">NA</a></code> for non-polymorphic loci.
If not given, they are estimated from the normal BAFs using
<code><a href="aroma.light.html#topic+callNaiveGenotypes">callNaiveGenotypes</a></code> as described in [2].</p>
</td></tr>
<tr><td><code id="PairedPSCNData_+3A_issnp">isSNP</code></td>
<td>
<p>An optional <code><a href="base.html#topic+logical">logical</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length J specifying
whether each locus is a SNP or not (non-polymorphic loci).</p>
</td></tr>
<tr><td><code id="PairedPSCNData_+3A_chromosome">chromosome</code></td>
<td>
<p>(Optional) An <code><a href="base.html#topic+integer">integer</a></code> scalar (or a <code><a href="base.html#topic+vector">vector</a></code> of length J),
which can be used to specify which chromosome each locus belongs to
in case multiple chromosomes are segments.
This argument is also used for annotation purposes.</p>
</td></tr>
<tr><td><code id="PairedPSCNData_+3A_x">x</code></td>
<td>
<p>Optional <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of J genomic locations.
If <code><a href="base.html#topic+NULL">NULL</a></code>, index locations <code>1:J</code> are used.</p>
</td></tr>
<tr><td><code id="PairedPSCNData_+3A_...">...</code></td>
<td>
<p>Optional named locus-specific signal <code><a href="base.html#topic+vector">vector</a></code>s of length J.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as.PairedPSCNData</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>callNaiveGenotypes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>callSegmentationOutliers</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>dropSegmentationOutliers</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getSignalColumnNames</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getTCNs</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getTotalCopyNumbers</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>normalizeTumorBoost</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plotTracks</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>segmentByCBS</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>segmentByPairedPSCBS</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AbstractPSCNData</b>:<br />
callSNPs, getSNPFields
</p>
<p><b>Methods inherited from AbstractCNData</b>:<br />
findLargeGaps, getChipType, getLocusData, getPlatform, hasKnownPositions, orderAlongGenome, setChipType, setPlatform
</p>
<p><b>Methods inherited from RawGenomicSignals</b>:<br />
*, +, -, addBy, append, applyBinaryOperator, as.character, as.data.frame, assertOneChromosome, binnedSmoothing, binnedSmoothingByField, clearCache, clone, divideBy, drawDensity, estimateStandardDeviation, extractChromosome, extractChromosomes, extractDataForSegmentation, extractRegion, extractRegions, extractSubset, gaussianSmoothing, getBasicField, getCXY, getChromosome, getChromosomes, getDefaultLocusFields, getLocusFields, getPositions, getSigma, getSignalColumnName, getSignalColumnNames, getSignals, getWeights, getXScale, getXY, getYScale, hasWeights, kernelSmoothing, lines, multiplyBy, nbrOfChromosomes, nbrOfLoci, plot, points, print, segmentByCBS, segmentByGLAD, segmentByHaarSeg, segmentByMPCBS, setBasicField, setSigma, setSignals, setWeights, setXScale, setYScale, signalRange, sort, subtractBy, xMax, xMin, xRange, xSeq, yMax, yMin, yRange
</p>
<p><b>Methods inherited from RichDataFrame</b>:<br />
$, $&lt;-, [, [[, [[&lt;-, as.data.frame, as.list, dim, dropVirtualColumn, getColumnNames, getColumnNamesTranslator, getFullName, getName, getTags, getVirtualColumn, getVirtualColumnFunction, getVirtualColumnNames, hasColumn, hasColumns, hasVirtualColumn, hasVirtualColumns, length, names, newInstance, print, rbind, setAttributes, setColumnNamesMap, setColumnNamesTranslator, setName, setTags, setVirtualColumn, subset, translateColumnNames
</p>
<p><b>Methods inherited from data.frame</b>:<br />
$&lt;-,data.frame-method, $&lt;-, Math, Ops,nonStructure,vector-method, Ops,structure,vector-method, Ops,vector,nonStructure-method, Ops,vector,structure-method, Ops, Summary, [, [&lt;-,data.frame-method, [&lt;-, [[, [[&lt;-,data.frame-method, [[&lt;-, aggregate, anyDuplicated, anyNA, as.NonPairedPSCNData, as.PairedPSCNData, as.data.frame, as.list, as.matrix, as.vector, attachLocally, by, callSegmentationOutliers, cbind, coerce,ANY,list-method, coerce,oldClass,S3-method, dim, dimnames, dimnames&lt;-, dropSegmentationOutliers, droplevels, duplicated, edit, findLargeGaps, format, formula, head, initialize,oldClass-method, is.na, merge, na.exclude, na.omit, plot, plotDensity, print, prompt, rbind, row.names, row.names&lt;-, rowsum, segmentByCBS, segmentByPairedPSCBS, show,oldClass-method, slotsFromS3,data.frame-method, split, split&lt;-, stack, str, subset, summary, t, tail, transform, type.convert, unique, unstack, unwrap, within, wrap, writeDataFrame, xtfrm
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='ParametersInterface'>The ParametersInterface class interface</h2><span id='topic+ParametersInterface'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class ParametersInterface</b><br />
</p>
<p><code><a href="R.oo.html#topic+Interface">Interface</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code>ParametersInterface</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AromaTransform">AromaTransform</a></em><br />
</p>
<p>public class <b>ParametersInterface</b><br />
extends <a href="R.oo.html#topic+Interface">Interface</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ParametersInterface(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ParametersInterface_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getParameters</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getParametersAsString</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from Interface</b>:<br />
extend, print, uses
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='plotTracks.PairedPSCNData'>Plots parental specific copy numbers along the genome</h2><span id='topic+plotTracks.PairedPSCNData'></span><span id='topic+PairedPSCNData.plotTracks'></span><span id='topic+plotTracks+2CPairedPSCNData-method'></span><span id='topic+plotTracks'></span>

<h3>Description</h3>

<p>Plots parental specific copy numbers along the genome for one or more chromosomes.
It is possible to specify what type of tracks to plot.
Each type of track is plotted in its own panel.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PairedPSCNData'
plotTracks(x, tracks=c("tcn", "dh", "tcn,c1,c2", "tcn,c1", "tcn,c2", "c1,c2", "betaN",
  "betaT", "betaTN")[1:3], pch=".", col=NULL, cex=1, grid=FALSE, xlim=NULL, Clim=c(0, 6),
  Blim=c(0, 1), xScale=1e-06, ..., add=FALSE, subplots=!add &amp;&amp; (length(tracks) &gt; 1),
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotTracks.PairedPSCNData_+3A_tracks">tracks</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying what types of tracks to plot.</p>
</td></tr>
<tr><td><code id="plotTracks.PairedPSCNData_+3A_pch">pch</code></td>
<td>
<p>The type of the scatter points, if any.</p>
</td></tr>
<tr><td><code id="plotTracks.PairedPSCNData_+3A_col">col</code></td>
<td>
<p>The color of the scatter points, if any.</p>
</td></tr>
<tr><td><code id="plotTracks.PairedPSCNData_+3A_cex">cex</code></td>
<td>
<p>The size of the scatter points, if any.</p>
</td></tr>
<tr><td><code id="plotTracks.PairedPSCNData_+3A_grid">grid</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, horizontal lines are displayed.</p>
</td></tr>
<tr><td><code id="plotTracks.PairedPSCNData_+3A_xlim">xlim</code></td>
<td>
<p>(Optional) The genomic range to plot.</p>
</td></tr>
<tr><td><code id="plotTracks.PairedPSCNData_+3A_clim">Clim</code></td>
<td>
<p>The range of copy numbers.</p>
</td></tr>
<tr><td><code id="plotTracks.PairedPSCNData_+3A_blim">Blim</code></td>
<td>
<p>The range of allele B fractions (BAFs) and
decrease of heterozygosity (DHs).</p>
</td></tr>
<tr><td><code id="plotTracks.PairedPSCNData_+3A_xscale">xScale</code></td>
<td>
<p>The scale factor used for genomic positions.</p>
</td></tr>
<tr><td><code id="plotTracks.PairedPSCNData_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="plotTracks.PairedPSCNData_+3A_add">add</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the panels plotted are added to the existing plot,
otherwise a new plot is created.</p>
</td></tr>
<tr><td><code id="plotTracks.PairedPSCNData_+3A_subplots">subplots</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, then subplots are automatically setup.</p>
</td></tr>
<tr><td><code id="plotTracks.PairedPSCNData_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns nothing.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='process.AromaTransform'>Processes the data set</h2><span id='topic+process.AromaTransform'></span><span id='topic+AromaTransform.process'></span><span id='topic+process+2CAromaTransform-method'></span>

<h3>Description</h3>

<p>Processes the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaTransform'
process(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="process.AromaTransform_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AromaTransform">AromaTransform</a></code>.
</p>

<hr>
<h2 id='process.ChromosomeExplorer'>Generates image files, scripts and dynamic pages for the explorer</h2><span id='topic+process.ChromosomeExplorer'></span><span id='topic+ChromosomeExplorer.process'></span><span id='topic+process+2CChromosomeExplorer-method'></span>

<h3>Description</h3>

<p>Generates image files, scripts and dynamic pages for the explorer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ChromosomeExplorer'
process(this, arrays=NULL, chromosomes=NULL, ..., zooms=getZooms(this), layers=FALSE,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="process.ChromosomeExplorer_+3A_arrays">arrays</code></td>
<td>
<p>A <code><a href="base.html#topic+vector">vector</a></code> of arrays specifying which arrays to
be considered.  If <code><a href="base.html#topic+NULL">NULL</a></code>, all are processed.</p>
</td></tr>
<tr><td><code id="process.ChromosomeExplorer_+3A_chromosomes">chromosomes</code></td>
<td>
<p>A <code><a href="base.html#topic+vector">vector</a></code> of chromosomes specifying which
chromosomes to be considered.  If <code><a href="base.html#topic+NULL">NULL</a></code>, all are processed.</p>
</td></tr>
<tr><td><code id="process.ChromosomeExplorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="process.ChromosomeExplorer_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns nothing.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ChromosomeExplorer">ChromosomeExplorer</a></code>.
</p>

<hr>
<h2 id='process.Explorer'>Generates image files, scripts and dynamic pages for the explorer</h2><span id='topic+process.Explorer'></span><span id='topic+Explorer.process'></span><span id='topic+process+2CExplorer-method'></span>

<h3>Description</h3>

<p>Generates image files, scripts and dynamic pages for the explorer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Explorer'
process(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="process.Explorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns nothing.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+Explorer">Explorer</a></code>.
</p>

<hr>
<h2 id='processTime'>Gets the running time of the R process and its children processes</h2><span id='topic+processTime.default'></span><span id='topic+processTime'></span>

<h3>Description</h3>

<p>Gets the running time of the R process and its children processes.
This function is a safe wrapper for <code><a href="base.html#topic+proc.time">proc.time</a></code>(), which might
not exist on all platforms.  It &quot;determines how much time (in seconds)
the currently running <span class="rlang"><b>R</b></span> process already consumed&quot;.  In addition it adds
descriptive names of the returned values.
For more details, see <code><a href="base.html#topic+proc.time">proc.time</a></code>().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
processTime(since=NULL, units=c("seconds", "milliseconds", "minutes", "hours", "days"),
  fmtstr=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="processTime_+3A_since">since</code></td>
<td>
<p>An optional <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> to be subtracted from the value
of <code><a href="base.html#topic+proc.time">proc.time</a></code>().  This is useful for calculating &quot;lap times&quot;.</p>
</td></tr>
<tr><td><code id="processTime_+3A_units">units</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying the unit of the
returned values.</p>
</td></tr>
<tr><td><code id="processTime_+3A_fmtstr">fmtstr</code></td>
<td>
<p>If given, a format string to convert the times to strings
via <code><a href="base.html#topic+sprintf">sprintf</a></code>().</p>
</td></tr>
<tr><td><code id="processTime_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length 5.
For more details, see <code><a href="base.html#topic+proc.time">proc.time</a></code>().
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+proc.time">proc.time</a></code>().
<code><a href="base.html#topic+system.time">system.time</a></code>().
<code><a href="base.html#topic+gc.time">gc.time</a></code>().
</p>

<hr>
<h2 id='RawAlleleBFractions'>The RawAlleleBFractions class</h2><span id='topic+RawAlleleBFractions'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class RawAlleleBFractions</b><br />
</p>
<p><code>data.frame</code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+RichDataFrame">RichDataFrame</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+RawGenomicSignals">RawGenomicSignals</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code>RawAlleleBFractions</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+RawMirroredAlleleBFractions">RawMirroredAlleleBFractions</a>, <a href="#topic+SegmentedAlleleBFractions">SegmentedAlleleBFractions</a><br />
</p>
<p>public class <b>RawAlleleBFractions</b><br />
extends <a href="#topic+RawGenomicSignals">RawGenomicSignals</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RawAlleleBFractions(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RawAlleleBFractions_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+RawGenomicSignals">RawGenomicSignals</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractRawMirroredAlleleBFractions</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plot</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from RawGenomicSignals</b>:<br />
*, +, -, addBy, append, applyBinaryOperator, as.character, as.data.frame, assertOneChromosome, binnedSmoothing, binnedSmoothingByField, clearCache, clone, divideBy, drawDensity, estimateStandardDeviation, extractChromosome, extractChromosomes, extractDataForSegmentation, extractRegion, extractRegions, extractSubset, gaussianSmoothing, getBasicField, getCXY, getChromosome, getChromosomes, getDefaultLocusFields, getLocusFields, getPositions, getSigma, getSignalColumnName, getSignalColumnNames, getSignals, getWeights, getXScale, getXY, getYScale, hasWeights, kernelSmoothing, lines, multiplyBy, nbrOfChromosomes, nbrOfLoci, plot, points, print, segmentByCBS, segmentByGLAD, segmentByHaarSeg, segmentByMPCBS, setBasicField, setSigma, setSignals, setWeights, setXScale, setYScale, signalRange, sort, subtractBy, xMax, xMin, xRange, xSeq, yMax, yMin, yRange
</p>
<p><b>Methods inherited from RichDataFrame</b>:<br />
$, $&lt;-, [, [[, [[&lt;-, as.data.frame, as.list, dim, dropVirtualColumn, getColumnNames, getColumnNamesTranslator, getFullName, getName, getTags, getVirtualColumn, getVirtualColumnFunction, getVirtualColumnNames, hasColumn, hasColumns, hasVirtualColumn, hasVirtualColumns, length, names, newInstance, print, rbind, setAttributes, setColumnNamesMap, setColumnNamesTranslator, setName, setTags, setVirtualColumn, subset, translateColumnNames
</p>
<p><b>Methods inherited from data.frame</b>:<br />
$&lt;-,data.frame-method, $&lt;-, Math, Ops,nonStructure,vector-method, Ops,structure,vector-method, Ops,vector,nonStructure-method, Ops,vector,structure-method, Ops, Summary, [, [&lt;-,data.frame-method, [&lt;-, [[, [[&lt;-,data.frame-method, [[&lt;-, aggregate, anyDuplicated, anyNA, as.NonPairedPSCNData, as.PairedPSCNData, as.data.frame, as.list, as.matrix, as.vector, attachLocally, by, callSegmentationOutliers, cbind, coerce,ANY,list-method, coerce,oldClass,S3-method, dim, dimnames, dimnames&lt;-, dropSegmentationOutliers, droplevels, duplicated, edit, findLargeGaps, format, formula, head, initialize,oldClass-method, is.na, merge, na.exclude, na.omit, plot, plotDensity, print, prompt, rbind, row.names, row.names&lt;-, rowsum, segmentByCBS, segmentByPairedPSCBS, show,oldClass-method, slotsFromS3,data.frame-method, split, split&lt;-, stack, str, subset, summary, t, tail, transform, type.convert, unique, unstack, unwrap, within, wrap, writeDataFrame, xtfrm
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='RawCopyNumberModel'>The RawCopyNumberModel class</h2><span id='topic+RawCopyNumberModel'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class RawCopyNumberModel</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+ChromosomalModel">ChromosomalModel</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+CopyNumberChromosomalModel">CopyNumberChromosomalModel</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><em><code>RawCopyNumberModel</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>RawCopyNumberModel</b><br />
extends <em><a href="#topic+CopyNumberChromosomalModel">CopyNumberChromosomalModel</a></em><br />
</p>
<p>This class represents an identity copy-number model which returns
the input as is.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RawCopyNumberModel(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RawCopyNumberModel_+3A_...">...</code></td>
<td>
<p>Passed to the constructor of the superclass.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from CopyNumberChromosomalModel</b>:<br />
as.character, calculateChromosomeStatistics, calculateRatios, estimateSds, extractRawCopyNumbers, fit, getChromosomeLength, getDataFileMatrix, getMaxNAFraction, getNames, getOptionalArguments, getPairedNames, getRefSetTuple, getReference, getReferenceSetTuple, isPaired, newPlot, plotAxesLayers, plotChromosomesLayers, plotCytobandLayers, plotFitLayers, plotGridHorizontalLayers, plotRawCopyNumbers, plotSampleLayers, setReference
</p>
<p><b>Methods inherited from ChromosomalModel</b>:<br />
as.character, fit, getAlias, getAromaGenomeTextFile, getAsteriskTags, getChipType, getChipTypes, getChromosomes, getFullName, getFullNames, getGenome, getGenomeData, getGenomeFile, getListOfAromaUgpFiles, getName, getNames, getParentPath, getPath, getReportPath, getRootPath, getSetTuple, getSets, getTags, indexOf, nbrOfArrays, nbrOfChipTypes, setChromosomes, setGenome
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='RawCopyNumbers'>The RawCopyNumbers class</h2><span id='topic+RawCopyNumbers'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class RawCopyNumbers</b><br />
</p>
<p><code>data.frame</code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+RichDataFrame">RichDataFrame</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+RawGenomicSignals">RawGenomicSignals</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code>RawCopyNumbers</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+SegmentedCopyNumbers">SegmentedCopyNumbers</a><br />
</p>
<p>public class <b>RawCopyNumbers</b><br />
extends <a href="#topic+RawGenomicSignals">RawGenomicSignals</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RawCopyNumbers(cn=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RawCopyNumbers_+3A_cn">cn</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length J specifying the copy number
at each loci.</p>
</td></tr>
<tr><td><code id="RawCopyNumbers_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+RawGenomicSignals">RawGenomicSignals</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>cnRange</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractRawCopyNumbers</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getSignals</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plot</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from RawGenomicSignals</b>:<br />
*, +, -, addBy, append, applyBinaryOperator, as.character, as.data.frame, assertOneChromosome, binnedSmoothing, binnedSmoothingByField, clearCache, clone, divideBy, drawDensity, estimateStandardDeviation, extractChromosome, extractChromosomes, extractDataForSegmentation, extractRegion, extractRegions, extractSubset, gaussianSmoothing, getBasicField, getCXY, getChromosome, getChromosomes, getDefaultLocusFields, getLocusFields, getPositions, getSigma, getSignalColumnName, getSignalColumnNames, getSignals, getWeights, getXScale, getXY, getYScale, hasWeights, kernelSmoothing, lines, multiplyBy, nbrOfChromosomes, nbrOfLoci, plot, points, print, segmentByCBS, segmentByGLAD, segmentByHaarSeg, segmentByMPCBS, setBasicField, setSigma, setSignals, setWeights, setXScale, setYScale, signalRange, sort, subtractBy, xMax, xMin, xRange, xSeq, yMax, yMin, yRange
</p>
<p><b>Methods inherited from RichDataFrame</b>:<br />
$, $&lt;-, [, [[, [[&lt;-, as.data.frame, as.list, dim, dropVirtualColumn, getColumnNames, getColumnNamesTranslator, getFullName, getName, getTags, getVirtualColumn, getVirtualColumnFunction, getVirtualColumnNames, hasColumn, hasColumns, hasVirtualColumn, hasVirtualColumns, length, names, newInstance, print, rbind, setAttributes, setColumnNamesMap, setColumnNamesTranslator, setName, setTags, setVirtualColumn, subset, translateColumnNames
</p>
<p><b>Methods inherited from data.frame</b>:<br />
$&lt;-,data.frame-method, $&lt;-, Math, Ops,nonStructure,vector-method, Ops,structure,vector-method, Ops,vector,nonStructure-method, Ops,vector,structure-method, Ops, Summary, [, [&lt;-,data.frame-method, [&lt;-, [[, [[&lt;-,data.frame-method, [[&lt;-, aggregate, anyDuplicated, anyNA, as.NonPairedPSCNData, as.PairedPSCNData, as.data.frame, as.list, as.matrix, as.vector, attachLocally, by, callSegmentationOutliers, cbind, coerce,ANY,list-method, coerce,oldClass,S3-method, dim, dimnames, dimnames&lt;-, dropSegmentationOutliers, droplevels, duplicated, edit, findLargeGaps, format, formula, head, initialize,oldClass-method, is.na, merge, na.exclude, na.omit, plot, plotDensity, print, prompt, rbind, row.names, row.names&lt;-, rowsum, segmentByCBS, segmentByPairedPSCBS, show,oldClass-method, slotsFromS3,data.frame-method, split, split&lt;-, stack, str, subset, summary, t, tail, transform, type.convert, unique, unstack, unwrap, within, wrap, writeDataFrame, xtfrm
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>Examples</h3>

<pre><code class='language-R'># - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Simulating copy-number data
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Number of loci
J &lt;- 1000

mu &lt;- double(J)
mu[200:300] &lt;- mu[200:300] + 1
mu[650:800] &lt;- mu[650:800] - 1
eps &lt;- rnorm(J, sd=1/2)
y &lt;- mu + eps
x &lt;- sort(runif(length(y), max=length(y)))


cn &lt;- RawCopyNumbers(y, x)
print(cn)

cn2 &lt;- extractSubset(cn, subset=xSeq(cn, by=5))
print(cn2)

# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Plot along genome
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
plot(cn, ylim=c(-3,3))
title(main="Complete and subsetted loci")
points(cn2, col="red", pch=176, cex=2)

legend("topright", pch=c(19,176), col=c("#999999", "red"),
       sprintf(c("raw [n=%d]", "every 5th [n=%d]"),
               c(nbrOfLoci(cn), nbrOfLoci(cn2))), bty="n")


# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Binned smoothing
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
plot(cn, col="#999999", ylim=c(-3,3))
title(main="Binned smoothing")

cnSa &lt;- binnedSmoothing(cn, by=3)
lines(cnSa, col="blue")
points(cnSa, col="blue")

cnSb &lt;- binnedSmoothing(cn, by=9)
lines(cnSb, col="red")
points(cnSb, col="red")

legend("topright", pch=19, col=c("#999999", "blue", "red"),
       sprintf(c("raw [n=%d]", "Bin(w=3) [n=%d]", "Bin(w=9) [n=%d]"),
       c(nbrOfLoci(cn), nbrOfLoci(cnSa), nbrOfLoci(cnSb))), bty="n")


# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Binned smoothing (by count)
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
plot(cn, col="#999999", ylim=c(-3,3))
title(main="Binned smoothing (by count)")

cnSa &lt;- binnedSmoothing(cn, by=3, byCount=TRUE)
lines(cnSa, col="blue")
points(cnSa, col="blue")

cnSb &lt;- binnedSmoothing(cn, by=9, byCount=TRUE)
lines(cnSb, col="red")
points(cnSb, col="red")

legend("topright", pch=19, col=c("#999999", "blue", "red"),
       sprintf(c("raw [n=%d]", "BinO(w=3) [n=%d]", "BinO(w=9) [n=%d]"),
       c(nbrOfLoci(cn), nbrOfLoci(cnSa), nbrOfLoci(cnSb))), bty="n")


# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Kernel smoothing (default is Gaussian)
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
plot(cn, col="#999999", ylim=c(-3,3))
title(main="Kernel smoothing w/ Gaussian kernel")

cnSa &lt;- kernelSmoothing(cn, h=2)
points(cnSa, col="blue")

cnSb &lt;- kernelSmoothing(cn, h=5)
points(cnSb, col="red")

legend("topright", pch=19, col=c("#999999", "blue", "red"),
       sprintf(c("raw [n=%d]", "N(.,sd=2) [n=%d]", "N(.,sd=5) [n=%d]"),
       c(nbrOfLoci(cn), nbrOfLoci(cnSa), nbrOfLoci(cnSb))), bty="n")


# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Kernel smoothing
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
plot(cn, col="#999999", ylim=c(-3,3))
title(main="Kernel smoothing w/ uniform kernel")

xOut &lt;- xSeq(cn, by=10)
cnSa &lt;- kernelSmoothing(cn, xOut=xOut, kernel="uniform", h=2)
lines(cnSa, col="blue")
points(cnSa, col="blue")

cnSb &lt;- kernelSmoothing(cn, xOut=xOut, kernel="uniform", h=5)
lines(cnSb, col="red")
points(cnSb, col="red")

legend("topright", pch=19, col=c("#999999", "blue", "red"),
       sprintf(c("raw [n=%d]", "U(w=2) [n=%d]", "U(w=5) [n=%d]"),
       c(nbrOfLoci(cn), nbrOfLoci(cnSa), nbrOfLoci(cnSb))), bty="n")
</code></pre>

<hr>
<h2 id='RawGenomicSignals'>The RawGenomicSignals class</h2><span id='topic+RawGenomicSignals'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class RawGenomicSignals</b><br />
</p>
<p><code>data.frame</code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+RichDataFrame">RichDataFrame</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code>RawGenomicSignals</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+AbstractCNData">AbstractCNData</a>, <a href="#topic+AbstractPSCNData">AbstractPSCNData</a>, <a href="#topic+NonPairedPSCNData">NonPairedPSCNData</a>, <a href="#topic+PairedPSCNData">PairedPSCNData</a>, <a href="#topic+RawAlleleBFractions">RawAlleleBFractions</a>, <a href="#topic+RawCopyNumbers">RawCopyNumbers</a>, <a href="#topic+RawMirroredAlleleBFractions">RawMirroredAlleleBFractions</a>, <a href="#topic+RawSequenceReads">RawSequenceReads</a>, <a href="#topic+SegmentedAlleleBFractions">SegmentedAlleleBFractions</a>, <a href="#topic+SegmentedCopyNumbers">SegmentedCopyNumbers</a><br />
</p>
<p>public class <b>RawGenomicSignals</b><br />
extends <a href="#topic+RichDataFrame">RichDataFrame</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RawGenomicSignals(y=NULL, x=NULL, w=NULL, chromosome=0L, name=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RawGenomicSignals_+3A_y">y</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length J specifying the signal
at each locus.</p>
</td></tr>
<tr><td><code id="RawGenomicSignals_+3A_x">x</code></td>
<td>
<p>A (optional) <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length J specifying the
position of each locus.</p>
</td></tr>
<tr><td><code id="RawGenomicSignals_+3A_w">w</code></td>
<td>
<p>A (optional) non-negative <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length J
specifying a weight of each locus.</p>
</td></tr>
<tr><td><code id="RawGenomicSignals_+3A_chromosome">chromosome</code></td>
<td>
<p>An (optional) <code><a href="base.html#topic+integer">integer</a></code> specifying the chromosome for
these genomic signals.</p>
</td></tr>
<tr><td><code id="RawGenomicSignals_+3A_name">name</code></td>
<td>
<p>An (optional) <code><a href="base.html#topic+character">character</a></code> string specifying the sample name.</p>
</td></tr>
<tr><td><code id="RawGenomicSignals_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>*</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>+</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>-</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>addBy</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>append</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as.data.frame</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>binnedSmoothing</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>divideBy</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>drawDensity</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>estimateStandardDeviation</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>gaussianSmoothing</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChromosome</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChromosomes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPositions</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getSigma</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getSignals</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getWeights</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getXScale</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getYScale</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>hasWeights</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>kernelSmoothing</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>lines</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>multiplyBy</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfChromosomes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfLoci</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plot</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>points</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>segmentByCBS</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>segmentByGLAD</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>segmentByHaarSeg</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>segmentByMPCBS</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setSigma</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setSignals</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setWeights</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setXScale</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setYScale</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>signalRange</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>sort</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>subtractBy</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>xMax</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>xMin</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>xRange</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>xSeq</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>yMax</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>yMin</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>yRange</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from RichDataFrame</b>:<br />
$, $&lt;-, [, [[, [[&lt;-, as.data.frame, as.list, dim, dropVirtualColumn, getColumnNames, getColumnNamesTranslator, getFullName, getName, getTags, getVirtualColumn, getVirtualColumnFunction, getVirtualColumnNames, hasColumn, hasColumns, hasVirtualColumn, hasVirtualColumns, length, names, newInstance, print, rbind, setAttributes, setColumnNamesMap, setColumnNamesTranslator, setName, setTags, setVirtualColumn, subset, translateColumnNames
</p>
<p><b>Methods inherited from data.frame</b>:<br />
$&lt;-,data.frame-method, $&lt;-, Math, Ops,nonStructure,vector-method, Ops,structure,vector-method, Ops,vector,nonStructure-method, Ops,vector,structure-method, Ops, Summary, [, [&lt;-,data.frame-method, [&lt;-, [[, [[&lt;-,data.frame-method, [[&lt;-, aggregate, anyDuplicated, anyNA, as.NonPairedPSCNData, as.PairedPSCNData, as.data.frame, as.list, as.matrix, as.vector, attachLocally, by, callSegmentationOutliers, cbind, coerce,ANY,list-method, coerce,oldClass,S3-method, dim, dimnames, dimnames&lt;-, dropSegmentationOutliers, droplevels, duplicated, edit, findLargeGaps, format, formula, head, initialize,oldClass-method, is.na, merge, na.exclude, na.omit, plot, plotDensity, print, prompt, rbind, row.names, row.names&lt;-, rowsum, segmentByCBS, segmentByPairedPSCBS, show,oldClass-method, slotsFromS3,data.frame-method, split, split&lt;-, stack, str, subset, summary, t, tail, transform, type.convert, unique, unstack, unwrap, within, wrap, writeDataFrame, xtfrm
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='RawMirroredAlleleBFractions'>The RawMirroredAlleleBFractions class</h2><span id='topic+RawMirroredAlleleBFractions'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class RawMirroredAlleleBFractions</b><br />
</p>
<p><code>data.frame</code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+RichDataFrame">RichDataFrame</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+RawGenomicSignals">RawGenomicSignals</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+RawAlleleBFractions">RawAlleleBFractions</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code>RawMirroredAlleleBFractions</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public class <b>RawMirroredAlleleBFractions</b><br />
extends <a href="#topic+RawAlleleBFractions">RawAlleleBFractions</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RawMirroredAlleleBFractions(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RawMirroredAlleleBFractions_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+RawAlleleBFractions">RawAlleleBFractions</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plot</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from RawAlleleBFractions</b>:<br />
extractRawMirroredAlleleBFractions, plot
</p>
<p><b>Methods inherited from RawGenomicSignals</b>:<br />
*, +, -, addBy, append, applyBinaryOperator, as.character, as.data.frame, assertOneChromosome, binnedSmoothing, binnedSmoothingByField, clearCache, clone, divideBy, drawDensity, estimateStandardDeviation, extractChromosome, extractChromosomes, extractDataForSegmentation, extractRegion, extractRegions, extractSubset, gaussianSmoothing, getBasicField, getCXY, getChromosome, getChromosomes, getDefaultLocusFields, getLocusFields, getPositions, getSigma, getSignalColumnName, getSignalColumnNames, getSignals, getWeights, getXScale, getXY, getYScale, hasWeights, kernelSmoothing, lines, multiplyBy, nbrOfChromosomes, nbrOfLoci, plot, points, print, segmentByCBS, segmentByGLAD, segmentByHaarSeg, segmentByMPCBS, setBasicField, setSigma, setSignals, setWeights, setXScale, setYScale, signalRange, sort, subtractBy, xMax, xMin, xRange, xSeq, yMax, yMin, yRange
</p>
<p><b>Methods inherited from RichDataFrame</b>:<br />
$, $&lt;-, [, [[, [[&lt;-, as.data.frame, as.list, dim, dropVirtualColumn, getColumnNames, getColumnNamesTranslator, getFullName, getName, getTags, getVirtualColumn, getVirtualColumnFunction, getVirtualColumnNames, hasColumn, hasColumns, hasVirtualColumn, hasVirtualColumns, length, names, newInstance, print, rbind, setAttributes, setColumnNamesMap, setColumnNamesTranslator, setName, setTags, setVirtualColumn, subset, translateColumnNames
</p>
<p><b>Methods inherited from data.frame</b>:<br />
$&lt;-,data.frame-method, $&lt;-, Math, Ops,nonStructure,vector-method, Ops,structure,vector-method, Ops,vector,nonStructure-method, Ops,vector,structure-method, Ops, Summary, [, [&lt;-,data.frame-method, [&lt;-, [[, [[&lt;-,data.frame-method, [[&lt;-, aggregate, anyDuplicated, anyNA, as.NonPairedPSCNData, as.PairedPSCNData, as.data.frame, as.list, as.matrix, as.vector, attachLocally, by, callSegmentationOutliers, cbind, coerce,ANY,list-method, coerce,oldClass,S3-method, dim, dimnames, dimnames&lt;-, dropSegmentationOutliers, droplevels, duplicated, edit, findLargeGaps, format, formula, head, initialize,oldClass-method, is.na, merge, na.exclude, na.omit, plot, plotDensity, print, prompt, rbind, row.names, row.names&lt;-, rowsum, segmentByCBS, segmentByPairedPSCBS, show,oldClass-method, slotsFromS3,data.frame-method, split, split&lt;-, stack, str, subset, summary, t, tail, transform, type.convert, unique, unstack, unwrap, within, wrap, writeDataFrame, xtfrm
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='RawSequenceReads'>The RawSequenceReads class</h2><span id='topic+RawSequenceReads'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class RawSequenceReads</b><br />
</p>
<p><code>data.frame</code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+RichDataFrame">RichDataFrame</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+RawGenomicSignals">RawGenomicSignals</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code>RawSequenceReads</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public class <b>RawSequenceReads</b><br />
extends <a href="#topic+RawGenomicSignals">RawGenomicSignals</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RawSequenceReads(x=NULL, y=rep(1L, length(x)), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RawSequenceReads_+3A_x">x</code></td>
<td>
<p>An <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length J specifying the read positions.</p>
</td></tr>
<tr><td><code id="RawSequenceReads_+3A_y">y</code></td>
<td>
<p>An (optional) <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length J specifying the number of reads at each position. Default is one read per position.</p>
</td></tr>
<tr><td><code id="RawSequenceReads_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+RawGenomicSignals">RawGenomicSignals</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>binnedSums</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractRawCopyNumbers</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfReads</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plot</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from RawGenomicSignals</b>:<br />
*, +, -, addBy, append, applyBinaryOperator, as.character, as.data.frame, assertOneChromosome, binnedSmoothing, binnedSmoothingByField, clearCache, clone, divideBy, drawDensity, estimateStandardDeviation, extractChromosome, extractChromosomes, extractDataForSegmentation, extractRegion, extractRegions, extractSubset, gaussianSmoothing, getBasicField, getCXY, getChromosome, getChromosomes, getDefaultLocusFields, getLocusFields, getPositions, getSigma, getSignalColumnName, getSignalColumnNames, getSignals, getWeights, getXScale, getXY, getYScale, hasWeights, kernelSmoothing, lines, multiplyBy, nbrOfChromosomes, nbrOfLoci, plot, points, print, segmentByCBS, segmentByGLAD, segmentByHaarSeg, segmentByMPCBS, setBasicField, setSigma, setSignals, setWeights, setXScale, setYScale, signalRange, sort, subtractBy, xMax, xMin, xRange, xSeq, yMax, yMin, yRange
</p>
<p><b>Methods inherited from RichDataFrame</b>:<br />
$, $&lt;-, [, [[, [[&lt;-, as.data.frame, as.list, dim, dropVirtualColumn, getColumnNames, getColumnNamesTranslator, getFullName, getName, getTags, getVirtualColumn, getVirtualColumnFunction, getVirtualColumnNames, hasColumn, hasColumns, hasVirtualColumn, hasVirtualColumns, length, names, newInstance, print, rbind, setAttributes, setColumnNamesMap, setColumnNamesTranslator, setName, setTags, setVirtualColumn, subset, translateColumnNames
</p>
<p><b>Methods inherited from data.frame</b>:<br />
$&lt;-,data.frame-method, $&lt;-, Math, Ops,nonStructure,vector-method, Ops,structure,vector-method, Ops,vector,nonStructure-method, Ops,vector,structure-method, Ops, Summary, [, [&lt;-,data.frame-method, [&lt;-, [[, [[&lt;-,data.frame-method, [[&lt;-, aggregate, anyDuplicated, anyNA, as.NonPairedPSCNData, as.PairedPSCNData, as.data.frame, as.list, as.matrix, as.vector, attachLocally, by, callSegmentationOutliers, cbind, coerce,ANY,list-method, coerce,oldClass,S3-method, dim, dimnames, dimnames&lt;-, dropSegmentationOutliers, droplevels, duplicated, edit, findLargeGaps, format, formula, head, initialize,oldClass-method, is.na, merge, na.exclude, na.omit, plot, plotDensity, print, prompt, rbind, row.names, row.names&lt;-, rowsum, segmentByCBS, segmentByPairedPSCBS, show,oldClass-method, slotsFromS3,data.frame-method, split, split&lt;-, stack, str, subset, summary, t, tail, transform, type.convert, unique, unstack, unwrap, within, wrap, writeDataFrame, xtfrm
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='readFooter.AromaTabularBinaryFile'>Reads the file footer in XML format into a named nested list</h2><span id='topic+readFooter.AromaTabularBinaryFile'></span><span id='topic+AromaTabularBinaryFile.readFooter'></span><span id='topic+readFooter+2CAromaTabularBinaryFile-method'></span>

<h3>Description</h3>

<p>Reads the file footer in XML format into a named nested list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaTabularBinaryFile'
readFooter(this, asXmlString=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readFooter.AromaTabularBinaryFile_+3A_asxmlstring">asXmlString</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the file footer is returned as
a <code><a href="base.html#topic+character">character</a></code> string.</p>
</td></tr>
<tr><td><code id="readFooter.AromaTabularBinaryFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named <code><a href="base.html#topic+list">list</a></code> structure (or a <code><a href="base.html#topic+character">character</a></code> string).
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+writeFooter.AromaTabularBinaryFile">*writeFooter</a>()</code>.
For more information see <code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code>.
</p>

<hr>
<h2 id='segmentByCBS.RawGenomicSignals'>Segment copy numbers using the CBS method</h2><span id='topic+segmentByCBS.RawGenomicSignals'></span><span id='topic+RawGenomicSignals.segmentByCBS'></span><span id='topic+segmentByCBS+2CRawGenomicSignals-method'></span>

<h3>Description</h3>

<p>Segment copy numbers using the CBS method of the <span class="pkg">DNAcopy</span> package.
For more details on the Circular Binary Segmentation (CBS) method
see [1,2].
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'RawGenomicSignals'
segmentByCBS(this, ..., seed=NULL, cache=FALSE, force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="segmentByCBS.RawGenomicSignals_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the segmentation function.</p>
</td></tr>
<tr><td><code id="segmentByCBS.RawGenomicSignals_+3A_seed">seed</code></td>
<td>
<p>An (optional) <code><a href="base.html#topic+integer">integer</a></code> specifying the random seed to be
set before calling the segmentation method.  The random seed is
set to its original state when exiting.  If <code><a href="base.html#topic+NULL">NULL</a></code>, it is not set.</p>
</td></tr>
<tr><td><code id="segmentByCBS.RawGenomicSignals_+3A_cache">cache</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, results are cached to file, otherwise not.</p>
</td></tr>
<tr><td><code id="segmentByCBS.RawGenomicSignals_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, cached results are ignored.</p>
</td></tr>
<tr><td><code id="segmentByCBS.RawGenomicSignals_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally <code><a href="DNAcopy.html#topic+segment">segment</a></code> is used to segment the signals.
This segmentation method support weighted segmentation.
</p>
<p>The &quot;DNAcopy::segment&quot; implementation of CBS uses approximation
through random sampling for some estimates.  Because of this,
repeated calls using the same signals may result in slightly
different results.
</p>


<h3>Value</h3>

<p>Returns the fit object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] A.B. Olshen, E.S. Venkatraman (aka Venkatraman E. Seshan),
R. Lucito and M. Wigler, <em>Circular binary segmentation for
the analysis of array-based DNA copy number data</em>,
Biostatistics, 2004.<br />
[2] E.S. Venkatraman and A.B. Olshen, <em>A faster circular binary
segmentation algorithm for the analysis of array CGH data</em>.
Bioinformatics, 2007.<br />
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+RawGenomicSignals">RawGenomicSignals</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>isEnvVarTRUE &lt;- function(name) {
  value &lt;- toupper(Sys.getenv(name))
  if (value == "yes") value &lt;- "TRUE"
  isTRUE(as.logical(value))
}

# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Simulating copy-number data
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Number of loci
J &lt;- 500

mu &lt;- double(J)
mu[100:150] &lt;- mu[100:150] + 1
mu[320:400] &lt;- mu[320:400] - 1
eps &lt;- rnorm(J, sd=1/2)
y &lt;- mu + eps
x &lt;- sort(runif(length(y), max=length(y))) * 1e5
w &lt;- runif(J)
w[320:400] &lt;- 0.001


cn &lt;- RawCopyNumbers(y, x)
print(cn)

plot(cn, ylim=c(-3,3), col="#aaaaaa", xlab="Position (Mb)")

cnS &lt;- binnedSmoothing(cn, by=500e3)
print(cnS)
lines(cnS, col="black", lwd=3)


# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Segment
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
legend &lt;- c()

if (require("DNAcopy")) {
  fit &lt;- segmentByCBS(cn)
  cnr &lt;- extractCopyNumberRegions(fit)
  print(cnr)
  drawLevels(cnr, col="red", lwd=3)
  legend &lt;- c(legend, red="CBS")
}

if (require("GLAD") &amp;&amp; packageVersion("GLAD") != "9.9.9" &amp;&amp;
    !isEnvVarTRUE("_R_S3_METHOD_LOOKUP_BASEENV_AFTER_GLOBALENV_")) {
  fit &lt;- segmentByGLAD(cn)
  cnr &lt;- extractCopyNumberRegions(fit)
  print(cnr)
  drawLevels(cnr, col="blue", lwd=3)
  legend &lt;- c(legend, blue="GLAD")
}

if (require("HaarSeg")) {
  fit &lt;- segmentByHaarSeg(cn)
  cnr &lt;- extractCopyNumberRegions(fit)
  print(cnr)
  drawLevels(cnr, col="orange", lwd=3)
  legend &lt;- c(legend, orange="HaarSeg")
}

if (require("mpcbs")) {
  fit &lt;- segmentByMPCBS(cn)
  cnr &lt;- extractCopyNumberRegions(fit)
  print(cnr)
  drawLevels(cnr, col="white", lwd=6)
  drawLevels(cnr, col="purple", lwd=3)
  legend &lt;- c(legend, purple="MPCBS")
}


if (length(legend) &gt; 0) {
  legend("topleft", pch=19, col=names(legend), legend, bty="n", horiz=TRUE)
}
</code></pre>

<hr>
<h2 id='segmentByGLAD.RawGenomicSignals'>Segment copy numbers using the GLAD method</h2><span id='topic+segmentByGLAD.RawGenomicSignals'></span><span id='topic+RawGenomicSignals.segmentByGLAD'></span><span id='topic+segmentByGLAD+2CRawGenomicSignals-method'></span>

<h3>Description</h3>

<p>Segment copy numbers using the GLAD method of the <span class="pkg">GLAD</span> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'RawGenomicSignals'
segmentByGLAD(this, ..., flavor=c("glad", "daglad"), cache=FALSE, force=FALSE,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="segmentByGLAD.RawGenomicSignals_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the segmentation function.</p>
</td></tr>
<tr><td><code id="segmentByGLAD.RawGenomicSignals_+3A_flavor">flavor</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying what flavor of
GLAD should be used.</p>
</td></tr>
<tr><td><code id="segmentByGLAD.RawGenomicSignals_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally <code><a href="GLAD.html#topic+glad">glad</a></code> or  <code><a href="GLAD.html#topic+daglad">daglad</a></code> is used to
segment the signals.
This segmentation method does not support weighted segmentation.
</p>


<h3>Value</h3>

<p>Returns the fit object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+RawGenomicSignals">RawGenomicSignals</a></code>.
</p>

<hr>
<h2 id='segmentByHaarSeg.RawGenomicSignals'>Segment copy numbers using the HaarSeg method</h2><span id='topic+segmentByHaarSeg.RawGenomicSignals'></span><span id='topic+RawGenomicSignals.segmentByHaarSeg'></span><span id='topic+segmentByHaarSeg+2CRawGenomicSignals-method'></span>

<h3>Description</h3>

<p>Segment copy numbers using the HaarSeg method of the <span class="pkg">HaarSeg</span> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'RawGenomicSignals'
segmentByHaarSeg(this, ..., cache=FALSE, force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="segmentByHaarSeg.RawGenomicSignals_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the segmentation function.</p>
</td></tr>
<tr><td><code id="segmentByHaarSeg.RawGenomicSignals_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally <code>haarSeg()</code> of the <span class="pkg">HaarSeg</span> is used to segment
the signals.
This segmentation method support weighted segmentation.
</p>


<h3>Value</h3>

<p>Returns the fit object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+RawGenomicSignals">RawGenomicSignals</a></code>.
</p>

<hr>
<h2 id='segmentByMPCBS.RawGenomicSignals'>Segment copy numbers using the multi-platform CBS (mpCBS) method</h2><span id='topic+segmentByMPCBS.RawGenomicSignals'></span><span id='topic+RawGenomicSignals.segmentByMPCBS'></span><span id='topic+segmentByMPCBS+2CRawGenomicSignals-method'></span>

<h3>Description</h3>

<p>Segment copy numbers using the multi-platform CBS (mpCBS) method of the <span class="pkg">mpcbs</span> package.
</p>
<p><em>WARNING: The <span class="pkg">mpcbs</span> package is an old package that is no
longer maintained. It also has '_R_CHECK_LENGTH_1_CONDITION_' and
'_R_CHECK_LENGTH_1_LOGIC2_' bugs, which give errors in R (&gt;= 4.2.0).
This means that 'segmentByMPCBS()' does not work in R (&gt;= 4.2.0).</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'RawGenomicSignals'
segmentByMPCBS(this, ..., cache=FALSE, force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="segmentByMPCBS.RawGenomicSignals_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the segmentation function.</p>
</td></tr>
<tr><td><code id="segmentByMPCBS.RawGenomicSignals_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally <code>mpcbs.mbic()</code> of the <span class="pkg">mpcbs</span> package is used
for segmenting the signals.
This segmentation method does not support weighted segmentation.
</p>


<h3>Value</h3>

<p>Returns the fit object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+RawGenomicSignals">RawGenomicSignals</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Simulating copy-number data from multiple platforms
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Piecewise-constant copy-number state function
cnState &lt;- function(x, ...) {
  n &lt;- length(x)
  mu &lt;- double(n)
  mu[20e6 &lt;= x &amp; x &lt;= 30e6] &lt;- +1
  mu[65e6 &lt;= x &amp; x &lt;= 80e6] &lt;- -1
  mu
} # cnState()

xMax &lt;- 100e6

Js &lt;- c(200, 400, 100)
bs &lt;- c(1, 1.4, 0.5)
as &lt;- c(0, +0.5, -0.5)
sds &lt;- c(0.5, 0.3, 0.8)

cnList &lt;- list()
for (kk in seq_along(Js)) {
  J &lt;- Js[kk]
  a &lt;- as[kk]
  b &lt;- bs[kk]
  sd &lt;- sds[kk]
  x &lt;- sort(runif(J, max=xMax))
  mu &lt;- a + b * cnState(x)
  eps &lt;- rnorm(J, sd=sd)
  y &lt;- mu + eps
  cn &lt;- RawCopyNumbers(y, x)
  cnList[[kk]] &lt;- cn
}


# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Merge platform data (record their sources in 'id')
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
cn &lt;- Reduce(append, cnList)
plot(cn, ylim=c(-3,3), col=cn$id)

# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Segment
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
legend &lt;- c()

if (require("DNAcopy")) {
  fit &lt;- segmentByCBS(cn)
  cnr &lt;- extractCopyNumberRegions(fit)
  print(cnr)
  drawLevels(cnr, col="white", lwd=6)
  drawLevels(cnr, col="red", lwd=3)
  legend &lt;- c(legend, red="CBS")
}


## WORKAROUND: There's a _R_CHECK_LENGTH_1_LOGIC2_ bug in
## mpcbs::mpcbs.mbic().  Until fixed, if ever, we cannot
## call segmentByMPCBS() here. /HB 2022-11-10
if (isTRUE(Sys.getenv("R_CHECK_FULL")) &amp;&amp; require("mpcbs")) {
  fit &lt;- segmentByMPCBS(cn)
  cnr &lt;- extractCopyNumberRegions(fit)
  print(cnr)
  drawLevels(cnr, col="white", lwd=6)
  drawLevels(cnr, col="blue", lwd=3)
  legend &lt;- c(legend, blue="MPCBS")
}


if (length(legend) &gt; 0) {
  legend("topleft", pch=19, col=names(legend), legend, bty="n", horiz=TRUE)
}
</code></pre>

<hr>
<h2 id='SegmentedAlleleBFractions'>The SegmentedAlleleBFractions class</h2><span id='topic+SegmentedAlleleBFractions'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class SegmentedAlleleBFractions</b><br />
</p>
<p><code>data.frame</code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+RichDataFrame">RichDataFrame</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+RawGenomicSignals">RawGenomicSignals</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+RawAlleleBFractions">RawAlleleBFractions</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+SegmentedGenomicSignalsInterface">SegmentedGenomicSignalsInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code>SegmentedAlleleBFractions</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public class <b>SegmentedAlleleBFractions</b><br />
extends <a href="#topic+SegmentedGenomicSignalsInterface">SegmentedGenomicSignalsInterface</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SegmentedAlleleBFractions(..., states=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SegmentedAlleleBFractions_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+RawAlleleBFractions">RawAlleleBFractions</a></code>.</p>
</td></tr>
<tr><td><code id="SegmentedAlleleBFractions_+3A_states">states</code></td>
<td>
<p>A <code><a href="base.html#topic+function">function</a></code> returning the copy-number states given a
<code><a href="base.html#topic+vector">vector</a></code> of locus positions.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from SegmentedGenomicSignalsInterface</b>:<br />
as.data.frame, binnedSmoothingByState, extractSubsetByState, findChangePointsByState, getStateColorMap, getStateColors, getStates, getUniqueStates, getVirtualField, getVirtualLocusFields, kernelSmoothingByState, plot, points, setStateColorMap, setStates
</p>
<p><b>Methods inherited from RawAlleleBFractions</b>:<br />
extractRawMirroredAlleleBFractions, plot
</p>
<p><b>Methods inherited from RawGenomicSignals</b>:<br />
*, +, -, addBy, append, applyBinaryOperator, as.character, as.data.frame, assertOneChromosome, binnedSmoothing, binnedSmoothingByField, clearCache, clone, divideBy, drawDensity, estimateStandardDeviation, extractChromosome, extractChromosomes, extractDataForSegmentation, extractRegion, extractRegions, extractSubset, gaussianSmoothing, getBasicField, getCXY, getChromosome, getChromosomes, getDefaultLocusFields, getLocusFields, getPositions, getSigma, getSignalColumnName, getSignalColumnNames, getSignals, getWeights, getXScale, getXY, getYScale, hasWeights, kernelSmoothing, lines, multiplyBy, nbrOfChromosomes, nbrOfLoci, plot, points, print, segmentByCBS, segmentByGLAD, segmentByHaarSeg, segmentByMPCBS, setBasicField, setSigma, setSignals, setWeights, setXScale, setYScale, signalRange, sort, subtractBy, xMax, xMin, xRange, xSeq, yMax, yMin, yRange
</p>
<p><b>Methods inherited from RichDataFrame</b>:<br />
$, $&lt;-, [, [[, [[&lt;-, as.data.frame, as.list, dim, dropVirtualColumn, getColumnNames, getColumnNamesTranslator, getFullName, getName, getTags, getVirtualColumn, getVirtualColumnFunction, getVirtualColumnNames, hasColumn, hasColumns, hasVirtualColumn, hasVirtualColumns, length, names, newInstance, print, rbind, setAttributes, setColumnNamesMap, setColumnNamesTranslator, setName, setTags, setVirtualColumn, subset, translateColumnNames
</p>
<p><b>Methods inherited from data.frame</b>:<br />
$&lt;-,data.frame-method, $&lt;-, Math, Ops,nonStructure,vector-method, Ops,structure,vector-method, Ops,vector,nonStructure-method, Ops,vector,structure-method, Ops, Summary, [, [&lt;-,data.frame-method, [&lt;-, [[, [[&lt;-,data.frame-method, [[&lt;-, aggregate, anyDuplicated, anyNA, as.NonPairedPSCNData, as.PairedPSCNData, as.data.frame, as.list, as.matrix, as.vector, attachLocally, by, callSegmentationOutliers, cbind, coerce,ANY,list-method, coerce,oldClass,S3-method, dim, dimnames, dimnames&lt;-, dropSegmentationOutliers, droplevels, duplicated, edit, findLargeGaps, format, formula, head, initialize,oldClass-method, is.na, merge, na.exclude, na.omit, plot, plotDensity, print, prompt, rbind, row.names, row.names&lt;-, rowsum, segmentByCBS, segmentByPairedPSCBS, show,oldClass-method, slotsFromS3,data.frame-method, split, split&lt;-, stack, str, subset, summary, t, tail, transform, type.convert, unique, unstack, unwrap, within, wrap, writeDataFrame, xtfrm
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='SegmentedCopyNumbers'>The SegmentedCopyNumbers class</h2><span id='topic+SegmentedCopyNumbers'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class SegmentedCopyNumbers</b><br />
</p>
<p><code>data.frame</code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+RichDataFrame">RichDataFrame</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+RawGenomicSignals">RawGenomicSignals</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+RawCopyNumbers">RawCopyNumbers</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+SegmentedGenomicSignalsInterface">SegmentedGenomicSignalsInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code>SegmentedCopyNumbers</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public class <b>SegmentedCopyNumbers</b><br />
extends <a href="#topic+SegmentedGenomicSignalsInterface">SegmentedGenomicSignalsInterface</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SegmentedCopyNumbers(..., states=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SegmentedCopyNumbers_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+RawCopyNumbers">RawCopyNumbers</a></code>.</p>
</td></tr>
<tr><td><code id="SegmentedCopyNumbers_+3A_states">states</code></td>
<td>
<p>A <code><a href="base.html#topic+function">function</a></code> returning the copy-number states given a
<code><a href="base.html#topic+vector">vector</a></code> of locus positions.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from SegmentedGenomicSignalsInterface</b>:<br />
as.data.frame, binnedSmoothingByState, extractSubsetByState, findChangePointsByState, getStateColorMap, getStateColors, getStates, getUniqueStates, getVirtualField, getVirtualLocusFields, kernelSmoothingByState, plot, points, setStateColorMap, setStates
</p>
<p><b>Methods inherited from RawCopyNumbers</b>:<br />
cnRange, extractRawCopyNumbers, getCNs, getCn, getSignals, plot
</p>
<p><b>Methods inherited from RawGenomicSignals</b>:<br />
*, +, -, addBy, append, applyBinaryOperator, as.character, as.data.frame, assertOneChromosome, binnedSmoothing, binnedSmoothingByField, clearCache, clone, divideBy, drawDensity, estimateStandardDeviation, extractChromosome, extractChromosomes, extractDataForSegmentation, extractRegion, extractRegions, extractSubset, gaussianSmoothing, getBasicField, getCXY, getChromosome, getChromosomes, getDefaultLocusFields, getLocusFields, getPositions, getSigma, getSignalColumnName, getSignalColumnNames, getSignals, getWeights, getXScale, getXY, getYScale, hasWeights, kernelSmoothing, lines, multiplyBy, nbrOfChromosomes, nbrOfLoci, plot, points, print, segmentByCBS, segmentByGLAD, segmentByHaarSeg, segmentByMPCBS, setBasicField, setSigma, setSignals, setWeights, setXScale, setYScale, signalRange, sort, subtractBy, xMax, xMin, xRange, xSeq, yMax, yMin, yRange
</p>
<p><b>Methods inherited from RichDataFrame</b>:<br />
$, $&lt;-, [, [[, [[&lt;-, as.data.frame, as.list, dim, dropVirtualColumn, getColumnNames, getColumnNamesTranslator, getFullName, getName, getTags, getVirtualColumn, getVirtualColumnFunction, getVirtualColumnNames, hasColumn, hasColumns, hasVirtualColumn, hasVirtualColumns, length, names, newInstance, print, rbind, setAttributes, setColumnNamesMap, setColumnNamesTranslator, setName, setTags, setVirtualColumn, subset, translateColumnNames
</p>
<p><b>Methods inherited from data.frame</b>:<br />
$&lt;-,data.frame-method, $&lt;-, Math, Ops,nonStructure,vector-method, Ops,structure,vector-method, Ops,vector,nonStructure-method, Ops,vector,structure-method, Ops, Summary, [, [&lt;-,data.frame-method, [&lt;-, [[, [[&lt;-,data.frame-method, [[&lt;-, aggregate, anyDuplicated, anyNA, as.NonPairedPSCNData, as.PairedPSCNData, as.data.frame, as.list, as.matrix, as.vector, attachLocally, by, callSegmentationOutliers, cbind, coerce,ANY,list-method, coerce,oldClass,S3-method, dim, dimnames, dimnames&lt;-, dropSegmentationOutliers, droplevels, duplicated, edit, findLargeGaps, format, formula, head, initialize,oldClass-method, is.na, merge, na.exclude, na.omit, plot, plotDensity, print, prompt, rbind, row.names, row.names&lt;-, rowsum, segmentByCBS, segmentByPairedPSCBS, show,oldClass-method, slotsFromS3,data.frame-method, split, split&lt;-, stack, str, subset, summary, t, tail, transform, type.convert, unique, unstack, unwrap, within, wrap, writeDataFrame, xtfrm
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>Examples</h3>

<pre><code class='language-R'># - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Simulating copy-number data
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# True CN states
stateFcn &lt;- function(x, ...) {
  states &lt;- integer(length(x))
  states[200 &lt;=x &amp; x &lt;= 300] &lt;- -1L
  states[650 &lt;=x &amp; x &lt;= 800] &lt;- +1L
  states
}

# Number of loci
J &lt;- 1000

y &lt;- rnorm(J, sd=1/2)
x &lt;- 1:length(y)
for (state in c(-1,+1)) {
  idxs &lt;- (stateFcn(x) == state)
  y[idxs] &lt;- y[idxs] + state
}


cn &lt;- SegmentedCopyNumbers(y, x, states=stateFcn)
print(cn)

# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Subsetting
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
plot(cn, ylim=c(-4,4))
title("Copy numbers annotated by state (and subset by state)")

cnS &lt;- extractSubsetByState(cn, states=c(0,+1L))
print(cnS)
points(cnS, pch=21, cex=1.2, lwd=2, col="purple")

legend("topright", pch=c(19, 21), col=c("#999999", "purple"),
       sprintf(c("raw [n=%d]", "CN in {0,1} [n=%d]"),
       c(nbrOfLoci(cn), nbrOfLoci(cnS))), bty="n")


# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Kernel smoothing stratified by state
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
plot(cn, col="#999999", ylim=c(-3,3))
title(main="Kernel smoothing stratified by state w/ Gaussian kernel")

cnSa &lt;- kernelSmoothingByState(cn, h=2)
points(cnSa, col="blue")

cnSb &lt;- kernelSmoothingByState(cn, h=5)
points(cnSb, col="red")

legend("topright", pch=19, col=c("#999999", "blue", "red"),
       sprintf(c("raw [n=%d]", "N(.,sd=2) [n=%d]", "N(.,sd=5) [n=%d]"),
       c(nbrOfLoci(cn), nbrOfLoci(cnSa), nbrOfLoci(cnSb))), bty="n")


# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Binned smoothing stratified by state
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
plot(cn, col="#999999", ylim=c(-3,3))
title(main="Binned smoothing stratified by state")

cnSa &lt;- binnedSmoothingByState(cn, by=3, verbose=-1)
lines(cnSa, col="blue")
points(cnSa, col="blue")

cnSb &lt;- binnedSmoothingByState(cn, by=9, verbose=-1)
lines(cnSb, col="red")
points(cnSb, col="red")

legend("topright", pch=19, col=c("#999999", "blue", "red"),
       sprintf(c("raw [n=%d]", "Bin(w=3) [n=%d]", "Bin(w=9) [n=%d]"),
       c(nbrOfLoci(cn), nbrOfLoci(cnSa), nbrOfLoci(cnSb))), bty="n")
</code></pre>

<hr>
<h2 id='SegmentedGenomicSignalsInterface'>The SegmentedGenomicSignalsInterface class interface</h2><span id='topic+SegmentedGenomicSignalsInterface'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class SegmentedGenomicSignalsInterface</b><br />
</p>
<p><code><a href="R.oo.html#topic+Interface">Interface</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code>SegmentedGenomicSignalsInterface</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+SegmentedAlleleBFractions">SegmentedAlleleBFractions</a>, <a href="#topic+SegmentedCopyNumbers">SegmentedCopyNumbers</a><br />
</p>
<p>public class <b>SegmentedGenomicSignalsInterface</b><br />
extends <a href="R.oo.html#topic+Interface">Interface</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SegmentedGenomicSignalsInterface(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SegmentedGenomicSignalsInterface_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as.data.frame</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>binnedSmoothingByState</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractSubsetByState</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>findChangePointsByState</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getStateColorMap</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getStateColors</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getStates</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUniqueStates</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>kernelSmoothingByState</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plot</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>points</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setStateColorMap</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setStates</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from Interface</b>:<br />
extend, print, uses
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='setAlias.Explorer'>Sets the alias of the output set</h2><span id='topic+setAlias.Explorer'></span><span id='topic+Explorer.setAlias'></span><span id='topic+setAlias+2CExplorer-method'></span>

<h3>Description</h3>

<p>Sets the alias of the output set.
If specified, the alias overrides the data set name, which is used by
default.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Explorer'
setAlias(this, alias=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="setAlias.Explorer_+3A_alias">alias</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string for the new alias of the output set.
The alias must consists of valid filename characters, and must not
contain commas, which are used to separate tags.</p>
</td></tr>
<tr><td><code id="setAlias.Explorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns itself invisibly.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+getAlias.Explorer">*getAlias</a>()</code>.
<code><a href="#topic+getName.Explorer">*getName</a>()</code>.
For more information see <code><a href="#topic+Explorer">Explorer</a></code>.
</p>

<hr>
<h2 id='setArrays.ChromosomeExplorer'>Sets the arrays</h2><span id='topic+setArrays.ChromosomeExplorer'></span><span id='topic+ChromosomeExplorer.setArrays'></span><span id='topic+setArrays+2CChromosomeExplorer-method'></span>

<h3>Description</h3>

<p>Sets the arrays to be processed by the explorer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ChromosomeExplorer'
setArrays(this, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="setArrays.ChromosomeExplorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ChromosomeExplorer">ChromosomeExplorer</a></code>.
</p>

<hr>
<h2 id='setArrays.Explorer'>Sets the arrays</h2><span id='topic+setArrays.Explorer'></span><span id='topic+Explorer.setArrays'></span><span id='topic+setArrays+2CExplorer-method'></span>

<h3>Description</h3>

<p>Sets the arrays to be processed by the explorer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Explorer'
setArrays(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="setArrays.Explorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+Explorer">Explorer</a></code>.
</p>

<hr>
<h2 id='UnitAnnotationDataFile'>The UnitAnnotationDataFile interface class</h2><span id='topic+UnitAnnotationDataFile'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class UnitAnnotationDataFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Interface">Interface</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code>UnitAnnotationDataFile</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+AromaUflFile">AromaUflFile</a>, <a href="#topic+AromaUgcFile">AromaUgcFile</a>, <a href="#topic+AromaUgpFile">AromaUgpFile</a>, <em><a href="#topic+AromaUnitChromosomeTabularBinaryFile">AromaUnitChromosomeTabularBinaryFile</a></em>, <a href="#topic+AromaUnitGcContentFile">AromaUnitGcContentFile</a>, <em><a href="#topic+AromaUnitTabularBinaryFile">AromaUnitTabularBinaryFile</a></em>, <a href="#topic+AromaUnitTypesFile">AromaUnitTypesFile</a>, <a href="#topic+TextUnitNamesFile">TextUnitNamesFile</a>, <em><a href="#topic+UnitNamesFile">UnitNamesFile</a></em>, <em><a href="#topic+UnitTypesFile">UnitTypesFile</a></em><br />
</p>
<p>public static class <b>UnitAnnotationDataFile</b><br />
extends <a href="R.oo.html#topic+Interface">Interface</a><br />
</p>
<p>A UnitAnnotationDataFile provides methods for querying certain types
of chip type annotation data by units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UnitAnnotationDataFile(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="UnitAnnotationDataFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="R.oo.html#topic+Interface">Interface</a></code>.</p>
</td></tr>
</table>


<h3>Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAromaUflFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAromaUgpFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPlatform</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from Interface</b>:<br />
extend, print, uses
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='UnitNamesFile'>The UnitNamesFile interface class</h2><span id='topic+UnitNamesFile'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class UnitNamesFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Interface">Interface</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+UnitAnnotationDataFile">UnitAnnotationDataFile</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><em><code>UnitNamesFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+TextUnitNamesFile">TextUnitNamesFile</a><br />
</p>
<p>public abstract static class <b>UnitNamesFile</b><br />
extends <a href="#topic+UnitAnnotationDataFile">UnitAnnotationDataFile</a><br />
</p>
<p>A UnitNamesFile provides methods for querying the unit names of
a given chip type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UnitNamesFile(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="UnitNamesFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+UnitAnnotationDataFile">UnitAnnotationDataFile</a></code>.</p>
</td></tr>
</table>


<h3>Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitNames</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>indexOf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from UnitAnnotationDataFile</b>:<br />
byChipType, getAromaUflFile, getAromaUgpFile, getChipType, getDefaultExtension, getPlatform, nbrOfUnits
</p>
<p><b>Methods inherited from Interface</b>:<br />
extend, print, uses
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='UnitTypesFile'>The UnitTypesFile interface class</h2><span id='topic+UnitTypesFile'></span>

<h3>Description</h3>

<p>Package:  aroma.core <br />
<b>Class UnitTypesFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Interface">Interface</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+UnitAnnotationDataFile">UnitAnnotationDataFile</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><em><code>UnitTypesFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+AromaUnitTypesFile">AromaUnitTypesFile</a><br />
</p>
<p>public abstract static class <b>UnitTypesFile</b><br />
extends <a href="#topic+UnitAnnotationDataFile">UnitAnnotationDataFile</a><br />
</p>
<p>A UnitTypesFile provides methods for querying the unit types of
a given chip type, e.g. genotyping or copy-number unit, exon unit etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UnitTypesFile(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="UnitTypesFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+UnitAnnotationDataFile">UnitAnnotationDataFile</a></code>.</p>
</td></tr>
</table>


<h3>Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitTypes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from UnitAnnotationDataFile</b>:<br />
byChipType, getAromaUflFile, getAromaUgpFile, getChipType, getDefaultExtension, getPlatform, nbrOfUnits
</p>
<p><b>Methods inherited from Interface</b>:<br />
extend, print, uses
</p>


<h3>The aroma unit-type map</h3>

<p>unknown=0,
expression=1,
genotyping=2,
resequencing=3,
tag=4,
copynumber=5,
genotypingcontrol=6,
expressioncontrol=7
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='updateSetupExplorerFile.ChromosomeExplorer'>Updates the Javascript file</h2><span id='topic+updateSetupExplorerFile.ChromosomeExplorer'></span><span id='topic+ChromosomeExplorer.updateSetupExplorerFile'></span><span id='topic+updateSetupExplorerFile+2CChromosomeExplorer-method'></span>

<h3>Description</h3>

<p>Updates the Javascript file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ChromosomeExplorer'
updateSetupExplorerFile(this, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="updateSetupExplorerFile.ChromosomeExplorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="updateSetupExplorerFile.ChromosomeExplorer_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns (invisibly) the pathname to the samples.js file.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ChromosomeExplorer">ChromosomeExplorer</a></code>.
</p>

<hr>
<h2 id='writeDataFrame.AromaUnitSignalBinaryFile'>Writes the data file as a tab-delimited text file</h2><span id='topic+writeDataFrame.AromaUnitSignalBinaryFile'></span><span id='topic+AromaUnitSignalBinaryFile.writeDataFrame'></span><span id='topic+writeDataFrame+2CAromaUnitSignalBinaryFile-method'></span>

<h3>Description</h3>

<p>Writes the data file as a tab-delimited text file with or without file header comments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaUnitSignalBinaryFile'
writeDataFrame(this, filename=sprintf("%s.txt", getFilename(this)),
  path=file.path(getRootName(this, tags = "*,txt"), getParentName(this),
  getChipType(this, fullname = FALSE)), ..., columns=c("unitName", "*"), sep="\t",
  addHeader=TRUE, createdBy=NULL, nbrOfDecimals=4L,
  columnNamesPrefix=c("fullname", "name", "none"), overwrite=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="writeDataFrame.AromaUnitSignalBinaryFile_+3A_filename">filename</code></td>
<td>
<p>The filename of the generated file.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitSignalBinaryFile_+3A_path">path</code></td>
<td>
<p>The path where the generated file should be written.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitSignalBinaryFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitSignalBinaryFile_+3A_columns">columns</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying which column names,
including optional annotation data column names, that should be
exported.  A string <code>"*"</code> corresponds to inserting
the column names of the source data file.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitSignalBinaryFile_+3A_sep">sep</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying the column separator</p>
</td></tr></table>
<p>.
</p>
<table role = "presentation">
<tr><td><code id="writeDataFrame.AromaUnitSignalBinaryFile_+3A_addheader">addHeader</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, file header comments will be added that
contain additional information about the source file and more.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitSignalBinaryFile_+3A_createdby">createdBy</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying the <code>createdBy</code>
file header comment. If <code><a href="base.html#topic+NULL">NULL</a></code>, the package version will be written.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitSignalBinaryFile_+3A_nbrofdecimals">nbrOfDecimals</code></td>
<td>
<p>An <code><a href="base.html#topic+integer">integer</a></code> specifying the number of decimals
for floating point columns.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitSignalBinaryFile_+3A_columnnamesprefix">columnNamesPrefix</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying what type
of prefix should be used for column names.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitSignalBinaryFile_+3A_overwrite">overwrite</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, an existing destination file will be
overwritten, otherwise an exception will be thrown.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitSignalBinaryFile_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the write data file as a <code><a href="R.filesets.html#topic+TabularTextFile">TabularTextFile</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a></code>.
</p>

<hr>
<h2 id='writeDataFrame.AromaUnitSignalBinarySet'>Writes the data set as a tab-delimited text file</h2><span id='topic+writeDataFrame.AromaUnitSignalBinarySet'></span><span id='topic+AromaUnitSignalBinarySet.writeDataFrame'></span><span id='topic+writeDataFrame+2CAromaUnitSignalBinarySet-method'></span>

<h3>Description</h3>

<p>Writes the data set as a tab-delimited text file with or without file header comments.
</p>
<p><em>We do neither recommend nor encourage the usage of this method -
it is only available due to popular demand</em>.
For more details, see below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaUnitSignalBinarySet'
writeDataFrame(this, filename=sprintf("%s.txt", getFullName(this)),
  path=file.path(getRootName(this, tags = "*,txt"), getFullName(this), getChipType(this,
  fullname = FALSE)), ..., units=NULL, columns=c("unitName", "*"), sep="\t",
  addHeader=TRUE, createdBy=NULL, nbrOfDecimals=4L, ram=1,
  columnNamesPrefix=c("fullnames", "names", "none"), overwrite=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="writeDataFrame.AromaUnitSignalBinarySet_+3A_filename">filename</code></td>
<td>
<p>The filename of the generated file.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitSignalBinarySet_+3A_path">path</code></td>
<td>
<p>The path where the generated file should be written.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitSignalBinarySet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitSignalBinarySet_+3A_units">units</code></td>
<td>
<p>The units to be written.  If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are considered.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitSignalBinarySet_+3A_columns">columns</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying which column names,
including optional annotation data column names, that should be
exported.  A string <code>"*"</code> corresponds to inserting
the column names of the source data files.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitSignalBinarySet_+3A_columnnamesprefix">columnNamesPrefix</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying what type
of prefix should be used for column names.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitSignalBinarySet_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the written data set as a <code><a href="R.filesets.html#topic+TabularTextFileSet">TabularTextFileSet</a></code> object.
</p>


<h3>Warning</h3>

<p>There is no limitation in how big the generated file can be.
The bigger the data set is, the greater the file size will be.
Because of this, <em>we do neither recommend nor encourage the usage
of this method</em>.  It is available only due to popular demand.
Instead we recommend to write each file to a separate file.
See <code>writeDataFrame()</code> of <code><a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a></code> for
more information.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>To write the data of one file, see <code>writeDataFrame()</code> for
<code><a href="#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a></code>.
For more information see <code><a href="#topic+AromaUnitSignalBinarySet">AromaUnitSignalBinarySet</a></code>.
</p>

<hr>
<h2 id='writeDataFrame.AromaUnitTabularBinaryFile'>Writes the data file as a tab-delimited text file</h2><span id='topic+writeDataFrame.AromaUnitTabularBinaryFile'></span><span id='topic+AromaUnitTabularBinaryFile.writeDataFrame'></span><span id='topic+writeDataFrame+2CAromaUnitTabularBinaryFile-method'></span>

<h3>Description</h3>

<p>Writes the data file as a tab-delimited text file with or without file header comments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaUnitTabularBinaryFile'
writeDataFrame(this, filename=sprintf("%s.txt", getFilename(this)),
  path=file.path(getRootName(this, tags = "*,txt"), getFullName(this), getChipType(this,
  fullname = FALSE)), ..., columns=c("unitName", "*"), sep="\t", addHeader=TRUE,
  createdBy=NULL, nbrOfDecimals=4L, columnNamesPrefix=c("fullname", "name", "none"),
  overwrite=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="writeDataFrame.AromaUnitTabularBinaryFile_+3A_filename">filename</code></td>
<td>
<p>The filename of generated file.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitTabularBinaryFile_+3A_path">path</code></td>
<td>
<p>The path where the generated file should be written.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitTabularBinaryFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitTabularBinaryFile_+3A_columns">columns</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying which column names,
including optional annotation data column names, that should be
exported.  A string <code>"*"</code> corresponds to inserting
the column names of the source data file.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitTabularBinaryFile_+3A_sep">sep</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying the column separator</p>
</td></tr></table>
<p>.
</p>
<table role = "presentation">
<tr><td><code id="writeDataFrame.AromaUnitTabularBinaryFile_+3A_addheader">addHeader</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, file header comments will be added that
contain additional information about the source file and more.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitTabularBinaryFile_+3A_createdby">createdBy</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying the <code>createdBy</code>
file header comment. If <code><a href="base.html#topic+NULL">NULL</a></code>, the package version will be written.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitTabularBinaryFile_+3A_nbrofdecimals">nbrOfDecimals</code></td>
<td>
<p>An <code><a href="base.html#topic+integer">integer</a></code> specifying the number of decimals
for floating point columns.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitTabularBinaryFile_+3A_columnnamesprefix">columnNamesPrefix</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying what type
of prefix should be used for column names.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitTabularBinaryFile_+3A_overwrite">overwrite</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, an existing destination file will be
overwritten, otherwise an exception will be thrown.</p>
</td></tr>
<tr><td><code id="writeDataFrame.AromaUnitTabularBinaryFile_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the write data file as a <code><a href="R.filesets.html#topic+TabularTextFile">TabularTextFile</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AromaUnitTabularBinaryFile">AromaUnitTabularBinaryFile</a></code>.
</p>

<hr>
<h2 id='writeFooter.AromaTabularBinaryFile'>Writes a named nested list to the file footer in XML format</h2><span id='topic+writeFooter.AromaTabularBinaryFile'></span><span id='topic+AromaTabularBinaryFile.writeFooter'></span><span id='topic+writeFooter+2CAromaTabularBinaryFile-method'></span>

<h3>Description</h3>

<p>Writes a named nested list to the file footer in XML format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaTabularBinaryFile'
writeFooter(this, footer, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="writeFooter.AromaTabularBinaryFile_+3A_footer">footer</code></td>
<td>
<p>A named <code><a href="base.html#topic+list">list</a></code> structure.</p>
</td></tr>
<tr><td><code id="writeFooter.AromaTabularBinaryFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns nothing.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+readFooter.AromaTabularBinaryFile">*readFooter</a>()</code>.
For more information see <code><a href="#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code>.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
