<!DOCTYPE html><html><head><title>Help for package aroma.affymetrix</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {aroma.affymetrix}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#aroma.affymetrix-package'><p>Package aroma.affymetrix</p></a></li>
<li><a href='#AbstractProbeSequenceNormalization'><p>The AbstractProbeSequenceNormalization class</p></a></li>
<li><a href='#AdditiveCovariatesNormalization'><p>The AdditiveCovariatesNormalization class</p></a></li>
<li><a href='#AffineCnPlm'><p>The AffineCnPlm class</p></a></li>
<li><a href='#AffinePlm'><p>The AffinePlm class</p></a></li>
<li><a href='#AffineSnpPlm'><p>The AffineSnpPlm class</p></a></li>
<li><a href='#AffymetrixCdfFile'><p>The AffymetrixCdfFile class</p></a></li>
<li><a href='#AffymetrixCdfFile+24findByChipType'><p>Locates a CDF file from its chip type</p></a></li>
<li><a href='#AffymetrixCdfFile+24fromFile'><p>Defines an AffymetrixCdfFile object from a CDF file</p></a></li>
<li><a href='#AffymetrixCelFile'><p>The AffymetrixCelFile class</p></a></li>
<li><a href='#AffymetrixCelFile+24allocateFromCdf'><p>Creates an empty CEL file from a template CDF</p></a></li>
<li><a href='#AffymetrixCelFile+24clearData'><p>Clears all or a subset of the fields in a CEL file</p></a></li>
<li><a href='#AffymetrixCelFile+24fromFile'><p>Defines an AffymetrixCelFile object from a CEL file</p></a></li>
<li><a href='#AffymetrixCelSet'><p>The AffymetrixCelSet class</p></a></li>
<li><a href='#AffymetrixCelSet+24importFromDChip'><p>Imports dChip-exported CEL files</p></a></li>
<li><a href='#AffymetrixCelSetReporter'><p>The AffymetrixCelSetReporter class</p></a></li>
<li><a href='#AffymetrixCelSetTuple'><p>The AffymetrixCelSetTuple class</p></a></li>
<li><a href='#AffymetrixCnChpSet'><p>The AffymetrixCnChpSet class</p></a></li>
<li><a href='#AffymetrixFile'><p>The abstract AffymetrixFile class</p></a></li>
<li><a href='#AffymetrixFileSet'><p>The AffymetrixFileSet class</p></a></li>
<li><a href='#AffymetrixFileSet+24byPath'><p>Defines an AffymetrixFileSet object by searching for Affymetrix files</p></a></li>
<li><a href='#AffymetrixFileSetReporter'><p>The AffymetrixFileSetReporter class</p></a></li>
<li><a href='#AffymetrixPgfFile'><p>The AffymetrixPgfFile class</p></a></li>
<li><a href='#AffymetrixPgfFile+24findByChipType'><p>Locates a PGF file from its chip type</p></a></li>
<li><a href='#AffymetrixPgfFile+24fromFile'><p>Defines an AffymetrixPgfFile object from a PGF file</p></a></li>
<li><a href='#AffymetrixProbeTabFile'><p>The AffymetrixProbeTabFile class</p></a></li>
<li><a href='#AlleleSummation'><p>The AlleleSummation class</p></a></li>
<li><a href='#AllelicCrosstalkCalibration'><p>The AllelicCrosstalkCalibration class</p></a></li>
<li><a href='#AromaChipTypeAnnotationFile'><p>The AromaChipTypeAnnotationFile class</p></a></li>
<li><a href='#AromaChipTypeAnnotationFile+24byChipType'><p>Defines an AromaChipTypeAnnotationFile object by chip type</p></a></li>
<li><a href='#AromaChipTypeAnnotationFile+24fromFile'><p>Sets up an AromaChipTypeAnnotationFile</p></a></li>
<li><a href='#AromaUnitTabularBinaryFile+24allocateFromCdf'><p>Creates an AromaUnitTabularBinaryFile mapping to a given CDF</p></a></li>
<li><a href='#ArrayExplorer'><p>The ArrayExplorer class</p></a></li>
<li><a href='#as.AffymetrixCelSet.AffymetrixCelSet'><p>Coerce an object to an AffymetrixCelSet object</p></a></li>
<li><a href='#as.AffymetrixCnChpSet.AffymetrixCnChpSet'><p>Coerce an object to an AffymetrixCnChpSet object</p></a></li>
<li><a href='#as.AffymetrixFileSet.AffymetrixFileSet'><p>Coerce an object to an AffymetrixFileSet object</p></a></li>
<li><a href='#as.character.AffymetrixCelSet'><p>Returns a short string describing the Affymetrix CEL set</p></a></li>
<li><a href='#as.character.AffymetrixCnChpSet'><p>Returns a short string describing the set</p></a></li>
<li><a href='#as.character.CnagCfhSet'><p>Returns a short string describing the CNAG CFH set</p></a></li>
<li><a href='#as.character.DChipDcpSet'><p>Returns a short string describing the DChip CHP set</p></a></li>
<li><a href='#as.CnagCfhSet.CnagCfhSet'><p>Coerce an object to an CnagCfhSet object</p></a></li>
<li><a href='#as.DChipDcpSet.DChipDcpSet'><p>Coerce an object to an DChipDcpSet object</p></a></li>
<li><a href='#averageQuantile.AffymetrixCelSet'><p>Calculates the average empirical distribution across all samples</p></a></li>
<li><a href='#AvgCnPlm'><p>The AvgCnPlm class</p></a></li>
<li><a href='#AvgPlm'><p>The AvgPlm class</p></a></li>
<li><a href='#AvgSnpPlm'><p>The AvgSnpPlm class</p></a></li>
<li><a href='#BackgroundCorrection'><p>The BackgroundCorrection class</p></a></li>
<li><a href='#BaseCountNormalization'><p>The BaseCountNormalization class</p></a></li>
<li><a href='#BasePositionNormalization'><p>The BasePositionNormalization class</p></a></li>
<li><a href='#bpmapCluster2Cdf'><p>Creates a CDF from tiling-array BPMAP file</p></a></li>
<li><a href='#byChipType.DChipGenomeInformation'><p>Defines a DChipGenomeInformation object by chip type</p></a></li>
<li><a href='#byChipType.DChipSnpInformation'><p>Defines a DChipSnpInformation object by chip type</p></a></li>
<li><a href='#byChipType.SnpInformation'><p>Static method to define a genome information set by chip type</p></a></li>
<li><a href='#calculateBaseline.ChipEffectSet'><p>Estimates the baseline signal chromosome by chromosome</p></a></li>
<li><a href='#calculateParametersGsb.AffymetrixCelSet'><p>Computes parameters for adjustment of specific binding</p></a></li>
<li><a href='#calculateResidualSet.FirmaModel'><p>Gets the set of residuals corresponding to the PLM</p></a></li>
<li><a href='#ChipEffectFile'><p>The ChipEffectFile class</p></a></li>
<li><a href='#ChipEffectFile+24fromDataFile'><p>Retrieve an existing CEL file, or create from CDF template if missing</p></a></li>
<li><a href='#ChipEffectGroupMerge'><p>The ChipEffectGroupMerge class</p></a></li>
<li><a href='#ChipEffectSet'><p>The ChipEffectSet class</p></a></li>
<li><a href='#ChipEffectTransform'><p>The ChipEffectTransform class</p></a></li>
<li><a href='#CnagCfhFile'><p>The CnagCfhFile class</p></a></li>
<li><a href='#CnagCfhFile+24fromFile'><p>Defines an CnagCfhFile object from a CFH file</p></a></li>
<li><a href='#CnagCfhSet'><p>The CnagCfhSet class</p></a></li>
<li><a href='#CnChipEffectFile'><p>The CnChipEffectFile class</p></a></li>
<li><a href='#CnChipEffectSet'><p>The CnChipEffectSet class</p></a></li>
<li><a href='#CnChipEffectSet+24importFromApt'><p>Imports copy-number estimates from an APT summary file</p></a></li>
<li><a href='#CnChipEffectSet+24importFromDChip'><p>Imports copy-number estimates from a dChip result file</p></a></li>
<li><a href='#CnPlm'><p>The CnPlm class</p></a></li>
<li><a href='#CnProbeAffinityFile'><p>The CnProbeAffinityFile class</p></a></li>
<li><a href='#compare.AffymetrixCdfFile'><p>Checks if two AffymetrixCdfFile objects are equal</p></a></li>
<li><a href='#computeAffinities.AffymetrixCdfFile'><p>Calculates probe affinities from sequence</p></a></li>
<li><a href='#convert.AffymetrixCdfFile'><p>Converts a CDF into the same CDF but with another format</p></a></li>
<li><a href='#convertUnits.AffymetrixCdfFile'><p>Gets and validates unit indices</p></a></li>
<li><a href='#createExonByTranscriptCdf.AffymetrixCdfFile'><p>Creates an exon-by-transcript CDF</p></a></li>
<li><a href='#createFrom.AffymetrixCelFile'><p>Creates a CEL file using another as a template</p></a></li>
<li><a href='#createMonocellCdf.AffymetrixCdfFile'><p>Creates a mono-cell version of the CDF</p></a></li>
<li><a href='#createUniqueCdf.AffymetrixCdfFile'><p>Creates a unique-cell version of the CDF</p></a></li>
<li><a href='#CrlmmParametersFile'><p>The CrlmmParametersFile class</p></a></li>
<li><a href='#CrlmmParametersSet'><p>The CrlmmParametersSet class</p></a></li>
<li><a href='#DChipCdfBinFile'><p>The DChipCdfBinFile class</p></a></li>
<li><a href='#DChipDcpFile'><p>The DChipDcpFile class</p></a></li>
<li><a href='#DChipDcpSet'><p>The DChipDcpSet class</p></a></li>
<li><a href='#DChipGenomeInformation'><p>The DChipGenomeInformation class</p></a></li>
<li><a href='#DChipQuantileNormalization'><p>The DChipQuantileNormalization class</p></a></li>
<li><a href='#DChipSnpInformation'><p>The DChipSnpInformation class</p></a></li>
<li><a href='#doCRMAv1'><p>Estimation and assessment of raw copy numbers at the single locus level (CRMA v1)</p></a></li>
<li><a href='#doCRMAv2'><p>A single-array preprocessing method for estimating full-resolution raw copy numbers from all Affymetrix genotyping arrays (CRMA v2)</p></a></li>
<li><a href='#doFIRMA'><p>Finding Isoforms using Robust Multichip Analysis (FIRMA)</p></a></li>
<li><a href='#doGCRMA'><p>Robust Multichip Analysis (GCRMA)</p></a></li>
<li><a href='#doRMA'><p>Robust Multichip Analysis (RMA)</p></a></li>
<li><a href='#ExonChipEffectFile'><p>The ExonChipEffectFile class</p></a></li>
<li><a href='#ExonChipEffectSet'><p>The ExonChipEffectSet class</p></a></li>
<li><a href='#ExonProbeAffinityFile'><p>The ExonProbeAffinityFile class</p></a></li>
<li><a href='#ExonRmaPlm'><p>The ExonRmaPlm class</p></a></li>
<li><a href='#extractAffyBatch.AffymetrixCelSet'><p>Extracts an in-memory AffyBatch object from the CEL set</p></a></li>
<li><a href='#extractDataFrame.ParameterCelSet'><p>Extract data as a data.frame for a set of arrays</p></a></li>
<li><a href='#extractExpressionSet.ChipEffectSet'><p>Extracts an in-memory ExpressionSet object</p></a></li>
<li><a href='#extractFeatureSet.AffymetrixCelSet'><p>Extracts CEL signals an in-memory FeatureSet object</p></a></li>
<li><a href='#extractMatrix.AffymetrixCelSet'><p>Extract data as a matrix for a set of arrays</p></a></li>
<li><a href='#extractMatrix.ParameterCelSet'><p>Extract data as a matrix for a set of arrays</p></a></li>
<li><a href='#findByChipType.AromaChipTypeAnnotationFile'><p>Locates an annotation file by its chip type</p></a></li>
<li><a href='#findUnitsTodo.FirmaModel'><p>Identifies non-fitted units</p></a></li>
<li><a href='#findUnitsTodo.ProbeLevelModel'><p>Identifies non-fitted units</p></a></li>
<li><a href='#findUnitsTodo.UnitModel'><p>Identifies non-fitted units</p></a></li>
<li><a href='#FirmaFile'><p>The FirmaFile class</p></a></li>
<li><a href='#FirmaModel'><p>The FirmaModel class</p></a></li>
<li><a href='#FirmaSet'><p>The FirmaSet class</p></a></li>
<li><a href='#fit.FirmaModel'><p>Estimates the model parameters</p></a></li>
<li><a href='#fit.Model'><p>Estimates the model parameters</p></a></li>
<li><a href='#fit.ProbeLevelModel'><p>Estimates the model parameters</p></a></li>
<li><a href='#fit.SingleArrayUnitModel'><p>Estimates the model parameters</p></a></li>
<li><a href='#fit.SmoothMultiarrayModel'><p>Fits the model for one chromosome across samples</p></a></li>
<li><a href='#fitQuantileNormFcn.AffymetrixCelFile'><p>Fits quantile normalization functions for the arrays in the data set</p></a></li>
<li><a href='#FragmentEquivalentClassNormalization'><p>The FragmentEquivalentClassNormalization class</p></a></li>
<li><a href='#FragmentLengthNormalization'><p>The FragmentLengthNormalization class</p></a></li>
<li><a href='#GcContentNormalization'><p>The GcContentNormalization class</p></a></li>
<li><a href='#GcContentNormalization2'><p>The GcContentNormalization2 class</p></a></li>
<li><a href='#GcRmaBackgroundCorrection'><p>The GcRmaBackgroundCorrection class</p></a></li>
<li><a href='#GenericReporter'><p>The GenericReporter class</p></a></li>
<li><a href='#GenomeInformation'><p>The GenomeInformation class</p></a></li>
<li><a href='#GenomeInformation+24byChipType'><p>Static method to define a genome information set by chip type</p></a></li>
<li><a href='#GenomeInformation+24fromCdf'><p>Static method to define a genome information set from a CDF</p></a></li>
<li><a href='#getAlias.GenericReporter'><p>Gets the alias of the report</p></a></li>
<li><a href='#getAlias.Model'><p>Gets the name alias for the model</p></a></li>
<li><a href='#getAlleleCellPairs.AffymetrixCdfFile'><p>Gets the cell indices of allele pairs</p></a></li>
<li><a href='#getAlleleProbePairs.AffymetrixCdfFile'><p>Gets the indices of probepairs with the same pair of SNP nucleotides</p></a></li>
<li><a href='#getAlleleProbePairs3.AffymetrixCdfFile'><p>Gets the indices of probepairs with the same pair of SNP nucleotides</p></a></li>
<li><a href='#getAM.ChipEffectFile'><p>Gets the log-intensities and log-ratios of chip effects for two arrays</p></a></li>
<li><a href='#getAM.ChipEffectSet'><p>Gets the log-intensities and log-ratios of chip effects of the set relative to a reference chip effect file</p></a></li>
<li><a href='#getAverageFile.AffymetrixCelSet'><p>Calculates the mean and the standard deviation of the cell signal (intensity, standard deviation etc.) across the CEL set</p></a></li>
<li><a href='#getAverageFile.CnagCfhSet'><p>Calculates the mean and the standard deviation of the cell signal (intensity, standard deviation etc.) across the CFH set</p></a></li>
<li><a href='#getBaseline.ChipEffectSet'><p>Gets the baseline signals across chromosomes</p></a></li>
<li><a href='#getCdf.AffymetrixCelFile'><p>Gets the CDF structure for this CEL file</p></a></li>
<li><a href='#getCdf.AffymetrixCelSet'><p>Gets the CDF structure for this CEL set</p></a></li>
<li><a href='#getCdf.CnagCfhFile'><p>Gets the CDF structure for this CEL file</p></a></li>
<li><a href='#getCdf.CnagCfhSet'><p>Gets the CDF structure for this CFH set</p></a></li>
<li><a href='#getCdf.Model'><p>Gets the CDF structure for this model</p></a></li>
<li><a href='#getCellIndices.AffymetrixCdfFile'><p>Gets the cell indices unit by unit</p></a></li>
<li><a href='#getCellIndices.ChipEffectFile'><p>Retrieves tree list of cell indices for a set of units</p></a></li>
<li><a href='#getCellIndices.CnChipEffectFile'><p>Retrieves tree list of cell indices for a set of units</p></a></li>
<li><a href='#getCellIndices.CnProbeAffinityFile'><p>Retrieves tree list of cell indices for a set of units</p></a></li>
<li><a href='#getCellIndices.ExonChipEffectFile'><p>Retrieves tree list of cell indices for a set of units</p></a></li>
<li><a href='#getCellIndices.ExonProbeAffinityFile'><p>Retrieves tree list of cell indices for a set of units</p></a></li>
<li><a href='#getCellIndices.FirmaFile'><p>Retrieves tree list of cell indices for a set of units</p></a></li>
<li><a href='#getCellIndices.ProbeAffinityFile'><p>Retrieves tree list of cell indices for a set of units</p></a></li>
<li><a href='#getCellIndices.ResidualFile'><p>Retrieves tree list of cell indices for a set of units</p></a></li>
<li><a href='#getCellIndices.SnpChipEffectFile'><p>Retrieves tree list of cell indices for a set of units</p></a></li>
<li><a href='#getCellIndices.SnpProbeAffinityFile'><p>Retrieves tree list of cell indices for a set of units</p></a></li>
<li><a href='#getCellIndices.UnitModel'><p>Gets the cell indices unit by unit</p></a></li>
<li><a href='#getCellIndices.WeightsFile'><p>Retrieves tree list of cell indices for a set of units</p></a></li>
<li><a href='#getChipEffectSet.AlleleSummation'><p>Gets the set of chip effects for this model</p></a></li>
<li><a href='#getChipEffectSet.ProbeLevelModel'><p>Gets the set of chip effects for this model</p></a></li>
<li><a href='#getChipType.AffymetrixCelFile'><p>Gets the chip type for this CEL file</p></a></li>
<li><a href='#getChipType.AffymetrixCelSet'><p>Gets the chip type for this CEL set</p></a></li>
<li><a href='#getChipType.CnagCfhFile'><p>Gets the chip type for this CEL file</p></a></li>
<li><a href='#getChipType.GenomeInformation'><p>Gets the chip type of this genome information set</p></a></li>
<li><a href='#getChipType.SnpInformation'><p>Gets the chip type of this genome information set</p></a></li>
<li><a href='#getCnNames.AffymetrixCdfFile'><p>Gets the names of the CN units</p></a></li>
<li><a href='#getData.GenomeInformation'><p>Gets all or a subset of the genome information data</p></a></li>
<li><a href='#getData.SnpInformation'><p>Gets all or a subset of the genome information data</p></a></li>
<li><a href='#getDataSet.AffymetrixCelSetReporter'><p>Gets the data set</p></a></li>
<li><a href='#getDataSet.ArrayExplorer'><p>Gets the data set</p></a></li>
<li><a href='#getDataSet.Model'><p>Gets the input data set for this model</p></a></li>
<li><a href='#getFirmaSet.FirmaModel'><p>Gets the set of FIRMA results for this model</p></a></li>
<li><a href='#getFitUnitGroupFunction.AffinePlm'><p>Gets the low-level function that fits the PLM</p></a></li>
<li><a href='#getFitUnitGroupFunction.AvgPlm'><p>Gets the low-level function that fits the PLM</p></a></li>
<li><a href='#getFitUnitGroupFunction.ExonRmaPlm'><p>Gets the low-level function that fits the Exon PLM</p></a></li>
<li><a href='#getFitUnitGroupFunction.FirmaModel'><p>Static method to get the low-level function that fits the PLM</p></a></li>
<li><a href='#getFitUnitGroupFunction.HetLogAddPlm'><p>Gets the low-level function that fits the PLM</p></a></li>
<li><a href='#getFitUnitGroupFunction.MbeiPlm'><p>Gets the low-level function that fits the PLM</p></a></li>
<li><a href='#getFitUnitGroupFunction.RmaPlm'><p>Gets the low-level function that fits the PLM</p></a></li>
<li><a href='#getFullName.Model'><p>Gets the full name of the output set</p></a></li>
<li><a href='#getFullName.TransformReport'><p>Gets the full name of the output data set</p></a></li>
<li><a href='#getGenomeInformation.AffymetrixCdfFile'><p>Gets genome information for this chip type</p></a></li>
<li><a href='#getHeader.AffymetrixCdfFile'><p>Gets the header of the CDF file</p></a></li>
<li><a href='#getHeader.AffymetrixCelFile'><p>Gets the header of the CEL file</p></a></li>
<li><a href='#getHeader.AffymetrixPgfFile'><p>Gets the header of the PGF file</p></a></li>
<li><a href='#getHeader.AromaChipTypeAnnotationFile'><p>Gets the header of the annotation file</p></a></li>
<li><a href='#getHeader.CnagCfhFile'><p>Gets the header of the CEL file</p></a></li>
<li><a href='#getImage.AffymetrixCelFile'><p>Creates an RGB Image object from a CEL file</p></a></li>
<li><a href='#getInputDataSet.TransformReport'><p>Gets the source data set</p></a></li>
<li><a href='#getIntensities.AffymetrixCelSet'><p>Gets cell intensities from a set of cells and a set of arrays</p></a></li>
<li><a href='#getName.GenericReporter'><p>Gets the name of the explorer</p></a></li>
<li><a href='#getName.Model'><p>Gets the name of the output data set</p></a></li>
<li><a href='#getName.TransformReport'><p>Gets the name of the output data set</p></a></li>
<li><a href='#getOutputDataSet.Transform'><p>Gets the transformed data set</p></a></li>
<li><a href='#getOutputDataSet.TransformReport'><p>Gets the transformed data set</p></a></li>
<li><a href='#getPath.Model'><p>Gets the path of this model</p></a></li>
<li><a href='#getPath.TransformReport'><p>Gets the path of the output data set</p></a></li>
<li><a href='#getPositions.GenomeInformation'><p>Gets the physical positions for a set of units</p></a></li>
<li><a href='#getProbeAffinityFile.ProbeLevelModel'><p>Gets the probe affinities for this model</p></a></li>
<li><a href='#getResiduals.QualityAssessmentModel'><p>Calculates the residuals from a probe-level model</p></a></li>
<li><a href='#getRootPath.Model'><p>Gets the root path of this model</p></a></li>
<li><a href='#getSnpNames.AffymetrixCdfFile'><p>Gets the names of the SNP units</p></a></li>
<li><a href='#getTags.GenericReporter'><p>Gets the tags of the reporter</p></a></li>
<li><a href='#getTags.Model'><p>Gets the tags of the output data set</p></a></li>
<li><a href='#getTags.TransformReport'><p>Gets the tags of the output data set</p></a></li>
<li><a href='#getTimestamp.AffymetrixCelFile'><p>Gets the timestamp in the CEL header</p></a></li>
<li><a href='#getUnitGroupCellMap.ChipEffectFile'><p>Gets a (unit, group, cell) index map</p></a></li>
<li><a href='#getUnitIndices.GenomeInformation'><p>Gets unit indices ordered along the chromosome</p></a></li>
<li><a href='#getUnitIntensities.AffymetrixCelSet'><p>Gets cell signals for a subset of units and a subset of arrays</p></a></li>
<li><a href='#getUnitNames.AffymetrixCdfFile'><p>Gets the names of each unit</p></a></li>
<li><a href='#getUnitNames.AffymetrixPgfFile'><p>Gets the names of each unit</p></a></li>
<li><a href='#getUnitTypes.AffymetrixCdfFile'><p>Gets the types of a set of units</p></a></li>
<li><a href='#getWeights.QualityAssessmentModel'><p>Calculates the weights from the robust fit to a probe-level model</p></a></li>
<li><a href='#getXAM.ChipEffectFile'><p>Gets the physical position, log-intensities and log-ratios of chip effects for two arrays</p></a></li>
<li><a href='#getXAM.ChipEffectSet'><p>Gets the physical position, log-intensities and log-ratios of chip effects for two arrays</p></a></li>
<li><a href='#groupUnitsByDimension.AffymetrixCdfFile'><p>Groups units by dimensions</p></a></li>
<li><a href='#HetLogAddCnPlm'><p>The HetLogAddCnPlm class</p></a></li>
<li><a href='#HetLogAddPlm'><p>The HetLogAddPlm class</p></a></li>
<li><a href='#HetLogAddSnpPlm'><p>The HetLogAddSnpPlm class</p></a></li>
<li><a href='#image270.AffymetrixCelFile'><p>Displays all or a subset of the data spatially</p></a></li>
<li><a href='#isDuplicated.AffymetrixCelSet'><p>Identifies duplicated CEL files</p></a></li>
<li><a href='#isDuplicated.CnagCfhSet'><p>Identifies duplicated CFH files</p></a></li>
<li><a href='#isPm.AffymetrixCdfFile'><p>Checks which cells (probes) are PMs and not</p></a></li>
<li><a href='#isResequenceChip.AffymetrixCdfFile'><p>Static method to check if a CDF is for a resequencing (CustomSeq) chip</p></a></li>
<li><a href='#isSnpChip.AffymetrixCdfFile'><p>Static method to check if a chip is a mapping (SNP) chip</p></a></li>
<li><a href='#justRMA'><p>Robust Multichip Analysis (RMA) reproducing the affy package</p></a></li>
<li><a href='#LimmaBackgroundCorrection'><p>The LimmaBackgroundCorrection class</p></a></li>
<li><a href='#LinearModelProbeSequenceNormalization'><p>The LinearModelProbeSequenceNormalization class</p></a></li>
<li><a href='#MatNormalization'><p>The MatNormalization class</p></a></li>
<li><a href='#MatSmoothing'><p>The MatSmoothing class</p></a></li>
<li><a href='#MbeiCnPlm'><p>The MbeiCnPlm class</p></a></li>
<li><a href='#MbeiPlm'><p>The MbeiPlm class</p></a></li>
<li><a href='#MbeiSnpPlm'><p>The MbeiSnpPlm class</p></a></li>
<li><a href='#Model'><p>The Model class</p></a></li>
<li><a href='#MultiArrayUnitModel'><p>The MultiArrayUnitModel class</p></a></li>
<li><a href='#MultiArrayUnitModel+24getFitUnitGroupFunction'><p>Static method to get the low-level function that fits the PLM</p></a></li>
<li><a href='#nbrOfGroupsPerUnit.AffymetrixCdfFile'><p>Gets the number of groups in each unit</p></a></li>
<li><a href='#nbrOfSnps.AffymetrixCdfFile'><p>Gets the number of SNPs</p></a></li>
<li><a href='#Non-documented+20objects'><p>Non-documented objects</p></a></li>
<li><a href='#normalizeQuantile.AffymetrixCelFile'><p>Normalizes the probe intensities to a target empirical distribution</p></a></li>
<li><a href='#normalizeQuantile.AffymetrixCelSet'><p>Normalizes samples to have the same empirical distribution</p></a></li>
<li><a href='#NormExpBackgroundCorrection'><p>The NormExpBackgroundCorrection class</p></a></li>
<li><a href='#OpticalBackgroundCorrection'><p>The OpticalBackgroundCorrection class</p></a></li>
<li><a href='#ParameterCelFile'><p>The ParameterCelFile class</p></a></li>
<li><a href='#ParameterCelSet'><p>The ParameterCelSet class</p></a></li>
<li><a href='#pdInfo2Cdf'><p>Generates an Affymetrix CDF file from a Platform Design (PD) package and a auxiliary CEL file for the same chip type</p></a></li>
<li><a href='#plotBoxplotStats.list'><p>Plots a (merged or non-merged) list of boxplot.stats() elements</p></a></li>
<li><a href='#plotDensity.AffymetrixCelFile'><p>Plots the density of the probe signals on the array</p></a></li>
<li><a href='#plotDensity.AffymetrixCelSet'><p>Plots the densities of all samples</p></a></li>
<li><a href='#plotDensity.GenomeInformation'><p>Plots the density of SNPs for a given chromosome</p></a></li>
<li><a href='#plotImage.AffymetrixCelFile'><p>Displays a spatial plot of a CEL file</p></a></li>
<li><a href='#plotMvsA.AffymetrixCelFile'><p>Plots log-ratio versus log-intensity in a scatter plot</p></a></li>
<li><a href='#plotMvsX.AffymetrixCelFile'><p>Plots log-ratio versus another variable in a scatter plot</p></a></li>
<li><a href='#ProbeAffinityFile'><p>The ProbeAffinityFile class</p></a></li>
<li><a href='#ProbeLevelModel'><p>The ProbeLevelModel class</p></a></li>
<li><a href='#ProbeLevelTransform'><p>The ProbeLevelTransform class</p></a></li>
<li><a href='#ProbeLevelTransform3'><p>The ProbeLevelTransform3 class</p></a></li>
<li><a href='#process.AbstractProbeSequenceNormalization'><p>Normalizes the data set</p></a></li>
<li><a href='#process.AdditiveCovariatesNormalization'><p>Normalizes the data set</p></a></li>
<li><a href='#process.AllelicCrosstalkCalibration'><p>Calibrates the data set</p></a></li>
<li><a href='#process.ArrayExplorer'><p>Generates image files, scripts and dynamic pages for the explorer</p></a></li>
<li><a href='#process.BackgroundCorrection'><p>Processes the data set</p></a></li>
<li><a href='#process.ChipEffectGroupMerge'><p>Normalizes the data set</p></a></li>
<li><a href='#process.DChipQuantileNormalization'><p>Normalizes the data set</p></a></li>
<li><a href='#process.FragmentEquivalentClassNormalization'><p>Normalizes the data set</p></a></li>
<li><a href='#process.FragmentLengthNormalization'><p>Normalizes the data set</p></a></li>
<li><a href='#process.GcContentNormalization'><p>Normalizes the data set</p></a></li>
<li><a href='#process.GcRmaBackgroundCorrection'><p>Performs background correction</p></a></li>
<li><a href='#process.GenericReporter'><p>Generates report</p></a></li>
<li><a href='#process.LimmaBackgroundCorrection'><p>Performs background correction</p></a></li>
<li><a href='#process.MatNormalization'><p>Normalizes the data set</p></a></li>
<li><a href='#process.MatSmoothing'><p>Processes the data set</p></a></li>
<li><a href='#process.OpticalBackgroundCorrection'><p>Performs background correction</p></a></li>
<li><a href='#process.QuantileNormalization'><p>Normalizes the data set</p></a></li>
<li><a href='#process.ReseqCrosstalkCalibration'><p>Calibrates the data set</p></a></li>
<li><a href='#process.RmaBackgroundCorrection'><p>Performs background correction</p></a></li>
<li><a href='#process.ScaleNormalization'><p>Normalizes the data set</p></a></li>
<li><a href='#process.ScaleNormalization3'><p>Normalizes the data set</p></a></li>
<li><a href='#process.SpatialReporter'><p>Generates image files, scripts and dynamic pages for the explorer</p></a></li>
<li><a href='#process.UnitTypeScaleNormalization'><p>Normalizes the data set</p></a></li>
<li><a href='#QualityAssessmentFile'><p>The QualityAssessmentFile class</p></a></li>
<li><a href='#QualityAssessmentModel'><p>The QualityAssessmentModel class</p></a></li>
<li><a href='#QualityAssessmentSet'><p>The QualityAssessmentSet class</p></a></li>
<li><a href='#QuantileNormalization'><p>The QuantileNormalization class</p></a></li>
<li><a href='#randomSeed'><p>Sets and resets the .Random.seed in the global environment</p></a></li>
<li><a href='#readRawData.AffymetrixCelFile'><p>Gets all or a subset of the fields in a CEL file</p></a></li>
<li><a href='#readUnits.AffymetrixCdfFile'><p>Reads CDF data unit by unit</p></a></li>
<li><a href='#readUnits.AffymetrixCelFile'><p>Reads CEL data unit by unit</p></a></li>
<li><a href='#readUnits.CnagCfhFile'><p>Reads CEL data unit by unit</p></a></li>
<li><a href='#readUnits.MultiArrayUnitModel'><p>Reads data unit by unit</p></a></li>
<li><a href='#readUnits.SingleArrayUnitModel'><p>Reads data unit by unit</p></a></li>
<li><a href='#readUnitsByQuartets.AffymetrixCdfFile'><p>Gets the cell quartets for each base position</p></a></li>
<li><a href='#ReseqCrosstalkCalibration'><p>The ReseqCrosstalkCalibration class</p></a></li>
<li><a href='#ResidualFile'><p>The ResidualFile class</p></a></li>
<li><a href='#ResidualSet'><p>The ResidualSet class</p></a></li>
<li><a href='#RmaBackgroundCorrection'><p>The RmaBackgroundCorrection class</p></a></li>
<li><a href='#RmaCnPlm'><p>The RmaCnPlm class</p></a></li>
<li><a href='#RmaPlm'><p>The RmaPlm class</p></a></li>
<li><a href='#RmaSnpPlm'><p>The RmaSnpPlm class</p></a></li>
<li><a href='#ScaleNormalization'><p>The ScaleNormalization class</p></a></li>
<li><a href='#ScaleNormalization3'><p>The ScaleNormalization3 class</p></a></li>
<li><a href='#setAlias.GenericReporter'><p>Sets the alias of the report</p></a></li>
<li><a href='#setAlias.Model'><p>Sets the name alias for the model</p></a></li>
<li><a href='#setArrays.ArrayExplorer'><p>Sets the arrays</p></a></li>
<li><a href='#setCdf.AffymetrixCelFile'><p>Sets the CDF structure for this CEL file</p></a></li>
<li><a href='#setCdf.AffymetrixCelSet'><p>Sets the CDF structure for this CEL set</p></a></li>
<li><a href='#setCdf.CnagCfhFile'><p>Sets the CDF structure for this CEL file</p></a></li>
<li><a href='#setCdf.CnagCfhSet'><p>Sets the CDF structure for this CFH set</p></a></li>
<li><a href='#setRestructor.AffymetrixCdfFile'><p>Specifies a function through which</p></a></li>
<li><a href='#setTags.Model'><p>Sets the tags to be appended</p></a></li>
<li><a href='#setupExampleData.AromaAffymetrix'><p>Setups example data in the current directory</p></a></li>
<li><a href='#SingleArrayUnitModel'><p>The SingleArrayUnitModel class</p></a></li>
<li><a href='#SingleArrayUnitModel+24getFitUnitGroupFunction'><p>Static method to get the low-level function that fits the PLM</p></a></li>
<li><a href='#SmoothMultiarrayModel'><p>The SmoothMultiarrayModel class</p></a></li>
<li><a href='#SmoothRmaModel'><p>The SmoothRmaModel class</p></a></li>
<li><a href='#smoothScatterMvsA.AffymetrixCelFile'><p>Plots log-ratio versus log-intensity in a smooth scatter plot</p></a></li>
<li><a href='#SnpChipEffectFile'><p>The SnpChipEffectFile class</p></a></li>
<li><a href='#SnpChipEffectGroupMerge'><p>The SnpChipEffectGroupMerge class</p></a></li>
<li><a href='#SnpChipEffectSet'><p>The SnpChipEffectSet class</p></a></li>
<li><a href='#SnpInformation'><p>The SnpInformation class</p></a></li>
<li><a href='#SnpInformation+24fromCdf'><p>Static method to define a genome information set from a CDF</p></a></li>
<li><a href='#SnpPlm'><p>The SnpPlm interface class</p></a></li>
<li><a href='#SnpProbeAffinityFile'><p>The SnpProbeAffinityFile class</p></a></li>
<li><a href='#SpatialReporter'><p>The SpatialReporter class</p></a></li>
<li><a href='#TransformReport'><p>The TransformReport class</p></a></li>
<li><a href='#UgpGenomeInformation'><p>The UgpGenomeInformation class</p></a></li>
<li><a href='#UgpGenomeInformation+24byChipType'><p>Defines a UgpGenomeInformation object by chip type</p></a></li>
<li><a href='#UnitModel'><p>The UnitModel class</p></a></li>
<li><a href='#UnitTypeScaleNormalization'><p>The UnitTypeScaleNormalization class</p></a></li>
<li><a href='#updateUnits.AffymetrixCelFile'><p>Updates CEL data unit by unit</p></a></li>
<li><a href='#verify.GenomeInformation'><p>Verifies the correctness of the underlying file</p></a></li>
<li><a href='#verify.SnpInformation'><p>Verifies the correctness of the underlying file</p></a></li>
<li><a href='#WeightsFile'><p>The WeightsFile class</p></a></li>
<li><a href='#WeightsSet'><p>The WeightsSet class</p></a></li>
<li><a href='#writeCdf.AffyGenePDInfo'><p>Generates an Affymetrix CDF file from a Platform Design (PD) package</p></a></li>
<li><a href='#writeImage.AffymetrixCelFile'><p>Writes a spatial image of the signals in the CEL file</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>3.2.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.0), R.utils (&ge; 2.9.0), aroma.core (&ge; 3.2.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, R.methodsS3 (&ge; 1.7.1), R.oo (&ge; 1.23.0), R.cache (&ge;
0.14.0), R.devices (&ge; 2.16.1), R.filesets (&ge; 2.13.0),
aroma.apd (&ge; 0.6.0), MASS, splines, matrixStats (&ge; 0.55.0),
listenv, future</td>
</tr>
<tr>
<td>Suggests:</td>
<td>DBI (&ge; 1.0.0), gsmoothr (&ge; 0.1.7), RColorBrewer (&ge; 1.1-2),
Biobase (&ge; 2.28.0), BiocGenerics (&ge; 0.14.0), affxparser (&ge;
1.40.0), affy (&ge; 1.46.0), affyPLM (&ge; 1.44.0), aroma.light (&ge;
2.4.0), gcrma (&ge; 2.40.0), limma (&ge; 3.24.1), oligo (&ge;
1.32.0), oligoClasses (&ge; 1.30.0), pdInfoBuilder (&ge; 1.32.0),
preprocessCore (&ge; 1.28.0), AffymetrixDataTestFiles, dChipIO
(&ge; 0.1.1)</td>
</tr>
<tr>
<td>SuggestsNote:</td>
<td>BioC (&gt;= 3.0), Recommended: preprocessCore, affyPLM,
aroma.light, affxparser, DNAcopy</td>
</tr>
<tr>
<td>Title:</td>
<td>Analysis of Large Affymetrix Microarray Data Sets</td>
</tr>
<tr>
<td>Description:</td>
<td>A cross-platform R framework that facilitates processing of any number of Affymetrix microarray samples regardless of computer system.  The only parameter that limits the number of chips that can be processed is the amount of available disk space.  The Aroma Framework has successfully been used in studies to process tens of thousands of arrays.  This package has actively been used since 2006.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-2.1">LGPL-2.1</a> | <a href="https://www.r-project.org/Licenses/LGPL-3">LGPL-3</a> [expanded from: LGPL (&ge; 2.1)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://www.aroma-project.org/">https://www.aroma-project.org/</a>,
<a href="https://github.com/HenrikBengtsson/aroma.affymetrix">https://github.com/HenrikBengtsson/aroma.affymetrix</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/HenrikBengtsson/aroma.affymetrix/issues">https://github.com/HenrikBengtsson/aroma.affymetrix/issues</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>TRUE</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Infrastructure, ProprietaryPlatforms, ExonArray, Microarray,
OneChannel, GUI, DataImport, DataRepresentation, Preprocessing,
QualityControl, Visualization, ReportWriting, aCGH,
CopyNumberVariants, DifferentialExpression, GeneExpression,
SNP, Transcription</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-18 19:43:01 UTC; henrik</td>
</tr>
<tr>
<td>Author:</td>
<td>Henrik Bengtsson [aut, cre, cph],
  James Bullard [ctb],
  Kasper Hansen [ctb],
  Pierre Neuvial [ctb],
  Elizabeth Purdom [ctb],
  Mark Robinson [ctb],
  Ken Simpson [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Henrik Bengtsson &lt;henrikb@braju.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-18 20:40:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='aroma.affymetrix-package'>Package aroma.affymetrix</h2><span id='topic+aroma.affymetrix-package'></span><span id='topic+aroma.affymetrix'></span>

<h3>Description</h3>

<p>A cross-platform R framework that facilitates processing of any number of Affymetrix microarray samples regardless of computer system.  The only parameter that limits the number of chips that can be processed is the amount of available disk space.  The Aroma Framework has successfully been used in studies to process tens of thousands of arrays.  This package has actively been used since 2006.
</p>


<h3>Installation and updates</h3>

<p>The preferred way to install this package is:
</p>
<pre>
   source("https://callr.org/install#aroma.affymetrix")
  </pre>


<h3>To get started</h3>

<p>To get started, see the online user guides and the vignettes
<a href="https://www.aroma-project.org/">https://www.aroma-project.org/</a>.
</p>


<h3>How to cite this package</h3>

<p>In order to keep improving and providing support for this project,
please cite references [1], [2], or any applicable publication listed on
<a href="https://aroma-project.org/publications/">https://aroma-project.org/publications/</a>, whenever you publish
work that have been used any of the Aroma Framework.
</p>


<h3>License</h3>

<p>The releases of this package is licensed under
LGPL version 2.1 or newer.
</p>
<p>The development code of the packages is under a private license
(where applicable) and patches sent to the author fall under the
latter license, but will be, if incorporated, released under the
&quot;release&quot; license above.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson, James Bullard, Kasper Hansen, Pierre Neuvial, Elizabeth Purdom, Mark Robinson, Ken Simpson</p>


<h3>References</h3>

<p>[1] H. Bengtsson, K. Simpson, J. Bullard, and K. Hansen, <em>aroma.affymetrix: A generic framework in R for analyzing small to very large Affymetrix data sets in bounded memory</em>, Tech Report #745, Department of Statistics, University of California, Berkeley, February 2008.
<br />
[2] H. Bengtsson, R. Irizarry, B. Carvalho, and T. Speed, <em>Estimation and assessment of raw copy numbers at the single locus level</em>, Bioinformatics, 2008.
<br />
[3] H. Bengtsson, <em>The R.oo package - Object-Oriented Programming with References Using Standard R Code</em>, In Kurt Hornik, Friedrich Leisch and Achim Zeileis, editors, Proceedings of the 3rd International Workshop on Distributed Statistical Computing (DSC 2003), March 20-22, Vienna, Austria. <a href="https://www.r-project.org/conferences/DSC-2003/Proceedings/">https://www.r-project.org/conferences/DSC-2003/Proceedings/</a>
<br />
</p>
<p>For a complete list, see <a href="https://aroma-project.org/publications/">https://aroma-project.org/publications/</a>.
</p>

<hr>
<h2 id='AbstractProbeSequenceNormalization'>The AbstractProbeSequenceNormalization class</h2><span id='topic+AbstractProbeSequenceNormalization'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AbstractProbeSequenceNormalization</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform3">ProbeLevelTransform3</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AbstractProbeSequenceNormalization</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+BaseCountNormalization">BaseCountNormalization</a>, <em><a href="#topic+BasePositionNormalization">BasePositionNormalization</a></em>, <em><a href="#topic+LinearModelProbeSequenceNormalization">LinearModelProbeSequenceNormalization</a></em>, <a href="#topic+MatNormalization">MatNormalization</a><br />
</p>
<p>public abstract static class <b>AbstractProbeSequenceNormalization</b><br />
extends <em><a href="#topic+ProbeLevelTransform3">ProbeLevelTransform3</a></em><br />
</p>
<p>This abstract class represents a normalization method that corrects for
systematic effects in the probe intensities due to differences in
probe sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AbstractProbeSequenceNormalization(..., target=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AbstractProbeSequenceNormalization_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+ProbeLevelTransform3">ProbeLevelTransform3</a></code>.</p>
</td></tr>
<tr><td><code id="AbstractProbeSequenceNormalization_+3A_target">target</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying type of &quot;target&quot; used.
If <code>"zero"</code>, all arrays are normalized to have no effects.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all arrays a normalized to have the same effect as
the average array has.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getTargetFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ProbeLevelTransform3</b>:<br />
getAsteriskTags, getCellsTo, getCellsToFit, getCellsToUpdate, getParameters, getUnitsTo, getUnitsToFit, getUnitsToUpdate, writeSignals
</p>
<p><b>Methods inherited from ProbeLevelTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Requirements</h3>

<p>This class requires that an <code><a href="aroma.core.html#topic+AromaCellSequenceFile">AromaCellSequenceFile</a></code> is
available for the chip type.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AdditiveCovariatesNormalization'>The AdditiveCovariatesNormalization class</h2><span id='topic+AdditiveCovariatesNormalization'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AdditiveCovariatesNormalization</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ChipEffectTransform">ChipEffectTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AdditiveCovariatesNormalization</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+GcContentNormalization2">GcContentNormalization2</a><br />
</p>
<p>public abstract static class <b>AdditiveCovariatesNormalization</b><br />
extends <em><a href="#topic+ChipEffectTransform">ChipEffectTransform</a></em><br />
</p>
<p>This class represents a normalization method that corrects for
GC-content effects on copy-number chip-effect estimates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AdditiveCovariatesNormalization(dataSet=NULL, ..., target=NULL, subsetToFit="-XY",
  shift=0, onMissing=c("median", "ignore"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AdditiveCovariatesNormalization_+3A_dataset">dataSet</code></td>
<td>
<p>A <code><a href="#topic+SnpChipEffectSet">SnpChipEffectSet</a></code>.</p>
</td></tr>
<tr><td><code id="AdditiveCovariatesNormalization_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the constructor of
<code><a href="#topic+ChipEffectTransform">ChipEffectTransform</a></code>.</p>
</td></tr>
<tr><td><code id="AdditiveCovariatesNormalization_+3A_target">target</code></td>
<td>
<p>(Optional) A <code><a href="base.html#topic+character">character</a></code> string or a <code><a href="base.html#topic+function">function</a></code>
specifying what to normalize toward.</p>
</td></tr>
<tr><td><code id="AdditiveCovariatesNormalization_+3A_subsettofit">subsetToFit</code></td>
<td>
<p>The units from which the normalization curve should
be estimated.  If <code><a href="base.html#topic+NULL">NULL</a></code>, all are considered.</p>
</td></tr>
<tr><td><code id="AdditiveCovariatesNormalization_+3A_onmissing">onMissing</code></td>
<td>
<p>Specifies how to normalize units for which the
GC contents are unknown.</p>
</td></tr>
<tr><td><code id="AdditiveCovariatesNormalization_+3A_shift">shift</code></td>
<td>
<p>An optional amount the data points should be shifted
(translated).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For SNPs, the normalization function is estimated based on the total
chip effects, i.e. the sum of the allele signals.  The normalizing
is done by rescale the chip effects on the intensity scale such that
the mean of the total chip effects are the same across samples for
any given GC content.  For allele-specific estimates, both alleles
are always rescaled by the same amount.  Thus, when normalizing
allele-specific chip effects, the total chip effects is change, but not
the relative allele signal, e.g. the allele B frequency.
</p>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ChipEffectTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AffineCnPlm'>The AffineCnPlm class</h2><span id='topic+AffineCnPlm'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AffineCnPlm</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffinePlm">AffinePlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffineSnpPlm">AffineSnpPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+SnpPlm">SnpPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CnPlm">CnPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AffineCnPlm</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>AffineCnPlm</b><br />
extends <a href="#topic+CnPlm">CnPlm</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AffineCnPlm(..., combineAlleles=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffineCnPlm_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AffineSnpPlm">AffineSnpPlm</a></code>.</p>
</td></tr>
<tr><td><code id="AffineCnPlm_+3A_combinealleles">combineAlleles</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">FALSE</a></code>, allele A and allele B are treated
separately, otherwise together.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from CnPlm</b>:<br />
getCellIndices, getChipEffectSet, getCombineAlleles, getParameters, getProbeAffinityFile, setCombineAlleles
</p>
<p><b>Methods inherited from SnpPlm</b>:<br />
getCellIndices, getChipEffectSet, getMergeStrands, getParameters, getProbeAffinityFile, setMergeStrands
</p>
<p><b>Methods inherited from AffineSnpPlm</b>:<br />
getAsteriskTags
</p>
<p><b>Methods inherited from AffinePlm</b>:<br />
getAsteriskTags, getFitUnitGroupFunction, getProbeAffinityFile
</p>
<p><b>Methods inherited from ProbeLevelModel</b>:<br />
calculateResidualSet, calculateWeights, fit, getAsteriskTags, getCalculateResidualsFunction, getChipEffectSet, getProbeAffinityFile, getResidualSet, getRootPath, getWeightsSet
</p>
<p><b>Methods inherited from MultiArrayUnitModel</b>:<br />
getListOfPriors, setListOfPriors, validate
</p>
<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AffinePlm'>The AffinePlm class</h2><span id='topic+AffinePlm'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AffinePlm</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AffinePlm</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AffineCnPlm">AffineCnPlm</a></em>, <em><a href="#topic+AffineSnpPlm">AffineSnpPlm</a></em><br />
</p>
<p>public abstract static class <b>AffinePlm</b><br />
extends <em><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></em><br />
</p>
<p>This class represents affine model in Bengtsson &amp; Hossjer (2006).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AffinePlm(..., background=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffinePlm_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code>.</p>
</td></tr>
<tr><td><code id="AffinePlm_+3A_background">background</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, background is estimate for each unit group,
otherwise not. That is, if <code><a href="base.html#topic+logical">FALSE</a></code>, a <em>linear</em> (proportional)
model without offset is fitted, resulting in very similar results as
obtained by the <code><a href="#topic+MbeiPlm">MbeiPlm</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getProbeAffinityFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ProbeLevelModel</b>:<br />
calculateResidualSet, calculateWeights, fit, getAsteriskTags, getCalculateResidualsFunction, getChipEffectSet, getProbeAffinityFile, getResidualSet, getRootPath, getWeightsSet
</p>
<p><b>Methods inherited from MultiArrayUnitModel</b>:<br />
getListOfPriors, setListOfPriors, validate
</p>
<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Model</h3>

<p>For a single unit group, the affine model is:
</p>
<p style="text-align: center;"><code class="reqn">y_{ik} = a + \theta_i \phi_k + \varepsilon_{ik}</code>
</p>

<p>where <code class="reqn">a</code> is an offset common to all probe signals,
<code class="reqn">\theta_i</code> are the chip effects for arrays <code class="reqn">i=1,...,I</code>,
and <code class="reqn">\phi_k</code> are the probe affinities for probes <code class="reqn">k=1,...,K</code>.
The <code class="reqn">\varepsilon_{ik}</code> are zero-mean noise with equal variance.
The model is constrained such that <code class="reqn">\prod_k \phi_k = 1</code>.
</p>
<p>Note that with the additional constraint <code class="reqn">a=0</code> (see arguments above),
the above model is very similar to <code><a href="#topic+MbeiPlm">MbeiPlm</a></code>.  The differences in
parameter estimates is due to difference is assumptions about the
error structure, which in turn affects how the model is estimated.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>Bengtsson &amp; Hossjer (2006). <br />
</p>

<hr>
<h2 id='AffineSnpPlm'>The AffineSnpPlm class</h2><span id='topic+AffineSnpPlm'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AffineSnpPlm</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffinePlm">AffinePlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+SnpPlm">SnpPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AffineSnpPlm</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AffineCnPlm">AffineCnPlm</a></em><br />
</p>
<p>public abstract static class <b>AffineSnpPlm</b><br />
extends <a href="#topic+SnpPlm">SnpPlm</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AffineSnpPlm(..., mergeStrands=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffineSnpPlm_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AffinePlm">AffinePlm</a></code>.</p>
</td></tr>
<tr><td><code id="AffineSnpPlm_+3A_mergestrands">mergeStrands</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the sense and the anti-sense strands are
fitted together, otherwise separately.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from SnpPlm</b>:<br />
getCellIndices, getChipEffectSet, getMergeStrands, getParameters, getProbeAffinityFile, setMergeStrands
</p>
<p><b>Methods inherited from AffinePlm</b>:<br />
getAsteriskTags, getFitUnitGroupFunction, getProbeAffinityFile
</p>
<p><b>Methods inherited from ProbeLevelModel</b>:<br />
calculateResidualSet, calculateWeights, fit, getAsteriskTags, getCalculateResidualsFunction, getChipEffectSet, getProbeAffinityFile, getResidualSet, getRootPath, getWeightsSet
</p>
<p><b>Methods inherited from MultiArrayUnitModel</b>:<br />
getListOfPriors, setListOfPriors, validate
</p>
<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AffymetrixCdfFile'>The AffymetrixCdfFile class</h2><span id='topic+AffymetrixCdfFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AffymetrixCdfFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaChipTypeAnnotationFile">AromaChipTypeAnnotationFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+UnitAnnotationDataFile">UnitAnnotationDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+UnitTypesFile">UnitTypesFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+UnitNamesFile">UnitNamesFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AffymetrixCdfFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>AffymetrixCdfFile</b><br />
extends <em><a href="aroma.core.html#topic+UnitNamesFile">UnitNamesFile</a></em><br />
</p>
<p>An AffymetrixCdfFile object represents a generic Affymetrix CDF file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AffymetrixCdfFile(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AromaChipTypeAnnotationFile">AromaChipTypeAnnotationFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>convert</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>createExonByTranscriptCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getACSFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAromaCellSequenceFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCellIndices</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getDimension</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFileFormat</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getGenomeInformation</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getImage</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPlatform</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitNames</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitTypes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>hasUnitTypes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>isMonocellCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>isPm</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>isUniqueCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfCells</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfColumns</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfQcUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfRows</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readDataFrame</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from UnitNamesFile</b>:<br />
getUnitNames, indexOf, nbrOfUnits
</p>
<p><b>Methods inherited from UnitTypesFile</b>:<br />
getUnitTypes, nbrOfUnits
</p>
<p><b>Methods inherited from UnitAnnotationDataFile</b>:<br />
byChipType, getAromaUflFile, getAromaUgpFile, getChipType, getDefaultExtension, getPlatform, nbrOfUnits
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaChipTypeAnnotationFile</b>:<br />
as.character, byChipType, byName, findByChipType, fromFile, getChipType, getDefaultExtension, getHeader, getPlatform
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson, Ken Simpson</p>

<hr>
<h2 id='AffymetrixCdfFile+24findByChipType'>Locates a CDF file from its chip type</h2><span id='topic+AffymetrixCdfFile+24findByChipType'></span><span id='topic+findByChipType.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.findByChipType'></span><span id='topic+findByChipType+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Locates a CDF file from its chip type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## AffymetrixCdfFile$findByChipType(chipType, tags=NULL, pattern=NULL, ...,
##   .useAffxparser=TRUE)

## Don't use the below:
## S3 method for class 'AffymetrixCdfFile'
findByChipType(static, chipType, tags=NULL, pattern=NULL, ..., .useAffxparser=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffymetrixCdfFile+2B24findByChipType_+3A_chiptype">chipType</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string.</p>
</td></tr>
<tr><td><code id="AffymetrixCdfFile+2B24findByChipType_+3A_tags">tags</code></td>
<td>
<p>An optional <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> of tags.</p>
</td></tr>
<tr><td><code id="AffymetrixCdfFile+2B24findByChipType_+3A_pattern">pattern</code></td>
<td>
<p>An optional <code><a href="base.html#topic+character">character</a></code> string.</p>
</td></tr>
<tr><td><code id="AffymetrixCdfFile+2B24findByChipType_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="AffymetrixCdfFile+2B24findByChipType_+3A_.useaffxparser">.useAffxparser</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, <code><a href="affxparser.html#topic+findCdf">findCdf</a></code> is used if
the CDF could not be located.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a pathname as a <code><a href="base.html#topic+character">character</a></code> string to the first CDF file found.
If non CDF with requested chip type was found, <code><a href="base.html#topic+NULL">NULL</a></code> is returned.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='AffymetrixCdfFile+24fromFile'>Defines an AffymetrixCdfFile object from a CDF file</h2><span id='topic+AffymetrixCdfFile+24fromFile'></span><span id='topic+fromFile.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.fromFile'></span><span id='topic+fromFile+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Defines an AffymetrixCdfFile object from a CDF file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## AffymetrixCdfFile$fromFile(filename, path=NULL, ...)

## Don't use the below:
## S3 method for class 'AffymetrixCdfFile'
fromFile(static, filename, path=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffymetrixCdfFile+2B24fromFile_+3A_filename">filename</code></td>
<td>
<p>The filename of to the file.</p>
</td></tr>
<tr><td><code id="AffymetrixCdfFile+2B24fromFile_+3A_path">path</code></td>
<td>
<p>The path to the file.</p>
</td></tr>
<tr><td><code id="AffymetrixCdfFile+2B24fromFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an instance of <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> or its subclasses.
If the file is not found or if it is of the wrong file format, an
error is thrown.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='AffymetrixCelFile'>The AffymetrixCelFile class</h2><span id='topic+AffymetrixCelFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AffymetrixCelFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AffymetrixCelFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+ChipEffectFile">ChipEffectFile</a></em>, <em><a href="#topic+CnChipEffectFile">CnChipEffectFile</a></em>, <em><a href="#topic+CnProbeAffinityFile">CnProbeAffinityFile</a></em>, <em><a href="#topic+ExonChipEffectFile">ExonChipEffectFile</a></em>, <em><a href="#topic+ExonProbeAffinityFile">ExonProbeAffinityFile</a></em>, <em><a href="#topic+FirmaFile">FirmaFile</a></em>, <em><a href="#topic+ParameterCelFile">ParameterCelFile</a></em>, <em><a href="#topic+ProbeAffinityFile">ProbeAffinityFile</a></em>, <em><a href="#topic+QualityAssessmentFile">QualityAssessmentFile</a></em>, <em><a href="#topic+ResidualFile">ResidualFile</a></em>, <em><a href="#topic+SnpChipEffectFile">SnpChipEffectFile</a></em>, <em><a href="#topic+SnpProbeAffinityFile">SnpProbeAffinityFile</a></em>, <em><a href="#topic+WeightsFile">WeightsFile</a></em><br />
</p>
<p>public abstract static class <b>AffymetrixCelFile</b><br />
extends <em><a href="#topic+AffymetrixFile">AffymetrixFile</a></em><br />
</p>
<p>An AffymetrixCelFile object represents a single Affymetrix CEL file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AffymetrixCelFile(..., cdf=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffymetrixCelFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code>.</p>
</td></tr>
<tr><td><code id="AffymetrixCelFile_+3A_cdf">cdf</code></td>
<td>
<p>An optional <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> making it possible to
override the default CDF file as specified by the CEL file header.
The requirement is that its number of cells must match that of
the CEL file.
If <code><a href="base.html#topic+NULL">NULL</a></code>, the CDF structure is inferred from the the chip type
as specified in the CEL file header.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractMatrix</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAm</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFileFormat</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getImage</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitNamesFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitTypesFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>image270</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfCells</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plotDensity</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plotImage</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plotMvsA</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plotMvsX</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>smoothScatterMvsA</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>writeImage</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>For developers</h3>

<p>If you subclass this class, please make sure to query the
<code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> object (see <code><a href="#topic+getCdf.AffymetrixCelFile">*getCdf</a>()</code>) whenever
querying CDF information.  Do not use the CDF file inferred from the
chip type in CEL header, unless you really want it to be hardwired that
way, otherwise you will break to possibility to override the CDF
structure.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>An object of this class is typically part of an <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>

<hr>
<h2 id='AffymetrixCelFile+24allocateFromCdf'>Creates an empty CEL file from a template CDF</h2><span id='topic+AffymetrixCelFile+24allocateFromCdf'></span><span id='topic+allocateFromCdf.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.allocateFromCdf'></span><span id='topic+allocateFromCdf+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Creates an empty CEL file from a template CDF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## AffymetrixCelFile$allocateFromCdf(cdf, name, tags=NULL, path=".", suffix=".CEL", ...,
##   overwrite=FALSE, verbose=FALSE)

## Don't use the below:
## S3 method for class 'AffymetrixCelFile'
allocateFromCdf(static, cdf, name, tags=NULL, path=".", suffix=".CEL", ...,
  overwrite=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffymetrixCelFile+2B24allocateFromCdf_+3A_cdf">cdf</code></td>
<td>
<p>A <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> to be used as a template.</p>
</td></tr>
<tr><td><code id="AffymetrixCelFile+2B24allocateFromCdf_+3A_name">name</code>, <code id="AffymetrixCelFile+2B24allocateFromCdf_+3A_tags">tags</code></td>
<td>
<p>The name and the tags of the file created.</p>
</td></tr>
<tr><td><code id="AffymetrixCelFile+2B24allocateFromCdf_+3A_path">path</code></td>
<td>
<p>The directory where the file is created.</p>
</td></tr>
<tr><td><code id="AffymetrixCelFile+2B24allocateFromCdf_+3A_suffix">suffix</code></td>
<td>
<p>Filename suffix.</p>
</td></tr>
<tr><td><code id="AffymetrixCelFile+2B24allocateFromCdf_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="affxparser.html#topic+createCel">createCel</a></code>.</p>
</td></tr>
<tr><td><code id="AffymetrixCelFile+2B24allocateFromCdf_+3A_overwrite">overwrite</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">FALSE</a></code> and the file already exists, then an
error is thrown.</p>
</td></tr>
<tr><td><code id="AffymetrixCelFile+2B24allocateFromCdf_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>

<hr>
<h2 id='AffymetrixCelFile+24clearData'>Clears all or a subset of the fields in a CEL file</h2><span id='topic+AffymetrixCelFile+24clearData'></span><span id='topic+clearData.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.clearData'></span><span id='topic+clearData+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Clears all or a subset of the fields in a CEL file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## AffymetrixCelFile$clearData(fields=c("intensities", "stdvs", "pixels"), value=0, ...,
##   .forSure=FALSE, verbose=TRUE)

## Don't use the below:
## S3 method for class 'AffymetrixCelFile'
clearData(this, fields=c("intensities", "stdvs", "pixels"), value=0, ..., .forSure=FALSE,
  verbose=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffymetrixCelFile+2B24clearData_+3A_fields">fields</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> of fields to be cleared.</p>
</td></tr>
<tr><td><code id="AffymetrixCelFile+2B24clearData_+3A_value">value</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> value to be written over the data.</p>
</td></tr>
<tr><td><code id="AffymetrixCelFile+2B24clearData_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="AffymetrixCelFile+2B24clearData_+3A_.forsure">.forSure</code></td>
<td>
<p>If not <code><a href="base.html#topic+logical">TRUE</a></code>, an exception is thrown asking if the
method was called by mistake.</p>
</td></tr>
<tr><td><code id="AffymetrixCelFile+2B24clearData_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns (invisibly) the names of the fields cleared.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
Internally, <code><a href="affxparser.html#topic+updateCel">updateCel</a></code> is used.
</p>

<hr>
<h2 id='AffymetrixCelFile+24fromFile'>Defines an AffymetrixCelFile object from a CEL file</h2><span id='topic+AffymetrixCelFile+24fromFile'></span><span id='topic+fromFile.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.fromFile'></span><span id='topic+fromFile+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Defines an AffymetrixCelFile object from a CEL file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## AffymetrixCelFile$fromFile(filename, path=NULL, ..., verbose=FALSE, .checkArgs=TRUE)

## Don't use the below:
## S3 method for class 'AffymetrixCelFile'
fromFile(static, filename, path=NULL, ..., verbose=FALSE, .checkArgs=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffymetrixCelFile+2B24fromFile_+3A_filename">filename</code></td>
<td>
<p>The filename of to the file.</p>
</td></tr>
<tr><td><code id="AffymetrixCelFile+2B24fromFile_+3A_path">path</code></td>
<td>
<p>The path to the file.</p>
</td></tr>
<tr><td><code id="AffymetrixCelFile+2B24fromFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code> object.
If the file is not found or if it is of the wrong file format, an
error is thrown.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>

<hr>
<h2 id='AffymetrixCelSet'>The AffymetrixCelSet class</h2><span id='topic+AffymetrixCelSet'></span><span id='topic+nbrOfArrays.AffymetrixCelSet'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AffymetrixCelSet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataSet">AromaMicroarrayDataSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>AffymetrixCelSet</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+ChipEffectSet">ChipEffectSet</a>, <a href="#topic+CnChipEffectSet">CnChipEffectSet</a>, <a href="#topic+ExonChipEffectSet">ExonChipEffectSet</a>, <a href="#topic+FirmaSet">FirmaSet</a>, <a href="#topic+ParameterCelSet">ParameterCelSet</a>, <a href="#topic+QualityAssessmentSet">QualityAssessmentSet</a>, <a href="#topic+ResidualSet">ResidualSet</a>, <a href="#topic+SnpChipEffectSet">SnpChipEffectSet</a>, <a href="#topic+WeightsSet">WeightsSet</a><br />
</p>
<p>public static class <b>AffymetrixCelSet</b><br />
extends <em><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></em><br />
</p>
<p>An AffymetrixCelSet object represents a set of Affymetrix CEL files
with <em>identical</em> chip types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AffymetrixCelSet(files=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffymetrixCelSet_+3A_files">files</code></td>
<td>
<p>A <code><a href="base.html#topic+list">list</a></code> of <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>:s.</p>
</td></tr>
<tr><td><code id="AffymetrixCelSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as.AffymetrixCelSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>doCRMAv1</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>doCRMAv2</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>doFIRMA</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>doGCRMA</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>doRMA</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractAffyBatch</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractFeatureSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractMatrix</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAverage</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAverageAsinh</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAverageFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAverageLog</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getData</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getIntensities</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPlatform</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getTimestamps</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitGroupCellMap</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitIntensities</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitNamesFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitTypesFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>justSNPRMA</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plotDensity</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>writeSgr</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AffymetrixFileSet</b>:<br />
as, as.AffymetrixFileSet, byPath, getDefaultFullName
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataSet</b>:<br />
as.AromaMicroarrayDataSetList, as.AromaMicroarrayDataSetTuple, getAromaFullNameTranslatorSet, getAverageFile, getChipType, getDefaultFullName, getPlatform, setAttributesBy, setAttributesBySampleAnnotationFile, setAttributesBySampleAnnotationSet, validate
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  for (zzz in 0) {

# Find any dataset
path &lt;- NULL
if (is.null(path))
  break

# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Define a dataset object based on all CEL files in a directory
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
ds &lt;- AffymetrixCelSet$fromFiles(path)
print(ds)

# Keep at most three arrays for this example
ds &lt;- ds[1:min(3,nbrOfArrays(ds))]
print(ds)

} # for (zzz in 0)
rm(zzz)


## End(Not run)</code></pre>

<hr>
<h2 id='AffymetrixCelSet+24importFromDChip'>Imports dChip-exported CEL files</h2><span id='topic+AffymetrixCelSet+24importFromDChip'></span><span id='topic+importFromDChip.AffymetrixCelSet'></span><span id='topic+AffymetrixCelSet.importFromDChip'></span><span id='topic+importFromDChip+2CAffymetrixCelSet-method'></span>

<h3>Description</h3>

<p>Imports dChip-exported CEL files into a directory structure recognized by this package.
ASCII CEL files are converted to binary CEL files, and for chip types
where the array data is rotated 90-degrees counter clockwise by dChip,
the data is rotated back.
</p>
<p>As of 2007-03-28, dChip rotates data for exon, tiling, and
Mapping 500K arrays.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## AffymetrixCelSet$importFromDChip(path, name=NULL, tags="dChip", rootPath="probeData",
##   rotateBack=NA, ..., skip=TRUE, verbose=FALSE)

## Don't use the below:
## S3 method for class 'AffymetrixCelSet'
importFromDChip(static, path, name=NULL, tags="dChip", rootPath="probeData",
  rotateBack=NA, ..., skip=TRUE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffymetrixCelSet+2B24importFromDChip_+3A_path">path</code></td>
<td>
<p>The path to all dChip-exported CEL files.</p>
</td></tr>
<tr><td><code id="AffymetrixCelSet+2B24importFromDChip_+3A_name">name</code></td>
<td>
<p>The name of the output data set.
If <code><a href="base.html#topic+NULL">NULL</a></code>, the name is inferred from the source path.</p>
</td></tr>
<tr><td><code id="AffymetrixCelSet+2B24importFromDChip_+3A_tags">tags</code></td>
<td>
<p>Tags added to the imported data set.</p>
</td></tr>
<tr><td><code id="AffymetrixCelSet+2B24importFromDChip_+3A_rootpath">rootPath</code></td>
<td>
<p>The root path where to store the data set.</p>
</td></tr>
<tr><td><code id="AffymetrixCelSet+2B24importFromDChip_+3A_rotateback">rotateBack</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the dChip-rotated array data is rotated
back. If <code><a href="base.html#topic+NA">NA</a></code>, this is inferred from the chip type name.</p>
</td></tr>
<tr><td><code id="AffymetrixCelSet+2B24importFromDChip_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>byPath()</code>.</p>
</td></tr>
<tr><td><code id="AffymetrixCelSet+2B24importFromDChip_+3A_skip">skip</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, already converted files are not re-converted.</p>
</td></tr>
<tr><td><code id="AffymetrixCelSet+2B24importFromDChip_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that dChip stores CEL intensities as 16-bit integers in its *.dcp
files, although CEL files are capable or holding floats.
This means that you might loose precision if first import data in to
dChip and then export it data again.
</p>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>

<hr>
<h2 id='AffymetrixCelSetReporter'>The AffymetrixCelSetReporter class</h2><span id='topic+AffymetrixCelSetReporter'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AffymetrixCelSetReporter</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+GenericReporter">GenericReporter</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+AffymetrixFileSetReporter">AffymetrixFileSetReporter</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><em><code>AffymetrixCelSetReporter</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+SpatialReporter">SpatialReporter</a></em><br />
</p>
<p>public abstract static class <b>AffymetrixCelSetReporter</b><br />
extends <em><a href="#topic+AffymetrixFileSetReporter">AffymetrixFileSetReporter</a></em><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AffymetrixCelSetReporter(..., .setClass="AffymetrixCelSet")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffymetrixCelSetReporter_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AffymetrixFileSetReporter">AffymetrixFileSetReporter</a></code>.</p>
</td></tr>
<tr><td><code id="AffymetrixCelSetReporter_+3A_.setclass">.setClass</code></td>
<td>
<p>The name of the class of the input set.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getDataSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AffymetrixFileSetReporter</b>:<br />
getFileSet, getInputName, getInputTags
</p>
<p><b>Methods inherited from GenericReporter</b>:<br />
as.character, getAlias, getAsteriskTags, getFullName, getInputName, getInputTags, getMainPath, getName, getPath, getReportSet, getRootPath, getTags, process, setAlias, setup
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AffymetrixCelSetTuple'>The AffymetrixCelSetTuple class</h2><span id='topic+AffymetrixCelSetTuple'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AffymetrixCelSetTuple</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSetList">GenericDataFileSetList</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataSetTuple">AromaMicroarrayDataSetTuple</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code>AffymetrixCelSetTuple</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+ChipEffectSetTuple">ChipEffectSetTuple</a>, <a href="#topic+CnChipEffectSetTuple">CnChipEffectSetTuple</a><br />
</p>
<p>public static class <b>AffymetrixCelSetTuple</b><br />
extends <em><a href="aroma.core.html#topic+AromaMicroarrayDataSetTuple">AromaMicroarrayDataSetTuple</a></em><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AffymetrixCelSetTuple(..., .setClass="AffymetrixCelSet")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffymetrixCelSetTuple_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="aroma.core.html#topic+AromaMicroarrayDataSetTuple">AromaMicroarrayDataSetTuple</a></code>.</p>
</td></tr>
<tr><td><code id="AffymetrixCelSetTuple_+3A_.setclass">.setClass</code></td>
<td>
<p>The name of the class of the input set.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byPath</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AromaMicroarrayDataSetTuple</b>:<br />
as, as.AromaMicroarrayDataSetTuple, byPath, getAsteriskTags, getChipTypes, getFullNames, getSets, getTags, indexOf, nbrOfChipTypes
</p>
<p><b>Methods inherited from GenericDataFileSetList</b>:<br />
as, as.GenericDataFileSetList, as.character, as.data.frame, as.list, assertDuplicates, clone, extract, getAsteriskTags, getDefaultFullName, getFileList, getFileListClass, getFullNames, getNames, getSet, getSets, getTags, indexOf, length, nbrOfFiles, nbrOfSets, setTags
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AffymetrixCnChpSet'>The AffymetrixCnChpSet class</h2><span id='topic+AffymetrixCnChpSet'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AffymetrixCnChpSet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataSet">AromaMicroarrayDataSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AffymetrixCnChpSet</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>AffymetrixCnChpSet</b><br />
extends <em><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></em><br />
</p>
<p>A AffymetrixCnChpSet object represents a set of AffymetrixCnChpFile:s
with <em>identical</em> chip types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AffymetrixCnChpSet(files=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffymetrixCnChpSet_+3A_files">files</code></td>
<td>
<p>A <code><a href="base.html#topic+list">list</a></code> of <code><a href="#topic+AffymetrixCnChpFile">AffymetrixCnChpFile</a></code>:s.</p>
</td></tr>
<tr><td><code id="AffymetrixCnChpSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as.AffymetrixCnChpSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>exportTotalCnRatioSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractLogRatios</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>findByName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AffymetrixFileSet</b>:<br />
as, as.AffymetrixFileSet, byPath, getDefaultFullName
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataSet</b>:<br />
as.AromaMicroarrayDataSetList, as.AromaMicroarrayDataSetTuple, getAromaFullNameTranslatorSet, getAverageFile, getChipType, getDefaultFullName, getPlatform, setAttributesBy, setAttributesBySampleAnnotationFile, setAttributesBySampleAnnotationSet, validate
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+AffymetrixCnChpFile">AffymetrixCnChpFile</a></code>.
</p>

<hr>
<h2 id='AffymetrixFile'>The abstract AffymetrixFile class</h2><span id='topic+AffymetrixFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AffymetrixFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AffymetrixFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></em>, <em><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></em>, <em><a href="#topic+AffymetrixCnChpFile">AffymetrixCnChpFile</a></em>, <em><a href="#topic+AffymetrixPgfFile">AffymetrixPgfFile</a></em>, <em><a href="#topic+AffymetrixTsvFile">AffymetrixTsvFile</a></em>, <em><a href="#topic+AromaChipTypeAnnotationFile">AromaChipTypeAnnotationFile</a></em>, <em><a href="#topic+ChipEffectFile">ChipEffectFile</a></em>, <em><a href="#topic+CnChipEffectFile">CnChipEffectFile</a></em>, <em><a href="#topic+CnProbeAffinityFile">CnProbeAffinityFile</a></em>, <em><a href="#topic+CnagCfhFile">CnagCfhFile</a></em>, <em><a href="#topic+DChipCdfBinFile">DChipCdfBinFile</a></em>, <em><a href="#topic+DChipDcpFile">DChipDcpFile</a></em>, <em><a href="#topic+ExonChipEffectFile">ExonChipEffectFile</a></em>, <em><a href="#topic+ExonProbeAffinityFile">ExonProbeAffinityFile</a></em>, <em><a href="#topic+FirmaFile">FirmaFile</a></em>, <em><a href="#topic+ParameterCelFile">ParameterCelFile</a></em>, <em><a href="#topic+ProbeAffinityFile">ProbeAffinityFile</a></em>, <em><a href="#topic+QualityAssessmentFile">QualityAssessmentFile</a></em>, <em><a href="#topic+ResidualFile">ResidualFile</a></em>, <em><a href="#topic+SnpChipEffectFile">SnpChipEffectFile</a></em>, <em><a href="#topic+SnpProbeAffinityFile">SnpProbeAffinityFile</a></em>, <em><a href="#topic+WeightsFile">WeightsFile</a></em><br />
</p>
<p>public abstract static class <b>AffymetrixFile</b><br />
extends <em><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></em><br />
</p>
<p>An AffymetrixFile object represents a single Affymetrix file,
e.g. an Affymetrix CEL file or an Affymetrix CDF file.
Note that this class is abstract and can not be instanciated, but
instead you have to use one of the subclasses or the generic
<code>fromFile()</code> method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AffymetrixFile(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffymetrixFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>An object of this class is typically part of an <code><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></code>.
</p>

<hr>
<h2 id='AffymetrixFileSet'>The AffymetrixFileSet class</h2><span id='topic+AffymetrixFileSet'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AffymetrixFileSet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataSet">AromaMicroarrayDataSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AffymetrixFileSet</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a>, <em><a href="#topic+AffymetrixCnChpSet">AffymetrixCnChpSet</a></em>, <a href="#topic+ChipEffectSet">ChipEffectSet</a>, <a href="#topic+CnChipEffectSet">CnChipEffectSet</a>, <em><a href="#topic+DChipDcpSet">DChipDcpSet</a></em>, <a href="#topic+ExonChipEffectSet">ExonChipEffectSet</a>, <a href="#topic+FirmaSet">FirmaSet</a>, <a href="#topic+ParameterCelSet">ParameterCelSet</a>, <a href="#topic+QualityAssessmentSet">QualityAssessmentSet</a>, <a href="#topic+ResidualSet">ResidualSet</a>, <a href="#topic+SnpChipEffectSet">SnpChipEffectSet</a>, <a href="#topic+WeightsSet">WeightsSet</a><br />
</p>
<p>public abstract static class <b>AffymetrixFileSet</b><br />
extends <em><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></em><br />
</p>
<p>An AffymetrixFileSet object represents a set of <code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code>s
with <em>identical</em> chip types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AffymetrixFileSet(files=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffymetrixFileSet_+3A_files">files</code></td>
<td>
<p>A <code><a href="base.html#topic+list">list</a></code> of <code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code>:s.</p>
</td></tr>
<tr><td><code id="AffymetrixFileSet_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as.AffymetrixFileSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byPath</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataSet</b>:<br />
as.AromaMicroarrayDataSetList, as.AromaMicroarrayDataSetTuple, getAromaFullNameTranslatorSet, getAverageFile, getChipType, getDefaultFullName, getPlatform, setAttributesBy, setAttributesBySampleAnnotationFile, setAttributesBySampleAnnotationSet, validate
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AffymetrixFileSet+24byPath'>Defines an AffymetrixFileSet object by searching for Affymetrix files</h2><span id='topic+AffymetrixFileSet+24byPath'></span><span id='topic+byPath.AffymetrixFileSet'></span><span id='topic+AffymetrixFileSet.byPath'></span><span id='topic+byPath+2CAffymetrixFileSet-method'></span>

<h3>Description</h3>

<p>Defines an AffymetrixFileSet object by searching for Affymetrix files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## AffymetrixFileSet$byPath(..., fileClass="AffymetrixFile")

## Don't use the below:
## S3 method for class 'AffymetrixFileSet'
byPath(static, ..., fileClass="AffymetrixFile")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffymetrixFileSet+2B24byPath_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the constructor of the static
(calling) class.</p>
</td></tr>
<tr><td><code id="AffymetrixFileSet+2B24byPath_+3A_fileclass">fileClass</code></td>
<td>
<p>The name of the <code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code> class.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></code> object.
</p>


<h3>Reserved filenames</h3>

<p>Note that files with names starting with a period <code>.</code> are not
searched for.  The reason for this is that such files are reserved for
internal use of this package.  For instance, the package store average
signals across CEL files in a file named as <code>.average&lt;something&gt;.CEL</code>
in the same directory as the CEL files, and when such a directory is
scanned we do not want such files to be interpreted as data.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></code>.
</p>

<hr>
<h2 id='AffymetrixFileSetReporter'>The AffymetrixFileSetReporter class</h2><span id='topic+AffymetrixFileSetReporter'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AffymetrixFileSetReporter</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+GenericReporter">GenericReporter</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><em><code>AffymetrixFileSetReporter</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AffymetrixCelSetReporter">AffymetrixCelSetReporter</a></em>, <em><a href="#topic+SpatialReporter">SpatialReporter</a></em><br />
</p>
<p>public abstract static class <b>AffymetrixFileSetReporter</b><br />
extends <em><a href="#topic+GenericReporter">GenericReporter</a></em><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AffymetrixFileSetReporter(set=NULL, ..., .setClass="AffymetrixFileSet")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffymetrixFileSetReporter_+3A_set">set</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></code> object.</p>
</td></tr>
<tr><td><code id="AffymetrixFileSetReporter_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+GenericReporter">GenericReporter</a></code>.</p>
</td></tr>
<tr><td><code id="AffymetrixFileSetReporter_+3A_.setclass">.setClass</code></td>
<td>
<p>The name of the class of the input set.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getInputName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getInputTags</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from GenericReporter</b>:<br />
as.character, getAlias, getAsteriskTags, getFullName, getInputName, getInputTags, getMainPath, getName, getPath, getReportSet, getRootPath, getTags, process, setAlias, setup
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AffymetrixPgfFile'>The AffymetrixPgfFile class</h2><span id='topic+AffymetrixPgfFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AffymetrixPgfFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AromaChipTypeAnnotationFile">AromaChipTypeAnnotationFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+UnitAnnotationDataFile">UnitAnnotationDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+UnitTypesFile">UnitTypesFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+UnitNamesFile">UnitNamesFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AffymetrixPgfFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>AffymetrixPgfFile</b><br />
extends <em><a href="aroma.core.html#topic+UnitNamesFile">UnitNamesFile</a></em><br />
</p>
<p>An AffymetrixPgfFile object represents a generic Affymetrix Probe Group
File (PGF).
A PGF file &quot;provides information about what probes are contained
within a probeset and information about the nature of the probes
necessary for analysis. The current PGF file format (version 1)
is only specified for expression style probesets.&quot; [1]
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AffymetrixPgfFile(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffymetrixPgfFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AromaChipTypeAnnotationFile">AromaChipTypeAnnotationFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getDimension</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPlatform</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitNames</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfCells</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfColumns</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfRows</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from UnitNamesFile</b>:<br />
getUnitNames, indexOf, nbrOfUnits
</p>
<p><b>Methods inherited from UnitTypesFile</b>:<br />
getUnitTypes, nbrOfUnits
</p>
<p><b>Methods inherited from UnitAnnotationDataFile</b>:<br />
byChipType, getAromaUflFile, getAromaUgpFile, getChipType, getDefaultExtension, getPlatform, nbrOfUnits
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaChipTypeAnnotationFile</b>:<br />
as.character, byChipType, byName, findByChipType, fromFile, getChipType, getDefaultExtension, getHeader, getPlatform
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] ... <br />
</p>

<hr>
<h2 id='AffymetrixPgfFile+24findByChipType'>Locates a PGF file from its chip type</h2><span id='topic+AffymetrixPgfFile+24findByChipType'></span><span id='topic+findByChipType.AffymetrixPgfFile'></span><span id='topic+AffymetrixPgfFile.findByChipType'></span><span id='topic+findByChipType+2CAffymetrixPgfFile-method'></span>

<h3>Description</h3>

<p>Locates a PGF file from its chip type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## AffymetrixPgfFile$findByChipType(chipType, tags=NULL, pattern=NULL, ...)

## Don't use the below:
## S3 method for class 'AffymetrixPgfFile'
findByChipType(static, chipType, tags=NULL, pattern=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffymetrixPgfFile+2B24findByChipType_+3A_chiptype">chipType</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string.</p>
</td></tr>
<tr><td><code id="AffymetrixPgfFile+2B24findByChipType_+3A_tags">tags</code></td>
<td>
<p>An optional <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> of tags.</p>
</td></tr>
<tr><td><code id="AffymetrixPgfFile+2B24findByChipType_+3A_pattern">pattern</code></td>
<td>
<p>An optional <code><a href="base.html#topic+character">character</a></code> string.</p>
</td></tr>
<tr><td><code id="AffymetrixPgfFile+2B24findByChipType_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a pathname as a <code><a href="base.html#topic+character">character</a></code> string to the first PGF file found.
If non PGF with requested chip type was found, <code><a href="base.html#topic+NULL">NULL</a></code> is returned.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixPgfFile">AffymetrixPgfFile</a></code>.
</p>

<hr>
<h2 id='AffymetrixPgfFile+24fromFile'>Defines an AffymetrixPgfFile object from a PGF file</h2><span id='topic+AffymetrixPgfFile+24fromFile'></span><span id='topic+fromFile.AffymetrixPgfFile'></span><span id='topic+AffymetrixPgfFile.fromFile'></span><span id='topic+fromFile+2CAffymetrixPgfFile-method'></span>

<h3>Description</h3>

<p>Defines an AffymetrixPgfFile object from a PGF file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## AffymetrixPgfFile$fromFile(filename, path=NULL, ...)

## Don't use the below:
## S3 method for class 'AffymetrixPgfFile'
fromFile(static, filename, path=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffymetrixPgfFile+2B24fromFile_+3A_filename">filename</code></td>
<td>
<p>The filename of to the file.</p>
</td></tr>
<tr><td><code id="AffymetrixPgfFile+2B24fromFile_+3A_path">path</code></td>
<td>
<p>The path to the file.</p>
</td></tr>
<tr><td><code id="AffymetrixPgfFile+2B24fromFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an instance of <code><a href="#topic+AffymetrixPgfFile">AffymetrixPgfFile</a></code> or its subclasses.
If the file is not found or if it is of the wrong file format, an
error is thrown.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixPgfFile">AffymetrixPgfFile</a></code>.
</p>

<hr>
<h2 id='AffymetrixProbeTabFile'>The AffymetrixProbeTabFile class</h2><span id='topic+AffymetrixProbeTabFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AffymetrixProbeTabFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+ColumnNamesInterface">ColumnNamesInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFile">GenericTabularFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+TabularTextFile">TabularTextFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AffymetrixProbeTabFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>AffymetrixProbeTabFile</b><br />
extends <em><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></em><br />
</p>
<p>An AffymetrixProbeTabFile represents an interface to query the data
contained in an Affymetrix probe tab file, e.g.
<code>Mapping250K_Nsp_probe_tab</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AffymetrixProbeTabFile(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AffymetrixProbeTabFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of <code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getColumnNames</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getData</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from TabularTextFile</b>:<br />
as.character, getCommentChar, getDefaultColumnClassPatterns, getDefaultColumnClasses, getDefaultColumnNames, getHeader, getReadArguments, hasColumnHeader, nbrOfLines, nbrOfRows, readColumnNames, readColumns, readDataFrame, readLines, readRawHeader, setCommentChar
</p>
<p><b>Methods inherited from GenericTabularFile</b>:<br />
[, as.character, dim, extractMatrix, head, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, tail, writeColumnsToFiles
</p>
<p><b>Methods inherited from ColumnNamesInterface</b>:<br />
appendColumnNamesTranslator, appendColumnNamesTranslatorByNULL, appendColumnNamesTranslatorBycharacter, appendColumnNamesTranslatorByfunction, appendColumnNamesTranslatorBylist, clearColumnNamesTranslator, clearListOfColumnNamesTranslators, getColumnNames, getColumnNamesTranslator, getDefaultColumnNames, getListOfColumnNamesTranslators, nbrOfColumns, setColumnNames, setColumnNamesTranslator, setListOfColumnNamesTranslators, updateColumnNames
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>About probe-tab files</h3>

<p>Probe-tab files are provided by Affymetrix and contains information
on the probes.  Note that not necessarily all probes are represented
in the files, e.g. typically only PM probes are given and MM are
left to be inferred from the PMs.
</p>
<p>The below is an extract of the <code>Mapping250K_Nsp_probe_tab</code> file
obtained from the Affymetrix website.  Note that columns are separated
by TABs.
</p>
<pre>
  SNP_A-1780270	1633	2398	3	TTGTTAAGCAAGTGAGTTATTTTAT	f	PM	C
  SNP_A-1780270	1633	2399	3	TTGTTAAGCAAGTGACTTATTTTAT	f	PM	G
  SNP_A-1780270	1951	1780	-4	GGATAAAATAAAATAACTCACTTGC	r	PM	C
  ...
  SNP_A-4241299	2553	1658	4	AAACACATTTTTGGGTCGTAAGGAA	f	PM	G
 </pre>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AlleleSummation'>The AlleleSummation class</h2><span id='topic+AlleleSummation'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AlleleSummation</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><em><code>AlleleSummation</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>AlleleSummation</b><br />
extends <em><a href="#topic+UnitModel">UnitModel</a></em><br />
</p>
<p>This class takes allele-specific chip effect estimates of a
SnpChipEffectSet and returns a CnChipEffectSet holding the summed
allele estimates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AlleleSummation(dataSet=NULL, ignoreNAs=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AlleleSummation_+3A_dataset">dataSet</code></td>
<td>
<p>A <code><a href="#topic+SnpChipEffectSet">SnpChipEffectSet</a></code>.</p>
</td></tr>
<tr><td><code id="AlleleSummation_+3A_ignorenas">ignoreNAs</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, missing values are excluded when summing
the signals from the two alleles.</p>
</td></tr>
<tr><td><code id="AlleleSummation_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+UnitModel">UnitModel</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>findUnitsTodo</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipEffectSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AllelicCrosstalkCalibration'>The AllelicCrosstalkCalibration class</h2><span id='topic+AllelicCrosstalkCalibration'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AllelicCrosstalkCalibration</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code>AllelicCrosstalkCalibration</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>AllelicCrosstalkCalibration</b><br />
extends <em><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></em><br />
</p>
<p>This class represents a calibration function that transforms the
probe-level signals such that the signals from the two alleles are
orthogonal.
The method fits and calibrates PM signals only.  MM signals will not
affect the model fitting and are unaffected.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AllelicCrosstalkCalibration(dataSet=NULL, ..., model=c("asis", "auto", "CRMA", "CRMAv2"),
  rescaleBy=c("auto", "groups", "all", "none"), targetAvg=c(2200, 2200),
  subsetToAvg="-XY", mergeShifts=TRUE, B=1, flavor=c("sfit", "expectile"),
  alpha=c(0.1, 0.075, 0.05, 0.03, 0.01), q=2, Q=98, lambda=2,
  pairBy=c("CDF", "sequence"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AllelicCrosstalkCalibration_+3A_dataset">dataSet</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.</p>
</td></tr>
<tr><td><code id="AllelicCrosstalkCalibration_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code>.</p>
</td></tr>
<tr><td><code id="AllelicCrosstalkCalibration_+3A_model">model</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string for quickly specifying default
parameter settings.</p>
</td></tr>
<tr><td><code id="AllelicCrosstalkCalibration_+3A_rescaleby">rescaleBy</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying what sets of cells
should be rescaled towards a target average, if any.
Default is to rescale all cells together.
If <code>"none"</code>, no rescaling is done.</p>
</td></tr>
<tr><td><code id="AllelicCrosstalkCalibration_+3A_targetavg">targetAvg</code></td>
<td>
<p>The signal(s) that either the average of the sum
(if one target value) or the average of each of the alleles
(if two target values) should have after calibration.
Only used if <code>rescaleBy != "none"</code>.</p>
</td></tr>
<tr><td><code id="AllelicCrosstalkCalibration_+3A_subsettoavg">subsetToAvg</code></td>
<td>
<p>The indices of the cells (taken as the intersect of
existing indices) used to calculate average in order to rescale to
the target average. If <code><a href="base.html#topic+NULL">NULL</a></code>, all probes are considered.</p>
</td></tr>
<tr><td><code id="AllelicCrosstalkCalibration_+3A_mergeshifts">mergeShifts</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the shift of the probe sequence
relative to the SNP position is ignored, otherwise not.</p>
</td></tr>
<tr><td><code id="AllelicCrosstalkCalibration_+3A_b">B</code></td>
<td>
<p>An <code><a href="base.html#topic+integer">integer</a></code> specifying by how many nucleotides the allelic
groups should be stratified by. If zero, all SNPs are put in one
group.</p>
</td></tr>
<tr><td><code id="AllelicCrosstalkCalibration_+3A_flavor">flavor</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying what algorithm is used
to fit the crosstalk calibration.</p>
</td></tr>
<tr><td><code id="AllelicCrosstalkCalibration_+3A_alpha">alpha</code>, <code id="AllelicCrosstalkCalibration_+3A_q">q</code>, <code id="AllelicCrosstalkCalibration_+3A_q">Q</code>, <code id="AllelicCrosstalkCalibration_+3A_lambda">lambda</code></td>
<td>
<p>Model fitting parameters.</p>
</td></tr>
<tr><td><code id="AllelicCrosstalkCalibration_+3A_pairby">pairBy</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying how allele probe pairs
are identified.</p>
</td></tr>
</table>


<h3>What probe signals are updated?</h3>

<p>Calibration for crosstalk between allele signals applies by definition
only SNP units.
Furthermore, it is only SNP units with two or four unit groups that
are calibrated.  For instance, in at least on custom SNP CDFs we
know of, there is a small number of SNP units that have six groups.
<em>Currently these units are not calibrated (at all).</em>
It is only PM probes that will be calibrated.
Note that, non-calibrated signals will be saved in the output files.
</p>


<h3>What probe signals are used to fit model?</h3>

<p>All PM probe pairs are used to fit the crosstalk model.
In the second step where signals are rescaled to a target average,
it is possible to specify the set of cells that should be included
when estimating the target average.
</p>


<h3>Important about rescaling towards target average</h3>

<p>Rescaling each allele-pair group (e.g. AC, AG, AT, CG, CT, GC)
towards a target average (<code>rescaleBy="groups"</code>)
<em>must not</em> be used for multi-enzyme chip types,
e.g. GenomeWideSNP_6.
If still done, due to confounded effects of non-perfect enzyme
mixtures etc, there will be a significant bias between raw CNs
for SNPs and CN probes.
Instead, for such chip types <em>all probe signals</em> should be
rescale together towards the target average (<code>rescaleBy="all"</code>).
</p>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ProbeLevelTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaChipTypeAnnotationFile'>The AromaChipTypeAnnotationFile class</h2><span id='topic+AromaChipTypeAnnotationFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AromaChipTypeAnnotationFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AromaChipTypeAnnotationFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></em>, <em><a href="#topic+AffymetrixPgfFile">AffymetrixPgfFile</a></em><br />
</p>
<p>public abstract static class <b>AromaChipTypeAnnotationFile</b><br />
extends <em><a href="#topic+AffymetrixFile">AffymetrixFile</a></em><br />
</p>
<p>An AromaChipTypeAnnotationFile object represents an annotation file for a
specific chip type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AromaChipTypeAnnotationFile(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AromaChipTypeAnnotationFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getHeader</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPlatform</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AromaChipTypeAnnotationFile+24byChipType'>Defines an AromaChipTypeAnnotationFile object by chip type</h2><span id='topic+AromaChipTypeAnnotationFile+24byChipType'></span><span id='topic+byChipType.AromaChipTypeAnnotationFile'></span><span id='topic+AromaChipTypeAnnotationFile.byChipType'></span><span id='topic+byChipType+2CAromaChipTypeAnnotationFile-method'></span><span id='topic+AromaChipTypeAnnotationFile.byName'></span><span id='topic+byName.AromaChipTypeAnnotationFile'></span><span id='topic+byName+2CAromaChipTypeAnnotationFile-method'></span>

<h3>Description</h3>

<p>Defines an AromaChipTypeAnnotationFile object by chip type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## AromaChipTypeAnnotationFile$byChipType(chipType, tags=NULL, ..., verbose=FALSE)

## Don't use the below:
## S3 method for class 'AromaChipTypeAnnotationFile'
byChipType(static, chipType, tags=NULL, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AromaChipTypeAnnotationFile+2B24byChipType_+3A_chiptype">chipType</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string.</p>
</td></tr>
<tr><td><code id="AromaChipTypeAnnotationFile+2B24byChipType_+3A_tags">tags</code></td>
<td>
<p>An optional <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> of tags.</p>
</td></tr>
<tr><td><code id="AromaChipTypeAnnotationFile+2B24byChipType_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="AromaChipTypeAnnotationFile+2B24byChipType_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AromaChipTypeAnnotationFile">AromaChipTypeAnnotationFile</a></code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fromFile.AromaChipTypeAnnotationFile">*fromFile</a>()</code>.
For more information see <code><a href="#topic+AromaChipTypeAnnotationFile">AromaChipTypeAnnotationFile</a></code>.
</p>

<hr>
<h2 id='AromaChipTypeAnnotationFile+24fromFile'>Sets up an AromaChipTypeAnnotationFile</h2><span id='topic+AromaChipTypeAnnotationFile+24fromFile'></span><span id='topic+fromFile.AromaChipTypeAnnotationFile'></span><span id='topic+AromaChipTypeAnnotationFile.fromFile'></span><span id='topic+fromFile+2CAromaChipTypeAnnotationFile-method'></span>

<h3>Description</h3>

<p>Sets up an AromaChipTypeAnnotationFile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## AromaChipTypeAnnotationFile$fromFile(filename, path=NULL, ...)

## Don't use the below:
## S3 method for class 'AromaChipTypeAnnotationFile'
fromFile(static, filename, path=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AromaChipTypeAnnotationFile+2B24fromFile_+3A_filename">filename</code></td>
<td>
<p>The filename of to the file.</p>
</td></tr>
<tr><td><code id="AromaChipTypeAnnotationFile+2B24fromFile_+3A_path">path</code></td>
<td>
<p>The path to the file.</p>
</td></tr>
<tr><td><code id="AromaChipTypeAnnotationFile+2B24fromFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an instance of <code><a href="#topic+AromaChipTypeAnnotationFile">AromaChipTypeAnnotationFile</a></code> or its subclasses.
If the file is not found or if it is of the wrong file format, an
error is thrown.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+byChipType.AromaChipTypeAnnotationFile">*byChipType</a>()</code>.
For more information see <code><a href="#topic+AromaChipTypeAnnotationFile">AromaChipTypeAnnotationFile</a></code>.
</p>

<hr>
<h2 id='AromaUnitTabularBinaryFile+24allocateFromCdf'>Creates an AromaUnitTabularBinaryFile mapping to a given CDF</h2><span id='topic+AromaUnitTabularBinaryFile+24allocateFromCdf'></span><span id='topic+allocateFromCdf.AromaUnitTabularBinaryFile'></span><span id='topic+AromaUnitTabularBinaryFile.allocateFromCdf'></span><span id='topic+allocateFromCdf+2CAromaUnitTabularBinaryFile-method'></span>

<h3>Description</h3>

<p>Creates an AromaUnitTabularBinaryFile mapping to a given CDF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## AromaUnitTabularBinaryFile$allocateFromCdf(cdf, ...)

## Don't use the below:
## S3 method for class 'AromaUnitTabularBinaryFile'
allocateFromCdf(static, cdf, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AromaUnitTabularBinaryFile+2B24allocateFromCdf_+3A_cdf">cdf</code></td>
<td>
<p>The <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> used as a template and from
which the (full) chip type is taken.</p>
</td></tr>
<tr><td><code id="AromaUnitTabularBinaryFile+2B24allocateFromCdf_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>allocate()</code> of
<code><a href="aroma.core.html#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="aroma.core.html#topic+AromaUnitTabularBinaryFile">AromaUnitTabularBinaryFile</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>To update to file footer afterwards, see <code>writeFooter()</code>.
</p>

<hr>
<h2 id='ArrayExplorer'>The ArrayExplorer class</h2><span id='topic+ArrayExplorer'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class ArrayExplorer</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+Explorer">Explorer</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><em><code>ArrayExplorer</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>ArrayExplorer</b><br />
extends <em><a href="aroma.core.html#topic+Explorer">Explorer</a></em><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ArrayExplorer(csTuple=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ArrayExplorer_+3A_cstuple">csTuple</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code> object.</p>
</td></tr>
<tr><td><code id="ArrayExplorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>addColorMap</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getColorMaps</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getDataSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getSetTuple</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfChipTypes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setArrays</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setColorMaps</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from Explorer</b>:<br />
addIncludes, addIndexFile, as.character, display, getAlias, getArraysOfInput, getAsteriskTags, getFullName, getIncludePath, getMainPath, getName, getNameOfInput, getNames, getPath, getReportPathPattern, getRootPath, getSampleLayerPrefix, getSubname, getTags, getTagsOfInput, getTemplatePath, getVersion, nbrOfArrays, process, setAlias, setArrays, setReportPathPattern, setSubname, setup, splitByReportPathPattern, updateSetupExplorerFile
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='as.AffymetrixCelSet.AffymetrixCelSet'>Coerce an object to an AffymetrixCelSet object</h2><span id='topic+as.AffymetrixCelSet.AffymetrixCelSet'></span><span id='topic+AffymetrixCelSet.as.AffymetrixCelSet'></span><span id='topic+as.AffymetrixCelSet+2CAffymetrixCelSet-method'></span><span id='topic+as.AffymetrixCelSet.list'></span><span id='topic+as.AffymetrixCelSet.default'></span>

<h3>Description</h3>

<p>Coerce an object to an AffymetrixCelSet object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelSet'
as.AffymetrixCelSet(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.AffymetrixCelSet.AffymetrixCelSet_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="base.html#topic+list.files">list.files</a></code>().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>

<hr>
<h2 id='as.AffymetrixCnChpSet.AffymetrixCnChpSet'>Coerce an object to an AffymetrixCnChpSet object</h2><span id='topic+as.AffymetrixCnChpSet.AffymetrixCnChpSet'></span><span id='topic+AffymetrixCnChpSet.as.AffymetrixCnChpSet'></span><span id='topic+as.AffymetrixCnChpSet+2CAffymetrixCnChpSet-method'></span><span id='topic+as.AffymetrixCnChpSet.list'></span><span id='topic+as.AffymetrixCnChpSet.default'></span>

<h3>Description</h3>

<p>Coerce an object to an AffymetrixCnChpSet object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCnChpSet'
as.AffymetrixCnChpSet(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.AffymetrixCnChpSet.AffymetrixCnChpSet_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="base.html#topic+list.files">list.files</a></code>().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AffymetrixCnChpSet">AffymetrixCnChpSet</a></code> object.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCnChpSet">AffymetrixCnChpSet</a></code>.
</p>

<hr>
<h2 id='as.AffymetrixFileSet.AffymetrixFileSet'>Coerce an object to an AffymetrixFileSet object</h2><span id='topic+as.AffymetrixFileSet.AffymetrixFileSet'></span><span id='topic+AffymetrixFileSet.as.AffymetrixFileSet'></span><span id='topic+as.AffymetrixFileSet+2CAffymetrixFileSet-method'></span><span id='topic+as.AffymetrixFileSet.list'></span><span id='topic+as.AffymetrixFileSet.default'></span>

<h3>Description</h3>

<p>Coerce an object to an AffymetrixFileSet object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixFileSet'
as.AffymetrixFileSet(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.AffymetrixFileSet.AffymetrixFileSet_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="base.html#topic+list.files">list.files</a></code>().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></code> object.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></code>.
</p>

<hr>
<h2 id='as.character.AffymetrixCelSet'>Returns a short string describing the Affymetrix CEL set</h2><span id='topic+as.character.AffymetrixCelSet'></span><span id='topic+AffymetrixCelSet.as.character'></span><span id='topic+as.character+2CAffymetrixCelSet-method'></span>

<h3>Description</h3>

<p>Returns a short string describing the Affymetrix CEL set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelSet'
as.character(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.character.AffymetrixCelSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>

<hr>
<h2 id='as.character.AffymetrixCnChpSet'>Returns a short string describing the set</h2><span id='topic+as.character.AffymetrixCnChpSet'></span><span id='topic+AffymetrixCnChpSet.as.character'></span><span id='topic+as.character+2CAffymetrixCnChpSet-method'></span>

<h3>Description</h3>

<p>Returns a short string describing the set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCnChpSet'
as.character(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.character.AffymetrixCnChpSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCnChpSet">AffymetrixCnChpSet</a></code>.
</p>

<hr>
<h2 id='as.character.CnagCfhSet'>Returns a short string describing the CNAG CFH set</h2><span id='topic+as.character.CnagCfhSet'></span><span id='topic+CnagCfhSet.as.character'></span><span id='topic+as.character+2CCnagCfhSet-method'></span>

<h3>Description</h3>

<p>Returns a short string describing the CNAG CFH set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CnagCfhSet'
as.character(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.character.CnagCfhSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+CnagCfhSet">CnagCfhSet</a></code>.
</p>

<hr>
<h2 id='as.character.DChipDcpSet'>Returns a short string describing the DChip CHP set</h2><span id='topic+as.character.DChipDcpSet'></span><span id='topic+DChipDcpSet.as.character'></span><span id='topic+as.character+2CDChipDcpSet-method'></span>

<h3>Description</h3>

<p>Returns a short string describing the DChip CHP set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'DChipDcpSet'
as.character(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.character.DChipDcpSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+DChipDcpSet">DChipDcpSet</a></code>.
</p>

<hr>
<h2 id='as.CnagCfhSet.CnagCfhSet'>Coerce an object to an CnagCfhSet object</h2><span id='topic+as.CnagCfhSet.CnagCfhSet'></span><span id='topic+CnagCfhSet.as.CnagCfhSet'></span><span id='topic+as.CnagCfhSet+2CCnagCfhSet-method'></span><span id='topic+as.CnagCfhSet.list'></span><span id='topic+as.CnagCfhSet.default'></span>

<h3>Description</h3>

<p>Coerce an object to an CnagCfhSet object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CnagCfhSet'
as.CnagCfhSet(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.CnagCfhSet.CnagCfhSet_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="base.html#topic+list.files">list.files</a></code>().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+CnagCfhSet">CnagCfhSet</a></code> object.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+CnagCfhSet">CnagCfhSet</a></code>.
</p>

<hr>
<h2 id='as.DChipDcpSet.DChipDcpSet'>Coerce an object to an DChipDcpSet object</h2><span id='topic+as.DChipDcpSet.DChipDcpSet'></span><span id='topic+DChipDcpSet.as.DChipDcpSet'></span><span id='topic+as.DChipDcpSet+2CDChipDcpSet-method'></span><span id='topic+as.DChipDcpSet.list'></span><span id='topic+as.DChipDcpSet.default'></span>

<h3>Description</h3>

<p>Coerce an object to an DChipDcpSet object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'DChipDcpSet'
as.DChipDcpSet(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.DChipDcpSet.DChipDcpSet_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="base.html#topic+list.files">list.files</a></code>().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+DChipDcpSet">DChipDcpSet</a></code> object.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+DChipDcpSet">DChipDcpSet</a></code>.
</p>

<hr>
<h2 id='averageQuantile.AffymetrixCelSet'>Calculates the average empirical distribution across all samples</h2><span id='topic+averageQuantile.AffymetrixCelSet'></span><span id='topic+AffymetrixCelSet.averageQuantile'></span><span id='topic+averageQuantile+2CAffymetrixCelSet-method'></span>

<h3>Description</h3>

<p>Calculates the average empirical distribution across all samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelSet'
averageQuantile(this, probes=NULL, excludeCells=NULL, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="averageQuantile.AffymetrixCelSet_+3A_probes">probes</code></td>
<td>
<p>An optional <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying what subset of
probes to be used to calculate the empirical distribution.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all probes are used.</p>
</td></tr>
<tr><td><code id="averageQuantile.AffymetrixCelSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="averageQuantile.AffymetrixCelSet_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This methods implements Step A2-A3 in the algorithm for quantile
normalization proposed by Bengtsson et al. (2008).
</p>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length N.
</p>


<h3>Missing values</h3>

<p>If <code><a href="base.html#topic+NA">NA</a></code>s are detected in a sample, these are excluded and the
<code>approx()</code> function (<code><a href="stats.html#topic+approx">approx</a></code>) is used to &quot;expand&quot;
the <code><a href="base.html#topic+vector">vector</a></code> of the remaining values so that the sorted <code><a href="base.html#topic+vector">vector</a></code>
(still) have length N.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] H. Bengtsson, R. Irizarry, B. Carvalho, &amp; T.P. Speed.
Estimation and assessment of raw copy numbers at the single
locus level, Bioinformatics, 2008.
</p>


<h3>See Also</h3>

<p><code><a href="aroma.light.html#topic+averageQuantile.list">averageQuantile.list</a></code>
For more information see <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>

<hr>
<h2 id='AvgCnPlm'>The AvgCnPlm class</h2><span id='topic+AvgCnPlm'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AvgCnPlm</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AvgPlm">AvgPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AvgSnpPlm">AvgSnpPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+SnpPlm">SnpPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CnPlm">CnPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AvgCnPlm</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>AvgCnPlm</b><br />
extends <a href="#topic+CnPlm">CnPlm</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AvgCnPlm(..., combineAlleles=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AvgCnPlm_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AvgSnpPlm">AvgSnpPlm</a></code>.</p>
</td></tr>
<tr><td><code id="AvgCnPlm_+3A_combinealleles">combineAlleles</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">FALSE</a></code>, allele A and allele B are treated
separately, otherwise together.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from CnPlm</b>:<br />
getCellIndices, getChipEffectSet, getCombineAlleles, getParameters, getProbeAffinityFile, setCombineAlleles
</p>
<p><b>Methods inherited from SnpPlm</b>:<br />
getCellIndices, getChipEffectSet, getMergeStrands, getParameters, getProbeAffinityFile, setMergeStrands
</p>
<p><b>Methods inherited from AvgSnpPlm</b>:<br />
getAsteriskTags
</p>
<p><b>Methods inherited from AvgPlm</b>:<br />
getAsteriskTags, getCalculateResidualsFunction, getParameters, validate
</p>
<p><b>Methods inherited from ProbeLevelModel</b>:<br />
calculateResidualSet, calculateWeights, fit, getAsteriskTags, getCalculateResidualsFunction, getChipEffectSet, getProbeAffinityFile, getResidualSet, getRootPath, getWeightsSet
</p>
<p><b>Methods inherited from MultiArrayUnitModel</b>:<br />
getListOfPriors, setListOfPriors, validate
</p>
<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AvgPlm'>The AvgPlm class</h2><span id='topic+AvgPlm'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AvgPlm</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AvgPlm</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AvgCnPlm">AvgCnPlm</a></em>, <em><a href="#topic+AvgSnpPlm">AvgSnpPlm</a></em><br />
</p>
<p>public abstract static class <b>AvgPlm</b><br />
extends <em><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></em><br />
</p>
<p>This class represents a PLM where the probe intensities are averaged
assuming identical probe affinities.
For instance, one may assume that replicated probes with identical
sequences have the same probe affinities, cf. the GenomeWideSNP_6
chip type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AvgPlm(..., flavor=c("median", "mean"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AvgPlm_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code>.</p>
</td></tr>
<tr><td><code id="AvgPlm_+3A_flavor">flavor</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying what model fitting algorithm
to be used.  This makes it possible to get identical estimates as other
packages.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from ProbeLevelModel</b>:<br />
calculateResidualSet, calculateWeights, fit, getAsteriskTags, getCalculateResidualsFunction, getChipEffectSet, getProbeAffinityFile, getResidualSet, getRootPath, getWeightsSet
</p>
<p><b>Methods inherited from MultiArrayUnitModel</b>:<br />
getListOfPriors, setListOfPriors, validate
</p>
<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Model</h3>

<p>For a single unit group, the averaging PLM of K probes is:
</p>
<p style="text-align: center;"><code class="reqn">y_{ik} = \theta_i + \varepsilon_{ik}</code>
</p>

<p>where <code class="reqn">\theta_i</code> are the chip effects for arrays <code class="reqn">i=1,...,I</code>.
The <code class="reqn">\varepsilon_{ik}</code> are zero-mean noise with equal variance.
</p>


<h3>Different flavors of model fitting</h3>

<p>The above model can be fitted in two ways, either robustly or
non-robustly.
Use argument <code>flavor="mean"</code> to fit the model non-robustly, i.e.
</p>
<p style="text-align: center;"><code class="reqn">\hat{\theta}_{i} = 1/K \sum_k y_{ik}</code>
</p>
<p>.
</p>
<p>Use argument <code>flavor="median"</code> to fit the model robustly, i.e.
</p>
<p style="text-align: center;"><code class="reqn">\hat{\theta}_{i} = median_k y_{ik}</code>
</p>
<p>.
</p>
<p>Missing values are always excluded.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='AvgSnpPlm'>The AvgSnpPlm class</h2><span id='topic+AvgSnpPlm'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class AvgSnpPlm</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AvgPlm">AvgPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+SnpPlm">SnpPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>AvgSnpPlm</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AvgCnPlm">AvgCnPlm</a></em><br />
</p>
<p>public abstract static class <b>AvgSnpPlm</b><br />
extends <a href="#topic+SnpPlm">SnpPlm</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AvgSnpPlm(..., mergeStrands=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AvgSnpPlm_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AvgPlm">AvgPlm</a></code>.</p>
</td></tr>
<tr><td><code id="AvgSnpPlm_+3A_mergestrands">mergeStrands</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the sense and the anti-sense strands are
fitted together, otherwise separately.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from SnpPlm</b>:<br />
getCellIndices, getChipEffectSet, getMergeStrands, getParameters, getProbeAffinityFile, setMergeStrands
</p>
<p><b>Methods inherited from AvgPlm</b>:<br />
getAsteriskTags, getCalculateResidualsFunction, getParameters, validate
</p>
<p><b>Methods inherited from ProbeLevelModel</b>:<br />
calculateResidualSet, calculateWeights, fit, getAsteriskTags, getCalculateResidualsFunction, getChipEffectSet, getProbeAffinityFile, getResidualSet, getRootPath, getWeightsSet
</p>
<p><b>Methods inherited from MultiArrayUnitModel</b>:<br />
getListOfPriors, setListOfPriors, validate
</p>
<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='BackgroundCorrection'>The BackgroundCorrection class</h2><span id='topic+BackgroundCorrection'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class BackgroundCorrection</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>BackgroundCorrection</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+GcRmaBackgroundCorrection">GcRmaBackgroundCorrection</a>, <a href="#topic+LimmaBackgroundCorrection">LimmaBackgroundCorrection</a>, <a href="#topic+NormExpBackgroundCorrection">NormExpBackgroundCorrection</a>, <a href="#topic+OpticalBackgroundCorrection">OpticalBackgroundCorrection</a>, <a href="#topic+RmaBackgroundCorrection">RmaBackgroundCorrection</a><br />
</p>
<p>public abstract static class <b>BackgroundCorrection</b><br />
extends <em><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></em><br />
</p>
<p>This class represents a background adjustment function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BackgroundCorrection(..., subsetToUpdate=NULL, typesToUpdate=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BackgroundCorrection_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code>.</p>
</td></tr>
<tr><td><code id="BackgroundCorrection_+3A_subsettoupdate">subsetToUpdate</code></td>
<td>
<p>The probes to be updated.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all probes are updated.</p>
</td></tr>
<tr><td><code id="BackgroundCorrection_+3A_typestoupdate">typesToUpdate</code></td>
<td>
<p>Types of probes to be updated.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ProbeLevelTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Ken Simpson, Henrik Bengtsson</p>

<hr>
<h2 id='BaseCountNormalization'>The BaseCountNormalization class</h2><span id='topic+BaseCountNormalization'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class BaseCountNormalization</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform3">ProbeLevelTransform3</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AbstractProbeSequenceNormalization">AbstractProbeSequenceNormalization</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>BaseCountNormalization</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>BaseCountNormalization</b><br />
extends <em><a href="#topic+AbstractProbeSequenceNormalization">AbstractProbeSequenceNormalization</a></em><br />
</p>
<p>This class represents a normalization method that corrects for systematic
effects in the probe intensities due to differences in the number of
A, C, G, and T:s in the probe sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BaseCountNormalization(..., model=c("robustSmoothSpline", "lm"), bootstrap=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BaseCountNormalization_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+AbstractProbeSequenceNormalization">AbstractProbeSequenceNormalization</a></code>.</p>
</td></tr>
<tr><td><code id="BaseCountNormalization_+3A_model">model</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying the model used to fit
the base-count effects.</p>
</td></tr>
<tr><td><code id="BaseCountNormalization_+3A_bootstrap">bootstrap</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the model fitting is done by bootstrap in
order to save memory.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from AbstractProbeSequenceNormalization</b>:<br />
fitOne, getAromaCellSequenceFile, getParameters, getTargetFile, indexOfMissingSequences, predictOne, process
</p>
<p><b>Methods inherited from ProbeLevelTransform3</b>:<br />
getAsteriskTags, getCellsTo, getCellsToFit, getCellsToUpdate, getParameters, getUnitsTo, getUnitsToFit, getUnitsToUpdate, writeSignals
</p>
<p><b>Methods inherited from ProbeLevelTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Requirements</h3>

<p>This class requires that an aroma probe sequence file is available
for the chip type.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='BasePositionNormalization'>The BasePositionNormalization class</h2><span id='topic+BasePositionNormalization'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class BasePositionNormalization</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform3">ProbeLevelTransform3</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AbstractProbeSequenceNormalization">AbstractProbeSequenceNormalization</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+LinearModelProbeSequenceNormalization">LinearModelProbeSequenceNormalization</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>BasePositionNormalization</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>BasePositionNormalization</b><br />
extends <em><a href="#topic+LinearModelProbeSequenceNormalization">LinearModelProbeSequenceNormalization</a></em><br />
</p>
<p>This class represents a normalization method that corrects for systematic
effects in the probe intensities due to differences in positioning of
A, C, G, and T:s in the probe sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BasePositionNormalization(..., model=c("smooth.spline"), df=5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BasePositionNormalization_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+LinearModelProbeSequenceNormalization">LinearModelProbeSequenceNormalization</a></code>.</p>
</td></tr>
<tr><td><code id="BasePositionNormalization_+3A_model">model</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying the model used to fit
the base-count effects.</p>
</td></tr>
<tr><td><code id="BasePositionNormalization_+3A_df">df</code></td>
<td>
<p>The degrees of freedom of the model.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from LinearModelProbeSequenceNormalization</b>:<br />
fitOne, getDesignMatrix, getNormalEquations, getSignalTransform, predictOne
</p>
<p><b>Methods inherited from AbstractProbeSequenceNormalization</b>:<br />
fitOne, getAromaCellSequenceFile, getParameters, getTargetFile, indexOfMissingSequences, predictOne, process
</p>
<p><b>Methods inherited from ProbeLevelTransform3</b>:<br />
getAsteriskTags, getCellsTo, getCellsToFit, getCellsToUpdate, getParameters, getUnitsTo, getUnitsToFit, getUnitsToUpdate, writeSignals
</p>
<p><b>Methods inherited from ProbeLevelTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson, Mark Robinson</p>

<hr>
<h2 id='bpmapCluster2Cdf'>Creates a CDF from tiling-array BPMAP file</h2><span id='topic+bpmapCluster2Cdf.default'></span><span id='topic+bpmapCluster2Cdf'></span>

<h3>Description</h3>

<p>Creates a CDF from tiling-array BPMAP file.<br />
</p>
<p><em>
NOTE: This method applies only to Affymetrix tiling arrays!
Furthermore, it is likely to be more useful for promoter tiling arrays
and less so for whole-genome tiling arrays.
</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
bpmapCluster2Cdf(pathname, chipType, tags=NULL, rows, cols, maxProbeDistance=3000L,
  minNbrOfProbes=30L, groupName=gsub("_.*", "", chipType), field="fullname",
  stringRemove=sprintf("%s:.*;", groupName), ..., flavor=c("v2", "v1"), path="*",
  verbose=-10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bpmapCluster2Cdf_+3A_pathname">pathname</code></td>
<td>
<p>The pathname to the BPMAP file.</p>
</td></tr>
<tr><td><code id="bpmapCluster2Cdf_+3A_chiptype">chipType</code>, <code id="bpmapCluster2Cdf_+3A_tags">tags</code></td>
<td>
<p>The chip type and optional tags of the CDF to
be written.</p>
</td></tr>
<tr><td><code id="bpmapCluster2Cdf_+3A_rows">rows</code>, <code id="bpmapCluster2Cdf_+3A_cols">cols</code></td>
<td>
<p>Two positive <code><a href="base.html#topic+integer">integer</a></code>s specifying the probe dimension
of the chip.  It is important to get this correct.  They can be
inferred from the CEL header of a CEL file for this chip,
cf. <code><a href="affxparser.html#topic+readCelHeader">readCelHeader</a></code>.</p>
</td></tr>
<tr><td><code id="bpmapCluster2Cdf_+3A_maxprobedistance">maxProbeDistance</code></td>
<td>
<p>A positive <code><a href="base.html#topic+integer">integer</a></code> specifying the maximum
genomic distance (in basepairs) allowed between two probes in order
to &quot;cluster&quot; those two probes into the same CDF units.  Whenever the
distance is greater, the two probes end up in two different CDF units.</p>
</td></tr>
<tr><td><code id="bpmapCluster2Cdf_+3A_minnbrofprobes">minNbrOfProbes</code></td>
<td>
<p>A positive <code><a href="base.html#topic+integer">integer</a></code> specifying the minimum number
of probes required in a CDF unit.  If fewer, those probes are dropped.</p>
</td></tr>
<tr><td><code id="bpmapCluster2Cdf_+3A_groupname">groupName</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying which BPMAP sequences
to keep.  Sequence with this group name is kept, all others are
excluded.</p>
</td></tr>
<tr><td><code id="bpmapCluster2Cdf_+3A_field">field</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string.</p>
</td></tr>
<tr><td><code id="bpmapCluster2Cdf_+3A_stringremove">stringRemove</code></td>
<td>
<p>An (optional) regular expression.</p>
</td></tr>
<tr><td><code id="bpmapCluster2Cdf_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to <code><a href="affxparser.html#topic+readBpmap">readBpmap</a></code>.</p>
</td></tr>
<tr><td><code id="bpmapCluster2Cdf_+3A_flavor">flavor</code></td>
<td>
<p>Specifying which version of BPMAP-to-CDF generator
to use. The default is always to use the most recent one, which
is also the recommended one.  Previous versions are kept only for
backward compatibility (and may be dropped at anytime).</p>
</td></tr>
<tr><td><code id="bpmapCluster2Cdf_+3A_path">path</code></td>
<td>
<p>The directory where the CDF file will be written.
If <code>"*"</code> (default), it will be written to
<code>annotationData/chipTypes/&lt;chipType&gt;/</code>.</p>
</td></tr>
<tr><td><code id="bpmapCluster2Cdf_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This method applies only to Affymetrix tiling arrays.  It is likely
to be useful for promoter tiling arrays and less so for whole-genome
tiling arrays.
Flavor <code>"v2"</code> replaced <code>"v1"</code> as aroma.affymetrix v2.5.4
(June 21, 2012). For details, see <code>news(package="aroma.affymetrix")</code>.
</p>


<h3>Value</h3>

<p>Returns (invisibly) a the pathname of the created CDF file.
The created CDF is written to the current directory.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson adopted from Mark Robinson standalone/online version
as of July 11, 2011.
</p>

<hr>
<h2 id='byChipType.DChipGenomeInformation'>Defines a DChipGenomeInformation object by chip type</h2><span id='topic+byChipType.DChipGenomeInformation'></span><span id='topic+DChipGenomeInformation.byChipType'></span><span id='topic+byChipType+2CDChipGenomeInformation-method'></span>

<h3>Description</h3>

<p>Defines a DChipGenomeInformation object by chip type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'DChipGenomeInformation'
byChipType(static, chipType, version=NULL, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="byChipType.DChipGenomeInformation_+3A_chiptype">chipType</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string.</p>
</td></tr>
<tr><td><code id="byChipType.DChipGenomeInformation_+3A_version">version</code></td>
<td>
<p>An optional <code><a href="base.html#topic+character">character</a></code> string specifying the version
string, if more than one version is available.</p>
</td></tr>
<tr><td><code id="byChipType.DChipGenomeInformation_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="byChipType.DChipGenomeInformation_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+DChipGenomeInformation">DChipGenomeInformation</a></code> object.
If no file was not found, an error is thrown.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+DChipGenomeInformation">DChipGenomeInformation</a></code>.
</p>

<hr>
<h2 id='byChipType.DChipSnpInformation'>Defines a DChipSnpInformation object by chip type</h2><span id='topic+byChipType.DChipSnpInformation'></span><span id='topic+DChipSnpInformation.byChipType'></span><span id='topic+byChipType+2CDChipSnpInformation-method'></span>

<h3>Description</h3>

<p>Defines a DChipSnpInformation object by chip type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'DChipSnpInformation'
byChipType(static, chipType, version=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="byChipType.DChipSnpInformation_+3A_chiptype">chipType</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string.</p>
</td></tr>
<tr><td><code id="byChipType.DChipSnpInformation_+3A_version">version</code></td>
<td>
<p>An optional <code><a href="base.html#topic+character">character</a></code> string specifying the version
string, if more than one version is available.</p>
</td></tr>
<tr><td><code id="byChipType.DChipSnpInformation_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+findByChipType.DChipSnpInformation">*findByChipType</a>()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+DChipSnpInformation">DChipSnpInformation</a></code> object.
If no file was not found, an error is thrown.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+DChipSnpInformation">DChipSnpInformation</a></code>.
</p>

<hr>
<h2 id='byChipType.SnpInformation'>Static method to define a genome information set by chip type</h2><span id='topic+byChipType.SnpInformation'></span><span id='topic+SnpInformation.byChipType'></span><span id='topic+byChipType+2CSnpInformation-method'></span>

<h3>Description</h3>

<p>Static method to define a genome information set by chip type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SnpInformation'
byChipType(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="byChipType.SnpInformation_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="#topic+SnpInformation">SnpInformation</a></code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fromCdf.SnpInformation">*fromCdf</a>()</code>.
For more information see <code><a href="#topic+SnpInformation">SnpInformation</a></code>.
</p>

<hr>
<h2 id='calculateBaseline.ChipEffectSet'>Estimates the baseline signal chromosome by chromosome</h2><span id='topic+calculateBaseline.ChipEffectSet'></span><span id='topic+ChipEffectSet.calculateBaseline'></span><span id='topic+calculateBaseline+2CChipEffectSet-method'></span>

<h3>Description</h3>

<p>Estimates the baseline signal chromosome by chromosome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ChipEffectSet'
calculateBaseline(this, chromosomes=NULL, ploidy=2, defaultPloidy=NA, all=FALSE,
  force=FALSE, verbose=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateBaseline.ChipEffectSet_+3A_chromosomes">chromosomes</code></td>
<td>
<p>An <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying for which chromsosomes
the baseline should be estimated.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all chromosomes are considered.</p>
</td></tr>
<tr><td><code id="calculateBaseline.ChipEffectSet_+3A_ploidy">ploidy</code></td>
<td>
<p>An <code><a href="base.html#topic+integer">integer</a></code> specifying the ploidy that the baseline
should have.</p>
</td></tr>
<tr><td><code id="calculateBaseline.ChipEffectSet_+3A_defaultploidy">defaultPloidy</code></td>
<td>
<p>An <code><a href="base.html#topic+integer">integer</a></code> specifying the default ploidy of
chromosomes that have not explicitly been allocated one.</p>
</td></tr>
<tr><td><code id="calculateBaseline.ChipEffectSet_+3A_all">all</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, signals are averaged also for cells that are not
on the genome.</p>
</td></tr>
<tr><td><code id="calculateBaseline.ChipEffectSet_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the CEL file that stores the is recreated.</p>
</td></tr>
<tr><td><code id="calculateBaseline.ChipEffectSet_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
<tr><td><code id="calculateBaseline.ChipEffectSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="aroma.core.html#topic+getAverageFile">getAverageFile</a></code>().
For more information see <code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code>.
</p>

<hr>
<h2 id='calculateParametersGsb.AffymetrixCelSet'>Computes parameters for adjustment of specific binding</h2><span id='topic+calculateParametersGsb.AffymetrixCelSet'></span><span id='topic+AffymetrixCelSet.calculateParametersGsb'></span><span id='topic+calculateParametersGsb+2CAffymetrixCelSet-method'></span>

<h3>Description</h3>

<p>Computes parameters for adjustment of specific binding.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelSet'
calculateParametersGsb(this, nbrOfPms=25000, affinities=NULL, seed=NULL, ...,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateParametersGsb.AffymetrixCelSet_+3A_nbrofpms">nbrOfPms</code></td>
<td>
<p>The number of random PMs to use in estimation.</p>
</td></tr>
<tr><td><code id="calculateParametersGsb.AffymetrixCelSet_+3A_affinities">affinities</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of probe affinities.</p>
</td></tr>
<tr><td><code id="calculateParametersGsb.AffymetrixCelSet_+3A_seed">seed</code></td>
<td>
<p>An (optional) <code><a href="base.html#topic+integer">integer</a></code> specifying a temporary random seed
to be used during processing.  The random seed is set to its original
state when done.  If <code><a href="base.html#topic+NULL">NULL</a></code>, it is not set.</p>
</td></tr>
<tr><td><code id="calculateParametersGsb.AffymetrixCelSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="calculateParametersGsb.AffymetrixCelSet_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This method is not constant in memory! /HB 2007-03-26
</p>


<h3>Author(s)</h3>

<p>Ken Simpson</p>

<hr>
<h2 id='calculateResidualSet.FirmaModel'>Gets the set of residuals corresponding to the PLM</h2><span id='topic+calculateResidualSet.FirmaModel'></span><span id='topic+FirmaModel.calculateResidualSet'></span><span id='topic+calculateResidualSet+2CFirmaModel-method'></span><span id='topic+FirmaModel.calculateResiduals'></span><span id='topic+calculateResiduals.FirmaModel'></span><span id='topic+calculateResiduals+2CFirmaModel-method'></span>

<h3>Description</h3>

<p>Gets the set of residuals corresponding to the PLM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FirmaModel'
calculateResidualSet(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateResidualSet.FirmaModel_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="#topic+ResidualSet">ResidualSet</a></code> object.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+FirmaModel">FirmaModel</a></code>.
</p>

<hr>
<h2 id='ChipEffectFile'>The ChipEffectFile class</h2><span id='topic+ChipEffectFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class ChipEffectFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ParameterCelFile">ParameterCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>ChipEffectFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+CnChipEffectFile">CnChipEffectFile</a></em>, <em><a href="#topic+ExonChipEffectFile">ExonChipEffectFile</a></em>, <em><a href="#topic+SnpChipEffectFile">SnpChipEffectFile</a></em><br />
</p>
<p>public abstract static class <b>ChipEffectFile</b><br />
extends <em><a href="#topic+ParameterCelFile">ParameterCelFile</a></em><br />
</p>
<p>This class represents estimates of chip effects in the probe-level models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ChipEffectFile(..., probeModel=c("pm"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ChipEffectFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+ParameterCelFile">ParameterCelFile</a></code>.</p>
</td></tr>
<tr><td><code id="ChipEffectFile_+3A_probemodel">probeModel</code></td>
<td>
<p>The specific type of model, e.g. <code>"pm"</code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractMatrix</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractTheta</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>findUnitsTodo</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAM</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ParameterCelFile</b>:<br />
extractDataFrame, extractMatrix, readUnits
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from AffymetrixCelFile</b>:<br />
allocateFromCdf, as.character, clone, createFrom, extractMatrix, fromFile, getAm, getCdf, getExtensionPattern, getFileFormat, getImage, getUnitNamesFile, getUnitTypesFile, highlight, image270, nbrOfCells, plotDensity, plotImage, plotMvsA, plotMvsX, range, setCdf, smoothScatterMvsA, writeImage
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>An object of this class is typically obtained through the
<code>getChipEffectSet()</code> method for the <code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code> class.
An object of this class is typically part of a <code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code>.
</p>

<hr>
<h2 id='ChipEffectFile+24fromDataFile'>Retrieve an existing CEL file, or create from CDF template if missing</h2><span id='topic+ChipEffectFile+24fromDataFile'></span><span id='topic+fromDataFile.ChipEffectFile'></span><span id='topic+ChipEffectFile.fromDataFile'></span><span id='topic+fromDataFile+2CChipEffectFile-method'></span>

<h3>Description</h3>

<p>Retrieve an existing CEL file, or create from CDF template if missing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## ChipEffectFile$fromDataFile(df=NULL, filename=sprintf("%s,chipEffects.CEL",
##   getFullName(df)), path, name=getName(df), cdf=NULL, ..., verbose=FALSE)

## Don't use the below:
## S3 method for class 'ChipEffectFile'
fromDataFile(static, df=NULL, filename=sprintf("%s,chipEffects.CEL", getFullName(df)),
  path, name=getName(df), cdf=NULL, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ChipEffectFile+2B24fromDataFile_+3A_static">static</code></td>
<td>
</td></tr>
<tr><td><code id="ChipEffectFile+2B24fromDataFile_+3A_df">df</code></td>
<td>
</td></tr>
<tr><td><code id="ChipEffectFile+2B24fromDataFile_+3A_filename">filename</code></td>
<td>
<p>The filename of the CEL file.</p>
</td></tr>
<tr><td><code id="ChipEffectFile+2B24fromDataFile_+3A_path">path</code></td>
<td>
<p>The path to the directory where to find/create the CEL file.</p>
</td></tr>
<tr><td><code id="ChipEffectFile+2B24fromDataFile_+3A_name">name</code></td>
<td>
<p>The name of the array to be stored in the CEL header.</p>
</td></tr>
<tr><td><code id="ChipEffectFile+2B24fromDataFile_+3A_cdf">cdf</code></td>
<td>
<p>The template <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> used for creating
a CEL file from scratch.</p>
</td></tr>
<tr><td><code id="ChipEffectFile+2B24fromDataFile_+3A_...">...</code></td>
<td>
<p>Passed to <code><a href="affxparser.html#topic+createCel">createCel</a></code>.</p>
</td></tr>
<tr><td><code id="ChipEffectFile+2B24fromDataFile_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="#topic+ChipEffectFile">ChipEffectFile</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code>allocateFromCdf()</code> of <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
For more information see <code><a href="#topic+ChipEffectFile">ChipEffectFile</a></code>.
</p>

<hr>
<h2 id='ChipEffectGroupMerge'>The ChipEffectGroupMerge class</h2><span id='topic+ChipEffectGroupMerge'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class ChipEffectGroupMerge</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ChipEffectTransform">ChipEffectTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code>ChipEffectGroupMerge</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+SnpChipEffectGroupMerge">SnpChipEffectGroupMerge</a><br />
</p>
<p>public static class <b>ChipEffectGroupMerge</b><br />
extends <em><a href="#topic+ChipEffectTransform">ChipEffectTransform</a></em><br />
</p>
<p>This class represents a method that merges chip effects across groups
unit by unit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ChipEffectGroupMerge(dataSet=NULL, fcn=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ChipEffectGroupMerge_+3A_dataset">dataSet</code></td>
<td>
<p>A <code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code>.</p>
</td></tr>
<tr><td><code id="ChipEffectGroupMerge_+3A_fcn">fcn</code></td>
<td>
<p>A <code><a href="base.html#topic+function">function</a></code> used to &quot;merge&quot; groups.</p>
</td></tr>
<tr><td><code id="ChipEffectGroupMerge_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the constructor of
<code><a href="#topic+ChipEffectTransform">ChipEffectTransform</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getMergeFunction</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setMergeFunction</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ChipEffectTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='ChipEffectSet'>The ChipEffectSet class</h2><span id='topic+ChipEffectSet'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class ChipEffectSet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataSet">AromaMicroarrayDataSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ParameterCelSet">ParameterCelSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>ChipEffectSet</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+CnChipEffectSet">CnChipEffectSet</a>, <a href="#topic+ExonChipEffectSet">ExonChipEffectSet</a>, <a href="#topic+SnpChipEffectSet">SnpChipEffectSet</a><br />
</p>
<p>public static class <b>ChipEffectSet</b><br />
extends <a href="#topic+ParameterCelSet">ParameterCelSet</a><br />
</p>
<p>This class represents estimates of chip effects in the probe-level models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ChipEffectSet(..., probeModel=c("pm"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ChipEffectSet_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.</p>
</td></tr>
<tr><td><code id="ChipEffectSet_+3A_probemodel">probeModel</code></td>
<td>
<p>The specific type of model, e.g. <code>"pm"</code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>boxplotStats</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractExpressionSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractMatrix</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractTheta</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>findUnitsTodo</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAM</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAverageFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCellIndices</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plotBoxplot</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ParameterCelSet</b>:<br />
extractDataFrame, extractMatrix
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from AffymetrixCelSet</b>:<br />
append, as, as.AffymetrixCelSet, as.character, averageQuantile, byName, byPath, clone, convertToUnique, doCRMAv1, doCRMAv2, doFIRMA, doGCRMA, doRMA, extractAffyBatch, extractFeatureSet, extractMatrix, extractSnpFeatureSet, findByName, getAverage, getAverageAsinh, getAverageFile, getAverageLog, getCdf, getChipType, getData, getIntensities, getPlatform, getTimestamps, getUnitGroupCellMap, getUnitIntensities, getUnitNamesFile, getUnitTypesFile, isDuplicated, justRMA, justSNPRMA, nbrOfArrays, normalizeQuantile, plotDensity, range, readUnits, setCdf, update2, writeSgr
</p>
<p><b>Methods inherited from AffymetrixFileSet</b>:<br />
as, as.AffymetrixFileSet, byPath, getDefaultFullName
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataSet</b>:<br />
as.AromaMicroarrayDataSetList, as.AromaMicroarrayDataSetTuple, getAromaFullNameTranslatorSet, getAverageFile, getChipType, getDefaultFullName, getPlatform, setAttributesBy, setAttributesBySampleAnnotationFile, setAttributesBySampleAnnotationSet, validate
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>An object of this class is typically obtained through the
<code>getChipEffectSet()</code> method for the <code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code> class.
</p>

<hr>
<h2 id='ChipEffectTransform'>The ChipEffectTransform class</h2><span id='topic+ChipEffectTransform'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class ChipEffectTransform</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><em><code>ChipEffectTransform</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AdditiveCovariatesNormalization">AdditiveCovariatesNormalization</a></em>, <a href="#topic+ChipEffectGroupMerge">ChipEffectGroupMerge</a>, <a href="#topic+FragmentEquivalentClassNormalization">FragmentEquivalentClassNormalization</a>, <a href="#topic+FragmentLengthNormalization">FragmentLengthNormalization</a>, <a href="#topic+GcContentNormalization">GcContentNormalization</a>, <a href="#topic+GcContentNormalization2">GcContentNormalization2</a>, <a href="#topic+SnpChipEffectGroupMerge">SnpChipEffectGroupMerge</a><br />
</p>
<p>public abstract static class <b>ChipEffectTransform</b><br />
extends <em><a href="#topic+Transform">Transform</a></em><br />
</p>
<p>This abstract class represents a transform that transforms chip-effect
estimates obtained from probe-level modeling.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ChipEffectTransform(dataSet=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ChipEffectTransform_+3A_dataset">dataSet</code></td>
<td>
<p>The input data set as an <code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code>.</p>
</td></tr>
<tr><td><code id="ChipEffectTransform_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of <code><a href="#topic+Transform">Transform</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Subclasses must implement the <code>process()</code> method.
</p>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='CnagCfhFile'>The CnagCfhFile class</h2><span id='topic+CnagCfhFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class CnagCfhFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>CnagCfhFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>CnagCfhFile</b><br />
extends <em><a href="#topic+AffymetrixFile">AffymetrixFile</a></em><br />
</p>
<p>A CnagCfhFile object represents a single CNAG CFH file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CnagCfhFile(..., cdf=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CnagCfhFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code>.</p>
</td></tr>
<tr><td><code id="CnagCfhFile_+3A_cdf">cdf</code></td>
<td>
<p>An optional <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code></p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfCells</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfSnps</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>An object of this class is typically part of an <code><a href="#topic+CnagCfhSet">CnagCfhSet</a></code>.
</p>

<hr>
<h2 id='CnagCfhFile+24fromFile'>Defines an CnagCfhFile object from a CFH file</h2><span id='topic+CnagCfhFile+24fromFile'></span><span id='topic+fromFile.CnagCfhFile'></span><span id='topic+CnagCfhFile.fromFile'></span><span id='topic+fromFile+2CCnagCfhFile-method'></span>

<h3>Description</h3>

<p>Defines an CnagCfhFile object from a CFH file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## CnagCfhFile$fromFile(filename, path=NULL, ..., verbose=FALSE, .checkArgs=TRUE)

## Don't use the below:
## S3 method for class 'CnagCfhFile'
fromFile(static, filename, path=NULL, ..., verbose=FALSE, .checkArgs=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CnagCfhFile+2B24fromFile_+3A_filename">filename</code></td>
<td>
<p>The filename of to the file.</p>
</td></tr>
<tr><td><code id="CnagCfhFile+2B24fromFile_+3A_path">path</code></td>
<td>
<p>The path to the file.</p>
</td></tr>
<tr><td><code id="CnagCfhFile+2B24fromFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+CnagCfhFile">CnagCfhFile</a></code> object.
If the file is not found or if it is of the wrong file format, an
error is thrown.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+CnagCfhFile">CnagCfhFile</a></code>.
</p>

<hr>
<h2 id='CnagCfhSet'>The CnagCfhSet class</h2><span id='topic+CnagCfhSet'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class CnagCfhSet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code>CnagCfhSet</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>CnagCfhSet</b><br />
extends <a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a><br />
</p>
<p>An CnagCfhSet object represents a set of CNAG CFH files
with <em>identical</em> chip types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CnagCfhSet(files=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CnagCfhSet_+3A_files">files</code></td>
<td>
<p>A <code><a href="base.html#topic+list">list</a></code> of <code><a href="#topic+CnagCfhFile">CnagCfhFile</a></code>:s.</p>
</td></tr>
<tr><td><code id="CnagCfhSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as.CnagCfhSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAverage</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAverageAsinh</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAverageFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAverageLog</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getData</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getTimestamps</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+CnagCfhFile">CnagCfhFile</a></code>.
</p>

<hr>
<h2 id='CnChipEffectFile'>The CnChipEffectFile class</h2><span id='topic+CnChipEffectFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class CnChipEffectFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ParameterCelFile">ParameterCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ChipEffectFile">ChipEffectFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+SnpChipEffectFile">SnpChipEffectFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CopyNumberDataFile">CopyNumberDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>CnChipEffectFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>CnChipEffectFile</b><br />
extends <em><a href="aroma.core.html#topic+CopyNumberDataFile">CopyNumberDataFile</a></em><br />
</p>
<p>This class represents estimates of chip effects in a copy-number probe-level
models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CnChipEffectFile(..., combineAlleles=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CnChipEffectFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+SnpChipEffectFile">SnpChipEffectFile</a></code>.</p>
</td></tr>
<tr><td><code id="CnChipEffectFile_+3A_combinealleles">combineAlleles</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> indicating if the signals from allele A
and allele B are combined or not.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>exportTotalAndFracB</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractTheta</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractTotalAndFreqB</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>hasAlleleBFractions</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>hasStrandiness</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>mergeStrands</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from CopyNumberDataFile</b>:<br />
as, as.CopyNumberDataFile, getNumberOfFilesAveraged, hasAlleleBFractions, hasStrandiness
</p>
<p><b>Methods inherited from SnpChipEffectFile</b>:<br />
exportTotalAndFracB, extractCNT, extractTheta, extractTotalAndFracB, getCellIndices, getExpandedCellMap, getParameters, mergeStrands, readUnits, writeCNT
</p>
<p><b>Methods inherited from ChipEffectFile</b>:<br />
as.character, extractChromosomalDataFrame, extractMatrix, extractTheta, findUnitsTodo, getAM, getAsFullCelFile, getCellIndices, getCellMapForMainCdf, getExpandedCellMap, getParameters, getUnitGroupCellArrayMap, getUnitGroupCellMatrixMap, getXAM, mergeGroups, readUnits, writeAsFullCelFile
</p>
<p><b>Methods inherited from ParameterCelFile</b>:<br />
extractDataFrame, extractMatrix, readUnits
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from AffymetrixCelFile</b>:<br />
allocateFromCdf, as.character, clone, createFrom, extractMatrix, fromFile, getAm, getCdf, getExtensionPattern, getFileFormat, getImage, getUnitNamesFile, getUnitTypesFile, highlight, image270, nbrOfCells, plotDensity, plotImage, plotMvsA, plotMvsX, range, setCdf, smoothScatterMvsA, writeImage
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>An object of this class is typically part of a <code><a href="#topic+CnChipEffectSet">CnChipEffectSet</a></code>.
</p>

<hr>
<h2 id='CnChipEffectSet'>The CnChipEffectSet class</h2><span id='topic+CnChipEffectSet'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class CnChipEffectSet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataSet">AromaMicroarrayDataSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ParameterCelSet">ParameterCelSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+SnpChipEffectSet">SnpChipEffectSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CopyNumberDataSet">CopyNumberDataSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>CnChipEffectSet</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>CnChipEffectSet</b><br />
extends <a href="aroma.core.html#topic+CopyNumberDataSet">CopyNumberDataSet</a><br />
</p>
<p>This class represents estimates of chip effects in the probe-level models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CnChipEffectSet(..., combineAlleles="byFirstFile")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CnChipEffectSet_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+SnpChipEffectSet">SnpChipEffectSet</a></code>.</p>
</td></tr>
<tr><td><code id="CnChipEffectSet_+3A_combinealleles">combineAlleles</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> indicating if the signals from
allele A and allele B are combined or not.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as.CopyNumberDataSetTuple</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>exportTotalAndFracB</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractTheta</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractTotalAndFreqB</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAverageFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCombineAlleles</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>hasAlleleBFractions</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>hasStrandiness</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setCombineAlleles</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>writeWig</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from CopyNumberDataSet</b>:<br />
as, as.CopyNumberDataSet, doCBS, hasAlleleBFractions, hasStrandiness
</p>
<p><b>Methods inherited from SnpChipEffectSet</b>:<br />
byPath, exportTotalAndFracB, extractAlleleSet, extractCNT, extractSnpCnvQSet, extractSnpQSet, extractTheta, extractTotalAndFreqB, getAverageFile, getBaseline, getMergeStrands, setMergeStrands, writeCNT
</p>
<p><b>Methods inherited from ChipEffectSet</b>:<br />
as.character, boxplotStats, byPath, calculateBaseline, calculateFieldBoxplotStats, calculateNuseBoxplotStats, calculateRleBoxplotStats, extractAffyBatch, extractChromosomalDataFrame, extractExpressionSet, extractMatrix, extractTheta, findByName, findUnitsTodo, fromDataSet, getAM, getAsFullCelSet, getAverageFile, getBaseline, getCellIndices, getXAM, plotBoxplot, readUnits, updateUnits
</p>
<p><b>Methods inherited from ParameterCelSet</b>:<br />
extractDataFrame, extractMatrix
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from AffymetrixCelSet</b>:<br />
append, as, as.AffymetrixCelSet, as.character, averageQuantile, byName, byPath, clone, convertToUnique, doCRMAv1, doCRMAv2, doFIRMA, doGCRMA, doRMA, extractAffyBatch, extractFeatureSet, extractMatrix, extractSnpFeatureSet, findByName, getAverage, getAverageAsinh, getAverageFile, getAverageLog, getCdf, getChipType, getData, getIntensities, getPlatform, getTimestamps, getUnitGroupCellMap, getUnitIntensities, getUnitNamesFile, getUnitTypesFile, isDuplicated, justRMA, justSNPRMA, nbrOfArrays, normalizeQuantile, plotDensity, range, readUnits, setCdf, update2, writeSgr
</p>
<p><b>Methods inherited from AffymetrixFileSet</b>:<br />
as, as.AffymetrixFileSet, byPath, getDefaultFullName
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataSet</b>:<br />
as.AromaMicroarrayDataSetList, as.AromaMicroarrayDataSetTuple, getAromaFullNameTranslatorSet, getAverageFile, getChipType, getDefaultFullName, getPlatform, setAttributesBy, setAttributesBySampleAnnotationFile, setAttributesBySampleAnnotationSet, validate
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='CnChipEffectSet+24importFromApt'>Imports copy-number estimates from an APT summary file</h2><span id='topic+CnChipEffectSet+24importFromApt'></span><span id='topic+importFromApt.CnChipEffectSet'></span><span id='topic+CnChipEffectSet.importFromApt'></span><span id='topic+importFromApt+2CCnChipEffectSet-method'></span>

<h3>Description</h3>

<p>Imports copy-number estimates from an APT summary file.
Currently only total copy-number estimates can be imported.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## CnChipEffectSet$importFromApt(filename, path=NULL, combineAlleles=TRUE, cdf, ...,
##   skip=TRUE, verbose=FALSE)

## Don't use the below:
## S3 method for class 'CnChipEffectSet'
importFromApt(static, filename, path=NULL, combineAlleles=TRUE, cdf, ..., skip=TRUE,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CnChipEffectSet+2B24importFromApt_+3A_filename">filename</code></td>
<td>
<p>The filename of the APT summary file.</p>
</td></tr>
<tr><td><code id="CnChipEffectSet+2B24importFromApt_+3A_path">path</code></td>
<td>
<p>An optional path to the file.</p>
</td></tr>
<tr><td><code id="CnChipEffectSet+2B24importFromApt_+3A_combinealleles">combineAlleles</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, ...</p>
</td></tr>
<tr><td><code id="CnChipEffectSet+2B24importFromApt_+3A_cdf">cdf</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> object.</p>
</td></tr>
<tr><td><code id="CnChipEffectSet+2B24importFromApt_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="CnChipEffectSet+2B24importFromApt_+3A_skip">skip</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, already imported arrays will be skipped.</p>
</td></tr>
<tr><td><code id="CnChipEffectSet+2B24importFromApt_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This import method is robust and memory efficient.  One array at the
time is imported by first writing to a temporary file which is then
renamed to the final name, if import was successful.  (If the import
failed, a temporary file will rename that has to be deleted manually).
</p>
<p>Since only one array at the time is imported, the memory overhead
will be bounded allowing to import very large tab-delimited data files
containing a large number of arrays.  Unfortunately, this approach
slows down the reading substantially, because in each import all but
one column is parsed but ignored.
</p>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+CnChipEffectSet">CnChipEffectSet</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+CnChipEffectSet">CnChipEffectSet</a></code>.
</p>

<hr>
<h2 id='CnChipEffectSet+24importFromDChip'>Imports copy-number estimates from a dChip result file</h2><span id='topic+CnChipEffectSet+24importFromDChip'></span><span id='topic+importFromDChip.CnChipEffectSet'></span><span id='topic+CnChipEffectSet.importFromDChip'></span><span id='topic+importFromDChip+2CCnChipEffectSet-method'></span>

<h3>Description</h3>

<p>Imports copy-number estimates from a dChip result file.
Currently only total copy-number estimates can be imported, that is
if dChip fitted the PLM with allele A and allele B combined.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## CnChipEffectSet$importFromDChip(filename, path=NULL, combineAlleles=TRUE, cdf, ...,
##   skip=TRUE, verbose=FALSE)

## Don't use the below:
## S3 method for class 'CnChipEffectSet'
importFromDChip(static, filename, path=NULL, combineAlleles=TRUE, cdf, ..., skip=TRUE,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CnChipEffectSet+2B24importFromDChip_+3A_filename">filename</code></td>
<td>
<p>The filename of the dChip result file.</p>
</td></tr>
<tr><td><code id="CnChipEffectSet+2B24importFromDChip_+3A_path">path</code></td>
<td>
<p>An optional path to the file.</p>
</td></tr>
<tr><td><code id="CnChipEffectSet+2B24importFromDChip_+3A_combinealleles">combineAlleles</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, ...</p>
</td></tr>
<tr><td><code id="CnChipEffectSet+2B24importFromDChip_+3A_cdf">cdf</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> object.</p>
</td></tr>
<tr><td><code id="CnChipEffectSet+2B24importFromDChip_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="CnChipEffectSet+2B24importFromDChip_+3A_skip">skip</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, already imported arrays will be skipped.</p>
</td></tr>
<tr><td><code id="CnChipEffectSet+2B24importFromDChip_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This import method is robust and memory efficient.  One array at the
time is imported by first writing to a temporary file which is then
renamed to the final name, if import was successful.  (If the import
failed, a temporary file will rename that has to be deleted manually).
</p>
<p>Since only one array at the time is imported, the memory overhead
will be bounded allowing to import very large tab-delimited data files
containing a large number of arrays.  Unfortunately, this approach
slows down the reading substantially, because in each import all but
one column is parsed but ignored.
</p>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+CnChipEffectSet">CnChipEffectSet</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+CnChipEffectSet">CnChipEffectSet</a></code>.
</p>

<hr>
<h2 id='CnPlm'>The CnPlm class</h2><span id='topic+CnPlm'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class CnPlm</b><br />
</p>
<p><code><a href="R.oo.html#topic+Interface">Interface</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+SnpPlm">SnpPlm</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code>CnPlm</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AffineCnPlm">AffineCnPlm</a></em>, <em><a href="#topic+AvgCnPlm">AvgCnPlm</a></em>, <em><a href="#topic+HetLogAddCnPlm">HetLogAddCnPlm</a></em>, <em><a href="#topic+MbeiCnPlm">MbeiCnPlm</a></em>, <em><a href="#topic+RmaCnPlm">RmaCnPlm</a></em><br />
</p>
<p>public class <b>CnPlm</b><br />
extends <a href="#topic+SnpPlm">SnpPlm</a><br />
</p>
<p>This support class represents a <code><a href="#topic+SnpPlm">SnpPlm</a></code> specially designed for
copy-number analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CnPlm(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CnPlm_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+SnpPlm">SnpPlm</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Models implementing this copy-number PLM, provides either
allele-specific or total copy-number estimates.
For allele-specific CNs the underlying <code><a href="#topic+SnpPlm">SnpPlm</a></code> model is fitted as
is, i.e. for each allele separately with or without the strands first
being merged.
</p>
<p>For total CNs the probe signals for the two alleles are combined
(=summed; not averaged) on the intensity scale before fitting
underlying <code><a href="#topic+SnpPlm">SnpPlm</a></code> model, again with or without the strands
first being merged.
</p>


<h3>Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCellIndices</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipEffectSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCombineAlleles</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getProbeAffinityFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setCombineAlleles</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from SnpPlm</b>:<br />
getCellIndices, getChipEffectSet, getMergeStrands, getParameters, getProbeAffinityFile, setMergeStrands
</p>
<p><b>Methods inherited from Interface</b>:<br />
extend, print, uses
</p>


<h3>Requirements</h3>

<p>Classes inheriting from this <code><a href="R.oo.html#topic+Interface">Interface</a></code> must provide the
following fields, in addition to the ones according to <code><a href="#topic+SnpPlm">SnpPlm</a></code>:
</p>

<dl>
<dt>combineAlleles</dt><dd><p>A <code><a href="base.html#topic+logical">logical</a></code> indicating if total or allele-specific
copy numbers should be estimated according to the above averaging.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='CnProbeAffinityFile'>The CnProbeAffinityFile class</h2><span id='topic+CnProbeAffinityFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class CnProbeAffinityFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ParameterCelFile">ParameterCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeAffinityFile">ProbeAffinityFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+SnpProbeAffinityFile">SnpProbeAffinityFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>CnProbeAffinityFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>CnProbeAffinityFile</b><br />
extends <em><a href="#topic+SnpProbeAffinityFile">SnpProbeAffinityFile</a></em><br />
</p>
<p>This class represents estimates of probe affinities in SNP probe-level models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CnProbeAffinityFile(..., combineAlleles=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CnProbeAffinityFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+SnpProbeAffinityFile">SnpProbeAffinityFile</a></code>.</p>
</td></tr>
<tr><td><code id="CnProbeAffinityFile_+3A_combinealleles">combineAlleles</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">FALSE</a></code>, allele A and allele B are treated
separately, otherwise together.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from SnpProbeAffinityFile</b>:<br />
getCellIndices, setMergeStrands
</p>
<p><b>Methods inherited from ProbeAffinityFile</b>:<br />
as.character, getCellIndices, getParameters, readUnits
</p>
<p><b>Methods inherited from ParameterCelFile</b>:<br />
extractDataFrame, extractMatrix, readUnits
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from AffymetrixCelFile</b>:<br />
allocateFromCdf, as.character, clone, createFrom, extractMatrix, fromFile, getAm, getCdf, getExtensionPattern, getFileFormat, getImage, getUnitNamesFile, getUnitTypesFile, highlight, image270, nbrOfCells, plotDensity, plotImage, plotMvsA, plotMvsX, range, setCdf, smoothScatterMvsA, writeImage
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='compare.AffymetrixCdfFile'>Checks if two AffymetrixCdfFile objects are equal</h2><span id='topic+compare.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.compare'></span><span id='topic+compare+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Checks if two AffymetrixCdfFile objects are equal.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
compare(this, other, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compare.AffymetrixCdfFile_+3A_other">other</code></td>
<td>
<p>The other <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> object to be compared to.</p>
</td></tr>
<tr><td><code id="compare.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="affxparser.html#topic+compareCdfs">compareCdfs</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code><a href="base.html#topic+logical">TRUE</a></code> if the two objects are equal, otherwise <code><a href="base.html#topic+logical">FALSE</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='computeAffinities.AffymetrixCdfFile'>Calculates probe affinities from sequence</h2><span id='topic+computeAffinities.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.computeAffinities'></span><span id='topic+computeAffinities+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Calculates probe affinities from sequence.
</p>
<p>Adapted from <code><a href="gcrma.html#topic+compute.affinities">compute.affinities</a></code> in the <span class="pkg">gcrma</span> package.
Attempts to find the tab-separated probe sequence file associated with
a particular CDF, and matches sequence to probe index in order to assign
an affinity to each probe.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
computeAffinities(this, safe=TRUE, force=FALSE, verbose=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="computeAffinities.AffymetrixCdfFile_+3A_safe">safe</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> argument passed to <code>getProbeSequenceData()</code>.</p>
</td></tr>
<tr><td><code id="computeAffinities.AffymetrixCdfFile_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">FALSE</a></code>, cached results is returned, if available.</p>
</td></tr>
<tr><td><code id="computeAffinities.AffymetrixCdfFile_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
<tr><td><code id="computeAffinities.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of (log2) probe affinities, of length equal
to the total number of features on the array.
</p>


<h3>Author(s)</h3>

<p>Ken Simpson, Henrik Bengtsson, Mark Robinson</p>

<hr>
<h2 id='convert.AffymetrixCdfFile'>Converts a CDF into the same CDF but with another format</h2><span id='topic+convert.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.convert'></span><span id='topic+convert+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Converts a CDF into the same CDF but with another format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
convert(this, chipType=getChipType(this), suffix=NULL, sep="-", path="cdf", ...,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert.AffymetrixCdfFile_+3A_chiptype">chipType</code></td>
<td>
<p>The chip type of the new CDF.</p>
</td></tr>
<tr><td><code id="convert.AffymetrixCdfFile_+3A_suffix">suffix</code></td>
<td>
<p>A suffix added to the chip type of the new CDF.</p>
</td></tr>
<tr><td><code id="convert.AffymetrixCdfFile_+3A_sep">sep</code></td>
<td>
<p>A string separating the chip type and the suffix string.</p>
</td></tr>
<tr><td><code id="convert.AffymetrixCdfFile_+3A_path">path</code></td>
<td>
<p>The path where to store the new CDF file.</p>
</td></tr>
<tr><td><code id="convert.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="affxparser.html#topic+convertCdf">convertCdf</a></code>.</p>
</td></tr>
<tr><td><code id="convert.AffymetrixCdfFile_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the new CDF as an <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> object.
</p>


<h3>See Also</h3>

<p>To compare two CDFs, use <code>equals()</code>.
Internally <code><a href="affxparser.html#topic+convertCdf">convertCdf</a></code> is used.
For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='convertUnits.AffymetrixCdfFile'>Gets and validates unit indices</h2><span id='topic+convertUnits.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.convertUnits'></span><span id='topic+convertUnits+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Gets and validates unit indices either by unit names or by a unit indices (validation).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
convertUnits(this, units=NULL, keepNULL=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convertUnits.AffymetrixCdfFile_+3A_units">units</code></td>
<td>
<p>Either a <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> with unit names, or an <code><a href="base.html#topic+integer">integer</a></code>
<code><a href="base.html#topic+vector">vector</a></code> with unit indices to be validated.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all unit indices are returned.</p>
</td></tr>
<tr><td><code id="convertUnits.AffymetrixCdfFile_+3A_keepnull">keepNULL</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, <code><a href="base.html#topic+NULL">NULL</a></code> returns <code><a href="base.html#topic+NULL">NULL</a></code>.</p>
</td></tr>
<tr><td><code id="convertUnits.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> with unit indices.
If some units are non existing, an error is thrown.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='createExonByTranscriptCdf.AffymetrixCdfFile'>Creates an exon-by-transcript CDF</h2><span id='topic+createExonByTranscriptCdf.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.createExonByTranscriptCdf'></span><span id='topic+createExonByTranscriptCdf+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Creates an exon-by-transcript CDF based on the probesets defined in an &quot;exon-only&quot; CDF
and transcript-exon mapping of a NetAffx probeset annotation data file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
createExonByTranscriptCdf(cdf, csv, tags=c("*"), path=getPath(cdf),
  type=c("all", "core", "extended", "full", "main", "control", "cds"), subsetBy=NULL,
  within=NULL, ..., overwrite=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="createExonByTranscriptCdf.AffymetrixCdfFile_+3A_cdf">cdf</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> specifying
an &quot;exon-only&quot; CDF, which defines the exon-specific probesets
that will go into the new CDF. For more details, see below.</p>
</td></tr>
<tr><td><code id="createExonByTranscriptCdf.AffymetrixCdfFile_+3A_csv">csv</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixNetAffxCsvFile">AffymetrixNetAffxCsvFile</a></code>
specifying the Affymetrix NetAffx CSV probeset annotation file
that contains the transcript-exon mapping.</p>
</td></tr>
<tr><td><code id="createExonByTranscriptCdf.AffymetrixCdfFile_+3A_tags">tags</code></td>
<td>
<p>Additional tags added to the filename of created CDF,
i.e. &lt;chiptype&gt;,&lt;tags&gt;.cdf.</p>
</td></tr>
<tr><td><code id="createExonByTranscriptCdf.AffymetrixCdfFile_+3A_path">path</code></td>
<td>
<p>The output path where the custom CDF will be written.</p>
</td></tr>
<tr><td><code id="createExonByTranscriptCdf.AffymetrixCdfFile_+3A_type">type</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying the type of CDF to be written.</p>
</td></tr>
<tr><td><code id="createExonByTranscriptCdf.AffymetrixCdfFile_+3A_subsetby">subsetBy</code></td>
<td>
<p>An optional <code><a href="base.html#topic+character">character</a></code> specifying the name of a column
in the annotation file to subset against.  The column will be parsed
as the data type of argument <code>within</code>.</p>
</td></tr>
<tr><td><code id="createExonByTranscriptCdf.AffymetrixCdfFile_+3A_within">within</code></td>
<td>
<p>A <code><a href="base.html#topic+vector">vector</a></code> of values accepted for the <code>subsetBy</code> column.</p>
</td></tr>
<tr><td><code id="createExonByTranscriptCdf.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>readDataFrame()</code> of
<code><a href="#topic+AffymetrixNetAffxCsvFile">AffymetrixNetAffxCsvFile</a></code>, e.g. <code>nrow</code>.</p>
</td></tr>
<tr><td><code id="createExonByTranscriptCdf.AffymetrixCdfFile_+3A_overwrite">overwrite</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, an existing CDF is overwritten.</p>
</td></tr>
<tr><td><code id="createExonByTranscriptCdf.AffymetrixCdfFile_+3A_verbose">verbose</code></td>
<td>
<p>...</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>


<h3>Requirements for the &quot;exon-only&quot; CDF</h3>

<p>The template CDF - argument <code>cdf</code> - should be an &quot;exon-only&quot; CDF:
each unit has one group/probeset, which is the exon.
An example of this is the &quot;unsupported&quot; HuEx-1_0-st-v2.cdf
from Affymetrix, which has 1,432,154 units.
Such &quot;exon-only&quot; CDFs do not contain information about clustering
exons/probesets into gene transcripts.
The CDF may also contain a number of non-exon probesets corresponding
to control probes, which can contain <em>very</em> large numbers of
probes per probeset. Such units are dropped/ignored by this method.
</p>


<h3>Ordering of transcripts and exons within transcripts</h3>

<p>The transcripts (=units) will be ordered as they appear in the
NetAffx annotation file.
Within each transcript (=unit), the exons (=groups) are ordered
lexicographically by their names.

</p>


<h3>Naming of transcripts and exons</h3>

<p>In the created CDF, each unit corresponds to one transcript cluster,
and each group within a unit corresponds to the exons within
that transcript cluster.  Thus, the unit names correspond to the
transcript cluster names and the group names correspond to the
exon names.
</p>
<p>The exon names are defined by unit names of the exon-only CDF,
whereas the transcript names are defined by the
<code>transcriptClusterId</code> column in the NetAffx annotation data file.
These transcript and exon names are often &quot;non-sense&quot; integers.
In order to map these to more genome-friendly names, use the various
annotations available in the NetAffx annotation data file.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson adopted from <code>createTranscriptCDF()</code> written
by Ken Simpson, Elizabeth Purdom and Mark Robinson.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# The exon-only CDF
cdf &lt;- AffymetrixCdfFile$byChipType("HuEx-1_0-st-v2")

# The NetAffx probeset annotation data file
csv &lt;- AffymetrixNetAffxCsvFile("HuEx-1_0-st-v2.na24.hg18.probeset.csv", path=getPath(cdf))

# Create a CDF containing all core probesets:
cdfT &lt;- createExonByTranscriptCdf(cdf, csv=csv, tags=c("*,HB20110911"))
print(cdfT)

# Create CDF containing the core probesets with 3 or 4 probes:
cdfT2 &lt;- createExonByTranscriptCdf(cdf, csv=csv,
            tags=c("*,bySize=3-4,HB20110911"),
            subsetBy="probeCount", within=c("3", "4"))
print(cdfT2)

## End(Not run)</code></pre>

<hr>
<h2 id='createFrom.AffymetrixCelFile'>Creates a CEL file using another as a template</h2><span id='topic+createFrom.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.createFrom'></span><span id='topic+createFrom+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Creates a CEL file using another as a template.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelFile'
createFrom(this, filename, path=NULL, overwrite=FALSE, skip=!overwrite,
  version=c("4", "3"), methods=c("copy", "create"), clear=FALSE, defValue=0, ...,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="createFrom.AffymetrixCelFile_+3A_filename">filename</code>, <code id="createFrom.AffymetrixCelFile_+3A_path">path</code></td>
<td>
<p>The filename and path of to the CEL
file to be created.</p>
</td></tr>
<tr><td><code id="createFrom.AffymetrixCelFile_+3A_version">version</code></td>
<td>
<p>The file-format version of the CEL file to be created.</p>
</td></tr>
<tr><td><code id="createFrom.AffymetrixCelFile_+3A_methods">methods</code></td>
<td>
<p>If <code>"copy"</code>, the new file is created as a copy of the
template file.  If <code>"create"</code>, the new file is created from
scratch from the template file.</p>
</td></tr>
<tr><td><code id="createFrom.AffymetrixCelFile_+3A_clear">clear</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the fields of the CEL file are cleared (zeroed),
otherwise they contain the same information as the source file.</p>
</td></tr>
<tr><td><code id="createFrom.AffymetrixCelFile_+3A_defvalue">defValue</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> value that cleared/allocated elements have.</p>
</td></tr>
<tr><td><code id="createFrom.AffymetrixCelFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="createFrom.AffymetrixCelFile_+3A_verbose">verbose</code></td>
<td>
<p>See &quot;R.utils::Verbose&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code> reference to the new CEL file.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>

<hr>
<h2 id='createMonocellCdf.AffymetrixCdfFile'>Creates a mono-cell version of the CDF</h2><span id='topic+createMonocellCdf.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.createMonocellCdf'></span><span id='topic+createMonocellCdf+2CAffymetrixCdfFile-method'></span><span id='topic+AffymetrixCdfFile.createMonoCell'></span><span id='topic+createMonoCell.AffymetrixCdfFile'></span><span id='topic+createMonoCell+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Creates a mono-cell version of the CDF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
createMonocellCdf(this, chipType=getChipType(this), tags="monocell", path=NULL,
  nbrOfCellsPerField=1, ..., ram=NULL, overwrite=FALSE, verbose=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="createMonocellCdf.AffymetrixCdfFile_+3A_chiptype">chipType</code></td>
<td>
<p>The chip type of the new CDF.</p>
</td></tr>
<tr><td><code id="createMonocellCdf.AffymetrixCdfFile_+3A_tags">tags</code></td>
<td>
<p>Tags added to the chip type of the new CDF.</p>
</td></tr>
<tr><td><code id="createMonocellCdf.AffymetrixCdfFile_+3A_path">path</code></td>
<td>
<p>The path where to store the new CDF file.</p>
</td></tr>
<tr><td><code id="createMonocellCdf.AffymetrixCdfFile_+3A_nbrofcellsperfield">nbrOfCellsPerField</code></td>
<td>
<p>Number of cells per group field the new CDF
should have.</p>
</td></tr>
<tr><td><code id="createMonocellCdf.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="affxparser.html#topic+writeCdf">writeCdf</a></code>.</p>
</td></tr>
<tr><td><code id="createMonocellCdf.AffymetrixCdfFile_+3A_ram">ram</code></td>
<td>
<p>A <code><a href="base.html#topic+double">double</a></code> saying if more or less units should be converted
per chunk.  A smaller value uses less memory.</p>
</td></tr>
<tr><td><code id="createMonocellCdf.AffymetrixCdfFile_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the monocell CDF as an <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson, Ken Simpson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='createUniqueCdf.AffymetrixCdfFile'>Creates a unique-cell version of the CDF</h2><span id='topic+createUniqueCdf.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.createUniqueCdf'></span><span id='topic+createUniqueCdf+2CAffymetrixCdfFile-method'></span><span id='topic+AffymetrixCdfFile.createUnique'></span><span id='topic+createUnique.AffymetrixCdfFile'></span><span id='topic+createUnique+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>In some cases, single probes map to multiple genome locations.  In cases
where you may later want to store a probe estimate (e.g. a probe effect
or a residual), you will not be able to store more than one per cell.
The unique CDF facilitates this by making the cell indices unique
(essentially copying the multimapping probes).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
createUniqueCdf(this, chipType=getChipType(this), tags="unique", path=NULL, units=NULL,
  ..., ram=NULL, verbose=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="createUniqueCdf.AffymetrixCdfFile_+3A_chiptype">chipType</code></td>
<td>
<p>The chip type of the new CDF.</p>
</td></tr>
<tr><td><code id="createUniqueCdf.AffymetrixCdfFile_+3A_tags">tags</code></td>
<td>
<p>Tags added to the chip type of the new CDF.</p>
</td></tr>
<tr><td><code id="createUniqueCdf.AffymetrixCdfFile_+3A_path">path</code></td>
<td>
<p>The path where to store the new CDF file.</p>
</td></tr>
<tr><td><code id="createUniqueCdf.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="affxparser.html#topic+writeCdf">writeCdf</a></code>.</p>
</td></tr>
<tr><td><code id="createUniqueCdf.AffymetrixCdfFile_+3A_ram">ram</code></td>
<td>
<p>A <code><a href="base.html#topic+double">double</a></code> saying if more or less units should be converted
per chunk.  A smaller value uses less memory.</p>
</td></tr>
<tr><td><code id="createUniqueCdf.AffymetrixCdfFile_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that the set of units in the &quot;unique&quot; CDF is identical to that
of the input CDF.  So are the unit groups in each unit.
Also the number of cells per unit group is preserved.
It is only the cell-index composition of each unit group that changes.
The cell indices in the unique CDF are guaranteed to occur only
once, whereas this may not be true for the input CDF.
</p>


<h3>Value</h3>

<p>Returns the unique CDF as an <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Mark Robinson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='CrlmmParametersFile'>The CrlmmParametersFile class</h2><span id='topic+CrlmmParametersFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class CrlmmParametersFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+ColumnNamesInterface">ColumnNamesInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFile">GenericTabularFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTabularBinaryFile">AromaTabularBinaryFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>CrlmmParametersFile</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>CrlmmParametersFile</b><br />
extends <a href="aroma.core.html#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a><br />
</p>
<p>An CrlmmParametersFile is a <code><a href="aroma.core.html#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CrlmmParametersFile(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CrlmmParametersFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="aroma.core.html#topic+AromaUnitSignalBinaryFile">AromaUnitSignalBinaryFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>allocate</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>findUnitsTodo</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readParameter</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>updateParameter</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AromaUnitSignalBinaryFile</b>:<br />
allocate, allocateFromUnitAnnotationDataFile, allocateFromUnitNamesFile, as.character, extractMatrix, extractRawGenomicSignals, fromFile, getChipType, getExtensionPattern, getFilenameExtension, getNumberOfFilesAveraged, getPlatform, isAverageFile, nbrOfUnits, readDataFrame, writeDataFrame
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaTabularBinaryFile</b>:<br />
[, [&lt;-, [[, allocate, as.character, colApply, colMeans, colStats, colSums, dimnames&lt;-, getBytesPerColumn, getColClasses, getDefaultColumnNames, getRootName, importFrom, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, readFooter, readHeader, readRawFooter, setAttributesByTags, subset, summary, updateData, updateDataColumn, writeFooter, writeRawFooter
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericTabularFile</b>:<br />
[, as.character, dim, extractMatrix, head, nbrOfColumns, nbrOfRows, readColumns, readDataFrame, tail, writeColumnsToFiles
</p>
<p><b>Methods inherited from ColumnNamesInterface</b>:<br />
appendColumnNamesTranslator, appendColumnNamesTranslatorByNULL, appendColumnNamesTranslatorBycharacter, appendColumnNamesTranslatorByfunction, appendColumnNamesTranslatorBylist, clearColumnNamesTranslator, clearListOfColumnNamesTranslators, getColumnNames, getColumnNamesTranslator, getDefaultColumnNames, getListOfColumnNamesTranslators, nbrOfColumns, setColumnNames, setColumnNamesTranslator, setListOfColumnNamesTranslators, updateColumnNames
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='CrlmmParametersSet'>The CrlmmParametersSet class</h2><span id='topic+CrlmmParametersSet'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class CrlmmParametersSet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericTabularFileSet">GenericTabularFileSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTabularBinarySet">AromaTabularBinarySet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaUnitSignalBinarySet">AromaUnitSignalBinarySet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>CrlmmParametersSet</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>CrlmmParametersSet</b><br />
extends <a href="aroma.core.html#topic+AromaUnitSignalBinarySet">AromaUnitSignalBinarySet</a><br />
</p>
<p>An CrlmmParametersSet object represents a set of
<code><a href="#topic+CrlmmParametersFile">CrlmmParametersFile</a></code>s with <em>identical</em> chip types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CrlmmParametersSet(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CrlmmParametersSet_+3A_...">...</code></td>
<td>
<p>Arguments passed to
<code><a href="aroma.core.html#topic+AromaUnitSignalBinarySet">AromaUnitSignalBinarySet</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byPath</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>findUnitsTodo</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AromaUnitSignalBinarySet</b>:<br />
byName, findByName, getAromaFullNameTranslatorSet, getAromaUgpFile, getChipType, getPlatform, validate, writeDataFrame
</p>
<p><b>Methods inherited from AromaTabularBinarySet</b>:<br />
getDefaultFullName, getRootName, setAttributesBy, setAttributesBySampleAnnotationFile, setAttributesBySampleAnnotationSet, setAttributesByTags
</p>
<p><b>Methods inherited from GenericTabularFileSet</b>:<br />
extractMatrix, calculateAverageColumnAcrossFiles
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='DChipCdfBinFile'>The DChipCdfBinFile class</h2><span id='topic+DChipCdfBinFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class DChipCdfBinFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+UnitAnnotationDataFile">UnitAnnotationDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+UnitNamesFile">UnitNamesFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>DChipCdfBinFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>DChipCdfBinFile</b><br />
extends <em><a href="aroma.core.html#topic+UnitNamesFile">UnitNamesFile</a></em><br />
</p>
<p>A DChipCdfBinFile object represents a DChip CDF.bin file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DChipCdfBinFile(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DChipCdfBinFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFileFormat</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getHeader</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPlatform</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitNames</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitSizes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfCells</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readDataFrame</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from UnitNamesFile</b>:<br />
getUnitNames, indexOf, nbrOfUnits
</p>
<p><b>Methods inherited from UnitAnnotationDataFile</b>:<br />
byChipType, getAromaUflFile, getAromaUgpFile, getChipType, getDefaultExtension, getPlatform, nbrOfUnits
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='DChipDcpFile'>The DChipDcpFile class</h2><span id='topic+DChipDcpFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class DChipDcpFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>DChipDcpFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>DChipDcpFile</b><br />
extends <em><a href="#topic+AffymetrixFile">AffymetrixFile</a></em><br />
</p>
<p>A DChipDcpFile object represents a DChip DCP file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DChipDcpFile(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DChipDcpFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>dim</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractTheta</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCalls</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getExcludes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFileFormat</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getHeader</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getNormalizedIntensities</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getRawIntensities</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getThetaStds</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getThetas</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getThetasAB</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>hasMbeiData</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>hasNormalizedData</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfCells</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+DChipDcpSet">DChipDcpSet</a></code>.
</p>

<hr>
<h2 id='DChipDcpSet'>The DChipDcpSet class</h2><span id='topic+DChipDcpSet'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class DChipDcpSet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataSet">AromaMicroarrayDataSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>DChipDcpSet</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>DChipDcpSet</b><br />
extends <em><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></em><br />
</p>
<p>A DChipDcpSet object represents a set of DChip DCP files
for <em>identical</em> chip types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DChipDcpSet(files=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DChipDcpSet_+3A_files">files</code></td>
<td>
<p>A <code><a href="base.html#topic+list">list</a></code> of <code><a href="#topic+DChipDcpFile">DChipDcpFile</a></code>:s.</p>
</td></tr>
<tr><td><code id="DChipDcpSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>as.DChipDcpSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>exportTotalAndFracB</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractTheta</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCdfBin</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AffymetrixFileSet</b>:<br />
as, as.AffymetrixFileSet, byPath, getDefaultFullName
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataSet</b>:<br />
as.AromaMicroarrayDataSetList, as.AromaMicroarrayDataSetTuple, getAromaFullNameTranslatorSet, getAverageFile, getChipType, getDefaultFullName, getPlatform, setAttributesBy, setAttributesBySampleAnnotationFile, setAttributesBySampleAnnotationSet, validate
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+DChipDcpFile">DChipDcpFile</a></code>.
</p>

<hr>
<h2 id='DChipGenomeInformation'>The DChipGenomeInformation class</h2><span id='topic+DChipGenomeInformation'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class DChipGenomeInformation</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+GenomeInformation">GenomeInformation</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>DChipGenomeInformation</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>DChipGenomeInformation</b><br />
extends <em><a href="#topic+GenomeInformation">GenomeInformation</a></em><br />
</p>
<p>This class represents dChip genome information files, which typically
contains information about chromosomal locations of the units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DChipGenomeInformation(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DChipGenomeInformation_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+GenomeInformation">GenomeInformation</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The dChip genome information files for various chip types can be
downloaded from <a href="https://sites.google.com/site/dchipsoft/">https://sites.google.com/site/dchipsoft/</a>.
Put each file in a
directory named identically as the corresponding chip type under the
<em>annotations/</em> directory, e.g.
<em>annotations/Mapping50K_Hind240/50k hind genome info AfAm
june 05 hg17.xls</em>.
Note that dChip changes the filename and file format slightly between
chip types, but currently the <code><a href="#topic+byChipType.DChipGenomeInformation">*byChipType</a>()</code> basically searches
for files with names consisting of <code>"genome info"</code> or
<code>"genome_info"</code>.  At least for the most common chip types, there
is no need to rename the files in order for this class to recognize them.
</p>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readDataFrame</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from GenomeInformation</b>:<br />
as.character, byChipType, fromCdf, fromDataSet, getChipType, getChromosomeStats, getChromosomes, getData, getPositions, getUnitIndices, getUnitsOnChromosome, getUnitsOnChromosomes, isCompatibleWithCdf, nbrOfUnits, plotDensity, readDataFrame, verify
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='DChipQuantileNormalization'>The DChipQuantileNormalization class</h2><span id='topic+DChipQuantileNormalization'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class DChipQuantileNormalization</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+QuantileNormalization">QuantileNormalization</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>DChipQuantileNormalization</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>DChipQuantileNormalization</b><br />
extends <a href="#topic+QuantileNormalization">QuantileNormalization</a><br />
</p>
<p>This class represents a special <code><a href="#topic+QuantileNormalization">QuantileNormalization</a></code>
using smooth-splines.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DChipQuantileNormalization(..., robust=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DChipQuantileNormalization_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+QuantileNormalization">QuantileNormalization</a></code>.</p>
</td></tr>
<tr><td><code id="DChipQuantileNormalization_+3A_robust">robust</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the normalization function is estimated
robustly, otherwise not.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This normalization method implements the two-pass algorithm described
in Bengtsson et al. (2008).
</p>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from QuantileNormalization</b>:<br />
findTargetDistributionFile, getParameters, process
</p>
<p><b>Methods inherited from ProbeLevelTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] H. Bengtsson, R. Irizarry, B. Carvalho, &amp; T.P. Speed.
Estimation and assessment of raw copy numbers at the single
locus level, Bioinformatics, 2008.
</p>

<hr>
<h2 id='DChipSnpInformation'>The DChipSnpInformation class</h2><span id='topic+DChipSnpInformation'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class DChipSnpInformation</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+SnpInformation">SnpInformation</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>DChipSnpInformation</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>DChipSnpInformation</b><br />
extends <em><a href="#topic+SnpInformation">SnpInformation</a></em><br />
</p>
<p>This class represents dChip genome information files, which typically
contains information on nucleotide sequences and fragment lengths
of the units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DChipSnpInformation(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DChipSnpInformation_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+SnpInformation">SnpInformation</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The dChip SNP information files for various chip types can be
downloaded from <a href="https://sites.google.com/site/dchipsoft/">https://sites.google.com/site/dchipsoft/</a>.
Put each file in a
directory named identically as the corresponding chip type under the
<em>annotations/</em> directory, e.g.
<em>annotations/Mapping50K_Hind240/50k hind snp info AfAm
june 05 hg17.xls</em>.
Note that dChip changes the filename and file format slightly between
chip types, but currently the <code><a href="#topic+byChipType.DChipSnpInformation">*byChipType</a>()</code> basically searches
for files with names consisting of <code>"snp info"</code> or
<code>"snp_info"</code>.  At least for the most common chip types, there
is no need to rename the files in order for this class to recognize them.
</p>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readDataFrame</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from SnpInformation</b>:<br />
as.character, byChipType, fromCdf, fromDataSet, getChipType, getData, getFields, getFragmentLengths, getFragmentStarts, getFragmentStops, isCompatibleWithCdf, nbrOfEnzymes, nbrOfUnits, readDataFrame, verify
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='doCRMAv1'>Estimation and assessment of raw copy numbers at the single locus level (CRMA v1)</h2><span id='topic+doCRMAv1.default'></span><span id='topic+doCRMAv1'></span><span id='topic+doCRMAv1.AffymetrixCelSet'></span><span id='topic+doASCRMAv1'></span><span id='topic+doASCRMAv1.default'></span>

<h3>Description</h3>

<p>Estimation and assessment of raw copy numbers at the single locus level (CRMA v1) based on [1].
The algorithm is processed in bounded memory, meaning virtually
any number of arrays can be analyzed on also very limited computer
systems.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  ## S3 method for class 'AffymetrixCelSet'
doCRMAv1(csR, shift=+300, combineAlleles=TRUE, lengthRange=NULL, arrays=NULL, drop=TRUE,
  verbose=FALSE, ...)
  ## Default S3 method:
doCRMAv1(dataSet, ..., verbose=FALSE)
  ## Default S3 method:
doASCRMAv1(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="doCRMAv1_+3A_csr">csR</code>, <code id="doCRMAv1_+3A_dataset">dataSet</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code> (or the name of an <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>).</p>
</td></tr>
<tr><td><code id="doCRMAv1_+3A_shift">shift</code></td>
<td>
<p>An tuning parameter specifying how much to shift the
probe signals before probe summarization.</p>
</td></tr>
<tr><td><code id="doCRMAv1_+3A_combinealleles">combineAlleles</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> specifying whether allele probe pairs
should be summed before modeling or not.</p>
</td></tr>
<tr><td><code id="doCRMAv1_+3A_lengthrange">lengthRange</code></td>
<td>
<p>An optional <code><a href="base.html#topic+numeric">numeric</a></code> vector of length two passed
to <code><a href="#topic+FragmentLengthNormalization">FragmentLengthNormalization</a></code>.</p>
</td></tr>
<tr><td><code id="doCRMAv1_+3A_arrays">arrays</code></td>
<td>
<p>A <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying the subset of arrays
to process.  If <code><a href="base.html#topic+NULL">NULL</a></code>, all arrays are considered.</p>
</td></tr>
<tr><td><code id="doCRMAv1_+3A_drop">drop</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the summaries are returned, otherwise
a named <code><a href="base.html#topic+list">list</a></code> of all intermediate and final results.</p>
</td></tr>
<tr><td><code id="doCRMAv1_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
<tr><td><code id="doCRMAv1_+3A_...">...</code></td>
<td>
<p>Additional arguments used to set up <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code> (when argument <code>dataSet</code> is specified).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named <code><a href="base.html#topic+list">list</a></code>, iff <code>drop == FALSE</code>, otherwise
only <code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code> object.
</p>


<h3>Allele-specific or only total-SNP signals</h3>

<p>If you wish to obtain allele-specific estimates for SNPs, which
are needed to call genotypes or infer parent-specific copy numbers,
then use argument <code>combineAlleles=FALSE</code>.  Total copy number
signals are still available.
If you know for certain that you will not use allele-specific
estimates, you will get slightly less noisy signals
(very small difference) if you use <code>combineAlleles=TRUE</code>.
</p>
<p><code>doASCRMAv1(...)</code> is a wrapper for
<code>doCRMAv1(..., combineAlleles=FALSE)</code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] H. Bengtsson, R. Irizarry, B. Carvalho &amp; T.P. Speed.
<em>Estimation and assessment of raw copy numbers at the
single locus level</em>,
Bioinformatics, 2008.<br />
</p>


<h3>See Also</h3>

<p>For CRMA v2 (recommended by authors), which is a single-array
improvement over CRMA v1, see <code><a href="#topic+doCRMAv2">doCRMAv2</a></code>().
</p>

<hr>
<h2 id='doCRMAv2'>A single-array preprocessing method for estimating full-resolution raw copy numbers from all Affymetrix genotyping arrays (CRMA v2)</h2><span id='topic+doCRMAv2.default'></span><span id='topic+doCRMAv2'></span><span id='topic+doCRMAv2.AffymetrixCelSet'></span><span id='topic+doASCRMAv2'></span><span id='topic+doASCRMAv2.default'></span>

<h3>Description</h3>

<p>A single-array preprocessing method for estimating full-resolution raw copy numbers from all Affymetrix genotyping arrays (CRMA v2) based on [1].
The algorithm is processed in bounded memory, meaning virtually
any number of arrays can be analyzed on also very limited computer
systems.
</p>
<p>We recommend CRMA v2 for estimating allele-specific as well total
SNP signals from Affymetrix SNP chips.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  ## S3 method for class 'AffymetrixCelSet'
doCRMAv2(csR, combineAlleles=TRUE, lengthRange=NULL, arrays=NULL,
  plm=c("AvgCnPlm", "RmaCnPlm"), drop=TRUE, verbose=FALSE, ...)
  ## Default S3 method:
doCRMAv2(dataSet, ..., verbose=FALSE)
  ## Default S3 method:
doASCRMAv2(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="doCRMAv2_+3A_csr">csR</code>, <code id="doCRMAv2_+3A_dataset">dataSet</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code> (or the name of an <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>).</p>
</td></tr>
<tr><td><code id="doCRMAv2_+3A_combinealleles">combineAlleles</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> specifying whether allele probe pairs
should be summed before modeling or not.</p>
</td></tr>
<tr><td><code id="doCRMAv2_+3A_lengthrange">lengthRange</code></td>
<td>
<p>An optional <code><a href="base.html#topic+numeric">numeric</a></code> vector of length two passed
to <code><a href="#topic+FragmentLengthNormalization">FragmentLengthNormalization</a></code>.</p>
</td></tr>
<tr><td><code id="doCRMAv2_+3A_arrays">arrays</code></td>
<td>
<p>A <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying the subset of arrays
to process.  If <code><a href="base.html#topic+NULL">NULL</a></code>, all arrays are considered.</p>
</td></tr>
<tr><td><code id="doCRMAv2_+3A_plm">plm</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying which type of
probe-summarization model to used.</p>
</td></tr>
<tr><td><code id="doCRMAv2_+3A_drop">drop</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the summaries are returned, otherwise
a named <code><a href="base.html#topic+list">list</a></code> of all intermediate and final results.</p>
</td></tr>
<tr><td><code id="doCRMAv2_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
<tr><td><code id="doCRMAv2_+3A_...">...</code></td>
<td>
<p>Additional arguments used to set up <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code> (when argument <code>dataSet</code> is specified).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named <code><a href="base.html#topic+list">list</a></code>, iff <code>drop == FALSE</code>, otherwise
only <code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code> object.
</p>


<h3>Allele-specific or only total-SNP signals</h3>

<p>If you wish to obtain allele-specific estimates for SNPs, which
are needed to call genotypes or infer parent-specific copy numbers,
then use argument <code>combineAlleles=FALSE</code>.  Total copy number
signals are still available.
If you know for certain that you will not use allele-specific
estimates, you will get slightly less noisy signals
(very small difference) if you use <code>combineAlleles=TRUE</code>.
</p>
<p><code>doASCRMAv2(...)</code> is a wrapper for
<code>doCRMAv2(..., combineAlleles=FALSE)</code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] H. Bengtsson, P. Wirapati &amp; T.P. Speed.
<em>A single-array preprocessing method for estimating
full-resolution raw copy numbers from all Affymetrix
genotyping arrays including GenomeWideSNP 5 &amp; 6</em>,
Bioinformatics, 2009.<br />
</p>


<h3>See Also</h3>

<p>For CRMA v1, which is a multi-array methods that precedes CRMA v2,
see <code><a href="#topic+doCRMAv1">doCRMAv1</a></code>().
</p>

<hr>
<h2 id='doFIRMA'>Finding Isoforms using Robust Multichip Analysis (FIRMA)</h2><span id='topic+doFIRMA.default'></span><span id='topic+doFIRMA'></span><span id='topic+doFIRMA.AffymetrixCelSet'></span>

<h3>Description</h3>

<p>Finding Isoforms using Robust Multichip Analysis (FIRMA) based on [1].
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  ## S3 method for class 'AffymetrixCelSet'
doFIRMA(csR, ..., flavor=c("v1b", "v1a"), drop=TRUE, verbose=FALSE)
  ## Default S3 method:
doFIRMA(dataSet, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="doFIRMA_+3A_csr">csR</code>, <code id="doFIRMA_+3A_dataset">dataSet</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code> (or the name of an <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>).</p>
</td></tr>
<tr><td><code id="doFIRMA_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+FirmaModel">FirmaModel</a></code>,
and to set up <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code> (when
argument <code>dataSet</code> is specified).</p>
</td></tr>
<tr><td><code id="doFIRMA_+3A_flavor">flavor</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying the flavor of FIRMA to use.</p>
</td></tr>
<tr><td><code id="doFIRMA_+3A_drop">drop</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the FIRMA scores are returned, otherwise
a named <code><a href="base.html#topic+list">list</a></code> of all intermediate and final results.</p>
</td></tr>
<tr><td><code id="doFIRMA_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named <code><a href="base.html#topic+list">list</a></code>, iff <code>drop == FALSE</code>, otherwise
only <code><a href="#topic+FirmaSet">FirmaSet</a></code> object (containing the FIRMA scores).
</p>


<h3>Using a custom exon-by-transcript CDF</h3>

<p>It is strongly recommended to use a custom CDF, e.g. &quot;core&quot;,
&quot;extended&quot; or &quot;full&quot; [1].  To use a custom CDF, set it before
calling this method, i.e. <code>setCdf(csR, cdf)</code>.
Do not set the standard &quot;non-supported&quot; Affymetrix CDF
(see also Section 'Flavors').
</p>


<h3>Flavors</h3>

<p>If <code>flavor == "v1b"</code> (default), then the standard
&quot;non-supported&quot; Affymetrix CDF is used for background correction
and the quantile normalization steps, and the custom CDF
is used for the probe summarization and everything that follows.
The advantage of this flavor is that those two first preprocessing
steps will remain the same if one later changes to a different
custom CDF.
</p>
<p>If <code>flavor == "v1a"</code>, then the custom CDF is used throughout
all steps of FIRMA, which means that if one changes the custom CDF
all steps will be redone.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] E. Purdom, K. Simpson, M. Robinson, J. Conboy, A. Lapuk &amp; T.P. Speed,
<em>FIRMA: a method for detection of alternative splicing from
exon array data</em>, Bioinformatics, 2008.<br />
</p>

<hr>
<h2 id='doGCRMA'>Robust Multichip Analysis (GCRMA)</h2><span id='topic+doGCRMA.default'></span><span id='topic+doGCRMA'></span><span id='topic+doGCRMA.AffymetrixCelSet'></span>

<h3>Description</h3>

<p>Robust Multichip Analysis (GCRMA) based on [1].
The algorithm is processed in bounded memory, meaning virtually
any number of arrays can be analyzed on also very limited computer
systems.
The method replicates the results of <code><a href="gcrma.html#topic+gcrma">gcrma</a></code>
(package <span class="pkg">gcrma</span>) with great precision.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  ## S3 method for class 'AffymetrixCelSet'
doGCRMA(csR, arrays=NULL, type=c("fullmodel", "affinities"), uniquePlm=FALSE, drop=TRUE,
  verbose=FALSE, ...)
  ## Default S3 method:
doGCRMA(dataSet, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="doGCRMA_+3A_csr">csR</code>, <code id="doGCRMA_+3A_dataset">dataSet</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code> (or the name of an <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>).</p>
</td></tr>
<tr><td><code id="doGCRMA_+3A_arrays">arrays</code></td>
<td>
<p>A <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying the subset of arrays
to process.  If <code><a href="base.html#topic+NULL">NULL</a></code>, all arrays are considered.</p>
</td></tr>
<tr><td><code id="doGCRMA_+3A_type">type</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying what type of model to
use for the GCRMA background correction.
For more details, see <code><a href="#topic+GcRmaBackgroundCorrection">GcRmaBackgroundCorrection</a></code>.</p>
</td></tr>
<tr><td><code id="doGCRMA_+3A_uniqueplm">uniquePlm</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the log-additive probe-summarization model
is done on probeset with <em>unique</em> sets of probes.
If <code><a href="base.html#topic+logical">FALSE</a></code>, the summarization is done on &quot;as-is&quot; probesets as
specified by the CDF.</p>
</td></tr>
<tr><td><code id="doGCRMA_+3A_drop">drop</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the summaries are returned, otherwise
a named <code><a href="base.html#topic+list">list</a></code> of all intermediate and final results.</p>
</td></tr>
<tr><td><code id="doGCRMA_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
<tr><td><code id="doGCRMA_+3A_...">...</code></td>
<td>
<p>Additional arguments used to set up <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code> (when argument <code>dataSet</code> is specified).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named <code><a href="base.html#topic+list">list</a></code>, iff <code>drop == FALSE</code>, otherwise
only <code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] Z. Wu, R. Irizarry, R. Gentleman, F.M. Murillo &amp; F. Spencer.
<em>A Model Based Background Adjustment for Oligonucleotide
Expression Arrays</em>, JASA, 2004.<br />
</p>

<hr>
<h2 id='doRMA'>Robust Multichip Analysis (RMA)</h2><span id='topic+doRMA.default'></span><span id='topic+doRMA'></span><span id='topic+doRMA.AffymetrixCelSet'></span>

<h3>Description</h3>

<p>Robust Multichip Analysis (RMA) based on [1].
The algorithm is processed in bounded memory, meaning virtually
any number of arrays can be analyzed on also very limited computer
systems.
The method replicates the results of <code><a href="affyPLM.html#topic+fitPLM">fitPLM</a></code>
(package <span class="pkg">affyPLM</span>) with great precision.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  ## S3 method for class 'AffymetrixCelSet'
doRMA(csR, arrays=NULL, flavor=c("affyPLM", "oligo"), uniquePlm=FALSE, drop=TRUE,
  verbose=FALSE, ...)
  ## Default S3 method:
doRMA(dataSet, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="doRMA_+3A_csr">csR</code>, <code id="doRMA_+3A_dataset">dataSet</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code> (or the name of an <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>).</p>
</td></tr>
<tr><td><code id="doRMA_+3A_arrays">arrays</code></td>
<td>
<p>A <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying the subset of arrays
to process.  If <code><a href="base.html#topic+NULL">NULL</a></code>, all arrays are considered.</p>
</td></tr>
<tr><td><code id="doRMA_+3A_flavor">flavor</code></td>
<td>
<p>A character string specifying what model fitting algorithm to be used, cf. <code><a href="#topic+RmaPlm">RmaPlm</a></code>.</p>
</td></tr>
<tr><td><code id="doRMA_+3A_uniqueplm">uniquePlm</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the log-additive probe-summarization model
is done on probeset with <em>unique</em> sets of probes.
If <code><a href="base.html#topic+logical">FALSE</a></code>, the summarization is done on &quot;as-is&quot; probesets as
specified by the CDF.</p>
</td></tr>
<tr><td><code id="doRMA_+3A_drop">drop</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the summaries are returned, otherwise
a named <code><a href="base.html#topic+list">list</a></code> of all intermediate and final results.</p>
</td></tr>
<tr><td><code id="doRMA_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
<tr><td><code id="doRMA_+3A_...">...</code></td>
<td>
<p>Additional arguments used to set up <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code> (when argument <code>dataSet</code> is specified).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named <code><a href="base.html#topic+list">list</a></code>, iff <code>drop == FALSE</code>, otherwise
only <code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] Irizarry et al.
<em>Summaries of Affymetrix GeneChip probe level data</em>.
NAR, 2003, 31, e15.<br />
</p>

<hr>
<h2 id='ExonChipEffectFile'>The ExonChipEffectFile class</h2><span id='topic+ExonChipEffectFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class ExonChipEffectFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ParameterCelFile">ParameterCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ChipEffectFile">ChipEffectFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>ExonChipEffectFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>ExonChipEffectFile</b><br />
extends <em><a href="#topic+ChipEffectFile">ChipEffectFile</a></em><br />
</p>
<p>This class represents estimates of chip effects in the probe-level models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExonChipEffectFile(..., mergeGroups=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExonChipEffectFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+ChipEffectFile">ChipEffectFile</a></code>.</p>
</td></tr>
<tr><td><code id="ExonChipEffectFile_+3A_mergegroups">mergeGroups</code></td>
<td>
<p>Specifies if the groups are merged or not for these
estimates.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ChipEffectFile</b>:<br />
as.character, extractChromosomalDataFrame, extractMatrix, extractTheta, findUnitsTodo, getAM, getAsFullCelFile, getCellIndices, getCellMapForMainCdf, getExpandedCellMap, getParameters, getUnitGroupCellArrayMap, getUnitGroupCellMatrixMap, getXAM, mergeGroups, readUnits, writeAsFullCelFile
</p>
<p><b>Methods inherited from ParameterCelFile</b>:<br />
extractDataFrame, extractMatrix, readUnits
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from AffymetrixCelFile</b>:<br />
allocateFromCdf, as.character, clone, createFrom, extractMatrix, fromFile, getAm, getCdf, getExtensionPattern, getFileFormat, getImage, getUnitNamesFile, getUnitTypesFile, highlight, image270, nbrOfCells, plotDensity, plotImage, plotMvsA, plotMvsX, range, setCdf, smoothScatterMvsA, writeImage
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Ken Simpson, Henrik Bengtsson</p>


<h3>See Also</h3>

<p>An object of this class is typically part of a <code><a href="#topic+ExonChipEffectSet">ExonChipEffectSet</a></code>.
</p>

<hr>
<h2 id='ExonChipEffectSet'>The ExonChipEffectSet class</h2><span id='topic+ExonChipEffectSet'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class ExonChipEffectSet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataSet">AromaMicroarrayDataSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ParameterCelSet">ParameterCelSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>ExonChipEffectSet</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>ExonChipEffectSet</b><br />
extends <a href="#topic+ChipEffectSet">ChipEffectSet</a><br />
</p>
<p>This class represents estimates of chip effects in the probe-level models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExonChipEffectSet(..., mergeGroups=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExonChipEffectSet_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code>.</p>
</td></tr>
<tr><td><code id="ExonChipEffectSet_+3A_mergegroups">mergeGroups</code></td>
<td>
<p>Specifies if groups (individual exons in a CDF
file) are merged or not for these estimates, i.e. whether
transcript-level expression is to be estimated.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>findUnitsTodo</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAverageFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getMergeGroups</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setMergeGroups</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ChipEffectSet</b>:<br />
as.character, boxplotStats, byPath, calculateBaseline, calculateFieldBoxplotStats, calculateNuseBoxplotStats, calculateRleBoxplotStats, extractAffyBatch, extractChromosomalDataFrame, extractExpressionSet, extractMatrix, extractTheta, findByName, findUnitsTodo, fromDataSet, getAM, getAsFullCelSet, getAverageFile, getBaseline, getCellIndices, getXAM, plotBoxplot, readUnits, updateUnits
</p>
<p><b>Methods inherited from ParameterCelSet</b>:<br />
extractDataFrame, extractMatrix
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from AffymetrixCelSet</b>:<br />
append, as, as.AffymetrixCelSet, as.character, averageQuantile, byName, byPath, clone, convertToUnique, doCRMAv1, doCRMAv2, doFIRMA, doGCRMA, doRMA, extractAffyBatch, extractFeatureSet, extractMatrix, extractSnpFeatureSet, findByName, getAverage, getAverageAsinh, getAverageFile, getAverageLog, getCdf, getChipType, getData, getIntensities, getPlatform, getTimestamps, getUnitGroupCellMap, getUnitIntensities, getUnitNamesFile, getUnitTypesFile, isDuplicated, justRMA, justSNPRMA, nbrOfArrays, normalizeQuantile, plotDensity, range, readUnits, setCdf, update2, writeSgr
</p>
<p><b>Methods inherited from AffymetrixFileSet</b>:<br />
as, as.AffymetrixFileSet, byPath, getDefaultFullName
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataSet</b>:<br />
as.AromaMicroarrayDataSetList, as.AromaMicroarrayDataSetTuple, getAromaFullNameTranslatorSet, getAverageFile, getChipType, getDefaultFullName, getPlatform, setAttributesBy, setAttributesBySampleAnnotationFile, setAttributesBySampleAnnotationSet, validate
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Ken Simpson, Henrik Bengtsson</p>

<hr>
<h2 id='ExonProbeAffinityFile'>The ExonProbeAffinityFile class</h2><span id='topic+ExonProbeAffinityFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class ExonProbeAffinityFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ParameterCelFile">ParameterCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeAffinityFile">ProbeAffinityFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>ExonProbeAffinityFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>ExonProbeAffinityFile</b><br />
extends <em><a href="#topic+ProbeAffinityFile">ProbeAffinityFile</a></em><br />
</p>
<p>This class represents estimates of probe affinities in exon array
probe-level models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExonProbeAffinityFile(..., mergeGroups=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExonProbeAffinityFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+ProbeAffinityFile">ProbeAffinityFile</a></code>.</p>
</td></tr>
<tr><td><code id="ExonProbeAffinityFile_+3A_mergegroups">mergeGroups</code></td>
<td>
<p>Specifies if the groups (exons) are merged or not for
these estimates.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from ProbeAffinityFile</b>:<br />
as.character, getCellIndices, getParameters, readUnits
</p>
<p><b>Methods inherited from ParameterCelFile</b>:<br />
extractDataFrame, extractMatrix, readUnits
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from AffymetrixCelFile</b>:<br />
allocateFromCdf, as.character, clone, createFrom, extractMatrix, fromFile, getAm, getCdf, getExtensionPattern, getFileFormat, getImage, getUnitNamesFile, getUnitTypesFile, highlight, image270, nbrOfCells, plotDensity, plotImage, plotMvsA, plotMvsX, range, setCdf, smoothScatterMvsA, writeImage
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Ken Simpson, Henrik Bengtsson</p>

<hr>
<h2 id='ExonRmaPlm'>The ExonRmaPlm class</h2><span id='topic+ExonRmaPlm'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class ExonRmaPlm</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+RmaPlm">RmaPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>ExonRmaPlm</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>ExonRmaPlm</b><br />
extends <em><a href="#topic+RmaPlm">RmaPlm</a></em><br />
</p>
<p>This class represents the log-additive model part of the Robust Multichip
Analysis (RMA) method described in Irizarry et al (2003), as implemented
for exon arrays.  The model may be fitted with exons merged into
transcripts (all probes fitted together) or on an individual exon basis
(probes within an exon treated as a group, but exons fitted separately).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExonRmaPlm(..., mergeGroups=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExonRmaPlm_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+RmaPlm">RmaPlm</a></code>.</p>
</td></tr>
<tr><td><code id="ExonRmaPlm_+3A_mergegroups">mergeGroups</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> flag specifying whether to merge exons
into transcripts.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCellIndices</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipEffectSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getProbeAffinityFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setMergeGroups</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from RmaPlm</b>:<br />
getAsteriskTags, getCalculateResidualsFunction, getParameters, getRlmFitFunctions
</p>
<p><b>Methods inherited from ProbeLevelModel</b>:<br />
calculateResidualSet, calculateWeights, fit, getAsteriskTags, getCalculateResidualsFunction, getChipEffectSet, getProbeAffinityFile, getResidualSet, getRootPath, getWeightsSet
</p>
<p><b>Methods inherited from MultiArrayUnitModel</b>:<br />
getListOfPriors, setListOfPriors, validate
</p>
<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Model</h3>

<p><code><a href="#topic+RmaPlm">RmaPlm</a></code>.
</p>


<h3>Author(s)</h3>

<p>Ken Simpson, Henrik Bengtsson, Elizabeth Purdom</p>


<h3>References</h3>

<p>Irizarry et al. <em>Summaries of Affymetrix GeneChip probe level data</em>.
NAR, 2003, 31, e15.<br />
</p>

<hr>
<h2 id='extractAffyBatch.AffymetrixCelSet'>Extracts an in-memory AffyBatch object from the CEL set</h2><span id='topic+extractAffyBatch.AffymetrixCelSet'></span><span id='topic+AffymetrixCelSet.extractAffyBatch'></span><span id='topic+extractAffyBatch+2CAffymetrixCelSet-method'></span><span id='topic+extractAffyBatch.ChipEffectSet'></span><span id='topic+extractAffyBatch'></span>

<h3>Description</h3>

<p>Extracts an in-memory AffyBatch object from the CEL set.
Note that any modifications done to the extract object will <em>not</em>
be reflected in the original CEL set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelSet'
extractAffyBatch(this, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractAffyBatch.AffymetrixCelSet_+3A_...">...</code></td>
<td>
<p>Argument passed to <code>ReadAffy()</code>
(<code><a href="affy.html#topic+read.affybatch">read.affybatch</a></code>).</p>
</td></tr>
<tr><td><code id="extractAffyBatch.AffymetrixCelSet_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Since the <span class="pkg">affy</span> package is making use of special CDF environment
packages, this method will warn if the needed package is missing and
explain that <span class="pkg">affy</span> will later try to download and install it
automatically.
</p>


<h3>Value</h3>

<p>Returns an <code><a href="affy.html#topic+AffyBatch-class">AffyBatch-class</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>Internally <code><a href="affy.html#topic+read.affybatch">read.affybatch</a></code> is used to read the data.
For more information see <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>

<hr>
<h2 id='extractDataFrame.ParameterCelSet'>Extract data as a data.frame for a set of arrays</h2><span id='topic+extractDataFrame.ParameterCelSet'></span><span id='topic+ParameterCelSet.extractDataFrame'></span><span id='topic+extractDataFrame+2CParameterCelSet-method'></span>

<h3>Description</h3>

<p>Extract data as a data.frame for a set of arrays.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ParameterCelSet'
extractDataFrame(this, addNames=FALSE, addUgcMap=TRUE, ..., drop=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractDataFrame.ParameterCelSet_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+extractMatrix.ParameterCelSet">*extractMatrix</a>()</code>.</p>
</td></tr>
<tr><td><code id="extractDataFrame.ParameterCelSet_+3A_addnames">addNames</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the first two columns contain the
unit names and the group names according the the CDF, otherwise
those two columns are not included.</p>
</td></tr>
<tr><td><code id="extractDataFrame.ParameterCelSet_+3A_addugcmap">addUgcMap</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the columns following the unit and
group names contains the (unit, group, cell) index map.</p>
</td></tr>
<tr><td><code id="extractDataFrame.ParameterCelSet_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a Jx(2+3+K) <code><a href="base.html#topic+data.frame">data.frame</a></code> where J is the number of units,
and K is the number of arrays.
The first two columns, if <code>addNames=TRUE</code>, contains the
unit names and the group names.
The next three columns contains the (unit, group, cell) index map.
The last K columns named by the arrays contain the data for the K arrays.
No names are set for the rows.
The rows are ordered according to <code>units</code> arguments.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+extractMatrix.ParameterCelSet">*extractMatrix</a>()</code>.
For more information see <code><a href="#topic+ParameterCelSet">ParameterCelSet</a></code>.
</p>

<hr>
<h2 id='extractExpressionSet.ChipEffectSet'>Extracts an in-memory ExpressionSet object</h2><span id='topic+extractExpressionSet.ChipEffectSet'></span><span id='topic+ChipEffectSet.extractExpressionSet'></span><span id='topic+extractExpressionSet+2CChipEffectSet-method'></span><span id='topic+extractExpressionSet'></span>

<h3>Description</h3>

<p>Extracts an in-memory ExpressionSet object from a <code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code> object.
Note that any modifications done to the extracted object will <em>not</em>
be reflected in the original chip-effect set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ChipEffectSet'
extractExpressionSet(this, ..., logBase=2, orderUnitsBy=c("asis", "lexicographic"),
  annotationPkg=NULL, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractExpressionSet.ChipEffectSet_+3A_...">...</code></td>
<td>
<p>Optional argument passed to <code><a href="#topic+extractMatrix.ChipEffectSet">*extractMatrix</a>()</code>.</p>
</td></tr>
<tr><td><code id="extractExpressionSet.ChipEffectSet_+3A_logbase">logBase</code></td>
<td>
<p>An <code><a href="base.html#topic+integer">integer</a></code> specifying the base to use when
log-transforming the signals.  If <code><a href="base.html#topic+NULL">NULL</a></code>, the signals are not
transformed, but kept as is.</p>
</td></tr>
<tr><td><code id="extractExpressionSet.ChipEffectSet_+3A_annotationpkg">annotationPkg</code></td>
<td>
<p>(optional) A <code><a href="base.html#topic+character">character</a></code> string specifies a
Bioconductor (ChipDb, CDF environment or PDInfo) annotation package,
which then sets the 'annotation' slot of the returned object.
If <code>"ChipDb"</code>, <code>"cdf"</code> or <code>"PDInfo"</code> th corresponding
ChipDB, CDF environment or PDInfo package is inferred from
the CDF's chip type.</p>
</td></tr>
<tr><td><code id="extractExpressionSet.ChipEffectSet_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="Biobase.html#topic+ExpressionSet-class">ExpressionSet-class</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code>.
</p>

<hr>
<h2 id='extractFeatureSet.AffymetrixCelSet'>Extracts CEL signals an in-memory FeatureSet object</h2><span id='topic+extractFeatureSet.AffymetrixCelSet'></span><span id='topic+AffymetrixCelSet.extractFeatureSet'></span><span id='topic+extractFeatureSet+2CAffymetrixCelSet-method'></span><span id='topic+extractFeatureSet'></span><span id='topic+AffymetrixCelSet.extractSnpFeatureSet'></span><span id='topic+extractSnpFeatureSet.AffymetrixCelSet'></span><span id='topic+extractSnpFeatureSet+2CAffymetrixCelSet-method'></span><span id='topic+extractSnpFeatureSet'></span>

<h3>Description</h3>

<p>Extracts CEL signals an in-memory FeatureSet object from a <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code> object.
Note that any modifications done to the extracted object will <em>not</em>
be reflected in the original CEL set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelSet'
extractFeatureSet(this, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractFeatureSet.AffymetrixCelSet_+3A_...">...</code></td>
<td>
<p>Additional argument passed to <code><a href="oligo.html#topic+read.celfiles">read.celfiles</a></code>.</p>
</td></tr>
<tr><td><code id="extractFeatureSet.AffymetrixCelSet_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="oligoClasses.html#topic+FeatureSet-class">FeatureSet-class</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>Internally <code><a href="oligo.html#topic+read.celfiles">read.celfiles</a></code> is used to read the data.
To read <em>summarized</em> data as a <code><a href="Biobase.html#topic+ExpressionSet-class">ExpressionSet-class</a></code>
object, see <code><a href="#topic+extractExpressionSet.ChipEffectSet">*extractExpressionSet</a>()</code>.
For more information see <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>

<hr>
<h2 id='extractMatrix.AffymetrixCelSet'>Extract data as a matrix for a set of arrays</h2><span id='topic+extractMatrix.AffymetrixCelSet'></span><span id='topic+AffymetrixCelSet.extractMatrix'></span><span id='topic+extractMatrix+2CAffymetrixCelSet-method'></span>

<h3>Description</h3>

<p>Extract data as a matrix for a set of arrays.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelSet'
extractMatrix(this, cells=NULL, ..., field=c("intensities", "stdvs", "pixels"),
  drop=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractMatrix.AffymetrixCelSet_+3A_cells">cells</code></td>
<td>
<p>(The subset of cells to be matched.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all cells are considered.</p>
</td></tr>
<tr><td><code id="extractMatrix.AffymetrixCelSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="extractMatrix.AffymetrixCelSet_+3A_field">field</code></td>
<td>
<p>The field to be extracted.</p>
</td></tr>
<tr><td><code id="extractMatrix.AffymetrixCelSet_+3A_drop">drop</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, singleton dimensions are dropped.</p>
</td></tr>
<tr><td><code id="extractMatrix.AffymetrixCelSet_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an JxK <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+matrix">matrix</a></code> where J is the number of units,
and K is the number of arrays.
The names of the columns are the names of the arrays.
No names are set for the rows.
The rows are ordered according to <code>cells</code> argument.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson, Mark Robinson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>

<hr>
<h2 id='extractMatrix.ParameterCelSet'>Extract data as a matrix for a set of arrays</h2><span id='topic+extractMatrix.ParameterCelSet'></span><span id='topic+ParameterCelSet.extractMatrix'></span><span id='topic+extractMatrix+2CParameterCelSet-method'></span>

<h3>Description</h3>

<p>Extract data as a matrix for a set of arrays.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ParameterCelSet'
extractMatrix(this, units=NULL, ..., field=c("intensities", "stdvs", "pixels"),
  returnUgcMap=FALSE, drop=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractMatrix.ParameterCelSet_+3A_units">units</code></td>
<td>
<p>(The subset of units to be matched.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are considered.</p>
</td></tr>
<tr><td><code id="extractMatrix.ParameterCelSet_+3A_...">...</code></td>
<td>
<p>Passed to <code><a href="base.html#topic+subset">subset</a></code>() operating on the UGC map.</p>
</td></tr>
<tr><td><code id="extractMatrix.ParameterCelSet_+3A_field">field</code></td>
<td>
<p>The field to be extracted.</p>
</td></tr>
<tr><td><code id="extractMatrix.ParameterCelSet_+3A_returnugcmap">returnUgcMap</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the (unit, group, cell) map is returned
as an attribute.</p>
</td></tr>
<tr><td><code id="extractMatrix.ParameterCelSet_+3A_drop">drop</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, singleton dimensions are dropped.</p>
</td></tr>
<tr><td><code id="extractMatrix.ParameterCelSet_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an JxK <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+matrix">matrix</a></code> where J is the number of units,
and K is the number of arrays.
The names of the columns are the names of the arrays.
No names are set for the rows.
The rows are ordered according to <code>units</code> arguments.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+extractDataFrame.ParameterCelSet">*extractDataFrame</a>()</code>.
For more information see <code><a href="#topic+ParameterCelSet">ParameterCelSet</a></code>.
</p>

<hr>
<h2 id='findByChipType.AromaChipTypeAnnotationFile'>Locates an annotation file by its chip type</h2><span id='topic+findByChipType.AromaChipTypeAnnotationFile'></span><span id='topic+AromaChipTypeAnnotationFile.findByChipType'></span><span id='topic+findByChipType+2CAromaChipTypeAnnotationFile-method'></span>

<h3>Description</h3>

<p>Locates an annotation file by its chip type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaChipTypeAnnotationFile'
findByChipType(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="findByChipType.AromaChipTypeAnnotationFile_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the pathname (as a <code><a href="base.html#topic+character">character</a></code> string) to the first annotation
chip type file found.  If no one was found, <code><a href="base.html#topic+NULL">NULL</a></code> is returned.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+byChipType.AromaChipTypeAnnotationFile">*byChipType</a>()</code>.
For more information see <code><a href="#topic+AromaChipTypeAnnotationFile">AromaChipTypeAnnotationFile</a></code>.
</p>

<hr>
<h2 id='findUnitsTodo.FirmaModel'>Identifies non-fitted units</h2><span id='topic+findUnitsTodo.FirmaModel'></span><span id='topic+FirmaModel.findUnitsTodo'></span><span id='topic+findUnitsTodo+2CFirmaModel-method'></span>

<h3>Description</h3>

<p>Identifies non-fitted units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FirmaModel'
findUnitsTodo(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="findUnitsTodo.FirmaModel_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> of unit indices.
</p>


<h3>See Also</h3>

<p>Internally this methods calls the same method for the
<code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code> class.
For more information see <code><a href="#topic+FirmaModel">FirmaModel</a></code>.
</p>

<hr>
<h2 id='findUnitsTodo.ProbeLevelModel'>Identifies non-fitted units</h2><span id='topic+findUnitsTodo.ProbeLevelModel'></span><span id='topic+ProbeLevelModel.findUnitsTodo'></span><span id='topic+findUnitsTodo+2CProbeLevelModel-method'></span>

<h3>Description</h3>

<p>Identifies non-fitted units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ProbeLevelModel'
findUnitsTodo(this, verbose=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="findUnitsTodo.ProbeLevelModel_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
<tr><td><code id="findUnitsTodo.ProbeLevelModel_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> of unit indices.
</p>


<h3>See Also</h3>

<p>Internally this methods calls the same method for the
<code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code> class.
For more information see <code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code>.
</p>

<hr>
<h2 id='findUnitsTodo.UnitModel'>Identifies non-fitted units</h2><span id='topic+findUnitsTodo.UnitModel'></span><span id='topic+UnitModel.findUnitsTodo'></span><span id='topic+findUnitsTodo+2CUnitModel-method'></span>

<h3>Description</h3>

<p>Identifies non-fitted units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'UnitModel'
findUnitsTodo(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="findUnitsTodo.UnitModel_+3A_...">...</code></td>
<td>
<p>Arguments specific to any subclass.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> of unit indices.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+UnitModel">UnitModel</a></code>.
</p>

<hr>
<h2 id='FirmaFile'>The FirmaFile class</h2><span id='topic+FirmaFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class FirmaFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ParameterCelFile">ParameterCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>FirmaFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>FirmaFile</b><br />
extends <em><a href="#topic+ParameterCelFile">ParameterCelFile</a></em><br />
</p>
<p>This class represents scores calculated by the FIRMA algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FirmaFile(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FirmaFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractMatrix</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>findUnitsTodo</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ParameterCelFile</b>:<br />
extractDataFrame, extractMatrix, readUnits
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from AffymetrixCelFile</b>:<br />
allocateFromCdf, as.character, clone, createFrom, extractMatrix, fromFile, getAm, getCdf, getExtensionPattern, getFileFormat, getImage, getUnitNamesFile, getUnitTypesFile, highlight, image270, nbrOfCells, plotDensity, plotImage, plotMvsA, plotMvsX, range, setCdf, smoothScatterMvsA, writeImage
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Ken Simpson, Henrik Bengtsson</p>


<h3>See Also</h3>

<p>An object of this class is typically part of a <code><a href="#topic+FirmaSet">FirmaSet</a></code>.
</p>

<hr>
<h2 id='FirmaModel'>The FirmaModel class</h2><span id='topic+FirmaModel'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class FirmaModel</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><em><code>FirmaModel</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>FirmaModel</b><br />
extends <em><a href="#topic+UnitModel">UnitModel</a></em><br />
</p>
<p>This class represents the FIRMA (Finding Isoforms using RMA) alternative
splicing model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FirmaModel(rmaPlm=NULL, summaryMethod=c("median", "upperQuartile", "max"),
  operateOn=c("residuals", "weights"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FirmaModel_+3A_rmaplm">rmaPlm</code></td>
<td>
<p>An @RmaPlm object.</p>
</td></tr>
<tr><td><code id="FirmaModel_+3A_summarymethod">summaryMethod</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> specifying what summarization method should be used.</p>
</td></tr>
<tr><td><code id="FirmaModel_+3A_operateon">operateOn</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> specifying what statistic to operate on.</p>
</td></tr>
<tr><td><code id="FirmaModel_+3A_...">...</code></td>
<td>
<p>Arguments passed to constructor of <code><a href="#topic+UnitModel">UnitModel</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>fit</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getDataSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFirmaSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPlm</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getTags</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Ken Simpson, Henrik Bengtsson</p>

<hr>
<h2 id='FirmaSet'>The FirmaSet class</h2><span id='topic+FirmaSet'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class FirmaSet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataSet">AromaMicroarrayDataSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ParameterCelSet">ParameterCelSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>FirmaSet</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>FirmaSet</b><br />
extends <a href="#topic+ParameterCelSet">ParameterCelSet</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FirmaSet(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FirmaSet_+3A_...">...</code></td>
<td>
<p>Arguments passed to constructor of <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractMatrix</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>findUnitsTodo</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCellIndices</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ParameterCelSet</b>:<br />
extractDataFrame, extractMatrix
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from AffymetrixCelSet</b>:<br />
append, as, as.AffymetrixCelSet, as.character, averageQuantile, byName, byPath, clone, convertToUnique, doCRMAv1, doCRMAv2, doFIRMA, doGCRMA, doRMA, extractAffyBatch, extractFeatureSet, extractMatrix, extractSnpFeatureSet, findByName, getAverage, getAverageAsinh, getAverageFile, getAverageLog, getCdf, getChipType, getData, getIntensities, getPlatform, getTimestamps, getUnitGroupCellMap, getUnitIntensities, getUnitNamesFile, getUnitTypesFile, isDuplicated, justRMA, justSNPRMA, nbrOfArrays, normalizeQuantile, plotDensity, range, readUnits, setCdf, update2, writeSgr
</p>
<p><b>Methods inherited from AffymetrixFileSet</b>:<br />
as, as.AffymetrixFileSet, byPath, getDefaultFullName
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataSet</b>:<br />
as.AromaMicroarrayDataSetList, as.AromaMicroarrayDataSetTuple, getAromaFullNameTranslatorSet, getAverageFile, getChipType, getDefaultFullName, getPlatform, setAttributesBy, setAttributesBySampleAnnotationFile, setAttributesBySampleAnnotationSet, validate
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Ken Simpson, Henrik Bengtsson</p>

<hr>
<h2 id='fit.FirmaModel'>Estimates the model parameters</h2><span id='topic+fit.FirmaModel'></span><span id='topic+FirmaModel.fit'></span><span id='topic+fit+2CFirmaModel-method'></span>

<h3>Description</h3>

<p>Estimates the model parameters for all or a subset of the units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FirmaModel'
fit(this, units="remaining", ..., ram=NULL, force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit.FirmaModel_+3A_units">units</code></td>
<td>
<p>The units to be fitted.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are considered.
If <code>remaining</code>, only non-fitted units are considered.
</p>
</td></tr>
<tr><td><code id="fit.FirmaModel_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code>readUnits()</code>.</p>
</td></tr>
<tr><td><code id="fit.FirmaModel_+3A_ram">ram</code></td>
<td>
<p>A <code><a href="base.html#topic+double">double</a></code> indicating if more or less units should
be loaded into memory at the same time.</p>
</td></tr>
<tr><td><code id="fit.FirmaModel_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, already fitted units are re-fitted, and
cached data is re-read.</p>
</td></tr>
<tr><td><code id="fit.FirmaModel_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All estimates are stored to file.
</p>
<p>The non-array specific parameter estimates together with standard deviation
estimates and convergence information are stored in one file.
</p>
<p>The parameter estimates specific to each array, typically &quot;chip effects&quot;,
are stored in array specific files.
</p>
<p>Data set specific estimates [L = number of probes]:
phi [L doubles] (probe affinities), sd(phi) [L doubles],
isOutlier(phi) [L logicals]
</p>
<p>Algorithm-specific results:
iter [1 integer], convergence1 [1 logical], convergence2 [1 logical]
dTheta [1 double]
sd(eps) - [1 double] estimated standard deviation of the error term
</p>
<p>Array-specific estimates [K = nbr of arrays]:
theta [K doubles] (chip effects), sd(theta) [K doubles],
isOutlier(theta) [K logicals]
</p>
<p>For each array and each unit group, we store:
1 theta, 1 sd(theta), 1 isOutlier(theta), i.e. (float, float, bit)
=&gt; For each array and each unit (with <code class="reqn">G_j</code> groups), we store:
<code class="reqn">G_j</code> theta, <code class="reqn">G_j</code> sd(theta), <code class="reqn">G_j</code> isOutlier(theta),
i.e. <code class="reqn">G_j</code>*(float, float, bit).
For optimal access we store all thetas first, then all sd(theta) and the
all isOutlier(theta).
To keep track of the number of groups in each unit, we have to have a
(unit, ngroups) map.  This can be obtained from getUnitNames() for the
AffymetrixCdfFile class.
</p>


<h3>Value</h3>

<p>Returns an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> of indices of the units fitted,
or <code><a href="base.html#topic+NULL">NULL</a></code> if no units was (had to be) fitted.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+FirmaModel">FirmaModel</a></code>.
</p>

<hr>
<h2 id='fit.Model'>Estimates the model parameters</h2><span id='topic+fit.Model'></span><span id='topic+Model.fit'></span><span id='topic+fit+2CModel-method'></span>

<h3>Description</h3>

<p>Estimates the model parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Model'
fit(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit.Model_+3A_...">...</code></td>
<td>
<p>Arguments specific to any subclass.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying what units where fitted.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+Model">Model</a></code>.
</p>

<hr>
<h2 id='fit.ProbeLevelModel'>Estimates the model parameters</h2><span id='topic+fit.ProbeLevelModel'></span><span id='topic+ProbeLevelModel.fit'></span><span id='topic+fit+2CProbeLevelModel-method'></span>

<h3>Description</h3>

<p>Estimates the model parameters for all or a subset of the units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ProbeLevelModel'
fit(this, units="remaining", ..., force=FALSE, ram=NULL, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit.ProbeLevelModel_+3A_units">units</code></td>
<td>
<p>The units to be fitted.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are considered.
If <code>remaining</code>, only non-fitted units are considered.
</p>
</td></tr>
<tr><td><code id="fit.ProbeLevelModel_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code>readUnits()</code>.</p>
</td></tr>
<tr><td><code id="fit.ProbeLevelModel_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, already fitted units are re-fitted, and
cached data is re-read.</p>
</td></tr>
<tr><td><code id="fit.ProbeLevelModel_+3A_ram">ram</code></td>
<td>
<p>A <code><a href="base.html#topic+double">double</a></code> indicating if more or less units should
be loaded into memory at the same time.</p>
</td></tr>
<tr><td><code id="fit.ProbeLevelModel_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All estimates are stored to file.
</p>
<p>The non-array specific parameter estimates together with standard deviation
estimates and convergence information are stored in one file.
</p>
<p>The parameter estimates specific to each array, typically &quot;chip effects&quot;,
are stored in array specific files.
</p>
<p>Data set specific estimates [L = number of probes]:
phi [L doubles] (probe affinities), sd(phi) [L doubles],
isOutlier(phi) [L logicals]
</p>
<p>Algorithm-specific results:
iter [1 integer], convergence1 [1 logical], convergence2 [1 logical]
dTheta [1 double]
sd(eps) - [1 double] estimated standard deviation of the error term
</p>
<p>Array-specific estimates [K = nbr of arrays]:
theta [K doubles] (chip effects), sd(theta) [K doubles],
isOutlier(theta) [K logicals]
</p>
<p>For each array and each unit group, we store:
1 theta, 1 sd(theta), 1 isOutlier(theta), i.e. (float, float, bit)
=&gt; For each array and each unit (with <code class="reqn">G_j</code> groups), we store:
<code class="reqn">G_j</code> theta, <code class="reqn">G_j</code> sd(theta), <code class="reqn">G_j</code> isOutlier(theta),
i.e. <code class="reqn">G_j</code>*(float, float, bit).
For optimal access we store all thetas first, then all sd(theta) and the
all isOutlier(theta).
To keep track of the number of groups in each unit, we have to have a
(unit, ngroups) map.  This can be obtained from getUnitNames() for the
AffymetrixCdfFile class.
</p>


<h3>Value</h3>

<p>Returns an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> of indices of the units fitted,
or <code><a href="base.html#topic+NULL">NULL</a></code> if no units was (had to be) fitted.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code>.
</p>

<hr>
<h2 id='fit.SingleArrayUnitModel'>Estimates the model parameters</h2><span id='topic+fit.SingleArrayUnitModel'></span><span id='topic+SingleArrayUnitModel.fit'></span><span id='topic+fit+2CSingleArrayUnitModel-method'></span>

<h3>Description</h3>

<p>Estimates the model parameters for all or a subset of the units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SingleArrayUnitModel'
fit(this, arrays=NULL, units="remaining", ..., force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit.SingleArrayUnitModel_+3A_arrays">arrays</code></td>
<td>
<p>The arrays to be fitted.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all arrays are considered.
If <code>remaining</code>, only non-fitted arrays are considered.
</p>
</td></tr>
<tr><td><code id="fit.SingleArrayUnitModel_+3A_units">units</code></td>
<td>
<p>The units to be fitted.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are considered.
If <code>remaining</code>, only non-fitted units are considered.
</p>
</td></tr>
<tr><td><code id="fit.SingleArrayUnitModel_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+readUnits.SingleArrayUnitModel">*readUnits</a>()</code>.</p>
</td></tr>
<tr><td><code id="fit.SingleArrayUnitModel_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, already fitted units are re-fitted, and
cached data is re-read.</p>
</td></tr>
<tr><td><code id="fit.SingleArrayUnitModel_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All estimates are stored to file.
</p>
<p>The parameter estimates specific to each array,
typically &quot;chip effects&quot;,
are stored in array specific files.
</p>
<p>Array-specific estimates [K = nbr of arrays]:
theta [K doubles] (chip effects), sd(theta) [K doubles],
isOutlier(theta) [K logicals]
</p>
<p>For each array and each unit group, we store:
1 theta, 1 sd(theta), 1 isOutlier(theta), i.e. (float, float, bit)
=&gt; For each array and each unit (with <code class="reqn">G_j</code> groups), we store:
<code class="reqn">G_j</code> theta, <code class="reqn">G_j</code> sd(theta), <code class="reqn">G_j</code> isOutlier(theta),
i.e. <code class="reqn">G_j</code>*(float, float, bit).
</p>


<h3>Value</h3>

<p>Returns nothing.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+SingleArrayUnitModel">SingleArrayUnitModel</a></code>.
</p>

<hr>
<h2 id='fit.SmoothMultiarrayModel'>Fits the model for one chromosome across samples</h2><span id='topic+fit.SmoothMultiarrayModel'></span><span id='topic+SmoothMultiarrayModel.fit'></span><span id='topic+fit+2CSmoothMultiarrayModel-method'></span>

<h3>Description</h3>

<p>Fits the model for one chromosome across samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SmoothMultiarrayModel'
fit(this, chromosome, force=FALSE, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit.SmoothMultiarrayModel_+3A_chromosome">chromosome</code></td>
<td>
<p>An <code><a href="base.html#topic+integer">integer</a></code> specifying the index of the chromosome to
be fitted.</p>
</td></tr>
<tr><td><code id="fit.SmoothMultiarrayModel_+3A_...">...</code></td>
<td>
<p>Additional arguments passed down to the internal fit function.</p>
</td></tr>
<tr><td><code id="fit.SmoothMultiarrayModel_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+SmoothMultiarrayModel">SmoothMultiarrayModel</a></code>.
</p>

<hr>
<h2 id='fitQuantileNormFcn.AffymetrixCelFile'>Fits quantile normalization functions for the arrays in the data set</h2><span id='topic+fitQuantileNormFcn.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.fitQuantileNormFcn'></span><span id='topic+fitQuantileNormFcn+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Fits quantile normalization functions for the arrays in the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelFile'
fitQuantileNormFcn(this, yTarget, subset=NULL, ..., spar=NULL, nknots=1024,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fitQuantileNormFcn.AffymetrixCelFile_+3A_ytarget">yTarget</code></td>
<td>
<p>The target probe signals.</p>
</td></tr>
<tr><td><code id="fitQuantileNormFcn.AffymetrixCelFile_+3A_subset">subset</code></td>
<td>
<p>An optional <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying the indices of the
subset of probes to be used to fit the normalization function.</p>
</td></tr>
<tr><td><code id="fitQuantileNormFcn.AffymetrixCelFile_+3A_spar">spar</code>, <code id="fitQuantileNormFcn.AffymetrixCelFile_+3A_nknots">nknots</code></td>
<td>
<p>Control parameters passed to
<code><a href="stats.html#topic+smooth.spline">smooth.spline</a></code>.</p>
</td></tr>
<tr><td><code id="fitQuantileNormFcn.AffymetrixCelFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="fitQuantileNormFcn.AffymetrixCelFile_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a normalization <code><a href="base.html#topic+function">function</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>

<hr>
<h2 id='FragmentEquivalentClassNormalization'>The FragmentEquivalentClassNormalization class</h2><span id='topic+FragmentEquivalentClassNormalization'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class FragmentEquivalentClassNormalization</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ChipEffectTransform">ChipEffectTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code>FragmentEquivalentClassNormalization</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>FragmentEquivalentClassNormalization</b><br />
extends <em><a href="#topic+ChipEffectTransform">ChipEffectTransform</a></em><br />
</p>
<p>This class represents a normalization method that corrects for systematic
effects between loci of different equivalent classes of pairs of
sequences that are recognized by the restriction enzymes that cut the
DNA studies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FragmentEquivalentClassNormalization(dataSet=NULL, ..., targetAvgs=NULL,
  subsetToFit="-XY")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FragmentEquivalentClassNormalization_+3A_dataset">dataSet</code></td>
<td>
<p>A <code><a href="#topic+CnChipEffectSet">CnChipEffectSet</a></code>.</p>
</td></tr>
<tr><td><code id="FragmentEquivalentClassNormalization_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the constructor of
<code><a href="#topic+ChipEffectTransform">ChipEffectTransform</a></code>.</p>
</td></tr>
<tr><td><code id="FragmentEquivalentClassNormalization_+3A_targetavgs">targetAvgs</code></td>
<td>
<p>An optional list of <code><a href="base.html#topic+function">function</a></code>s.
For each enzyme there is one target averages to which all arrays
should be normalized to.</p>
</td></tr>
<tr><td><code id="FragmentEquivalentClassNormalization_+3A_subsettofit">subsetToFit</code></td>
<td>
<p>The units from which the normalization curve should
be estimated.  If <code><a href="base.html#topic+NULL">NULL</a></code>, all are considered.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAromaUfcFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ChipEffectTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Requirements</h3>

<p>This class requires an UFC (Unit Fragment Class) annotation file.
</p>


<h3>Acknowledgments</h3>

<p>The idea of normalization signals stratified on enzyme
recognition sequences is credited to Jim Veitch and
Ben Bolstad at Affymetrix Inc. (2008) who have designed
a similar method for copy number estimation in the
Affymetrix' Genotype Console v2.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='FragmentLengthNormalization'>The FragmentLengthNormalization class</h2><span id='topic+FragmentLengthNormalization'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class FragmentLengthNormalization</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ChipEffectTransform">ChipEffectTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code>FragmentLengthNormalization</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>FragmentLengthNormalization</b><br />
extends <em><a href="#topic+ChipEffectTransform">ChipEffectTransform</a></em><br />
</p>
<p>This class represents a normalization method that corrects for PCR
fragment length effects on copy-number chip-effect estimates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FragmentLengthNormalization(dataSet=NULL, ..., target=targetFunctions, subsetToFit="-XY",
  lengthRange=NULL, onMissing=c("median", "ignore"), shift=0, targetFunctions=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FragmentLengthNormalization_+3A_dataset">dataSet</code></td>
<td>
<p>A <code><a href="#topic+SnpChipEffectSet">SnpChipEffectSet</a></code>.</p>
</td></tr>
<tr><td><code id="FragmentLengthNormalization_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the constructor of
<code><a href="#topic+ChipEffectTransform">ChipEffectTransform</a></code>.</p>
</td></tr>
<tr><td><code id="FragmentLengthNormalization_+3A_target">target</code></td>
<td>
<p>(Optional) A <code><a href="base.html#topic+character">character</a></code> string or a list of <code><a href="base.html#topic+function">function</a></code>s
specifying what to normalize toward.
For each enzyme there is one target function to which all arrays
should be normalized to.</p>
</td></tr>
<tr><td><code id="FragmentLengthNormalization_+3A_subsettofit">subsetToFit</code></td>
<td>
<p>The units from which the normalization curve should
be estimated.  If <code><a href="base.html#topic+NULL">NULL</a></code>, all are considered.</p>
</td></tr>
<tr><td><code id="FragmentLengthNormalization_+3A_lengthrange">lengthRange</code></td>
<td>
<p>If given, a <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length 2 specifying
the range of fragment lengths considered.  All fragments with lengths
outside this range are treated as if they were missing.</p>
</td></tr>
<tr><td><code id="FragmentLengthNormalization_+3A_onmissing">onMissing</code></td>
<td>
<p>Specifies how to normalize units for which the
fragment lengths are unknown.</p>
</td></tr>
<tr><td><code id="FragmentLengthNormalization_+3A_shift">shift</code></td>
<td>
<p>An optional amount the data points should be shifted
(translated).</p>
</td></tr>
<tr><td><code id="FragmentLengthNormalization_+3A_targetfunctions">targetFunctions</code></td>
<td>
<p>Deprecated.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For SNPs, the normalization function is estimated based on the total
chip effects, i.e. the sum of the allele signals.  The normalizing
is done by rescale the chip effects on the intensity scale such that
the mean of the total chip effects are the same across samples for
any given fragment length.  For allele-specific estimates, both alleles
are always rescaled by the same amount.  Thus, when normalizing
allele-specific chip effects, the total chip effects is change, but not
the relative allele signal, e.g. the allele B frequency.
</p>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ChipEffectTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Requirements</h3>

<p>This class requires a SNP information annotation file for the
chip type to be normalized.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='GcContentNormalization'>The GcContentNormalization class</h2><span id='topic+GcContentNormalization'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class GcContentNormalization</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ChipEffectTransform">ChipEffectTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code>GcContentNormalization</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>GcContentNormalization</b><br />
extends <em><a href="#topic+ChipEffectTransform">ChipEffectTransform</a></em><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GcContentNormalization(dataSet=NULL, ..., targetFunction=NULL, subsetToFit=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GcContentNormalization_+3A_dataset">dataSet</code></td>
<td>
<p>A <code><a href="#topic+CnChipEffectSet">CnChipEffectSet</a></code>.</p>
</td></tr>
<tr><td><code id="GcContentNormalization_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the constructor of
<code><a href="#topic+ChipEffectTransform">ChipEffectTransform</a></code>.</p>
</td></tr>
<tr><td><code id="GcContentNormalization_+3A_targetfunction">targetFunction</code></td>
<td>
<p>A <code><a href="base.html#topic+function">function</a></code>. The target function to which all arrays
should be normalized to.</p>
</td></tr>
<tr><td><code id="GcContentNormalization_+3A_subsettofit">subsetToFit</code></td>
<td>
<p>The units from which the normalization curve should
be estimated.  If <code><a href="base.html#topic+NULL">NULL</a></code>, all are considered.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ChipEffectTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Requirements</h3>

<p>This class requires an Aroma unit GC-content (UGC) file.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='GcContentNormalization2'>The GcContentNormalization2 class</h2><span id='topic+GcContentNormalization2'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class GcContentNormalization2</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ChipEffectTransform">ChipEffectTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AdditiveCovariatesNormalization">AdditiveCovariatesNormalization</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>GcContentNormalization2</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>GcContentNormalization2</b><br />
extends <em><a href="#topic+AdditiveCovariatesNormalization">AdditiveCovariatesNormalization</a></em><br />
</p>
<p>This class represents a normalization method that corrects for
annotation-data covariate effects on copy-number chip-effect estimates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GcContentNormalization2(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GcContentNormalization2_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+AdditiveCovariatesNormalization">AdditiveCovariatesNormalization</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plotCovariateEffects</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AdditiveCovariatesNormalization</b>:<br />
getAsteriskTags, getCdf, getCovariates, getOutputDataSet00, getParameters, process
</p>
<p><b>Methods inherited from ChipEffectTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='GcRmaBackgroundCorrection'>The GcRmaBackgroundCorrection class</h2><span id='topic+GcRmaBackgroundCorrection'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class GcRmaBackgroundCorrection</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+BackgroundCorrection">BackgroundCorrection</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>GcRmaBackgroundCorrection</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>GcRmaBackgroundCorrection</b><br />
extends <em><a href="#topic+BackgroundCorrection">BackgroundCorrection</a></em><br />
</p>
<p>This class represents the GCRMA background adjustment function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GcRmaBackgroundCorrection(..., indicesNegativeControl=NULL, affinities=NULL,
  type=c("fullmodel", "affinities"), opticalAdjust=TRUE, gsbAdjust=TRUE,
  gsbParameters=NULL, seed=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GcRmaBackgroundCorrection_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code>.</p>
</td></tr>
<tr><td><code id="GcRmaBackgroundCorrection_+3A_indicesnegativecontrol">indicesNegativeControl</code></td>
<td>
<p>Locations of any negative control
probes (e.g., the anti-genomic controls on the human exon array).
If <code><a href="base.html#topic+NULL">NULL</a></code> and <code>type == "affinities"</code>, then all non-PM probes
are used as the negative controls.</p>
</td></tr>
<tr><td><code id="GcRmaBackgroundCorrection_+3A_affinities">affinities</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of probe affinities, usually as
calculated by <code>computeAffinities()</code> of the
<code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> class.</p>
</td></tr>
<tr><td><code id="GcRmaBackgroundCorrection_+3A_type">type</code></td>
<td>
<p>Type (flavor) of background correction, which can
be either <code>"fullmodel"</code> (uses MMs; requires that the chip type
has PM/MM pairs) or <code>"affinities"</code> (uses probe sequence only).</p>
</td></tr>
<tr><td><code id="GcRmaBackgroundCorrection_+3A_gsbadjust">gsbAdjust</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, adjustment for specific binding is done,
otherwise not.</p>
</td></tr>
<tr><td><code id="GcRmaBackgroundCorrection_+3A_opticaladjust">opticalAdjust</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, correction for optical effect is done
first, utilizing <code><a href="#topic+OpticalBackgroundCorrection">OpticalBackgroundCorrection</a></code>.</p>
</td></tr>
<tr><td><code id="GcRmaBackgroundCorrection_+3A_gsbparameters">gsbParameters</code></td>
<td>
<p>Additional argument passed to the internal
<code>bgAdjustGcrma()</code> method.</p>
</td></tr>
<tr><td><code id="GcRmaBackgroundCorrection_+3A_seed">seed</code></td>
<td>
<p>An (optional) <code><a href="base.html#topic+integer">integer</a></code> specifying a temporary random seed
to be used during processing.  The random seed is set to its original
state when done.  If <code><a href="base.html#topic+NULL">NULL</a></code>, it is not set.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from BackgroundCorrection</b>:<br />
getParameters, process
</p>
<p><b>Methods inherited from ProbeLevelTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Ken Simpson, Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] Z. Wu, R. Irizarry, R. Gentleman, F.M. Murillo &amp; F. Spencer.
<em>A Model Based Background Adjustment for Oligonucleotide
Expression Arrays</em>, JASA, 2004.<br />
</p>

<hr>
<h2 id='GenericReporter'>The GenericReporter class</h2><span id='topic+GenericReporter'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class GenericReporter</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><em><code>GenericReporter</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AffymetrixCelSetReporter">AffymetrixCelSetReporter</a></em>, <em><a href="#topic+AffymetrixFileSetReporter">AffymetrixFileSetReporter</a></em>, <em><a href="#topic+SpatialReporter">SpatialReporter</a></em><br />
</p>
<p>public abstract static class <b>GenericReporter</b><br />
extends <a href="R.oo.html#topic+Object">Object</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GenericReporter(tags="*", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GenericReporter_+3A_tags">tags</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> of tags to be added to the output path.</p>
</td></tr>
<tr><td><code id="GenericReporter_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFullName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPath</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getTags</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setup</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='GenomeInformation'>The GenomeInformation class</h2><span id='topic+GenomeInformation'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class GenomeInformation</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>GenomeInformation</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AffymetrixCsvGenomeInformation">AffymetrixCsvGenomeInformation</a></em>, <em><a href="#topic+DChipGenomeInformation">DChipGenomeInformation</a></em>, <em><a href="#topic+UgpGenomeInformation">UgpGenomeInformation</a></em><br />
</p>
<p>public abstract static class <b>GenomeInformation</b><br />
extends <a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GenomeInformation(..., .verify=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GenomeInformation_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code>.</p>
</td></tr>
<tr><td><code id="GenomeInformation_+3A_.verify">.verify</code></td>
<td>
<p>For internal use only.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChromosomeStats</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChromosomes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getData</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPositions</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitsOnChromosome</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitsOnChromosomes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plotDensity</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='GenomeInformation+24byChipType'>Static method to define a genome information set by chip type</h2><span id='topic+GenomeInformation+24byChipType'></span><span id='topic+byChipType.GenomeInformation'></span><span id='topic+GenomeInformation.byChipType'></span><span id='topic+byChipType+2CGenomeInformation-method'></span>

<h3>Description</h3>

<p>Static method to define a genome information set by chip type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## GenomeInformation$byChipType()

## Don't use the below:
## S3 method for class 'GenomeInformation'
byChipType(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GenomeInformation+2B24byChipType_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="#topic+GenomeInformation">GenomeInformation</a></code> object.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+GenomeInformation">GenomeInformation</a></code>.
</p>

<hr>
<h2 id='GenomeInformation+24fromCdf'>Static method to define a genome information set from a CDF</h2><span id='topic+GenomeInformation+24fromCdf'></span><span id='topic+fromCdf.GenomeInformation'></span><span id='topic+GenomeInformation.fromCdf'></span><span id='topic+fromCdf+2CGenomeInformation-method'></span>

<h3>Description</h3>

<p>Static method to define a genome information set from a CDF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## GenomeInformation$fromCdf(cdf, ...)

## Don't use the below:
## S3 method for class 'GenomeInformation'
fromCdf(static, cdf, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GenomeInformation+2B24fromCdf_+3A_cdf">cdf</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.</p>
</td></tr>
<tr><td><code id="GenomeInformation+2B24fromCdf_+3A_...">...</code></td>
<td>
<p>Additional argument passed to <code><a href="#topic+byChipType.GenomeInformation">*byChipType</a>()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="#topic+GenomeInformation">GenomeInformation</a></code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+byChipType.GenomeInformation">*byChipType</a>()</code>.
For more information see <code><a href="#topic+GenomeInformation">GenomeInformation</a></code>.
</p>

<hr>
<h2 id='getAlias.GenericReporter'>Gets the alias of the report</h2><span id='topic+getAlias.GenericReporter'></span><span id='topic+GenericReporter.getAlias'></span><span id='topic+getAlias+2CGenericReporter-method'></span>

<h3>Description</h3>

<p>Gets the alias of the report.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GenericReporter'
getAlias(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAlias.GenericReporter_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code>, or <code><a href="base.html#topic+NULL">NULL</a></code> if no alias is set.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+setAlias.GenericReporter">*setAlias</a>()</code>.
<code><a href="#topic+getName.GenericReporter">*getName</a>()</code>.
For more information see <code><a href="#topic+GenericReporter">GenericReporter</a></code>.
</p>

<hr>
<h2 id='getAlias.Model'>Gets the name alias for the model</h2><span id='topic+getAlias.Model'></span><span id='topic+Model.getAlias'></span><span id='topic+getAlias+2CModel-method'></span>

<h3>Description</h3>

<p>Gets the name alias for the model, if set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Model'
getAlias(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAlias.Model_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string, or <code><a href="base.html#topic+NULL">NULL</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getName.Model">*getName</a>()</code>
<code><a href="#topic+setAlias.Model">*setAlias</a>()</code>
For more information see <code><a href="#topic+Model">Model</a></code>.
</p>

<hr>
<h2 id='getAlleleCellPairs.AffymetrixCdfFile'>Gets the cell indices of allele pairs</h2><span id='topic+getAlleleCellPairs.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.getAlleleCellPairs'></span><span id='topic+getAlleleCellPairs+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Gets the cell indices of allele pairs
in units of type &quot;genotyping&quot; with 2 or 4 unit groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
getAlleleCellPairs(this, units=NULL, stratifyBy=c("pm", "pmmm", "mm"), force=FALSE, ...,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAlleleCellPairs.AffymetrixCdfFile_+3A_units">units</code></td>
<td>
<p>A <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> of units to query.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are considered.</p>
</td></tr>
<tr><td><code id="getAlleleCellPairs.AffymetrixCdfFile_+3A_stratifyby">stratifyBy</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying what type of probes
to return.</p>
</td></tr>
<tr><td><code id="getAlleleCellPairs.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="getAlleleCellPairs.AffymetrixCdfFile_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a Nx2 <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+matrix">matrix</a></code> of cell indices, where each row is
a (PMA, PMB) probe pair.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='getAlleleProbePairs.AffymetrixCdfFile'>Gets the indices of probepairs with the same pair of SNP nucleotides</h2><span id='topic+getAlleleProbePairs.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.getAlleleProbePairs'></span><span id='topic+getAlleleProbePairs+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Gets the indices of probepairs with the same pair of SNP nucleotides.
Note that the order of allele A and allele B is irrelevant.
For instance, all probepairs with nucleotides (A,T) are calibrated
together with all probepairs with nucleotides (T,A) reversed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
getAlleleProbePairs(this, units=NULL, ignoreOrder=TRUE, force=FALSE, verbose=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAlleleProbePairs.AffymetrixCdfFile_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> object.</p>
</td></tr>
<tr><td><code id="getAlleleProbePairs.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named <code><a href="base.html#topic+list">list</a></code> where each element is a two-column <code><a href="base.html#topic+matrix">matrix</a></code> where
the column names are the nucleotides for the two alleles.
</p>


<h3>Benchmarking</h3>

<p>On an IBM Thinkpad A31 with 1.8GHz and 1GB RAM:
</p>

<dl>
<dt>Mapping10K_Xba142</dt><dd><p>10208 units &amp; 432964 cells: 11 seconds.</p>
</dd>
<dt>Mapping50K_Xba240</dt><dd><p>58960 SNPs &amp; 589,600 (PMA,PMB) probe pairs: 11 seconds.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='getAlleleProbePairs3.AffymetrixCdfFile'>Gets the indices of probepairs with the same pair of SNP nucleotides</h2><span id='topic+getAlleleProbePairs3.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.getAlleleProbePairs3'></span><span id='topic+getAlleleProbePairs3+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Gets the indices of probepairs with the same pair of SNP nucleotides.
Note that the order of allele A and allele B is irrelevant.
For instance, all probepairs with nucleotides (A,T) are calibrated
together with all probepairs with nucleotides (T,A) reversed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
getAlleleProbePairs3(this, units=NULL, ignoreOrder=TRUE, force=FALSE, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAlleleProbePairs3.AffymetrixCdfFile_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> object.</p>
</td></tr>
<tr><td><code id="getAlleleProbePairs3.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named <code><a href="base.html#topic+list">list</a></code> where each element is a two-column <code><a href="base.html#topic+matrix">matrix</a></code> where
the column names are the nucleotides for the two alleles.
</p>


<h3>Benchmarking</h3>

<p>On an IBM Thinkpad A31 with 1.8GHz and 1GB RAM:
</p>

<dl>
<dt>Mapping10K_Xba142</dt><dd><p>10208 units &amp; 432964 cells: 11 seconds.</p>
</dd>
<dt>Mapping50K_Xba240</dt><dd><p>58960 SNPs &amp; 589,600 (PMA,PMB) probe pairs: 11 seconds.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='getAM.ChipEffectFile'>Gets the log-intensities and log-ratios of chip effects for two arrays</h2><span id='topic+getAM.ChipEffectFile'></span><span id='topic+ChipEffectFile.getAM'></span><span id='topic+getAM+2CChipEffectFile-method'></span>

<h3>Description</h3>

<p>Gets the log-intensities and log-ratios of chip effects for two arrays.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ChipEffectFile'
getAM(this, other, units=NULL, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAM.ChipEffectFile_+3A_other">other</code></td>
<td>
<p>The second <code><a href="#topic+ChipEffectFile">ChipEffectFile</a></code> object used as the
reference.</p>
</td></tr>
<tr><td><code id="getAM.ChipEffectFile_+3A_units">units</code></td>
<td>
<p>(The subset of units to be matched.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are considered.</p>
</td></tr>
<tr><td><code id="getAM.ChipEffectFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="getAM.ChipEffectFile_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a Nx2 matrix where N is the number of units returned.
The names of the columns are A (log-intensities) and M (log-ratios).
The names of the rows are the unit indices (as indexed by the CDF).
The rows are ordered according to <code>units</code> arguments.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+getXAM.ChipEffectFile">*getXAM</a>()</code>.
For more information see <code><a href="#topic+ChipEffectFile">ChipEffectFile</a></code>.
</p>

<hr>
<h2 id='getAM.ChipEffectSet'>Gets the log-intensities and log-ratios of chip effects of the set relative to a reference chip effect file</h2><span id='topic+getAM.ChipEffectSet'></span><span id='topic+ChipEffectSet.getAM'></span><span id='topic+getAM+2CChipEffectSet-method'></span>

<h3>Description</h3>

<p>Gets the log-intensities and log-ratios of chip effects of the set relative to a reference chip effect file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ChipEffectSet'
getAM(this, other, units=NULL, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAM.ChipEffectSet_+3A_other">other</code></td>
<td>
<p>The second <code><a href="#topic+ChipEffectFile">ChipEffectFile</a></code> object used as the
reference.</p>
</td></tr>
<tr><td><code id="getAM.ChipEffectSet_+3A_units">units</code></td>
<td>
<p>(The subset of units to be matched.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are considered.</p>
</td></tr>
<tr><td><code id="getAM.ChipEffectSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="getAM.ChipEffectSet_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an Jx2xK <code><a href="base.html#topic+array">array</a></code> where J is the number of units, and K is
the number of arrays (arrays are always the last dimension).
The names of the columns are A (log-intensities) and M (log-ratios).
The names of the rows are the unit indices (as indexed by the CDF).
The rows are ordered according to <code>units</code> arguments.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code>.
</p>

<hr>
<h2 id='getAverageFile.AffymetrixCelSet'>Calculates the mean and the standard deviation of the cell signal (intensity, standard deviation etc.) across the CEL set</h2><span id='topic+getAverageFile.AffymetrixCelSet'></span><span id='topic+AffymetrixCelSet.getAverageFile'></span><span id='topic+getAverageFile+2CAffymetrixCelSet-method'></span>

<h3>Description</h3>

<p>Calculates the mean and the standard deviation of the cell signal (intensity, standard deviation etc.) across the CEL set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelSet'
getAverageFile(this, name=NULL, prefix="average", indices="remaining",
  field=c("intensities", "stdvs"), mean=c("median", "mean"), sd=c("mad", "sd"),
  na.rm=TRUE, g=NULL, h=NULL, ..., cellsPerChunk=moreCells * 10^7/length(this),
  moreCells=1, force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAverageFile.AffymetrixCelSet_+3A_name">name</code></td>
<td>
<p>The label of the calculated parameters.
If <code><a href="base.html#topic+NULL">NULL</a></code>, a default name format <code>&lt;prefix&gt;-&lt;mean&gt;-&lt;sd&gt;</code> is used.</p>
</td></tr>
<tr><td><code id="getAverageFile.AffymetrixCelSet_+3A_indices">indices</code></td>
<td>
<p>An <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying which cells to consider.
If <code>"remaining"</code>, only parameters for cells that have not been
are calculated.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all cells are used.</p>
</td></tr>
<tr><td><code id="getAverageFile.AffymetrixCelSet_+3A_mean">mean</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> of a <code><a href="base.html#topic+function">function</a></code> specifying the function used
to calculate the average.</p>
</td></tr>
<tr><td><code id="getAverageFile.AffymetrixCelSet_+3A_sd">sd</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> of a <code><a href="base.html#topic+function">function</a></code> specifying the function used
to calculate the standard deviation.</p>
</td></tr>
<tr><td><code id="getAverageFile.AffymetrixCelSet_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, <code><a href="base.html#topic+NA">NA</a></code>s are excluded before, otherwise not.</p>
</td></tr>
<tr><td><code id="getAverageFile.AffymetrixCelSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="getAverageFile.AffymetrixCelSet_+3A_cellsperchunk">cellsPerChunk</code></td>
<td>
<p>A <code><a href="base.html#topic+integer">integer</a></code> specifying the total number of cells
(across arrays) read into memory per chunk.</p>
</td></tr>
<tr><td><code id="getAverageFile.AffymetrixCelSet_+3A_morecells">moreCells</code></td>
<td>
<p>A <code><a href="base.html#topic+double">double</a></code> scalar indicating if more or less cells
should be used per chunk.</p>
</td></tr>
<tr><td><code id="getAverageFile.AffymetrixCelSet_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, parameters for cells already calculated are
recalculated, otherwise not.</p>
</td></tr>
<tr><td><code id="getAverageFile.AffymetrixCelSet_+3A_verbose">verbose</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, progress details are printed, otherwise not.
May also be a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The parameter estimates are stored as a CEL file of the same class as
the data files in the set.  The CEL file is named <code>&lt;name&gt;.cel</code>
and placed in the directory of the set.
Currently there is no specific data class for this file, but the average
cell signals are stored as &quot;intensities&quot;, the standard deviation of the
cell signals as &quot;stddevs&quot;, and the number of data points used for each
estimate is stored as &quot;pixels&quot;.
</p>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code> of the same class as the CEL set
averaged.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson, Ken Simpson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>

<hr>
<h2 id='getAverageFile.CnagCfhSet'>Calculates the mean and the standard deviation of the cell signal (intensity, standard deviation etc.) across the CFH set</h2><span id='topic+getAverageFile.CnagCfhSet'></span><span id='topic+CnagCfhSet.getAverageFile'></span><span id='topic+getAverageFile+2CCnagCfhSet-method'></span>

<h3>Description</h3>

<p>Calculates the mean and the standard deviation of the cell signal (intensity, standard deviation etc.) across the CFH set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CnagCfhSet'
getAverageFile(this, name=NULL, prefix="average", indices="remaining",
  field=c("intensities", "stdvs"), mean=c("median", "mean"), sd=c("mad", "sd"),
  na.rm=TRUE, g=NULL, h=NULL, ..., cellsPerChunk=moreCells * 10^7/length(this),
  moreCells=1, force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAverageFile.CnagCfhSet_+3A_name">name</code></td>
<td>
<p>The label of the calculated parameters.
If <code><a href="base.html#topic+NULL">NULL</a></code>, a default name format <code>&lt;prefix&gt;-&lt;mean&gt;-&lt;sd&gt;</code> is used.</p>
</td></tr>
<tr><td><code id="getAverageFile.CnagCfhSet_+3A_indices">indices</code></td>
<td>
<p>An <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying which cells to consider.
If <code>"remaining"</code>, only parameters for cells that have not been
are calculated.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all cells are used.</p>
</td></tr>
<tr><td><code id="getAverageFile.CnagCfhSet_+3A_mean">mean</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> of a <code><a href="base.html#topic+function">function</a></code> specifying the function used
to calculate the average.</p>
</td></tr>
<tr><td><code id="getAverageFile.CnagCfhSet_+3A_sd">sd</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> of a <code><a href="base.html#topic+function">function</a></code> specifying the function used
to calculate the standard deviation.</p>
</td></tr>
<tr><td><code id="getAverageFile.CnagCfhSet_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, <code><a href="base.html#topic+NA">NA</a></code>s are excluded before, otherwise not.</p>
</td></tr>
<tr><td><code id="getAverageFile.CnagCfhSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="getAverageFile.CnagCfhSet_+3A_cellsperchunk">cellsPerChunk</code></td>
<td>
<p>A <code><a href="base.html#topic+integer">integer</a></code> specifying the total number of cells
(across arrays) read into memory per chunk.</p>
</td></tr>
<tr><td><code id="getAverageFile.CnagCfhSet_+3A_morecells">moreCells</code></td>
<td>
<p>A <code><a href="base.html#topic+double">double</a></code> scalar indicating if more or less cells
should be used per chunk.</p>
</td></tr>
<tr><td><code id="getAverageFile.CnagCfhSet_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, parameters for cells already calculated are
recalculated, otherwise not.</p>
</td></tr>
<tr><td><code id="getAverageFile.CnagCfhSet_+3A_verbose">verbose</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, progress details are printed, otherwise not.
May also be a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The parameter estimates are stored as a CFH file of the same class as
the data files in the set.  The CFH file is named <code>&lt;name&gt;.cel</code>
and placed in the directory of the set.
Currently there is no specific data class for this file, but the average
cell signals are stored as &quot;intensities&quot;, the standard deviation of the
cell signals as &quot;stddevs&quot;, and the number of data points used for each
estimate is stored as &quot;pixels&quot;.
</p>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+CnagCfhSet">CnagCfhSet</a></code> of the same class as the CFH set
averaged.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+CnagCfhSet">CnagCfhSet</a></code>.
</p>

<hr>
<h2 id='getBaseline.ChipEffectSet'>Gets the baseline signals across chromosomes</h2><span id='topic+getBaseline.ChipEffectSet'></span><span id='topic+ChipEffectSet.getBaseline'></span><span id='topic+getBaseline+2CChipEffectSet-method'></span>

<h3>Description</h3>

<p>Gets the baseline signals across chromosomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ChipEffectSet'
getBaseline(this, force=FALSE, verbose=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getBaseline.ChipEffectSet_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the CEL file that stores the is recreated.</p>
</td></tr>
<tr><td><code id="getBaseline.ChipEffectSet_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
<tr><td><code id="getBaseline.ChipEffectSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code>.
</p>

<hr>
<h2 id='getCdf.AffymetrixCelFile'>Gets the CDF structure for this CEL file</h2><span id='topic+getCdf.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.getCdf'></span><span id='topic+getCdf+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Gets the CDF structure for this CEL file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelFile'
getCdf(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCdf.AffymetrixCelFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+setCdf.AffymetrixCelFile">*setCdf</a>()</code>.
For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>

<hr>
<h2 id='getCdf.AffymetrixCelSet'>Gets the CDF structure for this CEL set</h2><span id='topic+getCdf.AffymetrixCelSet'></span><span id='topic+AffymetrixCelSet.getCdf'></span><span id='topic+getCdf+2CAffymetrixCelSet-method'></span>

<h3>Description</h3>

<p>Gets the CDF structure for this CEL set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelSet'
getCdf(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCdf.AffymetrixCelSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+setCdf.AffymetrixCelSet">*setCdf</a>()</code>.
For more information see <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>

<hr>
<h2 id='getCdf.CnagCfhFile'>Gets the CDF structure for this CEL file</h2><span id='topic+getCdf.CnagCfhFile'></span><span id='topic+CnagCfhFile.getCdf'></span><span id='topic+getCdf+2CCnagCfhFile-method'></span>

<h3>Description</h3>

<p>Gets the CDF structure for this CEL file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CnagCfhFile'
getCdf(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCdf.CnagCfhFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+setCdf.CnagCfhFile">*setCdf</a>()</code>.
For more information see <code><a href="#topic+CnagCfhFile">CnagCfhFile</a></code>.
</p>

<hr>
<h2 id='getCdf.CnagCfhSet'>Gets the CDF structure for this CFH set</h2><span id='topic+getCdf.CnagCfhSet'></span><span id='topic+CnagCfhSet.getCdf'></span><span id='topic+getCdf+2CCnagCfhSet-method'></span>

<h3>Description</h3>

<p>Gets the CDF structure for this CFH set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CnagCfhSet'
getCdf(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCdf.CnagCfhSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+setCdf.CnagCfhSet">*setCdf</a>()</code>.
For more information see <code><a href="#topic+CnagCfhSet">CnagCfhSet</a></code>.
</p>

<hr>
<h2 id='getCdf.Model'>Gets the CDF structure for this model</h2><span id='topic+getCdf.Model'></span><span id='topic+Model.getCdf'></span><span id='topic+getCdf+2CModel-method'></span>

<h3>Description</h3>

<p>Gets the CDF structure for this model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Model'
getCdf(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCdf.Model_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> object.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+Model">Model</a></code>.
</p>

<hr>
<h2 id='getCellIndices.AffymetrixCdfFile'>Gets the cell indices unit by unit</h2><span id='topic+getCellIndices.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.getCellIndices'></span><span id='topic+getCellIndices+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Gets the cell indices unit by unit of all or a subset of the units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
getCellIndices(this, units=NULL, ..., useNames=TRUE, unlist=FALSE, force=FALSE,
  cache=TRUE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCellIndices.AffymetrixCdfFile_+3A_units">units</code></td>
<td>
<p>The units of interest. If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are considered.</p>
</td></tr>
<tr><td><code id="getCellIndices.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to
<code><a href="affxparser.html#topic+readCdfCellIndices">readCdfCellIndices</a></code>.</p>
</td></tr>
<tr><td><code id="getCellIndices.AffymetrixCdfFile_+3A_usenames">useNames</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, element names are returned, otherwise not.</p>
</td></tr>
<tr><td><code id="getCellIndices.AffymetrixCdfFile_+3A_unlist">unlist</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the unlisted result is returned. Using this
argument is more memory efficient that calling <code><a href="base.html#topic+unlist">unlist</a></code>()
afterwards.</p>
</td></tr>
<tr><td><code id="getCellIndices.AffymetrixCdfFile_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, cached values are ignored.</p>
</td></tr>
<tr><td><code id="getCellIndices.AffymetrixCdfFile_+3A_cache">cache</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, results are cached, if not too large.</p>
</td></tr>
<tr><td><code id="getCellIndices.AffymetrixCdfFile_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note, that it is much more memory efficient to do
<code>getCellIndices(cdf, useNames=FALSE, unlist=TRUE)</code>
compare with <code>unlist(getCellIndices(cdf), use.names=FALSE)</code>.
</p>


<h3>Value</h3>

<p>Returns the <code><a href="base.html#topic+list">list</a></code> structure returned by
<code><a href="affxparser.html#topic+readCdfCellIndices">readCdfCellIndices</a></code>.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+setRestructor.AffymetrixCdfFile">*setRestructor</a>()</code> to set a default re-constructor for
the returned CDF structure.
For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='getCellIndices.ChipEffectFile'>Retrieves tree list of cell indices for a set of units</h2><span id='topic+getCellIndices.ChipEffectFile'></span><span id='topic+ChipEffectFile.getCellIndices'></span><span id='topic+getCellIndices+2CChipEffectFile-method'></span>

<h3>Description</h3>

<p>Retrieves tree list of cell indices for a set of units from the associated CDF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ChipEffectFile'
getCellIndices(this, ..., .cache=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCellIndices.ChipEffectFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>getCellIndices()</code>
of <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.</p>
</td></tr>
<tr><td><code id="getCellIndices.ChipEffectFile_+3A_.cache">.cache</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> structure, where each element corresponds to a unit.
If argument <code>unlist=TRUE</code> is passed, an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> is returned.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ChipEffectFile">ChipEffectFile</a></code>.
</p>

<hr>
<h2 id='getCellIndices.CnChipEffectFile'>Retrieves tree list of cell indices for a set of units</h2><span id='topic+getCellIndices.CnChipEffectFile'></span><span id='topic+CnChipEffectFile.getCellIndices'></span><span id='topic+getCellIndices+2CCnChipEffectFile-method'></span>

<h3>Description</h3>

<p>Retrieves tree list of cell indices for a set of units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CnChipEffectFile'
getCellIndices(this, units=NULL, ..., unlist=FALSE, force=FALSE, .cache=TRUE,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCellIndices.CnChipEffectFile_+3A_units">units</code></td>
<td>
<p>Unit indices to be retrieved.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are considered.</p>
</td></tr>
<tr><td><code id="getCellIndices.CnChipEffectFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>getCellIndices()</code>
of <code><a href="#topic+ChipEffectFile">ChipEffectFile</a></code>.</p>
</td></tr>
<tr><td><code id="getCellIndices.CnChipEffectFile_+3A_unlist">unlist</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the cell indices are returned as a <code><a href="base.html#topic+vector">vector</a></code>.</p>
</td></tr>
<tr><td><code id="getCellIndices.CnChipEffectFile_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the cell indices are re-read regardless whether
they are already cached in memory or not.</p>
</td></tr>
<tr><td><code id="getCellIndices.CnChipEffectFile_+3A_.cache">.cache</code></td>
<td>
<p>(internal) If <code><a href="base.html#topic+logical">TRUE</a></code>, the result is cached in memory.</p>
</td></tr>
<tr><td><code id="getCellIndices.CnChipEffectFile_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> structure, where each element corresponds to a unit.
If argument <code>unlist=TRUE</code> is passed, an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> is returned.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+CnChipEffectFile">CnChipEffectFile</a></code>.
</p>

<hr>
<h2 id='getCellIndices.CnProbeAffinityFile'>Retrieves tree list of cell indices for a set of units</h2><span id='topic+getCellIndices.CnProbeAffinityFile'></span><span id='topic+CnProbeAffinityFile.getCellIndices'></span><span id='topic+getCellIndices+2CCnProbeAffinityFile-method'></span>

<h3>Description</h3>

<p>Retrieves tree list of cell indices for a set of units from the associated CDF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CnProbeAffinityFile'
getCellIndices(this, ..., unlist=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCellIndices.CnProbeAffinityFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>getCellIndices()</code>
of <code><a href="#topic+SnpProbeAffinityFile">SnpProbeAffinityFile</a></code>.</p>
</td></tr>
<tr><td><code id="getCellIndices.CnProbeAffinityFile_+3A_unlist">unlist</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the cell indices are returned as a <code><a href="base.html#topic+vector">vector</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> structure, where each element corresponds to a unit.
If argument <code>unlist=TRUE</code> is passed, an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> is returned.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+CnProbeAffinityFile">CnProbeAffinityFile</a></code>.
</p>

<hr>
<h2 id='getCellIndices.ExonChipEffectFile'>Retrieves tree list of cell indices for a set of units</h2><span id='topic+getCellIndices.ExonChipEffectFile'></span><span id='topic+ExonChipEffectFile.getCellIndices'></span><span id='topic+getCellIndices+2CExonChipEffectFile-method'></span>

<h3>Description</h3>

<p>Retrieves tree list of cell indices for a set of units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ExonChipEffectFile'
getCellIndices(this, ..., unlist=FALSE, force=FALSE, .cache=TRUE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCellIndices.ExonChipEffectFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>getCellIndices()</code>
of <code><a href="#topic+ChipEffectFile">ChipEffectFile</a></code>.</p>
</td></tr>
<tr><td><code id="getCellIndices.ExonChipEffectFile_+3A_unlist">unlist</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the cell indices are returned as a <code><a href="base.html#topic+vector">vector</a></code>.</p>
</td></tr>
<tr><td><code id="getCellIndices.ExonChipEffectFile_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the cell indices are re-read regardless whether
they are already cached in memory or not.</p>
</td></tr>
<tr><td><code id="getCellIndices.ExonChipEffectFile_+3A_.cache">.cache</code></td>
<td>
<p>(internal) If <code><a href="base.html#topic+logical">TRUE</a></code>, the result is cached in memory.</p>
</td></tr>
<tr><td><code id="getCellIndices.ExonChipEffectFile_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> structure, where each element corresponds to a unit.
If argument <code>unlist=TRUE</code> is passed, an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> is returned.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ExonChipEffectFile">ExonChipEffectFile</a></code>.
</p>

<hr>
<h2 id='getCellIndices.ExonProbeAffinityFile'>Retrieves tree list of cell indices for a set of units</h2><span id='topic+getCellIndices.ExonProbeAffinityFile'></span><span id='topic+ExonProbeAffinityFile.getCellIndices'></span><span id='topic+getCellIndices+2CExonProbeAffinityFile-method'></span>

<h3>Description</h3>

<p>Retrieves tree list of cell indices for a set of units from the associated CDF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ExonProbeAffinityFile'
getCellIndices(this, ..., unlist=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCellIndices.ExonProbeAffinityFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>getCellIndices()</code>
of <code><a href="#topic+ProbeAffinityFile">ProbeAffinityFile</a></code>.</p>
</td></tr>
<tr><td><code id="getCellIndices.ExonProbeAffinityFile_+3A_unlist">unlist</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the cell indices are returned as a <code><a href="base.html#topic+vector">vector</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> structure, where each element corresponds to a unit.
If argument <code>unlist=TRUE</code> is passed, an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> is returned.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ExonProbeAffinityFile">ExonProbeAffinityFile</a></code>.
</p>

<hr>
<h2 id='getCellIndices.FirmaFile'>Retrieves tree list of cell indices for a set of units</h2><span id='topic+getCellIndices.FirmaFile'></span><span id='topic+FirmaFile.getCellIndices'></span><span id='topic+getCellIndices+2CFirmaFile-method'></span>

<h3>Description</h3>

<p>Retrieves tree list of cell indices for a set of units from the associated CDF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FirmaFile'
getCellIndices(this, ..., .cache=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCellIndices.FirmaFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>getCellIndices()</code>
of <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.</p>
</td></tr>
<tr><td><code id="getCellIndices.FirmaFile_+3A_.cache">.cache</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> structure, where each element corresponds to a unit.
If argument <code>unlist=TRUE</code> is passed, an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> is returned.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+FirmaFile">FirmaFile</a></code>.
</p>

<hr>
<h2 id='getCellIndices.ProbeAffinityFile'>Retrieves tree list of cell indices for a set of units</h2><span id='topic+getCellIndices.ProbeAffinityFile'></span><span id='topic+ProbeAffinityFile.getCellIndices'></span><span id='topic+getCellIndices+2CProbeAffinityFile-method'></span>

<h3>Description</h3>

<p>Retrieves tree list of cell indices for a set of units from the associated CDF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ProbeAffinityFile'
getCellIndices(this, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCellIndices.ProbeAffinityFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>getCellIndices()</code>
of <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.</p>
</td></tr>
<tr><td><code id="getCellIndices.ProbeAffinityFile_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> structure, where each element corresponds to a unit.
If argument <code>unlist=TRUE</code> is passed, an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> is returned.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ProbeAffinityFile">ProbeAffinityFile</a></code>.
</p>

<hr>
<h2 id='getCellIndices.ResidualFile'>Retrieves tree list of cell indices for a set of units</h2><span id='topic+getCellIndices.ResidualFile'></span><span id='topic+ResidualFile.getCellIndices'></span><span id='topic+getCellIndices+2CResidualFile-method'></span>

<h3>Description</h3>

<p>Retrieves tree list of cell indices for a set of units from the associated CDF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ResidualFile'
getCellIndices(this, ..., .cache=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCellIndices.ResidualFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>getCellIndices()</code>
of <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.</p>
</td></tr>
<tr><td><code id="getCellIndices.ResidualFile_+3A_.cache">.cache</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> structure, where each element corresponds to a unit.
If argument <code>unlist=TRUE</code> is passed, an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> is returned.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ResidualFile">ResidualFile</a></code>.
</p>

<hr>
<h2 id='getCellIndices.SnpChipEffectFile'>Retrieves tree list of cell indices for a set of units</h2><span id='topic+getCellIndices.SnpChipEffectFile'></span><span id='topic+SnpChipEffectFile.getCellIndices'></span><span id='topic+getCellIndices+2CSnpChipEffectFile-method'></span>

<h3>Description</h3>

<p>Retrieves tree list of cell indices for a set of units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SnpChipEffectFile'
getCellIndices(this, units=NULL, ..., unlist=FALSE, force=FALSE, .cache=TRUE,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCellIndices.SnpChipEffectFile_+3A_units">units</code></td>
<td>
<p>Unit indices to be retrieved.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are considered.</p>
</td></tr>
<tr><td><code id="getCellIndices.SnpChipEffectFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>getCellIndices()</code>
of <code><a href="#topic+ChipEffectFile">ChipEffectFile</a></code>.</p>
</td></tr>
<tr><td><code id="getCellIndices.SnpChipEffectFile_+3A_unlist">unlist</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the cell indices are returned as a <code><a href="base.html#topic+vector">vector</a></code>.</p>
</td></tr>
<tr><td><code id="getCellIndices.SnpChipEffectFile_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the cell indices are re-read regardless whether
they are already cached in memory or not.</p>
</td></tr>
<tr><td><code id="getCellIndices.SnpChipEffectFile_+3A_.cache">.cache</code></td>
<td>
<p>(internal) If <code><a href="base.html#topic+logical">TRUE</a></code>, the result is cached in memory.</p>
</td></tr>
<tr><td><code id="getCellIndices.SnpChipEffectFile_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> structure, where each element corresponds to a unit.
If argument <code>unlist=TRUE</code> is passed, an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> is returned.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+SnpChipEffectFile">SnpChipEffectFile</a></code>.
</p>

<hr>
<h2 id='getCellIndices.SnpProbeAffinityFile'>Retrieves tree list of cell indices for a set of units</h2><span id='topic+getCellIndices.SnpProbeAffinityFile'></span><span id='topic+SnpProbeAffinityFile.getCellIndices'></span><span id='topic+getCellIndices+2CSnpProbeAffinityFile-method'></span>

<h3>Description</h3>

<p>Retrieves tree list of cell indices for a set of units from the associated CDF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SnpProbeAffinityFile'
getCellIndices(this, ..., unlist=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCellIndices.SnpProbeAffinityFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>getCellIndices()</code>
of <code><a href="#topic+ProbeAffinityFile">ProbeAffinityFile</a></code>.</p>
</td></tr>
<tr><td><code id="getCellIndices.SnpProbeAffinityFile_+3A_unlist">unlist</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the cell indices are returned as a <code><a href="base.html#topic+vector">vector</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> structure, where each element corresponds to a unit.
If argument <code>unlist=TRUE</code> is passed, an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> is returned.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+SnpProbeAffinityFile">SnpProbeAffinityFile</a></code>.
</p>

<hr>
<h2 id='getCellIndices.UnitModel'>Gets the cell indices unit by unit</h2><span id='topic+getCellIndices.UnitModel'></span><span id='topic+UnitModel.getCellIndices'></span><span id='topic+getCellIndices+2CUnitModel-method'></span>

<h3>Description</h3>

<p>Gets the cell indices unit by unit for all or a subset of units (probesets).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'UnitModel'
getCellIndices(this, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCellIndices.UnitModel_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>getCellIndices()</code>
of the <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> of the input data set.</p>
</td></tr>
<tr><td><code id="getCellIndices.UnitModel_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the <code><a href="base.html#topic+list">list</a></code> structure consisting of CDF cell indices.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+UnitModel">UnitModel</a></code>.
</p>

<hr>
<h2 id='getCellIndices.WeightsFile'>Retrieves tree list of cell indices for a set of units</h2><span id='topic+getCellIndices.WeightsFile'></span><span id='topic+WeightsFile.getCellIndices'></span><span id='topic+getCellIndices+2CWeightsFile-method'></span>

<h3>Description</h3>

<p>Retrieves tree list of cell indices for a set of units from the associated CDF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'WeightsFile'
getCellIndices(this, ..., .cache=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCellIndices.WeightsFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>getCellIndices()</code>
of <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.</p>
</td></tr>
<tr><td><code id="getCellIndices.WeightsFile_+3A_.cache">.cache</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> structure, where each element corresponds to a unit.
If argument <code>unlist=TRUE</code> is passed, an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> is returned.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+WeightsFile">WeightsFile</a></code>.
</p>

<hr>
<h2 id='getChipEffectSet.AlleleSummation'>Gets the set of chip effects for this model</h2><span id='topic+getChipEffectSet.AlleleSummation'></span><span id='topic+AlleleSummation.getChipEffectSet'></span><span id='topic+getChipEffectSet+2CAlleleSummation-method'></span>

<h3>Description</h3>

<p>Gets the set of chip effects for this model.
There is one chip-effect file per array.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AlleleSummation'
getChipEffectSet(this, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getChipEffectSet.AlleleSummation_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="getChipEffectSet.AlleleSummation_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code> object.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AlleleSummation">AlleleSummation</a></code>.
</p>

<hr>
<h2 id='getChipEffectSet.ProbeLevelModel'>Gets the set of chip effects for this model</h2><span id='topic+getChipEffectSet.ProbeLevelModel'></span><span id='topic+ProbeLevelModel.getChipEffectSet'></span><span id='topic+getChipEffectSet+2CProbeLevelModel-method'></span><span id='topic+ProbeLevelModel.getChipEffects'></span><span id='topic+getChipEffects.ProbeLevelModel'></span><span id='topic+getChipEffects+2CProbeLevelModel-method'></span>

<h3>Description</h3>

<p>Gets the set of chip effects for this model.
There is one chip-effect file per array.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ProbeLevelModel'
getChipEffectSet(this, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getChipEffectSet.ProbeLevelModel_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code>getMonocellCdf()</code> of
<code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.</p>
</td></tr>
<tr><td><code id="getChipEffectSet.ProbeLevelModel_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code> object.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code>.
</p>

<hr>
<h2 id='getChipType.AffymetrixCelFile'>Gets the chip type for this CEL file</h2><span id='topic+getChipType.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.getChipType'></span><span id='topic+getChipType+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Gets the chip type for this CEL file <em>according</em> to the <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelFile'
getChipType(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getChipType.AffymetrixCelFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>

<hr>
<h2 id='getChipType.AffymetrixCelSet'>Gets the chip type for this CEL set</h2><span id='topic+getChipType.AffymetrixCelSet'></span><span id='topic+AffymetrixCelSet.getChipType'></span><span id='topic+getChipType+2CAffymetrixCelSet-method'></span>

<h3>Description</h3>

<p>Gets the chip type for this CEL set.
The chip type is inferred from the current CDF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelSet'
getChipType(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getChipType.AffymetrixCelSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+getCdf.AffymetrixCelSet">*getCdf</a>()</code>.
For more information see <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>

<hr>
<h2 id='getChipType.CnagCfhFile'>Gets the chip type for this CEL file</h2><span id='topic+getChipType.CnagCfhFile'></span><span id='topic+CnagCfhFile.getChipType'></span><span id='topic+getChipType+2CCnagCfhFile-method'></span>

<h3>Description</h3>

<p>Gets the chip type for this CEL file <em>according</em> to the <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CnagCfhFile'
getChipType(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getChipType.CnagCfhFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+CnagCfhFile">CnagCfhFile</a></code>.
</p>

<hr>
<h2 id='getChipType.GenomeInformation'>Gets the chip type of this genome information set</h2><span id='topic+getChipType.GenomeInformation'></span><span id='topic+GenomeInformation.getChipType'></span><span id='topic+getChipType+2CGenomeInformation-method'></span>

<h3>Description</h3>

<p>Gets the chip type of this genome information set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GenomeInformation'
getChipType(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getChipType.GenomeInformation_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+GenomeInformation">GenomeInformation</a></code>.
</p>

<hr>
<h2 id='getChipType.SnpInformation'>Gets the chip type of this genome information set</h2><span id='topic+getChipType.SnpInformation'></span><span id='topic+SnpInformation.getChipType'></span><span id='topic+getChipType+2CSnpInformation-method'></span>

<h3>Description</h3>

<p>Gets the chip type of this genome information set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SnpInformation'
getChipType(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getChipType.SnpInformation_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+SnpInformation">SnpInformation</a></code>.
</p>

<hr>
<h2 id='getCnNames.AffymetrixCdfFile'>Gets the names of the CN units</h2><span id='topic+getCnNames.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.getCnNames'></span><span id='topic+getCnNames+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Gets the names of the CN units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
getCnNames(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCnNames.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+getSnpNames.AffymetrixCdfFile">*getSnpNames</a>()</code>.
Internally, <code><a href="#topic+getUnitTypes.AffymetrixCdfFile">*getUnitTypes</a>()</code>.
is used.
</p>
<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='getData.GenomeInformation'>Gets all or a subset of the genome information data</h2><span id='topic+getData.GenomeInformation'></span><span id='topic+GenomeInformation.getData'></span><span id='topic+getData+2CGenomeInformation-method'></span>

<h3>Description</h3>

<p>Gets all or a subset of the genome information data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GenomeInformation'
getData(this, units=NULL, fields=c("chromosome", "physicalPosition"), orderBy=NULL, ...,
  force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getData.GenomeInformation_+3A_units">units</code></td>
<td>
<p>The units for which the data should be returned.</p>
</td></tr>
<tr><td><code id="getData.GenomeInformation_+3A_fields">fields</code></td>
<td>
<p>The fields to be returned.</p>
</td></tr>
<tr><td><code id="getData.GenomeInformation_+3A_orderby">orderBy</code></td>
<td>
<p>The fields by which the returned data frame should be
ordered.</p>
</td></tr>
<tr><td><code id="getData.GenomeInformation_+3A_...">...</code></td>
<td>
<p>Named arguments used to select a subset of the units to be
returned.  Either a value to be compared to or a <code><a href="base.html#topic+function">function</a></code> returning
<code><a href="base.html#topic+logical">TRUE</a></code> or <code><a href="base.html#topic+logical">FALSE</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+data.frame">data.frame</a></code>, where the row names correspond to unit indices
as defined by the annotation unit names file.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getUnitIndices.GenomeInformation">*getUnitIndices</a>()</code>.
For more information see <code><a href="#topic+GenomeInformation">GenomeInformation</a></code>.
</p>

<hr>
<h2 id='getData.SnpInformation'>Gets all or a subset of the genome information data</h2><span id='topic+getData.SnpInformation'></span><span id='topic+SnpInformation.getData'></span><span id='topic+getData+2CSnpInformation-method'></span>

<h3>Description</h3>

<p>Gets all or a subset of the genome information data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SnpInformation'
getData(this, units=NULL, fields=c("fragmentLength", "start", "stop"), orderBy=NULL, ...,
  force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getData.SnpInformation_+3A_units">units</code></td>
<td>
<p>The units for which the data should be returned.</p>
</td></tr>
<tr><td><code id="getData.SnpInformation_+3A_fields">fields</code></td>
<td>
<p>The fields to be returned.</p>
</td></tr>
<tr><td><code id="getData.SnpInformation_+3A_orderby">orderBy</code></td>
<td>
<p>The fields by which the returned data frame should be
ordered.</p>
</td></tr>
<tr><td><code id="getData.SnpInformation_+3A_...">...</code></td>
<td>
<p>Named arguments used to select a subset of the units to be
returned.  Either a value to be compared to or a <code><a href="base.html#topic+function">function</a></code> returning
<code><a href="base.html#topic+logical">TRUE</a></code> or <code><a href="base.html#topic+logical">FALSE</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+data.frame">data.frame</a></code>, where the row names correspond to unit indices
as defined by the CDF.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+SnpInformation">SnpInformation</a></code>.
</p>

<hr>
<h2 id='getDataSet.AffymetrixCelSetReporter'>Gets the data set</h2><span id='topic+getDataSet.AffymetrixCelSetReporter'></span><span id='topic+AffymetrixCelSetReporter.getDataSet'></span><span id='topic+getDataSet+2CAffymetrixCelSetReporter-method'></span>

<h3>Description</h3>

<p>Gets the data set for which the reporter is generating output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelSetReporter'
getDataSet(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getDataSet.AffymetrixCelSetReporter_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelSetReporter">AffymetrixCelSetReporter</a></code>.
</p>

<hr>
<h2 id='getDataSet.ArrayExplorer'>Gets the data set</h2><span id='topic+getDataSet.ArrayExplorer'></span><span id='topic+ArrayExplorer.getDataSet'></span><span id='topic+getDataSet+2CArrayExplorer-method'></span>

<h3>Description</h3>

<p>Gets the data set for which the explorer is displaying it results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ArrayExplorer'
getDataSet(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getDataSet.ArrayExplorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ArrayExplorer">ArrayExplorer</a></code>.
</p>

<hr>
<h2 id='getDataSet.Model'>Gets the input data set for this model</h2><span id='topic+getDataSet.Model'></span><span id='topic+Model.getDataSet'></span><span id='topic+getDataSet+2CModel-method'></span>

<h3>Description</h3>

<p>Gets the input data set for this model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Model'
getDataSet(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getDataSet.Model_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+Model">Model</a></code>.
</p>

<hr>
<h2 id='getFirmaSet.FirmaModel'>Gets the set of FIRMA results for this model</h2><span id='topic+getFirmaSet.FirmaModel'></span><span id='topic+FirmaModel.getFirmaSet'></span><span id='topic+getFirmaSet+2CFirmaModel-method'></span><span id='topic+FirmaModel.getFirmaScores'></span><span id='topic+getFirmaScores.FirmaModel'></span><span id='topic+getFirmaScores+2CFirmaModel-method'></span>

<h3>Description</h3>

<p>Gets the set of FIRMA results for this model.
There is one chip-effect file per array.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FirmaModel'
getFirmaSet(this, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getFirmaSet.FirmaModel_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="getFirmaSet.FirmaModel_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="#topic+FirmaSet">FirmaSet</a></code> object.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+FirmaModel">FirmaModel</a></code>.
</p>

<hr>
<h2 id='getFitUnitGroupFunction.AffinePlm'>Gets the low-level function that fits the PLM</h2><span id='topic+getFitUnitGroupFunction.AffinePlm'></span><span id='topic+AffinePlm.getFitUnitGroupFunction'></span><span id='topic+getFitUnitGroupFunction+2CAffinePlm-method'></span>

<h3>Description</h3>

<p>Gets the low-level function that fits the PLM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffinePlm'
getFitUnitGroupFunction(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getFitUnitGroupFunction.AffinePlm_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+function">function</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="aroma.light.html#topic+calibrateMultiscan">calibrateMultiscan</a></code>.
For more information see <code><a href="#topic+AffinePlm">AffinePlm</a></code>.
</p>

<hr>
<h2 id='getFitUnitGroupFunction.AvgPlm'>Gets the low-level function that fits the PLM</h2><span id='topic+getFitUnitGroupFunction.AvgPlm'></span><span id='topic+AvgPlm.getFitUnitGroupFunction'></span><span id='topic+getFitUnitGroupFunction+2CAvgPlm-method'></span>

<h3>Description</h3>

<p>Gets the low-level function that fits the PLM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AvgPlm'
getFitUnitGroupFunction(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getFitUnitGroupFunction.AvgPlm_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+function">function</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AvgPlm">AvgPlm</a></code>.
</p>

<hr>
<h2 id='getFitUnitGroupFunction.ExonRmaPlm'>Gets the low-level function that fits the Exon PLM</h2><span id='topic+getFitUnitGroupFunction.ExonRmaPlm'></span><span id='topic+ExonRmaPlm.getFitUnitGroupFunction'></span><span id='topic+getFitUnitGroupFunction+2CExonRmaPlm-method'></span>

<h3>Description</h3>

<p>Gets the low-level function that fits the Exon PLM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ExonRmaPlm'
getFitUnitGroupFunction(this, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getFitUnitGroupFunction.ExonRmaPlm_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="getFitUnitGroupFunction.ExonRmaPlm_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+function">function</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson and Ken Simpson (WEHI) utilizing Ben Bolstad's
<span class="pkg">preprocessCore</span> package.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ExonRmaPlm">ExonRmaPlm</a></code>.
</p>

<hr>
<h2 id='getFitUnitGroupFunction.FirmaModel'>Static method to get the low-level function that fits the PLM</h2><span id='topic+getFitUnitGroupFunction.FirmaModel'></span><span id='topic+FirmaModel.getFitUnitGroupFunction'></span><span id='topic+getFitUnitGroupFunction+2CFirmaModel-method'></span>

<h3>Description</h3>

<p>Static method to get the low-level function that fits the PLM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FirmaModel'
getFitUnitGroupFunction(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getFitUnitGroupFunction.FirmaModel_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+function">function</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+FirmaModel">FirmaModel</a></code>.
</p>

<hr>
<h2 id='getFitUnitGroupFunction.HetLogAddPlm'>Gets the low-level function that fits the PLM</h2><span id='topic+getFitUnitGroupFunction.HetLogAddPlm'></span><span id='topic+HetLogAddPlm.getFitUnitGroupFunction'></span><span id='topic+getFitUnitGroupFunction+2CHetLogAddPlm-method'></span>

<h3>Description</h3>

<p>Gets the low-level function that fits the PLM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'HetLogAddPlm'
getFitUnitGroupFunction(this, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getFitUnitGroupFunction.HetLogAddPlm_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="getFitUnitGroupFunction.HetLogAddPlm_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+function">function</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson and Ken Simpson (WEHI) utilizing Ben Bolstad's
<span class="pkg">preprocessCore</span> package.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+HetLogAddPlm">HetLogAddPlm</a></code>.
</p>

<hr>
<h2 id='getFitUnitGroupFunction.MbeiPlm'>Gets the low-level function that fits the PLM</h2><span id='topic+getFitUnitGroupFunction.MbeiPlm'></span><span id='topic+MbeiPlm.getFitUnitGroupFunction'></span><span id='topic+getFitUnitGroupFunction+2CMbeiPlm-method'></span>

<h3>Description</h3>

<p>Gets the low-level function that fits the PLM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MbeiPlm'
getFitUnitGroupFunction(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getFitUnitGroupFunction.MbeiPlm_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+function">function</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="affy.html#topic+fit.li.wong">fit.li.wong</a></code>.
For more information see <code><a href="#topic+MbeiPlm">MbeiPlm</a></code>.
</p>

<hr>
<h2 id='getFitUnitGroupFunction.RmaPlm'>Gets the low-level function that fits the PLM</h2><span id='topic+getFitUnitGroupFunction.RmaPlm'></span><span id='topic+RmaPlm.getFitUnitGroupFunction'></span><span id='topic+getFitUnitGroupFunction+2CRmaPlm-method'></span>

<h3>Description</h3>

<p>Gets the low-level function that fits the PLM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'RmaPlm'
getFitUnitGroupFunction(this, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getFitUnitGroupFunction.RmaPlm_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="getFitUnitGroupFunction.RmaPlm_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+function">function</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson and Ken Simpson (WEHI) utilizing Ben Bolstad's
<span class="pkg">preprocessCore</span> package.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+RmaPlm">RmaPlm</a></code>.
</p>

<hr>
<h2 id='getFullName.Model'>Gets the full name of the output set</h2><span id='topic+getFullName.Model'></span><span id='topic+Model.getFullName'></span><span id='topic+getFullName+2CModel-method'></span>

<h3>Description</h3>

<p>Gets the full name of the output set, which consists of the name with appended
comma-separated tags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Model'
getFullName(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getFullName.Model_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+Model">Model</a></code>.
</p>

<hr>
<h2 id='getFullName.TransformReport'>Gets the full name of the output data set</h2><span id='topic+getFullName.TransformReport'></span><span id='topic+TransformReport.getFullName'></span><span id='topic+getFullName+2CTransformReport-method'></span>

<h3>Description</h3>

<p>Gets the full name of the output data set, which is the name with comma separated tags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'TransformReport'
getFullName(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getFullName.TransformReport_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+TransformReport">TransformReport</a></code>.
</p>

<hr>
<h2 id='getGenomeInformation.AffymetrixCdfFile'>Gets genome information for this chip type</h2><span id='topic+getGenomeInformation.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.getGenomeInformation'></span><span id='topic+getGenomeInformation+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Gets genome information for this chip type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
getGenomeInformation(this, types=c("UGP", "dChip"), ..., force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGenomeInformation.AffymetrixCdfFile_+3A_types">types</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying what type of genome
information sets to search for.</p>
</td></tr>
<tr><td><code id="getGenomeInformation.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="getGenomeInformation.AffymetrixCdfFile_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">FALSE</a></code>, cached information is retrieved, otherwise not.</p>
</td></tr>
<tr><td><code id="getGenomeInformation.AffymetrixCdfFile_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="#topic+GenomeInformation">GenomeInformation</a></code> object.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 
for (zzz in 0) {

# Define a CDF for a SNP chip
if (!exists("cdf")) {
  cdf &lt;- AffymetrixCdfFile$fromChipType(".*Sty.*")
}
print(cdf)

# Get the genome information
gi &lt;- getGenomeInformation(cdf)
print(gi)

# Get the units on chromosome X order position by default
# (First call will be slow be data is being cached)
units &lt;- getUnitIndices(gi, chromosome="X")
nunits &lt;- length(units)
cat(sprintf("Number of units on chromosome X: %d\n", nunits))
str(units)

# Plot the SNP density for this chromosome
plotDensity(gi, chromosome="X")

} # for (zzz in 0)
rm(zzz)


## End(Not run)</code></pre>

<hr>
<h2 id='getHeader.AffymetrixCdfFile'>Gets the header of the CDF file</h2><span id='topic+getHeader.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.getHeader'></span><span id='topic+getHeader+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Gets the header of the CDF file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
getHeader(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getHeader.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> structure as returned by <code><a href="affxparser.html#topic+readCdfHeader">readCdfHeader</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='getHeader.AffymetrixCelFile'>Gets the header of the CEL file</h2><span id='topic+getHeader.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.getHeader'></span><span id='topic+getHeader+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Gets the header of the CEL file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelFile'
getHeader(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getHeader.AffymetrixCelFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> structure as returned by <code><a href="affxparser.html#topic+readCelHeader">readCelHeader</a></code>.
If file does not exists, then <code><a href="base.html#topic+NULL">NULL</a></code> is returned.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>

<hr>
<h2 id='getHeader.AffymetrixPgfFile'>Gets the header of the PGF file</h2><span id='topic+getHeader.AffymetrixPgfFile'></span><span id='topic+AffymetrixPgfFile.getHeader'></span><span id='topic+getHeader+2CAffymetrixPgfFile-method'></span>

<h3>Description</h3>

<p>Gets the header of the PGF file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixPgfFile'
getHeader(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getHeader.AffymetrixPgfFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> structure as returned by <code><a href="affxparser.html#topic+readPgfHeader">readPgfHeader</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixPgfFile">AffymetrixPgfFile</a></code>.
</p>

<hr>
<h2 id='getHeader.AromaChipTypeAnnotationFile'>Gets the header of the annotation file</h2><span id='topic+getHeader.AromaChipTypeAnnotationFile'></span><span id='topic+AromaChipTypeAnnotationFile.getHeader'></span><span id='topic+getHeader+2CAromaChipTypeAnnotationFile-method'></span>

<h3>Description</h3>

<p>Gets the header of the annotation file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaChipTypeAnnotationFile'
getHeader(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getHeader.AromaChipTypeAnnotationFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> structure.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AromaChipTypeAnnotationFile">AromaChipTypeAnnotationFile</a></code>.
</p>

<hr>
<h2 id='getHeader.CnagCfhFile'>Gets the header of the CEL file</h2><span id='topic+getHeader.CnagCfhFile'></span><span id='topic+CnagCfhFile.getHeader'></span><span id='topic+getHeader+2CCnagCfhFile-method'></span>

<h3>Description</h3>

<p>Gets the header of the CEL file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CnagCfhFile'
getHeader(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getHeader.CnagCfhFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+list">list</a></code> structure as returned by <code><a href="affxparser.html#topic+readCelHeader">readCelHeader</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+CnagCfhFile">CnagCfhFile</a></code>.
</p>

<hr>
<h2 id='getImage.AffymetrixCelFile'>Creates an RGB Image object from a CEL file</h2><span id='topic+getImage.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.getImage'></span><span id='topic+getImage+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Creates an RGB Image object from a CEL file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelFile'
getImage(this, other=NULL, transforms=list(sqrt), xrange=c(0, Inf), yrange=xrange,
  zrange=c(0, sqrt(2^16)), field=c("intensities", "stdvs", "pixels"), zoom=1, ...,
  readRectFcn=NULL, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getImage.AffymetrixCelFile_+3A_other">other</code></td>
<td>
<p>An optional <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code> of the same chip type,
that is used for calculating the ratio (non-logged).  Note, to get
the log-ratios, the <code>log</code>() function has to be specified as
the first transform in the <code><a href="base.html#topic+list">list</a></code> of <code>transformations</code>.</p>
</td></tr>
<tr><td><code id="getImage.AffymetrixCelFile_+3A_xrange">xrange</code>, <code id="getImage.AffymetrixCelFile_+3A_yrange">yrange</code></td>
<td>
<p><code><a href="base.html#topic+vector">vector</a></code>s of length two specifying the
(x0,x1) and (y0,y1) regions to be extracted.  If <code><a href="base.html#topic+NULL">NULL</a></code>, the
complete regions is used.</p>
</td></tr>
<tr><td><code id="getImage.AffymetrixCelFile_+3A_field">field</code></td>
<td>
<p>One of the CEL file fields, i.e. <code>"intensities"</code>,
<code>stdvs</code>, or <code>pixels</code>.</p>
</td></tr>
<tr><td><code id="getImage.AffymetrixCelFile_+3A_zoom">zoom</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> scale factor in (0,+Inf) for resizing the
imaging. If <code>1</code>, no resizing is done.</p>
</td></tr>
<tr><td><code id="getImage.AffymetrixCelFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to
<code><a href="#topic+readRawDataRectangle.AffymetrixCelFile">*readRawDataRectangle</a>()</code> and more function.</p>
</td></tr>
<tr><td><code id="getImage.AffymetrixCelFile_+3A_readrectfcn">readRectFcn</code></td>
<td>
<p>A <code><a href="base.html#topic+function">function</a></code> taking arguments 'xrange' and 'yrange',
or <code><a href="base.html#topic+NULL">NULL</a></code> for the default read function.</p>
</td></tr>
<tr><td><code id="getImage.AffymetrixCelFile_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an Image object as defined by the EBImage package.
If <code>palette==NULL</code>, the color code is <code>Grayscale</code>, otherwise
<code>TrueColor</code>.
</p>


<h3>Author(s)</h3>

<p>Ken Simpson, Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>

<hr>
<h2 id='getInputDataSet.TransformReport'>Gets the source data set</h2><span id='topic+getInputDataSet.TransformReport'></span><span id='topic+TransformReport.getInputDataSet'></span><span id='topic+getInputDataSet+2CTransformReport-method'></span>

<h3>Description</h3>

<p>Gets the source data set that is to be (or has been) transformed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'TransformReport'
getInputDataSet(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getInputDataSet.TransformReport_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+TransformReport">TransformReport</a></code>.
</p>

<hr>
<h2 id='getIntensities.AffymetrixCelSet'>Gets cell intensities from a set of cells and a set of arrays</h2><span id='topic+getIntensities.AffymetrixCelSet'></span><span id='topic+AffymetrixCelSet.getIntensities'></span><span id='topic+getIntensities+2CAffymetrixCelSet-method'></span>

<h3>Description</h3>

<p>Gets cell intensities from a set of cells and a set of arrays.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelSet'
getIntensities(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getIntensities.AffymetrixCelSet_+3A_...">...</code></td>
<td>
<p>Passed to <code><a href="#topic+getData.AffymetrixCelSet">*getData</a>()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+numeric">numeric</a></code> <code class="reqn">NxK</code> matrix, where <code class="reqn">N</code> is the number of
cells read, and <code class="reqn">K</code> is the number of arrays in the data set.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>

<hr>
<h2 id='getName.GenericReporter'>Gets the name of the explorer</h2><span id='topic+getName.GenericReporter'></span><span id='topic+GenericReporter.getName'></span><span id='topic+getName+2CGenericReporter-method'></span>

<h3>Description</h3>

<p>Gets the name of the explorer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GenericReporter'
getName(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getName.GenericReporter_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If a name alias has not been set explicitly, the input name will be used.
</p>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+GenericReporter">GenericReporter</a></code>.
</p>

<hr>
<h2 id='getName.Model'>Gets the name of the output data set</h2><span id='topic+getName.Model'></span><span id='topic+Model.getName'></span><span id='topic+getName+2CModel-method'></span>

<h3>Description</h3>

<p>Gets the name of the output data set, which is the same as the name of the input data set, unless
an alias is set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Model'
getName(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getName.Model_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getAlias.Model">*getAlias</a>()</code>
For more information see <code><a href="#topic+Model">Model</a></code>.
</p>

<hr>
<h2 id='getName.TransformReport'>Gets the name of the output data set</h2><span id='topic+getName.TransformReport'></span><span id='topic+TransformReport.getName'></span><span id='topic+getName+2CTransformReport-method'></span>

<h3>Description</h3>

<p>Gets the name of the output data set, which is the same as the input data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'TransformReport'
getName(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getName.TransformReport_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+TransformReport">TransformReport</a></code>.
</p>

<hr>
<h2 id='getOutputDataSet.Transform'>Gets the transformed data set</h2><span id='topic+getOutputDataSet.Transform'></span><span id='topic+Transform.getOutputDataSet'></span><span id='topic+getOutputDataSet+2CTransform-method'></span>

<h3>Description</h3>

<p>Gets the transformed data set, if processed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Transform'
getOutputDataSet(this, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getOutputDataSet.Transform_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="getOutputDataSet.Transform_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="aroma.core.html#topic+AromaMicroarrayDataSet">AromaMicroarrayDataSet</a></code> or <code><a href="base.html#topic+NULL">NULL</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+Transform">Transform</a></code>.
</p>

<hr>
<h2 id='getOutputDataSet.TransformReport'>Gets the transformed data set</h2><span id='topic+getOutputDataSet.TransformReport'></span><span id='topic+TransformReport.getOutputDataSet'></span><span id='topic+getOutputDataSet+2CTransformReport-method'></span>

<h3>Description</h3>

<p>Gets the transformed data set, if processed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'TransformReport'
getOutputDataSet(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getOutputDataSet.TransformReport_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+TransformReport">TransformReport</a></code>.
</p>

<hr>
<h2 id='getPath.Model'>Gets the path of this model</h2><span id='topic+getPath.Model'></span><span id='topic+Model.getPath'></span><span id='topic+getPath+2CModel-method'></span>

<h3>Description</h3>

<p>Gets the path of this model where the parameter files are located.
The directory is created, if missing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Model'
getPath(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getPath.Model_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the path does not exist, it is created.
</p>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+Model">Model</a></code>.
</p>

<hr>
<h2 id='getPath.TransformReport'>Gets the path of the output data set</h2><span id='topic+getPath.TransformReport'></span><span id='topic+TransformReport.getPath'></span><span id='topic+getPath+2CTransformReport-method'></span>

<h3>Description</h3>

<p>Gets the path of the output data set.
If non-existing, then the directory is created.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'TransformReport'
getPath(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getPath.TransformReport_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+TransformReport">TransformReport</a></code>.
</p>

<hr>
<h2 id='getPositions.GenomeInformation'>Gets the physical positions for a set of units</h2><span id='topic+getPositions.GenomeInformation'></span><span id='topic+GenomeInformation.getPositions'></span><span id='topic+getPositions+2CGenomeInformation-method'></span>

<h3>Description</h3>

<p>Gets the physical positions for a set of units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GenomeInformation'
getPositions(this, ..., na.rm=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getPositions.GenomeInformation_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+getData.GenomeInformation">*getData</a>()</code>.</p>
</td></tr>
<tr><td><code id="getPositions.GenomeInformation_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, non-defined unit indices are excluded, otherwise
not.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getData.GenomeInformation">*getData</a>()</code>.
<code><a href="#topic+getUnitIndices.GenomeInformation">*getUnitIndices</a>()</code>.
For more information see <code><a href="#topic+GenomeInformation">GenomeInformation</a></code>.
</p>

<hr>
<h2 id='getProbeAffinityFile.ProbeLevelModel'>Gets the probe affinities for this model</h2><span id='topic+getProbeAffinityFile.ProbeLevelModel'></span><span id='topic+ProbeLevelModel.getProbeAffinityFile'></span><span id='topic+getProbeAffinityFile+2CProbeLevelModel-method'></span><span id='topic+ProbeLevelModel.getProbeAffinities'></span><span id='topic+getProbeAffinities.ProbeLevelModel'></span><span id='topic+getProbeAffinities+2CProbeLevelModel-method'></span>

<h3>Description</h3>

<p>Gets the probe affinities for this model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ProbeLevelModel'
getProbeAffinityFile(this, ..., .class=ProbeAffinityFile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getProbeAffinityFile.ProbeLevelModel_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="getProbeAffinityFile.ProbeLevelModel_+3A_.class">.class</code></td>
<td>
<p>A <code><a href="#topic+ProbeAffinityFile">ProbeAffinityFile</a></code> <em>class</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="#topic+ProbeAffinityFile">ProbeAffinityFile</a></code> object.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code>.
</p>

<hr>
<h2 id='getResiduals.QualityAssessmentModel'>Calculates the residuals from a probe-level model</h2><span id='topic+getResiduals.QualityAssessmentModel'></span><span id='topic+QualityAssessmentModel.getResiduals'></span><span id='topic+getResiduals+2CQualityAssessmentModel-method'></span>

<h3>Description</h3>

<p>Calculates the residuals from a probe-level model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'QualityAssessmentModel'
getResiduals(this, units=NULL, path=NULL, name="qcData", tags="*", ram=NULL, force=FALSE,
  verbose=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getResiduals.QualityAssessmentModel_+3A_...">...</code></td>
<td>
<p>Additional arguments passed <code>byPath()</code> of
<code><a href="#topic+QualityAssessmentSet">QualityAssessmentSet</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+QualityAssessmentSet">QualityAssessmentSet</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+QualityAssessmentModel">QualityAssessmentModel</a></code>.
</p>

<hr>
<h2 id='getRootPath.Model'>Gets the root path of this model</h2><span id='topic+getRootPath.Model'></span><span id='topic+Model.getRootPath'></span><span id='topic+getRootPath+2CModel-method'></span>

<h3>Description</h3>

<p>Gets the root path of this model.
By default, this is the string <code>"model"</code> appended by the capitalized
name of the model class, e.g. <code>"modelUnit"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Model'
getRootPath(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getRootPath.Model_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> string.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+Model">Model</a></code>.
</p>

<hr>
<h2 id='getSnpNames.AffymetrixCdfFile'>Gets the names of the SNP units</h2><span id='topic+getSnpNames.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.getSnpNames'></span><span id='topic+getSnpNames+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Gets the names of the SNP units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
getSnpNames(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSnpNames.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+getCnNames.AffymetrixCdfFile">*getCnNames</a>()</code>.
Internally, <code><a href="#topic+getUnitTypes.AffymetrixCdfFile">*getUnitTypes</a>()</code>.
is used.
</p>
<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='getTags.GenericReporter'>Gets the tags of the reporter</h2><span id='topic+getTags.GenericReporter'></span><span id='topic+GenericReporter.getTags'></span><span id='topic+getTags+2CGenericReporter-method'></span>

<h3>Description</h3>

<p>Gets the tags of the reporter, which are the input tags plus additional tags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GenericReporter'
getTags(this, collapse=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getTags.GenericReporter_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+GenericReporter">GenericReporter</a></code>.
</p>

<hr>
<h2 id='getTags.Model'>Gets the tags of the output data set</h2><span id='topic+getTags.Model'></span><span id='topic+Model.getTags'></span><span id='topic+getTags+2CModel-method'></span>

<h3>Description</h3>

<p>Gets the tags of the output data set, which consists of the tags of the input data set followed
by an additional set of tags added by the model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Model'
getTags(this, collapse=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getTags.Model_+3A_collapse">collapse</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string used to concatenate the tags.
If <code><a href="base.html#topic+NULL">NULL</a></code>, the tags are not concatenated.</p>
</td></tr>
<tr><td><code id="getTags.Model_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+setTags.Model">*setTags</a>()</code>
For more information see <code><a href="#topic+Model">Model</a></code>.
</p>

<hr>
<h2 id='getTags.TransformReport'>Gets the tags of the output data set</h2><span id='topic+getTags.TransformReport'></span><span id='topic+TransformReport.getTags'></span><span id='topic+getTags+2CTransformReport-method'></span>

<h3>Description</h3>

<p>Gets the tags of the output data set, which equals the tags of the input data set plus the tags
of this transformation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'TransformReport'
getTags(this, collapse=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getTags.TransformReport_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+TransformReport">TransformReport</a></code>.
</p>

<hr>
<h2 id='getTimestamp.AffymetrixCelFile'>Gets the timestamp in the CEL header</h2><span id='topic+getTimestamp.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.getTimestamp'></span><span id='topic+getTimestamp+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Gets the timestamp in the CEL header.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelFile'
getTimestamp(this, format="%m/%d/%y %H:%M:%S", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getTimestamp.AffymetrixCelFile_+3A_format">format</code></td>
<td>
<p>The default format string for parsing the time stamp.</p>
</td></tr>
<tr><td><code id="getTimestamp.AffymetrixCelFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code>POSIXct</code> object.
The parsed string containing the timestamp is returned as
attribute <code>text</code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>Internally, <code><a href="base.html#topic+strptime">strptime</a></code>() is used to parse the time stamp.
<code><a href="base.html#topic+DateTimeClasses">DateTimeClasses</a></code> for more information on <code>POSIXct</code>.
For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>

<hr>
<h2 id='getUnitGroupCellMap.ChipEffectFile'>Gets a (unit, group, cell) index map</h2><span id='topic+getUnitGroupCellMap.ChipEffectFile'></span><span id='topic+ChipEffectFile.getUnitGroupCellMap'></span><span id='topic+getUnitGroupCellMap+2CChipEffectFile-method'></span><span id='topic+ChipEffectFile.getCellMap'></span><span id='topic+getCellMap.ChipEffectFile'></span><span id='topic+getCellMap+2CChipEffectFile-method'></span>

<h3>Description</h3>

<p>Gets a (unit, group, cell) index map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ChipEffectFile'
getUnitGroupCellMap(this, units=NULL, force=FALSE, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getUnitGroupCellMap.ChipEffectFile_+3A_units">units</code></td>
<td>
<p>The units for which the map should be returned.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are considered.</p>
</td></tr>
<tr><td><code id="getUnitGroupCellMap.ChipEffectFile_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, cached cell indices are ignored.</p>
</td></tr>
<tr><td><code id="getUnitGroupCellMap.ChipEffectFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="getUnitGroupCellMap.ChipEffectFile_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+data.frame">data.frame</a></code> with <code><a href="base.html#topic+integer">integer</a></code> columns <code>unit</code>, <code>group</code>,
and <code>cell</code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ChipEffectFile">ChipEffectFile</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
     unit group cell
   1  104     1  335
   2  104     2  336
   3  105     1  337
   4  105     2  338
   5  105     3  339
   6  105     4  340

## End(Not run)</code></pre>

<hr>
<h2 id='getUnitIndices.GenomeInformation'>Gets unit indices ordered along the chromosome</h2><span id='topic+getUnitIndices.GenomeInformation'></span><span id='topic+GenomeInformation.getUnitIndices'></span><span id='topic+getUnitIndices+2CGenomeInformation-method'></span>

<h3>Description</h3>

<p>Gets unit indices ordered along the chromosome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GenomeInformation'
getUnitIndices(this, ..., na.rm=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getUnitIndices.GenomeInformation_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+getData.GenomeInformation">*getData</a>()</code>.</p>
</td></tr>
<tr><td><code id="getUnitIndices.GenomeInformation_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, non-defined unit indices are excluded, otherwise
not.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getData.GenomeInformation">*getData</a>()</code>.
For more information see <code><a href="#topic+GenomeInformation">GenomeInformation</a></code>.
</p>

<hr>
<h2 id='getUnitIntensities.AffymetrixCelSet'>Gets cell signals for a subset of units and a subset of arrays</h2><span id='topic+getUnitIntensities.AffymetrixCelSet'></span><span id='topic+AffymetrixCelSet.getUnitIntensities'></span><span id='topic+getUnitIntensities+2CAffymetrixCelSet-method'></span>

<h3>Description</h3>

<p>Gets cell signals for a subset of units and a subset of arrays.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelSet'
getUnitIntensities(this, units=NULL, ..., force=FALSE, cache=!is.null(units),
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getUnitIntensities.AffymetrixCelSet_+3A_units">units</code></td>
<td>
<p>An <code><a href="base.html#topic+integer">integer</a></code> index <code><a href="base.html#topic+vector">vector</a></code> specifying units to be read.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are read.</p>
</td></tr>
<tr><td><code id="getUnitIntensities.AffymetrixCelSet_+3A_...">...</code></td>
<td>
<p>Arguments passed to the low-level function for reading
CEL units, i.e. <code><a href="affxparser.html#topic+readCelUnits">readCelUnits</a></code>.
If <code>units</code> is not already a CDF <code><a href="base.html#topic+list">list</a></code> structure, these arguments
are also passed to <code>readUnits()</code> of <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.</p>
</td></tr>
<tr><td><code id="getUnitIntensities.AffymetrixCelSet_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, cached values are ignored.</p>
</td></tr>
<tr><td><code id="getUnitIntensities.AffymetrixCelSet_+3A_verbose">verbose</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, progress details are printed, otherwise not.
May also be a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named <code><a href="base.html#topic+list">list</a></code> structure.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>

<hr>
<h2 id='getUnitNames.AffymetrixCdfFile'>Gets the names of each unit</h2><span id='topic+getUnitNames.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.getUnitNames'></span><span id='topic+getUnitNames+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Gets the names of each unit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
getUnitNames(this, units=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getUnitNames.AffymetrixCdfFile_+3A_units">units</code></td>
<td>
<p>The units of interest. If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are considered.</p>
</td></tr>
<tr><td><code id="getUnitNames.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Once read from file, this information is cached in memory for efficiency.
The cache can be cleared by calling <code>gc(cdf)</code>.
</p>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+vector">vector</a></code> of <code><a href="base.html#topic+character">character</a></code> names.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='getUnitNames.AffymetrixPgfFile'>Gets the names of each unit</h2><span id='topic+getUnitNames.AffymetrixPgfFile'></span><span id='topic+AffymetrixPgfFile.getUnitNames'></span><span id='topic+getUnitNames+2CAffymetrixPgfFile-method'></span>

<h3>Description</h3>

<p>Gets the names of each unit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixPgfFile'
getUnitNames(this, units=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getUnitNames.AffymetrixPgfFile_+3A_units">units</code></td>
<td>
<p>The units of interest. If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are considered.</p>
</td></tr>
<tr><td><code id="getUnitNames.AffymetrixPgfFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Once read from file, this information is cached in memory for efficiency.
The cache can be cleared by calling <code>gc(pgf)</code>.
</p>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+vector">vector</a></code> of <code><a href="base.html#topic+character">character</a></code> names.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixPgfFile">AffymetrixPgfFile</a></code>.
</p>

<hr>
<h2 id='getUnitTypes.AffymetrixCdfFile'>Gets the types of a set of units</h2><span id='topic+getUnitTypes.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.getUnitTypes'></span><span id='topic+getUnitTypes+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Gets the types of a set of units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
getUnitTypes(this, units=NULL, ..., force=FALSE, .cache=TRUE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getUnitTypes.AffymetrixCdfFile_+3A_units">units</code></td>
<td>
<p>The units of interest. If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are considered.</p>
</td></tr>
<tr><td><code id="getUnitTypes.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Once read from file, this information is cached in memory for efficiency.
The cache can be cleared by calling <code>gc(cdf)</code>.
</p>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+vector">vector</a></code> of <code><a href="base.html#topic+integer">integer</a></code>s.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='getWeights.QualityAssessmentModel'>Calculates the weights from the robust fit to a probe-level model</h2><span id='topic+getWeights.QualityAssessmentModel'></span><span id='topic+QualityAssessmentModel.getWeights'></span><span id='topic+getWeights+2CQualityAssessmentModel-method'></span>

<h3>Description</h3>

<p>Calculates the weights from the robust fit to a probe-level model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'QualityAssessmentModel'
getWeights(this, path=NULL, name="qcData", tags="*", ..., ram=NULL, force=FALSE,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getWeights.QualityAssessmentModel_+3A_path">path</code></td>
<td>
</td></tr>
<tr><td><code id="getWeights.QualityAssessmentModel_+3A_name">name</code></td>
<td>
</td></tr>
<tr><td><code id="getWeights.QualityAssessmentModel_+3A_tags">tags</code></td>
<td>
</td></tr>
<tr><td><code id="getWeights.QualityAssessmentModel_+3A_...">...</code></td>
<td>
</td></tr>
<tr><td><code id="getWeights.QualityAssessmentModel_+3A_ram">ram</code></td>
<td>
</td></tr>
<tr><td><code id="getWeights.QualityAssessmentModel_+3A_force">force</code></td>
<td>
</td></tr>
<tr><td><code id="getWeights.QualityAssessmentModel_+3A_verbose">verbose</code></td>
<td>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+QualityAssessmentSet">QualityAssessmentSet</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+QualityAssessmentModel">QualityAssessmentModel</a></code>.
</p>

<hr>
<h2 id='getXAM.ChipEffectFile'>Gets the physical position, log-intensities and log-ratios of chip effects for two arrays</h2><span id='topic+getXAM.ChipEffectFile'></span><span id='topic+ChipEffectFile.getXAM'></span><span id='topic+getXAM+2CChipEffectFile-method'></span>

<h3>Description</h3>

<p>Gets the physical position, log-intensities and log-ratios of chip effects for two arrays of units on a certain chromosome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ChipEffectFile'
getXAM(this, other, chromosome, units=NULL, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getXAM.ChipEffectFile_+3A_other">other</code></td>
<td>
<p>The second <code><a href="#topic+ChipEffectFile">ChipEffectFile</a></code> object used as the
reference.</p>
</td></tr>
<tr><td><code id="getXAM.ChipEffectFile_+3A_chromosome">chromosome</code></td>
<td>
<p>(The chromosome for which results should be returned.</p>
</td></tr>
<tr><td><code id="getXAM.ChipEffectFile_+3A_units">units</code></td>
<td>
<p>(The subset of units to be matched.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are considered.</p>
</td></tr>
<tr><td><code id="getXAM.ChipEffectFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="getXAM.ChipEffectFile_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a Nx3 matrix where N is the number of units returned.
The names of the columns are X (physical position in a given chromosome),
A (log-intensities) and M (log-ratios).
The names of the rows are the unit indices (as indexed by the CDF).
<em>Note: The rows are ordered according to chromosomal position.</em>
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+getAM.ChipEffectFile">*getAM</a>()</code>.
For more information see <code><a href="#topic+ChipEffectFile">ChipEffectFile</a></code>.
</p>

<hr>
<h2 id='getXAM.ChipEffectSet'>Gets the physical position, log-intensities and log-ratios of chip effects for two arrays</h2><span id='topic+getXAM.ChipEffectSet'></span><span id='topic+ChipEffectSet.getXAM'></span><span id='topic+getXAM+2CChipEffectSet-method'></span>

<h3>Description</h3>

<p>Gets the physical position, log-intensities and log-ratios of chip effects for two arrays of units on a certain chromosome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ChipEffectSet'
getXAM(this, other, chromosome, units=NULL, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getXAM.ChipEffectSet_+3A_other">other</code></td>
<td>
<p>The second <code><a href="#topic+ChipEffectFile">ChipEffectFile</a></code> object used as the
reference.</p>
</td></tr>
<tr><td><code id="getXAM.ChipEffectSet_+3A_chromosome">chromosome</code></td>
<td>
<p>(The chromosome for which results should be returned.</p>
</td></tr>
<tr><td><code id="getXAM.ChipEffectSet_+3A_units">units</code></td>
<td>
<p>(The subset of units to be matched.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are considered.</p>
</td></tr>
<tr><td><code id="getXAM.ChipEffectSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="getXAM.ChipEffectSet_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a Nx3 matrix where N is the number of units returned.
The names of the columns are X (physical position in a given chromosome),
A (log-intensities) and M (log-ratios).
The names of the rows are the unit indices (as indexed by the CDF).
<em>Note: The rows are ordered according to chromosomal position.</em>
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+getAM.ChipEffectSet">*getAM</a>()</code>.
For more information see <code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code>.
</p>

<hr>
<h2 id='groupUnitsByDimension.AffymetrixCdfFile'>Groups units by dimensions</h2><span id='topic+groupUnitsByDimension.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.groupUnitsByDimension'></span><span id='topic+groupUnitsByDimension+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Groups units by dimensions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
groupUnitsByDimension(this, units=NULL, ..., sort=TRUE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="groupUnitsByDimension.AffymetrixCdfFile_+3A_units">units</code></td>
<td>
<p>An optional <code><a href="base.html#topic+integer">integer</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying the units to be
queried.</p>
</td></tr>
<tr><td><code id="groupUnitsByDimension.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="groupUnitsByDimension.AffymetrixCdfFile_+3A_sort">sort</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the sets are ordered by number of groups per
units and then by the number of cells per group.</p>
</td></tr>
<tr><td><code id="groupUnitsByDimension.AffymetrixCdfFile_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named <code><a href="base.html#topic+list">list</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='HetLogAddCnPlm'>The HetLogAddCnPlm class</h2><span id='topic+HetLogAddCnPlm'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class HetLogAddCnPlm</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+RmaPlm">RmaPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+HetLogAddPlm">HetLogAddPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+HetLogAddSnpPlm">HetLogAddSnpPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+SnpPlm">SnpPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CnPlm">CnPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>HetLogAddCnPlm</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>HetLogAddCnPlm</b><br />
extends <a href="#topic+CnPlm">CnPlm</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HetLogAddCnPlm(..., combineAlleles=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HetLogAddCnPlm_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+HetLogAddCnPlm">HetLogAddCnPlm</a></code>.</p>
</td></tr>
<tr><td><code id="HetLogAddCnPlm_+3A_combinealleles">combineAlleles</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">FALSE</a></code>, allele A and allele B are treated
separately, otherwise together.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from CnPlm</b>:<br />
getCellIndices, getChipEffectSet, getCombineAlleles, getParameters, getProbeAffinityFile, setCombineAlleles
</p>
<p><b>Methods inherited from SnpPlm</b>:<br />
getCellIndices, getChipEffectSet, getMergeStrands, getParameters, getProbeAffinityFile, setMergeStrands
</p>
<p><b>Methods inherited from HetLogAddSnpPlm</b>:<br />
getAsteriskTags
</p>
<p><b>Methods inherited from HetLogAddPlm</b>:<br />
getAsteriskTags, getFitUnitGroupFunction
</p>
<p><b>Methods inherited from RmaPlm</b>:<br />
getAsteriskTags, getCalculateResidualsFunction, getParameters, getRlmFitFunctions
</p>
<p><b>Methods inherited from ProbeLevelModel</b>:<br />
calculateResidualSet, calculateWeights, fit, getAsteriskTags, getCalculateResidualsFunction, getChipEffectSet, getProbeAffinityFile, getResidualSet, getRootPath, getWeightsSet
</p>
<p><b>Methods inherited from MultiArrayUnitModel</b>:<br />
getListOfPriors, setListOfPriors, validate
</p>
<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Model</h3>

<p>TO DO.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='HetLogAddPlm'>The HetLogAddPlm class</h2><span id='topic+HetLogAddPlm'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class HetLogAddPlm</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+RmaPlm">RmaPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>HetLogAddPlm</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+HetLogAddCnPlm">HetLogAddCnPlm</a></em>, <em><a href="#topic+HetLogAddSnpPlm">HetLogAddSnpPlm</a></em><br />
</p>
<p>public abstract static class <b>HetLogAddPlm</b><br />
extends <em><a href="#topic+RmaPlm">RmaPlm</a></em><br />
</p>
<p>This class represents a log-additive model similar to the one described
in Irizarry et al (2003), except that the errors may have different
variances for different probes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HetLogAddPlm(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HetLogAddPlm_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+RmaPlm">RmaPlm</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from RmaPlm</b>:<br />
getAsteriskTags, getCalculateResidualsFunction, getParameters, getRlmFitFunctions
</p>
<p><b>Methods inherited from ProbeLevelModel</b>:<br />
calculateResidualSet, calculateWeights, fit, getAsteriskTags, getCalculateResidualsFunction, getChipEffectSet, getProbeAffinityFile, getResidualSet, getRootPath, getWeightsSet
</p>
<p><b>Methods inherited from MultiArrayUnitModel</b>:<br />
getListOfPriors, setListOfPriors, validate
</p>
<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+RmaPlm">RmaPlm</a></code>.
</p>

<hr>
<h2 id='HetLogAddSnpPlm'>The HetLogAddSnpPlm class</h2><span id='topic+HetLogAddSnpPlm'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class HetLogAddSnpPlm</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+RmaPlm">RmaPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+HetLogAddPlm">HetLogAddPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+SnpPlm">SnpPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>HetLogAddSnpPlm</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+HetLogAddCnPlm">HetLogAddCnPlm</a></em><br />
</p>
<p>public abstract static class <b>HetLogAddSnpPlm</b><br />
extends <a href="#topic+SnpPlm">SnpPlm</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HetLogAddSnpPlm(..., mergeStrands=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HetLogAddSnpPlm_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+HetLogAddPlm">HetLogAddPlm</a></code>.</p>
</td></tr>
<tr><td><code id="HetLogAddSnpPlm_+3A_mergestrands">mergeStrands</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the sense and the anti-sense strands are
fitted together, otherwise separately.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from SnpPlm</b>:<br />
getCellIndices, getChipEffectSet, getMergeStrands, getParameters, getProbeAffinityFile, setMergeStrands
</p>
<p><b>Methods inherited from HetLogAddPlm</b>:<br />
getAsteriskTags, getFitUnitGroupFunction
</p>
<p><b>Methods inherited from RmaPlm</b>:<br />
getAsteriskTags, getCalculateResidualsFunction, getParameters, getRlmFitFunctions
</p>
<p><b>Methods inherited from ProbeLevelModel</b>:<br />
calculateResidualSet, calculateWeights, fit, getAsteriskTags, getCalculateResidualsFunction, getChipEffectSet, getProbeAffinityFile, getResidualSet, getRootPath, getWeightsSet
</p>
<p><b>Methods inherited from MultiArrayUnitModel</b>:<br />
getListOfPriors, setListOfPriors, validate
</p>
<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='image270.AffymetrixCelFile'>Displays all or a subset of the data spatially</h2><span id='topic+image270.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.image270'></span><span id='topic+image270+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Displays all or a subset of the data spatially.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelFile'
image270(this, xrange=c(0, Inf), yrange=c(0, Inf), takeLog=TRUE, interleaved=FALSE, ...,
  field=c("intensities", "stdvs", "pixels"), col=gray.colors(256), main=getName(this))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="image270.AffymetrixCelFile_+3A_xrange">xrange</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length two giving the left and right
coordinates of the cells to be returned.</p>
</td></tr>
<tr><td><code id="image270.AffymetrixCelFile_+3A_yrange">yrange</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length two giving the top and bottom
coordinates of the cells to be returned.</p>
</td></tr>
<tr><td><code id="image270.AffymetrixCelFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed <code><a href="graphics.html#topic+image">image</a></code> and [...].</p>
</td></tr>
<tr><td><code id="image270.AffymetrixCelFile_+3A_field">field</code></td>
<td>
<p>The data field to be displayed.</p>
</td></tr>
<tr><td><code id="image270.AffymetrixCelFile_+3A_col">col</code></td>
<td>
<p>The color map to be used.</p>
</td></tr>
<tr><td><code id="image270.AffymetrixCelFile_+3A_main">main</code></td>
<td>
<p>The main title of the plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the (270-degrees rotated) data <code><a href="base.html#topic+matrix">matrix</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+updateUnits.AffymetrixCelFile">*updateUnits</a>()</code>.
For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>

<hr>
<h2 id='isDuplicated.AffymetrixCelSet'>Identifies duplicated CEL files</h2><span id='topic+isDuplicated.AffymetrixCelSet'></span><span id='topic+AffymetrixCelSet.isDuplicated'></span><span id='topic+isDuplicated+2CAffymetrixCelSet-method'></span>

<h3>Description</h3>

<p>Identifies duplicated CEL files by comparing the timestamps in the CEL headers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelSet'
isDuplicated(this, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="isDuplicated.AffymetrixCelSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="isDuplicated.AffymetrixCelSet_+3A_verbose">verbose</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, progress details are printed, otherwise not.
May also be a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+logical">logical</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length equal to the number of files
in the set.
An element with value <code><a href="base.html#topic+logical">TRUE</a></code> indicates that the corresponding CEL file
has the same time stamp as another preceeding CEL file.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>Internally <code><a href="base.html#topic+duplicated">duplicated</a></code>() is used to compare timestamps.
For more information see <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>

<hr>
<h2 id='isDuplicated.CnagCfhSet'>Identifies duplicated CFH files</h2><span id='topic+isDuplicated.CnagCfhSet'></span><span id='topic+CnagCfhSet.isDuplicated'></span><span id='topic+isDuplicated+2CCnagCfhSet-method'></span>

<h3>Description</h3>

<p>Identifies duplicated CFH files by comparing the timestamps in the CFH headers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CnagCfhSet'
isDuplicated(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="isDuplicated.CnagCfhSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+logical">logical</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length equal to the number of files
in the set.
An element with value <code><a href="base.html#topic+logical">TRUE</a></code> indicates that the corresponding CFH file
has the same time stamp as another preceeding CFH file.
</p>


<h3>See Also</h3>

<p>Internally <code><a href="base.html#topic+duplicated">duplicated</a></code>() is used to compare timestamps.
For more information see <code><a href="#topic+CnagCfhSet">CnagCfhSet</a></code>.
</p>

<hr>
<h2 id='isPm.AffymetrixCdfFile'>Checks which cells (probes) are PMs and not</h2><span id='topic+isPm.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.isPm'></span><span id='topic+isPm+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Checks which cells (probes) are PMs and not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
isPm(this, units=NULL, force=FALSE, cache=TRUE, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="isPm.AffymetrixCdfFile_+3A_units">units</code></td>
<td>
<p>The units to be read. If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are read.</p>
</td></tr>
<tr><td><code id="isPm.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="affxparser.html#topic+readCdfUnits">readCdfUnits</a></code>.</p>
</td></tr>
<tr><td><code id="isPm.AffymetrixCdfFile_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, cached results are ignored.</p>
</td></tr>
<tr><td><code id="isPm.AffymetrixCdfFile_+3A_cache">cache</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, results are cached.</p>
</td></tr>
<tr><td><code id="isPm.AffymetrixCdfFile_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+logical">logical</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length K, where K equals the total number
of cells in the requested units.  Note that the cells are ordered as
they occur in the units, that is, <em>not</em> in incremental order.
</p>


<h3>Caching</h3>

<p>This method caches a <code><a href="base.html#topic+logical">logical</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length N, when N equals the
number of cells on the array. The size of this vector is approximately
4*N bytes.  The vector indicates if a cell is a PM or not.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='isResequenceChip.AffymetrixCdfFile'>Static method to check if a CDF is for a resequencing (CustomSeq) chip</h2><span id='topic+isResequenceChip.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.isResequenceChip'></span><span id='topic+isResequenceChip+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Static method to check if a CDF is for a resequencing (CustomSeq) chip.  Note, this method is not bullet proof.  Several
resequencing CDF does not carry that information.  For such, we add
tests based on their chip type, as we become aware of them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
isResequenceChip(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="isResequenceChip.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code><a href="base.html#topic+logical">TRUE</a></code> if the chip type refers to a resequence array,
otherwise <code><a href="base.html#topic+logical">FALSE</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='isSnpChip.AffymetrixCdfFile'>Static method to check if a chip is a mapping (SNP) chip</h2><span id='topic+isSnpChip.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.isSnpChip'></span><span id='topic+isSnpChip+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Static method to check if a chip is a mapping (SNP) chip.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
isSnpChip(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="isSnpChip.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code><a href="base.html#topic+logical">TRUE</a></code> if the chip type refers to a SNP array, otherwise <code><a href="base.html#topic+logical">FALSE</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='justRMA'>Robust Multichip Analysis (RMA) reproducing the affy package</h2><span id='topic+justRMA'></span><span id='topic+justRMA.AffymetrixCelSet'></span><span id='topic+justRMA.default'></span>

<h3>Description</h3>

<p>Robust Multichip Analysis (RMA) reproducing the affy package as far as possible.  The RMA method is described in [1].
</p>
<p>The algorithm is processed in bounded memory, meaning a very large
number of arrays can be analyzed on also very limited computer systems.
The only limitation is the amount of memory required to load the final
chip-effect estimates into memory (as a <code><a href="Biobase.html#topic+ExpressionSet">ExpressionSet</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  ## S3 method for class 'AffymetrixCelSet'
justRMA(csR, flavor=c("oligo", "affyPLM"), ..., verbose=FALSE)
  ## Default S3 method:
justRMA(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="justRMA_+3A_csr">csR</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.</p>
</td></tr>
<tr><td><code id="justRMA_+3A_flavor">flavor</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying the estimators used in
the RMA summarization step.</p>
</td></tr>
<tr><td><code id="justRMA_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+doRMA">doRMA</a></code>() used internally.</p>
</td></tr>
<tr><td><code id="justRMA_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an annotated <code><a href="Biobase.html#topic+ExpressionSet">ExpressionSet</a></code>.
</p>


<h3>Reproducibility of affy</h3>

<p>This implementation of the RMA method reproduces <code><a href="affy.html#topic+justRMA">justRMA</a></code>
in <span class="pkg">affy</span> package quite well.  It does so by still using a
constant memory profile, i.e. it is possible to use this implementation
to run RMA on a much large data set than what is possible with
<span class="pkg">affy</span>.  At least 20-50 <em>times</em> more samples should be
doable, if not more.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] Irizarry et al.
<em>Summaries of Affymetrix GeneChip probe level data</em>.
NAR, 2003, 31, e15.<br />
</p>


<h3>See Also</h3>

<p><code><a href="#topic+doRMA">doRMA</a></code>().
</p>

<hr>
<h2 id='LimmaBackgroundCorrection'>The LimmaBackgroundCorrection class</h2><span id='topic+LimmaBackgroundCorrection'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class LimmaBackgroundCorrection</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+BackgroundCorrection">BackgroundCorrection</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>LimmaBackgroundCorrection</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+NormExpBackgroundCorrection">NormExpBackgroundCorrection</a><br />
</p>
<p>public static class <b>LimmaBackgroundCorrection</b><br />
extends <em><a href="#topic+BackgroundCorrection">BackgroundCorrection</a></em><br />
</p>
<p>This class represents the various &quot;background&quot; correction methods
implemented in the <span class="pkg">limma</span> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LimmaBackgroundCorrection(..., args=NULL, addJitter=FALSE, jitterSd=0.2, seed=6022007)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LimmaBackgroundCorrection_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+BackgroundCorrection">BackgroundCorrection</a></code>.</p>
</td></tr>
<tr><td><code id="LimmaBackgroundCorrection_+3A_args">args</code></td>
<td>
<p>A <code><a href="base.html#topic+list">list</a></code> of additional arguments passed to the
correction algorithm.</p>
</td></tr>
<tr><td><code id="LimmaBackgroundCorrection_+3A_addjitter">addJitter</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, Zero-mean gaussian noise is added to the
signals before being background corrected.</p>
</td></tr>
<tr><td><code id="LimmaBackgroundCorrection_+3A_jittersd">jitterSd</code></td>
<td>
<p>Standard deviation of the jitter noise added.</p>
</td></tr>
<tr><td><code id="LimmaBackgroundCorrection_+3A_seed">seed</code></td>
<td>
<p>An (optional) <code><a href="base.html#topic+integer">integer</a></code> specifying a temporary random seed
to be used for generating the (optional) jitter.  The random seed
is set to its original state when done.  If <code><a href="base.html#topic+NULL">NULL</a></code>, it is not set.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, only PM signals are background corrected and MMs are
left unchanged.
</p>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from BackgroundCorrection</b>:<br />
getParameters, process
</p>
<p><b>Methods inherited from ProbeLevelTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Jitter noise</h3>

<p>The fitting algorithm of the normal+exponential background correction
model may not converge if there too many small and discrete signals.
To overcome this problem, a small amount of noise may be added to the
signals before fitting the model.  This is an ad hoc solution that
seems to work.
However, adding Gaussian noise may generate non-positive signals.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson.
Adopted from RmaBackgroundCorrection by Ken Simpson.</p>


<h3>See Also</h3>

<p>Internally, <code><a href="limma.html#topic+backgroundCorrect">backgroundCorrect</a></code> is used.
</p>

<hr>
<h2 id='LinearModelProbeSequenceNormalization'>The LinearModelProbeSequenceNormalization class</h2><span id='topic+LinearModelProbeSequenceNormalization'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class LinearModelProbeSequenceNormalization</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform3">ProbeLevelTransform3</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AbstractProbeSequenceNormalization">AbstractProbeSequenceNormalization</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>LinearModelProbeSequenceNormalization</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+BasePositionNormalization">BasePositionNormalization</a></em><br />
</p>
<p>public abstract static class <b>LinearModelProbeSequenceNormalization</b><br />
extends <em><a href="#topic+AbstractProbeSequenceNormalization">AbstractProbeSequenceNormalization</a></em><br />
</p>
<p>This abstract class represents a normalization method that corrects
for systematic effects in the probe intensities due to probe-sequence
dependent effects that can be modeled using a linear model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LinearModelProbeSequenceNormalization(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LinearModelProbeSequenceNormalization_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+AbstractProbeSequenceNormalization">AbstractProbeSequenceNormalization</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from AbstractProbeSequenceNormalization</b>:<br />
fitOne, getAromaCellSequenceFile, getParameters, getTargetFile, indexOfMissingSequences, predictOne, process
</p>
<p><b>Methods inherited from ProbeLevelTransform3</b>:<br />
getAsteriskTags, getCellsTo, getCellsToFit, getCellsToUpdate, getParameters, getUnitsTo, getUnitsToFit, getUnitsToUpdate, writeSignals
</p>
<p><b>Methods inherited from ProbeLevelTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Requirements</h3>

<p>This class requires that an aroma probe sequence file is available
for the chip type.
</p>


<h3>Memory usage</h3>

<p>The model fitting methods of this class are bounded in memory.
This is done by first building up the normal equations incrementally
in chunks of cells.  The generation of normal equations is otherwise
the step that consumes the most memory.
When the normal equations are available, the <code><a href="base.html#topic+solve">solve</a></code>()
method is used to solve the equations.  Note that this algorithm is
still exact.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='MatNormalization'>The MatNormalization class</h2><span id='topic+MatNormalization'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class MatNormalization</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform3">ProbeLevelTransform3</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AbstractProbeSequenceNormalization">AbstractProbeSequenceNormalization</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>MatNormalization</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>MatNormalization</b><br />
extends <em><a href="#topic+AbstractProbeSequenceNormalization">AbstractProbeSequenceNormalization</a></em><br />
</p>
<p>This class represents a normalization method that corrects for systematic
effects in the probe intensities due to differences in the number of
A, C, G, and T:s and the match scores according to MAT [1].
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MatNormalization(..., unitsToFit=NULL, model=c("lm"), nbrOfBins=200)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MatNormalization_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+AbstractProbeSequenceNormalization">AbstractProbeSequenceNormalization</a></code>.</p>
</td></tr>
<tr><td><code id="MatNormalization_+3A_unitstofit">unitsToFit</code></td>
<td>
<p>The units from which the normalization curve should
be estimated.  If <code><a href="base.html#topic+NULL">NULL</a></code>, all are considered.</p>
</td></tr>
<tr><td><code id="MatNormalization_+3A_model">model</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying the model used to fit
the base-count effects.</p>
</td></tr>
<tr><td><code id="MatNormalization_+3A_nbrofbins">nbrOfBins</code></td>
<td>
<p>The number of bins to use for the variance smoothing step.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AbstractProbeSequenceNormalization</b>:<br />
fitOne, getAromaCellSequenceFile, getParameters, getTargetFile, indexOfMissingSequences, predictOne, process
</p>
<p><b>Methods inherited from ProbeLevelTransform3</b>:<br />
getAsteriskTags, getCellsTo, getCellsToFit, getCellsToUpdate, getParameters, getUnitsTo, getUnitsToFit, getUnitsToUpdate, writeSignals
</p>
<p><b>Methods inherited from ProbeLevelTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Requirements</h3>

<p>This class requires that an aroma probe sequence file and aroma
match scores file is available for the chip type.
</p>


<h3>Author(s)</h3>

<p>Mark Robinson</p>


<h3>References</h3>

<p>[1] Johnson WE, Li W, Meyer CA, Gottardo R, Carroll JS, Brown M, Liu XS.
<em>Model-based analysis of tiling-arrays for ChIP-chip</em>, PNAS, 2006.
</p>

<hr>
<h2 id='MatSmoothing'>The MatSmoothing class</h2><span id='topic+MatSmoothing'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class MatSmoothing</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code>MatSmoothing</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>MatSmoothing</b><br />
extends <em><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></em><br />
</p>
<p>This class represents a function for smoothing data with a trimmed mean.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MatSmoothing(..., design=NULL, probeWindow=300, nProbes=10, meanTrim=0.1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MatSmoothing_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code>.</p>
</td></tr>
<tr><td><code id="MatSmoothing_+3A_design">design</code></td>
<td>
<p>A design <code><a href="base.html#topic+matrix">matrix</a></code>.</p>
</td></tr>
<tr><td><code id="MatSmoothing_+3A_probewindow">probeWindow</code></td>
<td>
<p>Bandwidth to use.  Effectively the width is
2*probeWindow since it looks probeWindow bases in either direction.</p>
</td></tr>
<tr><td><code id="MatSmoothing_+3A_nprobes">nProbes</code></td>
<td>
<p>The minimum number of probes to calculate a MAT score for.</p>
</td></tr>
<tr><td><code id="MatSmoothing_+3A_meantrim">meanTrim</code></td>
<td>
<p>The amount of trimming of the mean in [0,0.5].</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ProbeLevelTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Mark Robinson, Henrik Bengtsson</p>

<hr>
<h2 id='MbeiCnPlm'>The MbeiCnPlm class</h2><span id='topic+MbeiCnPlm'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class MbeiCnPlm</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+MbeiPlm">MbeiPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+MbeiSnpPlm">MbeiSnpPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+SnpPlm">SnpPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CnPlm">CnPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>MbeiCnPlm</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>MbeiCnPlm</b><br />
extends <a href="#topic+CnPlm">CnPlm</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MbeiCnPlm(..., combineAlleles=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MbeiCnPlm_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+MbeiSnpPlm">MbeiSnpPlm</a></code>.</p>
</td></tr>
<tr><td><code id="MbeiCnPlm_+3A_combinealleles">combineAlleles</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">FALSE</a></code>, allele A and allele B are treated
separately, otherwise together.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from CnPlm</b>:<br />
getCellIndices, getChipEffectSet, getCombineAlleles, getParameters, getProbeAffinityFile, setCombineAlleles
</p>
<p><b>Methods inherited from SnpPlm</b>:<br />
getCellIndices, getChipEffectSet, getMergeStrands, getParameters, getProbeAffinityFile, setMergeStrands
</p>
<p><b>Methods inherited from MbeiSnpPlm</b>:<br />
getAsteriskTags
</p>
<p><b>Methods inherited from MbeiPlm</b>:<br />
getAsteriskTags, getFitUnitGroupFunction
</p>
<p><b>Methods inherited from ProbeLevelModel</b>:<br />
calculateResidualSet, calculateWeights, fit, getAsteriskTags, getCalculateResidualsFunction, getChipEffectSet, getProbeAffinityFile, getResidualSet, getRootPath, getWeightsSet
</p>
<p><b>Methods inherited from MultiArrayUnitModel</b>:<br />
getListOfPriors, setListOfPriors, validate
</p>
<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='MbeiPlm'>The MbeiPlm class</h2><span id='topic+MbeiPlm'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class MbeiPlm</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>MbeiPlm</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+MbeiCnPlm">MbeiCnPlm</a></em>, <em><a href="#topic+MbeiSnpPlm">MbeiSnpPlm</a></em><br />
</p>
<p>public abstract static class <b>MbeiPlm</b><br />
extends <em><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></em><br />
</p>
<p>This class represents the <em>model-based expression indexes</em> (MBEI)
multiplicative model in Li &amp; Wong (2001).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MbeiPlm(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MbeiPlm_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from ProbeLevelModel</b>:<br />
calculateResidualSet, calculateWeights, fit, getAsteriskTags, getCalculateResidualsFunction, getChipEffectSet, getProbeAffinityFile, getResidualSet, getRootPath, getWeightsSet
</p>
<p><b>Methods inherited from MultiArrayUnitModel</b>:<br />
getListOfPriors, setListOfPriors, validate
</p>
<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Model</h3>

<p>For a single unit group, the multiplicative model is:
</p>
<p style="text-align: center;"><code class="reqn">y_{ik} = \theta_i \phi_k + \varepsilon_{ik}</code>
</p>

<p>where <code class="reqn">\theta_i</code> are the chip effects for arrays <code class="reqn">i=1,...,I</code>,
and <code class="reqn">\phi_k</code> are the probe affinities for probes <code class="reqn">k=1,...,K</code>.
The <code class="reqn">\varepsilon_{ik}</code> are zero-mean noise with equal variance.
To make to parameters identifiable, the constraint
<code class="reqn">\prod_k \phi_k = 1</code> is added.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>Li, C. and Wong, W.H. (2001), Genome Biology 2, 1-11.<br />
Li, C. and Wong, W.H. (2001), Proc. Natl. Acad. Sci USA 98, 31-36.<br />
</p>


<h3>See Also</h3>

<p>Internally <code><a href="affy.html#topic+fit.li.wong">fit.li.wong</a></code> is used.
</p>

<hr>
<h2 id='MbeiSnpPlm'>The MbeiSnpPlm class</h2><span id='topic+MbeiSnpPlm'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class MbeiSnpPlm</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+MbeiPlm">MbeiPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+SnpPlm">SnpPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>MbeiSnpPlm</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+MbeiCnPlm">MbeiCnPlm</a></em><br />
</p>
<p>public abstract static class <b>MbeiSnpPlm</b><br />
extends <a href="#topic+SnpPlm">SnpPlm</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MbeiSnpPlm(..., mergeStrands=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MbeiSnpPlm_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+MbeiPlm">MbeiPlm</a></code>.</p>
</td></tr>
<tr><td><code id="MbeiSnpPlm_+3A_mergestrands">mergeStrands</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the sense and the anti-sense strands are
fitted together, otherwise separately.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from SnpPlm</b>:<br />
getCellIndices, getChipEffectSet, getMergeStrands, getParameters, getProbeAffinityFile, setMergeStrands
</p>
<p><b>Methods inherited from MbeiPlm</b>:<br />
getAsteriskTags, getFitUnitGroupFunction
</p>
<p><b>Methods inherited from ProbeLevelModel</b>:<br />
calculateResidualSet, calculateWeights, fit, getAsteriskTags, getCalculateResidualsFunction, getChipEffectSet, getProbeAffinityFile, getResidualSet, getRootPath, getWeightsSet
</p>
<p><b>Methods inherited from MultiArrayUnitModel</b>:<br />
getListOfPriors, setListOfPriors, validate
</p>
<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='Model'>The Model class</h2><span id='topic+Model'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class Model</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><em><code>Model</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AffineCnPlm">AffineCnPlm</a></em>, <em><a href="#topic+AffinePlm">AffinePlm</a></em>, <em><a href="#topic+AffineSnpPlm">AffineSnpPlm</a></em>, <em><a href="#topic+AlleleSummation">AlleleSummation</a></em>, <em><a href="#topic+AvgCnPlm">AvgCnPlm</a></em>, <em><a href="#topic+AvgPlm">AvgPlm</a></em>, <em><a href="#topic+AvgSnpPlm">AvgSnpPlm</a></em>, <a href="#topic+CrlmmModel">CrlmmModel</a>, <em><a href="#topic+ExonRmaPlm">ExonRmaPlm</a></em>, <em><a href="#topic+FirmaModel">FirmaModel</a></em>, <em><a href="#topic+HetLogAddCnPlm">HetLogAddCnPlm</a></em>, <em><a href="#topic+HetLogAddPlm">HetLogAddPlm</a></em>, <em><a href="#topic+HetLogAddSnpPlm">HetLogAddSnpPlm</a></em>, <em><a href="#topic+MbeiCnPlm">MbeiCnPlm</a></em>, <em><a href="#topic+MbeiPlm">MbeiPlm</a></em>, <em><a href="#topic+MbeiSnpPlm">MbeiSnpPlm</a></em>, <em><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></em>, <em><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></em>, <em><a href="#topic+RmaCnPlm">RmaCnPlm</a></em>, <em><a href="#topic+RmaPlm">RmaPlm</a></em>, <em><a href="#topic+RmaSnpPlm">RmaSnpPlm</a></em>, <em><a href="#topic+SingleArrayUnitModel">SingleArrayUnitModel</a></em>, <em><a href="#topic+UnitModel">UnitModel</a></em><br />
</p>
<p>public abstract static class <b>Model</b><br />
extends <a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a><br />
</p>
<p>This class is abstract and represents a generic model that applies
to a data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Model(dataSet=NULL, tags="*", ..., .onUnknownArgs=c("error", "warning", "ignore"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Model_+3A_dataset">dataSet</code></td>
<td>
<p>The data set to which this model should be fitted.</p>
</td></tr>
<tr><td><code id="Model_+3A_tags">tags</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> of tags to be appended to the tags of
the input data set.</p>
</td></tr>
<tr><td><code id="Model_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="Model_+3A_.onunknownargs">.onUnknownArgs</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying what should occur
if there are unknown arguments in <code>...</code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>fit</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAsteriskTags</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getDataSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFullName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPath</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getTags</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setTags</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='MultiArrayUnitModel'>The MultiArrayUnitModel class</h2><span id='topic+MultiArrayUnitModel'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class MultiArrayUnitModel</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><em><code>MultiArrayUnitModel</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AffineCnPlm">AffineCnPlm</a></em>, <em><a href="#topic+AffinePlm">AffinePlm</a></em>, <em><a href="#topic+AffineSnpPlm">AffineSnpPlm</a></em>, <em><a href="#topic+AvgCnPlm">AvgCnPlm</a></em>, <em><a href="#topic+AvgPlm">AvgPlm</a></em>, <em><a href="#topic+AvgSnpPlm">AvgSnpPlm</a></em>, <em><a href="#topic+ExonRmaPlm">ExonRmaPlm</a></em>, <em><a href="#topic+HetLogAddCnPlm">HetLogAddCnPlm</a></em>, <em><a href="#topic+HetLogAddPlm">HetLogAddPlm</a></em>, <em><a href="#topic+HetLogAddSnpPlm">HetLogAddSnpPlm</a></em>, <em><a href="#topic+MbeiCnPlm">MbeiCnPlm</a></em>, <em><a href="#topic+MbeiPlm">MbeiPlm</a></em>, <em><a href="#topic+MbeiSnpPlm">MbeiSnpPlm</a></em>, <em><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></em>, <em><a href="#topic+RmaCnPlm">RmaCnPlm</a></em>, <em><a href="#topic+RmaPlm">RmaPlm</a></em>, <em><a href="#topic+RmaSnpPlm">RmaSnpPlm</a></em><br />
</p>
<p>public abstract static class <b>MultiArrayUnitModel</b><br />
extends <em><a href="#topic+UnitModel">UnitModel</a></em><br />
</p>
<p>This abstract class represents a unit model that fits one model per unit
based on signals for all arrays in the data set.
The nature of a multi-array unit model is that all arrays must be
available at the time of the fit and the estimated parameters will
depend on the data from all arrays.
Thus, if the signals in one array changes the model has to be refitted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MultiArrayUnitModel(..., listOfPriors=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MultiArrayUnitModel_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+UnitModel">UnitModel</a></code>.</p>
</td></tr>
<tr><td><code id="MultiArrayUnitModel_+3A_listofpriors">listOfPriors</code></td>
<td>
<p>A <code><a href="base.html#topic+list">list</a></code> of priors to be used when fitting
the model.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getListOfPriors</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setListOfPriors</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='MultiArrayUnitModel+24getFitUnitGroupFunction'>Static method to get the low-level function that fits the PLM</h2><span id='topic+MultiArrayUnitModel+24getFitUnitGroupFunction'></span><span id='topic+getFitUnitGroupFunction.MultiArrayUnitModel'></span><span id='topic+MultiArrayUnitModel.getFitUnitGroupFunction'></span><span id='topic+getFitUnitGroupFunction+2CMultiArrayUnitModel-method'></span>

<h3>Description</h3>

<p>Static method to get the low-level function that fits the PLM.
Any subclass model must provide this method, which should return
a <code><a href="base.html#topic+function">function</a></code> that accepts an IxK <code><a href="base.html#topic+matrix">matrix</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## MultiArrayUnitModel$getFitUnitGroupFunction()

## Don't use the below:
## S3 method for class 'MultiArrayUnitModel'
getFitUnitGroupFunction(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MultiArrayUnitModel+2B24getFitUnitGroupFunction_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+function">function</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></code>.
</p>

<hr>
<h2 id='nbrOfGroupsPerUnit.AffymetrixCdfFile'>Gets the number of groups in each unit</h2><span id='topic+nbrOfGroupsPerUnit.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.nbrOfGroupsPerUnit'></span><span id='topic+nbrOfGroupsPerUnit+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Gets the number of groups in each unit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
nbrOfGroupsPerUnit(this, units=NULL, ..., force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nbrOfGroupsPerUnit.AffymetrixCdfFile_+3A_units">units</code></td>
<td>
<p>The units of interest. If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are considered.</p>
</td></tr>
<tr><td><code id="nbrOfGroupsPerUnit.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="nbrOfGroupsPerUnit.AffymetrixCdfFile_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Once read from file, this information is cached in memory for efficiency.
The cache can be cleared by calling <code>gc(cdf)</code>.
</p>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+vector">vector</a></code> of <code><a href="base.html#topic+integer">integer</a></code>s.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='nbrOfSnps.AffymetrixCdfFile'>Gets the number of SNPs</h2><span id='topic+nbrOfSnps.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.nbrOfSnps'></span><span id='topic+nbrOfSnps+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Gets the number of SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
nbrOfSnps(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nbrOfSnps.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+getSnpNames.AffymetrixCdfFile">*getSnpNames</a>()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="base.html#topic+integer">integer</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>Internally, <code><a href="#topic+getSnpNames.AffymetrixCdfFile">*getSnpNames</a>()</code> is used to identify SNPs.
</p>
<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='Non-documented+20objects'>Non-documented objects</h2><span id='topic+Non-documented+20objects'></span><span id='topic+ASCRMAv2'></span><span id='topic+AffymetrixAptSummaryFile'></span><span id='topic+AffymetrixCnChpFile'></span><span id='topic+AffymetrixCsvFile'></span><span id='topic+AffymetrixCsvGenomeInformation'></span><span id='topic+AffymetrixNetAffxCsvFile'></span><span id='topic+AffymetrixPlatform'></span><span id='topic+AffymetrixTabularFile'></span><span id='topic+AffymetrixTsvFile'></span><span id='topic+AromaAffymetrix'></span><span id='topic+AromaCellMatchScoreFile'></span><span id='topic+AromaPipeline'></span><span id='topic+AromaUfcFile'></span><span id='topic+ChipEffectSetTuple'></span><span id='topic+CnChipEffectSetTuple'></span><span id='topic+CrlmmModel'></span><span id='topic+Env2Cdf'></span><span id='topic+SmoothSaModel'></span><span id='topic+SpatialRowColumnNormalization'></span><span id='topic+Transform'></span><span id='topic+UflSnpInformation'></span><span id='topic+addColorMap'></span><span id='topic+addColorMap.ArrayExplorer'></span><span id='topic+addColorMap.SpatialReporter'></span><span id='topic+addExclCells'></span><span id='topic+addExclCells.DChipQuantileNormalization'></span><span id='topic+allocate.AromaCellMatchScoreFile'></span><span id='topic+allocate.CrlmmParametersFile'></span><span id='topic+allocateFromCdf'></span><span id='topic+allocateFromCdf.AromaCellCpgFile'></span><span id='topic+allocateFromCdf.AromaCellMatchScoreFile'></span><span id='topic+allocateFromCdf.AromaCellPositionFile'></span><span id='topic+allocateFromCdf.AromaCellSequenceFile'></span><span id='topic+allocateFromCdf.AromaUfcFile'></span><span id='topic+allocateFromCdf.AromaUflFile'></span><span id='topic+allocateFromCdf.AromaUnitGcContentFile'></span><span id='topic+annotateMvsA'></span><span id='topic+annotateMvsA.AffymetrixCelFile'></span><span id='topic+append.AffymetrixCelSet'></span><span id='topic+append.CnagCfhSet'></span><span id='topic+applyAnyOrder'></span><span id='topic+applyAnyOrder.CopyNumberChromosomalModel'></span><span id='topic+applyCCF0'></span><span id='topic+applyCCF0.CopyNumberChromosomalModel'></span><span id='topic+applyCFC0'></span><span id='topic+applyCFC0.CopyNumberChromosomalModel'></span><span id='topic+applyToUnitIntensities'></span><span id='topic+applyToUnitIntensities.AffymetrixCelSet'></span><span id='topic+as.AffymetrixCelSet'></span><span id='topic+as.AffymetrixCnChpSet'></span><span id='topic+as.AffymetrixFileSet'></span><span id='topic+as.ChipEffectSetTuple'></span><span id='topic+as.ChipEffectSetTuple.ChipEffectSetTuple'></span><span id='topic+as.ChipEffectSetTuple.default'></span><span id='topic+as.CnChipEffectSetTuple'></span><span id='topic+as.CnChipEffectSetTuple.CnChipEffectSetTuple'></span><span id='topic+as.CnChipEffectSetTuple.default'></span><span id='topic+as.CnagCfhSet'></span><span id='topic+as.CopyNumberDataSetTuple.CnChipEffectSet'></span><span id='topic+as.DChipDcpSet'></span><span id='topic+as.character.AffymetrixCdfFile'></span><span id='topic+as.character.AffymetrixCelFile'></span><span id='topic+as.character.AffymetrixCelSetReporter'></span><span id='topic+as.character.AffymetrixCnChpFile'></span><span id='topic+as.character.AffymetrixPgfFile'></span><span id='topic+as.character.AffymetrixProbeTabFile'></span><span id='topic+as.character.AromaChipTypeAnnotationFile'></span><span id='topic+as.character.ArrayExplorer'></span><span id='topic+as.character.ChipEffectFile'></span><span id='topic+as.character.ChipEffectSet'></span><span id='topic+as.character.CnagCfhFile'></span><span id='topic+as.character.DChipCdfBinFile'></span><span id='topic+as.character.DChipDcpFile'></span><span id='topic+as.character.DChipQuantileNormalization'></span><span id='topic+as.character.FirmaModel'></span><span id='topic+as.character.GenericReporter'></span><span id='topic+as.character.GenomeInformation'></span><span id='topic+as.character.Model'></span><span id='topic+as.character.ProbeAffinityFile'></span><span id='topic+as.character.QualityAssessmentModel'></span><span id='topic+as.character.ResidualFile'></span><span id='topic+as.character.ResidualSet'></span><span id='topic+as.character.SmoothMultiarrayModel'></span><span id='topic+as.character.SnpInformation'></span><span id='topic+as.character.SpatialReporter'></span><span id='topic+as.character.TransformReport'></span><span id='topic+as.character.WeightsFile'></span><span id='topic+as.character.WeightsSet'></span><span id='topic+assertAnnotationData'></span><span id='topic+assertAnnotationData.ASCRMAv2'></span><span id='topic+averageQuantile'></span><span id='topic+bgAdjustGcrma'></span><span id='topic+bgAdjustOptical'></span><span id='topic+bgAdjustRma'></span><span id='topic+boxplotStats'></span><span id='topic+boxplotStats.ChipEffectSet'></span><span id='topic+byChipType.AffymetrixCsvGenomeInformation'></span><span id='topic+byChipType.AffymetrixProbeTabFile'></span><span id='topic+byChipType.AffymetrixTabularFile'></span><span id='topic+byChipType.AffymetrixTsvFile'></span><span id='topic+byChipType.AromaCellMatchScoreFile'></span><span id='topic+byChipType.DChipCdfBinFile'></span><span id='topic+byChipType.UflSnpInformation'></span><span id='topic+byName.AffymetrixCelSet'></span><span id='topic+byName.AffymetrixCnChpSet'></span><span id='topic+byName.CnagCfhSet'></span><span id='topic+byName.CrlmmParametersSet'></span><span id='topic+byName.DChipDcpSet'></span><span id='topic+byPath.AffymetrixCelSet'></span><span id='topic+byPath.AffymetrixCelSetTuple'></span><span id='topic+byPath.AffymetrixCnChpSet'></span><span id='topic+byPath.ChipEffectSet'></span><span id='topic+byPath.CnChipEffectSet'></span><span id='topic+byPath.CnagCfhSet'></span><span id='topic+byPath.CrlmmParametersSet'></span><span id='topic+byPath.DChipDcpSet'></span><span id='topic+byPath.ExonChipEffectSet'></span><span id='topic+byPath.FirmaSet'></span><span id='topic+byPath.ResidualSet'></span><span id='topic+byPath.SnpChipEffectSet'></span><span id='topic+byPath.WeightsSet'></span><span id='topic+calcRawCnStats'></span><span id='topic+calcRawCnStats.CopyNumberChromosomalModel'></span><span id='topic+calcRawCnStats.default'></span><span id='topic+calculateAffinities'></span><span id='topic+calculateAffinities.GcRmaBackgroundCorrection'></span><span id='topic+calculateAverages'></span><span id='topic+calculateAverages.FragmentEquivalentClassNormalization'></span><span id='topic+calculateBaseline'></span><span id='topic+calculateConfidenceScores'></span><span id='topic+calculateConfidenceScores.CrlmmModel'></span><span id='topic+calculateFieldBoxplotStats'></span><span id='topic+calculateFieldBoxplotStats.ChipEffectSet'></span><span id='topic+calculateMargins'></span><span id='topic+calculateMargins.SpatialReporter'></span><span id='topic+calculateNuseBoxplotStats'></span><span id='topic+calculateNuseBoxplotStats.ChipEffectSet'></span><span id='topic+calculateParametersGsb'></span><span id='topic+calculateResidualSet'></span><span id='topic+calculateResidualSet.ProbeLevelModel'></span><span id='topic+calculateRleBoxplotStats'></span><span id='topic+calculateRleBoxplotStats.ChipEffectSet'></span><span id='topic+calculateTargetDistribution'></span><span id='topic+calculateTargetDistribution.QuantileNormalization'></span><span id='topic+calculateWeights'></span><span id='topic+calculateWeights.ExonRmaPlm'></span><span id='topic+calculateWeights.FirmaModel'></span><span id='topic+calculateWeights.ProbeLevelModel'></span><span id='topic+calibrateOne'></span><span id='topic+calibrateOne.ReseqCrosstalkCalibration'></span><span id='topic+cdfMergeGroups'></span><span id='topic+clearCache.AffymetrixCelSet'></span><span id='topic+clearData'></span><span id='topic+clone.AffymetrixCelFile'></span><span id='topic+clone.AffymetrixCelSet'></span><span id='topic+clone.AffymetrixCnChpFile'></span><span id='topic+clone.CnagCfhFile'></span><span id='topic+clone.CnagCfhSet'></span><span id='topic+compare'></span><span id='topic+computeAffinities'></span><span id='topic+computeAffinitiesByACS'></span><span id='topic+computeAffinitiesByACS.AffymetrixCdfFile'></span><span id='topic+convert'></span><span id='topic+convertToUnique'></span><span id='topic+convertToUnique.AffymetrixCelSet'></span><span id='topic+convertUnits'></span><span id='topic+createExonByTranscriptCdf'></span><span id='topic+createFrom'></span><span id='topic+createMonocellCdf'></span><span id='topic+createOutputTuple'></span><span id='topic+createOutputTuple.SmoothMultiarrayModel'></span><span id='topic+createParamCdf'></span><span id='topic+createParamCdf.ChipEffectFile'></span><span id='topic+createParamCdf.FirmaFile'></span><span id='topic+createUniqueCdf'></span><span id='topic+decode'></span><span id='topic+decode.ParameterCelFile'></span><span id='topic+dim.DChipDcpFile'></span><span id='topic+dropCellsFromCdfList'></span><span id='topic+encode'></span><span id='topic+encode.ParameterCelFile'></span><span id='topic+env2Cdf'></span><span id='topic+excludeChrXFromFit'></span><span id='topic+excludeChrXFromFit.DChipQuantileNormalization'></span><span id='topic+exportAromaSignalBinaryFileList'></span><span id='topic+exportAromaSignalBinaryFileList.CnChipEffectFile'></span><span id='topic+exportTotalAndFracB'></span><span id='topic+exportTotalAndFracB.CnChipEffectFile'></span><span id='topic+exportTotalAndFracB.CnChipEffectSet'></span><span id='topic+exportTotalAndFracB.DChipDcpSet'></span><span id='topic+exportTotalAndFracB.SnpChipEffectFile'></span><span id='topic+exportTotalAndFracB.SnpChipEffectSet'></span><span id='topic+exportTotalCnRatioSet.AffymetrixCnChpSet'></span><span id='topic+extractAlleleSet'></span><span id='topic+extractAlleleSet.SnpChipEffectSet'></span><span id='topic+extractCNT'></span><span id='topic+extractCNT.SnpChipEffectFile'></span><span id='topic+extractCNT.SnpChipEffectSet'></span><span id='topic+extractChromosomalDataFrame'></span><span id='topic+extractChromosomalDataFrame.ChipEffectFile'></span><span id='topic+extractChromosomalDataFrame.ChipEffectSet'></span><span id='topic+extractDataFrame'></span><span id='topic+extractDataFrame.ParameterCelFile'></span><span id='topic+extractLogRatios'></span><span id='topic+extractLogRatios.AffymetrixCnChpFile'></span><span id='topic+extractLogRatios.AffymetrixCnChpSet'></span><span id='topic+extractMatrix.AffymetrixCelFile'></span><span id='topic+extractMatrix.ChipEffectFile'></span><span id='topic+extractMatrix.ChipEffectSet'></span><span id='topic+extractMatrix.FirmaFile'></span><span id='topic+extractMatrix.FirmaSet'></span><span id='topic+extractMatrix.ParameterCelFile'></span><span id='topic+extractSnpCnvQSet'></span><span id='topic+extractSnpCnvQSet.SnpChipEffectSet'></span><span id='topic+extractSnpQSet'></span><span id='topic+extractSnpQSet.SnpChipEffectSet'></span><span id='topic+extractTheta'></span><span id='topic+extractTheta.ChipEffectFile'></span><span id='topic+extractTheta.ChipEffectSet'></span><span id='topic+extractTheta.CnChipEffectFile'></span><span id='topic+extractTheta.CnChipEffectSet'></span><span id='topic+extractTheta.DChipDcpFile'></span><span id='topic+extractTheta.DChipDcpSet'></span><span id='topic+extractTheta.SnpChipEffectFile'></span><span id='topic+extractTheta.SnpChipEffectSet'></span><span id='topic+extractTheta.SnpCnvQSet'></span><span id='topic+extractTheta.SnpQSet'></span><span id='topic+extractTotalAndFracB'></span><span id='topic+extractTotalAndFracB.SnpChipEffectFile'></span><span id='topic+extractTotalAndFreqB'></span><span id='topic+extractTotalAndFreqB.CnChipEffectFile'></span><span id='topic+extractTotalAndFreqB.CnChipEffectSet'></span><span id='topic+extractTotalAndFreqB.SnpChipEffectSet'></span><span id='topic+extractTotalAndFreqB.default'></span><span id='topic+findByCdf2'></span><span id='topic+findByCdf2.default'></span><span id='topic+findByChipType.AffymetrixCsvFile'></span><span id='topic+findByChipType.AffymetrixCsvGenomeInformation'></span><span id='topic+findByChipType.AffymetrixNetAffxCsvFile'></span><span id='topic+findByChipType.AffymetrixProbeTabFile'></span><span id='topic+findByChipType.AffymetrixTabularFile'></span><span id='topic+findByChipType.AffymetrixTsvFile'></span><span id='topic+findByChipType.DChipCdfBinFile'></span><span id='topic+findByChipType.DChipGenomeInformation'></span><span id='topic+findByChipType.DChipSnpInformation'></span><span id='topic+findByChipType.UflSnpInformation'></span><span id='topic+findByChipType.UgpGenomeInformation'></span><span id='topic+findByName.AffymetrixCelSet'></span><span id='topic+findByName.AffymetrixCnChpSet'></span><span id='topic+findByName.ChipEffectSet'></span><span id='topic+findByName.CnagCfhSet'></span><span id='topic+findByName.DChipDcpSet'></span><span id='topic+findTargetDistributionFile'></span><span id='topic+findTargetDistributionFile.QuantileNormalization'></span><span id='topic+findUnitsTodo.AlleleSummation'></span><span id='topic+findUnitsTodo.ChipEffectFile'></span><span id='topic+findUnitsTodo.ChipEffectSet'></span><span id='topic+findUnitsTodo.CrlmmModel'></span><span id='topic+findUnitsTodo.CrlmmParametersFile'></span><span id='topic+findUnitsTodo.CrlmmParametersSet'></span><span id='topic+findUnitsTodo.ExonChipEffectSet'></span><span id='topic+findUnitsTodo.FirmaFile'></span><span id='topic+findUnitsTodo.FirmaSet'></span><span id='topic+findUnitsTodo.QualityAssessmentFile'></span><span id='topic+findUnitsTodo.ResidualFile'></span><span id='topic+findUnitsTodo.ResidualSet'></span><span id='topic+findUnitsTodo.WeightsFile'></span><span id='topic+findUnitsTodo.WeightsSet'></span><span id='topic+fit.CrlmmModel'></span><span id='topic+fitCnProbes'></span><span id='topic+fitCnProbes.UnitModel'></span><span id='topic+fitOne'></span><span id='topic+fitOne.AbstractProbeSequenceNormalization'></span><span id='topic+fitOne.BaseCountNormalization'></span><span id='topic+fitOne.LinearModelProbeSequenceNormalization'></span><span id='topic+fitOne.MatNormalization'></span><span id='topic+fitOne.ReseqCrosstalkCalibration'></span><span id='topic+fitOne.ScaleNormalization3'></span><span id='topic+fitOne.UnitTypeScaleNormalization'></span><span id='topic+fitOneArray'></span><span id='topic+fitOneArray.SingleArrayUnitModel'></span><span id='topic+fitOneChromosome'></span><span id='topic+fitOneChromosome.SmoothMultiarrayModel'></span><span id='topic+fitPlasqUnit'></span><span id='topic+fitPlasqUnit.matrix'></span><span id='topic+fitQuantileNormFcn'></span><span id='topic+fixSearchPath.AromaAffymetrix'></span><span id='topic+fromCdf'></span><span id='topic+fromCdf.AffymetrixProbeTabFile'></span><span id='topic+fromDataFile'></span><span id='topic+fromDataFile.FirmaFile'></span><span id='topic+fromDataFile.ResidualFile'></span><span id='topic+fromDataFile.WeightsFile'></span><span id='topic+fromDataSet'></span><span id='topic+fromDataSet.ChipEffectSet'></span><span id='topic+fromDataSet.FirmaSet'></span><span id='topic+fromDataSet.GenomeInformation'></span><span id='topic+fromDataSet.ResidualSet'></span><span id='topic+fromDataSet.SnpInformation'></span><span id='topic+fromDataSet.WeightsSet'></span><span id='topic+fromFile.AffymetrixCnChpFile'></span><span id='topic+fromFile.DChipCdfBinFile'></span><span id='topic+fromFile.DChipDcpFile'></span><span id='topic+getACSFile'></span><span id='topic+getACSFile.AffymetrixCdfFile'></span><span id='topic+getAlias.ArrayExplorer'></span><span id='topic+getAlias.TransformReport'></span><span id='topic+getAlleleCellPairs'></span><span id='topic+getAlleleProbePairs'></span><span id='topic+getAlleleProbePairs2'></span><span id='topic+getAlleleProbePairs2.AffymetrixCdfFile'></span><span id='topic+getAlleleProbePairs3'></span><span id='topic+getAm'></span><span id='topic+getAm.AffymetrixCelFile'></span><span id='topic+getAromaCellMatchScoreFile'></span><span id='topic+getAromaCellMatchScoreFile.MatNormalization'></span><span id='topic+getAromaCellPositionFile'></span><span id='topic+getAromaCellPositionFile.MatSmoothing'></span><span id='topic+getAromaCellSequenceFile'></span><span id='topic+getAromaCellSequenceFile.AbstractProbeSequenceNormalization'></span><span id='topic+getAromaCellSequenceFile.AffymetrixCdfFile'></span><span id='topic+getAromaUfcFile'></span><span id='topic+getAromaUfcFile.FragmentEquivalentClassNormalization'></span><span id='topic+getAromaUflFile.UflSnpInformation'></span><span id='topic+getAromaUgpFile.UgpGenomeInformation'></span><span id='topic+getAromaUnitFracBCnBinarySet'></span><span id='topic+getAromaUnitFracBCnBinarySet.default'></span><span id='topic+getAromaUnitTotalCnBinarySet'></span><span id='topic+getAromaUnitTotalCnBinarySet.default'></span><span id='topic+getArrayNames'></span><span id='topic+getArrayNames.AffymetrixAptSummaryFile'></span><span id='topic+getArraysOfInput.ArrayExplorer'></span><span id='topic+getAsFullCelFile'></span><span id='topic+getAsFullCelFile.ChipEffectFile'></span><span id='topic+getAsFullCelSet'></span><span id='topic+getAsFullCelSet.ChipEffectSet'></span><span id='topic+getAsteriskTags.AdditiveCovariatesNormalization'></span><span id='topic+getAsteriskTags.AffineCnPlm'></span><span id='topic+getAsteriskTags.AffinePlm'></span><span id='topic+getAsteriskTags.AffineSnpPlm'></span><span id='topic+getAsteriskTags.AlleleSummation'></span><span id='topic+getAsteriskTags.AllelicCrosstalkCalibration'></span><span id='topic+getAsteriskTags.ArrayExplorer'></span><span id='topic+getAsteriskTags.AvgCnPlm'></span><span id='topic+getAsteriskTags.AvgPlm'></span><span id='topic+getAsteriskTags.AvgSnpPlm'></span><span id='topic+getAsteriskTags.BaseCountNormalization'></span><span id='topic+getAsteriskTags.BasePositionNormalization'></span><span id='topic+getAsteriskTags.CrlmmModel'></span><span id='topic+getAsteriskTags.ExonRmaPlm'></span><span id='topic+getAsteriskTags.FirmaModel'></span><span id='topic+getAsteriskTags.FragmentEquivalentClassNormalization'></span><span id='topic+getAsteriskTags.FragmentLengthNormalization'></span><span id='topic+getAsteriskTags.GenericReporter'></span><span id='topic+getAsteriskTags.HetLogAddCnPlm'></span><span id='topic+getAsteriskTags.HetLogAddPlm'></span><span id='topic+getAsteriskTags.HetLogAddSnpPlm'></span><span id='topic+getAsteriskTags.LimmaBackgroundCorrection'></span><span id='topic+getAsteriskTags.MatNormalization'></span><span id='topic+getAsteriskTags.MbeiCnPlm'></span><span id='topic+getAsteriskTags.MbeiPlm'></span><span id='topic+getAsteriskTags.MbeiSnpPlm'></span><span id='topic+getAsteriskTags.Model'></span><span id='topic+getAsteriskTags.NormExpBackgroundCorrection'></span><span id='topic+getAsteriskTags.ProbeLevelModel'></span><span id='topic+getAsteriskTags.ProbeLevelTransform3'></span><span id='topic+getAsteriskTags.QualityAssessmentModel'></span><span id='topic+getAsteriskTags.ReseqCrosstalkCalibration'></span><span id='topic+getAsteriskTags.RmaCnPlm'></span><span id='topic+getAsteriskTags.RmaPlm'></span><span id='topic+getAsteriskTags.RmaSnpPlm'></span><span id='topic+getAsteriskTags.ScaleNormalization'></span><span id='topic+getAsteriskTags.SmoothMultiarrayModel'></span><span id='topic+getAsteriskTags.SmoothRmaModel'></span><span id='topic+getAsteriskTags.SmoothSaModel'></span><span id='topic+getAsteriskTags.SnpChipEffectGroupMerge'></span><span id='topic+getAsteriskTags.UnitModel'></span><span id='topic+getAverage'></span><span id='topic+getAverage.AffymetrixCelSet'></span><span id='topic+getAverage.CnagCfhSet'></span><span id='topic+getAverageAsinh'></span><span id='topic+getAverageAsinh.AffymetrixCelSet'></span><span id='topic+getAverageAsinh.CnagCfhSet'></span><span id='topic+getAverageFile.ChipEffectSet'></span><span id='topic+getAverageFile.CnChipEffectSet'></span><span id='topic+getAverageFile.ExonChipEffectSet'></span><span id='topic+getAverageFile.ResidualSet'></span><span id='topic+getAverageFile.SnpChipEffectSet'></span><span id='topic+getAverageFile.WeightsSet'></span><span id='topic+getAverageLog'></span><span id='topic+getAverageLog.AffymetrixCelSet'></span><span id='topic+getAverageLog.CnagCfhSet'></span><span id='topic+getBandwidth'></span><span id='topic+getBandwidth.SmoothMultiarrayModel'></span><span id='topic+getBaseline'></span><span id='topic+getBaseline.CnChipEffectSet'></span><span id='topic+getBaseline.SnpChipEffectSet'></span><span id='topic+getBlockSizes'></span><span id='topic+getBlockSizes.SpatialRowColumnNormalization'></span><span id='topic+getCalculateResidualsFunction'></span><span id='topic+getCalculateResidualsFunction.AvgPlm'></span><span id='topic+getCalculateResidualsFunction.ProbeLevelModel'></span><span id='topic+getCalculateResidualsFunction.RmaPlm'></span><span id='topic+getCallSet'></span><span id='topic+getCallSet.CrlmmModel'></span><span id='topic+getCalls.DChipDcpFile'></span><span id='topic+getCdf'></span><span id='topic+getCdf.AdditiveCovariatesNormalization'></span><span id='topic+getCdf.AffymetrixCnChpFile'></span><span id='topic+getCdf.AffymetrixCnChpSet'></span><span id='topic+getCdf.AffymetrixProbeTabFile'></span><span id='topic+getCdf.AffymetrixTsvFile'></span><span id='topic+getCdf.AromaUnitTabularBinaryFile'></span><span id='topic+getCdf.ChipEffectGroupMerge'></span><span id='topic+getCdf.FirmaModel'></span><span id='topic+getCdf.FragmentEquivalentClassNormalization'></span><span id='topic+getCdf.FragmentLengthNormalization'></span><span id='topic+getCdf.GcContentNormalization'></span><span id='topic+getCdf.QualityAssessmentModel'></span><span id='topic+getCdf.TransformReport'></span><span id='topic+getCdfBin'></span><span id='topic+getCdfBin.DChipDcpSet'></span><span id='topic+getCellDim'></span><span id='topic+getCellDim.DChipCdfBinFile'></span><span id='topic+getCellIndices'></span><span id='topic+getCellIndices.ChipEffectSet'></span><span id='topic+getCellIndices.CnPlm'></span><span id='topic+getCellIndices.ExonRmaPlm'></span><span id='topic+getCellIndices.FirmaSet'></span><span id='topic+getCellIndices.ResidualSet'></span><span id='topic+getCellIndices.SnpPlm'></span><span id='topic+getCellIndices.WeightsSet'></span><span id='topic+getCellMapForMainCdf'></span><span id='topic+getCellMapForMainCdf.ChipEffectFile'></span><span id='topic+getCellQuartets'></span><span id='topic+getCellQuartets.AffymetrixCdfFile'></span><span id='topic+getCellsTo'></span><span id='topic+getCellsTo.ProbeLevelTransform3'></span><span id='topic+getCellsToFit'></span><span id='topic+getCellsToFit.ProbeLevelTransform3'></span><span id='topic+getCellsToInternal'></span><span id='topic+getCellsToInternal.ProbeLevelTransform3'></span><span id='topic+getCellsToUpdate'></span><span id='topic+getCellsToUpdate.ProbeLevelTransform3'></span><span id='topic+getChipEffectFileClass'></span><span id='topic+getChipEffectFileClass.ChipEffectSet'></span><span id='topic+getChipEffectFileClass.CnChipEffectSet'></span><span id='topic+getChipEffectFileClass.ExonChipEffectSet'></span><span id='topic+getChipEffectFileClass.SnpChipEffectSet'></span><span id='topic+getChipEffectSet'></span><span id='topic+getChipEffectSet.CnPlm'></span><span id='topic+getChipEffectSet.ExonRmaPlm'></span><span id='topic+getChipEffectSet.QualityAssessmentModel'></span><span id='topic+getChipEffectSet.SnpPlm'></span><span id='topic+getChipEffectSetClass'></span><span id='topic+getChipEffectSetClass.AlleleSummation'></span><span id='topic+getChipEffectSetClass.CnPlm'></span><span id='topic+getChipEffectSetClass.ExonRmaPlm'></span><span id='topic+getChipEffectSetClass.ProbeLevelModel'></span><span id='topic+getChipEffectSetClass.SnpPlm'></span><span id='topic+getChipType.AffymetrixAptSummaryFile'></span><span id='topic+getChipType.AffymetrixCdfFile'></span><span id='topic+getChipType.AffymetrixCelSetReporter'></span><span id='topic+getChipType.AffymetrixCnChpFile'></span><span id='topic+getChipType.AffymetrixPgfFile'></span><span id='topic+getChipType.AffymetrixProbeTabFile'></span><span id='topic+getChipType.AffymetrixTsvFile'></span><span id='topic+getChipType.AromaChipTypeAnnotationFile'></span><span id='topic+getChipType.CrlmmModel'></span><span id='topic+getChipType.DChipCdfBinFile'></span><span id='topic+getChipType.DChipDcpSet'></span><span id='topic+getChipType.UflSnpInformation'></span><span id='topic+getChipType.UgpGenomeInformation'></span><span id='topic+getChromosomeStats'></span><span id='topic+getChromosomeStats.GenomeInformation'></span><span id='topic+getChromosomes.GenomeInformation'></span><span id='topic+getChromosomes.UgpGenomeInformation'></span><span id='topic+getCnNames'></span><span id='topic+getColorMaps'></span><span id='topic+getColorMaps.ArrayExplorer'></span><span id='topic+getColorMaps.SpatialReporter'></span><span id='topic+getColumnNames.AffymetrixProbeTabFile'></span><span id='topic+getCombineAlleles'></span><span id='topic+getCombineAlleles.CnChipEffectSet'></span><span id='topic+getCombineAlleles.CnPlm'></span><span id='topic+getConfidenceScoreSet'></span><span id='topic+getConfidenceScoreSet.CrlmmModel'></span><span id='topic+getCovariates'></span><span id='topic+getCovariates.AdditiveCovariatesNormalization'></span><span id='topic+getCovariates.GcContentNormalization2'></span><span id='topic+getCrlmmParametersSet'></span><span id='topic+getCrlmmParametersSet.CrlmmModel'></span><span id='topic+getCrlmmPriors'></span><span id='topic+getCrlmmPriors.CrlmmModel'></span><span id='topic+getCrlmmSNPs'></span><span id='topic+getCrlmmSNPs.CrlmmModel'></span><span id='topic+getCrlmmSNPsOnChrX'></span><span id='topic+getCrlmmSNPsOnChrX.CrlmmModel'></span><span id='topic+getCrlmmSplineParameters'></span><span id='topic+getCrlmmSplineParameters.CrlmmModel'></span><span id='topic+getData'></span><span id='topic+getData.AffymetrixCelSet'></span><span id='topic+getData.AffymetrixCnChpFile'></span><span id='topic+getData.AffymetrixProbeTabFile'></span><span id='topic+getData.AffymetrixTsvFile'></span><span id='topic+getData.CnagCfhSet'></span><span id='topic+getData.UflSnpInformation'></span><span id='topic+getData.UgpGenomeInformation'></span><span id='topic+getDataColumns'></span><span id='topic+getDataColumns.GenomeInformation'></span><span id='topic+getDataColumns.UflSnpInformation'></span><span id='topic+getDataColumns.UgpGenomeInformation'></span><span id='topic+getDataFlat'></span><span id='topic+getDataFlat.ChipEffectFile'></span><span id='topic+getDataFlat.FirmaFile'></span><span id='topic+getDataFlat.ResidualFile'></span><span id='topic+getDataFlat.WeightsFile'></span><span id='topic+getDataPairs'></span><span id='topic+getDataPairs.AllelicCrosstalkCalibration'></span><span id='topic+getDataSet'></span><span id='topic+getDataSet.FirmaModel'></span><span id='topic+getDataSet.QualityAssessmentModel'></span><span id='topic+getDefaultColumnNames.AromaCellMatchScoreFile'></span><span id='topic+getDefaultColumnNames.AromaUfcFile'></span><span id='topic+getDefaultExtension.AffymetrixCdfFile'></span><span id='topic+getDefaultExtension.AffymetrixCsvFile'></span><span id='topic+getDefaultExtension.AffymetrixCsvGenomeInformation'></span><span id='topic+getDefaultExtension.AffymetrixNetAffxCsvFile'></span><span id='topic+getDefaultExtension.AffymetrixPgfFile'></span><span id='topic+getDefaultExtension.AffymetrixTsvFile'></span><span id='topic+getDefaultExtension.AromaCellMatchScoreFile'></span><span id='topic+getDefaultExtension.AromaChipTypeAnnotationFile'></span><span id='topic+getDefaultFullName.AffymetrixFileSet'></span><span id='topic+getDefaultFullName.CnagCfhSet'></span><span id='topic+getDefaultSettings'></span><span id='topic+getDefaultSettings.AromaAffymetrix'></span><span id='topic+getDefaultSettings.Package'></span><span id='topic+getDesignMatrix'></span><span id='topic+getDesignMatrix.BaseCountNormalization'></span><span id='topic+getDesignMatrix.BasePositionNormalization'></span><span id='topic+getDesignMatrix.LinearModelProbeSequenceNormalization'></span><span id='topic+getDesignMatrix.MatNormalization'></span><span id='topic+getDimension'></span><span id='topic+getDimension.AffymetrixCdfFile'></span><span id='topic+getDimension.AffymetrixPgfFile'></span><span id='topic+getExclCells'></span><span id='topic+getExclCells.DChipQuantileNormalization'></span><span id='topic+getExclCells.QuantileNormalization'></span><span id='topic+getExcludes'></span><span id='topic+getExcludes.DChipDcpFile'></span><span id='topic+getExclusiveUnitSubsets'></span><span id='topic+getExclusiveUnitSubsets.FragmentEquivalentClassNormalization'></span><span id='topic+getExpandedCellMap'></span><span id='topic+getExpandedCellMap.ChipEffectFile'></span><span id='topic+getExpandedCellMap.CnChipEffectFile'></span><span id='topic+getExpandedCellMap.SnpChipEffectFile'></span><span id='topic+getExpectedOutputFullnames.MatSmoothing'></span><span id='topic+getExtensionPattern.AffymetrixCdfFile'></span><span id='topic+getExtensionPattern.AffymetrixCelFile'></span><span id='topic+getExtensionPattern.AffymetrixCnChpFile'></span><span id='topic+getExtensionPattern.AffymetrixCsvFile'></span><span id='topic+getExtensionPattern.AffymetrixPgfFile'></span><span id='topic+getExtensionPattern.AffymetrixTsvFile'></span><span id='topic+getExtensionPattern.AromaCellMatchScoreFile'></span><span id='topic+getExtensionPattern.AromaUfcFile'></span><span id='topic+getExtensionPattern.CnagCfhFile'></span><span id='topic+getExtensionPattern.DChipDcpFile'></span><span id='topic+getField'></span><span id='topic+getField.AffymetrixTsvFile'></span><span id='topic+getFields.SnpInformation'></span><span id='topic+getFields.UflSnpInformation'></span><span id='topic+getFileClass.FirmaSet'></span><span id='topic+getFileFormat'></span><span id='topic+getFileFormat.AffymetrixCdfFile'></span><span id='topic+getFileFormat.AffymetrixCelFile'></span><span id='topic+getFileFormat.AffymetrixCnChpFile'></span><span id='topic+getFileFormat.DChipCdfBinFile'></span><span id='topic+getFileFormat.DChipDcpFile'></span><span id='topic+getFileSet'></span><span id='topic+getFileSet.AffymetrixFileSetReporter'></span><span id='topic+getFileSetClass'></span><span id='topic+getFileSetClass.FirmaModel'></span><span id='topic+getFilenameExtension.AromaCellMatchScoreFile'></span><span id='topic+getFilenameExtension.AromaUfcFile'></span><span id='topic+getFilteredFragmentLengths'></span><span id='topic+getFilteredFragmentLengths.FragmentLengthNormalization'></span><span id='topic+getFirmaScores'></span><span id='topic+getFirmaSet'></span><span id='topic+getFirstCellIndices'></span><span id='topic+getFirstCellIndices.AffymetrixCdfFile'></span><span id='topic+getFirstCellPerUnitIndices'></span><span id='topic+getFirstCellPerUnitIndices.ExonChipEffectSet'></span><span id='topic+getFit'></span><span id='topic+getFit.BasePositionNormalization'></span><span id='topic+getFitSingleCellUnitFunction'></span><span id='topic+getFitSingleCellUnitFunction.UnitModel'></span><span id='topic+getFitUnitFunction'></span><span id='topic+getFitUnitFunction.CnPlm'></span><span id='topic+getFitUnitFunction.FirmaModel'></span><span id='topic+getFitUnitFunction.MultiArrayUnitModel'></span><span id='topic+getFitUnitFunction.SingleArrayUnitModel'></span><span id='topic+getFitUnitFunction.UnitModel'></span><span id='topic+getFitUnitGroupFunction'></span><span id='topic+getFitUnitGroupFunction.SmoothMultiarrayModel'></span><span id='topic+getFitUnitGroupFunction.SmoothRmaModel'></span><span id='topic+getFitUnitGroupFunction.SmoothSaModel'></span><span id='topic+getFragmentLengths'></span><span id='topic+getFragmentLengths.AffymetrixTsvFile'></span><span id='topic+getFragmentLengths.SnpInformation'></span><span id='topic+getFragmentLengths.UflSnpInformation'></span><span id='topic+getFragmentStarts'></span><span id='topic+getFragmentStarts.SnpInformation'></span><span id='topic+getFragmentStarts.UflSnpInformation'></span><span id='topic+getFragmentStops'></span><span id='topic+getFragmentStops.SnpInformation'></span><span id='topic+getFragmentStops.UflSnpInformation'></span><span id='topic+getFullName.GenericReporter'></span><span id='topic+getFullName.QualityAssessmentModel'></span><span id='topic+getFullNames.ChipEffectSetTuple'></span><span id='topic+getGc'></span><span id='topic+getGc.AffymetrixTsvFile'></span><span id='topic+getGcContent'></span><span id='topic+getGcContent.GcContentNormalization'></span><span id='topic+getGenomeBuild'></span><span id='topic+getGenomeBuild.AffymetrixNetAffxCsvFile'></span><span id='topic+getGenomeInformation'></span><span id='topic+getGroupDirections'></span><span id='topic+getGroupDirections.AffymetrixCdfFile'></span><span id='topic+getHeader.AffymetrixCnChpFile'></span><span id='topic+getHeader.DChipCdfBinFile'></span><span id='topic+getHeader.DChipDcpFile'></span><span id='topic+getHeaderAttributes'></span><span id='topic+getHeaderAttributes.AffymetrixNetAffxCsvFile'></span><span id='topic+getHeaderLength'></span><span id='topic+getHeaderLength.CnagCfhFile'></span><span id='topic+getHeaderV3'></span><span id='topic+getHeaderV3.AffymetrixCelFile'></span><span id='topic+getIdentifier'></span><span id='topic+getIdentifier.AffymetrixCelFile'></span><span id='topic+getIdentifier.AffymetrixCelSet'></span><span id='topic+getIdentifier.AffymetrixFileSet'></span><span id='topic+getIdentifier.CnagCfhFile'></span><span id='topic+getIdentifier.CnagCfhSet'></span><span id='topic+getImage.AffymetrixCdfFile'></span><span id='topic+getImage.AromaUnitTabularBinaryFile'></span><span id='topic+getImage.ResidualFile'></span><span id='topic+getImage.WeightsFile'></span><span id='topic+getIndexToRowMap'></span><span id='topic+getIndexToRowMap.AffymetrixProbeTabFile'></span><span id='topic+getInputName'></span><span id='topic+getInputName.AffymetrixFileSetReporter'></span><span id='topic+getInputName.GenericReporter'></span><span id='topic+getInputTags'></span><span id='topic+getInputTags.AffymetrixFileSetReporter'></span><span id='topic+getInputTags.GenericReporter'></span><span id='topic+getIntensities'></span><span id='topic+getLabel.Model'></span><span id='topic+getListOfCdfs'></span><span id='topic+getListOfCdfs.AffymetrixCelSetTuple'></span><span id='topic+getListOfGenomeInformations'></span><span id='topic+getListOfGenomeInformations.ChromosomalModel'></span><span id='topic+getListOfPriors'></span><span id='topic+getListOfPriors.MultiArrayUnitModel'></span><span id='topic+getListOfReporters'></span><span id='topic+getListOfReporters.ArrayExplorer'></span><span id='topic+getListOfUnitNamesFiles.AffymetrixCelSetTuple'></span><span id='topic+getListOfUnitNamesFiles.ArrayExplorer'></span><span id='topic+getListOfUnitTypesFiles.AffymetrixCelSetTuple'></span><span id='topic+getListOfUnitTypesFiles.ArrayExplorer'></span><span id='topic+getMainCdf'></span><span id='topic+getMainCdf.AffymetrixCdfFile'></span><span id='topic+getMainPath.GenericReporter'></span><span id='topic+getMergeFunction'></span><span id='topic+getMergeFunction.ChipEffectGroupMerge'></span><span id='topic+getMergeFunction.SnpChipEffectGroupMerge'></span><span id='topic+getMergeGroups'></span><span id='topic+getMergeGroups.ExonChipEffectSet'></span><span id='topic+getMergeStrands'></span><span id='topic+getMergeStrands.SnpChipEffectSet'></span><span id='topic+getMergeStrands.SnpPlm'></span><span id='topic+getMonocellCdf'></span><span id='topic+getMonocellCdf.AffymetrixCdfFile'></span><span id='topic+getName.AffymetrixPlatform'></span><span id='topic+getName.FirmaModel'></span><span id='topic+getName.QualityAssessmentModel'></span><span id='topic+getNameOfInput.ArrayExplorer'></span><span id='topic+getNetAffxBuild'></span><span id='topic+getNetAffxBuild.AffymetrixNetAffxCsvFile'></span><span id='topic+getNetAffxDate'></span><span id='topic+getNetAffxDate.AffymetrixNetAffxCsvFile'></span><span id='topic+getNormalEquations'></span><span id='topic+getNormalEquations.LinearModelProbeSequenceNormalization'></span><span id='topic+getNormalizeSignalsOne'></span><span id='topic+getNormalizeSignalsOne.ScaleNormalization3'></span><span id='topic+getNormalizeSignalsOne.UnitTypeScaleNormalization'></span><span id='topic+getNormalizedIntensities'></span><span id='topic+getNormalizedIntensities.DChipDcpFile'></span><span id='topic+getNumberOfFilesAveraged.CnChipEffectFile'></span><span id='topic+getOrderedFragmentPairMap'></span><span id='topic+getOrderedFragmentPairMap.AromaUfcFile'></span><span id='topic+getOrderedFragmentPairs'></span><span id='topic+getOrderedFragmentPairs.AromaUfcFile'></span><span id='topic+getOutputDataSet00'></span><span id='topic+getOutputDataSet00.AdditiveCovariatesNormalization'></span><span id='topic+getOutputDataSet00.FragmentEquivalentClassNormalization'></span><span id='topic+getOutputDataSet00.FragmentLengthNormalization'></span><span id='topic+getOutputDataSet00.GcContentNormalization'></span><span id='topic+getOutputFiles.Transform'></span><span id='topic+getOutputIdentifier'></span><span id='topic+getOutputIdentifier.QuantileNormalization'></span><span id='topic+getOutputTuple'></span><span id='topic+getOutputTuple.SmoothMultiarrayModel'></span><span id='topic+getParameter'></span><span id='topic+getParameter.AffymetrixAptSummaryFile'></span><span id='topic+getParameters.AbstractProbeSequenceNormalization'></span><span id='topic+getParameters.AdditiveCovariatesNormalization'></span><span id='topic+getParameters.AllelicCrosstalkCalibration'></span><span id='topic+getParameters.AvgPlm'></span><span id='topic+getParameters.BackgroundCorrection'></span><span id='topic+getParameters.BaseCountNormalization'></span><span id='topic+getParameters.BasePositionNormalization'></span><span id='topic+getParameters.ChipEffectFile'></span><span id='topic+getParameters.ChipEffectGroupMerge'></span><span id='topic+getParameters.CnChipEffectFile'></span><span id='topic+getParameters.CnPlm'></span><span id='topic+getParameters.CrlmmModel'></span><span id='topic+getParameters.DChipQuantileNormalization'></span><span id='topic+getParameters.ExonChipEffectFile'></span><span id='topic+getParameters.ExonRmaPlm'></span><span id='topic+getParameters.FragmentEquivalentClassNormalization'></span><span id='topic+getParameters.FragmentLengthNormalization'></span><span id='topic+getParameters.GcContentNormalization'></span><span id='topic+getParameters.GcRmaBackgroundCorrection'></span><span id='topic+getParameters.LimmaBackgroundCorrection'></span><span id='topic+getParameters.MatNormalization'></span><span id='topic+getParameters.MatSmoothing'></span><span id='topic+getParameters.NormExpBackgroundCorrection'></span><span id='topic+getParameters.OpticalBackgroundCorrection'></span><span id='topic+getParameters.ProbeAffinityFile'></span><span id='topic+getParameters.ProbeLevelTransform3'></span><span id='topic+getParameters.QuantileNormalization'></span><span id='topic+getParameters.ReseqCrosstalkCalibration'></span><span id='topic+getParameters.ResidualFile'></span><span id='topic+getParameters.ResidualSet'></span><span id='topic+getParameters.RmaBackgroundCorrection'></span><span id='topic+getParameters.RmaPlm'></span><span id='topic+getParameters.ScaleNormalization'></span><span id='topic+getParameters.ScaleNormalization3'></span><span id='topic+getParameters.SnpChipEffectFile'></span><span id='topic+getParameters.SnpChipEffectGroupMerge'></span><span id='topic+getParameters.SnpPlm'></span><span id='topic+getParameters.SpatialRowColumnNormalization'></span><span id='topic+getParameters.UnitModel'></span><span id='topic+getParameters.UnitTypeScaleNormalization'></span><span id='topic+getParameters.WeightsFile'></span><span id='topic+getParameters.WeightsSet'></span><span id='topic+getPath.AffymetrixCelSetReporter'></span><span id='topic+getPath.GenericReporter'></span><span id='topic+getPath.QualityAssessmentModel'></span><span id='topic+getPcuTheta'></span><span id='topic+getPcuTheta.ChromosomalModel'></span><span id='topic+getPlasqTypes'></span><span id='topic+getPlatform.AffymetrixCdfFile'></span><span id='topic+getPlatform.AffymetrixCelFile'></span><span id='topic+getPlatform.AffymetrixCelSet'></span><span id='topic+getPlatform.AffymetrixPgfFile'></span><span id='topic+getPlatform.AromaChipTypeAnnotationFile'></span><span id='topic+getPlatform.DChipCdfBinFile'></span><span id='topic+getPlatformDesignDB'></span><span id='topic+getPlatformDesignDB.CrlmmModel'></span><span id='topic+getPlm'></span><span id='topic+getPlm.FirmaModel'></span><span id='topic+getPlm.QualityAssessmentModel'></span><span id='topic+getPosition.AffymetrixTsvFile'></span><span id='topic+getPositionChipTypeUnit'></span><span id='topic+getPositionChipTypeUnit.ChromosomalModel'></span><span id='topic+getPositionChipTypeUnit.CopyNumberSegmentationModel'></span><span id='topic+getProbeAffinityFile'></span><span id='topic+getProbeAffinityFile.AffinePlm'></span><span id='topic+getProbeAffinityFile.AvgPlm'></span><span id='topic+getProbeAffinityFile.CnPlm'></span><span id='topic+getProbeAffinityFile.ExonRmaPlm'></span><span id='topic+getProbeAffinityFile.MbeiPlm'></span><span id='topic+getProbeAffinityFile.RmaPlm'></span><span id='topic+getProbeAffinityFile.SnpPlm'></span><span id='topic+getProbeSequenceData'></span><span id='topic+getProbeSequenceData.AffymetrixCdfFile'></span><span id='topic+getProbesetIds'></span><span id='topic+getProbesetIds.AffymetrixAptSummaryFile'></span><span id='topic+getQuantificationScale'></span><span id='topic+getQuantificationScale.AffymetrixAptSummaryFile'></span><span id='topic+getRawIntensities'></span><span id='topic+getRawIntensities.DChipDcpFile'></span><span id='topic+getReadArguments.AffymetrixAptSummaryFile'></span><span id='topic+getRectangle'></span><span id='topic+getRectangle.AffymetrixCelFile'></span><span id='topic+getReference.SpatialReporter'></span><span id='topic+getRegions.profileCGH'></span><span id='topic+getReportSet'></span><span id='topic+getReportSet.GenericReporter'></span><span id='topic+getReportSet.SpatialReporter'></span><span id='topic+getResidualFileClass'></span><span id='topic+getResidualFileClass.ResidualSet'></span><span id='topic+getResidualSet'></span><span id='topic+getResidualSet.ProbeLevelModel'></span><span id='topic+getResidualSetClass'></span><span id='topic+getResidualSetClass.ProbeLevelModel'></span><span id='topic+getResiduals'></span><span id='topic+getRestructor'></span><span id='topic+getRestructor.AffymetrixCdfFile'></span><span id='topic+getRlmFitFunctions'></span><span id='topic+getRlmFitFunctions.RmaPlm'></span><span id='topic+getRootPath.AlleleSummation'></span><span id='topic+getRootPath.ChipEffectTransform'></span><span id='topic+getRootPath.CrlmmModel'></span><span id='topic+getRootPath.FirmaModel'></span><span id='topic+getRootPath.GenericReporter'></span><span id='topic+getRootPath.ProbeLevelModel'></span><span id='topic+getRootPath.ProbeLevelTransform'></span><span id='topic+getRootPath.QualityAssessmentModel'></span><span id='topic+getRootPath.SmoothMultiarrayModel'></span><span id='topic+getRootPath.SmoothRmaModel'></span><span id='topic+getRootPath.TransformReport'></span><span id='topic+getSetTuple.ArrayExplorer'></span><span id='topic+getSetsOfProbes'></span><span id='topic+getSetsOfProbes.AllelicCrosstalkCalibration'></span><span id='topic+getSetsOfProbes.ReseqCrosstalkCalibration'></span><span id='topic+getSettings'></span><span id='topic+getSettings.Package'></span><span id='topic+getSignalTransform'></span><span id='topic+getSignalTransform.BasePositionNormalization'></span><span id='topic+getSignalTransform.LinearModelProbeSequenceNormalization'></span><span id='topic+getSnpInformation'></span><span id='topic+getSnpInformation.AffymetrixCdfFile'></span><span id='topic+getSnpNames'></span><span id='topic+getSpar'></span><span id='topic+getSpar.SpatialRowColumnNormalization'></span><span id='topic+getSteps'></span><span id='topic+getSteps.ASCRMAv2'></span><span id='topic+getSteps.AromaPipeline'></span><span id='topic+getSubsetOfCellIndices'></span><span id='topic+getSubsetOfCellIndices.AffymetrixCdfFile'></span><span id='topic+getSubsetOfUnits'></span><span id='topic+getSubsetOfUnits.AffymetrixCdfFile'></span><span id='topic+getSubsetToAvg'></span><span id='topic+getSubsetToAvg.AllelicCrosstalkCalibration'></span><span id='topic+getSubsetToAvg.QuantileNormalization'></span><span id='topic+getSubsetToAvg.ReseqCrosstalkCalibration'></span><span id='topic+getSubsetToAvg.ScaleNormalization'></span><span id='topic+getSubsetToFit'></span><span id='topic+getSubsetToFit.AdditiveCovariatesNormalization'></span><span id='topic+getSubsetToFit.FragmentEquivalentClassNormalization'></span><span id='topic+getSubsetToFit.FragmentLengthNormalization'></span><span id='topic+getSubsetToFit.GcContentNormalization'></span><span id='topic+getSubsetToUpdate'></span><span id='topic+getSubsetToUpdate.BackgroundCorrection'></span><span id='topic+getSubsetToUpdate.DChipQuantileNormalization'></span><span id='topic+getSubsetToUpdate.QuantileNormalization'></span><span id='topic+getSubsetToUpdate.ScaleNormalization'></span><span id='topic+getSubsetToUpdate0'></span><span id='topic+getSubsetToUpdate0.LimmaBackgroundCorrection'></span><span id='topic+getTags.FirmaModel'></span><span id='topic+getTags.QualityAssessmentModel'></span><span id='topic+getTagsOfInput.ArrayExplorer'></span><span id='topic+getTargetAverages'></span><span id='topic+getTargetAverages.FragmentEquivalentClassNormalization'></span><span id='topic+getTargetDistribution'></span><span id='topic+getTargetDistribution.QuantileNormalization'></span><span id='topic+getTargetDistributionIdentifier'></span><span id='topic+getTargetDistributionIdentifier.QuantileNormalization'></span><span id='topic+getTargetDistributionPathname'></span><span id='topic+getTargetDistributionPathname.QuantileNormalization'></span><span id='topic+getTargetFile'></span><span id='topic+getTargetFile.AbstractProbeSequenceNormalization'></span><span id='topic+getTargetFunction'></span><span id='topic+getTargetFunction.GcContentNormalization'></span><span id='topic+getTargetFunctions'></span><span id='topic+getTargetFunctions.AdditiveCovariatesNormalization'></span><span id='topic+getTargetFunctions.FragmentLengthNormalization'></span><span id='topic+getThetaStds'></span><span id='topic+getThetaStds.DChipDcpFile'></span><span id='topic+getThetas'></span><span id='topic+getThetas.DChipDcpFile'></span><span id='topic+getThetasAB'></span><span id='topic+getThetasAB.DChipDcpFile'></span><span id='topic+getTimeStamp'></span><span id='topic+getTimeStamp.AffymetrixAptSummaryFile'></span><span id='topic+getTimestamp'></span><span id='topic+getTimestamp.AffymetrixCnChpFile'></span><span id='topic+getTimestamp.CnagCfhFile'></span><span id='topic+getTimestamps'></span><span id='topic+getTimestamps.AffymetrixCelSet'></span><span id='topic+getTimestamps.CnagCfhSet'></span><span id='topic+getUniqueCdf'></span><span id='topic+getUniqueCdf.AffymetrixCdfFile'></span><span id='topic+getUnitGroupCellArrayMap'></span><span id='topic+getUnitGroupCellArrayMap.ChipEffectFile'></span><span id='topic+getUnitGroupCellChromosomePositionMap'></span><span id='topic+getUnitGroupCellChromosomePositionMap.AffymetrixCdfFile'></span><span id='topic+getUnitGroupCellChromosomePositionMap.ChipEffectFile'></span><span id='topic+getUnitGroupCellMap'></span><span id='topic+getUnitGroupCellMap.AffymetrixCdfFile'></span><span id='topic+getUnitGroupCellMap.AffymetrixCelSet'></span><span id='topic+getUnitGroupCellMap.FirmaFile'></span><span id='topic+getUnitGroupCellMap.ResidualFile'></span><span id='topic+getUnitGroupCellMap.WeightsFile'></span><span id='topic+getUnitGroupCellMapWithMonocell'></span><span id='topic+getUnitGroupCellMapWithMonocell.AffymetrixCdfFile'></span><span id='topic+getUnitGroupCellMapWithUnique'></span><span id='topic+getUnitGroupCellMapWithUnique.AffymetrixCdfFile'></span><span id='topic+getUnitGroupCellMatrixMap'></span><span id='topic+getUnitGroupCellMatrixMap.ChipEffectFile'></span><span id='topic+getUnitGroupNamesFromUgcMap'></span><span id='topic+getUnitGroupNamesFromUgcMap.AffymetrixCdfFile'></span><span id='topic+getUnitIndices'></span><span id='topic+getUnitIntensities'></span><span id='topic+getUnitNames.AffymetrixNetAffxCsvFile'></span><span id='topic+getUnitNames.DChipCdfBinFile'></span><span id='topic+getUnitNamesFile.AffymetrixCdfFile'></span><span id='topic+getUnitNamesFile.AffymetrixCelFile'></span><span id='topic+getUnitNamesFile.AffymetrixCelSet'></span><span id='topic+getUnitNamesFile.AffymetrixCnChpFile'></span><span id='topic+getUnitNamesFile.AffymetrixPgfFile'></span><span id='topic+getUnitNamesFile.AffymetrixPlatform'></span><span id='topic+getUnitNamesFile.TransformReport'></span><span id='topic+getUnitReadMap'></span><span id='topic+getUnitReadMap.AffymetrixCnChpFile'></span><span id='topic+getUnitSizes'></span><span id='topic+getUnitSizes.DChipCdfBinFile'></span><span id='topic+getUnitTypesFile.AffymetrixCdfFile'></span><span id='topic+getUnitTypesFile.AffymetrixCelFile'></span><span id='topic+getUnitTypesFile.AffymetrixCelSet'></span><span id='topic+getUnitTypesFile.AffymetrixCnChpFile'></span><span id='topic+getUnitTypesFile.AffymetrixPlatform'></span><span id='topic+getUnitTypesFile.TransformReport'></span><span id='topic+getUnitsOnChromosome.GenomeInformation'></span><span id='topic+getUnitsOnChromosome.UgpGenomeInformation'></span><span id='topic+getUnitsOnChromosomes.GenomeInformation'></span><span id='topic+getUnitsTo'></span><span id='topic+getUnitsTo.ProbeLevelTransform3'></span><span id='topic+getUnitsToFit'></span><span id='topic+getUnitsToFit.CrlmmModel'></span><span id='topic+getUnitsToFit.ProbeLevelTransform3'></span><span id='topic+getUnitsToUpdate'></span><span id='topic+getUnitsToUpdate.ProbeLevelTransform3'></span><span id='topic+getWeightsFileClass'></span><span id='topic+getWeightsFileClass.WeightsSet'></span><span id='topic+getWeightsSet'></span><span id='topic+getWeightsSet.ProbeLevelModel'></span><span id='topic+getWeightsSetClass'></span><span id='topic+getWeightsSetClass.ProbeLevelModel'></span><span id='topic+getYY'></span><span id='topic+getYY.TransformReport'></span><span id='topic+groupUnitsByDimension'></span><span id='topic+hasAlleleBFractions.CnChipEffectFile'></span><span id='topic+hasAlleleBFractions.CnChipEffectSet'></span><span id='topic+hasAlleleBFractions.CnChipEffectSetTuple'></span><span id='topic+hasColumnHeader.AffymetrixProbeTabFile'></span><span id='topic+hasMbeiData'></span><span id='topic+hasMbeiData.DChipDcpFile'></span><span id='topic+hasNormalizedData'></span><span id='topic+hasNormalizedData.DChipDcpFile'></span><span id='topic+hasStrandiness.CnChipEffectFile'></span><span id='topic+hasStrandiness.CnChipEffectSet'></span><span id='topic+hasStrandiness.CnChipEffectSetTuple'></span><span id='topic+hasUnitReadMap'></span><span id='topic+hasUnitReadMap.AffymetrixCnChpFile'></span><span id='topic+hasUnitTypes'></span><span id='topic+hasUnitTypes.AffymetrixCdfFile'></span><span id='topic+highlight'></span><span id='topic+highlight.AffymetrixCelFile'></span><span id='topic+identifyCells'></span><span id='topic+identifyCells.AffymetrixCdfFile'></span><span id='topic+image270'></span><span id='topic+importFrom.AromaUnitTabularBinaryFile'></span><span id='topic+importFromAffymetrixNetAffxCsvFile'></span><span id='topic+importFromAffymetrixNetAffxCsvFile.AromaUflFile'></span><span id='topic+importFromAffymetrixNetAffxCsvFile.AromaUgpFile'></span><span id='topic+importFromAffymetrixNetAffxCsvFile.AromaUnitGcContentFile'></span><span id='topic+importFromAffymetrixNetAffxCsvFile.AromaUnitTabularBinaryFile'></span><span id='topic+importFromAffymetrixProbeTabFile'></span><span id='topic+importFromAffymetrixProbeTabFile.AromaCellSequenceFile'></span><span id='topic+importFromAffymetrixTabularFile'></span><span id='topic+importFromAffymetrixTabularFile.AromaUfcFile'></span><span id='topic+importFromAffymetrixTabularFile.AromaUgpFile'></span><span id='topic+importFromAffymetrixTabularFile.AromaUnitTabularBinaryFile'></span><span id='topic+importFromApt'></span><span id='topic+importFromBpMap'></span><span id='topic+importFromBpMap.AromaCellMatchScoreFile'></span><span id='topic+importFromBpMap.AromaCellSequenceFile'></span><span id='topic+importFromDChip'></span><span id='topic+importFromDChipGenomeInformation'></span><span id='topic+importFromDChipGenomeInformation.AromaUnitTabularBinaryFile'></span><span id='topic+importFromGenericTabularFile.AromaUnitTabularBinaryFile'></span><span id='topic+importFromGenomeInformation'></span><span id='topic+importFromGenomeInformation.AromaUgpFile'></span><span id='topic+importFromGenomeInformation.AromaUnitTabularBinaryFile'></span><span id='topic+indexOfMissingSequences'></span><span id='topic+indexOfMissingSequences.AbstractProbeSequenceNormalization'></span><span id='topic+inferMmFromPm'></span><span id='topic+inferMmFromPm.AromaCellSequenceFile'></span><span id='topic+inferParameters'></span><span id='topic+inferParameters.CnChipEffectSet'></span><span id='topic+inferParameters.SnpChipEffectSet'></span><span id='topic+isCompatibleWithCdf'></span><span id='topic+isCompatibleWithCdf.GenomeInformation'></span><span id='topic+isCompatibleWithCdf.SnpInformation'></span><span id='topic+isCompatibleWithCdf.UflSnpInformation'></span><span id='topic+isCompatibleWithCdf.UgpGenomeInformation'></span><span id='topic+isDuplicated'></span><span id='topic+isMissing.AromaCellMatchScoreFile'></span><span id='topic+isMonocellCdf'></span><span id='topic+isMonocellCdf.AffymetrixCdfFile'></span><span id='topic+isPm'></span><span id='topic+isResequenceChip'></span><span id='topic+isSnpChip'></span><span id='topic+isUniqueCdf'></span><span id='topic+isUniqueCdf.AffymetrixCdfFile'></span><span id='topic+isUnitGroupCellMap'></span><span id='topic+isUnitGroupCellMap.data.frame'></span><span id='topic+isUnitGroupCellMap.default'></span><span id='topic+isUnitGroupCellMap.matrix'></span><span id='topic+justSNPRMA'></span><span id='topic+justSNPRMA.AffymetrixCelSet'></span><span id='topic+justSNPRMA.character'></span><span id='topic+mapToUnitNamesFile'></span><span id='topic+mapToUnitNamesFile.DChipCdfBinFile'></span><span id='topic+mergeGroups'></span><span id='topic+mergeGroups.ChipEffectFile'></span><span id='topic+mergeStrands'></span><span id='topic+mergeStrands.CnChipEffectFile'></span><span id='topic+mergeStrands.SnpChipEffectFile'></span><span id='topic+migrateTool'></span><span id='topic+migrateTool.CopyNumberSegmentationModel'></span><span id='topic+nbrOfArrays.CnagCfhSet'></span><span id='topic+nbrOfArrays.DChipDcpSet'></span><span id='topic+nbrOfArrays.AffymetrixCnChpSet'></span><span id='topic+nbrOfArrays.AffymetrixAptSummaryFile'></span><span id='topic+nbrOfArrays.AffymetrixCelSetReporter'></span><span id='topic+nbrOfArrays.QualityAssessmentModel'></span><span id='topic+nbrOfArrays.TransformReport'></span><span id='topic+nbrOfCells.AffymetrixCdfFile'></span><span id='topic+nbrOfCells.AffymetrixCelFile'></span><span id='topic+nbrOfCells.AffymetrixPgfFile'></span><span id='topic+nbrOfCells.CnagCfhFile'></span><span id='topic+nbrOfCells.DChipCdfBinFile'></span><span id='topic+nbrOfCells.DChipDcpFile'></span><span id='topic+nbrOfCellsPerUnit'></span><span id='topic+nbrOfCellsPerUnit.AffymetrixCdfFile'></span><span id='topic+nbrOfCellsPerUnitGroup'></span><span id='topic+nbrOfCellsPerUnitGroup.AffymetrixCdfFile'></span><span id='topic+nbrOfChipTypes.ArrayExplorer'></span><span id='topic+nbrOfColumns.AffymetrixCdfFile'></span><span id='topic+nbrOfColumns.AffymetrixPgfFile'></span><span id='topic+nbrOfEnzymes.AromaUfcFile'></span><span id='topic+nbrOfEnzymes.SnpInformation'></span><span id='topic+nbrOfEnzymes.UflSnpInformation'></span><span id='topic+nbrOfGroupsPerUnit'></span><span id='topic+nbrOfQcUnits'></span><span id='topic+nbrOfQcUnits.AffymetrixCdfFile'></span><span id='topic+nbrOfRows.AffymetrixCdfFile'></span><span id='topic+nbrOfRows.AffymetrixPgfFile'></span><span id='topic+nbrOfSnps'></span><span id='topic+nbrOfSnps.CnagCfhFile'></span><span id='topic+nbrOfSteps'></span><span id='topic+nbrOfSteps.AromaPipeline'></span><span id='topic+nbrOfUnits.AffymetrixCdfFile'></span><span id='topic+nbrOfUnits.AffymetrixPgfFile'></span><span id='topic+nbrOfUnits.DChipCdfBinFile'></span><span id='topic+nbrOfUnits.DChipDcpFile'></span><span id='topic+nbrOfUnits.GenomeInformation'></span><span id='topic+nbrOfUnits.SnpInformation'></span><span id='topic+nbrOfUnits.UflSnpInformation'></span><span id='topic+nbrOfUnits.UgpGenomeInformation'></span><span id='topic+normalizeOneArrayVector'></span><span id='topic+normalizeOneArrayVector.FragmentEquivalentClassNormalization'></span><span id='topic+normalizeQuantile'></span><span id='topic+plasqGaussian'></span><span id='topic+plotAllelePairs'></span><span id='topic+plotAllelePairs.AllelicCrosstalkCalibration'></span><span id='topic+plotBasepair'></span><span id='topic+plotBasepair.AllelicCrosstalkCalibration'></span><span id='topic+plotBoxplot'></span><span id='topic+plotBoxplot.ChipEffectSet'></span><span id='topic+plotBoxplotStats'></span><span id='topic+plotCovariateEffects'></span><span id='topic+plotCovariateEffects.GcContentNormalization2'></span><span id='topic+plotDensity'></span><span id='topic+plotImage'></span><span id='topic+plotMargins'></span><span id='topic+plotMargins.SpatialReporter'></span><span id='topic+plotMvsA'></span><span id='topic+plotMvsX'></span><span id='topic+plotNuse'></span><span id='topic+plotNuse.QualityAssessmentModel'></span><span id='topic+plotRle'></span><span id='topic+plotRle.QualityAssessmentModel'></span><span id='topic+plotXYCurve'></span><span id='topic+plotXYCurve.TransformReport'></span><span id='topic+plotXYCurveLog2'></span><span id='topic+plotXYCurveLog2.TransformReport'></span><span id='topic+predictOne'></span><span id='topic+predictOne.AbstractProbeSequenceNormalization'></span><span id='topic+predictOne.BaseCountNormalization'></span><span id='topic+predictOne.LinearModelProbeSequenceNormalization'></span><span id='topic+predictOne.MatNormalization'></span><span id='topic+process.AlleleSummation'></span><span id='topic+process.AromaPipeline'></span><span id='topic+process.SpatialRowColumnNormalization'></span><span id='topic+range.AffymetrixCelFile'></span><span id='topic+range.AffymetrixCelSet'></span><span id='topic+range.CnagCfhFile'></span><span id='topic+range.CnagCfhSet'></span><span id='topic+read10K'></span><span id='topic+read10K.DChipSnpInformation'></span><span id='topic+read250K'></span><span id='topic+read250K.DChipSnpInformation'></span><span id='topic+read250KHg17'></span><span id='topic+read250KHg17.DChipGenomeInformation'></span><span id='topic+read50K'></span><span id='topic+read50K.DChipSnpInformation'></span><span id='topic+read50KHg17'></span><span id='topic+read50KHg17.DChipGenomeInformation'></span><span id='topic+readArrays'></span><span id='topic+readArrays.AffymetrixAptSummaryFile'></span><span id='topic+readCdfGroupStrands'></span><span id='topic+readCfhHeader'></span><span id='topic+readCfhHeader.default'></span><span id='topic+readCfhUnits'></span><span id='topic+readCfhUnits.default'></span><span id='topic+readCfnHeader'></span><span id='topic+readCfnHeader.default'></span><span id='topic+readCfnUnits'></span><span id='topic+readCfnUnits.default'></span><span id='topic+readDataFrame.AffymetrixCdfFile'></span><span id='topic+readDataFrame.AffymetrixCsvGenomeInformation'></span><span id='topic+readDataFrame.AffymetrixTsvFile'></span><span id='topic+readDataFrame.AromaUfcFile'></span><span id='topic+readDataFrame.DChipCdfBinFile'></span><span id='topic+readDataFrame.DChipGenomeInformation'></span><span id='topic+readDataFrame.DChipSnpInformation'></span><span id='topic+readDataFrame.GenomeInformation'></span><span id='topic+readDataFrame.SnpInformation'></span><span id='topic+readDataFrame.UflSnpInformation'></span><span id='topic+readDataFrame.UgpGenomeInformation'></span><span id='topic+readDataFrame2'></span><span id='topic+readDataFrame2.AffymetrixProbeTabFile'></span><span id='topic+readDataUnitChromosomePosition'></span><span id='topic+readDataUnitChromosomePosition.AffymetrixNetAffxCsvFile'></span><span id='topic+readDataUnitFragmentLength'></span><span id='topic+readDataUnitFragmentLength.AffymetrixNetAffxCsvFile'></span><span id='topic+readGeneAssignments'></span><span id='topic+readGeneAssignments.AffymetrixNetAffxCsvFile'></span><span id='topic+readGenomeWideSNP'></span><span id='topic+readGenomeWideSNP.DChipGenomeInformation'></span><span id='topic+readHeader.AffymetrixAptSummaryFile'></span><span id='topic+readMatchScores'></span><span id='topic+readMatchScores.AromaCellMatchScoreFile'></span><span id='topic+readMouse430'></span><span id='topic+readMouse430.DChipGenomeInformation'></span><span id='topic+readMouse430KenHardwired'></span><span id='topic+readMouse430KenHardwired.DChipGenomeInformation'></span><span id='topic+readParameter'></span><span id='topic+readParameter.CrlmmParametersFile'></span><span id='topic+readPlasqTypes'></span><span id='topic+readPlasqTypes.AffymetrixCdfFile'></span><span id='topic+readPriorsByUnits'></span><span id='topic+readPriorsByUnits.MultiArrayUnitModel'></span><span id='topic+readProbesetIds'></span><span id='topic+readProbesetIds.AffymetrixAptSummaryFile'></span><span id='topic+readRawData'></span><span id='topic+readRawData.AffymetrixCnChpFile'></span><span id='topic+readRawData.AffymetrixPgfFile'></span><span id='topic+readRawDataRectangle'></span><span id='topic+readRawDataRectangle.AffymetrixCelFile'></span><span id='topic+readRawDataRectangle.SpatialReporter'></span><span id='topic+readSequenceDataFrame'></span><span id='topic+readSequenceDataFrame.AffymetrixProbeTabFile'></span><span id='topic+readTableInternal'></span><span id='topic+readTableInternal.GenomeInformation'></span><span id='topic+readTableInternal.SnpInformation'></span><span id='topic+readUnitNames'></span><span id='topic+readUnitNames.AffymetrixNetAffxCsvFile'></span><span id='topic+readUnits'></span><span id='topic+readUnits.AffymetrixCelSet'></span><span id='topic+readUnits.ChipEffectFile'></span><span id='topic+readUnits.ChipEffectSet'></span><span id='topic+readUnits.CnChipEffectFile'></span><span id='topic+readUnits.CnagCfhSet'></span><span id='topic+readUnits.ExonChipEffectFile'></span><span id='topic+readUnits.FirmaFile'></span><span id='topic+readUnits.FirmaSet'></span><span id='topic+readUnits.ParameterCelFile'></span><span id='topic+readUnits.ProbeAffinityFile'></span><span id='topic+readUnits.ResidualFile'></span><span id='topic+readUnits.ResidualSet'></span><span id='topic+readUnits.SnpChipEffectFile'></span><span id='topic+readUnits.WeightsFile'></span><span id='topic+readUnits.WeightsSet'></span><span id='topic+readUnitsByQuartets'></span><span id='topic+rescale.AllelicCrosstalkCalibration'></span><span id='topic+rescaleByAll'></span><span id='topic+rescaleByAll.AllelicCrosstalkCalibration'></span><span id='topic+rescaleByAll.ReseqCrosstalkCalibration'></span><span id='topic+rescaleByGroups'></span><span id='topic+rescaleByGroups.AllelicCrosstalkCalibration'></span><span id='topic+restruct'></span><span id='topic+restruct.AffymetrixCdfFile'></span><span id='topic+seq.TransformReport'></span><span id='topic+setAlias.ArrayExplorer'></span><span id='topic+setAlias.TransformReport'></span><span id='topic+setBandwidth'></span><span id='topic+setBandwidth.SmoothMultiarrayModel'></span><span id='topic+setCdf'></span><span id='topic+setCdf.AffymetrixCnChpFile'></span><span id='topic+setCdf.AffymetrixCnChpSet'></span><span id='topic+setColorMaps'></span><span id='topic+setColorMaps.ArrayExplorer'></span><span id='topic+setColorMaps.SpatialReporter'></span><span id='topic+setCombineAlleles'></span><span id='topic+setCombineAlleles.CnChipEffectSet'></span><span id='topic+setCombineAlleles.CnPlm'></span><span id='topic+setCombineAlleles.CnProbeAffinityFile'></span><span id='topic+setCustomFindCdf'></span><span id='topic+setDecodeFunction'></span><span id='topic+setDecodeFunction.ParameterCelFile'></span><span id='topic+setEncodeFunction'></span><span id='topic+setEncodeFunction.ParameterCelFile'></span><span id='topic+setGenomeInformation'></span><span id='topic+setGenomeInformation.AffymetrixCdfFile'></span><span id='topic+setLabel.Model'></span><span id='topic+setListOfPriors'></span><span id='topic+setListOfPriors.MultiArrayUnitModel'></span><span id='topic+setMergeFunction'></span><span id='topic+setMergeFunction.ChipEffectGroupMerge'></span><span id='topic+setMergeGroups'></span><span id='topic+setMergeGroups.ExonChipEffectSet'></span><span id='topic+setMergeGroups.ExonProbeAffinityFile'></span><span id='topic+setMergeGroups.ExonRmaPlm'></span><span id='topic+setMergeStrands'></span><span id='topic+setMergeStrands.SnpChipEffectSet'></span><span id='topic+setMergeStrands.SnpPlm'></span><span id='topic+setMergeStrands.SnpProbeAffinityFile'></span><span id='topic+setReference.SpatialReporter'></span><span id='topic+setRestructor'></span><span id='topic+setSettings'></span><span id='topic+setSettings.Package'></span><span id='topic+setSnpInformation'></span><span id='topic+setSnpInformation.AffymetrixCdfFile'></span><span id='topic+setUnitReadMap'></span><span id='topic+setUnitReadMap.AffymetrixCnChpFile'></span><span id='topic+setup.GenericReporter'></span><span id='topic+smoothScatterMvsA'></span><span id='topic+stextChipType.AffymetrixCdfFile'></span><span id='topic+transformAffine'></span><span id='topic+transformAffine.AffymetrixCelFile'></span><span id='topic+transformAffine.AffymetrixCelSet'></span><span id='topic+translateColumnNames.AffymetrixAptSummaryFile'></span><span id='topic+translateColumnNames.AffymetrixProbeTabFile'></span><span id='topic+translateColumnNames.AffymetrixTabularFile'></span><span id='topic+translateFullName.AffymetrixProbeTabFile'></span><span id='topic+update.AromaAffymetrix'></span><span id='topic+update2.AffymetrixCelSet'></span><span id='topic+updateDataFlat'></span><span id='topic+updateDataFlat.ChipEffectFile'></span><span id='topic+updateDataFlat.FirmaFile'></span><span id='topic+updateDataFlat.ResidualFile'></span><span id='topic+updateDataFlat.WeightsFile'></span><span id='topic+updateMatchScores'></span><span id='topic+updateMatchScores.AromaCellMatchScoreFile'></span><span id='topic+updateParameter'></span><span id='topic+updateParameter.CrlmmParametersFile'></span><span id='topic+updateSampleAnnotationSet'></span><span id='topic+updateSampleAnnotationSet.AffymetrixCelSet'></span><span id='topic+updateSettings'></span><span id='topic+updateSettings.Package'></span><span id='topic+updateSetupExplorerFile.ArrayExplorer'></span><span id='topic+updateUnits'></span><span id='topic+updateUnits.ChipEffectFile'></span><span id='topic+updateUnits.ChipEffectSet'></span><span id='topic+updateUnits.FirmaFile'></span><span id='topic+updateUnits.FirmaSet'></span><span id='topic+updateUnits.ParameterCelFile'></span><span id='topic+updateUnits.ProbeAffinityFile'></span><span id='topic+updateUnits.ResidualFile'></span><span id='topic+updateUnits.ResidualSet'></span><span id='topic+updateUnits.WeightsFile'></span><span id='topic+updateUnits.WeightsSet'></span><span id='topic+validate.AffymetrixCdfFile'></span><span id='topic+validate.AvgPlm'></span><span id='topic+validate.MultiArrayUnitModel'></span><span id='topic+verify.AffymetrixCsvGenomeInformation'></span><span id='topic+verify.AffymetrixTsvFile'></span><span id='topic+verify.DChipGenomeInformation'></span><span id='topic+verify.DChipSnpInformation'></span><span id='topic+verify.UflSnpInformation'></span><span id='topic+verify.UgpGenomeInformation'></span><span id='topic+writeAsFullCelFile'></span><span id='topic+writeAsFullCelFile.ChipEffectFile'></span><span id='topic+writeCNT'></span><span id='topic+writeCNT.SnpChipEffectFile'></span><span id='topic+writeCNT.SnpChipEffectSet'></span><span id='topic+writeCdf'></span><span id='topic+writeCdf.default'></span><span id='topic+writeCdfByExcludingCells'></span><span id='topic+writeCdfByExcludingCells.AffymetrixCdfFile'></span><span id='topic+writeImage.ResidualFile'></span><span id='topic+writeImage.WeightsFile'></span><span id='topic+writeImageCombined'></span><span id='topic+writeImageCombined.TransformReport'></span><span id='topic+writeImages'></span><span id='topic+writeImages.SpatialReporter'></span><span id='topic+writeImages.TransformReport'></span><span id='topic+writeRegions.profileCGH'></span><span id='topic+writeSgr'></span><span id='topic+writeSgr.AffymetrixCelSet'></span><span id='topic+writeSignals'></span><span id='topic+writeSignals.ProbeLevelTransform3'></span><span id='topic+writeWig'></span><span id='topic+writeWig.CnChipEffectSet'></span>

<h3>Description</h3>

<p>This page contains aliases for all &quot;non-documented&quot; objects that
<code>R CMD check</code> detects in this package.
</p>
<p>Almost all of them are <em>generic</em> functions that have specific
document for the corresponding method coupled to a specific class.
Other functions are re-defined by <code>setMethodS3()</code> to
<em>default</em> methods. Neither of these two classes are non-documented
in reality.
The rest are deprecated methods.
</p>

<hr>
<h2 id='normalizeQuantile.AffymetrixCelFile'>Normalizes the probe intensities to a target empirical distribution</h2><span id='topic+normalizeQuantile.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.normalizeQuantile'></span><span id='topic+normalizeQuantile+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Normalizes the probe intensities to a target empirical distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelFile'
normalizeQuantile(this, path=file.path("normQuantile", getChipType(this)), xTarget,
  subsetToUpdate=NULL, typesToUpdate=NULL, ..., overwrite=FALSE, skip=!overwrite,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalizeQuantile.AffymetrixCelFile_+3A_path">path</code></td>
<td>
<p>The path where to save the normalized data files.</p>
</td></tr>
<tr><td><code id="normalizeQuantile.AffymetrixCelFile_+3A_xtarget">xTarget</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code>.  The empirical distribution
to which all arrays should be normalized to.</p>
</td></tr>
<tr><td><code id="normalizeQuantile.AffymetrixCelFile_+3A_subsettoupdate">subsetToUpdate</code></td>
<td>
<p>The indices of the probes to be updated.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all are updated.</p>
</td></tr>
<tr><td><code id="normalizeQuantile.AffymetrixCelFile_+3A_typestoupdate">typesToUpdate</code></td>
<td>
<p>Types of probes to be updated.  For more details,
see argument <code>types</code> of <code>identifyCells()</code> for the
<code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> class.</p>
</td></tr>
<tr><td><code id="normalizeQuantile.AffymetrixCelFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>normalizeQuantile()</code>.</p>
</td></tr>
<tr><td><code id="normalizeQuantile.AffymetrixCelFile_+3A_overwrite">overwrite</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, already normalized arrays are overwritten,
unless skipped, otherwise an error is thrown.</p>
</td></tr>
<tr><td><code id="normalizeQuantile.AffymetrixCelFile_+3A_skip">skip</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the array is not normalized if it already exists.</p>
</td></tr>
<tr><td><code id="normalizeQuantile.AffymetrixCelFile_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the normalized <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson, Ken Simpson</p>


<h3>See Also</h3>

<p><code><a href="aroma.light.html#topic+normalizeQuantile">normalizeQuantile</a></code>
For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>

<hr>
<h2 id='normalizeQuantile.AffymetrixCelSet'>Normalizes samples to have the same empirical distribution</h2><span id='topic+normalizeQuantile.AffymetrixCelSet'></span><span id='topic+AffymetrixCelSet.normalizeQuantile'></span><span id='topic+normalizeQuantile+2CAffymetrixCelSet-method'></span>

<h3>Description</h3>

<p>Normalizes samples to have the same empirical distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelSet'
normalizeQuantile(this, path=NULL, name="normQuantile", subsetToUpdate=NULL,
  typesToUpdate=NULL, xTarget=NULL, subsetToAvg=subsetToUpdate, typesToAvg=typesToUpdate,
  ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalizeQuantile.AffymetrixCelSet_+3A_path">path</code></td>
<td>
<p>The path where to save the normalized data files.
If <code><a href="base.html#topic+NULL">NULL</a></code>, a default name is used.</p>
</td></tr>
<tr><td><code id="normalizeQuantile.AffymetrixCelSet_+3A_name">name</code></td>
<td>
<p>The name of the normalized data set, which will also be
part of the default path.</p>
</td></tr>
<tr><td><code id="normalizeQuantile.AffymetrixCelSet_+3A_subsettoupdate">subsetToUpdate</code></td>
<td>
<p>The probes to be updated.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all probes are updated.</p>
</td></tr>
<tr><td><code id="normalizeQuantile.AffymetrixCelSet_+3A_typestoupdate">typesToUpdate</code></td>
<td>
<p>Types of probes to be updated.</p>
</td></tr>
<tr><td><code id="normalizeQuantile.AffymetrixCelSet_+3A_xtarget">xTarget</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code>.  The empirical distribution
to which all arrays should be normalized to.</p>
</td></tr>
<tr><td><code id="normalizeQuantile.AffymetrixCelSet_+3A_subsettoavg">subsetToAvg</code></td>
<td>
<p>The probes to calculate average empirical
distribution over.  If a single <code><a href="base.html#topic+numeric">numeric</a></code> in (0,1), then this
fraction of all probes will be used.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all probes are considered.</p>
</td></tr>
<tr><td><code id="normalizeQuantile.AffymetrixCelSet_+3A_typestoavg">typesToAvg</code></td>
<td>
<p>Types of probes to be used when calculating the
average empirical distribution.
If <code>"pm"</code> and <code>"mm"</code> only perfect-match and mismatch
probes are used, respectively. If <code>"pmmm"</code> both types are used.
</p>
</td></tr>
<tr><td><code id="normalizeQuantile.AffymetrixCelSet_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>normalizeQuantile()</code>.</p>
</td></tr>
<tr><td><code id="normalizeQuantile.AffymetrixCelSet_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="aroma.light.html#topic+normalizeQuantileRank.numeric">normalizeQuantileRank.numeric</a></code>
For more information see <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>

<hr>
<h2 id='NormExpBackgroundCorrection'>The NormExpBackgroundCorrection class</h2><span id='topic+NormExpBackgroundCorrection'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class NormExpBackgroundCorrection</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+BackgroundCorrection">BackgroundCorrection</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+LimmaBackgroundCorrection">LimmaBackgroundCorrection</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>NormExpBackgroundCorrection</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>NormExpBackgroundCorrection</b><br />
extends <a href="#topic+LimmaBackgroundCorrection">LimmaBackgroundCorrection</a><br />
</p>
<p>This class represents the normal exponential background correction model.
Estimators of the <span class="pkg">limma</span> package is used.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NormExpBackgroundCorrection(..., method=c("rma", "saddle", "mle"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NormExpBackgroundCorrection_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+LimmaBackgroundCorrection">LimmaBackgroundCorrection</a></code>.</p>
</td></tr>
<tr><td><code id="NormExpBackgroundCorrection_+3A_method">method</code></td>
<td>
<p>The estimator used, cf. argument <code>normexp.method</code>
of <code><a href="limma.html#topic+backgroundCorrect">backgroundCorrect</a></code> in <span class="pkg">limma</span> for more details.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from LimmaBackgroundCorrection</b>:<br />
getAsteriskTags, getParameters, process
</p>
<p><b>Methods inherited from BackgroundCorrection</b>:<br />
getParameters, process
</p>
<p><b>Methods inherited from ProbeLevelTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>Internally, <code><a href="limma.html#topic+backgroundCorrect">backgroundCorrect</a></code> is used.
</p>

<hr>
<h2 id='OpticalBackgroundCorrection'>The OpticalBackgroundCorrection class</h2><span id='topic+OpticalBackgroundCorrection'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class OpticalBackgroundCorrection</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+BackgroundCorrection">BackgroundCorrection</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>OpticalBackgroundCorrection</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>OpticalBackgroundCorrection</b><br />
extends <em><a href="#topic+BackgroundCorrection">BackgroundCorrection</a></em><br />
</p>
<p>This class represents &quot;optical&quot; background adjustment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OpticalBackgroundCorrection(..., minimum=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OpticalBackgroundCorrection_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code>.</p>
</td></tr>
<tr><td><code id="OpticalBackgroundCorrection_+3A_minimum">minimum</code></td>
<td>
<p>The minimum signal allowed after adjustment.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from BackgroundCorrection</b>:<br />
getParameters, process
</p>
<p><b>Methods inherited from ProbeLevelTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Ken Simpson</p>

<hr>
<h2 id='ParameterCelFile'>The ParameterCelFile class</h2><span id='topic+ParameterCelFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class ParameterCelFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>ParameterCelFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+ChipEffectFile">ChipEffectFile</a></em>, <em><a href="#topic+CnChipEffectFile">CnChipEffectFile</a></em>, <em><a href="#topic+CnProbeAffinityFile">CnProbeAffinityFile</a></em>, <em><a href="#topic+ExonChipEffectFile">ExonChipEffectFile</a></em>, <em><a href="#topic+ExonProbeAffinityFile">ExonProbeAffinityFile</a></em>, <em><a href="#topic+FirmaFile">FirmaFile</a></em>, <em><a href="#topic+ProbeAffinityFile">ProbeAffinityFile</a></em>, <em><a href="#topic+ResidualFile">ResidualFile</a></em>, <em><a href="#topic+SnpChipEffectFile">SnpChipEffectFile</a></em>, <em><a href="#topic+SnpProbeAffinityFile">SnpProbeAffinityFile</a></em>, <em><a href="#topic+WeightsFile">WeightsFile</a></em><br />
</p>
<p>public abstract static class <b>ParameterCelFile</b><br />
extends <a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a><br />
</p>
<p>A ParameterCelFile object represents parameter estimates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ParameterCelFile(..., encodeFunction=NULL, decodeFunction=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ParameterCelFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.</p>
</td></tr>
<tr><td><code id="ParameterCelFile_+3A_encodefunction">encodeFunction</code></td>
<td>
<p>A <code><a href="base.html#topic+function">function</a></code> taking a single <code><a href="base.html#topic+list">list</a></code> structure
as its argument.</p>
</td></tr>
<tr><td><code id="ParameterCelFile_+3A_decodefunction">decodeFunction</code></td>
<td>
<p>A <code><a href="base.html#topic+function">function</a></code> taking a single <code><a href="base.html#topic+list">list</a></code> structure
as its argument.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractDataFrame</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractMatrix</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from AffymetrixCelFile</b>:<br />
allocateFromCdf, as.character, clone, createFrom, extractMatrix, fromFile, getAm, getCdf, getExtensionPattern, getFileFormat, getImage, getUnitNamesFile, getUnitTypesFile, highlight, image270, nbrOfCells, plotDensity, plotImage, plotMvsA, plotMvsX, range, setCdf, smoothScatterMvsA, writeImage
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>File format</h3>

<p>The idea behind this class is store data fields which by nature have
one value per probe (per field) in CEL files.  A perfect example is to
store probe-affinity estimates and their standard deviations.  There
is one probe affinity per probe so the structure of a CEL file (and
its coupled CDF file) is well suited to read/write such information.
</p>
<p>Consider a unit group with L probes.  A CEL file stores
<code>intensities</code> (L floats), <code>stdvs</code> (L floats), and
<code>pixels</code> (L integers).  Thus, for each probe l=1,...,L, a
(float, float, integer) tuple is stored.  We can use this for any
information we want.  If we want a slightly different structure,
we can choose to encode/decode our structure/information to fit the
structure of the CEL file.  This abstract class provides transparent
methods for encoding and decoding such information through methods
<code>encodeUnitGroup()</code> and <code>decodeUnitGroup()</code>.
By subclassing you can implement different types of data structures.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='ParameterCelSet'>The ParameterCelSet class</h2><span id='topic+ParameterCelSet'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class ParameterCelSet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataSet">AromaMicroarrayDataSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>ParameterCelSet</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+ChipEffectSet">ChipEffectSet</a>, <a href="#topic+CnChipEffectSet">CnChipEffectSet</a>, <a href="#topic+ExonChipEffectSet">ExonChipEffectSet</a>, <a href="#topic+FirmaSet">FirmaSet</a>, <a href="#topic+SnpChipEffectSet">SnpChipEffectSet</a><br />
</p>
<p>public static class <b>ParameterCelSet</b><br />
extends <a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a><br />
</p>
<p>A ParameterCelSet object represents a set of <code><a href="#topic+ParameterCelFile">ParameterCelFile</a></code>:s.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ParameterCelSet(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ParameterCelSet_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractDataFrame</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractMatrix</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from AffymetrixCelSet</b>:<br />
append, as, as.AffymetrixCelSet, as.character, averageQuantile, byName, byPath, clone, convertToUnique, doCRMAv1, doCRMAv2, doFIRMA, doGCRMA, doRMA, extractAffyBatch, extractFeatureSet, extractMatrix, extractSnpFeatureSet, findByName, getAverage, getAverageAsinh, getAverageFile, getAverageLog, getCdf, getChipType, getData, getIntensities, getPlatform, getTimestamps, getUnitGroupCellMap, getUnitIntensities, getUnitNamesFile, getUnitTypesFile, isDuplicated, justRMA, justSNPRMA, nbrOfArrays, normalizeQuantile, plotDensity, range, readUnits, setCdf, update2, writeSgr
</p>
<p><b>Methods inherited from AffymetrixFileSet</b>:<br />
as, as.AffymetrixFileSet, byPath, getDefaultFullName
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataSet</b>:<br />
as.AromaMicroarrayDataSetList, as.AromaMicroarrayDataSetTuple, getAromaFullNameTranslatorSet, getAverageFile, getChipType, getDefaultFullName, getPlatform, setAttributesBy, setAttributesBySampleAnnotationFile, setAttributesBySampleAnnotationSet, validate
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='pdInfo2Cdf'>Generates an Affymetrix CDF file from a Platform Design (PD) package and a auxiliary CEL file for the same chip type</h2><span id='topic+pdInfo2Cdf'></span>

<h3>Description</h3>

<p>Generates an Affymetrix CDF file from a Platform Design (PD) package and a auxiliary CEL file for the same chip type.
Platform Design (PD) packages are also known as &quot;pdInfo&quot; packages.
</p>
<p><em>Disclaimer: This is a user-contributed function.</em>
</p>
<p><em>Instead of using this method, we recommend to use
<code><a href="#topic+writeCdf.AffyGenePDInfo">writeCdf</a>()</code>
for the <code>AffyGenePDInfo</code> class.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pdInfo2Cdf(pdpkg, celfile, overwrite=FALSE, verbose=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pdInfo2Cdf_+3A_pdpkg">pdpkg</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string for an existing PD package.</p>
</td></tr>
<tr><td><code id="pdInfo2Cdf_+3A_celfile">celfile</code></td>
<td>
<p>The pathname to an auxiliary CEL for the same chip type.</p>
</td></tr>
<tr><td><code id="pdInfo2Cdf_+3A_overwrite">overwrite</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, an existing CDF is overwritten, otherwise
an exception is thrown.</p>
</td></tr>
<tr><td><code id="pdInfo2Cdf_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
<tr><td><code id="pdInfo2Cdf_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns (invisibly) the pathname to CDF written.
The CDF filename is generated from the name of the PD package.
</p>


<h3>Limitations</h3>

<p>The information available in the PD package is limited and does
not contain all information needed to populate a CDF file.
In order to workaround these limitations, certain CDF entries
are set to predefined/hardwired values.
The 'pbase' and 'tbase' entries of the generated CDF file is
hardwired to &quot;T&quot; and &quot;A&quot;, respectively.  Likewise, the 'groupdirection'
entry is hardwired to &quot;sense&quot;.
</p>


<h3>Author(s)</h3>

<p>Maintained by Mark Robinson.
Original code by Samuel Wuest.
Code improvements by Henrik Bengtsson.
</p>


<h3>See Also</h3>

<p>Instead of using this method, we recommend to use
<code><a href="#topic+writeCdf.AffyGenePDInfo">writeCdf</a>()</code>
for the <code>AffyGenePDInfo</code> class.
</p>

<hr>
<h2 id='plotBoxplotStats.list'>Plots a (merged or non-merged) list of boxplot.stats() elements</h2><span id='topic+plotBoxplotStats.list'></span>

<h3>Description</h3>

<p>Plots a (merged or non-merged) list of boxplot.stats() elements.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'list'
plotBoxplotStats(stats, ylim=NULL, outline=FALSE, las=2, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotBoxplotStats.list_+3A_stats">stats</code></td>
<td>
<p>A (merged or non-merged) <code><a href="base.html#topic+list">list</a></code> of
<code><a href="grDevices.html#topic+boxplot.stats">boxplot.stats</a></code> elements.</p>
</td></tr>
<tr><td><code id="plotBoxplotStats.list_+3A_ylim">ylim</code>, <code id="plotBoxplotStats.list_+3A_outline">outline</code>, <code id="plotBoxplotStats.list_+3A_las">las</code>, <code id="plotBoxplotStats.list_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="graphics.html#topic+bxp">bxp</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns (invisibly) the merged <code><a href="grDevices.html#topic+boxplot.stats">boxplot.stats</a></code> structure.
</p>


<h3>Author(s)</h3>

<p>Elizabeth Purdom</p>

<hr>
<h2 id='plotDensity.AffymetrixCelFile'>Plots the density of the probe signals on the array</h2><span id='topic+plotDensity.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.plotDensity'></span><span id='topic+plotDensity+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Plots the density of the probe signals on the array.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelFile'
plotDensity(this, subset=NULL, types=NULL, ..., xlim=c(0, 16), xlab=NULL,
  ylab="density (integrates to one)", log=TRUE, annotate=TRUE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotDensity.AffymetrixCelFile_+3A_subset">subset</code></td>
<td>
<p>The subset of probes to considered <em>before</em> any
filtering by probe type is applied.
If a <code><a href="base.html#topic+vector">vector</a></code> of <code><a href="base.html#topic+double">double</a></code>s, the cell indices.
If a scalar <code><a href="base.html#topic+double">double</a></code> in [0,1], the fraction of cells, which can
be used to speed up the plotting if approximate densities are
acceptable.
if <code><a href="base.html#topic+NULL">NULL</a></code>, all cells are considered.
</p>
</td></tr>
<tr><td><code id="plotDensity.AffymetrixCelFile_+3A_types">types</code></td>
<td>
<p>The type of probes to include, e.g. <code>"all"</code>,
<code>"pmmm"</code>, <code>"pm"</code>, and <code>"mm"</code>.</p>
</td></tr>
<tr><td><code id="plotDensity.AffymetrixCelFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to
<code><a href="aroma.light.html#topic+plotDensity.numeric">plotDensity.numeric</a></code>.</p>
</td></tr>
<tr><td><code id="plotDensity.AffymetrixCelFile_+3A_xlim">xlim</code></td>
<td>
<p>The range on the x axis.</p>
</td></tr>
<tr><td><code id="plotDensity.AffymetrixCelFile_+3A_xlab">xlab</code>, <code id="plotDensity.AffymetrixCelFile_+3A_ylab">ylab</code></td>
<td>
<p>The labels on the x and the y axes.</p>
</td></tr>
<tr><td><code id="plotDensity.AffymetrixCelFile_+3A_log">log</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the density of the log (base 2) values are
used, otherwise the non-logged values.</p>
</td></tr>
<tr><td><code id="plotDensity.AffymetrixCelFile_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns nothing.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>

<hr>
<h2 id='plotDensity.AffymetrixCelSet'>Plots the densities of all samples</h2><span id='topic+plotDensity.AffymetrixCelSet'></span><span id='topic+AffymetrixCelSet.plotDensity'></span><span id='topic+plotDensity+2CAffymetrixCelSet-method'></span>

<h3>Description</h3>

<p>Plots the densities of all samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelSet'
plotDensity(this, subset=NULL, types=NULL, ..., col=seq_along(this), lty=NULL, lwd=NULL,
  annotate=TRUE, add=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotDensity.AffymetrixCelSet_+3A_subset">subset</code></td>
<td>
<p>The subset of probes to considered <em>before</em> any
filtering by probe type is applied.
If a <code><a href="base.html#topic+vector">vector</a></code> of <code><a href="base.html#topic+double">double</a></code>s, the cell indices.
If a scalar <code><a href="base.html#topic+double">double</a></code> in [0,1], the fraction of cells, which can
be used to speed up the plotting if approximate densities are
acceptable.
if <code><a href="base.html#topic+NULL">NULL</a></code>, all cells are considered.
</p>
</td></tr>
<tr><td><code id="plotDensity.AffymetrixCelSet_+3A_types">types</code></td>
<td>
<p>The type of probes to include, e.g. <code>"all"</code>,
<code>"pmmm"</code>, <code>"pm"</code>, and <code>"mm"</code>.</p>
</td></tr>
<tr><td><code id="plotDensity.AffymetrixCelSet_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to
<code><a href="#topic+plotDensity.AffymetrixCelFile">plotDensity.AffymetrixCelFile</a></code>().</p>
</td></tr>
<tr><td><code id="plotDensity.AffymetrixCelSet_+3A_col">col</code></td>
<td>
<p>A <code><a href="base.html#topic+vector">vector</a></code> of colors for each of the arrays.</p>
</td></tr>
<tr><td><code id="plotDensity.AffymetrixCelSet_+3A_lty">lty</code></td>
<td>
<p>A <code><a href="base.html#topic+vector">vector</a></code> of line types for each of the arrays.</p>
</td></tr>
<tr><td><code id="plotDensity.AffymetrixCelSet_+3A_lwd">lwd</code></td>
<td>
<p>A <code><a href="base.html#topic+vector">vector</a></code> of line widths for each of the arrays.</p>
</td></tr>
<tr><td><code id="plotDensity.AffymetrixCelSet_+3A_add">add</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">FALSE</a></code>, a new plot is created, otherwise the generated
graphics is added to the current plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns nothing.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>

<hr>
<h2 id='plotDensity.GenomeInformation'>Plots the density of SNPs for a given chromosome</h2><span id='topic+plotDensity.GenomeInformation'></span><span id='topic+GenomeInformation.plotDensity'></span><span id='topic+plotDensity+2CGenomeInformation-method'></span>

<h3>Description</h3>

<p>Plots the density of SNPs for a given chromosome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GenomeInformation'
plotDensity(this, chromosome, ..., adjust=1/20, xlab=NULL, main=NULL, annotate=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotDensity.GenomeInformation_+3A_chromosome">chromosome</code></td>
<td>
<p>The chromsome to be displayed.</p>
</td></tr>
<tr><td><code id="plotDensity.GenomeInformation_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+getPositions.GenomeInformation">*getPositions</a>()</code>.</p>
</td></tr>
<tr><td><code id="plotDensity.GenomeInformation_+3A_adjust">adjust</code></td>
<td>
<p>A bandwidth parameter for the density estimator.</p>
</td></tr>
<tr><td><code id="plotDensity.GenomeInformation_+3A_xlab">xlab</code></td>
<td>
<p>The label on the x-axis.  If <code><a href="base.html#topic+NULL">NULL</a></code>, a default will generated.</p>
</td></tr>
<tr><td><code id="plotDensity.GenomeInformation_+3A_main">main</code></td>
<td>
<p>The title of the plot.  If <code><a href="base.html#topic+NULL">NULL</a></code>, a default will generated.</p>
</td></tr>
<tr><td><code id="plotDensity.GenomeInformation_+3A_annotate">annotate</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the plot is annotated with extra information.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns (invisibly) the estimated density.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+GenomeInformation">GenomeInformation</a></code>.
</p>

<hr>
<h2 id='plotImage.AffymetrixCelFile'>Displays a spatial plot of a CEL file</h2><span id='topic+plotImage.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.plotImage'></span><span id='topic+plotImage+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Displays a spatial plot of a CEL file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelFile'
plotImage(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotImage.AffymetrixCelFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+getImage.AffymetrixCelFile">*getImage</a>()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns (invisibly) the image displayed.
</p>


<h3>Author(s)</h3>

<p>Ken Simpson, Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>

<hr>
<h2 id='plotMvsA.AffymetrixCelFile'>Plots log-ratio versus log-intensity in a scatter plot</h2><span id='topic+plotMvsA.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.plotMvsA'></span><span id='topic+plotMvsA+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Plots log-ratio versus log-intensity in a scatter plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelFile'
plotMvsA(this, reference, indices=NULL, pch=176, xlim=c(0, 16),
  ylim=c(-1, 1) * diff(xlim), xlab=expression(A == 1/2 %*% log[2](y[1] * y[2])),
  ylab=expression(M == log[2](y[1]/y[2])), ..., annotate=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotMvsA.AffymetrixCelFile_+3A_reference">reference</code></td>
<td>
<p>The reference channel, i.e. the denominator in the
log ratios.</p>
</td></tr>
<tr><td><code id="plotMvsA.AffymetrixCelFile_+3A_indices">indices</code></td>
<td>
<p>Indices of the probe signals to be plotted.</p>
</td></tr>
<tr><td><code id="plotMvsA.AffymetrixCelFile_+3A_pch">pch</code></td>
<td>
<p>The plot symbol.</p>
</td></tr>
<tr><td><code id="plotMvsA.AffymetrixCelFile_+3A_xlim">xlim</code>, <code id="plotMvsA.AffymetrixCelFile_+3A_ylim">ylim</code></td>
<td>
<p>The range of the x and the y axes.</p>
</td></tr>
<tr><td><code id="plotMvsA.AffymetrixCelFile_+3A_xlab">xlab</code>, <code id="plotMvsA.AffymetrixCelFile_+3A_ylab">ylab</code></td>
<td>
<p>The labels on the x and the y axes.</p>
</td></tr>
<tr><td><code id="plotMvsA.AffymetrixCelFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="graphics.html#topic+plot">plot</a></code>().</p>
</td></tr>
<tr><td><code id="plotMvsA.AffymetrixCelFile_+3A_annotate">annotate</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the plot is annotated with information about
the data plotted, otherwise not.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns (invisibly) a <code><a href="base.html#topic+data.frame">data.frame</a></code> with the plotted data in the
first two columns.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+smoothScatterMvsA.AffymetrixCelFile">*smoothScatterMvsA</a>()</code>.
<code><a href="#topic+plotMvsX.AffymetrixCelFile">*plotMvsX</a>()</code>.
For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>

<hr>
<h2 id='plotMvsX.AffymetrixCelFile'>Plots log-ratio versus another variable in a scatter plot</h2><span id='topic+plotMvsX.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.plotMvsX'></span><span id='topic+plotMvsX+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Plots log-ratio versus another variable in a scatter plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelFile'
plotMvsX(this, reference, x, indices=NULL, pch=176, ylim=c(-1, 1) * 2, ylab=NULL, ...,
  what=c("M", "A"), add=FALSE, annotate=!add)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotMvsX.AffymetrixCelFile_+3A_reference">reference</code></td>
<td>
<p>The reference channel, i.e. the denominator in the
log ratios.</p>
</td></tr>
<tr><td><code id="plotMvsX.AffymetrixCelFile_+3A_x">x</code></td>
<td>
<p>The other variable.  A <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code>.</p>
</td></tr>
<tr><td><code id="plotMvsX.AffymetrixCelFile_+3A_indices">indices</code></td>
<td>
<p>Indices of the probe signals to be plotted.</p>
</td></tr>
<tr><td><code id="plotMvsX.AffymetrixCelFile_+3A_pch">pch</code></td>
<td>
<p>The plot symbol.</p>
</td></tr>
<tr><td><code id="plotMvsX.AffymetrixCelFile_+3A_ylim">ylim</code></td>
<td>
<p>The range of the y axes.</p>
</td></tr>
<tr><td><code id="plotMvsX.AffymetrixCelFile_+3A_ylab">ylab</code></td>
<td>
<p>The labels on the y axes.</p>
</td></tr>
<tr><td><code id="plotMvsX.AffymetrixCelFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="graphics.html#topic+plot">plot</a></code>().</p>
</td></tr>
<tr><td><code id="plotMvsX.AffymetrixCelFile_+3A_annotate">annotate</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the plot is annotated with information about
the data plotted, otherwise not.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns (invisibly) a <code><a href="base.html#topic+data.frame">data.frame</a></code> with the plotted data in the
first two columns, and remaining data in the following columns.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotMvsA.AffymetrixCelFile">*plotMvsA</a>()</code>.
<code><a href="#topic+smoothScatterMvsA.AffymetrixCelFile">*smoothScatterMvsA</a>()</code>.
For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>

<hr>
<h2 id='ProbeAffinityFile'>The ProbeAffinityFile class</h2><span id='topic+ProbeAffinityFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class ProbeAffinityFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ParameterCelFile">ParameterCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>ProbeAffinityFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+CnProbeAffinityFile">CnProbeAffinityFile</a></em>, <em><a href="#topic+ExonProbeAffinityFile">ExonProbeAffinityFile</a></em>, <em><a href="#topic+SnpProbeAffinityFile">SnpProbeAffinityFile</a></em><br />
</p>
<p>public abstract static class <b>ProbeAffinityFile</b><br />
extends <em><a href="#topic+ParameterCelFile">ParameterCelFile</a></em><br />
</p>
<p>This class represents estimates of probe affinities in probe-level models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ProbeAffinityFile(..., probeModel=c("pm", "mm", "pm-mm", "min1(pm-mm)", "pm+mm"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ProbeAffinityFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+ParameterCelFile">ParameterCelFile</a></code>.</p>
</td></tr>
<tr><td><code id="ProbeAffinityFile_+3A_probemodel">probeModel</code></td>
<td>
<p>The specific type of probe model.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ParameterCelFile</b>:<br />
extractDataFrame, extractMatrix, readUnits
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from AffymetrixCelFile</b>:<br />
allocateFromCdf, as.character, clone, createFrom, extractMatrix, fromFile, getAm, getCdf, getExtensionPattern, getFileFormat, getImage, getUnitNamesFile, getUnitTypesFile, highlight, image270, nbrOfCells, plotDensity, plotImage, plotMvsA, plotMvsX, range, setCdf, smoothScatterMvsA, writeImage
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>An object of this class is typically obtained through the
<code>getProbeAffinityFile()</code> method for the <code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code> class.
</p>

<hr>
<h2 id='ProbeLevelModel'>The ProbeLevelModel class</h2><span id='topic+ProbeLevelModel'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class ProbeLevelModel</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>ProbeLevelModel</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AffineCnPlm">AffineCnPlm</a></em>, <em><a href="#topic+AffinePlm">AffinePlm</a></em>, <em><a href="#topic+AffineSnpPlm">AffineSnpPlm</a></em>, <em><a href="#topic+AvgCnPlm">AvgCnPlm</a></em>, <em><a href="#topic+AvgPlm">AvgPlm</a></em>, <em><a href="#topic+AvgSnpPlm">AvgSnpPlm</a></em>, <em><a href="#topic+ExonRmaPlm">ExonRmaPlm</a></em>, <em><a href="#topic+HetLogAddCnPlm">HetLogAddCnPlm</a></em>, <em><a href="#topic+HetLogAddPlm">HetLogAddPlm</a></em>, <em><a href="#topic+HetLogAddSnpPlm">HetLogAddSnpPlm</a></em>, <em><a href="#topic+MbeiCnPlm">MbeiCnPlm</a></em>, <em><a href="#topic+MbeiPlm">MbeiPlm</a></em>, <em><a href="#topic+MbeiSnpPlm">MbeiSnpPlm</a></em>, <em><a href="#topic+RmaCnPlm">RmaCnPlm</a></em>, <em><a href="#topic+RmaPlm">RmaPlm</a></em>, <em><a href="#topic+RmaSnpPlm">RmaSnpPlm</a></em><br />
</p>
<p>public abstract static class <b>ProbeLevelModel</b><br />
extends <em><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></em><br />
</p>
<p>This abstract class represents a probe-level model (PLM) as defined
by the <span class="pkg">affyPLM</span> package:
&quot;A [...] PLM is a model that is fit to probe-intensity data.
More specifically, it is where we fit a model with probe level
and chip level parameters on a probeset by probeset basis&quot;,
where the more general case for a probeset is a <em>unit group</em>
in Affymetrix CDF terms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ProbeLevelModel(..., standardize=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ProbeLevelModel_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></code>.</p>
</td></tr>
<tr><td><code id="ProbeLevelModel_+3A_standardize">standardize</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, chip-effect and probe-affinity estimates are
rescaled such that the product of the probe affinities is one.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In order to minimize the risk for mistakes, but also to be able compare
results from different PLMs, all PLM subclasses must meet the following
criteria:
</p>

<ol>
<li><p> All parameter estimates must be (stored and returned) on the
intensity scale, e.g. log-additive models such as <code><a href="#topic+RmaPlm">RmaPlm</a></code>
have to transform the parameters on the log-scale to the intensity
scale.
</p>
</li>
<li><p> The probe-affinity estimates <code class="reqn">\phi_k</code> for a unit group
must be constrained such that <code class="reqn">\prod_k \phi_k = 1</code>,
or equivalently if <code class="reqn">\phi_k &gt; 0</code>,<code class="reqn">\sum_k \log(\phi_k) = 0</code>.
</p>
</li></ol>

<p>Note that the above probe-affinity constraint guarantees that the
estimated chip effects across models are on the same scale.
</p>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>fit</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipEffectSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getProbeAffinityFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getResidualSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getWeightsSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from MultiArrayUnitModel</b>:<br />
getListOfPriors, setListOfPriors, validate
</p>
<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more details on probe-level models, please see
the <span class="pkg">preprocessCore</span> package.
</p>

<hr>
<h2 id='ProbeLevelTransform'>The ProbeLevelTransform class</h2><span id='topic+ProbeLevelTransform'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class ProbeLevelTransform</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><em><code>ProbeLevelTransform</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AbstractProbeSequenceNormalization">AbstractProbeSequenceNormalization</a></em>, <a href="#topic+AllelicCrosstalkCalibration">AllelicCrosstalkCalibration</a>, <em><a href="#topic+BackgroundCorrection">BackgroundCorrection</a></em>, <a href="#topic+BaseCountNormalization">BaseCountNormalization</a>, <em><a href="#topic+BasePositionNormalization">BasePositionNormalization</a></em>, <a href="#topic+DChipQuantileNormalization">DChipQuantileNormalization</a>, <a href="#topic+GcRmaBackgroundCorrection">GcRmaBackgroundCorrection</a>, <a href="#topic+LimmaBackgroundCorrection">LimmaBackgroundCorrection</a>, <em><a href="#topic+LinearModelProbeSequenceNormalization">LinearModelProbeSequenceNormalization</a></em>, <a href="#topic+MatNormalization">MatNormalization</a>, <a href="#topic+MatSmoothing">MatSmoothing</a>, <a href="#topic+NormExpBackgroundCorrection">NormExpBackgroundCorrection</a>, <a href="#topic+OpticalBackgroundCorrection">OpticalBackgroundCorrection</a>, <em><a href="#topic+ProbeLevelTransform3">ProbeLevelTransform3</a></em>, <a href="#topic+QuantileNormalization">QuantileNormalization</a>, <a href="#topic+ReseqCrosstalkCalibration">ReseqCrosstalkCalibration</a>, <a href="#topic+RmaBackgroundCorrection">RmaBackgroundCorrection</a>, <a href="#topic+ScaleNormalization">ScaleNormalization</a>, <a href="#topic+ScaleNormalization3">ScaleNormalization3</a>, <a href="#topic+SpatialRowColumnNormalization">SpatialRowColumnNormalization</a>, <a href="#topic+UnitTypeScaleNormalization">UnitTypeScaleNormalization</a><br />
</p>
<p>public abstract static class <b>ProbeLevelTransform</b><br />
extends <em><a href="#topic+Transform">Transform</a></em><br />
</p>
<p>This abstract class represents a transformation methods that transforms
probe-level signals, typically intensities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ProbeLevelTransform(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ProbeLevelTransform_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of <code><a href="#topic+Transform">Transform</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Subclasses must implement the <code>process()</code> method.
</p>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='ProbeLevelTransform3'>The ProbeLevelTransform3 class</h2><span id='topic+ProbeLevelTransform3'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class ProbeLevelTransform3</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>ProbeLevelTransform3</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AbstractProbeSequenceNormalization">AbstractProbeSequenceNormalization</a></em>, <a href="#topic+BaseCountNormalization">BaseCountNormalization</a>, <em><a href="#topic+BasePositionNormalization">BasePositionNormalization</a></em>, <em><a href="#topic+LinearModelProbeSequenceNormalization">LinearModelProbeSequenceNormalization</a></em>, <a href="#topic+MatNormalization">MatNormalization</a>, <a href="#topic+ScaleNormalization3">ScaleNormalization3</a>, <a href="#topic+UnitTypeScaleNormalization">UnitTypeScaleNormalization</a><br />
</p>
<p>public abstract static class <b>ProbeLevelTransform3</b><br />
extends <em><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></em><br />
</p>
<p>This abstract class is specialized from <code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code> and
provides methods to identify subsets and types of probes that are used
for fitting and/or updating the signals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ProbeLevelTransform3(dataSet=NULL, ..., unitsToFit="-XY", typesToFit=typesToUpdate,
  unitsToUpdate=NULL, typesToUpdate="pm", shift=0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ProbeLevelTransform3_+3A_dataset">dataSet</code></td>
<td>
<p>A <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.</p>
</td></tr>
<tr><td><code id="ProbeLevelTransform3_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code>.</p>
</td></tr>
<tr><td><code id="ProbeLevelTransform3_+3A_unitstofit">unitsToFit</code></td>
<td>
<p>The units from which the normalization curve should
be estimated.  If <code><a href="base.html#topic+NULL">NULL</a></code>, all are considered.</p>
</td></tr>
<tr><td><code id="ProbeLevelTransform3_+3A_typestofit">typesToFit</code></td>
<td>
<p>Types of probes to be used when fitting the model.</p>
</td></tr>
<tr><td><code id="ProbeLevelTransform3_+3A_unitstoupdate">unitsToUpdate</code></td>
<td>
<p>The units to be updated.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all are considered.</p>
</td></tr>
<tr><td><code id="ProbeLevelTransform3_+3A_typestoupdate">typesToUpdate</code></td>
<td>
<p>Types of probes to be updated.</p>
</td></tr>
<tr><td><code id="ProbeLevelTransform3_+3A_shift">shift</code></td>
<td>
<p>An optional amount to shift data before fitting and updating.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from ProbeLevelTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='process.AbstractProbeSequenceNormalization'>Normalizes the data set</h2><span id='topic+process.AbstractProbeSequenceNormalization'></span><span id='topic+AbstractProbeSequenceNormalization.process'></span><span id='topic+process+2CAbstractProbeSequenceNormalization-method'></span>

<h3>Description</h3>

<p>Normalizes the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AbstractProbeSequenceNormalization'
process(this, ..., ram=NULL, force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.AbstractProbeSequenceNormalization_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="process.AbstractProbeSequenceNormalization_+3A_ram">ram</code></td>
<td>
<p>A positive <code><a href="base.html#topic+double">double</a></code> scale factor specifying how much more
memory to use relative to the default.</p>
</td></tr>
<tr><td><code id="process.AbstractProbeSequenceNormalization_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, data already normalized is re-normalized,
otherwise not.</p>
</td></tr>
<tr><td><code id="process.AbstractProbeSequenceNormalization_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AbstractProbeSequenceNormalization">AbstractProbeSequenceNormalization</a></code>.
</p>

<hr>
<h2 id='process.AdditiveCovariatesNormalization'>Normalizes the data set</h2><span id='topic+process.AdditiveCovariatesNormalization'></span><span id='topic+AdditiveCovariatesNormalization.process'></span><span id='topic+process+2CAdditiveCovariatesNormalization-method'></span>

<h3>Description</h3>

<p>Normalizes the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AdditiveCovariatesNormalization'
process(this, ..., force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.AdditiveCovariatesNormalization_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to
<code><a href="aroma.light.html#topic+normalizeFragmentLength">normalizeFragmentLength</a></code> (only for advanced users).</p>
</td></tr>
<tr><td><code id="process.AdditiveCovariatesNormalization_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, data already normalized is re-normalized,
otherwise not.</p>
</td></tr>
<tr><td><code id="process.AdditiveCovariatesNormalization_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AdditiveCovariatesNormalization">AdditiveCovariatesNormalization</a></code>.
</p>

<hr>
<h2 id='process.AllelicCrosstalkCalibration'>Calibrates the data set</h2><span id='topic+process.AllelicCrosstalkCalibration'></span><span id='topic+AllelicCrosstalkCalibration.process'></span><span id='topic+process+2CAllelicCrosstalkCalibration-method'></span>

<h3>Description</h3>

<p>Calibrates the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AllelicCrosstalkCalibration'
process(this, ..., force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.AllelicCrosstalkCalibration_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="process.AllelicCrosstalkCalibration_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, data already calibrated is re-calibrated,
otherwise not.</p>
</td></tr>
<tr><td><code id="process.AllelicCrosstalkCalibration_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AllelicCrosstalkCalibration">AllelicCrosstalkCalibration</a></code>.
</p>

<hr>
<h2 id='process.ArrayExplorer'>Generates image files, scripts and dynamic pages for the explorer</h2><span id='topic+process.ArrayExplorer'></span><span id='topic+ArrayExplorer.process'></span><span id='topic+process+2CArrayExplorer-method'></span>

<h3>Description</h3>

<p>Generates image files, scripts and dynamic pages for the explorer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ArrayExplorer'
process(this, arrays=NULL, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.ArrayExplorer_+3A_arrays">arrays</code></td>
<td>
<p>An optional <code><a href="base.html#topic+vector">vector</a></code> of arrays to be processed.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all arrays are considered.</p>
</td></tr>
<tr><td><code id="process.ArrayExplorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="process.ArrayExplorer_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns nothing.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ArrayExplorer">ArrayExplorer</a></code>.
</p>

<hr>
<h2 id='process.BackgroundCorrection'>Processes the data set</h2><span id='topic+process.BackgroundCorrection'></span><span id='topic+BackgroundCorrection.process'></span><span id='topic+process+2CBackgroundCorrection-method'></span>

<h3>Description</h3>

<p>Processes the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BackgroundCorrection'
process(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.BackgroundCorrection_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+BackgroundCorrection">BackgroundCorrection</a></code>.
</p>

<hr>
<h2 id='process.ChipEffectGroupMerge'>Normalizes the data set</h2><span id='topic+process.ChipEffectGroupMerge'></span><span id='topic+ChipEffectGroupMerge.process'></span><span id='topic+process+2CChipEffectGroupMerge-method'></span>

<h3>Description</h3>

<p>Normalizes the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ChipEffectGroupMerge'
process(this, ..., force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.ChipEffectGroupMerge_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="process.ChipEffectGroupMerge_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, data already normalized is re-normalized,
otherwise not.</p>
</td></tr>
<tr><td><code id="process.ChipEffectGroupMerge_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ChipEffectGroupMerge">ChipEffectGroupMerge</a></code>.
</p>

<hr>
<h2 id='process.DChipQuantileNormalization'>Normalizes the data set</h2><span id='topic+process.DChipQuantileNormalization'></span><span id='topic+DChipQuantileNormalization.process'></span><span id='topic+process+2CDChipQuantileNormalization-method'></span>

<h3>Description</h3>

<p>Normalizes the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'DChipQuantileNormalization'
process(this, ..., force=FALSE, skip=TRUE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.DChipQuantileNormalization_+3A_...">...</code></td>
<td>
<p>Arguments passed to
<code><a href="aroma.light.html#topic+normalizeQuantileSpline.numeric">normalizeQuantileSpline.numeric</a></code>.</p>
</td></tr>
<tr><td><code id="process.DChipQuantileNormalization_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, data already normalized is re-normalized,
otherwise not.</p>
</td></tr>
<tr><td><code id="process.DChipQuantileNormalization_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+DChipQuantileNormalization">DChipQuantileNormalization</a></code>.
</p>

<hr>
<h2 id='process.FragmentEquivalentClassNormalization'>Normalizes the data set</h2><span id='topic+process.FragmentEquivalentClassNormalization'></span><span id='topic+FragmentEquivalentClassNormalization.process'></span><span id='topic+process+2CFragmentEquivalentClassNormalization-method'></span>

<h3>Description</h3>

<p>Normalizes the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FragmentEquivalentClassNormalization'
process(this, ..., force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.FragmentEquivalentClassNormalization_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="process.FragmentEquivalentClassNormalization_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, data already normalized is re-normalized,
otherwise not.</p>
</td></tr>
<tr><td><code id="process.FragmentEquivalentClassNormalization_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+FragmentEquivalentClassNormalization">FragmentEquivalentClassNormalization</a></code>.
</p>

<hr>
<h2 id='process.FragmentLengthNormalization'>Normalizes the data set</h2><span id='topic+process.FragmentLengthNormalization'></span><span id='topic+FragmentLengthNormalization.process'></span><span id='topic+process+2CFragmentLengthNormalization-method'></span>

<h3>Description</h3>

<p>Normalizes the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FragmentLengthNormalization'
process(this, ..., force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.FragmentLengthNormalization_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to
<code><a href="aroma.light.html#topic+normalizeFragmentLength">normalizeFragmentLength</a></code> (only for advanced users).</p>
</td></tr>
<tr><td><code id="process.FragmentLengthNormalization_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, data already normalized is re-normalized,
otherwise not.</p>
</td></tr>
<tr><td><code id="process.FragmentLengthNormalization_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+FragmentLengthNormalization">FragmentLengthNormalization</a></code>.
</p>

<hr>
<h2 id='process.GcContentNormalization'>Normalizes the data set</h2><span id='topic+process.GcContentNormalization'></span><span id='topic+GcContentNormalization.process'></span><span id='topic+process+2CGcContentNormalization-method'></span>

<h3>Description</h3>

<p>Normalizes the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GcContentNormalization'
process(this, ..., force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.GcContentNormalization_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="process.GcContentNormalization_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, data already normalized is re-normalized,
otherwise not.</p>
</td></tr>
<tr><td><code id="process.GcContentNormalization_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+GcContentNormalization">GcContentNormalization</a></code>.
</p>

<hr>
<h2 id='process.GcRmaBackgroundCorrection'>Performs background correction</h2><span id='topic+process.GcRmaBackgroundCorrection'></span><span id='topic+GcRmaBackgroundCorrection.process'></span><span id='topic+process+2CGcRmaBackgroundCorrection-method'></span>

<h3>Description</h3>

<p>Performs background correction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GcRmaBackgroundCorrection'
process(this, ..., force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.GcRmaBackgroundCorrection_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="process.GcRmaBackgroundCorrection_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the output data set.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+GcRmaBackgroundCorrection">GcRmaBackgroundCorrection</a></code>.
</p>

<hr>
<h2 id='process.GenericReporter'>Generates report</h2><span id='topic+process.GenericReporter'></span><span id='topic+GenericReporter.process'></span><span id='topic+process+2CGenericReporter-method'></span>

<h3>Description</h3>

<p>Generates report.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GenericReporter'
process(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.GenericReporter_+3A_...">...</code></td>
<td>
<p>See subclasses.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns nothing.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+GenericReporter">GenericReporter</a></code>.
</p>

<hr>
<h2 id='process.LimmaBackgroundCorrection'>Performs background correction</h2><span id='topic+process.LimmaBackgroundCorrection'></span><span id='topic+LimmaBackgroundCorrection.process'></span><span id='topic+process+2CLimmaBackgroundCorrection-method'></span>

<h3>Description</h3>

<p>Performs background correction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'LimmaBackgroundCorrection'
process(this, ..., force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.LimmaBackgroundCorrection_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="process.LimmaBackgroundCorrection_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+LimmaBackgroundCorrection">LimmaBackgroundCorrection</a></code>.
</p>

<hr>
<h2 id='process.MatNormalization'>Normalizes the data set</h2><span id='topic+process.MatNormalization'></span><span id='topic+MatNormalization.process'></span><span id='topic+process+2CMatNormalization-method'></span>

<h3>Description</h3>

<p>Normalizes the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MatNormalization'
process(this, ..., ram=NULL, force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.MatNormalization_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="process.MatNormalization_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, data already normalized is re-normalized,
otherwise not.</p>
</td></tr>
<tr><td><code id="process.MatNormalization_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+MatNormalization">MatNormalization</a></code>.
</p>

<hr>
<h2 id='process.MatSmoothing'>Processes the data set</h2><span id='topic+process.MatSmoothing'></span><span id='topic+MatSmoothing.process'></span><span id='topic+process+2CMatSmoothing-method'></span>

<h3>Description</h3>

<p>Processes the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MatSmoothing'
process(this, ..., units=NULL, force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.MatSmoothing_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="process.MatSmoothing_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, data already processed is re-processed,
otherwise not.</p>
</td></tr>
<tr><td><code id="process.MatSmoothing_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+MatSmoothing">MatSmoothing</a></code>.
</p>

<hr>
<h2 id='process.OpticalBackgroundCorrection'>Performs background correction</h2><span id='topic+process.OpticalBackgroundCorrection'></span><span id='topic+OpticalBackgroundCorrection.process'></span><span id='topic+process+2COpticalBackgroundCorrection-method'></span>

<h3>Description</h3>

<p>Performs background correction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'OpticalBackgroundCorrection'
process(this, ..., force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.OpticalBackgroundCorrection_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="process.OpticalBackgroundCorrection_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the output data set.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+OpticalBackgroundCorrection">OpticalBackgroundCorrection</a></code>.
</p>

<hr>
<h2 id='process.QuantileNormalization'>Normalizes the data set</h2><span id='topic+process.QuantileNormalization'></span><span id='topic+QuantileNormalization.process'></span><span id='topic+process+2CQuantileNormalization-method'></span>

<h3>Description</h3>

<p>Normalizes the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'QuantileNormalization'
process(this, ..., force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.QuantileNormalization_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="process.QuantileNormalization_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, data already normalized is re-normalized,
otherwise not.</p>
</td></tr>
<tr><td><code id="process.QuantileNormalization_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+QuantileNormalization">QuantileNormalization</a></code>.
</p>

<hr>
<h2 id='process.ReseqCrosstalkCalibration'>Calibrates the data set</h2><span id='topic+process.ReseqCrosstalkCalibration'></span><span id='topic+ReseqCrosstalkCalibration.process'></span><span id='topic+process+2CReseqCrosstalkCalibration-method'></span>

<h3>Description</h3>

<p>Calibrates the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ReseqCrosstalkCalibration'
process(this, ..., force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.ReseqCrosstalkCalibration_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="process.ReseqCrosstalkCalibration_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, data already calibrated is re-calibrated,
otherwise not.</p>
</td></tr>
<tr><td><code id="process.ReseqCrosstalkCalibration_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ReseqCrosstalkCalibration">ReseqCrosstalkCalibration</a></code>.
</p>

<hr>
<h2 id='process.RmaBackgroundCorrection'>Performs background correction</h2><span id='topic+process.RmaBackgroundCorrection'></span><span id='topic+RmaBackgroundCorrection.process'></span><span id='topic+process+2CRmaBackgroundCorrection-method'></span>

<h3>Description</h3>

<p>Performs background correction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'RmaBackgroundCorrection'
process(this, ..., force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.RmaBackgroundCorrection_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="process.RmaBackgroundCorrection_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the output data set.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+RmaBackgroundCorrection">RmaBackgroundCorrection</a></code>.
</p>

<hr>
<h2 id='process.ScaleNormalization'>Normalizes the data set</h2><span id='topic+process.ScaleNormalization'></span><span id='topic+ScaleNormalization.process'></span><span id='topic+process+2CScaleNormalization-method'></span>

<h3>Description</h3>

<p>Normalizes the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ScaleNormalization'
process(this, ..., skip=FALSE, force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.ScaleNormalization_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="process.ScaleNormalization_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, data already normalized is re-normalized,
otherwise not.</p>
</td></tr>
<tr><td><code id="process.ScaleNormalization_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ScaleNormalization">ScaleNormalization</a></code>.
</p>

<hr>
<h2 id='process.ScaleNormalization3'>Normalizes the data set</h2><span id='topic+process.ScaleNormalization3'></span><span id='topic+ScaleNormalization3.process'></span><span id='topic+process+2CScaleNormalization3-method'></span>

<h3>Description</h3>

<p>Normalizes the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ScaleNormalization3'
process(this, ..., skip=FALSE, force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.ScaleNormalization3_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="process.ScaleNormalization3_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, data already normalized is re-normalized,
otherwise not.</p>
</td></tr>
<tr><td><code id="process.ScaleNormalization3_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ScaleNormalization3">ScaleNormalization3</a></code>.
</p>

<hr>
<h2 id='process.SpatialReporter'>Generates image files, scripts and dynamic pages for the explorer</h2><span id='topic+process.SpatialReporter'></span><span id='topic+SpatialReporter.process'></span><span id='topic+process+2CSpatialReporter-method'></span>

<h3>Description</h3>

<p>Generates image files, scripts and dynamic pages for the explorer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SpatialReporter'
process(this, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.SpatialReporter_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="process.SpatialReporter_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns nothing.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+SpatialReporter">SpatialReporter</a></code>.
</p>

<hr>
<h2 id='process.UnitTypeScaleNormalization'>Normalizes the data set</h2><span id='topic+process.UnitTypeScaleNormalization'></span><span id='topic+UnitTypeScaleNormalization.process'></span><span id='topic+process+2CUnitTypeScaleNormalization-method'></span>

<h3>Description</h3>

<p>Normalizes the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'UnitTypeScaleNormalization'
process(this, ..., skip=FALSE, force=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process.UnitTypeScaleNormalization_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="process.UnitTypeScaleNormalization_+3A_force">force</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, data already normalized is re-normalized,
otherwise not.</p>
</td></tr>
<tr><td><code id="process.UnitTypeScaleNormalization_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+double">double</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+UnitTypeScaleNormalization">UnitTypeScaleNormalization</a></code>.
</p>

<hr>
<h2 id='QualityAssessmentFile'>The QualityAssessmentFile class</h2><span id='topic+QualityAssessmentFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class QualityAssessmentFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>QualityAssessmentFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>QualityAssessmentFile</b><br />
extends <em><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></em><br />
</p>
<p>This class represents probe-level QC information (residuals, weights, etc.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QualityAssessmentFile(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QualityAssessmentFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>findUnitsTodo</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AffymetrixCelFile</b>:<br />
allocateFromCdf, as.character, clone, createFrom, extractMatrix, fromFile, getAm, getCdf, getExtensionPattern, getFileFormat, getImage, getUnitNamesFile, getUnitTypesFile, highlight, image270, nbrOfCells, plotDensity, plotImage, plotMvsA, plotMvsX, range, setCdf, smoothScatterMvsA, writeImage
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Ken Simpson</p>


<h3>See Also</h3>

<p>An object of this class is typically part of a <code><a href="#topic+QualityAssessmentSet">QualityAssessmentSet</a></code>.
</p>

<hr>
<h2 id='QualityAssessmentModel'>The QualityAssessmentModel class</h2><span id='topic+QualityAssessmentModel'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class QualityAssessmentModel</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code>QualityAssessmentModel</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>QualityAssessmentModel</b><br />
extends <a href="R.oo.html#topic+Object">Object</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QualityAssessmentModel(plm=NULL, tags="*", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QualityAssessmentModel_+3A_plm">plm</code></td>
<td>
<p>A <code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code>.</p>
</td></tr>
<tr><td><code id="QualityAssessmentModel_+3A_tags">tags</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> of tags.</p>
</td></tr>
<tr><td><code id="QualityAssessmentModel_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipEffectSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getDataSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFullName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPath</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPlm</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getResiduals</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getTags</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getWeights</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfArrays</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plotNuse</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plotRle</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Ken Simpson, Henrik Bengtsson</p>

<hr>
<h2 id='QualityAssessmentSet'>The QualityAssessmentSet class</h2><span id='topic+QualityAssessmentSet'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class QualityAssessmentSet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataSet">AromaMicroarrayDataSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>QualityAssessmentSet</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>QualityAssessmentSet</b><br />
extends <a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QualityAssessmentSet(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QualityAssessmentSet_+3A_...">...</code></td>
<td>
<p>Arguments passed to constructor of <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from AffymetrixCelSet</b>:<br />
append, as, as.AffymetrixCelSet, as.character, averageQuantile, byName, byPath, clone, convertToUnique, doCRMAv1, doCRMAv2, doFIRMA, doGCRMA, doRMA, extractAffyBatch, extractFeatureSet, extractMatrix, extractSnpFeatureSet, findByName, getAverage, getAverageAsinh, getAverageFile, getAverageLog, getCdf, getChipType, getData, getIntensities, getPlatform, getTimestamps, getUnitGroupCellMap, getUnitIntensities, getUnitNamesFile, getUnitTypesFile, isDuplicated, justRMA, justSNPRMA, nbrOfArrays, normalizeQuantile, plotDensity, range, readUnits, setCdf, update2, writeSgr
</p>
<p><b>Methods inherited from AffymetrixFileSet</b>:<br />
as, as.AffymetrixFileSet, byPath, getDefaultFullName
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataSet</b>:<br />
as.AromaMicroarrayDataSetList, as.AromaMicroarrayDataSetTuple, getAromaFullNameTranslatorSet, getAverageFile, getChipType, getDefaultFullName, getPlatform, setAttributesBy, setAttributesBySampleAnnotationFile, setAttributesBySampleAnnotationSet, validate
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Ken Simpson</p>

<hr>
<h2 id='QuantileNormalization'>The QuantileNormalization class</h2><span id='topic+QuantileNormalization'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class QuantileNormalization</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code>QuantileNormalization</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+DChipQuantileNormalization">DChipQuantileNormalization</a><br />
</p>
<p>public static class <b>QuantileNormalization</b><br />
extends <em><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></em><br />
</p>
<p>This class represents a normalization function that transforms the
probe-level signals towards the same empirical distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QuantileNormalization(..., subsetToUpdate=NULL, typesToUpdate=NULL,
  targetDistribution=NULL, subsetToAvg=subsetToUpdate, typesToAvg=typesToUpdate)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QuantileNormalization_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code>.</p>
</td></tr>
<tr><td><code id="QuantileNormalization_+3A_subsettoupdate">subsetToUpdate</code></td>
<td>
<p>The probes to be updated.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all probes are updated.</p>
</td></tr>
<tr><td><code id="QuantileNormalization_+3A_typestoupdate">typesToUpdate</code></td>
<td>
<p>Types of probes to be updated.</p>
</td></tr>
<tr><td><code id="QuantileNormalization_+3A_targetdistribution">targetDistribution</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code>.  The empirical
distribution to which all arrays should be normalized to.</p>
</td></tr>
<tr><td><code id="QuantileNormalization_+3A_subsettoavg">subsetToAvg</code></td>
<td>
<p>The probes to calculate average empirical
distribution over.  If a single <code><a href="base.html#topic+numeric">numeric</a></code> in (0,1), then this
fraction of all probes will be used.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all probes are considered.</p>
</td></tr>
<tr><td><code id="QuantileNormalization_+3A_typestoavg">typesToAvg</code></td>
<td>
<p>Types of probes to be used when calculating the
average empirical distribution.
If <code>"pm"</code> and <code>"mm"</code> only perfect-match and mismatch
probes are used, respectively. If <code>"pmmm"</code> both types are used.
</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ProbeLevelTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  for (zzz in 0) {

# Setup verbose output
verbose &lt;- Arguments$getVerbose(-2)
timestampOn(verbose)

# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Define an example dataset
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Find any dataset
path &lt;- NULL
if (is.null(path))
  break

ds &lt;- AffymetrixCelSet$fromFiles(path)
print(ds)

# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Normalization
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
norm &lt;- QuantileNormalization(ds, subsetToAvg=1/3)
dsQN &lt;- process(norm, verbose=verbose)
print(dsQN)

} # for (zzz in 0)
rm(zzz)


## End(Not run)</code></pre>

<hr>
<h2 id='randomSeed'>Sets and resets the .Random.seed in the global environment</h2><span id='topic+randomSeed'></span>

<h3>Description</h3>

<p>Sets and resets the .Random.seed in the global environment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> randomSeed(action=c("set", "advance", "reset", "get"), seed=NULL, kind=NULL, backup=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="randomSeed_+3A_action">action</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying the action.</p>
</td></tr>
<tr><td><code id="randomSeed_+3A_seed">seed</code></td>
<td>
<p>Random seed to be set; only for <code>action="set"</code>.
If <code>length(seed) == 1</code>, then <code>set.seed(seed)</code> is
used, otherwise <code>.Random.seed</code> is assigned the value.</p>
</td></tr>
<tr><td><code id="randomSeed_+3A_kind">kind</code></td>
<td>
<p>(optional) A <code><a href="base.html#topic+character">character</a></code> string specifying type of
random number generator to use, cf. <code><a href="base.html#topic+RNGkind">RNGkind</a></code>().</p>
</td></tr>
<tr><td><code id="randomSeed_+3A_backup">backup</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the previous (seed, kind) state is recorded
such that it can be reset later.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code>.Random.seed</code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='readRawData.AffymetrixCelFile'>Gets all or a subset of the fields in a CEL file</h2><span id='topic+readRawData.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.readRawData'></span><span id='topic+readRawData+2CAffymetrixCelFile-method'></span><span id='topic+AffymetrixCelFile.getData'></span><span id='topic+getData.AffymetrixCelFile'></span><span id='topic+getData+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Gets all or a subset of the fields in a CEL file for all or a subset of the cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelFile'
readRawData(this, indices=NULL, fields=c("xy", "intensities", "stdvs", "pixels"), ...,
  drop=FALSE, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readRawData.AffymetrixCelFile_+3A_indices">indices</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of cell indices.  If <code><a href="base.html#topic+NULL">NULL</a></code>, all cells
are considered.</p>
</td></tr>
<tr><td><code id="readRawData.AffymetrixCelFile_+3A_fields">fields</code></td>
<td>
<p>Names of fields to be retrieved.</p>
</td></tr>
<tr><td><code id="readRawData.AffymetrixCelFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="affxparser.html#topic+readCel">readCel</a></code>.</p>
</td></tr>
<tr><td><code id="readRawData.AffymetrixCelFile_+3A_drop">drop</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code> and a single field is returned, then data is
returned as a <code><a href="base.html#topic+vector">vector</a></code>, otherwise as a <code><a href="base.html#topic+data.frame">data.frame</a></code>.</p>
</td></tr>
<tr><td><code id="readRawData.AffymetrixCelFile_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+data.frame">data.frame</a></code> of the fields requested (unless dimension dropped).
</p>


<h3>Caching</h3>

<p>Neither in-memory nor on-file caching is done by this method.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>

<hr>
<h2 id='readUnits.AffymetrixCdfFile'>Reads CDF data unit by unit</h2><span id='topic+readUnits.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.readUnits'></span><span id='topic+readUnits+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Reads CDF data unit by unit for all or a subset of units (probeset).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
readUnits(this, units=NULL, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readUnits.AffymetrixCdfFile_+3A_units">units</code></td>
<td>
<p>The units to be read. If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are read.</p>
</td></tr>
<tr><td><code id="readUnits.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="affxparser.html#topic+readCdfUnits">readCdfUnits</a></code>.</p>
</td></tr>
<tr><td><code id="readUnits.AffymetrixCdfFile_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the <code><a href="base.html#topic+list">list</a></code> structure that <code><a href="affxparser.html#topic+readCdfUnits">readCdfUnits</a></code> returns
(possibly restructured).
</p>


<h3>Caching</h3>

<p>CDF data is neither cached in memory nor on file by this method.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='readUnits.AffymetrixCelFile'>Reads CEL data unit by unit</h2><span id='topic+readUnits.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.readUnits'></span><span id='topic+readUnits+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Reads CEL data unit by unit for all or a subset of units (probeset).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelFile'
readUnits(this, units=NULL, cdf=NULL, ..., stratifyBy=NULL, force=FALSE, cache=FALSE,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readUnits.AffymetrixCelFile_+3A_units">units</code></td>
<td>
<p>The units to be read. If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are read.</p>
</td></tr>
<tr><td><code id="readUnits.AffymetrixCelFile_+3A_cdf">cdf</code></td>
<td>
<p>An alternative CDF structure to be used.  This overrides
the <code>units</code> arguments.</p>
</td></tr>
<tr><td><code id="readUnits.AffymetrixCelFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code>getUnits()</code> of the
<code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> class (if <code>cdf</code> was not specified),
but also to the <code><a href="affxparser.html#topic+readCelUnits">readCelUnits</a></code> methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the <code><a href="base.html#topic+list">list</a></code> structure that <code><a href="affxparser.html#topic+readCelUnits">readCelUnits</a></code> returns.
</p>


<h3>Caching</h3>

<p>CEL data is neither cached in memory nor on file by this method.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+updateUnits.AffymetrixCelFile">*updateUnits</a>()</code>.
For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>

<hr>
<h2 id='readUnits.CnagCfhFile'>Reads CEL data unit by unit</h2><span id='topic+readUnits.CnagCfhFile'></span><span id='topic+CnagCfhFile.readUnits'></span><span id='topic+readUnits+2CCnagCfhFile-method'></span>

<h3>Description</h3>

<p>Reads CEL data unit by unit for all or a subset of units (probeset).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CnagCfhFile'
readUnits(this, units=NULL, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readUnits.CnagCfhFile_+3A_units">units</code></td>
<td>
<p>The units to be read. If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are read.</p>
</td></tr>
<tr><td><code id="readUnits.CnagCfhFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code>getUnits()</code> of the
<code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> class (if <code>cdf</code> was not specified),
but also to the <code><a href="affxparser.html#topic+readCelUnits">readCelUnits</a></code> methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the <code><a href="base.html#topic+list">list</a></code> structure that <code><a href="affxparser.html#topic+readCelUnits">readCelUnits</a></code> returns.
</p>


<h3>Caching</h3>

<p>CEL data is neither cached in memory nor on file by this method.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+CnagCfhFile">CnagCfhFile</a></code>.
</p>

<hr>
<h2 id='readUnits.MultiArrayUnitModel'>Reads data unit by unit</h2><span id='topic+readUnits.MultiArrayUnitModel'></span><span id='topic+MultiArrayUnitModel.readUnits'></span><span id='topic+readUnits+2CMultiArrayUnitModel-method'></span>

<h3>Description</h3>

<p>Reads data unit by unit for all or a subset of units (probeset)
specially structured for this PLM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MultiArrayUnitModel'
readUnits(this, units=NULL, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readUnits.MultiArrayUnitModel_+3A_units">units</code></td>
<td>
<p>The units to be read. If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are read.</p>
</td></tr>
<tr><td><code id="readUnits.MultiArrayUnitModel_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code>getCellIndices()</code> of the
<code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> class (if <code>cdf</code> was not specified),
but also to the <code>readUnits()</code> method of the
<code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code> class.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the <code><a href="base.html#topic+list">list</a></code> structure that <code>readUnits()</code> of
<code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code> returns.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></code>.
</p>

<hr>
<h2 id='readUnits.SingleArrayUnitModel'>Reads data unit by unit</h2><span id='topic+readUnits.SingleArrayUnitModel'></span><span id='topic+SingleArrayUnitModel.readUnits'></span><span id='topic+readUnits+2CSingleArrayUnitModel-method'></span>

<h3>Description</h3>

<p>Reads data unit by unit for all or a subset of units (probeset)
specially structured for this PLM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SingleArrayUnitModel'
readUnits(this, array, units=NULL, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readUnits.SingleArrayUnitModel_+3A_units">units</code></td>
<td>
<p>The units to be read. If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are read.</p>
</td></tr>
<tr><td><code id="readUnits.SingleArrayUnitModel_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code>getCellIndices()</code> of the
<code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> class (if <code>cdf</code> was not specified),
but also to the <code>readUnits()</code> method of the
<code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code> class.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the <code><a href="base.html#topic+list">list</a></code> structure that <code>readUnits()</code> of
<code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code> returns.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+SingleArrayUnitModel">SingleArrayUnitModel</a></code>.
</p>

<hr>
<h2 id='readUnitsByQuartets.AffymetrixCdfFile'>Gets the cell quartets for each base position</h2><span id='topic+readUnitsByQuartets.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.readUnitsByQuartets'></span><span id='topic+readUnitsByQuartets+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Gets the cell quartets for each base position.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
readUnitsByQuartets(this, units=NULL, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readUnitsByQuartets.AffymetrixCdfFile_+3A_units">units</code></td>
<td>
<p>Subset of units to be queried. If <code><a href="base.html#topic+NULL">NULL</a></code>, all units are used.</p>
</td></tr>
<tr><td><code id="readUnitsByQuartets.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+vector">vector</a></code> of <code><a href="base.html#topic+factor">factor</a></code>s.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='ReseqCrosstalkCalibration'>The ReseqCrosstalkCalibration class</h2><span id='topic+ReseqCrosstalkCalibration'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class ReseqCrosstalkCalibration</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code>ReseqCrosstalkCalibration</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>ReseqCrosstalkCalibration</b><br />
extends <em><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></em><br />
</p>
<p>This class represents a calibration function that transforms the
probe-level signals such that the signals from the four nucleotides
(A, C, G, T) are orthogonal.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ReseqCrosstalkCalibration(dataSet=NULL, ..., targetAvg=2200, subsetToAvg=NULL,
  mergeGroups=FALSE, flavor=c("sfit", "expectile"), alpha=c(0.1, 0.075, 0.05, 0.03,
  0.01), q=2, Q=98)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ReseqCrosstalkCalibration_+3A_dataset">dataSet</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.</p>
</td></tr>
<tr><td><code id="ReseqCrosstalkCalibration_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code>.</p>
</td></tr>
<tr><td><code id="ReseqCrosstalkCalibration_+3A_targetavg">targetAvg</code></td>
<td>
<p>The signal(s) that the average of the sum of the
probe quartets should have after calibration.</p>
</td></tr>
<tr><td><code id="ReseqCrosstalkCalibration_+3A_subsettoavg">subsetToAvg</code></td>
<td>
<p>The indices of the cells (taken as the intersect of
existing indices) used to calculate average in order to rescale to
the target average. If <code><a href="base.html#topic+NULL">NULL</a></code>, all probes are considered.</p>
</td></tr>
<tr><td><code id="ReseqCrosstalkCalibration_+3A_mergegroups">mergeGroups</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> ...</p>
</td></tr>
<tr><td><code id="ReseqCrosstalkCalibration_+3A_flavor">flavor</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying what algorithm is used
to fit the crosstalk calibration.</p>
</td></tr>
<tr><td><code id="ReseqCrosstalkCalibration_+3A_alpha">alpha</code>, <code id="ReseqCrosstalkCalibration_+3A_q">q</code>, <code id="ReseqCrosstalkCalibration_+3A_q">Q</code></td>
<td>
<p>Additional arguments passed to
<code>fitMultiDimensionalCone()</code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ProbeLevelTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='ResidualFile'>The ResidualFile class</h2><span id='topic+ResidualFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class ResidualFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ParameterCelFile">ParameterCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>ResidualFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>ResidualFile</b><br />
extends <em><a href="#topic+ParameterCelFile">ParameterCelFile</a></em><br />
</p>
<p>This class represents estimates of residuals in the probe-level models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ResidualFile(..., probeModel=c("pm"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ResidualFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+ParameterCelFile">ParameterCelFile</a></code>.</p>
</td></tr>
<tr><td><code id="ResidualFile_+3A_probemodel">probeModel</code></td>
<td>
<p>The specific type of model, e.g. <code>"pm"</code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>findUnitsTodo</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getImage</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>writeImage</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ParameterCelFile</b>:<br />
extractDataFrame, extractMatrix, readUnits
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from AffymetrixCelFile</b>:<br />
allocateFromCdf, as.character, clone, createFrom, extractMatrix, fromFile, getAm, getCdf, getExtensionPattern, getFileFormat, getImage, getUnitNamesFile, getUnitTypesFile, highlight, image270, nbrOfCells, plotDensity, plotImage, plotMvsA, plotMvsX, range, setCdf, smoothScatterMvsA, writeImage
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Ken Simpson, Henrik Bengtsson</p>


<h3>See Also</h3>

<p>An object of this class is typically obtained through the
<code>getResidualSet()</code> method for the <code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code> class.
An object of this class is typically part of a <code><a href="#topic+ResidualSet">ResidualSet</a></code>.
</p>

<hr>
<h2 id='ResidualSet'>The ResidualSet class</h2><span id='topic+ResidualSet'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class ResidualSet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataSet">AromaMicroarrayDataSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>ResidualSet</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>ResidualSet</b><br />
extends <a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a><br />
</p>
<p>This class represents probe-level residuals from probe-level models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ResidualSet(..., probeModel=c("pm"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ResidualSet_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.</p>
</td></tr>
<tr><td><code id="ResidualSet_+3A_probemodel">probeModel</code></td>
<td>
<p>The specific type of model, e.g. <code>"pm"</code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>findUnitsTodo</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAverageFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCellIndices</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from AffymetrixCelSet</b>:<br />
append, as, as.AffymetrixCelSet, as.character, averageQuantile, byName, byPath, clone, convertToUnique, doCRMAv1, doCRMAv2, doFIRMA, doGCRMA, doRMA, extractAffyBatch, extractFeatureSet, extractMatrix, extractSnpFeatureSet, findByName, getAverage, getAverageAsinh, getAverageFile, getAverageLog, getCdf, getChipType, getData, getIntensities, getPlatform, getTimestamps, getUnitGroupCellMap, getUnitIntensities, getUnitNamesFile, getUnitTypesFile, isDuplicated, justRMA, justSNPRMA, nbrOfArrays, normalizeQuantile, plotDensity, range, readUnits, setCdf, update2, writeSgr
</p>
<p><b>Methods inherited from AffymetrixFileSet</b>:<br />
as, as.AffymetrixFileSet, byPath, getDefaultFullName
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataSet</b>:<br />
as.AromaMicroarrayDataSetList, as.AromaMicroarrayDataSetTuple, getAromaFullNameTranslatorSet, getAverageFile, getChipType, getDefaultFullName, getPlatform, setAttributesBy, setAttributesBySampleAnnotationFile, setAttributesBySampleAnnotationSet, validate
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Ken Simpson, Henrik Bengtsson</p>


<h3>See Also</h3>

<p>An object of this class is typically obtained through the
<code>getResidualSet()</code> method for the <code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code> class.
</p>

<hr>
<h2 id='RmaBackgroundCorrection'>The RmaBackgroundCorrection class</h2><span id='topic+RmaBackgroundCorrection'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class RmaBackgroundCorrection</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+BackgroundCorrection">BackgroundCorrection</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>RmaBackgroundCorrection</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>RmaBackgroundCorrection</b><br />
extends <em><a href="#topic+BackgroundCorrection">BackgroundCorrection</a></em><br />
</p>
<p>This class represents the RMA background adjustment function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RmaBackgroundCorrection(..., addJitter=FALSE, jitterSd=0.2, seed=6022007)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RmaBackgroundCorrection_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+BackgroundCorrection">BackgroundCorrection</a></code>.</p>
</td></tr>
<tr><td><code id="RmaBackgroundCorrection_+3A_addjitter">addJitter</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, Zero-mean gaussian noise is added to the
signals before being background corrected.</p>
</td></tr>
<tr><td><code id="RmaBackgroundCorrection_+3A_jittersd">jitterSd</code></td>
<td>
<p>Standard deviation of the jitter noise added.</p>
</td></tr>
<tr><td><code id="RmaBackgroundCorrection_+3A_seed">seed</code></td>
<td>
<p>An (optional) <code><a href="base.html#topic+integer">integer</a></code> specifying a temporary random seed
to be used for generating the (optional) jitter.  The random seed
is set to its original state when done.  If <code><a href="base.html#topic+NULL">NULL</a></code>, it is not set.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally <code><a href="affy.html#topic+bg.adjust">bg.adjust</a></code> is used to background correct the
probe signals.  The default is to background correct PM signals only.
</p>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from BackgroundCorrection</b>:<br />
getParameters, process
</p>
<p><b>Methods inherited from ProbeLevelTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Jitter noise</h3>

<p>The fitting algorithm of the RMA background correction model may not
converge if there too many small and discrete signals.  To overcome
this problem, a small amount of noise may be added to the signals
before fitting the model.  This is an ad hoc solution that seems to
work.
However, add Gaussian noise may generate non-positive signals.
</p>


<h3>Author(s)</h3>

<p>Ken Simpson, Henrik Bengtsson</p>

<hr>
<h2 id='RmaCnPlm'>The RmaCnPlm class</h2><span id='topic+RmaCnPlm'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class RmaCnPlm</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+RmaPlm">RmaPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+RmaSnpPlm">RmaSnpPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+SnpPlm">SnpPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+CnPlm">CnPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>RmaCnPlm</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>RmaCnPlm</b><br />
extends <a href="#topic+CnPlm">CnPlm</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RmaCnPlm(..., combineAlleles=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RmaCnPlm_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+RmaSnpPlm">RmaSnpPlm</a></code>.</p>
</td></tr>
<tr><td><code id="RmaCnPlm_+3A_combinealleles">combineAlleles</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">FALSE</a></code>, allele A and allele B are treated
separately, otherwise together.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from CnPlm</b>:<br />
getCellIndices, getChipEffectSet, getCombineAlleles, getParameters, getProbeAffinityFile, setCombineAlleles
</p>
<p><b>Methods inherited from SnpPlm</b>:<br />
getCellIndices, getChipEffectSet, getMergeStrands, getParameters, getProbeAffinityFile, setMergeStrands
</p>
<p><b>Methods inherited from RmaSnpPlm</b>:<br />
getAsteriskTags
</p>
<p><b>Methods inherited from RmaPlm</b>:<br />
getAsteriskTags, getCalculateResidualsFunction, getParameters, getRlmFitFunctions
</p>
<p><b>Methods inherited from ProbeLevelModel</b>:<br />
calculateResidualSet, calculateWeights, fit, getAsteriskTags, getCalculateResidualsFunction, getChipEffectSet, getProbeAffinityFile, getResidualSet, getRootPath, getWeightsSet
</p>
<p><b>Methods inherited from MultiArrayUnitModel</b>:<br />
getListOfPriors, setListOfPriors, validate
</p>
<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Model</h3>

<p>TO DO.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='RmaPlm'>The RmaPlm class</h2><span id='topic+RmaPlm'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class RmaPlm</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>RmaPlm</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+ExonRmaPlm">ExonRmaPlm</a></em>, <em><a href="#topic+HetLogAddCnPlm">HetLogAddCnPlm</a></em>, <em><a href="#topic+HetLogAddPlm">HetLogAddPlm</a></em>, <em><a href="#topic+HetLogAddSnpPlm">HetLogAddSnpPlm</a></em>, <em><a href="#topic+RmaCnPlm">RmaCnPlm</a></em>, <em><a href="#topic+RmaSnpPlm">RmaSnpPlm</a></em><br />
</p>
<p>public abstract static class <b>RmaPlm</b><br />
extends <em><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></em><br />
</p>
<p>This class represents the log-additive model part of the Robust Multichip
Analysis (RMA) method described in Irizarry et al (2003).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RmaPlm(..., flavor=c("affyPLM", "oligo"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RmaPlm_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code>.</p>
</td></tr>
<tr><td><code id="RmaPlm_+3A_flavor">flavor</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying what model fitting algorithm
to be used.  This makes it possible to get identical estimates as other
packages.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from ProbeLevelModel</b>:<br />
calculateResidualSet, calculateWeights, fit, getAsteriskTags, getCalculateResidualsFunction, getChipEffectSet, getProbeAffinityFile, getResidualSet, getRootPath, getWeightsSet
</p>
<p><b>Methods inherited from MultiArrayUnitModel</b>:<br />
getListOfPriors, setListOfPriors, validate
</p>
<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Model</h3>

<p>For a single unit group, the log-additive model of RMA is:
</p>
<p style="text-align: center;"><code class="reqn">log_2(y_{ik}) = \beta_i + \alpha_k + \varepsilon_{ik}</code>
</p>

<p>where <code class="reqn">\beta_i</code> are the chip effects for arrays <code class="reqn">i=1,...,I</code>,
and <code class="reqn">\alpha_k</code> are the probe affinities for probes <code class="reqn">k=1,...,K</code>.
The <code class="reqn">\varepsilon_{ik}</code> are zero-mean noise with equal variance.
The model is constrained such that <code class="reqn">\sum_k{\alpha_k} = 0</code>.
</p>
<p>Note that all PLM classes must return parameters on the intensity scale.
For this class that means that <code class="reqn">\theta_i = 2^\beta_i</code> and
<code class="reqn">\phi_k = 2^\alpha_k</code> are returned.
</p>


<h3>Different flavors of model fitting</h3>

<p>There are a few differ algorithms available for fitting the same
probe-level model.  The default and recommended method
(<code>flavor="affyPLM"</code>) uses the implementation in the
<span class="pkg">preprocessCore</span> package which fits the model parameters robustly
using an M-estimator (the method used to be in <span class="pkg">affyPLM</span>).
</p>
<p>Alternatively, other model-fitting algorithms are available.
The algorithm (<code>flavor="oligo"</code>) used by the <span class="pkg">oligo</span> package,
which originates from the <span class="pkg">affy</span> packages, fits the model using
median polish, which is a non-robust estimator.  Note that this algorithm
does not constraint the probe-effect parameters to multiply to one on
the intensity scale.  Since the internal function does not return these
estimates, we can neither rescale them.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson, Ken Simpson</p>


<h3>References</h3>

<p>Irizarry et al. <em>Summaries of Affymetrix GeneChip probe level data</em>.
NAR, 2003, 31, e15.<br />
</p>

<hr>
<h2 id='RmaSnpPlm'>The RmaSnpPlm class</h2><span id='topic+RmaSnpPlm'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class RmaSnpPlm</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+RmaPlm">RmaPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+SnpPlm">SnpPlm</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>RmaSnpPlm</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+RmaCnPlm">RmaCnPlm</a></em><br />
</p>
<p>public abstract static class <b>RmaSnpPlm</b><br />
extends <a href="#topic+SnpPlm">SnpPlm</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RmaSnpPlm(..., mergeStrands=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RmaSnpPlm_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+RmaPlm">RmaPlm</a></code>.</p>
</td></tr>
<tr><td><code id="RmaSnpPlm_+3A_mergestrands">mergeStrands</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the sense and the anti-sense strands are
fitted together, otherwise separately.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from SnpPlm</b>:<br />
getCellIndices, getChipEffectSet, getMergeStrands, getParameters, getProbeAffinityFile, setMergeStrands
</p>
<p><b>Methods inherited from RmaPlm</b>:<br />
getAsteriskTags, getCalculateResidualsFunction, getParameters, getRlmFitFunctions
</p>
<p><b>Methods inherited from ProbeLevelModel</b>:<br />
calculateResidualSet, calculateWeights, fit, getAsteriskTags, getCalculateResidualsFunction, getChipEffectSet, getProbeAffinityFile, getResidualSet, getRootPath, getWeightsSet
</p>
<p><b>Methods inherited from MultiArrayUnitModel</b>:<br />
getListOfPriors, setListOfPriors, validate
</p>
<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='ScaleNormalization'>The ScaleNormalization class</h2><span id='topic+ScaleNormalization'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class ScaleNormalization</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code>ScaleNormalization</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>ScaleNormalization</b><br />
extends <em><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></em><br />
</p>
<p>This class represents a normalization function that transforms the
probe-level signals towards the same scale.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ScaleNormalization(dataSet=NULL, ..., targetAvg=4400, subsetToUpdate=NULL,
  typesToUpdate=NULL, subsetToAvg="-XY", typesToAvg=typesToUpdate, shift=0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ScaleNormalization_+3A_dataset">dataSet</code></td>
<td>
<p><code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code> to be normalized.</p>
</td></tr>
<tr><td><code id="ScaleNormalization_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code>.</p>
</td></tr>
<tr><td><code id="ScaleNormalization_+3A_targetavg">targetAvg</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> value.</p>
</td></tr>
<tr><td><code id="ScaleNormalization_+3A_subsettoupdate">subsetToUpdate</code></td>
<td>
<p>The probes to be updated.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all probes are updated.</p>
</td></tr>
<tr><td><code id="ScaleNormalization_+3A_typestoupdate">typesToUpdate</code></td>
<td>
<p>Types of probes to be updated.</p>
</td></tr>
<tr><td><code id="ScaleNormalization_+3A_subsettoavg">subsetToAvg</code></td>
<td>
<p>The probes to calculate average signal over.
If a single <code><a href="base.html#topic+numeric">numeric</a></code> in (0,1), then this
fraction of all probes will be used.
If <code><a href="base.html#topic+NULL">NULL</a></code>, all probes are considered.</p>
</td></tr>
<tr><td><code id="ScaleNormalization_+3A_typestoavg">typesToAvg</code></td>
<td>
<p>Types of probes to be used when calculating the
average signal.
If <code>"pm"</code> and <code>"mm"</code> only perfect-match and mismatch
probes are used, respectively. If <code>"pmmm"</code> both types are used.
</p>
</td></tr>
<tr><td><code id="ScaleNormalization_+3A_shift">shift</code></td>
<td>
<p>Optional amount of shift if data before fitting/normalizing.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ProbeLevelTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='ScaleNormalization3'>The ScaleNormalization3 class</h2><span id='topic+ScaleNormalization3'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class ScaleNormalization3</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform3">ProbeLevelTransform3</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>ScaleNormalization3</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>ScaleNormalization3</b><br />
extends <em><a href="#topic+ProbeLevelTransform3">ProbeLevelTransform3</a></em><br />
</p>
<p>This class represents a normalization function that transforms the
probe-level signals towards the same scale.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ScaleNormalization3(..., targetAvg=4400)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ScaleNormalization3_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+ProbeLevelTransform3">ProbeLevelTransform3</a></code>.</p>
</td></tr>
<tr><td><code id="ScaleNormalization3_+3A_targetavg">targetAvg</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> value.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ProbeLevelTransform3</b>:<br />
getAsteriskTags, getCellsTo, getCellsToFit, getCellsToUpdate, getParameters, getUnitsTo, getUnitsToFit, getUnitsToUpdate, writeSignals
</p>
<p><b>Methods inherited from ProbeLevelTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='setAlias.GenericReporter'>Sets the alias of the report</h2><span id='topic+setAlias.GenericReporter'></span><span id='topic+GenericReporter.setAlias'></span><span id='topic+setAlias+2CGenericReporter-method'></span>

<h3>Description</h3>

<p>Sets the alias of the report.
If specified, the alias overrides the report name, which is used by
default.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GenericReporter'
setAlias(this, alias=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setAlias.GenericReporter_+3A_alias">alias</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string for the new alias of the report.
The alias must consists of valid filename characters, and must not
contain commas, which are used to separate tags.</p>
</td></tr>
<tr><td><code id="setAlias.GenericReporter_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns nothing.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getAlias.GenericReporter">*getAlias</a>()</code>.
<code><a href="#topic+getName.GenericReporter">*getName</a>()</code>.
For more information see <code><a href="#topic+GenericReporter">GenericReporter</a></code>.
</p>

<hr>
<h2 id='setAlias.Model'>Sets the name alias for the model</h2><span id='topic+setAlias.Model'></span><span id='topic+Model.setAlias'></span><span id='topic+setAlias+2CModel-method'></span>

<h3>Description</h3>

<p>Sets the name alias for the model, if set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Model'
setAlias(this, alias=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setAlias.Model_+3A_alias">alias</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string, or <code><a href="base.html#topic+NULL">NULL</a></code>.</p>
</td></tr>
<tr><td><code id="setAlias.Model_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns nothing.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getAlias.Model">*getAlias</a>()</code>
For more information see <code><a href="#topic+Model">Model</a></code>.
</p>

<hr>
<h2 id='setArrays.ArrayExplorer'>Sets the arrays</h2><span id='topic+setArrays.ArrayExplorer'></span><span id='topic+ArrayExplorer.setArrays'></span><span id='topic+setArrays+2CArrayExplorer-method'></span>

<h3>Description</h3>

<p>Sets the arrays to be processed by the explorer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ArrayExplorer'
setArrays(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setArrays.ArrayExplorer_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+ArrayExplorer">ArrayExplorer</a></code>.
</p>

<hr>
<h2 id='setCdf.AffymetrixCelFile'>Sets the CDF structure for this CEL file</h2><span id='topic+setCdf.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.setCdf'></span><span id='topic+setCdf+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Sets the CDF structure for this CEL file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelFile'
setCdf(this, cdf, ..., .checkArgs=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setCdf.AffymetrixCelFile_+3A_cdf">cdf</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> object.</p>
</td></tr>
<tr><td><code id="setCdf.AffymetrixCelFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="setCdf.AffymetrixCelFile_+3A_.checkargs">.checkArgs</code></td>
<td>
<p>(Internal) If <code><a href="base.html#topic+logical">FALSE</a></code>, arguments are not validated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns nothing.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+getCdf.AffymetrixCelFile">*getCdf</a>()</code>.
For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>

<hr>
<h2 id='setCdf.AffymetrixCelSet'>Sets the CDF structure for this CEL set</h2><span id='topic+setCdf.AffymetrixCelSet'></span><span id='topic+AffymetrixCelSet.setCdf'></span><span id='topic+setCdf+2CAffymetrixCelSet-method'></span>

<h3>Description</h3>

<p>Sets the CDF structure for this CEL set.  This structures is assigned to all CEL files in the set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelSet'
setCdf(this, cdf, verbose=FALSE, ..., .checkArgs=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setCdf.AffymetrixCelSet_+3A_cdf">cdf</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> object.</p>
</td></tr>
<tr><td><code id="setCdf.AffymetrixCelSet_+3A_verbose">verbose</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, progress details are printed, otherwise not.
May also be a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> object.</p>
</td></tr>
<tr><td><code id="setCdf.AffymetrixCelSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="setCdf.AffymetrixCelSet_+3A_.checkargs">.checkArgs</code></td>
<td>
<p>(Internal) If <code><a href="base.html#topic+logical">FALSE</a></code>, arguments are not validated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns nothing.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+getCdf.AffymetrixCelSet">*getCdf</a>()</code>.
For more information see <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.
</p>

<hr>
<h2 id='setCdf.CnagCfhFile'>Sets the CDF structure for this CEL file</h2><span id='topic+setCdf.CnagCfhFile'></span><span id='topic+CnagCfhFile.setCdf'></span><span id='topic+setCdf+2CCnagCfhFile-method'></span>

<h3>Description</h3>

<p>Sets the CDF structure for this CEL file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CnagCfhFile'
setCdf(this, cdf, ..., .checkArgs=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setCdf.CnagCfhFile_+3A_cdf">cdf</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> object.</p>
</td></tr>
<tr><td><code id="setCdf.CnagCfhFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns nothing.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getCdf.CnagCfhFile">*getCdf</a>()</code>.
For more information see <code><a href="#topic+CnagCfhFile">CnagCfhFile</a></code>.
</p>

<hr>
<h2 id='setCdf.CnagCfhSet'>Sets the CDF structure for this CFH set</h2><span id='topic+setCdf.CnagCfhSet'></span><span id='topic+CnagCfhSet.setCdf'></span><span id='topic+setCdf+2CCnagCfhSet-method'></span>

<h3>Description</h3>

<p>Sets the CDF structure for this CFH set.  This structures is assigned to all CFH files in the set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CnagCfhSet'
setCdf(this, cdf, verbose=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setCdf.CnagCfhSet_+3A_cdf">cdf</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code> object.</p>
</td></tr>
<tr><td><code id="setCdf.CnagCfhSet_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns nothing.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getCdf.CnagCfhSet">*getCdf</a>()</code>.
For more information see <code><a href="#topic+CnagCfhSet">CnagCfhSet</a></code>.
</p>

<hr>
<h2 id='setRestructor.AffymetrixCdfFile'>Specifies a function through which</h2><span id='topic+setRestructor.AffymetrixCdfFile'></span><span id='topic+AffymetrixCdfFile.setRestructor'></span><span id='topic+setRestructor+2CAffymetrixCdfFile-method'></span>

<h3>Description</h3>

<p>Specifies a function through which of all or a subset of the units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCdfFile'
setRestructor(this, fcn=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setRestructor.AffymetrixCdfFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to
<code><a href="affxparser.html#topic+readCdfCellIndices">readCdfCellIndices</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the <code><a href="base.html#topic+list">list</a></code> structure returned by
<code><a href="affxparser.html#topic+readCdfCellIndices">readCdfCellIndices</a></code>.
</p>


<h3>Requirements</h3>

<p>The reconstructor function <em>have to</em>:
</p>

<ol>
<li><p> Accept a <code><a href="base.html#topic+list">list</a></code> structure as its first argument.
</p>
</li>
<li><p> Require no other arguments.
</p>
</li>
<li><p> Return a <code><a href="base.html#topic+list">list</a></code> structure of identical length as the input one.
In other words, it must not change the number of units nor
the order of units.
</p>
</li></ol>

<p>The reconstructor function <em>may</em>:
</p>

<ol>
<li><p> Rearrange the groups or change the number of groups, for
instance by utilizing <code><a href="affxparser.html#topic+applyCdfGroups">applyCdfGroups</a></code>.
</p>
</li>
<li><p> Exclude some cells, for instance by excluding MMs.
</p>
</li></ol>



<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.
</p>

<hr>
<h2 id='setTags.Model'>Sets the tags to be appended</h2><span id='topic+setTags.Model'></span><span id='topic+Model.setTags'></span><span id='topic+setTags+2CModel-method'></span>

<h3>Description</h3>

<p>Sets the tags to be appended to the tags of the input data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Model'
setTags(this, tags=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setTags.Model_+3A_tags">tags</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> of tags.
The tags may also be passed as comma-separated strings.</p>
</td></tr>
<tr><td><code id="setTags.Model_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns nothing.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getTags.Model">*getTags</a>()</code>
For more information see <code><a href="#topic+Model">Model</a></code>.
</p>

<hr>
<h2 id='setupExampleData.AromaAffymetrix'>Setups example data in the current directory</h2><span id='topic+setupExampleData.AromaAffymetrix'></span><span id='topic+AromaAffymetrix.setupExampleData'></span><span id='topic+setupExampleData+2CAromaAffymetrix-method'></span><span id='topic+setupExampleData'></span>

<h3>Description</h3>

<p>Setups example data in the current directory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AromaAffymetrix'
setupExampleData(pkg, dataset=NULL, chipType=NULL, dirs=c("annotationData", "rawData"),
  mustWork=TRUE, validate=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setupExampleData.AromaAffymetrix_+3A_dataset">dataset</code>, <code id="setupExampleData.AromaAffymetrix_+3A_chiptype">chipType</code></td>
<td>
<p><code><a href="base.html#topic+character">character</a></code> strings specifying the data set
and the chip type to install.</p>
</td></tr>
<tr><td><code id="setupExampleData.AromaAffymetrix_+3A_dirs">dirs</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> specifying which directories to setup.</p>
</td></tr>
<tr><td><code id="setupExampleData.AromaAffymetrix_+3A_mustwork">mustWork</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, an error is thrown if the requested data set
could not be installed, otherwise not.</p>
</td></tr>
<tr><td><code id="setupExampleData.AromaAffymetrix_+3A_validate">validate</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the installed files are also validated,
otherwise not.</p>
</td></tr>
<tr><td><code id="setupExampleData.AromaAffymetrix_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns (invisibly) <code><a href="base.html#topic+logical">TRUE</a></code> if all requested data was installed,
otherwise <code><a href="base.html#topic+logical">FALSE</a></code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='SingleArrayUnitModel'>The SingleArrayUnitModel class</h2><span id='topic+SingleArrayUnitModel'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class SingleArrayUnitModel</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+UnitModel">UnitModel</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><em><code>SingleArrayUnitModel</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>SingleArrayUnitModel</b><br />
extends <em><a href="#topic+UnitModel">UnitModel</a></em><br />
</p>
<p>This abstract class represents a unit model that fits one model per unit
based on signals from a single arrays.
The nature of a single-array unit model is that each array can be fitted
independently of the others.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SingleArrayUnitModel(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SingleArrayUnitModel_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+UnitModel">UnitModel</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>fit</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from UnitModel</b>:<br />
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='SingleArrayUnitModel+24getFitUnitGroupFunction'>Static method to get the low-level function that fits the PLM</h2><span id='topic+SingleArrayUnitModel+24getFitUnitGroupFunction'></span><span id='topic+getFitUnitGroupFunction.SingleArrayUnitModel'></span><span id='topic+SingleArrayUnitModel.getFitUnitGroupFunction'></span><span id='topic+getFitUnitGroupFunction+2CSingleArrayUnitModel-method'></span>

<h3>Description</h3>

<p>Static method to get the low-level function that fits the PLM.
Any subclass model must provide this method, which should return
a <code><a href="base.html#topic+function">function</a></code> that accepts a <code><a href="base.html#topic+numeric">numeric</a></code> <code><a href="base.html#topic+vector">vector</a></code> of length K, where K
is the number of probes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## SingleArrayUnitModel$getFitUnitGroupFunction()

## Don't use the below:
## S3 method for class 'SingleArrayUnitModel'
getFitUnitGroupFunction(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SingleArrayUnitModel+2B24getFitUnitGroupFunction_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="base.html#topic+function">function</a></code>.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+SingleArrayUnitModel">SingleArrayUnitModel</a></code>.
</p>

<hr>
<h2 id='SmoothMultiarrayModel'>The SmoothMultiarrayModel class</h2><span id='topic+SmoothMultiarrayModel'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class SmoothMultiarrayModel</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ChromosomalModel">ChromosomalModel</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><em><code>SmoothMultiarrayModel</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+SmoothRmaModel">SmoothRmaModel</a></em>, <em><a href="#topic+SmoothSaModel">SmoothSaModel</a></em><br />
</p>
<p>public abstract static class <b>SmoothMultiarrayModel</b><br />
extends <em><a href="aroma.core.html#topic+ChromosomalModel">ChromosomalModel</a></em><br />
</p>
<p>This abstract class represents a chromosomal smoothing method done
chromosome by chromosome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SmoothMultiarrayModel(..., typoOfWeights=c("none", "1/s2"), bandwidth=10000, tags="*")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SmoothMultiarrayModel_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of <code><a href="aroma.core.html#topic+ChromosomalModel">ChromosomalModel</a></code>.</p>
</td></tr>
<tr><td><code id="SmoothMultiarrayModel_+3A_typoofweights">typoOfWeights</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string.</p>
</td></tr>
<tr><td><code id="SmoothMultiarrayModel_+3A_bandwidth">bandwidth</code></td>
<td>
<p>A single <code><a href="base.html#topic+numeric">numeric</a></code> specifying the smoothing bandwidth
in units of nucleotides.</p>
</td></tr>
<tr><td><code id="SmoothMultiarrayModel_+3A_tags">tags</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> of tags to be added.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getBandwidth</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getOutputTuple</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setBandwidth</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ChromosomalModel</b>:<br />
as.character, fit, getAlias, getAromaGenomeTextFile, getAsteriskTags, getChipType, getChipTypes, getChromosomes, getFullName, getFullNames, getGenome, getGenomeData, getGenomeFile, getListOfAromaUgpFiles, getName, getNames, getParentPath, getPath, getReportPath, getRootPath, getSetTuple, getSets, getTags, indexOf, nbrOfArrays, nbrOfChipTypes, setChromosomes, setGenome, getListOfGenomeInformations, getPcuTheta, getPositionChipTypeUnit
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="aroma.core.html#topic+CopyNumberSegmentationModel">CopyNumberSegmentationModel</a></code>.
</p>

<hr>
<h2 id='SmoothRmaModel'>The SmoothRmaModel class</h2><span id='topic+SmoothRmaModel'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class SmoothRmaModel</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ChromosomalModel">ChromosomalModel</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+SmoothMultiarrayModel">SmoothMultiarrayModel</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><em><code>SmoothRmaModel</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>SmoothRmaModel</b><br />
extends <em><a href="#topic+SmoothMultiarrayModel">SmoothMultiarrayModel</a></em><br />
</p>
<p>This class represents the Chromosomal Smoothing Robust Multichip Analysis
method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SmoothRmaModel(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SmoothRmaModel_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+SmoothMultiarrayModel">SmoothMultiarrayModel</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from SmoothMultiarrayModel</b>:<br />
as.character, createOutputTuple, fitOneChromosome, getAsteriskTags, getBandwidth, getFitUnitGroupFunction, getOutputTuple, getRootPath, setBandwidth
</p>
<p><b>Methods inherited from ChromosomalModel</b>:<br />
as.character, fit, getAlias, getAromaGenomeTextFile, getAsteriskTags, getChipType, getChipTypes, getChromosomes, getFullName, getFullNames, getGenome, getGenomeData, getGenomeFile, getListOfAromaUgpFiles, getName, getNames, getParentPath, getPath, getReportPath, getRootPath, getSetTuple, getSets, getTags, indexOf, nbrOfArrays, nbrOfChipTypes, setChromosomes, setGenome, getListOfGenomeInformations, getPcuTheta, getPositionChipTypeUnit
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='smoothScatterMvsA.AffymetrixCelFile'>Plots log-ratio versus log-intensity in a smooth scatter plot</h2><span id='topic+smoothScatterMvsA.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.smoothScatterMvsA'></span><span id='topic+smoothScatterMvsA+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Plots log-ratio versus log-intensity in a smooth scatter plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelFile'
smoothScatterMvsA(this, reference, indices=NULL, pch=176, xlim=c(0, 16),
  ylim=c(-1, 1) * diff(xlim), xlab=expression(A == 1/2 %*% log[2](y[1] * y[2])),
  ylab=expression(M == log[2](y[1]/y[2])), ..., annotate=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="smoothScatterMvsA.AffymetrixCelFile_+3A_reference">reference</code></td>
<td>
<p>The reference channel, i.e. the denominator in the
log ratios.</p>
</td></tr>
<tr><td><code id="smoothScatterMvsA.AffymetrixCelFile_+3A_indices">indices</code></td>
<td>
<p>Indices of the probe signals to be plotted.</p>
</td></tr>
<tr><td><code id="smoothScatterMvsA.AffymetrixCelFile_+3A_pch">pch</code></td>
<td>
<p>The plot symbol.</p>
</td></tr>
<tr><td><code id="smoothScatterMvsA.AffymetrixCelFile_+3A_xlim">xlim</code>, <code id="smoothScatterMvsA.AffymetrixCelFile_+3A_ylim">ylim</code></td>
<td>
<p>The range of the x and the y axes.</p>
</td></tr>
<tr><td><code id="smoothScatterMvsA.AffymetrixCelFile_+3A_xlab">xlab</code>, <code id="smoothScatterMvsA.AffymetrixCelFile_+3A_ylab">ylab</code></td>
<td>
<p>The labels on the x and the y axes.</p>
</td></tr>
<tr><td><code id="smoothScatterMvsA.AffymetrixCelFile_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="graphics.html#topic+plot">plot</a></code>().</p>
</td></tr>
<tr><td><code id="smoothScatterMvsA.AffymetrixCelFile_+3A_annotate">annotate</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, the plot is annotated with information about
the data plotted, otherwise not.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns (invisibly) a <code><a href="base.html#topic+data.frame">data.frame</a></code> with the plotted data in the
first two columns.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotMvsA.AffymetrixCelFile">*plotMvsA</a>()</code>.
Internally <code><a href="graphics.html#topic+smoothScatter">smoothScatter</a></code> is used.
For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>

<hr>
<h2 id='SnpChipEffectFile'>The SnpChipEffectFile class</h2><span id='topic+SnpChipEffectFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class SnpChipEffectFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ParameterCelFile">ParameterCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ChipEffectFile">ChipEffectFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>SnpChipEffectFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+CnChipEffectFile">CnChipEffectFile</a></em><br />
</p>
<p>public abstract static class <b>SnpChipEffectFile</b><br />
extends <em><a href="#topic+ChipEffectFile">ChipEffectFile</a></em><br />
</p>
<p>This class represents estimates of chip effects in the probe-level models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SnpChipEffectFile(..., mergeStrands=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SnpChipEffectFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+ChipEffectFile">ChipEffectFile</a></code>.</p>
</td></tr>
<tr><td><code id="SnpChipEffectFile_+3A_mergestrands">mergeStrands</code></td>
<td>
<p>Specifies if the strands are merged or not for these
estimates.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractTheta</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractTotalAndFracB</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ChipEffectFile</b>:<br />
as.character, extractChromosomalDataFrame, extractMatrix, extractTheta, findUnitsTodo, getAM, getAsFullCelFile, getCellIndices, getCellMapForMainCdf, getExpandedCellMap, getParameters, getUnitGroupCellArrayMap, getUnitGroupCellMatrixMap, getXAM, mergeGroups, readUnits, writeAsFullCelFile
</p>
<p><b>Methods inherited from ParameterCelFile</b>:<br />
extractDataFrame, extractMatrix, readUnits
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from AffymetrixCelFile</b>:<br />
allocateFromCdf, as.character, clone, createFrom, extractMatrix, fromFile, getAm, getCdf, getExtensionPattern, getFileFormat, getImage, getUnitNamesFile, getUnitTypesFile, highlight, image270, nbrOfCells, plotDensity, plotImage, plotMvsA, plotMvsX, range, setCdf, smoothScatterMvsA, writeImage
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>An object of this class is typically part of a <code><a href="#topic+SnpChipEffectSet">SnpChipEffectSet</a></code>.
</p>

<hr>
<h2 id='SnpChipEffectGroupMerge'>The SnpChipEffectGroupMerge class</h2><span id='topic+SnpChipEffectGroupMerge'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class SnpChipEffectGroupMerge</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ChipEffectTransform">ChipEffectTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ChipEffectGroupMerge">ChipEffectGroupMerge</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>SnpChipEffectGroupMerge</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>SnpChipEffectGroupMerge</b><br />
extends <a href="#topic+ChipEffectGroupMerge">ChipEffectGroupMerge</a><br />
</p>
<p>This class represents a method that merges SNP chip effects across groups
unit by unit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SnpChipEffectGroupMerge(..., mergeStrands=FALSE, mean=c("arithmetic", "geometric"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SnpChipEffectGroupMerge_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+ChipEffectGroupMerge">ChipEffectGroupMerge</a></code>.</p>
</td></tr>
<tr><td><code id="SnpChipEffectGroupMerge_+3A_mergestrands">mergeStrands</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, group strands are merged.</p>
</td></tr>
<tr><td><code id="SnpChipEffectGroupMerge_+3A_mean">mean</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying what type of averaging
should be applied.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getMergeFunction</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ChipEffectGroupMerge</b>:<br />
getCdf, getMergeFunction, getParameters, process, setMergeFunction
</p>
<p><b>Methods inherited from ChipEffectTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='SnpChipEffectSet'>The SnpChipEffectSet class</h2><span id='topic+SnpChipEffectSet'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class SnpChipEffectSet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataSet">AromaMicroarrayDataSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ParameterCelSet">ParameterCelSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>SnpChipEffectSet</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<a href="#topic+CnChipEffectSet">CnChipEffectSet</a><br />
</p>
<p>public static class <b>SnpChipEffectSet</b><br />
extends <a href="#topic+ChipEffectSet">ChipEffectSet</a><br />
</p>
<p>This class represents estimates of chip effects in the probe-level models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SnpChipEffectSet(..., mergeStrands="byFirstFile")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SnpChipEffectSet_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+ChipEffectSet">ChipEffectSet</a></code>.</p>
</td></tr>
<tr><td><code id="SnpChipEffectSet_+3A_mergestrands">mergeStrands</code></td>
<td>
<p>Specifies if the strands are merged or not for these
estimates.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractAlleleSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractSnpCnvQSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractSnpQSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractTheta</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>extractTotalAndFreqB</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAverageFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getMergeStrands</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setMergeStrands</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ChipEffectSet</b>:<br />
as.character, boxplotStats, byPath, calculateBaseline, calculateFieldBoxplotStats, calculateNuseBoxplotStats, calculateRleBoxplotStats, extractAffyBatch, extractChromosomalDataFrame, extractExpressionSet, extractMatrix, extractTheta, findByName, findUnitsTodo, fromDataSet, getAM, getAsFullCelSet, getAverageFile, getBaseline, getCellIndices, getXAM, plotBoxplot, readUnits, updateUnits
</p>
<p><b>Methods inherited from ParameterCelSet</b>:<br />
extractDataFrame, extractMatrix
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from AffymetrixCelSet</b>:<br />
append, as, as.AffymetrixCelSet, as.character, averageQuantile, byName, byPath, clone, convertToUnique, doCRMAv1, doCRMAv2, doFIRMA, doGCRMA, doRMA, extractAffyBatch, extractFeatureSet, extractMatrix, extractSnpFeatureSet, findByName, getAverage, getAverageAsinh, getAverageFile, getAverageLog, getCdf, getChipType, getData, getIntensities, getPlatform, getTimestamps, getUnitGroupCellMap, getUnitIntensities, getUnitNamesFile, getUnitTypesFile, isDuplicated, justRMA, justSNPRMA, nbrOfArrays, normalizeQuantile, plotDensity, range, readUnits, setCdf, update2, writeSgr
</p>
<p><b>Methods inherited from AffymetrixFileSet</b>:<br />
as, as.AffymetrixFileSet, byPath, getDefaultFullName
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataSet</b>:<br />
as.AromaMicroarrayDataSetList, as.AromaMicroarrayDataSetTuple, getAromaFullNameTranslatorSet, getAverageFile, getChipType, getDefaultFullName, getPlatform, setAttributesBy, setAttributesBySampleAnnotationFile, setAttributesBySampleAnnotationSet, validate
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='SnpInformation'>The SnpInformation class</h2><span id='topic+SnpInformation'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class SnpInformation</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>SnpInformation</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+DChipSnpInformation">DChipSnpInformation</a></em>, <em><a href="#topic+UflSnpInformation">UflSnpInformation</a></em><br />
</p>
<p>public abstract static class <b>SnpInformation</b><br />
extends <a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SnpInformation(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SnpInformation_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getData</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFragmentLengths</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFragmentStarts</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFragmentStops</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfEnzymes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readDataFrame</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='SnpInformation+24fromCdf'>Static method to define a genome information set from a CDF</h2><span id='topic+SnpInformation+24fromCdf'></span><span id='topic+fromCdf.SnpInformation'></span><span id='topic+SnpInformation.fromCdf'></span><span id='topic+fromCdf+2CSnpInformation-method'></span>

<h3>Description</h3>

<p>Static method to define a genome information set from a CDF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## SnpInformation$fromCdf(cdf, ...)

## Don't use the below:
## S3 method for class 'SnpInformation'
fromCdf(static, cdf, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SnpInformation+2B24fromCdf_+3A_cdf">cdf</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixCdfFile">AffymetrixCdfFile</a></code>.</p>
</td></tr>
<tr><td><code id="SnpInformation+2B24fromCdf_+3A_...">...</code></td>
<td>
<p>Additional argument passed to <code><a href="#topic+byChipType.SnpInformation">*byChipType</a>()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code><a href="#topic+SnpInformation">SnpInformation</a></code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+byChipType.SnpInformation">*byChipType</a>()</code>.
For more information see <code><a href="#topic+SnpInformation">SnpInformation</a></code>.
</p>

<hr>
<h2 id='SnpPlm'>The SnpPlm interface class</h2><span id='topic+SnpPlm'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class SnpPlm</b><br />
</p>
<p><code><a href="R.oo.html#topic+Interface">Interface</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code>SnpPlm</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AffineCnPlm">AffineCnPlm</a></em>, <em><a href="#topic+AffineSnpPlm">AffineSnpPlm</a></em>, <em><a href="#topic+AvgCnPlm">AvgCnPlm</a></em>, <em><a href="#topic+AvgSnpPlm">AvgSnpPlm</a></em>, <a href="#topic+CnPlm">CnPlm</a>, <em><a href="#topic+HetLogAddCnPlm">HetLogAddCnPlm</a></em>, <em><a href="#topic+HetLogAddSnpPlm">HetLogAddSnpPlm</a></em>, <em><a href="#topic+MbeiCnPlm">MbeiCnPlm</a></em>, <em><a href="#topic+MbeiSnpPlm">MbeiSnpPlm</a></em>, <em><a href="#topic+RmaCnPlm">RmaCnPlm</a></em>, <em><a href="#topic+RmaSnpPlm">RmaSnpPlm</a></em><br />
</p>
<p>public class <b>SnpPlm</b><br />
extends <a href="R.oo.html#topic+Interface">Interface</a><br />
</p>
<p>An <code><a href="R.oo.html#topic+Interface">Interface</a></code> implementing methods special for
<code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code>s specific to SNP arrays.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SnpPlm(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SnpPlm_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCellIndices</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipEffectSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getMergeStrands</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getProbeAffinityFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setMergeStrands</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from Interface</b>:<br />
extend, print, uses
</p>


<h3>Requirements</h3>

<p>Classes inheriting from this <code><a href="R.oo.html#topic+Interface">Interface</a></code> must provide the
following fields:
</p>

<dl>
<dt>mergeStrands</dt><dd><p>A <code><a href="base.html#topic+logical">logical</a></code> value indicating if strands should be
merged or not.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>Examples</h3>

<pre><code class='language-R'>for (zzz in 0) {

# Setup verbose output
verbose &lt;- Arguments$getVerbose(-2)
timestampOn(verbose)

# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Define an example dataset using this path
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Find any SNP dataset
path &lt;- NULL
if (is.null(path))
  break

if (!exists("ds")) {
  ds &lt;- AffymetrixCelSet$fromFiles(path)
}
print(ds)

# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Create a set of various PLMs for this dataset
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
if (!exists("models", mode="list")) {
  mergeStrands &lt;- TRUE
  models &lt;- list(
    rma = RmaSnpPlm(ds, mergeStrands=mergeStrands),
    mbei = MbeiSnpPlm(ds, mergeStrands=mergeStrands)
#   affine = AffineSnpPlm(ds, background=FALSE, mergeStrands=mergeStrands)
  )
}
print(models)


# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# For each model, fit a few units
#
# Note, by fitting the same set of units across models, the internal
# caching mechanisms of aroma.affymetrix makes sure that the data is
# only read into memory once. See log for reading speed.
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
units &lt;- 55+1:100

for (model in models) {
  ruler(verbose)
  fit(model, units=units, force=TRUE, verbose=verbose)
}

# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# For each unit, plot the estimated (thetaB,thetaA) for all models
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Should we plot the on the log scale?
log &lt;- TRUE

# Do only user to press ENTER if more than one unit is plotted
opar &lt;- par(ask=(length(units) &gt; 1))

Alab &lt;- expression(theta[A])
Blab &lt;- expression(theta[B])
if (log) {
  lim &lt;- c(6, 16)
} else {
  lim &lt;- c(0, 2^15)
}

# For each unit...
for (unit in units) {
  # For all models...
  for (kk in seq_along(models)) {
    ces &lt;- getChipEffects(models[[kk]])
    ceUnit &lt;- ces[unit]
    snpName &lt;- names(ceUnit)[1]
    theta &lt;- ceUnit[[1]]
    thetaA &lt;- theta[[1]]$theta
    thetaB &lt;- theta[[2]]$theta
    if (log) {
      thetaA &lt;- log(thetaA, base=2)
      thetaB &lt;- log(thetaB, base=2)
    }

    # Create the plot?
    if (kk == 1) {
      plot(NA, xlim=lim, ylim=lim, xlab=Blab, ylab=Alab, main=snpName)
      abline(a=0, b=1, lty=2)
    }

    # Plot the estimated parameters
    points(thetaB, thetaA, col=kk, pch=19)
  }
} # for (unit ...)

# Reset graphical parameter settings
par(opar)

} # for (zzz in 0)
rm(zzz)
</code></pre>

<hr>
<h2 id='SnpProbeAffinityFile'>The SnpProbeAffinityFile class</h2><span id='topic+SnpProbeAffinityFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class SnpProbeAffinityFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ParameterCelFile">ParameterCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeAffinityFile">ProbeAffinityFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>SnpProbeAffinityFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+CnProbeAffinityFile">CnProbeAffinityFile</a></em><br />
</p>
<p>public abstract static class <b>SnpProbeAffinityFile</b><br />
extends <em><a href="#topic+ProbeAffinityFile">ProbeAffinityFile</a></em><br />
</p>
<p>This class represents estimates of probe affinities in SNP probe-level models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SnpProbeAffinityFile(..., mergeStrands=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SnpProbeAffinityFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+ProbeAffinityFile">ProbeAffinityFile</a></code>.</p>
</td></tr>
<tr><td><code id="SnpProbeAffinityFile_+3A_mergestrands">mergeStrands</code></td>
<td>
<p>Specifies if the strands are merged or not for these
estimates.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
<em>No methods defined</em>.
</p>
<p><b>Methods inherited from ProbeAffinityFile</b>:<br />
as.character, getCellIndices, getParameters, readUnits
</p>
<p><b>Methods inherited from ParameterCelFile</b>:<br />
extractDataFrame, extractMatrix, readUnits
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from AffymetrixCelFile</b>:<br />
allocateFromCdf, as.character, clone, createFrom, extractMatrix, fromFile, getAm, getCdf, getExtensionPattern, getFileFormat, getImage, getUnitNamesFile, getUnitTypesFile, highlight, image270, nbrOfCells, plotDensity, plotImage, plotMvsA, plotMvsX, range, setCdf, smoothScatterMvsA, writeImage
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='SpatialReporter'>The SpatialReporter class</h2><span id='topic+SpatialReporter'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class SpatialReporter</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="#topic+GenericReporter">GenericReporter</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+AffymetrixFileSetReporter">AffymetrixFileSetReporter</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelSetReporter">AffymetrixCelSetReporter</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><em><code>SpatialReporter</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>SpatialReporter</b><br />
extends <em><a href="#topic+AffymetrixCelSetReporter">AffymetrixCelSetReporter</a></em><br />
</p>
<p>A SpatialReporter generates image files of spatial representations of
cell signals for each of the arrays in the input set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SpatialReporter(..., reference=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SpatialReporter_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AffymetrixCelSetReporter">AffymetrixCelSetReporter</a></code>.</p>
</td></tr>
<tr><td><code id="SpatialReporter_+3A_reference">reference</code></td>
<td>
<p>An optional reference <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>addColorMap</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getColorMaps</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plotMargins</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>setColorMaps</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from AffymetrixCelSetReporter</b>:<br />
as.character, getChipType, getDataSet, getPath, nbrOfArrays
</p>
<p><b>Methods inherited from AffymetrixFileSetReporter</b>:<br />
getFileSet, getInputName, getInputTags
</p>
<p><b>Methods inherited from GenericReporter</b>:<br />
as.character, getAlias, getAsteriskTags, getFullName, getInputName, getInputTags, getMainPath, getName, getPath, getReportSet, getRootPath, getTags, process, setAlias, setup
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='TransformReport'>The TransformReport class</h2><span id='topic+TransformReport'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class TransformReport</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code>TransformReport</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>TransformReport</b><br />
extends <a href="R.oo.html#topic+Object">Object</a><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TransformReport(inSet=NULL, outSet=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TransformReport_+3A_inset">inSet</code></td>
<td>
<p>The input data set as an <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.</p>
</td></tr>
<tr><td><code id="TransformReport_+3A_outset">outSet</code></td>
<td>
<p>The output data set as an <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.</p>
</td></tr>
<tr><td><code id="TransformReport_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getFullName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getInputDataSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getName</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getOutputDataSet</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getPath</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getTags</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitNamesFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitTypesFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getYY</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfArrays</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plotXYCurve</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>plotXYCurveLog2</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>seq</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>writeImageCombined</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>writeImages</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='UgpGenomeInformation'>The UgpGenomeInformation class</h2><span id='topic+UgpGenomeInformation'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class UgpGenomeInformation</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+GenomeInformation">GenomeInformation</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>UgpGenomeInformation</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>UgpGenomeInformation</b><br />
extends <em><a href="#topic+GenomeInformation">GenomeInformation</a></em><br />
</p>
<p>This class represents Aroma UGP genome information files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UgpGenomeInformation(..., .ugp=NULL, .verify=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="UgpGenomeInformation_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+GenomeInformation">GenomeInformation</a></code>.</p>
</td></tr>
<tr><td><code id="UgpGenomeInformation_+3A_.ugp">.ugp</code></td>
<td>
<p>For internal use only.</p>
</td></tr>
<tr><td><code id="UgpGenomeInformation_+3A_.verify">.verify</code></td>
<td>
<p>For internal use only.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>byChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChipType</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getChromosomes</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getData</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getUnitsOnChromosome</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>isCompatibleWithCdf</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>nbrOfUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readDataFrame</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from GenomeInformation</b>:<br />
as.character, byChipType, fromCdf, fromDataSet, getChipType, getChromosomeStats, getChromosomes, getData, getPositions, getUnitIndices, getUnitsOnChromosome, getUnitsOnChromosomes, isCompatibleWithCdf, nbrOfUnits, plotDensity, readDataFrame, verify
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='UgpGenomeInformation+24byChipType'>Defines a UgpGenomeInformation object by chip type</h2><span id='topic+UgpGenomeInformation+24byChipType'></span><span id='topic+byChipType.UgpGenomeInformation'></span><span id='topic+UgpGenomeInformation.byChipType'></span><span id='topic+byChipType+2CUgpGenomeInformation-method'></span>

<h3>Description</h3>

<p>Defines a UgpGenomeInformation object by chip type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Static method (use this):
## UgpGenomeInformation$byChipType(chipType, tags=NULL, ..., nbrOfUnits=NULL,
##   verbose=FALSE)

## Don't use the below:
## S3 method for class 'UgpGenomeInformation'
byChipType(static, chipType, tags=NULL, ..., nbrOfUnits=NULL, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="UgpGenomeInformation+2B24byChipType_+3A_chiptype">chipType</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string.</p>
</td></tr>
<tr><td><code id="UgpGenomeInformation+2B24byChipType_+3A_tags">tags</code></td>
<td>
<p>...</p>
</td></tr>
<tr><td><code id="UgpGenomeInformation+2B24byChipType_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+UgpGenomeInformation">UgpGenomeInformation</a></code>.</p>
</td></tr>
<tr><td><code id="UgpGenomeInformation+2B24byChipType_+3A_verbose">verbose</code></td>
<td>
<p>See <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+UgpGenomeInformation">UgpGenomeInformation</a></code> object.
If no file was not found, an error is thrown.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+UgpGenomeInformation">UgpGenomeInformation</a></code>.
</p>

<hr>
<h2 id='UnitModel'>The UnitModel class</h2><span id='topic+UnitModel'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class UnitModel</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="#topic+Model">Model</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><em><code>UnitModel</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<em><a href="#topic+AffineCnPlm">AffineCnPlm</a></em>, <em><a href="#topic+AffinePlm">AffinePlm</a></em>, <em><a href="#topic+AffineSnpPlm">AffineSnpPlm</a></em>, <em><a href="#topic+AlleleSummation">AlleleSummation</a></em>, <em><a href="#topic+AvgCnPlm">AvgCnPlm</a></em>, <em><a href="#topic+AvgPlm">AvgPlm</a></em>, <em><a href="#topic+AvgSnpPlm">AvgSnpPlm</a></em>, <em><a href="#topic+ExonRmaPlm">ExonRmaPlm</a></em>, <em><a href="#topic+FirmaModel">FirmaModel</a></em>, <em><a href="#topic+HetLogAddCnPlm">HetLogAddCnPlm</a></em>, <em><a href="#topic+HetLogAddPlm">HetLogAddPlm</a></em>, <em><a href="#topic+HetLogAddSnpPlm">HetLogAddSnpPlm</a></em>, <em><a href="#topic+MbeiCnPlm">MbeiCnPlm</a></em>, <em><a href="#topic+MbeiPlm">MbeiPlm</a></em>, <em><a href="#topic+MbeiSnpPlm">MbeiSnpPlm</a></em>, <em><a href="#topic+MultiArrayUnitModel">MultiArrayUnitModel</a></em>, <em><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></em>, <em><a href="#topic+RmaCnPlm">RmaCnPlm</a></em>, <em><a href="#topic+RmaPlm">RmaPlm</a></em>, <em><a href="#topic+RmaSnpPlm">RmaSnpPlm</a></em>, <em><a href="#topic+SingleArrayUnitModel">SingleArrayUnitModel</a></em><br />
</p>
<p>public abstract static class <b>UnitModel</b><br />
extends <em><a href="#topic+Model">Model</a></em><br />
</p>
<p>This class is abstract and represents a generic unit model, i.e.
a model that is applied to each unit separately.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UnitModel(dataSet=NULL, probeModel=c("pm", "mm", "pm-mm", "min1(pm-mm)", "pm+mm"),
  shift=0, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="UnitModel_+3A_dataset">dataSet</code></td>
<td>
<p>An <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code> to which this model should
be fitted.</p>
</td></tr>
<tr><td><code id="UnitModel_+3A_probemodel">probeModel</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying how PM and MM values
should be modeled.  By default only PM signals are used.</p>
</td></tr>
<tr><td><code id="UnitModel_+3A_shift">shift</code></td>
<td>
<p>An optional amount the signals should be shifted
(translated) before fitting the model.</p>
</td></tr>
<tr><td><code id="UnitModel_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of <code><a href="#topic+Model">Model</a></code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>findUnitsTodo</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from Model</b>:<br />
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='UnitTypeScaleNormalization'>The UnitTypeScaleNormalization class</h2><span id='topic+UnitTypeScaleNormalization'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class UnitTypeScaleNormalization</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaTransform">AromaTransform</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="#topic+Transform">Transform</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform">ProbeLevelTransform</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ProbeLevelTransform3">ProbeLevelTransform3</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>UnitTypeScaleNormalization</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>UnitTypeScaleNormalization</b><br />
extends <em><a href="#topic+ProbeLevelTransform3">ProbeLevelTransform3</a></em><br />
</p>
<p>This class represents a normalization function that transforms the
probe signals such that each unit type gets the same average.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UnitTypeScaleNormalization(..., targetAvg=4400)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="UnitTypeScaleNormalization_+3A_...">...</code></td>
<td>
<p>Arguments passed to the constructor of
<code><a href="#topic+ProbeLevelTransform3">ProbeLevelTransform3</a></code>.</p>
</td></tr>
<tr><td><code id="UnitTypeScaleNormalization_+3A_targetavg">targetAvg</code></td>
<td>
<p>A <code><a href="base.html#topic+numeric">numeric</a></code> value.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>process</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ProbeLevelTransform3</b>:<br />
getAsteriskTags, getCellsTo, getCellsToFit, getCellsToUpdate, getParameters, getUnitsTo, getUnitsToFit, getUnitsToUpdate, writeSignals
</p>
<p><b>Methods inherited from ProbeLevelTransform</b>:<br />
getRootPath
</p>
<p><b>Methods inherited from Transform</b>:<br />
getOutputDataSet, getOutputFiles
</p>
<p><b>Methods inherited from AromaTransform</b>:<br />
as.character, findFilesTodo, getAsteriskTags, getExpectedOutputFiles, getExpectedOutputFullnames, getFullName, getInputDataSet, getName, getOutputDataSet, getOutputDataSet0, getOutputFiles, getPath, getRootPath, getTags, isDone, process, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>

<hr>
<h2 id='updateUnits.AffymetrixCelFile'>Updates CEL data unit by unit</h2><span id='topic+updateUnits.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.updateUnits'></span><span id='topic+updateUnits+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Updates CEL data unit by unit for all or a subset of units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelFile'
updateUnits(this, data, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="updateUnits.AffymetrixCelFile_+3A_data">data</code></td>
<td>
<p>A <code><a href="base.html#topic+list">list</a></code> structure consisting of grouped data similar to
what <code><a href="#topic+readUnits.AffymetrixCelFile">*readUnits</a>()</code> returns.</p>
</td></tr>
<tr><td><code id="updateUnits.AffymetrixCelFile_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the <code><a href="base.html#topic+list">list</a></code> structure that <code><a href="affxparser.html#topic+updateCelUnits">updateCelUnits</a></code> returns.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code><a href="#topic+updateUnits.AffymetrixCelFile">*updateUnits</a>()</code>.
For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>

<hr>
<h2 id='verify.GenomeInformation'>Verifies the correctness of the underlying file</h2><span id='topic+verify.GenomeInformation'></span><span id='topic+GenomeInformation.verify'></span><span id='topic+verify+2CGenomeInformation-method'></span>

<h3>Description</h3>

<p>Verifies the correctness of the underlying file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GenomeInformation'
verify(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="verify.GenomeInformation_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns (visibly) <code><a href="base.html#topic+logical">TRUE</a></code> if the file is valid, otherwise an error is
thrown.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+GenomeInformation">GenomeInformation</a></code>.
</p>

<hr>
<h2 id='verify.SnpInformation'>Verifies the correctness of the underlying file</h2><span id='topic+verify.SnpInformation'></span><span id='topic+SnpInformation.verify'></span><span id='topic+verify+2CSnpInformation-method'></span>

<h3>Description</h3>

<p>Verifies the correctness of the underlying file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SnpInformation'
verify(this, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="verify.SnpInformation_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns (visibly) <code><a href="base.html#topic+logical">TRUE</a></code> if the file is valid, otherwise an error is
thrown.
</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+SnpInformation">SnpInformation</a></code>.
</p>

<hr>
<h2 id='WeightsFile'>The WeightsFile class</h2><span id='topic+WeightsFile'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class WeightsFile</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFile">GenericDataFile</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+CacheKeyInterface">CacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+FileCacheKeyInterface">FileCacheKeyInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataFile">AromaMicroarrayDataFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFile">AffymetrixFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+ParameterCelFile">ParameterCelFile</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>WeightsFile</code></em><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public abstract static class <b>WeightsFile</b><br />
extends <em><a href="#topic+ParameterCelFile">ParameterCelFile</a></em><br />
</p>
<p>This class represents weights calculated from residuals of
probe-level models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WeightsFile(..., probeModel=c("pm"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="WeightsFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+ParameterCelFile">ParameterCelFile</a></code>.</p>
</td></tr>
<tr><td><code id="WeightsFile_+3A_probemodel">probeModel</code></td>
<td>
<p>The specific type of model, e.g. <code>"pm"</code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>findUnitsTodo</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getImage</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>writeImage</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ParameterCelFile</b>:<br />
extractDataFrame, extractMatrix, readUnits
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from AffymetrixCelFile</b>:<br />
allocateFromCdf, as.character, clone, createFrom, extractMatrix, fromFile, getAm, getCdf, getExtensionPattern, getFileFormat, getImage, getUnitNamesFile, getUnitTypesFile, highlight, image270, nbrOfCells, plotDensity, plotImage, plotMvsA, plotMvsX, range, setCdf, smoothScatterMvsA, writeImage
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataFile</b>:<br />
getAttributeXY, getChipType, getPlatform, getPloidy, getXAM, hasAttributeXY, isAverageFile, setAttributeXY, setAttributesByTags
</p>
<p><b>Methods inherited from FileCacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from CacheKeyInterface</b>:<br />
getCacheKey
</p>
<p><b>Methods inherited from GenericDataFile</b>:<br />
as.character, clone, compareChecksum, copyTo, equals, fromFile, getAttribute, getAttributes, getChecksum, getChecksumFile, getCreatedOn, getDefaultFullName, getExtension, getExtensionPattern, getFileSize, getFileType, getFilename, getFilenameExtension, getLastAccessedOn, getLastModifiedOn, getOutputExtension, getPath, getPathname, gunzip, gzip, hasBeenModified, is.na, isFile, isGzipped, linkTo, readChecksum, renameTo, renameToUpperCaseExt, setAttribute, setAttributes, setAttributesBy, setAttributesByTags, setExtensionPattern, testAttributes, validate, validateChecksum, writeChecksum, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson, Ken Simpson</p>


<h3>See Also</h3>

<p>An object of this class is typically obtained through the
<code>getWeightsSet()</code> method for the <code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code> class.
An object of this class is typically part of a <code><a href="#topic+WeightsSet">WeightsSet</a></code>.
</p>

<hr>
<h2 id='WeightsSet'>The WeightsSet class</h2><span id='topic+WeightsSet'></span>

<h3>Description</h3>

<p>Package:  aroma.affymetrix <br />
<b>Class WeightsSet</b><br />
</p>
<p><code><a href="R.oo.html#topic+Object">Object</a></code><br />
<code>~~|</code><br />
<code>~~+--</code><code><a href="R.filesets.html#topic+FullNameInterface">FullNameInterface</a></code><br />
<code>~~~~~~~|</code><br />
<code>~~~~~~~+--</code><code><a href="R.filesets.html#topic+GenericDataFileSet">GenericDataFileSet</a></code><br />
<code>~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaMicroarrayDataSet">AromaMicroarrayDataSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+AromaPlatformInterface">AromaPlatformInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixFileSet">AffymetrixFileSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code><a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a></code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br />
<code>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><code>WeightsSet</code><br />
</p>
<p><b>Directly known subclasses:</b><br />
<br />
</p>
<p>public static class <b>WeightsSet</b><br />
extends <a href="aroma.core.html#topic+ParametersInterface">ParametersInterface</a><br />
</p>
<p>This class represents probe-level weights.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WeightsSet(..., probeModel=c("pm"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="WeightsSet_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+AffymetrixCelSet">AffymetrixCelSet</a></code>.</p>
</td></tr>
<tr><td><code id="WeightsSet_+3A_probemodel">probeModel</code></td>
<td>
<p>The specific type of model, e.g. <code>"pm"</code>.</p>
</td></tr>
</table>


<h3>Fields and Methods</h3>

<p><b>Methods:</b><br />
</p>

<table>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>findUnitsTodo</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getAverageFile</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>getCellIndices</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
 </td><td style="text-align: left;"> <code>readUnits</code> </td><td style="text-align: left;">  -</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p><b>Methods inherited from ParametersInterface</b>:<br />
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from AffymetrixCelSet</b>:<br />
append, as, as.AffymetrixCelSet, as.character, averageQuantile, byName, byPath, clone, convertToUnique, doCRMAv1, doCRMAv2, doFIRMA, doGCRMA, doRMA, extractAffyBatch, extractFeatureSet, extractMatrix, extractSnpFeatureSet, findByName, getAverage, getAverageAsinh, getAverageFile, getAverageLog, getCdf, getChipType, getData, getIntensities, getPlatform, getTimestamps, getUnitGroupCellMap, getUnitIntensities, getUnitNamesFile, getUnitTypesFile, isDuplicated, justRMA, justSNPRMA, nbrOfArrays, normalizeQuantile, plotDensity, range, readUnits, setCdf, update2, writeSgr
</p>
<p><b>Methods inherited from AffymetrixFileSet</b>:<br />
as, as.AffymetrixFileSet, byPath, getDefaultFullName
</p>
<p><b>Methods inherited from AromaPlatformInterface</b>:<br />
getAromaPlatform, getAromaUflFile, getAromaUgpFile, getChipType, getPlatform, getUnitAnnotationDataFile, getUnitNamesFile, getUnitTypesFile, isCompatibleWith
</p>
<p><b>Methods inherited from AromaMicroarrayDataSet</b>:<br />
as.AromaMicroarrayDataSetList, as.AromaMicroarrayDataSetTuple, getAromaFullNameTranslatorSet, getAverageFile, getChipType, getDefaultFullName, getPlatform, setAttributesBy, setAttributesBySampleAnnotationFile, setAttributesBySampleAnnotationSet, validate
</p>
<p><b>Methods inherited from GenericDataFileSet</b>:<br />
[, [[, anyDuplicated, anyNA, append, appendFiles, appendFullNamesTranslator, appendFullNamesTranslatorByNULL, appendFullNamesTranslatorByTabularTextFile, appendFullNamesTranslatorByTabularTextFileSet, appendFullNamesTranslatorBydata.frame, appendFullNamesTranslatorByfunction, appendFullNamesTranslatorBylist, as.character, as.list, byName, byPath, c, clearCache, clearFullNamesTranslator, clone, copyTo, dsApplyInPairs, duplicated, equals, extract, findByName, findDuplicated, getChecksum, getChecksumFileSet, getChecksumObjects, getDefaultFullName, getFile, getFileClass, getFileSize, getFiles, getFullNames, getNames, getOneFile, getPath, getPathnames, getSubdirs, gunzip, gzip, hasFile, indexOf, is.na, names, nbrOfFiles, rep, resetFullNames, setFullNamesTranslator, sortBy, unique, update2, updateFullName, updateFullNames, validate, getFullNameTranslatorSet, getParentName
</p>
<p><b>Methods inherited from FullNameInterface</b>:<br />
appendFullNameTranslator, appendFullNameTranslatorByNULL, appendFullNameTranslatorByTabularTextFile, appendFullNameTranslatorByTabularTextFileSet, appendFullNameTranslatorBycharacter, appendFullNameTranslatorBydata.frame, appendFullNameTranslatorByfunction, appendFullNameTranslatorBylist, clearFullNameTranslator, clearListOfFullNameTranslators, getDefaultFullName, getFullName, getFullNameTranslator, getListOfFullNameTranslators, getName, getTags, hasTag, hasTags, resetFullName, setFullName, setFullNameTranslator, setListOfFullNameTranslators, setName, setTags, updateFullName
</p>
<p><b>Methods inherited from Object</b>:<br />
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson, Ken Simpson</p>


<h3>See Also</h3>

<p>An object of this class is typically obtained through the
<code>getWeightsSet()</code> method for the <code><a href="#topic+ProbeLevelModel">ProbeLevelModel</a></code> class.
</p>

<hr>
<h2 id='writeCdf.AffyGenePDInfo'>Generates an Affymetrix CDF file from a Platform Design (PD) package</h2><span id='topic+writeCdf.AffyGenePDInfo'></span><span id='topic+AffyGenePDInfo.writeCdf'></span><span id='topic+writeCdf+2CAffyGenePDInfo-method'></span><span id='topic+writeCdf.PDInfoList'></span><span id='topic+writeCdf.DBPDInfo'></span>

<h3>Description</h3>

<p>Generates an Affymetrix CDF file from a Platform Design (PD) package.
Platform Design (PD) packages are also known as &quot;pdInfo&quot; packages.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffyGenePDInfo'
writeCdf(this, tags=c("*"), unitsBy=c("transcript", "exon"), namesBy=c("fsetid", "id"),
  path=NULL, overwrite=FALSE, verbose=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writeCdf.AffyGenePDInfo_+3A_tags">tags</code></td>
<td>
<p>An optional <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> of tags to be added to the CDF
filename.</p>
</td></tr>
<tr><td><code id="writeCdf.AffyGenePDInfo_+3A_unitsby">unitsBy</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying how to group units.</p>
</td></tr>
<tr><td><code id="writeCdf.AffyGenePDInfo_+3A_path">path</code></td>
<td>
<p>The output path where the CDF file is written.
If <code><a href="base.html#topic+NULL">NULL</a></code> (default), then it is written to the corresponding
<code>annotationData/chipTypes/&lt;chipType&gt;/</code> directory.</p>
</td></tr>
<tr><td><code id="writeCdf.AffyGenePDInfo_+3A_overwrite">overwrite</code></td>
<td>
<p>If <code><a href="base.html#topic+logical">TRUE</a></code>, an existing CDF is overwritten, otherwise
an exception is thrown.</p>
</td></tr>
<tr><td><code id="writeCdf.AffyGenePDInfo_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or <code><a href="R.utils.html#topic+Verbose">Verbose</a></code>.</p>
</td></tr>
<tr><td><code id="writeCdf.AffyGenePDInfo_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The formal chip type of the CDF is inferred from the AffyGenePDInfo package.
The filename of the CDF is constructed from the chip type and any optional
tags.
To minimize the risk for a corrupt CDF file, the creation of the file
is atomic by first writing to a temporary file which is then renamed.
</p>


<h3>Value</h3>

<p>Returns (invisibly) the pathname to CDF written.
</p>


<h3>Limitations</h3>

<p>The information available in the PD package is limited and does
not contain all information needed to populate a CDF file.
In order to workaround these limitations, certain CDF entries
are set to predefined/hardwired values.
The 'pbase' and 'tbase' entries of the generated CDF file is
hardwired to &quot;T&quot; and &quot;A&quot;, respectively.
Likewise, the 'groupdirection' entry is hardwired to &quot;sense&quot;.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson and Guido Hooiveld adopted from <code>pdInfo2Cdf()</code>
written by Samuel Wuest and Mark Robinson.
</p>

<hr>
<h2 id='writeImage.AffymetrixCelFile'>Writes a spatial image of the signals in the CEL file</h2><span id='topic+writeImage.AffymetrixCelFile'></span><span id='topic+AffymetrixCelFile.writeImage'></span><span id='topic+writeImage+2CAffymetrixCelFile-method'></span>

<h3>Description</h3>

<p>Writes a spatial image of the signals in the CEL file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AffymetrixCelFile'
writeImage(this, filename=NULL, fullname=NULL, tags=c("*", "sqrt", "gray"),
  imgFormat="png", path=NULL, field=c("intensities", "stdvs", "pixels"), ..., skip=TRUE,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writeImage.AffymetrixCelFile_+3A_filename">filename</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying the filename of
the output file.</p>
</td></tr>
<tr><td><code id="writeImage.AffymetrixCelFile_+3A_fullname">fullname</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying the full name of
the output file.</p>
</td></tr>
<tr><td><code id="writeImage.AffymetrixCelFile_+3A_tags">tags</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> <code><a href="base.html#topic+vector">vector</a></code> of optional tags added to the
already existing tags of the CEL file.</p>
</td></tr>
<tr><td><code id="writeImage.AffymetrixCelFile_+3A_imgformat">imgFormat</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> string specifying the filename extension
which also defines the image file format.</p>
</td></tr>
<tr><td><code id="writeImage.AffymetrixCelFile_+3A_path">path</code></td>
<td>
<p>The path where the image file is stored.</p>
</td></tr>
<tr><td><code id="writeImage.AffymetrixCelFile_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+getImage.AffymetrixCelFile">*getImage</a>()</code>.</p>
</td></tr>
<tr><td><code id="writeImage.AffymetrixCelFile_+3A_verbose">verbose</code></td>
<td>
<p>A <code><a href="base.html#topic+logical">logical</a></code> or a <code><a href="R.utils.html#topic+Verbose">Verbose</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the pathname to the image file created.
</p>


<h3>Author(s)</h3>

<p>Ken Simpson, Henrik Bengtsson</p>


<h3>See Also</h3>

<p>For more information see <code><a href="#topic+AffymetrixCelFile">AffymetrixCelFile</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  yellow.colors &lt;- function(n) { hsv(h=0.15, v=0:(n-1)/(n-1)) }
  df &lt;- ds[[1]]
  writeImage(df, tags="gray", palette=gray.colors(256), xrange=c(0,200))
  writeImage(df, tags="yellow", palette=yellow.colors(256), xrange=c(0,200))
  writeImage(df, tags="heat", palette=heat.colors(256), xrange=c(0,200))
  writeImage(df, tags="terrain", palette=terrain.colors(256), xrange=c(0,200))
  writeImage(df, tags="topo", palette=topo.colors(256), xrange=c(0,200))
  writeImage(df, tags="cm", palette=cm.colors(256), xrange=c(0,200))
  writeImage(df, tags="rainbow", palette=rainbow(256), xrange=c(0,200))

## End(Not run)</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
