<!DOCTYPE html><html><head><title>Help for package aSPU</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {aSPU}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#aSPU'><p>Sum of Powered Score (SPU) tests and adaptive SPU (aSPU) test for single trait - SNP set association.</p></a></li>
<li><a href='#aSPUd'><p>Adaptive Sum of powered score (SPU) tests (SPU and aSPU) based on asymptotic distribution.</p></a></li>
<li><a href='#aSPUpath'><p>Pathway based Sum of Powered Score tests (SPUpath) and adaptive SPUpath (aSPUpath) test for single trait - pathway association.</p></a></li>
<li><a href='#aSPUpath2'><p>Pathway based Sum of Powered Score tests (SPUpath) and adaptive SPUpath (aSPUpath) test for single trait - pathway association. (vector version, fast when n is large)</p></a></li>
<li><a href='#aSPUr'><p>Robust Sum of powered score (SPU) tests and aSPU test for a quantitative trait</p></a></li>
<li><a href='#aSPUs'><p>Sum of Powered Score (SPUs) tests and adaptive SPU (aSPUs) test for single trait - SNP set association with GWAS summary statistics.</p></a></li>
<li><a href='#aSPUsD'><p>Sum of Powered Score (SPUs) tests and adaptive SPU (aSPUs) test for single trait - SNP set association with GWAS summary statistics (distribution based).</p></a></li>
<li><a href='#aSPUsPath'><p>Pathway based Sum of Powered Score tests (SPUsPath) and adaptive SPUpath (aSPUsPath) test for single trait - pathway association with GWAS summary statistics.</p></a></li>
<li><a href='#aSPUsPath2'><p>Pathway based Sum of Powered Score tests (SPUsPath) and adaptive SPUpath (aSPUsPath) test for single trait - pathway association with GWAS summary statistics. (vector version, fast when n is large)</p></a></li>
<li><a href='#aSPUw'><p>Inverse variance weighted Sum of Powered Score tests (SPUw) and adaptive SPUw (aSPUw) test for single trait - SNP set association.</p></a></li>
<li><a href='#estcov'><p>estcov</p></a></li>
<li><a href='#exdat'><p>An example data set</p></a></li>
<li><a href='#GATES2'><p>Gene-based Association Test that uses an extended Simes procedure (GATES) for single trait - SNP set association</p></a></li>
<li><a href='#GatesSimes'><p>GATES-Simes test for single trait - pathway association.</p></a></li>
<li><a href='#GEEaSPU'><p>The SPU and aSPU tests for multiple traits - single SNP association in generalized estimating equations.</p></a></li>
<li><a href='#getlogitp'><p>Get the p-values of the multiple SNPs based on the Wald test in a logistic regresion model.</p></a></li>
<li><a href='#Hyst'><p>HYST (Hybrid Set-based Test) for single trait - pathway association</p></a></li>
<li><a href='#kegg9'><p>A subset of coronary artery disease (CAD) data</p></a></li>
<li><a href='#minP'><p>minP test.</p></a></li>
<li><a href='#MTaSPUs'><p>The SPU and aSPU tests for multiple traits - single SNP association with GWAS summary statistics.</p></a></li>
<li><a href='#MTaSPUsSet'><p>gene-Multitrait Sum of Powered Score (MTSPUsSet) tests and adaptive MTSPUsSet (MTaSPUsSet) test for multi trait - SNP set association with GWAS summary statistics.</p></a></li>
<li><a href='#MTaSPUsSetPath'><p>Multitraits - Pathway based Sum of Powered Score tests (MTSPUsSetPath) and adaptive MTSPUsSetpath (MTaSPUsSetPath) test for multi trait - pathway association with GWAS summary statistics.</p></a></li>
<li><a href='#MTaSPUsSetScore'><p>gene-Multitrait Sum of Powered Score (MTSPUsSetScore) tests and adaptive MTSPUsSet (MTaSPUsSet) test for multi trait - SNP set association with GWAS summary statistics. (Score version added)</p></a></li>
<li><a href='#plotPmat'><p>Image plot of P-value matrix.</p></a></li>
<li><a href='#pruneSNP'><p>Prune SNPs (related to aSPUs, aSPUsPath, MTaSPUsSet)</p></a></li>
<li><a href='#SAMD11'><p>An example data for MTaSPUsSet test</p></a></li>
<li><a href='#simPathAR1Snp'><p>Simulating a pathway with multiple SNPs.</p></a></li>
<li><a href='#someGs'><p>An example data for MTaSPUsSet test</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.50</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-6-28</td>
</tr>
<tr>
<td>Title:</td>
<td>Adaptive Sum of Powered Score Test</td>
</tr>
<tr>
<td>Description:</td>
<td>R codes for the (adaptive) Sum of Powered Score ('SPU' and 'aSPU')
    tests, inverse variance weighted Sum of Powered score ('SPUw' and 'aSPUw') tests
    and gene-based and some pathway based association tests (Pathway based Sum of
    Powered Score tests ('SPUpath'), adaptive 'SPUpath' ('aSPUpath') test, 'GEEaSPU'
    test for multiple traits - single 'SNP' (single nucleotide polymorphism)
    association in generalized estimation equations, 'MTaSPUs' test for multiple
    traits - single 'SNP' association with Genome Wide Association Studies ('GWAS')
    summary statistics, Gene-based Association Test that uses an extended 'Simes'
    procedure ('GATES'), Hybrid Set-based Test ('HYST') and extended version
    of 'GATES' test for pathway-based association testing ('GATES-Simes'). ).
    The tests can be used with genetic and other data sets with covariates. The
    response variable is binary or quantitative. Summary; (1) Single trait-'SNP' set
    association with individual-level data ('aSPU', 'aSPUw', 'aSPUr'), (2) Single trait-'SNP'
    set association with summary statistics ('aSPUs'), (3) Single trait-pathway
    association with individual-level data ('aSPUpath'), (4) Single trait-pathway
    association with summary statistics ('aSPUsPath'), (5) Multiple traits-single
    'SNP' association with individual-level data ('GEEaSPU'), (6) Multiple traits-
    single 'SNP' association with summary statistics ('MTaSPUs'), (7) Multiple traits-'SNP' set association with summary statistics('MTaSPUsSet'), (8) Multiple traits-pathway association with summary statistics('MTaSPUsSetPath').</td>
</tr>
<tr>
<td>Author:</td>
<td>Il-Youp Kwak and others (See Author(s) in each function manual)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Il-Youp Kwak &lt;ikwak2@cau.ac.kr&gt;</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, markdown, knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.6.0), gee, MASS, mvtnorm, fields, matrixStats</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ikwak2/aSPU">https://github.com/ikwak2/aSPU</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-06-28 16:32:54 UTC; root</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-06-28 17:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='aSPU'>Sum of Powered Score (SPU) tests and adaptive SPU (aSPU) test for single trait - SNP set association.</h2><span id='topic+aSPU'></span>

<h3>Description</h3>

<p>It gives p-values of the SPU tests and aSPU test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aSPU(
  Y,
  X,
  cov = NULL,
  resample = c("perm", "boot", "sim"),
  model = c("gaussian", "binomial"),
  pow = c(1:8, Inf),
  n.perm = 1000,
  threshold = 10^5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aSPU_+3A_y">Y</code></td>
<td>
<p>Response or phenotype data. It can be a disease indicator; =0 for controls, =1 for cases.
Or it can be a quantitative trait. A vector with length n (number of observations).</p>
</td></tr>
<tr><td><code id="aSPU_+3A_x">X</code></td>
<td>
<p>Genotype or other data; each row for a subject, and each column
for an SNP (or a predictor). The value of each SNP is the # of the copies
for an allele. A matrix with dimension n by k (n : number of observation, k : number of SNPs (or predictors) ).</p>
</td></tr>
<tr><td><code id="aSPU_+3A_cov">cov</code></td>
<td>
<p>Covariates. A matrix with dimension n by p (n :number of observation, p : number of covariates).</p>
</td></tr>
<tr><td><code id="aSPU_+3A_resample">resample</code></td>
<td>
<p>Use &quot;perm&quot; for residual permutations, &quot;sim&quot; for simulations from the null distribution, and &quot;boot&quot; for parametric bootstrap.</p>
</td></tr>
<tr><td><code id="aSPU_+3A_model">model</code></td>
<td>
<p>Use &quot;gaussian&quot; for a quantitative trait, and use &quot;binomial&quot; for a binary trait.</p>
</td></tr>
<tr><td><code id="aSPU_+3A_pow">pow</code></td>
<td>
<p>power used in SPU test. A vector of the powers.</p>
</td></tr>
<tr><td><code id="aSPU_+3A_n.perm">n.perm</code></td>
<td>
<p>number of permutations or bootstraps.</p>
</td></tr>
<tr><td><code id="aSPU_+3A_threshold">threshold</code></td>
<td>
<p>When n.perm is less than the threshold, we use faster version of aSPU but use more memory. (default is 10^5)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object, Ts : test statistics for the SPU tests (in the order of the specified pow) and finally for the aSPU test.
pvs : p-values for the SPU and aSPU tests.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak, Junghi Kim, Yiwei Zhang and Wei Pan
</p>


<h3>References</h3>

<p>Wei Pan, Junghi Kim, Yiwei Zhang, Xiaotong Shen and Peng Wei (2014)
A powerful and adaptive association test for rare variants,
Genetics, 197(4), 1081-95
</p>
<p>Junghi Kim, Jeffrey R Wozniak, Bryon A Mueller, Xiaotong Shen and Wei Pan (2014) Comparison of statistical tests for group differences in brain functional networks, NeuroImage, 1;101:681-694
</p>


<h3>See Also</h3>

<p><code><a href="#topic+aSPUw">aSPUw</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(exdat)

## example analysis using aSPU test on exdat data.
# increase n.perm for better p.val 
## Not run: out &lt;- aSPU(exdat$Y, exdat$X, cov = NULL, resample = "boot",
            model = "binomial", pow = c(1:8, Inf), n.perm = 1000) 
## End(Not run)


out$Ts
# This is a vector of Test Statistics for SPU and aSPU tests.
# SPU1 to SPUInf corresponds with the option pow=c(1:8, Inf)
# They are SPU test statistics.
# The last element aSPU is minimum of them, aSPU statistic.

out$pvs
# This is a vector of p-values for SPU and aSPU tests.
# SPU1 to SPUInf corresponds with the option pow=c(1:8, Inf)
# They are p-values for corresponding SPU tests.
# The last element is p-value of aSPU test.

</code></pre>

<hr>
<h2 id='aSPUd'>Adaptive Sum of powered score (SPU) tests (SPU and aSPU) based on asymptotic distribution.</h2><span id='topic+aSPUd'></span>

<h3>Description</h3>

<p>It gives the p-values of the SPU(1), SPU(2) and minP tests and aSPUd test based on the asymptotic distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aSPUd(Y, X, cov = NULL, model = c("gaussian", "binomial"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aSPUd_+3A_y">Y</code></td>
<td>
<p>phenotype data. It can be disease lables; =0 for controls, =1 for cases.
or It can be any quantitative traits. Vector with length n (number of observations)</p>
</td></tr>
<tr><td><code id="aSPUd_+3A_x">X</code></td>
<td>
<p>genotype data; each row for a subject, and each column
for an SNP. The value of each element is the # of the copies
for an allele. Matrix with dimension n by k (n : number of observation, k : number of genotype data)</p>
</td></tr>
<tr><td><code id="aSPUd_+3A_cov">cov</code></td>
<td>
<p>covariates. Matrix with dimension n by p (n :number of observation, p : number of covariates)</p>
</td></tr>
<tr><td><code id="aSPUd_+3A_model">model</code></td>
<td>
<p>Use &quot;gaussian&quot; for quantitative trait (Default)
, and Use &quot;binomial&quot; for binary trait.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>p-values for SPU(1), SPU(2), minP tests and aSPU test.
</p>


<h3>References</h3>

<p>Gongjun Xu, Lifeng Lin, Peng Wei and Wei Pan (2016) An adaptive two-sample test for high-dimensional means, Biometrika (2016) 103 (3): 609-624.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+aSPU">aSPU</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(exdat)
out &lt;- aSPUd(exdat$Y, exdat$X, cov = NULL, model = "binomial")
out

</code></pre>

<hr>
<h2 id='aSPUpath'>Pathway based Sum of Powered Score tests (SPUpath) and adaptive SPUpath (aSPUpath) test for single trait - pathway association.</h2><span id='topic+aSPUpath'></span>

<h3>Description</h3>

<p>It gives p-values of the SPUpath tests and aSPUpath test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aSPUpath(
  Y,
  X,
  cov = NULL,
  model = c("binomial", "gaussian"),
  snp.info,
  gene.info,
  pow = c(1:8, Inf),
  pow2 = c(1, 2, 4, 8),
  n.perm = 200,
  usePCs = F,
  varprop = 0.95
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aSPUpath_+3A_y">Y</code></td>
<td>
<p>Response or phenotype data. It can be a disease indicator; =0 for controls, =1 for cases.
Or it can be a quantitative trait. A vector with length n (number of observations).</p>
</td></tr>
<tr><td><code id="aSPUpath_+3A_x">X</code></td>
<td>
<p>Genotype or other data; each row for a subject, and each column
for an SNP (or a predictor). The value of each SNP is the # of the copies
for an allele. A matrix with dimension n by k (n : number of observation, k : number of SNPs (or predictors) ).</p>
</td></tr>
<tr><td><code id="aSPUpath_+3A_cov">cov</code></td>
<td>
<p>Covariates. A matrix with dimension n by p (n :number of observation, p : number of covariates).</p>
</td></tr>
<tr><td><code id="aSPUpath_+3A_model">model</code></td>
<td>
<p>Use &quot;gaussian&quot; for a quantitative trait, and use &quot;binomial&quot; for a binary trait.</p>
</td></tr>
<tr><td><code id="aSPUpath_+3A_snp.info">snp.info</code></td>
<td>
<p>SNP information matrix, the 1st column is SNP id, 2nd column is chromosome #, 3rd column indicates SNP location.</p>
</td></tr>
<tr><td><code id="aSPUpath_+3A_gene.info">gene.info</code></td>
<td>
<p>GENE information matrix, The 1st column is GENE id, 2nd column is chromosome #, 3rd and 4th column indicate start and end positions of the gene.</p>
</td></tr>
<tr><td><code id="aSPUpath_+3A_pow">pow</code></td>
<td>
<p>SNP specific power(gamma values) used in SPUpath test.</p>
</td></tr>
<tr><td><code id="aSPUpath_+3A_pow2">pow2</code></td>
<td>
<p>GENE specific power(gamma values) used in SPUpath test.</p>
</td></tr>
<tr><td><code id="aSPUpath_+3A_n.perm">n.perm</code></td>
<td>
<p>number of permutations.</p>
</td></tr>
<tr><td><code id="aSPUpath_+3A_usepcs">usePCs</code></td>
<td>
<p>indicating whether to extract PCs and then use PCs of X.</p>
</td></tr>
<tr><td><code id="aSPUpath_+3A_varprop">varprop</code></td>
<td>
<p>the proportion of the variations explained (cutoff) that
determines how many top PCs to use.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>P-values for SPUpath tests and aSPUpath test.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak and Wei Pan
</p>


<h3>References</h3>

<p>Wei Pan, Il-Youp Kwak and Peng Wei (2015)
A Powerful and Pathway-Based Adaptive Test for Genetic Association With Common or Rare Variants, The American Journal of Human Genetics, 97, 86-98
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simPathAR1Snp">simPathAR1Snp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: dat1&lt;-simPathAR1Snp(nGenes=20, nGenes1=5, nSNPlim=c(1, 20),
	       nSNP0=1, LOR=.2, n=100, MAFlim=c(0.05, 0.4), p0=0.05 ) 
## End(Not run)


# p-values of SPUpath and aSPUpath tests.
## Not run: p.pathaspu&lt;- aSPUpath(dat1$Y, dat1$X, snp.info = dat1$snp.info,
         gene.info = dat1$gene.info,
         model = "binomial", pow=1:8, pow2=c(1, 2, 4, 8), n.perm=1000) 
## End(Not run)

p.pathaspu
## pow = 1:8 and pow2 = 1,2,4,8
## So, there are 8*4 = 32 SPUpath p-values.
## SPUpathi,j corresponds pow = i , pow2 = j
## The last element, aSPUpath gives aSPUpath p-value.

</code></pre>

<hr>
<h2 id='aSPUpath2'>Pathway based Sum of Powered Score tests (SPUpath) and adaptive SPUpath (aSPUpath) test for single trait - pathway association. (vector version, fast when n is large)</h2><span id='topic+aSPUpath2'></span>

<h3>Description</h3>

<p>It gives p-values of the SPUpath tests and aSPUpath test. Faster than aSPUsPath function when n is large (N &gt; 10^4).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aSPUpath2(
  Y,
  X,
  cov = NULL,
  model = c("binomial", "gaussian"),
  snp.info,
  gene.info,
  pow = c(1:8, Inf),
  pow2 = c(1, 2, 4, 8),
  n.perm = 200,
  usePCs = F,
  varprop = 0.95
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aSPUpath2_+3A_y">Y</code></td>
<td>
<p>Response or phenotype data. It can be a disease indicator; =0 for controls, =1 for cases.
Or it can be a quantitative trait. A vector with length n (number of observations).</p>
</td></tr>
<tr><td><code id="aSPUpath2_+3A_x">X</code></td>
<td>
<p>Genotype or other data; each row for a subject, and each column
for an SNP (or a predictor). The value of each SNP is the # of the copies
for an allele. A matrix with dimension n by k (n : number of observation, k : number of SNPs (or predictors) ).</p>
</td></tr>
<tr><td><code id="aSPUpath2_+3A_cov">cov</code></td>
<td>
<p>Covariates. A matrix with dimension n by p (n :number of observation, p : number of covariates).</p>
</td></tr>
<tr><td><code id="aSPUpath2_+3A_model">model</code></td>
<td>
<p>Use &quot;gaussian&quot; for a quantitative trait, and use &quot;binomial&quot; for a binary trait.</p>
</td></tr>
<tr><td><code id="aSPUpath2_+3A_snp.info">snp.info</code></td>
<td>
<p>SNP information matrix, the 1st column is SNP id, 2nd column is chromosome #, 3rd column indicates SNP location.</p>
</td></tr>
<tr><td><code id="aSPUpath2_+3A_gene.info">gene.info</code></td>
<td>
<p>GENE information matrix, The 1st column is GENE id, 2nd column is chromosome #, 3rd and 4th column indicate start and end positions of the gene.</p>
</td></tr>
<tr><td><code id="aSPUpath2_+3A_pow">pow</code></td>
<td>
<p>SNP specific power(gamma values) used in SPUpath test.</p>
</td></tr>
<tr><td><code id="aSPUpath2_+3A_pow2">pow2</code></td>
<td>
<p>GENE specific power(gamma values) used in SPUpath test.</p>
</td></tr>
<tr><td><code id="aSPUpath2_+3A_n.perm">n.perm</code></td>
<td>
<p>number of permutations.</p>
</td></tr>
<tr><td><code id="aSPUpath2_+3A_usepcs">usePCs</code></td>
<td>
<p>indicating whether to extract PCs and then use PCs of X.</p>
</td></tr>
<tr><td><code id="aSPUpath2_+3A_varprop">varprop</code></td>
<td>
<p>the proportion of the variations explained (cutoff) that
determines how many top PCs to use.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>P-values for SPUpath tests and aSPUpath test.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak and Wei Pan
</p>


<h3>References</h3>

<p>Wei Pan, Il-Youp Kwak and Peng Wei (2015)
A Powerful and Pathway-Based Adaptive Test for Genetic Association With Common or Rare Variants, The American Journal of Human Genetics, 97, 86-98
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simPathAR1Snp">simPathAR1Snp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: dat1&lt;-simPathAR1Snp(nGenes=20, nGenes1=5, nSNPlim=c(1, 20),
	       nSNP0=1, LOR=.2, n=100, MAFlim=c(0.05, 0.4), p0=0.05 ) 
## End(Not run)

# p-values of SPUpath and aSPUpath tests.
## Not run: p.pathaspu&lt;- aSPUpath2(dat1$Y, dat1$X, snp.info = dat1$snp.info,
         gene.info = dat1$gene.info,
         model = "binomial", pow=1:8, pow2=c(1, 2, 4, 8), n.perm=1000) 
## End(Not run)
#p.pathaspu
## pow = 1:8 and pow2 = 1,2,4,8
## So, there are 8*4 = 32 SPUpath p-values.
## SPUpathi,j corresponds pow = i , pow2 = j
## The last element, aSPUpath gives aSPUpath p-value.

</code></pre>

<hr>
<h2 id='aSPUr'>Robust Sum of powered score (SPU) tests and aSPU test for a quantitative trait</h2><span id='topic+aSPUr'></span>

<h3>Description</h3>

<p>The test is based on the Huber loss function and using the parametric
bootstrap for inference (i.e. bootstrapping residuals).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aSPUr(Y, X, cov = NULL, pow = c(1:8, Inf), B = 1000, C = 1.345)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aSPUr_+3A_y">Y</code></td>
<td>
<p>a vector of quantitative traits (QTs).</p>
</td></tr>
<tr><td><code id="aSPUr_+3A_x">X</code></td>
<td>
<p>Genotype or other data; each row for a subject, and each column
for an SNP (or a predictor). The value of each SNP is the # of the copies
for an allele. A matrix with dimension n by k (n : number of observation, k : number of SNPs (or predictors) ).</p>
</td></tr>
<tr><td><code id="aSPUr_+3A_cov">cov</code></td>
<td>
<p>Covariates. A matrix with dimension n by k2 (n :number of observation, k2 : number of covariates).</p>
</td></tr>
<tr><td><code id="aSPUr_+3A_pow">pow</code></td>
<td>
<p>power used in SPUr test. A vector of the powers.</p>
</td></tr>
<tr><td><code id="aSPUr_+3A_b">B</code></td>
<td>
<p>number of bootstraps.</p>
</td></tr>
<tr><td><code id="aSPUr_+3A_c">C</code></td>
<td>
<p>Constant in huber loss function. C = 1.345 is chosen to maintain a high efficiency for a Normal error.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>p-values of the SPUr tests in the order of supplied pow values; finally, the p-value of the aSPUr test (that combines the SPUs tests with pow by taking their min P-value and adjust for multiple testing).
</p>


<h3>Author(s)</h3>

<p>Yiwei Zhang and Wei Pan
</p>


<h3>References</h3>

<p>Peng Wei, Ying Cao, Yiwei Zhang, Zhiyuan Xu, Il-Youp Kwak, Eric Boerwinkle, Wei Pan (2016) On Robust Association Testing for Quantitative Traits and Rare Variants, G3, 6(12) 3941-3950.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+aSPU">aSPU</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(exdat)

## example analysis using aSPU test on exdat data.

QT &lt;- jitter(exdat$Y)

out &lt;- aSPUr(Y = QT, X = exdat$X, cov = NULL, B = 100)
out
## This is a vector of p-values for SPUr and aSPUr tests.
## SPU1 to SPUInf corresponds with the option pow=c(1:8, Inf)
## They are p-values for corresponding SPUr tests.
## The last element is p-value of aSPUr test.

</code></pre>

<hr>
<h2 id='aSPUs'>Sum of Powered Score (SPUs) tests and adaptive SPU (aSPUs) test for single trait - SNP set association with GWAS summary statistics.</h2><span id='topic+aSPUs'></span>

<h3>Description</h3>

<p>It gives p-values of the SPUs tests and aSPUs test with GWAS summary statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aSPUs(Zs, corSNP, pow = c(1:8, Inf), n.perm = 1000, Ps = FALSE, prune = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aSPUs_+3A_zs">Zs</code></td>
<td>
<p>Z-scores for each SNPs. It could be P-values if the Ps option is TRUE.</p>
</td></tr>
<tr><td><code id="aSPUs_+3A_corsnp">corSNP</code></td>
<td>
<p>Correlation matirx of the SNPs to be tested; estimated from a
reference panel (based on the same set of the reference alleles as
used in calculating Z-scores).</p>
</td></tr>
<tr><td><code id="aSPUs_+3A_pow">pow</code></td>
<td>
<p>power used in SPU test. A vector of the powers.</p>
</td></tr>
<tr><td><code id="aSPUs_+3A_n.perm">n.perm</code></td>
<td>
<p>number of permutations or bootstraps.</p>
</td></tr>
<tr><td><code id="aSPUs_+3A_ps">Ps</code></td>
<td>
<p>TRUE if input is p-value, FALSE if input is Z-scores. The default is FALSE.</p>
</td></tr>
<tr><td><code id="aSPUs_+3A_prune">prune</code></td>
<td>
<p>if it is TRUE, do pruing before the test using pruneSNP function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object, Ts : test statistics for the SPU tests (in the order of the specified pow) and finally for the aSPU test.
pvs : p-values for the SPUs and aSPUs tests.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak and Wei Pan
</p>


<h3>References</h3>

<p>Il-Youp Kwak, Wei Pan (2015)
Adaptive Gene- and Pathway-Trait Association Testing with GWAS Summary Statistics,
Bioinformatics, 32(8), 1178-1184.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+aSPUw">aSPUw</a></code> <code><a href="#topic+aSPU">aSPU</a></code> <code><a href="#topic+aSPUsPath">aSPUsPath</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(kegg9)
## example analysis using aSPUM test.
g &lt;- kegg9$gene.info[1,1]  #  SOAT1 
## Take snps mapped on gene "SOAT1" from the information of gene.info and snp.info. 
snps &lt;- which( ( kegg9$snp.info[,2] == kegg9$gene.info[kegg9$gene.info[,1] == g, 2] ) &amp;
                 (kegg9$snp.info[,3] &gt; kegg9$gene.info[kegg9$gene.info[,1] == g, 3] ) &amp;
                 (kegg9$snp.info[,3] &lt; kegg9$gene.info[kegg9$gene.info[,1] == g, 4] )  )
## Take subsets
newP &lt;- kegg9$nP[snps] ;
ldsub &lt;- kegg9$ldmatrix[snps, snps];

## Get p-value for gene SOAT1. Read vignette for details.
out &lt;- aSPUs(newP, corSNP=ldsub , pow=c(1:8, Inf), n.perm=100, Ps=TRUE)

out$Ts
# This is a vector of Test Statistics for SPUM and aSPUM tests.
# SPUs1 to SPUsInf corresponds with the option pow=c(1:8, Inf)
# They are SPUs test statistics.
# The last element aSPUs is minimum of them, aSPUs statistic.

out$pvs
# This is a vector of p-values for SPUs and aSPUs tests.
# SPUs1 to SPUsInf corresponds with the option pow=c(1:8, Inf)
# They are p-values for corresponding SPUs tests.
# The last element is p-value of aSPUs test.

</code></pre>

<hr>
<h2 id='aSPUsD'>Sum of Powered Score (SPUs) tests and adaptive SPU (aSPUs) test for single trait - SNP set association with GWAS summary statistics (distribution based).</h2><span id='topic+aSPUsD'></span>

<h3>Description</h3>

<p>It gives p-values of the SPUs tests and aSPUs test with GWAS summary statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aSPUsD(Zs, corrSNP, Ps = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aSPUsD_+3A_zs">Zs</code></td>
<td>
<p>Z-scores for each SNPs. It could be P-values if the Ps option is TRUE.</p>
</td></tr>
<tr><td><code id="aSPUsD_+3A_corrsnp">corrSNP</code></td>
<td>
<p>Correlation matirx of the SNPs to be tested; estimated from a
reference panel (based on the same set of the reference alleles as
used in calculating Z-scores).</p>
</td></tr>
<tr><td><code id="aSPUsD_+3A_ps">Ps</code></td>
<td>
<p>TRUE if input is p-value, FALSE if input is Z-scores. The default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pvs : p-values for the SPUsD and aSPUsD tests.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak and Wei Pan
</p>


<h3>References</h3>

<p>Gongjun Xu, Lifeng Lin, Peng Wei and Wei Pan (2016)
An adaptive two-sample test for high-dimensional means, Biometrika (2016) 103 (3): 609-624.
Il-Youp Kwak, Wei Pan (2015) Adaptive Gene- and Pathway-Trait Association Testing with GWAS Summary Statistics,
Bioinformatics, 32(8), 1178-1184
</p>


<h3>See Also</h3>

<p><code><a href="#topic+aSPUs">aSPUs</a></code> <code><a href="#topic+aSPU">aSPU</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(kegg9)
## example analysis using aSPUM test.
g &lt;- kegg9$gene.info[1,1]  #  SOAT1 
## Take snps mapped on gene "SOAT1" from the information of gene.info and snp.info. 
snps &lt;- which( ( kegg9$snp.info[,2] == kegg9$gene.info[kegg9$gene.info[,1] == g, 2] ) &amp;
                 (kegg9$snp.info[,3] &gt; kegg9$gene.info[kegg9$gene.info[,1] == g, 3] ) &amp;
                 (kegg9$snp.info[,3] &lt; kegg9$gene.info[kegg9$gene.info[,1] == g, 4] )  )
## Take subsets
newP &lt;- kegg9$nP[snps] ;
ldsub &lt;- kegg9$ldmatrix[snps, snps];
## Get p-value for gene SOAT1. Read vignette for details.
out &lt;- aSPUsD(newP, corrSNP=ldsub, Ps=TRUE)

out

</code></pre>

<hr>
<h2 id='aSPUsPath'>Pathway based Sum of Powered Score tests (SPUsPath) and adaptive SPUpath (aSPUsPath) test for single trait - pathway association with GWAS summary statistics.</h2><span id='topic+aSPUsPath'></span>

<h3>Description</h3>

<p>It gives p-values of the SPUsPath tests and aSPUsPath test with GWAS summary statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aSPUsPath(
  Zs,
  corSNP,
  pow = c(1, 2, 4, 8, Inf),
  pow2 = c(1, 2, 4, 8),
  snp.info,
  gene.info,
  n.perm = 1000,
  Ps = FALSE,
  prune = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aSPUsPath_+3A_zs">Zs</code></td>
<td>
<p>Z-scores for each SNPs. It could be P-values if the Ps option is TRUE.</p>
</td></tr>
<tr><td><code id="aSPUsPath_+3A_corsnp">corSNP</code></td>
<td>
<p>Correlation matirx of the SNPs to be tested; estimated from a
reference panel (based on the same set of the reference alleles as
used in calculating Z-scores).</p>
</td></tr>
<tr><td><code id="aSPUsPath_+3A_pow">pow</code></td>
<td>
<p>SNP specific power(gamma values) used in SPUpath test.</p>
</td></tr>
<tr><td><code id="aSPUsPath_+3A_pow2">pow2</code></td>
<td>
<p>GENE specific power(gamma values) used in SPUpath test.</p>
</td></tr>
<tr><td><code id="aSPUsPath_+3A_snp.info">snp.info</code></td>
<td>
<p>SNP information matrix, the 1st column is SNP id, 2nd column is chromosome #, 3rd column indicates SNP location.</p>
</td></tr>
<tr><td><code id="aSPUsPath_+3A_gene.info">gene.info</code></td>
<td>
<p>GENE information matrix, The 1st column is GENE id, 2nd column is chromosome #, 3rd and 4th column indicate start and end positions of the gene.</p>
</td></tr>
<tr><td><code id="aSPUsPath_+3A_n.perm">n.perm</code></td>
<td>
<p>number of permutations.</p>
</td></tr>
<tr><td><code id="aSPUsPath_+3A_ps">Ps</code></td>
<td>
<p>TRUE if input is p-value, FALSE if input is Z-scores. The default is FALSE.</p>
</td></tr>
<tr><td><code id="aSPUsPath_+3A_prune">prune</code></td>
<td>
<p>if it is TRUE, do pruing before the test using pruneSNP function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>P-values for SPUMpath tests and aSPUMpath test.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak and Wei Pan
</p>


<h3>References</h3>

<p>Il-Youp Kwak, Wei Pan (2015)
Adaptive Gene- and Pathway-Trait Association Testing with GWAS Summary Statistics,
Bioinformatics, 32(8):1178-1184
</p>


<h3>See Also</h3>

<p><code><a href="#topic+aSPUs">aSPUs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(kegg9)

# p-values of SPUpath and aSPUpath tests.
out.a &lt;- aSPUsPath(kegg9$nP, corSNP = kegg9$ldmatrix, pow=c(1:8, Inf),
                  pow2 = c(1,2,4,8), 
                  snp.info=kegg9$snp.info, gene.info = kegg9$gene.info,
                  n.perm=5, Ps = TRUE)

out.a

</code></pre>

<hr>
<h2 id='aSPUsPath2'>Pathway based Sum of Powered Score tests (SPUsPath) and adaptive SPUpath (aSPUsPath) test for single trait - pathway association with GWAS summary statistics. (vector version, fast when n is large)</h2><span id='topic+aSPUsPath2'></span>

<h3>Description</h3>

<p>It gives p-values of the SPUsPath tests and aSPUsPath test with GWAS summary statistics. Faster than aSPUsPath function when n is large (N &gt; 10^4).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aSPUsPath2(
  Zs,
  corSNP,
  pow = c(1, 2, 4, 8, Inf),
  pow2 = c(1, 2, 4, 8),
  snp.info,
  gene.info,
  n.perm = 1000,
  Ps = FALSE,
  prune = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aSPUsPath2_+3A_zs">Zs</code></td>
<td>
<p>Z-scores for each SNPs. It could be P-values if the Ps option is TRUE.</p>
</td></tr>
<tr><td><code id="aSPUsPath2_+3A_corsnp">corSNP</code></td>
<td>
<p>Correlation matirx of the SNPs to be tested; estimated from a
reference panel (based on the same set of the reference alleles as
used in calculating Z-scores).</p>
</td></tr>
<tr><td><code id="aSPUsPath2_+3A_pow">pow</code></td>
<td>
<p>SNP specific power(gamma values) used in SPUpath test.</p>
</td></tr>
<tr><td><code id="aSPUsPath2_+3A_pow2">pow2</code></td>
<td>
<p>GENE specific power(gamma values) used in SPUpath test.</p>
</td></tr>
<tr><td><code id="aSPUsPath2_+3A_snp.info">snp.info</code></td>
<td>
<p>SNP information matrix, the 1st column is SNP id, 2nd column is chromosome #, 3rd column indicates SNP location.</p>
</td></tr>
<tr><td><code id="aSPUsPath2_+3A_gene.info">gene.info</code></td>
<td>
<p>GENE information matrix, The 1st column is GENE id, 2nd column is chromosome #, 3rd and 4th column indicate start and end positions of the gene.</p>
</td></tr>
<tr><td><code id="aSPUsPath2_+3A_n.perm">n.perm</code></td>
<td>
<p>number of permutations.</p>
</td></tr>
<tr><td><code id="aSPUsPath2_+3A_ps">Ps</code></td>
<td>
<p>TRUE if input is p-value, FALSE if input is Z-scores. The default is FALSE.</p>
</td></tr>
<tr><td><code id="aSPUsPath2_+3A_prune">prune</code></td>
<td>
<p>if it is TRUE, do pruing before the test using pruneSNP function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>P-values for SPUMpath tests and aSPUMpath test.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak and Wei Pan
</p>


<h3>References</h3>

<p>Il-Youp Kwak, Wei Pan (2015)
Adaptive Gene- and Pathway-Trait Association Testing with GWAS Summary Statistics,
Bioinformatics, 32(8):1178-1184
</p>


<h3>See Also</h3>

<p><code><a href="#topic+aSPUs">aSPUs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(kegg9)

# p-values of SPUpath and aSPUpath tests.
## Not run: out.a &lt;- aSPUsPath2(kegg9$nP, corSNP = kegg9$ldmatrix, pow=c(1:8, Inf),
                  pow2 = c(1,2,4,8), 
                  snp.info=kegg9$snp.info, gene.info = kegg9$gene.info,
                  n.perm=1000, Ps = TRUE) 
## End(Not run)

#out.a

</code></pre>

<hr>
<h2 id='aSPUw'>Inverse variance weighted Sum of Powered Score tests (SPUw) and adaptive SPUw (aSPUw) test for single trait - SNP set association.</h2><span id='topic+aSPUw'></span>

<h3>Description</h3>

<p>It gives the p-values of the SPUw tests and aSPUw test based
on the permutations of the residuals or simulations from the null distripution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aSPUw(
  Y,
  X,
  cov = NULL,
  resample = c("perm", "boot", "sim"),
  model = c("gaussian", "binomial"),
  pow = c(1:8, Inf),
  n.perm = 1000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aSPUw_+3A_y">Y</code></td>
<td>
<p>Response or phenotype data. It can be a disease indicator; =0 for controls, =1 for cases.
Or it can be a quantitative trait. A vector with length n (number of observations).</p>
</td></tr>
<tr><td><code id="aSPUw_+3A_x">X</code></td>
<td>
<p>Genotype or other data; each row for a subject, and each column
for an SNP (or a predictor). The value of each SNP is the # of the copies
for an allele. A matrix with dimension n by k (n : number of observation, k : number of SNPs (or predictors) ).</p>
</td></tr>
<tr><td><code id="aSPUw_+3A_cov">cov</code></td>
<td>
<p>Covariates. A matrix with dimension n by p (n :number of observation, p : number of covariates).</p>
</td></tr>
<tr><td><code id="aSPUw_+3A_resample">resample</code></td>
<td>
<p>Use &quot;perm&quot; for residual permutations, &quot;sim&quot; for simulations from the null distribution, and &quot;boot&quot; for parametric bootstrap.</p>
</td></tr>
<tr><td><code id="aSPUw_+3A_model">model</code></td>
<td>
<p>Use &quot;gaussian&quot; for a quantitative trait, and use &quot;binomial&quot; for a binary trait.</p>
</td></tr>
<tr><td><code id="aSPUw_+3A_pow">pow</code></td>
<td>
<p>power used in SPU test. A vector of the powers.</p>
</td></tr>
<tr><td><code id="aSPUw_+3A_n.perm">n.perm</code></td>
<td>
<p>number of permutations or bootstraps.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object, Ts : Test Statistics for the SPUw and aSPUw test.
pvs : p-values for the SPUw and aSPUw test.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak, Junghi Kim and Wei Pan
</p>


<h3>References</h3>

<p>Junghi Kim, Jeffrey R Wozniak, Bryon A Mueller, Xiaotong Shen and Wei Pan (2014)
Comparison of statistical tests for group differences in brain functional networks,
Neuroimage, 1;101:681-694
</p>


<h3>See Also</h3>

<p><code><a href="#topic+aSPU">aSPU</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(exdat)

# increase n.perm for accurate p-value
## Not run: out &lt;- aSPUw(exdat$Y, exdat$X, pow = c(1:8, Inf), n.perm = 1000) 


out$Ts
# This is a vector of Test Statistics for SPU and aSPU tests.
# SPU1 to SPUInf corresponds with the option pow=c(1:8, Inf)
# They are SPU test statistics.
# The last element aSPU is minimum of them, aSPU statistic.

out$pvs
# This is a vector of p-values for SPU and aSPU tests.
# SPU1 to SPUInf corresponds with the option pow=c(1:8, Inf)
# They are p-values for corresponding SPU tests.
# The last element is p-value of aSPU test.

</code></pre>

<hr>
<h2 id='estcov'>estcov</h2><span id='topic+estcov'></span>

<h3>Description</h3>

<p>Estimate the covariance matrix of multiple traits based on their (null) summary Z-scores.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estcov(allZ, Ps = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estcov_+3A_allz">allZ</code></td>
<td>
<p>matrix of summary Z-scores for all SNP. each row for SNP; each column for single trait.</p>
</td></tr>
<tr><td><code id="estcov_+3A_ps">Ps</code></td>
<td>
<p>TRUE if input is p-value, FALSE if input is Z-scores. The default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>estimated correlation matrix.
</p>


<h3>Author(s)</h3>

<p>Junghi Kim, Yun Bai and Wei Pan
</p>


<h3>References</h3>

<p>Junghi Kim, Yun Bai and Wei Pan (2015) An Adaptive Association Test for Multiple Phenotypes with GWAS Summary Statistics, Genetic Epidemiology, 8:651-663
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MTaSPUs">MTaSPUs</a></code> <code><a href="#topic+minP">minP</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# -- n.snp: number of SNPs
# -- n.trait: number of traits
# -- n.subject: number of subjects

n.snp &lt;- 100
n.traits &lt;- 10
n.subjects &lt;- 1000
traits &lt;- matrix(rnorm(n.subjects*n.traits), n.subjects, n.traits)
v &lt;- cov(traits)
allZ &lt;- rmvnorm(n.snp, Sigma=v)
colnames(allZ) &lt;- paste("trait", 1:n.traits, sep="")
rownames(allZ) &lt;- paste("snp", 1:n.snp, sep="")


r &lt;- estcov(allZ)
MTaSPUs(Z = allZ, v = r, B = 100, pow = c(1:4, Inf), transform = FALSE)
MTaSPUs(Z = allZ[1,], v = r, B = 100, pow = c(1:4, Inf), transform = FALSE)
minP(Zi= allZ[1,], r = r)

</code></pre>

<hr>
<h2 id='exdat'>An example data set</h2><span id='topic+exdat'></span>

<h3>Description</h3>

<p>The exdat data set is list of three objects. exdat$Y is a vector of length
1000, 500 0s and 500 1s. exdat$X is a matrix of 1000 by 10. This
simulated X matrix is assumed to be rare variants. All
eliments are 0, 1 or 2. exdat$SNP0indx is a vector of length 10. The
values are 1 or 0, 0 indicate corresponding column of X matrix have no
association with Y vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(exdat)
</code></pre>


<h3>Examples</h3>

<pre><code class='language-R'>data(exdat)
exdat$X[c(1:10, 501:510),]
exdat$Y[c(1:10, 501:510)]

# 3rd and 10th column of X have no association with Y
exdat$SNP0indx

</code></pre>

<hr>
<h2 id='GATES2'>Gene-based Association Test that uses an extended Simes procedure (GATES) for single trait - SNP set association</h2><span id='topic+GATES2'></span>

<h3>Description</h3>

<p>Get the p-value of GATES. Usually it is used to get genomewise p-values.
This function is taken from postgwas package.
There is a little modification of the code GATES in postgwas package.
1) The approximated matrix may have negative eigen value, we modified it not to have negative values; 2) we added one more return (the key gene location) for Hyst method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GATES2(ldmatrix, p)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GATES2_+3A_ldmatrix">ldmatrix</code></td>
<td>
<p>numeric. A correlation matrix of SNPs,
dimensions matching the p and snps arguments.</p>
</td></tr>
<tr><td><code id="GATES2_+3A_p">p</code></td>
<td>
<p>p-value for each SNPs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A p-value of GATES and the key gene location (to be used by Hyst).
</p>


<h3>Author(s)</h3>

<p>Milan Hiersche(taken from pastgwas package), Il-Youp Kwak(modified a little)
</p>


<h3>References</h3>

<p>Miao-Xin Li, Hong-Sheng Gui, Johnny S.H. Kwan and Pak C. Sham (2011)
GATES: A Rapid and Powerful Gene-Based Association Test Using Extended Simes Procedure
The American Journal of Human Genetics 88, 283-293
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Hyst">Hyst</a></code> <code><a href="#topic+GatesSimes">GatesSimes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#simula &lt;- simPathAR1Snp(nGenes=20, nGenes1=1, nSNPlim=c(1, 20), nSNP0=1:3,
#                           LOR=.2, rholim=c(0,0),
#                           n=30, MAFlim=c(0.05, 0.4), p0=0.05)
#Ps &lt;- getlogitp(simula$Y, simula$X)

## get correlation of SNPs using controls
#ldmat &lt;- cor(simula$X[ simula$Y == 0, ])

#o.pvec = order(Ps)
# ldmat &lt;- ldmat[o.pvec, o.pvec]
#(gatesp &lt;- GATES2(ldmat, sort(Ps))[1])


</code></pre>

<hr>
<h2 id='GatesSimes'>GATES-Simes test for single trait - pathway association.</h2><span id='topic+GatesSimes'></span>

<h3>Description</h3>

<p>Get the p-value of GATES-Simes. It uses an extended Simes procedure to combine GATES p-values across multiple genes in a pathway.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GatesSimes(pvec, ldmatrix, snp.info, gene.info)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GatesSimes_+3A_pvec">pvec</code></td>
<td>
<p>p-values for each SNP.</p>
</td></tr>
<tr><td><code id="GatesSimes_+3A_ldmatrix">ldmatrix</code></td>
<td>
<p>numeric. A correlation matrix of SNPs with dimensions matching the length of pvec (the number of SNPs).</p>
</td></tr>
<tr><td><code id="GatesSimes_+3A_snp.info">snp.info</code></td>
<td>
<p>SNP information matrix, the 1st column is SNP id, 2nd column is chromosome #, 3rd column indicates SNP location.</p>
</td></tr>
<tr><td><code id="GatesSimes_+3A_gene.info">gene.info</code></td>
<td>
<p>GENE information matrix, The 1st column is GENE id, 2nd column is chromosome #, 3rd and 4th column indicate start and end positions of the gene.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A p-value.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak and Wei Pan
</p>


<h3>References</h3>

<p>Hongsheng Gui, Miaoxin Li, Pak C Sham and Stacey S Cherny (2011)
Comparisons of seven algorithms for pathway analysis using the WTCCC Crohn's Disease
BMC Research Notes, 4:386
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Hyst">Hyst</a></code>  <code><a href="#topic+GATES2">GATES2</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#simula &lt;- simPathAR1Snp(nGenes=20, nGenes1=1, nSNPlim=c(1, 20), nSNP0=1:3,
#                           LOR=.2, rholim=c(0,0),
#                           n=30, MAFlim=c(0.05, 0.4), p0=0.05)
#logitp &lt;- getlogitp(simula$Y, simula$X)

## get correlation of SNPs using controls
#ldmat &lt;- cor(simula$X[ simula$Y == 0, ])
#out &lt;- GatesSimes(pvec = logitp, ldmatrix = ldmat, snp.info = simula$snp.info,
#                  gene.info = simula$gene.info)
#out


</code></pre>

<hr>
<h2 id='GEEaSPU'>The SPU and aSPU tests for multiple traits - single SNP association in generalized estimating equations.</h2><span id='topic+GEEaSPU'></span>

<h3>Description</h3>

<p>It gives p-values of the GEESPU tests and GEEaSPU test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GEEaSPU(
  traits,
  geno,
  Z = NULL,
  model = c("binomial", "gaussian"),
  gamma = c(1:8, Inf),
  n.sim = 1000,
  corstr = "independence"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GEEaSPU_+3A_traits">traits</code></td>
<td>
<p>trait matrix. The row for individuals and the column for traits.</p>
</td></tr>
<tr><td><code id="GEEaSPU_+3A_geno">geno</code></td>
<td>
<p>A matrix of genetic information.</p>
</td></tr>
<tr><td><code id="GEEaSPU_+3A_z">Z</code></td>
<td>
<p>covariates.</p>
</td></tr>
<tr><td><code id="GEEaSPU_+3A_model">model</code></td>
<td>
<p>Use &quot;gaussian&quot; for a quantitative trait, and use &quot;binomial&quot; for a binary trait.</p>
</td></tr>
<tr><td><code id="GEEaSPU_+3A_gamma">gamma</code></td>
<td>
<p>power used in GEEaSPU test. A vector of the powers.</p>
</td></tr>
<tr><td><code id="GEEaSPU_+3A_n.sim">n.sim</code></td>
<td>
<p>number of simulations.</p>
</td></tr>
<tr><td><code id="GEEaSPU_+3A_corstr">corstr</code></td>
<td>
<p>a character string specifying the correlation structure. The following are permitted: &quot;independence&quot;, &quot;fixed&quot;, &quot;stat_M_dep&quot;, &quot;non_stat_M_dep&quot;, &quot;exchangeable&quot;, &quot;AR-M&quot; and &quot;unstructured&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>p-values for the GEE-SPU and GEE-aSPU test.
</p>


<h3>Author(s)</h3>

<p>Junghi Kim, Wei Pan and Il-Youp Kwak
</p>


<h3>References</h3>

<p>Yiwei Zhang, Zhiyuan Xu, Xiaotong Shen, Wei Pan (2014)
Testing for association with multiple traits in generalized estimation equations, with application to neuroimaging data.
Neuroimage. 96:309-25
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
traits &lt;- matrix(rnorm(100*5, 0,1), ncol=5)
Z &lt;- rnorm(100, 2, 0.5)
geno &lt;- rbinom(100, 2, 0.5)
out &lt;- GEEaSPU(traits, geno, Z = NULL, model = "gaussian", 
	  gamma = c(1:8,Inf), n.sim = 100)

</code></pre>

<hr>
<h2 id='getlogitp'>Get the p-values of the multiple SNPs based on the Wald test in a logistic regresion model.</h2><span id='topic+getlogitp'></span>

<h3>Description</h3>

<p>Get p-value using logistic regresion for each of the multiple SNPs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getlogitp(Y, X)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getlogitp_+3A_y">Y</code></td>
<td>
<p>Response or phenotype data. It can be a disease indicator; =0 for controls, =1 for cases.</p>
</td></tr>
<tr><td><code id="getlogitp_+3A_x">X</code></td>
<td>
<p>Genotype or other data; each row for a subject, and each column
for an SNP (or a predictor). The value of each SNP is the # of the copies
for an allele. A matrix with dimension n by p (n : number of observation, p : number of SNPs (or predictors) ).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>p-values for each SNPs.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GATES2">GATES2</a></code> <code><a href="#topic+GatesSimes">GatesSimes</a></code> <code><a href="#topic+Hyst">Hyst</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#simula &lt;- simPathAR1Snp(nGenes=20, nGenes1=1, nSNPlim=c(1, 20), nSNP0=1:3,
#                           LOR=.2, rholim=c(0,0),
#                           n=10, MAFlim=c(0.05, 0.4), p0=0.05)
#logitp &lt;- getlogitp(simula$Y, simula$X)


</code></pre>

<hr>
<h2 id='Hyst'>HYST (Hybrid Set-based Test) for single trait - pathway association</h2><span id='topic+Hyst'></span>

<h3>Description</h3>

<p>Get a p-value using HYST.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Hyst(pvec, ldmatrix, snp.info, gene.info)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Hyst_+3A_pvec">pvec</code></td>
<td>
<p>p-values for each SNP.</p>
</td></tr>
<tr><td><code id="Hyst_+3A_ldmatrix">ldmatrix</code></td>
<td>
<p>numeric. A correlation matrix of SNPs with dimensions matching the length of pvec (the number of SNPs).</p>
</td></tr>
<tr><td><code id="Hyst_+3A_snp.info">snp.info</code></td>
<td>
<p>SNP information matrix, the 1st column is SNP id, 2nd column is chromosome #, 3rd column indicates SNP location.</p>
</td></tr>
<tr><td><code id="Hyst_+3A_gene.info">gene.info</code></td>
<td>
<p>GENE information matrix, The 1st column is GENE id, 2nd column is chromosome #, 3rd and 4th column indicate start and end positions of the gene.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A p-value.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak and Wei Pan
</p>


<h3>References</h3>

<p>Miao-Xin Li, Johnny S.H. Kwan and Pak C. Sham (2012)
HYST: A Hybrid Set-Based Test for Genome-wide Association Studies, with Application to Protein-Protein Interaction-Based Association Analysis
The American Journal of Human Genetics, 91, 478-488.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GatesSimes">GatesSimes</a></code> <code><a href="#topic+GATES2">GATES2</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#simula &lt;- simPathAR1Snp(nGenes=20, nGenes1=1, nSNPlim=c(1, 20), nSNP0=1:3,
#                           LOR=.2, rholim=c(0,0),
#                           n=30, MAFlim=c(0.05, 0.4), p0=0.05)
#logitp &lt;- getlogitp(simula$Y, simula$X)

## get correlation of SNPs using controls
#ldmat &lt;- cor(simula$X[ simula$Y == 0, ])
#out &lt;- Hyst(pvec = logitp, ldmatrix = ldmat, snp.info = simula$snp.info,
#            gene.info = simula$gene.info)

</code></pre>

<hr>
<h2 id='kegg9'>A subset of coronary artery disease (CAD) data </h2><span id='topic+kegg9'></span>

<h3>Description</h3>

<p>Data on coronary artery disease, myocardial infarction have been
contributed by CARDIoGRAMplusC4D investigators and have been
downloaded from www.CARDIOGRAMPLUSC4D.ORG
</p>
<p>The data set contains P value data for coronary artery disease (CAD). 
We mapped these SNPs to the 9th KEGG pathway. 
</p>
<p>gene.info is a contains 16 gene information of 9th Kegg pathway. 
The 1st column is the name of the gene, 2nd column is the chromosome number, 
3rd column is where the gene starts and 4th column is where the gene ends.
</p>
<p>snp.info contains 330 snp informations mapped on this Kegg pathway. 
The 1st column is the rsID of the SNP, 2nd column is the chromosome number
and the 3rd column is the location of the SNP. 
</p>
<p>Ps is a vector containing p-value information for mapped SNPs.
</p>
<p>The 'PPs' is a list object contains the SNP information for each genes. For example, 
PPs$CEL contains snp.info matrix mapped on gene 'CEL'. 
The 1st column is the rsID of the SNP, 2nd column is the chromosome number,
the 3rd column is the location of the SNP and the 4th column is p-value of the SNP.
</p>
<p>nP is a subvector of Ps. These are SNPs on reference population(Hapmap CEU phase 2 data, 
downloadable from plink, SNPs with MAF less than 5 percent omitted) 
</p>
<p>ldmatrix is a correaltion matrix of SNPs from the reference population.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(kegg9)
</code></pre>


<h3>References</h3>

<p>Schunkert H, Konig IR, Kathiresan S, Reilly MP, Assimes TL, Holm H et
al. (2011) Large-scale association analysis identifies 13 new susceptibility
loci for coronary artery disease. Nat Genet. 43: 333-338
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(kegg9)

## gene informations
kegg9$gene.info

## SNPs mapped on 3rd and 4th gene in 9th Kegg pathway
kegg9$PPs[3:4]

## snp information
kegg9$snp.info

## The 1st 10 P-values of SNPs mapped on 9th Kegg pathway. 
kegg9$nP[1:10]

## correlation matrix among those SNPs
kegg9$ldmatrix[1:10,1:10]

</code></pre>

<hr>
<h2 id='minP'>minP test.</h2><span id='topic+minP'></span>

<h3>Description</h3>

<p>Return exact minP test p-value for multiple traits - single SNP association.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>minP(Zi, r)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="minP_+3A_zi">Zi</code></td>
<td>
<p>a vector of summary Z-scores for single SNP</p>
</td></tr>
<tr><td><code id="minP_+3A_r">r</code></td>
<td>
<p>estimated correlation matrix based on the summary Z-scores (output of estcov)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>return exact minP test
</p>


<h3>Author(s)</h3>

<p>Junghi Kim, Yun Bai and Wei Pan
</p>


<h3>References</h3>

<p>Junghi Kim, Yun Bai and Wei Pan (2015) An Adaptive Association Test for Multiple Phenotypes with GWAS Summary Statistics, Genetic Epidemiology, 8:651-663
</p>


<h3>See Also</h3>

<p><code><a href="#topic+estcov">estcov</a></code> <code><a href="#topic+MTaSPUs">MTaSPUs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# -- n.snp: number of SNPs
# -- n.trait: number of traits
# -- n.subject: number of subjects

n.snp &lt;- 100
n.traits &lt;- 10
n.subjects &lt;- 1000
traits &lt;- matrix(rnorm(n.subjects*n.traits), n.subjects, n.traits)
v &lt;- cov(traits)
allZ &lt;- rmvnorm(n.snp, Sigma=v)
colnames(allZ) &lt;- paste("trait", 1:n.traits, sep="")
rownames(allZ) &lt;- paste("snp", 1:n.snp, sep="")


r &lt;- estcov(allZ)
MTaSPUs(Z = allZ, v = r, B = 100, pow = c(1:4, Inf), transform = FALSE)
MTaSPUs(Z = allZ[1,], v = r, B = 100, pow = c(1:4, Inf), transform = FALSE)
minP(Zi= allZ[1,], r = r)

</code></pre>

<hr>
<h2 id='MTaSPUs'>The SPU and aSPU tests for multiple traits - single SNP association with GWAS summary statistics.</h2><span id='topic+MTaSPUs'></span>

<h3>Description</h3>

<p>SNP based adaptive association test for multiple phenotypes with GWAS summary statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MTaSPUs(Z, v, B, pow, transform = FALSE, Ps = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MTaSPUs_+3A_z">Z</code></td>
<td>
<p>matrix of summary Z-scores, SNPs in rows and traits in columns. Or a vector of summary Z-scores for a single snp</p>
</td></tr>
<tr><td><code id="MTaSPUs_+3A_v">v</code></td>
<td>
<p>estimated correlation matrix based on the summary Z-scores (output of estcov)</p>
</td></tr>
<tr><td><code id="MTaSPUs_+3A_b">B</code></td>
<td>
<p>number of Monte Carlo samples simulated to compute p-values, the maximum number of MC simulations is 1e8</p>
</td></tr>
<tr><td><code id="MTaSPUs_+3A_pow">pow</code></td>
<td>
<p>power used in SPU test. A vector of the powers.</p>
</td></tr>
<tr><td><code id="MTaSPUs_+3A_transform">transform</code></td>
<td>
<p>if TRUE, the inference is made on transformed Z</p>
</td></tr>
<tr><td><code id="MTaSPUs_+3A_ps">Ps</code></td>
<td>
<p>TRUE if input is p-value, FALSE if input is Z-scores. The default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>compute p-values for SPU(gamma) i.e. pow=1:8, and infinity
aSPU, based on the minimum p-values over SPU(power)
each row for single SNP
</p>


<h3>Author(s)</h3>

<p>Junghi Kim, Yun Bai and Wei Pan
</p>


<h3>References</h3>

<p>Junghi Kim, Yun Bai and Wei Pan (2015) An Adaptive Association Test for Multiple Phenotypes with GWAS Summary Statistics, Genetic Epidemiology, 8:651-663
</p>


<h3>See Also</h3>

<p><code><a href="#topic+minP">minP</a></code> <code><a href="#topic+estcov">estcov</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# -- n.snp: number of SNPs
# -- n.trait: number of traits
# -- n.subject: number of subjects

n.snp &lt;- 100
n.traits &lt;- 10
n.subjects &lt;- 1000
traits &lt;- matrix(rnorm(n.subjects*n.traits), n.subjects, n.traits)
v &lt;- cov(traits)
allZ &lt;- rmvnorm(n.snp, Sigma=v)
colnames(allZ) &lt;- paste("trait", 1:n.traits, sep="")
rownames(allZ) &lt;- paste("snp", 1:n.snp, sep="")


r &lt;- estcov(allZ)
MTaSPUs(Z = allZ, v = r, B = 100, pow = c(1:4, Inf), transform = FALSE)
MTaSPUs(Z = allZ[1,], v = r, B = 100, pow = c(1:4, Inf), transform = FALSE)
minP(Zi= allZ[1,], r = r)

</code></pre>

<hr>
<h2 id='MTaSPUsSet'>gene-Multitrait Sum of Powered Score (MTSPUsSet) tests and adaptive MTSPUsSet (MTaSPUsSet) test for multi trait - SNP set association with GWAS summary statistics.</h2><span id='topic+MTaSPUsSet'></span>

<h3>Description</h3>

<p>It gives p-values of the MTSPUsSet tests and MTaSPUsSet test with GWAS summary statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MTaSPUsSet(
  Zs,
  corSNP,
  corPhe,
  pow = c(1, 2, 4, 8),
  pow2 = c(1, 2, 4, 8),
  n.perm = 5000,
  Ps = FALSE,
  prune = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MTaSPUsSet_+3A_zs">Zs</code></td>
<td>
<p>Z-score matrix. row represent SNPs and column represent traits. It could be P-values if the Ps option is TRUE.</p>
</td></tr>
<tr><td><code id="MTaSPUsSet_+3A_corsnp">corSNP</code></td>
<td>
<p>Correlation matirx of the SNPs to be tested; estimated from a
reference panel (based on the same set of the reference alleles as
used in calculating Z-scores).</p>
</td></tr>
<tr><td><code id="MTaSPUsSet_+3A_corphe">corPhe</code></td>
<td>
<p>Correlation matirx of phenotypes to be tested; Estimated from Z-scores.</p>
</td></tr>
<tr><td><code id="MTaSPUsSet_+3A_pow">pow</code></td>
<td>
<p>SNP specific power(gamma values) used in MTSPUsSet test.</p>
</td></tr>
<tr><td><code id="MTaSPUsSet_+3A_pow2">pow2</code></td>
<td>
<p>GENE specific power(gamma values) used in MTSPUsSet test.</p>
</td></tr>
<tr><td><code id="MTaSPUsSet_+3A_n.perm">n.perm</code></td>
<td>
<p>number of permutations or bootstraps.</p>
</td></tr>
<tr><td><code id="MTaSPUsSet_+3A_ps">Ps</code></td>
<td>
<p>TRUE if input is p-value, FALSE if input is Z-scores. The default is FALSE.</p>
</td></tr>
<tr><td><code id="MTaSPUsSet_+3A_prune">prune</code></td>
<td>
<p>if it is TRUE, do pruing before the test using pruneSNP function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector object, MTSPUsSet test P values and MTaSPUsSet P value.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak and Wei Pan
</p>


<h3>References</h3>

<p>Il-Youp Kwak, Wei Pan (2017)
Gene- and pathway-based association tests for multiple
traits with GWAS summary statistics, Bioinformatics, 33(1), 64-71.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(SAMD11)
attach(SAMD11)
## example analysis using MTaSPUsSet test.
(outFZ &lt;- MTaSPUsSet(ZsF, corSNP=corSNPF, corPhe = corPheF,
      pow=c(1,2,4,8),  pow2 = c(1,2,4,8), n.perm=10, Ps=FALSE))

</code></pre>

<hr>
<h2 id='MTaSPUsSetPath'>Multitraits - Pathway based Sum of Powered Score tests (MTSPUsSetPath) and adaptive MTSPUsSetpath (MTaSPUsSetPath) test for multi trait - pathway association with GWAS summary statistics.</h2><span id='topic+MTaSPUsSetPath'></span>

<h3>Description</h3>

<p>It gives p-values of the MTSPUsSetPath tests and MTaSPUsSetPath test with GWAS summary statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MTaSPUsSetPath(
  Zs,
  corPhe,
  corSNP,
  pow1 = c(1, 2, 4, 8),
  pow2 = c(1, 2, 4, 8),
  pow3 = c(1, 2, 4, 8),
  snp.info,
  gene.info,
  n.perm = 1000,
  Ps = FALSE,
  prune = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MTaSPUsSetPath_+3A_zs">Zs</code></td>
<td>
<p>Z-score matrix. row represent SNPs and column represent traits. It could be P-values if the Ps option is TRUE.</p>
</td></tr>
<tr><td><code id="MTaSPUsSetPath_+3A_corphe">corPhe</code></td>
<td>
<p>Correlation matirx of phenotypes to be tested; Estimated from Z-scores.</p>
</td></tr>
<tr><td><code id="MTaSPUsSetPath_+3A_corsnp">corSNP</code></td>
<td>
<p>Correlation matirx of the SNPs to be tested; estimated from a
reference panel (based on the same set of the reference alleles as
used in calculating Z-scores).</p>
</td></tr>
<tr><td><code id="MTaSPUsSetPath_+3A_pow1">pow1</code></td>
<td>
<p>SNP specific power(gamma values) used in MTSPUsSetPath test.</p>
</td></tr>
<tr><td><code id="MTaSPUsSetPath_+3A_pow2">pow2</code></td>
<td>
<p>GENE specific power(gamma values) used in MTSPUsSetPath test.</p>
</td></tr>
<tr><td><code id="MTaSPUsSetPath_+3A_pow3">pow3</code></td>
<td>
<p>Trait specific power(gamma values) used in MTSPUsSetPath test.</p>
</td></tr>
<tr><td><code id="MTaSPUsSetPath_+3A_snp.info">snp.info</code></td>
<td>
<p>SNP information matrix, the 1st column is SNP id, 2nd column is chromosome #, 3rd column indicates SNP location.</p>
</td></tr>
<tr><td><code id="MTaSPUsSetPath_+3A_gene.info">gene.info</code></td>
<td>
<p>GENE information matrix, The 1st column is GENE id, 2nd column is chromosome #, 3rd and 4th column indicate start and end positions of the gene.</p>
</td></tr>
<tr><td><code id="MTaSPUsSetPath_+3A_n.perm">n.perm</code></td>
<td>
<p>number of permutations.</p>
</td></tr>
<tr><td><code id="MTaSPUsSetPath_+3A_ps">Ps</code></td>
<td>
<p>TRUE if input is p-value, FALSE if input is Z-scores. The default is FALSE.</p>
</td></tr>
<tr><td><code id="MTaSPUsSetPath_+3A_prune">prune</code></td>
<td>
<p>if it is TRUE, do pruing before the test using pruneSNP function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>P-values for MTSPUsSetpath tests and MTaSPUsSetpPath test.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak and Wei Pan
</p>


<h3>References</h3>

<p>Il-Youp Kwak, Wei Pan (2017)
Gene- and pathway-based association tests for multiple
traits with GWAS summary statistics, Bioinformatics, 33(1), 64-71
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MTaSPUsSet">MTaSPUsSet</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
Zs &lt;- cbind ( c( 0.3, 0.2, 0.1,0.5,1.2),
          c(-.1, .3,-.1,.1,1.2) )

varSNP = cbind( c( 1, .1,0, 0, .11),
               c(.1,1, 0, 0, 0),
               c(0,0, 1, 0, 0),
               c(0,0, 0, 1, 0),
               c(.11,0,0,0,1) )

varPhe = cbind( c( 1, -.1),
               c(-.1,1) )

gene.info = data.frame( Gnm = c( "G1", "G2"), chr = c(1,3),
                    loc1 = c(0, 0), loc2 = c(10,10) )
snp.info = data.frame( rsid = c("rs1", "rs2", "rs3", "rs4", "rs5"),
                chr = c(1,1,3,3,3), loc = c(1,2,1,2,3) )
 out &lt;- MTaSPUsSetPath(Zs, corPhe = varPhe, corSNP=varSNP,
            n.perm = 100, snp.info = snp.info, gene.info = gene.info)
out

</code></pre>

<hr>
<h2 id='MTaSPUsSetScore'>gene-Multitrait Sum of Powered Score (MTSPUsSetScore) tests and adaptive MTSPUsSet (MTaSPUsSet) test for multi trait - SNP set association with GWAS summary statistics. (Score version added)</h2><span id='topic+MTaSPUsSetScore'></span>

<h3>Description</h3>

<p>It gives p-values of the MTSPUsSet tests MTScore and MTaSPUsSet_Score test with GWAS summary statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MTaSPUsSetScore(
  Zs,
  corSNP,
  corPhe,
  pow = c(1, 2, 4, 8),
  pow2 = c(1, 2, 4, 8),
  n.perm = 5000,
  Ps = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MTaSPUsSetScore_+3A_zs">Zs</code></td>
<td>
<p>Z-score matrix. row represent SNPs and column represent traits. It could be P-values if the Ps option is TRUE.</p>
</td></tr>
<tr><td><code id="MTaSPUsSetScore_+3A_corsnp">corSNP</code></td>
<td>
<p>Correlation matirx of the SNPs to be tested; estimated from a
reference panel (based on the same set of the reference alleles as
used in calculating Z-scores).</p>
</td></tr>
<tr><td><code id="MTaSPUsSetScore_+3A_corphe">corPhe</code></td>
<td>
<p>Correlation matirx of phenotypes to be tested; Estimated from Z-scores.</p>
</td></tr>
<tr><td><code id="MTaSPUsSetScore_+3A_pow">pow</code></td>
<td>
<p>SNP specific power(gamma values) used in MTSPUsSetScore test.</p>
</td></tr>
<tr><td><code id="MTaSPUsSetScore_+3A_pow2">pow2</code></td>
<td>
<p>GENE specific power(gamma values) used in MTSPUsSetScore test.</p>
</td></tr>
<tr><td><code id="MTaSPUsSetScore_+3A_n.perm">n.perm</code></td>
<td>
<p>number of permutations or bootstraps.</p>
</td></tr>
<tr><td><code id="MTaSPUsSetScore_+3A_ps">Ps</code></td>
<td>
<p>TRUE if input is p-value, FALSE if input is Z-scores. The default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector object, MTSPUsSet test, MTScore test P values and MTaSPUsSet_Score P value.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak and Wei Pan
</p>


<h3>References</h3>

<p>Il-Youp Kwak, Wei Pan (2017)
Gene- and pathway-based association tests for multiple
traits with GWAS summary statistics, Bioinformatics. 33(1), 64-71
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(SAMD11)
attach(SAMD11)
## example analysis using aSPUM test.
(outFZ &lt;- MTaSPUsSetScore(ZsF, corSNP=corSNPF, corPhe = corPheF,
      pow=c(1,2,4,8),  pow2 = c(1,2,4,8), n.perm=10, Ps=FALSE))

</code></pre>

<hr>
<h2 id='plotPmat'>Image plot of P-value matrix.</h2><span id='topic+plotPmat'></span>

<h3>Description</h3>

<p>It gives a P-value image for a given P-value matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPmat(
  Ps,
  zlim = NULL,
  main = NULL,
  yt = NULL,
  xlab = "SNPs",
  thresh = -log(5e-08, 10),
  trait.names = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotPmat_+3A_ps">Ps</code></td>
<td>
<p>P-value matrix. Row represent traits, column represent SNPs</p>
</td></tr>
<tr><td><code id="plotPmat_+3A_zlim">zlim</code></td>
<td>
<p>-log 10 transformed p value range.</p>
</td></tr>
<tr><td><code id="plotPmat_+3A_main">main</code></td>
<td>
<p>main title.</p>
</td></tr>
<tr><td><code id="plotPmat_+3A_yt">yt</code></td>
<td>
<p>it controls the location of trait names.</p>
</td></tr>
<tr><td><code id="plotPmat_+3A_xlab">xlab</code></td>
<td>
<p>lable for x axis. Default is &quot;SNPs&quot;.</p>
</td></tr>
<tr><td><code id="plotPmat_+3A_thresh">thresh</code></td>
<td>
<p>Default is set to widely used genome wise threshold -log(5e-8,10).</p>
</td></tr>
<tr><td><code id="plotPmat_+3A_trait.names">trait.names</code></td>
<td>
<p>A vector of trait names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Image of P-values matrix.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak and Wei Pan
</p>


<h3>References</h3>

<p>Il-Youp Kwak, Wei Pan (2017)
Gene- and pathway-based association tests for multiple traits with GWAS summary statistics, Bioinformatics. 33(1), 64-71
</p>


<h3>See Also</h3>

<p><code><a href="#topic+aSPUs">aSPUs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Say we have 3 traits and their p-values at 5 SNPs. 
Ps &lt;- rbind( c(0.001, 0.4, 0.5, 0.00000001, .1),
           c(0.03, 0.3, 0.3, 0.00001, .2),
           c(0.01, 0.2, 0.4, 0.001, .0001) )

## We can visualize it using plotPmat function.
plotPmat(Ps)


</code></pre>

<hr>
<h2 id='pruneSNP'>Prune SNPs (related to aSPUs, aSPUsPath, MTaSPUsSet)</h2><span id='topic+pruneSNP'></span>

<h3>Description</h3>

<p>When correlation matirx have highly correlated SNPs, the performance of aSPUs, aSPUsPath and MTaSPUsSet are not very good. Do pruning using this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pruneSNP(corSNP, rup = 0.95, rdown = -0.95)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pruneSNP_+3A_corsnp">corSNP</code></td>
<td>
<p>Correlation matirx of the SNPs to be tested; estimated from a
reference panel (based on the same set of the reference alleles as
used in calculating Z-scores).</p>
</td></tr>
<tr><td><code id="pruneSNP_+3A_rup">rup</code></td>
<td>
<p>pruning criteria, erase one SNP when correlation between two SNPs are larger than this value.</p>
</td></tr>
<tr><td><code id="pruneSNP_+3A_rdown">rdown</code></td>
<td>
<p>pruning criteria, erase one SNP when correlation between two SNPs are smaller than this value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list object pruned.corSNP and to.erase. pruned.corSNP is pruned correlation matrix. to.erase is SNP index to erase to get purned object. (i.e. corSNP[-to.erase, -to.erase] = pruned.corSNP )
</p>


<h3>Author(s)</h3>

<p>Il Youp Kwak
</p>


<h3>See Also</h3>

<p><code><a href="#topic+aSPUs">aSPUs</a></code> <code><a href="#topic+aSPUsPath">aSPUsPath</a></code> <code><a href="#topic+MTaSPUsSet">MTaSPUsSet</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(kegg9)
## example analysis using aSPUM test.
g &lt;- kegg9$gene.info[1,1]  #  SOAT1 
## Take snps mapped on gene "SOAT1" from the information of gene.info and snp.info. 
snps &lt;- which( ( kegg9$snp.info[,2] == kegg9$gene.info[kegg9$gene.info[,1] == g, 2] ) &amp;
                 (kegg9$snp.info[,3] &gt; kegg9$gene.info[kegg9$gene.info[,1] == g, 3] ) &amp;
                 (kegg9$snp.info[,3] &lt; kegg9$gene.info[kegg9$gene.info[,1] == g, 4] )  )
## Take subsets
newP &lt;- kegg9$nP[snps] ;
ldsub &lt;- kegg9$ldmatrix[snps, snps];

prSNP &lt;- pruneSNP(ldsub) 
newP &lt;- newP[-prSNP$to.erase]
ldsub &lt;- ldsub[-prSNP$to.erase, -prSNP$to.erase ]

## Get p-value for gene SOAT1. Read vignette for details.
out &lt;- aSPUs(newP, corSNP=ldsub , pow=c(1:8, Inf), n.perm=100, Ps=TRUE)

out$Ts
# This is a vector of Test Statistics for SPUM and aSPUM tests.
# SPUs1 to SPUsInf corresponds with the option pow=c(1:8, Inf)
# They are SPUs test statistics.
# The last element aSPUs is minimum of them, aSPUs statistic.

out$pvs
# This is a vector of p-values for SPUs and aSPUs tests.
# SPUs1 to SPUsInf corresponds with the option pow=c(1:8, Inf)
# They are p-values for corresponding SPUs tests.
# The last element is p-value of aSPUs test.

</code></pre>

<hr>
<h2 id='SAMD11'>An example data for MTaSPUsSet test</h2><span id='topic+SAMD11'></span>

<h3>Description</h3>

<p>Genetic Investigation of ANthropometric Traits (GIANT) consortium
data contain P-values of 2.7 million SNPs with six anthropometric
traits that are well established to represent body size and shape:
height, weight, BMI, waist circumference (WC), hip
circumference(HIP). 
</p>
<p>We mapped SNPs on gene SAMD11. This subdata contains P values, Z
scores of SNPs mapped on gene SAMD11. It also contains correlation
among SNPs and correlation among phenotypes for demostrated example of
MTaSPUsSet test. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(SAMD11)
</code></pre>


<h3>References</h3>

<p>Shungin D, Winkler TW, Croteau-Chonka DC, Ferreira T, Locke AE, Magi R,
Strawbridge R, Pers TH, Fischer K, Justice AE, Workalemahu T, Wu JM, et
al. (2015). New genetic loci link adipose and insulin biology to body
fat distribution. Nature 518: 187-196.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SAMD11)

## Z-scores for man
SAMD11$ZsM

## P-values for man
SAMD11$PsM

## correlation among SNPs for man
SAMD11$corSNPM

## correlation among phenotypes for man
SAMD11$corPheM

</code></pre>

<hr>
<h2 id='simPathAR1Snp'>Simulating a pathway with multiple SNPs.</h2><span id='topic+simPathAR1Snp'></span>

<h3>Description</h3>

<p>It gives a simulated SNPs consisting of multiple genes in a pathway. Each SNPs from a latent multivariate Gaussian variable with an AR1 correlation structure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simPathAR1Snp(
  nGenes = 10,
  nGenes1 = 5,
  nSNPs = NULL,
  ncSNPs = NULL,
  nSNPlim = c(1, 20),
  nSNP0 = 1:3,
  LOR = 0.3,
  n = 100,
  MAFlim = c(0.05, 0.4),
  rholim = c(0, 0),
  p0 = 0.05,
  noncausal = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simPathAR1Snp_+3A_ngenes">nGenes</code></td>
<td>
<p>The number of total genes.</p>
</td></tr>
<tr><td><code id="simPathAR1Snp_+3A_ngenes1">nGenes1</code></td>
<td>
<p>The number of causal genes.</p>
</td></tr>
<tr><td><code id="simPathAR1Snp_+3A_nsnps">nSNPs</code></td>
<td>
<p>A vector, length matched with total number of genes. Each elements of vector indicate the number of SNPs in the gene. Default is nSNPs = NULL, in this case the number of nSNPs randomly selected from nSNPlow to nSNPup.</p>
</td></tr>
<tr><td><code id="simPathAR1Snp_+3A_ncsnps">ncSNPs</code></td>
<td>
<p>A vector, length matched with total number of genes. Each elements of vector indicate the number of causal SNPs in the gene. Default is ncSNPs = NULL, in this case the number of ncSNPs are randomly selected from nSNP0.</p>
</td></tr>
<tr><td><code id="simPathAR1Snp_+3A_nsnplim">nSNPlim</code></td>
<td>
<p>If nSNPs = NULL, the number of SNPs in Gene randomly selected in nSNPlim.</p>
</td></tr>
<tr><td><code id="simPathAR1Snp_+3A_nsnp0">nSNP0</code></td>
<td>
<p>If ncSNPs = NULL, the number of causal SNPs in Gene randomly selected from nSNP0. Default is 1:3.</p>
</td></tr>
<tr><td><code id="simPathAR1Snp_+3A_lor">LOR</code></td>
<td>
<p>Association in log OR between a causal SNP and outcome.</p>
</td></tr>
<tr><td><code id="simPathAR1Snp_+3A_n">n</code></td>
<td>
<p># of cases (= # of controls).</p>
</td></tr>
<tr><td><code id="simPathAR1Snp_+3A_maflim">MAFlim</code></td>
<td>
<p>MAF's of the SNPs are drawn from Unif(MAFlim[1], MAFlim[2]).</p>
</td></tr>
<tr><td><code id="simPathAR1Snp_+3A_rholim">rholim</code></td>
<td>
<p>the SNPs in eahc gene are from a latent Normal
variable with a AR(rho) corr structure, rho's are drawn
from Unif(rholim[1], rholim[2]); the SNPs in diff genes are independant.</p>
</td></tr>
<tr><td><code id="simPathAR1Snp_+3A_p0">p0</code></td>
<td>
<p>background disease prevalence;i.e. intercept=log(p0/(1-p0)).</p>
</td></tr>
<tr><td><code id="simPathAR1Snp_+3A_noncausal">noncausal</code></td>
<td>
<p>exclude causal SNPs if TRUE, it is the simulation set up d in the paper(Pan et al 2015).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of the binary outcome Y (=0 or 1) and SNPs (=0, 1 or 2);
Y is a vector of length 2n; X is a matrix of 2n by nSNP.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+aSPUpath">aSPUpath</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Simulation set up A a) in the paper (Pan et al 2015)
## Not run:  simula &lt;- simPathAR1Snp(nGenes=20, nGenes1=1, nSNPlim=c(1, 20), nSNP0=1:3,
                           LOR=.2, rholim=c(0,0),
                           n=100, MAFlim=c(0.05, 0.4), p0=0.05) 
## End(Not run)


# Simulation set up A b) in the paper
#simulb &lt;- simPathAR1Snp(nGenes=20, nGenes1=1, nSNPlim=c(1, 100), nSNP0=1:3,
#                           LOR=.2, rholim=c(0,0),
#                           n=100, MAFlim=c(0.05, 0.4), p0=0.05)



</code></pre>

<hr>
<h2 id='someGs'>An example data for MTaSPUsSet test</h2><span id='topic+someGs'></span>

<h3>Description</h3>

<p>Genetic Investigation of ANthropometric Traits (GIANT) consortium
data contain P-values of 2.7 million SNPs with six anthropometric
traits that are well established to represent body size and shape:
height, weight, BMI, waist circumference (WC), hip
circumference(HIP). 
</p>
<p>We mapped SNPs on gene LCORL, RASA2, STK33 and RPGRIP1L. This subdata
contains P values, correlation matrices for demostrated example of
MTaSPUsSet test. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(someGs)
</code></pre>


<h3>References</h3>

<p>Shungin D, Winkler TW, Croteau-Chonka DC, Ferreira T, Locke AE, Magi R,
Strawbridge R, Pers TH, Fischer K, Justice AE, Workalemahu T, Wu JM, et
al. (2015). New genetic loci link adipose and insulin biology to body
fat distribution. Nature 518: 187-196.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(someGs)

## P-values for LCORL
someGs$LCORL[[1]]

## correlation matrix for LCORL
someGs$LCORL[[2]]

## P-values for RASA2
someGs$RASA2[[1]]

## correlation matrix for LCORL
someGs$RASA2[[2]]

# See vignettes for more details.

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
