<!DOCTYPE html><html><head><title>Help for package assignPOP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {assignPOP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#accuracy.kfold'><p>Estimate assignment accuracies of K-fold cross-validation results</p></a></li>
<li><a href='#accuracy.MC'><p>Estimate assignment accuracies of Monte-Carlo cross-validation results</p></a></li>
<li><a href='#accuracy.plot'><p>Make a boxplot (ggplot2 style) of assignment accuracy from cross-validation results</p></a></li>
<li><a href='#assign.kfold'><p>Population assignment test using K-fold cross-validation</p></a></li>
<li><a href='#assign.matrix'><p>Make an assignment maxtrix from cross-validation results</p></a></li>
<li><a href='#assign.MC'><p>Population assignment test using Monte-Carlo cross-validation</p></a></li>
<li><a href='#assign.X'><p>Perform a population assignment test on unknown individuals using known data</p></a></li>
<li><a href='#check.loci'><p>Check which loci frequently have high Fst across training sets</p></a></li>
<li><a href='#compile.data'><p>Compile genetic and other non-genetic data</p></a></li>
<li><a href='#membership.plot'><p>Make a membership probability plot using results from K-fold cross-validation (ggplot2 style)</p></a></li>
<li><a href='#read.Genepop'><p>Read GENEPOP format file</p></a></li>
<li><a href='#read.Structure'><p>Read Structure format file</p></a></li>
<li><a href='#reduce.allele'><p>Remove low variance alleles (dimensionality reduction)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Population Assignment using Genetic, Non-Genetic or Integrated
Data in a Machine Learning Framework</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Kuan-Yu (Alex) Chen [aut, cre], Elizabeth A. Marschall [aut], Michael
    G. Sovic [aut], Anthony C. Fries [aut], H. Lisle Gibbs [aut], Stuart A. Ludsin
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kuan-Yu (Alex) Chen &lt;alexkychen@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Use Monte-Carlo and K-fold cross-validation coupled with machine-
    learning classification algorithms to perform population assignment, with
    functionalities of evaluating discriminatory power of independent training
    samples, identifying informative loci, reducing data dimensionality for genomic
    data, integrating genetic and non-genetic data, and visualizing results.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/alexkychen/assignPOP">https://github.com/alexkychen/assignPOP</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.3.2)</td>
</tr>
<tr>
<td>Imports:</td>
<td>caret, doParallel, e1071, foreach, ggplot2, MASS, parallel,
randomForest, reshape2, stringr, tree, rlang,</td>
</tr>
<tr>
<td>Suggests:</td>
<td>gtable, iterators, klaR, stringi, knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-13 08:13:55 UTC; alexk</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-13 08:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='accuracy.kfold'>Estimate assignment accuracies of K-fold cross-validation results</h2><span id='topic+accuracy.kfold'></span>

<h3>Description</h3>

<p>This function allows you to estimate assignment accuracies of K-fold cross-validation results. The output results can be used to make assignment accuracy plots (use function accuracy.plot) and membership probability plot (use function membership.plot)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>accuracy.kfold(dir = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="accuracy.kfold_+3A_dir">dir</code></td>
<td>
<p>A character string to specify the folder that has your K-fold cross-validation results. A slash should be included at the end (e.g., dir=&quot;YourFolderName/&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function outputs the results in a text file (a table). It can return a data frame when a returning object is specified.
</p>

<hr>
<h2 id='accuracy.MC'>Estimate assignment accuracies of Monte-Carlo cross-validation results</h2><span id='topic+accuracy.MC'></span>

<h3>Description</h3>

<p>This function allows you to estimate assignment accuracies of Monte-Carlo cross-validation results. The output results can be used to make assignment accuracy plots (use function accuracy.plot).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>accuracy.MC(dir = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="accuracy.MC_+3A_dir">dir</code></td>
<td>
<p>A character string to specify the folder that has your Monte-Carlo cross-validation results. A slash should be included at the end (e.g., dir=&quot;YourFolderName/&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function outputs the results in a text file (a table). It can return a data frame when a returning object is specified.
</p>

<hr>
<h2 id='accuracy.plot'>Make a boxplot (ggplot2 style) of assignment accuracy from cross-validation results</h2><span id='topic+accuracy.plot'></span>

<h3>Description</h3>

<p>This functions allows you to make a boxplot of assignment accuracies estimated from Monte-Carlo or K-fold cross-validation results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>accuracy.plot(df, pop = "all")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="accuracy.plot_+3A_df">df</code></td>
<td>
<p>A dataframe of your assignment accuracy results. It could be the object returned from the function accuracy.MC() or accuracy.kfold() or a data frame imported to R via other functions (e.g., read.table(...)).</p>
</td></tr>
<tr><td><code id="accuracy.plot_+3A_pop">pop</code></td>
<td>
<p>Population names (one or multiple string characters) for making the plot. By default, it uses &quot;all&quot;, meaning overall assignment accuracies. It creates faceted plot with one population per panel, if multiple population names are given. The specified population name should match what you entered in read.genpop() earlier.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a boxplot plot using the ggplot2 library. Users can modified the plot (e.g., change color, text, etc.) using functions provided by ggplot2 library.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Your_df &lt;- read.table(system.file("extdata/Rate.txt", package="assignPOP"), header=TRUE)
accuracy.plot(Your_df, pop="all")
</code></pre>

<hr>
<h2 id='assign.kfold'>Population assignment test using K-fold cross-validation</h2><span id='topic+assign.kfold'></span>

<h3>Description</h3>

<p>This function employs K-fold cross-validation for assignment tests. The results help estimate membership probabilities of every individual. It accepts genetic-only [object returned from read.genpop() or reducel.allele()], integrated [object returned from compile.data()], or non-genetic [R data frame with header] data as input, and outputs results to text files. Several built-in options are provided. See below for more details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>assign.kfold(
  x,
  k.fold = c(3, 4, 5),
  train.loci = c(0.1, 0.25, 0.5, 1),
  loci.sample = "fst",
  dir = NULL,
  scaled = FALSE,
  pca.method = "mixed",
  pca.PCs = "kaiser-guttman",
  pca.loadings = F,
  model = "svm",
  svm.kernel = "linear",
  svm.cost = 1,
  ntree = 50,
  processors = 999,
  multiprocess = TRUE,
  skipQ = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="assign.kfold_+3A_x">x</code></td>
<td>
<p>An input object which should be the object (list) returned from the function read.genpop(), reduce.allele(), or compile.data(). It could also be a data frame (with column name) returned from read.csv() or read.table() if you're analyzing non-genetic data, such as morphormetrics, chemistry data. The non-genetic data frame should have sample ID in the first column and population label in the last column.</p>
</td></tr>
<tr><td><code id="assign.kfold_+3A_k.fold">k.fold</code></td>
<td>
<p>The number of groups to be divided for each population. Use a numeric vector to specify multiple sets of k-folds.</p>
</td></tr>
<tr><td><code id="assign.kfold_+3A_train.loci">train.loci</code></td>
<td>
<p>The proportion (float between 0 and 1) of loci to be used as training data. Use a numeric vector to specify multiple sets of training loci. This argument will be ignored if you're analyzing non-genetic data.</p>
</td></tr>
<tr><td><code id="assign.kfold_+3A_loci.sample">loci.sample</code></td>
<td>
<p>Locus sampling method, &quot;fst&quot; or &quot;random&quot;. If loci.sample=&quot;fst&quot; (default) and train.loci=0.1, it means that top 10 percent of high Fst loci will be sampled as training loci. On the other hand, if loci.sample=&quot;random&quot;, then random 10 percent of loci will be sampled as training loci. This argument will be ignored if you're analyzing non-genetic data.</p>
</td></tr>
<tr><td><code id="assign.kfold_+3A_dir">dir</code></td>
<td>
<p>A character string to specify the folder name for saving output files. A slash at the end must be included (e.g., dir=&quot;YourFolderName/&quot;). Otherwise, the files will be saved under your working directory.</p>
</td></tr>
<tr><td><code id="assign.kfold_+3A_scaled">scaled</code></td>
<td>
<p>A logical variable (TRUE or FALSE) to specify whether to center (make mean of each feature to 0) and scale (make standard deviation of each feature to 1) the entire dataset before performing PCA and cross-validation. Default is FALSE. As genetic data has converted to numeric data between 0 and 1, to scale or not to scale the genetic data should not be critical. However, it is recommended to set scaled=TRUE when integrated data contains various scales of features.</p>
</td></tr>
<tr><td><code id="assign.kfold_+3A_pca.method">pca.method</code></td>
<td>
<p>Either a character string (&quot;mixed&quot;, &quot;independent&quot;, or &quot;original&quot;) or logical variable (TRUE or FALSE) to specify how to perform PCA on non-genetic data (PCA is always performed on genetic data). The character strings are used when analyzing integrated (genetic plus non-genetic) data. If using &quot;mixed&quot; (default), PCA is perfromed across the genetic and non-genetic data, resulting in each PC summarizing mixed variations of genetic and non-genetic data. If using &quot;independent&quot;, PCA is independently performed on non-genetic data. Genetic PCs and non-genetic PCs are then used as new features. If using &quot;original&quot;, original non-genetic data and genetic PCs are used as features. The logical variable is used when analyzing non-genetic data alone. If TRUE, it performs PCA on the training data and applys the loadings to the test data. Scores of training and test data will be used as new features.</p>
</td></tr>
<tr><td><code id="assign.kfold_+3A_pca.pcs">pca.PCs</code></td>
<td>
<p>A criterion (&quot;Kaiser-Guttman&quot;,&quot;broken-stick&quot;, or numeric) to retain number of PCs. By default, it uses Kaiser-Guttman criterion that any PC has the eigenvalue greater than 1 will be retained as the new variable/feature. Users can set an integer to specify the number of PCs to be retained.</p>
</td></tr>
<tr><td><code id="assign.kfold_+3A_pca.loadings">pca.loadings</code></td>
<td>
<p>A logical variable (False or True) to determine whether it prints the loadings of training data to output text files. Default is False, if set True, the overall output files could be large.</p>
</td></tr>
<tr><td><code id="assign.kfold_+3A_model">model</code></td>
<td>
<p>A character string to specify which classifier to use for creating predictive models. The current options include &quot;lda&quot;, &quot;svm&quot;, &quot;naiveBayes&quot;, &quot;tree&quot;, and &quot;randomForest&quot;.</p>
</td></tr>
<tr><td><code id="assign.kfold_+3A_svm.kernel">svm.kernel</code></td>
<td>
<p>A character string to specify which kernel to be used when using &quot;svm&quot; classifier. Default is &quot;linear&quot;. Other options include &quot;polynomial&quot;, &quot;radial&quot;, and &quot;sigmoid&quot;. Look up R pacakge e1071 for more details about SVM, or see a guidance at https://www.csie.ntu.edu.tw/~cjlin/papers/guide/guide.pdf</p>
</td></tr>
<tr><td><code id="assign.kfold_+3A_svm.cost">svm.cost</code></td>
<td>
<p>A number to specify the cost for &quot;svm&quot; method.</p>
</td></tr>
<tr><td><code id="assign.kfold_+3A_ntree">ntree</code></td>
<td>
<p>A integer to specify how many trees to build when using &quot;randomForest&quot; method.</p>
</td></tr>
<tr><td><code id="assign.kfold_+3A_processors">processors</code></td>
<td>
<p>The number of processors to be used for parallel running. By default, it uses N-1 processors in your computer.</p>
</td></tr>
<tr><td><code id="assign.kfold_+3A_multiprocess">multiprocess</code></td>
<td>
<p>A logical variable to determine whether using multiprocess. Default is TRUE. If set FALSE, it will only use single core to run the program.</p>
</td></tr>
<tr><td><code id="assign.kfold_+3A_skipq">skipQ</code></td>
<td>
<p>A logical variable to determine whether prompting interactive dialogue when analyzing non-genetic data. If set TRUE, default data type and original values of non-genetic data will be used.</p>
</td></tr>
<tr><td><code id="assign.kfold_+3A_...">...</code></td>
<td>
<p>Other arguments that could be potentially used for various models</p>
</td></tr>
</table>


<h3>Value</h3>

<p>You don't need to specify a name for the returned object when using this function. It automatically outputs results in text files to your designated folder.
</p>

<hr>
<h2 id='assign.matrix'>Make an assignment maxtrix from cross-validation results</h2><span id='topic+assign.matrix'></span>

<h3>Description</h3>

<p>This function generates a pairwise assignment matrix with mean and variation of assignment accuracies estimated across all assignment tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>assign.matrix(
  dir = NULL,
  train.loci = "all",
  train.inds = "all",
  k.fold = "all"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="assign.matrix_+3A_dir">dir</code></td>
<td>
<p>A character string to specify the folder that has your cross-validation assignment results.</p>
</td></tr>
<tr><td><code id="assign.matrix_+3A_train.loci">train.loci</code></td>
<td>
<p>Choose your proportions of training loci used in Monte-Carlo or K-fold cross-validation. Default is &quot;all&quot;.</p>
</td></tr>
<tr><td><code id="assign.matrix_+3A_train.inds">train.inds</code></td>
<td>
<p>Choose your numbers or proportions of training individuals used in Monte-Carlo cross-validation. Default is &quot;all&quot;.</p>
</td></tr>
<tr><td><code id="assign.matrix_+3A_k.fold">k.fold</code></td>
<td>
<p>Choose the k fold values used in K-fold cross-validation. Default is &quot;all&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a matrix in R console as well as a file named &quot;assignment_matrix.txt&quot; in the folder.
</p>

<hr>
<h2 id='assign.MC'>Population assignment test using Monte-Carlo cross-validation</h2><span id='topic+assign.MC'></span>

<h3>Description</h3>

<p>This function employs Monte-Carlo cross-validation for assignment tests. The results help evaluate if known data set has sufficient discriminatory power. It accepts genetic-only [object returned from read.genpop() or reducel.allele()], integrated [object returned from compile.data()], or non-genetic [R data frame with header] data as input, and outputs results to text files. Several built-in options are provided. See below for more details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>assign.MC(
  x,
  train.inds = c(0.5, 0.7, 0.9),
  train.loci = c(0.1, 0.25, 0.5, 1),
  loci.sample = "fst",
  iterations = 20,
  dir = NULL,
  scaled = FALSE,
  pca.method = "mixed",
  pca.PCs = "kaiser-guttman",
  pca.loadings = F,
  model = "svm",
  svm.kernel = "linear",
  svm.cost = 1,
  ntree = 50,
  multiprocess = TRUE,
  processors = 999,
  skipQ = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="assign.MC_+3A_x">x</code></td>
<td>
<p>An input object which should be the object (list) returned from the function read.genpop(), reduce.allele(), or compile.data(). It could also be a data frame (with column name) returned from read.csv() or read.table() if you're analyzing non-genetic data, such as morphormetrics, chemistry data. The non-genetic data frame should have sample ID in the first column and population label in the last column.</p>
</td></tr>
<tr><td><code id="assign.MC_+3A_train.inds">train.inds</code></td>
<td>
<p>The number (integer greater than 1) or proportion (float between 0 and 1) of individuals (observations) from each population to be used as training data. Use a numeric vector to specify multiple sets of training individuals. No mixture of integer and float in a vector.</p>
</td></tr>
<tr><td><code id="assign.MC_+3A_train.loci">train.loci</code></td>
<td>
<p>The proportion (float between 0 and 1) of loci to be used as training data. Use a numeric vector to specify multiple sets of training loci. This argument will be ignored if you're analyzing non-genetic data.</p>
</td></tr>
<tr><td><code id="assign.MC_+3A_loci.sample">loci.sample</code></td>
<td>
<p>Locus sampling method, &quot;fst&quot; or &quot;random&quot;. If loci.sample=&quot;fst&quot; (default) and train.loci=0.1, it means that top 10 percent of high Fst loci will be sampled as training loci. On the other hand, if loci.sample=&quot;random&quot;, then random 10 percent of loci will be sampled as training loci. This argument will be ignored if you're analyzing non-genetic data.</p>
</td></tr>
<tr><td><code id="assign.MC_+3A_iterations">iterations</code></td>
<td>
<p>Resampling times (an integer) for each combination of training individuals and loci.</p>
</td></tr>
<tr><td><code id="assign.MC_+3A_dir">dir</code></td>
<td>
<p>A character string to specify the folder name for saving output files. A slash at the end must be included (e.g., dir=&quot;YourFolderName/&quot;). Otherwise, the files will be saved under your working directory.</p>
</td></tr>
<tr><td><code id="assign.MC_+3A_scaled">scaled</code></td>
<td>
<p>A logical variable (TRUE or FALSE) to specify whether to center (make mean of each feature to 0) and scale (make standard deviation of each feature to 1) the entire dataset before performing PCA and cross-validation. Default is FALSE. As genetic data has converted to numeric data between 0 and 1, to scale or not to scale the genetic data should not be critical. However, it is recommended to set scaled=TRUE when integrated data contains various scales of features.</p>
</td></tr>
<tr><td><code id="assign.MC_+3A_pca.method">pca.method</code></td>
<td>
<p>Either a character string (&quot;mixed&quot;, &quot;independent&quot;, or &quot;original&quot;) or logical variable (TRUE or FALSE) to specify how to perform PCA on non-genetic data (PCA is always performed on genetic data). The character strings are used when analyzing integrated (genetic plus non-genetic) data. If using &quot;mixed&quot; (default), PCA is perfromed across the genetic and non-genetic data, resulting in each PC summarizing mixed variations of genetic and non-genetic data. If using &quot;independent&quot;, PCA is independently performed on non-genetic data. Genetic PCs and non-genetic PCs are then used as new features. If using &quot;original&quot;, original non-genetic data and genetic PCs are used as features. The logical variable is used when analyzing non-genetic data.If TRUE, it performs PCA on the training data and applys the loadings to the test data. Scores of training and test data will be used as new features.</p>
</td></tr>
<tr><td><code id="assign.MC_+3A_pca.pcs">pca.PCs</code></td>
<td>
<p>A criterion (&quot;Kaiser-Guttman&quot;,&quot;broken-stick&quot;, or numeric) to retain number of PCs. By default, it uses Kaiser-Guttman criterion that any PC has the eigenvalue greater than 1 will be retained as the new variable/feature. Users can set an integer to specify the number of PCs to be retained.</p>
</td></tr>
<tr><td><code id="assign.MC_+3A_pca.loadings">pca.loadings</code></td>
<td>
<p>A logical variable (TRUE or FALSE) to determine whether to output the loadings of training data to text files. Default is FALSE. Just a heads-up, the output files could take some storage space, if set TRUE.</p>
</td></tr>
<tr><td><code id="assign.MC_+3A_model">model</code></td>
<td>
<p>A character string to specify which classifier to use for creating predictive models. The current options include &quot;lda&quot;, &quot;svm&quot;, &quot;naiveBayes&quot;, &quot;tree&quot;, and &quot;randomForest&quot;. Default is &quot;svm&quot;(support vector machine).</p>
</td></tr>
<tr><td><code id="assign.MC_+3A_svm.kernel">svm.kernel</code></td>
<td>
<p>A character string to specify which kernel to be used when using &quot;svm&quot; classifier. Default is &quot;linear&quot;. Other options include &quot;polynomial&quot;, &quot;radial&quot;, and &quot;sigmoid&quot;. Look up R pacakge e1071 for more details about SVM, or see a guidance at https://www.csie.ntu.edu.tw/~cjlin/papers/guide/guide.pdf</p>
</td></tr>
<tr><td><code id="assign.MC_+3A_svm.cost">svm.cost</code></td>
<td>
<p>A number to specify the cost for &quot;svm&quot; method.</p>
</td></tr>
<tr><td><code id="assign.MC_+3A_ntree">ntree</code></td>
<td>
<p>A integer to specify how many trees to build when using &quot;randomForest&quot; method.</p>
</td></tr>
<tr><td><code id="assign.MC_+3A_multiprocess">multiprocess</code></td>
<td>
<p>A logical variable to determine whether using multiprocess. Default is TRUE. If set FALSE, it will only use single core to run the program.</p>
</td></tr>
<tr><td><code id="assign.MC_+3A_processors">processors</code></td>
<td>
<p>The number of processors to be used for parallel running. By default, it uses N-1 processors in your computer.</p>
</td></tr>
<tr><td><code id="assign.MC_+3A_skipq">skipQ</code></td>
<td>
<p>A logical variable to determine whether prompting interactive dialogue when analyzing non-genetic data. If set TRUE, default data type and original values of non-genetic data will be used.</p>
</td></tr>
<tr><td><code id="assign.MC_+3A_...">...</code></td>
<td>
<p>Other arguments that could be potentially used for various models</p>
</td></tr>
</table>


<h3>Value</h3>

<p>You don't need to specify a name for the returned object when using this function. It automatically outputs results in text files to your designated folder.
</p>

<hr>
<h2 id='assign.X'>Perform a population assignment test on unknown individuals using known data</h2><span id='topic+assign.X'></span>

<h3>Description</h3>

<p>This function assigns unknown individuals to possible source populations based on known individuals and genetic or non-genetic or integrated data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>assign.X(
  x1,
  x2,
  dir = NULL,
  common = T,
  scaled = F,
  pca.method = "mixed",
  pca.PCs = "kaiser-guttman",
  pca.loadings = F,
  model = "svm",
  svm.kernel = "linear",
  svm.cost = 1,
  ntree = 50,
  mplot = T,
  skipQ = F,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="assign.X_+3A_x1">x1</code></td>
<td>
<p>An input object containing data from known individuals for building predictive models. It could be a list object returned from the function read.genpop(), reduce.allele() or compile.data(). Or, it could be a data frame containing non-genetic data returned from read.csv() or read.table().</p>
</td></tr>
<tr><td><code id="assign.X_+3A_x2">x2</code></td>
<td>
<p>An input object containing data from unknown individuals to be predicted. It could be a list object returned from read.genpop(), reduce.allele(), or compile.data(). Or, it could be a data frame containing non-genetic data returned from read.csv() or read.table(). The x1 and x2 should be the same type (both are either lists or data frames).</p>
</td></tr>
<tr><td><code id="assign.X_+3A_dir">dir</code></td>
<td>
<p>A character string to specify the folder name for saving output files. A slash at the end must be included (e.g., dir=&quot;YourFolderName/&quot;). Otherwise, the files will be saved under your working directory.</p>
</td></tr>
<tr><td><code id="assign.X_+3A_common">common</code></td>
<td>
<p>A logical variable (TRUE or FALSE) to specify whether exclusively using features, the name of which is in common, between known and unknown data sets. Default is TRUE. If it is FALSE, it will stop performing analysis when inconsistent feature names were found.</p>
</td></tr>
<tr><td><code id="assign.X_+3A_scaled">scaled</code></td>
<td>
<p>A logical variable (TRUE or FALSE) to specify whether to center (make mean of each feature to 0) and scale (make standard deviation of each feature to 1) the dataset before performing PCA and cross-validation. Default is FALSE. As genetic data has converted to numeric data between 0 and 1, to scale or not to scale the genetic data should not be critical. However, it is recommended to set scaled=TRUE when integrated data contains various scales of features.</p>
</td></tr>
<tr><td><code id="assign.X_+3A_pca.method">pca.method</code></td>
<td>
<p>Either a character string (&quot;mixed&quot;, &quot;independent&quot;, or &quot;original&quot;) or logical variable (TRUE or FALSE) to specify how to perform PCA on non-genetic data (PCA is always performed on genetic data). The character strings are used when analyzing integrated (genetic plus non-genetic) data. If using &quot;mixed&quot; (default), PCA is perfromed across the genetic and non-genetic data, resulting in each PC summarizing mixed variations of genetic and non-genetic data. If using &quot;independent&quot;, PCA is independently performed on non-genetic data. Genetic PCs and non-genetic PCs are then used as new features. If using &quot;original&quot;, original non-genetic data and genetic PCs are used as features. The logical variable is used when analyzing non-genetic data.If TRUE, it performs PCA on the training data and applys the loadings to the test data. Scores of training and test data will be used as new features.</p>
</td></tr>
<tr><td><code id="assign.X_+3A_pca.pcs">pca.PCs</code></td>
<td>
<p>A criterion (&quot;Kaiser-Guttman&quot;,&quot;broken-stick&quot;, or numeric) to retain number of PCs. By default, it uses Kaiser-Guttman criterion that any PC has the eigenvalue greater than 1 will be retained as the new variable/feature. Users can set an integer to specify the number of PCs to be retained.</p>
</td></tr>
<tr><td><code id="assign.X_+3A_pca.loadings">pca.loadings</code></td>
<td>
<p>A logical variable (TRUE or FALSE) to determine whether to output the loadings of training data to text files. Default is FALSE. Just a heads-up, the output files could take some storage space, if set TRUE.</p>
</td></tr>
<tr><td><code id="assign.X_+3A_model">model</code></td>
<td>
<p>A character string to specify which classifier to use for creating predictive models. The current options include &quot;lda&quot;, &quot;svm&quot;, &quot;naiveBayes&quot;, &quot;tree&quot;, and &quot;randomForest&quot;. Default is &quot;svm&quot;(support vector machine).</p>
</td></tr>
<tr><td><code id="assign.X_+3A_svm.kernel">svm.kernel</code></td>
<td>
<p>A character string to specify which kernel to be used when using &quot;svm&quot; classifier. Default is &quot;linear&quot;. Other options include &quot;polynomial&quot;, &quot;radial&quot;, and &quot;sigmoid&quot;. Look up R pacakge e1071 for more details about SVM, or see a guidance at https://www.csie.ntu.edu.tw/~cjlin/papers/guide/guide.pdf</p>
</td></tr>
<tr><td><code id="assign.X_+3A_svm.cost">svm.cost</code></td>
<td>
<p>A number to specify the cost for &quot;svm&quot; method.</p>
</td></tr>
<tr><td><code id="assign.X_+3A_ntree">ntree</code></td>
<td>
<p>A integer to specify how many trees to build when using &quot;randomForest&quot; method.</p>
</td></tr>
<tr><td><code id="assign.X_+3A_mplot">mplot</code></td>
<td>
<p>A logical variable (TRUE or FALSE) to specify whether making a membership probability plot right after the assignment test is done. Default is TRUE.</p>
</td></tr>
<tr><td><code id="assign.X_+3A_skipq">skipQ</code></td>
<td>
<p>A logical variable (TRUE or FALSE) to skip data type checking on non-genetic data. Default is FALSE and will prompt questions to confirm data type. If it is TRUE, it will skip the confirmation and use data type by default (integer and float will be numeric data).</p>
</td></tr>
<tr><td><code id="assign.X_+3A_...">...</code></td>
<td>
<p>Other arguments that could be potentially used for various models</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function outputs assignment results and other analytical information in text files that will be saved under your designated folder. It also outputs a membership probability plot, if permitted.
</p>

<hr>
<h2 id='check.loci'>Check which loci frequently have high Fst across training sets</h2><span id='topic+check.loci'></span>

<h3>Description</h3>

<p>This function reads through training locus file for each assignment test and counts the frequency of those loci and outputs the results in a text file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check.loci(dir = NULL, top.loci = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check.loci_+3A_dir">dir</code></td>
<td>
<p>A character string to specify the folder with your cross-validation results. A slash should be entered at the end.</p>
</td></tr>
<tr><td><code id="check.loci_+3A_top.loci">top.loci</code></td>
<td>
<p>An integer to specify how many top informative loci to output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function output the results in a text file. It includes the top N informative loci in N rows, and each row has a list of loci sorted by its occurrence.
</p>

<hr>
<h2 id='compile.data'>Compile genetic and other non-genetic data</h2><span id='topic+compile.data'></span>

<h3>Description</h3>

<p>This function allows you to combine genetic and other non-genetic data, such as morphometrics, of the observations for assignment tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compile.data(x, add.x, method = "common", skipQ = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compile.data_+3A_x">x</code></td>
<td>
<p>A returned object (list) from the function read.genpop() or reduce.allele().</p>
</td></tr>
<tr><td><code id="compile.data_+3A_add.x">add.x</code></td>
<td>
<p>A file containing non-genetic data that has sample ID in the first column. The sample ID must be the same as your GENEPOP file.</p>
</td></tr>
<tr><td><code id="compile.data_+3A_method">method</code></td>
<td>
<p>A method to match sample ID between genetic and non-genetic data. The &quot;common&quot; method only concatenate the data that has sample ID in both files. If an individual only exists in one of the files, this individual will be discarded.</p>
</td></tr>
<tr><td><code id="compile.data_+3A_skipq">skipQ</code></td>
<td>
<p>A logical variable to determine whether prompting interactive dialogue. If set TRUE, input data type will be recognized as default type and not be verified by the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a new object (list) that comprises 5 items. [[1]] data matrix including genetic and non-genetic data, [[2]] a sample ID vector, [[3]] a locus name vector, [[4]] a vector of non-genetic variable names, and [[5]] the number of non-genetic variables.
</p>

<hr>
<h2 id='membership.plot'>Make a membership probability plot using results from K-fold cross-validation (ggplot2 style)</h2><span id='topic+membership.plot'></span>

<h3>Description</h3>

<p>This function allows you to make a membership probability plot (stacked-bar plot) using results estimated from K-fold cross-validation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>membership.plot(
  dir = NULL,
  style = NULL,
  non.genetic = FALSE,
  plot.k = NULL,
  plot.loci = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="membership.plot_+3A_dir">dir</code></td>
<td>
<p>A character string to specify the folder that has your K-fold cross-validation assignment results. A slash should be entered at the end.</p>
</td></tr>
<tr><td><code id="membership.plot_+3A_style">style</code></td>
<td>
<p>An option for output style. If style=1, it creates the plot which individuals on the x-axis are in random order. If style=2, individuals are sorted by probabilities within each population. If style=3, individuals of different folds are in seperate plots. If style=4, individuals are separated by fold and sorted by probability.</p>
</td></tr>
<tr><td><code id="membership.plot_+3A_non.genetic">non.genetic</code></td>
<td>
<p>A logical variable to specify if data are non-genetic. Set it TRUE if you're analyzing non-genetic alone.</p>
</td></tr>
<tr><td><code id="membership.plot_+3A_plot.k">plot.k</code></td>
<td>
<p>A number to specify which K of the data set should be plotted. If not given, it will prompt the question.</p>
</td></tr>
<tr><td><code id="membership.plot_+3A_plot.loci">plot.loci</code></td>
<td>
<p>The proportion of training loci used in your K-fold cross-validation analysis. Specify one of the numbers here to skip question prompt.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a stacked-bar plot using the ggplot2 library. Users can modified the plot (e.g., change color, text, etc.) using functions provided by ggplot2 library.
</p>

<hr>
<h2 id='read.Genepop'>Read GENEPOP format file</h2><span id='topic+read.Genepop'></span>

<h3>Description</h3>

<p>This function allows you to import a GENEPOP format file into R. Population names can be specified in the argument. See http://genepop.curtin.edu.au/help_input.html for details about GENEPOP format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.Genepop(x, pop.names = NULL, haploid = FALSE, pos = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.Genepop_+3A_x">x</code></td>
<td>
<p>GENEPOP file or path to the file. The filename extension (e.g., .txt) should be included.</p>
</td></tr>
<tr><td><code id="read.Genepop_+3A_pop.names">pop.names</code></td>
<td>
<p>A character string vector for population names. The order of the name should be the same with the order (top to down) in your GENEPOP file.</p>
</td></tr>
<tr><td><code id="read.Genepop_+3A_haploid">haploid</code></td>
<td>
<p>A logical variable (TRUE or FALSE) to specify whether your dataset is haploid data. Default is FALSE.</p>
</td></tr>
<tr><td><code id="read.Genepop_+3A_pos">pos</code></td>
<td>
<p>A parameter for program development use; users can ignore it.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a list comprising three elements. 1. YOU_NAME_IT$DataMatrix: A matrix of genetic data with a population name label ($popNameVector) in the last column. 2. YOU_NAME_IT$SampleID: A vector of sample ID. 3. YOU_NAME_IT$LocusName: A vector of locus name.
</p>


<h3>References</h3>

<p>Rousset, F. 2008. Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. Mol. Ecol. Resources 8: 103-106
</p>


<h3>Examples</h3>

<pre><code class='language-R'># infile &lt;- read.Genepop("Your_Genepop_File.txt", pop.names=c("pop_A", "pop_B", "pop_C"))
</code></pre>

<hr>
<h2 id='read.Structure'>Read Structure format file</h2><span id='topic+read.Structure'></span>

<h3>Description</h3>

<p>This function allows you to import a STRUCTURE format file into R. The first row should be locus name (either with or withour column names for sample ID and population label); the first column should be sample ID; the second column should be population label; the rest are genotype. Use &quot;-9&quot; for missing alleles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.Structure(x, ploidy = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.Structure_+3A_x">x</code></td>
<td>
<p>STRUCTURE file or path to the file. The filename extension (e.g., .txt) should be included.</p>
</td></tr>
<tr><td><code id="read.Structure_+3A_ploidy">ploidy</code></td>
<td>
<p>An integer of 1, 2, 3, or 4, to indicate haploid, diploid, triploid, or tetraploid data. Default is 2 (diploid).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a list comprising three elements. 1. YOU_NAME_IT$DataMatrix: A matrix of genetic data with a population name label ($popNameVector) in the last column. 2. YOU_NAME_IT$SampleID: A vector of sample ID. 3. YOU_NAME_IT$LocusName: A vector of locus name.
</p>


<h3>References</h3>

<p>Pritchard, J.K., Stephens, M. and Donnelly, P., 2000. Inference of population structure using multilocus genotype data. Genetics, 155(2), pp.945-959.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># infile &lt;- read.Structure("Your_Structure_File.txt")
</code></pre>

<hr>
<h2 id='reduce.allele'>Remove low variance alleles (dimensionality reduction)</h2><span id='topic+reduce.allele'></span>

<h3>Description</h3>

<p>This function helps remove alleles that have low variance in the data set such that it can speed up further analyses for a large data set (e.g., &gt; 10K SNPs).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reduce.allele(x, p = 0.95)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reduce.allele_+3A_x">x</code></td>
<td>
<p>A returned object (a list) from the function read.genpop().</p>
</td></tr>
<tr><td><code id="reduce.allele_+3A_p">p</code></td>
<td>
<p>A threshold of variance for the alleles to be removed. For example, if p = 0.95 (default setting), an allele occupied more than 95 percents across all the samples will be removed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function return the same object as the function read.genpop() except that the number of columns in the matrix [[1]] is reduced and so is the locus name [[3]].
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
