<!DOCTYPE html><html><head><title>Help for package AssocAFC</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {AssocAFC}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#afcSKAT'>
<p>Allele Frequency Comparison Sequence Kernel Association Test</p></a></li>
<li><a href='#AssocAFC'>
<p>Allele Frequency Comparison</p></a></li>
<li><a href='#cor.afc'>
<p>AFC Correlation Example</p></a></li>
<li><a href='#geno.afc'>
<p>AFC Genotype Example</p></a></li>
<li><a href='#kin.afc'>
<p>AFC Kinship Example</p></a></li>
<li><a href='#maf.afc'>
<p>AFC MAF Example</p></a></li>
<li><a href='#phenotype.afc'>
<p>AFC Phenotype Example</p></a></li>
<li><a href='#Wcorrected'>
<p>Corrected Chi Squared Test</p></a></li>
<li><a href='#weights.afc'>
<p>AFC Weights Example</p></a></li>
<li><a href='#Wqls'>
<p>W Quasi-Likelihood Score</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Allele Frequency Comparison</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Author:</td>
<td>Khalid B. Kunji [aut, cre],
  Mohamad Saad [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Khalid B. Kunji &lt;kkunji@hbku.edu.qa&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Description:</td>
<td>When doing association analysis one does not always have the 
   genotypes for the control population.  In such cases it may be 
   necessary to fall back on frequency based tests using well known sources 
   for the frequencies in the control population, for instance, from the 
   1000 Genomes Project.  The Allele Frequency Comparison ('AssocAFC') package 
   performs multiple rare variant association analyses in both population and 
   family-based GWAS (Genome-Wide Association Study) designs. It includes 
   three score tests that are based on the difference of the sum of allele 
   frequencies between cases and controls.  Two of these tests, Wcorrected() 
   and Wqls(), are collapsing-based tests and suffer from having protective 
   and risk variants. The third test, afcSKAT(), is a score test that 
   overcomes the mix of SNP (Single-Nucleotide Polymorphism) effect 
   directions. For more details see 
   Saad M and Wijsman EM (2017) &lt;<a href="https://doi.org/10.1093%2Fbib%2Fbbx107">doi:10.1093/bib/bbx107</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://www.r-project.org">https://www.r-project.org</a>, <a href="https://doi.org/10.1093/bib/bbx107">https://doi.org/10.1093/bib/bbx107</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>CompQuadForm</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.0)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>KeepSource:</td>
<td>TRUE</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-05-05 09:10:46 UTC; kunji</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-05-05 09:34:22 UTC</td>
</tr>
</table>
<hr>
<h2 id='afcSKAT'>
Allele Frequency Comparison Sequence Kernel Association Test
</h2><span id='topic+afcSKAT'></span>

<h3>Description</h3>

<p>Sequence Kernel Association Test, afcSKAT(), for multiple rare variant association
test using the difference of sum of minor allele frequencies between cases and controls.
This test handles related individuals, unrelated individuals, or both. Plus, it is
robust againt the inclusion of both protective and risk variants in the same model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>afcSKAT(MAF, Pheno, Kin, Correlation, Weights)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="afcSKAT_+3A_maf">MAF</code></td>
<td>

<p>matrix (#Snps * 2): First column contains Minor Allele Frequency (MAF) in cases; Second column contains MAF in controls.
</p>
</td></tr>
<tr><td><code id="afcSKAT_+3A_pheno">Pheno</code></td>
<td>

<p>matrix (#subjects * 1): this one-column matrix contains 0's amd 1's: 1 for cases and 0 for controls. No missing values are allowed.
</p>
</td></tr>
<tr><td><code id="afcSKAT_+3A_kin">Kin</code></td>
<td>

<p>The kinship matrix (#subjects * #subjects): the subjects must be ordered as the Pheno variable.
</p>
</td></tr>
<tr><td><code id="afcSKAT_+3A_correlation">Correlation</code></td>
<td>

<p>Correlation matrix between SNPs (#Snps * #Snps). The user should calculate this matrix beforehand. Either based on own genotype data (in cases, controls, or both) or based on public databases (e.g., 1000 Genomes Projects, ESP, etc.). NA values are not allowed. They have to be replaced by zeros.
</p>
</td></tr>
<tr><td><code id="afcSKAT_+3A_weights">Weights</code></td>
<td>

<p>The weights values that can be used to up-weight or down-weight SNPs. This size of this
vector is the number of Snps by 1. By default, the weights are 1 for all Snps.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector with two values is returned, the 1st is the p-value calculated by the Satterwaite method
and the second is calculated via Davies.
</p>


<h3>References</h3>

<p>Saad M and Wijsman EM, Association score testing for rare variants and binary traits in family
data with shared controls, Briefings in Bioinformatics, 2017.
Schaid DJ , McDonnell SK , Sinnwell JP , et al.   Multiple genetic variant association testing
by collapsing and kernel methods with pedigree or population structured data.
Genet Epidemiol 2013 ;37 :409 –18.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+AssocAFC">AssocAFC</a></code>
<code><a href="#topic+Wqls">Wqls</a></code>
<code><a href="#topic+Wcorrected">Wcorrected</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
P_afcSKAT_Satterwaite &lt;- vector("numeric")
P_afcSKAT_Davies &lt;- vector("numeric")
#This data corresponds to what is used in the 1st iteration with the raw data
data("maf.afc")
data("phenotype.afc")
data("kin.afc")
data("cor.afc")
data("weights.afc")
SKAT &lt;- afcSKAT(MAF = maf.afc , Pheno = phenotype.afc, Kin = kin.afc , Correlation=cor.afc,
                Weights = weights.afc)
P_afcSKAT_Satterwaite &lt;- c(P_afcSKAT_Satterwaite, SKAT[1])
P_afcSKAT_Davies &lt;- c(P_afcSKAT_Davies, SKAT[2])
print(P_afcSKAT_Satterwaite)
print(P_afcSKAT_Davies)

## Not run: 
#This example shows processing the raw data and uses kinship2,
#which AFC does not depend on

library(kinship2)
library(CompQuadForm)

P_afcSKAT_Satterwaite &lt;- vector("numeric")
P_afcSKAT_Davies &lt;- vector("numeric")

for (j in 1:10)
{
  geno.afc &lt;- read.table(system.file("extdata", "Additive_Genotyped_Truncated.txt",
                         package = "AFC"), header = TRUE)
  geno.afc[ , "IID"] &lt;- paste(geno.afc[ , "FID"]  , geno.afc[ , "IID"]  ,sep=".")
  geno.afc[geno.afc[,"FA"]!=0 , "FA"] &lt;- paste(geno.afc[geno.afc[,"FA"]!=0 , "FID"],
                                                geno.afc[geno.afc[,"FA"]!=0 , "FA"]  ,sep=".")
  geno.afc[geno.afc[,"FA"]!=0 , "MO"] &lt;- paste(geno.afc[geno.afc[,"FA"]!=0 , "FID"],
                                                geno.afc[geno.afc[,"FA"]!=0 , "MO"]  ,sep=".")
  Kinship &lt;- makekinship(geno.afc$FID , geno.afc$IID , geno.afc$FA, geno.afc$MO)
  kin.afc &lt;- as.matrix(Kinship)
  pheno.afc &lt;- read.table(system.file("extdata", "Phenotype", package = "AFC"))
  phenotype.afc &lt;- matrix(pheno.afc[,j],nc=1,nr=nrow(pheno.afc))
  geno.afc &lt;- geno.afc[,7:ncol(geno.afc)]
  Na &lt;- nrow(pheno.afc[pheno.afc[,j]==1,])
  Nu &lt;- nrow(pheno.afc[pheno.afc[,j]==0,])
  N &lt;- Nu + Na
  maf.afc &lt;- matrix(NA , nr=ncol(geno.afc) , nc=2)
  maf.afc[,1] &lt;- colMeans(geno.afc[phenotype.afc==1,])/2;
  maf.afc[,2] &lt;- colMeans(geno.afc[phenotype.afc==0,])/2;
  P  &lt;- (maf.afc[,1]*Na + maf.afc[,2]*Nu)/N
  Set &lt;- which(P&lt;0.05)
  maf.afc &lt;- maf.afc[c(Set),]
  cor.afc &lt;- cor(geno.afc[,c(Set)])
  cor.afc[is.na(cor.afc)] &lt;- 0
  geno.afc &lt;- as.matrix(geno.afc[,Set])

  weights.afc &lt;- matrix(1/(maf.afc[,2]+1),nc=1,nr=length(Set))
  SKAT &lt;- afcSKAT(MAF = maf.afc , Pheno = phenotype.afc, Kin = kin.afc , Correlation=cor.afc,
                  Weights = weights.afc)
  P_afcSKAT_Satterwaite &lt;- c(P_afcSKAT_Satterwaite, SKAT[1])
  P_afcSKAT_Davies &lt;- c(P_afcSKAT_Davies, SKAT[2])
}
print(P_afcSKAT_Satterwaite)
print(P_afcSKAT_Davies)

## End(Not run)

## The function is currently defined as
function(MAF, Pheno, Kin, Correlation, Weights)
{
  Na     &lt;- length(Pheno[Pheno[,1]==1,])
  Nu     &lt;- length(Pheno[Pheno[,1]==0,])
  N      &lt;- Na + Nu

  # The three following lines: prepare the phenotype variables
  OneN  &lt;- matrix(1, ncol=1, nrow = N)
  Y  &lt;- Pheno
  OneHat &lt;- matrix( Na/N, ncol=1 , nrow=N)

  # Estimate MAF in all subjects
  P  &lt;- (MAF[,1]*Na + MAF[,2]*Nu)/N
  if (is.null(Weights))
  {
    # Variance of SNPs (2p(1-p))
    VarSnps &lt;- sqrt(P*(1-P))
  } else
  {
    # Variance of SNPs (2p(1-p)) accounting for the prespecified Snp weights
    VarSnps &lt;- Weights*sqrt(P*(1-P))
  }
  VarSnps &lt;- matrix(VarSnps,ncol=1)
  cz &lt;- 2* sum((Y - OneHat) %*% t(Y - OneHat) * Kin )
  Vz &lt;- cz * VarSnps %*% t(VarSnps)*Correlation
  if (is.null(Weights))
  {
  	# Quadratic form without Weights
    Q &lt;-  4*Na^2*(sum ( (MAF[,1] - P)^2 ))
  } else
  {
  	# Quadratic form with Weights
    Q &lt;-  4*Na^2*(sum ( Weights*(MAF[,1] - P)^2 ))
  }
  # Satterwaite approximation (1)
  E_Q &lt;- sum(diag(Vz))
  # Satterwaite approximation (2)
  V_Q &lt;- 2* sum( diag (Vz %*% Vz))
  # Satterwaite approximation (3)
  Delta &lt;- V_Q/(2*E_Q)
  # Satterwaite approximation (4)
  df&lt;- 2*E_Q^2/V_Q
  # Satterwaite approximation (5)
  Qscaled &lt;- Q / Delta
  # Satterwaite approximation (6)
  Pvalue_Sat &lt;- 1-pchisq(Qscaled, df)

	# Davies approximation (1)
  eig &lt;- eigen(Vz, symmetric = T, only.values = T)
  # Davies approximation (2)
  evals &lt;- eig$values[eig$values &gt; 1e-06 * eig$values[1]]
  # Davies approximation (3)
  Pvalue_Dav &lt;-davies(Q, evals, acc = 1e-5)$Qq
  out &lt;- t(data.frame(c(Pvalue_Sat,Pvalue_Dav)))
  colnames(out)&lt;- c("Satterwaite","Davies")
  rownames(out) &lt;- "Pvalue"
  return(out)
}
</code></pre>

<hr>
<h2 id='AssocAFC'>
Allele Frequency Comparison
</h2><span id='topic+AssocAFC'></span>

<h3>Description</h3>

<p>When doing association analysis one does not always have the 
genotypes for the control population.  In such cases it may be 
necessary to fall back on frequency based tests using well known sources 
for the frequencies in the control population, for instance, from the 
1000 Genomes Project.  The Allele Frequency Comparison ('AssocAFC') package 
performs multiple rare variant association analyses in both population and 
family-based GWAS (Genome-Wide Association Study) designs. It includes 
three score tests that are based on the difference of the sum of allele 
frequencies between cases and controls.  Two of these tests, Wcorrected() 
and Wqls(), are collapsing-based tests and suffer from having protective 
and risk variants. The third test, afcSKAT(), is a score test that 
overcomes the mix of SNP (Single-Nucleotide Polymorphism) effect 
directions. For more details see 
Saad M and Wijsman EM (2017) &lt;doi:10.1093/bib/bbx107&gt;.
</p>


<h3>Author(s)</h3>

<p>Khalid B. Kunji [aut, cre], Mohamad Saad [aut]
</p>
<p>Maintainer: Khalid B. Kunji &lt;kkunji@hbku.edu.qa&gt;
</p>


<h3>References</h3>

<p>Mohamad Saad, Ellen M. Wijsman; Association score testing for rare variants and binary traits in family data with shared controls, Briefings in Bioinformatics, , bbx107, https://doi.org/10.1093/bib/bbx107
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Wcorrected">Wcorrected</a></code>
<code><a href="#topic+Wqls">Wqls</a></code>
<code><a href="#topic+afcSKAT">afcSKAT</a></code>
</p>

<hr>
<h2 id='cor.afc'>
AFC Correlation Example
</h2><span id='topic+cor.afc'></span>

<h3>Description</h3>

<p>An example correlation matrix that corresponds to the the one derived in the unrun example's
first iteration.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("cor.afc")</code></pre>


<h3>Format</h3>

<p>The format is:
num [1:75, 1:75] 1 0.2153 0.14829 -0.00285 0.0536 ...
- attr(*, &quot;dimnames&quot;)=List of 2
..$ : chr [1:75] &quot;rs3745120_1&quot; &quot;chr19.22817596_1&quot; &quot;chr19.22817734_1&quot; &quot;chr19.22817955_1&quot; ...
..$ : chr [1:75] &quot;rs3745120_1&quot; &quot;chr19.22817596_1&quot; &quot;chr19.22817734_1&quot; &quot;chr19.22817955_1&quot; ...
</p>


<h3>Source</h3>

<p>This is simulated data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("cor.afc")
</code></pre>

<hr>
<h2 id='geno.afc'>
AFC Genotype Example
</h2><span id='topic+geno.afc'></span>

<h3>Description</h3>

<p>An example genotype file that corresponds to the the one derived in the unrun example's
first iteration.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("geno.afc")</code></pre>


<h3>Format</h3>

<p>The format is:
int [1:1800, 1:75] 0 0 0 0 0 0 0 0 0 0 ...
- attr(*, &quot;dimnames&quot;)=List of 2
..$ : NULL
..$ : chr [1:75] &quot;rs3745120_1&quot; &quot;chr19.22817596_1&quot; &quot;chr19.22817734_1&quot; &quot;chr19.22817955_1&quot; ...
</p>


<h3>Source</h3>

<p>This is simulated data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("geno.afc")
</code></pre>

<hr>
<h2 id='kin.afc'>
AFC Kinship Example
</h2><span id='topic+kin.afc'></span>

<h3>Description</h3>

<p>An example kinship matrix that corresponds to the the one derived in the unrun example's
first iteration.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("kin.afc")</code></pre>


<h3>Format</h3>

<p>The format is:
num [1:1800, 1:1800] 0.5 0 0.25 0.25 0.25 0.25 0.25 0.25 0.25 0.25 ...
- attr(*, &quot;dimnames&quot;)=List of 2
..$ : chr [1:1800] &quot;1.1&quot; &quot;1.2&quot; &quot;1.3&quot; &quot;1.4&quot; ...
..$ : chr [1:1800] &quot;1.1&quot; &quot;1.2&quot; &quot;1.3&quot; &quot;1.4&quot; ...
</p>


<h3>Source</h3>

<p>This is simulated data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("kin.afc")
</code></pre>

<hr>
<h2 id='maf.afc'>
AFC MAF Example
</h2><span id='topic+maf.afc'></span>

<h3>Description</h3>

<p>An example MAF (Mean Allele Frequency) file that corresponds to the the one derived in the unrun
example's first iteration.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("maf.afc")</code></pre>


<h3>Format</h3>

<p>The format is:
num [1:75, 1:2] 0.00874 0.00541 0.00333 0 0.00291 ...
</p>


<h3>Source</h3>

<p>This is simulated data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("maf.afc")
</code></pre>

<hr>
<h2 id='phenotype.afc'>
AFC Phenotype Example
</h2><span id='topic+phenotype.afc'></span>

<h3>Description</h3>

<p>An example phenotype file that corresponds to the the one derived in the unrun example's
first iteration.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("phenotype.afc")</code></pre>


<h3>Format</h3>

<p>The format is:
int [1:1800, 1] 1 1 1 0 0 1 1 1 1 1 ...
</p>


<h3>Source</h3>

<p>This is simulated data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("phenotype.afc")
</code></pre>

<hr>
<h2 id='Wcorrected'>
Corrected Chi Squared Test
</h2><span id='topic+Wcorrected'></span>

<h3>Description</h3>

<p>Corrected Chi Squared Test, Wcorrected(), for multiple rare variant association using the
difference of the sum of minor allele frequencies between cases and controls. This test handles
related individuals, unrelated individuals, or both.  Note: this is referred to as X (Chi) Squared Corrected in the reference rather than W Corrected.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Wcorrected(MAF, Pheno, Kin, Correlation, Weights)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Wcorrected_+3A_maf">MAF</code></td>
<td>

<p>matrix (#Snps * 2): First column contains Minor Allele Frequency (MAF) in cases; Second column contains MAF in controls.
</p>
</td></tr>
<tr><td><code id="Wcorrected_+3A_pheno">Pheno</code></td>
<td>

<p>matrix (#subjects * 1): this one-column matrix contains 0's amd 1's: 1 for cases and 0 for controls. No missing values are allowed.
</p>
</td></tr>
<tr><td><code id="Wcorrected_+3A_kin">Kin</code></td>
<td>

<p>The kinship matrix (#subjects * #subjects): the subjects must be ordered as the Pheno variable.
</p>
</td></tr>
<tr><td><code id="Wcorrected_+3A_correlation">Correlation</code></td>
<td>

<p>Correlation matrix between SNPs (#Snps * #Snps). The user should calculate this matrix beforehand. Either based on own genotype data (in cases, controls, or both) or based on public databases (e.g., 1000 Genomes Projects, ESP, etc.). NA values are not allowed. They have to be replaced by zeros.
</p>
</td></tr>
<tr><td><code id="Wcorrected_+3A_weights">Weights</code></td>
<td>

<p>The weights values that can be used to up-weight or down-weight SNPs. This size of this
vector is the number of Snps by 1. By default, the weights are 1 for all Snps.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector with the following values: the sum of MAF for cases, the sum of MAF for controls, the sum of MAF for all weighted by the phenotype, the numerator of the test, the denominator of the test, the Wcorrected value (the main value calculated by the test), and the P-value.
</p>


<h3>References</h3>

<p>Saad M and Wijsman EM, Association score testing for rare variants and binary traits in family
data with shared controls, Briefings in Bioinformatics, 2017.
Schaid DJ , McDonnell SK , Sinnwell JP , et al.   Multiple genetic variant association testing
by collapsing and kernel methods with pedigree or population structured data.
Genet Epidemiol 2013 ;37 :409 –18.
Choi Y , Wijsman EM , Weir BS.  Case-control association testing in the presence of unknown
relationships. Genet Epidemiol 2009 ;33 :668 –78.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+AssocAFC">AssocAFC</a></code>
<code><a href="#topic+Wqls">Wqls</a></code>
<code><a href="#topic+afcSKAT">afcSKAT</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
P_Wcorrected &lt;- vector("numeric")
#This data corresponds to what is used in the 1st iteration with the raw data
data("maf.afc")
data("phenotype.afc")
data("kin.afc")
data("cor.afc")
data("weights.afc")
CORREC &lt;- Wcorrected(MAF = maf.afc , Pheno = phenotype.afc, Kin = kin.afc , Correlation=cor.afc,
                                                                Weights = weights.afc)
P_Wcorrected &lt;- c(P_Wcorrected, CORREC[7])
print(P_Wcorrected)


## Not run: 
#This example shows processing the raw data and uses kinship2,
#which AFC does not depend on

library(kinship2)
library(CompQuadForm)

P_Wcorrected &lt;- vector("numeric")

for (j in 1:10)
{
  geno.afc &lt;- read.table(system.file("extdata", "Additive_Genotyped_Truncated.txt",
                         package = "AFC"), header = TRUE)
  geno.afc[ , "IID"] &lt;- paste(geno.afc[ , "FID"]  , geno.afc[ , "IID"]  ,sep=".")
  geno.afc[geno.afc[,"FA"]!=0 , "FA"] &lt;- paste(geno.afc[geno.afc[,"FA"]!=0 , "FID"],
                                      geno.afc[geno.afc[,"FA"]!=0 , "FA"]  ,sep=".")
  geno.afc[geno.afc[,"FA"]!=0 , "MO"] &lt;- paste(geno.afc[geno.afc[,"FA"]!=0 , "FID"],
                                      geno.afc[geno.afc[,"FA"]!=0 , "MO"]  ,sep=".")
  Kinship &lt;- makekinship(geno.afc$FID , geno.afc$IID , geno.afc$FA, geno.afc$MO)
  kin.afc &lt;- as.matrix(Kinship)
  pheno.afc &lt;- read.table(system.file("extdata", "Phenotype", package = "AFC"))
  phenotype.afc &lt;- matrix(pheno.afc[,j],nc=1,nr=nrow(pheno.afc))
  geno.afc &lt;- geno.afc[,7:ncol(geno.afc)]
  Na &lt;- nrow(pheno.afc[pheno.afc[,j]==1,])
  Nu &lt;- nrow(pheno.afc[pheno.afc[,j]==0,])
  N &lt;- Nu + Na
  maf.afc &lt;- matrix(NA , nr=ncol(geno.afc) , nc=2)
  maf.afc[,1] &lt;- colMeans(geno.afc[phenotype.afc==1,])/2;
  maf.afc[,2] &lt;- colMeans(geno.afc[phenotype.afc==0,])/2;
  P  &lt;- (maf.afc[,1]*Na + maf.afc[,2]*Nu)/N
  Set &lt;- which(P&lt;0.05)
  maf.afc &lt;- maf.afc[c(Set),]
  cor.afc &lt;- cor(geno.afc[,c(Set)])
  cor.afc[is.na(cor.afc)] &lt;- 0

  weights.afc &lt;- matrix(1/(maf.afc[,2]+1),nc=1,nr=length(Set))
  CORREC &lt;- Wcorrected(MAF = maf.afc , Pheno = phenotype.afc, Kin = kin.afc , Correlation=cor.afc,
                                                                Weights = weights.afc)
  P_Wcorrected &lt;- c(P_Wcorrected, CORREC[7])
}
print(P_Wcorrected)

## End(Not run)

## The function is currently defined as
function(MAF, Pheno, Kin, Correlation, Weights)
{
  Na     &lt;- length(Pheno[Pheno[, 1] == 1,])
  Nu     &lt;- length(Pheno[Pheno[, 1] == 0,])
  N      &lt;- Na + Nu

  # The three following lines: prepare the phenotype variables
  OneN  &lt;- matrix(1, ncol = 1, nrow = N)
  Y  &lt;- Pheno
  OneHat &lt;- matrix(Na / N, ncol = 1 , nrow = N)

  # Estimate MAF in all subjects
  P  &lt;- (MAF[, 1] * Na + MAF[, 2] * Nu) / N
  if (is.null(Weights))
  {
    # Variance of SNPs (2p(1-p))
    VarSnps &lt;- sqrt(P * (1 - P))
  } else
  {
    # Variance of SNPs (2p(1-p)) accounting for the prespecified Snp weights
    VarSnps &lt;- Weights * sqrt(P * (1 - P))
  }
  VarSnps &lt;- matrix(VarSnps, ncol = 1)
  # This value will account for the correlation between Snps.
  cs &lt;- 2 * t(VarSnps) %*% Correlation %*% VarSnps

  if (is.null(Weights))

  {
    # Numerator of the Xcorrec test statistic
    num &lt;- 4 * (sum (Na * MAF[, 1] - Na * P)) ^ 2
  } else{
    # Numerator of the Xcorrec test statistic
    num &lt;- 4 * (sum (Na * Weights * MAF[, 1] - Na * Weights * P)) ^ 2
  }
  # Denominator of the Xcorrec test statistic
  denom &lt;- 2 * as.numeric(cs) * t(Y - OneHat) %*% Kin %*% (Y - OneHat)
  # Xcorrec test statistic
  W &lt;- num / denom
  # Pvalue from a chi-square proba distribution
  Pvalue &lt;- 1 - pchisq(W, 1)
  out &lt;- t(data.frame(c(sum(MAF[,1]), sum(MAF[,2]), sum(P), num, denom, W, Pvalue)))
  colnames(out) &lt;- c("Sum MAF Cases", "Sum MAF Controls", "Sum MAF All Weighted", "Numerator",
                     "Denominator", "Wcorrected", "Pvalue")
  rownames(out) &lt;- "Statistics"
  return(out)
}

</code></pre>

<hr>
<h2 id='weights.afc'>
AFC Weights Example
</h2><span id='topic+weights.afc'></span>

<h3>Description</h3>

<p>An example weights file that corresponds to the the one derived in the unrun example's
first iteration.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("weights.afc")</code></pre>


<h3>Format</h3>

<p>The format is:
num [1:75, 1] 0.996 0.995 0.995 0.999 0.998 ...
</p>


<h3>Source</h3>

<p>This is simulated data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("weights.afc")
</code></pre>

<hr>
<h2 id='Wqls'>
W Quasi-Likelihood Score
</h2><span id='topic+Wqls'></span>

<h3>Description</h3>

<p>The W Quasi-Likelihood Score, Wqls(), is a score for multiple rare variant association tests
using the difference of the sum of minor allele frequencies between cases and controls.
This test handles related individuals, unrelated individuals, or both. This test is expected
to be similar to a linear mixed model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Wqls(Genotypes, MAF, Pheno, Kin, Correlation, Weights)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Wqls_+3A_genotypes">Genotypes</code></td>
<td>

<p>matrix(#Subjects * #Snps). The genotypes are coded as 0, 1, or 2 copies of the minor allele.
</p>
</td></tr>
<tr><td><code id="Wqls_+3A_maf">MAF</code></td>
<td>

<p>matrix (#Snps * 2): First column contains Minor Allele Frequency (MAF) in cases; Second column contains MAF in controls.
</p>
</td></tr>
<tr><td><code id="Wqls_+3A_pheno">Pheno</code></td>
<td>

<p>matrix (#subjects * 1): this one-column matrix contains 0's amd 1's: 1 for cases and 0 for controls. No missing values are allowed.
</p>
</td></tr>
<tr><td><code id="Wqls_+3A_kin">Kin</code></td>
<td>

<p>The kinship matrix (#subjects * #subjects): the subjects must be ordered as the Pheno variable.
</p>
</td></tr>
<tr><td><code id="Wqls_+3A_correlation">Correlation</code></td>
<td>

<p>Correlation matrix between SNPs (#Snps * #Snps). The user should calculate this matrix beforehand. Either based on own genotype data (in cases, controls, or both) or based on public databases (e.g., 1000 Genomes Projects, ESP, etc.). NA values are not allowed. They have to be replaced by zeros.
</p>
</td></tr>
<tr><td><code id="Wqls_+3A_weights">Weights</code></td>
<td>

<p>The weights values that can be used to up-weight or down-weight SNPs. This size of this
vector is the number of Snps by 1. By default, the weights are 1 for all Snps.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector with the following values: the sum of MAF for cases, the sum of MAF for controls, the sum of MAF for all weighted by the phenotype, the numerator of the test, the denominator of the test, the Wqls value (the main value calculated by the test), and the P-value.
</p>


<h3>References</h3>

<p>Saad M and Wijsman EM, Association score testing for rare variants and binary traits in family
data with shared controls, Briefings in Bioinformatics, 2017.
Bourgain C , Hoffjan S , Nicolae R , et al.   Novel case-control test in a founder population
identifies P-selectin as an atopy-susceptibility locus . Am J Hum Genet 2003 ;73 :612 –26.
Thornton T , McPeek MS.  Case-control association testing with related individuals: a more
powerful quasi-likelihood score test. Am J Hum Genet 2007 ;81 :321 –37.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+AssocAFC">AssocAFC</a></code>
<code><a href="#topic+Wcorrected">Wcorrected</a></code>
<code><a href="#topic+afcSKAT">afcSKAT</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
P_WQLS &lt;- vector("numeric")
#This data corresponds to what is used in the 1st iteration with the raw data
data("geno.afc")
data("maf.afc")
data("phenotype.afc")
data("kin.afc")
data("cor.afc")
data("weights.afc")
QLS &lt;- Wqls(Genotypes = geno.afc, MAF = maf.afc, Pheno = phenotype.afc, Kin = kin.afc,
                                        Correlation = cor.afc, Weights = weights.afc)
P_WQLS &lt;- c(P_WQLS, QLS[7])
print(P_WQLS)

## Not run: 
#This example shows processing the raw data and uses kinship2,
#which AFC does not depend on

library(kinship2)
library(CompQuadForm)

P_WQLS &lt;- vector("numeric")

for (j in 1:10)
{
  geno.afc &lt;- read.table(system.file("extdata", "Additive_Genotyped_Truncated.txt",
                         package = "AFC"), header = TRUE)
  geno.afc[ , "IID"] &lt;- paste(geno.afc[ , "FID"]  , geno.afc[ , "IID"]  ,sep=".")
  geno.afc[geno.afc[,"FA"]!=0 , "FA"] &lt;- paste(geno.afc[geno.afc[,"FA"]!=0 , "FID"],
                                      geno.afc[geno.afc[,"FA"]!=0 , "FA"]  ,sep=".")
  geno.afc[geno.afc[,"FA"]!=0 , "MO"] &lt;- paste(geno.afc[geno.afc[,"FA"]!=0 , "FID"],
                                      geno.afc[geno.afc[,"FA"]!=0 , "MO"]  ,sep=".")
  Kinship &lt;- makekinship(geno.afc$FID , geno.afc$IID , geno.afc$FA, geno.afc$MO)
  kin.afc &lt;- as.matrix(Kinship)
  pheno.afc &lt;- read.table(system.file("extdata", "Phenotype", package = "AFC"))
  phenotype.afc &lt;- matrix(pheno.afc[,j],nc=1,nr=nrow(pheno.afc))
  geno.afc &lt;- geno.afc[,7:ncol(geno.afc)]
  Na &lt;- nrow(pheno.afc[pheno.afc[,j]==1,])
  Nu &lt;- nrow(pheno.afc[pheno.afc[,j]==0,])
  N &lt;- Nu + Na
  maf.afc &lt;- matrix(NA , nr=ncol(geno.afc) , nc=2)
  maf.afc[,1] &lt;- colMeans(geno.afc[phenotype.afc==1,])/2;
  maf.afc[,2] &lt;- colMeans(geno.afc[phenotype.afc==0,])/2;
  P  &lt;- (maf.afc[,1]*Na + maf.afc[,2]*Nu)/N
  Set &lt;- which(P&lt;0.05)
  maf.afc &lt;- maf.afc[c(Set),]
  cor.afc &lt;- cor(geno.afc[,c(Set)])
  cor.afc[is.na(cor.afc)] &lt;- 0
  geno.afc &lt;- as.matrix(geno.afc[,Set])
  weights.afc &lt;- matrix(1/(maf.afc[,2]+1),nc=1,nr=length(Set))
  QLS &lt;- Wqls(Genotypes = geno.afc, MAF = maf.afc, Pheno = phenotype.afc, Kin = kin.afc,
                                        Correlation = cor.afc, Weights = weights.afc)
  P_WQLS &lt;- c(P_WQLS, QLS[7])
}
print(P_WQLS)

## End(Not run)

## The function is currently defined as
function(Genotypes, MAF, Pheno, Kin, Correlation, Weights)
{
  Na     &lt;- length(Pheno[Pheno[,1]==1,])
  Nu     &lt;- length(Pheno[Pheno[,1]==0,])
  N      &lt;- Na + Nu

  # The three following lines: prepare the Phenotype variables
  OneN  &lt;- matrix(1, ncol=1, nrow = N)
  Y  &lt;- Pheno
  OneHat &lt;- matrix( Na/N, ncol=1 , nrow=N)
  temp &lt;- Y - OneHat

  # Estimate MAF in all subjects
  P  &lt;- (MAF[,1]*Na + MAF[,2]*Nu)/N
  # Number of Snps
  Nsnps   &lt;- nrow(MAF)
  if (is.null(Weights))
  {
    # Variance of SNPs (2p(1-p))
    VarSnps &lt;- sqrt(P*(1-P))
  } else
  {
    # Variance of SNPs (2p(1-p)) accounting for the prespecified Snp weights
    VarSnps &lt;- Weights*sqrt(P*(1-P))
  }
  VarSnps &lt;- matrix(VarSnps,ncol=1)
  # This value will account for the correlation between Snps.
  cs      &lt;- 2*t(VarSnps) %*% Correlation %*% VarSnps

  if (Nsnps==1)
  {
    S&lt;-Genotypes
  }else
  {
    if (is.null(Weights))
    {
      # Rare variant score: sum of minor alleles accross all Snps.
      S &lt;- apply(Genotypes , 1 , sum)
    }else
    {
      # Rare variant score: sum of minor alleles accross all Snps.
      S &lt;- Genotypes %*% Weights
    }
  }
  S &lt;- matrix(S,ncol=1,nrow=length(S))
  Kin &lt;- 2*Kin
  KinInv &lt;- solve(Kin)
  A &lt;- as.numeric(t(Y) %*% KinInv %*% OneN %*% solve(t(OneN) %*% KinInv %*% OneN))
  V &lt;- KinInv %*% Y - A * KinInv %*% OneN
  if (is.null(Weights))
  {
    num &lt;- (sum((S[Y==1]) * rowSums(KinInv[Y==1,Y==1]))*(1-A) -2*sum(MAF[,2])*(A)*Nu)^2;
  }else
  {
    num &lt;- (sum((S[Y==1]) * rowSums(KinInv[Y==1,Y==1]))*(1-A) -2*sum(Weights*MAF[,2])*(A)*Nu)^2;
  }
  denom &lt;- as.numeric(cs) * (t(V)%*% Kin %*% V) ;
  W &lt;- num/denom ;
  Pvalue &lt;- 1-pchisq(W,1) ;
  out &lt;- t(data.frame(c(sum(MAF[,1]), sum(MAF[,2]), sum(P), num, denom, W, Pvalue)))
  colnames(out) &lt;- c("Sum MAF Cases", "Sum MAF Controls", "Sum MAF All Weighted", "Numerator",
                     "Denominator", "Wqls", "Pvalue")
  rownames(out) &lt;- "Statistics"
  return(out)
}
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
