<!DOCTYPE html><html><head><title>Help for package AssocTests</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {AssocTests}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#dr'><p>Distance regression</p></a></li>
<li><a href='#drS.eg'><p>A toy similarity matrix for dr</p></a></li>
<li><a href='#eigenstrat'><p>EIGENSTRAT for correcting for population stratification</p></a></li>
<li><a href='#max3'><p>Maximum Test: maximum value of the three Cochran-Armitage</p>
trend tests under the recessive, additive, and dominant models</a></li>
<li><a href='#nmax3'><p>NMAX3 based on the maximum value of the three nonparametric trend tests</p>
under the recessive, additive, and dominant models</a></li>
<li><a href='#npt'><p>Nonparametric trend test based on the nonparametric</p>
risk under a given genetic model</a></li>
<li><a href='#pcoc'><p>PCoC for correcting for population stratification</p></a></li>
<li><a href='#smt'><p>Single-marker test</p></a></li>
<li><a href='#tw'><p>Tracy-Widom test</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genetic Association Studies</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0-1</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-1-2</td>
</tr>
<tr>
<td>Author:</td>
<td>Lin Wang [aut],
    Wei Zhang [aut],
    Qizhai Li [aut],
    Weicheng Zhu [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lin Wang &lt;wanglin2009@amss.ac.cn&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>cluster, mvtnorm, combinat, fExtremes, R(&ge; 2.10.0)</td>
</tr>
<tr>
<td>Description:</td>
<td>Some procedures including EIGENSTRAT (a procedure for detecting and correcting 
	for population stratification through searching for the eigenvectors in genetic association 
	studies), PCoC (a procedure for correcting for population stratification through calculating 
	the principal coordinates and the clustering of the subjects), Tracy-Widom test (a procedure 
	for detecting the significant eigenvalues of a matrix), distance regression (a procedure for 
	detecting the association between a distance matrix and some independent variants of interest), 
	single-marker test (a procedure for identifying the association between the genotype at a 
	biallelic marker and a trait using the Wald test or the Fisher's exact test), MAX3 (a procedure 
	for testing for the association between a single nucleotide polymorphism and a binary phenotype 
	using the maximum value of the three test statistics derived for the recessive, additive, and 
	dominant models), nonparametric trend test (a procedure for testing for the association between 
	a genetic variant and a non-normal distributed quantitative trait based on the nonparametric 
	risk), and nonparametric MAX3 (a procedure for testing for the association between a biallelic 
	single nucleotide polymorphism and a quantitative trait using the maximum value of the three 
	nonparametric trend tests derived for the recessive, additive, and dominant models), which are 
	commonly used in genetic association studies. To cite this package in publications use: 
	Lin Wang, Wei Zhang, and Qizhai Li. AssocTests: An R Package for Genetic Association Studies. 
	Journal of Statistical Software. 2020; 94(5): 1-26.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Collate:</td>
<td>'AssociationTestWithCorrectPS.R' 'CalculateGapK.R'
'CalculateWAll.R' 'CalExpect.R' 'ChangeX.R' 'CorrMatNRTest.R'
'DR_main.R' 'EigenStrat_main.R' 'FindCNumRandom.R'
'MAX3_main.R' 'MAX3Sign.R' 'MDS.R' 'PCoC_main.R'
'ModifyNormalization.R' 'NMAX3_main.R' 'NPT_main.R'
'RhombusFormula.R' 'ScoreTest.R' 'SimilarityMatrix.R'
'SMT_main.R' 'Str2Num.R' 'TrendTest.R' 'TW_main.R'
'UniformSample.R' 'UniformTest.R' 'WaldTest.R'</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-01-02 13:29:51 UTC; admin</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-01-10 11:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='dr'>Distance regression</h2><span id='topic+dr'></span>

<h3>Description</h3>

<p>Conduct the distance regression with or without the adjustment of
the covariates to detect the association between a distance matrix
and some independent variants of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dr(
  simi.mat,
  null.space,
  x.mat,
  permute = TRUE,
  n.MonteCarlo = 1000,
  seed = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dr_+3A_simi.mat">simi.mat</code></td>
<td>
<p>a similarity matrix among the subjects.</p>
</td></tr>
<tr><td><code id="dr_+3A_null.space">null.space</code></td>
<td>
<p>a numeric vector to show the column numbers of
the null space in <code>x.mat</code>.</p>
</td></tr>
<tr><td><code id="dr_+3A_x.mat">x.mat</code></td>
<td>
<p>the covariate matrix which combines the null space
and the matrix of interest.</p>
</td></tr>
<tr><td><code id="dr_+3A_permute">permute</code></td>
<td>
<p>logical. If <code>TRUE</code>, the Monte Carlo sampling is used
without replacement; otherwise, with replacement. The default is
<code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="dr_+3A_n.montecarlo">n.MonteCarlo</code></td>
<td>
<p>the number of times for the Monte Carlo
procedure. The default is <code>1000</code>.</p>
</td></tr>
<tr><td><code id="dr_+3A_seed">seed</code></td>
<td>
<p>if it is not <code>NULL</code>, set the random number generator
state for random number generation. The default is <code>NULL</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The pseudo <em>F</em> statistic based on the distance regression with
or without the adjustment of the covariates detects the
association between a distance matrix and some independent
variants of interest. A distance matrix can be transformed into a
similarity matrix easily.
</p>


<h3>Value</h3>

<p>A list with class &quot;<code>htest</code>&quot; containing the following components:
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>statistic</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the observed value of the test statistic.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>p.value</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the p-value for the test.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>alternative</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string describing the alternative hypothesis.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string indicating the type of test performed.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>data.name</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string giving the names of the data.
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li.
</p>


<h3>References</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li. AssocTests: An R Package 
for Genetic Association Studies. <em>Journal of Statistical Software</em>. 
2020; 94(5): 1-26.
</p>
<p>Q Li, S Wacholder, DJ Hunter, RN Hoover, S Chanock, G
Thomas, and K Yu. Genetic Background Comparison Using
Distance-Based Regression, with Applications in Population
Stratification Evaluation and Adjustment. <em>Genetic
Epidemiology</em>. 2009; 33(5): 432-441.
</p>
<p>J Wessel and NJ Schork. Generalized Genomic
Distance-Based Regression Methodology for Multilocus Association
Analysis. <em>American Journal of Human Genetics</em>. 2006; 79(5):
792-806.
</p>
<p>MA Zapala and NJ Schork. Multivariate Regression
Analysis of Distance Matrices for Testing Associations Between
Gene Expression Patterns and Related Variables. <em>Proceedings
of the National Academy of Sciences of the United States of
America</em>. 2006; 103(51): 19430-19435.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(drS.eg)
null.space &lt;- 1
x.mat &lt;- matrix(c(rep(1, 600), rep(0, 200)), ncol=2)
dr(drS.eg, null.space, x.mat, permute = TRUE, n.MonteCarlo = 50, seed = NULL)
</code></pre>

<hr>
<h2 id='drS.eg'>A toy similarity matrix for <a href="#topic+dr">dr</a></h2><span id='topic+drS.eg'></span>

<h3>Description</h3>

<p>This data set is the toy similarity matrix among the subjects calculated from a toy genotype data set 
which contains of the genotypes of 400 subjects at 200 markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(drS.eg)</code></pre>

<hr>
<h2 id='eigenstrat'>EIGENSTRAT for correcting for population stratification</h2><span id='topic+eigenstrat'></span>

<h3>Description</h3>

<p>Find the eigenvectors of the similarity matrix among the subjects
used for correcting for population stratification in the
population-based genetic association studies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eigenstrat(
  genoFile,
  outFile.Robj = "out.list",
  outFile.txt = "out.txt",
  rm.marker.index = NULL,
  rm.subject.index = NULL,
  miss.val = 9,
  num.splits = 10,
  topK = NULL,
  signt.eigen.level = 0.01,
  signal.outlier = FALSE,
  iter.outlier = 5,
  sigma.thresh = 6
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="eigenstrat_+3A_genofile">genoFile</code></td>
<td>
<p>a txt file containing the genotypes (0, 1, 2, or
9). The element of the file in Row <em>i</em> and Column <em>j</em>
represents the genotype at the <em>i</em>th marker of the <em>j</em>th
subject. 0, 1, and 2 denote the number of risk alleles, and 9
(default) is for the missing genotype.</p>
</td></tr>
<tr><td><code id="eigenstrat_+3A_outfile.robj">outFile.Robj</code></td>
<td>
<p>the name of an R object for saving the list of
the results which is the same as the return value of this
function. The default is &quot;<code>out.list</code>&quot;.</p>
</td></tr>
<tr><td><code id="eigenstrat_+3A_outfile.txt">outFile.txt</code></td>
<td>
<p>a txt file for saving the eigenvectors
corresponding to the top significant eigenvalues.</p>
</td></tr>
<tr><td><code id="eigenstrat_+3A_rm.marker.index">rm.marker.index</code></td>
<td>
<p>a numeric vector for the indices of the
removed markers. The default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="eigenstrat_+3A_rm.subject.index">rm.subject.index</code></td>
<td>
<p>a numeric vector for the indices of the
removed subjects. The default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="eigenstrat_+3A_miss.val">miss.val</code></td>
<td>
<p>the number representing the missing data in the
input data. The default is <code>9</code>. The element 9 for the missing data
in the <code>genoFile</code> should be changed according to the value of
<code>miss.val</code>.</p>
</td></tr>
<tr><td><code id="eigenstrat_+3A_num.splits">num.splits</code></td>
<td>
<p>the number of groups into which the markers are
split. The default is <code>10</code>.</p>
</td></tr>
<tr><td><code id="eigenstrat_+3A_topk">topK</code></td>
<td>
<p>the number of eigenvectors to return. If <code>NULL</code>, it is
calculated by the Tracy-Widom test. The default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="eigenstrat_+3A_signt.eigen.level">signt.eigen.level</code></td>
<td>
<p>a numeric value which is the significance
level of the Tracy-Widom test. It should be <code>0.05</code>, <code>0.01</code>, <code>0.005</code>, or
<code>0.001</code>. The default is <code>0.01</code>.</p>
</td></tr>
<tr><td><code id="eigenstrat_+3A_signal.outlier">signal.outlier</code></td>
<td>
<p>logical. If <code>TRUE</code>, delete the outliers of the
subjects; otherwise, do not search for the outliers. The default
is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="eigenstrat_+3A_iter.outlier">iter.outlier</code></td>
<td>
<p>a numeric value that is the iteration time for
finding the outliers of the subjects. The default is <code>5</code>.</p>
</td></tr>
<tr><td><code id="eigenstrat_+3A_sigma.thresh">sigma.thresh</code></td>
<td>
<p>a numeric value that is the lower limit for
eliminating the outliers. The default is <code>6</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose that a total of <em>n</em> cases and controls are randomly
enrolled in the source population and a panel of <em>m</em>
single-nucleotide polymorphisms are genotyped. The genotype at a
marker locus is coded as 0, 1, or 2, with the value corresponding
to the copy number of risk alleles. All the genotypes are given in
the form of a <em>m*n</em> matrix, in which the element in the
<em>i</em>th row and the <em>j</em>th column represents the genotype
of the <em>j</em>th subject at the <em>i</em>th marker. This function
calculates the top eigenvectors or the eigenvectors with
significant eigenvalues of the similarity matrix among the
subjects to infer the potential population structure. See also
<a href="#topic+tw">tw</a>.
</p>


<h3>Value</h3>

<p><code>eigenstrat</code> returns a list, which contains the following components:
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>num.markers</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the number of markers excluding the removed markers.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>num.subjects</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the number of subjects excluding the outliers.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>rm.marker.index</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the indices of the removed markers.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>rm.subject.index</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the indices of the removed subjects.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TW.level</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the significance level of the Tracy-Widom test.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>signal.outlier</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> dealing with the outliers in the subjects or not.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>iter.outlier</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the iteration time for finding the outliers.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sigma.thresh</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the lower limit for eliminating the outliers.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>num.outliers</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the number of outliers.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>outliers.index</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the indices of the outliers.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>num.used.subjects</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the number of the used subjects.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>used.subjects.index</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the indices of the used subjects.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>similarity.matrix</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the similarity matrix among the subjects.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>eigenvalues</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the eigenvalues of the similarity matrix.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>eigenvectors</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the eigenvectors corresponding to the eigenvalues.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>topK</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the number of significant eigenvalues.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TW.stat</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the observed values of the Tracy-Widom statistics.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>topK.eigenvalues</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the top eigenvalues.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>topK.eigenvectors</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the eigenvectors corresponding to the top eigenvalues.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>runtime</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the running time of this function.
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li.
</p>


<h3>References</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li. AssocTests: An R Package 
for Genetic Association Studies. <em>Journal of Statistical Software</em>. 
2020; 94(5): 1-26.
</p>
<p>AL Price, NJ Patterson, RM Plenge, ME Weinblatt, NA
Shadick, and D Reich. Principal Components Analysis Corrects for
Stratification in Genome-Wide Association Studies. <em>Nature
Genetics</em>. 2006; 38(8): 904-909.
</p>
<p>N Patterson, AL Price, and D Reich. Population
Structure and Eigenanalysis. <em>PloS Genetics</em>. 2006; 2(12):
2074-2093.
</p>
<p>CA Tracy and H Widom. Level-Spacing Distributions and
the Airy Kernel. <em>Communications in Mathematical
Physics</em>. 1994; 159(1): 151-174.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>eigenstratG.eg &lt;- matrix(rbinom(3000, 2, 0.5), ncol = 30)
write.table(eigenstratG.eg, file = "eigenstratG.eg.txt", quote = FALSE,
            sep = "", row.names = FALSE, col.names = FALSE)
eigenstrat(genoFile = "eigenstratG.eg.txt", outFile.Robj = "eigenstrat.result.list",
             outFile.txt = "eigenstrat.result.txt", rm.marker.index = NULL,
             rm.subject.index = NULL, miss.val = 9, num.splits = 10,
             topK = NULL, signt.eigen.level = 0.01, signal.outlier = FALSE,
             iter.outlier = 5, sigma.thresh = 6)
file.remove("eigenstratG.eg.txt", "eigenstrat.result.list", "eigenstrat.result.txt")
</code></pre>

<hr>
<h2 id='max3'>Maximum Test: maximum value of the three Cochran-Armitage
trend tests under the recessive, additive, and dominant models</h2><span id='topic+max3'></span>

<h3>Description</h3>

<p>Conduct MAX3 (the maximal value of the three Cochran-Armitage
trend tests derived for the recessive, additive, and dominant
models) based on the trend tests without the adjustment of the
covariates or based on the Wald tests with the adjustment of the
covariates to test for the association between a single-nucleotide
polymorphism and the binary phenotype.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>max3(
  y,
  g,
  covariates = NULL,
  Score.test = TRUE,
  Wald.test = FALSE,
  rhombus.formula = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="max3_+3A_y">y</code></td>
<td>
<p>a numeric vector of the observed trait values in which
the <em>i</em>th element is for the <em>i</em>th subject. The elements
should be <code>0</code> or <code>1</code>.</p>
</td></tr>
<tr><td><code id="max3_+3A_g">g</code></td>
<td>
<p>a numeric vector of the observed genotype values (<code>0</code>, <code>1</code>,
or <code>2</code> denotes the number of risk alleles) in which the <em>i</em>th
element is for the <em>i</em>th subject. The missing value is
represented by <code>NA</code>. <code>g</code> has the same length as <code>y</code>.</p>
</td></tr>
<tr><td><code id="max3_+3A_covariates">covariates</code></td>
<td>
<p>a numeric matrix for the covariates used in the
model. Each column is for one covariate. The default is <code>NULL</code>, that
is, there are no covariates to be adjusted for.</p>
</td></tr>
<tr><td><code id="max3_+3A_score.test">Score.test</code></td>
<td>
<p>logical. If <code>TRUE</code>, the score tests are used. One
of <code>Score.test</code> and <code>Wald.test</code> should be <code>FALSE</code>,
and the other should be <code>TRUE</code>. The default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="max3_+3A_wald.test">Wald.test</code></td>
<td>
<p>logical. If <code>TRUE</code>, the Wald tests are used. One of
<code>Score.test</code> and <code>Wald.test</code> should be <code>FALSE</code>,
and the other should be <code>TRUE</code>. The default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="max3_+3A_rhombus.formula">rhombus.formula</code></td>
<td>
<p>logical. If <code>TRUE</code>, the p-value for the MAX3
is approximated by the rhombus formula. IF <code>FALSE</code>, the 2-fold
integration is used to calculate the p-value. The default is
<code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In an association study, the genetic inheritance models
(recessive, additive, or dominant) are unknown beforehand. This
function can account for the uncertainty of the underlying genetic
models and test for the association between a single-nucleotide
polymorphism and a binary phenotype with or without correcting for
the covariates.
</p>


<h3>Value</h3>

<p>A list with class &quot;<code>htest</code>&quot; containing the following components:
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>statistic</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the observed value of the test statistic.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>p.value</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the p-value for the test.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>alternative</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string describing the alternative hypothesis.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string indicating the type of test performed.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>data.name</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string giving the names of the data.
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li.
</p>


<h3>References</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li. AssocTests: An R Package 
for Genetic Association Studies. <em>Journal of Statistical Software</em>. 
2020; 94(5): 1-26.
</p>
<p>Q Li, G Zheng, Z Li, and K Yu. Efficient Approximation
of P Value of the Maximum of Correlated Tests, with Applications
to Genome-Wide Association Studies. <em>Annals of Human
Genetics</em>. 2008; 72(3): 397-406.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>y &lt;- rep(c(0, 1), 5)
g &lt;- sample(c(0, 1, 2), 10, replace = TRUE)
max3(y, g, covariates = NULL, Score.test = TRUE, Wald.test = FALSE,
       rhombus.formula = FALSE)
max3(y, g, covariates = matrix(sample(c(0,1), 20, replace = TRUE), ncol=2),
       Score.test = TRUE, Wald.test = FALSE, rhombus.formula = FALSE)
</code></pre>

<hr>
<h2 id='nmax3'>NMAX3 based on the maximum value of the three nonparametric trend tests
under the recessive, additive, and dominant models</h2><span id='topic+nmax3'></span>

<h3>Description</h3>

<p>Test for the association between a biallelic SNP and a
quantitative trait using the maximum value of the three
nonparametric trend tests derived for the recessive, additive, and
dominant models. It is a robust procedure against the genetic
models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nmax3(y, g)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nmax3_+3A_y">y</code></td>
<td>
<p>a numeric vector of the observed quantitative trait
values in which the <em>i</em>th element is the trait value of the
<em>i</em>th subject.</p>
</td></tr>
<tr><td><code id="nmax3_+3A_g">g</code></td>
<td>
<p>a numeric vector of the observed genotype values (<code>0</code>, <code>1</code>,
or <code>2</code> denotes the number of risk alleles) in which the <em>i</em>th
element is the genotype value of the <em>i</em>th subject for a
biallelic SNP. <code>g</code> has the same length as <code>y</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Under the null hypothesis of no association, the vector of the
three nonparametric tests under the recessive, additive, and
dominant models asymptotically follows a three-dimensional normal
distribution. The p-value can be calculated using the function
<a href="mvtnorm.html#topic+pmvnorm">pmvnorm</a> in the R package &quot;<span class="pkg">mvtnorm</span>&quot;.
</p>
<p>This test is different from the MAX3 test using in the function
<a href="#topic+max3">max3</a>. On one hand, the NMAX3 applies to the quantitative
traits association studies. However, the MAX3 is used in the
case-control association studies. On the other hand, the NMAX3 is
based on the nonparametric trend test. However, the MAX3 is based
on the Cochran-Armitage trend test.
</p>


<h3>Value</h3>

<p>A list with class &quot;<code>htest</code>&quot; containing the following components:
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>statistic</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the observed value of the test statistic.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>p.value</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the p-value for the test.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>alternative</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string describing the alternative hypothesis.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string indicating the type of test performed.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>data.name</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string giving the names of the data.
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li.
</p>


<h3>References</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li. AssocTests: An R Package 
for Genetic Association Studies. <em>Journal of Statistical Software</em>. 
2020; 94(5): 1-26.
</p>
<p>W Zhang and Q Li. Nonparametric Risk and Nonparametric
Odds in Quantitative Genetic Association Studies. <em>Science
Reports (2nd revision)</em>. 2015.
</p>
<p>B Freidlin, G Zheng, Z Li, and JL Gastwirth. Trend
Tests for Case-Control Studies of Genetic Markers: Power, Sample
Size and Robustness. <em>Human Heredity</em>. 2002; 53:146-152.
</p>
<p>WG Cochran. Some Methods for Strengthening the Common
Chi-Square Tests. <em>Biometrics</em>. 1954; 10:417-451.
</p>
<p>P Armitage. Tests for Linear Trends in Proportions and
Frequencies. <em>Biometrics</em>. 1955; 11:375-386.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>g &lt;- rbinom(1500, 2, 0.3)
y &lt;- 0.5 + 0.25 * g + rgev(1500, 0, 0, 5)
nmax3(y, g)
</code></pre>

<hr>
<h2 id='npt'>Nonparametric trend test based on the nonparametric
risk under a given genetic model</h2><span id='topic+npt'></span>

<h3>Description</h3>

<p>Test for the association between a genetic variant and a
non-normal distributed quantitative trait based on the
nonparametric risk under a specific genetic model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>npt(y, g, varphi)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="npt_+3A_y">y</code></td>
<td>
<p>a numeric vector of the observed quantitative trait
values in which the <em>i</em>th element corresponds to the trait
value of the <em>i</em>th subject.</p>
</td></tr>
<tr><td><code id="npt_+3A_g">g</code></td>
<td>
<p>a numeric vector of the observed genotype values (<code>0</code>, <code>1</code>,
or <code>2</code> denotes the number of risk alleles) in which the <em>i</em>th
element is the genotype value of the <em>i</em>th subject for a
biallelic SNP. <code>g</code> has the same length as <code>y</code>.</p>
</td></tr>
<tr><td><code id="npt_+3A_varphi">varphi</code></td>
<td>
<p>a numeric value which represents the genetic
model. It should be <code>0</code>, <code>0.5</code>, or <code>1</code>, which indicates that the
calculation is performed under the recessive, additive, or
dominant model, respectively.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For a non-normal distributed quantitative trait, three genetic
models (recessive, additive and dominant) used commonly are
defined in terms of the nonparametric risk (NR). The recessive,
additive, and dominant models can be classified based on the
nonparametric risks. More specifically, the recessive, additive,
and dominant models refer to NR20 &gt; NR10 = 1/2, NR12 = NR10&gt;1/2, and
NR10 = NR20 &gt; 1/2, respectively, where NR10 and NR20 are the
nonparametric risks of the groups with the genotypes 1 and 2
relative to the group with the genotype 0, respectively, and NR12
is the nonparametric risk of the group with the genotype 2
relative to the group with the genotype 1.
</p>
<p><code>varphi</code> can be <code>0</code>, <code>0.5</code>, or <code>1</code> for the recessive, additive, or
dominant model, respectively. When <code>varphi</code> is <code>0</code>, the test is
constructed under the recessive model by pooling together the
subjects with the genotypes 0 and 1. Similarly, when <code>varphi</code>
is <code>1</code>, the test is constructed under the dominant model by pooling
together the subjects with the genotypes 1 and 2. When
<code>varphi</code> is <code>0.5</code>, the test is based on the weighted sum of
NR10 and NR12.
</p>


<h3>Value</h3>

<p>A list with class &quot;<code>htest</code>&quot; containing the following components:
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>statistic</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the observed value of the test statistic.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>p.value</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the p-value for the test.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>alternative</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string describing the alternative hypothesis.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string indicating the type of test performed.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>data.name</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string giving the names of the data.
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li.
</p>


<h3>References</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li. AssocTests: An R Package 
for Genetic Association Studies. <em>Journal of Statistical Software</em>. 
2020; 94(5): 1-26.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>g &lt;- rbinom(1500, 2, 0.3)
y &lt;- 0.5 + 0.25 * g + rgev(1500, 0, 0, 5)
npt(y, g, 0.5)
</code></pre>

<hr>
<h2 id='pcoc'>PCoC for correcting for population stratification</h2><span id='topic+pcoc'></span>

<h3>Description</h3>

<p>Identify the clustered and continuous patterns of the genetic
variation using the PCoC, which calculates the principal
coordinates and the clustering of the subjects for correcting for
PS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pcoc(
  genoFile,
  outFile.txt = "pcoc.result.txt",
  n.MonteCarlo = 1000,
  num.splits = 10,
  miss.val = 9
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pcoc_+3A_genofile">genoFile</code></td>
<td>
<p>a txt file containing the genotypes (0, 1, 2, or
9). The element of the file in Row <em>i</em> and Column <em>j</em>
represents the genotype at the <em>i</em>th marker of the <em>j</em>th
subject. 0, 1, and 2 denote the number of risk alleles, and 9
(default) is for the missing genotype.</p>
</td></tr>
<tr><td><code id="pcoc_+3A_outfile.txt">outFile.txt</code></td>
<td>
<p>a txt file for saving the result of this
function. The default is &quot;<code>pcoc.result.txt</code>&quot;.</p>
</td></tr>
<tr><td><code id="pcoc_+3A_n.montecarlo">n.MonteCarlo</code></td>
<td>
<p>the number of times for the Monte Carlo
procedure. The default is <code>1000</code>.</p>
</td></tr>
<tr><td><code id="pcoc_+3A_num.splits">num.splits</code></td>
<td>
<p>the number of groups into which the markers are
split. The default is <code>10</code>.</p>
</td></tr>
<tr><td><code id="pcoc_+3A_miss.val">miss.val</code></td>
<td>
<p>the number representing the missing data in the
input data. The default is <code>9</code>. The element 9 for the missing data
in the <code>genoFile</code> should be changed according to the value of
<code>miss.val</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The hidden population structure is a possible confounding effect
in the large-scale genome-wide association studies. Cases and
controls might have systematic differences because of the
unrecognized population structure. The PCoC procedure uses the
techniques from the multidimensional scaling and the clustering to
correct for the population stratification. The PCoC could be seen
as an extension of the EIGENSTRAT.
</p>


<h3>Value</h3>

<p>A list of <code>principal.coordinates</code> and
<code>cluster</code>. <code>principal.coordinates</code> is the principal
coordinates and <code>cluster</code> is the clustering of the
subjects. If the number of clusters is only one,
<code>cluster</code> is omitted.
</p>


<h3>Author(s)</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li.
</p>


<h3>References</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li. AssocTests: An R Package 
for Genetic Association Studies. <em>Journal of Statistical Software</em>. 
2020; 94(5): 1-26.
</p>
<p>Q Li and K Yu. Improved Correction for Population
Stratification in Genome-Wide Association Studies by Identifying
Hidden Population Structures. <em>Genetic Epidemiology</em>. 2008;
32(3): 215-226.
</p>
<p>KV Mardia, JT Kent, and JM Bibby. Multivariate
Analysis. <em>New York: Academic Press</em>. 1976.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pcocG.eg &lt;- matrix(rbinom(4000, 2, 0.5), ncol = 40)
write.table(pcocG.eg, file = "pcocG.eg.txt", quote = FALSE,
       sep = "", row.names = FALSE, col.names = FALSE)
pcoc(genoFile = "pcocG.eg.txt", outFile.txt = "pcoc.result.txt",
       n.MonteCarlo = 50, num.splits = 10, miss.val = 9)
file.remove("pcocG.eg.txt", "pcoc.result.txt")
</code></pre>

<hr>
<h2 id='smt'>Single-marker test</h2><span id='topic+smt'></span>

<h3>Description</h3>

<p>Conduct the single-marker test in an association study to test for
the association between the genotype at a biallelic marker and a
trait.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>smt(
  y,
  g,
  covariates = NULL,
  min.count = 5,
  missing.rate = 0.2,
  y.continuous = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="smt_+3A_y">y</code></td>
<td>
<p>a numeric vector of the observed trait values in which
the <em>i</em>th element is for the <em>i</em>th subject. The elements
could be discrete (<code>0</code> or <code>1</code>) or continuous. The missing value is
represented by <code>NA</code>.</p>
</td></tr>
<tr><td><code id="smt_+3A_g">g</code></td>
<td>
<p>a numeric vector of the observed genotype values (<code>0</code>, <code>1</code>,
or <code>2</code> denotes the number of risk alleles) in which the <em>i</em>th
element is for the <em>i</em>th subject. The missing value is
represented by <code>NA</code>. <code>g</code> has the same length as <code>y</code>.</p>
</td></tr>
<tr><td><code id="smt_+3A_covariates">covariates</code></td>
<td>
<p>an optional data frame, list or environment
containing the covariates used in the model. The default is <code>NULL</code>,
that is, there are no covariates.</p>
</td></tr>
<tr><td><code id="smt_+3A_min.count">min.count</code></td>
<td>
<p>a critical value to decide which method is used
to calculate the p-value when the trait is discrete and <code>covariates
= NULL</code>. If the minimum number of the elements given a specific
trait value and a specific genotype value is less than
<code>min.count</code>, the Fisher's exact test is adopted; otherwise, the
Wald test is adopted. The default is <code>5</code>.</p>
</td></tr>
<tr><td><code id="smt_+3A_missing.rate">missing.rate</code></td>
<td>
<p>the highest missing value rate of the genotype
values that this function can tolerate. The default is <code>0.2</code>.</p>
</td></tr>
<tr><td><code id="smt_+3A_y.continuous">y.continuous</code></td>
<td>
<p>logical. If <code>TRUE</code>, <code>y</code> is continuous;
otherwise, <code>y</code> is discrete. The default is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Single-marker analysis is a core in many gene-based or
pathway-based procedures, such as the truncated p-value
combination and the minimal p-value.
</p>


<h3>Value</h3>

<p><code>smt</code> returns a list with class &quot;<code>htest</code>&quot;.
</p>
<p>If y is continuous, the list contains the following components:
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>statistic</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the observed value of the test statistic.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>p.value</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the p-value for the test.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>alternative</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string describing the alternative hypothesis.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string indicating the type of test performed.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>data.name</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string giving the names of the data. </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sample.size</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a vector giving the numbers of the subjects with the genotypes <code>0</code>, <code>1</code>, and <code>2</code> (<code>n0</code>, </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>n1</code>, and <code>n2</code>, respectively).
</td>
</tr>

</table>

<p>If y is discrete, the list contains the following components:
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>statistic</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the observed value of the test statistic.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>p.value</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the p-value for the test.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>alternative</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string describing the alternative hypothesis.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string indicating the type of test performed.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>data.name</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string giving the names of the data. </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sample.size</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a vector giving </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the number of subjects with the trait value <code>1</code> and the genotype <code>0</code> (<code>r0</code>), </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the number of subjects with the trait value <code>1</code> and the genotype <code>1</code> (<code>r1</code>), </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the number of subjects with the trait value <code>1</code> and the genotype <code>2</code> (<code>r2</code>), </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the number of subjects with the trait value <code>0</code> and the genotype <code>0</code> (<code>s0</code>), </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the number of subjects with the trait value <code>0</code> and the genotype <code>1</code> (<code>s1</code>), </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> and the number of subjects with the trait value <code>0</code> and the genotype <code>2</code> (<code>s2</code>).</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>bad.obs</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a vector giving the number of missing genotype values with the trait value <code>1</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> (<code>r.miss</code>), the number of missing genotype values with the trait value <code>0</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> (<code>s.miss</code>), and the total number of the missing genotype values (<code>n.miss</code>).
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li.
</p>


<h3>References</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li. AssocTests: An R Package 
for Genetic Association Studies. <em>Journal of Statistical Software</em>. 
2020; 94(5): 1-26.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>y &lt;- rep(c(0, 1), 25)
g &lt;- sample(c(0, 1, 2), 50, replace = TRUE)
smt(y, g, covariates = NULL, min.count=5,
        missing.rate=0.20, y.continuous = FALSE)
</code></pre>

<hr>
<h2 id='tw'>Tracy-Widom test</h2><span id='topic+tw'></span>

<h3>Description</h3>

<p>Find the significant eigenvalues of a matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tw(eigenvalues, eigenL, criticalpoint = 2.0234)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tw_+3A_eigenvalues">eigenvalues</code></td>
<td>
<p>a numeric vector whose elements are the
eigenvalues of a matrix. The values should be sorted in the
descending order.</p>
</td></tr>
<tr><td><code id="tw_+3A_eigenl">eigenL</code></td>
<td>
<p>the number of eigenvalues.</p>
</td></tr>
<tr><td><code id="tw_+3A_criticalpoint">criticalpoint</code></td>
<td>
<p>a numeric value corresponding to the
significance level. If the significance level is 0.05, 0.01,
0.005, or 0.001, the <code>criticalpoint</code> should be set to be <code>0.9793</code>,
<code>2.0234</code>, <code>2.4224</code>, or <code>3.2724</code>, accordingly. The default is <code>2.0234</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with class &quot;<code>htest</code>&quot; containing the following components:
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>statistic</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a vector of the Tracy-Widom statistics.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>alternative</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string describing the alternative hypothesis.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string indicating the type of test performed.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>data.name</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> a character string giving the name of the data. </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>SigntEigenL</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> the number of significant eigenvalues.
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li.
</p>


<h3>References</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li. AssocTests: An R Package 
for Genetic Association Studies. <em>Journal of Statistical Software</em>. 
2020; 94(5): 1-26.
</p>
<p>N Patterson, AL Price, and D Reich. Population
Structure and Eigenanalysis. <em>PloS Genetics</em>. 2006; 2(12):
2074-2093.
</p>
<p>CA Tracy and H Widom. Level-Spacing Distributions and
the Airy Kernel. <em>Communications in Mathematical
Physics</em>. 1994; 159(1): 151-174.
</p>
<p>A Bejan. Tracy-Widom and Painleve II: Computational
Aspects and Realisation in S-Plus. In <em>First Workshop of the ERCIM
Working Group on Computing and Statistics</em>. 2008, Neuchatel, Switzerland.
</p>
<p>A Bejan. Largest eigenvalues and sample covariance matrices. 
<em>MSc Dissertation, the university of Warwick</em>. 2005. (This function
was written by A Bejan and publicly downloadable.)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tw(eigenvalues = c(5, 3, 1, 0), eigenL = 4, criticalpoint = 2.0234)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
