<!DOCTYPE html><html lang="en"><head><title>Help for package asymLD</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {asymLD}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#asymLD-package'><p>asymLD: a package for computing asymmetric Linkage Disequilibrium (ALD) between two polymorphic genetic loci</p></a></li>
<li><a href='#compute.ALD'><p>Compute ALD.</p></a></li>
<li><a href='#compute.AShomz'><p>Compute allele specific homozygosity.</p></a></li>
<li><a href='#hla.freqs'><p>Pairwise haplotype frequencies for 7 HLA loci</p></a></li>
<li><a href='#lsort'><p>Sort a data.frame.</p></a></li>
<li><a href='#snp.freqs'><p>Pairwise haplotype frequencies for 4 SNP loci</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Asymmetric Linkage Disequilibrium (ALD) for Polymorphic Genetic
Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-01-29</td>
</tr>
<tr>
<td>Author:</td>
<td>Richard M. Single</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Richard M. Single &lt;rsingle@uvm.edu&gt;</td>
</tr>
<tr>
<td>Suggests:</td>
<td>haplo.stats, fields</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>Yes</td>
</tr>
<tr>
<td>Description:</td>
<td>Computes asymmetric LD measures (ALD) for multi-allelic genetic data. These measures are identical to the correlation measure (r) for bi-allelic data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-01-30 15:57:20 UTC; Rich</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-01-30 19:00:02</td>
</tr>
</table>
<hr>
<h2 id='asymLD-package'>asymLD: a package for computing asymmetric Linkage Disequilibrium (ALD) between two polymorphic genetic loci</h2><span id='topic+asymLD'></span><span id='topic+asymLD-package'></span>

<h3>Description</h3>

<p>Computes asymmetric LD measures (ALD) for polymorphic genetic data.
These measures are identical to the correlation measure (r) for bi-allelic data.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
 Package: </td><td style="text-align: left;"> asymLD</td>
</tr>
<tr>
 <td style="text-align: left;">
 Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
 Version: </td><td style="text-align: left;"> 0.1</td>
</tr>
<tr>
 <td style="text-align: left;">
 Date: </td><td style="text-align: left;"> 2015-03-13</td>
</tr>
<tr>
 <td style="text-align: left;">
 License: </td><td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
 <td style="text-align: left;">
 </td>
</tr>

</table>



<h3>asymLD functions</h3>

<p>The function compute.ALD() calculates asymmetric LD for haplotype frequency data.
The function compute.AShomz() calculates allele specific homozygosity values for haplotype frequency data.
</p>


<h3>References</h3>

<p>Thomson G, Single RM. Conditional Asymmetric Linkage Disequilibrium (ALD): Extending the Bi-Allelic r^2 Measure. Genetics. 2014  198(1):321-331. PMID:25023400
</p>

<hr>
<h2 id='compute.ALD'>Compute ALD.</h2><span id='topic+compute.ALD'></span>

<h3>Description</h3>

<p>A function to compute asymmetric Linkage Disequilibrium measures (ALD) for polymorphic genetic
data. These measures are identical to the correlation measure (r) for bi-allelic data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compute.ALD(dat, tolerance = 0.01)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compute.ALD_+3A_dat">dat</code></td>
<td>
<p>A data.frame with 5 required variables (having the names listed below):
</p>

<table>
<tr>
 <td style="text-align: left;">
     <code>haplo.freq</code> </td><td style="text-align: left;"> A numeric vector of haplotype frequencies.</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>locus1</code> </td><td style="text-align: left;"> A character vector indentifying the first locus.</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>locus2</code> </td><td style="text-align: left;"> A character vector indentifying the second locus.</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>allele1</code> </td><td style="text-align: left;"> A character vector indentifying the allele at locus 1.</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>allele2</code> </td><td style="text-align: left;"> A character vector indentifying the allele at locus 2.</td>
</tr>
<tr>
 <td style="text-align: left;">
      </td>
</tr>

</table>
</td></tr>
<tr><td><code id="compute.ALD_+3A_tolerance">tolerance</code></td>
<td>
<p>A threshold for the sum of the haplotype frequencies.
If the sum of the haplotype frequencies is greater than 1+tolerance or less
than 1-tolerance an error is returned. The default is 0.01.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The return value is a dataframe with the following components:
</p>

<table>
<tr>
 <td style="text-align: left;">
 <code>locus1</code>	</td><td style="text-align: left;"> The name of the first locus.</td>
</tr>
<tr>
 <td style="text-align: left;">
 <code>locus2</code>	</td><td style="text-align: left;"> The name of the second locus.</td>
</tr>
<tr>
 <td style="text-align: left;">
 <code>F.1</code>	</td><td style="text-align: left;"> Homozygosity (expected under HWP) for locus 1.</td>
</tr>
<tr>
 <td style="text-align: left;">
 <code>F.1.2</code>	</td><td style="text-align: left;"> Conditional homozygosity* for locus1 given locus2.</td>
</tr>
<tr>
 <td style="text-align: left;">
 <code>F.2</code>	</td><td style="text-align: left;"> Homozygosity (expected under HWP) for locus 2.</td>
</tr>
<tr>
 <td style="text-align: left;">
 <code>F.2.1</code>	</td><td style="text-align: left;"> Conditional homozygosity* for locus2 given locus1.</td>
</tr>
<tr>
 <td style="text-align: left;">
 <code>ALD.1.2</code> </td><td style="text-align: left;"> Asymmetric LD for locus1 given locus2.</td>
</tr>
<tr>
 <td style="text-align: left;">
 <code>ALD.2.1</code> </td><td style="text-align: left;">  Asymmetric LD for locus2 given locus1.</td>
</tr>
<tr>
 <td style="text-align: left;">
 </td>
</tr>

</table>

<p>*Overall weighted haplotype-specific homozygosity for the first locus given the second locus.
</p>


<h3>Details</h3>

<p>A warning message is given if the sum of the haplotype frequencies is greater than 1.01 or less
than 0.99 (regardless of the <code>tolerance</code> setting). The haplotype frequencies that are passed
to the function are normalized within the function to sum to 1.0 by dividing each frequency by
the sum of the passed frequencies.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(asymLD)

# An example using haplotype frequencies from Wilson(2010)
data(hla.freqs)
hla.a_b &lt;- hla.freqs[hla.freqs$locus1=="A" &amp; hla.freqs$locus2=="B",]
compute.ALD(hla.a_b)
hla.freqs$locus &lt;- paste(hla.freqs$locus1, hla.freqs$locus2, sep="-")
compute.ALD(hla.freqs[hla.freqs$locus=="C-B",])
# Note: additonal columns on the input dataframe (e.g., "locus" above) are allowed, but
# ignored by the function.

# An example using genotype data from the haplo.stats package
require(haplo.stats)
data(hla.demo)
geno &lt;- hla.demo[,5:8]  #DPB-DPA
label &lt;- unique(gsub(".a(1|2)", "", colnames(geno)))
label &lt;- paste("HLA*",label,sep="")
keep &lt;- !apply(is.na(geno) | geno==0, 1, any)
em.keep  &lt;- haplo.em(geno=geno[keep,], locus.label=label)
hapfreqs.df &lt;- cbind(em.keep$haplotype, em.keep$hap.prob)
#format dataframe for ALD function
names(hapfreqs.df)[dim(hapfreqs.df)[2]] &lt;- "haplo.freq"
names(hapfreqs.df)[1] &lt;- "allele1"
names(hapfreqs.df)[2] &lt;- "allele2"
hapfreqs.df$locus1 &lt;- label[1]
hapfreqs.df$locus2 &lt;- label[2]
head(hapfreqs.df)
compute.ALD(hapfreqs.df)
# Note that there is substantially less variablity (higher ALD) for HLA*DPA1
# conditional on HLA*DPB1 than for HLA*DPB1 conditional on HLA*DPA1, indicating
# that the overall variation for DPA1 is relatively low given specific DPB1 alleles

# An example using SNP data where results are symmetric and equal to the ordinary
# correlation measure (r)
data(snp.freqs)
snps &lt;- c("rs1548306", "rs6923504", "rs4434496", "rs7766854")
compute.ALD(snp.freqs[snp.freqs$locus1==snps[2] &amp; snp.freqs$locus2==snps[3],])

snp.freqs$locus &lt;- paste(snp.freqs$locus1, snp.freqs$locus2, sep="-")
by(snp.freqs,list(snp.freqs$locus),compute.ALD)

# SNP1 &amp; SNP2 : the r correlation &amp; ALD measures are equivalent due to symmetry for
# bi-allelic SNPs
p.AB &lt;- snp.freqs$haplo.freq[1]
p.Ab &lt;- snp.freqs$haplo.freq[2]
p.aB &lt;- snp.freqs$haplo.freq[3]
p.ab &lt;- snp.freqs$haplo.freq[4]
p.A &lt;- p.AB + p.Ab
p.B &lt;- p.AB + p.aB
r.squared &lt;- (p.AB - p.A*p.B)^2 / (p.A*(1-p.A)*p.B*(1-p.B))
sqrt(r.squared) #the r correlation measure
compute.ALD(snp.freqs[snp.freqs$locus1==snps[1] &amp; snp.freqs$locus2==snps[2],])
</code></pre>

<hr>
<h2 id='compute.AShomz'>Compute allele specific homozygosity.</h2><span id='topic+compute.AShomz'></span>

<h3>Description</h3>

<p>A function to compute allele specific homozygosity values for haplotype frequency data.
The allele specific homozygosity is the homozygosity statistic computed for alleles at
one locus that are found on haplotypes with a specific allele (the focal allele) at the
other locus (the focal locus).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compute.AShomz(dat, tolerance = 0.01, sort.var = c("focal", "allele"),
  sort.asc = rep(TRUE, length(sort.var)))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compute.AShomz_+3A_dat">dat</code></td>
<td>
<p>A data.frame with 5 required variables (having the names listed below):
</p>

<table>
<tr>
 <td style="text-align: left;">
     <code>haplo.freq</code> </td><td style="text-align: left;"> A numeric vector of haplotype frequencies.</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>locus1</code> </td><td style="text-align: left;"> A character vector indentifying the first locus.</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>locus2</code> </td><td style="text-align: left;"> A character vector indentifying the second locus.</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>allele1</code> </td><td style="text-align: left;"> A character vector indentifying the allele at locus 1.</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>allele2</code> </td><td style="text-align: left;"> A character vector indentifying the allele at locus 1.</td>
</tr>
<tr>
 <td style="text-align: left;">
      </td>
</tr>

</table>
</td></tr>
<tr><td><code id="compute.AShomz_+3A_tolerance">tolerance</code></td>
<td>
<p>A threshold for the sum of the haplotype frequencies.
If the sum of the haplotype frequencies is greater than 1+tolerance or less
than 1-tolerance an error is returned. The default is 0.01.</p>
</td></tr>
<tr><td><code id="compute.AShomz_+3A_sort.var">sort.var</code></td>
<td>
<p>a vector of variable names specifying the &quot;sort by&quot; variables.
The default is c(&quot;focal&quot;,&quot;allele&quot;).</p>
</td></tr>
<tr><td><code id="compute.AShomz_+3A_sort.asc">sort.asc</code></td>
<td>
<p>a vector of TRUE/FALSE values, with the same length as &quot;sort.var&quot;,
indicating whether sorting of each variable is in ascending order.
The default order is ascending.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The return value is a dataframe with the following components:
</p>

<table>
<tr>
 <td style="text-align: left;">
 <code>loci</code>	</td><td style="text-align: left;"> The locus names separated by "-".</td>
</tr>
<tr>
 <td style="text-align: left;">
 <code>focal</code>	</td><td style="text-align: left;"> The name of the focal locus (locus conditioned on).</td>
</tr>
<tr>
 <td style="text-align: left;">
 <code>allele</code>	</td><td style="text-align: left;"> The name of the focal allele (allele conditioned on).</td>
</tr>
<tr>
 <td style="text-align: left;">
 <code>allele.freq</code>	</td><td style="text-align: left;"> The frequency of the focal allele.</td>
</tr>
<tr>
 <td style="text-align: left;">
 <code>as.homz</code>	</td><td style="text-align: left;"> The allele specific homozygosity (on haplotypes with the focal allele).</td>
</tr>
<tr>
 <td style="text-align: left;">
 </td>
</tr>

</table>



<h3>Details</h3>

<p>A warning message is given if the sum of the haplotype frequencies is greater than 1.01 or less
than 0.99 (regardless of the <code>tolerance</code> setting). The haplotype frequencies that are
passed to the function are normalized within the function to sum to 1.0 by dividing each
frequency by the sum of the passed frequencies.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(asymLD)

# An example using haplotype frequencies from Wilson(2010)
data(hla.freqs)
hla.dr_dq &lt;- hla.freqs[hla.freqs$locus1=="DRB1" &amp; hla.freqs$locus2=="DQB1",]
compute.ALD(hla.dr_dq)
compute.AShomz(hla.dr_dq, sort.var=c("focal","allele"), sort.asc=c(TRUE,TRUE))
compute.AShomz(hla.dr_dq, sort.var=c("focal","allele.freq"), sort.asc=c(FALSE,FALSE))
# Note that there is substantially less variablity (higher ALD) for HLA*DQB1
# conditional on HLA*DRB1 than for HLA*DRB1 conditional on HLA*DQB1, indicating
# that the overall variation for DQB1 is relatively low given specific DRB1 alleles.
# The largest contributors to ALD{DQB1|DRB1} are the DRB1*0301 and DRB1*1501 focal
# alleles, which have high allele frequencies and also have high allele specific
# homozygosity values.
</code></pre>

<hr>
<h2 id='hla.freqs'>Pairwise haplotype frequencies for 7 HLA loci</h2><span id='topic+hla.freqs'></span>

<h3>Description</h3>

<p>HLA haplotype frequencies for 21 pairs of HLA loci in a set of 300 controls
from a study of myopericarditis incidence following smallpox vaccination.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hla.freqs)
</code></pre>


<h3>Format</h3>

<p>A data frame with 3063 observations on the following 5 variables.
</p>

<dl>
<dt><code>haplo.freq</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>locus1</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>locus2</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>allele1</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>allele2</code></dt><dd><p>a character vector</p>
</dd>
</dl>


<h3>Source</h3>

<p>Wilson, C., 2010 Identifying polymorphisms associated with risk for the development of myopericarditis following smallpox vaccine. The Immunology Database and Analysis Portal (ImmPort), Study #26.
</p>


<h3>References</h3>

<p>https://immport.niaid.nih.gov/immportWeb/clinical/study/displayStudyDetails.do?itemList=SDY26
</p>

<hr>
<h2 id='lsort'>Sort a data.frame.</h2><span id='topic+lsort'></span>

<h3>Description</h3>

<p>A function to sort a data.frame on specific columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lsort(dat, by = 1:dim(dat)[2], asc = rep(TRUE, length(by)),
  na.last = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lsort_+3A_dat">dat</code></td>
<td>
<p>a dataframe or a matrix (&quot;dimnames&quot; are used as the variable names for a matrix)</p>
</td></tr>
<tr><td><code id="lsort_+3A_by">by</code></td>
<td>
<p>a vector or a list of variable names or column indices specifying the &quot;sort by&quot;
variables, the default is to sort by all variables in the order they appear in the data set.</p>
</td></tr>
<tr><td><code id="lsort_+3A_asc">asc</code></td>
<td>
<p>a vector with the same length as &quot;by&quot; indicating whether the sorting of each &quot;by&quot;
variable is in ascending order, the default order is ascending.</p>
</td></tr>
<tr><td><code id="lsort_+3A_na.last">na.last</code></td>
<td>
<p>a flag indicating whether missing values are placed as the last
elements in the data set, the default is TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The return value is a sorted dataframe.
</p>


<h3>Details</h3>

<p>The input dataframe is not modified. The code is adapted from code posted to an old s-news
listserve.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(asymLD)
data(snp.freqs)

# sort snp.freqs by "locus1" (ascending) and "allele1" (descending)
newdata &lt;- lsort(snp.freqs, by=c("locus1","allele1"), asc=c(T,F))
head(newdata)
# sort snp.freqs by the fourth and the second variable (ascending)
newdata &lt;- lsort(snp.freqs, by=c(4,2))
# sort "snp.freqs" by "locus1" and the 5th variable (ascending)
newdata &lt;- lsort(snp.freqs, by=list("locus1",5))


## End(Not run)
</code></pre>

<hr>
<h2 id='snp.freqs'>Pairwise haplotype frequencies for 4 SNP loci</h2><span id='topic+snp.freqs'></span>

<h3>Description</h3>

<p>HLA haplotype frequencies for 6 pairs of SNP loci from the de Bakker et al. 2006 data for 90 unrelated individuals with European ancestry (CEU) from the Centre d'Etude du Polymorphisme Humain (CEPH)
collection obtained from the Tagger/MHC webpage.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(snp.freqs)
</code></pre>


<h3>Format</h3>

<p>A data frame with 20 observations on the following 5 variables.
</p>

<dl>
<dt><code>locus1</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>locus2</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>allele1</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>allele2</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>haplo.freq</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>


<h3>Details</h3>

<p>For bi-allelic SNP data the ALD measures are symmetric and equivalent to the r correlation measure of LD.
</p>


<h3>Source</h3>

<p>de Bakker, P. I., G. McVean, P. C. Sabeti, M. M. Miretti, T. Green et al., 2006 A high-resolution HLA and SNP haplotype map for disease association studies in the extended human MHC. Nat.Genet. 38: 1166-1172.
</p>


<h3>References</h3>

<p>http://www.broadinstitute.org/mpg/tagger/mhc.html
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
