<!DOCTYPE html><html><head><title>Help for package augSIMEX</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {augSIMEX}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#augSIMEX-package'>
<p>R package for Analysis of Data with Mixed Measurement Error and Misclassification in Covariates</p></a></li>
<li><a href='#augSIMEX'>
<p>Analysis of Data with Mixed Measurement Error and Misclassification in Covariates</p></a></li>
<li><a href='#coef.augSIMEX'>
<p>Extract the coefficients from the fitted augSIMEX object</p></a></li>
<li><a href='#GeneRepeat'><p>Example data for univariate error-prone covariates in repeated measurements case</p></a></li>
<li><a href='#GeneUni'><p>Example of genetic data for univariate error-prone covariates</p></a></li>
<li><a href='#glmscore'>
<p>Score Value in Generalized Linear Model</p></a></li>
<li><a href='#logLik.augSIMEX'>
<p>Extract the likelihood from the fitted augSIMEX object</p></a></li>
<li><a href='#plot.augSIMEX'>
<p>Plot of Extrapolation</p></a></li>
<li><a href='#predict.augSIMEX'>
<p>Predict Method for the model fits by augSIMEX</p></a></li>
<li><a href='#residuals.augSIMEX'>
<p>Residuals of the Fits Made by augSIMEX</p></a></li>
<li><a href='#summary.augSIMEX'>
<p>Summarizing the Adjusted Fits of Generalized Linear Model</p></a></li>
<li><a href='#ToyMult'><p>Toy example data for multivariate error-prone covariates</p></a></li>
<li><a href='#ToyRepeat'><p>Toy example data for univariate error-prone covariates in repeated measurements case</p></a></li>
<li><a href='#ToyUni'><p>Toy example data for univariate error-prone covariates</p></a></li>
<li><a href='#vcov.augSIMEX'>
<p>Extract the variance-covariance matrix from the fitted augSIMEX object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Analysis of Data with Mixed Measurement Error and
Misclassification in Covariates</td>
</tr>
<tr>
<td>Version:</td>
<td>3.7.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-04-23</td>
</tr>
<tr>
<td>Author:</td>
<td>Qihuang Zhang &lt;qihuang.zhang@uwaterloo.ca&gt;, Grace Y. Yi &lt;yyi@uwaterloo.ca&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Qihuang Zhang &lt;qihuang.zhang@uwaterloo.ca&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implementation of the augmented
            Simulation-Extrapolation (SIMEX) algorithm proposed by Yi et al. (2015) &lt;<a href="https://doi.org/10.1080%2F01621459.2014.922777">doi:10.1080/01621459.2014.922777</a>&gt;
            for analyzing the data with mixed measurement error and misclassification. The main
            function provides a similar summary output as that of glm() function. Both parametric and
            empirical SIMEX are considered in the package.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.12.11), MASS, stats, graphics, Formula, nleqslv</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-04-23 13:33:47 UTC; Qihuang</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-04-23 13:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='augSIMEX-package'>
R package for Analysis of Data with Mixed Measurement Error and Misclassification in Covariates
</h2><span id='topic+augSIMEX-package'></span>

<h3>Description</h3>

<p>Implementation of the augmented SIMEX algorithm for data with mixed measurement error and misclassification in covariates. The package is allowing for both parametric SIMEX and empirical SIMEX.
</p>


<h3>Details</h3>

<p>The <code>SIMEX</code> package implements the method developed by Yi et al. (2015) to adjust for both measurement error and misclassification in the generalized linear models. The main function is <code><a href="#topic+augSIMEX">augSIMEX</a></code> which returns an &quot;augSIMEX&quot; object. The user can summarize the returned augSIMEX object in a similar format as in <code><a href="stats.html#topic+glm">glm</a></code>. The extrapolation can also be visualized through <code>plot</code> function. Other implementation tools are also provided in the package.
</p>


<h3>Author(s)</h3>

<p>Qihuang Zhang and Grace Y. Yi.
</p>
<p>Maintainer: Qihuang Zhang &lt;qihuang.zhang@uwaterloo.ca&gt;
</p>


<h3>References</h3>

<p>Yi G Y, Ma Y, Spiegelman D, et al. Functional and structural methods with mixed measurement error and misclassification in covariates[J]. Journal of the American Statistical Association, 2015, 110(510): 681-696.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+augSIMEX">augSIMEX</a></code>
</p>

<hr>
<h2 id='augSIMEX'>
Analysis of Data with Mixed Measurement Error and Misclassification in Covariates
</h2><span id='topic+augSIMEX'></span>

<h3>Description</h3>

<p>Implementation of the SIMEX algorithm for data with mixed measurement error and misclassification in covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>augSIMEX(mainformula = formula(data), mismodel = pi | qi ~ 1,
                 meformula = NULL, family = gaussian, data,
                 validationdata, err.var, mis.var, err.true, mis.true,
                 err.mat = NULL, cppmethod = TRUE, repeated = FALSE,
                 repind = list(), subset, offset, weights, na.action,
                 scorefunction = NULL, lambda = NULL, M = 5, B = 20,
                 nBoot = 50, extrapolation = c("quadratic", "linear"), bound = 8,
                 initial = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="augSIMEX_+3A_mainformula">mainformula</code></td>
<td>

<p>an object of class &ldquo;formula&rdquo;: an object of class &ldquo;formula&rdquo;: a symbolic description of the model of the response variable, error-prone covariates, and other covariates.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_mismodel">mismodel</code></td>
<td>

<p>an object of class &ldquo;Formula&rdquo;. A symbolic description in modeling the misclassification rates. See details for the specification of the model.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_meformula">meformula</code></td>
<td>

<p>an object of &ldquo;Formula&rdquo; specifying the measurement error model for each error-prone covariate. The number of responses should equal the number of error-prone covariates. The default choice will be classic additive models. See details for the specification of the model.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_family">family</code></td>
<td>

<p>an object of class &ldquo;family&rdquo; (same as in <code><a href="stats.html#topic+glm">glm</a></code> function in stats package).
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_data">data</code></td>
<td>

<p>a data frame or a matrix of main data. The variable in the main data includes the response, observed covariates matrix which is subject to measurement error, observed binary covariate vector that is prone to misclassification and may also contain the precisely measured covariates matrix. The default choice includes all covariates that mentioned in the <code>mainformula</code>.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_validationdata">validationdata</code></td>
<td>

<p>a data frame or matrix of validation data. The variable in the model includes an observed covariates matrix of that is subject to measurement error and their corresponding precisely measured covariates, an observed binary covariate vector that is prone to misclassification and their corresponding precisely measured covariates, a covariates matrix of precisely measured covariates. 
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_err.var">err.var</code></td>
<td>

<p>a vector of character specifying the name of covariates that are subject to measurement error.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_mis.var">mis.var</code></td>
<td>

<p>a string specifying the variable name of the binary variable subject to misclassification.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_err.true">err.true</code></td>
<td>

<p>a vector of character specifying the names of the all the counterparts of <code>err.var</code> that are precisely measured covariates. 
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_mis.true">mis.true</code></td>
<td>

<p>a string specifying the variable name of the precisely measured binary variable.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_err.mat">err.mat</code></td>
<td>

<p>a matrix indicating the variance-covariance matrix of generated Normal distribution in simulation step. 
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_cppmethod">cppmethod</code></td>
<td>

<p>a logical value indicating whether solving the score function via C++ functions. The function involves <code>Rcpp</code> package. The C++ based method is much faster and more computationally efficient than the R based method. See the discussion section.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_repeated">repeated</code></td>
<td>

<p>a logic value indicating whether repeated measurements are involved.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_repind">repind</code></td>
<td>

<p>a list of vectors of repeated measurement for error-prone covariates. See detail below.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_subset">subset</code></td>
<td>

<p>a logical vector indicating which subjects should be included in the fitting.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_offset">offset</code></td>
<td>

<p>a numeric vector indicating the offset. The default choice is null.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_weights">weights</code></td>
<td>

<p>an optional numeric vector of the weights should be used in the fitting.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_na.action">na.action</code></td>
<td>

<p>a function indicating what method should be applied to deal with missing data.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_scorefunction">scorefunction</code></td>
<td>

<p>a function of score function. To allow for the generality, the users can specify their own score function. It should be a function of parameters, response, covariates, weights and offset. A matrix of score value for each individual and each parameter should be returned in the function. An example is shown in <code><a href="#topic+glmscore">glmscore</a></code>
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_lambda">lambda</code></td>
<td>

<p>if the <code>M</code> value is not specified, the user can specify a positive sequence of lambda directly. The first element is usually set to be 0.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_m">M</code></td>
<td>

<p>the number of predetermined lambda vector if <code>lambda</code> is missing. The default is chosen to be 50.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_b">B</code></td>
<td>

<p>the number of the dataset generated in the simulation step. The default value is set to be 200.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_nboot">nBoot</code></td>
<td>

<p>the number of the iterations repeated in bootstrap. The default value is set to be 100.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_extrapolation">extrapolation</code></td>
<td>

<p>specifies the regression model that involves in the extrapolation step. The options include &ldquo;linear&rdquo; and &ldquo;quadratic&rdquo;. The default is set to be &ldquo;quadratic&rdquo;.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_bound">bound</code></td>
<td>
<p> a value or vector specififying the bound for the absolute value of the regression coefficients. During the simulation, the parameters out of bound will be filter out before extrapolation step.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_initial">initial</code></td>
<td>
<p> the initial value of the parameters.
</p>
</td></tr>
<tr><td><code id="augSIMEX_+3A_...">...</code></td>
<td>

<p>other arguments that pass into the function.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The misclassification models are set in &quot;Formula&quot; format, where the misclassification rates for both classes of the binary variable is set simultaneously. The left-hand side sets the responses of the misclassification model. The response should always set be to pi (indicating pimodel) and qi (indicating qimodel), separated by &quot;|&quot;. On the right hand side of the formula sets the covariates of the misclassification model. If the covariates for both models are different, use &quot;|&quot; to separate them in the same order as the response. See example.
</p>
<p>The measurement error models are also set in &quot;Formula&quot; format. The left-hand side sets the responses of the measurement error model, which should be consistent with the specification of <code>err.var</code>. Each response is separated by &quot;|&quot;. On the right-hand side of the formula, the covariates are set. If the covariates for each response are different, use &quot;|&quot; to separate the specifications of covariates and the order should correspond to that of the responses of the measurement error model. See example.
</p>
<p>In the case of repeated measurements, the users can pick one of the measurements into the formula of the main model. If more than one covariates have multiple replicates, the users should name the vector of repeated measurements in the list of <code>repind</code> by the corresponding representative measurement in the formula of the main model. See example.
</p>
<p>The number of bootstrap repetitions should be at least 2. i.e., <code>nBoot</code>&gt;2. Otherwise, an error might occur. 
</p>
<p>The examples are mainly for illustration purpose. NA's are possible to be generated because the bootstrap simulation parameter is only set as 2. To obtain precise results, simulation parameters are supposed to be set on a larger scale, which also involves more time for computing. 
</p>


<h3>Value</h3>

<table>
<tr><td><code>coefficients</code></td>
<td>
<p>the coefficient of the main model after correction.</p>
</td></tr>
<tr><td><code>vcov</code></td>
<td>
<p>a adjusted variance-covariance matrix estimated by bootstrap.</p>
</td></tr>
<tr><td><code>fitted.values</code></td>
<td>
<p>the fitted mean values, obtained by transforming the linear predictors by the inverse of the link function.</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>a vector of values that involved in extrapolation step.</p>
</td></tr>
<tr><td><code>coefmatrix</code></td>
<td>
<p>a matrix of coefficient before doing extrapolation. Each row correspond to a specified lambda. Each column represent a component of coefficient. This output if for the plotting purpose.</p>
</td></tr>
<tr><td><code>deviance</code></td>
<td>
<p>minus twice the maximized log-likelihood.</p>
</td></tr>
<tr><td><code>null.deviance</code></td>
<td>
<p>the deviance of null model which only includes intercept and offset term. It is comparable to the deviance. </p>
</td></tr>
<tr><td><code>df.residual</code></td>
<td>
<p>the degree of freedom.</p>
</td></tr>
<tr><td><code>df.null</code></td>
<td>
<p>the degree of freedom of null model.</p>
</td></tr>
<tr><td><code>aic</code></td>
<td>
<p>Akaike's Information Criterion, minus twice the maximized log-likelihood plus twice the number of parameters.</p>
</td></tr>
<tr><td><code>...</code></td>
<td>
<p>Other components are the arguments that have been used in the function.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Qihuang Zhang and Grace Y. Yi
</p>


<h3>References</h3>

<p>Yi G Y, Ma Y, Spiegelman D, et al. Functional and structural methods with mixed measurement error and misclassification in covariates[J]. Journal of the American Statistical Association, 2015, 110(510): 681-696.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+glm">glm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Example 1: Univariate Case
data(ToyUni)
example1&lt;-augSIMEX(mainformula = Y ~ Xstar + Zstar + W, family = binomial(link = logit),
  mismodel = pi|qi ~ W, 
  meformula = Xstar ~ X + Z + W,
  data = ToyUni$Main,validationdata = ToyUni$Validation, subset = NULL,
  err.var = "Xstar", mis.var = "Zstar", err.true = "X", mis.true = "Z", 
  err.mat = NULL,
  lambda = NULL, M = 5, B = 2, nBoot = 2, extrapolation="quadratic")                   
summary(example1)

## Without adjustment
example1_naive &lt;- glm(formula = Y ~ Xstar + Zstar + W,
family = binomial(link = logit),data = ToyUni$Main)
summary(example1_naive)

## using  accurate data
example1_true &lt;- glm(Y~Xstar+Zstar+W, family = binomial(link=logit), 
   data=ToyUni$True)


### Example 2: Multivariate Case
data(ToyMult)
ErrorFormula&lt;-Xstar.X1|Xstar.X2~-1+X.X1|-1+X.X2   ## measurement error model
example2&lt;-augSIMEX(mainformula = Y~Xstar.X1+Xstar.X2+Zstar+W.W1+W.W2, 
  mismodel=pi|qi~X.X1+X.X2+W.W1+W.W2, family = binomial,
  data=ToyMult$Main,meformula=ErrorFormula,
  validationdata=ToyMult$Validation, subset=NULL,
  err.var=c("Xstar.X1","Xstar.X2"), mis.var="Zstar", err.true=c("X.X1","X.X2"), 
  mis.true="Z", err.mat = NULL,
  lambda=NULL, M=5, B=2, nBoot=2, extrapolation="quadratic")
summary(example2)


### Example 3
data(ToyRepeat)
example3&lt;-augSIMEX(mainformula = Y~Xstar1+Zstar+W, family = binomial(link=logit),
  mismodel = pi|qi ~ W, meformula = Xstar ~ X + Z + W,
  data=ToyRepeat$Main,validationdata=ToyRepeat$Validation, 
  subset=NULL, err.var="Xstar1", mis.var="Zstar", err.true="X", mis.true="Z", 
  err.mat = NULL, repeated = TRUE,repind=list(Xstar1=c("Xstar1","Xstar2")),
  lambda=NULL, M=5, B=2, nBoot=2, extrapolation="quadratic")
summary(example3)
</code></pre>

<hr>
<h2 id='coef.augSIMEX'>
Extract the coefficients from the fitted augSIMEX object
</h2><span id='topic+coef.augSIMEX'></span>

<h3>Description</h3>

<p>This function returns the vector of fitted coefficients from a fitted augSIMEX object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'augSIMEX'

## S3 method for class 'augSIMEX'
coef(object, ...)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coef.augSIMEX_+3A_object">object</code></td>
<td>

<p>the &ldquo;augSIMEX&quot; object gotten from <code><a href="#topic+augSIMEX">augSIMEX</a></code> function.
</p>
</td></tr>
<tr><td><code id="coef.augSIMEX_+3A_...">...</code></td>
<td>
<p> other arguments to pass to the function.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Qihuang Zhang and Grace Y. Yi.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+coef">coef</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ToyUni)
example &lt;- augSIMEX(mainformula = Y ~ Xstar + Zstar + W, family = binomial(link = logit),
  mismodel = pi|qi ~ W, 
  meformula = Xstar ~ X + Z + W,
  data = ToyUni$Main,validationdata = ToyUni$Validation, subset = NULL,
  err.var = "Xstar", mis.var = "Zstar", err.true = "X", mis.true = "Z", 
  err.mat = NULL,
  lambda = NULL, M = 5, B = 2, nBoot = 2, extrapolation="quadratic")
coef(example)
</code></pre>

<hr>
<h2 id='GeneRepeat'>Example data for univariate error-prone covariates in repeated measurements case</h2><span id='topic+GeneRepeat'></span>

<h3>Description</h3>

<p>This dataset gives an example data for the illustration of usage of <code><a href="#topic+augSIMEX">augSIMEX</a></code> function. The data set is adapted from the outbred Carworth Farms White mice data (Parker et. al., 2016). The dataset contains main data and validation data. We are interested to study the association between the response genotype of rs223979909 and the locomotor response to methamphetamine injections, which is subject to mismeasurement. 
</p>
<p>Tibia_5 - Tibia_30: the repeated measurement of tibia length
Tibia_M: the error-prone measurement of tibia length
Batch_T: the precisely labeled batch effect
Batch_O: the observed batch effect
Weight:  the body weights of the mice
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GeneRepeat)</code></pre>


<h3>Format</h3>

<p>A list of two data frames. Main data (rs38916331, Totdist5,  Totdist10, Totdist15, Totdist20, Totdist25, Totdist30, Weight, Batch_O) and validation data (Batch_T, Batch_O).</p>


<h3>References</h3>

<p>Parker C C, Gopalakrishnan S, Carbonetto P, et al. Genome-wide association study of behavioral, physiological and gene expression traits in outbred CFW mice[J]. Nature genetics, 2016, 48(8): 919.
</p>

<hr>
<h2 id='GeneUni'>Example of genetic data for univariate error-prone covariates</h2><span id='topic+GeneUni'></span>

<h3>Description</h3>

<p>This data set gives an example data for the illustration of usage of <code><a href="#topic+augSIMEX">augSIMEX</a></code> function. The data set is adapted from the outbred Carworth Farms White mice data (Parker et. al., 2016). The dataset contains main data and validation data. We are interested to study the association between the response genotype of rs38916331 and the tibia length, which is subject to mismeasurement. 
</p>
<p>Tibia_T: the precise measurement of tibia length
Tibia_M: the error-prone measurement of tibia length
Batch_T: the precisely labeled batch effect
Batch_O: the observed batch effect
Weight:  the body weights of the mice
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GeneUni)</code></pre>


<h3>Format</h3>

<p>A list of two data frames. Main data (rs38916331, Weight, Tibia_M, Batch_O) and validation data (Tibia_T, Batch_T, Weight, Tibia_M, Batch_O). </p>


<h3>References</h3>

<p>Parker C C, Gopalakrishnan S, Carbonetto P, et al. Genome-wide association study of behavioral, physiological and gene expression traits in outbred CFW mice[J]. Nature genetics, 2016, 48(8): 919.
</p>

<hr>
<h2 id='glmscore'>
Score Value in Generalized Linear Model
</h2><span id='topic+glmscore'></span>

<h3>Description</h3>

<p>This function can be used to calculate the value of score function. This function can serve as a tool to assist the users in debugging their self-defined score function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glmscore(beta, Y, data, weight, offset, family)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="glmscore_+3A_beta">beta</code></td>
<td>

<p>a vector specifying the value of parameters.
</p>
</td></tr>
<tr><td><code id="glmscore_+3A_y">Y</code></td>
<td>

<p>a vector specifying the response.
</p>
</td></tr>
<tr><td><code id="glmscore_+3A_data">data</code></td>
<td>

<p>a matrix or data frame of covariates. If this is for score function debugging purpose, see detail for the rearrangement of the dataset.
</p>
</td></tr>
<tr><td><code id="glmscore_+3A_weight">weight</code></td>
<td>

<p>a vector of weight.
</p>
</td></tr>
<tr><td><code id="glmscore_+3A_offset">offset</code></td>
<td>

<p>a vector of offset.
</p>
</td></tr>
<tr><td><code id="glmscore_+3A_family">family</code></td>
<td>

<p>a <code><a href="stats.html#topic+glm">glm</a></code> &ldquo;family&rdquo; object.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In a general setting, Y can be treated as a response and data as covariates. 
</p>
<p>This function helps to debug the user specified score function. The data are needed to be specially arranged such that the score function output can be successfully passed into <code>augSIMEX</code> function in a correct order. The data should be a matrix or data frame of a combination with error-prone covariates (in the order of <code>err.var</code> as in <code><a href="#topic+augSIMEX">augSIMEX</a></code>), other covariates and binary variable that is prone to misclassification. The intercept, if considered, should be included the category of &ldquo;other covariates&rdquo;.
</p>


<h3>Value</h3>

<p>A matrix of n times m, where n is the number of observations and m is the number of parameters. Each entry in the matrix represents the calculated score value for subject <code class="reqn">i</code> on parameter <code class="reqn">j</code>. The vector of row sum will be the score value vector.
</p>


<h3>Author(s)</h3>

<p>Qihuang Zhang and Grace Y. Yi.
</p>


<h3>References</h3>

<p>McCullagh P. Generalized linear models[J]. European Journal of Operational Research, 1984, 16(3): 285-292.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+glm">glm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### The user specified function to be checked. (logit link in binomial family)
scorefunction=function(beta,Y,data,weight,offset){
  results&lt;-lapply(1:dim(data)[2],
                  FUN=function(i){
                    S&lt;-lapply(1:dim(data)[1],function(x){
                      eta&lt;- matrix(beta,nrow=1) 
                      return(weight[x]*Y[x]*data[x,i]-weight[x]*exp(eta)/(1+exp(eta))*data[x,i])})
                    return(S)}
  )
  return(matrix(unlist(results),ncol=dim(data)[2]))
}

data(ToyUni)

### Data need to rearranged. See detail.
nsize&lt;-length(ToyUni$Main[,"Y"])
data.in.score&lt;-data.frame(intercept=1,X=ToyUni$Main[,"Xstar"],
                          W=ToyUni$Main[,"W"],Z=ToyUni$Main[,"Zstar"])

## compare. The results should be identical.
glmscore(rep(0,4),ToyUni$Main[,"Y"],data.in.score,rep(1,nsize),
         rep(0,nsize),family=binomial(link=logit))
scorefunction(rep(0,4),ToyUni$Main[,"Y"],data.in.score,rep(1,nsize),rep(0,nsize))
</code></pre>

<hr>
<h2 id='logLik.augSIMEX'>
Extract the likelihood from the fitted augSIMEX object
</h2><span id='topic+logLik.augSIMEX'></span>

<h3>Description</h3>

<p>This function outputs the likelihood from a fitted augSIMEX object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'augSIMEX'

## S3 method for class 'augSIMEX'
logLik(object, ...)


</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logLik.augSIMEX_+3A_object">object</code></td>
<td>

<p>the &ldquo;augSIMEX&quot; object gotten from <code><a href="#topic+augSIMEX">augSIMEX</a></code> function.
</p>
</td></tr>
<tr><td><code id="logLik.augSIMEX_+3A_...">...</code></td>
<td>
<p> other arguments to pass to the function.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Qihuang Zhang and Grace Y. Yi.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+logLik">logLik</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ToyUni)
example &lt;- augSIMEX(mainformula = Y ~ Xstar + Zstar + W, family = binomial(link = logit),
  mismodel = pi|qi ~ W, 
  meformula = Xstar ~ X + Z + W,
  data = ToyUni$Main,validationdata = ToyUni$Validation, subset = NULL,
  err.var = "Xstar", mis.var = "Zstar", err.true = "X", mis.true = "Z", 
  err.mat = NULL,
  lambda = NULL, M = 5, B = 2, nBoot = 2, extrapolation="quadratic")
logLik(example)
</code></pre>

<hr>
<h2 id='plot.augSIMEX'>
Plot of Extrapolation
</h2><span id='topic+plot.augSIMEX'></span>

<h3>Description</h3>

<p>This function visualizes the extrapolation step. It plots the simulated value for each lambda value and the curve of extrapolation. The function provide the plots for both methods simultaneously, and guides the user to choose a proper extrapolation method. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'augSIMEX'

## S3 method for class 'augSIMEX'
plot(x,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.augSIMEX_+3A_x">x</code></td>
<td>

<p>the &ldquo;augSIMEX&quot; object gotten from <code><a href="#topic+augSIMEX">augSIMEX</a></code> function.
</p>
</td></tr>
<tr><td><code id="plot.augSIMEX_+3A_...">...</code></td>
<td>

<p>other arguments that are passed into the function.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The user may need to adjust the range of y axis for a proper display.
</p>


<h3>Author(s)</h3>

<p>Qihuang Zhang and Grace Y. Yi.
</p>


<h3>See Also</h3>

<p><code><a href="graphics.html#topic+plot">plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ToyUni)
example&lt;-augSIMEX(mainformula = Y ~ Xstar + Zstar + W, family = binomial(link = logit),
  mismodel = pi|qi ~ W, 
  meformula = Xstar ~ X + Z + W,
  data = ToyUni$Main,validationdata = ToyUni$Validation, subset = NULL,
  err.var = "Xstar", mis.var = "Zstar", err.true = "X", mis.true = "Z", 
  err.mat = NULL,
  lambda = NULL, M = 5, B = 2, nBoot = 2, extrapolation="quadratic")
         
plot(example, ylim = c(-1,0.4))
</code></pre>

<hr>
<h2 id='predict.augSIMEX'>
Predict Method for the model fits by augSIMEX
</h2><span id='topic+predict.augSIMEX'></span>

<h3>Description</h3>

<p>This function returns the predictions and optionally estimates the standard errors of the predictions from a fitted augSIMEX object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'augSIMEX'

## S3 method for class 'augSIMEX'
predict(object, newdata = NULL, type = c("link", "response","terms"), 
se.fit = FALSE, dispersion = NULL, terms = NULL, na.action = na.pass, ...) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.augSIMEX_+3A_object">object</code></td>
<td>

<p>the &ldquo;augSIMEX&quot; object gotten from <code><a href="#topic+augSIMEX">augSIMEX</a></code> function.
</p>
</td></tr>
<tr><td><code id="predict.augSIMEX_+3A_newdata">newdata</code></td>
<td>

<p>An optional data frame in which to look for variables with which to predict. If not specified, the prediction will be conducted on the original main data.
</p>
</td></tr>
<tr><td><code id="predict.augSIMEX_+3A_type">type</code></td>
<td>

<p>the type of prediction needed. The default is on the scale of the linear predictors; the alternative &quot;<code>response</code>&quot; is on the scale of the response variable. The setting is similar to the <code><a href="stats.html#topic+predict.glm">predict.glm</a></code>. The &quot;<code>terms</code>&quot; option returns a matrix including the fitted values of each term in the model formula on the linear predictor scale.
</p>
</td></tr>
<tr><td><code id="predict.augSIMEX_+3A_se.fit">se.fit</code></td>
<td>

<p>a logical variable indicating if standard errors are required.
</p>
</td></tr>
<tr><td><code id="predict.augSIMEX_+3A_dispersion">dispersion</code></td>
<td>

<p>a numeric variable specifying the dispersion of the fit to be assumed when computing the standard errors.
</p>
</td></tr>
<tr><td><code id="predict.augSIMEX_+3A_terms">terms</code></td>
<td>

<p>a character vector specifies which terms are to be returned. This is the case for <code>type = "terms"</code>.
</p>
</td></tr>
<tr><td><code id="predict.augSIMEX_+3A_na.action">na.action</code></td>
<td>

<p>function determining what should be done with missing values in <code>newdata</code>. The default is to predict <code>NA</code>.
</p>
</td></tr>
<tr><td><code id="predict.augSIMEX_+3A_...">...</code></td>
<td>

<p>other arguments that are passed into the function.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The specifications of the arugments are the same as the setting in <code>glm</code> function. The user may refer to the <code><a href="stats.html#topic+predict.glm">predict.glm</a></code>.
</p>


<h3>Author(s)</h3>

<p>Qihuang Zhang and Grace Y. Yi.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+predict">predict</a></code>,<code><a href="stats.html#topic+predict.glm">predict.glm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ToyUni)
example &lt;- augSIMEX(mainformula = Y ~ Xstar + Zstar + W, family = binomial(link = logit),
  mismodel = pi|qi ~ W, 
  meformula = Xstar ~ X + Z + W,
  data = ToyUni$Main,validationdata = ToyUni$Validation, subset = NULL,
  err.var = "Xstar", mis.var = "Zstar", err.true = "X", mis.true = "Z", 
  err.mat = NULL,
  lambda = NULL, M = 5, B = 2, nBoot = 2, extrapolation="quadratic")
predict(example)
</code></pre>

<hr>
<h2 id='residuals.augSIMEX'>
Residuals of the Fits Made by augSIMEX
</h2><span id='topic+residuals.augSIMEX'></span>

<h3>Description</h3>

<p>This function outputs the residuals from a fitted augSIMEX object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'augSIMEX'

## S3 method for class 'augSIMEX'
residuals(object, type = c("deviance", "pearson", "working",
   "response", "partial"), ...) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="residuals.augSIMEX_+3A_object">object</code></td>
<td>

<p>the &ldquo;augSIMEX&quot; object gotten from <code><a href="#topic+augSIMEX">augSIMEX</a></code> function.
</p>
</td></tr>
<tr><td><code id="residuals.augSIMEX_+3A_type">type</code></td>
<td>

<p>a character specifying what type of residuals to be returned. The option includes the deviance, Pearson, working, response and partial.
</p>
</td></tr>
<tr><td><code id="residuals.augSIMEX_+3A_...">...</code></td>
<td>

<p>other arguments that are passed into the function.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Qihuang Zhang and Grace Y. Yi.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+residuals">residuals</a></code>,<code><a href="stats.html#topic+residuals.glm">residuals.glm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ToyUni)
example &lt;- augSIMEX(mainformula = Y ~ Xstar + Zstar + W, family = binomial(link = logit),
  mismodel = pi|qi ~ W, 
  meformula = Xstar ~ X + Z + W,
  data = ToyUni$Main,validationdata = ToyUni$Validation, subset = NULL,
  err.var = "Xstar", mis.var = "Zstar", err.true = "X", mis.true = "Z", 
  err.mat = NULL,
  lambda = NULL, M = 5, B = 2, nBoot = 2, extrapolation="quadratic")
residuals(example)
</code></pre>

<hr>
<h2 id='summary.augSIMEX'>
Summarizing the Adjusted Fits of Generalized Linear Model 
</h2><span id='topic+summary.augSIMEX'></span>

<h3>Description</h3>

<p>This function is a method for <code>augSIMEX</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'augSIMEX'

## S3 method for class 'augSIMEX'
summary(object, dispersion = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.augSIMEX_+3A_object">object</code></td>
<td>

<p>the &ldquo;augSIMEX&rdquo; object gotten from <code><a href="#topic+augSIMEX">augSIMEX</a></code> function.
</p>
</td></tr>
<tr><td><code id="summary.augSIMEX_+3A_dispersion">dispersion</code></td>
<td>

<p>the dispersion parameter for the family used.
</p>
</td></tr>
<tr><td><code id="summary.augSIMEX_+3A_...">...</code></td>
<td>

<p>other arguments passed to the function.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function provides summary output similar to <code><a href="stats.html#topic+glm">glm</a></code>. 
</p>


<h3>Value</h3>

<table>
<tr><td><code>coefficients</code></td>
<td>
<p>the matrix of coefficients, standard errors, z-values and p-values.</p>
</td></tr>
<tr><td><code>dispersion</code></td>
<td>
<p>the component from arguments of the function</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Qihuang Zhang and Grace Y. Yi.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+summary.glm">summary.glm</a></code>,<code><a href="#topic+augSIMEX">augSIMEX</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Please see the example in augSIMEX function
</code></pre>

<hr>
<h2 id='ToyMult'>Toy example data for multivariate error-prone covariates</h2><span id='topic+ToyMult'></span>

<h3>Description</h3>

<p>This dataset gives an example data for the illustration of usage of <code><a href="#topic+augSIMEX">augSIMEX</a></code> function. The dataset contains main data and validation data. The main data contains response variable Y and error-prone covariates. Validation data have both correctly observed data and error-prone covariates. See example 2 of <code><a href="#topic+augSIMEX">augSIMEX</a></code> for the usage.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ToyMult)</code></pre>


<h3>Format</h3>

<p>A list of two data frames. Main data, a data frame 1000 observations of 6 variables. Validation data, 500 observations of 8 variables.</p>

<hr>
<h2 id='ToyRepeat'>Toy example data for univariate error-prone covariates in repeated measurements case</h2><span id='topic+ToyRepeat'></span>

<h3>Description</h3>

<p>This data set gives an example data for the illustration of usage of <code><a href="#topic+augSIMEX">augSIMEX</a></code> function in adjusting for the case of repeated measurements error. The dataset contains main data and validation data. The main data includes response variable Y and error-prone covariates. Validation data have both correctly observed data and error-prone covariates. See example 3 of <code><a href="#topic+augSIMEX">augSIMEX</a></code> for the usage.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ToyRepeat)</code></pre>


<h3>Format</h3>

<p>A list of two data frames. Main data, a data frame with 1000 observations of 5 variables. Validation data, 500 observations of 4 variables.</p>

<hr>
<h2 id='ToyUni'>Toy example data for univariate error-prone covariates</h2><span id='topic+ToyUni'></span>

<h3>Description</h3>

<p>This data set gives an example data for the illustration of usage of <code><a href="#topic+augSIMEX">augSIMEX</a></code> function. The dataset contains main data, validation data, and true data. True data correspond to the main data, but the all covariates are corrected observed. The main data and true data contains response variable Y. Both main data and validation data contains the error-prone covariates. Validation data and true data have correctly observed data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ToyUni)</code></pre>


<h3>Format</h3>

<p>A list of three data frames. Main data, a data frame 1000 observations of 4 variables. Validation data, 500 observations of 5 variables. True data, 1000 observations of 4 variables. </p>

<hr>
<h2 id='vcov.augSIMEX'>
Extract the variance-covariance matrix from the fitted augSIMEX object
</h2><span id='topic+vcov.augSIMEX'></span>

<h3>Description</h3>

<p>This function return the variance-covariance matrix from a fitted augSIMEX object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'augSIMEX'

## S3 method for class 'augSIMEX'
vcov(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vcov.augSIMEX_+3A_object">object</code></td>
<td>

<p>the &ldquo;augSIMEX&quot; object gotten from <code><a href="#topic+augSIMEX">augSIMEX</a></code> function.
</p>
</td></tr>
<tr><td><code id="vcov.augSIMEX_+3A_...">...</code></td>
<td>
<p> other arguments to pass to the function.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Qihuang Zhang and Grace Y. Yi.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+vcov">vcov</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ToyUni)
example &lt;- augSIMEX(mainformula = Y ~ Xstar + Zstar + W, family = binomial(link = logit),
  mismodel = pi|qi ~ W, 
  meformula = Xstar ~ X + Z + W,
  data = ToyUni$Main,validationdata = ToyUni$Validation, subset = NULL,
  err.var = "Xstar", mis.var = "Zstar", err.true = "X", mis.true = "Z", 
  err.mat = NULL,
  lambda = NULL, M = 5, B = 2, nBoot = 2, extrapolation="quadratic")
vcov(example)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
