<!DOCTYPE html><html><head><title>Help for package auk</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {auk}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#auk-package'><p><code>auk</code>: eBird Data Extraction and Processing in R</p></a></li>
<li><a href='#auk_bbox'><p>Filter the eBird data by spatial bounding box</p></a></li>
<li><a href='#auk_bcr'><p>Filter the eBird data by Bird Conservation Region</p></a></li>
<li><a href='#auk_breeding'><p>Filter to only include observations with breeding codes</p></a></li>
<li><a href='#auk_clean'><p>Clean an eBird data file (Deprecated)</p></a></li>
<li><a href='#auk_complete'><p>Filter out incomplete checklists from the eBird data</p></a></li>
<li><a href='#auk_country'><p>Filter the eBird data by country</p></a></li>
<li><a href='#auk_county'><p>Filter the eBird data by county</p></a></li>
<li><a href='#auk_date'><p>Filter the eBird data by date</p></a></li>
<li><a href='#auk_distance'><p>Filter eBird data by distance travelled</p></a></li>
<li><a href='#auk_duration'><p>Filter the eBird data by duration</p></a></li>
<li><a href='#auk_ebd'><p>Reference to eBird data file</p></a></li>
<li><a href='#auk_ebd_version'><p>Get the EBD version and associated taxonomy version</p></a></li>
<li><a href='#auk_exotic'><p>Filter the eBird data by exotic code</p></a></li>
<li><a href='#auk_extent'><p>Filter the eBird data by spatial extent</p></a></li>
<li><a href='#auk_filter'><p>Filter the eBird file using AWK</p></a></li>
<li><a href='#auk_get_awk_path'><p>OS specific path to AWK executable</p></a></li>
<li><a href='#auk_get_ebd_path'><p>Return EBD data path</p></a></li>
<li><a href='#auk_last_edited'><p>Filter the eBird data by last edited date</p></a></li>
<li><a href='#auk_observer'><p>Filter the eBird data by observer</p></a></li>
<li><a href='#auk_project'><p>Filter the eBird data by project code</p></a></li>
<li><a href='#auk_protocol'><p>Filter the eBird data by protocol</p></a></li>
<li><a href='#auk_rollup'><p>Roll up eBird taxonomy to species</p></a></li>
<li><a href='#auk_sampling'><p>Reference to eBird sampling event file</p></a></li>
<li><a href='#auk_select'><p>Select a subset of columns</p></a></li>
<li><a href='#auk_set_awk_path'><p>Set a custom path to AWK executable</p></a></li>
<li><a href='#auk_set_ebd_path'><p>Set the path to EBD text files</p></a></li>
<li><a href='#auk_species'><p>Filter the eBird data by species</p></a></li>
<li><a href='#auk_split'><p>Split an eBird data file by species</p></a></li>
<li><a href='#auk_state'><p>Filter the eBird data by state</p></a></li>
<li><a href='#auk_time'><p>Filter the eBird data by checklist start time</p></a></li>
<li><a href='#auk_unique'><p>Remove duplicate group checklists</p></a></li>
<li><a href='#auk_version'><p>Versions of auk, the EBD, and the eBird taxonomy</p></a></li>
<li><a href='#auk_year'><p>Filter the eBird data to a set of years</p></a></li>
<li><a href='#auk_zerofill'><p>Read and zero-fill an eBird data file</p></a></li>
<li><a href='#bcr_codes'><p>BCR Codes</p></a></li>
<li><a href='#ebird_species'><p>Lookup species in eBird taxonomy</p></a></li>
<li><a href='#ebird_states'><p>eBird States</p></a></li>
<li><a href='#ebird_taxonomy'><p>eBird Taxonomy</p></a></li>
<li><a href='#filter_repeat_visits'><p>Filter observations to repeat visits for hierarchical modeling</p></a></li>
<li><a href='#format_unmarked_occu'><p>Format EBD data for occupancy modeling with <code>unmarked</code></p></a></li>
<li><a href='#get_ebird_taxonomy'><p>Get eBird taxonomy via the eBird API</p></a></li>
<li><a href='#read_ebd'><p>Read an EBD file</p></a></li>
<li><a href='#reexports'><p>Objects exported from other packages</p></a></li>
<li><a href='#valid_protocols'><p>Valid Protocols</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>eBird Data Extraction and Processing in R</td>
</tr>
<tr>
<td>Version:</td>
<td>0.7.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Extract and process bird sightings records from
    eBird (<a href="http://ebird.org">http://ebird.org</a>), an online tool for recording bird
    observations.  Public access to the full eBird database is via the
    eBird Basic Dataset (EBD; see <a href="http://ebird.org/ebird/data/download">http://ebird.org/ebird/data/download</a>
    for access), a downloadable text file. This package is an interface to
    AWK for extracting data from the EBD based on taxonomic, spatial, or
    temporal filters, to produce a manageable file size that can be
    imported into R.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/CornellLabofOrnithology/auk">https://github.com/CornellLabofOrnithology/auk</a>,
<a href="https://cornelllabofornithology.github.io/auk/">https://cornelllabofornithology.github.io/auk/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/CornellLabofOrnithology/auk/issues">https://github.com/CornellLabofOrnithology/auk/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.2)</td>
</tr>
<tr>
<td>Imports:</td>
<td>assertthat, countrycode (&ge; 1.0.0), dplyr (&ge; 0.7.8), httr,
magrittr, readr (&ge; 2.0.0), rlang (&ge; 0.3.0), stringi, stringr,
tidyr (&ge; 0.8.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>covr, knitr, rmarkdown, sf, testthat, unmarked, withr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-13 22:32:25 UTC; mes335</td>
</tr>
<tr>
<td>Author:</td>
<td>Matthew Strimas-Mackey
    <a href="https://orcid.org/0000-0001-8929-7776"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Eliot Miller [aut],
  Wesley Hochachka [aut],
  Cornell Lab of Ornithology [cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Matthew Strimas-Mackey &lt;mes335@cornell.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-14 09:43:22 UTC</td>
</tr>
</table>
<hr>
<h2 id='auk-package'><code>auk</code>: eBird Data Extraction and Processing in R</h2><span id='topic+auk'></span><span id='topic+auk-package'></span>

<h3>Description</h3>

<p>Tools for extracting and processing eBird data from the eBird Basic Dataset
(EBD).
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Matthew Strimas-Mackey <a href="mailto:mes335@cornell.edu">mes335@cornell.edu</a> (<a href="https://orcid.org/0000-0001-8929-7776">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Eliot Miller
</p>
</li>
<li><p> Wesley Hochachka
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Cornell Lab of Ornithology [copyright holder]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/CornellLabofOrnithology/auk">https://github.com/CornellLabofOrnithology/auk</a>
</p>
</li>
<li> <p><a href="https://cornelllabofornithology.github.io/auk/">https://cornelllabofornithology.github.io/auk/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/CornellLabofOrnithology/auk/issues">https://github.com/CornellLabofOrnithology/auk/issues</a>
</p>
</li></ul>


<hr>
<h2 id='auk_bbox'>Filter the eBird data by spatial bounding box</h2><span id='topic+auk_bbox'></span>

<h3>Description</h3>

<p>Define a filter for the eBird Basic Dataset (EBD) based on spatial bounding
box. This function only defines the filter and, once all filters have been
defined, <code><a href="#topic+auk_filter">auk_filter()</a></code> should be used to call AWK and perform the filtering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_bbox(x, bbox)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_bbox_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> or <code>auk_sampling</code> object; reference to file created by
<code><a href="#topic+auk_ebd">auk_ebd()</a></code> or <code><a href="#topic+auk_sampling">auk_sampling()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_bbox_+3A_bbox">bbox</code></td>
<td>
<p>numeric or <code>sf</code> or <code style="white-space: pre;">&#8288;Raster*&#8288;</code> object; spatial bounding box
expressed as the range of latitudes and longitudes in decimal degrees:
<code>c(lng_min, lat_min, lng_max, lat_max)</code>. Note that longitudes in the
Western Hemisphere and latitudes sound of the equator should be given as
negative numbers. Alternatively, a spatial object from either the <code>sf</code> or
<code>raster</code> packages can be provided and the bounding box will be extracted
from this object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can also work with on an <code>auk_sampling</code> object if the
user only wishes to filter the sampling event data.
</p>


<h3>Value</h3>

<p>An <code>auk_ebd</code> object.
</p>


<h3>See Also</h3>

<p>Other filter: 
<code><a href="#topic+auk_bcr">auk_bcr</a>()</code>,
<code><a href="#topic+auk_breeding">auk_breeding</a>()</code>,
<code><a href="#topic+auk_complete">auk_complete</a>()</code>,
<code><a href="#topic+auk_country">auk_country</a>()</code>,
<code><a href="#topic+auk_county">auk_county</a>()</code>,
<code><a href="#topic+auk_date">auk_date</a>()</code>,
<code><a href="#topic+auk_distance">auk_distance</a>()</code>,
<code><a href="#topic+auk_duration">auk_duration</a>()</code>,
<code><a href="#topic+auk_exotic">auk_exotic</a>()</code>,
<code><a href="#topic+auk_extent">auk_extent</a>()</code>,
<code><a href="#topic+auk_filter">auk_filter</a>()</code>,
<code><a href="#topic+auk_last_edited">auk_last_edited</a>()</code>,
<code><a href="#topic+auk_observer">auk_observer</a>()</code>,
<code><a href="#topic+auk_project">auk_project</a>()</code>,
<code><a href="#topic+auk_protocol">auk_protocol</a>()</code>,
<code><a href="#topic+auk_species">auk_species</a>()</code>,
<code><a href="#topic+auk_state">auk_state</a>()</code>,
<code><a href="#topic+auk_time">auk_time</a>()</code>,
<code><a href="#topic+auk_year">auk_year</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># fliter to locations roughly in the Pacific Northwest
system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  auk_bbox(bbox = c(-125, 37, -120, 52))
  
# alternatively, without pipes
ebd &lt;- auk_ebd(system.file("extdata/ebd-sample.txt", package = "auk"))
auk_bbox(ebd, bbox = c(-125, 37, -120, 52))
</code></pre>

<hr>
<h2 id='auk_bcr'>Filter the eBird data by Bird Conservation Region</h2><span id='topic+auk_bcr'></span>

<h3>Description</h3>

<p>Define a filter for the eBird Basic Dataset (EBD) to extract data for a set
of <a href="https://nabci-us.org/resources/bird-conservation-regions/">Bird Conservation Regions</a> (BCRs).
BCRs are ecologically distinct regions in North America with similar bird
communities, habitats, and resource management issues. This function only
defines the filter and, once all filters have been defined, <code><a href="#topic+auk_filter">auk_filter()</a></code>
should be used to call AWK and perform the filtering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_bcr(x, bcr, replace = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_bcr_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> or <code>auk_sampling</code> object; reference to file created by
<code><a href="#topic+auk_ebd">auk_ebd()</a></code> or <code><a href="#topic+auk_sampling">auk_sampling()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_bcr_+3A_bcr">bcr</code></td>
<td>
<p>integer; BCRs to filter by. BCRs are identified by an integer,
from 1 to 66, that can be looked up in the <a href="#topic+bcr_codes">bcr_codes</a> table.</p>
</td></tr>
<tr><td><code id="auk_bcr_+3A_replace">replace</code></td>
<td>
<p>logical; multiple calls to <code>auk_bcr()</code> are additive,
unless <code>replace = FALSE</code>, in which case the previous list of states to
filter by will be removed and replaced by that in the current call.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can also work with on an <code>auk_sampling</code> object if the
user only wishes to filter the sampling event data.
</p>


<h3>Value</h3>

<p>An <code>auk_ebd</code> object.
</p>


<h3>See Also</h3>

<p>Other filter: 
<code><a href="#topic+auk_bbox">auk_bbox</a>()</code>,
<code><a href="#topic+auk_breeding">auk_breeding</a>()</code>,
<code><a href="#topic+auk_complete">auk_complete</a>()</code>,
<code><a href="#topic+auk_country">auk_country</a>()</code>,
<code><a href="#topic+auk_county">auk_county</a>()</code>,
<code><a href="#topic+auk_date">auk_date</a>()</code>,
<code><a href="#topic+auk_distance">auk_distance</a>()</code>,
<code><a href="#topic+auk_duration">auk_duration</a>()</code>,
<code><a href="#topic+auk_exotic">auk_exotic</a>()</code>,
<code><a href="#topic+auk_extent">auk_extent</a>()</code>,
<code><a href="#topic+auk_filter">auk_filter</a>()</code>,
<code><a href="#topic+auk_last_edited">auk_last_edited</a>()</code>,
<code><a href="#topic+auk_observer">auk_observer</a>()</code>,
<code><a href="#topic+auk_project">auk_project</a>()</code>,
<code><a href="#topic+auk_protocol">auk_protocol</a>()</code>,
<code><a href="#topic+auk_species">auk_species</a>()</code>,
<code><a href="#topic+auk_state">auk_state</a>()</code>,
<code><a href="#topic+auk_time">auk_time</a>()</code>,
<code><a href="#topic+auk_year">auk_year</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># bcr codes can be looked up in bcr_codes
dplyr::filter(bcr_codes, bcr_name == "Central Hardwoods")
system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  auk_bcr(bcr = 24)
  
# filter to bcr 24
ebd &lt;- auk_ebd(system.file("extdata/ebd-sample.txt", package = "auk"))
auk_bcr(ebd, bcr = 24)
</code></pre>

<hr>
<h2 id='auk_breeding'>Filter to only include observations with breeding codes</h2><span id='topic+auk_breeding'></span>

<h3>Description</h3>

<p>eBird users have the option of specifying breeding bird atlas codes for their
observations, for example, if nesting building behaviour is observed. Use
this filter to select only those observations with an associated breeding
code. This function only defines the filter and, once all filters have been
defined, <code><a href="#topic+auk_filter">auk_filter()</a></code> should be used to call AWK and perform the filtering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_breeding(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_breeding_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> object; reference to basic dataset file created by
<code><a href="#topic+auk_ebd">auk_ebd()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>auk_ebd</code> object.
</p>


<h3>See Also</h3>

<p>Other filter: 
<code><a href="#topic+auk_bbox">auk_bbox</a>()</code>,
<code><a href="#topic+auk_bcr">auk_bcr</a>()</code>,
<code><a href="#topic+auk_complete">auk_complete</a>()</code>,
<code><a href="#topic+auk_country">auk_country</a>()</code>,
<code><a href="#topic+auk_county">auk_county</a>()</code>,
<code><a href="#topic+auk_date">auk_date</a>()</code>,
<code><a href="#topic+auk_distance">auk_distance</a>()</code>,
<code><a href="#topic+auk_duration">auk_duration</a>()</code>,
<code><a href="#topic+auk_exotic">auk_exotic</a>()</code>,
<code><a href="#topic+auk_extent">auk_extent</a>()</code>,
<code><a href="#topic+auk_filter">auk_filter</a>()</code>,
<code><a href="#topic+auk_last_edited">auk_last_edited</a>()</code>,
<code><a href="#topic+auk_observer">auk_observer</a>()</code>,
<code><a href="#topic+auk_project">auk_project</a>()</code>,
<code><a href="#topic+auk_protocol">auk_protocol</a>()</code>,
<code><a href="#topic+auk_species">auk_species</a>()</code>,
<code><a href="#topic+auk_state">auk_state</a>()</code>,
<code><a href="#topic+auk_time">auk_time</a>()</code>,
<code><a href="#topic+auk_year">auk_year</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  auk_breeding()
</code></pre>

<hr>
<h2 id='auk_clean'>Clean an eBird data file (Deprecated)</h2><span id='topic+auk_clean'></span>

<h3>Description</h3>

<p>This function is no longer required by current versions of the eBird Basic
Dataset (EBD).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_clean(f_in, f_out, sep = "\t", remove_text = FALSE, overwrite = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_clean_+3A_f_in">f_in</code></td>
<td>
<p>character; input file. If file is not found as specified, it will
be looked for in the directory specified by the <code>EBD_PATH</code> environment
variable.</p>
</td></tr>
<tr><td><code id="auk_clean_+3A_f_out">f_out</code></td>
<td>
<p>character; output file.</p>
</td></tr>
<tr><td><code id="auk_clean_+3A_sep">sep</code></td>
<td>
<p>character; the input field separator, the basic dataset is tab
separated by default. Must only be a single character and space delimited
is not allowed since spaces appear in many of the fields.</p>
</td></tr>
<tr><td><code id="auk_clean_+3A_remove_text">remove_text</code></td>
<td>
<p>logical; whether all free text entry columns should be
removed. These columns include comments, location names, and observer
names. These columns cause import errors due to special characters and
increase the file size, yet are rarely valuable for analytical
applications, so may be removed. Setting this argument to <code>TRUE</code> can lead
to a significant reduction in file size.</p>
</td></tr>
<tr><td><code id="auk_clean_+3A_overwrite">overwrite</code></td>
<td>
<p>logical; overwrite output file if it already exists.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If AWK ran without errors, the output filename is returned, however,
if an error was encountered the exit code is returned.
</p>


<h3>See Also</h3>

<p>Other text: 
<code><a href="#topic+auk_select">auk_select</a>()</code>,
<code><a href="#topic+auk_split">auk_split</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# get the path to the example data included in the package
f &lt;- system.file("extdata/ebd-sample.txt", package = "auk")
# output to a temp file for example
# in practice, provide path to output file
# e.g. f_out &lt;- "output/ebd_clean.txt"
f_out &lt;- tempfile()

# clean file to remove problem rows
# note: this function is deprecated and no longer does anything
auk_clean(f, f_out)

## End(Not run)
</code></pre>

<hr>
<h2 id='auk_complete'>Filter out incomplete checklists from the eBird data</h2><span id='topic+auk_complete'></span>

<h3>Description</h3>

<p>Define a filter for the eBird Basic Dataset (EBD) to only keep complete
checklists, i.e. those for which all birds seen or heard were recorded. These
checklists are the most valuable for scientific uses since they provide
presence and absence data.This function only defines the filter and, once all
filters have been defined, <code><a href="#topic+auk_filter">auk_filter()</a></code> should be used to call AWK and
perform the filtering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_complete(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_complete_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> or <code>auk_sampling</code> object; reference to file created by
<code><a href="#topic+auk_ebd">auk_ebd()</a></code> or <code><a href="#topic+auk_sampling">auk_sampling()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can also work with on an <code>auk_sampling</code> object if the
user only wishes to filter the sampling event data.
</p>


<h3>Value</h3>

<p>An <code>auk_ebd</code> object.
</p>


<h3>See Also</h3>

<p>Other filter: 
<code><a href="#topic+auk_bbox">auk_bbox</a>()</code>,
<code><a href="#topic+auk_bcr">auk_bcr</a>()</code>,
<code><a href="#topic+auk_breeding">auk_breeding</a>()</code>,
<code><a href="#topic+auk_country">auk_country</a>()</code>,
<code><a href="#topic+auk_county">auk_county</a>()</code>,
<code><a href="#topic+auk_date">auk_date</a>()</code>,
<code><a href="#topic+auk_distance">auk_distance</a>()</code>,
<code><a href="#topic+auk_duration">auk_duration</a>()</code>,
<code><a href="#topic+auk_exotic">auk_exotic</a>()</code>,
<code><a href="#topic+auk_extent">auk_extent</a>()</code>,
<code><a href="#topic+auk_filter">auk_filter</a>()</code>,
<code><a href="#topic+auk_last_edited">auk_last_edited</a>()</code>,
<code><a href="#topic+auk_observer">auk_observer</a>()</code>,
<code><a href="#topic+auk_project">auk_project</a>()</code>,
<code><a href="#topic+auk_protocol">auk_protocol</a>()</code>,
<code><a href="#topic+auk_species">auk_species</a>()</code>,
<code><a href="#topic+auk_state">auk_state</a>()</code>,
<code><a href="#topic+auk_time">auk_time</a>()</code>,
<code><a href="#topic+auk_year">auk_year</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  auk_complete()
</code></pre>

<hr>
<h2 id='auk_country'>Filter the eBird data by country</h2><span id='topic+auk_country'></span>

<h3>Description</h3>

<p>Define a filter for the eBird Basic Dataset (EBD) based on a set of
countries. This function only defines the filter and, once all filters have
been defined, <code><a href="#topic+auk_filter">auk_filter()</a></code> should be used to call AWK and perform the
filtering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_country(x, country, replace = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_country_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> or <code>auk_sampling</code> object; reference to file created by
<code><a href="#topic+auk_ebd">auk_ebd()</a></code> or <code><a href="#topic+auk_sampling">auk_sampling()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_country_+3A_country">country</code></td>
<td>
<p>character; countries to filter by. Countries can either be
expressed as English names or
<a href="https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2">ISO 2-letter country codes</a>.
English names are matched via regular expressions using
<a href="countrycode.html#topic+countrycode">countrycode</a>, so there is some flexibility in names.</p>
</td></tr>
<tr><td><code id="auk_country_+3A_replace">replace</code></td>
<td>
<p>logical; multiple calls to <code>auk_country()</code> are additive,
unless <code>replace = FALSE</code>, in which case the previous list of countries to
filter by will be removed and replaced by that in the current call.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can also work with on an <code>auk_sampling</code> object if the
user only wishes to filter the sampling event data.
</p>


<h3>Value</h3>

<p>An <code>auk_ebd</code> object.
</p>


<h3>See Also</h3>

<p>Other filter: 
<code><a href="#topic+auk_bbox">auk_bbox</a>()</code>,
<code><a href="#topic+auk_bcr">auk_bcr</a>()</code>,
<code><a href="#topic+auk_breeding">auk_breeding</a>()</code>,
<code><a href="#topic+auk_complete">auk_complete</a>()</code>,
<code><a href="#topic+auk_county">auk_county</a>()</code>,
<code><a href="#topic+auk_date">auk_date</a>()</code>,
<code><a href="#topic+auk_distance">auk_distance</a>()</code>,
<code><a href="#topic+auk_duration">auk_duration</a>()</code>,
<code><a href="#topic+auk_exotic">auk_exotic</a>()</code>,
<code><a href="#topic+auk_extent">auk_extent</a>()</code>,
<code><a href="#topic+auk_filter">auk_filter</a>()</code>,
<code><a href="#topic+auk_last_edited">auk_last_edited</a>()</code>,
<code><a href="#topic+auk_observer">auk_observer</a>()</code>,
<code><a href="#topic+auk_project">auk_project</a>()</code>,
<code><a href="#topic+auk_protocol">auk_protocol</a>()</code>,
<code><a href="#topic+auk_species">auk_species</a>()</code>,
<code><a href="#topic+auk_state">auk_state</a>()</code>,
<code><a href="#topic+auk_time">auk_time</a>()</code>,
<code><a href="#topic+auk_year">auk_year</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># country names and ISO2 codes can be mixed
# not case sensitive
country &lt;- c("CA", "United States", "mexico")
system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  auk_country(country)
  
# alternatively, without pipes
ebd &lt;- auk_ebd(system.file("extdata/ebd-sample.txt", package = "auk"))
auk_country(ebd, country)
</code></pre>

<hr>
<h2 id='auk_county'>Filter the eBird data by county</h2><span id='topic+auk_county'></span>

<h3>Description</h3>

<p>Define a filter for the eBird Basic Dataset (EBD) based on a set of
counties This function only defines the filter and, once all filters have
been defined, <code><a href="#topic+auk_filter">auk_filter()</a></code> should be used to call AWK and perform the
filtering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_county(x, county, replace = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_county_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> or <code>auk_sampling</code> object; reference to file created by
<code><a href="#topic+auk_ebd">auk_ebd()</a></code> or <code><a href="#topic+auk_sampling">auk_sampling()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_county_+3A_county">county</code></td>
<td>
<p>character; counties to filter by. eBird uses county codes
consisting of three parts, the 2-letter ISO country code, a 1-3 character
state code, and a county code, all separated by a dash. For example,
<code>"US-NY-109"</code> corresponds to Tompkins, NY, US. The easiest way to find a
county code is to find the corresponding <a href="https://ebird.org/explore">explore region</a> page and look at the URL.</p>
</td></tr>
<tr><td><code id="auk_county_+3A_replace">replace</code></td>
<td>
<p>logical; multiple calls to <code>auk_county()</code> are additive,
unless <code>replace = FALSE</code>, in which case the previous list of states to
filter by will be removed and replaced by that in the current call.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is not possible to filter by both county as well as country or
state, so calling <code>auk_county()</code> will reset these filters to all countries
and states, and vice versa.
</p>
<p>This function can also work with on an <code>auk_sampling</code> object if the user only
wishes to filter the sampling event data.
</p>


<h3>Value</h3>

<p>An <code>auk_ebd</code> object.
</p>


<h3>See Also</h3>

<p>Other filter: 
<code><a href="#topic+auk_bbox">auk_bbox</a>()</code>,
<code><a href="#topic+auk_bcr">auk_bcr</a>()</code>,
<code><a href="#topic+auk_breeding">auk_breeding</a>()</code>,
<code><a href="#topic+auk_complete">auk_complete</a>()</code>,
<code><a href="#topic+auk_country">auk_country</a>()</code>,
<code><a href="#topic+auk_date">auk_date</a>()</code>,
<code><a href="#topic+auk_distance">auk_distance</a>()</code>,
<code><a href="#topic+auk_duration">auk_duration</a>()</code>,
<code><a href="#topic+auk_exotic">auk_exotic</a>()</code>,
<code><a href="#topic+auk_extent">auk_extent</a>()</code>,
<code><a href="#topic+auk_filter">auk_filter</a>()</code>,
<code><a href="#topic+auk_last_edited">auk_last_edited</a>()</code>,
<code><a href="#topic+auk_observer">auk_observer</a>()</code>,
<code><a href="#topic+auk_project">auk_project</a>()</code>,
<code><a href="#topic+auk_protocol">auk_protocol</a>()</code>,
<code><a href="#topic+auk_species">auk_species</a>()</code>,
<code><a href="#topic+auk_state">auk_state</a>()</code>,
<code><a href="#topic+auk_time">auk_time</a>()</code>,
<code><a href="#topic+auk_year">auk_year</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># choose tompkins county, ny, united states
system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  auk_county("US-NY-109")
  
# alternatively, without pipes
ebd &lt;- auk_ebd(system.file("extdata/ebd-sample.txt", package = "auk"))
auk_county(ebd, "US-NY-109")
</code></pre>

<hr>
<h2 id='auk_date'>Filter the eBird data by date</h2><span id='topic+auk_date'></span>

<h3>Description</h3>

<p>Define a filter for the eBird Basic Dataset (EBD) based on a range of dates.
This function only defines the filter and, once all filters have been
defined, <code><a href="#topic+auk_filter">auk_filter()</a></code> should be used to call AWK and perform the
filtering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_date(x, date)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_date_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> or <code>auk_sampling</code> object; reference to file created by
<code><a href="#topic+auk_ebd">auk_ebd()</a></code> or <code><a href="#topic+auk_sampling">auk_sampling()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_date_+3A_date">date</code></td>
<td>
<p>character or date; date range to filter by, provided either as a
character vector in the format <code>"2015-12-31"</code> or a vector of Date objects.
To filter on a range of dates, regardless of year, use <code>"*"</code> in place of
the year.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To select observations from a range of dates, regardless of year,
the  wildcard <code>"*"</code> can be used in place of the year. For example, using
<code>date = c("*-05-01", "*-06-30")</code> will return observations from May and June
of <em>any year</em>. When using wildcards, dates can wrap around the year end.
</p>
<p>This function can also work with on an <code>auk_sampling</code> object if the user only
wishes to filter the sampling event data.
</p>


<h3>Value</h3>

<p>An <code>auk_ebd</code> object.
</p>


<h3>See Also</h3>

<p>Other filter: 
<code><a href="#topic+auk_bbox">auk_bbox</a>()</code>,
<code><a href="#topic+auk_bcr">auk_bcr</a>()</code>,
<code><a href="#topic+auk_breeding">auk_breeding</a>()</code>,
<code><a href="#topic+auk_complete">auk_complete</a>()</code>,
<code><a href="#topic+auk_country">auk_country</a>()</code>,
<code><a href="#topic+auk_county">auk_county</a>()</code>,
<code><a href="#topic+auk_distance">auk_distance</a>()</code>,
<code><a href="#topic+auk_duration">auk_duration</a>()</code>,
<code><a href="#topic+auk_exotic">auk_exotic</a>()</code>,
<code><a href="#topic+auk_extent">auk_extent</a>()</code>,
<code><a href="#topic+auk_filter">auk_filter</a>()</code>,
<code><a href="#topic+auk_last_edited">auk_last_edited</a>()</code>,
<code><a href="#topic+auk_observer">auk_observer</a>()</code>,
<code><a href="#topic+auk_project">auk_project</a>()</code>,
<code><a href="#topic+auk_protocol">auk_protocol</a>()</code>,
<code><a href="#topic+auk_species">auk_species</a>()</code>,
<code><a href="#topic+auk_state">auk_state</a>()</code>,
<code><a href="#topic+auk_time">auk_time</a>()</code>,
<code><a href="#topic+auk_year">auk_year</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  auk_date(date = c("2010-01-01", "2010-12-31"))
  
# alternatively, without pipes
ebd &lt;- auk_ebd(system.file("extdata/ebd-sample.txt", package = "auk"))
auk_date(ebd, date = c("2010-01-01", "2010-12-31"))

# the * wildcard can be used in place of year to select dates from all years
system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  # may-june records from all years
  auk_date(date = c("*-05-01", "*-06-30"))
  
# dates can also wrap around the end of the year
system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  # dec-jan records from all years
  auk_date(date = c("*-12-01", "*-01-31"))
</code></pre>

<hr>
<h2 id='auk_distance'>Filter eBird data by distance travelled</h2><span id='topic+auk_distance'></span>

<h3>Description</h3>

<p>Define a filter for the eBird Basic Dataset (EBD) based on the distance
travelled on the checklist. This function only defines the filter and, once
all filters have been defined, <code><a href="#topic+auk_filter">auk_filter()</a></code> should be used to call AWK and
perform the filtering. Note that stationary checklists (i.e. point counts)
have no distance associated with them, however, since these checklists can
be assumed to have 0 distance they will be kept if 0 is in the range defined
by <code>distance</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_distance(x, distance, distance_units)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_distance_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> or <code>auk_sampling</code> object; reference to file created by
<code><a href="#topic+auk_ebd">auk_ebd()</a></code> or <code><a href="#topic+auk_sampling">auk_sampling()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_distance_+3A_distance">distance</code></td>
<td>
<p>integer; 2 element vector specifying the range of distances
to filter by. The default is to accept distances in kilometers, use
<code>distance_units = "miles"</code> for miles.</p>
</td></tr>
<tr><td><code id="auk_distance_+3A_distance_units">distance_units</code></td>
<td>
<p>character; whether distances are provided in kilometers
(the default) or miles.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can also work with on an <code>auk_sampling</code> object if the
user only wishes to filter the sampling event data.
</p>


<h3>Value</h3>

<p>An <code>auk_ebd</code> object.
</p>


<h3>See Also</h3>

<p>Other filter: 
<code><a href="#topic+auk_bbox">auk_bbox</a>()</code>,
<code><a href="#topic+auk_bcr">auk_bcr</a>()</code>,
<code><a href="#topic+auk_breeding">auk_breeding</a>()</code>,
<code><a href="#topic+auk_complete">auk_complete</a>()</code>,
<code><a href="#topic+auk_country">auk_country</a>()</code>,
<code><a href="#topic+auk_county">auk_county</a>()</code>,
<code><a href="#topic+auk_date">auk_date</a>()</code>,
<code><a href="#topic+auk_duration">auk_duration</a>()</code>,
<code><a href="#topic+auk_exotic">auk_exotic</a>()</code>,
<code><a href="#topic+auk_extent">auk_extent</a>()</code>,
<code><a href="#topic+auk_filter">auk_filter</a>()</code>,
<code><a href="#topic+auk_last_edited">auk_last_edited</a>()</code>,
<code><a href="#topic+auk_observer">auk_observer</a>()</code>,
<code><a href="#topic+auk_project">auk_project</a>()</code>,
<code><a href="#topic+auk_protocol">auk_protocol</a>()</code>,
<code><a href="#topic+auk_species">auk_species</a>()</code>,
<code><a href="#topic+auk_state">auk_state</a>()</code>,
<code><a href="#topic+auk_time">auk_time</a>()</code>,
<code><a href="#topic+auk_year">auk_year</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># only keep checklists that are less than 10 km long
system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  auk_distance(distance = c(0, 10))
  
# alternatively, without pipes
ebd &lt;- auk_ebd(system.file("extdata/ebd-sample.txt", package = "auk"))
auk_distance(ebd, distance = c(0, 10))
</code></pre>

<hr>
<h2 id='auk_duration'>Filter the eBird data by duration</h2><span id='topic+auk_duration'></span>

<h3>Description</h3>

<p>Define a filter for the eBird Basic Dataset (EBD) based on the duration of
the checklist. This function only defines the filter and, once all filters
have been defined, <code><a href="#topic+auk_filter">auk_filter()</a></code> should be used to call AWK and perform the
filtering. Note that checklists with no effort, such as incidental
observations, will be excluded if this filter is used since they have no
associated duration information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_duration(x, duration)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_duration_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> or <code>auk_sampling</code> object; reference to file created by
<code><a href="#topic+auk_ebd">auk_ebd()</a></code> or <code><a href="#topic+auk_sampling">auk_sampling()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_duration_+3A_duration">duration</code></td>
<td>
<p>integer; 2 element vector specifying the range of durations
in minutes to filter by.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can also work with on an <code>auk_sampling</code> object if the
user only wishes to filter the sampling event data.
</p>


<h3>Value</h3>

<p>An <code>auk_ebd</code> object.
</p>


<h3>See Also</h3>

<p>Other filter: 
<code><a href="#topic+auk_bbox">auk_bbox</a>()</code>,
<code><a href="#topic+auk_bcr">auk_bcr</a>()</code>,
<code><a href="#topic+auk_breeding">auk_breeding</a>()</code>,
<code><a href="#topic+auk_complete">auk_complete</a>()</code>,
<code><a href="#topic+auk_country">auk_country</a>()</code>,
<code><a href="#topic+auk_county">auk_county</a>()</code>,
<code><a href="#topic+auk_date">auk_date</a>()</code>,
<code><a href="#topic+auk_distance">auk_distance</a>()</code>,
<code><a href="#topic+auk_exotic">auk_exotic</a>()</code>,
<code><a href="#topic+auk_extent">auk_extent</a>()</code>,
<code><a href="#topic+auk_filter">auk_filter</a>()</code>,
<code><a href="#topic+auk_last_edited">auk_last_edited</a>()</code>,
<code><a href="#topic+auk_observer">auk_observer</a>()</code>,
<code><a href="#topic+auk_project">auk_project</a>()</code>,
<code><a href="#topic+auk_protocol">auk_protocol</a>()</code>,
<code><a href="#topic+auk_species">auk_species</a>()</code>,
<code><a href="#topic+auk_state">auk_state</a>()</code>,
<code><a href="#topic+auk_time">auk_time</a>()</code>,
<code><a href="#topic+auk_year">auk_year</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># only keep checklists that are less than an hour long
system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  auk_duration(duration = c(0, 60))
  
# alternatively, without pipes
ebd &lt;- auk_ebd(system.file("extdata/ebd-sample.txt", package = "auk"))
auk_duration(ebd, duration = c(0, 60))
</code></pre>

<hr>
<h2 id='auk_ebd'>Reference to eBird data file</h2><span id='topic+auk_ebd'></span>

<h3>Description</h3>

<p>Create a reference to an eBird Basic Dataset (EBD) file in preparation for
filtering using AWK.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_ebd(file, file_sampling, sep = "\t")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_ebd_+3A_file">file</code></td>
<td>
<p>character; input file. If file is not found as specified, it will
be looked for in the directory specified by the <code>EBD_PATH</code> environment
variable.</p>
</td></tr>
<tr><td><code id="auk_ebd_+3A_file_sampling">file_sampling</code></td>
<td>
<p>character; optional input sampling event data (i.e.
checklists) file, required if you intend to zero-fill the data to produce a
presence-absence data set. This file consists of just effort information
for every eBird checklist. Any species not appearing in the EBD for a given
checklist is implicitly considered to have a count of 0. This file should
be downloaded at the same time as the basic dataset to ensure they are in
sync. If file is not found as specified, it will be looked for in the
directory specified by the <code>EBD_PATH</code> environment variable.</p>
</td></tr>
<tr><td><code id="auk_ebd_+3A_sep">sep</code></td>
<td>
<p>character; the input field separator, the eBird data are tab
separated so this should generally not be modified. Must only be a single
character and space delimited is not allowed since spaces appear in many of
the fields.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>eBird data can be downloaded as a tab-separated text file from the
<a href="http://ebird.org/ebird/data/download">eBird website</a> after submitting a
request for access. As of February 2017, this file is nearly 150 GB making it
challenging to work with. If you're only interested in a single species or a
small region it is possible to submit a custom download request. This
approach is suggested to speed up processing time.
</p>
<p>There are two potential pathways for preparing eBird data. Users wishing to
produce presence only data, should download the
<a href="http://ebird.org/ebird/data/download/">eBird Basic Dataset</a> and reference
this file when calling <code>auk_ebd()</code>. Users wishing to produce zero-filled,
presence absence data should additionally download the sampling event data
file associated with the basic dataset This file contains only checklist
information and can be used to infer absences. The sampling event data file
should be provided to <code>auk_ebd()</code> via the <code>file_sampling</code> argument. For
further details consult the vignettes.
</p>


<h3>Value</h3>

<p>An <code>auk_ebd</code> object storing the file reference and the desired
filters once created with other package functions.
</p>


<h3>See Also</h3>

<p>Other objects: 
<code><a href="#topic+auk_sampling">auk_sampling</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># get the path to the example data included in the package
# in practice, provide path to ebd, e.g. f &lt;- "data/ebd_relFeb-2018.txt
f &lt;- system.file("extdata/ebd-sample.txt", package = "auk")
auk_ebd(f)
# to produce zero-filled data, provide a checklist file
f_ebd &lt;- system.file("extdata/zerofill-ex_ebd.txt", package = "auk")
f_cl &lt;- system.file("extdata/zerofill-ex_sampling.txt", package = "auk")
auk_ebd(f_ebd, file_sampling = f_cl)
</code></pre>

<hr>
<h2 id='auk_ebd_version'>Get the EBD version and associated taxonomy version</h2><span id='topic+auk_ebd_version'></span>

<h3>Description</h3>

<p>Based on the filename of eBird Basic Dataset (EBD) or sampling event data,
determine the version (i.e. release date) of this EBD. Also determine the
corresponding taxonomy version. The eBird taxonomy is updated annually in
August.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_ebd_version(x, check_exists = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_ebd_version_+3A_x">x</code></td>
<td>
<p>filename of EBD of sampling event data file, <code>auk_ebd</code> object, or
<code>auk_sampling</code> object.</p>
</td></tr>
<tr><td><code id="auk_ebd_version_+3A_check_exists">check_exists</code></td>
<td>
<p>logical; should the file be checked for existence before
processing. If <code>check_exists = TRUE</code> and the file does not exists, the
function will raise an error.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with two elements:
</p>

<ul>
<li> <p><code>ebd_version</code>: a date object specifying the release date of the EBD.
</p>
</li>
<li> <p><code>taxonomy_version</code>: the year of the taxonomy used in this EBD.
</p>
</li></ul>

<p>Both elements will be NA if an EBD version cannot be extracted from the
filename.
</p>


<h3>See Also</h3>

<p>Other helpers: 
<code><a href="#topic+auk_version">auk_version</a>()</code>,
<code><a href="#topic+ebird_species">ebird_species</a>()</code>,
<code><a href="#topic+get_ebird_taxonomy">get_ebird_taxonomy</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>auk_ebd_version("ebd_relAug-2018.txt", check_exists = FALSE)
</code></pre>

<hr>
<h2 id='auk_exotic'>Filter the eBird data by exotic code</h2><span id='topic+auk_exotic'></span>

<h3>Description</h3>

<p>Exotic codes are applied to eBird observations when the species is believe to
be non-native to the given location. This function defines a filter for the
eBird Basic Dataset (EBD) to subset observations to one or more of the exotic
codes: &quot;&quot; (i.e. no code, meaning it is a native species), &quot;N&quot; (naturalized),
&quot;P&quot; (provisional), or &quot;X&quot; (escapee). This function only defines the filter
and, once all filters have been defined, <code><a href="#topic+auk_filter">auk_filter()</a></code> should be used to
call AWK and perform the filtering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_exotic(x, exotic_code, replace = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_exotic_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> or <code>auk_sampling</code> object; reference to file created by
<code><a href="#topic+auk_ebd">auk_ebd()</a></code> or <code><a href="#topic+auk_sampling">auk_sampling()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_exotic_+3A_exotic_code">exotic_code</code></td>
<td>
<p>characterr; exotic codes to filter by. Note that an empty
string (&quot;&quot;), meaning no exotic code, is used for native species.</p>
</td></tr>
<tr><td><code id="auk_exotic_+3A_replace">replace</code></td>
<td>
<p>logical; multiple calls to <code>auk_exotic()</code> are additive,
unless <code>replace = FALSE</code>, in which case the previous list of states to
filter by will be removed and replaced by that in the current call.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>auk_ebd</code> object.
</p>


<h3>See Also</h3>

<p>Other filter: 
<code><a href="#topic+auk_bbox">auk_bbox</a>()</code>,
<code><a href="#topic+auk_bcr">auk_bcr</a>()</code>,
<code><a href="#topic+auk_breeding">auk_breeding</a>()</code>,
<code><a href="#topic+auk_complete">auk_complete</a>()</code>,
<code><a href="#topic+auk_country">auk_country</a>()</code>,
<code><a href="#topic+auk_county">auk_county</a>()</code>,
<code><a href="#topic+auk_date">auk_date</a>()</code>,
<code><a href="#topic+auk_distance">auk_distance</a>()</code>,
<code><a href="#topic+auk_duration">auk_duration</a>()</code>,
<code><a href="#topic+auk_extent">auk_extent</a>()</code>,
<code><a href="#topic+auk_filter">auk_filter</a>()</code>,
<code><a href="#topic+auk_last_edited">auk_last_edited</a>()</code>,
<code><a href="#topic+auk_observer">auk_observer</a>()</code>,
<code><a href="#topic+auk_project">auk_project</a>()</code>,
<code><a href="#topic+auk_protocol">auk_protocol</a>()</code>,
<code><a href="#topic+auk_species">auk_species</a>()</code>,
<code><a href="#topic+auk_state">auk_state</a>()</code>,
<code><a href="#topic+auk_time">auk_time</a>()</code>,
<code><a href="#topic+auk_year">auk_year</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># filter to only native observations
ebd &lt;- auk_ebd(system.file("extdata/ebd-sample.txt", package = "auk"))
auk_exotic(ebd, exotic_code = "")

# filter to native and naturalized observations
auk_exotic(ebd, exotic_code = c("", "N"))
</code></pre>

<hr>
<h2 id='auk_extent'>Filter the eBird data by spatial extent</h2><span id='topic+auk_extent'></span>

<h3>Description</h3>

<p><strong>Deprecated</strong>, use <code><a href="#topic+auk_bbox">auk_bbox()</a></code> instead.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_extent(x, extent)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_extent_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> or <code>auk_sampling</code> object; reference to file created by
<code><a href="#topic+auk_ebd">auk_ebd()</a></code> or <code><a href="#topic+auk_sampling">auk_sampling()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_extent_+3A_extent">extent</code></td>
<td>
<p>numeric; spatial extent expressed as the range of latitudes and
longitudes in decimal degrees: <code>c(lng_min, lat_min, lng_max, lat_max)</code>.
Note that longitudes in the Western Hemisphere and latitudes sound of the
equator should be given as negative numbers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>auk_ebd</code> object.
</p>


<h3>See Also</h3>

<p>Other filter: 
<code><a href="#topic+auk_bbox">auk_bbox</a>()</code>,
<code><a href="#topic+auk_bcr">auk_bcr</a>()</code>,
<code><a href="#topic+auk_breeding">auk_breeding</a>()</code>,
<code><a href="#topic+auk_complete">auk_complete</a>()</code>,
<code><a href="#topic+auk_country">auk_country</a>()</code>,
<code><a href="#topic+auk_county">auk_county</a>()</code>,
<code><a href="#topic+auk_date">auk_date</a>()</code>,
<code><a href="#topic+auk_distance">auk_distance</a>()</code>,
<code><a href="#topic+auk_duration">auk_duration</a>()</code>,
<code><a href="#topic+auk_exotic">auk_exotic</a>()</code>,
<code><a href="#topic+auk_filter">auk_filter</a>()</code>,
<code><a href="#topic+auk_last_edited">auk_last_edited</a>()</code>,
<code><a href="#topic+auk_observer">auk_observer</a>()</code>,
<code><a href="#topic+auk_project">auk_project</a>()</code>,
<code><a href="#topic+auk_protocol">auk_protocol</a>()</code>,
<code><a href="#topic+auk_species">auk_species</a>()</code>,
<code><a href="#topic+auk_state">auk_state</a>()</code>,
<code><a href="#topic+auk_time">auk_time</a>()</code>,
<code><a href="#topic+auk_year">auk_year</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># fliter to locations roughly in the Pacific Northwest
system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  auk_bbox(bbox = c(-125, 37, -120, 52))
  
# alternatively, without pipes
ebd &lt;- auk_ebd(system.file("extdata/ebd-sample.txt", package = "auk"))
auk_bbox(ebd, bbox = c(-125, 37, -120, 52))
</code></pre>

<hr>
<h2 id='auk_filter'>Filter the eBird file using AWK</h2><span id='topic+auk_filter'></span><span id='topic+auk_filter.auk_ebd'></span><span id='topic+auk_filter.auk_sampling'></span>

<h3>Description</h3>

<p>Convert the filters defined in an <code>auk_ebd</code> object into an AWK script and run
this script to produce a filtered eBird Reference Dataset (ERD). The initial
creation of the <code>auk_ebd</code> object should be done with <code><a href="#topic+auk_ebd">auk_ebd()</a></code> and filters
can be defined using the various other functions in this package, e.g.
<code><a href="#topic+auk_species">auk_species()</a></code> or <code><a href="#topic+auk_country">auk_country()</a></code>. <strong>Note that this function typically takes
at least a couple hours to run on the full dataset</strong>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_filter(x, file, ...)

## S3 method for class 'auk_ebd'
auk_filter(
  x,
  file,
  file_sampling,
  keep,
  drop,
  awk_file,
  sep = "\t",
  filter_sampling = TRUE,
  execute = TRUE,
  overwrite = FALSE,
  ...
)

## S3 method for class 'auk_sampling'
auk_filter(
  x,
  file,
  keep,
  drop,
  awk_file,
  sep = "\t",
  execute = TRUE,
  overwrite = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_filter_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> or <code>auk_sampling</code> object; reference to file created by
<code><a href="#topic+auk_ebd">auk_ebd()</a></code> or <code><a href="#topic+auk_sampling">auk_sampling()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_filter_+3A_file">file</code></td>
<td>
<p>character; output file.</p>
</td></tr>
<tr><td><code id="auk_filter_+3A_...">...</code></td>
<td>
<p>arguments passed on to methods.</p>
</td></tr>
<tr><td><code id="auk_filter_+3A_file_sampling">file_sampling</code></td>
<td>
<p>character; optional output file for sampling data.</p>
</td></tr>
<tr><td><code id="auk_filter_+3A_keep">keep</code></td>
<td>
<p>character; a character vector specifying the names of the columns
to keep in the output file. Columns should be as they appear in the header
of the EBD; however, names are not case sensitive and spaces may be
replaced by underscores, e.g. <code>"COMMON NAME"</code>, <code>"common name"</code>, and
<code>"common_NAME"</code> are all valid.</p>
</td></tr>
<tr><td><code id="auk_filter_+3A_drop">drop</code></td>
<td>
<p>character; a character vector of columns to drop in the same
format as <code>keep</code>. Ignored if <code>keep</code> is supplied.</p>
</td></tr>
<tr><td><code id="auk_filter_+3A_awk_file">awk_file</code></td>
<td>
<p>character; output file to optionally save the awk script to.</p>
</td></tr>
<tr><td><code id="auk_filter_+3A_sep">sep</code></td>
<td>
<p>character; the input field separator, the eBird file is tab
separated by default. Must only be a single character and space delimited
is not allowed since spaces appear in many of the fields.</p>
</td></tr>
<tr><td><code id="auk_filter_+3A_filter_sampling">filter_sampling</code></td>
<td>
<p>logical; whether the sampling event data should also
be filtered.</p>
</td></tr>
<tr><td><code id="auk_filter_+3A_execute">execute</code></td>
<td>
<p>logical; whether to execute the awk script, or output it to a
file for manual execution. If this flag is <code>FALSE</code>, <code>awk_file</code> must be
provided.</p>
</td></tr>
<tr><td><code id="auk_filter_+3A_overwrite">overwrite</code></td>
<td>
<p>logical; overwrite output file if it already exists</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If a sampling file is provided in the <a href="#topic+auk_ebd">auk_ebd</a> object, this
function will filter both the eBird Basic Dataset and the sampling data using
the same set of filters. This ensures that the files are in sync, i.e. that
they contain data on the same set of checklists.
</p>
<p>The AWK script can be saved for future reference by providing an output
filename to <code>awk_file</code>. The default behavior of this function is to generate
and run the AWK script, however, by setting <code>execute = FALSE</code> the AWK script
will be generated but not run. In this case, <code>file</code> is ignored and <code>awk_file</code>
must be specified.
</p>
<p>Calling this function requires that the command line utility AWK is
installed. Linux and Mac machines should have AWK by default, Windows users
will likely need to install <a href="https://www.cygwin.com">Cygwin</a>.
</p>


<h3>Value</h3>

<p>An <code>auk_ebd</code> object with the output files set. If <code>execute = FALSE</code>,
then the path to the AWK script is returned instead.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>auk_filter(auk_ebd)</code>: <code>auk_ebd</code> object
</p>
</li>
<li> <p><code>auk_filter(auk_sampling)</code>: <code>auk_sampling</code> object
</p>
</li></ul>


<h3>See Also</h3>

<p>Other filter: 
<code><a href="#topic+auk_bbox">auk_bbox</a>()</code>,
<code><a href="#topic+auk_bcr">auk_bcr</a>()</code>,
<code><a href="#topic+auk_breeding">auk_breeding</a>()</code>,
<code><a href="#topic+auk_complete">auk_complete</a>()</code>,
<code><a href="#topic+auk_country">auk_country</a>()</code>,
<code><a href="#topic+auk_county">auk_county</a>()</code>,
<code><a href="#topic+auk_date">auk_date</a>()</code>,
<code><a href="#topic+auk_distance">auk_distance</a>()</code>,
<code><a href="#topic+auk_duration">auk_duration</a>()</code>,
<code><a href="#topic+auk_exotic">auk_exotic</a>()</code>,
<code><a href="#topic+auk_extent">auk_extent</a>()</code>,
<code><a href="#topic+auk_last_edited">auk_last_edited</a>()</code>,
<code><a href="#topic+auk_observer">auk_observer</a>()</code>,
<code><a href="#topic+auk_project">auk_project</a>()</code>,
<code><a href="#topic+auk_protocol">auk_protocol</a>()</code>,
<code><a href="#topic+auk_species">auk_species</a>()</code>,
<code><a href="#topic+auk_state">auk_state</a>()</code>,
<code><a href="#topic+auk_time">auk_time</a>()</code>,
<code><a href="#topic+auk_year">auk_year</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># get the path to the example data included in the package
# in practice, provide path to ebd, e.g. f &lt;- "data/ebd_relFeb-2018.txt"
f &lt;- system.file("extdata/ebd-sample.txt", package = "auk")
# define filters
filters &lt;- auk_ebd(f) %&gt;%
  auk_species(species = c("Canada Jay", "Blue Jay")) %&gt;%
  auk_country(country = c("US", "Canada")) %&gt;%
  auk_bbox(bbox = c(-100, 37, -80, 52)) %&gt;%
  auk_date(date = c("2012-01-01", "2012-12-31")) %&gt;%
  auk_time(start_time = c("06:00", "09:00")) %&gt;%
  auk_duration(duration = c(0, 60)) %&gt;%
  auk_complete()
  
# alternatively, without pipes
ebd &lt;- auk_ebd(system.file("extdata/ebd-sample.txt", package = "auk"))
filters &lt;- auk_species(ebd, species = c("Canada Jay", "Blue Jay"))
filters &lt;- auk_country(filters, country = c("US", "Canada"))
filters &lt;- auk_bbox(filters, bbox = c(-100, 37, -80, 52))
filters &lt;- auk_date(filters, date = c("2012-01-01", "2012-12-31"))
filters &lt;- auk_time(filters, start_time = c("06:00", "09:00"))
filters &lt;- auk_duration(filters, duration = c(0, 60))
filters &lt;- auk_complete(filters)

# apply filters
## Not run: 
# output to a temp file for example
# in practice, provide path to output file
# e.g. f_out &lt;- "output/ebd_filtered.txt"
f_out &lt;- tempfile()
filtered &lt;- auk_filter(filters, file = f_out)
str(read_ebd(filtered))

## End(Not run)
</code></pre>

<hr>
<h2 id='auk_get_awk_path'>OS specific path to AWK executable</h2><span id='topic+auk_get_awk_path'></span>

<h3>Description</h3>

<p>Return the OS specific path to AWK (e.g. <code>"C:/cygwin64/bin/gawk.exe"</code> or
<code>"/usr/bin/awk"</code>), or highlights if it's not installed. To manually set the
path to AWK, set the <code>AWK_PATH</code> environment variable in your <code>.Renviron</code>
file, which can be accomplished with the helper function
<code>auk_set_awk_path(path)</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_get_awk_path()
</code></pre>


<h3>Value</h3>

<p>Path to AWK or <code>NA</code> if AWK wasn't found.
</p>


<h3>See Also</h3>

<p>Other paths: 
<code><a href="#topic+auk_get_ebd_path">auk_get_ebd_path</a>()</code>,
<code><a href="#topic+auk_set_awk_path">auk_set_awk_path</a>()</code>,
<code><a href="#topic+auk_set_ebd_path">auk_set_ebd_path</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>auk_get_awk_path()
</code></pre>

<hr>
<h2 id='auk_get_ebd_path'>Return EBD data path</h2><span id='topic+auk_get_ebd_path'></span>

<h3>Description</h3>

<p>Returns the environment variable <code>EBD_PATH</code>, which users are encouraged to
set to the directory that stores the eBird Basic Dataset (EBD) text files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_get_ebd_path()
</code></pre>


<h3>Value</h3>

<p>The path stored in the <code>EBD_PATH</code> environment variable.
</p>


<h3>See Also</h3>

<p>Other paths: 
<code><a href="#topic+auk_get_awk_path">auk_get_awk_path</a>()</code>,
<code><a href="#topic+auk_set_awk_path">auk_set_awk_path</a>()</code>,
<code><a href="#topic+auk_set_ebd_path">auk_set_ebd_path</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>auk_get_ebd_path()
</code></pre>

<hr>
<h2 id='auk_last_edited'>Filter the eBird data by last edited date</h2><span id='topic+auk_last_edited'></span>

<h3>Description</h3>

<p>Define a filter for the eBird Basic Dataset (EBD) based on a range of last
edited dates. Last edited date is typically used to extract just new or
recently edited data. This function only defines the filter and, once all
filters have been defined, <code><a href="#topic+auk_filter">auk_filter()</a></code> should be used to call AWK and
perform the filtering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_last_edited(x, date)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_last_edited_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> or <code>auk_sampling</code> object; reference to file created by
<code><a href="#topic+auk_ebd">auk_ebd()</a></code> or <code><a href="#topic+auk_sampling">auk_sampling()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_last_edited_+3A_date">date</code></td>
<td>
<p>character or date; date range to filter by, provided either as a
character vector in the format <code>"2015-12-31"</code> or a vector of Date objects.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can also work with on an <code>auk_sampling</code> object if the
user only wishes to filter the sampling event data.
</p>


<h3>Value</h3>

<p>An <code>auk_ebd</code> object.
</p>


<h3>See Also</h3>

<p>Other filter: 
<code><a href="#topic+auk_bbox">auk_bbox</a>()</code>,
<code><a href="#topic+auk_bcr">auk_bcr</a>()</code>,
<code><a href="#topic+auk_breeding">auk_breeding</a>()</code>,
<code><a href="#topic+auk_complete">auk_complete</a>()</code>,
<code><a href="#topic+auk_country">auk_country</a>()</code>,
<code><a href="#topic+auk_county">auk_county</a>()</code>,
<code><a href="#topic+auk_date">auk_date</a>()</code>,
<code><a href="#topic+auk_distance">auk_distance</a>()</code>,
<code><a href="#topic+auk_duration">auk_duration</a>()</code>,
<code><a href="#topic+auk_exotic">auk_exotic</a>()</code>,
<code><a href="#topic+auk_extent">auk_extent</a>()</code>,
<code><a href="#topic+auk_filter">auk_filter</a>()</code>,
<code><a href="#topic+auk_observer">auk_observer</a>()</code>,
<code><a href="#topic+auk_project">auk_project</a>()</code>,
<code><a href="#topic+auk_protocol">auk_protocol</a>()</code>,
<code><a href="#topic+auk_species">auk_species</a>()</code>,
<code><a href="#topic+auk_state">auk_state</a>()</code>,
<code><a href="#topic+auk_time">auk_time</a>()</code>,
<code><a href="#topic+auk_year">auk_year</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  auk_last_edited(date = c("2010-01-01", "2010-12-31"))
</code></pre>

<hr>
<h2 id='auk_observer'>Filter the eBird data by observer</h2><span id='topic+auk_observer'></span>

<h3>Description</h3>

<p>Define a filter for the eBird Basic Dataset (EBD) based on a set of
observer IDs This function only defines the filter and, once all filters have
been defined, <code><a href="#topic+auk_filter">auk_filter()</a></code> should be used to call AWK and perform the
filtering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_observer(x, observer_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_observer_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> or <code>auk_sampling</code> object; reference to file created by
<code><a href="#topic+auk_ebd">auk_ebd()</a></code> or <code><a href="#topic+auk_sampling">auk_sampling()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_observer_+3A_observer_id">observer_id</code></td>
<td>
<p>character or integer; observers to filter by. Observer IDs
can be provided either as integer (e.g. 12345) or character with the &quot;obsr&quot;
prefix as they appear in the EBD (e.g. &quot;obsr12345&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>auk_ebd</code> or 'auk_sampling&ldquo; object.
</p>


<h3>See Also</h3>

<p>Other filter: 
<code><a href="#topic+auk_bbox">auk_bbox</a>()</code>,
<code><a href="#topic+auk_bcr">auk_bcr</a>()</code>,
<code><a href="#topic+auk_breeding">auk_breeding</a>()</code>,
<code><a href="#topic+auk_complete">auk_complete</a>()</code>,
<code><a href="#topic+auk_country">auk_country</a>()</code>,
<code><a href="#topic+auk_county">auk_county</a>()</code>,
<code><a href="#topic+auk_date">auk_date</a>()</code>,
<code><a href="#topic+auk_distance">auk_distance</a>()</code>,
<code><a href="#topic+auk_duration">auk_duration</a>()</code>,
<code><a href="#topic+auk_exotic">auk_exotic</a>()</code>,
<code><a href="#topic+auk_extent">auk_extent</a>()</code>,
<code><a href="#topic+auk_filter">auk_filter</a>()</code>,
<code><a href="#topic+auk_last_edited">auk_last_edited</a>()</code>,
<code><a href="#topic+auk_project">auk_project</a>()</code>,
<code><a href="#topic+auk_protocol">auk_protocol</a>()</code>,
<code><a href="#topic+auk_species">auk_species</a>()</code>,
<code><a href="#topic+auk_state">auk_state</a>()</code>,
<code><a href="#topic+auk_time">auk_time</a>()</code>,
<code><a href="#topic+auk_year">auk_year</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  auk_observer("obsr313215")
  
# alternatively, without pipes
ebd &lt;- auk_ebd(system.file("extdata/ebd-sample.txt", package = "auk"))
auk_observer(ebd, observer = 313215)
</code></pre>

<hr>
<h2 id='auk_project'>Filter the eBird data by project code</h2><span id='topic+auk_project'></span>

<h3>Description</h3>

<p>Some eBird records are collected as part of a particular project (e.g. the
Virginia Breeding Bird Survey) and have an associated project code in the
eBird dataset (e.g. EBIRD_ATL_VA). This function only defines the filter and,
once all filters have been defined, <code><a href="#topic+auk_filter">auk_filter()</a></code> should be used to call AWK
and perform the filtering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_project(x, project)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_project_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> or <code>auk_sampling</code> object; reference to file created by
<code><a href="#topic+auk_ebd">auk_ebd()</a></code> or <code><a href="#topic+auk_sampling">auk_sampling()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_project_+3A_project">project</code></td>
<td>
<p>character; project code to filter by (e.g. <code>"EBIRD_MEX"</code>).
Multiple codes are accepted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can also work with on an <code>auk_sampling</code> object if the
user only wishes to filter the sampling event data.
</p>


<h3>Value</h3>

<p>An <code>auk_ebd</code> object.
</p>


<h3>See Also</h3>

<p>Other filter: 
<code><a href="#topic+auk_bbox">auk_bbox</a>()</code>,
<code><a href="#topic+auk_bcr">auk_bcr</a>()</code>,
<code><a href="#topic+auk_breeding">auk_breeding</a>()</code>,
<code><a href="#topic+auk_complete">auk_complete</a>()</code>,
<code><a href="#topic+auk_country">auk_country</a>()</code>,
<code><a href="#topic+auk_county">auk_county</a>()</code>,
<code><a href="#topic+auk_date">auk_date</a>()</code>,
<code><a href="#topic+auk_distance">auk_distance</a>()</code>,
<code><a href="#topic+auk_duration">auk_duration</a>()</code>,
<code><a href="#topic+auk_exotic">auk_exotic</a>()</code>,
<code><a href="#topic+auk_extent">auk_extent</a>()</code>,
<code><a href="#topic+auk_filter">auk_filter</a>()</code>,
<code><a href="#topic+auk_last_edited">auk_last_edited</a>()</code>,
<code><a href="#topic+auk_observer">auk_observer</a>()</code>,
<code><a href="#topic+auk_protocol">auk_protocol</a>()</code>,
<code><a href="#topic+auk_species">auk_species</a>()</code>,
<code><a href="#topic+auk_state">auk_state</a>()</code>,
<code><a href="#topic+auk_time">auk_time</a>()</code>,
<code><a href="#topic+auk_year">auk_year</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  auk_project("EBIRD_MEX")
  
# alternatively, without pipes
ebd &lt;- auk_ebd(system.file("extdata/ebd-sample.txt", package = "auk"))
auk_project(ebd, "EBIRD_MEX")
</code></pre>

<hr>
<h2 id='auk_protocol'>Filter the eBird data by protocol</h2><span id='topic+auk_protocol'></span>

<h3>Description</h3>

<p>Filter to just data collected following a specific search protocol:
stationary, traveling, or casual. This function only defines the filter and,
once all filters have been defined, <code><a href="#topic+auk_filter">auk_filter()</a></code> should be used to call AWK
and perform the filtering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_protocol(x, protocol)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_protocol_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> or <code>auk_sampling</code> object; reference to file created by
<code><a href="#topic+auk_ebd">auk_ebd()</a></code> or <code><a href="#topic+auk_sampling">auk_sampling()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_protocol_+3A_protocol">protocol</code></td>
<td>
<p>character. Many protocols exist in the database, however, the
most commonly used are:
</p>

<ul>
<li><p> Stationary
</p>
</li>
<li><p> Traveling
</p>
</li>
<li><p> Area
</p>
</li>
<li><p> Incidental
</p>
</li></ul>

<p>A complete list of valid protocols is contained within the vector
<code>valid_protocols</code> within this package. Multiple protocols are allowed at
the same time.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can also work with on an <code>auk_sampling</code> object if the
user only wishes to filter the sampling event data.
</p>


<h3>Value</h3>

<p>An <code>auk_ebd</code> object.
</p>


<h3>See Also</h3>

<p>Other filter: 
<code><a href="#topic+auk_bbox">auk_bbox</a>()</code>,
<code><a href="#topic+auk_bcr">auk_bcr</a>()</code>,
<code><a href="#topic+auk_breeding">auk_breeding</a>()</code>,
<code><a href="#topic+auk_complete">auk_complete</a>()</code>,
<code><a href="#topic+auk_country">auk_country</a>()</code>,
<code><a href="#topic+auk_county">auk_county</a>()</code>,
<code><a href="#topic+auk_date">auk_date</a>()</code>,
<code><a href="#topic+auk_distance">auk_distance</a>()</code>,
<code><a href="#topic+auk_duration">auk_duration</a>()</code>,
<code><a href="#topic+auk_exotic">auk_exotic</a>()</code>,
<code><a href="#topic+auk_extent">auk_extent</a>()</code>,
<code><a href="#topic+auk_filter">auk_filter</a>()</code>,
<code><a href="#topic+auk_last_edited">auk_last_edited</a>()</code>,
<code><a href="#topic+auk_observer">auk_observer</a>()</code>,
<code><a href="#topic+auk_project">auk_project</a>()</code>,
<code><a href="#topic+auk_species">auk_species</a>()</code>,
<code><a href="#topic+auk_state">auk_state</a>()</code>,
<code><a href="#topic+auk_time">auk_time</a>()</code>,
<code><a href="#topic+auk_year">auk_year</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  auk_protocol("Stationary")
  
# alternatively, without pipes
ebd &lt;- auk_ebd(system.file("extdata/ebd-sample.txt", package = "auk"))
auk_protocol(ebd, "Stationary")
</code></pre>

<hr>
<h2 id='auk_rollup'>Roll up eBird taxonomy to species</h2><span id='topic+auk_rollup'></span>

<h3>Description</h3>

<p>The eBird Basic Dataset (EBD) includes both true species and every other
field-identifiable taxon that could be relevant for birders to report. This
includes taxa not identifiable to a species (e.g. hybrids) and taxa reported
below the species level (e.g. subspecies). This function produces a list of
observations of true species, by removing the former and rolling the latter
up to the species level. In the resulting EBD data.frame,
<code>category</code> will be <code>"species"</code> for all records and the subspecies fields will
be dropped. By default, <code><a href="#topic+read_ebd">read_ebd()</a></code> calls <code>ebd_rollup()</code> when importing an
eBird data file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_rollup(x, taxonomy_version, drop_higher = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_rollup_+3A_x">x</code></td>
<td>
<p>data.frame; data frame of eBird data, typically as imported by
<code><a href="#topic+read_ebd">read_ebd()</a></code></p>
</td></tr>
<tr><td><code id="auk_rollup_+3A_taxonomy_version">taxonomy_version</code></td>
<td>
<p>integer; the version (i.e. year) of the taxonomy. In
most cases, this should be left empty to use the version of the taxonomy
included in the package. See <code><a href="#topic+get_ebird_taxonomy">get_ebird_taxonomy()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_rollup_+3A_drop_higher">drop_higher</code></td>
<td>
<p>logical; whether to remove taxa above species during the
rollup process, e.g. &quot;spuhs&quot; like &quot;duck sp.&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When rolling observations up to species level the observed counts
are summed across any taxa that resolve to the same species. However, if
any of these taxa have a count of &quot;X&quot; (i.e. the observer did not enter a
count), then the rolled up record will get an &quot;X&quot; as well. For example, if
an observer saw 3 Myrtle and 2 Audubon's Warblers, this will roll up to 5
Yellow-rumped Warblers. However, if an &quot;X&quot; was entered for Myrtle, this
would roll up to &quot;X&quot; for Yellow-rumped Warbler.
</p>
<p>The eBird taxonomy groups taxa into eight different categories. These
categories, and the way they are treated by <code><a href="#topic+auk_rollup">auk_rollup()</a></code> are as follows:
</p>

<ul>
<li> <p><strong>Species:</strong> e.g., Mallard. Combined with lower level taxa if present on
the same checklist.
</p>
</li>
<li> <p><strong>ISSF or Identifiable Sub-specific Group:</strong> Identifiable subspecies or
group of subspecies, e.g., Mallard (Mexican). Rolled-up to species level.
</p>
</li>
<li> <p><strong>Intergrade:</strong> Hybrid between two ISSF (subspecies or subspecies
groups), e.g., Mallard (Mexican intergrade. Rolled-up to species level.
</p>
</li>
<li> <p><strong>Form:</strong> Miscellaneous other taxa, including recently-described species
yet to be accepted or distinctive forms that are not universally accepted
(Red-tailed Hawk (Northern), Upland Goose (Bar-breasted)). If the checklist
contains multiple taxa corresponding to the same species, the lower level
taxa are rolled up, otherwise these records are left as is.
</p>
</li>
<li> <p><strong>Spuh:</strong>  Genus or identification at broad level &ndash; e.g., duck sp.,
dabbling duck sp.. Dropped by <code>auk_rollup()</code>.
</p>
</li>
<li> <p><strong>Slash:</strong> Identification to Species-pair e.g., American Black
Duck/Mallard). Dropped by <code>auk_rollup()</code>.
</p>
</li>
<li> <p><strong>Hybrid:</strong> Hybrid between two species, e.g., American Black Duck x
Mallard (hybrid). Dropped by <code>auk_rollup()</code>.
</p>
</li>
<li> <p><strong>Domestic:</strong> Distinctly-plumaged domesticated varieties that may be
free-flying (these do not count on personal lists) e.g., Mallard (Domestic
type). Dropped by <code>auk_rollup()</code>.
</p>
</li></ul>

<p>The rollup process is based on the eBird taxonomy, which is updated once a
year in August. The <code>auk</code> package includes a copy of the eBird taxonomy,
current at the time of release; however, if the EBD and <code>auk</code> versions are
not aligned, you may need to explicitly specify which version of the
taxonomy to use, in which case the eBird API will be queried to get the
correct version of the taxonomy.
</p>


<h3>Value</h3>

<p>A data frame of the eBird data with taxonomic rollup applied.
</p>


<h3>References</h3>

<p>Consult the <a href="https://ebird.org/science/use-ebird-data/the-ebird-taxonomy">eBird taxonomy</a>
page for further details.
</p>


<h3>See Also</h3>

<p>Other pre: 
<code><a href="#topic+auk_unique">auk_unique</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># get the path to the example data included in the package
# in practice, provide path to ebd, e.g. f &lt;- "data/ebd_relFeb-2018.txt
f &lt;- system.file("extdata/ebd-rollup-ex.txt", package = "auk")
# read in data without rolling up
ebd &lt;- read_ebd(f, rollup = FALSE)
# rollup
ebd_ru &lt;- auk_rollup(ebd)
# keep higher taxa
ebd_higher &lt;- auk_rollup(ebd, drop_higher = FALSE)

# all taxa not identifiable to species are dropped
unique(ebd$category)
unique(ebd_ru$category)
unique(ebd_higher$category)

# yellow-rump warbler subspecies rollup
library(dplyr)
# without rollup, there are three observations
ebd %&gt;%
  filter(common_name == "Yellow-rumped Warbler") %&gt;% 
  select(checklist_id, category, common_name, subspecies_common_name, 
         observation_count)
# with rollup, they have been combined
ebd_ru %&gt;%
  filter(common_name == "Yellow-rumped Warbler") %&gt;% 
  select(checklist_id, category, common_name, observation_count)
</code></pre>

<hr>
<h2 id='auk_sampling'>Reference to eBird sampling event file</h2><span id='topic+auk_sampling'></span>

<h3>Description</h3>

<p>Create a reference to an eBird sampling event file in preparation for
filtering using AWK. For working with the sightings data use <code>auk_ebd()</code>,
only use <code>auk_sampling()</code> if you intend to only work with checklist-level
data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_sampling(file, sep = "\t")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_sampling_+3A_file">file</code></td>
<td>
<p>character; input sampling event data file, which contains
checklist data from eBird.</p>
</td></tr>
<tr><td><code id="auk_sampling_+3A_sep">sep</code></td>
<td>
<p>character; the input field separator, the eBird data are tab
separated so this should generally not be modified. Must only be a single
character and space delimited is not allowed since spaces appear in many of
the fields.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>eBird data can be downloaded as a tab-separated text file from the
<a href="http://ebird.org/ebird/data/download">eBird website</a> after submitting a
request for access. In the eBird Basic Dataset (EBD) each row corresponds
to a observation of a single bird species on a single checklist, while the
sampling event data file contains a single row for every checklist. This
function creates an R object to reference only the sampling data.
</p>


<h3>Value</h3>

<p>An <code>auk_sampling</code> object storing the file reference and the desired
filters once created with other package functions.
</p>


<h3>See Also</h3>

<p>Other objects: 
<code><a href="#topic+auk_ebd">auk_ebd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># get the path to the example data included in the package
# in practice, provide path to the sampling event data
# e.g. f &lt;- "data/ebd_sampling_relFeb-2018.txt"
f &lt;- system.file("extdata/zerofill-ex_sampling.txt", package = "auk")
auk_sampling(f)
</code></pre>

<hr>
<h2 id='auk_select'>Select a subset of columns</h2><span id='topic+auk_select'></span>

<h3>Description</h3>

<p>Select a subset of columns from the eBird Basic Dataset (EBD) or the sampling
events file. Subsetting the columns can significantly decrease file size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_select(x, select, file, sep = "\t", overwrite = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_select_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> or <code>auk_sampling</code> object; reference to file created by
<code><a href="#topic+auk_ebd">auk_ebd()</a></code> or <code><a href="#topic+auk_sampling">auk_sampling()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_select_+3A_select">select</code></td>
<td>
<p>character; a character vector specifying the names of the
columns to select. Columns should be as they appear in the header of the
EBD; however, names are not case sensitive and spaces may be replaced by
underscores, e.g. <code>"COMMON NAME"</code>, <code>"common name"</code>, and <code>"common_NAME"</code> are
all valid.</p>
</td></tr>
<tr><td><code id="auk_select_+3A_file">file</code></td>
<td>
<p>character; output file.</p>
</td></tr>
<tr><td><code id="auk_select_+3A_sep">sep</code></td>
<td>
<p>character; the input field separator, the eBird file is tab
separated by default. Must only be a single character and space delimited
is not allowed since spaces appear in many of the fields.</p>
</td></tr>
<tr><td><code id="auk_select_+3A_overwrite">overwrite</code></td>
<td>
<p>logical; overwrite output file if it already exists</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly returns the filename of the output file.
</p>


<h3>See Also</h3>

<p>Other text: 
<code><a href="#topic+auk_clean">auk_clean</a>()</code>,
<code><a href="#topic+auk_split">auk_split</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# select a minimal set of columns
out_file &lt;- tempfile()
ebd &lt;- auk_ebd(system.file("extdata/ebd-sample.txt", package = "auk"))
cols &lt;- c("latitude", "longitude",
          "group identifier", "sampling event identifier", 
          "scientific name", "observation count",
          "observer_id")
selected &lt;- auk_select(ebd, select = cols, file = out_file)
str(read_ebd(selected))

## End(Not run)
</code></pre>

<hr>
<h2 id='auk_set_awk_path'>Set a custom path to AWK executable</h2><span id='topic+auk_set_awk_path'></span>

<h3>Description</h3>

<p>If AWK has been installed in a non-standard location, the environment
variable <code>AWK_PATH</code> must be set to specify the location of the executable.
Use this function to set <code>AWK_PATH</code> in your .Renviron file. <strong>Most users
should NOT set <code>AWK_PATH</code>, only do so if you have installed AWK in
non-standard location and <code>auk</code> cannot find it.</strong> This function first looks
for for an .Renviron location defined by <code>R_ENVIRON_USER</code>, then defaults to
~/.Renviron.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_set_awk_path(path, overwrite = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_set_awk_path_+3A_path">path</code></td>
<td>
<p>character; path to the AWK executable on your system, e.g.
<code>"C:/cygwin64/bin/gawk.exe"</code> or <code>"/usr/bin/awk"</code>.</p>
</td></tr>
<tr><td><code id="auk_set_awk_path_+3A_overwrite">overwrite</code></td>
<td>
<p>logical; should the existing <code>AWK_PATH</code> be overwritten if it
has already been set in .Renviron.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Edits .Renviron, sets <code>AWK_PATH</code> for the current session, then
returns the EBD path invisibly.
</p>


<h3>See Also</h3>

<p>Other paths: 
<code><a href="#topic+auk_get_awk_path">auk_get_awk_path</a>()</code>,
<code><a href="#topic+auk_get_ebd_path">auk_get_ebd_path</a>()</code>,
<code><a href="#topic+auk_set_ebd_path">auk_set_ebd_path</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
auk_set_awk_path("/usr/bin/awk")

## End(Not run)
</code></pre>

<hr>
<h2 id='auk_set_ebd_path'>Set the path to EBD text files</h2><span id='topic+auk_set_ebd_path'></span>

<h3>Description</h3>

<p>Users of <code>auk</code> are encouraged to set the path to the directory containing the
eBird Basic Dataset (EBD) text files in the <code>EBD_PATH</code> environment variable.
All functions referencing the EBD or sampling event data files will check in
this directory to find the files, thus avoiding the need to specify the full
path every time. This will increase the portability of your code. Use this
function to set <code>EBD_PATH</code> in your .Renviron file; it is also possible to
manually edit the file. This function first looks for for an .Renviron
location defined by <code>R_ENVIRON_USER</code>, then defaults to ~/.Renviron.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_set_ebd_path(path, overwrite = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_set_ebd_path_+3A_path">path</code></td>
<td>
<p>character; directory where the EBD text files are stored, e.g.
<code>"/home/matt/ebd"</code>.</p>
</td></tr>
<tr><td><code id="auk_set_ebd_path_+3A_overwrite">overwrite</code></td>
<td>
<p>logical; should the existing <code>EBD_PATH</code> be overwritten if it
has already been set in .Renviron.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Edits .Renviron, sets <code>EBD_PATH</code> for the current session, then
returns the EBD path invisibly.
</p>


<h3>See Also</h3>

<p>Other paths: 
<code><a href="#topic+auk_get_awk_path">auk_get_awk_path</a>()</code>,
<code><a href="#topic+auk_get_ebd_path">auk_get_ebd_path</a>()</code>,
<code><a href="#topic+auk_set_awk_path">auk_set_awk_path</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
auk_set_ebd_path("/home/matt/ebd")

## End(Not run)
</code></pre>

<hr>
<h2 id='auk_species'>Filter the eBird data by species</h2><span id='topic+auk_species'></span>

<h3>Description</h3>

<p>Define a filter for the eBird Basic Dataset (EBD) based on species. This
function only defines the filter and, once all filters have been defined,
<code><a href="#topic+auk_filter">auk_filter()</a></code> should be used to call AWK and perform the filtering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_species(x, species, taxonomy_version, replace = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_species_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> object; reference to object created by <code><a href="#topic+auk_ebd">auk_ebd()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_species_+3A_species">species</code></td>
<td>
<p>character; species to filter by, provided as scientific or
English common names, or a mixture of both. These names must match the
official eBird Taxomony (<a href="#topic+ebird_taxonomy">ebird_taxonomy</a>).</p>
</td></tr>
<tr><td><code id="auk_species_+3A_taxonomy_version">taxonomy_version</code></td>
<td>
<p>integer; the version (i.e. year) of the taxonomy. In
most cases, this should be left empty to use the version of the taxonomy
included in the package. See <code><a href="#topic+get_ebird_taxonomy">get_ebird_taxonomy()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_species_+3A_replace">replace</code></td>
<td>
<p>logical; multiple calls to <code>auk_species()</code> are additive,
unless <code>replace = FALSE</code>, in which case the previous list of species to
filter by will be removed and replaced by that in the current call.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The list of species is checked against the eBird taxonomy for
validity. This taxonomy is updated once a year in August. The <code>auk</code> package
includes a copy of the eBird taxonomy, current at the time of release;
however, if the EBD and <code>auk</code> versions are not aligned, you may need to
explicitly specify which version of the taxonomy to use, in which case
the eBird API will be queried to get the correct version of the taxonomy.
</p>


<h3>Value</h3>

<p>An <code>auk_ebd</code> object.
</p>


<h3>See Also</h3>

<p>Other filter: 
<code><a href="#topic+auk_bbox">auk_bbox</a>()</code>,
<code><a href="#topic+auk_bcr">auk_bcr</a>()</code>,
<code><a href="#topic+auk_breeding">auk_breeding</a>()</code>,
<code><a href="#topic+auk_complete">auk_complete</a>()</code>,
<code><a href="#topic+auk_country">auk_country</a>()</code>,
<code><a href="#topic+auk_county">auk_county</a>()</code>,
<code><a href="#topic+auk_date">auk_date</a>()</code>,
<code><a href="#topic+auk_distance">auk_distance</a>()</code>,
<code><a href="#topic+auk_duration">auk_duration</a>()</code>,
<code><a href="#topic+auk_exotic">auk_exotic</a>()</code>,
<code><a href="#topic+auk_extent">auk_extent</a>()</code>,
<code><a href="#topic+auk_filter">auk_filter</a>()</code>,
<code><a href="#topic+auk_last_edited">auk_last_edited</a>()</code>,
<code><a href="#topic+auk_observer">auk_observer</a>()</code>,
<code><a href="#topic+auk_project">auk_project</a>()</code>,
<code><a href="#topic+auk_protocol">auk_protocol</a>()</code>,
<code><a href="#topic+auk_state">auk_state</a>()</code>,
<code><a href="#topic+auk_time">auk_time</a>()</code>,
<code><a href="#topic+auk_year">auk_year</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># common and scientific names can be mixed
species &lt;- c("Canada Jay", "Pluvialis squatarola")
system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  auk_species(species)
  
# alternatively, without pipes
ebd &lt;- auk_ebd(system.file("extdata/ebd-sample.txt", package = "auk"))
auk_species(ebd, species)
</code></pre>

<hr>
<h2 id='auk_split'>Split an eBird data file by species</h2><span id='topic+auk_split'></span>

<h3>Description</h3>

<p>Given an eBird Basic Dataset (EBD) and a list of species, split the file into
multiple text files, one for each species. This function is typically used
after <code><a href="#topic+auk_filter">auk_filter()</a></code> has been applied if the resulting file is too large to
be read in all at once.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_split(
  file,
  species,
  prefix,
  taxonomy_version,
  sep = "\t",
  overwrite = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_split_+3A_file">file</code></td>
<td>
<p>character; input file.</p>
</td></tr>
<tr><td><code id="auk_split_+3A_species">species</code></td>
<td>
<p>character; species to filter and split by, provided as
scientific or English common names, or a mixture of both. These names must
match the official eBird Taxomony (<a href="#topic+ebird_taxonomy">ebird_taxonomy</a>).</p>
</td></tr>
<tr><td><code id="auk_split_+3A_prefix">prefix</code></td>
<td>
<p>character; a file and directory prefix. For example, if
splitting by species &quot;A&quot; and &quot;B&quot; and <code>prefix = "data/ebd_"</code>, the resulting
files will be &quot;data/ebd_A.txt&quot; and &quot;data/ebd_B.txt&quot;.</p>
</td></tr>
<tr><td><code id="auk_split_+3A_taxonomy_version">taxonomy_version</code></td>
<td>
<p>integer; the version (i.e. year) of the taxonomy. In
most cases, this should be left empty to use the version of the taxonomy
included in the package. See <code><a href="#topic+get_ebird_taxonomy">get_ebird_taxonomy()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_split_+3A_sep">sep</code></td>
<td>
<p>character; the input field separator, the eBird file is tab
separated by default. Must only be a single character and space delimited
is not allowed since spaces appear in many of the fields.</p>
</td></tr>
<tr><td><code id="auk_split_+3A_overwrite">overwrite</code></td>
<td>
<p>logical; overwrite output files if they already exists.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The list of species is checked against the eBird taxonomy for
validity. This taxonomy is updated once a year in August. The <code>auk</code> package
includes a copy of the eBird taxonomy, current at the time of release;
however, if the EBD and <code>auk</code> versions are not aligned, you may need to
explicitly specify which version of the taxonomy to use, in which case
the eBird API will be queried to get the correct version of the taxonomy.
</p>


<h3>Value</h3>

<p>A vector of output filenames, one for each species.
</p>


<h3>See Also</h3>

<p>Other text: 
<code><a href="#topic+auk_clean">auk_clean</a>()</code>,
<code><a href="#topic+auk_select">auk_select</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
species &lt;- c("Canada Jay", "Cyanocitta stelleri")
# get the path to the example data included in the package
# in practice, provide path to a filtered ebd file
# e.g. f &lt;- "data/ebd_filtered.txt
f &lt;- system.file("extdata/ebd-sample.txt", package = "auk")
# output to a temporary directory for example
# in practice, provide the path to the output location
# e.g. prefix &lt;- "output/ebd_"
prefix &lt;- file.path(tempdir(), "ebd_")
species_files &lt;- auk_split(f, species = species, prefix = prefix)

## End(Not run)
</code></pre>

<hr>
<h2 id='auk_state'>Filter the eBird data by state</h2><span id='topic+auk_state'></span>

<h3>Description</h3>

<p>Define a filter for the eBird Basic Dataset (EBD) based on a set of
states. This function only defines the filter and, once all filters have
been defined, <code><a href="#topic+auk_filter">auk_filter()</a></code> should be used to call AWK and perform the
filtering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_state(x, state, replace = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_state_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> or <code>auk_sampling</code> object; reference to file created by
<code><a href="#topic+auk_ebd">auk_ebd()</a></code> or <code><a href="#topic+auk_sampling">auk_sampling()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_state_+3A_state">state</code></td>
<td>
<p>character; states to filter by. eBird uses 4 to 6 character
state codes consisting of two parts, the 2-letter ISO country code and a
1-3 character state code, separated by a dash. For example, <code>"US-NY"</code>
corresponds to New York State in the United States. Refer to the data frame
<a href="#topic+ebird_states">ebird_states</a> for look up state codes.</p>
</td></tr>
<tr><td><code id="auk_state_+3A_replace">replace</code></td>
<td>
<p>logical; multiple calls to <code>auk_state()</code> are additive,
unless <code>replace = FALSE</code>, in which case the previous list of states to
filter by will be removed and replaced by that in the current call.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is not possible to filter by both country and state, so calling
<code>auk_state()</code> will reset the country filter to all countries, and vice versa.
</p>
<p>This function can also work with on an <code>auk_sampling</code> object if the user only
wishes to filter the sampling event data.
</p>


<h3>Value</h3>

<p>An <code>auk_ebd</code> object.
</p>


<h3>See Also</h3>

<p>Other filter: 
<code><a href="#topic+auk_bbox">auk_bbox</a>()</code>,
<code><a href="#topic+auk_bcr">auk_bcr</a>()</code>,
<code><a href="#topic+auk_breeding">auk_breeding</a>()</code>,
<code><a href="#topic+auk_complete">auk_complete</a>()</code>,
<code><a href="#topic+auk_country">auk_country</a>()</code>,
<code><a href="#topic+auk_county">auk_county</a>()</code>,
<code><a href="#topic+auk_date">auk_date</a>()</code>,
<code><a href="#topic+auk_distance">auk_distance</a>()</code>,
<code><a href="#topic+auk_duration">auk_duration</a>()</code>,
<code><a href="#topic+auk_exotic">auk_exotic</a>()</code>,
<code><a href="#topic+auk_extent">auk_extent</a>()</code>,
<code><a href="#topic+auk_filter">auk_filter</a>()</code>,
<code><a href="#topic+auk_last_edited">auk_last_edited</a>()</code>,
<code><a href="#topic+auk_observer">auk_observer</a>()</code>,
<code><a href="#topic+auk_project">auk_project</a>()</code>,
<code><a href="#topic+auk_protocol">auk_protocol</a>()</code>,
<code><a href="#topic+auk_species">auk_species</a>()</code>,
<code><a href="#topic+auk_time">auk_time</a>()</code>,
<code><a href="#topic+auk_year">auk_year</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># state codes for a given country can be looked up in ebird_states
dplyr::filter(ebird_states, country == "Costa Rica")
# choose texas, united states and puntarenas, cost rica
states &lt;- c("US-TX", "CR-P")
system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  auk_state(states)
  
# alternatively, without pipes
ebd &lt;- auk_ebd(system.file("extdata/ebd-sample.txt", package = "auk"))
auk_state(ebd, states)
</code></pre>

<hr>
<h2 id='auk_time'>Filter the eBird data by checklist start time</h2><span id='topic+auk_time'></span>

<h3>Description</h3>

<p>Define a filter for the eBird Basic Dataset (EBD) based on a range of start
times for the checklist. This function only defines the filter and, once all
filters have been defined, <code><a href="#topic+auk_filter">auk_filter()</a></code> should be used to call AWK and
perform the filtering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_time(x, start_time)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_time_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> or <code>auk_sampling</code> object; reference to file created by
<code><a href="#topic+auk_ebd">auk_ebd()</a></code> or <code><a href="#topic+auk_sampling">auk_sampling()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_time_+3A_start_time">start_time</code></td>
<td>
<p>character; 2 element character vector giving the range of
times in 24 hour format, e.g. <code>"06:30"</code> or <code>"16:22"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can also work with on an <code>auk_sampling</code> object if the
user only wishes to filter the sampling event data.
</p>


<h3>Value</h3>

<p>An <code>auk_ebd</code> object.
</p>


<h3>See Also</h3>

<p>Other filter: 
<code><a href="#topic+auk_bbox">auk_bbox</a>()</code>,
<code><a href="#topic+auk_bcr">auk_bcr</a>()</code>,
<code><a href="#topic+auk_breeding">auk_breeding</a>()</code>,
<code><a href="#topic+auk_complete">auk_complete</a>()</code>,
<code><a href="#topic+auk_country">auk_country</a>()</code>,
<code><a href="#topic+auk_county">auk_county</a>()</code>,
<code><a href="#topic+auk_date">auk_date</a>()</code>,
<code><a href="#topic+auk_distance">auk_distance</a>()</code>,
<code><a href="#topic+auk_duration">auk_duration</a>()</code>,
<code><a href="#topic+auk_exotic">auk_exotic</a>()</code>,
<code><a href="#topic+auk_extent">auk_extent</a>()</code>,
<code><a href="#topic+auk_filter">auk_filter</a>()</code>,
<code><a href="#topic+auk_last_edited">auk_last_edited</a>()</code>,
<code><a href="#topic+auk_observer">auk_observer</a>()</code>,
<code><a href="#topic+auk_project">auk_project</a>()</code>,
<code><a href="#topic+auk_protocol">auk_protocol</a>()</code>,
<code><a href="#topic+auk_species">auk_species</a>()</code>,
<code><a href="#topic+auk_state">auk_state</a>()</code>,
<code><a href="#topic+auk_year">auk_year</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># only keep checklists started between 6 and 8 in the morning
system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  auk_time(start_time = c("06:00", "08:00"))
  
# alternatively, without pipes
ebd &lt;- auk_ebd(system.file("extdata/ebd-sample.txt", package = "auk"))
auk_time(ebd, start_time = c("06:00", "08:00"))
</code></pre>

<hr>
<h2 id='auk_unique'>Remove duplicate group checklists</h2><span id='topic+auk_unique'></span>

<h3>Description</h3>

<p>eBird checklists can be shared among a group of multiple observers, in which
case observations will be duplicated in the database. This functions removes
these duplicates from the eBird Basic Dataset (EBD) or the EBD sampling event
data (with <code>checklists_only = TRUE</code>), creating a set of unique bird
observations. This function is called automatically by <code><a href="#topic+read_ebd">read_ebd()</a></code> and
<code><a href="#topic+read_sampling">read_sampling()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_unique(
  x,
  group_id = "group_identifier",
  checklist_id = "sampling_event_identifier",
  species_id = "scientific_name",
  observer_id = "observer_id",
  checklists_only = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_unique_+3A_x">x</code></td>
<td>
<p>data.frame; the EBD data frame, typically as imported by
<code><a href="#topic+read_ebd">read_ebd()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_unique_+3A_group_id">group_id</code></td>
<td>
<p>character; the name of the group ID column.</p>
</td></tr>
<tr><td><code id="auk_unique_+3A_checklist_id">checklist_id</code></td>
<td>
<p>character; the name of the checklist ID column, each
checklist within a group will get a unique value for this field. The record
with the lowest <code>checklist_id</code> will be picked as the unique record within
each group. In the output dataset, this field will be updated to have a
full list of the checklist IDs that went into this group checklist.</p>
</td></tr>
<tr><td><code id="auk_unique_+3A_species_id">species_id</code></td>
<td>
<p>character; the name of the column identifying species
uniquely. This is required to ensure that removing duplicates is done
independently for each species. Note that this will not treat sub-species
independently and, if that behavior is desired, the user will have to
generate a column uniquely identifying species and subspecies and pass that
column's name to this argument.</p>
</td></tr>
<tr><td><code id="auk_unique_+3A_observer_id">observer_id</code></td>
<td>
<p>character; the name of the column identifying the owner
of this instance of the group checklist. In the output dataset, the full
list of observer IDs will be stored (comma separated) in the new
<code>observer_id</code> field. The order of these IDs will match the order of the
comma separated checklist IDs.</p>
</td></tr>
<tr><td><code id="auk_unique_+3A_checklists_only">checklists_only</code></td>
<td>
<p>logical; whether the dataset provided only contains
checklist information as with the sampling event data file. If this
argument is <code>TRUE</code>, then the <code>species_id</code> argument is ignored and removing
of duplicated records is done at the checklist level not the species level.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function chooses the checklist within in each that has the
lowest value for the field specified by <code>checklist_id</code>. A new column is
also created, <code>checklist_id</code>, whose value is the taken from the field
specified in the <code>checklist_id</code> parameter for non-group checklists and from
the field specified by the <code>group_id</code> parameter for grouped checklists.
</p>
<p>All the checklist and observer IDs for the checklists that comprise a given
group checklist will be retained as a comma separated string ordered by
checklist ID.
</p>


<h3>Value</h3>

<p>A data frame with unique observations, and an additional field,
<code>checklist_id</code>, which is a combination of the sampling event and group IDs.
</p>


<h3>See Also</h3>

<p>Other pre: 
<code><a href="#topic+auk_rollup">auk_rollup</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read in an ebd file and don't automatically remove duplicates
f &lt;- system.file("extdata/ebd-sample.txt", package = "auk")
ebd &lt;- read_ebd(f, unique = FALSE)
# remove duplicates
ebd_unique &lt;- auk_unique(ebd)
nrow(ebd)
nrow(ebd_unique)
</code></pre>

<hr>
<h2 id='auk_version'>Versions of auk, the EBD, and the eBird taxonomy</h2><span id='topic+auk_version'></span>

<h3>Description</h3>

<p>This package depends on the version of the EBD and on the eBird taxonomy. Use
this function to determine the currently installed version of <code>auk</code>, the
version of the EBD that this <code>auk</code> version works with, and the version of the
eBird taxonomy included in the packages. The EBD is update quarterly, in
March, June, September, and December, while the taxonomy is updated annually
in August or September. To ensure proper functioning, always use the latest
version of the auk package and the EBD.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_version()
</code></pre>


<h3>Value</h3>

<p>A list with three elements:
</p>

<ul>
<li> <p><code>auk_version</code>: the version of <code>auk</code>, e.g. <code>"auk 0.4.1"</code>.
</p>
</li>
<li> <p><code>ebd_version</code>: a date object specifying the release date of the EBD
version that this <code>auk</code> version is designed to work with.
</p>
</li>
<li> <p><code>taxonomy_version</code>: the year of the taxonomy built in to this version of
<code>auk</code>, i.e. the one stored in <a href="#topic+ebird_taxonomy">ebird_taxonomy</a>.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other helpers: 
<code><a href="#topic+auk_ebd_version">auk_ebd_version</a>()</code>,
<code><a href="#topic+ebird_species">ebird_species</a>()</code>,
<code><a href="#topic+get_ebird_taxonomy">get_ebird_taxonomy</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>auk_version()
</code></pre>

<hr>
<h2 id='auk_year'>Filter the eBird data to a set of years</h2><span id='topic+auk_year'></span>

<h3>Description</h3>

<p>Define a filter for the eBird Basic Dataset (EBD) based on a set of
years. This function only defines the filter and, once all filters have
been defined, <code><a href="#topic+auk_filter">auk_filter()</a></code> should be used to call AWK and perform the
filtering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_year(x, year, replace = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_year_+3A_x">x</code></td>
<td>
<p><code>auk_ebd</code> or <code>auk_sampling</code> object; reference to file created by
<code><a href="#topic+auk_ebd">auk_ebd()</a></code> or <code><a href="#topic+auk_sampling">auk_sampling()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_year_+3A_year">year</code></td>
<td>
<p>integer; years to filter to.</p>
</td></tr>
<tr><td><code id="auk_year_+3A_replace">replace</code></td>
<td>
<p>logical; multiple calls to <code>auk_year()</code> are additive,
unless <code>replace = FALSE</code>, in which case the previous list of years to
filter by will be removed and replaced by that in the current call.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For filtering to a range of dates use <code>auk_date()</code>; however,
sometimes the goal is to extract data for a given year or set of years, in
which case <code>auk_year()</code> is simpler. In addition, <code>auk_year()</code> can be used
to get data from discontiguous sets of years (e.g. 2010 and 2012, but not
2011), which is not possible with <code>auk_date()</code>. Finally, <code>auk_year()</code> can
be used in conjunction with <code>auk_date()</code> to extract data from a given range
of dates within a set of years (see example below).
</p>
<p>This function can also work with on an <code>auk_sampling</code> object if the user
only wishes to filter the sampling event data.
</p>


<h3>Value</h3>

<p>An <code>auk_ebd</code> object.
</p>


<h3>See Also</h3>

<p>Other filter: 
<code><a href="#topic+auk_bbox">auk_bbox</a>()</code>,
<code><a href="#topic+auk_bcr">auk_bcr</a>()</code>,
<code><a href="#topic+auk_breeding">auk_breeding</a>()</code>,
<code><a href="#topic+auk_complete">auk_complete</a>()</code>,
<code><a href="#topic+auk_country">auk_country</a>()</code>,
<code><a href="#topic+auk_county">auk_county</a>()</code>,
<code><a href="#topic+auk_date">auk_date</a>()</code>,
<code><a href="#topic+auk_distance">auk_distance</a>()</code>,
<code><a href="#topic+auk_duration">auk_duration</a>()</code>,
<code><a href="#topic+auk_exotic">auk_exotic</a>()</code>,
<code><a href="#topic+auk_extent">auk_extent</a>()</code>,
<code><a href="#topic+auk_filter">auk_filter</a>()</code>,
<code><a href="#topic+auk_last_edited">auk_last_edited</a>()</code>,
<code><a href="#topic+auk_observer">auk_observer</a>()</code>,
<code><a href="#topic+auk_project">auk_project</a>()</code>,
<code><a href="#topic+auk_protocol">auk_protocol</a>()</code>,
<code><a href="#topic+auk_species">auk_species</a>()</code>,
<code><a href="#topic+auk_state">auk_state</a>()</code>,
<code><a href="#topic+auk_time">auk_time</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># years to filter to
years &lt;- c(2010, 2012)
# set up filter
system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  auk_year(year = years)
  
# alternatively, without pipes
ebd &lt;- auk_ebd(system.file("extdata/ebd-sample.txt", package = "auk"))
auk_year(ebd, years)

# filter to may and june of 2010 and 2012
system.file("extdata/ebd-sample.txt", package = "auk") %&gt;%
  auk_ebd() %&gt;%
  auk_year(year = c(2010, 2012)) %&gt;% 
  auk_date(date = c("*-05-01", "*-06-30"))
</code></pre>

<hr>
<h2 id='auk_zerofill'>Read and zero-fill an eBird data file</h2><span id='topic+auk_zerofill'></span><span id='topic+auk_zerofill.data.frame'></span><span id='topic+auk_zerofill.character'></span><span id='topic+auk_zerofill.auk_ebd'></span><span id='topic+collapse_zerofill'></span>

<h3>Description</h3>

<p>Read an eBird Basic Dataset (EBD) file, and associated sampling event data
file, to produce a zero-filled, presence-absence dataset. The EBD contains
bird sightings and the sampling event data is a set of all checklists, they
can be combined to infer absence data by assuming any species not reported on
a checklist was had a count of zero.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auk_zerofill(x, ...)

## S3 method for class 'data.frame'
auk_zerofill(
  x,
  sampling_events,
  species,
  taxonomy_version,
  collapse = FALSE,
  unique = TRUE,
  rollup = TRUE,
  drop_higher = TRUE,
  complete = TRUE,
  ...
)

## S3 method for class 'character'
auk_zerofill(
  x,
  sampling_events,
  species,
  taxonomy_version,
  collapse = FALSE,
  unique = TRUE,
  rollup = TRUE,
  drop_higher = TRUE,
  complete = TRUE,
  sep = "\t",
  ...
)

## S3 method for class 'auk_ebd'
auk_zerofill(
  x,
  species,
  taxonomy_version,
  collapse = FALSE,
  unique = TRUE,
  rollup = TRUE,
  drop_higher = TRUE,
  complete = TRUE,
  sep = "\t",
  ...
)

collapse_zerofill(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auk_zerofill_+3A_x">x</code></td>
<td>
<p>filename, <code>data.frame</code> of eBird observations, or <code>auk_ebd</code> object
with associated output files as created by <code><a href="#topic+auk_filter">auk_filter()</a></code>. If a filename is
provided, it must point to the EBD and the <code>sampling_events</code> argument must
point to the sampling event data file. If a <code>data.frame</code> is provided it
should have been imported with <code><a href="#topic+read_ebd">read_ebd()</a></code>, to ensure the variables names
have been set correctly, and it must have been passed through
<code><a href="#topic+auk_unique">auk_unique()</a></code> to ensure duplicate group checklists have been removed.</p>
</td></tr>
<tr><td><code id="auk_zerofill_+3A_...">...</code></td>
<td>
<p>additional arguments passed to methods.</p>
</td></tr>
<tr><td><code id="auk_zerofill_+3A_sampling_events">sampling_events</code></td>
<td>
<p>character or <code>data.frame</code>; filename for the sampling
event data or a <code>data.frame</code> of the same data. If a <code>data.frame</code> is
provided it should have been imported with <code><a href="#topic+read_sampling">read_sampling()</a></code>, to ensure the
variables names have been set correctly, and it must have been passed
through <code><a href="#topic+auk_unique">auk_unique()</a></code> to ensure duplicate group checklists have been
removed.</p>
</td></tr>
<tr><td><code id="auk_zerofill_+3A_species">species</code></td>
<td>
<p>character; species to include in zero-filled dataset, provided
as scientific or English common names, or a mixture of both. These names
must match the official eBird Taxomony (<a href="#topic+ebird_taxonomy">ebird_taxonomy</a>). To include all
species, leave this argument blank.</p>
</td></tr>
<tr><td><code id="auk_zerofill_+3A_taxonomy_version">taxonomy_version</code></td>
<td>
<p>integer; the version (i.e. year) of the taxonomy. In
most cases, this should be left empty to use the version of the taxonomy
included in the package. See <code><a href="#topic+get_ebird_taxonomy">get_ebird_taxonomy()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_zerofill_+3A_collapse">collapse</code></td>
<td>
<p>logical; whether to call <code>collapse_zerofill()</code> to return a
data frame rather than an <code>auk_zerofill</code> object.</p>
</td></tr>
<tr><td><code id="auk_zerofill_+3A_unique">unique</code></td>
<td>
<p>logical; should <code><a href="#topic+auk_unique">auk_unique()</a></code> be run on the input data if it
hasn't already.</p>
</td></tr>
<tr><td><code id="auk_zerofill_+3A_rollup">rollup</code></td>
<td>
<p>logical; should <code><a href="#topic+auk_rollup">auk_rollup()</a></code> be run on the input data if it
hasn't already.</p>
</td></tr>
<tr><td><code id="auk_zerofill_+3A_drop_higher">drop_higher</code></td>
<td>
<p>logical; whether to remove taxa above species during the
rollup process, e.g. &quot;spuhs&quot; like &quot;duck sp.&quot;. See <code><a href="#topic+auk_rollup">auk_rollup()</a></code>.</p>
</td></tr>
<tr><td><code id="auk_zerofill_+3A_complete">complete</code></td>
<td>
<p>logical; if <code>TRUE</code> (the default) all checklists are required
to be complete prior to zero-filling.</p>
</td></tr>
<tr><td><code id="auk_zerofill_+3A_sep">sep</code></td>
<td>
<p>character; single character used to separate fields within a row.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>auk_zerofill()</code> generates an <code>auk_zerofill</code> object consisting of a list with
elements <code>observations</code> and <code>sampling_events</code>. <code>observations</code> is a data frame
giving counts and binary presence/absence data for each species.
<code>sampling_events</code> is a data frame with checklist level information. The two
data frames can be connected via the <code>checklist_id</code> field. This format is
efficient for storage since the checklist columns are not duplicated for each
species, however, working with the data often requires joining the two data
frames together.
</p>
<p>To return a data frame, set <code>collapse = TRUE</code>. Alternatively,
<code>zerofill_collapse()</code> generates a data frame from an <code>auk_zerofill</code> object,
by joining the two data frames together to produce a single data frame in
which each row provides both checklist and species information for a
sighting.
</p>
<p>The list of species is checked against the eBird taxonomy for validity. This
taxonomy is updated once a year in August. The <code>auk</code> package includes a copy
of the eBird taxonomy, current at the time of release; however, if the EBD
and <code>auk</code> versions are not aligned, you may need to explicitly specify which
version of the taxonomy to use, in which case the eBird API will be queried
to get the correct version of the taxonomy.
</p>


<h3>Value</h3>

<p>By default, an <code>auk_zerofill</code> object, or a data frame if <code>collapse = TRUE</code>.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>auk_zerofill(data.frame)</code>: EBD data frame.
</p>
</li>
<li> <p><code>auk_zerofill(character)</code>: Filename of EBD.
</p>
</li>
<li> <p><code>auk_zerofill(auk_ebd)</code>: <code>auk_ebd</code> object output from <code><a href="#topic+auk_filter">auk_filter()</a></code>. Must
have had a sampling event data file set in the original call to
<code><a href="#topic+auk_ebd">auk_ebd()</a></code>.
</p>
</li></ul>


<h3>See Also</h3>

<p>Other import: 
<code><a href="#topic+read_ebd">read_ebd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read and zero-fill the ebd data
f_ebd &lt;- system.file("extdata/zerofill-ex_ebd.txt", package = "auk")
f_smpl &lt;- system.file("extdata/zerofill-ex_sampling.txt", package = "auk")
auk_zerofill(x = f_ebd, sampling_events = f_smpl)

# use the species argument to only include a subset of species
auk_zerofill(x = f_ebd, sampling_events = f_smpl,
             species = "Collared Kingfisher")

# to return a data frame use collapse = TRUE
ebd_df &lt;- auk_zerofill(x = f_ebd, sampling_events = f_smpl, collapse = TRUE)
</code></pre>

<hr>
<h2 id='bcr_codes'>BCR Codes</h2><span id='topic+bcr_codes'></span>

<h3>Description</h3>

<p>A data frame of Bird Conservation Region (BCR) codes. BCRs are ecologically
distinct regions in North America with similar bird communities, habitats,
and resource management issues. These codes are required to filter by BCR
using <code><a href="#topic+auk_bcr">auk_bcr()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bcr_codes
</code></pre>


<h3>Format</h3>

<p>A data frame with two variables and 66 rows:
</p>

<ul>
<li> <p><code>bcr_code</code>: integer code from 1 to 66.
</p>
</li>
<li> <p><code>bcr_name</code>: name of BCR.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other data: 
<code><a href="#topic+ebird_states">ebird_states</a></code>,
<code><a href="#topic+ebird_taxonomy">ebird_taxonomy</a></code>,
<code><a href="#topic+valid_protocols">valid_protocols</a></code>
</p>

<hr>
<h2 id='ebird_species'>Lookup species in eBird taxonomy</h2><span id='topic+ebird_species'></span>

<h3>Description</h3>

<p>Given a list of common or scientific names, check that they appear in the
official eBird taxonomy and convert them all to scientific names, common
names, or species codes. Un-matched species are returned as <code>NA</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ebird_species(
  x,
  type = c("scientific", "common", "code", "all"),
  taxonomy_version
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ebird_species_+3A_x">x</code></td>
<td>
<p>character; species to look up, provided as scientific or
English common names, or a mixture of both. Case insensitive.</p>
</td></tr>
<tr><td><code id="ebird_species_+3A_type">type</code></td>
<td>
<p>character; whether to return scientific names (<code>scientific</code>),
English common names (<code>common</code>), or 6-letter eBird species codes (<code>code</code>).
Alternatively, use <code>all</code> to return a data frame with the all the taxonomy
information.</p>
</td></tr>
<tr><td><code id="ebird_species_+3A_taxonomy_version">taxonomy_version</code></td>
<td>
<p>integer; the version (i.e. year) of the taxonomy.
Leave empty to use the version of the taxonomy included in the package.
See <code><a href="#topic+get_ebird_taxonomy">get_ebird_taxonomy()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character vector of species identified by scientific name, common
name, or species code. If <code>type = "all"</code> a data frame of the taxonomy of
the requested species is returned.
</p>


<h3>See Also</h3>

<p>Other helpers: 
<code><a href="#topic+auk_ebd_version">auk_ebd_version</a>()</code>,
<code><a href="#topic+auk_version">auk_version</a>()</code>,
<code><a href="#topic+get_ebird_taxonomy">get_ebird_taxonomy</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># mix common and scientific names, case-insensitive
species &lt;- c("Blackburnian Warbler", "Poecile atricapillus",
             "american dipper", "Caribou")
# note that species not in the ebird taxonomy return NA
ebird_species(species)

# use taxonomy_version to query older taxonomy versions
## Not run: 
ebird_species("Cordillera Azul Antbird")
ebird_species("Cordillera Azul Antbird", taxonomy_version = 2017)

## End(Not run)
</code></pre>

<hr>
<h2 id='ebird_states'>eBird States</h2><span id='topic+ebird_states'></span>

<h3>Description</h3>

<p>A data frame of state codes used by eBird. These codes are 4 to 6 characters,
consisting of two parts, the 2-letter ISO country code and a 1-3 character
state code, separated by a dash. For example, <code>"US-NY"</code> corresponds to New
York State in the United States. These state codes are required to filter by
state using <code><a href="#topic+auk_state">auk_state()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ebird_states
</code></pre>


<h3>Format</h3>

<p>A data frame with four variables and 3,145 rows:
</p>

<ul>
<li> <p><code>country</code>: short form of English country name.
</p>
</li>
<li> <p><code>country_code</code>: 2-letter ISO country code.
</p>
</li>
<li> <p><code>state</code>: state name.
</p>
</li>
<li> <p><code>state_code</code>: 4 to 6 character state code.
</p>
</li></ul>



<h3>Details</h3>

<p>Note that some countries are not broken into states in eBird and therefore do
not appear in this data frame.
</p>


<h3>See Also</h3>

<p>Other data: 
<code><a href="#topic+bcr_codes">bcr_codes</a></code>,
<code><a href="#topic+ebird_taxonomy">ebird_taxonomy</a></code>,
<code><a href="#topic+valid_protocols">valid_protocols</a></code>
</p>

<hr>
<h2 id='ebird_taxonomy'>eBird Taxonomy</h2><span id='topic+ebird_taxonomy'></span>

<h3>Description</h3>

<p>A simplified version of the taxonomy used by eBird. Includes proper species
as well as various other categories such as <code>spuh</code> (e.g. <em>duck sp.</em>) and
<em>slash</em> (e.g. <em>American Black Duck/Mallard</em>). This taxonomy is based on the
Clements Checklist, which is updated annually, typically in the late summer.
Non-ASCII characters (e.g. those with accents) have been converted to ASCII
equivalents in this data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ebird_taxonomy
</code></pre>


<h3>Format</h3>

<p>A data frame with eight variables and 16,248 rows:
</p>

<ul>
<li> <p><code>scientific_name</code>: scientific name.
</p>
</li>
<li> <p><code>common_name</code>: common name, defaults to English, but different languages
can be selected using the <code>locale</code> parameter.
</p>
</li>
<li> <p><code>species_code</code>: a unique alphanumeric code identifying each species.
</p>
</li>
<li> <p><code>category</code>: whether the entry is for a species or another
field-identifiable taxon, such as <code>spuh</code>, <code>slash</code>, <code>hybrid</code>, etc.
</p>
</li>
<li> <p><code>taxon_order</code>: numeric value used to sort rows in taxonomic order.
</p>
</li>
<li> <p><code>order</code>: the scientific name of the order that the species belongs to.
</p>
</li>
<li> <p><code>family</code>: the scientific name of the family that the species belongs to.
</p>
</li>
<li> <p><code>report_as</code>: for taxa that can be resolved to true species (i.e. species,
subspecies, and recognizable forms), this field links to the corresponding
species code. For taxa that can't be resolved, this field is <code>NA</code>.
</p>
</li></ul>

<p>For further details, see <a href="https://support.ebird.org/support/solutions/articles/48000837816-the-ebird-taxonomy">https://support.ebird.org/support/solutions/articles/48000837816-the-ebird-taxonomy</a>
</p>


<h3>See Also</h3>

<p>Other data: 
<code><a href="#topic+bcr_codes">bcr_codes</a></code>,
<code><a href="#topic+ebird_states">ebird_states</a></code>,
<code><a href="#topic+valid_protocols">valid_protocols</a></code>
</p>

<hr>
<h2 id='filter_repeat_visits'>Filter observations to repeat visits for hierarchical modeling</h2><span id='topic+filter_repeat_visits'></span>

<h3>Description</h3>

<p>Hierarchical modeling of abundance and occurrence requires repeat visits to
sites to estimate detectability. These visits should be all be within a
period of closure, i.e. when the population can be assumed to be closed.
eBird data, and many other data sources, do not explicitly follow this
protocol; however, subsets of the data can be extracted to produce data
suitable for hierarchical modeling. This function extracts a subset of
observation data that have a desired number of repeat visits within a period
of closure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_repeat_visits(
  x,
  min_obs = 2L,
  max_obs = 10L,
  annual_closure = TRUE,
  n_days = NULL,
  date_var = "observation_date",
  site_vars = c("locality_id", "observer_id"),
  ll_digits = 6L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_repeat_visits_+3A_x">x</code></td>
<td>
<p><code>data.frame</code>; observation data, e.g. data from the eBird Basic
Dataset (EBD) zero-filled with <code><a href="#topic+auk_zerofill">auk_zerofill()</a></code>. This function will also
work with an <code>auk_zerofill</code> object, in which case it will be converted to
a data frame with <code><a href="#topic+collapse_zerofill">collapse_zerofill()</a></code>.
<strong>Note that these data must for a single species</strong>.</p>
</td></tr>
<tr><td><code id="filter_repeat_visits_+3A_min_obs">min_obs</code></td>
<td>
<p>integer; minimum number of observations required for each
site.</p>
</td></tr>
<tr><td><code id="filter_repeat_visits_+3A_max_obs">max_obs</code></td>
<td>
<p>integer; maximum number of observations allowed for each site.</p>
</td></tr>
<tr><td><code id="filter_repeat_visits_+3A_annual_closure">annual_closure</code></td>
<td>
<p>logical; whether the entire year should be treated as
the period of closure (the default). This can be useful, for example, if
the data have been subset to a period of closure prior to calling
<code><a href="#topic+filter_repeat_visits">filter_repeat_visits()</a></code>.</p>
</td></tr>
<tr><td><code id="filter_repeat_visits_+3A_n_days">n_days</code></td>
<td>
<p>integer; number of days defining the temporal length of
closure. If <code>annual_closure = TRUE</code> closure periods will be split at year
boundaries. If <code>annual_closure = FALSE</code> the closure periods will ignore
year boundaries.</p>
</td></tr>
<tr><td><code id="filter_repeat_visits_+3A_date_var">date_var</code></td>
<td>
<p>character; column name of the variable in <code>x</code> containing the
date. This column should either be in <code>Date</code> format or convertible to
<code>Date</code> format with <code><a href="base.html#topic+as.Date">as.Date()</a></code>.</p>
</td></tr>
<tr><td><code id="filter_repeat_visits_+3A_site_vars">site_vars</code></td>
<td>
<p>character; names of one of more columns in <code>x</code> that define a
site, typically the location (e.g. latitude/longitude) and observer ID.</p>
</td></tr>
<tr><td><code id="filter_repeat_visits_+3A_ll_digits">ll_digits</code></td>
<td>
<p>integer; the number of digits to round latitude and longitude
to. If latitude and/or longitude are used as <code>site_vars</code>, it's usually best
to round them prior to identifying sites, otherwise locations that are only
slightly offset (e.g. a few centimeters) will be treated as different. This
argument can also be used to group sites together that are close but not
identical. Note that 1 degree of latitude is approximately 100 km, so the
default value of 6 for <code>ll_digits</code> is equivalent to about 10 cm.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In addition to specifying the minimum and maximum number of
observations per site, users must specify the variables in the dataset that
define a &quot;site&quot;. This is typically a combination of IDs defining the
geographic site and the unique observer (repeat visits are meant to be
conducted by the same observer). Finally, the closure period must be
defined, which is a period within which the population of the focal species
can reasonably be assumed to be closed. This can be done using a
combination of the <code>n_days</code> and <code>annual_closure</code> arguments.
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> filtered to only retain observations from sites with
the allowed number of observations within the period of closure. The
results will be sorted such that sites are together and in chronological
order. The following variables are added to the data frame:
</p>

<ul>
<li> <p><code>site</code>: a unique identifier for each &quot;site&quot; corresponding to all the
variables in <code>site_vars</code> and <code>closure_id</code> concatenated together with
underscore separators.
</p>
</li>
<li> <p><code>closure_id</code>: a unique ID for each closure period. If <code>annual_closure =   TRUE</code> this ID will include the year. If <code>n_days</code> is used an index given the
number of blocks of <code>n_days</code> days since the earliest observation will be
included. Note that in this case, there may be gaps in the IDs.
</p>
</li>
<li> <p><code>n_observations</code>: number of observations at each site after all
filtering.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other modeling: 
<code><a href="#topic+format_unmarked_occu">format_unmarked_occu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read and zero-fill the ebd data
f_ebd &lt;- system.file("extdata/zerofill-ex_ebd.txt", package = "auk")
f_smpl &lt;- system.file("extdata/zerofill-ex_sampling.txt", package = "auk")
# data must be for a single species
ebd_zf &lt;- auk_zerofill(x = f_ebd, sampling_events = f_smpl,
                       species = "Collared Kingfisher",
                       collapse = TRUE)
filter_repeat_visits(ebd_zf, n_days = 30)
</code></pre>

<hr>
<h2 id='format_unmarked_occu'>Format EBD data for occupancy modeling with <code>unmarked</code></h2><span id='topic+format_unmarked_occu'></span>

<h3>Description</h3>

<p>Prepare a data frame of species observations for ingestion into the package
<code>unmarked</code> for hierarchical modeling of abundance and occurrence. The
function <code><a href="unmarked.html#topic+formatWideLong">unmarked::formatWide()</a></code> takes a data frame and converts it to one
of several <code>unmarked</code> objects, which can then be used for modeling. This
function converts data from a format in which each row is an observation
(e.g. as in the eBird Basic Dataset) to the esoteric format required by
<code><a href="unmarked.html#topic+formatWideLong">unmarked::formatWide()</a></code> in which each row is a site.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>format_unmarked_occu(
  x,
  site_id = "site",
  response = "species_observed",
  site_covs,
  obs_covs
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="format_unmarked_occu_+3A_x">x</code></td>
<td>
<p><code>data.frame</code>; observation data, e.g. from the eBird Basic Dataset
(EBD), for <strong>a single species</strong>, that has been filtered to those with
repeat visits by <code><a href="#topic+filter_repeat_visits">filter_repeat_visits()</a></code>.</p>
</td></tr>
<tr><td><code id="format_unmarked_occu_+3A_site_id">site_id</code></td>
<td>
<p>character; a unique idenitifer for each &quot;site&quot;, typically
identifying observations from a unique location by the same observer
within a period of temporal closure. Data output from
<code><a href="#topic+filter_repeat_visits">filter_repeat_visits()</a></code> will have a <code>.site_id</code> variable that meets these
requirements.</p>
</td></tr>
<tr><td><code id="format_unmarked_occu_+3A_response">response</code></td>
<td>
<p>character; the variable that will act as the response in
modeling efforts, typically a binary variable indicating presence or
absence or a count of individuals seen.</p>
</td></tr>
<tr><td><code id="format_unmarked_occu_+3A_site_covs">site_covs</code></td>
<td>
<p>character; the variables that will act as site-level
covariates, i.e. covariates that vary at the site level, for example,
latitude/longitude or habitat predictors. If this parameter is missing, it
will be assumed that any variable that is not an observation-level
covariate (<code>obs_covs</code>) or the <code>site_id</code>, is a site-level covariate.</p>
</td></tr>
<tr><td><code id="format_unmarked_occu_+3A_obs_covs">obs_covs</code></td>
<td>
<p>character; the variables that will act as observation-level
covariates, i.e. covariates that vary within sites, at the level of
observations, for example, time or length of observation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Hierarchical modeling requires repeat observations at each &quot;site&quot; to
estimate detectability. A &quot;site&quot; is typically defined as a geographic
location visited by the same observer within a period of temporal closure.
To define these sites and filter out observations that do not correspond to
repeat visits, users should use <code><a href="#topic+filter_repeat_visits">filter_repeat_visits()</a></code>, then pass the
output to this function.
</p>
<p><code><a href="#topic+format_unmarked_occu">format_unmarked_occu()</a></code> is designed to prepare data to be converted into
an <code>unmarkedFrameOccu</code> object for occupancy modeling with
<code><a href="unmarked.html#topic+occu">unmarked::occu()</a></code>; however, it can also be used to prepare data for
conversion to an <code>unmarkedFramePCount</code> object for abundance modeling with
<code><a href="unmarked.html#topic+pcount">unmarked::pcount()</a></code>.
</p>


<h3>Value</h3>

<p>A data frame that can be processed by <code><a href="unmarked.html#topic+formatWideLong">unmarked::formatWide()</a></code>.
Each row will correspond to a unqiue site and, assuming there are a maximum
of <code>N</code> observations per site, columns will be as follows:
</p>

<ol>
<li><p> The unique site identifier, named &quot;site&quot;.
</p>
</li>
<li> <p><code>N</code> response columns, one for each observation, named &quot;y.1&quot;, ..., &quot;y.N&quot;.
</p>
</li>
<li><p> Columns for each of the site-level covariates.
</p>
</li>
<li><p> Groups of <code>N</code> columns of observation-level covariates, one column per
covariate per observation, names &quot;covariate_name.1&quot;, ...,
&quot;covariate_name.N&quot;.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other modeling: 
<code><a href="#topic+filter_repeat_visits">filter_repeat_visits</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read and zero-fill the ebd data
f_ebd &lt;- system.file("extdata/zerofill-ex_ebd.txt", package = "auk")
f_smpl &lt;- system.file("extdata/zerofill-ex_sampling.txt", package = "auk")
# data must be for a single species
ebd_zf &lt;- auk_zerofill(x = f_ebd, sampling_events = f_smpl,
                       species = "Collared Kingfisher",
                       collapse = TRUE)
occ &lt;- filter_repeat_visits(ebd_zf, n_days = 30)
# format for unmarked
# typically one would join in habitat covariates prior to this step
occ_wide &lt;- format_unmarked_occu(occ,
                                 response = "species_observed",
                                 site_covs = c("latitude", "longitude"),
                                 obs_covs = c("effort_distance_km", 
                                              "duration_minutes"))
# create an unmarked object
if (requireNamespace("unmarked", quietly = TRUE)) {
  occ_um &lt;- unmarked::formatWide(occ_wide, type = "unmarkedFrameOccu")
  unmarked::summary(occ_um)
}

# this function can also be used for abundance modeling
abd &lt;- ebd_zf %&gt;% 
  # convert count to integer, drop records with no count
  dplyr::mutate(observation_count = as.integer(observation_count)) %&gt;% 
  dplyr::filter(!is.na(observation_count)) %&gt;% 
  # filter to repeated visits
  filter_repeat_visits(n_days = 30)
# prepare for conversion to unmarkedFramePCount object
abd_wide &lt;- format_unmarked_occu(abd,
                                 response = "observation_count",
                                 site_covs = c("latitude", "longitude"),
                                 obs_covs = c("effort_distance_km", 
                                              "duration_minutes"))
# create an unmarked object
if (requireNamespace("unmarked", quietly = TRUE)) {
  abd_um &lt;- unmarked::formatWide(abd_wide, type = "unmarkedFrameOccu")
  unmarked::summary(abd_um)
}
</code></pre>

<hr>
<h2 id='get_ebird_taxonomy'>Get eBird taxonomy via the eBird API</h2><span id='topic+get_ebird_taxonomy'></span>

<h3>Description</h3>

<p>Get the taxonomy used in eBird via the eBird API.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_ebird_taxonomy(version, locale)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_ebird_taxonomy_+3A_version">version</code></td>
<td>
<p>integer; the version (i.e. year) of the taxonomy. The eBird
taxonomy is updated once a year in August. Leave this parameter blank to
get the current taxonomy.</p>
</td></tr>
<tr><td><code id="get_ebird_taxonomy_+3A_locale">locale</code></td>
<td>
<p>character; the <a href="https://support.ebird.org/support/solutions/articles/48000804865-bird-names-in-ebird">locale for the common names</a>,
defaults to English.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of all species in the eBird taxonomy, consisting of the
following columns:
</p>

<ul>
<li> <p><code>scientific_name</code>: scientific name.
</p>
</li>
<li> <p><code>common_name</code>: common name, defaults to English, but different languages
can be selected using the <code>locale</code> parameter.
</p>
</li>
<li> <p><code>species_code</code>: a unique alphanumeric code identifying each species.
</p>
</li>
<li> <p><code>category</code>: whether the entry is for a species or another
field-identifiable taxon, such as <code>spuh</code>, <code>slash</code>, <code>hybrid</code>, etc.
</p>
</li>
<li> <p><code>taxon_order</code>: numeric value used to sort rows in taxonomic order.
</p>
</li>
<li> <p><code>order</code>: the scientific name of the order that the species belongs to.
</p>
</li>
<li> <p><code>family</code>: the scientific name of the family that the species belongs to.
</p>
</li>
<li> <p><code>report_as</code>: for taxa that can be resolved to true species (i.e. species,
subspecies, and recognizable forms), this field links to the corresponding
species code. For taxa that can't be resolved, this field is <code>NA</code>.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other helpers: 
<code><a href="#topic+auk_ebd_version">auk_ebd_version</a>()</code>,
<code><a href="#topic+auk_version">auk_version</a>()</code>,
<code><a href="#topic+ebird_species">ebird_species</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_ebird_taxonomy()

## End(Not run)
</code></pre>

<hr>
<h2 id='read_ebd'>Read an EBD file</h2><span id='topic+read_ebd'></span><span id='topic+read_ebd.character'></span><span id='topic+read_ebd.auk_ebd'></span><span id='topic+read_sampling'></span><span id='topic+read_sampling.character'></span><span id='topic+read_sampling.auk_ebd'></span><span id='topic+read_sampling.auk_sampling'></span>

<h3>Description</h3>

<p>Read an eBird Basic Dataset file using <code><a href="readr.html#topic+read_delim">readr::read_delim()</a></code>. <code>read_ebd()</code>
reads the EBD itself, while read_sampling()' reads a sampling event data
file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_ebd(x, sep = "\t", unique = TRUE, rollup = TRUE)

## S3 method for class 'character'
read_ebd(x, sep = "\t", unique = TRUE, rollup = TRUE)

## S3 method for class 'auk_ebd'
read_ebd(x, sep = "\t", unique = TRUE, rollup = TRUE)

read_sampling(x, sep = "\t", unique = TRUE)

## S3 method for class 'character'
read_sampling(x, sep = "\t", unique = TRUE)

## S3 method for class 'auk_ebd'
read_sampling(x, sep = "\t", unique = TRUE)

## S3 method for class 'auk_sampling'
read_sampling(x, sep = "\t", unique = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_ebd_+3A_x">x</code></td>
<td>
<p>filename or <code>auk_ebd</code> object with associated output
files as created by <code><a href="#topic+auk_filter">auk_filter()</a></code>.</p>
</td></tr>
<tr><td><code id="read_ebd_+3A_sep">sep</code></td>
<td>
<p>character; single character used to separate fields within a row.</p>
</td></tr>
<tr><td><code id="read_ebd_+3A_unique">unique</code></td>
<td>
<p>logical; should duplicate grouped checklists be removed. If
<code>unique = TRUE</code>, <code><a href="#topic+auk_unique">auk_unique()</a></code> is called on the EBD before returning.</p>
</td></tr>
<tr><td><code id="read_ebd_+3A_rollup">rollup</code></td>
<td>
<p>logical; should taxonomic rollup to species level be applied.
If <code>rollup = TRUE</code>, <code><a href="#topic+auk_rollup">auk_rollup()</a></code> is called on the EBD before returning.
Note that this process can be time consuming for large files, try turning
rollup off if reading is taking too long.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This functions performs the following processing steps:
</p>

<ul>
<li><p> Data types for columns are manually set based on column names used in the
February 2017 EBD. If variables are added or names are changed in later
releases, any new variables will have data types inferred by the import
function used.
</p>
</li>
<li><p> Variables names are converted to <code>snake_case</code>.
</p>
</li>
<li><p> Duplicate observations resulting from group checklists are removed using
<code><a href="#topic+auk_unique">auk_unique()</a></code>, unless <code>unique = FALSE</code>.
</p>
</li></ul>



<h3>Value</h3>

<p>A data frame of EBD observations. An additional column,
<code>checklist_id</code>, is added to output files if <code>unique = TRUE</code>, that uniquely
identifies the checklist from which the observation came. This field is
equal to <code>sampling_event_identifier</code> for non-group checklists, and
<code>group_identifier</code> for group checklists.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>read_ebd(character)</code>: Filename of EBD.
</p>
</li>
<li> <p><code>read_ebd(auk_ebd)</code>: <code>auk_ebd</code> object output from <code><a href="#topic+auk_filter">auk_filter()</a></code>
</p>
</li></ul>


<h3>Functions</h3>


<ul>
<li> <p><code>read_sampling(character)</code>: Filename of sampling event data file
</p>
</li>
<li> <p><code>read_sampling(auk_ebd)</code>: <code>auk_ebd</code> object output from <code><a href="#topic+auk_filter">auk_filter()</a></code>. Must have
had a sampling event data file set in the original call to <code><a href="#topic+auk_ebd">auk_ebd()</a></code>.
</p>
</li>
<li> <p><code>read_sampling(auk_sampling)</code>: <code>auk_sampling</code> object output from <code><a href="#topic+auk_filter">auk_filter()</a></code>.
</p>
</li></ul>


<h3>See Also</h3>

<p>Other import: 
<code><a href="#topic+auk_zerofill">auk_zerofill</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>f &lt;- system.file("extdata/ebd-sample.txt", package = "auk")
read_ebd(f)
# read a sampling event data file
x &lt;- system.file("extdata/zerofill-ex_sampling.txt", package = "auk") %&gt;%
  read_sampling()
</code></pre>

<hr>
<h2 id='reexports'>Objects exported from other packages</h2><span id='topic+reexports'></span><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>magrittr</dt><dd><p><code><a href="magrittr.html#topic+pipe">%&gt;%</a></code></p>
</dd>
</dl>

<hr>
<h2 id='valid_protocols'>Valid Protocols</h2><span id='topic+valid_protocols'></span>

<h3>Description</h3>

<p>A vector of valid protocol names, e.g. &quot;Traveling&quot;, &quot;Stationary&quot;, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>valid_protocols
</code></pre>


<h3>Format</h3>

<p>A vector with 42 elements.
</p>


<h3>See Also</h3>

<p>Other data: 
<code><a href="#topic+bcr_codes">bcr_codes</a></code>,
<code><a href="#topic+ebird_states">ebird_states</a></code>,
<code><a href="#topic+ebird_taxonomy">ebird_taxonomy</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
