<!DOCTYPE html><html><head><title>Help for package ausplotsR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ausplotsR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ausplots_trajectory'><p>Visualisation of species composition and community index revisit trajectories</p></a></li>
<li><a href='#ausplots_visual'><p>Graphical illustration of sample TERN AusPlots data</p></a></li>
<li><a href='#ausplotsR-package'>
<p>Live extraction, preparation, visualisation and analysis of TERN AusPlots ecosystem monitoring data.</p></a></li>
<li><a href='#basal_area'><p>Basal area of tree trunks in TERN AusPlots</p></a></li>
<li><a href='#fractional_cover'><p>Fractional cover based on TERN AusPlots data.</p></a></li>
<li><a href='#get_ausplots'><p>Extract AusPlots data for sites, soils and vegetation</p></a></li>
<li><a href='#growth_form_table'><p>Plant growth form occurrence matrices for TERN AusPlots</p></a></li>
<li><a href='#optim_species'><p>Optimisation methods for plot selection using different diversity indices in order to maximise species accumulation</p></a></li>
<li><a href='#plot_opt'><p>Plot a set of species accumulation curves together</p></a></li>
<li><a href='#single_cover_value'><p>Vegetation cover by height or growth form using TERN AusPlots data.</p></a></li>
<li><a href='#species_list'><p>Species lists for AusPlots</p></a></li>
<li><a href='#species_table'><p>Species occurrence matrices for AusPlots</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>TERN AusPlots Australian Ecosystem Monitoring Data</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-11-7</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Greg Guerin &lt;ggueri01@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0), ggplot2, mapdata</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, markdown, formatR, sf</td>
</tr>
<tr>
<td>Imports:</td>
<td>vegan, plyr, R.utils, httr, jsonlite, gtools, jose, betapart,
curl, r2r, stringr, progress</td>
</tr>
<tr>
<td>Description:</td>
<td>Extraction, preparation, visualisation and analysis of TERN AusPlots ecosystem monitoring data. Direct access to plot-based data on vegetation and soils across Australia, including physical sample barcode numbers. Simple function calls extract the data and merge them into species occurrence matrices for downstream analysis, or calculate things like basal area and fractional cover. TERN AusPlots is a national field plot-based ecosystem surveillance monitoring method and dataset for Australia. The data have been collected across a national network of plots and transects by the Terrestrial Ecosystem Research Network (TERN - <a href="https://www.tern.org.au">https://www.tern.org.au</a>), an Australian Government NCRIS-enabled project, and its Ecosystem Surveillance platform (<a href="https://www.tern.org.au/tern-land-observatory/ecosystem-surveillance-and-environmental-monitoring/">https://www.tern.org.au/tern-land-observatory/ecosystem-surveillance-and-environmental-monitoring/</a>).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-17 02:13:42 UTC; a1038758</td>
</tr>
<tr>
<td>Author:</td>
<td>Greg Guerin [aut, cre],
  Tom Saleeba [aut],
  Samantha Munroe [aut],
  Bernardo Blanco-Martin [aut],
  Irene Martín-Forés [aut],
  Walid Al Naim [aut],
  Andrew Tokmakoff [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-17 04:43:50 UTC</td>
</tr>
</table>
<hr>
<h2 id='ausplots_trajectory'>Visualisation of species composition and community index revisit trajectories
</h2><span id='topic+ausplots_trajectory'></span>

<h3>Description</h3>

<p>Generates 2D plots visualising the trajectory of species composition (PCoA) and diversity (scatterplot or time series) for revisit series. Trajectories are calculated from the TERN AusPlots point intercept module (<code>$veg.PI</code>) as generated by <code><a href="#topic+get_ausplots">get_ausplots</a></code> and other pre-processing functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ausplots_trajectory(my.ausplots.object, choices = c("PCoA", "diversity"),
min.revisits = 3, plot_select = NULL, outfile = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ausplots_trajectory_+3A_my.ausplots.object">my.ausplots.object</code></td>
<td>
<p>Input raw data object (list) as generated in a call from the <code><a href="#topic+get_ausplots">get_ausplots</a></code> function. See Details.
</p>
</td></tr>
<tr><td><code id="ausplots_trajectory_+3A_choices">choices</code></td>
<td>
<p>Character. User selection of PCoA and/or diversity plots. Both are generated by default.
</p>
</td></tr>
<tr><td><code id="ausplots_trajectory_+3A_min.revisits">min.revisits</code></td>
<td>
<p>Integer. Minimum number of revisits required to include a plot.
</p>
</td></tr>
<tr><td><code id="ausplots_trajectory_+3A_plot_select">plot_select</code></td>
<td>
<p>Optional character vector of AusPlots plot IDs (<code>site_location_name</code>) to request data for a specific set of plots. Note PCoA will not run on one or very few plots/surveys.
</p>
</td></tr>
<tr><td><code id="ausplots_trajectory_+3A_outfile">outfile</code></td>
<td>
<p>Optional file path and name to which the output pdf file is written.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function primarily applies the functionality of R package <code>ecotraj</code> (De Cáceres et al. 2019; Sturbois et al. 2021) to the TERN AusPlots vegetation point intercept data by including the required pre-processing of the raw data and modifications to the output plot format. The function is intended as a rapid, visual means to compare revisit trajectories in species composition and diversity (or other community indices) among sites. For in-depth change/trajectory analysis, users are directed to work with the raw data and the extensive functionality in <code>ecotraj</code> or other approaches.
</p>
<p>The default graphical outputs require <code>$site.info</code>, to order visits by date, and <code>$veg.PI</code>, for plant species abundance, in the AusPlots data extract. User-generated community indices (any single value per survey/<code>$site_unique</code>) can be plotted in place of diversity if <code>$community.indices</code> appears in the <code>my.ausplots.object</code>. This table must have a <code>$site_unique</code> column matching surveys in <code>$site.info</code> and any number of additional columns representing any community indices intended for plotting out over time. See Examples.
</p>
<p>Two output options are available. 'PCoA' displays the first two axes (and their respective deviance explained as axis labels) of a metric ordination based on species abundances and so shows the trajectory of plots in species composition space. Similar trajectories on the plot implies parallel trends in species presence and abundance.
</p>
<p>Next, 'diversity' by default displays a simple scatterplot with trajectories between revisits. The function presently computes species richness and Shannon diversity metrics using the <code>vegan</code> package (Oksanen et al. 2022) and displays these as trajectories on a 2D scatterplot, as well as individually against year. By adding a data table <code>$community.indices</code> to those available via <code><a href="#topic+get_ausplots">get_ausplots</a></code>, users can also plot out the trajectory of other bespoke community level indices. For &gt; 2 indices, only individual plots against year of survey are generated. 
</p>
<p>Note that revisits (and revisit series &gt; 2) are available only for a subset of the full plot network and that <code>ausplots_trajectory</code> may display trajectories better for targeted sets of plots.
</p>


<h3>Value</h3>

<p>Plots are written to a pdf file to ensure consistent formatting, saved by default in the working directory.
</p>


<h3>Author(s)</h3>

<p>Greg Guerin
</p>


<h3>References</h3>

<p>De Cáceres, M., Coll, L., Legendre, P., Allen, R.B., Wiser, S.K., Fortin, M.J., Condit, R. &amp; Hubbell, S. (2019). Trajectory analysis in community ecology. Ecological Monographs 89: e01350
</p>
<p>Oksanen J, Simpson G, Blanchet F, Kindt R, Legendre P, Minchin P, O'Hara R, Solymos P, Stevens M, Szoecs E, Wagner H, Barbour M, Bedward M, Bolker B, Borcard D, Carvalho G, Chirico M, De Caceres M, Durand S, Evangelista H, FitzJohn R, Friendly M, Furneaux B, Hannigan G, Hill M, Lahti L, McGlinn D, Ouellette M, Ribeiro Cunha E, Smith T, Stier A, Ter Braak C, Weedon J (2022). _vegan: Community Ecology Package_. R package version 2.6-4, &lt;https://CRAN.R-project.org/package=vegan&gt;.
</p>
<p>Sturbois, A., De Cáceres, M., Sánchez-Pinillos, M., Schaal, G., Gauthier, O., Le Mao, P., Ponsero, A., &amp; Desroy, N. (2021). Extending community trajectory analysis : New metrics and representation. Ecological Modelling 440: 109400
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_ausplots">get_ausplots</a></code>
<br />
<br />
<code><a href="#topic+ausplots_visual">ausplots_visual</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Extract site data first:
my.data &lt;- get_ausplots()

#Extract vegetation point intercept data for plots with revisits
#Here, subsetting my.Plots_IDs to revisits in one bioregion
my.data &lt;- get_ausplots(veg.PI = TRUE, 
my.Plot_IDs = unique(subset(my.data$site.info, visit_number &gt; 1 
&amp; bioregion_name == "MAC", select=site_location_name))[,1])

#Display species composition trajectory for all plots with at least 3 revisits
ausplots_trajectory(my.data, choices = "PCoA")

#Display fractional cover trajectories (community index example)
my.data$community.indices &lt;- fractional_cover(my.data$veg.PI)[,c("site_unique",
 "bare", "brown", "green")]

ausplots_trajectory(my.data, choices = "diversity")

## End(Not run)
</code></pre>

<hr>
<h2 id='ausplots_visual'>Graphical illustration of sample TERN AusPlots data
</h2><span id='topic+ausplots_visual'></span><span id='topic+map_ausplots'></span>

<h3>Description</h3>

<p>Generates a set of graphical displays representing TERN AusPlots data based on geographic locations and vegetation attributes from the plot-based point intercepts (as generated by <code><a href="#topic+get_ausplots">get_ausplots</a></code> and other pre-processing functions).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ausplots_visual(my.ausplots.object = NULL, map = TRUE, map.attribute = TRUE, 
fraction.pie = TRUE, growthform.pie = TRUE, cumulative.cover = TRUE, 
whittaker = TRUE, outfile=NULL, max.plots = 5)

map_ausplots(my.ausplots.object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ausplots_visual_+3A_my.ausplots.object">my.ausplots.object</code></td>
<td>
<p>Optional (but required for <code>map_ausplots</code>) input raw data object as generated in a call from the <code><a href="#topic+get_ausplots">get_ausplots</a></code> function. The default graphical outputs require $site.info (for locations) and $veg.PI (for vegetation attributes) to be included in the AusPlots data extract.
</p>
</td></tr>
<tr><td><code id="ausplots_visual_+3A_map">map</code></td>
<td>
<p>Logical, whether to include a basic map of site locations in the context of Australia.
</p>
</td></tr>
<tr><td><code id="ausplots_visual_+3A_map.attribute">map.attribute</code></td>
<td>
<p>Logical, whether to include a map depicting site locations in Australia coded by cover of trees over 5 m high.
</p>
</td></tr>
<tr><td><code id="ausplots_visual_+3A_fraction.pie">fraction.pie</code></td>
<td>
<p>Logical, whether to include pie charts depicting <code><a href="#topic+fractional_cover">fractional_cover</a></code>.
</p>
</td></tr>
<tr><td><code id="ausplots_visual_+3A_growthform.pie">growthform.pie</code></td>
<td>
<p>Logical, whether to include pie charts depicting relative cover of plant growth forms.
</p>
</td></tr>
<tr><td><code id="ausplots_visual_+3A_cumulative.cover">cumulative.cover</code></td>
<td>
<p>Logical, whether to include graphs of cumulative cover estimates by species as point intercepts are taken across each AusPlot.
</p>
</td></tr>
<tr><td><code id="ausplots_visual_+3A_whittaker">whittaker</code></td>
<td>
<p>Logical, whether to include Whittaker plots depicting  species abundance distributions.
</p>
</td></tr>
<tr><td><code id="ausplots_visual_+3A_outfile">outfile</code></td>
<td>
<p>Optional file path and name to which the output pdf file is written.
</p>
</td></tr>
<tr><td><code id="ausplots_visual_+3A_max.plots">max.plots</code></td>
<td>
<p>Maximum number of sites for which to display individual graphics panels. By default, <code>max.plots</code> sites are selected randomly. If you provide existing AusPlots data, <code>max.plots</code> sites are processed (in order) whenever <code>fractional.pie</code> or <code>cumulative.cover</code> are selected, because those calculations can be slow across a large number of plots. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, extracts sample data, allowing the user to explore the dataset in a single call, or to easily visualise an extracted AusPlots dataset. Included in the default demonstration plots are maps of plot locations with and without coding by tree cover, and then a set of four panels for a smaller sample of AusPlots, depicting data from the main vegetation tables in the package, i.e., fractional cover, growth form cover, species cover and species relative abundance. This function is intended as a simple demonstration of the data available. Refer to TERN tutorials for many other ways to visualise and explore AusPlots.
</p>


<h3>Value</h3>

<p>Plots are written to a pdf file to ensure consistent formatting, saved by default in the working directory.
</p>


<h3>Author(s)</h3>

<p>Greg Guerin, Bernardo Blanco-Martin
</p>


<h3>References</h3>

<p>Blanco-Martin, B. (2019) Tutorial: Understanding and using the 'ausplotsR' package and AusPlots data. Terrestrial Ecology Research Network. Version 2019.04.0, April 2019.
https://github.com/ternaustralia/TERN-Data-Skills/
</p>
<p>Guerin, G.R., Sparrow, B., Tokmakoff, A., Smyth, A., Leitch, E., Baruch, Z., Lowe, A.J. (2017) Opportunities for integrated ecological analysis across inland Australia with standardised data from AusPlots Rangelands. <em>PLoS ONE</em> <strong>12</strong>, e0170137.
</p>
<p>Whittaker, R.H. (1965) Dominance and diversity in land plant communities: numerical relations of species express the importance of competition in community function and evolution. <em>Science</em>, <strong>147</strong>, 250-260.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_ausplots">get_ausplots</a></code>   
<br />
<br />
<code><a href="#topic+fractional_cover">fractional_cover</a></code>   
<br />
<br />
<code><a href="#topic+growth_form_table">growth_form_table</a></code>   
<br />
<br />
<code><a href="#topic+single_cover_value">single_cover_value</a></code>   
<br />
<br />
<code><a href="#topic+species_table">species_table</a></code>
<br />
<br />
<code><a href="#topic+ausplots_trajectory">ausplots_trajectory</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Default output
#Plot locations; visualisations for 5 random plots (including revisits):
ausplots_visual()

#Map tree cover only for all sites:

#Extract data first:
my.data &lt;- get_ausplots(veg.PI=TRUE)

#Set maximum number of plots to all available in the veg.PI data:
ausplots_visual(my.data, map = FALSE, map.attribute = TRUE, 
fraction.pie = FALSE, growthform.pie = FALSE, cumulative.cover = FALSE, 
whittaker = FALSE, outfile = "AusPlots_treeCover.pdf", 
max.plots=length(unique(my.data$veg.PI$site_location_name))) 

## End(Not run)
</code></pre>

<hr>
<h2 id='ausplotsR-package'>
Live extraction, preparation, visualisation and analysis of TERN AusPlots ecosystem monitoring data.
</h2><span id='topic+ausplotsR-package'></span><span id='topic+ausplotsR'></span>

<h3>Description</h3>

<p>Through ausplotsR, users can directly access plot-based data on vegetation and soils across Australia, including physical sample barcode numbers. Simple function calls extract the data and merge them into species occurrence matrices for downstream analysis or calculate things like basal area and fractional cover.
</p>
<p>ausplotsR whas been tested for R version 4 (recommended).
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> ausplotsR</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.0.5</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2023-11-7</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>TERN AusPlots is a national field plot-based ecosystem surveillance monitoring method and dataset for Australia. A standardised national network of plots enables consistent ecological assessment and monitoring. The data are open access under Creative Commons – Attribution 4.0 International (CC BY 4.0), and have been collected across a national network of plots and transects by the Terrestrial Ecosystem Research Network (TERN - <a href="https://www.tern.org.au">https://www.tern.org.au</a>), an Australian Government NCRIS-enabled project, and its Ecosystem Surveillance platform (<a href="https://www.tern.org.au/tern-land-observatory/ecosystem-surveillance-and-environmental-monitoring/">https://www.tern.org.au/tern-land-observatory/ecosystem-surveillance-and-environmental-monitoring/</a>).
</p>
<p>The AusPlots survey method is made up of a series of individual modules, which are described in full in the protocols manual (White et al. 2012, Sparrow et al. 2020). Data from the following modules can currently be accessed through ausplotsR: plot selection and layout; vascular plant vouchering; point intercept; basal area; structural summary; and soils. Field data are recorded directly onto mobile (tablet) devices before being stored in cloud-based server infrastructure. <br />
</p>
<p><em>Method summaries:</em>
</p>
<p><em>Plot layout</em>: 1 ha (100 x 100 m) plots are permanently marked over a homogenous patch of terrestrial vegetation.
</p>
<p><em>Soil sampling</em>: soils are characterised in a pit at the SW corner of the plot to 1 m depth and 500 g samples are collected at 10 cm depth intervals; soil bulk density is measured; nine soil subsites are located across the plots and 500 g samples are taken at 10 cm depths to 30 cm. Additionally, 200 g surface soil (~3 cm depth) samples are taken at the nine subsites and silica-dried for metagenomic analysis.
</p>
<p><em>Structural summary</em>: the dominant species in three vegetation strata (lower, middle and upper) are identified visually by the observer. The vegetation is then categorised into MVGs according to the Australian National Vegetation Information System (NVIS 2017).
</p>
<p><em>Vascular plant vouchering</em>: all vascular plants within the plot are detected visually, with a herbarium voucher taken for each unique taxon, which is assigned a barcode and identified then permanently stored in a herbarium. Additionally, silica-dried material is sampled from each species for genetic and isotopic analysis, and five replicate samples are taken for dominant species. Bryophytes are sometimes recorded opportunistically (e.g. when prominent) but full diversity is not systematically collected.
</p>
<p><em>Point-intercept</em> : 10 x 100 m long transects are laid out within the plot in a grid pattern. A staff with laser pointer and densiometer is used to record species, growth form and height plus substrate type every 1 m along the transects, resulting in a total of 1010 (10 x 101) point-intercept hits for the plot.
</p>
<p><em>Basal wedge</em> : a basal wedge is used to determine the basal area of trees and shrubs at 1.3 m, with measures of hits per species taken at nine locations across the plot.
</p>
<p>Additional modules in the AusPlots method not available through this package include: full DGPS coordinate data (only the coordinates of the SW corner are provided here), photo panoramas, Leaf Area Index. <br />
</p>
<p><em>Acknowledging TERN AusPlots data in publications:</em>
</p>
<p>It is important that users of data extracted via ausplotsR acknowledge TERN, the AusPlots dataset and ausplotsR by including an appropriate citation in the Acknowledgments or References section of any published papers/reports/theses that make use of the data. To make this easy, a data citation is automatically generated when you extract data (see <code><a href="#topic+get_ausplots">get_ausplots</a></code>). It has the following format:
</p>
<p>TERN (&quot;year&quot;) AusPlots ecosystem surveillance monitoring dataset 
(URL: http://aekos.org.au/). Obtained via the ausplotsR R package (URL: https://github.com/ternaustralia/ausplotsR), accessed &quot;day month year&quot;.
</p>
<p>To cite the ausplotsR package itself and the functions it contains, enter: <code>citation('ausplotsR')</code>, or see links to citable archived versions with DOI at <a href="https://github.com/ternaustralia/ausplotsR">https://github.com/ternaustralia/ausplotsR</a>. <br />
</p>
<p><em>Key functions in the ausplotsR package:</em>
</p>
<p><code><a href="#topic+get_ausplots">get_ausplots</a></code> :  This function is the starting point for accessing data through the ausplotsR package for the following survey modules: Site information; Vegetation point intercept; Vegetation vouchers; Basal wedge; Vegetation structural summaries; Soils and soil samples information.
</p>
<p><code><a href="#topic+ausplots_visual">ausplots_visual</a></code> :  Quick maps and demonstration graphics for AusPlots.
</p>
<p><code><a href="#topic+species_table">species_table</a></code> :  Species occurrence matrices for AusPlots.
</p>
<p><code><a href="#topic+growth_form_table">growth_form_table</a></code> :  Plant growth form occurrence matrices for TERN AusPlots.
</p>
<p><code><a href="#topic+fractional_cover">fractional_cover</a></code> :  Fractional cover based on TERN AusPlots data.
</p>
<p><code><a href="#topic+basal_area">basal_area</a></code> :  Basal area of tree trunks in TERN AusPlots.
</p>
<p><code><a href="#topic+single_cover_value">single_cover_value</a></code> :  Vegetation cover in TERN AusPlots by height and growth form subsets.
</p>
<p><code><a href="#topic+species_list">species_list</a></code> :  Simplified plant species lists for TERN AusPlots based on verified vouchers.
</p>
<p><code><a href="#topic+ausplots_trajectory">ausplots_trajectory</a></code> :  Visualisation of species composition and community index revisit trajectories.
</p>
<p><code><a href="#topic+optim_species">optim_species</a></code> :  Species accumulation optimisation for community data including TERN AusPlots.
</p>


<h3>Author(s)</h3>

<p>Greg Guerin, Tom Saleeba, Samantha Munroe, Bernardo Blanco-Martin, Irene Martin-Fores, Walid Al Naim, Andrew Tokmakoff
</p>
<p>Maintainer: Greg Guerin &lt;ggueri01@gmail.com&gt;
</p>


<h3>References</h3>

<p>Guerin, G. R., Sparrow, B., Tokmakoff, A., Smyth, A., Leitch, E., Baruch, Z., &amp; Lowe, A. J. (2017). Opportunities for integrated ecological analysis across inland Australia with standardised data from Ausplots Rangelands. <em>PLoS ONE</em>, <strong>12</strong>, e0170137.
</p>
<p>Munroe, S., Guerin, G.R., Saleeba, T., Martin-Fores, M., Blanco-Martin, B., Sparrow, B. &amp; Tokmakoff, A. (2021). ausplotsR: An R package for rapid extraction and analysis of vegetation and soil data collected by Australia’s Terrestrial Ecosystem Research Network. <em>Journal of Vegetation Science</em>, <strong>32</strong>, e13046.
</p>
<p>NVIS Technical Working Group (2017). Australian Vegetation Attribute Manual: National Vegetation Information System, Version 7.0. Department of the Environment and Energy, Canberra. Prep by Bolton, M.P., deLacey, C. and Bossard, K.B. (Eds)
</p>
<p>Sparrow, B., Foulkes, J., Wardle, G., Leitch, E., Caddy-Retalic, S., van Leeuwen, S., Tokmakoff, A., Thurgate, N., Guerin, G.R. and Lowe, A.J. (2020). A vegetation and soil survey method for surveillance monitoring of rangeland environments. <em>Frontiers in Ecology and Evolution</em>, <strong>8</strong>, 157.
</p>
<p>Tokmakoff, A., Sparrow, B., Turner, D., &amp; Lowe, A. (2016). AusPlots Rangelands field data collection and publication: Infrastructure for ecological monitoring. <em>Future Generation Computer Systems</em>, <strong>56</strong>, 537-549.
</p>
<p>White, A., Sparrow, B., Leitch, E., Foulkes, J., Flitton, R., Lowe, A. J., &amp; Caddy-Retalic, S. (2012). AUSPLOTS rangelands survey protocols manual. University of Adelaide Press.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_ausplots">get_ausplots</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#load package:
library(ausplotsR)

#get data for selected plots:
my.data &lt;- get_ausplots(my.Plot_IDs=c("SATFLB0004", "QDAMGD0022", "NTASTU0002"), 
basal.wedge=TRUE, structural_summaries=TRUE)

#view module:
my.data$struct.summ

#get site info for all available plot visits:
get_ausplots() 

#get a species by sites table:
my.data &lt;- get_ausplots(my.Plot_IDs=c("SATFLB0004", "QDAMGD0022", "NTASTU0002"),
veg.PI=TRUE)

species_table(my.data$veg.PI, m_kind="percent_cover", cover_type="PFC",
species_name="SN")

## End(Not run)
</code></pre>

<hr>
<h2 id='basal_area'>Basal area of tree trunks in TERN AusPlots
</h2><span id='topic+basal_area'></span>

<h3>Description</h3>

<p>Calculates basal area (or number of basal wedge hits) for each plot, using the raw basal wedge data returned from <code>get_ausplots</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>basal_area(veg.basal, by.spp=FALSE, by.hits=FALSE, 
species_name=c("SN","HD","GS"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="basal_area_+3A_veg.basal">veg.basal</code></td>
<td>
<p>Raw basal wedge data generated from <code><a href="#topic+get_ausplots">get_ausplots</a></code>.
</p>
</td></tr>
<tr><td><code id="basal_area_+3A_by.spp">by.spp</code></td>
<td>
<p>Set to TRUE to get values per individual species in plots rather than combined values per plot.
</p>
</td></tr>
<tr><td><code id="basal_area_+3A_by.hits">by.hits</code></td>
<td>
<p>Set to TRUE to return the number of individual basal wedge hits rather than basal area per se.
</p>
</td></tr>
<tr><td><code id="basal_area_+3A_species_name">species_name</code></td>
<td>
<p>If <code>by.spp=TRUE</code>, set to &quot;SN&quot; to use &quot;standardised_names&quot; (the default), &quot;HD&quot; to use &quot;herbarium_determination&quot;, and &quot;GS&quot; to use &quot;genus_species&quot;. See details for explanation of each setting.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Stand basal area is the cross-sectional area of tree trunks per unit area, and is measured at each plot using a basal wedge sweep. Measurement is by species at nine points, and basal area (number of hits at a defined wedge angle multiplied by the basal wedge factor) is averaged across them (Finch 1957). Raw tree stem hits are also recorded, noting that the number of hits is dependent on the basal wedge factor in use. Note that species-level estimates of basal area (if requested in the function) may be unstable if there were fewer than 7 hits of that species.
</p>
<p><code>species_name</code> provides multiple options for calculating basal area if <code>by.spp=TRUE</code>. For example, if <code>species_name="HD"</code>, the function will use the &quot;herbarium_determination&quot; provided by state herbaria to identify unique species. &quot;herbarium_determination&quot; are made to the lowest possible taxonomic rank, which may be the family, genus, species, subspecies or variety. If <code>species_name="SN"</code>, the default, the function will use the the &quot;standardised_name&quot; to identify unique species. Values are based on the <code>veg.PI$herbarium_determination</code> but have been matched to the accepted scientific name according to the Australian Plant Census (APC: https://www.anbg.gov.au/cpbr/program/hc/hc-APC.html) and the Australian Plant Name Index (APNI: https://www.anbg.gov.au/cpbr/program/hc/hc-APNI.html). The standardised name field therefore helps to ensure taxonomic consistency. It excludes any records that don't match the APC. If <code>species_name="GS"</code>, only the standardised genus species rank (&quot;genus_species&quot; in the veg.basal input table) is used to identify species. <code>species_name="GS"</code> maximises consistency in species comparisons between plots, but reduces the taxonomic detail.
</p>


<h3>Value</h3>

<p>Returns a dataframe with rows for plots (or species by plots if <code>by.spp=TRUE</code>) and basal area (m2/ha) or hit scores as mean per plot/species across the 9 sweep points. Species means (if selected) are mean across all 9 sweep points, even if a species is recorded at fewer points.
</p>


<h3>Author(s)</h3>

<p>Greg R. Guerin, Samantha E.M. Munroe
</p>


<h3>References</h3>

<p>Finch, H.D.S. (1957) Plotless enumeration with angle gauges.Forestry 30, 173-192.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_ausplots">get_ausplots</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#get basal wedge data for selected plots:
my.data &lt;- get_ausplots(my.Plot_IDs=c("SATFLB0004", "QDAMGD0022", "NTASTU0002"),
site_info=FALSE, basal.wedge=TRUE)

#calculate basal area:
basal_area(my.data$veg.basal)

## End(Not run)
</code></pre>

<hr>
<h2 id='fractional_cover'>Fractional cover based on TERN AusPlots data.
</h2><span id='topic+fractional_cover'></span>

<h3>Description</h3>

<p>Calculates fractional cover (i.e., the proportional cover of green vegetation, dead vegetation and bare substrate) based on plot-based point-intercept data from AusPlots (as generated by <code><a href="#topic+get_ausplots">get_ausplots</a></code>). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fractional_cover(veg.PI, ground_fractional=FALSE, in_canopy_sky=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fractional_cover_+3A_veg.pi">veg.PI</code></td>
<td>
<p>The input raw point-intercept data as generated in the <code><a href="#topic+get_ausplots">get_ausplots</a></code> function.
</p>
</td></tr>
<tr><td><code id="fractional_cover_+3A_ground_fractional">ground_fractional</code></td>
<td>
<p>Logical. Set to TRUE to calculate fractional GROUND cover.
</p>
</td></tr>
<tr><td><code id="fractional_cover_+3A_in_canopy_sky">in_canopy_sky</code></td>
<td>
<p>Logical. Set to TRUE to include in green fraction, thereby calculating opaque canopy cover.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Cover fractions are assigned according to the following:
</p>
<p>'Green' or 'photosynthetic vegetation' is living vascular plant cover.
</p>
<p>'Brown' or 'non-photosynthetic vegetation' is either vascular plant cover scored as 'dead' or substrate scored as litter, coarse woody debris or cryptogam (see below) that has no other veg cover.
</p>
<p>'Bare' or 'bare ground' is substrate that is rock, outcrop, gravel or bare soil with no veg cover.
</p>
<p>A height rule is applied so that coding to green/brown/bare of the uppermost substrate/vegetation stratum hit at a given point intercept location overrides the others. That is, a dead tree overrides a living shrub beneath and vice versa; substrate coding is overriden by any vegetation cover etc. This means for each of the (usually) 1010 intercepts, there is a single coding and percentage is the number of hits assigned to each fraction, divided by the total number of PIs taken (usually 1010 but can vary) times 100.
</p>
<p>There is an option via argument 'ground_fractional' to calculate fractional ground cover - the same concept applied to only grasses (hummock, tussock, other); sedge; rush; forb; fern; and vine plant growth forms. Presently, cryptogam cover is excluded and included in the non-photosynthetic fraction.
</p>
<p>'In canopy sky' is excluded by default (only the substrate is considered for those hits) and applies only to regular fractional cover (as trees are excluded in the green fraction for ground fractional cover by default).
</p>
<p>Currently, cryptogam substrate is assigned to the non-photosynthetic fraction.
</p>
<p>Occasionally substrate type was not collected ('NC') or could not be assigned to one of the above categories ('Unknwn'), in which case a percent cover will be returned under an 'NA' fraction if there was no veg cover above those points.
</p>


<h3>Value</h3>

<p>Returns a data frame in which plots are rows, columns are fractions (bare, brown, green and NA) and values are percent cover.
</p>


<h3>Author(s)</h3>

<p>Greg Guerin
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_ausplots">get_ausplots</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#get veg.PI data for selected plots:
my.data &lt;- get_ausplots(my.Plot_IDs=c("SATFLB0004", "QDAMGD0022", "NTASTU0002"),
	veg.PI=TRUE, site_info=FALSE)

#calculate fractional cover:
fractional_cover(my.data$veg.PI)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_ausplots'>Extract AusPlots data for sites, soils and vegetation
</h2><span id='topic+get_ausplots'></span>

<h3>Description</h3>

<p>This function is the starting point for accessing data through the ausplotsR package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_ausplots(my.Plot_IDs="none", site_info=TRUE, structural_summaries=FALSE,
	veg.vouchers=FALSE, veg.PI=FALSE, basal.wedge=FALSE, soil_subsites=FALSE,
	soil_bulk_density=FALSE, soil_character=FALSE, plot_search=NULL, 
	bounding_box="none", herbarium_determination_search=NULL, family_search=NULL, 
	standardised_name_search=NULL, dictionary=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_ausplots_+3A_my.plot_ids">my.Plot_IDs</code></td>
<td>
<p>Optional character vector of AusPlots plot IDs (One of <code>site_location_name</code> or <code>site_unique</code> fields) to request data for specific set of plots/visits.
</p>
</td></tr>
<tr><td><code id="get_ausplots_+3A_site_info">site_info</code></td>
<td>
<p>Whether site summary data are required (includes plot and visit details, landform data, geographic coordinates, notes etc). Default is to include.
</p>
</td></tr>
<tr><td><code id="get_ausplots_+3A_structural_summaries">structural_summaries</code></td>
<td>
<p>Whether site vegetation structural summaries are required.
</p>
</td></tr>
<tr><td><code id="get_ausplots_+3A_veg.vouchers">veg.vouchers</code></td>
<td>
<p>Whether vegetation vouchers data are requested - contains a complete set of species records for the plot determined by a herbarium plus ID numbers for silica-dried tissue samples.
</p>
</td></tr>
<tr><td><code id="get_ausplots_+3A_veg.pi">veg.PI</code></td>
<td>
<p>Whether point-intercept data are requested; includes data on substrate, plant species, growth form and height etc at each of (typically) 1010 points per plot.
</p>
</td></tr>
<tr><td><code id="get_ausplots_+3A_basal.wedge">basal.wedge</code></td>
<td>
<p>Whether basal wedge data raw hits are required to enable calculation of basal area by species by plot.
</p>
</td></tr>
<tr><td><code id="get_ausplots_+3A_soil_subsites">soil_subsites</code></td>
<td>
<p>Whether soil subsite information is required. This includes information on what soil and soil metagenomics samples were taken at nine locations across the plot and their identification barcode numbers.
</p>
</td></tr>
<tr><td><code id="get_ausplots_+3A_soil_bulk_density">soil_bulk_density</code></td>
<td>
<p>Whether soil bulk density data are required.
</p>
</td></tr>
<tr><td><code id="get_ausplots_+3A_soil_character">soil_character</code></td>
<td>
<p>Whether soil characterisation and sample ID data at 10 cm increments to a depth of 1 m are required.
</p>
</td></tr>
<tr><td><code id="get_ausplots_+3A_plot_search">plot_search</code></td>
<td>
<p>Optional character string filter for selecting AusPlots based on wildcard/partial match to the <code>site_location_name</code>.
</p>
</td></tr>
<tr><td><code id="get_ausplots_+3A_bounding_box">bounding_box</code></td>
<td>
<p>An additional optional spatial filter for selecting AusPlots based on a rectangular box, in the format of e.g. c(xmin, xmax, ymin, ymax). AusPlots location data are are in longlat, therefore x is the longitude and y is the latitude of the box/extent object. e.g., c(120, 140, -30, -10).
</p>
</td></tr>
<tr><td><code id="get_ausplots_+3A_herbarium_determination_search">herbarium_determination_search</code></td>
<td>
<p>Optional character string filter for selecting AusPlots based on the presence of a genus or species as recorded in the &quot;herbarium_determination&quot; column. See details for search term settings.
</p>
</td></tr>
<tr><td><code id="get_ausplots_+3A_family_search">family_search</code></td>
<td>
<p>Optional character string filter for selecting AusPlots based on the presence of a plant family as recorded in the 'family' column.  Will be NA if no family identification was possible. See details for search term settings.
</p>
</td></tr>
<tr><td><code id="get_ausplots_+3A_standardised_name_search">standardised_name_search</code></td>
<td>
<p>Optional character string filter for selecting AusPlots based on the presence of a genus and/or species as recorded in the 'standardised_name' column. Will be NA if no genus rank (at minimum) was possible. See details for search term settings.
</p>
</td></tr>
<tr><td><code id="get_ausplots_+3A_dictionary">dictionary</code></td>
<td>
<p>Whether the metadata variable dictionary is required. Provides a table that describes many of the variables and values in each data frame.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, the function will extract and compile a site information table including details of all available plot visits. Arguments allow the user to select sites by plot ID and visit codes, geographic bounding box, or species, and select which modules data are drawn from. Partial/wildcard site code matches are enabled in the <code>plot_search</code> argument, and further spatial filtering of sites from a master list is possible by using the 'bioregion_name', 'state', and spatial coordinate fields in the site table. Additional tables such as vegetation vouchers, point intercept modules and soil attributes can be requested by setting the desired modules to 'TRUE'.
</p>
<p>The datasets returned by the function are used directly by further formatting and analysis functions in ausplotsR (such as <code><a href="#topic+species_table">species_table</a></code> to generate a species by sites matrix for community analysis), making it simple to go from raw data extract to ecological analysis.
</p>
<p>AusPlots surveys are identified by a plot ID called <code>site_location_name</code> and unique visit number, that are combined to identify a unique measurement for sites with repeat visits, identified in the extracted data tables as 'site_unique'.
</p>
<p>&quot;herbarium_determination_search&quot;, &quot;family search&quot;, and &quot;standardised_name_search&quot; search terms are not case sensitive and do not require an exact taxonomic match (e.g. &quot;Eucalyptus moderata&quot;, &quot;Eucalyptus&quot;, and &quot;euca&quot; are all acceptable search terms). If <code>veg.vouch=TRUE</code>, veg.vouch will return a data frame that only includes voucher records that match the species or family search. If <code>veg.PI=TRUE</code>, veg.PI will return point intercept data for all plots where the species or family occurs. If <code>basal.wedge=TRUE</code>, veg.basal will return a dataframe that only includes the raw basal wedge data of records that match the species or family. For all other data.frames, get_ausplots will return data for all the plots where the species or family occurs. 
</p>
<p><code>herbarium determinations</code> are provided by state herbaria and are the most commonly used scientific names in the given state. However, scientific names may differ between states and with international databases due to disagreements on taxonomy/nomenclature. Herbarium determinations are made to the lowest possible taxonomic rank, which may be the family, genus, species, subspecies or variety.
</p>
<p><code>standardised_name_search</code> values are based on the <code>veg.PI$herbarium_determination</code> but have been matched to the accepted scientific name according to the Australian Plant Census (APC: https://www.anbg.gov.au/cpbr/program/hc/hc-APC.html) and the Australian Plant Name Index (APNI: https://www.anbg.gov.au/cpbr/program/hc/hc-APNI.html). The standardised name field therefore helps to ensure taxonomic consistency. <code>standardised_name_search</code> determinations are made to the lowest possible taxonomic rank, which may be the family, genus, species, subspecies or variety. It excludes any records that don't match the APC. Additional taxonomic fields based on the standardised_name are listed as separate columns in veg.PI, veg.vouch, and veg.basal data tables.  
</p>
<p>The metadata variable dictionary returned by <code>dictionary=T</code> may not include all of the the variables and values in each data frame, but will be expanded over time.
</p>


<h3>Value</h3>

<p>Returns a list with following elements, depending on the AusPlots modules requested in the function call.
</p>
<table>
<tr><td><code>$site.info</code></td>
<td>
<p>Data frame with basic site information including location.</p>
</td></tr>
<tr><td><code>$struct.summ</code></td>
<td>
<p>Data frame with vegetation structural summaries for each plot.</p>
</td></tr>
<tr><td><code>$veg.vouch</code></td>
<td>
<p>Data frame with rows for each voucher and information on species determinations and silica-dried tissue samples.</p>
</td></tr>
<tr><td><code>$veg.PI</code></td>
<td>
<p>Data frame with individual point-intercept data.</p>
</td></tr>
<tr><td><code>$veg.basal</code></td>
<td>
<p>Data frame with compiled raw basal wedge hit data.</p>
</td></tr>
<tr><td><code>$soil.sub</code></td>
<td>
<p>Data frame with details of soil subsites within each plot including sample IDs.</p>
</td></tr>
<tr><td><code>$soil.bulk</code></td>
<td>
<p>Data frame with raw bulk density data from each plot.</p>
</td></tr>
<tr><td><code>$soil.char</code></td>
<td>
<p>Data frame with soil characterisation data from the 1 m pit at the SW corner of each plot.</p>
</td></tr>
<tr><td><code>$metadata.dictionary</code></td>
<td>
<p>Data frame that lists and describes each variable and corresponding values in each data frame.</p>
</td></tr>
<tr><td><code>$citation</code></td>
<td>
<p>Auto-generated citation for the data extracted. Please cite ausplotsR and the TERN AusPlots data you use.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Greg Guerin, Andrew Tokmakoff, Tom Saleeba, Samantha Munroe
</p>


<h3>See Also</h3>

<p><code><a href="#topic+species_table">species_table</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#get data for selected plots:
my.data &lt;- get_ausplots(my.Plot_IDs=c("SATFLB0004", "QDAMGD0022", "NTASTU0002")
, basal.wedge=TRUE, structural_summaries=TRUE)

#View module:
my.data$struct.summ

#Get data citation and extraction date info:
my.data$citation

#Get site info for all available plots:
my.data &lt;- get_ausplots()

##Subset site records to plots with revisits only
#vector of plots with revisits
revisited_plots &lt;- unique(my.data$site.info$site_location_name[
which(my.data$site.info$visit_number &gt; 1)])

#subset site table to revisited plots
revisited_site.info &lt;- my.data$site.info[
which(my.data$site.info$site_location_name 

#Get voucher records for "Eucalyptus moderata" using "herbarium_determination":
Eucalyptus_moderata &lt;- get_ausplots(site_info=FALSE, veg.vouchers=TRUE, 
herbarium_determination_search="Eucalyptus moderata")

head(Eucalyptus_moderata$veg.vouch, 10)

#Get veg voucher records for all species in the family "Myrtaceae"
Myrtaceae &lt;- get_ausplots(site_info=FALSE, veg.vouchers=TRUE, family_search="Myrtaceae")
head(Myrtaceae$veg.vouch,10)

## End(Not run)
</code></pre>

<hr>
<h2 id='growth_form_table'>Plant growth form occurrence matrices for TERN AusPlots
</h2><span id='topic+growth_form_table'></span>

<h3>Description</h3>

<p>Generates occurrence matrices for NVIS plant growth forms in plots as desired based on presence/absence, percent cover or species richness (i.e., the number of species assigned to a particular growth form). The input is a data frame of raw point intercept data from AusPlots generated using the <code><a href="#topic+get_ausplots">get_ausplots</a></code> function. Alternatively, cover by strata can be calculated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>growth_form_table(veg.PI, m_kind=c("PA", "percent_cover", "richness"),	
cover_type=c("PFC", "OCC"), cumulative=TRUE, by_strata=FALSE, 
species_name=c("SN","HD","GS"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="growth_form_table_+3A_veg.pi">veg.PI</code></td>
<td>
<p>The input raw point intercept data generated by <code><a href="#topic+get_ausplots">get_ausplots</a></code>
</p>
</td></tr>
<tr><td><code id="growth_form_table_+3A_m_kind">m_kind</code></td>
<td>
<p>The desired scoring method: binary (presence/absence; PA), percent cover or species richness.
</p>
</td></tr>
<tr><td><code id="growth_form_table_+3A_cover_type">cover_type</code></td>
<td>
<p>A choice between PFC: 'projected foliage cover' and OCC: 'opaque canopy cover' and only applies to percent_cover. If 'PFC', the default, is selected (suitable as an abundance or cover measure for most applications), hits scored as 'in canopy sky' are removed, whereas they are retained as cover for that species for 'OCC' (suitable for application such as comparisons to remotely sensed cover).
</p>
</td></tr>
<tr><td><code id="growth_form_table_+3A_cumulative">cumulative</code></td>
<td>
<p>Logical. Should the cover of each growth form be calculated cumulatively across species (i.e., the sum of species covers in that growth form category; the default) or as absolute cover (percent of points where that growth form was hit)? Ignored except when <code>m_kind="percent_cover"</code>.
</p>
</td></tr>
<tr><td><code id="growth_form_table_+3A_by_strata">by_strata</code></td>
<td>
<p>Logical. If set to <code>TRUE</code>, cover is calculated by strata (as defined by growth forms).  Ignored except when <code>m_kind="percent_cover"</code>.
</p>
</td></tr>
<tr><td><code id="growth_form_table_+3A_species_name">species_name</code></td>
<td>
<p>Select &quot;SN&quot; to use &quot;standardised_names&quot; (the default), &quot;HD&quot; to use &quot;herbarium_determination&quot;, and &quot;GS&quot; to use &quot;genus_species&quot;. See details for an explanation of each setting.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Probably the most useful data output from this function is the matrix of plant growth forms against sites with percent cover. This could be used in manova, ordination, classification etc. It also provides a multivariate structural and functional response variable.
</p>
<p><code>species_name</code> provides multiple options for calculating plant growth form cover. If <code>species_name="HD"</code>, the function will use the &quot;herbarium_determination&quot; provided by state herbaria to identify unique species. &quot;herbarium_determination&quot; are made to the lowest possible taxonomic rank, which may be the family, genus, species, subspecies or variety. If <code>species_name="SN"</code>, the default, the function will use the the &quot;standardised_name&quot; to identify unique species. Values are based on the <code>veg.PI$herbarium_determination</code> but have been matched to the accepted scientific name according to the Australian Plant Census (APC: https://www.anbg.gov.au/cpbr/program/hc/hc-APC.html) and the Australian Plant Name Index (APNI: https://www.anbg.gov.au/cpbr/program/hc/hc-APNI.html). The standardised name field therefore helps to ensure taxonomic consistency. It excludes any records that don't match the APC. If <code>species_name="GS"</code>, only the standardised genus species rank (&quot;genus_species&quot; in the veg.PI input table) is used to identify species. <code>species_name="GS"</code> maximises consistency in species comparisons between plots, but reduces the taxonomic detail.
</p>
<p>Note sometimes the same species is recorded with different growth forms in a plot and therefore the rowSums of the occurrence matrix when m_kind is set to &quot;richness&quot; can be higher than the observed species richness because the same species can count towards the weights for multiple growth forms.
</p>
<p>By default, cumulative percent cover (percent of hits out of total number of point intercepts) is returned, meaning multiple species of the same growth form hit at a point add to the score, which can therefore exceed 100 percent. If <code>cumulative=FALSE</code>, non-cumulative cover is returned, whereby only one hit is counted per growth form per point-intercept, with a maximum of 100.
</p>
<p>If <code>by_strata=TRUE</code>, returns percent cover of three pre-defined strata: Upper, Mid and Lower, based on the crude assignment of growth forms such that Upper includes &quot;Tree Mallee&quot;, &quot;Tree/Palm&quot;, &quot;Tree-fern&quot;, &quot;Epiphyte&quot;; Mid includes &quot;Shrub Mallee&quot;, &quot;Shrub&quot;, &quot;Grass-tree&quot;, &quot;Chenopod&quot;, &quot;Heath-shrub&quot;; and Lower includes &quot;Tussock grass&quot;, &quot;Forb&quot;, &quot;Vine&quot;, &quot;Hummock grass&quot;, &quot;Fern&quot;, &quot;Sedge&quot;, &quot;Rush&quot;. The argument <code>cumulative</code> still applies for cover by strata, where the default (<code>TRUE</code>) is to calculate the cumulative cover of species in each stratum. Absolute cover (i.e., maximum value of 100) of strata can be calculated by setting <code>cumulative=FALSE</code>.
</p>
<p>The returned occurrence matrix (plant growth forms against sites) can be converted to long format (with rows as individual occurrences of growth forms) using:
</p>
<p><code>my.matrix &lt;- growth_form_table(my.veg.PI, m_kind="percent_cover",
cover_type="PFC")</code> 
</p>
<p><code>library(reshape2)</code>
</p>
<p><code>my.matrix &lt;- as.data.frame(melt(as.matrix(my.matrix)))</code>
</p>
<p><code>my.matrix &lt;- my.matrix[-which(my.matrix $value == 0),]</code>
</p>
<p>The <code><a href="base.html#topic+round">round</a></code> function can be used on the returned matrix to estimate percent covers to the nearest percent if desired (i.e., with digits=0)
</p>


<h3>Value</h3>

<p>Returns a data frame with plant growth forms as columns and sites as rows. Values are determined by argument m_kind. For 'PA', binary presence/absence (0/1) is returned; for 'richness', the numbers of species (integer) in growth form categories are returned; for 'percent_cover', percent cover from point-intercept is returned (see <code>Details</code>).
</p>


<h3>Author(s)</h3>

<p>Greg R. Guerin, Samantha E.M. Munroe
</p>


<h3>References</h3>

<p>NVIS Technical Working Group (2017) Australian Vegetation Attribute Manual: National Vegetation Information System, Version 7.0. Department of the Environment and Energy, Canberra. Prep by Bolton, M.P., deLacey, C. and Bossard, K.B. (Eds)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_ausplots">get_ausplots</a></code>
<br />
<br />
<code><a href="#topic+single_cover_value">single_cover_value</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#get veg.PI data for selected plots:
my.data &lt;- get_ausplots(my.Plot_IDs=c("SATFLB0004", "QDAMGD0022", "NTASTU0002"),
	veg.PI=TRUE, site_info=FALSE)

#generate a percent cover occurrence matrix for plant growth forms:
growth_form_table(my.data$veg.PI, m_kind="percent_cover", cover_type="PFC",
cumulative=FALSE, species_name="SN")

#calculate cover of strata:
growth_form_table(my.data$veg.PI, m_kind="percent_cover", 
cover_type="PFC", cumulative=FALSE, by_strata=TRUE, species_name="SN")

## End(Not run)
</code></pre>

<hr>
<h2 id='optim_species'>Optimisation methods for plot selection using different diversity indices in order to maximise species accumulation</h2><span id='topic+optim_species'></span>

<h3>Description</h3>

<p>This function applies different optimisation methods to select a subset of plots that maximise species accumulation. Optimisation methods vary based on the desired biodiversity parameters chosen by the user. The function operates under the 'Maximum covering problem' framework, which seeks to identify the plots that will protect the maximum number of species in a specified number of plots. The user specifies the number of plots to select, and each optimiser selects the plots that represent the largest number of species for that given number of plots.</p>


<h3>Usage</h3>

<pre><code class='language-R'>  optim_species(speciesVsitesMatrix, n.plt=250, richness=TRUE, RRR=TRUE, 
  CWE=TRUE, shannon=TRUE, simpson=TRUE, simpson_beta=TRUE, 
  start="fixed", plot_name= NULL, frequent=TRUE, random=FALSE, 
  iterations=10, plot=TRUE, verbose=TRUE)
  </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="optim_species_+3A_speciesvsitesmatrix">speciesVsitesMatrix</code></td>
<td>
<p>dataframe or matrix, formatted as species (presence/absence or abundance) versus sites, where rows are sites and columns are species. See <code>Details</code> for more information on formatting.</p>
</td></tr>
<tr><td><code id="optim_species_+3A_n.plt">n.plt</code></td>
<td>
<p>Number of plots, n, conforming the subset to be optimised.</p>
</td></tr>
<tr><td><code id="optim_species_+3A_richness">richness</code></td>
<td>
<p>Species richness selected as optimiser. By default is TRUE.</p>
</td></tr>
<tr><td><code id="optim_species_+3A_rrr">RRR</code></td>
<td>
<p>Range rarity richness selected as optimiser. By default is TRUE.</p>
</td></tr>
<tr><td><code id="optim_species_+3A_cwe">CWE</code></td>
<td>
<p>Corrected Weight Endemism selected as optimiser. By default is TRUE.</p>
</td></tr>
<tr><td><code id="optim_species_+3A_shannon">shannon</code></td>
<td>
<p>Shannon-Wiener diversity index selected as optimiser. By default is TRUE.</p>
</td></tr>
<tr><td><code id="optim_species_+3A_simpson">simpson</code></td>
<td>
<p>Simpson diversity index selected as optimiser. By default is TRUE.</p>
</td></tr>
<tr><td><code id="optim_species_+3A_simpson_beta">simpson_beta</code></td>
<td>
<p>Simpson dissimilarity selected as optimiser. By default is TRUE.</p>
</td></tr>
<tr><td><code id="optim_species_+3A_start">start</code></td>
<td>
<p>Choose under which condition the first plot for the simpson_beta optimiser is selected. The default, &quot;fixed&quot;, uses the most speciose plot as the start seed. See details for additional options.</p>
</td></tr>
<tr><td><code id="optim_species_+3A_plot_name">plot_name</code></td>
<td>
<p>Specific plot to start the optimisation when the user has selected start = &quot;defined&quot;. By default, <code>plot_name = NULL</code>. If rownames are provided, the  <code>plot_name</code> should be in character format and must match the relevant rowname in the input species matrix. If there are no rownames, <code>plot_name</code> can be numeric and will specify the row to use as the start seed.</p>
</td></tr>
<tr><td><code id="optim_species_+3A_frequent">frequent</code></td>
<td>
<p>See Details. By default is TRUE.</p>
</td></tr>
<tr><td><code id="optim_species_+3A_iterations">iterations</code></td>
<td>
<p>Number of random seed replications for <code>simpson_beta</code> if <code>frequent=TRUE</code> and also number of random replicates if <code>random=TRUE</code>.</p>
</td></tr>
<tr><td><code id="optim_species_+3A_random">random</code></td>
<td>
<p>Generation of a set of random species accumulations. By default is FALSE</p>
</td></tr>
<tr><td><code id="optim_species_+3A_plot">plot</code></td>
<td>
<p>Whether or not to immediately plot the species accumulation curves for all the optimisers.</p>
</td></tr>
<tr><td><code id="optim_species_+3A_verbose">verbose</code></td>
<td>
<p>Logical, whether to print progress of iterations to the console.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input is a species versus sites matrix or dataframe. If a dataframe, the first column can list the site names. Site names must contain text, they cannot be numeric. Depending on the desired optimiser, the input data needs to be either species presence/absence or abundance.  For richness, RRR, CWE, and simpson_beta optimisers, a presence/absence input matrix is required. For shannon and simpson optimisers, an abundance matrix is required. If the input data include abundance, the function automatically generates binary presence/absence data when required. TERN plot species occurrence matrices generated by the ausplotsR function <code><a href="#topic+species_table">species_table</a></code> can be directly incorporated into optimiser function.
</p>
<p>The <code>start</code> refers to the first plot used as the starting seed when using the simpson_beta optimiser. If &quot;fixed&quot;, the plot with the greatest species richness is chosen as the start seed. If &quot;random&quot;, a random plot is selected as the start seed. If &quot;defined&quot; the user assigns a specific starting plot as the start seed. Note that if <code>start = "defined"</code> then the <code>plot_name</code> argument must be provided. 
</p>
<p><code>plot_name</code> must be provided when start = &quot;defined&quot; for the simpson_beta optimiser. If the input matrix or dataframe contains site names, then <code>plot_name</code> must match the rowname of the desired site to be used as the starting seed. If the input matrix does not include site names, then <code>plot_name</code> should specify the row number to be used as the starting seed.
</p>
<p><code>richness</code> refers to Species richness, which is the count of the number of species present in a given site. It is best used when the goal is to identify biodiversity hotspots.
</p>
<p><code>RRR</code> refers to Range rarity richness, which is a rarity-weighted richness calculated as the inverse of the number of sites in which a species occur. It is best used when the goal is to identify areas of high biodiversity and biological uniqueness. 
</p>
<p><code>CWE</code> refers to Corrected Weight Endemism, which is calculated as range rarity richness (RRR) divided by species richness. It is best used when the goal is to identify centers of endemism highlighting range-restricted species. 
</p>
<p><code>shannon</code> refers to the Shannon&ndash;Wiener diversity index, which combines species richness and the evenness or equitability by computing the species' relative abundances. The Shannon-Wiener diversity index assumes that all species are represented in a sample and that they are randomly sampled. 
The Shannon&ndash;Wiener index is defined as
<code class="reqn">H = -\sum_i p_i \log_{b} p_i</code>, where
<code class="reqn">p_i</code> is the proportional abundance of species <code class="reqn">i</code> and <code class="reqn">b</code>
is the base of the logarithm.  It is most popular to use natural logarithms, but some argue for base <code class="reqn">b = 2</code> (which makes sense, but no real difference).
</p>
<p><code>simpson</code> refers to the Simpson diversity index which combines species richness and the evenness or equitability by computing the species' relative abundances. The Simpson diversity index is a dominance index, giving more weight to
common or dominant species. 
The Simpson diversity index is based on
<code class="reqn">D = \sum p_i^2</code>
and returns <code class="reqn">1-D</code> 
</p>
<p><code>simpson_beta</code> refers to the Simpson dissimilarity, which is based on diversity partitioning, which separates species replacement (i.e. turnover) from species loss (i.e. nestedness). 
The Simpson dissimilarity corresponds to the turnover component of the Sorensen dissimilarity. Thus, it is used to maximise species turnover. 
</p>
<p>By default, the function will extract and compile data from the top n selected plots (as specified by the user) based on all the different optimisers. 
The starting seed by default will be &quot;fixed&quot; which will correspond to the site with the highest value of species richness. 
</p>
<p><code>frequent</code> refers to the most frequent sites that have been selected using the simpson_beta optimiser in a certain number of iterations that must be defined by the user. 
In order to do so, the starting seed for the simpson_beta optimiser must be &quot;random&quot;. The result will display two different accumulation curves:
one with the most frequent selected plots and another one with the mean and standard deviation of all the species accumulation curves obtained with a random starting seed for all the iterations
</p>
<p><code>random</code> refers to the species accumulation when the sites are selected randomly.
</p>
<p><code>plot</code> calls a function that allows plotting all the species accumulation curves obtained for each of the optimisers included in the optim_species function, see <code><a href="#topic+plot_opt">plot_opt</a></code>.
</p>


<h3>Value</h3>

<p>Returns a list containing, for each optimiser within the function, a species accumulation object (see <code><a href="vegan.html#topic+specaccum">specaccum</a></code>, object as returned from vegan R package). The species accumulation refers to the cumulative curve or the number of species for a certain number of selected sites or individuals. Likewise, a list of the sites selected in order to maximise the value of the species accumulated using each optimiser will be returned. Additionally, the function will plot by default of the species cumulative curves obtained by the different optimisers.
</p>


<h3>Author(s)</h3>

<p>Irene Martin-Fores, Samantha Munroe and Greg Guerin.</p>


<h3>References</h3>

<p>Albuquerque, F. &amp; Beier, P. (2015) Rarity-weighted richness: a simple and reliable alternative to integer programming and heuristic algorithms for minimum set and maximum coverage problems in conservation planning. <em>PLoS ONE</em> <strong>10</strong>, e0119905.
</p>
<p>Baselga, A. (2010) Multiplicative partition of true diversity yields independent alpha and beta components; additive partition does not.  <em>Ecology</em> <strong>91</strong>, 1974-1981.
</p>
<p>Baselga, A. (2012) The relationship between species replacement, dissimilarity derived from nestedness, and nestedness. <em>Global Ecology and Biogeography</em> <strong>21</strong>, 1223-1232.
</p>
<p>Baselga, A. &amp; Leprieur, F. (2015) Comparing methods to separate components of beta diversity. <em>Methods in Ecology and Evolution</em> <strong>6</strong>, 1069-1079.
</p>
<p>Guerin, G.R. &amp; Lowe, A.J. (2015) 'Sum of inverse range-sizes'(SIR), a biodiversity metric with many names and interpretations.  <em>Biodiversity and conservation</em> <strong>24</strong>, 2877-2882.
</p>
<p>Guerin, G.R., Ruokolainen, L. &amp; Lowe, A.J. (2015). A georeferenced implementation of weighted endemism. <em>Methods in Ecology and Evolution</em> <strong>6</strong>, 845-852.
</p>
<p>Jost, L. (2007). Partitioning diversity into independent alpha and beta components. <em>Ecology</em> <strong>88</strong>, 2427&ndash;2439.
</p>
<p>Koleff, P., Gaston, K. J., &amp; Lennon, J. J. (2003). Measuring beta diversity for presence-absence data. <em>Journal of Animal Ecology</em> <strong>72</strong>, 367-382. 
</p>
<p>Martín‐Forés, I., Guerin, G. R., Munroe, S. E., &amp; Sparrow, B. (2021). Applying conservation reserve design strategies to define ecosystem monitoring priorities. <em>Ecology and Evolution</em> <strong>11</strong>, 17060-17070.
</p>
<p>Oksanen, J. et al. (2016) vegan: Community Ecology Package. R package version 2.4-3. Vienna: R Foundation for Statistical Computing.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot_opt">plot_opt</a></code>
<br />
<br />
<code><a href="vegan.html#topic+specaccum">specaccum</a></code>
<br />
<br />
<code><a href="vegan.html#topic+fisher.alpha">fisher.alpha</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  #example with dune database from vegan
  library(vegan)
  data(dune)
  
  optim_species(dune, n.plt=15, frequent=FALSE)
  
  #example with auplots database from ausplotsR
  ## Not run: 
  library(ausplotsR)
  
  ausplotsdata &lt;- get_ausplots(veg.PI=TRUE)
  
  ausplotsPAdata &lt;- species_table(ausplotsdata$veg.PI, m_kind="percent_cover", 
  cover_type="PFC", species_name="SN")
  
  optim_species(ausplotsPAdata, n.plt=5, iterations= 5)
  
  optim_species(ausplotsPAdata, n.plt=5, start="defined", plot=TRUE, 
  plot_name="WAANUL0001-56966", frequent=FALSE, random=TRUE, iterations=20)
  
## End(Not run)
</code></pre>

<hr>
<h2 id='plot_opt'>Plot a set of species accumulation curves together</h2><span id='topic+plot_opt'></span>

<h3>Description</h3>

<p>This function plots different species accumulation curves obtained through different optimisers in the <code><a href="#topic+optim_species">optim_species</a></code> function.</p>


<h3>Usage</h3>

<pre><code class='language-R'>  plot_opt(optim_result, choices=c("Richness", "RRR", "CWE", "Shannon", 
  "Simpson", "SimpsonBeta", "Frequent", "SimpsonBeta_randSeed", "Random"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_opt_+3A_optim_result">optim_result</code></td>
<td>
<p>Object returned from the <code><a href="#topic+optim_species">optim_species</a></code> function, which is a list of species accumulation objects (see <code><a href="vegan.html#topic+specaccum">specaccum</a></code>) obtained with the different optimisers.</p>
</td></tr>
<tr><td><code id="plot_opt_+3A_choices">choices</code></td>
<td>
<p>select the optimisers from which the species accumulations curves are going to be plotted. The choices can be &quot;Richness&quot;, &quot;RRR&quot;, &quot;CWE&quot;, &quot;Shannon&quot;, &quot;Simpson&quot;, &quot;SimpsonBeta&quot;, &quot;Frequent&quot;, &quot;SimpsonBeta_randSeed&quot;, &quot;Random&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Line colours are assigned randomly on each run.
</p>


<h3>Value</h3>

<p>This function will return a plot of the species cumulative curves obtained by the different optimisers in the <code><a href="#topic+optim_species">optim_species</a></code> function.
</p>


<h3>Author(s)</h3>

<p>Greg R. Guerin</p>


<h3>See Also</h3>

<p><code><a href="#topic+optim_species">optim_species</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  #example with dune database from vegan
  library(vegan)
  data(dune)
  
  example1 &lt;- optim_species(dune, n.plt=15, frequent=FALSE, plot=FALSE)
  
  plot_opt(example1)
  plot_opt(example1, choices=c("Richness", "SimpsonBeta"))
      
  #example with auplots database from ausplotsR
  ## Not run: 
  library(ausplotsR)
  
  ausplotsdata &lt;- get_ausplots(veg.PI=TRUE)
  ausplotsPAdata &lt;- species_table(ausplotsdata$veg.PI, m_kind="PA", 
  species_name="SN")
  
  example2 &lt;- optim_species(ausplotsPAdata, n.plt=5, iterations=5, plot=FALSE)
  plot_opt(example2)
  plot_opt(example2, 
  choices=c("RRR", "SimpsonBeta", "Frequent", "SimpsonBeta_randSeed"))
  
## End(Not run)
</code></pre>

<hr>
<h2 id='single_cover_value'>Vegetation cover by height or growth form using TERN AusPlots data.
</h2><span id='topic+single_cover_value'></span>

<h3>Description</h3>

<p>Calculates a single vegetation cover value per site based on the plot-based point-intercept data from AusPlots (as generated by <code><a href="#topic+get_ausplots">get_ausplots</a></code>). Cover can be subsetted to vegetation that has a specified minimum or maximum height and/or by plant growth forms, allowing user defined definitions of e.g. grass or tree cover. For example, by default, forest cover is calculated per plot for tree growth forms of 5 metres or higher.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>single_cover_value(veg.PI, in_canopy_sky=FALSE, 
by.growth_form=TRUE, min.height=5, max.height=NULL, 
my.growth_forms=c("Tree/Palm", "Tree Mallee"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="single_cover_value_+3A_veg.pi">veg.PI</code></td>
<td>
<p>The input raw point intercept data ($veg.PI) as generated in the <code><a href="#topic+get_ausplots">get_ausplots</a></code> function.
</p>
</td></tr>
<tr><td><code id="single_cover_value_+3A_in_canopy_sky">in_canopy_sky</code></td>
<td>
<p>Logical, whether to include in canopy sky hits to calculate 'opaque canopy cover' instead of 'projected foliage cover'. Set to <code>TRUE</code> to calculate opaque canopy cover (e.g., suitable for some applications such as comparison to remotely sensed cover products).
</p>
</td></tr>
<tr><td><code id="single_cover_value_+3A_by.growth_form">by.growth_form</code></td>
<td>
<p>Logical, whether to subset by plant growth form. If set to FALSE then only height subsetting occurs. Set to TRUE to calculate cover subsetted to specified growth forms.
</p>
</td></tr>
<tr><td><code id="single_cover_value_+3A_min.height">min.height</code></td>
<td>
<p>Numeric, sets the height in m below which PI hits are ignored. This argument is used to calculate e.g. cover of trees 5 m high or higher. Can be set to zero to ignore height and include any plant hit. If you set min.height to a negative number you will get a nonsensical output!
</p>
</td></tr>
<tr><td><code id="single_cover_value_+3A_max.height">max.height</code></td>
<td>
<p>Numeric, sets the height in m above which PI hits are ignored. This argument can be used to calculate ground cover of desired growth forms, for example. 
</p>
</td></tr>
<tr><td><code id="single_cover_value_+3A_my.growth_forms">my.growth_forms</code></td>
<td>
<p>A user-supplied character vector specifying growth forms to include in the cover calculation. Defaults to trees. Applies only when <code>by.growth_form=TRUE</code>, otherwise this argument is ignored and only the height rule is applied. For allowable growth forms, see <code>unique(veg.PI$growth_form)</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All included point intercept hits are merged to give a single percent cover value with a maximum of 100. Cover partitioned by multiple plant growth forms can be calculated using <code>growth_form_table</code>, by species using <code>species_table</code> and fractional (green, brown or bare) cover using <code>fractional_cover</code>. Note, when min.height is set to zero and by.growth_form is set to FALSE, the output is nearly the same as the green cover fraction returned from <code>fractional_cover</code>. The values can differ, however, because <code>fractional_cover</code> applies a height rule in which the highest intercept at a given point is taken, whereas <code>single_cover_value</code> finds any green cover, leading to differences, most obviously, for example, when dead trees overhang green understorey. For such general cover purposes, we recommend <code>fractional_cover</code>. <code>single_cover_value</code> is best suited to cover subset by height and growth form. 
</p>


<h3>Value</h3>

<p>Returns a data.frame in which plots are rows identified by a unique site and visit identifier ('site_unique') and the returned cover value is given in the percentCover column.
</p>


<h3>Author(s)</h3>

<p>Greg Guerin
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_ausplots">get_ausplots</a></code>
<br />
<br />
<code><a href="#topic+fractional_cover">fractional_cover</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
my.dat &lt;- get_ausplots(bounding_box=c(140, 142, -20, -18), veg.PI=TRUE, 
site_info=FALSE)

#Default settings, where cover of trees 5 m or higher is calculated:
tree.cover &lt;- single_cover_value(my.dat$veg.PI)
tree.cover

#Tussock grasses of any height (minimum height set to zero):
grass.cover &lt;- single_cover_value(my.dat$veg.PI, 
my.growth_forms=c("Tussock grass"), min.height=0) 
grass.cover

#Any green veg at least 2 m in height:
veg.height.eg &lt;- single_cover_value(my.dat$veg.PI, by.growth_form=FALSE, 
min.height=2)
veg.height.eg

#Any green veg up to 1 m high:
ground.cover &lt;- single_cover_value(my.dat$veg.PI, by.growth_form=FALSE, 
min.height=0, max.height=1)
ground.cover

## End(Not run)
</code></pre>

<hr>
<h2 id='species_list'>Species lists for AusPlots
</h2><span id='topic+species_list'></span>

<h3>Description</h3>

<p>Nice species lists based on the vegetation voucher module of Ausplots generated using the <code><a href="#topic+get_ausplots">get_ausplots</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>species_list(veg.vouch, grouping=c("by_site", "by_visit", "collapse"), 
species_name=c("SN","HD","GS"), strip_bryophytes=FALSE, 
append_family=FALSE, writefile=FALSE, outfile="my_species_lists.txt")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="species_list_+3A_veg.vouch">veg.vouch</code></td>
<td>
<p>The input raw plant voucher data generated by <code><a href="#topic+get_ausplots">get_ausplots</a></code>
</p>
</td></tr>
<tr><td><code id="species_list_+3A_grouping">grouping</code></td>
<td>
<p>One of <code>c("by_site", "by_visit", "collapse")</code>. Determines whether species names are  pooled over all visits to each plot (<code>site_location_name</code>, the default), or gathered by individual site visits (<code>site_unique</code>). The final option 'collapse' returns a complete list of species over the plots provided.
</p>
</td></tr>
<tr><td><code id="species_list_+3A_species_name">species_name</code></td>
<td>
<p>Defaults to &quot;SN&quot; (standardised_name); set to  &quot;HD&quot; to use &quot;herbarium_determination&quot;, and &quot;GS&quot; to use standardised &quot;genus_species&quot; format. See details for an explanation of each setting.
</p>
</td></tr>
<tr><td><code id="species_list_+3A_strip_bryophytes">strip_bryophytes</code></td>
<td>
<p>Logical. Switch to <code>TRUE</code> to exclude bryophyte records and only return vascular plants in Angiosperm (flowering plants), Gymnosperm (conifers) and Pteridophyte (ferns and allies) groups. Although there are a number of bryophyte records in Ausplots, they are not surveyed systematically.
</p>
</td></tr>
<tr><td><code id="species_list_+3A_append_family">append_family</code></td>
<td>
<p>Logical. If set to <code>TRUE</code>, names in the the returned species lists are appended with (and sorted by) family standardised name.
</p>
</td></tr>
<tr><td><code id="species_list_+3A_writefile">writefile</code></td>
<td>
<p>Logical. Whether or not to write the result to a text file.
</p>
</td></tr>
<tr><td><code id="species_list_+3A_outfile">outfile</code></td>
<td>
<p>Character denoting path/file/extension to attempt to write to if <code>writefile=TRUE</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The aim of the function is to provide simple, cleaned lists of species that have been recorded at various Ausplots sites, simplifying from the more detailed voucher and identification information in the <code>$veg.vouch</code> vegetation voucher module. The output is not designed for downstream analysis (which is perhaps better performed by manipulating the raw data tables) but as a rapid  means of viewing recorded species.
</p>
<p><code>species_name</code> provides multiple options for how names are presented. If <code>species_name="HD"</code>, the function will use the &quot;herbarium_determination&quot; provided by state herbaria to identify unique species. Herbarium determinations are made to the lowest possible taxonomic rank, which may be the family, genus, species, subspecies or variety. If <code>species_name="SN"</code>, the default, the function will use the the &quot;standardised_name&quot; to identify unique species. Values are based on the <code>veg.PI$herbarium_determination</code> but have been matched to the accepted scientific name according to the Australian Plant Census (APC: https://www.anbg.gov.au/cpbr/program/hc/hc-APC.html) and the Australian Plant Name Index (APNI: https://www.anbg.gov.au/cpbr/program/hc/hc-APNI.html). The standardised name field therefore helps to ensure taxonomic consistency. It excludes any records that don't match the APC. If <code>species_name="GS"</code>, only the standardised genus species rank (&quot;genus_species&quot; in the veg.vouch input table) is used to identify species. <code>species_name="GS"</code> maximises consistency in species comparisons between plots, but reduces the taxonomic detail.
</p>


<h3>Value</h3>

<p>Returns a list containing a character vector of species names for each plot/visit grouping. If <code>grouping="collapse"</code>, a single character vector is returned. If <code>writefile=TRUE</code>, a text file is written to path <code>outfile</code> in the working directory.
</p>


<h3>Author(s)</h3>

<p>Greg Guerin, Samantha Munroe
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_ausplots">get_ausplots</a></code>
<br />
<br />
<code><a href="#topic+species_table">species_table</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## Not run: 
  #get veg.vouch data for selected plots:
  my.data &lt;- get_ausplots(site_info=FALSE, veg.vouchers=TRUE, 
  my.Plot_IDs=c("SAAFLB0005", "SAAFLB0006"))
  
  #generate 'genus_species' lists by site:
  species_list(my.data$veg.vouch, grouping="by_site", 
  species_name="GS", writefile=FALSE)
  
  #generate a complete species list ordered by family:
  species_list(my.data$veg.vouch, grouping="collapse", 
  species_name="SN", append_family=TRUE, writefile=FALSE)
  
## End(Not run)
</code></pre>

<hr>
<h2 id='species_table'>Species occurrence matrices for AusPlots
</h2><span id='topic+species_table'></span>

<h3>Description</h3>

<p>This function takes a data frame of individual raw point-intercept hits from AusPlots generated using the <code><a href="#topic+get_ausplots">get_ausplots</a></code> function, and generates species occurrence matrices as desired based on presence/absence, cover, frequency or IVI.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>species_table(veg.PI, m_kind=c("PA", "percent_cover", "freq", "IVI"),
	cover_type=c("PFC", "OCC"), species_name=c("SN","HD","GS"), 
	strip_bryophytes=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="species_table_+3A_veg.pi">veg.PI</code></td>
<td>
<p>The input raw point-intercept data generated by <code><a href="#topic+get_ausplots">get_ausplots</a></code>
</p>
</td></tr>
<tr><td><code id="species_table_+3A_m_kind">m_kind</code></td>
<td>
<p>The desired species scoring method: PA: binary (presence/absence), percent cover, frequency (based on occurrences on the 10 point-intercept transects within plots) or IVI (combination of cover and frequency - see Details).
</p>
</td></tr>
<tr><td><code id="species_table_+3A_cover_type">cover_type</code></td>
<td>
<p>A choice between PFC: 'projected foliage cover' and OCC: 'opaque canopy cover' and only applies to percent_cover and IVI. If 'PFC' is selected (the default), hits scored as 'in canopy sky' are removed, whereas they are retained as cover for that species for 'OCC'. 'PFC' may be more suitable for some applications such as comparison to remotely sensed cover products.
</p>
</td></tr>
<tr><td><code id="species_table_+3A_species_name">species_name</code></td>
<td>
<p>Set to &quot;SN&quot; to use &quot;standardised_names&quot; (the default), &quot;HD&quot; to use &quot;herbarium_determination&quot;, and &quot;GS&quot; to use &quot;genus_species&quot; See details for an explanation of each setting.
</p>
</td></tr>
<tr><td><code id="species_table_+3A_strip_bryophytes">strip_bryophytes</code></td>
<td>
<p>Logical. Switch to <code>TRUE</code> to exclude bryophyte records and only return vascular plants in Angiosperm (flowering plants), Gymnosperm (conifers) and Pteridophyte (ferns and allies) groups. Although there are a number of bryophyte records in Ausplots, they are not surveyed systematically.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Species occurrence matrices are the basis for a wide range of downstream analyses in community ecology, such as classification, ordination, alpha and beta diversity, abundance distributions and many more. Many of these analyses can be performed in the vegan R package using the returned matrix.
</p>
<p>IVI (Importance Value Index; Ellenberg &amp; Mueller-Dombois 1974) is calculated as the sum of relative abundance (percent cover) and relative frequency (percent frequency of occurrence on the 10 transects used to gather the point-intercept data). It has a theoretical range of 0 - 200 and is calculated as follows:
</p>
<p style="text-align: center;"><code class="reqn">(cover / total cover) * 100 + (frequency / total frequency) * 100</code>
</p>

<p><code>species_name</code> provides multiple options for calculating plant cover, presence or frequency. If <code>species_name="HD"</code>, the function will use the &quot;herbarium_determination&quot; provided by state herbaria to identify unique species. &quot;herbarium_determination&quot; are made to the lowest possible taxonomic rank, which may be the family, genus, species, subspecies or variety. If <code>species_name="SN"</code>, the default, the function will use the the &quot;standardised_name&quot; to identify unique species. Values are based on the <code>veg.PI$herbarium_determination</code> but have been matched to the accepted scientific name according to the Australian Plant Census (APC: https://www.anbg.gov.au/cpbr/program/hc/hc-APC.html) and the Australian Plant Name Index (APNI: https://www.anbg.gov.au/cpbr/program/hc/hc-APNI.html). The standardised name field therefore helps to ensure taxonomic consistency. It excludes any records that don't match the APC. If <code>species_name="GS"</code>, only the standardised genus species rank (i.e., &quot;genus_species&quot; in the veg.PI input table) is used to identify species. <code>species_name="GS"</code> maximises consistency in species comparisons between plots, but reduces the taxonomic detail.
</p>
<p>The returned occurrence matrix (species against sites) can be converted to 'long' format (with rows as species records) using:
</p>
<p><code>my.matrix &lt;- species_table(my.veg.PI, m_kind="PA", cover_type="PFC")</code>
<br />
<code>library(reshape2)</code>
<br />
<code>my.matrix &lt;- as.data.frame(melt(as.matrix(my.matrix)))</code>
<br />
<code>my.matrix &lt;- my.matrix[-which(my.matrix $value == 0),]</code>
<br />
<br />
</p>
<p>The <code><a href="base.html#topic+round">round</a></code> function can be used on the returned matrix to estimate percent covers to the nearest percent if desired (i.e., with digits=0)
</p>


<h3>Value</h3>

<p>Returns a data frame with species as columns and sites as rows. Values are determined by argument <code>m_kind</code>.
</p>


<h3>Author(s)</h3>

<p>Greg R. Guerin, Samantha E.M. Munroe
</p>


<h3>References</h3>

<p>Ellenberg, D. &amp; Mueller-Dombois, D. (1974) Aims and methods of vegetation ecology. Wiley, New York, NY.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_ausplots">get_ausplots</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#get veg.PI data for selected plots:
my.data &lt;- get_ausplots(my.Plot_IDs=c("SATFLB0004", "QDAMGD0022", "NTASTU0002"),
	veg.PI=TRUE, site_info=FALSE)

#generate a percent cover occurrence matrix using "genus_species" species identification:
species_table(my.data$veg.PI, m_kind="percent_cover", cover_type="PFC", 
species_name="GS")

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
