<!DOCTYPE html><html><head><title>Help for package autoGO</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {autoGO}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#autoGO'><p>autoGO: Auto-GO: Reproducible, Robust and High Quality Ontology Enrichment Visualizations</p></a></li>
<li><a href='#barplotGO'><p>barplotGO</p></a></li>
<li><a href='#choose_database'><p>choose_database</p></a></li>
<li><a href='#comparisons'><p>Example comparison data.</p></a></li>
<li><a href='#counts'><p>Example raw count data.</p></a></li>
<li><a href='#deseq_analysis'><p>Differential Gene Expression Analysis</p></a></li>
<li><a href='#filtering_DE'><p>Filtering DESeq2 results</p></a></li>
<li><a href='#groups'><p>Example group data.</p></a></li>
<li><a href='#heatmapGO'><p>HeatmapGO</p></a></li>
<li><a href='#lolliGO'><p>lolliGO</p></a></li>
<li><a href='#read_enrich_tables'><p>Read enrichment results from tables</p></a></li>
<li><a href='#read_gene_lists'><p>Read all the genes lists</p></a></li>
<li><a href='#volcanoplot'><p>Volcano Plot</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Auto-GO: Reproducible, Robust and High Quality Ontology
Enrichment Visualizations</td>
</tr>
<tr>
<td>Version:</td>
<td>0.9.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Auto-GO is a framework that enables automated, high quality Gene Ontology enrichment analysis visualizations. It also features a handy wrapper for Differential Expression analysis around the 'DESeq2' package described in Love et al. (2014) &lt;<a href="https://doi.org/10.1186%2Fs13059-014-0550-8">doi:10.1186/s13059-014-0550-8</a>&gt;. The whole framework is structured in different, independent functions, in order to let the user decide which steps of the analysis to perform and which plot to produce.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/mpallocc/auto-go/issues">https://github.com/mpallocc/auto-go/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), readr (&ge; 2.1.2), dplyr (&ge; 1.1.0), enrichR</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, ComplexHeatmap, DESeq2, dichromat, ggplot2, ggrepel,
grDevices, GSVA, imguR, openxlsx, purrr, RColorBrewer,
reshape2, stats, stringr, SummarizedExperiment, textshape,
tidyr, tidyselect, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rlang, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-02-13 22:07:53 UTC; fabio</td>
</tr>
<tr>
<td>Author:</td>
<td>Isabella Grassucci
    <a href="https://orcid.org/0000-0002-2484-8570"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Eleonora Sperandio
    <a href="https://orcid.org/0000-0002-8572-6788"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Fabio Ticconi <a href="https://orcid.org/0000-0001-8905-6657"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [cre],
  Alice Massacci <a href="https://orcid.org/0000-0001-7780-809X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Lorenzo D'Ambrosio
    <a href="https://orcid.org/0000-0002-9656-7436"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Matteo Pallocca <a href="https://orcid.org/0000-0002-7756-3579"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Fabio Ticconi &lt;fabio.ticconi@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-02-13 22:50:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='autoGO'>autoGO: Auto-GO: Reproducible, Robust and High Quality Ontology Enrichment Visualizations</h2><span id='topic+autoGO'></span><span id='topic+_PACKAGE'></span><span id='topic+autoGO-package'></span>

<h3>Description</h3>

<p>Auto-GO is a framework that enables automated, high quality Gene Ontology enrichment analysis visualizations. It also features a handy wrapper for Differential Expression analysis around the 'DESeq2' package described in Love et al. (2014) <a href="https://doi.org/10.1186/s13059-014-0550-8">doi:10.1186/s13059-014-0550-8</a>. The whole framework is structured in different, independent functions, in order to let the user decide which steps of the analysis to perform and which plot to produce.
</p>
<p>Perform enrichment analysis on all the desired gene lists. This function take advantage of the 'enrichR' package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>autoGO(
  list_of_genes,
  dbs = c("GO_Molecular_Function_2021", "GO_Biological_Process_2021", "KEGG_2021_Human"),
  my_comparison = NULL,
  ensembl = FALSE,
  excel = FALSE,
  outfolder = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="autoGO_+3A_list_of_genes">list_of_genes</code></td>
<td>
<p>it can be a list of dataframes containing gene names (i.e. from the output of read_gene_lists()), a single dataframe or character vector of gene names, or a path to a .txt file containing one gene name per row.</p>
</td></tr>
<tr><td><code id="autoGO_+3A_dbs">dbs</code></td>
<td>
<p>Databases over which the enrichment will be performed, based on the enrichR libraries. Default are GO_Molecular_Function_2021, GO_Cellular_Component_2021, GO_Biological_Process_2021, KEGG_2021_Human. Run choose_database() to see all the possible databases.</p>
</td></tr>
<tr><td><code id="autoGO_+3A_my_comparison">my_comparison</code></td>
<td>
<p>Name of the comparison (or the analysis) the user would like to inspect. Ignored if list_of_genes is a list. taken from the path.</p>
</td></tr>
<tr><td><code id="autoGO_+3A_ensembl">ensembl</code></td>
<td>
<p>(Default = FALSE). Set to TRUE if the provided gene list contains Ensembl IDs. A conversion to HGNC will be performed.</p>
</td></tr>
<tr><td><code id="autoGO_+3A_excel">excel</code></td>
<td>
<p>(Default = FALSE). Set to TRUE if you want to save output tables in .xlsx format.</p>
</td></tr>
<tr><td><code id="autoGO_+3A_outfolder">outfolder</code></td>
<td>
<p>Default to NULL. The name to assign to the folder in which outputs are saved. Ignored if list_of_genes is a list, taken from the path.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. Files will be produced as part of normal execution.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Fabio Ticconi <a href="mailto:fabio.ticconi@gmail.com">fabio.ticconi@gmail.com</a> (<a href="https://orcid.org/0000-0001-8905-6657">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Isabella Grassucci <a href="mailto:isabella.grassucci@gmail.com">isabella.grassucci@gmail.com</a> (<a href="https://orcid.org/0000-0002-2484-8570">ORCID</a>)
</p>
</li>
<li><p> Eleonora Sperandio <a href="mailto:eleonora.sperandio@ifo.it">eleonora.sperandio@ifo.it</a> (<a href="https://orcid.org/0000-0002-8572-6788">ORCID</a>)
</p>
</li>
<li><p> Alice Massacci <a href="mailto:alice.massacci@ifo.it">alice.massacci@ifo.it</a> (<a href="https://orcid.org/0000-0001-7780-809X">ORCID</a>)
</p>
</li>
<li><p> Lorenzo D'Ambrosio <a href="mailto:lorenzo.dambrosio@ifo.it">lorenzo.dambrosio@ifo.it</a> (<a href="https://orcid.org/0000-0002-9656-7436">ORCID</a>)
</p>
</li>
<li><p> Matteo Pallocca <a href="mailto:matteo.pallocca@ifo.it">matteo.pallocca@ifo.it</a> (<a href="https://orcid.org/0000-0002-7756-3579">ORCID</a>)
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li><p> Report bugs at <a href="https://github.com/mpallocc/auto-go/issues">https://github.com/mpallocc/auto-go/issues</a>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
autoGO(
  list_of_genes = gene_lists,
  dbs = c("GO_Molecular_Function_2021", "GO_Biological_Process_2021", "KEGG_2021_Human"),
  my_comparison = NULL,
  ensembl = F,
  excel = F,
  outfolder = NULL
)

## End(Not run)
</code></pre>

<hr>
<h2 id='barplotGO'>barplotGO</h2><span id='topic+barplotGO'></span>

<h3>Description</h3>

<p>The function barplotGO.R implement the barplot of the first 15 enrichment terms.
</p>
<p>For each enrichment result table a barplot is produced. Results are stored in the &quot;enrichment_plots&quot; subfolder for each comparison.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>barplotGO(
  enrich_tables,
  title = NULL,
  outfolder = NULL,
  outfile = "barplotGO.png",
  from_autoGO = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="barplotGO_+3A_enrich_tables">enrich_tables</code></td>
<td>
<p>Dataframe containing the enrichment results or a path to your .tsv file containing the enrichment results. Columns 'Term' and 'Adjusted.P.Value' are required.</p>
</td></tr>
<tr><td><code id="barplotGO_+3A_title">title</code></td>
<td>
<p>Default to NULL, only specify if from_autoGO is FALSE. When enrich_tables is not from autoGO and thus from read_enrich_tables, the user can specify title and subtitle of the plot as 2-element character vector, for example: c(&quot;This is the title&quot;, &quot;this is the subtitle&quot;)</p>
</td></tr>
<tr><td><code id="barplotGO_+3A_outfolder">outfolder</code></td>
<td>
<p>Default to NULL, only specify if from_autoGO is FALSE. The name to assign to the folder for output saving.</p>
</td></tr>
<tr><td><code id="barplotGO_+3A_outfile">outfile</code></td>
<td>
<p>Default to &quot;barplotGO.png&quot;, is ignored if from_autoGO is TRUE. The name of the barplot filename.</p>
</td></tr>
<tr><td><code id="barplotGO_+3A_from_autogo">from_autoGO</code></td>
<td>
<p>Default is TRUE, set to FALSE if the enrichment tables you want to use are not from a differential expression analysis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. Files will be produced as part of normal execution.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
barplotGO(
  enrich_tables = enrich_tables,
  title = NULL,
  outfolder = NULL,
  outfile = NULL,
  from_autoGO = TRUE
)

## End(Not run)
</code></pre>

<hr>
<h2 id='choose_database'>choose_database</h2><span id='topic+choose_database'></span>

<h3>Description</h3>

<p>It allows the user to choose the databases on which to perform the enrichment analysis. Either it returns all the possible databases or a subset of them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>choose_database(db_search = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="choose_database_+3A_db_search">db_search</code></td>
<td>
<p>(Default = NULL), is the string pattern to be matched against the list of enrichR databases. Any matching DBs will be returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of database names as a character vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
choose_database(db_search = "KEGG")

## End(Not run)
</code></pre>

<hr>
<h2 id='comparisons'>Example comparison data.</h2><span id='topic+comparisons'></span>

<h3>Description</h3>

<p>It is the comparison table needed by DESeq2, constituted by the
column <em>treatment</em> and by the column <em>control</em>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>comparisons
</code></pre>


<h3>Format</h3>

<p>A data frame with 6 rows and 2 variable:
</p>

<hr>
<h2 id='counts'>Example raw count data.</h2><span id='topic+counts'></span>

<h3>Description</h3>

<p>Genes as rows and samples as columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>counts
</code></pre>


<h3>Format</h3>

<p>A data frame with 29744 rows and 19 variable:
</p>

<hr>
<h2 id='deseq_analysis'>Differential Gene Expression Analysis</h2><span id='topic+deseq_analysis'></span>

<h3>Description</h3>

<p>This function allows to perform a differential gene expression analysis using the DESeq2 package.
</p>
<p>The principal DESeq2 workflow is employed. Raw counts are rounded, if it is necessary, because integers are needed for DESeq2 to run. In case the user provides an .rds file, the tool makes sure that in the assay of the SummarizedExperiment only counts are stored. Then the function DESeqDataSetFromMatrix (or DESeqDataSet for .rds) is employed. A prefiltering is applied in order to remove all genes having sum along the subjects less than 10. The Differential Expression Analysis is performed by employing the function DE and the normalized data matrix is stored in the working directory. This will be useful for further analysis and visualizations. Results are extracted for each comparison of interest. Subfolders will be generated in the &quot;outfolder&quot; (default: ./results) with the name of the comparisons made. NOTE: as standard we use the nomenclature &quot;CONTROL_vs_TREATMENT&quot;, i.e. the control is on the left. Inside each comparison subfolder we will find a .tsv file with the complete differential analysis and other subfolder based on the pvalue and the log2FC thresholds; inside this we will find a .tsv file with the results of the only filtered genes. This subfolders will be divided in other subfolders as &quot;up_genes&quot;, &quot;down_genes&quot; and &quot;up_down_genes&quot;. Look the path flow chart at the end of this tutorial (Figure 1).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deseq_analysis(
  counts,
  groups,
  comparisons,
  padj_threshold = 0.05,
  log2FC_threshold = 0,
  pre_filtering = TRUE,
  save_excel = FALSE,
  outfolder = "./results",
  del_csv = ","
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="deseq_analysis_+3A_counts">counts</code></td>
<td>
<p>The path to raw counts file. Accepted file formats are tab or comma-separated files (.tsv, .csv), .txt files, .rds. Genes must be on rows, samples on columns.</p>
</td></tr>
<tr><td><code id="deseq_analysis_+3A_groups">groups</code></td>
<td>
<p>Sample information table needed by DESeq2 (e.g. 'colData'). A data frame with at least two columns: one for samples, one for a grouping variable (See examples).</p>
</td></tr>
<tr><td><code id="deseq_analysis_+3A_comparisons">comparisons</code></td>
<td>
<p>Table of comparisons based on the grouping variable in 'groups' table (See examples). It should be a data.frame with column 'treatment' and column 'control'. It is possible to provide the path to a .txt file.</p>
</td></tr>
<tr><td><code id="deseq_analysis_+3A_padj_threshold">padj_threshold</code></td>
<td>
<p>Threshold value for adjusted p-value significance (Defaults to 0.05).</p>
</td></tr>
<tr><td><code id="deseq_analysis_+3A_log2fc_threshold">log2FC_threshold</code></td>
<td>
<p>Threshold value for log2(Fold Change) for considering genes as differentially expressed.</p>
</td></tr>
<tr><td><code id="deseq_analysis_+3A_pre_filtering">pre_filtering</code></td>
<td>
<p>Removes genes which sum in the raw counts is less than 10 (Default = TRUE).</p>
</td></tr>
<tr><td><code id="deseq_analysis_+3A_save_excel">save_excel</code></td>
<td>
<p>Allows to save all the output tables in .xlsx format (Default = FALSE).</p>
</td></tr>
<tr><td><code id="deseq_analysis_+3A_outfolder">outfolder</code></td>
<td>
<p>The name to assign to the folder for output saving. (Default = &quot;./results&quot;).</p>
</td></tr>
<tr><td><code id="deseq_analysis_+3A_del_csv">del_csv</code></td>
<td>
<p>Specify the delimiter of the .csv file, default is &quot;,&quot;. This is because opening .csv files with Excel messes up the format and changes the delimiter in &quot;;&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. Files will be produced as part of normal execution.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sample &lt;- c("Pat_1", "Pat_2", "Pat_3", "Pat_4", "Pat_5", "Pat_6")
group &lt;- c("CTRL", "CTRL", "TREAT_A", "TREAT_A", "TREAT_B", "TREAT_B")
groups &lt;- data.frame(sample, group)
treatment &lt;- c("TREAT_A", "TREAT_B", "TREAT_A")
control &lt;- c("CTRL", "CTRL", "TREAT_B")
comparisons &lt;- data.frame(treatment, control)
## Not run: 
deseq_analysis(counts,
  groups,
  comparisons,
  padj_threshold = 0.05,
  log2FC_threshold = 0,
  pre_filtering = T,
  save_excel = F,
  outfolder = "./results",
  del_csv = ","
)

## End(Not run)
</code></pre>

<hr>
<h2 id='filtering_DE'>Filtering DESeq2 results</h2><span id='topic+filtering_DE'></span>

<h3>Description</h3>

<p>We could be in a position to carry out multiple filters on the results of the differential analysis, in order not to repeat all the deseq_analysis.R code which provides for the actual computation of the differential analysis, the filtering_DE.R function has been implemented to be able to filter the file(s) &quot; * _allres.tsv &quot; and generate all the folders and files associated with the specific filters applied.
</p>
<p>The function automatically searches inside the folders where_results and outfolder the file(s) (See ?deseq_analysis()) &quot;_allres.tsv&quot; and generates folders and files in the same folders with the new filters for foldchange and pvalue respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filtering_DE(
  padj_threshold = 0.05,
  log2FC_threshold = 1,
  outfolder = "./results",
  save_excel = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filtering_DE_+3A_padj_threshold">padj_threshold</code></td>
<td>
<p>(Default = 0.05) Threshold value for adjusted p-value filtering.</p>
</td></tr>
<tr><td><code id="filtering_DE_+3A_log2fc_threshold">log2FC_threshold</code></td>
<td>
<p>(Default = 0) Threshold value for log2(Fold Change) filtering.</p>
</td></tr>
<tr><td><code id="filtering_DE_+3A_outfolder">outfolder</code></td>
<td>
<p>(Default = &quot;./results&quot;) Name of the folder in which the new output is written.</p>
</td></tr>
<tr><td><code id="filtering_DE_+3A_save_excel">save_excel</code></td>
<td>
<p>(Default = FALSE) Write output in MS Excel file format (.xlsx).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. Files will be produced as part of normal execution.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
filtering_DE(
  padj_threshold = 0.05,
  log2FC_threshold = 1,
  outfolder = "./results",
  save_excel = F
)

## End(Not run)
</code></pre>

<hr>
<h2 id='groups'>Example group data.</h2><span id='topic+groups'></span>

<h3>Description</h3>

<p>It is the sample information table needed by DESeq2, constituted by the
column <em>sample</em>, including sample barcodes and by the column <em>group</em>
including the condition to which each sample belongs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>groups
</code></pre>


<h3>Format</h3>

<p>A data frame with 18 rows and 2 variable:
</p>

<hr>
<h2 id='heatmapGO'>HeatmapGO</h2><span id='topic+heatmapGO'></span>

<h3>Description</h3>

<p>If the analysis has been performed on more conditions it is interest to have a look at the difference in the enrichment results between the groups. This can be performed by heatmapGO().
</p>
<p>The function automatically reads all the enrichment results of the chosen database. A heatmap is produced for each database, all the terms are merged together and a filter is applied as follows: only terms with a significant pvalue (i.e. less than padj_threshold) in at least one comparison will be retained and plotted. These plots will be saved in the &quot;Comparison_Heatmap&quot; folder. In order to have readable plots, if many terms are enriched for a database several images will be created (indexed _1, _2, ...).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmapGO(
  db,
  outfolder = "./results",
  log2FC_threshold = 0,
  padj_threshold = 0.05,
  min_term_per_row = 2,
  which_list = c("up_genes", "down_genes", "up_down_genes", "not_from_DE")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heatmapGO_+3A_db">db</code></td>
<td>
<p>Database of choice to plot the heatmap. It has to be one for which the enrichment analysis has been performed.</p>
</td></tr>
<tr><td><code id="heatmapGO_+3A_outfolder">outfolder</code></td>
<td>
<p>The name to assign to the folder for output saving. (Default = &quot;./results&quot;). NOTE: please add &quot;/&quot; at the end.</p>
</td></tr>
<tr><td><code id="heatmapGO_+3A_log2fc_threshold">log2FC_threshold</code></td>
<td>
<p>Threshold value for log2(Fold Change) for considering genes as differentially expressed (Default = 0).</p>
</td></tr>
<tr><td><code id="heatmapGO_+3A_padj_threshold">padj_threshold</code></td>
<td>
<p>Threshold value for adjusted p-value significance (Defaults to 0.05).</p>
</td></tr>
<tr><td><code id="heatmapGO_+3A_min_term_per_row">min_term_per_row</code></td>
<td>
<p>Minimum of comparisons or enriched lists on which a certain term must be significant (Defaults to 2).</p>
</td></tr>
<tr><td><code id="heatmapGO_+3A_which_list">which_list</code></td>
<td>
<p>One of c(&quot;up_genes&quot;, &quot;down_genes&quot;,&quot;up_down_genes&quot;, &quot;not_from_DE&quot;): select data to plot. Respectively, only up regulated genes (up_genes), only down regulated genes (&quot;down_genes&quot;), enrichment on both up and down regulated genes (up_down_genes) or select &quot;not_from_DE&quot; if the enrichment will be made on a list of genes that does not come from a differential expression analysis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. Files will be produced as part of normal execution.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
heatmapGO(
  db = "GO_Biological_Process_2021",
  outfolder = "./results",
  log2FC_threshold = 0,
  padj_threshold = 0.05,
  min_term_per_row = 3,
  which_list = "down_genes"
)

## End(Not run)
</code></pre>

<hr>
<h2 id='lolliGO'>lolliGO</h2><span id='topic+lolliGO'></span>

<h3>Description</h3>

<p>The function lolliGO.R implement the lollipop plot of the first 20 enrichment terms.
</p>
<p>For each enrichment result table a lollipop plot is produced. Results are stored in the &quot;enrichment_plots&quot; subfolder for each comparison.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lolliGO(
  enrich_tables,
  title = NULL,
  outfolder = NULL,
  outfile = "lolliGO.png",
  from_autoGO = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lolliGO_+3A_enrich_tables">enrich_tables</code></td>
<td>
<p>Dataframe containing the enrichment results or a path to your .tsv file containing the enrichment results. Columns 'Term' and 'Adjusted.P.Value' are required.</p>
</td></tr>
<tr><td><code id="lolliGO_+3A_title">title</code></td>
<td>
<p>Default to NULL, only specify if from_autoGO is FALSE. When enrich_tables is not from autoGO and thus from read_enrich_tables, the user can specify title and subtitle of the plot as 2-element character vector, for example: c(&quot;This is the title&quot;, &quot;this is the subtitle&quot;)</p>
</td></tr>
<tr><td><code id="lolliGO_+3A_outfolder">outfolder</code></td>
<td>
<p>Default to NULL, only specify if from_autoGO is FALSE. The name to assign to the folder for output saving.</p>
</td></tr>
<tr><td><code id="lolliGO_+3A_outfile">outfile</code></td>
<td>
<p>Default to &quot;lolliGO.png&quot;, is ignored if from_autoGO is TRUE. The name of the lolli filename.</p>
</td></tr>
<tr><td><code id="lolliGO_+3A_from_autogo">from_autoGO</code></td>
<td>
<p>Default is TRUE, set to FALSE if the enrichment tables you want to use are not from a differential expression analysis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. Files will be produced as part of normal execution.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
lolliGO(
  enrich_tables = enrich_tables,
  title = NULL,
  outfolder = NULL,
  outfile = NULL,
  from_autoGO = TRUE
)

## End(Not run)
</code></pre>

<hr>
<h2 id='read_enrich_tables'>Read enrichment results from tables</h2><span id='topic+read_enrich_tables'></span>

<h3>Description</h3>

<p>Helper function to read all the enrichment results in order to proceed with the visualization in an automated way.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_enrich_tables(
  enrich_table_path = "./results",
  log2FC_threshold = 0,
  padj_threshold = 0.05,
  which_list = c("up_genes", "down_genes", "up_down_genes", "everything"),
  from_autoGO = TRUE,
  files_format = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_enrich_tables_+3A_enrich_table_path">enrich_table_path</code></td>
<td>
<p>Specify the full path to the folder where enrichment tables have to read from (all fitting files in any subdirectory will be loaded).</p>
</td></tr>
<tr><td><code id="read_enrich_tables_+3A_log2fc_threshold">log2FC_threshold</code></td>
<td>
<p>Threshold value for log2(Fold Change) for considering genes as differentially expressed (default = 0).</p>
</td></tr>
<tr><td><code id="read_enrich_tables_+3A_padj_threshold">padj_threshold</code></td>
<td>
<p>Threshold value for adjusted p-value significance (Defaults to 0.05).</p>
</td></tr>
<tr><td><code id="read_enrich_tables_+3A_which_list">which_list</code></td>
<td>
<p>It can be: &quot;up_genes&quot;,&quot;down_genes&quot;,&quot;up_down_genes&quot;,&quot;everything&quot;. Select a list of genes to perform the enrichment. Respectively, only up regulated genes (up_genes), only down regulated genes (down_genes), both up and down regulated genes (up_down_genes), or (everything) allow to load all the three kind of lists separately and it is employed also for lists not from differential analysis.</p>
</td></tr>
<tr><td><code id="read_enrich_tables_+3A_from_autogo">from_autoGO</code></td>
<td>
<p>Default is TRUE, set to FALSE if the lists you want to upload are not from a differential expression analysis.</p>
</td></tr>
<tr><td><code id="read_enrich_tables_+3A_files_format">files_format</code></td>
<td>
<p>Default is NULL, when from_autoGO = FALSE it is mandatory to provide the extension of the list of genes you want to upload.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of enrichment tables, each one being a tibble object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
enrich_tables &lt;- read_enrich_tables(
  enrich_table_path = "./results",
  log2FC_threshold = 0,
  padj_threshold = 0.05,
  which_list = "down_genes",
  from_autoGO = T,
  files_format = NULL
)

## End(Not run)
</code></pre>

<hr>
<h2 id='read_gene_lists'>Read all the genes lists</h2><span id='topic+read_gene_lists'></span>

<h3>Description</h3>

<p>Function employed for reading all the gene lists on which the enrichment will be performed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_gene_lists(
  gene_lists_path = "./results",
  log2FC_threshold = 0,
  padj_threshold = 0.05,
  which_list = c("up_genes", "down_genes", "up_down_genes", "everything"),
  from_autoGO = TRUE,
  files_format = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_gene_lists_+3A_gene_lists_path">gene_lists_path</code></td>
<td>
<p>Specify the full path to the folder where the gene lists have to read from (all fitting files in any subdirectory will be loaded).</p>
</td></tr>
<tr><td><code id="read_gene_lists_+3A_log2fc_threshold">log2FC_threshold</code></td>
<td>
<p>Threshold value for log2(Fold Change) for considering genes as differentially expressed (default = 0).</p>
</td></tr>
<tr><td><code id="read_gene_lists_+3A_padj_threshold">padj_threshold</code></td>
<td>
<p>Threshold value for adjusted p-value significance (Defaults to 0.05).</p>
</td></tr>
<tr><td><code id="read_gene_lists_+3A_which_list">which_list</code></td>
<td>
<p>It can be: &quot;up_genes&quot;,&quot;down_genes&quot;,&quot;up_down_genes&quot;,&quot;everything&quot;. Select a list of genes to perform the enrichment. Respectively, both up and down regulated genes (up_down_genes), only up regulated genes (up_genes), only down regulated genes (down_genes), or (everything) allow to load all the three kind of lists separately and it is employed also for lists not from differential analysis..</p>
</td></tr>
<tr><td><code id="read_gene_lists_+3A_from_autogo">from_autoGO</code></td>
<td>
<p>Default is TRUE, set FALSE if the gene list you want to upload are not from a differential expression analysis.</p>
</td></tr>
<tr><td><code id="read_gene_lists_+3A_files_format">files_format</code></td>
<td>
<p>(Default = NULL). When from_autoGO = FALSE it is mandatory to provide the extension of the list of genes to upload.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of gene lists, each one being a one-dimensional data.frame.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
gene_lists &lt;- read_gene_lists(
  gene_lists_path = "./results",
  log2FC_threshold = 0,
  padj_threshold = 0.05,
  which_list = "down_genes",
  from_autoGO = T,
  files_format = NULL
)

## End(Not run)
</code></pre>

<hr>
<h2 id='volcanoplot'>Volcano Plot</h2><span id='topic+volcanoplot'></span>

<h3>Description</h3>

<p>Once the differential analysis has been performed, it is possible to visualize the volcano plots employing this function.
</p>
<p>The volcano plot is generated by the employment of ggplot2, setting xlimit and ylimit based on the data. If there are genes with pvalue equal to infinity, those are forced to the maximum value of the pvalue. If 'my_comparison' paramater is not provided (default: NULL), the function will extract the name of the first subfolder inside &quot;./results&quot; and use it. The volcano plots are saved in the a subfolder for each comparison (Figure 1).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>volcanoplot(
  DE_results,
  my_comparison = NULL,
  highlight_genes = NULL,
  log2FC_thresh = 0,
  padj_thresh = 0.05,
  del_csv = ",",
  outfolder = "./results"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="volcanoplot_+3A_de_results">DE_results</code></td>
<td>
<p>DESeq2 results table. Accepts both a file (.tsv, .csv, tab-separated .txt) or a dataframe (see example below). Requires 'log2FoldChange' and 'padj' columns.</p>
</td></tr>
<tr><td><code id="volcanoplot_+3A_my_comparison">my_comparison</code></td>
<td>
<p>The comparison to plot (control_vs_treatment, a_vs_b, ...). If the user is employing the whole workflow use exactly the name of the comparison indicated in the comparison dataframe.</p>
</td></tr>
<tr><td><code id="volcanoplot_+3A_highlight_genes">highlight_genes</code></td>
<td>
<p>A (optional) list of genes the user would like to highlight (label) in the volcano plot. It accepts a dataframe, a character vector or a path to a file in .txt format.</p>
</td></tr>
<tr><td><code id="volcanoplot_+3A_log2fc_thresh">log2FC_thresh</code></td>
<td>
<p>Threshold value for log2(Fold Change) to highlight genes as differentially expressed (default = 0).</p>
</td></tr>
<tr><td><code id="volcanoplot_+3A_padj_thresh">padj_thresh</code></td>
<td>
<p>Threshold value for adjusted p-value to highlight genes as  significant (default = 0.05).</p>
</td></tr>
<tr><td><code id="volcanoplot_+3A_del_csv">del_csv</code></td>
<td>
<p>Specify the delimiter of the .csv file (default = &quot;,&quot;). This is because opening .csv files with Excel messes up the format and changes the delimiter to &quot;;&quot;.</p>
</td></tr>
<tr><td><code id="volcanoplot_+3A_outfolder">outfolder</code></td>
<td>
<p>The name to assign to the folder for output saving. (Default = &quot;./results&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. Files will be produced as part of normal execution.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
filename &lt;- "./results/H460.2D_vs_H460.3D.2p/DE_H460.2D_vs_H460.3D.2p_allres.tsv"
volcanoplot(
  DE_results = filename,
  my_comparison = "H460.2D_vs_H460.3D.2p",
  log2FC_thresh = 0,
  padj_thresh = 0.05,
  highlight_genes = NULL,
  del_csv = ",",
  outfolder = "./results"
)

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
