<!DOCTYPE html><html><head><title>Help for package AutoPipe</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {AutoPipe}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AutoPipe_tSNE'><p>Implemented t-distributed stochastic neighbor embedding</p></a></li>
<li><a href='#cons_clust'><p>A function to plot do a Consensus clustering to validate the results</p></a></li>
<li><a href='#Groups_Sup'><p>cluster the samples</p></a></li>
<li><a href='#read_expression_file'><p>Input Expression File</p></a></li>
<li><a href='#rna'><p>rna egene expression of 48 meningiomas</p></a></li>
<li><a href='#Supervised_Cluster_Heatmap'><p>Produce a Heatmap using a Supervised clustering Algorithm</p></a></li>
<li><a href='#top_supervised'><p>A Function for  Assisting Supervised Clustering</p></a></li>
<li><a href='#TopPAM'><p>Compute Top genes</p></a></li>
<li><a href='#UnSuperClassifier'><p>Unsupervised Clustering</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Automated Transcriptome Classifier Pipeline: Comprehensive
Transcriptome Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.6</td>
</tr>
<tr>
<td>Author:</td>
<td>Karam Daka [cre, aut],
  Dieter Henrik Heiland [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Karam Daka &lt;k.dacca@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>An unsupervised fully-automated pipeline for transcriptome analysis or a supervised option to identify characteristic genes from predefined subclasses.
              We rely on the 'pamr' <a href="http://www.bioconductor.org/packages//2.7/bioc/html/pamr.html">http://www.bioconductor.org/packages//2.7/bioc/html/pamr.html</a> clustering algorithm to cluster the Data and then draw a heatmap of the clusters with the most significant genes and the 
              least significant genes according to the 'pamr' algorithm. This way we get easy to grasp heatmaps that show us for each cluster which are the clusters most defining genes.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>cluster ,pamr ,siggenes ,annotate ,fgsea ,org.Hs.eg.db
,RColorBrewer ,ConsensusClusterPlus ,Rtsne ,clusterProfiler
,msigdbr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-02-18 09:38:09 UTC; Karam</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-02-27 17:00:36 UTC</td>
</tr>
</table>
<hr>
<h2 id='AutoPipe_tSNE'>Implemented t-distributed stochastic neighbor embedding</h2><span id='topic+AutoPipe_tSNE'></span>

<h3>Description</h3>

<p>This function is used to upload a table into R for further use in the AutoPipe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AutoPipe_tSNE(me,perplexity=30,max_iter=500,groups_men)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AutoPipe_tSNE_+3A_me">me</code></td>
<td>
<p>The path of the expression table</p>
</td></tr>
<tr><td><code id="AutoPipe_tSNE_+3A_perplexity">perplexity</code></td>
<td>
<p>numeric; Perplexity parameter</p>
</td></tr>
<tr><td><code id="AutoPipe_tSNE_+3A_max_iter">max_iter</code></td>
<td>
<p>integer; Number of iterations (default: 1000)</p>
</td></tr>
<tr><td><code id="AutoPipe_tSNE_+3A_groups_men">groups_men</code></td>
<td>
<p>the data frame with the group clustering that the function Groups_Sup or top_supervised (2. place on the list) returns with
the data about each sample and its coressponding cluster.</p>
</td></tr>
</table>

<hr>
<h2 id='cons_clust'>A function to plot do a Consensus clustering to validate the results</h2><span id='topic+cons_clust'></span>

<h3>Description</h3>

<p>this function  calls the ConsensusClusterPlus function with thedaraset and plots a plot
with the heatmaps of the clustering for each number of clusters from 2 to max_clust
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cons_clust(data,max_clust,TOPgenes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cons_clust_+3A_data">data</code></td>
<td>
<p>this is the data for the ConsensusClusterPlus</p>
</td></tr>
<tr><td><code id="cons_clust_+3A_max_clust">max_clust</code></td>
<td>
<p>the max number of clusters that should be evaluated.</p>
</td></tr>
<tr><td><code id="cons_clust_+3A_topgenes">TOPgenes</code></td>
<td>
<p>the number of the top genes to choose for the clustering</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plots a plot with all the heatmaps from the ConsensusClusterPlus for the number ofd clusters 2 to max_clust
the same return value as the COnsensusClusterPlus
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(rna)
cons_clust(rna,5,TOPgenes=50)

</code></pre>

<hr>
<h2 id='Groups_Sup'>cluster the samples</h2><span id='topic+Groups_Sup'></span>

<h3>Description</h3>

<p>This function clusters the samples into x clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Groups_Sup(me_TOP, me, number_of_k,TRw)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Groups_Sup_+3A_me_top">me_TOP</code></td>
<td>
<p>the matrix with the n top genes, usually the  from
output of the function TopPAM</p>
</td></tr>
<tr><td><code id="Groups_Sup_+3A_me">me</code></td>
<td>
<p>the original expression matrix. (with genes in rows and samples in columns).</p>
</td></tr>
<tr><td><code id="Groups_Sup_+3A_number_of_k">number_of_k</code></td>
<td>
<p>the number of clusters</p>
</td></tr>
<tr><td><code id="Groups_Sup_+3A_trw">TRw</code></td>
<td>
<p>threshold for the elemenation of the samples with a Silhouette width lower than TRw.
Default value is -1.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
## load data
library(org.Hs.eg.db)
data(rna)
me_x=rna
res&lt;-AutoPipe::TopPAM(me_x,max_clusters = 8, TOP=100)
me_TOP=res[[1]]
number_of_k=res[[3]]
File_genes=Groups_Sup(me_TOP, me=me_x, number_of_k,TRw=-1)
groups_men=File_genes[[2]]
me_x=File_genes[[1]]

</code></pre>

<hr>
<h2 id='read_expression_file'>Input Expression File</h2><span id='topic+read_expression_file'></span>

<h3>Description</h3>

<p>This function is used to upload a table into R for further use in the AutoPipe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_expression_file(file, format = "csv", sep=";",gene_name="SYMBOL", Trans=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_expression_file_+3A_file">file</code></td>
<td>
<p>The path of the expression table</p>
</td></tr>
<tr><td><code id="read_expression_file_+3A_format">format</code></td>
<td>
<p>The format of the table &quot;csv&quot; or &quot;txt&quot;</p>
</td></tr>
<tr><td><code id="read_expression_file_+3A_sep">sep</code></td>
<td>
<p>The seperator of the input table</p>
</td></tr>
<tr><td><code id="read_expression_file_+3A_gene_name">gene_name</code></td>
<td>
<p>Genes are given in &quot;SYMBOL&quot; or &quot;ENTREZID&quot;</p>
</td></tr>
<tr><td><code id="read_expression_file_+3A_trans">Trans</code></td>
<td>
<p>Need Matrix Transpose TRUE or FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with a gene expression matrix
</p>

<hr>
<h2 id='rna'>rna egene expression of 48 meningiomas</h2><span id='topic+rna'></span>

<h3>Description</h3>

<p>A dataset containing the gene expression data od 48 meningioma tumors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rna
</code></pre>


<h3>Format</h3>

<p>A data frame with 200 rows and 48 variables:
</p>

<dl>
<dt>BT_1008</dt><dd><p>sample BT_1008, </p>
</dd>
<dt>BT_1017</dt><dd><p>sample BT_1017, </p>
</dd>
<dt>BT_1025</dt><dd><p>sample BT_1025, </p>
</dd>
<dt>BT_1042</dt><dd><p>sample BT_1042, </p>
</dd>
<dt>BT_1050</dt><dd><p>sample BT_1050, </p>
</dd>
<dt>BT_1056</dt><dd><p>sample BT_1056, </p>
</dd>
<dt>BT_1065</dt><dd><p>sample BT_1065, </p>
</dd>
<dt>BT_1067</dt><dd><p>sample BT_1067, </p>
</dd>
<dt>BT_1072</dt><dd><p>sample BT_1072, </p>
</dd>
<dt>BT_1078</dt><dd><p>sample BT_1078, </p>
</dd>
<dt>BT_1082</dt><dd><p>sample BT_1082, </p>
</dd>
<dt>BT_1091</dt><dd><p>sample BT_1091, </p>
</dd>
<dt>BT_1094</dt><dd><p>sample BT_1094, </p>
</dd>
<dt>BT_1097</dt><dd><p>sample BT_1097, </p>
</dd>
<dt>BT_1115</dt><dd><p>sample BT_1115, </p>
</dd>
<dt>BT_605</dt><dd><p>sample BT_605, </p>
</dd>
<dt>BT_617</dt><dd><p>sample BT_617, </p>
</dd>
<dt>BT_619</dt><dd><p>sample BT_619, </p>
</dd>
<dt>BT_633</dt><dd><p>sample BT_633, </p>
</dd>
<dt>BT_634</dt><dd><p>sample BT_634, </p>
</dd>
<dt>BT_644</dt><dd><p>sample BT_644, </p>
</dd>
<dt>BT_654</dt><dd><p>sample BT_654, </p>
</dd>
<dt>BT_659</dt><dd><p>sample BT_659, </p>
</dd>
<dt>BT_690</dt><dd><p>sample BT_690, </p>
</dd>
<dt>BT_695</dt><dd><p>sample BT_695, </p>
</dd>
<dt>BT_700</dt><dd><p>sample BT_700, </p>
</dd>
<dt>BT_738</dt><dd><p>sample BT_738, </p>
</dd>
<dt>BT_751</dt><dd><p>sample BT_751, </p>
</dd>
<dt>BT_771</dt><dd><p>sample BT_771, </p>
</dd>
<dt>BT_797</dt><dd><p>sample BT_797, </p>
</dd>
<dt>BT_803</dt><dd><p>sample BT_803, </p>
</dd>
<dt>BT_808</dt><dd><p>sample BT_808, </p>
</dd>
<dt>BT_820</dt><dd><p>sample BT_820, </p>
</dd>
<dt>BT_837</dt><dd><p>sample BT_837, </p>
</dd>
<dt>BT_855</dt><dd><p>sample BT_855, </p>
</dd>
<dt>BT_862</dt><dd><p>sample BT_862, </p>
</dd>
<dt>BT_873</dt><dd><p>sample BT_873, </p>
</dd>
<dt>BT_882</dt><dd><p>sample BT_882, </p>
</dd>
<dt>BT_887</dt><dd><p>sample BT_887, </p>
</dd>
<dt>BT_900</dt><dd><p>sample BT_900, </p>
</dd>
<dt>BT_905</dt><dd><p>sample BT_905, </p>
</dd>
<dt>BT_907</dt><dd><p>sample BT_907, </p>
</dd>
<dt>BT_920</dt><dd><p>sample BT_920, </p>
</dd>
<dt>BT_944</dt><dd><p>sample BT_944, </p>
</dd>
<dt>BT_962</dt><dd><p>sample BT_962, </p>
</dd>
<dt>BT_963</dt><dd><p>sample BT_963, </p>
</dd>
<dt>BT_982</dt><dd><p>sample BT_982, </p>
</dd>
<dt>BT_990</dt><dd><p>sample BT_990, </p>
</dd>
</dl>
<p>...
</p>

<hr>
<h2 id='Supervised_Cluster_Heatmap'>Produce a Heatmap using a Supervised clustering Algorithm</h2><span id='topic+Supervised_Cluster_Heatmap'></span>

<h3>Description</h3>

<p>This function produces a plot with a Heatmap using
a supervised clustering algorithm which the user choses.
with a the  mean Silhouette width plotted on the right top corner
and the Silhouette width for each sample on top.
On the right side of the plot the n highest and lowest scoring
genes for each cluster will added. And next to them the coressponding pathways
(see Details)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Supervised_Cluster_Heatmap(groups_men, gene_matrix,
method="PAMR",TOP=1000,TOP_Cluster=150,
show_sil=FALSE,show_clin=FALSE,genes_to_print=5,
print_genes=FALSE,samples_data=NULL,colors="RdBu",
GSE=FALSE,topPaths=5,db="c2",plot_mean_sil=FALSE,stats_clust =NULL,threshold=2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Supervised_Cluster_Heatmap_+3A_groups_men">groups_men</code></td>
<td>
<p>the data frame with the group clustering that the function Groups_Sup or top_supervised (2. place on the list) returns with
the data about each sample and its coressponding cluster.</p>
</td></tr>
<tr><td><code id="Supervised_Cluster_Heatmap_+3A_gene_matrix">gene_matrix</code></td>
<td>
<p>the matrix of n selected genes that the function Groups_Sup returns</p>
</td></tr>
<tr><td><code id="Supervised_Cluster_Heatmap_+3A_method">method</code></td>
<td>
<p>the method to cluster of Clustering. The default is &quot;PAMR&quot; which uses the pamr library.
other methods are SAM and our own &quot;EXReg&quot; (see details)</p>
</td></tr>
<tr><td><code id="Supervised_Cluster_Heatmap_+3A_top">TOP</code></td>
<td>
<p>the number of the top genes to take. the default value is 1000.</p>
</td></tr>
<tr><td><code id="Supervised_Cluster_Heatmap_+3A_top_cluster">TOP_Cluster</code></td>
<td>
<p>a numeric variable for the number of genes to include in the clusters. Default is 150.</p>
</td></tr>
<tr><td><code id="Supervised_Cluster_Heatmap_+3A_show_sil">show_sil</code></td>
<td>
<p>a logical value that indicates if the function should show
the Silhouette width for each sample. Default is FALSE.</p>
</td></tr>
<tr><td><code id="Supervised_Cluster_Heatmap_+3A_show_clin">show_clin</code></td>
<td>
<p>a logical value if TRUE the function will plot the clinical data provided
by the user. Default value is FALSE.</p>
</td></tr>
<tr><td><code id="Supervised_Cluster_Heatmap_+3A_genes_to_print">genes_to_print</code></td>
<td>
<p>the number of genes to print for each cluster. this function
adds on the right side.
of the heatmap the n highest expressed genes and the n lowest expressed genes for each
cluster. Default value is 5.</p>
</td></tr>
<tr><td><code id="Supervised_Cluster_Heatmap_+3A_print_genes">print_genes</code></td>
<td>
<p>a logical value indicating if or not to plot the TOP genes for each
cluster.Default value is FALSE.</p>
</td></tr>
<tr><td><code id="Supervised_Cluster_Heatmap_+3A_samples_data">samples_data</code></td>
<td>
<p>the clinical data provided by the user to plot under the heatmap. it will be
plotted only if show_clin is TRUE. Default value is NULL. see details for format.</p>
</td></tr>
<tr><td><code id="Supervised_Cluster_Heatmap_+3A_colors">colors</code></td>
<td>
<p>the colors for the Heatmap. The function RColorBrewer palletes.</p>
</td></tr>
<tr><td><code id="Supervised_Cluster_Heatmap_+3A_gse">GSE</code></td>
<td>
<p>a logical variable that indicates wether to  plot thr Gene Set Enrichment Analysis
next to the heatmap. Default value is FALSE.</p>
</td></tr>
<tr><td><code id="Supervised_Cluster_Heatmap_+3A_toppaths">topPaths</code></td>
<td>
<p>a numerical value that says how many pathways the Gene Set Enrichment
plots should contain fo each cluster. Default value is 5.</p>
</td></tr>
<tr><td><code id="Supervised_Cluster_Heatmap_+3A_db">db</code></td>
<td>
<p>a value for the database for the GSE to be used. Default value is &quot;c1&quot;.
the paramater can one of the values: &quot;c1&quot;,&quot;c2&quot;,&quot;c3&quot;,c4&quot;,&quot;c5&quot;,&quot;c6&quot;,&quot;c7&quot;,&quot;h&quot;. See the
broad institue GSE <a href="http://software.broadinstitute.org/gsea/index.jsp">GSE webpage</a> for further information in each dataset.</p>
</td></tr>
<tr><td><code id="Supervised_Cluster_Heatmap_+3A_plot_mean_sil">plot_mean_sil</code></td>
<td>
<p>A logical value. if TRUE the function plots the mean of the
Silhouette width for each cluster number or gap statistic.</p>
</td></tr>
<tr><td><code id="Supervised_Cluster_Heatmap_+3A_stats_clust">stats_clust</code></td>
<td>
<p>A vector with the mean Silhouette widths or gap statistic for the number of clusters. The first
value should be for 2 Clusters. 2nd is for 3 clusters and so on.</p>
</td></tr>
<tr><td><code id="Supervised_Cluster_Heatmap_+3A_threshold">threshold</code></td>
<td>
<p>the threshhold for the pam analysis default is 2.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>sample data should be a data.frame with the sample names
as rownames and the clinical triats as columns.
each trait must be a numeric variable.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load the org.Hs.eg Library
library(org.Hs.eg.db)
## load data
data(rna)
me_x=rna
## calculate best number of clusters and
res&lt;-AutoPipe::TopPAM(me_x,max_clusters = 6, TOP=100)
me_TOP=res[[1]]
number_of_k=res[[3]]
File_genes=Groups_Sup(me_TOP, me=me_x, number_of_k,TRw=-1)
groups_men=File_genes[[2]]
me_x=File_genes[[1]]
o_g&lt;-Supervised_Cluster_Heatmap(groups_men = groups_men, gene_matrix=me_x,
    method="PAMR",show_sil=TRUE,print_genes=TRUE,threshold=0,
    TOP = 100,GSE=FALSE,plot_mean_sil=TRUE,stats_clust=res[[2]])

</code></pre>

<hr>
<h2 id='top_supervised'>A Function for  Assisting Supervised Clustering</h2><span id='topic+top_supervised'></span>

<h3>Description</h3>

<p>when perfoming a supervised clustering the user should run this function in order to get the best results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>top_supervised(me,TOP=1000,cluster_which,TRw=-1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="top_supervised_+3A_me">me</code></td>
<td>
<p>the matrix of the gene exporessions, the olums should be the samples  and the colnames the sample names
the rownames should be the genes . at best the ENTEREZID</p>
</td></tr>
<tr><td><code id="top_supervised_+3A_top">TOP</code></td>
<td>
<p>the top genes to choose, default is 100.</p>
</td></tr>
<tr><td><code id="top_supervised_+3A_cluster_which">cluster_which</code></td>
<td>
<p>a dataframe with the supervised clustering arrangment of the samples. the dataframe should have the
sample names in the first column and the clustering in the secound column.</p>
</td></tr>
<tr><td><code id="top_supervised_+3A_trw">TRw</code></td>
<td>
<p>the threshhold for excluding samples with silhouette width &lt; TRw</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list. the first place is the expression matrix, the secound is the silhouette for each sample.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(org.Hs.eg.db)
data(rna)
cluster_which&lt;-cbind(colnames(rna),c(rep(1,times=24),rep(2,times=24)))
me_x=rna
## calculate best number of clusters and
res&lt;-top_supervised(me_x,TOP = 100,cluster_which)
me_TOP=res[[1]]
number_of_k=2
groups_men=res[[2]]
me_x=me_TOP
colnames(me_x)
o_g&lt;-Supervised_Cluster_Heatmap(groups_men = groups_men, gene_matrix=me_x,
                               method="PAMR",show_sil=TRUE,print_genes=TRUE,threshold = 0,
                               TOP = 100,GSE=FALSE,plot_mean_sil=FALSE,stats_clust=res[[2]],
                               samples_data = as.data.frame(groups_men[,1,drop=FALSE]))
                               
</code></pre>

<hr>
<h2 id='TopPAM'>Compute Top genes</h2><span id='topic+TopPAM'></span>

<h3>Description</h3>

<p>This function computes the n=TOP genes and the the best number of clusters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TopPAM(me, max_clusters=15,TOP=1000,B=100,clusterboot=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TopPAM_+3A_me">me</code></td>
<td>
<p>a matrix with genes in rows and samples in columns</p>
</td></tr>
<tr><td><code id="TopPAM_+3A_max_clusters">max_clusters</code></td>
<td>
<p>max. number of clusters to check</p>
</td></tr>
<tr><td><code id="TopPAM_+3A_top">TOP</code></td>
<td>
<p>the number of genes to take.</p>
</td></tr>
<tr><td><code id="TopPAM_+3A_b">B</code></td>
<td>
<p>integer, number of Monte Carlo (“bootstrap”) samples.</p>
</td></tr>
<tr><td><code id="TopPAM_+3A_clusterboot">clusterboot</code></td>
<td>
<p>A logical value indicating wether or not to calculate the Gap statistic and to bootstrap.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>we use the clusGap algorithm from the package cluster to calculate the Gap statistic.
</p>


<h3>Value</h3>

<p>a list of 1. A matrix with the top genes
2. A list of means of the Silhouette width for each number of clusters. 3. The optimal number of clusters. 4. gap_st the gap statistic of the clustering
5. best number of clusters according to the gap statistic.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load the org.Hs.eg Library
library(org.Hs.eg.db)
#' ## load data
data(rna)
me_x=rna
res&lt;-AutoPipe::TopPAM(me_x,max_clusters = 8, TOP=100,clusterboot=FALSE)
me_TOP=res[[1]]
number_of_k=res[[3]]

</code></pre>

<hr>
<h2 id='UnSuperClassifier'>Unsupervised Clustering</h2><span id='topic+UnSuperClassifier'></span>

<h3>Description</h3>

<p>A function for unsupervised Clustering of the data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UnSuperClassifier(data,clinical_data=NULL,thr=2,TOP_Cluster=150,TOP=100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="UnSuperClassifier_+3A_data">data</code></td>
<td>
<p>the data for the clustering. Data should be in the following format: samples in columns and
the genes in the rows (colnames and rownames accordingly). The rownames should be Entrez ID in order to
plot a gene set enrichment analysis.</p>
</td></tr>
<tr><td><code id="UnSuperClassifier_+3A_clinical_data">clinical_data</code></td>
<td>
<p>the clinical data provided by the user to plot under the heatmap. it will be
plotted only if show_clin is TRUE. Default value is NULL. see details for format.</p>
</td></tr>
<tr><td><code id="UnSuperClassifier_+3A_thr">thr</code></td>
<td>
<p>The threshold for the PAMR algorithm default is 2.</p>
</td></tr>
<tr><td><code id="UnSuperClassifier_+3A_top_cluster">TOP_Cluster</code></td>
<td>
<p>numeric; Number of genes in each cluster.</p>
</td></tr>
<tr><td><code id="UnSuperClassifier_+3A_top">TOP</code></td>
<td>
<p>numeric; the number of the TOP genes to take from the gene exoression matrix see TopPAM TOP.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>sample data should be a data.frame with the sample names
as rownames and the clinical triats as columns.
each trait must be a numeric variable.
@return the function is an autated Pipeline for clustering it plot cluster analysis for the geneset
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
