<!DOCTYPE html><html><head><title>Help for package babelgene</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {babelgene}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#orthologs'><p>Retrieve gene orthologs/homologs</p></a></li>
<li><a href='#species'><p>Retrieve the available species</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gene Orthologs for Model Organisms in a Tidy Data Format</td>
</tr>
<tr>
<td>Version:</td>
<td>22.9</td>
</tr>
<tr>
<td>Description:</td>
<td>Genomic analysis of model organisms frequently requires the
    use of databases based on human data or making comparisons to
    patient-derived resources. This requires harmonization of gene names
    into the same gene space. The 'babelgene' R package converts between
    human and non-human gene orthologs/homologs. The package integrates
    orthology assertion predictions sourced from multiple databases as
    compiled by the HGNC Comparison of Orthology Predictions (HCOP)
    (Wright et al. 2005 &lt;<a href="https://doi.org/10.1007%2Fs00335-005-0103-2">doi:10.1007/s00335-005-0103-2</a>&gt;, Eyre et al. 2007
    &lt;<a href="https://doi.org/10.1093%2Fbib%2Fbbl030">doi:10.1093/bib/bbl030</a>&gt;, Seal et al. 2011 &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgkq892">doi:10.1093/nar/gkq892</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://igordot.github.io/babelgene/">https://igordot.github.io/babelgene/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/igordot/babelgene/issues">https://github.com/igordot/babelgene/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4)</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, methods, rlang</td>
</tr>
<tr>
<td>Suggests:</td>
<td>covr, knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-09-28 16:45:01 UTC; id460</td>
</tr>
<tr>
<td>Author:</td>
<td>Igor Dolgalev <a href="https://orcid.org/0000-0003-4451-126X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Igor Dolgalev &lt;igor.dolgalev@nyumc.org&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-09-29 09:40:17 UTC</td>
</tr>
</table>
<hr>
<h2 id='orthologs'>Retrieve gene orthologs/homologs</h2><span id='topic+orthologs'></span>

<h3>Description</h3>

<p>Retrieve gene orthologs/homologs for a set of genes.
Converts between human and non-human analogs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>orthologs(genes, species, human = TRUE, min_support = 3, top = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="orthologs_+3A_genes">genes</code></td>
<td>
<p>A vector of gene symbols or Entrez/Ensembl IDs.</p>
</td></tr>
<tr><td><code id="orthologs_+3A_species">species</code></td>
<td>
<p>Species name, such as <code>Mus musculus</code> or <code>mouse</code> (see <code><a href="#topic+species">species()</a></code> for options).</p>
</td></tr>
<tr><td><code id="orthologs_+3A_human">human</code></td>
<td>
<p>A logical scalar indicating if the input genes are human. If <code>TRUE</code>, the input genes are human. If <code>FALSE</code>, the input genes correspond to the non-human species and the output will be the human equivalents.</p>
</td></tr>
<tr><td><code id="orthologs_+3A_min_support">min_support</code></td>
<td>
<p>Minimum number of supporting source databases. Gene pairs available in this package are supported by 2 to 12 databases (the maximum varies depending on the species).</p>
</td></tr>
<tr><td><code id="orthologs_+3A_top">top</code></td>
<td>
<p>For each gene, output only the match with the highest support level if there are multiple hits.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of gene pairs (human and given species).
</p>


<h3>References</h3>

<p>Wright MW, Eyre TA, Lush MJ, Povey S, Bruford EA. HCOP: the HGNC comparison of orthology predictions search tool. <em>Mamm Genome</em>. 2005 Nov;16(11):827-8. <a href="https://doi.org/10.1007/s00335-005-0103-2">doi:10.1007/s00335-005-0103-2</a>
</p>
<p>Eyre TA, Wright MW, Lush MJ, Bruford EA. HCOP: a searchable database of human orthology predictions. <em>Brief Bioinform</em>. 2007 Jan;8(1):2-5. <a href="https://doi.org/10.1093/bib/bbl030">doi:10.1093/bib/bbl030</a>
</p>
<p>Seal RL, Gordon SM, Lush MJ, Wright MW, Bruford EA. genenames.org: the HGNC resources in 2011. <em>Nucleic Acids Res</em>. 2011 Jan;39:D514-9. <a href="https://doi.org/10.1093/nar/gkq892">doi:10.1093/nar/gkq892</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>orthologs(genes = "TP53", species = "mouse", human = TRUE)
orthologs(genes = c("Ptprc", "Cd34"), species = "mouse", human = FALSE)
</code></pre>

<hr>
<h2 id='species'>Retrieve the available species</h2><span id='topic+species'></span>

<h3>Description</h3>

<p>List the species with available human orthologs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>species(species = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="species_+3A_species">species</code></td>
<td>
<p>Species name, such as <code>Mus musculus</code> or <code>mouse</code>. If specified, will return results for the given species only.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of the available species.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>species()
species("Mus musculus")
species("mouse")
species("rat")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
