<!DOCTYPE html><html><head><title>Help for package babette</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {babette}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#babette-package'><p>babette: A package for Bayesian phylogenetics.</p></a></li>
<li><a href='#bbt_continue'><p>Continue a BEAST2 run</p></a></li>
<li><a href='#bbt_delete_temp_files'><p>Delete all the temporary files created by bbt_run_from_model</p></a></li>
<li><a href='#bbt_run'><p>Run BEAST2</p></a></li>
<li><a href='#bbt_run_from_model'><p>Run BEAST2</p></a></li>
<li><a href='#bbt_self_test'><p>Do a self test to verify babette that works correctly.</p></a></li>
<li><a href='#check_beast2_pkgs'><p>Checks if <code>bbt_run</code> has the 'BEAST2' packages needed to process</p>
its arguments. Will stop if not.</a></li>
<li><a href='#create_test_bbt_run_output'><p>Get an example output of <code>bbt_run</code></p>
or <code>bbt_run_from_model</code>.</a></li>
<li><a href='#create_test_ns_output'><p>Create NS testing output</p></a></li>
<li><a href='#default_params_doc'><p>This function does nothing. It is intended to inherit is parameters'</p>
documentation.</a></li>
<li><a href='#get_alignment_ids_from_xml'><p>Get the alignment IDs from one or more 'BEAST2' XML input files.</p></a></li>
<li><a href='#get_babette_path'><p>Get the full path of a file in the <code>inst/extdata</code> folder</p></a></li>
<li><a href='#get_babette_paths'><p>Get the full paths of files in the <code>inst/extdata</code> folder</p></a></li>
<li><a href='#parse_beast2_output'><p>Process the 'BEAST2' output dependent on 'BEAST2' package specifics</p></a></li>
<li><a href='#parse_beast2_output_to_ns'><p>Parse BEAST2 NS output</p></a></li>
<li><a href='#plot_densitree'><p>Draw multiple trees on top of one another.</p></a></li>
<li><a href='#prepare_file_creation'><p>Internal function to prepare for 'BEAST2' creating files</p></a></li>
<li><a href='#update_babette'><p>Deprecated function.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Control 'BEAST2'</td>
</tr>
<tr>
<td>Version:</td>
<td>2.3.4</td>
</tr>
<tr>
<td>Description:</td>
<td>'BEAST2' (<a href="https://www.beast2.org">https://www.beast2.org</a>) is a widely used
    Bayesian phylogenetic tool, that uses DNA/RNA/protein data
    and many model priors to create a posterior of jointly estimated 
    phylogenies and parameters.
    'BEAST2' is commonly accompanied by 'BEAUti 2', 'Tracer' and 'DensiTree'.
    'babette' provides for an alternative workflow of using 
    all these tools separately. This allows doing complex Bayesian
    phylogenetics easily and reproducibly from 'R'. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://docs.ropensci.org/babette/">https://docs.ropensci.org/babette/</a> (website),
<a href="https://github.com/ropensci/babette">https://github.com/ropensci/babette</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ropensci/babette/issues">https://github.com/ropensci/babette/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>beautier (&ge; 2.6.12), beastier (&ge; 2.5), mauricer (&ge; 2.5.2),
R (&ge; 3.5.0), tracerer</td>
</tr>
<tr>
<td>Imports:</td>
<td>phangorn, rlang (&ge; 1.1.0), stringr, xml2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ape, ggplot2, knitr, lintr, nLTT, rappdirs, rmarkdown,
spelling, testthat (&ge; 2.1.0)</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>BEAST2 (https://www.beast2.org/)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-06-24 09:17:09 UTC; richel</td>
</tr>
<tr>
<td>Author:</td>
<td>Richèl J.C. Bilderbeek
    <a href="https://orcid.org/0000-0003-1107-7049"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Joëlle Barido-Sottani [rev] (Joëlle reviewed the package for rOpenSci,
    see https://github.com/richelbilderbeek/onboarding/issues/209),
  David Winter [rev] (David reviewed the package for rOpenSci, see
    https://github.com/richelbilderbeek/onboarding/issues/209),
  Jason Griffiths <a href="https://orcid.org/0000-0002-1667-8233"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Olivier Roy [ctb],
  Giovanni Laudanno [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Richèl J.C. Bilderbeek &lt;rjcbilderbeek@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-06-25 11:20:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='babette-package'>babette: A package for Bayesian phylogenetics.</h2><span id='topic+babette'></span><span id='topic+babette-package'></span>

<h3>Description</h3>

<p>'babette' provides for an alternative workflow of using
the popular phylogenetics tool 'BEAST2', including
it peripheral tools. From an alignment and
inference model, a posterior of jointly estimated
phylogenies and parameter estimates is generated.
</p>


<h3>Note</h3>

<p>the imports are created by script 'scripts/create_imports.R'
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+bbt_self_test">bbt_self_test</a> to do verify <a href="#topic+babette">babette</a> is installed
correctly.<br />
</p>
<p>These are packages associated with 'babette':
</p>

<ul>
<li>
<p>'<a href="beautier.html#topic+beautier">beautier</a>' creates 'BEAST2' input files.

</p>
</li>
<li>
<p>'<a href="beastier.html#topic+beastier">beastier</a>' runs 'BEAST2'.

</p>
</li>
<li>
<p>'<a href="mauricer.html#topic+mauricer">mauricer</a>' does 'BEAST2' package management.

</p>
</li>
<li>
<p>'<a href="tracerer.html#topic+tracerer">tracerer</a>' parses 'BEAST2' output files.

</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>if (beautier::is_on_ci() &amp;&amp; is_beast2_installed()) {
  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()

  inference_model &lt;- create_test_inference_model()
  beast2_options &lt;- create_beast2_options()

  bbt_run_from_model(
    fasta_filename = get_babette_path("anthus_aco.fas"),
    inference_model = inference_model,
    beast2_options = beast2_options
  )

  # Clean up temporary files created by babette
  bbt_delete_temp_files(
    inference_model = inference_model,
    beast2_options = beast2_options
  )
  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()
}
beastier::remove_beaustier_folders()
</code></pre>

<hr>
<h2 id='bbt_continue'>Continue a BEAST2 run</h2><span id='topic+bbt_continue'></span>

<h3>Description</h3>

<p>Do a full run: create a 'BEAST2' configuration file (like 'BEAUti 2'),
run 'BEAST2', parse results (like 'Tracer')
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bbt_continue(fasta_filename, inference_model, beast2_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bbt_continue_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>a FASTA filename</p>
</td></tr>
<tr><td><code id="bbt_continue_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model,
as returned by <a href="beautier.html#topic+create_inference_model">create_inference_model</a></p>
</td></tr>
<tr><td><code id="bbt_continue_+3A_beast2_options">beast2_options</code></td>
<td>
<p>'BEAST2'  options,
as can be created by <a href="beastier.html#topic+create_beast2_options">create_beast2_options</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with the following elements:<br />
</p>

<ul>
<li>
<p><code>estimates</code>: a data frame with 'BEAST2'
parameter estimates

</p>
</li>
<li>
<p><code>[alignment_id]_trees</code>: a <code>multiPhylo</code>
containing the phylogenies
in the 'BEAST2' posterior. <code>[alignment_id]</code>
is the ID of the alignment. For example,
when running <code>bbt_run_from_model</code> with
<code>anthus_aco.fas</code>, this element will have
name <code>anthus_aco_trees</code>

</p>
</li>
<li>
<p><code>operators</code>: a data frame with the
'BEAST2' MCMC operator acceptances

</p>
</li>
<li>
<p><code>output</code>: a numeric vector with the output
sent to standard output and error streams

</p>
</li>
<li>
<p><code>ns</code>: (optional) the results of a marginal likelihood estimation,
will exist only when <code>create_ns_mcmc</code> was
used for <code>mcmc</code>.
This structure will contain the following elements:
</p>

<ul>
<li> <p><code>marg_log_lik</code> the marginal log likelihood estimate
</p>
</li>
<li> <p><code>marg_log_lik_sd</code> the standard deviation around the estimate
</p>
</li>
<li> <p><code>estimates</code> the parameter estimates
created during the marginal likelihood estimation
</p>
</li>
<li> <p><code>trees</code> the trees
created during the marginal likelihood estimation
</p>
</li></ul>


</li></ul>



<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code><a href="tracerer.html#topic+remove_burn_ins">remove_burn_ins</a></code>
to remove the burn-ins from
the posterior's estimates (<code>posterior$estimates</code>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (beautier::is_on_ci() &amp;&amp; is_beast2_installed()) {
  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()

  # A simple FASTA file
  fasta_filename &lt;- beautier::get_beautier_path("test_output_0.fas")

  # Simple short inference
  inference_model &lt;- create_test_inference_model()

  # Default BEAST2 options
  beast2_options &lt;- create_beast2_options()

  bbt_run_from_model(
    fasta_filename = fasta_filename,
    inference_model = inference_model,
    beast2_options = beast2_options
  )

  bbt_continue(
    fasta_filename = fasta_filename,
    inference_model = inference_model,
    beast2_options = beast2_options
  )

  # Cleanup
  bbt_delete_temp_files(
    inference_model = inference_model,
    beast2_options = beast2_options
  )
  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()
}
</code></pre>

<hr>
<h2 id='bbt_delete_temp_files'>Delete all the temporary files created by <a href="#topic+bbt_run_from_model">bbt_run_from_model</a></h2><span id='topic+bbt_delete_temp_files'></span>

<h3>Description</h3>

<p>Delete all the temporary files created by <a href="#topic+bbt_run_from_model">bbt_run_from_model</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bbt_delete_temp_files(inference_model, beast2_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bbt_delete_temp_files_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model,
as returned by <a href="beautier.html#topic+create_inference_model">create_inference_model</a></p>
</td></tr>
<tr><td><code id="bbt_delete_temp_files_+3A_beast2_options">beast2_options</code></td>
<td>
<p>'BEAST2'  options,
as can be created by <a href="beastier.html#topic+create_beast2_options">create_beast2_options</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (beautier::is_on_ci() &amp;&amp; is_beast2_installed()) {
  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()

  # Do a minimal run
  inference_model &lt;- create_test_inference_model()
  beast2_options &lt;- create_beast2_options()
  bbt_run_from_model(
    fasta_filename = get_fasta_filename(),
    inference_model = inference_model,
    beast2_options = beast2_options
  )

  # Cleanup
  bbt_delete_temp_files(
    inference_model = inference_model,
    beast2_options = beast2_options
  )

  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()
}
</code></pre>

<hr>
<h2 id='bbt_run'>Run BEAST2</h2><span id='topic+bbt_run'></span>

<h3>Description</h3>

<p>Do a full BEAST2 run: create a 'BEAST2' configuration file (like 'BEAUti 2'),
run 'BEAST2', parse results (like 'Tracer')
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bbt_run(
  fasta_filename,
  tipdates_filename = NA,
  site_model = beautier::create_jc69_site_model(),
  clock_model = beautier::create_strict_clock_model(),
  tree_prior = beautier::create_yule_tree_prior(),
  mrca_prior = NA,
  mcmc = beautier::create_mcmc(),
  beast2_input_filename = beastier::create_temp_input_filename(),
  rng_seed = 1,
  beast2_output_state_filename = beastier::create_temp_state_filename(),
  beast2_path = beastier::get_default_beast2_path(),
  overwrite = TRUE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bbt_run_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>a FASTA filename</p>
</td></tr>
<tr><td><code id="bbt_run_+3A_tipdates_filename">tipdates_filename</code></td>
<td>
<p>name of the file containing tip dates</p>
</td></tr>
<tr><td><code id="bbt_run_+3A_site_model">site_model</code></td>
<td>
<p>one site model,
see <a href="beautier.html#topic+create_site_models">create_site_models</a></p>
</td></tr>
<tr><td><code id="bbt_run_+3A_clock_model">clock_model</code></td>
<td>
<p>one clock model,
see <a href="beautier.html#topic+create_clock_model">create_clock_model</a></p>
</td></tr>
<tr><td><code id="bbt_run_+3A_tree_prior">tree_prior</code></td>
<td>
<p>one tree priors,
as created by <a href="beautier.html#topic+create_tree_prior">create_tree_prior</a></p>
</td></tr>
<tr><td><code id="bbt_run_+3A_mrca_prior">mrca_prior</code></td>
<td>
<p>one Most Recent Common Ancestor prior,
as returned by <code><a href="beautier.html#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
<tr><td><code id="bbt_run_+3A_mcmc">mcmc</code></td>
<td>
<p>the MCMC options,
see <a href="beautier.html#topic+create_mcmc">create_mcmc</a></p>
</td></tr>
<tr><td><code id="bbt_run_+3A_beast2_input_filename">beast2_input_filename</code></td>
<td>
<p>path of the 'BEAST2'  configuration file.
By default, this file is put in a temporary folder with a random
filename, as the user needs not read it: it is used as input of 'BEAST2'.
Specifying a <code>beast2_input_filename</code> allows to store that file
in a more permanently stored location.</p>
</td></tr>
<tr><td><code id="bbt_run_+3A_rng_seed">rng_seed</code></td>
<td>
<p>the random number generator seed. Must be either
<code>NA</code> or a positive non-zero value. An RNG seed of <code>NA</code>
results in 'BEAST2'  picking a random seed.</p>
</td></tr>
<tr><td><code id="bbt_run_+3A_beast2_output_state_filename">beast2_output_state_filename</code></td>
<td>
<p>name of the final state file created
by 'BEAST2', containing the operator acceptances. By default, this
file is put a temporary folder with a random
filename, as the user needs not read it: its content
is parsed and returned by this function.
Specifying a <code>beast2_output_state_filename</code> allows to store that file
in a more permanently stored location.</p>
</td></tr>
<tr><td><code id="bbt_run_+3A_beast2_path">beast2_path</code></td>
<td>
<p>name of either a 'BEAST2'  binary file
(usually simply <code>beast</code>)
or a 'BEAST2'  jar file
(usually has a <code>.jar</code> extension).
Use <code>get_default_beast2_bin_path</code> to get
the default BEAST binary file's path
Use <code>get_default_beast2_jar_path</code> to get
the default BEAST jar file's path</p>
</td></tr>
<tr><td><code id="bbt_run_+3A_overwrite">overwrite</code></td>
<td>
<p>will 'BEAST2'  overwrite files? Like 'BEAST2',
this is set to <a href="base.html#topic+TRUE">TRUE</a> by default.
If <a href="base.html#topic+TRUE">TRUE</a>, 'BEAST2'  will overwrite the
<code>beast2_options$output_state_filename</code> if its present.
If <a href="base.html#topic+FALSE">FALSE</a>, 'BEAST2'  will not overwrite the
<code>beast2_options$output_state_filename</code> if its present
and <a href="#topic+babette">babette</a> will give an error message.
Note that if <code>overwrite</code> is set to <a href="base.html#topic+FALSE">FALSE</a> when
a <code>tracelog</code> (see <a href="beautier.html#topic+create_tracelog">create_tracelog</a>),
<code>screenlog</code> (see <a href="beautier.html#topic+create_screenlog">create_screenlog</a>)
or <code>treelog</code> (see <a href="beautier.html#topic+create_treelog">create_treelog</a>)
file already exists,
'BEAST2'  (and thus <a href="#topic+babette">babette</a>) will freeze.</p>
</td></tr>
<tr><td><code id="bbt_run_+3A_verbose">verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Prefer using <code><a href="#topic+bbt_run_from_model">bbt_run_from_model</a></code>, as it has a cleaner interface.
</p>


<h3>Value</h3>

<p>a list with the following elements:<br />
</p>

<ul>
<li>
<p><code>estimates</code>: a data frame with 'BEAST2'
parameter estimates

</p>
</li>
<li>
<p><code>[alignment_id]_trees</code>: a <code>multiPhylo</code>
containing the phylogenies
in the 'BEAST2' posterior. <code>[alignment_id]</code>
is the ID of the alignment. For example,
when running <code><a href="#topic+bbt_run">bbt_run</a></code> with
<code>anthus_aco.fas</code>, this element will have
name <code>anthus_aco_trees</code>

</p>
</li>
<li>
<p><code>operators</code>: a data frame with the
'BEAST2' MCMC operator acceptances

</p>
</li>
<li>
<p><code>output</code>: a numeric vector with the output
sent to standard output and error streams

</p>
</li>
<li>
<p><code>ns</code>: (optional) the results of a marginal likelihood estimation,
will exist only when <code><a href="beautier.html#topic+create_ns_mcmc">create_ns_mcmc</a></code> was
used for the MCMC.
This structure will contain the following elements:
</p>

<ul>
<li> <p><code>marg_log_lik</code> the marginal log likelihood estimate
</p>
</li>
<li> <p><code>marg_log_lik_sd</code> the standard deviation around the estimate
</p>
</li>
<li> <p><code>estimates</code> the parameter estimates
created during the marginal likelihood estimation
</p>
</li>
<li> <p><code>trees</code> the trees
created during the marginal likelihood estimation
</p>
</li></ul>


</li></ul>



<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code><a href="tracerer.html#topic+remove_burn_ins">remove_burn_ins</a></code>
to remove the burn-ins from
the posterior's estimates (<code>posterior$estimates</code>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (beautier::is_on_ci() &amp;&amp; is_beast2_installed()) {
  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()

  # Setup for a short run
  mcmc &lt;- create_test_mcmc()

  # Store filenames for cleanup.
  # Note that 'bbt_run_from_model allows for easier cleanup
  mcmc$tracelog$filename &lt;- tempfile()
  mcmc$treelog$filename &lt;- tempfile()
  mcmc$screenlog$filename &lt;- tempfile()
  beast2_input_filename &lt;- tempfile()
  beast2_output_state_filename &lt;- tempfile()

  bbt_run(
    fasta_filename = get_babette_path("anthus_aco.fas"),
    beast2_input_filename = beast2_input_filename,
    beast2_output_state_filename = beast2_output_state_filename,
    mcmc = mcmc
  )

  # Cleanup
  # Again, note that 'bbt_run_from_model allows for easier cleanup
  file.remove(mcmc$tracelog$filename)
  file.remove(mcmc$treelog$filename)
  file.remove(mcmc$screenlog$filename)
  file.remove(beast2_input_filename)
  file.remove(beast2_output_state_filename)
  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()
}
</code></pre>

<hr>
<h2 id='bbt_run_from_model'>Run BEAST2</h2><span id='topic+bbt_run_from_model'></span>

<h3>Description</h3>

<p>Do a full run: create a 'BEAST2' configuration file (like 'BEAUti 2'),
run 'BEAST2', parse results (like 'Tracer')
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bbt_run_from_model(
  fasta_filename,
  inference_model = beautier::create_inference_model(),
  beast2_options = beastier::create_beast2_options()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bbt_run_from_model_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>a FASTA filename</p>
</td></tr>
<tr><td><code id="bbt_run_from_model_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model,
as returned by <a href="beautier.html#topic+create_inference_model">create_inference_model</a></p>
</td></tr>
<tr><td><code id="bbt_run_from_model_+3A_beast2_options">beast2_options</code></td>
<td>
<p>'BEAST2'  options,
as can be created by <a href="beastier.html#topic+create_beast2_options">create_beast2_options</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with the following elements:<br />
</p>

<ul>
<li>
<p><code>estimates</code>: a data frame with 'BEAST2'
parameter estimates

</p>
</li>
<li>
<p><code>[alignment_id]_trees</code>: a <code>multiPhylo</code>
containing the phylogenies
in the 'BEAST2' posterior. <code>[alignment_id]</code>
is the ID of the alignment. For example,
when running <code>bbt_run_from_model</code> with
<code>anthus_aco.fas</code>, this element will have
name <code>anthus_aco_trees</code>

</p>
</li>
<li>
<p><code>operators</code>: a data frame with the
'BEAST2' MCMC operator acceptances

</p>
</li>
<li>
<p><code>output</code>: a numeric vector with the output
sent to standard output and error streams

</p>
</li>
<li>
<p><code>ns</code>: (optional) the results of a marginal likelihood estimation,
will exist only when <code>create_ns_mcmc</code> was
used for <code>mcmc</code>.
This structure will contain the following elements:
</p>

<ul>
<li> <p><code>marg_log_lik</code> the marginal log likelihood estimate
</p>
</li>
<li> <p><code>marg_log_lik_sd</code> the standard deviation around the estimate
</p>
</li>
<li> <p><code>estimates</code> the parameter estimates
created during the marginal likelihood estimation
</p>
</li>
<li> <p><code>trees</code> the trees
created during the marginal likelihood estimation
</p>
</li></ul>


</li></ul>



<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code><a href="tracerer.html#topic+remove_burn_ins">remove_burn_ins</a></code>
to remove the burn-ins from
the posterior's estimates (<code>posterior$estimates</code>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (beautier::is_on_ci() &amp;&amp; is_beast2_installed()) {
  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()

  # Simple short inference
  inference_model &lt;- create_test_inference_model()

  # Default BEAST2 options
  beast2_options &lt;- create_beast2_options()

  bbt_run_from_model(
    fasta_filename = get_babette_path("anthus_aco.fas"),
    inference_model = inference_model,
    beast2_options = beast2_options
  )

  # Cleanup
  bbt_delete_temp_files(
    inference_model = inference_model,
    beast2_options = beast2_options
  )
  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()
}
</code></pre>

<hr>
<h2 id='bbt_self_test'>Do a self test to verify <a href="#topic+babette">babette</a> that works correctly.</h2><span id='topic+bbt_self_test'></span>

<h3>Description</h3>

<p>Do a self test to verify <a href="#topic+babette">babette</a> that works correctly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bbt_self_test(beast2_options = beastier::create_beast2_options())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bbt_self_test_+3A_beast2_options">beast2_options</code></td>
<td>
<p>'BEAST2'  options,
as can be created by <a href="beastier.html#topic+create_beast2_options">create_beast2_options</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing. Will raise an exception if something is wrong.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (beautier::is_on_ci() &amp;&amp; is_beast2_installed()) {
  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()

  bbt_self_test()

  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()
}
</code></pre>

<hr>
<h2 id='check_beast2_pkgs'>Checks if <code><a href="#topic+bbt_run">bbt_run</a></code> has the 'BEAST2' packages needed to process
its arguments. Will <a href="base.html#topic+stop">stop</a> if not.</h2><span id='topic+check_beast2_pkgs'></span>

<h3>Description</h3>

<p>For example, to use a Nested Sampling MCMC, the 'BEAST2' 'NS' package
needs to be installed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_beast2_pkgs(mcmc, beast2_path = get_default_beast2_bin_path())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_beast2_pkgs_+3A_mcmc">mcmc</code></td>
<td>
<p>the MCMC options,
see <a href="beautier.html#topic+create_mcmc">create_mcmc</a></p>
</td></tr>
<tr><td><code id="check_beast2_pkgs_+3A_beast2_path">beast2_path</code></td>
<td>
<p>name of either a 'BEAST2'  binary file
(usually simply <code>beast</code>)
or a 'BEAST2'  jar file
(usually has a <code>.jar</code> extension).
Use <code>get_default_beast2_bin_path</code> to get
the default BEAST binary file's path
Use <code>get_default_beast2_jar_path</code> to get
the default BEAST jar file's path</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (beautier::is_on_ci() &amp;&amp; is_beast2_installed()) {
  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()

  # Minimal BEAST2 setup
  check_beast2_pkgs(mcmc = create_mcmc())

  # BEAST2 with NS package installed
  if (is_beast2_ns_pkg_installed()) {
    check_beast2_pkgs(mcmc = create_ns_mcmc())
  }

  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()
}
</code></pre>

<hr>
<h2 id='create_test_bbt_run_output'>Get an example output of <code><a href="#topic+bbt_run">bbt_run</a></code>
or <code><a href="#topic+bbt_run_from_model">bbt_run_from_model</a></code>.</h2><span id='topic+create_test_bbt_run_output'></span>

<h3>Description</h3>

<p>This output is used in testing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_test_bbt_run_output()
</code></pre>


<h3>Value</h3>

<p>the same results as <code><a href="#topic+bbt_run">bbt_run</a></code>
or <code><a href="#topic+bbt_run_from_model">bbt_run_from_model</a></code>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>beastier::remove_beaustier_folders()
beastier::check_empty_beaustier_folders()

create_test_bbt_run_output()

beastier::remove_beaustier_folders()
beastier::check_empty_beaustier_folders()
</code></pre>

<hr>
<h2 id='create_test_ns_output'>Create NS testing output</h2><span id='topic+create_test_ns_output'></span>

<h3>Description</h3>

<p>Create testing output similar to when running a 'BEAST2' run
with nested sampling
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_test_ns_output()
</code></pre>


<h3>Value</h3>

<p>a text of type character vector.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+parse_beast2_output_to_ns">parse_beast2_output_to_ns</a> to parse
this output to a Nested Sampling result.
See <a href="beautier.html#topic+create_ns_mcmc">create_ns_mcmc</a> to see how to do a marginal
likelihood estimation using Nested Sampling.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>beastier::remove_beaustier_folders()
beastier::check_empty_beaustier_folders()

create_test_ns_output()

beastier::remove_beaustier_folders()
beastier::check_empty_beaustier_folders()
</code></pre>

<hr>
<h2 id='default_params_doc'>This function does nothing. It is intended to inherit is parameters'
documentation.</h2><span id='topic+default_params_doc'></span>

<h3>Description</h3>

<p>This function does nothing. It is intended to inherit is parameters'
documentation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>default_params_doc(
  beast2_input_filename,
  beast2_options,
  beast2_output_log_filename,
  beast2_output_state_filename,
  beast2_output_trees_filenames,
  beast2_path,
  beast2_working_dir,
  cleanup,
  clock_model,
  clock_models,
  fasta_filename,
  fasta_filenames,
  inference_model,
  mcmc,
  mrca_prior,
  mrca_priors,
  overwrite,
  rng_seed,
  site_model,
  site_models,
  tipdates_filename,
  tree_prior,
  tree_priors,
  verbose
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="default_params_doc_+3A_beast2_input_filename">beast2_input_filename</code></td>
<td>
<p>path of the 'BEAST2'  configuration file.
By default, this file is put in a temporary folder with a random
filename, as the user needs not read it: it is used as input of 'BEAST2'.
Specifying a <code>beast2_input_filename</code> allows to store that file
in a more permanently stored location.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_beast2_options">beast2_options</code></td>
<td>
<p>'BEAST2'  options,
as can be created by <a href="beastier.html#topic+create_beast2_options">create_beast2_options</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_beast2_output_log_filename">beast2_output_log_filename</code></td>
<td>
<p>name of the log file created by 'BEAST2',
containing the parameter estimates in time. By default, this
file is put a temporary folder with a random
filename, as the user needs not read it: its content
is parsed and returned by this function.
Specifying a <code>beast2_output_log_filename</code> allows to store that file
in a more permanently stored location.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_beast2_output_state_filename">beast2_output_state_filename</code></td>
<td>
<p>name of the final state file created
by 'BEAST2', containing the operator acceptances. By default, this
file is put a temporary folder with a random
filename, as the user needs not read it: its content
is parsed and returned by this function.
Specifying a <code>beast2_output_state_filename</code> allows to store that file
in a more permanently stored location.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_beast2_output_trees_filenames">beast2_output_trees_filenames</code></td>
<td>
<p>name of the one or more trees
files created by 'BEAST2', one per alignment. By default, these
files are put a temporary folder with a random
filename, as the user needs not read it: their content
is parsed and returned by this function.
Specifying <code>beast2_output_trees_filenames</code> allows to store these
one or more files in a more permanently stored location.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_beast2_path">beast2_path</code></td>
<td>
<p>name of either a 'BEAST2'  binary file
(usually simply <code>beast</code>)
or a 'BEAST2'  jar file
(usually has a <code>.jar</code> extension).
Use <code>get_default_beast2_bin_path</code> to get
the default BEAST binary file's path
Use <code>get_default_beast2_jar_path</code> to get
the default BEAST jar file's path</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_beast2_working_dir">beast2_working_dir</code></td>
<td>
<p>the folder 'BEAST2'  will work in. This is
an (empty) temporary folder by default. This allows to call
'BEAST2'  in multiple parallel processes, as each process can have
its own working directory</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_cleanup">cleanup</code></td>
<td>
<p>set to FALSE to keep all temporary files</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_clock_model">clock_model</code></td>
<td>
<p>one clock model,
see <a href="beautier.html#topic+create_clock_model">create_clock_model</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_clock_models">clock_models</code></td>
<td>
<p>one or more clock models,
see <a href="beautier.html#topic+create_clock_models">create_clock_models</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>a FASTA filename</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_fasta_filenames">fasta_filenames</code></td>
<td>
<p>one or more FASTA filename, each with one alignment</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model,
as returned by <a href="beautier.html#topic+create_inference_model">create_inference_model</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_mcmc">mcmc</code></td>
<td>
<p>the MCMC options,
see <a href="beautier.html#topic+create_mcmc">create_mcmc</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_mrca_prior">mrca_prior</code></td>
<td>
<p>one Most Recent Common Ancestor prior,
as returned by <code><a href="beautier.html#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_mrca_priors">mrca_priors</code></td>
<td>
<p>a list of one or more Most Recent Common Ancestor priors,
as returned by <code><a href="beautier.html#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_overwrite">overwrite</code></td>
<td>
<p>will 'BEAST2'  overwrite files? Like 'BEAST2',
this is set to <a href="base.html#topic+TRUE">TRUE</a> by default.
If <a href="base.html#topic+TRUE">TRUE</a>, 'BEAST2'  will overwrite the
<code>beast2_options$output_state_filename</code> if its present.
If <a href="base.html#topic+FALSE">FALSE</a>, 'BEAST2'  will not overwrite the
<code>beast2_options$output_state_filename</code> if its present
and <a href="#topic+babette">babette</a> will give an error message.
Note that if <code>overwrite</code> is set to <a href="base.html#topic+FALSE">FALSE</a> when
a <code>tracelog</code> (see <a href="beautier.html#topic+create_tracelog">create_tracelog</a>),
<code>screenlog</code> (see <a href="beautier.html#topic+create_screenlog">create_screenlog</a>)
or <code>treelog</code> (see <a href="beautier.html#topic+create_treelog">create_treelog</a>)
file already exists,
'BEAST2'  (and thus <a href="#topic+babette">babette</a>) will freeze.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_rng_seed">rng_seed</code></td>
<td>
<p>the random number generator seed. Must be either
<code>NA</code> or a positive non-zero value. An RNG seed of <code>NA</code>
results in 'BEAST2'  picking a random seed.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_site_model">site_model</code></td>
<td>
<p>one site model,
see <a href="beautier.html#topic+create_site_models">create_site_models</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_site_models">site_models</code></td>
<td>
<p>one or more site models,
see <a href="beautier.html#topic+create_site_models">create_site_models</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_tipdates_filename">tipdates_filename</code></td>
<td>
<p>name of the file containing tip dates</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_tree_prior">tree_prior</code></td>
<td>
<p>one tree priors,
as created by <a href="beautier.html#topic+create_tree_prior">create_tree_prior</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_tree_priors">tree_priors</code></td>
<td>
<p>one or more tree priors,
see <a href="beautier.html#topic+create_tree_priors">create_tree_priors</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_verbose">verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This is an internal function, so it should be marked with
<code>@noRd</code>. This is not done, as this will disallow all
functions to find the documentation parameters
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='get_alignment_ids_from_xml'>Get the alignment IDs from one or more 'BEAST2' XML input files.</h2><span id='topic+get_alignment_ids_from_xml'></span>

<h3>Description</h3>

<p>Get the alignment IDs from one or more 'BEAST2' XML input files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_alignment_ids_from_xml(xml_filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_alignment_ids_from_xml_+3A_xml_filename">xml_filename</code></td>
<td>
<p>name of a 'BEAST2' XML input file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector with one or more alignment IDs.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>beastier::remove_beaustier_folders()
beastier::check_empty_beaustier_folders()

alignment_ids &lt;- get_alignment_ids_from_xml(
  get_babette_path("anthus_2_4.xml")
)

beastier::remove_beaustier_folders()
beastier::check_empty_beaustier_folders()
</code></pre>

<hr>
<h2 id='get_babette_path'>Get the full path of a file in the <code>inst/extdata</code> folder</h2><span id='topic+get_babette_path'></span>

<h3>Description</h3>

<p>Get the full path of a file in the <code>inst/extdata</code> folder
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_babette_path(filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_babette_path_+3A_filename">filename</code></td>
<td>
<p>the file's name, without the path</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the full path of the filename, if and only if
the file is present. Will stop otherwise.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>for more files, use <code><a href="#topic+get_babette_paths">get_babette_paths</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>beastier::remove_beaustier_folders()
beastier::check_empty_beaustier_folders()

get_babette_path("anthus_aco.fas")

beastier::remove_beaustier_folders()
beastier::check_empty_beaustier_folders()
</code></pre>

<hr>
<h2 id='get_babette_paths'>Get the full paths of files in the <code>inst/extdata</code> folder</h2><span id='topic+get_babette_paths'></span>

<h3>Description</h3>

<p>Get the full paths of files in the <code>inst/extdata</code> folder
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_babette_paths(filenames)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_babette_paths_+3A_filenames">filenames</code></td>
<td>
<p>the files' names, without the path</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the filenames' full paths, if and only if
all files are present. Will stop otherwise.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>for one file, use <code><a href="#topic+get_babette_path">get_babette_path</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>beastier::remove_beaustier_folders()
beastier::check_empty_beaustier_folders()

get_babette_paths(c("anthus_aco.fas", "anthus_nd2.fas"))

beastier::remove_beaustier_folders()
beastier::check_empty_beaustier_folders()
</code></pre>

<hr>
<h2 id='parse_beast2_output'>Process the 'BEAST2' output dependent on 'BEAST2' package specifics</h2><span id='topic+parse_beast2_output'></span>

<h3>Description</h3>

<p>Process the 'BEAST2' output dependent on 'BEAST2' package specifics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parse_beast2_output(out, inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parse_beast2_output_+3A_out">out</code></td>
<td>
<p>a list with the complete babette output, with elements:
</p>

<ul>
<li> <p><code>output</code> textual output of a 'BEAST2' run
</p>
</li></ul>
</td></tr>
<tr><td><code id="parse_beast2_output_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model,
as returned by <a href="beautier.html#topic+create_inference_model">create_inference_model</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>complete babette output with added attributes,
which depends on the 'BEAST2' package.
</p>

<ul>
<li> <p><code>marg_log_lik</code> the marginal log likelihood estimate
</p>
</li>
<li> <p><code>marg_log_lik_sd</code> the standard deviation around the estimate
</p>
</li>
<li> <p><code>estimates</code> the parameter estimates
created during the marginal likelihood estimation
</p>
</li>
<li> <p><code>trees</code> the trees
created during the marginal likelihood estimation
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='parse_beast2_output_to_ns'>Parse BEAST2 NS output</h2><span id='topic+parse_beast2_output_to_ns'></span>

<h3>Description</h3>

<p>Parse the BEAST2 output when run with the BEAST2 NS ('Nested Sampling')
package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parse_beast2_output_to_ns(output)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parse_beast2_output_to_ns_+3A_output">output</code></td>
<td>
<p>screen output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with the following elements:
</p>

<ul>
<li> <p><code>marg_log_lik</code> the marginal log likelihood estimate
</p>
</li>
<li> <p><code>marg_log_lik_sd</code> the standard deviation around the estimate
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <code><a href="#topic+create_test_ns_output">create_test_ns_output</a></code> to obtain
a test screen output.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>beastier::remove_beaustier_folders()
beastier::check_empty_beaustier_folders()

parse_beast2_output_to_ns(
  output = create_test_ns_output()
)

beastier::remove_beaustier_folders()
beastier::check_empty_beaustier_folders()
</code></pre>

<hr>
<h2 id='plot_densitree'>Draw multiple trees on top of one another.</h2><span id='topic+plot_densitree'></span>

<h3>Description</h3>

<p>Draw multiple trees on top of one another.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_densitree(phylos, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_densitree_+3A_phylos">phylos</code></td>
<td>
<p>one or more phylogenies, must be of class <code>multiPhylo</code></p>
</td></tr>
<tr><td><code id="plot_densitree_+3A_...">...</code></td>
<td>
<p>options to be passed to <code>phangorn</code>'s
<a href="phangorn.html#topic+densiTree">densiTree</a> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing. Will produce a plot.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (beautier::is_on_ci() &amp;&amp; is_beast2_installed()) {
  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()

  inference_model &lt;- create_test_inference_model()
  beast2_options &lt;- create_beast2_options()

   out &lt;- bbt_run_from_model(
    get_babette_path("anthus_aco.fas"),
    inference_model = inference_model,
    beast2_options = beast2_options
  )
  bbt_delete_temp_files(
    inference_model = inference_model,
    beast2_options = beast2_options
  )

  plot_densitree(out$anthus_aco_trees)

  # Clean up temporary files created by babette
  bbt_delete_temp_files(
    inference_model = inference_model,
    beast2_options = beast2_options
  )
  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()
}
</code></pre>

<hr>
<h2 id='prepare_file_creation'>Internal function to prepare for 'BEAST2' creating files</h2><span id='topic+prepare_file_creation'></span>

<h3>Description</h3>

<p>The inference model and 'BEAST2' options contain paths that may point
to sub-sub-sub folders. Create those folders and test
if these folders can be written to
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_file_creation(inference_model, beast2_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare_file_creation_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model,
as returned by <a href="beautier.html#topic+create_inference_model">create_inference_model</a></p>
</td></tr>
<tr><td><code id="prepare_file_creation_+3A_beast2_options">beast2_options</code></td>
<td>
<p>'BEAST2'  options,
as can be created by <a href="beastier.html#topic+create_beast2_options">create_beast2_options</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># This example will fail on the CRAN
# r-oldrel-macos-x86_64 platform
if (rappdirs::app_dir()$os != "mac") {
  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()

  # For a test inference model, the files can be prepared
  inference_model &lt;- create_test_inference_model()
  beast2_options &lt;- create_beast2_options()
  prepare_file_creation(inference_model, beast2_options)

  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()
}
</code></pre>

<hr>
<h2 id='update_babette'>Deprecated function.</h2><span id='topic+update_babette'></span>

<h3>Description</h3>

<p>Update all babette dependencies, by installing their
latest versions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>update_babette(upgrade = "default")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="update_babette_+3A_upgrade">upgrade</code></td>
<td>
<p>Deprecated.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See <a href="https://github.com/richelbilderbeek/babetteinstall">https://github.com/richelbilderbeek/babetteinstall</a>
how to do this.
</p>


<h3>Value</h3>

<p>Nothing.
</p>


<h3>Author(s)</h3>

<p>Giovanni Laudanno, Richèl J.C. Bilderbeek
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
