<!DOCTYPE html><html><head><title>Help for package Ball</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Ball}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ArcticLake'><p>Arctic lake sediment samples of different water depth</p></a></li>
<li><a href='#bcor'><p>Ball Covariance and Correlation Statistics</p></a></li>
<li><a href='#bcorsis'><p>Ball Correlation based Sure Independence Screening (BCor-SIS)</p></a></li>
<li><a href='#bcov.test'><p>Ball Covariance Test</p></a></li>
<li><a href='#bd'><p>Ball Divergence statistic</p></a></li>
<li><a href='#bd.gwas.test'><p>Fast K-sample Ball Divergence Test for GWAS Data</p></a></li>
<li><a href='#bd.test'><p>Ball Divergence based Equality of Distributions Test</p></a></li>
<li><a href='#bdvmf'><p>Simulated von Mises-Fisher Data</p></a></li>
<li><a href='#genlung'><p>Lung cancer genomic data</p></a></li>
<li><a href='#macaques'><p>Male and Female macaque data</p></a></li>
<li><a href='#meteorology'><p>meteorological data</p></a></li>
<li><a href='#nhdist'><p>Distance Matrix Computation for Non-Hilbert Data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Statistical Inference and Sure Independence Screening via Ball
Statistics</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.13</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-02-12</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jin Zhu &lt;zhuj37@mail2.sysu.edu.cn&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Hypothesis tests and sure independence screening (SIS) procedure based on ball statistics, including ball divergence &lt;<a href="https://doi.org/10.1214%2F17-AOS1579">doi:10.1214/17-AOS1579</a>&gt;, ball covariance &lt;<a href="https://doi.org/10.1080%2F01621459.2018.1543600">doi:10.1080/01621459.2018.1543600</a>&gt;, and ball correlation &lt;<a href="https://doi.org/10.1080%2F01621459.2018.1462709">doi:10.1080/01621459.2018.1462709</a>&gt;, are developed to analyze complex data in metric spaces, e.g, shape, directional, compositional and symmetric positive definite matrix data. The ball divergence and ball covariance based distribution-free tests are implemented to detecting distribution difference and association in metric spaces &lt;<a href="https://doi.org/10.18637%2Fjss.v097.i06">doi:10.18637/jss.v097.i06</a>&gt;. Furthermore, several generic non-parametric feature selection procedures based on ball correlation, BCor-SIS and all of its variants, are implemented to tackle the challenge in the context of ultra high dimensional data. A fast implementation for large-scale multiple K-sample testing with ball divergence &lt;<a href="https://doi.org/10.1002%2Fgepi.22423">doi:10.1002/gepi.22423</a>&gt; is supported, which is particularly helpful for genome-wide association study.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-02-12 09:18:06 UTC; zhujin</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>utils, gam, survival, mvtnorm</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://mamba413.github.io/Ball/">https://mamba413.github.io/Ball/</a>, <a href="https://github.com/Mamba413/Ball">https://github.com/Mamba413/Ball</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Mamba413/Ball/issues">https://github.com/Mamba413/Ball/issues</a></td>
</tr>
<tr>
<td>Author:</td>
<td>Jin Zhu <a href="https://orcid.org/0000-0001-8550-5822"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Wenliang Pan [aut],
  Yuan Tian [aut],
  Weinan Xiao [aut],
  Chengfeng Liu [aut],
  Ruihuang Liu [aut],
  Yue Hu [aut],
  Hongtu Zhu [aut],
  Heping Zhang [aut],
  Xueqin Wang <a href="https://orcid.org/0000-0001-5205-9950"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-02-12 10:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ArcticLake'>Arctic lake sediment samples of different water depth</h2><span id='topic+ArcticLake'></span>

<h3>Description</h3>

<p>Sand, silt and clay compositions of 39 sediment samples of different water 
depth in an Arctic lake.
</p>


<h3>Format</h3>


<p><code>ArcticLake$depth</code>: water depth (in meters). 
</p>
<p><code>ArcticLake$x</code>: compositions of three covariates: sand, silt, and clay.

</p>


<h3>Details</h3>

<p>Sand, silt and clay compositions of 39 sediment samples at different water 
depth (in meters) in an Arctic lake. The additional feature is a concomitant variable or 
covariate, water depth, which may account for some of the variation in the compositions. 
In statistical terminology, we have a multivariate regression problem with sediment 
composition as predictors and water depth as a response. All row percentage sums to 100, 
except for rounding errors.
</p>


<h3>Note</h3>

<p>Courtesy of J. Aitchison
</p>


<h3>Source</h3>

<p>Aitchison: CODA microcomputer statistical package, 1986, the file name ARCTIC.DAT, here included under the GNU Public Library Licence Version 2 or newer.
</p>


<h3>References</h3>

<p>Aitchison: The Statistical Analysis of Compositional Data, 1986, Data 5, pp5.
</p>

<hr>
<h2 id='bcor'>Ball Covariance and Correlation Statistics</h2><span id='topic+bcor'></span><span id='topic+bcov'></span>

<h3>Description</h3>

<p>Computes Ball Covariance and Ball Correlation statistics, 
which are generic dependence measures in Banach spaces.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bcor(x, y, distance = FALSE, weight = FALSE)

bcov(x, y, distance = FALSE, weight = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bcor_+3A_x">x</code></td>
<td>
<p>a numeric vector, matrix, data.frame, or a list containing at least two numeric vectors, matrices, or data.frames.</p>
</td></tr>
<tr><td><code id="bcor_+3A_y">y</code></td>
<td>
<p>a numeric vector, matrix, or data.frame.</p>
</td></tr>
<tr><td><code id="bcor_+3A_distance">distance</code></td>
<td>
<p>if <code>distance = TRUE</code>, the elements of <code>x</code> and <code>y</code> are considered as distance matrices.</p>
</td></tr>
<tr><td><code id="bcor_+3A_weight">weight</code></td>
<td>
<p>a logical or character string used to choose the weight form of Ball Covariance statistic.. 
If input is a character string, it must be one of <code>"constant"</code>, <code>"probability"</code>, or <code>"chisquare"</code>. 
Any unambiguous substring can be given. 
If input is a logical value, it is equivalent to <code>weight = "probability"</code> if <code>weight = TRUE</code> while 
equivalent to <code>weight = "constant"</code> if <code>weight = FALSE</code>.
Default: <code>weight = FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The sample sizes of the two variables must agree,  and samples must not contain missing and infinite values. 
If we set <code>distance = TRUE</code>, arguments <code>x</code>, <code>y</code> can be a <code>dist</code> object or a
symmetric numeric matrix recording distance between samples; otherwise, these arguments are treated as data.
</p>
<p><code>bcov</code> and <code>bcor</code> compute Ball Covariance and Ball Correlation statistics.
</p>
<p>Ball Covariance statistics is a generic dependence measure in Banach spaces. It enjoys the following properties: 
</p>

<ul>
<li><p> It is nonnegative and it is equal to zero if and only if variables are unassociated; 
</p>
</li>
<li><p> It is highly robust;   
</p>
</li>
<li><p> It is distribution-free and model-free;   
</p>
</li>
<li><p> it is interesting that the HHG is a special case of Ball Covariance statistics. 
</p>
</li></ul>

<p>Ball correlation statistics, a normalized version of Ball Covariance statistics, generalizes Pearson correlation in two fundamental ways: 
</p>

<ul>
<li><p> It is well-defined for random variables in arbitrary dimension in Banach spaces
</p>
</li>
<li><p> BCor is equal to zero implies random variables are unassociated.
</p>
</li></ul>

<p>The definitions of the Ball Covariance and Ball Correlation statistics between two random variables are as follows.
Suppose, we are given pairs of independent observations 
<code class="reqn">\{(x_1, y_1),...,(x_n,y_n)\}</code>, where <code class="reqn">x_i</code> and <code class="reqn">y_i</code> can be of any dimension 
and the dimensionality of <code class="reqn">x_i</code> and <code class="reqn">y_i</code> need not be the same.
Then, we define sample version Ball Covariance as:
</p>
<p style="text-align: center;"><code class="reqn">\mathbf{BCov}_{\omega, n}^{2}(X, Y)=\frac{1}{n^{2}}\sum_{i,j=1}^{n}{(\Delta_{ij,n}^{X,Y}-\Delta_{ij,n}^{X}\Delta_{ij,n}^{Y})^{2}} </code>
</p>

<p>where:
</p>
<p style="text-align: center;"><code class="reqn"> \Delta_{ij,n}^{X,Y}=\frac{1}{n}\sum_{k=1}^{n}{\delta_{ij,k}^{X} \delta_{ij,k}^{Y}}, 
\Delta_{ij,n}^{X}=\frac{1}{n}\sum_{k=1}^{n}{\delta_{ij,k}^{X}}, 
\Delta_{ij,n}^{Y}=\frac{1}{n}\sum_{k=1}^{n}{\delta_{ij,k}^{Y}} </code>
</p>

<p style="text-align: center;"><code class="reqn">\delta_{ij,k}^{X} = I(x_{k} \in \bar{B}(x_{i}, \rho(x_{i}, x_{j}))), 
\delta_{ij,k}^{Y} = I(y_{k} \in \bar{B}(y_{i}, \rho(y_{i}, y_{j})))</code>
</p>

<p>Among them, <code class="reqn">\bar{B}(x_{i}, \rho(x_{i}, x_{j}))</code> is a closed ball 
with center <code class="reqn">x_{i}</code> and radius <code class="reqn">\rho(x_{i}, x_{j})</code>.
Similarly, we can define <code class="reqn"> \mathbf{BCov}_{\omega,n}^2(\mathbf{X},\mathbf{X}) </code> 
and <code class="reqn"> \mathbf{BCov}_{\omega,n}^2(\mathbf{Y},\mathbf{Y}) </code>. 
We define Ball Correlation statistic as follows.
</p>
<p style="text-align: center;"><code class="reqn">\mathbf{BCor}_{\omega,n}^2(\mathbf{X},\mathbf{Y})=
\mathbf{BCov}_{\omega,n}^2(\mathbf{X},\mathbf{Y})/\sqrt{\mathbf{BCov}_{\omega,n}^2(\mathbf{X},\mathbf{X})\mathbf{BCov}_{\omega,n}^2(\mathbf{Y},\mathbf{Y})}
</code>
</p>

<p>We can extend <code class="reqn">\mathbf{BCov}_{\omega,n}</code> to measure the mutual independence between <code class="reqn">K</code> random variables:
</p>
<p style="text-align: center;"><code class="reqn">\frac{1}{n^{2}}\sum_{i,j=1}^{n}{\left[ (\Delta_{ij,n}^{X_{1}, ..., X_{K}}-\prod_{k=1}^{K}\Delta_{ij,n}^{X_{k}})^{2}\prod_{k=1}^{K}{\hat{\omega}_{k}(X_{ki},X_{kj})} \right]}</code>
</p>

<p>where <code class="reqn">X_{k}(k=1,\ldots,K)</code> are random variables and <code class="reqn">X_{ki}</code> is the <code class="reqn">i</code>-th observations of <code class="reqn">X_{k}</code>. 
</p>
<p>See <code><a href="#topic+bcov.test">bcov.test</a></code> for a test of independence based on the Ball Covariance statistic.
</p>


<h3>Value</h3>

<table>
<tr><td><code>bcor</code></td>
<td>
<p> Ball Correlation statistic.</p>
</td></tr>
<tr><td><code>bcov</code></td>
<td>
<p> Ball Covariance statistic.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Wenliang Pan, Xueqin Wang, Heping Zhang, Hongtu Zhu &amp; Jin Zhu (2019) Ball Covariance: A Generic Measure of Dependence in Banach Space, Journal of the American Statistical Association, DOI: 10.1080/01621459.2018.1543600
</p>
<p>Wenliang Pan, Xueqin Wang, Weinan Xiao &amp; Hongtu Zhu (2018) A Generic Sure Independence Screening Procedure, Journal of the American Statistical Association, DOI: 10.1080/01621459.2018.1462709
</p>
<p>Jin Zhu, Wenliang Pan, Wei Zheng, and Xueqin Wang (2021). Ball: An R Package for Detecting Distribution Difference and Association in Metric Spaces, Journal of Statistical Software, Vol.97(6), doi: 10.18637/jss.v097.i06.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bcov.test">bcov.test</a></code>, <code><a href="#topic+bcorsis">bcorsis</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>############# Ball Correlation #############
num &lt;- 50
x &lt;- 1:num
y &lt;- 1:num
bcor(x, y)
bcor(x, y, weight = "prob")
bcor(x, y, weight = "chisq")
############# Ball Covariance #############
num &lt;- 50
x &lt;- rnorm(num)
y &lt;- rnorm(num)
bcov(x, y)
bcov(x, y, weight = "prob")
bcov(x, y, weight = "chisq")
</code></pre>

<hr>
<h2 id='bcorsis'>Ball Correlation based Sure Independence Screening (BCor-SIS)</h2><span id='topic+bcorsis'></span>

<h3>Description</h3>

<p>Generic non-parametric sure independence screening (SIS) procedure based on Ball Correlation.
Ball correlation is a generic measure of dependence in Banach spaces.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bcorsis(
  x,
  y,
  d = "small",
  weight = c("constant", "probability", "chisquare"),
  method = "standard",
  distance = FALSE,
  category = FALSE,
  parms = list(d1 = 5, d2 = 5, df = 3),
  num.threads = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bcorsis_+3A_x">x</code></td>
<td>
<p>a numeric matrix or data.frame included <code class="reqn">n</code> rows and <code class="reqn">p</code> columns. 
Each row is an observation vector and each column corresponding to a explanatory variable, generally <code class="reqn">p &gt;&gt; n</code>.</p>
</td></tr>
<tr><td><code id="bcorsis_+3A_y">y</code></td>
<td>
<p>a numeric vector, matrix, or data.frame.</p>
</td></tr>
<tr><td><code id="bcorsis_+3A_d">d</code></td>
<td>
<p>the hard cutoff rule suggests selecting <code class="reqn">d</code> variables. Setting <code>d = "large"</code> or 
<code>d = "small"</code> means <code>n - 1</code> or <code>floor(n/log(n))</code> 
variables are selected. If <code>d</code> is a integer, <code>d</code> variables are selected. Default: <code>d = "small"</code>.</p>
</td></tr>
<tr><td><code id="bcorsis_+3A_weight">weight</code></td>
<td>
<p>a logical or character string used to choose the weight form of Ball Covariance statistic.. 
If input is a character string, it must be one of <code>"constant"</code>, <code>"probability"</code>, or <code>"chisquare"</code>. 
Any unambiguous substring can be given. 
If input is a logical value, it is equivalent to <code>weight = "probability"</code> if <code>weight = TRUE</code> while 
equivalent to <code>weight = "constant"</code> if <code>weight = FALSE</code>.
Default: <code>weight = FALSE</code>.</p>
</td></tr>
<tr><td><code id="bcorsis_+3A_method">method</code></td>
<td>
<p>specific method for the BCor-SIS procedure. It must be one of <code>"standard"</code>,
<code>"lm"</code>, <code>"gam"</code>, <code>"interaction"</code>, or <code>"survival"</code>.
Setting <code>method = "standard"</code> means performing standard SIS procedure 
while the options <code>"lm"</code> and <code>"gam"</code> mean carrying out iterative SIS procedure with ordinary 
linear regression and generalized additive models, respectively.
The options <code>"interaction"</code> and <code>"survival"</code> are designed for detecting variables 
with potential linear interaction and associated with left censored responses, respectively. 
Any unambiguous substring can be given. Default: <code>method = "standard"</code>.</p>
</td></tr>
<tr><td><code id="bcorsis_+3A_distance">distance</code></td>
<td>
<p>if <code>distance = TRUE</code>, <code>y</code> will be considered as a distance matrix. 
Arguments only available when <code>method = "standard"</code> and <code>method = "interaction"</code>. Default: <code>distance = FALSE</code>.</p>
</td></tr>
<tr><td><code id="bcorsis_+3A_category">category</code></td>
<td>
<p>a logical value or integer vector indicating columns to be selected as categorical variables.
If <code>category</code> is an integer vector, the positive/negative integers select/discard the corresponding columns;
If <code>category</code> is a logical value, <code>category = TRUE</code> select all columns, <code>category = FALSE</code> select none column.
Default: <code>category = FALSE</code>.</p>
</td></tr>
<tr><td><code id="bcorsis_+3A_parms">parms</code></td>
<td>
<p>parameters list only available when <code>method = "lm"</code> or <code>"gam"</code>. 
It contains three parameters: <code>d1</code>, <code>d2</code>, and <code>df</code>. <code>d1</code> is the
number of initially selected variables, <code>d2</code> is the number of variables added in each iteration.
<code>df</code> is a degree freedom of basis in generalized additive models playing a role only when <code>method = "gam"</code>. 
Default: <code>parms = list(d1 = 5, d2 = 5, df = 3)</code>.</p>
</td></tr>
<tr><td><code id="bcorsis_+3A_num.threads">num.threads</code></td>
<td>
<p>number of threads. If <code>num.threads = 0</code>, then all of available cores will be used. Default <code>num.threads = 0</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>bcorsis</code> performs a model-free generic sure independence screening procedure, 
BCor-SIS, to pick out variables from <code>x</code> which are potentially associated with <code>y</code>. 
BCor-SIS relies on Ball correlation, a universal dependence measure in Banach spaces.
Ball correlation (BCor) ranges from 0 to 1. A larger BCor implies they are likely to be associated while 
Bcor is equal to 0 implies they are unassociated. (See <code><a href="#topic+bcor">bcor</a></code> for details.)
Consequently, BCor-SIS pick out variables with larger Bcor values with <code>y</code>.
</p>
<p>Theory and numerical result indicate that BCor-SIS has following advantages:
</p>

<ul>
<li><p> BCor-SIS can retain the efficient variables even when the dimensionality (i.e., <code>ncol(x)</code>) is 
an exponential order of the sample size (i.e., <code>exp(nrow(x))</code>);
</p>
</li>
<li><p> It is distribution-free and model-free;
</p>
</li>
<li><p> It is very robust;
</p>
</li>
<li><p> It is works well for complex data, such as shape and survival data;
</p>
</li></ul>

<p>If <code>x</code> is a matrix, the sample sizes of <code>x</code> and <code>y</code> must agree.
If <code>x</code> is a <code><a href="base.html#topic+list">list</a></code> object, each element of this <code>list</code> must with the same sample size.
<code>x</code> and <code>y</code> must not contain missing or infinite values. 
</p>
<p>When <code>method = "survival"</code>, the matrix or data.frame pass to <code>y</code> must have exactly two columns, where the first column is 
event (failure) time while the second column is a dichotomous censored status.
</p>


<h3>Value</h3>

<table>
<tr><td><code>ix</code></td>
<td>
<p> the indices vector corresponding to variables selected by BCor-SIS.</p>
</td></tr> 
<tr><td><code>method</code></td>
<td>
<p> the method used.</p>
</td></tr> 
<tr><td><code>weight</code></td>
<td>
<p> the weight used.</p>
</td></tr> 
<tr><td><code>complete.info</code></td>
<td>
<p> a <code>list</code> mainly containing a <code class="reqn">p \times 3</code> matrix, 
where each row is a variable and each column is a weight Ball Correlation statistic. 
If <code>method = "gam"</code> or <code>method = "lm"</code>, <code>complete.info</code> is an empty list.</p>
</td></tr>
</table>


<h3>Note</h3>

<p><code>bcorsis</code> simultaneously computing Ball Correlation statistics with 
<code>"constant"</code>, <code>"probability"</code>, and <code>"chisquare"</code> weights.
Users can get other Ball Correlation statistics with different weight in the <code>complete.info</code> element of output. 
We give a quick example below to illustrate.
</p>


<h3>Author(s)</h3>

<p>Wenliang Pan, Weinan Xiao, Xueqin Wang, Hongtu Zhu, Jin Zhu
</p>


<h3>References</h3>

<p>Wenliang Pan, Xueqin Wang, Weinan Xiao &amp; Hongtu Zhu (2018) A Generic Sure Independence Screening Procedure, Journal of the American Statistical Association, DOI: 10.1080/01621459.2018.1462709
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bcor">bcor</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

############### Quick Start for bcorsis function ###############
set.seed(1)
n &lt;- 150
p &lt;- 3000
x &lt;- matrix(rnorm(n * p), nrow = n)
eps &lt;- rnorm(n)
y &lt;- 3 * x[, 1] + 5 * (x[, 3])^2 + eps
res &lt;- bcorsis(y = y, x = x)
head(res[["ix"]])
head(res[["complete.info"]][["statistic"]])

############### BCor-SIS: Censored Data Example ###############
data("genlung")
result &lt;- bcorsis(x = genlung[["covariate"]], y = genlung[["survival"]], 
                  method = "survival")
index &lt;- result[["ix"]]
top_gene &lt;- colnames(genlung[["covariate"]])[index]
head(top_gene, n = 1)


############### BCor-SIS: Interaction Pursuing ###############
set.seed(1)
n &lt;- 150
p &lt;- 3000
x &lt;- matrix(rnorm(n * p), nrow = n)
eps &lt;- rnorm(n)
y &lt;- 3 * x[, 1] * x[, 5] * x[, 10] + eps
res &lt;- bcorsis(y = y, x = x, method = "interaction")
head(res[["ix"]])

############### BCor-SIS: Iterative Method ###############
library(mvtnorm)
set.seed(1)
n &lt;- 150
p &lt;- 3000
sigma_mat &lt;- matrix(0.5, nrow = p, ncol = p)
diag(sigma_mat) &lt;- 1
x &lt;- rmvnorm(n = n, sigma = sigma_mat)
eps &lt;- rnorm(n)
rm(sigma_mat); gc(reset = TRUE)
y &lt;- 3 * (x[, 1])^2 + 5 * (x[, 2])^2 + 5 * x[, 8] - 8 * x[, 16] + eps
res &lt;- bcorsis(y = y, x = x, method = "lm", d = 15)
res &lt;- bcorsis(y = y, x = x, method = "gam", d = 15)
res[["ix"]]

############### Weighted BCor-SIS: Probability weight ###############
set.seed(1)
n &lt;- 150
p &lt;- 3000
x &lt;- matrix(rnorm(n * p), nrow = n)
eps &lt;- rnorm(n)
y &lt;- 3 * x[, 1] + 5 * (x[, 3])^2 + eps
res &lt;- bcorsis(y = y, x = x, weight = "prob")
head(res[["ix"]])
# Alternative, chisq weight:
res &lt;- bcorsis(y = y, x = x, weight = "chisq")
head(res[["ix"]])

############### BCor-SIS: GWAS data ###############
set.seed(1)
n &lt;- 150
p &lt;- 3000
x &lt;- sapply(1:p, function(i) {
  sample(0:2, size = n, replace = TRUE)
})
eps &lt;- rnorm(n)
y &lt;- 6 * x[, 1] - 7 * x[, 2] + 5 * x[, 3] + eps
res &lt;- bcorsis(x = x, y = y, category = TRUE)
head(res[["ix"]])
head(res[["complete.info"]][["statistic"]])

x &lt;- cbind(matrix(rnorm(n * 2), ncol = 2), x)
# remove the first two columns:
res &lt;- bcorsis(x = x, y = y, category = c(-1, -2))
head(res[["ix"]])

x &lt;- cbind(x[, 3:5], matrix(rnorm(n * p), ncol = p))
res &lt;- bcorsis(x = x, y = y, category = 1:3)
head(res[["ix"]], n = 10)

## End(Not run)
</code></pre>

<hr>
<h2 id='bcov.test'>Ball Covariance Test</h2><span id='topic+bcov.test'></span><span id='topic+bcov.test.default'></span><span id='topic+bcov.test.formula'></span>

<h3>Description</h3>

<p>Ball Covariance test of independence. 
Ball covariance are generic dependence measures in Banach spaces.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bcov.test(x, ...)

## Default S3 method:
bcov.test(
  x,
  y = NULL,
  num.permutations = 99,
  method = c("permutation", "limit"),
  distance = FALSE,
  weight = FALSE,
  seed = 1,
  num.threads = 0,
  ...
)

## S3 method for class 'formula'
bcov.test(formula, data, subset, na.action, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bcov.test_+3A_x">x</code></td>
<td>
<p>a numeric vector, matrix, data.frame, or a list containing at least two numeric vectors, matrices, or data.frames.</p>
</td></tr>
<tr><td><code id="bcov.test_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to or from methods.</p>
</td></tr>
<tr><td><code id="bcov.test_+3A_y">y</code></td>
<td>
<p>a numeric vector, matrix, or data.frame.</p>
</td></tr>
<tr><td><code id="bcov.test_+3A_num.permutations">num.permutations</code></td>
<td>
<p>the number of permutation replications. When <code>num.permutations = 0</code>, the function just returns
the Ball Covariance statistic. Default: <code>num.permutations = 99</code>.</p>
</td></tr>
<tr><td><code id="bcov.test_+3A_method">method</code></td>
<td>
<p>if <code>method = "permutation"</code>, a permutation procedure is carried out to compute the <code class="reqn">p</code>-value;
if <code> method = "limit"</code>, an approximate null distribution is used when <code>weight = "constant"</code>.
Any unambiguous substring can be given. Default <code>method = "permutation"</code>.</p>
</td></tr>
<tr><td><code id="bcov.test_+3A_distance">distance</code></td>
<td>
<p>if <code>distance = TRUE</code>, the elements of <code>x</code> and <code>y</code> are considered as distance matrices.</p>
</td></tr>
<tr><td><code id="bcov.test_+3A_weight">weight</code></td>
<td>
<p>a logical or character string used to choose the weight form of Ball Covariance statistic.. 
If input is a character string, it must be one of <code>"constant"</code>, <code>"probability"</code>, or <code>"chisquare"</code>. 
Any unambiguous substring can be given. 
If input is a logical value, it is equivalent to <code>weight = "probability"</code> if <code>weight = TRUE</code> while 
equivalent to <code>weight = "constant"</code> if <code>weight = FALSE</code>.
Default: <code>weight = FALSE</code>.</p>
</td></tr>
<tr><td><code id="bcov.test_+3A_seed">seed</code></td>
<td>
<p>the random seed. Default <code>seed = 1</code>.</p>
</td></tr>
<tr><td><code id="bcov.test_+3A_num.threads">num.threads</code></td>
<td>
<p>number of threads. If <code>num.threads = 0</code>, then all of available cores will be used. Default <code>num.threads = 0</code>.</p>
</td></tr>
<tr><td><code id="bcov.test_+3A_formula">formula</code></td>
<td>
<p>a formula of the form <code>~ u + v</code>, where each of <code>u</code> and <code>v</code> are numeric variables giving the data values for one sample. The samples must be of the same length.</p>
</td></tr>
<tr><td><code id="bcov.test_+3A_data">data</code></td>
<td>
<p>an optional matrix or data frame (or similar: see <code>model.frame</code>) containing the variables in the formula formula. By default the variables are taken from environment(formula).</p>
</td></tr>
<tr><td><code id="bcov.test_+3A_subset">subset</code></td>
<td>
<p>an optional vector specifying a subset of observations to be used.</p>
</td></tr>
<tr><td><code id="bcov.test_+3A_na.action">na.action</code></td>
<td>
<p>a function which indicates what should happen when the data contain <code>NA</code>s. Defaults to <code>getOption("na.action")</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>bcov.test</code> is non-parametric tests of independence in Banach spaces. 
It can detect the dependence between two random objects (variables) and 
the mutual dependence among at least three random objects (variables).
</p>
<p>If two samples are pass to arguments <code>x</code> and <code>y</code>, the sample sizes (i.e. number of rows or length of the vector) 
of the two variables must agree. If a <code><a href="base.html#topic+list">list</a></code> object is passed to <code>x</code>, this <code>list</code> must contain at least
two numeric vectors, matrices, or data.frames, and each element of this <code>list</code> 
must with the same sample size. Moreover, data pass to <code>x</code> or <code>y</code> 
must not contain missing or infinite values. 
If <code>distance = TRUE</code>, <code>x</code> is considered as a distance matrix or a list containing distance matrices, 
and <code>y</code> is considered as a distance matrix; otherwise, these arguments are treated as data.
</p>
<p><code>bcov.test</code> utilizes the Ball Covariance statistics (see <code><a href="#topic+bcov">bcov</a></code>) to measure dependence and 
derives a <code class="reqn">p</code>-value via replicating the random permutation <code>num.permutations</code> times.
</p>
<p>See Pan et al 2018 for theoretical properties of the test, including statistical consistency.
</p>


<h3>Value</h3>

<p>If <code>num.permutations &gt; 0</code>, <code>bcov.test</code> returns a <code>htest</code> class object containing the following components:
</p>
<table>
<tr><td><code>statistic</code></td>
<td>
<p>Ball Covariance statistic.</p>
</td></tr>            
<tr><td><code>p.value</code></td>
<td>
<p>the p-value for the test.</p>
</td></tr>  
<tr><td><code>replicates</code></td>
<td>
<p>permutation replications of the test statistic.</p>
</td></tr>
<tr><td><code>size</code></td>
<td>
<p>sample size.</p>
</td></tr> 
<tr><td><code>complete.info</code></td>
<td>
<p>a <code>list</code> mainly containing two vectors, the first vector is the Ball Covariance statistics 
with different weights, the second is the <code class="reqn">p</code>-values of weighted Ball Covariance tests.</p>
</td></tr>
<tr><td><code>alternative</code></td>
<td>
<p>a character string describing the alternative hypothesis.</p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p>a character string indicating what type of test was performed.</p>
</td></tr>
<tr><td><code>data.name</code></td>
<td>
<p>description of data.</p>
</td></tr>
</table>
<p>If <code>num.permutations = 0</code>, <code>bcov.test</code> returns a statistic value.
</p>


<h3>Note</h3>

<p>Actually, <code>bcov.test</code> simultaneously computing Ball Covariance statistics with 
<code>"constant"</code>, <code>"probability"</code>, and <code>"chisquare"</code> weights.
Users can get other Ball Covariance statistics with different weight and their corresponding <code class="reqn">p</code>-values 
in the <code>complete.info</code> element of output. We give a quick example below to illustrate.
</p>


<h3>Author(s)</h3>

<p>Wenliang Pan, Xueqin Wang, Heping Zhang, Hongtu Zhu, Jin Zhu
</p>


<h3>References</h3>

<p>Wenliang Pan, Xueqin Wang, Heping Zhang, Hongtu Zhu &amp; Jin Zhu (2019) Ball Covariance: A Generic Measure of Dependence in Banach Space, Journal of the American Statistical Association, DOI: 10.1080/01621459.2018.1543600
</p>
<p>Jin Zhu, Wenliang Pan, Wei Zheng, and Xueqin Wang (2021). Ball: An R Package for Detecting Distribution Difference and Association in Metric Spaces, Journal of Statistical Software, Vol.97(6), doi: 10.18637/jss.v097.i06.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bcov">bcov</a></code>, <code><a href="#topic+bcor">bcor</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)

################# Quick Start #################
noise &lt;- runif(50, min = -0.3, max = 0.3)
x &lt;- runif(50, 0, 4*pi)
y &lt;- cos(x) + noise
# plot(x, y)
res &lt;- bcov.test(x, y)
res
## get all Ball Covariance statistics:
res[["complete.info"]][["statistic"]]
## get all test result:
res[["complete.info"]][["p.value"]]

################# Quick Start #################
x &lt;- matrix(runif(50 * 2, -pi, pi), nrow = 50, ncol = 2)
noise &lt;- runif(50, min = -0.1, max = 0.1)
y &lt;- sin(x[,1] + x[,2]) + noise
bcov.test(x = x, y = y, weight = "prob")

################# Ball Covariance Test for Non-Hilbert Data #################
# load data:
data("ArcticLake")
# Distance matrix between y:
Dy &lt;- nhdist(ArcticLake[["x"]], method = "compositional")
# Distance matrix between x:
Dx &lt;- dist(ArcticLake[["depth"]])
# hypothesis test with BCov:
bcov.test(x = Dx, y = Dy, distance = TRUE)

################  Weighted Ball Covariance Test  #################
data("ArcticLake")
Dy &lt;- nhdist(ArcticLake[["x"]], method = "compositional")
Dx &lt;- dist(ArcticLake[["depth"]])
# hypothesis test with weighted BCov:
bcov.test(x = Dx, y = Dy, distance = TRUE, weight = "prob")

################# Mutual Independence Test #################
x &lt;- rnorm(50)
y &lt;- (x &gt; 0) * x + rnorm(50)
z &lt;- (x &lt;= 0) * x + rnorm(50)
data_list &lt;- list(x, y, z)
bcov.test(data_list)
data_list &lt;- lapply(data_list, function(x) {
  as.matrix(dist(x))
})
bcov.test(data_list, distance = TRUE)
bcov.test(data_list, distance = FALSE, weight = "chi")

################# Mutual Independence Test for Meteorology data #################
data("meteorology")
bcov.test(meteorology)

################  Testing via approximate limit distribution  #################
## Not run: 
set.seed(1)
n &lt;- 2000
x &lt;- rnorm(n)
y &lt;- rnorm(n)
bcov.test(x, y, method = "limit")
bcov.test(x, y)

## End(Not run)

################  Formula interface  ################
## independence test:
bcov.test(~ CONT + INTG, data = USJudgeRatings)
## independence test with chisquare weight:
bcov.test(~ CONT + INTG, data = USJudgeRatings, weight = "chi")
## mutual independence test:
bcov.test(~ CONT + INTG + DMNR, data = USJudgeRatings)
</code></pre>

<hr>
<h2 id='bd'>Ball Divergence statistic</h2><span id='topic+bd'></span>

<h3>Description</h3>

<p>Compute Ball Divergence statistic, which is a generic dispersion measure in Banach spaces.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bd(
  x,
  y = NULL,
  distance = FALSE,
  size = NULL,
  num.threads = 1,
  kbd.type = c("sum", "maxsum", "max")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bd_+3A_x">x</code></td>
<td>
<p>a numeric vector, matrix, data.frame, or a list containing at least two numeric vectors, matrices, or data.frames.</p>
</td></tr>
<tr><td><code id="bd_+3A_y">y</code></td>
<td>
<p>a numeric vector, matrix, data.frame.</p>
</td></tr>
<tr><td><code id="bd_+3A_distance">distance</code></td>
<td>
<p>if <code>distance = TRUE</code>, the elements of <code>x</code> will be considered as a distance matrix. Default: <code>distance = FALSE</code>.</p>
</td></tr>
<tr><td><code id="bd_+3A_size">size</code></td>
<td>
<p>a vector recording sample size of each group.</p>
</td></tr>
<tr><td><code id="bd_+3A_num.threads">num.threads</code></td>
<td>
<p>number of threads. If <code>num.threads = 0</code>, then all of available cores will be used. Default <code>num.threads = 0</code>.</p>
</td></tr>
<tr><td><code id="bd_+3A_kbd.type">kbd.type</code></td>
<td>
<p>a character string specifying the <code class="reqn">K</code>-sample Ball Divergence test statistic, 
must be one of <code>"sum"</code>, <code>"summax"</code>, or <code>"max"</code>. Any unambiguous substring can be given. 
Default <code>kbd.type = "sum"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Given the samples not containing missing values, <code>bd</code> returns Ball Divergence statistics.
If we set <code>distance = TRUE</code>, arguments <code>x</code>, <code>y</code> can be a <code>dist</code> object or a
symmetric numeric matrix recording distance between samples; 
otherwise, these arguments are treated as data.
</p>
<p>Ball divergence statistic measure the distribution difference of two datasets in Banach spaces. 
The Ball divergence statistic is proven to be zero if and only if two datasets are identical.
</p>
<p>The definition of the Ball Divergence statistics is as follows.
Given two independent samples <code class="reqn"> \{x_{1}, \ldots, x_{n}\} </code> with the associated probability measure <code class="reqn">\mu</code> and 
<code class="reqn"> \{y_{1}, \ldots, y_{m}\} </code> with <code class="reqn">\nu</code>, where the observations in each sample are <em>i.i.d</em>.
Let <code class="reqn">\delta(x,y,z)=I(z\in \bar{B}(x, \rho(x,y)))</code>, 
where <code class="reqn">\delta(x,y,z)</code> indicates whether <code class="reqn">z</code> is located in the closed ball <code class="reqn">\bar{B}(x, \rho(x,y))</code> 
with center <code class="reqn">x</code> and radius <code class="reqn">\rho(x, y)</code>. 
We denote:
</p>
<p style="text-align: center;"><code class="reqn">
A_{ij}^{X}=\frac{1}{n}\sum_{u=1}^{n}{\delta(X_i,X_j,X_u)}, \quad A_{ij}^{Y}=\frac{1}{m}\sum_{v=1}^{m}{\delta(X_i,X_j,Y_v)},
</code>
</p>

<p style="text-align: center;"><code class="reqn">
C_{kl}^{X}=\frac{1}{n}\sum_{u=1}^{n}{\delta(Y_k,Y_l,X_u)}, \quad C_{kl}^{Y}=\frac{1}{m}\sum_{v=1}^{m}{\delta(Y_k,Y_l,Y_v)}.
</code>
</p>

<p><code class="reqn">A_{ij}^X</code> represents the proportion of samples <code class="reqn"> \{x_{1}, \ldots, x_{n}\} </code> located in the 
ball <code class="reqn">\bar{B}(X_i,\rho(X_i,X_j))</code> and <code class="reqn">A_{ij}^Y</code> represents the proportion of samples <code class="reqn"> \{y_{1}, \ldots, y_{m}\} </code> 
located in the ball <code class="reqn">\bar{B}(X_i,\rho(X_i,X_j))</code>. 
Meanwhile, <code class="reqn">C_{kl}^X</code> and <code class="reqn">C_{kl}^Y</code> represent the corresponding proportions located in the ball <code class="reqn">\bar{B}(Y_k,\rho(Y_k,Y_l))</code>.
The Ball Divergence statistic is defined as:
</p>
<p style="text-align: center;"><code class="reqn">D_{n,m}=A_{n,m}+C_{n,m}</code>
</p>

<p>Ball Divergence can be generalized to the <em>K</em>-sample test problem. Suppose we 
have <code class="reqn">K</code> group samples, each group include <code class="reqn">n_{k}</code> samples. 
The definition of <code class="reqn">K</code>-sample Ball Divergence statistic could be 
to directly sum up the two-sample Ball Divergence statistics of all sample pairs (<code>kbd.type = "sum"</code>)
</p>
<p style="text-align: center;"><code class="reqn">\sum_{1 \leq k &lt; l \leq K}{D_{n_{k},n_{l}}},</code>
</p>

<p>or to find one sample with the largest difference to the others (<code>kbd.type = "maxsum"</code>)
</p>
<p style="text-align: center;"><code class="reqn">\max_{t}{\sum_{s=1, s \neq t}^{K}{D_{n_{s}, n_{t}}},}</code>
</p>

<p>to aggregate the <code class="reqn">K-1</code> most significant different two-sample Ball Divergence statistics (<code>kbd.type = "max"</code>)
</p>
<p style="text-align: center;"><code class="reqn">\sum_{k=1}^{K-1}{D_{(k)}},</code>
</p>

<p>where <code class="reqn">D_{(1)}, \ldots, D_{(K-1)}</code> are the largest <code class="reqn">K-1</code> two-sample Ball Divergence statistics among 
<code class="reqn">\{D_{n_s, n_t}| 1 \leq s &lt; t \leq K\}</code>. When <code class="reqn">K=2</code>,
the three types of Ball Divergence statistics degenerate into two-sample Ball Divergence statistic.
</p>
<p>See <code><a href="#topic+bd.test">bd.test</a></code> for a test of distribution equality based on the Ball Divergence.
</p>


<h3>Value</h3>

<table>
<tr><td><code>bd</code></td>
<td>
<p> Ball Divergence statistic</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Wenliang Pan, Yuan Tian, Xueqin Wang, Heping Zhang
</p>


<h3>References</h3>

<p>Wenliang Pan, Yuan Tian, Xueqin Wang, Heping Zhang. Ball Divergence: Nonparametric two sample test. Ann. Statist. 46 (2018), no. 3, 1109&ndash;1137. doi:10.1214/17-AOS1579. https://projecteuclid.org/euclid.aos/1525313077
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bd.test">bd.test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>############# Ball Divergence #############
x &lt;- rnorm(50)
y &lt;- rnorm(50)
bd(x, y)
</code></pre>

<hr>
<h2 id='bd.gwas.test'>Fast K-sample Ball Divergence Test for GWAS Data</h2><span id='topic+bd.gwas.test'></span>

<h3>Description</h3>

<p>Fast K-sample Ball Divergence Test for GWAS Data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bd.gwas.test(
  x,
  snp,
  screening.method = c("permute", "spectrum"),
  refine = TRUE,
  num.permutations,
  distance = FALSE,
  alpha,
  screening.result = NULL,
  verbose = TRUE,
  seed = 1,
  num.threads = 0,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bd.gwas.test_+3A_x">x</code></td>
<td>
<p>a numeric vector, matrix, data.frame, dist object.</p>
</td></tr>
<tr><td><code id="bd.gwas.test_+3A_snp">snp</code></td>
<td>
<p>a numeric matrix recording the values of single nucleotide polymorphism (SNP). Each column must be an integer vector.</p>
</td></tr>
<tr><td><code id="bd.gwas.test_+3A_screening.method">screening.method</code></td>
<td>
<p>if <code>screening.method = "spectrum"</code>, the spectrum method is applied to
screening the candidate SNPs, or otherwise, the permutation method is applied. Default: <code>screening.method = "permute"</code>.</p>
</td></tr>
<tr><td><code id="bd.gwas.test_+3A_refine">refine</code></td>
<td>
<p>a logical value. If <code>refine = TRUE</code>, a <code class="reqn">p</code>-values refining process is applied to
the SNPs which passes the pre-screening process. Default: <code>refine = TRUE</code> (At present, <code>refine = FALSE</code> is not available).</p>
</td></tr>
<tr><td><code id="bd.gwas.test_+3A_num.permutations">num.permutations</code></td>
<td>
<p>the number of permutation replications. When <code>num.permutations = 0</code>,
the function just returns the Ball Divergence statistic. Default: <code>num.permutations = 100 * ncol(snp)</code></p>
</td></tr>
<tr><td><code id="bd.gwas.test_+3A_distance">distance</code></td>
<td>
<p>if <code>distance = TRUE</code>, the elements of <code>x</code> will be considered as a distance matrix. Default: <code>distance = FALSE</code>.</p>
</td></tr>
<tr><td><code id="bd.gwas.test_+3A_alpha">alpha</code></td>
<td>
<p>the significance level. Default: <code>0.05 / ncol(snp)</code>.</p>
</td></tr>
<tr><td><code id="bd.gwas.test_+3A_screening.result">screening.result</code></td>
<td>
<p>A object return by <code>bd.gwas.test</code> that
preserving the pre-screening result.
It works only if the pre-screening is available.
Default: <code>screening.result = NULL</code>.</p>
</td></tr>
<tr><td><code id="bd.gwas.test_+3A_verbose">verbose</code></td>
<td>
<p>Show computation status and estimated runtimes. Default: <code>verbose = FALSE</code>.</p>
</td></tr>
<tr><td><code id="bd.gwas.test_+3A_seed">seed</code></td>
<td>
<p>the random seed. Default <code>seed = 1</code>.</p>
</td></tr>
<tr><td><code id="bd.gwas.test_+3A_num.threads">num.threads</code></td>
<td>
<p>number of threads. If <code>num.threads = 0</code>, then all of available cores will be used. Default <code>num.threads = 0</code>.</p>
</td></tr>
<tr><td><code id="bd.gwas.test_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to or from methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>bd.gwas.test returns a list containing the following components:
</p>
<table>
<tr><td><code>statistic</code></td>
<td>
<p>ball divergence statistics vector.</p>
</td></tr>
<tr><td><code>permuted.statistic</code></td>
<td>
<p>a data.frame containing permuted ball divergence statistic for pre-screening SNPs.
If <code>refine = FALSE</code>, it takes value <code>NULL</code>.</p>
</td></tr>
<tr><td><code>eigenvalue</code></td>
<td>
<p>the eigenvalue of spectrum decomposition. If <code>refine = TRUE</code>, it takes value <code>NULL</code>.</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>the p-values of ball divergence test.</p>
</td></tr>
<tr><td><code>refined.snp</code></td>
<td>
<p>the SNPs have been refined.</p>
</td></tr>
<tr><td><code>refined.p.value</code></td>
<td>
<p>the refined <code class="reqn">p</code>-value of significant snp.</p>
</td></tr>
<tr><td><code>refined.permuted.statistic</code></td>
<td>
<p>a data.frame containing permuted ball divergence statistics for refining <code class="reqn">p</code>-values.</p>
</td></tr>
<tr><td><code>screening.result</code></td>
<td>
<p>a list containing the result of screening.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jin Zhu
</p>


<h3>References</h3>

<p>Yue Hu, Haizhu Tan, Cai Li, and Heping Zhang. (2021). Identifying genetic risk variants associated with brain volumetric phenotypes via K-sample Ball Divergence method. Genetic Epidemiology, 1â€“11. https://doi.org/10.1002/gepi.22423
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bd">bd</a></code>, <code><a href="#topic+bd.test">bd.test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(Ball)
set.seed(1234)
num &lt;- 200
snp_num &lt;- 500
p &lt;- 5
x &lt;- matrix(rnorm(num * p), nrow = num)
snp &lt;- sapply(1:snp_num, function(i) {
  sample(0:2, size = num, replace = TRUE)
})
snp1 &lt;- sapply(1:snp_num, function(i) {
  sample(1:2, size = num, replace = TRUE)
})
snp &lt;- cbind(snp, snp1)
res &lt;- Ball::bd.gwas.test(x = x, snp = snp)
mean(res[["p.value"]] &lt; 0.05)
mean(res[["p.value"]] &lt; 0.005)

## only return the test statistics;
res &lt;- Ball::bd.gwas.test(x = x, snp = snp, num.permutation = 0)

## save pre-screening process results:
x &lt;- matrix(rnorm(num * p), nrow = num)
snp &lt;- sapply(1:snp_num, function(i) {
  sample(0:2, size = num, replace = TRUE, prob = c(1/2, 1/4, 1/4))
})
snp_screening &lt;- Ball::bd.gwas.test(x = x, snp = snp,
                                    alpha = 5*10^-4, 
                                    num.permutations = 19999)
mean(res[["p.value"]] &lt; 0.05)
mean(res[["p.value"]] &lt; 0.005)
mean(res[["p.value"]] &lt; 0.0005)
## refine p-value according to the pre-screening process result:
res &lt;- Ball::bd.gwas.test(x = x, snp = snp, alpha = 5*10^-4,
                          num.permutations = 19999,
                          screening.result = snp_screening[["screening.result"]])

</code></pre>

<hr>
<h2 id='bd.test'>Ball Divergence based Equality of Distributions Test</h2><span id='topic+bd.test'></span><span id='topic+bd.test.default'></span><span id='topic+bd.test.formula'></span>

<h3>Description</h3>

<p>Performs the nonparametric two-sample or <code class="reqn">K</code>-sample Ball Divergence test for
equality of multivariate distributions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bd.test(x, ...)

## Default S3 method:
bd.test(
  x,
  y = NULL,
  num.permutations = 99,
  method = c("permutation", "limit"),
  distance = FALSE,
  size = NULL,
  seed = 1,
  num.threads = 0,
  kbd.type = c("sum", "maxsum", "max"),
  weight = c("constant", "variance"),
  ...
)

## S3 method for class 'formula'
bd.test(formula, data, subset, na.action, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bd.test_+3A_x">x</code></td>
<td>
<p>a numeric vector, matrix, data.frame, or a list containing at least two numeric vectors, matrices, or data.frames.</p>
</td></tr>
<tr><td><code id="bd.test_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to or from methods.</p>
</td></tr>
<tr><td><code id="bd.test_+3A_y">y</code></td>
<td>
<p>a numeric vector, matrix, data.frame.</p>
</td></tr>
<tr><td><code id="bd.test_+3A_num.permutations">num.permutations</code></td>
<td>
<p>the number of permutation replications. When <code>num.permutations = 0</code>, the function just returns
the Ball Divergence statistic. Default: <code>num.permutations = 99</code>.</p>
</td></tr>
<tr><td><code id="bd.test_+3A_method">method</code></td>
<td>
<p>if <code>method = "permutation"</code>, a permutation procedure is carried out to compute the <code class="reqn">p</code>-value;
if <code> method = "limit"</code>, an approximate null distribution is used when <code>weight = "constant"</code>.
Any unambiguous substring can be given. Default <code>method = "permutation"</code>.</p>
</td></tr>
<tr><td><code id="bd.test_+3A_distance">distance</code></td>
<td>
<p>if <code>distance = TRUE</code>, the elements of <code>x</code> will be considered as a distance matrix. Default: <code>distance = FALSE</code>.</p>
</td></tr>
<tr><td><code id="bd.test_+3A_size">size</code></td>
<td>
<p>a vector recording sample size of each group.</p>
</td></tr>
<tr><td><code id="bd.test_+3A_seed">seed</code></td>
<td>
<p>the random seed. Default <code>seed = 1</code>.</p>
</td></tr>
<tr><td><code id="bd.test_+3A_num.threads">num.threads</code></td>
<td>
<p>number of threads. If <code>num.threads = 0</code>, then all of available cores will be used. Default <code>num.threads = 0</code>.</p>
</td></tr>
<tr><td><code id="bd.test_+3A_kbd.type">kbd.type</code></td>
<td>
<p>a character string specifying the <code class="reqn">K</code>-sample Ball Divergence test statistic, 
must be one of <code>"sum"</code>, <code>"summax"</code>, or <code>"max"</code>. Any unambiguous substring can be given. 
Default <code>kbd.type = "sum"</code>.</p>
</td></tr>
<tr><td><code id="bd.test_+3A_weight">weight</code></td>
<td>
<p>a character string specifying the weight form of Ball Divergence statistic.
It must be one of <code>"constant"</code> or <code>"variance"</code>. 
Any unambiguous substring can be given. Default: <code>weight = "constant"</code>.</p>
</td></tr>
<tr><td><code id="bd.test_+3A_formula">formula</code></td>
<td>
<p>a formula of the form <code>response ~ group</code> where <code>response</code> gives the data values and <code>group</code> a vector or factor of the corresponding groups.</p>
</td></tr>
<tr><td><code id="bd.test_+3A_data">data</code></td>
<td>
<p>an optional matrix or data frame (or similar: see <code>model.frame</code>) containing the variables in the formula <code>formula</code>. By default the variables are taken from <code>environment(formula)</code>.</p>
</td></tr>
<tr><td><code id="bd.test_+3A_subset">subset</code></td>
<td>
<p>an optional vector specifying a subset of observations to be used.</p>
</td></tr>
<tr><td><code id="bd.test_+3A_na.action">na.action</code></td>
<td>
<p>a function which indicates what should happen when the data contain <code>NA</code>s. Defaults to <code>getOption("na.action")</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>bd.test</code> is nonparametric test for the two-sample or <code class="reqn">K</code>-sample problem. 
It can detect distribution difference between <code class="reqn">K(K \geq 2)</code> sample even though sample size are imbalanced.
This test can cope well multivariate dataset or complex dataset. 
</p>
<p>If only <code>x</code> is given, the statistic is 
computed from the original pooled samples, stacked in 
matrix where each row is a multivariate observation, or from the distance matrix 
when <code>distance = TRUE</code>. The first <code>sizes[1]</code> rows of <code>x</code> are the first sample, the next 
<code>sizes[2]</code> rows of <code>x</code> are the second sample, etc.
If <code>x</code> is a <code>list</code>, its elements are taken as the samples to be compared, 
and hence, this <code>list</code> must contain at least two numeric data vectors, matrices or data.frames.
</p>
<p><code>bd.test</code> utilizes the Ball Divergence statistics (see <code><a href="#topic+bd">bd</a></code>) to measure dispersion and 
derives a <code class="reqn">p</code>-value via replicating the random permutation <code>num.permutations</code> times. 
The function simply returns the test statistic 
when <code>num.permutations = 0</code>. 
</p>
<p>The time complexity of <code>bd.test</code> is around <code class="reqn">O(R \times n^2)</code>,
where <code class="reqn">R</code> = <code>num.permutations</code> and <code class="reqn">n</code> is sample size.
</p>


<h3>Value</h3>

<p>If <code>num.permutations &gt; 0</code>, <code>bd.test</code> returns a <code>htest</code> class object containing the following components:
</p>
<table>
<tr><td><code>statistic</code></td>
<td>
<p>Ball Divergence statistic.</p>
</td></tr>            
<tr><td><code>p.value</code></td>
<td>
<p>the <code class="reqn">p</code>-value for the test.</p>
</td></tr>
<tr><td><code>replicates</code></td>
<td>
<p>permutation replications of the test statistic.</p>
</td></tr>
<tr><td><code>size</code></td>
<td>
<p>sample sizes.</p>
</td></tr>
<tr><td><code>complete.info</code></td>
<td>
<p>a <code>list</code> mainly containing two vectors, the first vector is the Ball Divergence statistics 
with different aggregation strategy and weight, the second vector is the <code class="reqn">p</code>-values of tests.</p>
</td></tr>
<tr><td><code>alternative</code></td>
<td>
<p>a character string describing the alternative hypothesis.</p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p>a character string indicating what type of test was performed.</p>
</td></tr>
<tr><td><code>data.name</code></td>
<td>
<p>description of data.</p>
</td></tr>
</table>
<p>If <code>num.permutations = 0</code>, <code>bd.test</code> returns a statistic value.
</p>


<h3>Note</h3>

<p>Actually, <code>bd.test</code> simultaneously computing <code>"sum"</code>, <code>"summax"</code>, and <code>"max"</code> Ball Divergence statistics 
when <code class="reqn">K \geq 3</code>.
Users can get other Ball Divergence statistics and their corresponding <code class="reqn">p</code>-values 
in the <code>complete.info</code> element of output. We give a quick example below to illustrate.
</p>


<h3>Author(s)</h3>

<p>Wenliang Pan, Yuan Tian, Xueqin Wang, Heping Zhang, Jin Zhu
</p>


<h3>References</h3>

<p>Wenliang Pan, Yuan Tian, Xueqin Wang, Heping Zhang. Ball Divergence: Nonparametric two sample test. Annals of Statistics. 46 (2018), no. 3, 1109&ndash;1137. doi:10.1214/17-AOS1579. https://projecteuclid.org/euclid.aos/1525313077
</p>
<p>Jin Zhu, Wenliang Pan, Wei Zheng, and Xueqin Wang (2021). Ball: An R Package for Detecting Distribution Difference and Association in Metric Spaces, Journal of Statistical Software, Vol.97(6), doi: 10.18637/jss.v097.i06.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bd">bd</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>################# Quick Start #################
set.seed(1)
x &lt;- rnorm(50)
y &lt;- rnorm(50, mean = 1)
# plot(density(x))
# lines(density(y), col = "red")
bd.test(x = x, y = y)

################# Quick Start #################
x &lt;- matrix(rnorm(100), nrow = 50, ncol = 2)
y &lt;- matrix(rnorm(100, mean = 3), nrow = 50, ncol = 2)
# Hypothesis test with Standard Ball Divergence:
bd.test(x = x, y = y)

################# Simlated Non-Hilbert data #################
data("bdvmf")
## Not run: 
library(scatterplot3d)
scatterplot3d(bdvmf[["x"]], color = bdvmf[["group"]], 
              xlab = "X1", ylab = "X2", zlab = "X3")

## End(Not run)
# calculate geodesic distance between sample:
Dmat &lt;- nhdist(bdvmf[["x"]], method = "geodesic")
# hypothesis test with BD :
bd.test(x = Dmat, size = c(150, 150), num.permutations = 99, distance = TRUE)

################# Non-Hilbert Real Data #################
# load data:
data("macaques")
# number of femala and male Macaca fascicularis:
table(macaques[["group"]])
# calculate Riemannian shape distance matrix:
Dmat &lt;- nhdist(macaques[["x"]], method = "riemann")
# hypothesis test with BD:
bd.test(x = Dmat, num.permutations = 99, size = c(9, 9), distance = TRUE)

################  K-sample Test  #################
n &lt;- 150
bd.test(rnorm(n), size = c(40, 50, 60))
# alternative input method:
x &lt;- lapply(c(40, 50, 60), rnorm)
res &lt;- bd.test(x)
res
## get all Ball Divergence statistics:
res[["complete.info"]][["statistic"]]
## get all test result:
res[["complete.info"]][["p.value"]]

################  Testing via approximate limit distribution  #################
## Not run: 
set.seed(1)
n &lt;- 1000
x &lt;- rnorm(n)
y &lt;- rnorm(n)
res &lt;- bd.test(x, y, method = "limit")
bd.test(x, y)

## End(Not run)

################  Formula interface  ################
## Two-sample test
bd.test(extra ~ group, data = sleep)
## K-sample test
bd.test(Sepal.Width ~ Species, data = iris)
bd.test(Sepal.Width ~ Species, data = iris, kbd.type = "max")
</code></pre>

<hr>
<h2 id='bdvmf'>Simulated von Mises-Fisher Data</h2><span id='topic+bdvmf'></span>

<h3>Description</h3>

<p>Simulated random vectors following the von Mises-Fisher distribution 
with mean direction <code class="reqn">\mu_{x}=(1, 0, 0)</code> and <code class="reqn">\mu_{y}=(1, 1, 1)</code>, 
and concentration parameter is <code class="reqn">\kappa = 3</code>.
</p>


<h3>Format</h3>


<p><code>bdvmf$x</code>: A <code class="reqn">300 \times 3</code> numeric matrix containing simulated von Mises-Fisher data. 
</p>
<p><code>bdvmf$group</code>: A group index vector.

</p>


<h3>Details</h3>

<p>In directional statistics, the von Misesâ€“Fisher distribution 
(named after Ronald Fisher and Richard von Mises), is a probability distribution
on the <code class="reqn">(p-1)</code>-dimensional sphere in <code class="reqn">R^{p}</code>
</p>
<p>The parameters <code class="reqn">\mu</code>, and <code class="reqn">\kappa</code>, are called the mean direction and concentration
parameter, respectively. The greater the value of <code class="reqn">\kappa</code>, 
the higher the concentration of the distribution around the mean 
direction <code class="reqn">\mu</code>,. The distribution is unimodal for <code class="reqn">\kappa</code>, 
and is uniform on the sphere for <code class="reqn">\kappa=0</code>.
</p>


<h3>References</h3>

<p>Embleton, N. I. Fisher, T. Lewis, B. J. J. (1993). Statistical analysis of spherical data (1st pbk. ed.). Cambridge: Cambridge University Press. pp. 115â€“116. ISBN 0-521-45699-1.
</p>

<hr>
<h2 id='genlung'>Lung cancer genomic data</h2><span id='topic+genlung'></span>

<h3>Description</h3>

<p>Publicly available lung cancer genomic data from the Chemores Cohort Study, containing 
the expression levels of mRNA, miRNA, artificial noise variables as well as clinical variables and response.
</p>


<h3>Format</h3>


<p><code>genlung$survival</code>: A data.frame containing <code class="reqn">n=123</code> complete observations. 
The first column is disease-free survival time and the 
second column is censoring status. 
</p>
<p><code>genlung$covariate</code>: A data.frame containing <code class="reqn">p=2000</code> covariates.

</p>


<h3>Details</h3>

<p>Tissue samples were analysed from a cohort of 123 patients, who underwent complete surgical resection at the Institut Mutualiste
Montsouris (Paris, France) between 30 January 2002 and 26 June 2006. The studied outcome was the &quot;Disease-Free Survival Time&quot;.
Patients were followed until the first relapse occurred or administrative censoring. In this genomic dataset, 
the expression levels of Agilent miRNA probes (<code class="reqn">p=939</code>) were included from the <code class="reqn">n=123</code> cohort samples. 
The miRNA data contains normalized expression levels. See below the paper by Lazar et al. (2013) and Array Express 
data repository for the complete description of the samples, tissue preparation, Agilent array technology, and data normalization. 
In addition to the genomic data, five clinical variables, also evaluated on the cohort samples, are included as 
continuous variable ('Age') and nominal variables ('Type','KRAS.status','EGFR.status','P53.status'). 
See Lazar et al. (2013) for more details. Moreover, we add 1056 standard gaussian variables 
which are independent with the censored response as noise covariates. This dataset represents a situation where the number of 
covariates dominates the number of complete observations or <code class="reqn">p &gt;&gt; n</code> case.
</p>


<h3>References</h3>

<p>Lazar V. et al. (2013). Integrated molecular portrait of non-small cell lung cancers. BMC Medical Genomics 6:53-65.
</p>

<hr>
<h2 id='macaques'>Male and Female macaque data</h2><span id='topic+macaques'></span>

<h3>Description</h3>

<p>Male and female macaque skull data. 7 landmarks in 3 dimensions, 
18 individuals (9 males, 9 females)
</p>


<h3>Format</h3>


<p><code>macaques$x</code>: An array of dimension <code class="reqn">7 \times 3 \times 18</code>
</p>
<p><code>macaques$group</code>: A factor indicating the sex ('m' for male and 'f' for female)

</p>


<h3>Details</h3>

<p>In an investigation into sex differences in the crania of a species of 
Macaca fascicularis (a type of monkey), random samples of 9 male and 9 female 
skulls were obtained by Paul Oâ€™Higgins (Hull-York Medical School) (Dryden and Mardia 1993). 
A subset of seven anatomical landmarks was located on each cranium and the three-dimensional (3D) 
coordinates of each point were recorded.
</p>


<h3>Note</h3>

<p>Dryden, I.L. and Mardia, K.V. (1998). Statistical Shape Analysis, Wiley, Chichester.
</p>


<h3>References</h3>

<p>Dryden, I. L. and Mardia, K. V. (1993). Multivariate shape analysis. Sankhya Series A, 55, 460-480.
</p>

<hr>
<h2 id='meteorology'>meteorological data</h2><span id='topic+meteorology'></span>

<h3>Description</h3>

<p>A meteorological data include 46 records about air, soil, humidity, wind and evaporation.
</p>


<h3>Format</h3>


<p><code>meteorology$air</code>: A data.frame containing 3 variables: maximum, minimum and average daily air temperature
</p>
<p><code>meteorology$soil</code>: A data.frame containing 3 covariates: maximum, minimum and average daily soil temperature
</p>
<p><code>meteorology$humidity</code>: A data.frame containing 3 covariates: maximum, minimum and average daily humidity temperature, 
</p>
<p><code>meteorology$wind</code>: a vector object record total wind, measured in miles per day
</p>
<p><code>meteorology$evaporation</code>: a vector object record evaporation

</p>


<h3>Details</h3>

<p>This meteorological data containing 46 observations on five groups of variables: air temperature, soil temperature, 
relative humidity, wind speed as well as evaporation. 
Among them, maximum, minimum and average value for air temperature, soil temperature, and 
relative humidity are recorded. As regards to wind speed and evaporation, 
there are univariate numerical variables. We desire to test the independence of these five 
groups of variables.
</p>

<hr>
<h2 id='nhdist'>Distance Matrix Computation for Non-Hilbert Data</h2><span id='topic+nhdist'></span>

<h3>Description</h3>

<p>This function computes and returns the numeric distance matrix 
computed by using the specified distance measure to compute 
the distances between the rows of a data matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nhdist(x, method = "geodesic")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nhdist_+3A_x">x</code></td>
<td>
<p>a numeric matrix, data frame or numeric array of dimension <code class="reqn">k \times m \times n</code> 
containing <code class="reqn">n</code> samples in <code class="reqn">k \times m</code> dimension.</p>
</td></tr>
<tr><td><code id="nhdist_+3A_method">method</code></td>
<td>
<p>the distance measure to be used. This must be one of <code>"geodesic"</code>, <code>"compositional"</code>, or <code>"riemann"</code>. 
Any unambiguous substring can be given.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Available distance measures are geodesic, compositional and riemann.
Denoting any two sample in the dataset as <code class="reqn">x</code> and <code class="reqn">y</code>, 
we give the definition of distance measures as follows.
</p>
<p>geodesic:
</p>
<p>The shortest route between two points on the Earth's surface, namely, a segment of a great circle.
</p>
<p style="text-align: center;"><code class="reqn">\arccos(x^{T}y), \|x\|_{2} = \|y\|_{2} = 1</code>
</p>

<p>compositional:
</p>
<p>First, we apply scale transformation to it, i.e., <code class="reqn">(x_{i1}/t, ..., x_{ip}/t_{i}), t_{i} = \sum_{d=1}^{p}{x_{d}}</code> 
. Then, apply the square root transformation to data and calculate the geodesic distance between samples.
</p>
<p>riemann:
</p>
<p><code class="reqn">k \times m \times n</code> array where <code class="reqn">k</code> = number of landmarks, <code class="reqn">m</code> = number of dimensions and <code class="reqn">n</code> = sample size. Detail about
riemannian shape distance was given in Kendall, D. G. (1984).
</p>


<h3>Value</h3>

<p><code class="reqn">n \times n</code> numeric distance matrix
</p>


<h3>References</h3>

<p>Kendall, D. G. (1984). Shape manifolds, Procrustean metrics and complex projective spaces, Bulletin of the London Mathematical Society, 16, 81-121.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('bdvmf')
Dmat &lt;- nhdist(bdvmf[['x']], method = "geodesic")

data("ArcticLake")
Dmat &lt;- nhdist(ArcticLake[['x']], method = "compositional")

data("macaques")
Dmat &lt;- nhdist(macaques[["x"]], method = "riemann")

# unambiguous substring also available:
Dmat &lt;- nhdist(macaques[["x"]], method = "rie")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
