<!DOCTYPE html><html><head><title>Help for package BALLI</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BALLI}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#balli'><p>BALLI</p></a></li>
<li><a href='#Balli-class'><p>Class Balli</p>
Class <code>Balli</code> holds results from BALLI</a></li>
<li><a href='#balliFit'><p>balliFit</p></a></li>
<li><a href='#LargeDataObject-class'><p>Class LargeDataObject</p>
Class <code>LargeDataObject</code> holds large data such as technical variance and results from BALLI fit</a></li>
<li><a href='#tecVarEstim'><p>Technical Variance Estimation</p></a></li>
<li><a href='#TecVarList-class'><p>Class TecVarList</p>
Class <code>TecVarList</code> holds technical variance</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Expression RNA-Seq Data Analysis Based on Linear Mixed Model</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Kyungtaek Park &lt;qkrrudxor147@snu.ac.kr&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kyungtaek Park &lt;qkrrudxor147@snu.ac.kr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Analysis of gene expression RNA-seq data using Bartlett-Adjusted Likelihood-based LInear model (BALLI). Based on likelihood ratio test, it provides comparisons for effect of one or more variables. See Kyungtaek Park (2018) &lt;<a href="https://doi.org/10.1101%2F344929">doi:10.1101/344929</a>&gt; for more information. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.15.0), edgeR, limma, MASS, parallel, stats, methods</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-04-24 13:21:01 UTC; KyungtaekPark</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-04-25 10:40:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='balli'>BALLI</h2><span id='topic+balli'></span>

<h3>Description</h3>

<p>DEG analysis using BALLI algorithm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>balli(object, intV = 2, logcpm = NULL, tecVar = NULL,
  design = NULL, numCores = NULL, threshold = 1e-06, maxiter = 200)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="balli_+3A_object">object</code></td>
<td>
<p>a TecVarList object</p>
</td></tr>
<tr><td><code id="balli_+3A_intv">intV</code></td>
<td>
<p>numeric vector designating interest variable(s) which is(are) column number(s) of design matrix</p>
</td></tr>
<tr><td><code id="balli_+3A_logcpm">logcpm</code></td>
<td>
<p>logcpm values for each gene and each sample</p>
</td></tr>
<tr><td><code id="balli_+3A_tecvar">tecVar</code></td>
<td>
<p>estimated technical variance values for each gene and each sample</p>
</td></tr>
<tr><td><code id="balli_+3A_design">design</code></td>
<td>
<p>design matrix with samples in row and covariable(s) to be estimated in column</p>
</td></tr>
<tr><td><code id="balli_+3A_numcores">numCores</code></td>
<td>
<p>number of cores to be used for multithreding. If NULL, a single core is used</p>
</td></tr>
<tr><td><code id="balli_+3A_threshold">threshold</code></td>
<td>
<p>threshold for convergence</p>
</td></tr>
<tr><td><code id="balli_+3A_maxiter">maxiter</code></td>
<td>
<p>maximum number of iteration to converge of estimated biological variance. If not, biological variance is estimated by using Brent method</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an Balli object including Result and topGenes list. Following components are shown by Result (same order of genes with input data) and topGenes (ordered by pBALLI in Result) :
</p>
<table>
<tr><td><code>log2FC</code></td>
<td>
<p>log2 fold changes of interest variable(s)</p>
</td></tr>
<tr><td><code>lLLI</code></td>
<td>
<p>log-likelihoods estimated by LLI</p>
</td></tr>
<tr><td><code>lBALLI</code></td>
<td>
<p>log-likelihoods estimated by BALLI</p>
</td></tr>
<tr><td><code>pLLI</code></td>
<td>
<p>p-values estimated by LLI</p>
</td></tr>
<tr><td><code>pBALLI</code></td>
<td>
<p>p-values estimated by BALLI</p>
</td></tr>
<tr><td><code>BCF</code></td>
<td>
<p>Bartlett's correction factor</p>
</td></tr>
</table>
<p>expr &lt;- data.frame(t(sapply(1:1000,function(x)rnbinom(20,mu=500,size=50))))
group &lt;- c(rep(&quot;A&quot;,10),rep(&quot;B&quot;,10))
design &lt;- model.matrix(~group, data = expr)
dge &lt;- DGEList(counts=expr, group=group)
dge &lt;- calcNormFactors(dge)
tV &lt;- tecVarEstim(dge,design)
balli(tV,intV=2)
</p>

<hr>
<h2 id='Balli-class'>Class Balli
Class <code>Balli</code> holds results from BALLI</h2><span id='topic+Balli-class'></span>

<h3>Description</h3>

<p>Class Balli
Class <code>Balli</code> holds results from BALLI
</p>

<hr>
<h2 id='balliFit'>balliFit</h2><span id='topic+balliFit'></span>

<h3>Description</h3>

<p>Estimates likelihood and Bartlett correction factor using BALLI algorithm of each gene
</p>


<h3>Usage</h3>

<pre><code class='language-R'>balliFit(y_mat, x_mat, tecVar, intVar = 2, full = T, cfault = 0,
  miter = 200, conv = 1e-06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="balliFit_+3A_y_mat">y_mat</code></td>
<td>
<p>numeric vector containing log-cpm values of each gene and each sample</p>
</td></tr>
<tr><td><code id="balliFit_+3A_x_mat">x_mat</code></td>
<td>
<p>design matrix with samples in row and covariable(s) to be estimated in column</p>
</td></tr>
<tr><td><code id="balliFit_+3A_tecvar">tecVar</code></td>
<td>
<p>numeric vector containing estimated technical variance of a gene of each sample</p>
</td></tr>
<tr><td><code id="balliFit_+3A_intvar">intVar</code></td>
<td>
<p>numeric vector designating interest variable(s) which is(are) column number(s) of x_mat</p>
</td></tr>
<tr><td><code id="balliFit_+3A_full">full</code></td>
<td>
<p>logical value designating full model (TRUE) or reduced model (FALSE).</p>
</td></tr>
<tr><td><code id="balliFit_+3A_cfault">cfault</code></td>
<td>
<p>initial value of index showing whether converged (0) or not (1).</p>
</td></tr>
<tr><td><code id="balliFit_+3A_miter">miter</code></td>
<td>
<p>maximum number of iteration to converge.</p>
</td></tr>
<tr><td><code id="balliFit_+3A_conv">conv</code></td>
<td>
<p>threshold for convergence</p>
</td></tr>
</table>


<h3>Value</h3>

<p>following components are estimated
</p>
<table>
<tr><td><code>ll</code></td>
<td>
<p>log-likelihoods</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>coefficients of interested variable(s)</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>coefficients of nuisance variable(s)</p>
</td></tr>
<tr><td><code>BCF</code></td>
<td>
<p>Bartlett's correction factor</p>
</td></tr>
<tr><td><code>cfault</code></td>
<td>
<p>index whether converged or not</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>expr &lt;- data.frame(t(sapply(1:1000,function(x)rnbinom(20,mu=500,size=50))))
group &lt;- c(rep("A",10),rep("B",10))
design &lt;- model.matrix(~group, data = expr)
dge &lt;- DGEList(counts=expr, group=group)
dge &lt;- calcNormFactors(dge)
tV &lt;- tecVarEstim(dge,design)
gtv &lt;- tV$tecVar[1,]
gdat &lt;- data.frame(logcpm=tV$logcpm[1,],design,tecVar=gtv)
gy &lt;- matrix(unlist(gdat[,1]),ncol=1)
gx &lt;- matrix(unlist(gdat[,2:(ncol(gdat)-1)]),ncol=ncol(gdat)-2)
balliFit(y_mat=gy,x_mat=gx,tecVar=gtv,intVar=2,full=TRUE,cfault=0,miter=200,conv=1e-6)
</code></pre>

<hr>
<h2 id='LargeDataObject-class'>Class LargeDataObject
Class <code>LargeDataObject</code> holds large data such as technical variance and results from BALLI fit</h2><span id='topic+LargeDataObject-class'></span>

<h3>Description</h3>

<p>Class LargeDataObject
Class <code>LargeDataObject</code> holds large data such as technical variance and results from BALLI fit
</p>

<hr>
<h2 id='tecVarEstim'>Technical Variance Estimation</h2><span id='topic+tecVarEstim'></span>

<h3>Description</h3>

<p>Estimate technical variance by using voom-trend. The code is derived from voom function in limma package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tecVarEstim(counts, design = NULL, lib.size = NULL, span = 0.5, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tecVarEstim_+3A_counts">counts</code></td>
<td>
<p>a DGEList object</p>
</td></tr>
<tr><td><code id="tecVarEstim_+3A_design">design</code></td>
<td>
<p>design matrix with samples in row and coefficient(s) to be estimated in column</p>
</td></tr>
<tr><td><code id="tecVarEstim_+3A_lib.size">lib.size</code></td>
<td>
<p>numeric vector containing total library sizes for each sample</p>
</td></tr>
<tr><td><code id="tecVarEstim_+3A_span">span</code></td>
<td>
<p>width of the lowess smoothing window as a proportion</p>
</td></tr>
<tr><td><code id="tecVarEstim_+3A_...">...</code></td>
<td>
<p>other arguments are passed to lmFit.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an TecVarList object with the following components:
</p>
<table>
<tr><td><code>targets</code></td>
<td>
<p>matrix containing covariables, library sizes and normalization foctors of each sample</p>
</td></tr>
<tr><td><code>design</code></td>
<td>
<p>design matrix with samples in row and covariable(s) to be estimated in column</p>
</td></tr>
<tr><td><code>logcpm</code></td>
<td>
<p>logcpm values of each gene and each sample</p>
</td></tr>
<tr><td><code>tecVar</code></td>
<td>
<p>estimated techical variance of each gene and each sample</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>expr &lt;- data.frame(t(sapply(1:1000,function(x)rnbinom(20,mu=500,size=50))))
group &lt;- c(rep("A",10),rep("B",10))
design &lt;- model.matrix(~group, data = expr)
dge &lt;- DGEList(counts=expr, group=group)
dge &lt;- calcNormFactors(dge)
tecVarEstim(dge,design)
</code></pre>

<hr>
<h2 id='TecVarList-class'>Class TecVarList
Class <code>TecVarList</code> holds technical variance</h2><span id='topic+TecVarList-class'></span>

<h3>Description</h3>

<p>Class TecVarList
Class <code>TecVarList</code> holds technical variance
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
