<!DOCTYPE html><html><head><title>Help for package bamm</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {bamm}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#adj_mat'><p>adj_mat: Function to compute the adjacency matrix of an area.</p></a></li>
<li><a href='#bam_clusters'><p>bam_clusters: Function to estimate the connectivity of suitable areas</p></a></li>
<li><a href='#bam_sim'><p>bam_sim: Simulate dispersal dynamics using the set B of the BAM framework.</p></a></li>
<li><a href='#bam_ssim'><p>bam_ssim: Simulate dispersal dynamics using the set B of the BAM framework.</p></a></li>
<li><a href='#bam-class'><p>Class <code>bam</code> digram</p></a></li>
<li><a href='#bioindex_sparse-class'><p>Class  <code>bioindex_sparse</code></p></a></li>
<li><a href='#bioindex-class'><p>Class  <code>bioindex</code></p></a></li>
<li><a href='#community_sim'><p>community_bam: Community <code>bamm</code></p></a></li>
<li><a href='#community_sim-class'><p>Class  <code>community_sim</code> digram</p></a></li>
<li><a href='#csd_estimate'><p>csd_estimate: Estimate the connectivity suitability and dispersal plot</p></a></li>
<li><a href='#csd-class'><p>Class  <code>csd</code></p></a></li>
<li><a href='#csim2pam'><p>csim2pam: Converts community simulation to a Presence Absence Matrix (PAM)</p></a></li>
<li><a href='#diversity_range_analysis'><p>range_diversity_analysis: diversity analysis</p></a></li>
<li><a href='#diversity_range-class'><p>Class  <code>diversity_range</code></p></a></li>
<li><a href='#eigen_bam'><p>eigen_bam: Compute the Eigen system of two bam objects</p></a></li>
<li><a href='#g_area-class'><p>S4 classes to organize data and results of <code>bamm</code> objects</p></a></li>
<li><a href='#jaccard'><p>jaccard: Estimates the Jaccard index for comparing two binary maps</p></a></li>
<li><a href='#leaflet-class'><p>Class  <code>leaflet</code> leaflet</p></a></li>
<li><a href='#model2sparse'><p>model2sparse: Converts a niche model into a diagonal sparse matrix</p></a></li>
<li><a href='#models2pam'><p>models2pam: Converts binary rasters to a PAM</p></a></li>
<li><a href='#null_dispersion_field_distribution'><p>null_dispersion_field_distribution: Null distribution of the</p>
dispersion field</a></li>
<li><a href='#occs2sparse'><p>occs2sparse: Converts occurrence data into a sparse matrix object</p></a></li>
<li><a href='#pam-class'><p>Class  <code>pam</code> Presence-Absence Matrix</p></a></li>
<li><a href='#pam2bioindex'><p>pam2bioindex: PAM to biodiversity index</p></a></li>
<li><a href='#pam2richness'><p>pam2richness: Converts Presence Absence Matrix (pam object) to</p>
richness raster</a></li>
<li><a href='#permute_pam'><p>permute_pam: Function to permute a Presence-Absence-Matrix.</p></a></li>
<li><a href='#plot+2Cdiversity_range+2CANY-method'><p>Plot method for objects of class diversity_range <span class="pkg">bamm</span>.</p></a></li>
<li><a href='#predict+2Cbam-method'><p>Predict method of the package <span class="pkg">bamm</span>.</p></a></li>
<li><a href='#sdm_sim'><p>sdm_sim: Simulate single species dispersal dynamics using the BAM framework.</p></a></li>
<li><a href='#setA-class'><p>Class for the A set of the BAM diagram</p></a></li>
<li><a href='#setM-class'><p>Class for the M set of the <code>bamm</code> diagram</p></a></li>
<li><a href='#shape2Grid'><p>shape2Grid: Function to create a grid given a spatial polygon</p></a></li>
<li><a href='#show+2CsetA-method'><p>Show information in setA class <span class="pkg">bamm</span>.</p></a></li>
<li><a href='#sim2Animation'><p>sim2Animation: Animate BAM simulation object.</p></a></li>
<li><a href='#sim2Raster'><p>sim2Raster: Convert a BAM simulation object to RasterStack</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Species Distribution Models as a Function of Biotic, Abiotic and
Movement Factors (BAM)</td>
</tr>
<tr>
<td>Version:</td>
<td>0.5.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Luis Osorio-Olvera &lt;luismurao@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Species Distribution Modeling (SDM) is a practical
    methodology that aims to estimate the area of distribution of a
    species. However, most of the work has focused on estimating static
    expressions of the correlation between environmental variables. The
    outputs of correlative species distribution models can be interpreted
    as maps of the suitable environment for a species but not generally as
    maps of its actual distribution.  Soberón and Peterson (2005)
    &lt;<a href="https://doi.org/10.17161%2Fbi.v2i0.4">doi:10.17161/bi.v2i0.4</a>&gt; presented the BAM scheme, a heuristic
    framework that states that the occupied area of a species occurs on
    sites that have been accessible through dispersal (M) and have both
    favorable biotic (B) and abiotic conditions (A).  The 'bamm' package
    implements classes and functions to operate on each element of the BAM
    and by using a cellular automata model where the occupied area of a
    species at time t is estimated by the multiplication of three binary
    matrices: one matrix represents movements (M), another abiotic -niche-
    tolerances (A), and a third, biotic interactions (B).  The theoretical
    background of the package can be found in Soberón and Osorio-Olvera
    (2023) &lt;<a href="https://doi.org/10.1111%2Fjbi.14587">doi:10.1111/jbi.14587</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://luismurao.github.io/bamm/">https://luismurao.github.io/bamm/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/luismurao/bamm/issues">https://github.com/luismurao/bamm/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>animation (&ge; 2.3), crosstalk, dplyr (&ge; 0.8.0), furrr (&ge;
0.1.0), future (&ge; 1.18.0), graphics, grDevices, igraph (&ge;
1.2), leaflet (&ge; 2.0), magrittr (&ge; 1.2), Matrix (&ge; 1.2.14),
methods (&ge; 3.3), plotly, purrr (&ge; 0.2), raster (&ge; 3.4-13),
Rcpp (&ge; 0.12.18), Rdpack (&ge; 0.11.0), RSpectra (&ge; 0.13.1), sp
(&ge; 1.3.0), stats, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RdMacros:</td>
<td>Rdpack</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>GDAL (&gt;= 2.2.3): gdal-bin (deb), libgdal-dev (deb)
or gdal-devel (rmp), GEOS (&gt;= 3.4.0), PROJ (&gt;= 4.9.3):
libproj-dev (deb), sqlite3, ImageMagick++: imagemagick (deb),
libmagic-dev (deb), libmagick++-dev (deb) or
ImageMagick-c++-devel (rpm) ImageMagick
(http://imagemagick.org) or GraphicsMagick
(http://www.graphicsmagick.org) or LyX (http://www.lyx.org) for
saveGIF(); (PDF)LaTeX for saveLatex(); SWF Tools
(http://swftools.org) for saveSWF(); FFmpeg (http://ffmpeg.org)
or avconv (https://libav.org/avconv.html) for saveVideo()</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-07-06 19:28:23 UTC; luis.osorio</td>
</tr>
<tr>
<td>Author:</td>
<td>Luis Osorio-Olvera
    <a href="https://orcid.org/0000-0003-0701-5398"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Jorge Soberón <a href="https://orcid.org/0000-0003-2160-4148"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Rusby G. Contreras-Díaz
    <a href="https://orcid.org/0000-0002-0569-8984"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-07-06 20:22:11 UTC</td>
</tr>
</table>
<hr>
<h2 id='adj_mat'>adj_mat: Function to compute the adjacency matrix of an area.</h2><span id='topic+adj_mat'></span>

<h3>Description</h3>

<p>Creates an adjacency matrix of an area of interest.
This could be the accessible area (M) of a species or any geographic
region of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adj_mat(modelsparse, ngbs = 1, eigen_sys = FALSE, which_eigs = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adj_mat_+3A_modelsparse">modelsparse</code></td>
<td>
<p>A setA object returned by the function
<code><a href="#topic+model2sparse">model2sparse</a></code>.</p>
</td></tr>
<tr><td><code id="adj_mat_+3A_ngbs">ngbs</code></td>
<td>
<p>Numeric. Number of neighbors (see details).</p>
</td></tr>
<tr><td><code id="adj_mat_+3A_eigen_sys">eigen_sys</code></td>
<td>
<p>Logical. If TRUE the eigen analyses of the
adjacency matrix will be returned.</p>
</td></tr>
<tr><td><code id="adj_mat_+3A_which_eigs">which_eigs</code></td>
<td>
<p>Numeric. Which eigen value and eigen vector will
be returned.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The model is a raster object of the area where the dispersal
process will occur.
The number of neighbors depends on the dispersal abilities of the species
and the spatial resolution of the niche model; for example, a species's
with big dispersal abilities will move throughout more than 1 km^2 per day,
so the idea is to give an approximate number of moving neighbors (pixels)
per unit of time.
For more information about see adjacency matrices in the context of
the theory of area of distribution (Soberon and Osorio-Olvera, 2022).
</p>


<h3>Value</h3>

<p>Returns an object of class <code><a href="#topic+setM">setM</a></code> with 7 slots.
The first contains the adjacency matrix. A n x n sparse matrix (n=number of
non-NA cells of the niche model) where connected cells are represented by 1.
The second slot has the adjacency list. It is a list of matrices with four
columns (FromRasCell -from cell ID of the raster-, -to cell ID of the
raster-, -from non-NA cell-, -to non-NA cell-). Other slots contain
information about initial coordinates where dispersal occurs
(initial_points), number of cells used to define the neighborhood (ngbs),
non-NA coordinates (coordinates), and a matrix of eigen vectors (eigen_vec).
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>References</h3>

<p>Soberón J, Osorio-Olvera L (2023).
&ldquo;A dynamic theory of the area of distribution.&rdquo;
<em>Journal of Biogeography6</em>, <b>50</b>, 1037-1048.
<a href="https://doi.org/10.1111/jbi.14587">doi:10.1111/jbi.14587</a>, <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/jbi.14587">https://onlinelibrary.wiley.com/doi/abs/10.1111/jbi.14587</a>..
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x_coord &lt;- c(-106.5699, -111.3737,-113.9332,
             -110.8913, -106.4262, -106.5699)
y_coord &lt;- c(16.62661, 17.72373, 19.87618,
             22.50763, 21.37728, 16.62661)
xy &lt;- cbind(x_coord, y_coord)
p &lt;- sp::Polygon(xy)
ps &lt;- sp::Polygons(list(p),1)
sps &lt;- sp::SpatialPolygons(list(ps))
mx_grid &lt;- bamm::shape2Grid(sps,resolution = 0.25,ones = TRUE)
mx_sparse &lt;- bamm::model2sparse(model=mx_grid, threshold = 0.1)
adj_mx &lt;- bamm::adj_mat(modelsparse=mx_sparse,
                        ngbs=1,eigen_sys=TRUE,which_eigs=1)
print(adj_mx)
mx_grid_eigen &lt;- mx_grid
mx_grid_eigen[mx_sparse@cellIDs] &lt;- adj_mx@eigen_vec
raster::plot(mx_grid_eigen)

</code></pre>

<hr>
<h2 id='bam_clusters'>bam_clusters: Function to estimate the connectivity of suitable areas</h2><span id='topic+bam_clusters'></span>

<h3>Description</h3>

<p>Function to estimate the connectivity of suitable areas given
an adjacency matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bam_clusters(model, ngbs = 1, plot_model = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bam_clusters_+3A_model">model</code></td>
<td>
<p>A niche model in raster format or a <code><a href="#topic+setA">setA</a></code>
object (see <code><a href="#topic+model2sparse">model2sparse</a></code>).</p>
</td></tr>
<tr><td><code id="bam_clusters_+3A_ngbs">ngbs</code></td>
<td>
<p>Numeric. Number of neighbors (see details).</p>
</td></tr>
<tr><td><code id="bam_clusters_+3A_plot_model">plot_model</code></td>
<td>
<p>Logical. Indicates whether to plot the niche model using a
leaflet map, connected suitable cells shown in the same color.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The main result of the function is the Connectivity-Suitability-Diagram
(CSD). In this diagram connected suitable cells make clusters of pixels.
For more details about the CSD see (Soberon and Osorio-Olvera, 2022).
</p>


<h3>Value</h3>

<p>An object of class <code><a href="#topic+csd">csd</a></code>. It contains three slots.
1) connections: a data.frame with three columns where first and the second
represent (x and y) centroid coordinates of the niche model
and the third column with the cluster ID where they belong.
2) interactive_map: a leaflet map of connected suitable pixels shown in
the same color. 3) A RasterLayer of connected suitable pixels.
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>References</h3>

<p>Soberón J, Osorio-Olvera L (2023).
&ldquo;A dynamic theory of the area of distribution.&rdquo;
<em>Journal of Biogeography6</em>, <b>50</b>, 1037-1048.
<a href="https://doi.org/10.1111/jbi.14587">doi:10.1111/jbi.14587</a>, <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/jbi.14587">https://onlinelibrary.wiley.com/doi/abs/10.1111/jbi.14587</a>..
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(891)
model_path &lt;- system.file("extdata/Lepus_californicus_cont.tif",
                          package = "bamm")
model &lt;- raster::raster(model_path)
model &lt;- model &gt; 0.7
clusterin &lt;- bamm::bam_clusters(model,ngbs=1,plot_model=TRUE)
raster::plot(clusterin@raster_map)

clusterin@interactive_map

</code></pre>

<hr>
<h2 id='bam_sim'>bam_sim: Simulate dispersal dynamics using the set B of the BAM framework.</h2><span id='topic+bam_sim'></span>

<h3>Description</h3>

<p>bam_sim: Simulate dispersal dynamics using the set B of the BAM framework.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bam_sim(
  sp1,
  sp2,
  set_M,
  initial_points,
  periods_toxic,
  periods_suitable,
  nsteps,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bam_sim_+3A_sp1">sp1</code></td>
<td>
<p>Niche model of the focal species (the one that disperses).</p>
</td></tr>
<tr><td><code id="bam_sim_+3A_sp2">sp2</code></td>
<td>
<p>Niche model of the species with whom sp1 interacts
(currently no dispersal dynamics for this species).</p>
</td></tr>
<tr><td><code id="bam_sim_+3A_set_m">set_M</code></td>
<td>
<p>A setM object containing the adjacency matrix for sp1.
See <code><a href="#topic+adj_mat">adj_mat</a></code></p>
</td></tr>
<tr><td><code id="bam_sim_+3A_initial_points">initial_points</code></td>
<td>
<p>A sparse vector returned by the function
<code><a href="#topic+occs2sparse">occs2sparse</a></code></p>
</td></tr>
<tr><td><code id="bam_sim_+3A_periods_toxic">periods_toxic</code></td>
<td>
<p>Time periods that sps2 takes to develop defense
mechanisms (i.e. toxic).</p>
</td></tr>
<tr><td><code id="bam_sim_+3A_periods_suitable">periods_suitable</code></td>
<td>
<p>This is the time that sp2 takes to become non-toxic</p>
</td></tr>
<tr><td><code id="bam_sim_+3A_nsteps">nsteps</code></td>
<td>
<p>Number of steps to run the simulation</p>
</td></tr>
<tr><td><code id="bam_sim_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Show progress bar</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The returned object inherits from <code><a href="#topic+setA">setA</a></code>,
<code><a href="#topic+setM">setM</a></code> classes. Details about the dynamic model
can be found in Soberon and Osorio-Olvera (2022). The model is cellular
automata where the occupied area of a species at time <code class="reqn">t+1</code> is
estimated by the multiplication of three
binary matrices: one matrix represents movements (M), another
abiotic -niche- tolerances (A), and a third, biotic interactions (B)
(Soberon and Osorio-Olvera, 2022).
</p>
<p style="text-align: center;"><code class="reqn">\mathbf{G}_j(t+1) =\mathbf{B}_j(t)\mathbf{A}_j(t)\mathbf{M}_j
\mathbf{G}_j(t)</code>
</p>



<h3>Value</h3>

<p>An object of class bam. The object contains 12 slots of information
(see details) from which simulation results are stored in sdm_sim object,
a list of sparse matrices with results of each simulation step.
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>References</h3>

<p>Soberón J, Osorio-Olvera L (2023).
&ldquo;A dynamic theory of the area of distribution.&rdquo;
<em>Journal of Biogeography6</em>, <b>50</b>, 1037-1048.
<a href="https://doi.org/10.1111/jbi.14587">doi:10.1111/jbi.14587</a>, <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/jbi.14587">https://onlinelibrary.wiley.com/doi/abs/10.1111/jbi.14587</a>..
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Compute dispersal dynamics of Urania boisduvalii as a function of
# palatable Omphalea
urap &lt;- system.file("extdata/urania_omph/urania_guanahacabibes.tif",
                                  package = "bamm")
ura &lt;- raster::raster(urap)
ompp &lt;- system.file("extdata/urania_omph/omphalea_guanahacabibes.tif",
                                  package = "bamm")
omp &lt;- raster::raster(ompp)
msparse &lt;- bamm::model2sparse(ura)
init_coordsdf &lt;- data.frame(x=-84.38751, y= 22.02932)
initial_points &lt;- bamm::occs2sparse(modelsparse = msparse,init_coordsdf)
set_M &lt;- bamm::adj_mat(modelsparse = msparse,ngbs = 1)
ura_sim &lt;- bamm::bam_sim(sp1=ura, sp2=omp, set_M=set_M,
                         initial_points=initial_points,
                         periods_toxic=5,
                         periods_suitable=1,
                         nsteps=40)
ura_omp &lt;- bamm::sim2Raster(ura_sim)
raster::plot(ura_omp[[c(1,5,10,15,20,30,35,40)]])

if(requireNamespace("animation")){
# Animation example
anp &lt;-tempfile(pattern = "simulation_results_",fileext = ".gif")
# new_sim &lt;- bamm::sim2Animation(sdm_simul = ura_sim,
#                                which_steps = seq_len(ura_sim@sim_steps),
#                                fmt = "GIF",
#                               filename = anp)
}

</code></pre>

<hr>
<h2 id='bam_ssim'>bam_ssim: Simulate dispersal dynamics using the set B of the BAM framework.</h2><span id='topic+bam_ssim'></span>

<h3>Description</h3>

<p>bam_ssim: Simulate dispersal dynamics using the set B of the BAM framework.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bam_ssim(
  sp1,
  sp2,
  set_M,
  initial_points,
  periods_toxic,
  periods_suitable,
  dispersal_prob = 0.85,
  palatable_matrices = FALSE,
  nsteps,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bam_ssim_+3A_sp1">sp1</code></td>
<td>
<p>Niche model of the focal species (the one that disperses).</p>
</td></tr>
<tr><td><code id="bam_ssim_+3A_sp2">sp2</code></td>
<td>
<p>Niche model of the species with whom sp1 interacts
(currently no dispersal dynamics for this species).</p>
</td></tr>
<tr><td><code id="bam_ssim_+3A_set_m">set_M</code></td>
<td>
<p>A setM object containing the adjacency matrix for sp1.
See <code><a href="#topic+adj_mat">adj_mat</a></code></p>
</td></tr>
<tr><td><code id="bam_ssim_+3A_initial_points">initial_points</code></td>
<td>
<p>A sparse vector returned by the function
<code><a href="#topic+occs2sparse">occs2sparse</a></code></p>
</td></tr>
<tr><td><code id="bam_ssim_+3A_periods_toxic">periods_toxic</code></td>
<td>
<p>Time periods that sps2 takes to develop defense
mechanisms (i.e. toxic).</p>
</td></tr>
<tr><td><code id="bam_ssim_+3A_periods_suitable">periods_suitable</code></td>
<td>
<p>This is the time that sp2 takes to become non-toxic</p>
</td></tr>
<tr><td><code id="bam_ssim_+3A_dispersal_prob">dispersal_prob</code></td>
<td>
<p>A numeric value indicating the probability to disperse
to neighboring cells.
This probability is assumed to be binomially distributed</p>
</td></tr>
<tr><td><code id="bam_ssim_+3A_palatable_matrices">palatable_matrices</code></td>
<td>
<p>Logical. If TRUE palatable matrices for each time
will be returned.</p>
</td></tr>
<tr><td><code id="bam_ssim_+3A_nsteps">nsteps</code></td>
<td>
<p>Number of steps to run the simulation</p>
</td></tr>
<tr><td><code id="bam_ssim_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Show progress bar</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The returned object inherits from <code><a href="#topic+setA">setA</a></code>,
<code><a href="#topic+setM">setM</a></code> classes. Details about the dynamic model
can be found in Soberon and Osorio-Olvera (2022).
</p>


<h3>Value</h3>

<p>An object of class bam. The object contains 12 slots of information
(see details) from which simulation results are stored in sdm_sim object,
a list of sparse matrices with results of each simulation step. Palatable
matrices are returned as a list of sparse matrices with information about
palatable pixels for each step of the simulation.
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>References</h3>

<p>Soberón J, Osorio-Olvera L (2023).
&ldquo;A dynamic theory of the area of distribution.&rdquo;
<em>Journal of Biogeography6</em>, <b>50</b>, 1037-1048.
<a href="https://doi.org/10.1111/jbi.14587">doi:10.1111/jbi.14587</a>, <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/jbi.14587">https://onlinelibrary.wiley.com/doi/abs/10.1111/jbi.14587</a>..
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
urap &lt;- system.file("extdata/urania_omph/urania_guanahacabibes.tif",
                                  package = "bamm")
ura &lt;- raster::raster(urap)
ompp &lt;- system.file("extdata/urania_omph/omphalea_guanahacabibes.tif",
                                  package = "bamm")
omp &lt;- raster::raster(ompp)
msparse &lt;- bamm::model2sparse(ura)
init_coordsdf &lt;- data.frame(x=-84.38751, y= 22.02932)
initial_points &lt;- bamm::occs2sparse(modelsparse = msparse,init_coordsdf)
set_M &lt;- bamm::adj_mat(modelsparse = msparse,ngbs = 1)
ura_ssim &lt;- bamm::bam_ssim(sp1=ura, sp2=omp, set_M=set_M,
                           dispersal_prob = 0.75,
                           initial_points=initial_points,
                           periods_toxic=5,
                           periods_suitable=1,
                           nsteps=40)
ura_omp &lt;- bamm::sim2Raster(ura_ssim)
raster::plot(ura_omp[[c(1,2,5,10,15,20,30,35,40)]])

if(requireNamespace("animation")){
# Animation example
anp &lt;-tempfile(pattern = "simulation_results_",fileext = ".gif")
#new_sim &lt;- bamm::sim2Animation(sdm_simul = ura_ssim,
#                               which_steps = seq_len(ura_ssim@sim_steps),
#                               fmt = "GIF",
#                               filename = anp)
}

</code></pre>

<hr>
<h2 id='bam-class'>Class <code>bam</code> digram</h2><span id='topic+bam-class'></span><span id='topic+bam'></span>

<h3>Description</h3>

<p>Class <code>bam</code> digram
</p>


<h3>Value</h3>

<p>An object of class bam
</p>


<h3>Slots</h3>


<dl>
<dt><code>sdm_sim</code></dt><dd><p>A list of sparse vectors representing the area occupied</p>
</dd>
<dt><code>palatable_matrices</code></dt><dd><p>A list of sparse vectors representing palatable
sites.</p>
</dd>
<dt><code>sim_steps</code></dt><dd><p>Number of simulation steps
by the species</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>

<hr>
<h2 id='bioindex_sparse-class'>Class  <code>bioindex_sparse</code></h2><span id='topic+bioindex_sparse-class'></span><span id='topic+bioindex_sparse'></span><span id='topic+biodiversity_index_sparse'></span>

<h3>Description</h3>

<p>Class  <code>bioindex_sparse</code>
</p>


<h3>Value</h3>

<p>An object of class bioindex_sparse
</p>


<h3>Slots</h3>


<dl>
<dt><code>alpha</code></dt><dd><p>A sparse matrix with the richness of species per site</p>
</dd>
<dt><code>omega</code></dt><dd><p>A sparse matrix with the range size of every species</p>
</dd>
<dt><code>wBeta</code></dt><dd><p>A numeric value with Whittaker’s multiplicative beta index</p>
</dd>
<dt><code>laBeta</code></dt><dd><p>A numeric value with Lande’s additive beta index</p>
</dd>
<dt><code>leBeta</code></dt><dd><p>A numeric value with Legendre’s beta index</p>
</dd>
<dt><code>nestedness</code></dt><dd><p>A numeric value with Wright and Reeves' nestedness</p>
</dd>
<dt><code>dispersion_field</code></dt><dd><p>A sparse matrix with the set of ranges of all species
that occur in at each locality</p>
</dd>
<dt><code>richness_field</code></dt><dd><p>A sparse matrix with the number of  shared species in
each site</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>

<hr>
<h2 id='bioindex-class'>Class  <code>bioindex</code></h2><span id='topic+bioindex-class'></span><span id='topic+bioindex'></span><span id='topic+biodiversity_index'></span>

<h3>Description</h3>

<p>Class  <code>bioindex</code>
</p>


<h3>Value</h3>

<p>An object of class bioindex
</p>


<h3>Slots</h3>


<dl>
<dt><code>alpha</code></dt><dd><p>A matrix with the richness of species per site</p>
</dd>
<dt><code>omega</code></dt><dd><p>A matrix with the range size of every species</p>
</dd>
<dt><code>wBeta</code></dt><dd><p>A numeric value with Whittaker’s multiplicative beta index</p>
</dd>
<dt><code>laBeta</code></dt><dd><p>A numeric value with Lande’s additive beta index</p>
</dd>
<dt><code>leBeta</code></dt><dd><p>A numeric value with Legendre’s beta index</p>
</dd>
<dt><code>nestedness</code></dt><dd><p>A numeric value with Wright and Reeves' nestedness</p>
</dd>
<dt><code>dispersion_field</code></dt><dd><p>A matrix with the set of ranges of all species that
occur in at each locality</p>
</dd>
<dt><code>richness_field</code></dt><dd><p>A matrix with the number of  shared species in
each site</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>

<hr>
<h2 id='community_sim'>community_bam: Community <code>bamm</code></h2><span id='topic+community_sim'></span>

<h3>Description</h3>

<p>Estimate community dynamics using the <code>bamm</code> framework
</p>


<h3>Usage</h3>

<pre><code class='language-R'>community_sim(
  en_models,
  ngbs_vect,
  init_coords,
  nsteps,
  threshold_vec = NULL,
  stochastic_dispersal = FALSE,
  disp_prop2_suitability = TRUE,
  disper_prop = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="community_sim_+3A_en_models">en_models</code></td>
<td>
<p>A stack or directory with the ecological niche models for
each species in the community.</p>
</td></tr>
<tr><td><code id="community_sim_+3A_ngbs_vect">ngbs_vect</code></td>
<td>
<p>A vector containing the number of neighbors for each
adjacency matrix of each species in the community
see <code><a href="#topic+adj_mat">adj_mat</a></code>.</p>
</td></tr>
<tr><td><code id="community_sim_+3A_init_coords">init_coords</code></td>
<td>
<p>A data.frame with 3 columns: sp_name, x and
y; x is the longitude and y is the latitude of initial dispersal points</p>
</td></tr>
<tr><td><code id="community_sim_+3A_nsteps">nsteps</code></td>
<td>
<p>Number of iteration steps for the simulation.</p>
</td></tr>
<tr><td><code id="community_sim_+3A_threshold_vec">threshold_vec</code></td>
<td>
<p>A vector of threshold values used to bnarize niche
models.</p>
</td></tr>
<tr><td><code id="community_sim_+3A_stochastic_dispersal">stochastic_dispersal</code></td>
<td>
<p>Logical. If dispersal depends on a probability of
visiting neighbor cells (Moore neighborhood).</p>
</td></tr>
<tr><td><code id="community_sim_+3A_disp_prop2_suitability">disp_prop2_suitability</code></td>
<td>
<p>Logical. If probability of dispersal
is proportional to the suitability of reachable cells. The proportional
value must be declared in the parameter 'disper_prop'.</p>
</td></tr>
<tr><td><code id="community_sim_+3A_disper_prop">disper_prop</code></td>
<td>
<p>Probability of dispersal to reachable cells.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each element in community_sim is an object of class. For more
details about the simulation see <code><a href="#topic+sdm_sim">sdm_sim</a></code>.
<code><a href="#topic+bam">bam</a></code>.
</p>


<h3>Value</h3>

<p>An object of class community_sim. The object contains simulation
results for each species in the community.
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberon
</p>


<h3>References</h3>

<p>Soberón J, Osorio-Olvera L (2023).
&ldquo;A dynamic theory of the area of distribution.&rdquo;
<em>Journal of Biogeography6</em>, <b>50</b>, 1037-1048.
<a href="https://doi.org/10.1111/jbi.14587">doi:10.1111/jbi.14587</a>, <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/jbi.14587">https://onlinelibrary.wiley.com/doi/abs/10.1111/jbi.14587</a>..
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
lagos_path &lt;- system.file("extdata/conejos",
                          package = "bamm")
enm_path &lt;- list.files(lagos_path,
                       pattern = ".tif",
                       full.names = TRUE)[seq(1,10)]
en_models &lt;- raster::stack(enm_path)
ngbs_vect &lt;- sample(1:2,replace = TRUE,
                    size = raster::nlayers(en_models))
init_coords &lt;- read.csv(file.path(lagos_path,
                                  "lagos_initit.csv"))[seq(1,10),]
nsteps &lt;- 12
sdm_comm &lt;- bamm::community_sim(en_models = en_models,
                                ngbs_vect = ngbs_vect,
                                init_coords = init_coords,
                                nsteps = nsteps)

com_pam &lt;- bamm::csim2pam(sdm_comm,which_steps = seq(1,nsteps))
rich_pam &lt;- pam2richness(com_pam,which_steps = c(1,5,10))
raster::plot(rich_pam)

</code></pre>

<hr>
<h2 id='community_sim-class'>Class  <code>community_sim</code> digram</h2><span id='topic+community_sim-class'></span><span id='topic+community_bam'></span><span id='topic+community-class'></span>

<h3>Description</h3>

<p>Class  <code>community_sim</code> digram
</p>


<h3>Value</h3>

<p>An object of class community_sim
</p>


<h3>Slots</h3>


<dl>
<dt><code>community_sim</code></dt><dd><p>A list of sparse vectors representing the area occupied
by the species</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>

<hr>
<h2 id='csd_estimate'>csd_estimate: Estimate the connectivity suitability and dispersal plot</h2><span id='topic+csd_estimate'></span>

<h3>Description</h3>

<p>csd_plot gives an estimate of the number of geographic clusters
given a set of dispersal hypothesis and a suitability raster
</p>


<h3>Usage</h3>

<pre><code class='language-R'>csd_estimate(model, dispersal_steps = c(2, 4, 8, 16, 32, 64))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="csd_estimate_+3A_model">model</code></td>
<td>
<p>A raster model or a setA object representing the
suitability model</p>
</td></tr>
<tr><td><code id="csd_estimate_+3A_dispersal_steps">dispersal_steps</code></td>
<td>
<p>A numeric vector with elements representing
the dispersal hypothesis to test.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more information about the Connectivity-Suitability-Diagram
see <code><a href="#topic+bam_clusters">bam_clusters</a></code>
</p>


<h3>Value</h3>

<p>A list of length three. The first element contains the Connectivity-
Suitability-Diagram information estimated for each element in the vector
of dispersal_steps. The second is tbl_df object with a summary of the number
of cluster of each dispersal step and the mean number of connected clusters.
The last element is base plot showing the information contained in
the tbl_df object.
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>References</h3>

<p>Soberón J, Osorio-Olvera L (2023).
&ldquo;A dynamic theory of the area of distribution.&rdquo;
<em>Journal of Biogeography6</em>, <b>50</b>, 1037-1048.
<a href="https://doi.org/10.1111/jbi.14587">doi:10.1111/jbi.14587</a>, <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/jbi.14587">https://onlinelibrary.wiley.com/doi/abs/10.1111/jbi.14587</a>..
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
model_path &lt;- system.file("extdata/Lepus_californicus_cont.tif",
                          package = "bamm")
model &lt;- raster::raster(model_path)
model &lt;- model &gt; 0.7
csd_plot &lt;- bamm::csd_estimate(model,
                         dispersal_steps=c(2,4,8))
csd_plot$plot

</code></pre>

<hr>
<h2 id='csd-class'>Class  <code>csd</code></h2><span id='topic+csd-class'></span><span id='topic+csd'></span>

<h3>Description</h3>

<p>Class  <code>csd</code>
</p>


<h3>Value</h3>

<p>An object of class csd
</p>


<h3>Slots</h3>


<dl>
<dt><code>connections</code></dt><dd><p>A data.frame with four columns: x, y, clusterID and
cluster_size</p>
</dd>
<dt><code>interactive_map</code></dt><dd><p>A leaflet map with markers showing the geographical
clusters</p>
</dd>
<dt><code>raster_map</code></dt><dd><p>A raster map with cluster IDs as values.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>

<hr>
<h2 id='csim2pam'>csim2pam: Converts community simulation to a Presence Absence Matrix (PAM)</h2><span id='topic+csim2pam'></span>

<h3>Description</h3>

<p>Converts community simulation object into a
Presence Absence Matrices (PAM) for a given simulation steps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>csim2pam(community_sim, which_steps)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="csim2pam_+3A_community_sim">community_sim</code></td>
<td>
<p>An object of class <code><a href="#topic+community_bam">community_bam</a></code>.</p>
</td></tr>
<tr><td><code id="csim2pam_+3A_which_steps">which_steps</code></td>
<td>
<p>Steps in the simulation object to be converted into a PAM</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For details about the object community_sim see
<code><a href="#topic+community_sim">community_sim</a></code>
</p>


<h3>Value</h3>

<p>An object of class <code><a href="#topic+pam">pam</a></code>; it contains five slots.
1) pams: a list of sparse matrices with Presence-Absence information (PAMs).
2) which_steps: time steps corresponding to each PAM. 3) sp_names: a
vector of species names. 4) the grid area used in the simulation. 5) Non NA
cell (pixel) IDs.
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>References</h3>

<p>Soberón J, Osorio-Olvera L (2023).
&ldquo;A dynamic theory of the area of distribution.&rdquo;
<em>Journal of Biogeography6</em>, <b>50</b>, 1037-1048.
<a href="https://doi.org/10.1111/jbi.14587">doi:10.1111/jbi.14587</a>, <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/jbi.14587">https://onlinelibrary.wiley.com/doi/abs/10.1111/jbi.14587</a>..
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
lagos_path &lt;- system.file("extdata/conejos",
                          package = "bamm")
enm_path &lt;- list.files(lagos_path,
                       pattern = ".tif",
                       full.names = TRUE)[seq(1,10)]
en_models &lt;- raster::stack(enm_path)
ngbs_vect &lt;- sample(1:2,replace = TRUE,
                    size = raster::nlayers(en_models))
init_coords &lt;- read.csv(file.path(lagos_path,
                                  "lagos_initit.csv"))[seq(1,10),]
nsteps &lt;- 10
sdm_comm &lt;- bamm::community_sim(en_models = en_models,
                               ngbs_vect = ngbs_vect,
                               init_coords = init_coords,
                               nsteps = nsteps,
                               threshold = 0.1)

pamt10 &lt;- bamm::csim2pam(community_sim = sdm_comm ,
                        which_steps = 10)
pams &lt;- bamm::csim2pam(community_sim = sdm_comm ,
                       which_steps = seq_len(10))
rich_pam &lt;- bamm::pam2richness(pams,which_steps = c(1,5))
print(rich_pam)


</code></pre>

<hr>
<h2 id='diversity_range_analysis'>range_diversity_analysis: diversity analysis</h2><span id='topic+diversity_range_analysis'></span>

<h3>Description</h3>

<p>diversity_range_analysis biodiversity indices related to
diversity-range plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diversity_range_analysis(
  pam,
  xy_mat = NULL,
  lower_interval = 0.05,
  upper_interval = 0.95,
  raster_templete = NULL,
  niter = 100,
  return_null_dfield = FALSE,
  randal = "indep_swap",
  parallel = TRUE,
  n_cores = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diversity_range_analysis_+3A_pam">pam</code></td>
<td>
<p>A Presence-Absence-Matrix of matrix class or sparse matrix.</p>
</td></tr>
<tr><td><code id="diversity_range_analysis_+3A_xy_mat">xy_mat</code></td>
<td>
<p>A two dimensional matrix with longitude and latitude data.</p>
</td></tr>
<tr><td><code id="diversity_range_analysis_+3A_lower_interval">lower_interval</code></td>
<td>
<p>Lower interval.</p>
</td></tr>
<tr><td><code id="diversity_range_analysis_+3A_upper_interval">upper_interval</code></td>
<td>
<p>Upper interval.</p>
</td></tr>
<tr><td><code id="diversity_range_analysis_+3A_raster_templete">raster_templete</code></td>
<td>
<p>A raster template.</p>
</td></tr>
<tr><td><code id="diversity_range_analysis_+3A_niter">niter</code></td>
<td>
<p>Number of iterations to obtain the distribution.</p>
</td></tr>
<tr><td><code id="diversity_range_analysis_+3A_return_null_dfield">return_null_dfield</code></td>
<td>
<p>If TRUE the null distribution of dispersal field
will be returned.</p>
</td></tr>
<tr><td><code id="diversity_range_analysis_+3A_randal">randal</code></td>
<td>
<p>Randomization algorithm applied to the PAM.
Possible choices &quot;curveball&quot;,&quot;fastball&quot; and &quot;indep_swap&quot;.</p>
</td></tr>
<tr><td><code id="diversity_range_analysis_+3A_parallel">parallel</code></td>
<td>
<p>If TRUE the computations will be performed in parallel.</p>
</td></tr>
<tr><td><code id="diversity_range_analysis_+3A_n_cores">n_cores</code></td>
<td>
<p>Number of cores for the parallel computation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more information about the biodiversity indices
see Soberon and Cavner (2015). For detail about the diversity range analysis
see Soberon et al. (2022). To plot diversity range results use
<code><a href="#topic+plot">plot</a></code> method for objects of class
<code><a href="#topic+diversity_range">diversity_range</a></code>.
</p>
<p>For details about randomization algorithms applied to the PAM see
<code><a href="#topic+null_dispersion_field_distribution">null_dispersion_field_distribution</a></code>.
</p>


<h3>Value</h3>

<p>An object of class  <code><a href="#topic+diversity_range">diversity_range</a></code>. The main
result is the diversity range analysis which shows jointly two indices
describing the community composition of every cell in the grid:
(1) the relative number of species, and (2) the mean dispersion field
(see plot method for <code><a href="#topic+plot">plot</a></code> (Soberon et al. 2022).
The contains 12 slots with different measurements of biodiversity such
as alpha diversity (species richness in each site or pixel),
omega (size of the area of distribution of each species),
dispersion field (the standardized size of the area of distribution of
all species occurring in each pixel).
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>References</h3>

<p>Soberón J, Cobos ME, Nuñez-Penichet C (2021).
&ldquo;Visualizing species richness and site similarity from presence-absence matrices.&rdquo;
<em>Biodiversity Informatics</em>, <b>16</b>(1), 20–27.
<a href="https://doi.org/10.17161/bi.v16i1.14782">doi:10.17161/bi.v16i1.14782</a>, <a href="https://journals.ku.edu/jbi/article/view/14782">https://journals.ku.edu/jbi/article/view/14782</a>..
</p>
<p>Soberon J, Cavner J (2015).
&ldquo;Indices of Biodiversity Pattern Based on Presence-Absence Matrices: A GIS Implementation.&rdquo;
<em>Biodiversity Informatics</em>, <b>10</b>, 22&ndash;34..
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(111)
pam &lt;- matrix(rbinom(10000,1,0.5),nrow = 100,ncol = 1000)
rdivan &lt;- bamm::diversity_range_analysis(pam=pam,
                                         parallel = FALSE,
                                         niter = 10,
                                         return_null_dfield=TRUE)
bamm::plot(rdivan,plot_type="diversity_range")
# Lagomorphos

lagos_path &lt;- system.file("extdata/conejos",
                          package = "bamm")
enm_path &lt;- list.files(lagos_path,
                       pattern = ".tif",
                       full.names = TRUE)
en_models &lt;- raster::stack(enm_path) &gt;0.01
nonas &lt;- which(!is.na(en_models[[1]][]))
xy_mat &lt;- sp::coordinates(en_models[[1]])[ nonas,]
pam &lt;- bamm::models2pam(en_models,sparse=FALSE)

rdivan &lt;- bamm::diversity_range_analysis(pam=pam,
                                         xy_mat=xy_mat,
                                         raster_templete = en_models[[1]],
                                         parallel=TRUE,
                                         n_cores=2,
                                         return_null_dfield=TRUE)
bamm::plot(rdivan,plot_type="diversity_range")
bamm::plot(rdivan,plot_type="diversity_range_map")
if(require(plotly) &amp;&amp; require(crosstalk)){
#bamm::plot(rdivan,plot_type="diversity_range_interactive")
}

</code></pre>

<hr>
<h2 id='diversity_range-class'>Class  <code>diversity_range</code></h2><span id='topic+diversity_range-class'></span><span id='topic+diversity_range'></span><span id='topic+diversityrange'></span>

<h3>Description</h3>

<p>Class  <code>diversity_range</code>
</p>


<h3>Value</h3>

<p>An object of class diversity_range
</p>


<h3>Slots</h3>


<dl>
<dt><code>alpha</code></dt><dd><p>A column vector with species richness per site</p>
</dd>
<dt><code>omega</code></dt><dd><p>A column vector with the size of the area of distribution per
species.</p>
</dd>
<dt><code>alpha_raster</code></dt><dd><p>Species richness in raster format.</p>
</dd>
<dt><code>dispersion_field</code></dt><dd><p>A matrix with the set of ranges of all species that
occur in at each locality.</p>
</dd>
<dt><code>dispersion_field_raster</code></dt><dd><p>Raster object with the observed values of
dispersion field.</p>
</dd>
<dt><code>diversity_range_raster</code></dt><dd><p>Raster object of diversity range.</p>
</dd>
<dt><code>diversity_range_colors</code></dt><dd><p>Colors to plot endemism levels.</p>
</dd>
<dt><code>null_dispersion_field_dist</code></dt><dd><p>A matrix with dispersion field null
distribution.</p>
</dd>
<dt><code>xy_coordinates</code></dt><dd><p>A matrix of geographical coordinates</p>
</dd>
<dt><code>n_iterations</code></dt><dd><p>Number of iterations used to estimate the dispersion
field null distribution.</p>
</dd>
<dt><code>nsps</code></dt><dd><p>Number of species in the PAM.</p>
</dd>
<dt><code>nsites</code></dt><dd><p>Number of sites in the PAM.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>

<hr>
<h2 id='eigen_bam'>eigen_bam: Compute the Eigen system of two bam objects</h2><span id='topic+eigen_bam'></span>

<h3>Description</h3>

<p>Calculates the Eigen values and Eigen vectors of bam objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eigen_bam(A = NULL, M = NULL, AM = TRUE, which_eigen = 1, rmap = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="eigen_bam_+3A_a">A</code></td>
<td>
<p>A bam object of class setA.</p>
</td></tr>
<tr><td><code id="eigen_bam_+3A_m">M</code></td>
<td>
<p>A bam object of class setM.</p>
</td></tr>
<tr><td><code id="eigen_bam_+3A_am">AM</code></td>
<td>
<p>A logical value to specify whether to use the product AM or MA.
If true the AM will be returned else the product MA will be returned.</p>
</td></tr>
<tr><td><code id="eigen_bam_+3A_which_eigen">which_eigen</code></td>
<td>
<p>An integer representing the which eigen value and eigen
vector will be computed.</p>
</td></tr>
<tr><td><code id="eigen_bam_+3A_rmap">rmap</code></td>
<td>
<p>Logical. If TRUE the function will return a map of the eigen
vector of the product AM.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The eigenvector associated with the dominant eigenvalue of an
adjacency matrix provides information about the number of forms in
which a cell can be visited from other cells. Details about the
eigen analysis in the context of the area of distribution can be
found in Soberon and Osorio-Olvera (2022).
</p>


<h3>Value</h3>

<p>A list with four objects. 1) eigen_values (these are indicated in
which_eigen parameter of the function), 2) eigen_vectors (the corresponding
eigen vectors of each eigen value), 3) Standardized eigen vectors (0 to 1),
4) A RasterLayer depicting the information of the first eigen vector of
the system.
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>References</h3>

<p>Soberón J, Osorio-Olvera L (2023).
&ldquo;A dynamic theory of the area of distribution.&rdquo;
<em>Journal of Biogeography6</em>, <b>50</b>, 1037-1048.
<a href="https://doi.org/10.1111/jbi.14587">doi:10.1111/jbi.14587</a>, <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/jbi.14587">https://onlinelibrary.wiley.com/doi/abs/10.1111/jbi.14587</a>..
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
model_path &lt;- system.file("extdata/Lepus_californicus_cont.tif",
                          package = "bamm")
model &lt;- raster::raster(model_path)
sparse_mod &lt;- bamm::model2sparse(model = model,0.75)
plot(sparse_mod@niche_model)
adj_mod &lt;- bamm::adj_mat(sparse_mod,ngbs = 1,eigen_sys = TRUE)
# Product AM
eig_bam_am &lt;- bamm::eigen_bam(A=sparse_mod,M=adj_mod,AM=TRUE)
raster::plot(eig_bam_am$map)
# Product MA
eig_bam_ma &lt;- bamm::eigen_bam(A=sparse_mod,M=adj_mod,AM=FALSE)
raster::plot(eig_bam_ma$map)

</code></pre>

<hr>
<h2 id='g_area-class'>S4 classes to organize data and results of <code>bamm</code> objects</h2><span id='topic+g_area-class'></span><span id='topic+g_area'></span><span id='topic+coordinates-class'></span>

<h3>Description</h3>

<p>S4 classes to organize data and results of <code>bamm</code> objects
</p>


<h3>Value</h3>

<p>An object of class g_area
</p>


<h3>Slots</h3>


<dl>
<dt><code>coordinates</code></dt><dd><p>A two column matrix with coordinates</p>
</dd>
<dt><code>eigen_vec</code></dt><dd><p>Eigen vector of adjacency matrix</p>
</dd>
<dt><code>eigen_val</code></dt><dd><p>Eigen value of adjacency matrix
slot g_model A raster representing the geographic area
slot g_sparse A sparse matrix of the geographic area</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>

<hr>
<h2 id='jaccard'>jaccard: Estimates the Jaccard index for comparing two binary maps</h2><span id='topic+jaccard'></span>

<h3>Description</h3>

<p>Estimates the Jaccard index for comparing two binary maps
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jaccard(m1, m2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jaccard_+3A_m1">m1</code></td>
<td>
<p>A binary raster A or an object of class setA returned by
the function <code><a href="#topic+model2sparse">model2sparse</a></code>.</p>
</td></tr>
<tr><td><code id="jaccard_+3A_m2">m2</code></td>
<td>
<p>A binary raster A or an object of class setA returned by
the function <code><a href="#topic+model2sparse">model2sparse</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Jaccard index is computed as follows
</p>
<p style="text-align: center;"><code class="reqn">J(A,B)={{|A\cap B|}\over{|A\cup B|}}={{|A\cap B|}\over{|A|+|B|-|A\cap B|}}.
</code>
</p>



<h3>Value</h3>

<p>Returns a data.frame with three values: 1) jaccard (Jaccard index),
2) percentage_m1 (the percentage of m1 that the
intersection <code class="reqn">|A \cap B|</code> represents), and 3) percentage_m2
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m1_path &lt;- system.file("extdata/conejos/Lepus_othus_cont.tif",
                       package = "bamm")
m2_path &lt;- system.file("extdata/conejos/Brachylagus_idahoensis_cont.tif",
                       package = "bamm")
m1 &lt;- raster::raster(m1_path) &gt; 0.01
m2 &lt;- raster::raster(m2_path) &gt;0.01
jcc &lt;- bamm::jaccard(m1,m2)
print(jcc)
</code></pre>

<hr>
<h2 id='leaflet-class'>Class  <code>leaflet</code> leaflet</h2><span id='topic+leaflet-class'></span><span id='topic+leaflet'></span>

<h3>Description</h3>

<p>Class  <code>leaflet</code> leaflet
</p>


<h3>Value</h3>

<p>An object of class leaflet
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>

<hr>
<h2 id='model2sparse'>model2sparse: Converts a niche model into a diagonal sparse matrix</h2><span id='topic+model2sparse'></span>

<h3>Description</h3>

<p>model2sparse: Converts a niche model into a diagonal sparse matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>model2sparse(model, threshold = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="model2sparse_+3A_model">model</code></td>
<td>
<p>A raster object representing the geographic projection
of a niche model.</p>
</td></tr>
<tr><td><code id="model2sparse_+3A_threshold">threshold</code></td>
<td>
<p>A threshold to convert a continuous model into a
binary model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>threshold parameter represents the suitability value used to
convert continuous model into a binary model.
</p>


<h3>Value</h3>

<p>An object of class <code><a href="#topic+setA">setA</a></code>. The niche model
is stored as diagonal sparse matrix (slot sparse_model).
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>Examples</h3>

<pre><code class='language-R'>model_path &lt;- system.file("extdata/Lepus_californicus_cont.tif",
                          package = "bamm")
model &lt;- raster::raster(model_path)

sparse_mod &lt;- bamm::model2sparse(model, threshold=0.75)
print(sparse_mod)
raster::plot(sparse_mod@niche_model)
</code></pre>

<hr>
<h2 id='models2pam'>models2pam: Converts binary rasters to a PAM</h2><span id='topic+models2pam'></span>

<h3>Description</h3>

<p>Function to convert binary raster models to a
Presence Absences Matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>models2pam(
  mods_stack,
  return_coords = FALSE,
  sparse = TRUE,
  parallel = FALSE,
  ncores = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="models2pam_+3A_mods_stack">mods_stack</code></td>
<td>
<p>A raster stack containing binary models of each
species in the community.</p>
</td></tr>
<tr><td><code id="models2pam_+3A_return_coords">return_coords</code></td>
<td>
<p>Logical. If TRUE the pam will be returned with
coordinates in the first two columns.</p>
</td></tr>
<tr><td><code id="models2pam_+3A_sparse">sparse</code></td>
<td>
<p>Logical. If TRUE the PAM will be returned as a sparse matrix.</p>
</td></tr>
<tr><td><code id="models2pam_+3A_parallel">parallel</code></td>
<td>
<p>Logical. If TRUE computations will be done in parallel</p>
</td></tr>
<tr><td><code id="models2pam_+3A_ncores">ncores</code></td>
<td>
<p>Integer. Number of cores to run the parallel process.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more information about PAM see Soberon and Cavner (2015).
</p>


<h3>Value</h3>

<p>A presence-absence matrix (PAM).
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>References</h3>

<p>Soberon J, Cavner J (2015).
&ldquo;Indices of Biodiversity Pattern Based on Presence-Absence Matrices: A GIS Implementation.&rdquo;
<em>Biodiversity Informatics</em>, <b>10</b>, 22&ndash;34..
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
lagos_path &lt;- system.file("extdata/conejos",
                          package = "bamm")
enm_path &lt;- list.files(lagos_path,
                       pattern = ".tif",
                       full.names = TRUE)[1:10]
en_models &lt;- raster::stack(enm_path) &gt;0.01
pam &lt;- bamm::models2pam(en_models,
                        return_coords=TRUE,
                        sparse=FALSE,
                        parallel=FALSE,ncores=2)
head(pam)

</code></pre>

<hr>
<h2 id='null_dispersion_field_distribution'>null_dispersion_field_distribution: Null distribution of the
dispersion field</h2><span id='topic+null_dispersion_field_distribution'></span>

<h3>Description</h3>

<p>null_dispersion_field_distribution estimates a
random distribution of the dispersion field values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>null_dispersion_field_distribution(
  pam,
  n_iter = 10,
  randal = "indep_swap",
  parallel = TRUE,
  n_cores = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="null_dispersion_field_distribution_+3A_pam">pam</code></td>
<td>
<p>A Presence-Absence-Matrix of matrix class or sparse matrix.</p>
</td></tr>
<tr><td><code id="null_dispersion_field_distribution_+3A_n_iter">n_iter</code></td>
<td>
<p>Number of iterations to obtain the distribution.</p>
</td></tr>
<tr><td><code id="null_dispersion_field_distribution_+3A_randal">randal</code></td>
<td>
<p>Randomization algorithm applied to the PAM.
Possible choices &quot;curveball&quot;, &quot;fastball&quot;, and &quot;indep_swap&quot;.</p>
</td></tr>
<tr><td><code id="null_dispersion_field_distribution_+3A_parallel">parallel</code></td>
<td>
<p>If TRUE the computations will be performed in parallel.</p>
</td></tr>
<tr><td><code id="null_dispersion_field_distribution_+3A_n_cores">n_cores</code></td>
<td>
<p>Number of cores for the parallel computation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Estimates a random distribution of the dispersion field values. To obtain
random values it uses the function <code><a href="#topic+permute_pam">permute_pam</a></code>
at each step of the iterations. Randomization of the PAM can be
performed using the &quot;fastball&quot; (Godard and Neal, 2022) and the
&quot;curveball&quot; (Strona et al., 2014), and  and the independent
swap (Kembel et al. 2010) algorithms.
The implementation of the &quot;fastball&quot; in C++ is provided
in <a href="https://github.com/zpneal/fastball/blob/main/fastball.cpp">https://github.com/zpneal/fastball/blob/main/fastball.cpp</a>
</p>


<h3>Value</h3>

<p>A data matrix of size nrow(pam) X n_iter with dispersion
field values.
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>References</h3>

<p>Soberon J, Cavner J (2015).
&ldquo;Indices of Biodiversity Pattern Based on Presence-Absence Matrices: A GIS Implementation.&rdquo;
<em>Biodiversity Informatics</em>, <b>10</b>, 22&ndash;34.
</p>
<p>Strona G, Nappo D, Boccacci F, Fattorini S, San-Miguel-Ayanz J (2014).
&ldquo;A fast and unbiased procedure to randomize ecological binary matrices with fixed row and column totals.&rdquo;
<em>Nature Communications</em>, <b>5</b>(1), 1&ndash;9.
ISSN 20411723, <a href="https://doi.org/10.1038/ncomms5114">doi:10.1038/ncomms5114</a>, <a href="https://www.r-project.org">https://www.r-project.org</a>.
</p>
<p>Godard K, Neal ZP (2022).
&ldquo;fastball: a fast algorithm to randomly sample bipartite graphs with fixed degree sequences.&rdquo;
<em>Journal of Complex Networks</em>, <b>10</b>(6), cnac049.
ISSN 2051-1329, <a href="https://doi.org/10.1093/comnet/cnac049">doi:10.1093/comnet/cnac049</a>, https://academic.oup.com/comnet/article-pdf/10/6/cnac049/47758701/cnac049.pdf.
</p>
<p>Kembel SW, Cowan PD, Helmus MR, Cornwell WK, Morlon H, Ackerly DD, Blomberg SP, Webb CO (2010).
&ldquo;Picante: R tools for integrating phylogenies and ecology.&rdquo;
<em>Bioinformatics</em>, <b>26</b>, 1463&ndash;1464.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(111)
pam &lt;- matrix(rbinom(100,1,0.3),nrow = 10,ncol = 10)
dfield_rand &lt;- bamm::null_dispersion_field_distribution(pam,n_iter=10,
                                                       parallel=FALSE,
                                                       randal="indep_swap",
                                                       n_cores = 2)
head(dfield_rand)
</code></pre>

<hr>
<h2 id='occs2sparse'>occs2sparse: Converts occurrence data into a sparse matrix object</h2><span id='topic+occs2sparse'></span>

<h3>Description</h3>

<p>occs2sparse: Converts occurrence data into a sparse matrix object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>occs2sparse(modelsparse, occs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="occs2sparse_+3A_modelsparse">modelsparse</code></td>
<td>
<p>A setA object returned by the function
<code><a href="#topic+model2sparse">model2sparse</a></code></p>
</td></tr>
<tr><td><code id="occs2sparse_+3A_occs">occs</code></td>
<td>
<p>A matrix or a data.frame containing two columns.
The first one is the longitude and the second is the latitude.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Rows of this column vector represent non NA pixels of the
niche model.
</p>


<h3>Value</h3>

<p>A sparse vector of zeros (presences) and ones (absences).
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
model_path &lt;- system.file("extdata/Lepus_californicus_cont.tif",
                          package = "bamm")
model &lt;- raster::raster(model_path)

sparse_mod &lt;- bamm::model2sparse(model,threshold=0.05)

occs_lep_cal &lt;- data.frame(longitude = c(-115.10417,
                                         -104.90417),
                           latitude = c(29.61846,
                                        29.81846))

occs_sparse &lt;- bamm::occs2sparse(modelsparse = sparse_mod,
                                occs = occs_lep_cal)

head(occs_sparse)
</code></pre>

<hr>
<h2 id='pam-class'>Class  <code>pam</code> Presence-Absence Matrix</h2><span id='topic+pam-class'></span><span id='topic+pam'></span><span id='topic+PAM'></span>

<h3>Description</h3>

<p>Class  <code>pam</code> Presence-Absence Matrix
</p>


<h3>Value</h3>

<p>An object of class pam
</p>


<h3>Slots</h3>


<dl>
<dt><code>pams</code></dt><dd><p>A list of sparse matrices representing Presence-Absence Matrix for
each simulation time</p>
</dd>
<dt><code>which_steps</code></dt><dd><p>Simulation steps</p>
</dd>
<dt><code>sp_names</code></dt><dd><p>Names of species in the PAM</p>
</dd>
<dt><code>grid</code></dt><dd><p>Raster grid of the studied area</p>
</dd>
<dt><code>cellIDs</code></dt><dd><p>Cells with ids of the PAM sites</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>

<hr>
<h2 id='pam2bioindex'>pam2bioindex: PAM to biodiversity index</h2><span id='topic+pam2bioindex'></span>

<h3>Description</h3>

<p>pam2bioindex estimates various biodiversity indices for a
certain PAM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pam2bioindex(pam, biodiv_index = "dispersion_field", as_sparse = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pam2bioindex_+3A_pam">pam</code></td>
<td>
<p>A Presence-Absence-Matrix of matrix class or sparse matrix.</p>
</td></tr>
<tr><td><code id="pam2bioindex_+3A_biodiv_index">biodiv_index</code></td>
<td>
<p>Possible values are alpha, omega,
wbeta (Whittaker’s multiplicative beta index),
laBeta (Lande’s additive beta index)
dispersion_field, all.</p>
</td></tr>
<tr><td><code id="pam2bioindex_+3A_as_sparse">as_sparse</code></td>
<td>
<p>Return indices as sparse objects</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The biodiversity indices can be found in Soberón and Cavner (2015).
</p>


<h3>Value</h3>

<p>An object of class <code><a href="#topic+bioindex">bioindex</a></code> with three slots
each represents a matrix of diversity indices: alpha, omega, and
dispersion field, richness_field.
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>References</h3>

<p>Soberon J, Cavner J (2015).
&ldquo;Indices of Biodiversity Pattern Based on Presence-Absence Matrices: A GIS Implementation.&rdquo;
<em>Biodiversity Informatics</em>, <b>10</b>, 22&ndash;34.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(111)
pam &lt;- matrix(rbinom(100,1,0.3),nrow = 10,ncol = 10)
bioindices &lt;- bamm::pam2bioindex(pam=pam,biodiv_index="all")
# Return results as sparse models
bioindices &lt;- bamm::pam2bioindex(pam=pam,biodiv_index="all",as_sparse=TRUE)
bioindices@alpha
bioindices@omega
bioindices@dispersion_field
</code></pre>

<hr>
<h2 id='pam2richness'>pam2richness: Converts Presence Absence Matrix (pam object) to
richness raster</h2><span id='topic+pam2richness'></span>

<h3>Description</h3>

<p>Converts Presence Absence Matrix (pam object) to
richness raster
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pam2richness(pamobj, which_steps)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pam2richness_+3A_pamobj">pamobj</code></td>
<td>
<p>An object of class pam see <code><a href="#topic+csim2pam">csim2pam</a></code></p>
</td></tr>
<tr><td><code id="pam2richness_+3A_which_steps">which_steps</code></td>
<td>
<p>Time steps in the pam to convert</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A RasterStack richness for each simulation step
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
lagos_path &lt;- system.file("extdata/conejos",
                          package = "bamm")
enm_path &lt;- list.files(lagos_path,
                       pattern = ".tif",
                    full.names = TRUE)[seq(1,10)]
en_models &lt;- raster::stack(enm_path)
ngbs_vect &lt;- sample(2,replace = TRUE,
                    size = raster::nlayers(en_models))
init_coords &lt;- read.csv(file.path(lagos_path,
                                  "lagos_initit.csv"))[seq(1,10),]
nsteps &lt;- 10
sdm_comm &lt;- bamm::community_sim(en_models = en_models,
                                ngbs_vect = ngbs_vect,
                                init_coords = init_coords,
                                nsteps = nsteps,
                                threshold = 0.1)

pams &lt;-bamm::csim2pam(community_sim = sdm_comm ,
                      which_steps = seq_len(nsteps))
richness_stack &lt;- bamm::pam2richness(pamobj = pams,
                                     which_steps = pams@which_steps)
raster::plot(richness_stack)

</code></pre>

<hr>
<h2 id='permute_pam'>permute_pam: Function to permute a Presence-Absence-Matrix.</h2><span id='topic+permute_pam'></span>

<h3>Description</h3>

<p>permute_pam: Function to permute a Presence-Absence-Matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>permute_pam(m, niter = NULL, as_sparse = FALSE, randal = "indep_swap")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="permute_pam_+3A_m">m</code></td>
<td>
<p>Presence-Absence-Matrix (PAM) or a binary matrix with columns
representing species and rows sites.</p>
</td></tr>
<tr><td><code id="permute_pam_+3A_niter">niter</code></td>
<td>
<p>Number of iterations to permute the PAM.</p>
</td></tr>
<tr><td><code id="permute_pam_+3A_as_sparse">as_sparse</code></td>
<td>
<p>If TRUE the PAM will be returned as a sparse matrix</p>
</td></tr>
<tr><td><code id="permute_pam_+3A_randal">randal</code></td>
<td>
<p>Randomization algorithm applied to the PAM.
Possible choices &quot;curveball&quot;, &quot;fastball&quot;, and &quot;indep_swap&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can use the &quot;curveball&quot; (Strona et al., 2014), the
fastball (Godard and Neal, 2022), and the independent swap
algorithms. The implementation of the &quot;fastball&quot; in C++ is provided
in <a href="https://github.com/zpneal/fastball/blob/main/fastball.cpp">https://github.com/zpneal/fastball/blob/main/fastball.cpp</a>.
Please when using the &quot;fastball&quot; algorithm for publications cite
Godard and Neal (2022). When using the &quot;curveball&quot; cite
Strona et al. (2014). When using independent swap (&quot;indep_swap&quot;)
cite Kembel et al. (2010)
</p>


<h3>Value</h3>

<p>Returns a permuted matrix of the same dimensions of m
(same number of rows and columns). Note that the sum of each row and column
of this permuted matrix is equal to that of m.
species.
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>References</h3>

<p>Strona G, Nappo D, Boccacci F, Fattorini S, San-Miguel-Ayanz J (2014).
&ldquo;A fast and unbiased procedure to randomize ecological binary matrices with fixed row and column totals.&rdquo;
<em>Nature Communications</em>, <b>5</b>(1), 1&ndash;9.
ISSN 20411723, <a href="https://doi.org/10.1038/ncomms5114">doi:10.1038/ncomms5114</a>, <a href="https://www.r-project.org">https://www.r-project.org</a>.
</p>
<p>Godard K, Neal ZP (2022).
&ldquo;fastball: a fast algorithm to randomly sample bipartite graphs with fixed degree sequences.&rdquo;
<em>Journal of Complex Networks</em>, <b>10</b>(6), cnac049.
ISSN 2051-1329, <a href="https://doi.org/10.1093/comnet/cnac049">doi:10.1093/comnet/cnac049</a>, https://academic.oup.com/comnet/article-pdf/10/6/cnac049/47758701/cnac049.pdf.
</p>
<p>Kembel SW, Cowan PD, Helmus MR, Cornwell WK, Morlon H, Ackerly DD, Blomberg SP, Webb CO (2010).
&ldquo;Picante: R tools for integrating phylogenies and ecology.&rdquo;
<em>Bioinformatics</em>, <b>26</b>, 1463&ndash;1464.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(111)
pam &lt;- matrix(rbinom(100,1,0.3),nrow = 10,ncol = 10)
ppam &lt;- bamm::permute_pam(m = pam,niter = NULL,as_sparse = FALSE)
# Check if matrices are different
all(pam == ppam)
# Check if row totals are the same
all(Matrix::rowSums(pam) == Matrix::rowSums(ppam))
# Check if column total are the same
all(Matrix::colSums(pam) == Matrix::colSums(ppam))
</code></pre>

<hr>
<h2 id='plot+2Cdiversity_range+2CANY-method'>Plot method for objects of class diversity_range <span class="pkg">bamm</span>.</h2><span id='topic+plot+2Cdiversity_range+2CANY-method'></span>

<h3>Description</h3>

<p>Plot method for objects of class diversity_range <span class="pkg">bamm</span>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'diversity_range,ANY'
plot(
  x,
  xlab = NULL,
  plot_type = "diversity_range",
  legend = TRUE,
  legend_position = "bottomright",
  ylab = NULL,
  col = NULL,
  pch = NULL,
  pch_legend = 19,
  radius = 0.5,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Cdiversity_range+2B2CANY-method_+3A_x">x</code></td>
<td>
<p>An object of class diversity_range</p>
</td></tr>
<tr><td><code id="plot+2B2Cdiversity_range+2B2CANY-method_+3A_xlab">xlab</code></td>
<td>
<p>x label</p>
</td></tr>
<tr><td><code id="plot+2B2Cdiversity_range+2B2CANY-method_+3A_plot_type">plot_type</code></td>
<td>
<p>Plot type: possible options: &quot;diversity_range&quot;
(range-diversity plot),
&quot;diversity_range_map&quot; (a raster map with
diversity_range categories),
&quot;alpha&quot; (a raster map with alpha diversity values),
&quot;dispersion_field&quot; (a raster with dispersion field)</p>
</td></tr>
<tr><td><code id="plot+2B2Cdiversity_range+2B2CANY-method_+3A_legend">legend</code></td>
<td>
<p>Logical. If TRUE the legend of the categorical diversity
range values will appear.</p>
</td></tr>
<tr><td><code id="plot+2B2Cdiversity_range+2B2CANY-method_+3A_legend_position">legend_position</code></td>
<td>
<p>Legend position.</p>
</td></tr>
<tr><td><code id="plot+2B2Cdiversity_range+2B2CANY-method_+3A_ylab">ylab</code></td>
<td>
<p>y label</p>
</td></tr>
<tr><td><code id="plot+2B2Cdiversity_range+2B2CANY-method_+3A_col">col</code></td>
<td>
<p>Plot colors.</p>
</td></tr>
<tr><td><code id="plot+2B2Cdiversity_range+2B2CANY-method_+3A_pch">pch</code></td>
<td>
<p>Patch type.</p>
</td></tr>
<tr><td><code id="plot+2B2Cdiversity_range+2B2CANY-method_+3A_pch_legend">pch_legend</code></td>
<td>
<p>Patch type for legends.</p>
</td></tr>
<tr><td><code id="plot+2B2Cdiversity_range+2B2CANY-method_+3A_radius">radius</code></td>
<td>
<p>Size of the patch for the interactive map.</p>
</td></tr>
<tr><td><code id="plot+2B2Cdiversity_range+2B2CANY-method_+3A_...">...</code></td>
<td>
<p>Graphical parameters. Any argument that can be passed to 1)
base::plot, such as axes=FALSE, main='title', ylab='latitude' 2)
leaflet::leaflet or 3) leaflet::addCircleMarkers.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To show interactive diversity_range plots install the 'plotly' R
package.
</p>


<h3>Value</h3>

<p>Plot of the results of the diversity_range analysis
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>

<hr>
<h2 id='predict+2Cbam-method'>Predict method of the package <span class="pkg">bamm</span>.</h2><span id='topic+predict+2Cbam-method'></span><span id='topic+predict'></span><span id='topic+bam-method'></span>

<h3>Description</h3>

<p>predicts species' distribution under suitability changes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'bam'
predict(
  object,
  niche_layers,
  nbgs_vec = NULL,
  nsteps_vec,
  stochastic_dispersal = FALSE,
  disp_prop2_suitability = TRUE,
  disper_prop = 0.5,
  animate = FALSE,
  period_names = NULL,
  fmt = "GIF",
  filename,
  bg_color = "#F6F2E5",
  suit_color = "#0076BE",
  occupied_color = "#03C33F",
  png_keyword = "sdm_sim",
  ani.width = 1200,
  ani.height = 1200,
  ani.res = 300
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict+2B2Cbam-method_+3A_object">object</code></td>
<td>
<p>a of class bam.</p>
</td></tr>
<tr><td><code id="predict+2B2Cbam-method_+3A_niche_layers">niche_layers</code></td>
<td>
<p>A raster or RasterStack with the niche models for
each time period</p>
</td></tr>
<tr><td><code id="predict+2B2Cbam-method_+3A_nbgs_vec">nbgs_vec</code></td>
<td>
<p>A vector with the number of neighbors for the adjacency
matrices</p>
</td></tr>
<tr><td><code id="predict+2B2Cbam-method_+3A_nsteps_vec">nsteps_vec</code></td>
<td>
<p>Number of simulation steps for each time period.</p>
</td></tr>
<tr><td><code id="predict+2B2Cbam-method_+3A_stochastic_dispersal">stochastic_dispersal</code></td>
<td>
<p>Logical. If dispersal depends on a probability of
visiting neighbor cells (Moore neighborhood).</p>
</td></tr>
<tr><td><code id="predict+2B2Cbam-method_+3A_disp_prop2_suitability">disp_prop2_suitability</code></td>
<td>
<p>Logical. If probability of dispersal is
proportional
to the suitability of reachable cells. The proportional value must be
declared
in the parameter 'disper_prop'.</p>
</td></tr>
<tr><td><code id="predict+2B2Cbam-method_+3A_disper_prop">disper_prop</code></td>
<td>
<p>Probability of dispersal to reachable cells.</p>
</td></tr>
<tr><td><code id="predict+2B2Cbam-method_+3A_animate">animate</code></td>
<td>
<p>Logical. If TRUE a dispersal animation on climate change
scenarios will be created</p>
</td></tr>
<tr><td><code id="predict+2B2Cbam-method_+3A_period_names">period_names</code></td>
<td>
<p>Character vector with the names of periods that will
be animated. Default NULL.</p>
</td></tr>
<tr><td><code id="predict+2B2Cbam-method_+3A_fmt">fmt</code></td>
<td>
<p>Animation format. Possible values are GIF and HTML</p>
</td></tr>
<tr><td><code id="predict+2B2Cbam-method_+3A_filename">filename</code></td>
<td>
<p>File name.</p>
</td></tr>
<tr><td><code id="predict+2B2Cbam-method_+3A_bg_color">bg_color</code></td>
<td>
<p>Color for unsuitable pixels. Default &quot;#F6F2E5&quot;.</p>
</td></tr>
<tr><td><code id="predict+2B2Cbam-method_+3A_suit_color">suit_color</code></td>
<td>
<p>Color for suitable pixels. Default &quot;#0076BE&quot;.</p>
</td></tr>
<tr><td><code id="predict+2B2Cbam-method_+3A_occupied_color">occupied_color</code></td>
<td>
<p>Color for occupied pixels. Default &quot;#03C33F&quot;.</p>
</td></tr>
<tr><td><code id="predict+2B2Cbam-method_+3A_png_keyword">png_keyword</code></td>
<td>
<p>A keyword name for the png images generated by
the function</p>
</td></tr>
<tr><td><code id="predict+2B2Cbam-method_+3A_ani.width">ani.width</code></td>
<td>
<p>Animation width unit in px</p>
</td></tr>
<tr><td><code id="predict+2B2Cbam-method_+3A_ani.height">ani.height</code></td>
<td>
<p>Animation height unit in px</p>
</td></tr>
<tr><td><code id="predict+2B2Cbam-method_+3A_ani.res">ani.res</code></td>
<td>
<p>Animation resolution unit in px</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A RasterStack of predictions of dispersal dynamics as a function
of environmental change scenarios.
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>Examples</h3>

<pre><code class='language-R'># rm(list = ls())
# Read raster model for Lepus californicus
model_path &lt;- system.file("extdata/Lepus_californicus_cont.tif",
                          package = "bamm")
model &lt;- raster::raster(model_path)
# Convert model to sparse
sparse_mod &lt;- bamm::model2sparse(model = model,threshold=0.1)
# Compute adjacency matrix
adj_mod &lt;- bamm::adj_mat(sparse_mod,ngbs=1)

# Initial points to start dispersal process

occs_lep_cal &lt;- data.frame(longitude = c(-115.10417,
                                         -104.90417),
                           latitude = c(29.61846,
                                        29.81846))
# Convert to sparse the initial points
occs_sparse &lt;- bamm::occs2sparse(modelsparse = sparse_mod,
                                occs = occs_lep_cal)

# Run the bam (sdm) simultation for 100 time steps
smd_lep_cal &lt;- bamm::sdm_sim(set_A = sparse_mod,
                             set_M = adj_mod,
                             initial_points = occs_sparse,
                             nsteps = 10)
#----------------------------------------------------------------------------
# Predict species' distribution under suitability change
# scenarios (could be climate chage scenarios).
#----------------------------------------------------------------------------

# Read suitability layers (two suitability change scenarios)
layers_path &lt;- system.file("extdata/suit_change",
                           package = "bamm")
niche_mods_stack &lt;- raster::stack(list.files(layers_path,
                                             pattern = ".tif$",
                                             full.names = TRUE))
raster::plot(niche_mods_stack)
# Predict
new_preds &lt;- predict(object = smd_lep_cal,
                     niche_layers = niche_mods_stack,
                     nsteps_vec = c(50,100))

# Generate the dispersal animation for time period 1 and 2

if(requireNamespace("animation")){
ani_prd &lt;- tempfile(pattern = "prediction_",fileext = ".gif")
#new_preds &lt;- predict(object = smd_lep_cal,
#                      niche_layers = niche_mods_stack,
#                      nsteps_vec = c(10,10),
#                      animate=TRUE,
#                      filename=ani_prd,
#                    fmt="GIF")

}

</code></pre>

<hr>
<h2 id='sdm_sim'>sdm_sim: Simulate single species dispersal dynamics using the BAM framework.</h2><span id='topic+sdm_sim'></span>

<h3>Description</h3>

<p>sdm_sim: Simulate single species dispersal dynamics using the BAM framework.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sdm_sim(
  set_A,
  set_M,
  initial_points,
  nsteps,
  stochastic_dispersal = TRUE,
  disp_prop2_suitability = TRUE,
  disper_prop = 0.5,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sdm_sim_+3A_set_a">set_A</code></td>
<td>
<p>A setA object returned by the function
<code><a href="#topic+model2sparse">model2sparse</a></code></p>
</td></tr>
<tr><td><code id="sdm_sim_+3A_set_m">set_M</code></td>
<td>
<p>A setM object containing the adjacency matrix of the
study area.
See <code><a href="#topic+adj_mat">adj_mat</a></code></p>
</td></tr>
<tr><td><code id="sdm_sim_+3A_initial_points">initial_points</code></td>
<td>
<p>A sparse vector returned by the function
<code><a href="#topic+occs2sparse">occs2sparse</a></code></p>
</td></tr>
<tr><td><code id="sdm_sim_+3A_nsteps">nsteps</code></td>
<td>
<p>Number of steps to run the simulation</p>
</td></tr>
<tr><td><code id="sdm_sim_+3A_stochastic_dispersal">stochastic_dispersal</code></td>
<td>
<p>Logical. If dispersal depends on a probability of
visiting neighbor cells (Moore neighborhood).</p>
</td></tr>
<tr><td><code id="sdm_sim_+3A_disp_prop2_suitability">disp_prop2_suitability</code></td>
<td>
<p>Logical. If probability of dispersal
is proportional to the suitability of reachable cells. The proportional
value must be declared in the parameter 'disper_prop'.</p>
</td></tr>
<tr><td><code id="sdm_sim_+3A_disper_prop">disper_prop</code></td>
<td>
<p>Probability of dispersal to reachable cells.</p>
</td></tr>
<tr><td><code id="sdm_sim_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Show progress bar</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The model is cellular automata where the occupied area
of a species at time <code class="reqn">t+1</code> is estimated by the multiplication of two
binary matrices: one matrix represents movements (M), another
abiotic -niche- tolerances (A)
(Soberon and Osorio-Olvera, 2022).
</p>
<p style="text-align: center;"><code class="reqn">\mathbf{G}_j(t+1) =\mathbf{A}_j(t)\mathbf{M}_j
\mathbf{G}_j(t)</code>
</p>

<p>The equation describes a very simple process: To find the occupied patches
in <code class="reqn">t+1</code> start with those occupied at time <code class="reqn">t</code> denoted by
<code class="reqn">\mathbf{G}_j(t)</code>, allow the individuals to disperse among
adjacent patches, as defined by <code class="reqn">\mathbf{M}_j</code>, then remove individuals
from patches that are unsuitable, as defined by <code class="reqn">\mathbf{A}_j(t)</code>.
</p>


<h3>Value</h3>

<p>An object of class <code><a href="#topic+bam">bam</a></code> with simulation results.
The simulation are stored in the sdm_sim slot (a list of sparse matrices).
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>References</h3>

<p>Soberón J, Osorio-Olvera L (2023).
&ldquo;A dynamic theory of the area of distribution.&rdquo;
<em>Journal of Biogeography6</em>, <b>50</b>, 1037-1048.
<a href="https://doi.org/10.1111/jbi.14587">doi:10.1111/jbi.14587</a>, <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/jbi.14587">https://onlinelibrary.wiley.com/doi/abs/10.1111/jbi.14587</a>..
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
model_path &lt;- system.file("extdata/Lepus_californicus_cont.tif",
                          package = "bamm")
model &lt;- raster::raster(model_path)

sparse_mod &lt;- bamm::model2sparse(model,threshold=0.05)
adj_mod &lt;- bamm::adj_mat(sparse_mod,ngbs=1)
occs_lep_cal &lt;- data.frame(longitude = c(-110.08880,
                                         -98.89638),
                           latitude = c(30.43455,
                                        25.19919))

occs_sparse &lt;- bamm::occs2sparse(modelsparse = sparse_mod,
                                occs = occs_lep_cal)
sdm_lep_cal &lt;- bamm::sdm_sim(set_A = sparse_mod,
                             set_M = adj_mod,
                             initial_points = occs_sparse,
                             nsteps = 10,
                             stochastic_dispersal = TRUE,
                             disp_prop2_suitability=TRUE,
                             disper_prop=0.5,
                             progress_bar=TRUE)

sim_res &lt;- bamm::sim2Raster(sdm_lep_cal)
raster::plot(sim_res)


</code></pre>

<hr>
<h2 id='setA-class'>Class for the A set of the BAM diagram</h2><span id='topic+setA-class'></span><span id='topic+setA'></span><span id='topic+A-class'></span>

<h3>Description</h3>

<p>A class for the A set of the BAM diagram. It contains raster models and
IDs of pixels with values different than NA.
</p>


<h3>Value</h3>

<p>An object of class setA
showClass(&quot;setA&quot;)
</p>


<h3>Slots</h3>


<dl>
<dt><code>niche_model</code></dt><dd><p>A niche model in raster format. It can be a binary model
or continuous.
If the model is in a continuous format.</p>
</dd>
<dt><code>suit_threshold</code></dt><dd><p>Suitability value used to binarize continuous model</p>
</dd>
<dt><code>cellIDs</code></dt><dd><p>A numeric vector with the IDs of the cells with prediction
values</p>
</dd>
<dt><code>suit_values</code></dt><dd><p>A numeric vector with suitability value of the continuous
map</p>
</dd>
<dt><code>sparse_model</code></dt><dd><p>A niche model in sparse matrix format</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>

<hr>
<h2 id='setM-class'>Class for the M set of the <code>bamm</code> diagram</h2><span id='topic+setM-class'></span><span id='topic+setM'></span><span id='topic+M-class'></span>

<h3>Description</h3>

<p>Class for the M set of the <code>bamm</code> diagram
</p>


<h3>Value</h3>

<p>An object of class setM
</p>


<h3>Slots</h3>


<dl>
<dt><code>adj_matix</code></dt><dd><p>An adjacency matrix</p>
</dd>
<dt><code>adj_list</code></dt><dd><p>An adjacency list</p>
</dd>
<dt><code>initial_points</code></dt><dd><p>A presence-absence vector with species' occurrences</p>
</dd>
<dt><code>n_initial_points</code></dt><dd><p>Number of initial points used to start the dispersal
process</p>
</dd>
<dt><code>ngbs</code></dt><dd><p>Number of neighbors</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("setM")
</code></pre>

<hr>
<h2 id='shape2Grid'>shape2Grid: Function to create a grid given a spatial polygon</h2><span id='topic+shape2Grid'></span>

<h3>Description</h3>

<p>shape2Grid creates a raster grid given a spatial polygon and
a grid resolution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shape2Grid(shpolygon, resolution, ones = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="shape2Grid_+3A_shpolygon">shpolygon</code></td>
<td>
<p>A SpatialPolygon, SpatialPolygonDataFrame representing
the desired shape of the grid.</p>
</td></tr>
<tr><td><code id="shape2Grid_+3A_resolution">resolution</code></td>
<td>
<p>Numeric. Spatial resolution of the grid.</p>
</td></tr>
<tr><td><code id="shape2Grid_+3A_ones">ones</code></td>
<td>
<p>Logical. Fill with ones the values of the raster. If not the
values will be written as cellID values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a raster object with the shape of 'shpolygon' of a given
resolution.
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x_coord &lt;- c(-106.5699, -111.3737,-113.9332, -110.8913, -106.4262, -106.5699)
y_coord &lt;- c(16.62661, 17.72373, 19.87618, 22.50763, 21.37728, 16.62661)
xy &lt;- cbind(x_coord, y_coord)
p &lt;- sp::Polygon(xy)
ps &lt;- sp::Polygons(list(p),1)
sps &lt;- sp::SpatialPolygons(list(ps))
r1 &lt;- bamm::shape2Grid(sps,resolution = 0.1,ones = FALSE)
plot(r1)
sp::plot(sps,add=TRUE)

</code></pre>

<hr>
<h2 id='show+2CsetA-method'>Show information in setA class <span class="pkg">bamm</span>.</h2><span id='topic+show+2CsetA-method'></span><span id='topic+show+2Ccsd-method'></span><span id='topic+show+2Cpam-method'></span><span id='topic+show+2Cbioindex_sparse-method'></span><span id='topic+show+2CsetM-method'></span><span id='topic+show+2Cdiversity_range-method'></span>

<h3>Description</h3>

<p>Show information in setA class <span class="pkg">bamm</span>.
</p>
<p>Show information in csd class <span class="pkg">bamm</span>.
</p>
<p>Show information in pam class <span class="pkg">bamm</span>.
</p>
<p>Show information in pam class <span class="pkg">bamm</span>.
</p>
<p>Show information in setA class <span class="pkg">bamm</span>.
</p>
<p>Show information in diversity_range class <span class="pkg">bamm</span>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'setA'
show(object)

## S4 method for signature 'csd'
show(object)

## S4 method for signature 'pam'
show(object)

## S4 method for signature 'bioindex_sparse'
show(object)

## S4 method for signature 'setM'
show(object)

## S4 method for signature 'diversity_range'
show(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show+2B2CsetA-method_+3A_object">object</code></td>
<td>
<p>An object of class diversity_range</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Display information about the setA object
</p>
<p>Display information about the csd object
</p>
<p>Display information about the pam object
</p>
<p>Display information about the bioindex_spars object
</p>
<p>Display information about the setM object
</p>
<p>Display information about the diversity_range object
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>

<hr>
<h2 id='sim2Animation'>sim2Animation: Animate BAM simulation object.</h2><span id='topic+sim2Animation'></span>

<h3>Description</h3>

<p>Animates BAM simulation object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim2Animation(
  sdm_simul,
  which_steps,
  fmt = "GIF",
  filename,
  png_keyword = "sdm_sim",
  extra_legend = NULL,
  bg_color = "#F6F2E5",
  suit_color = "#0076BE",
  occupied_color = "#03C33F",
  gif_vel = 0.8,
  ani.width = 1200,
  ani.height = 1200,
  ani.res = 300
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim2Animation_+3A_sdm_simul">sdm_simul</code></td>
<td>
<p>A bam object. See <code><a href="#topic+sdm_sim">sdm_sim</a></code></p>
</td></tr>
<tr><td><code id="sim2Animation_+3A_which_steps">which_steps</code></td>
<td>
<p>A numeric vector indicating the simulation steps that
are going to be converted into raster layers.</p>
</td></tr>
<tr><td><code id="sim2Animation_+3A_fmt">fmt</code></td>
<td>
<p>Animation format. Possible values are GIF and HTML</p>
</td></tr>
<tr><td><code id="sim2Animation_+3A_filename">filename</code></td>
<td>
<p>File name.</p>
</td></tr>
<tr><td><code id="sim2Animation_+3A_png_keyword">png_keyword</code></td>
<td>
<p>A keyword name for the png images generated by the
function</p>
</td></tr>
<tr><td><code id="sim2Animation_+3A_extra_legend">extra_legend</code></td>
<td>
<p>A legend  to add to the animation.</p>
</td></tr>
<tr><td><code id="sim2Animation_+3A_bg_color">bg_color</code></td>
<td>
<p>Color for unsuitable pixels. Default &quot;#F6F2E5&quot;.</p>
</td></tr>
<tr><td><code id="sim2Animation_+3A_suit_color">suit_color</code></td>
<td>
<p>Color for suitable pixels. Default &quot;#0076BE&quot;.</p>
</td></tr>
<tr><td><code id="sim2Animation_+3A_occupied_color">occupied_color</code></td>
<td>
<p>Color for occupied pixels. Default &quot;#03C33F&quot;.</p>
</td></tr>
<tr><td><code id="sim2Animation_+3A_gif_vel">gif_vel</code></td>
<td>
<p>A value that regulates the velocity of frame transitions.
The bigger it is the transition will be slower
default 0.8</p>
</td></tr>
<tr><td><code id="sim2Animation_+3A_ani.width">ani.width</code></td>
<td>
<p>Animation width unit in px</p>
</td></tr>
<tr><td><code id="sim2Animation_+3A_ani.height">ani.height</code></td>
<td>
<p>Animation height unit in px</p>
</td></tr>
<tr><td><code id="sim2Animation_+3A_ani.res">ani.res</code></td>
<td>
<p>Animation resolution unit in px</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The animation can be saved in a GIF or HTML format. Note that
the generation of the GIF can be time consuming for large simulation
(simulations with more than 60 time steps).
</p>


<h3>Value</h3>

<p>A RasterStack of species' distribution at each simulation step
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>Examples</h3>

<pre><code class='language-R'>model_path &lt;- system.file("extdata/Lepus_californicus_cont.tif",
                          package = "bamm")
model &lt;- raster::raster(model_path)
sparse_mod &lt;- bamm::model2sparse(model,0.1)
adj_mod &lt;- bamm::adj_mat(sparse_mod,ngbs=2)
occs_lep_cal &lt;- data.frame(longitude = c(-115.10417,
                                         -104.90417),
                           latitude = c(29.61846,
                                        29.81846))
occs_sparse &lt;- bamm::occs2sparse(modelsparse = sparse_mod,
                                occs = occs_lep_cal)
sdm_lep_cal &lt;- bamm::sdm_sim(set_A = sparse_mod,
                            set_M = adj_mod,
                            initial_points = occs_sparse,
                            nsteps = 50)

if(requireNamespace("animation")){
ani_name &lt;- tempfile(pattern = "simulation_",fileext = ".html")
#sdm_lep_cal_st &lt;- bamm::sim2Animation(sdm_simul = sdm_lep_cal,
#                                      which_steps = seq(1,50,by=1),
#                                      fmt = "HTML",ani.width = 1200,
#                                      ani.height = 1200,
#                                      filename = ani_name)
}

</code></pre>

<hr>
<h2 id='sim2Raster'>sim2Raster: Convert a BAM simulation object to RasterStack</h2><span id='topic+sim2Raster'></span>

<h3>Description</h3>

<p>Convert a BAM simulation object to RasterStack.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim2Raster(sdm_simul, which_steps = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim2Raster_+3A_sdm_simul">sdm_simul</code></td>
<td>
<p>A bam object. See <code><a href="#topic+sdm_sim">sdm_sim</a></code></p>
</td></tr>
<tr><td><code id="sim2Raster_+3A_which_steps">which_steps</code></td>
<td>
<p>A numeric vector indicating the simulation steps that
are going to be converted into raster layers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A RasterStack of species' distribution at each simulation step
</p>


<h3>Author(s)</h3>

<p>Luis Osorio-Olvera &amp; Jorge Soberón
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
model_path &lt;- system.file("extdata/Lepus_californicus_cont.tif",
                          package = "bamm")
model &lt;- raster::raster(model_path)
sparse_mod &lt;- bamm::model2sparse(model,threshold=0.1)
adj_mod &lt;- bamm::adj_mat(sparse_mod,ngbs = 1)
occs_lep_cal &lt;- data.frame(longitude = c(-115.10417,
                                         -104.90417),
                           latitude = c(29.61846,
                                        29.81846))
occs_sparse &lt;- bamm::occs2sparse(modelsparse = sparse_mod,
                                occs = occs_lep_cal)
sdm_lep_cal &lt;- bamm::sdm_sim(set_A = sparse_mod,
                            set_M = adj_mod,
                            initial_points = occs_sparse,
                            nsteps = 10)
sdm_lep_cal_st &lt;- bamm::sim2Raster(sdm_simul = sdm_lep_cal,
                                  which_steps = seq(1,10,by=1))

raster::plot(sdm_lep_cal_st)


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
