<!DOCTYPE html><html lang="en"><head><title>Help for package bapred</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {bapred}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bapred-package'>
<p>The bapred package</p></a></li>
<li><a href='#autism'>
<p>Autism dataset</p></a></li>
<li><a href='#avedist'>
<p>Average minimal distance between batches</p></a></li>
<li><a href='#ba'>
<p>Batch effect adjustment using a method of choice</p></a></li>
<li><a href='#baaddon'>
<p>Addon batch effect adjustment</p></a></li>
<li><a href='#bametric'>
<p>Diverse metrics for quality of (adjusted) batch data</p></a></li>
<li><a href='#bapred-internal'><p>Internal bapred functions</p></a></li>
<li><a href='#batch'>
<p>batch variable of dataset <code>autism</code></p></a></li>
<li><a href='#combatba'>
<p>Batch effect adjustment using ComBat</p></a></li>
<li><a href='#combatbaaddon'>
<p>Addon batch effect adjustment using ComBat</p></a></li>
<li><a href='#corba'>
<p>Mean correlation before and after batch effect adjustment</p></a></li>
<li><a href='#diffexprm'>
<p>Measure for performance of differential expression analysis (after batch effect adjustment)</p></a></li>
<li><a href='#fabatch'>
<p>Batch effect adjustment using FAbatch</p></a></li>
<li><a href='#fabatchaddon'>
<p>Addon batch effect adjustment using FAbatch</p></a></li>
<li><a href='#kldist'>
<p>Kullback-Leibler divergence between density of within and between batch pairwise distances</p></a></li>
<li><a href='#meancenter'>
<p>Batch effect adjustment by mean-centering</p></a></li>
<li><a href='#meancenteraddon'>
<p>Addon batch effect adjustment for mean-centering</p></a></li>
<li><a href='#noba'>
<p>No batch effect adjustment</p></a></li>
<li><a href='#nobaaddon'>
<p>No addon batch effect adjustment</p></a></li>
<li><a href='#pcplot'>
<p>Visualization of batch effects using Principal Component Analysis</p></a></li>
<li><a href='#pvcam'>
<p>Proportion of variation induced by class signal estimated by Principal Variance Component Analysis</p></a></li>
<li><a href='#qunormaddon'>
<p>Addon quantile normalization using &ldquo;documentation by value&rdquo; (Kostka &amp; Spang, 2008)</p></a></li>
<li><a href='#qunormtrain'>
<p>Quantile normalization with &ldquo;documentation by value&rdquo; (Kostka &amp; Spang, 2008)</p></a></li>
<li><a href='#ratioa'>
<p>Batch effect adjustment using Ratio-A</p></a></li>
<li><a href='#ratioaaddon'>
<p>Addon batch effect adjustment for Ratio-A</p></a></li>
<li><a href='#ratiog'>
<p>Batch effect adjustment using Ratio-G</p></a></li>
<li><a href='#ratiogaddon'>
<p>Addon batch effect adjustment for Ratio-G</p></a></li>
<li><a href='#rmaaddon'>
<p>Addon RMA normalization using &ldquo;documentation by value&rdquo; (Kostka &amp; Spang, 2008)</p></a></li>
<li><a href='#rmatrain'>
<p>RMA normalization with &ldquo;documentation by value&rdquo; (Kostka &amp; Spang, 2008)</p></a></li>
<li><a href='#sepscore'>
<p>Separation score as described in Hornung et al. (2016)</p></a></li>
<li><a href='#skewdiv'>
<p>Skewness divergence score</p></a></li>
<li><a href='#standardize'>
<p>Batch effect adjustment by standardization</p></a></li>
<li><a href='#standardizeaddon'>
<p>Addon batch effect adjustment for standardization</p></a></li>
<li><a href='#svaba'>
<p>Batch effect adjustment using SVA</p></a></li>
<li><a href='#svabaaddon'>
<p>Addon batch effect adjustment using frozen SVA</p></a></li>
<li><a href='#X'>
<p>Covariate matrix of dataset <code>autism</code></p></a></li>
<li><a href='#y'>
<p>Target variable of dataset <code>autism</code></p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Batch Effect Removal and Addon Normalization (in Phenotype
Prediction using Gene Data)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-06-21</td>
</tr>
<tr>
<td>Author:</td>
<td>Roman Hornung, David Causeur</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Roman Hornung &lt;hornung@ibe.med.uni-muenchen.de&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0), glmnet, lme4, MASS, sva, affyPLM</td>
</tr>
<tr>
<td>Imports:</td>
<td>FNN, fuzzyRankTests, methods, mnormt, affy, Biobase</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ArrayExpress</td>
</tr>
<tr>
<td>Description:</td>
<td>Various tools dealing with batch effects, in particular enabling the 
  removal of discrepancies between training and test sets in prediction scenarios.
  Moreover, addon quantile normalization and addon RMA normalization (Kostka &amp; Spang, 
  2008) is implemented to enable integrating the quantile normalization step into 
  prediction rules. The following batch effect removal methods are implemented: 
  FAbatch, ComBat, (f)SVA, mean-centering, standardization, Ratio-A and Ratio-G. 
  For each of these we provide  an additional function which enables a posteriori 
  ('addon') batch effect removal in independent batches ('test data'). Here, the
  (already batch effect adjusted) training data is not altered. For evaluating the
  success of batch effect adjustment several metrics are provided. Moreover, the 
  package implements a plot for the visualization of batch effects using principal 
  component analysis. The main functions of the package for batch effect adjustment 
  are ba() and baaddon() which enable batch effect removal and addon batch effect 
  removal, respectively, with one of the seven methods mentioned above. Another 
  important function here is bametric() which is a wrapper function for all implemented
  methods for evaluating the success of batch effect removal. For (addon) quantile 
  normalization and (addon) RMA normalization the functions qunormtrain(), 
  qunormaddon(), rmatrain() and rmaaddon() can be used.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-21 17:44:28 UTC; hornung</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-06-22 08:20:13 UTC</td>
</tr>
</table>
<hr>
<h2 id='bapred-package'>
The bapred package
</h2><span id='topic+bapred-package'></span><span id='topic+bapred'></span>

<h3>Description</h3>

<p>This is a short summary of the features of <span class="pkg">bapred</span>, a package in <span class="rlang"><b>R</b></span> for the treatment and analysis of batch effects based in high-dimensional molecular data via batch effect adjustment and addon quantile normalization. Here, a special focus is set on phenotype prediction in the presence of batch effects.
</p>


<h3>Details</h3>

<p>Various tools dealing with batch effects, in particular enabling the 
removal of discrepancies between training and test sets in prediction scenarios.
Moreover, addon quantile normalization and addon RMA normalization (Kostka &amp; Spang, 
2008) is implemented to enable integrating the quantile normalization step into 
prediction rules. The following batch effect removal methods are implemented: 
FAbatch, ComBat, (f)SVA, mean-centering, standardization, Ratio-A and Ratio-G. 
For each of these we provide  an additional function which enables a posteriori 
('addon') batch effect removal in independent batches ('test data'). Here, the
(already batch effect adjusted) training data is not altered. For evaluating the
success of batch effect adjustment several metrics are provided. Moreover, the 
package implements a plot for the visualization of batch effects using principal 
component analysis. The main functions of the package for batch effect adjustment 
are <code><a href="#topic+ba">ba</a>()</code> and <code><a href="#topic+baaddon">baaddon</a>()</code> which enable batch effect removal and addon batch effect 
removal, respectively, with one of the seven methods mentioned above. Another 
important function here is <code><a href="#topic+bametric">bametric</a>()</code> which is a wrapper function for all implemented
methods for evaluating the success of batch effect removal. For (addon) quantile 
normalization and (addon) RMA normalization the functions <code><a href="#topic+qunormtrain">qunormtrain</a>()</code>, 
<code><a href="#topic+qunormaddon">qunormaddon</a>()</code>, <code><a href="#topic+rmatrain">rmatrain</a>()</code> and <code><a href="#topic+rmaaddon">rmaaddon</a>()</code> can be used.
</p>


<h3>Author(s)</h3>

<p>Roman Hornung, David Causeur
</p>
<p>Maintainer: Roman Hornung
<a href="mailto:hornung@ibe.med.uni-muenchen.de">hornung@ibe.med.uni-muenchen.de</a>
</p>


<h3>References</h3>

<p>Hornung, R., Boulesteix, A.-L., Causeur, D. (2016). Combining location-and-scale batch effect adjustment with data cleaning by latent factor adjustment. BMC Bioinformatics 17:27, &lt;doi: <a href="https://doi.org/10.1186/s12859-015-0870-z">10.1186/s12859-015-0870-z</a>&gt;.
</p>
<p>Hornung, R., Causeur, D., Bernau, C., Boulesteix, A.-L. (2017). Improving cross-study prediction through addon batch effect adjustment and addon normalization. Bioinformatics 33(3):397–404, &lt;doi: <a href="https://doi.org/10.1093/bioinformatics/btw650">10.1093/bioinformatics/btw650</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load example dataset:

data(autism)



# Subset this example dataset to reduce the
# computational burden of the toy examples:

# Random subset of 150 variables:
set.seed(1234)
Xsub &lt;- X[,sample(1:ncol(X), size=150)]

# In cases of batches with more than 20 observations
# select 20 observations at random:
subinds &lt;- unlist(sapply(1:length(levels(batch)), function(x) {
  indbatch &lt;- which(batch==x)
  if(length(indbatch) &gt; 20)
    indbatch &lt;- sort(sample(indbatch, size=20))
  indbatch
}))
Xsub &lt;- Xsub[subinds,]
batchsub &lt;- batch[subinds]
ysub &lt;- y[subinds]



# Split into training and test sets:

trainind &lt;- which(batchsub %in% c(1,2))

Xsubtrain &lt;- Xsub[trainind,]
ysubtrain &lt;- ysub[trainind]
batchsubtrain &lt;- factor(as.numeric(batchsub[trainind]), levels=c(1,2))


testind &lt;- which(batchsub %in% c(3,4))

Xsubtest &lt;- Xsub[testind,]
ysubtest &lt;- ysub[testind]

batchsubtest &lt;- as.numeric(batchsub[testind])
batchsubtest[batchsubtest==3] &lt;- 1
batchsubtest[batchsubtest==4] &lt;- 2
batchsubtest &lt;- factor(batchsubtest, levels=c(1,2))



# Batch effect adjustment:

combatparams &lt;- ba(x=Xsubtrain, y=ysubtrain, batch=batchsubtrain, 
  method = "combat")
Xsubtraincombat &lt;- combatparams$xadj

meancenterparams &lt;- ba(x=Xsubtrain, y=ysubtrain, batch=batchsubtrain, 
  method = "meancenter")
Xsubtrainmeancenter &lt;- meancenterparams$xadj



# Addon batch effect adjustment:

Xsubtestcombat &lt;- baaddon(params=combatparams, x=Xsubtest, 
  batch=batchsubtest)

Xsubtestmeancenter &lt;- baaddon(params=meancenterparams, x=Xsubtest, 
  batch=batchsubtest)



# Metrics for evaluating the success of batch effect adjustment:

bametric(xba=Xsubtrain, batch=batchsubtrain, metric = "sep")
bametric(xba=Xsubtrainmeancenter, batch=batchsubtrain, metric = "sep")

bametric(x=Xsubtrain, batch=batchsubtrain, y=ysubtrain, 
  metric = "diffexpr", method = "meancenter")

bametric(xba=Xsubtrainmeancenter, x=Xsubtrain, metric = "cor")



# Principal component analysis plots for the visualization 
# of batch effects:

par(mfrow=c(1,3))
pcplot(x=Xsub, batch=batchsub, y=ysub, alpha=0.25, main="alpha = 0.25")
pcplot(x=Xsub, batch=batchsub, y=ysub, alpha=0.75, main="alpha = 0.75")
pcplot(x=Xsub, batch=batchsub, y=ysub, col=1:length(unique(batchsub)), 
  main="col = 1:length(unique(batchsub))")
par(mfrow=c(1,1))



# (Addon) quantile normalization:

qunormparams &lt;- qunormtrain(x=Xsubtrain)

Xtrainnorm &lt;- qunormparams$xnorm

Xtestaddonnorm &lt;- qunormaddon(qunormparams, x=Xsubtest)
</code></pre>

<hr>
<h2 id='autism'>
Autism dataset
</h2><span id='topic+autism'></span>

<h3>Description</h3>

<p>Total RNA obtained from lmyphoblast cell lines derived from 250 individuals, 137 of which suffer from autism and 113 are healthy. The dataset consists of four batches of sizes 101, 96, 45 and 8.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(autism)</code></pre>


<h3>Format</h3>

<p>1) <code>X</code> - the covariate matrix: a matrix of dimension 250 x 1000, containing the numerical transcript values 
</p>
<p>2) <code>batch</code> - the batch variable: a factor with levels '1', '2', '3' and '4'
</p>
<p>3) <code>y</code> - the target variable: a factor with levels '1' corresponding to 'healthy' and '2' corresponding to 'autism'
</p>


<h3>Details</h3>

<p>The RNA measurements were obtained by the Illumina HumanRef-8 v3.0 Expression BeadChip featuring 24,526 transcripts. To reduce computational burden of potential analyses performed using this dataset we randomly selected 1,000 of these 24,526 transcripts. Moreover, the original dataset consisted of five batches and contained measurements of 439 individuals. Again to reduce computational burden of potential analyses we excluded the biggest batch featuring 189 individuals resulting in the 250 individuals included in the dataset made available in <code>bapred</code>.
</p>


<h3>Source</h3>

<p>ArrayExpress, accession number: E-GEOD-37772
</p>


<h3>References</h3>

<p>Luo, R., Sanders, S. J., Tian, Y., Voineagu, I., Huang, N., Chu, S. H., Klei, L., Cai, C., Ou, J., Lowe, J. K., Hurles, M. E., Devlin, B., State, M. W., Geschwind, D. H. (2012). Genome-wide Transcriptome Profiling Reveals the Functional Impact of Rare De Novo and Recurrent CNVs in Autism Spectrum Disorders. The American Journal of Human Genetics 91:38–55, &lt;doi: <a href="https://doi.org/10.1016/j.ajhg.2012.05.011">10.1016/j.ajhg.2012.05.011</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)
</code></pre>

<hr>
<h2 id='avedist'>
Average minimal distance between batches
</h2><span id='topic+avedist'></span>

<h3>Description</h3>

<p>This metric is concerned with the minimal distances between pairs of batches.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>avedist(xba, batch)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="avedist_+3A_xba">xba</code></td>
<td>

<p>matrix. The covariate matrix, raw or after batch effect adjustment. observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="avedist_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable. Each factor level (or 'category') corresponds to one of the batches. For example, if there are four batches, this variable would have four factor levels and observations with the same factor level would belong to the same batch.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For two batches j and j* (see next paragraph for the case with more batches): 1) for each observation in batch j the euclidean 
distance to the nearest observation in batch j* is calculated; 2) the roles 
of j and j* are switched and 1) is re-performed; 3) the average is taken over all n_j + n_j* minimal distances. 
</p>
<p>For more than two batches: 1) for all possible pairs of batches: calculate the metric as described above; 2) calculate
the weighted average of the values in 1) with weights proportional to the sum of the sample sizes in the two respective batches. 
</p>
<p>The variables are standardized before the calculation to make the metric independent of scale.
</p>


<h3>Value</h3>

<p>Value of the metric
</p>


<h3>Note</h3>

<p>The smaller the values of this metric, the better.
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Lazar, C., Meganck, S., Taminau, J., Steenhoff, D., Coletta, A., Molter,C., Weiss-Solís, D. Y., Duque, R., Bersini, H., Nowé, A. (2012). Batch effect removal methods for microarray gene expression data integration: a survey. Briefings in Bioinformatics 14(4):469-490, &lt;doi: <a href="https://doi.org/10.1093/bib/bbs037">10.1093/bib/bbs037</a>&gt;.
</p>
<p>Hornung, R., Boulesteix, A.-L., Causeur, D. (2016). Combining location-and-scale batch effect adjustment with data cleaning by latent factor adjustment. BMC Bioinformatics 17:27, &lt;doi: <a href="https://doi.org/10.1186/s12859-015-0870-z">10.1186/s12859-015-0870-z</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

avedist(xba=X, batch=batch)
</code></pre>

<hr>
<h2 id='ba'>
Batch effect adjustment using a method of choice
</h2><span id='topic+ba'></span>

<h3>Description</h3>

<p>Performs batch effect adjustment using one of the following methods: FAbatch, ComBat, SVA, mean-centering, standardization, Ratio-A, Ratio-G or &quot;no batch effect adjustment&quot;. Additionally returns information necessary for addon batch effect adjustment with the respective method. The latter can be done using <code><a href="#topic+baaddon">baaddon</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ba(x, y, batch, method = c("fabatch", "combat", "sva", "meancenter", 
  "standardize", "ratioa", "ratiog", "none"), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ba_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix. observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="ba_+3A_y">y</code></td>
<td>

<p>factor. Binary target variable. Has to have two factor levels, where each of them correponds to one of the two classes of the target variable. Only used for <code>method = "fabatch"</code> and <code>method = "sva"</code>.
</p>
</td></tr>
<tr><td><code id="ba_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable. Each factor level (or 'category') corresponds to one of the batches. For example, if there are four batches, this variable would have four factor levels and observations with the same factor level would belong to the same batch.
</p>
</td></tr>
<tr><td><code id="ba_+3A_method">method</code></td>
<td>

<p>character. Batch effect adjustment method. 
</p>
</td></tr>
<tr><td><code id="ba_+3A_...">...</code></td>
<td>

<p>additional arguments to be passed to <code>fabatch</code> or <code>svaba</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is merely for convenience - a wrapper function for <code><a href="#topic+fabatch">fabatch</a></code>, <code><a href="#topic+combatba">combatba</a></code>, <code><a href="#topic+svaba">svaba</a></code>, <code><a href="#topic+meancenter">meancenter</a></code>, <code><a href="#topic+standardize">standardize</a></code>, <code><a href="#topic+ratioa">ratioa</a></code>, <code><a href="#topic+ratiog">ratiog</a></code> and <code><a href="#topic+noba">noba</a></code>.
</p>


<h3>Value</h3>

<p>The output of <code><a href="#topic+fabatch">fabatch</a></code>, <code><a href="#topic+combatba">combatba</a></code>, <code><a href="#topic+svaba">svaba</a></code>, <code><a href="#topic+meancenter">meancenter</a></code>, <code><a href="#topic+standardize">standardize</a></code>, <code><a href="#topic+ratioa">ratioa</a></code>, <code><a href="#topic+ratiog">ratiog</a></code> or <code><a href="#topic+noba">noba</a></code> respectively.
</p>


<h3>Note</h3>

<p><b>The following methods are NOT recommended in cross-study prediction settings</b>: FAbatch (<code><a href="#topic+fabatch">fabatch</a></code>), frozen SVA (<code><a href="#topic+svaba">svaba</a></code>), standardization (<code><a href="#topic+standardize">standardize</a></code>) as well as no addon batch effect adjustment (<code><a href="#topic+noba">noba</a></code>). 
</p>
<p>Given a not too small test set, <b>the following methods are recommended</b> (Hornung et al., 2016b): ComBat (<code><a href="#topic+combatba">combatba</a></code>), mean-centering (<code><a href="#topic+meancenter">meancenter</a></code>), Ratio-A (<code><a href="#topic+ratioa">ratioa</a></code>), Ratio-G (<code><a href="#topic+ratiog">ratiog</a></code>).
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Hornung, R., Boulesteix, A.-L., Causeur, D. (2016). Combining location-and-scale batch effect adjustment with data cleaning by latent factor adjustment. BMC Bioinformatics 17:27, &lt;doi: <a href="https://doi.org/10.1186/s12859-015-0870-z">10.1186/s12859-015-0870-z</a>&gt;.
</p>
<p>Hornung, R., Causeur, D., Bernau, C., Boulesteix, A.-L. (2017). Improving cross-study prediction through addon batch effect adjustment and addon normalization. Bioinformatics 33(3):397–404, &lt;doi: <a href="https://doi.org/10.1093/bioinformatics/btw650">10.1093/bioinformatics/btw650</a>&gt;.
</p>
<p>Johnson, W. E., Rabinovic, A., Li, C. (2007). Adjusting batch effects in microarray expression data using empirical bayes methods. Biostatistics 8:118-127, &lt;doi: <a href="https://doi.org/10.1093/biostatistics/kxj037">10.1093/biostatistics/kxj037</a>&gt;.
</p>
<p>Leek, J. T., Storey, J. D. (2007). Capturing Heterogeneity in Gene Expression Studies by Surrogate Variable Analysis. PLoS Genetics 3:1724-1735, &lt;doi: <a href="https://doi.org/10.1371/journal.pgen.0030161">10.1371/journal.pgen.0030161</a>&gt;.
</p>
<p>Luo, J., Schumacher, M., Scherer, A., Sanoudou, D., Megherbi, D., Davison, T., Shi, T., Tong, W., Shi, L., Hong, H., Zhao, C., Elloumi, F., Shi, W., Thomas, R., Lin, S., Tillinghast, G., Liu, G., Zhou, Y., Herman, D., Li, Y., Deng, Y., Fang, H., Bushel, P., Woods, M., Zhang, J. (2010). A comparison of batch effect removal methods for enhancement of prediction performance using maqc-ii microarray gene expression data. The Pharmacogenomics Journal 10:278-291, &lt;doi: <a href="https://doi.org/10.1038/tpj.2010.57">10.1038/tpj.2010.57</a>&gt;.
</p>
<p>Parker, H. S., Bravo, H. C., Leek, J. T. (2014). Removing batch effects for prediction problems with frozen surrogate variable analysis. PeerJ 2:e561, &lt;doi: <a href="https://doi.org/10.7717/peerj.561">10.7717/peerj.561</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

# Random subset of 150 variables:
set.seed(1234)
Xsub &lt;- X[,sample(1:ncol(X), size=150)]

# In cases of batches with more than 20 observations
# select 20 observations at random:
subinds &lt;- unlist(sapply(1:length(levels(batch)), function(x) {
  indbatch &lt;- which(batch==x)
  if(length(indbatch) &gt; 20)
    indbatch &lt;- sort(sample(indbatch, size=20))
  indbatch
}))
Xsub &lt;- Xsub[subinds,]
batchsub &lt;- batch[subinds]
ysub &lt;- y[subinds]


somemethods &lt;- c("fabatch", "combat", "meancenter", "none")

adjusteddata &lt;- list()

for(i in seq(along=somemethods)) {
  cat(paste("Adjusting using method = \"", somemethods[i], "\"", 
    sep=""), "\n")
  adjusteddata[[i]] &lt;- ba(x=Xsub, y=ysub, batch=batchsub, 
    method = somemethods[i])$xadj
}
</code></pre>

<hr>
<h2 id='baaddon'>
Addon batch effect adjustment
</h2><span id='topic+baaddon'></span>

<h3>Description</h3>

<p>Performs addon batch effect adjustment for a method of choice: takes the output of <code><a href="#topic+ba">ba</a></code> or that of one of the functions performing a specific batch effect adjustment method (e.g. <code>fabatch</code> or <code>svaba</code>) and new batch data. Then performs the respective batch effect adjustment method on the new batch data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>baaddon(params, x, batch)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="baaddon_+3A_params">params</code></td>
<td>

<p>object of class <code>fabatch</code>, <code>combat</code>, <code>svatrain</code>, <code>meancenter</code>, <code>standardize</code>, <code>ratioa</code>, <code>ratiog</code> or <code>noba</code>. Contains parameters necessary for addon batch effect adjustment according to the respective method.
</p>
</td></tr>
<tr><td><code id="baaddon_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix of the new data. Observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="baaddon_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable of the new data. Currently has to have levels: '1', '2', '3' and so on.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The adjusted covariate matrix of the test data.
</p>


<h3>Note</h3>

<p><b>The following methods are NOT recommended in cross-study prediction settings</b>: FAbatch (<code><a href="#topic+fabatch">fabatch</a></code>), frozen SVA (<code><a href="#topic+svaba">svaba</a></code>), standardization (<code><a href="#topic+standardize">standardize</a></code>) as well as no addon batch effect adjustment (<code><a href="#topic+noba">noba</a></code>). 
</p>
<p>Given a not too small test set, <b>the following methods are recommended</b> (Hornung et al., 2016b): ComBat (<code><a href="#topic+combatba">combatba</a></code>), mean-centering (<code><a href="#topic+meancenter">meancenter</a></code>), Ratio-A (<code><a href="#topic+ratioa">ratioa</a></code>), Ratio-G (<code><a href="#topic+ratiog">ratiog</a></code>).
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Hornung, R., Boulesteix, A.-L., Causeur, D. (2016). Combining location-and-scale batch effect adjustment with data cleaning by latent factor adjustment. BMC Bioinformatics 17:27, &lt;doi: <a href="https://doi.org/10.1186/s12859-015-0870-z">10.1186/s12859-015-0870-z</a>&gt;.
</p>
<p>Hornung, R., Causeur, D., Bernau, C., Boulesteix, A.-L. (2017). Improving cross-study prediction through addon batch effect adjustment and addon normalization. Bioinformatics 33(3):397–404, &lt;doi: <a href="https://doi.org/10.1093/bioinformatics/btw650">10.1093/bioinformatics/btw650</a>&gt;.
</p>
<p>Johnson, W. E., Rabinovic, A., Li, C. (2007). Adjusting batch effects in microarray expression data using empirical bayes methods. Biostatistics 8:118-127, &lt;doi: <a href="https://doi.org/10.1093/biostatistics/kxj037">10.1093/biostatistics/kxj037</a>&gt;.
</p>
<p>Leek, J. T., Storey, J. D. (2007). Capturing Heterogeneity in Gene Expression Studies by Surrogate Variable Analysis. PLoS Genetics 3:1724-1735, &lt;doi: <a href="https://doi.org/10.1371/journal.pgen.0030161">10.1371/journal.pgen.0030161</a>&gt;.
</p>
<p>Luo, J., Schumacher, M., Scherer, A., Sanoudou, D., Megherbi, D., Davison, T., Shi, T., Tong, W., Shi, L., Hong, H., Zhao, C., Elloumi, F., Shi, W., Thomas, R., Lin, S., Tillinghast, G., Liu, G., Zhou, Y., Herman, D., Li, Y., Deng, Y., Fang, H., Bushel, P., Woods, M., Zhang, J. (2010). A comparison of batch effect removal methods for enhancement of prediction performance using maqc-ii microarray gene expression data. The Pharmacogenomics Journal 10:278-291, &lt;doi: <a href="https://doi.org/10.1038/tpj.2010.57">10.1038/tpj.2010.57</a>&gt;.
</p>
<p>Parker, H. S., Bravo, H. C., Leek, J. T. (2014). Removing batch effects for prediction problems with frozen surrogate variable analysis. PeerJ 2:e561, &lt;doi: <a href="https://doi.org/10.7717/peerj.561">10.7717/peerj.561</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

# Random subset of 150 variables:
set.seed(1234)
Xsub &lt;- X[,sample(1:ncol(X), size=150)]

# In cases of batches with more than 20 observations
# select 20 observations at random:
subinds &lt;- unlist(sapply(1:length(levels(batch)), function(x) {
  indbatch &lt;- which(batch==x)
  if(length(indbatch) &gt; 20)
    indbatch &lt;- sort(sample(indbatch, size=20))
  indbatch
}))
Xsub &lt;- Xsub[subinds,]
batchsub &lt;- batch[subinds]
ysub &lt;- y[subinds]



trainind &lt;- which(batchsub %in% c(1,2))

Xsubtrain &lt;- Xsub[trainind,]
ysubtrain &lt;- ysub[trainind]
batchsubtrain &lt;- factor(as.numeric(batchsub[trainind]), levels=c(1,2))


testind &lt;- which(batchsub %in% c(3,4))

Xsubtest &lt;- Xsub[testind,]
ysubtest &lt;- ysub[testind]

batchsubtest &lt;- as.numeric(batchsub[testind])
batchsubtest[batchsubtest==3] &lt;- 1
batchsubtest[batchsubtest==4] &lt;- 2
batchsubtest &lt;- factor(batchsubtest, levels=c(1,2))



somemethods &lt;- c("fabatch", "combat", "meancenter", "none")

adjustedtestdata &lt;- list()

for(i in seq(along=somemethods)) {
  cat(paste("Adjusting training data using method = \"", somemethods[i], 
    "\"", sep=""), "\n")
  paramstemp &lt;- ba(x=Xsubtrain, y=ysubtrain, batch=batchsubtrain, 
    method = somemethods[i])
  cat(paste("Addon adjusting test data using method = \"", 
    somemethods[i], "\"", sep=""), "\n")
  adjustedtestdata[[i]] &lt;- baaddon(params=paramstemp, x=Xsubtest, 
    batch=batchsubtest)
}
</code></pre>

<hr>
<h2 id='bametric'>
Diverse metrics for quality of (adjusted) batch data
</h2><span id='topic+bametric'></span>

<h3>Description</h3>

<p>Diverse metrics measuring the severeness of batch effects in (batch effect adjusted) data or the performance of certain analyses performed using the latter. This is a wrapper function for the following functions, where each of them calculates a certain metric: <code><a href="#topic+sepscore">sepscore</a></code>, <code><a href="#topic+avedist">avedist</a></code>, <code><a href="#topic+kldist">kldist</a></code>, <code><a href="#topic+skewdiv">skewdiv</a></code>, <code><a href="#topic+pvcam">pvcam</a></code>, <code><a href="#topic+diffexprm">diffexprm</a></code> and <code><a href="#topic+corba">corba</a></code>. For details see the documentation of the latter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bametric(xba, batch, y, x, metric = c("sep", "avedist", "kldist", 
  "skew", "pvca", "diffexpr", "cor"), method, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bametric_+3A_xba">xba</code></td>
<td>

<p>matrix. The covariate matrix, raw or after batch effect adjustment. observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="bametric_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable. Each factor level (or 'category') corresponds to one of the batches. For example, if there are four batches, this variable would have four factor levels and observations with the same factor level would belong to the same batch.
</p>
</td></tr>
<tr><td><code id="bametric_+3A_y">y</code></td>
<td>

<p>factor. Binary target variable. Has to have two factor levels, where each of them correponds to one of the two classes of the target variable. Only used for <code>metric = "pvca"</code> and <code>metric = "diffexpr"</code>.
</p>
</td></tr>
<tr><td><code id="bametric_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix before batch effect adjustment. observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="bametric_+3A_metric">metric</code></td>
<td>

<p>character. The metric to use. 
</p>
</td></tr>
<tr><td><code id="bametric_+3A_method">method</code></td>
<td>

<p>character. The batch effect adjustment method to use for <code>metric = "diffexpr"</code>.
</p>
</td></tr>
<tr><td><code id="bametric_+3A_...">...</code></td>
<td>

<p>additional arguments to be passed to <code>sepscore</code> or <code>pvcam</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Value of the respective metric
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Boltz, S., Debreuve, E., Barlaud, M. (2009). High-dimensional statistical measure for region-of-interest tracking. Transactions in Image Processing 18(6):1266-1283, &lt;doi: <a href="https://doi.org/10.1109/TIP.2009.2015158">10.1109/TIP.2009.2015158</a>&gt;.
</p>
<p>Geyer, C. J., Meeden, G., D. (2005). Fuzzy and randomized confidence intervals and p-values (with discussion). Statistical Science 20(4):358-387, &lt;doi: <a href="https://doi.org/10.1214/088342305000000340">10.1214/088342305000000340</a>&gt;.
</p>
<p>Hornung, R., Boulesteix, A.-L., Causeur, D. (2016). Combining location-and-scale batch effect adjustment with data cleaning by latent factor adjustment. BMC Bioinformatics 17:27, &lt;doi: <a href="https://doi.org/10.1186/s12859-015-0870-z">10.1186/s12859-015-0870-z</a>&gt;.
</p>
<p>Lazar, C., Meganck, S., Taminau, J., Steenhoff, D., Coletta, A., Molter,C., Weiss-Solís, D. Y., Duque, R., Bersini, H., Nowé, A. (2012). Batch effect removal methods for microarray gene expression data integration: a survey. Briefings in Bioinformatics 14(4):469-490, &lt;doi: <a href="https://doi.org/10.1093/bib/bbs037">10.1093/bib/bbs037</a>&gt;.
</p>
<p>Lee, J. A., Dobbin, K. K., Ahn, J. (2014). Covariance adjustment for batch effect in gene expression data. Statistics in Medicine 33:2681-2695, &lt;doi: <a href="https://doi.org/10.1002/sim.6157">10.1002/sim.6157</a>&gt;.
</p>
<p>Li, J., Bushel, P., Chu, T.-M., Wolfinger, R.D. (2009). Principal variance components analysis: Estimating batch effects in microarray gene expression data. In: Scherer, A. (ed) Batch Effects and Noise in Microarray Experiments: Sources and Solutions, John Wiley &amp; Sons, Chichester, UK, &lt;doi: <a href="https://doi.org/10.1002/9780470685983.ch12">10.1002/9780470685983.ch12</a>&gt;.
</p>
<p>Shabalin, A. A., Tjelmeland, H., Fan, C., Perou, C. M., Nobel, A. B. (2008). Merging two gene-expression studies via cross-platform normalization. Bioinformatics 24(9):1154-1160, &lt;doi: <a href="https://doi.org/10.1093/bioinformatics/btn083">10.1093/bioinformatics/btn083</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

# Random subset of 150 variables:
set.seed(1234)
Xsub &lt;- X[,sample(1:ncol(X), size=150)]

# In cases of batches with more than 20 observations
# select 20 observations at random:
subinds &lt;- unlist(sapply(1:length(levels(batch)), function(x) {
  indbatch &lt;- which(batch==x)
  if(length(indbatch) &gt; 20)
    indbatch &lt;- sort(sample(indbatch, size=20))
  indbatch
}))
Xsub &lt;- Xsub[subinds,]
batchsub &lt;- batch[subinds]
ysub &lt;- y[subinds]




Xsubadj &lt;- ba(x=Xsub, y=ysub, batch=batchsub, method = "combat")$xadj


bametric(xba=Xsub, batch=batchsub, metric = "sep")
bametric(xba=Xsubadj, batch=batchsub, metric = "sep")

bametric(xba=Xsub, batch=batchsub, metric = "avedist")
bametric(xba=Xsubadj, batch=batchsub, metric = "avedist")

bametric(xba=Xsub, batch=batchsub, metric = "kldist")
bametric(xba=Xsubadj, batch=batchsub, metric = "kldist")

bametric(xba=Xsub, batch=batchsub, metric = "skew")
bametric(xba=Xsubadj, batch=batchsub, metric = "skew")

bametric(xba=Xsub, batch=batchsub, y=ysub, metric = "pvca")
bametric(xba=Xsubadj, batch=batchsub, y=ysub, metric = "pvca")

bametric(x=Xsub, batch=batchsub, y=ysub, metric = "diffexpr", 
  method = "combat")

bametric(xba=Xsubadj, x=Xsub, metric = "cor")
</code></pre>

<hr>
<h2 id='bapred-internal'>Internal bapred functions</h2><span id='topic+Dt'></span><span id='topic+VarInflation'></span><span id='topic+aprior'></span><span id='topic+avedistTwo'></span><span id='topic+bivprob'></span><span id='topic+bprior'></span><span id='topic+build.design'></span><span id='topic+design.mat'></span><span id='topic+diffexprmAfterBR'></span><span id='topic+emfahighdim'></span><span id='topic+extractAffybatch'></span><span id='topic+fuzzywilcox'></span><span id='topic+it.sol'></span><span id='topic+kldistTwo'></span><span id='topic+list.batch'></span><span id='topic+mypvcaBatchAssess'></span><span id='topic+nbfactors'></span><span id='topic+normalizeAffyBatchqntval'></span><span id='topic+normalizeqntadd'></span><span id='topic+normalizeqntadd2'></span><span id='topic+normalizeqntval'></span><span id='topic+plotcomp'></span><span id='topic+postmean'></span><span id='topic+postvar'></span><span id='topic+print.combat'></span><span id='topic+print.fabatch'></span><span id='topic+print.meancenter'></span><span id='topic+print.noba'></span><span id='topic+print.qunormtrain'></span><span id='topic+print.ratioa'></span><span id='topic+print.ratiog'></span><span id='topic+print.rmatrain'></span><span id='topic+print.standardize'></span><span id='topic+print.svatrain'></span><span id='topic+sepscoreTwo'></span><span id='topic+skewdivTwo'></span><span id='topic+summarizeadd2'></span><span id='topic+summarizeval2'></span>

<h3>Description</h3>

<p>Internal bapred functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Dt(rho) 
VarInflation(dta, Blist, maxnbfactors, dig) 
aprior(gamma.hat) 
avedistTwo(xb1, xb2) 
bivprob(rho, lower, upper = -lower, mean = 0) 
bprior(gamma.hat) 
build.design(vec, des = NULL, start = 2) 
design.mat(mod, numCov) 
diffexprmAfterBR(x, xbrlist, y, batch, batchessuited) 
emfahighdim(eps, nbf, minerr = 1e-06, maxiter = 100) 
extractAffybatch(inds, abo)
fuzzywilcox(x, y) 
it.sol(sdat, g.hat, d.hat, g.bar, t2, a, b, conv = 1e-04) 
kldistTwo(xb1, xb2) 
list.batch(mod) 
mypvcaBatchAssess(X, factordata, threshold) 
nbfactors(dta, maxnbfactors = 12, diagnostic.plot = FALSE, minerr = 0.001, 
  maxiter = 100) 
normalizeAffyBatchqntval(abatch, type = c("separate", "pmonly", "mmonly", "together")) 
normalizeqntadd(abo, qnt.scale)
normalizeqntadd2(x, qnt.scale)
normalizeqntval(x)
plotcomp(xp, groups, y, alpha, col = rainbow(length(groupsun)), 
  xlim, ylim, ...) 
postmean(g.hat, g.bar, n, d.star, t2) 
postvar(sum2, n, a, b) 
## S3 method for class 'combat'
print(x, ...) 
## S3 method for class 'fabatch'
print(x, ...) 
## S3 method for class 'meancenter'
print(x, ...) 
## S3 method for class 'noba'
print(x, ...) 
## S3 method for class 'qunormtrain'
print(x, ...) 
## S3 method for class 'ratioa'
print(x, ...) 
## S3 method for class 'ratiog'
print(x, ...) 
## S3 method for class 'rmatrain'
print(x, ...)
## S3 method for class 'standardize'
print(x, ...) 
## S3 method for class 'svatrain'
print(x, ...) 
sepscoreTwo(xb1, xb2, k = 10) 
skewdivTwo(xb1, xb2)
summarizeadd2(abo, probe.effects) 
summarizeval2(abo)
</code></pre>


<h3>Details</h3>

<p>These internal <span class="pkg">bapred</span> functions are not usually called
directly by the user.
</p>

<hr>
<h2 id='batch'>
batch variable of dataset <code>autism</code>
</h2><span id='topic+batch'></span>

<h3>Description</h3>

<p>See dataset <code><a href="#topic+autism">autism</a></code>
</p>

<hr>
<h2 id='combatba'>
Batch effect adjustment using ComBat
</h2><span id='topic+combatba'></span>

<h3>Description</h3>

<p>Performs batch effect adjustment using the parametric version of ComBat and additionally returns information necessary for addon batch effect adjustment with ComBat.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combatba(x, batch)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="combatba_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix. Observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="combatba_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable. Each factor level (or 'category') corresponds to one of the batches. For example, if there are four batches, this variable would have four factor levels and observations with the same factor level would belong to the same batch.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>combatba</code> returns an object of class <code>combat</code>.
An object of class &quot;<code>combat</code>&quot; is a list containing the following components: 
</p>
<table role = "presentation">
<tr><td><code>xadj</code></td>
<td>
<p> matrix of adjusted (training) data </p>
</td></tr>
<tr><td><code>meanoverall</code></td>
<td>
<p> vector containing the overall means of the variables. Used in addon adjustment. </p>
</td></tr>
<tr><td><code>var.pooled</code></td>
<td>
<p> vector containing the pooled variances of the variables. Used in addon adjustment. </p>
</td></tr>
<tr><td><code>batch</code></td>
<td>
<p> batch variable </p>
</td></tr>
<tr><td><code>nbatches</code></td>
<td>
<p> number of batches </p>
</td></tr>
</table>


<h3>Note</h3>

<p>The original ComBat-code is used in <code>combatba</code>:
<a href="http://www.bu.edu/jlab/wp-assets/ComBat/Download.html">http://www.bu.edu/jlab/wp-assets/ComBat/Download.html</a> (Access date: 2015/06/19)
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Johnson, W. E., Rabinovic, A., Li, C. (2007). Adjusting batch effects in microarray expression data using empirical bayes methods. Biostatistics 8:118-127, &lt;doi: <a href="https://doi.org/10.1093/biostatistics/kxj037">10.1093/biostatistics/kxj037</a>&gt;.
</p>
<p>Luo, J., Schumacher, M., Scherer, A., Sanoudou, D., Megherbi, D., Davison, T., Shi, T., Tong, W., Shi, L., Hong, H., Zhao, C., Elloumi, F., Shi, W., Thomas, R., Lin, S., Tillinghast, G., Liu, G., Zhou, Y., Herman, D., Li, Y., Deng, Y., Fang, H., Bushel, P., Woods, M., Zhang, J. (2010). A comparison of batch effect removal methods for enhancement of prediction performance using maqc-ii microarray gene expression data. The Pharmacogenomics Journal 10:278-291, &lt;doi: <a href="https://doi.org/10.1038/tpj.2010.57">10.1038/tpj.2010.57</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

combatba(x=X, batch=batch)
</code></pre>

<hr>
<h2 id='combatbaaddon'>
Addon batch effect adjustment using ComBat
</h2><span id='topic+combatbaaddon'></span>

<h3>Description</h3>

<p>Performs addon batch effect adjustment using ComBat. Takes the output of performing <code><a href="#topic+combatba">combatba</a></code> on a training data set and new batch data and correspondingly adjusts the test data to better match the adjusted training data according to the ComBat model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combatbaaddon(params, x, batch)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="combatbaaddon_+3A_params">params</code></td>
<td>

<p>object of class <code>combat</code>. Contains parameters necessary for addon batch effect adjustment.
</p>
</td></tr>
<tr><td><code id="combatbaaddon_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix of the new data. Observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="combatbaaddon_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable of the new data. Currently has to have levels: '1', '2', '3' and so on.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The adjusted covariate matrix of the test data.
</p>


<h3>Note</h3>

<p>The original ComBat-code is used in <code>combatbaaddon</code>: 
<a href="http://www.bu.edu/jlab/wp-assets/ComBat/Download.html">http://www.bu.edu/jlab/wp-assets/ComBat/Download.html</a> (Access date: 2015/06/19)
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Johnson, W. E., Rabinovic, A., Li, C. (2007). Adjusting batch effects in microarray expression data using empirical bayes methods. Biostatistics 8:118-127, &lt;doi: <a href="https://doi.org/10.1093/biostatistics/kxj037">10.1093/biostatistics/kxj037</a>&gt;.
</p>
<p>Luo, J., Schumacher, M., Scherer, A., Sanoudou, D., Megherbi, D., Davison, T., Shi, T., Tong, W., Shi, L., Hong, H., Zhao, C., Elloumi, F., Shi, W., Thomas, R., Lin, S., Tillinghast, G., Liu, G., Zhou, Y., Herman, D., Li, Y., Deng, Y., Fang, H., Bushel, P., Woods, M., Zhang, J. (2010). A comparison of batch effect removal methods for enhancement of prediction performance using maqc-ii microarray gene expression data. The Pharmacogenomics Journal 10:278-291, &lt;doi: <a href="https://doi.org/10.1038/tpj.2010.57">10.1038/tpj.2010.57</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

trainind &lt;- which(batch %in% c(1,2))

Xtrain &lt;- X[trainind,]
ytrain &lt;- y[trainind]
batchtrain &lt;- factor(as.numeric(batch[trainind]), levels=c(1,2))


testind &lt;- which(batch %in% c(3,4))

Xtest &lt;- X[testind,]
ytest &lt;- y[testind]

batchtest &lt;- as.numeric(batch[testind])
batchtest[batchtest==3] &lt;- 1
batchtest[batchtest==4] &lt;- 2
batchtest &lt;- factor(batchtest, levels=c(1,2))


params &lt;- combatba(x=Xtrain, batch=batchtrain)

Xtestaddon &lt;- combatbaaddon(params, x=Xtest, batch=batchtest)
</code></pre>

<hr>
<h2 id='corba'>
Mean correlation before and after batch effect adjustment
</h2><span id='topic+corba'></span>

<h3>Description</h3>

<p>For each variable Pearson's correlation of the values before and after batch effect adjustment is calculated. Then the mean is taken over all these correlations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>corba(xba, x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="corba_+3A_xba">xba</code></td>
<td>

<p>matrix. The covariate matrix after batch effect adjustment. observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="corba_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix before batch effect adjustment. observations in rows, variables in columns.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Value of the metric
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Lazar, C., Meganck, S., Taminau, J., Steenhoff, D., Coletta, A., Molter,C., Weiss-Solís, D. Y., Duque, R., Bersini, H., Nowé, A. (2012). Batch effect removal methods for microarray gene expression data integration: a survey. Briefings in Bioinformatics 14(4):469-490, &lt;doi: <a href="https://doi.org/10.1093/bib/bbs037">10.1093/bib/bbs037</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

params &lt;- ba(x=X, y=y, batch=batch, method = "combat")
Xadj &lt;- params$xadj

corba(xba=Xadj, x=X)
</code></pre>

<hr>
<h2 id='diffexprm'>
Measure for performance of differential expression analysis (after batch effect adjustment)
</h2><span id='topic+diffexprm'></span>

<h3>Description</h3>

<p>This metric is similar to the idea presented in Lazar et al (2012) which consists 
in comparing the list of the most differentially expressed genes obtained using a batch 
effect adjusted dataset to the list obtained using an independent dataset. For each 
batch the following is done by <code>diffexprm</code>: 1) the respective batch is left out
and batch effect adjustment 
is performed using the remaining batches; 2) differential expression analysis is performed 
once using the left-out batch and once using the remaining batch-effect adjusted data; 
3) the overlap between the two lists of genes found differentially expressed in the 
two subsets is measured. See below for further details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diffexprm(x, batch, y, method = c("fabatch", "combat", "sva", 
  "meancenter", "standardize", "ratioa", "ratiog", "none"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="diffexprm_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix. Observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="diffexprm_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable. Each factor level (or 'category') corresponds to one of the batches. For example, if there are four batches, this variable would have four factor levels and observations with the same factor level would belong to the same batch.
</p>
</td></tr>
<tr><td><code id="diffexprm_+3A_y">y</code></td>
<td>

<p>factor. Binary target variable. Has to have two factor levels, where each of them correponds to one of the two classes of the target variable.
</p>
</td></tr>
<tr><td><code id="diffexprm_+3A_method">method</code></td>
<td>

<p>character. Method for batch effect adjustment. The following are supported: <code>fabatch</code>, <code>combat</code>, <code>fsva</code>, <code>meancenter</code>, <code>standardize</code>, <code>ratioa</code>, <code>ratiog</code> and <code>none</code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following procedure is performed: 1) For each batch j leave this batch out and 
perform batch effect adjustment on the rest of the dataset. Derive two lists of the 5 percent 
of variables which are most differentially expressed (see next paragraph): one using the batch 
effect adjusted dataset - where batch j was left out - and one using the data from batch j. 
Calculate the number of variables appearing in both lists and divide this number by the common 
length of the lists. 2) Calculate a weighted average of the values obtained in 1) with weights 
proportional to the number of observations in the corresponding left-out batches. 
</p>
<p>Differential expression is measured as follows. For each variable a randomized p-value out 
of the Whitney-Wilcoxon rank sum test is drawn, see Geyer and Meeden (2005) for details. 
Then those 5 percent variables are considered differentially expressed, which are associated 
with the smallest p-values.
</p>


<h3>Value</h3>

<p>Value of the metric
</p>


<h3>Note</h3>

<p>The larger the values of this metric, the better.
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Hornung, R., Boulesteix, A.-L., Causeur, D. (2016). Combining location-and-scale batch effect adjustment with data cleaning by latent factor adjustment. BMC Bioinformatics 17:27, &lt;doi: <a href="https://doi.org/10.1186/s12859-015-0870-z">10.1186/s12859-015-0870-z</a>&gt;.
</p>
<p>Lazar, C., Meganck, S., Taminau, J., Steenhoff, D., Coletta, A., Molter,C., Weiss-Solís, D. Y., Duque, R., Bersini, H., Nowé, A. (2012). Batch effect removal methods for microarray gene expression data integration: a survey. Briefings in Bioinformatics 14(4):469-490, &lt;doi: <a href="https://doi.org/10.1093/bib/bbs037">10.1093/bib/bbs037</a>&gt;.
</p>
<p>Geyer, C. J., Meeden, G., D. (2005). Fuzzy and randomized confidence intervals and p-values (with discussion). Statistical Science 20(4):358-387, &lt;doi: <a href="https://doi.org/10.1214/088342305000000340">10.1214/088342305000000340</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

# Random subset of 150 variables:
set.seed(1234)
Xsub &lt;- X[,sample(1:ncol(X), size=150)]

# In cases of batches with more than 20 observations
# select 20 observations at random:
subinds &lt;- unlist(sapply(1:length(levels(batch)), function(x) {
  indbatch &lt;- which(batch==x)
  if(length(indbatch) &gt; 20)
    indbatch &lt;- sort(sample(indbatch, size=20))
  indbatch
}))
Xsub &lt;- Xsub[subinds,]
batchsub &lt;- batch[subinds]
ysub &lt;- y[subinds]



diffexprm(x=Xsub, batch=batchsub, y=ysub, method = "ratiog")
diffexprm(x=Xsub, batch=batchsub, y=ysub, method = "none")
</code></pre>

<hr>
<h2 id='fabatch'>
Batch effect adjustment using FAbatch
</h2><span id='topic+fabatch'></span>

<h3>Description</h3>

<p>Performs batch effect adjustment using the FAbatch-method described in Hornung et al. (2016) and additionally returns information necessary for addon batch effect adjustment with FAbatch.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fabatch(x, y, batch, nbf = NULL, minerr = 1e-06, 
  probcrossbatch = TRUE, maxiter = 100, maxnbf = 12)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fabatch_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix. Observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="fabatch_+3A_y">y</code></td>
<td>

<p>factor. Binary target variable. Has to have two factor levels, where each of them correponds to one of the two classes of the target variable.
</p>
</td></tr>
<tr><td><code id="fabatch_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable. Each factor level (or 'category') corresponds to one of the batches. For example, if there are four batches, this variable would have four factor levels and observations with the same factor level would belong to the same batch.
</p>
</td></tr>
<tr><td><code id="fabatch_+3A_nbf">nbf</code></td>
<td>

<p>integer. Number of factors to estimate in all batches. If not given the number of factors is estimated automatically for each batch. Recommended to leave unspecified.
</p>
</td></tr>
<tr><td><code id="fabatch_+3A_minerr">minerr</code></td>
<td>

<p>numeric. Maximal mean quadratic deviations between the estimated residual variances from two consecutive iterations. The iteration stops when this value is undercut.
</p>
</td></tr>
<tr><td><code id="fabatch_+3A_probcrossbatch">probcrossbatch</code></td>
<td>

<p>logical. Default is <code>TRUE</code>. If <code>TRUE</code> the preliminary probabilities are estimated through leave-one-batch-out cross-validation. If set to <code>FALSE</code> ordinary cross-validation is used for estimating the preliminary probabilities. This might result in an artificially increased class signal in comparison to that in the data in independent batches. Is automatically set to <code>FALSE</code>, when only one batch is present in the training data.
</p>
</td></tr>
<tr><td><code id="fabatch_+3A_maxiter">maxiter</code></td>
<td>

<p>integer. Maximal number of iterations in the estimation of the latent factors by Maximum Likelihood.
</p>
</td></tr>
<tr><td><code id="fabatch_+3A_maxnbf">maxnbf</code></td>
<td>

<p>integer. Maximal number of factors if <code>nbf</code> is not given. Default is the largest integer smaller than half the number of observations in the corresponding batch.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>fabatch</code> returns an object of class <code>fabatch</code>.
An object of class &quot;<code>fabatch</code>&quot; is a list containing the following components: 
</p>
<table role = "presentation">
<tr><td><code>xadj</code></td>
<td>
<p> matrix of adjusted (training) data </p>
</td></tr>
<tr><td><code>m1</code></td>
<td>
<p> means of the standardized variables in class '1' </p>
</td></tr>
<tr><td><code>m2</code></td>
<td>
<p> means of the standardized variables in class '2' </p>
</td></tr>
<tr><td><code>b0</code></td>
<td>
<p> intercept out of the L2-penalized logistic regression performed for estimation of the class probabilities </p>
</td></tr>
<tr><td><code>b</code></td>
<td>
<p> variable coefficients out of the L2-penalized logistic regression performed for estimation of the class probabilities </p>
</td></tr>
<tr><td><code>pooledsds</code></td>
<td>
<p> vector containing the pooled standard deviations of the variables </p>
</td></tr>
<tr><td><code>meanoverall</code></td>
<td>
<p> vector containing the variable means </p>
</td></tr>
<tr><td><code>minerr</code></td>
<td>
<p> maximal mean quadratic deviations between the estimated residual variances from two consecutive iterations </p>
</td></tr>
<tr><td><code>nbfinput</code></td>
<td>
<p> user-specified number of latent factors <code>nbf</code> in all batches. <code>NULL</code> if <code>nbf</code> was not specified. </p>
</td></tr>
<tr><td><code>badvariables</code></td>
<td>
<p> indices of those variables which are constant in at least one batch </p>
</td></tr>
<tr><td><code>nbatches</code></td>
<td>
<p> number of batches </p>
</td></tr>
<tr><td><code>batch</code></td>
<td>
<p> batch variable </p>
</td></tr>
<tr><td><code>nbfvec</code></td>
<td>
<p> vector containing the numbers of factors in the individual batches </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Hornung, R., Boulesteix, A.-L., Causeur, D. (2016). Combining location-and-scale batch effect adjustment with data cleaning by latent factor adjustment. BMC Bioinformatics 17:27, &lt;doi: <a href="https://doi.org/10.1186/s12859-015-0870-z">10.1186/s12859-015-0870-z</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

# Random subset of 150 variables:
set.seed(1234)
Xsub &lt;- X[,sample(1:ncol(X), size=150)]

# In cases of batches with more than 20 observations
# select 20 observations at random:
subinds &lt;- unlist(sapply(1:length(levels(batch)), function(x) {
  indbatch &lt;- which(batch==x)
  if(length(indbatch) &gt; 20)
    indbatch &lt;- sort(sample(indbatch, size=20))
  indbatch
}))
Xsub &lt;- Xsub[subinds,]
batchsub &lt;- batch[subinds]
ysub &lt;- y[subinds]



fabatch(x=Xsub, y=ysub, batch=batchsub)
</code></pre>

<hr>
<h2 id='fabatchaddon'>
Addon batch effect adjustment using FAbatch
</h2><span id='topic+fabatchaddon'></span>

<h3>Description</h3>

<p>Performs addon batch effect adjustment using FAbatch. Takes the output of performing <code><a href="#topic+fabatch">fabatch</a></code> on a training data set and new batch data and correspondingly adjusts the test data to better match the adjusted training data according to the FAbatch model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fabatchaddon(params, x, batch)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fabatchaddon_+3A_params">params</code></td>
<td>

<p>object of class <code>fabatch</code>. Contains parameters necessary for addon batch effect adjustment.
</p>
</td></tr>
<tr><td><code id="fabatchaddon_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix of the new data. Observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="fabatchaddon_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable of the new data. Currently has to have levels: '1', '2', '3' and so on.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The adjusted covariate matrix of the test data.
</p>


<h3>Note</h3>

<p>It is <b>not recommended</b> to perform FAbatch in cross-study prediction settings, because it has been observed to (strongly) impair prediction performance. Given a not too small test set, the following methods are recommended (Hornung et al., 2016b): <code><a href="#topic+combatba">combatba</a></code>, <code><a href="#topic+meancenter">meancenter</a></code>, <code><a href="#topic+ratioa">ratioa</a></code>, <code><a href="#topic+ratiog">ratiog</a></code>.
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Hornung, R., Boulesteix, A.-L., Causeur, D. (2016). Combining location-and-scale batch effect adjustment with data cleaning by latent factor adjustment. BMC Bioinformatics 17:27, &lt;doi: <a href="https://doi.org/10.1186/s12859-015-0870-z">10.1186/s12859-015-0870-z</a>&gt;.
</p>
<p>Hornung, R., Causeur, D., Bernau, C., Boulesteix, A.-L. (2017). Improving cross-study prediction through addon batch effect adjustment and addon normalization. Bioinformatics 33(3):397–404, &lt;doi: <a href="https://doi.org/10.1093/bioinformatics/btw650">10.1093/bioinformatics/btw650</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

# Random subset of 150 variables:
set.seed(1234)
Xsub &lt;- X[,sample(1:ncol(X), size=150)]

# In cases of batches with more than 20 observations
# select 20 observations at random:
subinds &lt;- unlist(sapply(1:length(levels(batch)), function(x) {
  indbatch &lt;- which(batch==x)
  if(length(indbatch) &gt; 20)
    indbatch &lt;- sort(sample(indbatch, size=20))
  indbatch
}))
Xsub &lt;- Xsub[subinds,]
batchsub &lt;- batch[subinds]
ysub &lt;- y[subinds]



trainind &lt;- which(batchsub %in% c(1,2))

Xsubtrain &lt;- Xsub[trainind,]
ysubtrain &lt;- ysub[trainind]
batchsubtrain &lt;- factor(as.numeric(batchsub[trainind]), levels=c(1,2))


testind &lt;- which(batchsub %in% c(3,4))

Xsubtest &lt;- Xsub[testind,]
ysubtest &lt;- ysub[testind]

batchsubtest &lt;- as.numeric(batchsub[testind])
batchsubtest[batchsubtest==3] &lt;- 1
batchsubtest[batchsubtest==4] &lt;- 2
batchsubtest &lt;- factor(batchsubtest, levels=c(1,2))



params &lt;- fabatch(x=Xsubtrain, y=ysubtrain, batch=batchsubtrain)

Xsubtestaddon &lt;- fabatchaddon(params, x=Xsubtest, 
  batch=batchsubtest)
</code></pre>

<hr>
<h2 id='kldist'>
Kullback-Leibler divergence between density of within and between batch pairwise distances
</h2><span id='topic+kldist'></span>

<h3>Description</h3>

<p>This metric estimates the Kullback-Leibler divergences between the distributions 
of the within and that of the between batch euclidian distances of pairs of 
observations. These distributions should be similar in the abscence of stronger 
batch effects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kldist(xba, batch)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="kldist_+3A_xba">xba</code></td>
<td>

<p>matrix. The covariate matrix, raw or after batch effect adjustment. observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="kldist_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable. Each factor level (or 'category') corresponds to one of the batches. For example, if there are four batches, this variable would have four factor levels and observations with the same factor level would belong to the same batch.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For two batches j and j* (see next paragraph for the case with more batches): 1) the distances between all pairs of observations in batch j - denoted as {dist_j} - and 
the distances between all such pairs in batch j* - denoted as {dist_j*} - are calculated; 
2) for each observation in j the distances to all observations in j* are calculated, 
resulting in n_j x n_j* distances denoted as {dist_jj*}; calculate the 
Kullback-Leibler divergence between the densities of {dist_j} and {dist_jj*} and that 
between the densities of {dist_j*} and {dist_jj*} - using the k-nearest neighbours based 
method by Boltz et al (2009) with k=5; 3) take the weighted mean of the values of 
these two divergences with weights proportional to n_j and n_j*. 
</p>
<p>For more than two batches: 1) for all possible pairs of batches: calculate the metric as described above; 2) calculate
the weighted average of the values in 1) with weights proportional to the sum of the sample sizes in the two respective batches. 
</p>
<p>The variables are standardized before the calculation to make the metric independent of scale.
</p>


<h3>Value</h3>

<p>Value of the metric
</p>


<h3>Note</h3>

<p>The smaller the values of this metric, the better.
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Lee, J. A., Dobbin, K. K., Ahn, J. (2014). Covariance adjustment for batch effect in gene expression data. Statistics in Medicine 33:2681-2695, &lt;doi: <a href="https://doi.org/10.1002/sim.6157">10.1002/sim.6157</a>&gt;.
</p>
<p>Boltz, S., Debreuve, E., Barlaud, M. (2009). High-dimensional statistical measure for region-of-interest tracking. Transactions in Image Processing 18(6):1266-1283, &lt;doi: <a href="https://doi.org/10.1109/TIP.2009.2015158">10.1109/TIP.2009.2015158</a>&gt;.
</p>
<p>Hornung, R., Boulesteix, A.-L., Causeur, D. (2016). Combining location-and-scale batch effect adjustment with data cleaning by latent factor adjustment. BMC Bioinformatics 17:27, &lt;doi: <a href="https://doi.org/10.1186/s12859-015-0870-z">10.1186/s12859-015-0870-z</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

kldist(xba=X, batch=batch)
</code></pre>

<hr>
<h2 id='meancenter'>
Batch effect adjustment by mean-centering
</h2><span id='topic+meancenter'></span>

<h3>Description</h3>

<p>Performs batch effect adjustment by centering the variables within batches to have zero mean.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meancenter(x, batch)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="meancenter_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix of the new data. Observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="meancenter_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable. Each factor level (or 'category') corresponds to one of the batches. For example, if there are four batches, this variable would have four factor levels and observations with the same factor level would belong to the same batch.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>meancenter</code> returns an object of class <code>meancenter</code>.
An object of class &quot;<code>meancenter</code>&quot; is a list containing the following components: 
</p>
<table role = "presentation">
<tr><td><code>xadj</code></td>
<td>
<p> matrix of adjusted (training) data </p>
</td></tr>
<tr><td><code>nbatches</code></td>
<td>
<p> number of batches </p>
</td></tr>
<tr><td><code>batch</code></td>
<td>
<p> batch variable </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

params &lt;- meancenter(x=X, batch=batch)
</code></pre>

<hr>
<h2 id='meancenteraddon'>
Addon batch effect adjustment for mean-centering
</h2><span id='topic+meancenteraddon'></span>

<h3>Description</h3>

<p>Performs addon batch effect adjustment for mean-centering: 
1) takes the output of <code><a href="#topic+meancenter">meancenter</a></code> applied to a training 
data set together with new batch data; 
2) checks whether the training data was also adjusted using mean-centering and whether the same number of variables is present in training and new data;
3) performs mean-centering on the new batch data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meancenteraddon(params, x, batch)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="meancenteraddon_+3A_params">params</code></td>
<td>

<p>object of class <code>meancenter</code>.
</p>
</td></tr>
<tr><td><code id="meancenteraddon_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix of the new data. Observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="meancenteraddon_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable of the new data. Currently has to have levels: '1', '2', '3' and so on.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The adjusted covariate matrix of the test data.
</p>


<h3>Note</h3>

<p>Because mean-centering is performed &quot;batch by batch&quot; the &quot;addon procedure&quot; for mean-centering consists of plain mean-centering on the new test batches.
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Hornung, R., Boulesteix, A.-L., Causeur, D. (2016). Combining location-and-scale batch effect adjustment with data cleaning by latent factor adjustment. BMC Bioinformatics 17:27, &lt;doi: <a href="https://doi.org/10.1186/s12859-015-0870-z">10.1186/s12859-015-0870-z</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

trainind &lt;- which(batch %in% c(1,2))

Xtrain &lt;- X[trainind,]
ytrain &lt;- y[trainind]
batchtrain &lt;- factor(as.numeric(batch[trainind]), levels=c(1,2))


testind &lt;- which(batch %in% c(3,4))

Xtest &lt;- X[testind,]
ytest &lt;- y[testind]

batchtest &lt;- as.numeric(batch[testind])
batchtest[batchtest==3] &lt;- 1
batchtest[batchtest==4] &lt;- 2
batchtest &lt;- factor(batchtest, levels=c(1,2))


params &lt;- meancenter(x=Xtrain, batch=batchtrain)

Xtestaddon &lt;- meancenteraddon(params=params, x=Xtest, batch=batchtest)
</code></pre>

<hr>
<h2 id='noba'>
No batch effect adjustment
</h2><span id='topic+noba'></span>

<h3>Description</h3>

<p>This function is merely included for consistency. It returns the raw dataset not adjusted for batch effects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>noba(x, batch)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="noba_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix. Observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="noba_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable. Each factor level (or 'category') corresponds to one of the batches. For example, if there are four batches, this variable would have four factor levels and observations with the same factor level would belong to the same batch.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>noba</code> returns an object of class <code>noba</code>.
An object of class &quot;<code>noba</code>&quot; is a list containing the following components: 
</p>
<table role = "presentation">
<tr><td><code>xadj</code></td>
<td>
<p> matrix of (training) data </p>
</td></tr>
<tr><td><code>nbatches</code></td>
<td>
<p> number of batches </p>
</td></tr>
<tr><td><code>batch</code></td>
<td>
<p> batch variable </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

Xadj &lt;- noba(x=X, batch=batch)$adj

all(as.vector(Xadj)==as.vector(X))
</code></pre>

<hr>
<h2 id='nobaaddon'>
No addon batch effect adjustment
</h2><span id='topic+nobaaddon'></span>

<h3>Description</h3>

<p>This function is merely included for consistency. It does the following:
1) takes the output of <code><a href="#topic+noba">noba</a></code> applied to a training 
data set together with new batch data; 
2) checks whether the training data has also not been adjusted using a batch effect adjustment method and whether the same number of variables is present in training and new data;
3) returns the new batch data not adjusted for batch effects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nobaaddon(params, x, batch)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nobaaddon_+3A_params">params</code></td>
<td>

<p>object of class <code>noba</code>.
</p>
</td></tr>
<tr><td><code id="nobaaddon_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix of the new data. Observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="nobaaddon_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable of the new data. Currently has to have levels: '1', '2', '3' and so on.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The unadjusted covariate matrix <code>x</code> of the test data.
</p>


<h3>Note</h3>

<p>It is <b>not recommended</b> to perform no addon batch effect adjustment in cross-study prediction settings. Given a not too small test set, the following methods are recommended (Hornung et al., 2016): <code><a href="#topic+combatba">combatba</a></code>, <code><a href="#topic+meancenter">meancenter</a></code>, <code><a href="#topic+ratioa">ratioa</a></code>, <code><a href="#topic+ratiog">ratiog</a></code>.
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Hornung, R., Causeur, D., Bernau, C., Boulesteix, A.-L. (2017). Improving cross-study prediction through addon batch effect adjustment and addon normalization. Bioinformatics 33(3):397–404, &lt;doi: <a href="https://doi.org/10.1093/bioinformatics/btw650">10.1093/bioinformatics/btw650</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

trainind &lt;- which(batch %in% c(1,2))

Xtrain &lt;- X[trainind,]
ytrain &lt;- y[trainind]
batchtrain &lt;- factor(as.numeric(batch[trainind]), levels=c(1,2))


testind &lt;- which(batch %in% c(3,4))

Xtest &lt;- X[testind,]
ytest &lt;- y[testind]

batchtest &lt;- as.numeric(batch[testind])
batchtest[batchtest==3] &lt;- 1
batchtest[batchtest==4] &lt;- 2
batchtest &lt;- factor(batchtest, levels=c(1,2))


params &lt;- noba(x=Xtrain, batch=batchtrain)

Xtestaddon &lt;- nobaaddon(params=params, x=Xtest, batch=batchtest)


all(as.vector(Xtestaddon)==as.vector(Xtest))
</code></pre>

<hr>
<h2 id='pcplot'>
Visualization of batch effects using Principal Component Analysis
</h2><span id='topic+pcplot'></span>

<h3>Description</h3>

<p>This function performs principal component analysis on the covariate matrix and plots the first two principal components against each other. Different batches are distinguished by different colors and (optionally) the two classes of the target variable by different plot symbols.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pcplot(x, batch, y, alpha = 0.35, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pcplot_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix. Observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="pcplot_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable. Each factor level (or 'category') corresponds to one of the batches. For example, if there are four batches, this variable would have four factor levels and observations with the same factor level would belong to the same batch.
</p>
</td></tr>
<tr><td><code id="pcplot_+3A_y">y</code></td>
<td>

<p>optional factor. Binary target variable. Has to have two factor levels, where each of them correponds to one of the two classes of the target variable.
</p>
</td></tr>
<tr><td><code id="pcplot_+3A_alpha">alpha</code></td>
<td>

<p>optional numeric between 0 and 1. Alpha transparency of the contour lines of the batch-specific two-dimensional density estimates. Only applicable when default color scheme (<code>rainbow</code>) is used.
</p>
</td></tr>
<tr><td><code id="pcplot_+3A_...">...</code></td>
<td>

<p>additional arguments to be passed to <code>plot</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For the data corresponding to each batch a two-dimensional kernel density estimate is obtained using the function <code>kde2d()</code> from the <span class="pkg">MASS</span>-package. These estimates are depicted through contour lines (using <code>contour</code>). 
</p>


<h3>Value</h3>

<p><code>NULL</code>
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

par(mfrow=c(1,3))
pcplot(x=X, batch=batch, y=y, alpha=0.25, main="alpha = 0.25")
pcplot(x=X, batch=batch, y=y, alpha=0.75, main="alpha = 0.75")
pcplot(x=X, batch=batch, y=y, col=1:length(unique(batch)), 
  main="col = 1:length(unique(batch))")
par(mfrow=c(1,1))
</code></pre>

<hr>
<h2 id='pvcam'>
Proportion of variation induced by class signal estimated by Principal Variance Component Analysis
</h2><span id='topic+pvcam'></span>

<h3>Description</h3>

<p>Principal Variance Component Analysis (PVCA) (Li et al, 2009) allows the estimation of the contribution of several sources of variability. <code>pvcam</code> uses it to estimate the proportion of variance in the data explained by the class signal. See below for a more detailed explanation of what the function does.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pvcam(xba, batch, y, threshold = 0.6)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pvcam_+3A_xba">xba</code></td>
<td>

<p>matrix. The covariate matrix, raw or after batch effect adjustment. observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="pvcam_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable. Each factor level (or 'category') corresponds to one of the batches. For example, if there are four batches, this variable would have four factor levels and observations with the same factor level would belong to the same batch.
</p>
</td></tr>
<tr><td><code id="pvcam_+3A_y">y</code></td>
<td>

<p>factor. Binary target variable. Has to have two factor levels, where each of them correponds to one of the two classes of the target variable.
</p>
</td></tr>
<tr><td><code id="pvcam_+3A_threshold">threshold</code></td>
<td>

<p>numeric. Minimal proportion of variance explained by the principal components used.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In PVCA, first principal component analysis is performed on the n x n covariance matrix between the
observations. Then, using a random effects model
the principal components are regressed on arbitrary factors of variability, such
as &quot;batch&quot; and &quot;(phenotype) class&quot;. Ultimately, estimated proportions of variance induced by each factor and that of the residual variance are obtained. 
In <code>pvcam</code> the factors included into the model are: &quot;batch&quot;, &quot;class&quot; and the interaction of these two into.
The metric calculated by <code>pvcam</code> is the proportion of variance
explained by &quot;class&quot;.
</p>
<p><code>pvcam</code> uses a slightly altered version of the function <code>pvcaBatchAssess()</code> from the Bioconductor package <code>pvca</code>.
The latter was altered to take the covariate data as a <code>matrix</code> instead of as an object of class <code>ExpressionSet</code>.
</p>


<h3>Value</h3>

<p>Value of the metric
</p>


<h3>Note</h3>

<p>Higher values of this metric indicate a better preservation or exposure, respectively, of the biological signal of interest. 
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Li, J., Bushel, P., Chu, T.-M., Wolfinger, R.D. (2009). Principal variance components analysis: Estimating batch effects in microarray gene expression data. In: Scherer, A. (ed) Batch Effects and Noise in Microarray Experiments: Sources and Solutions, John Wiley &amp; Sons, Chichester, UK, &lt;doi: <a href="https://doi.org/10.1002/9780470685983.ch12">10.1002/9780470685983.ch12</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

Xadj &lt;- ba(x=X, y=y, batch=batch, method = "combat")$xadj

pvcam(xba = X, batch = batch, y = y)
pvcam(xba = Xadj, batch = batch, y = y)
</code></pre>

<hr>
<h2 id='qunormaddon'>
Addon quantile normalization using &ldquo;documentation by value&rdquo; (Kostka &amp; Spang, 2008)
</h2><span id='topic+qunormaddon'></span>

<h3>Description</h3>

<p>Performs addon quantile normalization by using documentation by value (Kostka &amp; Spang, 2008).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qunormaddon(params, x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qunormaddon_+3A_params">params</code></td>
<td>

<p>object of class <code>qunormtrain</code>. Contains parameters necessary for addon quantile normalization.
</p>
</td></tr>
<tr><td><code id="qunormaddon_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix of the new data. Observations in rows, variables in columns.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses code from the off-CRAN package <code>docval</code>, version 1.0.
</p>


<h3>Value</h3>

<p>The covariate matrix of the test data after addon quantile normalization. Observations in rows, variables in columns.
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Kostka, D., Spang, R. (2008). Microarray based diagnosis profits from better documentation of gene expression signatures. PLoS Computational Biology 4(2):e22, &lt;doi: <a href="https://doi.org/10.1371/journal.pcbi.0040022">10.1371/journal.pcbi.0040022</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)
Xtrain &lt;- X[batch==1,]
Xtest &lt;- X[batch==2,]

params &lt;- qunormtrain(x=Xtrain)

Xtrainnorm &lt;- params$xnorm

Xtestaddonnorm &lt;- qunormaddon(params, x=Xtest)
</code></pre>

<hr>
<h2 id='qunormtrain'>
Quantile normalization with &ldquo;documentation by value&rdquo; (Kostka &amp; Spang, 2008)
</h2><span id='topic+qunormtrain'></span>

<h3>Description</h3>

<p>Performs quantile normalization for a covariate matrix and returns the normalized dataset together with information necessary for addon quantile normalization (Kostka &amp; Spang, 2008) using <code><a href="#topic+qunormaddon">qunormaddon</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qunormtrain(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qunormtrain_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix. observations in rows, variables in columns.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses code from the off-CRAN package <code>docval</code>, version 1.0.
</p>


<h3>Value</h3>

<p><code>qunormtrain</code> returns an object of class <code>qunormtrain</code>.
An object of class &quot;<code>qunormtrain</code>&quot; is a list containing the following components: 
</p>
<table role = "presentation">
<tr><td><code>xnorm</code></td>
<td>
<p> matrix of quantile normalized (training) data. Observations in rows, variables in columns. </p>
</td></tr>
<tr><td><code>mqnts</code></td>
<td>
<p> Vector of length ncol(xnorm). Averages of sorted values, with averages taken across observations. </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Kostka, D., Spang, R. (2008). Microarray based diagnosis profits from better documentation of gene expression signatures. PLoS Computational Biology 4(2):e22, &lt;doi: <a href="https://doi.org/10.1371/journal.pcbi.0040022">10.1371/journal.pcbi.0040022</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)
Xtrain &lt;- X[batch==1,]

params &lt;- qunormtrain(x=Xtrain)

Xtrainnorm &lt;- params$xnorm
</code></pre>

<hr>
<h2 id='ratioa'>
Batch effect adjustment using Ratio-A
</h2><span id='topic+ratioa'></span>

<h3>Description</h3>

<p>Performs batch effect adjustment using Ratio-A. Here, the variable values are divided by the batch-specific arithmetic mean of the corresponding variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ratioa(x, batch)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ratioa_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix. Observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="ratioa_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable. Each factor level (or 'category') corresponds to one of the batches. For example, if there are four batches, this variable would have four factor levels and observations with the same factor level would belong to the same batch.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>ratioa</code> returns an object of class <code>ratioa</code>.
An object of class &quot;<code>ratioa</code>&quot; is a list containing the following components: 
</p>
<table role = "presentation">
<tr><td><code>xadj</code></td>
<td>
<p> matrix of adjusted (training) data </p>
</td></tr>
<tr><td><code>nbatches</code></td>
<td>
<p> number of batches </p>
</td></tr>
<tr><td><code>batch</code></td>
<td>
<p> batch variable </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Luo, J., Schumacher, M., Scherer, A., Sanoudou, D., Megherbi, D., Davison, T., Shi, T., Tong, W., Shi, L., Hong, H., Zhao, C., Elloumi, F., Shi, W., Thomas, R., Lin, S., Tillinghast, G., Liu, G., Zhou, Y., Herman, D., Li, Y., Deng, Y., Fang, H., Bushel, P., Woods, M., Zhang, J. (2010). A comparison of batch effect removal methods for enhancement of prediction performance using maqc-ii microarray gene expression data. The Pharmacogenomics Journal 10:278-291, &lt;doi: <a href="https://doi.org/10.1038/tpj.2010.57">10.1038/tpj.2010.57</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

params &lt;- ratioa(x=X, batch=batch)
</code></pre>

<hr>
<h2 id='ratioaaddon'>
Addon batch effect adjustment for Ratio-A
</h2><span id='topic+ratioaaddon'></span>

<h3>Description</h3>

<p>Performs addon batch effect adjustment for Ratio-A: 
1) takes the output of <code><a href="#topic+ratioa">ratioa</a></code> applied to a training 
data set together with new batch data; 
2) checks whether the training data was also adjusted using Ratio-A and whether the same number of variables is present in training and new data;
3) performs Ratio-A on the new batch data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ratioaaddon(params, x, batch)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ratioaaddon_+3A_params">params</code></td>
<td>

<p>object of class <code>ratioa</code>.
</p>
</td></tr>
<tr><td><code id="ratioaaddon_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix of the new data. Observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="ratioaaddon_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable of the new data. Currently has to have levels: '1', '2', '3' and so on.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The adjusted covariate matrix of the test data.
</p>


<h3>Note</h3>

<p>Because Ratio-A is performed &quot;batch by batch&quot; the &quot;addon procedure&quot; for Ratio-A consists of plain Ratio-A on the new test batches.
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Luo, J., Schumacher, M., Scherer, A., Sanoudou, D., Megherbi, D., Davison, T., Shi, T., Tong, W., Shi, L., Hong, H., Zhao, C., Elloumi, F., Shi, W., Thomas, R., Lin, S., Tillinghast, G., Liu, G., Zhou, Y., Herman, D., Li, Y., Deng, Y., Fang, H., Bushel, P., Woods, M., Zhang, J. (2010). A comparison of batch effect removal methods for enhancement of prediction performance using maqc-ii microarray gene expression data. The Pharmacogenomics Journal 10:278-291, &lt;doi: <a href="https://doi.org/10.1038/tpj.2010.57">10.1038/tpj.2010.57</a>&gt;.
</p>
<p>Hornung, R., Boulesteix, A.-L., Causeur, D. (2016). Combining location-and-scale batch effect adjustment with data cleaning by latent factor adjustment. BMC Bioinformatics 17:27, &lt;doi: <a href="https://doi.org/10.1186/s12859-015-0870-z">10.1186/s12859-015-0870-z</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

trainind &lt;- which(batch %in% c(1,2))

Xtrain &lt;- X[trainind,]
ytrain &lt;- y[trainind]
batchtrain &lt;- factor(as.numeric(batch[trainind]), levels=c(1,2))


testind &lt;- which(batch %in% c(3,4))

Xtest &lt;- X[testind,]
ytest &lt;- y[testind]

batchtest &lt;- as.numeric(batch[testind])
batchtest[batchtest==3] &lt;- 1
batchtest[batchtest==4] &lt;- 2
batchtest &lt;- factor(batchtest, levels=c(1,2))


params &lt;- ratioa(x=Xtrain, batch=batchtrain)

Xtestaddon &lt;- ratioaaddon(params=params, x=Xtest, batch=batchtest)
</code></pre>

<hr>
<h2 id='ratiog'>
Batch effect adjustment using Ratio-G
</h2><span id='topic+ratiog'></span>

<h3>Description</h3>

<p>Performs batch effect adjustment using Ratio-G. Here, the variable values are divided by the batch-specific geometric mean of the corresponding variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ratiog(x, batch)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ratiog_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix. Observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="ratiog_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable. Each factor level (or 'category') corresponds to one of the batches. For example, if there are four batches, this variable would have four factor levels and observations with the same factor level would belong to the same batch.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>ratiog</code> returns an object of class <code>ratiog</code>.
An object of class &quot;<code>ratiog</code>&quot; is a list containing the following components: 
</p>
<table role = "presentation">
<tr><td><code>xadj</code></td>
<td>
<p> matrix of adjusted (training) data </p>
</td></tr>
<tr><td><code>nbatches</code></td>
<td>
<p> number of batches </p>
</td></tr>
<tr><td><code>batch</code></td>
<td>
<p> batch variable </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Luo, J., Schumacher, M., Scherer, A., Sanoudou, D., Megherbi, D., Davison, T., Shi, T., Tong, W., Shi, L., Hong, H., Zhao, C., Elloumi, F., Shi, W., Thomas, R., Lin, S., Tillinghast, G., Liu, G., Zhou, Y., Herman, D., Li, Y., Deng, Y., Fang, H., Bushel, P., Woods, M., Zhang, J. (2010). A comparison of batch effect removal methods for enhancement of prediction performance using maqc-ii microarray gene expression data. The Pharmacogenomics Journal 10:278-291, &lt;doi: <a href="https://doi.org/10.1038/tpj.2010.57">10.1038/tpj.2010.57</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

params &lt;- ratiog(x=X, batch=batch)
</code></pre>

<hr>
<h2 id='ratiogaddon'>
Addon batch effect adjustment for Ratio-G
</h2><span id='topic+ratiogaddon'></span>

<h3>Description</h3>

<p>Performs addon batch effect adjustment for Ratio-G: 
1) takes the output of <code><a href="#topic+ratiog">ratiog</a></code> applied to a training 
data set together with new batch data; 
2) checks whether the training data was also adjusted using Ratio-G and whether the same number of variables is present in training and new data;
3) performs Ratio-G on the new batch data.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ratiogaddon(params, x, batch)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ratiogaddon_+3A_params">params</code></td>
<td>

<p>object of class <code>ratiog</code>.
</p>
</td></tr>
<tr><td><code id="ratiogaddon_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix of the new data. Observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="ratiogaddon_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable of the new data. Currently has to have levels: '1', '2', '3' and so on.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The adjusted covariate matrix of the test data.
</p>


<h3>Note</h3>

<p>Because Ratio-G is performed &quot;batch by batch&quot; the &quot;addon procedure&quot; for Ratio-G consists of plain Ratio-G on the new test batches.
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Luo, J., Schumacher, M., Scherer, A., Sanoudou, D., Megherbi, D., Davison, T., Shi, T., Tong, W., Shi, L., Hong, H., Zhao, C., Elloumi, F., Shi, W., Thomas, R., Lin, S., Tillinghast, G., Liu, G., Zhou, Y., Herman, D., Li, Y., Deng, Y., Fang, H., Bushel, P., Woods, M., Zhang, J. (2010). A comparison of batch effect removal methods for enhancement of prediction performance using maqc-ii microarray gene expression data. The Pharmacogenomics Journal 10:278-291, &lt;doi: <a href="https://doi.org/10.1038/tpj.2010.57">10.1038/tpj.2010.57</a>&gt;.
</p>
<p>Hornung, R., Boulesteix, A.-L., Causeur, D. (2016). Combining location-and-scale batch effect adjustment with data cleaning by latent factor adjustment. BMC Bioinformatics 17:27, &lt;doi: <a href="https://doi.org/10.1186/s12859-015-0870-z">10.1186/s12859-015-0870-z</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

trainind &lt;- which(batch %in% c(1,2))

Xtrain &lt;- X[trainind,]
ytrain &lt;- y[trainind]
batchtrain &lt;- factor(as.numeric(batch[trainind]), levels=c(1,2))


testind &lt;- which(batch %in% c(3,4))

Xtest &lt;- X[testind,]
ytest &lt;- y[testind]

batchtest &lt;- as.numeric(batch[testind])
batchtest[batchtest==3] &lt;- 1
batchtest[batchtest==4] &lt;- 2
batchtest &lt;- factor(batchtest, levels=c(1,2))


params &lt;- ratiog(x=Xtrain, batch=batchtrain)

Xtestaddon &lt;- ratiogaddon(params=params, x=Xtest, batch=batchtest)
</code></pre>

<hr>
<h2 id='rmaaddon'>
Addon RMA normalization using &ldquo;documentation by value&rdquo; (Kostka &amp; Spang, 2008)
</h2><span id='topic+rmaaddon'></span>

<h3>Description</h3>

<p>Performs RMA with addon quantile normalization by using documentation by value (Kostka &amp; Spang, 2008).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rmaaddon(params, affybatchtest)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rmaaddon_+3A_params">params</code></td>
<td>

<p>object of class <code>rmatrain</code>. The normalized training data together with the information necesssary for addon normalization.
</p>
</td></tr>
<tr><td><code id="rmaaddon_+3A_affybatchtest">affybatchtest</code></td>
<td>

<p>object of class <code>AffyBatch</code>, that is Affymetrix GeneChip probe level data. Test data to be used for addon normalization.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses code from the off-CRAN package <code>docval</code>, version 1.0.
</p>


<h3>Value</h3>

<p>The covariate matrix of the test data after addon normalization. Observations in rows, variables in columns.
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Kostka, D., Spang, R. (2008). Microarray based diagnosis profits from better documentation of gene expression signatures. PLoS Computational Biology 4(2):e22, &lt;doi: <a href="https://doi.org/10.1371/journal.pcbi.0040022">10.1371/journal.pcbi.0040022</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Read in example data from ArrayExpress-webpage:

library("ArrayExpress")

expFiles &lt;- getAE("E-GEOD-62837", path = tempdir(), type = "raw")

rawfiles &lt;- file.path(tempdir(), expFiles$rawFiles)

library("affy")
# Training data:
affybatchtrain &lt;- ReadAffy(filenames=rawfiles[1:3])
# Test data:
affybatchtest &lt;- ReadAffy(filenames=rawfiles[4:5])

try(file.remove(file.path(tempdir(), expFiles$rawFiles)))
try(file.remove(file.path(tempdir(), expFiles$processedFiles)))
try(file.remove(file.path(tempdir(), expFiles$sdrf)))
try(file.remove(file.path(tempdir(), expFiles$idf)))
try(file.remove(file.path(tempdir(), expFiles$adf)))
try(file.remove(file.path(tempdir(), expFiles$rawArchive)))
try(file.remove(file.path(tempdir(), expFiles$processedArchive)))


# RMA normalization with documentation by value:
rmaparams &lt;- rmatrain(affybatchtrain)
Xtrainnorm &lt;- rmaparams$xnorm
dim(Xtrainnorm)

# RMA addon normalization:
Xtestaddonnorm &lt;- rmaaddon(rmaparams, affybatchtest)
dim(Xtestaddonnorm)

## End(Not run)
</code></pre>

<hr>
<h2 id='rmatrain'>
RMA normalization with &ldquo;documentation by value&rdquo; (Kostka &amp; Spang, 2008)
</h2><span id='topic+rmatrain'></span>

<h3>Description</h3>

<p>Performs RMA normalization and returns the normalized dataset together with information necessary for addon RMA normalization (Kostka &amp; Spang, 2008) using <code><a href="#topic+rmaaddon">rmaaddon</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rmatrain(affybatchtrain)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rmatrain_+3A_affybatchtrain">affybatchtrain</code></td>
<td>

<p>object of class <code>AffyBatch</code>. Affymetrix GeneChip probe level data.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses code from the off-CRAN package <code>docval</code>, version 1.0.
</p>


<h3>Value</h3>

<p><code>rmatrain</code> returns an object of class <code>rmatrain</code>.
An object of class &quot;<code>rmatrain</code>&quot; is a list containing the following components: 
</p>
<table role = "presentation">
<tr><td><code>xnorm</code></td>
<td>
<p> matrix of RMA normalized (training) data. Observations in rows, variables in columns. </p>
</td></tr>
<tr><td><code>rmadoc</code>, <code>sumdoc.rma</code>, <code>nfeature</code></td>
<td>
<p> information necessary for addon RMA normalization </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Kostka, D., Spang, R. (2008). Microarray based diagnosis profits from better documentation of gene expression signatures. PLoS Computational Biology 4(2):e22, &lt;doi: <a href="https://doi.org/10.1371/journal.pcbi.0040022">10.1371/journal.pcbi.0040022</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Read in example data from ArrayExpress-webpage:

library("ArrayExpress")

expFiles &lt;- getAE("E-GEOD-62837", path = tempdir(), type = "raw")

rawfiles &lt;- file.path(tempdir(), expFiles$rawFiles)

library("affy")
# Training data:
affybatchtrain &lt;- ReadAffy(filenames=rawfiles[1:3])

try(file.remove(file.path(tempdir(), expFiles$rawFiles)))
try(file.remove(file.path(tempdir(), expFiles$processedFiles)))
try(file.remove(file.path(tempdir(), expFiles$sdrf)))
try(file.remove(file.path(tempdir(), expFiles$idf)))
try(file.remove(file.path(tempdir(), expFiles$adf)))
try(file.remove(file.path(tempdir(), expFiles$rawArchive)))
try(file.remove(file.path(tempdir(), expFiles$processedArchive)))


# RMA normalization with documentation by value:
rmaparams &lt;- rmatrain(affybatchtrain)
Xtrainnorm &lt;- rmaparams$xnorm
dim(Xtrainnorm)

## End(Not run)
</code></pre>

<hr>
<h2 id='sepscore'>
Separation score as described in Hornung et al. (2016)
</h2><span id='topic+sepscore'></span>

<h3>Description</h3>

<p>This metric described in Hornung et al. (2016) was derived from the mixture score 
presented in Lazar et al. (2012). In contrast to the mixture score the 
separation score does not measure the degree of mixing but the degree of 
separation between the batches. Moreover it is less dependent on the 
relative sizes of the involved batches.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sepscore(xba, batch, k = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sepscore_+3A_xba">xba</code></td>
<td>

<p>matrix. The covariate matrix, raw or after batch effect adjustment. observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="sepscore_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable. Each factor level (or 'category') corresponds to one of the batches. For example, if there are four batches, this variable would have four factor levels and observations with the same factor level would belong to the same batch.
</p>
</td></tr>
<tr><td><code id="sepscore_+3A_k">k</code></td>
<td>

<p>integer. Number of nearest neighbors.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For two batches j and j* (see next paragraph for the case with more batches): 1) for each observation in batch j its k nearest neighbours in both batches j and j* 
simultaneously with respect to the euclidean distance are determined. Here, the proportion 
of those of these nearest neighbours, which belong to batch j* is calculated; 2) the 
average - denoted as MS_j - is taken over the thus obtained n_j proportions. This value 
is the mixture score as in Lazar et al. (2012); 3) to obtain a measure for the separation 
of the two batches the absolute difference between MS_j and its value expected in the 
absence of batch effects is taken: |MS_j - n_j* /(n_j + n_j* - 1)|; 4) the separation score 
is defined as the simple average of the latter quantity and the corresponding quantity when 
the roles of j and j* are switched. If the supplied number <code>k</code> of nearest neighbours 
is larger than n_j + n_j*, <code>k</code> is set to n_j + n_j* - 1 internally. 
</p>
<p>For more than two batches: 1) for all possible pairs of batches: calculate the metric as described above; 2) calculate
the weighted average of the values in 1) with weights proportional to the sum of the sample sizes in the two respective batches.
</p>


<h3>Value</h3>

<p>Value of the metric
</p>


<h3>Note</h3>

<p>The smaller the values of this metric, the better.
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Hornung, R., Boulesteix, A.-L., Causeur, D. (2016). Combining location-and-scale batch effect adjustment with data cleaning by latent factor adjustment. BMC Bioinformatics 17:27, &lt;doi: <a href="https://doi.org/10.1186/s12859-015-0870-z">10.1186/s12859-015-0870-z</a>&gt;.
</p>
<p>Lazar, C., Meganck, S., Taminau, J., Steenhoff, D., Coletta, A., Molter,C., Weiss-Solís, D. Y., Duque, R., Bersini, H., Nowé, A. (2012). Batch effect removal methods for microarray gene expression data integration: a survey. Briefings in Bioinformatics 14(4):469-490, &lt;doi: <a href="https://doi.org/10.1093/bib/bbs037">10.1093/bib/bbs037</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

# Random subset of 150 variables:
set.seed(1234)
Xsub &lt;- X[,sample(1:ncol(X), size=150)]

# In cases of batches with more than 20 observations
# select 20 observations at random:
subinds &lt;- unlist(sapply(1:length(levels(batch)), function(x) {
  indbatch &lt;- which(batch==x)
  if(length(indbatch) &gt; 20)
    indbatch &lt;- sort(sample(indbatch, size=20))
  indbatch
}))
Xsub &lt;- Xsub[subinds,]
batchsub &lt;- batch[subinds]
ysub &lt;- y[subinds]



sepscore(xba=Xsub, batch=batchsub, k=5)

params &lt;- ba(x=Xsub, y=ysub, batch=batchsub, method = "ratiog")
Xsubadj &lt;- params$xadj

sepscore(xba=Xsubadj, batch=batchsub, k=5)

params &lt;- ba(x=Xsub, y=ysub, batch=batchsub, method = "combat")
Xsubadj &lt;- params$xadj

sepscore(xba=Xsubadj, batch=batchsub, k=5)
</code></pre>

<hr>
<h2 id='skewdiv'>
Skewness divergence score
</h2><span id='topic+skewdiv'></span>

<h3>Description</h3>

<p>This metric presented in Shabalin et al. (2008) is concerned with the 
dissimilarity across batches of the skewnesses of the observation-wise 
empirical distributions of the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>skewdiv(xba, batch)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="skewdiv_+3A_xba">xba</code></td>
<td>

<p>matrix. The covariate matrix, raw or after batch effect adjustment. observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="skewdiv_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable. Each factor level (or 'category') corresponds to one of the batches. For example, if there are four batches, this variable would have four factor levels and observations with the same factor level would belong to the same batch.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For two batches j and j* (see next paragraph for the case with more batches): 1) for each observation 
calculate the difference between the mean and the median of the data as a measure for 
the skewness of the distribution of the variable values; 2) determine the area between 
the two batch-wise empirical cumulative density functions of the values out of 1). The value 
obtained in 2) can be regarded as a measure for the disparity of the batches with respect 
to the skewness of the observation-wise empirical distributions. 
</p>
<p>For more than two batches: 1) for all possible pairs of batches: calculate the metric as described above; 2) calculate
the weighted average of the values in 1) with weights proportional to the sum of the sample sizes in the two respective batches. 
</p>
<p>The variables are standardized before the calculation to make the metric independent of scale. 
</p>


<h3>Value</h3>

<p>Value of the metric
</p>


<h3>Note</h3>

<p>The smaller the values of this metric, the better.
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Shabalin, A. A., Tjelmeland, H., Fan, C., Perou, C. M., Nobel, A. B. (2008). Merging two gene-expression studies via cross-platform normalization. Bioinformatics 24(9):1154-1160, &lt;doi: <a href="https://doi.org/10.1093/bioinformatics/btn083">10.1093/bioinformatics/btn083</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

skewdiv(xba=X, batch=batch)

params &lt;- ba(x=X, y=y, batch=batch, method = "ratiog")
Xadj &lt;- params$xadj

skewdiv(xba=Xadj, batch=batch)

params &lt;- ba(x=X, y=y, batch=batch, method = "combat")
Xadj &lt;- params$xadj

skewdiv(xba=Xadj, batch=batch)
</code></pre>

<hr>
<h2 id='standardize'>
Batch effect adjustment by standardization
</h2><span id='topic+standardize'></span>

<h3>Description</h3>

<p>Performs batch effect adjustment by standardizing the variables within batches to have zero mean and variance one.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>standardize(x, batch)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="standardize_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix. Observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="standardize_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable. Each factor level (or 'category') corresponds to one of the batches. For example, if there are four batches, this variable would have four factor levels and observations with the same factor level would belong to the same batch.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>standardize</code> returns an object of class <code>standardize</code>.
An object of class &quot;<code>standardize</code>&quot; is a list containing the following components: 
</p>
<table role = "presentation">
<tr><td><code>xadj</code></td>
<td>
<p> matrix of adjusted (training) data </p>
</td></tr>
<tr><td><code>nbatches</code></td>
<td>
<p> number of batches </p>
</td></tr>
<tr><td><code>batch</code></td>
<td>
<p> batch variable </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

params &lt;- standardize(x=X, batch=batch)
</code></pre>

<hr>
<h2 id='standardizeaddon'>
Addon batch effect adjustment for standardization
</h2><span id='topic+standardizeaddon'></span>

<h3>Description</h3>

<p>Performs addon batch effect adjustment for standardization: 
1) takes the output of <code><a href="#topic+standardize">standardize</a></code> applied to a training 
data set together with new batch data; 
2) checks whether the training data was also adjusted using standardization and whether the same number of variables is present in training and new data;
3) performs standardization on the new batch data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>standardizeaddon(params, x, batch)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="standardizeaddon_+3A_params">params</code></td>
<td>

<p>object of class <code>standardize</code>.
</p>
</td></tr>
<tr><td><code id="standardizeaddon_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix of the new data. Observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="standardizeaddon_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable of the new data. Currently has to have levels: '1', '2', '3' and so on.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The adjusted covariate matrix of the test data.
</p>


<h3>Note</h3>

<p>It is <b>not recommended</b> to perform standardization in cross-study prediction settings, because for some classifiers the prediction performance can be strongly impaired by this method. Given a not too small test set, the following methods are recommended (Hornung et al., 2016a): <code><a href="#topic+combatba">combatba</a></code>, <code><a href="#topic+meancenter">meancenter</a></code>, <code><a href="#topic+ratioa">ratioa</a></code>, <code><a href="#topic+ratiog">ratiog</a></code>.
</p>
<p>Because standardization is performed &quot;batch by batch&quot; the &quot;addon procedure&quot; for standardization consists of plain standardization on the new test batches.
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Hornung, R., Causeur, D., Bernau, C., Boulesteix, A.-L. (2017). Improving cross-study prediction through addon batch effect adjustment and addon normalization. Bioinformatics 33(3):397–404, &lt;doi: <a href="https://doi.org/10.1093/bioinformatics/btw650">10.1093/bioinformatics/btw650</a>&gt;.
</p>
<p>Hornung, R., Boulesteix, A.-L., Causeur, D. (2016). Combining location-and-scale batch effect adjustment with data cleaning by latent factor adjustment. BMC Bioinformatics 17:27, &lt;doi: <a href="https://doi.org/10.1186/s12859-015-0870-z">10.1186/s12859-015-0870-z</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

trainind &lt;- which(batch %in% c(1,2))

Xtrain &lt;- X[trainind,]
ytrain &lt;- y[trainind]
batchtrain &lt;- factor(as.numeric(batch[trainind]), levels=c(1,2))


testind &lt;- which(batch %in% c(3,4))

Xtest &lt;- X[testind,]
ytest &lt;- y[testind]

batchtest &lt;- as.numeric(batch[testind])
batchtest[batchtest==3] &lt;- 1
batchtest[batchtest==4] &lt;- 2
batchtest &lt;- factor(batchtest, levels=c(1,2))


params &lt;- standardize(x=Xtrain, batch=batchtrain)

Xtestaddon &lt;- standardizeaddon(params=params, x=Xtest, batch=batchtest)
</code></pre>

<hr>
<h2 id='svaba'>
Batch effect adjustment using SVA
</h2><span id='topic+svaba'></span>

<h3>Description</h3>

<p>Performs batch effect adjustment using Surrogate Variable Analysis (SVA) and additionally returns information necessary for addon batch effect adjustment with frozen SVA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>svaba(x, y, batch, nbf = NULL, algorithm = "fast")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="svaba_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix. Observations in rows, variables in columns.
</p>
</td></tr>
<tr><td><code id="svaba_+3A_y">y</code></td>
<td>

<p>factor. Binary target variable. Has to have two factor levels, where each of them correponds to one of the two classes of the target variable.
</p>
</td></tr>
<tr><td><code id="svaba_+3A_batch">batch</code></td>
<td>

<p>factor. Batch variable. Each factor level (or 'category') corresponds to one of the batches. For example, if there are four batches, this variable would have four factor levels and observations with the same factor level would belong to the same batch.
</p>
</td></tr>
<tr><td><code id="svaba_+3A_nbf">nbf</code></td>
<td>

<p>integer. Number of latent factors to estimate.
</p>
</td></tr>
<tr><td><code id="svaba_+3A_algorithm">algorithm</code></td>
<td>

<p>character. If <code>method = "fast"</code> the &quot;approximate fSVA algorithm&quot; will be used in frozen SVA. If <code>method = "exact"</code> the &quot;exact fSVA algorithm&quot; will be used. See Parker et al. (2014).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is essentially a wrapper function of the function <code>sva()</code> from the Bioconductor package of the same name.
</p>


<h3>Value</h3>

<p><code>svaba</code> returns an object of class <code>svatrain</code>.
An object of class &quot;<code>svatrain</code>&quot; is a list containing the following components: 
</p>
<table role = "presentation">
<tr><td><code>xadj</code></td>
<td>
<p> matrix of adjusted (training) data </p>
</td></tr>
<tr><td><code>xtrain</code></td>
<td>
<p> the unadjusted covariate matrix. Used in frozen SVA. </p>
</td></tr>
<tr><td><code>ytrain</code></td>
<td>
<p> binary target variable. Used in frozen SVA. </p>
</td></tr>
<tr><td><code>svobj</code></td>
<td>
<p> output of the function <code>sva()</code>. Used in frozen SVA. </p>
</td></tr>
<tr><td><code>algorithm</code></td>
<td>
<p> algorithm to use in frozen SVA </p>
</td></tr>
<tr><td><code>nbatches</code></td>
<td>
<p> number of batches </p>
</td></tr>
<tr><td><code>batch</code></td>
<td>
<p> batch variable </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Leek, J. T., Storey, J. D. (2007). Capturing Heterogeneity in Gene Expression Studies by Surrogate Variable Analysis. PLoS Genetics 3:1724-1735, &lt;doi: <a href="https://doi.org/10.1371/journal.pgen.0030161">10.1371/journal.pgen.0030161</a>&gt;.
</p>
<p>Parker, H. S., Bravo, H. C., Leek, J. T. (2014). Removing batch effects for prediction problems with frozen surrogate variable analysis. PeerJ 2:e561, &lt;doi: <a href="https://doi.org/10.7717/peerj.561">10.7717/peerj.561</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

# Random subset of 150 variables:
set.seed(1234)
Xsub &lt;- X[,sample(1:ncol(X), size=150)]

# In cases of batches with more than 20 observations
# select 20 observations at random:
subinds &lt;- unlist(sapply(1:length(levels(batch)), function(x) {
  indbatch &lt;- which(batch==x)
  if(length(indbatch) &gt; 20)
    indbatch &lt;- sort(sample(indbatch, size=20))
  indbatch
}))
Xsub &lt;- Xsub[subinds,]
batchsub &lt;- batch[subinds]
ysub &lt;- y[subinds]



params &lt;- svaba(x=Xsub, y=ysub, batch=batchsub)
</code></pre>

<hr>
<h2 id='svabaaddon'>
Addon batch effect adjustment using frozen SVA
</h2><span id='topic+svabaaddon'></span>

<h3>Description</h3>

<p>Performs addon batch effect adjustment using frozen SVA. Takes the output of performing <code>svaba</code> on a training data set and new batch data and correspondingly adjusts the test data to better match the adjusted training data according to the SVA model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>svabaaddon(params, x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="svabaaddon_+3A_params">params</code></td>
<td>

<p>object of class <code>svatrain</code>. Contains parameters necessary for addon batch effect adjustment with frozen SVA.
</p>
</td></tr>
<tr><td><code id="svabaaddon_+3A_x">x</code></td>
<td>

<p>matrix. The covariate matrix of the new data. Observations in rows, variables in columns.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The adjusted covariate matrix of the test data.
</p>


<h3>Note</h3>

<p>It is <b>not recommended</b> to perform frozen SVA in cross-study prediction settings, because it assumes similarity between training and test set and has been observed to (strongly) impair prediction performance in cases where this assumption is not given. Given a not too small test set, the following methods are recommended (Hornung et al., 2016): <code><a href="#topic+combatba">combatba</a></code>, <code><a href="#topic+meancenter">meancenter</a></code>, <code><a href="#topic+ratioa">ratioa</a></code>, <code><a href="#topic+ratiog">ratiog</a></code>.
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Leek, J. T., Storey, J. D. (2007). Capturing Heterogeneity in Gene Expression Studies by Surrogate Variable Analysis. PLoS Genetics 3:1724-1735, &lt;doi: <a href="https://doi.org/10.1371/journal.pgen.0030161">10.1371/journal.pgen.0030161</a>&gt;.
</p>
<p>Parker, H. S., Bravo, H. C., Leek, J. T. (2014). Removing batch effects for prediction problems with frozen surrogate variable analysis. PeerJ 2:e561, &lt;doi: <a href="https://doi.org/10.7717/peerj.561">10.7717/peerj.561</a>&gt;.
</p>
<p>Hornung, R., Causeur, D., Bernau, C., Boulesteix, A.-L. (2017). Improving cross-study prediction through addon batch effect adjustment and addon normalization. Bioinformatics 33(3):397–404, &lt;doi: <a href="https://doi.org/10.1093/bioinformatics/btw650">10.1093/bioinformatics/btw650</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(autism)

# Random subset of 150 variables:
set.seed(1234)
Xsub &lt;- X[,sample(1:ncol(X), size=150)]

# In cases of batches with more than 20 observations
# select 20 observations at random:
subinds &lt;- unlist(sapply(1:length(levels(batch)), function(x) {
  indbatch &lt;- which(batch==x)
  if(length(indbatch) &gt; 20)
    indbatch &lt;- sort(sample(indbatch, size=20))
  indbatch
}))
Xsub &lt;- Xsub[subinds,]
batchsub &lt;- batch[subinds]
ysub &lt;- y[subinds]



trainind &lt;- which(batchsub %in% c(1,2))

Xsubtrain &lt;- Xsub[trainind,]
ysubtrain &lt;- ysub[trainind]
batchsubtrain &lt;- factor(as.numeric(batchsub[trainind]), levels=c(1,2))


testind &lt;- which(batchsub %in% c(3,4))

Xsubtest &lt;- Xsub[testind,]
ysubtest &lt;- ysub[testind]

batchsubtest &lt;- as.numeric(batchsub[testind])
batchsubtest[batchsubtest==3] &lt;- 1
batchsubtest[batchsubtest==4] &lt;- 2
batchsubtest &lt;- factor(batchsubtest, levels=c(1,2))



params &lt;- svaba(x=Xsubtrain, y=ysubtrain, batch=batchsubtrain)

Xsubtestaddon &lt;- svabaaddon(params, x=Xsubtest)
</code></pre>

<hr>
<h2 id='X'>
Covariate matrix of dataset <code>autism</code>
</h2><span id='topic+X'></span>

<h3>Description</h3>

<p>See dataset <code><a href="#topic+autism">autism</a></code>
</p>

<hr>
<h2 id='y'>
Target variable of dataset <code>autism</code>
</h2><span id='topic+y'></span>

<h3>Description</h3>

<p>See dataset <code><a href="#topic+autism">autism</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
