<!DOCTYPE html><html lang="en"><head><title>Help for package BarcodingR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BarcodingR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#barcodes.eval'><p>Barcodes Evaluation</p></a></li>
<li><a href='#barcoding.gap'><p>Barcoding Gap Calculation</p></a></li>
<li><a href='#barcoding.spe.identify'><p>Species Identification using Protein-coding Barcodes</p></a></li>
<li><a href='#barcoding.spe.identify2'><p>Species Identification Based on Fuzzy-set Method and kmer</p></a></li>
<li><a href='#bbsik'><p>Bp Barcoding Species Identify using Kmer</p></a></li>
<li><a href='#char2NumVector'><p>Character to Integer Vector</p></a></li>
<li><a href='#compare2delimitations'><p>Comparision between two Delimitations</p></a></li>
<li><a href='#consensus.identify'><p>Consensus Identification</p></a></li>
<li><a href='#digitize.DNA'><p>Digitize DNAbin</p></a></li>
<li><a href='#DNAbin2kmerFreqMatrix'><p>Calculation of Kmer Frequency Matrix from DNAbin for Both Reference and Query Sequences</p></a></li>
<li><a href='#FMF'><p>Fuzzy Membership Function Value</p></a></li>
<li><a href='#FMFtheta12'><p>Calculate Intraspecific and Interspecific Variation</p></a></li>
<li><a href='#NAMES'><p>Extracts Lables of Samples</p></a></li>
<li><a href='#optimize.kmer'><p>Optimize kmer Length</p></a></li>
<li><a href='#pineMothCOI'>
<p>pine Moth COI</p></a></li>
<li><a href='#pineMothITS1'>
<p>pine Moth ITS1</p></a></li>
<li><a href='#pineMothITS2'>
<p>pine Moth ITS2</p></a></li>
<li><a href='#sample.ref'><p>Sample Random Datasets from References (DNAbin)</p></a></li>
<li><a href='#save.ids'><p>Save Identifications</p></a></li>
<li><a href='#summarize.ref'><p>Summarize Reference Data</p></a></li>
<li><a href='#TDR2'><p>TDR2 Species Membership Value</p></a></li>
<li><a href='#TibetanMoth'>
<p>Tibetan Moth</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Species Identification using DNA Barcodes</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0-3</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-04-10</td>
</tr>
<tr>
<td>Description:</td>
<td>To perform species identification using DNA barcodes.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.1),ape,nnet,class,stats,sp</td>
</tr>
<tr>
<td>Suggests:</td>
<td>cluster,compiler,utils</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>true</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-04-14 13:47:36 UTC; Administrator</td>
</tr>
<tr>
<td>Author:</td>
<td>Ai-bing ZHANG [aut, cre],
  Meng-di HAO [aut],
  Cai-qing YANG [aut],
  Zhi-yong SHI [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ai-bing ZHANG &lt;Zhangab2008@mail.cnu.edu.cn&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-04-14 15:00:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='barcodes.eval'>Barcodes Evaluation</h2><span id='topic+barcodes.eval'></span>

<h3>Description</h3>

<p>Evaluate two barcodes using species identification success rate criteria.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>barcodes.eval(barcode1, barcode2, kmer1 = kmer1, kmer2 = kmer2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="barcodes.eval_+3A_barcode1">barcode1</code></td>
<td>
<p>object of class &quot;DNAbin&quot; based on barcode1, which contains taxon information.</p>
</td></tr>
<tr><td><code id="barcodes.eval_+3A_barcode2">barcode2</code></td>
<td>
<p>object of class &quot;DNAbin&quot; based on barcode2, which contains taxon information.</p>
</td></tr>
<tr><td><code id="barcodes.eval_+3A_kmer1">kmer1</code></td>
<td>
<p>a numeric to indicate the length of kmer1 for barcode1, the optimal kmer could be found by the function
optimize.kmer() before running this function.</p>
</td></tr>
<tr><td><code id="barcodes.eval_+3A_kmer2">kmer2</code></td>
<td>
<p>a numeric to indicate the length of kmer2 for barcode2, see above.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing p_value of prop.test(), and so on.
</p>


<h3>Author(s)</h3>

<p>Ai-bing ZHANG, PhD. CNU, Beijing, CHINA.
</p>


<h3>References</h3>

<p>zhangab2008 (at) mail. cnu. edu. cn.
</p>


<h3>See Also</h3>

<p>prop.test()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(TibetanMoth) 
barcode1&lt;-as.DNAbin(as.character(TibetanMoth[1:30,]))
barcode2&lt;-barcode1
b.eval&lt;-barcodes.eval(barcode1,barcode2,kmer1=1,kmer2=3)
b.eval
</code></pre>

<hr>
<h2 id='barcoding.gap'>Barcoding Gap Calculation</h2><span id='topic+barcoding.gap'></span>

<h3>Description</h3>

<p>Calculation of DNA barcoding gap. Besides K2P distance, raw distance and euclidean could
also be used for calculation DNA barcoding gap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>barcoding.gap(ref, dist = dist)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="barcoding.gap_+3A_ref">ref</code></td>
<td>
<p>object of class &quot;DNAbin&quot; used as a reference dataset, which contains taxon information.</p>
</td></tr>
<tr><td><code id="barcoding.gap_+3A_dist">dist</code></td>
<td>
<p>a character string which takes one of (&quot;raw&quot;,&quot;K80&quot;,&quot;euclidean&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list indicates the summary statistics of interspecific and intraspecific genetic distance,
such as k2P distance.
</p>


<h3>Note</h3>

<p>the current version of the function can only be used for protein-coding barcodes,
such as, COI. The futuren version may incorporate calculation for non-coding barcodes,for
instance, ITS1, ITS2.
</p>


<h3>Author(s)</h3>

<p>Ai-bing ZHANG, PhD. CNU, Beijing, CHINA, contact at zhangab2008(at)mail.cnu.edu.cn
</p>


<h3>References</h3>

<p>Meyer, Christopher P., and Gustav Paulay. (2005). &rdquo;DNA barcoding: error rates based on
comprehensive sampling.&rdquo;.PLoS biology 3.12: e422.
</p>
<p>F.Jiang, Q. Jin, L. Liang, A.B. Zhang,and Z.H. Li.(2014). Existence of Species
Complex Largely Reduced Barcoding Success for Invasive Species of Tephritidae: A Case Study in
Bactrocera spp. Mol Ecol Resour. 14(6):1114-1128 DOI: 10.1111/1755-0998.12259.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(TibetanMoth)
TibetanMoth&lt;-as.DNAbin(as.character(TibetanMoth[1:20,]))
b.gap&lt;-barcoding.gap(ref=TibetanMoth,dist="K80")
b.gap
</code></pre>

<hr>
<h2 id='barcoding.spe.identify'>Species Identification using Protein-coding Barcodes</h2><span id='topic+barcoding.spe.identify'></span>

<h3>Description</h3>

<p>Species identification using protein-coding barcodes with different methods,including BP-based method
(Zhang et al. 2008), fuzzy-set based method (Zhang et al. 2012), Bayesian-based method (Jin et al. 2013).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>barcoding.spe.identify(ref, que, method = "bpNewTraining")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="barcoding.spe.identify_+3A_ref">ref</code></td>
<td>
<p>object of class &quot;DNAbin&quot; used as a reference dataset, which contains taxon information.</p>
</td></tr>
<tr><td><code id="barcoding.spe.identify_+3A_que">que</code></td>
<td>
<p>object of class &quot;DNAbin&quot;, whose identities (species names) need to be inferred.</p>
</td></tr>
<tr><td><code id="barcoding.spe.identify_+3A_method">method</code></td>
<td>
<p>a character string indicating which method will be used to train model and/or infer species membership. 
One of these methods (&quot;fuzzyId&quot;, &quot;bpNewTraining&quot;, &quot;bpNewTrainingOnly&quot;, &quot;bpUseTrained&quot;,&quot;Bayesian&quot;) should be specified.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing model parameters used, species identification success rates using references,
query sequences, species inferred, and corresponding confidence levels 
(bp probability for BP-based method / FMF values for fuzzy set theory based method / posterior probability for Bayesian method) when available.
</p>


<h3>Note</h3>

<p>functions fasta2DNAbin() from package:adegenet and read.dna() from package:ape were used to obtain DNAbin object in our package.
The former is used to read large aligned coding DNA barcodes, the latter unaligned ones. ref and que
should be aligned with identical sequence length. We provided a pipeline to perform fast
sequences alignment for reference and query sequences. Windows users could contact zhangab2008(at)mail.cnu.edu.cn
for an exec version of the package. For very large DNA dataset, read.fas() package:phyloch is strongly suggested instead of
fasta2DNAbin() since the latter is very slow.
</p>


<h3>Author(s)</h3>

<p>Ai-bing ZHANG, PhD. CNU, Beijing, CHINA.
zhangab2008(at)mail.cnu.edu.cn
</p>


<h3>References</h3>

<p>Zhang, A. B., M. D. Hao, C. Q. Yang, and Z. Y. Shi. (2017). BarcodingR: an integrated R package for species identification using DNA barcodes. Methods Ecol Evol. 8(5):627-634.
https://besjournals.onlinelibrary.wiley.com/doi/10.1111/2041-210X.12682.
</p>
<p>Jin,Q., H.L. Han, X.M. Hu, X.H. Li,C.D. Zhu,S. Y. W. Ho, R. D. Ward, A.B. Zhang . (2013). Quantifying Species Diversity with a DNA Barcoding-Based Method: Tibetan Moth Species (Noctuidae) on the Qinghai-Tibetan Plateau. PloS One 8: e644.
https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0064428.
</p>
<p>Zhang, A. B., C. Muster, H.B. Liang, C.D. Zhu, R. Crozier, P. Wan, J. Feng, R. D. Ward.(2012). A fuzzy-set-theory-based approach to analyse species membership in DNA barcoding. Molecular Ecology, 21(8):1848-63.
https://onlinelibrary.wiley.com/doi/10.1111/j.1365-294X.2011.05235.x
</p>
<p>Zhang, A. B., D. S. Sikes, C. Muster, S. Q. Li. (2008). Inferring Species Membership using DNA sequences with Back-propagation Neural Networks. Systematic Biology, 57(2):202-215.
https://besjournals.onlinelibrary.wiley.com/doi/10.1111/2041-210X.12682
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(TibetanMoth)
ref&lt;-as.DNAbin(as.character(TibetanMoth[1:5,]))
que&lt;-as.DNAbin(as.character(TibetanMoth[50:55,]))
bsi&lt;-barcoding.spe.identify(ref, que, method = "fuzzyId")
bsi
bsi&lt;-barcoding.spe.identify(ref, que, method = "bpNewTraining")
bsi
bsi&lt;-barcoding.spe.identify(ref, que, method = "Bayesian")
bsi
</code></pre>

<hr>
<h2 id='barcoding.spe.identify2'>Species Identification Based on Fuzzy-set Method and kmer</h2><span id='topic+barcoding.spe.identify2'></span>

<h3>Description</h3>

<p>Species identification based on fuzzy-set method (Zhang et al. 2012)and kmer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>barcoding.spe.identify2(ref, que, kmer = kmer, optimization = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="barcoding.spe.identify2_+3A_ref">ref</code></td>
<td>
<p>object of class &quot;DNAbin&quot; used as a reference dataset, which contains taxon information.</p>
</td></tr>
<tr><td><code id="barcoding.spe.identify2_+3A_que">que</code></td>
<td>
<p>object of class &quot;DNAbin&quot;, whose identities (species names) need to be inferred.</p>
</td></tr>
<tr><td><code id="barcoding.spe.identify2_+3A_kmer">kmer</code></td>
<td>
<p>a numeric to indicate the length of maximum kmer to try in the range of 1 to kmer in case of
optimization = TRUE, otherwise, only a certain length of kmer is used.</p>
</td></tr>
<tr><td><code id="barcoding.spe.identify2_+3A_optimization">optimization</code></td>
<td>
<p>a character string, indicating whether different length of kmer (up to kmer) will be used or
just a specified length of kmer will be used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list indicating the identified species.
</p>


<h3>Note</h3>

<p>read.dna() from package ape was used to obtain DNAbin object for unaligned non-coding barcodes.
</p>


<h3>Author(s)</h3>

<p>Ai-bing ZHANG, Cai-qing YANG, Meng-di HAO, CNU, Beijing, CHINA, contact at zhangab2008 (at) mail. cnu. edu. cn.
</p>


<h3>References</h3>

<p>Zhang, A. B., M. D. Hao, C. Q. Yang, and Z. Y. Shi. (2017). BarcodingR: an integrated R package for species identification using DNA barcodes. Methods Ecol Evol. 8(5):627-634.
https://besjournals.onlinelibrary.wiley.com/doi/10.1111/2041-210X.12682.
</p>
<p>Jin,Q., H.L. Han, X.M. Hu, X.H. Li,C.D. Zhu,S. Y. W. Ho, R. D. Ward, A.B. Zhang . (2013). Quantifying Species Diversity with a DNA Barcoding-Based Method: Tibetan Moth Species (Noctuidae) on the Qinghai-Tibetan Plateau. PloS One 8: e644.
https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0064428.
</p>
<p>Zhang, A. B., C. Muster, H.B. Liang, C.D. Zhu, R. Crozier, P. Wan, J. Feng, R. D. Ward.(2012). A fuzzy-set-theory-based approach to analyse species membership in DNA barcoding. Molecular Ecology, 21(8):1848-63.
https://onlinelibrary.wiley.com/doi/10.1111/j.1365-294X.2011.05235.x
</p>
<p>Zhang, A. B., D. S. Sikes, C. Muster, S. Q. Li. (2008). Inferring Species Membership using DNA sequences with Back-propagation Neural Networks. Systematic Biology, 57(2):202-215.
https://besjournals.onlinelibrary.wiley.com/doi/10.1111/2041-210X.12682
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(pineMothITS2) 
ref&lt;-pineMothITS2
que&lt;-ref
spe.id&lt;-barcoding.spe.identify2(ref,que, kmer = 1, optimization = FALSE)
spe.id
 
</code></pre>

<hr>
<h2 id='bbsik'>Bp Barcoding Species Identify using Kmer</h2><span id='topic+bbsik'></span>

<h3>Description</h3>

<p>Species identification using BP-based method for both protein-coding
barcodes, for instance, COI, and non-coding barcodes, such as, ITS, using kmer statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bbsik(ref, que, kmer = kmer, UseBuiltModel = FALSE, lr = 5e-05, maxit = 1e+06)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bbsik_+3A_ref">ref</code></td>
<td>
<p>object of class &quot;DNAbin&quot; used as a reference dataset, which contains taxon information.</p>
</td></tr>
<tr><td><code id="bbsik_+3A_que">que</code></td>
<td>
<p>object of class &quot;DNAbin&quot;, which needs to be inferred.</p>
</td></tr>
<tr><td><code id="bbsik_+3A_kmer">kmer</code></td>
<td>
<p>a numeric indicating the length of kmer used.</p>
</td></tr>
<tr><td><code id="bbsik_+3A_usebuiltmodel">UseBuiltModel</code></td>
<td>
<p>logic value to indicate whether a built model is used 
or not.</p>
</td></tr>
<tr><td><code id="bbsik_+3A_lr">lr</code></td>
<td>
<p>parameter for weight decay. Default 5e-5.</p>
</td></tr>
<tr><td><code id="bbsik_+3A_maxit">maxit</code></td>
<td>
<p>maximum number of iterations. Default 1e+6.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing model parameters used, species identification success rates using references,
query sequences, species inferred, and corresponding confidence levels (bp probability for BP-based method).
</p>


<h3>Author(s)</h3>

<p>Ai-bing ZHANG, Meng-di HAO, Cai-qing YANG, CNU, Beijing, CHINA. zhangab2008 (at) mail. cnu. edu.cn
</p>


<h3>References</h3>

<p>Zhang, A. B., D. S. Sikes, C. Muster, S. Q. Li. (2008). Inferring Species Membership 
using DNA sequences with Back-propagation Neural Networks. Systematic Biology, 57(2):202-215. 
https://academic.oup.com/sysbio/article/57/2/202/1622290
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(TibetanMoth) 
ref&lt;-as.DNAbin(as.character(TibetanMoth[1:50,]))
que&lt;-as.DNAbin(as.character(TibetanMoth[51:60,]))
out&lt;-bbsik(ref, que, kmer = 1, UseBuiltModel = FALSE)
out
out$convergence
out$success.rates.ref

data(pineMothITS2) 
ref&lt;-pineMothITS2
que&lt;-pineMothITS2
out&lt;-bbsik(ref, que, kmer = 1, UseBuiltModel = FALSE)
out
out$convergence
out$success.rates.ref



</code></pre>

<hr>
<h2 id='char2NumVector'>Character to Integer Vector</h2><span id='topic+char2NumVector'></span>

<h3>Description</h3>

<p>Conversion from a character vector to an integer vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>char2NumVector(c)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="char2NumVector_+3A_c">c</code></td>
<td>
<p>character vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an integer vector.
</p>


<h3>Author(s)</h3>

<p>Ai-bing ZHANG, PhD. CNU, Beijing, CHINA. zhangab2008 (at) mail. cnu. edu.cn.
</p>


<h3>References</h3>

<p>zhangab2008 (at) mail. cnu. edu. cn.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>c&lt;-c("a","a","b")
num&lt;-char2NumVector(c)
num

</code></pre>

<hr>
<h2 id='compare2delimitations'>Comparision between two Delimitations</h2><span id='topic+compare2delimitations'></span>

<h3>Description</h3>

<p>Comparision between two delimitations of a group of samples, for instance,
traditionally morphological delimitation and molecular delimitation (MOTU).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compare2delimitations(deli1, deli2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compare2delimitations_+3A_deli1">deli1</code></td>
<td>
<p>a character array (vector),containing a set of, for example, morphological identification (species names), to compare with</p>
</td></tr>
<tr><td><code id="compare2delimitations_+3A_deli2">deli2</code></td>
<td>
<p>a character array (vector),containing a set of, molecular delimitation (MOTU).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing the adjusted Rand index comparing the two partitions (a scalar). This index has zero expected value in the case
of random partition, and it is bounded above by 1 in the case of perfect agreement between two partitions; the numbers of matches,
splits,merges, and corresponding percentage.
</p>


<h3>Note</h3>

<p>This is for the same set of samples with two partitions/delimitations.
</p>


<h3>Author(s)</h3>

<p>Ai-bing ZHANG, PhD. CNU, Beijing, CHINA.
</p>


<h3>References</h3>

<p>L. Hubert and P. Arabie (1985) Comparing Partitions, Journal of the Classification 2:193-218.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>




deli1&lt;-c(1,1,1,1,1,1)
deli2&lt;-c(1,1,2,1,1,3)
out&lt;-compare2delimitations(deli1,deli2)
out
</code></pre>

<hr>
<h2 id='consensus.identify'>Consensus Identification</h2><span id='topic+consensus.identify'></span>

<h3>Description</h3>

<p>Make consensus for identifications from two or more methods, usually for 
a set of query sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>consensus.identify(identifiedBy2orMore)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="consensus.identify_+3A_identifiedby2ormore">identifiedBy2orMore</code></td>
<td>
<p>an object of class &quot;data.frame&quot;, containing (queIDs, as rownames), identifiedByMethod1,identifiedByMethod2,and so on.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame with consensus.identification, and corresponding votes.
</p>


<h3>Note</h3>

<p>Suitable for case where a set of queries were identified by more than two methods.
</p>


<h3>Author(s)</h3>

<p>Ai-bing ZHANG, PhD. CNU, Beijing, CHINA, contact at zhangab2008(at)mail.cnu.edu.cn
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
queIDs&lt;-c("q1","q2","q3")

bp&lt;-c("sp1","sp1","sp1")
bpk&lt;-c("sp1","sp1","sp2")
bayes&lt;-c("sp2","sp1","sp3")
fuzzyID&lt;-c("sp1","sp1","sp2")
identifiedBy2orMore&lt;-data.frame(bp=bp,bpk=bpk,bayes=bayes,fuzzyID=fuzzyID)
rownames(identifiedBy2orMore)&lt;-queIDs&lt;-c("q1","q2","q3")
ccs&lt;-consensus.identify(identifiedBy2orMore)
</code></pre>

<hr>
<h2 id='digitize.DNA'>Digitize DNAbin</h2><span id='topic+digitize.DNA'></span>

<h3>Description</h3>

<p>Digitize an object of DNAbin.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>digitize.DNA(seqs)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="digitize.DNA_+3A_seqs">seqs</code></td>
<td>
<p>an object of DNAbin.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a numeric matrix of DNA sequences digitized.
</p>


<h3>Author(s)</h3>

<p>Ai-bing ZHANG, PhD. CNU, Beijing, CHINA.
</p>


<h3>References</h3>

<p>zhangab2008(at)mail.cnu.edu.cn
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(TibetanMoth)
digitized.DNA&lt;-digitize.DNA(seqs=TibetanMoth)
digitized.DNA


</code></pre>

<hr>
<h2 id='DNAbin2kmerFreqMatrix'>Calculation of Kmer Frequency Matrix from DNAbin for Both Reference and Query Sequences</h2><span id='topic+DNAbin2kmerFreqMatrix'></span>

<h3>Description</h3>

<p>Calculation of kmer frequency matrices from DNAbin for both reference and query sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DNAbin2kmerFreqMatrix(ref, que, kmer = kmer)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DNAbin2kmerFreqMatrix_+3A_ref">ref</code></td>
<td>
<p>Object of class &quot;DNAbin&quot; used as a reference dataset, which contains taxon information.</p>
</td></tr>
<tr><td><code id="DNAbin2kmerFreqMatrix_+3A_que">que</code></td>
<td>
<p>Object of class &quot;DNAbin&quot;, which needs to be inferred.</p>
</td></tr>
<tr><td><code id="DNAbin2kmerFreqMatrix_+3A_kmer">kmer</code></td>
<td>
<p>a numeric to indicate the length of kmer used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>kmer frequency matrices for both ref and que sequences, but only based on kmers found in ref!!!
new kmers in que will be ignored.
</p>


<h3>Author(s)</h3>

<p>Ai-bing ZHANG, PhD. CNU, Beijing, CHINA. zhangab2008(at)mail.cnu.edu.cn
</p>


<h3>References</h3>

<p>zhangab2008(at)mail.cnu.edu.cn
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(TibetanMoth) 
ref&lt;-as.DNAbin(as.character(TibetanMoth[1:50,]))
que&lt;-as.DNAbin(as.character(TibetanMoth[51:60,]))
out&lt;-DNAbin2kmerFreqMatrix(ref,que,kmer=3)
out


</code></pre>

<hr>
<h2 id='FMF'>Fuzzy Membership Function Value</h2><span id='topic+FMF'></span>

<h3>Description</h3>

<p>Calculation fuzzy membership function value given a distance from query to a potenial
species, maximal intraspecific variation of the potential species theta1, and minimal interspecific
distance (here, the distance between the potential species and its nearest neighbor theta2) 
(fuzzy-set based method, Zhang et al. 2012), different definition of distances could also be used.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FMF(xtheta12)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FMF_+3A_xtheta12">xtheta12</code></td>
<td>
<p>a numerical vector containing three elements, a distance from query to a potenial species,
maximal or sd of intraspecific variation of the potential species theta1,minimal or mean interspecific
distance.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a numeric between 0 and 1.
</p>


<h3>Note</h3>

<p>different definitions of distances could also be used.
</p>


<h3>Author(s)</h3>

<p>Ai-bing ZHANG, Zhi-yong SHI. CNU, Beijing, CHINA, contact at zhangab2008(at)mail.cnu.edu.cn
</p>


<h3>References</h3>

<p>Zhang, A. B., C. Muster, H.B. Liang, C.D. Zhu, R. Crozier, P. Wan, J. Feng, R. D. Ward.(2012). A fuzzy-set-theory-based approach 
to analyse species membership in DNA barcoding. Molecular Ecology, 21(8):1848-63.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
xtheta12&lt;-c(0.6289163,0.1465522,0.6379375) 
FMF.out&lt;-FMF(xtheta12)
FMF.out
</code></pre>

<hr>
<h2 id='FMFtheta12'>Calculate Intraspecific and Interspecific Variation</h2><span id='topic+FMFtheta12'></span>

<h3>Description</h3>

<p>Calculation intraspecific variation (sd) of the potential species theta1, and mean interspecific
distance (here, the mean distance between the potential species and its nearest neighbor theta2) 
(fuzzy-set based method,slightly modified from Zhang et al. 2012). The calculation was done for all species in the 
reference dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FMFtheta12(ref)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FMFtheta12_+3A_ref">ref</code></td>
<td>
<p>object of class &quot;DNAbin&quot; used as a reference dataset, which contains taxon information.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame containing intraspecific (sd, theta1) and interspefic variation (mean) of all species,
and their corresponding nearest neighbor (NN).
</p>


<h3>Author(s)</h3>

<p>Ai-bing ZHANG, PhD. CNU, Beijing, CHINA, contact at zhangab2008 (at) mail.cnu.edu.cn.
</p>


<h3>References</h3>

<p>Zhang, A. B., C. Muster, H.B. Liang, C.D. Zhu, R. Crozier, P. Wan, J. Feng, R. D. Ward.(2012). A fuzzy-set-theory-based approach 
to analyse species membership in DNA barcoding. Molecular Ecology, 21(8):1848-63.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(TibetanMoth) 
ref&lt;-as.DNAbin(as.character(TibetanMoth[1:50,]))
FMF.theta12&lt;-FMFtheta12(ref)
FMF.theta12
</code></pre>

<hr>
<h2 id='NAMES'>Extracts Lables of Samples</h2><span id='topic+NAMES'></span>

<h3>Description</h3>

<p>Extract sequence names from different objects of DNAbin, including generated from fasta2DNAbin() (package:adegenet),
and read.dna() (package:ape).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NAMES(seqs)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="NAMES_+3A_seqs">seqs</code></td>
<td>
<p>object of class &quot;DNAbin&quot;, generated from fasta2DNAbin() (package:adegenet), and read.dna() (package:ape).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character string array/vector.
</p>


<h3>Author(s)</h3>

<p>Ai-bing ZHANG, PhD. CNU, Beijing, CHINA.
</p>


<h3>References</h3>

<p>zhangab2008(at)mail.cnu.edu.cn
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(TibetanMoth) 
seqNames&lt;-NAMES(TibetanMoth)
seqNames
</code></pre>

<hr>
<h2 id='optimize.kmer'>Optimize kmer Length</h2><span id='topic+optimize.kmer'></span>

<h3>Description</h3>

<p>Optimize kmer length by trying kmers which length is in the range from 1 to max.kmer. 
The optimal kmer will have maximal species identification success rate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>optimize.kmer(ref, max.kmer = max.kmer)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="optimize.kmer_+3A_ref">ref</code></td>
<td>
<p>object of class &quot;DNAbin&quot; used as a reference dataset, which contains taxon information.</p>
</td></tr>
<tr><td><code id="optimize.kmer_+3A_max.kmer">max.kmer</code></td>
<td>
<p>a numeric to indicate the length of maximal kmer.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a numeric indicating the optimal kmer in the range examined.
</p>


<h3>Author(s)</h3>

<p>Ai-bing ZHANG, Cai-qing YANG, Meng-di HAO, CNU, Beijing, CHINA.
</p>


<h3>References</h3>

<p>zhangab2008 (at) mail. cnu. edu. cn/zhangab2008 (at) gmail.com.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(TibetanMoth) 
ref&lt;-TibetanMoth[1:10,]
optimial.kmer&lt;-optimize.kmer(ref,max.kmer=5)



</code></pre>

<hr>
<h2 id='pineMothCOI'>
pine Moth COI
</h2><span id='topic+pineMothCOI'></span>

<h3>Description</h3>

<p>COI DNA barcodes of Pine Moth in China.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("pineMothCOI")</code></pre>


<h3>Format</h3>

<p>The format is:
'DNAbin' raw [1:140, 1:652] a t a a ...
- attr(*, &quot;dimnames&quot;)=List of 2
..$ : chr [1:140] &quot;A111,Lasiocampidae_Dendrolimus_punctatus&quot; &quot;A22,Lasiocampidae_Dendrolimus_punctatus&quot; &quot;A23,Lasiocampidae_Dendrolimus_punctatus&quot; &quot;A27,Lasiocampidae_Dendrolimus_punctatus&quot; ...
..$ : NULL
</p>


<h3>Details</h3>

<p>COI DNA barcodes of Pine Moth (Lasiocampidae) in China.
</p>


<h3>Source</h3>

<p>http://dx.plos.org/10.1371/journal.pone.0064428.
</p>


<h3>References</h3>

<p>Dai Q-Y, Gao Q, Wu C-S, Chesters D, Zhu C-D, and A.B. Zhang*. (2012) Phylogenetic Reconstruction and DNA Barcoding for Closely Related Pine Moth Species (Dendrolimus) in China with Multiple Gene Markers. PLoS ONE 7(4): e32544.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pineMothCOI)
pineMothCOI
</code></pre>

<hr>
<h2 id='pineMothITS1'>
pine Moth ITS1
</h2><span id='topic+pineMothITS1'></span>

<h3>Description</h3>

<p>ITS1 sequences of seven closely related pine moths species sampled through China.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("pineMothITS1")</code></pre>


<h3>Format</h3>

<p>The format is:
List of 69
$ A43,Lasiocampidae_ Dendrolimus_ punctatus: raw [1:698] 48 48 48 28 ...
$ A54,Lasiocampidae_ Dendrolimus_ punctatus: raw [1:696] 48 18 28 18 ...
- attr(*, &quot;class&quot;)= chr &quot;DNAbin&quot;
</p>


<h3>Details</h3>

<p>ITS1 used as DNA barcodes for closely related rine moth species (Dendrolimus) in China.
</p>


<h3>Source</h3>

<p>http://dx.plos.org/ 10.1371/ journal.pone.0064428.
</p>


<h3>References</h3>

<p>Dai Q-Y, Gao Q, Wu C-S, Chesters D, Zhu C-D, and A.B. Zhang*. (2012) Phylogenetic Reconstruction and DNA Barcoding for Closely Related Pine Moth Species (Dendrolimus) in China with Multiple Gene Markers. PLoS ONE 7(4): e32544.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pineMothITS1)
pineMothITS1
</code></pre>

<hr>
<h2 id='pineMothITS2'>
pine Moth ITS2
</h2><span id='topic+pineMothITS2'></span>

<h3>Description</h3>

<p>ITS2 sequences of seven closely related pine moths species sampled in China.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("pineMothITS2")</code></pre>


<h3>Format</h3>

<p>The format is:
List of 97
$ A22,Lasiocampidae_ Dendrolimus_ punctatus: raw [1:568] 48 48 48 28 ...
$ A23,Lasiocampidae_ Dendrolimus_ punctatus: raw [1:574] 48 18 28 18 ...
$ A29,Lasiocampidae_ Dendrolimus_ punctatus: raw [1:569] 88 48 48 48 ...
$ A52,Lasiocampidae_ Dendrolimus_ punctatus: raw [1:570] 48 18 28 18 ...
- attr(*, &quot;class&quot;)= chr &quot;DNAbin&quot;
</p>


<h3>Details</h3>

<p>ITS2 used as DNA barcodes for closely related rine moth species (Dendrolimus) in China.
</p>


<h3>Source</h3>

<p>http://dx.plos.org /10.1371/ journal.pone.0064428.
</p>


<h3>References</h3>

<p>Dai Q-Y, Gao Q, Wu C-S, Chesters D, Zhu C-D, and A.B. Zhang*. (2012) Phylogenetic Reconstruction and DNA Barcoding for Closely Related Pine Moth Species (Dendrolimus) in China with Multiple Gene Markers. PLoS ONE 7(4): e32544.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pineMothITS2)
pineMothITS2
</code></pre>

<hr>
<h2 id='sample.ref'>Sample Random Datasets from References (DNAbin)</h2><span id='topic+sample.ref'></span>

<h3>Description</h3>

<p>Randomly sample reference data at different levels of taxon.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample.ref(ref, sample.porp = 0.5, sample.level = "full")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sample.ref_+3A_ref">ref</code></td>
<td>
<p>Object of class &quot;DNAbin&quot; used as a reference dataset, which contains taxon information.</p>
</td></tr>
<tr><td><code id="sample.ref_+3A_sample.porp">sample.porp</code></td>
<td>
<p>a numeric value between 0 and 1, indicating proportion of samples to draw 
at each level of taxon.</p>
</td></tr>
<tr><td><code id="sample.ref_+3A_sample.level">sample.level</code></td>
<td>
<p>a character string choosing from c(&quot;full&quot;,&quot;family&quot;,&quot;genus&quot;,&quot;species&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing the selected samples and the samples left, in DNAbin format stored in a matrix or a list.
</p>


<h3>Note</h3>

<p>the ref must contain information on taxonomy, in format like, &quot;&gt;LS0909030M,Noctuidae_Himalaea_unica&quot;,
i.e., &quot;seqID,family_genus_species&quot;, or &quot;&gt;LS0909030M,Himalaea_unica&quot;; in case there is only one sample/individual
for a taxon level, this sample will be retained in ref.selected.
</p>


<h3>Author(s)</h3>

<p>Ai-bing ZHANG, PhD. CNU, Beijing, CHINA.
</p>


<h3>References</h3>

<p>zhangab2008(at)mail.cnu.edu.cn;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(TibetanMoth) 
data(pineMothITS2)
ref&lt;-TibetanMoth
ref2&lt;-pineMothITS2
out&lt;-sample.ref(ref,sample.porp=0.5,sample.level="full")
out
out2&lt;-sample.ref(ref2,sample.porp=0.5,sample.level="full")
out2


</code></pre>

<hr>
<h2 id='save.ids'>Save Identifications</h2><span id='topic+save.ids'></span>

<h3>Description</h3>

<p>Output identified results to an outfile in temporty directory (found by tempdir() function).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>save.ids(outfile = "identified.txt", ids)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="save.ids_+3A_outfile">outfile</code></td>
<td>
<p>character string to indicate outfile name.</p>
</td></tr>
<tr><td><code id="save.ids_+3A_ids">ids</code></td>
<td>
<p>object of class &quot;BarcodingR&quot;, which contains identified taxon information.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no value returned,but an output file.
</p>


<h3>Author(s)</h3>

<p>Ai-bing ZHANG, PhD. CNU, Beijing, CHINA.
</p>


<h3>References</h3>

<p>zhangab2008(at)mail.cnu.edu.cn
</p>


<h3>See Also</h3>

<p>barcoding.spe.identify()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(TibetanMoth) 
ref&lt;-as.DNAbin(as.character(TibetanMoth[1:50,]))
que&lt;-as.DNAbin(as.character(TibetanMoth[50:60,]))
bsi&lt;-barcoding.spe.identify(ref, que, method = "fuzzyId")
bsi
save.ids(outfile="identified.txt",bsi)
</code></pre>

<hr>
<h2 id='summarize.ref'>Summarize Reference Data</h2><span id='topic+summarize.ref'></span>

<h3>Description</h3>

<p>Summarize taxon information, sequence statistics,barcodes numbers per species for reference dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summarize.ref(ref, taxonStat = TRUE, seqStat = TRUE, barcodeStat = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summarize.ref_+3A_ref">ref</code></td>
<td>
<p>object of class &quot;DNAbin&quot; used as a reference dataset, which contains taxon information, or just an array
containing taxon information only.</p>
</td></tr>
<tr><td><code id="summarize.ref_+3A_taxonstat">taxonStat</code></td>
<td>
<p>logic value to indicate whether the item is calculated only when ref is an object of class &quot;DNAbin&quot;.</p>
</td></tr>
<tr><td><code id="summarize.ref_+3A_seqstat">seqStat</code></td>
<td>
<p>logic value to indicate whether the item is calculated</p>
</td></tr>
<tr><td><code id="summarize.ref_+3A_barcodestat">barcodeStat</code></td>
<td>
<p>logic value to indicate whether the item is calculated</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing taxon statistics, sequence statistics, population parameters,barcoding statistics ()
</p>


<h3>Author(s)</h3>

<p>Ai-bing ZHANG, Meng-di HAO, CNU, Beijing, CHINA.
</p>


<h3>References</h3>

<p>zhangab2008(at)mail.cnu.edu.cn./zhangab2008(at)gmail.com.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(TibetanMoth)
s.r&lt;-summarize.ref(TibetanMoth,taxonStat=TRUE,seqStat=TRUE,barcodeStat=TRUE)
s.r
</code></pre>

<hr>
<h2 id='TDR2'>TDR2 Species Membership Value</h2><span id='topic+TDR2'></span>

<h3>Description</h3>

<p>To calculate TDR value for a set of queries and one potential species. Its value is in the range of [0,1],
0 indicates extremly weak species membership, values close 1 indicating strong species membership.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TDR2(oneSpe, que, boot, boot2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="TDR2_+3A_onespe">oneSpe</code></td>
<td>
<p>object of class &quot;DNAbin&quot; which contains DNA sequences from one species</p>
</td></tr>
<tr><td><code id="TDR2_+3A_que">que</code></td>
<td>
<p>object of class &quot;DNAbin&quot; which contains DNA sequences different samples</p>
</td></tr>
<tr><td><code id="TDR2_+3A_boot">boot</code></td>
<td>
<p>a numeric value indicating times of resampling along sequence columns</p>
</td></tr>
<tr><td><code id="TDR2_+3A_boot2">boot2</code></td>
<td>
<p>a numeric value indicating times of resampling along sequence rows (different samples)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a numeric vector represents TDR values for each query against the species
</p>


<h3>Note</h3>

<p>oneSpe and que should be the same in sequence length, i.e., they should be aligned in prior.
It's strongly recommended that oneSpe should have large enough sample size,e.g., 20.
</p>


<h3>Author(s)</h3>

<p>Ai-bing ZHANG, PhD. CNU, Beijing, CHINA, contact at zhangab2008(at)mail.cnu.edu.cn
</p>


<h3>References</h3>

<p>Jin Q, L,J.He, A.B. Zhang* (2012). A Simple 2D Non-Parametric Resampling Statistical Approach to 
Assess Confidence in Species Identification in DNA Barcoding-An Alternative to Likelihood and Bayesian Approaches.
PLoS ONE 7(12): e50831. doi:10.1371/ journal. pone. 0050831. http://dx.plos.org/ 10.1371/ journal. pone. 0050831.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(TibetanMoth)  
sampleSpeNames&lt;-NAMES(TibetanMoth)
Spp&lt;-gsub(".+,","",sampleSpeNames)
oneSpe&lt;-TibetanMoth[grep("Macdunnoughia_crassisigna", Spp, value = FALSE,fixed = TRUE),] 
oneSpe&lt;-as.DNAbin(as.character(oneSpe[1:5,]))
que&lt;-TibetanMoth[grep("Agrotis_justa", Spp, value = FALSE,fixed = TRUE),] 
que2&lt;-oneSpe[1:2,]
out&lt;-TDR2(oneSpe,que, boot=10,boot2=10) ### true false identification
</code></pre>

<hr>
<h2 id='TibetanMoth'>
Tibetan Moth
</h2><span id='topic+TibetanMoth'></span>

<h3>Description</h3>

<p>COI DNA barcodes of Tibetan Moth in China.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("TibetanMoth")</code></pre>


<h3>Format</h3>

<p>The format is:
'DNAbin' raw [1:319, 1:630] a t a a ...
- attr(*, &quot;dimnames&quot;)=List of 2
..$ : chr [1:319] &quot;LS0909030M,Noctuidae_Himalaea_unica&quot; &quot;LZ0827026M,Noctuidae_Amphipyra_pyramidea&quot; &quot;ML0829010M,Noctuidae_Auchmis_saga&quot; &quot;BM0830055M,Noctuidae_Auchmis_saga&quot; ...
..$ : NULL
</p>


<h3>Details</h3>

<p>COI DNA barcodes of Tibetan Moth (Noctuidae) in China.
</p>


<h3>Source</h3>

<p>http://dx.plos.org/10.1371/journal.pone.0064428.
</p>


<h3>References</h3>

<p>Q. Jin, H.L. Han, X.M. Hu, X.H. Li,C.D. Zhu,S. Y. W. Ho, R. D. Ward, A.B. Zhang* . (2013).Quantifying Species Diversity with a DNA Barcoding-Based Method: Tibetan Moth Species (Noctuidae) on the Qinghai-Tibetan Plateau. PloS One 8: e644.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(TibetanMoth)
TibetanMoth
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
