<!DOCTYPE html><html><head><title>Help for package baseq</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {baseq}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#clean_DNA_file'><p>Clean DNA file</p></a></li>
<li><a href='#clean_DNA_sequence'><p>Clean DNA sequence</p></a></li>
<li><a href='#clean_RNA_file'><p>Clean RNA file</p></a></li>
<li><a href='#clean_RNA_sequence'><p>Clean RNA sequence</p></a></li>
<li><a href='#clean_sequence'><p>Clean DNA or RNA sequence</p></a></li>
<li><a href='#count_bases'><p>Count the number of A's, C's, G's, and T's in a DNA sequence</p></a></li>
<li><a href='#count_seq_pattern'><p>Count frequency of a pattern in a sequence</p></a></li>
<li><a href='#dna_to_protein'><p>Translation of a DNA sequence</p></a></li>
<li><a href='#dna_to_rna'><p>Transcription of a DNA sequence</p></a></li>
<li><a href='#fastq_to_fasta'><p>Convert a FASTQ file to a FASTA file</p></a></li>
<li><a href='#gc_content'><p>Calculate GC content of a DNA sequence</p></a></li>
<li><a href='#gc_content_file'><p>GC content of sequences in a multi FASTA file</p></a></li>
<li><a href='#read.fasta_to_df'><p>Read a fasta file into a dataframe and assign to the environment</p></a></li>
<li><a href='#read.fasta_to_list'><p>Read a fasta file into a list and assign to the environment</p></a></li>
<li><a href='#read.fastq_to_df'><p>Read a Fastq file and store it as a dataframe</p></a></li>
<li><a href='#read.fastq_to_list'><p>Read a Fastq file and store it as a list</p></a></li>
<li><a href='#reverse_complement'><p>Generate Reverse Complement of DNA sequence</p></a></li>
<li><a href='#rna_reverse_complement'><p>Generate Reverse Complement of DNA sequence</p></a></li>
<li><a href='#rna_to_dna'><p>Reverse Transcription of a RNA sequence</p></a></li>
<li><a href='#rna_to_protein'><p>Translation of a RNA sequence</p></a></li>
<li><a href='#write.df_to_fasta'><p>Write a data frame to a fasta file</p></a></li>
<li><a href='#write.df_to_fastq'><p>Write a FASTQ file from a dataframe of reads</p></a></li>
<li><a href='#write.dna_to_rna'><p>Convert DNA file to RNA file</p></a></li>
<li><a href='#write.list_to_fasta'><p>Write a list of sequences to a FASTA file</p></a></li>
<li><a href='#write.list_to_fastq'><p>Write a list of sequence_bases and quality scores to a FASTQ file</p></a></li>
<li><a href='#write.rna_to_dna'><p>Convert RNA file to DNA file</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Basic Sequence Processing Tool for Biological Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.4</td>
</tr>
<tr>
<td>Description:</td>
<td>Primarily created as an easy and understanding way to do basic sequences surrounding the central dogma of molecular biology.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ambuvjyn/baseq">https://github.com/ambuvjyn/baseq</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ambuvjyn/baseq/issues">https://github.com/ambuvjyn/baseq/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-05-02 19:07:38 UTC; ambuv</td>
</tr>
<tr>
<td>Author:</td>
<td>Ambu Vijayan <a href="https://orcid.org/0000-0001-8924-5685"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  J. Sreekumar <a href="https://orcid.org/0000-0002-4253-6378"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut] (Principal Scientist, ICAR - Central Tuber Crops Research
    Institute)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ambu Vijayan &lt;ambuvjyn@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-05-03 11:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='clean_DNA_file'>Clean DNA file</h2><span id='topic+clean_DNA_file'></span>

<h3>Description</h3>

<p>This function reads a multi FASTA file containing DNA sequences, removes any characters other
than A, T, G, and C, and writes the cleaned sequences to a new multi FASTA file. The output
file name is generated from the input file name with the suffix '_clean.fasta'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clean_DNA_file(input_file, output_dir = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clean_DNA_file_+3A_input_file">input_file</code></td>
<td>
<p>The name of the input multi FASTA file.</p>
</td></tr>
<tr><td><code id="clean_DNA_file_+3A_output_dir">output_dir</code></td>
<td>
<p>The directory where the output file will be saved. If not given,
the output file will be saved in the same directory as the input file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string specifying the path to the output FASTA file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#sample_file_path_three &lt;- system.file("extdata", "sample2_fa.fasta", package = "baseq")
#tempdir &lt;- tempdir()
#temp_file_path &lt;- file.path(tempdir, basename(sample_file_path_three))
#file.copy(sample_file_path_three, temp_file_path, overwrite = TRUE)
#clean_DNA_file(temp_file_path, output_dir = tempdir)

# Write to working directory
# clean_DNA_file(file_path)

# Write to custom directory
# clean_DNA_file(file_path, output_dir = "/path/to/directory/")

</code></pre>

<hr>
<h2 id='clean_DNA_sequence'>Clean DNA sequence</h2><span id='topic+clean_DNA_sequence'></span>

<h3>Description</h3>

<p>This function takes a DNA sequence as input and removes any characters other than A, C, G, and T.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clean_DNA_sequence(sequence)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clean_DNA_sequence_+3A_sequence">sequence</code></td>
<td>
<p>DNA sequence to be cleaned</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Cleaned DNA sequence
</p>


<h3>Examples</h3>

<pre><code class='language-R'>clean_DNA_sequence("ATGTCGTAGCTAGCTN")
# Output: "ATGTCGTAGCTAGCT"

</code></pre>

<hr>
<h2 id='clean_RNA_file'>Clean RNA file</h2><span id='topic+clean_RNA_file'></span>

<h3>Description</h3>

<p>This function reads a multi FASTA file containing RNA sequences, removes any characters other
than A, T, G, and C, and writes the cleaned sequences to a new multi FASTA file. The output
file name is generated from the input file name with the suffix '_clean.fasta'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clean_RNA_file(input_file, output_dir = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clean_RNA_file_+3A_input_file">input_file</code></td>
<td>
<p>The name of the input multi FASTA file.</p>
</td></tr>
<tr><td><code id="clean_RNA_file_+3A_output_dir">output_dir</code></td>
<td>
<p>The directory where the output file will be saved. If not given,
the output file will be saved in the same directory as the input file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string specifying the path to the output FASTA file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#sample_file_path_three &lt;- system.file("extdata", "sample2_fa.fasta", package = "baseq")
#tempdir &lt;- tempdir()
#temp_file_path &lt;- file.path(tempdir, basename(sample_file_path_three))
#file.copy(sample_file_path_three, temp_file_path, overwrite = TRUE)
#clean_RNA_file(temp_file_path, output_dir = tempdir)

# Write to working directory
# clean_RNA_file(file_path)

# Write to custom directory
# clean_RNA_file(file_path, output_dir = "/path/to/directory/")

</code></pre>

<hr>
<h2 id='clean_RNA_sequence'>Clean RNA sequence</h2><span id='topic+clean_RNA_sequence'></span>

<h3>Description</h3>

<p>This function takes a RNA sequence as input and removes any characters other than A, C, G, and T.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clean_RNA_sequence(sequence)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clean_RNA_sequence_+3A_sequence">sequence</code></td>
<td>
<p>RNA sequence to be cleaned</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Cleaned RNA sequence
</p>


<h3>Examples</h3>

<pre><code class='language-R'>clean_RNA_sequence("AUGUCGTAGCTAGCTN")
# Output: "AUGUCGAGCAGC"

</code></pre>

<hr>
<h2 id='clean_sequence'>Clean DNA or RNA sequence</h2><span id='topic+clean_sequence'></span>

<h3>Description</h3>

<p>This function takes a DNA or RNA sequence as input and removes any characters that are not A, C, G, T (for DNA) or A, C, G, U (for RNA).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clean_sequence(sequence, type = "DNA")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clean_sequence_+3A_sequence">sequence</code></td>
<td>
<p>A character string containing the DNA or RNA sequence to be cleaned.</p>
</td></tr>
<tr><td><code id="clean_sequence_+3A_type">type</code></td>
<td>
<p>A character string indicating the type of sequence. The default is &quot;DNA&quot;. If set to &quot;RNA&quot;, the function will remove any characters that are not A, C, G, U.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string containing the cleaned DNA or RNA sequence.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>clean_sequence("atgcNnRYMK") # Returns "ATGC"
clean_sequence("auggcuuNnRYMK", type = "RNA") # Returns "AUGGCUU"
</code></pre>

<hr>
<h2 id='count_bases'>Count the number of A's, C's, G's, and T's in a DNA sequence</h2><span id='topic+count_bases'></span>

<h3>Description</h3>

<p>This function takes a single argument, a DNA sequence as a character string,
and counts the number of A's, C's, G's, and T's in the sequence. The counts
are returned as a named vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count_bases(sequence)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="count_bases_+3A_sequence">sequence</code></td>
<td>
<p>a character string containing a DNA sequence</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named integer vector containing the counts of A's, C's, G's, and T's
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sequence &lt;- "ATCGAGCTAGCTAGCTAGCTAGCT"
count_bases(sequence)
# A  C  G  T
# 6  6  6  6
</code></pre>

<hr>
<h2 id='count_seq_pattern'>Count frequency of a pattern in a sequence</h2><span id='topic+count_seq_pattern'></span>

<h3>Description</h3>

<p>This function counts the frequency of a specific character or pattern in a given sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count_seq_pattern(seq, pattern)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="count_seq_pattern_+3A_seq">seq</code></td>
<td>
<p>A character vector representing the sequence to count the pattern in.</p>
</td></tr>
<tr><td><code id="count_seq_pattern_+3A_pattern">pattern</code></td>
<td>
<p>A character string representing the pattern to count in the sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer representing the count of the pattern in the sequence.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>seq &lt;- "ATGGTGCTCCGTGCGCGCGCGCGCGCGCGCGCGCGCGCGCGCGCGCGCGCGCGCGCGCGCGCGCGCGCGCGCTACGTAG"
count_seq_pattern(seq, "CG")
# [1] 31

</code></pre>

<hr>
<h2 id='dna_to_protein'>Translation of a DNA sequence</h2><span id='topic+dna_to_protein'></span>

<h3>Description</h3>

<p>This function takes a DNA sequence as input and translates it in all six reading frames.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dna_to_protein(sequence)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dna_to_protein_+3A_sequence">sequence</code></td>
<td>
<p>A character string representing a DNA sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of character strings representing the translated protein sequences in all six frames.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sequence &lt;- "ATCGAGCTAGCTAGCTAGCTAGCT"
dna_to_protein(sequence)
# Returns a list containing the translated protein sequences in all six frames:
# $`Frame F1`
# [1] "IELAS"
#
# $`Frame F2`
# [1] "SS"
#
# $`Frame F3`
# [1] "RAS"
#
# $`Frame R1`
# [1] "S"
#
# $`Frame R2`
# [1] "AS"
#
# $`Frame R3`
# [1] "LAS"
</code></pre>

<hr>
<h2 id='dna_to_rna'>Transcription of a DNA sequence</h2><span id='topic+dna_to_rna'></span>

<h3>Description</h3>

<p>This function takes a DNA sequence as input and returns its RNA transcript.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dna_to_rna(sequence)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dna_to_rna_+3A_sequence">sequence</code></td>
<td>
<p>A character string representing a DNA sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string representing the RNA transcript of the input DNA sequence.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sequence &lt;- "ATCGAGCTAGCTAGCTAGCTAGCT"
dna_to_rna(sequence)
# Returns "AUCGAGCUAGCUAGCUAGCUAGCU"
</code></pre>

<hr>
<h2 id='fastq_to_fasta'>Convert a FASTQ file to a FASTA file</h2><span id='topic+fastq_to_fasta'></span>

<h3>Description</h3>

<p>This function converts a FASTQ file to a FASTA file. The output file has the same name as the input
FASTQ file, but with the extension changed to <code>.fasta</code>. This function removes the <code>@</code> symbol at the beginning
of FASTQ sequence names and replaces it with the <code>&gt;</code> symbol for the FASTA format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fastq_to_fasta(fastq_file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fastq_to_fasta_+3A_fastq_file">fastq_file</code></td>
<td>
<p>A character string specifying the path to the input FASTQ file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string specifying the path to the output FASTA file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#sample_file_path_two &lt;- system.file("extdata", "sample_fq.fastq", package = "baseq")
#tempdir &lt;- tempdir()
#temp_file_path &lt;- file.path(tempdir, basename(sample_file_path_two))
#file.copy(sample_file_path_two, temp_file_path, overwrite = TRUE)
#fastq_to_fasta(temp_file_path)

# Output: "path/to/Temp/tempfoldername/sample_fq.fasta"

</code></pre>

<hr>
<h2 id='gc_content'>Calculate GC content of a DNA sequence</h2><span id='topic+gc_content'></span>

<h3>Description</h3>

<p>Calculates the percentage of nucleotides in a DNA sequence that are either guanine (G) or cytosine (C).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gc_content(sequence)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gc_content_+3A_sequence">sequence</code></td>
<td>
<p>A character string containing the DNA sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric value representing the percentage of nucleotides in the sequence that are G or C.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sequence &lt;- "ATCGAGCTAGCTAGCTAGCTAGCT"
gc_content(sequence)
50
</code></pre>

<hr>
<h2 id='gc_content_file'>GC content of sequences in a multi FASTA file</h2><span id='topic+gc_content_file'></span>

<h3>Description</h3>

<p>Function to calculate GC content of sequences in a multi FASTA file
and write the results to a new FASTA file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gc_content_file(input_file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gc_content_file_+3A_input_file">input_file</code></td>
<td>
<p>A string indicating the path and name of the input multi-FASTA file</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>#sample_file_path &lt;- system.file("extdata", "sample_fa.fasta", package = "baseq")
#clean_DNA_file(sample_file_path)

</code></pre>

<hr>
<h2 id='read.fasta_to_df'>Read a fasta file into a dataframe and assign to the environment</h2><span id='topic+read.fasta_to_df'></span>

<h3>Description</h3>

<p>This function reads a fasta file and creates a dataframe with two columns:
Header and Sequence. The dataframe is then assigned to the environment with the
name same as the fasta file name but without the .fasta extension.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.fasta_to_df(fasta_file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.fasta_to_df_+3A_fasta_file">fasta_file</code></td>
<td>
<p>The path to the fasta file to be read.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function does not return anything. It assigns the resulting dataframe
to the environment.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Read in sequences from a FASTA file

sample_file_path &lt;- system.file("extdata", "sample_fa.fasta", package = "baseq")
read.fasta_to_df(sample_file_path)


</code></pre>

<hr>
<h2 id='read.fasta_to_list'>Read a fasta file into a list and assign to the environment</h2><span id='topic+read.fasta_to_list'></span>

<h3>Description</h3>

<p>This function reads a fasta file and creates a list with two columns:
Header and Sequence. The list is then assigned to the environment with the
name same as the fasta file name but without the .fasta extension.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.fasta_to_list(fasta_file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.fasta_to_list_+3A_fasta_file">fasta_file</code></td>
<td>
<p>The path to the fasta file to be read.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function does not return anything. It assigns the resulting list
to the environment.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Read in sequences from a FASTA file

sample_file_path &lt;- system.file("extdata", "sample_fa.fasta", package = "baseq")
read.fasta_to_list(sample_file_path)

# Access a specific sequence by name
# sample_fa[["sample_seq.1"]]

</code></pre>

<hr>
<h2 id='read.fastq_to_df'>Read a Fastq file and store it as a dataframe</h2><span id='topic+read.fastq_to_df'></span>

<h3>Description</h3>

<p>This function reads a Fastq file and stores it as a dataframe with three columns: Header, Sequence, and QualityScore.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.fastq_to_df(fastq_file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.fastq_to_df_+3A_fastq_file">fastq_file</code></td>
<td>
<p>A character string specifying the path to the Fastq file to be read.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a dataframe with three columns: Header, Sequence, and QualityScore.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# Read in sequences from a FASTQ file

#sample_file_path_two &lt;- system.file("extdata", "sample_fq.fastq", package = "baseq")
#read.fastq_to_df(sample_file_path_two)


</code></pre>

<hr>
<h2 id='read.fastq_to_list'>Read a Fastq file and store it as a list</h2><span id='topic+read.fastq_to_list'></span>

<h3>Description</h3>

<p>This function reads a Fastq file and stores it as a list with three columns: Header, Sequence, and QualityScore.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.fastq_to_list(fastq_file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.fastq_to_list_+3A_fastq_file">fastq_file</code></td>
<td>
<p>A character string specifying the path to the Fastq file to be read.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a list with three columns: Header, Sequence, and QualityScore.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Read in sequences from a FASTQ file

sample_file_path_two &lt;- system.file("extdata", "sample_fq.fastq", package = "baseq")
read.fastq_to_list(sample_file_path_two)

</code></pre>

<hr>
<h2 id='reverse_complement'>Generate Reverse Complement of DNA sequence</h2><span id='topic+reverse_complement'></span>

<h3>Description</h3>

<p>Given a DNA sequence, the function generates the reverse complement of the sequence and returns it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reverse_complement(sequence)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reverse_complement_+3A_sequence">sequence</code></td>
<td>
<p>A character string containing the DNA sequence to be reversed and complemented</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string containing the reverse complement of the input DNA sequence
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sequence &lt;- "ATCGAGCTAGCTAGCTAGCTAGCT"
reverse_complement(sequence)
# [1] "AGCTAGCTAGCTAGCTAGCTCGAT"
</code></pre>

<hr>
<h2 id='rna_reverse_complement'>Generate Reverse Complement of DNA sequence</h2><span id='topic+rna_reverse_complement'></span>

<h3>Description</h3>

<p>Given a DNA sequence, the function generates the reverse complement of the sequence and returns it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rna_reverse_complement(sequence)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rna_reverse_complement_+3A_sequence">sequence</code></td>
<td>
<p>A character string containing the DNA sequence to be reversed and complemented</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string containing the reverse complement of the input DNA sequence
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sequence &lt;- "AUCGAGCUAGCUAGCUAGCUAGCU"
rna_reverse_complement(sequence)
# [1] "AGCUAGCUAGCUAGCUAGCUCGAU"
</code></pre>

<hr>
<h2 id='rna_to_dna'>Reverse Transcription of a RNA sequence</h2><span id='topic+rna_to_dna'></span>

<h3>Description</h3>

<p>This function takes a RNA sequence as input and returns its DNA transcript.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rna_to_dna(sequence)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rna_to_dna_+3A_sequence">sequence</code></td>
<td>
<p>A character string representing a RNA sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string representing the RNA transcript of the input RNA sequence.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sequence &lt;- "AUCGAGCUAGCUAGCUAGCUAGCU"
rna_to_dna(sequence)
# Returns "ATCGAGCTAGCTAGCTAGCTAGCT"
</code></pre>

<hr>
<h2 id='rna_to_protein'>Translation of a RNA sequence</h2><span id='topic+rna_to_protein'></span>

<h3>Description</h3>

<p>This function takes a RNA sequence as input and translates it in all six reading frames.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rna_to_protein(sequence)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rna_to_protein_+3A_sequence">sequence</code></td>
<td>
<p>A character string representing a RNA sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of character strings representing the translated protein sequences in all six frames.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sequence &lt;- "AUCGAGCUAGCUAGCUAGCUAGCU"
rna_to_protein(sequence)
# Returns a list containing the translated protein sequences in all six frames:
# $`Frame F1`
# [1] "IELAS"
#
# $`Frame F2`
# [1] "SS"
#
# $`Frame F3`
# [1] "RAS"
#
# $`Frame R1`
# [1] "S"
#
# $`Frame R2`
# [1] "AS"
#
# $`Frame R3`
# [1] "LAS"
</code></pre>

<hr>
<h2 id='write.df_to_fasta'>Write a data frame to a fasta file</h2><span id='topic+write.df_to_fasta'></span>

<h3>Description</h3>

<p>This function writes a data frame to a fasta file with the same name as the data frame. The data frame is assumed to have two columns, &quot;Header&quot; and &quot;Sequence&quot;, which represent the header and sequence lines of each fasta record, respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.df_to_fasta(df, output_dir = getwd())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.df_to_fasta_+3A_df">df</code></td>
<td>
<p>A data frame containing fasta records with &quot;Header&quot; and &quot;Sequence&quot; columns.</p>
</td></tr>
<tr><td><code id="write.df_to_fasta_+3A_output_dir">output_dir</code></td>
<td>
<p>The directory path where the output file should be written. If not provided, the working directory will be used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function does not return a value, but writes a fasta file to the specified output directory or the working directory.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#sample_file_path &lt;- system.file("extdata", "sample_fa.fasta", package = "baseq")
#tempdir &lt;- tempdir()
#temp_file_path &lt;- file.path(tempdir, basename(sample_file_path))
#file.copy(sample_file_path, temp_file_path, overwrite = TRUE)
#read.fasta_to_df(sample_file_path)
#write.df_to_fasta(sample_fa, output_dir = tempdir)

# Write to working directory
# write.df_to_fasta(sample_fa)

# Write to custom directory
# write.df_to_fasta(sample_fa, output_dir = "/path/to/directory/")

</code></pre>

<hr>
<h2 id='write.df_to_fastq'>Write a FASTQ file from a dataframe of reads</h2><span id='topic+write.df_to_fastq'></span>

<h3>Description</h3>

<p>Write a FASTQ file from a dataframe of reads
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.df_to_fastq(df, output_dir = getwd())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.df_to_fastq_+3A_df">df</code></td>
<td>
<p>A dataframe containing reads in the format &quot;Header&quot;, &quot;Sequence&quot;, and &quot;QualityScore&quot;.</p>
</td></tr>
<tr><td><code id="write.df_to_fastq_+3A_output_dir">output_dir</code></td>
<td>
<p>An optional argument specifying the directory where the FASTQ file should be saved. If not specified, the file will be saved in the working directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A FASTQ file with the same name as the input dataframe.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#sample_file_path_two &lt;- system.file("extdata", "sample_fq.fastq", package = "baseq")
#tempdir &lt;- tempdir()
#temp_file_path &lt;- file.path(tempdir, basename(sample_file_path_two))
#file.copy(sample_file_path_two, temp_file_path, overwrite = TRUE)
#read.fastq_to_df(sample_file_path_two)
#write.df_to_fastq(sample_fq, output_dir = tempdir)

# Write to working directory
# write.df_to_fastq(sample_fq)

# Write to custom directory
# write.df_to_fastq(sample_fq, output_dir = "/path/to/directory/")

</code></pre>

<hr>
<h2 id='write.dna_to_rna'>Convert DNA file to RNA file</h2><span id='topic+write.dna_to_rna'></span>

<h3>Description</h3>

<p>This function reads a multi FASTA file containing DNA sequences, converts each DNA sequence to RNA sequence,
and writes the RNA sequences to a new multi FASTA file. The output file name is generated from the input file
name with the suffix '_rna.fasta'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.dna_to_rna(input_file, output_dir = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.dna_to_rna_+3A_input_file">input_file</code></td>
<td>
<p>The name of the input multi FASTA file.</p>
</td></tr>
<tr><td><code id="write.dna_to_rna_+3A_output_dir">output_dir</code></td>
<td>
<p>The directory where the output file will be saved. If not given,
the output file will be saved in the same directory as the input file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string specifying the path to the output FASTA file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#sample_file_path &lt;- system.file("extdata", "sample_fa.fasta", package = "baseq")
#tempdir &lt;- tempdir()
#temp_file_path &lt;- file.path(tempdir, basename(sample_file_path))
#file.copy(sample_file_path, temp_file_path, overwrite = TRUE)
#write.dna_to_rna(temp_file_path, output_dir = tempdir)

# Write to working directory
# write.dna_to_rna(file_path)

# Write to custom directory
# write.dna_to_rna(file_path, output_dir = "/path/to/directory/")

</code></pre>

<hr>
<h2 id='write.list_to_fasta'>Write a list of sequences to a FASTA file</h2><span id='topic+write.list_to_fasta'></span>

<h3>Description</h3>

<p>This function takes a list of sequences and writes them to a FASTA file. The name of the list
is used as the base name for the output file with the .fasta extension. Each sequence in the
list is written to the output file in FASTA format with the sequence name as the header.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.list_to_fasta(sequence_list, output_dir = getwd())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.list_to_fasta_+3A_sequence_list">sequence_list</code></td>
<td>
<p>A list of sequences where each element of the list is a character string
representing a single sequence.</p>
</td></tr>
<tr><td><code id="write.list_to_fasta_+3A_output_dir">output_dir</code></td>
<td>
<p>The directory path where the output file should be written. If not provided, the working directory will be used.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>sequences &lt;- list("ACGT", "ATCG")
tempdir &lt;- tempdir()
write.list_to_fasta(sequences, output_dir = tempdir)

# Write to working directory
# write.list_to_fasta(sequences)

# Write to custom directory
# write.list_to_fasta(sequences, output_dir = "/path/to/directory/")

</code></pre>

<hr>
<h2 id='write.list_to_fastq'>Write a list of sequence_bases and quality scores to a FASTQ file</h2><span id='topic+write.list_to_fastq'></span>

<h3>Description</h3>

<p>This function takes a list of sequence_bases and quality scores and writes them to a FASTQ file.
The name of the list is used as the base name for the output file with the .fastq extension.
Each sequence in the list is written to the output file in FASTQ format with the sequence
name as the header and the quality scores on the following line.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.list_to_fastq(sequence_list, output_dir = getwd())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.list_to_fastq_+3A_sequence_list">sequence_list</code></td>
<td>
<p>A list of sequence_bases where each element of the list is a named list
containing &quot;Sequence&quot; and &quot;QualityScore&quot; elements.</p>
</td></tr>
<tr><td><code id="write.list_to_fastq_+3A_output_dir">output_dir</code></td>
<td>
<p>The directory path where the output file should be written. If not provided, the working directory will be used.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>sequence_bases &lt;- list("ACGT", "ATCG")
quality_scores &lt;- list("IIII", "JJJJ")
sequences &lt;- list(seq1=list(Sequence=sequence_bases[[1]], QualityScore=quality_scores[[1]]),
                       seq2=list(Sequence=sequence_bases[[2]], QualityScore=quality_scores[[2]]))
tempdir &lt;- tempdir()
write.list_to_fastq(sequences, output_dir = tempdir)

# Write to working directory
# write.list_to_fastq(sequences)

# Write to custom directory
# write.list_to_fastq(sequences, output_dir = "/path/to/directory/")

</code></pre>

<hr>
<h2 id='write.rna_to_dna'>Convert RNA file to DNA file</h2><span id='topic+write.rna_to_dna'></span>

<h3>Description</h3>

<p>This function reads a multi FASTA file containing RNA sequences, converts each RNA sequence to DNA sequence,
and writes the DNA sequences to a new multi FASTA file. The output file name is generated from the input file
name with the suffix '_rna.fasta'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.rna_to_dna(input_file, output_dir = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.rna_to_dna_+3A_input_file">input_file</code></td>
<td>
<p>The name of the input multi FASTA file.</p>
</td></tr>
<tr><td><code id="write.rna_to_dna_+3A_output_dir">output_dir</code></td>
<td>
<p>The directory where the output file will be saved. If not given,
the output file will be saved in the same directory as the input file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string specifying the path to the output FASTA file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#sample_file_path &lt;- system.file("extdata", "sample3_fa.fasta", package = "baseq")
#tempdir &lt;- tempdir()
#temp_file_path &lt;- file.path(tempdir, basename(sample_file_path))
#file.copy(sample_file_path, temp_file_path, overwrite = TRUE)
#write.rna_to_dna(temp_file_path, output_dir = tempdir)

# Write to working directory
# write.rna_to_dna(file_path)

# Write to custom directory
# write.rna_to_dna(file_path, output_dir = "/path/to/directory/")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
