<!DOCTYPE html><html lang="en"><head><title>Help for package batchtma</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {batchtma}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#adjust_batch'><p>Adjust for batch effects</p></a></li>
<li><a href='#batchtma'><p>batchtma: Methods to address batch effects</p></a></li>
<li><a href='#diagnose_models'><p>Model diagnostics after batch adjustment</p></a></li>
<li><a href='#plot_batch'><p>Plot biomarkers by batch</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Batch Effect Adjustments</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.6</td>
</tr>
<tr>
<td>Description:</td>
<td>Different adjustment methods for batch effects in biomarker data,
  such as from tissue microarrays. Some methods attempt to retain differences 
  between batches that may be due to between-batch differences in "biological"
  factors that influence biomarker values.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>broom (&ge; 0.7.0), dplyr (&ge; 1.0.0), geepack, ggplot2, limma,
nnet, purrr (&ge; 0.3.0), quantreg, rlang (&ge; 0.4.0), stringr,
tibble, tidyr (&ge; 1.1.0), magrittr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, tidyverse</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://stopsack.github.io/batchtma/">https://stopsack.github.io/batchtma/</a>,
<a href="https://github.com/stopsack/batchtma">https://github.com/stopsack/batchtma</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-12-04 23:17:25 UTC; stopsack</td>
</tr>
<tr>
<td>Author:</td>
<td>Konrad Stopsack <a href="https://orcid.org/0000-0002-0722-1311"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Travis Gerke <a href="https://orcid.org/0000-0002-9500-8907"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Konrad Stopsack &lt;stopsack@post.harvard.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-12-06 08:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='adjust_batch'>Adjust for batch effects</h2><span id='topic+adjust_batch'></span>

<h3>Description</h3>

<p><code>adjust_batch</code> generates biomarker levels for the variable(s)
<code>markers</code> in the dataset <code>data</code> that are corrected
(adjusted) for batch effects, i.e. differential measurement
error between levels of <code>batch</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjust_batch(
  data,
  markers,
  batch,
  method = c("simple", "standardize", "ipw", "quantreg", "quantnorm"),
  confounders = NULL,
  suffix = "_adjX",
  ipw_truncate = c(0.025, 0.975),
  quantreg_tau = c(0.25, 0.75),
  quantreg_method = "fn"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="adjust_batch_+3A_data">data</code></td>
<td>
<p>Data set</p>
</td></tr>
<tr><td><code id="adjust_batch_+3A_markers">markers</code></td>
<td>
<p>Variable name(s) to batch-adjust. Select
multiple variables with tidy evaluation, e.g.,
<code>markers = starts_with("biomarker")</code>.</p>
</td></tr>
<tr><td><code id="adjust_batch_+3A_batch">batch</code></td>
<td>
<p>Categorical variable indicating batch.</p>
</td></tr>
<tr><td><code id="adjust_batch_+3A_method">method</code></td>
<td>
<p>Method for batch effect correction:
</p>

<ul>
<li> <p><code>simple</code>  Simple means per batch will be subtracted.
No adjustment for confounders.
</p>
</li>
<li> <p><code>standardize</code>  Means per batch after standardization
for confounders in linear models will be subtracted.
If no <code>confounders</code> are supplied, <code>method = simple</code>
is equivalent and will be used.
</p>
</li>
<li> <p><code>ipw</code>  Means per batch after inverse-probability
weighting for assignment to a specific batch in multinomial
models, conditional on confounders, will be subtracted.
Stabilized weights are used, truncated at quantiles
defined by the <code>ipw_truncate</code> parameters. If no
<code>confounders</code> are supplied, <code>method = simple</code>
is equivalent and will be used.
</p>
</li>
<li> <p><code>quantreg</code>  Lower quantiles (default: 25th percentile)
and ranges between a lower and an upper quantile (default: 75th
percentile) will be unified between batches, allowing for
differences in both parameters due to confounders. Set the two
quantiles using the <code>quantreg_tau</code> parameters.
</p>
</li>
<li> <p><code>quantnorm</code>  Quantile normalization between batches. No
adjustment for confounders.
</p>
</li></ul>
</td></tr>
<tr><td><code id="adjust_batch_+3A_confounders">confounders</code></td>
<td>
<p>Optional: Confounders, i.e. determinants of
biomarker levels that differ between batches. Only used if
<code>method = standardize</code>, <code>method = ipw</code>, or
<code>method = quantreg</code>, i.e. methods that attempt to retain
some of these &quot;true&quot; between-batch differences. Select multiple
confounders with tidy evaluation, e.g.,
<code>confounders = c(age, age_squared, sex)</code>.</p>
</td></tr>
<tr><td><code id="adjust_batch_+3A_suffix">suffix</code></td>
<td>
<p>Optional: What string to append to variable names
after batch adjustment. Defaults to <code>"_adjX"</code>, with
<code>X</code> depending on <code>method</code>:
</p>

<ul>
<li> <p><code>_adj2</code> from <code>method = simple</code>
</p>
</li>
<li> <p><code>_adj3</code> from <code>method = standardize</code>
</p>
</li>
<li> <p><code>_adj4</code> from <code>method = ipw</code>
</p>
</li>
<li> <p><code>_adj5</code> from <code>method = quantreg</code>
</p>
</li>
<li> <p><code>_adj6</code> from <code>method = quantnorm</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="adjust_batch_+3A_ipw_truncate">ipw_truncate</code></td>
<td>
<p>Optional and used for <code>method = ipw</code> only:
Lower and upper quantiles for truncation of stabilized
weights. Defaults to <code>c(0.025, 0.975)</code>.</p>
</td></tr>
<tr><td><code id="adjust_batch_+3A_quantreg_tau">quantreg_tau</code></td>
<td>
<p>Optional and used for <code>method = quantreg</code> only:
Quantiles to scale. Defaults to <code>c(0.25, 0.75)</code>.</p>
</td></tr>
<tr><td><code id="adjust_batch_+3A_quantreg_method">quantreg_method</code></td>
<td>
<p>Optional and used for <code>method = quantreg</code> only:
Algorithmic method to fit quantile regression. Defaults to
<code>"fn"</code>. See parameter <code>method</code> of <code><a href="quantreg.html#topic+rq">rq</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If no true differences between batches are expected, because
samples have been randomized to batches, then a <code>method</code>
that returns adjusted values with equal means
(<code>method = simple</code>) or with equal rank values
(<code>method = quantnorm</code>) for all batches is appropriate.
</p>
<p>If the distribution of determinants of biomarker values
(<code>confounders</code>) differs between batches, then a
<code>method</code> that retains these &quot;true&quot; differences
between batches while adjusting for batch effects
may be appropriate: <code>method = standardize</code> and
<code>method = ipw</code> address means; <code>method = quantreg</code>
addresses lower values and dynamic range separately.
</p>
<p>Which <code>method</code> to choose depends on the properties of
batch effects (affecting means or also variance?) and
the presence and strength of confounding. For the two
mean-only confounder-adjusted methods, the choice may depend
on  whether the confounder&ndash;batch association (<code>method = ipw</code>)
or the confounder&ndash;biomarker association
(<code>method = standardize</code>) can be modeled better.
Generally, if batch effects are present, any adjustment
method tends to perform better than no adjustment in
reducing bias and increasing between-study reproducibility.
See references.
</p>
<p>All adjustment approaches except <code>method = quantnorm</code>
are based on linear models. It is recommended that variables
for <code>markers</code> and <code>confounders</code> first be transformed
as necessary (e.g., <code><a href="base.html#topic+log">log</a></code> transformations or
<code><a href="splines.html#topic+splines">splines</a></code>). Scaling or mean centering are not necessary,
and adjusted values are returned on the original scale.
Parameters <code>markers</code>, <code>batch</code>, and <code>confounders</code>
support tidy evaluation.
</p>
<p>Observations with missing values for the <code>markers</code> and
<code>confounders</code> will be ignored in the estimation of adjustment
parameters, as are empty batches. Batch effect-adjusted values
for observations with existing marker values but missing
confounders are based on adjustment parameters derived from the
other observations in a batch with non-missing confounders.
</p>


<h3>Value</h3>

<p>The <code>data</code> dataset with batch effect-adjusted
variable(s) added at the end. Model diagnostics, using
the attribute <code>.batchtma</code> of this dataset, are available
via the <code><a href="#topic+diagnose_models">diagnose_models</a></code> function.
</p>


<h3>Author(s)</h3>

<p>Konrad H. Stopsack
</p>


<h3>References</h3>

<p>Stopsack KH, Tyekucheva S, Wang M, Gerke TA, Vaselkiv JB, Penney KL,
Kantoff PW, Finn SP, Fiorentino M, Loda M, Lotan TL, Parmigiani G+,
Mucci LA+ (+ equal contribution). Extent, impact, and mitigation of
batch effects in tumor biomarker studies using tissue microarrays.
bioRxiv 2021.06.29.450369; doi: https://doi.org/10.1101/2021.06.29.450369
(This R package, all methods descriptions, and further recommendations.)
</p>
<p>Rosner B, Cook N, Portman R, Daniels S, Falkner B.
Determination of blood pressure percentiles in
normal-weight children: some methodological issues.
Am J Epidemiol 2008;167(6):653-66. (Basis for
<code>method = standardize</code>)
</p>
<p>Bolstad BM, Irizarry RA, Åstrand M, Speed TP.
A comparison of normalization methods for high density
oligonucleotide array data based on variance and bias.
Bioinformatics 2003;19:185–193. (<code>method = quantnorm</code>)
</p>


<h3>See Also</h3>

<p><a href="https://stopsack.github.io/batchtma/">https://stopsack.github.io/batchtma/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Data frame with two batches
# Batch 2 has higher values of biomarker and confounder
df &lt;- data.frame(
  tma = rep(1:2, times = 10),
  biomarker = rep(1:2, times = 10) +
    runif(max = 5, n = 20),
  confounder = rep(0:1, times = 10) +
    runif(max = 10, n = 20)
)

# Adjust for batch effects
# Using simple means, ignoring the confounder:
adjust_batch(
  data = df,
  markers = biomarker,
  batch = tma,
  method = simple
)
# Returns data set with new variable "biomarker_adj2"

# Use quantile regression, include the confounder,
# change suffix of returned variable:
adjust_batch(
  data = df,
  markers = biomarker,
  batch = tma,
  method = quantreg,
  confounders = confounder,
  suffix = "_batchadjusted"
)
# Returns data set with new variable "biomarker_batchadjusted"
</code></pre>

<hr>
<h2 id='batchtma'>batchtma: Methods to address batch effects</h2><span id='topic+batchtma'></span>

<h3>Description</h3>

<p>The goal of the batchtma is to provide functions for
batch effect-adjusting biomarker data. It implements different
methods that address batch effects while retaining differences
between batches that may be due to “true” underlying differences
in factors that drive biomarker values (confounders).
</p>


<h3>Functions</h3>

<p><code><a href="#topic+adjust_batch">adjust_batch</a></code>: Adjust for batch effects
</p>
<p><code><a href="#topic+diagnose_models">diagnose_models</a></code>: Model diagnostics after batch adjustment
</p>
<p><code><a href="#topic+plot_batch">plot_batch</a></code>: Plot biomarkers by batch
</p>


<h3>References</h3>

<p>Stopsack KH, Tyekucheva S, Wang M, Gerke TA, Vaselkiv JB, Penney KL,
Kantoff PW, Finn SP, Fiorentino M, Loda M, Lotan TL, Parmigiani G+,
Mucci LA+ (+ equal contribution). Extent, impact, and mitigation of
batch effects in tumor biomarker studies using tissue microarrays.
bioRxiv 2021.06.29.450369; doi: https://doi.org/10.1101/2021.06.29.450369
</p>


<h3>See Also</h3>

<p><a href="https://stopsack.github.io/batchtma/">https://stopsack.github.io/batchtma/</a>
</p>

<hr>
<h2 id='diagnose_models'>Model diagnostics after batch adjustment</h2><span id='topic+diagnose_models'></span>

<h3>Description</h3>

<p>After <code><a href="#topic+adjust_batch">adjust_batch</a></code> has performed
adjustment for batch effects, <code>diagnose_models</code>
provides an overview of parameters and adjustment models.
Information is only available about the most recent
run of <code><a href="#topic+adjust_batch">adjust_batch</a></code> on a dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diagnose_models(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="diagnose_models_+3A_data">data</code></td>
<td>
<p>Batch-adjusted dataset (in which
<code><a href="#topic+adjust_batch">adjust_batch</a></code> has stored information on batch
adjustment in the attribute <code>.batchtma</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List:
</p>

<ul>
<li> <p><code>adjust_method</code> Method used for batch adjustment
(see <code><a href="#topic+adjust_batch">adjust_batch</a></code>).
</p>
</li>
<li> <p><code>markers</code> Variables of biomarkers for adjustment
</p>
</li>
<li> <p><code>suffix</code> Suffix appended to variable names
</p>
</li>
<li> <p><code>batchvar</code> Variable indicating batch
</p>
</li>
<li> <p><code>confounders</code> Confounders, i.e. determinants of
biomarker levels that differ between batches.
Returned only if used by the model.
</p>
</li>
<li> <p><code>adjust_parameters</code> Tibble of parameters used to
obtain adjust biomarker levels. Parameters differ between
methods:
</p>

<ul>
<li> <p><code>simple</code>, <code>standardize</code>, and <code>ipw</code>: Estimated adjustment
parameters are a tibble with one <code>batchmean</code> per <code>marker</code>
and <code>.batchvar</code>.
</p>
</li>
<li> <p><code>quantreg</code> returns a tibble with numerous values per
<code>marker</code> and <code>.batchvar</code>: unadjusted (<code>un_...</code>) and
adjusted (<code>ad_...</code>) estimates of the lower (<code>..._lo</code>) and
upper quantile (<code>..._hi</code>) and interquantile range (<code>..._iq</code>),
plus the lower (<code>all_lo</code>) and upper quantiles (<code>all_hi</code>)
across all batches.
</p>
</li>
<li> <p><code>quantnorm</code> does not explicitly estimate parameters.
</p>
</li></ul>

</li>
<li> <p><code>model_fits</code> List of model fit objects, one
per biomarker. Models differ between methods:
</p>

<ul>
<li> <p><code>standardize</code>: Linear regression model for the biomarker with
<code>.batchvar</code> and <code>confounders</code> as predictors, from which
marginal predictions of batch means for each batch are obtained.
</p>
</li>
<li> <p><code>ipw</code>: Logistic (2 batches) or multinomial models for assignment
to a specific batch with <code>.batchvar</code> as the response and
<code>confounders</code> as the predictors, used to generate stabilized
inverse-probability weights that are then used in a linear regression
model to estimate marginally standardized batch means.
</p>
</li>
<li> <p><code>quantreg</code>: Quantile regression with the marker as the response
variable and <code>.batchvar</code> and <code>confounders</code> as predictors.
</p>
</li>
<li> <p><code>simple</code> and <code>quantnorm</code> do not fit any regression models.
</p>
</li></ul>

</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Data frame with two batches
# Batch 2 has higher values of biomarker and confounder
df &lt;- data.frame(
  tma = rep(1:2, times = 10),
  biomarker = rep(1:2, times = 10) +
    runif(max = 5, n = 20),
  confounder = rep(0:1, times = 10) +
    runif(max = 10, n = 20)
)

# Adjust for batch effects
df2 &lt;- adjust_batch(
  data = df,
  markers = biomarker,
  batch = tma,
  method = quantreg,
  confounders = confounder
)

# Show overview of model diagnostics:
diagnose_models(data = df2)

# Obtain first fitted regression model:
fit &lt;- diagnose_models(data = df2)$model_fits[[1]][[1]]

# Obtain residuals for this model:
residuals(fit)
</code></pre>

<hr>
<h2 id='plot_batch'>Plot biomarkers by batch</h2><span id='topic+plot_batch'></span>

<h3>Description</h3>

<p>To provide a simple visualization of potential batch
effects, <code>plot_batch</code> generates a Tukey
box plot overlaid by a jittered
dot plot, inspired by the Stata plugin <code>stripplot</code>.
</p>
<p>Boxes span from the 1st to the 3rd quartile; thick lines
indicate medians; whiskers span up to 1.5 times the
interquartile range; and asterisks indicate means.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_batch(
  data,
  marker,
  batch,
  color = NULL,
  maxlevels = 15,
  title = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_batch_+3A_data">data</code></td>
<td>
<p>Dataset.</p>
</td></tr>
<tr><td><code id="plot_batch_+3A_marker">marker</code></td>
<td>
<p>Variable indicating the biomarker.</p>
</td></tr>
<tr><td><code id="plot_batch_+3A_batch">batch</code></td>
<td>
<p>Variable indicating the batch.</p>
</td></tr>
<tr><td><code id="plot_batch_+3A_color">color</code></td>
<td>
<p>Optional: third variable to use for symbol
color and shape. For example, <code>color</code> can be used
to show differences in a confounder.</p>
</td></tr>
<tr><td><code id="plot_batch_+3A_maxlevels">maxlevels</code></td>
<td>
<p>Optional: Maximum number of
levels for <code>color</code> parameter to accept as a discrete
variable, rather than a continuous variable.
Defaults to <code>15</code>.</p>
</td></tr>
<tr><td><code id="plot_batch_+3A_title">title</code></td>
<td>
<p>Optional: character string that specifies plot
title</p>
</td></tr>
<tr><td><code id="plot_batch_+3A_...">...</code></td>
<td>
<p>Optional: Passed on to <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object, which can be further
modified using standard ggplot2 functions. See examples.
</p>


<h3>References</h3>

<p>Cox NJ (2003). STRIPPLOT: Stata module for strip plots
(one-way dot plots). Statistical Software Components
S433401, Boston College Department of Economics,
revised 11 Oct 2020.
</p>
<p>Manimaran S, Selby HM, Okrah K, Ruberman C, Leek JT,
Quackenbush J, Haibe-Kains B, Bravo HC, Johnson WE (2016).
BatchQC: interactive software for evaluating sample
and batch effects in genomic data. Bioinformatics.
doi:10.1093/bioinformatics/btw538
</p>


<h3>See Also</h3>

<p>More powerful visualizations of batch effects exist
in the BatchQC package:
</p>
<p><a href="http://bioconductor.org/packages/release/bioc/html/BatchQC.html">http://bioconductor.org/packages/release/bioc/html/BatchQC.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Define example data
df &lt;- data.frame(
  tma = rep(1:2, times = 10),
  biomarker = rep(1:2, times = 10) +
    runif(max = 5, n = 20),
  confounder = rep(0:1, times = 10) +
    runif(max = 10, n = 20)
)

# Visualize batch effects:
plot_batch(
  data = df,
  marker = biomarker,
  batch = tma,
  color = confounder
)

# Label y-axis, changing graph like other ggplots:
plot_batch(
  data = df,
  marker = biomarker,
  batch = tma,
  color = confounder
) +
  ggplot2::labs(y = "Biomarker (variable 'noisy')")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
