<!DOCTYPE html><html lang="en"><head><title>Help for package BayesSurvive</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BayesSurvive}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BayesSurvive-package'><p>BayesSurvive: Bayesian Survival Models for High-Dimensional Data</p></a></li>
<li><a href='#BayesSurvive'><p>Fit Bayesian Cox Models</p></a></li>
<li><a href='#coef.BayesSurvive'><p>Create a dataframe of estimated coefficients</p></a></li>
<li><a href='#func_MCMC'><p>Function to run MCMC sampling</p></a></li>
<li><a href='#func_MCMC_graph'><p>Function to learn MRF graph</p></a></li>
<li><a href='#plot.BayesSurvive'><p>Create a plot of estimated coefficients</p></a></li>
<li><a href='#plotBrier'><p>Time-dependent Brier scores</p></a></li>
<li><a href='#predict.BayesSurvive'><p>Predict survival risk</p></a></li>
<li><a href='#simData'><p>Simulated survival data</p></a></li>
<li><a href='#UpdateGamma'><p>Subfunctions to update parameters</p></a></li>
<li><a href='#UpdateRPlee11'><p>Update coefficients of Bayesian Cox Models</p></a></li>
<li><a href='#VS'><p>Function to perform variable selection</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Bayesian Survival Models for High-Dimensional Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-06-03</td>
</tr>
<tr>
<td>Description:</td>
<td>An implementation of Bayesian survival models with graph-structured selection priors for sparse identification of omics features predictive of survival (Madjar et al., 2021 &lt;<a href="https://doi.org/10.1186%2Fs12859-021-04483-z">doi:10.1186/s12859-021-04483-z</a>&gt;) and its extension to use a fixed graph via a Markov Random Field (MRF) prior for capturing known structure of omics features, e.g. disease-specific pathways from the Kyoto Encyclopedia of Genes and Genomes database.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ocbe-uio/BayesSurvive">https://github.com/ocbe-uio/BayesSurvive</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ocbe-uio/BayesSurvive/issues">https://github.com/ocbe-uio/BayesSurvive/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo (&ge; 0.9.000)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, ggplot2, GGally, mvtnorm, survival, riskRegression,
utils, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-06-03 19:22:26 UTC; zhiz</td>
</tr>
<tr>
<td>Author:</td>
<td>Zhi Zhao [aut, cre],
  Katrin Madjar [aut],
  Tobias Østmo Hermansen [aut],
  Manuela Zucknick [ctb],
  Jörg Rahnenführer [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zhi Zhao &lt;zhi.zhao@medisin.uio.no&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-06-04 13:20:12 UTC</td>
</tr>
</table>
<hr>
<h2 id='BayesSurvive-package'>BayesSurvive: Bayesian Survival Models for High-Dimensional Data</h2><span id='topic+BayesSurvive-package'></span>

<h3>Description</h3>

<p>An implementation of Bayesian survival models with graph-structured selection priors for sparse identification of omics features predictive of survival (Madjar et al., 2021 <a href="https://doi.org/10.1186/s12859-021-04483-z">doi:10.1186/s12859-021-04483-z</a>) and its extension to use a fixed graph via a Markov Random Field (MRF) prior for capturing known structure of omics features, e.g. disease-specific pathways from the Kyoto Encyclopedia of Genes and Genomes database.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Zhi Zhao <a href="mailto:zhi.zhao@medisin.uio.no">zhi.zhao@medisin.uio.no</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Katrin Madjar
</p>
</li>
<li><p> Tobias Østmo Hermansen
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Manuela Zucknick [contributor]
</p>
</li>
<li><p> Jörg Rahnenführer [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/ocbe-uio/BayesSurvive">https://github.com/ocbe-uio/BayesSurvive</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/ocbe-uio/BayesSurvive/issues">https://github.com/ocbe-uio/BayesSurvive/issues</a>
</p>
</li></ul>


<hr>
<h2 id='BayesSurvive'>Fit Bayesian Cox Models</h2><span id='topic+BayesSurvive'></span>

<h3>Description</h3>

<p>This is the main function to fit a Bayesian Cox model with graph-structured
selection priors for sparse identification of high-dimensional covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BayesSurvive(
  survObj,
  model.type = "Pooled",
  MRF2b = FALSE,
  MRF.G = TRUE,
  g.ini = 0,
  hyperpar = NULL,
  initial = NULL,
  nIter = 1,
  burnin = 0,
  thin = 1,
  output_graph_para = FALSE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BayesSurvive_+3A_survobj">survObj</code></td>
<td>
<p>a list containing observed data from <code>n</code> subjects with
components <code>t</code>, <code>di</code>, <code>X</code>. For graphical learning of the
Markov random field prior, <code>survObj</code> should be a list of the list with
survival and covariates data. For subgroup models with or without graphical
learning, <code>survObj</code> should be a list of multiple lists with each
component list representing each subgroup's survival and covariates data</p>
</td></tr>
<tr><td><code id="BayesSurvive_+3A_model.type">model.type</code></td>
<td>
<p>a method option from
<code>c("Pooled", "CoxBVSSL", "Sub-struct")</code>. To enable graphical learning
for &quot;Pooled&quot; model, please specify <code>list(survObj)</code> where <code>survObj</code>
is the list of <code>t</code>, <code>di</code> and <code>X</code></p>
</td></tr>
<tr><td><code id="BayesSurvive_+3A_mrf2b">MRF2b</code></td>
<td>
<p>logical value. <code>MRF2b = TRUE</code> means two different
hyperparameters b in MRF prior (values b01 and b02) and <code>MRF2b = FALSE</code>
means one hyperparamter b in MRF prior</p>
</td></tr>
<tr><td><code id="BayesSurvive_+3A_mrf.g">MRF.G</code></td>
<td>
<p>logical value. <code>MRF.G = TRUE</code> is to fix the MRF graph which
is provided in the argument <code>hyperpar</code>, and <code>MRF.G = FALSE</code> is to
use graphical model for leanring the MRF graph</p>
</td></tr>
<tr><td><code id="BayesSurvive_+3A_g.ini">g.ini</code></td>
<td>
<p>initial values for latent edge inclusion indicators in graph,
should be a value in [0,1]. 0 or 1: set all random edges to 0 or 1; value in
(0,1): rate of indicators randomly set to 1, the remaining indicators are 0</p>
</td></tr>
<tr><td><code id="BayesSurvive_+3A_hyperpar">hyperpar</code></td>
<td>
<p>a list containing prior parameter values</p>
</td></tr>
<tr><td><code id="BayesSurvive_+3A_initial">initial</code></td>
<td>
<p>a list containing prior parameters' initial values</p>
</td></tr>
<tr><td><code id="BayesSurvive_+3A_niter">nIter</code></td>
<td>
<p>the number of iterations of the chain</p>
</td></tr>
<tr><td><code id="BayesSurvive_+3A_burnin">burnin</code></td>
<td>
<p>number of iterations to discard at the start of the chain.
Default is 0</p>
</td></tr>
<tr><td><code id="BayesSurvive_+3A_thin">thin</code></td>
<td>
<p>thinning MCMC intermediate results to be stored</p>
</td></tr>
<tr><td><code id="BayesSurvive_+3A_output_graph_para">output_graph_para</code></td>
<td>
<p>allow (<code>TRUE</code>) or suppress (<code>FALSE</code>) the
output for parameters 'G', 'V', 'C' and 'Sig' in the graphical model
if <code>MRF.G = FALSE</code></p>
</td></tr>
<tr><td><code id="BayesSurvive_+3A_verbose">verbose</code></td>
<td>
<p>logical value to display the progess of MCMC</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>BayesSurvive</code> is saved as
<code>obj_BayesSurvive.rda</code> in the output file, including the following components:
</p>

<ul>
<li><p> input - a list of all input parameters by the user
</p>
</li>
<li><p> output - a list of the all output estimates:
</p>

<ul>
<li><p> &quot;<code>gamma.p</code>&quot; - a matrix with MCMC intermediate estimates of the indicator variables of regression coefficients.
</p>
</li>
<li><p> &quot;<code>beta.p</code>&quot; - a matrix with MCMC intermediate estimates of the regression coefficients.
</p>
</li>
<li><p> &quot;<code>h.p</code>&quot; - a matrix with MCMC intermediate estimates of the increments in the cumulative baseline hazard in each interval.
</p>
</li></ul>

</li>
<li><p> call - the matched call.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
library("BayesSurvive")
set.seed(123)

# Load the example dataset
data("simData", package = "BayesSurvive")

dataset &lt;- list(
  "X" = simData[[1]]$X,
  "t" = simData[[1]]$time,
  "di" = simData[[1]]$status
)

# Initial value: null model without covariates
initial &lt;- list("gamma.ini" = rep(0, ncol(dataset$X)))
# Hyperparameters
hyperparPooled &lt;- list(
  "c0"     = 2, # prior of baseline hazard
  "tau"    = 0.0375, # sd (spike) for coefficient prior
  "cb"     = 20, # sd (spike) for coefficient prior
  "pi.ga"  = 0.02, # prior variable selection probability for standard Cox models
  "a"      = -4, # hyperparameter in MRF prior
  "b"      = 0.1, # hyperparameter in MRF prior
  "G"      = simData$G # hyperparameter in MRF prior
)


# run Bayesian Cox with graph-structured priors
fit &lt;- BayesSurvive(
  survObj = dataset, hyperpar = hyperparPooled,
  initial = initial, nIter = 100
)

# show posterior mean of coefficients and 95% credible intervals
library("GGally")
plot(fit) +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 90, size = 7))


</code></pre>

<hr>
<h2 id='coef.BayesSurvive'>Create a dataframe of estimated coefficients</h2><span id='topic+coef.BayesSurvive'></span>

<h3>Description</h3>

<p>Estimate regression coefficients with posterior mean/median, credible
intervals, standard deviation, or MPM estimates, posterior gammas
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BayesSurvive'
coef(
  object,
  MPM = FALSE,
  type = "mean",
  CI = 95,
  SD = FALSE,
  subgroup = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="coef.BayesSurvive_+3A_object">object</code></td>
<td>
<p>an object of class <code>BayesSurvive</code></p>
</td></tr>
<tr><td><code id="coef.BayesSurvive_+3A_mpm">MPM</code></td>
<td>
<p>logical value to obtain MPM coefficients. Default: FALSE</p>
</td></tr>
<tr><td><code id="coef.BayesSurvive_+3A_type">type</code></td>
<td>
<p>type of point estimates of regression coefficients. One of
<code>c("mean", "median")</code>. Default is <code>mean</code></p>
</td></tr>
<tr><td><code id="coef.BayesSurvive_+3A_ci">CI</code></td>
<td>
<p>size (level, as a percentage) of the credible interval to report.
Default: 95, i.e. a 95% credible interval</p>
</td></tr>
<tr><td><code id="coef.BayesSurvive_+3A_sd">SD</code></td>
<td>
<p>logical value to show each coefficient's standard deviation over
MCMC iterations</p>
</td></tr>
<tr><td><code id="coef.BayesSurvive_+3A_subgroup">subgroup</code></td>
<td>
<p>index of the subgroup for visualizing posterior coefficients</p>
</td></tr>
<tr><td><code id="coef.BayesSurvive_+3A_...">...</code></td>
<td>
<p>other arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library("BayesSurvive")
set.seed(123)

# Load the example dataset
data("simData", package = "BayesSurvive")

dataset &lt;- list(
  "X" = simData[[1]]$X,
  "t" = simData[[1]]$time,
  "di" = simData[[1]]$status
)

# Initial value: null model without covariates
initial &lt;- list("gamma.ini" = rep(0, ncol(dataset$X)))
# Hyperparameters
hyperparPooled &lt;- list(
  "c0"     = 2, # prior of baseline hazard
  "tau"    = 0.0375, # sd for coefficient prior
  "cb"     = 20, # sd for coefficient prior
  "pi.ga"  = 0.02, # prior variable selection probability for standard Cox models
  "a"      = -4, # hyperparameter in MRF prior
  "b"      = 0.1, # hyperparameter in MRF prior
  "G"      = simData$G # hyperparameter in MRF prior
)


# run Bayesian Cox with graph-structured priors
fit &lt;- BayesSurvive(
  survObj = dataset, hyperpar = hyperparPooled,
  initial = initial, nIter = 100
)

# show posterior coefficients
betas &lt;- coef(fit)
head(betas)


</code></pre>

<hr>
<h2 id='func_MCMC'>Function to run MCMC sampling</h2><span id='topic+func_MCMC'></span>

<h3>Description</h3>

<p>This an internal function for MCMC sampling
</p>


<h3>Usage</h3>

<pre><code class='language-R'>func_MCMC(
  survObj,
  hyperpar,
  initial,
  nIter,
  thin,
  burnin,
  S,
  method,
  MRF_2b,
  MRF_G,
  output_graph_para,
  verbose
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="func_MCMC_+3A_survobj">survObj</code></td>
<td>
<p>a list containing observed data from <code>n</code> subjects;
<code>t</code>, <code>di</code>, <code>X</code>. See details for more information</p>
</td></tr>
<tr><td><code id="func_MCMC_+3A_hyperpar">hyperpar</code></td>
<td>
<p>a list containing prior parameter values</p>
</td></tr>
<tr><td><code id="func_MCMC_+3A_initial">initial</code></td>
<td>
<p>a list containing prior parameters' initial values</p>
</td></tr>
<tr><td><code id="func_MCMC_+3A_niter">nIter</code></td>
<td>
<p>the number of iterations of the chain</p>
</td></tr>
<tr><td><code id="func_MCMC_+3A_thin">thin</code></td>
<td>
<p>thinning MCMC intermediate results to be stored</p>
</td></tr>
<tr><td><code id="func_MCMC_+3A_burnin">burnin</code></td>
<td>
<p>number of iterations to discard at the start of the chain.
Default is 0</p>
</td></tr>
<tr><td><code id="func_MCMC_+3A_s">S</code></td>
<td>
<p>the number of subgroups</p>
</td></tr>
<tr><td><code id="func_MCMC_+3A_method">method</code></td>
<td>
<p>a method option from
<code>c("Pooled", "CoxBVSSL", "Sub-struct")</code></p>
</td></tr>
<tr><td><code id="func_MCMC_+3A_mrf_2b">MRF_2b</code></td>
<td>
<p>two different b in MRF prior for subgraphs G_ss and G_rs</p>
</td></tr>
<tr><td><code id="func_MCMC_+3A_mrf_g">MRF_G</code></td>
<td>
<p>logical value. <code>MRF_G = TRUE</code> is to fix the MRF graph which
is provided in the argument <code>hyperpar</code>, and <code>MRF_G = FALSE</code> is to
use graphical model for leanring the MRF graph</p>
</td></tr>
<tr><td><code id="func_MCMC_+3A_output_graph_para">output_graph_para</code></td>
<td>
<p>allow (<code>TRUE</code>) or suppress (<code>FALSE</code>) the
output for parameters 'G', 'V', 'C' and 'Sig' in the graphical model
if <code>MRF_G = FALSE</code></p>
</td></tr>
<tr><td><code id="func_MCMC_+3A_verbose">verbose</code></td>
<td>
<p>logical value to display the progess of MCMC</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object saving the MCMC results with components including
'gamma.p', 'beta.p', 'h.p', 'gamma.margin', 'beta.margin', 's', 'eta0',
'kappa0', 'c0', 'pi.ga', 'tau', 'cb', 'accept.RW', 'log.jpost', 'log.like',
'post.gamma'
</p>

<hr>
<h2 id='func_MCMC_graph'>Function to learn MRF graph</h2><span id='topic+func_MCMC_graph'></span>

<h3>Description</h3>

<p>This an internal function for MCMC sampling
</p>


<h3>Usage</h3>

<pre><code class='language-R'>func_MCMC_graph(sobj, hyperpar, ini, S, method, MRF_2b)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="func_MCMC_graph_+3A_sobj">sobj</code></td>
<td>
<p>a list containing observed data from <code>n</code> subjects;
<code>t</code>, <code>di</code>, <code>X</code>. See details for more information</p>
</td></tr>
<tr><td><code id="func_MCMC_graph_+3A_hyperpar">hyperpar</code></td>
<td>
<p>a list containing prior parameter values</p>
</td></tr>
<tr><td><code id="func_MCMC_graph_+3A_ini">ini</code></td>
<td>
<p>a list containing prior parameters' ini values</p>
</td></tr>
<tr><td><code id="func_MCMC_graph_+3A_s">S</code></td>
<td>
<p>the number of subgroups</p>
</td></tr>
<tr><td><code id="func_MCMC_graph_+3A_method">method</code></td>
<td>
<p>a method option from
<code>c("Pooled", "CoxBVSSL", "Sub-struct")</code></p>
</td></tr>
<tr><td><code id="func_MCMC_graph_+3A_mrf_2b">MRF_2b</code></td>
<td>
<p>two different b in MRF prior for subgraphs G_ss and G_rs</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object with components &quot;Sig&quot; the updated covariance matrices,
&quot;G.ini&quot; the updated graph, &quot;V.ini&quot; the updated variances for precision
matrices in all subgroups, &quot;C.ini&quot; the updated precision matrices omega for
each subgroup
</p>

<hr>
<h2 id='plot.BayesSurvive'>Create a plot of estimated coefficients</h2><span id='topic+plot.BayesSurvive'></span>

<h3>Description</h3>

<p>Plot point estimates of regression coefficients and 95% credible intervals
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BayesSurvive'
plot(x, type = "mean", interval = TRUE, subgroup = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.BayesSurvive_+3A_x">x</code></td>
<td>
<p>an object of class <code>BayesSurvive</code> or a matrix. If <code>x</code>
is a matrix, use <code>BayesSurvive:::plot.BayesSurvive(x)</code></p>
</td></tr>
<tr><td><code id="plot.BayesSurvive_+3A_type">type</code></td>
<td>
<p>type of point estimates of regression coefficients. One of
<code>c("mean", "median")</code>. Default is <code>mean</code></p>
</td></tr>
<tr><td><code id="plot.BayesSurvive_+3A_interval">interval</code></td>
<td>
<p>logical argument to show 95% credible intervals. Default
is <code>TRUE</code></p>
</td></tr>
<tr><td><code id="plot.BayesSurvive_+3A_subgroup">subgroup</code></td>
<td>
<p>index of the subgroup for visualizing posterior coefficients</p>
</td></tr>
<tr><td><code id="plot.BayesSurvive_+3A_...">...</code></td>
<td>
<p>additional arguments sent to <code>ggplot2::geom_point()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library("BayesSurvive")
set.seed(123)

# Load the example dataset
data("simData", package = "BayesSurvive")

dataset &lt;- list(
  "X" = simData[[1]]$X,
  "t" = simData[[1]]$time,
  "di" = simData[[1]]$status
)

# Initial value: null model without covariates
initial &lt;- list("gamma.ini" = rep(0, ncol(dataset$X)))
# Hyperparameters
hyperparPooled &lt;- list(
  "c0"     = 2, # prior of baseline hazard
  "tau"    = 0.0375, # sd for coefficient prior
  "cb"     = 20, # sd for coefficient prior
  "pi.ga"  = 0.02, # prior variable selection probability for standard Cox models
  "a"      = -4, # hyperparameter in MRF prior
  "b"      = 0.1, # hyperparameter in MRF prior
  "G"      = simData$G # hyperparameter in MRF prior
)


# run Bayesian Cox with graph-structured priors
fit &lt;- BayesSurvive(
  survObj = dataset, hyperpar = hyperparPooled,
  initial = initial, nIter = 100
)

# show posterior mean of coefficients and 95% credible intervals
library("GGally")
plot(fit) +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 90, size = 7))


</code></pre>

<hr>
<h2 id='plotBrier'>Time-dependent Brier scores</h2><span id='topic+plotBrier'></span>

<h3>Description</h3>

<p>Predict time-dependent Brier scores based on Cox regression models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotBrier(
  object,
  survObj.new = NULL,
  method = "mean",
  times = NULL,
  subgroup = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotBrier_+3A_object">object</code></td>
<td>
<p>fitted object obtained with <code>BayesSurvive</code></p>
</td></tr>
<tr><td><code id="plotBrier_+3A_survobj.new">survObj.new</code></td>
<td>
<p>a list containing observed data from new subjects with
components <code>t</code>, <code>di</code>, <code>X</code></p>
</td></tr>
<tr><td><code id="plotBrier_+3A_method">method</code></td>
<td>
<p>option to use the posterior mean (<code>"mean"</code>) of coefficients
for prediction or Bayesian model averaging (<code>"BMA"</code>) for prediction</p>
</td></tr>
<tr><td><code id="plotBrier_+3A_times">times</code></td>
<td>
<p>maximum time point to evaluate the prediction</p>
</td></tr>
<tr><td><code id="plotBrier_+3A_subgroup">subgroup</code></td>
<td>
<p>index of the subgroup in <code>survObj.new</code> for prediction.
Default value is 1</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot2::ggplot</code> object. See <code>?ggplot2::ggplot</code> for more
details of the object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library("BayesSurvive")
set.seed(123)

# Load the example dataset
data("simData", package = "BayesSurvive")

dataset &lt;- list(
  "X" = simData[[1]]$X,
  "t" = simData[[1]]$time,
  "di" = simData[[1]]$status
)

# Initial value: null model without covariates
initial &lt;- list("gamma.ini" = rep(0, ncol(dataset$X)))
# Hyperparameters
hyperparPooled &lt;- list(
  "c0"     = 2, # prior of baseline hazard
  "tau"    = 0.0375, # sd for coefficient prior
  "cb"     = 20, # sd for coefficient prior
  "pi.ga"  = 0.02, # prior variable selection probability for standard Cox models
  "a"      = -4, # hyperparameter in MRF prior
  "b"      = 0.1, # hyperparameter in MRF prior
  "G"      = simData$G # hyperparameter in MRF prior
)


# run Bayesian Cox with graph-structured priors
fit &lt;- BayesSurvive(
  survObj = dataset, hyperpar = hyperparPooled,
  initial = initial, nIter = 100
)
# predict survival probabilities of the train data
plotBrier(fit, survObj.new = dataset)


</code></pre>

<hr>
<h2 id='predict.BayesSurvive'>Predict survival risk</h2><span id='topic+predict.BayesSurvive'></span>

<h3>Description</h3>

<p>Predict survival probability, (cumulative) hazard or (integrated) Brier
scores based on Cox regression models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BayesSurvive'
predict(
  object,
  survObj.new,
  type = "brier",
  method = "mean",
  times = NULL,
  subgroup = 1,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predict.BayesSurvive_+3A_object">object</code></td>
<td>
<p>fitted object obtained with <code>BayesSurvive</code></p>
</td></tr>
<tr><td><code id="predict.BayesSurvive_+3A_survobj.new">survObj.new</code></td>
<td>
<p>a list containing observed data from new subjects with
components <code>t</code>, <code>di</code>, <code>x</code>. If <code>type</code> is among
<code>c("hazard", "cumhazard", "survival")</code>, only <code>survObj.new$X</code> is
needed.</p>
</td></tr>
<tr><td><code id="predict.BayesSurvive_+3A_type">type</code></td>
<td>
<p>option to chose for predicting brier scores with <code>type="brier"</code>
or one of <code>type=c("brier", "hazard", "cumhazard", "survival")</code>)</p>
</td></tr>
<tr><td><code id="predict.BayesSurvive_+3A_method">method</code></td>
<td>
<p>option to use the posterior mean (<code>"mean"</code>) of coefficients
for prediction or Bayesian model averaging (<code>"BMA"</code>) for prediction</p>
</td></tr>
<tr><td><code id="predict.BayesSurvive_+3A_times">times</code></td>
<td>
<p>time points at which to evaluate the risks. If <code>NULL</code>
(default), the event/censoring times are used. If <code>type="brier"</code>, the
largest one of the <code>times</code> is used</p>
</td></tr>
<tr><td><code id="predict.BayesSurvive_+3A_subgroup">subgroup</code></td>
<td>
<p>index of the subgroup in <code>survObj.new</code> for prediction.
Default value is 1</p>
</td></tr>
<tr><td><code id="predict.BayesSurvive_+3A_verbose">verbose</code></td>
<td>
<p>logical value to print IBS of the NULL model and the Bayesian
Cox model</p>
</td></tr>
<tr><td><code id="predict.BayesSurvive_+3A_...">...</code></td>
<td>
<p>not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object including seven components with the first compoment as
the specified argument <code>type</code>. The other components of the list are
&quot;se&quot;, &quot;band&quot;, &quot;type&quot;, &quot;diag&quot;, &quot;baseline&quot; and &quot;times&quot;, see function
<code>riskRegression::predictCox</code> for details
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library("BayesSurvive")
set.seed(123)

# Load the example dataset
data("simData", package = "BayesSurvive")

dataset &lt;- list(
  "X" = simData[[1]]$X,
  "t" = simData[[1]]$time,
  "di" = simData[[1]]$status
)

# Initial value: null model without covariates
initial &lt;- list("gamma.ini" = rep(0, ncol(dataset$X)))
# Hyperparameters
hyperparPooled &lt;- list(
  "c0"     = 2, # prior of baseline hazard
  "tau"    = 0.0375, # sd for coefficient prior
  "cb"     = 20, # sd for coefficient prior
  "pi.ga"  = 0.02, # prior variable selection probability for standard Cox models
  "a"      = -4, # hyperparameter in MRF prior
  "b"      = 0.1, # hyperparameter in MRF prior
  "G"      = simData$G # hyperparameter in MRF prior
)


# run Bayesian Cox with graph-structured priors
fit &lt;- BayesSurvive(
  survObj = dataset, hyperpar = hyperparPooled,
  initial = initial, nIter = 100
)
# predict survival probabilities of the train data
predict(fit, survObj.new = dataset)


</code></pre>

<hr>
<h2 id='simData'>Simulated survival data</h2><span id='topic+simData'></span>

<h3>Description</h3>

<p>Simulated data set for a quick test. The data set is a list with six
components:  covariates <code>"X"</code>, survival times <code>"time"</code>,
event status <code>"status"</code>. The R code for generating the simulated data
is given in the Examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simData
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 3.
</p>

<hr>
<h2 id='UpdateGamma'>Subfunctions to update parameters</h2><span id='topic+UpdateGamma'></span>

<h3>Description</h3>

<p>This contains subfunctions to update parameters gammas, betas, baseline
hazard and graph learning parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UpdateGamma(sobj, hyperpar, ini, S, method, MRF_G, MRF_2b)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="UpdateGamma_+3A_sobj">sobj</code></td>
<td>
<p>a list containing observed data</p>
</td></tr>
<tr><td><code id="UpdateGamma_+3A_hyperpar">hyperpar</code></td>
<td>
<p>a list containing prior parameter values</p>
</td></tr>
<tr><td><code id="UpdateGamma_+3A_ini">ini</code></td>
<td>
<p>a list containing prior parameters' initial values</p>
</td></tr>
<tr><td><code id="UpdateGamma_+3A_s">S</code></td>
<td>
<p>the number of subgroups</p>
</td></tr>
<tr><td><code id="UpdateGamma_+3A_method">method</code></td>
<td>
<p>a method option from
<code>c("Pooled", "CoxBVSSL", "Sub-struct", "Subgroup")</code></p>
</td></tr>
<tr><td><code id="UpdateGamma_+3A_mrf_g">MRF_G</code></td>
<td>
<p>logical value. <code>MRF_G = TRUE</code> is to fix the MRF graph which
is provided in the argument <code>hyperpar</code>, and <code>MRF_G = FALSE</code> is to
use graphical model for leanring the MRF graph</p>
</td></tr>
<tr><td><code id="UpdateGamma_+3A_mrf_2b">MRF_2b</code></td>
<td>
<p>two different b in MRF prior for subgraphs G_ss and G_rs</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object with two components for the latent variable selection
indicators gamma with either independent Bernoulli prior
</p>

<hr>
<h2 id='UpdateRPlee11'>Update coefficients of Bayesian Cox Models</h2><span id='topic+UpdateRPlee11'></span>

<h3>Description</h3>

<p>This an internal function to update coefficients of the Bayesian Cox Lasso Model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UpdateRPlee11(sobj, hyperpar, ini, S, method, MRF_G)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="UpdateRPlee11_+3A_sobj">sobj</code></td>
<td>
<p>a list containing observed data</p>
</td></tr>
<tr><td><code id="UpdateRPlee11_+3A_hyperpar">hyperpar</code></td>
<td>
<p>a list containing prior parameter values</p>
</td></tr>
<tr><td><code id="UpdateRPlee11_+3A_ini">ini</code></td>
<td>
<p>a list containing prior parameters' initial values</p>
</td></tr>
<tr><td><code id="UpdateRPlee11_+3A_s">S</code></td>
<td>
<p>the number of subgroups</p>
</td></tr>
<tr><td><code id="UpdateRPlee11_+3A_method">method</code></td>
<td>
<p>a method option from
<code>c("Pooled", "CoxBVSSL", "Sub-struct")</code></p>
</td></tr>
<tr><td><code id="UpdateRPlee11_+3A_mrf_g">MRF_G</code></td>
<td>
<p>logical value. <code>MRF_G = TRUE</code> is to fix the MRF graph which
is provided in the argument <code>hyperpar</code>, and <code>MRF_G = FALSE</code> is to
use graphical model for leanring the MRF graph</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object with component 'beta.ini' for the updated coefficients
and component 'acceptlee' for the MCMC acceptance rate
</p>

<hr>
<h2 id='VS'>Function to perform variable selection</h2><span id='topic+VS'></span>

<h3>Description</h3>

<p>Perform variable selection using the 95
neighborhood criterion (SNC), median probability model (MPM) or Bayesian
false discovery rate (FDR). Note that the Bayesian FDR only applies for each
subgroup if there are subgroups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VS(x, method = "FDR", threshold = NA, subgroup = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="VS_+3A_x">x</code></td>
<td>
<p>fitted object obtained with <code>BayesSurvive</code></p>
</td></tr>
<tr><td><code id="VS_+3A_method">method</code></td>
<td>
<p>variable selection method to choose from
<code>c("CI", "SNC", "MPM", "FDR")</code>. Default is &quot;FDR&quot;</p>
</td></tr>
<tr><td><code id="VS_+3A_threshold">threshold</code></td>
<td>
<p>SNC threshold value (default 0.5) or the Bayesian expected false
discovery rate threshold (default 0.05)</p>
</td></tr>
<tr><td><code id="VS_+3A_subgroup">subgroup</code></td>
<td>
<p>index of the subgroup for visualizing posterior coefficients</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A boolean vector of selected (= TRUE) and rejected (= FALSE)
variables
</p>


<h3>References</h3>

<p>Lee KH, Chakraborty S, Sun J (2015). Survival prediction and variable
selection with simultaneous shrinkage and grouping priors. Statistical
Analysis and Data Mining, 8:114-127
</p>
<p>Newton MA, Noueiry A, Sarkar D, Ahlquist P (2004). Detecting differential
gene expression with a semiparametric hierarchical mixture method.
Biostatistics, 5(2), 155-76
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library("BayesSurvive")
set.seed(123)

# Load the example dataset
data("simData", package = "BayesSurvive")

dataset &lt;- list(
  "X" = simData[[1]]$X,
  "t" = simData[[1]]$time,
  "di" = simData[[1]]$status
)

# Initial value: null model without covariates
initial &lt;- list("gamma.ini" = rep(0, ncol(dataset$X)))
# Hyperparameters
hyperparPooled &lt;- list(
  "c0"     = 2, # prior of baseline hazard
  "tau"    = 0.0375, # sd for coefficient prior
  "cb"     = 20, # sd for coefficient prior
  "pi.ga"  = 0.02, # prior variable selection probability for standard Cox models
  "a"      = -4, # hyperparameter in MRF prior
  "b"      = 0.1, # hyperparameter in MRF prior
  "G"      = simData$G # hyperparameter in MRF prior
)


# run Bayesian Cox with graph-structured priors
fit &lt;- BayesSurvive(
  survObj = dataset, hyperpar = hyperparPooled,
  initial = initial, nIter = 100
)
# show variable selection
VS(fit, method = "FDR")


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
