<!DOCTYPE html><html lang="en"><head><title>Help for package BayesNetBP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BayesNetBP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AbsorbEvidence'><p>Absorb evidence into the model</p></a></li>
<li><a href='#bn_to_graphNEL'><p>Convert a bn object to graphNEL object</p></a></li>
<li><a href='#chest'><p>A simulated data from the Chest Clinic example</p></a></li>
<li><a href='#ClusterTree-class'><p>An S4 class of the cluster tree.</p></a></li>
<li><a href='#ClusterTreeCompile'><p>Compile the cluster tree</p></a></li>
<li><a href='#ComputeKLDs'><p>Compute signed and symmetric Kullback-Leibler divergence</p></a></li>
<li><a href='#ElimTreeInitialize'><p>Initialize the elimination tree</p></a></li>
<li><a href='#emission'><p>A ClusterTree Example of Emission Model</p></a></li>
<li><a href='#emission1000'><p>A simulated data from the Emission example</p></a></li>
<li><a href='#FactorQuery'><p>Queries of discrete variable distributions</p></a></li>
<li><a href='#GetValue'><p>Possible values of a discrete variable</p></a></li>
<li><a href='#Initializer'><p>Initialize a ClusterTree object</p></a></li>
<li><a href='#liver'><p>Mus Musculus HDL QTL data from Leduc et. al. (2012)</p></a></li>
<li><a href='#LocalModelCompile'><p>Model compilation</p></a></li>
<li><a href='#Marginals'><p>Obtain marginal distributions</p></a></li>
<li><a href='#PlotCGBN'><p>Plot the Bayesian network</p></a></li>
<li><a href='#PlotMarginals'><p>Plot the marginal distributions</p></a></li>
<li><a href='#PlotTree'><p>Plot the cluster tree</p></a></li>
<li><a href='#Propagate'><p>Propagate the cluster tree</p></a></li>
<li><a href='#qtlnet_to_graphNEL'><p>Convert qtlnet to graphNEL object</p></a></li>
<li><a href='#runBayesNetApp'><p>Launch the BayesNetBP Shiny App</p></a></li>
<li><a href='#Sampler'><p>Sampling from the Bayesian network</p></a></li>
<li><a href='#SummaryMarginals'><p>Summary a continuous marginal distribution</p></a></li>
<li><a href='#toytree'><p>A ClusterTree Example of Liver Model</p></a></li>
<li><a href='#yeast'><p>Saccharomyces Cerevisiae eQTL data from Kruglak et. al. (2005)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Bayesian Network Belief Propagation</td>
</tr>
<tr>
<td>Version:</td>
<td>1.6.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-05-08</td>
</tr>
<tr>
<td>Author:</td>
<td>Han Yu, Rachael Blair, Janhavi Moharil, Andrew Yan</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Han Yu &lt;hyu9@buffalo.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0), stats, utils</td>
</tr>
<tr>
<td>Imports:</td>
<td>igraph, RColorBrewer, fields, doBy, methods, graph, bnlearn,
graphics</td>
</tr>
<tr>
<td>Suggests:</td>
<td>Rgraphviz, shiny, googleVis, cyjShiny, qtl, qtlnet</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/hyu-ub/BayesNetBP/issues">https://github.com/hyu-ub/BayesNetBP/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Description:</td>
<td>Belief propagation methods in Bayesian Networks to propagate evidence through the network. The implementation of these methods are based on the article: Cowell, RG (2005). Local Propagation in Conditional Gaussian Bayesian Networks <a href="https://www.jmlr.org/papers/v6/cowell05a.html">https://www.jmlr.org/papers/v6/cowell05a.html</a>. For details please see Yu et. al. (2020) BayesNetBP: An R Package for Probabilistic Reasoning in Bayesian Networks &lt;<a href="https://doi.org/10.18637%2Fjss.v094.i03">doi:10.18637/jss.v094.i03</a>&gt;. The optional 'cyjShiny' package for running the Shiny app is available at <a href="https://github.com/cytoscape/cyjShiny">https://github.com/cytoscape/cyjShiny</a>. Please see the example in the documentation of 'runBayesNetApp' function for installing 'cyjShiny' package from GitHub. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-08 16:38:42 UTC; yuhan</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-08 17:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='AbsorbEvidence'>Absorb evidence into the model</h2><span id='topic+AbsorbEvidence'></span>

<h3>Description</h3>

<p>Absorb evidence into the model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AbsorbEvidence(tree, vars, values)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AbsorbEvidence_+3A_tree">tree</code></td>
<td>
<p>a <code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object</p>
</td></tr>
<tr><td><code id="AbsorbEvidence_+3A_vars">vars</code></td>
<td>
<p>a <code>vector</code> of the names of observed variables</p>
</td></tr>
<tr><td><code id="AbsorbEvidence_+3A_values">values</code></td>
<td>
<p>a <code>list</code> of observed values of the variables. Aside from a single value,
The element of the list can also be a vector of likelihood values</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Absorb multiple types and pieces of evidences into a <code><a href="#topic+ClusterTree-class">ClusterTree</a></code>
object. The discrete compartment of the <code><a href="#topic+ClusterTree-class">ClusterTree</a></code> will be automatically
propagated after evidence absorption, so that the object will be ready for making
queries and absorbing additional evidence.
</p>


<h3>Value</h3>

<p><code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object with the evidence absorbed
</p>


<h3>Author(s)</h3>

<p>Han Yu
</p>


<h3>References</h3>

<p>Cowell, R. G. (2005). Local propagation in conditional Gaussian Bayesian networks.
Journal of Machine Learning Research, 6(Sep), 1517-1550. <br />
<br />
Lauritzen, S. L., &amp; Spiegelhalter, D. J. (1988). Local computations with probabilities on
graphical structures and their application to expert systems. Journal of the Royal Statistical
Society. Series B (Methodological), 157-224. <br />
<br />
Yu H, Moharil J, Blair RH (2020). BayesNetBP: An R Package for Probabilistic Reasoning in Bayesian
Networks. Journal of Statistical Software, 94(3), 1-31. &lt;doi:10.18637/jss.v094.i03&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(liver)
tree.init.p &lt;- Initializer(dag=liver$dag, data=liver$data,
                           node.class=liver$node.class,
                           propagate = TRUE)
tree.post &lt;- AbsorbEvidence(tree.init.p, c("Nr1i3", "chr1_42.65"), list(1,"1"))

</code></pre>

<hr>
<h2 id='bn_to_graphNEL'>Convert a bn object to graphNEL object</h2><span id='topic+bn_to_graphNEL'></span>

<h3>Description</h3>

<p>Convert a bn object to graphNEL object while removing isolated nodes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bn_to_graphNEL(graph_bn)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bn_to_graphNEL_+3A_graph_bn">graph_bn</code></td>
<td>
<p>a <code>bn</code> object of Bayesian network</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>graphNEL</code> object
</p>


<h3>Author(s)</h3>

<p>Han Yu
</p>

<hr>
<h2 id='chest'>A simulated data from the Chest Clinic example</h2><span id='topic+chest'></span>

<h3>Description</h3>

<p>Simulated data from the Chest Clinic example (also known as the Asia example) 
from Lauritzen and Spiegelhalter, 1988.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(chest)
</code></pre>


<h3>Format</h3>

<p>The data set <code>chest</code> contains two objects: 
</p>

<dl>
<dt><code>data</code></dt><dd><p>a <code>data.frame</code> object of 10000 observations and 8 discrete variables.</p>
</dd>
<dt><code>dag</code></dt><dd><p>a <code>graphNEL</code> object specifying the network structure.</p>
</dd>
</dl>



<h3>References</h3>

<p>Lauritzen and Spiegelhalter (1988) Local Computations with Probabilities on Graphical Structures and 
their Application to Expert Systems (with Discussion). J. Roy. Stat. Soc. 50, p. 157-224. <br />
<br />
Dethlefsen, C., &amp; Hojsgaard, S. (2005). A common platform for graphical models in R: The gRbase package. 
Journal of Statistical Software, 14(17), 1-12.
</p>

<hr>
<h2 id='ClusterTree-class'>An S4 class of the cluster tree.</h2><span id='topic+ClusterTree-class'></span>

<h3>Description</h3>

<p>The <code>ClusterTree</code> object is the computational object for belief propagation.
</p>


<h3>Slots</h3>


<dl>
<dt><code>cluster</code></dt><dd><p>A <code>vector</code> storing the name of clusters in the cluster tree.</p>
</dd>
<dt><code>node</code></dt><dd><p>A <code>vector</code> storing the name of nodes in the Bayesian network.</p>
</dd>
<dt><code>graph</code></dt><dd><p>A <code>list</code> of two graphNEL objects: <code>$dag</code> stores the graph of Bayesian network,
<code>$tree</code> stores the graph of the cluster tree.</p>
</dd>
<dt><code>member</code></dt><dd><p>A named <code>list</code> of the node cluster membership.</p>
</dd>
<dt><code>parent</code></dt><dd><p>A named <code>vector</code> indicating the parent node of a given cluster in the cluster tree.</p>
</dd>
<dt><code>cluster.class</code></dt><dd><p>A named <code>vector</code> of logical values indicating whether a cluster is continuous or discrete.</p>
</dd>
<dt><code>node.class</code></dt><dd><p>A named <code>vector</code> of logical values indicating whether a node is continuous or discrete.</p>
</dd>
<dt><code>assignment</code></dt><dd><p>A named <code>list</code> indicating the assignment of discrete nodes discrete clusters.</p>
</dd>
<dt><code>propagated</code></dt><dd><p>A <code>logical</code> value indicating whether the discrete compartment has been propagated.</p>
</dd>
<dt><code>cpt</code></dt><dd><p>A named <code>list</code> of the conditional probability tables.</p>
</dd>
<dt><code>jpt</code></dt><dd><p>A named <code>list</code> of the joint distribution tables.</p>
</dd>
<dt><code>lppotential</code></dt><dd><p>A named <code>list</code> of the linear predictor potentials assigned to each cluster in the lppotential slots.</p>
</dd>
<dt><code>postbag</code></dt><dd><p>A named <code>list</code> of the linear predictor potentials assigned to each cluster in the postbag slots.</p>
</dd>
<dt><code>activeflag</code></dt><dd><p>A named <code>vector</code> of logical values indicating whether a continuous cluster is active.</p>
</dd>
<dt><code>absorbed.variables</code></dt><dd><p>A <code>vector</code> of characters indicating variables observed with hard evidence.</p>
</dd>
<dt><code>absorbed.values</code></dt><dd><p>A <code>list</code> indicating the values of the variables observed with hard evidence.</p>
</dd>
<dt><code>absorbed.soft.variables</code></dt><dd><p>A <code>vector</code> of characters indicating variables observed with soft or likelihood evidence.</p>
</dd>
<dt><code>absorbed.soft.values</code></dt><dd><p>A <code>list</code> of the likelihoods of the soft or likelihood evidence.</p>
</dd>
</dl>

<hr>
<h2 id='ClusterTreeCompile'>Compile the cluster tree</h2><span id='topic+ClusterTreeCompile'></span>

<h3>Description</h3>

<p>Get the cluster sets and strong semi-elimination tree from the Bayesian network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ClusterTreeCompile(dag, node.class)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ClusterTreeCompile_+3A_dag">dag</code></td>
<td>
<p>a <code>graphNEL</code> object of the Bayesian network</p>
</td></tr>
<tr><td><code id="ClusterTreeCompile_+3A_node.class">node.class</code></td>
<td>
<p>a named <code>vector</code> of <code>logical</code> values, <code>TRUE</code> if node
is discrete, <code>FASLE</code> if otherwise</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function forms the cluster sets and the semi-elimination tree graph
from the Bayesian network. The procedures include acquiring the elimination order,
moralization, triangulation, obtaining cluster sets, forming strong elimination
tree and strong semi-elimination tree. The cluster sets and the semi-elimination
tree are required to initialize the cluster tree.
</p>


<h3>Value</h3>


<dl>
<dt><code>tree.graph</code></dt><dd><p>a <code>graphNEL</code> object of semi-elimination tree.</p>
</dd>
<dt><code>dag</code></dt><dd><p>a <code>graphNEL</code> object of original Bayesian network.</p>
</dd>
<dt><code>cluster.sets</code></dt><dd><p>a <code>list</code> of members of each cluster.</p>
</dd>
<dt><code>node.class</code></dt><dd><p>a named <code>vector</code> of <code>logical</code> values, <code>TRUE</code> if node
is discrete, <code>FASLE</code> if otherwise</p>
</dd>
<dt><code>elimination.order</code></dt><dd><p>a <code>vector</code> of node names sorted by the elimination order.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Han Yu
</p>


<h3>References</h3>

<p>Cowell, R. G. (2005). Local propagation in conditional Gaussian Bayesian networks.
Journal of Machine Learning Research, 6(Sep), 1517-1550. <br />
<br />
Yu H, Moharil J, Blair RH (2020). BayesNetBP: An R Package for Probabilistic Reasoning in Bayesian
Networks. Journal of Statistical Software, 94(3), 1-31. &lt;doi:10.18637/jss.v094.i03&gt;.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ElimTreeInitialize">ElimTreeInitialize</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(liver)
cst &lt;- ClusterTreeCompile(dag=liver$dag, node.class=liver$node.class)

</code></pre>

<hr>
<h2 id='ComputeKLDs'>Compute signed and symmetric Kullback-Leibler divergence</h2><span id='topic+ComputeKLDs'></span>

<h3>Description</h3>

<p>Compute signed and symmetric Kullback-Leibler divergence of variables over a spectrum of evidence
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ComputeKLDs(
  tree,
  var0,
  vars,
  seq,
  pbar = TRUE,
  method = "gaussian",
  epsilon = 10^-6
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ComputeKLDs_+3A_tree">tree</code></td>
<td>
<p>a <code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object</p>
</td></tr>
<tr><td><code id="ComputeKLDs_+3A_var0">var0</code></td>
<td>
<p>the variable to have evidence absrobed</p>
</td></tr>
<tr><td><code id="ComputeKLDs_+3A_vars">vars</code></td>
<td>
<p>the variables to have divergence computed</p>
</td></tr>
<tr><td><code id="ComputeKLDs_+3A_seq">seq</code></td>
<td>
<p>a <code>vector</code> of numeric values as the evidences</p>
</td></tr>
<tr><td><code id="ComputeKLDs_+3A_pbar">pbar</code></td>
<td>
<p><code>logical(1)</code> whether to show progress bar</p>
</td></tr>
<tr><td><code id="ComputeKLDs_+3A_method">method</code></td>
<td>
<p>method for divergence computation:
<code>gaussian</code> for Gaussian approximation,  for Monte Carlo integration</p>
</td></tr>
<tr><td><code id="ComputeKLDs_+3A_epsilon">epsilon</code></td>
<td>
<p><code>numeric(1)</code> the KL divergence is undefined if certain states of a discrete variable 
have probabilities of 0. In this case, a small positive number epsilon is assigned as their probabilities for
calculating the divergence. The probabilities of other states are shrunked proportionally to ensure they sum up to 1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Compute signed and symmetric Kullback-Leibler divergence of variables over a spectrum of evidence.
The signed and symmetric Kullback-Leibler divergence is also known as Jeffery's signed information (JSI) for
continuous variables.
</p>


<h3>Value</h3>

<p>a <code>data.frame</code> of the divergence
</p>


<h3>Author(s)</h3>

<p>Han Yu
</p>


<h3>References</h3>

<p>Cowell, R. G. (2005). Local propagation in conditional Gaussian Bayesian networks.
Journal of Machine Learning Research, 6(Sep), 1517-1550. <br />
<br />
Yu H, Moharil J, Blair RH (2020). BayesNetBP: An R Package for Probabilistic Reasoning in Bayesian
Networks. Journal of Statistical Software, 94(3), 1-31. &lt;doi:10.18637/jss.v094.i03&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(liver)
tree.init.p &lt;- Initializer(dag=liver$dag, data=liver$data,
                           node.class=liver$node.class,
                           propagate = TRUE)
klds &lt;- ComputeKLDs(tree=tree.init.p, var0="Nr1i3",
                    vars=setdiff(tree.init.p@node, "Nr1i3"),
                    seq=seq(-3,3,0.5))
head(klds)

## End(Not run)
</code></pre>

<hr>
<h2 id='ElimTreeInitialize'>Initialize the elimination tree</h2><span id='topic+ElimTreeInitialize'></span>

<h3>Description</h3>

<p>Initialize the elimination tree with the local models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ElimTreeInitialize(tree, dag, model, node.sets, node.class)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ElimTreeInitialize_+3A_tree">tree</code></td>
<td>
<p>a <code>graphNEL</code> object of the elimination tree</p>
</td></tr>
<tr><td><code id="ElimTreeInitialize_+3A_dag">dag</code></td>
<td>
<p>a <code>graphNEL</code> object of the Bayesian network</p>
</td></tr>
<tr><td><code id="ElimTreeInitialize_+3A_model">model</code></td>
<td>
<p>a <code>list</code> of local models built from <code><a href="#topic+LocalModelCompile">LocalModelCompile</a></code> function</p>
</td></tr>
<tr><td><code id="ElimTreeInitialize_+3A_node.sets">node.sets</code></td>
<td>
<p>a <code>list</code> of cluster sets obtained from <code><a href="#topic+ClusterTreeCompile">ClusterTreeCompile</a></code> function</p>
</td></tr>
<tr><td><code id="ElimTreeInitialize_+3A_node.class">node.class</code></td>
<td>
<p>a named <code>vector</code> of <code>logical</code> values, <code>TRUE</code> if node
is discrete, <code>FASLE</code> if otherwise</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Initialize the elimination tree with the local models
</p>


<h3>Value</h3>

<p><code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object with the local models incorporated
</p>


<h3>Author(s)</h3>

<p>Han Yu
</p>


<h3>References</h3>

<p>Cowell, R. G. (2005). Local propagation in conditional Gaussian Bayesian networks.
Journal of Machine Learning Research, 6(Sep), 1517-1550. <br />
<br />
Yu H, Moharil J, Blair RH (2020). BayesNetBP: An R Package for Probabilistic Reasoning in Bayesian
Networks. Journal of Statistical Software, 94(3), 1-31. &lt;doi:10.18637/jss.v094.i03&gt;.
</p>


<h3>See Also</h3>

<p>The functions <code><a href="#topic+ClusterTreeCompile">ClusterTreeCompile</a></code> and <code><a href="#topic+LocalModelCompile">LocalModelCompile</a></code> provide necessary
objects to obtain <code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object by initializing the elimination tree through this function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(liver)
cst &lt;- ClusterTreeCompile(dag=liver$dag, node.class=liver$node.class)
models &lt;- LocalModelCompile(data=liver$data, dag=liver$dag, node.class=liver$node.class)
tree.init &lt;- ElimTreeInitialize(tree=cst$tree.graph,
                                dag=cst$dag,
                                model=models,
                                node.sets=cst$cluster.sets,
                                node.class=cst$node.class)

</code></pre>

<hr>
<h2 id='emission'>A ClusterTree Example of Emission Model</h2><span id='topic+emission'></span>

<h3>Description</h3>

<p>A propagated <code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object named <code>emission</code>. This model contains nine variables, indlucing
three discrete: Filter State (Fs), Waste Type (W), Burning Regimen (B) and six continuous variables: Metals in Waste (Min), 
Metals Emission (Mout), Filter Efficiency (E), Dust Emission (D), CO2 Concentration in Emission (C), Light Penetrability (L).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(emission)
</code></pre>


<h3>Format</h3>

<p>The data set contains a propagated <code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object <code>emission</code> 
ready for evidence absorption and making queries.
</p>


<h3>References</h3>

<p>Lauritzen, Steffen L., and Frank Jensen. Stable local computation with conditional Gaussian distributions. 
Statistics and Computing 11.2 (2001): 191-203.
</p>

<hr>
<h2 id='emission1000'>A simulated data from the Emission example</h2><span id='topic+emission1000'></span>

<h3>Description</h3>

<p>Simulated data from the Emission example (also known as the Waste Incinerator example)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(emission1000)
</code></pre>


<h3>Format</h3>

<p>The data set <code>emission1000</code> contains two objects: 
</p>

<dl>
<dt><code>data</code></dt><dd><p>a <code>data.frame</code> object of 1000 observations and 3 discrete variables and 6 continuous variables.</p>
</dd>
<dt><code>dag</code></dt><dd><p>a <code>graphNEL</code> object specifying the network structure.</p>
</dd>
</dl>



<h3>References</h3>

<p>Lauritzen, S. L., &amp; Jensen, F. (2001). 
Stable local computation with conditional Gaussian distributions. Statistics and Computing, 11(2), 191-203.
</p>

<hr>
<h2 id='FactorQuery'>Queries of discrete variable distributions</h2><span id='topic+FactorQuery'></span>

<h3>Description</h3>

<p>Obtain the joint, marginal, and conditional distributions of discrete variables
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FactorQuery(tree, vars = c(), mode = c("joint", "conditional", "list"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FactorQuery_+3A_tree">tree</code></td>
<td>
<p>a <code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object</p>
</td></tr>
<tr><td><code id="FactorQuery_+3A_vars">vars</code></td>
<td>
<p>the variables to be queried</p>
</td></tr>
<tr><td><code id="FactorQuery_+3A_mode">mode</code></td>
<td>
<p>type of desired distribution</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Query the joint distribution of any combination of discrete variables when
mode is &quot;joint&quot;, or conditional distribution of a discrete variable. The mode &quot;list&quot;
return a <code>list</code> of variable combinations, such that joint distributions of any subset
of them are ready for extraction. Queries outside this list are also supported but may
take longer computing time. This function will also return marginal distribution if only
one variable is queried.
</p>


<h3>Value</h3>

<p><code>data.frame</code> object specifying a joint or conditional distribution.
</p>


<h3>Author(s)</h3>

<p>Han Yu
</p>


<h3>References</h3>

<p>Cowell, R. G. (2005). Local propagation in conditional Gaussian Bayesian networks.
Journal of Machine Learning Research, 6(Sep), 1517-1550. <br />
<br />
Yu H, Moharil J, Blair RH (2020). BayesNetBP: An R Package for Probabilistic Reasoning in Bayesian
Networks. Journal of Statistical Software, 94(3), 1-31. &lt;doi:10.18637/jss.v094.i03&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(chest)
dag &lt;- chest$dag
node.class &lt;- rep(TRUE, length(dag@nodes))
names(node.class) &lt;- dag@nodes
tree.init.p &lt;- Initializer(dag=dag, data=chest$data,
                           node.class=node.class,
                           propagate=TRUE)
# joint distribution
FactorQuery(tree=tree.init.p, vars=c("tub", "xray", "dysp", "asia"), mode="joint")

# conditional distribution
FactorQuery(tree=tree.init.p, vars=c("xray"), mode="conditional")

</code></pre>

<hr>
<h2 id='GetValue'>Possible values of a discrete variable</h2><span id='topic+GetValue'></span>

<h3>Description</h3>

<p>Obtain all the possible values of a discrete variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetValue(tree, var, message = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetValue_+3A_tree">tree</code></td>
<td>
<p>a <code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object</p>
</td></tr>
<tr><td><code id="GetValue_+3A_var">var</code></td>
<td>
<p>the variables to be queried</p>
</td></tr>
<tr><td><code id="GetValue_+3A_message">message</code></td>
<td>
<p>type of desired distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>vector</code> of the possible values of discrete variable. If the variable is continuous, 
the returned value will be <code>NULL</code>.
</p>


<h3>Author(s)</h3>

<p>Han Yu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(toytree)
GetValue(toytree, "HDL")

</code></pre>

<hr>
<h2 id='Initializer'>Initialize a ClusterTree object</h2><span id='topic+Initializer'></span>

<h3>Description</h3>

<p>Initialize a ClusterTree object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Initializer(dag, data, node.class, propagate = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Initializer_+3A_dag">dag</code></td>
<td>
<p>a <code>graphNEL</code> object of the Bayesian network</p>
</td></tr>
<tr><td><code id="Initializer_+3A_data">data</code></td>
<td>
<p>a <code>data.frame</code> object</p>
</td></tr>
<tr><td><code id="Initializer_+3A_node.class">node.class</code></td>
<td>
<p>a named <code>vector</code> of <code>logical</code> values, <code>TRUE</code> if node
is discrete, <code>FASLE</code> if otherwise</p>
</td></tr>
<tr><td><code id="Initializer_+3A_propagate">propagate</code></td>
<td>
<p><code>logical</code> <code>TRUE</code> if the discrete part of the <code><a href="#topic+ClusterTree-class">ClusterTree</a></code>
to be propagated</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A wrapper function to initialize a <code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object. It combines
the functions of <code><a href="#topic+ClusterTreeCompile">ClusterTreeCompile</a></code>, <code><a href="#topic+LocalModelCompile">LocalModelCompile</a></code>,
<code><a href="#topic+ElimTreeInitialize">ElimTreeInitialize</a></code> and <code><a href="#topic+Propagate">Propagate</a></code>, thus initialize the <code><a href="#topic+ClusterTree-class">ClusterTree</a></code>
object in a single step.
</p>


<h3>Value</h3>

<p><code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object
</p>


<h3>Author(s)</h3>

<p>Han Yu
</p>


<h3>References</h3>

<p>Cowell, R. G. (2005). Local propagation in conditional Gaussian Bayesian networks.
Journal of Machine Learning Research, 6(Sep), 1517-1550. <br />
<br />
Yu H, Moharil J, Blair RH (2020). BayesNetBP: An R Package for Probabilistic Reasoning in Bayesian
Networks. Journal of Statistical Software, 94(3), 1-31. &lt;doi:10.18637/jss.v094.i03&gt;.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ClusterTreeCompile">ClusterTreeCompile</a></code>, <code><a href="#topic+LocalModelCompile">LocalModelCompile</a></code>, <code><a href="#topic+ElimTreeInitialize">ElimTreeInitialize</a></code>,
<code><a href="#topic+Propagate">Propagate</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(liver)
tree.init.p &lt;- Initializer(dag=liver$dag, data=liver$data,
                           node.class=liver$node.class,
                           propagate = TRUE)
</code></pre>

<hr>
<h2 id='liver'>Mus Musculus HDL QTL data from Leduc et. al. (2012)</h2><span id='topic+liver'></span>

<h3>Description</h3>

<p>Liver QTL data was obtained from a F2 inner-cross between inbred MRL/MpJ and SM/J 
strains of mice.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(liver)
</code></pre>


<h3>Format</h3>

<p>The data set <code>liver</code> contains three objects: the data, a learned Bayesian network structure 
and <code>vector</code> specifying node type. The fields are described as follows:  
</p>

<dl>
<dt><code>data</code></dt><dd><p>a <code>data.frame</code> object that contains 280 samples (rows) and 15 variables: genotype data 
(genotype states at 5 SNP markers) and phenotype data (HDL levels and normalized expression values of 10 genes).  
Three of these phenotypes are dichotomized, including Cyp2b10, Spgl1 and HDL.  Genotypes and dichotomized phenotypes 
are of class <code>factor</code> and continuous phenotypes are of class <code>numeric</code>.</p>
</dd>
<dt><code>dag</code></dt><dd><p>a <code>graphNEL</code> object, which is the network structure learned by <code>qtlnet</code> package.</p>
</dd>
<dt><code>node.class</code></dt><dd><p>a named <code>vector</code> of <code>logical</code> values indicating whether each node is discrete.</p>
</dd>
</dl>



<h3>References</h3>

<p>Leduc MS, Blair RH, Verdugo RA, Tsaih SW, Walsh K, Churchill GA, Paigen B.(2012). 
&quot;Using bioinformatics and systems genetics to dissect HDL-cholesterol genetics in an MRL/MpJ 
x SM/J intercross.&quot; J Lipid Res., 6, 1163-75.
</p>

<hr>
<h2 id='LocalModelCompile'>Model compilation</h2><span id='topic+LocalModelCompile'></span>

<h3>Description</h3>

<p>Compile the local models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LocalModelCompile(data, dag = NULL, node.class = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="LocalModelCompile_+3A_data">data</code></td>
<td>
<p>a <code>data.frame</code> object or a <code>qtlnet</code> object</p>
</td></tr>
<tr><td><code id="LocalModelCompile_+3A_dag">dag</code></td>
<td>
<p><code>NULL</code> if data is <code>qtlnet</code> object, or a <code>graphNEL</code> object of conditional
Gaussian Bayesian network if data is <code>data.frame</code>.</p>
</td></tr>
<tr><td><code id="LocalModelCompile_+3A_node.class">node.class</code></td>
<td>
<p><code>NULL</code> if data is <code>qtlnet</code> object, or a <code>vector</code> of logical values
named by node names, <code>TRUE</code> for discrete, <code>FALSE</code> for continuous variables if data
is <code>data.frame</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function compiles the local models, including the conditional
probability tables for discrete variables, and linear predictor potentials
for continuous variables. The qtlnet and qtl package need to be installed if data is
a <code>qtlnet</code> object.
</p>


<h3>Value</h3>


<dl>
<dt><code>pots</code></dt><dd><p>a <code>list</code> of discrete potentials (conditional probability tables)
for each discrete variable. </p>
</dd>
<dt><code>bags</code></dt><dd><p>a <code>list</code> of sets of continuous potentials (lppotentials), each set for a
continuous variables.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Han Yu
</p>


<h3>References</h3>

<p>Cowell, R. G. (2005). Local propagation in conditional Gaussian Bayesian networks.
Journal of Machine Learning Research, 6(Sep), 1517-1550. <br />
<br />
Yu H, Moharil J, Blair RH (2020). BayesNetBP: An R Package for Probabilistic Reasoning in Bayesian
Networks. Journal of Statistical Software, 94(3), 1-31. &lt;doi:10.18637/jss.v094.i03&gt;.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ElimTreeInitialize">ElimTreeInitialize</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(liver)
models &lt;- LocalModelCompile(data=liver$data, dag=liver$dag, node.class=liver$node.class)

</code></pre>

<hr>
<h2 id='Marginals'>Obtain marginal distributions</h2><span id='topic+Marginals'></span>

<h3>Description</h3>

<p>Get the marginal distributions of multiple variables
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Marginals(tree, vars)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Marginals_+3A_tree">tree</code></td>
<td>
<p>a <code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object</p>
</td></tr>
<tr><td><code id="Marginals_+3A_vars">vars</code></td>
<td>
<p>a <code>vector</code> of variables for query of marginal distributions</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Get the marginal distributions of multiple variables. The function <code>Marginals</code>
returns a <code>list</code> of marginal distributions. The marginal distribution of a discrete variable
is a named vector of probabilities. Meanwhile, the marginal distributions of
continous variables in a CG-BN model are mixtures of Gaussian distributions.
To fully represent this information, the marginal of a continuous variable is represented by
a <code>data.frame</code> with three columns to specify
parameters for each Gaussian distribution in the mixture, which are
</p>

<dl>
<dt><code>mean</code></dt><dd><p>the mean value of a Gaussian distribution.</p>
</dd>
<dt><code>sd</code></dt><dd><p>the standard deviation of a Gaussian distribution.</p>
</dd>
<dt><code>n</code></dt><dd><p>the number of Gaussian mixtures</p>
</dd>
</dl>



<h3>Value</h3>


<dl>
<dt><code>marginals</code></dt><dd><p>a <code>list</code> of marginal distributions</p>
</dd>
<dt><code>types</code></dt><dd><p>a named <code>vector</code> indicating the types of the variables whose
marginals are queried: <code>TRUE</code> for discrete, <code>FALSE</code> for continuous.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Han Yu
</p>


<h3>References</h3>

<p>Cowell, R. G. (2005). Local propagation in conditional Gaussian Bayesian networks.
Journal of Machine Learning Research, 6(Sep), 1517-1550. <br />
<br />
Yu H, Moharil J, Blair RH (2020). BayesNetBP: An R Package for Probabilistic Reasoning in Bayesian
Networks. Journal of Statistical Software, 94(3), 1-31. &lt;doi:10.18637/jss.v094.i03&gt;.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PlotMarginals">PlotMarginals</a></code> for visualization of the marginal distributions,
<code><a href="#topic+SummaryMarginals">SummaryMarginals</a></code> for summarization of the marginal distributions of
continuous variables.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(liver)
tree.init.p &lt;- Initializer(dag=liver$dag, data=liver$data,
                           node.class=liver$node.class,
                           propagate = TRUE)
tree.post &lt;- AbsorbEvidence(tree.init.p, c("Nr1i3", "chr1_42.65"), list(1,"1"))
marg &lt;- Marginals(tree.post, c("HDL", "Ppap2a"))
marg$marginals$HDL
head(marg$marginals$Ppap2a)

</code></pre>

<hr>
<h2 id='PlotCGBN'>Plot the Bayesian network</h2><span id='topic+PlotCGBN'></span>

<h3>Description</h3>

<p>Plot and compare two Bayesian networks with different evidence(s) absorbed and propagated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotCGBN(
  tree.1,
  tree.2,
  fontsize = NULL,
  pbar = FALSE,
  plotting = TRUE,
  epsilon = 10^-6
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotCGBN_+3A_tree.1">tree.1</code></td>
<td>
<p>a <code><a href="#topic+ClusterTree-class">ClusterTree</a></code></p>
</td></tr>
<tr><td><code id="PlotCGBN_+3A_tree.2">tree.2</code></td>
<td>
<p>a <code><a href="#topic+ClusterTree-class">ClusterTree</a></code></p>
</td></tr>
<tr><td><code id="PlotCGBN_+3A_fontsize">fontsize</code></td>
<td>
<p>font size for the node labels</p>
</td></tr>
<tr><td><code id="PlotCGBN_+3A_pbar">pbar</code></td>
<td>
<p><code>logical(1)</code> whether to show progress bar</p>
</td></tr>
<tr><td><code id="PlotCGBN_+3A_plotting">plotting</code></td>
<td>
<p><code>logical(1)</code> whether to output plot</p>
</td></tr>
<tr><td><code id="PlotCGBN_+3A_epsilon">epsilon</code></td>
<td>
<p><code>numeric(1)</code> the KL divergence is undefined if certain states of a discrete variable 
have probabilities of 0. In this case, a small positive number epsilon is assigned as their probabilities for calculating
the divergence. The probabilities of other states are shrunked proportionally to ensure they sum up to 1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Network visualization of the node-specific differences between Bayesian Networks
with the same topology, but evidence that has been absorbed and propagated.  The change of
marginal distribution of each node is measured by signed and symmetric Kullback-Leibler
divergence.  The sign indicates the direction of change, with <code>tree.1</code> considered as the baseline.
The magnitude of the change is reflected by the value.  Nodes that are white are d-separated
from the evidence. This function requires <code>Rgraphviz</code> package.
</p>


<h3>Value</h3>

<p>a plot of Bayesian network
</p>
<p>a <code>vector</code> of signed symmetric Kullback-Leibler divergence
</p>


<h3>Author(s)</h3>

<p>Han Yu
</p>


<h3>References</h3>

<p>Cowell, R. G. (2005). Local propagation in conditional Gaussian Bayesian networks.
Journal of Machine Learning Research, 6(Sep), 1517-1550. <br />
<br />
Yu H, Moharil J, Blair RH (2020). BayesNetBP: An R Package for Probabilistic Reasoning in Bayesian
Networks. Journal of Statistical Software, 94(3), 1-31. &lt;doi:10.18637/jss.v094.i03&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library("Rgraphviz")
data(toytree)
tree.post &lt;- AbsorbEvidence(toytree, c("Nr1i3"), list(1))
PlotCGBN(tree.1=toytree, tree.2=tree.post)

## End(Not run)
</code></pre>

<hr>
<h2 id='PlotMarginals'>Plot the marginal distributions</h2><span id='topic+PlotMarginals'></span>

<h3>Description</h3>

<p>Plot the marginal distributions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotMarginals(marginals, groups = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotMarginals_+3A_marginals">marginals</code></td>
<td>
<p>the marginal distributions returned by <code>Marginals</code> for plotting</p>
</td></tr>
<tr><td><code id="PlotMarginals_+3A_groups">groups</code></td>
<td>
<p>names of the marginals to be shown on plots</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plot the marginal distributions. Marginals of discrete variables are plotted as
bar plots, while those of continuous variables as density plots.
</p>


<h3>Author(s)</h3>

<p>Han Yu
</p>


<h3>References</h3>

<p>Cowell, R. G. (2005). Local propagation in conditional Gaussian Bayesian networks.
Journal of Machine Learning Research, 6(Sep), 1517-1550.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Marginals">Marginals</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(toytree)
marg &lt;- Marginals(toytree, c("Neu1", "Nr1i3", "chr1_42.65", "Spgl1"))
PlotMarginals(marginals=marg, groups=NULL)

</code></pre>

<hr>
<h2 id='PlotTree'>Plot the cluster tree</h2><span id='topic+PlotTree'></span>

<h3>Description</h3>

<p>Plot the structure of a <code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotTree(tree, color = "gray90")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotTree_+3A_tree">tree</code></td>
<td>
<p>a <code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object</p>
</td></tr>
<tr><td><code id="PlotTree_+3A_color">color</code></td>
<td>
<p>nodes color</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plot the structure of <code>clustertree</code> object, with the nodes labeled by corresponding
elimination node. The circles represent continuous clusters, while the boxes represent discrete clusters.
This function requires <code>Rgraphviz</code> package.
</p>


<h3>Author(s)</h3>

<p>Han Yu
</p>


<h3>References</h3>

<p>Cowell, R. G. (2005). Local propagation in conditional Gaussian Bayesian networks.
Journal of Machine Learning Research, 6(Sep), 1517-1550. <br />
<br />
Yu H, Moharil J, Blair RH (2020). BayesNetBP: An R Package for Probabilistic Reasoning in Bayesian
Networks. Journal of Statistical Software, 94(3), 1-31. &lt;doi:10.18637/jss.v094.i03&gt;.
</p>
<p>Cowell, R. G. (2005). Local propagation in conditional Gaussian Bayesian networks.
Journal of Machine Learning Research, 6(Sep), 1517-1550.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
library("Rgraphviz")
data(toytree)
PlotTree(toytree)

## End(Not run)

</code></pre>

<hr>
<h2 id='Propagate'>Propagate the cluster tree</h2><span id='topic+Propagate'></span>

<h3>Description</h3>

<p>This function propagates the discrete compartment of a <code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Propagate(tree, targets = NA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Propagate_+3A_tree">tree</code></td>
<td>
<p>an initialized <code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object</p>
</td></tr>
<tr><td><code id="Propagate_+3A_targets">targets</code></td>
<td>
<p>the cluster involved in evidence propagation, usually set by default</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The discrete compartment must be propagted to get the joint distributions
of discrete variables in each discrete clusters. A <code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object must be propagated
before absorbing evidence and making queries.
</p>


<h3>Value</h3>

<p>a <code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object
</p>


<h3>References</h3>

<p>Cowell, R. G. (2005). Local propagation in conditional Gaussian Bayesian networks.
Journal of Machine Learning Research, 6(Sep), 1517-1550. <br />
<br />
Lauritzen, S. L., &amp; Spiegelhalter, D. J. (1988). Local computations with probabilities on
graphical structures and their application to expert systems. Journal of the Royal Statistical
Society. Series B (Methodological), 157-224. <br />
<br />
Yu H, Moharil J, Blair RH (2020). BayesNetBP: An R Package for Probabilistic Reasoning in Bayesian
Networks. Journal of Statistical Software, 94(3), 1-31. &lt;doi:10.18637/jss.v094.i03&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(liver)
tree.init &lt;- Initializer(dag=liver$dag, data=liver$data,
                         node.class=liver$node.class,
                         propagate = FALSE)
tree.init@propagated
tree.init.p &lt;- Propagate(tree.init)
tree.init.p@propagated

</code></pre>

<hr>
<h2 id='qtlnet_to_graphNEL'>Convert qtlnet to graphNEL object</h2><span id='topic+qtlnet_to_graphNEL'></span>

<h3>Description</h3>

<p>Extract network structure from qtlnet object and convert to graphNEL object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qtlnet_to_graphNEL(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qtlnet_to_graphNEL_+3A_data">data</code></td>
<td>
<p>a <code>qtlnet</code> object</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function extracts network structure from qtlnet object and convert to graphNEL object.
The example data can be downloaded from &lt;https://github.com/hyu-ub/BayesNetBP&gt;.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>graphNEL</code></td>
<td>
<p>a <code>graphNEL</code> object.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Han Yu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
load(liverqtl.rda)
qtlnet_to_graphNEL(liverqtl$qtlnet.fit)

## End(Not run)

</code></pre>

<hr>
<h2 id='runBayesNetApp'>Launch the BayesNetBP Shiny App</h2><span id='topic+runBayesNetApp'></span>

<h3>Description</h3>

<p>Launch the BayesNetBP Shiny App
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runBayesNetApp(launch.browser = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="runBayesNetApp_+3A_launch.browser">launch.browser</code></td>
<td>
<p><code>logical(1)</code> whether launch the App in browser</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>runBayesNetApp</code> lauches the <code>Shiny App</code> accompanied
with this package. The app loads the <code>toytree</code> example by default and allows users
to load customized <code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object. In order to use this feature, a
<code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object should be built, propagated and named <code>tree.init.p</code>, and
then saved as a <code>.RDATA</code> file. This file can be read in by the app. <br />
</p>
<p>The console of <code>BayesNetBP</code> Shiny App comprises three panels. The first
part controls the model loading, visualization and subnetwork selection. The <code>Fit</code> function fits
the entire graph in the window. The <code>Fit Selected</code> function fits the selected subnetwork to the window.
The user can subset the network for visualization. The <code>Expand</code> function can trace the one hop neighbor of
selected nodes in a stepwise manner.
After selecting desired node sets, the user can subset the graph by the <code>Subset</code> function.
<br />
</p>
<p>The second panel is used for absorption of fixed and hard evidences.
The users can add multiple pieces of evidence to a list and absorb them into the model simultaneously.
Marginals of other nodes can be quried as
density or bar plots by node types. If a set of evidence has been absorbed, the marginals both
before and after absorption will be returned to facilitate comparison. To query the marginals, the user can
select the node of interest in the graph, and then click <code>Marginal of Selected</code>. The <code>Shift in Marginals</code>
function computes the signed and symmetric Kullback-Liebler divergence for all applicable nodes
in the network, and colors the nodes by their divergence and change in directions. <br />
</p>
<p>The function for systematic assessment of variable marginal shifts is provided in the third panel.
It allows user to specify which node to absorb the spectrum of evidence in the select menu and click <code>Select Observed</code>, and to select whose
divergence to be calculated by selecting the node in the menu and then clicking <code>Add to Plot</code>.
Alternatively, the user can use <code>Add All</code> function to select all applicable nodes into the plotting list.
The result is visualized in an interactive plot. The <code>Min</code>, <code>Max</code> and <code>Step</code> controls the range of values
of the evidence to be absorbed.
</p>


<h3>Author(s)</h3>

<p>Han Yu
</p>


<h3>References</h3>

<p>Yu H, Moharil J, Blair RH (2020). BayesNetBP: An R Package for Probabilistic Reasoning in Bayesian
Networks. Journal of Statistical Software, 94(3), 1-31. &lt;doi:10.18637/jss.v094.i03&gt;.
</p>
<p>Yu H, Moharil J, Blair RH (2020). BayesNetBP: An R Package for Probabilistic Reasoning in Bayesian
Networks. Journal of Statistical Software, 94(3), 1-31. &lt;doi:10.18637/jss.v094.i03&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
# load or install required packages to run App
library("shiny")
library("googleVis")
library("devtools")
devtools::install_github("cytoscape/cyjShiny")
library("cyjShiny")
# run the App in browser
runBayesNetApp(launch.browser=TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='Sampler'>Sampling from the Bayesian network</h2><span id='topic+Sampler'></span>

<h3>Description</h3>

<p>Sampling from the joint distribution of all applicable nodes in the Bayesian network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Sampler(tree, n)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Sampler_+3A_tree">tree</code></td>
<td>
<p>a <code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object</p>
</td></tr>
<tr><td><code id="Sampler_+3A_n">n</code></td>
<td>
<p>a <code>integer</code> number of observations to generate</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>dataframe</code> of generated data
</p>


<h3>Author(s)</h3>

<p>Han Yu
</p>


<h3>References</h3>

<p>Cowell, R. G. (2005). Local propagation in conditional Gaussian Bayesian networks.
Journal of Machine Learning Research, 6(Sep), 1517-1550. <br />
<br />
Yu H, Moharil J, Blair RH (2020). BayesNetBP: An R Package for Probabilistic Reasoning in Bayesian
Networks. Journal of Statistical Software, 94(3), 1-31. &lt;doi:10.18637/jss.v094.i03&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(toytree)
Sampler(tree = toytree, n = 10)

</code></pre>

<hr>
<h2 id='SummaryMarginals'>Summary a continuous marginal distribution</h2><span id='topic+SummaryMarginals'></span>

<h3>Description</h3>

<p>This function summary the marginal distributions of continuous variables by outputing the
mean, standard deviation, and number of subpopulations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SummaryMarginals(marginals)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SummaryMarginals_+3A_marginals">marginals</code></td>
<td>
<p>the marginal distributions obtained from <code><a href="#topic+Marginals">Marginals</a></code> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>data.frame</code> object containing information about the marginal distributions for continuous variables.
The marginal distributions of continous variables in a CG-BN model are mixtures of Gaussian distributions.
Therefore, besides the mean and standard deviation, the object has an additional column to specify the number of Gaussian
mixtures.
</p>

<dl>
<dt><code>mean</code></dt><dd><p>the mean value of a Gaussian distribution.</p>
</dd>
<dt><code>sd</code></dt><dd><p>the standard deviation of a Gaussian distribution.</p>
</dd>
<dt><code>n</code></dt><dd><p>the number of Gaussian distributions in the mixture.</p>
</dd>
</dl>



<h3>References</h3>

<p>Cowell, R. G. (2005). Local propagation in conditional Gaussian Bayesian networks.
Journal of Machine Learning Research, 6(Sep), 1517-1550. <br />
<br />
Yu H, Moharil J, Blair RH (2020). BayesNetBP: An R Package for Probabilistic Reasoning in Bayesian
Networks. Journal of Statistical Software, 94(3), 1-31. &lt;doi:10.18637/jss.v094.i03&gt;.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Marginals">Marginals</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(liver)
tree.init.p &lt;- Initializer(dag=liver$dag, data=liver$data,
                           node.class=liver$node.class,
                           propagate = TRUE)
marg &lt;- Marginals(tree.init.p, c("HDL", "Ppap2a", "Neu1"))
SummaryMarginals(marginals=marg)

</code></pre>

<hr>
<h2 id='toytree'>A ClusterTree Example of Liver Model</h2><span id='topic+toytree'></span>

<h3>Description</h3>

<p>A propagated <code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object named <code>toytree</code>, obtained from 
<code><a href="#topic+liver">liver</a></code> QTL data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(toytree)
</code></pre>


<h3>Format</h3>

<p>The data set contains a propagated <code><a href="#topic+ClusterTree-class">ClusterTree</a></code> object <code>toytree</code>, 
which is ready for evidence absorption and making queries.
</p>

<hr>
<h2 id='yeast'>Saccharomyces Cerevisiae eQTL data from Kruglak et. al. (2005)</h2><span id='topic+yeast'></span>

<h3>Description</h3>

<p>eQTL data from 112 F1 segregants from a cross between BY4716 and 
RM11-1a strains of <em>Saccharomyces Cerevisiae</em>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(yeast)
</code></pre>


<h3>Format</h3>

<p>The data set <code>yeast</code> is a data frame of 112 observations of 50 variables: genotype 
data (genotype states at 12 SNP markers) and phenotype data (normalized and discretized 
expression values of 38 genes). Both genotypes and phenotypes are of class <code>factor</code>.
</p>


<h3>Details</h3>

<p>The <code>yeast</code> dataset is a subset of the widely studied yeast expression 
dataset comprising of 112 F1 segregants from a cross between BY4716 and RM11-1a 
strains of <em>Saccharomyces Cerevisiae</em>. The original dataset consists of 
expression values reported as log2(sample/ BY reference) for 6216 genes. 
The data can be accessed in Gene Expression Omnibus (GEO) by accession number (GSE1990). 
After linkage analysis and filtering based on location and significance of QTL, 
a final set of 38 genes and their corresponding 12 SNP markers were identified and 
included in the yeast dataset. The gene expression values are discretized around 
the median and have two states, 1 (above or equal to median) and -1 (below median). 
re are two genotype states: 1 or 2. Thus the final dataset is a data frame of 112 observations 
(genotype) of 12 variables (SNP markers) and normalized gene expression of 38 variables (genes).
</p>


<h3>References</h3>

<p>Brem RB, Kruglyak L. The landscape of genetic complexity across 5,700 gene expression traits in yeast. 
Proc Natl Acad Sci U S A 2005 Feb 1;102(5):1572-7.<br />
<br />
Brem RB, Storey JD, Whittle J, Kruglyak L. Genetic interactions between polymorphisms that affect gene 
expression in yeast. Nature 2005 Aug 4;436(7051):701-3.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
