<!DOCTYPE html><html lang="en-gb"><head><title>Help for package bayefdr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {bayefdr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bayefdr-package'><p>The 'nibbles' package.</p></a></li>
<li><a href='#cumplot'><p>Plot the cumulative median, mean, and 95% high posterior density region.</p></a></li>
<li><a href='#efdr'><p>EFDR and EFNR estimation</p></a></li>
<li><a href='#efdr_search'><p>Bayesian EFDR optimisation.</p></a></li>
<li><a href='#optimal'><p>Retrieve the index of the optimal probability threshold.</p></a></li>
<li><a href='#plot.bayefdr'><p>Plot the EFDR, EFNR grids of a bayefdr object.</p></a></li>
<li><a href='#print.bayefdr'><p>Print methods for bayefdr objects.</p></a></li>
<li><a href='#traceplot'><p>Trace, marginal density histogram, and autocorrelation plot of MCMC draws.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Bayesian Estimation and Optimisation of Expected False Discovery
Rate</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-10-26</td>
</tr>
<tr>
<td>Description:</td>
<td>
    Implements the Bayesian FDR control described by 
    Newton et al. (2004), &lt;<a href="https://doi.org/10.1093%2Fbiostatistics%2F5.2.155">doi:10.1093/biostatistics/5.2.155</a>&gt;.
    Allows optimisation and visualisation of expected error rates based on
    tail posterior probability tests.
    Based on code written by Catalina Vallejos for BASiCS, see
    Beyond comparisons of means: understanding changes in gene expression at the
    single-cell level Vallejos et al. (2016) &lt;<a href="https://doi.org/10.1186%2Fs13059-016-0930-3">doi:10.1186/s13059-016-0930-3</a>&gt;.</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, reshape2, assertthat, utils, cowplot, ggExtra, stats</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/VallejosGroup/bayefdr/issues">https://github.com/VallejosGroup/bayefdr/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/VallejosGroup/bayefdr">https://github.com/VallejosGroup/bayefdr</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, pkgdown</td>
</tr>
<tr>
<td>Language:</td>
<td>en-gb</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-10-26 19:20:25 UTC; alan</td>
</tr>
<tr>
<td>Author:</td>
<td>Alan O'Callaghan [aut, cre],
  Catalina Vallejos [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alan O'Callaghan &lt;alan.ocallaghan@outlook.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-10-26 19:35:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='bayefdr-package'>The 'nibbles' package.</h2><span id='topic+bayefdr-package'></span><span id='topic+bayefdr'></span>

<h3>Description</h3>

<p>A DESCRIPTION OF THE PACKAGE
</p>


<h3>References</h3>

<p>Detecting differential gene expression with a semiparametric hierarchical 
mixture method
Michael A. Newton, Amine Noueiry, Deepayan Sarkar, Paul Ahlquist
https://doi.org/10.1093/biostatistics/5.2.155
</p>

<hr>
<h2 id='cumplot'>Plot the cumulative median, mean, and 95% high posterior density region.</h2><span id='topic+cumplot'></span>

<h3>Description</h3>

<p>Plot the cumulative median, mean, and 95% high posterior density region.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cumplot(x, ylab = NULL, burn = 0, thin = 1, hpd_level = 0.95)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cumplot_+3A_x">x</code></td>
<td>
<p>An vector of MCMC draws.</p>
</td></tr>
<tr><td><code id="cumplot_+3A_ylab">ylab</code></td>
<td>
<p>An optional y-axis label.</p>
</td></tr>
<tr><td><code id="cumplot_+3A_burn">burn</code></td>
<td>
<p>Integer specifying the number of initial iterations to be
discarded.</p>
</td></tr>
<tr><td><code id="cumplot_+3A_thin">thin</code></td>
<td>
<p>Integer specifying the thinning factor to be used on
the MCMC steps.</p>
</td></tr>
<tr><td><code id="cumplot_+3A_hpd_level">hpd_level</code></td>
<td>
<p>Floating point specifying the desired HPD level.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot showing the cumulative mean, median and HPD.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- rnorm(1000)
cumplot(x)
</code></pre>

<hr>
<h2 id='efdr'>EFDR and EFNR estimation</h2><span id='topic+efdr'></span><span id='topic+efnr'></span>

<h3>Description</h3>

<p>Calculate the Expected False Discovery Rate (EFDR) or Expected False 
Negative Rate (EFNR) in a vector of probabilities, given a specified
evidence threshold.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>efdr(evidence_threshold, probs)

efnr(evidence_threshold, probs)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="efdr_+3A_evidence_threshold">evidence_threshold</code></td>
<td>
<p>Scalar value specifying the evidence threshold
at which the EFDR or EFNR should be evaluated.</p>
</td></tr>
<tr><td><code id="efdr_+3A_probs">probs</code></td>
<td>
<p>Vector of probabilities.</p>
</td></tr>
</table>

<hr>
<h2 id='efdr_search'>Bayesian EFDR optimisation.</h2><span id='topic+efdr_search'></span>

<h3>Description</h3>

<p>Given a vector of probabilities, this function finds the probability 
threshold that matches a target expected false discovery rate as closely
as possible.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>efdr_search(
  probs,
  target_efdr,
  min_threshold = 0.7,
  prob_thresholds = seq(0.5, 0.9995, by = 0.00025)
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="efdr_search_+3A_probs">probs</code></td>
<td>
<p>Vector of probabilities.</p>
</td></tr>
<tr><td><code id="efdr_search_+3A_target_efdr">target_efdr</code></td>
<td>
<p>Numeric scalar specifying the expected false discovery 
rate to match.</p>
</td></tr>
<tr><td><code id="efdr_search_+3A_min_threshold">min_threshold</code></td>
<td>
<p>Minimum probability threshold. If the optimal 
probability threshold is below this number, it is rejected and 
<code>min_threshold</code> is used instead.</p>
</td></tr>
<tr><td><code id="efdr_search_+3A_prob_thresholds">prob_thresholds</code></td>
<td>
<p>Vector for probability thresholds to scan,
with the aim of finding the threshold that matches the target EFDR.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class &quot;bayefdr&quot; containing the probability thresholds 
tested, the EFDR and EFNR at each probability threshold, and the optimal 
threshold.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> probs &lt;- runif(100)
 efdr &lt;- efdr_search(probs, target_efdr = 0.1)
 plot(efdr)
</code></pre>

<hr>
<h2 id='optimal'>Retrieve the index of the optimal probability threshold.</h2><span id='topic+optimal'></span>

<h3>Description</h3>

<p>Retrieve the index of the optimal probability threshold.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>optimal(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="optimal_+3A_x">x</code></td>
<td>
<p>An object of class &quot;bayefdr&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The integer index of the optimal probability threshold.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> probs &lt;- runif(100)
 e &lt;- efdr_search(probs, target_efdr = 0.1)
 optimal(e)
 e[optimal(e), ]
</code></pre>

<hr>
<h2 id='plot.bayefdr'>Plot the EFDR, EFNR grids of a bayefdr object.</h2><span id='topic+plot.bayefdr'></span>

<h3>Description</h3>

<p>Plot the EFDR, EFNR grids of a bayefdr object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'bayefdr'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.bayefdr_+3A_x">x</code></td>
<td>
<p>An object of class bayefdr.</p>
</td></tr>
<tr><td><code id="plot.bayefdr_+3A_...">...</code></td>
<td>
<p>Unused.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot.
</p>

<hr>
<h2 id='print.bayefdr'>Print methods for bayefdr objects.</h2><span id='topic+print.bayefdr'></span><span id='topic+head.bayefdr'></span>

<h3>Description</h3>

<p>Print methods for bayefdr objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'bayefdr'
print(x, ...)

## S3 method for class 'bayefdr'
head(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.bayefdr_+3A_x">x</code></td>
<td>
<p>An object of class bayefdr.</p>
</td></tr>
<tr><td><code id="print.bayefdr_+3A_...">...</code></td>
<td>
<p>Unused.</p>
</td></tr>
</table>

<hr>
<h2 id='traceplot'>Trace, marginal density histogram, and autocorrelation plot of MCMC draws.</h2><span id='topic+traceplot'></span>

<h3>Description</h3>

<p>Trace, marginal density histogram, and autocorrelation plot of MCMC draws.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>traceplot(x, ylab = NULL, log = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="traceplot_+3A_x">x</code></td>
<td>
<p>A vector of MCMC draws.</p>
</td></tr>
<tr><td><code id="traceplot_+3A_ylab">ylab</code></td>
<td>
<p>An optional y-axis label.</p>
</td></tr>
<tr><td><code id="traceplot_+3A_log">log</code></td>
<td>
<p>Logical scalar controlling whether the y-axis should be logged.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot created using <code><a href="cowplot.html#topic+plot_grid">plot_grid</a></code> showing
the trace, marginal density histogram, and autocorrelation function of the
MCMC draws in <code>x</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- rnorm(1000)
traceplot(x)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
