<!DOCTYPE html><html><head><title>Help for package Bayenet</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Bayenet}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Bayenet-package'><p>Bayesian Quantile Elastic Net for Genetic Study</p></a></li>
<li><a href='#Bayenet'><p>fit a robust Bayesian elastic net variable selection model for genetic study.</p></a></li>
<li><a href='#dat'><p>simulated data for demonstrating the features of Bayenet.</p></a></li>
<li><a href='#predict.Bayenet'><p>make predictions from a Bayenet object</p></a></li>
<li><a href='#print.Bayenet'><p>print a Bayenet object</p></a></li>
<li><a href='#print.Bayenet.pred'><p>print a predict.Bayenet object</p></a></li>
<li><a href='#print.Selection'><p>print a Selection object</p></a></li>
<li><a href='#Selection'><p>Variable selection for a Bayenet object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Bayesian Quantile Elastic Net for Genetic Study</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-04-04</td>
</tr>
<tr>
<td>Description:</td>
<td>As heavy-tailed error distribution and outliers in the response variable widely exist, models which are robust to data contamination are highly demanded. Here, we develop a novel robust Bayesian variable selection method with elastic net penalty for quantile regression in genetic analysis. In particular, the spike-and-slab priors have been incorporated to impose sparsity. An efficient Gibbs sampler has been developed to facilitate computation.The core modules of the package have been developed in 'C++' and R.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, stats, MCMCpack, base, gsl, VGAM, MASS, hbmem, SuppDists</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-05 15:21:09 UTC; xilu0</td>
</tr>
<tr>
<td>Author:</td>
<td>Xi Lu [aut, cre],
  Cen Wu [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xi Lu &lt;xilu@ksu.edu&gt;</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-05 15:43:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='Bayenet-package'>Bayesian Quantile Elastic Net for Genetic Study</h2><span id='topic+Bayenet-package'></span>

<h3>Description</h3>

<p>In this package, we provide a set of robust Bayesian quantile variable selection methods for genetic analysis. A Bayesian formulation of the quantile regression has been adopted to accommodate data contamination and heavy-tailed distributions in the response. 
The proposed method conducts a robust quantile variable selection by accounting for structural sparsity. In particular, the spike-and-slab priors are imposed to identify important genetic effects.
In addition to the default method, users can also choose different structures (robust or non-robust) and penalty (lasso or elastic net) with or without spike-and-slab priors.
</p>


<h3>Details</h3>

<p>_package
</p>
<p>The user friendly, integrated interface <strong>Bayenet()</strong> allows users to flexibly choose the fitting methods they prefer. There are three arguments in Bayenet() that control the fitting method:
robust: whether to use robust methods; sparse: whether to use the spike-and-slab priors to create sparsity; penalty: use lasso or elastic net as penalty.
The function Bayenet() returns a Bayenet object that contains the posterior estimates of each coefficients. 
predict.Bayenet() and print.Bayenet() are implemented for Bayenet objects.
predict.Bayenet() takes a Bayenet object and returns the predicted values for new observations.
</p>


<h3>References</h3>

<p>Lu, X. and Wu, C. (2023). Bayesian quantile elastic net with spike-and-slab priors.
</p>
<p>Lu, X., Fan, K., Ren, J., and Wu, C. (2021). Identifying Gene–Environment Interactions With Robust Marginal Bayesian Variable Selection.
<em>Frontiers in Genetics</em>, 12:667074 <a href="https://doi.org/10.3389/fgene.2021.667074">doi:10.3389/fgene.2021.667074</a>
</p>
<p>Zhou, F., Ren, J., Lu, X., Ma, S. and Wu, C. (2020). Gene–Environment Interaction: a Variable Selection Perspective. Epistasis. Methods in Molecular Biology.
<em>Humana Press</em> (Accepted) <a href="https://arxiv.org/abs/2003.02930">https://arxiv.org/abs/2003.02930</a>
</p>
<p>Wu, C., Cui, Y., and Ma, S. (2014). Integrative analysis of gene–environment interactions under a multi–response partially linear varying coefficient model.
<em>Statistics in Medicine</em>, 33(28), 4988–4998 <a href="https://doi.org/10.1002/sim.6287">doi:10.1002/sim.6287</a>
</p>
<p>Li, Q. and Lin, N. (2010). The Bayesian elastic net. <em>Bayesian Anal</em>, 5(1): 151-170 <a href="https://doi.org/10.1214/10-BA506">doi:10.1214/10-BA506</a>
</p>
<p>Li, Q., Xi, R. and Lin, N. (2010). The Bayesian regularized quantile regression. <em>Bayesian Analysis</em>, 5(3): 533-556
<a href="https://doi.org/10.1214/10-BA521">doi:10.1214/10-BA521</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Bayenet">Bayenet</a></code>
</p>

<hr>
<h2 id='Bayenet'>fit a robust Bayesian elastic net variable selection model for genetic study.</h2><span id='topic+Bayenet'></span>

<h3>Description</h3>

<p>fit a robust Bayesian elastic net variable selection model for genetic study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Bayenet(
  X,
  Y,
  clin,
  max.steps = 10000,
  robust = TRUE,
  sparse = TRUE,
  penalty = c("lasso", "elastic net"),
  debugging = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Bayenet_+3A_x">X</code></td>
<td>
<p>the matrix of predictors (genetic factors). Each row should be an observation vector.</p>
</td></tr>
<tr><td><code id="Bayenet_+3A_y">Y</code></td>
<td>
<p>the continuous response variable.</p>
</td></tr>
<tr><td><code id="Bayenet_+3A_clin">clin</code></td>
<td>
<p>a matrix of clinical variables. Clinical variables are not subject to penalize. Clinical variables will be centered and a column of 1 will be added to the Clinical matrix as the intercept.</p>
</td></tr>
<tr><td><code id="Bayenet_+3A_max.steps">max.steps</code></td>
<td>
<p>the number of MCMC iterations.</p>
</td></tr>
<tr><td><code id="Bayenet_+3A_robust">robust</code></td>
<td>
<p>logical flag. If TRUE, robust methods will be used.</p>
</td></tr>
<tr><td><code id="Bayenet_+3A_sparse">sparse</code></td>
<td>
<p>logical flag. If TRUE, spike-and-slab priors will be used to shrink coefficients of irrelevant covariates to zero exactly.</p>
</td></tr>
<tr><td><code id="Bayenet_+3A_penalty">penalty</code></td>
<td>
<p>two choices are available. &quot;lasso&quot; for lasso penalty. &quot;elastic net&quot; for elastic net penalty.</p>
</td></tr>
<tr><td><code id="Bayenet_+3A_debugging">debugging</code></td>
<td>
<p>logical flag. If TRUE, progress will be output to the console and extra information will be returned.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Consider the data model described in &quot;<code><a href="#topic+dat">dat</a></code>&quot;:
</p>
<p style="text-align: center;"><code class="reqn">Y_{i} = \alpha_{0} + \sum_{k=1}^{q}\gamma_{k}C_{ik}+\sum_{j=1}^{p}\beta_{j}X_{ij}+\epsilon_{i},</code>
</p>

<p>where <code class="reqn">\alpha_{0}</code> is the intercept, <code class="reqn">\gamma_{k}</code>'s and <code class="reqn">\beta_{j}</code>'s are the regression coefficients corresponding to effects of clinical factors and genetic variants, respectively.
</p>
<p>When penalty=&quot;elastic net&quot; (default), the elastic net penalty is adopted. If penalty=&quot;lasso&quot;, the lasso penalty is used.
</p>
<p>When sparse=TRUE (default), spike&ndash;and&ndash;slab priors are imposed to identify important main and interaction effects. If sparse=FALSE, Laplacian shrinkage will be used.
</p>
<p>When robust=TRUE (default), the distribution of <code class="reqn">\epsilon_{i}</code> is defined as a Laplace distribution with density
<code class="reqn">
f(\epsilon_{i}|\nu) = \frac{\nu}{2}\exp\left\{-\nu |\epsilon_{i}|\right\}
</code>, (<code class="reqn">i=1,\dots,n</code>), which leads to a Bayesian formulation of LAD regression. If robust=FALSE, <code class="reqn">\epsilon_{i}</code> follows a normal distribution.
</p>
<p>Both <code class="reqn">X</code> and <code class="reqn">clin</code> will be standardized before the generation of interaction terms to avoid the multicollinearity between main effects and interaction terms.
</p>
<p>Please check the references for more details about the prior distributions.
</p>


<h3>Value</h3>

<p>an object of class &lsquo;Bayenet&rsquo; is returned, which is a list with component:
</p>
<table>
<tr><td><code>posterior</code></td>
<td>
<p>the posterior samples of coefficients from the MCMC.</p>
</td></tr>
<tr><td><code>coefficient</code></td>
<td>
<p>the estimated value of coefficients.</p>
</td></tr>
<tr><td><code>burn.in</code></td>
<td>
<p>the total number of burn-ins.</p>
</td></tr>
<tr><td><code>iterations</code></td>
<td>
<p>the total number of iterations.</p>
</td></tr>
<tr><td><code>design</code></td>
<td>
<p>the design matrix of all effects.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Lu, X. and Wu, C. (2023). Bayesian quantile elastic net with spike-and-slab priors.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Selection">Selection</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dat)

max.steps=5000
fit= Bayenet(X, Y, clin, max.steps, penalty="lasso")

## coefficients of parameters
fit$coefficient

## Estimated values of main G effects 
fit$coefficient$G

## Estimated values of clincal effects 
fit$coefficient$clin


</code></pre>

<hr>
<h2 id='dat'>simulated data for demonstrating the features of Bayenet.</h2><span id='topic+dat'></span><span id='topic+X'></span><span id='topic+Y'></span><span id='topic+clin'></span><span id='topic+coef'></span>

<h3>Description</h3>

<p>Simulated gene expression data for demonstrating the features of Bayenet.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("dat")
</code></pre>


<h3>Format</h3>

<p>dat consists of four components: X, Y, clin, coef.
</p>


<h3>Details</h3>

<p><strong>The data model for generating Y</strong>
</p>
<p>Use subscript <code class="reqn">i</code> to denote the <code class="reqn">i</code>th subject. Let <code class="reqn">(Y_{i}, X_{i}, clin_{i})</code> (<code class="reqn">i=1,\ldots,n</code>) be
independent and identically distributed random vectors. <code class="reqn">Y_{i}</code> is a continuous response variable representing the
cancer outcome and disease phenotype. <code class="reqn">X_{i}</code> is the <code class="reqn">p</code>&ndash;dimensional vector of genetic factors. The clinical factors
is denoted as the <code class="reqn">q</code>-dimensional vector <code class="reqn">clin_{i}</code>.
The <code class="reqn">\epsilon</code> follows some heavy-tailed distribution. Considering the following model:
</p>
<p style="text-align: center;"><code class="reqn">Y_{i} = \alpha_{0} + \sum_{k=1}^{q}\gamma_{k}C_{ik}+\sum_{j=1}^{p}\beta_{j}X_{ij}+\epsilon_{i},</code>
</p>

<p>where <code class="reqn">\alpha_{0}</code> is the intercept, <code class="reqn">\gamma_{k}</code>'s and <code class="reqn">\beta_{j}</code>'s are the regression coefficients corresponding to effects of clinical factors and genetic variants, respectively.
Denote <code class="reqn">\gamma=(\gamma_{1}, \ldots, \gamma_{q})^{T}</code>, <code class="reqn">\beta=(\beta_{1}, \ldots, \beta_{p})^{T}</code>. 
Then model can be written as 
</p>
<p style="text-align: center;"><code class="reqn">Y_{i} = C_{i}\gamma + X_{i}\beta + \epsilon_{i}.</code>
</p>



<h3>See Also</h3>

<p><code><a href="#topic+Bayenet">Bayenet</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dat)
dim(X)
</code></pre>

<hr>
<h2 id='predict.Bayenet'>make predictions from a Bayenet object</h2><span id='topic+predict.Bayenet'></span>

<h3>Description</h3>

<p>make predictions from a Bayenet object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Bayenet'
predict(object, X.new, clin.new, Y.new, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.Bayenet_+3A_object">object</code></td>
<td>
<p>Bayenet object.</p>
</td></tr>
<tr><td><code id="predict.Bayenet_+3A_x.new">X.new</code></td>
<td>
<p>a matrix of new values for X at which predictions are to be made.</p>
</td></tr>
<tr><td><code id="predict.Bayenet_+3A_clin.new">clin.new</code></td>
<td>
<p>a vector or matrix of new values for clin at which predictions are to be made.</p>
</td></tr>
<tr><td><code id="predict.Bayenet_+3A_y.new">Y.new</code></td>
<td>
<p>a vector of the response of new observations. If provided, the prediction error will be computed based on Y.new.</p>
</td></tr>
<tr><td><code id="predict.Bayenet_+3A_...">...</code></td>
<td>
<p>other predict arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>X.new must have the same number of columns as X used for fitting the model. If clin was provided when fit the model, clin.new
must not be NULL, and vice versa. The predictions are made based on the posterior estimates of coefficients in the Bayenet object.
Note that the effects of clinical factors are not subject to selection.
</p>
<p>If Y.new is provided, the prediction error will be computed. For robust methods, the prediction mean absolute deviations (PMAD) will be computed.
For non-robust methods, the prediction mean squared error (PMSE) will be computed.
</p>


<h3>Value</h3>

<p>an object of class &lsquo;Bayenet.pred&rsquo; is returned, which is a list with components:
</p>
<table>
<tr><td><code>error</code></td>
<td>
<p>prediction error. error is NULL is Y.new=NULL.</p>
</td></tr>
<tr><td><code>y.pred</code></td>
<td>
<p>predicted values of the new observations.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+Bayenet">Bayenet</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dat)
test=sample((1:nrow(X)), floor(nrow(X)/5))
fit=Bayenet(X[-test,], Y[-test], clin[-test,], max.steps=500,penalty="lasso")
predict(fit, X[test,], clin[test,], Y[test,])

</code></pre>

<hr>
<h2 id='print.Bayenet'>print a Bayenet object</h2><span id='topic+print.Bayenet'></span>

<h3>Description</h3>

<p>Print a summary of a Bayenet object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Bayenet'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.Bayenet_+3A_x">x</code></td>
<td>
<p>Bayenet object.</p>
</td></tr>
<tr><td><code id="print.Bayenet_+3A_digits">digits</code></td>
<td>
<p>significant digits in printout.</p>
</td></tr>
<tr><td><code id="print.Bayenet_+3A_...">...</code></td>
<td>
<p>other print arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Bayenet">Bayenet</a></code>
</p>

<hr>
<h2 id='print.Bayenet.pred'>print a predict.Bayenet object</h2><span id='topic+print.Bayenet.pred'></span>

<h3>Description</h3>

<p>Print a summary of a predict.Bayenet object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Bayenet.pred'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.Bayenet.pred_+3A_x">x</code></td>
<td>
<p>predict.Bayenet object.</p>
</td></tr>
<tr><td><code id="print.Bayenet.pred_+3A_digits">digits</code></td>
<td>
<p>significant digits in printout.</p>
</td></tr>
<tr><td><code id="print.Bayenet.pred_+3A_...">...</code></td>
<td>
<p>other print arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+predict.Bayenet">predict.Bayenet</a></code>
</p>

<hr>
<h2 id='print.Selection'>print a Selection object</h2><span id='topic+print.Selection'></span>

<h3>Description</h3>

<p>Print a summary of a Selection object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Selection'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.Selection_+3A_x">x</code></td>
<td>
<p>Selection object.</p>
</td></tr>
<tr><td><code id="print.Selection_+3A_digits">digits</code></td>
<td>
<p>significant digits in printout.</p>
</td></tr>
<tr><td><code id="print.Selection_+3A_...">...</code></td>
<td>
<p>other print arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Selection">Selection</a></code>
</p>

<hr>
<h2 id='Selection'>Variable selection for a Bayenet object</h2><span id='topic+Selection'></span>

<h3>Description</h3>

<p>Variable selection for a Bayenet object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Selection(obj, sparse)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Selection_+3A_obj">obj</code></td>
<td>
<p>Bayenet object.</p>
</td></tr>
<tr><td><code id="Selection_+3A_sparse">sparse</code></td>
<td>
<p>logical flag. If TRUE, spike-and-slab priors will be used to shrink coefficients of irrelevant covariates to zero exactly.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For class &lsquo;Sparse&rsquo;,  the inclusion probability is used to indicate the importance of predictors. 
Here we use a binary indicator <code class="reqn">\phi</code> to denote the membership of the non-spike distribution. 
Take the main effect of the <code class="reqn">j</code>th genetic factor, <code class="reqn">X_{j}</code>, as an example. 
Suppose we have collected H posterior samples from MCMC after burn-ins. The <code class="reqn">j</code>th G factor is included 
in the final model at the <code class="reqn">j</code>th MCMC iteration if the corresponding indicator is 1, i.e., <code class="reqn">\phi_j^{(h)} = 1</code>. 
Subsequently, the posterior probability of retaining the <code class="reqn">j</code>th genetic main effect in the final model is defined as the average of all the indicators for the <code class="reqn">j</code>th G factor among the H posterior samples. 
That is, <code class="reqn">p_j = \hat{\pi} (\phi_j = 1|y) = \frac{1}{H} \sum_{h=1}^{H} \phi_j^{(h)}, \;  j = 1, \dots,p.</code>
A larger posterior inclusion probability of <code class="reqn">j</code>th indicates a stronger empirical evidence that the <code class="reqn">j</code>th genetic main effect has a non-zero coefficient, i.e., a stronger association with the phenotypic trait. 
Here, we use 0.5 as a cutting-off point. If <code class="reqn">p_j &gt; 0.5</code>, then the <code class="reqn">j</code>th genetic main effect is included in the final model. Otherwise, the <code class="reqn">j</code>th genetic main effect is excluded in the final model.
For class &lsquo;NonSparse&rsquo;, variable selection is based on 95% credible interval.
Please check the references for more details about the variable selection.
</p>


<h3>Value</h3>

<p>an object of class &lsquo;Selection&rsquo; is returned, which is a list with components:
</p>
<table>
<tr><td><code>method</code></td>
<td>
<p>method used for identifying important effects.</p>
</td></tr>
<tr><td><code>effects</code></td>
<td>
<p>a list of indicators of selected effects.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Lu, X. and Wu, C. (2023). Bayesian quantile elastic net with spike-and-slab priors.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Bayenet">Bayenet</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dat)
max.steps=5000
fit= Bayenet(X, Y, clin, max.steps, penalty="lasso")
selected=Selection(fit,sparse=TRUE)
selected$Main.G


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
