<!DOCTYPE html><html><head><title>Help for package bbknnR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {bbknnR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bbknnR-package'><p>bbknnR: Perform Batch Balanced KNN in R</p></a></li>
<li><a href='#panc8_small'><p>A small example version of the pancreas scRNA-seq dataset</p></a></li>
<li><a href='#RidgeRegression'><p>Perform ridge regression on scaled expression data</p></a></li>
<li><a href='#RunBBKNN'><p>Perform batch balanced KNN</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Perform Batch Balanced KNN in R</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-02-13</td>
</tr>
<tr>
<td>Description:</td>
<td>A fast and intuitive batch effect removal tool for single-cell data. BBKNN is originally used in the 'scanpy' python package, and now can be used with 'Seurat' seamlessly.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0), methods, utils</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp (&ge; 1.0.8)</td>
</tr>
<tr>
<td>Imports:</td>
<td>future, glmnet, Matrix, magrittr, Rcpp, RcppAnnoy, reticulate,
rlang, Rtsne, Seurat, SeuratObject, tidytable, uwot (&ge; 0.1.14)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ycli1995/bbknnR">https://github.com/ycli1995/bbknnR</a>,
<a href="https://github.com/Teichlab/bbknn">https://github.com/Teichlab/bbknn</a>,
<a href="https://bbknn.readthedocs.io/en/latest/">https://bbknn.readthedocs.io/en/latest/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ycli1995/bbknnR/issues">https://github.com/ycli1995/bbknnR/issues</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>dplyr, knitr, rmarkdown, testthat (&ge; 3.0.0), patchwork</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-13 09:43:11 UTC; mydocker</td>
</tr>
<tr>
<td>Author:</td>
<td>Yuchen Li [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yuchen Li &lt;ycli1995@outlook.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-13 10:20:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='bbknnR-package'>bbknnR: Perform Batch Balanced KNN in R</h2><span id='topic+bbknnR'></span><span id='topic+bbknnR-package'></span>

<h3>Description</h3>

<p>A fast and intuitive batch effect removal tool for single-cell data. BBKNN is originally used in the 'scanpy' python package, and now can be used with 'Seurat' seamlessly.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Yuchen Li <a href="mailto:ycli1995@outlook.com">ycli1995@outlook.com</a>
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/ycli1995/bbknnR">https://github.com/ycli1995/bbknnR</a>
</p>
</li>
<li> <p><a href="https://github.com/Teichlab/bbknn">https://github.com/Teichlab/bbknn</a>
</p>
</li>
<li> <p><a href="https://bbknn.readthedocs.io/en/latest/">https://bbknn.readthedocs.io/en/latest/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/ycli1995/bbknnR/issues">https://github.com/ycli1995/bbknnR/issues</a>
</p>
</li></ul>


<hr>
<h2 id='panc8_small'>A small example version of the pancreas scRNA-seq dataset</h2><span id='topic+panc8_small'></span>

<h3>Description</h3>

<p>A subsetted version of the pancreas scRNA-seq dataset to test BBKNN
</p>


<h3>Usage</h3>

<pre><code class='language-R'>panc8_small
</code></pre>


<h3>Format</h3>

<p>A Seurat object with the following slots filled
</p>

<dl>
<dt>assays</dt><dd>
<p>Currently only contains one assay (&quot;RNA&quot; - scRNA-seq expression data)
</p>
<ul>
<li><p>counts - Raw expression data
</p>
</li>
<li><p>data - Normalized expression data
</p>
</li>
<li><p>scale.data - Scaled expression data
</p>
</li>
<li><p>var.features - names of the current features selected as variable
</p>
</li>
<li><p>meta.features - Assay level metadata such as mean and variance
</p>
</li></ul>
</dd>
<dt>meta.data</dt><dd><p>Cell level metadata</p>
</dd>
<dt>active.assay</dt><dd><p>Current default assay</p>
</dd>
<dt>active.ident</dt><dd><p>Current default idents</p>
</dd>
<dt>graphs</dt><dd><p>Empty</p>
</dd>
<dt>reductions</dt><dd><p>Dimensional reductions: currently PCA</p>
</dd>
<dt>version</dt><dd><p>Seurat version used to create the object</p>
</dd>
<dt>commands</dt><dd><p>Command history</p>
</dd>
</dl>



<h3>Source</h3>

<p>SeuratData <a href="https://github.com/satijalab/seurat-data">https://github.com/satijalab/seurat-data</a>
</p>

<hr>
<h2 id='RidgeRegression'>Perform ridge regression on scaled expression data</h2><span id='topic+RidgeRegression'></span><span id='topic+RidgeRegression.default'></span><span id='topic+RidgeRegression.Seurat'></span>

<h3>Description</h3>

<p>Perform ridge regression on scaled expression data, accepting both technical 
and biological categorical variables. The effect of the technical variables 
is removed while the effect of the biological variables is retained. This is 
a preprocessing step that can aid BBKNN integration.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RidgeRegression(object, ...)

## Default S3 method:
RidgeRegression(
  object,
  latent_data,
  batch_key,
  confounder_key,
  lambda = 1,
  seed = 42,
  verbose = TRUE,
  ...
)

## S3 method for class 'Seurat'
RidgeRegression(
  object,
  batch_key,
  confounder_key,
  assay = NULL,
  features = NULL,
  lambda = 1,
  run_pca = TRUE,
  npcs = 50,
  reduction.name = "pca",
  reduction.key = "PC_",
  replace = FALSE,
  seed = 42,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RidgeRegression_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="RidgeRegression_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="RidgeRegression_+3A_latent_data">latent_data</code></td>
<td>
<p>Extra data to regress out, should be cells x latent data</p>
</td></tr>
<tr><td><code id="RidgeRegression_+3A_batch_key">batch_key</code></td>
<td>
<p>Variables to regress out as technical effects. Must be 
included in column names of latent_data</p>
</td></tr>
<tr><td><code id="RidgeRegression_+3A_confounder_key">confounder_key</code></td>
<td>
<p>Variables to to retain as biological effects. Must be 
included in column names of latent_data</p>
</td></tr>
<tr><td><code id="RidgeRegression_+3A_lambda">lambda</code></td>
<td>
<p>A user supplied lambda sequence. pass to 
<code><a href="glmnet.html#topic+glmnet">glmnet</a></code></p>
</td></tr>
<tr><td><code id="RidgeRegression_+3A_seed">seed</code></td>
<td>
<p>Set a random seed. By default, sets the seed to 42. Setting NULL 
will not set a seed.</p>
</td></tr>
<tr><td><code id="RidgeRegression_+3A_verbose">verbose</code></td>
<td>
<p>Whether or not to print output to the console</p>
</td></tr>
<tr><td><code id="RidgeRegression_+3A_assay">assay</code></td>
<td>
<p>Name of Assay ridge regression is being run on</p>
</td></tr>
<tr><td><code id="RidgeRegression_+3A_features">features</code></td>
<td>
<p>Features to compute ridge regression on. If features=NULL, 
ridge regression will be run using the variable features for the Assay.</p>
</td></tr>
<tr><td><code id="RidgeRegression_+3A_run_pca">run_pca</code></td>
<td>
<p>Whether or not to run pca with regressed expression data 
(TRUE by default)</p>
</td></tr>
<tr><td><code id="RidgeRegression_+3A_npcs">npcs</code></td>
<td>
<p>Total Number of PCs to compute and store (50 by default)</p>
</td></tr>
<tr><td><code id="RidgeRegression_+3A_reduction.name">reduction.name</code></td>
<td>
<p>Dimensional reduction name (pca by default)</p>
</td></tr>
<tr><td><code id="RidgeRegression_+3A_reduction.key">reduction.key</code></td>
<td>
<p>Dimensional reduction key, specifies the string before 
the number for the dimension names (PC by default)</p>
</td></tr>
<tr><td><code id="RidgeRegression_+3A_replace">replace</code></td>
<td>
<p>Whether or not to replace original scale.data with regressed 
expression data (TRUE by default)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a Seurat object.
</p>


<h3>References</h3>

<p>Park, Jong-Eun, et al. &quot;A cell atlas of human thymic development 
defines T cell repertoire formation.&quot; Science 367.6480 (2020): eaay3224.
</p>

<hr>
<h2 id='RunBBKNN'>Perform batch balanced KNN</h2><span id='topic+RunBBKNN'></span><span id='topic+RunBBKNN.default'></span><span id='topic+RunBBKNN.Seurat'></span>

<h3>Description</h3>

<p>Batch balanced KNN, altering the KNN procedure to identify each cellâ€™s top 
neighbours in each batch separately instead of the entire cell pool with no 
accounting for batch. The nearest neighbours for each batch are then merged 
to create a final list of neighbours for the cell. Aligns batches in a quick 
and lightweight manner.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RunBBKNN(object, ...)

## Default S3 method:
RunBBKNN(
  object,
  batch_list,
  n_pcs = 50L,
  neighbors_within_batch = 3L,
  trim = NULL,
  approx = TRUE,
  use_annoy = TRUE,
  annoy_n_trees = 10L,
  pynndescent_n_neighbors = 30L,
  pynndescent_random_state = 0L,
  use_faiss = TRUE,
  metric = "euclidean",
  set_op_mix_ratio = 1,
  local_connectivity = 1,
  seed = 42,
  verbose = TRUE,
  ...
)

## S3 method for class 'Seurat'
RunBBKNN(
  object,
  batch_key,
  assay = NULL,
  reduction = "pca",
  n_pcs = 50L,
  graph_name = "bbknn",
  set_op_mix_ratio = 1,
  local_connectivity = 1,
  run_TSNE = TRUE,
  TSNE_name = "tsne",
  TSNE_key = "tSNE_",
  run_UMAP = TRUE,
  UMAP_name = "umap",
  UMAP_key = "UMAP_",
  min_dist = 0.3,
  spread = 1,
  seed = 42,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RunBBKNN_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_batch_list">batch_list</code></td>
<td>
<p>A character vector with the same length as nrow(pca)</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_n_pcs">n_pcs</code></td>
<td>
<p>Number of dimensions to use. Default is 50.</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_neighbors_within_batch">neighbors_within_batch</code></td>
<td>
<p>How many top neighbours to report for each 
batch; total number of neighbours in the initial k-nearest-neighbours 
computation will be this number times the number of batches. This then serves 
as the basis for the construction of a symmetrical matrix of connectivities.</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_trim">trim</code></td>
<td>
<p>Trim the neighbours of each cell to these many top 
connectivities. May help with population independence and improve the 
tidiness of clustering. The lower the value the more independent the 
individual populations, at the cost of more conserved batch effect. Default 
is 10 times neighbors_within_batch times the number of batches. Set to 0 to 
skip.</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_approx">approx</code></td>
<td>
<p>If TRUE, use approximate neighbour finding - RcppAnnoy or 
pyNNDescent. This results in a quicker run time for large datasets while also 
potentially increasing the degree of batch correction.</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_use_annoy">use_annoy</code></td>
<td>
<p>Only used when approx = TRUE. If TRUE, will use RcppAnnoy 
for neighbour finding. If FALSE, will use pyNNDescent instead.</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_annoy_n_trees">annoy_n_trees</code></td>
<td>
<p>Only used with annoy neighbour identification. The 
number of trees to construct in the annoy forest. More trees give higher 
precision when querying, at the cost of increased run time and resource 
intensity.</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_pynndescent_n_neighbors">pynndescent_n_neighbors</code></td>
<td>
<p>Only used with pyNNDescent neighbour 
identification. The number of neighbours to include in the approximate 
neighbour graph. More neighbours give higher precision when querying, at the 
cost of increased run time and resource intensity.</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_pynndescent_random_state">pynndescent_random_state</code></td>
<td>
<p>Only used with pyNNDescent neighbour 
identification. The RNG seed to use when creating the graph.</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_use_faiss">use_faiss</code></td>
<td>
<p>If approx = FALSE and the metric is &quot;euclidean&quot;, use the 
faiss package to compute nearest neighbours if installed. This improves 
performance at a minor cost to numerical precision as faiss operates on 
float32.</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_metric">metric</code></td>
<td>
<p>What distance metric to use. The options depend on the choice 
of neighbour algorithm. &quot;euclidean&quot;, the default, is always available.</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_set_op_mix_ratio">set_op_mix_ratio</code></td>
<td>
<p>Pass to 'set_op_mix_ratio' parameter for 
<code><a href="uwot.html#topic+umap">umap</a></code></p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_local_connectivity">local_connectivity</code></td>
<td>
<p>Pass to 'local_connectivity' parameter for 
<code><a href="uwot.html#topic+umap">umap</a></code></p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_seed">seed</code></td>
<td>
<p>Set a random seed. By default, sets the seed to 42. Setting 
<code>NULL</code> will not set a seed.</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_verbose">verbose</code></td>
<td>
<p>Whether or not to print output to the console</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_batch_key">batch_key</code></td>
<td>
<p>Column name in meta.data discriminating between your 
batches.</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_assay">assay</code></td>
<td>
<p>used to construct Graph.</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_reduction">reduction</code></td>
<td>
<p>Which dimensional reduction to use for the BBKNN input. 
Default is PCA</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_graph_name">graph_name</code></td>
<td>
<p>Name of the generated BBKNN graph. Default is bbknn.</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_run_tsne">run_TSNE</code></td>
<td>
<p>Whether or not to run t-SNE based on BBKNN results.</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_tsne_name">TSNE_name</code></td>
<td>
<p>Name to store t-SNE dimensional reduction.</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_tsne_key">TSNE_key</code></td>
<td>
<p>Specifies the string before the number of the t-SNE dimension 
names. tSNE by default.</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_run_umap">run_UMAP</code></td>
<td>
<p>Whether or not to run UMAP based on BBKNN results.</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_umap_name">UMAP_name</code></td>
<td>
<p>Name to store UMAP dimensional reduction.</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_umap_key">UMAP_key</code></td>
<td>
<p>Specifies the string before the number of the UMAP dimension 
names. tSNE by default.</p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_min_dist">min_dist</code></td>
<td>
<p>Pass to 'min_dist' parameter for <code><a href="uwot.html#topic+umap">umap</a></code></p>
</td></tr>
<tr><td><code id="RunBBKNN_+3A_spread">spread</code></td>
<td>
<p>Pass to 'spread' parameter for <code><a href="uwot.html#topic+umap">umap</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a Seurat object containing a new BBKNN Graph. If run t-SNE or 
UMAP, will also return corresponded reduction objects.
</p>


<h3>References</h3>

<p>PolaÅ„ski, Krzysztof, et al. &quot;BBKNN: fast batch alignment of single cell 
transcriptomes.&quot; Bioinformatics 36.3 (2020): 964-965.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
