<!DOCTYPE html><html><head><title>Help for package bbmix</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {bbmix}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bbmix-package'><p>The 'bbmix' package.</p></a></li>
<li><a href='#call_gt'><p>Call genotypes using beta binomial after model training</p></a></li>
<li><a href='#call_help'><p>call gt helper, calculate mean dbetabinom from all posterior samples</p></a></li>
<li><a href='#ex_alt_hom'><p>Exclude fSNPs with no alternative allele in any sample. Also exclude fSNPs if all samples are hom.</p></a></li>
<li><a href='#fit_bb'><p>Fit beta binomial distribution to allelic counts for homozygous reference, heterozygous, homozygous alternative</p></a></li>
<li><a href='#gt_help'><p>call gt helper, get posterior mean, expected gt and sd expected gt across all samples</p></a></li>
<li><a href='#poolreads'><p>Pool randomly selected reads from different files</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Bayesian Model for Genotyping using RNA-Seq</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>The method models RNA-seq reads using a mixture of 3 beta-binomial distributions to generate posterior probabilities for genotyping bi-allelic single nucleotide polymorphisms. Elena Vigorito, Anne Barton, Costantino Pitzalis, Myles J. Lewis and Chris Wallace (2023) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtad393">doi:10.1093/bioinformatics/btad393</a>&gt; "BBmix: a Bayesian beta-binomial mixture model for accurate genotyping from RNA-sequencing."</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Biarch:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, Rcpp (&ge; 0.12.0), rstan (&ge; 2.18.1), R.utils,
data.table, rmutil</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>BH (&ge; 1.66.0), Rcpp (&ge; 0.12.0), RcppEigen (&ge; 0.3.3.3.0),
rstan (&ge; 2.18.1), StanHeaders (&ge; 2.18.0)</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>GNU make</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.0</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-07-05 18:36:26 UTC; ev250</td>
</tr>
<tr>
<td>Author:</td>
<td>Elena Vigorito [aut, cre],
  Chris Wallace [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Elena Vigorito &lt;elena.vigorito@mrc-bsu.cam.ac.uk&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-07-06 14:00:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='bbmix-package'>The 'bbmix' package.</h2><span id='topic+bbmix-package'></span><span id='topic+bbmix'></span>

<h3>Description</h3>

<p>Bayesian Beta-Binomial mixture model for RNA-seq genotyping
</p>


<h3>References</h3>

<p>Stan Development Team (2018). RStan: the R interface to Stan. R package version 2.18.2. https://mc-stan.org
</p>

<hr>
<h2 id='call_gt'>Call genotypes using beta binomial after model training</h2><span id='topic+call_gt'></span>

<h3>Description</h3>

<p>Call genotypes using beta binomial after model training
</p>


<h3>Usage</h3>

<pre><code class='language-R'>call_gt(
  allele_counts_f,
  depth = 10,
  stan_f = NULL,
  legend_f,
  pop = "EUR",
  prob = 0.99,
  fisher_f = NULL,
  fisher = 30,
  cluster_f = NULL,
  out
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="call_gt_+3A_allele_counts_f">allele_counts_f</code></td>
<td>
<p>vector with file names with allele counts for SNPs</p>
</td></tr>
<tr><td><code id="call_gt_+3A_depth">depth</code></td>
<td>
<p>min read count to call variant</p>
</td></tr>
<tr><td><code id="call_gt_+3A_stan_f">stan_f</code></td>
<td>
<p>full name to stan object with model fit to extract mean of parameters. Defaults to the model trained with genome in a bottle reads. Otherwise this object can be generated with fit_bb function.</p>
</td></tr>
<tr><td><code id="call_gt_+3A_legend_f">legend_f</code></td>
<td>
<p>full name for file with SNP info to get allele frequency for prior</p>
</td></tr>
<tr><td><code id="call_gt_+3A_pop">pop</code></td>
<td>
<p>population to select AF for GT prior, defaults to EUR</p>
</td></tr>
<tr><td><code id="call_gt_+3A_prob">prob</code></td>
<td>
<p>cut-off for making hard calls, defaults to 0.99</p>
</td></tr>
<tr><td><code id="call_gt_+3A_fisher_f">fisher_f</code></td>
<td>
<p>file with Fisher test to detect strand bias</p>
</td></tr>
<tr><td><code id="call_gt_+3A_fisher">fisher</code></td>
<td>
<p>cut_off for Fisher test to detect strand bias</p>
</td></tr>
<tr><td><code id="call_gt_+3A_cluster_f">cluster_f</code></td>
<td>
<p>file with info about SNP clusters</p>
</td></tr>
<tr><td><code id="call_gt_+3A_out">out</code></td>
<td>
<p>character with file name to save genotype output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data table with genotype probabilities
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Retrive input files for running call_gt
counts_f &lt;- system.file("extdata/input", "NA12878.chr22.Q20.allelicCounts.txt",
package = "bbmix",
mustWork = TRUE)

legend &lt;- system.file("extdata/input", "1000GP_Phase3_chr22.legend",
package = "bbmix", mustWork = TRUE)

fisher_f &lt;- system.file("extdata/input", "chr22.FS.Q20.alleleCounts.txt",
package = "bbmix", mustWork = TRUE)

cluster_f &lt;- system.file("extdata/input", "fSNPs_22_RP_maf0_01_cluster3window35.txt",
package = "bbmix", mustWork = TRUE)

out &lt;- paste0(tempdir() , "/NA12878.chrom22.gt.txt")

## Run call_gt:
call_gt(allele_counts_f = counts_f,
legend_f = legend,
fisher_f = fisher_f,
cluster_f = cluster_f,
out = out)

unlink(out)

</code></pre>

<hr>
<h2 id='call_help'>call gt helper, calculate mean dbetabinom from all posterior samples</h2><span id='topic+call_help'></span>

<h3>Description</h3>

<p>call gt helper, calculate mean dbetabinom from all posterior samples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>call_help(n, m, mu, lambda)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="call_help_+3A_n">n</code></td>
<td>
<p>counts alt allele</p>
</td></tr>
<tr><td><code id="call_help_+3A_m">m</code></td>
<td>
<p>total counts</p>
</td></tr>
<tr><td><code id="call_help_+3A_mu">mu</code></td>
<td>
<p>vector with posterior draws for mu param</p>
</td></tr>
<tr><td><code id="call_help_+3A_lambda">lambda</code></td>
<td>
<p>vector with posterior draws for lambda param</p>
</td></tr>
</table>


<h3>Value</h3>

<p>mean of dbetabinom
</p>

<hr>
<h2 id='ex_alt_hom'>Exclude fSNPs with no alternative allele in any sample. Also exclude fSNPs if all samples are hom.</h2><span id='topic+ex_alt_hom'></span>

<h3>Description</h3>

<p>Exclude fSNPs with no alternative allele in any sample. Also exclude fSNPs if all samples are hom.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ex_alt_hom(gt_f, out)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ex_alt_hom_+3A_gt_f">gt_f</code></td>
<td>
<p>character vector with file names with genotype calls per sample</p>
</td></tr>
<tr><td><code id="ex_alt_hom_+3A_out">out</code></td>
<td>
<p>file name to save output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>save file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
gt_f &lt;- system.file("extdata/output", "gt.NA12878.chr22.txt",
package = "bbmix",
mustWork = TRUE)
out &lt;- tempfile()

## Running function
ex_alt_hom(gt_f, out)

unlink(out)

</code></pre>

<hr>
<h2 id='fit_bb'>Fit beta binomial distribution to allelic counts for homozygous reference, heterozygous, homozygous alternative</h2><span id='topic+fit_bb'></span>

<h3>Description</h3>

<p>Fit beta binomial distribution to allelic counts for homozygous reference, heterozygous, homozygous alternative
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_bb(
  counts_f,
  depth = 10,
  N = 1000,
  prefix = NULL,
  k = 3,
  alpha_p = c(1, 10, 499),
  beta_p = c(499, 10, 1),
  out,
  mc.cores = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_bb_+3A_counts_f">counts_f</code></td>
<td>
<p>file name with allele counts for SNPs</p>
</td></tr>
<tr><td><code id="fit_bb_+3A_depth">depth</code></td>
<td>
<p>depth cut-off to use to select SNPs to fit  distributions</p>
</td></tr>
<tr><td><code id="fit_bb_+3A_n">N</code></td>
<td>
<p>number of SNPs to use for fitting</p>
</td></tr>
<tr><td><code id="fit_bb_+3A_prefix">prefix</code></td>
<td>
<p>charcter with prefix to add for saving files, defaults to NULL</p>
</td></tr>
<tr><td><code id="fit_bb_+3A_k">k</code></td>
<td>
<p>number of components for mixture model, defaults to 3</p>
</td></tr>
<tr><td><code id="fit_bb_+3A_alpha_p">alpha_p</code></td>
<td>
<p>alpha parameter for the k components of alpha parameter</p>
</td></tr>
<tr><td><code id="fit_bb_+3A_beta_p">beta_p</code></td>
<td>
<p>beta paramenter for the k components of Beta parameter</p>
</td></tr>
<tr><td><code id="fit_bb_+3A_out">out</code></td>
<td>
<p>character with dir name to save output</p>
</td></tr>
<tr><td><code id="fit_bb_+3A_mc.cores">mc.cores</code></td>
<td>
<p>number of cores to use, defaults to parallel detected cores</p>
</td></tr>
</table>


<h3>Value</h3>

<p>saves stan object to file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
## Retrive input files for running call_gt
counts_f &lt;- system.file("extdata/input", "NA12878.chr22.Q20.allelicCounts.txt",
package = "bbmix",
mustWork = TRUE)

out &lt;- tempdir()
fit_bb(counts_f = counts_f, N=10,
out = out, mc.cores=1)
unlink(out)

## End(Not run)

</code></pre>

<hr>
<h2 id='gt_help'>call gt helper, get posterior mean, expected gt and sd expected gt across all samples</h2><span id='topic+gt_help'></span>

<h3>Description</h3>

<p>call gt helper, get posterior mean, expected gt and sd expected gt across all samples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gt_help(stan_samples, pop, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gt_help_+3A_stan_samples">stan_samples</code></td>
<td>
<p>matrix with samples extracted from stan fit object, params mu and lambda</p>
</td></tr>
<tr><td><code id="gt_help_+3A_pop">pop</code></td>
<td>
<p>population to select AF for GT prior, defaults to EUR</p>
</td></tr>
<tr><td><code id="gt_help_+3A_data">data</code></td>
<td>
<p>data table 1 row with counts and EAF to apply model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>gt_help()
</p>

<hr>
<h2 id='poolreads'>Pool randomly selected reads from different files</h2><span id='topic+poolreads'></span>

<h3>Description</h3>

<p>Pool randomly selected reads from different files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>poolreads(files, N = 1000, d = 10, out)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="poolreads_+3A_files">files</code></td>
<td>
<p>names for files to extract reads</p>
</td></tr>
<tr><td><code id="poolreads_+3A_n">N</code></td>
<td>
<p>number of reads to extract</p>
</td></tr>
<tr><td><code id="poolreads_+3A_d">d</code></td>
<td>
<p>depth for reads</p>
</td></tr>
<tr><td><code id="poolreads_+3A_out">out</code></td>
<td>
<p>file name to save reads</p>
</td></tr>
</table>


<h3>Value</h3>

<p>save files
</p>


<h3>Examples</h3>

<pre><code class='language-R'>counts_f &lt;- system.file("extdata/input", "NA12878.chr22.Q20.allelicCounts.txt",
package = "bbmix",
mustWork = TRUE)

## In this example we only use one file and we take a pool of 10 reads

out &lt;- tempfile()

poolreads(files=counts_f,
N=10,
d=10,
out = out)

unlink(out)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
