<!DOCTYPE html><html><head><title>Help for package BcDiag</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BcDiag}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BcDiag-package'><p>The BCDiag package</p></a></li>
<li><a href='#anomedOnlybic'><p> The anomedOnlybic function</p></a></li>
<li><a href='#breastc'><p>Gene Expression Data Example</p></a></li>
<li><a href='#dlbcl'><p>Gene Expression Data Example</p></a></li>
<li><a href='#exploreBic'><p>The exploreBic function</p></a></li>
<li><a href='#exploreOnlybic'><p>The exploreOnlybic function</p></a></li>
<li><a href='#profileBic'>
<p>The  profileBic function.</p></a></li>
<li><a href='#writeBic'><p>The writeBic function</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.0.10</td>
</tr>
<tr>
<td>Date:</td>
<td>2015-10-16</td>
</tr>
<tr>
<td>Title:</td>
<td>Diagnostics Plots for Bicluster Data</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>biclust, isa2</td>
</tr>
<tr>
<td>Author:</td>
<td>Aregay Mengsteab, Martin Otava, Tatsiana Khamiakova, Ewoud De Troyer</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ewoud De Troyer &lt;ewoud.detroyer@uhasselt.be&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Diagnostic tools based on two-way
        anova and median-polish residual plots for Bicluster output
        obtained from packages; "biclust" by Kaiser et al.(2008),"isa2"
        by Csardi et al. (2010) and "fabia" by Hochreiter et al.
        (2010). Moreover, It provides visualization tools for bicluster
        output and corresponding non-bicluster rows- or columns
        outcomes.  It has also extended the idea of Kaiser et al.(2008)
        which is, extracting bicluster output in a text format, by
        adding two bicluster methods from the fabia and isa2 R
        packages.</td>
</tr>
<tr>
<td>Imports:</td>
<td>fabia, methods, graphics, stats</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2015-10-16 21:05:23 UTC; rforge</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2015-10-17 15:53:37</td>
</tr>
<tr>
<td>Repository/R-Forge/Project:</td>
<td>bcdiag</td>
</tr>
<tr>
<td>Repository/R-Forge/Revision:</td>
<td>26</td>
</tr>
<tr>
<td>Repository/R-Forge/DateTimeStamp:</td>
<td>2015-10-16 20:54:29</td>
</tr>
</table>
<hr>
<h2 id='BcDiag-package'>The BCDiag package</h2><span id='topic+BcDiag-package'></span>

<h3>Description</h3>

<p>Bicluster Diagnostics plots</p>


<h3>Introduction</h3>

<p>The Bicluster Diagnostics plots(BcDiag) package is a visualization technique, 
for profiling and summarizing Bicluster data, particularly for gene expression level data.
Target data matrix are bicluster genes(rows) and conditions(columns) versus clustered genes or conditions.
</p>


<h3>Main task</h3>

<p>A BicDiag is a package of visualaization bicluster data, which is a 
subset matrix that have similar characterstics in terms of row(genes) 
and columns(conditions).
</p>
<p>It has used three different types of bicluster algorithms to extract the biculsterd data; 'biclust','isa2' and 'fabia'. plots such as boxplot,histogram, line plot,3D plot  are some of the plots that have used to visualize the data. 
</p>
<p>Major taskes of the package can be categorized in to three sections;
</p>

<ol>
<li><p> profiling and summarizing the biclustered vs. the clustered simultaneously
</p>
</li>
<li><p> profiling and summarizing only the biclusterd data.
</p>
</li>
<li><p> exploring the biclusterd data using anova and median polish techniques.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Mengsteab Aregay <a href="mailto:mycs.zab@gmail.com">mycs.zab@gmail.com</a></p>


<h3>References</h3>

<p>Hochreiter, S., Bodenhofer, U., Heusel, M.<em>et al.</em> (2010).FABIA: factor analysis for bicluster acquisition. <em>Bioinformatices</em>, 26, 1520-1527.
</p>
<p>Kaiser S. and Leisch F. (2008). A Toolbox for Bicluster Analysis in R. <em>Ludwigstrasse.</em> 33.
</p>
<p>Csardi G., Kutalik Z., and Bergmann S.(2010). Modular analysis of gene expression data with R. <em>Bioinformatics,</em> 26, 1376-7
</p>


<h3>See Also</h3>

<p>The Bicluster algorithms in the packages <span class="pkg">biclust</span>,<span class="pkg">fabia</span> and  <span class="pkg">isa2</span>.
</p>

<hr>
<h2 id='anomedOnlybic'> The anomedOnlybic function
</h2><span id='topic+anomedOnlybic'></span>

<h3>Description</h3>

<p>Provides ANOVA and median polish residual plots for biclustered data.</p>


<h3>Usage</h3>

<pre><code class='language-R'>anomedOnlybic(dset, bres, fit="boxplot", mname="biclust", bnum=1, 
fabia.thresZ=0.5,fabia.thresL=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="anomedOnlybic_+3A_dset">dset</code></td>
<td>
<p>data matrix.</p>
</td></tr>
<tr><td><code id="anomedOnlybic_+3A_bres">bres</code></td>
<td>
<p>bicluster result.</p>
</td></tr>
<tr><td><code id="anomedOnlybic_+3A_fit">fit</code></td>
<td>
<p>a string value to fit a plot;
'aplot','mplot','anovbplot','mpolishbplot','boxplot'.</p>
</td></tr>
<tr><td><code id="anomedOnlybic_+3A_mname">mname</code></td>
<td>
<p>method name; 'biclust', 'isa2', 'fabia' or 'bicare'.</p>
</td></tr>
<tr><td><code id="anomedOnlybic_+3A_bnum">bnum</code></td>
<td>
<p>existing biclusters; '1','2'...</p>
</td></tr>
<tr><td><code id="anomedOnlybic_+3A_fabia.thresz">fabia.thresZ</code></td>
<td>
<p>Bicluster threshold for <code>mname="fabia"</code>. Threshold for sample belonging to bicluster; default 0.5.</p>
</td></tr>
<tr><td><code id="anomedOnlybic_+3A_fabia.thresl">fabia.thresL</code></td>
<td>
<p>Bicluster threshold for <code>mname="fabia"</code>. Threshold for loading belonging to bicluster (if not given it is estimated).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A function provides residuals plots for biclustered data based on ANOVA and median polish.
</p>
<p>The function checks the required parameter values and fit the plot according to the user requirements.
</p>
<p>Note that the <code>"biclust"</code> option for <code>mname</code> will also accept results from the packages <span class="pkg">iBBiG</span> and <span class="pkg">rqubic</span>.
</p>


<h3>Value</h3>

<p>Residual plots or residual box plots.
</p>


<h3>Author(s)</h3>

<p>Mengsteab Aregay <a href="mailto:mycs.zab@gmail.com">mycs.zab@gmail.com</a></p>


<h3>References</h3>

  
<p>Van't Veer, L.J., Dai, H., van de Vijver, M.J., He, Y.D., Hart, A.A. et al. (2002). Gene expression profiling predicts clinical outcome of breast cancer,<em>Nature</em>, 415, 530-536.
</p>
<p>Kaiser S. and Leisch F. (2008). A Toolbox for Bicluster Analysis in R. <em>Ludwigstrasse.</em> 33.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(breastc)
library(biclust)
# find bicluster using one of biclust algorithms

bic &lt;- biclust(breastc, method=BCPlaid())
# fit residual boxplot from ANOVA
anomedOnlybic(dset=breastc,bres=bic,fit="boxplot",mname="biclust")</code></pre>

<hr>
<h2 id='breastc'>Gene Expression Data Example</h2><span id='topic+breastc'></span>

<h3>Description</h3>

<p>Microarray data set of van't Veer breast cancer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(breastc)</code></pre>


<h3>Format</h3>

<p>A data matrix with 1213 genes and 97 samples.
</p>


<h3>References</h3>

<p>Van't Veer, L.J., Dai, H., van de Vijver, M.J., He, Y.D., Hart, A.A. et al. (2002). Gene expression profiling predicts clinical outcome of breast cancer,<em>Nature</em>, 415, 530-536.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  
data(breastc)
  
head(breastc)

</code></pre>

<hr>
<h2 id='dlbcl'>Gene Expression Data Example</h2><span id='topic+dlbcl'></span>

<h3>Description</h3>

<p>Log transformed Microarray data set of Rosenwald diffuse large-B-cell lymphoma.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(dlbcl)</code></pre>


<h3>Format</h3>

<p>A data matrix with 661 genes and 141 samples.
</p>


<h3>References</h3>

<p>Rosenwald, A., Wright, G., Chan, W.C., Connors, J.M., Campo, E. et al. (2002). The use of molecular profiling to predict survival after chemotherapy for diffuse large-B-cell lymphoma, <em>New Engl. J. Med.</em>, 346. 1937-1947.</p>


<h3>Examples</h3>

<pre><code class='language-R'>  
data(dlbcl)
  
head(dlbcl)

</code></pre>

<hr>
<h2 id='exploreBic'>The exploreBic function</h2><span id='topic+exploreBic'></span>

<h3>Description</h3>

<p>Provides exploratory plots for biclustered and clustered data.</p>


<h3>Usage</h3>

<pre><code class='language-R'>exploreBic(dset, bres, gby ="genes", pfor ="mean", mname ="biclust", bnum =1, 
fabia.thresZ=0.5,fabia.thresL=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exploreBic_+3A_dset">dset</code></td>
<td>
<p>data matrix.</p>
</td></tr>
<tr><td><code id="exploreBic_+3A_bres">bres</code></td>
<td>
<p>bicluster result.</p>
</td></tr>
<tr><td><code id="exploreBic_+3A_gby">gby</code></td>
<td>
<p>dimension to plot; 'genes' or 'conditions'.</p>
</td></tr>
<tr><td><code id="exploreBic_+3A_pfor">pfor</code></td>
<td>
<p>plot for 'mean', 'median', 'variance', 'mad', 'all', or 'quant' (quantile).</p>
</td></tr>
<tr><td><code id="exploreBic_+3A_mname">mname</code></td>
<td>
<p>method name; 'biclust', 'isa2', 'fabia' or 'bicare'</p>
</td></tr>
<tr><td><code id="exploreBic_+3A_bnum">bnum</code></td>
<td>
<p>existing biclusters; '1','2'...</p>
</td></tr>
<tr><td><code id="exploreBic_+3A_fabia.thresz">fabia.thresZ</code></td>
<td>
<p>Bicluster threshold for <code>mname="fabia"</code>. Threshold for sample belonging to bicluster; default 0.5.</p>
</td></tr>
<tr><td><code id="exploreBic_+3A_fabia.thresl">fabia.thresL</code></td>
<td>
<p>Bicluster threshold for <code>mname="fabia"</code>. Threshold for loading belonging to bicluster (if not given it is estimated).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The exploreBic function is mainly used for exploratory data analysis. It provides summary plots for mean, median, variance, MAD and quantile plot.
</p>
<p>The <code><a href="#topic+exploreBic">exploreBic</a></code> function checks if the parameters are appropriately submitted and then identifies the biclusters submatrix and calculates its summary statistics. Finally, the results are displayed on the required plot.
</p>
<p>Note that the <code>"biclust"</code> option for <code>mname</code> will also accept results from the packages <span class="pkg">iBBiG</span> and <span class="pkg">rqubic</span>.
</p>


<h3>Value</h3>

<p>Summary plot will display according to the user specification.
</p>


<h3>Author(s)</h3>

<p>Mengsteab Aregay <a href="mailto:mycs.zab@gmail.com">mycs.zab@gmail.com</a></p>


<h3>References</h3>

<p>Van't Veer, L.J., Dai, H., van de Vijver, M.J., He, Y.D., Hart, A.A. et al. (2002). Gene expression profiling predicts clinical outcome of breast cancer,<em>Nature</em>, 415, 530-536.
</p>
<p>Hochreiter, S., Bodenhofer, U., Heusel, M.<em>et al.</em> (2010).FABIA: factor analysis for bicluster acquisition. <em>Bioinformatices</em>, 26, 1520-1527.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+exploreOnlybic">exploreOnlybic</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(breastc)
# find bicluster using biclust package
library(biclust)
bic &lt;- biclust(breastc,method=BCPlaid())
# Plot the mean of biclusterd and clustered genes parallely.
exploreBic(dset=breastc,bres=bic,gby="conditions",pfor="mean",mname="biclust")
</code></pre>

<hr>
<h2 id='exploreOnlybic'>The exploreOnlybic function</h2><span id='topic+exploreOnlybic'></span>

<h3>Description</h3>

<p>Provides exploratory plots only for biclustering results.</p>


<h3>Usage</h3>

<pre><code class='language-R'>exploreOnlybic(dset, bres, pfor= "all", gby= "genes", mname="biclust",bnum=1, 
fabia.thresZ=0.5,fabia.thresL=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exploreOnlybic_+3A_dset">dset</code></td>
<td>
<p>data matrix.</p>
</td></tr>
<tr><td><code id="exploreOnlybic_+3A_bres">bres</code></td>
<td>
<p>biclustering result.</p>
</td></tr>
<tr><td><code id="exploreOnlybic_+3A_gby">gby</code></td>
<td>
<p>group bicluster; 'genes' or 'conditions'.</p>
</td></tr>
<tr><td><code id="exploreOnlybic_+3A_pfor">pfor</code></td>
<td>
<p>fit a plot for 'mean', 'median', 'variance', 'mad', 'all', or 'quant' (quantile).</p>
</td></tr>
<tr><td><code id="exploreOnlybic_+3A_mname">mname</code></td>
<td>
<p>method name; 'biclust', 'isa2', 'fabia' or 'bicare'.</p>
</td></tr>
<tr><td><code id="exploreOnlybic_+3A_bnum">bnum</code></td>
<td>
<p>existing biclusters; '1','2'...</p>
</td></tr>
<tr><td><code id="exploreOnlybic_+3A_fabia.thresz">fabia.thresZ</code></td>
<td>
<p>Bicluster threshold for <code>mname="fabia"</code>. Threshold for sample belonging to bicluster; default 0.5.</p>
</td></tr>
<tr><td><code id="exploreOnlybic_+3A_fabia.thresl">fabia.thresL</code></td>
<td>
<p>Bicluster threshold for <code>mname="fabia"</code>. Threshold for loading belonging to bicluster (if not given it is estimated).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The exploreOnlybic function has similar function with <code><a href="#topic+exploreBic">exploreBic</a></code>. The only difference is that it provides exploratory plots only for biclustered data.</p>


<h3>Value</h3>

<p>Summary plot will display only for biclustered data.
</p>
<p>Note that the <code>"biclust"</code> option for <code>mname</code> will also accept results from the packages <span class="pkg">iBBiG</span> and <span class="pkg">rqubic</span>.
</p>


<h3>Author(s)</h3>

<p>Mengsteab Aregay <a href="mailto:mycs.zab@gmail.com">mycs.zab@gmail.com</a></p>


<h3>References</h3>

<p>Van't Veer, L.J., Dai, H., van de Vijver, M.J., He, Y.D., Hart, A.A. et al. (2002). Gene expression profiling predicts clinical outcome of breast cancer,<em>Nature</em>, 415, 530-536.
</p>
<p>Hochreiter, S., Bodenhofer, U., Heusel, M.<em>et al.</em> (2010).FABIA: factor analysis for bicluster acquisition. <em>Bioinformatices</em>, 26, 1520-1527.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+exploreBic">exploreBic</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(breastc)
# find bicluster using biclust algorithm
library(biclust)
bic &lt;- biclust(breastc,method=BCPlaid())
# Plot the median of biclusterd data.
exploreOnlybic(dset=breastc, bres=bic, pfor="all", gby="genes", mname="biclust", bnum=1)
</code></pre>

<hr>
<h2 id='profileBic'>
The  profileBic function.
</h2><span id='topic+profileBic'></span>

<h3>Description</h3>

<p>Provides profile plots for biclustered and clustered data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>profileBic(dset, bres, mname = c("fabia", "isa2", "biclust","bicare"), bplot = "all",
gby = "genes", bnum = 1, teta = 120, ph = 30, fabia.thresZ=0.5,fabia.thresL=NULL,
BClabel=TRUE,gene.lines=NULL,condition.lines=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="profileBic_+3A_dset">dset</code></td>
<td>
<p>data matrix.</p>
</td></tr>
<tr><td><code id="profileBic_+3A_bres">bres</code></td>
<td>
<p>biclustering result.</p>
</td></tr>
<tr><td><code id="profileBic_+3A_mname">mname</code></td>
<td>
<p>method name; 'biclust', 'isa2', 'fabia' or 'bicare'.</p>
</td></tr>
<tr><td><code id="profileBic_+3A_bplot">bplot</code></td>
<td>
<p>types of plots; 'all','lines', 'boxplot', 'histogram' or '3D'.</p>
</td></tr>
<tr><td><code id="profileBic_+3A_gby">gby</code></td>
<td>
<p>grouped by; 'genes', or 'conditions'.</p>
</td></tr>
<tr><td><code id="profileBic_+3A_bnum">bnum</code></td>
<td>
<p>Existing biclusters; '1','2',...</p>
</td></tr>
<tr><td><code id="profileBic_+3A_teta">teta</code></td>
<td>
<p>numerical value to rotate the 3D; 0, 90, 180,...</p>
</td></tr>
<tr><td><code id="profileBic_+3A_ph">ph</code></td>
<td>
<p>numerical value to rotate the 3D; 0, 90, 180,...</p>
</td></tr>
<tr><td><code id="profileBic_+3A_fabia.thresz">fabia.thresZ</code></td>
<td>
<p>Bicluster threshold for <code>mname="fabia"</code>. Threshold for sample belonging to bicluster; default 0.5.</p>
</td></tr>
<tr><td><code id="profileBic_+3A_fabia.thresl">fabia.thresL</code></td>
<td>
<p>Bicluster threshold for <code>mname="fabia"</code>. Threshold for loading belonging to bicluster (if not given it is estimated).</p>
</td></tr>
<tr><td><code id="profileBic_+3A_bclabel">BClabel</code></td>
<td>
<p><code>TRUE</code>/<code>FALSE</code> to show BC labels on the lines plot.</p>
</td></tr>
<tr><td><code id="profileBic_+3A_gene.lines">gene.lines</code></td>
<td>
<p>Vector of indices or names of genes inside of Bicluster <code>bnum</code>. These gene profiles will be highlighted in the line plot (<code>bplot='lines'</code>).</p>
</td></tr>
<tr><td><code id="profileBic_+3A_condition.lines">condition.lines</code></td>
<td>
<p>Vector of indices or names of conditions inside of Bicluster <code>bnum</code>. These condition profiles will be highlighted in the line plot (<code>bplot='lines'</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The profile.bic function checks if all parameters are correctly submitted and then identifies the biclustered and clustered data.
</p>
<p>Note that the <code>"biclust"</code> option for <code>mname</code> will also accept results from the packages <span class="pkg">iBBiG</span> and <span class="pkg">rqubic</span>.</p>


<h3>Value</h3>

<p>profile.bic(dset, bres, mname=&quot;biclust&quot;, bplot=&quot;all&quot;, gby=&quot;genes&quot;, bnum=1, teta=120, ph=30)
</p>


<h3>Author(s)</h3>

<p>Mengsteab Aregay <a href="mailto:mycs.zab@gmail.com">mycs.zab@gmail.com</a></p>


<h3>References</h3>

<p>Van't Veer, L.J., Dai, H., van de Vijver, M.J., He, Y.D., Hart, A.A. et al. (2002). Gene expression profiling predicts clinical outcome of breast cancer,<em>Nature</em>, 415, 530-536.
</p>
<p>Kaiser S. and Leisch F. (2008). A Toolbox for Bicluster Analysis in R. <em>Ludwigstrasse.</em> 33.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># create the biclustering result
data(breastc)
library(biclust)
bic&lt;- biclust(breastc, method=BCPlaid())
# 3 biclusters found

# 3D profile plot for biclustered and clustered data.
profileBic(dset=breastc,bres=bic,mname="biclust",
bplot="3D",gby="genes",teta=-30,ph=50,bnum=1)
</code></pre>

<hr>
<h2 id='writeBic'>The writeBic function</h2><span id='topic+writeBic'></span>

<h3>Description</h3>

<p>Provides a summary output in a text format, extracted from 'biclust','isa2' and 'fabia' bicluster algorithms. </p>


<h3>Usage</h3>

<pre><code class='language-R'>writeBic(dset, fileName, bicResult, bicname, 
mname = c("fabia", "isa2", "biclust","bicare"), append = TRUE, delimiter = " ", 
fabia.thresZ=0.5,fabia.thresL=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writeBic_+3A_dset">dset</code></td>
<td>
<p>data matrix</p>
</td></tr>
<tr><td><code id="writeBic_+3A_filename">fileName</code></td>
<td>
<p>the name of the bicluster file to be saved.</p>
</td></tr>
<tr><td><code id="writeBic_+3A_bicresult">bicResult</code></td>
<td>
<p>bicluster result obtained from 'biclust','isa2' or 'fabia'</p>
</td></tr>
<tr><td><code id="writeBic_+3A_bicname">bicname</code></td>
<td>
<p>the title to be given for the biclustered data.</p>
</td></tr>
<tr><td><code id="writeBic_+3A_mname">mname</code></td>
<td>
<p>method name; 'biclust', 'isa2', 'fabia' or 'bicare'</p>
</td></tr>
<tr><td><code id="writeBic_+3A_append">append</code></td>
<td>
<p>logical value; TRUE as default</p>
</td></tr>
<tr><td><code id="writeBic_+3A_delimiter">delimiter</code></td>
<td>
<p>delimiter in created output file; default value is &quot; &quot;.</p>
</td></tr>
<tr><td><code id="writeBic_+3A_fabia.thresz">fabia.thresZ</code></td>
<td>
<p>Bicluster threshold for <code>mname="fabia"</code>. Threshold for sample belonging to bicluster; default 0.5.</p>
</td></tr>
<tr><td><code id="writeBic_+3A_fabia.thresl">fabia.thresL</code></td>
<td>
<p>Bicluster threshold for <code>mname="fabia"</code>. Threshold for loading belonging to bicluster (if not given it is estimated).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The original function was developed in 'biclust' package by Kaiser <em>et.al</em> (2008). We extend the function to be used for further bicluster algorithms, such as; 'isa2', 'fabia' and 'bicare'.
</p>
<p>Note that the <code>"biclust"</code> option for <code>mname</code> will also accept results from the packages <span class="pkg">iBBiG</span> and <span class="pkg">rqubic</span>.
</p>


<h3>Value</h3>

<p>Biclustered text file with title, total number of biclustered, dimension and name of the biclustered genes(rows) or conditions(columns).</p>


<h3>Author(s)</h3>

<p>Mengsteab Aregay 
</p>
<p><a href="mailto:mycs.zab@gmail.com">mycs.zab@gmail.com</a></p>


<h3>References</h3>

<p>Van't Veer, L.J., Dai, H., van de Vijver, M.J., He, Y.D., Hart, A.A. et al. (2002). Gene expression profiling predicts clinical outcome of breast cancer,<em>Nature</em>, 415, 530-536.
</p>
<p>Kaiser S. and Leisch F. (2008). A Toolbox for Bicluster Analysis in R. <em>Ludwigstrasse.</em> 33.
</p>
<p>Csardi G., Kutalik Z., and Bergmann S.(2010). Modular analysis of gene expression data with R. <em>Bioinformatics,</em> 26, 1376-7
</p>


<h3>See Also</h3>

<p><span class="pkg">biclust</span></p>


<h3>Examples</h3>

<pre><code class='language-R'># create the biclustering result
data(breastc)
library(fabia)
fab&lt;- fabia(breastc)
# write the biclustering result into a text file
writeBic(dset=breastc,fileName="fabiabreast.txt",
bicResult=fab, bicname="Biclust results for fabia",
mname="fabia")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
