<!DOCTYPE html><html><head><title>Help for package BDgraph</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BDgraph}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BDgraph-package'><p> Bayesian Structure Learning in Graphical Models</p></a></li>
<li><a href='#adj2link'><p> Extract links from an adjacency matrix</p></a></li>
<li><a href='#auc'><p> Compute the area under the ROC curve</p></a></li>
<li><a href='#bdgraph'><p>	Search algorithm in graphical models</p></a></li>
<li><a href='#bdgraph.dw'><p>	Search algorithm for Gaussian copula graphical models for count data</p></a></li>
<li><a href='#bdgraph.mpl'><p>	Search algorithm in graphical models using marginal pseudo-likehlihood</p></a></li>
<li><a href='#bdgraph.npn'><p>Nonparametric transfer</p></a></li>
<li><a href='#bdgraph.sim'><p> Graph data simulation</p></a></li>
<li><a href='#bdw.reg'><p>	Bayesian estimation of (zero-inflated) Discrete Weibull regression</p></a></li>
<li><a href='#bf'><p> Bayes factor for two graphs</p></a></li>
<li><a href='#compare'><p> Graph structure comparison</p></a></li>
<li><a href='#conf.mat'><p>	Confusion Matrix</p></a></li>
<li><a href='#conf.mat.plot'><p>	Plot Confusion Matrix</p></a></li>
<li><a href='#covariance'><p> Estimated covariance matrix</p></a></li>
<li><a href='#Discrete Weibull'><p> The Discrete Weibull Distribution (Type 1)</p></a></li>
<li><a href='#geneExpression'><p> Human gene expression dataset</p></a></li>
<li><a href='#gnorm'><p> Normalizing constant for G-Wishart</p></a></li>
<li><a href='#graph.sim'><p> Graph simulation</p></a></li>
<li><a href='#link2adj'><p> Extract links from an adjacency matrix</p></a></li>
<li><a href='#pgraph'><p> Posterior probabilities of the graphs</p></a></li>
<li><a href='#plinks'><p> Estimated posterior link probabilities</p></a></li>
<li><a href='#plot.bdgraph'><p> Plot function for <code>S3</code> class &quot;<code>bdgraph</code>&quot;</p></a></li>
<li><a href='#plot.graph'><p> Plot function for <code>S3</code> class <code>"graph"</code></p></a></li>
<li><a href='#plot.sim'><p> Plot function for <code>S3</code> class &quot;<code>sim</code>&quot;</p></a></li>
<li><a href='#plotcoda'><p> Convergence plot</p></a></li>
<li><a href='#plotroc'><p> ROC plot</p></a></li>
<li><a href='#posterior.predict'><p> Posterior Predictive Samples</p></a></li>
<li><a href='#precision'><p> Estimated precision matrix</p></a></li>
<li><a href='#predict.bdgraph'><p> Predict function for <code>S3</code> class &quot;<code>bdgraph</code>&quot;</p></a></li>
<li><a href='#print.bdgraph'><p>Print function for <code>S3</code> class &quot;<code>bdgraph</code>&quot;</p></a></li>
<li><a href='#print.sim'><p> Print function for <code>S3</code> class &quot;<code>sim</code>&quot;</p></a></li>
<li><a href='#reinis'><p> Risk factors of coronary heart disease</p></a></li>
<li><a href='#rgwish'><p>Sampling from G-Wishart distribution</p></a></li>
<li><a href='#rmvnorm'><p> Generate data from the multivariate Normal distribution</p></a></li>
<li><a href='#roc'><p> Build a ROC curve</p></a></li>
<li><a href='#rwish'><p> Sampling from Wishart distribution</p></a></li>
<li><a href='#select'><p> Graph selection</p></a></li>
<li><a href='#sparsity'><p> Compute the sparsity of a graph</p></a></li>
<li><a href='#summary.bdgraph'><p> Summary function for <code>S3</code> class &quot;<code>bdgraph</code>&quot;</p></a></li>
<li><a href='#surveyData'><p> Labor force survey data</p></a></li>
<li><a href='#traceplot'><p> Trace plot of graph size</p></a></li>
<li><a href='#transfer'><p> transfer for count data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Bayesian Structure Learning in Graphical Models using
Birth-Death MCMC</td>
</tr>
<tr>
<td>Version:</td>
<td>2.72</td>
</tr>
<tr>
<td>Description:</td>
<td>Statistical tools for Bayesian structure learning in undirected graphical models for continuous, ordinal/discrete/count, and mixed data. The package is implemented the recent improvements in the Bayesian graphical models' literature, including Mohammadi and Wit (2015) &lt;<a href="https://doi.org/10.1214%2F14-BA889">doi:10.1214/14-BA889</a>&gt;, Mohammadi et al. (2021) &lt;<a href="https://doi.org/10.1080%2F01621459.2021.1996377">doi:10.1080/01621459.2021.1996377</a>&gt;, and Dobra and Mohammadi (2018) &lt;<a href="https://doi.org/10.1214%2F18-AOAS1164">doi:10.1214/18-AOAS1164</a>&gt;. </td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://www.uva.nl/profile/a.mohammadi">https://www.uva.nl/profile/a.mohammadi</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>igraph, ggplot2, pROC</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ssgraph, huge, tmvtnorm, skimr, knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Author:</td>
<td>Reza Mohammadi <a href="https://orcid.org/0000-0001-9538-0648"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Ernst Wit <a href="https://orcid.org/0000-0002-3671-9610"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Adrian Dobra <a href="https://orcid.org/0000-0001-7793-2197"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Reza Mohammadi &lt;a.mohammadi@uva.nl&gt;</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-12-24 12:57:22 UTC; reza</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-12-25 07:20:14 UTC</td>
</tr>
</table>
<hr>
<h2 id='BDgraph-package'> Bayesian Structure Learning in Graphical Models </h2><span id='topic+BDgraph-package'></span><span id='topic+BDgraph'></span><span id='topic+get_Ds_tgm_R'></span><span id='topic+get_Ts_R'></span><span id='topic+update_tu_R'></span><span id='topic+update_mu_R'></span><span id='topic+compute_tp_fp'></span><span id='topic+compute_measures'></span><span id='topic+get_graph'></span><span id='topic+get_g_prior'></span><span id='topic+get_g_start'></span><span id='topic+get_K_start'></span><span id='topic+get_S_n_p'></span><span id='topic+get_cores'></span><span id='topic+hill_climb_mpl'></span><span id='topic+local_mb_hc'></span><span id='topic+global_hc'></span><span id='topic+log_mpl_disrete'></span><span id='topic+hill_climb_mpl_binary'></span><span id='topic+local_mb_hc_binary'></span><span id='topic+global_hc_binary'></span><span id='topic+log_mpl_binary'></span><span id='topic+detect_cores'></span><span id='topic+sample_ug'></span><span id='topic+generate_clique_factors'></span><span id='topic+calc_joint_dist'></span><span id='topic+get_bounds_dw'></span><span id='topic+ddweibull_reg'></span><span id='topic+log_post_cond_dw'></span><span id='topic+near_positive_definite'></span>

<h3>Description</h3>

<p>The <code>R</code> package <span class="pkg">BDgraph</span> provides statistical tools for Bayesian structure learning in undirected graphical models for continuous, ordinal/count/dicrete, binary, and mixed data. The package is implemented the recent improvements in the Bayesian graphical models' literature, including Mohammadi and Wit (2015), Mohammadi et al. (2021), Mohammadi et al. (2017), and Dobra and Mohammadi (2018). To speed up the computations, the intensive tasks of the package are implemented in parallel using <span class="pkg">OpenMP</span> in <code>C++</code> and interfaced with <code>R</code>. Besides, the package contains several functions for simulation and visualization, as well as several multivariate datasets taken from the literature.
</p>


<h3>How to cite this package</h3>

<p>To cite <span class="pkg">BDgraph</span> in publications use:
</p>
<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a>
</p>


<h3>Author(s)</h3>

<p>Reza Mohammadi [aut, cre] (&lt;https://orcid.org/0000-0001-9538-0648&gt;), <br />
Ernst Wit [aut] (&lt;https://orcid.org/0000-0002-3671-9610&gt;), <br />
Adrian Dobra [ctb] (&lt;https://orcid.org/0000-0001-7793-2197&gt;). <br />
Maintainer: Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a>
</p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a>
</p>
<p>Mohammadi, A. and Wit, E. C. (2015). Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Bayesian Analysis</em>, 10(1):109-138, <a href="https://doi.org/10.1214/14-BA889">doi:10.1214/14-BA889</a>
</p>
<p>Mohammadi, R., Massam, H. and Letac, G. (2021). Accelerating Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Journal of the American Statistical Association</em>, <a href="https://doi.org/10.1080/01621459.2021.1996377">doi:10.1080/01621459.2021.1996377</a> 
</p>
<p>Mohammadi, A., et al (2017). Bayesian modelling of Dupuytren disease by using Gaussian copula graphical models, <em>Journal of the Royal Statistical Society: Series C</em>, 66(3):629-645, <a href="https://doi.org/10.1111/rssc.12171">doi:10.1111/rssc.12171</a> 
</p>
<p>Dobra, A. and Mohammadi, R. (2018). Loglinear Model Selection and Human Mobility, <em>Annals of Applied Statistics</em>, 12(2):815-845, <a href="https://doi.org/10.1214/18-AOAS1164">doi:10.1214/18-AOAS1164</a>
</p>
<p>Vinciotti, V., Behrouzi, P., and Mohammadi, R. (2022) Bayesian structural learning of microbiota systems from count metagenomic data, <em>arXiv preprint</em>, <a href="https://doi.org/10.48550/arXiv.2203.10118">doi:10.48550/arXiv.2203.10118</a>
</p>
<p>Dobra, A. and Lenkoski, A. (2011). Copula Gaussian graphical models and their application to modeling functional disability data, <em>The Annals of Applied Statistics</em>, 5(2A):969-93, <a href="https://doi.org/10.1214/10-AOAS397">doi:10.1214/10-AOAS397</a>
</p>
<p>Dobra, A., et al. (2011). Bayesian inference for general Gaussian graphical models with application to multivariate lattice data. <em>Journal of the American Statistical Association</em>, 106(496):1418-33, <a href="https://doi.org/10.1198/jasa.2011.tm10465">doi:10.1198/jasa.2011.tm10465</a>
</p>
<p>Mohammadi, A. and Dobra, A. (2017). The <code>R</code> Package <span class="pkg">BDgraph</span> for Bayesian Structure Learning in Graphical Models, <em>ISBA Bulletin</em>, 24(4):11-16
</p>
<p>Lenkoski, A. (2013). A direct sampler for G-Wishart variates, <em>Stat</em>, 2(1):119-28, <a href="https://doi.org/10.1002/sta4.23">doi:10.1002/sta4.23</a>
</p>
<p>Pensar, J. et al (2017) Marginal pseudo-likelihood learning of discrete Markov network structures, <em>Bayesian Analysis</em>, 12(4):1195-215, <a href="https://doi.org/10.1214/16-BA1032">doi:10.1214/16-BA1032</a>
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code>, <code><a href="#topic+bdgraph.dw">bdgraph.dw</a></code>, <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>, <code><a href="#topic+compare">compare</a></code>, <code><a href="#topic+rgwish">rgwish</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library( BDgraph )

set.seed( 10 )

# Generating multivariate normal data from a 'scale-free' graph
data.sim &lt;- bdgraph.sim( n = 100, p = 10, graph = "scale-free", vis = TRUE )

# Running algorithm based on GGMs
bdgraph.obj &lt;- bdgraph( data = data.sim, iter = 5000 )

summary( bdgraph.obj )

# To compare the result with true graph
compare( bdgraph.obj, data.sim, main = c( "Target", "BDgraph" ), vis = TRUE )

# Confusion Matrix
conf.mat( actual = data.sim, pred = bdgraph.obj )

conf.mat.plot( actual = data.sim, pred = bdgraph.obj )

# Running algorithm based on GGMs and marginal pseudo-likelihood
bdgraph.mpl.obj &lt;- bdgraph.mpl( data = data.sim, iter = 5000 )

summary( bdgraph.mpl.obj )

# Confusion Matrix
conf.mat( actual = data.sim, pred = bdgraph.mpl.obj )

conf.mat.plot( actual = data.sim, pred = bdgraph.mpl.obj )

# To compare the results of both algorithms with true graph
compare( list( bdgraph.obj, bdgraph.mpl.obj ), data.sim, 
         main = c( "Target", "BDgraph", "BDgraph_mpl" ), vis = TRUE )

## End(Not run)
</code></pre>

<hr>
<h2 id='adj2link'> Extract links from an adjacency matrix </h2><span id='topic+adj2link'></span>

<h3>Description</h3>

<p> Extract links from an adjacency matrix or an object of calsses &quot;<code>sim</code>&quot; from function <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code> and
<code>"graph"</code> from function <code><a href="#topic+graph.sim">graph.sim</a></code>.</p>


<h3>Usage</h3>

<pre><code class='language-R'> adj2link( adj ) </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adj2link_+3A_adj">adj</code></td>
<td>
     
<p>adjacency matrix corresponding to a graph structure in which <code class="reqn">a_{ij}=1</code> if there is a link between notes <code class="reqn">i</code> and <code class="reqn">j</code>, otherwise <code class="reqn">a_{ij}=0</code>.
It can be an object with <code>S3</code> class &quot;<code>sim</code>&quot; from function <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>.
It can be an object with <code>S3</code> class <code>"graph"</code> from function <code><a href="#topic+graph.sim">graph.sim</a></code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p> matrix corresponding to the extracted links from graph structure.</p>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a>
</p>
<p>Mohammadi, A. and Wit, E. C. (2015). Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Bayesian Analysis</em>, 10(1):109-138, <a href="https://doi.org/10.1214/14-BA889">doi:10.1214/14-BA889</a>
</p>
<p>Mohammadi, R., Massam, H. and Letac, G. (2021). Accelerating Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Journal of the American Statistical Association</em>, <a href="https://doi.org/10.1080/01621459.2021.1996377">doi:10.1080/01621459.2021.1996377</a> 
</p>
<p>Mohammadi, A. et al (2017). Bayesian modelling of Dupuytren disease by using Gaussian copula graphical models, <em>Journal of the Royal Statistical Society: Series C</em>, 66(3):629-645, <a href="https://doi.org/10.1111/rssc.12171">doi:10.1111/rssc.12171</a> 
</p>
<p>Dobra, A. and Mohammadi, R. (2018). Loglinear Model Selection and Human Mobility, <em>Annals of Applied Statistics</em>, 12(2):815-845, <a href="https://doi.org/10.1214/18-AOAS1164">doi:10.1214/18-AOAS1164</a>
</p>
<p>Pensar, J. et al (2017) Marginal pseudo-likelihood learning of discrete Markov network structures, <em>Bayesian Analysis</em>, 12(4):1195-215, <a href="https://doi.org/10.1214/16-BA1032">doi:10.1214/16-BA1032</a>
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+link2adj">link2adj</a></code>, <code><a href="#topic+graph.sim">graph.sim</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'># Generating a 'random' graph 
adj &lt;- graph.sim( p = 6, vis = TRUE )

adj2link( adj )
</code></pre>

<hr>
<h2 id='auc'> Compute the area under the ROC curve </h2><span id='topic+auc'></span>

<h3>Description</h3>

 
<p>This function computes the numeric value of area under the ROC curve (AUC) specifically for graph structure learning.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> 
auc( pred, actual, cut = 200, calibrate = TRUE )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auc_+3A_pred">pred</code></td>
<td>

<p>adjacency matrix corresponding to an estimated graph. 
It can be an object with <code>S3</code> class &quot;<code>bdgraph</code>&quot; from function <code><a href="#topic+bdgraph">bdgraph</a></code>. 
It can be an object of <code>S3</code> class <code>"ssgraph"</code>, from the function <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code> of <code>R</code> package <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code>.
It can be a <code><a href="base.html#topic+numeric">numeric</a></code> or <code><a href="base.html#topic+ordered">ordered</a></code> vector of the same length than <code>actual</code>, containing the predicted
value of each observation.
</p>
</td></tr>
<tr><td><code id="auc_+3A_actual">actual</code></td>
<td>
<p> adjacency matrix corresponding to the actual graph structure in which <code class="reqn">a_{ij}=1</code> if there is a link between notes <code class="reqn">i</code> and <code class="reqn">j</code>, otherwise <code class="reqn">a_{ij}=0</code>.
It can be an object with <code>S3</code> class &quot;<code>sim</code>&quot; from function <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>.
It can be an object with <code>S3</code> class <code>"graph"</code> from function <code><a href="#topic+graph.sim">graph.sim</a></code>.
It can be a factor, numeric or character vector of responses (true class), typically encoded with 0 (controls) and 1 (cases).
Only two classes can be used in a ROC curve. </p>
</td></tr>
<tr><td><code id="auc_+3A_cut">cut</code></td>
<td>
<p> number of cut points.</p>
</td></tr>
<tr><td><code id="auc_+3A_calibrate">calibrate</code></td>
<td>
<p>If <code>TRUE</code>, compute the value of AUC by taking the level of the probabilities into account.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The numeric AUC value</p>


<h3>Author(s)</h3>

<p>Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a>; Lucas Vogels <a href="mailto:l.f.o.vogels@uva.nl">l.f.o.vogels@uva.nl</a></p>


<h3>References</h3>

<p>Tom Fawcett (2006) &ldquo;An introduction to ROC analysis&rdquo;. <em>Pattern
Recognition Letters</em> <b>27</b>, 861&ndash;874, <a href="https://doi.org/10.1016/j.patrec.2005.10.010">doi:10.1016/j.patrec.2005.10.010</a>
</p>
<p>Xavier Robin, Natacha Turck, Alexandre Hainard, <em>et al.</em>
(2011) &ldquo;pROC: an open-source package for R and S+ to analyze and
compare ROC curves&rdquo;. <em>BMC Bioinformatics</em>, <b>7</b>, 77, 
<a href="https://doi.org/10.1186/1471-2105-12-77">doi:10.1186/1471-2105-12-77</a>.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+plotroc">plotroc</a></code>, <code><a href="pROC.html#topic+plot.roc">pROC::plot.roc()</a></code>, <code><a href="pROC.html#topic+auc">pROC::auc()</a></code>, <code><a href="pROC.html#topic+print.roc">pROC::print.roc()</a></code>, <code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code>, <code><a href="#topic+compare">compare</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set.seed( 5 )

# Generating multivariate normal data from a 'scale-free' graph
data.sim = bdgraph.sim( n = 200, p = 15, graph = "scale-free", vis = TRUE )

# Running BDMCMC algorithm 
sample.bdmcmc = bdgraph( data = data.sim, algorithm = "bdmcmc", iter = 10000 )

BDgraph::auc( pred = sample.bdmcmc, actual = data.sim, calibrate = TRUE )


## End(Not run)
</code></pre>

<hr>
<h2 id='bdgraph'>	Search algorithm in graphical models </h2><span id='topic+bdgraph'></span>

<h3>Description</h3>

<p>As the main function of the <span class="pkg">BDgraph</span> package, 
this function consists of several MCMC sampling algorithms for Bayesian model determination in undirected graphical models. 
To speed up the computations, the birth-death MCMC sampling algorithms are implemented in parallel using <span class="pkg">OpenMP</span> in <code>C++</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bdgraph( data, n = NULL, method = "ggm", algorithm = "bdmcmc", iter = 5000, 
         burnin = iter / 2, not.cont = NULL, g.prior = 0.2, df.prior = 3, 
         g.start = "empty", jump = NULL, save = FALSE, 
         cores = NULL, threshold = 1e-8, verbose = TRUE, nu = 1 )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bdgraph_+3A_data">data</code></td>
<td>

<p>there are two options: (1) an (<code class="reqn">n \times p</code>) <code>matrix</code> or a <code>data.frame</code> corresponding to the data, 
(2) an (<code class="reqn">p \times p</code>) covariance matrix as <code class="reqn">S=X'X</code> which <code class="reqn">X</code> is the data matrix 
(<code class="reqn">n</code> is the sample size and <code class="reqn">p</code> is the number of variables). 
It also could be an object of class &quot;<code>sim</code>&quot;, from function <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>.
The input matrix is automatically identified by checking the symmetry.
</p>
</td></tr>
<tr><td><code id="bdgraph_+3A_n">n</code></td>
<td>
<p>number of observations. It is needed if the &quot;<code>data</code>&quot; is a covariance matrix.</p>
</td></tr>
<tr><td><code id="bdgraph_+3A_method">method</code></td>
<td>

<p>character with two options &quot;<code>ggm</code>&quot; (default) and &quot;<code>gcgm</code>&quot;. 
Option &quot;<code>ggm</code>&quot; is for Gaussian graphical models based on Gaussianity assumption.
Option &quot;<code>gcgm</code>&quot; is for Gaussian copula graphical models for the data that not follow Gaussianity assumption (e.g. continuous non-Gaussian, count, or mixed dataset).
</p>
</td></tr>
<tr><td><code id="bdgraph_+3A_algorithm">algorithm</code></td>
<td>

<p>character with two options &quot;<code>bdmcmc</code>&quot; (default) and &quot;<code>rjmcmc</code>&quot;. 
Option &quot;<code>bdmcmc</code>&quot; is based on birth-death MCMC algorithm.
Option &quot;<code>rjmcmc</code>&quot; is based on reverible jump MCMC algorithm.
Option &quot;<code>bd-dmh</code>&quot; is based on birth-death MCMC algorithm using double Metropolis Hasting.
Option &quot;<code>rj-dmh</code>&quot; is based on reverible jump MCMC algorithm using double Metropolis Hasting.
</p>
</td></tr>
<tr><td><code id="bdgraph_+3A_iter">iter</code></td>
<td>
<p>number of iteration for the sampling algorithm.</p>
</td></tr>
<tr><td><code id="bdgraph_+3A_burnin">burnin</code></td>
<td>
<p>number of burn-in iteration for the sampling algorithm.</p>
</td></tr>
<tr><td><code id="bdgraph_+3A_not.cont">not.cont</code></td>
<td>
<p> for the case <code>method = "gcgm"</code>, a vector with binary values in which <code class="reqn">1</code> indicates not continuous variables. </p>
</td></tr>
<tr><td><code id="bdgraph_+3A_g.prior">g.prior</code></td>
<td>

<p>for determining the prior distribution of each edge in the graph. 
There are two options: a single value between <code class="reqn">0</code> and <code class="reqn">1</code> (e.g. <code class="reqn">0.5</code> as a noninformative prior) 
or an (<code class="reqn">p \times p</code>) matrix with elements between <code class="reqn">0</code> and <code class="reqn">1</code>.
</p>
</td></tr>
<tr><td><code id="bdgraph_+3A_df.prior">df.prior</code></td>
<td>

<p>degree of freedom for G-Wishart distribution, <code class="reqn">W_G(b,D)</code>, which is a prior distribution of the precision matrix.
</p>
</td></tr>
<tr><td><code id="bdgraph_+3A_g.start">g.start</code></td>
<td>

<p>corresponds to a starting point of the graph. It could be an (<code class="reqn">p \times p</code>) matrix, &quot;<code>empty</code>&quot; (default), or &quot;<code>full</code>&quot;. 
Option &quot;<code>empty</code>&quot; means the initial graph is an empty graph and &quot;<code>full</code>&quot; means a full graph. 
It also could be an object with <code>S3</code> class &quot;<code>bdgraph</code>&quot; of <code>R</code> package <code><a href="#topic+BDgraph">BDgraph</a></code> or the class <code>"ssgraph"</code> of <code>R</code> package <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code>; 
this option can be used to run the sampling algorithm from the last objects of previous run (see examples).     
</p>
</td></tr> 
<tr><td><code id="bdgraph_+3A_jump">jump</code></td>
<td>

<p>it is only for the BDMCMC algorithm (<code>algorithm</code> = &quot;<code>bdmcmc</code>&quot;).
It is for simultaneously updating multiple links at the same time to update graph in the BDMCMC algorithm.
</p>
</td></tr>	
<tr><td><code id="bdgraph_+3A_save">save</code></td>
<td>

<p>logical: if FALSE (default), the adjacency matrices are NOT saved. 
If TRUE, the adjacency matrices after burn-in are saved.
</p>
</td></tr>
<tr><td><code id="bdgraph_+3A_cores">cores</code></td>
<td>
<p> number of cores to use for parallel execution. 
The case <code>cores</code> = &quot;<code>all</code>&quot; means all CPU cores to use for parallel execution. 

</p>
</td></tr>	
<tr><td><code id="bdgraph_+3A_threshold">threshold</code></td>
<td>
<p> threshold value for the convergence of sampling algorithm from G-Wishart for the precision matrix.</p>
</td></tr>
<tr><td><code id="bdgraph_+3A_verbose">verbose</code></td>
<td>
<p> logical: if TRUE (default), report/print the MCMC running time. </p>
</td></tr>
<tr><td><code id="bdgraph_+3A_nu">nu</code></td>
<td>
<p> prior parameter for option <code>method</code> = &quot;<code>tgm</code>&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object with <code>S3</code> class &quot;<code>bdgraph</code>&quot; is returned:
</p>
<table>
<tr><td><code>p_links</code></td>
<td>
<p> upper triangular matrix which corresponds the estimated posterior probabilities of all possible links. </p>
</td></tr>
<tr><td><code>K_hat</code></td>
<td>
<p> posterior estimation of the precision matrix. </p>
</td></tr>
</table>
<p>For the case &quot;<code>save</code> = <code>TRUE</code>&quot; is returned:
</p>
<table>
<tr><td><code>sample_graphs</code></td>
<td>
<p> vector of strings which includes the adjacency matrices of visited graphs after burn-in.</p>
</td></tr>
<tr><td><code>graph_weights</code></td>
<td>
<p> vector which includes the waiting times of visited graphs after burn-in. </p>
</td></tr>
<tr><td><code>all_graphs</code></td>
<td>
<p>vector which includes the identity of the adjacency matrices for all iterations after burn-in. 
It is needed for monitoring the convergence of the BD-MCMC algorithm.</p>
</td></tr>
<tr><td><code>all_weights</code></td>
<td>
<p>vector which includes the waiting times for all iterations after burn-in. 
It is needed for monitoring the convergence of the BD-MCMC algorithm.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> and Ernst Wit </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>
<p>Mohammadi, A. and Wit, E. C. (2015). Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Bayesian Analysis</em>, 10(1):109-138, <a href="https://doi.org/10.1214/14-BA889">doi:10.1214/14-BA889</a>
</p>
<p>Mohammadi, R., Massam, H. and Letac, G. (2021). Accelerating Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Journal of the American Statistical Association</em>, <a href="https://doi.org/10.1080/01621459.2021.1996377">doi:10.1080/01621459.2021.1996377</a> 
</p>
<p>Mohammadi, A. et al (2017). Bayesian modelling of Dupuytren disease by using Gaussian copula graphical models, <em>Journal of the Royal Statistical Society: Series C</em>, 66(3):629-645, <a href="https://doi.org/10.1111/rssc.12171">doi:10.1111/rssc.12171</a>
</p>
<p>Dobra, A. and Mohammadi, R. (2018). Loglinear Model Selection and Human Mobility, <em>Annals of Applied Statistics</em>, 12(2):815-845, <a href="https://doi.org/10.1214/18-AOAS1164">doi:10.1214/18-AOAS1164</a>
</p>
<p>Mohammadi, A. and Dobra A. (2017). The <code>R</code> Package <span class="pkg">BDgraph</span> for Bayesian Structure Learning in Graphical Models, <em>ISBA Bulletin</em>, 24(4):11-16
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code>, <code><a href="#topic+bdgraph.dw">bdgraph.dw</a></code>, <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>, <code><a href="#topic+summary.bdgraph">summary.bdgraph</a></code>, <code><a href="#topic+compare">compare</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set.seed( 10 )

# - - Example 1

# Generating multivariate normal data from a 'random' graph
data.sim &lt;- bdgraph.sim( n = 100, p = 10, size = 15, vis = TRUE )
   
bdgraph.obj &lt;- bdgraph( data = data.sim, iter = 1000, save = TRUE )
  
summary( bdgraph.obj )

# Confusion Matrix
conf.mat( actual = data.sim, pred = bdgraph.obj )

conf.mat.plot( actual = data.sim, pred = bdgraph.obj )
   
# To compare our result with true graph
compare( bdgraph.obj, data.sim, main = c( "Target", "BDgraph" ), vis = T )

# Running algorithm with starting points from previous run
bdgraph.obj2 &lt;- bdgraph( data = data.sim, g.start = bdgraph.obj )
    
compare( list( bdgraph.obj, bdgraph.obj2 ), data.sim, 
         main = c( "Target", "Frist run", "Second run" ) )

# - - Example 2

# Generating mixed data from a 'scale-free' graph
data.sim &lt;- bdgraph.sim( n = 200, p = 7, type = "mixed", graph = "scale-free", vis = TRUE )
   
bdgraph.obj &lt;- bdgraph( data = data.sim, method = "gcgm" )
  
summary( bdgraph.obj )
   
compare( bdgraph.obj, data.sim, vis = T )

conf.mat( actual = data.sim, pred = bdgraph.obj )

conf.mat.plot( actual = data.sim, pred = bdgraph.obj )

## End(Not run)	  
</code></pre>

<hr>
<h2 id='bdgraph.dw'>	Search algorithm for Gaussian copula graphical models for count data </h2><span id='topic+bdgraph.dw'></span>

<h3>Description</h3>

<p>This function consists of several sampling algorithms for Bayesian structure learning in undirected graphical models for count data. It is based on Gaussian copula graphical models with discrete Weibull distributed marginals.
To speed up the computations, the birth-death MCMC sampling algorithms are implemented in parallel using <span class="pkg">OpenMP</span> in <code>C++</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bdgraph.dw( data, x = NULL, formula = y ~ ., 
            n = NULL, algorithm = "bdmcmc", iter = 5000, 
            burnin = iter / 2, g.prior = 0.2, df.prior = 3,
            ZI = FALSE, iter_bdw = 5000,
            g.start = "empty", jump = NULL, save = FALSE, 
            q = NULL, beta = NULL, pii = NULL,
            cores = NULL, threshold = 1e-8, verbose = TRUE )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bdgraph.dw_+3A_data">data</code></td>
<td>

<p>(<code class="reqn">n \times p</code>) <code>matrix</code> or a <code>data.frame</code> corresponding to the data on the <code>p</code> nodes of the graph. It can also be an object of class &quot;<code>sim</code>&quot;, from the function <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>.
</p>
</td></tr>
<tr><td><code id="bdgraph.dw_+3A_x">x</code></td>
<td>
<p> (<code class="reqn">n \times k</code>) <code>matrix</code> or a <code>data.frame</code> corresponding to the predictors. </p>
</td></tr>
<tr><td><code id="bdgraph.dw_+3A_formula">formula</code></td>
<td>
<p> object of class <a href="stats.html#topic+formula">formula</a> as a symbolic description of the model for linking each node to the predictors. For the case of <code>data.frame</code>, it is taken as the model frame (see <code><a href="stats.html#topic+model.frame">model.frame</a>)</code>.</p>
</td></tr>
<tr><td><code id="bdgraph.dw_+3A_n">n</code></td>
<td>
<p>number of observations. It is needed if the &quot;<code>data</code>&quot; is a covariance matrix.</p>
</td></tr>
<tr><td><code id="bdgraph.dw_+3A_algorithm">algorithm</code></td>
<td>

<p>character with two options &quot;<code>bdmcmc</code>&quot; (default) and &quot;<code>rjmcmc</code>&quot;. 
Option &quot;<code>bdmcmc</code>&quot; is based on a birth-death MCMC algorithm.
Option &quot;<code>rjmcmc</code>&quot; is based on a reversible jump MCMC algorithm.
</p>
</td></tr>
<tr><td><code id="bdgraph.dw_+3A_iter">iter</code></td>
<td>
<p>number of iterations for the sampling algorithm for graph learning.</p>
</td></tr>
<tr><td><code id="bdgraph.dw_+3A_burnin">burnin</code></td>
<td>
<p>number of burn-in iterations for the sampling algorithm for graph learning.</p>
</td></tr>
<tr><td><code id="bdgraph.dw_+3A_g.prior">g.prior</code></td>
<td>

<p>for determining the prior distribution of each edge in the graph. 
There are two options: a single value between <code class="reqn">0</code> and <code class="reqn">1</code> (e.g. <code class="reqn">0.5</code> as a noninformative prior) 
or a (<code class="reqn">p \times p</code>) matrix with elements between <code class="reqn">0</code> and <code class="reqn">1</code>.
</p>
</td></tr>
<tr><td><code id="bdgraph.dw_+3A_df.prior">df.prior</code></td>
<td>
<p>degree of freedom for G-Wishart distribution, <code class="reqn">W_G(b,D)</code>, which is a prior distribution for the precision matrix.</p>
</td></tr>
<tr><td><code id="bdgraph.dw_+3A_zi">ZI</code></td>
<td>
<p>logical. If FALSE (default), the conditional distribution of each response variable is assumed to be Discrete Weibull given the predictors <code>x</code>. If TRUE, a zero-inflated model will be applied to each response. <code>ZI</code> can be passed also as a vector, in order to specify which of the (<code>p</code> variables) should be fitted with zero-inflation (TRUE) or not (FALSE).
</p>
</td></tr>
<tr><td><code id="bdgraph.dw_+3A_iter_bdw">iter_bdw</code></td>
<td>
<p>number of iterations for the sampling algorithm to estimate the regression parameters for the Discrete Weibull distribution. It is passed to                     the <code><a href="#topic+bdw.reg">bdw.reg</a></code> function. </p>
</td></tr>
<tr><td><code id="bdgraph.dw_+3A_g.start">g.start</code></td>
<td>

<p>corresponds to a starting point of the graph. It could be an (<code class="reqn">p \times p</code>) matrix, &quot;<code>empty</code>&quot; (default), or &quot;<code>full</code>&quot;. 
Option &quot;<code>empty</code>&quot; means that the initial graph is an empty graph and &quot;<code>full</code>&quot; means a full graph. 
It also could be an object with <code>S3</code> class &quot;<code>bdgraph</code>&quot; of <code>R</code> package <code><a href="#topic+BDgraph">BDgraph</a></code> or the class <code>"ssgraph"</code> of <code>R</code> package <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code>; 
this option can be used to run the sampling algorithm from the last objects of the previous run (see examples).     
</p>
</td></tr> 
<tr><td><code id="bdgraph.dw_+3A_jump">jump</code></td>
<td>

<p>it is only for the BDMCMC algorithm (<code>algorithm</code> = &quot;<code>bdmcmc</code>&quot;).
It is for simultaneously updating multiple links at the same time while updating the graph in the BDMCMC algorithm.
</p>
</td></tr>	
<tr><td><code id="bdgraph.dw_+3A_save">save</code></td>
<td>

<p>logical: if FALSE (default), the adjacency matrices are NOT saved. 
If TRUE, the adjacency matrices  after burn-in are saved.
</p>
</td></tr>
<tr><td><code id="bdgraph.dw_+3A_q">q</code>, <code id="bdgraph.dw_+3A_beta">beta</code></td>
<td>
<p> parameters of the discrete Weibull distribution used for the marginals. They should be given either as a (<code class="reqn">n \times p</code>) <code>matrix</code> (if covariates are present) or as a vector (if covariates are not present). If NULL (default), these parameters are estimated by the <code><a href="#topic+bdw.reg">bdw.reg</a></code> function.</p>
</td></tr>
<tr><td><code id="bdgraph.dw_+3A_pii">pii</code></td>
<td>
<p> vector of zero-inflation parameters of the zero-inflated discrete Weibull distributions used for the marginals. If NULL (default), this parameter is estimated by the <code><a href="#topic+bdw.reg">bdw.reg</a></code> function when <code>ZI = TRUE</code>. </p>
</td></tr>
<tr><td><code id="bdgraph.dw_+3A_cores">cores</code></td>
<td>
<p> number of cores to use for parallel execution. 
The case <code>cores</code> = &quot;<code>all</code>&quot; means all CPU cores to use for parallel execution. 

</p>
</td></tr>	
<tr><td><code id="bdgraph.dw_+3A_threshold">threshold</code></td>
<td>
<p> threshold value for the convergence of the sampling algorithm from G-Wishart for the precision matrix.</p>
</td></tr>
<tr><td><code id="bdgraph.dw_+3A_verbose">verbose</code></td>
<td>
<p> logical: if TRUE (default), report/print the MCMC running time. </p>
</td></tr>	
</table>


<h3>Value</h3>

<p>An object with <code>S3</code> class &quot;<code>bdgraph</code>&quot; is returned, containing:
</p>
<table>
<tr><td><code>p_links</code></td>
<td>
<p> upper triangular matrix corresponding to the estimated posterior probabilities of all possible links. </p>
</td></tr>
<tr><td><code>K_hat</code></td>
<td>
<p> posterior estimation of the precision matrix. </p>
</td></tr>
<tr><td><code>sample_marginals</code></td>
<td>
<p> posterior samples of the regression coefficients of the marginal distributions. </p>
</td></tr>
</table>
<p>For the case &quot;<code>save</code> = <code>TRUE</code>&quot;, the code returns:
</p>
<table>
<tr><td><code>sample_graphs</code></td>
<td>
<p> vector of strings which includes the adjacency matrices of the graphs visited after burn-in.</p>
</td></tr>
<tr><td><code>graph_weights</code></td>
<td>
<p> vector which includes the waiting times of the graphs visited after burn-in. </p>
</td></tr>
<tr><td><code>all_graphs</code></td>
<td>
<p>vector which includes the identity of the adjacency matrices for all iterations after burn-in. 
It is needed for monitoring the convergence of the BDMCMC algorithm.</p>
</td></tr>
<tr><td><code>all_weights</code></td>
<td>
<p>vector which includes the waiting times for all iterations after burn-in. 
It is needed for monitoring the convergence of the BDMCMC algorithm.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a>, Veronica Vinciotti, and Pariya Behrouzi </p>


<h3>References</h3>

<p>Vinciotti, V., Behrouzi, P., and Mohammadi, R. (2022) Bayesian structural learning of microbiota systems from count metagenomic data, <em>arXiv preprint</em>, <a href="https://doi.org/10.48550/arXiv.2203.10118">doi:10.48550/arXiv.2203.10118</a>
</p>
<p>Peluso, A., Vinciotti, V., and Yu, K. (2018) Discrete Weibull generalized additive model: an application to count fertility, <em>Journal of the Royal Statistical Society: Series C</em>, 68(3):565-583, <a href="https://doi.org/10.1111/rssc.12311">doi:10.1111/rssc.12311</a>  
</p>
<p>Haselimashhadi, H., Vinciotti, V., and Yu, K. (2018) A novel Bayesian regression model for counts with an application to health data, <em>Journal of Applied Statistics,</em> 45(6):1085-1105, <a href="https://doi.org/10.1080/02664763.2017.1342782">doi:10.1080/02664763.2017.1342782</a>
</p>
<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a>  
</p>
<p>Mohammadi, A. and Wit, E. C. (2015). Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Bayesian Analysis</em>, 10(1):109-138, <a href="https://doi.org/10.1214/14-BA889">doi:10.1214/14-BA889</a>
</p>
<p>Mohammadi, A. et al (2017). Bayesian modelling of Dupuytren disease by using Gaussian copula graphical models, <em>Journal of the Royal Statistical Society: Series C</em>, 66(3):629-645, <a href="https://doi.org/10.1111/rssc.12171">doi:10.1111/rssc.12171</a>
</p>
<p>Mohammadi, R., Massam, H. and Letac, G. (2021). Accelerating Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Journal of the American Statistical Association</em>, <a href="https://doi.org/10.1080/01621459.2021.1996377">doi:10.1080/01621459.2021.1996377</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+bdgraph">bdgraph</a></code>, <a href="#topic+bdgraph.mpl">bdgraph.mpl</a>, <a href="#topic+bdw.reg">bdw.reg</a>, <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>, <code><a href="#topic+summary.bdgraph">summary.bdgraph</a></code>, <code><a href="#topic+compare">compare</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# - - Example 1

# Generating multivariate Discrete Weibull data based on 'random' graph
data.sim &lt;- bdgraph.sim( n = 100, p = 10, type = "dw", vis = TRUE )

bdgraph.obj &lt;- bdgraph.dw( data = data.sim, iter = 5000 )

summary( bdgraph.obj )

# To compare the result with true graph
compare( bdgraph.obj, data.sim, main = c( "Target", "BDgraph" ), vis = TRUE )

# - - Example 2

# Generating multivariate Discrete Weibull data based on a 'scale-free' graph
data.sim &lt;- bdgraph.sim( n = 100, p = 10, type = "dw", graph = "scale-free", vis = TRUE )

bdgraph.obj &lt;- bdgraph.dw( data = data.sim, iter = 10000 )

summary( bdgraph.obj )

compare( bdgraph.obj, data.sim, main = c( "Target", "BDgraph" ), vis = TRUE )

## End(Not run)	  
</code></pre>

<hr>
<h2 id='bdgraph.mpl'>	Search algorithm in graphical models using marginal pseudo-likehlihood </h2><span id='topic+bdgraph.mpl'></span>

<h3>Description</h3>

<p>This function consists of several sampling algorithms for Bayesian model determination in undirected graphical models based on mariginal pseudo-likelihood. 
To speed up the computations, the birth-death MCMC sampling algorithms are implemented in parallel using <span class="pkg">OpenMP</span> in <code>C++</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bdgraph.mpl( data, n = NULL, method = "ggm", transfer = TRUE, 
             algorithm = "bdmcmc", iter = 5000, burnin = iter / 2, 
             g.prior = 0.2, g.start = "empty", 
             jump = NULL, alpha = 0.5, save = FALSE, 
             cores = NULL, operator = "or", verbose = TRUE )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bdgraph.mpl_+3A_data">data</code></td>
<td>

<p>there are two options: (1) an (<code class="reqn">n \times p</code>) <code>matrix</code> or a <code>data.frame</code> corresponding to the data, 
(2) an (<code class="reqn">p \times p</code>) covariance matrix as <code class="reqn">S=X'X</code> which <code class="reqn">X</code> is the data matrix 
(<code class="reqn">n</code> is the sample size and <code class="reqn">p</code> is the number of variables). 
It also could be an object of class &quot;<code>sim</code>&quot;, from function <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>.
The input matrix is automatically identified by checking the symmetry.
</p>
</td></tr>
<tr><td><code id="bdgraph.mpl_+3A_n">n</code></td>
<td>
<p>number of observations. It is needed if the &quot;<code>data</code>&quot; is a covariance matrix.</p>
</td></tr>
<tr><td><code id="bdgraph.mpl_+3A_method">method</code></td>
<td>

<p>character with two options &quot;<code>ggm</code>&quot; (default), &quot;<code>dgm</code>&quot; and &quot;<code>dgm-binary</code>&quot;. 
Option &quot;<code>ggm</code>&quot; is for Gaussian graphical models based on Gaussianity assumption.
Option &quot;<code>dgm</code>&quot; is for discrete graphical models for the count data.
Option &quot;<code>dgm-binary</code>&quot; is for discrete graphical models for the data that are binary.
</p>
</td></tr>
<tr><td><code id="bdgraph.mpl_+3A_transfer">transfer</code></td>
<td>
<p> for only <code>'count'</code> data which <code>method</code> = &quot;<code>dgm</code>&quot; or <code>method</code> = &quot;<code>dgm-binary</code>&quot;. </p>
</td></tr>
<tr><td><code id="bdgraph.mpl_+3A_algorithm">algorithm</code></td>
<td>

<p>character with two options &quot;<code>bdmcmc</code>&quot; (default) and &quot;<code>rjmcmc</code>&quot;. 
Option &quot;<code>bdmcmc</code>&quot; is based on birth-death MCMC algorithm.
Option &quot;<code>rjmcmc</code>&quot; is based on reverible jump MCMC algorithm.
Option <code>"hc"</code>     is based on hill-climbing algorithm; this algorithm is only for count data which <code>method</code> = &quot;<code>dgm</code>&quot; or <code>method</code> = &quot;<code>dgm-binary</code>&quot;.
</p>
</td></tr>
<tr><td><code id="bdgraph.mpl_+3A_iter">iter</code></td>
<td>
<p>   number of iteration for the sampling algorithm. </p>
</td></tr>
<tr><td><code id="bdgraph.mpl_+3A_burnin">burnin</code></td>
<td>
<p> number of burn-in iteration for the sampling algorithm. </p>
</td></tr>
<tr><td><code id="bdgraph.mpl_+3A_g.prior">g.prior</code></td>
<td>

<p>for determining the prior distribution of each edge in the graph. 
There are two options: a single value between <code class="reqn">0</code> and <code class="reqn">1</code> (e.g. <code class="reqn">0.5</code> as a noninformative prior) 
or an (<code class="reqn">p \times p</code>) matrix with elements between <code class="reqn">0</code> and <code class="reqn">1</code>.
</p>
</td></tr>
<tr><td><code id="bdgraph.mpl_+3A_g.start">g.start</code></td>
<td>

<p>corresponds to a starting point of the graph. It could be an (<code class="reqn">p \times p</code>) matrix, &quot;<code>empty</code>&quot; (default), or &quot;<code>full</code>&quot;. 
Option &quot;<code>empty</code>&quot; means the initial graph is an empty graph and &quot;<code>full</code>&quot; means a full graph. 
It also could be an object with <code>S3</code> class &quot;<code>bdgraph</code>&quot; of <code>R</code> package <code><a href="#topic+BDgraph">BDgraph</a></code> or the class <code>"ssgraph"</code> of <code>R</code> package <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code>; 
this option can be used to run the sampling algorithm from the last objects of previous run (see examples).     
</p>
</td></tr> 
<tr><td><code id="bdgraph.mpl_+3A_jump">jump</code></td>
<td>

<p>it is only for the BDMCMC algorithm (<code>algorithm</code> = &quot;<code>bdmcmc</code>&quot;).
It is for simultaneously updating multiple links at the same time to update graph in the BDMCMC algorithm.
</p>
</td></tr>	
<tr><td><code id="bdgraph.mpl_+3A_alpha">alpha</code></td>
<td>
<p> value of the hyper parameter of Dirichlet, which is a prior distribution. </p>
</td></tr>
<tr><td><code id="bdgraph.mpl_+3A_save">save</code></td>
<td>

<p>logical: if FALSE (default), the adjacency matrices are NOT saved. 
If TRUE, the adjacency matrices after burn-in are saved.
</p>
</td></tr>
<tr><td><code id="bdgraph.mpl_+3A_cores">cores</code></td>
<td>
<p> number of cores to use for parallel execution. 
The case <code>cores</code> = &quot;<code>all</code>&quot; means all CPU cores to use for parallel execution. 

</p>
</td></tr>	
<tr><td><code id="bdgraph.mpl_+3A_operator">operator</code></td>
<td>
<p> character with two options &quot;<code>or</code>&quot; (default) and  &quot;<code>and</code>&quot;. It is for hill-climbing algorithm. </p>
</td></tr>	   	
<tr><td><code id="bdgraph.mpl_+3A_verbose">verbose</code></td>
<td>
<p> logical: if TRUE (default), report/print the MCMC running time. </p>
</td></tr>	
</table>


<h3>Value</h3>

<p>An object with <code>S3</code> class &quot;<code>bdgraph</code>&quot; is returned:
</p>
<table>
<tr><td><code>p_links</code></td>
<td>
<p> upper triangular matrix which corresponds the estimated posterior probabilities of all possible links. </p>
</td></tr>
</table>
<p>For the case &quot;<code>save</code> = <code>TRUE</code>&quot; is returned:
</p>
<table>
<tr><td><code>sample_graphs</code></td>
<td>
<p> vector of strings which includes the adjacency matrices of visited graphs after burn-in.</p>
</td></tr>
<tr><td><code>graph_weights</code></td>
<td>
<p> vector which includes the waiting times of visited graphs after burn-in. </p>
</td></tr>
<tr><td><code>all_graphs</code></td>
<td>
<p>vector which includes the identity of the adjacency matrices for all iterations after burn-in. 
It is needed for monitoring the convergence of the BD-MCMC algorithm.</p>
</td></tr>
<tr><td><code>all_weights</code></td>
<td>
<p>vector which includes the waiting times for all iterations after burn-in. It is needed for monitoring the convergence of the BD-MCMC algorithm.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a>, Adrian Dobra, and Johan Pensar </p>


<h3>References</h3>

<p>Dobra, A. and Mohammadi, R. (2018). Loglinear Model Selection and Human Mobility, <em>Annals of Applied Statistics</em>, 12(2):815-845, <a href="https://doi.org/10.1214/18-AOAS1164">doi:10.1214/18-AOAS1164</a>
</p>
<p>Mohammadi, A. and Wit, E. C. (2015). Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Bayesian Analysis</em>, 10(1):109-138, <a href="https://doi.org/10.1214/14-BA889">doi:10.1214/14-BA889</a>
</p>
<p>Mohammadi, A. and Dobra, A. (2017). The <code>R</code> Package <span class="pkg">BDgraph</span> for Bayesian Structure Learning in Graphical Models, <em>ISBA Bulletin</em>, 24(4):11-16
</p>
<p>Pensar, J. et al (2017) Marginal pseudo-likelihood learning of discrete Markov network structures, <em>Bayesian Analysis</em>, 12(4):1195-215, <a href="https://doi.org/10.1214/16-BA1032">doi:10.1214/16-BA1032</a>
</p>
<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+bdgraph.dw">bdgraph.dw</a></code>, <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>, <code><a href="#topic+summary.bdgraph">summary.bdgraph</a></code>, <code><a href="#topic+compare">compare</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'># Generating multivariate normal data from a 'random' graph
data.sim &lt;- bdgraph.sim( n = 70, p = 5, size = 7, vis = TRUE )
   
bdgraph.obj &lt;- bdgraph.mpl( data = data.sim, iter = 500 )
  
summary( bdgraph.obj )
   
# To compare the result with true graph
compare( bdgraph.obj, data.sim, main = c( "Target", "BDgraph" ) )
</code></pre>

<hr>
<h2 id='bdgraph.npn'>Nonparametric transfer</h2><span id='topic+bdgraph.npn'></span>

<h3>Description</h3>

<p> Transfers non-Gaussian data to Gaussian.  </p>


<h3>Usage</h3>

<pre><code class='language-R'> bdgraph.npn( data, npn = "shrinkage", npn.thresh = NULL ) </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bdgraph.npn_+3A_data">data</code></td>
<td>
<p> (<code class="reqn">n \times p</code>) <code>matrix</code> or a <code>data.frame</code> corresponding to the data (<code class="reqn">n</code> is the sample size and <code class="reqn">p</code> is the number of variables). </p>
</td></tr>
<tr><td><code id="bdgraph.npn_+3A_npn">npn</code></td>
<td>
 
<p>character with three options <code>"shrinkage"</code> (default), <code>"truncation"</code>, and <code>"skeptic"</code>. Option <code>"shrinkage"</code> is for the 
shrunken transformation, option <code>"truncation"</code> is for the truncated transformation and option <code>"skeptic"</code> is for the 
non-paranormal skeptic transformation. For more details see references.
</p>
</td></tr>
<tr><td><code id="bdgraph.npn_+3A_npn.thresh">npn.thresh</code></td>
<td>
<p> truncation threshold; it is only for the truncated transformation (<code>npn=</code> <code>"truncation"</code>). 
The default value is <code class="reqn">1/(4n^{1/4} \sqrt{\pi \log(n)})</code>. </p>
</td></tr>
</table>


<h3>Value</h3>

<p> (<code class="reqn">n \times p</code>) matrix of transferred data, if <code>npn = "shrinkage" or "truncation"</code>, and 
a non-paranormal correlation (<code class="reqn">p \times p</code>) matrix, if <code>npn = "skeptic"</code>.  
</p>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> </p>


<h3>References</h3>

<p>Liu, H., et al (2012). High Dimensional Semiparametric Gaussian Copula Graphical Models, <em>Annals of Statistics</em>, 40(4):2293-2326
</p>
<p>Zhao, T. and Liu, H. (2012). The <span class="pkg">huge</span> Package for High-dimensional Undirected Graph Estimation in <code>R</code>, <em>Journal of Machine Learning Research</em>, 13:1059-1062
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>, <code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Generating multivariate normal data from a 'random' graph
data.sim &lt;- bdgraph.sim( n = 6, p = 4, size = 4 )

data     &lt;- ( data.sim $ data - 3 ) ^ 4
data
   
# Transfer the data by truncation 
bdgraph.npn( data, npn = "truncation" )
  
# Transfer the data by shrunken 
bdgraph.npn( data, npn = "shrunken" )
  
# Transfer the data by skeptic 
bdgraph.npn( data, npn = "skeptic" )

## End(Not run)
</code></pre>

<hr>
<h2 id='bdgraph.sim'> Graph data simulation </h2><span id='topic+bdgraph.sim'></span>

<h3>Description</h3>

<p>Simulating multivariate distributions with different types of underlying graph structures, including 
&quot;<code>random</code>&quot;, &quot;<code>cluster</code>&quot;, &quot;<code>smallworld</code>&quot;, &quot;<code>scale-free</code>&quot;, &quot;<code>lattice</code>&quot;, &quot;<code>hub</code>&quot;, &quot;<code>star</code>&quot;, &quot;<code>circle</code>&quot;, &quot;<code>AR(1)</code>&quot;, and &quot;<code>AR(2)</code>&quot;.
Based on the underlying graph structure, the function generates different types of <em>multivariate</em> data, including &quot;<em>Gaussian</em>&quot;, &quot;<em>non-Gaussian</em>&quot;, &quot;<em>categorical</em>&quot;, &quot;<em>pois</em>&quot; (Poisson), &quot;<em>nbinom</em>&quot; (negative binomial), &quot;<em>dweibull</em>&quot; (discrete Weibull), &quot;<em>binary</em>&quot;, &quot;<em>t</em>&quot; (t-distribution), &quot;<em>alternative-t</em>&quot;, or &quot;<em>mixed</em>&quot; data. 
This function can be used also for simulating only graphs by setting the option <code>n=0</code> (default). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bdgraph.sim( p = 10, graph = "random", n = 0, type = "Gaussian", prob = 0.2, 
             size = NULL, mean = 0, class = NULL, cut = 4, b = 3,
             D = diag( p ), K = NULL, sigma = NULL, 
             q = exp(-1), beta = 1, vis = FALSE, rewire = 0.05,
             range.mu = c( 3, 5 ), range.dispersion = c( 0.01, 0.1 ), nu = 1 )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bdgraph.sim_+3A_p">p</code></td>
<td>
<p>number of variables (nodes).</p>
</td></tr>
<tr><td><code id="bdgraph.sim_+3A_graph">graph</code></td>
<td>
<p>graph structure with options 
&quot;<code>random</code>&quot;, &quot;<code>cluster</code>&quot;, &quot;<code>smallworld</code>&quot;, &quot;<code>scale-free</code>&quot;, &quot;<code>lattice</code>&quot;, &quot;<code>hub</code>&quot;, &quot;<code>star</code>&quot;, &quot;<code>circle</code>&quot;, 
&quot;<code>AR(1)</code>&quot;, and &quot;<code>AR(2)</code>&quot;.
It could also be an adjacency matrix corresponding to a graph structure (an upper triangular matrix in which 
<code class="reqn">g_{ij}=1</code> if there is a link between nodes <code class="reqn">i</code> and <code class="reqn">j</code>, otherwise <code class="reqn">g_{ij}=0</code>). 
</p>
</td></tr>
<tr><td><code id="bdgraph.sim_+3A_n">n</code></td>
<td>
<p>number of samples required. Note that for the case <code>n = 0</code>, only the graph is generated. </p>
</td></tr>
<tr><td><code id="bdgraph.sim_+3A_type">type</code></td>
<td>
<p>type of data with options &quot;<code>Gaussian</code>&quot; (default), &quot;<code>non-Gaussian</code>&quot;, &quot;<code>categorical</code>&quot;, &quot;<code>pois</code>&quot;, &quot;<code>nbinom</code>&quot;, &quot;<code>dweibull</code>&quot;, &quot;<code>binary</code>&quot;, &quot;<code>mixed</code>&quot;, &quot;<code>t</code>&quot;, and &quot;<code>alternative-t</code>&quot;.    
For the option &quot;<code>Gaussian</code>&quot;, data are generated from a multivariate normal distribution.
For the option &quot;<code>non-Gaussian</code>&quot;, data are transfered from a multivariate normal distribution to a continuous multivariate non-Gaussian distribution via Exponential marginals.
For the option &quot;<code>categorical</code>&quot;, data are transfered from a multivariate normal distribution to multivariate 'categorical' data.
For the option &quot;<code>pois</code>&quot;, data are transfered from a multivariate normal distribution to a multivariate Poisson distribution.	  
For the option &quot;<code>nbinom</code>&quot;, data are transfered from a multivariate normal distribution to a multivariate Negative Binomial distribution.	  
For the option &quot;<code>dweibull</code>&quot;, data are transfered from a multivariate normal distribution to a multivariate discrete Weibull distribution with parameters <code>q</code> and <code>beta</code>.	  
For the option &quot;<code>binary</code>&quot;, data are generated directly from the joint distribution, in this case <code class="reqn">p</code> must be less than <code class="reqn">17</code>. 
For the option &quot;<code>mixed</code>&quot;, data are transfered from a multivariate normal distribution to a mixture of 'categorical', 'non-Gaussian', 'binary' and 'Gaussian', respectively. 
</p>
</td></tr>
<tr><td><code id="bdgraph.sim_+3A_prob">prob</code></td>
<td>
<p> if <code>graph</code> = &quot;<code>random</code>&quot;, it is the probability that a pair of nodes has a link.</p>
</td></tr>
<tr><td><code id="bdgraph.sim_+3A_size">size</code></td>
<td>
<p>number of links in the true graph (graph size).</p>
</td></tr>
<tr><td><code id="bdgraph.sim_+3A_mean">mean</code></td>
<td>
<p>vector specifying the mean of the variables.</p>
</td></tr>
<tr><td><code id="bdgraph.sim_+3A_class">class</code></td>
<td>
<p> if <code>graph</code> = &quot;<code>cluster</code>&quot;, it is the number of classes. </p>
</td></tr>
<tr><td><code id="bdgraph.sim_+3A_cut">cut</code></td>
<td>
<p> if <code>type</code> = &quot;<code>categorical</code>&quot;, it is the number of categories for simulating 'categorical' data.</p>
</td></tr>
<tr><td><code id="bdgraph.sim_+3A_b">b</code></td>
<td>
<p>degree of freedom for G-Wishart distribution, <code class="reqn">W_G(b, D)</code>.</p>
</td></tr>
<tr><td><code id="bdgraph.sim_+3A_d">D</code></td>
<td>
<p>positive definite <code class="reqn">(p \times p)</code> &quot;scale&quot; matrix for G-Wishart distribution, <code class="reqn">W_G(b, D)</code>. The default is an identity matrix.</p>
</td></tr>
<tr><td><code id="bdgraph.sim_+3A_k">K</code></td>
<td>
<p>     if <code>graph</code> = &quot;<code>fixed</code>&quot;, it is a positive-definite symmetric matrix, corresponding to the true precision matrix. </p>
</td></tr>
<tr><td><code id="bdgraph.sim_+3A_sigma">sigma</code></td>
<td>
<p> if <code>graph</code> = &quot;<code>fixed</code>&quot;, it is a positive-definite symmetric matrix corresponding to the true covariance matrix.</p>
</td></tr>
<tr><td><code id="bdgraph.sim_+3A_q">q</code>, <code id="bdgraph.sim_+3A_beta">beta</code></td>
<td>
<p> if <code>type</code> = &quot;<code>dweibull</code>&quot;, they are the parameters of the discrete Weibull distribution with density 
</p>
<p style="text-align: center;"><code class="reqn">p( x, q, \beta ) = q^{x^{\beta}}-q^{(x+1)^{\beta}}, \quad \forall x = \{ 0, 1, 2, \ldots \}.</code>
</p>
<p> They can be given either as a vector of length p or as an  (<code class="reqn">n \times p</code>) matrix, e.g. if covariates are available and a regression model is used.</p>
</td></tr>
<tr><td><code id="bdgraph.sim_+3A_vis">vis</code></td>
<td>
<p>visualize the true graph structure.</p>
</td></tr>
<tr><td><code id="bdgraph.sim_+3A_rewire">rewire</code></td>
<td>
<p>rewiring probability for smallworld network. Must be between 0 and 1.</p>
</td></tr>
<tr><td><code id="bdgraph.sim_+3A_range.mu">range.mu</code>, <code id="bdgraph.sim_+3A_range.dispersion">range.dispersion</code></td>
<td>
<p>if <code>type</code> = &quot;<code>nbinom</code>&quot;, vector with two elements specifying the range of parameters for the Negative Binomial distribution.</p>
</td></tr>
<tr><td><code id="bdgraph.sim_+3A_nu">nu</code></td>
<td>
<p> if <code>type</code> = &quot;<code>t</code>&quot; or &quot;<code>alternative-t</code>&quot;, it is the parameter of the t distribution with density.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object with <code>S3</code> class &quot;<code>sim</code>&quot; is returned:
</p>
<table>
<tr><td><code>data</code></td>
<td>
<p>generated data as an (<code class="reqn">n \times p</code>) matrix.</p>
</td></tr>
<tr><td><code>sigma</code></td>
<td>
<p>covariance matrix of the generated data.</p>
</td></tr>
<tr><td><code>K</code></td>
<td>
<p>precision matrix of the generated data.</p>
</td></tr>
<tr><td><code>G</code></td>
<td>
<p>adjacency matrix corresponding to the true graph structure.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a>, Pariya Behrouzi, Veronica Vinciotti, Ernst Wit, and Alexander Christensen </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+graph.sim">graph.sim</a></code>, <code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Generating multivariate normal data from a 'random' graph
data.sim &lt;- bdgraph.sim( p = 10, n = 50, prob = 0.3, vis = TRUE )

print( data.sim )
     
# Generating multivariate normal data from a 'hub' graph
data.sim &lt;- bdgraph.sim( p = 6, n = 3, graph = "hub", vis = FALSE )

round( data.sim $ data, 2 )
     
# Generating mixed data from a 'hub' graph 
data.sim &lt;- bdgraph.sim( p = 8, n = 10, graph = "hub", type = "mixed" )

round( data.sim $ data, 2 )

# Generating only a 'scale-free' graph (with no data) 
graph.sim &lt;- bdgraph.sim( p = 8, graph = "scale-free" )

plot( graph.sim )

graph.sim $ G

## End(Not run)
</code></pre>

<hr>
<h2 id='bdw.reg'>	Bayesian estimation of (zero-inflated) Discrete Weibull regression </h2><span id='topic+bdw.reg'></span>

<h3>Description</h3>

<p>Bayesian estimation of the parameters for Discrete Weibull (DW) regression. The conditional distribution of the response given the predictors is assumed to be DW with parameters q and beta, dependent on the predictors, and,  with an additional parameter pi under zero inflation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bdw.reg( data, formula = NA, iter = 5000, burnin = NULL, 
         dist.q = dnorm, dist.beta = dnorm, 
         par.q = c( 0, 1 ), par.beta = c( 0, 1 ), par.pi = c( 1, 1 ), 
         initial.q = NULL, initial.beta = NULL, initial.pi = NULL, 
         ZI = FALSE, scale.proposal = NULL, adapt = TRUE, print = TRUE )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bdw.reg_+3A_data">data</code></td>
<td>
<p><code>data.frame</code> or <code>matrix</code> corresponding to the data, containing the variables in the model.</p>
</td></tr>
<tr><td><code id="bdw.reg_+3A_formula">formula</code></td>
<td>
<p> object of class <a href="stats.html#topic+formula">formula</a> as a symbolic description of the model to be fitted. For the case of <code>data.frame</code>, it is taken as the model frame (see <code><a href="stats.html#topic+model.frame">model.frame</a>)</code>.</p>
</td></tr>
<tr><td><code id="bdw.reg_+3A_iter">iter</code></td>
<td>
<p> number of iterations for the sampling algorithm.</p>
</td></tr>
<tr><td><code id="bdw.reg_+3A_burnin">burnin</code></td>
<td>
<p>number of burn-in iterations for the sampling algorithm.</p>
</td></tr>
<tr><td><code id="bdw.reg_+3A_dist.q">dist.q</code></td>
<td>
<p> Prior density for the regression coefficients associated to the parameter <code>q</code>. The default is a Normal distribution (<code>dnorm</code>). 
Any density function which has two parameters and can support the <code>log = TRUE</code> flag can be used, e.g. <code><a href="stats.html#topic+dnorm">dnorm</a></code>, <code><a href="stats.html#topic+dlnorm">dlnorm</a></code>, <code><a href="stats.html#topic+dunif">dunif</a></code> etc. 
</p>
</td></tr>
<tr><td><code id="bdw.reg_+3A_dist.beta">dist.beta</code></td>
<td>
<p> Prior density for the regression coefficients associated to the parameter <code>beta</code>. The default is a Normal distribution (<code><a href="stats.html#topic+dnorm">dnorm</a></code>). 
Any density function which has two parameters and can support the <code>log = TRUE</code> flag can be used, e.g. <code><a href="stats.html#topic+dnorm">dnorm</a></code>, <code><a href="stats.html#topic+dlnorm">dlnorm</a></code>, <code><a href="stats.html#topic+dunif">dunif</a></code> etc. 
</p>
</td></tr>
<tr><td><code id="bdw.reg_+3A_par.q">par.q</code></td>
<td>
<p> vector of length two corresponding to the parameters of <code>dist.q</code>. </p>
</td></tr>
<tr><td><code id="bdw.reg_+3A_par.beta">par.beta</code></td>
<td>
<p> vector of length two corresponding to the parameters of <code>dist.beta</code>. </p>
</td></tr>
<tr><td><code id="bdw.reg_+3A_par.pi">par.pi</code></td>
<td>
<p> vector of length two corresponding to the parameters of the <code><a href="base.html#topic+beta">beta</a></code> prior density on <code>pi</code>. </p>
</td></tr>
<tr><td><code id="bdw.reg_+3A_initial.q">initial.q</code>, <code id="bdw.reg_+3A_initial.beta">initial.beta</code>, <code id="bdw.reg_+3A_initial.pi">initial.pi</code></td>
<td>
<p> vector of initial values for the regression coefficients and for <code>pi</code> (if <code>ZI</code> = <code>TRUE</code>). </p>
</td></tr>
<tr><td><code id="bdw.reg_+3A_zi">ZI</code></td>
<td>
 
<p>logical: if FALSE (default), the conditional distribution of the response given the predictors is assumed to be DW with parameters <code>q</code> and <code>beta</code>. 
If TRUE, a zero-inflated DW distribution will be applied.
</p>
</td></tr>
<tr><td><code id="bdw.reg_+3A_scale.proposal">scale.proposal</code></td>
<td>
<p> scale of the proposal function. Setting to lower values results in an increase in the acceptance rate of the sampler. </p>
</td></tr>
<tr><td><code id="bdw.reg_+3A_adapt">adapt</code></td>
<td>
<p> logical: if TRUE (default), the proposals will be adapted. If FALSE, no adapting will be applied. </p>
</td></tr>
<tr><td><code id="bdw.reg_+3A_print">print</code></td>
<td>
<p> logical: if TRUE (default), tracing information is printed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The regression model uses a logit link function on <code>q</code> and a log link function on <code>beta</code>, the two parameters of a DW distribution, with probability mass function given by 
</p>
<p style="text-align: center;"><code class="reqn">
DW(y) = q^{y^\beta} - q^{(y+1)^\beta}, y = 0, 1, 2, \ldots
</code>
</p>

<p>For the case of zero inflation (<code>ZI</code> = <code>TRUE</code>), a zero-inflated DW is considered:
</p>
<p style="text-align: center;"><code class="reqn">
f(y) = ( 1 - pi) I(y = 0 ) + pi DW(y)
</code>
</p>

<p>where <code class="reqn">0 \leq pi \leq 1</code> and <code class="reqn">I(y = 0 )</code> is an indicator for the point mass at zero for the response <code>y</code>. 
</p>


<h3>Value</h3>

<table>
<tr><td><code>sample</code></td>
<td>
<p> MCMC samples </p>
</td></tr>
<tr><td><code>q.est</code></td>
<td>
<p> posterior estimates of <code>q</code> </p>
</td></tr>
<tr><td><code>beta.est</code></td>
<td>
<p> posterior estimates of <code>beta</code> </p>
</td></tr>
<tr><td><code>pi.est</code></td>
<td>
<p> posterior estimates of <code>pi</code> </p>
</td></tr>
<tr><td><code>accept.rate</code></td>
<td>
<p> acceptance rate of the MCMC algorithm </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Veronica Vinciotti, Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a>, and Pariya Behrouzi </p>


<h3>References</h3>

<p>Vinciotti, V., Behrouzi, P., and Mohammadi, R. (2022) Bayesian structural learning of microbiota systems from count metagenomic data, <em>arXiv preprint</em>, <a href="https://doi.org/10.48550/arXiv.2203.10118">doi:10.48550/arXiv.2203.10118</a>
</p>
<p>Peluso, A., Vinciotti, V., and Yu, K. (2018) Discrete Weibull generalized additive model: an application to count fertility, <em>Journal of the Royal Statistical Society: Series C</em>, 68(3):565-583, <a href="https://doi.org/10.1111/rssc.12311">doi:10.1111/rssc.12311</a>  
</p>
<p>Haselimashhadi, H., Vinciotti, V. and Yu, K. (2018) A novel Bayesian regression model for counts with an application to health data, <em>Journal of Applied Statistics,</em> 45(6):1085-1105, <a href="https://doi.org/10.1080/02664763.2017.1342782">doi:10.1080/02664763.2017.1342782</a>
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+bdgraph.dw">bdgraph.dw</a></code>, <a href="#topic+bdgraph">bdgraph</a>, <a href="#topic+ddweibull">ddweibull</a>, <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# - - Example 1

q    = 0.6
beta = 1.1
n    = 500

y = BDgraph::rdweibull( n = n, q = q, beta = beta )

output = bdw.reg( data = y, y ~ ., iter = 5000 )

output $ q.est
output $ beta.est

traceplot( output $ sample[ , 1 ], acf = T, pacf = T )
traceplot( output $ sample[ , 2 ], acf = T, pacf = T )

# - - Example 2

q    = 0.6
beta = 1.1
pii  = 0.8
n    = 500

y_dw = BDgraph::rdweibull( n = n, q = q, beta = beta )
z = rbinom( n = n, size = 1, prob = pii ) 
y = z * y_dw

output = bdw.reg( data = y, iter = 5000, ZI = TRUE )

output $ q.est
output $ beta.est
output $ pi.est

traceplot( output $ sample[ , 1 ], acf = T, pacf = T )
traceplot( output $ sample[ , 2 ], acf = T, pacf = T )
traceplot( output $ sample[ , 3 ], acf = T, pacf = T )

# - - Example 3

theta.q    = c( 0.1, -0.1, 0.34 )  # true parameter
theta.beta = c( 0.1, -.15, 0.5  )  # true parameter

n  = 500

x1 = runif( n = n, min = 0, max = 1.5 )
x2 = runif( n = n, min = 0, max = 1.5 )

reg_q = theta.q[ 1 ] + x1 * theta.q[ 2 ] + x2 * theta.q[ 3 ]
q     = 1 / ( 1 + exp( - reg_q ) )

reg_beta = theta.beta[ 1 ] + x1 * theta.beta[ 2 ] + x2 * theta.beta[ 3 ]
beta     = exp( reg_beta )

y = BDgraph::rdweibull( n = n, q = q, beta = beta )

data = data.frame( x1, x2, y ) 

output = bdw.reg( data, y ~. , iter = 5000 )

# - - Example 4

theta.q    = c( 1, -1, 0.8 )  # true parameter
theta.beta = c( 1, -1, 0.3 )  # true parameter
pii = 0.8

n  = 500

x1 = runif( n = n, min = 0, max = 1.5 )
x2 = runif( n = n, min = 0, max = 1.5 )

reg_q = theta.q[ 1 ] + x1 * theta.q[ 2 ] + x2 * theta.q[ 3 ]
q     = 1 / ( 1 + exp( - reg_q ) )

reg_beta = theta.beta[ 1 ] + x1 * theta.beta[ 2 ] + x2 * theta.beta[ 3 ]
beta     = exp( reg_beta )

y_dw = BDgraph::rdweibull( n = n, q = q, beta = beta )
z    = rbinom( n = n, size = 1, prob = pii ) 
y    = z * y_dw

data = data.frame( x1, x2, y ) 

output = bdw.reg( data, y ~. , iter = 5000 )

## End(Not run)	  
</code></pre>

<hr>
<h2 id='bf'> Bayes factor for two graphs </h2><span id='topic+bf'></span>

<h3>Description</h3>

<p> Compute the Bayes factor for two graph structures. </p>


<h3>Usage</h3>

<pre><code class='language-R'> 
    bf( num, den, bdgraph.obj, log = TRUE ) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bf_+3A_num">num</code>, <code id="bf_+3A_den">den</code></td>
<td>

<p>adjacency matrix corresponding to the true graph structure in which <code class="reqn">a_{ij}=1</code> if there is a link between notes <code class="reqn">i</code> and <code class="reqn">j</code>, otherwise <code class="reqn">a_{ij}=0</code>.
It can be an object with <code>S3</code> class <code>"graph"</code> from function <code><a href="#topic+graph.sim">graph.sim</a></code>.
It can be an object with <code>S3</code> class &quot;<code>sim</code>&quot; from function <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>.
</p>
</td></tr>
<tr><td><code id="bf_+3A_bdgraph.obj">bdgraph.obj</code></td>
<td>
<p>object of <code>S3</code> class &quot;<code>bdgraph</code>&quot;, from function <code><a href="#topic+bdgraph">bdgraph</a></code>. 
It also can be an object of <code>S3</code> class <code>"ssgraph"</code>, from the function <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code> of <code>R</code> package <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code>.
</p>
</td></tr>
<tr><td><code id="bf_+3A_log">log</code></td>
<td>
<p> character value. If TRUE the Bayes factor is given as log(BF). </p>
</td></tr>
</table>


<h3>Value</h3>

<p> single numeric value, the Bayes factor of the two graph structures <code>num</code> and <code>den</code>.</p>


<h3>Author(s)</h3>

<p>Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a></p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code>, <code><a href="#topic+compare">compare</a></code>, <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>    ## Not run: 
        # Generating multivariate normal data from a 'circle' graph
        data.sim &lt;- bdgraph.sim( n = 50, p = 6, graph = "circle", vis = TRUE )

        # Running sampling algorithm
        bdgraph.obj &lt;- bdgraph( data = data.sim )

        graph_1 &lt;- graph.sim( p = 6, vis = TRUE )
        
        graph_2 &lt;- graph.sim( p = 6, vis = TRUE )

        bf( num = graph_1, den = graph_2, bdgraph.obj = bdgraph.obj )
    
## End(Not run)
</code></pre>

<hr>
<h2 id='compare'> Graph structure comparison </h2><span id='topic+compare'></span>

<h3>Description</h3>

<p> This function provides several measures to assess the performance of the graphical structure learning. </p>


<h3>Usage</h3>

<pre><code class='language-R'> compare( pred, actual, main = NULL, vis = FALSE ) </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compare_+3A_pred">pred</code></td>
<td>

<p>adjacency matrix corresponding to an estimated graph. 
It can be an object with <code>S3</code> class &quot;<code>bdgraph</code>&quot; from function <code><a href="#topic+bdgraph">bdgraph</a></code>. 
It can be an object of <code>S3</code> class <code>"ssgraph"</code>, from the function <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code> of <code>R</code> package <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code>.
It can be an object of <code>S3</code> class <code>"select"</code>, from the function <code><a href="huge.html#topic+huge.select">huge.select</a></code> of <code>R</code> package <code><a href="huge.html#topic+huge">huge</a></code>.
It also can be a <code>list</code> of above objects for comparing two or more different approaches.
</p>
</td></tr>
<tr><td><code id="compare_+3A_actual">actual</code></td>
<td>

<p>adjacency matrix corresponding to the true graph structure in which <code class="reqn">a_{ij}=1</code> if there is a link between notes <code class="reqn">i</code> and <code class="reqn">j</code>, otherwise <code class="reqn">a_{ij}=0</code>.
It can be an object with <code>S3</code> class &quot;<code>sim</code>&quot; from function <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>.
It can be an object with <code>S3</code> class <code>"graph"</code> from function <code><a href="#topic+graph.sim">graph.sim</a></code>.
</p>
</td></tr>
<tr><td><code id="compare_+3A_main">main</code></td>
<td>
<p> character vector giving the names for the result table. </p>
</td></tr>
<tr><td><code id="compare_+3A_vis">vis</code></td>
<td>
<p> logical: if TRUE, visualize the true graph and estimated graph structures. </p>
</td></tr> 
</table>


<h3>Value</h3>

<table>
<tr><td><code>True positive</code></td>
<td>
<p> number of correctly estimated links.</p>
</td></tr>
<tr><td><code>True negative</code></td>
<td>
<p> number of true non-existing links which is correctly estimated.</p>
</td></tr>
<tr><td><code>False positive</code></td>
<td>
<p> number of links which they are not in the true graph, but are incorrectly estimated.</p>
</td></tr>
<tr><td><code>False negative</code></td>
<td>
<p> number of links which they are in the true graph, but are not estimated.</p>
</td></tr>
<tr><td><code>F1-score</code></td>
<td>
<p> weighted average of the <code>"positive predictive"</code> and <code>"true positive rate"</code>. The F1-score value reaches its best value at 1 and worst score at 0.</p>
</td></tr>
<tr><td><code>Specificity</code></td>
<td>
<p> Specificity value reaches its best value at 1 and worst score at 0.</p>
</td></tr>
<tr><td><code>Sensitivity</code></td>
<td>
<p> Sensitivity value reaches its best value at 1 and worst score at 0.</p>
</td></tr>
<tr><td><code>MCC</code></td>
<td>
<p> Matthews Correlation Coefficients (MCC) value reaches its best value at 1 and worst score at 0.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a>, Antonio Abbruzzo, and Ivan Vujacic</p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code>, <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>, <code><a href="#topic+plotroc">plotroc</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Generating multivariate normal data from a 'random' graph
data.sim &lt;- bdgraph.sim( n = 50, p = 6, size = 7, vis = TRUE )
    
# Running sampling algorithm based on GGMs 
sample.ggm &lt;- bdgraph( data = data.sim, method = "ggm", iter = 10000 )
   
# Comparing the results
compare( sample.ggm, data.sim, main = c( "True", "GGM" ), vis = TRUE )
      
# Running sampling algorithm based on GCGMs
sample.gcgm &lt;- bdgraph( data = data.sim, method = "gcgm", iter = 10000 )

# Comparing GGM and GCGM methods
compare( list( sample.ggm, sample.gcgm ), data.sim, 
         main = c( "True", "GGM", "GCGM" ), vis = TRUE )

## End(Not run)
</code></pre>

<hr>
<h2 id='conf.mat'>	Confusion Matrix </h2><span id='topic+conf.mat'></span>

<h3>Description</h3>

<p> Create a Confusion Matrix. </p>


<h3>Usage</h3>

<pre><code class='language-R'> 
conf.mat( pred, actual, cutoff = 0.5, proportion = FALSE, 
                     dnn = c( "Prediction", "Actual" ), ... )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="conf.mat_+3A_pred">pred</code></td>
<td>

<p>adjacency matrix corresponding to an estimated graph. 
It can be an object with <code>S3</code> class &quot;<code>bdgraph</code>&quot; from function <code><a href="#topic+bdgraph">bdgraph</a></code>. 
It can be an object of <code>S3</code> class <code>"ssgraph"</code>, from the function <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code> of <code>R</code> package <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code>.
</p>
</td></tr>
<tr><td><code id="conf.mat_+3A_actual">actual</code></td>
<td>
<p> adjacency matrix corresponding to the actual graph structure in which <code class="reqn">a_{ij}=1</code> if there is a link between notes <code class="reqn">i</code> and <code class="reqn">j</code>, otherwise <code class="reqn">a_{ij}=0</code>.
It can be an object with <code>S3</code> class &quot;<code>sim</code>&quot; from function <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>.
It can be an object with <code>S3</code> class <code>"graph"</code> from function <code><a href="#topic+graph.sim">graph.sim</a></code>.
It can be a factor, numeric or character vector of responses (true class), typically encoded with 0 (controls) and 1 (cases).
Only two classes can be used in a ROC curve. </p>
</td></tr>
<tr><td><code id="conf.mat_+3A_cutoff">cutoff</code></td>
<td>
<p>cutoff value for the case that <code>pred</code> is vector of probabilites. The default is 0.5. </p>
</td></tr>  
<tr><td><code id="conf.mat_+3A_proportion">proportion</code></td>
<td>

<p>logical: FALSE (default) for a confusion matrix with number of cases. 
TRUE, for a confusion matrix with the proportion of cases.
</p>
</td></tr>
<tr><td><code id="conf.mat_+3A_dnn">dnn</code></td>
<td>
<p>names to be given to the dimensions in the result (the dimnames names). </p>
</td></tr>
<tr><td><code id="conf.mat_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to <code>table</code>. </p>
</td></tr>
</table>


<h3>Value</h3>

<p> the results of <code>table</code> on <code>pred</code> and <code>actual</code>. </p>


<h3>Author(s)</h3>

<p>Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+conf.mat.plot">conf.mat.plot</a></code>, <code><a href="#topic+compare">compare</a></code>, <code><a href="#topic+roc">roc</a></code>, <code><a href="#topic+bdgraph">bdgraph</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set.seed( 100 )

# Generating multivariate normal data from a 'random' graph
data.sim &lt;- bdgraph.sim( n = 50, p = 6, size = 7, vis = TRUE )

# Running sampling algorithm based on GGMs 
sample.ggm &lt;- bdgraph( data = data.sim, method = "ggm", iter = 10000 )

# Confusion Matrix for GGM method
conf.mat( pred = sample.ggm, actual = data.sim )

## End(Not run)
</code></pre>

<hr>
<h2 id='conf.mat.plot'>	Plot Confusion Matrix </h2><span id='topic+conf.mat.plot'></span>

<h3>Description</h3>

<p> Plot a Confusion Matrix. </p>


<h3>Usage</h3>

<pre><code class='language-R'> 
conf.mat.plot( pred, actual, cutoff = 0.5, conf.level = 0, margin = 1, 
                          color = c( "#ff83a8", "#83ff9b" ), ... )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="conf.mat.plot_+3A_pred">pred</code></td>
<td>

<p>adjacency matrix corresponding to an estimated graph. 
It can be an object with <code>S3</code> class &quot;<code>bdgraph</code>&quot; from function <code><a href="#topic+bdgraph">bdgraph</a></code>. 
It can be an object of <code>S3</code> class <code>"ssgraph"</code>, from the function <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code> of <code>R</code> package <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code>.
</p>
</td></tr>
<tr><td><code id="conf.mat.plot_+3A_actual">actual</code></td>
<td>
<p> adjacency matrix corresponding to the actual graph structure in which <code class="reqn">a_{ij}=1</code> if there is a link between notes <code class="reqn">i</code> and <code class="reqn">j</code>, otherwise <code class="reqn">a_{ij}=0</code>.
It can be an object with <code>S3</code> class &quot;<code>sim</code>&quot; from function <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>.
It can be an object with <code>S3</code> class <code>"graph"</code> from function <code><a href="#topic+graph.sim">graph.sim</a></code>.
It can be a factor, numeric or character vector of responses (true class), typically encoded with 0 (controls) and 1 (cases).
Only two classes can be used in a ROC curve. </p>
</td></tr>
<tr><td><code id="conf.mat.plot_+3A_cutoff">cutoff</code></td>
<td>
<p>cutoff value for the case that <code>pred</code> is vector of probabilites. The default is 0.5. </p>
</td></tr>  
<tr><td><code id="conf.mat.plot_+3A_conf.level">conf.level</code></td>
<td>
<p>confidence level used for the confidence rings on the odds ratios. Must be a single nonnegative number less than 1; if set to 0 (the default), confidence rings are suppressed. </p>
</td></tr>  
<tr><td><code id="conf.mat.plot_+3A_margin">margin</code></td>
<td>
<p>numeric vector with the margins to equate. Must be one of 1 (the default), 2, or c(1, 2), which corresponds to standardizing the row, column, or both margins in each 2 by 2 table. Only used if std equals &quot;margins&quot;. </p>
</td></tr>  
<tr><td><code id="conf.mat.plot_+3A_color">color</code></td>
<td>
<p>vector of length 2 specifying the colors to use for the smaller and larger diagonals of each 2 by 2 table. </p>
</td></tr>  
<tr><td><code id="conf.mat.plot_+3A_...">...</code></td>
<td>
<p>options to be passed to <code>fourfoldplot</code>. </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+conf.mat">conf.mat</a></code>, <code><a href="#topic+compare">compare</a></code>, <code><a href="#topic+roc">roc</a></code>, <code><a href="#topic+bdgraph">bdgraph</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set.seed( 100 )

# Generating multivariate normal data from a 'random' graph
data.sim &lt;- bdgraph.sim( n = 50, p = 6, size = 7, vis = TRUE )

# Running sampling algorithm based on GGMs 
sample.ggm &lt;- bdgraph( data = data.sim, method = "ggm", iter = 10000 )

# Confusion Matrix for GGM method
conf.mat.plot( pred = sample.ggm, actual = data.sim )

## End(Not run)
</code></pre>

<hr>
<h2 id='covariance'> Estimated covariance matrix </h2><span id='topic+covariance'></span>

<h3>Description</h3>

 
<p>Provides the estimated covariance matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> covariance( bdgraph.obj, round = 2 ) </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="covariance_+3A_bdgraph.obj">bdgraph.obj</code></td>
<td>
<p>object of <code>S3</code> class &quot;<code>bdgraph</code>&quot;, from function <code><a href="#topic+bdgraph">bdgraph</a></code>. 
It also can be an object of <code>S3</code> class <code>"ssgraph"</code>, from the function <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code> of <code>R</code> package <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code>.
</p>
</td></tr>
<tr><td><code id="covariance_+3A_round">round</code></td>
<td>
<p> value for rounding all probabilities to the specified number of decimal places. </p>
</td></tr>
</table>


<h3>Value</h3>

<p> matrix which corresponds the estimated covariance matrix.</p>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>
<p>Mohammadi, A. and Wit, E. C. (2015). Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Bayesian Analysis</em>, 10(1):109-138, <a href="https://doi.org/10.1214/14-BA889">doi:10.1214/14-BA889</a>
</p>
<p>Mohammadi, R., Massam, H. and Letac, G. (2021). Accelerating Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Journal of the American Statistical Association</em>, <a href="https://doi.org/10.1080/01621459.2021.1996377">doi:10.1080/01621459.2021.1996377</a> 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+precision">precision</a></code>, <code><a href="#topic+plinks">plinks</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Generating multivariate normal data from a 'circle' graph
data.sim &lt;- bdgraph.sim( n = 70, p = 6, graph = "circle", vis = TRUE )

bdgraph.obj &lt;- bdgraph( data = data.sim )

covariance( bdgraph.obj ) # Estimated covariance matrix
  
data.sim $ sigma   # True covariance matrix

## End(Not run)
</code></pre>

<hr>
<h2 id='Discrete+20Weibull'> The Discrete Weibull Distribution (Type 1) </h2><span id='topic+ddweibull'></span><span id='topic+pdweibull'></span><span id='topic+qdweibull'></span><span id='topic+rdweibull'></span>

<h3>Description</h3>

<p>Density, distribution function, quantile function and random generation for the discrete Weibull distribution (type I) with parameters <code class="reqn">q</code> and <code class="reqn">\beta</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ddweibull( x, q = exp( -1 ), beta = 1, zero = TRUE )
pdweibull( x, q = exp( -1 ), beta = 1, zero = TRUE )
qdweibull( p, q = exp( -1 ), beta = 1, zero = TRUE )
rdweibull( n, q = exp( -1 ), beta = 1, zero = TRUE )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Discrete+2B20Weibull_+3A_x">x</code></td>
<td>
<p> vector of quantiles.</p>
</td></tr>
<tr><td><code id="Discrete+2B20Weibull_+3A_p">p</code></td>
<td>
<p> vector of probabilities. </p>
</td></tr>
<tr><td><code id="Discrete+2B20Weibull_+3A_q">q</code>, <code id="Discrete+2B20Weibull_+3A_beta">beta</code></td>
<td>
<p> shape and scale parameters, the latter defaulting to 1. </p>
</td></tr>
<tr><td><code id="Discrete+2B20Weibull_+3A_zero">zero</code></td>
<td>
<p> logical; if <code>TRUE</code> (default), the support contains <code class="reqn">0</code>; <code>FALSE</code> otherwise. </p>
</td></tr>
<tr><td><code id="Discrete+2B20Weibull_+3A_n">n</code></td>
<td>
<p> number of observations. If <code>length(n) &gt; 1</code>, the length is taken to be the number required. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The discrete Weibull distribution has density given by 
</p>
<p style="text-align: center;"><code class="reqn">
f(x) = q^{x^\beta} - q^{(x+1)^\beta}, x = 0, 1, 2, \ldots
</code>
</p>

<p>For the case <code>zero</code> = <code>FALSE</code>:
</p>
<p style="text-align: center;"><code class="reqn">
f(x) = q^{(x-1)^\beta} - q^{x^\beta}, x = 1, 2, \ldots
</code>
</p>

<p>Cumulative distribution function
</p>
<p style="text-align: center;"><code class="reqn">
F(x) = 1-q^{(x+1)^\beta}
</code>
</p>

<p>For the case <code>zero</code> = <code>FALSE</code>, <code>x+1</code> should replaced by <code>x</code>.
</p>


<h3>Value</h3>

<p><code>ddweibull</code> gives the density, <code>pdweibull</code> gives the distribution function, <code>qdweibull</code> gives the quantile function, and <code>rdweibull</code> generates random values.
</p>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a>, Pariya Behrouzi, Veronica Vinciotti </p>


<h3>References</h3>

<p>Nakagawa, T. and Osaki, S. (1975). The Discrete Weibull Distribution. <em>IEEE Transactions on Reliability</em>, R-24, 300-301, <a href="https://doi.org/10.1109/TR.1975.5214915">doi:10.1109/TR.1975.5214915</a>
</p>


<h3>See Also</h3>

 <p><code><a href="stats.html#topic+dweibull">dweibull</a></code>, <code><a href="#topic+bdw.reg">bdw.reg</a></code>, <a href="#topic+bdgraph.dw">bdgraph.dw</a> </p>


<h3>Examples</h3>

<pre><code class='language-R'>n    = 1000
q    = 0.4
beta = 0.8

set.seed( 7 )

rdw = rdweibull( n = n, q = q, beta = beta )

plot( prop.table( table( rdw ) ), type = "h", col = "gray50" )

x = 0:max( rdw )

lines( x, ddweibull( x = x, q = q, beta = beta ), type = "o", col = "blue", lwd = 2 )

hist( pdweibull( x = rdw, q = q, beta = beta ) )

plot( ecdf( rdw ) )
lines( x, pdweibull( x, q = q, beta = beta ), col = "blue", lwd = 2, type = "s" )
</code></pre>

<hr>
<h2 id='geneExpression'> Human gene expression dataset </h2><span id='topic+geneExpression'></span>

<h3>Description</h3>

 
<p>The dataset contains human gene expression of <code class="reqn">100</code> transcripts (with unique Illumina TargetID) measured on <code class="reqn">60</code> unrelated individuals. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'> data( geneExpression ) </code></pre>


<h3>Format</h3>

<p> The format is a matrix with 60 rows (number of individuals) and 100 column (number of transcripts). </p>


<h3>References</h3>

<p>Bhadra, A. and Mallick, B. K. (2013). Joint High Dimensional Bayesian Variable and Covariance Selection with an Application to eQTL Analysis, <em>Biometrics</em>, 69(2):447-457, <a href="https://doi.org/10.1111/biom.12021">doi:10.1111/biom.12021</a>
</p>
<p>Mohammadi, A. and Wit, E. C. (2015). Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Bayesian Analysis</em>, 10(1):109-138, <a href="https://doi.org/10.1214/14-BA889">doi:10.1214/14-BA889</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data( geneExpression )

dim( geneExpression )
head( geneExpression )
</code></pre>

<hr>
<h2 id='gnorm'> Normalizing constant for G-Wishart </h2><span id='topic+gnorm'></span>

<h3>Description</h3>

<p>Calculates log of the normalizing constant of G-Wishart distribution based on the Monte Carlo method, developed by Atay-Kayis and Massam (2005).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gnorm( adj, b = 3, D = diag( ncol( adj ) ), iter = 100 )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gnorm_+3A_adj">adj</code></td>
<td>

<p>adjacency matrix corresponding to the graph structure. It is an upper triangular matrix in which 
<code class="reqn">a_{ij}=1</code> if there is a link between notes <code class="reqn">i</code> and <code class="reqn">j</code>, 
otherwise <code class="reqn">a_{ij}=0</code>.
</p>
</td></tr>
<tr><td><code id="gnorm_+3A_b">b</code></td>
<td>
<p> degree of freedom for G-Wishart distribution, <code class="reqn">W_G(b, D)</code>. </p>
</td></tr>
<tr><td><code id="gnorm_+3A_d">D</code></td>
<td>

<p>positive definite <code class="reqn">(p \times p)</code> &quot;scale&quot; matrix for G-Wishart distribution, 
<code class="reqn">W_G(b,D)</code>. The default is an identity matrix.
</p>
</td></tr>
<tr><td><code id="gnorm_+3A_iter">iter</code></td>
<td>
<p>number of iteration for the Monte Carlo approximation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Log of the normalizing constant approximation using Monte Carlo method for a G-Wishart distribution, <code class="reqn">K \sim W_G(b, D)</code>, with density:
</p>
<p style="text-align: center;"><code class="reqn">Pr(K) = \frac{1}{I(b, D)} |K| ^ {(b - 2) / 2} \exp \left\{- \frac{1}{2} \mbox{trace}(K \times D)\right\}.</code>
</p>



<h3>Value</h3>

<p> Log of the normalizing constant of G-Wishart distribution. </p>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> </p>


<h3>References</h3>

<p>Atay-Kayis, A. and Massam, H. (2005). A monte carlo method for computing the marginal likelihood in nondecomposable Gaussian graphical models, <em>Biometrika</em>, 92(2):317-335, <a href="https://doi.org/10.1093/biomet/92.2.317">doi:10.1093/biomet/92.2.317</a>
</p>
<p>Mohammadi, R., Massam, H. and Letac, G. (2021). Accelerating Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Journal of the American Statistical Association</em>, <a href="https://doi.org/10.1080/01621459.2021.1996377">doi:10.1080/01621459.2021.1996377</a> 
</p>
<p>Uhler, C., et al (2018) Exact formulas for the normalizing constants of Wishart distributions for graphical models, <em>The Annals of Statistics</em> 46(1):90-118, <a href="https://doi.org/10.1214/17-AOS1543">doi:10.1214/17-AOS1543</a>
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+rgwish">rgwish</a></code>, <code><a href="#topic+rwish">rwish</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# adj: adjacency matrix of graph with 3 nodes and 2 links
adj &lt;- matrix( c( 0, 0, 1,
                  0, 0, 1,
                  0, 0, 0 ), 3, 3, byrow = TRUE )		                
   
gnorm( adj, b = 3, D = diag( 3 ) )

## End(Not run)
</code></pre>

<hr>
<h2 id='graph.sim'> Graph simulation </h2><span id='topic+graph.sim'></span>

<h3>Description</h3>

<p>Simulating undirected graph structures, including 
&quot;<code>random</code>&quot;, &quot;<code>cluster</code>&quot;, &quot;<code>scale-free</code>&quot;, &quot;<code>lattice</code>&quot;, &quot;<code>hub</code>&quot;, &quot;<code>star</code>&quot;, and &quot;<code>circle</code>&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>graph.sim( p = 10, graph = "random", prob = 0.2, size = NULL, class = NULL, vis = FALSE,
           rewire = 0.05 )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="graph.sim_+3A_p">p</code></td>
<td>
<p>number of variables (nodes).</p>
</td></tr>
<tr><td><code id="graph.sim_+3A_graph">graph</code></td>
<td>
<p> undirected graph with options 
&quot;<code>random</code>&quot;, &quot;<code>cluster</code>&quot;, &quot;<code>smallworld</code>&quot;, &quot;<code>scale-free</code>&quot;, &quot;<code>lattice</code>&quot;, &quot;<code>hub</code>&quot;, &quot;<code>star</code>&quot;, and &quot;<code>circle</code>&quot;. 
It also could be an adjacency matrix corresponding to a graph structure (an upper triangular matrix in which 
<code class="reqn">g_{ij}=1</code> if there is a link between notes <code class="reqn">i</code> and <code class="reqn">j</code>, otherwise <code class="reqn">g_{ij}=0</code>). 
</p>
</td></tr>
<tr><td><code id="graph.sim_+3A_prob">prob</code></td>
<td>
<p> if <code>graph</code> = &quot;<code>random</code>&quot;, it is the probability that a pair of nodes has a link.</p>
</td></tr>
<tr><td><code id="graph.sim_+3A_size">size</code></td>
<td>
<p>number of links in the true graph (graph size).</p>
</td></tr>
<tr><td><code id="graph.sim_+3A_class">class</code></td>
<td>
<p> if <code>graph</code> = &quot;<code>cluster</code>&quot;, it is the number of classes. </p>
</td></tr>
<tr><td><code id="graph.sim_+3A_vis">vis</code></td>
<td>
<p>visualize the true graph structure.</p>
</td></tr>
<tr><td><code id="graph.sim_+3A_rewire">rewire</code></td>
<td>
<p>rewiring probability for smallworld network. Must be between 0 and 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p> The adjacency matrix corresponding to the simulated graph structure, as an object with <code>S3</code> class <code>"graph"</code>.</p>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> and Alexander Christensen </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>, <code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'># Generating a 'hub' graph 
adj &lt;- graph.sim( p = 8, graph = "scale-free" )

plot( adj )

adj
</code></pre>

<hr>
<h2 id='link2adj'> Extract links from an adjacency matrix </h2><span id='topic+link2adj'></span>

<h3>Description</h3>

<p> Extract links from an adjacency matrix or an object of calsses &quot;<code>sim</code>&quot; from function <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code> and
<code>"graph"</code> from function <code><a href="#topic+graph.sim">graph.sim</a></code>.</p>


<h3>Usage</h3>

<pre><code class='language-R'> link2adj( link, p = NULL ) </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="link2adj_+3A_link">link</code></td>
<td>
<p> (<code class="reqn">2 \times p</code>) <code>matrix</code> or a <code>data.frame</code> corresponding to the links from the graph structure. </p>
</td></tr>
<tr><td><code id="link2adj_+3A_p">p</code></td>
<td>
<p>number of nodes of the graph.</p>
</td></tr>
</table>


<h3>Value</h3>

<p> An adjacency matrix corresponding to a graph structure in which <code class="reqn">a_{ij}=1</code> if there is a link between notes <code class="reqn">i</code> and <code class="reqn">j</code>, otherwise <code class="reqn">a_{ij}=0</code>.</p>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+adj2link">adj2link</a></code>, <code><a href="#topic+graph.sim">graph.sim</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'># Generating a 'random' graph 
adj &lt;- graph.sim( p = 6, vis = TRUE )

link &lt;- adj2link( adj )

link2adj( link, p = 6 )
   
</code></pre>

<hr>
<h2 id='pgraph'> Posterior probabilities of the graphs </h2><span id='topic+pgraph'></span>

<h3>Description</h3>

<p>Provides the estimated posterior probabilities for the most likely graphs or a specific graph.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> pgraph( bdgraph.obj, number.g = 4, adj = NULL ) </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pgraph_+3A_bdgraph.obj">bdgraph.obj</code></td>
<td>
<p> object of <code>S3</code> class &quot;<code>bdgraph</code>&quot;, from function <code><a href="#topic+bdgraph">bdgraph</a></code>. </p>
</td></tr>
<tr><td><code id="pgraph_+3A_number.g">number.g</code></td>
<td>
<p> number of graphs with the highest posterior probabilities to be shown.
This option is ignored if <code>'adj'</code> is specified. </p>
</td></tr>
<tr><td><code id="pgraph_+3A_adj">adj</code></td>
<td>
<p> adjacency matrix corresponding to a graph structure. It is an upper triangular matrix in which 
<code class="reqn">a_{ij}=1</code> if there is a link between notes <code class="reqn">i</code> and <code class="reqn">j</code>, otherwise <code class="reqn">a_{ij}=0</code>.
It also can be an object of <code>S3</code> class &quot;<code>sim</code>&quot;, from function <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>. </p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>selected_g</code></td>
<td>
<p> adjacency matrices which corresponding to the graphs with the highest posterior probabilities.</p>
</td></tr>
<tr><td><code>prob_g</code></td>
<td>
<p>vector of the posterior probabilities of the graphs corresponding to <code>'selected\_g'</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> and Ernst Wit </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>
<p>Mohammadi, A. and Wit, E. C. (2015). Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Bayesian Analysis</em>, 10(1):109-138, <a href="https://doi.org/10.1214/14-BA889">doi:10.1214/14-BA889</a>
</p>
<p>Mohammadi, R., Massam, H. and Letac, G. (2021). Accelerating Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Journal of the American Statistical Association</em>, <a href="https://doi.org/10.1080/01621459.2021.1996377">doi:10.1080/01621459.2021.1996377</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Generating multivariate normal data from a 'random' graph
data.sim &lt;- bdgraph.sim( n = 50, p = 6, size = 6, vis = TRUE )
   
bdgraph.obj &lt;- bdgraph( data = data.sim, save = TRUE )
   
# Estimated posterior probability of the true graph
pgraph( bdgraph.obj, adj = data.sim )
   
# Estimated posterior probability of first and second graphs with highest probabilities
pgraph( bdgraph.obj, number.g = 2 )

## End(Not run)
</code></pre>

<hr>
<h2 id='plinks'> Estimated posterior link probabilities </h2><span id='topic+plinks'></span>

<h3>Description</h3>

 
<p>Provides the estimated posterior link probabilities for all possible links in the graph.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> plinks( bdgraph.obj, round = 2, burnin = NULL ) </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plinks_+3A_bdgraph.obj">bdgraph.obj</code></td>
<td>
<p>object of <code>S3</code> class &quot;<code>bdgraph</code>&quot;, from function <code><a href="#topic+bdgraph">bdgraph</a></code>. 
It also can be an object of <code>S3</code> class <code>"ssgraph"</code>, from the function <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code> of <code>R</code> package <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code>.
</p>
</td></tr>
<tr><td><code id="plinks_+3A_round">round</code></td>
<td>
<p> value for rounding all probabilities to the specified number of decimal places. </p>
</td></tr>
<tr><td><code id="plinks_+3A_burnin">burnin</code></td>
<td>
<p> number of burn-in iteration to scape.  </p>
</td></tr>
</table>


<h3>Value</h3>

<p> An upper triangular matrix which corresponds the estimated posterior probabilities for all possible links.</p>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> and Ernst Wit </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>
<p>Mohammadi, A. and Wit, E. C. (2015). Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Bayesian Analysis</em>, 10(1):109-138, <a href="https://doi.org/10.1214/14-BA889">doi:10.1214/14-BA889</a>
</p>
<p>Mohammadi, R., Massam, H. and Letac, G. (2021). Accelerating Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Journal of the American Statistical Association</em>, <a href="https://doi.org/10.1080/01621459.2021.1996377">doi:10.1080/01621459.2021.1996377</a> 
</p>
<p>Dobra, A. and Mohammadi, R. (2018). Loglinear Model Selection and Human Mobility, <em>Annals of Applied Statistics</em>, 12(2):815-845, <a href="https://doi.org/10.1214/18-AOAS1164">doi:10.1214/18-AOAS1164</a>
</p>
<p>Mohammadi, A. et al (2017). Bayesian modelling of Dupuytren disease by using Gaussian copula graphical models, <em>Journal of the Royal Statistical Society: Series C</em>, 66(3):629-645, <a href="https://doi.org/10.1111/rssc.12171">doi:10.1111/rssc.12171</a> 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Generating multivariate normal data from a 'circle' graph
data.sim &lt;- bdgraph.sim( n = 70, p = 6, graph = "circle", vis = TRUE )

bdgraph.obj   &lt;- bdgraph( data = data.sim, iter = 10000 )

plinks( bdgraph.obj, round = 2 )

## End(Not run)
</code></pre>

<hr>
<h2 id='plot.bdgraph'> Plot function for <code>S3</code> class &quot;<code>bdgraph</code>&quot; </h2><span id='topic+plot.bdgraph'></span>

<h3>Description</h3>

<p> Visualizes structure of the selected graphs which could be a graph with links for which their estimated posterior probabilities are greater than 0.5
or graph with the highest posterior probability. </p>


<h3>Usage</h3>

<pre><code class='language-R'> ## S3 method for class 'bdgraph'
plot( x, cut = 0.5, number.g = NULL, main = NULL, 
           layout = igraph::layout_with_fr, vertex.size = 2, vertex.color = "orange", 
           vertex.frame.color = "orange", vertex.label = NULL, vertex.label.dist = 0.5, 
           vertex.label.color = "blue", edge.color = "lightblue", ... ) </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.bdgraph_+3A_x">x</code></td>
<td>
<p> object of <code>S3</code> class &quot;<code>bdgraph</code>&quot;, from function <code><a href="#topic+bdgraph">bdgraph</a></code>. </p>
</td></tr>
<tr><td><code id="plot.bdgraph_+3A_cut">cut</code></td>
<td>
<p> threshold for including the links in the selected graph based on the estimated posterior probabilities of the links; See the examples. </p>
</td></tr> 
<tr><td><code id="plot.bdgraph_+3A_number.g">number.g</code></td>
<td>
<p> number of graphs with the highest probabilities. This option works for the case running function <code>bdgraph()</code> with option <code>save = TRUE</code>; See the examples. </p>
</td></tr>
<tr><td><code id="plot.bdgraph_+3A_main">main</code></td>
<td>
<p> Graphical parameter (see plot).</p>
</td></tr>
<tr><td><code id="plot.bdgraph_+3A_layout">layout</code></td>
<td>
<p> vertex placement which is according to <code>R</code> package <code><a href="igraph.html#topic+igraph">igraph</a></code>; For different layouts, see <code><a href="igraph.html#topic+layout">layout</a></code> of <code>R</code> package <code>igraph</code>. </p>
</td></tr>
<tr><td><code id="plot.bdgraph_+3A_vertex.size">vertex.size</code></td>
<td>
<p> vertex size which is according to <code>R</code> package <code><a href="igraph.html#topic+igraph">igraph</a></code>. </p>
</td></tr>
<tr><td><code id="plot.bdgraph_+3A_vertex.color">vertex.color</code></td>
<td>
<p> vertex color which is according to <code>R</code> package <code><a href="igraph.html#topic+igraph">igraph</a></code>. </p>
</td></tr>
<tr><td><code id="plot.bdgraph_+3A_vertex.frame.color">vertex.frame.color</code></td>
<td>
<p> vertex frame color which is according to <code>R</code> package <code><a href="igraph.html#topic+igraph">igraph</a></code>. </p>
</td></tr>
<tr><td><code id="plot.bdgraph_+3A_vertex.label">vertex.label</code></td>
<td>
<p> vertex label. The default vertex labels are the vertex ids. </p>
</td></tr>
<tr><td><code id="plot.bdgraph_+3A_vertex.label.dist">vertex.label.dist</code></td>
<td>
<p> vertex label distance which is according to <code>R</code> package <code><a href="igraph.html#topic+igraph">igraph</a></code>. </p>
</td></tr>
<tr><td><code id="plot.bdgraph_+3A_vertex.label.color">vertex.label.color</code></td>
<td>
<p> vertex label color which is according to <code>R</code> package <code><a href="igraph.html#topic+igraph">igraph</a></code>. </p>
</td></tr>
<tr><td><code id="plot.bdgraph_+3A_edge.color">edge.color</code></td>
<td>
<p> edge color which is according to <code>R</code> package <code><a href="igraph.html#topic+igraph">igraph</a></code>. </p>
</td></tr>
<tr><td><code id="plot.bdgraph_+3A_...">...</code></td>
<td>
<p> additional plotting parameters. For the complete list, see <code><a href="igraph.html#topic+igraph.plotting">igraph.plotting</a></code> of <code>R</code> package <code>igraph</code>. </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> and Ernst Wit </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set.seed( 100 )

# Generating multivariate normal data from a 'random' graph
data.sim &lt;- bdgraph.sim( n = 100, p = 15, graph = "random", prob = 0.2, vis = TRUE )
   
bdgraph.obj &lt;- bdgraph( data = data.sim )
   
plot( bdgraph.obj )
   
bdgraph.obj &lt;- bdgraph( data = data.sim, save = TRUE )
   
plot( bdgraph.obj, cut = 0.5 )

plot( bdgraph.obj, number.g = 4 )

## End(Not run)
</code></pre>

<hr>
<h2 id='plot.graph'> Plot function for <code>S3</code> class <code>"graph"</code> </h2><span id='topic+plot.graph'></span>

<h3>Description</h3>

<p> Visualizes structure of the graph. </p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'graph'
plot( x, cut = 0.5, mode = "undirected", diag = FALSE, main = NULL, 
           layout = igraph::layout_with_fr, vertex.size = 2, vertex.color = "orange", 
           vertex.frame.color = "orange", vertex.label = NULL, vertex.label.dist = 0.5, 
           vertex.label.color = "blue", edge.color = "lightblue", ... )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.graph_+3A_x">x</code></td>
<td>
<p>   object of <code>S3</code> class <code>"graph"</code>, from function <code><a href="#topic+graph.sim">graph.sim</a></code>.  </p>
</td></tr>
<tr><td><code id="plot.graph_+3A_cut">cut</code></td>
<td>
<p> for the case where input '<code>x</code>' is the object of class &quot;bdgraph&quot; or &quot;ssgraph&quot;. Threshold for including the links in the selected graph based on the estimated posterior probabilities of the links. </p>
</td></tr> 
<tr><td><code id="plot.graph_+3A_mode">mode</code></td>
<td>
<p> type of graph which is according to <code>R</code> package <code><a href="igraph.html#topic+igraph">igraph</a></code>.</p>
</td></tr>
<tr><td><code id="plot.graph_+3A_diag">diag</code></td>
<td>
<p> logical which is according to <code>R</code> package <code><a href="igraph.html#topic+igraph">igraph</a></code>.</p>
</td></tr>
<tr><td><code id="plot.graph_+3A_main">main</code></td>
<td>
<p> graphical parameter (see plot).</p>
</td></tr>
<tr><td><code id="plot.graph_+3A_layout">layout</code></td>
<td>
<p> vertex placement which is according to <code>R</code> package <code><a href="igraph.html#topic+igraph">igraph</a></code>; For different layouts, see <code><a href="igraph.html#topic+layout">layout</a></code> of <code>R</code> package <code>igraph</code>. </p>
</td></tr>
<tr><td><code id="plot.graph_+3A_vertex.size">vertex.size</code></td>
<td>
<p> vertex size which is according to <code>R</code> package <code><a href="igraph.html#topic+igraph">igraph</a></code>. </p>
</td></tr>
<tr><td><code id="plot.graph_+3A_vertex.color">vertex.color</code></td>
<td>
<p> vertex color which is according to <code>R</code> package <code><a href="igraph.html#topic+igraph">igraph</a></code>. </p>
</td></tr>
<tr><td><code id="plot.graph_+3A_vertex.frame.color">vertex.frame.color</code></td>
<td>
<p> vertex frame color which is according to <code>R</code> package <code><a href="igraph.html#topic+igraph">igraph</a></code>. </p>
</td></tr>
<tr><td><code id="plot.graph_+3A_vertex.label">vertex.label</code></td>
<td>
<p> vertex label. The default vertex labels are the vertex ids. </p>
</td></tr>
<tr><td><code id="plot.graph_+3A_vertex.label.dist">vertex.label.dist</code></td>
<td>
<p> vertex label distance which is according to <code>R</code> package <code><a href="igraph.html#topic+igraph">igraph</a></code>. </p>
</td></tr>
<tr><td><code id="plot.graph_+3A_vertex.label.color">vertex.label.color</code></td>
<td>
<p> vertex label color which is according to <code>R</code> package <code><a href="igraph.html#topic+igraph">igraph</a></code>. </p>
</td></tr>
<tr><td><code id="plot.graph_+3A_edge.color">edge.color</code></td>
<td>
<p> edge color which is according to <code>R</code> package <code><a href="igraph.html#topic+igraph">igraph</a></code>. </p>
</td></tr>
<tr><td><code id="plot.graph_+3A_...">...</code></td>
<td>
<p> additional plotting parameters. For the complete list, see <code><a href="igraph.html#topic+igraph.plotting">igraph.plotting</a></code> of <code>R</code> package <code>igraph</code>. </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+graph.sim">graph.sim</a></code>, <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>, <code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'># Generating a 'scale-free' graph 
adj &lt;- graph.sim( p = 20, graph = "scale-free" )

plot( adj )
</code></pre>

<hr>
<h2 id='plot.sim'> Plot function for <code>S3</code> class &quot;<code>sim</code>&quot; </h2><span id='topic+plot.sim'></span>

<h3>Description</h3>

<p> Visualizes structure of the simulated graph for an object of <code>S3</code> class &quot;<code>sim</code>&quot;, from function <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>. </p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sim'
plot( x, ... )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.sim_+3A_x">x</code></td>
<td>
<p> object of <code>S3</code> class &quot;<code>sim</code>&quot;, from function <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>.  </p>
</td></tr>
<tr><td><code id="plot.sim_+3A_...">...</code></td>
<td>
<p> additional plotting parameters. See <code><a href="#topic+plot.graph">plot.graph</a></code> and for the complete list <code><a href="igraph.html#topic+igraph.plotting">igraph.plotting</a></code> of <code>R</code> package <code>igraph</code>. </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> and Ernst Wit </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+graph.sim">graph.sim</a></code>, <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>, <code><a href="#topic+plot.graph">plot.graph</a></code>, <code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed( 10 )

# Generating synthetic multivariate normal data from a 'random' graph
data.sim &lt;- bdgraph.sim( n = 10, p = 15 )
  
plot( data.sim )
</code></pre>

<hr>
<h2 id='plotcoda'> Convergence plot </h2><span id='topic+plotcoda'></span>

<h3>Description</h3>

<p>Visualizes the cumulative occupancy fractions of all possible links in the graph.
It can be used for monitoring the convergence of the sampling algorithms, BDMCMC and RJMCMC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> plotcoda( bdgraph.obj, thin = NULL, control = TRUE, main = NULL, 
                  verbose = TRUE, ... ) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotcoda_+3A_bdgraph.obj">bdgraph.obj</code></td>
<td>
<p> object of <code>S3</code> class &quot;<code>bdgraph</code>&quot;, from function <code><a href="#topic+bdgraph">bdgraph</a></code>. 
It also can be an object of <code>S3</code> class <code>"ssgraph"</code>, from the function <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code> of <code>R</code> package <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code>.
</p>
</td></tr>
<tr><td><code id="plotcoda_+3A_thin">thin</code></td>
<td>
<p> option for getting fast result for a cumulative plot according to part of the iteration.</p>
</td></tr>
<tr><td><code id="plotcoda_+3A_control">control</code></td>
<td>
<p> logical: if TRUE (default) and the number of nodes is greater than 15, then 100 links randomly is selected for visualization. </p>
</td></tr> 
<tr><td><code id="plotcoda_+3A_main">main</code></td>
<td>
<p> graphical parameter (see plot). </p>
</td></tr>
<tr><td><code id="plotcoda_+3A_verbose">verbose</code></td>
<td>
<p> logical: if TRUE (default), report/print the calculation progress. </p>
</td></tr>	
<tr><td><code id="plotcoda_+3A_...">...</code></td>
<td>
<p> system reserved (no specific usage). </p>
</td></tr>  
</table>


<h3>Details</h3>

<p>Note that a spending time for this function depends on the number of nodes. 
For fast result, you can choose bigger value for the <code>'thin'</code> option.
</p>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code>, <code><a href="#topic+traceplot">traceplot</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Generating multivariate normal data from a 'circle' graph
data.sim &lt;- bdgraph.sim( n = 50, p = 6, graph = "circle", vis = TRUE )
  
bdgraph.obj &lt;- bdgraph( data = data.sim, iter = 10000, burnin = 0 , save = TRUE )
   
plotcoda( bdgraph.obj )  

## End(Not run)
</code></pre>

<hr>
<h2 id='plotroc'> ROC plot </h2><span id='topic+plotroc'></span>

<h3>Description</h3>

<p>Draws the receiver operating characteristic (ROC) curve according to the true graph structure for object of <code>S3</code> class &quot;<code>bdgraph</code>&quot;, from function <code><a href="#topic+bdgraph">bdgraph</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> 
plotroc( pred, actual, cut = 200, smooth = FALSE, calibrate = TRUE, 
         linetype = NULL, color = NULL, size = 1, main = "ROC Curve", 
         xlab = "False Postive Rate", ylab = "True Postive Rate",
         legend = TRUE, legend.size = 17, legend.position = c( 0.7, 0.3 ),
         labels = NULL, auc = TRUE, theme = ggplot2::theme_minimal() ) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotroc_+3A_pred">pred</code></td>
<td>

<p>upper triangular matrix corresponding to the estimated posterior probabilities for all possible links.
It can be an object with <code>S3</code> class &quot;<code>bdgraph</code>&quot; from function <code><a href="#topic+bdgraph">bdgraph</a></code>. 
It can be an object of <code>S3</code> class <code>"ssgraph"</code>, from the function <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code> of <code>R</code> package <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code>.
It can be an object of <code>S3</code> class <code>"select"</code>, from the function <code><a href="huge.html#topic+huge.select">huge.select</a></code> of <code>R</code> package <code><a href="huge.html#topic+huge">huge</a></code>.
Options <code>est2, est3</code> and <code>est4</code> are for comparing two or more different approaches.
</p>
</td></tr>
<tr><td><code id="plotroc_+3A_actual">actual</code></td>
<td>

<p>adjacency matrix corresponding to the true graph structure in which <code class="reqn">a_{ij}=1</code> if there is a link between notes <code class="reqn">i</code> and <code class="reqn">j</code>, otherwise <code class="reqn">a_{ij}=0</code>.
It can be an object with <code>S3</code> class &quot;<code>sim</code>&quot; from function <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>.
It can be an object with <code>S3</code> class <code>"graph"</code> from function <code><a href="#topic+graph.sim">graph.sim</a></code>.
</p>
</td></tr>
<tr><td><code id="plotroc_+3A_cut">cut</code></td>
<td>
<p> number of cut points.</p>
</td></tr>
<tr><td><code id="plotroc_+3A_smooth">smooth</code></td>
<td>
<p> logical: for smoothing the ROC curve.</p>
</td></tr> 
<tr><td><code id="plotroc_+3A_calibrate">calibrate</code></td>
<td>
<p>If <code>TRUE</code>, compute the value of AUC by taking the level of the probabilities into account.</p>
</td></tr>
<tr><td><code id="plotroc_+3A_linetype">linetype</code></td>
<td>
<p> specification for the default plotting line type.</p>
</td></tr> 
<tr><td><code id="plotroc_+3A_color">color</code></td>
<td>
<p> specification for the default plotting color.</p>
</td></tr> 
<tr><td><code id="plotroc_+3A_size">size</code></td>
<td>
<p> specification for the default plotting line size.</p>
</td></tr> 
<tr><td><code id="plotroc_+3A_main">main</code></td>
<td>
<p> overall title for the plot.</p>
</td></tr> 
<tr><td><code id="plotroc_+3A_xlab">xlab</code></td>
<td>
<p> title for the x axis.</p>
</td></tr> 
<tr><td><code id="plotroc_+3A_ylab">ylab</code></td>
<td>
<p> title for the y axis.</p>
</td></tr> 
<tr><td><code id="plotroc_+3A_legend">legend</code></td>
<td>
<p> logical: for adding legend to the ROC plot.</p>
</td></tr> 
<tr><td><code id="plotroc_+3A_legend.size">legend.size</code></td>
<td>
<p> title for the x axis.</p>
</td></tr> 
<tr><td><code id="plotroc_+3A_legend.position">legend.position</code></td>
<td>
<p> title for the y axis.</p>
</td></tr> 
<tr><td><code id="plotroc_+3A_labels">labels</code></td>
<td>
<p> for legends of the legend to the ROC plot.</p>
</td></tr> 
<tr><td><code id="plotroc_+3A_auc">auc</code></td>
<td>
<p> logical: to report AUC with legend.</p>
</td></tr> 
<tr><td><code id="plotroc_+3A_theme">theme</code></td>
<td>
<p> theme for the plot from the function <code><a href="ggplot2.html#topic+ggplot">ggplot2::ggplot()</a></code> of <code>R</code> package <code><a href="ggplot2.html#topic+ggplot">ggplot2::ggplot()</a></code>.</p>
</td></tr> 
</table>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a>; Lucas Vogels <a href="mailto:l.f.o.vogels@uva.nl">l.f.o.vogels@uva.nl</a> </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>
<p>Mohammadi, A. and Wit, E. C. (2015). Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Bayesian Analysis</em>, 10(1):109-138, <a href="https://doi.org/10.1214/14-BA889">doi:10.1214/14-BA889</a>
</p>
<p>Mohammadi, R., Massam, H. and Letac, G. (2021). Accelerating Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Journal of the American Statistical Association</em>, <a href="https://doi.org/10.1080/01621459.2021.1996377">doi:10.1080/01621459.2021.1996377</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+roc">roc</a></code>, <code><a href="pROC.html#topic+plot.roc">pROC::plot.roc()</a></code>, <code><a href="pROC.html#topic+auc">pROC::auc()</a></code>, <code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code>, <code><a href="#topic+compare">compare</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# To generate multivariate normal data from a 'random' graph
data.sim &lt;- bdgraph.sim( n = 30, p = 6, size = 7, vis = TRUE )
   
# To Run sampling algorithm
bdgraph.obj &lt;- bdgraph( data = data.sim, iter = 10000 )

# To compare the results
plotroc( bdgraph.ob2j, data.sim )
   
# To compare the results based on CGGMs approach
bdgraph.obj2 &lt;- bdgraph( data = data.sim, method = "gcgm", iter = 10000 )

# To Compare the resultss
plotroc( list( bdgraph.obj, bdgraph.obj2 ), data.sim, legend = FALSE )

## End(Not run)
</code></pre>

<hr>
<h2 id='posterior.predict'> Posterior Predictive Samples </h2><span id='topic+posterior.predict'></span>

<h3>Description</h3>

<p>Provides samples from the posterior predictive distribution.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>posterior.predict( object, iter = 1, ... )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="posterior.predict_+3A_object">object</code></td>
<td>
<p> object of <code>S3</code> class &quot;<code>bdgraph</code>&quot;, from function <code><a href="#topic+bdgraph">bdgraph</a></code>. </p>
</td></tr>
<tr><td><code id="posterior.predict_+3A_iter">iter</code></td>
<td>
<p> number of predictions. </p>
</td></tr>
<tr><td><code id="posterior.predict_+3A_...">...</code></td>
<td>
<p> additional parameters. </p>
</td></tr>
</table>


<h3>Value</h3>

 
<p>a <code>matrix</code> containing the predicted datasets, corresponding to the samples from the joint posterior disribtuion. 
</p>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> and Ernst Wit </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>
<p>Mohammadi, A. and Wit, E. C. (2015). Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Bayesian Analysis</em>, 10(1):109-138, <a href="https://doi.org/10.1214/14-BA889">doi:10.1214/14-BA889</a>
</p>
<p>Mohammadi, R., Massam, H. and Letac, G. (2021). Accelerating Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Journal of the American Statistical Association</em>, <a href="https://doi.org/10.1080/01621459.2021.1996377">doi:10.1080/01621459.2021.1996377</a> 
</p>
<p>Dobra, A. and Mohammadi, R. (2018). Loglinear Model Selection and Human Mobility, <em>Annals of Applied Statistics</em>, 12(2):815-845, <a href="https://doi.org/10.1214/18-AOAS1164">doi:10.1214/18-AOAS1164</a>
</p>
<p>Mohammadi, A. et al (2017). Bayesian modelling of Dupuytren disease by using Gaussian copula graphical models, <em>Journal of the Royal Statistical Society: Series C</em>, 66(3):629-645, <a href="https://doi.org/10.1111/rssc.12171">doi:10.1111/rssc.12171</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code>, <code><a href="#topic+bdgraph.dw">bdgraph.dw</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Generating multivariate normal data from a 'random' graph
data.sim &lt;- bdgraph.sim( n = 50, p = 6, size = 7, vis = TRUE )
   
bdgraph.obj &lt;- bdgraph( data = data.sim )
   
posterior.predict( bdgraph.obj, iter = 20 )

## End(Not run)
</code></pre>

<hr>
<h2 id='precision'> Estimated precision matrix </h2><span id='topic+precision'></span>

<h3>Description</h3>

 
<p>Provides the estimated precision matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> precision( bdgraph.obj, round = 2 ) </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="precision_+3A_bdgraph.obj">bdgraph.obj</code></td>
<td>
<p>object of <code>S3</code> class &quot;<code>bdgraph</code>&quot;, from function <code><a href="#topic+bdgraph">bdgraph</a></code>. 
It also can be an object of <code>S3</code> class <code>"ssgraph"</code>, from the function <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code> of <code>R</code> package <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code>.
</p>
</td></tr>
<tr><td><code id="precision_+3A_round">round</code></td>
<td>
<p> value for rounding all probabilities to the specified number of decimal places. </p>
</td></tr>
</table>


<h3>Value</h3>

<p> matrix which corresponds the estimated precision matrix.</p>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+covariance">covariance</a></code>, <code><a href="#topic+plinks">plinks</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Generating multivariate normal data from a 'circle' graph
data.sim &lt;- bdgraph.sim( n = 70, p = 6, graph = "circle", vis = TRUE )

bdgraph.obj   &lt;- bdgraph( data = data.sim )

precision( bdgraph.obj ) # Estimated precision matrix
  
data.sim $ K   # True precision matrix

## End(Not run)
</code></pre>

<hr>
<h2 id='predict.bdgraph'> Predict function for <code>S3</code> class &quot;<code>bdgraph</code>&quot; </h2><span id='topic+predict.bdgraph'></span>

<h3>Description</h3>

<p>Provides predict values of the results for function <code><a href="#topic+bdgraph">bdgraph</a></code>.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'bdgraph'
predict( object, iter = 1, ... )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.bdgraph_+3A_object">object</code></td>
<td>
<p> object of <code>S3</code> class &quot;<code>bdgraph</code>&quot;, from function <code><a href="#topic+bdgraph">bdgraph</a></code>. </p>
</td></tr>
<tr><td><code id="predict.bdgraph_+3A_iter">iter</code></td>
<td>
<p> number of predictions. </p>
</td></tr>
<tr><td><code id="predict.bdgraph_+3A_...">...</code></td>
<td>
<p> additional parameters. </p>
</td></tr>
</table>


<h3>Value</h3>

 
<p>a <code>matrix</code> containing the predicted datasets, corresponding to the samples from the joint posterior disribtuion. 
</p>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> and Ernst Wit </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>
<p>Mohammadi, A. and Wit, E. C. (2015). Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Bayesian Analysis</em>, 10(1):109-138, <a href="https://doi.org/10.1214/14-BA889">doi:10.1214/14-BA889</a>
</p>
<p>Mohammadi, R., Massam, H. and Letac, G. (2021). Accelerating Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Journal of the American Statistical Association</em>, <a href="https://doi.org/10.1080/01621459.2021.1996377">doi:10.1080/01621459.2021.1996377</a> 
</p>
<p>Dobra, A. and Mohammadi, R. (2018). Loglinear Model Selection and Human Mobility, <em>Annals of Applied Statistics</em>, 12(2):815-845, <a href="https://doi.org/10.1214/18-AOAS1164">doi:10.1214/18-AOAS1164</a>
</p>
<p>Mohammadi, A. et al (2017). Bayesian modelling of Dupuytren disease by using Gaussian copula graphical models, <em>Journal of the Royal Statistical Society: Series C</em>, 66(3):629-645, <a href="https://doi.org/10.1111/rssc.12171">doi:10.1111/rssc.12171</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code>, <code><a href="#topic+bdgraph.dw">bdgraph.dw</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Generating multivariate normal data from a 'random' graph
data.sim &lt;- bdgraph.sim( n = 50, p = 6, size = 7, vis = TRUE )
   
bdgraph.obj &lt;- bdgraph( data = data.sim )
   
predict( bdgraph.obj, iter = 20 )

## End(Not run)
</code></pre>

<hr>
<h2 id='print.bdgraph'>Print function for <code>S3</code> class &quot;<code>bdgraph</code>&quot; </h2><span id='topic+print.bdgraph'></span>

<h3>Description</h3>

<p>Prints the information about the selected graph which could be a graph with links for which their estimated posterior probabilities are greater than 0.5
or graph with the highest posterior probability.
It provides adjacency matrix, size and posterior probability of the selected graph.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'bdgraph'
print( x, ... )</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.bdgraph_+3A_x">x</code></td>
<td>
<p> object of <code>S3</code> class &quot;<code>bdgraph</code>&quot;, from function <code><a href="#topic+bdgraph">bdgraph</a></code>. </p>
</td></tr>
<tr><td><code id="print.bdgraph_+3A_...">...</code></td>
<td>
<p> system reserved (no specific usage). </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> and Ernst Wit </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Generating multivariate normal data from a 'random' graph
data.sim &lt;- bdgraph.sim( n = 50, p = 6, vis = TRUE )
  
bdgraph.obj &lt;- bdgraph( data = data.sim )
   
print( bdgraph.obj )

## End(Not run)
</code></pre>

<hr>
<h2 id='print.sim'> Print function for <code>S3</code> class &quot;<code>sim</code>&quot; </h2><span id='topic+print.sim'></span>

<h3>Description</h3>

<p> Prints the information about the type of data, the sample size, the graph type, the number of nodes, number of links and sparsity of the true graph. </p>


<h3>Usage</h3>

<pre><code class='language-R'> ## S3 method for class 'sim'
print( x, ... ) </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.sim_+3A_x">x</code></td>
<td>
<p> object of <code>S3</code> class &quot;<code>sim</code>&quot;, from function <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>.  </p>
</td></tr>
<tr><td><code id="print.sim_+3A_...">...</code></td>
<td>
<p> system reserved (no specific usage). </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> and Ernst Wit </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+graph.sim">graph.sim</a></code>, <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'># Generating multivariate normal data from a 'random' graph 
data.sim &lt;- bdgraph.sim( n = 20, p = 10, vis = TRUE )
   
print( data.sim )
</code></pre>

<hr>
<h2 id='reinis'> Risk factors of coronary heart disease </h2><span id='topic+reinis'></span>

<h3>Description</h3>

 
<p>The dataset consist of <code class="reqn">6</code> discrete variables as the potential risk factors of coronary heart disease.
The data collected from <code class="reqn">1841</code> men employed of a car factory in Czechoslovakia (Reinis et al. 1981). </p>


<h3>Usage</h3>

<pre><code class='language-R'> data( reinis ) </code></pre>


<h3>Format</h3>

<p> The format is a matrix with <code class="reqn">1841</code> rows (number of individuals) and <code class="reqn">6</code> column (number of variables). </p>


<h3>References</h3>

<p>Edwards and Havranek (1985). A fast procedure for model search in multidimensional contingency tables, <em>Biometrika</em>, 72:339-351
</p>
<p>Reinis et al (1981). Prognostic significance of the risk profile in the prevention of coronary heart disease, <em>Bratis. lek. Listy</em>, 76:137-150
</p>
<p>Mohammadi, A. and Dobra, A. (2017). The <code>R</code> Package <span class="pkg">BDgraph</span> for Bayesian Structure Learning in Graphical Models, <em>ISBA Bulletin</em>, 24(4):11-16
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data( reinis )

summary( reinis )
</code></pre>

<hr>
<h2 id='rgwish'>Sampling from G-Wishart distribution</h2><span id='topic+rgwish'></span>

<h3>Description</h3>

<p>Generates random matrices, distributed according to the G-Wishart distribution with parameters <code class="reqn">b</code> and <code class="reqn">D</code>, <code class="reqn">W_G(b, D)</code> with respect to the graph structure <code class="reqn">G</code>. 
Note this fuction works for both non-decomposable and decomposable graphs. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'> rgwish( n = 1, adj = NULL, b = 3, D = NULL, threshold = 1e-8 ) </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rgwish_+3A_n">n</code></td>
<td>
<p>number of samples required. </p>
</td></tr>
<tr><td><code id="rgwish_+3A_adj">adj</code></td>
<td>

<p>adjacency matrix corresponding to the graph structure which can be non-decomposable or decomposable. It should be an upper triangular matrix in which <code class="reqn">a_{ij}=1</code> 
if there is a link between notes <code class="reqn">i</code> and <code class="reqn">j</code>, otherwise <code class="reqn">a_{ij}=0</code>.
<code>adj</code> could be an object of class <code>"graph"</code>, from function <code><a href="#topic+graph.sim">graph.sim</a></code>.
It also could be an object of class &quot;<code>sim</code>&quot;, from function <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>.
It also could be an object of class &quot;<code>bdgraph</code>&quot;, from functions <code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code> or <code><a href="#topic+bdgraph">bdgraph</a></code>.
</p>
</td></tr>
<tr><td><code id="rgwish_+3A_b">b</code></td>
<td>
<p>degree of freedom for G-Wishart distribution, <code class="reqn">W_G(b, D)</code>. </p>
</td></tr>
<tr><td><code id="rgwish_+3A_d">D</code></td>
<td>
<p>positive definite <code class="reqn">(p \times p)</code> &quot;scale&quot; matrix for G-Wishart distribution, <code class="reqn">W_G(b, D)</code>. 
The default is an identity matrix.</p>
</td></tr>
<tr><td><code id="rgwish_+3A_threshold">threshold</code></td>
<td>
<p> threshold value for the convergence of sampling algorithm from G-Wishart.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Sampling from G-Wishart distribution, <code class="reqn">K \sim W_G(b, D)</code>, with density:
</p>
<p style="text-align: center;"><code class="reqn">Pr(K) \propto |K| ^ {(b - 2) / 2} \exp \left\{- \frac{1}{2} \mbox{trace}(K \times D)\right\},</code>
</p>

<p>which <code class="reqn">b &gt; 2</code> is the degree of freedom and <code class="reqn">D</code> is a symmetric positive definite matrix.
</p>


<h3>Value</h3>

<p>A numeric array, say <code class="reqn">A</code>, of dimension <code class="reqn">(p \times p \times n)</code>, where each <code class="reqn">A[,,i]</code> is a positive 
definite matrix, a realization of the G-Wishart distribution, <code class="reqn">W_G(b, D)</code>.
Note, for the case <code class="reqn">n=1</code>, the output is a matrix.
</p>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> </p>


<h3>References</h3>

<p>Lenkoski, A. (2013). A direct sampler for G-Wishart variates, <em>Stat</em>, 2:119-128, <a href="https://doi.org/10.1002/sta4.23">doi:10.1002/sta4.23</a>
</p>
<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>
<p>Mohammadi, A. and Wit, E. C. (2015). Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Bayesian Analysis</em>, 10(1):109-138, <a href="https://doi.org/10.1214/14-BA889">doi:10.1214/14-BA889</a>
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+gnorm">gnorm</a></code>, <code><a href="#topic+rwish">rwish</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'># Generating a 'circle' graph as a non-decomposable graph
adj &lt;- graph.sim( p = 5, graph = "circle" )
adj    # adjacency of graph with 5 nodes
  
sample &lt;- rgwish( n = 1, adj = adj, b = 3, D = diag( 5 ) )
round( sample, 2 ) 

sample &lt;- rgwish( n = 5, adj = adj )
round( sample, 2 )  
</code></pre>

<hr>
<h2 id='rmvnorm'> Generate data from the multivariate Normal distribution </h2><span id='topic+rmvnorm'></span>

<h3>Description</h3>

<p>Random generation function from the multivariate Normal distribution with mean equal to <code class="reqn">mean</code> and covariance matrix <code class="reqn">sigma</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> 
    rmvnorm( n = 10, mean = rep( 0, length = ncol( sigma ) ), 
             sigma = diag( length( mean ) ) ) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rmvnorm_+3A_n">n</code></td>
<td>
<p> Number of observations. </p>
</td></tr>
<tr><td><code id="rmvnorm_+3A_mean">mean</code></td>
<td>
<p> Mean vector, default is <code class="reqn">rep( 0, length = ncol( sigma ) )</code>. </p>
</td></tr>
<tr><td><code id="rmvnorm_+3A_sigma">sigma</code></td>
<td>
<p> positive definite covariance matrix, default is <code class="reqn"> diag( length( mean ) ) </code>. </p>
</td></tr>
</table>


<h3>Value</h3>

<p> A numeric matrix with rows equal to <code class="reqn">n</code> and columns equal to <code class="reqn">length( mean )</code>. </p>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> </p>


<h3>See Also</h3>

<p><code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>, <code><a href="#topic+rwish">rwish</a></code>, <code><a href="#topic+rgwish">rgwish</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>mean  &lt;- c( 5, 20 )       
sigma &lt;- matrix( c( 4, 2,
                    2, 5 ), 2, 2 )  # covariance matrix

sample &lt;- rmvnorm( n = 500, mean = mean, sigma = sigma )
plot( sample )
</code></pre>

<hr>
<h2 id='roc'> Build a ROC curve </h2><span id='topic+roc'></span>

<h3>Description</h3>

 
<p>This function builds a ROC curve specifically for graph structure learning and returns a &ldquo;roc&rdquo; object, a list of class
&ldquo;roc&rdquo;. This object can be <code>prin</code>ted, <code>plot</code>ted, or
passed to the functions <code><a href="pROC.html#topic+roc">pROC::roc()</a></code>, <code><a href="pROC.html#topic+ci">pROC::ci()</a></code>, <code><a href="pROC.html#topic+smooth.roc">pROC::smooth.roc()</a></code>
and <code><a href="pROC.html#topic+coords">pROC::coords()</a></code>. Additionally, two <code>roc</code> objects can be compared with <code><a href="pROC.html#topic+roc.test">pROC::roc.test()</a></code>. 
This function is based on the <code><a href="pROC.html#topic+roc">roc</a></code> function of <code>R</code> package <code>pROC</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> 
roc( pred, actual, auc = TRUE, smooth = FALSE, plot = FALSE, quiet = TRUE, ... )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="roc_+3A_pred">pred</code></td>
<td>

<p>adjacency matrix (or a vector) corresponding to an estimated graph. 
It can be an object with <code>S3</code> class &quot;<code>bdgraph</code>&quot; from function <code><a href="#topic+bdgraph">bdgraph</a></code>. 
It can be an object of <code>S3</code> class <code>"ssgraph"</code>, from the function <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code> of <code>R</code> package <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code>.
It can be a <code><a href="base.html#topic+numeric">numeric</a></code> or <code><a href="base.html#topic+ordered">ordered</a></code> vector of the same length than <code>actual</code>, containing the predicted
value of each observation.
</p>
</td></tr>
<tr><td><code id="roc_+3A_actual">actual</code></td>
<td>
<p> adjacency matrix (or a vector) corresponding to the actual graph structure in which <code class="reqn">a_{ij}=1</code> if there is a link between notes <code class="reqn">i</code> and <code class="reqn">j</code>, otherwise <code class="reqn">a_{ij}=0</code>.
It can be an object with <code>S3</code> class &quot;<code>sim</code>&quot; from function <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>.
It can be an object with <code>S3</code> class <code>"graph"</code> from function <code><a href="#topic+graph.sim">graph.sim</a></code>.
It can be a factor, numeric or character vector of responses (true class), typically encoded with 0 (controls) and 1 (cases).
Only two classes can be used in a ROC curve. </p>
</td></tr>
<tr><td><code id="roc_+3A_smooth">smooth</code></td>
<td>
<p>if TRUE, the ROC curve is passed to <code><a href="stats.html#topic+smooth">smooth</a></code>
to be smoothed.
</p>
</td></tr>
<tr><td><code id="roc_+3A_auc">auc</code></td>
<td>
<p>compute the area under the curve (AUC)? If <code>TRUE</code>
(default), additional arguments can be passed to <code><a href="pROC.html#topic+auc">pROC::auc()</a></code>.
</p>
</td></tr>
<tr><td><code id="roc_+3A_plot">plot</code></td>
<td>
<p>plot the ROC curve? If <code>TRUE</code>, additional
arguments can be passed to <code><a href="pROC.html#topic+plot.roc">pROC::plot.roc()</a></code>.
</p>
</td></tr>
<tr><td><code id="roc_+3A_quiet">quiet</code></td>
<td>
<p>if <code>TRUE</code>, turn off <code><a href="base.html#topic+message">message</a></code>s when <code>direction</code> and <code>levels</code> are auto-detected.
</p>
</td></tr>
<tr><td><code id="roc_+3A_...">...</code></td>
<td>
<p> further arguments to be passed to <code><a href="pROC.html#topic+roc">pROC::roc()</a></code>. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>If the data contained any <code>NA</code> value and <code>na.rm=FALSE</code>, <code>NA</code> is
returned. Otherwise, if <code>smooth=FALSE</code>, a list of class
&ldquo;roc&rdquo; with the following fields: 
</p>
<table>
<tr><td><code>auc</code></td>
<td>
<p>if called with <code>auc=TRUE</code>, a numeric of class &ldquo;auc&rdquo; as
defined in <code><a href="pROC.html#topic+auc">pROC::auc()</a></code>.
</p>
</td></tr>
<tr><td><code>ci</code></td>
<td>
<p>if called with <code>ci=TRUE</code>, a numeric of class &ldquo;ci&rdquo; as
defined in <code><a href="pROC.html#topic+ci">pROC::ci()</a></code>.
</p>
</td></tr>
<tr><td><code>response</code></td>
<td>
<p>the response vector. Patients whose response is not
<code><a href="base.html#topic++25in+25">%in%</a></code> <code>levels</code> are discarded. If <code>NA</code> values
were removed, a <code>na.action</code> attribute similar
to <code><a href="stats.html#topic+na.omit">na.omit</a></code> stores the row numbers. 
</p>
</td></tr>
<tr><td><code>predictor</code></td>
<td>
<p>predictor vector converted to numeric as used to build the ROC
curve. Patients whose response is not <code><a href="base.html#topic++25in+25">%in%</a></code> <code>levels</code> are discarded. If
<code>NA</code> values were removed, a <code>na.action</code> attribute similar
to <code><a href="stats.html#topic+na.omit">na.omit</a></code> stores the row numbers.
</p>
</td></tr>
<tr><td><code>original.predictor</code>, <code>original.response</code></td>
<td>
<p>response and predictor vectors as passed in argument.</p>
</td></tr>
<tr><td><code>levels</code></td>
<td>
<p>levels of the response as defined in argument.</p>
</td></tr>
<tr><td><code>controls</code></td>
<td>
<p>predictor values for the control observations.</p>
</td></tr>
<tr><td><code>cases</code></td>
<td>
<p>predictor values for the cases.</p>
</td></tr>
<tr><td><code>percent</code></td>
<td>
<p>if the sensitivities, specificities and AUC are
reported in percent, as defined in argument.
</p>
</td></tr>
<tr><td><code>direction</code></td>
<td>
<p>direction of the comparison, as defined in argument.</p>
</td></tr>
<tr><td><code>fun.sesp</code></td>
<td>
<p>function used to compute sensitivities and specificities.
Will be re-used in bootstrap operations.</p>
</td></tr>
<tr><td><code>sensitivities</code></td>
<td>
<p>sensitivities defining the ROC curve.</p>
</td></tr>
<tr><td><code>specificities</code></td>
<td>
<p>specificities defining the ROC curve.</p>
</td></tr>
<tr><td><code>thresholds</code></td>
<td>
<p>thresholds at which the sensitivities and
specificities were computed. See below for details.
</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>how the function was called. See <code><a href="base.html#topic+match.call">match.call</a></code> for
more details.
</p>
</td></tr>
</table>
<p>If <code>smooth=TRUE</code> a list of class &ldquo;smooth.roc&rdquo; as returned
by <code><a href="pROC.html#topic+smooth">pROC::smooth()</a></code>, with or without additional elements
<code>auc</code> and <code>ci</code> (according to the call).
</p>


<h3>Author(s)</h3>

<p>Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a>; Lucas Vogels <a href="mailto:l.f.o.vogels@uva.nl">l.f.o.vogels@uva.nl</a></p>


<h3>References</h3>

<p>Tom Fawcett (2006) &ldquo;An introduction to ROC analysis&rdquo;. <em>Pattern
Recognition Letters</em> <b>27</b>, 861&ndash;874, <a href="https://doi.org/10.1016/j.patrec.2005.10.010">doi:10.1016/j.patrec.2005.10.010</a>
</p>
<p>Xavier Robin, Natacha Turck, Alexandre Hainard, <em>et al.</em>
(2011) &ldquo;pROC: an open-source package for R and S+ to analyze and
compare ROC curves&rdquo;. <em>BMC Bioinformatics</em>, <b>7</b>, 77, 
<a href="https://doi.org/10.1186/1471-2105-12-77">doi:10.1186/1471-2105-12-77</a>.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+plotroc">plotroc</a></code>, <code><a href="pROC.html#topic+plot.roc">pROC::plot.roc()</a></code>, <code><a href="pROC.html#topic+auc">pROC::auc()</a></code>, <code><a href="pROC.html#topic+print.roc">pROC::print.roc()</a></code>, <code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code>, <code><a href="#topic+compare">compare</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set.seed( 5 )

# Generating multivariate normal data from a 'scale-free' graph
data.sim = bdgraph.sim( n = 200, p = 15, graph = "scale-free", vis = TRUE )

# Running BDMCMC algorithm 
sample.bdmcmc = bdgraph( data = data.sim, algorithm = "bdmcmc", iter = 10000 )

# ROC curve for BDMCMC algorithm 
roc.bdmcmc = BDgraph::roc( pred = sample.bdmcmc, actual = data.sim, plot = TRUE )

# Running RJMCMC algorithm
sample.rjmcmc = bdgraph( data = data.sim, algorithm = "rjmcmc", iter = 10000 )

# ROC curve for RJMCMC algorithm
roc.rjmcmc = BDgraph::roc( pred = sample.rjmcmc, actual = data.sim, plot = TRUE )

# ROC curve for both BDMCMC and RJMCMC algorithms
pROC::ggroc( list( BDMCMC = roc.bdmcmc, RJMCMC = roc.rjmcmc ) )

## End(Not run)
</code></pre>

<hr>
<h2 id='rwish'> Sampling from Wishart distribution </h2><span id='topic+rwish'></span>

<h3>Description</h3>

<p>Generates random matrices, distributed according to the Wishart distribution with parameters <code class="reqn">b</code> and <code class="reqn">D</code>, <code class="reqn">W(b, D)</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> rwish( n = 1, p = 2, b = 3, D = diag( p ) ) </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rwish_+3A_n">n</code></td>
<td>
<p> number of samples required. </p>
</td></tr>
<tr><td><code id="rwish_+3A_p">p</code></td>
<td>
<p> number of variables (nodes). </p>
</td></tr>
<tr><td><code id="rwish_+3A_b">b</code></td>
<td>
<p> degree of freedom for Wishart distribution, <code class="reqn">W(b, D)</code>. </p>
</td></tr>
<tr><td><code id="rwish_+3A_d">D</code></td>
<td>
<p> positive definite <code class="reqn">(p \times p)</code> &quot;scale&quot; matrix for Wishart distribution, <code class="reqn">W(b, D)</code>. 
The default is an identity matrix. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Sampling from Wishart distribution, <code class="reqn">K \sim W(b, D)</code>, with density:
</p>
<p style="text-align: center;"><code class="reqn">Pr(K) \propto |K| ^ {(b - 2) / 2} \exp \left\{- \frac{1}{2} \mbox{trace}(K \times D)\right\},</code>
</p>

<p>which <code class="reqn">b &gt; 2</code> is the degree of freedom and <code class="reqn">D</code> is a symmetric positive definite matrix.
</p>


<h3>Value</h3>

<p>A numeric array, say <code class="reqn">A</code>, of dimension <code class="reqn">(p \times p \times n)</code>, where each <code class="reqn">A[,,i]</code> is a positive 
definite matrix, a realization of the Wishart distribution <code class="reqn">W(b, D)</code>.
Note, for the case <code class="reqn">n=1</code>, the output is a matrix.
</p>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> </p>


<h3>References</h3>

<p>Lenkoski, A. (2013). A direct sampler for G-Wishart variates, <em>Stat</em>, 2:119-128, <a href="https://doi.org/10.1002/sta4.23">doi:10.1002/sta4.23</a>
</p>
<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+gnorm">gnorm</a></code>, <code><a href="#topic+rgwish">rgwish</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>sample &lt;- rwish( n = 3, p = 5, b = 3, D = diag( 5 ) )

round( sample, 2 )  
</code></pre>

<hr>
<h2 id='select'> Graph selection </h2><span id='topic+select'></span>

<h3>Description</h3>

<p>Provides the selected graph which, based on input, could be a graph with links for which their estimated posterior probabilities are greater than 0.5 (default)
or a graph with the highest posterior probability; see examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> select( bdgraph.obj, cut = NULL, vis = FALSE ) </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="select_+3A_bdgraph.obj">bdgraph.obj</code></td>
<td>
 
<p>matrix in which each element response to the weight of the links.
It can be an object of <code>S3</code> class &quot;<code>bdgraph</code>&quot;, from function <code><a href="#topic+bdgraph">bdgraph</a></code>.
It can be an object of <code>S3</code> class <code>"ssgraph"</code>, from the function <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code> of <code>R</code> package <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code>.
</p>
</td></tr>
<tr><td><code id="select_+3A_cut">cut</code></td>
<td>
<p> threshold for including the links in the selected graph based on the estimated posterior probabilities of the links; see the examples. </p>
</td></tr>  
<tr><td><code id="select_+3A_vis">vis</code></td>
<td>
<p> visualize the selected graph structure. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>	An adjacency matrix corresponding to the selected graph.</p>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> and Ernst Wit </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>
<p>Mohammadi, A. and Wit, E. C. (2015). Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Bayesian Analysis</em>, 10(1):109-138, <a href="https://doi.org/10.1214/14-BA889">doi:10.1214/14-BA889</a>
</p>
<p>Mohammadi, R., Massam, H. and Letac, G. (2021). Accelerating Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Journal of the American Statistical Association</em>, <a href="https://doi.org/10.1080/01621459.2021.1996377">doi:10.1080/01621459.2021.1996377</a> 
</p>
<p>Dobra, A. and Mohammadi, R. (2018). Loglinear Model Selection and Human Mobility, <em>Annals of Applied Statistics</em>, 12(2):815-845, <a href="https://doi.org/10.1214/18-AOAS1164">doi:10.1214/18-AOAS1164</a>
</p>
<p>Mohammadi, A. et al (2017). Bayesian modelling of Dupuytren disease by using Gaussian copula graphical models, <em>Journal of the Royal Statistical Society: Series C</em>, 66(3):629-645, <a href="https://doi.org/10.1111/rssc.12171">doi:10.1111/rssc.12171</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Generating multivariate normal data from a 'random' graph
data.sim &lt;- bdgraph.sim( n = 50, p = 6, size = 7, vis = TRUE )
  
bdgraph.obj &lt;- bdgraph( data = data.sim )
   
select( bdgraph.obj )
  
bdgraph.obj &lt;- bdgraph( data = data.sim, save = TRUE )
  
select( bdgraph.obj )
  
select( bdgraph.obj, cut = 0.5, vis = TRUE )

## End(Not run)
</code></pre>

<hr>
<h2 id='sparsity'> Compute the sparsity of a graph </h2><span id='topic+sparsity'></span>

<h3>Description</h3>

<p> Compute the sparsity of a graph/network or an object of calss <code>"graph"</code> from function <code><a href="#topic+graph.sim">graph.sim</a></code> or an object of calss &quot;<code>sim</code>&quot; from function <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>. </p>


<h3>Usage</h3>

<pre><code class='language-R'> sparsity( adj ) </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sparsity_+3A_adj">adj</code></td>
<td>
     
<p>adjacency matrix corresponding to a graph structure in which <code class="reqn">a_{ij}=1</code> if there is a link between notes <code class="reqn">i</code> and <code class="reqn">j</code>, otherwise <code class="reqn">a_{ij}=0</code>.
It can be an object with <code>S3</code> class <code>"graph"</code> from function <code><a href="#topic+graph.sim">graph.sim</a></code>.
It can be an object with <code>S3</code> class &quot;<code>sim</code>&quot; from function <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p> value corresponding to the graph sparsity which is the proportion of the non-links (non-zero elements) in <code>adj</code>. </p>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+graph.sim">graph.sim</a></code>, <code><a href="#topic+adj2link">adj2link</a></code>, <code><a href="#topic+link2adj">link2adj</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'># Generating a 'random' graph 
adj &lt;- graph.sim( p = 10, graph = "random", prob = 0.4, vis = TRUE )

sparsity( adj )
</code></pre>

<hr>
<h2 id='summary.bdgraph'> Summary function for <code>S3</code> class &quot;<code>bdgraph</code>&quot; </h2><span id='topic+summary.bdgraph'></span>

<h3>Description</h3>

<p>Provides a summary of the results for function <code><a href="#topic+bdgraph">bdgraph</a></code>.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'bdgraph'
summary( object, round = 2, vis = TRUE, ... )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.bdgraph_+3A_object">object</code></td>
<td>
<p> object of <code>S3</code> class &quot;<code>bdgraph</code>&quot;, from function <code><a href="#topic+bdgraph">bdgraph</a></code>. </p>
</td></tr>
<tr><td><code id="summary.bdgraph_+3A_round">round</code></td>
<td>
<p> value for rounding all probabilities to the specified number of decimal places. </p>
</td></tr>
<tr><td><code id="summary.bdgraph_+3A_vis">vis</code></td>
<td>
<p> visualize the results. </p>
</td></tr>
<tr><td><code id="summary.bdgraph_+3A_...">...</code></td>
<td>
<p> additional plotting parameters for the case <code>vis = TRUE</code>. See <code><a href="#topic+plot.graph">plot.graph</a></code>. </p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>selected_g</code></td>
<td>
<p>adjacency matrix corresponding to the selected graph which has the highest posterior probability.</p>
</td></tr>
<tr><td><code>p_links</code></td>
<td>
<p>upper triangular matrix corresponding to the posterior probabilities of all possible links.</p>
</td></tr>
<tr><td><code>K_hat</code></td>
<td>
<p>estimated precision matrix.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> and Ernst Wit </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Generating multivariate normal data from a 'random' graph
data.sim &lt;- bdgraph.sim( n = 50, p = 6, size = 7, vis = TRUE )
   
bdgraph.obj &lt;- bdgraph( data = data.sim )
   
summary( bdgraph.obj )
   
bdgraph.obj &lt;- bdgraph( data = data.sim, save = TRUE )
   
summary( bdgraph.obj )
   
summary( bdgraph.obj, vis = FALSE )

## End(Not run)
</code></pre>

<hr>
<h2 id='surveyData'> Labor force survey data </h2><span id='topic+surveyData'></span>

<h3>Description</h3>

 
<p>The survey dataset concerns 1002 males in the U.S labor force, described by Hoff (2007). 
The seven observed variables which have been measured on various scales are as follow: 
the income (income), degree (degree), the number of children (children), parents income (pincome), 
parents degree (pdegree), number of parents children (pchildren), and age (age).  
</p>


<h3>Usage</h3>

<pre><code class='language-R'> data( surveyData ) </code></pre>


<h3>Format</h3>

<p> The format is a matrix with 1002 rows (number of individuals) and 7 column (number of variables). </p>


<h3>References</h3>

<p>Hoff, P. (2007). Extending the rank likelihood for semiparametric copula estimation, <em>The Annals of Applied Statistics, 1(1)</em>, 265-283.
</p>
<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data( surveyData )

summary( surveyData )
</code></pre>

<hr>
<h2 id='traceplot'> Trace plot of graph size </h2><span id='topic+traceplot'></span>

<h3>Description</h3>

<p>Trace plot for graph size for the objects of <code>S3</code> class &quot;<code>bdgraph</code>&quot;, from function <code><a href="#topic+bdgraph">bdgraph</a></code>. 
It is a tool for monitoring the convergence of the sampling algorithms, BDMCMC and RJMCMC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> traceplot ( bdgraph.obj, acf = FALSE, pacf = FALSE, main = NULL, ... ) </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="traceplot_+3A_bdgraph.obj">bdgraph.obj</code></td>
<td>
<p> object of <code>S3</code> class &quot;<code>bdgraph</code>&quot;, from function <code><a href="#topic+bdgraph">bdgraph</a></code>. 
It also can be an object of <code>S3</code> class <code>"ssgraph"</code>, from the function <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code> of <code>R</code> package <code><a href="ssgraph.html#topic+ssgraph">ssgraph::ssgraph()</a></code>.
</p>
</td></tr>
<tr><td><code id="traceplot_+3A_acf">acf</code></td>
<td>
<p> visualize the autocorrelation functions for graph size. </p>
</td></tr>
<tr><td><code id="traceplot_+3A_pacf">pacf</code></td>
<td>
<p> visualize the partial autocorrelations for graph size. </p>
</td></tr>
<tr><td><code id="traceplot_+3A_main">main</code></td>
<td>
<p> graphical parameter (see plot). </p>
</td></tr>
<tr><td><code id="traceplot_+3A_...">...</code></td>
<td>
<p> system reserved (no specific usage). </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> and Ernst Wit </p>


<h3>References</h3>

<p>Mohammadi, R. and Wit, E. C. (2019). <span class="pkg">BDgraph</span>: An <code>R</code> Package for Bayesian Structure Learning in Graphical Models, <em>Journal of Statistical Software</em>, 89(3):1-30, <a href="https://doi.org/10.18637/jss.v089.i03">doi:10.18637/jss.v089.i03</a> 
</p>
<p>Mohammadi, A. and Wit, E. C. (2015). Bayesian Structure Learning in Sparse Gaussian Graphical Models, <em>Bayesian Analysis</em>, 10(1):109-138, <a href="https://doi.org/10.1214/14-BA889">doi:10.1214/14-BA889</a>
</p>
<p>Dobra, A. and Mohammadi, R. (2018). Loglinear Model Selection and Human Mobility, <em>Annals of Applied Statistics</em>, 12(2):815-845, <a href="https://doi.org/10.1214/18-AOAS1164">doi:10.1214/18-AOAS1164</a>
</p>
<p>Mohammadi, A. et al (2017). Bayesian modelling of Dupuytren disease by using Gaussian copula graphical models, <em>Journal of the Royal Statistical Society: Series C</em>, 66(3):629-645, <a href="https://doi.org/10.1111/rssc.12171">doi:10.1111/rssc.12171</a>
</p>
<p>Mohammadi, A. and Dobra, A. (2017). The <code>R</code> Package <span class="pkg">BDgraph</span> for Bayesian Structure Learning in Graphical Models, <em>ISBA Bulletin</em>, 24(4):11-16
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+plotcoda">plotcoda</a></code>, <code><a href="#topic+bdgraph">bdgraph</a></code>, <code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Generating multivariate normal data from a 'random' graph
data.sim &lt;- bdgraph.sim( n = 50, p = 6, size = 7, vis = TRUE )
   
bdgraph.obj &lt;- bdgraph( data = data.sim, iter = 10000, burnin = 0, save = TRUE )
   
traceplot( bdgraph.obj )
   
traceplot( bdgraph.obj, acf = TRUE, pacf = TRUE )

## End(Not run)
</code></pre>

<hr>
<h2 id='transfer'> transfer for count data </h2><span id='topic+transfer'></span>

<h3>Description</h3>

<p> Transfers count data, by counting the duplicated rows.  </p>


<h3>Usage</h3>

<pre><code class='language-R'> transfer( r_data ) </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="transfer_+3A_r_data">r_data</code></td>
<td>
<p> (<code class="reqn">n \times p</code>) <code>matrix</code> or a <code>data.frame</code> corresponding to the data (<code class="reqn">n</code> is the sample size and <code class="reqn">p</code> is the number of variables). </p>
</td></tr>
</table>


<h3>Value</h3>

<p> (<code class="reqn">n \times p+1</code>) matrix of transferred data, in which the last column is the frequency of duplicated rows.
</p>


<h3>Author(s)</h3>

<p> Reza Mohammadi <a href="mailto:a.mohammadi@uva.nl">a.mohammadi@uva.nl</a> and Adrian Dobra </p>


<h3>References</h3>

<p>Mohammadi, A. and Dobra, A. (2017). The <code>R</code> Package <span class="pkg">BDgraph</span> for Bayesian Structure Learning in Graphical Models, <em>ISBA Bulletin</em>, 24(4):11-16
</p>
<p>Dobra, A. and Mohammadi, R. (2018). Loglinear Model Selection and Human Mobility, <em>Annals of Applied Statistics</em>, 12(2):815-845, <a href="https://doi.org/10.1214/18-AOAS1164">doi:10.1214/18-AOAS1164</a>
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+bdgraph.mpl">bdgraph.mpl</a></code>, <code><a href="#topic+bdgraph.sim">bdgraph.sim</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'># Generating multivariate binary data from a 'random' graph
data.sim &lt;- bdgraph.sim( n = 12, p = 4, size = 4, type = "binary" )
r_data   &lt;- data.sim $ data
r_data
   
# Transfer the data
transfer( r_data )
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
