<!DOCTYPE html><html><head><title>Help for package beautier</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {beautier}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#alpha_parameter_to_xml'><p>Internal function</p></a></li>
<li><a href='#are_clock_models'><p>Determine if x consists out of clock_models objects</p></a></li>
<li><a href='#are_equal_mcmcs'><p>Determine if two MCMCs are equal.</p></a></li>
<li><a href='#are_equal_screenlogs'><p>Determine if two screenlogs are equal.</p></a></li>
<li><a href='#are_equal_tracelogs'><p>Determine if two tracelogs are equal.</p></a></li>
<li><a href='#are_equal_treelogs'><p>Determine if two treelogs are equal.</p></a></li>
<li><a href='#are_equal_xml_files'><p>Determine if XML files result in equal trees</p></a></li>
<li><a href='#are_equal_xml_lines'><p>Determine if XML lines result in equal trees</p></a></li>
<li><a href='#are_equivalent_xml_files'><p>Internal function</p></a></li>
<li><a href='#are_equivalent_xml_lines'><p>Determine if XML lines result in equivalent trees</p></a></li>
<li><a href='#are_equivalent_xml_lines_all'><p>Determine if XML lines result in equivalent trees</p></a></li>
<li><a href='#are_equivalent_xml_lines_loggers'><p>Determine if XML operator lines result in equivalent trees</p></a></li>
<li><a href='#are_equivalent_xml_lines_operators'><p>Determine if XML operator lines result in equivalent trees</p></a></li>
<li><a href='#are_equivalent_xml_lines_section'><p>Determine if XML lines result in equivalent trees</p></a></li>
<li><a href='#are_fasta_filenames'><p>Checks if all filenames have a FASTA filename extension</p></a></li>
<li><a href='#are_ids'><p>Determine if x consists out of IDs</p></a></li>
<li><a href='#are_init_clock_models'><p>Determine if x consists out of initialized clock_models objects</p></a></li>
<li><a href='#are_init_mrca_priors'><p>Determine if x consists out of initialized MRCA priors</p></a></li>
<li><a href='#are_init_site_models'><p>Determine if x consists out of initialized site_models objects</p></a></li>
<li><a href='#are_init_tree_priors'><p>Determine if x consists out of initialized tree_priors objects</p></a></li>
<li><a href='#are_mrca_align_ids_in_fasta'><p>Determine if the MRCA priors' alignment IDs are present in the FASTA files</p></a></li>
<li><a href='#are_mrca_priors'><p>Determine if x consists out of MRCA priors</p></a></li>
<li><a href='#are_mrca_taxon_names_in_fasta'><p>Determine if the MRCA priors' taxa names are present in the FASTA files</p></a></li>
<li><a href='#are_rln_clock_models'><p>Are the clock models Relaxed Log-Normal clock models?</p></a></li>
<li><a href='#are_site_models'><p>Determine if x consists out of site_models objects</p></a></li>
<li><a href='#are_tree_priors'><p>Determine if x consists out of tree_priors objects</p></a></li>
<li><a href='#b_pop_sizes_param_to_xml'><p>Internal function</p></a></li>
<li><a href='#b_pop_sizes_parameter_to_xml'><p>Internal function</p></a></li>
<li><a href='#bd_tree_prior_to_xml_prior_distr'><p>Creates the tree prior section in the prior section of</p>
the prior section of the distribution section
of a BEAST2 XML parameter file for a Birth-Death tree prior</a></li>
<li><a href='#beautier-package'><p><code>beautier</code>: A package to create a <code>BEAST2</code> input file.</p></a></li>
<li><a href='#beta_parameter_to_xml'><p>Internal function</p></a></li>
<li><a href='#cbs_tree_prior_to_xml_prior_distr'><p>Creates the tree prior section in the prior section of</p>
the prior section of the distribution section
of a BEAST2 XML parameter file for a Birth-Death tree prior</a></li>
<li><a href='#ccp_tree_prior_to_xml_prior_distr'><p>Creates the tree prior section in the prior section of</p>
the prior section of the distribution section
of a BEAST2 XML parameter file for a
Coalescent Constant Population tree prior</a></li>
<li><a href='#ccp_tree_prior_to_xml_state'><p>Convert a CCP tree prior</p>
to the XML as part of the <code>state</code> section</a></li>
<li><a href='#cep_tree_prior_to_xml_prior_distr'><p>Creates the tree prior section in the prior section of</p>
the prior section of the distribution section
of a BEAST2 XML parameter file for a
Coalescent Exponential Population tree prior</a></li>
<li><a href='#check_alignment_id'><p>Check if the <code>alignment_id</code> is valid.</p></a></li>
<li><a href='#check_beauti_options'><p>Check if the <code>beauti_options</code> is a valid <code>beauti_options</code> object.</p></a></li>
<li><a href='#check_clock_model'><p>Check if the clock model is a valid clock model.</p></a></li>
<li><a href='#check_clock_models'><p>Check if the object is a list of one or more clock models.</p></a></li>
<li><a href='#check_empty_beautier_folder'><p>Internal function</p></a></li>
<li><a href='#check_file_and_model_agree'><p>Checks if the input FASTA file and the inference model agree.</p></a></li>
<li><a href='#check_file_exists'><p>Function to check if a file exists.</p>
Calls <code>stop</code> if the file is absent</a></li>
<li><a href='#check_filename'><p>Check if the 'filename' is valid</p></a></li>
<li><a href='#check_gamma_site_model'><p>Checks if the parameter is a valid gamma site model</p></a></li>
<li><a href='#check_gamma_site_model_names'><p>Checks if the gamma site model has the right list elements' names</p></a></li>
<li><a href='#check_gtr_site_model'><p>Check if the <code>gtr_site_model</code> is a valid</p>
GTR nucleotide substitution model.</a></li>
<li><a href='#check_gtr_site_model_names'><p>Check if the <code>gtr_site_model</code> has the list elements</p>
of a valid <code>gtr_site_model</code> object.</a></li>
<li><a href='#check_inference_model'><p>Check if the supplied object is a valid</p>
Bayesian phylogenetic inference model.</a></li>
<li><a href='#check_inference_models'><p>Check if the <code>inference_model</code> is a valid BEAUti inference model.</p></a></li>
<li><a href='#check_is_monophyletic'><p>Check if <code>is_monophyletic</code> has a valid value.</p></a></li>
<li><a href='#check_log_mode'><p>Check if the supplied <code>mode</code> is a valid logging mode.</p></a></li>
<li><a href='#check_log_sort'><p>Check if the supplied <code>sort</code> is a valid logging sorting option.</p></a></li>
<li><a href='#check_mcmc'><p>Check if the MCMC is a valid MCMC object.</p></a></li>
<li><a href='#check_mcmc_list_element_names'><p>Check if the MCMC has the list elements of a valid MCMC object.</p></a></li>
<li><a href='#check_mcmc_values'><p>Check if the MCMC has the list elements with valid values</p>
for being a valid MCMC object.</a></li>
<li><a href='#check_mrca_prior'><p>Check if the MRCA prior is a valid MRCA prior.</p></a></li>
<li><a href='#check_mrca_prior_name'><p>Check if <code>mrca_prior_name</code> is a valid MRCA prior name.</p></a></li>
<li><a href='#check_mrca_prior_names'><p>Check if the MRCA prior,</p>
which is a list, has all the named elements.</a></li>
<li><a href='#check_mrca_prior_taxa_names'><p>Check the MRCA prior's taxon names are valid.</p></a></li>
<li><a href='#check_ns_mcmc'><p>Check if this an MCMC that uses Nested Sampling</p>
to estimate a marginal likelihood.</a></li>
<li><a href='#check_param'><p>Check if the parameter is a valid parameter</p></a></li>
<li><a href='#check_param_names'><p>Check if the <code>param</code> has the list elements</p>
of a valid <code>param</code> object.</a></li>
<li><a href='#check_param_types'><p>Check if the <code>param</code> has the list elements</p>
of the right type for a valid <code>param</code> object.</a></li>
<li><a href='#check_phylogeny'><p>Check if the phylogeny is a valid phylogeny object.</p></a></li>
<li><a href='#check_rename_fun'><p>Check if the rename function is a valid filename rename function</p></a></li>
<li><a href='#check_rln_clock_model'><p>Check if the clock model is a valid clock model.</p></a></li>
<li><a href='#check_screenlog'><p>Check if a <code>screenlog</code> is valid.</p></a></li>
<li><a href='#check_screenlog_names'><p>Check if the <code>screenlog</code> has the list elements</p>
of a valid <code>screenlog</code> object.</a></li>
<li><a href='#check_screenlog_values'><p>Check if the screenlog has the list elements with valid values</p>
for being a valid screenlog object.</a></li>
<li><a href='#check_site_model'><p>Check if the site model is a valid site model</p></a></li>
<li><a href='#check_site_model_names'><p>Check if the <code>site_model</code> has the list elements</p>
of a valid <code>site_model</code> object.</a></li>
<li><a href='#check_site_model_types'><p>Check if the <code>site_model</code> has the list elements</p>
of the right type for a valid <code>site_model</code> object.</a></li>
<li><a href='#check_site_models'><p>Check if the object is a list of one or more site models.</p></a></li>
<li><a href='#check_store_every'><p>Check if <code>store_every</code> holds a valid value</p></a></li>
<li><a href='#check_strict_clock_model'><p>Check if the clock model is a valid clock model.</p></a></li>
<li><a href='#check_tn93_site_model'><p>Check if the <code>tn93_site_model</code> is a valid</p>
TN93 nucleotide substitution model.</a></li>
<li><a href='#check_tn93_site_model_names'><p>Check if the <code>tn93_site_model</code> has the list elements</p>
of a valid <code>tn93_site_model</code> object.</a></li>
<li><a href='#check_tracelog'><p>Check if a <code>tracelog</code> is valid.</p></a></li>
<li><a href='#check_tracelog_names'><p>Check if the <code>tracelog</code> has the list elements</p>
of a valid <code>tracelog</code> object.</a></li>
<li><a href='#check_tracelog_values'><p>Check if the tracelog has the list elements with valid values</p>
for being a valid tracelog object.</a></li>
<li><a href='#check_tree_prior'><p>Check if the tree prior is a valid tree prior</p></a></li>
<li><a href='#check_tree_priors'><p>Check if the object is a list of one or more tree priors.</p></a></li>
<li><a href='#check_treelog'><p>Check if a <code>treelog</code> is valid.</p></a></li>
<li><a href='#check_treelog_names'><p>Check if the <code>treelog</code> has the list elements</p>
of a valid <code>treelog</code> object.</a></li>
<li><a href='#check_treelog_values'><p>Check if the treelog has the list elements with valid values</p>
for being a valid treelog object.</a></li>
<li><a href='#clock_model_to_xml_operators'><p>Internal function</p></a></li>
<li><a href='#clock_model_to_xml_prior_distr'><p>Internal function</p></a></li>
<li><a href='#clock_model_to_xml_state'><p>Internal function</p></a></li>
<li><a href='#clock_model_to_xml_tracelog'><p>Internal function</p></a></li>
<li><a href='#clock_model_to_xml_treelogger'><p>Convert a clock model to the XML of the <code>TreeLogger</code></p></a></li>
<li><a href='#clock_rate_param_to_xml'><p>Internal function</p></a></li>
<li><a href='#compare_lines'><p>Internal function</p></a></li>
<li><a href='#count_trailing_spaces'><p>Count the number of spaces before the first character</p></a></li>
<li><a href='#create_alpha_param'><p>Create a parameter called alpha</p></a></li>
<li><a href='#create_b_pop_sizes_param'><p>Create a parameter called 'b_pop_sizes'.</p></a></li>
<li><a href='#create_bd_tree_prior'><p>Create a Birth-Death tree prior</p></a></li>
<li><a href='#create_beast2_beast_xml'><p>Create the <code>&lt;beast ...&gt;</code> XML</p></a></li>
<li><a href='#create_beast2_input'><p>Create a BEAST2 XML input text</p></a></li>
<li><a href='#create_beast2_input_beast'><p>Creates the XML text for the <code>beast</code> tag of a BEAST2 parameter file.</p></a></li>
<li><a href='#create_beast2_input_data'><p>Creates the <code>data</code> section of a BEAST2 XML parameter file</p></a></li>
<li><a href='#create_beast2_input_data_sequences'><p>Creates the data section of a BEAST2 XML parameter file</p></a></li>
<li><a href='#create_beast2_input_distr'><p>Creates the distribution section of a BEAST2 XML parameter file.</p></a></li>
<li><a href='#create_beast2_input_distr_lh'><p>Creates the XML text for the <code>distribution</code> tag</p>
with the <code>likelihood</code> ID,
of a BEAST2 parameter file.</a></li>
<li><a href='#create_beast2_input_distr_prior'><p>Creates the prior section in the distribution section</p>
of a BEAST2 XML parameter file</a></li>
<li><a href='#create_beast2_input_file'><p>Create a BEAST2 input file</p></a></li>
<li><a href='#create_beast2_input_file_from_model'><p>Create a BEAST2 input file from an inference model</p></a></li>
<li><a href='#create_beast2_input_from_model'><p>Create a BEAST2 XML input text from an inference model</p></a></li>
<li><a href='#create_beast2_input_init'><p>Creates the <code>init</code> section of a BEAST2 XML parameter file</p></a></li>
<li><a href='#create_beast2_input_map'><p>Creates the map section of a BEAST2 XML parameter file</p></a></li>
<li><a href='#create_beast2_input_operators'><p>Creates the operators section of a BEAST2 XML parameter file</p></a></li>
<li><a href='#create_beast2_input_run'><p>Creates the '<code>run</code>' section of a BEAST2 XML parameter file</p></a></li>
<li><a href='#create_beast2_input_state'><p>Creates the '<code>state</code>' section of a BEAST2 XML parameter file</p></a></li>
<li><a href='#create_beauti_options'><p>Function to create a set of 'BEAUti' options.</p></a></li>
<li><a href='#create_beauti_options_v2_4'><p>Function to create the BEAUti options for version 2.4.</p></a></li>
<li><a href='#create_beauti_options_v2_6'><p>Function to create the BEAUti options for version 2.6.</p></a></li>
<li><a href='#create_beautier_tempfolder'><p>Create the default 'beautier' temporary folder</p></a></li>
<li><a href='#create_beta_distr'><p>Create a beta distribution</p></a></li>
<li><a href='#create_beta_param'><p>Create a parameter called beta</p></a></li>
<li><a href='#create_branch_rate_model_xml'><p>Internal function to create the <code>branchRateModel</code> section</p>
of the XML as text.</a></li>
<li><a href='#create_cbs_tree_prior'><p>Create a Coalescent Bayesian Skyline tree prior</p></a></li>
<li><a href='#create_ccp_tree_prior'><p>Create a Coalescent Constant Population tree prior</p></a></li>
<li><a href='#create_cep_tree_prior'><p>Create a Coalescent Exponential Population tree prior</p></a></li>
<li><a href='#create_clock_model'><p>General function to create a clock model</p></a></li>
<li><a href='#create_clock_model_from_name'><p>Create a clock model from name</p></a></li>
<li><a href='#create_clock_models'><p>Creates all supported clock models,</p>
which is a list of the types returned by
<code>create_rln_clock_model</code>,
and <code>create_strict_clock_model</code></a></li>
<li><a href='#create_clock_models_from_names'><p>Create clock models from their names</p></a></li>
<li><a href='#create_clock_rate_param'><p>Create a parameter called <code>clock_rate</code>,</p>
as needed by <code>create_strict_clock_model</code></a></li>
<li><a href='#create_clock_rate_state_node_parameter_xml'><p>Internal function</p></a></li>
<li><a href='#create_data_xml'><p>Create the <code>&lt;data ..&gt;</code> XML</p></a></li>
<li><a href='#create_distr'><p>General function to create a distribution.</p></a></li>
<li><a href='#create_exp_distr'><p>Create an exponential distribution</p></a></li>
<li><a href='#create_freq_param'><p>Create a parameter called freq</p></a></li>
<li><a href='#create_gamma_distr'><p>Create a gamma distribution</p></a></li>
<li><a href='#create_gamma_site_model'><p>Create a gamma site model, part of a site model</p></a></li>
<li><a href='#create_gtr_site_model'><p>Create a GTR site model</p></a></li>
<li><a href='#create_gtr_subst_model_xml'><p>Converts a GTR site model to XML,</p>
used in the <code>substModel</code> section</a></li>
<li><a href='#create_hky_site_model'><p>Create an HKY site model</p></a></li>
<li><a href='#create_hky_subst_model_xml'><p>Converts a site model to XML,</p>
used in the <code>substModel</code> section</a></li>
<li><a href='#create_inference_model'><p>Create a Bayesian phylogenetic inference model.</p></a></li>
<li><a href='#create_inv_gamma_distr'><p>Create an inverse-gamma distribution</p></a></li>
<li><a href='#create_jc69_site_model'><p>Create a JC69 site model</p></a></li>
<li><a href='#create_jc69_subst_model_xml'><p>Converts a JC69 site model to XML,</p>
used in the <code>substModel</code> section</a></li>
<li><a href='#create_kappa_1_param'><p>Create a parameter called kappa 1</p></a></li>
<li><a href='#create_kappa_2_param'><p>Create a parameter called kappa 2</p></a></li>
<li><a href='#create_kappa_param'><p>Create a parameter called kappa</p></a></li>
<li><a href='#create_lambda_param'><p>Create a parameter called lambda</p></a></li>
<li><a href='#create_laplace_distr'><p>Create a Laplace distribution</p></a></li>
<li><a href='#create_log_normal_distr'><p>Create a log-normal distribution</p></a></li>
<li><a href='#create_loggers_xml'><p>Creates the three logger sections of a BEAST2 XML parameter file</p></a></li>
<li><a href='#create_m_param'><p>Create a parameter called m</p></a></li>
<li><a href='#create_mcmc'><p>Create an MCMC configuration.</p></a></li>
<li><a href='#create_mean_param'><p>Create a parameter called mean</p></a></li>
<li><a href='#create_mrca_prior'><p>Create a Most Recent Common Ancestor prior</p></a></li>
<li><a href='#create_mu_param'><p>Create a parameter called mu</p></a></li>
<li><a href='#create_normal_distr'><p>Create an normal distribution</p></a></li>
<li><a href='#create_ns_inference_model'><p>Create an inference model to measure the evidence of.</p></a></li>
<li><a href='#create_ns_mcmc'><p>Create an MCMC object to estimate the marginal likelihood</p>
using Nested Sampling.</a></li>
<li><a href='#create_one_div_x_distr'><p>Create a 1/x distribution</p></a></li>
<li><a href='#create_param'><p>General function to create a parameter.</p></a></li>
<li><a href='#create_poisson_distr'><p>Create a Poisson distribution</p></a></li>
<li><a href='#create_rate_ac_param'><p>Create a parameter called 'rate AC'</p></a></li>
<li><a href='#create_rate_ag_param'><p>Create a parameter called 'rate AG'</p></a></li>
<li><a href='#create_rate_at_param'><p>Create a parameter called 'rate AT'</p></a></li>
<li><a href='#create_rate_categories_state_node_xml'><p>Internal function</p></a></li>
<li><a href='#create_rate_cg_param'><p>Create a parameter called 'rate CG'</p></a></li>
<li><a href='#create_rate_ct_param'><p>Create a parameter called 'rate CT'</p></a></li>
<li><a href='#create_rate_gt_param'><p>Create a parameter called 'rate GT'</p></a></li>
<li><a href='#create_rln_clock_branch_rate_model_xml'><p>Internal function</p></a></li>
<li><a href='#create_rln_clock_model'><p>Create a relaxed log-normal clock model</p></a></li>
<li><a href='#create_s_param'><p>Create a parameter called s</p></a></li>
<li><a href='#create_scale_param'><p>Create a parameter called scale</p></a></li>
<li><a href='#create_screenlog'><p>Create a <code>screenlog</code> object</p></a></li>
<li><a href='#create_screenlog_xml'><p>Creates the <code>screenlog</code> section of the <code>logger</code> section</p>
of a BEAST2 XML parameter file</a></li>
<li><a href='#create_sigma_param'><p>Create a parameter called sigma</p></a></li>
<li><a href='#create_site_model'><p>General function to create a site model.</p></a></li>
<li><a href='#create_site_model_from_name'><p>Create a site model from name</p></a></li>
<li><a href='#create_site_model_parameters_xml'><p>Internal function to creates the XML text for the</p>
<code>parameter</code>s within the <code>siteModel</code> section
of a BEAST2 parameter file.</a></li>
<li><a href='#create_site_model_xml'><p>Internal function to creates the XML text for the <code>siteModel</code> tag</p>
of a BEAST2 parameter file.</a></li>
<li><a href='#create_site_models'><p>Creates all supported site models</p>
which is a list of the types returned by
<code>create_gtr_site_model</code>,
<code>create_hky_site_model</code>,
<code>create_jc69_site_model</code>
and <code>create_tn93_site_model</code></a></li>
<li><a href='#create_site_models_from_names'><p>Create site models from their names</p></a></li>
<li><a href='#create_strict_clock_branch_rate_model_xml'><p>Internal function.</p></a></li>
<li><a href='#create_strict_clock_model'><p>Create a strict clock model</p></a></li>
<li><a href='#create_strict_clock_rate_scaler_operator_xml'><p>Internal function</p></a></li>
<li><a href='#create_subst_model_xml'><p>Internal function to create the <code>substModel</code> section</p></a></li>
<li><a href='#create_temp_screenlog_filename'><p>Create a filename for a temporary 'screenlog' file</p></a></li>
<li><a href='#create_temp_tracelog_filename'><p>Create a filename for a temporary 'tracelog' file</p></a></li>
<li><a href='#create_temp_treelog_filename'><p>Create a filename for a temporary 'treelog' file</p></a></li>
<li><a href='#create_test_inference_model'><p>Create a testing inference model.</p></a></li>
<li><a href='#create_test_mcmc'><p>Create an MCMC configuration for testing.</p></a></li>
<li><a href='#create_test_ns_inference_model'><p>Create an inference model to be tested by Nested Sampling</p></a></li>
<li><a href='#create_test_ns_mcmc'><p>Create an NS MCMC object for testing</p></a></li>
<li><a href='#create_test_screenlog'><p>Create a <code>screenlog</code> object, to be used in testing</p></a></li>
<li><a href='#create_test_tracelog'><p>Create a <code>tracelog</code> object, as used for testing</p></a></li>
<li><a href='#create_test_treelog'><p>Create a <code>treelog</code> object to be used in testing</p></a></li>
<li><a href='#create_tn93_site_model'><p>Create a TN93 site model</p></a></li>
<li><a href='#create_tn93_subst_model_xml'><p>Converts a TN93 site model to XML,</p>
used in the <code>substModel</code> section</a></li>
<li><a href='#create_tracelog'><p>Create a <code>tracelog</code> object</p></a></li>
<li><a href='#create_tracelog_xml'><p>Internal function</p></a></li>
<li><a href='#create_trait_set_string'><p>Create a trait set string.</p></a></li>
<li><a href='#create_tree_likelihood_distr_xml'><p>Creates the XML text for the <code>distribution</code> tag</p>
with the <code>treeLikelihood</code> ID,
of a BEAST2 parameter file.</a></li>
<li><a href='#create_tree_prior'><p>Internal function to create a tree prior</p></a></li>
<li><a href='#create_tree_priors'><p>Creates all supported tree priors,</p>
which is a list of the types returned by
<code>create_bd_tree_prior</code>,
<code>create_cbs_tree_prior</code>,
<code>create_ccp_tree_prior</code>,
<code>create_cep_tree_prior</code>
and <code>create_yule_tree_prior</code></a></li>
<li><a href='#create_treelog'><p>Create a <code>treelog</code> object</p></a></li>
<li><a href='#create_treelog_xml'><p>Creates the XML text for the 'logger' tag with ID 'treelog'.</p>
This section has these elements:
&ldquo;'
&lt;logger id=&quot;treelog.t:test_output_0&quot; spec=&quot;Logger&quot; fileName=&quot;my_treelog.trees&quot; logEvery=&quot;345000&quot; mode=&quot;tree&quot; sanitiseHeaders=&quot;true&quot; sort=&quot;smart&quot;&gt; # nolint indeed long
&lt;log id=&quot;TreeWithMetaDataLogger.t:test_output_0&quot; spec=&quot;beast.evolution.tree.TreeWithMetaDataLogger&quot; tree=&quot;@Tree.t:test_output_0&quot;/&gt; # nolint indeed long
&lt;/logger&gt;
&ldquo;'</a></li>
<li><a href='#create_ucld_mean_state_node_param_xml'><p>Internal function</p></a></li>
<li><a href='#create_ucld_stdev_state_node_param_xml'><p>Internal function</p></a></li>
<li><a href='#create_uniform_distr'><p>Create a uniform distribution</p></a></li>
<li><a href='#create_xml_declaration'><p>Create the XML declaration of the BEAST2 XML input file</p></a></li>
<li><a href='#create_yule_tree_prior'><p>Create a Yule tree prior</p></a></li>
<li><a href='#default_parameters_doc'><p>Documentation of parameters (for example, <code>create_param</code>.</p>
This function does nothing. It is intended to inherit documentation from.</a></li>
<li><a href='#default_params_doc'><p>Documentation of general function arguments.</p>
This function does nothing.
It is intended to inherit function argument documentation.</a></li>
<li><a href='#distr_to_xml'><p>Internal function</p></a></li>
<li><a href='#distr_to_xml_beta'><p>Internal function</p></a></li>
<li><a href='#distr_to_xml_exp'><p>Internal function</p></a></li>
<li><a href='#distr_to_xml_inv_gamma'><p>Internal function</p></a></li>
<li><a href='#distr_to_xml_laplace'><p>Internal function</p></a></li>
<li><a href='#distr_to_xml_log_normal'><p>Internal function</p></a></li>
<li><a href='#distr_to_xml_normal'><p>Internal function</p></a></li>
<li><a href='#distr_to_xml_one_div_x'><p>Internal function</p></a></li>
<li><a href='#distr_to_xml_poisson'><p>Internal function</p></a></li>
<li><a href='#distr_to_xml_uniform'><p>Internal function</p></a></li>
<li><a href='#extract_xml_loggers_from_lines'><p>Extract everything between first loggers and last loggers line</p></a></li>
<li><a href='#extract_xml_operators_from_lines'><p>Extract everything between first operators and last operators line</p></a></li>
<li><a href='#extract_xml_section_from_lines'><p>Get the lines of an XML section, including the section tags</p></a></li>
<li><a href='#fasta_file_to_sequences'><p>Convert a FASTA file to a table of sequences</p></a></li>
<li><a href='#find_clock_model'><p>Finds a clock model with a certain ID</p></a></li>
<li><a href='#find_first_regex_line'><p>Find the first line that satisfies a regex</p></a></li>
<li><a href='#find_first_xml_opening_tag_line'><p>Find the line number of the first section's opening tag</p></a></li>
<li><a href='#find_last_regex_line'><p>Find the index of the last line that matches a regex</p></a></li>
<li><a href='#find_last_xml_closing_tag_line'><p>Find the highest line number of a section's closing tag</p></a></li>
<li><a href='#freq_equilibrium_to_xml'><p>Creates the <code>freq_equilibrium</code> as XML</p></a></li>
<li><a href='#freq_param_to_xml'><p>Internal function</p></a></li>
<li><a href='#gamma_distr_to_xml'><p>Internal function</p></a></li>
<li><a href='#gamma_site_model_to_xml_prior_distr'><p>Internal function.</p></a></li>
<li><a href='#gamma_site_model_to_xml_state'><p>Converts a gamma site model to XML,</p>
used in the <code>state</code> section</a></li>
<li><a href='#gamma_site_models_to_xml_prior_distr'><p>Deprecated function</p></a></li>
<li><a href='#get_alignment_id'><p>Conclude the ID from a FASTA filename.</p></a></li>
<li><a href='#get_alignment_ids'><p>Get the alignment IDs from one or more files.</p></a></li>
<li><a href='#get_alignment_ids_from_fasta_filenames'><p>Get the alignment ID from one or more FASTA filenames.</p></a></li>
<li><a href='#get_beautier_folder'><p>Get the path to the beautier temporary files folder</p></a></li>
<li><a href='#get_beautier_path'><p>Get the full path of a file in the <code>inst/extdata</code> folder</p></a></li>
<li><a href='#get_beautier_paths'><p>Get the full paths of files in the <code>inst/extdata</code> folder</p></a></li>
<li><a href='#get_beautier_tempfilename'><p>Get a temporary filename</p></a></li>
<li><a href='#get_clock_model_name'><p>Get the BEAUti name for a clock model</p></a></li>
<li><a href='#get_clock_model_names'><p>Get the clock model names</p></a></li>
<li><a href='#get_clock_models_ids'><p>Collect the IDs of the list of clock models</p></a></li>
<li><a href='#get_crown_age'><p>Obtain the crown age of a phylogeny.</p></a></li>
<li><a href='#get_default_beast_namespace'><p>Get the default 'namespace' element value of the 'beast' XML tag.</p></a></li>
<li><a href='#get_default_beast_namespace_v2_4'><p>Get the default 'namespace' element value of the 'beast' XML tag</p>
for BEAST 2.4</a></li>
<li><a href='#get_default_beast_namespace_v2_6'><p>Get the default 'namespace' element value of the 'beast' XML tag</p>
for BEAST 2.6</a></li>
<li><a href='#get_distr_n_params'><p>Get the number of parameters a distribution uses</p></a></li>
<li><a href='#get_distr_names'><p>Get the distribution names</p></a></li>
<li><a href='#get_fasta_filename'><p>Get the path of a FASTA file used in testing</p></a></li>
<li><a href='#get_file_base_sans_ext'><p>Get the base of the filename base without extension</p></a></li>
<li><a href='#get_freq_equilibrium_names'><p>Returns valid values for the <code>freq_equilibrium</code> argument</p></a></li>
<li><a href='#get_gamma_site_model_n_distrs'><p>Get the number of distributions in a gamma site model</p></a></li>
<li><a href='#get_gamma_site_model_n_params'><p>Get the number of distributions a site model has</p></a></li>
<li><a href='#get_has_non_strict_clock_model'><p>Determines if there is at least one non-strict clock model</p>
in the list of one or more clock models</a></li>
<li><a href='#get_inference_model_filenames'><p>Get the filenames stored in an inference model.</p></a></li>
<li><a href='#get_log_modes'><p>Get the possible log modes</p></a></li>
<li><a href='#get_log_sorts'><p>Get the possible log sorts</p></a></li>
<li><a href='#get_mcmc_filenames'><p>Get the filenames stored in an MCMC.</p></a></li>
<li><a href='#get_n_taxa'><p>Extract the number of taxa from a file</p></a></li>
<li><a href='#get_operator_id_pre'><p>Get the prefix of operator IDs</p></a></li>
<li><a href='#get_param_names'><p>Get the parameter names</p></a></li>
<li><a href='#get_remove_dir_fun'><p>Get a function that, from a filename, returns the part</p>
without the directory.</a></li>
<li><a href='#get_remove_hex_fun'><p>Get a function that removes the hex string from filenames.</p></a></li>
<li><a href='#get_replace_dir_fun'><p>Get a function to replace the directory of a filename</p></a></li>
<li><a href='#get_site_model_n_distrs'><p>Get the number of distributions a site model has</p></a></li>
<li><a href='#get_site_model_n_params'><p>Get the number of distributions a site model has</p></a></li>
<li><a href='#get_site_model_names'><p>Get the site models' names</p></a></li>
<li><a href='#get_site_models_n_distrs'><p>Get the number of distributions a site model has</p></a></li>
<li><a href='#get_site_models_n_params'><p>Get the number of distributions one or more site models have</p></a></li>
<li><a href='#get_taxa_names'><p>Extract the names of taxa from a file</p></a></li>
<li><a href='#get_tree_prior_n_distrs'><p>Get the number of distributions a tree prior has</p></a></li>
<li><a href='#get_tree_prior_n_params'><p>Get the number of parameters a tree prior has</p></a></li>
<li><a href='#get_tree_prior_names'><p>Get the tree prior names</p></a></li>
<li><a href='#get_tree_priors_n_distrs'><p>Get the number of distributions a tree prior has</p></a></li>
<li><a href='#get_tree_priors_n_params'><p>Get the number of parameters a list of tree priors has</p></a></li>
<li><a href='#get_xml_closing_tag'><p>Get the XML closing tag</p></a></li>
<li><a href='#get_xml_opening_tag'><p>Get the XML opening tag</p></a></li>
<li><a href='#gtr_site_model_to_xml_prior_distr'><p>Internal function</p></a></li>
<li><a href='#gtr_site_model_to_xml_state'><p>Converts a site model to XML,</p>
used in the <code>state</code> section</a></li>
<li><a href='#has_mrca_prior'><p>Determines if the inference model has an MRCA prior.</p></a></li>
<li><a href='#has_mrca_prior_with_distr'><p>See if the inference model has one MRCA prior with a distribution</p></a></li>
<li><a href='#has_rln_clock_model'><p>Determine if the <code>inference_model</code> uses</p>
a relaxed log-normal clock model.</a></li>
<li><a href='#has_strict_clock_model'><p>Determine if the <code>inference_model</code> uses a strict clock model.</p></a></li>
<li><a href='#has_tip_dating'><p>Determine if the <code>inference_model</code> uses tip dating.</p></a></li>
<li><a href='#has_xml_closing_tag'><p>Is an XML closing tag with the value of <code>section</code></p>
present among the lines of
the text?</a></li>
<li><a href='#has_xml_opening_tag'><p>Is an XML opening tag with value 'section' present among the lines of</p>
the text?</a></li>
<li><a href='#has_xml_short_closing_tag'><p>Is an XML closing tag with short closing text in</p>
one of the lines of the text?</a></li>
<li><a href='#hky_site_model_to_xml_prior_distr'><p>Internal function</p></a></li>
<li><a href='#hky_site_model_to_xml_state'><p>Converts a site model to XML,</p>
used in the <code>state</code> section</a></li>
<li><a href='#indent'><p>Indent text for a certain number of spaces.</p>
If the text is only whitespace, leave it as such</a></li>
<li><a href='#init_bd_tree_prior'><p>Initializes a Birth-Death tree prior</p></a></li>
<li><a href='#init_beta_distr'><p>Initializes a beta distribution</p></a></li>
<li><a href='#init_ccp_tree_prior'><p>Initializes a Coalescent Constant Population tree prior</p></a></li>
<li><a href='#init_cep_tree_prior'><p>Initializes a Coalescent Exponential Population tree prior</p></a></li>
<li><a href='#init_clock_models'><p>Initializes all clock models</p></a></li>
<li><a href='#init_distr'><p>Initializes a distribution</p></a></li>
<li><a href='#init_exp_distr'><p>Initializes an exponential distribution</p></a></li>
<li><a href='#init_gamma_distr'><p>Initializes a gamma distribution</p></a></li>
<li><a href='#init_gamma_site_model'><p>Initializes a gamma site model</p></a></li>
<li><a href='#init_gtr_site_model'><p>Initializes a GTR site model</p></a></li>
<li><a href='#init_hky_site_model'><p>Initializes an HKY site model</p></a></li>
<li><a href='#init_inference_model'><p>Initialize an inference model</p></a></li>
<li><a href='#init_inv_gamma_distr'><p>Initializes an inverse gamma distribution</p></a></li>
<li><a href='#init_jc69_site_model'><p>Initializes a JC69 site model</p></a></li>
<li><a href='#init_laplace_distr'><p>Initializes an Laplace distribution</p></a></li>
<li><a href='#init_log_normal_distr'><p>Initializes an log-normal distribution</p></a></li>
<li><a href='#init_mrca_prior'><p>Initialize the MRCA prior.</p></a></li>
<li><a href='#init_mrca_priors'><p>Initializes all MRCA priors</p></a></li>
<li><a href='#init_normal_distr'><p>Initializes an normal distribution</p></a></li>
<li><a href='#init_one_div_x_distr'><p>Initializes an one-divided-by-x distribution</p></a></li>
<li><a href='#init_param'><p>Initializes a parameter</p></a></li>
<li><a href='#init_poisson_distr'><p>Initializes an Poisson distribution</p></a></li>
<li><a href='#init_rln_clock_model'><p>Initializes a Relaxed Log-Normal clock model</p></a></li>
<li><a href='#init_site_models'><p>Initializes all site models</p></a></li>
<li><a href='#init_strict_clock_model'><p>Initializes a strict clock model</p></a></li>
<li><a href='#init_tn93_site_model'><p>Initializes a TN93 site model</p></a></li>
<li><a href='#init_tree_priors'><p>Initializes all tree priors</p></a></li>
<li><a href='#init_uniform_distr'><p>Initializes a uniform distribution</p></a></li>
<li><a href='#init_yule_tree_prior'><p>Initializes a Yule tree prior</p></a></li>
<li><a href='#interspace'><p>Puts spaces in between the lines</p></a></li>
<li><a href='#is_alpha_param'><p>Determine if the object is a valid</p>
alpha parameter</a></li>
<li><a href='#is_b_pop_sizes_param'><p>Determine if the object is a valid</p>
b_pop_sizes parameter</a></li>
<li><a href='#is_bd_tree_prior'><p>Determine if the object is a valid Birth Death tree prior</p></a></li>
<li><a href='#is_beauti_options'><p>Determine if the object is a valid <code>beauti_options</code></p></a></li>
<li><a href='#is_beta_distr'><p>Determine if the object is a valid</p>
beta distribution,
as created by <code>create_beta_distr</code></a></li>
<li><a href='#is_beta_param'><p>Determine if the object is a valid</p>
beta parameter</a></li>
<li><a href='#is_cbs_tree_prior'><p>Determine if the object is a valid constant coalescent Bayesian skyline prior</p></a></li>
<li><a href='#is_ccp_tree_prior'><p>Determine if the object is a valid</p>
constant coalescence population tree prior</a></li>
<li><a href='#is_cep_tree_prior'><p>Determine if the object is a valid</p>
coalescent exponential population tree prior</a></li>
<li><a href='#is_clock_model'><p>Determine if the object is a valid clock_model</p></a></li>
<li><a href='#is_clock_model_name'><p>Determines if the name is a valid clock model name</p></a></li>
<li><a href='#is_clock_rate_param'><p>Determine if the object is a valid</p>
clock_rate parameter</a></li>
<li><a href='#is_default_mcmc'><p>Determine if the MCMC is a default MCMC</p></a></li>
<li><a href='#is_distr'><p>Determine if the object is a valid distribution</p></a></li>
<li><a href='#is_distr_name'><p>Determines if the name is a valid distribution name</p></a></li>
<li><a href='#is_exp_distr'><p>Determine if the object is a valid</p>
exponential distribution
as created by <code>create_exp_distr</code></a></li>
<li><a href='#is_freq_equilibrium_name'><p>Checks if <code>name</code> is a valid <code>freq_equilibrium</code> argument value</p></a></li>
<li><a href='#is_freq_param'><p>Determine if the object is a valid freq parameter</p></a></li>
<li><a href='#is_gamma_distr'><p>Determine if the object is a valid</p>
gamma distribution,
as created by <code>create_gamma_distr</code></a></li>
<li><a href='#is_gamma_site_model'><p>Is object x a gamma site model?</p></a></li>
<li><a href='#is_gtr_site_model'><p>Determine if the object is a valid GTR site model,</p>
as created by <code>create_gtr_site_model</code></a></li>
<li><a href='#is_hky_site_model'><p>Determine if the object is a valid HKY site model,</p>
as created by <code>create_hky_site_model</code></a></li>
<li><a href='#is_id'><p>Determine if the object is a valid ID</p></a></li>
<li><a href='#is_in_patterns'><p>Is there at least one regular expression having a match with the line?</p></a></li>
<li><a href='#is_inference_model'><p>Determine if the input is an inference model</p></a></li>
<li><a href='#is_init_bd_tree_prior'><p>Determine if x is an initialized Birth-Death tree_prior object</p></a></li>
<li><a href='#is_init_beta_distr'><p>Determine if x is an initialized beta distribution object</p>
as created by <code>create_beta_distr</code></a></li>
<li><a href='#is_init_cbs_tree_prior'><p>Determine if x is an initialized Coalescent Bayesian Skyline</p>
tree_prior object</a></li>
<li><a href='#is_init_ccp_tree_prior'><p>Determine if x is an initialized Coalescent Constant Population</p>
tree_prior object</a></li>
<li><a href='#is_init_cep_tree_prior'><p>Determine if x is an initialized Coalescent Exponential Population</p>
tree_prior object</a></li>
<li><a href='#is_init_clock_model'><p>Determine if x is an initialized clock_model object,</p>
as created by <code>create_clock_model</code></a></li>
<li><a href='#is_init_distr'><p>Determine if x is an initialized distribution object</p>
as created by <code>create_distr</code></a></li>
<li><a href='#is_init_exp_distr'><p>Determine if x is an initialized exponential distribution object</p>
as created by <code>create_exp_distr</code></a></li>
<li><a href='#is_init_gamma_distr'><p>Determine if x is an initialized gamma distribution object</p></a></li>
<li><a href='#is_init_gamma_site_model'><p>Determine if x is an initialized gamma site model,</p>
as created by <code>create_gamma_site_model</code></a></li>
<li><a href='#is_init_gtr_site_model'><p>Determine if x is an initialized GTR site model</p>
as created by <code>create_gtr_site_model</code></a></li>
<li><a href='#is_init_hky_site_model'><p>Determine if x is an initialized HKY site model</p>
as created by <code>create_hky_site_model</code></a></li>
<li><a href='#is_init_inv_gamma_distr'><p>Determine if x is an initialized inverse-gamma distribution</p>
as created by <code>create_inv_gamma_distr</code></a></li>
<li><a href='#is_init_jc69_site_model'><p>Determine if x is an initialized JC69 site model</p>
as created by <code>create_jc69_site_model</code></a></li>
<li><a href='#is_init_laplace_distr'><p>Determine if x is an initialized Laplace distribution</p>
as created by <code>create_laplace_distr</code></a></li>
<li><a href='#is_init_log_normal_distr'><p>Determine if x is an initialized log_normal distribution object</p>
as created by <code>create_log_normal_distr</code></a></li>
<li><a href='#is_init_mrca_prior'><p>Determine if x is an initialized MRCA prior</p></a></li>
<li><a href='#is_init_normal_distr'><p>Determine if x is an initialized normal distribution object</p>
as created by <code>create_normal_distr</code></a></li>
<li><a href='#is_init_one_div_x_distr'><p>Determine if x is an initialized one_div_x distribution object</p>
as created by <code>create_one_div_x_distr</code></a></li>
<li><a href='#is_init_param'><p>Determine if x is an initialized parameter,</p>
as created by create_param</a></li>
<li><a href='#is_init_poisson_distr'><p>Determine if x is an initialized Poisson distribution object</p>
as created by <code>create_poisson_distr</code></a></li>
<li><a href='#is_init_rln_clock_model'><p>Determine if x is an initialized relaxed log-normal clock_model object</p></a></li>
<li><a href='#is_init_site_model'><p>Determine if x is an initialized site model,</p>
as created by <code>create_site_model</code></a></li>
<li><a href='#is_init_strict_clock_model'><p>Determine if x is an initialized strict clock_model object</p></a></li>
<li><a href='#is_init_tn93_site_model'><p>Determine if x is an initialized tn93 site model</p>
as created by <code>create_tn93_site_model</code></a></li>
<li><a href='#is_init_tree_prior'><p>Determine if x is an initialized tree_prior objects</p></a></li>
<li><a href='#is_init_uniform_distr'><p>Determine if x is an initialized uniform distribution object</p>
as created by <code>create_uniform_distr</code></a></li>
<li><a href='#is_init_yule_tree_prior'><p>Determine if x is an initialized Yule tree_prior object</p></a></li>
<li><a href='#is_inv_gamma_distr'><p>Determine if the object is a valid</p>
inverse-gamma distribution
as created by <code>create_inv_gamma_distr</code></a></li>
<li><a href='#is_jc69_site_model'><p>Determine if the object is a valid JC69 site model</p></a></li>
<li><a href='#is_kappa_1_param'><p>Determine if the object is a valid kappa 1 parameter</p></a></li>
<li><a href='#is_kappa_2_param'><p>Determine if the object is a valid kappa 2 parameter</p></a></li>
<li><a href='#is_kappa_param'><p>Determine if the object is a valid kappa parameter</p></a></li>
<li><a href='#is_lambda_param'><p>Determine if the object is a valid lambda parameter</p></a></li>
<li><a href='#is_laplace_distr'><p>Determine if the object is a valid</p>
Laplace distribution,
as created by <code>create_laplace_distr</code></a></li>
<li><a href='#is_log_normal_distr'><p>Determine if the object is a valid</p>
log-normal distribution,
as created by <code>create_log_normal_distr</code></a></li>
<li><a href='#is_m_param'><p>Determine if the object is a valid</p>
m parameter</a></li>
<li><a href='#is_mcmc'><p>Determine if the object is a valid MCMC</p></a></li>
<li><a href='#is_mcmc_nested_sampling'><p>Determine if the object is a valid Nested-Sampling MCMC,</p>
as used in [1]</a></li>
<li><a href='#is_mean_param'><p>Determine if the object is a valid mean parameter</p></a></li>
<li><a href='#is_mrca_align_id_in_fasta'><p>Determine if an MRCA prior's alignment IDs is present in the FASTA file</p></a></li>
<li><a href='#is_mrca_align_ids_in_fastas'><p>Determine if an MRCA prior's alignment IDs are present in the FASTA files</p></a></li>
<li><a href='#is_mrca_prior'><p>Determine of the object is an empty (<code>NA</code>) or valid MRCA prior.</p></a></li>
<li><a href='#is_mrca_prior_with_distr'><p>See if x is one MRCA prior with a distribution</p></a></li>
<li><a href='#is_mu_param'><p>Determine if the object is a valid</p>
mu parameter</a></li>
<li><a href='#is_normal_distr'><p>Determine if the object is a valid</p>
normal distribution
as created by <code>create_normal_distr</code></a></li>
<li><a href='#is_on_appveyor'><p>Determines if the environment is AppVeyor</p></a></li>
<li><a href='#is_on_ci'><p>Determines if the environment is a continuous integration service</p></a></li>
<li><a href='#is_on_github_actions'><p>Determines if the environment is GitHub Actions</p></a></li>
<li><a href='#is_on_travis'><p>Determines if the environment is Travis CI</p></a></li>
<li><a href='#is_one_bool'><p>Check if the argument is one boolean</p></a></li>
<li><a href='#is_one_div_x_distr'><p>Determine if the object is a valid</p>
1/x distribution,
as created by <code>create_one_div_x_distr</code></a></li>
<li><a href='#is_one_double'><p>Determines if the argument is a double</p></a></li>
<li><a href='#is_one_empty_string'><p>Determine if an object is one empty string</p></a></li>
<li><a href='#is_one_int'><p>Determines if the argument is a whole number</p></a></li>
<li><a href='#is_one_na'><p>Determines if x is one NA</p></a></li>
<li><a href='#is_one_string'><p>Determines if the argument is one string</p></a></li>
<li><a href='#is_one_string_that_is_a_number'><p>General function to create a distribution.</p></a></li>
<li><a href='#is_param'><p>Determine if the object is a valid parameter</p></a></li>
<li><a href='#is_param_name'><p>Determines if the name is a valid parameter name</p></a></li>
<li><a href='#is_phylo'><p>Checks if the input is a phylogeny</p></a></li>
<li><a href='#is_poisson_distr'><p>Determine if the object is a valid</p>
Poisson distribution
as created by <code>create_poisson_distr</code></a></li>
<li><a href='#is_rate_ac_param'><p>Determine if the object is a valid</p>
'rate AC' parameter</a></li>
<li><a href='#is_rate_ag_param'><p>Determine if the object is a valid</p>
'rate AG' parameter</a></li>
<li><a href='#is_rate_at_param'><p>Determine if the object is a valid</p>
'rate AT' parameter</a></li>
<li><a href='#is_rate_cg_param'><p>Determine if the object is a valid</p>
'rate CG' parameter</a></li>
<li><a href='#is_rate_ct_param'><p>Determine if the object is a valid</p>
'rate CT' parameter</a></li>
<li><a href='#is_rate_gt_param'><p>Determine if the object is a valid</p>
'rate GT' parameter</a></li>
<li><a href='#is_rln_clock_model'><p>Determine if the object is a valid relaxed log normal clock model</p></a></li>
<li><a href='#is_s_param'><p>Determine if the object is a valid</p>
s parameter</a></li>
<li><a href='#is_scale_param'><p>Determine if the object is a valid</p>
scale parameter</a></li>
<li><a href='#is_sigma_param'><p>Determine if the object is a valid</p>
sigma parameter</a></li>
<li><a href='#is_site_model'><p>Determine if the object is a valid site_model</p></a></li>
<li><a href='#is_site_model_name'><p>Determines if the name is a valid site_model name</p></a></li>
<li><a href='#is_strict_clock_model'><p>Determine if the object is a valid strict clock model,</p>
as returned by <code>create_strict_clock_model</code></a></li>
<li><a href='#is_tn93_site_model'><p>Determine if the object is a valid TN93 site model,</p></a></li>
<li><a href='#is_tree_prior'><p>Determine if an object is a valid tree prior</p></a></li>
<li><a href='#is_tree_prior_name'><p>Determines if the name is a valid tree prior name</p></a></li>
<li><a href='#is_uniform_distr'><p>Determine if the object is a valid</p>
uniform distribution
as created by <code>create_uniform_distr</code></a></li>
<li><a href='#is_xml'><p>Checks if the text is a valid XML node, that is,</p>
it has a opening and matching closing tag</a></li>
<li><a href='#is_yule_tree_prior'><p>Determine if the object is a valid Yule tree prior,</p></a></li>
<li><a href='#jc69_site_model_to_xml_state'><p>Converts a site model to XML,</p>
used in the <code>state</code> section</a></li>
<li><a href='#kappa_param_to_xml'><p>Internal function</p></a></li>
<li><a href='#m_param_to_xml'><p>Internal function</p></a></li>
<li><a href='#mcmc_to_xml_run'><p>Converts an MCMC object to the run section's XML</p></a></li>
<li><a href='#mcmc_to_xml_run_default'><p>Converts an MCMC object to the run section's XML for a default MCMC</p></a></li>
<li><a href='#mcmc_to_xml_run_nested_sampling'><p>Converts an MCMC object to the run section's XML for a Nested-Sampling MCMC</p></a></li>
<li><a href='#mrca_prior_to_xml_prior_distr'><p>Creates the distribution section in the prior section of the</p>
distribution section of a BEAST2 XML parameter file.</a></li>
<li><a href='#mrca_prior_to_xml_state'><p>Internal function to create the XML of an MRCA prior,</p>
as used in the <code>state</code> section</a></li>
<li><a href='#mrca_prior_to_xml_taxonset'><p>Creates the <code>taxonset</code> section in the prior section of the</p>
distribution section of a BEAST2 XML parameter file.</a></li>
<li><a href='#mrca_prior_to_xml_tracelog'><p>Internal function</p></a></li>
<li><a href='#mrca_priors_to_xml_prior_distr'><p>Creates the the <code>distribution</code>'s prior section (which is part of</p>
a posterior distribution section) of a BEAST2 XML parameter file.</a></li>
<li><a href='#no_taxa_to_xml_tree'><p>Internal function</p></a></li>
<li><a href='#parameter_to_xml'><p>Internal function</p></a></li>
<li><a href='#parameter_to_xml_kappa_1'><p>Internal function</p></a></li>
<li><a href='#parameter_to_xml_kappa_2'><p>Internal function</p></a></li>
<li><a href='#parameter_to_xml_lambda'><p>Internal function</p></a></li>
<li><a href='#parameter_to_xml_mean'><p>Internal function</p></a></li>
<li><a href='#parameter_to_xml_mu'><p>Internal function</p></a></li>
<li><a href='#parameter_to_xml_rate_ac'><p>Internal function</p></a></li>
<li><a href='#parameter_to_xml_rate_ag'><p>Internal function</p></a></li>
<li><a href='#parameter_to_xml_rate_at'><p>Internal function</p></a></li>
<li><a href='#parameter_to_xml_rate_cg'><p>Internal function</p></a></li>
<li><a href='#parameter_to_xml_rate_ct'><p>Internal function</p></a></li>
<li><a href='#parameter_to_xml_rate_gt'><p>Internal function</p></a></li>
<li><a href='#parameter_to_xml_scale'><p>Internal function</p></a></li>
<li><a href='#parameter_to_xml_sigma'><p>Internal function</p></a></li>
<li><a href='#remove_beautier_folder'><p>Check there are no files in the default beautier folder</p></a></li>
<li><a href='#remove_empty_lines'><p>Remove all lines that are only whitespace</p></a></li>
<li><a href='#remove_multiline'><p>Remove consecutive lines</p></a></li>
<li><a href='#rename_inference_model_filenames'><p>Rename the filenames in an inference model</p></a></li>
<li><a href='#rename_mcmc_filenames'><p>Rename the filenames within an MCMC</p></a></li>
<li><a href='#rln_clock_model_to_xml_mean_rate_prior'><p>Internal function</p></a></li>
<li><a href='#rln_clock_model_to_xml_operators'><p>Internal function</p></a></li>
<li><a href='#rln_clock_model_to_xml_prior_distr'><p>Internal function</p></a></li>
<li><a href='#rln_clock_model_to_xml_state'><p>Internal function</p></a></li>
<li><a href='#rln_clock_model_to_xml_tracelog'><p>Internal function</p></a></li>
<li><a href='#rnd_phylo_to_xml_init'><p>Creates the XML of a random phylogeny,</p>
as used in the <code>init</code> section</a></li>
<li><a href='#s_parameter_to_xml'><p>Internal function</p></a></li>
<li><a href='#site_model_to_xml_operators'><p>Converts a site model to XML,</p>
used in the <code>operators</code> section</a></li>
<li><a href='#site_model_to_xml_prior_distr'><p>Internal function</p></a></li>
<li><a href='#site_model_to_xml_state'><p>Internal function to convert a site model to XML,</p>
used in the 'state' section</a></li>
<li><a href='#site_model_to_xml_tracelog'><p>Creates the site model's XML for the tracelog section</p></a></li>
<li><a href='#site_models_to_xml_operators'><p>Write the XML <code>operators</code> section from the site models.</p></a></li>
<li><a href='#site_models_to_xml_prior_distr'><p>Represent the site models as XML</p></a></li>
<li><a href='#site_models_to_xml_tracelog'><p>Creates the site models' XML for the tracelog section</p></a></li>
<li><a href='#strict_clock_model_to_xml_operators'><p>Internal function</p></a></li>
<li><a href='#strict_clock_model_to_xml_prior_distr'><p>Internal function</p></a></li>
<li><a href='#strict_clock_model_to_xml_state'><p>Internal function</p></a></li>
<li><a href='#strict_clock_model_to_xml_tracelog'><p>Internal function</p></a></li>
<li><a href='#taxa_to_xml_tree'><p>Internal function</p></a></li>
<li><a href='#tipdate_taxa_to_xml_trait'><p>Internal function</p></a></li>
<li><a href='#tipdate_taxa_to_xml_tree'><p>Internal function</p></a></li>
<li><a href='#tn93_site_model_to_xml_prior_distr'><p>Internal function</p></a></li>
<li><a href='#tn93_site_model_to_xml_state'><p>Converts a site model to XML,</p>
used in the <code>state</code> section</a></li>
<li><a href='#tree_model_to_tracelog_xml'><p>Internal function</p></a></li>
<li><a href='#tree_prior_to_xml_operators'><p>Internal function</p></a></li>
<li><a href='#tree_prior_to_xml_prior_distr'><p>Creates the distribution section in the prior section of the</p>
distribution section of a BEAST2 XML parameter file.</a></li>
<li><a href='#tree_prior_to_xml_state'><p>Creates the XML of a tree prior,</p>
as used in the <code>state</code> section</a></li>
<li><a href='#tree_prior_to_xml_tracelog'><p>Creates the tree prior's XML for the tracelog section</p></a></li>
<li><a href='#tree_priors_to_xml_prior_distr'><p>Creates the distribution section in the prior section of the</p>
distribution section of a BEAST2 XML parameter file.</a></li>
<li><a href='#tree_priors_to_xml_tracelog'><p>Creates the tree priors' XML for the tracelog section</p></a></li>
<li><a href='#unindent'><p>Unindents text</p></a></li>
<li><a href='#yule_tree_prior_to_xml_operators'><p>Internal function</p></a></li>
<li><a href='#yule_tree_prior_to_xml_prior_distr'><p>Creates the <code>prior</code> section in the prior section of</p>
the prior section of the distribution section
of a BEAST2 XML parameter file for a Yule tree prior</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>'BEAUti' from R</td>
</tr>
<tr>
<td>Version:</td>
<td>2.6.11</td>
</tr>
<tr>
<td>Description:</td>
<td>'BEAST2' (<a href="https://www.beast2.org">https://www.beast2.org</a>) is a widely used
  Bayesian phylogenetic tool, that uses DNA/RNA/protein data
  and many model priors to create a posterior of jointly estimated 
  phylogenies and parameters.
  'BEAUti 2' (which is part of 'BEAST2') is a GUI tool 
  that allows users to specify the many possible setups
  and generates the XML file 'BEAST2' needs to run.
  This package provides a way to create 'BEAST2' input
  files without active user input, but using
  R function calls instead.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://docs.ropensci.org/beautier/">https://docs.ropensci.org/beautier/</a>,
<a href="https://github.com/ropensci/beautier">https://github.com/ropensci/beautier</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ropensci/beautier/issues">https://github.com/ropensci/beautier/issues</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, rappdirs, purrr, rlang (&ge; 1.1.0), seqinr, stringr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>spelling, knitr, markdown, readr, rmarkdown, testthat (&ge;
2.1.0)</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-02 11:24:27 UTC; richel</td>
</tr>
<tr>
<td>Author:</td>
<td>Richèl J.C. Bilderbeek
    <a href="https://orcid.org/0000-0003-1107-7049"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Joëlle Barido-Sottani [rev] (Joëlle reviewed the package for rOpenSci,
    see https://github.com/ropensci/onboarding/issues/209),
  David Winter [rev] (David reviewed the package for rOpenSci, see
    https://github.com/ropensci/onboarding/issues/209),
  Paul Van Els <a href="https://orcid.org/0000-0002-9499-8873"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Raphael Scherrer <a href="https://orcid.org/0000-0002-1447-7630"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Yacine B. Chehida <a href="https://orcid.org/0000-0001-7269-9082"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Katharine S. Walter
    <a href="https://orcid.org/0000-0003-0065-2204"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Gary Napier <a href="https://orcid.org/0000-0002-1077-0055"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Jason Griffiths <a href="https://orcid.org/0000-0002-1667-8233"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Thijs Janzen <a href="https://orcid.org/0000-0002-4162-1140"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Pedro Taucce <a href="https://orcid.org/0000-0002-3088-4543"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Olivier Roy [ctb],
  Jana Riederer <a href="https://orcid.org/0000-0001-6951-9984"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Richèl J.C. Bilderbeek &lt;richel@richelbilderbeek.nl&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-02 13:20:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='alpha_parameter_to_xml'>Internal function</h2><span id='topic+alpha_parameter_to_xml'></span>

<h3>Description</h3>

<p>Converts an alpha parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alpha_parameter_to_xml(alpha_parameter, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alpha_parameter_to_xml_+3A_alpha_parameter">alpha_parameter</code></td>
<td>
<p>an alpha parameter,
as created by <a href="#topic+create_alpha_param">create_alpha_param</a></p>
</td></tr>
<tr><td><code id="alpha_parameter_to_xml_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>remove_beautier_folder()
check_empty_beautier_folder()

# The alpha parameter must be initialized, i.e. have an ID
alpha_parameter_to_xml(
  alpha_parameter = create_alpha_param(id = "1"),
  beauti_options = create_beauti_options()
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='are_clock_models'>Determine if x consists out of clock_models objects</h2><span id='topic+are_clock_models'></span>

<h3>Description</h3>

<p>Determine if x consists out of clock_models objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_clock_models(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_clock_models_+3A_x">x</code></td>
<td>
<p>the object to check if it consists out of clock_models objects</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x, or all elements of x, are clock_model objects
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

rln_clock_model &lt;- create_rln_clock_model()
strict_clock_model &lt;- create_strict_clock_model()
both_clock_models &lt;- list(rln_clock_model, strict_clock_model)
# TRUE
are_clock_models(rln_clock_model)
are_clock_models(strict_clock_model)
are_clock_models(both_clock_models)

# FALSE
are_clock_models(NA)
are_clock_models(NULL)
are_clock_models("nonsense")
are_clock_models(create_jc69_site_model())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='are_equal_mcmcs'>Determine if two MCMCs are equal.</h2><span id='topic+are_equal_mcmcs'></span>

<h3>Description</h3>

<p>Will <a href="base.html#topic+stop">stop</a> if the arguments are not MCMCs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_equal_mcmcs(mcmc_1, mcmc_2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_equal_mcmcs_+3A_mcmc_1">mcmc_1</code></td>
<td>
<p>an MCMC, as created by <code><a href="#topic+create_mcmc">create_mcmc</a></code></p>
</td></tr>
<tr><td><code id="are_equal_mcmcs_+3A_mcmc_2">mcmc_2</code></td>
<td>
<p>an MCMC, as created by <code><a href="#topic+create_mcmc">create_mcmc</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the two MCMCs are equal
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create an MCMC
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  check_empty_beautier_folder()

  mcmc_1 &lt;- create_mcmc(chain_length = 1000)
  mcmc_2 &lt;- create_mcmc(chain_length = 314)
  # TRUE
  are_equal_mcmcs(mcmc_1, mcmc_1)
  # FALSE
  are_equal_mcmcs(mcmc_1, mcmc_2)

  check_empty_beautier_folder()
}
</code></pre>

<hr>
<h2 id='are_equal_screenlogs'>Determine if two screenlogs are equal.</h2><span id='topic+are_equal_screenlogs'></span>

<h3>Description</h3>

<p>Will <a href="base.html#topic+stop">stop</a> if the arguments are not screenlogs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_equal_screenlogs(screenlog_1, screenlog_2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_equal_screenlogs_+3A_screenlog_1">screenlog_1</code></td>
<td>
<p>an screenlog, as created by <a href="#topic+create_screenlog">create_screenlog</a></p>
</td></tr>
<tr><td><code id="are_equal_screenlogs_+3A_screenlog_2">screenlog_2</code></td>
<td>
<p>an screenlog, as created by <a href="#topic+create_screenlog">create_screenlog</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the two screenlogs are equal
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_screenlog">create_screenlog</a> to create an screenlog
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

screenlog_1 &lt;- create_screenlog(log_every = 1000)
screenlog_2 &lt;- create_screenlog(log_every = 314)
# TRUE
are_equal_screenlogs(screenlog_1, screenlog_1)
# FALSE
are_equal_screenlogs(screenlog_1, screenlog_2)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='are_equal_tracelogs'>Determine if two tracelogs are equal.</h2><span id='topic+are_equal_tracelogs'></span>

<h3>Description</h3>

<p>Will <a href="base.html#topic+stop">stop</a> if the arguments are not tracelogs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_equal_tracelogs(tracelog_1, tracelog_2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_equal_tracelogs_+3A_tracelog_1">tracelog_1</code></td>
<td>
<p>an tracelog, as created by <a href="#topic+create_tracelog">create_tracelog</a></p>
</td></tr>
<tr><td><code id="are_equal_tracelogs_+3A_tracelog_2">tracelog_2</code></td>
<td>
<p>an tracelog, as created by <a href="#topic+create_tracelog">create_tracelog</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the two tracelogs are equal
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_tracelog">create_tracelog</a> to create an tracelog
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

tracelog_1 &lt;- create_tracelog(log_every = 1000)
tracelog_2 &lt;- create_tracelog(log_every = 314)
# TRUE
are_equal_tracelogs(tracelog_1, tracelog_1)
# FALSE
are_equal_tracelogs(tracelog_1, tracelog_2)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='are_equal_treelogs'>Determine if two treelogs are equal.</h2><span id='topic+are_equal_treelogs'></span>

<h3>Description</h3>

<p>Will <a href="base.html#topic+stop">stop</a> if the arguments are not treelogs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_equal_treelogs(treelog_1, treelog_2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_equal_treelogs_+3A_treelog_1">treelog_1</code></td>
<td>
<p>an treelog, as created by <a href="#topic+create_treelog">create_treelog</a></p>
</td></tr>
<tr><td><code id="are_equal_treelogs_+3A_treelog_2">treelog_2</code></td>
<td>
<p>an treelog, as created by <a href="#topic+create_treelog">create_treelog</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the two treelogs are equal
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_treelog">create_treelog</a> to create an treelog
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

treelog_1 &lt;- create_treelog(log_every = 1000)
treelog_2 &lt;- create_treelog(log_every = 314)
# TRUE
are_equal_treelogs(treelog_1, treelog_1)
# FALSE
are_equal_treelogs(treelog_1, treelog_2)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='are_equal_xml_files'>Determine if XML files result in equal trees</h2><span id='topic+are_equal_xml_files'></span>

<h3>Description</h3>

<p>Determine if XML files result in equal trees
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_equal_xml_files(filename_1, filename_2, section)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_equal_xml_files_+3A_filename_1">filename_1</code></td>
<td>
<p>name of a first XML file</p>
</td></tr>
<tr><td><code id="are_equal_xml_files_+3A_filename_2">filename_2</code></td>
<td>
<p>name of a second XML file</p>
</td></tr>
<tr><td><code id="are_equal_xml_files_+3A_section">section</code></td>
<td>
<p>name of an XML section.
Assumes that there is one line that starts with <code>&lt;section</code>
(excluding whitespace)
and one line that is <code>&lt;/section&gt;</code> (also excluding whitespace)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the two sections of the XML files are equal,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>to check for equivalence, use <code><a href="#topic+are_equivalent_xml_files">are_equivalent_xml_files</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

are_equal_xml_files(
  filename_1 = get_beautier_path("2_4.xml"),
  filename_2 = get_beautier_path("2_6_0.xml"),
  section = "taxonset"
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='are_equal_xml_lines'>Determine if XML lines result in equal trees</h2><span id='topic+are_equal_xml_lines'></span>

<h3>Description</h3>

<p>Determine if XML lines result in equal trees
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_equal_xml_lines(lines_1, lines_2, section)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_equal_xml_lines_+3A_lines_1">lines_1</code></td>
<td>
<p>lines of a first XML file</p>
</td></tr>
<tr><td><code id="are_equal_xml_lines_+3A_lines_2">lines_2</code></td>
<td>
<p>lines of a second XML file</p>
</td></tr>
<tr><td><code id="are_equal_xml_lines_+3A_section">section</code></td>
<td>
<p>name of an XML section.
Assumes that there is one line that starts with <code>&lt;section</code>
(excluding whitespace)
and one line that is <code>&lt;/section&gt;</code> (also excluding whitespace)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the two sections of the XML files are equal,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

are_equal_xml_lines(
  lines_1 = readLines(get_beautier_path("2_4.xml")),
  lines_2 = readLines(get_beautier_path("2_6_0.xml")),
  section = "taxonset"
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='are_equivalent_xml_files'>Internal function</h2><span id='topic+are_equivalent_xml_files'></span>

<h3>Description</h3>

<p>Internal function used for debugging to
determine if XML files result in equivalent trees
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_equivalent_xml_files(filename_1, filename_2, section = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_equivalent_xml_files_+3A_filename_1">filename_1</code></td>
<td>
<p>name of a first XML file</p>
</td></tr>
<tr><td><code id="are_equivalent_xml_files_+3A_filename_2">filename_2</code></td>
<td>
<p>name of a second XML file</p>
</td></tr>
<tr><td><code id="are_equivalent_xml_files_+3A_section">section</code></td>
<td>
<p>the name of the XML section, use NA to check the whole file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the two XML files result in equivalent trees,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>to check for equality, use <code>are_equal_xml_files</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

are_equivalent_xml_files(
  filename_1 = get_beautier_path("2_4.xml"),
  filename_2 = get_beautier_path("2_6_0.xml")
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='are_equivalent_xml_lines'>Determine if XML lines result in equivalent trees</h2><span id='topic+are_equivalent_xml_lines'></span>

<h3>Description</h3>

<p>Determine if XML lines result in equivalent trees
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_equivalent_xml_lines(lines_1, lines_2, section = NA, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_equivalent_xml_lines_+3A_lines_1">lines_1</code></td>
<td>
<p>lines of a first XML file</p>
</td></tr>
<tr><td><code id="are_equivalent_xml_lines_+3A_lines_2">lines_2</code></td>
<td>
<p>lines of a second XML file</p>
</td></tr>
<tr><td><code id="are_equivalent_xml_lines_+3A_section">section</code></td>
<td>
<p>the name of the XML section</p>
</td></tr>
<tr><td><code id="are_equivalent_xml_lines_+3A_verbose">verbose</code></td>
<td>
<p>if TRUE, additional information is displayed, that
is potentially useful in debugging</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the two XML lines result in equivalent trees,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='are_equivalent_xml_lines_all'>Determine if XML lines result in equivalent trees</h2><span id='topic+are_equivalent_xml_lines_all'></span>

<h3>Description</h3>

<p>Determine if XML lines result in equivalent trees
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_equivalent_xml_lines_all(lines_1, lines_2, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_equivalent_xml_lines_all_+3A_lines_1">lines_1</code></td>
<td>
<p>lines of a first XML file</p>
</td></tr>
<tr><td><code id="are_equivalent_xml_lines_all_+3A_lines_2">lines_2</code></td>
<td>
<p>lines of a second XML file</p>
</td></tr>
<tr><td><code id="are_equivalent_xml_lines_all_+3A_verbose">verbose</code></td>
<td>
<p>if TRUE, additional information is displayed, that
is potentially useful in debugging</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the two XML lines result in equivalent trees,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='are_equivalent_xml_lines_loggers'>Determine if XML operator lines result in equivalent trees</h2><span id='topic+are_equivalent_xml_lines_loggers'></span>

<h3>Description</h3>

<p>Determine if XML operator lines result in equivalent trees
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_equivalent_xml_lines_loggers(lines_1, lines_2, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_equivalent_xml_lines_loggers_+3A_lines_1">lines_1</code></td>
<td>
<p>lines of a first XML file</p>
</td></tr>
<tr><td><code id="are_equivalent_xml_lines_loggers_+3A_lines_2">lines_2</code></td>
<td>
<p>lines of a second XML file</p>
</td></tr>
<tr><td><code id="are_equivalent_xml_lines_loggers_+3A_verbose">verbose</code></td>
<td>
<p>if TRUE, additional information is displayed, that
is potentially useful in debugging</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the two XML lines result in equivalent trees,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='are_equivalent_xml_lines_operators'>Determine if XML operator lines result in equivalent trees</h2><span id='topic+are_equivalent_xml_lines_operators'></span>

<h3>Description</h3>

<p>Determine if XML operator lines result in equivalent trees
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_equivalent_xml_lines_operators(lines_1, lines_2, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_equivalent_xml_lines_operators_+3A_lines_1">lines_1</code></td>
<td>
<p>lines of a first XML file</p>
</td></tr>
<tr><td><code id="are_equivalent_xml_lines_operators_+3A_lines_2">lines_2</code></td>
<td>
<p>lines of a second XML file</p>
</td></tr>
<tr><td><code id="are_equivalent_xml_lines_operators_+3A_verbose">verbose</code></td>
<td>
<p>if TRUE, additional information is displayed, that
is potentially useful in debugging</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the two XML lines result in equivalent trees,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='are_equivalent_xml_lines_section'>Determine if XML lines result in equivalent trees</h2><span id='topic+are_equivalent_xml_lines_section'></span>

<h3>Description</h3>

<p>Determine if XML lines result in equivalent trees
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_equivalent_xml_lines_section(lines_1, lines_2, section, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_equivalent_xml_lines_section_+3A_lines_1">lines_1</code></td>
<td>
<p>lines of a first XML file</p>
</td></tr>
<tr><td><code id="are_equivalent_xml_lines_section_+3A_lines_2">lines_2</code></td>
<td>
<p>lines of a second XML file</p>
</td></tr>
<tr><td><code id="are_equivalent_xml_lines_section_+3A_section">section</code></td>
<td>
<p>the name of the XML section</p>
</td></tr>
<tr><td><code id="are_equivalent_xml_lines_section_+3A_verbose">verbose</code></td>
<td>
<p>if TRUE, additional information is displayed, that
is potentially useful in debugging</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the two XML lines result in equivalent trees,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='are_fasta_filenames'>Checks if all filenames have a FASTA filename extension</h2><span id='topic+are_fasta_filenames'></span>

<h3>Description</h3>

<p>Checks if all filenames have a FASTA filename extension
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_fasta_filenames(filenames)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_fasta_filenames_+3A_filenames">filenames</code></td>
<td>
<p>filenames</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if all filenames have a FASTA filename extension
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
are_fasta_filenames("1.fas")
are_fasta_filenames("1.fasta")
are_fasta_filenames("1.FAS")
are_fasta_filenames("1.FASTA")
are_fasta_filenames(c("1.fas", "2.fas"))

# FALSE
are_fasta_filenames("")
are_fasta_filenames(NA)
are_fasta_filenames(NULL)
are_fasta_filenames(Inf)
are_fasta_filenames("1.fasX")
are_fasta_filenames(c("1.fas", "2.exe"))
are_fasta_filenames(c("1.bat", "2.exe"))

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='are_ids'>Determine if x consists out of IDs</h2><span id='topic+are_ids'></span>

<h3>Description</h3>

<p>Determine if x consists out of IDs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_ids(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_ids_+3A_x">x</code></td>
<td>
<p>the object to check if it consists out of IDs</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x, or all elements of x, are IDs
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>to check one ID, use <a href="#topic+is_id">is_id</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
are_ids("anthus_aco")
are_ids(c("anthus_aco", "anthus_nd2"))
are_ids(list("anthus_aco", "anthus_nd2"))
are_ids(c(1, 2))
are_ids(1)

# FALSE
are_ids(NULL)
are_ids(NA)
are_ids(c())
are_ids(ape::rcoal(3))
are_ids(c(ape::rcoal(3), ape::rcoal(4)))

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='are_init_clock_models'>Determine if x consists out of initialized clock_models objects</h2><span id='topic+are_init_clock_models'></span>

<h3>Description</h3>

<p>Determine if x consists out of initialized clock_models objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_init_clock_models(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_init_clock_models_+3A_x">x</code></td>
<td>
<p>the object to check if it consists out of
initialized clock_models objects</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x, or all elements of x, are initialized clock_model objects
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='are_init_mrca_priors'>Determine if x consists out of initialized MRCA priors</h2><span id='topic+are_init_mrca_priors'></span>

<h3>Description</h3>

<p>Determine if x consists out of initialized MRCA priors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_init_mrca_priors(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_init_mrca_priors_+3A_x">x</code></td>
<td>
<p>the object to check if it consists out of
initialized MRCA priors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x, or all elements of x, are initialized MRCA priors
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='are_init_site_models'>Determine if x consists out of initialized site_models objects</h2><span id='topic+are_init_site_models'></span>

<h3>Description</h3>

<p>Determine if x consists out of initialized site_models objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_init_site_models(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_init_site_models_+3A_x">x</code></td>
<td>
<p>the object to check if it consists out of
initialized site_models objects</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x, or all elements of x, are initialized site_model objects
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='are_init_tree_priors'>Determine if x consists out of initialized tree_priors objects</h2><span id='topic+are_init_tree_priors'></span>

<h3>Description</h3>

<p>Determine if x consists out of initialized tree_priors objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_init_tree_priors(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_init_tree_priors_+3A_x">x</code></td>
<td>
<p>the object to check if it consists out of
initialized tree_priors objects</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x, or all elements of x, are initialized tree_prior objects
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='are_mrca_align_ids_in_fasta'>Determine if the MRCA priors' alignment IDs are present in the FASTA files</h2><span id='topic+are_mrca_align_ids_in_fasta'></span>

<h3>Description</h3>

<p>Determine if the MRCA priors' alignment IDs are present in the FASTA files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_mrca_align_ids_in_fasta(mrca_prior, fasta_filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_mrca_align_ids_in_fasta_+3A_mrca_prior">mrca_prior</code></td>
<td>
<p>a Most Recent Common Ancestor prior,
as returned by <code><a href="#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
<tr><td><code id="are_mrca_align_ids_in_fasta_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>a FASTA filename.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.
Note that BEAST2 also supports missing data,
by using a dash (<code>-</code>) or question mark (<code>?</code>)
as a sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if all the MRCA priors' alignment IDs
are present in the FASTA files.
Returns FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='are_mrca_priors'>Determine if x consists out of MRCA priors</h2><span id='topic+are_mrca_priors'></span>

<h3>Description</h3>

<p>Determine if x consists out of MRCA priors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_mrca_priors(mrca_priors)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_mrca_priors_+3A_mrca_priors">mrca_priors</code></td>
<td>
<p>a list of one or more Most Recent Common Ancestor priors,
as returned by <code><a href="#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x, or all elements of x, are MRCA priors.
Returns FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='are_mrca_taxon_names_in_fasta'>Determine if the MRCA priors' taxa names are present in the FASTA files</h2><span id='topic+are_mrca_taxon_names_in_fasta'></span>

<h3>Description</h3>

<p>Determine if the MRCA priors' taxa names are present in the FASTA files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_mrca_taxon_names_in_fasta(mrca_prior, fasta_filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_mrca_taxon_names_in_fasta_+3A_mrca_prior">mrca_prior</code></td>
<td>
<p>a Most Recent Common Ancestor prior,
as returned by <code><a href="#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
<tr><td><code id="are_mrca_taxon_names_in_fasta_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>a FASTA filename.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.
Note that BEAST2 also supports missing data,
by using a dash (<code>-</code>) or question mark (<code>?</code>)
as a sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the MRCA priors' taxa names are
present in the FASTA files. FALSE otherwise.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='are_rln_clock_models'>Are the clock models Relaxed Log-Normal clock models?</h2><span id='topic+are_rln_clock_models'></span>

<h3>Description</h3>

<p>Are the clock models Relaxed Log-Normal clock models?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_rln_clock_models(clock_models)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_rln_clock_models_+3A_clock_models">clock_models</code></td>
<td>
<p>a list of one or more clock models,
as returned by <code><a href="#topic+create_clock_model">create_clock_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of booleans with the same length
as the number of clock models in <code>clock_models</code>.
Each nth element is TRUE if the nth element
in <code>clock_models</code> is a relaxed log-normal
clock model, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='are_site_models'>Determine if x consists out of site_models objects</h2><span id='topic+are_site_models'></span>

<h3>Description</h3>

<p>Determine if x consists out of site_models objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_site_models(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_site_models_+3A_x">x</code></td>
<td>
<p>the object to check if it consists out of site_models objects</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x, or all elements of x, are site_model objects
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_site_model">create_site_model</a> to create a site model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

jc69_site_model &lt;- create_jc69_site_model()
gtr_site_model &lt;- create_gtr_site_model()
both_site_models &lt;- list(jc69_site_model, gtr_site_model)

# TRUE
are_site_models(jc69_site_model)

# TRUE
are_site_models(gtr_site_model)

# TRUE
are_site_models(both_site_models)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='are_tree_priors'>Determine if x consists out of tree_priors objects</h2><span id='topic+are_tree_priors'></span>

<h3>Description</h3>

<p>Determine if x consists out of tree_priors objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_tree_priors(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_tree_priors_+3A_x">x</code></td>
<td>
<p>the object to check if it consists out of tree_priors objects</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x, or all elements of x, are tree_prior objects
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_yule_tree_prior">create_yule_tree_prior</a> to create a Yule tree prior
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

yule_tree_prior &lt;- create_yule_tree_prior()
bd_tree_prior &lt;- create_bd_tree_prior()
both_tree_priors &lt;- list(yule_tree_prior, bd_tree_prior)
# TRUE
are_tree_priors(yule_tree_prior)
# TRUE
are_tree_priors(bd_tree_prior)
# TRUE
are_tree_priors(both_tree_priors)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='b_pop_sizes_param_to_xml'>Internal function</h2><span id='topic+b_pop_sizes_param_to_xml'></span>

<h3>Description</h3>

<p>Converts a 'bPopSizes' parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>b_pop_sizes_param_to_xml(
  b_pop_sizes_param,
  beauti_options = create_beauti_options()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="b_pop_sizes_param_to_xml_+3A_b_pop_sizes_param">b_pop_sizes_param</code></td>
<td>
<p>a Bayesian population size parameter,
as created by <a href="#topic+create_b_pop_sizes_param">create_b_pop_sizes_param</a></p>
</td></tr>
<tr><td><code id="b_pop_sizes_param_to_xml_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='b_pop_sizes_parameter_to_xml'>Internal function</h2><span id='topic+b_pop_sizes_parameter_to_xml'></span>

<h3>Description</h3>

<p>Converts a Bayesian population sizes parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>b_pop_sizes_parameter_to_xml(
  b_pop_sizes_parameter,
  beauti_options = create_beauti_options()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="b_pop_sizes_parameter_to_xml_+3A_b_pop_sizes_parameter">b_pop_sizes_parameter</code></td>
<td>
<p>a Bayesian population size parameter,
as created by <a href="#topic+create_b_pop_sizes_param">create_b_pop_sizes_param</a></p>
</td></tr>
<tr><td><code id="b_pop_sizes_parameter_to_xml_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b_pop_sizes_parameter_to_xml(
  b_pop_sizes_parameter = create_b_pop_sizes_param(id = 42),
  beauti_options = create_beauti_options()
)
b_pop_sizes_parameter_to_xml(
  b_pop_sizes_parameter = create_b_pop_sizes_param(id = 42, upper = Inf),
  beauti_options = create_beauti_options()
)
</code></pre>

<hr>
<h2 id='bd_tree_prior_to_xml_prior_distr'>Creates the tree prior section in the prior section of
the prior section of the distribution section
of a BEAST2 XML parameter file for a Birth-Death tree prior</h2><span id='topic+bd_tree_prior_to_xml_prior_distr'></span>

<h3>Description</h3>

<p>Creates the tree prior section in the prior section of
the prior section of the distribution section
of a BEAST2 XML parameter file for a Birth-Death tree prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bd_tree_prior_to_xml_prior_distr(bd_tree_prior, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bd_tree_prior_to_xml_prior_distr_+3A_bd_tree_prior">bd_tree_prior</code></td>
<td>
<p>a Birth-Death tree prior, as created
by <code><a href="#topic+create_bd_tree_prior">create_bd_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="bd_tree_prior_to_xml_prior_distr_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

 # &lt;distribution id="posterior" spec="util.CompoundDistribution"&gt;
 #     &lt;distribution id="prior" spec="util.CompoundDistribution"&gt;
 #       HERE, where the ID of the distribution is 'prior'
 #     &lt;/distribution&gt;
 #     &lt;distribution id="likelihood" ...&gt;
 #     &lt;/distribution&gt;
 # &lt;/distribution&gt;

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='beautier-package'><code>beautier</code>: A package to create a <code>BEAST2</code> input file.</h2><span id='topic+beautier'></span><span id='topic+beautier-package'></span>

<h3>Description</h3>

<p><code>beautier</code> allows to create a <code>BEAST2</code> input file, using
an R interface. <code>beautier</code> closely follows the interface
of <code>BEAUti 2</code>, a GUI tool bundled with <code>BEAST2</code>, including
its default settings.
</p>


<h3>Details</h3>

<p>See the documentation of <code>create_inference_model</code> to see the
features of BEAST2 that <code>beautier</code> supports.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>These are packages associated with <code>beautier</code>:
</p>

<ul>
<li>
<p>The package <code>beastier</code> can run
BEAST2 from R

</p>
</li>
<li>
<p>The package <code>tracerer</code> can parse
BEAST2 output files from R

</p>
</li>
<li>
<p>The package <code>mauricer</code> manages
BEAST2 packages from R

</p>
</li>
<li>
<p>The package <code>babette</code> combines the
functionality of <code>beautier</code>,
<code>beastier</code>, <code>mauricer</code> and <code>tracerer</code>
into a single workflow

</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  check_empty_beautier_folder()

  # Get an example FASTA file
  input_filename &lt;- get_fasta_filename()

  # The file created by beautier, a BEAST2 input file
  output_filename &lt;- get_beautier_tempfilename()

  # Use the default BEAUti settings to create a BEAST2 input file
  create_beast2_input_file_from_model(
    input_filename,
    output_filename,
    inference_model = create_inference_model()
  )

  # Remove the folder that contains the beautier temporary files
  remove_beautier_folder()
  check_empty_beautier_folder()
}
</code></pre>

<hr>
<h2 id='beta_parameter_to_xml'>Internal function</h2><span id='topic+beta_parameter_to_xml'></span>

<h3>Description</h3>

<p>Converts a beta parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>beta_parameter_to_xml(beta_parameter, beauti_options = create_beauti_options())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="beta_parameter_to_xml_+3A_beta_parameter">beta_parameter</code></td>
<td>
<p>a beta parameter,
as created by <a href="#topic+create_beta_param">create_beta_param</a></p>
</td></tr>
<tr><td><code id="beta_parameter_to_xml_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='cbs_tree_prior_to_xml_prior_distr'>Creates the tree prior section in the prior section of
the prior section of the distribution section
of a BEAST2 XML parameter file for a Birth-Death tree prior</h2><span id='topic+cbs_tree_prior_to_xml_prior_distr'></span>

<h3>Description</h3>

<p>Creates the tree prior section in the prior section of
the prior section of the distribution section
of a BEAST2 XML parameter file for a Birth-Death tree prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cbs_tree_prior_to_xml_prior_distr(cbs_tree_prior, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cbs_tree_prior_to_xml_prior_distr_+3A_cbs_tree_prior">cbs_tree_prior</code></td>
<td>
<p>a Coalescent Bayesian Skyline tree prior,
as returned by <code><a href="#topic+create_cbs_tree_prior">create_cbs_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="cbs_tree_prior_to_xml_prior_distr_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

 # &lt;distribution id="posterior" spec="util.CompoundDistribution"&gt;
 #     &lt;distribution id="prior" spec="util.CompoundDistribution"&gt;
 #       HERE, where the ID of the distribution is 'prior'
 #     &lt;/distribution&gt;
 #     &lt;distribution id="likelihood" ...&gt;
 #     &lt;/distribution&gt;
 # &lt;/distribution&gt;

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='ccp_tree_prior_to_xml_prior_distr'>Creates the tree prior section in the prior section of
the prior section of the distribution section
of a BEAST2 XML parameter file for a
Coalescent Constant Population tree prior</h2><span id='topic+ccp_tree_prior_to_xml_prior_distr'></span>

<h3>Description</h3>

<p>Creates the tree prior section in the prior section of
the prior section of the distribution section
of a BEAST2 XML parameter file for a
Coalescent Constant Population tree prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ccp_tree_prior_to_xml_prior_distr(ccp_tree_prior, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ccp_tree_prior_to_xml_prior_distr_+3A_ccp_tree_prior">ccp_tree_prior</code></td>
<td>
<p>a Coalescent Constant Population tree prior,
as returned by <code><a href="#topic+create_ccp_tree_prior">create_ccp_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="ccp_tree_prior_to_xml_prior_distr_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

 # &lt;distribution id="posterior" spec="util.CompoundDistribution"&gt;
 #     &lt;distribution id="prior" spec="util.CompoundDistribution"&gt;
 #       HERE, where the ID of the distribution is 'prior'
 #     &lt;/distribution&gt;
 #     &lt;distribution id="likelihood" ...&gt;
 #     &lt;/distribution&gt;
 # &lt;/distribution&gt;

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='ccp_tree_prior_to_xml_state'>Convert a CCP tree prior
to the XML as part of the <code>state</code> section</h2><span id='topic+ccp_tree_prior_to_xml_state'></span>

<h3>Description</h3>

<p>Convert a CCP tree prior
to the XML as part of the <code>state</code> section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ccp_tree_prior_to_xml_state(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ccp_tree_prior_to_xml_state_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>XML as text
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# Need an ID and inital value
inference_model &lt;- create_inference_model(
  tree_prior = create_ccp_tree_prior(
    id = "anthus_nd2_sub",
    pop_size_distr = create_normal_distr(
      id = 123,
      value = 3.14
    )
  )
)

ccp_tree_prior_to_xml_state(inference_model)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='cep_tree_prior_to_xml_prior_distr'>Creates the tree prior section in the prior section of
the prior section of the distribution section
of a BEAST2 XML parameter file for a
Coalescent Exponential Population tree prior</h2><span id='topic+cep_tree_prior_to_xml_prior_distr'></span>

<h3>Description</h3>

<p>Creates the tree prior section in the prior section of
the prior section of the distribution section
of a BEAST2 XML parameter file for a
Coalescent Exponential Population tree prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cep_tree_prior_to_xml_prior_distr(cep_tree_prior, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cep_tree_prior_to_xml_prior_distr_+3A_cep_tree_prior">cep_tree_prior</code></td>
<td>
<p>a Coalescent Exponential Population tree prior,
as returned by <code><a href="#topic+create_cep_tree_prior">create_cep_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="cep_tree_prior_to_xml_prior_distr_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

 # &lt;distribution id="posterior" spec="util.CompoundDistribution"&gt;
 #     &lt;distribution id="prior" spec="util.CompoundDistribution"&gt;
 #       HERE, where the ID of the distribution is 'prior'
 #     &lt;/distribution&gt;
 #     &lt;distribution id="likelihood" ...&gt;
 #     &lt;/distribution&gt;
 # &lt;/distribution&gt;

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_alignment_id'>Check if the <code>alignment_id</code> is valid.</h2><span id='topic+check_alignment_id'></span>

<h3>Description</h3>

<p>Will <a href="base.html#topic+stop">stop</a> if not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_alignment_id(alignment_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_alignment_id_+3A_alignment_id">alignment_id</code></td>
<td>
<p>ID of the alignment,
as returned by <a href="#topic+get_alignment_id">get_alignment_id</a>.
Keep at <code>NA</code> to have it initialized automatically</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing, will <a href="base.html#topic+stop">stop</a> if needed
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# anthus_aco_sub
alignment_id &lt;- get_alignment_id("/home/homer/anthus_aco_sub.fas")
check_alignment_id(alignment_id)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_beauti_options'>Check if the <code>beauti_options</code> is a valid <code>beauti_options</code> object.</h2><span id='topic+check_beauti_options'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if the <code>beauti_options</code> object is invalid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_beauti_options(beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_beauti_options_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_beauti_options">create_beauti_options</a> to create a valid
BEAUti options setup
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

check_beauti_options(create_beauti_options())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_clock_model'>Check if the clock model is a valid clock model.</h2><span id='topic+check_clock_model'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if the clock model is invalid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_clock_model(clock_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_clock_model_+3A_clock_model">clock_model</code></td>
<td>
<p>a clock model,
as returned by <code><a href="#topic+create_clock_model">create_clock_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if <code>clock_model</code> is a valid clock model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_clock_model">create_clock_model</a> to create a valid clock model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

check_clock_model(create_strict_clock_model())
check_clock_model(create_rln_clock_model())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_clock_models'>Check if the object is a list of one or more clock models.</h2><span id='topic+check_clock_models'></span>

<h3>Description</h3>

<p>Will <a href="base.html#topic+stop">stop</a> if the object is not a list of one or more clock models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_clock_models(clock_models)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_clock_models_+3A_clock_models">clock_models</code></td>
<td>
<p>the object to be checked if it is a list of one
or more valid clock models</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing.
Will <a href="base.html#topic+stop">stop</a> if the object is not a list of one or more clock models.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_clock_model">create_clock_model</a> to create a valid clock model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

check_clock_models(create_strict_clock_model())
check_clock_models(list(create_strict_clock_model()))
check_clock_models(
  list(create_strict_clock_model(), create_rln_clock_model())
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_empty_beautier_folder'>Internal function</h2><span id='topic+check_empty_beautier_folder'></span>

<h3>Description</h3>

<p>Internal function to verify that, if there are 'beautier'
temporary files created, these are also cleaned up,
as by CRAN policy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_empty_beautier_folder(beautier_folder = get_beautier_folder())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_empty_beautier_folder_+3A_beautier_folder">beautier_folder</code></td>
<td>
<p>the path to
the <a href="#topic+beautier">beautier</a> temporary files folder</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the 'beautier' folder does not exist, this function
does nothing.
If there are folder and/or files in the 'beautier' folder,
an error is given.
</p>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <a href="#topic+remove_beautier_folder">remove_beautier_folder</a> to remove the default
'beautier' folder
</p>


<h3>Examples</h3>

<pre><code class='language-R'>remove_beautier_folder()

check_empty_beautier_folder()

remove_beautier_folder()
</code></pre>

<hr>
<h2 id='check_file_and_model_agree'>Checks if the input FASTA file and the inference model agree.</h2><span id='topic+check_file_and_model_agree'></span>

<h3>Description</h3>

<p>Will <a href="base.html#topic+stop">stop</a> if not
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_file_and_model_agree(input_filename, inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_file_and_model_agree_+3A_input_filename">input_filename</code></td>
<td>
<p>A FASTA filename.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.</p>
</td></tr>
<tr><td><code id="check_file_and_model_agree_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing, will <a href="base.html#topic+stop">stop</a> if needed
</p>

<hr>
<h2 id='check_file_exists'>Function to check if a file exists.
Calls <code>stop</code> if the file is absent</h2><span id='topic+check_file_exists'></span>

<h3>Description</h3>

<p>Function to check if a file exists.
Calls <code>stop</code> if the file is absent
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_file_exists(filename, filename_description = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_file_exists_+3A_filename">filename</code></td>
<td>
<p>name of the file</p>
</td></tr>
<tr><td><code id="check_file_exists_+3A_filename_description">filename_description</code></td>
<td>
<p>description of the filename</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing. Will <code>stop</code> if the file is absent,
with a proper error message
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

check_file_exists(get_beautier_path("anthus_aco_sub.fas"))

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_filename'>Check if the 'filename' is valid</h2><span id='topic+check_filename'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if the filename is invalid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_filename(filename, allow_empty_str = FALSE, allow_na = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_filename_+3A_filename">filename</code></td>
<td>
<p>a filename, as can be checked by <a href="#topic+check_filename">check_filename</a></p>
</td></tr>
<tr><td><code id="check_filename_+3A_allow_empty_str">allow_empty_str</code></td>
<td>
<p>allow a string to be empty</p>
</td></tr>
<tr><td><code id="check_filename_+3A_allow_na">allow_na</code></td>
<td>
<p>allow <a href="base.html#topic+NA">NA</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>The filename (invisibly)
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

check_filename("trace.log")
check_filename("my.trees")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_gamma_site_model'>Checks if the parameter is a valid gamma site model</h2><span id='topic+check_gamma_site_model'></span>

<h3>Description</h3>

<p>Checks if the parameter is a valid gamma site model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_gamma_site_model(gamma_site_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_gamma_site_model_+3A_gamma_site_model">gamma_site_model</code></td>
<td>
<p>a site model's gamma site model,
as returned by <code><a href="#topic+create_gamma_site_model">create_gamma_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing. Will call <code>stop</code> if the argument is not a valid
gamma site model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

check_gamma_site_model(create_gamma_site_model())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_gamma_site_model_names'>Checks if the gamma site model has the right list elements' names</h2><span id='topic+check_gamma_site_model_names'></span>

<h3>Description</h3>

<p>Checks if the gamma site model has the right list elements' names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_gamma_site_model_names(gamma_site_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_gamma_site_model_names_+3A_gamma_site_model">gamma_site_model</code></td>
<td>
<p>a site model's gamma site model,
as returned by <code><a href="#topic+create_gamma_site_model">create_gamma_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing. Will call <code>stop</code> if the argument is not a valid
gamma site model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='check_gtr_site_model'>Check if the <code>gtr_site_model</code> is a valid
GTR nucleotide substitution model.</h2><span id='topic+check_gtr_site_model'></span>

<h3>Description</h3>

<p>Use <a href="#topic+create_gtr_site_model">create_gtr_site_model</a> to create a valid
GTR nucleotide substitution model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_gtr_site_model(gtr_site_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_gtr_site_model_+3A_gtr_site_model">gtr_site_model</code></td>
<td>
<p>a GTR site model,
as returned by <code><a href="#topic+create_gtr_site_model">create_gtr_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE is the <code>gtr_site_model</code> is a valid
GTR nucleotide substitution model, FALSE otherwise
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

check_gtr_site_model(create_gtr_site_model())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_gtr_site_model_names'>Check if the <code>gtr_site_model</code> has the list elements
of a valid <code>gtr_site_model</code> object.</h2><span id='topic+check_gtr_site_model_names'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if an element is missing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_gtr_site_model_names(gtr_site_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_gtr_site_model_names_+3A_gtr_site_model">gtr_site_model</code></td>
<td>
<p>a GTR site model,
as returned by <code><a href="#topic+create_gtr_site_model">create_gtr_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_gtr_site_model">create_gtr_site_model</a>
to create a valid <code>gtr_site_model</code>
</p>

<hr>
<h2 id='check_inference_model'>Check if the supplied object is a valid
Bayesian phylogenetic inference model.</h2><span id='topic+check_inference_model'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if the supplied object is not a valid
Bayesian phylogenetic inference model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_inference_model(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_inference_model_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_inference_model">create_inference_model</a> to create a valid Bayesian
phylogenetic inference model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

check_inference_model(create_inference_model())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_inference_models'>Check if the <code>inference_model</code> is a valid BEAUti inference model.</h2><span id='topic+check_inference_models'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_inference_models(inference_models)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_inference_models_+3A_inference_models">inference_models</code></td>
<td>
<p>a list of one or more inference models,
as can be created by <a href="#topic+create_inference_model">create_inference_model</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_inference_model">create_inference_model</a> to create a valid
BEAST2 options object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

check_inference_models(list(create_inference_model()))

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_is_monophyletic'>Check if <code>is_monophyletic</code> has a valid value.</h2><span id='topic+check_is_monophyletic'></span>

<h3>Description</h3>

<p>Will <a href="base.html#topic+stop">stop</a> if not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_is_monophyletic(is_monophyletic)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_is_monophyletic_+3A_is_monophyletic">is_monophyletic</code></td>
<td>
<p>boolean to indicate monophyly is assumed in
a Most Recent Common Ancestor prior,
as returned by <code><a href="#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='check_log_mode'>Check if the supplied <code>mode</code> is a valid logging mode.</h2><span id='topic+check_log_mode'></span>

<h3>Description</h3>

<p>Check if the supplied <code>mode</code> is a valid logging mode.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_log_mode(mode)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_log_mode_+3A_mode">mode</code></td>
<td>
<p>mode how to log.
Valid are <code>tree</code>, <code>autodetect</code> and <code>compound</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='check_log_sort'>Check if the supplied <code>sort</code> is a valid logging sorting option.</h2><span id='topic+check_log_sort'></span>

<h3>Description</h3>

<p>Check if the supplied <code>sort</code> is a valid logging sorting option.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_log_sort(sort)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_log_sort_+3A_sort">sort</code></td>
<td>
<p>how to sort the entries in a log.
Valid are <code>smart</code>, <code>none</code> and <code>alphabetic</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='check_mcmc'>Check if the MCMC is a valid MCMC object.</h2><span id='topic+check_mcmc'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if the MCMC is invalid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_mcmc(mcmc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_mcmc_+3A_mcmc">mcmc</code></td>
<td>
<p>one MCMC.
Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create an MCMC.
Use <code><a href="#topic+create_ns_mcmc">create_ns_mcmc</a></code> to create an MCMC
for a Nested Sampling run.
Use <code><a href="#topic+check_mcmc">check_mcmc</a></code> to check if an MCMC is valid.
Use <code><a href="#topic+rename_mcmc_filenames">rename_mcmc_filenames</a></code> to rename the filenames in an MCMC.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create a valid MCMC
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

check_mcmc(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_mcmc_list_element_names'>Check if the MCMC has the list elements of a valid MCMC object.</h2><span id='topic+check_mcmc_list_element_names'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if an element is missing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_mcmc_list_element_names(mcmc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_mcmc_list_element_names_+3A_mcmc">mcmc</code></td>
<td>
<p>one MCMC.
Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create an MCMC.
Use <code><a href="#topic+create_ns_mcmc">create_ns_mcmc</a></code> to create an MCMC
for a Nested Sampling run.
Use <code><a href="#topic+check_mcmc">check_mcmc</a></code> to check if an MCMC is valid.
Use <code><a href="#topic+rename_mcmc_filenames">rename_mcmc_filenames</a></code> to rename the filenames in an MCMC.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create a valid MCMC
</p>

<hr>
<h2 id='check_mcmc_values'>Check if the MCMC has the list elements with valid values
for being a valid MCMC object.</h2><span id='topic+check_mcmc_values'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if a value is invalid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_mcmc_values(mcmc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_mcmc_values_+3A_mcmc">mcmc</code></td>
<td>
<p>one MCMC.
Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create an MCMC.
Use <code><a href="#topic+create_ns_mcmc">create_ns_mcmc</a></code> to create an MCMC
for a Nested Sampling run.
Use <code><a href="#topic+check_mcmc">check_mcmc</a></code> to check if an MCMC is valid.
Use <code><a href="#topic+rename_mcmc_filenames">rename_mcmc_filenames</a></code> to rename the filenames in an MCMC.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create a valid MCMC
</p>

<hr>
<h2 id='check_mrca_prior'>Check if the MRCA prior is a valid MRCA prior.</h2><span id='topic+check_mrca_prior'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if the MRCA prior is invalid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_mrca_prior(mrca_prior)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_mrca_prior_+3A_mrca_prior">mrca_prior</code></td>
<td>
<p>a Most Recent Common Ancestor prior,
as returned by <code><a href="#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_mrca_prior">create_mrca_prior</a> to create a valid MRCA prior
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

fasta_filename &lt;- get_beautier_path("anthus_aco.fas")
mrca_prior &lt;- create_mrca_prior(
  alignment_id = get_alignment_id(fasta_filename = fasta_filename),
  taxa_names = get_taxa_names(filename = fasta_filename)
)
mrca_prior &lt;- create_mrca_prior(
 alignment_id = get_alignment_id(fasta_filename = fasta_filename),
 taxa_names = get_taxa_names(filename = fasta_filename)
)
check_mrca_prior(mrca_prior)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_mrca_prior_name'>Check if <code>mrca_prior_name</code> is a valid MRCA prior name.</h2><span id='topic+check_mrca_prior_name'></span>

<h3>Description</h3>

<p>A valid MRCA prior name is either <a href="base.html#topic+NA">NA</a> or one character string.
Will <a href="base.html#topic+stop">stop</a> if not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_mrca_prior_name(mrca_prior_name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_mrca_prior_name_+3A_mrca_prior_name">mrca_prior_name</code></td>
<td>
<p>the unique name of the MRCA prior,
for example a genus, family,
order or even class name.
Leave at <a href="base.html#topic+NA">NA</a> to have it named automatically.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='check_mrca_prior_names'>Check if the MRCA prior,
which is a list, has all the named elements.</h2><span id='topic+check_mrca_prior_names'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_mrca_prior_names(mrca_prior)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_mrca_prior_names_+3A_mrca_prior">mrca_prior</code></td>
<td>
<p>a Most Recent Common Ancestor prior,
as returned by <code><a href="#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+check_mrca_prior">check_mrca_prior</a> to check the entire MRCA prior
</p>

<hr>
<h2 id='check_mrca_prior_taxa_names'>Check the MRCA prior's taxon names are valid.</h2><span id='topic+check_mrca_prior_taxa_names'></span>

<h3>Description</h3>

<p>Will <a href="base.html#topic+stop">stop</a> if not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_mrca_prior_taxa_names(taxa_names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_mrca_prior_taxa_names_+3A_taxa_names">taxa_names</code></td>
<td>
<p>names of the taxa,
as returned by <code><a href="#topic+get_taxa_names">get_taxa_names</a></code>.
Keep at <code>NA</code> to have it initialized automatically,
using all taxa in the alignment</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='check_ns_mcmc'>Check if this an MCMC that uses Nested Sampling
to estimate a marginal likelihood.</h2><span id='topic+check_ns_mcmc'></span><span id='topic+check_mcmc_nested_sampling'></span><span id='topic+check_nested_sampling_mcmc'></span>

<h3>Description</h3>

<p>Will <a href="base.html#topic+stop">stop</a> if not, else will do nothing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_ns_mcmc(mcmc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_ns_mcmc_+3A_mcmc">mcmc</code></td>
<td>
<p>one MCMC.
Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create an MCMC.
Use <code><a href="#topic+create_ns_mcmc">create_ns_mcmc</a></code> to create an MCMC
for a Nested Sampling run.
Use <code><a href="#topic+check_mcmc">check_mcmc</a></code> to check if an MCMC is valid.
Use <code><a href="#topic+rename_mcmc_filenames">rename_mcmc_filenames</a></code> to rename the filenames in an MCMC.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <code><a href="#topic+create_ns_mcmc">create_ns_mcmc</a></code>
to create an MCMC that uses Nested Sampling
to estimate a marginal likelihood
</p>

<hr>
<h2 id='check_param'>Check if the parameter is a valid parameter</h2><span id='topic+check_param'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if the parameter is invalid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_param(param)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_param_+3A_param">param</code></td>
<td>
<p>a parameter, as can be created by <code><a href="#topic+create_param">create_param</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_param">create_param</a> to create a valid parameter
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

check_param(create_alpha_param())
check_param(create_beta_param())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_param_names'>Check if the <code>param</code> has the list elements
of a valid <code>param</code> object.</h2><span id='topic+check_param_names'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if an element is missing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_param_names(param)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_param_names_+3A_param">param</code></td>
<td>
<p>a parameter, as can be created by <code><a href="#topic+create_param">create_param</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_param">create_param</a> to create a valid <code>param</code>
</p>

<hr>
<h2 id='check_param_types'>Check if the <code>param</code> has the list elements
of the right type for a valid <code>param</code> object.</h2><span id='topic+check_param_types'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if an element has the incorrect type
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_param_types(param)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_param_types_+3A_param">param</code></td>
<td>
<p>a parameter, as can be created by <code><a href="#topic+create_param">create_param</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_param">create_param</a> to create a valid <code>param</code>
</p>

<hr>
<h2 id='check_phylogeny'>Check if the phylogeny is a valid phylogeny object.</h2><span id='topic+check_phylogeny'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if the phylogeny is invalid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_phylogeny(phylogeny)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_phylogeny_+3A_phylogeny">phylogeny</code></td>
<td>
<p>a phylogeny of type <code>phylo</code> from the <code>ape</code>
package</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code>ape::read.tree</code> to create a phylogeny
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# Must do nothing on phylogenies
phylogeny &lt;- ape::read.tree(text = "(A:1, B:1):1;")
check_phylogeny(phylogeny)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_rename_fun'>Check if the rename function is a valid filename rename function</h2><span id='topic+check_rename_fun'></span>

<h3>Description</h3>

<p>Will <a href="base.html#topic+stop">stop</a> if not
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_rename_fun(rename_fun)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_rename_fun_+3A_rename_fun">rename_fun</code></td>
<td>
<p>a function to rename a filename,
as can be checked by <a href="#topic+check_rename_fun">check_rename_fun</a>. This function should
have one argument, which will be a filename or <a href="base.html#topic+NA">NA</a>. The
function should <a href="base.html#topic+return">return</a> one filename (when passed one filename) or
one <a href="base.html#topic+NA">NA</a> (when passed one <a href="base.html#topic+NA">NA</a>).
Example rename functions are:
</p>

<ul>
<li> <p><a href="#topic+get_remove_dir_fun">get_remove_dir_fun</a> get a function that removes the directory
paths from the filenames, in effect turning these into local files
</p>
</li>
<li> <p><a href="#topic+get_replace_dir_fun">get_replace_dir_fun</a> get a function that replaces the directory
paths from the filenames
</p>
</li>
<li> <p><a href="#topic+get_remove_hex_fun">get_remove_hex_fun</a> get a function that removes the
hex string from filenames.
For example, <code>tracelog_82c1a522040.log</code> becomes <code>tracelog.log</code>
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='check_rln_clock_model'>Check if the clock model is a valid clock model.</h2><span id='topic+check_rln_clock_model'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if the clock model is invalid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_rln_clock_model(clock_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_rln_clock_model_+3A_clock_model">clock_model</code></td>
<td>
<p>a clock model,
as returned by <code><a href="#topic+create_clock_model">create_clock_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if <code>clock_model</code> is a valid clock model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_clock_model">create_clock_model</a> to create a valid clock model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

check_rln_clock_model(create_rln_clock_model())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_screenlog'>Check if a <code>screenlog</code> is valid.</h2><span id='topic+check_screenlog'></span>

<h3>Description</h3>

<p>Will call <a href="base.html#topic+stop">stop</a> if not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_screenlog(screenlog)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_screenlog_+3A_screenlog">screenlog</code></td>
<td>
<p>a <code>screenlog</code>,
as created by <a href="#topic+create_screenlog">create_screenlog</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

check_screenlog(create_test_screenlog())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_screenlog_names'>Check if the <code>screenlog</code> has the list elements
of a valid <code>screenlog</code> object.</h2><span id='topic+check_screenlog_names'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if an element is missing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_screenlog_names(screenlog)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_screenlog_names_+3A_screenlog">screenlog</code></td>
<td>
<p>a <code>screenlog</code>,
as created by <a href="#topic+create_screenlog">create_screenlog</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_screenlog">create_screenlog</a> to create a valid <code>screenlog</code>
</p>

<hr>
<h2 id='check_screenlog_values'>Check if the screenlog has the list elements with valid values
for being a valid screenlog object.</h2><span id='topic+check_screenlog_values'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if a value is invalid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_screenlog_values(screenlog)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_screenlog_values_+3A_screenlog">screenlog</code></td>
<td>
<p>a <code>screenlog</code>,
as created by <a href="#topic+create_screenlog">create_screenlog</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_screenlog">create_screenlog</a> to create a valid screenlog
</p>

<hr>
<h2 id='check_site_model'>Check if the site model is a valid site model</h2><span id='topic+check_site_model'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if the site models are invalid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_site_model(site_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_site_model_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_site_model">create_site_model</a> to create a valid site model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

check_site_model(create_jc69_site_model())
check_site_model(create_hky_site_model())
check_site_model(create_tn93_site_model())
check_site_model(create_gtr_site_model())

# Can use list of one site model
check_site_model(list(create_jc69_site_model()))

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_site_model_names'>Check if the <code>site_model</code> has the list elements
of a valid <code>site_model</code> object.</h2><span id='topic+check_site_model_names'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if an element is missing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_site_model_names(site_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_site_model_names_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_site_model">create_site_model</a> to create a valid <code>site_model</code>
</p>

<hr>
<h2 id='check_site_model_types'>Check if the <code>site_model</code> has the list elements
of the right type for a valid <code>site_model</code> object.</h2><span id='topic+check_site_model_types'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if an element has the incorrect type
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_site_model_types(site_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_site_model_types_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_site_model">create_site_model</a> to create a valid <code>site_model</code>
</p>

<hr>
<h2 id='check_site_models'>Check if the object is a list of one or more site models.</h2><span id='topic+check_site_models'></span>

<h3>Description</h3>

<p>Will <a href="base.html#topic+stop">stop</a> if the object is not a list of one or more site models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_site_models(site_models)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_site_models_+3A_site_models">site_models</code></td>
<td>
<p>the object to be checked if it is a list of one
or more valid site models</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing.
Will <a href="base.html#topic+stop">stop</a> if the object is not a list of one or more site models.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_site_model">create_site_model</a> to create a valid site model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

check_site_models(create_jc69_site_model())
check_site_models(list(create_jc69_site_model()))
check_site_models(
  list(create_jc69_site_model(), create_gtr_site_model())
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_store_every'>Check if <code>store_every</code> holds a valid value</h2><span id='topic+check_store_every'></span>

<h3>Description</h3>

<p>Will <a href="base.html#topic+stop">stop</a> if not
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_store_every(store_every)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_store_every_+3A_store_every">store_every</code></td>
<td>
<p>number of states the MCMC will process
before the posterior's state will be saved to file.
Use -1 or <code>NA</code> to use the default frequency.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='check_strict_clock_model'>Check if the clock model is a valid clock model.</h2><span id='topic+check_strict_clock_model'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if the clock model is invalid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_strict_clock_model(clock_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_strict_clock_model_+3A_clock_model">clock_model</code></td>
<td>
<p>a clock model,
as returned by <code><a href="#topic+create_clock_model">create_clock_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if <code>clock_model</code> is a valid clock model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_clock_model">create_clock_model</a> to create a valid clock model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

check_strict_clock_model(create_strict_clock_model())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_tn93_site_model'>Check if the <code>tn93_site_model</code> is a valid
TN93 nucleotide substitution model.</h2><span id='topic+check_tn93_site_model'></span>

<h3>Description</h3>

<p>Use <a href="#topic+create_tn93_site_model">create_tn93_site_model</a> to create a valid
TN93 nucleotide substitution model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_tn93_site_model(tn93_site_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_tn93_site_model_+3A_tn93_site_model">tn93_site_model</code></td>
<td>
<p>a TN93 site model,
as returned by <code><a href="#topic+create_tn93_site_model">create_tn93_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

check_tn93_site_model(create_tn93_site_model())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_tn93_site_model_names'>Check if the <code>tn93_site_model</code> has the list elements
of a valid <code>tn93_site_model</code> object.</h2><span id='topic+check_tn93_site_model_names'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if an element is missing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_tn93_site_model_names(tn93_site_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_tn93_site_model_names_+3A_tn93_site_model">tn93_site_model</code></td>
<td>
<p>a TN93 site model,
as returned by <code><a href="#topic+create_tn93_site_model">create_tn93_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_tn93_site_model">create_tn93_site_model</a>
to create a valid <code>tn93_site_model</code>
</p>

<hr>
<h2 id='check_tracelog'>Check if a <code>tracelog</code> is valid.</h2><span id='topic+check_tracelog'></span>

<h3>Description</h3>

<p>Will call <a href="base.html#topic+stop">stop</a> if not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_tracelog(tracelog)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_tracelog_+3A_tracelog">tracelog</code></td>
<td>
<p>a <code>tracelog</code>,
as created by <a href="#topic+create_tracelog">create_tracelog</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='check_tracelog_names'>Check if the <code>tracelog</code> has the list elements
of a valid <code>tracelog</code> object.</h2><span id='topic+check_tracelog_names'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if an element is missing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_tracelog_names(tracelog)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_tracelog_names_+3A_tracelog">tracelog</code></td>
<td>
<p>a <code>tracelog</code>,
as created by <a href="#topic+create_tracelog">create_tracelog</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_tracelog">create_tracelog</a> to create a valid <code>tracelog</code>
</p>

<hr>
<h2 id='check_tracelog_values'>Check if the tracelog has the list elements with valid values
for being a valid tracelog object.</h2><span id='topic+check_tracelog_values'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if a value is invalid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_tracelog_values(tracelog)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_tracelog_values_+3A_tracelog">tracelog</code></td>
<td>
<p>a <code>tracelog</code>,
as created by <a href="#topic+create_tracelog">create_tracelog</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_tracelog">create_tracelog</a> to create a valid tracelog
</p>

<hr>
<h2 id='check_tree_prior'>Check if the tree prior is a valid tree prior</h2><span id='topic+check_tree_prior'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if the tree priors are invalid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_tree_prior(tree_prior)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_tree_prior_+3A_tree_prior">tree_prior</code></td>
<td>
<p>a tree priors,
as returned by <code><a href="#topic+create_tree_prior">create_tree_prior</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_tree_prior">create_tree_prior</a> to create a valid tree prior
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

check_tree_prior(create_yule_tree_prior())
check_tree_prior(create_bd_tree_prior())
check_tree_prior(create_cbs_tree_prior())
check_tree_prior(create_ccp_tree_prior())
check_tree_prior(create_cep_tree_prior())

# Can use list of one tree prior
check_tree_prior(list(create_yule_tree_prior()))

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_tree_priors'>Check if the object is a list of one or more tree priors.</h2><span id='topic+check_tree_priors'></span>

<h3>Description</h3>

<p>Will <a href="base.html#topic+stop">stop</a> if the object is not a list of one or more tree priors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_tree_priors(tree_priors)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_tree_priors_+3A_tree_priors">tree_priors</code></td>
<td>
<p>the object to be checked if it is a list of one
or more valid tree priors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing.
Will <a href="base.html#topic+stop">stop</a> if the object is not a list of one or more tree priors.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_tree_prior">create_tree_prior</a> to create a valid tree prior
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

check_tree_priors(create_yule_tree_prior())
check_tree_priors(list(create_yule_tree_prior()))
check_tree_priors(list(create_yule_tree_prior(), create_bd_tree_prior()))

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_treelog'>Check if a <code>treelog</code> is valid.</h2><span id='topic+check_treelog'></span>

<h3>Description</h3>

<p>Will call <a href="base.html#topic+stop">stop</a> if not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_treelog(treelog)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_treelog_+3A_treelog">treelog</code></td>
<td>
<p>a <code>treelog</code>,
as created by <a href="#topic+create_treelog">create_treelog</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

check_treelog(create_test_treelog())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='check_treelog_names'>Check if the <code>treelog</code> has the list elements
of a valid <code>treelog</code> object.</h2><span id='topic+check_treelog_names'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if an element is missing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_treelog_names(treelog)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_treelog_names_+3A_treelog">treelog</code></td>
<td>
<p>a <code>treelog</code>,
as created by <a href="#topic+create_treelog">create_treelog</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_treelog">create_treelog</a> to create a valid <code>treelog</code>
</p>

<hr>
<h2 id='check_treelog_values'>Check if the treelog has the list elements with valid values
for being a valid treelog object.</h2><span id='topic+check_treelog_values'></span>

<h3>Description</h3>

<p>Calls <code>stop</code> if a value is invalid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_treelog_values(treelog)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_treelog_values_+3A_treelog">treelog</code></td>
<td>
<p>a <code>treelog</code>,
as created by <a href="#topic+create_treelog">create_treelog</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_treelog">create_treelog</a> to create a valid treelog
</p>

<hr>
<h2 id='clock_model_to_xml_operators'>Internal function</h2><span id='topic+clock_model_to_xml_operators'></span>

<h3>Description</h3>

<p>Converts a clock model to the <code>operators</code> section of the
XML as text
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clock_model_to_xml_operators(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clock_model_to_xml_operators_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of XML strings
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='clock_model_to_xml_prior_distr'>Internal function</h2><span id='topic+clock_model_to_xml_prior_distr'></span>

<h3>Description</h3>

<p>Internal function to converts a clock model
to the <code>prior</code> section of the XML as text
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clock_model_to_xml_prior_distr(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clock_model_to_xml_prior_distr_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of XML strings
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()
 # &lt;distribution id="posterior" spec="util.CompoundDistribution"&gt;
 #     &lt;distribution id="prior" spec="util.CompoundDistribution"&gt;
 #       HERE, where the ID of the distribution is 'prior'
 #     &lt;/distribution&gt;
 #     &lt;distribution id="likelihood" ...&gt;
 #     &lt;/distribution&gt;
 # &lt;/distribution&gt;
clock_model_to_xml_prior_distr(
  inference_model = create_inference_model()
)
check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='clock_model_to_xml_state'>Internal function</h2><span id='topic+clock_model_to_xml_state'></span>

<h3>Description</h3>

<p>Converts a clock model to the <code>state</code> section of the
XML as text
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clock_model_to_xml_state(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clock_model_to_xml_state_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text, without indentation nor <code>state</code>
tags
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='clock_model_to_xml_tracelog'>Internal function</h2><span id='topic+clock_model_to_xml_tracelog'></span>

<h3>Description</h3>

<p>Creates the clock model's XML for the tracelog section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clock_model_to_xml_tracelog(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clock_model_to_xml_tracelog_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of XML strings
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# &lt;logger id="tracelog" ...&gt;
#'   # Here
# &lt;/logger&gt;

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='clock_model_to_xml_treelogger'>Convert a clock model to the XML of the <code>TreeLogger</code></h2><span id='topic+clock_model_to_xml_treelogger'></span>

<h3>Description</h3>

<p>Convert a clock model to the XML of the <code>TreeLogger</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clock_model_to_xml_treelogger(clock_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clock_model_to_xml_treelogger_+3A_clock_model">clock_model</code></td>
<td>
<p>a clock model,
as returned by <code><a href="#topic+create_clock_model">create_clock_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of XML strings
</p>


<h3>Note</h3>

<p>This is an internal function, so it should be marked with
<code>@export</code>. This is not done, as this will disallow all
functions to find the documentation parameters
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='clock_rate_param_to_xml'>Internal function</h2><span id='topic+clock_rate_param_to_xml'></span>

<h3>Description</h3>

<p>Converts a <code>clockRate</code> parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clock_rate_param_to_xml(
  clock_rate_param,
  beauti_options = create_beauti_options()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clock_rate_param_to_xml_+3A_clock_rate_param">clock_rate_param</code></td>
<td>
<p>a <code>clockRate</code> parameter,
a numeric value,
as created by <a href="#topic+create_clock_rate_param">create_clock_rate_param</a></p>
</td></tr>
<tr><td><code id="clock_rate_param_to_xml_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='compare_lines'>Internal function</h2><span id='topic+compare_lines'></span>

<h3>Description</h3>

<p>Internal debug function to compare the actually created
lines to expected lines using any diff tool
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compare_lines(
  lines,
  expected,
  section = NA,
  created_lines_filename = get_beautier_tempfilename(pattern = "created", fileext =
    ".xml"),
  expected_lines_filename = get_beautier_tempfilename(pattern = "expected", fileext =
    ".xml")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compare_lines_+3A_lines">lines</code></td>
<td>
<p>the created lines</p>
</td></tr>
<tr><td><code id="compare_lines_+3A_expected">expected</code></td>
<td>
<p>the expected/goal/target lines</p>
</td></tr>
<tr><td><code id="compare_lines_+3A_section">section</code></td>
<td>
<p>the XML section. Leave at NA to compare all lines</p>
</td></tr>
<tr><td><code id="compare_lines_+3A_created_lines_filename">created_lines_filename</code></td>
<td>
<p>name of the file where the (section of
the) created lines are stored</p>
</td></tr>
<tr><td><code id="compare_lines_+3A_expected_lines_filename">expected_lines_filename</code></td>
<td>
<p>name of the file where the (section of
the) expected lines are stored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing. Instead, two files are created, with the
names <code>created_lines_filename</code>
and <code>expected_lines_filename</code> that contain the
section under investigation, so that a diff tool
can compare these
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# Creates temporary files in beautier folder
compare_lines(
  lines = readLines(get_beautier_path("bd_2_4.xml")),
  expected = readLines(get_beautier_path("bd_2_4.xml"))
)

remove_beautier_folder()
check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='count_trailing_spaces'>Count the number of spaces before the first character</h2><span id='topic+count_trailing_spaces'></span>

<h3>Description</h3>

<p>Count the number of spaces before the first character
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count_trailing_spaces(line)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="count_trailing_spaces_+3A_line">line</code></td>
<td>
<p>line of text</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the number of spaces before the first character
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# 0
count_trailing_spaces("x")
# 1
count_trailing_spaces(" y")
# 2
count_trailing_spaces("  &lt;")
# 0
count_trailing_spaces("")
# 1
count_trailing_spaces(" ")
# 2
count_trailing_spaces("  ")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_alpha_param'>Create a parameter called alpha</h2><span id='topic+create_alpha_param'></span><span id='topic+create_param_alpha'></span>

<h3>Description</h3>

<p>Create a parameter called alpha
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_alpha_param(id = NA, value = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_alpha_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_alpha_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called alpha
</p>


<h3>Note</h3>

<p>this parameter is used in a beta distribution
(as returned by <code><a href="#topic+create_beta_distr">create_beta_distr</a></code>)
and gamma distribution
(as returned by <code><a href="#topic+create_gamma_distr">create_gamma_distr</a></code>)
and inverse-gamma distribution
(as returned by <code><a href="#topic+create_inv_gamma_distr">create_inv_gamma_distr</a></code>).
It cannot be estimated (as a hyper parameter) yet.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_param">create_param</a></code> contains a list
of all parameters that can be created
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  # Create the parameter
  alpha_param &lt;- create_alpha_param()

  # Use the parameter in a distribution
  beta_distr &lt;- create_beta_distr(
    alpha = alpha_param
  )

  # Use the distribution to create a BEAST2 input file
  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = create_yule_tree_prior(
      birth_rate_distr = beta_distr
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_b_pop_sizes_param'>Create a parameter called 'b_pop_sizes'.</h2><span id='topic+create_b_pop_sizes_param'></span><span id='topic+create_param_b_pop_sizes'></span>

<h3>Description</h3>

<p>Create a parameter called 'b_pop_sizes'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_b_pop_sizes_param(id = NA, value = 1, upper = "380000.0")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_b_pop_sizes_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_b_pop_sizes_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
<tr><td><code id="create_b_pop_sizes_param_+3A_upper">upper</code></td>
<td>
<p>upper value of the parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called b_pop_sizes
</p>


<h3>Note</h3>

<p>this parameter is used in a CBS model,
as created by <a href="#topic+create_cbs_tree_prior">create_cbs_tree_prior</a>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_param">create_param</a></code> contains a list
of all parameters that can be created
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create the parameter
b_pop_sizes_param &lt;- create_b_pop_sizes_param()
</code></pre>

<hr>
<h2 id='create_bd_tree_prior'>Create a Birth-Death tree prior</h2><span id='topic+create_bd_tree_prior'></span><span id='topic+create_tree_prior_bd'></span>

<h3>Description</h3>

<p>Create a Birth-Death tree prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_bd_tree_prior(
  id = NA,
  birth_rate_distr = create_uniform_distr(),
  death_rate_distr = create_uniform_distr()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_bd_tree_prior_+3A_id">id</code></td>
<td>
<p>the ID of the alignment</p>
</td></tr>
<tr><td><code id="create_bd_tree_prior_+3A_birth_rate_distr">birth_rate_distr</code></td>
<td>
<p>the birth rate distribution,
as created by a <code><a href="#topic+create_distr">create_distr</a></code> function</p>
</td></tr>
<tr><td><code id="create_bd_tree_prior_+3A_death_rate_distr">death_rate_distr</code></td>
<td>
<p>the death rate distribution,
as created by a <code><a href="#topic+create_distr">create_distr</a></code> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Birth-Death tree_prior
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>An alignment ID can be extracted from
its FASTA filename using <code><a href="#topic+get_alignment_id">get_alignment_id</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  bd_tree_prior &lt;- create_bd_tree_prior()

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = bd_tree_prior
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_beast2_beast_xml'>Create the <code>&lt;beast ...&gt;</code> XML</h2><span id='topic+create_beast2_beast_xml'></span>

<h3>Description</h3>

<p>The <code>&lt;beast ...&gt;</code> XML is the XML at the start of a BEAST2
XML input file, directly after the general XML declaration (as
created by <a href="#topic+create_xml_declaration">create_xml_declaration</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beast2_beast_xml(beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_beast2_beast_xml_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the XML
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>remove_beautier_folder()

create_beast2_beast_xml(
  beauti_options = create_beauti_options_v2_6()
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_beast2_input'>Create a BEAST2 XML input text</h2><span id='topic+create_beast2_input'></span>

<h3>Description</h3>

<p>Create a BEAST2 XML input text
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beast2_input(
  input_filename,
  tipdates_filename = NA,
  site_model = create_jc69_site_model(),
  clock_model = create_strict_clock_model(),
  tree_prior = create_yule_tree_prior(),
  mrca_prior = NA,
  mcmc = create_mcmc(),
  beauti_options = create_beauti_options()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_beast2_input_+3A_input_filename">input_filename</code></td>
<td>
<p>A FASTA filename.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.</p>
</td></tr>
<tr><td><code id="create_beast2_input_+3A_tipdates_filename">tipdates_filename</code></td>
<td>
<p>name of the file containing the tip dates.
This file is assumed to have two columns, separated by a tab.
The first column contains the taxa names, the second column contains
the date.</p>
</td></tr>
<tr><td><code id="create_beast2_input_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
<tr><td><code id="create_beast2_input_+3A_clock_model">clock_model</code></td>
<td>
<p>a clock model,
as returned by <code><a href="#topic+create_clock_model">create_clock_model</a></code></p>
</td></tr>
<tr><td><code id="create_beast2_input_+3A_tree_prior">tree_prior</code></td>
<td>
<p>a tree priors,
as returned by <code><a href="#topic+create_tree_prior">create_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="create_beast2_input_+3A_mrca_prior">mrca_prior</code></td>
<td>
<p>a Most Recent Common Ancestor prior,
as returned by <code><a href="#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
<tr><td><code id="create_beast2_input_+3A_mcmc">mcmc</code></td>
<td>
<p>one MCMC.
Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create an MCMC.
Use <code><a href="#topic+create_ns_mcmc">create_ns_mcmc</a></code> to create an MCMC
for a Nested Sampling run.
Use <code><a href="#topic+check_mcmc">check_mcmc</a></code> to check if an MCMC is valid.
Use <code><a href="#topic+rename_mcmc_filenames">rename_mcmc_filenames</a></code> to rename the filenames in an MCMC.</p>
</td></tr>
<tr><td><code id="create_beast2_input_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of XML strings
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_beast2_input_from_model">create_beast2_input_from_model</a> to create the BEAST2 XML
input text from an inference model
Use <a href="#topic+create_beast2_input_file">create_beast2_input_file</a> to also save it to file.
</p>
<p><code><a href="#topic+create_beast2_input_file">create_beast2_input_file</a></code> shows more examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {
  create_beast2_input(
    input_filename = get_fasta_filename()
  )
}
</code></pre>

<hr>
<h2 id='create_beast2_input_beast'>Creates the XML text for the <code>beast</code> tag of a BEAST2 parameter file.</h2><span id='topic+create_beast2_input_beast'></span>

<h3>Description</h3>

<p>Creates the XML text for the <code>beast</code> tag of a BEAST2 parameter file,
which is directly after the XML
declaration (created by <a href="#topic+create_xml_declaration">create_xml_declaration</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beast2_input_beast(
  input_filename,
  inference_model = create_inference_model()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_beast2_input_beast_+3A_input_filename">input_filename</code></td>
<td>
<p>A FASTA filename.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.</p>
</td></tr>
<tr><td><code id="create_beast2_input_beast_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>beast</code> tag has these elements:
</p>
<pre>
  &lt;beast[...]&gt;
      &lt;data
      [...]
      &lt;/data&gt;
      [map names]
      &lt;run[...]&gt;
      [...]
      &lt;/run&gt;
  &lt;/beast&gt;
</pre>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_beast2_input_from_model">create_beast2_input_from_model</a> to create the complete XML text.
Use <a href="#topic+create_beast2_input_data">create_beast2_input_data</a> to create the XML text for
the <code>data</code> tag only.
Use <a href="#topic+create_beast2_input_map">create_beast2_input_map</a> to create the XML text for
the <code>[map names]</code> part.
Use <a href="#topic+create_beast2_input_run">create_beast2_input_run</a> to create the XML text for
the <code>run</code> tag only.
</p>

<hr>
<h2 id='create_beast2_input_data'>Creates the <code>data</code> section of a BEAST2 XML parameter file</h2><span id='topic+create_beast2_input_data'></span>

<h3>Description</h3>

<p>Creates the <code>data</code> section of a BEAST2 XML parameter file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beast2_input_data(
  input_filename,
  beauti_options = create_beauti_options()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_beast2_input_data_+3A_input_filename">input_filename</code></td>
<td>
<p>A FASTA filename.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.</p>
</td></tr>
<tr><td><code id="create_beast2_input_data_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

create_beast2_input_data(
  input_filename = get_fasta_filename(),
  beauti_options = create_beauti_options_v2_4()
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_beast2_input_data_sequences'>Creates the data section of a BEAST2 XML parameter file</h2><span id='topic+create_beast2_input_data_sequences'></span>

<h3>Description</h3>

<p>Creates the data section of a BEAST2 XML parameter file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beast2_input_data_sequences(
  input_fasta_filename,
  beauti_options = create_beauti_options()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_beast2_input_data_sequences_+3A_input_fasta_filename">input_fasta_filename</code></td>
<td>
<p>one FASTA filename</p>
</td></tr>
<tr><td><code id="create_beast2_input_data_sequences_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='create_beast2_input_distr'>Creates the distribution section of a BEAST2 XML parameter file.</h2><span id='topic+create_beast2_input_distr'></span>

<h3>Description</h3>

<p>Creates the distribution section of a BEAST2 XML parameter file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beast2_input_distr(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_beast2_input_distr_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Note</h3>

<p>this function is not intended for regular use, thus its
long name length is accepted
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_beast2_input">create_beast2_input</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

inference_model &lt;- init_inference_model(
  input_filename = get_fasta_filename(),
  inference_model = create_inference_model(
    beauti_options = create_beauti_options_v2_4()
  )
)
create_beast2_input_distr(
  inference_model = inference_model
)
 # &lt;distribution id="posterior" spec="util.CompoundDistribution"&gt;
 #     &lt;distribution id="prior" spec="util.CompoundDistribution"&gt;
 #       HERE, where the ID of the distribution is 'prior'
 #     &lt;/distribution&gt;
 #     &lt;distribution id="likelihood" ...&gt;
 #     &lt;/distribution&gt;
 # &lt;/distribution&gt;

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_beast2_input_distr_lh'>Creates the XML text for the <code>distribution</code> tag
with the <code>likelihood</code> ID,
of a BEAST2 parameter file.</h2><span id='topic+create_beast2_input_distr_lh'></span>

<h3>Description</h3>

<p>Creates the XML text for the <code>distribution</code> tag
with the <code>likelihood</code> ID,
of a BEAST2 parameter file,
in an unindented form
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beast2_input_distr_lh(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_beast2_input_distr_lh_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>distribution</code> tag (with ID equals <code>likelihood</code>)
has these elements:
</p>
<pre>
  &lt;distribution id="likelihood"[...]&gt;
     &lt;distribution id="treeLikelihood"[...]&gt;
       [...]
     &lt;/distribution&gt;
  &lt;/distribution&gt;
</pre>
<p>The <code>distribution</code> section with ID <code>treeLikelihood</code>
is created by <a href="#topic+create_tree_likelihood_distr_xml">create_tree_likelihood_distr_xml</a>.
</p>
<p>Zooming out:
</p>
<pre>
  &lt;beast[...]&gt;
    &lt;run[...]&gt;
      &lt;distribution id="posterior"[...]&gt;
        &lt;distribution id="likelihood"[...]&gt;
          [this section]
        &lt;/distribution&gt;
      &lt;/distribution&gt;
    &lt;/run&gt;
  &lt;/beast&gt;
</pre>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Note</h3>

<p>this function is not intended for regular use, thus its
long name length is accepted
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>this function is called by <code>create_beast2_input_distr</code>,
together with <code>create_beast2_input_distr_prior</code>
</p>

<hr>
<h2 id='create_beast2_input_distr_prior'>Creates the prior section in the distribution section
of a BEAST2 XML parameter file</h2><span id='topic+create_beast2_input_distr_prior'></span>

<h3>Description</h3>

<p>Creates the prior section in the distribution section
of a BEAST2 XML parameter file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beast2_input_distr_prior(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_beast2_input_distr_prior_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Note</h3>

<p>this function is not intended for regular use, thus its
long name length is accepted
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>this function is called by <code>create_beast2_input_distr</code>,
together with <code>create_beast2_input_distr_lh</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

 # &lt;distribution id="posterior" spec="util.CompoundDistribution"&gt;
 #     &lt;distribution id="prior" spec="util.CompoundDistribution"&gt;
 #       HERE, where the ID of the distribution is 'prior'
 #     &lt;/distribution&gt;
 #     &lt;distribution id="likelihood" ...&gt;
 #     &lt;/distribution&gt;
 # &lt;/distribution&gt;

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_beast2_input_file'>Create a BEAST2 input file</h2><span id='topic+create_beast2_input_file'></span>

<h3>Description</h3>

<p>Create a BEAST2 input file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beast2_input_file(
  input_filename,
  output_filename,
  site_model = create_jc69_site_model(),
  clock_model = create_strict_clock_model(),
  tree_prior = create_yule_tree_prior(),
  mrca_prior = NA,
  mcmc = create_mcmc(),
  beauti_options = create_beauti_options(),
  tipdates_filename = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_beast2_input_file_+3A_input_filename">input_filename</code></td>
<td>
<p>A FASTA filename.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.</p>
</td></tr>
<tr><td><code id="create_beast2_input_file_+3A_output_filename">output_filename</code></td>
<td>
<p>Name of the XML parameter file created by this
function. BEAST2 uses this file as input.</p>
</td></tr>
<tr><td><code id="create_beast2_input_file_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
<tr><td><code id="create_beast2_input_file_+3A_clock_model">clock_model</code></td>
<td>
<p>a clock model,
as returned by <code><a href="#topic+create_clock_model">create_clock_model</a></code></p>
</td></tr>
<tr><td><code id="create_beast2_input_file_+3A_tree_prior">tree_prior</code></td>
<td>
<p>a tree priors,
as returned by <code><a href="#topic+create_tree_prior">create_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="create_beast2_input_file_+3A_mrca_prior">mrca_prior</code></td>
<td>
<p>a Most Recent Common Ancestor prior,
as returned by <code><a href="#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
<tr><td><code id="create_beast2_input_file_+3A_mcmc">mcmc</code></td>
<td>
<p>one MCMC.
Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create an MCMC.
Use <code><a href="#topic+create_ns_mcmc">create_ns_mcmc</a></code> to create an MCMC
for a Nested Sampling run.
Use <code><a href="#topic+check_mcmc">check_mcmc</a></code> to check if an MCMC is valid.
Use <code><a href="#topic+rename_mcmc_filenames">rename_mcmc_filenames</a></code> to rename the filenames in an MCMC.</p>
</td></tr>
<tr><td><code id="create_beast2_input_file_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
<tr><td><code id="create_beast2_input_file_+3A_tipdates_filename">tipdates_filename</code></td>
<td>
<p>name of the file containing the tip dates.
This file is assumed to have two columns, separated by a tab.
The first column contains the taxa names, the second column contains
the date.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_beast2_input_file_from_model">create_beast2_input_file_from_model</a> to do the same with an
inference model.
See <code><a href="#topic+create_site_model">create_site_model</a></code> for examples with
different site models. See <code><a href="#topic+create_clock_model">create_clock_model</a></code> for examples
with clock models. See <code><a href="#topic+create_tree_prior">create_tree_prior</a></code> for examples with
different tree priors. See <code><a href="#topic+create_mcmc">create_mcmc</a></code> for examples with
a different MCMC setup.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  check_empty_beautier_folder()

  # Get an example FASTA file
  input_filename &lt;- get_fasta_filename()

  # The file created by beautier, a BEAST2 input file
  output_filename &lt;- get_beautier_tempfilename()

  create_beast2_input_file(
    input_filename,
    output_filename
  )
  file.remove(output_filename)

  remove_beautier_folder()
  check_empty_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_beast2_input_file_from_model'>Create a BEAST2 input file from an inference model</h2><span id='topic+create_beast2_input_file_from_model'></span>

<h3>Description</h3>

<p>Create a BEAST2 input file from an inference model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beast2_input_file_from_model(
  input_filename,
  output_filename,
  inference_model = create_inference_model()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_beast2_input_file_from_model_+3A_input_filename">input_filename</code></td>
<td>
<p>A FASTA filename.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.</p>
</td></tr>
<tr><td><code id="create_beast2_input_file_from_model_+3A_output_filename">output_filename</code></td>
<td>
<p>Name of the XML parameter file created by this
function. BEAST2 uses this file as input.</p>
</td></tr>
<tr><td><code id="create_beast2_input_file_from_model_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <a href="#topic+create_beast2_input_from_model">create_beast2_input_from_model</a> to
get the BEAST2 input file as text
</p>
<p>See <code><a href="#topic+create_site_model">create_site_model</a></code> for examples with
different site models.
See <code><a href="#topic+create_clock_model">create_clock_model</a></code> for examples
with clock models.
See <code><a href="#topic+create_tree_prior">create_tree_prior</a></code> for examples with
different tree priors.
See <code><a href="#topic+create_mcmc">create_mcmc</a></code> for examples with
a different MCMC setup.
Use <a href="#topic+create_beast2_input_file">create_beast2_input_file</a> to do the same with the elements
of an inference model.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {
  check_empty_beautier_folder()

  output_filename &lt;- get_beautier_tempfilename()
  create_beast2_input_file_from_model(
    input_filename = get_fasta_filename(),
    output_filename = output_filename,
    inference_model = create_inference_model()
  )
  file.remove(output_filename)

  remove_beautier_folder()
  check_empty_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_beast2_input_from_model'>Create a BEAST2 XML input text from an inference model</h2><span id='topic+create_beast2_input_from_model'></span>

<h3>Description</h3>

<p>The main two XML tags are these:
</p>
<pre>
  &lt;?xml[...]?&gt;&lt;beast[...]&gt;
  [...]
  &lt;/beast&gt;
</pre>


<h3>Usage</h3>

<pre><code class='language-R'>create_beast2_input_from_model(input_filename, inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_beast2_input_from_model_+3A_input_filename">input_filename</code></td>
<td>
<p>A FASTA filename.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.</p>
</td></tr>
<tr><td><code id="create_beast2_input_from_model_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of XML strings
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_beast2_input_file_from_model">create_beast2_input_file_from_model</a> to also save it to file.
Use <a href="#topic+create_xml_declaration">create_xml_declaration</a>
to create the XML text of the XML declaration.
Use <a href="#topic+create_beast2_input_beast">create_beast2_input_beast</a> to create
to create the XML text of the <code>beast</code> tag.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  check_empty_beautier_folder()

  text &lt;- create_beast2_input_from_model(
    input_filename = get_fasta_filename(),
    inference_model = create_inference_model()
  )

  check_empty_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_beast2_input_init'>Creates the <code>init</code> section of a BEAST2 XML parameter file</h2><span id='topic+create_beast2_input_init'></span>

<h3>Description</h3>

<p>Creates the <code>init</code> section of a BEAST2 XML parameter file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beast2_input_init(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_beast2_input_init_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

inference_model &lt;- init_inference_model(
  input_filename = get_fasta_filename(),
  inference_model = create_test_inference_model()
)
xml &lt;- create_beast2_input_init(
  inference_model = inference_model
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_beast2_input_map'>Creates the map section of a BEAST2 XML parameter file</h2><span id='topic+create_beast2_input_map'></span>

<h3>Description</h3>

<p>Creates the map section of a BEAST2 XML parameter file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beast2_input_map(beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_beast2_input_map_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='create_beast2_input_operators'>Creates the operators section of a BEAST2 XML parameter file</h2><span id='topic+create_beast2_input_operators'></span>

<h3>Description</h3>

<p>Creates the operators section of a BEAST2 XML parameter file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beast2_input_operators(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_beast2_input_operators_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='create_beast2_input_run'>Creates the '<code>run</code>' section of a BEAST2 XML parameter file</h2><span id='topic+create_beast2_input_run'></span>

<h3>Description</h3>

<p>Creates the '<code>run</code>' section of a BEAST2 XML parameter file,
without being indented.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beast2_input_run(
  input_filename,
  inference_model = create_inference_model()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_beast2_input_run_+3A_input_filename">input_filename</code></td>
<td>
<p>A FASTA filename.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.</p>
</td></tr>
<tr><td><code id="create_beast2_input_run_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>run</code> tag has these elements:
</p>
<pre>
   &lt;run[...]&gt;
       &lt;state[...]&gt;
       [...]
       &lt;/state&gt;
       &lt;init[...]&gt;
       [...]
       &lt;/init&gt;
       &lt;distribution[...]&gt;
       [...]
       &lt;/distribution&gt;
       [operator ids]
       [loggers]
    &lt;/run&gt;
</pre>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_beast2_input_state">create_beast2_input_state</a>
to create the XML text of the <code>state</code> tag.
Use <a href="#topic+create_beast2_input_init">create_beast2_input_init</a>
to create the XML text of the <code>init</code> tag.
Use <a href="#topic+create_beast2_input_distr">create_beast2_input_distr</a>
to create the XML text of the <code>distribution</code> tag.
Use <a href="#topic+create_beast2_input_operators">create_beast2_input_operators</a>
to create the XML text of the <code>[operator ids]</code> section.
Use <a href="#topic+create_loggers_xml">create_loggers_xml</a>
to create the XML text of the <code>[loggers]</code> part.
</p>

<hr>
<h2 id='create_beast2_input_state'>Creates the '<code>state</code>' section of a BEAST2 XML parameter file</h2><span id='topic+create_beast2_input_state'></span>

<h3>Description</h3>

<p>Creates the '<code>state</code>' section of a BEAST2 XML parameter file,
without being indented.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beast2_input_state(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_beast2_input_state_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>state</code> tag has these elements:
</p>
<pre>
   &lt;state[...]&gt;
       &lt;tree[...]&gt;
       [...]
       &lt;/tree&gt;
       [parameters]
    &lt;/run&gt;
</pre>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_beast2_input_state">create_beast2_input_state</a>
to create the XML text of the <code>tree</code> tag.
to create the XML text of the <code>[parameters]</code> section.
</p>

<hr>
<h2 id='create_beauti_options'>Function to create a set of 'BEAUti' options.</h2><span id='topic+create_beauti_options'></span>

<h3>Description</h3>

<p>'BEAUti' options are settings that differ between 'BEAUti'
version. The use of these options is mostly for testing
older versions
Whatever option chosen here, the created XML file will be valid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beauti_options(
  capitalize_first_char_id = FALSE,
  nucleotides_uppercase = FALSE,
  beast2_version = "2.4",
  required = "",
  sequence_indent = 20,
  status = "",
  namespace = get_default_beast_namespace_v2_4()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_beauti_options_+3A_capitalize_first_char_id">capitalize_first_char_id</code></td>
<td>
<p>must the ID of alignment start with a
capital? TRUE if yes, FALSE if it can be left lower case (if it is
lowercase)</p>
</td></tr>
<tr><td><code id="create_beauti_options_+3A_nucleotides_uppercase">nucleotides_uppercase</code></td>
<td>
<p>must the nucleotides of the DNA sequence be
in uppercase?</p>
</td></tr>
<tr><td><code id="create_beauti_options_+3A_beast2_version">beast2_version</code></td>
<td>
<p>the BEAST2 version</p>
</td></tr>
<tr><td><code id="create_beauti_options_+3A_required">required</code></td>
<td>
<p>things that may be required,
for example <code>BEAST v2.5.0</code></p>
</td></tr>
<tr><td><code id="create_beauti_options_+3A_sequence_indent">sequence_indent</code></td>
<td>
<p>the number of spaces the XML <code>sequence</code>
lines are indented</p>
</td></tr>
<tr><td><code id="create_beauti_options_+3A_status">status</code></td>
<td>
<p>the BEAUti status</p>
</td></tr>
<tr><td><code id="create_beauti_options_+3A_namespace">namespace</code></td>
<td>
<p>the 'namespace' XML element in the 'beast' XML tag.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Available BEAUti options are:<br />
* <a href="#topic+create_beauti_options_v2_4">create_beauti_options_v2_4</a>
* <a href="#topic+create_beauti_options_v2_6">create_beauti_options_v2_6</a>
</p>
<p>'beautier' uses v2.4 by default, as this is when the first tests
were written.
</p>


<h3>Value</h3>

<p>a BEAUti options structure
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  check_empty_beautier_folder()

  beauti_options &lt;- create_beauti_options_v2_4()
  xml &lt;- create_beast2_input(
    get_fasta_filename(),
    beauti_options = beauti_options
  )

  check_empty_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_beauti_options_v2_4'>Function to create the BEAUti options for version 2.4.</h2><span id='topic+create_beauti_options_v2_4'></span>

<h3>Description</h3>

<p>Function to create the BEAUti options for version 2.4, by
calling <a href="#topic+create_beauti_options">create_beauti_options</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beauti_options_v2_4()
</code></pre>


<h3>Value</h3>

<p>a BEAUti options structure
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {
  check_empty_beautier_folder()

  beauti_options &lt;- create_beauti_options_v2_4()
  xml &lt;- create_beast2_input(
    get_fasta_filename(),
    beauti_options = beauti_options
  )

  check_empty_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_beauti_options_v2_6'>Function to create the BEAUti options for version 2.6.</h2><span id='topic+create_beauti_options_v2_6'></span>

<h3>Description</h3>

<p>Function to create the BEAUti options for version 2.6, by
calling <a href="#topic+create_beauti_options">create_beauti_options</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beauti_options_v2_6(
  beast2_version = "2.6",
  sequence_indent = 8,
  nucleotides_uppercase = FALSE,
  status = "",
  namespace = get_default_beast_namespace_v2_6(),
  required = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_beauti_options_v2_6_+3A_beast2_version">beast2_version</code></td>
<td>
<p>the BEAST2 version</p>
</td></tr>
<tr><td><code id="create_beauti_options_v2_6_+3A_sequence_indent">sequence_indent</code></td>
<td>
<p>the number of spaces the XML <code>sequence</code>
lines are indented</p>
</td></tr>
<tr><td><code id="create_beauti_options_v2_6_+3A_nucleotides_uppercase">nucleotides_uppercase</code></td>
<td>
<p>must the nucleotides of the DNA sequence be
in uppercase?</p>
</td></tr>
<tr><td><code id="create_beauti_options_v2_6_+3A_status">status</code></td>
<td>
<p>the BEAUti status</p>
</td></tr>
<tr><td><code id="create_beauti_options_v2_6_+3A_namespace">namespace</code></td>
<td>
<p>the 'namespace' XML element in the 'beast' XML tag.</p>
</td></tr>
<tr><td><code id="create_beauti_options_v2_6_+3A_required">required</code></td>
<td>
<p>things that may be required,
for example <code>BEAST v2.5.0</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a BEAUti options structure
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>see <a href="#topic+create_beauti_options_v2_4">create_beauti_options_v2_4</a> for using the older v2.4
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  check_empty_beautier_folder()

  beauti_options &lt;- create_beauti_options_v2_6()
  xml &lt;- create_beast2_input(
    get_fasta_filename(),
    beauti_options = beauti_options
  )

  check_empty_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_beautier_tempfolder'>Create the default 'beautier' temporary folder</h2><span id='topic+create_beautier_tempfolder'></span>

<h3>Description</h3>

<p>Create the default 'beautier' temporary folder
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beautier_tempfolder()
</code></pre>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>create_beautier_tempfolder()

remove_beautier_folder()

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_beta_distr'>Create a beta distribution</h2><span id='topic+create_beta_distr'></span><span id='topic+create_distr_beta'></span>

<h3>Description</h3>

<p>Create a beta distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beta_distr(
  id = NA,
  alpha = 0,
  beta = 1,
  value = NA,
  lower = NA,
  upper = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_beta_distr_+3A_id">id</code></td>
<td>
<p>the distribution's ID</p>
</td></tr>
<tr><td><code id="create_beta_distr_+3A_alpha">alpha</code></td>
<td>
<p>the alpha shape parameter,
a numeric value.
The value
of alpha must be at least 0.0.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_alpha_param">create_alpha_param</a></code>.</p>
</td></tr>
<tr><td><code id="create_beta_distr_+3A_beta">beta</code></td>
<td>
<p>the beta shape parameter,
a numeric value.
The value
of beta must be at least 1.0.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_beta_param">create_beta_param</a></code>.</p>
</td></tr>
<tr><td><code id="create_beta_distr_+3A_value">value</code></td>
<td>
<p>the initial value for the MCMC</p>
</td></tr>
<tr><td><code id="create_beta_distr_+3A_lower">lower</code></td>
<td>
<p>the lower bound, the lowest possible value</p>
</td></tr>
<tr><td><code id="create_beta_distr_+3A_upper">upper</code></td>
<td>
<p>an upper limit of the uniform distribution.
If the upper limits needs to be infinity, set <code>upper</code> to <code>Inf</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a beta distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_distr">create_distr</a></code> shows an overview
of all supported distributions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  beta_distr &lt;- create_beta_distr()

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = create_yule_tree_prior(
      birth_rate_distr = beta_distr
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_beta_param'>Create a parameter called beta</h2><span id='topic+create_beta_param'></span><span id='topic+create_param_beta'></span>

<h3>Description</h3>

<p>Create a parameter called beta
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_beta_param(id = NA, value = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_beta_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_beta_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called beta
</p>


<h3>Note</h3>

<p>this parameter is used in a beta distribution
(as returned by <code><a href="#topic+create_beta_distr">create_beta_distr</a></code>)
and gamma distribution
(as returned by <code><a href="#topic+create_gamma_distr">create_gamma_distr</a></code>)
and inverse-gamma distribution
(as returned by <code><a href="#topic+create_inv_gamma_distr">create_inv_gamma_distr</a></code>).
It cannot be estimated (as a hyper parameter) yet.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_param">create_param</a></code> contains a list
of all parameters that can be created
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  # Create the parameter
  beta_param &lt;- create_beta_param()

  # Use the parameter in a distribution
  gamma_distr &lt;- create_gamma_distr(
    beta = beta_param
  )

  # Use the distribution to create a BEAST2 input file
  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = create_yule_tree_prior(
      birth_rate_distr = gamma_distr
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_branch_rate_model_xml'>Internal function to create the <code>branchRateModel</code> section
of the XML as text.</h2><span id='topic+create_branch_rate_model_xml'></span>

<h3>Description</h3>

<p>Creates the <code>branchRateModel</code> section
of the XML as text.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_branch_rate_model_xml(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_branch_rate_model_xml_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>distribution</code> tag (with ID equals <code>treeLikelihood</code>)
has these elements:
</p>
<pre>
  &lt;branchRateModel[...]&gt;
    [...]
  &lt;/branchRateModel&gt;
</pre>
<p>When there is a strict clock,
<a href="#topic+create_strict_clock_branch_rate_model_xml">create_strict_clock_branch_rate_model_xml</a> is called.
When there is an RLN clock,
<a href="#topic+create_rln_clock_branch_rate_model_xml">create_rln_clock_branch_rate_model_xml</a> is called.
</p>
<p>Zooming out:
</p>
<pre>
  &lt;beast[...]&gt;
    &lt;run[...]&gt;
      &lt;distribution id="posterior"[...]&gt;
        &lt;distribution id="likelihood"[...]&gt;
          &lt;distribution id="treeLikelihood"[...]&gt;
             [...]

             [this section]
          &lt;/distribution&gt;
        &lt;/distribution&gt;
      &lt;/distribution&gt;
    &lt;/run&gt;
  &lt;/beast&gt;
</pre>


<h3>Value</h3>

<p>a character vector of XML strings
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='create_cbs_tree_prior'>Create a Coalescent Bayesian Skyline tree prior</h2><span id='topic+create_cbs_tree_prior'></span><span id='topic+create_tree_prior_cbs'></span>

<h3>Description</h3>

<p>Create a Coalescent Bayesian Skyline tree prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_cbs_tree_prior(
  id = NA,
  group_sizes_dimension = 5,
  b_pop_sizes_param = create_b_pop_sizes_param(),
  pop_sizes_scaler_scale_factor = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_cbs_tree_prior_+3A_id">id</code></td>
<td>
<p>an alignment's IDs.
An ID can be extracted from its FASTA filename
with <code><a href="#topic+get_alignment_ids_from_fasta_filenames">get_alignment_ids_from_fasta_filenames</a></code>)</p>
</td></tr>
<tr><td><code id="create_cbs_tree_prior_+3A_group_sizes_dimension">group_sizes_dimension</code></td>
<td>
<p>the group sizes' dimension,
as used by the CBS tree prior (see <code><a href="#topic+create_cbs_tree_prior">create_cbs_tree_prior</a></code>)</p>
</td></tr>
<tr><td><code id="create_cbs_tree_prior_+3A_b_pop_sizes_param">b_pop_sizes_param</code></td>
<td>
<p>a Bayesian population size parameter,
as created by <a href="#topic+create_b_pop_sizes_param">create_b_pop_sizes_param</a></p>
</td></tr>
<tr><td><code id="create_cbs_tree_prior_+3A_pop_sizes_scaler_scale_factor">pop_sizes_scaler_scale_factor</code></td>
<td>
<p>the scale factor used by the
population sizes scaler operator</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Coalescent Bayesian Skyline tree_prior
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>An alignment ID can be extracted from
its FASTA filename using <code><a href="#topic+get_alignment_id">get_alignment_id</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  cbs_tree_prior &lt;- create_cbs_tree_prior()

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_beautier_path("test_output_6.fas"),
    beast2_input_file,
    tree_prior = cbs_tree_prior
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_ccp_tree_prior'>Create a Coalescent Constant Population tree prior</h2><span id='topic+create_ccp_tree_prior'></span><span id='topic+create_tree_prior_ccp'></span>

<h3>Description</h3>

<p>Create a Coalescent Constant Population tree prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_ccp_tree_prior(
  id = NA,
  pop_size_distr = create_one_div_x_distr(value = 0.3)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_ccp_tree_prior_+3A_id">id</code></td>
<td>
<p>the ID of the alignment</p>
</td></tr>
<tr><td><code id="create_ccp_tree_prior_+3A_pop_size_distr">pop_size_distr</code></td>
<td>
<p>the population distribution,
as created by a <code><a href="#topic+create_distr">create_distr</a></code> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Coalescent Constant Population tree_prior
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>An alignment ID can be extracted from
its FASTA filename using <code><a href="#topic+get_alignment_id">get_alignment_id</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  ccp_tree_prior &lt;- create_ccp_tree_prior()

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = ccp_tree_prior
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_cep_tree_prior'>Create a Coalescent Exponential Population tree prior</h2><span id='topic+create_cep_tree_prior'></span><span id='topic+create_tree_prior_cep'></span>

<h3>Description</h3>

<p>Create a Coalescent Exponential Population tree prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_cep_tree_prior(
  id = NA,
  pop_size_distr = create_one_div_x_distr(),
  growth_rate_distr = create_laplace_distr()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_cep_tree_prior_+3A_id">id</code></td>
<td>
<p>the ID of the alignment</p>
</td></tr>
<tr><td><code id="create_cep_tree_prior_+3A_pop_size_distr">pop_size_distr</code></td>
<td>
<p>the population distribution,
as created by a <code><a href="#topic+create_distr">create_distr</a></code> function</p>
</td></tr>
<tr><td><code id="create_cep_tree_prior_+3A_growth_rate_distr">growth_rate_distr</code></td>
<td>
<p>the growth rate distribution,
as created by a <code><a href="#topic+create_distr">create_distr</a></code> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Coalescent Exponential Population tree_prior
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>An alignment ID can be extracted from
its FASTA filename using <code><a href="#topic+get_alignment_id">get_alignment_id</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  cep_tree_prior &lt;- create_cep_tree_prior()

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = cep_tree_prior
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_clock_model'>General function to create a clock model</h2><span id='topic+create_clock_model'></span>

<h3>Description</h3>

<p>General function to create a clock model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_clock_model(name, id, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_clock_model_+3A_name">name</code></td>
<td>
<p>the clock model name. Valid
names can be found in <code>get_clock_model_names</code></p>
</td></tr>
<tr><td><code id="create_clock_model_+3A_id">id</code></td>
<td>
<p>a clock model's ID</p>
</td></tr>
<tr><td><code id="create_clock_model_+3A_...">...</code></td>
<td>
<p>specific clock model parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a valid clock model
</p>


<h3>Note</h3>

<p>Prefer using the named function
<code><a href="#topic+create_rln_clock_model">create_rln_clock_model</a></code>
and <code><a href="#topic+create_strict_clock_model">create_strict_clock_model</a></code>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>An alignment ID can be extracted from
its FASTA filename using <code><a href="#topic+get_alignment_id">get_alignment_id</a></code>.
For more examples about creating a relaxed log-normal clock
model, see <code><a href="#topic+create_rln_clock_model">create_rln_clock_model</a></code>.
For more examples about creating a strict clock
model, see <code><a href="#topic+create_strict_clock_model">create_strict_clock_model</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {
  # Can use any of these models
  strict_clock_model &lt;- create_strict_clock_model()
  rln_clock_model &lt;- create_rln_clock_model()

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    get_fasta_filename(),
    beast2_input_file,
    clock_model = strict_clock_model
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_clock_model_from_name'>Create a clock model from name</h2><span id='topic+create_clock_model_from_name'></span>

<h3>Description</h3>

<p>Create a clock model from name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_clock_model_from_name(clock_model_name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_clock_model_from_name_+3A_clock_model_name">clock_model_name</code></td>
<td>
<p>name of a clock model,
must be a name as returned by <code><a href="#topic+get_clock_model_names">get_clock_model_names</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a clock model,
as can be created by <a href="#topic+create_clock_model">create_clock_model</a>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+create_clock_model">create_clock_model</a></code> to create a clock model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

create_clock_model_from_name(get_clock_model_names()[1])

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_clock_models'>Creates all supported clock models,
which is a list of the types returned by
<code><a href="#topic+create_rln_clock_model">create_rln_clock_model</a></code>,
and <code><a href="#topic+create_strict_clock_model">create_strict_clock_model</a></code></h2><span id='topic+create_clock_models'></span>

<h3>Description</h3>

<p>Creates all supported clock models,
which is a list of the types returned by
<code><a href="#topic+create_rln_clock_model">create_rln_clock_model</a></code>,
and <code><a href="#topic+create_strict_clock_model">create_strict_clock_model</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_clock_models()
</code></pre>


<h3>Value</h3>

<p>a list of site_models
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_clock_model">create_clock_model</a> to create a clock model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

clock_models &lt;- create_clock_models()
is_rln_clock_model(clock_models[[1]])
is_strict_clock_model(clock_models[[2]])

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_clock_models_from_names'>Create clock models from their names</h2><span id='topic+create_clock_models_from_names'></span>

<h3>Description</h3>

<p>Create clock models from their names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_clock_models_from_names(clock_model_names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_clock_models_from_names_+3A_clock_model_names">clock_model_names</code></td>
<td>
<p>one or more names of a clock model,
must be name among those returned by <code><a href="#topic+get_clock_model_names">get_clock_model_names</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of one or more clock models
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_clock_models">create_clock_models</a> to get all clock models
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

create_clock_models_from_names(get_clock_model_names())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_clock_rate_param'>Create a parameter called <code>clock_rate</code>,
as needed by <code><a href="#topic+create_strict_clock_model">create_strict_clock_model</a></code></h2><span id='topic+create_clock_rate_param'></span><span id='topic+create_param_clock_rate'></span>

<h3>Description</h3>

<p>Create a parameter called <code>clock_rate</code>,
as needed by <code><a href="#topic+create_strict_clock_model">create_strict_clock_model</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_clock_rate_param(value = "1.0", estimate = FALSE, id = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_clock_rate_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
<tr><td><code id="create_clock_rate_param_+3A_estimate">estimate</code></td>
<td>
<p>TRUE if this parameter is to be estimated by BEAST2,
FALSE otherwise</p>
</td></tr>
<tr><td><code id="create_clock_rate_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called rate
</p>


<h3>Note</h3>

<p>It cannot be estimated (as a hyper parameter) yet.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_param">create_param</a></code> contains a list
of all parameters that can be created
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  clock_rate_param &lt;- create_clock_rate_param(
    id = "anthus_aco", value = 1.0
  )

  # Use the parameter in a clock model
  strict_clock_model &lt;- create_strict_clock_model(
    clock_rate_param = clock_rate_param
  )

  # Use the distribution to create a BEAST2 input file
  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    clock_model = strict_clock_model
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_clock_rate_state_node_parameter_xml'>Internal function</h2><span id='topic+create_clock_rate_state_node_parameter_xml'></span>

<h3>Description</h3>

<p>Creates the <code>clockRate</code> parameter with the name <code>stateNode</code>,
such as:
<code>
  &lt;parameter id="ucldStdev.c:[id]" [...] name="stateNode"&gt;0.1&lt;/parameter&gt;
</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_clock_rate_state_node_parameter_xml(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_clock_rate_state_node_parameter_xml_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the following XML:
<code>
  &lt;parameter id="ucldStdev.c:[id]" lower="0.0" name="stateNode"&gt;
    0.1
  &lt;/parameter&gt;
</code>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

create_ucld_stdev_state_node_param_xml(
  create_inference_model(
    clock_model = create_rln_clock_model(id = 314)
  )
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_data_xml'>Create the <code>&lt;data ..&gt;</code> XML</h2><span id='topic+create_data_xml'></span>

<h3>Description</h3>

<p>Create the <code>&lt;data ..&gt;</code> XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_data_xml(id, beast2_version)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_data_xml_+3A_id">id</code></td>
<td>
<p>an alignment's IDs.
An ID can be extracted from its FASTA filename
with <code><a href="#topic+get_alignment_ids_from_fasta_filenames">get_alignment_ids_from_fasta_filenames</a></code>)</p>
</td></tr>
<tr><td><code id="create_data_xml_+3A_beast2_version">beast2_version</code></td>
<td>
<p>BEAST2 version, for example, <code>"2.5"</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='create_distr'>General function to create a distribution.</h2><span id='topic+create_distr'></span>

<h3>Description</h3>

<p>General function to create a distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_distr(name, id, value = NA, lower = NA, upper = NA, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_distr_+3A_name">name</code></td>
<td>
<p>the distribution name. Valid
names can be found in <code>get_distr_names</code></p>
</td></tr>
<tr><td><code id="create_distr_+3A_id">id</code></td>
<td>
<p>the distribution's ID</p>
</td></tr>
<tr><td><code id="create_distr_+3A_value">value</code></td>
<td>
<p>the initial value for the MCMC</p>
</td></tr>
<tr><td><code id="create_distr_+3A_lower">lower</code></td>
<td>
<p>the lower bound, the lowest possible value</p>
</td></tr>
<tr><td><code id="create_distr_+3A_upper">upper</code></td>
<td>
<p>an upper limit of the uniform distribution.
If the upper limits needs to be infinity, set <code>upper</code> to <code>Inf</code>.</p>
</td></tr>
<tr><td><code id="create_distr_+3A_...">...</code></td>
<td>
<p>specific distribution parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a distribution
</p>


<h3>Note</h3>

<p>Prefer using the
named functions
<code><a href="#topic+create_beta_distr">create_beta_distr</a></code>,
<code><a href="#topic+create_exp_distr">create_exp_distr</a></code>,
<code><a href="#topic+create_gamma_distr">create_gamma_distr</a></code>,
<code><a href="#topic+create_inv_gamma_distr">create_inv_gamma_distr</a></code>,
<code><a href="#topic+create_laplace_distr">create_laplace_distr</a></code>,
<code><a href="#topic+create_log_normal_distr">create_log_normal_distr</a></code>,
<code><a href="#topic+create_normal_distr">create_normal_distr</a></code>,
<code><a href="#topic+create_one_div_x_distr">create_one_div_x_distr</a></code>,
<code><a href="#topic+create_poisson_distr">create_poisson_distr</a></code>
and <code><a href="#topic+create_uniform_distr">create_uniform_distr</a></code>
</p>
<p>See
<code><a href="#topic+create_beta_distr">create_beta_distr</a></code>,
<code><a href="#topic+create_exp_distr">create_exp_distr</a></code>,
<code><a href="#topic+create_gamma_distr">create_gamma_distr</a></code>,
<code><a href="#topic+create_inv_gamma_distr">create_inv_gamma_distr</a></code>,
<code><a href="#topic+create_laplace_distr">create_laplace_distr</a></code>,
<code><a href="#topic+create_log_normal_distr">create_log_normal_distr</a></code>,
<code><a href="#topic+create_normal_distr">create_normal_distr</a></code>,
<code><a href="#topic+create_one_div_x_distr">create_one_div_x_distr</a></code>,
<code><a href="#topic+create_poisson_distr">create_poisson_distr</a></code>
and <code><a href="#topic+create_uniform_distr">create_uniform_distr</a></code>
for examples how to use those distributions
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  # Use any distribution
  distr &lt;- create_beta_distr()

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = create_yule_tree_prior(
      birth_rate_distr = distr
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_exp_distr'>Create an exponential distribution</h2><span id='topic+create_exp_distr'></span><span id='topic+create_distr_exp'></span>

<h3>Description</h3>

<p>Create an exponential distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_exp_distr(id = NA, mean = 1, value = NA, lower = NA, upper = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_exp_distr_+3A_id">id</code></td>
<td>
<p>the distribution's ID</p>
</td></tr>
<tr><td><code id="create_exp_distr_+3A_mean">mean</code></td>
<td>
<p>the mean parameter,
a numeric value.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_mean_param">create_mean_param</a></code></p>
</td></tr>
<tr><td><code id="create_exp_distr_+3A_value">value</code></td>
<td>
<p>the initial value for the MCMC</p>
</td></tr>
<tr><td><code id="create_exp_distr_+3A_lower">lower</code></td>
<td>
<p>the lower bound, the lowest possible value</p>
</td></tr>
<tr><td><code id="create_exp_distr_+3A_upper">upper</code></td>
<td>
<p>an upper limit of the uniform distribution.
If the upper limits needs to be infinity, set <code>upper</code> to <code>Inf</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an exponential distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_distr">create_distr</a></code> shows an overview
of all supported distributions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  exp_distr &lt;- create_exp_distr()

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = create_yule_tree_prior(
      birth_rate_distr = exp_distr
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_freq_param'>Create a parameter called freq</h2><span id='topic+create_freq_param'></span><span id='topic+create_param_freq'></span>

<h3>Description</h3>

<p>Create a parameter called freq
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_freq_param(
  id = NA,
  lower = "0.0",
  upper = "1.0",
  value = "0.25",
  estimate = TRUE,
  dimension = 4
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_freq_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_freq_param_+3A_lower">lower</code></td>
<td>
<p>lowest possible value of the parameter. If the parameter
is estimated, <code>lower</code> must be less than <code>value</code></p>
</td></tr>
<tr><td><code id="create_freq_param_+3A_upper">upper</code></td>
<td>
<p>upper value of the parameter</p>
</td></tr>
<tr><td><code id="create_freq_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
<tr><td><code id="create_freq_param_+3A_estimate">estimate</code></td>
<td>
<p>TRUE if this parameter is to be estimated by BEAST2,
FALSE otherwise</p>
</td></tr>
<tr><td><code id="create_freq_param_+3A_dimension">dimension</code></td>
<td>
<p>the number of dimensions, for example, as used in
<a href="#topic+create_freq_param">create_freq_param</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called freq
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='create_gamma_distr'>Create a gamma distribution</h2><span id='topic+create_gamma_distr'></span><span id='topic+create_distr_gamma'></span>

<h3>Description</h3>

<p>Create a gamma distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_gamma_distr(
  id = NA,
  alpha = 0.5396,
  beta = 0.3819,
  value = NA,
  lower = NA,
  upper = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_gamma_distr_+3A_id">id</code></td>
<td>
<p>the distribution's ID</p>
</td></tr>
<tr><td><code id="create_gamma_distr_+3A_alpha">alpha</code></td>
<td>
<p>the alpha shape parameter,
a numeric value.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_alpha_param">create_alpha_param</a></code></p>
</td></tr>
<tr><td><code id="create_gamma_distr_+3A_beta">beta</code></td>
<td>
<p>the beta shape parameter,
a numeric value.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_beta_param">create_beta_param</a></code></p>
</td></tr>
<tr><td><code id="create_gamma_distr_+3A_value">value</code></td>
<td>
<p>the initial value for the MCMC</p>
</td></tr>
<tr><td><code id="create_gamma_distr_+3A_lower">lower</code></td>
<td>
<p>the lower bound, the lowest possible value</p>
</td></tr>
<tr><td><code id="create_gamma_distr_+3A_upper">upper</code></td>
<td>
<p>an upper limit of the uniform distribution.
If the upper limits needs to be infinity, set <code>upper</code> to <code>Inf</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a gamma distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_distr">create_distr</a></code> shows an overview
of all supported distributions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  gamma_distr &lt;- create_gamma_distr(
    alpha = 0.05,
    beta = 10.0
  )

  gtr_site_model &lt;- create_gtr_site_model(
    rate_ac_prior_distr = gamma_distr
  )

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    site_model = gtr_site_model
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_gamma_site_model'>Create a gamma site model, part of a site model</h2><span id='topic+create_gamma_site_model'></span>

<h3>Description</h3>

<p>Create a gamma site model, part of a site model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_gamma_site_model(
  gamma_cat_count = "0",
  gamma_shape = "1.0",
  prop_invariant = "0.0",
  gamma_shape_prior_distr = NA,
  freq_equilibrium = "estimated",
  freq_prior_uniform_distr_id = 1000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_gamma_site_model_+3A_gamma_cat_count">gamma_cat_count</code></td>
<td>
<p>the number of gamma categories, must
be an integer with value zero or more</p>
</td></tr>
<tr><td><code id="create_gamma_site_model_+3A_gamma_shape">gamma_shape</code></td>
<td>
<p>gamma curve shape parameter</p>
</td></tr>
<tr><td><code id="create_gamma_site_model_+3A_prop_invariant">prop_invariant</code></td>
<td>
<p>the proportion invariant, must be a value
from 0.0 to 1.0</p>
</td></tr>
<tr><td><code id="create_gamma_site_model_+3A_gamma_shape_prior_distr">gamma_shape_prior_distr</code></td>
<td>
<p>the distribution of the gamma shape prior.
<code>gamma_shape_prior_distr</code> must be <code>NA</code> for
a <code>gamma_cat_count</code> of zero or one.
For a <code>gamma_cat_count</code> of two or more,
leaving <code>gamma_shape_prior_distr</code> equal to its default
value of <code>NA</code>, a default distribution is used.
Else <code>gamma_shape_prior_distr</code> must be a
distribution, as can be created by <code><a href="#topic+create_distr">create_distr</a></code></p>
</td></tr>
<tr><td><code id="create_gamma_site_model_+3A_freq_equilibrium">freq_equilibrium</code></td>
<td>
<p>the frequency in which the rates are at equilibrium
are either <code>estimated</code>, <code>empirical</code> or <code>all_equal</code>.
<code>get_freq_equilibrium_names</code> returns the possible values
for <code>freq_equilibrium</code></p>
</td></tr>
<tr><td><code id="create_gamma_site_model_+3A_freq_prior_uniform_distr_id">freq_prior_uniform_distr_id</code></td>
<td>
<p>the ID of the &lsquo;FrequenciesPrior'&rsquo;s
uniform distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a gamma site model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+create_gamma_site_model">create_gamma_site_model</a></code>
to create a gamma site model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  gamma_site_model &lt;- create_gamma_site_model(prop_invariant = 0.5)

  site_model &lt;- create_hky_site_model(gamma_site_model = gamma_site_model)

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    get_fasta_filename(),
    beast2_input_file,
    site_model = site_model
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_gtr_site_model'>Create a GTR site model</h2><span id='topic+create_gtr_site_model'></span><span id='topic+create_site_model_gtr'></span>

<h3>Description</h3>

<p>Create a GTR site model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_gtr_site_model(
  id = NA,
  gamma_site_model = create_gamma_site_model(),
  rate_ac_prior_distr = create_gamma_distr(alpha = 0.05, beta = create_beta_param(value =
    "10.0")),
  rate_ag_prior_distr = create_gamma_distr(alpha = 0.05, beta = create_beta_param(value =
    "20.0")),
  rate_at_prior_distr = create_gamma_distr(alpha = 0.05, beta = create_beta_param(value =
    "10.0")),
  rate_cg_prior_distr = create_gamma_distr(alpha = 0.05, beta = create_beta_param(value =
    "10.0")),
  rate_gt_prior_distr = create_gamma_distr(alpha = 0.05, beta = create_beta_param(value =
    "10.0")),
  rate_ac_param = create_rate_ac_param(),
  rate_ag_param = create_rate_ag_param(),
  rate_at_param = create_rate_at_param(),
  rate_cg_param = create_rate_cg_param(),
  rate_ct_param = create_rate_ct_param(),
  rate_gt_param = create_rate_gt_param(),
  freq_equilibrium = "estimated",
  freq_param = create_freq_param()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_gtr_site_model_+3A_id">id</code></td>
<td>
<p>the IDs of the alignment (can be extracted from
the FASTA filename using <code><a href="#topic+get_alignment_id">get_alignment_id</a></code>)</p>
</td></tr>
<tr><td><code id="create_gtr_site_model_+3A_gamma_site_model">gamma_site_model</code></td>
<td>
<p>a gamma site model, as created
by <a href="#topic+create_gamma_site_model">create_gamma_site_model</a></p>
</td></tr>
<tr><td><code id="create_gtr_site_model_+3A_rate_ac_prior_distr">rate_ac_prior_distr</code></td>
<td>
<p>the AC rate prior distribution,
as returned by <code><a href="#topic+create_distr">create_distr</a></code></p>
</td></tr>
<tr><td><code id="create_gtr_site_model_+3A_rate_ag_prior_distr">rate_ag_prior_distr</code></td>
<td>
<p>the AG rate prior distribution,
as returned by <code><a href="#topic+create_distr">create_distr</a></code></p>
</td></tr>
<tr><td><code id="create_gtr_site_model_+3A_rate_at_prior_distr">rate_at_prior_distr</code></td>
<td>
<p>the AT rate prior distribution,
as returned by <code><a href="#topic+create_distr">create_distr</a></code></p>
</td></tr>
<tr><td><code id="create_gtr_site_model_+3A_rate_cg_prior_distr">rate_cg_prior_distr</code></td>
<td>
<p>the CG rate prior distribution,
as returned by <code><a href="#topic+create_distr">create_distr</a></code></p>
</td></tr>
<tr><td><code id="create_gtr_site_model_+3A_rate_gt_prior_distr">rate_gt_prior_distr</code></td>
<td>
<p>the GT rate prior distribution,
as returned by <code><a href="#topic+create_distr">create_distr</a></code></p>
</td></tr>
<tr><td><code id="create_gtr_site_model_+3A_rate_ac_param">rate_ac_param</code></td>
<td>
<p>the 'rate AC' parameter,
a numeric value.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_rate_ac_param">create_rate_ac_param</a></code></p>
</td></tr>
<tr><td><code id="create_gtr_site_model_+3A_rate_ag_param">rate_ag_param</code></td>
<td>
<p>the 'rate AG' parameter,
a numeric value.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_rate_ag_param">create_rate_ag_param</a></code></p>
</td></tr>
<tr><td><code id="create_gtr_site_model_+3A_rate_at_param">rate_at_param</code></td>
<td>
<p>the 'rate AT' parameter,
a numeric value.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_rate_at_param">create_rate_at_param</a></code></p>
</td></tr>
<tr><td><code id="create_gtr_site_model_+3A_rate_cg_param">rate_cg_param</code></td>
<td>
<p>the 'rate CG' parameter,
a numeric value.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_rate_cg_param">create_rate_cg_param</a></code></p>
</td></tr>
<tr><td><code id="create_gtr_site_model_+3A_rate_ct_param">rate_ct_param</code></td>
<td>
<p>the 'rate CT' parameter,
a numeric value.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_rate_ct_param">create_rate_ct_param</a></code></p>
</td></tr>
<tr><td><code id="create_gtr_site_model_+3A_rate_gt_param">rate_gt_param</code></td>
<td>
<p>the 'rate GT' parameter,
a numeric value.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_rate_gt_param">create_rate_gt_param</a></code></p>
</td></tr>
<tr><td><code id="create_gtr_site_model_+3A_freq_equilibrium">freq_equilibrium</code></td>
<td>
<p>the frequency in which the rates are at equilibrium
are either <code>estimated</code>, <code>empirical</code> or <code>all_equal</code>.
<code>get_freq_equilibrium_names</code> returns the possible values
for <code>freq_equilibrium</code></p>
</td></tr>
<tr><td><code id="create_gtr_site_model_+3A_freq_param">freq_param</code></td>
<td>
<p>a 'freq' parameter,
as created by <a href="#topic+create_freq_param">create_freq_param</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a GTR site_model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  gtr_site_model &lt;- create_gtr_site_model(
    rate_ac_param = 1.2,
    rate_ag_param = 2.3,
    rate_at_param = 3.4,
    rate_cg_param = 4.5,
    rate_ct_param = 5.6,
    rate_gt_param = 6.7
  )

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    site_model = gtr_site_model
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_gtr_subst_model_xml'>Converts a GTR site model to XML,
used in the <code>substModel</code> section</h2><span id='topic+create_gtr_subst_model_xml'></span>

<h3>Description</h3>

<p>Converts a GTR site model to XML,
used in the <code>substModel</code> section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_gtr_subst_model_xml(site_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_gtr_subst_model_xml_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the site model as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='create_hky_site_model'>Create an HKY site model</h2><span id='topic+create_hky_site_model'></span><span id='topic+create_site_model_hky'></span>

<h3>Description</h3>

<p>Create an HKY site model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_hky_site_model(
  id = NA,
  kappa = "obsolete",
  kappa_param = create_kappa_param(value = "2.0"),
  gamma_site_model = create_gamma_site_model(),
  kappa_prior_distr = create_log_normal_distr(m = create_m_param(value = "1.0"), s =
    1.25),
  freq_equilibrium = "estimated",
  freq_param = create_freq_param()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_hky_site_model_+3A_id">id</code></td>
<td>
<p>the IDs of the alignment (can be extracted from
the FASTA filename using <code><a href="#topic+get_alignment_id">get_alignment_id</a></code>)</p>
</td></tr>
<tr><td><code id="create_hky_site_model_+3A_kappa">kappa</code></td>
<td>
<p>obsoleted parameter. It is the value in the 'kappa_param'
argument</p>
</td></tr>
<tr><td><code id="create_hky_site_model_+3A_kappa_param">kappa_param</code></td>
<td>
<p>a 'kappa' parameter,
as created by <a href="#topic+create_kappa_param">create_kappa_param</a></p>
</td></tr>
<tr><td><code id="create_hky_site_model_+3A_gamma_site_model">gamma_site_model</code></td>
<td>
<p>a gamma site model, as created
by <a href="#topic+create_gamma_site_model">create_gamma_site_model</a></p>
</td></tr>
<tr><td><code id="create_hky_site_model_+3A_kappa_prior_distr">kappa_prior_distr</code></td>
<td>
<p>the distribution of the kappa prior,
which is a log-normal distribution
(as created by <code><a href="#topic+create_log_normal_distr">create_log_normal_distr</a></code>)
by default</p>
</td></tr>
<tr><td><code id="create_hky_site_model_+3A_freq_equilibrium">freq_equilibrium</code></td>
<td>
<p>the frequency in which the rates are at equilibrium
are either <code>estimated</code>, <code>empirical</code> or <code>all_equal</code>.
<code>get_freq_equilibrium_names</code> returns the possible values
for <code>freq_equilibrium</code></p>
</td></tr>
<tr><td><code id="create_hky_site_model_+3A_freq_param">freq_param</code></td>
<td>
<p>a 'freq' parameter,
as created by <a href="#topic+create_freq_param">create_freq_param</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>an HKY site_model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  hky_site_model &lt;- create_hky_site_model()
  output_filename &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    output_filename = output_filename,
    site_model = hky_site_model
  )
  file.remove(output_filename)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_hky_subst_model_xml'>Converts a site model to XML,
used in the <code>substModel</code> section</h2><span id='topic+create_hky_subst_model_xml'></span>

<h3>Description</h3>

<p>Converts a site model to XML,
used in the <code>substModel</code> section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_hky_subst_model_xml(site_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_hky_subst_model_xml_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the site model as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='create_inference_model'>Create a Bayesian phylogenetic inference model.</h2><span id='topic+create_inference_model'></span>

<h3>Description</h3>

<p>Create a Bayesian phylogenetic inference model,
as can be done by BEAUti.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_inference_model(
  site_model = create_jc69_site_model(),
  clock_model = create_strict_clock_model(),
  tree_prior = create_yule_tree_prior(),
  mrca_prior = NA,
  mcmc = create_mcmc(),
  beauti_options = create_beauti_options(),
  tipdates_filename = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_inference_model_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
<tr><td><code id="create_inference_model_+3A_clock_model">clock_model</code></td>
<td>
<p>a clock model,
as returned by <code><a href="#topic+create_clock_model">create_clock_model</a></code></p>
</td></tr>
<tr><td><code id="create_inference_model_+3A_tree_prior">tree_prior</code></td>
<td>
<p>a tree priors,
as returned by <code><a href="#topic+create_tree_prior">create_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="create_inference_model_+3A_mrca_prior">mrca_prior</code></td>
<td>
<p>a Most Recent Common Ancestor prior,
as returned by <code><a href="#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
<tr><td><code id="create_inference_model_+3A_mcmc">mcmc</code></td>
<td>
<p>one MCMC.
Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create an MCMC.
Use <code><a href="#topic+create_ns_mcmc">create_ns_mcmc</a></code> to create an MCMC
for a Nested Sampling run.
Use <code><a href="#topic+check_mcmc">check_mcmc</a></code> to check if an MCMC is valid.
Use <code><a href="#topic+rename_mcmc_filenames">rename_mcmc_filenames</a></code> to rename the filenames in an MCMC.</p>
</td></tr>
<tr><td><code id="create_inference_model_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
<tr><td><code id="create_inference_model_+3A_tipdates_filename">tipdates_filename</code></td>
<td>
<p>name of the file containing the tip dates.
This file is assumed to have two columns, separated by a tab.
The first column contains the taxa names, the second column contains
the date.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an inference model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_test_inference_model">create_test_inference_model</a> to create an inference model
with a short MCMC, to be used in testing.
Use <a href="#topic+create_ns_inference_model">create_ns_inference_model</a> to create an inference model
to estimate the marginal likelihood
(aka evidence) using a nested sampling approach.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  check_empty_beautier_folder()

  # Create an MCMC chain with 50 states
  inference_model &lt;- create_inference_model(
    mcmc = create_mcmc(chain_length = 50000, store_every = 1000)
  )

  output_filename &lt;- get_beautier_tempfilename()
  create_beast2_input_file_from_model(
    input_filename = get_fasta_filename(),
    output_filename = output_filename,
    inference_model = inference_model
  )
  file.remove(output_filename)

  remove_beautier_folder()
  check_empty_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_inv_gamma_distr'>Create an inverse-gamma distribution</h2><span id='topic+create_inv_gamma_distr'></span><span id='topic+create_distr_inv_gamma'></span>

<h3>Description</h3>

<p>Create an inverse-gamma distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_inv_gamma_distr(
  id = NA,
  alpha = 0,
  beta = 1,
  value = NA,
  lower = NA,
  upper = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_inv_gamma_distr_+3A_id">id</code></td>
<td>
<p>the distribution's ID</p>
</td></tr>
<tr><td><code id="create_inv_gamma_distr_+3A_alpha">alpha</code></td>
<td>
<p>the alpha shape parameter,
a numeric value.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_alpha_param">create_alpha_param</a></code></p>
</td></tr>
<tr><td><code id="create_inv_gamma_distr_+3A_beta">beta</code></td>
<td>
<p>the beta shape parameter,
a numeric value.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_beta_param">create_beta_param</a></code></p>
</td></tr>
<tr><td><code id="create_inv_gamma_distr_+3A_value">value</code></td>
<td>
<p>the initial value for the MCMC</p>
</td></tr>
<tr><td><code id="create_inv_gamma_distr_+3A_lower">lower</code></td>
<td>
<p>the lower bound, the lowest possible value</p>
</td></tr>
<tr><td><code id="create_inv_gamma_distr_+3A_upper">upper</code></td>
<td>
<p>an upper limit of the uniform distribution.
If the upper limits needs to be infinity, set <code>upper</code> to <code>Inf</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an inverse-gamma distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_distr">create_distr</a></code> shows an overview
of all supported distributions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  inv_gamma_distr &lt;- create_inv_gamma_distr()

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = create_yule_tree_prior(
      birth_rate_distr = inv_gamma_distr
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_jc69_site_model'>Create a JC69 site model</h2><span id='topic+create_jc69_site_model'></span><span id='topic+create_site_model_jc69'></span>

<h3>Description</h3>

<p>Create a JC69 site model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_jc69_site_model(id = NA, gamma_site_model = create_gamma_site_model())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_jc69_site_model_+3A_id">id</code></td>
<td>
<p>the IDs of the alignment (can be extracted from
the FASTA filename using <code><a href="#topic+get_alignment_id">get_alignment_id</a></code>)</p>
</td></tr>
<tr><td><code id="create_jc69_site_model_+3A_gamma_site_model">gamma_site_model</code></td>
<td>
<p>a gamma site model, as created
by <a href="#topic+create_gamma_site_model">create_gamma_site_model</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a JC69 site_model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  jc69_site_model &lt;- create_jc69_site_model()

  output_filename &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    output_filename = output_filename,
    site_model = jc69_site_model
  )
  file.remove(output_filename)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_jc69_subst_model_xml'>Converts a JC69 site model to XML,
used in the <code>substModel</code> section</h2><span id='topic+create_jc69_subst_model_xml'></span>

<h3>Description</h3>

<p>Converts a JC69 site model to XML,
used in the <code>substModel</code> section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_jc69_subst_model_xml(site_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_jc69_subst_model_xml_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the site model as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='create_kappa_1_param'>Create a parameter called kappa 1</h2><span id='topic+create_kappa_1_param'></span><span id='topic+create_param_kappa_1'></span>

<h3>Description</h3>

<p>Create a parameter called kappa 1
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_kappa_1_param(id = NA, lower = "0.0", value = "2.0", estimate = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_kappa_1_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_kappa_1_param_+3A_lower">lower</code></td>
<td>
<p>lowest possible value of the parameter. If the parameter
is estimated, <code>lower</code> must be less than <code>value</code></p>
</td></tr>
<tr><td><code id="create_kappa_1_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
<tr><td><code id="create_kappa_1_param_+3A_estimate">estimate</code></td>
<td>
<p>TRUE if this parameter is to be estimated by BEAST2,
FALSE otherwise</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called kappa 1
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='create_kappa_2_param'>Create a parameter called kappa 2</h2><span id='topic+create_kappa_2_param'></span><span id='topic+create_param_kappa_2'></span>

<h3>Description</h3>

<p>Create a parameter called kappa 2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_kappa_2_param(id = NA, lower = "0.0", value = "2.0", estimate = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_kappa_2_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_kappa_2_param_+3A_lower">lower</code></td>
<td>
<p>lowest possible value of the parameter. If the parameter
is estimated, <code>lower</code> must be less than <code>value</code></p>
</td></tr>
<tr><td><code id="create_kappa_2_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
<tr><td><code id="create_kappa_2_param_+3A_estimate">estimate</code></td>
<td>
<p>TRUE if this parameter is to be estimated by BEAST2,
FALSE otherwise</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called kappa 2
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='create_kappa_param'>Create a parameter called kappa</h2><span id='topic+create_kappa_param'></span><span id='topic+create_param_kappa'></span>

<h3>Description</h3>

<p>Create a parameter called kappa
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_kappa_param(id = NA, lower = "0.0", value = "2.0", estimate = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_kappa_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_kappa_param_+3A_lower">lower</code></td>
<td>
<p>lowest possible value of the parameter. If the parameter
is estimated, <code>lower</code> must be less than <code>value</code></p>
</td></tr>
<tr><td><code id="create_kappa_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
<tr><td><code id="create_kappa_param_+3A_estimate">estimate</code></td>
<td>
<p>TRUE if this parameter is to be estimated by BEAST2,
FALSE otherwise</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called kappa
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='create_lambda_param'>Create a parameter called lambda</h2><span id='topic+create_lambda_param'></span><span id='topic+create_param_lambda'></span>

<h3>Description</h3>

<p>Create a parameter called lambda
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_lambda_param(id = NA, value = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_lambda_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_lambda_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called lambda
</p>


<h3>Note</h3>

<p>this parameter is used in a Poisson distribution
(as returned by <code><a href="#topic+create_poisson_distr">create_poisson_distr</a></code>)
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_param">create_param</a></code> contains a list
of all parameters that can be created
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  # Create the parameter
  lambda_param &lt;- create_lambda_param()

  # Use the parameter in a distribution
  poisson_distr &lt;- create_poisson_distr(
    lambda = lambda_param
  )

  # Use the distribution to create a BEAST2 input file
  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = create_yule_tree_prior(
      birth_rate_distr = poisson_distr
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_laplace_distr'>Create a Laplace distribution</h2><span id='topic+create_laplace_distr'></span><span id='topic+create_distr_laplace'></span>

<h3>Description</h3>

<p>Create a Laplace distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_laplace_distr(
  id = NA,
  mu = 0,
  scale = 1,
  value = NA,
  lower = NA,
  upper = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_laplace_distr_+3A_id">id</code></td>
<td>
<p>the distribution's ID</p>
</td></tr>
<tr><td><code id="create_laplace_distr_+3A_mu">mu</code></td>
<td>
<p>the mu parameter,
a numeric value.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_mu_param">create_mu_param</a></code></p>
</td></tr>
<tr><td><code id="create_laplace_distr_+3A_scale">scale</code></td>
<td>
<p>the scale parameter,
a numeric value.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_scale_param">create_scale_param</a></code></p>
</td></tr>
<tr><td><code id="create_laplace_distr_+3A_value">value</code></td>
<td>
<p>the initial value for the MCMC</p>
</td></tr>
<tr><td><code id="create_laplace_distr_+3A_lower">lower</code></td>
<td>
<p>the lower bound, the lowest possible value</p>
</td></tr>
<tr><td><code id="create_laplace_distr_+3A_upper">upper</code></td>
<td>
<p>an upper limit of the uniform distribution.
If the upper limits needs to be infinity, set <code>upper</code> to <code>Inf</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Laplace distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_distr">create_distr</a></code> shows an overview
of all supported distributions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  laplace_distr &lt;- create_laplace_distr()

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = create_yule_tree_prior(
      birth_rate_distr = laplace_distr
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_log_normal_distr'>Create a log-normal distribution</h2><span id='topic+create_log_normal_distr'></span><span id='topic+create_distr_log_normal'></span>

<h3>Description</h3>

<p>Create a log-normal distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_log_normal_distr(
  id = NA,
  m = 0,
  s = 0,
  value = NA,
  lower = NA,
  upper = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_log_normal_distr_+3A_id">id</code></td>
<td>
<p>the distribution's ID</p>
</td></tr>
<tr><td><code id="create_log_normal_distr_+3A_m">m</code></td>
<td>
<p>the m parameter,
a numeric value.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_m_param">create_m_param</a></code></p>
</td></tr>
<tr><td><code id="create_log_normal_distr_+3A_s">s</code></td>
<td>
<p>the s parameter,
a numeric value.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_s_param">create_s_param</a></code></p>
</td></tr>
<tr><td><code id="create_log_normal_distr_+3A_value">value</code></td>
<td>
<p>the initial value for the MCMC</p>
</td></tr>
<tr><td><code id="create_log_normal_distr_+3A_lower">lower</code></td>
<td>
<p>the lower bound, the lowest possible value</p>
</td></tr>
<tr><td><code id="create_log_normal_distr_+3A_upper">upper</code></td>
<td>
<p>an upper limit of the uniform distribution.
If the upper limits needs to be infinity, set <code>upper</code> to <code>Inf</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a log-normal distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_distr">create_distr</a></code> shows an overview
of all supported distributions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  log_normal_distr &lt;- create_log_normal_distr()

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = create_yule_tree_prior(
      birth_rate_distr = log_normal_distr
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_loggers_xml'>Creates the three logger sections of a BEAST2 XML parameter file</h2><span id='topic+create_loggers_xml'></span>

<h3>Description</h3>

<p>The logger section has these elements:
</p>
<pre>
 &lt;logger id="tracelog" [...]&gt;
     [...]
 &lt;/logger&gt;
 &lt;logger id="screenlog" [...]&gt;
     [...]
 &lt;/logger&gt;
 &lt;logger id="treelog.t:[alignment ID]"  [...]&gt;
     [...]
 &lt;/logger&gt;
</pre>


<h3>Usage</h3>

<pre><code class='language-R'>create_loggers_xml(input_filename, inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_loggers_xml_+3A_input_filename">input_filename</code></td>
<td>
<p>A FASTA filename.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.</p>
</td></tr>
<tr><td><code id="create_loggers_xml_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_tracelog_xml">create_tracelog_xml</a> to create the XML text
of the logger with the <code>tracelog</code> ID.
Use <a href="#topic+create_screenlog_xml">create_screenlog_xml</a> to create the XML text
of the logger with the <code>screenlog</code> ID.
Use <a href="#topic+create_treelog_xml">create_treelog_xml</a> to create the XML text
of the loggers with the <code>treelog</code> ID.
</p>

<hr>
<h2 id='create_m_param'>Create a parameter called m</h2><span id='topic+create_m_param'></span><span id='topic+create_param_m'></span>

<h3>Description</h3>

<p>Create a parameter called m
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_m_param(id = NA, estimate = FALSE, lower = NA, upper = NA, value = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_m_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_m_param_+3A_estimate">estimate</code></td>
<td>
<p>TRUE if this parameter is to be estimated by BEAST2,
FALSE otherwise</p>
</td></tr>
<tr><td><code id="create_m_param_+3A_lower">lower</code></td>
<td>
<p>lowest possible value of the parameter. If the parameter
is estimated, <code>lower</code> must be less than <code>value</code></p>
</td></tr>
<tr><td><code id="create_m_param_+3A_upper">upper</code></td>
<td>
<p>upper value of the parameter</p>
</td></tr>
<tr><td><code id="create_m_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called m
</p>


<h3>Note</h3>

<p>this parameter is used in a log-normal distribution
(as returned by <code><a href="#topic+create_log_normal_distr">create_log_normal_distr</a></code>)
It cannot be estimated (as a hyper parameter) yet.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_param">create_param</a></code> contains a list
of all parameters that can be created
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  # Create the parameter
  m_param &lt;- create_m_param()

  # Use the parameter in a distribution
  log_normal_distr &lt;- create_log_normal_distr(
    m = m_param
  )

  # Use the distribution to create a BEAST2 input file
  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = create_yule_tree_prior(
      birth_rate_distr = log_normal_distr
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()

}
</code></pre>

<hr>
<h2 id='create_mcmc'>Create an MCMC configuration.</h2><span id='topic+create_mcmc'></span>

<h3>Description</h3>

<p>Create an MCMC configuration, as in the BEAUti MCMC tab.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_mcmc(
  chain_length = 1e+07,
  store_every = -1,
  pre_burnin = 0,
  n_init_attempts = 10,
  sample_from_prior = FALSE,
  tracelog = create_tracelog(),
  screenlog = create_screenlog(),
  treelog = create_treelog()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_mcmc_+3A_chain_length">chain_length</code></td>
<td>
<p>length of the MCMC chain</p>
</td></tr>
<tr><td><code id="create_mcmc_+3A_store_every">store_every</code></td>
<td>
<p>number of states the MCMC will process
before the posterior's state will be saved to file.
Use -1 or <code>NA</code> to use the default frequency.</p>
</td></tr>
<tr><td><code id="create_mcmc_+3A_pre_burnin">pre_burnin</code></td>
<td>
<p>number of burn in samples taken before entering
the main loop</p>
</td></tr>
<tr><td><code id="create_mcmc_+3A_n_init_attempts">n_init_attempts</code></td>
<td>
<p>number of initialization attempts before failing</p>
</td></tr>
<tr><td><code id="create_mcmc_+3A_sample_from_prior">sample_from_prior</code></td>
<td>
<p>set to <a href="base.html#topic+TRUE">TRUE</a> to sample from the prior</p>
</td></tr>
<tr><td><code id="create_mcmc_+3A_tracelog">tracelog</code></td>
<td>
<p>a <code>tracelog</code>,
as created by <a href="#topic+create_tracelog">create_tracelog</a></p>
</td></tr>
<tr><td><code id="create_mcmc_+3A_screenlog">screenlog</code></td>
<td>
<p>a <code>screenlog</code>,
as created by <a href="#topic+create_screenlog">create_screenlog</a></p>
</td></tr>
<tr><td><code id="create_mcmc_+3A_treelog">treelog</code></td>
<td>
<p>a <code>treelog</code>,
as created by <a href="#topic+create_treelog">create_treelog</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are four things that can be saved:
* <code>store_every</code>: saves the state of the MCMC to file,
as a <code>.state.xml</code> file
* <code>tracelog</code>: stores the trace of the state of the MCMC
to file. See <code>create_tracelog</code>
how to specify the filename
* <code>screenlog</code>: stores the screen output
to file. See <code>create_screenlog</code>
how to specify the filename
* <code>treelog</code>: stores the estimated phylogenies
to file. See <code>create_treelog</code>
how to specify the filename
</p>


<h3>Value</h3>

<p>an MCMC configuration
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+create_test_mcmc">create_test_mcmc</a></code> to create a short regular MCMC,
that can be used for testing runs.
Use <code><a href="#topic+create_ns_mcmc">create_ns_mcmc</a></code> to create an MCMC for a Nested Sampling run.
Use <code><a href="#topic+check_mcmc">check_mcmc</a></code> to check if an MCMC is valid.
Use <code><a href="#topic+rename_mcmc_filenames">rename_mcmc_filenames</a></code> to rename the filenames in an MCMC.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  # Create an MCMC chain with 50 states
  mcmc &lt;- create_mcmc(chain_length = 50000, store_every = 1000)

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    get_fasta_filename(),
    beast2_input_file,
    mcmc = mcmc
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_mean_param'>Create a parameter called mean</h2><span id='topic+create_mean_param'></span><span id='topic+create_param_mean'></span>

<h3>Description</h3>

<p>Create a parameter called mean
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_mean_param(id = NA, value = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_mean_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_mean_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called mean
</p>


<h3>Note</h3>

<p>this parameter is used in an exponential distribution
(as returned by <code><a href="#topic+create_exp_distr">create_exp_distr</a></code>)
and normal distribution
(as returned by <code><a href="#topic+create_normal_distr">create_normal_distr</a></code>).
It cannot be estimated (as a hyper parameter) yet.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_param">create_param</a></code> contains a list
of all parameters that can be created
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  # Create the parameter
  mean_param &lt;- create_mean_param(value = 1.0)

  # Use the parameter in a distribution
  exp_distr &lt;- create_exp_distr(
    mean = mean_param
  )

  # Use the distribution to create a BEAST2 input file
  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = create_yule_tree_prior(
      birth_rate_distr = exp_distr
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_mrca_prior'>Create a Most Recent Common Ancestor prior</h2><span id='topic+create_mrca_prior'></span>

<h3>Description</h3>

<p>Create a Most Recent Common Ancestor prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_mrca_prior(
  alignment_id = NA,
  taxa_names = NA,
  is_monophyletic = FALSE,
  mrca_distr = NA,
  name = NA,
  clock_prior_distr_id = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_mrca_prior_+3A_alignment_id">alignment_id</code></td>
<td>
<p>ID of the alignment,
as returned by <a href="#topic+get_alignment_id">get_alignment_id</a>.
Keep at <code>NA</code> to have it initialized automatically</p>
</td></tr>
<tr><td><code id="create_mrca_prior_+3A_taxa_names">taxa_names</code></td>
<td>
<p>names of the taxa,
as returned by <code><a href="#topic+get_taxa_names">get_taxa_names</a></code>.
Keep at <code>NA</code> to have it initialized automatically,
using all taxa in the alignment</p>
</td></tr>
<tr><td><code id="create_mrca_prior_+3A_is_monophyletic">is_monophyletic</code></td>
<td>
<p>boolean to indicate monophyly is assumed in
a Most Recent Common Ancestor prior,
as returned by <code><a href="#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
<tr><td><code id="create_mrca_prior_+3A_mrca_distr">mrca_distr</code></td>
<td>
<p>the distribution used by the MRCA prior.
Can be NA (the default) or any distribution
returned by <code><a href="#topic+create_distr">create_distr</a></code></p>
</td></tr>
<tr><td><code id="create_mrca_prior_+3A_name">name</code></td>
<td>
<p>the unique name of the MRCA prior, for example a genus, family,
order or even class name.
Leave at <a href="base.html#topic+NA">NA</a> to have it named automatically.</p>
</td></tr>
<tr><td><code id="create_mrca_prior_+3A_clock_prior_distr_id">clock_prior_distr_id</code></td>
<td>
<p>ID of an MRCA clock model's distribution.
Keep at <code>NA</code> to have it initialized automatically</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an MRCA prior
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

fasta_filename &lt;- get_beautier_path("anthus_aco.fas")

 # The first two taxa are sister species
 mrca_prior &lt;- create_mrca_prior(
   taxa_names = get_taxa_names(filename = fasta_filename)[1:2]
 )

 # The taxa are monophyletic
 mrca_prior &lt;- create_mrca_prior(
   taxa_names = get_taxa_names(filename = fasta_filename),
   is_monophyletic = TRUE
 )

 # Set the crown age to 10
 mrca_prior &lt;- create_mrca_prior(
   taxa_names = get_taxa_names(fasta_filename),
   mrca_distr = create_normal_distr(mean = 10, sigma = 0.1)
 )

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_mu_param'>Create a parameter called mu</h2><span id='topic+create_mu_param'></span><span id='topic+create_param_mu'></span>

<h3>Description</h3>

<p>Create a parameter called mu
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_mu_param(id = NA, value = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_mu_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_mu_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called mu
</p>


<h3>Note</h3>

<p>this parameter is used in a Laplace distribution
(as returned by <code><a href="#topic+create_laplace_distr">create_laplace_distr</a></code>).
It cannot be estimated (as a hyper parameter) yet.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_param">create_param</a></code> contains a list
of all parameters that can be created
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  # Create the parameter
  mu_param &lt;- create_mu_param()

  # Use the parameter in a distribution
  laplace_distr &lt;- create_laplace_distr(
    mu = mu_param
  )

  # Use the distribution to create a BEAST2 input file
  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = create_yule_tree_prior(
      birth_rate_distr = laplace_distr
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_normal_distr'>Create an normal distribution</h2><span id='topic+create_normal_distr'></span><span id='topic+create_distr_normal'></span>

<h3>Description</h3>

<p>Create an normal distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_normal_distr(
  id = NA,
  mean = 0,
  sigma = 1,
  value = NA,
  lower = NA,
  upper = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_normal_distr_+3A_id">id</code></td>
<td>
<p>the distribution's ID</p>
</td></tr>
<tr><td><code id="create_normal_distr_+3A_mean">mean</code></td>
<td>
<p>the mean parameter,
a numeric value.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_mean_param">create_mean_param</a></code></p>
</td></tr>
<tr><td><code id="create_normal_distr_+3A_sigma">sigma</code></td>
<td>
<p>the sigma parameter,
a numeric value.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_sigma_param">create_sigma_param</a></code></p>
</td></tr>
<tr><td><code id="create_normal_distr_+3A_value">value</code></td>
<td>
<p>the initial value for the MCMC</p>
</td></tr>
<tr><td><code id="create_normal_distr_+3A_lower">lower</code></td>
<td>
<p>the lower bound, the lowest possible value</p>
</td></tr>
<tr><td><code id="create_normal_distr_+3A_upper">upper</code></td>
<td>
<p>an upper limit of the uniform distribution.
If the upper limits needs to be infinity, set <code>upper</code> to <code>Inf</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a normal distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_distr">create_distr</a></code> shows an overview
of all supported distributions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  normal_distr &lt;- create_normal_distr()

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = create_yule_tree_prior(
      birth_rate_distr = normal_distr
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_ns_inference_model'>Create an inference model to measure the evidence of.</h2><span id='topic+create_ns_inference_model'></span>

<h3>Description</h3>

<p>Create an inference model to measure the evidence of.
To do so, the inference model is created as usual (see
<a href="#topic+create_inference_model">create_inference_model</a>), except
for using a Nested Sampling MCMC (see <a href="#topic+create_ns_mcmc">create_ns_mcmc</a>)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_ns_inference_model(
  site_model = create_jc69_site_model(),
  clock_model = create_strict_clock_model(),
  tree_prior = create_yule_tree_prior(),
  mcmc = create_ns_mcmc()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_ns_inference_model_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
<tr><td><code id="create_ns_inference_model_+3A_clock_model">clock_model</code></td>
<td>
<p>a clock model,
as returned by <code><a href="#topic+create_clock_model">create_clock_model</a></code></p>
</td></tr>
<tr><td><code id="create_ns_inference_model_+3A_tree_prior">tree_prior</code></td>
<td>
<p>a tree priors,
as returned by <code><a href="#topic+create_tree_prior">create_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="create_ns_inference_model_+3A_mcmc">mcmc</code></td>
<td>
<p>one MCMC.
Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create an MCMC.
Use <code><a href="#topic+create_ns_mcmc">create_ns_mcmc</a></code> to create an MCMC
for a Nested Sampling run.
Use <code><a href="#topic+check_mcmc">check_mcmc</a></code> to check if an MCMC is valid.
Use <code><a href="#topic+rename_mcmc_filenames">rename_mcmc_filenames</a></code> to rename the filenames in an MCMC.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an inference model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_inference_model">create_inference_model</a> to create a
regular  inference model.
Use <a href="#topic+create_test_ns_inference_model">create_test_ns_inference_model</a> to create an inference model
to estimate the marginal likelihood with a short MCMC,
to be used in testing.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

inference_model &lt;- create_ns_inference_model()

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_ns_mcmc'>Create an MCMC object to estimate the marginal likelihood
using Nested Sampling.</h2><span id='topic+create_ns_mcmc'></span><span id='topic+create_mcmc_nested_sampling'></span>

<h3>Description</h3>

<p>This will result in a BEAST run that estimates the marginal
likelihood until convergence is achieved.
In this context, <code>chain_length</code> is only an upper bound
to the length of that run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_ns_mcmc(
  chain_length = 1e+07,
  store_every = -1,
  pre_burnin = 0,
  n_init_attempts = 3,
  particle_count = 1,
  sub_chain_length = 5000,
  epsilon = "1e-12",
  tracelog = create_tracelog(),
  screenlog = create_screenlog(),
  treelog = create_treelog()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_ns_mcmc_+3A_chain_length">chain_length</code></td>
<td>
<p>upper bound to the length of the MCMC chain</p>
</td></tr>
<tr><td><code id="create_ns_mcmc_+3A_store_every">store_every</code></td>
<td>
<p>number of states the MCMC will process
before the posterior's state will be saved to file.
Use -1 or <code>NA</code> to use the default frequency.</p>
</td></tr>
<tr><td><code id="create_ns_mcmc_+3A_pre_burnin">pre_burnin</code></td>
<td>
<p>number of burn in samples taken before entering
the main loop</p>
</td></tr>
<tr><td><code id="create_ns_mcmc_+3A_n_init_attempts">n_init_attempts</code></td>
<td>
<p>number of initialization attempts before failing</p>
</td></tr>
<tr><td><code id="create_ns_mcmc_+3A_particle_count">particle_count</code></td>
<td>
<p>number of particles</p>
</td></tr>
<tr><td><code id="create_ns_mcmc_+3A_sub_chain_length">sub_chain_length</code></td>
<td>
<p>sub-chain length</p>
</td></tr>
<tr><td><code id="create_ns_mcmc_+3A_epsilon">epsilon</code></td>
<td>
<p>epsilon</p>
</td></tr>
<tr><td><code id="create_ns_mcmc_+3A_tracelog">tracelog</code></td>
<td>
<p>a <code>tracelog</code>,
as created by <a href="#topic+create_tracelog">create_tracelog</a></p>
</td></tr>
<tr><td><code id="create_ns_mcmc_+3A_screenlog">screenlog</code></td>
<td>
<p>a <code>screenlog</code>,
as created by <a href="#topic+create_screenlog">create_screenlog</a></p>
</td></tr>
<tr><td><code id="create_ns_mcmc_+3A_treelog">treelog</code></td>
<td>
<p>a <code>treelog</code>,
as created by <a href="#topic+create_treelog">create_treelog</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>an MCMC object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>References</h3>

<p>* [1] Patricio Maturana Russel, Brendon J Brewer, Steffen Klaere,
Remco R Bouckaert; Model Selection and Parameter Inference in
Phylogenetics Using Nested Sampling, Systematic Biology, 2018,
syy050, https://doi.org/10.1093/sysbio/syy050
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create a regular MCMC.
Use <code><a href="#topic+create_test_ns_mcmc">create_test_ns_mcmc</a></code> to create an NS MCMC for testing,
with, among others, a short MCMC chain length.
Use <code><a href="#topic+check_ns_mcmc">check_ns_mcmc</a></code> to check that an NS MCMC object is valid.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  mcmc &lt;- create_ns_mcmc(
    chain_length = 1e7,
    store_every = 1000,
    particle_count = 1,
    sub_chain_length = 1000,
    epsilon = 1e-12
  )

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    get_fasta_filename(),
    beast2_input_file,
    mcmc = mcmc
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_one_div_x_distr'>Create a 1/x distribution</h2><span id='topic+create_one_div_x_distr'></span><span id='topic+create_distr_one_div_x'></span>

<h3>Description</h3>

<p>Create a 1/x distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_one_div_x_distr(id = NA, value = NA, lower = NA, upper = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_one_div_x_distr_+3A_id">id</code></td>
<td>
<p>the distribution's ID</p>
</td></tr>
<tr><td><code id="create_one_div_x_distr_+3A_value">value</code></td>
<td>
<p>the initial value for the MCMC</p>
</td></tr>
<tr><td><code id="create_one_div_x_distr_+3A_lower">lower</code></td>
<td>
<p>the lower bound, the lowest possible value</p>
</td></tr>
<tr><td><code id="create_one_div_x_distr_+3A_upper">upper</code></td>
<td>
<p>an upper limit of the uniform distribution.
If the upper limits needs to be infinity, set <code>upper</code> to <code>Inf</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a 1/x distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_distr">create_distr</a></code> shows an overview
of all supported distributions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  one_div_x_distr &lt;- create_one_div_x_distr()

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = create_yule_tree_prior(
      birth_rate_distr = one_div_x_distr
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_param'>General function to create a parameter.</h2><span id='topic+create_param'></span>

<h3>Description</h3>

<p>General function to create a parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_param(name, id, value, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_param_+3A_name">name</code></td>
<td>
<p>the parameters' name. Valid
names can be found in <code>get_param_names</code></p>
</td></tr>
<tr><td><code id="create_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
<tr><td><code id="create_param_+3A_...">...</code></td>
<td>
<p>specific parameter parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter
</p>


<h3>Note</h3>

<p>Prefer using the
named functions
<code><a href="#topic+create_alpha_param">create_alpha_param</a></code>,
<code><a href="#topic+create_beta_param">create_beta_param</a></code>,
<code><a href="#topic+create_clock_rate_param">create_clock_rate_param</a></code>,
<code><a href="#topic+create_kappa_1_param">create_kappa_1_param</a></code>,
<code><a href="#topic+create_kappa_2_param">create_kappa_2_param</a></code>,
<code><a href="#topic+create_lambda_param">create_lambda_param</a></code>,
<code><a href="#topic+create_m_param">create_m_param</a></code>,
<code><a href="#topic+create_mean_param">create_mean_param</a></code>,
<code><a href="#topic+create_mu_param">create_mu_param</a></code>,
<code><a href="#topic+create_rate_ac_param">create_rate_ac_param</a></code>,
<code><a href="#topic+create_rate_ag_param">create_rate_ag_param</a></code>,
<code><a href="#topic+create_rate_at_param">create_rate_at_param</a></code>,
<code><a href="#topic+create_rate_cg_param">create_rate_cg_param</a></code>,
<code><a href="#topic+create_rate_ct_param">create_rate_ct_param</a></code>,
<code><a href="#topic+create_rate_gt_param">create_rate_gt_param</a></code>,
<code><a href="#topic+create_s_param">create_s_param</a></code>,
<code><a href="#topic+create_scale_param">create_scale_param</a></code>,
and <code><a href="#topic+create_sigma_param">create_sigma_param</a></code>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  # Create an alpha parameter
  alpha_param &lt;- create_alpha_param()

  # Use the parameter in a distribution
  beta_distr &lt;- create_beta_distr(
    alpha = alpha_param
  )

  # Use the distribution to create a BEAST2 input file
  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = create_yule_tree_prior(
      birth_rate_distr = beta_distr
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_poisson_distr'>Create a Poisson distribution</h2><span id='topic+create_poisson_distr'></span><span id='topic+create_distr_poisson'></span>

<h3>Description</h3>

<p>Create a Poisson distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_poisson_distr(id = NA, lambda = 0, value = NA, lower = NA, upper = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_poisson_distr_+3A_id">id</code></td>
<td>
<p>the distribution's ID</p>
</td></tr>
<tr><td><code id="create_poisson_distr_+3A_lambda">lambda</code></td>
<td>
<p>the lambda parameter,
a numeric value.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_lambda_param">create_lambda_param</a></code></p>
</td></tr>
<tr><td><code id="create_poisson_distr_+3A_value">value</code></td>
<td>
<p>the initial value for the MCMC</p>
</td></tr>
<tr><td><code id="create_poisson_distr_+3A_lower">lower</code></td>
<td>
<p>the lower bound, the lowest possible value</p>
</td></tr>
<tr><td><code id="create_poisson_distr_+3A_upper">upper</code></td>
<td>
<p>an upper limit of the uniform distribution.
If the upper limits needs to be infinity, set <code>upper</code> to <code>Inf</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Poisson distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_distr">create_distr</a></code> shows an overview
of all supported distributions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  poisson_distr &lt;- create_poisson_distr()

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = create_yule_tree_prior(
      birth_rate_distr = poisson_distr
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_rate_ac_param'>Create a parameter called 'rate AC'</h2><span id='topic+create_rate_ac_param'></span><span id='topic+create_param_rate_ac'></span>

<h3>Description</h3>

<p>Create a parameter called 'rate AC'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_rate_ac_param(id = NA, estimate = TRUE, value = "1.0", lower = "0.0")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_rate_ac_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_rate_ac_param_+3A_estimate">estimate</code></td>
<td>
<p>TRUE if this parameter is to be estimated by BEAST2,
FALSE otherwise</p>
</td></tr>
<tr><td><code id="create_rate_ac_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
<tr><td><code id="create_rate_ac_param_+3A_lower">lower</code></td>
<td>
<p>lowest possible value of the parameter. If the parameter
is estimated, <code>lower</code> must be less than <code>value</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called 'rate AC'
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_param">create_param</a></code> contains a list
of all parameters that can be created
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  # Create parameter
  rate_ac_param &lt;- create_rate_ac_param(value = 1, estimate = FALSE)

  # Use the parameter to create a BEAST2 input file
  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    site_model = create_gtr_site_model(
      rate_ac_param = rate_ac_param
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_rate_ag_param'>Create a parameter called 'rate AG'</h2><span id='topic+create_rate_ag_param'></span><span id='topic+create_param_rate_ag'></span>

<h3>Description</h3>

<p>Create a parameter called 'rate AG'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_rate_ag_param(id = NA, estimate = TRUE, value = "1.0", lower = "0.0")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_rate_ag_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_rate_ag_param_+3A_estimate">estimate</code></td>
<td>
<p>TRUE if this parameter is to be estimated by BEAST2,
FALSE otherwise</p>
</td></tr>
<tr><td><code id="create_rate_ag_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
<tr><td><code id="create_rate_ag_param_+3A_lower">lower</code></td>
<td>
<p>lowest possible value of the parameter. If the parameter
is estimated, <code>lower</code> must be less than <code>value</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called 'rate AG'
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_param">create_param</a></code> contains a list
of all parameters that can be created
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  # Create parameter
  rate_ag_param &lt;- create_rate_ag_param(value = 1, estimate = FALSE)

  # Use the parameter to create a BEAST2 input file
  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    site_model = create_gtr_site_model(
      rate_ag_param = rate_ag_param
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_rate_at_param'>Create a parameter called 'rate AT'</h2><span id='topic+create_rate_at_param'></span><span id='topic+create_param_rate_at'></span>

<h3>Description</h3>

<p>Create a parameter called 'rate AT'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_rate_at_param(id = NA, estimate = TRUE, value = "1.0", lower = "0.0")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_rate_at_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_rate_at_param_+3A_estimate">estimate</code></td>
<td>
<p>TRUE if this parameter is to be estimated by BEAST2,
FALSE otherwise</p>
</td></tr>
<tr><td><code id="create_rate_at_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
<tr><td><code id="create_rate_at_param_+3A_lower">lower</code></td>
<td>
<p>lowest possible value of the parameter. If the parameter
is estimated, <code>lower</code> must be less than <code>value</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called 'rate AT'
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_param">create_param</a></code> contains a list
of all parameters that can be created
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  # Create parameter
  rate_at_param &lt;- create_rate_at_param(value = 1, estimate = FALSE)

  # Use the parameter to create a BEAST2 input file
  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    site_model = create_gtr_site_model(
      rate_at_param = rate_at_param
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_rate_categories_state_node_xml'>Internal function</h2><span id='topic+create_rate_categories_state_node_xml'></span>

<h3>Description</h3>

<p>Creates the <code>rateCategories</code> state node,
such as:
<code>
  "&lt;stateNode id=\"rateCategories.c:[id]\"
    spec=\"parameter.IntegerParameter\"
    dimension=\"[dimension]\"&gt;
  1
  &lt;/stateNode&gt;"
</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_rate_categories_state_node_xml(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_rate_categories_state_node_xml_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the following XML:
<code>
  "&lt;stateNode id=\"rateCategories.c:[id]\"
  spec=\"parameter.IntegerParameter\" dimension=\"[dimension]\"&gt;
  1
  &lt;/stateNode&gt;"
</code>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

create_rate_categories_state_node_xml(
  create_inference_model(
    clock_model = create_rln_clock_model(
      id = 314,
      dimension = 1
    )
  )
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_rate_cg_param'>Create a parameter called 'rate CG'</h2><span id='topic+create_rate_cg_param'></span><span id='topic+create_param_rate_cg'></span>

<h3>Description</h3>

<p>Create a parameter called 'rate CG'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_rate_cg_param(id = NA, estimate = TRUE, value = "1.0", lower = "0.0")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_rate_cg_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_rate_cg_param_+3A_estimate">estimate</code></td>
<td>
<p>TRUE if this parameter is to be estimated by BEAST2,
FALSE otherwise</p>
</td></tr>
<tr><td><code id="create_rate_cg_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
<tr><td><code id="create_rate_cg_param_+3A_lower">lower</code></td>
<td>
<p>lowest possible value of the parameter. If the parameter
is estimated, <code>lower</code> must be less than <code>value</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called 'rate CG'
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_param">create_param</a></code> contains a list
of all parameters that can be created
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  # Create parameter
  rate_cg_param &lt;- create_rate_cg_param(value = 1, estimate = FALSE)

  # Use the parameter to create a BEAST2 input file
  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    site_model = create_gtr_site_model(
      rate_cg_param = rate_cg_param
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_rate_ct_param'>Create a parameter called 'rate CT'</h2><span id='topic+create_rate_ct_param'></span><span id='topic+create_param_rate_ct'></span>

<h3>Description</h3>

<p>Create a parameter called 'rate CT'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_rate_ct_param(id = NA, value = "1.0", lower = "0.0")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_rate_ct_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_rate_ct_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
<tr><td><code id="create_rate_ct_param_+3A_lower">lower</code></td>
<td>
<p>lowest possible value of the parameter. If the parameter
is estimated, <code>lower</code> must be less than <code>value</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called 'rate CT'
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_param">create_param</a></code> contains a list
of all parameters that can be created
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  # Create parameter
  rate_ct_param &lt;- create_rate_ct_param(value = 1)

  # Use the parameter to create a BEAST2 input file
  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    site_model = create_gtr_site_model(
      rate_ct_param = rate_ct_param
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_rate_gt_param'>Create a parameter called 'rate GT'</h2><span id='topic+create_rate_gt_param'></span><span id='topic+create_param_rate_gt'></span>

<h3>Description</h3>

<p>Create a parameter called 'rate GT'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_rate_gt_param(id = NA, estimate = TRUE, value = "1.0", lower = "0.0")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_rate_gt_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_rate_gt_param_+3A_estimate">estimate</code></td>
<td>
<p>TRUE if this parameter is to be estimated by BEAST2,
FALSE otherwise</p>
</td></tr>
<tr><td><code id="create_rate_gt_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
<tr><td><code id="create_rate_gt_param_+3A_lower">lower</code></td>
<td>
<p>lowest possible value of the parameter. If the parameter
is estimated, <code>lower</code> must be less than <code>value</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called 'rate GT'
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_param">create_param</a></code> contains a list
of all parameters that can be created
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  # Create parameter
  rate_gt_param &lt;- create_rate_gt_param(value = 1, estimate = FALSE)

  # Use the parameter to create a BEAST2 input file
  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    site_model = create_gtr_site_model(
      rate_gt_param = rate_gt_param
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_rln_clock_branch_rate_model_xml'>Internal function</h2><span id='topic+create_rln_clock_branch_rate_model_xml'></span>

<h3>Description</h3>

<p>Internal function to call <a href="#topic+create_branch_rate_model_xml">create_branch_rate_model_xml</a>
for a relaxed log-normal clock.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_rln_clock_branch_rate_model_xml(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_rln_clock_branch_rate_model_xml_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of XML strings
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='create_rln_clock_model'>Create a relaxed log-normal clock model</h2><span id='topic+create_rln_clock_model'></span><span id='topic+create_clock_model_rln'></span>

<h3>Description</h3>

<p>Create a relaxed log-normal clock model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_rln_clock_model(
  id = NA,
  mean_rate_prior_distr = create_uniform_distr(),
  ucldstdev_distr = create_gamma_distr(),
  mparam_id = NA,
  mean_clock_rate = "1.0",
  n_rate_categories = -1,
  normalize_mean_clock_rate = FALSE,
  dimension = NA,
  rate_scaler_factor = 0.75
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_rln_clock_model_+3A_id">id</code></td>
<td>
<p>an alignment's IDs.
An ID can be extracted from its FASTA filename
with <code><a href="#topic+get_alignment_ids_from_fasta_filenames">get_alignment_ids_from_fasta_filenames</a></code>)</p>
</td></tr>
<tr><td><code id="create_rln_clock_model_+3A_mean_rate_prior_distr">mean_rate_prior_distr</code></td>
<td>
<p>the mean clock rate prior distribution,
as created by a <code><a href="#topic+create_distr">create_distr</a></code> function</p>
</td></tr>
<tr><td><code id="create_rln_clock_model_+3A_ucldstdev_distr">ucldstdev_distr</code></td>
<td>
<p>the standard deviation of the uncorrelated
log-normal distribution,
as created by a <code><a href="#topic+create_distr">create_distr</a></code> function</p>
</td></tr>
<tr><td><code id="create_rln_clock_model_+3A_mparam_id">mparam_id</code></td>
<td>
<p>the ID of the M parameter in the <code>branchRateModel</code>,
set to NA to have it initialized</p>
</td></tr>
<tr><td><code id="create_rln_clock_model_+3A_mean_clock_rate">mean_clock_rate</code></td>
<td>
<p>the mean clock rate, 1.0 by default
(is called <code>ucld_stdev</code> in XML, where <code>ucld_stdev</code> is always 0.1)</p>
</td></tr>
<tr><td><code id="create_rln_clock_model_+3A_n_rate_categories">n_rate_categories</code></td>
<td>
<p>the number of rate categories.
-1 is default,
0 denotes as much rates as branches</p>
</td></tr>
<tr><td><code id="create_rln_clock_model_+3A_normalize_mean_clock_rate">normalize_mean_clock_rate</code></td>
<td>
<p>normalize the mean clock rate</p>
</td></tr>
<tr><td><code id="create_rln_clock_model_+3A_dimension">dimension</code></td>
<td>
<p>the dimensionality of the relaxed clock model.
Leave NA to let beautier calculate it.
Else, the dimensionality of the clock
equals twice the number of taxa minus two.</p>
</td></tr>
<tr><td><code id="create_rln_clock_model_+3A_rate_scaler_factor">rate_scaler_factor</code></td>
<td>
<p>the strict clock model's operator scaler
for the rate.
Use an empty string to indicate the default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a relaxed log-normal clock_model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  rln_clock_model &lt;- create_rln_clock_model()

  beast2_input_file &lt;- get_beautier_tempfilename()

  create_beast2_input_file(
    get_fasta_filename(),
    beast2_input_file,
    clock_model = rln_clock_model
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_s_param'>Create a parameter called s</h2><span id='topic+create_s_param'></span><span id='topic+create_param_s'></span>

<h3>Description</h3>

<p>Create a parameter called s
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_s_param(id = NA, value = 0, lower = 0, upper = Inf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_s_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_s_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
<tr><td><code id="create_s_param_+3A_lower">lower</code></td>
<td>
<p>lowest possible value of the parameter. If the parameter
is estimated, <code>lower</code> must be less than <code>value</code></p>
</td></tr>
<tr><td><code id="create_s_param_+3A_upper">upper</code></td>
<td>
<p>upper value of the parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called s
</p>


<h3>Note</h3>

<p>this parameter is used in a log-normal distribution
(as returned by <code><a href="#topic+create_log_normal_distr">create_log_normal_distr</a></code>)
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_param">create_param</a></code> contains a list
of all parameters that can be created
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  # Create the parameter
  s_param &lt;- create_s_param()

  # Use the parameter in a distribution
  log_normal_distr &lt;- create_log_normal_distr(
    s = s_param
  )

  # Use the distribution to create a BEAST2 input file
  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = create_yule_tree_prior(
      birth_rate_distr = log_normal_distr
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_scale_param'>Create a parameter called scale</h2><span id='topic+create_scale_param'></span><span id='topic+create_param_scale'></span>

<h3>Description</h3>

<p>Create a parameter called scale
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_scale_param(id = NA, value = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_scale_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_scale_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called scale
</p>


<h3>Note</h3>

<p>this parameter is used in a Laplace distribution
(as returned by <code><a href="#topic+create_laplace_distr">create_laplace_distr</a></code>)
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_param">create_param</a></code> contains a list
of all parameters that can be created
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {
  # Create the parameter
  scale_param &lt;- create_scale_param()

  # Use the parameter in a distribution
  laplace_distr &lt;- create_laplace_distr(
    scale = scale_param
  )

  # Use the distribution to create a BEAST2 input file
  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = create_yule_tree_prior(
      birth_rate_distr = laplace_distr
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_screenlog'>Create a <code>screenlog</code> object</h2><span id='topic+create_screenlog'></span>

<h3>Description</h3>

<p>Create a <code>screenlog</code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_screenlog(
  filename = "",
  log_every = 1000,
  mode = "autodetect",
  sanitise_headers = FALSE,
  sort = "none"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_screenlog_+3A_filename">filename</code></td>
<td>
<p>name of the file to store the posterior screens
phylogenies to. By default, this is <code>$(screen).screens</code></p>
</td></tr>
<tr><td><code id="create_screenlog_+3A_log_every">log_every</code></td>
<td>
<p>number of MCMC states between writing to file</p>
</td></tr>
<tr><td><code id="create_screenlog_+3A_mode">mode</code></td>
<td>
<p>mode how to log.
Valid values are the ones returned by <a href="#topic+get_log_modes">get_log_modes</a></p>
</td></tr>
<tr><td><code id="create_screenlog_+3A_sanitise_headers">sanitise_headers</code></td>
<td>
<p>set to <a href="base.html#topic+TRUE">TRUE</a> to sanitise the headers of the
log file</p>
</td></tr>
<tr><td><code id="create_screenlog_+3A_sort">sort</code></td>
<td>
<p>how to sort the log.
Valid values are the ones returned by <a href="#topic+get_log_sorts">get_log_sorts</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>screenlog</code> object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

create_screenlog()

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_screenlog_xml'>Creates the <code>screenlog</code> section of the <code>logger</code> section
of a BEAST2 XML parameter file</h2><span id='topic+create_screenlog_xml'></span>

<h3>Description</h3>

<p>Creates the <code>screenlog</code> section of the <code>logger</code> section
of a BEAST2 XML parameter file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_screenlog_xml(inference_model = create_inference_model())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_screenlog_xml_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='create_sigma_param'>Create a parameter called sigma</h2><span id='topic+create_sigma_param'></span><span id='topic+create_param_sigma'></span>

<h3>Description</h3>

<p>Create a parameter called sigma
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_sigma_param(id = NA, value = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_sigma_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="create_sigma_param_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a parameter called sigma
</p>


<h3>Note</h3>

<p>this parameter is used in a normal distribution
(as returned by <code><a href="#topic+create_normal_distr">create_normal_distr</a></code>)
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_param">create_param</a></code> contains a list
of all parameters that can be created
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  # Create the parameter
  sigma_param &lt;- create_sigma_param()

  # Use the parameter in a distribution
  normal_distr &lt;- create_normal_distr(
    sigma = sigma_param
  )

  # Use the distribution to create a BEAST2 input file
  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = create_yule_tree_prior(
      birth_rate_distr = normal_distr
    )
  )
 file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_site_model'>General function to create a site model.</h2><span id='topic+create_site_model'></span>

<h3>Description</h3>

<p>General function to create a site model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_site_model(name, id, gamma_site_model = create_gamma_site_model(), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_site_model_+3A_name">name</code></td>
<td>
<p>the site model name. Valid
names can be found in <code>get_site_model_names</code></p>
</td></tr>
<tr><td><code id="create_site_model_+3A_id">id</code></td>
<td>
<p>the IDs of the alignment (can be extracted from
the FASTA filename using <code><a href="#topic+get_alignment_id">get_alignment_id</a></code>)</p>
</td></tr>
<tr><td><code id="create_site_model_+3A_gamma_site_model">gamma_site_model</code></td>
<td>
<p>a gamma site model, as created
by <a href="#topic+create_gamma_site_model">create_gamma_site_model</a></p>
</td></tr>
<tr><td><code id="create_site_model_+3A_...">...</code></td>
<td>
<p>specific site model parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a site_model
</p>


<h3>Note</h3>

<p>Prefer using the
named functions
<code><a href="#topic+create_gtr_site_model">create_gtr_site_model</a></code>,
<code><a href="#topic+create_hky_site_model">create_hky_site_model</a></code>,,
<code><a href="#topic+create_jc69_site_model">create_jc69_site_model</a></code>,
and <code><a href="#topic+create_tn93_site_model">create_tn93_site_model</a></code>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+create_gtr_site_model">create_gtr_site_model</a></code> for more examples
with a GTR site model. See <code><a href="#topic+create_hky_site_model">create_hky_site_model</a></code>
for more examples with an HKY site model. See
<code><a href="#topic+create_jc69_site_model">create_jc69_site_model</a></code> for more examples with a JC69
site model. See <code><a href="#topic+create_tn93_site_model">create_tn93_site_model</a></code> for more
examples with a TN93 site model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  check_empty_beautier_folder()

  input_filename &lt;- get_fasta_filename()
  gtr_site_model &lt;- create_gtr_site_model()
  hk_site_model &lt;- create_hky_site_model()
  jc69_site_model &lt;- create_jc69_site_model()
  tn93_site_model &lt;- create_tn93_site_model()

  # Can use any site model
  output_filename &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = input_filename,
    output_filename = output_filename,
    site_model = jc69_site_model
  )
  file.remove(output_filename)

  remove_beautier_folder()
  check_empty_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_site_model_from_name'>Create a site model from name</h2><span id='topic+create_site_model_from_name'></span>

<h3>Description</h3>

<p>Create a site model from name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_site_model_from_name(site_model_name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_site_model_from_name_+3A_site_model_name">site_model_name</code></td>
<td>
<p>name of a site model,
must be a name as returned by <code><a href="#topic+get_site_model_names">get_site_model_names</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a site model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_site_model">create_site_model</a> to create a site model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

site_model &lt;- create_site_model_from_name(get_site_model_names()[1])

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_site_model_parameters_xml'>Internal function to creates the XML text for the
<code>parameter</code>s within the <code>siteModel</code> section
of a BEAST2 parameter file.</h2><span id='topic+create_site_model_parameters_xml'></span>

<h3>Description</h3>

<p>Internal function to creates the XML text for the
<code>parameter</code>s within the <code>siteModel</code> section,
which is part of the <code>siteModel</code> section
of a BEAST2 parameter file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_site_model_parameters_xml(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_site_model_parameters_xml_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>parameter</code>s sections has these elements:
</p>
<pre>
   [parameters]
</pre>
<p><code>[parameters]</code> can be a combination of these:
</p>
<pre>
  &lt;parameter id="mutationRate.s[...]&gt;
  &lt;parameter id="gammaShape.s[...]&gt;
  &lt;parameter id="proportionInvariant.s[...]&gt;
</pre>


<h3>Value</h3>

<p>the site model as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_site_model_xml'>Internal function to creates the XML text for the <code>siteModel</code> tag
of a BEAST2 parameter file.</h2><span id='topic+create_site_model_xml'></span>

<h3>Description</h3>

<p>Creates the XML text for the <code>siteModel</code> tag of
a BEAST2 parameter file,
which is part of the <code>distribution</code> node for the
<code>treeLikelihood</code> ID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_site_model_xml(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_site_model_xml_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>siteModel</code> tag has these elements:
</p>
<pre>
  &lt;siteModel[...]&gt;

      [parameters]

      &lt;substModel[...]&gt;
        [...]
      &lt;/substModel&gt;
  &lt;/siteModel&gt;
</pre>
<p>The <code>parameter</code> section is created by
<a href="#topic+create_site_model_parameters_xml">create_site_model_parameters_xml</a>
The <code>substModel</code> section is created by <a href="#topic+create_subst_model_xml">create_subst_model_xml</a>
</p>


<h3>Value</h3>

<p>the site model as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()
 # &lt;distribution id="posterior"[...]"&gt;
 #     &lt;distribution id="likelihood" [...]&gt;
 #       &lt;siteModel...&gt;
 #         [parameters]
 #       &lt;/siteModel&gt;
 #     &lt;/distribution&gt;
 # &lt;/distribution&gt;

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_site_models'>Creates all supported site models
which is a list of the types returned by
<code><a href="#topic+create_gtr_site_model">create_gtr_site_model</a></code>,
<code><a href="#topic+create_hky_site_model">create_hky_site_model</a></code>,
<code><a href="#topic+create_jc69_site_model">create_jc69_site_model</a></code>
and <code><a href="#topic+create_tn93_site_model">create_tn93_site_model</a></code></h2><span id='topic+create_site_models'></span>

<h3>Description</h3>

<p>Creates all supported site models
which is a list of the types returned by
<code><a href="#topic+create_gtr_site_model">create_gtr_site_model</a></code>,
<code><a href="#topic+create_hky_site_model">create_hky_site_model</a></code>,
<code><a href="#topic+create_jc69_site_model">create_jc69_site_model</a></code>
and <code><a href="#topic+create_tn93_site_model">create_tn93_site_model</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_site_models()
</code></pre>


<h3>Value</h3>

<p>a list of site_models
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_site_model">create_site_model</a> to create a site model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# All created site models are a kind of site model
site_models &lt;- create_site_models()

# TRUE
is_gtr_site_model(site_models[[1]])
is_hky_site_model(site_models[[2]])
is_jc69_site_model(site_models[[3]])
is_tn93_site_model(site_models[[4]])

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_site_models_from_names'>Create site models from their names</h2><span id='topic+create_site_models_from_names'></span>

<h3>Description</h3>

<p>Create site models from their names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_site_models_from_names(site_model_names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_site_models_from_names_+3A_site_model_names">site_model_names</code></td>
<td>
<p>one or more names of a site model,
must be name among those returned by <code><a href="#topic+get_site_model_names">get_site_model_names</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>one or more site models
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_site_model">create_site_model</a> to create a site model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

create_site_models_from_names(get_site_model_names())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_strict_clock_branch_rate_model_xml'>Internal function.</h2><span id='topic+create_strict_clock_branch_rate_model_xml'></span>

<h3>Description</h3>

<p>Internal function to create the <code>branchRateModel</code> section
of the XML as text, for a strict clock model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_strict_clock_branch_rate_model_xml(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_strict_clock_branch_rate_model_xml_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of XML strings
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='create_strict_clock_model'>Create a strict clock model</h2><span id='topic+create_strict_clock_model'></span><span id='topic+create_clock_model_strict'></span>

<h3>Description</h3>

<p>Create a strict clock model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_strict_clock_model(
  id = NA,
  clock_rate_param = create_clock_rate_param(),
  clock_rate_distr = create_uniform_distr(),
  rate_scaler_factor = 0.75
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_strict_clock_model_+3A_id">id</code></td>
<td>
<p>an alignment's IDs.
An ID can be extracted from its FASTA filename
with <code><a href="#topic+get_alignment_ids_from_fasta_filenames">get_alignment_ids_from_fasta_filenames</a></code>)</p>
</td></tr>
<tr><td><code id="create_strict_clock_model_+3A_clock_rate_param">clock_rate_param</code></td>
<td>
<p>the clock rate's parameter,
a numeric value.
For advanced usage, use the structure
as created by the <code><a href="#topic+create_clock_rate_param">create_clock_rate_param</a></code> function</p>
</td></tr>
<tr><td><code id="create_strict_clock_model_+3A_clock_rate_distr">clock_rate_distr</code></td>
<td>
<p>the clock rate's distribution,
as created by a <code><a href="#topic+create_distr">create_distr</a></code> function</p>
</td></tr>
<tr><td><code id="create_strict_clock_model_+3A_rate_scaler_factor">rate_scaler_factor</code></td>
<td>
<p>the strict clock model's operator scaler
for the rate.
Use an empty string to indicate the default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a strict clock_model
</p>


<h3>Note</h3>

<p>I am unsure about the relationship between 'clock_rate_param'
and 'clock_rate_distr'. Please contact me if you know
the most natural architecture
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {
  strict_clock_model &lt;- create_strict_clock_model(
    clock_rate_param = 1.0,
    clock_rate_distr = create_uniform_distr()
  )

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    get_fasta_filename(),
    beast2_input_file,
    clock_model = strict_clock_model
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_strict_clock_rate_scaler_operator_xml'>Internal function</h2><span id='topic+create_strict_clock_rate_scaler_operator_xml'></span>

<h3>Description</h3>

<p>Creates the <code>StrictClockRateScaler</code> operator
such as:
<code>
  ...
</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_strict_clock_rate_scaler_operator_xml(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_strict_clock_rate_scaler_operator_xml_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the following XML:
<code>
  ...
</code>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

create_strict_clock_rate_scaler_operator_xml(
  create_inference_model(
    clock_model = create_strict_clock_model(id = 314)
  )
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_subst_model_xml'>Internal function to create the <code>substModel</code> section</h2><span id='topic+create_subst_model_xml'></span>

<h3>Description</h3>

<p>Internal function to create the <code>substModel</code> section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_subst_model_xml(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_subst_model_xml_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the <code>substModel</code> section as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# Inference model must be initialized
inference_model &lt;- create_inference_model(
  site_model = create_jc69_site_model(id = 123)
)
create_subst_model_xml(
  inference_model = inference_model
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_temp_screenlog_filename'>Create a filename for a temporary 'screenlog' file</h2><span id='topic+create_temp_screenlog_filename'></span>

<h3>Description</h3>

<p>Create a filename for a temporary 'screenlog' file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_temp_screenlog_filename()
</code></pre>


<h3>Value</h3>

<p>a filename for a temporary 'screenlog' file
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <a href="#topic+create_screenlog">create_screenlog</a> to create a screenlog.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

create_temp_screenlog_filename()

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_temp_tracelog_filename'>Create a filename for a temporary 'tracelog' file</h2><span id='topic+create_temp_tracelog_filename'></span>

<h3>Description</h3>

<p>Create a filename for a temporary 'tracelog' file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_temp_tracelog_filename()
</code></pre>


<h3>Value</h3>

<p>a filename for a temporary 'tracelog' file
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <a href="#topic+create_tracelog">create_tracelog</a> to create a tracelog.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

create_temp_tracelog_filename()

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_temp_treelog_filename'>Create a filename for a temporary 'treelog' file</h2><span id='topic+create_temp_treelog_filename'></span>

<h3>Description</h3>

<p>Create a filename for a temporary 'treelog' file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_temp_treelog_filename()
</code></pre>


<h3>Value</h3>

<p>a filename for a temporary 'treelog' file
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <a href="#topic+create_treelog">create_treelog</a> to create a treelog.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

create_temp_treelog_filename()

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_test_inference_model'>Create a testing inference model.</h2><span id='topic+create_test_inference_model'></span>

<h3>Description</h3>

<p>Creates a simple inference model with a short MCMC chain,
to be used in testing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_test_inference_model(
  site_model = create_jc69_site_model(),
  clock_model = create_strict_clock_model(),
  tree_prior = create_yule_tree_prior(),
  mrca_prior = NA,
  mcmc = create_test_mcmc(),
  beauti_options = create_beauti_options(),
  tipdates_filename = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_test_inference_model_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
<tr><td><code id="create_test_inference_model_+3A_clock_model">clock_model</code></td>
<td>
<p>a clock model,
as returned by <code><a href="#topic+create_clock_model">create_clock_model</a></code></p>
</td></tr>
<tr><td><code id="create_test_inference_model_+3A_tree_prior">tree_prior</code></td>
<td>
<p>a tree priors,
as returned by <code><a href="#topic+create_tree_prior">create_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="create_test_inference_model_+3A_mrca_prior">mrca_prior</code></td>
<td>
<p>a Most Recent Common Ancestor prior,
as returned by <code><a href="#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
<tr><td><code id="create_test_inference_model_+3A_mcmc">mcmc</code></td>
<td>
<p>one MCMC.
Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create an MCMC.
Use <code><a href="#topic+create_ns_mcmc">create_ns_mcmc</a></code> to create an MCMC
for a Nested Sampling run.
Use <code><a href="#topic+check_mcmc">check_mcmc</a></code> to check if an MCMC is valid.
Use <code><a href="#topic+rename_mcmc_filenames">rename_mcmc_filenames</a></code> to rename the filenames in an MCMC.</p>
</td></tr>
<tr><td><code id="create_test_inference_model_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
<tr><td><code id="create_test_inference_model_+3A_tipdates_filename">tipdates_filename</code></td>
<td>
<p>name of the file containing the tip dates.
This file is assumed to have two columns, separated by a tab.
The first column contains the taxa names, the second column contains
the date.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an inference model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_inference_model">create_inference_model</a> to create a
regular inference model.
Use <a href="#topic+create_test_ns_inference_model">create_test_ns_inference_model</a> to create an inference model
to estimate the marginal likelihood with a short MCMC, to be
used in testing
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {
  check_empty_beautier_folder()

  inference_model &lt;- create_test_inference_model()

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file_from_model(
    get_fasta_filename(),
    beast2_input_file,
    inference_model = inference_model
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
  check_empty_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_test_mcmc'>Create an MCMC configuration for testing.</h2><span id='topic+create_test_mcmc'></span>

<h3>Description</h3>

<p>Create an MCMC configuration for testing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_test_mcmc(
  chain_length = 3000,
  store_every = 1000,
  pre_burnin = 0,
  n_init_attempts = 10,
  sample_from_prior = FALSE,
  tracelog = create_test_tracelog(),
  screenlog = create_test_screenlog(),
  treelog = create_test_treelog()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_test_mcmc_+3A_chain_length">chain_length</code></td>
<td>
<p>length of the MCMC chain</p>
</td></tr>
<tr><td><code id="create_test_mcmc_+3A_store_every">store_every</code></td>
<td>
<p>number of states the MCMC will process
before the posterior's state will be saved to file.
Use -1 or <code>NA</code> to use the default frequency.</p>
</td></tr>
<tr><td><code id="create_test_mcmc_+3A_pre_burnin">pre_burnin</code></td>
<td>
<p>number of burn in samples taken before entering
the main loop</p>
</td></tr>
<tr><td><code id="create_test_mcmc_+3A_n_init_attempts">n_init_attempts</code></td>
<td>
<p>number of initialization attempts before failing</p>
</td></tr>
<tr><td><code id="create_test_mcmc_+3A_sample_from_prior">sample_from_prior</code></td>
<td>
<p>set to <a href="base.html#topic+TRUE">TRUE</a> to sample from the prior</p>
</td></tr>
<tr><td><code id="create_test_mcmc_+3A_tracelog">tracelog</code></td>
<td>
<p>a <code>tracelog</code>,
as created by <a href="#topic+create_tracelog">create_tracelog</a></p>
</td></tr>
<tr><td><code id="create_test_mcmc_+3A_screenlog">screenlog</code></td>
<td>
<p>a <code>screenlog</code>,
as created by <a href="#topic+create_screenlog">create_screenlog</a></p>
</td></tr>
<tr><td><code id="create_test_mcmc_+3A_treelog">treelog</code></td>
<td>
<p>a <code>treelog</code>,
as created by <a href="#topic+create_treelog">create_treelog</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>an MCMC configuration
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create a default BEAST2 MCMC
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  # Create an MCMC chain with 50 states
  mcmc &lt;- create_test_mcmc()

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    get_fasta_filename(),
    beast2_input_file,
    mcmc = mcmc
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_test_ns_inference_model'>Create an inference model to be tested by Nested Sampling</h2><span id='topic+create_test_ns_inference_model'></span>

<h3>Description</h3>

<p>Create an inference model to be tested by Nested Sampling
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_test_ns_inference_model(
  site_model = create_jc69_site_model(),
  clock_model = create_strict_clock_model(),
  tree_prior = create_yule_tree_prior(),
  mcmc = create_test_ns_mcmc()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_test_ns_inference_model_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
<tr><td><code id="create_test_ns_inference_model_+3A_clock_model">clock_model</code></td>
<td>
<p>a clock model,
as returned by <code><a href="#topic+create_clock_model">create_clock_model</a></code></p>
</td></tr>
<tr><td><code id="create_test_ns_inference_model_+3A_tree_prior">tree_prior</code></td>
<td>
<p>a tree priors,
as returned by <code><a href="#topic+create_tree_prior">create_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="create_test_ns_inference_model_+3A_mcmc">mcmc</code></td>
<td>
<p>one MCMC.
Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create an MCMC.
Use <code><a href="#topic+create_ns_mcmc">create_ns_mcmc</a></code> to create an MCMC
for a Nested Sampling run.
Use <code><a href="#topic+check_mcmc">check_mcmc</a></code> to check if an MCMC is valid.
Use <code><a href="#topic+rename_mcmc_filenames">rename_mcmc_filenames</a></code> to rename the filenames in an MCMC.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an inference model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_test_inference_model">create_test_inference_model</a> to create a
regular inference model with a short MCMC, to be used in testing.
Use <a href="#topic+create_ns_inference_model">create_ns_inference_model</a> to create an inference model
to estimate the marginal likelihood.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

inference_model &lt;- create_test_ns_inference_model()

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_test_ns_mcmc'>Create an NS MCMC object for testing</h2><span id='topic+create_test_ns_mcmc'></span>

<h3>Description</h3>

<p>Create an NS MCMC object for testing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_test_ns_mcmc(
  chain_length = 2000,
  store_every = 1000,
  pre_burnin = 0,
  n_init_attempts = 3,
  particle_count = 1,
  sub_chain_length = 500,
  epsilon = 1e-12,
  tracelog = create_test_tracelog(),
  screenlog = create_test_screenlog(),
  treelog = create_test_treelog()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_test_ns_mcmc_+3A_chain_length">chain_length</code></td>
<td>
<p>upper bound to the length of the MCMC chain</p>
</td></tr>
<tr><td><code id="create_test_ns_mcmc_+3A_store_every">store_every</code></td>
<td>
<p>number of states the MCMC will process
before the posterior's state will be saved to file.
Use -1 or <code>NA</code> to use the default frequency.</p>
</td></tr>
<tr><td><code id="create_test_ns_mcmc_+3A_pre_burnin">pre_burnin</code></td>
<td>
<p>number of burn in samples taken before entering
the main loop</p>
</td></tr>
<tr><td><code id="create_test_ns_mcmc_+3A_n_init_attempts">n_init_attempts</code></td>
<td>
<p>number of initialization attempts before failing</p>
</td></tr>
<tr><td><code id="create_test_ns_mcmc_+3A_particle_count">particle_count</code></td>
<td>
<p>number of particles</p>
</td></tr>
<tr><td><code id="create_test_ns_mcmc_+3A_sub_chain_length">sub_chain_length</code></td>
<td>
<p>sub-chain length</p>
</td></tr>
<tr><td><code id="create_test_ns_mcmc_+3A_epsilon">epsilon</code></td>
<td>
<p>epsilon</p>
</td></tr>
<tr><td><code id="create_test_ns_mcmc_+3A_tracelog">tracelog</code></td>
<td>
<p>a <code>tracelog</code>,
as created by <a href="#topic+create_tracelog">create_tracelog</a></p>
</td></tr>
<tr><td><code id="create_test_ns_mcmc_+3A_screenlog">screenlog</code></td>
<td>
<p>a <code>screenlog</code>,
as created by <a href="#topic+create_screenlog">create_screenlog</a></p>
</td></tr>
<tr><td><code id="create_test_ns_mcmc_+3A_treelog">treelog</code></td>
<td>
<p>a <code>treelog</code>,
as created by <a href="#topic+create_treelog">create_treelog</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>an MCMC object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+create_ns_mcmc">create_ns_mcmc</a></code> to create a default
nested sampling MCMC
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  mcmc &lt;- create_test_ns_mcmc()
  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    get_fasta_filename(),
    beast2_input_file,
    mcmc = mcmc
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_test_screenlog'>Create a <code>screenlog</code> object, to be used in testing</h2><span id='topic+create_test_screenlog'></span>

<h3>Description</h3>

<p>Create a <code>screenlog</code> object, to be used in testing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_test_screenlog(
  filename = create_temp_screenlog_filename(),
  log_every = 1000,
  mode = "autodetect",
  sanitise_headers = FALSE,
  sort = "none"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_test_screenlog_+3A_filename">filename</code></td>
<td>
<p>name of the file to store the posterior screens
phylogenies to. By default, this is <code>$(screen).screens</code></p>
</td></tr>
<tr><td><code id="create_test_screenlog_+3A_log_every">log_every</code></td>
<td>
<p>number of MCMC states between writing to file</p>
</td></tr>
<tr><td><code id="create_test_screenlog_+3A_mode">mode</code></td>
<td>
<p>mode how to log.
Valid values are the ones returned by <a href="#topic+get_log_modes">get_log_modes</a></p>
</td></tr>
<tr><td><code id="create_test_screenlog_+3A_sanitise_headers">sanitise_headers</code></td>
<td>
<p>set to <a href="base.html#topic+TRUE">TRUE</a> to sanitise the headers of the
log file</p>
</td></tr>
<tr><td><code id="create_test_screenlog_+3A_sort">sort</code></td>
<td>
<p>how to sort the log.
Valid values are the ones returned by <a href="#topic+get_log_sorts">get_log_sorts</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>screenlog</code> object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>create_test_screenlog()
</code></pre>

<hr>
<h2 id='create_test_tracelog'>Create a <code>tracelog</code> object, as used for testing</h2><span id='topic+create_test_tracelog'></span>

<h3>Description</h3>

<p>Create a <code>tracelog</code> object, as used for testing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_test_tracelog(
  filename = create_temp_tracelog_filename(),
  log_every = 1000,
  mode = "autodetect",
  sanitise_headers = TRUE,
  sort = "smart"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_test_tracelog_+3A_filename">filename</code></td>
<td>
<p>name of the file to store the posterior traces.
Use <a href="base.html#topic+NA">NA</a> to use the filename <code>[alignment_id].log</code>,
where <code>alignment_id</code> is obtained using <a href="#topic+get_alignment_id">get_alignment_id</a></p>
</td></tr>
<tr><td><code id="create_test_tracelog_+3A_log_every">log_every</code></td>
<td>
<p>number of MCMC states between writing to file</p>
</td></tr>
<tr><td><code id="create_test_tracelog_+3A_mode">mode</code></td>
<td>
<p>mode how to log.
Valid values are the ones returned by <a href="#topic+get_log_modes">get_log_modes</a></p>
</td></tr>
<tr><td><code id="create_test_tracelog_+3A_sanitise_headers">sanitise_headers</code></td>
<td>
<p>set to <a href="base.html#topic+TRUE">TRUE</a> to sanitise the headers of the
log file</p>
</td></tr>
<tr><td><code id="create_test_tracelog_+3A_sort">sort</code></td>
<td>
<p>how to sort the log.
Valid values are the ones returned by <a href="#topic+get_log_sorts">get_log_sorts</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>tracelog</code> object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>create_test_tracelog()
</code></pre>

<hr>
<h2 id='create_test_treelog'>Create a <code>treelog</code> object to be used in testing</h2><span id='topic+create_test_treelog'></span>

<h3>Description</h3>

<p>Create a <code>treelog</code> object to be used in testing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_test_treelog(
  filename = create_temp_treelog_filename(),
  log_every = 1000,
  mode = "tree",
  sanitise_headers = FALSE,
  sort = "none"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_test_treelog_+3A_filename">filename</code></td>
<td>
<p>name of the file to store the posterior trees</p>
</td></tr>
<tr><td><code id="create_test_treelog_+3A_log_every">log_every</code></td>
<td>
<p>number of MCMC states between writing to file</p>
</td></tr>
<tr><td><code id="create_test_treelog_+3A_mode">mode</code></td>
<td>
<p>mode how to log.
Valid values are the ones returned by <a href="#topic+get_log_modes">get_log_modes</a></p>
</td></tr>
<tr><td><code id="create_test_treelog_+3A_sanitise_headers">sanitise_headers</code></td>
<td>
<p>set to <a href="base.html#topic+TRUE">TRUE</a> to sanitise the headers of the
log file</p>
</td></tr>
<tr><td><code id="create_test_treelog_+3A_sort">sort</code></td>
<td>
<p>how to sort the log.
Valid values are the ones returned by <a href="#topic+get_log_sorts">get_log_sorts</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>treelog</code> object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>create_test_treelog()
</code></pre>

<hr>
<h2 id='create_tn93_site_model'>Create a TN93 site model</h2><span id='topic+create_tn93_site_model'></span><span id='topic+create_site_model_tn93'></span>

<h3>Description</h3>

<p>Create a TN93 site model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_tn93_site_model(
  id = NA,
  gamma_site_model = create_gamma_site_model(),
  kappa_1_param = create_kappa_1_param(),
  kappa_2_param = create_kappa_2_param(),
  kappa_1_prior_distr = create_log_normal_distr(m = 1, s = 1.25),
  kappa_2_prior_distr = create_log_normal_distr(m = 1, s = 1.25),
  freq_equilibrium = "estimated",
  freq_param = create_freq_param()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_tn93_site_model_+3A_id">id</code></td>
<td>
<p>the IDs of the alignment (can be extracted from
the FASTA filename using <code><a href="#topic+get_alignment_id">get_alignment_id</a></code>)</p>
</td></tr>
<tr><td><code id="create_tn93_site_model_+3A_gamma_site_model">gamma_site_model</code></td>
<td>
<p>a gamma site model, as created
by <a href="#topic+create_gamma_site_model">create_gamma_site_model</a></p>
</td></tr>
<tr><td><code id="create_tn93_site_model_+3A_kappa_1_param">kappa_1_param</code></td>
<td>
<p>the 'kappa 1' parameter,
a numeric value.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_kappa_1_param">create_kappa_1_param</a></code></p>
</td></tr>
<tr><td><code id="create_tn93_site_model_+3A_kappa_2_param">kappa_2_param</code></td>
<td>
<p>the 'kappa 2' parameter,
a numeric value.
For advanced usage, use the structure
as returned by <code><a href="#topic+create_kappa_2_param">create_kappa_2_param</a></code></p>
</td></tr>
<tr><td><code id="create_tn93_site_model_+3A_kappa_1_prior_distr">kappa_1_prior_distr</code></td>
<td>
<p>the distribution of the kappa 1 prior,
which is a log-normal distribution
(as created by <code><a href="#topic+create_log_normal_distr">create_log_normal_distr</a></code>)
by default</p>
</td></tr>
<tr><td><code id="create_tn93_site_model_+3A_kappa_2_prior_distr">kappa_2_prior_distr</code></td>
<td>
<p>the distribution of the kappa 2 prior,
which is a log-normal distribution
(as created by <code><a href="#topic+create_log_normal_distr">create_log_normal_distr</a></code>)
by default</p>
</td></tr>
<tr><td><code id="create_tn93_site_model_+3A_freq_equilibrium">freq_equilibrium</code></td>
<td>
<p>the frequency in which the rates are at equilibrium
are either <code>estimated</code>, <code>empirical</code> or <code>all_equal</code>.
<code>get_freq_equilibrium_names</code> returns the possible values
for <code>freq_equilibrium</code></p>
</td></tr>
<tr><td><code id="create_tn93_site_model_+3A_freq_param">freq_param</code></td>
<td>
<p>a 'freq' parameter,
as created by <a href="#topic+create_freq_param">create_freq_param</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a TN93 site_model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  tn93_site_model &lt;- create_tn93_site_model(
    kappa_1_param = 2.0,
    kappa_2_param = 2.0
  )

  output_filename &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    output_filename = output_filename,
    site_model = tn93_site_model
  )
  file.remove(output_filename)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_tn93_subst_model_xml'>Converts a TN93 site model to XML,
used in the <code>substModel</code> section</h2><span id='topic+create_tn93_subst_model_xml'></span>

<h3>Description</h3>

<p>Converts a TN93 site model to XML,
used in the <code>substModel</code> section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_tn93_subst_model_xml(site_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_tn93_subst_model_xml_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the site model as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='create_tracelog'>Create a <code>tracelog</code> object</h2><span id='topic+create_tracelog'></span>

<h3>Description</h3>

<p>Create a <code>tracelog</code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_tracelog(
  filename = NA,
  log_every = 1000,
  mode = "autodetect",
  sanitise_headers = TRUE,
  sort = "smart"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_tracelog_+3A_filename">filename</code></td>
<td>
<p>name of the file to store the posterior traces.
Use <a href="base.html#topic+NA">NA</a> to use the filename <code>[alignment_id].log</code>,
where <code>alignment_id</code> is obtained using <a href="#topic+get_alignment_id">get_alignment_id</a></p>
</td></tr>
<tr><td><code id="create_tracelog_+3A_log_every">log_every</code></td>
<td>
<p>number of MCMC states between writing to file</p>
</td></tr>
<tr><td><code id="create_tracelog_+3A_mode">mode</code></td>
<td>
<p>mode how to log.
Valid values are the ones returned by <a href="#topic+get_log_modes">get_log_modes</a></p>
</td></tr>
<tr><td><code id="create_tracelog_+3A_sanitise_headers">sanitise_headers</code></td>
<td>
<p>set to <a href="base.html#topic+TRUE">TRUE</a> to sanitise the headers of the
log file</p>
</td></tr>
<tr><td><code id="create_tracelog_+3A_sort">sort</code></td>
<td>
<p>how to sort the log.
Valid values are the ones returned by <a href="#topic+get_log_sorts">get_log_sorts</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>tracelog</code> object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>create_tracelog()
</code></pre>

<hr>
<h2 id='create_tracelog_xml'>Internal function</h2><span id='topic+create_tracelog_xml'></span>

<h3>Description</h3>

<p>Creates the <code>tracelog</code> section of the <code>logger</code> section
of a BEAST2 XML parameter file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_tracelog_xml(input_filename, inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_tracelog_xml_+3A_input_filename">input_filename</code></td>
<td>
<p>A FASTA filename.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.</p>
</td></tr>
<tr><td><code id="create_tracelog_xml_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='create_trait_set_string'>Create a trait set string.</h2><span id='topic+create_trait_set_string'></span>

<h3>Description</h3>

<p>For example, a data frame with row <code>A 1</code>
and another row <code>B 2</code>, the trait set string will be
<code>A=1,B=2</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_trait_set_string(df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_trait_set_string_+3A_df">df</code></td>
<td>
<p>a data frame with two columns</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the trait set string
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='create_tree_likelihood_distr_xml'>Creates the XML text for the <code>distribution</code> tag
with the <code>treeLikelihood</code> ID,
of a BEAST2 parameter file.</h2><span id='topic+create_tree_likelihood_distr_xml'></span>

<h3>Description</h3>

<p>Creates the XML text for the <code>distribution</code> tag
with the <code>treeLikelihood</code> ID,
of a BEAST2 parameter file,
in an unindented form
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_tree_likelihood_distr_xml(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_tree_likelihood_distr_xml_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>distribution</code> tag (with ID equals <code>treeLikelihood</code>)
has these elements:
</p>
<pre>
   &lt;distribution id="treeLikelihood"[...]&gt;
      &lt;siteModel[...]&gt;
        [...]
      &lt;/siteModel&gt;
      &lt;branchRateModel[...]&gt;
        [...]
      &lt;/branchRateModel&gt;
   &lt;/distribution&gt;
</pre>
<p>The <code>siteModel</code> section
is created by <a href="#topic+create_site_model_xml">create_site_model_xml</a>.
The <code>branchRateModel</code> section
is created by <a href="#topic+create_branch_rate_model_xml">create_branch_rate_model_xml</a>.
</p>
<p>Zooming out:
</p>
<pre>
  &lt;beast[...]&gt;
    &lt;run[...]&gt;
      &lt;distribution id="posterior"[...]&gt;
        &lt;distribution id="likelihood"[...]&gt;
          [this section]
        &lt;/distribution&gt;
      &lt;/distribution&gt;
    &lt;/run&gt;
  &lt;/beast&gt;
</pre>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Note</h3>

<p>this function is not intended for regular use, thus its
long name length is accepted
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>this function is called by <code>create_beast2_input_distr</code>,
together with <code>create_beast2_input_distr_prior</code>
</p>

<hr>
<h2 id='create_tree_prior'>Internal function to create a tree prior</h2><span id='topic+create_tree_prior'></span>

<h3>Description</h3>

<p>Internal function to create a tree prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_tree_prior(name, id, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_tree_prior_+3A_name">name</code></td>
<td>
<p>the tree prior name. Can be any name
in <code>get_tree_prior_names</code></p>
</td></tr>
<tr><td><code id="create_tree_prior_+3A_id">id</code></td>
<td>
<p>the ID of the alignment</p>
</td></tr>
<tr><td><code id="create_tree_prior_+3A_...">...</code></td>
<td>
<p>specific tree prior parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a tree_prior
</p>


<h3>Note</h3>

<p>Prefer the use the named functions
<code><a href="#topic+create_bd_tree_prior">create_bd_tree_prior</a></code>,
<code><a href="#topic+create_cbs_tree_prior">create_cbs_tree_prior</a></code>,
<code><a href="#topic+create_ccp_tree_prior">create_ccp_tree_prior</a></code>
<code><a href="#topic+create_cep_tree_prior">create_cep_tree_prior</a></code>
and <code><a href="#topic+create_yule_tree_prior">create_yule_tree_prior</a></code>
instead
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>See
<code><a href="#topic+create_bd_tree_prior">create_bd_tree_prior</a></code>,
<code><a href="#topic+create_cbs_tree_prior">create_cbs_tree_prior</a></code>,
<code><a href="#topic+create_ccp_tree_prior">create_ccp_tree_prior</a></code>
<code><a href="#topic+create_cep_tree_prior">create_cep_tree_prior</a></code>
and <code><a href="#topic+create_yule_tree_prior">create_yule_tree_prior</a></code>
for more examples using those functions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  check_empty_beautier_folder()

  beast2_input_file &lt;- get_beautier_tempfilename()

  bd_tree_prior &lt;- create_bd_tree_prior()
  cbs_tree_prior &lt;- create_cbs_tree_prior()
  ccp_tree_prior &lt;- create_ccp_tree_prior()
  cep_tree_prior &lt;- create_cep_tree_prior()
  yule_tree_prior &lt;- create_yule_tree_prior()

  # Use any of the above tree priors
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = bd_tree_prior
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
  check_empty_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_tree_priors'>Creates all supported tree priors,
which is a list of the types returned by
<code><a href="#topic+create_bd_tree_prior">create_bd_tree_prior</a></code>,
<code><a href="#topic+create_cbs_tree_prior">create_cbs_tree_prior</a></code>,
<code><a href="#topic+create_ccp_tree_prior">create_ccp_tree_prior</a></code>,
<code><a href="#topic+create_cep_tree_prior">create_cep_tree_prior</a></code>
and <code><a href="#topic+create_yule_tree_prior">create_yule_tree_prior</a></code></h2><span id='topic+create_tree_priors'></span>

<h3>Description</h3>

<p>Creates all supported tree priors,
which is a list of the types returned by
<code><a href="#topic+create_bd_tree_prior">create_bd_tree_prior</a></code>,
<code><a href="#topic+create_cbs_tree_prior">create_cbs_tree_prior</a></code>,
<code><a href="#topic+create_ccp_tree_prior">create_ccp_tree_prior</a></code>,
<code><a href="#topic+create_cep_tree_prior">create_cep_tree_prior</a></code>
and <code><a href="#topic+create_yule_tree_prior">create_yule_tree_prior</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_tree_priors()
</code></pre>


<h3>Value</h3>

<p>a list of tree_priors
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

tree_priors &lt;- create_tree_priors()
# TRUE
is_bd_tree_prior(tree_priors[[1]])
is_cbs_tree_prior(tree_priors[[2]])
is_ccp_tree_prior(tree_priors[[3]])
is_cep_tree_prior(tree_priors[[4]])
is_yule_tree_prior(tree_priors[[5]])

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_treelog'>Create a <code>treelog</code> object</h2><span id='topic+create_treelog'></span>

<h3>Description</h3>

<p>Create a <code>treelog</code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_treelog(
  filename = "$(tree).trees",
  log_every = 1000,
  mode = "tree",
  sanitise_headers = FALSE,
  sort = "none"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_treelog_+3A_filename">filename</code></td>
<td>
<p>name of the file to store the posterior trees</p>
</td></tr>
<tr><td><code id="create_treelog_+3A_log_every">log_every</code></td>
<td>
<p>number of MCMC states between writing to file</p>
</td></tr>
<tr><td><code id="create_treelog_+3A_mode">mode</code></td>
<td>
<p>mode how to log.
Valid values are the ones returned by <a href="#topic+get_log_modes">get_log_modes</a></p>
</td></tr>
<tr><td><code id="create_treelog_+3A_sanitise_headers">sanitise_headers</code></td>
<td>
<p>set to <a href="base.html#topic+TRUE">TRUE</a> to sanitise the headers of the
log file</p>
</td></tr>
<tr><td><code id="create_treelog_+3A_sort">sort</code></td>
<td>
<p>how to sort the log.
Valid values are the ones returned by <a href="#topic+get_log_sorts">get_log_sorts</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a 'treelog', as can be checked by <a href="#topic+check_treelog">check_treelog</a>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

create_treelog()

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_treelog_xml'>Creates the XML text for the 'logger' tag with ID 'treelog'.
This section has these elements:
&ldquo;'
&lt;logger id=&quot;treelog.t:test_output_0&quot; spec=&quot;Logger&quot; fileName=&quot;my_treelog.trees&quot; logEvery=&quot;345000&quot; mode=&quot;tree&quot; sanitiseHeaders=&quot;true&quot; sort=&quot;smart&quot;&gt; # nolint indeed long
&lt;log id=&quot;TreeWithMetaDataLogger.t:test_output_0&quot; spec=&quot;beast.evolution.tree.TreeWithMetaDataLogger&quot; tree=&quot;@Tree.t:test_output_0&quot;/&gt; # nolint indeed long
&lt;/logger&gt;
&ldquo;'</h2><span id='topic+create_treelog_xml'></span>

<h3>Description</h3>

<p>Creates the XML text for the 'logger' tag with ID 'treelog'.
This section has these elements:
&ldquo;'
&lt;logger id=&quot;treelog.t:test_output_0&quot; spec=&quot;Logger&quot; fileName=&quot;my_treelog.trees&quot; logEvery=&quot;345000&quot; mode=&quot;tree&quot; sanitiseHeaders=&quot;true&quot; sort=&quot;smart&quot;&gt; # nolint indeed long
&lt;log id=&quot;TreeWithMetaDataLogger.t:test_output_0&quot; spec=&quot;beast.evolution.tree.TreeWithMetaDataLogger&quot; tree=&quot;@Tree.t:test_output_0&quot;/&gt; # nolint indeed long
&lt;/logger&gt;
&ldquo;'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_treelog_xml(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_treelog_xml_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='create_ucld_mean_state_node_param_xml'>Internal function</h2><span id='topic+create_ucld_mean_state_node_param_xml'></span>

<h3>Description</h3>

<p>Creates the <code>ucldMean.c</code> parameter with the name <code>stateNode</code>,
such as:
<code>
  &lt;parameter id=\"ucldMean.c:[id]\" spec=\"parameter.RealParameter\"
    name=\"stateNode\"&gt;1.0&lt;/parameter&gt;
</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_ucld_mean_state_node_param_xml(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_ucld_mean_state_node_param_xml_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the XML
<code>
  &lt;parameter id=\"ucldMean.c:[id]\" spec=\"parameter.RealParameter\"
    name=\"stateNode\"&gt;1.0&lt;/parameter&gt;
</code>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

create_ucld_mean_state_node_param_xml(
  create_inference_model(
    clock_model = create_rln_clock_model(id = 314),
    beauti_options = create_beauti_options_v2_6()
  )
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_ucld_stdev_state_node_param_xml'>Internal function</h2><span id='topic+create_ucld_stdev_state_node_param_xml'></span>

<h3>Description</h3>

<p>Creates the <code>ucldStdev</code> parameter with the name <code>stateNode</code>,
such as:
<code>
  &lt;parameter id="ucldStdev.c:[id]" [...] name="stateNode"&gt;0.1&lt;/parameter&gt;
</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_ucld_stdev_state_node_param_xml(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_ucld_stdev_state_node_param_xml_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the following XML:
<code>
  &lt;parameter id="ucldStdev.c:[id]" lower="0.0" name="stateNode"&gt;
    0.1
  &lt;/parameter&gt;
</code>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

create_ucld_stdev_state_node_param_xml(
  create_inference_model(
    clock_model = create_rln_clock_model(id = 314)
  )
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_uniform_distr'>Create a uniform distribution</h2><span id='topic+create_uniform_distr'></span><span id='topic+create_distr_uniform'></span>

<h3>Description</h3>

<p>Create a uniform distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_uniform_distr(id = NA, value = NA, lower = NA, upper = Inf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_uniform_distr_+3A_id">id</code></td>
<td>
<p>the distribution's ID</p>
</td></tr>
<tr><td><code id="create_uniform_distr_+3A_value">value</code></td>
<td>
<p>the initial value for the MCMC</p>
</td></tr>
<tr><td><code id="create_uniform_distr_+3A_lower">lower</code></td>
<td>
<p>the lower bound, the lowest possible value</p>
</td></tr>
<tr><td><code id="create_uniform_distr_+3A_upper">upper</code></td>
<td>
<p>an upper limit of the uniform distribution.
If the upper limits needs to be infinity, set <code>upper</code> to <code>Inf</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a uniform distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the function <code><a href="#topic+create_distr">create_distr</a></code> shows an overview
of all supported distributions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  uniform_distr &lt;- create_uniform_distr()

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
    input_filename = get_fasta_filename(),
    beast2_input_file,
    tree_prior = create_yule_tree_prior(
      birth_rate_distr = uniform_distr
    )
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='create_xml_declaration'>Create the XML declaration of the BEAST2 XML input file</h2><span id='topic+create_xml_declaration'></span>

<h3>Description</h3>

<p>Create the XML declaration of the BEAST2 XML input file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_xml_declaration()
</code></pre>


<h3>Value</h3>

<p>one line of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

create_xml_declaration()

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='create_yule_tree_prior'>Create a Yule tree prior</h2><span id='topic+create_yule_tree_prior'></span><span id='topic+create_tree_prior_yule'></span>

<h3>Description</h3>

<p>Create a Yule tree prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_yule_tree_prior(
  id = NA,
  birth_rate_distr = create_uniform_distr()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_yule_tree_prior_+3A_id">id</code></td>
<td>
<p>the ID of the alignment</p>
</td></tr>
<tr><td><code id="create_yule_tree_prior_+3A_birth_rate_distr">birth_rate_distr</code></td>
<td>
<p>the birth rate distribution,
as created by a <code><a href="#topic+create_distr">create_distr</a></code> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Yule tree_prior
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>An alignment ID can be extracted from
its FASTA filename using <code><a href="#topic+get_alignment_id">get_alignment_id</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  yule_tree_prior &lt;- create_yule_tree_prior()

  beast2_input_file &lt;- get_beautier_tempfilename()
  create_beast2_input_file(
     input_filename = get_fasta_filename(),
     beast2_input_file,
     tree_prior = yule_tree_prior
  )
  file.remove(beast2_input_file)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='default_parameters_doc'>Documentation of parameters (for example, <code>create_param</code>.
This function does nothing. It is intended to inherit documentation from.</h2><span id='topic+default_parameters_doc'></span>

<h3>Description</h3>

<p>Documentation of parameters (for example, <code>create_param</code>.
This function does nothing. It is intended to inherit documentation from.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>default_parameters_doc(dimension, estimate, id, lower, name, upper, value, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="default_parameters_doc_+3A_dimension">dimension</code></td>
<td>
<p>the number of dimensions, for example, as used in
<a href="#topic+create_freq_param">create_freq_param</a></p>
</td></tr>
<tr><td><code id="default_parameters_doc_+3A_estimate">estimate</code></td>
<td>
<p>TRUE if this parameter is to be estimated by BEAST2,
FALSE otherwise</p>
</td></tr>
<tr><td><code id="default_parameters_doc_+3A_id">id</code></td>
<td>
<p>the parameter's ID</p>
</td></tr>
<tr><td><code id="default_parameters_doc_+3A_lower">lower</code></td>
<td>
<p>lowest possible value of the parameter. If the parameter
is estimated, <code>lower</code> must be less than <code>value</code></p>
</td></tr>
<tr><td><code id="default_parameters_doc_+3A_name">name</code></td>
<td>
<p>the parameters' name. Valid
names can be found in <code>get_param_names</code></p>
</td></tr>
<tr><td><code id="default_parameters_doc_+3A_upper">upper</code></td>
<td>
<p>upper value of the parameter</p>
</td></tr>
<tr><td><code id="default_parameters_doc_+3A_value">value</code></td>
<td>
<p>value of the parameter</p>
</td></tr>
<tr><td><code id="default_parameters_doc_+3A_...">...</code></td>
<td>
<p>specific parameter parameters</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This is an internal function, so it should be marked with
<code>@export</code>. This is not done, as this will disallow all
functions to find the documentation parameters
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='default_params_doc'>Documentation of general function arguments.
This function does nothing.
It is intended to inherit function argument documentation.</h2><span id='topic+default_params_doc'></span>

<h3>Description</h3>

<p>Documentation of general function arguments.
This function does nothing.
It is intended to inherit function argument documentation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>default_params_doc(
  alignment_id,
  allow_empty_str,
  allow_na,
  alpha_parameter,
  b_pop_sizes_param,
  b_pop_sizes_parameter,
  bd_tree_prior,
  beautier_folder,
  cbs_tree_prior,
  beast2_version,
  beauti_options,
  beta_parameter,
  ccp_tree_prior,
  cep_tree_prior,
  chain_length,
  clock_model,
  clock_model_name,
  clock_model_names,
  clock_models,
  clock_prior_distr_id,
  clock_rate_param,
  crown_age,
  crown_ages,
  distr_id,
  fasta_filename,
  fasta_filenames,
  filename,
  fixed_crown_age,
  fixed_crown_ages,
  freq_param,
  gamma_distr,
  gamma_site_model,
  group_sizes_dimension,
  gtr_site_model,
  has_non_strict_clock_model,
  has_tip_dating,
  hky_site_model,
  id,
  ids,
  inference_model,
  inference_models,
  initial_phylogenies,
  input_filename,
  input_filenames,
  is_monophyletic,
  jc69_site_model,
  kappa_param,
  log_every,
  m_param,
  mcmc,
  mode,
  mrca_prior,
  mrca_priors,
  mrca_prior_name,
  n_init_attempts,
  output_filename,
  param,
  param_id,
  phylogeny,
  pop_sizes_scaler_scale_factor,
  pre_burnin,
  rate_scaler_factor,
  rename_fun,
  rln_clock_model,
  sample_from_prior,
  sanitise_headers,
  screenlog,
  sequence_length,
  site_model,
  site_model_name,
  site_model_names,
  site_models,
  sort,
  store_every,
  strict_clock_model,
  taxa_names,
  tipdates_filename,
  tn93_site_model,
  tracelog,
  treelog,
  tree_prior,
  tree_prior_name,
  tree_prior_names,
  tree_priors,
  verbose,
  yule_tree_prior
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="default_params_doc_+3A_alignment_id">alignment_id</code></td>
<td>
<p>ID of the alignment,
as returned by <a href="#topic+get_alignment_id">get_alignment_id</a>.
Keep at <code>NA</code> to have it initialized automatically</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_allow_empty_str">allow_empty_str</code></td>
<td>
<p>allow a string to be empty</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_allow_na">allow_na</code></td>
<td>
<p>allow <a href="base.html#topic+NA">NA</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_alpha_parameter">alpha_parameter</code></td>
<td>
<p>an alpha parameter,
as created by <a href="#topic+create_alpha_param">create_alpha_param</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_b_pop_sizes_param">b_pop_sizes_param</code></td>
<td>
<p>a Bayesian population size parameter,
as created by <a href="#topic+create_b_pop_sizes_param">create_b_pop_sizes_param</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_b_pop_sizes_parameter">b_pop_sizes_parameter</code></td>
<td>
<p>a Bayesian population size parameter,
as created by <a href="#topic+create_b_pop_sizes_param">create_b_pop_sizes_param</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_bd_tree_prior">bd_tree_prior</code></td>
<td>
<p>a Birth-Death tree prior, as created
by <code><a href="#topic+create_bd_tree_prior">create_bd_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_beautier_folder">beautier_folder</code></td>
<td>
<p>the path to
the <a href="#topic+beautier">beautier</a> temporary files folder</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_cbs_tree_prior">cbs_tree_prior</code></td>
<td>
<p>a Coalescent Bayesian Skyline tree prior,
as returned by <code><a href="#topic+create_cbs_tree_prior">create_cbs_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_beast2_version">beast2_version</code></td>
<td>
<p>BEAST2 version, for example, <code>"2.5"</code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_beta_parameter">beta_parameter</code></td>
<td>
<p>a beta parameter,
as created by <a href="#topic+create_beta_param">create_beta_param</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_ccp_tree_prior">ccp_tree_prior</code></td>
<td>
<p>a Coalescent Constant Population tree prior,
as returned by <code><a href="#topic+create_ccp_tree_prior">create_ccp_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_cep_tree_prior">cep_tree_prior</code></td>
<td>
<p>a Coalescent Exponential Population tree prior,
as returned by <code><a href="#topic+create_cep_tree_prior">create_cep_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_chain_length">chain_length</code></td>
<td>
<p>length of the MCMC chain</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_clock_model">clock_model</code></td>
<td>
<p>a clock model,
as returned by <code><a href="#topic+create_clock_model">create_clock_model</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_clock_model_name">clock_model_name</code></td>
<td>
<p>name of a clock model,
must be a name as returned by <code><a href="#topic+get_clock_model_names">get_clock_model_names</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_clock_model_names">clock_model_names</code></td>
<td>
<p>one or more names of a clock model,
must be name among those returned by <code><a href="#topic+get_clock_model_names">get_clock_model_names</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_clock_models">clock_models</code></td>
<td>
<p>a list of one or more clock models,
as returned by <code><a href="#topic+create_clock_model">create_clock_model</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_clock_prior_distr_id">clock_prior_distr_id</code></td>
<td>
<p>ID of an MRCA clock model's distribution.
Keep at <code>NA</code> to have it initialized automatically</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_clock_rate_param">clock_rate_param</code></td>
<td>
<p>a <code>clockRate</code> parameter,
a numeric value,
as created by <a href="#topic+create_clock_rate_param">create_clock_rate_param</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_crown_age">crown_age</code></td>
<td>
<p>the crown age of the phylogeny</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_crown_ages">crown_ages</code></td>
<td>
<p>the crown ages of the phylogenies. Set to NA
if the crown age needs to be estimated</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_distr_id">distr_id</code></td>
<td>
<p>a distributions' ID</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>a FASTA filename.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.
Note that BEAST2 also supports missing data,
by using a dash (<code>-</code>) or question mark (<code>?</code>)
as a sequence.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_fasta_filenames">fasta_filenames</code></td>
<td>
<p>One or more FASTA filenames.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_filename">filename</code></td>
<td>
<p>a filename, as can be checked by <a href="#topic+check_filename">check_filename</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_fixed_crown_age">fixed_crown_age</code></td>
<td>
<p>determines if the phylogeny's crown age is
fixed. If FALSE, crown age is estimated by BEAST2. If TRUE,
the crown age is fixed to the crown age
of the initial phylogeny.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_fixed_crown_ages">fixed_crown_ages</code></td>
<td>
<p>one or more booleans to determine if the
phylogenies' crown ages are fixed.
If FALSE, crown age is estimated by BEAST2. If TRUE,
the crown age is fixed to the crown age
of the initial phylogeny.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_freq_param">freq_param</code></td>
<td>
<p>a 'freq' parameter,
as created by <a href="#topic+create_freq_param">create_freq_param</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_gamma_distr">gamma_distr</code></td>
<td>
<p>a gamma distribution,
as created by <code><a href="#topic+create_gamma_distr">create_gamma_distr</a></code>)</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_gamma_site_model">gamma_site_model</code></td>
<td>
<p>a site model's gamma site model,
as returned by <code><a href="#topic+create_gamma_site_model">create_gamma_site_model</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_group_sizes_dimension">group_sizes_dimension</code></td>
<td>
<p>the group sizes' dimension,
as used by the CBS tree prior (see <code><a href="#topic+create_cbs_tree_prior">create_cbs_tree_prior</a></code>)</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_gtr_site_model">gtr_site_model</code></td>
<td>
<p>a GTR site model,
as returned by <code><a href="#topic+create_gtr_site_model">create_gtr_site_model</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_has_non_strict_clock_model">has_non_strict_clock_model</code></td>
<td>
<p>boolean to indicate that the is
already at least one non-strict (i.e. relaxed log-normal) clock model</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_has_tip_dating">has_tip_dating</code></td>
<td>
<p>TRUE if the user has supplied tip dates,
FALSE otherwise</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_hky_site_model">hky_site_model</code></td>
<td>
<p>an HKY site model,
as returned by <code><a href="#topic+create_hky_site_model">create_hky_site_model</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_id">id</code></td>
<td>
<p>an alignment's IDs.
An ID can be extracted from its FASTA filename
with <code><a href="#topic+get_alignment_ids_from_fasta_filenames">get_alignment_ids_from_fasta_filenames</a></code>)</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_ids">ids</code></td>
<td>
<p>one or more alignments' IDs.
IDs can be extracted from their FASTA filenames
with <code><a href="#topic+get_alignment_ids_from_fasta_filenames">get_alignment_ids_from_fasta_filenames</a></code>)</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_inference_models">inference_models</code></td>
<td>
<p>a list of one or more inference models,
as can be created by <a href="#topic+create_inference_model">create_inference_model</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_initial_phylogenies">initial_phylogenies</code></td>
<td>
<p>one or more MCMC chain's initial phylogenies.
Each one set to <code>NA</code> will result in BEAST2 using a random phylogeny.
Else the phylogeny is assumed to be of class <code>phylo</code> from the
<code>ape</code> package</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_input_filename">input_filename</code></td>
<td>
<p>A FASTA filename.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_input_filenames">input_filenames</code></td>
<td>
<p>One or more FASTA filenames.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_is_monophyletic">is_monophyletic</code></td>
<td>
<p>boolean to indicate monophyly is assumed in
a Most Recent Common Ancestor prior,
as returned by <code><a href="#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_jc69_site_model">jc69_site_model</code></td>
<td>
<p>a JC69 site model,
as returned by <code><a href="#topic+create_jc69_site_model">create_jc69_site_model</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_kappa_param">kappa_param</code></td>
<td>
<p>a kappa parameter,
as created by <a href="#topic+create_kappa_param">create_kappa_param</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_log_every">log_every</code></td>
<td>
<p>number of MCMC states between writing to file</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_m_param">m_param</code></td>
<td>
<p>an m parameter,
as created by <a href="#topic+create_m_param">create_m_param</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_mcmc">mcmc</code></td>
<td>
<p>one MCMC.
Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create an MCMC.
Use <code><a href="#topic+create_ns_mcmc">create_ns_mcmc</a></code> to create an MCMC
for a Nested Sampling run.
Use <code><a href="#topic+check_mcmc">check_mcmc</a></code> to check if an MCMC is valid.
Use <code><a href="#topic+rename_mcmc_filenames">rename_mcmc_filenames</a></code> to rename the filenames in an MCMC.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_mode">mode</code></td>
<td>
<p>mode how to log.
Valid values are the ones returned by <a href="#topic+get_log_modes">get_log_modes</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_mrca_prior">mrca_prior</code></td>
<td>
<p>a Most Recent Common Ancestor prior,
as returned by <code><a href="#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_mrca_priors">mrca_priors</code></td>
<td>
<p>a list of one or more Most Recent Common Ancestor priors,
as returned by <code><a href="#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_mrca_prior_name">mrca_prior_name</code></td>
<td>
<p>the unique name of the MRCA prior,
for example a genus, family,
order or even class name.
Leave at <a href="base.html#topic+NA">NA</a> to have it named automatically.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_n_init_attempts">n_init_attempts</code></td>
<td>
<p>number of initialization attempts before failing</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_output_filename">output_filename</code></td>
<td>
<p>Name of the XML parameter file created by this
function. BEAST2 uses this file as input.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_param">param</code></td>
<td>
<p>a parameter, as can be created by <code><a href="#topic+create_param">create_param</a></code>.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_param_id">param_id</code></td>
<td>
<p>a parameter's ID</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_phylogeny">phylogeny</code></td>
<td>
<p>a phylogeny of type <code>phylo</code> from the <code>ape</code>
package</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_pop_sizes_scaler_scale_factor">pop_sizes_scaler_scale_factor</code></td>
<td>
<p>the scale factor used by the
population sizes scaler operator</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_pre_burnin">pre_burnin</code></td>
<td>
<p>number of burn in samples taken before entering
the main loop</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_rate_scaler_factor">rate_scaler_factor</code></td>
<td>
<p>the strict clock model's operator scaler
for the rate.
Use an empty string to indicate the default.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_rename_fun">rename_fun</code></td>
<td>
<p>a function to rename a filename,
as can be checked by <a href="#topic+check_rename_fun">check_rename_fun</a>. This function should
have one argument, which will be a filename or <a href="base.html#topic+NA">NA</a>. The
function should <a href="base.html#topic+return">return</a> one filename (when passed one filename) or
one <a href="base.html#topic+NA">NA</a> (when passed one <a href="base.html#topic+NA">NA</a>).
Example rename functions are:
</p>

<ul>
<li> <p><a href="#topic+get_remove_dir_fun">get_remove_dir_fun</a> get a function that removes the directory
paths from the filenames, in effect turning these into local files
</p>
</li>
<li> <p><a href="#topic+get_replace_dir_fun">get_replace_dir_fun</a> get a function that replaces the directory
paths from the filenames
</p>
</li>
<li> <p><a href="#topic+get_remove_hex_fun">get_remove_hex_fun</a> get a function that removes the
hex string from filenames.
For example, <code>tracelog_82c1a522040.log</code> becomes <code>tracelog.log</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="default_params_doc_+3A_rln_clock_model">rln_clock_model</code></td>
<td>
<p>a Relaxed Log-Normal clock model,
as returned by <code><a href="#topic+create_rln_clock_model">create_rln_clock_model</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_sample_from_prior">sample_from_prior</code></td>
<td>
<p>set to <a href="base.html#topic+TRUE">TRUE</a> to sample from the prior</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_sanitise_headers">sanitise_headers</code></td>
<td>
<p>set to <a href="base.html#topic+TRUE">TRUE</a> to sanitise the headers of the
log file</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_screenlog">screenlog</code></td>
<td>
<p>a <code>screenlog</code>,
as created by <a href="#topic+create_screenlog">create_screenlog</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_sequence_length">sequence_length</code></td>
<td>
<p>a DNA sequence length, in base pairs</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_site_model_name">site_model_name</code></td>
<td>
<p>name of a site model,
must be a name as returned by <code><a href="#topic+get_site_model_names">get_site_model_names</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_site_model_names">site_model_names</code></td>
<td>
<p>one or more names of a site model,
must be name among those returned by <code><a href="#topic+get_site_model_names">get_site_model_names</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_site_models">site_models</code></td>
<td>
<p>one or more site models,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_sort">sort</code></td>
<td>
<p>how to sort the log.
Valid values are the ones returned by <a href="#topic+get_log_sorts">get_log_sorts</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_store_every">store_every</code></td>
<td>
<p>number of states the MCMC will process
before the posterior's state will be saved to file.
Use -1 or <code>NA</code> to use the default frequency.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_strict_clock_model">strict_clock_model</code></td>
<td>
<p>a strict clock model,
as returned by <code><a href="#topic+create_strict_clock_model">create_strict_clock_model</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_taxa_names">taxa_names</code></td>
<td>
<p>names of the taxa,
as returned by <code><a href="#topic+get_taxa_names">get_taxa_names</a></code>.
Keep at <code>NA</code> to have it initialized automatically,
using all taxa in the alignment</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_tipdates_filename">tipdates_filename</code></td>
<td>
<p>name of the file containing the tip dates.
This file is assumed to have two columns, separated by a tab.
The first column contains the taxa names, the second column contains
the date.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_tn93_site_model">tn93_site_model</code></td>
<td>
<p>a TN93 site model,
as returned by <code><a href="#topic+create_tn93_site_model">create_tn93_site_model</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_tracelog">tracelog</code></td>
<td>
<p>a <code>tracelog</code>,
as created by <a href="#topic+create_tracelog">create_tracelog</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_treelog">treelog</code></td>
<td>
<p>a <code>treelog</code>,
as created by <a href="#topic+create_treelog">create_treelog</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_tree_prior">tree_prior</code></td>
<td>
<p>a tree priors,
as returned by <code><a href="#topic+create_tree_prior">create_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_tree_prior_name">tree_prior_name</code></td>
<td>
<p>name of a tree prior,
must be a name as returned by <code><a href="#topic+get_tree_prior_names">get_tree_prior_names</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_tree_prior_names">tree_prior_names</code></td>
<td>
<p>one or more names of a tree prior,
must be a name among those returned by <code><a href="#topic+get_tree_prior_names">get_tree_prior_names</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_tree_priors">tree_priors</code></td>
<td>
<p>one or more tree priors,
as returned by <code><a href="#topic+create_tree_prior">create_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_verbose">verbose</code></td>
<td>
<p>if TRUE, additional information is displayed, that
is potentially useful in debugging</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_yule_tree_prior">yule_tree_prior</code></td>
<td>
<p>a Yule tree_prior,
as created by <code><a href="#topic+create_yule_tree_prior">create_yule_tree_prior</a></code></p>
</td></tr>
</table>


<h3>Note</h3>

<p>This is an internal function, so it should be marked with
<code>@export</code>. This is not done, as this will disallow all
functions to find the documentation parameters
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='distr_to_xml'>Internal function</h2><span id='topic+distr_to_xml'></span>

<h3>Description</h3>

<p>Converts a distribution to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distr_to_xml(distr, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="distr_to_xml_+3A_distr">distr</code></td>
<td>
<p>a distribution,
as created by <code><a href="#topic+create_distr">create_distr</a></code>)</p>
</td></tr>
<tr><td><code id="distr_to_xml_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the distribution as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

distr_to_xml(
  create_uniform_distr(id = 1),
  beauti_options = create_beauti_options()
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='distr_to_xml_beta'>Internal function</h2><span id='topic+distr_to_xml_beta'></span>

<h3>Description</h3>

<p>Converts a beta distribution to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distr_to_xml_beta(distr, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="distr_to_xml_beta_+3A_distr">distr</code></td>
<td>
<p>a beta distribution,
as created by <code><a href="#topic+create_beta_distr">create_beta_distr</a></code>)</p>
</td></tr>
<tr><td><code id="distr_to_xml_beta_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the distribution as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='distr_to_xml_exp'>Internal function</h2><span id='topic+distr_to_xml_exp'></span>

<h3>Description</h3>

<p>Converts an exponential distribution to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distr_to_xml_exp(distr, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="distr_to_xml_exp_+3A_distr">distr</code></td>
<td>
<p>an exponential distribution,
as created by <code><a href="#topic+create_exp_distr">create_exp_distr</a></code>)</p>
</td></tr>
<tr><td><code id="distr_to_xml_exp_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the distribution as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='distr_to_xml_inv_gamma'>Internal function</h2><span id='topic+distr_to_xml_inv_gamma'></span>

<h3>Description</h3>

<p>Converts an inverse-gamma distribution to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distr_to_xml_inv_gamma(distr, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="distr_to_xml_inv_gamma_+3A_distr">distr</code></td>
<td>
<p>an inverse-gamma distribution,
as created by <code><a href="#topic+create_inv_gamma_distr">create_inv_gamma_distr</a></code>)</p>
</td></tr>
<tr><td><code id="distr_to_xml_inv_gamma_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the distribution as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='distr_to_xml_laplace'>Internal function</h2><span id='topic+distr_to_xml_laplace'></span>

<h3>Description</h3>

<p>Converts a Laplace distribution to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distr_to_xml_laplace(distr, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="distr_to_xml_laplace_+3A_distr">distr</code></td>
<td>
<p>a Laplace distribution
as created by <code><a href="#topic+create_laplace_distr">create_laplace_distr</a></code>)</p>
</td></tr>
<tr><td><code id="distr_to_xml_laplace_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the distribution as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='distr_to_xml_log_normal'>Internal function</h2><span id='topic+distr_to_xml_log_normal'></span>

<h3>Description</h3>

<p>Converts a log-normal distribution to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distr_to_xml_log_normal(distr, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="distr_to_xml_log_normal_+3A_distr">distr</code></td>
<td>
<p>a log-normal distribution,
as created by <code><a href="#topic+create_log_normal_distr">create_log_normal_distr</a></code>)</p>
</td></tr>
<tr><td><code id="distr_to_xml_log_normal_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the distribution as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='distr_to_xml_normal'>Internal function</h2><span id='topic+distr_to_xml_normal'></span>

<h3>Description</h3>

<p>Converts a normal distribution to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distr_to_xml_normal(distr, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="distr_to_xml_normal_+3A_distr">distr</code></td>
<td>
<p>a normal distribution,
as created by <code><a href="#topic+create_normal_distr">create_normal_distr</a></code>)</p>
</td></tr>
<tr><td><code id="distr_to_xml_normal_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the distribution as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='distr_to_xml_one_div_x'>Internal function</h2><span id='topic+distr_to_xml_one_div_x'></span>

<h3>Description</h3>

<p>Converts a 1/x distribution to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distr_to_xml_one_div_x(distr, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="distr_to_xml_one_div_x_+3A_distr">distr</code></td>
<td>
<p>a 1/x distribution,
as created by <code><a href="#topic+create_one_div_x_distr">create_one_div_x_distr</a></code>)</p>
</td></tr>
<tr><td><code id="distr_to_xml_one_div_x_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the distribution as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='distr_to_xml_poisson'>Internal function</h2><span id='topic+distr_to_xml_poisson'></span>

<h3>Description</h3>

<p>Converts a Poisson distribution to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distr_to_xml_poisson(distr, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="distr_to_xml_poisson_+3A_distr">distr</code></td>
<td>
<p>a Poisson distribution,
as created by <code><a href="#topic+create_poisson_distr">create_poisson_distr</a></code>)</p>
</td></tr>
<tr><td><code id="distr_to_xml_poisson_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the distribution as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='distr_to_xml_uniform'>Internal function</h2><span id='topic+distr_to_xml_uniform'></span>

<h3>Description</h3>

<p>Converts a uniform distribution to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distr_to_xml_uniform(distr, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="distr_to_xml_uniform_+3A_distr">distr</code></td>
<td>
<p>a uniform distribution,
as created by <code><a href="#topic+create_uniform_distr">create_uniform_distr</a></code>)</p>
</td></tr>
<tr><td><code id="distr_to_xml_uniform_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the distribution as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='extract_xml_loggers_from_lines'>Extract everything between first loggers and last loggers line</h2><span id='topic+extract_xml_loggers_from_lines'></span>

<h3>Description</h3>

<p>Extract everything between first loggers and last loggers line
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract_xml_loggers_from_lines(lines)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extract_xml_loggers_from_lines_+3A_lines">lines</code></td>
<td>
<p>lines of text</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of text from the first to and including the last operators line
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='extract_xml_operators_from_lines'>Extract everything between first operators and last operators line</h2><span id='topic+extract_xml_operators_from_lines'></span>

<h3>Description</h3>

<p>Extract everything between first operators and last operators line
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract_xml_operators_from_lines(lines)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extract_xml_operators_from_lines_+3A_lines">lines</code></td>
<td>
<p>lines of text</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of text from the first to and including the last operators line
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='extract_xml_section_from_lines'>Get the lines of an XML section, including the section tags</h2><span id='topic+extract_xml_section_from_lines'></span>

<h3>Description</h3>

<p>Get the lines of an XML section, including the section tags
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract_xml_section_from_lines(lines, section)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extract_xml_section_from_lines_+3A_lines">lines</code></td>
<td>
<p>lines of the XML text</p>
</td></tr>
<tr><td><code id="extract_xml_section_from_lines_+3A_section">section</code></td>
<td>
<p>the XML section name</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the section's lines of XML text, including the tags
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='fasta_file_to_sequences'>Convert a FASTA file to a table of sequences</h2><span id='topic+fasta_file_to_sequences'></span>

<h3>Description</h3>

<p>Convert a FASTA file to a table of sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fasta_file_to_sequences(fasta_filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fasta_file_to_sequences_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>One existing FASTA filenames</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a table of sequences
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

fasta_file_to_sequences(fasta_filename = get_fasta_filename())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='find_clock_model'>Finds a clock model with a certain ID</h2><span id='topic+find_clock_model'></span>

<h3>Description</h3>

<p>Finds a clock model with a certain ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_clock_model(clock_models, id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_clock_model_+3A_clock_models">clock_models</code></td>
<td>
<p>a list of one or more clock models,
as returned by <code><a href="#topic+create_clock_model">create_clock_model</a></code></p>
</td></tr>
<tr><td><code id="find_clock_model_+3A_id">id</code></td>
<td>
<p>the ID of the clock model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the clock models with the desired ID, NULL if such a clock model is
absent
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='find_first_regex_line'>Find the first line that satisfies a regex</h2><span id='topic+find_first_regex_line'></span>

<h3>Description</h3>

<p>Find the first line that satisfies a regex
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_first_regex_line(lines, regex)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_first_regex_line_+3A_lines">lines</code></td>
<td>
<p>lines of text</p>
</td></tr>
<tr><td><code id="find_first_regex_line_+3A_regex">regex</code></td>
<td>
<p>the regex as text</p>
</td></tr>
</table>


<h3>Value</h3>

<p>index of the line
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='find_first_xml_opening_tag_line'>Find the line number of the first section's opening tag</h2><span id='topic+find_first_xml_opening_tag_line'></span>

<h3>Description</h3>

<p>Find the line number of the first section's opening tag
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_first_xml_opening_tag_line(lines, section)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_first_xml_opening_tag_line_+3A_lines">lines</code></td>
<td>
<p>the lines of an XML text</p>
</td></tr>
<tr><td><code id="find_first_xml_opening_tag_line_+3A_section">section</code></td>
<td>
<p>the name of the XML section</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the line number's index (which is 1 for the first line) if the
opening tag is found, else NA
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='find_last_regex_line'>Find the index of the last line that matches a regex</h2><span id='topic+find_last_regex_line'></span>

<h3>Description</h3>

<p>Find the index of the last line that matches a regex
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_last_regex_line(lines, regex)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_last_regex_line_+3A_lines">lines</code></td>
<td>
<p>lines of text</p>
</td></tr>
<tr><td><code id="find_last_regex_line_+3A_regex">regex</code></td>
<td>
<p>regex string</p>
</td></tr>
</table>


<h3>Value</h3>

<p>index of the line
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='find_last_xml_closing_tag_line'>Find the highest line number of a section's closing tag</h2><span id='topic+find_last_xml_closing_tag_line'></span>

<h3>Description</h3>

<p>Find the highest line number of a section's closing tag
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_last_xml_closing_tag_line(lines, section)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_last_xml_closing_tag_line_+3A_lines">lines</code></td>
<td>
<p>the lines of an XML text</p>
</td></tr>
<tr><td><code id="find_last_xml_closing_tag_line_+3A_section">section</code></td>
<td>
<p>the name of the XML section</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the line number's index (which is 1 for the first line) if the
opening tag is found, else NA
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='freq_equilibrium_to_xml'>Creates the <code>freq_equilibrium</code> as XML</h2><span id='topic+freq_equilibrium_to_xml'></span>

<h3>Description</h3>

<p>Creates the <code>freq_equilibrium</code> as XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>freq_equilibrium_to_xml(freq_equilibrium, id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="freq_equilibrium_to_xml_+3A_freq_equilibrium">freq_equilibrium</code></td>
<td>
<p>a <code>freq_equilibrium</code> name</p>
</td></tr>
<tr><td><code id="freq_equilibrium_to_xml_+3A_id">id</code></td>
<td>
<p>a site model's name</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the <code>freq_equilibrium</code> as XML
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

freq_equilibrium_to_xml(freq_equilibrium = "estimated", id = "my_id")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='freq_param_to_xml'>Internal function</h2><span id='topic+freq_param_to_xml'></span>

<h3>Description</h3>

<p>Converts a 'freq' parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>freq_param_to_xml(freq_param, beauti_options = create_beauti_options())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="freq_param_to_xml_+3A_freq_param">freq_param</code></td>
<td>
<p>a 'freq' parameter,
as created by <a href="#topic+create_freq_param">create_freq_param</a></p>
</td></tr>
<tr><td><code id="freq_param_to_xml_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# The freq parameter must be initialized, i.e. have an ID
freq_param_to_xml(freq_param = create_freq_param(id = "1"))

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='gamma_distr_to_xml'>Internal function</h2><span id='topic+gamma_distr_to_xml'></span>

<h3>Description</h3>

<p>Converts a gamma distribution to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gamma_distr_to_xml(gamma_distr, beauti_options = create_beauti_options())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gamma_distr_to_xml_+3A_gamma_distr">gamma_distr</code></td>
<td>
<p>a gamma distribution,
as created by <code><a href="#topic+create_gamma_distr">create_gamma_distr</a></code>)</p>
</td></tr>
<tr><td><code id="gamma_distr_to_xml_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the distribution as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# gamma distribution must be initialized
gamma_distr_to_xml(
  gamma_distr = create_gamma_distr(
    id = "0",
    alpha = create_alpha_param(id = "2", value = "0.5396"),
    beta = create_beta_param(id = "3", value = "0.3819")
  )
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='gamma_site_model_to_xml_prior_distr'>Internal function.</h2><span id='topic+gamma_site_model_to_xml_prior_distr'></span>

<h3>Description</h3>

<p>Creates the gamma site models section in the distribution section
of a BEAST2 XML parameter file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gamma_site_model_to_xml_prior_distr(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gamma_site_model_to_xml_prior_distr_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='gamma_site_model_to_xml_state'>Converts a gamma site model to XML,
used in the <code>state</code> section</h2><span id='topic+gamma_site_model_to_xml_state'></span>

<h3>Description</h3>

<p>Converts a gamma site model to XML,
used in the <code>state</code> section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gamma_site_model_to_xml_state(gamma_site_model, id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gamma_site_model_to_xml_state_+3A_gamma_site_model">gamma_site_model</code></td>
<td>
<p>a gamma site model,
as created by <code><a href="#topic+create_gamma_site_model">create_gamma_site_model</a></code>)</p>
</td></tr>
<tr><td><code id="gamma_site_model_to_xml_state_+3A_id">id</code></td>
<td>
<p>the site model's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the gamma_site model as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='gamma_site_models_to_xml_prior_distr'>Deprecated function</h2><span id='topic+gamma_site_models_to_xml_prior_distr'></span>

<h3>Description</h3>

<p>Internal function to
creates the gamma site models section in the distribution section
of a BEAST2 XML parameter file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gamma_site_models_to_xml_prior_distr(
  site_models,
  beauti_options = create_beauti_options()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gamma_site_models_to_xml_prior_distr_+3A_site_models">site_models</code></td>
<td>
<p>one or more site models,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
<tr><td><code id="gamma_site_models_to_xml_prior_distr_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='get_alignment_id'>Conclude the ID from a FASTA filename.</h2><span id='topic+get_alignment_id'></span>

<h3>Description</h3>

<p>This is done in the same way as BEAST2 will do so.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_alignment_id(fasta_filename, capitalize_first_char_id = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_alignment_id_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>a FASTA filename.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.
Note that BEAST2 also supports missing data,
by using a dash (<code>-</code>) or question mark (<code>?</code>)
as a sequence.</p>
</td></tr>
<tr><td><code id="get_alignment_id_+3A_capitalize_first_char_id">capitalize_first_char_id</code></td>
<td>
<p>if TRUE, the first character will
be capitalized</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an alignment's ID
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+check_alignment_id">check_alignment_id</a> to check if an alignment
ID is valid.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# Path need not exist, use UNIX path as example
# anthus_aco_sub
alignment_id &lt;- get_alignment_id("/home/homer/anthus_aco_sub.fas")
check_alignment_id(alignment_id)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_alignment_ids'>Get the alignment IDs from one or more files.</h2><span id='topic+get_alignment_ids'></span>

<h3>Description</h3>

<p>This is done in the same way as BEAST2 does by default
The file extension will be used to determine which
type of file is worked on.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_alignment_ids(filenames)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_alignment_ids_+3A_filenames">filenames</code></td>
<td>
<p>names of the files to be checked</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the IDs extracted from the one or more files
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+get_alignment_ids_from_fasta_filenames">get_alignment_ids_from_fasta_filenames</a> to
get the alignment IDs from files known to be FASTA files
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

get_alignment_ids(
  get_beautier_paths(c("anthus_aco.fas", "anthus_nd2.fas"))
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_alignment_ids_from_fasta_filenames'>Get the alignment ID from one or more FASTA filenames.</h2><span id='topic+get_alignment_ids_from_fasta_filenames'></span>

<h3>Description</h3>

<p>This is done in the same way as BEAST2 does by default.
The files are assumed to be FASTA. If this is not the case, there
may be any kind of error message when calling this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_alignment_ids_from_fasta_filenames(fasta_filenames)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_alignment_ids_from_fasta_filenames_+3A_fasta_filenames">fasta_filenames</code></td>
<td>
<p>One or more FASTA filenames.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the IDs from one or more FASTA files
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+get_alignment_ids">get_alignment_ids</a> to get the alignment IDs from multiple
kids of files.
Use <a href="#topic+are_fasta_filenames">are_fasta_filenames</a> to
see if the filenames all have a common FASTA filename extension.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

get_alignment_ids_from_fasta_filenames(
  get_beautier_paths(c("anthus_aco.fas", "anthus_nd2.fas"))
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_beautier_folder'>Get the path to the <a href="#topic+beautier">beautier</a> temporary files folder</h2><span id='topic+get_beautier_folder'></span>

<h3>Description</h3>

<p>Get the path to the <a href="#topic+beautier">beautier</a> temporary files folder
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_beautier_folder()
</code></pre>


<h3>Value</h3>

<p>the path to the <a href="#topic+beautier">beautier</a> temporary files folder
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

get_beautier_folder()

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_beautier_path'>Get the full path of a file in the <code>inst/extdata</code> folder</h2><span id='topic+get_beautier_path'></span>

<h3>Description</h3>

<p>Get the full path of a file in the <code>inst/extdata</code> folder
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_beautier_path(filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_beautier_path_+3A_filename">filename</code></td>
<td>
<p>the file's name, without the path</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the full path of the filename
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>for more files, use <code><a href="#topic+get_beautier_paths">get_beautier_paths</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

get_beautier_path("test_output_0.fas")
get_beautier_path("anthus_aco.fas")
get_beautier_path("anthus_nd2.fas")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_beautier_paths'>Get the full paths of files in the <code>inst/extdata</code> folder</h2><span id='topic+get_beautier_paths'></span>

<h3>Description</h3>

<p>Get the full paths of files in the <code>inst/extdata</code> folder
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_beautier_paths(filenames)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_beautier_paths_+3A_filenames">filenames</code></td>
<td>
<p>the files' names, without the path</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the filenames' full paths
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+get_beautier_path">get_beautier_path</a> to get the path of one file
</p>
<p>for one file, use <code><a href="#topic+get_beautier_path">get_beautier_path</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

 get_beautier_paths(
   c("test_output_0.fas", "anthus_aco.fas", "anthus_nd2.fas")
 )

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_beautier_tempfilename'>Get a temporary filename</h2><span id='topic+get_beautier_tempfilename'></span>

<h3>Description</h3>

<p>Get a temporary filename, similar to <a href="base.html#topic+tempfile">tempfile</a>,
except that it always writes to a temporary folder
named <a href="#topic+beautier">beautier</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_beautier_tempfilename(pattern = "file", fileext = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_beautier_tempfilename_+3A_pattern">pattern</code></td>
<td>
<p>a non-empty character vector
giving the initial part of the name.</p>
</td></tr>
<tr><td><code id="get_beautier_tempfilename_+3A_fileext">fileext</code></td>
<td>
<p>a non-empty character vector
giving the file extension</p>
</td></tr>
</table>


<h3>Value</h3>

<p>name for a temporary file
</p>


<h3>Note</h3>

<p>this function is added to make sure no temporary
cache files are left undeleted
</p>

<hr>
<h2 id='get_clock_model_name'>Get the BEAUti name for a clock model</h2><span id='topic+get_clock_model_name'></span>

<h3>Description</h3>

<p>Will <a href="base.html#topic+stop">stop</a> if the clock model is an invalid clock model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_clock_model_name(clock_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_clock_model_name_+3A_clock_model">clock_model</code></td>
<td>
<p>a clock model,
as returned by <code><a href="#topic+create_clock_model">create_clock_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>name of the clock model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# StrictClock
get_clock_model_name(create_strict_clock_model())

# RelaxedClock
get_clock_model_name(create_rln_clock_model())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_clock_model_names'>Get the clock model names</h2><span id='topic+get_clock_model_names'></span>

<h3>Description</h3>

<p>Get the clock model names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_clock_model_names()
</code></pre>


<h3>Value</h3>

<p>the clock model names
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_clock_models">create_clock_models</a> to create a list
with all clock models
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

get_clock_model_names()

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_clock_models_ids'>Collect the IDs of the list of clock models</h2><span id='topic+get_clock_models_ids'></span>

<h3>Description</h3>

<p>Collect the IDs of the list of clock models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_clock_models_ids(clock_models)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_clock_models_ids_+3A_clock_models">clock_models</code></td>
<td>
<p>a list of one or more clock models,
as returned by <code><a href="#topic+create_clock_model">create_clock_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>IDs of the clock models
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='get_crown_age'>Obtain the crown age of a phylogeny.</h2><span id='topic+get_crown_age'></span>

<h3>Description</h3>

<p>The crown age of a phylogeny is the time between
the present and the moment of at which the first
diversification (resulting in two lineages) happened.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_crown_age(phylogeny)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_crown_age_+3A_phylogeny">phylogeny</code></td>
<td>
<p>The phylogeny to obtain the crown age of</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the crown age of the phylogeny
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

phylogeny &lt;- ape::read.tree(text = "(a:15,b:15):1;")
get_crown_age(phylogeny = phylogeny)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_default_beast_namespace'>Get the default 'namespace' element value of the 'beast' XML tag.</h2><span id='topic+get_default_beast_namespace'></span>

<h3>Description</h3>

<p>Get the default 'namespace' element value of the 'beast' XML tag.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_default_beast_namespace()
</code></pre>


<h3>Value</h3>

<p>the default 'namespace' element value of the 'beast' XML tag.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>* For BEAST2 v2.4, use <a href="#topic+get_default_beast_namespace_v2_4">get_default_beast_namespace_v2_4</a>
* For BEAST2 v2.6, use <a href="#topic+get_default_beast_namespace_v2_6">get_default_beast_namespace_v2_6</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_default_beast_namespace()
</code></pre>

<hr>
<h2 id='get_default_beast_namespace_v2_4'>Get the default 'namespace' element value of the 'beast' XML tag
for BEAST 2.4</h2><span id='topic+get_default_beast_namespace_v2_4'></span>

<h3>Description</h3>

<p>Get the default 'namespace' element value of the 'beast' XML tag
for BEAST 2.4
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_default_beast_namespace_v2_4()
</code></pre>


<h3>Value</h3>

<p>the default 'namespace' element value of the 'beast' XML tag
for BEAST 2.4
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_default_beast_namespace_v2_4()
</code></pre>

<hr>
<h2 id='get_default_beast_namespace_v2_6'>Get the default 'namespace' element value of the 'beast' XML tag
for BEAST 2.6</h2><span id='topic+get_default_beast_namespace_v2_6'></span>

<h3>Description</h3>

<p>Get the default 'namespace' element value of the 'beast' XML tag
for BEAST 2.6
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_default_beast_namespace_v2_6()
</code></pre>


<h3>Value</h3>

<p>the default 'namespace' element value of the 'beast' XML tag
for BEAST 2.6
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_default_beast_namespace_v2_6()
</code></pre>

<hr>
<h2 id='get_distr_n_params'>Get the number of parameters a distribution uses</h2><span id='topic+get_distr_n_params'></span>

<h3>Description</h3>

<p>Get the number of parameters a distribution uses
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_distr_n_params(distr)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_distr_n_params_+3A_distr">distr</code></td>
<td>
<p>a distribution,
as created by <code><a href="#topic+create_distr">create_distr</a></code> or (preferable)
its named functions</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the number of parameters that distribution uses
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

get_distr_n_params(create_beta_distr())
get_distr_n_params(create_exp_distr())
get_distr_n_params(create_gamma_distr())
get_distr_n_params(create_inv_gamma_distr())
get_distr_n_params(create_laplace_distr())
get_distr_n_params(create_log_normal_distr())
get_distr_n_params(create_normal_distr())
get_distr_n_params(create_one_div_x_distr())
get_distr_n_params(create_poisson_distr())
get_distr_n_params(create_uniform_distr())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_distr_names'>Get the distribution names</h2><span id='topic+get_distr_names'></span>

<h3>Description</h3>

<p>Get the distribution names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_distr_names()
</code></pre>


<h3>Value</h3>

<p>the distribution names
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

get_distr_names()

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_fasta_filename'>Get the path of a FASTA file used in testing</h2><span id='topic+get_fasta_filename'></span>

<h3>Description</h3>

<p>Get the path of a FASTA file used in testing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_fasta_filename()
</code></pre>


<h3>Value</h3>

<p>the path of a FASTA file used in testing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  input_filename &lt;- get_fasta_filename()
  output_filename &lt;- get_beautier_tempfilename()

  create_beast2_input_file(
    input_filename = input_filename,
    output_filename = output_filename
  )
  file.remove(output_filename)

  remove_beautier_folder()
}
</code></pre>

<hr>
<h2 id='get_file_base_sans_ext'>Get the base of the filename base without extension</h2><span id='topic+get_file_base_sans_ext'></span>

<h3>Description</h3>

<p>The path need not to actually exist
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_file_base_sans_ext(filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_file_base_sans_ext_+3A_filename">filename</code></td>
<td>
<p>A filename</p>
</td></tr>
</table>


<h3>Value</h3>

<p>That filename without its full path and extension
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# Path need not exist, use UNIX path as example
# test
get_file_base_sans_ext("/home/homer/test.txt")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_freq_equilibrium_names'>Returns valid values for the <code>freq_equilibrium</code> argument</h2><span id='topic+get_freq_equilibrium_names'></span>

<h3>Description</h3>

<p>Returns valid values for the <code>freq_equilibrium</code> argument
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_freq_equilibrium_names()
</code></pre>


<h3>Value</h3>

<p>the valid values for the <code>freq_equilibrium</code> argument
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the <code>freq_equilibrium</code> argument is used in
<code><a href="#topic+create_gtr_site_model">create_gtr_site_model</a></code>,
<code><a href="#topic+create_hky_site_model">create_hky_site_model</a></code>,
and <code><a href="#topic+create_tn93_site_model">create_tn93_site_model</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

get_freq_equilibrium_names()

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_gamma_site_model_n_distrs'>Get the number of distributions in a gamma site model</h2><span id='topic+get_gamma_site_model_n_distrs'></span>

<h3>Description</h3>

<p>Get the number of distributions in a gamma site model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gamma_site_model_n_distrs(gamma_site_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_gamma_site_model_n_distrs_+3A_gamma_site_model">gamma_site_model</code></td>
<td>
<p>a site model's gamma site model,
as returned by <code><a href="#topic+create_gamma_site_model">create_gamma_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the number of distributions a gamma site model has
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_gamma_site_model">create_gamma_site_model</a> to create a gamma site model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# zero distributions
gamma_site_model &lt;- create_gamma_site_model()
get_gamma_site_model_n_distrs(
  gamma_site_model
)

gamma_site_model &lt;- create_gamma_site_model(
 gamma_cat_count = 2,
 gamma_shape_prior_distr = create_exp_distr()
)

# one distribution
get_gamma_site_model_n_distrs(gamma_site_model)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_gamma_site_model_n_params'>Get the number of distributions a site model has</h2><span id='topic+get_gamma_site_model_n_params'></span>

<h3>Description</h3>

<p>Get the number of distributions a site model has
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gamma_site_model_n_params(gamma_site_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_gamma_site_model_n_params_+3A_gamma_site_model">gamma_site_model</code></td>
<td>
<p>a site model's gamma site model,
as returned by <code><a href="#topic+create_gamma_site_model">create_gamma_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the number of parameters a site model has
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# 0 parameters
get_gamma_site_model_n_params(
  create_gamma_site_model(gamma_cat_count = 0)
)

# 0 parameters
get_gamma_site_model_n_params(
  create_gamma_site_model(gamma_cat_count = 1)
)

# 1 parameter
get_gamma_site_model_n_params(
    create_gamma_site_model(
    gamma_cat_count = 2,
    gamma_shape_prior_distr = create_exp_distr()
  )
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_has_non_strict_clock_model'>Determines if there is at least one non-strict clock model
in the list of one or more clock models</h2><span id='topic+get_has_non_strict_clock_model'></span>

<h3>Description</h3>

<p>Determines if there is at least one non-strict clock model
in the list of one or more clock models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_has_non_strict_clock_model(clock_models)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_has_non_strict_clock_model_+3A_clock_models">clock_models</code></td>
<td>
<p>a list of one or more clock models,
as returned by <code><a href="#topic+create_clock_model">create_clock_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if there is at least one non-strict clock model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='get_inference_model_filenames'>Get the filenames stored in an inference model.</h2><span id='topic+get_inference_model_filenames'></span>

<h3>Description</h3>

<p>If there is no name for a <code>tipdates</code> file specified (as done by
setting <code>inference_model$tipdates_filename</code> to <a href="base.html#topic+NA">NA</a>,
there will be one filename less returned
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_inference_model_filenames(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_inference_model_filenames_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the filenames stored in an inference model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

inference_model &lt;- create_inference_model()
filenames &lt;- get_inference_model_filenames(inference_model)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_log_modes'>Get the possible log modes</h2><span id='topic+get_log_modes'></span>

<h3>Description</h3>

<p>Get the possible log modes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_log_modes()
</code></pre>


<h3>Value</h3>

<p>the possible log modes
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_log_modes()
</code></pre>

<hr>
<h2 id='get_log_sorts'>Get the possible log sorts</h2><span id='topic+get_log_sorts'></span>

<h3>Description</h3>

<p>Get the possible log sorts
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_log_sorts()
</code></pre>


<h3>Value</h3>

<p>the possible log sorts
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_log_sorts()
</code></pre>

<hr>
<h2 id='get_mcmc_filenames'>Get the filenames stored in an MCMC.</h2><span id='topic+get_mcmc_filenames'></span>

<h3>Description</h3>

<p>If a filename is set to an empty string, to indicate a certain log file
need not be created, this (non-)filename will not be returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mcmc_filenames(mcmc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_mcmc_filenames_+3A_mcmc">mcmc</code></td>
<td>
<p>one MCMC.
Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create an MCMC.
Use <code><a href="#topic+create_ns_mcmc">create_ns_mcmc</a></code> to create an MCMC
for a Nested Sampling run.
Use <code><a href="#topic+check_mcmc">check_mcmc</a></code> to check if an MCMC is valid.
Use <code><a href="#topic+rename_mcmc_filenames">rename_mcmc_filenames</a></code> to rename the filenames in an MCMC.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the filenames stored in an MCMC
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

mcmc &lt;- create_mcmc()
mcmc$tracelog$filename &lt;- "/home/john/trace.log"
mcmc$screenlog$filename &lt;- "/home/john/screen.log"
mcmc$treelog$filename &lt;- "/home/john/tree.log"

# 3 filenames
filenames &lt;- get_mcmc_filenames(mcmc)

# If there is no need to write to the screenlog file ...
mcmc$screenlog$filename &lt;- ""

# 2 filenames
# ... one file less will be created
filenames &lt;- get_mcmc_filenames(mcmc)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_n_taxa'>Extract the number of taxa from a file</h2><span id='topic+get_n_taxa'></span>

<h3>Description</h3>

<p>Extract the number of taxa from a file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_n_taxa(filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_n_taxa_+3A_filename">filename</code></td>
<td>
<p>name of a FASTA file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the number of taxa
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

fasta_filename &lt;- get_beautier_path("test_output_5.fas")
# 5
get_n_taxa(fasta_filename)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_operator_id_pre'>Get the prefix of operator IDs</h2><span id='topic+get_operator_id_pre'></span>

<h3>Description</h3>

<p>Get the prefix of operator IDs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_operator_id_pre(tree_prior)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_operator_id_pre_+3A_tree_prior">tree_prior</code></td>
<td>
<p>a tree priors,
as returned by <code><a href="#topic+create_tree_prior">create_tree_prior</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the prefix of operator IDs, similar to the name of a tree prior
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# BirthDeath
get_operator_id_pre(
  tree_prior = create_bd_tree_prior()
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_param_names'>Get the parameter names</h2><span id='topic+get_param_names'></span>

<h3>Description</h3>

<p>Get the parameter names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_param_names()
</code></pre>


<h3>Value</h3>

<p>the parameter names
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

names &lt;- get_param_names()

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_remove_dir_fun'>Get a function that, from a filename, returns the part
without the directory.</h2><span id='topic+get_remove_dir_fun'></span>

<h3>Description</h3>

<p>Or: get a function that returns the local version of a filename.
Also, the function will return <a href="base.html#topic+NA">NA</a> if the filename is <a href="base.html#topic+NA">NA</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_remove_dir_fun()
</code></pre>


<h3>Value</h3>

<p>a function to remove the folder name from a path
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>see <a href="#topic+check_rename_fun">check_rename_fun</a>
for an overview of file renaming functions
</p>

<hr>
<h2 id='get_remove_hex_fun'>Get a function that removes the hex string from filenames.</h2><span id='topic+get_remove_hex_fun'></span>

<h3>Description</h3>

<p>The default filenames created by <a href="#topic+beautier">beautier</a> are temporary files,
such as <code>/home/john/.cache/tracelog_82c5888db98.log</code> (on Linux),
where <code>/home/john/.cache</code> is the location to a temporary folder
(on Linux) and <code>tracelog_82c5888db98.log</code> the filename.
The filename ends with a hex string (as is common for temporary files,
as <a href="base.html#topic+tempfile">tempfile</a> does so).
Because <a href="#topic+beautier">beautier</a> puts an underscore
between the filename description (<code>tracelog</code>) and the hex
string, this function removes both.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_remove_hex_fun()
</code></pre>


<h3>Value</h3>

<p>a function to remove the hext string from filenames
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

f &lt;- get_remove_hex_fun()
# /home/john/beast2.xml.state
f("/home/john/beast2_186c7404208c.xml.state")

# beast2.xml.state
f("beast2_186c7404208c.xml.state")

# NA
f(NA)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_replace_dir_fun'>Get a function to replace the directory of a filename</h2><span id='topic+get_replace_dir_fun'></span>

<h3>Description</h3>

<p>Get a function to replace the directory of a filename
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_replace_dir_fun(new_dir_name = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_replace_dir_fun_+3A_new_dir_name">new_dir_name</code></td>
<td>
<p>the new directory name</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a function to replace the directory of a filename
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='get_site_model_n_distrs'>Get the number of distributions a site model has</h2><span id='topic+get_site_model_n_distrs'></span>

<h3>Description</h3>

<p>Get the number of distributions a site model has
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_site_model_n_distrs(site_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_site_model_n_distrs_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the number of distributions a site model has
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# 5: rates AC, AG, AT, CG and GT
get_site_model_n_distrs(create_gtr_site_model())

# 1: kappa
get_site_model_n_distrs(create_hky_site_model())

# 0: npne
get_site_model_n_distrs(create_jc69_site_model())

# 2: kappa 1 and kappa 2
get_site_model_n_distrs(create_tn93_site_model())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_site_model_n_params'>Get the number of distributions a site model has</h2><span id='topic+get_site_model_n_params'></span>

<h3>Description</h3>

<p>Get the number of distributions a site model has
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_site_model_n_params(site_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_site_model_n_params_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the number of parameters a site model has
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# Ten
get_site_model_n_params(create_gtr_site_model())

# Two
get_site_model_n_params(create_hky_site_model())

# Zero
get_site_model_n_params(create_jc69_site_model())

# Four
get_site_model_n_params(create_tn93_site_model())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_site_model_names'>Get the site models' names</h2><span id='topic+get_site_model_names'></span>

<h3>Description</h3>

<p>Get the site models' names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_site_model_names()
</code></pre>


<h3>Value</h3>

<p>the site model names
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_site_models">create_site_models</a> to get all site models
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

get_site_model_names()

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_site_models_n_distrs'>Get the number of distributions a site model has</h2><span id='topic+get_site_models_n_distrs'></span>

<h3>Description</h3>

<p>Get the number of distributions a site model has
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_site_models_n_distrs(site_models)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_site_models_n_distrs_+3A_site_models">site_models</code></td>
<td>
<p>one or more site models,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the number of distributions the site models have
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# 5
get_site_models_n_distrs(list(create_gtr_site_model()))
# 1
get_site_models_n_distrs(list(create_hky_site_model()))
# 0
get_site_models_n_distrs(list(create_jc69_site_model()))
# 2
get_site_models_n_distrs(list(create_tn93_site_model()))

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_site_models_n_params'>Get the number of distributions one or more site models have</h2><span id='topic+get_site_models_n_params'></span>

<h3>Description</h3>

<p>Get the number of distributions one or more site models have
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_site_models_n_params(site_models)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_site_models_n_params_+3A_site_models">site_models</code></td>
<td>
<p>one or more site models,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the number of parameters the site models have
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# Ten
get_site_models_n_params(list(create_gtr_site_model()))

# Two
get_site_models_n_params(list(create_hky_site_model()))

# Zero
get_site_models_n_params(list(create_jc69_site_model()))

# Four
get_site_models_n_params(list(create_tn93_site_model()))

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_taxa_names'>Extract the names of taxa from a file</h2><span id='topic+get_taxa_names'></span>

<h3>Description</h3>

<p>Extract the names of taxa from a file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_taxa_names(filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_taxa_names_+3A_filename">filename</code></td>
<td>
<p>name of a FASTA file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the taxa names
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

get_taxa_names(get_beautier_path("anthus_aco_sub.fas"))

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_tree_prior_n_distrs'>Get the number of distributions a tree prior has</h2><span id='topic+get_tree_prior_n_distrs'></span>

<h3>Description</h3>

<p>Get the number of distributions a tree prior has
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_tree_prior_n_distrs(tree_prior)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_tree_prior_n_distrs_+3A_tree_prior">tree_prior</code></td>
<td>
<p>a tree priors,
as returned by <code><a href="#topic+create_tree_prior">create_tree_prior</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the number of distributions a tree prior has
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# 2: birth_rate_distr and death_rate_distr
get_tree_prior_n_distrs(create_bd_tree_prior())

# 0:none
get_tree_prior_n_distrs(create_cbs_tree_prior())

# 1: pop_size_distr
get_tree_prior_n_distrs(create_ccp_tree_prior())

 # 2:pop_size_distr and growth_rate_distr
get_tree_prior_n_distrs(create_cep_tree_prior())

# 1: birth_rate_distr
get_tree_prior_n_distrs(create_yule_tree_prior())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_tree_prior_n_params'>Get the number of parameters a tree prior has</h2><span id='topic+get_tree_prior_n_params'></span>

<h3>Description</h3>

<p>Get the number of parameters a tree prior has
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_tree_prior_n_params(tree_prior)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_tree_prior_n_params_+3A_tree_prior">tree_prior</code></td>
<td>
<p>a tree_prior,
as created by <code><a href="#topic+create_tree_prior">create_tree_prior</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the number of parameters a tree prior has
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# birth_rate_distr is uniform, which has zero parameters
# death_rate_distr is uniform, which has zero parameters
get_tree_prior_n_params(create_bd_tree_prior())

# no distributions, no parameters
get_tree_prior_n_params(create_cbs_tree_prior())

# pop_size_distr is 1/x, which has zero parameters
get_tree_prior_n_params(create_ccp_tree_prior())

# pop_size_distr is 1/x, which has zero parameters
# growth_rate_distr is Laplace, which has two parameters
get_tree_prior_n_params(create_cep_tree_prior())

# birth_rate_distr is uniform, which has zero parameters
get_tree_prior_n_params(create_yule_tree_prior())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_tree_prior_names'>Get the tree prior names</h2><span id='topic+get_tree_prior_names'></span>

<h3>Description</h3>

<p>Get the tree prior names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_tree_prior_names()
</code></pre>


<h3>Value</h3>

<p>the tree prior names
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_tree_priors">create_tree_priors</a> to get all tree priors
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

get_tree_prior_names()

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_tree_priors_n_distrs'>Get the number of distributions a tree prior has</h2><span id='topic+get_tree_priors_n_distrs'></span>

<h3>Description</h3>

<p>Get the number of distributions a tree prior has
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_tree_priors_n_distrs(tree_priors)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_tree_priors_n_distrs_+3A_tree_priors">tree_priors</code></td>
<td>
<p>one or more tree priors,
as returned by <code><a href="#topic+create_tree_prior">create_tree_prior</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the number of distributions a tree prior has
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# Three distrubutions
get_tree_priors_n_distrs(
  list(
    create_bd_tree_prior(), # has two distributions
    create_ccp_tree_prior() # has one distribution
  )
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_tree_priors_n_params'>Get the number of parameters a list of tree priors has</h2><span id='topic+get_tree_priors_n_params'></span>

<h3>Description</h3>

<p>Get the number of parameters a list of tree priors has
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_tree_priors_n_params(tree_priors)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_tree_priors_n_params_+3A_tree_priors">tree_priors</code></td>
<td>
<p>one or more tree priors,
as returned by <code><a href="#topic+create_tree_prior">create_tree_prior</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the number of parameters the tree priors have
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# Two
get_tree_priors_n_params(
  list(
    create_bd_tree_prior(), # zero
    create_cep_tree_prior() # two
  )
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_xml_closing_tag'>Get the XML closing tag</h2><span id='topic+get_xml_closing_tag'></span>

<h3>Description</h3>

<p>Get the XML closing tag
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_xml_closing_tag(text)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_xml_closing_tag_+3A_text">text</code></td>
<td>
<p>lines of XML to extract the XML closing tag from</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the closing tag if found, else NA
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# my_tag
get_xml_closing_tag("&lt;my_tag text=something&gt;&lt;/my_tag&gt;")

# Will return NA
get_xml_closing_tag("&lt;my_tag text=something/&gt;")
get_xml_closing_tag("no_xml")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='get_xml_opening_tag'>Get the XML opening tag</h2><span id='topic+get_xml_opening_tag'></span>

<h3>Description</h3>

<p>Get the XML opening tag
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_xml_opening_tag(text)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_xml_opening_tag_+3A_text">text</code></td>
<td>
<p>text to be determined to be valid</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the opening tag if found, else NA
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# my_tag
get_xml_opening_tag("&lt;my_tag text=something/&gt;")

# NA when there is no opening tag
get_xml_opening_tag("no_xml")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='gtr_site_model_to_xml_prior_distr'>Internal function</h2><span id='topic+gtr_site_model_to_xml_prior_distr'></span>

<h3>Description</h3>

<p>Converts a GTR site model to XML,
used in the <code>prior</code> section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtr_site_model_to_xml_prior_distr(site_model, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gtr_site_model_to_xml_prior_distr_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
<tr><td><code id="gtr_site_model_to_xml_prior_distr_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the site model as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gtr_site_model_to_xml_prior_distr(
  site_model = create_gtr_site_model(
    id = 1,
    rate_ac_prior_distr = create_uniform_distr(id = 2),
    rate_ag_prior_distr = create_uniform_distr(id = 3),
    rate_at_prior_distr = create_uniform_distr(id = 4),
    rate_cg_prior_distr = create_uniform_distr(id = 5),
    rate_gt_prior_distr = create_uniform_distr(id = 6)
  ),
  beauti_options = create_beauti_options()
)
</code></pre>

<hr>
<h2 id='gtr_site_model_to_xml_state'>Converts a site model to XML,
used in the <code>state</code> section</h2><span id='topic+gtr_site_model_to_xml_state'></span>

<h3>Description</h3>

<p>Converts a site model to XML,
used in the <code>state</code> section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtr_site_model_to_xml_state(
  site_model,
  beauti_options = create_beauti_options()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gtr_site_model_to_xml_state_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
<tr><td><code id="gtr_site_model_to_xml_state_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the site model as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='has_mrca_prior'>Determines if the inference model has an MRCA prior.</h2><span id='topic+has_mrca_prior'></span>

<h3>Description</h3>

<p>Will <a href="base.html#topic+stop">stop</a> if the inference model is invalid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>has_mrca_prior(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="has_mrca_prior_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the inference model has an MRCA prior,
FALSE otherwise
</p>


<h3>Note</h3>

<p>MRCA: 'Most Recent Common Ancestor'
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+create_inference_model">create_inference_model</a></code>: create an inference model
</p>
</li>
<li> <p><code><a href="#topic+create_mrca_prior">create_mrca_prior</a></code>: create an MRCA prior
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# No MRCA prior
inference_model &lt;- create_inference_model(
  mrca_prior = NA
)
has_mrca_prior(inference_model) # Returns FALSE

# A default MRCA prior
inference_model &lt;- create_inference_model(
  mrca_prior = create_mrca_prior()
)
has_mrca_prior(inference_model) # Returns TRUE

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='has_mrca_prior_with_distr'>See if the inference model has one MRCA prior with a distribution</h2><span id='topic+has_mrca_prior_with_distr'></span>

<h3>Description</h3>

<p>See if the inference model has one MRCA prior with a distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>has_mrca_prior_with_distr(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="has_mrca_prior_with_distr_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the inference model has one MRCA prior with a distribution,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='has_rln_clock_model'>Determine if the <code>inference_model</code> uses
a relaxed log-normal clock model.</h2><span id='topic+has_rln_clock_model'></span>

<h3>Description</h3>

<p>Determine if the <code>inference_model</code> uses
a relaxed log-normal clock model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>has_rln_clock_model(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="has_rln_clock_model_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the <code>inference_model</code> uses
a relaxed log-normal clock model,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {
  check_empty_beautier_folder()

  # Yes, has a RLN clock model
  has_rln_clock_model(
    create_inference_model(clock_model = create_rln_clock_model())
  )

  # No RLN clock model
  has_rln_clock_model(
    create_inference_model(clock_model = create_strict_clock_model())
  )

  check_empty_beautier_folder()
}
</code></pre>

<hr>
<h2 id='has_strict_clock_model'>Determine if the <code>inference_model</code> uses a strict clock model.</h2><span id='topic+has_strict_clock_model'></span>

<h3>Description</h3>

<p>Determine if the <code>inference_model</code> uses a strict clock model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>has_strict_clock_model(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="has_strict_clock_model_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the <code>inference_model</code> uses a strict clock model,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# Yes, has a strict clock model
has_strict_clock_model(
  create_inference_model(clock_model = create_strict_clock_model())
)

# No strict clock model
has_strict_clock_model(
  create_inference_model(clock_model = create_rln_clock_model())
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='has_tip_dating'>Determine if the <code>inference_model</code> uses tip dating.</h2><span id='topic+has_tip_dating'></span>

<h3>Description</h3>

<p>Determine if the <code>inference_model</code> uses tip dating
</p>


<h3>Usage</h3>

<pre><code class='language-R'>has_tip_dating(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="has_tip_dating_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the <code>inference_model</code> uses tip dating,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# Yes, has tip dating
has_strict_clock_model(
  create_inference_model(
    tipdates_filename = get_beautier_path("test_output_0_tipdates.tsv")
  )
)

# No tip dating
has_strict_clock_model(
  create_inference_model()
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='has_xml_closing_tag'>Is an XML closing tag with the value of <code>section</code>
present among the lines of
the text?</h2><span id='topic+has_xml_closing_tag'></span>

<h3>Description</h3>

<p>Is an XML closing tag with the value of <code>section</code>
present among the lines of
the text?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>has_xml_closing_tag(lines, section)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="has_xml_closing_tag_+3A_lines">lines</code></td>
<td>
<p>lines of the XML text</p>
</td></tr>
<tr><td><code id="has_xml_closing_tag_+3A_section">section</code></td>
<td>
<p>the XML section</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if there is an XML closing tag with the value of
<code>section</code> present. FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='has_xml_opening_tag'>Is an XML opening tag with value 'section' present among the lines of
the text?</h2><span id='topic+has_xml_opening_tag'></span>

<h3>Description</h3>

<p>Is an XML opening tag with value 'section' present among the lines of
the text?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>has_xml_opening_tag(lines, section = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="has_xml_opening_tag_+3A_lines">lines</code></td>
<td>
<p>lines of an XML text</p>
</td></tr>
<tr><td><code id="has_xml_opening_tag_+3A_section">section</code></td>
<td>
<p>if NA, this function returns TRUE if there is any
XML opening tag. If <code>section</code> is set to a certain word,
this function returns TRUE if that tag matches <code>section</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='has_xml_short_closing_tag'>Is an XML closing tag with short closing text in
one of the lines of the text?</h2><span id='topic+has_xml_short_closing_tag'></span>

<h3>Description</h3>

<p>Is an XML closing tag with short closing text in
one of the lines of the text?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>has_xml_short_closing_tag(lines)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="has_xml_short_closing_tag_+3A_lines">lines</code></td>
<td>
<p>lines of an XML text</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if there is an XML tag that also closes present in the lines
of text, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
has_xml_short_closing_tag("&lt;my_tag id=1/&gt;")
# FALSE
has_xml_short_closing_tag("&lt;my_tag id=1&gt;text&lt;/my_tag&gt;")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='hky_site_model_to_xml_prior_distr'>Internal function</h2><span id='topic+hky_site_model_to_xml_prior_distr'></span>

<h3>Description</h3>

<p>Converts an HKY site model to XML,
used in the <code>prior</code> section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hky_site_model_to_xml_prior_distr(site_model, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hky_site_model_to_xml_prior_distr_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
<tr><td><code id="hky_site_model_to_xml_prior_distr_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the site model as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>hky_site_model_to_xml_prior_distr(
  site_model = create_hky_site_model(
    id = 1,
    kappa_prior_distr = create_uniform_distr(id = 2)
  ),
  beauti_options = create_beauti_options()
)
</code></pre>

<hr>
<h2 id='hky_site_model_to_xml_state'>Converts a site model to XML,
used in the <code>state</code> section</h2><span id='topic+hky_site_model_to_xml_state'></span>

<h3>Description</h3>

<p>Converts a site model to XML,
used in the <code>state</code> section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hky_site_model_to_xml_state(
  site_model,
  beauti_options = create_beauti_options()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hky_site_model_to_xml_state_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
<tr><td><code id="hky_site_model_to_xml_state_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the site model as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='indent'>Indent text for a certain number of spaces.
If the text is only whitespace, leave it as such</h2><span id='topic+indent'></span>

<h3>Description</h3>

<p>Indent text for a certain number of spaces.
If the text is only whitespace, leave it as such
</p>


<h3>Usage</h3>

<pre><code class='language-R'>indent(text, n_spaces = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="indent_+3A_text">text</code></td>
<td>
<p>the text to indent</p>
</td></tr>
<tr><td><code id="indent_+3A_n_spaces">n_spaces</code></td>
<td>
<p>the number of spaces to add before the text. BEAUti uses
four spaces by default</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the indented text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_bd_tree_prior'>Initializes a Birth-Death tree prior</h2><span id='topic+init_bd_tree_prior'></span>

<h3>Description</h3>

<p>Initializes a Birth-Death tree prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_bd_tree_prior(bd_tree_prior, distr_id, param_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_bd_tree_prior_+3A_bd_tree_prior">bd_tree_prior</code></td>
<td>
<p>a Birth-Death tree prior, as created
by <code><a href="#topic+create_bd_tree_prior">create_bd_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="init_bd_tree_prior_+3A_distr_id">distr_id</code></td>
<td>
<p>a distributions' ID</p>
</td></tr>
<tr><td><code id="init_bd_tree_prior_+3A_param_id">param_id</code></td>
<td>
<p>a parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized Birth-Death tree prior
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_beta_distr'>Initializes a beta distribution</h2><span id='topic+init_beta_distr'></span>

<h3>Description</h3>

<p>Initializes a beta distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_beta_distr(beta_distr, distr_id = 0, param_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_beta_distr_+3A_beta_distr">beta_distr</code></td>
<td>
<p>a beta distribution,
using <a href="#topic+create_beta_distr">create_beta_distr</a></p>
</td></tr>
<tr><td><code id="init_beta_distr_+3A_distr_id">distr_id</code></td>
<td>
<p>the first distribution's ID</p>
</td></tr>
<tr><td><code id="init_beta_distr_+3A_param_id">param_id</code></td>
<td>
<p>the first parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized beta distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_ccp_tree_prior'>Initializes a Coalescent Constant Population tree prior</h2><span id='topic+init_ccp_tree_prior'></span>

<h3>Description</h3>

<p>Initializes a Coalescent Constant Population tree prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_ccp_tree_prior(ccp_tree_prior, distr_id, param_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_ccp_tree_prior_+3A_ccp_tree_prior">ccp_tree_prior</code></td>
<td>
<p>a Coalescent Constant Population tree prior,
as returned by <code><a href="#topic+create_ccp_tree_prior">create_ccp_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="init_ccp_tree_prior_+3A_distr_id">distr_id</code></td>
<td>
<p>a distributions' ID</p>
</td></tr>
<tr><td><code id="init_ccp_tree_prior_+3A_param_id">param_id</code></td>
<td>
<p>a parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized Coalescent Constant Population tree prior
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_cep_tree_prior'>Initializes a Coalescent Exponential Population tree prior</h2><span id='topic+init_cep_tree_prior'></span>

<h3>Description</h3>

<p>Initializes a Coalescent Exponential Population tree prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_cep_tree_prior(cep_tree_prior, distr_id, param_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_cep_tree_prior_+3A_cep_tree_prior">cep_tree_prior</code></td>
<td>
<p>a Coalescent Exponential Population tree prior,
as returned by <code><a href="#topic+create_cep_tree_prior">create_cep_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="init_cep_tree_prior_+3A_distr_id">distr_id</code></td>
<td>
<p>a distributions' ID</p>
</td></tr>
<tr><td><code id="init_cep_tree_prior_+3A_param_id">param_id</code></td>
<td>
<p>a parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized Coalescent Exponential Population tree prior
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_clock_models'>Initializes all clock models</h2><span id='topic+init_clock_models'></span>

<h3>Description</h3>

<p>Initializes all clock models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_clock_models(fasta_filenames, clock_models, distr_id = 0, param_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_clock_models_+3A_fasta_filenames">fasta_filenames</code></td>
<td>
<p>One or more FASTA filenames.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.</p>
</td></tr>
<tr><td><code id="init_clock_models_+3A_clock_models">clock_models</code></td>
<td>
<p>a list of one or more clock models,
as returned by <code><a href="#topic+create_clock_model">create_clock_model</a></code></p>
</td></tr>
<tr><td><code id="init_clock_models_+3A_distr_id">distr_id</code></td>
<td>
<p>the first distributions' ID</p>
</td></tr>
<tr><td><code id="init_clock_models_+3A_param_id">param_id</code></td>
<td>
<p>the first parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of initialized clock models
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_distr'>Initializes a distribution</h2><span id='topic+init_distr'></span>

<h3>Description</h3>

<p>Initializes a distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_distr(distr, distr_id = 0, param_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_distr_+3A_distr">distr</code></td>
<td>
<p>a distribution,
using <code><a href="#topic+create_distr">create_distr</a></code></p>
</td></tr>
<tr><td><code id="init_distr_+3A_distr_id">distr_id</code></td>
<td>
<p>the first distribution's ID</p>
</td></tr>
<tr><td><code id="init_distr_+3A_param_id">param_id</code></td>
<td>
<p>the first parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_exp_distr'>Initializes an exponential distribution</h2><span id='topic+init_exp_distr'></span>

<h3>Description</h3>

<p>Initializes an exponential distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_exp_distr(exp_distr, distr_id = 0, param_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_exp_distr_+3A_exp_distr">exp_distr</code></td>
<td>
<p>a exponential distribution,
using <a href="#topic+create_exp_distr">create_exp_distr</a></p>
</td></tr>
<tr><td><code id="init_exp_distr_+3A_distr_id">distr_id</code></td>
<td>
<p>the first distribution's ID</p>
</td></tr>
<tr><td><code id="init_exp_distr_+3A_param_id">param_id</code></td>
<td>
<p>the first parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized exponential distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_gamma_distr'>Initializes a gamma distribution</h2><span id='topic+init_gamma_distr'></span>

<h3>Description</h3>

<p>Initializes a gamma distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_gamma_distr(gamma_distr, distr_id = 0, param_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_gamma_distr_+3A_gamma_distr">gamma_distr</code></td>
<td>
<p>a gamma distribution,
using <a href="#topic+create_gamma_distr">create_gamma_distr</a></p>
</td></tr>
<tr><td><code id="init_gamma_distr_+3A_distr_id">distr_id</code></td>
<td>
<p>the first distribution's ID</p>
</td></tr>
<tr><td><code id="init_gamma_distr_+3A_param_id">param_id</code></td>
<td>
<p>the first parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized gamma distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_gamma_site_model'>Initializes a gamma site model</h2><span id='topic+init_gamma_site_model'></span>

<h3>Description</h3>

<p>Initializes a gamma site model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_gamma_site_model(gamma_site_model, distr_id = 0, param_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_gamma_site_model_+3A_gamma_site_model">gamma_site_model</code></td>
<td>
<p>a site model's gamma site model,
as returned by <code><a href="#topic+create_gamma_site_model">create_gamma_site_model</a></code></p>
</td></tr>
<tr><td><code id="init_gamma_site_model_+3A_distr_id">distr_id</code></td>
<td>
<p>the first distributions' ID</p>
</td></tr>
<tr><td><code id="init_gamma_site_model_+3A_param_id">param_id</code></td>
<td>
<p>the first parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized gamma site model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

gamma_site_model &lt;- create_gamma_site_model(
  gamma_cat_count = 2,
  gamma_shape_prior_distr = create_one_div_x_distr(id = NA)
)
# FALSE: not yet initialized
is_init_gamma_site_model(gamma_site_model)
gamma_site_model &lt;- init_gamma_site_model(gamma_site_model)
# TRUE: now it is initialized
is_init_gamma_site_model(gamma_site_model)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='init_gtr_site_model'>Initializes a GTR site model</h2><span id='topic+init_gtr_site_model'></span>

<h3>Description</h3>

<p>Initializes a GTR site model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_gtr_site_model(gtr_site_model, distr_id = 0, param_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_gtr_site_model_+3A_gtr_site_model">gtr_site_model</code></td>
<td>
<p>a GTR site model,
as returned by <code><a href="#topic+create_gtr_site_model">create_gtr_site_model</a></code></p>
</td></tr>
<tr><td><code id="init_gtr_site_model_+3A_distr_id">distr_id</code></td>
<td>
<p>a distributions' ID</p>
</td></tr>
<tr><td><code id="init_gtr_site_model_+3A_param_id">param_id</code></td>
<td>
<p>a parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized GTR site model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

gtr_site_model &lt;- create_gtr_site_model()
# FALSE
is_init_gtr_site_model(gtr_site_model)
gtr_site_model &lt;- init_gtr_site_model(gtr_site_model)
# TRUE
is_init_gtr_site_model(gtr_site_model)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='init_hky_site_model'>Initializes an HKY site model</h2><span id='topic+init_hky_site_model'></span>

<h3>Description</h3>

<p>Initializes an HKY site model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_hky_site_model(hky_site_model, distr_id = 0, param_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_hky_site_model_+3A_hky_site_model">hky_site_model</code></td>
<td>
<p>an HKY site model,
as returned by <code><a href="#topic+create_hky_site_model">create_hky_site_model</a></code></p>
</td></tr>
<tr><td><code id="init_hky_site_model_+3A_distr_id">distr_id</code></td>
<td>
<p>a distributions' ID</p>
</td></tr>
<tr><td><code id="init_hky_site_model_+3A_param_id">param_id</code></td>
<td>
<p>a parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized HKY site model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

hky_site_model &lt;- create_hky_site_model()
is_init_hky_site_model(hky_site_model)
hky_site_model &lt;- init_hky_site_model(hky_site_model)
is_init_hky_site_model(hky_site_model)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='init_inference_model'>Initialize an inference model</h2><span id='topic+init_inference_model'></span>

<h3>Description</h3>

<p>Initialize an inference model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_inference_model(input_filename, inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_inference_model_+3A_input_filename">input_filename</code></td>
<td>
<p>A FASTA filename.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.</p>
</td></tr>
<tr><td><code id="init_inference_model_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized inference model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_inv_gamma_distr'>Initializes an inverse gamma distribution</h2><span id='topic+init_inv_gamma_distr'></span>

<h3>Description</h3>

<p>Initializes an inverse gamma distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_inv_gamma_distr(inv_gamma_distr, distr_id = 0, param_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_inv_gamma_distr_+3A_inv_gamma_distr">inv_gamma_distr</code></td>
<td>
<p>an inverse gamma distribution,
using <a href="#topic+create_inv_gamma_distr">create_inv_gamma_distr</a></p>
</td></tr>
<tr><td><code id="init_inv_gamma_distr_+3A_distr_id">distr_id</code></td>
<td>
<p>the first distribution's ID</p>
</td></tr>
<tr><td><code id="init_inv_gamma_distr_+3A_param_id">param_id</code></td>
<td>
<p>the first parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized inverse gamma distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_jc69_site_model'>Initializes a JC69 site model</h2><span id='topic+init_jc69_site_model'></span>

<h3>Description</h3>

<p>Initializes a JC69 site model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_jc69_site_model(jc69_site_model, distr_id = 0, param_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_jc69_site_model_+3A_jc69_site_model">jc69_site_model</code></td>
<td>
<p>a JC69 site model,
as returned by <code><a href="#topic+create_jc69_site_model">create_jc69_site_model</a></code></p>
</td></tr>
<tr><td><code id="init_jc69_site_model_+3A_distr_id">distr_id</code></td>
<td>
<p>a distributions' ID</p>
</td></tr>
<tr><td><code id="init_jc69_site_model_+3A_param_id">param_id</code></td>
<td>
<p>a parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized HKY site model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

hky_site_model &lt;- create_hky_site_model()
is_init_hky_site_model(hky_site_model)
hky_site_model &lt;- init_hky_site_model(hky_site_model)
is_init_hky_site_model(hky_site_model)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='init_laplace_distr'>Initializes an Laplace distribution</h2><span id='topic+init_laplace_distr'></span>

<h3>Description</h3>

<p>Initializes an Laplace distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_laplace_distr(laplace_distr, distr_id = 0, param_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_laplace_distr_+3A_laplace_distr">laplace_distr</code></td>
<td>
<p>a Laplace distribution,
using <a href="#topic+create_laplace_distr">create_laplace_distr</a></p>
</td></tr>
<tr><td><code id="init_laplace_distr_+3A_distr_id">distr_id</code></td>
<td>
<p>the first distribution's ID</p>
</td></tr>
<tr><td><code id="init_laplace_distr_+3A_param_id">param_id</code></td>
<td>
<p>the first parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized Laplace distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_log_normal_distr'>Initializes an log-normal distribution</h2><span id='topic+init_log_normal_distr'></span>

<h3>Description</h3>

<p>Initializes an log-normal distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_log_normal_distr(log_normal_distr, distr_id = 0, param_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_log_normal_distr_+3A_log_normal_distr">log_normal_distr</code></td>
<td>
<p>a log-normal distribution,
using <a href="#topic+create_log_normal_distr">create_log_normal_distr</a></p>
</td></tr>
<tr><td><code id="init_log_normal_distr_+3A_distr_id">distr_id</code></td>
<td>
<p>the first distribution's ID</p>
</td></tr>
<tr><td><code id="init_log_normal_distr_+3A_param_id">param_id</code></td>
<td>
<p>the first parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized log-normal distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_mrca_prior'>Initialize the MRCA prior.</h2><span id='topic+init_mrca_prior'></span>

<h3>Description</h3>

<p>Initialized by
</p>

<ul>
<li><p> if no alignment ID is set,
it is set by reading it from the alignment file
</p>
</li>
<li><p> if no taxa names are set,
these are set by reading these from the alignment file
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>init_mrca_prior(input_filename, inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_mrca_prior_+3A_input_filename">input_filename</code></td>
<td>
<p>A FASTA filename.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.</p>
</td></tr>
<tr><td><code id="init_mrca_prior_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized MRCA prior
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_mrca_priors'>Initializes all MRCA priors</h2><span id='topic+init_mrca_priors'></span>

<h3>Description</h3>

<p>Initializes all MRCA priors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_mrca_priors(mrca_priors, distr_id = 0, param_id = 0, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_mrca_priors_+3A_mrca_priors">mrca_priors</code></td>
<td>
<p>a list of one or more Most Recent Common Ancestor priors,
as returned by <code><a href="#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
<tr><td><code id="init_mrca_priors_+3A_distr_id">distr_id</code></td>
<td>
<p>the first distributions' ID</p>
</td></tr>
<tr><td><code id="init_mrca_priors_+3A_param_id">param_id</code></td>
<td>
<p>the first parameter's ID</p>
</td></tr>
<tr><td><code id="init_mrca_priors_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of initialized MRCA priors
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_normal_distr'>Initializes an normal distribution</h2><span id='topic+init_normal_distr'></span>

<h3>Description</h3>

<p>Initializes an normal distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_normal_distr(normal_distr, distr_id = 0, param_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_normal_distr_+3A_normal_distr">normal_distr</code></td>
<td>
<p>a normal distribution,
using <a href="#topic+create_normal_distr">create_normal_distr</a></p>
</td></tr>
<tr><td><code id="init_normal_distr_+3A_distr_id">distr_id</code></td>
<td>
<p>the first distribution's ID</p>
</td></tr>
<tr><td><code id="init_normal_distr_+3A_param_id">param_id</code></td>
<td>
<p>the first parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized normal distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_one_div_x_distr'>Initializes an one-divided-by-x distribution</h2><span id='topic+init_one_div_x_distr'></span>

<h3>Description</h3>

<p>Initializes an one-divided-by-x distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_one_div_x_distr(one_div_x_distr, distr_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_one_div_x_distr_+3A_one_div_x_distr">one_div_x_distr</code></td>
<td>
<p>a one-divided-by-x distribution,
using <a href="#topic+create_one_div_x_distr">create_one_div_x_distr</a></p>
</td></tr>
<tr><td><code id="init_one_div_x_distr_+3A_distr_id">distr_id</code></td>
<td>
<p>the first distribution's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized one-divided-by-x distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_param'>Initializes a parameter</h2><span id='topic+init_param'></span>

<h3>Description</h3>

<p>Initializes a parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_param(param, id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_param_+3A_param">param</code></td>
<td>
<p>a parameter,
using <code><a href="#topic+create_param">create_param</a></code></p>
</td></tr>
<tr><td><code id="init_param_+3A_id">id</code></td>
<td>
<p>the parameter's ID. Will be ignored if the parameter already
has an ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized parameter
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_poisson_distr'>Initializes an Poisson distribution</h2><span id='topic+init_poisson_distr'></span>

<h3>Description</h3>

<p>Initializes an Poisson distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_poisson_distr(poisson_distr, distr_id = 0, param_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_poisson_distr_+3A_poisson_distr">poisson_distr</code></td>
<td>
<p>a Poisson distribution,
using <a href="#topic+create_poisson_distr">create_poisson_distr</a></p>
</td></tr>
<tr><td><code id="init_poisson_distr_+3A_distr_id">distr_id</code></td>
<td>
<p>the first distribution's ID</p>
</td></tr>
<tr><td><code id="init_poisson_distr_+3A_param_id">param_id</code></td>
<td>
<p>the first parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized Poisson distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_rln_clock_model'>Initializes a Relaxed Log-Normal clock model</h2><span id='topic+init_rln_clock_model'></span>

<h3>Description</h3>

<p>Initializes a Relaxed Log-Normal clock model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_rln_clock_model(rln_clock_model, distr_id = 0, param_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_rln_clock_model_+3A_rln_clock_model">rln_clock_model</code></td>
<td>
<p>a Relaxed Log-Normal clock model,
as returned by <code><a href="#topic+create_rln_clock_model">create_rln_clock_model</a></code></p>
</td></tr>
<tr><td><code id="init_rln_clock_model_+3A_distr_id">distr_id</code></td>
<td>
<p>a distributions' ID</p>
</td></tr>
<tr><td><code id="init_rln_clock_model_+3A_param_id">param_id</code></td>
<td>
<p>a parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized Relaxed Log-Normal clock model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

rln_clock_model &lt;- create_rln_clock_model()
# FALSE: not yet initialized
is_init_rln_clock_model(rln_clock_model)
rln_clock_model &lt;- init_rln_clock_model(rln_clock_model)
# Dimension is set to NA by default, for unknown reasons.
# Because 'init_rln_clock_model' does not initialize it (for
# unknown reasons), set it manually
rln_clock_model$dimension &lt;- 42
# TRUE: now it is initialized
is_init_rln_clock_model(rln_clock_model)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='init_site_models'>Initializes all site models</h2><span id='topic+init_site_models'></span>

<h3>Description</h3>

<p>Initializes all site models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_site_models(site_models, ids, distr_id = 0, param_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_site_models_+3A_site_models">site_models</code></td>
<td>
<p>one or more site models,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
<tr><td><code id="init_site_models_+3A_ids">ids</code></td>
<td>
<p>one or more alignments' IDs.
IDs can be extracted from their FASTA filenames
with <code><a href="#topic+get_alignment_ids_from_fasta_filenames">get_alignment_ids_from_fasta_filenames</a></code>)</p>
</td></tr>
<tr><td><code id="init_site_models_+3A_distr_id">distr_id</code></td>
<td>
<p>the first distributions' ID</p>
</td></tr>
<tr><td><code id="init_site_models_+3A_param_id">param_id</code></td>
<td>
<p>the first parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of initialized site models
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_strict_clock_model'>Initializes a strict clock model</h2><span id='topic+init_strict_clock_model'></span>

<h3>Description</h3>

<p>Initializes a strict clock model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_strict_clock_model(strict_clock_model, distr_id = 0, param_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_strict_clock_model_+3A_strict_clock_model">strict_clock_model</code></td>
<td>
<p>a strict clock model,
as returned by <code><a href="#topic+create_strict_clock_model">create_strict_clock_model</a></code></p>
</td></tr>
<tr><td><code id="init_strict_clock_model_+3A_distr_id">distr_id</code></td>
<td>
<p>a distributions' ID</p>
</td></tr>
<tr><td><code id="init_strict_clock_model_+3A_param_id">param_id</code></td>
<td>
<p>a parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized strict clock model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

strict_clock_model &lt;- create_strict_clock_model()
# FALSE: not yet initialized
is_init_strict_clock_model(strict_clock_model)
strict_clock_model &lt;- init_strict_clock_model(strict_clock_model)
# TRUE: initialized
is_init_strict_clock_model(strict_clock_model)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='init_tn93_site_model'>Initializes a TN93 site model</h2><span id='topic+init_tn93_site_model'></span>

<h3>Description</h3>

<p>Initializes a TN93 site model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_tn93_site_model(tn93_site_model, distr_id = 0, param_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_tn93_site_model_+3A_tn93_site_model">tn93_site_model</code></td>
<td>
<p>a TN93 site model,
as returned by <code><a href="#topic+create_tn93_site_model">create_tn93_site_model</a></code></p>
</td></tr>
<tr><td><code id="init_tn93_site_model_+3A_distr_id">distr_id</code></td>
<td>
<p>a distributions' ID</p>
</td></tr>
<tr><td><code id="init_tn93_site_model_+3A_param_id">param_id</code></td>
<td>
<p>a parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized TN93 site model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

tn93_site_model &lt;- create_tn93_site_model()
is_init_tn93_site_model(tn93_site_model)
tn93_site_model &lt;- init_tn93_site_model(tn93_site_model)
is_init_tn93_site_model(tn93_site_model)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='init_tree_priors'>Initializes all tree priors</h2><span id='topic+init_tree_priors'></span>

<h3>Description</h3>

<p>Initializes all tree priors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_tree_priors(tree_priors, ids, distr_id = 0, param_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_tree_priors_+3A_tree_priors">tree_priors</code></td>
<td>
<p>one or more tree priors,
as returned by <code><a href="#topic+create_tree_prior">create_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="init_tree_priors_+3A_ids">ids</code></td>
<td>
<p>one or more alignments' IDs.
IDs can be extracted from their FASTA filenames
with <code><a href="#topic+get_alignment_ids_from_fasta_filenames">get_alignment_ids_from_fasta_filenames</a></code>)</p>
</td></tr>
<tr><td><code id="init_tree_priors_+3A_distr_id">distr_id</code></td>
<td>
<p>the first distributions' ID</p>
</td></tr>
<tr><td><code id="init_tree_priors_+3A_param_id">param_id</code></td>
<td>
<p>the first parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of initialized tree priors
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_uniform_distr'>Initializes a uniform distribution</h2><span id='topic+init_uniform_distr'></span>

<h3>Description</h3>

<p>Initializes a uniform distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_uniform_distr(uniform_distr, distr_id = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_uniform_distr_+3A_uniform_distr">uniform_distr</code></td>
<td>
<p>a uniform distribution,
using <a href="#topic+create_uniform_distr">create_uniform_distr</a></p>
</td></tr>
<tr><td><code id="init_uniform_distr_+3A_distr_id">distr_id</code></td>
<td>
<p>the first distribution's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized uniform distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='init_yule_tree_prior'>Initializes a Yule tree prior</h2><span id='topic+init_yule_tree_prior'></span>

<h3>Description</h3>

<p>Initializes a Yule tree prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_yule_tree_prior(yule_tree_prior, distr_id, param_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_yule_tree_prior_+3A_yule_tree_prior">yule_tree_prior</code></td>
<td>
<p>a Yule tree_prior,
as created by <code><a href="#topic+create_yule_tree_prior">create_yule_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="init_yule_tree_prior_+3A_distr_id">distr_id</code></td>
<td>
<p>a distributions' ID</p>
</td></tr>
<tr><td><code id="init_yule_tree_prior_+3A_param_id">param_id</code></td>
<td>
<p>a parameter's ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an initialized Yule tree prior
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='interspace'>Puts spaces in between the lines</h2><span id='topic+interspace'></span>

<h3>Description</h3>

<p>Puts spaces in between the lines
</p>


<h3>Usage</h3>

<pre><code class='language-R'>interspace(lines)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="interspace_+3A_lines">lines</code></td>
<td>
<p>lines of text</p>
</td></tr>
</table>


<h3>Value</h3>

<p>interspaced lines of text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_alpha_param'>Determine if the object is a valid
alpha parameter</h2><span id='topic+is_alpha_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
alpha parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_alpha_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_alpha_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
alpha parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid alpha parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_alpha_param(create_alpha_param())
is_alpha_param(create_beta_param())
is_alpha_param(create_clock_rate_param())
is_alpha_param(create_kappa_1_param())
is_alpha_param(create_kappa_2_param())
is_alpha_param(create_lambda_param())
is_alpha_param(create_m_param())
is_alpha_param(create_mean_param())
is_alpha_param(create_mu_param())
is_alpha_param(create_rate_ac_param())
is_alpha_param(create_rate_ag_param())
is_alpha_param(create_rate_at_param())
is_alpha_param(create_rate_cg_param())
is_alpha_param(create_rate_ct_param())
is_alpha_param(create_rate_gt_param())
is_alpha_param(create_s_param())
is_alpha_param(create_scale_param())
is_alpha_param(create_sigma_param())

is_alpha_param(NA)
is_alpha_param(NULL)
is_alpha_param("nonsense")
is_alpha_param(create_jc69_site_model())
is_alpha_param(create_strict_clock_model())
is_alpha_param(create_yule_tree_prior())
is_alpha_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_b_pop_sizes_param'>Determine if the object is a valid
b_pop_sizes parameter</h2><span id='topic+is_b_pop_sizes_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
b_pop_sizes parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_b_pop_sizes_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_b_pop_sizes_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
b_pop_sizes parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid b_pop_sizes parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_b_pop_sizes_param(create_alpha_param())
is_b_pop_sizes_param(create_b_pop_sizes_param())
is_b_pop_sizes_param(create_beta_param())
is_b_pop_sizes_param(create_clock_rate_param())
is_b_pop_sizes_param(create_kappa_1_param())
is_b_pop_sizes_param(create_kappa_2_param())
is_b_pop_sizes_param(create_lambda_param())
is_b_pop_sizes_param(create_m_param())
is_b_pop_sizes_param(create_mean_param())
is_b_pop_sizes_param(create_mu_param())
is_b_pop_sizes_param(create_rate_ac_param())
is_b_pop_sizes_param(create_rate_ag_param())
is_b_pop_sizes_param(create_rate_at_param())
is_b_pop_sizes_param(create_rate_cg_param())
is_b_pop_sizes_param(create_rate_ct_param())
is_b_pop_sizes_param(create_rate_gt_param())
is_b_pop_sizes_param(create_s_param())
is_b_pop_sizes_param(create_scale_param())
is_b_pop_sizes_param(create_sigma_param())

is_b_pop_sizes_param(NA)
is_b_pop_sizes_param(NULL)
is_b_pop_sizes_param("nonsense")
is_b_pop_sizes_param(create_jc69_site_model())
is_b_pop_sizes_param(create_strict_clock_model())
is_b_pop_sizes_param(create_yule_tree_prior())
is_b_pop_sizes_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_bd_tree_prior'>Determine if the object is a valid Birth Death tree prior</h2><span id='topic+is_bd_tree_prior'></span>

<h3>Description</h3>

<p>Determine if the object is a valid Birth Death tree prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_bd_tree_prior(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_bd_tree_prior_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid birth death tree prior</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid birth death tree prior, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+create_bd_tree_prior">create_bd_tree_prior</a></code> to create a valid
Birth-Death tree prior
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

  is_bd_tree_prior(create_bd_tree_prior())
  !is_bd_tree_prior(create_cbs_tree_prior())
  !is_bd_tree_prior(create_ccp_tree_prior())
  !is_bd_tree_prior(create_cep_tree_prior())
  !is_bd_tree_prior(create_yule_tree_prior())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_beauti_options'>Determine if the object is a valid <code>beauti_options</code></h2><span id='topic+is_beauti_options'></span>

<h3>Description</h3>

<p>Determine if the object is a valid <code>beauti_options</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_beauti_options(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_beauti_options_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a <code>beauti_options</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><a href="base.html#topic+TRUE">TRUE</a> if the object is a valid <code>beauti_options</code>,
<a href="base.html#topic+FALSE">FALSE</a> otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <a href="#topic+create_beauti_options">create_beauti_options</a> to create a valid
<code>beauti_options</code> object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_beauti_options(create_beauti_options())

# FALSE
is_beauti_options("nonsense")
is_beauti_options(NA)
is_beauti_options(NULL)
is_beauti_options("")
is_beauti_options(c())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_beta_distr'>Determine if the object is a valid
beta distribution,
as created by <code><a href="#topic+create_beta_distr">create_beta_distr</a></code></h2><span id='topic+is_beta_distr'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
beta distribution,
as created by <code><a href="#topic+create_beta_distr">create_beta_distr</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_beta_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_beta_distr_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
beta distribution,</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid beta distribution,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <code><a href="#topic+is_distr">is_distr</a></code> to see if x is any
distribution
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_beta_distr(create_beta_distr())
# FALSE
is_beta_distr(create_exp_distr())
is_beta_distr(NA)
is_beta_distr(NULL)
is_beta_distr("nonsense")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_beta_param'>Determine if the object is a valid
beta parameter</h2><span id='topic+is_beta_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
beta parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_beta_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_beta_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
beta parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid beta parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_beta_param(create_alpha_param())
is_beta_param(create_beta_param())
is_beta_param(create_clock_rate_param())
is_beta_param(create_kappa_1_param())
is_beta_param(create_kappa_2_param())
is_beta_param(create_lambda_param())
is_beta_param(create_m_param())
is_beta_param(create_mean_param())
is_beta_param(create_mu_param())
is_beta_param(create_rate_ac_param())
is_beta_param(create_rate_ag_param())
is_beta_param(create_rate_at_param())
is_beta_param(create_rate_cg_param())
is_beta_param(create_rate_ct_param())
is_beta_param(create_rate_gt_param())
is_beta_param(create_s_param())
is_beta_param(create_scale_param())
is_beta_param(create_sigma_param())

is_beta_param(NA)
is_beta_param(NULL)
is_beta_param("nonsense")
is_beta_param(create_jc69_site_model())
is_beta_param(create_strict_clock_model())
is_beta_param(create_yule_tree_prior())
is_beta_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_cbs_tree_prior'>Determine if the object is a valid constant coalescent Bayesian skyline prior</h2><span id='topic+is_cbs_tree_prior'></span>

<h3>Description</h3>

<p>Determine if the object is a valid constant coalescent Bayesian skyline prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_cbs_tree_prior(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_cbs_tree_prior_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid constant coalescent
Bayesian skyline prior</p>
</td></tr>
</table>


<h3>Value</h3>

<p>'TRUE' if 'x' is a valid constant coalescent Bayesian skyline prior,
'FALSE' otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+create_cbs_tree_prior">create_cbs_tree_prior</a></code> to create a valid
coalescent Bayes skyline tree prior
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_cbs_tree_prior(create_cbs_tree_prior())

# FALSE
is_cbs_tree_prior(create_bd_tree_prior())
is_cbs_tree_prior(create_ccp_tree_prior())
is_cbs_tree_prior(create_cep_tree_prior())
is_cbs_tree_prior(create_yule_tree_prior())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_ccp_tree_prior'>Determine if the object is a valid
constant coalescence population tree prior</h2><span id='topic+is_ccp_tree_prior'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
constant coalescence population tree prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_ccp_tree_prior(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_ccp_tree_prior_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
constant coalescence population tree prior</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid constant coalescence population tree prior,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+create_ccp_tree_prior">create_ccp_tree_prior</a></code> to create a valid
constant coalescence population tree prior
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()
  !is_ccp_tree_prior(create_bd_tree_prior())
  !is_ccp_tree_prior(create_cbs_tree_prior())
  is_ccp_tree_prior(create_ccp_tree_prior())
  !is_ccp_tree_prior(create_cep_tree_prior())
  !is_ccp_tree_prior(create_yule_tree_prior())
check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_cep_tree_prior'>Determine if the object is a valid
coalescent exponential population tree prior</h2><span id='topic+is_cep_tree_prior'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
coalescent exponential population tree prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_cep_tree_prior(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_cep_tree_prior_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
constant coalescent exponential population tree prior</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid coalescent exponential population tree prior,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+create_cep_tree_prior">create_cep_tree_prior</a></code> to create a valid
coalescent exponential population tree prior
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

  !is_cep_tree_prior(create_bd_tree_prior())
  !is_cep_tree_prior(create_cbs_tree_prior())
  !is_cep_tree_prior(create_ccp_tree_prior())
  is_cep_tree_prior(create_cep_tree_prior())
  !is_cep_tree_prior(create_yule_tree_prior())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_clock_model'>Determine if the object is a valid clock_model</h2><span id='topic+is_clock_model'></span>

<h3>Description</h3>

<p>Determine if the object is a valid clock_model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_clock_model(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_clock_model_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a clock_model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the clock_model is a valid clock_model, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>see <code><a href="#topic+create_clock_model">create_clock_model</a></code> for an overview of functions
to create valid clock model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_clock_model(create_strict_clock_model())
is_clock_model(create_rln_clock_model())

# FALSE
is_clock_model(NA)
is_clock_model(NULL)
is_clock_model("nonsense")
is_clock_model(create_jc69_site_model())
is_clock_model(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_clock_model_name'>Determines if the name is a valid clock model name</h2><span id='topic+is_clock_model_name'></span>

<h3>Description</h3>

<p>Determines if the name is a valid clock model name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_clock_model_name(name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_clock_model_name_+3A_name">name</code></td>
<td>
<p>the name to be tested</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the name is a valid clock_model name, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_clock_model_name("relaxed_log_normal")
is_clock_model_name("strict")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_clock_rate_param'>Determine if the object is a valid
clock_rate parameter</h2><span id='topic+is_clock_rate_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
clock_rate parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_clock_rate_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_clock_rate_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
clock_rate parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid clock_rate parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_clock_rate_param(create_alpha_param())
is_clock_rate_param(create_beta_param())
is_clock_rate_param(create_clock_rate_param())
is_clock_rate_param(create_kappa_1_param())
is_clock_rate_param(create_kappa_2_param())
is_clock_rate_param(create_lambda_param())
is_clock_rate_param(create_m_param())
is_clock_rate_param(create_mean_param())
is_clock_rate_param(create_mu_param())
is_clock_rate_param(create_rate_ac_param())
is_clock_rate_param(create_rate_ag_param())
is_clock_rate_param(create_rate_at_param())
is_clock_rate_param(create_rate_cg_param())
is_clock_rate_param(create_rate_ct_param())
is_clock_rate_param(create_rate_gt_param())
is_clock_rate_param(create_s_param())
is_clock_rate_param(create_scale_param())
is_clock_rate_param(create_sigma_param())

is_clock_rate_param(NA)
is_clock_rate_param(NULL)
is_clock_rate_param("nonsense")
is_clock_rate_param(create_jc69_site_model())
is_clock_rate_param(create_strict_clock_model())
is_clock_rate_param(create_yule_tree_prior())
is_clock_rate_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_default_mcmc'>Determine if the MCMC is a default MCMC</h2><span id='topic+is_default_mcmc'></span>

<h3>Description</h3>

<p>Determine if the MCMC is a default MCMC
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_default_mcmc(mcmc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_default_mcmc_+3A_mcmc">mcmc</code></td>
<td>
<p>one MCMC.
Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create an MCMC.
Use <code><a href="#topic+create_ns_mcmc">create_ns_mcmc</a></code> to create an MCMC
for a Nested Sampling run.
Use <code><a href="#topic+check_mcmc">check_mcmc</a></code> to check if an MCMC is valid.
Use <code><a href="#topic+rename_mcmc_filenames">rename_mcmc_filenames</a></code> to rename the filenames in an MCMC.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the MCMC is a default MCMC
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  check_empty_beautier_folder()

  # TRUE: An MCMC created by 'create_mcmc' is default.
  is_default_mcmc(create_mcmc())

  # FALSE: An MCMC created by 'create_ns_mcmc' is not
  is_default_mcmc(create_ns_mcmc())

  check_empty_beautier_folder()
}
</code></pre>

<hr>
<h2 id='is_distr'>Determine if the object is a valid distribution</h2><span id='topic+is_distr'></span>

<h3>Description</h3>

<p>Determine if the object is a valid distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_distr_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid distribution,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use
<code><a href="#topic+is_beta_distr">is_beta_distr</a></code>,
<code><a href="#topic+is_exp_distr">is_exp_distr</a></code>,
<code><a href="#topic+is_gamma_distr">is_gamma_distr</a></code>,
<code><a href="#topic+is_inv_gamma_distr">is_inv_gamma_distr</a></code>,
<code><a href="#topic+is_laplace_distr">is_laplace_distr</a></code>,
<code><a href="#topic+is_log_normal_distr">is_log_normal_distr</a></code>,
<code><a href="#topic+is_normal_distr">is_normal_distr</a></code>,
<code><a href="#topic+is_one_div_x_distr">is_one_div_x_distr</a></code>,
<code><a href="#topic+is_poisson_distr">is_poisson_distr</a></code>,
or <code><a href="#topic+is_uniform_distr">is_uniform_distr</a></code>,
to check for more specific distribution
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_distr(create_beta_distr())
is_distr(create_exp_distr())
is_distr(create_gamma_distr())
is_distr(create_inv_gamma_distr())
is_distr(create_laplace_distr())
is_distr(create_log_normal_distr())
is_distr(create_normal_distr())
is_distr(create_one_div_x_distr())
is_distr(create_poisson_distr())
is_distr(create_uniform_distr())

# FALSE
is_distr(NA)
is_distr(NULL)
is_distr("nonsense")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_distr_name'>Determines if the name is a valid distribution name</h2><span id='topic+is_distr_name'></span>

<h3>Description</h3>

<p>Determines if the name is a valid distribution name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_distr_name(name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_distr_name_+3A_name">name</code></td>
<td>
<p>the name to be tested</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the name is a valid distribution name, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_distr_name("uniform")
is_distr_name("normal")
is_distr_name("one_div_x")
is_distr_name("log_normal")
is_distr_name("exponential")
is_distr_name("gamma")
is_distr_name("beta")
is_distr_name("laplace")
is_distr_name("inv_gamma")
is_distr_name("poisson")
# FALSE
is_distr_name("nonsense")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_exp_distr'>Determine if the object is a valid
exponential distribution
as created by <code><a href="#topic+create_exp_distr">create_exp_distr</a></code></h2><span id='topic+is_exp_distr'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
exponential distribution
as created by <code><a href="#topic+create_exp_distr">create_exp_distr</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_exp_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_exp_distr_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
exponential distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid exponential distribution,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <code><a href="#topic+is_distr">is_distr</a></code> to see if x is any
distribution
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_exp_distr(create_exp_distr())
# FALSE
is_exp_distr(create_gamma_distr())
is_exp_distr(NA)
is_exp_distr(NULL)
is_exp_distr("nonsense")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_freq_equilibrium_name'>Checks if <code>name</code> is a valid <code>freq_equilibrium</code> argument value</h2><span id='topic+is_freq_equilibrium_name'></span>

<h3>Description</h3>

<p>Checks if <code>name</code> is a valid <code>freq_equilibrium</code> argument value
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_freq_equilibrium_name(name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_freq_equilibrium_name_+3A_name">name</code></td>
<td>
<p>the name to check if it is a valid <code>freq_equilibrium</code>
argument value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the name is a valid <code>freq_equilibrium</code> value
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the <code>freq_equilibrium</code> argument is used by
<code><a href="#topic+create_gtr_site_model">create_gtr_site_model</a></code>,
<code><a href="#topic+create_hky_site_model">create_hky_site_model</a></code>,
and <code><a href="#topic+create_tn93_site_model">create_tn93_site_model</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_freq_equilibrium_name("estimated")
is_freq_equilibrium_name("empirical")
is_freq_equilibrium_name("all_equal")
# FALSE
is_freq_equilibrium_name("nonsense")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_freq_param'>Determine if the object is a valid freq parameter</h2><span id='topic+is_freq_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid freq parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_freq_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_freq_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
freq parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid freq parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>freq parameters are returned by
<code><a href="#topic+create_freq_param">create_freq_param</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_freq_param(create_alpha_param())
is_freq_param(create_beta_param())
is_freq_param(create_clock_rate_param())
is_freq_param(create_freq_param())
is_freq_param(create_freq_param())
is_freq_param(create_kappa_param())
is_freq_param(create_kappa_1_param())
is_freq_param(create_kappa_2_param())
is_freq_param(create_lambda_param())
is_freq_param(create_m_param())
is_freq_param(create_mean_param())
is_freq_param(create_mu_param())
is_freq_param(create_rate_ac_param())
is_freq_param(create_rate_ag_param())
is_freq_param(create_rate_at_param())
is_freq_param(create_rate_cg_param())
is_freq_param(create_rate_ct_param())
is_freq_param(create_rate_gt_param())
is_freq_param(create_s_param())
is_freq_param(create_scale_param())
is_freq_param(create_sigma_param())

is_freq_param(NA)
is_freq_param(NULL)
is_freq_param("nonsense")
is_freq_param(create_jc69_site_model())
is_freq_param(create_strict_clock_model())
is_freq_param(create_yule_tree_prior())
is_freq_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_gamma_distr'>Determine if the object is a valid
gamma distribution,
as created by <code><a href="#topic+create_gamma_distr">create_gamma_distr</a></code></h2><span id='topic+is_gamma_distr'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
gamma distribution,
as created by <code><a href="#topic+create_gamma_distr">create_gamma_distr</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_gamma_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_gamma_distr_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
gamma distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid gamma distribution,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <code><a href="#topic+is_distr">is_distr</a></code> to see if x is any
distribution
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_gamma_distr(create_gamma_distr())
# FALSE
is_gamma_distr(create_inv_gamma_distr())
is_gamma_distr(NA)
is_gamma_distr(NULL)
is_gamma_distr("nonsense")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_gamma_site_model'>Is object x a gamma site model?</h2><span id='topic+is_gamma_site_model'></span>

<h3>Description</h3>

<p>Is object x a gamma site model?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_gamma_site_model(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_gamma_site_model_+3A_x">x</code></td>
<td>
<p>the object to be determined if it is a valid gamma site object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid gamma site object, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_gamma_site_model(create_gamma_site_model())

# FALSE
is_gamma_site_model("nonsense")
is_gamma_site_model(NA)
is_gamma_site_model(NULL)
is_gamma_site_model("")
is_gamma_site_model(c())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_gtr_site_model'>Determine if the object is a valid GTR site model,
as created by <code><a href="#topic+create_gtr_site_model">create_gtr_site_model</a></code></h2><span id='topic+is_gtr_site_model'></span>

<h3>Description</h3>

<p>Determine if the object is a valid GTR site model,
as created by <code><a href="#topic+create_gtr_site_model">create_gtr_site_model</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_gtr_site_model(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_gtr_site_model_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid GTR site model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid GTR site model, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# site models
is_gtr_site_model(create_gtr_site_model())
is_gtr_site_model(create_hky_site_model())
is_gtr_site_model(create_jc69_site_model())
is_gtr_site_model(create_tn93_site_model())

# other models
is_gtr_site_model(NA)
is_gtr_site_model(NULL)
is_gtr_site_model("nonsense")
is_gtr_site_model(create_strict_clock_model())
is_gtr_site_model(create_bd_tree_prior())
is_gtr_site_model(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_hky_site_model'>Determine if the object is a valid HKY site model,
as created by <code><a href="#topic+create_hky_site_model">create_hky_site_model</a></code></h2><span id='topic+is_hky_site_model'></span>

<h3>Description</h3>

<p>Determine if the object is a valid HKY site model,
as created by <code><a href="#topic+create_hky_site_model">create_hky_site_model</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_hky_site_model(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_hky_site_model_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid HKY site model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid HKY site model, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# site models
is_hky_site_model(create_hky_site_model())
is_hky_site_model(create_gtr_site_model())
is_hky_site_model(create_jc69_site_model())
is_hky_site_model(create_tn93_site_model())

# other models
is_hky_site_model(NA)
is_hky_site_model(NULL)
is_hky_site_model("nonsense")
is_hky_site_model(create_strict_clock_model())
is_hky_site_model(create_bd_tree_prior())
is_hky_site_model(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_id'>Determine if the object is a valid ID</h2><span id='topic+is_id'></span>

<h3>Description</h3>

<p>Determine if the object is a valid ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_id(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_id_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid ID, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>to check multiple IDs, use <a href="#topic+are_ids">are_ids</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_id("anthus_aco")
is_id(3)

# FALSE
is_id(ape::rcoal(3))
is_id(NULL)
is_id(NA)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_in_patterns'>Is there at least one regular expression having a match with the line?</h2><span id='topic+is_in_patterns'></span>

<h3>Description</h3>

<p>Is there at least one regular expression having a match with the line?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_in_patterns(line, patterns)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_in_patterns_+3A_line">line</code></td>
<td>
<p>a line of text</p>
</td></tr>
<tr><td><code id="is_in_patterns_+3A_patterns">patterns</code></td>
<td>
<p>one or more regular expression patterns</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if there is at least one match found
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_inference_model'>Determine if the input is an inference model</h2><span id='topic+is_inference_model'></span>

<h3>Description</h3>

<p>Determine if the input is an inference model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_inference_model(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_inference_model_+3A_x">x</code></td>
<td>
<p>object to be determined of if it is an inference model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the object is an inference model
</p>

<hr>
<h2 id='is_init_bd_tree_prior'>Determine if x is an initialized Birth-Death tree_prior object</h2><span id='topic+is_init_bd_tree_prior'></span>

<h3>Description</h3>

<p>Determine if x is an initialized Birth-Death tree_prior object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_bd_tree_prior(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_bd_tree_prior_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized Birth-Death tree prior object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized Birth-Death tree_prior object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_beta_distr'>Determine if x is an initialized beta distribution object
as created by <code><a href="#topic+create_beta_distr">create_beta_distr</a></code></h2><span id='topic+is_init_beta_distr'></span>

<h3>Description</h3>

<p>Determine if x is an initialized beta distribution object
as created by <code><a href="#topic+create_beta_distr">create_beta_distr</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_beta_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_beta_distr_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized beta distribution object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized beta distribution object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_cbs_tree_prior'>Determine if x is an initialized Coalescent Bayesian Skyline
tree_prior object</h2><span id='topic+is_init_cbs_tree_prior'></span>

<h3>Description</h3>

<p>Determine if x is an initialized Coalescent Bayesian Skyline
tree_prior object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_cbs_tree_prior(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_cbs_tree_prior_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized Coalescent Bayesian Skyline tree prior object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized Coalescent Bayesian Skyline
tree prior object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_ccp_tree_prior'>Determine if x is an initialized Coalescent Constant Population
tree_prior object</h2><span id='topic+is_init_ccp_tree_prior'></span>

<h3>Description</h3>

<p>Determine if x is an initialized Coalescent Constant Population
tree_prior object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_ccp_tree_prior(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_ccp_tree_prior_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized Coalescent Constant Population tree prior object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized Coalescent Constant Population
tree prior object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_cep_tree_prior'>Determine if x is an initialized Coalescent Exponential Population
tree_prior object</h2><span id='topic+is_init_cep_tree_prior'></span>

<h3>Description</h3>

<p>Determine if x is an initialized Coalescent Exponential Population
tree_prior object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_cep_tree_prior(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_cep_tree_prior_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized Coalescent Exponential Population tree prior object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized Coalescent Exponential Population
tree prior object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_clock_model'>Determine if x is an initialized clock_model object,
as created by <code><a href="#topic+create_clock_model">create_clock_model</a></code></h2><span id='topic+is_init_clock_model'></span>

<h3>Description</h3>

<p>Determine if x is an initialized clock_model object,
as created by <code><a href="#topic+create_clock_model">create_clock_model</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_clock_model(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_clock_model_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized clock_models object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized clock_model object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_distr'>Determine if x is an initialized distribution object
as created by <code><a href="#topic+create_distr">create_distr</a></code></h2><span id='topic+is_init_distr'></span>

<h3>Description</h3>

<p>Determine if x is an initialized distribution object
as created by <code><a href="#topic+create_distr">create_distr</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_distr_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized distribution object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized distribution object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_exp_distr'>Determine if x is an initialized exponential distribution object
as created by <code><a href="#topic+create_exp_distr">create_exp_distr</a></code></h2><span id='topic+is_init_exp_distr'></span>

<h3>Description</h3>

<p>Determine if x is an initialized exponential distribution object
as created by <code><a href="#topic+create_exp_distr">create_exp_distr</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_exp_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_exp_distr_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized exponential distribution object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized exponential distribution object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_gamma_distr'>Determine if x is an initialized gamma distribution object</h2><span id='topic+is_init_gamma_distr'></span>

<h3>Description</h3>

<p>Determine if x is an initialized gamma distribution object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_gamma_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_gamma_distr_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized gamma distribution object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized gamma distribution object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_gamma_site_model'>Determine if x is an initialized gamma site model,
as created by <code><a href="#topic+create_gamma_site_model">create_gamma_site_model</a></code></h2><span id='topic+is_init_gamma_site_model'></span>

<h3>Description</h3>

<p>Determine if x is an initialized gamma site model,
as created by <code><a href="#topic+create_gamma_site_model">create_gamma_site_model</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_gamma_site_model(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_gamma_site_model_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized gamma site_models object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized gamma site model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_gtr_site_model'>Determine if x is an initialized GTR site model
as created by <code><a href="#topic+create_gtr_site_model">create_gtr_site_model</a></code></h2><span id='topic+is_init_gtr_site_model'></span>

<h3>Description</h3>

<p>Determine if x is an initialized GTR site model
as created by <code><a href="#topic+create_gtr_site_model">create_gtr_site_model</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_gtr_site_model(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_gtr_site_model_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized GTR site model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized GTR site model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

gtr_site_model &lt;- create_gtr_site_model()
# FALSE: not yet initialized
is_init_gtr_site_model(gtr_site_model)
gtr_site_model &lt;- init_gtr_site_model(gtr_site_model)
# TRUE: now it is initialized
is_init_gtr_site_model(gtr_site_model)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_init_hky_site_model'>Determine if x is an initialized HKY site model
as created by <code><a href="#topic+create_hky_site_model">create_hky_site_model</a></code></h2><span id='topic+is_init_hky_site_model'></span>

<h3>Description</h3>

<p>Determine if x is an initialized HKY site model
as created by <code><a href="#topic+create_hky_site_model">create_hky_site_model</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_hky_site_model(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_hky_site_model_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized HKY site model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized HKY site model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

hky_site_model &lt;- create_hky_site_model()
# FALSE: not yet initialized
is_init_hky_site_model(hky_site_model)
hky_site_model &lt;- init_hky_site_model(hky_site_model)
# TRUE: now it is initialized
is_init_hky_site_model(hky_site_model)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_init_inv_gamma_distr'>Determine if x is an initialized inverse-gamma distribution
as created by <code><a href="#topic+create_inv_gamma_distr">create_inv_gamma_distr</a></code></h2><span id='topic+is_init_inv_gamma_distr'></span>

<h3>Description</h3>

<p>Determine if x is an initialized inverse-gamma distribution
as created by <code><a href="#topic+create_inv_gamma_distr">create_inv_gamma_distr</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_inv_gamma_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_inv_gamma_distr_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized inverse-gamma distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized inverse-gamma distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_jc69_site_model'>Determine if x is an initialized JC69 site model
as created by <code><a href="#topic+create_jc69_site_model">create_jc69_site_model</a></code></h2><span id='topic+is_init_jc69_site_model'></span>

<h3>Description</h3>

<p>Determine if x is an initialized JC69 site model
as created by <code><a href="#topic+create_jc69_site_model">create_jc69_site_model</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_jc69_site_model(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_jc69_site_model_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized JC69 site model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized JC69 site model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

jc69_site_model &lt;- create_jc69_site_model(
  gamma_site_model = create_gamma_site_model(
    gamma_cat_count = 2,
    gamma_shape_prior_distr = create_normal_distr()
  )
)
# FALSE: not yet initialized
is_init_jc69_site_model(jc69_site_model)
jc69_site_model &lt;- init_jc69_site_model(jc69_site_model)
# TRUE: now it is initialized
is_init_jc69_site_model(jc69_site_model)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_init_laplace_distr'>Determine if x is an initialized Laplace distribution
as created by <code><a href="#topic+create_laplace_distr">create_laplace_distr</a></code></h2><span id='topic+is_init_laplace_distr'></span>

<h3>Description</h3>

<p>Determine if x is an initialized Laplace distribution
as created by <code><a href="#topic+create_laplace_distr">create_laplace_distr</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_laplace_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_laplace_distr_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized Laplace distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized Laplace distribution
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_log_normal_distr'>Determine if x is an initialized log_normal distribution object
as created by <code><a href="#topic+create_log_normal_distr">create_log_normal_distr</a></code></h2><span id='topic+is_init_log_normal_distr'></span>

<h3>Description</h3>

<p>Determine if x is an initialized log_normal distribution object
as created by <code><a href="#topic+create_log_normal_distr">create_log_normal_distr</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_log_normal_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_log_normal_distr_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized log_normal distribution object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized log_normal distribution object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_mrca_prior'>Determine if x is an initialized MRCA prior</h2><span id='topic+is_init_mrca_prior'></span>

<h3>Description</h3>

<p>Determine if x is an initialized MRCA prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_mrca_prior(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_mrca_prior_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized MRCA prior</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized MRCA prior
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_normal_distr'>Determine if x is an initialized normal distribution object
as created by <code><a href="#topic+create_normal_distr">create_normal_distr</a></code></h2><span id='topic+is_init_normal_distr'></span>

<h3>Description</h3>

<p>Determine if x is an initialized normal distribution object
as created by <code><a href="#topic+create_normal_distr">create_normal_distr</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_normal_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_normal_distr_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized normal distribution object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized normal distribution object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_one_div_x_distr'>Determine if x is an initialized one_div_x distribution object
as created by <code><a href="#topic+create_one_div_x_distr">create_one_div_x_distr</a></code></h2><span id='topic+is_init_one_div_x_distr'></span>

<h3>Description</h3>

<p>Determine if x is an initialized one_div_x distribution object
as created by <code><a href="#topic+create_one_div_x_distr">create_one_div_x_distr</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_one_div_x_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_one_div_x_distr_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized one_div_x distribution object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized one_div_x distribution object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_param'>Determine if x is an initialized parameter,
as created by <a href="#topic+create_param">create_param</a></h2><span id='topic+is_init_param'></span>

<h3>Description</h3>

<p>Determine if x is an initialized parameter,
as created by <a href="#topic+create_param">create_param</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_param_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p><a href="base.html#topic+TRUE">TRUE</a> if x is an initialized parameter,
<a href="base.html#topic+FALSE">FALSE</a> otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_poisson_distr'>Determine if x is an initialized Poisson distribution object
as created by <code><a href="#topic+create_poisson_distr">create_poisson_distr</a></code></h2><span id='topic+is_init_poisson_distr'></span>

<h3>Description</h3>

<p>Determine if x is an initialized Poisson distribution object
as created by <code><a href="#topic+create_poisson_distr">create_poisson_distr</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_poisson_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_poisson_distr_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized Poisson distribution object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized Poisson distribution object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_rln_clock_model'>Determine if x is an initialized relaxed log-normal clock_model object</h2><span id='topic+is_init_rln_clock_model'></span>

<h3>Description</h3>

<p>Determine if x is an initialized relaxed log-normal clock_model object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_rln_clock_model(rln_clock_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_rln_clock_model_+3A_rln_clock_model">rln_clock_model</code></td>
<td>
<p>a Relaxed Log-Normal clock model,
as returned by <code><a href="#topic+create_rln_clock_model">create_rln_clock_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized relaxed log-normal clock_model object,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_site_model'>Determine if x is an initialized site model,
as created by <code><a href="#topic+create_site_model">create_site_model</a></code></h2><span id='topic+is_init_site_model'></span>

<h3>Description</h3>

<p>Determine if x is an initialized site model,
as created by <code><a href="#topic+create_site_model">create_site_model</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_site_model(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_site_model_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized site_models object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized site model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_strict_clock_model'>Determine if x is an initialized strict clock_model object</h2><span id='topic+is_init_strict_clock_model'></span>

<h3>Description</h3>

<p>Determine if x is an initialized strict clock_model object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_strict_clock_model(strict_clock_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_strict_clock_model_+3A_strict_clock_model">strict_clock_model</code></td>
<td>
<p>a strict clock model,
as returned by <code><a href="#topic+create_strict_clock_model">create_strict_clock_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized strict clock_model object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_tn93_site_model'>Determine if x is an initialized tn93 site model
as created by <code><a href="#topic+create_tn93_site_model">create_tn93_site_model</a></code></h2><span id='topic+is_init_tn93_site_model'></span>

<h3>Description</h3>

<p>Determine if x is an initialized tn93 site model
as created by <code><a href="#topic+create_tn93_site_model">create_tn93_site_model</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_tn93_site_model(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_tn93_site_model_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized TN93 site model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized TN93 site model
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

tn93_site_model &lt;- create_tn93_site_model()
# FALSE: not yet initialized
is_init_tn93_site_model(tn93_site_model)
tn93_site_model &lt;- init_tn93_site_model(tn93_site_model)
# TRUE: now it is initialized
is_init_tn93_site_model(tn93_site_model)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_init_tree_prior'>Determine if x is an initialized tree_prior objects</h2><span id='topic+is_init_tree_prior'></span>

<h3>Description</h3>

<p>Determine if x is an initialized tree_prior objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_tree_prior(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_tree_prior_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized tree_priors object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized tree_prior object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_uniform_distr'>Determine if x is an initialized uniform distribution object
as created by <code><a href="#topic+create_uniform_distr">create_uniform_distr</a></code></h2><span id='topic+is_init_uniform_distr'></span>

<h3>Description</h3>

<p>Determine if x is an initialized uniform distribution object
as created by <code><a href="#topic+create_uniform_distr">create_uniform_distr</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_uniform_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_uniform_distr_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized uniform distribution object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized uniform distribution object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_init_yule_tree_prior'>Determine if x is an initialized Yule tree_prior object</h2><span id='topic+is_init_yule_tree_prior'></span>

<h3>Description</h3>

<p>Determine if x is an initialized Yule tree_prior object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_init_yule_tree_prior(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_init_yule_tree_prior_+3A_x">x</code></td>
<td>
<p>the object to check if it is an
initialized Yule tree prior object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is an initialized Yule tree_prior object
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_inv_gamma_distr'>Determine if the object is a valid
inverse-gamma distribution
as created by <code><a href="#topic+create_inv_gamma_distr">create_inv_gamma_distr</a></code></h2><span id='topic+is_inv_gamma_distr'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
inverse-gamma distribution
as created by <code><a href="#topic+create_inv_gamma_distr">create_inv_gamma_distr</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_inv_gamma_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_inv_gamma_distr_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
inverse-gamma distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid inverse-gamma distribution,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <code><a href="#topic+is_distr">is_distr</a></code> to see if x is any
distribution
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_inv_gamma_distr(create_inv_gamma_distr())
# FALSE
is_inv_gamma_distr(create_laplace_distr())
is_inv_gamma_distr(NA)
is_inv_gamma_distr(NULL)
is_inv_gamma_distr("nonsense")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_jc69_site_model'>Determine if the object is a valid JC69 site model</h2><span id='topic+is_jc69_site_model'></span>

<h3>Description</h3>

<p>Determine if the object is a valid JC69 site model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_jc69_site_model(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_jc69_site_model_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid JC69 site model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid JC69 site model, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# site models
is_jc69_site_model(create_gtr_site_model())
is_jc69_site_model(create_hky_site_model())
is_jc69_site_model(create_jc69_site_model())
is_jc69_site_model(create_tn93_site_model())

# other models
is_jc69_site_model(NA)
is_jc69_site_model(NULL)
is_jc69_site_model("nonsense")
is_jc69_site_model(create_strict_clock_model())
is_jc69_site_model(create_bd_tree_prior())
is_jc69_site_model(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_kappa_1_param'>Determine if the object is a valid kappa 1 parameter</h2><span id='topic+is_kappa_1_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid kappa 1 parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_kappa_1_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_kappa_1_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
kappa 1 parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid kappa 1 parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>kappa 1 parameters are returned by
<code><a href="#topic+create_kappa_1_param">create_kappa_1_param</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_kappa_1_param(create_alpha_param())
is_kappa_1_param(create_beta_param())
is_kappa_1_param(create_clock_rate_param())
is_kappa_1_param(create_kappa_param())
is_kappa_1_param(create_kappa_1_param())
is_kappa_1_param(create_kappa_2_param())
is_kappa_1_param(create_lambda_param())
is_kappa_1_param(create_m_param())
is_kappa_1_param(create_mean_param())
is_kappa_1_param(create_mu_param())
is_kappa_1_param(create_rate_ac_param())
is_kappa_1_param(create_rate_ag_param())
is_kappa_1_param(create_rate_at_param())
is_kappa_1_param(create_rate_cg_param())
is_kappa_1_param(create_rate_ct_param())
is_kappa_1_param(create_rate_gt_param())
is_kappa_1_param(create_s_param())
is_kappa_1_param(create_scale_param())
is_kappa_1_param(create_sigma_param())

is_kappa_1_param(NA)
is_kappa_1_param(NULL)
is_kappa_1_param("nonsense")
is_kappa_1_param(create_jc69_site_model())
is_kappa_1_param(create_strict_clock_model())
is_kappa_1_param(create_yule_tree_prior())
is_kappa_1_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_kappa_2_param'>Determine if the object is a valid kappa 2 parameter</h2><span id='topic+is_kappa_2_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid kappa 2 parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_kappa_2_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_kappa_2_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
kappa 2 parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid kappa_2 parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>kappa 2 parameters are returned by
<code><a href="#topic+create_kappa_2_param">create_kappa_2_param</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_kappa_2_param(create_alpha_param())
is_kappa_2_param(create_beta_param())
is_kappa_2_param(create_clock_rate_param())
is_kappa_2_param(create_kappa_1_param())
is_kappa_2_param(create_kappa_2_param())
is_kappa_2_param(create_lambda_param())
is_kappa_2_param(create_m_param())
is_kappa_2_param(create_mean_param())
is_kappa_2_param(create_mu_param())
is_kappa_2_param(create_rate_ac_param())
is_kappa_2_param(create_rate_ag_param())
is_kappa_2_param(create_rate_at_param())
is_kappa_2_param(create_rate_cg_param())
is_kappa_2_param(create_rate_ct_param())
is_kappa_2_param(create_rate_gt_param())
is_kappa_2_param(create_s_param())
is_kappa_2_param(create_scale_param())
is_kappa_2_param(create_sigma_param())

is_kappa_2_param(NA)
is_kappa_2_param(NULL)
is_kappa_2_param("nonsense")
is_kappa_2_param(create_jc69_site_model())
is_kappa_2_param(create_strict_clock_model())
is_kappa_2_param(create_yule_tree_prior())
is_kappa_2_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_kappa_param'>Determine if the object is a valid kappa parameter</h2><span id='topic+is_kappa_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid kappa parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_kappa_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_kappa_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
kappa parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid kappa parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>kappa parameters are returned by
<code><a href="#topic+create_kappa_param">create_kappa_param</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_kappa_param(create_alpha_param())
is_kappa_param(create_beta_param())
is_kappa_param(create_clock_rate_param())
is_kappa_param(create_kappa_param())
is_kappa_param(create_kappa_1_param())
is_kappa_param(create_kappa_2_param())
is_kappa_param(create_lambda_param())
is_kappa_param(create_m_param())
is_kappa_param(create_mean_param())
is_kappa_param(create_mu_param())
is_kappa_param(create_rate_ac_param())
is_kappa_param(create_rate_ag_param())
is_kappa_param(create_rate_at_param())
is_kappa_param(create_rate_cg_param())
is_kappa_param(create_rate_ct_param())
is_kappa_param(create_rate_gt_param())
is_kappa_param(create_s_param())
is_kappa_param(create_scale_param())
is_kappa_param(create_sigma_param())

is_kappa_param(NA)
is_kappa_param(NULL)
is_kappa_param("nonsense")
is_kappa_param(create_jc69_site_model())
is_kappa_param(create_strict_clock_model())
is_kappa_param(create_yule_tree_prior())
is_kappa_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_lambda_param'>Determine if the object is a valid lambda parameter</h2><span id='topic+is_lambda_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid lambda parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_lambda_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_lambda_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
lambda parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid lambda parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>lambda parameters are returned by <code><a href="#topic+create_lambda_param">create_lambda_param</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_lambda_param(create_alpha_param())
is_lambda_param(create_beta_param())
is_lambda_param(create_clock_rate_param())
is_lambda_param(create_kappa_1_param())
is_lambda_param(create_kappa_2_param())
is_lambda_param(create_lambda_param())
is_lambda_param(create_m_param())
is_lambda_param(create_mean_param())
is_lambda_param(create_mu_param())
is_lambda_param(create_rate_ac_param())
is_lambda_param(create_rate_ag_param())
is_lambda_param(create_rate_at_param())
is_lambda_param(create_rate_cg_param())
is_lambda_param(create_rate_ct_param())
is_lambda_param(create_rate_gt_param())
is_lambda_param(create_s_param())
is_lambda_param(create_scale_param())
is_lambda_param(create_sigma_param())

is_lambda_param(NA)
is_lambda_param(NULL)
is_lambda_param("nonsense")
is_lambda_param(create_jc69_site_model())
is_lambda_param(create_strict_clock_model())
is_lambda_param(create_yule_tree_prior())
is_lambda_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_laplace_distr'>Determine if the object is a valid
Laplace distribution,
as created by <code><a href="#topic+create_laplace_distr">create_laplace_distr</a></code></h2><span id='topic+is_laplace_distr'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
Laplace distribution,
as created by <code><a href="#topic+create_laplace_distr">create_laplace_distr</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_laplace_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_laplace_distr_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
Laplace distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid Laplace distribution,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <code><a href="#topic+is_distr">is_distr</a></code> to see if x is any
distribution
</p>


<h3>Examples</h3>

<pre><code class='language-R'># TRUE
is_laplace_distr(create_laplace_distr())
# FALSE
is_laplace_distr(create_log_normal_distr())
is_laplace_distr(NA)
is_laplace_distr(NULL)
is_laplace_distr("nonsense")
</code></pre>

<hr>
<h2 id='is_log_normal_distr'>Determine if the object is a valid
log-normal distribution,
as created by <code><a href="#topic+create_log_normal_distr">create_log_normal_distr</a></code></h2><span id='topic+is_log_normal_distr'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
log-normal distribution,
as created by <code><a href="#topic+create_log_normal_distr">create_log_normal_distr</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_log_normal_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_log_normal_distr_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
log-normal distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid log-normal distribution,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <code><a href="#topic+is_distr">is_distr</a></code> to see if x is any
distribution
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_log_normal_distr(create_log_normal_distr())
# FALSE
is_log_normal_distr(create_normal_distr())
is_distr(NA)
is_distr(NULL)
is_distr("nonsense")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_m_param'>Determine if the object is a valid
m parameter</h2><span id='topic+is_m_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
m parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_m_param(m_param)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_m_param_+3A_m_param">m_param</code></td>
<td>
<p>an m parameter,
as created by <a href="#topic+create_m_param">create_m_param</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid m parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_m_param(create_alpha_param())
is_m_param(create_beta_param())
is_m_param(create_clock_rate_param())
is_m_param(create_kappa_1_param())
is_m_param(create_kappa_2_param())
is_m_param(create_lambda_param())
is_m_param(create_m_param())
is_m_param(create_mean_param())
is_m_param(create_mu_param())
is_m_param(create_rate_ac_param())
is_m_param(create_rate_ag_param())
is_m_param(create_rate_at_param())
is_m_param(create_rate_cg_param())
is_m_param(create_rate_ct_param())
is_m_param(create_rate_gt_param())
is_m_param(create_s_param())
is_m_param(create_scale_param())
is_m_param(create_sigma_param())

is_m_param(NA)
is_m_param(NULL)
is_m_param("nonsense")
is_m_param(create_jc69_site_model())
is_m_param(create_strict_clock_model())
is_m_param(create_yule_tree_prior())
is_m_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_mcmc'>Determine if the object is a valid MCMC</h2><span id='topic+is_mcmc'></span>

<h3>Description</h3>

<p>Determine if the object is a valid MCMC
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_mcmc(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_mcmc_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid MCMC</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid MCMC, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create an MCMC
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {
  check_empty_beautier_folder()

  # Returns TRUE
  is_mcmc(create_mcmc())
  is_mcmc(create_ns_mcmc())

  # Returns FALSE
  is_mcmc("nonsense")
  is_mcmc(NULL)
  is_mcmc(NA)
  is_mcmc("")
  is_mcmc(c())

  check_empty_beautier_folder()
}
</code></pre>

<hr>
<h2 id='is_mcmc_nested_sampling'>Determine if the object is a valid Nested-Sampling MCMC,
as used in [1]</h2><span id='topic+is_mcmc_nested_sampling'></span><span id='topic+is_nested_sampling_mcmc'></span>

<h3>Description</h3>

<p>Determine if the object is a valid Nested-Sampling MCMC,
as used in [1]
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_mcmc_nested_sampling(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_mcmc_nested_sampling_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid MCMC</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid Nested-Sampling MCMC, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>References</h3>

<p>* [1] Patricio Maturana Russel, Brendon J Brewer, Steffen Klaere,
Remco R Bouckaert; Model Selection and Parameter Inference in
Phylogenetics Using Nested Sampling, Systematic Biology, 2018,
syy050, https://doi.org/10.1093/sysbio/syy050
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+create_ns_mcmc">create_ns_mcmc</a> to create an NS MCMC
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {
  check_empty_beautier_folder()

  # TRUE
  is_nested_sampling_mcmc(create_ns_mcmc())
  # FALSE
  is_nested_sampling_mcmc(create_mcmc())
  is_nested_sampling_mcmc("nonsense")

  check_empty_beautier_folder()
}
</code></pre>

<hr>
<h2 id='is_mean_param'>Determine if the object is a valid mean parameter</h2><span id='topic+is_mean_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid mean parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_mean_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_mean_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid mean parameter,
as created by <code><a href="#topic+create_mean_param">create_mean_param</a></code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid mean parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_mean_param(create_alpha_param())
is_mean_param(create_beta_param())
is_mean_param(create_clock_rate_param())
is_mean_param(create_kappa_1_param())
is_mean_param(create_kappa_2_param())
is_mean_param(create_lambda_param())
is_mean_param(create_m_param())
is_mean_param(create_mean_param())
is_mean_param(create_mu_param())
is_mean_param(create_rate_ac_param())
is_mean_param(create_rate_ag_param())
is_mean_param(create_rate_at_param())
is_mean_param(create_rate_cg_param())
is_mean_param(create_rate_ct_param())
is_mean_param(create_rate_gt_param())
is_mean_param(create_s_param())
is_mean_param(create_scale_param())
is_mean_param(create_sigma_param())

is_mean_param(NA)
is_mean_param(NULL)
is_mean_param("nonsense")
is_mean_param(create_jc69_site_model())
is_mean_param(create_strict_clock_model())
is_mean_param(create_yule_tree_prior())
is_mean_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_mrca_align_id_in_fasta'>Determine if an MRCA prior's alignment IDs is present in the FASTA file</h2><span id='topic+is_mrca_align_id_in_fasta'></span>

<h3>Description</h3>

<p>Determine if an MRCA prior's alignment IDs is present in the FASTA file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_mrca_align_id_in_fasta(mrca_prior, fasta_filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_mrca_align_id_in_fasta_+3A_mrca_prior">mrca_prior</code></td>
<td>
<p>a Most Recent Common Ancestor prior,
as returned by <code><a href="#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
<tr><td><code id="is_mrca_align_id_in_fasta_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>a FASTA filename.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.
Note that BEAST2 also supports missing data,
by using a dash (<code>-</code>) or question mark (<code>?</code>)
as a sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the MRCA prior's alignment IDs
is present in the FASTA file.
Returns FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_mrca_align_ids_in_fastas'>Determine if an MRCA prior's alignment IDs are present in the FASTA files</h2><span id='topic+is_mrca_align_ids_in_fastas'></span>

<h3>Description</h3>

<p>Determine if an MRCA prior's alignment IDs are present in the FASTA files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_mrca_align_ids_in_fastas(mrca_prior, fasta_filenames)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_mrca_align_ids_in_fastas_+3A_mrca_prior">mrca_prior</code></td>
<td>
<p>a Most Recent Common Ancestor prior,
as returned by <code><a href="#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
<tr><td><code id="is_mrca_align_ids_in_fastas_+3A_fasta_filenames">fasta_filenames</code></td>
<td>
<p>One or more FASTA filenames.
Use <code><a href="#topic+get_fasta_filename">get_fasta_filename</a></code> to obtain a testing FASTA filename.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the MRCA prior's alignment IDs
is present in the FASTA files.
Returns FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_mrca_prior'>Determine of the object is an empty (<code>NA</code>) or valid MRCA prior.</h2><span id='topic+is_mrca_prior'></span>

<h3>Description</h3>

<p>Determine of the object is an empty (<code>NA</code>) or valid MRCA prior.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_mrca_prior(mrca_prior)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_mrca_prior_+3A_mrca_prior">mrca_prior</code></td>
<td>
<p>a Most Recent Common Ancestor prior,
as returned by <code><a href="#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if <code>x</code> is an MRCA prior, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_mrca_prior(create_mrca_prior())
# Also 'NA' is a valid MRCA prior,
# denoting that there no MRCA priors
is_mrca_prior(NA)

# FALSE
is_mrca_prior(NULL)
is_mrca_prior("nonsense")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_mrca_prior_with_distr'>See if x is one MRCA prior with a distribution</h2><span id='topic+is_mrca_prior_with_distr'></span>

<h3>Description</h3>

<p>See if x is one MRCA prior with a distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_mrca_prior_with_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_mrca_prior_with_distr_+3A_x">x</code></td>
<td>
<p>the object to be tested</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is one MRCA prior with a distribution,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_mu_param'>Determine if the object is a valid
mu parameter</h2><span id='topic+is_mu_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
mu parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_mu_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_mu_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
mu parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid mu parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_mu_param">create_mu_param</a></code> creates a mu parameter
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_mu_param(create_alpha_param())
is_mu_param(create_beta_param())
is_mu_param(create_clock_rate_param())
is_mu_param(create_kappa_1_param())
is_mu_param(create_kappa_2_param())
is_mu_param(create_lambda_param())
is_mu_param(create_m_param())
is_mu_param(create_mean_param())
is_mu_param(create_mu_param())
is_mu_param(create_rate_ac_param())
is_mu_param(create_rate_ag_param())
is_mu_param(create_rate_at_param())
is_mu_param(create_rate_cg_param())
is_mu_param(create_rate_ct_param())
is_mu_param(create_rate_gt_param())
is_mu_param(create_s_param())
is_mu_param(create_scale_param())
is_mu_param(create_sigma_param())

is_mu_param(NA)
is_mu_param(NULL)
is_mu_param("nonsense")
is_mu_param(create_jc69_site_model())
is_mu_param(create_strict_clock_model())
is_mu_param(create_yule_tree_prior())
is_mu_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_normal_distr'>Determine if the object is a valid
normal distribution
as created by <code><a href="#topic+create_normal_distr">create_normal_distr</a></code></h2><span id='topic+is_normal_distr'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
normal distribution
as created by <code><a href="#topic+create_normal_distr">create_normal_distr</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_normal_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_normal_distr_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
normal distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid normal distribution,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <code><a href="#topic+is_distr">is_distr</a></code> to see if x is any
distribution
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_normal_distr(create_normal_distr())
# FALSE
is_normal_distr(create_one_div_x_distr())
is_normal_distr(NA)
is_normal_distr(NULL)
is_normal_distr("nonsense")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_on_appveyor'>Determines if the environment is AppVeyor</h2><span id='topic+is_on_appveyor'></span>

<h3>Description</h3>

<p>Determines if the environment is AppVeyor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_on_appveyor()
</code></pre>


<h3>Value</h3>

<p><a href="base.html#topic+TRUE">TRUE</a> if run on AppVeyor, <a href="base.html#topic+FALSE">FALSE</a> otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  if (is_on_appveyor()) {
    message("Running on AppVeyor")
  }
</code></pre>

<hr>
<h2 id='is_on_ci'>Determines if the environment is a continuous integration service</h2><span id='topic+is_on_ci'></span>

<h3>Description</h3>

<p>Determines if the environment is a continuous integration service
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_on_ci()
</code></pre>


<h3>Value</h3>

<p><a href="base.html#topic+TRUE">TRUE</a> if run on AppVeyor or Travis CI, <a href="base.html#topic+FALSE">FALSE</a> otherwise
</p>


<h3>Note</h3>

<p>It is possible to fake being on continuous integration service,
in this case GitHub Actions, using:
</p>
<p>&ldquo;'r
Sys.setenv(GITHUB_ACTIONS = &quot;I fake being on GitHub Actions&quot;)
is_on_ci() # Will be true
&ldquo;'
</p>
<p>To undo this, do
</p>
<p>&ldquo;'r
Sys.setenv(GITHUB_ACTIONS = &quot;&quot;)
is_on_ci() # Will be false
&ldquo;'
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>is_on_ci()
</code></pre>

<hr>
<h2 id='is_on_github_actions'>Determines if the environment is GitHub Actions</h2><span id='topic+is_on_github_actions'></span>

<h3>Description</h3>

<p>Determines if the environment is GitHub Actions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_on_github_actions()
</code></pre>


<h3>Value</h3>

<p><a href="base.html#topic+TRUE">TRUE</a> if run on GitHub Actions, <a href="base.html#topic+FALSE">FALSE</a> otherwise
</p>


<h3>Note</h3>

<p>It is possible to fake being on GitHub Actions, using:
</p>
<p>&ldquo;'r
Sys.setenv(GITHUB_ACTIONS = &quot;I fake being on GitHub Actions&quot;)
is_on_github_actions() # Will be true
&ldquo;'
</p>
<p>To undo this, do
</p>
<p>&ldquo;'r
Sys.setenv(GITHUB_ACTIONS = &quot;&quot;)
is_on_github_actions() # Will be false
&ldquo;'
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_github_actions()) {
  message("Running on GitHub Actions")
}
</code></pre>

<hr>
<h2 id='is_on_travis'>Determines if the environment is Travis CI</h2><span id='topic+is_on_travis'></span>

<h3>Description</h3>

<p>Determines if the environment is Travis CI
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_on_travis()
</code></pre>


<h3>Value</h3>

<p><a href="base.html#topic+TRUE">TRUE</a> if run on Travis CI, <a href="base.html#topic+FALSE">FALSE</a> otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  if (is_on_ci()) {
    message("Running on Travis CI")
  }
</code></pre>

<hr>
<h2 id='is_one_bool'>Check if the argument is one boolean</h2><span id='topic+is_one_bool'></span>

<h3>Description</h3>

<p>Check if the argument is one boolean
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_one_bool(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_one_bool_+3A_x">x</code></td>
<td>
<p>the argument to be tested to be boolean</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the argument is one boolean, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_one_bool(TRUE)
is_one_bool(FALSE)

# FALSE
is_one_bool(NULL)
is_one_bool(NA)
is_one_bool(c())
is_one_bool("nonsense")
is_one_bool(is_one_bool)
is_one_bool(c(TRUE, FALSE))

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_one_div_x_distr'>Determine if the object is a valid
1/x distribution,
as created by <code><a href="#topic+create_one_div_x_distr">create_one_div_x_distr</a></code></h2><span id='topic+is_one_div_x_distr'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
1/x distribution,
as created by <code><a href="#topic+create_one_div_x_distr">create_one_div_x_distr</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_one_div_x_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_one_div_x_distr_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
1/x distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid 1/x distribution,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <code><a href="#topic+is_distr">is_distr</a></code> to see if x is any
distribution
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_one_div_x_distr(create_one_div_x_distr())
# FALSE
is_one_div_x_distr(create_poisson_distr())
is_one_div_x_distr(NA)
is_one_div_x_distr(NULL)
is_one_div_x_distr("nonsense")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_one_double'>Determines if the argument is a double</h2><span id='topic+is_one_double'></span>

<h3>Description</h3>

<p>Determines if the argument is a double
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_one_double(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_one_double_+3A_x">x</code></td>
<td>
<p>the object to be determined of if it is one double</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the argument is one floating point value, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_one_double(314)
is_one_double(0)
is_one_double(-314)
is_one_double(3.14)

# FALSE
is_one_double(NULL)
is_one_double(NA)
is_one_double(Inf)
is_one_double("nonsense")
is_one_double(is_one_double)
is_one_double(c())
is_one_double(c(1, 2))

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_one_empty_string'>Determine if an object is one empty string</h2><span id='topic+is_one_empty_string'></span>

<h3>Description</h3>

<p>Determine if an object is one empty string
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_one_empty_string(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_one_empty_string_+3A_x">x</code></td>
<td>
<p>the object that may be one string that may be empty</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE is 'x' is one string that is empty
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'># TRUE
is_one_empty_string("")

# FALSE
is_one_empty_string("3.14")
is_one_empty_string(c("", ""))
is_one_empty_string(42)
is_one_empty_string("nonsense")
</code></pre>

<hr>
<h2 id='is_one_int'>Determines if the argument is a whole number</h2><span id='topic+is_one_int'></span>

<h3>Description</h3>

<p>Determines if the argument is a whole number
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_one_int(x, tolerance = .Machine$double.eps^0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_one_int_+3A_x">x</code></td>
<td>
<p>the object to be determined of if it is one integer</p>
</td></tr>
<tr><td><code id="is_one_int_+3A_tolerance">tolerance</code></td>
<td>
<p>tolerance to rounding errors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the argument is one int, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_one_int(314)
is_one_int(0)
is_one_int(-314)
# FALSE
is_one_int(3.14)
is_one_int(NULL)
is_one_int(NA)
is_one_int(Inf)
is_one_int("nonsense")
is_one_int(c())
is_one_int(c(1, 2))

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_one_na'>Determines if x is one NA</h2><span id='topic+is_one_na'></span>

<h3>Description</h3>

<p>Determines if x is one NA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_one_na(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_one_na_+3A_x">x</code></td>
<td>
<p>the object to be determined if it is one NA</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is one NA, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()


check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_one_string'>Determines if the argument is one string</h2><span id='topic+is_one_string'></span>

<h3>Description</h3>

<p>Determines if the argument is one string
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_one_string(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_one_string_+3A_x">x</code></td>
<td>
<p>the object to be determined of if it is one string</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the argument is one string, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_one_string("This is one string")

# FALSE
is_one_string(NULL)
is_one_string(NA)
is_one_string(Inf)
is_one_string(314)
is_one_string(0)
is_one_string(-314)
is_one_string(3.14)
is_one_string(c("a", "b"))
is_one_string(is_one_string)
is_one_string(c())
is_one_string(c(1, 2))

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_one_string_that_is_a_number'>General function to create a distribution.</h2><span id='topic+is_one_string_that_is_a_number'></span>

<h3>Description</h3>

<p>General function to create a distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_one_string_that_is_a_number(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_one_string_that_is_a_number_+3A_x">x</code></td>
<td>
<p>the object that may be one string that may be a number</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE is 'x' is one string that is a number
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'># TRUE
is_one_string_that_is_a_number("3.14")

# FALSE
is_one_string_that_is_a_number(c("3.14", "42"))
is_one_string_that_is_a_number("")
is_one_string_that_is_a_number(42)
is_one_string_that_is_a_number("nonsense")
</code></pre>

<hr>
<h2 id='is_param'>Determine if the object is a valid parameter</h2><span id='topic+is_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid parameter,
as created by <code><a href="#topic+create_param">create_param</a></code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_param(create_alpha_param())
is_param(create_beta_param())
is_param(create_clock_rate_param())
is_param(create_kappa_1_param())
is_param(create_kappa_2_param())
is_param(create_lambda_param())
is_param(create_m_param())
is_param(create_mean_param())
is_param(create_mu_param())
is_param(create_rate_ac_param())
is_param(create_rate_ag_param())
is_param(create_rate_at_param())
is_param(create_rate_cg_param())
is_param(create_rate_ct_param())
is_param(create_rate_gt_param())
is_param(create_s_param())
is_param(create_scale_param())
is_param(create_sigma_param())

# FALSE
is_param(NA)
is_param(NULL)
is_param("nonsense")
is_param(create_jc69_site_model())
is_param(create_strict_clock_model())
is_param(create_yule_tree_prior())
is_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_param_name'>Determines if the name is a valid parameter name</h2><span id='topic+is_param_name'></span>

<h3>Description</h3>

<p>Determines if the name is a valid parameter name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_param_name(name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_param_name_+3A_name">name</code></td>
<td>
<p>the name to be tested</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the name is a valid parameter name, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_param_name("alpha")
is_param_name("beta")
is_param_name("clock_rate")
is_param_name("kappa_1")
is_param_name("kappa_2")
is_param_name("lambda")
is_param_name("m")
is_param_name("mean")
is_param_name("mu")
is_param_name("rate_ac")
is_param_name("rate_ag")
is_param_name("rate_at")
is_param_name("rate_cg")
is_param_name("rate_ct")
is_param_name("rate_gt")
is_param_name("s")
is_param_name("scale")
is_param_name("sigma")

# FALSE
is_param_name("nonsense")
is_param_name(NA)
is_param_name(NULL)
is_param_name("")
is_param_name(c())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_phylo'>Checks if the input is a phylogeny</h2><span id='topic+is_phylo'></span>

<h3>Description</h3>

<p>Checks if the input is a phylogeny
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_phylo(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_phylo_+3A_x">x</code></td>
<td>
<p>input to be checked</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE or FALSE
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+check_phylogeny">check_phylogeny</a> to check for a phylogeny
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
phylogeny &lt;- ape::read.tree(text = "(a:15,b:15):1;")
is_phylo(phylogeny)

# FALSE
is_phylo("nonsense")
is_phylo(NA)
is_phylo(NULL)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_poisson_distr'>Determine if the object is a valid
Poisson distribution
as created by <code><a href="#topic+create_poisson_distr">create_poisson_distr</a></code></h2><span id='topic+is_poisson_distr'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
Poisson distribution
as created by <code><a href="#topic+create_poisson_distr">create_poisson_distr</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_poisson_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_poisson_distr_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
Poisson distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid Poisson distribution,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <code><a href="#topic+is_distr">is_distr</a></code> to see if x is any
distribution
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_poisson_distr(create_poisson_distr())
# FALSE
is_poisson_distr(create_uniform_distr())
is_distr(NA)
is_distr(NULL)
is_distr("nonsense")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_rate_ac_param'>Determine if the object is a valid
'rate AC' parameter</h2><span id='topic+is_rate_ac_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
'rate AC' parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_rate_ac_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_rate_ac_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
'rate AC' parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid 'rate AC' parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_rate_ac_param">create_rate_ac_param</a></code> creates a 'rate AC' parameter
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_rate_ac_param(create_alpha_param())
is_rate_ac_param(create_beta_param())
is_rate_ac_param(create_clock_rate_param())
is_rate_ac_param(create_kappa_1_param())
is_rate_ac_param(create_kappa_2_param())
is_rate_ac_param(create_lambda_param())
is_rate_ac_param(create_m_param())
is_rate_ac_param(create_mean_param())
is_rate_ac_param(create_mu_param())
is_rate_ac_param(create_rate_ac_param())
is_rate_ac_param(create_rate_ag_param())
is_rate_ac_param(create_rate_at_param())
is_rate_ac_param(create_rate_cg_param())
is_rate_ac_param(create_rate_ct_param())
is_rate_ac_param(create_rate_gt_param())
is_rate_ac_param(create_s_param())
is_rate_ac_param(create_scale_param())
is_rate_ac_param(create_sigma_param())

is_rate_ac_param(NA)
is_rate_ac_param(NULL)
is_rate_ac_param("nonsense")
is_rate_ac_param(create_jc69_site_model())
is_rate_ac_param(create_strict_clock_model())
is_rate_ac_param(create_yule_tree_prior())
is_rate_ac_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_rate_ag_param'>Determine if the object is a valid
'rate AG' parameter</h2><span id='topic+is_rate_ag_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
'rate AG' parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_rate_ag_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_rate_ag_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
'rate AG' parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid 'rate AG' parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_rate_ag_param">create_rate_ag_param</a></code> creates a 'rate AG' parameter
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_rate_ag_param(create_alpha_param())
is_rate_ag_param(create_beta_param())
is_rate_ag_param(create_clock_rate_param())
is_rate_ag_param(create_kappa_1_param())
is_rate_ag_param(create_kappa_2_param())
is_rate_ag_param(create_lambda_param())
is_rate_ag_param(create_m_param())
is_rate_ag_param(create_mean_param())
is_rate_ag_param(create_mu_param())
is_rate_ag_param(create_rate_ac_param())
is_rate_ag_param(create_rate_ag_param())
is_rate_ag_param(create_rate_at_param())
is_rate_ag_param(create_rate_cg_param())
is_rate_ag_param(create_rate_ct_param())
is_rate_ag_param(create_rate_gt_param())
is_rate_ag_param(create_s_param())
is_rate_ag_param(create_scale_param())
is_rate_ag_param(create_sigma_param())

is_rate_ag_param(NA)
is_rate_ag_param(NULL)
is_rate_ag_param("nonsense")
is_rate_ag_param(create_jc69_site_model())
is_rate_ag_param(create_strict_clock_model())
is_rate_ag_param(create_yule_tree_prior())
is_rate_ag_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_rate_at_param'>Determine if the object is a valid
'rate AT' parameter</h2><span id='topic+is_rate_at_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
'rate AT' parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_rate_at_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_rate_at_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
'rate AT' parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid 'rate AT' parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_rate_at_param">create_rate_at_param</a></code> creates a 'rate AT' parameter
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_rate_at_param(create_alpha_param())
is_rate_at_param(create_beta_param())
is_rate_at_param(create_clock_rate_param())
is_rate_at_param(create_kappa_1_param())
is_rate_at_param(create_kappa_2_param())
is_rate_at_param(create_lambda_param())
is_rate_at_param(create_m_param())
is_rate_at_param(create_mean_param())
is_rate_at_param(create_mu_param())
is_rate_at_param(create_rate_ac_param())
is_rate_at_param(create_rate_ag_param())
is_rate_at_param(create_rate_at_param())
is_rate_at_param(create_rate_cg_param())
is_rate_at_param(create_rate_ct_param())
is_rate_at_param(create_rate_gt_param())
is_rate_at_param(create_s_param())
is_rate_at_param(create_scale_param())
is_rate_at_param(create_sigma_param())

is_rate_at_param(NA)
is_rate_at_param(NULL)
is_rate_at_param("nonsense")
is_rate_at_param(create_jc69_site_model())
is_rate_at_param(create_strict_clock_model())
is_rate_at_param(create_yule_tree_prior())
is_rate_at_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_rate_cg_param'>Determine if the object is a valid
'rate CG' parameter</h2><span id='topic+is_rate_cg_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
'rate CG' parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_rate_cg_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_rate_cg_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
'rate CG' parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid 'rate CG' parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_rate_cg_param">create_rate_cg_param</a></code> creates a 'rate CG' parameter
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_rate_cg_param(create_alpha_param())
is_rate_cg_param(create_beta_param())
is_rate_cg_param(create_clock_rate_param())
is_rate_cg_param(create_kappa_1_param())
is_rate_cg_param(create_kappa_2_param())
is_rate_cg_param(create_lambda_param())
is_rate_cg_param(create_m_param())
is_rate_cg_param(create_mean_param())
is_rate_cg_param(create_mu_param())
is_rate_cg_param(create_rate_ac_param())
is_rate_cg_param(create_rate_ag_param())
is_rate_cg_param(create_rate_at_param())
is_rate_cg_param(create_rate_cg_param())
is_rate_cg_param(create_rate_ct_param())
is_rate_cg_param(create_rate_gt_param())
is_rate_cg_param(create_s_param())
is_rate_cg_param(create_scale_param())
is_rate_cg_param(create_sigma_param())

is_rate_cg_param(NA)
is_rate_cg_param(NULL)
is_rate_cg_param("nonsense")
is_rate_cg_param(create_jc69_site_model())
is_rate_cg_param(create_strict_clock_model())
is_rate_cg_param(create_yule_tree_prior())
is_rate_cg_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_rate_ct_param'>Determine if the object is a valid
'rate CT' parameter</h2><span id='topic+is_rate_ct_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
'rate CT' parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_rate_ct_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_rate_ct_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
'rate CT' parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid 'rate CG' parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_rate_ct_param">create_rate_ct_param</a></code> creates a 'rate CT' parameter
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_rate_ct_param(create_alpha_param())
is_rate_ct_param(create_beta_param())
is_rate_ct_param(create_clock_rate_param())
is_rate_ct_param(create_kappa_1_param())
is_rate_ct_param(create_kappa_2_param())
is_rate_ct_param(create_lambda_param())
is_rate_ct_param(create_m_param())
is_rate_ct_param(create_mean_param())
is_rate_ct_param(create_mu_param())
is_rate_ct_param(create_rate_ac_param())
is_rate_ct_param(create_rate_ag_param())
is_rate_ct_param(create_rate_at_param())
is_rate_ct_param(create_rate_cg_param())
is_rate_ct_param(create_rate_ct_param())
is_rate_ct_param(create_rate_gt_param())
is_rate_ct_param(create_s_param())
is_rate_ct_param(create_scale_param())
is_rate_ct_param(create_sigma_param())

is_rate_ct_param(NA)
is_rate_ct_param(NULL)
is_rate_ct_param("nonsense")
is_rate_ct_param(create_jc69_site_model())
is_rate_ct_param(create_strict_clock_model())
is_rate_ct_param(create_yule_tree_prior())
is_rate_ct_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_rate_gt_param'>Determine if the object is a valid
'rate GT' parameter</h2><span id='topic+is_rate_gt_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
'rate GT' parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_rate_gt_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_rate_gt_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
'rate GT' parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid 'rate GT' parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_rate_gt_param">create_rate_gt_param</a></code> creates a 'rate GT' parameter
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_rate_gt_param(create_alpha_param())
is_rate_gt_param(create_beta_param())
is_rate_gt_param(create_clock_rate_param())
is_rate_gt_param(create_kappa_1_param())
is_rate_gt_param(create_kappa_2_param())
is_rate_gt_param(create_lambda_param())
is_rate_gt_param(create_m_param())
is_rate_gt_param(create_mean_param())
is_rate_gt_param(create_mu_param())
is_rate_gt_param(create_rate_ac_param())
is_rate_gt_param(create_rate_ag_param())
is_rate_gt_param(create_rate_at_param())
is_rate_gt_param(create_rate_cg_param())
is_rate_gt_param(create_rate_ct_param())
is_rate_gt_param(create_rate_gt_param())
is_rate_gt_param(create_s_param())
is_rate_gt_param(create_scale_param())
is_rate_gt_param(create_sigma_param())

is_rate_gt_param(NA)
is_rate_gt_param(NULL)
is_rate_gt_param("nonsense")
is_rate_gt_param(create_jc69_site_model())
is_rate_gt_param(create_strict_clock_model())
is_rate_gt_param(create_yule_tree_prior())
is_rate_gt_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_rln_clock_model'>Determine if the object is a valid relaxed log normal clock model</h2><span id='topic+is_rln_clock_model'></span>

<h3>Description</h3>

<p>Determine if the object is a valid relaxed log normal clock model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_rln_clock_model(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_rln_clock_model_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
relaxed log normal clock model,
as created by <code><a href="#topic+create_rln_clock_model">create_rln_clock_model</a></code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid relaxed log normal clock model, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_clock_model">create_clock_model</a></code> shows an overview of
functions to create a clock model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_rln_clock_model(create_strict_clock_model())
is_rln_clock_model(create_rln_clock_model())

is_rln_clock_model(NA)
is_rln_clock_model(NULL)
is_rln_clock_model("nonsense")
is_rln_clock_model(create_jc69_site_model())
is_rln_clock_model(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_s_param'>Determine if the object is a valid
s parameter</h2><span id='topic+is_s_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
s parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_s_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_s_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
s parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid s parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_s_param(create_alpha_param())
is_s_param(create_beta_param())
is_s_param(create_clock_rate_param())
is_s_param(create_kappa_1_param())
is_s_param(create_kappa_2_param())
is_s_param(create_lambda_param())
is_s_param(create_m_param())
is_s_param(create_mean_param())
is_s_param(create_mu_param())
is_s_param(create_rate_ac_param())
is_s_param(create_rate_ag_param())
is_s_param(create_rate_at_param())
is_s_param(create_rate_cg_param())
is_s_param(create_rate_ct_param())
is_s_param(create_rate_gt_param())
is_s_param(create_s_param())
is_s_param(create_scale_param())
is_s_param(create_sigma_param())

is_s_param(NA)
is_s_param(NULL)
is_s_param("nonsense")
is_s_param(create_jc69_site_model())
is_s_param(create_strict_clock_model())
is_s_param(create_yule_tree_prior())
is_s_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_scale_param'>Determine if the object is a valid
scale parameter</h2><span id='topic+is_scale_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
scale parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_scale_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_scale_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
scale parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid scale parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_scale_param(create_alpha_param())
is_scale_param(create_beta_param())
is_scale_param(create_clock_rate_param())
is_scale_param(create_kappa_1_param())
is_scale_param(create_kappa_2_param())
is_scale_param(create_lambda_param())
is_scale_param(create_m_param())
is_scale_param(create_mean_param())
is_scale_param(create_mu_param())
is_scale_param(create_rate_ac_param())
is_scale_param(create_rate_ag_param())
is_scale_param(create_rate_at_param())
is_scale_param(create_rate_cg_param())
is_scale_param(create_rate_ct_param())
is_scale_param(create_rate_gt_param())
is_scale_param(create_s_param())
is_scale_param(create_scale_param())
is_scale_param(create_sigma_param())

is_scale_param(NA)
is_scale_param(NULL)
is_scale_param("nonsense")
is_scale_param(create_jc69_site_model())
is_scale_param(create_strict_clock_model())
is_scale_param(create_yule_tree_prior())
is_scale_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_sigma_param'>Determine if the object is a valid
sigma parameter</h2><span id='topic+is_sigma_param'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
sigma parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_sigma_param(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_sigma_param_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
sigma parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid sigma parameter,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_sigma_param(create_alpha_param())
is_sigma_param(create_beta_param())
is_sigma_param(create_clock_rate_param())
is_sigma_param(create_kappa_1_param())
is_sigma_param(create_kappa_2_param())
is_sigma_param(create_lambda_param())
is_sigma_param(create_m_param())
is_sigma_param(create_mean_param())
is_sigma_param(create_mu_param())
is_sigma_param(create_rate_ac_param())
is_sigma_param(create_rate_ag_param())
is_sigma_param(create_rate_at_param())
is_sigma_param(create_rate_cg_param())
is_sigma_param(create_rate_ct_param())
is_sigma_param(create_rate_gt_param())
is_sigma_param(create_s_param())
is_sigma_param(create_scale_param())
is_sigma_param(create_sigma_param())

is_sigma_param(NA)
is_sigma_param(NULL)
is_sigma_param("nonsense")
is_sigma_param(create_jc69_site_model())
is_sigma_param(create_strict_clock_model())
is_sigma_param(create_yule_tree_prior())
is_sigma_param(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_site_model'>Determine if the object is a valid site_model</h2><span id='topic+is_site_model'></span>

<h3>Description</h3>

<p>Determine if the object is a valid site_model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_site_model(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_site_model_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a site_model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the site_model is a valid site_model, FALSE otherwise
</p>


<h3>See Also</h3>

<p>A site model can be created using <code><a href="#topic+create_site_model">create_site_model</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_site_model(create_gtr_site_model())
is_site_model(create_hky_site_model())
is_site_model(create_jc69_site_model())
is_site_model(create_tn93_site_model())

# FALSE
is_site_model(NA)
is_site_model(NULL)
is_site_model("nonsense")
is_site_model(create_strict_clock_model())
is_site_model(create_bd_tree_prior())
is_site_model(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_site_model_name'>Determines if the name is a valid site_model name</h2><span id='topic+is_site_model_name'></span>

<h3>Description</h3>

<p>Determines if the name is a valid site_model name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_site_model_name(name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_site_model_name_+3A_name">name</code></td>
<td>
<p>the name to be tested</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the name is a valid site_model name, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_site_model_name("JC69")
is_site_model_name("HKY")
is_site_model_name("TN93")
is_site_model_name("GTR")
# FALSE
is_site_model_name("nonsense")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_strict_clock_model'>Determine if the object is a valid strict clock model,
as returned by <code><a href="#topic+create_strict_clock_model">create_strict_clock_model</a></code></h2><span id='topic+is_strict_clock_model'></span>

<h3>Description</h3>

<p>Determine if the object is a valid strict clock model,
as returned by <code><a href="#topic+create_strict_clock_model">create_strict_clock_model</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_strict_clock_model(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_strict_clock_model_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid strict clock model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid strict clock model, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_clock_model">create_clock_model</a></code> shows an overview of
functions to create a clock model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_strict_clock_model(create_strict_clock_model())
is_strict_clock_model(create_rln_clock_model())

is_strict_clock_model(NA)
is_strict_clock_model(NULL)
is_strict_clock_model("nonsense")
is_strict_clock_model(create_jc69_site_model())
is_strict_clock_model(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_tn93_site_model'>Determine if the object is a valid TN93 site model,</h2><span id='topic+is_tn93_site_model'></span>

<h3>Description</h3>

<p>Determine if the object is a valid TN93 site model,
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_tn93_site_model(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_tn93_site_model_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid TN93 site model,
as created by <code><a href="#topic+create_tn93_site_model">create_tn93_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid TN93 site model, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# site models
is_tn93_site_model(create_gtr_site_model())
is_tn93_site_model(create_hky_site_model())
is_tn93_site_model(create_jc69_site_model())
is_tn93_site_model(create_tn93_site_model())

# other models
is_tn93_site_model(NA)
is_tn93_site_model(NULL)
is_tn93_site_model("nonsense")
is_tn93_site_model("")
is_tn93_site_model(c())
is_tn93_site_model(create_strict_clock_model())
is_tn93_site_model(create_bd_tree_prior())
is_tn93_site_model(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_tree_prior'>Determine if an object is a valid tree prior</h2><span id='topic+is_tree_prior'></span>

<h3>Description</h3>

<p>Determine if an object is a valid tree prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_tree_prior(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_tree_prior_+3A_x">x</code></td>
<td>
<p>an object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid tree_prior, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>tree priors can be created by <code><a href="#topic+create_tree_prior">create_tree_prior</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

is_tree_prior(create_bd_tree_prior())
is_tree_prior(create_yule_tree_prior())
!is_tree_prior("nonsense")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_tree_prior_name'>Determines if the name is a valid tree prior name</h2><span id='topic+is_tree_prior_name'></span>

<h3>Description</h3>

<p>Determines if the name is a valid tree prior name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_tree_prior_name(name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_tree_prior_name_+3A_name">name</code></td>
<td>
<p>the name to be tested</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the name is a valid tree_prior name, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_tree_prior_name("birth_death")
is_tree_prior_name("coalescent_bayesian_skyline")
is_tree_prior_name("coalescent_constant_population")
is_tree_prior_name("coalescent_exp_population")
is_tree_prior_name("yule")
# FALSE
is_tree_prior_name("nonsense")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_uniform_distr'>Determine if the object is a valid
uniform distribution
as created by <code><a href="#topic+create_uniform_distr">create_uniform_distr</a></code></h2><span id='topic+is_uniform_distr'></span>

<h3>Description</h3>

<p>Determine if the object is a valid
uniform distribution
as created by <code><a href="#topic+create_uniform_distr">create_uniform_distr</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_uniform_distr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_uniform_distr_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid
uniform distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid uniform distribution,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <code><a href="#topic+is_distr">is_distr</a></code> to see if x is any
distribution
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_uniform_distr(create_uniform_distr())
# FALSE
is_uniform_distr(create_beta_distr())
is_uniform_distr(NA)
is_uniform_distr(NULL)
is_uniform_distr("nonsense")

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='is_xml'>Checks if the text is a valid XML node, that is,
it has a opening and matching closing tag</h2><span id='topic+is_xml'></span>

<h3>Description</h3>

<p>Checks if the text is a valid XML node, that is,
it has a opening and matching closing tag
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_xml(text)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_xml_+3A_text">text</code></td>
<td>
<p>text to be determined to be valid</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the text is valid XML, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_yule_tree_prior'>Determine if the object is a valid Yule tree prior,</h2><span id='topic+is_yule_tree_prior'></span>

<h3>Description</h3>

<p>Determine if the object is a valid Yule tree prior,
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_yule_tree_prior(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_yule_tree_prior_+3A_x">x</code></td>
<td>
<p>an object, to be determined if it is a valid Yule tree prior</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid Yule tree prior, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+create_yule_tree_prior">create_yule_tree_prior</a></code> to create a valid
Yule tree prior
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# TRUE
is_yule_tree_prior(create_yule_tree_prior())

# FALSE
is_yule_tree_prior(create_bd_tree_prior())
is_yule_tree_prior(create_cbs_tree_prior())
is_yule_tree_prior(create_ccp_tree_prior())
is_yule_tree_prior(create_cep_tree_prior())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='jc69_site_model_to_xml_state'>Converts a site model to XML,
used in the <code>state</code> section</h2><span id='topic+jc69_site_model_to_xml_state'></span>

<h3>Description</h3>

<p>Converts a site model to XML,
used in the <code>state</code> section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jc69_site_model_to_xml_state(
  site_model,
  beauti_options = create_beauti_options()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jc69_site_model_to_xml_state_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
<tr><td><code id="jc69_site_model_to_xml_state_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the site model as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='kappa_param_to_xml'>Internal function</h2><span id='topic+kappa_param_to_xml'></span>

<h3>Description</h3>

<p>Converts an kappa parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kappa_param_to_xml(kappa_param, beauti_options = create_beauti_options())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kappa_param_to_xml_+3A_kappa_param">kappa_param</code></td>
<td>
<p>a kappa parameter,
as created by <a href="#topic+create_kappa_param">create_kappa_param</a></p>
</td></tr>
<tr><td><code id="kappa_param_to_xml_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# The kappa parameter must be initialized, i.e. have an ID
kappa_param_to_xml(kappa_param = create_kappa_param(id = "1"))

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='m_param_to_xml'>Internal function</h2><span id='topic+m_param_to_xml'></span>

<h3>Description</h3>

<p>Converts an m parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>m_param_to_xml(m_param, beauti_options = create_beauti_options())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="m_param_to_xml_+3A_m_param">m_param</code></td>
<td>
<p>an m parameter,
as created by <a href="#topic+create_m_param">create_m_param</a></p>
</td></tr>
<tr><td><code id="m_param_to_xml_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='mcmc_to_xml_run'>Converts an MCMC object to the run section's XML</h2><span id='topic+mcmc_to_xml_run'></span>

<h3>Description</h3>

<p>Converts an MCMC object to the run section's XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mcmc_to_xml_run(mcmc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mcmc_to_xml_run_+3A_mcmc">mcmc</code></td>
<td>
<p>one MCMC.
Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create an MCMC.
Use <code><a href="#topic+create_ns_mcmc">create_ns_mcmc</a></code> to create an MCMC
for a Nested Sampling run.
Use <code><a href="#topic+check_mcmc">check_mcmc</a></code> to check if an MCMC is valid.
Use <code><a href="#topic+rename_mcmc_filenames">rename_mcmc_filenames</a></code> to rename the filenames in an MCMC.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the XML as text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_ci()) {

  check_empty_beautier_folder()

  # &lt;run id=\"mcmc\" spec=\"MCMC\" chainLength=\"1e+07\"&gt;
  mcmc_to_xml_run(create_mcmc())

  check_empty_beautier_folder()
}
</code></pre>

<hr>
<h2 id='mcmc_to_xml_run_default'>Converts an MCMC object to the run section's XML for a default MCMC</h2><span id='topic+mcmc_to_xml_run_default'></span>

<h3>Description</h3>

<p>Converts an MCMC object to the run section's XML for a default MCMC
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mcmc_to_xml_run_default(mcmc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mcmc_to_xml_run_default_+3A_mcmc">mcmc</code></td>
<td>
<p>one MCMC.
Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create an MCMC.
Use <code><a href="#topic+create_ns_mcmc">create_ns_mcmc</a></code> to create an MCMC
for a Nested Sampling run.
Use <code><a href="#topic+check_mcmc">check_mcmc</a></code> to check if an MCMC is valid.
Use <code><a href="#topic+rename_mcmc_filenames">rename_mcmc_filenames</a></code> to rename the filenames in an MCMC.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the XML as text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# &lt;run id=\"mcmc\" spec=\"MCMC\" chainLength=\"1e+07\"&gt;
xml &lt;- mcmc_to_xml_run_default(create_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='mcmc_to_xml_run_nested_sampling'>Converts an MCMC object to the run section's XML for a Nested-Sampling MCMC</h2><span id='topic+mcmc_to_xml_run_nested_sampling'></span>

<h3>Description</h3>

<p>Converts an MCMC object to the run section's XML for a Nested-Sampling MCMC
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mcmc_to_xml_run_nested_sampling(mcmc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mcmc_to_xml_run_nested_sampling_+3A_mcmc">mcmc</code></td>
<td>
<p>one MCMC.
Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create an MCMC.
Use <code><a href="#topic+create_ns_mcmc">create_ns_mcmc</a></code> to create an MCMC
for a Nested Sampling run.
Use <code><a href="#topic+check_mcmc">check_mcmc</a></code> to check if an MCMC is valid.
Use <code><a href="#topic+rename_mcmc_filenames">rename_mcmc_filenames</a></code> to rename the filenames in an MCMC.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the XML as text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

#  "&lt;run id=\"mcmc\" spec=\"beast.gss.NS\" chainLength=\"1e+07\" "
#  "particleCount=\"1\" subChainLength=\"5000\" epsilon=\"1e-12\"&gt;"
mcmc_to_xml_run_nested_sampling(create_ns_mcmc())

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='mrca_prior_to_xml_prior_distr'>Creates the distribution section in the prior section of the
distribution section of a BEAST2 XML parameter file.</h2><span id='topic+mrca_prior_to_xml_prior_distr'></span>

<h3>Description</h3>

<p>These lines start with '&lt;distribution id='
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mrca_prior_to_xml_prior_distr(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mrca_prior_to_xml_prior_distr_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

 # &lt;distribution id="posterior" spec="util.CompoundDistribution"&gt;
 #     &lt;distribution id="prior" spec="util.CompoundDistribution"&gt;
 #       HERE, where the ID of the distribution is 'prior'
 #     &lt;/distribution&gt;
 #     &lt;distribution id="likelihood" ...&gt;
 #     &lt;/distribution&gt;
 # &lt;/distribution&gt;

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='mrca_prior_to_xml_state'>Internal function to create the XML of an MRCA prior,
as used in the <code>state</code> section</h2><span id='topic+mrca_prior_to_xml_state'></span>

<h3>Description</h3>

<p>Internal function to create the XML of an MRCA prior,
as used in the <code>state</code> section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mrca_prior_to_xml_state(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mrca_prior_to_xml_state_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the tree prior as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

mrca_prior_to_xml_state(
  inference_model = create_inference_model(
    mrca_prior = create_mrca_prior(
      alignment_id = "test_output_0",
      mrca_distr = create_normal_distr(id = 42)
    ),
    clock_model = create_strict_clock_model()
  )
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='mrca_prior_to_xml_taxonset'>Creates the <code>taxonset</code> section in the prior section of the
distribution section of a BEAST2 XML parameter file.</h2><span id='topic+mrca_prior_to_xml_taxonset'></span>

<h3>Description</h3>

<p>Creates the <code>taxonset</code> section in the prior section of the
distribution section of a BEAST2 XML parameter file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mrca_prior_to_xml_taxonset(mrca_prior, taxa_names_with_ids = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mrca_prior_to_xml_taxonset_+3A_mrca_prior">mrca_prior</code></td>
<td>
<p>a Most Recent Common Ancestor prior,
as returned by <code><a href="#topic+create_mrca_prior">create_mrca_prior</a></code></p>
</td></tr>
<tr><td><code id="mrca_prior_to_xml_taxonset_+3A_taxa_names_with_ids">taxa_names_with_ids</code></td>
<td>
<p>taxa names that already have received
an ID. Causes the XML to <code>idref</code> these</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>
  &lt;taxonset id="all" spec="TaxonSet"&gt;
      &lt;taxon id="626029_aco" spec="Taxon"/&gt;
      &lt;taxon id="630116_aco" spec="Taxon"/&gt;
      &lt;taxon id="630210_aco" spec="Taxon"/&gt;
      &lt;taxon id="B25702_aco" spec="Taxon"/&gt;
      &lt;taxon id="61430_aco" spec="Taxon"/&gt;
  &lt;/taxonset&gt;
</code>
</p>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='mrca_prior_to_xml_tracelog'>Internal function</h2><span id='topic+mrca_prior_to_xml_tracelog'></span>

<h3>Description</h3>

<p>Internal function to creates the MRCA prior's XML for the tracelog section.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mrca_prior_to_xml_tracelog(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mrca_prior_to_xml_tracelog_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>
  &lt;logger id="tracelog" ...&gt;
    # Here
  &lt;/logger&gt;
</code>
</p>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='mrca_priors_to_xml_prior_distr'>Creates the the <code>distribution</code>'s prior section (which is part of
a posterior distribution section) of a BEAST2 XML parameter file.</h2><span id='topic+mrca_priors_to_xml_prior_distr'></span>

<h3>Description</h3>

<p>These lines start with '<code>&lt;distribution id="prior"</code>'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mrca_priors_to_xml_prior_distr(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mrca_priors_to_xml_prior_distr_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>
   &lt;distribution id="posterior" spec="util.CompoundDistribution"&gt;
       &lt;distribution id="prior" spec="util.CompoundDistribution"&gt;
         HERE, where the ID of the distribution is 'prior'
       &lt;/distribution&gt;
       &lt;distribution id="likelihood" ...&gt;
       &lt;/distribution&gt;
  &lt;/distribution&gt;
</code>
</p>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='no_taxa_to_xml_tree'>Internal function</h2><span id='topic+no_taxa_to_xml_tree'></span>

<h3>Description</h3>

<p>Creates the '<code>tree</code>' section of a BEAST2 XML parameter file,
which is part of a '<code>state</code>' section,
without being indented,
when there is no tip-dating
</p>


<h3>Usage</h3>

<pre><code class='language-R'>no_taxa_to_xml_tree(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="no_taxa_to_xml_tree_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>tree</code> tag has these elements:
</p>
<pre>
   &lt;tree[...]&gt;
       &lt;taxonset[...]&gt;
       [...]
       &lt;/taxonset&gt;
    &lt;/run&gt;
</pre>


<h3>Value</h3>

<p>the random phylogeny as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='parameter_to_xml'>Internal function</h2><span id='topic+parameter_to_xml'></span>

<h3>Description</h3>

<p>Converts a parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameter_to_xml(parameter, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameter_to_xml_+3A_parameter">parameter</code></td>
<td>
<p>a parameter,
as created by <code><a href="#topic+create_param">create_param</a></code>)</p>
</td></tr>
<tr><td><code id="parameter_to_xml_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

parameter_to_xml(
  create_alpha_param(id = 1),
  beauti_options = create_beauti_options()
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='parameter_to_xml_kappa_1'>Internal function</h2><span id='topic+parameter_to_xml_kappa_1'></span>

<h3>Description</h3>

<p>Converts a kappa 1 parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameter_to_xml_kappa_1(parameter, beauti_options = create_beauti_options())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameter_to_xml_kappa_1_+3A_parameter">parameter</code></td>
<td>
<p>a kappa 1 parameter,
a numeric value.
For advanced usage, use the structure
as created by <code><a href="#topic+create_kappa_1_param">create_kappa_1_param</a></code>)</p>
</td></tr>
<tr><td><code id="parameter_to_xml_kappa_1_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='parameter_to_xml_kappa_2'>Internal function</h2><span id='topic+parameter_to_xml_kappa_2'></span>

<h3>Description</h3>

<p>Converts a kappa 2 parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameter_to_xml_kappa_2(parameter, beauti_options = create_beauti_options())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameter_to_xml_kappa_2_+3A_parameter">parameter</code></td>
<td>
<p>a kappa 2 parameter,
a numeric value.
For advanced usage, use the structure
as created by <code><a href="#topic+create_kappa_2_param">create_kappa_2_param</a></code>)</p>
</td></tr>
<tr><td><code id="parameter_to_xml_kappa_2_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='parameter_to_xml_lambda'>Internal function</h2><span id='topic+parameter_to_xml_lambda'></span>

<h3>Description</h3>

<p>Converts a lambda parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameter_to_xml_lambda(parameter, beauti_options = create_beauti_options())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameter_to_xml_lambda_+3A_parameter">parameter</code></td>
<td>
<p>a lambda parameter,
a numeric value.
For advanced usage, use the structure
as created by <code><a href="#topic+create_lambda_param">create_lambda_param</a></code>)</p>
</td></tr>
<tr><td><code id="parameter_to_xml_lambda_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='parameter_to_xml_mean'>Internal function</h2><span id='topic+parameter_to_xml_mean'></span>

<h3>Description</h3>

<p>Converts a mean parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameter_to_xml_mean(parameter, beauti_options = create_beauti_options())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameter_to_xml_mean_+3A_parameter">parameter</code></td>
<td>
<p>a mean parameter,
a numeric value.
For advanced usage, use the structure
as created by <code><a href="#topic+create_mean_param">create_mean_param</a></code>)</p>
</td></tr>
<tr><td><code id="parameter_to_xml_mean_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='parameter_to_xml_mu'>Internal function</h2><span id='topic+parameter_to_xml_mu'></span>

<h3>Description</h3>

<p>Converts a mu parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameter_to_xml_mu(parameter, beauti_options = create_beauti_options())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameter_to_xml_mu_+3A_parameter">parameter</code></td>
<td>
<p>a mu parameter,
a numeric value.
For advanced usage, use the structure
as created by <code><a href="#topic+create_mu_param">create_mu_param</a></code>)</p>
</td></tr>
<tr><td><code id="parameter_to_xml_mu_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='parameter_to_xml_rate_ac'>Internal function</h2><span id='topic+parameter_to_xml_rate_ac'></span>

<h3>Description</h3>

<p>Converts a 'rate AC' parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameter_to_xml_rate_ac(
  parameter,
  beauti_options = create_beauti_options(),
  which_name = "state_node"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameter_to_xml_rate_ac_+3A_parameter">parameter</code></td>
<td>
<p>a 'rate AC' parameter,
a numeric value.
For advanced usage, use the structure
as created by <code><a href="#topic+create_rate_ac_param">create_rate_ac_param</a></code>)</p>
</td></tr>
<tr><td><code id="parameter_to_xml_rate_ac_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
<tr><td><code id="parameter_to_xml_rate_ac_+3A_which_name">which_name</code></td>
<td>
<p>the name, can be <code>state_node</code> or <code>rate_name</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='parameter_to_xml_rate_ag'>Internal function</h2><span id='topic+parameter_to_xml_rate_ag'></span>

<h3>Description</h3>

<p>Converts a 'rate AG' parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameter_to_xml_rate_ag(
  parameter,
  beauti_options = create_beauti_options(),
  which_name = "state_node"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameter_to_xml_rate_ag_+3A_parameter">parameter</code></td>
<td>
<p>a 'rate AG' parameter,
a numeric value.
For advanced usage, use the structure
as created by <code><a href="#topic+create_rate_ag_param">create_rate_ag_param</a></code>)</p>
</td></tr>
<tr><td><code id="parameter_to_xml_rate_ag_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
<tr><td><code id="parameter_to_xml_rate_ag_+3A_which_name">which_name</code></td>
<td>
<p>the name, can be <code>state_node</code> or <code>rate_name</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='parameter_to_xml_rate_at'>Internal function</h2><span id='topic+parameter_to_xml_rate_at'></span>

<h3>Description</h3>

<p>Converts a 'rate AT' parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameter_to_xml_rate_at(
  parameter,
  beauti_options = create_beauti_options(),
  which_name = "state_node"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameter_to_xml_rate_at_+3A_parameter">parameter</code></td>
<td>
<p>a 'rate AT' parameter,
a numeric value.
For advanced usage, use the structure
as created by <code><a href="#topic+create_rate_at_param">create_rate_at_param</a></code>)</p>
</td></tr>
<tr><td><code id="parameter_to_xml_rate_at_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
<tr><td><code id="parameter_to_xml_rate_at_+3A_which_name">which_name</code></td>
<td>
<p>the name, can be <code>state_node</code> or <code>rate_name</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='parameter_to_xml_rate_cg'>Internal function</h2><span id='topic+parameter_to_xml_rate_cg'></span>

<h3>Description</h3>

<p>Converts a 'rate CG' parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameter_to_xml_rate_cg(
  parameter,
  beauti_options = create_beauti_options(),
  which_name = "state_node"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameter_to_xml_rate_cg_+3A_parameter">parameter</code></td>
<td>
<p>a 'rate CG' parameter,
a numeric value.
For advanced usage, use the structure
as created by <code><a href="#topic+create_rate_cg_param">create_rate_cg_param</a></code>)</p>
</td></tr>
<tr><td><code id="parameter_to_xml_rate_cg_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
<tr><td><code id="parameter_to_xml_rate_cg_+3A_which_name">which_name</code></td>
<td>
<p>the name, can be <code>state_node</code> or <code>rate_name</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='parameter_to_xml_rate_ct'>Internal function</h2><span id='topic+parameter_to_xml_rate_ct'></span>

<h3>Description</h3>

<p>Converts a 'rate CT' parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameter_to_xml_rate_ct(
  parameter,
  beauti_options = create_beauti_options(),
  which_name = "state_node"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameter_to_xml_rate_ct_+3A_parameter">parameter</code></td>
<td>
<p>a 'rate CT' parameter,
a numeric value.
For advanced usage, use the structure
as created by <code><a href="#topic+create_rate_ct_param">create_rate_ct_param</a></code>)</p>
</td></tr>
<tr><td><code id="parameter_to_xml_rate_ct_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
<tr><td><code id="parameter_to_xml_rate_ct_+3A_which_name">which_name</code></td>
<td>
<p>the name, can be <code>state_node</code> or <code>rate_name</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='parameter_to_xml_rate_gt'>Internal function</h2><span id='topic+parameter_to_xml_rate_gt'></span>

<h3>Description</h3>

<p>Converts a 'rate GT' parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameter_to_xml_rate_gt(
  parameter,
  beauti_options = create_beauti_options(),
  which_name = "state_node"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameter_to_xml_rate_gt_+3A_parameter">parameter</code></td>
<td>
<p>a 'rate GT' parameter,
a numeric value.
For advanced usage, use the structure
as created by <code><a href="#topic+create_rate_gt_param">create_rate_gt_param</a></code>)</p>
</td></tr>
<tr><td><code id="parameter_to_xml_rate_gt_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
<tr><td><code id="parameter_to_xml_rate_gt_+3A_which_name">which_name</code></td>
<td>
<p>the name, can be <code>state_node</code> or <code>rate_name</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='parameter_to_xml_scale'>Internal function</h2><span id='topic+parameter_to_xml_scale'></span>

<h3>Description</h3>

<p>Converts a scale parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameter_to_xml_scale(parameter, beauti_options = create_beauti_options())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameter_to_xml_scale_+3A_parameter">parameter</code></td>
<td>
<p>a scale parameter,
a numeric value.
For advanced usage, use the structure
as created by <code><a href="#topic+create_scale_param">create_scale_param</a></code>)</p>
</td></tr>
<tr><td><code id="parameter_to_xml_scale_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='parameter_to_xml_sigma'>Internal function</h2><span id='topic+parameter_to_xml_sigma'></span>

<h3>Description</h3>

<p>Converts a sigma parameter to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameter_to_xml_sigma(parameter, beauti_options = create_beauti_options())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameter_to_xml_sigma_+3A_parameter">parameter</code></td>
<td>
<p>a sigma parameter,
a numeric value.
For advanced usage, use the structure
as created by <code><a href="#topic+create_sigma_param">create_sigma_param</a></code>)</p>
</td></tr>
<tr><td><code id="parameter_to_xml_sigma_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='remove_beautier_folder'>Check there are no files in the default <a href="#topic+beautier">beautier</a> folder</h2><span id='topic+remove_beautier_folder'></span>

<h3>Description</h3>

<p>Check there are no files in the default <a href="#topic+beautier">beautier</a> folder.
The goal is to make sure no temporary files are left undeleted.
Will <a href="base.html#topic+stop">stop</a> if there are files in the <a href="#topic+beautier">beautier</a> folder.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_beautier_folder()
</code></pre>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <a href="#topic+remove_beautier_folder">remove_beautier_folder</a> to remove the default
'beautier' folder
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

remove_beautier_folder()

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='remove_empty_lines'>Remove all lines that are only whitespace</h2><span id='topic+remove_empty_lines'></span>

<h3>Description</h3>

<p>Remove all lines that are only whitespace
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_empty_lines(lines, trim = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="remove_empty_lines_+3A_lines">lines</code></td>
<td>
<p>character vector with text</p>
</td></tr>
<tr><td><code id="remove_empty_lines_+3A_trim">trim</code></td>
<td>
<p>FALSE if indentation must be preserved,
TRUE will remove all surrounding whitespace</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the lines with text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='remove_multiline'>Remove consecutive lines</h2><span id='topic+remove_multiline'></span>

<h3>Description</h3>

<p>Remove consecutive lines
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_multiline(text, lines_to_remove)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="remove_multiline_+3A_text">text</code></td>
<td>
<p>lines of characters</p>
</td></tr>
<tr><td><code id="remove_multiline_+3A_lines_to_remove">lines_to_remove</code></td>
<td>
<p>lines of character that need to be removed from text</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='rename_inference_model_filenames'>Rename the filenames in an inference model</h2><span id='topic+rename_inference_model_filenames'></span>

<h3>Description</h3>

<p>Rename the filenames in an inference model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rename_inference_model_filenames(inference_model, rename_fun)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rename_inference_model_filenames_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
<tr><td><code id="rename_inference_model_filenames_+3A_rename_fun">rename_fun</code></td>
<td>
<p>a function to rename a filename,
as can be checked by <a href="#topic+check_rename_fun">check_rename_fun</a>. This function should
have one argument, which will be a filename or <a href="base.html#topic+NA">NA</a>. The
function should <a href="base.html#topic+return">return</a> one filename (when passed one filename) or
one <a href="base.html#topic+NA">NA</a> (when passed one <a href="base.html#topic+NA">NA</a>).
Example rename functions are:
</p>

<ul>
<li> <p><a href="#topic+get_remove_dir_fun">get_remove_dir_fun</a> get a function that removes the directory
paths from the filenames, in effect turning these into local files
</p>
</li>
<li> <p><a href="#topic+get_replace_dir_fun">get_replace_dir_fun</a> get a function that replaces the directory
paths from the filenames
</p>
</li>
<li> <p><a href="#topic+get_remove_hex_fun">get_remove_hex_fun</a> get a function that removes the
hex string from filenames.
For example, <code>tracelog_82c1a522040.log</code> becomes <code>tracelog.log</code>
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>an inference model with the renamed filenames
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

inference_model &lt;- create_inference_model()
inference_model$mcmc$tracelog$filename &lt;- "trace.log"
inference_model$mcmc$screenlog$filename &lt;- "screen.log"
inference_model$mcmc$treelog$filename &lt;- "tree.log"
inference_model$tipdates_filename &lt;- "tipdates.csv"

# Nah, put the files in a folder
inference_model &lt;- rename_inference_model_filenames(
  inference_model = inference_model,
  rename_fun = get_replace_dir_fun("/home/john")
)

# Nah, put the files in anoth folder
inference_model &lt;- rename_inference_model_filenames(
  inference_model = inference_model,
  rename_fun = get_replace_dir_fun("/home/doe")
)

# Nah, store the files locally
rename_inference_model_filenames(
  inference_model = inference_model,
  rename_fun = get_remove_dir_fun()
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='rename_mcmc_filenames'>Rename the filenames within an MCMC</h2><span id='topic+rename_mcmc_filenames'></span>

<h3>Description</h3>

<p>Rename the filenames within an MCMC
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rename_mcmc_filenames(mcmc, rename_fun)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rename_mcmc_filenames_+3A_mcmc">mcmc</code></td>
<td>
<p>one MCMC.
Use <code><a href="#topic+create_mcmc">create_mcmc</a></code> to create an MCMC.
Use <code><a href="#topic+create_ns_mcmc">create_ns_mcmc</a></code> to create an MCMC
for a Nested Sampling run.
Use <code><a href="#topic+check_mcmc">check_mcmc</a></code> to check if an MCMC is valid.
Use <code><a href="#topic+rename_mcmc_filenames">rename_mcmc_filenames</a></code> to rename the filenames in an MCMC.</p>
</td></tr>
<tr><td><code id="rename_mcmc_filenames_+3A_rename_fun">rename_fun</code></td>
<td>
<p>a function to rename a filename,
as can be checked by <a href="#topic+check_rename_fun">check_rename_fun</a>. This function should
have one argument, which will be a filename or <a href="base.html#topic+NA">NA</a>. The
function should <a href="base.html#topic+return">return</a> one filename (when passed one filename) or
one <a href="base.html#topic+NA">NA</a> (when passed one <a href="base.html#topic+NA">NA</a>).
Example rename functions are:
</p>

<ul>
<li> <p><a href="#topic+get_remove_dir_fun">get_remove_dir_fun</a> get a function that removes the directory
paths from the filenames, in effect turning these into local files
</p>
</li>
<li> <p><a href="#topic+get_replace_dir_fun">get_replace_dir_fun</a> get a function that replaces the directory
paths from the filenames
</p>
</li>
<li> <p><a href="#topic+get_remove_hex_fun">get_remove_hex_fun</a> get a function that removes the
hex string from filenames.
For example, <code>tracelog_82c1a522040.log</code> becomes <code>tracelog.log</code>
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>an 'mcmc' (see <a href="#topic+create_mcmc">create_mcmc</a>) with renamed filenames
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# Create an MCMC with local filenames
mcmc &lt;- create_mcmc()
mcmc$tracelog$filename &lt;- "trace.log"
mcmc$screenlog$filename &lt;- "screen.log"
mcmc$treelog$filename &lt;- "tree.log"

# Nah, files should be put in '/home/john' folder
mcmc &lt;- rename_mcmc_filenames(
  mcmc = mcmc,
  rename_fun = get_replace_dir_fun("/home/john")
)

# Nah, files should be put in '/home/doe' folder instead
mcmc &lt;- rename_mcmc_filenames(
  mcmc = mcmc,
  rename_fun = get_replace_dir_fun("/home/doe")
)

# Nah, files should be put in local folder instead
mcmc &lt;- rename_mcmc_filenames(
  mcmc = mcmc,
  rename_fun = get_remove_dir_fun()
)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='rln_clock_model_to_xml_mean_rate_prior'>Internal function</h2><span id='topic+rln_clock_model_to_xml_mean_rate_prior'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rln_clock_model_to_xml_mean_rate_prior(rln_clock_model, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rln_clock_model_to_xml_mean_rate_prior_+3A_rln_clock_model">rln_clock_model</code></td>
<td>
<p>a Relaxed Log-Normal clock model,
as returned by <code><a href="#topic+create_rln_clock_model">create_rln_clock_model</a></code></p>
</td></tr>
<tr><td><code id="rln_clock_model_to_xml_mean_rate_prior_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='rln_clock_model_to_xml_operators'>Internal function</h2><span id='topic+rln_clock_model_to_xml_operators'></span>

<h3>Description</h3>

<p>Converts an RLN clock model to the <code>operators</code> section of the
XML as text
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rln_clock_model_to_xml_operators(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rln_clock_model_to_xml_operators_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of XML strings
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='rln_clock_model_to_xml_prior_distr'>Internal function</h2><span id='topic+rln_clock_model_to_xml_prior_distr'></span>

<h3>Description</h3>

<p>Internal function to converts a relaxed log-normal clock model
to the <code>prior</code> section of the XML as text
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rln_clock_model_to_xml_prior_distr(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rln_clock_model_to_xml_prior_distr_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of XML strings
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

 # &lt;distribution id="posterior" spec="util.CompoundDistribution"&gt;
 #     &lt;distribution id="prior" spec="util.CompoundDistribution"&gt;
 #       HERE, where the ID of the distribution is 'prior'
 #     &lt;/distribution&gt;
 #     &lt;distribution id="likelihood" ...&gt;
 #     &lt;/distribution&gt;
 # &lt;/distribution&gt;

# Must be an initialized inference model
inference_model &lt;- create_inference_model(
  clock_model = create_rln_clock_model(
    id = "test_output_0",
    ucldstdev_distr = create_gamma_distr(
      id = 0,
      alpha = create_alpha_param(id = 2, value = "0.5396"),
      beta = create_beta_param(id = 3, value = "0.3819")
    ),
    mean_rate_prior_distr = create_uniform_distr(id = 1),
    mparam_id = 1
  )
)

rln_clock_model_to_xml_prior_distr(inference_model)

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='rln_clock_model_to_xml_state'>Internal function</h2><span id='topic+rln_clock_model_to_xml_state'></span>

<h3>Description</h3>

<p>Converts an RLN clock model to the 'state' section of the
XML as text
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rln_clock_model_to_xml_state(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rln_clock_model_to_xml_state_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text, without indentation nor <code>state</code>
tags
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='rln_clock_model_to_xml_tracelog'>Internal function</h2><span id='topic+rln_clock_model_to_xml_tracelog'></span>

<h3>Description</h3>

<p>Creates the RLN clock model's XML for the tracelog section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rln_clock_model_to_xml_tracelog(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rln_clock_model_to_xml_tracelog_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of XML strings
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# &lt;logger id="tracelog" ...&gt;
#'   # Here
# &lt;/logger&gt;

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='rnd_phylo_to_xml_init'>Creates the XML of a random phylogeny,
as used in the <code>init</code> section</h2><span id='topic+rnd_phylo_to_xml_init'></span>

<h3>Description</h3>

<p>Creates the XML text for the <code>beast</code> tag of a BEAST2 parameter file,
which is directly after the XML
declaration (created by <a href="#topic+create_xml_declaration">create_xml_declaration</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rnd_phylo_to_xml_init(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rnd_phylo_to_xml_init_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>init</code> tag has these elements:
</p>
<pre>
  &lt;init id=\"RandomTree.t:[...]&gt;
      &lt;populationModel[...]&gt;
      [...]
      &lt;/populationModel&gt;
  &lt;/init&gt;
</pre>


<h3>Value</h3>

<p>the phylogeny as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='s_parameter_to_xml'>Internal function</h2><span id='topic+s_parameter_to_xml'></span>

<h3>Description</h3>

<p>Converts an 's_param' to XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>s_parameter_to_xml(parameter, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="s_parameter_to_xml_+3A_parameter">parameter</code></td>
<td>
<p>a s parameter,
a numeric value.
For advanced usage, use the structure
as created by <code><a href="#topic+create_s_param">create_s_param</a></code>)</p>
</td></tr>
<tr><td><code id="s_parameter_to_xml_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the parameter as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>s_parameter_to_xml(
  create_s_param(id = 4, value = 1.25),
  beauti_options = create_beauti_options_v2_4()
)
s_parameter_to_xml(
  create_s_param(id = 4, value = 1.25),
  beauti_options = create_beauti_options_v2_6()
)
</code></pre>

<hr>
<h2 id='site_model_to_xml_operators'>Converts a site model to XML,
used in the <code>operators</code> section</h2><span id='topic+site_model_to_xml_operators'></span>

<h3>Description</h3>

<p>Converts a site model to XML,
used in the <code>operators</code> section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>site_model_to_xml_operators(site_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="site_model_to_xml_operators_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the site model as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='site_model_to_xml_prior_distr'>Internal function</h2><span id='topic+site_model_to_xml_prior_distr'></span>

<h3>Description</h3>

<p>Converts a site model to XML,
used in the <code>prior</code> section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>site_model_to_xml_prior_distr(site_model, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="site_model_to_xml_prior_distr_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
<tr><td><code id="site_model_to_xml_prior_distr_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the site model as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>site_model_to_xml_prior_distr(
  site_model = create_jc69_site_model(id = 1),
  beauti_options = create_beauti_options()
)
site_model_to_xml_prior_distr(
  site_model = create_hky_site_model(
    id = 1,
    kappa_prior_distr = create_uniform_distr(id = 2)
  ),
  beauti_options = create_beauti_options()
)
site_model_to_xml_prior_distr(
  site_model = create_tn93_site_model(
    id = 1,
    kappa_1_prior_distr = create_uniform_distr(id = 2),
    kappa_2_prior_distr = create_uniform_distr(id = 3)
  ),
  beauti_options = create_beauti_options()
)
site_model_to_xml_prior_distr(
  site_model = create_gtr_site_model(
    id = 1,
    rate_ac_prior_distr = create_uniform_distr(id = 2),
    rate_ag_prior_distr = create_uniform_distr(id = 3),
    rate_at_prior_distr = create_uniform_distr(id = 4),
    rate_cg_prior_distr = create_uniform_distr(id = 5),
    rate_gt_prior_distr = create_uniform_distr(id = 6)
  ),
  beauti_options = create_beauti_options()
)
</code></pre>

<hr>
<h2 id='site_model_to_xml_state'>Internal function to convert a site model to XML,
used in the 'state' section</h2><span id='topic+site_model_to_xml_state'></span>

<h3>Description</h3>

<p>Internal function to convert a site model to XML,
used in the 'state' section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>site_model_to_xml_state(site_model, beauti_options = create_beauti_options())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="site_model_to_xml_state_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
<tr><td><code id="site_model_to_xml_state_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the site model as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='site_model_to_xml_tracelog'>Creates the site model's XML for the tracelog section</h2><span id='topic+site_model_to_xml_tracelog'></span>

<h3>Description</h3>

<p>Creates the site model's XML for the tracelog section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>site_model_to_xml_tracelog(site_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="site_model_to_xml_tracelog_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>all site models' tracelog section is created
by <code><a href="#topic+site_models_to_xml_tracelog">site_models_to_xml_tracelog</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# &lt;logger id="tracelog" ...&gt;
#'   # Here
# &lt;/logger&gt;

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='site_models_to_xml_operators'>Write the XML <code>operators</code> section from the site models.</h2><span id='topic+site_models_to_xml_operators'></span>

<h3>Description</h3>

<p>Write the XML <code>operators</code> section from the site models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>site_models_to_xml_operators(site_models)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="site_models_to_xml_operators_+3A_site_models">site_models</code></td>
<td>
<p>one or more site models,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='site_models_to_xml_prior_distr'>Represent the site models as XML</h2><span id='topic+site_models_to_xml_prior_distr'></span>

<h3>Description</h3>

<p>Represent the site models as XML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>site_models_to_xml_prior_distr(site_models, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="site_models_to_xml_prior_distr_+3A_site_models">site_models</code></td>
<td>
<p>one or more site models,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
<tr><td><code id="site_models_to_xml_prior_distr_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

 # &lt;distribution id="posterior" spec="util.CompoundDistribution"&gt;
 #     &lt;distribution id="prior" spec="util.CompoundDistribution"&gt;
 #       HERE, where the ID of the distribution is 'prior'
 #     &lt;/distribution&gt;
 #     &lt;distribution id="likelihood" ...&gt;
 #     &lt;/distribution&gt;
 # &lt;/distribution&gt;

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='site_models_to_xml_tracelog'>Creates the site models' XML for the tracelog section</h2><span id='topic+site_models_to_xml_tracelog'></span>

<h3>Description</h3>

<p>Creates the site models' XML for the tracelog section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>site_models_to_xml_tracelog(site_models)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="site_models_to_xml_tracelog_+3A_site_models">site_models</code></td>
<td>
<p>one or more site models,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the complete tracelog section is created
by <code><a href="#topic+create_tracelog_xml">create_tracelog_xml</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# &lt;logger id="tracelog" ...&gt;
#'   # Here
# &lt;/logger&gt;

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='strict_clock_model_to_xml_operators'>Internal function</h2><span id='topic+strict_clock_model_to_xml_operators'></span>

<h3>Description</h3>

<p>Converts a clock model to the <code>operators</code> section of the
XML as text
</p>


<h3>Usage</h3>

<pre><code class='language-R'>strict_clock_model_to_xml_operators(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="strict_clock_model_to_xml_operators_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of XML strings
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='strict_clock_model_to_xml_prior_distr'>Internal function</h2><span id='topic+strict_clock_model_to_xml_prior_distr'></span>

<h3>Description</h3>

<p>Internal function to converts a strict clock model
to the <code>prior</code> section of the XML as text
</p>


<h3>Usage</h3>

<pre><code class='language-R'>strict_clock_model_to_xml_prior_distr(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="strict_clock_model_to_xml_prior_distr_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of XML strings
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()
 # &lt;distribution id="posterior" spec="util.CompoundDistribution"&gt;
 #     &lt;distribution id="prior" spec="util.CompoundDistribution"&gt;
 #       HERE, where the ID of the distribution is 'prior'
 #     &lt;/distribution&gt;
 #     &lt;distribution id="likelihood" ...&gt;
 #     &lt;/distribution&gt;
 # &lt;/distribution&gt;
strict_clock_model_to_xml_prior_distr(
  inference_model = create_inference_model()
)
check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='strict_clock_model_to_xml_state'>Internal function</h2><span id='topic+strict_clock_model_to_xml_state'></span>

<h3>Description</h3>

<p>Converts a strict clock model to the 'state' section of the
XML as text
</p>


<h3>Usage</h3>

<pre><code class='language-R'>strict_clock_model_to_xml_state(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="strict_clock_model_to_xml_state_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text, without indentation nor <code>state</code>
tags
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='strict_clock_model_to_xml_tracelog'>Internal function</h2><span id='topic+strict_clock_model_to_xml_tracelog'></span>

<h3>Description</h3>

<p>Creates a strict clock model's XML for the tracelog section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>strict_clock_model_to_xml_tracelog(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="strict_clock_model_to_xml_tracelog_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of XML strings
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# &lt;logger id="tracelog" ...&gt;
#'   # Here
# &lt;/logger&gt;

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='taxa_to_xml_tree'>Internal function</h2><span id='topic+taxa_to_xml_tree'></span>

<h3>Description</h3>

<p>Internal function to creates the '<code>tree</code>' section
of a BEAST2 XML parameter file,
which is part of a '<code>state</code>' section,
without being indented.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxa_to_xml_tree(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxa_to_xml_tree_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>tree</code> tag has these elements:
</p>
<pre>
   &lt;tree[...]&gt;
       &lt;taxonset[...]&gt;
       [...]
       &lt;/taxonset&gt;
    &lt;/run&gt;
</pre>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='tipdate_taxa_to_xml_trait'>Internal function</h2><span id='topic+tipdate_taxa_to_xml_trait'></span>

<h3>Description</h3>

<p>Internal function to creates the '<code>trait</code>' section
of a BEAST2 XML parameter file,
which is part of a '<code>tree</code>' section,
without being indented.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tipdate_taxa_to_xml_trait(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tipdate_taxa_to_xml_trait_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>tree</code> tag has these elements:
</p>
<pre>
&lt;run[...]&gt;
  &lt;state[...]&gt;
    &lt;tree[...]&gt;
      &lt;trait[...]&gt;
      This part
      &lt;/trait&gt;
    &lt;/tree&gt;
  &lt;/run&gt;
&lt;/state&gt;
</pre>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='tipdate_taxa_to_xml_tree'>Internal function</h2><span id='topic+tipdate_taxa_to_xml_tree'></span>

<h3>Description</h3>

<p>Creates the <code>tree</code> section
(part of the <code>state</code> section)
when there is tip-dating
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tipdate_taxa_to_xml_tree(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tipdate_taxa_to_xml_tree_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the random phylogeny as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='tn93_site_model_to_xml_prior_distr'>Internal function</h2><span id='topic+tn93_site_model_to_xml_prior_distr'></span>

<h3>Description</h3>

<p>Converts a TN93 site model to XML,
used in the <code>prior</code> section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tn93_site_model_to_xml_prior_distr(site_model, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tn93_site_model_to_xml_prior_distr_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
<tr><td><code id="tn93_site_model_to_xml_prior_distr_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the site model as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tn93_site_model_to_xml_prior_distr(
  site_model = create_tn93_site_model(
    id = 1,
    kappa_1_prior_distr = create_uniform_distr(id = 2),
    kappa_2_prior_distr = create_uniform_distr(id = 3)
  ),
  beauti_options = create_beauti_options()
)
</code></pre>

<hr>
<h2 id='tn93_site_model_to_xml_state'>Converts a site model to XML,
used in the <code>state</code> section</h2><span id='topic+tn93_site_model_to_xml_state'></span>

<h3>Description</h3>

<p>Converts a site model to XML,
used in the <code>state</code> section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tn93_site_model_to_xml_state(
  site_model,
  beauti_options = create_beauti_options()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tn93_site_model_to_xml_state_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as returned by <code><a href="#topic+create_site_model">create_site_model</a></code></p>
</td></tr>
<tr><td><code id="tn93_site_model_to_xml_state_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the site model as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='tree_model_to_tracelog_xml'>Internal function</h2><span id='topic+tree_model_to_tracelog_xml'></span>

<h3>Description</h3>

<p>Creates the tree models' XML for the tracelog section.
That is, all XML tags that have the word 'tree' in them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_model_to_tracelog_xml(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tree_model_to_tracelog_xml_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Note</h3>

<p>use site_models just because it contains all IDs
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the complete tracelog section is created
by <code><a href="#topic+create_tracelog_xml">create_tracelog_xml</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# &lt;logger id="tracelog" ...&gt;
#'   # Here
# &lt;/logger&gt;

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='tree_prior_to_xml_operators'>Internal function</h2><span id='topic+tree_prior_to_xml_operators'></span>

<h3>Description</h3>

<p>Creates the XML of a tree prior,
as used in the <code>operators</code> section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_prior_to_xml_operators(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tree_prior_to_xml_operators_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the tree prior as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='tree_prior_to_xml_prior_distr'>Creates the distribution section in the prior section of the
distribution section of a BEAST2 XML parameter file.</h2><span id='topic+tree_prior_to_xml_prior_distr'></span>

<h3>Description</h3>

<p>These lines start with '&lt;distribution id='
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_prior_to_xml_prior_distr(tree_prior, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tree_prior_to_xml_prior_distr_+3A_tree_prior">tree_prior</code></td>
<td>
<p>a tree priors,
as returned by <code><a href="#topic+create_tree_prior">create_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="tree_prior_to_xml_prior_distr_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

 # &lt;distribution id="posterior" spec="util.CompoundDistribution"&gt;
 #     &lt;distribution id="prior" spec="util.CompoundDistribution"&gt;
 #       HERE, where the ID of the distribution is 'prior'
 #     &lt;/distribution&gt;
 #     &lt;distribution id="likelihood" ...&gt;
 #     &lt;/distribution&gt;
 # &lt;/distribution&gt;

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='tree_prior_to_xml_state'>Creates the XML of a tree prior,
as used in the <code>state</code> section</h2><span id='topic+tree_prior_to_xml_state'></span>

<h3>Description</h3>

<p>Creates the XML of a tree prior,
as used in the <code>state</code> section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_prior_to_xml_state(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tree_prior_to_xml_state_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the tree prior as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='tree_prior_to_xml_tracelog'>Creates the tree prior's XML for the tracelog section</h2><span id='topic+tree_prior_to_xml_tracelog'></span>

<h3>Description</h3>

<p>Creates the tree prior's XML for the tracelog section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_prior_to_xml_tracelog(tree_prior)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tree_prior_to_xml_tracelog_+3A_tree_prior">tree_prior</code></td>
<td>
<p>a tree priors,
as returned by <code><a href="#topic+create_tree_prior">create_tree_prior</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>all tree priors' tracelog section is created
by <code><a href="#topic+tree_priors_to_xml_tracelog">tree_priors_to_xml_tracelog</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# &lt;logger id="tracelog" ...&gt;
#'   # Here
# &lt;/logger&gt;

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='tree_priors_to_xml_prior_distr'>Creates the distribution section in the prior section of the
distribution section of a BEAST2 XML parameter file.</h2><span id='topic+tree_priors_to_xml_prior_distr'></span>

<h3>Description</h3>

<p>These lines start with '&lt;distribution id='
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_priors_to_xml_prior_distr(tree_priors, beauti_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tree_priors_to_xml_prior_distr_+3A_tree_priors">tree_priors</code></td>
<td>
<p>one or more tree priors,
as returned by <code><a href="#topic+create_tree_prior">create_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="tree_priors_to_xml_prior_distr_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

 # &lt;distribution id="posterior" spec="util.CompoundDistribution"&gt;
 #     &lt;distribution id="prior" spec="util.CompoundDistribution"&gt;
 #       HERE, where the ID of the distribution is 'prior'
 #     &lt;/distribution&gt;
 #     &lt;distribution id="likelihood" ...&gt;
 #     &lt;/distribution&gt;
 # &lt;/distribution&gt;

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='tree_priors_to_xml_tracelog'>Creates the tree priors' XML for the tracelog section</h2><span id='topic+tree_priors_to_xml_tracelog'></span>

<h3>Description</h3>

<p>Creates the tree priors' XML for the tracelog section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_priors_to_xml_tracelog(tree_priors)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tree_priors_to_xml_tracelog_+3A_tree_priors">tree_priors</code></td>
<td>
<p>one or more tree priors,
as returned by <code><a href="#topic+create_tree_prior">create_tree_prior</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>the complete tracelog section is created
by <code><a href="#topic+create_tracelog_xml">create_tracelog_xml</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

# &lt;logger id="tracelog" ...&gt;
#'   # Here
# &lt;/logger&gt;

check_empty_beautier_folder()
</code></pre>

<hr>
<h2 id='unindent'>Unindents text</h2><span id='topic+unindent'></span>

<h3>Description</h3>

<p>Unindents text
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unindent(text)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unindent_+3A_text">text</code></td>
<td>
<p>one or more lines of text</p>
</td></tr>
</table>


<h3>Value</h3>

<p>unindented lines of text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='yule_tree_prior_to_xml_operators'>Internal function</h2><span id='topic+yule_tree_prior_to_xml_operators'></span>

<h3>Description</h3>

<p>Creates the XML of a Yule tree prior,
as used in the <code>operators</code> section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>yule_tree_prior_to_xml_operators(inference_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="yule_tree_prior_to_xml_operators_+3A_inference_model">inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use <a href="#topic+create_inference_model">create_inference_model</a> to create an inference model.
Use <a href="#topic+check_inference_model">check_inference_model</a> to check if  an inference model is valid.
Use <a href="#topic+rename_inference_model_filenames">rename_inference_model_filenames</a> to rename the files in an
inference model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the tree prior as XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='yule_tree_prior_to_xml_prior_distr'>Creates the <code>prior</code> section in the prior section of
the prior section of the distribution section
of a BEAST2 XML parameter file for a Yule tree prior</h2><span id='topic+yule_tree_prior_to_xml_prior_distr'></span>

<h3>Description</h3>

<p>Creates the <code>prior</code> section in the prior section of
the prior section of the distribution section
of a BEAST2 XML parameter file for a Yule tree prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>yule_tree_prior_to_xml_prior_distr(
  yule_tree_prior,
  beauti_options = create_beauti_options()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="yule_tree_prior_to_xml_prior_distr_+3A_yule_tree_prior">yule_tree_prior</code></td>
<td>
<p>a Yule tree_prior,
as created by <code><a href="#topic+create_yule_tree_prior">create_yule_tree_prior</a></code></p>
</td></tr>
<tr><td><code id="yule_tree_prior_to_xml_prior_distr_+3A_beauti_options">beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code><a href="#topic+create_beauti_options">create_beauti_options</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>lines of XML text
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_empty_beautier_folder()

 # &lt;distribution id="posterior" spec="util.CompoundDistribution"&gt;
 #     &lt;distribution id="prior" spec="util.CompoundDistribution"&gt;
 #       HERE, where the ID of the distribution is 'prior'
 #     &lt;/distribution&gt;
 #     &lt;distribution id="likelihood" ...&gt;
 #     &lt;/distribution&gt;
 # &lt;/distribution&gt;

check_empty_beautier_folder()
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
