<!DOCTYPE html><html lang="en"><head><title>Help for package BED</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BED}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BED'><p>Biological Entity Dictionary (BED)</p></a></li>
<li><a href='#bedCall'><p>Call a function on the BED graph</p></a></li>
<li><a href='#bedImport'><p>Feeding BED: Imports a data.frame in the BED graph database</p></a></li>
<li><a href='#BEIDList'><p>Create a BEIDList</p></a></li>
<li><a href='#BEIDs'><p>Get the BEIDs from an object</p></a></li>
<li><a href='#beidsServer'><p>Shiny module for searching BEIDs</p></a></li>
<li><a href='#beIDsToAllScopes'><p>Find all BEID and ProbeID corresponding to a BE</p></a></li>
<li><a href='#cacheBedCall'><p>Cached neo4j call</p></a></li>
<li><a href='#cacheBedResult'><p>Put a BED query result in cache</p></a></li>
<li><a href='#checkBedCache'><p>Check BED cache</p></a></li>
<li><a href='#checkBedConn'><p>Check if there is a connection to a BED database</p></a></li>
<li><a href='#checkBeIds'><p>Check biological entities (BE) identifiers</p></a></li>
<li><a href='#cleanDubiousXRef'><p>Identify and remove dubious cross-references</p></a></li>
<li><a href='#clearBedCache'><p>Clear the BED cache SQLite database</p></a></li>
<li><a href='#compareBedInstances'><p>Compare 2 BED database instances</p></a></li>
<li><a href='#connectToBed'><p>Connect to a neo4j BED database</p></a></li>
<li><a href='#convBeIdLists'><p>Converts lists of BE IDs</p></a></li>
<li><a href='#convBeIds'><p>Converts BE IDs</p></a></li>
<li><a href='#convDfBeIds'><p>Add BE ID conversion to a data frame</p></a></li>
<li><a href='#dumpEnsCore'><p>Feeding BED: Dump table from the Ensembl core database</p></a></li>
<li><a href='#dumpNcbiDb'><p>Feeding BED: Dump tables from the NCBI gene DATA</p></a></li>
<li><a href='#dumpNcbiTax'><p>Feeding BED: Dump tables with taxonomic information from NCBI</p></a></li>
<li><a href='#dumpUniprotDb'><p>Feeding BED: Dump and preprocess flat data files from Uniprot</p></a></li>
<li><a href='#exploreBe'><p>Explore BE identifiers</p></a></li>
<li><a href='#exploreConvPath'><p>Explore the shortest convertion path between two identifiers</p></a></li>
<li><a href='#filterByBEID'><p>Filter an object to keep only a set of BEIDs</p></a></li>
<li><a href='#findBe'><p>Find Biological Entity</p></a></li>
<li><a href='#findBeids'><p>Find Biological Entity identifiers</p></a></li>
<li><a href='#firstCommonUpstreamBe'><p>First common upstream BE</p></a></li>
<li><a href='#focusOnScope'><p>Focus a BE related object on a specific identifier (BEID) scope</p></a></li>
<li><a href='#focusOnScope.BEIDList'><p>Convert a BEIDList object in a specific identifier (BEID) scope</p></a></li>
<li><a href='#forgetBedConnection'><p>Forget a BED connection</p></a></li>
<li><a href='#genBePath'><p>Construct CQL sub-query to map 2 biological entity</p></a></li>
<li><a href='#geneIDsToAllScopes'><p>Find all GeneID, ObjectID, TranscriptID, PeptideID and ProbeID corresponding to a Gene in any organism</p></a></li>
<li><a href='#genProbePath'><p>Identify the biological entity (BE) targeted by probes and</p>
construct the CQL sub-query to map probes to the BE</a></li>
<li><a href='#getAllBeIdSources'><p>List all the source databases of BE identifiers whatever the BE type</p></a></li>
<li><a href='#getBeIdConvTable'><p>Get a conversion table between biological entity (BE) identifiers</p></a></li>
<li><a href='#getBeIdDescription'><p>Get description of Biological Entity identifiers</p></a></li>
<li><a href='#getBeIdNames'><p>Get names of Biological Entity identifiers</p></a></li>
<li><a href='#getBeIdNameTable'><p>Get a table of biological entity (BE) identifiers and names</p></a></li>
<li><a href='#getBeIds'><p>Get biological entities identifiers</p></a></li>
<li><a href='#getBeIdSymbols'><p>Get symbols of Biological Entity identifiers</p></a></li>
<li><a href='#getBeIdSymbolTable'><p>Get a table of biological entity (BE) identifiers and symbols</p></a></li>
<li><a href='#getBeIdURL'><p>Get reference URLs for BE IDs</p></a></li>
<li><a href='#getDirectOrigin'><p>Get the direct origin of BE identifiers</p></a></li>
<li><a href='#getDirectProduct'><p>Get the direct product of BE identifiers</p></a></li>
<li><a href='#getEnsemblGeneIds'><p>Feeding BED: Download Ensembl DB and load gene information in BED</p></a></li>
<li><a href='#getEnsemblPeptideIds'><p>Feeding BED: Download Ensembl DB and load peptide information in BED</p></a></li>
<li><a href='#getEnsemblTranscriptIds'><p>Feeding BED: Download Ensembl DB and load transcript information in BED</p></a></li>
<li><a href='#getGeneDescription'><p>Get description of genes corresponding to Biological Entity identifiers</p></a></li>
<li><a href='#getHomTable'><p>Get gene homologs between 2 organisms</p></a></li>
<li><a href='#getNcbiGeneTransPep'><p>Feeding BED: Download NCBI gene DATA and load gene, transcript and peptide</p>
information in BED</a></li>
<li><a href='#getOrgNames'><p>Get organism names from taxonomy IDs</p></a></li>
<li><a href='#getRelevantIds'><p>Get relevant IDs for a formerly identified BE in a context of interest</p></a></li>
<li><a href='#getTargetedBe'><p>Identify the biological entity (BE) targeted by probes</p></a></li>
<li><a href='#getTaxId'><p>Get taxonomy ID of an organism name</p></a></li>
<li><a href='#getUniprot'><p>Feeding BED: Download Uniprot information in BED</p></a></li>
<li><a href='#guessIdScope'><p>Guess biological entity (BE), database source and organism of a vector</p>
of identifiers.</a></li>
<li><a href='#identicalScopes'><p>Check if two objects have the same BEID scope</p></a></li>
<li><a href='#is.BEIDList'><p>Check if the provided object is a BEIDList</p></a></li>
<li><a href='#largestBeSource'><p>Autoselect source of biological entity identifiers</p></a></li>
<li><a href='#listBe'><p>Lists all the biological entities (BE) available in the BED database</p></a></li>
<li><a href='#listBeIdSources'><p>Lists all the databases taken into account in the BED database</p>
for a biological entity (BE)</a></li>
<li><a href='#listDBAttributes'><p>List all attributes provided by a BEDB</p></a></li>
<li><a href='#listOrganisms'><p>Lists all the organisms available in the BED database</p></a></li>
<li><a href='#listPlatforms'><p>Lists all the probe platforms available in the BED database</p></a></li>
<li><a href='#loadBE'><p>Feeding BED: Load biological entities in BED</p></a></li>
<li><a href='#loadBeAttribute'><p>Feeding BED: Load attributes for biological entities in BED</p></a></li>
<li><a href='#loadBedModel'><p>Feeding BED: Load BED data model in neo4j</p></a></li>
<li><a href='#loadBedOtherIndexes'><p>Feeding BED: Load additional indexes in neo4j</p></a></li>
<li><a href='#loadBedResult'><p>Get a BED query result from cache</p></a></li>
<li><a href='#loadBENames'><p>Feeding BED: Load names associated to BEIDs</p></a></li>
<li><a href='#loadBESymbols'><p>Feeding BED: Load symbols associated to BEIDs</p></a></li>
<li><a href='#loadBEVersion'><p>Feeding BED: Load biological entities in BED with information about</p>
DB version</a></li>
<li><a href='#loadCodesFor'><p>Feeding BED: Load correspondance between genes and objects as coding events</p></a></li>
<li><a href='#loadCorrespondsTo'><p>Feeding BED: Load correspondances between BE IDs</p></a></li>
<li><a href='#loadHistory'><p>Feeding BED: Load history of BEIDs</p></a></li>
<li><a href='#loadIsAssociatedTo'><p>Feeding BED: Load BE ID associations</p></a></li>
<li><a href='#loadIsExpressedAs'><p>Feeding BED: Load correspondance between genes and transcripts as</p>
expression events</a></li>
<li><a href='#loadIsHomologOf'><p>Feeding BED: Load homology between BE IDs</p></a></li>
<li><a href='#loadIsTranslatedIn'><p>Feeding BED: Load correspondance between transcripts and peptides as</p>
translation events</a></li>
<li><a href='#loadLuceneIndexes'><p>Feeding BED: Create Lucene indexes in neo4j</p></a></li>
<li><a href='#loadNCBIEntrezGOFunctions'><p>Feeding BED: Load in BED GO functions associated to Entrez gene IDs</p>
from NCBI</a></li>
<li><a href='#loadNcbiTax'><p>Feeding BED: Load taxonomic information from NCBI</p></a></li>
<li><a href='#loadOrganisms'><p>Feeding BED: Load organisms in BED</p></a></li>
<li><a href='#loadPlf'><p>Feeding BED: Load a probes platform</p></a></li>
<li><a href='#loadProbes'><p>Feeding BED: Load probes targeting BE IDs</p></a></li>
<li><a href='#lsBedCache'><p>List all the BED queries in cache and the total size of the cache</p></a></li>
<li><a href='#lsBedConnections'><p>List all registered BED connection</p></a></li>
<li><a href='#metadata'><p>Get object metadata</p></a></li>
<li><a href='#metadata+26lt+3B-'><p>Set object metadata</p></a></li>
<li><a href='#registerBEDB'><p>Feeding BED: Register a database of biological entities in BED DB</p></a></li>
<li><a href='#scope'><p>Get the BEID scope of an object</p></a></li>
<li><a href='#scopes'><p>Get the BEID scopes of an object</p></a></li>
<li><a href='#searchBeid'><p>Search a BEID</p></a></li>
<li><a href='#searchId'><p>Search identifier, symbol or name information</p></a></li>
<li><a href='#setBedVersion'><p>Feeding BED: Set the BED version</p></a></li>
<li><a href='#showBedDataModel'><p>Show the data model of BED</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Biological Entity Dictionary (BED)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.6.0</td>
</tr>
<tr>
<td>Description:</td>
<td>An interface for the 'Neo4j' database providing
    mapping between different identifiers of biological entities.
    This Biological Entity Dictionary (BED)
    has been developed to address three main challenges.
    The first one is related to the completeness of identifier mappings.
    Indeed, direct mapping information provided by the different systems
    are not always complete and can be enriched by mappings provided by other
    resources.
    More interestingly, direct mappings not identified by any of these
    resources can be indirectly inferred by using mappings to a third reference.
    For example, many human Ensembl gene ID are not directly mapped to any
    Entrez gene ID but such mappings can be inferred using respective mappings
    to HGNC ID. The second challenge is related to the mapping of deprecated
    identifiers. Indeed, entity identifiers can change from one resource
    release to another. The identifier history is provided by some resources,
    such as Ensembl or the NCBI, but it is generally not used by mapping tools.
    The third challenge is related to the automation of the mapping process
    according to the relationships between the biological entities of interest.
    Indeed, mapping between gene and protein ID scopes should not be done
    the same way than between two scopes regarding gene ID.
    Also, converting identifiers from different organisms should be possible
    using gene orthologs information.
    The method has been published by
    Godard and van Eyll (2018) &lt;<a href="https://doi.org/10.12688%2Ff1000research.13925.3">doi:10.12688/f1000research.13925.3</a>&gt;.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://patzaw.github.io/BED/">https://patzaw.github.io/BED/</a>, <a href="https://github.com/patzaw/BED">https://github.com/patzaw/BED</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/patzaw/BED/issues">https://github.com/patzaw/BED/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6), neo2R (&ge; 2.4.1), visNetwork</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, readr, stringr, utils, shiny (&ge; 0.13), DT, miniUI (&ge;
0.1.1), rstudioapi (&ge; 0.5)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, biomaRt, GEOquery, base64enc, htmltools,
webshot2, RCurl</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-11-09 16:03:05 UTC; pgodard</td>
</tr>
<tr>
<td>Author:</td>
<td>Patrice Godard <a href="https://orcid.org/0000-0001-6257-9730"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Patrice Godard &lt;patrice.godard@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-11-09 16:40:04 UTC</td>
</tr>
</table>
<hr>
<h2 id='BED'>Biological Entity Dictionary (BED)</h2><span id='topic+BED'></span>

<h3>Description</h3>

<p>An interface for the neo4j database providing
mapping between different identifiers of biological entities.
This Biological Entity Dictionary (BED)
has been developed to address three main challenges.
The first one is related to the completeness of identifier mappings.
Indeed, direct mapping information provided by the different systems
are not always complete and can be enriched by mappings provided by other
resources.
More interestingly, direct mappings not identified by any of these
resources can be indirectly inferred by using mappings to a third reference.
For example, many human Ensembl gene ID are not directly mapped to any
Entrez gene ID but such mappings can be inferred using respective mappings
to HGNC ID. The second challenge is related to the mapping of deprecated
identifiers. Indeed, entity identifiers can change from one resource
release to another. The identifier history is provided by some resources,
such as Ensembl or the NCBI, but it is generally not used by mapping tools.
The third challenge is related to the automation of the mapping process
according to the relationships between the biological entities of interest.
Indeed, mapping between gene and protein ID scopes should not be done
the same way than between two scopes regarding gene ID.
Also, converting identifiers from different organisms should be possible
using gene orthologs information.
</p>

<ul>
<li> <p><a href="../doc/BED.html">Vignette</a>
</p>
</li>
<li><p> Available database instance: <a href="https://github.com/patzaw/BED#bed-database-instance-available-as-a-docker-image">https://github.com/patzaw/BED#bed-database-instance-available-as-a-docker-image</a>
</p>
</li>
<li><p> Building a database instance: <a href="https://github.com/patzaw/BED#build-a-bed-database-instance">https://github.com/patzaw/BED#build-a-bed-database-instance</a>
</p>
</li>
<li><p> Repository: <a href="https://github.com/patzaw/BED">https://github.com/patzaw/BED</a>
</p>
</li>
<li><p> Bug reports: <a href="https://github.com/patzaw/BED/issues">https://github.com/patzaw/BED/issues</a>
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Patrice Godard
</p>

<hr>
<h2 id='bedCall'>Call a function on the BED graph</h2><span id='topic+bedCall'></span>

<h3>Description</h3>

<p>Call a function on the BED graph
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bedCall(f, ..., bedCheck = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bedCall_+3A_f">f</code></td>
<td>
<p>the function to call</p>
</td></tr>
<tr><td><code id="bedCall_+3A_...">...</code></td>
<td>
<p>params for f</p>
</td></tr>
<tr><td><code id="bedCall_+3A_bedcheck">bedCheck</code></td>
<td>
<p>check if a connection to BED exists (default: FALSE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output of the called function.
</p>


<h3>See Also</h3>

<p><a href="#topic+checkBedConn">checkBedConn</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result &lt;- bedCall(
   cypher,
   query=prepCql(
      'MATCH (n:BEID)',
      'WHERE n.value IN $values',
      'RETURN n.value AS value, n.labels, n.database'
   ),
   parameters=list(values=c("10", "100"))
)

## End(Not run)

</code></pre>

<hr>
<h2 id='bedImport'>Feeding BED: Imports a data.frame in the BED graph database</h2><span id='topic+bedImport'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bedImport(cql, toImport, periodicCommit = 10000, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bedImport_+3A_cql">cql</code></td>
<td>
<p>the CQL query to be applied on each row of toImport</p>
</td></tr>
<tr><td><code id="bedImport_+3A_toimport">toImport</code></td>
<td>
<p>the data.frame to be imported as &quot;row&quot;.
Use &quot;row.FIELD&quot; in the cql query to refer to one FIELD of the toImport
data.frame</p>
</td></tr>
<tr><td><code id="bedImport_+3A_periodiccommit">periodicCommit</code></td>
<td>
<p>use periodic commit when loading the data
(default: 1000).</p>
</td></tr>
<tr><td><code id="bedImport_+3A_...">...</code></td>
<td>
<p>additional parameters for <a href="#topic+bedCall">bedCall</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the results of the query
</p>


<h3>See Also</h3>

<p><a href="#topic+bedCall">bedCall</a>, <a href="neo2R.html#topic+import_from_df">neo2R::import_from_df</a>
</p>

<hr>
<h2 id='BEIDList'>Create a BEIDList</h2><span id='topic+BEIDList'></span>

<h3>Description</h3>

<p>Create a BEIDList
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BEIDList(l, metadata, scope)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BEIDList_+3A_l">l</code></td>
<td>
<p>a named list of BEID vectors</p>
</td></tr>
<tr><td><code id="BEIDList_+3A_metadata">metadata</code></td>
<td>
<p>a data.frame with rownames or
a column &quot;<strong>.lname</strong>&quot; all in names of l.
If missing, the metadata is constructed with <strong>.lname</strong> being the names
of l.</p>
</td></tr>
<tr><td><code id="BEIDList_+3A_scope">scope</code></td>
<td>
<p>a list with 3 character vectors of length one named &quot;be&quot;,
&quot;source&quot; and &quot;organism&quot;.
If missing, it is guessed from l.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A BEIDList object which is a list of BEID vectors with 2 additional
attributes:
</p>

<ul>
<li> <p><strong>metadata</strong>: a data.frame with metadata about list elements.
The &quot;<strong>.lname</strong>&quot; column correspond to the names of the BEIDList.
</p>
</li>
<li> <p><strong>scope</strong>: the BEID scope (&quot;be&quot;, &quot;source&quot; and &quot;organism&quot;)
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
bel &lt;- BEIDList(
   l=list(
      kinases=c("117283", "3706", "3707", "51447", "80271", "9807"),
      phosphatases=c(
         "130367", "249", "283871", "493911", "57026", "5723", "81537"
      )
   ),
   scope=list(be="Gene", source="EntrezGene", organism="Homo sapiens")
)
scope(bel)
metadata(bel)
metadata(bel) &lt;- dplyr::mutate(
   metadata(bel),
   "description"=c("A few kinases", "A few phosphatases")
)
metadata(bel)

## End(Not run)

</code></pre>

<hr>
<h2 id='BEIDs'>Get the BEIDs from an object</h2><span id='topic+BEIDs'></span>

<h3>Description</h3>

<p>Get the BEIDs from an object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BEIDs(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BEIDs_+3A_x">x</code></td>
<td>
<p>an object representing a collection of BEID (e.g. BEIDList)</p>
</td></tr>
<tr><td><code id="BEIDs_+3A_...">...</code></td>
<td>
<p>method specific parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with at least 4 columns:
</p>

<ul>
<li><p> value
</p>
</li>
<li><p> be
</p>
</li>
<li><p> source
</p>
</li>
<li><p> organism
</p>
</li>
<li><p> ...
</p>
</li></ul>


<hr>
<h2 id='beidsServer'>Shiny module for searching BEIDs</h2><span id='topic+beidsServer'></span><span id='topic+beidsUI'></span>

<h3>Description</h3>

<p>Shiny module for searching BEIDs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>beidsServer(
  id,
  toGene = TRUE,
  excludeTechID = FALSE,
  multiple = FALSE,
  beOfInt = NULL,
  selectBe = TRUE,
  orgOfInt = NULL,
  selectOrg = TRUE,
  oneColumn = FALSE,
  withId = FALSE,
  maxHits = 75,
  compact = FALSE,
  tableHeight = 150,
  highlightStyle = "",
  highlightClass = "bed-search"
)

beidsUI(id)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="beidsServer_+3A_id">id</code></td>
<td>
<p>an identifier for the module instance</p>
</td></tr>
<tr><td><code id="beidsServer_+3A_togene">toGene</code></td>
<td>
<p>focus on gene entities (default=TRUE): matches from other
BE are converted to genes.</p>
</td></tr>
<tr><td><code id="beidsServer_+3A_excludetechid">excludeTechID</code></td>
<td>
<p>do not display BED technical BEIDs</p>
</td></tr>
<tr><td><code id="beidsServer_+3A_multiple">multiple</code></td>
<td>
<p>allow multiple selections (default=FALSE)</p>
</td></tr>
<tr><td><code id="beidsServer_+3A_beofint">beOfInt</code></td>
<td>
<p>if toGene==FALSE, BE to consider (default=NULL ==&gt; all)</p>
</td></tr>
<tr><td><code id="beidsServer_+3A_selectbe">selectBe</code></td>
<td>
<p>if toGene==FALSE, display an interface for selecting BE</p>
</td></tr>
<tr><td><code id="beidsServer_+3A_orgofint">orgOfInt</code></td>
<td>
<p>organism to consider (default=NULL ==&gt; all)</p>
</td></tr>
<tr><td><code id="beidsServer_+3A_selectorg">selectOrg</code></td>
<td>
<p>display an interface for selecting organisms</p>
</td></tr>
<tr><td><code id="beidsServer_+3A_onecolumn">oneColumn</code></td>
<td>
<p>if TRUE the hits are displayed in only one column</p>
</td></tr>
<tr><td><code id="beidsServer_+3A_withid">withId</code></td>
<td>
<p>if FALSE and one column, the BEIDs are not shown</p>
</td></tr>
<tr><td><code id="beidsServer_+3A_maxhits">maxHits</code></td>
<td>
<p>maximum number of raw hits to return</p>
</td></tr>
<tr><td><code id="beidsServer_+3A_compact">compact</code></td>
<td>
<p>compact display (default: FALSE)</p>
</td></tr>
<tr><td><code id="beidsServer_+3A_tableheight">tableHeight</code></td>
<td>
<p>height of the result table (default: 150)</p>
</td></tr>
<tr><td><code id="beidsServer_+3A_highlightstyle">highlightStyle</code></td>
<td>
<p>style to apply to the text to highlight</p>
</td></tr>
<tr><td><code id="beidsServer_+3A_highlightclass">highlightClass</code></td>
<td>
<p>class to apply to the text to highlight</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A reactive data.frame with the following columns:
</p>

<ul>
<li> <p><strong>beid</strong>: the BE identifier
</p>
</li>
<li> <p><strong>preferred</strong>: preferred identifier for the same BE in the same scope
</p>
</li>
<li> <p><strong>be</strong>: the type of biological entity
</p>
</li>
<li> <p><strong>source</strong>: the source of the identifier
</p>
</li>
<li> <p><strong>organism</strong>: the BE organism
</p>
</li>
<li> <p><strong>entity</strong>:  internal identifier of the BE
</p>
</li>
<li> <p><strong>match</strong>: the matching character string
</p>
</li></ul>



<h3>Functions</h3>


<ul>
<li> <p><code>beidsUI()</code>: 
</p>
</li></ul>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(shiny)
library(BED)
library(DT)

ui &lt;- fluidPage(
   beidsUI("be"),
   fluidRow(
      column(
         12,
         tags$br(),
         h3("Selected gene entities"),
         DTOutput("result")
      )
   )
)

server &lt;- function(input, output){
   found &lt;- beidsServer("be", toGene=TRUE, multiple=TRUE, tableHeight=250)
   output$result &lt;- renderDT({
      req(found())
      toRet &lt;- found()
      datatable(toRet, rownames=FALSE)
   })
}

shinyApp(ui = ui, server = server)

## End(Not run)

</code></pre>

<hr>
<h2 id='beIDsToAllScopes'>Find all BEID and ProbeID corresponding to a BE</h2><span id='topic+beIDsToAllScopes'></span>

<h3>Description</h3>

<p>Find all BEID and ProbeID corresponding to a BE
</p>


<h3>Usage</h3>

<pre><code class='language-R'>beIDsToAllScopes(
  beids,
  be,
  source,
  organism,
  entities = NULL,
  canonical_symbols = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="beIDsToAllScopes_+3A_beids">beids</code></td>
<td>
<p>a character vector of gene identifiers</p>
</td></tr>
<tr><td><code id="beIDsToAllScopes_+3A_be">be</code></td>
<td>
<p>one BE. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="beIDsToAllScopes_+3A_source">source</code></td>
<td>
<p>the source of gene identifiers. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="beIDsToAllScopes_+3A_organism">organism</code></td>
<td>
<p>the gene organism. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="beIDsToAllScopes_+3A_entities">entities</code></td>
<td>
<p>a numeric vector of gene entity. If NULL (default),
beids, source and organism arguments are used to identify BEs.
Be carefull when using entities as these identifiers are not stable.</p>
</td></tr>
<tr><td><code id="beIDsToAllScopes_+3A_canonical_symbols">canonical_symbols</code></td>
<td>
<p>return only canonical symbols (default: TRUE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with the following fields:
</p>

<ul>
<li> <p><strong>value</strong>: the identifier
</p>
</li>
<li> <p><strong>be</strong>: the type of BE
</p>
</li>
<li> <p><strong>source</strong>: the source of the identifier
</p>
</li>
<li> <p><strong>organism</strong>: the BE organism
</p>
</li>
<li> <p><strong>symbol</strong>: canonical symbol of the identifier
</p>
</li>
<li> <p><strong>BE_entity</strong>: the BE entity input
</p>
</li>
<li> <p><strong>BEID</strong> (optional): the BE ID input
</p>
</li>
<li> <p><strong>BE_source</strong> (optional): the BE source input
</p>
</li></ul>


<hr>
<h2 id='cacheBedCall'>Cached neo4j call</h2><span id='topic+cacheBedCall'></span>

<h3>Description</h3>

<p>This function calls neo4j DB the first time a query is sent and puts
the result in the cache SQLite database. The next time the same query is
called, it loads the results directly from cache SQLite database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cacheBedCall(..., tn, recache = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cacheBedCall_+3A_...">...</code></td>
<td>
<p>params for <a href="#topic+bedCall">bedCall</a></p>
</td></tr>
<tr><td><code id="cacheBedCall_+3A_tn">tn</code></td>
<td>
<p>the name of the cached table</p>
</td></tr>
<tr><td><code id="cacheBedCall_+3A_recache">recache</code></td>
<td>
<p>boolean indicating if the CQL query should be run even if
the table is already in cache</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Use only with &quot;row&quot; result returned by DB request.
</p>
<p>Internal use.
</p>


<h3>Value</h3>

<p>The results of the <a href="#topic+bedCall">bedCall</a>.
</p>


<h3>See Also</h3>

<p><a href="#topic+cacheBedResult">cacheBedResult</a>, <a href="#topic+bedCall">bedCall</a>
</p>

<hr>
<h2 id='cacheBedResult'>Put a BED query result in cache</h2><span id='topic+cacheBedResult'></span>

<h3>Description</h3>

<p>Internal use
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cacheBedResult(value, name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cacheBedResult_+3A_value">value</code></td>
<td>
<p>the result to cache</p>
</td></tr>
<tr><td><code id="cacheBedResult_+3A_name">name</code></td>
<td>
<p>the name of the query</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><a href="#topic+cacheBedCall">cacheBedCall</a>, <a href="#topic+loadBedResult">loadBedResult</a>
</p>

<hr>
<h2 id='checkBedCache'>Check BED cache</h2><span id='topic+checkBedCache'></span>

<h3>Description</h3>

<p>This function checks information recorded into BED cache
and resets it if not relevant.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkBedCache(newCon = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkBedCache_+3A_newcon">newCon</code></td>
<td>
<p>if TRUE for the loading of the system information file</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internal use.
</p>


<h3>See Also</h3>

<p><a href="#topic+clearBedCache">clearBedCache</a>, <a href="#topic+lsBedCache">lsBedCache</a>
</p>

<hr>
<h2 id='checkBedConn'>Check if there is a connection to a BED database</h2><span id='topic+checkBedConn'></span>

<h3>Description</h3>

<p>Check if there is a connection to a BED database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkBedConn(verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkBedConn_+3A_verbose">verbose</code></td>
<td>
<p>if TRUE print information about the BED connection
(default: FALSE).</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> TRUE if the connection can be established
</p>
</li>
<li><p> Or FALSE if the connection cannot be established or the &quot;System&quot; node
does not exist or does not have &quot;BED&quot; as name or any version recorded.
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="#topic+connectToBed">connectToBed</a>
</p>

<hr>
<h2 id='checkBeIds'>Check biological entities (BE) identifiers</h2><span id='topic+checkBeIds'></span>

<h3>Description</h3>

<p>This function takes a vector of identifiers and verify if they can
be found in the provided source database according to the BE type and
the organism of interest. If an ID is in the DB but not linked directly nor
indirectly to any entity then it is considered as not found.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkBeIds(ids, be, source, organism, stopThr = 1, caseSensitive = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkBeIds_+3A_ids">ids</code></td>
<td>
<p>a vector of identifiers to be checked</p>
</td></tr>
<tr><td><code id="checkBeIds_+3A_be">be</code></td>
<td>
<p>biological entity. See <a href="#topic+getBeIds">getBeIds</a>. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="checkBeIds_+3A_source">source</code></td>
<td>
<p>source of the ids. See <a href="#topic+getBeIds">getBeIds</a>. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="checkBeIds_+3A_organism">organism</code></td>
<td>
<p>the organism of interest. See <a href="#topic+getBeIds">getBeIds</a>.
<strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="checkBeIds_+3A_stopthr">stopThr</code></td>
<td>
<p>proportion of non-recognized IDs above which an error is
thrown. Default: 1 ==&gt; no check</p>
</td></tr>
<tr><td><code id="checkBeIds_+3A_casesensitive">caseSensitive</code></td>
<td>
<p>if FALSE (default) the case is not taken into account
when checking ids.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>invisible(TRUE). Stop if too many (see stopThr parameter) ids are not
found. Warning if any id is not found.
</p>


<h3>See Also</h3>

<p><a href="#topic+getBeIds">getBeIds</a>, <a href="#topic+listBeIdSources">listBeIdSources</a>, <a href="#topic+getAllBeIdSources">getAllBeIdSources</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
checkBeIds(
   ids=c("10", "100"), be="Gene", source="EntrezGene", organism="human"
)
checkBeIds(
   ids=c("10", "100"), be="Gene", source="Ens_gene", organism="human"
)

## End(Not run)

</code></pre>

<hr>
<h2 id='cleanDubiousXRef'>Identify and remove dubious cross-references</h2><span id='topic+cleanDubiousXRef'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cleanDubiousXRef(d, strict = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cleanDubiousXRef_+3A_d">d</code></td>
<td>
<p>a cross-reference data.frame with 2 columns.</p>
</td></tr>
<tr><td><code id="cleanDubiousXRef_+3A_strict">strict</code></td>
<td>
<p>if TRUE (default), the function returns only unambiguous
mappings</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns d without dubious cross-references.
Issues are reported in attr(d, &quot;issues&quot;).
</p>

<hr>
<h2 id='clearBedCache'>Clear the BED cache SQLite database</h2><span id='topic+clearBedCache'></span>

<h3>Description</h3>

<p>Clear the BED cache SQLite database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clearBedCache(queries = NULL, force = FALSE, hard = FALSE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clearBedCache_+3A_queries">queries</code></td>
<td>
<p>a character vector of the names of queries to remove.
If NULL all queries are removed.</p>
</td></tr>
<tr><td><code id="clearBedCache_+3A_force">force</code></td>
<td>
<p>if TRUE clear the BED cache table even if cache
file is not found</p>
</td></tr>
<tr><td><code id="clearBedCache_+3A_hard">hard</code></td>
<td>
<p>if TRUE remove everything in cache without checking file names</p>
</td></tr>
<tr><td><code id="clearBedCache_+3A_verbose">verbose</code></td>
<td>
<p>display some information during the process</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><a href="#topic+lsBedCache">lsBedCache</a>
</p>

<hr>
<h2 id='compareBedInstances'>Compare 2 BED database instances</h2><span id='topic+compareBedInstances'></span>

<h3>Description</h3>

<p>Compare 2 BED database instances
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compareBedInstances(connections)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compareBedInstances_+3A_connections">connections</code></td>
<td>
<p>a numeric vector of length 1 or 2 providing connections
from <a href="#topic+lsBedConnections">lsBedConnections</a> to be compared.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The current connection is restored when exiting this function.
</p>


<h3>Value</h3>

<p>If only one connection is provided, the function returns a list
with information about BEID and platforms available for the connection
along with DB version information.
If two connections are provided the same information as above is provided
for the 2 connection named V1 and V2 in that order. In addition,
differences observed between the 2 instances are reported for BEID and
platforms.
</p>

<hr>
<h2 id='connectToBed'>Connect to a neo4j BED database</h2><span id='topic+connectToBed'></span>

<h3>Description</h3>

<p>Connect to a neo4j BED database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>connectToBed(
  url = NULL,
  username = NULL,
  password = NULL,
  connection = 1,
  remember = FALSE,
  useCache = NA,
  importPath = NULL,
  .opts = list()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="connectToBed_+3A_url">url</code></td>
<td>
<p>a character string. The host and the port are sufficient
(e.g: &quot;localhost:5454&quot;)</p>
</td></tr>
<tr><td><code id="connectToBed_+3A_username">username</code></td>
<td>
<p>a character string</p>
</td></tr>
<tr><td><code id="connectToBed_+3A_password">password</code></td>
<td>
<p>a character string</p>
</td></tr>
<tr><td><code id="connectToBed_+3A_connection">connection</code></td>
<td>
<p>the id of the connection already registered to use. By
default the first registered connection is used.</p>
</td></tr>
<tr><td><code id="connectToBed_+3A_remember">remember</code></td>
<td>
<p>if TRUE connection information is saved localy in a file
and used to automatically connect the next time.
The default is set to FALSE.
All the connections that have been saved can be listed
with <a href="#topic+lsBedConnections">lsBedConnections</a> and any of
them can be forgotten with <a href="#topic+forgetBedConnection">forgetBedConnection</a>.</p>
</td></tr>
<tr><td><code id="connectToBed_+3A_usecache">useCache</code></td>
<td>
<p>if TRUE the results of large queries can be saved locally
in a file. The default is FALSE for policy reasons.
But it is recommended to set it to TRUE to improve the speed
of recurrent queries.
If NA (default parameter) the value is taken from former connection if
it exists or it is set to FALSE.</p>
</td></tr>
<tr><td><code id="connectToBed_+3A_importpath">importPath</code></td>
<td>
<p>the path to the import folder for loading information
in BED (used only when feeding the database ==&gt; default: NULL)</p>
</td></tr>
<tr><td><code id="connectToBed_+3A_.opts">.opts</code></td>
<td>
<p>a named list identifying the curl
options for the handle (see <code><a href="neo2R.html#topic+startGraph">neo2R::startGraph()</a></code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Be careful that you should reconnect to BED database each time
the environment is reloaded. It is done automatically if <code>remember</code> is
set to TRUE.
</p>
<p>Information about how to get an instance of the BED 'Neo4j' database is
provided here:
</p>

<ul>
<li> <p><a href="https://github.com/patzaw/BED#bed-database-instance-available-as-a-docker-image">https://github.com/patzaw/BED#bed-database-instance-available-as-a-docker-image</a>
</p>
</li>
<li> <p><a href="https://github.com/patzaw/BED#build-a-bed-database-instance">https://github.com/patzaw/BED#build-a-bed-database-instance</a>
</p>
</li></ul>



<h3>Value</h3>

<p>This function does not return any value. It prepares the BED
environment to allow transparent DB calls.
</p>


<h3>See Also</h3>

<p><a href="#topic+checkBedConn">checkBedConn</a>, <a href="#topic+lsBedConnections">lsBedConnections</a>, <a href="#topic+forgetBedConnection">forgetBedConnection</a>
</p>

<hr>
<h2 id='convBeIdLists'>Converts lists of BE IDs</h2><span id='topic+convBeIdLists'></span>

<h3>Description</h3>

<p>Converts lists of BE IDs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convBeIdLists(idList, entity = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convBeIdLists_+3A_idlist">idList</code></td>
<td>
<p>a list of IDs lists</p>
</td></tr>
<tr><td><code id="convBeIdLists_+3A_entity">entity</code></td>
<td>
<p>if TRUE returns BE instead of BEID (default: FALSE).
BE CAREFUL, THIS INTERNAL ID IS NOT STABLE AND CANNOT BE USED AS A REFERENCE.
This internal identifier is useful to avoid biases related to identifier
redundancy. See &lt;../doc/BED.html#3_managing_identifiers&gt;</p>
</td></tr>
<tr><td><code id="convBeIdLists_+3A_...">...</code></td>
<td>
<p>params for the <a href="#topic+convBeIds">convBeIds</a> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of <a href="#topic+convBeIds">convBeIds</a> ouput ids.
Scope (&quot;be&quot;, &quot;source&quot; &quot;organism&quot; and &quot;entity&quot; (see Arguments))
is provided as a named list
in the &quot;scope&quot; attributes: <code>attr(x, "scope")</code>
</p>


<h3>See Also</h3>

<p><a href="#topic+convBeIds">convBeIds</a>, <a href="#topic+convDfBeIds">convDfBeIds</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
convBeIdLists(
   idList=list(a=c("10", "100"), b=c("1000")),
   from="Gene",
   from.source="EntrezGene",
   from.org="human",
   to.source="Ens_gene"
)

## End(Not run)

</code></pre>

<hr>
<h2 id='convBeIds'>Converts BE IDs</h2><span id='topic+convBeIds'></span>

<h3>Description</h3>

<p>Converts BE IDs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convBeIds(
  ids,
  from,
  from.source,
  from.org,
  to,
  to.source,
  to.org,
  caseSensitive = FALSE,
  canonical = FALSE,
  prefFilter = FALSE,
  restricted = TRUE,
  recache = FALSE,
  limForCache = 2000
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convBeIds_+3A_ids">ids</code></td>
<td>
<p>list of identifiers</p>
</td></tr>
<tr><td><code id="convBeIds_+3A_from">from</code></td>
<td>
<p>a character corresponding to the biological entity or Probe.
<strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="convBeIds_+3A_from.source">from.source</code></td>
<td>
<p>a character corresponding to the ID source.
<strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="convBeIds_+3A_from.org">from.org</code></td>
<td>
<p>a character corresponding to the organism.
<strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="convBeIds_+3A_to">to</code></td>
<td>
<p>a character corresponding to the biological entity or Probe</p>
</td></tr>
<tr><td><code id="convBeIds_+3A_to.source">to.source</code></td>
<td>
<p>a character corresponding to the ID source</p>
</td></tr>
<tr><td><code id="convBeIds_+3A_to.org">to.org</code></td>
<td>
<p>a character corresponding to the organism</p>
</td></tr>
<tr><td><code id="convBeIds_+3A_casesensitive">caseSensitive</code></td>
<td>
<p>if TRUE the case of provided symbols
is taken into account
during search. This option will only affect the conversion from &quot;Symbol&quot;
(default: caseSensitive=FALSE).
All the other conversion will be case sensitive.</p>
</td></tr>
<tr><td><code id="convBeIds_+3A_canonical">canonical</code></td>
<td>
<p>if TRUE, only returns the canonical &quot;Symbol&quot;.
(default: FALSE)</p>
</td></tr>
<tr><td><code id="convBeIds_+3A_preffilter">prefFilter</code></td>
<td>
<p>boolean indicating if the results should be filter
to keep only preferred BEID of BE when they exist (default: FALSE).
If there are several
preferred BEID of a BE, all are kept. If there are no preferred BEID
of a BE, all non-preferred BEID are kept.</p>
</td></tr>
<tr><td><code id="convBeIds_+3A_restricted">restricted</code></td>
<td>
<p>boolean indicating if the results should be restricted to
current version of to BEID db. If FALSE former BEID are also returned:
<strong>Depending on history it can take a very long time to return</strong>
<strong>a very large result!</strong></p>
</td></tr>
<tr><td><code id="convBeIds_+3A_recache">recache</code></td>
<td>
<p>a logical value indicating if the results should be taken from
cache or recomputed</p>
</td></tr>
<tr><td><code id="convBeIds_+3A_limforcache">limForCache</code></td>
<td>
<p>if there are more ids than limForCache. Results are
collected for all IDs (beyond provided ids) and cached for futur queries.
If not, results are collected only for provided ids and not cached.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame with the following columns:
</p>

<ul>
<li> <p><strong>from</strong>: the input IDs
</p>
</li>
<li> <p><strong>to</strong>: the corresponding IDs in <code>to.source</code>
</p>
</li>
<li> <p><strong>to.preferred</strong>: boolean indicating if the to ID is a preferred
ID for the corresponding entity.
</p>
</li>
<li> <p><strong>to.entity</strong>: the entity technical ID of the <code>to</code> IDs
</p>
</li></ul>

<p>This data.frame can be filtered in order to remove duplicated
from/to.entity associations which can lead information bias.
Scope (&quot;be&quot;, &quot;source&quot; and &quot;organism&quot;) is provided as a named list
in the &quot;scope&quot; attributes: <code>attr(x, "scope")</code>
</p>


<h3>See Also</h3>

<p><a href="#topic+getBeIdConvTable">getBeIdConvTable</a>, <a href="#topic+convBeIdLists">convBeIdLists</a>, <a href="#topic+convDfBeIds">convDfBeIds</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
oriId &lt;- c("10", "100")
convBeIds(
   ids=oriId,
   from="Gene",
   from.source="EntrezGene",
   from.org="human",
   to.source="Ens_gene"
)
convBeIds(
   ids=oriId,
   from="Gene",
   from.source="EntrezGene",
   from.org="human",
   to="Peptide",
   to.source="Ens_translation"
)
convBeIds(
   ids=oriId,
   from="Gene",
   from.source="EntrezGene",
   from.org="human",
   to="Peptide",
   to.source="Ens_translation",
   to.org="mouse"
)

## End(Not run)

</code></pre>

<hr>
<h2 id='convDfBeIds'>Add BE ID conversion to a data frame</h2><span id='topic+convDfBeIds'></span>

<h3>Description</h3>

<p>Add BE ID conversion to a data frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convDfBeIds(df, idCol = NULL, entity = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convDfBeIds_+3A_df">df</code></td>
<td>
<p>the data.frame to be converted</p>
</td></tr>
<tr><td><code id="convDfBeIds_+3A_idcol">idCol</code></td>
<td>
<p>the column in which ID to convert are. If NULL (default)
the row names are taken.</p>
</td></tr>
<tr><td><code id="convDfBeIds_+3A_entity">entity</code></td>
<td>
<p>if TRUE returns BE instead of BEID (default: FALSE).
BE CAREFUL, THIS INTERNAL ID IS NOT STABLE AND CANNOT BE USED AS A REFERENCE.
This internal identifier is useful to avoid biases related to identifier
redundancy. See <a href="../doc/BED.html#3_managing_identifiers">../doc/BED.html#3_managing_identifiers</a></p>
</td></tr>
<tr><td><code id="convDfBeIds_+3A_...">...</code></td>
<td>
<p>params for the <a href="#topic+convBeIds">convBeIds</a> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with converted IDs.
Scope (&quot;be&quot;, &quot;source&quot;, &quot;organism&quot; and &quot;entity&quot; (see Arguments))
is provided as a named list
in the &quot;scope&quot; attributes: <code>attr(x, "scope")</code>.
</p>


<h3>See Also</h3>

<p><a href="#topic+convBeIds">convBeIds</a>, <a href="#topic+convBeIdLists">convBeIdLists</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
toConv &lt;- data.frame(a=1:2, b=3:4)
rownames(toConv) &lt;- c("10", "100")
convDfBeIds(
   df=toConv,
   from="Gene",
   from.source="EntrezGene",
   from.org="human",
   to.source="Ens_gene"
)

## End(Not run)

</code></pre>

<hr>
<h2 id='dumpEnsCore'>Feeding BED: Dump table from the Ensembl core database</h2><span id='topic+dumpEnsCore'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dumpEnsCore(
  organism,
  release,
  gv,
  ddir,
  toDump = c("attrib_type", "gene_attrib", "transcript", "external_db", "gene",
    "translation", "external_synonym", "object_xref", "xref", "stable_id_event"),
  env = parent.frame(n = 1)
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dumpEnsCore_+3A_organism">organism</code></td>
<td>
<p>the organism to download (e.g. &quot;Homo sapiens&quot;).</p>
</td></tr>
<tr><td><code id="dumpEnsCore_+3A_release">release</code></td>
<td>
<p>Ensembl release (e.g. &quot;83&quot;)</p>
</td></tr>
<tr><td><code id="dumpEnsCore_+3A_gv">gv</code></td>
<td>
<p>version of the genome (e.g. &quot;38&quot;)</p>
</td></tr>
<tr><td><code id="dumpEnsCore_+3A_ddir">ddir</code></td>
<td>
<p>path to the directory where the data should be saved</p>
</td></tr>
<tr><td><code id="dumpEnsCore_+3A_todump">toDump</code></td>
<td>
<p>the list of tables to download</p>
</td></tr>
<tr><td><code id="dumpEnsCore_+3A_env">env</code></td>
<td>
<p>the R environment in which to load the tables when downloaded</p>
</td></tr>
</table>

<hr>
<h2 id='dumpNcbiDb'>Feeding BED: Dump tables from the NCBI gene DATA</h2><span id='topic+dumpNcbiDb'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dumpNcbiDb(
  taxOfInt,
  reDumpThr,
  ddir,
  toLoad = c("gene_info", "gene2ensembl", "gene_group", "gene_orthologs", "gene_history",
    "gene2refseq"),
  env = parent.frame(n = 1),
  curDate
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dumpNcbiDb_+3A_taxofint">taxOfInt</code></td>
<td>
<p>the organism to download (e.g. &quot;9606&quot;).</p>
</td></tr>
<tr><td><code id="dumpNcbiDb_+3A_redumpthr">reDumpThr</code></td>
<td>
<p>time difference threshold between 2 downloads</p>
</td></tr>
<tr><td><code id="dumpNcbiDb_+3A_ddir">ddir</code></td>
<td>
<p>path to the directory where the data should be saved</p>
</td></tr>
<tr><td><code id="dumpNcbiDb_+3A_toload">toLoad</code></td>
<td>
<p>the list of tables to load</p>
</td></tr>
<tr><td><code id="dumpNcbiDb_+3A_env">env</code></td>
<td>
<p>the R environment in which to load the tables when downloaded</p>
</td></tr>
<tr><td><code id="dumpNcbiDb_+3A_curdate">curDate</code></td>
<td>
<p>current date as given by <a href="base.html#topic+Sys.Date">Sys.Date</a></p>
</td></tr>
</table>

<hr>
<h2 id='dumpNcbiTax'>Feeding BED: Dump tables with taxonomic information from NCBI</h2><span id='topic+dumpNcbiTax'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dumpNcbiTax(
  reDumpThr,
  ddir,
  toDump = c("names.dmp"),
  env = parent.frame(n = 1),
  curDate
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dumpNcbiTax_+3A_redumpthr">reDumpThr</code></td>
<td>
<p>time difference threshold between 2 downloads</p>
</td></tr>
<tr><td><code id="dumpNcbiTax_+3A_ddir">ddir</code></td>
<td>
<p>path to the directory where the data should be saved</p>
</td></tr>
<tr><td><code id="dumpNcbiTax_+3A_todump">toDump</code></td>
<td>
<p>the list of tables to load</p>
</td></tr>
<tr><td><code id="dumpNcbiTax_+3A_env">env</code></td>
<td>
<p>the R environment in which to load the tables when downloaded</p>
</td></tr>
<tr><td><code id="dumpNcbiTax_+3A_curdate">curDate</code></td>
<td>
<p>current date as given by <a href="base.html#topic+Sys.Date">Sys.Date</a></p>
</td></tr>
</table>

<hr>
<h2 id='dumpUniprotDb'>Feeding BED: Dump and preprocess flat data files from Uniprot</h2><span id='topic+dumpUniprotDb'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dumpUniprotDb(
  taxOfInt,
  divOfInt,
  release,
  ddir,
  ftp = "ftp://ftp.expasy.org/databases/uniprot",
  env = parent.frame(n = 1)
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dumpUniprotDb_+3A_taxofint">taxOfInt</code></td>
<td>
<p>the organism of interest (e.g., &quot;9606&quot; for human,
&quot;10090&quot; for mouse or &quot;10116&quot; for rat)</p>
</td></tr>
<tr><td><code id="dumpUniprotDb_+3A_divofint">divOfInt</code></td>
<td>
<p>the taxonomic division to which the organism belong
(e.g., &quot;human&quot;, &quot;rodents&quot;, &quot;mammals&quot;, &quot;vertebrates&quot;)</p>
</td></tr>
<tr><td><code id="dumpUniprotDb_+3A_release">release</code></td>
<td>
<p>the release of interest (check if already downloaded)</p>
</td></tr>
<tr><td><code id="dumpUniprotDb_+3A_ddir">ddir</code></td>
<td>
<p>path to the directory where the data should be saved</p>
</td></tr>
<tr><td><code id="dumpUniprotDb_+3A_ftp">ftp</code></td>
<td>
<p>location of the ftp site</p>
</td></tr>
<tr><td><code id="dumpUniprotDb_+3A_env">env</code></td>
<td>
<p>the R environment in which to load the tables when built</p>
</td></tr>
</table>

<hr>
<h2 id='exploreBe'>Explore BE identifiers</h2><span id='topic+exploreBe'></span>

<h3>Description</h3>

<p>This function uses visNetwork to draw all the identifiers
corresponding to one BE (including ProbeID and BESymbol)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exploreBe(
  id,
  source,
  be,
  showBE = FALSE,
  showProbes = FALSE,
  showLegend = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="exploreBe_+3A_id">id</code></td>
<td>
<p>one ID for the BE</p>
</td></tr>
<tr><td><code id="exploreBe_+3A_source">source</code></td>
<td>
<p>the ID source database. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="exploreBe_+3A_be">be</code></td>
<td>
<p>the type of BE. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="exploreBe_+3A_showbe">showBE</code></td>
<td>
<p>boolean. If TRUE the Biological Entity corresponding to the
id is shown. If id is isolated (not mapped to any other ID or symbol)
BE is shown anyway.</p>
</td></tr>
<tr><td><code id="exploreBe_+3A_showprobes">showProbes</code></td>
<td>
<p>boolean. If TRUE, probes targeting any BEID are shown.</p>
</td></tr>
<tr><td><code id="exploreBe_+3A_showlegend">showLegend</code></td>
<td>
<p>boolean. If TRUE the legend is displayed.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
exploreBe("Gene", "100", "EntrezGene")

## End(Not run)

</code></pre>

<hr>
<h2 id='exploreConvPath'>Explore the shortest convertion path between two identifiers</h2><span id='topic+exploreConvPath'></span>

<h3>Description</h3>

<p>This function uses visNetwork to draw all the shortest convertion paths
between two identifiers (including ProbeID).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exploreConvPath(
  from.id,
  to.id,
  from,
  from.source,
  to,
  to.source,
  edgeDirection = FALSE,
  showLegend = TRUE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="exploreConvPath_+3A_from.id">from.id</code></td>
<td>
<p>the first identifier</p>
</td></tr>
<tr><td><code id="exploreConvPath_+3A_to.id">to.id</code></td>
<td>
<p>the second identifier</p>
</td></tr>
<tr><td><code id="exploreConvPath_+3A_from">from</code></td>
<td>
<p>the type of entity: <code>listBe()</code> or Probe.
<strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="exploreConvPath_+3A_from.source">from.source</code></td>
<td>
<p>the identifier source: database or platform.
<strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="exploreConvPath_+3A_to">to</code></td>
<td>
<p>the type of entity: <code>listBe()</code> or Probe.
<strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="exploreConvPath_+3A_to.source">to.source</code></td>
<td>
<p>the identifier source: database or platform.
<strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="exploreConvPath_+3A_edgedirection">edgeDirection</code></td>
<td>
<p>a logical value indicating if the direction of the
edges should be drawn.</p>
</td></tr>
<tr><td><code id="exploreConvPath_+3A_showlegend">showLegend</code></td>
<td>
<p>boolean. If TRUE the legend is displayed.</p>
</td></tr>
<tr><td><code id="exploreConvPath_+3A_verbose">verbose</code></td>
<td>
<p>if TRUE the cypher query is shown</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
exploreConvPath(
   from.id="ENST00000413465",
   from="Transcript", from.source="Ens_transcript",
   to.id="ENSMUST00000108658",
   to="Transcript", to.source="Ens_transcript"
)

## End(Not run)

</code></pre>

<hr>
<h2 id='filterByBEID'>Filter an object to keep only a set of BEIDs</h2><span id='topic+filterByBEID'></span>

<h3>Description</h3>

<p>Filter an object to keep only a set of BEIDs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterByBEID(x, toKeep, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filterByBEID_+3A_x">x</code></td>
<td>
<p>an object representing a collection of BEID (e.g. BEIDList)</p>
</td></tr>
<tr><td><code id="filterByBEID_+3A_tokeep">toKeep</code></td>
<td>
<p>a vector of elements to keep</p>
</td></tr>
<tr><td><code id="filterByBEID_+3A_...">...</code></td>
<td>
<p>method specific parameters</p>
</td></tr>
</table>

<hr>
<h2 id='findBe'>Find Biological Entity</h2><span id='topic+findBe'></span>

<h3>Description</h3>

<p>Find Biological Entity in BED based on their IDs, symbols and names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findBe(
  be = NULL,
  organism = NULL,
  ncharSymb = 4,
  ncharName = 8,
  restricted = TRUE,
  by = 20,
  exclude = c("BEDTech_gene", "BEDTech_transcript")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="findBe_+3A_be">be</code></td>
<td>
<p>optional. If provided the search is focused on provided BEs.</p>
</td></tr>
<tr><td><code id="findBe_+3A_organism">organism</code></td>
<td>
<p>optional.  If provided the search is focused on provided
organisms.</p>
</td></tr>
<tr><td><code id="findBe_+3A_ncharsymb">ncharSymb</code></td>
<td>
<p>The minimum number of characters in searched to consider
incomplete symbol matches.</p>
</td></tr>
<tr><td><code id="findBe_+3A_ncharname">ncharName</code></td>
<td>
<p>The minimum number of characters in searched to consider
incomplete name matches.</p>
</td></tr>
<tr><td><code id="findBe_+3A_restricted">restricted</code></td>
<td>
<p>boolean indicating if the results should be restricted to
current version of to BEID db. If FALSE former BEID are also returned:
<strong>Depending on history it can take a very long time to return</strong>
<strong>a very large result!</strong></p>
</td></tr>
<tr><td><code id="findBe_+3A_by">by</code></td>
<td>
<p>number of found items to be converted into relevant IDs.</p>
</td></tr>
<tr><td><code id="findBe_+3A_exclude">exclude</code></td>
<td>
<p>database to exclude from possible selection. Used to filter
out technical database names such as &quot;BEDTech_gene&quot; and &quot;BEDTech_transcript&quot;
used to manage orphan IDs (not linked to any gene based on information
taken from sources)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the following fields:
</p>

<ul>
<li> <p><strong>found</strong>: the element found in BED corresponding to the searched term
</p>
</li>
<li> <p><strong>be</strong>: the type of the element
</p>
</li>
<li> <p><strong>source</strong>: the source of the element
</p>
</li>
<li> <p><strong>organism</strong>: the related organism
</p>
</li>
<li> <p><strong>entity</strong>: the related entity internal ID
</p>
</li>
<li> <p><strong>ebe</strong>: the BE of the related entity
</p>
</li>
<li> <p><strong>canonical</strong>: if the symbol is canonical
</p>
</li>
<li> <p><strong>Relevant ID</strong>: the seeked element id
</p>
</li>
<li> <p><strong>Symbol</strong>: the symbol(s) of the corresponding gene(s)
</p>
</li>
<li> <p><strong>Name</strong>: the symbol(s) of the corresponding gene(s)
</p>
</li></ul>

<p>Scope (&quot;be&quot;, &quot;source&quot; and &quot;organism&quot;) is provided as a named list
in the &quot;scope&quot; attributes: 'attr(x, &quot;scope&quot;)&ldquo;
</p>

<hr>
<h2 id='findBeids'>Find Biological Entity identifiers</h2><span id='topic+findBeids'></span>

<h3>Description</h3>

<p>Find Biological Entity identifiers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findBeids(toGene = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="findBeids_+3A_togene">toGene</code></td>
<td>
<p>focus on gene entities (default=TRUE): matches from other
BE are converted to genes.</p>
</td></tr>
<tr><td><code id="findBeids_+3A_...">...</code></td>
<td>
<p>parameters for <a href="#topic+beidsServer">beidsServer</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL if not any result, or a data.frame with the selected
values and the following column:
</p>

<ul>
<li> <p><strong>value</strong>: the BE identifier
</p>
</li>
<li> <p><strong>preferred</strong>: preferred identifier for the same BE in the same scope
</p>
</li>
<li> <p><strong>be</strong>: the type of biological entity
</p>
</li>
<li> <p><strong>source</strong>: the source of the identifier
</p>
</li>
<li> <p><strong>organism</strong>: the organism of the BE
</p>
</li>
<li> <p><strong>canonical</strong> (if toGene==TRUE): canonical gene product? (if known)
</p>
</li>
<li> <p><strong>symbol</strong>: the symbol of the identifier (if any)
</p>
</li></ul>


<hr>
<h2 id='firstCommonUpstreamBe'>First common upstream BE</h2><span id='topic+firstCommonUpstreamBe'></span>

<h3>Description</h3>

<p>Returns the first common Biological Entity (BE) upstream a set of BE.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>firstCommonUpstreamBe(beList = listBe(), uniqueOrg = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="firstCommonUpstreamBe_+3A_belist">beList</code></td>
<td>
<p>a character vector containing BE</p>
</td></tr>
<tr><td><code id="firstCommonUpstreamBe_+3A_uniqueorg">uniqueOrg</code></td>
<td>
<p>a logical value indicating if as single organism is under
focus. If false &quot;Gene&quot; is returned.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is used to identified the level at which different BE should be
compared. Peptides and transcripts should be compared at the level of
transcripts whereas transcripts and objects should be compared at the
level of genes. BE from different organism should be compared at the level
of genes using homologs.
</p>


<h3>See Also</h3>

<p><a href="#topic+listBe">listBe</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
firstCommonUpstreamBe(c("Object", "Transcript"))
firstCommonUpstreamBe(c("Peptide", "Transcript"))
firstCommonUpstreamBe(c("Peptide", "Transcript"), uniqueOrg=FALSE)

## End(Not run)

</code></pre>

<hr>
<h2 id='focusOnScope'>Focus a BE related object on a specific identifier (BEID) scope</h2><span id='topic+focusOnScope'></span>

<h3>Description</h3>

<p>Focus a BE related object on a specific identifier (BEID) scope
</p>


<h3>Usage</h3>

<pre><code class='language-R'>focusOnScope(
  x,
  be,
  source,
  organism,
  scope,
  force,
  restricted,
  prefFilter,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="focusOnScope_+3A_x">x</code></td>
<td>
<p>an object representing a collection of BEID (e.g. BEIDList)</p>
</td></tr>
<tr><td><code id="focusOnScope_+3A_be">be</code></td>
<td>
<p>the type of biological entity to focus on.
Used if <code>is.null(scope)</code></p>
</td></tr>
<tr><td><code id="focusOnScope_+3A_source">source</code></td>
<td>
<p>the source of BEID to focus on.
Used if <code>is.null(scope)</code></p>
</td></tr>
<tr><td><code id="focusOnScope_+3A_organism">organism</code></td>
<td>
<p>the organism of BEID to focus on.
Used if <code>is.null(scope)</code></p>
</td></tr>
<tr><td><code id="focusOnScope_+3A_scope">scope</code></td>
<td>
<p>a list with the following element:
</p>

<ul>
<li> <p><strong>be</strong>
</p>
</li>
<li> <p><strong>source</strong>
</p>
</li>
<li> <p><strong>organism</strong>
</p>
</li></ul>
</td></tr>
<tr><td><code id="focusOnScope_+3A_force">force</code></td>
<td>
<p>if TRUE the conversion is done even between identical scopes
(default: FALSE)</p>
</td></tr>
<tr><td><code id="focusOnScope_+3A_restricted">restricted</code></td>
<td>
<p>if TRUE (default) the BEID are limited to current version
of the source</p>
</td></tr>
<tr><td><code id="focusOnScope_+3A_preffilter">prefFilter</code></td>
<td>
<p>if TRUE (default) the BEID are limited to prefered
identifiers when they exist</p>
</td></tr>
<tr><td><code id="focusOnScope_+3A_...">...</code></td>
<td>
<p>method specific parameters for BEID conversion</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Depends on the class of x
</p>

<hr>
<h2 id='focusOnScope.BEIDList'>Convert a BEIDList object in a specific identifier (BEID) scope</h2><span id='topic+focusOnScope.BEIDList'></span>

<h3>Description</h3>

<p>Convert a BEIDList object in a specific identifier (BEID) scope
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BEIDList'
focusOnScope(
  x,
  be = NULL,
  source = NULL,
  organism = NULL,
  scope = NULL,
  force = FALSE,
  restricted = TRUE,
  prefFilter = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="focusOnScope.BEIDList_+3A_x">x</code></td>
<td>
<p>the BEIDList to be converted</p>
</td></tr>
<tr><td><code id="focusOnScope.BEIDList_+3A_be">be</code></td>
<td>
<p>the type of biological entity to focus on.
If NULL (default), it's taken from <code>scope(x)</code>.
Used if <code>is.null(scope)</code></p>
</td></tr>
<tr><td><code id="focusOnScope.BEIDList_+3A_source">source</code></td>
<td>
<p>the source of BEID to focus on.
If NULL (default), it's taken from <code>scope(x)</code>.
Used if <code>is.null(scope)</code></p>
</td></tr>
<tr><td><code id="focusOnScope.BEIDList_+3A_organism">organism</code></td>
<td>
<p>the organism of BEID to focus on.
If NULL (default), it's taken from <code>scope(x)</code>.
Used if <code>is.null(scope)</code></p>
</td></tr>
<tr><td><code id="focusOnScope.BEIDList_+3A_scope">scope</code></td>
<td>
<p>a list with the following element:
</p>

<ul>
<li> <p><strong>be</strong>
</p>
</li>
<li> <p><strong>source</strong>
</p>
</li>
<li> <p><strong>organism</strong>
</p>
</li></ul>
</td></tr>
<tr><td><code id="focusOnScope.BEIDList_+3A_force">force</code></td>
<td>
<p>if TRUE the conversion is done even between identical scopes
(default: FALSE)</p>
</td></tr>
<tr><td><code id="focusOnScope.BEIDList_+3A_restricted">restricted</code></td>
<td>
<p>if TRUE (default) the BEID are limited to current version
of the source</p>
</td></tr>
<tr><td><code id="focusOnScope.BEIDList_+3A_preffilter">prefFilter</code></td>
<td>
<p>if TRUE (default) the BEID are limited to prefered
identifiers when they exist</p>
</td></tr>
<tr><td><code id="focusOnScope.BEIDList_+3A_...">...</code></td>
<td>
<p>additional parameters to the BEID conversion function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A BEIDList
</p>

<hr>
<h2 id='forgetBedConnection'>Forget a BED connection</h2><span id='topic+forgetBedConnection'></span>

<h3>Description</h3>

<p>Forget a BED connection
</p>


<h3>Usage</h3>

<pre><code class='language-R'>forgetBedConnection(connection, save = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="forgetBedConnection_+3A_connection">connection</code></td>
<td>
<p>the id of the connection to forget.</p>
</td></tr>
<tr><td><code id="forgetBedConnection_+3A_save">save</code></td>
<td>
<p>a logical. Should be set to TRUE to save the updated list of
connections in the file space (default to FALSE to comply with CRAN
policies).</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><a href="#topic+lsBedConnections">lsBedConnections</a>, <a href="#topic+checkBedConn">checkBedConn</a>, <a href="#topic+connectToBed">connectToBed</a>
</p>

<hr>
<h2 id='genBePath'>Construct CQL sub-query to map 2 biological entity</h2><span id='topic+genBePath'></span>

<h3>Description</h3>

<p>Internal use
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genBePath(from, to, onlyR = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genBePath_+3A_from">from</code></td>
<td>
<p>one biological entity (BE)</p>
</td></tr>
<tr><td><code id="genBePath_+3A_to">to</code></td>
<td>
<p>one biological entity (BE)</p>
</td></tr>
<tr><td><code id="genBePath_+3A_onlyr">onlyR</code></td>
<td>
<p>logical. If TRUE (default: FALSE) it returns only the names
of the relationships and not the cypher sub-query</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character value corresponding to the sub-query.
Or, if onlyR, a character vector with the names of the relationships.
</p>


<h3>See Also</h3>

<p><a href="#topic+genProbePath">genProbePath</a>, <a href="#topic+listBe">listBe</a>
</p>

<hr>
<h2 id='geneIDsToAllScopes'>Find all GeneID, ObjectID, TranscriptID, PeptideID and ProbeID corresponding to a Gene in any organism</h2><span id='topic+geneIDsToAllScopes'></span>

<h3>Description</h3>

<p>Find all GeneID, ObjectID, TranscriptID, PeptideID and ProbeID corresponding to a Gene in any organism
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geneIDsToAllScopes(
  geneids,
  source,
  organism,
  entities = NULL,
  orthologs = TRUE,
  canonical_symbols = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geneIDsToAllScopes_+3A_geneids">geneids</code></td>
<td>
<p>a character vector of gene identifiers</p>
</td></tr>
<tr><td><code id="geneIDsToAllScopes_+3A_source">source</code></td>
<td>
<p>the source of gene identifiers. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="geneIDsToAllScopes_+3A_organism">organism</code></td>
<td>
<p>the gene organism. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="geneIDsToAllScopes_+3A_entities">entities</code></td>
<td>
<p>a numeric vector of gene entity. If NULL (default),
geneids, source and organism arguments are used to identify genes.
Be carefull when using entities as these identifiers are not stable.</p>
</td></tr>
<tr><td><code id="geneIDsToAllScopes_+3A_orthologs">orthologs</code></td>
<td>
<p>return identifiers from orthologs</p>
</td></tr>
<tr><td><code id="geneIDsToAllScopes_+3A_canonical_symbols">canonical_symbols</code></td>
<td>
<p>return only canonical symbols (default: TRUE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with the following fields:
</p>

<ul>
<li> <p><strong>value</strong>: the identifier
</p>
</li>
<li> <p><strong>preferred</strong>: preferred identifier for the same BE in the same scope
</p>
</li>
<li> <p><strong>be</strong>: the type of BE
</p>
</li>
<li> <p><strong>organism</strong>: the BE organism
</p>
</li>
<li> <p><strong>source</strong>: the source of the identifier
</p>
</li>
<li> <p><strong>canonical</strong>: canonical gene product (logical)
</p>
</li>
<li> <p><strong>symbol</strong>: canonical symbol of the identifier
</p>
</li>
<li> <p><strong>Gene_entity</strong>: the gene entity input
</p>
</li>
<li> <p><strong>GeneID</strong> (optional): the gene ID input
</p>
</li>
<li> <p><strong>Gene_source</strong> (optional): the gene source input
</p>
</li>
<li> <p><strong>Gene_organism</strong> (optional): the gene organism input
</p>
</li></ul>


<hr>
<h2 id='genProbePath'>Identify the biological entity (BE) targeted by probes and
construct the CQL sub-query to map probes to the BE</h2><span id='topic+genProbePath'></span>

<h3>Description</h3>

<p>Internal use
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genProbePath(platform)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genProbePath_+3A_platform">platform</code></td>
<td>
<p>the platform of the probes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character value corresponding to the sub-query.
The <code>attr(,"be")</code> correspond to the BE targeted by probes
</p>


<h3>See Also</h3>

<p><a href="#topic+genBePath">genBePath</a>, <a href="#topic+listPlatforms">listPlatforms</a>
</p>

<hr>
<h2 id='getAllBeIdSources'>List all the source databases of BE identifiers whatever the BE type</h2><span id='topic+getAllBeIdSources'></span>

<h3>Description</h3>

<p>List all the source databases of BE identifiers whatever the BE type
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAllBeIdSources(recache = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getAllBeIdSources_+3A_recache">recache</code></td>
<td>
<p>boolean indicating if the CQL query should be run even if
the table is already in cache</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame indicating the BE related to the ID source (database).
</p>


<h3>See Also</h3>

<p><a href="#topic+listBeIdSources">listBeIdSources</a>, <a href="#topic+listPlatforms">listPlatforms</a>
</p>

<hr>
<h2 id='getBeIdConvTable'>Get a conversion table between biological entity (BE) identifiers</h2><span id='topic+getBeIdConvTable'></span>

<h3>Description</h3>

<p>Get a conversion table between biological entity (BE) identifiers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getBeIdConvTable(
  from,
  to = from,
  from.source,
  to.source,
  organism,
  caseSensitive = FALSE,
  canonical = FALSE,
  restricted = TRUE,
  entity = TRUE,
  verbose = FALSE,
  recache = FALSE,
  filter = NULL,
  limForCache = 100
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getBeIdConvTable_+3A_from">from</code></td>
<td>
<p>one BE or &quot;Probe&quot;</p>
</td></tr>
<tr><td><code id="getBeIdConvTable_+3A_to">to</code></td>
<td>
<p>one BE or &quot;Probe&quot;</p>
</td></tr>
<tr><td><code id="getBeIdConvTable_+3A_from.source">from.source</code></td>
<td>
<p>the from BE ID database if BE or
the from probe platform if Probe</p>
</td></tr>
<tr><td><code id="getBeIdConvTable_+3A_to.source">to.source</code></td>
<td>
<p>the to BE ID database if BE or
the to probe platform if Probe</p>
</td></tr>
<tr><td><code id="getBeIdConvTable_+3A_organism">organism</code></td>
<td>
<p>organism name</p>
</td></tr>
<tr><td><code id="getBeIdConvTable_+3A_casesensitive">caseSensitive</code></td>
<td>
<p>if TRUE the case of provided symbols
is taken into account
during the conversion and selection.
This option will only affect the conversion from &quot;Symbol&quot;
(default: caseSensitive=FALSE).
All the other conversion will be case sensitive.</p>
</td></tr>
<tr><td><code id="getBeIdConvTable_+3A_canonical">canonical</code></td>
<td>
<p>if TRUE, only returns the canonical &quot;Symbol&quot;.
(default: FALSE)</p>
</td></tr>
<tr><td><code id="getBeIdConvTable_+3A_restricted">restricted</code></td>
<td>
<p>boolean indicating if the results should be restricted to
current version of to BEID db. If FALSE former BEID are also returned:
<strong>Depending on history it can take a very long time to return
a very large result!</strong></p>
</td></tr>
<tr><td><code id="getBeIdConvTable_+3A_entity">entity</code></td>
<td>
<p>boolean indicating if the technical ID of to BE should be
returned</p>
</td></tr>
<tr><td><code id="getBeIdConvTable_+3A_verbose">verbose</code></td>
<td>
<p>boolean indicating if the CQL query should be displayed</p>
</td></tr>
<tr><td><code id="getBeIdConvTable_+3A_recache">recache</code></td>
<td>
<p>boolean indicating if the CQL query should be run even if
the table is already in cache</p>
</td></tr>
<tr><td><code id="getBeIdConvTable_+3A_filter">filter</code></td>
<td>
<p>character vector on which to filter from IDs.
If NULL (default), the result is not filtered:
all from IDs are taken into account.</p>
</td></tr>
<tr><td><code id="getBeIdConvTable_+3A_limforcache">limForCache</code></td>
<td>
<p>if there are more filter than limForCache results are
collected for all IDs (beyond provided ids) and cached for futur queries.
If not, results are collected only for provided ids and not cached.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame mapping BE IDs with the
following fields:
</p>

<ul>
<li> <p><strong>from</strong>: the from BE ID
</p>
</li>
<li> <p><strong>to</strong>: the to BE ID
</p>
</li>
<li> <p><strong>entity</strong>: (optional) the technical ID of to BE
</p>
</li>
<li> <p><strong>preferred</strong>: true if &quot;to&quot; is the preferred identifier for the entity
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="#topic+getHomTable">getHomTable</a>, <a href="#topic+listBe">listBe</a>, <a href="#topic+listPlatforms">listPlatforms</a>, <a href="#topic+listBeIdSources">listBeIdSources</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getBeIdConvTable(
    from="Gene", from.source="EntrezGene",
    to.source="Ens_gene",
    organism="human"
)

## End(Not run)

</code></pre>

<hr>
<h2 id='getBeIdDescription'>Get description of Biological Entity identifiers</h2><span id='topic+getBeIdDescription'></span>

<h3>Description</h3>

<p>This description can be used for annotating tables or graph based on BE IDs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getBeIdDescription(ids, be, source, organism, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getBeIdDescription_+3A_ids">ids</code></td>
<td>
<p>list of identifiers</p>
</td></tr>
<tr><td><code id="getBeIdDescription_+3A_be">be</code></td>
<td>
<p>one BE. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="getBeIdDescription_+3A_source">source</code></td>
<td>
<p>the BE ID database. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="getBeIdDescription_+3A_organism">organism</code></td>
<td>
<p>organism name. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="getBeIdDescription_+3A_...">...</code></td>
<td>
<p>further arguments
for <a href="#topic+getBeIdNames">getBeIdNames</a> and <a href="#topic+getBeIdSymbols">getBeIdSymbols</a> functions</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame providing for each BE IDs
(row.names are provided BE IDs):
</p>

<ul>
<li> <p><strong>id</strong>: the BE ID
</p>
</li>
<li> <p><strong>symbol</strong>: the BE symbol
</p>
</li>
<li> <p><strong>name</strong>: the corresponding name
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="#topic+getBeIdNames">getBeIdNames</a>, <a href="#topic+getBeIdSymbols">getBeIdSymbols</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getBeIdDescription(
   ids=c("10", "100"),
   be="Gene",
   source="EntrezGene",
   organism="human"
)

## End(Not run)

</code></pre>

<hr>
<h2 id='getBeIdNames'>Get names of Biological Entity identifiers</h2><span id='topic+getBeIdNames'></span>

<h3>Description</h3>

<p>Get names of Biological Entity identifiers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getBeIdNames(ids, be, source, organism, limForCache = 4000, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getBeIdNames_+3A_ids">ids</code></td>
<td>
<p>list of identifiers</p>
</td></tr>
<tr><td><code id="getBeIdNames_+3A_be">be</code></td>
<td>
<p>one BE. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="getBeIdNames_+3A_source">source</code></td>
<td>
<p>the BE ID database. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="getBeIdNames_+3A_organism">organism</code></td>
<td>
<p>organism name. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="getBeIdNames_+3A_limforcache">limForCache</code></td>
<td>
<p>if there are more ids than limForCache results are
collected for all IDs (beyond provided ids) and cached for futur queries.
If not, results are collected only for provided ids and not cached.</p>
</td></tr>
<tr><td><code id="getBeIdNames_+3A_...">...</code></td>
<td>
<p>params for the <a href="#topic+getBeIdNameTable">getBeIdNameTable</a> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame mapping BE IDs and names with the
following fields:
</p>

<ul>
<li> <p><strong>id</strong>: the BE ID
</p>
</li>
<li> <p><strong>name</strong>: the corresponding name
</p>
</li>
<li> <p><strong>canonical</strong>: true if the name is canonical for the direct BE ID
(often FALSE for backward compatibility)
</p>
</li>
<li> <p><strong>direct</strong>: true if the name is directly related to the BE ID
</p>
</li>
<li> <p><strong>preferred</strong>: true if the id is the preferred identifier for the BE
</p>
</li>
<li> <p><strong>entity</strong>: (optional) the technical ID of to BE
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="#topic+getBeIdNameTable">getBeIdNameTable</a>, <a href="#topic+getBeIdSymbols">getBeIdSymbols</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getBeIdNames(
   ids=c("10", "100"),
   be="Gene",
   source="EntrezGene",
   organism="human"
)

## End(Not run)

</code></pre>

<hr>
<h2 id='getBeIdNameTable'>Get a table of biological entity (BE) identifiers and names</h2><span id='topic+getBeIdNameTable'></span>

<h3>Description</h3>

<p>Get a table of biological entity (BE) identifiers and names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getBeIdNameTable(
  be,
  source,
  organism,
  restricted,
  entity = TRUE,
  verbose = FALSE,
  recache = FALSE,
  filter = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getBeIdNameTable_+3A_be">be</code></td>
<td>
<p>one BE</p>
</td></tr>
<tr><td><code id="getBeIdNameTable_+3A_source">source</code></td>
<td>
<p>the BE ID database</p>
</td></tr>
<tr><td><code id="getBeIdNameTable_+3A_organism">organism</code></td>
<td>
<p>organism name</p>
</td></tr>
<tr><td><code id="getBeIdNameTable_+3A_restricted">restricted</code></td>
<td>
<p>boolean indicating if the results should be restricted to
direct names</p>
</td></tr>
<tr><td><code id="getBeIdNameTable_+3A_entity">entity</code></td>
<td>
<p>boolean indicating if the technical ID of BE should be
returned</p>
</td></tr>
<tr><td><code id="getBeIdNameTable_+3A_verbose">verbose</code></td>
<td>
<p>boolean indicating if the CQL query should be displayed</p>
</td></tr>
<tr><td><code id="getBeIdNameTable_+3A_recache">recache</code></td>
<td>
<p>boolean indicating if the CQL query should be run even if
the table is already in cache</p>
</td></tr>
<tr><td><code id="getBeIdNameTable_+3A_filter">filter</code></td>
<td>
<p>character vector on which to filter id. If NULL (default),
the result is not filtered: all IDs are taken into account.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame with the
following fields:
</p>

<ul>
<li> <p><strong>id</strong>: the from BE ID
</p>
</li>
<li> <p><strong>name</strong>: the BE name
</p>
</li>
<li> <p><strong>direct</strong>: false if the symbol is not directly associated to the BE ID
</p>
</li>
<li> <p><strong>preferred</strong>: true if the id is the preferred identifier for the BE
</p>
</li>
<li> <p><strong>entity</strong>: (optional) the technical ID of to BE
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="#topic+getBeIdNames">getBeIdNames</a>, <a href="#topic+getBeIdSymbolTable">getBeIdSymbolTable</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getBeIdNameTable(
   be="Gene",
   source="EntrezGene",
   organism="human"
)

## End(Not run)

</code></pre>

<hr>
<h2 id='getBeIds'>Get biological entities identifiers</h2><span id='topic+getBeIds'></span>

<h3>Description</h3>

<p>Get biological entities identifiers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getBeIds(
  be = c(listBe(), "Probe"),
  source,
  organism = NA,
  restricted,
  entity = TRUE,
  attributes = NULL,
  verbose = FALSE,
  recache = FALSE,
  filter = NULL,
  caseSensitive = FALSE,
  limForCache = 100,
  bef = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getBeIds_+3A_be">be</code></td>
<td>
<p>one BE or &quot;Probe&quot;</p>
</td></tr>
<tr><td><code id="getBeIds_+3A_source">source</code></td>
<td>
<p>the BE ID database or &quot;Symbol&quot; if BE or
the probe platform if Probe</p>
</td></tr>
<tr><td><code id="getBeIds_+3A_organism">organism</code></td>
<td>
<p>organism name</p>
</td></tr>
<tr><td><code id="getBeIds_+3A_restricted">restricted</code></td>
<td>
<p>boolean indicating if the results should be restricted to
current version of to BEID db. If FALSE former BEID are also returned.</p>
</td></tr>
<tr><td><code id="getBeIds_+3A_entity">entity</code></td>
<td>
<p>boolean indicating if the technical ID of BE should be
returned</p>
</td></tr>
<tr><td><code id="getBeIds_+3A_attributes">attributes</code></td>
<td>
<p>a character vector listing attributes that should be
returned.</p>
</td></tr>
<tr><td><code id="getBeIds_+3A_verbose">verbose</code></td>
<td>
<p>boolean indicating if the CQL query should be displayed</p>
</td></tr>
<tr><td><code id="getBeIds_+3A_recache">recache</code></td>
<td>
<p>boolean indicating if the CQL query should be run even if
the table is already in cache</p>
</td></tr>
<tr><td><code id="getBeIds_+3A_filter">filter</code></td>
<td>
<p>character vector on which to filter id. If NULL (default),
the result is not filtered: all IDs are taken into account.</p>
</td></tr>
<tr><td><code id="getBeIds_+3A_casesensitive">caseSensitive</code></td>
<td>
<p>if TRUE the case of provided symbols
is taken into account.
This option will only affect &quot;Symbol&quot; source
(default: caseSensitive=FALSE).</p>
</td></tr>
<tr><td><code id="getBeIds_+3A_limforcache">limForCache</code></td>
<td>
<p>if there are more filter than limForCache results are
collected for all IDs (beyond provided ids) and cached for futur queries.
If not, results are collected only for provided ids and not cached.</p>
</td></tr>
<tr><td><code id="getBeIds_+3A_bef">bef</code></td>
<td>
<p>For internal use only</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame mapping BE IDs with the
following fields:
</p>

<ul>
<li> <p><strong>id</strong>: the BE ID
</p>
</li>
<li> <p><strong>preferred</strong>: true if the id is the preferred identifier for the BE
</p>
</li>
<li> <p><strong>BE</strong>: IF entity is TRUE the technical ID of BE
</p>
</li>
<li> <p><strong>db.version</strong>: IF be is not &quot;Probe&quot; and source not &quot;Symbol&quot;
the version of the DB
</p>
</li>
<li> <p><strong>db.deprecated</strong>: IF be is not &quot;Probe&quot; and source not &quot;Symbol&quot;
a value if the BE ID is deprecated or FALSE if it's not
</p>
</li>
<li> <p><strong>canonical</strong>: IF source is &quot;Symbol&quot; TRUE if the symbol is canonical
</p>
</li>
<li> <p><strong>organism</strong>: IF be is &quot;Probe&quot; the organism of the targeted BE
</p>
</li></ul>

<p>If attributes are part of the query, additional columns for each of them.
Scope (&quot;be&quot;, &quot;source&quot; and &quot;organism&quot;) is provided as a named list
in the &quot;scope&quot; attributes: <code>attr(x, "scope")</code>
</p>


<h3>See Also</h3>

<p><a href="#topic+listPlatforms">listPlatforms</a>, <a href="#topic+listBeIdSources">listBeIdSources</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
beids &lt;- getBeIds(be="Gene", source="EntrezGene", organism="human", restricted=TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='getBeIdSymbols'>Get symbols of Biological Entity identifiers</h2><span id='topic+getBeIdSymbols'></span>

<h3>Description</h3>

<p>Get symbols of Biological Entity identifiers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getBeIdSymbols(ids, be, source, organism, limForCache = 4000, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getBeIdSymbols_+3A_ids">ids</code></td>
<td>
<p>list of identifiers</p>
</td></tr>
<tr><td><code id="getBeIdSymbols_+3A_be">be</code></td>
<td>
<p>one BE. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="getBeIdSymbols_+3A_source">source</code></td>
<td>
<p>the BE ID database. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="getBeIdSymbols_+3A_organism">organism</code></td>
<td>
<p>organism name. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="getBeIdSymbols_+3A_limforcache">limForCache</code></td>
<td>
<p>if there are more ids than limForCache. Results are
collected for all IDs (beyond provided ids) and cached for futur queries.
If not, results are collected only for provided ids and not cached.</p>
</td></tr>
<tr><td><code id="getBeIdSymbols_+3A_...">...</code></td>
<td>
<p>params for the <a href="#topic+getBeIdSymbolTable">getBeIdSymbolTable</a> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame with the
following fields:
</p>

<ul>
<li> <p><strong>id</strong>: the from BE ID
</p>
</li>
<li> <p><strong>symbol</strong>: the BE symbol
</p>
</li>
<li> <p><strong>canonical</strong>: true if the symbol is canonical for the direct BE ID
</p>
</li>
<li> <p><strong>direct</strong>: false if the symbol is not directly associated to the BE ID
</p>
</li>
<li> <p><strong>preferred</strong>: true if the id is the preferred identifier for the BE
</p>
</li>
<li> <p><strong>entity</strong>: (optional) the technical ID of to BE
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="#topic+getBeIdSymbolTable">getBeIdSymbolTable</a>, <a href="#topic+getBeIdNames">getBeIdNames</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getBeIdSymbols(
   ids=c("10", "100"),
   be="Gene",
   source="EntrezGene",
   organism="human"
)

## End(Not run)

</code></pre>

<hr>
<h2 id='getBeIdSymbolTable'>Get a table of biological entity (BE) identifiers and symbols</h2><span id='topic+getBeIdSymbolTable'></span>

<h3>Description</h3>

<p>Get a table of biological entity (BE) identifiers and symbols
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getBeIdSymbolTable(
  be,
  source,
  organism,
  restricted,
  entity = TRUE,
  verbose = FALSE,
  recache = FALSE,
  filter = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getBeIdSymbolTable_+3A_be">be</code></td>
<td>
<p>one BE</p>
</td></tr>
<tr><td><code id="getBeIdSymbolTable_+3A_source">source</code></td>
<td>
<p>the BE ID database</p>
</td></tr>
<tr><td><code id="getBeIdSymbolTable_+3A_organism">organism</code></td>
<td>
<p>organism name</p>
</td></tr>
<tr><td><code id="getBeIdSymbolTable_+3A_restricted">restricted</code></td>
<td>
<p>boolean indicating if the results should be restricted to
direct symbols</p>
</td></tr>
<tr><td><code id="getBeIdSymbolTable_+3A_entity">entity</code></td>
<td>
<p>boolean indicating if the technical ID of BE should be
returned</p>
</td></tr>
<tr><td><code id="getBeIdSymbolTable_+3A_verbose">verbose</code></td>
<td>
<p>boolean indicating if the CQL query should be displayed</p>
</td></tr>
<tr><td><code id="getBeIdSymbolTable_+3A_recache">recache</code></td>
<td>
<p>boolean indicating if the CQL query should be run even if
the table is already in cache</p>
</td></tr>
<tr><td><code id="getBeIdSymbolTable_+3A_filter">filter</code></td>
<td>
<p>character vector on which to filter id. If NULL (default),
the result is not filtered: all IDs are taken into account.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame with the
following fields:
</p>

<ul>
<li> <p><strong>id</strong>: the from BE ID
</p>
</li>
<li> <p><strong>symbol</strong>: the BE symbol
</p>
</li>
<li> <p><strong>canonical</strong>: true if the symbol is canonical for the direct BE ID
</p>
</li>
<li> <p><strong>direct</strong>: false if the symbol is not directly associated to the BE ID
</p>
</li>
<li> <p><strong>preferred</strong>: true if the id is the preferred identifier for the BE
</p>
</li>
<li> <p><strong>entity</strong>: (optional) the technical ID of to BE
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="#topic+getBeIdSymbols">getBeIdSymbols</a>, <a href="#topic+getBeIdNameTable">getBeIdNameTable</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getBeIdSymbolTable(
   be="Gene",
   source="EntrezGene",
   organism="human"
)

## End(Not run)

</code></pre>

<hr>
<h2 id='getBeIdURL'>Get reference URLs for BE IDs</h2><span id='topic+getBeIdURL'></span>

<h3>Description</h3>

<p>Get reference URLs for BE IDs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getBeIdURL(ids, databases)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getBeIdURL_+3A_ids">ids</code></td>
<td>
<p>the BE ID</p>
</td></tr>
<tr><td><code id="getBeIdURL_+3A_databases">databases</code></td>
<td>
<p>the databases from which each ID has been taken
(if only one database is provided it is chosen for all ids)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of the same length than ids
corresponding to the relevant URLs.
NA is returned is there is no URL corresponding to the provided database.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getBeIdURL(c("100", "ENSG00000145335"), c("EntrezGene", "Ens_gene"))

## End(Not run)

</code></pre>

<hr>
<h2 id='getDirectOrigin'>Get the direct origin of BE identifiers</h2><span id='topic+getDirectOrigin'></span>

<h3>Description</h3>

<p>The origin is directly taken as provided by the original database.
This function does not return indirect relationships.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDirectOrigin(
  ids,
  sources = NULL,
  process = c("is_expressed_as", "is_translated_in", "codes_for")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getDirectOrigin_+3A_ids">ids</code></td>
<td>
<p>list of product identifiers</p>
</td></tr>
<tr><td><code id="getDirectOrigin_+3A_sources">sources</code></td>
<td>
<p>a character vector corresponding to the possible product ID
sources. If NULL (default), all sources are considered</p>
</td></tr>
<tr><td><code id="getDirectOrigin_+3A_process">process</code></td>
<td>
<p>the production process among:
&quot;is_expressed_as&quot;, &quot;is_translated_in&quot;, &quot;codes_for&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame with the following columns:
</p>

<ul>
<li> <p><strong>origin</strong>: the origin BE identifiers
</p>
</li>
<li> <p><strong>osource</strong>: the origin database
</p>
</li>
<li> <p><strong>product</strong>: the product BE identifiers
</p>
</li>
<li> <p><strong>psource</strong>: the production database
</p>
</li>
<li> <p><strong>canonical</strong>: whether the production process is canonical or not
</p>
</li></ul>

<p>The process is also returned as an attribute of the data.frame.
</p>


<h3>See Also</h3>

<p><a href="#topic+getDirectOrigin">getDirectOrigin</a>, <a href="#topic+convBeIds">convBeIds</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
oriId &lt;- c("XP_016868427", "NP_001308979")
res &lt;- getDirectOrigin(
   ids=oriId,
   source="RefSeq_peptide",
   process="is_translated_in"
)
attr(res, "process")

## End(Not run)

</code></pre>

<hr>
<h2 id='getDirectProduct'>Get the direct product of BE identifiers</h2><span id='topic+getDirectProduct'></span>

<h3>Description</h3>

<p>The product is directly taken as provided by the original database.
This function does not return indirect relationships.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDirectProduct(
  ids,
  sources = NULL,
  process = c("is_expressed_as", "is_translated_in", "codes_for"),
  canonical = NA
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getDirectProduct_+3A_ids">ids</code></td>
<td>
<p>list of origin identifiers</p>
</td></tr>
<tr><td><code id="getDirectProduct_+3A_sources">sources</code></td>
<td>
<p>a character vector corresponding to the possible origin ID
sources. If NULL (default), all sources are considered</p>
</td></tr>
<tr><td><code id="getDirectProduct_+3A_process">process</code></td>
<td>
<p>the production process among:
&quot;is_expressed_as&quot;, &quot;is_translated_in&quot;, &quot;codes_for&quot;.</p>
</td></tr>
<tr><td><code id="getDirectProduct_+3A_canonical">canonical</code></td>
<td>
<p>If TRUE returns only canonical production process.
If FALSE returns only non-canonical production processes.
If NA (default) canonical information is taken into account.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame with the following columns:
</p>

<ul>
<li> <p><strong>origin</strong>: the origin BE identifiers
</p>
</li>
<li> <p><strong>osource</strong>: the origin database
</p>
</li>
<li> <p><strong>product</strong>: the product BE identifiers
</p>
</li>
<li> <p><strong>psource</strong>: the production database
</p>
</li>
<li> <p><strong>canonical</strong>: whether the production process is canonical or not
</p>
</li></ul>

<p>The process is also returned as an attribute of the data.frame.
</p>


<h3>See Also</h3>

<p><a href="#topic+getDirectOrigin">getDirectOrigin</a>, <a href="#topic+convBeIds">convBeIds</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
oriId &lt;- c("10", "100")
res &lt;- getDirectProduct(
   ids=oriId,
   source="EntrezGene",
   process="is_expressed_as",
   canonical=NA
)
attr(res, "process")

## End(Not run)

</code></pre>

<hr>
<h2 id='getEnsemblGeneIds'>Feeding BED: Download Ensembl DB and load gene information in BED</h2><span id='topic+getEnsemblGeneIds'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getEnsemblGeneIds(organism, release, gv, ddir, dbCref, dbAss, canChromosomes)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getEnsemblGeneIds_+3A_organism">organism</code></td>
<td>
<p>character vector of 1 element corresponding to the organism
of interest (e.g. &quot;Homo sapiens&quot;)</p>
</td></tr>
<tr><td><code id="getEnsemblGeneIds_+3A_release">release</code></td>
<td>
<p>the Ensembl release of interest (e.g. &quot;83&quot;)</p>
</td></tr>
<tr><td><code id="getEnsemblGeneIds_+3A_gv">gv</code></td>
<td>
<p>the genome version (e.g. &quot;38&quot;)</p>
</td></tr>
<tr><td><code id="getEnsemblGeneIds_+3A_ddir">ddir</code></td>
<td>
<p>path to the directory where the data should be saved</p>
</td></tr>
<tr><td><code id="getEnsemblGeneIds_+3A_dbcref">dbCref</code></td>
<td>
<p>a named vector of characters providing cross-reference DB of
interest. These DB are also used to find indirect ID associations.</p>
</td></tr>
<tr><td><code id="getEnsemblGeneIds_+3A_dbass">dbAss</code></td>
<td>
<p>a named vector of characters providing associated DB of
interest. Unlike the DB in dbCref parameter, these DB are not used for
indirect ID associations: the IDs are only linked to Ensembl IDs.</p>
</td></tr>
<tr><td><code id="getEnsemblGeneIds_+3A_canchromosomes">canChromosomes</code></td>
<td>
<p>canonical chromosmomes to be considered as preferred
ID (e.g. c(1:22, &quot;X&quot;, &quot;Y&quot;, &quot;MT&quot;) for human)</p>
</td></tr>
</table>

<hr>
<h2 id='getEnsemblPeptideIds'>Feeding BED: Download Ensembl DB and load peptide information in BED</h2><span id='topic+getEnsemblPeptideIds'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getEnsemblPeptideIds(organism, release, gv, ddir, dbCref, canChromosomes)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getEnsemblPeptideIds_+3A_organism">organism</code></td>
<td>
<p>character vector of 1 element corresponding to the organism
of interest (e.g. &quot;Homo sapiens&quot;)</p>
</td></tr>
<tr><td><code id="getEnsemblPeptideIds_+3A_release">release</code></td>
<td>
<p>the Ensembl release of interest (e.g. &quot;83&quot;)</p>
</td></tr>
<tr><td><code id="getEnsemblPeptideIds_+3A_gv">gv</code></td>
<td>
<p>the genome version (e.g. &quot;38&quot;)</p>
</td></tr>
<tr><td><code id="getEnsemblPeptideIds_+3A_ddir">ddir</code></td>
<td>
<p>path to the directory where the data should be saved</p>
</td></tr>
<tr><td><code id="getEnsemblPeptideIds_+3A_dbcref">dbCref</code></td>
<td>
<p>a named vector of characters providing cross-reference DB of
interest. These DB are also used to find indirect ID associations.</p>
</td></tr>
<tr><td><code id="getEnsemblPeptideIds_+3A_canchromosomes">canChromosomes</code></td>
<td>
<p>canonical chromosmomes to be considered as preferred
ID (e.g. c(1:22, &quot;X&quot;, &quot;Y&quot;, &quot;MT&quot;) for human)</p>
</td></tr>
</table>

<hr>
<h2 id='getEnsemblTranscriptIds'>Feeding BED: Download Ensembl DB and load transcript information in BED</h2><span id='topic+getEnsemblTranscriptIds'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getEnsemblTranscriptIds(organism, release, gv, ddir, dbCref, canChromosomes)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getEnsemblTranscriptIds_+3A_organism">organism</code></td>
<td>
<p>character vector of 1 element corresponding to the organism
of interest (e.g. &quot;Homo sapiens&quot;)</p>
</td></tr>
<tr><td><code id="getEnsemblTranscriptIds_+3A_release">release</code></td>
<td>
<p>the Ensembl release of interest (e.g. &quot;83&quot;)</p>
</td></tr>
<tr><td><code id="getEnsemblTranscriptIds_+3A_gv">gv</code></td>
<td>
<p>the genome version (e.g. &quot;38&quot;)</p>
</td></tr>
<tr><td><code id="getEnsemblTranscriptIds_+3A_ddir">ddir</code></td>
<td>
<p>path to the directory where the data should be saved</p>
</td></tr>
<tr><td><code id="getEnsemblTranscriptIds_+3A_dbcref">dbCref</code></td>
<td>
<p>a named vector of characters providing cross-reference DB of
interest. These DB are also used to find indirect ID associations.</p>
</td></tr>
<tr><td><code id="getEnsemblTranscriptIds_+3A_canchromosomes">canChromosomes</code></td>
<td>
<p>canonical chromosmomes to be considered as preferred
ID (e.g. c(1:22, &quot;X&quot;, &quot;Y&quot;, &quot;MT&quot;) for human)</p>
</td></tr>
</table>

<hr>
<h2 id='getGeneDescription'>Get description of genes corresponding to Biological Entity identifiers</h2><span id='topic+getGeneDescription'></span>

<h3>Description</h3>

<p>This description can be used for annotating tables or graph based on BE IDs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGeneDescription(
  ids,
  be,
  source,
  organism,
  gsource = largestBeSource(be = "Gene", organism = organism, rel = "is_known_as",
    restricted = TRUE),
  limForCache = 2000
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getGeneDescription_+3A_ids">ids</code></td>
<td>
<p>list of identifiers</p>
</td></tr>
<tr><td><code id="getGeneDescription_+3A_be">be</code></td>
<td>
<p>one BE. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="getGeneDescription_+3A_source">source</code></td>
<td>
<p>the BE ID database. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="getGeneDescription_+3A_organism">organism</code></td>
<td>
<p>organism name. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="getGeneDescription_+3A_gsource">gsource</code></td>
<td>
<p>the source of the gene IDs to use. It's chosen automatically
by default.</p>
</td></tr>
<tr><td><code id="getGeneDescription_+3A_limforcache">limForCache</code></td>
<td>
<p>The number of ids above which the description
is gathered for all be IDs and cached for  futur queries.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame providing for each BE IDs
(row.names are provided BE IDs):
</p>

<ul>
<li> <p><strong>id</strong>: the BE ID
</p>
</li>
<li> <p><strong>gsource</strong>: the Gene ID the column name provides the source of the
used identifier
</p>
</li>
<li> <p><strong>symbol</strong>: the associated gene symbols
</p>
</li>
<li> <p><strong>name</strong>: the associated gene names
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="#topic+getBeIdDescription">getBeIdDescription</a>, <a href="#topic+getBeIdNames">getBeIdNames</a>, <a href="#topic+getBeIdSymbols">getBeIdSymbols</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getGeneDescription(
   ids=c("1438_at", "1552335_at"),
   be="Probe",
   source="GPL570",
   organism="human"
)

## End(Not run)

</code></pre>

<hr>
<h2 id='getHomTable'>Get gene homologs between 2 organisms</h2><span id='topic+getHomTable'></span>

<h3>Description</h3>

<p>Get gene homologs between 2 organisms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getHomTable(
  from.org,
  to.org,
  from.source = "Ens_gene",
  to.source = from.source,
  restricted = TRUE,
  verbose = FALSE,
  recache = FALSE,
  filter = NULL,
  limForCache = 100
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getHomTable_+3A_from.org">from.org</code></td>
<td>
<p>organism name</p>
</td></tr>
<tr><td><code id="getHomTable_+3A_to.org">to.org</code></td>
<td>
<p>organism name</p>
</td></tr>
<tr><td><code id="getHomTable_+3A_from.source">from.source</code></td>
<td>
<p>the from gene ID database</p>
</td></tr>
<tr><td><code id="getHomTable_+3A_to.source">to.source</code></td>
<td>
<p>the to gene ID database</p>
</td></tr>
<tr><td><code id="getHomTable_+3A_restricted">restricted</code></td>
<td>
<p>boolean indicating if the results should be restricted to
current version of to BEID db. If FALSE former BEID are also returned:
<strong>Depending on history it can take a very long time to return a very</strong>
<strong>large result!</strong></p>
</td></tr>
<tr><td><code id="getHomTable_+3A_verbose">verbose</code></td>
<td>
<p>boolean indicating if the CQL query should be displayed</p>
</td></tr>
<tr><td><code id="getHomTable_+3A_recache">recache</code></td>
<td>
<p>boolean indicating if the CQL query should be run even if
the table is already in cache</p>
</td></tr>
<tr><td><code id="getHomTable_+3A_filter">filter</code></td>
<td>
<p>character vector on which to filter from IDs.
If NULL (default), the result is not filtered:
all from IDs are taken into account.</p>
</td></tr>
<tr><td><code id="getHomTable_+3A_limforcache">limForCache</code></td>
<td>
<p>if there are more filter than limForCache results are
collected for all IDs (beyond provided ids) and cached for futur queries.
If not, results are collected only for provided ids and not cached.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame mapping gene IDs with the
following fields:
</p>

<ul>
<li> <p><strong>from</strong>: the from gene ID
</p>
</li>
<li> <p><strong>to</strong>: the to gene ID
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="#topic+getBeIdConvTable">getBeIdConvTable</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getHomTable(
   from.org="human",
   to.org="mouse"
)

## End(Not run)

</code></pre>

<hr>
<h2 id='getNcbiGeneTransPep'>Feeding BED: Download NCBI gene DATA and load gene, transcript and peptide
information in BED</h2><span id='topic+getNcbiGeneTransPep'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getNcbiGeneTransPep(organism, reDumpThr = 1e+05, ddir, curDate)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getNcbiGeneTransPep_+3A_organism">organism</code></td>
<td>
<p>character vector of 1 element corresponding to the organism
of interest (e.g. &quot;Homo sapiens&quot;)</p>
</td></tr>
<tr><td><code id="getNcbiGeneTransPep_+3A_redumpthr">reDumpThr</code></td>
<td>
<p>time difference threshold between 2 downloads</p>
</td></tr>
<tr><td><code id="getNcbiGeneTransPep_+3A_ddir">ddir</code></td>
<td>
<p>path to the directory where the data should be saved</p>
</td></tr>
<tr><td><code id="getNcbiGeneTransPep_+3A_curdate">curDate</code></td>
<td>
<p>current date as given by <a href="base.html#topic+Sys.Date">Sys.Date</a></p>
</td></tr>
</table>

<hr>
<h2 id='getOrgNames'>Get organism names from taxonomy IDs</h2><span id='topic+getOrgNames'></span>

<h3>Description</h3>

<p>Get organism names from taxonomy IDs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getOrgNames(taxID = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getOrgNames_+3A_taxid">taxID</code></td>
<td>
<p>a vector of taxonomy IDs. If NULL (default) the function lists
all taxonomy IDs available in the DB.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame mapping taxonomy IDs to organism names with the
following fields:
</p>

<ul>
<li> <p><strong>taxID</strong>: the taxonomy ID
</p>
</li>
<li> <p><strong>name</strong>: the organism name
</p>
</li>
<li> <p><strong>nameClass</strong>: the class of the name
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="#topic+getTaxId">getTaxId</a>, <a href="#topic+listOrganisms">listOrganisms</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getOrgNames(c("9606", "10090"))
getOrgNames("9606")

## End(Not run)

</code></pre>

<hr>
<h2 id='getRelevantIds'>Get relevant IDs for a formerly identified BE in a context of interest</h2><span id='topic+getRelevantIds'></span>

<h3>Description</h3>

<p><strong>DEPRECATED: use <a href="#topic+searchBeid">searchBeid</a> and <a href="#topic+geneIDsToAllScopes">geneIDsToAllScopes</a> instead.</strong>
This function is meant to be used with <a href="#topic+searchId">searchId</a> in order
to implement a dictonary of identifiers of interest. First
the <a href="#topic+searchId">searchId</a> function is used to search a term.
Then the <a href="#topic+getRelevantIds">getRelevantIds</a> function
is used to find the corresponding IDs in a context of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getRelevantIds(
  d,
  selected = 1,
  be = c(listBe(), "Probe"),
  source,
  organism,
  restricted = TRUE,
  simplify = TRUE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getRelevantIds_+3A_d">d</code></td>
<td>
<p>the data.frame returned by <a href="#topic+searchId">searchId</a>.</p>
</td></tr>
<tr><td><code id="getRelevantIds_+3A_selected">selected</code></td>
<td>
<p>the rows of interest in d</p>
</td></tr>
<tr><td><code id="getRelevantIds_+3A_be">be</code></td>
<td>
<p>the BE in the context of interest</p>
</td></tr>
<tr><td><code id="getRelevantIds_+3A_source">source</code></td>
<td>
<p>the source of the identifier in the context of interest</p>
</td></tr>
<tr><td><code id="getRelevantIds_+3A_organism">organism</code></td>
<td>
<p>the organism in the context of interest</p>
</td></tr>
<tr><td><code id="getRelevantIds_+3A_restricted">restricted</code></td>
<td>
<p>boolean indicating if the results should be restricted to
current version of to BEID db. If FALSE former BEID are also returned:
<strong>Depending on history it can take a very long time to return</strong>
<strong>a very large result!</strong></p>
</td></tr>
<tr><td><code id="getRelevantIds_+3A_simplify">simplify</code></td>
<td>
<p>if TRUE (default) duplicated IDs are removed from the output</p>
</td></tr>
<tr><td><code id="getRelevantIds_+3A_verbose">verbose</code></td>
<td>
<p>if TRUE, the CQL query is shown</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The d data.frame with a new column providing the relevant ID
in the context of interest and without the gene field.
Scope (&quot;be&quot;, &quot;source&quot; and &quot;organism&quot;) is provided as a named list
in the &quot;scope&quot; attributes: <code>attr(x, "scope")</code>
</p>


<h3>See Also</h3>

<p><a href="#topic+searchId">searchId</a>
</p>

<hr>
<h2 id='getTargetedBe'>Identify the biological entity (BE) targeted by probes</h2><span id='topic+getTargetedBe'></span>

<h3>Description</h3>

<p>Identify the biological entity (BE) targeted by probes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getTargetedBe(platform)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getTargetedBe_+3A_platform">platform</code></td>
<td>
<p>the platform of the probes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The BE targeted by the platform
</p>


<h3>See Also</h3>

<p><a href="#topic+listPlatforms">listPlatforms</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getTargetedBe("GPL570")

## End(Not run)

</code></pre>

<hr>
<h2 id='getTaxId'>Get taxonomy ID of an organism name</h2><span id='topic+getTaxId'></span>

<h3>Description</h3>

<p>Get taxonomy ID of an organism name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getTaxId(name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getTaxId_+3A_name">name</code></td>
<td>
<p>the name of the organism</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of taxonomy ID
</p>


<h3>See Also</h3>

<p><a href="#topic+getOrgNames">getOrgNames</a>, <a href="#topic+listOrganisms">listOrganisms</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getTaxId("human")

## End(Not run)

</code></pre>

<hr>
<h2 id='getUniprot'>Feeding BED: Download Uniprot information in BED</h2><span id='topic+getUniprot'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getUniprot(organism, taxDiv, release, ddir)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getUniprot_+3A_organism">organism</code></td>
<td>
<p>character vector of 1 element corresponding to the organism
of interest (e.g. &quot;Homo sapiens&quot;)</p>
</td></tr>
<tr><td><code id="getUniprot_+3A_taxdiv">taxDiv</code></td>
<td>
<p>the taxonomic division to which the organism belong
(e.g., &quot;human&quot;, &quot;rodents&quot;, &quot;mammals&quot;, &quot;vertebrates&quot;)</p>
</td></tr>
<tr><td><code id="getUniprot_+3A_release">release</code></td>
<td>
<p>the release of interest (check if already downloaded)</p>
</td></tr>
<tr><td><code id="getUniprot_+3A_ddir">ddir</code></td>
<td>
<p>path to the directory where the data should be saved</p>
</td></tr>
</table>

<hr>
<h2 id='guessIdScope'>Guess biological entity (BE), database source and organism of a vector
of identifiers.</h2><span id='topic+guessIdScope'></span><span id='topic+guessIdOrigin'></span>

<h3>Description</h3>

<p>Guess biological entity (BE), database source and organism of a vector
of identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>guessIdScope(ids, be, source, organism, tcLim = 100)

guessIdOrigin(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="guessIdScope_+3A_ids">ids</code></td>
<td>
<p>a character vector of identifiers</p>
</td></tr>
<tr><td><code id="guessIdScope_+3A_be">be</code></td>
<td>
<p>one BE or &quot;Probe&quot;. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="guessIdScope_+3A_source">source</code></td>
<td>
<p>the BE ID database or &quot;Symbol&quot; if BE or
the probe platform if Probe. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="guessIdScope_+3A_organism">organism</code></td>
<td>
<p>organism name. <strong>Guessed if not provided</strong></p>
</td></tr>
<tr><td><code id="guessIdScope_+3A_tclim">tcLim</code></td>
<td>
<p>number of identifiers to check to guess origin for the whole set.
Inf ==&gt; no limit.</p>
</td></tr>
<tr><td><code id="guessIdScope_+3A_...">...</code></td>
<td>
<p>params for <code>guessIdScope</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list (NULL if no match):
</p>

<ul>
<li> <p><strong>be</strong>: a character vector of length 1 providing the best BE guess
(NA if inconsistent with user input: be, source or organism)
</p>
</li>
<li> <p><strong>source</strong>: a character vector of length 1 providing the best source
guess (NA if inconsistent with user input: be, source or organism)
</p>
</li>
<li><p> *<em>organism</em>$: a character vector of length 1 providing the best organism
guess (NA if inconsistent with user input: be, source or organism)
</p>
</li></ul>

<p>The &quot;details&quot; attribute ('attr(x, &quot;details&quot;)&ldquo;) is a data frame providing
numbers supporting the guess
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>guessIdOrigin()</code>: Deprecated version of guessIdScope
</p>
</li></ul>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
guessIdScope(ids=c("10", "100"))

## End(Not run)

</code></pre>

<hr>
<h2 id='identicalScopes'>Check if two objects have the same BEID scope</h2><span id='topic+identicalScopes'></span>

<h3>Description</h3>

<p>Check if two objects have the same BEID scope
</p>


<h3>Usage</h3>

<pre><code class='language-R'>identicalScopes(x, y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="identicalScopes_+3A_x">x</code></td>
<td>
<p>the object to test</p>
</td></tr>
<tr><td><code id="identicalScopes_+3A_y">y</code></td>
<td>
<p>the object to test</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical indicating if the 2 scopes are identical
</p>

<hr>
<h2 id='is.BEIDList'>Check if the provided object is a <a href="#topic+BEIDList">BEIDList</a></h2><span id='topic+is.BEIDList'></span>

<h3>Description</h3>

<p>Check if the provided object is a <a href="#topic+BEIDList">BEIDList</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.BEIDList(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is.BEIDList_+3A_x">x</code></td>
<td>
<p>the object to check</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical value
</p>

<hr>
<h2 id='largestBeSource'>Autoselect source of biological entity identifiers</h2><span id='topic+largestBeSource'></span>

<h3>Description</h3>

<p>The selection is based on direct identifiers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>largestBeSource(
  be,
  organism,
  rel = NA,
  restricted = TRUE,
  exclude = c("BEDTech_gene", "BEDTech_transcript")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="largestBeSource_+3A_be">be</code></td>
<td>
<p>the biological entity under focus</p>
</td></tr>
<tr><td><code id="largestBeSource_+3A_organism">organism</code></td>
<td>
<p>the organism under focus</p>
</td></tr>
<tr><td><code id="largestBeSource_+3A_rel">rel</code></td>
<td>
<p>a type of relationship to consider in the query
(e.g. &quot;is_member_of&quot;) in order to focus on specific information.
If NA (default) all be are taken into account whatever their available
relationships.</p>
</td></tr>
<tr><td><code id="largestBeSource_+3A_restricted">restricted</code></td>
<td>
<p>boolean indicating if the results should be restricted to
current version of to BEID db. If FALSE former BEID are also taken
into account.</p>
</td></tr>
<tr><td><code id="largestBeSource_+3A_exclude">exclude</code></td>
<td>
<p>database to exclude from possible selection. Used to filter
out technical database names such as &quot;BEDTech_gene&quot; and &quot;BEDTech_transcript&quot;
used to manage orphan IDs (not linked to any gene based on information
taken from sources)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The name of the selected source. The selected source will be the one
providing the largest number of current identifiers.
</p>


<h3>See Also</h3>

<p><a href="#topic+listBeIdSources">listBeIdSources</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
largestBeSource(be="Gene", "Mus musculus")

## End(Not run)

</code></pre>

<hr>
<h2 id='listBe'>Lists all the biological entities (BE) available in the BED database</h2><span id='topic+listBe'></span>

<h3>Description</h3>

<p>Lists all the biological entities (BE) available in the BED database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>listBe()
</code></pre>


<h3>Value</h3>

<p>A character vector of biological entities (BE)
</p>


<h3>See Also</h3>

<p><a href="#topic+listPlatforms">listPlatforms</a>, <a href="#topic+listBeIdSources">listBeIdSources</a>,
<a href="#topic+listOrganisms">listOrganisms</a>
</p>

<hr>
<h2 id='listBeIdSources'>Lists all the databases taken into account in the BED database
for a biological entity (BE)</h2><span id='topic+listBeIdSources'></span>

<h3>Description</h3>

<p>Lists all the databases taken into account in the BED database
for a biological entity (BE)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>listBeIdSources(
  be = listBe(),
  organism,
  direct = FALSE,
  rel = NA,
  restricted = FALSE,
  recache = FALSE,
  verbose = FALSE,
  exclude = c()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="listBeIdSources_+3A_be">be</code></td>
<td>
<p>the BE on which to focus</p>
</td></tr>
<tr><td><code id="listBeIdSources_+3A_organism">organism</code></td>
<td>
<p>the name of the organism to focus on.</p>
</td></tr>
<tr><td><code id="listBeIdSources_+3A_direct">direct</code></td>
<td>
<p>a logical value indicating if only &quot;direct&quot; BE identifiers
should be considered</p>
</td></tr>
<tr><td><code id="listBeIdSources_+3A_rel">rel</code></td>
<td>
<p>a type of relationship to consider in the query
(e.g. &quot;is_member_of&quot;) in order to focus on specific information.
If NA (default) all be are taken into account whatever their available
relationships.</p>
</td></tr>
<tr><td><code id="listBeIdSources_+3A_restricted">restricted</code></td>
<td>
<p>boolean indicating if the results should be restricted to
current version of to BEID db. If FALSE former BEID are also returned.
There is no impact if direct is set to TRUE.</p>
</td></tr>
<tr><td><code id="listBeIdSources_+3A_recache">recache</code></td>
<td>
<p>boolean indicating if the CQL query should be run even if
the table is already in cache</p>
</td></tr>
<tr><td><code id="listBeIdSources_+3A_verbose">verbose</code></td>
<td>
<p>boolean indicating if the CQL query should be shown.</p>
</td></tr>
<tr><td><code id="listBeIdSources_+3A_exclude">exclude</code></td>
<td>
<p>database to exclude from possible selection. Used to filter
out technical database names such as &quot;BEDTech_gene&quot; and &quot;BEDTech_transcript&quot;
used to manage orphan IDs (not linked to any gene based on information
taken from sources)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame indicating the number of ID in each available database
with the following fields:
</p>

<ul>
<li> <p><strong>database</strong>: the database name
</p>
</li>
<li> <p><strong>nbBe</strong>: number of distinct entities
</p>
</li>
<li> <p><strong>nbId</strong>: number of identifiers
</p>
</li>
<li> <p><strong>be</strong>: the BE under focus
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="#topic+listBe">listBe</a>, <a href="#topic+largestBeSource">largestBeSource</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
listBeIdSources(be="Transcript", organism="mouse")

## End(Not run)

</code></pre>

<hr>
<h2 id='listDBAttributes'>List all attributes provided by a BEDB</h2><span id='topic+listDBAttributes'></span>

<h3>Description</h3>

<p>List all attributes provided by a BEDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>listDBAttributes(dbname)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="listDBAttributes_+3A_dbname">dbname</code></td>
<td>
<p>the name of the database</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of attribute names
</p>

<hr>
<h2 id='listOrganisms'>Lists all the organisms available in the BED database</h2><span id='topic+listOrganisms'></span>

<h3>Description</h3>

<p>Lists all the organisms available in the BED database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>listOrganisms()
</code></pre>


<h3>Value</h3>

<p>A character vector of organism scientific names
</p>


<h3>See Also</h3>

<p><a href="#topic+listPlatforms">listPlatforms</a>, <a href="#topic+listBeIdSources">listBeIdSources</a>,
<a href="#topic+listBe">listBe</a>, <a href="#topic+getTaxId">getTaxId</a>, <a href="#topic+getOrgNames">getOrgNames</a>
</p>

<hr>
<h2 id='listPlatforms'>Lists all the probe platforms available in the BED database</h2><span id='topic+listPlatforms'></span>

<h3>Description</h3>

<p>Lists all the probe platforms available in the BED database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>listPlatforms(be = c(NA, listBe()))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="listPlatforms_+3A_be">be</code></td>
<td>
<p>a character vector of BE on which to focus.
if NA (default) all the BE are considered.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame mapping platforms to BE with the following fields:
</p>

<ul>
<li> <p><strong>name</strong>: the platform nam
</p>
</li>
<li> <p><strong>description</strong>: platform description
</p>
</li>
<li> <p><strong>focus</strong>: Targeted BE
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="#topic+listBe">listBe</a>, <a href="#topic+listBeIdSources">listBeIdSources</a>,
<a href="#topic+listOrganisms">listOrganisms</a>, <a href="#topic+getTargetedBe">getTargetedBe</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
listPlatforms(be="Gene")
listPlatforms()

## End(Not run)

</code></pre>

<hr>
<h2 id='loadBE'>Feeding BED: Load biological entities in BED</h2><span id='topic+loadBE'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadBE(
  d,
  be = "Gene",
  dbname,
  version = NA,
  deprecated = NA,
  taxId = NA,
  onlyId = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadBE_+3A_d">d</code></td>
<td>
<p>a data.frame with information about the entities to be loaded.
It should contain the following fields: &quot;id&quot;.
If there is a boolean column named &quot;preferred&quot;, the value is loaded.</p>
</td></tr>
<tr><td><code id="loadBE_+3A_be">be</code></td>
<td>
<p>a character corresponding to the BE type (default: &quot;Gene&quot;)</p>
</td></tr>
<tr><td><code id="loadBE_+3A_dbname">dbname</code></td>
<td>
<p>the DB from which the BE ID are taken</p>
</td></tr>
<tr><td><code id="loadBE_+3A_version">version</code></td>
<td>
<p>the version of the DB from which the BE IDs are taken</p>
</td></tr>
<tr><td><code id="loadBE_+3A_deprecated">deprecated</code></td>
<td>
<p>NA (default) or the date when the ID was deprecated</p>
</td></tr>
<tr><td><code id="loadBE_+3A_taxid">taxId</code></td>
<td>
<p>the taxonomy ID of the BE organism</p>
</td></tr>
<tr><td><code id="loadBE_+3A_onlyid">onlyId</code></td>
<td>
<p>a logical. If TRUE, only an BEID is created and not the
corresponding BE.</p>
</td></tr>
</table>

<hr>
<h2 id='loadBeAttribute'>Feeding BED: Load attributes for biological entities in BED</h2><span id='topic+loadBeAttribute'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadBeAttribute(d, be = "Gene", dbname, attribute)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadBeAttribute_+3A_d">d</code></td>
<td>
<p>a data.frame providing for each BE ID (&quot;id&quot; column) an attribute
value (&quot;value&quot; column). There can be several values for each id.</p>
</td></tr>
<tr><td><code id="loadBeAttribute_+3A_be">be</code></td>
<td>
<p>a character corresponding to the BE type (default: &quot;Gene&quot;)</p>
</td></tr>
<tr><td><code id="loadBeAttribute_+3A_dbname">dbname</code></td>
<td>
<p>the DB from which the BE ID are taken</p>
</td></tr>
<tr><td><code id="loadBeAttribute_+3A_attribute">attribute</code></td>
<td>
<p>the name of the attribute to be loaded</p>
</td></tr>
</table>

<hr>
<h2 id='loadBedModel'>Feeding BED: Load BED data model in neo4j</h2><span id='topic+loadBedModel'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadBedModel()
</code></pre>

<hr>
<h2 id='loadBedOtherIndexes'>Feeding BED: Load additional indexes in neo4j</h2><span id='topic+loadBedOtherIndexes'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadBedOtherIndexes()
</code></pre>

<hr>
<h2 id='loadBedResult'>Get a BED query result from cache</h2><span id='topic+loadBedResult'></span>

<h3>Description</h3>

<p>Internal use
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadBedResult(name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadBedResult_+3A_name">name</code></td>
<td>
<p>the name of the query</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><a href="#topic+cacheBedCall">cacheBedCall</a>, <a href="#topic+cacheBedResult">cacheBedResult</a>
</p>

<hr>
<h2 id='loadBENames'>Feeding BED: Load names associated to BEIDs</h2><span id='topic+loadBENames'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadBENames(d, be = "Gene", dbname)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadBENames_+3A_d">d</code></td>
<td>
<p>a data.frame with information about the names
to be loaded. It should contain the following fields: &quot;id&quot;, &quot;name&quot;
and &quot;canonical&quot; (optional).</p>
</td></tr>
<tr><td><code id="loadBENames_+3A_be">be</code></td>
<td>
<p>a character corresponding to the BE type (default: &quot;Gene&quot;)</p>
</td></tr>
<tr><td><code id="loadBENames_+3A_dbname">dbname</code></td>
<td>
<p>the DB of BEID</p>
</td></tr>
</table>

<hr>
<h2 id='loadBESymbols'>Feeding BED: Load symbols associated to BEIDs</h2><span id='topic+loadBESymbols'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadBESymbols(d, be = "Gene", dbname)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadBESymbols_+3A_d">d</code></td>
<td>
<p>a data.frame with information about the symbols
to be loaded. It should contain the following fields: &quot;id&quot;, &quot;symbol&quot;
and &quot;canonical&quot; (optional).</p>
</td></tr>
<tr><td><code id="loadBESymbols_+3A_be">be</code></td>
<td>
<p>a character corresponding to the BE type (default: &quot;Gene&quot;)</p>
</td></tr>
<tr><td><code id="loadBESymbols_+3A_dbname">dbname</code></td>
<td>
<p>the DB of BEID</p>
</td></tr>
</table>

<hr>
<h2 id='loadBEVersion'>Feeding BED: Load biological entities in BED with information about
DB version</h2><span id='topic+loadBEVersion'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadBEVersion(d, be = "Gene", dbname, taxId = NA, onlyId = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadBEVersion_+3A_d">d</code></td>
<td>
<p>a data.frame with information about the entities to be loaded.
It should contain the following fields: &quot;id&quot;, &quot;version&quot; and &quot;deprecated&quot;.</p>
</td></tr>
<tr><td><code id="loadBEVersion_+3A_be">be</code></td>
<td>
<p>a character corresponding to the BE type (default: &quot;Gene&quot;)</p>
</td></tr>
<tr><td><code id="loadBEVersion_+3A_dbname">dbname</code></td>
<td>
<p>the DB from which the BE ID are taken</p>
</td></tr>
<tr><td><code id="loadBEVersion_+3A_taxid">taxId</code></td>
<td>
<p>the taxonomy ID of the BE organism</p>
</td></tr>
<tr><td><code id="loadBEVersion_+3A_onlyid">onlyId</code></td>
<td>
<p>a logical. If TRUE, only an BEID is created and not the
corresponding BE.</p>
</td></tr>
</table>

<hr>
<h2 id='loadCodesFor'>Feeding BED: Load correspondance between genes and objects as coding events</h2><span id='topic+loadCodesFor'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadCodesFor(d, gdb, odb)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadCodesFor_+3A_d">d</code></td>
<td>
<p>a data.frame with information about the coding events.
It should contain the following fields: &quot;gid&quot; and &quot;oid&quot;</p>
</td></tr>
<tr><td><code id="loadCodesFor_+3A_gdb">gdb</code></td>
<td>
<p>the DB of Gene IDs</p>
</td></tr>
<tr><td><code id="loadCodesFor_+3A_odb">odb</code></td>
<td>
<p>the DB of Object IDs</p>
</td></tr>
</table>

<hr>
<h2 id='loadCorrespondsTo'>Feeding BED: Load correspondances between BE IDs</h2><span id='topic+loadCorrespondsTo'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadCorrespondsTo(d, db1, db2, be = "Gene")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadCorrespondsTo_+3A_d">d</code></td>
<td>
<p>a data.frame with information about the correspondances
to be loaded. It should contain the following fields: &quot;id1&quot; and &quot;id2&quot;.</p>
</td></tr>
<tr><td><code id="loadCorrespondsTo_+3A_db1">db1</code></td>
<td>
<p>the DB of id1</p>
</td></tr>
<tr><td><code id="loadCorrespondsTo_+3A_db2">db2</code></td>
<td>
<p>the DB of id2</p>
</td></tr>
<tr><td><code id="loadCorrespondsTo_+3A_be">be</code></td>
<td>
<p>a character corresponding to the BE type (default: &quot;Gene&quot;)</p>
</td></tr>
</table>

<hr>
<h2 id='loadHistory'>Feeding BED: Load history of BEIDs</h2><span id='topic+loadHistory'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadHistory(d, dbname, be = "Gene")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadHistory_+3A_d">d</code></td>
<td>
<p>a data.frame with information about the history.
It should contain the following fields: &quot;old&quot; and &quot;new&quot;.</p>
</td></tr>
<tr><td><code id="loadHistory_+3A_dbname">dbname</code></td>
<td>
<p>the DB of BEID</p>
</td></tr>
<tr><td><code id="loadHistory_+3A_be">be</code></td>
<td>
<p>a character corresponding to the BE type (default: &quot;Gene&quot;)</p>
</td></tr>
</table>

<hr>
<h2 id='loadIsAssociatedTo'>Feeding BED: Load BE ID associations</h2><span id='topic+loadIsAssociatedTo'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadIsAssociatedTo(d, db1, db2, be = "Gene")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadIsAssociatedTo_+3A_d">d</code></td>
<td>
<p>a data.frame with information about the associations
to be loaded. It should contain the following fields: &quot;id1&quot; and &quot;id2&quot;.
At the end id1 is associated to id2 (this way and not the other).</p>
</td></tr>
<tr><td><code id="loadIsAssociatedTo_+3A_db1">db1</code></td>
<td>
<p>the DB of id1</p>
</td></tr>
<tr><td><code id="loadIsAssociatedTo_+3A_db2">db2</code></td>
<td>
<p>the DB of id2</p>
</td></tr>
<tr><td><code id="loadIsAssociatedTo_+3A_be">be</code></td>
<td>
<p>a character corresponding to the BE type (default: &quot;Gene&quot;)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When associating one id1 to id2, the BE identified by id1
is deleted after that its production edges have been transferred
to the BE identified by id2.
After this operation all id &quot;corresponding_to&quot; id1  do not
directly identify any BE as they are supposed to do. Thus,
to run this function with id1 involved in &quot;corresponds_to&quot; edges.
</p>

<hr>
<h2 id='loadIsExpressedAs'>Feeding BED: Load correspondance between genes and transcripts as
expression events</h2><span id='topic+loadIsExpressedAs'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadIsExpressedAs(d, gdb, tdb)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadIsExpressedAs_+3A_d">d</code></td>
<td>
<p>a data.frame with information about the expression events.
It should contain the following fields: &quot;gid&quot;, &quot;tid&quot;
and &quot;canonical&quot; (optional).</p>
</td></tr>
<tr><td><code id="loadIsExpressedAs_+3A_gdb">gdb</code></td>
<td>
<p>the DB of Gene IDs</p>
</td></tr>
<tr><td><code id="loadIsExpressedAs_+3A_tdb">tdb</code></td>
<td>
<p>the DB of Transcript IDs</p>
</td></tr>
</table>

<hr>
<h2 id='loadIsHomologOf'>Feeding BED: Load homology between BE IDs</h2><span id='topic+loadIsHomologOf'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadIsHomologOf(d, db1, db2, be = "Gene")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadIsHomologOf_+3A_d">d</code></td>
<td>
<p>a data.frame with information about the homologies
to be loaded. It should contain the following fields: &quot;id1&quot; and &quot;id2&quot;.</p>
</td></tr>
<tr><td><code id="loadIsHomologOf_+3A_db1">db1</code></td>
<td>
<p>the DB of id1</p>
</td></tr>
<tr><td><code id="loadIsHomologOf_+3A_db2">db2</code></td>
<td>
<p>the DB of id2</p>
</td></tr>
<tr><td><code id="loadIsHomologOf_+3A_be">be</code></td>
<td>
<p>a character corresponding to the BE type (default: &quot;Gene&quot;)</p>
</td></tr>
</table>

<hr>
<h2 id='loadIsTranslatedIn'>Feeding BED: Load correspondance between transcripts and peptides as
translation events</h2><span id='topic+loadIsTranslatedIn'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadIsTranslatedIn(d, tdb, pdb)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadIsTranslatedIn_+3A_d">d</code></td>
<td>
<p>a data.frame with information about the translation events.
It should contain the following fields: &quot;tid&quot;, &quot;pid&quot;
and &quot;canonical&quot; (optional).</p>
</td></tr>
<tr><td><code id="loadIsTranslatedIn_+3A_tdb">tdb</code></td>
<td>
<p>the DB of Transcript IDs</p>
</td></tr>
<tr><td><code id="loadIsTranslatedIn_+3A_pdb">pdb</code></td>
<td>
<p>the DB of Peptide IDs</p>
</td></tr>
</table>

<hr>
<h2 id='loadLuceneIndexes'>Feeding BED: Create Lucene indexes in neo4j</h2><span id='topic+loadLuceneIndexes'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadLuceneIndexes()
</code></pre>

<hr>
<h2 id='loadNCBIEntrezGOFunctions'>Feeding BED: Load in BED GO functions associated to Entrez gene IDs
from NCBI</h2><span id='topic+loadNCBIEntrezGOFunctions'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadNCBIEntrezGOFunctions(organism, reDumpThr = 1e+05, ddir, curDate)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadNCBIEntrezGOFunctions_+3A_organism">organism</code></td>
<td>
<p>character vector of 1 element corresponding to the organism
of interest (e.g. &quot;Homo sapiens&quot;)</p>
</td></tr>
<tr><td><code id="loadNCBIEntrezGOFunctions_+3A_redumpthr">reDumpThr</code></td>
<td>
<p>time difference threshold between 2 downloads</p>
</td></tr>
<tr><td><code id="loadNCBIEntrezGOFunctions_+3A_ddir">ddir</code></td>
<td>
<p>path to the directory where the data should be saved</p>
</td></tr>
<tr><td><code id="loadNCBIEntrezGOFunctions_+3A_curdate">curDate</code></td>
<td>
<p>current date as given by <a href="base.html#topic+Sys.Date">Sys.Date</a></p>
</td></tr>
</table>

<hr>
<h2 id='loadNcbiTax'>Feeding BED: Load taxonomic information from NCBI</h2><span id='topic+loadNcbiTax'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadNcbiTax(reDumpThr, ddir, orgOfInt = c("human", "rat", "mouse"), curDate)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadNcbiTax_+3A_redumpthr">reDumpThr</code></td>
<td>
<p>time difference threshold between 2 downloads</p>
</td></tr>
<tr><td><code id="loadNcbiTax_+3A_ddir">ddir</code></td>
<td>
<p>path to the directory where the data should be saved</p>
</td></tr>
<tr><td><code id="loadNcbiTax_+3A_orgofint">orgOfInt</code></td>
<td>
<p>organisms of interest: a character vector</p>
</td></tr>
<tr><td><code id="loadNcbiTax_+3A_curdate">curDate</code></td>
<td>
<p>current date as given by <a href="base.html#topic+Sys.Date">Sys.Date</a></p>
</td></tr>
</table>

<hr>
<h2 id='loadOrganisms'>Feeding BED: Load organisms in BED</h2><span id='topic+loadOrganisms'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadOrganisms(d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadOrganisms_+3A_d">d</code></td>
<td>
<p>a data.frame with 2 columns named &quot;tax_id&quot; and &quot;name_txt&quot; providing
the taxonomic ID for each organism name</p>
</td></tr>
</table>

<hr>
<h2 id='loadPlf'>Feeding BED: Load a probes platform</h2><span id='topic+loadPlf'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadPlf(name, description, be)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadPlf_+3A_name">name</code></td>
<td>
<p>the name of the platform</p>
</td></tr>
<tr><td><code id="loadPlf_+3A_description">description</code></td>
<td>
<p>a description of the platform</p>
</td></tr>
<tr><td><code id="loadPlf_+3A_be">be</code></td>
<td>
<p>the type of BE targeted by the platform</p>
</td></tr>
</table>

<hr>
<h2 id='loadProbes'>Feeding BED: Load probes targeting BE IDs</h2><span id='topic+loadProbes'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadProbes(d, be = "Transcript", platform, dbname)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadProbes_+3A_d">d</code></td>
<td>
<p>a data.frame with information about the entities to be loaded.
It should contain the following fields: &quot;id&quot; and &quot;probeID&quot;.</p>
</td></tr>
<tr><td><code id="loadProbes_+3A_be">be</code></td>
<td>
<p>a character corresponding to the BE
targeted by the probes (default: &quot;Transcript&quot;)</p>
</td></tr>
<tr><td><code id="loadProbes_+3A_platform">platform</code></td>
<td>
<p>the plateform gathering the probes</p>
</td></tr>
<tr><td><code id="loadProbes_+3A_dbname">dbname</code></td>
<td>
<p>the DB from which the BE ID are taken</p>
</td></tr>
</table>

<hr>
<h2 id='lsBedCache'>List all the BED queries in cache and the total size of the cache</h2><span id='topic+lsBedCache'></span>

<h3>Description</h3>

<p>List all the BED queries in cache and the total size of the cache
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lsBedCache(verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lsBedCache_+3A_verbose">verbose</code></td>
<td>
<p>if TRUE (default) prints a message displaying the total
size of the cache</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame giving for each query (row names) its size in Bytes
(column &quot;size&quot;) and in human readable format (column &quot;hr&quot;). The
attribute &quot;Total&quot; corresponds to the sum of all the file size.
</p>


<h3>See Also</h3>

<p><a href="#topic+clearBedCache">clearBedCache</a>
</p>

<hr>
<h2 id='lsBedConnections'>List all registered BED connection</h2><span id='topic+lsBedConnections'></span>

<h3>Description</h3>

<p>List all registered BED connection
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lsBedConnections()
</code></pre>


<h3>See Also</h3>

<p><a href="#topic+connectToBed">connectToBed</a>, <a href="#topic+forgetBedConnection">forgetBedConnection</a>, <a href="#topic+checkBedConn">checkBedConn</a>
</p>

<hr>
<h2 id='metadata'>Get object metadata</h2><span id='topic+metadata'></span>

<h3>Description</h3>

<p>Get object metadata
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metadata(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="metadata_+3A_x">x</code></td>
<td>
<p>an object representing a collection of BEID (e.g. BEIDList)</p>
</td></tr>
<tr><td><code id="metadata_+3A_...">...</code></td>
<td>
<p>method specific parameters</p>
</td></tr>
</table>

<hr>
<h2 id='metadata+26lt+3B-'>Set object metadata</h2><span id='topic+metadata+3C-'></span>

<h3>Description</h3>

<p>Set object metadata
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metadata(x) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="metadata+2B26lt+2B3B-_+3A_x">x</code></td>
<td>
<p>an object representing a collection of BEID (e.g. BEIDList)</p>
</td></tr>
<tr><td><code id="metadata+2B26lt+2B3B-_+3A_value">value</code></td>
<td>
<p>a data.frame with rownames or
a column &quot;<strong>.lname</strong>&quot; all in names of l.</p>
</td></tr>
</table>

<hr>
<h2 id='registerBEDB'>Feeding BED: Register a database of biological entities in BED DB</h2><span id='topic+registerBEDB'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>registerBEDB(name, description = NA, currentVersion = NA, idURL = NA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="registerBEDB_+3A_name">name</code></td>
<td>
<p>of the database (e.g. &quot;Ens_gene&quot;)</p>
</td></tr>
<tr><td><code id="registerBEDB_+3A_description">description</code></td>
<td>
<p>a short description of the database (e.g. &quot;Ensembl gene&quot;)</p>
</td></tr>
<tr><td><code id="registerBEDB_+3A_currentversion">currentVersion</code></td>
<td>
<p>the version taken into account in BED (e.g. 83)</p>
</td></tr>
<tr><td><code id="registerBEDB_+3A_idurl">idURL</code></td>
<td>
<p>the URL template to use to retrieve id information. A '%s'
corresponding to the ID should be present in this character vector of
length one.</p>
</td></tr>
</table>

<hr>
<h2 id='scope'>Get the BEID scope of an object</h2><span id='topic+scope'></span>

<h3>Description</h3>

<p>Get the BEID scope of an object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scope(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scope_+3A_x">x</code></td>
<td>
<p>an object representing a collection of BEID (e.g. BEIDList)</p>
</td></tr>
<tr><td><code id="scope_+3A_...">...</code></td>
<td>
<p>method specific parameters</p>
</td></tr>
</table>

<hr>
<h2 id='scopes'>Get the BEID scopes of an object</h2><span id='topic+scopes'></span>

<h3>Description</h3>

<p>Get the BEID scopes of an object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scopes(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scopes_+3A_x">x</code></td>
<td>
<p>an object representing a collection of BEID (e.g. BEIDList)</p>
</td></tr>
<tr><td><code id="scopes_+3A_...">...</code></td>
<td>
<p>method specific parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with 4 columns:
</p>

<ul>
<li><p> be
</p>
</li>
<li><p> source
</p>
</li>
<li><p> organism
</p>
</li>
<li><p> Freq
</p>
</li></ul>


<hr>
<h2 id='searchBeid'>Search a BEID</h2><span id='topic+searchBeid'></span>

<h3>Description</h3>

<p>Search a BEID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>searchBeid(x, maxHits = 75, clean_id_search = TRUE, clean_name_search = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="searchBeid_+3A_x">x</code></td>
<td>
<p>a character value to search</p>
</td></tr>
<tr><td><code id="searchBeid_+3A_maxhits">maxHits</code></td>
<td>
<p>maximum number of raw hits to return</p>
</td></tr>
<tr><td><code id="searchBeid_+3A_clean_id_search">clean_id_search</code></td>
<td>
<p>clean x to avoid error during ID search.
Default: TRUE. Set it to false if you're sure of your lucene query.</p>
</td></tr>
<tr><td><code id="searchBeid_+3A_clean_name_search">clean_name_search</code></td>
<td>
<p>clean x to avoid error during ID search.
Default: TRUE. Set it to false if you're sure of your lucene query.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL if there is not any match or
a data.frame with the following columns:
</p>

<ul>
<li> <p><strong>value</strong>: the matching term
</p>
</li>
<li> <p><strong>from</strong>: the type of the matched term (e.g. BESymbol, GeneID...)
</p>
</li>
<li> <p><strong>be</strong>: the matching biological entity (BE)
</p>
</li>
<li> <p><strong>beid</strong>: the BE identifier
</p>
</li>
<li> <p><strong>source</strong>: the BEID reference database
</p>
</li>
<li> <p><strong>preferred</strong>: TRUE if the BEID is considered as a preferred identifier
</p>
</li>
<li> <p><strong>symbol</strong>: BEID canonical symbol
</p>
</li>
<li> <p><strong>name</strong>: BEID name
</p>
</li>
<li> <p><strong>entity</strong>: technical BE identifier
</p>
</li>
<li> <p><strong>GeneID</strong>: Corresponding gene identifier
</p>
</li>
<li> <p><strong>Gene_source</strong>: Gene ID database
</p>
</li>
<li> <p><strong>preferred_gene</strong>: TRUE if the GeneID is considered as a preferred identifier
</p>
</li>
<li> <p><strong>Gene_symbol</strong>: Gene symbol
</p>
</li>
<li> <p><strong>Gene_name</strong>: Gene name
</p>
</li>
<li> <p><strong>Gene_entity</strong>: technical gene identifier
</p>
</li>
<li> <p><strong>organism</strong>: gene organism (scientific name)
</p>
</li>
<li> <p><strong>score</strong>: score of the fuzzy search
</p>
</li>
<li> <p><strong>included</strong>: is the search term fully included in the value
</p>
</li>
<li> <p><strong>exact</strong>: is the value an exact match of the term
</p>
</li></ul>


<hr>
<h2 id='searchId'>Search identifier, symbol or name information</h2><span id='topic+searchId'></span>

<h3>Description</h3>

<p><strong>DEPRECATED: use <a href="#topic+searchBeid">searchBeid</a> and <a href="#topic+geneIDsToAllScopes">geneIDsToAllScopes</a> instead.</strong>
This function is meant to be used with <a href="#topic+getRelevantIds">getRelevantIds</a> in order
to implement a dictonary of identifiers of interest. First
the <a href="#topic+searchId">searchId</a> function is used to search a term.
Then the <a href="#topic+getRelevantIds">getRelevantIds</a> function
is used to find the corresponding ID in a context of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>searchId(
  searched,
  be = NULL,
  organism = NULL,
  ncharSymb = 4,
  ncharName = 8,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="searchId_+3A_searched">searched</code></td>
<td>
<p>the searched term. Identifiers are searched by exact match.
Symbols and names are also searched for partial match when searched is
greater than ncharSymb and ncharName respectively.</p>
</td></tr>
<tr><td><code id="searchId_+3A_be">be</code></td>
<td>
<p>optional. If provided the search is focused on provided BEs.</p>
</td></tr>
<tr><td><code id="searchId_+3A_organism">organism</code></td>
<td>
<p>optional.  If provided the search is focused on provided
organisms.</p>
</td></tr>
<tr><td><code id="searchId_+3A_ncharsymb">ncharSymb</code></td>
<td>
<p>The minimum number of characters in searched to consider
incomplete symbol matches.</p>
</td></tr>
<tr><td><code id="searchId_+3A_ncharname">ncharName</code></td>
<td>
<p>The minimum number of characters in searched to consider
incomplete name matches.</p>
</td></tr>
<tr><td><code id="searchId_+3A_verbose">verbose</code></td>
<td>
<p>boolean indicating if the CQL queries should be displayed</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the following fields:
</p>

<ul>
<li> <p><strong>found</strong>: the element found in BED corresponding to the searched term
</p>
</li>
<li> <p><strong>be</strong>: the type of the element
</p>
</li>
<li> <p><strong>source</strong>: the source of the element
</p>
</li>
<li> <p><strong>organism</strong>: the related organism
</p>
</li>
<li> <p><strong>entity</strong>: the related entity internal ID
</p>
</li>
<li> <p><strong>ebe</strong>: the BE of the related entity
</p>
</li>
<li> <p><strong>canonical</strong>: if the symbol is canonical
</p>
</li>
<li> <p><strong>gene</strong>: list of the related genes BE internal ID
</p>
</li></ul>

<p>Exact matches are returned first folowed by the shortest elements.
</p>


<h3>See Also</h3>

<p><a href="#topic+getRelevantIds">getRelevantIds</a>
</p>

<hr>
<h2 id='setBedVersion'>Feeding BED: Set the BED version</h2><span id='topic+setBedVersion'></span>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
This function is used when modifying the BED content.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setBedVersion(bedInstance, bedVersion)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="setBedVersion_+3A_bedinstance">bedInstance</code></td>
<td>
<p>instance of BED to be set</p>
</td></tr>
<tr><td><code id="setBedVersion_+3A_bedversion">bedVersion</code></td>
<td>
<p>version of BED to be set</p>
</td></tr>
</table>

<hr>
<h2 id='showBedDataModel'>Show the data model of BED</h2><span id='topic+showBedDataModel'></span>

<h3>Description</h3>

<p>Show the shema of the BED data model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>showBedDataModel()
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
