<!DOCTYPE html><html lang="en"><head><title>Help for package BEDMatrix</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BEDMatrix}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BEDMatrix-package'><p>A Package to Extract Genotypes from a PLINK .bed File</p></a></li>
<li><a href='#BEDMatrix'><p>Create a BEDMatrix Object from a PLINK .bed File</p></a></li>
<li><a href='#BEDMatrix-class'><p>A Class to Extract Genotypes from a PLINK .bed File</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>2.0.4</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Title:</td>
<td>Extract Genotypes from a PLINK .bed File</td>
</tr>
<tr>
<td>Description:</td>
<td>A matrix-like data structure that allows for efficient,
    convenient, and scalable subsetting of binary genotype/phenotype files
    generated by PLINK (<a href="https://www.cog-genomics.org/plink2">https://www.cog-genomics.org/plink2</a>), the whole
    genome association analysis toolset, without loading the entire file into
    memory.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/QuantGen/BEDMatrix">https://github.com/QuantGen/BEDMatrix</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/QuantGen/BEDMatrix/issues">https://github.com/QuantGen/BEDMatrix/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, crochet (&ge; 2.3.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>data.table, LinkedMatrix, tinytest</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-09-06 18:54:35 UTC; agrueneberg</td>
</tr>
<tr>
<td>Author:</td>
<td>Alexander Grueneberg [aut, cre],
  Gustavo de los Campos [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alexander Grueneberg &lt;cran@agrueneberg.info&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-06 19:20:14 UTC</td>
</tr>
</table>
<hr>
<h2 id='BEDMatrix-package'>A Package to Extract Genotypes from a PLINK .bed File</h2><span id='topic+BEDMatrix-package'></span>

<h3>Description</h3>

<p>The <code>BEDMatrix</code> package provides a matrix-like wrapper around
<a href="https://www.cog-genomics.org/plink2/formats#bed">.bed</a> files, one of
the genotype/phenotype file formats of
<a href="https://www.cog-genomics.org/plink2">PLINK</a>, the whole genome
association analysis toolset. <code>BEDMatrix</code> objects are created by
simply providing the path to a .bed file and once created, they behave
similarly to regular matrices with the advantage that genotypes are
retrieved on demand without loading the entire file into memory. This
allows handling of very large files with limited use of memory.
</p>


<h3>.bed Files</h3>

<p><a href="https://www.cog-genomics.org/plink2/formats#bed">.bed</a> files
(sometimes referred to as binary .ped files) are binary representations of
genotype calls at biallelic variants. This very compact file format (2 bits
per genotype call) is used and generated by
<a href="https://www.cog-genomics.org/plink2/">PLINK</a>. .bed files should not
be confused with the
<a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format1">UCSC Genome
Browser's BED format</a>, which is totally different.
</p>
<p>A .bed file can be created from a
<a href="https://www.cog-genomics.org/plink2/formats#ped">.ped</a> file with
<a href="https://www.cog-genomics.org/plink2">PLINK</a> using <code>plink --file
    myfile --make-bed</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BEDMatrix-class">BEDMatrix-class</a></code> to learn more about the <code>BEDMatrix</code>
class.
</p>

<hr>
<h2 id='BEDMatrix'>Create a BEDMatrix Object from a PLINK .bed File</h2><span id='topic+BEDMatrix'></span>

<h3>Description</h3>

<p>This function constructs a new <code>BEDMatrix</code> object by mapping the
specified <a href="https://www.cog-genomics.org/plink2/formats#bed">PLINK
.bed</a> file into memory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BEDMatrix(path, n = NULL, p = NULL, simple_names = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BEDMatrix_+3A_path">path</code></td>
<td>

<p>Path to the
<a href="https://www.cog-genomics.org/plink2/formats#bed">.bed</a> file (with
or without extension).
</p>
</td></tr>
<tr><td><code id="BEDMatrix_+3A_n">n</code></td>
<td>

<p>The number of samples. If <code>NULL</code> (the default), this number will
be determined from the accompanying
<a href="https://www.cog-genomics.org/plink2/formats#fam">.fam</a> file (of
the same name as the
<a href="https://www.cog-genomics.org/plink2/formats#bed">.bed</a> file). If
a positive integer, the .fam file is not read and <code>rownames</code> will
be set to <code>NULL</code> and have to be provided manually.
</p>
</td></tr>
<tr><td><code id="BEDMatrix_+3A_p">p</code></td>
<td>

<p>The number of variants. If <code>NULL</code> (the default) the number of
variants will be determined from the accompanying
<a href="https://www.cog-genomics.org/plink2/formats#bim">.bim</a> file (of
the same name as the
<a href="https://www.cog-genomics.org/plink2/formats#bed">.bed</a> file). If
a positive integer, the .bim file is not read and <code>colnames</code> will
be set to <code>NULL</code> and have to be provided manually.
</p>
</td></tr>
<tr><td><code id="BEDMatrix_+3A_simple_names">simple_names</code></td>
<td>

<p>Whether to simplify the format of the dimension names. If <code>FALSE</code>
(the default), row names are concatenations of family IDs, <code>_</code>,
and within-family IDs, while column names are concatenations of variant
names, <code>_</code>, and allele codes for the first allele of the
<a href="https://www.cog-genomics.org/plink2/formats#bim">.bim</a> file
(<code>A1</code>). If <code>TRUE</code>, row names are within-family IDs only and
column names are variant names only.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><a href="https://www.cog-genomics.org/plink2/formats#bed">.bed</a> files must be
accompanied by <a href="https://www.cog-genomics.org/plink2/formats#fam">.fam</a>
and <a href="https://www.cog-genomics.org/plink2/formats#bim">.bim</a> files:
.fam files contain sample information, and .bim files contain variant
information. If the name of the .bed file is <em>plink</em>.bed then the
names of the .fam and .bim files have to be <em>plink</em>.fam and
<em>plink</em>.bim, respectively. The .fam and .bim files are used to extract
the number and names of samples and variants.
</p>
<p>For very large .bed files, reading the .fam and .bim files can take a long
time. If <code>n</code> and <code>p</code> are provided, these files are not read and
<code>dimnames</code> have to be provided manually.
</p>
<p>Currently, only the variant-major mode of .bed files is supported.
<a href="https://www.cog-genomics.org/plink2/">PLINK2</a> &quot;dropped&quot; support for
the sample-major mode by automatically converting files in this format to
the variant-major mode. Therefore, it is recommended to run files in
sample-major mode through PLINK2 first.
</p>


<h3>Value</h3>

<p>A <code>BEDMatrix</code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BEDMatrix-package">BEDMatrix-package</a></code> to learn more about the <code>BEDMatrix</code>
package, <code><a href="#topic+BEDMatrix-class">BEDMatrix-class</a></code> to learn more about the
<code>BEDMatrix</code> class.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get the path to the example .bed file
path &lt;- system.file("extdata", "example.bed",
                    package = "BEDMatrix")

# Create a BEDMatrix object the example .bed file
m1 &lt;- BEDMatrix(path)

# Create a BEDMatrix object the example .bed file without loading
# the .fam and .bim files
m2 &lt;- BEDMatrix(path, n = 50, p = 1000)
</code></pre>

<hr>
<h2 id='BEDMatrix-class'>A Class to Extract Genotypes from a PLINK .bed File</h2><span id='topic+BEDMatrix-class'></span><span id='topic+show+2CBEDMatrix-method'></span>

<h3>Description</h3>

<p><code>BEDMatrix</code> is a class that maps a
<a href="https://www.cog-genomics.org/plink2/formats#bed">PLINK .bed</a> file
into memory and behaves similarly to a regular <code>matrix</code> by
implementing key methods such as <code>[</code>, <code>dim</code>, and <code>dimnames</code>.
Subsets are extracted directly and on-demand from the .bed file without
loading the entire file into memory.
</p>


<h3>Details</h3>

<p>The subsets extracted from a <code>BEDMatrix</code> object are coded as the
allelic dosages of the first allele in the .bim file (A1), similarly to
<a href="https://www.cog-genomics.org/plink2/formats#raw">.raw</a> files
generated with the <code>--recode A</code> argument in
<a href="https://www.cog-genomics.org/plink2/">PLINK</a>).
</p>
<p>Internally, this class is an S4 class with the following slots that should
not be relied upon in actual code: <code>xptr</code>, <code>dims</code>, <code>dnames</code>,
and <code>path</code>. The .bed file is mapped into memory using <code>mmap</code> on
Unix and <code>MapViewOfFile</code> on Windows.
</p>


<h3>Slots</h3>


<dl>
<dt><code>xptr</code>:</dt><dd>
<p>An external pointer to the underlying <code>C</code> code.
</p>
</dd>
<dt><code>dims</code>:</dt><dd>
<p>An integer vector specifying the number of samples and variants as
determined by the the accompanying
<a href="https://www.cog-genomics.org/plink2/formats#fam">.fam</a> and
<a href="https://www.cog-genomics.org/plink2/formats#bim">.bim</a> files
or by the <code>n</code> and <code>p</code> parameters of the <code>BEDMatrix</code>
constructor function.
</p>
</dd>
<dt><code>dnames</code>:</dt><dd>
<p>A list describing the row names and column names of the object as
determined by the accompanying
<a href="https://www.cog-genomics.org/plink2/formats#fam">.fam</a> and
<a href="https://www.cog-genomics.org/plink2/formats#bim">.bim</a> files,
or <code>NULL</code> if the <code>n</code> and <code>p</code> parameters of the
<code>BEDMatrix</code> constructor function were provided.
</p>
</dd>
<dt><code>path</code>:</dt><dd>
<p>A character string containing the path to the .bed file.
</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt><code>[</code>:</dt><dd><p>Extract parts of an object</p>
</dd>
<dt><code>dim</code>:</dt><dd><p>Retrieve the dimension of an object</p>
</dd>
<dt><code>dimnames</code>:</dt><dd><p>Retrieve the dimnames of an object</p>
</dd>
<dt><code>dimnames&lt;-</code>:</dt><dd><p>Set the dimnames of an object</p>
</dd>
<dt><code>as.matrix</code>:</dt><dd><p>Turn the object into a matrix</p>
</dd>
<dt><code>is.matrix</code>:</dt><dd><p>Test if the object is a matrix</p>
</dd>
<dt><code>length</code>:</dt><dd><p>Get the length of an object</p>
</dd>
<dt><code>str</code>:</dt><dd><p>Display the internal structure of an object</p>
</dd>
<dt><code>show</code>:</dt><dd><p>Display the object</p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+BEDMatrix">BEDMatrix</a></code> to create a <code>BEDMatrix</code> object from a .bed
file, <code><a href="#topic+BEDMatrix-package">BEDMatrix-package</a></code> to learn more about the
<code>BEDMatrix</code> package,
<code><a href="LinkedMatrix.html#topic+LinkedMatrix-package">LinkedMatrix</a></code> to link
several <code>BEDMatrix</code> objects together.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get the path to the example .bed file
path &lt;- system.file("extdata", "example.bed",
                    package = "BEDMatrix")

# Create a BEDMatrix object the example .bed file
m &lt;- BEDMatrix(path)

# Get the dimensions of the BEDMatrix object
dim(m)

# Get the row names of the BEDMatrix object
rownames(m)

# Get the column names of the BEDMatrix object
colnames(m)

# Extract genotypes for the specified sample(s)
m[1, ]
m[1:3, ]
m["per0_per0", ]
m[c("per0_per0", "per1_per1", "per2_per2"), ]

# Extract genotypes for a particular variant
m[, 1]
m[, c("snp0_A", "snp1_C", "snp2_G")]

# Extract genotypes for the specified samples and variants
m[
    c("per0_per0", "per1_per1", "per2_per2"),
    c("snp0_A", "snp1_C", "snp2_G")
]
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
