<!DOCTYPE html><html><head><title>Help for package behaviorchange</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {behaviorchange}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#abcd'><p>Acyclic Behavior Change Diagram</p></a></li>
<li><a href='#abcd_specs_examples'><p>Simple example datasets for ABCDs</p></a></li>
<li><a href='#apply_graph_theme'><p>Apply multiple DiagrammeR global graph attributes</p></a></li>
<li><a href='#behaviorchange-package'><p>behaviorchange: Tools for Behavior Change Researchers and Professionals</p></a></li>
<li><a href='#cat0'><p>Concatenate to screen without spaces</p></a></li>
<li><a href='#CIBER'><p>Confidence Interval-Based Estimation of Relevance (CIBER)</p></a></li>
<li><a href='#CIBERlite'><p>CIBERlite</p></a></li>
<li><a href='#complecs'><p>Create a COMPLECS graph</p></a></li>
<li><a href='#complecs_to_precede'><p>Represent a COMPLECS specification as a PRECEDE model</p></a></li>
<li><a href='#convert.threshold.to.er'><p>Visualising Numbers Needed for Change</p></a></li>
<li><a href='#determinant_selection_table'><p>Potential for Change Index and the Determinant Selection Table</p></a></li>
<li><a href='#determinantStructure'><p>Determinant Structure specification</p></a></li>
<li><a href='#detStructAddVarLabels'><p>Functions to preprocess determinant structures</p></a></li>
<li><a href='#dMCD'><p>Estimate Cohen's d corresponding to a Meaningful Change Definition</p></a></li>
<li><a href='#lm_rSq_ci'><p>Obtaining an R squared confidence interval estimate for an lm regression</p></a></li>
<li><a href='#nnc'><p>Numbers Needed for Change</p></a></li>
<li><a href='#opts'><p>Options for the behaviorchange package</p></a></li>
<li><a href='#partypanelData'><p>Subsets of Party Panel datasets</p></a></li>
<li><a href='#pies'><p>Practically Important Effect Sizes</p></a></li>
<li><a href='#repeatStr'><p>Repeat a string a number of times</p></a></li>
<li><a href='#vecTxt'><p>Easily parse a vector into a character value</p></a></li>
<li><a href='#wrapVector'><p>Wrap all elements in a vector</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Tools for Behavior Change Researchers and Professionals</td>
</tr>
<tr>
<td>Version:</td>
<td>0.5.5</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Gjalt-Jorn Peters &lt;behaviorchange@opens.science&gt;</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Contains specialised analyses and
    visualisation tools for behavior change science.
    These facilitate conducting determinant studies
    (for example, using confidence interval-based
    estimation of relevance, CIBER, or CIBERlite
    plots, see Crutzen, Noijen &amp; Peters (2017)
    &lt;<a href="https://doi.org/10%2Fghtfz9">doi:10/ghtfz9</a>&gt;),
    systematically developing, reporting,
    and analysing interventions (for example, using
    Acyclic Behavior Change Diagrams), and reporting
    about intervention effectiveness (for example, using
    the Numbers Needed for Change, see Gruijters &amp; Peters
    (2017) &lt;<a href="https://doi.org/10%2Fjzkt">doi:10/jzkt</a>&gt;), and computing the
    required sample size (using the Meaningful Change
    Definition, see Gruijters &amp; Peters (2020)
    &lt;<a href="https://doi.org/10%2Fghpnx8">doi:10/ghpnx8</a>&gt;).
    This package is especially useful for
    researchers in the field of behavior change or
    health psychology and to behavior change
    professionals such as intervention developers and
    prevention workers.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://r-packages.gitlab.io/behaviorchange">https://r-packages.gitlab.io/behaviorchange</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://gitlab.com/r-packages/behaviorchange/-/issues">https://gitlab.com/r-packages/behaviorchange/-/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>BiasedUrn (&ge; 1.07), data.tree (&ge; 0.7.5), DiagrammeR (&ge;
1.0.0), DiagrammeRsvg (&ge; 0.1.0), ggplot2 (&ge; 2.2.1),
googlesheets4 (&ge; 0.2.0), gridExtra (&ge; 2.3), gtable (&ge;
0.2.0), knitr (&ge; 1.0), methods (&ge; 3.0), rmdpartials (&ge;
0.5.0), ufs (&ge; 0.3.2), viridis (&ge; 0.5.1), yum (&ge; 0.0.1)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>htmltools, kableExtra, openxlsx, png (&ge; 0.1), rmarkdown,
rstudioapi, rsvg, webshot</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-03-04 22:07:43 UTC; gjalt</td>
</tr>
<tr>
<td>Author:</td>
<td>Gjalt-Jorn Peters <a href="https://orcid.org/0000-0002-0336-9589"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Rik Crutzen <a href="https://orcid.org/0000-0002-3731-6610"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Stefan Gruijters <a href="https://orcid.org/0000-0003-0141-0071"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-03-04 23:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='abcd'>Acyclic Behavior Change Diagram</h2><span id='topic+abcd'></span><span id='topic+print.abcdiagram'></span>

<h3>Description</h3>

<p>This function generates an acyclic behavior change diagram (ABCD)
from a specification in a google sheet or .csv file. An ABCD is
a logic model that illustrates the assumptions underlying a
behavior change intervention. Specifically, the ABCD shows the
assumed causal and structural assumptions, thereby showing what is
assumed to cause what (e.g. which elements of the intervention are
assumed to influence which aspects of the target population's
psychology?) and what is assumed to consist of what (e.g. which
determinants are assumed to contain which specific aspects of the
target population's psychology?).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>abcd(
  specs,
  specCols = c("bcps", "cnds", "apps", "sdts", "dets", "pobs", "behs"),
  localBackup = NULL,
  title = "Acyclic Behavior Change Diagram\n\n",
  outputFile = NULL,
  outputWidth = 3000,
  outputHeight = 1500,
  includeColNames = TRUE,
  maxLabelLength = 30,
  nodeFontSize = 10,
  edgeFontSize = 8,
  colNameFontSize = nodeFontSize,
  grayscale = FALSE,
  colorTheme = behaviorchange::opts$get("aabbcc"),
  penWidth = 1,
  silent = FALSE,
  returnGraphOnly = FALSE,
  returnSvgOnly = FALSE,
  columnWarning = TRUE,
  graphTheme = list(c("fontname", "Arial", "node")),
  regExReplacements = behaviorchange::opts$get("diagrammerSanitization")
)

## S3 method for class 'abcdiagram'
print(
  x,
  width = x$input$width,
  height = x$input$height,
  title = DiagrammeR::get_graph_name(x$output$graph),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="abcd_+3A_specs">specs</code></td>
<td>
<p>The specifications: either a google sheets URL,
the path to a local file, a character vector with both,
or a matrix or data frame</p>
</td></tr>
<tr><td><code id="abcd_+3A_speccols">specCols</code></td>
<td>
<p>The order of the columns. This character vector
specified the order of the elements of an ABCD. In the default
order, from left to right, these are (see below for definitions
and more details):
</p>

<ul>
<li> <p><code>bcps</code> = <strong>Behavior Change Principles (BCPs)</strong>;
</p>
</li>
<li> <p><code>cnds</code> = <strong>Conditions for effectiveness</strong>;
</p>
</li>
<li> <p><code>apps</code> = <strong>Applications</strong>;
</p>
</li>
<li> <p><code>sdts</code> = <strong>Sub-determinants</strong>;
</p>
</li>
<li> <p><code>dets</code> = <strong>Determinants</strong>;
</p>
</li>
<li> <p><code>pobs</code> = <strong>Performance Objectives</strong>;
</p>
</li>
<li> <p><code>behs</code> = <strong>Behaviors</strong>;
</p>
</li></ul>
</td></tr>
<tr><td><code id="abcd_+3A_localbackup">localBackup</code></td>
<td>
<p>Whether to write the specifications
to a local backup</p>
</td></tr>
<tr><td><code id="abcd_+3A_title">title</code></td>
<td>
<p>The title of the diagram</p>
</td></tr>
<tr><td><code id="abcd_+3A_outputfile">outputFile</code></td>
<td>
<p>If specified, the ABCD is written to
this file using <a href="DiagrammeR.html#topic+export_graph">DiagrammeR::export_graph</a>.</p>
</td></tr>
<tr><td><code id="abcd_+3A_outputwidth">outputWidth</code>, <code id="abcd_+3A_outputheight">outputHeight</code></td>
<td>
<p>If an <code>outputFile</code> is
specified, these determine its width and height (in pixels)</p>
</td></tr>
<tr><td><code id="abcd_+3A_includecolnames">includeColNames</code></td>
<td>
<p>Whether to include the column names
as titles/legend for the entities in each 'column'
of the ABCD.</p>
</td></tr>
<tr><td><code id="abcd_+3A_maxlabellength">maxLabelLength</code></td>
<td>
<p>At which width to word wrap the
labels.</p>
</td></tr>
<tr><td><code id="abcd_+3A_nodefontsize">nodeFontSize</code>, <code id="abcd_+3A_edgefontsize">edgeFontSize</code>, <code id="abcd_+3A_colnamefontsize">colNameFontSize</code></td>
<td>
<p>Font sizes of the nodes (i.e.
the text in boxes), edges (basically the conditions for effectiveness) and the
column names (at the bottom).</p>
</td></tr>
<tr><td><code id="abcd_+3A_grayscale">grayscale</code></td>
<td>
<p>Whether to use the <code>colorTheme</code> or produce a grayscale ABCD.</p>
</td></tr>
<tr><td><code id="abcd_+3A_colortheme">colorTheme</code></td>
<td>
<p>The color theme, a named list containing the colors,
each a character vector with three HTML (hex) color values. The list elements
have to be named <code>bcp</code>, <code>condition_for_effectiveness</code>, <code>application</code>,
<code>sub_determinant</code>, <code>determinant</code>, <code>sub_behavior</code>, and <code>target_behavior</code>, and
each must contain a named vector with two elements named <code>fill</code>, <code>stroke</code>,
and <code>text</code>, containing the color codes for the fill, stroke, and text,
respectively; see <code>behaviorchange::opts$get("aabbcc")</code> for an example.</p>
</td></tr>
<tr><td><code id="abcd_+3A_penwidth">penWidth</code></td>
<td>
<p>The width of the pen to draw the strokes.</p>
</td></tr>
<tr><td><code id="abcd_+3A_silent">silent</code></td>
<td>
<p>Whether to suppress (<code>TRUE</code>) or show
(<code>FALSE</code>) more detailed information.</p>
</td></tr>
<tr><td><code id="abcd_+3A_returngraphonly">returnGraphOnly</code>, <code id="abcd_+3A_returnsvgonly">returnSvgOnly</code></td>
<td>
<p>Whether to return the full results
object or only either the <a href="DiagrammeR.html#topic+DiagrammeR">DiagrammeR::DiagrammeR</a> graph or a one-value
character vector containing a Scalable Vector Graphic as produced by
<code><a href="DiagrammeRsvg.html#topic+export_svg">DiagrammeRsvg::export_svg()</a></code>.</p>
</td></tr>
<tr><td><code id="abcd_+3A_columnwarning">columnWarning</code></td>
<td>
<p>Can be used to suppress the warning if the number of
columns is too large.</p>
</td></tr>
<tr><td><code id="abcd_+3A_graphtheme">graphTheme</code></td>
<td>
<p>Specific settings to apply to the graph
using <code><a href="#topic+apply_graph_theme">apply_graph_theme()</a></code>; a list of vectors, where each vector has
three elements: the setting, the value, and what to apply it to ('node',
'edge', or 'graph').</p>
</td></tr>
<tr><td><code id="abcd_+3A_regexreplacements">regExReplacements</code></td>
<td>
<p>A list of pairs of regular
expressions that will be applied to the specifications
before generating the ABCD. This can be used to sanitize
problematic characters (e.g. ', &quot; and \).</p>
</td></tr>
<tr><td><code id="abcd_+3A_x">x</code></td>
<td>
<p>The ABCD object to print (as generated by a call to <code>abcd</code>).</p>
</td></tr>
<tr><td><code id="abcd_+3A_width">width</code>, <code id="abcd_+3A_height">height</code></td>
<td>
<p>Width and height to use when printing the ABCD.</p>
</td></tr>
<tr><td><code id="abcd_+3A_...">...</code></td>
<td>
<p>Any additional arguments are passed on to
<code><a href="DiagrammeR.html#topic+render_graph">DiagrammeR::render_graph()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Specifically, a full ABCD is a model that shows the following
elements:
</p>

<ul>
<li> <p><strong>Behavior Change Principles (BCPs)</strong>: The specific
psychological principles engaged to influence the relevant
sub-determinants, usually selected using the determinants
to which the sub-determinants 'belong'. These are also
known as methods of behavior change in the Intervention
Mapping framework, or behavior change techniques, BCTs,
in the Behavior Change Wheel approach. For a list of 99
BCPs, see Kok et al. (2016).
</p>
</li>
<li> <p><strong>Conditions for effectiveness</strong>: The conditions
that need to be met for a Behavior Change Principle (BCP) to
be effective. These conditions depend on the specific
underlying Evolutionary Learning Processes (ELPs) that the
BCP engages (Crutzen &amp; Peters, 2018). If the conditions for
effectiveness (called <em>parameters</em> for effectiveness in the
Intervention Mapping framework) are not met, the method will
likely not be effective, or at least, not achieve its
maximum effectiveness.
</p>
</li>
<li> <p><strong>Applications</strong>: Since BCP's describe aspects of
human psychology in general, they are necessarily formulated
on a generic level. Therefore, using them in an intervention
requires translating them to the specific target population,
culture, available means, and context. The result of this
translation is the application of the BCP. Multiple BCPs can
be combined into one application; and one BCP can be applied in
multiple applications (see Kok, 2014).
</p>
</li>
<li> <p><strong>Sub-determinants</strong>: Behavior change interventions
engage specific aspects of the human psychology (ideally, they
specifically, target those aspects found most important in
predicting the target behavior, as can be established with
<code><a href="#topic+CIBER">CIBER</a></code> plots. These aspects are
called sub-determinants (the Intervention Mapping framework
references <em>Change Objectives</em>, which are sub-determinants
formulated according to specific guidelines). In some
theoretical traditions, sub-determinants are called <em>beliefs</em>.
</p>
</li>
<li> <p><strong>Determinants</strong>: The overarching psychological constructs that
are defined as clusters of specific aspects of the human
psychology that explain humans' behavior (and are targeted
by behavior change interventions). Psychological theories
contain specific definitions of such determinants, and make
statements about how they relate to each other and to human
behavior. There are also theories (and exists empirical
evidence) on how these determinants can be changed (i.e. BCPs),
so althought the sub-determinants are what is targeted in an
intervention, the selection of feasible BCPs requires knowing
to which determinants those sub-determinants belong.
</p>
</li>
<li> <p><strong>Performance objectives</strong>: The specific sub-behaviors that often
underlie (or make up) the ultimate target behavior. These are
distinguished from the overarching target behavior because
the relevant determinants of these sub-behaviors can be
different: for example, the reasons why people do or do not
<em>buy</em> condoms can be very different from the reasons why they
do or do not <em>carry</em> condoms or why they do or do not
<em>negotiate</em> condom use with a sexual partner.
</p>
</li>
<li> <p><strong>Behavior</strong>: The ultimate target behavior of the intervention,
usually an umbrella that implicitly contains multiple
performance objectives.
</p>
</li></ul>

<p>For details, see Peters et al. (2019).
</p>


<h3>Value</h3>

<p>A list consisting of an <code>input</code>, <code>intermediate</code>, and
<code>output</code> list, where the ABCD is stored in the <code>output</code> list
as a <a href="DiagrammeR.html#topic+DiagrammeR">DiagrammeR::DiagrammeR</a> called <code>graph</code>.
</p>


<h3>Author(s)</h3>

<p>Gjalt-Jorn Peters, <a href="mailto:gjalt-jorn@a-bc.eu">gjalt-jorn@a-bc.eu</a>, with
contributions from Matti Heino and Sander Eggers.
</p>


<h3>References</h3>

<p>Crutzen, R., &amp; Peters, G.-J. Y. (2018). Evolutionary
learning processes as the foundation for behaviour change.
<em>Health Psychology Review,</em> 12(1), 43–57.
https://doi.org/10.1080/17437199.2017.1362569
</p>
<p>Kok, G. (2014). A practical guide to effective behavior
change: How to apply theory- and evidence-based behavior
change methods in an intervention. <em>European Health Psychologist</em>,
16(5), 156–170. https://doi.org/10.31234/osf.io/r78wh
</p>
<p>Kok, G., Gottlieb, N. H., Peters, G.-J. Y., Mullen,
P. D., Parcel, G. S., Ruiter, R. A. C., … Bartholomew,
L. K. (2016). A taxonomy of behavior change methods:
an Intervention Mapping approach. <em>Health Psychology
Review</em>, 10(3), 297–312.
https://doi.org/10.1080/17437199.2015.1077155
</p>
<p>Peters, G.-J. Y., et al. (2019) The core of
behavior change: introducing the Acyclic Behavior Change
Diagram to report and analyze interventions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Load one of the ABCD matrices supplied
### with the behaviorchange package
data(abcd_specification_example_xtc);

### Create ABCD matrix (using 'print' to allow pkgdown() to print properly).
print(behaviorchange::abcd(abcd_specification_example_xtc));

### Other examples not executed during testing as creating ABCDs takes long

## Not run: 
### Change the appearance; note that many attributes are specified
### for specific elements, and element-level settings always override
### the global settings that can be specified here.
print(
  behaviorchange::abcd(
    abcd_specification_example_xtc,
    graphTheme = list(
      c("fontname", "Courier New", "node")
    )
  )
);

## End(Not run)

</code></pre>

<hr>
<h2 id='abcd_specs_examples'>Simple example datasets for ABCDs</h2><span id='topic+abcd_specs_examples'></span><span id='topic+abcd_specs_complete'></span><span id='topic+abcd_specs_without_conditions'></span><span id='topic+abcd_specs_single_po_without_conditions'></span><span id='topic+abcd_specs_dutch_xtc'></span><span id='topic+abcd_specification_example_xtc'></span><span id='topic+abcd_specification_empty'></span>

<h3>Description</h3>

<p>This are three (nested) datasets illustrating the logic model of change for
a simple condom use intervention in a way that can be visualised using
the <a href="#topic+abcd">abcd</a> function. The full dataset is <code>abcd_specs_full</code>, a subset
that does not explicitly include the conditions for effectiveness
(instead showing letters that can then be explained in, for example,
the manuscript text) is called <code>abcd_specs_without_conditions</code>, and
a version that only contains the information about one sub-behavior
(performance objective) is available as
<code>abcd_specs_single_po_without_conditions</code>. The variables in the full
dataset are:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(abcd_specs_complete)

data(abcd_specs_without_conditions)

data(abcd_specs_single_po_without_conditions)

data(abcd_specification_example_xtc)

data(abcd_specs_dutch_xtc)

data(abcd_specification_empty)
</code></pre>


<h3>Format</h3>

<p>For <code>abcd_specs_complete</code>, a data frame with 7 variables and 7 rows;
for <code>abcd_specs_without_conditions</code>, a data frame with 6 variables and 7 rows;
for <code>abcd_specs_single_po_without_conditions</code>, a data frame with 5 variables and 4 rows;
for <code>abcd_specification_example_xtc</code> and <code>abcd_specs_dutch_xtc</code>,
a data frame with 7 variables and 5 rows' and
for <code>abcd_specification_empty</code>, a data frame with 7 variables and 1 row.
</p>


<h3>Details</h3>


<ul>
<li> <p><code style="white-space: pre;">&#8288;Behavior Change Principles&#8288;</code>: The behavior change principles (BCPs), also known as methods for behavior change or 'behavior change techniques' (BCTs), that describe the psychological principles that are assumed to realise the change in the (sub-)determinants.
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;Conditions for effectiveness\\n(e.g. parameters for use)&#8288;</code>: The conditions for effectiveness that describe the constraints and considerations taken into account in the translation of the BCPs to practical applications for the relevant target population, context, culture, etc.
</p>
</li>
<li> <p><code>Applications</code>: The applications of these BCPs. Where the BCPs describe theoretical principles, the applications are more or less tangible intervention elements.
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;Sub-determinants\\n(e.g. beliefs; can be formulated as Change Objectives)&#8288;</code>: The specific aspects of teh target population's psychology that are targeted by the BCPs (e.g. beliefs, or in Intervention Mapping vocabulary, Change Objectives).
</p>
</li>
<li> <p><code>Determinants</code>: The determinants, psychological constructs, that the targeted sub-determinants are a part of, and that together predict the Performance Objectives (sub-behaviors).
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;Performance Objectives&#8288;</code>: Explicitly defined sub-behaviors at a level of specificity that distinguishes them from other sub-behaviors, and that together form the target behavior.
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;Target Behavior&#8288;</code>: The ultimate target behavior, usually defined at a relatively general level.
</p>
</li></ul>

<p>In addition to these three datasets, a Dutch example specification
is included named <code>abcd_specs_dutch_xtc</code>, and the same in English as <code>abcd_specification_example_xtc</code>.
</p>
<p>Finally, <code>abcd_specification_empty</code> is an empty 'template' ABCD matrix.
</p>

<hr>
<h2 id='apply_graph_theme'>Apply multiple DiagrammeR global graph attributes</h2><span id='topic+apply_graph_theme'></span>

<h3>Description</h3>

<p>Apply multiple DiagrammeR global graph attributes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>apply_graph_theme(graph, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="apply_graph_theme_+3A_graph">graph</code></td>
<td>
<p>The <a href="DiagrammeR.html#topic+DiagrammeR">DiagrammeR::DiagrammeR</a> graph to apply the attributes to.</p>
</td></tr>
<tr><td><code id="apply_graph_theme_+3A_...">...</code></td>
<td>
<p>One or more character vectors of length three, where the first element is
the attribute, the second the value, and the third, the attribute type (<code>graph</code>,
<code>node</code>, or <code>edge</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The <a href="DiagrammeR.html#topic+DiagrammeR">DiagrammeR::DiagrammeR</a> graph.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>abcd_complete  &lt;- behaviorchange::abcd(behaviorchange::abcd_specs_complete)$output$graph;
abcd_complete &lt;- apply_graph_theme(abcd_complete,
                                   c("penwidth", 5, "node"),
                                   c("penwidth", 15, "edge"));
</code></pre>

<hr>
<h2 id='behaviorchange-package'>behaviorchange: Tools for Behavior Change Researchers and Professionals</h2><span id='topic+behaviorchange'></span><span id='topic+behaviorchange-package'></span>

<h3>Description</h3>

<p>Contains specialised analyses and visualisation tools for behavior change science. These facilitate conducting determinant studies (for example, using confidence interval-based estimation of relevance, CIBER, or CIBERlite plots, see Crutzen, Noijen &amp; Peters (2017) <a href="https://doi.org/10/ghtfz9">doi:10/ghtfz9</a>), systematically developing, reporting, and analysing interventions (for example, using Acyclic Behavior Change Diagrams), and reporting about intervention effectiveness (for example, using the Numbers Needed for Change, see Gruijters &amp; Peters (2017) <a href="https://doi.org/10/jzkt">doi:10/jzkt</a>), and computing the required sample size (using the Meaningful Change Definition, see Gruijters &amp; Peters (2020) <a href="https://doi.org/10/ghpnx8">doi:10/ghpnx8</a>). This package is especially useful for researchers in the field of behavior change or health psychology and to behavior change professionals such as intervention developers and prevention workers.
</p>


<h3>Details</h3>

<p>Contains specialised analyses and
visualisation tools for behavior change science.
These facilitate conducting determinant studies
(for example, using confidence interval-based
estimation of relevance, CIBER, or CIBERlite
plots), systematically developing, reporting,
and analysing interventions (for example, using
acyclic behavior change diagrams), and reporting
about intervention effectiveness (for example, using
the Numbers Needed for Change), and computing the
required sample size (using the Meaningful Change
Definition). This package is especially useful for
researchers in the field of behavior change or
health psychology and to behavior change
professionals such as intervention developers and
prevention workers.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Gjalt-Jorn Peters <a href="mailto:behaviorchange@opens.science">behaviorchange@opens.science</a> (<a href="https://orcid.org/0000-0002-0336-9589">ORCID</a>)
</p>
<p>Other contributors:
</p>

<ul>
<li><p> Rik Crutzen <a href="mailto:rik.crutzen@maastrichtuniversity.nl">rik.crutzen@maastrichtuniversity.nl</a> (<a href="https://orcid.org/0000-0002-3731-6610">ORCID</a>) [contributor]
</p>
</li>
<li><p> Stefan Gruijters <a href="mailto:mail@stefangruijters.nl">mail@stefangruijters.nl</a> (<a href="https://orcid.org/0000-0003-0141-0071">ORCID</a>) [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://r-packages.gitlab.io/behaviorchange">https://r-packages.gitlab.io/behaviorchange</a>
</p>
</li>
<li><p> Report bugs at <a href="https://gitlab.com/r-packages/behaviorchange/-/issues">https://gitlab.com/r-packages/behaviorchange/-/issues</a>
</p>
</li></ul>


<hr>
<h2 id='cat0'>Concatenate to screen without spaces</h2><span id='topic+cat0'></span>

<h3>Description</h3>

<p>The cat0 function is to cat what paste0 is to paste; it
simply makes concatenating many strings without a separator easier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cat0(..., sep = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cat0_+3A_...">...</code></td>
<td>
<p>The character vector(s) to print; passed to <a href="base.html#topic+cat">cat</a>.</p>
</td></tr>
<tr><td><code id="cat0_+3A_sep">sep</code></td>
<td>
<p>The separator to pass to <a href="base.html#topic+cat">cat</a>, of course, <code>""</code> by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing (invisible <code>NULL</code>, like <a href="base.html#topic+cat">cat</a>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cat0("The first variable is '", names(mtcars)[1], "'.");
</code></pre>

<hr>
<h2 id='CIBER'>Confidence Interval-Based Estimation of Relevance (CIBER)</h2><span id='topic+CIBER'></span><span id='topic+detStructCIBER'></span><span id='topic+binaryCIBER'></span>

<h3>Description</h3>

<p>This function generates a high-level plot consisting of several diamond
plots. This function is useful for estimating the relative relevance of a
set of determinants of, for example, behavior. The plot in the left hand
panel shows each determinant's distribution with a diamond representing the
confidence interval. The right hand plot shows the determinants'
associations to one or more 'target' variables, such as behavior or
determinants of behavior.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CIBER(
  data,
  determinants,
  targets,
  conf.level = list(means = 0.9999, associations = 0.95),
  subQuestions = NULL,
  leftAnchors = rep("Lo", length(determinants)),
  rightAnchors = rep("Hi", length(determinants)),
  outputFile = NULL,
  outputWidth = NULL,
  outputHeight = NULL,
  outputUnits = "in",
  outputParams = list(),
  orderBy = NULL,
  decreasing = NULL,
  numberSubQuestions = FALSE,
  generateColors = list(means = c("red", "blue", "green"), associations = c("red",
    "grey", "green")),
  strokeColors = viridis::viridis(length(targets)),
  vLines = c(-0.5, 0, 0.5),
  vLineColors = "grey",
  titlePrefix = "Means and associations (r) with",
  titleVarLabels = NULL,
  titleSuffix = "",
  fullColorRange = NULL,
  associationsAlpha = 0.5,
  returnPlotOnly = TRUE,
  drawPlot = TRUE,
  jitterWidth = 0.45,
  baseSize = 0.8,
  dotSize = 2.5 * baseSize,
  baseFontSize = 10 * baseSize,
  theme = ggplot2::theme_bw(base_size = baseFontSize),
  xbreaks = NULL,
  rsq = TRUE,
  ...
)

binaryCIBER(
  data,
  determinants,
  targets,
  conf.level = list(means = 0.9999, associations = 0.95),
  subQuestions = NULL,
  leftAnchors = rep("Lo", length(determinants)),
  rightAnchors = rep("Hi", length(determinants)),
  outputFile = NULL,
  outputWidth = NULL,
  outputHeight = NULL,
  outputUnits = "in",
  outputParams = list(),
  orderBy = NULL,
  decreasing = NULL,
  numberSubQuestions = FALSE,
  comparisonColors = viridis::viridis(2, end = 0.5),
  categoryLabels = NULL,
  generateColors = list(means = c("red", "blue", "green"), associations = c("red",
    "grey", "green")),
  strokeColors = viridis::viridis(length(targets)),
  vLines = c(-0.8, 0, 0.8),
  vLineColors = "grey",
  titlePrefix = "Means and associations (d) with",
  titleVarLabels = NULL,
  titleSuffix = "",
  fullColorRange = NULL,
  associationsAlpha = 0.5,
  returnPlotOnly = TRUE,
  drawPlot = TRUE,
  baseSize = 0.8,
  dotSize = 2.5 * baseSize,
  baseFontSize = 10 * baseSize,
  theme = ggplot2::theme_bw(base_size = baseFontSize),
  xbreaks = NULL,
  ...
)

detStructCIBER(
  determinantStructure,
  data,
  conf.level = list(means = 0.9999, associations = 0.95),
  subQuestions = NULL,
  leftAnchors = rep("Lo", length(determinants)),
  rightAnchors = rep("Hi", length(determinants)),
  orderBy = 1,
  decreasing = NULL,
  generateColors = list(means = c("red", "blue", "green"), associations = c("red",
    "grey", "green")),
  strokeColors = NULL,
  titlePrefix = "Means and associations with",
  titleVarLabels = NULL,
  titleSuffix = "",
  fullColorRange = NULL,
  associationsAlpha = 0.5,
  baseSize = 0.8,
  dotSize = 2.5 * baseSize,
  baseFontSize = 10 * baseSize,
  theme = ggplot2::theme_bw(base_size = baseFontSize),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CIBER_+3A_data">data</code></td>
<td>
<p>The dataframe containing the variables.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_determinants">determinants</code></td>
<td>
<p>The 'determinants': the predictors (or 'covariates') of
the target variables(s) (or 'criteria').</p>
</td></tr>
<tr><td><code id="CIBER_+3A_targets">targets</code></td>
<td>
<p>The 'targets' or 'criteria' variables: the variables
predicted by the determinants.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_conf.level">conf.level</code></td>
<td>
<p>The confidence levels for the confidence intervals: has to
be a named list with two elements: <code>means</code> and <code>associations</code>,
specifying the desired confidence levels for the means and associations,
respectively. The confidence level for the associations is also used for the
intervals for the proportions of explained variance.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_subquestions">subQuestions</code></td>
<td>
<p>The subquestions used to measure each determinants. This
can also be used to provide pretty names for the variables if the
determinants were not measured by one question each. Must have the same
length as <code>determinants</code>.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_leftanchors">leftAnchors</code></td>
<td>
<p>The anchors to display on the left side of the left hand
panel. If the determinants were measured with one variable each, this can be
used to show the anchors that were used for the respective scales. Must have
the same length as <code>determinants</code>.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_rightanchors">rightAnchors</code></td>
<td>
<p>The anchors to display on the left side of the left hand
panel. If the determinants were measured with one variable each, this can be
used to show the anchors that were used for the respective scales. Must have
the same length as <code>determinants</code>.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_outputfile">outputFile</code></td>
<td>
<p>The file to write the output to (the plot is not stored to
disk if <code>NULL</code>). The extension can be specified to change the file type.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_outputwidth">outputWidth</code>, <code id="CIBER_+3A_outputheight">outputHeight</code>, <code id="CIBER_+3A_outputunits">outputUnits</code></td>
<td>
<p>The width, height,
and units for the output file.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_outputparams">outputParams</code></td>
<td>
<p>More advanced parameters for the output file. This can be
used to pass arguments to <code><a href="ggplot2.html#topic+ggsave">ggplot2::ggsave()</a></code>, such as passing
<code>outputParams=list(type="cairo-png")</code> to use anti-aliasing when saving a PNG file.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_orderby">orderBy</code></td>
<td>
<p>Whether to sort the determinants. Set to <code>NULL</code> to not
sort at all; specify the name or index of one of the <code>target</code>s to sort
by the point estimates of the associations with that target variable. Use
<code>decreasing</code> to determine whether to sort in ascending or descending
order. For convenience, if <code>orderBy</code> is not <code>NULL</code>, but
<code>decreasing</code> is, the determinants are sorted in descending (decreasing)
order.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_decreasing">decreasing</code></td>
<td>
<p>Whether to sort the determinants. Specify <code>NULL</code> to
not sort at all, <code>TRUE</code> to sort in descending order, and <code>FALSE</code>
to sort in ascending order. If <code>decreasing</code> is nor <code>NULL</code>, but
<code>orderBy</code> is <code>NULL</code>, the determinants are sorted by their means.
For convenience, if <code>orderBy</code> is not <code>NULL</code>, but <code>decreasing</code>
is, the determinants are sorted in descending (decreasing) order.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_numbersubquestions">numberSubQuestions</code></td>
<td>
<p>Whether or not to number the subquestions. If they
are numbered, they are numbered from the top to the bottom.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_generatecolors">generateColors</code></td>
<td>
<p>The colors to use to generate the gradients for
coloring the diamonds representing the confidence intervals. Has to be a
named list with two elements: <code>means</code> and <code>associations</code>,
specifying the desired colors for the means and associations, respectively.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_strokecolors">strokeColors</code></td>
<td>
<p>The palette to use to color the stroke of the confidence
intervals for the associations between the determinants and the targets.
Successive colors from this palette are used for the targets.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_vlines">vLines</code>, <code id="CIBER_+3A_vlinecolors">vLineColors</code></td>
<td>
<p>In the association plot, vertical lines can
be plotted to facilitate interpretation. Specify their locations and
colors here, or set one or both to <code>NULL</code> to eliminate them.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_titleprefix">titlePrefix</code></td>
<td>
<p>Text to add before the list of target names and the
proportions of explained variance for each target. This plot title also
serves as legend to indicate which target 'gets' which each color.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_titlevarlabels">titleVarLabels</code></td>
<td>
<p>Optionally, variable labels to use in the plot title.
Has to be the exact same length as <code>targets</code>.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_titlesuffix">titleSuffix</code></td>
<td>
<p>Text to add after the list of target names and the
proportions of explained variance for each target.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_fullcolorrange">fullColorRange</code></td>
<td>
<p>If colors are specified, this can be used to specify
which values, for the determinant confidence intervals in the left hand
panel, are the minimum and maximum. This is useful if those scores are not
actually in the data (e.g. for extremely skewed distributions). If
<code>NULL</code>, the range of all individual scores on the determinants is used.
For the associations, <code>c(-1, 1)</code> is always used as
<code>fullColorRange</code>.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_associationsalpha">associationsAlpha</code></td>
<td>
<p>The alpha level (transparency) of the confidence
interval diamonds in the right hand plot. Value between 0 and 1, where 0
signifies complete transparency (i.e. invisibility) and 1 signifies complete
'opaqueness'.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_returnplotonly">returnPlotOnly</code></td>
<td>
<p>Whether to return the entire object that is generated
(including all intermediate objects) or only the plot.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_drawplot">drawPlot</code></td>
<td>
<p>Whether the draw the plot, or only return it.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_jitterwidth">jitterWidth</code></td>
<td>
<p>How much to jitter the data points in the left hand plot.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_basesize">baseSize</code></td>
<td>
<p>This can be used to efficiently change the size of most plot
elements.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_dotsize">dotSize</code></td>
<td>
<p>This is the size of the points used to show the individual
data points in the left hand plot.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_basefontsize">baseFontSize</code></td>
<td>
<p>This can be used to set the font size separately from
the <code>baseSize</code>.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_theme">theme</code></td>
<td>
<p>This is the theme that is used for the plots.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_xbreaks">xbreaks</code></td>
<td>
<p>Which breaks to use on the X axis (can be useful to override
<code><a href="ggplot2.html#topic+ggplot2">ggplot2</a></code>'s defaults).</p>
</td></tr>
<tr><td><code id="CIBER_+3A_rsq">rsq</code></td>
<td>
<p>Whether to compute the R squared values.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_...">...</code></td>
<td>
<p>These arguments are passed on to
<code>biAxisDiamondPlot</code> (for the left panel) and
<code>diamondPlot</code> (for the right panel).  Note that all argument are
passed to both those functions.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_comparisoncolors">comparisonColors</code></td>
<td>
<p>Colors to use for the two groups in a
binary CIBER plot with one (dichotomous) target.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_categorylabels">categoryLabels</code></td>
<td>
<p>Labels for the two values of the target.</p>
</td></tr>
<tr><td><code id="CIBER_+3A_determinantstructure">determinantStructure</code></td>
<td>
<p>When using <code>detStructCIBER</code>, the
determinant structure as generated by <code><a href="#topic+determinantStructure">determinantStructure</a></code> is
included here. <code>determinants</code>, <code>targets</code>, <code>subQuestions</code>,
<code>leftAnchors</code>, and <code>rightAnchors</code> are then read from the
<code><a href="#topic+determinantStructure">determinantStructure</a></code> object. In other words: once a
<code><a href="#topic+determinantStructure">determinantStructure</a></code> has been generated, only <code>dat</code> and
<code><a href="#topic+determinantStructure">determinantStructure</a></code> have to be provided as argument to
generate a CIBER diamond plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Details are explained in Crutzen &amp; Peters (2017).
</p>


<h3>Value</h3>

<p>Depending on the value of <code>returnPlotOnly</code>, either the plot
only (a <code><a href="gtable.html#topic+gtable">gtable</a></code> object) or an object containing most objects
created along the way (in which case the plot is stored in
<code>$output$plot</code>).
</p>
<p>The plot has <code>width</code> and <code>height</code> attributes which can be used
when saving the plot.
</p>


<h3>References</h3>

<p>Crutzen, R., Peters, G.-J. Y., &amp; Noijen, J. (2017). How to
Select Relevant Social-Cognitive Determinants and Use them in the
Development of Behaviour Change Interventions? Confidence Interval-Based
Estimation of Relevance. http://dx.doi.org/
</p>


<h3>See Also</h3>

<p><code><a href="#topic+determinantStructure">determinantStructure</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### This example uses the determinant study Party Panel 17.1;
### see ?behaviorchange::BBC_data for more information.
data(BBC_pp17.1);
behaviorchange::CIBER(data=BBC_pp17.1,
                      determinants=c('epw_AttExpect_hearingDamage',
                                     'epw_AttExpect_highTone',
                                     'epw_AttExpect_musicVolume',
                                     'epw_AttExpect_musicFidelity',
                                     'epw_AttExpect_loudConversation',
                                     'epw_AttExpect_musicFocus',
                                     'epw_AttExpect_musicEnjoy'),
                      targets=c('epw_attitude'));

### With a binary target
data(BBC_pp17.1);
behaviorchange::binaryCIBER(data=BBC_pp17.1,
                            determinants=c('epGeneralBeliefs_loudnessPreference',
                                           'epGeneralBeliefs_loudnessGenre',
                                           'epGeneralBeliefs_loudnessTooMuch',
                                           'epGeneralBeliefs_priceFoam',
                                           'epGeneralBeliefs_priceSilicon',
                                           'epGeneralBeliefs_priceCustom'),
                            targets=c('epPossession'),
                            categoryLabels = c('no',
                                               'yes'));


</code></pre>

<hr>
<h2 id='CIBERlite'>CIBERlite</h2><span id='topic+CIBERlite'></span>

<h3>Description</h3>

<p>CIBERlite plots can be used to quickly get an idea of means and correlations
of a small number of determinants. They were developed to facilitate
conducting and interpreting determinant studies by prevention
professionals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CIBERlite(
  data,
  determinants,
  targets,
  determinantOrder = NULL,
  determinantLabels = NULL,
  subDeterminantLabels = NULL,
  title = NULL,
  conf.level = 0.95,
  scaleRange = NULL,
  determinantAesthetics = list(fill = "black", color = NA, alpha = 0.5),
  subDeterminantAesthetics = list(fill = "black", color = NA, alpha = 0.5),
  rDiamondAesthetics = list(fill = "#c4c4c4", color = NA, alpha = 0.75)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CIBERlite_+3A_data">data</code></td>
<td>
<p>The dataframe containing the variables.</p>
</td></tr>
<tr><td><code id="CIBERlite_+3A_determinants">determinants</code></td>
<td>
<p>Either a character vector with the names of the
determinants, or a list of named character vectors, where each vector
contains a number of subdeterminants, and each vector's name is the name
of the more proximal determinant (i.e. that 'contains' those subdeterminants).</p>
</td></tr>
<tr><td><code id="CIBERlite_+3A_targets">targets</code></td>
<td>
<p>A character vector with the names of the targets (i.e. more
proximal determinants, behavior, etc).</p>
</td></tr>
<tr><td><code id="CIBERlite_+3A_determinantorder">determinantOrder</code></td>
<td>
<p>The order in which to display the determinants (if
this needs to be different from the order as provided in <code>determinants</code>).</p>
</td></tr>
<tr><td><code id="CIBERlite_+3A_determinantlabels">determinantLabels</code></td>
<td>
<p>The labels to use for the determinants.</p>
</td></tr>
<tr><td><code id="CIBERlite_+3A_subdeterminantlabels">subDeterminantLabels</code></td>
<td>
<p>The labels to use for the subdeterminants.</p>
</td></tr>
<tr><td><code id="CIBERlite_+3A_title">title</code></td>
<td>
<p>The title of the plot.</p>
</td></tr>
<tr><td><code id="CIBERlite_+3A_conf.level">conf.level</code></td>
<td>
<p>The confidence levels: a list with two named values; the
confidence level for the means, named <code>means</code>, and the confidence level
for the associations, named <code>associations</code>.</p>
</td></tr>
<tr><td><code id="CIBERlite_+3A_scalerange">scaleRange</code></td>
<td>
<p>The full range of the scale of the
determinants/subdeterminants; the minimum and maximum values are used if
this is not provided.</p>
</td></tr>
<tr><td><code id="CIBERlite_+3A_determinantaesthetics">determinantAesthetics</code>, <code id="CIBERlite_+3A_subdeterminantaesthetics">subDeterminantAesthetics</code>, <code id="CIBERlite_+3A_rdiamondaesthetics">rDiamondAesthetics</code></td>
<td>
<p>The
aesthetics for the determinants, subdeterminants, and correlation diamonds,
each a list containing three named values: <code>fill</code>, <code>color</code>, and <code>alpha</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>More details will be provided in a forthcoming paper; until then, see
<a href="https://CIBERlite.com">https://CIBERlite.com</a>
</p>


<h3>Value</h3>

<p>A <code>ggplot</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### This example uses the determinant study Party Panel 15.1;
### see ?behaviorchange::BBC_data for more information.
data(BBC_pp15.1);
CIBERlite(data=BBC_pp15.1,
          determinants=c('highDose_attitude',
                         'highDose_perceivedNorm',
                         'highDose_pbc'),
          targets=c('highDose_intention'));

</code></pre>

<hr>
<h2 id='complecs'>Create a COMPLECS graph</h2><span id='topic+complecs'></span><span id='topic+print.complecs'></span>

<h3>Description</h3>

<p>COMPLECS was developed to help make sense of complex systems. It reads data from a
number of worksheets in a spreadsheet and generates a diagram according to those
specifications. Originally, COMPLECS was developed to visualise a problem during
the needs assessment phase of intervention development.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>complecs(
  input,
  title = "COMPLECS overview",
  layout = "fdp",
  graph_styling = list(c("outputorder", "edgesfirst", "graph"), c("overlap", "false",
    "graph"), c("fixedsize", "false", "node"), c("fontname", "Arial", "graph"),
    c("fontname", "Arial", "node"), c("fontname", "Arial", "edge"), c("headclip", "true",
    "edge"), c("tailclip", "false", "edge")),
  directed = TRUE,
  outputFile = NULL,
  outputWidth = 1600,
  outputHeight = NULL,
  returnDotOnly = FALSE,
  returnSvgOnly = FALSE,
  returnGraphOnly = TRUE,
  maxLabelLength = 20,
  regExReplacements = opts$get("diagrammerSanitization"),
  silent = opts$get("silent")
)

## S3 method for class 'complecs'
print(
  x,
  width = x$input$width,
  height = x$input$height,
  title = DiagrammeR::get_graph_name(x$output$graph),
  ...
)

## S3 method for class 'complecs'
print(
  x,
  width = x$input$width,
  height = x$input$height,
  title = DiagrammeR::get_graph_name(x$output$graph),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="complecs_+3A_input">input</code></td>
<td>
<p>Either a link to a Google Sheet, or a path to an Excel file.</p>
</td></tr>
<tr><td><code id="complecs_+3A_title">title</code></td>
<td>
<p>The title of the COMPLECS graph.</p>
</td></tr>
<tr><td><code id="complecs_+3A_layout">layout</code></td>
<td>
<p>The layout to use; has to be one of the <code>DiagrammeR</code> layout
types (<code>dot</code>, <code>neato</code>, <code>circo</code> and <code>twopi</code>).</p>
</td></tr>
<tr><td><code id="complecs_+3A_graph_styling">graph_styling</code></td>
<td>
<p>Additional styling to apply; a list with three-element
vectors, where the three elements correspond to, respectively, the <code>attr</code>,
<code>value</code>, and <code>attr_type</code> arguments for <code><a href="DiagrammeR.html#topic+add_global_graph_attrs">DiagrammeR::add_global_graph_attrs()</a></code>.
Note that these attributes may override attributes specified in the COMPLECS
specification.</p>
</td></tr>
<tr><td><code id="complecs_+3A_directed">directed</code></td>
<td>
<p>Whether to draw directed arrows or not.</p>
</td></tr>
<tr><td><code id="complecs_+3A_outputfile">outputFile</code></td>
<td>
<p>A character vector where each element is one path (including
filename) to write the graph to.</p>
</td></tr>
<tr><td><code id="complecs_+3A_outputwidth">outputWidth</code>, <code id="complecs_+3A_outputheight">outputHeight</code></td>
<td>
<p>If not <code>NULL</code>, a way to override the width and
height when calling <code>complecs</code> to generate a COMPLECS overview.</p>
</td></tr>
<tr><td><code id="complecs_+3A_returndotonly">returnDotOnly</code></td>
<td>
<p>Whether to only return the produced DOT code.</p>
</td></tr>
<tr><td><code id="complecs_+3A_returnsvgonly">returnSvgOnly</code></td>
<td>
<p>Whether to only return the SVG in a character vector.</p>
</td></tr>
<tr><td><code id="complecs_+3A_returngraphonly">returnGraphOnly</code></td>
<td>
<p>Whether to only return the produced graph.</p>
</td></tr>
<tr><td><code id="complecs_+3A_maxlabellength">maxLabelLength</code></td>
<td>
<p>The number of characters where to wrap the labels.</p>
</td></tr>
<tr><td><code id="complecs_+3A_regexreplacements">regExReplacements</code></td>
<td>
<p>A list of pairs of regular
expressions that will be applied to the specifications
before generating the ABCD. This can be used to sanitize
problematic characters (e.g. ', &quot; and \).</p>
</td></tr>
<tr><td><code id="complecs_+3A_silent">silent</code></td>
<td>
<p>Whether to be chatty or silent.</p>
</td></tr>
<tr><td><code id="complecs_+3A_x">x</code></td>
<td>
<p>The object to print (i.e. a result of a call to <code>complecs</code>).</p>
</td></tr>
<tr><td><code id="complecs_+3A_width">width</code>, <code id="complecs_+3A_height">height</code></td>
<td>
<p>If not <code>NULL</code>, a way to override the width and
height when calling <code>print</code> to print a COMPLECS overview.</p>
</td></tr>
<tr><td><code id="complecs_+3A_...">...</code></td>
<td>
<p>Any additional arguments for the <code><a href="base.html#topic+print">print()</a></code> method are passed
to <code><a href="DiagrammeR.html#topic+render_graph">DiagrammeR::render_graph()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>COMPLECS is a recursive acronym for COMPLECS Organises Multiple Players &amp; Linked
Environments using Connected Specifications.
</p>


<h3>Value</h3>

<p>A <code>complecs</code> object that includes the graph and the graph in SVG in
<code>output$graph</code> and <code>output$graphSvg</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
### Path in the package with example COMPLECS
exampleCOMPLECS &lt;-
  system.file(
    "extdata",
    "COMPLECS-spec-example.xlsx",
    package = "behaviorchange"
  );

behaviorchange::complecs(
  exampleCOMPLECS
);

### Loading that COMPLECS from a google sheet - but note that
### this requires an internet connection!
behaviorchange::complecs(
  paste0(
    "https://docs.google.com/spreadsheets/d/",
    "1WMO15xroy4a0RfpuZ8GhT-NfDoxwS34w9PrWp8rGjjk"
  )
);

## End(Not run)
</code></pre>

<hr>
<h2 id='complecs_to_precede'>Represent a COMPLECS specification as a PRECEDE model</h2><span id='topic+complecs_to_precede'></span>

<h3>Description</h3>

<p>This function reads in a complecs specification and draw a PRECEDE model,
with a number of assumptions (see Details section).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>complecs_to_precede(
  input,
  title = "PRECEDE diagram",
  layout = "fdp",
  graph_styling = list(c("outputorder", "edgesfirst", "graph"), c("rankdir", "LR",
    "graph"), c("overlap", "false", "graph"), c("fixedsize", "false", "node"),
    c("fontname", "Arial", "graph"), c("fontname", "Arial", "node"), c("fillcolor",
    "White", "node"), c("shape", "box", "node"), c("style", "filled", "node"),
    c("fontname", "Arial", "edge"), c("headclip", "true", "edge"), c("tailclip", "false",
    "edge")),
  directed = TRUE,
  outputFile = NULL,
  outputWidth = 1600,
  outputHeight = NULL,
  returnDotOnly = FALSE,
  returnSvgOnly = FALSE,
  returnGraphOnly = TRUE,
  maxLabelLength = 60,
  regExReplacements = opts$get("diagrammerSanitization"),
  silent = opts$get("silent")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="complecs_to_precede_+3A_input">input</code></td>
<td>
<p>Either a link to a Google Sheet, or a path to an Excel file.</p>
</td></tr>
<tr><td><code id="complecs_to_precede_+3A_title">title</code></td>
<td>
<p>The title of the COMPLECS graph.</p>
</td></tr>
<tr><td><code id="complecs_to_precede_+3A_layout">layout</code></td>
<td>
<p>The layout to use; has to be one of the <code>DiagrammeR</code> layout
types (<code>dot</code>, <code>neato</code>, <code>circo</code> and <code>twopi</code>).</p>
</td></tr>
<tr><td><code id="complecs_to_precede_+3A_graph_styling">graph_styling</code></td>
<td>
<p>Additional styling to apply; a list with three-element
vectors, where the three elements correspond to, respectively, the <code>attr</code>,
<code>value</code>, and <code>attr_type</code> arguments for <code><a href="DiagrammeR.html#topic+add_global_graph_attrs">DiagrammeR::add_global_graph_attrs()</a></code>.
Note that these attributes may override attributes specified in the COMPLECS
specification.</p>
</td></tr>
<tr><td><code id="complecs_to_precede_+3A_directed">directed</code></td>
<td>
<p>Whether to draw directed arrows or not.</p>
</td></tr>
<tr><td><code id="complecs_to_precede_+3A_outputfile">outputFile</code></td>
<td>
<p>A character vector where each element is one path (including
filename) to write the graph to.</p>
</td></tr>
<tr><td><code id="complecs_to_precede_+3A_outputwidth">outputWidth</code>, <code id="complecs_to_precede_+3A_outputheight">outputHeight</code></td>
<td>
<p>If not <code>NULL</code>, a way to override the width and
height when calling <code>complecs</code> to generate a COMPLECS overview.</p>
</td></tr>
<tr><td><code id="complecs_to_precede_+3A_returndotonly">returnDotOnly</code></td>
<td>
<p>Whether to only return the produced DOT code.</p>
</td></tr>
<tr><td><code id="complecs_to_precede_+3A_returnsvgonly">returnSvgOnly</code></td>
<td>
<p>Whether to only return the SVG in a character vector.</p>
</td></tr>
<tr><td><code id="complecs_to_precede_+3A_returngraphonly">returnGraphOnly</code></td>
<td>
<p>Whether to only return the produced graph.</p>
</td></tr>
<tr><td><code id="complecs_to_precede_+3A_maxlabellength">maxLabelLength</code></td>
<td>
<p>The number of characters where to wrap the labels.</p>
</td></tr>
<tr><td><code id="complecs_to_precede_+3A_regexreplacements">regExReplacements</code></td>
<td>
<p>A list of pairs of regular
expressions that will be applied to the specifications
before generating the ABCD. This can be used to sanitize
problematic characters (e.g. ', &quot; and \).</p>
</td></tr>
<tr><td><code id="complecs_to_precede_+3A_silent">silent</code></td>
<td>
<p>Whether to be chatty or silent.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Only entities with the following entity types are used from the COMPLECS
specification:
</p>

<ul>
<li> <p><code>person</code>
</p>
</li>
<li> <p><code>organization</code>
</p>
</li>
<li> <p><code>environmental_condition</code>
</p>
</li>
<li> <p><code>behavior</code>
</p>
</li>
<li> <p><code>determinant</code>
</p>
</li>
<li> <p><code>outcome</code>
</p>
</li></ul>

<p>Furthermore, it will be assumed that the only direct connections from
<code>behavior</code> entities to <code>outcome</code> entities belong to the focal population;
therefore, if behaviors of environmental actors are important for an
outcome, those behaviors' effects must be represented as
<code>environmental_condition</code> entities - otherwise the relevant <code>person</code>s or
<code>organizations</code>s will be erroneously considered as focal population members.
</p>


<h3>Value</h3>

<p>A <code>complecs</code> object that includes the graph and the graph in SVG in
<code>output$graph</code> and <code>output$graphSvg</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
### Path in the package with example COMPLECS
exampleCOMPLECS &lt;-
  system.file(
    "extdata",
    "COMPLECS-spec-example.xlsx",
    package = "behaviorchange"
  );

behaviorchange::complecs_to_precede(
  exampleCOMPLECS
);

### Loading that COMPLECS from a google sheet - but note that
### this requires an internet connection!
behaviorchange::complecs_to_precede(
  paste0(
    "https://docs.google.com/spreadsheets/d/",
    "1WMO15xroy4a0RfpuZ8GhT-NfDoxwS34w9PrWp8rGjjk"
  )
);

## End(Not run)
</code></pre>

<hr>
<h2 id='convert.threshold.to.er'>Visualising Numbers Needed for Change</h2><span id='topic+convert.threshold.to.er'></span><span id='topic+convert.er.to.threshold'></span><span id='topic+erDataSeq'></span><span id='topic+nncvis'></span><span id='topic+ggNNC'></span>

<h3>Description</h3>

<p>These functions can be used to visualise Numbers Needed for Change (or
Numbers Needed to Treat).
<code>erDataSeq</code> is a helper function to generate an Event Rate Data
Sequence, and it uses <code>convert.threshold.to.er</code> and
<code>convert.er.to.threshold</code> to convert thresholds to event rates and vice
versa.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert.threshold.to.er(
  threshold,
  mean,
  sd,
  eventIfHigher = TRUE,
  pdist = stats::pnorm
)

convert.er.to.threshold(
  er,
  mean,
  sd,
  eventIfHigher = TRUE,
  qdist = stats::qnorm
)

erDataSeq(
  er = NULL,
  threshold = NULL,
  mean = NULL,
  sd = NULL,
  eventIfHigher = TRUE,
  pRange = c(1e-06, 0.99999),
  xStep = 0.01
)

ggNNC(
  cerDataSeq,
  d = NULL,
  eventDesirable = TRUE,
  r = 1,
  xlab = "Continuous outcome",
  plotTitle = c("Numbers Needed for Change = ", ""),
  theme = ggplot2::theme_bw(),
  lineSize = 1,
  cerColor = "#EBF2F8",
  eerColor = "#172F47",
  cerLineColor = "#888888",
  eerLineColor = "#000000",
  dArrowColor = "#000000",
  cerAlpha = 0.66,
  eerAlpha = 0.66,
  xLim = NULL,
  xLimAutoDensityTolerance = 0.001,
  showLegend = TRUE,
  verticalLineColor = "#172F47",
  desirableColor = "#00FF00",
  desirableAlpha = 0.2,
  undesirableColor = "#FF0000",
  undesirableAlpha = 0.2,
  desirableTextColor = "#009900",
  undesirableTextColor = "#990000",
  dArrowDistance = 0.04 * max(cerDataSeq$density),
  dLabelDistance = 0.08 * max(cerDataSeq$density)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert.threshold.to.er_+3A_threshold">threshold</code></td>
<td>
<p>If the event rate is not available, a threshold value can
be specified instead, which is then used in conjunction with the mean
(<code>mean</code>) and standard deviation (<code>sd</code>) and assuming a normal
distribution to compute the event rate.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_mean">mean</code></td>
<td>
<p>The mean of the control group distribution.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_sd">sd</code></td>
<td>
<p>The standard deviation (of the control distribution, but assumed
to be the same for both distributions).</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_eventifhigher">eventIfHigher</code></td>
<td>
<p>Whether scores above or below the threshold are
considered 'an event'.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_pdist">pdist</code>, <code id="convert.threshold.to.er_+3A_qdist">qdist</code></td>
<td>
<p>Distributions to use when converting thresholds to event
rates and vice versa; defaults to the normal distribution.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_er">er</code></td>
<td>
<p>Event rate to visualise (or convert).</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_prange">pRange</code></td>
<td>
<p>The range of probabilities for which to so the distribution.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_xstep">xStep</code></td>
<td>
<p>Precision of the drawn distribution; higher values mean lower
precision/granularity/resolution.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_cerdataseq">cerDataSeq</code></td>
<td>
<p>The <code>cerDataSeq</code> object.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_d">d</code></td>
<td>
<p>The value of Cohen's <em>d</em>.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_eventdesirable">eventDesirable</code></td>
<td>
<p>Whether an event is desirable or undesirable.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_r">r</code></td>
<td>
<p>The correlation between the determinant and behavior (for mediated
NNC's).</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_xlab">xlab</code></td>
<td>
<p>The label to display for the X axis.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_plottitle">plotTitle</code></td>
<td>
<p>The title of the plot; either one character value, this
value if used; if two, they are considered a prefix and suffix to be
pre/appended to the NNC value.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_theme">theme</code></td>
<td>
<p>The theme to use for the plot.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_linesize">lineSize</code></td>
<td>
<p>The thickness of the lines in the plot.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_cercolor">cerColor</code></td>
<td>
<p>The color to use for the event rate portion of the control
group distribution.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_eercolor">eerColor</code></td>
<td>
<p>The color to use for the event rate portion of the
experimental group distribution.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_cerlinecolor">cerLineColor</code></td>
<td>
<p>The line color to use for the control group
distribution.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_eerlinecolor">eerLineColor</code></td>
<td>
<p>The line color to use for the experimental group
distribution.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_darrowcolor">dArrowColor</code></td>
<td>
<p>The color of the arrow to show the effect size.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_ceralpha">cerAlpha</code></td>
<td>
<p>The alpha value (transparency) to use for the control group
distribution.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_eeralpha">eerAlpha</code></td>
<td>
<p>The alpha value (transparency) to use for the control group
distribution.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_xlim">xLim</code></td>
<td>
<p>This can be used to manually specify the limits for the X axis;
if <code>NULL</code>, sensible limits will be derived using
<code>xLimAutoDensityTolerance</code>.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_xlimautodensitytolerance">xLimAutoDensityTolerance</code></td>
<td>
<p>If <code>xLim</code> is <code>NULL</code>, the limits
will be set where the density falls below this proportion of its maximum
value.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_showlegend">showLegend</code></td>
<td>
<p>Whether to show the legend (only if showing two
distributions).</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_verticallinecolor">verticalLineColor</code></td>
<td>
<p>The color of the vertical line used to indicate the
threshold.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_desirablecolor">desirableColor</code></td>
<td>
<p>The color for the desirable portion of the X axis.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_desirablealpha">desirableAlpha</code></td>
<td>
<p>The alpha for the desirable portion of the X axis.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_undesirablecolor">undesirableColor</code></td>
<td>
<p>The color for the undesirable portion of the X axis.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_undesirablealpha">undesirableAlpha</code></td>
<td>
<p>The color for the undesirable portion of the X axis.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_desirabletextcolor">desirableTextColor</code></td>
<td>
<p>The color for the text to indicate the desirable
portion of the X axis.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_undesirabletextcolor">undesirableTextColor</code></td>
<td>
<p>The color for the text to indicate the
undesirable portion of the X axis.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_darrowdistance">dArrowDistance</code></td>
<td>
<p>The distance of the effect size arrow from the top of
the distributions.</p>
</td></tr>
<tr><td><code id="convert.threshold.to.er_+3A_dlabeldistance">dLabelDistance</code></td>
<td>
<p>The distance of the effect size label from the top of
the distributions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions are used by <code><a href="#topic+nnc">nnc()</a></code> to show the distributions, and
event rates. They probably won't be used much on their own.
</p>


<h3>Value</h3>

<p><code>erDataSeq</code> returns a data sequence; <code>ggNNC</code> a
<code><a href="ggplot2.html#topic+ggplot">ggplot2::ggplot()</a></code>.
</p>


<h3>Author(s)</h3>

<p>Gjalt-Jorn Peters &amp; Stefan Gruijters
</p>
<p>Maintainer: Gjalt-Jorn Peters <a href="mailto:gjalt-jorn@userfriendlyscience.com">gjalt-jorn@userfriendlyscience.com</a>
</p>


<h3>References</h3>

<p>Gruijters, S. L., &amp; Peters, G. Y. (2019). Gauging the
impact of behavior change interventions: A tutorial on the Numbers
Needed to Treat. <em>PsyArXiv.</em> <a href="https://doi.org/10.31234/osf.io/2bau7">doi:10.31234/osf.io/2bau7</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+nnc">nnc()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
### Show distribution for an event rate value of 125
behaviorchange::ggNNC(behaviorchange::erDataSeq(threshold=125, mean=90, sd=30));

### If the event occurs under the threshold instead of
### above it
behaviorchange::ggNNC(behaviorchange::erDataSeq(threshold=125,
                                                mean=90, sd=30,
                      eventIfHigher = FALSE));

### ... And for undesirable events (note how
### desirability is an argument for ggNNC, whereas
### whether an event occurs 'above' or 'below' the
### threshold is an argument for erDataSeq):
behaviorchange::ggNNC(behaviorchange::erDataSeq(threshold=125,
                                                mean=90, sd=30,
                      eventIfHigher = FALSE),
      eventDesirable = FALSE);

### Show event rate for both experimental and
### control conditions, and show the numbers
### needed for change
behaviorchange::ggNNC(behaviorchange::erDataSeq(threshold=125,
                                                mean=90, sd=30),
                      d=.5);

### Illustration of how even with very large effect
### sizes, if the control event rate is very high,
### you'll still need a high number of NNC
behaviorchange::ggNNC(behaviorchange::erDataSeq(er=.9),
                      d=1);

</code></pre>

<hr>
<h2 id='determinant_selection_table'>Potential for Change Index and the Determinant Selection Table</h2><span id='topic+determinant_selection_table'></span><span id='topic+determinantSelectionTable_partial'></span><span id='topic+knit_print.determinantSelectionTable'></span><span id='topic+print.determinantSelectionTable'></span><span id='topic+potential_for_change_index'></span><span id='topic+P_delta'></span><span id='topic+intervention_potential'></span><span id='topic+room_for_improvement'></span>

<h3>Description</h3>

<p>These functions compute the Potential for Change Index for one or multiple
(sub-)determinants, the room for improvement (an intermediate estimate),
and produce a convenient table with an overview of all (sub-)determinants.
Note that for determinant selection purposes, quantitative estimates such
as the Potential for Change Index should never be used without also
thoroughly inspecting the visualisations of the univariate distributions
and the confidence intervals for the associations to the ultimate
intervention targets (usually the target behavior or a proxy measure). For
this purpose, the Confidence Interval-Based Estimation of Relevance plots
can be used (see <code><a href="#topic+CIBER">CIBER()</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>determinant_selection_table(
  data,
  determinants,
  target,
  determinantLabels = NULL,
  targetLabel = NULL,
  sortBy = NULL,
  sortByAbs = TRUE,
  decreasing = TRUE,
  digits = 3,
  increasesAreImprovements = TRUE,
  minimum = base::min,
  maximum = base::max,
  center = base::mean,
  weight = stats::cor,
  type = NULL,
  minimumArgs = list(na.rm = TRUE),
  maximumArgs = list(na.rm = TRUE),
  centerArgs = list(na.rm = TRUE),
  weightArgs = list(use = "complete.obs"),
  potentialScale = NULL,
  headingLevel = 3,
  output = behaviorchange::opts$get("tableOutput")
)

determinantSelectionTable_partial(
  x,
  digits = attr(x, "digits"),
  headingLevel = attr(x, "headingLevel"),
  echoPartial = FALSE,
  partialFile = NULL,
  quiet = TRUE,
  ...
)

## S3 method for class 'determinantSelectionTable'
knit_print(
  x,
  digits = attr(x, "digits"),
  headingLevel = attr(x, "headingLevel"),
  echoPartial = FALSE,
  partialFile = NULL,
  quiet = TRUE,
  ...
)

## S3 method for class 'determinantSelectionTable'
print(
  x,
  digits = attr(x, "digits"),
  headingLevel = attr(x, "headingLevel"),
  output = attr(x, "output"),
  forceKnitrOutput = FALSE,
  ...
)

potential_for_change_index(
  data,
  determinants,
  target,
  increasesAreImprovements = TRUE,
  sampleLevel = FALSE,
  minimum = base::min,
  maximum = base::max,
  center = base::mean,
  weight = stats::cor,
  type = NULL,
  minimumArgs = list(na.rm = TRUE),
  maximumArgs = list(na.rm = TRUE),
  centerArgs = list(na.rm = TRUE),
  weightArgs = list(use = "complete.obs")
)

room_for_improvement(
  x,
  increasesAreImprovements = TRUE,
  sampleLevel = FALSE,
  minimum = base::min,
  maximum = base::max,
  center = base::mean,
  minimumArgs = list(na.rm = TRUE),
  maximumArgs = list(na.rm = TRUE),
  centerArgs = list(na.rm = TRUE),
  varName = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="determinant_selection_table_+3A_data">data</code></td>
<td>
<p>The dataframe containing the variables.</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_determinants">determinants</code></td>
<td>
<p>The name(s) of the determinant(s).</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_target">target</code></td>
<td>
<p>The target (e.g. behavior or intention).</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_determinantlabels">determinantLabels</code>, <code id="determinant_selection_table_+3A_targetlabel">targetLabel</code></td>
<td>
<p>Optionally, labels to use for the
(sub-)determinants and the target. The <code>determinantLabels</code> must have the
same order as the <code>determinants</code> vector.</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_sortby">sortBy</code></td>
<td>
<p>The column to sort the results by; if not <code>NULL</code>, a number
from 1-6 that corresponds to the six columns of the Determinant Selection
Table.</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_sortbyabs">sortByAbs</code></td>
<td>
<p>Whether to sort by raw values (<code>FALSE</code>) or their
absolute value (<code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_decreasing">decreasing</code></td>
<td>
<p>Whether to sort in decreasing (<code>TRUE</code>) or increasing
(<code>FALSE</code>) order.</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_digits">digits</code></td>
<td>
<p>The number of digits to round to.</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_increasesareimprovements">increasesAreImprovements</code></td>
<td>
<p>Whether increases are improvements (<code>TRUE</code>)
or decreases are improvements (<code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_minimum">minimum</code>, <code id="determinant_selection_table_+3A_maximum">maximum</code></td>
<td>
<p>The minimum and maximum, as functions that take
a vector and return the minimum and maximum scores, as numbers, or as
vectors of numbers specifying the minimum and maximum to use for each
column in <code>x</code> or <code>determinants</code>, or a lists of functions specifying the functions to use
for each column in <code>x</code> or <code>determinants</code>.</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_center">center</code></td>
<td>
<p>For the sample-level version, a function that takes a
vector and returns the center (e.g. mean, median, etc), or a list of
functions specifying the function to use for each column in <code>x</code> or
<code>determinants</code>.</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_weight">weight</code></td>
<td>
<p>The function to return the weight/multiplier to use in the
computation.</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_type">type</code></td>
<td>
<p>The type of potential for change index. Currently implemented
are type <code>'1'</code> and type <code>'2'</code> - see details for more information.</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_minimumargs">minimumArgs</code>, <code id="determinant_selection_table_+3A_maximumargs">maximumArgs</code>, <code id="determinant_selection_table_+3A_centerargs">centerArgs</code>, <code id="determinant_selection_table_+3A_weightargs">weightArgs</code></td>
<td>
<p>lists with arguments
to pass to the corresponding functions. Note that these are not vectorized.</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_potentialscale">potentialScale</code></td>
<td>
<p>The scale with minimum and maximum possible values
for the Potential for Change index. If <code>NULL</code>, the minimum is set to <code>0</code>
and the maximum is set to the highest observed value.</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_headinglevel">headingLevel</code></td>
<td>
<p>The number of hashes to print in front of the headings
when printing while knitting</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_output">output</code></td>
<td>
<p>Whether to only output to the viewer (if possible;
<code>output='viewer'</code>), or only to the console (<code>output='console'</code>), or to
both (<code>output=c('viewer', 'console')</code>). Note that displaying in the viewer
requires the <code>htmltools</code> package.</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_x">x</code></td>
<td>
<p>For room for improvement, either a numeric vector with scores on
a (sub-)determinant, or a data frame with multiple such vectors. For the
Determinant Selection Table functions, the object to print/knit.</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_echopartial">echoPartial</code></td>
<td>
<p>Whether to show the executed code in the R Markdown
partial (<code>TRUE</code>) or not (<code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_partialfile">partialFile</code></td>
<td>
<p>This can be used to specify a custom partial file. The
file will have object <code>x</code> available.</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_quiet">quiet</code></td>
<td>
<p>Passed on to <code><a href="knitr.html#topic+knit">knitr::knit()</a></code> whether it should e
chatty (<code>FALSE</code>) or quiet (<code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_...">...</code></td>
<td>
<p>Any additional arguments are passed to the default print method
by the print method, and to <code><a href="rmdpartials.html#topic+partial">rmdpartials::partial()</a></code> when knitting a
RMarkdown partial.</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_forceknitroutput">forceKnitrOutput</code></td>
<td>
<p>Force knitr output.</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_samplelevel">sampleLevel</code></td>
<td>
<p>Whether to return sample-level estimates (<code>TRUE</code>) or
individual-level estimates (<code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="determinant_selection_table_+3A_varname">varName</code></td>
<td>
<p>For internal use.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Potential for Change index was developed by Keegan et al. and is a
numerical representation of a number of important features in <code><a href="#topic+CIBER">CIBER()</a></code>
plots (for more details, please see the references below). It turned out
a similar measure, the Intervention Potential, was developed by Huber &amp;
Mosler (2013). The latter uses regression coefficients as weights, which
is problematic for a number of reasons (see Crutzen, Peters &amp; Noijen, 2017),
and has therefore not been implemented as a default, but it is possible to
use regression coefficients by specifying a custom weight function.
</p>
<p>The original Potential for Change Index was conceptualized to optimize
intervention tailoring and improve the prediction of individual-level
intervention effectiveness. A second conceptualization of the Potential
for Change Index can facilitate sub-determinant selection.
</p>
<p>In addition to using the <code>minimum</code>, <code>maximum</code>, <code>center</code>, and <code>weight</code>
functions to specify custom functions, specific types have also been
implemented to quickly use a prespecified combination of functions.
</p>
<p>The first (<code>type = '1'</code>) is computed as follows:
</p>

<ul>
<li><p> For sub-determinants with a positive zero-order correlation with
behavior, the sample mean is subtracted from the observed maximum score,
and the result is multiplied by the zero-order correlation;
</p>
</li>
<li><p> For sub-determinants with a negative zero-order correlation with
behavior, the sample mean is subtracted from the observed minimum score,
and the result is multiplied by the zero-order correlation.
</p>
</li></ul>

<p>The second (<code>type = '2'</code>) is computed as follows:
</p>

<ul>
<li><p> For sub-determinants with a positive zero-order correlation with
behavior, the sample mean is subtracted from the .95 quantile of the
scores, and the result is multiplied by the squared zero-order
correlation (i.e. the proportion of explained variance);
</p>
</li>
<li><p> For sub-determinants with a negative zero-order correlation with
behavior, the sample mean is subtracted from the .05 quantile of the
scores, and the result is multiplied by the squared zero-order
correlation (i.e. the proportion of explained variance);
</p>
</li></ul>

<p>The second variant effectively takes the 5% trimmed maximum and minimum,
rendering it less sensitive to outliers, penalizes weak associations with
behavior more severely, and decreases sensitivity to differences between
correlations. These differences should render the second variant a bit more
robust over different samples.
</p>
<p>The room for improvement is one of the ingredients of the Potential
for Change Index or P_delta, a generalized version of the Intervention
Potential. The Determinant Selection Table efficiently presents the
Potential for Change Indices for a set of (sub-)determinants.
</p>


<h3>Value</h3>

<p>For the individual-level version, a vector or data
frame with the same dimensions as provided; for the sample-level version, if
a vector is provided, a single number, and if a data frame is provided, a
vector with as many values as the data frame has columns. For
Determinant Selection Table, a data frame.
</p>


<h3>References</h3>

<p>Knittle, K. P., Peters, G.-J. Y., Heino, M. T. J., Tobias, R., &amp;
Hankonen, N. (2019). Potential for change: New metrics for tailoring and
predicting response to behavior change interventions. <a href="https://doi.org/10/ghqmg3">doi:10/ghqmg3</a>
</p>
<p>Huber, A. C. &amp; Mosler, H.-J. (2013) Determining behavioral factors for
interventions to increase safe water consumption: a cross-sectional field
study in rural Ethiopia, International Journal of Environmental Health
Research, 23:2, 96-107 <a href="https://doi.org/10.1080/09603123.2012.699032">doi:10.1080/09603123.2012.699032</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Get example data
dat &lt;- get(data("BBC_pp15.1", package="behaviorchange"));

### Individual-level version, for one sub-determinant
P_delta_example &lt;-
  behaviorchange::potential_for_change_index(
    data=dat,
    determinants='highDose_attitude',
    target='highDose_intention'
  );

head(P_delta_example);
hist(P_delta_example);

### Sample-level version
behaviorchange::potential_for_change_index(
  data=dat,
  determinants='highDose_attitude',
  target='highDose_intention',
  sampleLevel = TRUE
);

### Individual-level for multiple determinants
P_delta_example &lt;-
  behaviorchange::potential_for_change_index(
    data=dat,
    determinants=c('highDose_attitude', 'highDose_perceivedNorm'),
    target='highDose_intention'
  );
head(P_delta_example);

### Sample-level version for multiple determinants
behaviorchange::potential_for_change_index(
  data=dat,
  determinants=c('highDose_attitude', 'highDose_perceivedNorm'),
  target='highDose_intention',
  sampleLevel = TRUE
);

### Get the Potential for Change Index Type 2
behaviorchange::potential_for_change_index(
  data=dat,
  determinants=c('highDose_attitude', 'highDose_perceivedNorm'),
  target='highDose_intention',
  type = '2',
  sampleLevel = TRUE
);

### Get a Determinant Selection Table
behaviorchange::determinant_selection_table(
  data=dat,
  determinants = c('highDose_AttBeliefs_long',
                   'highDose_AttBeliefs_intensity',
                   'highDose_AttBeliefs_euphoria'),
  target = 'highDose_intention',
  sortBy = 6
);

### R Markdown partials can smoothly be included in RMarkdown documents
behaviorchange::determinantSelectionTable_partial(
  behaviorchange::determinant_selection_table(
    data=dat,
    determinants = c('highDose_AttBeliefs_long',
                     'highDose_AttBeliefs_intensity',
                     'highDose_AttBeliefs_euphoria'),
    target = 'highDose_intention',
    sortBy = 6
  )
);

### Room for improvement for one variable
head(
  room_for_improvement(
    dat$highDose_AttBeliefs_long
  )
);

room_for_improvement(
  dat$highDose_AttBeliefs_long,
  sampleLevel = TRUE
);

### For multiple (sub-)determinants
head(
  room_for_improvement(
    dat[, c('highDose_AttBeliefs_long',
            'highDose_AttBeliefs_intensity',
            'highDose_AttBeliefs_euphoria')]
  )
);

room_for_improvement(
  dat[, c('highDose_AttBeliefs_long',
          'highDose_AttBeliefs_intensity',
          'highDose_AttBeliefs_euphoria')],
  sampleLevel = TRUE
);

</code></pre>

<hr>
<h2 id='determinantStructure'>Determinant Structure specification</h2><span id='topic+determinantStructure'></span><span id='topic+determinantVar'></span><span id='topic+subdeterminants'></span><span id='topic+subdeterminantProducts'></span><span id='topic+plot.determinantStructure'></span><span id='topic+print.determinantStructure'></span>

<h3>Description</h3>

<p>These functions can be used to specify a determinant structure: a
hierarchical structure of determinants that can then be
conveniently plotted and analysed, for example using
<a href="#topic+detStructCIBER">detStructCIBER</a>. These functions are made to be used together;
see the example and the forthcoming article for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>determinantStructure(name, selection = NULL, ...)

determinantVar(name, selection = NULL, ...)

subdeterminants(name, selection = NULL, ...)

subdeterminantProducts(name, selection = NULL, ...)

## S3 method for class 'determinantStructure'
plot(x, useDiagrammeR = FALSE, ...)

## S3 method for class 'determinantStructure'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="determinantStructure_+3A_name">name</code></td>
<td>
<p>The name of the variable that is specified.</p>
</td></tr>
<tr><td><code id="determinantStructure_+3A_selection">selection</code></td>
<td>
<p>A regular expression to use to select the
variables in a dataframe that are considered items that
together form this variable. For <code>determinantStructure</code>, a
list can be provided that also contains a named regular
expression with the name 'behaviorRegEx', which specifies
the name of the behavior to which this determinant structure
pertains.</p>
</td></tr>
<tr><td><code id="determinantStructure_+3A_...">...</code></td>
<td>
<p>Any additional arguments are other determinant structure
building functions. These are used to construct the determinant
structure 'tree'.</p>
</td></tr>
<tr><td><code id="determinantStructure_+3A_x">x</code></td>
<td>
<p>The <code>determinantStructure</code> object to print or plot.</p>
</td></tr>
<tr><td><code id="determinantStructure_+3A_usediagrammer">useDiagrammeR</code></td>
<td>
<p>Whether to simply use <code>print(plot(x))</code>
(if <code>FALSE</code>) or whether to use <a href="data.tree.html#topic+ToDiagrammeRGraph">data.tree::ToDiagrammeRGraph</a>,
tweak it a bit, by setting global graph attributes, and then
using <a href="DiagrammeR.html#topic+render_graph">DiagrammeR::render_graph</a> (if <code>TRUE</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This family of functions will be explained more in detail in a
forthcoming paper.
</p>
<p><code>plot</code> and <code>print</code> methods plot and print a
<code>determinantStructure</code> object.
</p>


<h3>Value</h3>

<p>A <code>determinantStructure</code> object, which is a
data.tree object.
</p>


<h3>Author(s)</h3>

<p>Gjalt-Jorn Peters, <a href="mailto:gjalt-jorn@a-bc.eu">gjalt-jorn@a-bc.eu</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+detStructAddVarLabels">detStructAddVarLabels</a></code>,
<code><a href="#topic+detStructAddVarNames">detStructAddVarNames</a></code>,
<code><a href="#topic+detStructComputeProducts">detStructComputeProducts</a></code>,
<code><a href="#topic+detStructComputeScales">detStructComputeScales</a></code>,
<code><a href="#topic+detStructCIBER">detStructCIBER</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>determinantStructure('using R',
                     list('using R',
                          behaviorRegEx = 'some RegEx'),
                     determinantVar("Intention",
                                    "another RegEx",
                                    determinantVar("Attitude",
                                                   "third RegEX",
                                                   subdeterminants("Likelihood",
                                                                   "4th RegEx"),
                                                  subdeterminants("Evaluation",
                                                                  "5th RegEx"),
                                                  subdeterminantProducts("attProduct",
                                                                         c("4th RegEx",
                                                                           "5th RegEx"))),
                                    determinantVar("perceivedNorm",
                                                   "6th RegEx",
                                                   subdeterminants("Approval",
                                                                   "7th RegEx"),
                                                   subdeterminants("Motivation to comply",
                                                                   "8th RegEx"),
                                                   subdeterminantProducts("normProduct",
                                                                          c("7th RegEx",
                                                                            "8th RegEx"))),
                                              determinantVar("pbc",
                                                             "9th RegEx",
                                                             subdeterminants("Control beliefs",
                                                                             "10th RegEx"))));

</code></pre>

<hr>
<h2 id='detStructAddVarLabels'>Functions to preprocess determinant structures</h2><span id='topic+detStructAddVarLabels'></span><span id='topic+detStructPreprocessing'></span><span id='topic+detStructAddVarNames'></span><span id='topic+detStructComputeProducts'></span><span id='topic+detStructComputeScales'></span>

<h3>Description</h3>

<p>These functions are used in conjunction with the
<code><a href="#topic+determinantStructure">determinantStructure</a></code> family of funtions to conveniently work
with determinant structures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>detStructAddVarLabels(
  determinantStructure,
  varLabelDf,
  varNameCol = "varNames.cln",
  leftAnchorCol = "leftAnchors",
  rightAnchorCol = "rightAnchors",
  subQuestionCol = "subQuestions",
  questionTextCol = "questionText"
)

detStructAddVarNames(determinantStructure, names)

detStructComputeProducts(determinantStructure, data, append = TRUE)

detStructComputeScales(
  determinantStructure,
  data,
  append = TRUE,
  separator = "_"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="detStructAddVarLabels_+3A_determinantstructure">determinantStructure</code></td>
<td>
<p>The <code><a href="#topic+determinantStructure">determinantStructure</a></code> object.</p>
</td></tr>
<tr><td><code id="detStructAddVarLabels_+3A_varlabeldf">varLabelDf</code></td>
<td>
<p>The variable label dataframe as generated by the
<code>processLSvarLabels</code> in the <code>userfriendlyscience</code> package.  It is also possible
to specify a 'homemade' dataframe, in which case the column names have to
specified (see the next arguments).</p>
</td></tr>
<tr><td><code id="detStructAddVarLabels_+3A_varnamecol">varNameCol</code></td>
<td>
<p>The name of the column of the <code>varLabelDf</code> that
contains the variable name. Only needs to be changed from the default value
if <code>varLabelDf</code> is not a dataframe as produced by
<code>processLSvarLabels</code>.</p>
</td></tr>
<tr><td><code id="detStructAddVarLabels_+3A_leftanchorcol">leftAnchorCol</code></td>
<td>
<p>The name of the column of the <code>varLabelDf</code> that
contains the left anchor. Only needs to be changed from the default value if
<code>varLabelDf</code> is not a dataframe as produced by
<code>processLSvarLabels</code>.</p>
</td></tr>
<tr><td><code id="detStructAddVarLabels_+3A_rightanchorcol">rightAnchorCol</code></td>
<td>
<p>The name of the column of the <code>varLabelDf</code> that
contains the right anchor. Only needs to be changed from the default value
if <code>varLabelDf</code> is not a dataframe as produced by
<code>processLSvarLabels</code>.</p>
</td></tr>
<tr><td><code id="detStructAddVarLabels_+3A_subquestioncol">subQuestionCol</code></td>
<td>
<p>The name of the column of the <code>varLabelDf</code> that
contains the subquestion. Only needs to be changed from the default value if
<code>varLabelDf</code> is not a dataframe as produced by
<code>processLSvarLabels</code>.</p>
</td></tr>
<tr><td><code id="detStructAddVarLabels_+3A_questiontextcol">questionTextCol</code></td>
<td>
<p>The name of the column of the <code>varLabelDf</code> that
contains the question text. Only needs to be changed from the default value
if <code>varLabelDf</code> is not a dataframe as produced by
<code>processLSvarLabels</code>.</p>
</td></tr>
<tr><td><code id="detStructAddVarLabels_+3A_names">names</code></td>
<td>
<p>A character vector with the variable names. These are matched
against the regular expressions as specified in the
<code><a href="#topic+determinantStructure">determinantStructure</a></code> object, and any matches will be stored in
the <code><a href="#topic+determinantStructure">determinantStructure</a></code> object.</p>
</td></tr>
<tr><td><code id="detStructAddVarLabels_+3A_data">data</code></td>
<td>
<p>The dataframe containing the data; the variables names specified
in <code>names</code> (when calling <code>detStructAddVarNames</code>) must be present
in this dataframe.</p>
</td></tr>
<tr><td><code id="detStructAddVarLabels_+3A_append">append</code></td>
<td>
<p>Whether to only return the products or scales, or whether to
append these to the dataframe and return the entire dataframe.</p>
</td></tr>
<tr><td><code id="detStructAddVarLabels_+3A_separator">separator</code></td>
<td>
<p>The separator to use when constructing the scale variables
names.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This family of functions will be explained more in detail in a forthcoming
paper.
</p>


<h3>Value</h3>

<p><code>detStructAddVarLabels</code> and <code>detStructAddVarNames</code> just
change the <code><a href="#topic+determinantStructure">determinantStructure</a></code> object;
<code>detStructComputeProducts</code> and <code>detStructComputeScales</code> return
either the dataframe with the new variables appended (if <code>append</code> =
<code>TRUE</code>) or just a dataframe with the new variables (if <code>append</code> =
<code>FALSE</code>).
</p>


<h3>References</h3>

<p>(Forthcoming)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+determinantStructure">determinantStructure</a></code>, <code><a href="#topic+determinantVar">determinantVar</a></code>,
<code><a href="#topic+subdeterminants">subdeterminants</a></code>, <code><a href="#topic+subdeterminantProducts">subdeterminantProducts</a></code>,
<code><a href="#topic+detStructCIBER">detStructCIBER</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Create some bogus determinant data
detStudy &lt;- mtcars[, c(1, 3:7)];
names(detStudy) &lt;- c('rUse_behav',
                     'rUse_intention',
                     'rUse_attitude1',
                     'rUse_attitude2',
                     'rUse_expAtt1',
                     'rUse_expAtt2');

### Specify the determinant structure

### First a subdeterminant
expAtt &lt;-
  behaviorchange::subdeterminants("Subdeterminants",
                                  "expAtt");

### Then two determinants
attitude &lt;-
  behaviorchange::determinantVar("Determinant",
                                 "attitude",
                                 expAtt);

intention &lt;-
  behaviorchange::determinantVar("ProximalDeterminant",
                                 "intention",
                                 attitude);

### Then the entire determinant strcture
detStruct &lt;-
  behaviorchange::determinantStructure('Behavior',
                                       list('behav',
                                       behaviorRegEx = 'rUse'),
                                       intention);

### Add the variable names
behaviorchange::detStructAddVarNames(detStruct,
                                     names(detStudy));

### Add the determinant scale variable to the dataframe
detStudyPlus &lt;-
  behaviorchange::detStructComputeScales(detStruct,
                                         data=detStudy);

### Show its presence
names(detStudyPlus);
mean(detStudyPlus$rUse_Determinant);

</code></pre>

<hr>
<h2 id='dMCD'>Estimate Cohen's d corresponding to a Meaningful Change Definition</h2><span id='topic+dMCD'></span><span id='topic+print.dMCD'></span>

<h3>Description</h3>

<p>This function uses a base rate (Control Event Rate, argument <code>cer</code>) and a
Meaningful Change Definitions (MCD, argument <code>mcd</code>) to compute the
corresponding Cohen's d. See Gruijters &amp; Peters (2019) for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dMCD(
  cer,
  mcd = NULL,
  eer = NULL,
  plot = TRUE,
  mcdOnX = FALSE,
  plotResultValues = TRUE,
  resultValueLineColor = "blue",
  resultValueLineSize = 1,
  returnLineLayerOnly = FALSE,
  theme = ggplot2::theme_bw(),
  highestPossibleEER = 0.999999999,
  xLab = ifelse(mcdOnX, "Meaningful Change Definition", "Control Event Rate"),
  yLab = "Cohen's d",
  dist = "norm",
  distArgs = list(),
  distNS = "stats",
  ...
)

## S3 method for class 'dMCD'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dMCD_+3A_cer">cer</code></td>
<td>
<p>The Control Event Rate (or base rate): how many people already
perform the target behavior in the population (as a proportion)?</p>
</td></tr>
<tr><td><code id="dMCD_+3A_mcd">mcd</code></td>
<td>
<p>The Meaningful Change Definitions: by which percentage (as a
proportion) should the event rate increase to render an effect meaningful?</p>
</td></tr>
<tr><td><code id="dMCD_+3A_eer">eer</code></td>
<td>
<p>Instead of the MCD, it is also possible to specify the Experimental
Event Rate (EER), in which case the MCD is computed by taking the difference
with the CER.</p>
</td></tr>
<tr><td><code id="dMCD_+3A_plot">plot</code></td>
<td>
<p>Whether to show a plot.</p>
</td></tr>
<tr><td><code id="dMCD_+3A_mcdonx">mcdOnX</code></td>
<td>
<p>Whether to plot the Meaningful Change Definition on the X axis
(by default, the CER is plotted on the X axis).</p>
</td></tr>
<tr><td><code id="dMCD_+3A_plotresultvalues">plotResultValues</code></td>
<td>
<p>Whether to plot the result values.</p>
</td></tr>
<tr><td><code id="dMCD_+3A_resultvaluelinecolor">resultValueLineColor</code>, <code id="dMCD_+3A_resultvaluelinesize">resultValueLineSize</code></td>
<td>
<p>If plotting the result values,
lines of this color and size are used.</p>
</td></tr>
<tr><td><code id="dMCD_+3A_returnlinelayeronly">returnLineLayerOnly</code></td>
<td>
<p>Whether to only return a layer with the plotted
line (which can be used to quickly stack lines for different MCDs).</p>
</td></tr>
<tr><td><code id="dMCD_+3A_theme">theme</code></td>
<td>
<p>The <code>ggplot2</code> theme to use.</p>
</td></tr>
<tr><td><code id="dMCD_+3A_highestpossibleeer">highestPossibleEER</code></td>
<td>
<p>The highest possible EER to include in the plot.</p>
</td></tr>
<tr><td><code id="dMCD_+3A_xlab">xLab</code>, <code id="dMCD_+3A_ylab">yLab</code></td>
<td>
<p>The labels for the X and Y axes.</p>
</td></tr>
<tr><td><code id="dMCD_+3A_dist">dist</code>, <code id="dMCD_+3A_distargs">distArgs</code>, <code id="dMCD_+3A_distns">distNS</code></td>
<td>
<p>Used to specify the distribution to use to convert
between Cohen's d and the CER and EER. distArgs can be used to specify
additional arguments to the corresponding <code>q</code> and <code>p</code> functions, and
distNS to specify the namespace (i.e. package) from where to get the
distribution functions.</p>
</td></tr>
<tr><td><code id="dMCD_+3A_...">...</code></td>
<td>
<p>Any additional arguments to dMCD are passed on to the <code>ggplot2::geom_line</code>
used to draw the line showing the different Cohen's d values as a function of
the base rate (or MCD) on the X axis. Additional arguments for the print method
are passed on to the default print method.</p>
</td></tr>
<tr><td><code id="dMCD_+3A_x">x</code></td>
<td>
<p>The object to print (i.e. a result from a call to dMCD).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The Cohen's d value, optionally with a <code>ggplot2</code> plot stored in an
attribute (which is only a <code>ggplot2</code> layer if <code>returnLineLayerOnly=TRUE</code>).
</p>


<h3>References</h3>

<p>Gruijters, S. L. K., &amp; Peters, G.-J. Y. (2020). Meaningful change
definitions: Sample size planning for experimental intervention research.
<em>PsyArXiv</em>. <a href="https://doi.org/10.31234/osf.io/jc295">doi:10.31234/osf.io/jc295</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dMCD(.2, .05);
</code></pre>

<hr>
<h2 id='lm_rSq_ci'>Obtaining an R squared confidence interval estimate for an lm regression</h2><span id='topic+lm_rSq_ci'></span>

<h3>Description</h3>

<p>The <code>lm_rSq_ci</code> function uses the base R <code>lm</code> function to conduct
a regression analysis and then computes the confidence interval for R squared.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lm_rSq_ci(
  formula,
  data = NULL,
  conf.level = 0.95,
  ci.method = c("widest", "r.con", "olkinfinn"),
  env = parent.frame()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lm_rSq_ci_+3A_formula">formula</code></td>
<td>
<p>The formula of the regression analysis, of the form <code>y ~
x1 + x2</code>, where y is the dependent variable and x1 and x2 are the
predictors.</p>
</td></tr>
<tr><td><code id="lm_rSq_ci_+3A_data">data</code></td>
<td>
<p>If the terms in the formula aren't vectors but variable names,
this should be the dataframe where those variables are stored.</p>
</td></tr>
<tr><td><code id="lm_rSq_ci_+3A_conf.level">conf.level</code></td>
<td>
<p>The confidence of the confidence interval around the
regression coefficients.</p>
</td></tr>
<tr><td><code id="lm_rSq_ci_+3A_ci.method">ci.method</code></td>
<td>
<p>Which method to use for the confidence
interval around R squared.</p>
</td></tr>
<tr><td><code id="lm_rSq_ci_+3A_env">env</code></td>
<td>
<p>The enviroment where to evaluate the formula.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The confidence interval
</p>


<h3>Author(s)</h3>

<p>Gjalt-Jorn Peters
</p>
<p>Maintainer: Gjalt-Jorn Peters <a href="mailto:gjalt-jorn@a-bc.eu">gjalt-jorn@a-bc.eu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
### Do a simple regression analysis
lm_rSq_ci(age ~ circumference, dat=Orange);

</code></pre>

<hr>
<h2 id='nnc'>Numbers Needed for Change</h2><span id='topic+nnc'></span><span id='topic+nnt'></span><span id='topic+print.nnc'></span>

<h3>Description</h3>

<p>This function computes the Numbers Needed for Change, and shows a
visualisation to illustrate them. Numbers Needed for Change is
the name for a Numbers Needed to Treat estimate that was computed
for a continuous outcome as is common in behavior change research.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nnc(
  d = NULL,
  cer = NULL,
  r = 1,
  n = NULL,
  threshold = NULL,
  mean = 0,
  sd = 1,
  poweredFor = NULL,
  thresholdSensitivity = NULL,
  eventDesirable = TRUE,
  eventIfHigher = TRUE,
  conf.level = 0.95,
  dReliability = 1,
  d.ci = NULL,
  cer.ci = NULL,
  r.ci = NULL,
  d.n = NULL,
  cer.n = NULL,
  r.n = NULL,
  plot = TRUE,
  returnPlot = TRUE,
  silent = FALSE
)

## S3 method for class 'nnc'
print(x, digits = 2, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nnc_+3A_d">d</code></td>
<td>
<p>The value of Cohen's <em>d</em>.</p>
</td></tr>
<tr><td><code id="nnc_+3A_cer">cer</code></td>
<td>
<p>The Control Event Rate.</p>
</td></tr>
<tr><td><code id="nnc_+3A_r">r</code></td>
<td>
<p>The correlation between the determinant and behavior (for mediated
Numbers Needed for Change).</p>
</td></tr>
<tr><td><code id="nnc_+3A_n">n</code></td>
<td>
<p>The sample size.</p>
</td></tr>
<tr><td><code id="nnc_+3A_threshold">threshold</code></td>
<td>
<p>If the event rate is not available, a threshold value can
be specified instead, which is then used in conjunction with the mean
(<code>mean</code>) and standard deviation (<code>sd</code>) and assuming a normal
distribution to compute the event rate.</p>
</td></tr>
<tr><td><code id="nnc_+3A_mean">mean</code></td>
<td>
<p>The mean value, used to draw the plot, or, if no CER is provided
but instead the threshold value, to compute the CER.</p>
</td></tr>
<tr><td><code id="nnc_+3A_sd">sd</code></td>
<td>
<p>The standard deviation, used to draw the plot (and to compute the
CER if a threshold value is supplied instead of the CER).</p>
</td></tr>
<tr><td><code id="nnc_+3A_poweredfor">poweredFor</code></td>
<td>
<p>The Cohen's <em>d</em> value for which the study was
powered. This expected Cohen's <em>d</em> value can be used to compute the
threshold, which then in turn is used to compute the CER. To use this
approach, also specify the mean and the standard deviation.</p>
</td></tr>
<tr><td><code id="nnc_+3A_thresholdsensitivity">thresholdSensitivity</code></td>
<td>
<p>This argument can be used to provide a vector of
potential threshold values, each of which is used to compute an NNC. This
enables easy inspection of whether the value chosen as threshold matters
much for the NNC.</p>
</td></tr>
<tr><td><code id="nnc_+3A_eventdesirable">eventDesirable</code></td>
<td>
<p>Whether an event is desirable or undesirable.</p>
</td></tr>
<tr><td><code id="nnc_+3A_eventifhigher">eventIfHigher</code></td>
<td>
<p>Whether scores above or below the threshold are
considered 'an event'.</p>
</td></tr>
<tr><td><code id="nnc_+3A_conf.level">conf.level</code></td>
<td>
<p>The confidence level of the confidence interval.</p>
</td></tr>
<tr><td><code id="nnc_+3A_dreliability">dReliability</code></td>
<td>
<p>If Cohen's d was not measured with perfect reliability,
<code>nnc</code> can disattenuate it to correct for the resulting attenuation using
<code><a href="ufs.html#topic+disattenuate.d">ufs::disattenuate.d()</a></code> before computating the Experimental Event Rate. Use
this argument to specify the reliability of the outcome measure. By default,
the setting of 1 means that no disattenuation is applied.</p>
</td></tr>
<tr><td><code id="nnc_+3A_d.ci">d.ci</code></td>
<td>
<p>Instead of providing a point estimate for Cohen's <em>d</em>, a
confidence interval can be provided.</p>
</td></tr>
<tr><td><code id="nnc_+3A_cer.ci">cer.ci</code></td>
<td>
<p>Instead of providing a point estimate for the Control Event
Rate, a confidence interval can be provided.</p>
</td></tr>
<tr><td><code id="nnc_+3A_r.ci">r.ci</code></td>
<td>
<p>Instead of providing a point estimate for the correlation, a
confidence interval can be provided.</p>
</td></tr>
<tr><td><code id="nnc_+3A_d.n">d.n</code></td>
<td>
<p>In addition to providing a point estimate for Cohen's <em>d</em>, a
sample size can be provided; if it is, the confidence interval is computed.</p>
</td></tr>
<tr><td><code id="nnc_+3A_cer.n">cer.n</code></td>
<td>
<p>In addition to providing a point estimate for the Control Event
Rate, a sample size can be provided; if it is, the confidence interval is
computed.</p>
</td></tr>
<tr><td><code id="nnc_+3A_r.n">r.n</code></td>
<td>
<p>In addition to providing a point estimate for the correlation, a
sample size can be provided; if it is, the confidence interval is computed.</p>
</td></tr>
<tr><td><code id="nnc_+3A_plot">plot</code></td>
<td>
<p>Whether to generate and show the plot.</p>
</td></tr>
<tr><td><code id="nnc_+3A_returnplot">returnPlot</code></td>
<td>
<p>Whether to return the plot (as an attribute), or to only
display it.</p>
</td></tr>
<tr><td><code id="nnc_+3A_silent">silent</code></td>
<td>
<p>Whether to suppress notifications.</p>
</td></tr>
<tr><td><code id="nnc_+3A_x">x</code></td>
<td>
<p>The <code>nnc</code> object to print.</p>
</td></tr>
<tr><td><code id="nnc_+3A_digits">digits</code></td>
<td>
<p>The number of digits to round to.</p>
</td></tr>
<tr><td><code id="nnc_+3A_...">...</code></td>
<td>
<p>Any additional arguments are passed to the <code>print</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Numbers Needed to Treat is a common and very useful effect size
measure in use in the medical sciences. It is computed based on the
Control Event Rate (CER) and the Experimental Event Rate (EER), and
expresses how many people would need to received a treatment to yield
a beneficial result for one person. In behavior change research, a
similar measure would be useful, but the outcome is normally not
dichotomous as is common in the medical literature (i.e. whether a
participants survives or is cured), but continuous. Numbers Needed
for Change fills this lacuna: it is simply the Numbers Needed to Treat,
but converted from a Cohen's d value. <code>nnt</code> is an alias for <code>nnc</code>.
</p>
<p>For more details, see Gruijters &amp; Peters (2019) for details.
</p>


<h3>Value</h3>

<p>The Numbers Needed for Change (NNC), potentially with a plot
visualising the NNC in an attribute.
</p>


<h3>Author(s)</h3>

<p>Gjalt-Jorn Peters &amp; Stefan Gruijters
</p>
<p>Maintainer: Gjalt-Jorn Peters <a href="mailto:gjalt-jorn@userfriendlyscience.com">gjalt-jorn@userfriendlyscience.com</a>
</p>


<h3>References</h3>

<p>Gruijters, S. L., &amp; Peters, G. Y. (2019). Gauging the
impact of behavior change interventions: A tutorial on the Numbers
Needed to Treat. <em>PsyArXiv.</em> <a href="https://doi.org/10.31234/osf.io/2bau7">doi:10.31234/osf.io/2bau7</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
### Simple example
behaviorchange::nnc(d=.4, cer=.3);

### Or for a scenario where events are undesirable, and the
### intervention effective (therefore having a negative value for d):
behaviorchange::nnc(d=-.4, cer=.3, eventDesirable=FALSE);

</code></pre>

<hr>
<h2 id='opts'>Options for the behaviorchange package</h2><span id='topic+opts'></span><span id='topic+set'></span><span id='topic+get'></span><span id='topic+reset'></span>

<h3>Description</h3>

<p>The <code>behaviorchange::opts</code> object contains three functions to set, get, and reset
options used by the escalc package. Use <code>behaviorchange::opts$set</code> to set options,
<code>behaviorchange::opts$get</code> to get options, or <code>behaviorchange::opts$reset</code> to reset specific or
all options to their default values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>opts
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 4.
</p>


<h3>Details</h3>

<p>It is normally not necessary to get or set <code>behaviorchange</code> options.
</p>
<p>The following arguments can be passed:
</p>

<dl>
<dt>...</dt><dd><p>For <code>behaviorchange::opts$set</code>, the dots can be used to specify the options
to set, in the format <code>option = value</code>, for example,
<code>EFFECTSIZE_POINTESTIMATE_NAME_IN_DF = "\n"</code>. For
<code>behaviorchange::opts$reset</code>, a list of options to be reset can be passed.</p>
</dd>
<dt>option</dt><dd><p>For <code>behaviorchange::opts$set</code>, the name of the option to set.</p>
</dd>
<dt>default</dt><dd><p>For <code>behaviorchange::opts$get</code>, the default value to return if the
option has not been manually specified.</p>
</dd>
</dl>

<p>To see the full list of options and their default values,
use <code>behaviorchange::opts$default()</code>. Some examples are:
</p>

<dl>
<dt>aabbcc</dt><dd><p>A color theme for <code><a href="#topic+abcd">abcd()</a></code>.</p>
</dd>
<dt>complecs_*</dt><dd><p>The worksheet and columns names for <code><a href="#topic+complecs">complecs()</a></code>.</p>
</dd>
<dt>silent</dt><dd><p>Whether to be chatty or silent.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>### Get the default utteranceMarker
behaviorchange::opts$get(complecs_entitySheet);

### Set it to a custom version, so that every line starts with a pipe
behaviorchange::opts$set(complecs_entitySheet = "sheet_with_entities");

### Check that it worked
behaviorchange::opts$get(complecs_entitySheet);

### Reset this option to its default value
behaviorchange::opts$reset(complecs_entitySheet);

### Check that the reset worked, too
behaviorchange::opts$get(complecs_entitySheet);

</code></pre>

<hr>
<h2 id='partypanelData'>Subsets of Party Panel datasets</h2><span id='topic+partypanelData'></span><span id='topic+BBC_data'></span><span id='topic+BBC_pp15.1'></span><span id='topic+BBC_pp16.1'></span><span id='topic+BBC_pp17.1'></span><span id='topic+BBC_pp18.1'></span>

<h3>Description</h3>

<p>These are subsets of Party Panel datasets. Party Panel is an annual
semi-panel determinant study among Dutch nightlife patrons, where every
year, the determinants of another nightlife-related risk behavior are
mapped.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(BBC_pp15.1)

data(BBC_pp16.1)

data(BBC_pp17.1)

data(BBC_pp18.1)
</code></pre>


<h3>Format</h3>

<p>For BBC_pp15.1, a <code>data.frame</code> with 123 columns and 829 rows.
For BBC_pp16.1, a <code>data.frame</code> with 63 columns and 1077 rows.
For BBC_pp17.1, a <code>data.frame</code> with 94 columns and 943 rows.
For BBC_pp18.1, a <code>data.frame</code> with 84 columns and 880 rows.
Note that many rows contain missing values; the columns and rows
were taken directly from the original Party Panel datasets, and
represent all participants that made it past a given behavior.
</p>


<h3>Details</h3>

<p>The behaviors of the Party Panel waves were:
</p>

<ul>
<li><p> 2015: Behaviors related to using highly dosed ecstasy pills
</p>
</li>
<li><p> 2016: Behaviors related to visiting nightlife first-aid facilities
</p>
</li>
<li><p> 2017: Behaviors related to hearing protection
</p>
</li>
<li><p> 2018: Behaviors related to flirting and boundary crossing
</p>
</li>
<li><p> 2019: Behaviors related to sleeping hygiene surrounding nightlife participation
</p>
</li></ul>

<p>The full datasets are publicly available through the Open Science
Framework (https://osf.io/s4fmu/). Also see the GitLab
repositories (https://gitlab.com/partypanel) and the website
at https://partypanel.eu.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('BBC_pp17.1', package='behaviorchange');
behaviorchange::CIBERlite(data=BBC_pp17.1,
                          determinants=c("epw_attitude",
                                         "epw_perceivedNorm",
                                         "epw_pbc",
                                         "epw_habit"),
                          targets=c("epw_intention"));
</code></pre>

<hr>
<h2 id='pies'>Practically Important Effect Sizes</h2><span id='topic+pies'></span>

<h3>Description</h3>

<p>Practically Important Effect Sizes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pies(
  data = NULL,
  controlCol = NULL,
  expCol = NULL,
  d = NULL,
  cer = NULL,
  r = 1,
  n = NULL,
  threshold = NULL,
  mean = 0,
  sd = 1,
  bootstrapA = FALSE,
  conf.level = 0.95
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pies_+3A_data">data</code></td>
<td>
<p>Optionally, if you want to get A, a data frame.</p>
</td></tr>
<tr><td><code id="pies_+3A_controlcol">controlCol</code>, <code id="pies_+3A_expcol">expCol</code></td>
<td>
<p>Optionally, if you want to get A, the names of
the columns with control and experimental data.</p>
</td></tr>
<tr><td><code id="pies_+3A_d">d</code></td>
<td>
<p>Cohen's d.</p>
</td></tr>
<tr><td><code id="pies_+3A_cer">cer</code></td>
<td>
<p>The control even rate (see <code><a href="#topic+nnt">nnt()</a></code>).</p>
</td></tr>
<tr><td><code id="pies_+3A_r">r</code>, <code id="pies_+3A_threshold">threshold</code>, <code id="pies_+3A_mean">mean</code>, <code id="pies_+3A_sd">sd</code></td>
<td>
<p>Arguments for the <code><a href="#topic+nnt">nnt()</a></code>
function.</p>
</td></tr>
<tr><td><code id="pies_+3A_n">n</code></td>
<td>
<p>The sample size.</p>
</td></tr>
<tr><td><code id="pies_+3A_bootstrapa">bootstrapA</code></td>
<td>
<p>Whether to use bootstrapping to compute A.</p>
</td></tr>
<tr><td><code id="pies_+3A_conf.level">conf.level</code></td>
<td>
<p>The confidence level of confidence intervals.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with all values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pies(d = .5, n = 100, cer = .2, threshold = 2);
</code></pre>

<hr>
<h2 id='repeatStr'>Repeat a string a number of times</h2><span id='topic+repeatStr'></span><span id='topic+repStr'></span>

<h3>Description</h3>

<p>Repeat a string a number of times
</p>


<h3>Usage</h3>

<pre><code class='language-R'>repeatStr(n = 1, str = " ")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="repeatStr_+3A_n">n</code>, <code id="repeatStr_+3A_str">str</code></td>
<td>
<p>Normally, respectively the frequency with which to
repeat the string and the string to repeat; but the order of the
inputs can be switched as well.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of length 1.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### 10 spaces:
repStr(10);

### Three euro symbols:
repStr("\u20ac", 3);
</code></pre>

<hr>
<h2 id='vecTxt'>Easily parse a vector into a character value</h2><span id='topic+vecTxt'></span><span id='topic+vecTxtQ'></span>

<h3>Description</h3>

<p>Easily parse a vector into a character value
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vecTxt(
  vector,
  delimiter = ", ",
  useQuote = "",
  firstDelimiter = NULL,
  lastDelimiter = " &amp; ",
  firstElements = 0,
  lastElements = 1,
  lastHasPrecedence = TRUE
)

vecTxtQ(vector, useQuote = "'", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vecTxt_+3A_vector">vector</code></td>
<td>
<p>The vector to process.</p>
</td></tr>
<tr><td><code id="vecTxt_+3A_delimiter">delimiter</code>, <code id="vecTxt_+3A_firstdelimiter">firstDelimiter</code>, <code id="vecTxt_+3A_lastdelimiter">lastDelimiter</code></td>
<td>
<p>The delimiters
to use for respectively the middle, first
<code>firstElements</code>, and last <code>lastElements</code> elements.</p>
</td></tr>
<tr><td><code id="vecTxt_+3A_usequote">useQuote</code></td>
<td>
<p>This character string is pre- and appended to all elements;
so use this to quote all elements (<code>useQuote="'"</code>), doublequote all
elements (<code>useQuote='"'</code>), or anything else (e.g. <code>useQuote='|'</code>).
The only difference between <code>vecTxt</code> and <code>vecTxtQ</code> is that the
latter by default quotes the elements.</p>
</td></tr>
<tr><td><code id="vecTxt_+3A_firstelements">firstElements</code>, <code id="vecTxt_+3A_lastelements">lastElements</code></td>
<td>
<p>The number of elements for which to use
the first respective last delimiters</p>
</td></tr>
<tr><td><code id="vecTxt_+3A_lasthasprecedence">lastHasPrecedence</code></td>
<td>
<p>If the vector is very short, it's possible that the
sum of firstElements and lastElements is larger than the vector length. In
that case, downwardly adjust the number of elements to separate with the
first delimiter (<code>TRUE</code>) or the number of elements to separate with the
last delimiter (<code>FALSE</code>)?</p>
</td></tr>
<tr><td><code id="vecTxt_+3A_...">...</code></td>
<td>
<p>Any addition arguments to <code>vecTxtQ</code> are passed on to
<code>vecTxt</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of length 1.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vecTxtQ(names(mtcars));
</code></pre>

<hr>
<h2 id='wrapVector'>Wrap all elements in a vector</h2><span id='topic+wrapVector'></span>

<h3>Description</h3>

<p>Wrap all elements in a vector
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wrapVector(x, width = 0.9 * getOption("width"), sep = "\n", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wrapVector_+3A_x">x</code></td>
<td>
<p>The character vector</p>
</td></tr>
<tr><td><code id="wrapVector_+3A_width">width</code></td>
<td>
<p>The number of</p>
</td></tr>
<tr><td><code id="wrapVector_+3A_sep">sep</code></td>
<td>
<p>The glue with which to combine the new lines</p>
</td></tr>
<tr><td><code id="wrapVector_+3A_...">...</code></td>
<td>
<p>Other arguments are passed to <code><a href="base.html#topic+strwrap">strwrap()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>res &lt;- wrapVector(
  c(
    "This is a sentence ready for wrapping",
    "So is this one, although it's a bit longer"
  ),
  width = 10
);

print(res);
cat(res, sep="\n");
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
