<!DOCTYPE html><html lang="en"><head><title>Help for package benthos</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {benthos}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#benthos-package'><p>Marine Benthic Ecosystem Analysis</p></a></li>
<li><a href='#abundance'><p>Abundance</p></a></li>
<li><a href='#ambi'><p>AZTI Marine Biotic Index (AMBI)</p></a></li>
<li><a href='#as_accepted'><p>Convert Taxon Names to Comply with WoRMS/TWN</p></a></li>
<li><a href='#bray_curtis'><p>Bray-Curtis Dissimilarity</p></a></li>
<li><a href='#eqr'><p>Ecological Quality Ratio (EQR)</p></a></li>
<li><a href='#genus_to_species'><p>Genus to Species Conversion</p></a></li>
<li><a href='#get_ambi'><p>Get Supplementary AMBI Sensitivity Groups</p></a></li>
<li><a href='#get_iti'><p>Get Infaunal Trophic Index</p></a></li>
<li><a href='#harmonize'><p>Harmonize Case</p></a></li>
<li><a href='#hill'><p>Hill's Diversity Numbers</p></a></li>
<li><a href='#hpie'><p>Hurlbert's Probability of Interspecific Encounter (PIE)</p></a></li>
<li><a href='#hurlbert'><p>Hurlbert's Expected Number of Species</p></a></li>
<li><a href='#is_azoic'><p>Test for Azoic Samples</p></a></li>
<li><a href='#is_binomen'><p>Binomial Names</p>
</p>
<p><code>is_binomial</code> tests for valid binomial names,</p>
<code>generic_name</code> extracts the genus to which the species belongs,
<code>specific_name</code> extracts the species within the genus.</a></li>
<li><a href='#iti'><p>Infaunal Trophic Index (ITI)</p></a></li>
<li><a href='#margalef'><p>Margalef Index of Diversity</p></a></li>
<li><a href='#northsea'><p>MWTL North Sea Bentos Data</p></a></li>
<li><a href='#oosterschelde'><p>Oosterschelde Marine Benthos Data</p></a></li>
<li><a href='#pool'><p>Pooling</p></a></li>
<li><a href='#read_ambi'><p>Read and Validate AMBI Sensitivity Data</p></a></li>
<li><a href='#read_beqi2'><p>Read and Validate BEQI2 Input Files</p></a></li>
<li><a href='#read_iti'><p>Read and Validate Infaunal Trophic Index Files</p></a></li>
<li><a href='#read_ref'><p>Read and Validate Habitat References Files</p></a></li>
<li><a href='#read_taxa'><p>Read and Validate Taxa Data</p></a></li>
<li><a href='#read_twn'><p>Read and Validate Taxa Waterbeheer Nederland (TWN) Data</p></a></li>
<li><a href='#rygg'><p>Rygg's Index of Diversity</p></a></li>
<li><a href='#shannon'><p>Shannon's Index or Entropy</p></a></li>
<li><a href='#simpson'><p>Simpson's Measure of Concentration</p></a></li>
<li><a href='#species_richness'><p>Species Richness</p></a></li>
<li><a href='#strip_spaces'><p>Remove Redundant Spaces</p></a></li>
<li><a href='#to_worms'><p>Convert Taxon Names to Comply with WoRMS</p></a></li>
<li><a href='#total_abundance'><p>Total Abundance</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Marine Benthic Ecosystem Analysis</td>
</tr>
<tr>
<td>Description:</td>
<td>Preprocessing tools and biodiversity measures 
    (species abundance, species richness, population heterogeneity and 
    sensitivity) for analysing marine benthic data. See Van Loon et al.
    (2015) &lt;<a href="https://doi.org/10.1016%2Fj.seares.2015.05.002">doi:10.1016/j.seares.2015.05.002</a>&gt; for an application of 
    these tools.</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3-8</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-08-22</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr (&ge; 0.7.0), lazyeval, readr, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 2.1.0), rmarkdown, knitr, ggplot2, tidyr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-08-22 09:49:38 UTC; dennis</td>
</tr>
<tr>
<td>Author:</td>
<td>Dennis Walvoort [aut, cre, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dennis Walvoort &lt;dennis.Walvoort@wur.nl&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-08-22 10:10:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='benthos-package'>Marine Benthic Ecosystem Analysis</h2><span id='topic+benthos'></span><span id='topic+benthos-package'></span>

<h3>Description</h3>

<p><span class="pkg">benthos</span> provides functions for facilitating the analysis of marine
benthos data. Examples are indicators like species abundance, 
species richness, Margalef's index of diversity, Shannon's Entropy, 
AZTI's Marine Biotic Index, and the Infaunal Trophic Index (ITI). 
In addition functions for data pooling,
genus-to-species conversion and validation and conversion of species names 
to those recommended by the World Register of Marine Species are provided.
</p>


<h3>Details</h3>

<p>All functions are designed to work seamlessly with the
<span class="pkg">dplyr</span>-package which implements a grammar for structured
data manimpulation.
</p>
<p>The <span class="pkg">benthos</span>-package contains functions for estimating various
species abundance, species richness, species heterogeneity and species
sensitivity measures:
</p>

<ul>
<li><p> total abundance (<code><a href="#topic+total_abundance">total_abundance</a></code>)
</p>
</li>
<li><p> abundance (<code><a href="#topic+abundance">abundance</a></code>)
</p>
</li>
<li><p> species richness (<code><a href="#topic+species_richness">species_richness</a></code>)
</p>
</li>
<li><p> Margalef's index of diversity (<code><a href="#topic+margalef">margalef</a></code>)
</p>
</li>
<li><p> Rygg's index of diversity (<code><a href="#topic+rygg">rygg</a></code>)
</p>
</li>
<li><p> Hurlbert's Expected Number of Species (<code><a href="#topic+hurlbert">hurlbert</a></code>)
</p>
</li>
<li><p> Simpson's measure of concentration (<code><a href="#topic+simpson">simpson</a></code>)
</p>
</li>
<li><p> Hurlbert's probability of interspecific encounter (PIE) (<code><a href="#topic+hpie">hpie</a></code>)
</p>
</li>
<li><p> Shannon's index or entropy (<code><a href="#topic+shannon">shannon</a></code>)
</p>
</li>
<li><p> Hill's diversity number (<code><a href="#topic+hill">hill</a></code>)
</p>
</li>
<li><p> AZTI Marine Biotic Index (AMBI) (<code><a href="#topic+ambi">ambi</a></code>)
</p>
</li>
<li><p> Infaunal Trophic Index (ITI) (<code><a href="#topic+iti">iti</a></code>)
</p>
</li>
<li><p> Bray-Curtis dissimilarity (<code><a href="#topic+bray_curtis">bray_curtis</a></code>)
</p>
</li></ul>

<p>In addition, functions are available for data preparation, e.g.:
</p>

<ul>
<li><p> data pooling (<code><a href="#topic+pool">pool</a></code>)
</p>
</li>
<li><p> genus to species conversion (<code><a href="#topic+genus_to_species">genus_to_species</a></code>)
</p>
</li></ul>

<p>For an overview of all the functions in the package click on the index link
at the bottom of this page.
</p>


<h3>Author(s)</h3>

<p>Dennis Walvoort <a href="mailto:dennis.walvoort@wur.nl">dennis.walvoort@wur.nl</a>
</p>


<h3>See Also</h3>

<p>The <span class="pkg">BEQI2</span>-package on CRAN, and the package vignettes.
</p>

<hr>
<h2 id='abundance'>Abundance</h2><span id='topic+abundance'></span><span id='topic+abundance_'></span>

<h3>Description</h3>

<p>The number of indiviuals in each taxon.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>abundance(.data = NULL, taxon = NULL, count)

abundance_(.data = NULL, taxon = NULL, count)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="abundance_+3A_.data">.data</code></td>
<td>
<p>data in a <code>data.frame</code>, <code>tibble</code>, 
<code>data.table</code>, database etc.</p>
</td></tr>
<tr><td><code id="abundance_+3A_taxon">taxon</code></td>
<td>
<p>name of column in <code>.data</code> containing taxa</p>
</td></tr>
<tr><td><code id="abundance_+3A_count">count</code></td>
<td>
<p>name of column in <code>.data</code> containing counts</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code> vector with abundance per taxon.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>abundance_()</code>: version suitable for calling from a function
(see package <span class="pkg">lazyeval</span>).
</p>
</li></ul>


<h3>Note</h3>

<p>due to pooling, the abundance is not necessarily an integer
</p>


<h3>Examples</h3>

<pre><code class='language-R'> abundance(
     taxon = c("Euspira pulchella", "Nephtys cirrosa"), 
     count = c(4, 6)
 )
</code></pre>

<hr>
<h2 id='ambi'>AZTI Marine Biotic Index (AMBI)</h2><span id='topic+ambi'></span><span id='topic+ambi_'></span><span id='topic+has_ambi'></span><span id='topic+has_ambi_'></span>

<h3>Description</h3>

<p>AZTI Marine Biotic Index (AMBI) according to Borja et al. (2000)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ambi(.data = NULL, taxon, count, group = NULL)

ambi_(.data = NULL, taxon, count, group = NULL)

has_ambi(.data = NULL, taxon, group = NULL)

has_ambi_(.data = NULL, taxon, group = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ambi_+3A_.data">.data</code></td>
<td>
<p>data in a <code>data.frame</code>, <code>tibble</code>, 
<code>data.table</code>, database etc.</p>
</td></tr>
<tr><td><code id="ambi_+3A_taxon">taxon</code></td>
<td>
<p>species names</p>
</td></tr>
<tr><td><code id="ambi_+3A_count">count</code></td>
<td>
<p>counts of individuals (<code>numeric</code>)</p>
</td></tr>
<tr><td><code id="ambi_+3A_group">group</code></td>
<td>
<p>sensitivity groups I, II, III, IV, or V</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The index is given by:
</p>
<p style="text-align: center;"><code class="reqn">c_\mathrm{b} = \frac{3}{2} \sum_{i=2}^5 (i-1) p_i</code>
</p>

<p>where <code class="reqn">p_i</code> is the proportion of species in sensitivity group <code class="reqn">i</code>.
</p>


<h3>Value</h3>

<p>numeric vector of length 1 containing the AMBI
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>ambi_()</code>: version suitable for calling from a function
(see package <span class="pkg">lazyeval</span>).
</p>
</li>
<li> <p><code>has_ambi()</code>: tests if an AMBI sensitivity group is available for <code>taxon</code>
(returns <code>TRUE</code> (available) or <code>FALSE</code> (unavailable))
</p>
</li>
<li> <p><code>has_ambi_()</code>: version suitable for calling from a function
(see package <span class="pkg">lazyeval</span>).
</p>
</li></ul>


<h3>References</h3>

<p>Borja, A., J. Franco and V. Perez, 2000. A Marine Biotic Index 
to Establish the Ecological Quality of Soft-Bottom Benthos Within 
European Estuarine and Coastal Environments. 
Marine Pollution Bulletin 40:1100-1114
</p>


<h3>Examples</h3>

<pre><code class='language-R'> ambi(
     taxon = c("Euspira pulchella", "Nephtys cirrosa"), 
     count = c(4, 6)
 )
 
     data(oosterschelde)
     has_ambi(oosterschelde, TAXON)
 
</code></pre>

<hr>
<h2 id='as_accepted'>Convert Taxon Names to Comply with WoRMS/TWN</h2><span id='topic+as_accepted'></span>

<h3>Description</h3>

<p>Taxon names are standardized according to the World Register of 
Marine Species (WoRMS) database. The conversion is case-insensitive.
For this conversion, the TWN-list (Taxa Water management the Netherlands) 
is used, extended with species of the Southern North Sea. 
See references below for download locations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as_accepted(taxon, taxa = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="as_accepted_+3A_taxon">taxon</code></td>
<td>
<p><code><a href="base.html#topic+character">character</a></code> vector, containing taxon names</p>
</td></tr>
<tr><td><code id="as_accepted_+3A_taxa">taxa</code></td>
<td>
<p>an optional table usually created with <code><a href="#topic+read_taxa">read_taxa</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector with WoRMS/TWN compliant species names
</p>


<h3>References</h3>

<p><a href="https://www.marinespecies.org/">https://www.marinespecies.org/</a>
</p>
<p><a href="https://taxainfo.nl/">https://taxainfo.nl/</a>
</p>

<hr>
<h2 id='bray_curtis'>Bray-Curtis Dissimilarity</h2><span id='topic+bray_curtis'></span>

<h3>Description</h3>

<p>Bray-Curtis Dissimilarity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bray_curtis(n1, n2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bray_curtis_+3A_n1">n1</code></td>
<td>
<p>abundances of species at site 1</p>
</td></tr>
<tr><td><code id="bray_curtis_+3A_n2">n2</code></td>
<td>
<p>abundances of species at site 2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Bray-Curtis dissimilarity (0..1, 0 = equal, 1 = different)
</p>


<h3>Note</h3>

<p>species in n1 and n2 need to be aligned
</p>


<h3>Examples</h3>

<pre><code class='language-R'>     n1 &lt;- c(11,  0, 7,  8, 0)
     n2 &lt;- c(24, 37, 5, 18, 1)
     bray_curtis(n1, n2)
</code></pre>

<hr>
<h2 id='eqr'>Ecological Quality Ratio (EQR)</h2><span id='topic+eqr'></span>

<h3>Description</h3>

<p>The ecological quality ratio is the ratio beween a parameter value and its
reference value:  
</p>
<p style="text-align: center;"><code class="reqn">EQR = \frac{x-bad}{ref-bad}</code>
</p>

<p>Depending on <code>bad</code> and <code>ref</code>, the EQR usually 
(but not necessarily!) varies between 0 (bad ecological quality) and 1 (
ecological quality equals the reference status).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eqr(x, bad, ref)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="eqr_+3A_x">x</code></td>
<td>
<p>numeric vector containing benthic indices</p>
</td></tr>
<tr><td><code id="eqr_+3A_bad">bad</code></td>
<td>
<p>the value for a bad status</p>
</td></tr>
<tr><td><code id="eqr_+3A_ref">ref</code></td>
<td>
<p>the value for a reference status</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric vector with EQR-values: low values indicate bad ecological
quality and high values indicate good ecological quality.
</p>

<hr>
<h2 id='genus_to_species'>Genus to Species Conversion</h2><span id='topic+genus_to_species'></span>

<h3>Description</h3>

<p>This alorithm reallocates the counts of taxa, that are only indentified at
the genus level to taxa in the same sampling unit and of the same genus 
but that are identified on the species level. The redistribution of counts 
is proportional to the number of counts at the species level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genus_to_species(is_genus, count)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genus_to_species_+3A_is_genus">is_genus</code></td>
<td>
<p><code>logical</code> vector with elements <code>TRUE</code> if the 
corresponding taxon is on the genus level, and <code>FALSE</code> if it is on 
the  <code>species</code> level.</p>
</td></tr>
<tr><td><code id="genus_to_species_+3A_count">count</code></td>
<td>
<p><code>numeric</code> vector with elements giving the counts of each
corresponding taxon.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code> vector with updated counts. The counts for the 
taxon on the genus level have been set to zero.
</p>


<h3>Note</h3>

<p>Parameters <code>is_genus</code> and <code>count</code> are of the same length and
correspond to the same taxon.
</p>
<p>The resulting counts are not necessarily integers.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>     genus_to_species(is_genus = c(TRUE, FALSE, FALSE), count = c(3, 10, 20))
     genus_to_species(is_genus = c(TRUE, FALSE, FALSE), count = c(1, 10, 20))
</code></pre>

<hr>
<h2 id='get_ambi'>Get Supplementary AMBI Sensitivity Groups</h2><span id='topic+get_ambi'></span>

<h3>Description</h3>

<p>This function gets sensitivity groups that are supplementary to the AMBI of
Borja et al., (2000)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_ambi(which = "NL")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_ambi_+3A_which">which</code></td>
<td>
<p>which AMBI supplement? Currently only the Dutch supplement is
available (<code>which</code> = &quot;NL&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame with columns <code>TAXON</code> containing taxa and 
<code>GROUP</code> containing Dutch AMBI-groups
</p>


<h3>References</h3>

<p>Borja, A., J. Franco and V. Perez, 2000. A Marine Biotic Index 
to Establish the Ecological Quality of Soft-Bottom Benthos Within 
European Estuarine and Coastal Environments. 
Marine Pollution Bulletin 40:1100-1114
</p>

<hr>
<h2 id='get_iti'>Get Infaunal Trophic Index</h2><span id='topic+get_iti'></span>

<h3>Description</h3>

<p>This function gets the sensitivity groups to estimate the infaunal 
trophic index of Gittenberger et al., (2011)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_iti()
</code></pre>


<h3>Value</h3>

<p>a data frame with columns <code>TAXON</code> containing taxa and 
<code>GROUP</code> containing the ITI-groups of Gittenberger &amp; Van Loon (2013).
</p>


<h3>References</h3>

<p>Gittenberger A. and  W. van Loon, 2013. 
Sensitivities of marine macrozoobenthos to environmental pressures 
in the Netherlands. Nederlandse Faunistische 
Mededelingen 41: 79-112.
</p>

<hr>
<h2 id='harmonize'>Harmonize Case</h2><span id='topic+harmonize'></span>

<h3>Description</h3>

<p>Convert text to the most occuring case. In case of ties, the first
occurence in sorted order will be taken.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>harmonize(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="harmonize_+3A_x">x</code></td>
<td>
<p>character vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector with harmonized names (i.e., same case)
</p>


<h3>Examples</h3>

<pre><code class='language-R'> x &lt;- c("FOO", "Foo", "bar", "FOO", "bar", "FOO", "Bar")
 y &lt;- harmonize(x)
 stopifnot(all.equal(y, c("FOO", "FOO", "bar", "FOO", "bar", "FOO", "bar")))
 
</code></pre>

<hr>
<h2 id='hill'>Hill's Diversity Numbers</h2><span id='topic+hill'></span><span id='topic+hill_'></span><span id='topic+hill0'></span><span id='topic+hill0_'></span><span id='topic+hill1'></span><span id='topic+hill1_'></span><span id='topic+hill2'></span><span id='topic+hill2_'></span>

<h3>Description</h3>

<p>According to Hill (1973): <em>&quot;a diversity number is figuratively a 
measure of how many species are present if we examine the sample down to a 
certain depth among its rarities. If we examine superficially (e.g.,
by using <code class="reqn">N_2</code>) we shall see only the more abundant species. If we 
look deeply (e.g., by using <code class="reqn">N_0</code>) we shall see all the 
species present.&quot;</em>
</p>
<p>Hill's diversity numbers are given by:
</p>
<p style="text-align: center;"><code class="reqn">N_a=\sum{i=1}^S (p_i^a)^{1/(1-a)}</code>
</p>

<p>Special cases are:
</p>

<dl>
<dt><code class="reqn">N_{-\infty}</code></dt><dd><p>reciprocal of the proportional
abundance of the rarest species;</p>
</dd>
<dt><code class="reqn">N_0</code></dt><dd><p>total number of species present;</p>
</dd>
<dt><code class="reqn">N_1</code></dt><dd><p>exp(H), where H: Shannon's index (see also 
<code><a href="#topic+shannon">shannon</a></code>);</p>
</dd>
<dt><code class="reqn">N_2</code></dt><dd><p>reciprocal of Simpson's index (see also 
<code><a href="#topic+simpson">simpson</a></code>);</p>
</dd>
<dt><code class="reqn">N_{\infty}</code></dt><dd><p>reciprocal of the proportional
abundance of the commonest species.</p>
</dd>
</dl>



<h3>Usage</h3>

<pre><code class='language-R'>hill(.data = NULL, taxon, count, a = 0)

hill_(.data = NULL, taxon, count, a = 0)

hill0(.data = NULL, taxon, count)

hill0_(.data = NULL, taxon, count)

hill1(.data = NULL, taxon, count)

hill1_(.data = NULL, taxon, count)

hill2(.data = NULL, taxon, count)

hill2_(.data = NULL, taxon, count)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hill_+3A_.data">.data</code></td>
<td>
<p>data in a <code>data.frame</code>, <code>tibble</code>, 
<code>data.table</code>, database etc.</p>
</td></tr>
<tr><td><code id="hill_+3A_taxon">taxon</code></td>
<td>
<p>name of column in <code>.data</code> containing taxa</p>
</td></tr>
<tr><td><code id="hill_+3A_count">count</code></td>
<td>
<p>name of column in <code>.data</code> containing counts</p>
</td></tr>
<tr><td><code id="hill_+3A_a">a</code></td>
<td>
<p>exponent in Hill's diversity number (R, with special cases for 
<code>a</code> in 0, 1, 2 (see details))</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric vector of Hill's numbers
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>hill_()</code>: version suitable for calling from a function
(see package <span class="pkg">lazyeval</span>).
</p>
</li>
<li> <p><code>hill0()</code>: <code class="reqn">N_0</code>
</p>
</li>
<li> <p><code>hill0_()</code>: <code class="reqn">N_0</code>, version suitable for calling 
from a function (see package <span class="pkg">lazyeval</span>).
</p>
</li>
<li> <p><code>hill1()</code>: <code class="reqn">N_1</code>
</p>
</li>
<li> <p><code>hill1_()</code>: <code class="reqn">N_1</code>, version suitable for calling 
from a function (see package <span class="pkg">lazyeval</span>).
</p>
</li>
<li> <p><code>hill2()</code>: <code class="reqn">N_2</code>
</p>
</li>
<li> <p><code>hill2_()</code>: <code class="reqn">N_2</code>, version suitable for calling 
from a function (see package <span class="pkg">lazyeval</span>).
</p>
</li></ul>


<h3>References</h3>

<p>Hill, M.O., 1973. Diversity and Evenness: 
A Unifying Notation and Its Consequences. Ecology 54:427-432
</p>


<h3>See Also</h3>

<p><code><a href="#topic+species_richness">species_richness</a></code>, <code><a href="#topic+shannon">shannon</a></code>, 
<code><a href="#topic+simpson">simpson</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>     hill(
         taxon = c("Euspira pulchella", "Nephtys cirrosa"),  
         count = c(6, 12),
         a = 0
     )
     hill0(
         taxon = c("Euspira pulchella", "Nephtys cirrosa"),  
         count = c(6, 12)
     )
 
</code></pre>

<hr>
<h2 id='hpie'>Hurlbert's Probability of Interspecific Encounter (PIE)</h2><span id='topic+hpie'></span><span id='topic+hpie_'></span>

<h3>Description</h3>

<p>The probability that two individuals selected at random (<em>without</em> 
replacement) from a sample will belong to different species is given by 
(Hurlbert, 1971, p.579, Eq. 3):
</p>
<p style="text-align: center;"><code class="reqn">\Delta_1 = \sum_{i=1}^S (\frac{N_i}{N})(\frac{N-N_i}{N-1}) =
(\frac{N}{N-1})\Delta_2</code>
</p>

<p>where <code class="reqn">\Delta_2</code> (Hurlbert, 1971, p.579, Eq. 4) is the 
probability that two individuals selected at random (<em>with</em> 
replacement) from a sample will belong to different species:
</p>
<p style="text-align: center;"><code class="reqn">\Delta_2 = 1 - \sum_{i=1}^S \pi_i^2</code>
</p>

<p>where  <code class="reqn">N_i</code> is the number of individuals of the <em>i</em>th 
species in the community, <code class="reqn">N</code> is the total number of individuals in the
community, <code class="reqn">\pi_i = N_i/N</code>, and <code class="reqn">S</code> is the number of 
species in the community.
Note that Hurlbert's PIE <code>hpie</code> is the complement of 
<code><a href="#topic+simpson">simpson</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hpie(.data = NULL, taxon, count)

hpie_(.data = NULL, taxon, count)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hpie_+3A_.data">.data</code></td>
<td>
<p>data in a <code>data.frame</code>, <code>tibble</code>, 
<code>data.table</code>, database etc.</p>
</td></tr>
<tr><td><code id="hpie_+3A_taxon">taxon</code></td>
<td>
<p>name of column in <code>.data</code> containing taxa</p>
</td></tr>
<tr><td><code id="hpie_+3A_count">count</code></td>
<td>
<p>name of column in <code>.data</code> containing counts</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector with the probability of interspecific encounter
(PIE).
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>hpie_()</code>: suitable for calling from a function
(see package <span class="pkg">lazyeval</span>).
</p>
</li></ul>


<h3>References</h3>

<p>Hurlbert, S.H., 1971. The Nonconcept of Species Diversity: 
A Critique and Alternative Parameters. Ecology 52:577-586.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simpson">simpson</a></code>, <code><a href="#topic+hurlbert">hurlbert</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>     hpie(
         taxon = c("Euspira pulchella", "Nephtys cirrosa"), 
         count = c(6, 12)
     )
 
</code></pre>

<hr>
<h2 id='hurlbert'>Hurlbert's Expected Number of Species</h2><span id='topic+hurlbert'></span><span id='topic+hurlbert_'></span>

<h3>Description</h3>

<p>The expected number of species in a sample of <code>n</code> individuals:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hurlbert(.data = NULL, taxon, count, n = 100L)

hurlbert_(.data = NULL, taxon, count, n = 100L)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hurlbert_+3A_.data">.data</code></td>
<td>
<p>data in a <code>data.frame</code>, <code>tibble</code>, 
<code>data.table</code>, database etc.</p>
</td></tr>
<tr><td><code id="hurlbert_+3A_taxon">taxon</code></td>
<td>
<p>name of column in <code>.data</code> containing taxa</p>
</td></tr>
<tr><td><code id="hurlbert_+3A_count">count</code></td>
<td>
<p>name of column in <code>.data</code> containing counts</p>
</td></tr>
<tr><td><code id="hurlbert_+3A_n">n</code></td>
<td>
<p>number of individuals in a standard sample</p>
</td></tr>
</table>


<h3>Value</h3>

<p>expected number of species in a sample of <code>n</code> individuals
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>hurlbert_()</code>: version suitable for calling from a function
(see package <span class="pkg">lazyeval</span>).
</p>
</li></ul>


<h3>References</h3>

<p>Hurlbert, S.H., 1971. The Nonconcept of Species Diversity: 
A Critique and Alternative Parameters. Ecology 52:577-586.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>     hurlbert(
         taxon = c("Euspira pulchella", "Nephtys cirrosa"), 
         count = c(4, 6),
         n = 8
     )
     
</code></pre>

<hr>
<h2 id='is_azoic'>Test for Azoic Samples</h2><span id='topic+is_azoic'></span>

<h3>Description</h3>

<p>Case-insensitive test for taxa starting with 'azoi'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_azoic(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is_azoic_+3A_x">x</code></td>
<td>
<p>character vector containing taxa</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Azoic samples need special attention during data analysis. 
They should be marked as 'azoic', and taken care of during
analysis. Note that an azoic sample is not the same as a
record where a taxon has zero counts. The latter should be
removed from further analysis, whereas the former provides
important information.
</p>


<h3>Value</h3>

<p>logical vector, with elements <code>TRUE</code> for azoic samples, 
and <code>FALSE</code> otherwise.
</p>

<hr>
<h2 id='is_binomen'>Binomial Names
<code>is_binomial</code> tests for valid binomial names,
<code>generic_name</code> extracts the genus to which the species belongs,
<code>specific_name</code> extracts the species within the genus.</h2><span id='topic+is_binomen'></span><span id='topic+generic_name'></span><span id='topic+specific_name'></span><span id='topic+strip_sp'></span>

<h3>Description</h3>

<p>Binomial Names
</p>
<p><code>is_binomial</code> tests for valid binomial names,
<code>generic_name</code> extracts the genus to which the species belongs,
<code>specific_name</code> extracts the species within the genus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_binomen(x)

generic_name(x)

specific_name(x)

strip_sp(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is_binomen_+3A_x">x</code></td>
<td>
<p><code><a href="base.html#topic+character">character</a></code> vector, containing the binomial name(s) of 
species (a.k.a. binomen or scientific name)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector with either the generic name or the specific name
of the species.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>generic_name()</code>: extracts the genus to which the species belongs
</p>
</li>
<li> <p><code>specific_name()</code>: extracts the species within the genus
</p>
</li>
<li> <p><code>strip_sp()</code>: strips postfix sp. or spp. from a binomen
</p>
</li></ul>


<h3>Examples</h3>

<pre><code class='language-R'> is_binomen("Venerupis corrugata") # TRUE
 generic_name("Venerupis corrugata") # Venerupis
 specific_name("Venerupis corrugata") # corrugata
 generic_name("venerupis corrugata") # NA (genus part should be capitalized)

</code></pre>

<hr>
<h2 id='iti'>Infaunal Trophic Index (ITI)</h2><span id='topic+iti'></span><span id='topic+iti_'></span><span id='topic+has_iti'></span><span id='topic+has_iti_'></span>

<h3>Description</h3>

<p>Computes the Infaunal Trophic Index (ITI) according to 
Gittenberger &amp; van Loon (2013).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iti(.data = NULL, taxon, count, group = NULL)

iti_(.data = NULL, taxon, count, group = NULL)

has_iti(.data = NULL, taxon, group = NULL)

has_iti_(.data = NULL, taxon, group = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="iti_+3A_.data">.data</code></td>
<td>
<p>data in a <code>data.frame</code>, <code>tibble</code>, 
<code>data.table</code>, database etc.</p>
</td></tr>
<tr><td><code id="iti_+3A_taxon">taxon</code></td>
<td>
<p>species names</p>
</td></tr>
<tr><td><code id="iti_+3A_count">count</code></td>
<td>
<p>counts of individuals (<code>numeric</code>)</p>
</td></tr>
<tr><td><code id="iti_+3A_group">group</code></td>
<td>
<p>sensitivity groups I, II, III, or IV</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Infaunal Trophic Index (ITI) is given by
</p>
<p style="text-align: center;"><code class="reqn">\mathrm{ITI} = 100 \sum_{i=1}^3 \frac{(4-i)}{3} p_i</code>
</p>

<p>where <code class="reqn">p_i</code> is the proportion of species in class <code class="reqn">i</code>, where
</p>

<ul>
<li><p> group   I are suspension feeders (highest quality);
</p>
</li>
<li><p> group  II are interface feeders
</p>
</li>
<li><p> group III are surface deposit feeders and
</p>
</li>
<li><p> group  IV are subsurface deposit feeders (lowest quality). 
</p>
</li></ul>



<h3>Value</h3>

<p>numeric vector of length 1 containing the ITI
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>iti_()</code>: version suitable for calling from a function
(see package <span class="pkg">lazyeval</span>).
</p>
</li>
<li> <p><code>has_iti()</code>: tests if an ITI sensitivity group is available for <code>taxon</code>
(returns <code>TRUE</code> (available) or <code>FALSE</code> (unavailable))
</p>
</li>
<li> <p><code>has_iti_()</code>: version suitable for calling from a function
(see package <span class="pkg">lazyeval</span>).
</p>
</li></ul>


<h3>References</h3>

<p>Gittenberger A. and  W. van Loon, 2013. 
Sensitivities of marine macrozoobenthos to environmental pressures 
in the Netherlands. Nederlandse Faunistische 
Mededelingen 41: 79-112.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>     iti(taxon = c("Euspira pulchella", "Nephtys cirrosa"), count = c(4, 6))
     
 
   data(oosterschelde)
   has_iti(oosterschelde, TAXON)
 
</code></pre>

<hr>
<h2 id='margalef'>Margalef Index of Diversity</h2><span id='topic+margalef'></span><span id='topic+margalef_'></span>

<h3>Description</h3>

<p>Margalef Index of Diversity is given by
</p>
<p style="text-align: center;"><code class="reqn">D = \frac{S-1}{\ln(N)}</code>
</p>

<p>For <code class="reqn">N=1</code>, the index is set to 0.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>margalef(.data = NULL, taxon, count)

margalef_(.data = NULL, taxon, count)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="margalef_+3A_.data">.data</code></td>
<td>
<p>data in a <code>data.frame</code>, <code>tibble</code>, 
<code>data.table</code>, database etc.</p>
</td></tr>
<tr><td><code id="margalef_+3A_taxon">taxon</code></td>
<td>
<p>taxa names (<code>character</code>)</p>
</td></tr>
<tr><td><code id="margalef_+3A_count">count</code></td>
<td>
<p>counts (<code>numeric</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Margalef diversity index (<code>numeric</code> <code>vector</code> of length 1)
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>margalef_()</code>: version suitable for calling from a function
(see package <span class="pkg">lazyeval</span>).
</p>
</li></ul>


<h3>Examples</h3>

<pre><code class='language-R'>margalef(
    taxon = c("Euspira pulchella", "Nephtys cirrosa"), 
    count = c(4, 6)
)
 
</code></pre>

<hr>
<h2 id='northsea'>MWTL North Sea Bentos Data</h2><span id='topic+northsea'></span>

<h3>Description</h3>

<p>MWTL North Sea Bentos Data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>northsea
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 24983 rows and 9 columns.
</p>

<hr>
<h2 id='oosterschelde'>Oosterschelde Marine Benthos Data</h2><span id='topic+oosterschelde'></span>

<h3>Description</h3>

<p>Oosterschelde data set. The Oosterschelde is located in the southwest of
the Netherlands.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oosterschelde
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 4269 rows and 8 columns.
</p>


<h3>Details</h3>

<p>The Oosterschelde data contains the following columns:
</p>

<ul>
<li><p> ID sample identifier
</p>
</li>
<li><p> HABITAT specification of the habitat
</p>
</li>
<li><p> AREA sampled area
</p>
</li>
<li><p> DATE sampling date (YYYY-MM-DD, ISO 8601)
</p>
</li>
<li><p> TAXON standardized taxon code (see WoRMS-website 
<a href="https://www.marinespecies.org/">https://www.marinespecies.org/</a>)
</p>
</li>
<li><p> COUNT number of individuals of 'TAXON'
</p>
</li></ul>



<h3>Note</h3>

<p>This is not the original data set, but a simplified version of it meant
for didactic purposes only! For instance it only contains taxa
identified at the species level. Other taxa have been removed.
</p>


<h3>Source</h3>

<p>Rijkswaterstaat Water, Transport and Living Environment,
Department of Information Management, Lelystad, 
The Netherland (contact: servicedesk-data@rws.nl)
</p>

<hr>
<h2 id='pool'>Pooling</h2><span id='topic+pool'></span><span id='topic+.pool'></span>

<h3>Description</h3>

<p>This function randomly assigns samples to pools of approximately equal area
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pool(sample_id = 1:length(area), area, target_area, max_try = 100L)

.pool(sample_id = 1:length(area), area, target_area, max_try = 100L)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pool_+3A_sample_id">sample_id</code></td>
<td>
<p>sample identifier</p>
</td></tr>
<tr><td><code id="pool_+3A_area">area</code></td>
<td>
<p>sampling area of <code>sample_id</code> (in the same units as 
<code>target_area</code>)</p>
</td></tr>
<tr><td><code id="pool_+3A_target_area">target_area</code></td>
<td>
<p>vector of length 2 containing the lower and upper bound 
of the pooled area (same units as <code>area</code>)</p>
</td></tr>
<tr><td><code id="pool_+3A_max_try">max_try</code></td>
<td>
<p>maximum number of unsuccessful pooling tries before the 
algorithm gives up.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector with idenitifiers (integers) indicating the pool to which 
each sample belongs (NA for samples that could not be pooled)
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>.pool()</code>: internal function not supposed to be called directly.
</p>
</li></ul>

<hr>
<h2 id='read_ambi'>Read and Validate AMBI Sensitivity Data</h2><span id='topic+read_ambi'></span><span id='topic+validate_ambi'></span>

<h3>Description</h3>

<p>This function reads and checks files with AMBI sensitivity data. 
The data should be stored in 'comma separated values' format (csv) 
consisting of two columns:
</p>

<ul>
<li><p>TAXON species name;
</p>
</li>
<li><p>GROUP Roman numeral (I, II, III, IV, V) giving the sensitivity
group
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>read_ambi(filename)

validate_ambi(.data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_ambi_+3A_filename">filename</code></td>
<td>
<p>name of the AMBI sensitivity file (character)</p>
</td></tr>
<tr><td><code id="read_ambi_+3A_.data">.data</code></td>
<td>
<p>table in AMBI-format</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function performs the following tasks:
</p>

<ul>
<li><p>checks the existence of <code>filename</code>;
</p>
</li>
<li><p>checks availablitity of required columns (case insensitive);
</p>
</li>
<li><p>removes redundant spaces;
</p>
</li>
<li><p>removes duplicated records.
</p>
</li></ul>



<h3>Functions</h3>


<ul>
<li> <p><code>validate_ambi()</code>: validator for AMBI-format
</p>
</li></ul>


<h3>References</h3>

<p>Borja, A., J. Franco and V. Perez, 2000. A Marine Biotic Index 
to Establish the Ecological Quality of Soft-Bottom Benthos Within 
European Estuarine and Coastal Environments. 
Marine Pollution Bulletin 40:1100-1114
</p>

<hr>
<h2 id='read_beqi2'>Read and Validate BEQI2 Input Files</h2><span id='topic+read_beqi2'></span><span id='topic+validate_beqi2'></span>

<h3>Description</h3>

<p>This function reads and checks BEQI2 input files. The format has been
specified in Van Loon (2013) and is described in the vignette of the BENMMI-package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_beqi2(filename)

validate_beqi2(.data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_beqi2_+3A_filename">filename</code></td>
<td>
<p>name of BEQI2 input file (<code>character</code>)</p>
</td></tr>
<tr><td><code id="read_beqi2_+3A_.data">.data</code></td>
<td>
<p>table in BEQI2-format</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function performs the following tasks:
</p>

<ul>
<li><p>checks the existence of <code>filename</code>;
</p>
</li>
<li><p>checks availablitity of required columns (case insensitive);
</p>
</li>
<li><p>make column names with aggregation data case-insensitive;
</p>
</li>
<li><p>removes redundant spaces;
</p>
</li>
<li><p>checks if DATE-field adheres to ISO 8601 (YYYY-mm-dd);
</p>
</li>
<li><p>constructs a unique identifier <code>ID</code> by concatenating 
columns <code>OBJECTID</code> and <code>DATE</code>;
</p>
</li>
<li><p>checks that each <code>ID</code> has a unique <code>AREA</code>;
</p>
</li>
<li><p>checks azoic samples for VALUE=0;
</p>
</li>
<li><p>removes records with VALUE=0, not belonging to azoic samples;
</p>
</li>
<li><p>checks VALUE-field on missing values;
</p>
</li>
<li><p>checks if VALUE-field is an integer;
</p>
</li></ul>



<h3>Functions</h3>


<ul>
<li> <p><code>validate_beqi2()</code>: validator for BEQI2-format
</p>
</li></ul>


<h3>References</h3>

<p>Willem van Loon, 2013. BEQI2 INPUT FORMAT. See the package-vignette of the BENMMI-package.
</p>

<hr>
<h2 id='read_iti'>Read and Validate Infaunal Trophic Index Files</h2><span id='topic+read_iti'></span><span id='topic+validate_iti'></span>

<h3>Description</h3>

<p>This function reads and checks files containing Infaunal Trophic Index 
(ITI) data (Gittenberger &amp; Van Loon, 2013)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_iti(filename)

validate_iti(.data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_iti_+3A_filename">filename</code></td>
<td>
<p>name of the ITI file (character).</p>
</td></tr>
<tr><td><code id="read_iti_+3A_.data">.data</code></td>
<td>
<p>table in ITI-format</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function performs the following tasks:
</p>

<ul>
<li><p>checks the existence of <code>filename</code>;
</p>
</li>
<li><p>checks availablitity of required columns (case insensitive), 
i.e., TAXON and GROUP;
</p>
</li>
<li><p>removes redundant spaces;
</p>
</li>
<li><p>removes duplicated records.
</p>
</li>
<li><p>checks if all ITI classes are I, II, III, or IV
</p>
</li></ul>

<p>The column 'GROUP' contains the Roman numerals I, II, III, and IV, with
the following meaning:
</p>

<ul>
<li><p>  I:  suspension feeders;
</p>
</li>
<li><p> II:  interface feeders;
</p>
</li>
<li><p>III:  surface deposit feeders;
</p>
</li>
<li><p> IV:  subsurface deposit feeders.
</p>
</li></ul>



<h3>Value</h3>

<p>A data frame with columns <code>TAXON</code> containing taxa and 
<code>GROUP</code> containing user-defined ITI-groups
(see Gittenberger &amp; Van Loon, 2013).
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>validate_iti()</code>: validator for ITI-format
</p>
</li></ul>


<h3>References</h3>

<p>Gittenberger A. and  W. van Loon, 2013. 
Sensitivities of marine macrozoobenthos to environmental pressures 
in the Netherlands. Nederlandse Faunistische 
Mededelingen 41: 79-112.
</p>

<hr>
<h2 id='read_ref'>Read and Validate Habitat References Files</h2><span id='topic+read_ref'></span><span id='topic+validate_ref'></span>

<h3>Description</h3>

<p>This function reads and checks files with reference values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_ref(filename, indicators = c("S", "H", "AMBI"))

validate_ref(.data, indicators = c("S", "H", "AMBI"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_ref_+3A_filename">filename</code></td>
<td>
<p>name of the habitat reference file (<code>character</code>)</p>
</td></tr>
<tr><td><code id="read_ref_+3A_indicators">indicators</code></td>
<td>
<p>indicators to be processed (<code>character</code>, 
see details)</p>
</td></tr>
<tr><td><code id="read_ref_+3A_.data">.data</code></td>
<td>
<p>table in REF-format</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function performs the following tasks:
</p>

<ul>
<li><p>checks the existence of <code>filename</code>;
</p>
</li>
<li><p>checks availablitity of required columns (case insensitive);
</p>
</li>
<li><p>removes redundant spaces
</p>
</li>
<li><p>removes duplicated records
</p>
</li></ul>

<p>Argument <code>indicators</code> is a <code>character</code> vector of additional benthic 
indicators to be checked for. For example, if <code>indicators = "ITI"</code>, then
the habitat reference file should also contain columns ITIREF and ITIBAD.
Implemented indicators are N, LNN, S, D, SN, SNA, H, L, AMBI, ITI, PIE, N2 (see package vignette).
</p>
<p>The format of the habitat reference file is documented in the 
BEQI2-package vignette.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>validate_ref()</code>: validator for REF-format
</p>
</li></ul>


<h3>References</h3>

<p>Van Loon, W, 2013. Loon2013-BEQI2-Specs-Ecotopes-27nov.doc
</p>

<hr>
<h2 id='read_taxa'>Read and Validate Taxa Data</h2><span id='topic+read_taxa'></span><span id='topic+get_taxa'></span><span id='topic+validate_taxa'></span>

<h3>Description</h3>

<p>This function reads files in the taxa format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_taxa(filename)

get_taxa()

validate_taxa(.data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_taxa_+3A_filename">filename</code></td>
<td>
<p>name of taxa file</p>
</td></tr>
<tr><td><code id="read_taxa_+3A_.data">.data</code></td>
<td>
<p>table in taxa-format</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Taxa files have the following format:
</p>

<ul>
<li><p>group taxonomic group
</p>
</li>
<li><p>provided provided taxon name
</p>
</li>
<li><p>accepted accepted taxon name
</p>
</li>
<li><p>level taxonomic level
</p>
</li></ul>

<p>Other columns are allowed, but silently ingored.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>get_taxa()</code>: get default taxa list (TWN list extended with species Southern North Sea)
</p>
</li>
<li> <p><code>validate_taxa()</code>: validator for taxa-format
</p>
</li></ul>

<hr>
<h2 id='read_twn'>Read and Validate Taxa Waterbeheer Nederland (TWN) Data</h2><span id='topic+read_twn'></span><span id='topic+get_worms'></span><span id='topic+validate_twn'></span>

<h3>Description</h3>

<p>This function reads files in the Taxa Waterbeheer Nederland (TWN) format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_twn(filename)

get_worms()

validate_twn(.data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_twn_+3A_filename">filename</code></td>
<td>
<p>name of TWN file (<code>character</code>)</p>
</td></tr>
<tr><td><code id="read_twn_+3A_.data">.data</code></td>
<td>
<p>table in TWN-format</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function adds a new column <code>taxon</code>. Its contents depending 
on TWN-status:
</p>

<ul>
<li><p>status = 10 taxonname
</p>
</li>
<li><p>status = 20 prefername
</p>
</li>
<li><p>status = 80 parentname
</p>
</li></ul>



<h3>Value</h3>

<p>a <code>tibble</code> with four columns:
</p>

<ul>
<li><p>GROUP TWN/WoRMS taxon group
</p>
</li>
<li><p>LEVEL TWN/WoRMS taxon level
</p>
</li>
<li><p>FROM taxon name to convert from
</p>
</li>
<li><p>TO taxon name to convert to
</p>
</li></ul>



<h3>Functions</h3>


<ul>
<li> <p><code>get_worms()</code>: get default WoRMS list (TWN list extended with species Southern North Sea)
</p>
</li>
<li> <p><code>validate_twn()</code>: validator for TWN-format
</p>
</li></ul>


<h3>References</h3>

<p><a href="https://taxainfo.nl/">https://taxainfo.nl/</a>
</p>

<hr>
<h2 id='rygg'>Rygg's Index of Diversity</h2><span id='topic+rygg'></span><span id='topic+rygg_'></span>

<h3>Description</h3>

<p>Rygg's index of diversity is given by
</p>
<p style="text-align: center;"><code class="reqn">SN = \frac{\ln(S)}{\ln(\ln(N))}</code>
</p>

<p>The adjusted version of Rygg's index which gives more consistent values
for smaller <code>S=2, N=2, N=3</code> and <code>S=3, N=3</code> is
</p>
<p style="text-align: center;"><code class="reqn">SN = \frac{\ln(S)}{\ln(\ln(N+1)+1)}</code>
</p>



<h3>Usage</h3>

<pre><code class='language-R'>rygg(.data = NULL, taxon, count, adjusted = FALSE)

rygg_(.data = NULL, taxon, count, adjusted = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rygg_+3A_.data">.data</code></td>
<td>
<p>data in a <code>data.frame</code>, <code>tibble</code>, 
<code>data.table</code>, database etc.</p>
</td></tr>
<tr><td><code id="rygg_+3A_taxon">taxon</code></td>
<td>
<p>taxa names (<code>character</code>)</p>
</td></tr>
<tr><td><code id="rygg_+3A_count">count</code></td>
<td>
<p>counts (<code>numeric</code>)</p>
</td></tr>
<tr><td><code id="rygg_+3A_adjusted">adjusted</code></td>
<td>
<p>(defaults to <code>FALSE</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Rygg's index of diversity (<code>numeric</code> <code>vector</code> of length 1)
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>rygg_()</code>: version suitable for calling from a function
(see package <span class="pkg">lazyeval</span>).
</p>
</li></ul>


<h3>Note</h3>

<p>Rygg's index is not defined for <code class="reqn">N=exp(1)</code>. For 
<code class="reqn">N \leq exp(1)</code>, <code>rygg</code> returns 
<code><a href="base.html#topic+NA_real_">NA_real_</a></code>.
</p>


<h3>References</h3>

<p>Rygg, B. (2006). Developing indices for quality-status 
classification of marine soft-bottom fauna in Norway. 
Norwegian Institute for Water Research, Oslo, Norway. 
NIVA Report SNO 5208-2006.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> rygg(
     taxon = c("Euspira pulchella", "Nephtys cirrosa"), 
     count = c(4, 6)
 )
 
</code></pre>

<hr>
<h2 id='shannon'>Shannon's Index or Entropy</h2><span id='topic+shannon'></span><span id='topic+shannon_'></span>

<h3>Description</h3>

<p>Compute entropy according to Shannon (1948)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shannon(.data = NULL, taxon, count, base = 2)

shannon_(.data = NULL, taxon, count, base = 2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="shannon_+3A_.data">.data</code></td>
<td>
<p>data in a <code>data.frame</code>, <code>tibble</code>, 
<code>data.table</code>, database etc.</p>
</td></tr>
<tr><td><code id="shannon_+3A_taxon">taxon</code></td>
<td>
<p>taxa names (<code>character</code>)</p>
</td></tr>
<tr><td><code id="shannon_+3A_count">count</code></td>
<td>
<p>counts (<code>numeric</code>)</p>
</td></tr>
<tr><td><code id="shannon_+3A_base">base</code></td>
<td>
<p>the base with respect to which logarithms are computed. 
Defaults to 2 (unit: bits).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Shannon's entropy
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>shannon_()</code>: version suitable for calling from a function
(see package <span class="pkg">lazyeval</span>).
</p>
</li></ul>


<h3>References</h3>

<p>Shannon, C. E., 1948. A Mathematical Theory of Communication.
Bell System Technical Journal 27: 379-423.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> shannon(
     taxon = c("Euspira pulchella", "Nephtys cirrosa"), 
     count = c(4, 6)
 )

</code></pre>

<hr>
<h2 id='simpson'>Simpson's Measure of Concentration</h2><span id='topic+simpson'></span><span id='topic+simpson_'></span>

<h3>Description</h3>

<p>The probability that two inidividuals selected at random (with replacement, 
Hurlbert, 1971, p.579) from a sample will belong to the same species. For
an infinite sample Simpson's Index is given by (Peet, 1974):
</p>
<p style="text-align: center;"><code class="reqn">\lambda = \sum_{i=1}^S p_i^2</code>
</p>

<p>For a finite sample by:
</p>
<p style="text-align: center;"><code class="reqn">L = \sum_{i=1}^S \frac{n_i (n_i-1)}{N (N-1)}</code>
</p>

<p>where <code class="reqn">p_i</code> the proportion of the individuals in species <code class="reqn">i</code>, 
<code class="reqn">n_i</code> the number of individuals in species 
<code class="reqn">i</code> (relative <code><a href="#topic+abundance">abundance</a></code>), and <code class="reqn">N</code> the total number 
of individuals (<code><a href="#topic+total_abundance">total_abundance</a></code>). The finite sample case 
has been implemented in function <code>simpson</code> (and <code>simpson_</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simpson(.data = NULL, taxon, count)

simpson_(.data = NULL, taxon, count)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simpson_+3A_.data">.data</code></td>
<td>
<p>data in a <code>data.frame</code>, <code>tibble</code>, 
<code>data.table</code>, database etc.</p>
</td></tr>
<tr><td><code id="simpson_+3A_taxon">taxon</code></td>
<td>
<p>name of column in <code>.data</code> containing taxa</p>
</td></tr>
<tr><td><code id="simpson_+3A_count">count</code></td>
<td>
<p>name of column in <code>.data</code> containing counts</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The probability that two inidividuals selected at random from a 
sample will belong to the same species.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>simpson_()</code>: version suitable for calling from a function
(see package <span class="pkg">lazyeval</span>).
</p>
</li></ul>


<h3>References</h3>

<p>Peet, R. K. 1974, The Measurement of Species Diversity. Annual 
Review of Ecology and Systematics 5:285-307.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hpie">hpie</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>     simpson(
         taxon = c("Euspira pulchella", "Nephtys cirrosa"),  
         count = c(6, 12)
     )
 
</code></pre>

<hr>
<h2 id='species_richness'>Species Richness</h2><span id='topic+species_richness'></span><span id='topic+species_richness_'></span>

<h3>Description</h3>

<p>Species richness (<code class="reqn">S</code>) is defined as the number of taxa 
(lowest identification level possible) per sampling unit 
(data pool or box core sample).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>species_richness(.data = NULL, taxon, count = NULL)

species_richness_(.data = NULL, taxon, count = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="species_richness_+3A_.data">.data</code></td>
<td>
<p>data in a <code>data.frame</code>, <code>tibble</code>, 
<code>data.table</code>, database etc.</p>
</td></tr>
<tr><td><code id="species_richness_+3A_taxon">taxon</code></td>
<td>
<p>taxa names (<code>character</code>)</p>
</td></tr>
<tr><td><code id="species_richness_+3A_count">count</code></td>
<td>
<p>number of individuals for each taxon (<code>numeric</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>species richness (<code>integer</code> vector of length 1)
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>species_richness_()</code>: version suitable for calling from a function
(see package <span class="pkg">lazyeval</span>).
</p>
</li></ul>


<h3>Examples</h3>

<pre><code class='language-R'> species_richness(
     taxon = c("Euspira pulchella", "Nephtys cirrosa"), 
     count = c(4, 6)
 )
 
</code></pre>

<hr>
<h2 id='strip_spaces'>Remove Redundant Spaces</h2><span id='topic+strip_spaces'></span>

<h3>Description</h3>

<p>This function removes redundant spaces from character vectors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>strip_spaces(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="strip_spaces_+3A_x">x</code></td>
<td>
<p>character vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector without trailing or multiple spaces
</p>

<hr>
<h2 id='to_worms'>Convert Taxon Names to Comply with WoRMS</h2><span id='topic+to_worms'></span><span id='topic+is_worms'></span><span id='topic+is_worms_'></span><span id='topic+is_accepted'></span><span id='topic+is_accepted_'></span>

<h3>Description</h3>

<p>Taxon names are standardized according to the World Register of 
Marine Species (WoRMS) database. The conversion is case-insensitive.
For this conversion, the TWN-list (Taxa Water management the Netherlands) 
is used, extended with species of the Southern North Sea. 
See references below for download locations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>to_worms(taxon, worms = NULL)

is_worms(.data = NULL, taxon)

is_worms_(.data, taxon)

is_accepted(.data = NULL, taxon)

is_accepted_(.data, taxon)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="to_worms_+3A_taxon">taxon</code></td>
<td>
<p><code><a href="base.html#topic+character">character</a></code> vector, containing taxon names</p>
</td></tr>
<tr><td><code id="to_worms_+3A_worms">worms</code></td>
<td>
<p>an optional table usually created with <code><a href="#topic+read_twn">read_twn</a></code>.</p>
</td></tr>
<tr><td><code id="to_worms_+3A_.data">.data</code></td>
<td>
<p>data in a <code>data.frame</code>, <code>tibble</code>, 
<code>data.table</code>, database etc.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector with WoRMS compliant species names
</p>
<p><code>TRUE</code> for WoRMS compliant species names, 
<code>FALSE</code> otherwise.
</p>
<p><code>TRUE</code> for WoRMS/TWN compliant species names, 
<code>FALSE</code> otherwise.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>is_worms()</code>: check if a taxon complies with WoRMS
</p>
</li>
<li> <p><code>is_worms_()</code>: as <code>is_worms</code> but suitable for calling from a 
function (see package <span class="pkg">lazyeval</span>).
</p>
</li>
<li> <p><code>is_accepted()</code>: check if a taxon complies with WoRMS/TWN
</p>
</li>
<li> <p><code>is_accepted_()</code>: as <code>is_accepted</code> but suitable for calling from a 
function (see package <span class="pkg">lazyeval</span>).
</p>
</li></ul>


<h3>References</h3>

<p><a href="https://www.marinespecies.org/">https://www.marinespecies.org/</a>
</p>
<p><a href="https://taxainfo.nl/">https://taxainfo.nl/</a>
</p>

<hr>
<h2 id='total_abundance'>Total Abundance</h2><span id='topic+total_abundance'></span><span id='topic+total_abundance_'></span><span id='topic+lnn'></span><span id='topic+lnn_'></span>

<h3>Description</h3>

<p>The total number of individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>total_abundance(.data = NULL, count, na.rm = FALSE)

total_abundance_(.data = NULL, count, na.rm = FALSE)

lnn(.data = NULL, count, na.rm = FALSE)

lnn_(.data = NULL, count, na.rm = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="total_abundance_+3A_.data">.data</code></td>
<td>
<p>data in a <code>data.frame</code>, <code>tibble</code>, 
<code>data.table</code>, database etc.</p>
</td></tr>
<tr><td><code id="total_abundance_+3A_count">count</code></td>
<td>
<p>counts (<code>numeric</code>)</p>
</td></tr>
<tr><td><code id="total_abundance_+3A_na.rm">na.rm</code></td>
<td>
<p>Should missing values (including <code>NaN</code>) 
be removed? (<code>logical</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>total number of individuals (<code>integer</code>)
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>total_abundance_()</code>: version suitable for calling from a function
(see package <span class="pkg">lazyeval</span>).
</p>
</li>
<li> <p><code>lnn()</code>: natural log of total abundance + 1
(see package <span class="pkg">lazyeval</span>).
</p>
</li>
<li> <p><code>lnn_()</code>: version of lnn suitable for calling from a function
(see package <span class="pkg">lazyeval</span>).
</p>
</li></ul>


<h3>Examples</h3>

<pre><code class='language-R'> total_abundance(count = c(4, 6))
     
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
