<!DOCTYPE html><html lang="en"><head><title>Help for package betaclust</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {betaclust}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AUC_WD_metric'><p>AUC and WD function</p></a></li>
<li><a href='#beta_k'><p>Fit the K.. model</p></a></li>
<li><a href='#beta_kn'><p>Fit the KN. model</p></a></li>
<li><a href='#beta_kr'><p>Fit the K.R Model</p></a></li>
<li><a href='#betaclust'><p>The betaclust wrapper function</p></a></li>
<li><a href='#DMC_identification'><p>The DMC identification function</p></a></li>
<li><a href='#ecdf.betaclust'><p>The empirical cumulative distribution function plot</p></a></li>
<li><a href='#em_aic'><p>Akaike Information Criterion</p></a></li>
<li><a href='#em_bic'><p>Bayesian Information Criterion</p></a></li>
<li><a href='#em_icl'><p>Integrated Complete-data Likelihood (ICL) Criterion</p></a></li>
<li><a href='#legacy.data'><p>MethylationEPIC manifest data.</p></a></li>
<li><a href='#pca.methylation.data'><p>DNA methylation data from patients with prostate cancer</p></a></li>
<li><a href='#plot.betaclust'><p>Plots for visualizing the betaclust class object</p></a></li>
<li><a href='#summary.betaclust'><p>Summarizing the beta mixture model fits</p></a></li>
<li><a href='#threshold'><p>Thresholds under the K.. and the KN. models</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Family of Beta Mixture Models for Clustering Beta-Valued DNA
Methylation Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.4</td>
</tr>
<tr>
<td>Description:</td>
<td>A family of novel beta mixture models (BMMs) has been developed by Majumdar et al. (2022) &lt;<a href="https://doi.org/10.48550%2FarXiv.2211.01938">doi:10.48550/arXiv.2211.01938</a>&gt; to appositely model the beta-valued cytosine-guanine dinucleotide (CpG) sites, to objectively identify methylation state thresholds and to identify the differentially methylated CpG (DMC) sites using a model-based clustering approach. The family of beta mixture models employs different parameter constraints applicable to different study settings. The EM algorithm is used for parameter estimation, with a novel approximation during the M-step providing tractability and ensuring computational feasibility.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>foreach, doParallel, stats, utils, ggplot2, plotly, scales,
pROC</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Koyel Majumdar &lt;koyel.majumdar@ucdconnect.ie&gt;</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-09-24 09:00:43 UTC; koyel</td>
</tr>
<tr>
<td>Author:</td>
<td>Koyel Majumdar <a href="https://orcid.org/0000-0003-4692-9002"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Romina Silva [aut],
  Antoinette Sabrina Perry [aut],
  Ronald William Watson [aut],
  Andrea Rau <a href="https://orcid.org/0000-0001-6469-488X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Florence Jaffrezic [aut],
  Thomas Brendan Murphy
    <a href="https://orcid.org/0000-0002-5668-7046"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Isobel Claire Gormley
    <a href="https://orcid.org/0000-0001-7713-681X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-24 09:30:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='AUC_WD_metric'>AUC and WD function</h2><span id='topic+AUC_WD_metric'></span>

<h3>Description</h3>

<p>Function to find the level of similarities between the <code class="reqn">R</code> cumulative distributions estimated in each of the <code class="reqn">K</code> clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AUC_WD_metric(alpha, delta, K, R)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AUC_WD_metric_+3A_alpha">alpha</code></td>
<td>
<p>The first shape parameter estimated for the beta mixture model.</p>
</td></tr>
<tr><td><code id="AUC_WD_metric_+3A_delta">delta</code></td>
<td>
<p>The second shape parameter estimated for the beta mixture model.</p>
</td></tr>
<tr><td><code id="AUC_WD_metric_+3A_k">K</code></td>
<td>
<p>The number of clusters estimated.</p>
</td></tr>
<tr><td><code id="AUC_WD_metric_+3A_r">R</code></td>
<td>
<p>The number of sample types in the dataset.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function to find the level of similarities between the <code class="reqn">R</code> cumulative distributions estimated in each of the <code class="reqn">K</code> clusters.
</p>


<h3>Value</h3>

<p>The list with AUC and WD values.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+betaclust">betaclust</a></code>
</p>

<hr>
<h2 id='beta_k'>Fit the K.. model</h2><span id='topic+beta_k'></span>

<h3>Description</h3>

<p>Fit the K.. model from the <code><a href="#topic+betaclust">betaclust</a></code> family of beta mixture models for DNA methylation data.
The K.. model analyses a single DNA sample type and identifies the thresholds between the different methylation states.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>beta_k(data, M = 3, parallel_process = FALSE, seed = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="beta_k_+3A_data">data</code></td>
<td>
<p>A dataframe of dimension <code class="reqn">C \times N</code> containing methylation values for <code class="reqn">C</code> CpG sites from <code class="reqn">R = 1</code> sample type collected from <code class="reqn">N</code> patients.
Samples are grouped together in the dataframe such that the columns are ordered as Sample1_Patient1, Sample1_Patient2, etc.</p>
</td></tr>
<tr><td><code id="beta_k_+3A_m">M</code></td>
<td>
<p>Number of methylation states to be identified in a DNA sample type.</p>
</td></tr>
<tr><td><code id="beta_k_+3A_parallel_process">parallel_process</code></td>
<td>
<p>The &quot;TRUE&quot; option results in parallel processing of the models for increased computational efficiency. The default option has been set as &quot;FALSE&quot; due to package testing limitations.</p>
</td></tr>
<tr><td><code id="beta_k_+3A_seed">seed</code></td>
<td>
<p>Seed to allow for reproducibility (default = NULL).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The K.. model clusters each of the <code class="reqn">C</code> CpG sites into one of <code class="reqn">K</code> methylation states, based on data from <code class="reqn">N</code> patients for one DNA sample type (i.e. <code class="reqn">R = 1</code>).
As each CpG site can belong to any of the <code class="reqn">M = 3</code> methylation states (hypomethylated, hemimethylated or hypermethylated), the default value of <code class="reqn">K = M = 3</code>.
Under the K.. model the shape parameters of each cluster are constrained to be equal for each patient. The returned object from this function can be passed as an input parameter to the
<code><a href="#topic+threshold">threshold</a></code> function available in this package to calculate the thresholds between the methylation states.
</p>


<h3>Value</h3>

<p>A list containing:
</p>

<ul>
<li><p> cluster_size - The total number of CpG sites in each of the K clusters.
</p>
</li>
<li><p> llk - A vector containing the log-likelihood value at each step of the EM algorithm.
</p>
</li>
<li><p> alpha - The first shape parameter for the beta mixture model.
</p>
</li>
<li><p> delta - The second shape parameter for the beta mixture model.
</p>
</li>
<li><p> tau - The estimated mixing proportion for each cluster.
</p>
</li>
<li><p> z - A matrix of dimension <code class="reqn">C \times K</code> containing the posterior probability of each CpG site belonging to each of the <code class="reqn">K</code> clusters.
</p>
</li>
<li><p> classification - The classification corresponding to z, i.e. map(z).
</p>
</li>
<li><p> uncertainty - The uncertainty of each CpG site's clustering.    </p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+beta_kn">beta_kn</a></code>
</p>
<p><code><a href="#topic+betaclust">betaclust</a></code>
</p>
<p><code><a href="#topic+threshold">threshold</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>my.seed &lt;- 190
M &lt;- 3
data_output &lt;- beta_k(pca.methylation.data[1:30,2:5], M,
                      parallel_process = FALSE, seed = my.seed)
thresholds &lt;- threshold(data_output, pca.methylation.data[1:30,2:5], "K..")
</code></pre>

<hr>
<h2 id='beta_kn'>Fit the KN. model</h2><span id='topic+beta_kn'></span>

<h3>Description</h3>

<p>Fit the KN. model from the <code><a href="#topic+betaclust">betaclust</a></code> family of beta mixture models for DNA methylation data.
The KN. model analyses a single DNA sample type and identifies the thresholds between the different methylation states.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>beta_kn(data, M = 3, parallel_process = FALSE, seed = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="beta_kn_+3A_data">data</code></td>
<td>
<p>A dataframe of dimension <code class="reqn">C \times N</code> containing methylation values for <code class="reqn">C</code> CpG sites from <code class="reqn">R = 1</code> sample type collected from <code class="reqn">N</code> patients.
Samples are grouped together in the dataframe such that the columns are ordered as Sample1_Patient1, Sample1_Patient2, etc.</p>
</td></tr>
<tr><td><code id="beta_kn_+3A_m">M</code></td>
<td>
<p>Number of methylation states to be identified in a DNA sample type.</p>
</td></tr>
<tr><td><code id="beta_kn_+3A_parallel_process">parallel_process</code></td>
<td>
<p>The &quot;TRUE&quot; option results in parallel processing of the models for increased computational efficiency. The default option has been set as &quot;FALSE&quot; due to package testing limitations.</p>
</td></tr>
<tr><td><code id="beta_kn_+3A_seed">seed</code></td>
<td>
<p>Seed to allow for reproducibility (default = NULL).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The KN. model clusters each of the <code class="reqn">C</code> CpG sites into one of <code class="reqn">K</code> methylation states, based on data from <code class="reqn">N</code> patients for one DNA sample type (i.e. <code class="reqn">R = 1</code>).
As each CpG site can belong to any of the <code class="reqn">M = 3</code> methylation states (hypomethylated, hemimethylated or hypermethylated), the default value of <code class="reqn">K = M = 3</code>.
The KN. model differs from the K.. model as it is less parsimonious, allowing cluster and patient-specific shape parameters. The returned object can be passed as an input parameter to the
<code><a href="#topic+threshold">threshold</a></code> function available in this package to calculate the thresholds between the methylation states.
</p>


<h3>Value</h3>

<p>A list containing:
</p>

<ul>
<li><p> cluster_size - The total number of CpG sites in each of the K clusters.
</p>
</li>
<li><p> llk - A vector containing the log-likelihood value at each step of the EM algorithm.
</p>
</li>
<li><p> alpha - The first shape parameter for the beta mixture model.
</p>
</li>
<li><p> delta - The second shape parameter for the mixture model.
</p>
</li>
<li><p> tau - The estimated mixing proportion for each cluster.
</p>
</li>
<li><p> z - A matrix of dimension <code class="reqn">C \times K</code> containing the posterior probability of each CpG site belonging to each of the <code class="reqn">K</code> clusters.
</p>
</li>
<li><p> classification - The classification corresponding to z, i.e. map(z).
</p>
</li>
<li><p> uncertainty - The uncertainty of each CpG site's clustering.    </p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+beta_k">beta_k</a></code>
</p>
<p><code><a href="#topic+betaclust">betaclust</a></code>
</p>
<p><code><a href="#topic+threshold">threshold</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>my.seed &lt;- 190
M &lt;- 3
data_output &lt;- beta_kn(pca.methylation.data[1:30,2:5], M,
                       parallel_process = FALSE, seed = my.seed)
thresholds &lt;- threshold(data_output, pca.methylation.data[1:30,2:5], "KN.")
</code></pre>

<hr>
<h2 id='beta_kr'>Fit the K.R Model</h2><span id='topic+beta_kr'></span>

<h3>Description</h3>

<p>A beta mixture model for identifying differentially methylated CpG sites between <code class="reqn">R</code> DNA sample types collected from <code class="reqn">N</code> patients.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>beta_kr(data, M = 3, N, R, parallel_process = FALSE, seed = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="beta_kr_+3A_data">data</code></td>
<td>
<p>A dataframe of dimension <code class="reqn">C \times NR</code> containing methylation values for <code class="reqn">C</code> CpG sites from <code class="reqn">R</code> sample types collected from <code class="reqn">N</code> patients.
Samples are grouped together in the dataframe such that the columns are ordered as Sample1_Patient1, Sample1_Patient2, Sample2_Patient1, Sample2_Patient2, etc.</p>
</td></tr>
<tr><td><code id="beta_kr_+3A_m">M</code></td>
<td>
<p>Number of methylation states to be identified.</p>
</td></tr>
<tr><td><code id="beta_kr_+3A_n">N</code></td>
<td>
<p>Number of patients in the study.</p>
</td></tr>
<tr><td><code id="beta_kr_+3A_r">R</code></td>
<td>
<p>Number of sample types collected from each patient for study.</p>
</td></tr>
<tr><td><code id="beta_kr_+3A_parallel_process">parallel_process</code></td>
<td>
<p>The &quot;TRUE&quot; option results in parallel processing of the models for increased computational efficiency. The default option has been set as &quot;FALSE&quot; due to package testing limitations.</p>
</td></tr>
<tr><td><code id="beta_kr_+3A_seed">seed</code></td>
<td>
<p>Seed to allow for reproducibility (default = NULL).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The K.R model allows identification of the differentially methylated CpG sites between the <code class="reqn">R</code> DNA sample types collected from each of <code class="reqn">N</code> patients.
As each CpG site in a DNA sample can belong to one of <code class="reqn">M</code> methylation states, there can be <code class="reqn">K=M^R</code> methylation state changes between <code class="reqn">R</code> DNA sample types.
The shape parameters vary for each DNA sample type but are constrained to be equal for each patient. An initial clustering using k-means is performed to identify <code class="reqn">K</code> clusters. The resulting clustering solution is provided as
starting values to the Expectation-Maximisation algorithm. A digamma approximation is used to obtain the maximised
parameters in the M-step.
</p>


<h3>Value</h3>

<p>A list containing:
</p>

<ul>
<li><p> cluster_size - The total number of CpG sites in each of the K clusters.
</p>
</li>
<li><p> llk - A vector containing the log-likelihood value at each step of the EM algorithm.
</p>
</li>
<li><p> alpha - The first shape parameter for the beta mixture model.
</p>
</li>
<li><p> delta - The second shape parameter for the beta mixture model.
</p>
</li>
<li><p> tau - The estimated mixing proportion for each cluster.
</p>
</li>
<li><p> z - A matrix of dimension <code class="reqn">C \times K</code> containing the posterior probability of each CpG site belonging to each of the <code class="reqn">K</code> clusters.
</p>
</li>
<li><p> classification - The classification corresponding to z, i.e. map(z).
</p>
</li>
<li><p> uncertainty - The uncertainty of each CpG site's clustering.
</p>
</li>
<li><p> DM - The AUC and WD metric for distribution similarity in each cluster. </p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+betaclust">betaclust</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>my.seed &lt;- 190
M &lt;- 3
N &lt;- 4
R &lt;- 2
data_output = beta_kr(pca.methylation.data[1:30,2:9], M, N, R,
                      parallel_process = FALSE, seed = my.seed)
</code></pre>

<hr>
<h2 id='betaclust'>The betaclust wrapper function</h2><span id='topic+betaclust'></span>

<h3>Description</h3>

<p>A family of model-based clustering techniques to identify methylation states in beta-valued DNA methylation data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>betaclust(
  data,
  M = 3,
  N,
  R,
  model_names = "K..",
  model_selection = "BIC",
  parallel_process = FALSE,
  seed = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="betaclust_+3A_data">data</code></td>
<td>
<p>A dataframe of dimension <code class="reqn">C \times NR</code> containing methylation values for <code class="reqn">C</code> CpG sites from <code class="reqn">R</code> sample types collected from <code class="reqn">N</code> patients.
Samples are grouped together in the dataframe such that the columns are ordered as Sample1_Patient1, Sample1_Patient2, Sample2_Patient1, Sample2_Patient2, etc.</p>
</td></tr>
<tr><td><code id="betaclust_+3A_m">M</code></td>
<td>
<p>Number of methylation states to be identified in a DNA sample type.</p>
</td></tr>
<tr><td><code id="betaclust_+3A_n">N</code></td>
<td>
<p>Number of patients in the study.</p>
</td></tr>
<tr><td><code id="betaclust_+3A_r">R</code></td>
<td>
<p>Number of sample types collected from each patient for the study.</p>
</td></tr>
<tr><td><code id="betaclust_+3A_model_names">model_names</code></td>
<td>
<p>Models to run from the set of models, K.., KN. and K.R, default = K.. . See details.</p>
</td></tr>
<tr><td><code id="betaclust_+3A_model_selection">model_selection</code></td>
<td>
<p>Information criterion used for model selection. Options are AIC, BIC or ICL (default = BIC).</p>
</td></tr>
<tr><td><code id="betaclust_+3A_parallel_process">parallel_process</code></td>
<td>
<p>The &quot;TRUE&quot; option results in parallel processing of the models for increased computational efficiency. The default option has been set as &quot;FALSE&quot; due to package testing limitations.</p>
</td></tr>
<tr><td><code id="betaclust_+3A_seed">seed</code></td>
<td>
<p>Seed to allow for reproducibility (default = NULL).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is a wrapper function which can be used to fit all three models (K.., KN., K.R) within a single function.
</p>
<p>The K.. and KN. models are used to analyse a single DNA sample type (<code class="reqn">R = 1</code>) and cluster the <code class="reqn">C</code> CpG sites into the <code class="reqn">K</code> clusters which represent the different methylation states in a DNA sample type. As each CpG site can belong to any of the <code class="reqn">M=3</code> methylation states (hypomethylation, hemimethylation and hypermethylation), the default value for <code class="reqn">K=M=3</code>.
The thresholds between methylation states are objectively inferred from the clustering solution.
</p>
<p>The K.R model is used to analyse <code class="reqn">R</code> independent sample types collected from <code class="reqn">N</code> patients, where each sample contains <code class="reqn">C</code> CpG sites, and cluster
the dataset into <code class="reqn">K=M^R</code> clusters to identify the differentially methylated CpG (DMC) sites between the <code class="reqn">R</code> DNA sample types.
</p>


<h3>Value</h3>

<p>The function returns an object of the <code><a href="#topic+betaclust">betaclust</a></code> class which contains the following values:
</p>

<ul>
<li><p> information_criterion - The information criterion used to select the optimal model.
</p>
</li>
<li><p> ic_output - The information criterion value calculated for each model.
</p>
</li>
<li><p> optimal_model - The model selected as optimal.
</p>
</li>
<li><p> function_call - The parameters passed as arguments to the function <code><a href="#topic+betaclust">betaclust</a></code>.
</p>
</li>
<li><p> K - The number of clusters identified using the beta mixture models.
</p>
</li>
<li><p> C - The number of CpG sites analysed using the beta mixture models.
</p>
</li>
<li><p> N - The number of patients analysed using the beta mixture models.
</p>
</li>
<li><p> R - The number of sample types analysed using the beta mixture models.
</p>
</li>
<li><p> optimal_model_results - Information from the optimal model. Specifically,
</p>

<ul>
<li><p> cluster_size - The total number of CpG sites in each of the K clusters.
</p>
</li>
<li><p> llk - A vector containing the log-likelihood value at each step of the EM algorithm.
</p>
</li>
<li><p> alpha - This contains the first shape parameter for the beta mixture model.
</p>
</li>
<li><p> delta - This contains the second shape parameter for the beta mixture model.
</p>
</li>
<li><p> tau - The proportion of CpG sites in each cluster.
</p>
</li>
<li><p> z - A matrix of dimension <code class="reqn">C \times K</code> containing the posterior probability of each CpG site belonging to each of the <code class="reqn">K</code> clusters.
</p>
</li>
<li><p> classification - The classification corresponding to z, i.e. map(z).
</p>
</li>
<li><p> uncertainty - The uncertainty of each CpG site's clustering.
</p>
</li>
<li><p> thresholds - Threshold points calculated under the K.. or the KN. model.
</p>
</li>
<li><p> DM - The AUC and WD metric for distribution similarity in each cluster.
</p>
</li></ul>

</li></ul>



<h3>References</h3>

<p>Silva, R., Moran, B., Russell, N.M., Fahey, C., Vlajnic, T., Manecksha, R.P., Finn, S.P., Brennan, D.J., Gallagher, W.M., Perry, A.S.: Evaluating liquid biopsies for methylomic profiling of prostate cancer. Epigenetics 15(6-7), 715-727 (2020). doi: <a href="https://doi.org/10.1080/15592294.2020.1712876">10.1080/15592294.2020.1712876</a>.
</p>
<p>Majumdar, K., Silva, R., Perry, A.S., Watson, R.W., Murphy, T.B., Gormley, I.C.: betaclust: a family of mixture models for beta valued DNA methylation data. arXiv [stat.ME] (2022). doi: <a href="https://doi.org/10.48550/ARXIV.2211.01938">10.48550/ARXIV.2211.01938</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+beta_k">beta_k</a></code>
</p>
<p><code><a href="#topic+beta_kn">beta_kn</a></code>
</p>
<p><code><a href="#topic+beta_kr">beta_kr</a></code>
</p>
<p><code><a href="#topic+pca.methylation.data">pca.methylation.data</a></code>
</p>
<p><code><a href="#topic+plot.betaclust">plot.betaclust</a></code>
</p>
<p><code><a href="#topic+summary.betaclust">summary.betaclust</a></code>
</p>
<p><code><a href="#topic+threshold">threshold</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
my.seed &lt;- 190
M &lt;- 3
N &lt;- 4
R &lt;- 2
data_output &lt;- betaclust(pca.methylation.data[1:30,2:9], M, N, R,
            model_names = c("K..","KN.","K.R"), model_selection = "BIC",
            parallel_process = FALSE, seed = my.seed)


</code></pre>

<hr>
<h2 id='DMC_identification'>The DMC identification function</h2><span id='topic+DMC_identification'></span>

<h3>Description</h3>

<p>A function to identify the most differentially
methylated clusters from <code class="reqn">K</code> clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DMC_identification(
  object,
  data,
  CpG_site_list,
  threshold = 0.65,
  metric = "AUC"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DMC_identification_+3A_object">object</code></td>
<td>
<p>A betaclust object</p>
</td></tr>
<tr><td><code id="DMC_identification_+3A_data">data</code></td>
<td>
<p>A dataframe of dimension <code class="reqn">C \times NR</code> containing methylation
values for <code class="reqn">C</code> CpG sites from <code class="reqn">R</code> samples collected from <code class="reqn">N</code>
patients which was passed as an argument to the
<code><a href="#topic+betaclust">betaclust</a></code> function.</p>
</td></tr>
<tr><td><code id="DMC_identification_+3A_cpg_site_list">CpG_site_list</code></td>
<td>
<p>The IlmnID of all the CpG sites analysed by
<code><a href="#topic+betaclust">betaclust</a></code> function.</p>
</td></tr>
<tr><td><code id="DMC_identification_+3A_threshold">threshold</code></td>
<td>
<p>The threshold value/s for selecting the most differentially
methylated clusters, default= 0.65</p>
</td></tr>
<tr><td><code id="DMC_identification_+3A_metric">metric</code></td>
<td>
<p>The metric (AUC or WD selected). default=&quot;AUC&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function selects the most diffentially methylated clusters
based on AUC and WD metric calculated and returns the CpG sites
belonging to those clusters.
</p>


<h3>Value</h3>

<p>The function returns a dataframe of CpG sites and
methylation values identified to belong to the most
differentially methylated clusters
</p>


<h3>See Also</h3>

<p><code><a href="#topic+beta_kr">beta_kr</a></code>
</p>
<p><code><a href="#topic+pca.methylation.data">pca.methylation.data</a></code>
</p>
<p><code><a href="#topic+plot.betaclust">plot.betaclust</a></code>
</p>
<p><code><a href="#topic+summary.betaclust">summary.betaclust</a></code>
</p>
<p><code><a href="#topic+betaclust">betaclust</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
my.seed &lt;- 190
M &lt;- 3
N &lt;- 4
R &lt;- 2
data_output &lt;- betaclust(pca.methylation.data[1:30,2:9], M, N, R,
            model_names = "K.R",
            parallel_process = FALSE, seed = my.seed)
dmc_df &lt;-DMC_identification(data_output,pca.methylation.data[1:30,2:9],
pca.methylation.data[1:30,1],
 threshold = 0.65, metric = "AUC")


</code></pre>

<hr>
<h2 id='ecdf.betaclust'>The empirical cumulative distribution function plot</h2><span id='topic+ecdf.betaclust'></span>

<h3>Description</h3>

<p>An empirical cumulative distribution function (ECDF) plot for a <code><a href="#topic+betaclust">betaclust</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ecdf.betaclust(x, R = 2, sample_name = NULL, title = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ecdf.betaclust_+3A_x">x</code></td>
<td>
<p>A dataframe containing methylation values of identified differentially methylated regions related to a gene.
Samples are grouped together in the dataframe such that the columns are ordered as Sample1_Patient1, Sample1_Patient2, Sample2_Patient1, Sample2_Patient2, etc.</p>
</td></tr>
<tr><td><code id="ecdf.betaclust_+3A_r">R</code></td>
<td>
<p>Number of tissue sample types from which DNA methylation data are collected (default R = 2).</p>
</td></tr>
<tr><td><code id="ecdf.betaclust_+3A_sample_name">sample_name</code></td>
<td>
<p>The order in which the sample types are grouped in the dataframe. If no value is specified then default values of sample names, e.g. Sample 1, Sample 2, etc are used (default = NULL).</p>
</td></tr>
<tr><td><code id="ecdf.betaclust_+3A_title">title</code></td>
<td>
<p>The title that the user wants to display on the graph. The default is &quot;NULL&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function plots the ECDF of the differentially methylated CpG sites identified using the K.R model for all patient samples.
The plot visualises the methylation state changes between the different DNA samples for each patient.
</p>


<h3>Value</h3>

<p>The ECDF plot for the selected CpG sites for all patients and their DNA sample types.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+betaclust">betaclust</a></code>
</p>
<p><code><a href="#topic+beta_kr">beta_kr</a></code>
</p>

<hr>
<h2 id='em_aic'>Akaike Information Criterion</h2><span id='topic+em_aic'></span>

<h3>Description</h3>

<p>Compute the AIC value for the optimal model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>em_aic(llk, C, M, N, R, model_name = "K..")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="em_aic_+3A_llk">llk</code></td>
<td>
<p>Log-likelihood value.</p>
</td></tr>
<tr><td><code id="em_aic_+3A_c">C</code></td>
<td>
<p>Number of CpG sites.</p>
</td></tr>
<tr><td><code id="em_aic_+3A_m">M</code></td>
<td>
<p>Number of methylation states identified in a DNA sample.</p>
</td></tr>
<tr><td><code id="em_aic_+3A_n">N</code></td>
<td>
<p>Number of patients.</p>
</td></tr>
<tr><td><code id="em_aic_+3A_r">R</code></td>
<td>
<p>Number of DNA sample types collected from each patient.</p>
</td></tr>
<tr><td><code id="em_aic_+3A_model_name">model_name</code></td>
<td>
<p>Fitted mixture model. Options are &quot;K..&quot;, &quot;KN.&quot; and/or &quot;K.R&quot; (default = &quot;K..&quot;).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Computes the AIC for a specified model given the log-likelihood, the dimension of the data, and the model names.
</p>


<h3>Value</h3>

<p>The AIC value for the selected model.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+em_bic">em_bic</a></code>
</p>
<p><code><a href="#topic+em_icl">em_icl</a></code>
</p>

<hr>
<h2 id='em_bic'>Bayesian Information Criterion</h2><span id='topic+em_bic'></span>

<h3>Description</h3>

<p>Compute the BIC value for the optimal model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>em_bic(llk, C, M, N, R, model_name = "K..")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="em_bic_+3A_llk">llk</code></td>
<td>
<p>Log-likelihood value.</p>
</td></tr>
<tr><td><code id="em_bic_+3A_c">C</code></td>
<td>
<p>Number of CpG sites.</p>
</td></tr>
<tr><td><code id="em_bic_+3A_m">M</code></td>
<td>
<p>Number of methylation states identified in a DNA sample.</p>
</td></tr>
<tr><td><code id="em_bic_+3A_n">N</code></td>
<td>
<p>Number of patients.</p>
</td></tr>
<tr><td><code id="em_bic_+3A_r">R</code></td>
<td>
<p>Number of DNA sample types collected from each patient.</p>
</td></tr>
<tr><td><code id="em_bic_+3A_model_name">model_name</code></td>
<td>
<p>Fitted mixture model. Options are &quot;K..&quot;, &quot;KN.&quot; and/or &quot;K.R&quot; (default = &quot;K..&quot;).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Computes the BIC for a specified model given the log-likelihood, the dimension of the data, and the model names.
</p>


<h3>Value</h3>

<p>The BIC value for the selected model.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+em_aic">em_aic</a></code>
</p>
<p><code><a href="#topic+em_icl">em_icl</a></code>
</p>

<hr>
<h2 id='em_icl'>Integrated Complete-data Likelihood (ICL) Criterion</h2><span id='topic+em_icl'></span>

<h3>Description</h3>

<p>Compute the ICL value for the optimal model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>em_icl(llk, C, M, N, R, model_name = "K..", z)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="em_icl_+3A_llk">llk</code></td>
<td>
<p>Log-likelihood value.</p>
</td></tr>
<tr><td><code id="em_icl_+3A_c">C</code></td>
<td>
<p>Number of CpG sites.</p>
</td></tr>
<tr><td><code id="em_icl_+3A_m">M</code></td>
<td>
<p>Number of methylation states identified in a DNA sample.</p>
</td></tr>
<tr><td><code id="em_icl_+3A_n">N</code></td>
<td>
<p>Number of patients.</p>
</td></tr>
<tr><td><code id="em_icl_+3A_r">R</code></td>
<td>
<p>Number of DNA sample types collected from each patient.</p>
</td></tr>
<tr><td><code id="em_icl_+3A_model_name">model_name</code></td>
<td>
<p>Fitted mixture model. Options are &quot;K..&quot;, &quot;KN.&quot; and/or &quot;K.R&quot; (default = &quot;K..&quot;).</p>
</td></tr>
<tr><td><code id="em_icl_+3A_z">z</code></td>
<td>
<p>A matrix of posterior probabilities of cluster membership, stored as z in the object from <code><a href="#topic+beta_k">beta_k</a></code>, <code><a href="#topic+beta_kn">beta_kn</a></code> and <code><a href="#topic+beta_kr">beta_kr</a></code> functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Computes the ICL for a specified model given the log-likelihood, the dimension of the data, and the model names. This criterion penalises the BIC by including an entropy term favouring well separated clusters.
</p>


<h3>Value</h3>

<p>The ICL value for the selected model.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+em_aic">em_aic</a></code>
</p>
<p><code><a href="#topic+em_bic">em_bic</a></code>
</p>

<hr>
<h2 id='legacy.data'>MethylationEPIC manifest data.</h2><span id='topic+legacy.data'></span>

<h3>Description</h3>

<p>A dataset containing a subset of the manifest data from the Illumina MethylationEPIC beadchip array. A subset of the complete dataset has been uploaded in the package for testing purpose. The complete dataset is available on <a href="https://github.com/koyelucd/betaclust">GitHub</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(legacy.data)
</code></pre>


<h3>Format</h3>

<p>A data frame with 5,080 rows and 6 columns.
</p>

<ul>
<li><p>IlmnID: The unique identifier from the Illumina CG database, i.e. the probe ID.
</p>
</li>
<li><p>Genome_Build: The genome build referenced by the Infinium MethylationEPIC manifest.
</p>
</li>
<li><p>CHR: The chromosome containing the CpG (Genome_Build = 37).
</p>
</li>
<li><p>MAPINFO: The chromosomal coordinates of the CpG sites.
</p>
</li>
<li><p>UCSC_RefGene_Name: The target gene name(s), from the UCSC database.
Note: multiple listings of the same gene name indicate splice variants.
</p>
</li>
<li><p>UCSC_CpG_Islands_Name: The chromosomal coordinates of the CpG Island from UCSC.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+pca.methylation.data">pca.methylation.data</a></code>
</p>

<hr>
<h2 id='pca.methylation.data'>DNA methylation data from patients with prostate cancer</h2><span id='topic+pca.methylation.data'></span>

<h3>Description</h3>

<p>A dataset containing pre-processed beta methylation values from <code class="reqn">R=2</code> sample types, collected from <code class="reqn">N=4</code> patients with prostate cancer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pca.methylation.data)
</code></pre>


<h3>Format</h3>

<p>A data frame with 5,067 rows and 9 columns. The data contain no missing values.
</p>

<ul>
<li><p>IlmnID: The unique identifier from the Illumina CG database, i.e. the probe ID.
</p>
</li>
<li><p>Benign_Patient_1: Methylation values from benign prostate tissue from patient 1.
</p>
</li>
<li><p>Benign_Patient_2: Methylation values from benign prostate tissue from patient 2.
</p>
</li>
<li><p>Benign_Patient_3: Methylation values from benign prostate tissue from patient 3.
</p>
</li>
<li><p>Benign_Patient_4: Methylation values from benign prostate tissue from patient 4.
</p>
</li>
<li><p>Tumour_Patient_1: Methylation values from tumor prostate tissue from patient 1.
</p>
</li>
<li><p>Tumour_Patient_2: Methylation values from tumor prostate tissue from patient 2.
</p>
</li>
<li><p>Tumour_Patient_3: Methylation values from tumor prostate tissue from patient 3.
</p>
</li>
<li><p>Tumour_Patient_4: Methylation values from tumor prostate tissue from patient 4.
</p>
</li></ul>



<h3>Details</h3>

<p>The raw methylation array data was first quality controlled and preprocessed using the <a href="https://rnbeads.org/">RnBeads</a> package. The array data were then normalized and and probes located outside of CpG sites and on the sex chromosome were filtered out. The CpG sites with missing values were removed from the resulting dataset. A subset of the complete dataset has been uploaded in the package for testing purposes. The complete dataset is available on <a href="https://github.com/koyelucd/betaclust">GitHub</a>.
</p>


<h3>References</h3>

<p>Mueller F, Scherer M, Assenov Y, Lutsik P, Walter J, Lengauer T, Bock C (2019). “RnBeads 2.0: comprehensive analysis of DNA methylation data.” Genome Biology, 20(55).
</p>
<p>Assenov Y, Mueller F, Lutsik P, Walter J, Lengauer T, Bock C (2014). “Comprehensive Analysis of DNA Methylation Data with RnBeads.” Nature Methods, 11(11), 1138–1140.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+legacy.data">legacy.data</a></code>
</p>

<hr>
<h2 id='plot.betaclust'>Plots for visualizing the betaclust class object</h2><span id='topic+plot.betaclust'></span>

<h3>Description</h3>

<p>Visualise a <code><a href="#topic+betaclust">betaclust</a></code> clustering solution by plotting the fitted and kernel density estimates, the uncertainty and the information criterion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'betaclust'
plot(
  x,
  what = "fitted density",
  plot_type = "ggplot",
  data = NULL,
  sample_name = NULL,
  title = NULL,
  patient_number = 1,
  threshold = FALSE,
  scale_param = "free_y",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.betaclust_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+betaclust">betaclust</a></code> object.</p>
</td></tr>
<tr><td><code id="plot.betaclust_+3A_what">what</code></td>
<td>
<p>The different plots that can be obtained are either &quot;fitted density&quot;,&quot;kernel density&quot;,&quot;uncertainty&quot; or &quot;information criterion&quot; (default = &quot;fitted density&quot;).</p>
</td></tr>
<tr><td><code id="plot.betaclust_+3A_plot_type">plot_type</code></td>
<td>
<p>The plot type to be displayed are either &quot;ggplot&quot; or &quot;plotly&quot; (default = &quot;ggplot&quot;).</p>
</td></tr>
<tr><td><code id="plot.betaclust_+3A_data">data</code></td>
<td>
<p>A dataframe of dimension <code class="reqn">C \times NR</code> containing methylation values for <code class="reqn">C</code> CpG sites from <code class="reqn">R</code> samples collected from <code class="reqn">N</code> patients which was passed as an argument to the <code><a href="#topic+betaclust">betaclust</a></code> function.
The data is not required as an input when generating &quot;uncertainty&quot; or &quot;information criterion&quot; plots and the default has been set as &quot;NULL&quot;. The data needs to be passed as an argument to this function when generating either &quot;fitted density&quot; or &quot;kernel density&quot; plots.</p>
</td></tr>
<tr><td><code id="plot.betaclust_+3A_sample_name">sample_name</code></td>
<td>
<p>The names of DNA sample types in the dataset analysed by the K.R model. If no value is passed then default values of sample names, e.g. Sample 1, Sample 2, etc are used as legend text (default = NULL).</p>
</td></tr>
<tr><td><code id="plot.betaclust_+3A_title">title</code></td>
<td>
<p>The title that the user wants to display. If no title is to be displayed the default is &quot;NULL&quot;.</p>
</td></tr>
<tr><td><code id="plot.betaclust_+3A_patient_number">patient_number</code></td>
<td>
<p>The column number representing the patient in the patient-wise ordered dataset selected for visualizing the clustering solution of the K.. or KN. model (default = 1).</p>
</td></tr>
<tr><td><code id="plot.betaclust_+3A_threshold">threshold</code></td>
<td>
<p>The &quot;TRUE&quot; option displays the threshold points in the graph for the K.. and the KN. model (default = &quot;FALSE&quot;).</p>
</td></tr>
<tr><td><code id="plot.betaclust_+3A_scale_param">scale_param</code></td>
<td>
<p>The position scales can be fixed or allowed to vary between different panels generated for the density estimate plots for visualizing the K.R clustering solution. Options are &quot;fixed&quot;, &quot;free_y&quot;,&quot;free_x&quot; or &quot;free&quot; (default = &quot;free_y&quot;). The option &quot;fixed&quot; results in the x and y scales being fixed across all panels, &quot;free&quot; varies the x and y scales across the panels, &quot;free_x&quot; fixes the y scale and lets the x scale vary across all panels and &quot;free_y&quot; fixes the x scale and lets the y scale vary across all panels.</p>
</td></tr>
<tr><td><code id="plot.betaclust_+3A_...">...</code></td>
<td>
<p>Other graphics parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The fitted density estimates can be visualized under the optimal clustering solution by specifying what = &quot;fitted density&quot; and kernel density estimates under the optimal clustering solution by specifying what = &quot;kernel density&quot;.
</p>
<p>The threshold inferred can be visualized by specifying threshold = TRUE.
The KN. model calculates different pairs of threshold points for each patient as the shape parameters are allowed to vary for each patient.
So the patient for whom the threshold needs to be displayed can be specified by inputting the column number representing the patient in the patient-wise ordered dataset in the parameter patient_number.
</p>
<p>Interactive plots can also be produced using plot_type = &quot;plotly&quot;. The uncertainty in the clustering solution can be plotted using what = &quot;uncertainty&quot;.
The information criterion values for all fitted models can be plotted using what = &quot;information criterion&quot;.
</p>


<h3>Value</h3>

<p>This function displays the following plots as requested by the user:
</p>

<ul>
<li><p> fitted density estimates - Plot showing the fitted density estimates of the clustering solution under the optimal model selected.
</p>
</li>
<li><p> kernel density estimates - Plot showing the kernel density estimates of the clustering solution under the optimal model selected.
</p>
</li>
<li><p> uncertainty -  A boxplot showing the uncertainties in the optimal clustering solution.
</p>
</li>
<li><p> information criterion - Plot showing the information criterion values for all models fitted to support the selection of the optimal model.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+betaclust">betaclust</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
my.seed &lt;- 190
M &lt;- 3
N &lt;- 4
R &lt;- 2
data_output &lt;- betaclust(pca.methylation.data[1:100,2:9], M, N, R,
            model_names = c("K..","KN.","K.R"), model_selection = "BIC",
            parallel_process = FALSE, seed = my.seed)
plot(data_output, what = "fitted density", plot_type = "ggplot",
     data=pca.methylation.data[1:100,2:9])
plot(data_output, what = "uncertainty", plot_type = "ggplot")
plot(data_output, what = "information criterion", plot_type = "ggplot")

</code></pre>

<hr>
<h2 id='summary.betaclust'>Summarizing the beta mixture model fits</h2><span id='topic+summary.betaclust'></span>

<h3>Description</h3>

<p>Summary method for a <code><a href="#topic+betaclust">betaclust</a></code> object containing the results under the optimal model selected.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'betaclust'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.betaclust_+3A_object">object</code></td>
<td>
<p>A <code><a href="#topic+betaclust">betaclust</a></code> object.</p>
</td></tr>
<tr><td><code id="summary.betaclust_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code><a href="#topic+summary.betaclust">summary.betaclust</a></code> which contains the following list:
</p>

<ul>
<li><p> C - The number of CpG sites analysed using the beta mixture models.
</p>
</li>
<li><p> N - The number of patients analysed using the beta mixture models.
</p>
</li>
<li><p> R - The number of sample types analysed using the beta mixture models.
</p>
</li>
<li><p> K - The number of methylation states in R DNA samples.
</p>
</li>
<li><p> modelName - The optimal model selected.
</p>
</li>
<li><p> loglik - The log-likelihood value for the selected optimal model.
</p>
</li>
<li><p> information_criterion - The information criterion used to select the optimal model.
</p>
</li>
<li><p> ic_output - This stores the information criterion value calculated for each model.
</p>
</li>
<li><p> classification - The total number of CpG sites in each cluster.
</p>
</li>
<li><p> prop_data - The estimated mixing proportion for each cluster.</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+betaclust">betaclust</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
my.seed &lt;- 190
M &lt;- 3
N &lt;- 4
R &lt;- 2
data_output &lt;- betaclust(pca.methylation.data[1:30,2:9], M, N, R,
            model_names=c("K..","KN.","K.R"), model_selection="BIC",
            parallel_process = FALSE, seed=my.seed)
summary(data_output)
</code></pre>

<hr>
<h2 id='threshold'>Thresholds under the K.. and the KN. models</h2><span id='topic+threshold'></span>

<h3>Description</h3>

<p>An objective method to calculate the threshold points for the clustering solution of the K.. and the KN. models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>threshold(object, data, model_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="threshold_+3A_object">object</code></td>
<td>
<p>A <code><a href="#topic+beta_k">beta_k</a></code> or <code><a href="#topic+beta_kn">beta_kn</a></code> object.</p>
</td></tr>
<tr><td><code id="threshold_+3A_data">data</code></td>
<td>
<p>A dataframe of dimension <code class="reqn">C \times NR</code> containing methylation values for <code class="reqn">C</code> CpG sites from <code class="reqn">R</code> sample types collected from <code class="reqn">N</code> patients which was passed as an argument to the <code><a href="#topic+betaclust">betaclust</a></code> function.</p>
</td></tr>
<tr><td><code id="threshold_+3A_model_name">model_name</code></td>
<td>
<p>The name of the model for which the thresholds need to be calculated.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As the K.. model constrains the shape parameters to be equal for all patients, a single pair of threshold points are calculated for all patients. The KN. model allows patient-specific shape parameters which results in a pair of threshold points for each patient based on the shape parameters for that patient.
The first threshold point means that any CpG site with beta value less than this value is likely to be hypomethylated.
The second threshold point means that any CpG site with beta value greater than this is highly likely to be hypermethylated.
A CpG site with beta value lying between the two threshold points is likely to be hemimethylated.
</p>


<h3>Value</h3>

<p>thresholds - The threshold points calculated for the selected model. A vector containing two threshold points are returned for the K.. model whereas a matrix containing two threshold points for each patient is returned for the KN. model.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+beta_k">beta_k</a></code>
</p>
<p><code><a href="#topic+beta_kn">beta_kn</a></code>
</p>
<p><code><a href="#topic+betaclust">betaclust</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
