<!DOCTYPE html><html lang="en"><head><title>Help for package BGGE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BGGE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BGGE'><p>Genotype x Environment models using regression kernel</p></a></li>
<li><a href='#getK'><p>Kernel matrix for GE genomic selection models</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Bayesian Genomic Linear Models Applied to GE Genome Selection</td>
</tr>
<tr>
<td>Version:</td>
<td>0.6.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-08-10</td>
</tr>
<tr>
<td>Description:</td>
<td>Application of genome prediction for a continuous variable, focused 
             on genotype by environment (GE) genomic selection models (GS). It consists a group of functions 
             that help to create regression kernels for some GE genomic models proposed by Jarquín et al. (2014) &lt;<a href="https://doi.org/10.1007%2Fs00122-013-2243-1">doi:10.1007/s00122-013-2243-1</a>&gt;
             and Lopez-Cruz et al. (2015) &lt;<a href="https://doi.org/10.1534%2Fg3.114.016097">doi:10.1534/g3.114.016097</a>&gt;. Also, it computes genomic predictions based on Bayesian approaches.
             The prediction function uses an orthogonal transformation of the data and specific priors
             present by Cuevas et al. (2014) &lt;<a href="https://doi.org/10.1534%2Fg3.114.013094">doi:10.1534/g3.114.013094</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats,</td>
</tr>
<tr>
<td>Suggests:</td>
<td>BGLR, coda</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-08-10 14:54:06 UTC; Italo Granato</td>
</tr>
<tr>
<td>Author:</td>
<td>Italo Granato [aut, cre],
  Luna-Vázquez Francisco J. [aut],
  Cuevas Jaime [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Italo Granato &lt;italo.granato@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-08-10 16:40:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='BGGE'>Genotype x Environment models using regression kernel</h2><span id='topic+BGGE'></span>

<h3>Description</h3>

<p>BGGE function fits Bayesian regression for continuous observations through regression kernels
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BGGE(y, K, XF = NULL, ne, ite = 1000, burn = 200, thin = 3, verbose = FALSE, 
            tol = 1e-10, R2 = 0.5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BGGE_+3A_y">y</code></td>
<td>
<p>Vector of data. Should be numeric and NAs are allowed.</p>
</td></tr>
<tr><td><code id="BGGE_+3A_k">K</code></td>
<td>
<p>list A two-level list Specify the regression kernels (co-variance matrix). The former is the <code>Kernel</code>,
where is included the regression kernels. The later is the <code>Type</code>, specifying if the matrix is either <code>D</code> Dense or
<code>BD</code> Block Diagonal. A number of regression kernels or random effects to be fitted are specified in this list.</p>
</td></tr>
<tr><td><code id="BGGE_+3A_xf">XF</code></td>
<td>
<p>matrix Design matrix (<code class="reqn">n \times p</code>) for fixed effects</p>
</td></tr>
<tr><td><code id="BGGE_+3A_ne">ne</code></td>
<td>
<p>vector Number of genotypes by environment.</p>
</td></tr>
<tr><td><code id="BGGE_+3A_ite">ite</code></td>
<td>
<p>numeric Number of iterations.</p>
</td></tr>
<tr><td><code id="BGGE_+3A_burn">burn</code></td>
<td>
<p>numeric Number of iterations to be discarded as burn-in.</p>
</td></tr>
<tr><td><code id="BGGE_+3A_thin">thin</code></td>
<td>
<p>numeric Thinin interval.</p>
</td></tr>
<tr><td><code id="BGGE_+3A_verbose">verbose</code></td>
<td>
<p>Should iteration history be printed on console? If TRUE or 1 then it is printed,
otherwise, if another number $n$ is choosen the history is printed every $n$ times. The default is FALSE.</p>
</td></tr>
<tr><td><code id="BGGE_+3A_tol">tol</code></td>
<td>
<p>a numeric tolerance level. Eigenvalues lower than <code>tol</code> are discarded. Default is 1e-10.</p>
</td></tr>
<tr><td><code id="BGGE_+3A_r2">R2</code></td>
<td>
<p>the proportion of variance expected to be explained by the regression.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The goal is to fit genomic prediction models for continuous outcomes through Gibbs sampler. BGGE uses a proposal for dimension reduction
through an orthogonal transformation of observed data (y) as well as differential shrinkage because of the prior variance assigned 
to regression parameters. Further details on this approach can be found in Cuevas et al. (2014).
The primaty genetic model is
</p>
<p style="text-align: center;"><code class="reqn">y = g + e</code>
</p>

<p>where <code class="reqn">y</code> is the response, <code class="reqn">g</code> is the unknown random effect and <code class="reqn">e</code> is the residual effect.
You can specify a number of random effects <code class="reqn">g</code>, as many as desired, through a list of regression kernels related to each random effect in the
argument <code>K</code>.
The structure of <code>K</code> is a two level list, where the first element on the second level is the Kernel and the second element is a definition of
type of matrix. There are two definitions, either matrix is <code>D</code> (dense) or <code>BD</code> (Block Diagonal). As we make the spectral decomposition 
on the kernels, for block diagonal matrices, we take advantage of its structure and make decomposition on the submatrices instead of one big matrix.
For example, the regression kernels should be an structure like K = list(list(Kernel = G, Type = &quot;D&quot;), list(Kernel = G, Type = &quot;BD&quot;)). 
The definition of one matrix as a block diagonal must be followed by the number of subjects in each submatrix in the block diagonal,
present in the <code>ne</code>, which allows sub matrices to be drawn. Some genotype by environment models has the block diagonal matrix type or similar.
The genotype x environment deviation matrix in MDs model (Sousa et al., 2017) has the structure of block diagonal. 
Also, the matrices for environment-specific variance in MDe models (Sousa et al., 2017) if summed, can form a structure of block diagonal, 
where is possible to extract sub matrices for each environment. In the case of all kernel be of the dense type, <code>ne</code> is ignored.
</p>


<h3>Value</h3>

<p>A list with estimated posterior means of residual and genetic variance component for each term in the linear model and the genetic value predicted. Also the 
values along with the chains are released.
</p>


<h3>References</h3>

<p>Cuevas, J., Perez-Elizalde, S., Soberanis, V., Perez-Rodriguez, P., Gianola, D., &amp; Crossa, J. (2014).
Bayesian genomic-enabled prediction as an inverse problem. G3: Genes, Genomes, Genetics, 4(10), 1991-2001.
</p>
<p>Sousa, M. B., Cuevas, J., Oliveira, E. G. C., Perez-Rodriguez, P., Jarquin, D., Fritsche-Neto, R., Burgueno, J. 
&amp; Crossa, J. (2017). Genomic-enabled prediction in maize using kernel models with genotype x environment interaction.
G3: Genes, Genomes, Genetics, 7(6), 1995-2014.
</p>


<h3>See Also</h3>

<p><code><a href="BGLR.html#topic+BGLR">BGLR</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># multi-environment main genotypic model
library(BGLR)
data(wheat)
X&lt;-wheat.X[1:200,1:600]  # Subset of 200 subjects and 600 markers
rownames(X) &lt;- 1:200
Y&lt;-wheat.Y[1:200,]
A&lt;-wheat.A[1:200,1:200] # Pedigree

GB&lt;-tcrossprod(X)/ncol(X)
K&lt;-list(G = list(Kernel = GB, Type = "D"))
y&lt;-Y[,1]
fit&lt;-BGGE(y = y,K = K, ne = length(y), ite = 300, burn = 100, thin = 2)

# multi-environment main genotypic model
Env &lt;- as.factor(c(2,3)) #subset of 2 environments
pheno_geno &lt;- data.frame(env = gl(n = 2, k = nrow(Y), labels = Env),
                         GID = gl(n = nrow(Y), k = 1,length = nrow(Y) * length(Env)),
                         value = as.vector(Y[,2:3]))

K &lt;- getK(Y = pheno_geno, X = X, kernel = "GB", model = "MM")
y &lt;- pheno_geno[,3]
fit &lt;- BGGE(y = y, K = K, ne = rep(nrow(Y), length(Env)), ite = 300, burn = 100,thin = 1)


</code></pre>

<hr>
<h2 id='getK'>Kernel matrix for GE genomic selection models</h2><span id='topic+getK'></span>

<h3>Description</h3>

<p>Create kernel matrix for GE genomic prediction models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getK(Y, X, kernel = c("GK", "GB"), setKernel = NULL, bandwidth = 1,
             model = c("SM", "MM", "MDs", "MDe"), quantil = 0.5,
             intercept.random = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getK_+3A_y">Y</code></td>
<td>
<p><code>data.frame</code> Phenotypic data with three columns. The first column is a factor for environments,
the second column is a factor identifying genotypes, and the third column contains the trait of interest</p>
</td></tr>
<tr><td><code id="getK_+3A_x">X</code></td>
<td>
<p>Marker matrix with individuals in rows and markers in columns. Missing markers are not allowed.</p>
</td></tr>
<tr><td><code id="getK_+3A_kernel">kernel</code></td>
<td>
<p>Kernel to be created internally. Methods currently implemented are the Gaussian <code>GK</code> and the linear <code>GBLUP</code> kernel</p>
</td></tr>
<tr><td><code id="getK_+3A_setkernel">setKernel</code></td>
<td>
<p><code>matrix</code> Single kernel matrix in case it is necessary to use a different kernel from <code>GK</code> or <code>GBLUP</code></p>
</td></tr>
<tr><td><code id="getK_+3A_bandwidth">bandwidth</code></td>
<td>
<p><code>vector</code> Bandwidth parameter to create the Gaussian Kernel (GK) matrix. The default for the <code>bandwidth</code> is 1.
Estimation of this parameter can be made using a Bayesian approach as presented in Perez-Elizalde et al. (2015)</p>
</td></tr>
<tr><td><code id="getK_+3A_model">model</code></td>
<td>
<p>Specifies the genotype <code class="reqn">\times</code> environment model to be fitted. It currently supported the 
models  <code>SM</code>, <code>MM</code>, <code>MDs</code> and <code>MDe</code>. See Details</p>
</td></tr>
<tr><td><code id="getK_+3A_quantil">quantil</code></td>
<td>
<p>Specifies the quantile to create the Gaussian kernel.</p>
</td></tr>
<tr><td><code id="getK_+3A_intercept.random">intercept.random</code></td>
<td>
<p>if <code>TRUE</code>, kernel related to random intercept of genotype is included.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The aim is to create kernels to fit GE interaction models applied to genomic prediction.
Two standard genomic kernels are currently supported:
<code>GB</code> creates a linear kernel resulted from the cross-product of centered and standardized 
marker genotypes divide by the number of markers <code class="reqn">p</code>:
</p>
<p style="text-align: center;"><code class="reqn">GB = \frac{XX^T}{p}</code>
</p>

<p>Another alternative is the Gaussian Kernel <code>GK</code>, resulted from:
</p>
<p style="text-align: center;"><code class="reqn"> GK (x_i, x_{i'}) = exp(\frac{-h d_{ii'}^2}{q(d)})</code>
</p>

<p>where <code class="reqn">d_{ii'}^2</code> is the genetic distance between individuals based on markers scaled 
by some percentile <code class="reqn">{q(d)}</code> and <code class="reqn">bandwidth</code> is the bandwidth parameter. However, 
other kernels can be provided through <code>setKernel</code>. In this case, arguments <code>X</code>, 
<code>kernel</code> and <code>h</code> are ignored.
</p>
<p>Currently, the supported models for GE kernels are:
</p>

<ul>
<li> <p><code>SM</code>: is the single-environment main genotypic effect model - It fits the data for a 
single environment, and only one kernel is produced. 
</p>
</li>
<li> <p><code>MM</code>: is the multi-environment main genotypic effect model - It consideres the main
random genetic effects across environments. Thus, just one kernel is produced, of order 
<code class="reqn">n \times n</code>, related to the main effect across environments. 
</p>
</li>
<li> <p><code>MDs</code>: is the multi-environment single variance genotype x environment deviation 
model - It is an extension of <code>MM</code> by adding the random interaction effect of 
environments with genotype information. Thus, two kernels are created, one related to the 
main effect across environment, and the second is associated with single genotype by environment effect.
</p>
</li>
<li> <p><code>MDe</code>: is the multi-environment, environment-specific variance genotype x environment 
deviation model - It separates the genetic effects into the main genetic 
effects and the specific genetic effects (for each environment). Thus, one kernel 
for across environments effect and <code class="reqn">j</code> kernels are created, one for each 
environment.
</p>
</li></ul>

<p>These GE genomic models were compared and named by Sousa et al. (2017) and can be increased by using 
the kernel related to random intercept of genotype through <code>intercept.random</code>.
</p>


<h3>Value</h3>

<p>This function returns a two-level list, which specifies the kernel and the type of matrix. 
The latter is a classification according to its structure, i. e.,
if the matrix is dense or a block diagonal. For the main effect (<code>G</code>), 
the matrix is classified as dense (D). On the other hand, matrices for environment-specific and 
genotype by environment effect (<code>GE</code>) are considered diagonal block (BD). This classification is used 
as part of the prediction through the BGGE function.
</p>


<h3>References</h3>

<p>Jarquin, D., J. Crossa, X. Lacaze, P. Du Cheyron, J. Daucourt, J. Lorgeou, F. Piraux, L. Guerreiro, P. Pérez, M. Calus, J. Burgueño,
and G. de los Campos. 2014. A reaction norm model for genomic selection using high-dimensional genomic and 
environmental data. Theor. Appl. Genet. 127(3): 595-607.
</p>
<p>Lopez-Cruz, M., J. Crossa, D. Bonnett, S. Dreisigacker, J. Poland, J.-L. Jannink, R.P. Singh, E. Autrique,
and G. de los Campos. 2015. Increased prediction accuracy in wheat breeding trials using a marker × environment
interaction genomic selection model. G3: Genes, Genomes, Genetics. 5(4): 569-82.
</p>
<p>Perez- Elizalde, S. J. Cuevas, P. Perez-Rodriguez, and J. Crossa. 2015. Selection of the
Bandwidth Parameter in a Bayesian Kernel Regression Model for Genomic-Enabled Prediction. 
Journal of Agricultural, Biological, and Environmental Statistics (JABES), 20(4):512-532.
</p>
<p>Sousa, M. B., Cuevas, J., Oliveira, E. G. C., Perez-Rodriguez, P., Jarquin, D., Fritsche-Neto, R., Burgueno, J. 
&amp; Crossa, J. (2017). Genomic-enabled prediction in maize using kernel models with genotype x environment interaction.
G3: Genes, Genomes, Genetics, 7(6), 1995-2014.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># create kernel matrix for model MDs using wheat dataset
library(BGLR)

data(wheat)
X &lt;- scale(wheat.X, scale = TRUE, center = TRUE)
rownames(X) &lt;- 1:599
pheno_geno &lt;- data.frame(env = gl(n = 4, k = 599), 
               GID = gl(n=599, k=1, length = 599*4),
               value = as.vector(wheat.Y))
               
 K &lt;- getK(Y = pheno_geno, X = X, kernel = "GB", model = "MDs")              



</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
