<!DOCTYPE html><html><head><title>Help for package bgsmtr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {bgsmtr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bgsmtr'><p>Bayesian Group Sparse Multi-Task Regression for Imaging Genetics</p></a></li>
<li><a href='#bgsmtr_example_data'><p>Example Structural Neuroimaging and Genetic Data</p></a></li>
<li><a href='#sp_bgsmtr'><p>A Bayesian Spatial Model for Imaging Genetics</p></a></li>
<li><a href='#sp_bgsmtr_example_data'><p>Example Structural Neuroimaging and Genetic Data for Spatial Model.</p></a></li>
<li><a href='#sp_bgsmtr_path'><p>A Bayesian Spatial Model for Imaging Genetics</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Bayesian Group Sparse Multi-Task Regression</td>
</tr>
<tr>
<td>Version:</td>
<td>0.7</td>
</tr>
<tr>
<td>Description:</td>
<td>Implementation of Bayesian multi-task regression models and was developed within the context of imaging genetics. The package can currently fit two models. The Bayesian group sparse multi-task regression model of Greenlaw et al. (2017)&lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtx215">doi:10.1093/bioinformatics/btx215</a>&gt; can be fit with implementation using Gibbs sampling. An extension of this model developed by Song, Ge et al. to accommodate both spatial correlation as well as correlation across brain hemispheres can also be fit using either mean-field variational Bayes or Gibbs sampling. The model can also be used more generally for multivariate (non-imaging) phenotypes with spatial correlation.</td>
</tr>
<tr>
<td>Author:</td>
<td>Yin Song, Shufei Ge, Liangliang Wang, Jiguo Cao, Keelin Greenlaw, Mary Lesperance, Farouk S. Nathoo</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yin Song &lt;yinsong@uvic.ca&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), Matrix (&ge; 1.2.6), mvtnorm (&ge;
1.0.5),matrixcalc(&ge; 1.0.3), miscTools (&ge; 0.6.22)</td>
</tr>
<tr>
<td>Imports:</td>
<td>coda (&ge; 0.18.1), EDISON (&ge; 1.1.1),statmod (&ge;
1.4.26),methods (&ge; 3.3.3), sparseMVN (&gt; 0.2.0),inline (&ge;
0.3.15),LaplacesDemon (&ge; 16.1.0), glmnet(&ge;
2.0.13),CholWishart (&ge; 0.9.3),mnormt (&ge; 1.5.4),Rcpp(&ge;
0.12.14),</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.0.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-12-13 03:34:48 UTC; guest1</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-12-13 05:20:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='bgsmtr'>Bayesian Group Sparse Multi-Task Regression for Imaging Genetics</h2><span id='topic+bgsmtr'></span>

<h3>Description</h3>

<p>Runs the the Gibbs sampling algorithm to fit a Bayesian group sparse multi-task regression model.
Tuning parameters can be chosen using either the MCMC samples and the WAIC (multiple runs) or using an approximation to
the posterior mode and five-fold cross-validation (single run).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bgsmtr(
  X,
  Y,
  group,
  tuning = "CV.mode",
  lam_1_fixed = NULL,
  lam_2_fixed = NULL,
  iter_num = 10000,
  burn_in = 5001
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bgsmtr_+3A_x">X</code></td>
<td>
<p>A d-by-n matrix; d is the number of SNPs and n is the number of subjects. Each row of X should correspond to a particular SNP
and each column should correspond to a particular subject. Each element of X should give the number of minor alleles for the corresponding
SNP and subject. The function will center each row of X to have mean zero prior to running the Gibbs sampling algorithm.</p>
</td></tr>
<tr><td><code id="bgsmtr_+3A_y">Y</code></td>
<td>
<p>A c-by-n matrix; c is the number of phenotypes (brain imaging measures) and n is the number of subjects. Each row of
Y should correspond to a particular phenotype and each column should correspond to a particular subject. Each element of Y should give
the measured value for the corresponding phentoype and subject. The function will center and scale each row of Y to have mean zero and unit
variance prior to running the Gibbs sampling algorithm.</p>
</td></tr>
<tr><td><code id="bgsmtr_+3A_group">group</code></td>
<td>
<p>A vector of length d; d is the number of SNPs. Each element of this vector is a string representing a gene or group
label associated with each SNP. The SNPs represented by this vector should be ordered according to the rows of X.</p>
</td></tr>
<tr><td><code id="bgsmtr_+3A_tuning">tuning</code></td>
<td>
<p>A string, either 'WAIC' or 'CV.mode'. If 'WAIC', the Gibbs sampler is run with fixed values of the tuning
parameters specified by the arguments <em>lam_1_fixed</em> and  <em>lam_2_fixed</em> and the WAIC is computed based on the sampling output. This
can then be used to choose optimal values for <em>lam_1_fixed</em> and <em>lam_2_fixed</em> based on multiple runs with each run using different
values of <em>lam_1_fixed</em> and <em>lam_2_fixed</em>. This option is best suited for either comparing a small set of tuning parameter values or
for computation on a high performance computing cluster where different nodes can be used to run the function with different
values of <em>lam_1_fixed</em> and <em>lam_2_fixed</em>. Posterior inference is then based on the run that produces the lowest value for the WAIC.
The option 'CV.mode', which is the default, is best suited for computation using just a single processor. In this case the
tuning parameters are chosen based on five-fold cross-validation over a grid of possible values with out-of-sample prediction based on an
approximate posterior mode. The Gibbs sampler is then run using the chosen values of the tuning parameters. When tuning = 'CV.mode' the values
for the arguments <em>lam_1_fixed</em> and <em>lam_2_fixed</em> are not required.</p>
</td></tr>
<tr><td><code id="bgsmtr_+3A_lam_1_fixed">lam_1_fixed</code></td>
<td>
<p>Only required if tuning = 'WAIC'. A positive number giving the value for the gene-specific tuning parameter. Larger values lead to a larger
degree of shrinkage to zero of estimated regression coefficients at the gene level (across all SNPs and phenotypes).</p>
</td></tr>
<tr><td><code id="bgsmtr_+3A_lam_2_fixed">lam_2_fixed</code></td>
<td>
<p>Only required if tuning = 'WAIC'. A positive number giving the value for the SNP-specific tuning parameter. Larger values lead to a larger
degree of shrinkage to zero of estimated regression coefficients at the SNP level (across all phenotypes).</p>
</td></tr>
<tr><td><code id="bgsmtr_+3A_iter_num">iter_num</code></td>
<td>
<p>Positive integer representing the total number of iterations to run the Gibbs sampler. Defaults to 10,000.</p>
</td></tr>
<tr><td><code id="bgsmtr_+3A_burn_in">burn_in</code></td>
<td>
<p>Nonnegative integer representing the number of MCMC samples to discard as burn-in. Defaults to 5001.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the elements
</p>
<table>
<tr><td><code>WAIC</code></td>
<td>
<p>If tuning = 'WAIC' this is the value of the WAIC computed from the MCMC output. If tuning = 'CV.mode' this component is excluded.</p>
</td></tr>
<tr><td><code>Gibbs_setup</code></td>
<td>
<p>A list providing values for the input parameters of the function.</p>
</td></tr>
<tr><td><code>Gibbs_W_summaries</code></td>
<td>
<p>A list with five components, each component being a d-by-c matrix giving some posterior summary of the regression parameter
matrix W, where the ij-th element of W represents the association between the i-th SNP and j-th phenotype.
</p>
<p>-Gibbs_W_summaries$W_post_mean is a d-by-c matrix giving the posterior mean of W.
</p>
<p>-Gibbs_W_summaries$W_post_mode is a d-by-c matrix giving the posterior mode of W.
</p>
<p>-Gibbs_W_summaries$W_post_sd is a d-by-c matrix giving the posterior standard deviation for each element of W.
</p>
<p>-Gibbs_W_summaries$W_2.5_quantile is a d-by-c matrix giving the posterior 2.5 percent quantile for each element of W.
</p>
<p>-Gibbs_W_summaries$W_97.5_quantile is a d-by-c matrix giving the posterior 97.5 percent quantile for each element of W.'</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Farouk S. Nathoo, <a href="mailto:nathoo@uvic.ca">nathoo@uvic.ca</a>
</p>
<p>Keelin Greenlaw,  <a href="mailto:keelingreenlaw@gmail.com">keelingreenlaw@gmail.com</a>
</p>
<p>Mary Lesperance, <a href="mailto:mlespera@uvic.ca">mlespera@uvic.ca</a>
</p>


<h3>References</h3>

<p>Greenlaw, Keelin, Elena Szefer, Jinko Graham, Mary Lesperance, and Farouk S. Nathoo. &quot;A Bayesian Group Sparse Multi-Task Regression Model for Imaging Genetics.&quot; arXiv preprint arXiv:1605.02234 (2016).
</p>
<p>Nathoo, Farouk S., Keelin Greenlaw, and Mary Lesperance. &quot;Regularization Parameter Selection for a Bayesian Multi-Level Group Lasso Regression Model with Application to Imaging Genomics.&quot; arXiv preprint arXiv:1603.08163 (2016).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bgsmtr_example_data)
names(bgsmtr_example_data)


## Not run: 
## test run the sampler for 100 iterations with fixed tunning parameters and compute WAIC
## we recomend at least 5,000 iterations for actual use
fit = bgsmtr(X = bgsmtr_example_data$SNP_data, Y = bgsmtr_example_data$BrainMeasures,
group = bgsmtr_example_data$SNP_groups, tuning = 'WAIC', lam_1_fixed = 2, lam_2_fixed = 2,
iter_num = 100, burn_in = 50)
## posterior mean for regression parameter relating 100th SNP to 14th phenotype
fit$Gibbs_W_summaries$W_post_mean[100,14]
## posterior mode for regression parameter relating 100th SNP to 14th phenotype
fit$Gibbs_W_summaries$W_post_mode[100,14]
## posterior standard deviation for regression parameter relating 100th SNP to 14th phenotype
fit$Gibbs_W_summaries$W_post_sd[100,14]
## 95
c(fit$Gibbs_W_summaries$W_2.5_quantile[100,14],fit$Gibbs_W_summaries$W_97.5_quantile[100,14])

## End(Not run)

## Not run: 
## run the sampler for 10,000 iterations with tuning parameters set using cross-validation
## On a standard computer with a small numer of cores this is the recomended option
fit = bgsmtr(X = bgsmtr_example_data$SNP_data, Y = bgsmtr_example_data$BrainMeasures,
group = bgsmtr_example_data$SNP_groups, tuning = 'CV.mode',iter_num = 10000, burn_in = 5000)

## End(Not run)

</code></pre>

<hr>
<h2 id='bgsmtr_example_data'>Example Structural Neuroimaging and Genetic Data</h2><span id='topic+bgsmtr_example_data'></span>

<h3>Description</h3>

<p>Simulated data with 632 subjects, 486 SNPs from 33 genes, 15 structural neuroimaging measures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(bgsmtr_example_data)
</code></pre>


<h3>Format</h3>

<p>A list with three components: &quot;SNP_data&quot;, &quot;SNP_groups&quot;, &quot;BrainMeasures&quot;.
SNP_data is a 486-by-632 matrix containing minor allele counts for 632 subjects and 486 SNPs.
SNP_groups is a vector of length 486 with labels partitioning the 486 SNPs into 33 genes.
BrainMeasures is a 15-by-632 matrix containing simulated volumetric and cortical thickness measures for 15 regions of interest.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bgsmtr_example_data)
names(bgsmtr_example_data)
dim(bgsmtr_example_data$SNP_data)
dim(bgsmtr_example_data$BrainMeasures)
unique(bgsmtr_example_data$SNP_groups)
</code></pre>

<hr>
<h2 id='sp_bgsmtr'>A Bayesian Spatial Model for Imaging Genetics</h2><span id='topic+sp_bgsmtr'></span>

<h3>Description</h3>

<p>Fits a Bayesian spatial model that allows for two types of correlation typically seen in structural brain imaging data. First, the spatial correlation in the imaging phenotypes obtained from neighbouring regions of the brain. Second, the correlation between corresponding measures on opposite hemispheres.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sp_bgsmtr(
  X,
  Y,
  method = "MCMC",
  rho = NULL,
  lambdasq = NULL,
  alpha = NULL,
  A = NULL,
  c.star = NULL,
  FDR_opt = TRUE,
  WAIC_opt = TRUE,
  iter_num = 10000,
  burn_in = 5001
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sp_bgsmtr_+3A_x">X</code></td>
<td>
<p>A d-by-n matrix; d is the number of SNPs and n is the number of subjects. Each row of X should correspond to a particular SNP
and each column should correspond to a particular subject. Each element of X should give the number of minor alleles for the corresponding
SNP and subject. The function will center each row of X to have mean zero prior to running the Gibbs sampling algorithm.</p>
</td></tr>
<tr><td><code id="sp_bgsmtr_+3A_y">Y</code></td>
<td>
<p>A c-by-n matrix; c is the number of phenotypes (brain imaging measures) and n is the number of subjects. Each row of
Y should correspond to a particular phenotype and each column should correspond to a particular subject. Each element of Y should give
the measured value for the corresponding phentoype and subject. The function will center and scale each row of Y to have mean zero and unit
variance prior to running the Gibbs sampling algorithm.</p>
</td></tr>
<tr><td><code id="sp_bgsmtr_+3A_method">method</code></td>
<td>
<p>A string, either 'MCMC' or 'MFVB'. If 'MCMC', the Gibbs sampling method will be used. If 'MFVB', mean field variational Bayes method will be used.</p>
</td></tr>
<tr><td><code id="sp_bgsmtr_+3A_rho">rho</code></td>
<td>
<p>Spatial cohesion paramter. If no value is assigned, it takes 0.95 by default.</p>
</td></tr>
<tr><td><code id="sp_bgsmtr_+3A_lambdasq">lambdasq</code></td>
<td>
<p>A tuning paratmeter. If no value is assigned, the algorithm will estimate and assign a value for it based on a moment estimate.</p>
</td></tr>
<tr><td><code id="sp_bgsmtr_+3A_alpha">alpha</code></td>
<td>
<p>Bayesian False Discovery Rate (FDR) level. Default level is 0.05.</p>
</td></tr>
<tr><td><code id="sp_bgsmtr_+3A_a">A</code></td>
<td>
<p>A c/2 by c/2 neighborhood structure matrix for different brain regions.</p>
</td></tr>
<tr><td><code id="sp_bgsmtr_+3A_c.star">c.star</code></td>
<td>
<p>The threshold for computing posterior tail probabilities p_ij for Bayesian FDR as defined in Section 3.2 of  Song, Ge et al.(2019). If not specified the default is to set this threshold as the minimum posterior standard deviation, where the minimum is taken over all regression coefficients in the model.</p>
</td></tr>
<tr><td><code id="sp_bgsmtr_+3A_fdr_opt">FDR_opt</code></td>
<td>
<p>A logical operator for computing Bayesian FDR. By default, it's TRUE.</p>
</td></tr>
<tr><td><code id="sp_bgsmtr_+3A_waic_opt">WAIC_opt</code></td>
<td>
<p>A logical operator for computing WAIC from MCMC method. By default, it's TRUE.</p>
</td></tr>
<tr><td><code id="sp_bgsmtr_+3A_iter_num">iter_num</code></td>
<td>
<p>Positive integer representing the total number of iterations to run the Gibbs sampler. Defaults to 10,000.</p>
</td></tr>
<tr><td><code id="sp_bgsmtr_+3A_burn_in">burn_in</code></td>
<td>
<p>Nonnegative integer representing the number of MCMC samples to discard as burn-in. Defaults to 5001.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with following elements
</p>
<table>
<tr><td><code>WAIC</code></td>
<td>
<p>WAIC is computed from the MCMC output if &quot;MCMC&quot; is chosen for method.</p>
</td></tr>
<tr><td><code>lower_boud</code></td>
<td>
<p>Lower bound from MFVB output if &quot;MFVB is choosen for method.</p>
</td></tr>
<tr><td><code>Gibbs_setup</code></td>
<td>
<p>A list providing values for the input parameters of the function.</p>
</td></tr>
<tr><td><code>lambdasq_est</code></td>
<td>
<p>Estimated value for tunning parameter lambda-squared.</p>
</td></tr>
<tr><td><code>Gibbs_W_summaries</code></td>
<td>
<p>A list with five components, each component being a d-by-c matrix giving some posterior summary of the regression parameter
matrix W, where the ij-th element of W represents the association between the i-th SNP and j-th phenotype.
</p>
<p>-Gibbs_W_summaries$W_post_mean is a d-by-c matrix giving the posterior mean of W.
</p>
<p>-Gibbs_W_summaries$W_post_mode is a d-by-c matrix giving the posterior mode of W.
</p>
<p>-Gibbs_W_summaries$W_post_sd is a d-by-c matrix giving the posterior standard deviation for each element of W.
</p>
<p>-Gibbs_W_summaries$W_2.5_quantile is a d-by-c matrix giving the posterior 2.5 percent quantile for each element of W.
</p>
<p>-Gibbs_W_summaries$W_97.5_quantile is a d-by-c matrix giving the posterior 97.5 percent quantile for each element of W.'</p>
</td></tr>
<tr><td><code>FDR_summaries</code></td>
<td>
<p>A list with three components providing the summaries for estimated Bayesian FDR results for both MCMC and MFVB methods. Details for Bayesian FDR computation could be found at Morris et al.(2008).
</p>
<p>-sensitivity_rate is the estimated sensitivity rate for each region.
</p>
<p>-specificity_rate is the estimated specificity rate for each region.
</p>
<p>-significant_snp_idx is the index of estimated significant/important SNPs for each region.
</p>
</td></tr>
<tr><td><code>MFVB_summaries</code></td>
<td>
<p>A list with four components, each component is the mean field variational bayes approximation summary of model paramters.
</p>
<p>-Number of Iteration is how many iterations it takes for convergence.
</p>
<p>-W_post_mean is MFVB approximation of W.
</p>
<p>-Sigma_post_mean is MFVB approximation of Sigma.
</p>
<p>-omega_post_mean is MFVB approximation of Omega.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yin Song, <a href="mailto:yinsong@uvic.ca">yinsong@uvic.ca</a>
</p>
<p>Shufei Ge, <a href="mailto:shufeig@sfu.ca">shufeig@sfu.ca</a>
</p>
<p>Farouk S. Nathoo, <a href="mailto:nathoo@uvic.ca">nathoo@uvic.ca</a>
</p>
<p>Liangliang Wang, <a href="mailto:lwa68@sfu.ca">lwa68@sfu.ca</a>
</p>
<p>Jiguo Cao, <a href="mailto:jiguo_cao@sfu.ca">jiguo_cao@sfu.ca</a>
</p>


<h3>References</h3>

<p>Song, Y., Ge, S., Cao, J., Wang, L., Nathoo, F.S., A Bayesian Spatial Model for Imaging Genetics. arXiv:1901.00068.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sp_bgsmtr_example_data)
names(sp_bgsmtr_example_data)


## Not run: 

# Run the example data with Gibbs sampling and compute Bayesian FDR as follow:

fit_mcmc = sp_bgsmtr(X = sp_bgsmtr_example_data$SNP_data,
Y = sp_bgsmtr_example_data$BrainMeasures, method = "MCMC",
A = sp_bgsmtr_example_data$neighborhood_structure, rho = 0.8,
FDR_opt = TRUE, WAIC_opt = TRUE,lambdasq = 1000, iter_num = 10000)

# MCMC estimation results for regression parameter W and estimated Bayesian FDR summaries

fit_mcmc$Gibbs_W_summaries
fit_mcmc$FDR_summaries

# The WAIC could be also obtained as:

fit_mcmc$WAIC

# Run the example data with mean field variational Bayes and compute Bayesian FDR as follow:

fit_mfvb = sp_bgsmtr(X = sp_bgsmtr_example_data$SNP_data,
Y = sp_bgsmtr_example_data$BrainMeasures, method = "MFVB",
A = sp_bgsmtr_example_data$neighborhood_structure, rho = 0.8,FDR_opt = FALSE,
lambdasq = 1000, iter_num = 10000)

# MFVB estimated results for regression parameter W and estimated Bayesian FDR summaries
fit_mfvb$MFVB_summaries
fit_mfvb$FDR_summaries

# The corresponding lower bound of MFVB method after convergence is obtained as:
fit_mfvb$lower_boud


## End(Not run)

</code></pre>

<hr>
<h2 id='sp_bgsmtr_example_data'>Example Structural Neuroimaging and Genetic Data for Spatial Model.</h2><span id='topic+sp_bgsmtr_example_data'></span>

<h3>Description</h3>

<p>This example dataset has simulated dataset with 632 subjects, 486 SNPs from 24 structural neuroimaging measures. It also contains example datasets used for regularization path plotting function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sp_bgsmtr_example_data)
</code></pre>


<h3>Format</h3>

<p>A list with four components: &quot;SNP_data&quot;, &quot;SNP_groups&quot;, &quot;BrainMeasures&quot;, &quot;path_data&quot;
SNP_data is a 486-by-632 matrix containing minor allele counts for 632 subjects and 486 SNPs.
neighbourhood_structure is a 12 by 12 first order neighbourhood structure matrix.
BrainMeasures is a 24-by-632 matrix containing simulated volumetric and cortical thickness measures for 24 regions of interest.
</p>
<p>path_data is a list of two elements. The fist element is &quot;lambda_v&quot;, which is a vector of different lambda squared values. The second element is &quot;W_est_list&quot;, which is a list containing estimated coefficients matrices W for each corresponding lambda squared value.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sp_bgsmtr_example_data)
names(sp_bgsmtr_example_data)
dim(sp_bgsmtr_example_data$SNP_data)
dim(sp_bgsmtr_example_data$BrainMeasures)
dim(sp_bgsmtr_example_data$neighbourhood_structure)
length(sp_bgsmtr_example_data$path_data$lambda_v)
length(sp_bgsmtr_example_data$path_data$W_est_list)
</code></pre>

<hr>
<h2 id='sp_bgsmtr_path'>A Bayesian Spatial Model for Imaging Genetics</h2><span id='topic+sp_bgsmtr_path'></span>

<h3>Description</h3>

<p>A plotting function can be used to demonstrate the regularization paths for estimating parameters of each ROI when the spatial model is fitted with multiple values of tuning parameter lambda-squared.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sp_bgsmtr_path(lambda_v, W_est_list)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sp_bgsmtr_path_+3A_lambda_v">lambda_v</code></td>
<td>
<p>A vector containing all the different tuning parameter lambda-squared values for model fitting.</p>
</td></tr>
<tr><td><code id="sp_bgsmtr_path_+3A_w_est_list">W_est_list</code></td>
<td>
<p>A list containing all the estimated coefficients matrices W for each corresponding lambda squared value used in lambda_v for model fitting. Each element of this list is a d-by-c matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Regularization plots files in PDF format for each ROI.
</p>


<h3>Author(s)</h3>

<p>Yin Song, <a href="mailto:yinsong@uvic.ca">yinsong@uvic.ca</a>
</p>
<p>Shufei Ge, <a href="mailto:shufeig@sfu.ca">shufeig@sfu.ca</a>
</p>
<p>Farouk S. Nathoo, <a href="mailto:nathoo@uvic.ca">nathoo@uvic.ca</a>
</p>
<p>Liangliang Wang, <a href="mailto:lwa68@sfu.ca">lwa68@sfu.ca</a>
</p>
<p>Jiguo Cao, <a href="mailto:jiguo_cao@sfu.ca">jiguo_cao@sfu.ca</a>
</p>


<h3>References</h3>

<p>Song, Y., Ge, S., Cao, J., Wang, L., Nathoo, F.S., A Bayesian Spatial Model for Imaging Genetics. arXiv:1901.00068.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sp_bgsmtr_example_data$path_data)


## Not run: 

# Creating the regulazaition path plots  as follow:
sp_bgsmtr_path(lambda_v = sp_bgsmtr_example_data$path_data$lambda_v,
 W_est_list = sp_bgsmtr_example_data$path_data$W_est_list )


## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
