<!DOCTYPE html><html lang="en"><head><title>Help for package BICORN</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BICORN}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#A'><p>TF-gene regulation strength matrix</p></a></li>
<li><a href='#A_old'><p>TF-gene regulation strength matrix sampled from the previous round</p></a></li>
<li><a href='#A_sampling'><p>Regulation Strength Sampling Function</p></a></li>
<li><a href='#alpha'><p>Inverse-gamma distribution hyper-parameter alpha</p></a></li>
<li><a href='#base_line'><p>Gene baseline expression</p></a></li>
<li><a href='#base_line_old'><p>Gene baseline expression sampled from the previous round.</p></a></li>
<li><a href='#baseline_sampling'><p>Gene Baseline Expression Sampling Function</p></a></li>
<li><a href='#beta'><p>Inverse-gamma distribution hyper-parameter beta</p></a></li>
<li><a href='#BICORN'><p>BICORN Algorithm Function</p></a></li>
<li><a href='#Binding_genes'><p>Genes in the prior binding network</p></a></li>
<li><a href='#Binding_matrix'><p>Prior TF-gene binding network</p></a></li>
<li><a href='#Binding_TFs'><p>TFs in the prior binding network</p></a></li>
<li><a href='#C'><p>TF-gene binding network</p></a></li>
<li><a href='#C_old'><p>TF-gene binding network sampled from the previous round</p></a></li>
<li><a href='#C_prior'><p>Prior TF-gene binding network</p></a></li>
<li><a href='#C_sampling_cluster'><p>cis-Regulatory Module Sampling Function</p></a></li>
<li><a href='#data_integration'><p>Data Initialization for BICORN</p></a></li>
<li><a href='#Exp_data'><p>Gene expression data</p></a></li>
<li><a href='#Exp_genes'><p>Genes in the expression data</p></a></li>
<li><a href='#sigma_A'><p>Regulation strength variance</p></a></li>
<li><a href='#sigma_baseline'><p>Variance of baseline gene expression.</p></a></li>
<li><a href='#sigma_noise'><p>Variance of gene expression fitting residuals.</p></a></li>
<li><a href='#sigma_X'><p>Transcription factor activity variance</p></a></li>
<li><a href='#sigmanoise_sampling'><p>Fitting Residule Variance Sampling Function</p></a></li>
<li><a href='#X'><p>Transcription factr activity matrix</p></a></li>
<li><a href='#X_old'><p>Transcription factr activity matrix sampled from the previous round</p></a></li>
<li><a href='#X_sampling'><p>Transcription Factor Activity Sampling Function</p></a></li>
<li><a href='#Y'><p>Gene expression data used for module inference</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Integrative Inference of De Novo Cis-Regulatory Modules</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Prior transcription factor binding knowledge and target gene expression data are integrated in a Bayesian framework for functional cis-regulatory module inference. Using Gibbs sampling, we iteratively estimate transcription factor associations for each gene, regulation strength for each binding event and the hidden activity for each transcription factor.  </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-06-04 19:37:06 UTC; xichen</td>
</tr>
<tr>
<td>Author:</td>
<td>Xi Chen [aut, cre],
  Jianhua Xuan [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xi Chen &lt;xichen86@vt.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-06-06 10:30:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='A'>TF-gene regulation strength matrix</h2><span id='topic+A'></span>

<h3>Description</h3>

<p>A matrix of TF-gene regulation strength with genes as rows and TFs as columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>A</code></pre>


<h3>Format</h3>

<p>numeric matrix</p>

<hr>
<h2 id='A_old'>TF-gene regulation strength matrix sampled from the previous round</h2><span id='topic+A_old'></span>

<h3>Description</h3>

<p>A matrix of TF-gene regulation strength with genes as rows and TFs as columns, sampled from the previous round. During the Gibbs sampling process, this matrix is used as prior for a new round of regulation strength sampling.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>A_old</code></pre>


<h3>Format</h3>

<p>numeric matrix</p>

<hr>
<h2 id='A_sampling'>Regulation Strength Sampling Function</h2><span id='topic+A_sampling'></span>

<h3>Description</h3>

<p>Function 'A_sampling' estimates a regulation strength for each sampled binding event in C,
according to a posterior Gaussian distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>A_sampling(Y, C, A_old, X, base_line, C_prior, sigma_noise, sigma_A,
  sigma_baseline, sigma_X)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="A_sampling_+3A_y">Y</code></td>
<td>
<p>gene expression data matrix</p>
</td></tr>
<tr><td><code id="A_sampling_+3A_c">C</code></td>
<td>
<p>sampled TF-gene binding network</p>
</td></tr>
<tr><td><code id="A_sampling_+3A_a_old">A_old</code></td>
<td>
<p>regulatory strength sampled from the previous round,
used as a prior in current function</p>
</td></tr>
<tr><td><code id="A_sampling_+3A_x">X</code></td>
<td>
<p>sampled transcription factor activity matrix</p>
</td></tr>
<tr><td><code id="A_sampling_+3A_base_line">base_line</code></td>
<td>
<p>sampled gene expression baseline activity</p>
</td></tr>
<tr><td><code id="A_sampling_+3A_c_prior">C_prior</code></td>
<td>
<p>prior TF-gene binding network</p>
</td></tr>
<tr><td><code id="A_sampling_+3A_sigma_noise">sigma_noise</code></td>
<td>
<p>variance of gene expression fitting residuals</p>
</td></tr>
<tr><td><code id="A_sampling_+3A_sigma_a">sigma_A</code></td>
<td>
<p>variance of regulatory strength</p>
</td></tr>
<tr><td><code id="A_sampling_+3A_sigma_baseline">sigma_baseline</code></td>
<td>
<p>variance of gene expression baseline activity</p>
</td></tr>
<tr><td><code id="A_sampling_+3A_sigma_x">sigma_X</code></td>
<td>
<p>variance of transcription factor activity</p>
</td></tr>
</table>

<hr>
<h2 id='alpha'>Inverse-gamma distribution hyper-parameter alpha</h2><span id='topic+alpha'></span>

<h3>Description</h3>

<p>Hyper-parameter alpha of inverse-gamma distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alpha</code></pre>


<h3>Format</h3>

<p>scalar</p>

<hr>
<h2 id='base_line'>Gene baseline expression</h2><span id='topic+base_line'></span>

<h3>Description</h3>

<p>A vector of baseline expression for all genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>base_line</code></pre>


<h3>Format</h3>

<p>numeric vector</p>

<hr>
<h2 id='base_line_old'>Gene baseline expression sampled from the previous round.</h2><span id='topic+base_line_old'></span>

<h3>Description</h3>

<p>A vector of baseline expression for all genes, sampled from the previous round. During the Gibbs Samplig process, this is used as a prior for a new round of gene baseline expression sampling. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>base_line_old</code></pre>


<h3>Format</h3>

<p>numeric vector</p>

<hr>
<h2 id='baseline_sampling'>Gene Baseline Expression Sampling Function</h2><span id='topic+baseline_sampling'></span>

<h3>Description</h3>

<p>Function 'baseline_sampling' estimates a baseline expression for each gene,
according to a posterior Gaussian distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>baseline_sampling(Y, C, A, X, base_line_old, C_prior, sigma_noise, sigma_A,
  sigma_baseline, sigma_X)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="baseline_sampling_+3A_y">Y</code></td>
<td>
<p>gene expression data matrix</p>
</td></tr>
<tr><td><code id="baseline_sampling_+3A_c">C</code></td>
<td>
<p>sampled TF-gene binding network</p>
</td></tr>
<tr><td><code id="baseline_sampling_+3A_a">A</code></td>
<td>
<p>sampled regulatory strength matrix</p>
</td></tr>
<tr><td><code id="baseline_sampling_+3A_x">X</code></td>
<td>
<p>sampled transcription factor activity matrix</p>
</td></tr>
<tr><td><code id="baseline_sampling_+3A_base_line_old">base_line_old</code></td>
<td>
<p>prior gene expression baseline activity</p>
</td></tr>
<tr><td><code id="baseline_sampling_+3A_c_prior">C_prior</code></td>
<td>
<p>prior TF-gene binding network</p>
</td></tr>
<tr><td><code id="baseline_sampling_+3A_sigma_noise">sigma_noise</code></td>
<td>
<p>variance of gene expression fitting residuals</p>
</td></tr>
<tr><td><code id="baseline_sampling_+3A_sigma_a">sigma_A</code></td>
<td>
<p>variance of regulatory strength</p>
</td></tr>
<tr><td><code id="baseline_sampling_+3A_sigma_baseline">sigma_baseline</code></td>
<td>
<p>variance of gene expression baseline activity</p>
</td></tr>
<tr><td><code id="baseline_sampling_+3A_sigma_x">sigma_X</code></td>
<td>
<p>variance of transcription factor activity</p>
</td></tr>
</table>

<hr>
<h2 id='beta'>Inverse-gamma distribution hyper-parameter beta</h2><span id='topic+beta'></span>

<h3>Description</h3>

<p>Hyper-parameter beta of inverse-gamma distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>beta</code></pre>


<h3>Format</h3>

<p>scalar</p>

<hr>
<h2 id='BICORN'>BICORN Algorithm Function</h2><span id='topic+BICORN'></span>

<h3>Description</h3>

<p>Function 'BICORN' infers a posterior module-gene regulatory network by iteratively sampling regulatory strength,
transcription factor activity and several key model parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BICORN(BICORN_input = NULL, L = 100, output_threshold = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BICORN_+3A_bicorn_input">BICORN_input</code></td>
<td>
<p>this list structure contains TF symbols, gene symbols and candidate modules</p>
</td></tr>
<tr><td><code id="BICORN_+3A_l">L</code></td>
<td>
<p>total rounds of Gibbs Sampling.</p>
</td></tr>
<tr><td><code id="BICORN_+3A_output_threshold">output_threshold</code></td>
<td>
<p>number of rounds after which we start to record results.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
# load in the sample data input
data("sample.input")

# Data initialization (Integerate prior binding network and gene expression data)
BICORN_input&lt;-data_integration(Binding_matrix = Binding_matrix, Binding_TFs = Binding_TFs,
Binding_genes = Binding_genes, Exp_data = Exp_data, Exp_genes = Exp_genes,
Minimum_gene_per_module_regulate = 2)

# Infer cis-regulatory modules (TF combinations) and their target genes
BICORN_output&lt;-BICORN(BICORN_input, L = 2, output_threshold = 1)
</code></pre>

<hr>
<h2 id='Binding_genes'>Genes in the prior binding network</h2><span id='topic+Binding_genes'></span>

<h3>Description</h3>

<p>A list of offical gene symbols in the binary binding network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Binding_genes</code></pre>


<h3>Format</h3>

<p>character vector</p>

<hr>
<h2 id='Binding_matrix'>Prior TF-gene binding network</h2><span id='topic+Binding_matrix'></span>

<h3>Description</h3>

<p>A prior binary TF-gene regulatory network with each unit either 1 (binding) or 0 (non-binding).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Binding_matrix
</code></pre>


<h3>Format</h3>

<p>numeric matrix</p>

<hr>
<h2 id='Binding_TFs'>TFs in the prior binding network</h2><span id='topic+Binding_TFs'></span>

<h3>Description</h3>

<p>A list of transcription factors in the prior binding network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Binding_TFs</code></pre>


<h3>Format</h3>

<p>character vector</p>

<hr>
<h2 id='C'>TF-gene binding network</h2><span id='topic+C'></span>

<h3>Description</h3>

<p>A matrix of TF-gene regulatory network with each unit either 1 (binding) or 0 (non-binding).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>C</code></pre>


<h3>Format</h3>

<p>numeric matrix</p>

<hr>
<h2 id='C_old'>TF-gene binding network sampled from the previous round</h2><span id='topic+C_old'></span>

<h3>Description</h3>

<p>A matrix of TF-gene binding network sampled from the previous round, with each unit either 1 (binding) or 0 (non-binding). During the Gibbs sampling process, this is used as a prior for a new round of binding network sampling.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>C_old</code></pre>


<h3>Format</h3>

<p>numeric matrix</p>

<hr>
<h2 id='C_prior'>Prior TF-gene binding network</h2><span id='topic+C_prior'></span>

<h3>Description</h3>

<p>A matrix of prior TF-gene binding events, with each unit either 1 (binding) or 0 (non-binding). Such a prior network can be obtained from TF-gene binding database, motif searching, ChIP-seq peaks or ATAC-seq peaks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>C_prior</code></pre>


<h3>Format</h3>

<p>numeric matrix</p>

<hr>
<h2 id='C_sampling_cluster'>cis-Regulatory Module Sampling Function</h2><span id='topic+C_sampling_cluster'></span>

<h3>Description</h3>

<p>Function 'C_sampling_cluster' samples a candidate cis-regulatory module for each gene,
according to a discrete posterior probability distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>C_sampling_cluster(Y, C_old, A_old, X_old, base_line_old, C_prior, sigma_noise,
  sigma_A, sigma_baseline, sigma_X, BICORN_input)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="C_sampling_cluster_+3A_y">Y</code></td>
<td>
<p>gene expression data matrix</p>
</td></tr>
<tr><td><code id="C_sampling_cluster_+3A_c_old">C_old</code></td>
<td>
<p>TF-gene binding network sampled from the previous round</p>
</td></tr>
<tr><td><code id="C_sampling_cluster_+3A_a_old">A_old</code></td>
<td>
<p>regulatory strength matrix sampled from the previous round</p>
</td></tr>
<tr><td><code id="C_sampling_cluster_+3A_x_old">X_old</code></td>
<td>
<p>transcription factor activity matrix sampled from the previous round</p>
</td></tr>
<tr><td><code id="C_sampling_cluster_+3A_base_line_old">base_line_old</code></td>
<td>
<p>gene expression baseline activity sampled from the previous round</p>
</td></tr>
<tr><td><code id="C_sampling_cluster_+3A_c_prior">C_prior</code></td>
<td>
<p>prior TF-gene binding network</p>
</td></tr>
<tr><td><code id="C_sampling_cluster_+3A_sigma_noise">sigma_noise</code></td>
<td>
<p>variance of gene expression fitting residuals</p>
</td></tr>
<tr><td><code id="C_sampling_cluster_+3A_sigma_a">sigma_A</code></td>
<td>
<p>variance of regulatory strength</p>
</td></tr>
<tr><td><code id="C_sampling_cluster_+3A_sigma_baseline">sigma_baseline</code></td>
<td>
<p>variance of gene expression baseline activity</p>
</td></tr>
<tr><td><code id="C_sampling_cluster_+3A_sigma_x">sigma_X</code></td>
<td>
<p>variance of transcription factor activity</p>
</td></tr>
<tr><td><code id="C_sampling_cluster_+3A_bicorn_input">BICORN_input</code></td>
<td>
<p>this list structure contains TF symbols, Gene symbols and candidate modules</p>
</td></tr>
</table>

<hr>
<h2 id='data_integration'>Data Initialization for BICORN</h2><span id='topic+data_integration'></span>

<h3>Description</h3>

<p>Function 'data_integration' integrates the prior TF-gene binding network and gene expression data together. It will remove any genes missing either TF bindings or gene expression and identify a list of candidate cis-regulatory modules.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_integration(Binding_matrix = NULL, Binding_TFs = NULL,
  Binding_genes = NULL, Exp_data, Exp_genes = NULL,
  Minimum_gene_per_module_regulate = 2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="data_integration_+3A_binding_matrix">Binding_matrix</code></td>
<td>
<p>loaded prior binding network</p>
</td></tr>
<tr><td><code id="data_integration_+3A_binding_tfs">Binding_TFs</code></td>
<td>
<p>loaded transcription factors</p>
</td></tr>
<tr><td><code id="data_integration_+3A_binding_genes">Binding_genes</code></td>
<td>
<p>loaded genes in the prior binding network</p>
</td></tr>
<tr><td><code id="data_integration_+3A_exp_data">Exp_data</code></td>
<td>
<p>loaded properly normalized gene expression data</p>
</td></tr>
<tr><td><code id="data_integration_+3A_exp_genes">Exp_genes</code></td>
<td>
<p>loaded genes in the gene expression data</p>
</td></tr>
<tr><td><code id="data_integration_+3A_minimum_gene_per_module_regulate">Minimum_gene_per_module_regulate</code></td>
<td>
<p>the minimum number of genes regulated by each module, used for candidate module filtering.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
# load in the sample data input
data("sample.input")

# Data initialization (Integerate prior binding network and gene expression data)
BICORN_input&lt;-data_integration(Binding_matrix = Binding_matrix, Binding_TFs = Binding_TFs,
Binding_genes = Binding_genes, Exp_data = Exp_data, Exp_genes = Exp_genes,
Minimum_gene_per_module_regulate = 2)
</code></pre>

<hr>
<h2 id='Exp_data'>Gene expression data</h2><span id='topic+Exp_data'></span>

<h3>Description</h3>

<p>A matrix of normalized gene expression data with genes as rows and samples as columns. The gene expression data can be either time-course data measured under multiple time points or steady state data generated from at least two different conditions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Exp_data</code></pre>


<h3>Format</h3>

<p>numeric matrix</p>

<hr>
<h2 id='Exp_genes'>Genes in the expression data</h2><span id='topic+Exp_genes'></span>

<h3>Description</h3>

<p>A list of official gene symbols in the gene expression data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Exp_genes</code></pre>


<h3>Format</h3>

<p>character vector</p>

<hr>
<h2 id='sigma_A'>Regulation strength variance</h2><span id='topic+sigma_A'></span>

<h3>Description</h3>

<p>Variance of regulation strength matrix A.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sigma_A</code></pre>


<h3>Format</h3>

<p>scalar</p>

<hr>
<h2 id='sigma_baseline'>Variance of baseline gene expression.</h2><span id='topic+sigma_baseline'></span>

<h3>Description</h3>

<p>Variance of baseline gene expression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sigma_baseline</code></pre>


<h3>Format</h3>

<p>scalar</p>

<hr>
<h2 id='sigma_noise'>Variance of gene expression fitting residuals.</h2><span id='topic+sigma_noise'></span>

<h3>Description</h3>

<p>Variance of gene expression fitting residuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sigma_noise</code></pre>


<h3>Format</h3>

<p>scalar</p>

<hr>
<h2 id='sigma_X'>Transcription factor activity variance</h2><span id='topic+sigma_X'></span>

<h3>Description</h3>

<p>Variance of transcription factor activity matrix X.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sigma_X</code></pre>


<h3>Format</h3>

<p>scalar</p>

<hr>
<h2 id='sigmanoise_sampling'>Fitting Residule Variance Sampling Function</h2><span id='topic+sigmanoise_sampling'></span>

<h3>Description</h3>

<p>Function 'sigmanoise_sampling' estimates the variance of overal gene expression fitting residuals,
according to an inverse-gamma distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sigmanoise_sampling(Y, C, A, X, base_line, C_prior, sigma_A, sigma_baseline,
  sigma_X, alpha, beta)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sigmanoise_sampling_+3A_y">Y</code></td>
<td>
<p>gene expression data matrix</p>
</td></tr>
<tr><td><code id="sigmanoise_sampling_+3A_c">C</code></td>
<td>
<p>sampled TF-gene binding network</p>
</td></tr>
<tr><td><code id="sigmanoise_sampling_+3A_a">A</code></td>
<td>
<p>sampled regulatory strength matrix</p>
</td></tr>
<tr><td><code id="sigmanoise_sampling_+3A_x">X</code></td>
<td>
<p>sampled transcription factor activity matrix</p>
</td></tr>
<tr><td><code id="sigmanoise_sampling_+3A_base_line">base_line</code></td>
<td>
<p>sampled gene expression baseline activity</p>
</td></tr>
<tr><td><code id="sigmanoise_sampling_+3A_c_prior">C_prior</code></td>
<td>
<p>prior TF-gene binding network</p>
</td></tr>
<tr><td><code id="sigmanoise_sampling_+3A_sigma_a">sigma_A</code></td>
<td>
<p>variance of regulatory strength</p>
</td></tr>
<tr><td><code id="sigmanoise_sampling_+3A_sigma_baseline">sigma_baseline</code></td>
<td>
<p>variance of gene expression baseline activity</p>
</td></tr>
<tr><td><code id="sigmanoise_sampling_+3A_sigma_x">sigma_X</code></td>
<td>
<p>variance of transcription factor activity</p>
</td></tr>
<tr><td><code id="sigmanoise_sampling_+3A_alpha">alpha</code></td>
<td>
<p>hyper-parameter for inverse-gamma distribution</p>
</td></tr>
<tr><td><code id="sigmanoise_sampling_+3A_beta">beta</code></td>
<td>
<p>hyper-parameter for inverse-gamma distribution</p>
</td></tr>
</table>

<hr>
<h2 id='X'>Transcription factr activity matrix</h2><span id='topic+X'></span>

<h3>Description</h3>

<p>A matrix of hidden transcription factr activity estimated from gene expression data, with transcription factrs as rows and samples as columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>X</code></pre>


<h3>Format</h3>

<p>numeric matrix</p>

<hr>
<h2 id='X_old'>Transcription factr activity matrix sampled from the previous round</h2><span id='topic+X_old'></span>

<h3>Description</h3>

<p>A matrix of hidden transcription factr activity estimated from gene expression data, with transcription factrs as rows and samples as columns, sampled from the previous round. During the Gibbs sampling process, this is used as a prior for a new round of transcription factor activity sampling.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>X_old</code></pre>


<h3>Format</h3>

<p>numeric matrix</p>

<hr>
<h2 id='X_sampling'>Transcription Factor Activity Sampling Function</h2><span id='topic+X_sampling'></span>

<h3>Description</h3>

<p>Function 'X_sampling' estimates the hidden activities of each transcription factor,
according to a posterior Gaussian random process.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>X_sampling(Y, C, A, X_old, base_line, C_prior, sigma_noise, sigma_A,
  sigma_baseline, sigma_X)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="X_sampling_+3A_y">Y</code></td>
<td>
<p>gene expression data matrix</p>
</td></tr>
<tr><td><code id="X_sampling_+3A_c">C</code></td>
<td>
<p>sampled TF-gene binding network</p>
</td></tr>
<tr><td><code id="X_sampling_+3A_a">A</code></td>
<td>
<p>sampled regulatory strength matrix</p>
</td></tr>
<tr><td><code id="X_sampling_+3A_x_old">X_old</code></td>
<td>
<p>sampled transcription factor activity matrix from the previous round</p>
</td></tr>
<tr><td><code id="X_sampling_+3A_base_line">base_line</code></td>
<td>
<p>sampled gene expression baseline activity</p>
</td></tr>
<tr><td><code id="X_sampling_+3A_c_prior">C_prior</code></td>
<td>
<p>prior TF-gene binding network</p>
</td></tr>
<tr><td><code id="X_sampling_+3A_sigma_noise">sigma_noise</code></td>
<td>
<p>variance of gene expression fitting residuals</p>
</td></tr>
<tr><td><code id="X_sampling_+3A_sigma_a">sigma_A</code></td>
<td>
<p>variance of regulatory strength</p>
</td></tr>
<tr><td><code id="X_sampling_+3A_sigma_baseline">sigma_baseline</code></td>
<td>
<p>variance of gene expression baseline activity</p>
</td></tr>
<tr><td><code id="X_sampling_+3A_sigma_x">sigma_X</code></td>
<td>
<p>variance of transcription factor activity</p>
</td></tr>
</table>

<hr>
<h2 id='Y'>Gene expression data used for module inference</h2><span id='topic+Y'></span>

<h3>Description</h3>

<p>A matrix of normalized gene expression for common genes of prior binding input and gene expression input, with genes as rows and samples as columns. Y is the matrix used for cis-regulatory mudole inference.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Y</code></pre>


<h3>Format</h3>

<p>numeric matrix</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
