<!DOCTYPE html><html lang="en"><head><title>Help for package BIGDAWG</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BIGDAWG}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#A'><p>Amino Acid Analysis Function</p></a></li>
<li><a href='#A.wrapper'><p>Amino Acid Wrapper</p></a></li>
<li><a href='#AA.df.check'><p>Contingency Table Check</p></a></li>
<li><a href='#AA.df.cs'><p>Contingency Table Amino Acid ChiSq Testing</p></a></li>
<li><a href='#AAtable.builder'><p>Amino Acid Contingency Table Build</p></a></li>
<li><a href='#AlignmentFilter'><p>Alignment Filter</p></a></li>
<li><a href='#AlignObj.Create'><p>Alignment Object Creator</p></a></li>
<li><a href='#AlignObj.Update'><p>Updated Alignment Object Creator</p></a></li>
<li><a href='#Append.System'><p>Append Genetic System Locus Designation to Allele String</p></a></li>
<li><a href='#BIGDAWG'><p>BIGDAWG Main Wrapper Function</p></a></li>
<li><a href='#Build.Matrix'><p>Build Output Matrix for GL2Tab Conversion</p></a></li>
<li><a href='#buildHAPnames'><p>Haplotype Name Builder</p></a></li>
<li><a href='#buildHAPsets'><p>Haplotype List Builder</p></a></li>
<li><a href='#cci'><p>Case-Control Odds ratio calculation and graphing</p></a></li>
<li><a href='#cci.pval'><p>Case Control Odds Ratio Calculation from Epicalc</p></a></li>
<li><a href='#cci.pval.list'><p>Case Control Odds Ratio Calculation from Epicalc list variation</p></a></li>
<li><a href='#Check.Cores'><p>Check Cores Parameters</p></a></li>
<li><a href='#Check.Data'><p>Check Data Structure</p></a></li>
<li><a href='#Check.Params'><p>Check Input Parameters</p></a></li>
<li><a href='#Check.Params.GLS'><p>Check Input Parameters for GLS conversion</p></a></li>
<li><a href='#CheckAlleles'><p>HLA Allele Legitimacy Check for Amino Acid Analysis</p></a></li>
<li><a href='#CheckHLA'><p>HLA Formatting Check for Amino Acid Analysis</p></a></li>
<li><a href='#CheckLoci'><p>HLA Loci Legitimacy Check for Amino Acid Analysis</p></a></li>
<li><a href='#CheckRelease'><p>Function to Check Release Versions</p></a></li>
<li><a href='#CheckString.Allele'><p>GL String Allele Check</p></a></li>
<li><a href='#CheckString.Locus'><p>GL String Locus Check</p></a></li>
<li><a href='#Condense.EPL'><p>Condensing Exon Specific Alignments to Single Dataframe</p></a></li>
<li><a href='#Create.Null.Table'><p>Create Empty Table</p></a></li>
<li><a href='#DRB345.Check.Wrapper'><p>DRB345 haplotype zygosity wrapper</p></a></li>
<li><a href='#DRB345.Check.Zygosity'><p>DRB345 haplotype zygosity checker single locus</p></a></li>
<li><a href='#DRB345.Exp'><p>DRB345 Expected</p></a></li>
<li><a href='#DRB345.parser'><p>DRB345 Column Processing</p></a></li>
<li><a href='#Err.Log'><p>Error Code Display and Logging</p></a></li>
<li><a href='#EVSremoval'><p>Expression Variant Suffix Removal</p></a></li>
<li><a href='#Exon.Filter'><p>Filter Exon Specific Alignment Sections</p></a></li>
<li><a href='#ExonPtnAlign.Create'><p>Protein Exon Alignment Formatter</p></a></li>
<li><a href='#ExonPtnList'><p>Exon protein alignments.</p></a></li>
<li><a href='#Filler'><p>Replace or Fill 00:00 allele strings</p></a></li>
<li><a href='#Format.Allele'><p>Ambiguous Alleles Locus Name Formatting</p></a></li>
<li><a href='#Format.Tab'><p>Tabular Data Locus Format Tool</p></a></li>
<li><a href='#getAllele.Count'><p>Recompute number of alleles</p></a></li>
<li><a href='#getCS.Mat'><p>Chi square matrices</p></a></li>
<li><a href='#getCS.stat'><p>Chi square test statistic</p></a></li>
<li><a href='#GetField'><p>HLA trimming function</p></a></li>
<li><a href='#getFileName'><p>File Name Extraction</p></a></li>
<li><a href='#GetFiles'><p>File Fetcher</p></a></li>
<li><a href='#getHap'><p>Haplotype Table Maker</p></a></li>
<li><a href='#getObsFreq'><p>Observed Frequency</p></a></li>
<li><a href='#GL2Tab.Loci'><p>Locus Ordering for GL2Tab</p></a></li>
<li><a href='#GL2Tab.Sub'><p>Genotype List String Expander</p></a></li>
<li><a href='#GL2Tab.wrapper'><p>Genotype List String to Tabular Data Conversion</p></a></li>
<li><a href='#GLSconvert'><p>Genotype List String Conversion</p></a></li>
<li><a href='#H.MC'><p>Haplotype Analysis Function for Multicore</p></a></li>
<li><a href='#H.MC.wrapper'><p>Haplotype Wrapper for Multicore</p></a></li>
<li><a href='#HLA_data'><p>Example HLA Dataset</p></a></li>
<li><a href='#HWE'><p>Hardy Weinbergy Equilibrium Function</p></a></li>
<li><a href='#HWE.ChiSq'><p>Hardy Weinbergy Equilibrium Function</p></a></li>
<li><a href='#HWE.wrapper'><p>Hardy-Weinbery Wrapper</p></a></li>
<li><a href='#L'><p>Locus Analysis Function</p></a></li>
<li><a href='#L.wrapper'><p>Locus Wrapper</p></a></li>
<li><a href='#make2x2'><p>Creation of a 2x2 table using the indicated orientation.</p></a></li>
<li><a href='#makeComb'><p>Genotype Combination Maker</p></a></li>
<li><a href='#MergeData_Output'><p>Data Object Merge and Output</p></a></li>
<li><a href='#PgrpExtract'><p>HLA P group Finder</p></a></li>
<li><a href='#PgrpFormat'><p>HLA P group File Formatter</p></a></li>
<li><a href='#PreCheck'><p>Data Summary Function</p></a></li>
<li><a href='#prepData'><p>Prepare imported data</p></a></li>
<li><a href='#rmABstrings'><p>Replace absent allele strings</p></a></li>
<li><a href='#RunChiSq'><p>Strict Chi-squared Contingency Table Test</p></a></li>
<li><a href='#RunChiSq_c'><p>Contextual Binning Chi-squared Contingency Table Test</p></a></li>
<li><a href='#Stripper'><p>Removes System and Locus from Alleles</p></a></li>
<li><a href='#summaryGeno.2'><p>Haplotype missing Allele summary function</p></a></li>
<li><a href='#Tab2GL.Loci'><p>Locus Condenser for Tab2GL</p></a></li>
<li><a href='#Tab2GL.Sub'><p>Genotype List String Condenser</p></a></li>
<li><a href='#Tab2GL.wrapper'><p>Genotype List String to Tabular Data Conversion</p></a></li>
<li><a href='#TableMaker'><p>Table Maker</p></a></li>
<li><a href='#UpdateRelease'><p>Update function for protein aligment upon new IMGT HLA data release</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Case-Control Analysis of Multi-Allelic Loci</td>
</tr>
<tr>
<td>Version:</td>
<td>3.0.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-11-01</td>
</tr>
<tr>
<td>Author:</td>
<td>Derek Pappas &lt;djpappas75@gmail.com&gt;, Steve Mack &lt;Steven.Mack@ucsf.edu&gt;, Jill Hollenbach &lt;Jill.Hollenbach@ucsf.edu&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Steve Mack &lt;Steven.Mack@ucsf.edu&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://tools.immunogenomics.org/">http://tools.immunogenomics.org/</a>,
<a href="https://github.com/IgDAWG/BIGDAWG">https://github.com/IgDAWG/BIGDAWG</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/IgDAWG/BIGDAWG/issues">https://github.com/IgDAWG/BIGDAWG/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Data sets and functions for chi-squared Hardy-Weinberg and case-control association tests of highly polymorphic genetic data [e.g., human leukocyte antigen (HLA) data]. Performs association tests at multiple levels of polymorphism (haplotype, locus and HLA amino-acids) as described in Pappas DJ, Marin W, Hollenbach JA, Mack SJ (2016) &lt;<a href="https://doi.org/10.1016%2Fj.humimm.2015.12.006">doi:10.1016/j.humimm.2015.12.006</a>&gt;. Combines rare variants to a common class to account for sparse cells in tables as described by Hollenbach JA, Mack SJ, Thomson G, Gourraud PA (2012) &lt;<a href="https://doi.org/10.1007%2F978-1-61779-842-9_14">doi:10.1007/978-1-61779-842-9_14</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>XML, httr, haplo.stats, parallel</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-11-16 16:53:55 UTC; derek</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-11-17 11:50:14 UTC</td>
</tr>
</table>
<hr>
<h2 id='A'>Amino Acid Analysis Function</h2><span id='topic+A'></span>

<h3>Description</h3>

<p>This is the workhorse function for the amino acid analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>A(Locus, loci.ColNames, genos, grp, Strict.Bin, ExonAlign, Cores)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="A_+3A_locus">Locus</code></td>
<td>
<p>Locus being analyzed.</p>
</td></tr>
<tr><td><code id="A_+3A_loci.colnames">loci.ColNames</code></td>
<td>
<p>The column names of the loci being analyzed.</p>
</td></tr>
<tr><td><code id="A_+3A_genos">genos</code></td>
<td>
<p>Genotype table.</p>
</td></tr>
<tr><td><code id="A_+3A_grp">grp</code></td>
<td>
<p>Case/Control or Phenotype groupings.</p>
</td></tr>
<tr><td><code id="A_+3A_strict.bin">Strict.Bin</code></td>
<td>
<p>Logical specify if strict rare cell binning should be used in ChiSq test.</p>
</td></tr>
<tr><td><code id="A_+3A_exonalign">ExonAlign</code></td>
<td>
<p>Exon protein alignment filtered for locus.</p>
</td></tr>
<tr><td><code id="A_+3A_cores">Cores</code></td>
<td>
<p>Number of cores to use for analysis.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='A.wrapper'>Amino Acid Wrapper</h2><span id='topic+A.wrapper'></span>

<h3>Description</h3>

<p>Wrapper function for amino acid analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>A.wrapper(
  loci,
  loci.ColNames,
  genos,
  grp,
  Exon,
  EPL,
  Cores,
  Strict.Bin,
  Output,
  Verbose
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="A.wrapper_+3A_loci">loci</code></td>
<td>
<p>Loci being analyzed.</p>
</td></tr>
<tr><td><code id="A.wrapper_+3A_loci.colnames">loci.ColNames</code></td>
<td>
<p>The column names of the loci being analyzed.</p>
</td></tr>
<tr><td><code id="A.wrapper_+3A_genos">genos</code></td>
<td>
<p>Genotype table.</p>
</td></tr>
<tr><td><code id="A.wrapper_+3A_grp">grp</code></td>
<td>
<p>Case/Control or Phenotype groupings.</p>
</td></tr>
<tr><td><code id="A.wrapper_+3A_exon">Exon</code></td>
<td>
<p>Exon(s)for targeted analysis.</p>
</td></tr>
<tr><td><code id="A.wrapper_+3A_epl">EPL</code></td>
<td>
<p>Protein Alignment List.</p>
</td></tr>
<tr><td><code id="A.wrapper_+3A_cores">Cores</code></td>
<td>
<p>Number of cores to use for analysis.</p>
</td></tr>
<tr><td><code id="A.wrapper_+3A_strict.bin">Strict.Bin</code></td>
<td>
<p>Logical specify if strict rare cell binning should be used in ChiSq test</p>
</td></tr>
<tr><td><code id="A.wrapper_+3A_output">Output</code></td>
<td>
<p>Data return carryover from main BIGDAWG function</p>
</td></tr>
<tr><td><code id="A.wrapper_+3A_verbose">Verbose</code></td>
<td>
<p>Summary display carryover from main BIGDAWG function</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='AA.df.check'>Contingency Table Check</h2><span id='topic+AA.df.check'></span>

<h3>Description</h3>

<p>Checks amino acid contingency table data frame to ensure required variation exists.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AA.df.check(x, Strict.Bin)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AA.df.check_+3A_x">x</code></td>
<td>
<p>contingency table.</p>
</td></tr>
<tr><td><code id="AA.df.check_+3A_strict.bin">Strict.Bin</code></td>
<td>
<p>Logical specify if strict rare cell binning should be used in ChiSq test.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='AA.df.cs'>Contingency Table Amino Acid ChiSq Testing</h2><span id='topic+AA.df.cs'></span>

<h3>Description</h3>

<p>Runs ChiSq test on amino acid contingency table data frames.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AA.df.cs(x, Strict.Bin)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AA.df.cs_+3A_x">x</code></td>
<td>
<p>contingency table.</p>
</td></tr>
<tr><td><code id="AA.df.cs_+3A_strict.bin">Strict.Bin</code></td>
<td>
<p>Logical specify if strict rare cell binning should be used in ChiSq test.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='AAtable.builder'>Amino Acid Contingency Table Build</h2><span id='topic+AAtable.builder'></span>

<h3>Description</h3>

<p>Build Contingency Tables for Amino Acid Analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AAtable.builder(x, y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AAtable.builder_+3A_x">x</code></td>
<td>
<p>Filtered alignmnet list element.</p>
</td></tr>
<tr><td><code id="AAtable.builder_+3A_y">y</code></td>
<td>
<p>Phenotype groupings.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='AlignmentFilter'>Alignment Filter</h2><span id='topic+AlignmentFilter'></span>

<h3>Description</h3>

<p>Filter Protein Exon Alignment File for Specific Alleles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AlignmentFilter(Align, Alleles, Locus)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AlignmentFilter_+3A_align">Align</code></td>
<td>
<p>Protein Alignment Object.</p>
</td></tr>
<tr><td><code id="AlignmentFilter_+3A_alleles">Alleles</code></td>
<td>
<p>to be pulled.</p>
</td></tr>
<tr><td><code id="AlignmentFilter_+3A_locus">Locus</code></td>
<td>
<p>Locus to be filtered against.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='AlignObj.Create'>Alignment Object Creator</h2><span id='topic+AlignObj.Create'></span>

<h3>Description</h3>

<p>Create Object for Exon Protein Alignments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AlignObj.Create(Loci, Release, RefTab)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AlignObj.Create_+3A_loci">Loci</code></td>
<td>
<p>Loci to be bundled.</p>
</td></tr>
<tr><td><code id="AlignObj.Create_+3A_release">Release</code></td>
<td>
<p>IMGT/HLA database release version.</p>
</td></tr>
<tr><td><code id="AlignObj.Create_+3A_reftab">RefTab</code></td>
<td>
<p>Data of reference exons used for protein alignment creation.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='AlignObj.Update'>Updated Alignment Object Creator</h2><span id='topic+AlignObj.Update'></span>

<h3>Description</h3>

<p>Synthesize Object for Exon Protein Alignments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AlignObj.Update(Loci, Release, RefTab)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AlignObj.Update_+3A_loci">Loci</code></td>
<td>
<p>Loci to be bundled.</p>
</td></tr>
<tr><td><code id="AlignObj.Update_+3A_release">Release</code></td>
<td>
<p>IMGT/HLA database release version.</p>
</td></tr>
<tr><td><code id="AlignObj.Update_+3A_reftab">RefTab</code></td>
<td>
<p>Data of reference exons used for protein alignment creation.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='Append.System'>Append Genetic System Locus Designation to Allele String</h2><span id='topic+Append.System'></span>

<h3>Description</h3>

<p>Adds genetic system (HLA/KIR) to each allele name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Append.System(x, df.name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Append.System_+3A_x">x</code></td>
<td>
<p>Vector Column genotypes to append</p>
</td></tr>
<tr><td><code id="Append.System_+3A_df.name">df.name</code></td>
<td>
<p>String SystemLocus name for each allele.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only.
</p>

<hr>
<h2 id='BIGDAWG'>BIGDAWG Main Wrapper Function</h2><span id='topic+BIGDAWG'></span>

<h3>Description</h3>

<p>This is the main wrapper function for each analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BIGDAWG(
  Data,
  HLA = TRUE,
  Run.Tests,
  Loci.Set,
  Exon,
  All.Pairwise = FALSE,
  Trim = FALSE,
  Res = 2,
  EVS.rm = FALSE,
  Missing = 2,
  Strict.Bin = FALSE,
  Cores.Lim = 1L,
  Results.Dir,
  Return = FALSE,
  Output = TRUE,
  Merge.Output = FALSE,
  Verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BIGDAWG_+3A_data">Data</code></td>
<td>
<p>Name of the genotype data file.</p>
</td></tr>
<tr><td><code id="BIGDAWG_+3A_hla">HLA</code></td>
<td>
<p>Logical Indicating whether data is HLA class I/II genotyping data only.</p>
</td></tr>
<tr><td><code id="BIGDAWG_+3A_run.tests">Run.Tests</code></td>
<td>
<p>Specifics which tests to run.</p>
</td></tr>
<tr><td><code id="BIGDAWG_+3A_loci.set">Loci.Set</code></td>
<td>
<p>Input list defining which loci to use for analyses (combinations permitted).</p>
</td></tr>
<tr><td><code id="BIGDAWG_+3A_exon">Exon</code></td>
<td>
<p>Numeric Exon(s) for targeted amino acid analysis.</p>
</td></tr>
<tr><td><code id="BIGDAWG_+3A_all.pairwise">All.Pairwise</code></td>
<td>
<p>Logical indicating whether all pairwise loci should be analyzed in haplotype analysis.</p>
</td></tr>
<tr><td><code id="BIGDAWG_+3A_trim">Trim</code></td>
<td>
<p>Logical indicating if HLA alleles should be trimmed to a set resolution.</p>
</td></tr>
<tr><td><code id="BIGDAWG_+3A_res">Res</code></td>
<td>
<p>Numeric setting what desired resolution to trim HLA alleles.</p>
</td></tr>
<tr><td><code id="BIGDAWG_+3A_evs.rm">EVS.rm</code></td>
<td>
<p>Logical indicating if expression variant suffixes should be removed.</p>
</td></tr>
<tr><td><code id="BIGDAWG_+3A_missing">Missing</code></td>
<td>
<p>Numeric setting allowable missing data for running analysis (may use &quot;ignore&quot;).</p>
</td></tr>
<tr><td><code id="BIGDAWG_+3A_strict.bin">Strict.Bin</code></td>
<td>
<p>Logical specify if strict rare cell binning should be used in ChiSq test.</p>
</td></tr>
<tr><td><code id="BIGDAWG_+3A_cores.lim">Cores.Lim</code></td>
<td>
<p>Integer setting the number of cores accessible to BIGDAWG (Windows limit is 1 core).</p>
</td></tr>
<tr><td><code id="BIGDAWG_+3A_results.dir">Results.Dir</code></td>
<td>
<p>Optional, string of full path directory name for BIGDAWG output.</p>
</td></tr>
<tr><td><code id="BIGDAWG_+3A_return">Return</code></td>
<td>
<p>Logical Should analysis results be returned as list.</p>
</td></tr>
<tr><td><code id="BIGDAWG_+3A_output">Output</code></td>
<td>
<p>Logical Should analysis results be written to output directory.</p>
</td></tr>
<tr><td><code id="BIGDAWG_+3A_merge.output">Merge.Output</code></td>
<td>
<p>Logical Should analysis results be merged into a single file for easy access.</p>
</td></tr>
<tr><td><code id="BIGDAWG_+3A_verbose">Verbose</code></td>
<td>
<p>Logical Should a summary of each analysis be displayed in console.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
### The following examples use the synthetic data set bundled with BIGDAWG

# Haplotype analysis with no missing genotypes for two loci sets
# Significant haplotype association with phenotype
# BIGDAWG(Data="HLA_data", Run.Tests="H", Missing=0, Loci.Set=list(c("DRB1","DQB1")))

# Hardy-Weinberg and Locus analysis ignoring missing data
# Significant locus associations with phenotype at all but DQB1
# BIGDAWG(Data="HLA_data", Run.Tests="L", Missing="ignore")

# Hardy-Weinberg analysis trimming data to 2-Field resolution with no output to files (console only)
# Significant locus deviation at DQB1
BIGDAWG(Data="HLA_data", Run.Tests="HWE", Trim=TRUE, Res=2, Output=FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='Build.Matrix'>Build Output Matrix for GL2Tab Conversion</h2><span id='topic+Build.Matrix'></span>

<h3>Description</h3>

<p>Initializes output matrix format for GL2Tab conversion
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Build.Matrix(System, Loci)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Build.Matrix_+3A_system">System</code></td>
<td>
<p>Character Genetic system HLA- or KIR</p>
</td></tr>
<tr><td><code id="Build.Matrix_+3A_loci">Loci</code></td>
<td>
<p>The loci for header names</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only.
</p>

<hr>
<h2 id='buildHAPnames'>Haplotype Name Builder</h2><span id='topic+buildHAPnames'></span>

<h3>Description</h3>

<p>Builds table of names for HAPsets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildHAPnames(Combn, loci)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buildHAPnames_+3A_combn">Combn</code></td>
<td>
<p>Combination of loci to extraction from genos</p>
</td></tr>
<tr><td><code id="buildHAPnames_+3A_loci">loci</code></td>
<td>
<p>Character vector of unique loci being analyzed.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='buildHAPsets'>Haplotype List Builder</h2><span id='topic+buildHAPsets'></span>

<h3>Description</h3>

<p>Builds table of haplotypes from combinations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildHAPsets(Combn, genos, loci, loci.ColNames)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buildHAPsets_+3A_combn">Combn</code></td>
<td>
<p>Combination of loci to extraction from genos</p>
</td></tr>
<tr><td><code id="buildHAPsets_+3A_genos">genos</code></td>
<td>
<p>The genotype columns of the loci set being analyzed.</p>
</td></tr>
<tr><td><code id="buildHAPsets_+3A_loci">loci</code></td>
<td>
<p>Character vector of unique loci being analyzed.</p>
</td></tr>
<tr><td><code id="buildHAPsets_+3A_loci.colnames">loci.ColNames</code></td>
<td>
<p>Character vector of genos column names.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='cci'>Case-Control Odds ratio calculation and graphing</h2><span id='topic+cci'></span>

<h3>Description</h3>

<p>cci function port epicalc version 2.15.1.0 (Virasakdi Chongsuvivatwong, 2012)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cci(
  caseexp,
  controlex,
  casenonex,
  controlnonex,
  cctable = NULL,
  graph = TRUE,
  design = "cohort",
  main,
  xlab,
  ylab,
  xaxis,
  yaxis,
  alpha = 0.05,
  fisher.or = FALSE,
  exact.ci.or = TRUE,
  decimal = 2
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cci_+3A_caseexp">caseexp</code></td>
<td>
<p>Number of cases exposed</p>
</td></tr>
<tr><td><code id="cci_+3A_controlex">controlex</code></td>
<td>
<p>Number of controls exposed</p>
</td></tr>
<tr><td><code id="cci_+3A_casenonex">casenonex</code></td>
<td>
<p>Number of cases not exosed</p>
</td></tr>
<tr><td><code id="cci_+3A_controlnonex">controlnonex</code></td>
<td>
<p>Number of controls not exposed</p>
</td></tr>
<tr><td><code id="cci_+3A_cctable">cctable</code></td>
<td>
<p>A 2-by-2 table. If specified, will supercede the outcome and exposure variables</p>
</td></tr>
<tr><td><code id="cci_+3A_graph">graph</code></td>
<td>
<p>If TRUE (default), produces an odds ratio plot</p>
</td></tr>
<tr><td><code id="cci_+3A_design">design</code></td>
<td>
<p>Specification for graph; can be &quot;case control&quot;,&quot;case-control&quot;, &quot;cohort&quot; or &quot;prospective&quot;</p>
</td></tr>
<tr><td><code id="cci_+3A_main">main</code></td>
<td>
<p>main title of the graph</p>
</td></tr>
<tr><td><code id="cci_+3A_xlab">xlab</code></td>
<td>
<p>label on X axis</p>
</td></tr>
<tr><td><code id="cci_+3A_ylab">ylab</code></td>
<td>
<p>label on Y axis</p>
</td></tr>
<tr><td><code id="cci_+3A_xaxis">xaxis</code></td>
<td>
<p>two categories of exposure in graph</p>
</td></tr>
<tr><td><code id="cci_+3A_yaxis">yaxis</code></td>
<td>
<p>two categories of outcome in graph</p>
</td></tr>
<tr><td><code id="cci_+3A_alpha">alpha</code></td>
<td>
<p>level of significance</p>
</td></tr>
<tr><td><code id="cci_+3A_fisher.or">fisher.or</code></td>
<td>
<p>whether odds ratio should be computed by the exact method</p>
</td></tr>
<tr><td><code id="cci_+3A_exact.ci.or">exact.ci.or</code></td>
<td>
<p>whether confidence limite of the odds ratio should be computed by the exact method</p>
</td></tr>
<tr><td><code id="cci_+3A_decimal">decimal</code></td>
<td>
<p>number of decimal places displayed</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='cci.pval'>Case Control Odds Ratio Calculation from Epicalc</h2><span id='topic+cci.pval'></span>

<h3>Description</h3>

<p>Calculates odds ratio and pvalues from 2x2 table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cci.pval(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cci.pval_+3A_x">x</code></td>
<td>
<p>List of 2x2 matrices for calculation, output of TableMaker.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='cci.pval.list'>Case Control Odds Ratio Calculation from Epicalc list variation</h2><span id='topic+cci.pval.list'></span>

<h3>Description</h3>

<p>Variation of the cci.pvalue function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cci.pval.list(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cci.pval.list_+3A_x">x</code></td>
<td>
<p>List of 2x2 matrices to apply the cci.pvalue function. List output of TableMaker.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='Check.Cores'>Check Cores Parameters</h2><span id='topic+Check.Cores'></span>

<h3>Description</h3>

<p>Check cores limitation for OS compatibility
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Check.Cores(Cores.Lim, Output)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Check.Cores_+3A_cores.lim">Cores.Lim</code></td>
<td>
<p>Integer How many cores can be used.</p>
</td></tr>
<tr><td><code id="Check.Cores_+3A_output">Output</code></td>
<td>
<p>Logical Should analysis results be written to output directory.</p>
</td></tr>
</table>

<hr>
<h2 id='Check.Data'>Check Data Structure</h2><span id='topic+Check.Data'></span>

<h3>Description</h3>

<p>Check data structure for successful conversion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Check.Data(Data, System, Convert)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Check.Data_+3A_data">Data</code></td>
<td>
<p>String Type of output.</p>
</td></tr>
<tr><td><code id="Check.Data_+3A_system">System</code></td>
<td>
<p>Character Genetic system HLA or KIR</p>
</td></tr>
<tr><td><code id="Check.Data_+3A_convert">Convert</code></td>
<td>
<p>String Direction for conversion.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only.
</p>

<hr>
<h2 id='Check.Params'>Check Input Parameters</h2><span id='topic+Check.Params'></span>

<h3>Description</h3>

<p>Check input parameters for invalid entries.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Check.Params(
  HLA,
  Loci.Set,
  Exon,
  All.Pairwise,
  Trim,
  Res,
  EVS.rm,
  Missing,
  Cores.Lim,
  Return,
  Output,
  Merge.Output,
  Verbose
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Check.Params_+3A_hla">HLA</code></td>
<td>
<p>Logical indicating whether data is HLA class I/II genotyping data only.</p>
</td></tr>
<tr><td><code id="Check.Params_+3A_loci.set">Loci.Set</code></td>
<td>
<p>Input list defining which loci to use for analyses (combinations permitted).</p>
</td></tr>
<tr><td><code id="Check.Params_+3A_exon">Exon</code></td>
<td>
<p>Numeric Exon(s) for targeted amino acid analysis.</p>
</td></tr>
<tr><td><code id="Check.Params_+3A_all.pairwise">All.Pairwise</code></td>
<td>
<p>Logical indicating whether all pairwise loci should be analyzed in haplotype analysis.</p>
</td></tr>
<tr><td><code id="Check.Params_+3A_trim">Trim</code></td>
<td>
<p>Logical indicating if HLA alleles should be trimmed to a set resolution.</p>
</td></tr>
<tr><td><code id="Check.Params_+3A_res">Res</code></td>
<td>
<p>Numeric setting what desired resolution to trim HLA alleles.</p>
</td></tr>
<tr><td><code id="Check.Params_+3A_evs.rm">EVS.rm</code></td>
<td>
<p>Logical indicating if expression variant suffixes should be removed.</p>
</td></tr>
<tr><td><code id="Check.Params_+3A_missing">Missing</code></td>
<td>
<p>Numeric setting allowable missing data for running analysis (may use &quot;ignore&quot;).</p>
</td></tr>
<tr><td><code id="Check.Params_+3A_cores.lim">Cores.Lim</code></td>
<td>
<p>Integer setting the number of cores accessible to BIGDAWG (Windows limit is 1 core).</p>
</td></tr>
<tr><td><code id="Check.Params_+3A_return">Return</code></td>
<td>
<p>Logical Should analysis results be returned as list.</p>
</td></tr>
<tr><td><code id="Check.Params_+3A_output">Output</code></td>
<td>
<p>Logical Should analysis results be written to output directory.</p>
</td></tr>
<tr><td><code id="Check.Params_+3A_merge.output">Merge.Output</code></td>
<td>
<p>Logical Should analysis results be merged into a single file for easy access.</p>
</td></tr>
<tr><td><code id="Check.Params_+3A_verbose">Verbose</code></td>
<td>
<p>Logical Should a summary of each analysis be displayed in console.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only.
</p>

<hr>
<h2 id='Check.Params.GLS'>Check Input Parameters for GLS conversion</h2><span id='topic+Check.Params.GLS'></span>

<h3>Description</h3>

<p>Check input parameters for invalid entries.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Check.Params.GLS(
  Convert,
  File.Output,
  System,
  HZY.Red,
  DRB345.Check,
  Cores.Lim
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Check.Params.GLS_+3A_convert">Convert</code></td>
<td>
<p>String Direction for conversion.</p>
</td></tr>
<tr><td><code id="Check.Params.GLS_+3A_file.output">File.Output</code></td>
<td>
<p>String Type of output.</p>
</td></tr>
<tr><td><code id="Check.Params.GLS_+3A_system">System</code></td>
<td>
<p>String Genetic system (HLA or KIR) of the data being converted</p>
</td></tr>
<tr><td><code id="Check.Params.GLS_+3A_hzy.red">HZY.Red</code></td>
<td>
<p>Logical Reduction of homozygote genotypes to single allele.</p>
</td></tr>
<tr><td><code id="Check.Params.GLS_+3A_drb345.check">DRB345.Check</code></td>
<td>
<p>Logical Check DR haplotypes for consistency and flag unusual haplotypes.</p>
</td></tr>
<tr><td><code id="Check.Params.GLS_+3A_cores.lim">Cores.Lim</code></td>
<td>
<p>Integer How many cores can be used.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only.
</p>

<hr>
<h2 id='CheckAlleles'>HLA Allele Legitimacy Check for Amino Acid Analysis</h2><span id='topic+CheckAlleles'></span>

<h3>Description</h3>

<p>Checks available alleles against data to ensure complete overlap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckAlleles(x, y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CheckAlleles_+3A_x">x</code></td>
<td>
<p>Exon protein list alignment object.</p>
</td></tr>
<tr><td><code id="CheckAlleles_+3A_y">y</code></td>
<td>
<p>Genotypes from data file</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='CheckHLA'>HLA Formatting Check for Amino Acid Analysis</h2><span id='topic+CheckHLA'></span>

<h3>Description</h3>

<p>Checks data to see if HLA data is properly formatted .
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckHLA(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CheckHLA_+3A_x">x</code></td>
<td>
<p>All columns of HLA genotyping data.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='CheckLoci'>HLA Loci Legitimacy Check for Amino Acid Analysis</h2><span id='topic+CheckLoci'></span>

<h3>Description</h3>

<p>Checks available loci against data to ensure complete overlap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckLoci(x, y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CheckLoci_+3A_x">x</code></td>
<td>
<p>Loci available in exon protein list alignment object.</p>
</td></tr>
<tr><td><code id="CheckLoci_+3A_y">y</code></td>
<td>
<p>Unique column names</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='CheckRelease'>Function to Check Release Versions</h2><span id='topic+CheckRelease'></span>

<h3>Description</h3>

<p>This updates the protein aligment used in checking HLA loci and alleles as well as in the amino acid analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckRelease(Package = T, Alignment = T, Output = F)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CheckRelease_+3A_package">Package</code></td>
<td>
<p>Logical to check for BIGDAWG package versions</p>
</td></tr>
<tr><td><code id="CheckRelease_+3A_alignment">Alignment</code></td>
<td>
<p>Logical to check the IMGT/HLA database version for the alignment bundled with BIGDAWG.</p>
</td></tr>
<tr><td><code id="CheckRelease_+3A_output">Output</code></td>
<td>
<p>Should any error be written to a file</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Requires active internet connection.
</p>

<hr>
<h2 id='CheckString.Allele'>GL String Allele Check</h2><span id='topic+CheckString.Allele'></span>

<h3>Description</h3>

<p>GL String check for allele ambiguity formatting
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckString.Allele(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CheckString.Allele_+3A_x">x</code></td>
<td>
<p>GL String to check against</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only.
</p>

<hr>
<h2 id='CheckString.Locus'>GL String Locus Check</h2><span id='topic+CheckString.Locus'></span>

<h3>Description</h3>

<p>Check GL string for loci appearing in multiple gene fields.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckString.Locus(x, Loci)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CheckString.Locus_+3A_x">x</code></td>
<td>
<p>GL String to check against</p>
</td></tr>
<tr><td><code id="CheckString.Locus_+3A_loci">Loci</code></td>
<td>
<p>Loci to check</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only.
</p>

<hr>
<h2 id='Condense.EPL'>Condensing Exon Specific Alignments to Single Dataframe</h2><span id='topic+Condense.EPL'></span>

<h3>Description</h3>

<p>Combines multiple Exon Specific Alignments into a single Alignment object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Condense.EPL(EPL.Exon)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Condense.EPL_+3A_epl.exon">EPL.Exon</code></td>
<td>
<p>Exon-Locus Specific Amino Acid Alignment.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='Create.Null.Table'>Create Empty Table</h2><span id='topic+Create.Null.Table'></span>

<h3>Description</h3>

<p>Creates matrix of NA for no result tables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Create.Null.Table(Locus, Names, nr)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Create.Null.Table_+3A_locus">Locus</code></td>
<td>
<p>Locus being analyzed.</p>
</td></tr>
<tr><td><code id="Create.Null.Table_+3A_names">Names</code></td>
<td>
<p>Column names for final matrix.</p>
</td></tr>
<tr><td><code id="Create.Null.Table_+3A_nr">nr</code></td>
<td>
<p>Number of rows.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='DRB345.Check.Wrapper'>DRB345 haplotype zygosity wrapper</h2><span id='topic+DRB345.Check.Wrapper'></span>

<h3>Description</h3>

<p>Checks DR haplotypes for correct zygosity and flags unanticipated haplotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DRB345.Check.Wrapper(Genotype, Loci.DR)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DRB345.Check.Wrapper_+3A_genotype">Genotype</code></td>
<td>
<p>Row of data set data frame following DRB345 parsing</p>
</td></tr>
<tr><td><code id="DRB345.Check.Wrapper_+3A_loci.dr">Loci.DR</code></td>
<td>
<p>DRBx Loci of interest to test for consistency</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only.
</p>

<hr>
<h2 id='DRB345.Check.Zygosity'>DRB345 haplotype zygosity checker single locus</h2><span id='topic+DRB345.Check.Zygosity'></span>

<h3>Description</h3>

<p>Checks DR haplotypes for correct zygosity and flags unanticipated haplotypes for a single DRBx
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DRB345.Check.Zygosity(Locus, Genotype)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DRB345.Check.Zygosity_+3A_locus">Locus</code></td>
<td>
<p>Locus of interest to test for consistency</p>
</td></tr>
<tr><td><code id="DRB345.Check.Zygosity_+3A_genotype">Genotype</code></td>
<td>
<p>Row of data set data frame following DRB345 parsing</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only.
</p>

<hr>
<h2 id='DRB345.Exp'>DRB345 Expected</h2><span id='topic+DRB345.Exp'></span>

<h3>Description</h3>

<p>Checks DRB1 Genotype and Returns Expected DR345 Loci
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DRB345.Exp(DRB1.Genotype)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DRB345.Exp_+3A_drb1.genotype">DRB1.Genotype</code></td>
<td>
<p>DRB1 Subject Genotypes</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only.
</p>

<hr>
<h2 id='DRB345.parser'>DRB345 Column Processing</h2><span id='topic+DRB345.parser'></span>

<h3>Description</h3>

<p>Separates DRB345 column pair into separate columns for each locus
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DRB345.parser(Tab)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DRB345.parser_+3A_tab">Tab</code></td>
<td>
<p>Data frame of sampleIDs, phenotypes, and genotypes</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='Err.Log'>Error Code Display and Logging</h2><span id='topic+Err.Log'></span>

<h3>Description</h3>

<p>Displays error codes attributable to data formatting and Locus/Allele naming. Writes to log file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Err.Log(Output, x, y = NULL, z = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Err.Log_+3A_output">Output</code></td>
<td>
<p>Logical indicating if Error logging should be written to a file.</p>
</td></tr>
<tr><td><code id="Err.Log_+3A_x">x</code></td>
<td>
<p>Log Code.</p>
</td></tr>
<tr><td><code id="Err.Log_+3A_y">y</code></td>
<td>
<p>Misc information relevant to error.</p>
</td></tr>
<tr><td><code id="Err.Log_+3A_z">z</code></td>
<td>
<p>Misc information relevant to error.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='EVSremoval'>Expression Variant Suffix Removal</h2><span id='topic+EVSremoval'></span>

<h3>Description</h3>

<p>Removes expression variant suffixes from HLA alleles in the exon protein alignment object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EVSremoval(Locus, EPList)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="EVSremoval_+3A_locus">Locus</code></td>
<td>
<p>Locus to be filtered against.</p>
</td></tr>
<tr><td><code id="EVSremoval_+3A_eplist">EPList</code></td>
<td>
<p>Exon Protein Alignment Object</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='Exon.Filter'>Filter Exon Specific Alignment Sections</h2><span id='topic+Exon.Filter'></span>

<h3>Description</h3>

<p>Filters the ExonPtnAlign object by locus and exon.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Exon.Filter(Locus, Exon, EPL.Locus, RefExons, E.Ptn.Starts)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Exon.Filter_+3A_locus">Locus</code></td>
<td>
<p>Locus being analyzed.</p>
</td></tr>
<tr><td><code id="Exon.Filter_+3A_exon">Exon</code></td>
<td>
<p>Exon being analyzed.</p>
</td></tr>
<tr><td><code id="Exon.Filter_+3A_epl.locus">EPL.Locus</code></td>
<td>
<p>ExonPtnAlign object filtered by Locus</p>
</td></tr>
<tr><td><code id="Exon.Filter_+3A_refexons">RefExons</code></td>
<td>
<p>Reference Exon Table</p>
</td></tr>
<tr><td><code id="Exon.Filter_+3A_e.ptn.starts">E.Ptn.Starts</code></td>
<td>
<p>Exon Protein Overlay Map</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='ExonPtnAlign.Create'>Protein Exon Alignment Formatter</h2><span id='topic+ExonPtnAlign.Create'></span>

<h3>Description</h3>

<p>Dynamically creates an alignmnet of Allele exons for Analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExonPtnAlign.Create(Locus, RefTab)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ExonPtnAlign.Create_+3A_locus">Locus</code></td>
<td>
<p>Locus alignment to be formatted.</p>
</td></tr>
<tr><td><code id="ExonPtnAlign.Create_+3A_reftab">RefTab</code></td>
<td>
<p>Reference exon protein information for alignment formatting.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='ExonPtnList'>Exon protein alignments.</h2><span id='topic+ExonPtnList'></span>

<h3>Description</h3>

<p>Alignment object for use in the amino acid analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExonPtnList
</code></pre>


<h3>Format</h3>

<p>A list where each element is an alignment dataframe for a single locus.
</p>

<hr>
<h2 id='Filler'>Replace or Fill 00:00 allele strings</h2><span id='topic+Filler'></span>

<h3>Description</h3>

<p>Replaces or Fills absent allele strings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Filler(x, Locus = NULL, Type)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Filler_+3A_x">x</code></td>
<td>
<p>Genotype</p>
</td></tr>
<tr><td><code id="Filler_+3A_locus">Locus</code></td>
<td>
<p>Locus column to adjust.</p>
</td></tr>
<tr><td><code id="Filler_+3A_type">Type</code></td>
<td>
<p>String specifying whether to pad ('Fill') or leave blank ('Remove') absent calls</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only.
</p>

<hr>
<h2 id='Format.Allele'>Ambiguous Alleles Locus Name Formatting</h2><span id='topic+Format.Allele'></span>

<h3>Description</h3>

<p>Remove or Append Locus name from/to allele in an ambiguous allele string
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Format.Allele(x, Type)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Format.Allele_+3A_x">x</code></td>
<td>
<p>Allele String</p>
</td></tr>
<tr><td><code id="Format.Allele_+3A_type">Type</code></td>
<td>
<p>String specifying whether to strip ('off') or append ('on') locus prefix</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only.
</p>

<hr>
<h2 id='Format.Tab'>Tabular Data Locus Format Tool</h2><span id='topic+Format.Tab'></span>

<h3>Description</h3>

<p>Correctly orders the expanded GL string
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Format.Tab(x, Order)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Format.Tab_+3A_x">x</code></td>
<td>
<p>Single row of converted GL string</p>
</td></tr>
<tr><td><code id="Format.Tab_+3A_order">Order</code></td>
<td>
<p>Single row data frame for mapping converted GL strings</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only.
</p>

<hr>
<h2 id='getAllele.Count'>Recompute number of alleles</h2><span id='topic+getAllele.Count'></span>

<h3>Description</h3>

<p>Using Freq.Final, recompute number of alleles
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAllele.Count(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getAllele.Count_+3A_x">x</code></td>
<td>
<p>Locus specific contingency matrix getCS.Mat output.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='getCS.Mat'>Chi square matrices</h2><span id='topic+getCS.Mat'></span>

<h3>Description</h3>

<p>Chi Square contingency matrix builder with rare cell binning
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCS.Mat(Locus, genos.sub, Allele.Freq, Allele.Combn)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getCS.Mat_+3A_locus">Locus</code></td>
<td>
<p>Locus of interest.</p>
</td></tr>
<tr><td><code id="getCS.Mat_+3A_genos.sub">genos.sub</code></td>
<td>
<p>Genotypes for locus of interest.</p>
</td></tr>
<tr><td><code id="getCS.Mat_+3A_allele.freq">Allele.Freq</code></td>
<td>
<p>Allele frequencies.</p>
</td></tr>
<tr><td><code id="getCS.Mat_+3A_allele.combn">Allele.Combn</code></td>
<td>
<p>Allele combinations.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='getCS.stat'>Chi square test statistic</h2><span id='topic+getCS.stat'></span>

<h3>Description</h3>

<p>Calculate chi square test statistic
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCS.stat(Locus, Freq.Final)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getCS.stat_+3A_locus">Locus</code></td>
<td>
<p>Locus of interest.</p>
</td></tr>
<tr><td><code id="getCS.stat_+3A_freq.final">Freq.Final</code></td>
<td>
<p>Contingency Matrix getCS.Mat output.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='GetField'>HLA trimming function</h2><span id='topic+GetField'></span>

<h3>Description</h3>

<p>Trim a properly formatted HLA allele to desired number of fields.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetField(x, Res)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetField_+3A_x">x</code></td>
<td>
<p>HLA allele.</p>
</td></tr>
<tr><td><code id="GetField_+3A_res">Res</code></td>
<td>
<p>Resolution desired.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='getFileName'>File Name Extraction</h2><span id='topic+getFileName'></span>

<h3>Description</h3>

<p>Function to extract file path.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getFileName(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getFileName_+3A_x">x</code></td>
<td>
<p>File name.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only.
</p>

<hr>
<h2 id='GetFiles'>File Fetcher</h2><span id='topic+GetFiles'></span>

<h3>Description</h3>

<p>Download Protein Alignment and Accessory Files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetFiles(Loci)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetFiles_+3A_loci">Loci</code></td>
<td>
<p>HLA Loci to be fetched. Limited Loci available.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='getHap'>Haplotype Table Maker</h2><span id='topic+getHap'></span>

<h3>Description</h3>

<p>Builds table of haplotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getHap(SID, HaploEM)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getHap_+3A_sid">SID</code></td>
<td>
<p>Index number (i.e., row number) of sample ID from genotype matrix.</p>
</td></tr>
<tr><td><code id="getHap_+3A_haploem">HaploEM</code></td>
<td>
<p>Haplotype output object from haplo.stat::haplo.em function.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='getObsFreq'>Observed Frequency</h2><span id='topic+getObsFreq'></span>

<h3>Description</h3>

<p>Get observed frequency of genotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getObsFreq(x, genos.locus)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getObsFreq_+3A_x">x</code></td>
<td>
<p>Single genotype.</p>
</td></tr>
<tr><td><code id="getObsFreq_+3A_genos.locus">genos.locus</code></td>
<td>
<p>Locus genotypes.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='GL2Tab.Loci'>Locus Ordering for GL2Tab</h2><span id='topic+GL2Tab.Loci'></span>

<h3>Description</h3>

<p>Orders Locus Calls
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GL2Tab.Loci(Locus, Genotype, System)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GL2Tab.Loci_+3A_locus">Locus</code></td>
<td>
<p>Locus to condense</p>
</td></tr>
<tr><td><code id="GL2Tab.Loci_+3A_genotype">Genotype</code></td>
<td>
<p>Row of loci to condense</p>
</td></tr>
<tr><td><code id="GL2Tab.Loci_+3A_system">System</code></td>
<td>
<p>Character Genetic system HLA or KIR</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only.
</p>

<hr>
<h2 id='GL2Tab.Sub'>Genotype List String Expander</h2><span id='topic+GL2Tab.Sub'></span>

<h3>Description</h3>

<p>Expands GL string into a table of adjacent loci
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GL2Tab.Sub(x, System)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GL2Tab.Sub_+3A_x">x</code></td>
<td>
<p>Character GL string to expand</p>
</td></tr>
<tr><td><code id="GL2Tab.Sub_+3A_system">System</code></td>
<td>
<p>Character Genetic system HLA or KIR</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only.
</p>

<hr>
<h2 id='GL2Tab.wrapper'>Genotype List String to Tabular Data Conversion</h2><span id='topic+GL2Tab.wrapper'></span>

<h3>Description</h3>

<p>Expands GL strings to columns of adjacent locus pairs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GL2Tab.wrapper(df, System, Strip.Prefix, Abs.Fill, Cores)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GL2Tab.wrapper_+3A_df">df</code></td>
<td>
<p>Data frame containing GL strings</p>
</td></tr>
<tr><td><code id="GL2Tab.wrapper_+3A_system">System</code></td>
<td>
<p>Character Genetic system HLA or KIR</p>
</td></tr>
<tr><td><code id="GL2Tab.wrapper_+3A_strip.prefix">Strip.Prefix</code></td>
<td>
<p>Logical Should System/Locus prefixes be stripped from table data.</p>
</td></tr>
<tr><td><code id="GL2Tab.wrapper_+3A_abs.fill">Abs.Fill</code></td>
<td>
<p>Logical Should absent loci special designations be used.</p>
</td></tr>
<tr><td><code id="GL2Tab.wrapper_+3A_cores">Cores</code></td>
<td>
<p>Integer How many cores can be used</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only
</p>

<hr>
<h2 id='GLSconvert'>Genotype List String Conversion</h2><span id='topic+GLSconvert'></span>

<h3>Description</h3>

<p>Main Workhorse wrapper for cross converting columnar table to GL string representaion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GLSconvert(
  Data,
  Convert,
  File.Output = "txt",
  System = "HLA",
  HZY.Red = FALSE,
  DRB345.Check = FALSE,
  Strip.Prefix = TRUE,
  Abs.Fill = FALSE,
  Cores.Lim = 1L
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GLSconvert_+3A_data">Data</code></td>
<td>
<p>String File name or R Data Frame.</p>
</td></tr>
<tr><td><code id="GLSconvert_+3A_convert">Convert</code></td>
<td>
<p>String Direction for conversion.</p>
</td></tr>
<tr><td><code id="GLSconvert_+3A_file.output">File.Output</code></td>
<td>
<p>String Type of File.Output.</p>
</td></tr>
<tr><td><code id="GLSconvert_+3A_system">System</code></td>
<td>
<p>String Genetic system (HLA or KIR) of the data being converted</p>
</td></tr>
<tr><td><code id="GLSconvert_+3A_hzy.red">HZY.Red</code></td>
<td>
<p>Logical Reduction of homozygote genotypes to single allele.</p>
</td></tr>
<tr><td><code id="GLSconvert_+3A_drb345.check">DRB345.Check</code></td>
<td>
<p>Logical Check DR haplotypes for consistency and flag unusual haplotypes.</p>
</td></tr>
<tr><td><code id="GLSconvert_+3A_strip.prefix">Strip.Prefix</code></td>
<td>
<p>Logical Should System/Locus prefixes be stripped from table data.</p>
</td></tr>
<tr><td><code id="GLSconvert_+3A_abs.fill">Abs.Fill</code></td>
<td>
<p>Logical Should absent loci special designations be used.</p>
</td></tr>
<tr><td><code id="GLSconvert_+3A_cores.lim">Cores.Lim</code></td>
<td>
<p>Integer How many cores can be used.</p>
</td></tr>
</table>

<hr>
<h2 id='H.MC'>Haplotype Analysis Function for Multicore</h2><span id='topic+H.MC'></span>

<h3>Description</h3>

<p>This is the workhorse function for the haplotype analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>H.MC(genos.sub, grp, Strict.Bin, Verbose)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="H.MC_+3A_genos.sub">genos.sub</code></td>
<td>
<p>The genotype columns of the loci(locus) set being analyzed.</p>
</td></tr>
<tr><td><code id="H.MC_+3A_grp">grp</code></td>
<td>
<p>Case/Control or Phenotype groupings.</p>
</td></tr>
<tr><td><code id="H.MC_+3A_strict.bin">Strict.Bin</code></td>
<td>
<p>Logical specify if strict rare cell binning should be used in ChiSq test</p>
</td></tr>
<tr><td><code id="H.MC_+3A_verbose">Verbose</code></td>
<td>
<p>Summary display carryover from main BIGDAWG function</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='H.MC.wrapper'>Haplotype Wrapper for Multicore</h2><span id='topic+H.MC.wrapper'></span>

<h3>Description</h3>

<p>Wrapper for main H function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>H.MC.wrapper(
  SID,
  Tabsub,
  loci,
  loci.ColNames,
  genos,
  grp,
  All.Pairwise,
  Strict.Bin,
  Output,
  Verbose,
  Cores
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="H.MC.wrapper_+3A_sid">SID</code></td>
<td>
<p>Character vector of subject IDs.</p>
</td></tr>
<tr><td><code id="H.MC.wrapper_+3A_tabsub">Tabsub</code></td>
<td>
<p>Data frame of genotype calls for set being analyzed.</p>
</td></tr>
<tr><td><code id="H.MC.wrapper_+3A_loci">loci</code></td>
<td>
<p>Character vector of unique loci being analyzed.</p>
</td></tr>
<tr><td><code id="H.MC.wrapper_+3A_loci.colnames">loci.ColNames</code></td>
<td>
<p>Character vector of genos column names.</p>
</td></tr>
<tr><td><code id="H.MC.wrapper_+3A_genos">genos</code></td>
<td>
<p>The genotype columns of the loci set being analyzed.</p>
</td></tr>
<tr><td><code id="H.MC.wrapper_+3A_grp">grp</code></td>
<td>
<p>Case/Control or Phenotype groupings.</p>
</td></tr>
<tr><td><code id="H.MC.wrapper_+3A_all.pairwise">All.Pairwise</code></td>
<td>
<p>Haplotype argument carryover from main BIGDAWG function</p>
</td></tr>
<tr><td><code id="H.MC.wrapper_+3A_strict.bin">Strict.Bin</code></td>
<td>
<p>Logical specify if strict rare cell binning should be used in ChiSq test</p>
</td></tr>
<tr><td><code id="H.MC.wrapper_+3A_output">Output</code></td>
<td>
<p>Data return carryover from main BIGDAWG function</p>
</td></tr>
<tr><td><code id="H.MC.wrapper_+3A_verbose">Verbose</code></td>
<td>
<p>Summary display carryover from main BIGDAWG function</p>
</td></tr>
<tr><td><code id="H.MC.wrapper_+3A_cores">Cores</code></td>
<td>
<p>Cores carryover from main BIGDAWG function</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='HLA_data'>Example HLA Dataset</h2><span id='topic+HLA_data'></span>

<h3>Description</h3>

<p>A synthetic dataset of HLA genotypes for using bigdawg.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HLA_data
</code></pre>


<h3>Format</h3>

<p>A data frame with 2000 rows and 14 variables
</p>

<hr>
<h2 id='HWE'>Hardy Weinbergy Equilibrium Function</h2><span id='topic+HWE'></span>

<h3>Description</h3>

<p>This is the main function for the HWE analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWE(Tab)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWE_+3A_tab">Tab</code></td>
<td>
<p>data frame of genotype files post processing.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='HWE.ChiSq'>Hardy Weinbergy Equilibrium Function</h2><span id='topic+HWE.ChiSq'></span>

<h3>Description</h3>

<p>This is the workhorse function for each group analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWE.ChiSq(genos.sub, loci, nloci)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWE.ChiSq_+3A_genos.sub">genos.sub</code></td>
<td>
<p>data frame of genotype files post processing.</p>
</td></tr>
<tr><td><code id="HWE.ChiSq_+3A_loci">loci</code></td>
<td>
<p>list of loci.</p>
</td></tr>
<tr><td><code id="HWE.ChiSq_+3A_nloci">nloci</code></td>
<td>
<p>number of loci in list</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='HWE.wrapper'>Hardy-Weinbery Wrapper</h2><span id='topic+HWE.wrapper'></span>

<h3>Description</h3>

<p>Wrapper for main HWE function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWE.wrapper(Tab, Output, Verbose)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWE.wrapper_+3A_tab">Tab</code></td>
<td>
<p>Data frame of genotype files post processing.</p>
</td></tr>
<tr><td><code id="HWE.wrapper_+3A_output">Output</code></td>
<td>
<p>Data return carryover from main BIGDAWG function</p>
</td></tr>
<tr><td><code id="HWE.wrapper_+3A_verbose">Verbose</code></td>
<td>
<p>Summary display carryover from main BIGDAWG function</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='L'>Locus Analysis Function</h2><span id='topic+L'></span>

<h3>Description</h3>

<p>This is the workhorse function for the locus level analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>L(loci.ColNames, Locus, genos, grp, Strict.Bin)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="L_+3A_loci.colnames">loci.ColNames</code></td>
<td>
<p>The column names of the loci being analyzed.</p>
</td></tr>
<tr><td><code id="L_+3A_locus">Locus</code></td>
<td>
<p>Locus being analyzed.</p>
</td></tr>
<tr><td><code id="L_+3A_genos">genos</code></td>
<td>
<p>Genotype table</p>
</td></tr>
<tr><td><code id="L_+3A_grp">grp</code></td>
<td>
<p>Case/Control or Phenotype groupings.</p>
</td></tr>
<tr><td><code id="L_+3A_strict.bin">Strict.Bin</code></td>
<td>
<p>Logical specify if strict rare cell binning should be used in ChiSq test</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='L.wrapper'>Locus Wrapper</h2><span id='topic+L.wrapper'></span>

<h3>Description</h3>

<p>Wrapper for main L function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>L.wrapper(nloci, loci, loci.ColNames, genos, grp, Strict.Bin, Output, Verbose)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="L.wrapper_+3A_nloci">nloci</code></td>
<td>
<p>Number of loci being analyzed.</p>
</td></tr>
<tr><td><code id="L.wrapper_+3A_loci">loci</code></td>
<td>
<p>Loci being analyzed.</p>
</td></tr>
<tr><td><code id="L.wrapper_+3A_loci.colnames">loci.ColNames</code></td>
<td>
<p>The column names of the loci being analyzed.</p>
</td></tr>
<tr><td><code id="L.wrapper_+3A_genos">genos</code></td>
<td>
<p>Genotype table</p>
</td></tr>
<tr><td><code id="L.wrapper_+3A_grp">grp</code></td>
<td>
<p>Case/Control or Phenotype groupings.</p>
</td></tr>
<tr><td><code id="L.wrapper_+3A_strict.bin">Strict.Bin</code></td>
<td>
<p>Logical specify if strict rare cell binning should be used in ChiSq test</p>
</td></tr>
<tr><td><code id="L.wrapper_+3A_output">Output</code></td>
<td>
<p>Data return carryover from main BIGDAWG function</p>
</td></tr>
<tr><td><code id="L.wrapper_+3A_verbose">Verbose</code></td>
<td>
<p>Summary display carryover from main BIGDAWG function</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='make2x2'>Creation of a 2x2 table using the indicated orientation.</h2><span id='topic+make2x2'></span>

<h3>Description</h3>

<p>make2x2 function port epicalc version 2.15.1.0 (Virasakdi Chongsuvivatwong, 2012)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make2x2(caseexp, controlex, casenonex, controlnonex)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="make2x2_+3A_caseexp">caseexp</code></td>
<td>
<p>Number of cases exposed</p>
</td></tr>
<tr><td><code id="make2x2_+3A_controlex">controlex</code></td>
<td>
<p>Number of controls exposed</p>
</td></tr>
<tr><td><code id="make2x2_+3A_casenonex">casenonex</code></td>
<td>
<p>Number of cases not exosed</p>
</td></tr>
<tr><td><code id="make2x2_+3A_controlnonex">controlnonex</code></td>
<td>
<p>Number of controls not exposed</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='makeComb'>Genotype Combination Maker</h2><span id='topic+makeComb'></span>

<h3>Description</h3>

<p>Make data frame of possible genotype combinations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makeComb(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="makeComb_+3A_x">x</code></td>
<td>
<p>Number of alleles.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='MergeData_Output'>Data Object Merge and Output</h2><span id='topic+MergeData_Output'></span>

<h3>Description</h3>

<p>Whole data set table construction of per haplotype for odds ratio, confidence intervals, and pvalues
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MergeData_Output(BD.out, Run, OutDir)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MergeData_Output_+3A_bd.out">BD.out</code></td>
<td>
<p>Output of analysis as list.</p>
</td></tr>
<tr><td><code id="MergeData_Output_+3A_run">Run</code></td>
<td>
<p>Tests that are to be run as defined by Run.Tests.</p>
</td></tr>
<tr><td><code id="MergeData_Output_+3A_outdir">OutDir</code></td>
<td>
<p>Output directory defined by Results.Dir or default.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='PgrpExtract'>HLA P group Finder</h2><span id='topic+PgrpExtract'></span>

<h3>Description</h3>

<p>Identify P group for a given allele if exists.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PgrpExtract(x, y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PgrpExtract_+3A_x">x</code></td>
<td>
<p>Allele of interest.</p>
</td></tr>
<tr><td><code id="PgrpExtract_+3A_y">y</code></td>
<td>
<p>Formatted P groups.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='PgrpFormat'>HLA P group File Formatter</h2><span id='topic+PgrpFormat'></span>

<h3>Description</h3>

<p>Format the hla_nom_p.txt read table object for a specific locus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PgrpFormat(x, Locus)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PgrpFormat_+3A_x">x</code></td>
<td>
<p>P group object from read.table command.</p>
</td></tr>
<tr><td><code id="PgrpFormat_+3A_locus">Locus</code></td>
<td>
<p>Locus to be filtered on.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='PreCheck'>Data Summary Function</h2><span id='topic+PreCheck'></span>

<h3>Description</h3>

<p>Summary function for sample population within data file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PreCheck(Tab, All.ColNames, rescall, HLA, Verbose, Output)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PreCheck_+3A_tab">Tab</code></td>
<td>
<p>Loci available in exon protein list alignment object.</p>
</td></tr>
<tr><td><code id="PreCheck_+3A_all.colnames">All.ColNames</code></td>
<td>
<p>Column names from genotype data.</p>
</td></tr>
<tr><td><code id="PreCheck_+3A_rescall">rescall</code></td>
<td>
<p>HLA resolution set for analysis.</p>
</td></tr>
<tr><td><code id="PreCheck_+3A_hla">HLA</code></td>
<td>
<p>HLA BIGDAWG argument passed to function</p>
</td></tr>
<tr><td><code id="PreCheck_+3A_verbose">Verbose</code></td>
<td>
<p>Summary display carryover from BIGDAWG function.</p>
</td></tr>
<tr><td><code id="PreCheck_+3A_output">Output</code></td>
<td>
<p>Data output carryover form BIGDAWG function</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='prepData'>Prepare imported data</h2><span id='topic+prepData'></span>

<h3>Description</h3>

<p>Prepare imported data for processing, checks, and analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepData(Tab)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prepData_+3A_tab">Tab</code></td>
<td>
<p>Genotypes dataframe.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='rmABstrings'>Replace absent allele strings</h2><span id='topic+rmABstrings'></span>

<h3>Description</h3>

<p>Replaces allowable absent allele strings with ^ symbol.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rmABstrings(df)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rmABstrings_+3A_df">df</code></td>
<td>
<p>Genotypes dataframe.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='RunChiSq'>Strict Chi-squared Contingency Table Test</h2><span id='topic+RunChiSq'></span>

<h3>Description</h3>

<p>Calculates chi-squared contingency table tests and bins all rare cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RunChiSq(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RunChiSq_+3A_x">x</code></td>
<td>
<p>Contingency table.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='RunChiSq_c'>Contextual Binning Chi-squared Contingency Table Test</h2><span id='topic+RunChiSq_c'></span>

<h3>Description</h3>

<p>Calculates chi-squared contingency table tests and bins rare cells at 20
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RunChiSq_c(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RunChiSq_c_+3A_x">x</code></td>
<td>
<p>Contingency table.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='Stripper'>Removes System and Locus from Alleles</h2><span id='topic+Stripper'></span>

<h3>Description</h3>

<p>Removes the System and Locus designations for alleles calls in GL2Tab
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Stripper(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Stripper_+3A_x">x</code></td>
<td>
<p>Allele</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only.
</p>

<hr>
<h2 id='summaryGeno.2'>Haplotype missing Allele summary function</h2><span id='topic+summaryGeno.2'></span>

<h3>Description</h3>

<p>Summary function for identifying missing alleles in a matrix of genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summaryGeno.2(geno, miss.val = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summaryGeno.2_+3A_geno">geno</code></td>
<td>
<p>Matrix of genotypes.</p>
</td></tr>
<tr><td><code id="summaryGeno.2_+3A_miss.val">miss.val</code></td>
<td>
<p>Vector of codes for allele missing values.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only and is ported from haplo.stats.
</p>

<hr>
<h2 id='Tab2GL.Loci'>Locus Condenser for Tab2GL</h2><span id='topic+Tab2GL.Loci'></span>

<h3>Description</h3>

<p>Condenses alleles calls of a single locus string using &quot;+&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Tab2GL.Loci(Locus, Genotype, System, HZY.Red)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Tab2GL.Loci_+3A_locus">Locus</code></td>
<td>
<p>Locus to condense</p>
</td></tr>
<tr><td><code id="Tab2GL.Loci_+3A_genotype">Genotype</code></td>
<td>
<p>Row of loci to condense</p>
</td></tr>
<tr><td><code id="Tab2GL.Loci_+3A_system">System</code></td>
<td>
<p>Character Genetic system HLA or KIR</p>
</td></tr>
<tr><td><code id="Tab2GL.Loci_+3A_hzy.red">HZY.Red</code></td>
<td>
<p>Logical Should homozygote genotypes be a single allele for non-DRB345.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only.
</p>

<hr>
<h2 id='Tab2GL.Sub'>Genotype List String Condenser</h2><span id='topic+Tab2GL.Sub'></span>

<h3>Description</h3>

<p>Condenses column of loci into a GL string using &quot;^&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Tab2GL.Sub(x, System, HZY.Red)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Tab2GL.Sub_+3A_x">x</code></td>
<td>
<p>Row of loci to condense</p>
</td></tr>
<tr><td><code id="Tab2GL.Sub_+3A_system">System</code></td>
<td>
<p>Character Genetic system HLA or KIR</p>
</td></tr>
<tr><td><code id="Tab2GL.Sub_+3A_hzy.red">HZY.Red</code></td>
<td>
<p>Logical Should homozygote genotypes be a single allele for non-DRB345.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only.
</p>

<hr>
<h2 id='Tab2GL.wrapper'>Genotype List String to Tabular Data Conversion</h2><span id='topic+Tab2GL.wrapper'></span>

<h3>Description</h3>

<p>Expands GL strings to columns of adjacent locus pairs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Tab2GL.wrapper(df, System, HZY.Red, Abs.Fill, Cores)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Tab2GL.wrapper_+3A_df">df</code></td>
<td>
<p>Data frame containing GL strings</p>
</td></tr>
<tr><td><code id="Tab2GL.wrapper_+3A_system">System</code></td>
<td>
<p>Character Genetic system HLA or KIR</p>
</td></tr>
<tr><td><code id="Tab2GL.wrapper_+3A_hzy.red">HZY.Red</code></td>
<td>
<p>Logical Should homozygote genotypes be a single allele for non-DRB345.</p>
</td></tr>
<tr><td><code id="Tab2GL.wrapper_+3A_abs.fill">Abs.Fill</code></td>
<td>
<p>Logical Should absent loci special designations be used</p>
</td></tr>
<tr><td><code id="Tab2GL.wrapper_+3A_cores">Cores</code></td>
<td>
<p>Integer How many cores can be used.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal use only.
</p>

<hr>
<h2 id='TableMaker'>Table Maker</h2><span id='topic+TableMaker'></span>

<h3>Description</h3>

<p>Table construction of per haplotype for odds ratio, confidence intervals, and pvalues
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TableMaker(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="TableMaker_+3A_x">x</code></td>
<td>
<p>Contingency table with binned rare cells.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>

<hr>
<h2 id='UpdateRelease'>Update function for protein aligment upon new IMGT HLA data release</h2><span id='topic+UpdateRelease'></span>

<h3>Description</h3>

<p>This updates the protein aligment used in checking HLA loci and alleles as well as in the amino acid analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UpdateRelease(Force = F, Restore = F, Output = F)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="UpdateRelease_+3A_force">Force</code></td>
<td>
<p>Logical specifiying if update should be forced.</p>
</td></tr>
<tr><td><code id="UpdateRelease_+3A_restore">Restore</code></td>
<td>
<p>Logical specifying if the original alignment file be restored.</p>
</td></tr>
<tr><td><code id="UpdateRelease_+3A_output">Output</code></td>
<td>
<p>Logical indicating if error reporting should be written to file.</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
