<!DOCTYPE html><html><head><title>Help for package BinaryDosage</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BinaryDosage}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bdapply'><p>Apply a function to each SNP in a</p>
binary dosage file</a></li>
<li><a href='#bdmerge'><p>Merge binary dosage files together</p></a></li>
<li><a href='#genapply'><p>Apply a function to each SNP in a</p>
gen, impute2, file</a></li>
<li><a href='#gentobd'><p>Convert a gen file to a binary dosage file</p></a></li>
<li><a href='#getaaf'><p>Calculate alternate allele frequency</p></a></li>
<li><a href='#getbdinfo'><p>Get information about a binary dosage file</p></a></li>
<li><a href='#getgeninfo'><p>Get information about a gen, impute2, file</p></a></li>
<li><a href='#getmaf'><p>Calculate minor allele frequency</p></a></li>
<li><a href='#getrsq'><p>Calculate imputation r squared</p></a></li>
<li><a href='#getsnp'><p>Read SNP data from a binary dosage file</p></a></li>
<li><a href='#getvcfinfo'><p>Get information about a vcf file</p></a></li>
<li><a href='#vcfapply'><p>Apply a function to each SNP in a vcf file</p></a></li>
<li><a href='#vcftobd'><p>Convert a VCF file to a binary dosage file</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Creates, Merges, and Reads Binary Dosage Files</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Tools to create binary dosage from either VCF
    or GEN files, merge binary dosage files, and read
    binary dosage files.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 2.1.0), covr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.0.2</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, digest, prodlim</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-01-09 23:00:33 UTC; jmorr</td>
</tr>
<tr>
<td>Author:</td>
<td>John Morrison [aut, cre],
  NIEHS [fnd] (P01 CA196559),
  NIEHS [fnd] (R01 CA201407),
  NIEHS [fnd] (P30 ES007048),
  NIEHS [fnd] (P01 HL115606)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>John Morrison &lt;jmorr@usc.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-01-13 16:20:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='bdapply'>Apply a function to each SNP in a
binary dosage file</h2><span id='topic+bdapply'></span>

<h3>Description</h3>

<p>A routine that reads in the SNP data serially
from a binary dosage file and applies a user
specified function to the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bdapply(bdinfo, func, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bdapply_+3A_bdinfo">bdinfo</code></td>
<td>
<p>List with information about the
binary dosage file returned from getbdinfo</p>
</td></tr>
<tr><td><code id="bdapply_+3A_func">func</code></td>
<td>
<p>A user supplied function to apply
to the data for each snp. The function must be
provide with the following parameters, dosage,
p0, p1, and p2, where dosage is the dosage values
for each subject and p0, p1, and p2 are the
probabilities that a subject has zero, one,
and two copies of the alternate allele,
respectively.</p>
</td></tr>
<tr><td><code id="bdapply_+3A_...">...</code></td>
<td>
<p>Additional parameters needed by the
user supplied function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with length equal to the number
of SNPs in the binary dosage file. Each element
of the list is the value returned by the user
supplied function
</p>


<h3>See Also</h3>

<p>Other Iterating functions: 
<code><a href="#topic+genapply">genapply</a>()</code>,
<code><a href="#topic+vcfapply">vcfapply</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get information about a binary dosage file

vcf1abdfile &lt;- system.file("extdata", "vcf1a.bdose", package = "BinaryDosage")
bdinfo &lt;- getbdinfo(bdfiles = vcf1abdfile)

# Apply the getaaf, get alternate allele frequency, function
# to all the SNPs in the binary dosage file

aaf &lt;- bdapply(bdinfo = bdinfo,
               func = BinaryDosage:::getaaf)
</code></pre>

<hr>
<h2 id='bdmerge'>Merge binary dosage files together</h2><span id='topic+bdmerge'></span>

<h3>Description</h3>

<p>Routine to merge binary dosage files together. The files
don't have to be in the same format. They will be merged
into a file with the format specified. Information about
the SNPs, aaf, maf, avgcall, rsq, can be maintained for
each file, or recalculated for the merged set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bdmerge(
  mergefiles,
  format = 4,
  subformat = 0L,
  bdfiles,
  famfiles = character(),
  mapfiles = character(),
  onegroup = TRUE,
  bdoptions = character(),
  snpjoin = "inner"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bdmerge_+3A_mergefiles">mergefiles</code></td>
<td>
<p>Vector of file names for the merged binary
files. The first is the
binary dosage data containing the dosages and genetic
probabilities. The second file name is the family information
file. The third file name is the SNP information file.
The family and SNP information files are not used if the
binary dosage file is in format 4. For this format the
family and SNP information are in the file with the dosages
and genetic probabilities.</p>
</td></tr>
<tr><td><code id="bdmerge_+3A_format">format</code></td>
<td>
<p>The format of the output binary dosage file.
Allowed values are 1, 2, 3, and 4. The default value is 4.
Using the default value is recommended.</p>
</td></tr>
<tr><td><code id="bdmerge_+3A_subformat">subformat</code></td>
<td>
<p>The subformat of the format of the output
binary dosage file. A value of 1 or 3 indicates that only the
dosage value is saved. A value of 2 or 4 indicates
the dosage and genetic probabilities will be output. Values
of 3 or 4 are only allowed with formats 3 and 4. If a value
of zero if provided, and genetic probabilities are in the vcf
file, subformat 2 will be used for formats 1 and 2, and
subformat 4 will be used for formats 3 and 4. If the vcf file
does not contain genetic probabilities, subformat 1 will be
used for formats 1 and 2, and subformat 3 will be used for
formats 3 and 4. The default value is 0.</p>
</td></tr>
<tr><td><code id="bdmerge_+3A_bdfiles">bdfiles</code></td>
<td>
<p>Vector of binary dosage file names to be merged.</p>
</td></tr>
<tr><td><code id="bdmerge_+3A_famfiles">famfiles</code></td>
<td>
<p>Vector of family file names that correspond to
the names in bdfiles. If the binary dosage files are all in format
4, this may be an empty character array. Default value is character().</p>
</td></tr>
<tr><td><code id="bdmerge_+3A_mapfiles">mapfiles</code></td>
<td>
<p>Vector of map file names that correspond to the
names in bdfiles. If the binary dosage files are all in format
4, this may be an empty character array. Default value is character().</p>
</td></tr>
<tr><td><code id="bdmerge_+3A_onegroup">onegroup</code></td>
<td>
<p>Indicator to combine all the samples in one group.
If this is FALSE, the groups in each binary dosage file are
maintained and any binary dosage file with one group is made into
its own group. Default value is TRUE.</p>
</td></tr>
<tr><td><code id="bdmerge_+3A_bdoptions">bdoptions</code></td>
<td>
<p>Options indicating what information to calculate
and store for each SNP. These can be aaf, maf, and rsq. This option
is only available if format is equal to 4 and onegroup is TRUE.
Default value is character().</p>
</td></tr>
<tr><td><code id="bdmerge_+3A_snpjoin">snpjoin</code></td>
<td>
<p>Character value that can be either &quot;inner&quot; or &quot;outer&quot;.
This indicates whether to do an inner or outer join of the SNPs in
each binary dosage file. Default value is &quot;inner&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bdvcf1afile &lt;- system.file("extdata", "vcf1a.bdose", package = "BinaryDosage")
bdvcf1bfile &lt;- system.file("extdata", "vcf1b.bdose", package = "BinaryDosage")
mergefiles &lt;- tempfile()

BinaryDosage:::bdmerge(mergefiles = mergefiles,
                       bdfiles = c(bdvcf1afile, bdvcf1bfile),
                       bdoptions = "maf")
bdinfo &lt;- getbdinfo(mergefiles)
</code></pre>

<hr>
<h2 id='genapply'>Apply a function to each SNP in a
gen, impute2, file</h2><span id='topic+genapply'></span>

<h3>Description</h3>

<p>A routine that reads in the SNP data serially
from a gen file and applies a user
specified function to the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genapply(geninfo, func, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genapply_+3A_geninfo">geninfo</code></td>
<td>
<p>List with information about the
gen, impute2, file returned from <a href="#topic+getgeninfo">getgeninfo</a></p>
</td></tr>
<tr><td><code id="genapply_+3A_func">func</code></td>
<td>
<p>A user supplied function to apply
to the data for each snp. The function must be
provide with the following parameters, dosage,
p0, p1, and p2, where dosage is the dosage values
for each subject and p0, p1, and p2 are the
probabilities that a subject has zero, one,
and two copies of the alternate allele,
respectively.</p>
</td></tr>
<tr><td><code id="genapply_+3A_...">...</code></td>
<td>
<p>Additional parameters needed by the
user supplied function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with length equal to the number
of SNPs in the vcf file. Each element
of the list is the value returned by the user
supplied function
</p>


<h3>See Also</h3>

<p>Other Iterating functions: 
<code><a href="#topic+bdapply">bdapply</a>()</code>,
<code><a href="#topic+vcfapply">vcfapply</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get information about a gen, impute2, file

gen1afile &lt;- system.file("extdata", "set1a.imp", package = "BinaryDosage")
geninfo &lt;- getgeninfo(genfiles = gen1afile,
                      snpcolumns = c(1L, 3L, 2L, 4L, 5L),
                      header = TRUE)


aaf &lt;- genapply(geninfo = geninfo,
                func = BinaryDosage:::getaaf)
</code></pre>

<hr>
<h2 id='gentobd'>Convert a gen file to a binary dosage file</h2><span id='topic+gentobd'></span>

<h3>Description</h3>

<p>Routine to read information from a gen file and create
a binary dosage file. Note: This routine can take a long
time to run if the gen file is large.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gentobd(
  genfiles,
  snpcolumns = 1L:5L,
  startcolumn = 6L,
  impformat = 3L,
  chromosome = character(),
  header = c(FALSE, TRUE),
  gz = FALSE,
  sep = "\t",
  bdfiles,
  format = 4L,
  subformat = 0L,
  snpidformat = 0L,
  bdoptions = character(0)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gentobd_+3A_genfiles">genfiles</code></td>
<td>
<p>A vector of file names.
The first is the name of the gen file. The
second is name of the sample file that contains
the subject information.</p>
</td></tr>
<tr><td><code id="gentobd_+3A_snpcolumns">snpcolumns</code></td>
<td>
<p>Column numbers containing chromosome,
snpid, location, reference allele, alternate allele,
respectively. This must be an integer vector. All
values must be positive except for the chromosome.
The value for the chromosome may be -1 or -0.
-1 indicates that the chromosome value is passed to
the routine using the chromosome parameter.
0 indicates that the chromosome value is in the snpid
and that the snpid has the format chromosome:other_data.
Default value is c(1L, 2L, 3L, 4L, 5L).</p>
</td></tr>
<tr><td><code id="gentobd_+3A_startcolumn">startcolumn</code></td>
<td>
<p>Column number of first column with
genetic probabilities or dosages. Must
be an integer value. Default value is 6L.</p>
</td></tr>
<tr><td><code id="gentobd_+3A_impformat">impformat</code></td>
<td>
<p>Number of genetic data values per
subject. 1 indicates dosage only, 2 indicates P(g=0)
and P(g=1) only, 3 indicates P(g=0), P(g=1), and
P(g=2). Default value is 3L.</p>
</td></tr>
<tr><td><code id="gentobd_+3A_chromosome">chromosome</code></td>
<td>
<p>Chromosome value to use if the
first value of the snpcolumns is equal to 0.
Default value is character().</p>
</td></tr>
<tr><td><code id="gentobd_+3A_header">header</code></td>
<td>
<p>Indicators if the gen and sample files
have headers. If the gen file does not have a
header. A sample file must be included.
Default value is c(FALSE, TRUE).</p>
</td></tr>
<tr><td><code id="gentobd_+3A_gz">gz</code></td>
<td>
<p>Indicator if file is compressed using gzip.
Default value is FALSE.</p>
</td></tr>
<tr><td><code id="gentobd_+3A_sep">sep</code></td>
<td>
<p>Separator used in the gen file. Default
value is <code>"\t"</code></p>
</td></tr>
<tr><td><code id="gentobd_+3A_bdfiles">bdfiles</code></td>
<td>
<p>Vector of names of the output files.
The binary dosage file name is first. The family and
map files follow. For format 4, no family and map file
names are needed.</p>
</td></tr>
<tr><td><code id="gentobd_+3A_format">format</code></td>
<td>
<p>The format of the output binary dosage file.
Allowed values are 1, 2, 3, and 4. The default value is 4.
Using the default value is recommended.</p>
</td></tr>
<tr><td><code id="gentobd_+3A_subformat">subformat</code></td>
<td>
<p>The subformat of the format of the output
binary dosage file. A value of 1 or 3 indicates that only the
dosage value is saved. A value of 2 or 4 indicates
the dosage and genetic probabilities will be output. Values
of 3 or 4 are only allowed with formats 3 and 4. If a value
of zero if provided, and genetic probabilities are in the vcf
file, subformat 2 will be used for formats 1 and 2, and
subformat 4 will be used for formats 3 and 4. If the vcf file
does not contain genetic probabilities, subformat 1 will be
used for formats 1 and 2, and subformat 3 will be used for
formats 3 and 4. The default value is 0.</p>
</td></tr>
<tr><td><code id="gentobd_+3A_snpidformat">snpidformat</code></td>
<td>
<p>The format that the SNP ID will be saved as.
-1 - SNP ID not written.
0 - same as in the VCF file.
1 - chromosome:location.
2 - chromosome:location:reference_allele:alternate_allele.
If snpidformat is 1 and the VCF file uses format 2, an error is
generated. Default value is 0.</p>
</td></tr>
<tr><td><code id="gentobd_+3A_bdoptions">bdoptions</code></td>
<td>
<p>Character array containing any of the following
value, &quot;aaf&quot;, &quot;maf&quot;, &quot;rsq&quot;. The presence of any of these
values indicates that the specified values should be
calculates and stored in the binary dosage file. These values only
apply to format 4.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Find the gen file names
gen3afile &lt;- system.file("extdata", "set3a.imp", package = "BinaryDosage")
gen3asample &lt;- system.file("extdata", "set3a.sample", package = "BinaryDosage")
# Get temporary output file name
bdfiles &lt;- tempfile()
# Convert the file
gentobd(genfiles = c(gen3afile, gen3asample),
        snpcolumns = c(0L, 2L:5L),
        bdfiles = bdfiles)
# Verify the file was written correctly
bdinfo &lt;- getbdinfo(bdfiles = bdfiles)
</code></pre>

<hr>
<h2 id='getaaf'>Calculate alternate allele frequency</h2><span id='topic+getaaf'></span>

<h3>Description</h3>

<p>Routine to calculate the alternate allele frequency given the dosages.
Missing values for dosage ignored. This function is used internally and
is exported for use in examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getaaf(dosage, p0, p1, p2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getaaf_+3A_dosage">dosage</code></td>
<td>
<p>Dosage values</p>
</td></tr>
<tr><td><code id="getaaf_+3A_p0">p0</code></td>
<td>
<p>Pr(g=0) - unused</p>
</td></tr>
<tr><td><code id="getaaf_+3A_p1">p1</code></td>
<td>
<p>Pr(g=1) - unused</p>
</td></tr>
<tr><td><code id="getaaf_+3A_p2">p2</code></td>
<td>
<p>Pr(g=2) - unused</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Alternate allele frequency
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get information about binary dosage file
bdfile &lt;- system.file("extdata", "vcf1a.bdose", package = "BinaryDosage")
bdinfo &lt;- getbdinfo(bdfiles = bdfile)
snp1 &lt;- getsnp(bdinfo = bdinfo, 1)
aaf &lt;- getaaf(snp1$dosage)
</code></pre>

<hr>
<h2 id='getbdinfo'>Get information about a binary dosage file</h2><span id='topic+getbdinfo'></span>

<h3>Description</h3>

<p>Routine to return information about a binary dosage file.
This information is used by other routines to
allow for quicker extraction of values from the
file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getbdinfo(bdfiles)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getbdinfo_+3A_bdfiles">bdfiles</code></td>
<td>
<p>Vector of file names. The first is the
binary dosage data containing the dosages and genetic
probabilities. The second file name is the family information
file. The third file name is the SNP information file.
The family and SNP information files are not used if the
binary dosage file is in format 4. For this format the
family and SNP information are in the file with the dosages
and genetic probabilities.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with information about the binary dosage file.
This includes family and subject IDs along with
a list of the SNPs in the file. Other information needed
to read the file is also included.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vcf1abdfile &lt;- system.file("extdata", "vcf1a.bdose", package = "BinaryDosage")
bdinfo &lt;- getbdinfo(bdfiles = vcf1abdfile)
</code></pre>

<hr>
<h2 id='getgeninfo'>Get information about a gen, impute2, file</h2><span id='topic+getgeninfo'></span>

<h3>Description</h3>

<p>Routine to return information about a gen file.
This information is used by other routines to
allow for quicker extraction of values from the
file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getgeninfo(
  genfiles,
  snpcolumns = 1L:5L,
  startcolumn = 6L,
  impformat = 3L,
  chromosome = character(),
  header = c(FALSE, TRUE),
  gz = FALSE,
  index = TRUE,
  snpidformat = 0L,
  sep = c("\t", "\t")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getgeninfo_+3A_genfiles">genfiles</code></td>
<td>
<p>A vector of file names.
The first is the name of the gen file. The
second is name of the sample file that contains
the subject information.</p>
</td></tr>
<tr><td><code id="getgeninfo_+3A_snpcolumns">snpcolumns</code></td>
<td>
<p>Column numbers containing chromosome,
snpid, location, reference allele, alternate allele,
respectively. This must be an integer vector. All
values must be positive except for the chromosome.
The value for the chromosome may be -1 or -0.
-1 indicates that the chromosome value is passed to
the routine using the chromosome parameter.
0 indicates that the chromosome value is in the snpid
and that the snpid has the format chromosome:other_data.
Default value is c(1L, 2L, 3L, 4L, 5L).</p>
</td></tr>
<tr><td><code id="getgeninfo_+3A_startcolumn">startcolumn</code></td>
<td>
<p>Column number of first column with
genetic probabilities or dosages. Must
be an integer value. Default value is 6L.</p>
</td></tr>
<tr><td><code id="getgeninfo_+3A_impformat">impformat</code></td>
<td>
<p>Number of genetic data values per
subject. 1 indicates dosage only, 2 indicates P(g=0)
and P(g=1) only, 3 indicates P(g=0), P(g=1), and
P(g=2). Default value is 3L.</p>
</td></tr>
<tr><td><code id="getgeninfo_+3A_chromosome">chromosome</code></td>
<td>
<p>Chromosome value to use if the
first value of the snpcolumns is equal to 0.
Default value is character().</p>
</td></tr>
<tr><td><code id="getgeninfo_+3A_header">header</code></td>
<td>
<p>Indicators if the gen and sample files
have headers. If the gen file does not have a
header. A sample file must be included.
Default value is c(FALSE, TRUE).</p>
</td></tr>
<tr><td><code id="getgeninfo_+3A_gz">gz</code></td>
<td>
<p>Indicator if file is compressed using gzip.
Default value is FALSE.</p>
</td></tr>
<tr><td><code id="getgeninfo_+3A_index">index</code></td>
<td>
<p>Indicator if file should be indexed. This
allows for faster reading of the file. Indexing a gzipped
file is not supported.
Default value is TRUE.</p>
</td></tr>
<tr><td><code id="getgeninfo_+3A_snpidformat">snpidformat</code></td>
<td>
<p>Format to change the snpid to.
0 indicates to use the snpid format in the file.
1 indicates to change the snpid into chromosome:location,
2 indicates to change the snpid into chromosome:location:referenceallele:alternateallele,
3 indicates to change the snpid into chromosome:location_referenceallele_alternateallele,
Default value is 0.</p>
</td></tr>
<tr><td><code id="getgeninfo_+3A_sep">sep</code></td>
<td>
<p>Separators used in the gen file and sample files,
respectively. If only value is provided it is used for both
files. Default value is c(<code>"\t"</code>, <code>"\t"</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with information about the gen file.
This includes family and subject IDs along with
a list of the SNPs in the file. Other information needed
to read the file is also included.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get file names of th gen and sample file
gen3afile &lt;- system.file("extdata", "set3a.imp", package = "BinaryDosage")
gen3ainfo &lt;- system.file("extdata", "set3a.sample", package = "BinaryDosage")

# Get the information about the gen file
geninfo &lt;- getgeninfo(genfiles = c(gen3afile, gen3ainfo),
                      snpcolumns = c(0L, 2L:5L))
</code></pre>

<hr>
<h2 id='getmaf'>Calculate minor allele frequency</h2><span id='topic+getmaf'></span>

<h3>Description</h3>

<p>Routine to calculate the minor allele frequency given the dosages.
Missing values for dosage ignored. This function is used internally and
is exported for use in examples. Note: The minor allele in one data set
may be different from another data set. This can make comparing minor
allele frequencies between data sets nonsensical.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getmaf(dosage, p0, p1, p2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getmaf_+3A_dosage">dosage</code></td>
<td>
<p>Dosage values</p>
</td></tr>
<tr><td><code id="getmaf_+3A_p0">p0</code></td>
<td>
<p>Pr(g=0) - unused</p>
</td></tr>
<tr><td><code id="getmaf_+3A_p1">p1</code></td>
<td>
<p>Pr(g=1) - unused</p>
</td></tr>
<tr><td><code id="getmaf_+3A_p2">p2</code></td>
<td>
<p>Pr(g=2) - unused</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Minor allele frequency
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get information about binary dosage file
bdfile &lt;- system.file("extdata", "vcf1a.bdose", package = "BinaryDosage")
bdinfo &lt;- getbdinfo(bdfiles = bdfile)
snp1 &lt;- getsnp(bdinfo = bdinfo, 1)
maf &lt;- getmaf(snp1$dosage)
</code></pre>

<hr>
<h2 id='getrsq'>Calculate imputation r squared</h2><span id='topic+getrsq'></span>

<h3>Description</h3>

<p>Routine to calculate the imputation r squared given the dosages
and Pr(g=2).
This is an estimate for the imputation r squared returned from
minimac and impute2. The r squared values are calculated slightly
differently between the programs. This estimate is based on the
method used by minimac. It does well for minor allele frequencies
above 5%. This function is used internally and is exported for
use in examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getrsq(dosage, p0, p1, p2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getrsq_+3A_dosage">dosage</code></td>
<td>
<p>Dosage values</p>
</td></tr>
<tr><td><code id="getrsq_+3A_p0">p0</code></td>
<td>
<p>Pr(g=0) - unused</p>
</td></tr>
<tr><td><code id="getrsq_+3A_p1">p1</code></td>
<td>
<p>Pr(g=1) - unused</p>
</td></tr>
<tr><td><code id="getrsq_+3A_p2">p2</code></td>
<td>
<p>Pr(g=2)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Imputation r squared
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get information about binary dosage file
bdfile &lt;- system.file("extdata", "vcf1a.bdose", package = "BinaryDosage")
bdinfo &lt;- getbdinfo(bdfiles = bdfile)
snp1 &lt;- getsnp(bdinfo = bdinfo, 1, dosageonly = FALSE)
rsq &lt;- BinaryDosage:::getrsq(snp1$dosage, p2 = snp1$p2)
</code></pre>

<hr>
<h2 id='getsnp'>Read SNP data from a binary dosage file</h2><span id='topic+getsnp'></span>

<h3>Description</h3>

<p>Routine to read the dosage and genetic probabilities about
a SNP from a binary dosage file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getsnp(bdinfo, snp, dosageonly = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getsnp_+3A_bdinfo">bdinfo</code></td>
<td>
<p>Information about a binary dosage file return
from getbdinfo</p>
</td></tr>
<tr><td><code id="getsnp_+3A_snp">snp</code></td>
<td>
<p>The SNP to read the information about. This may
be the SNP ID or the index of the SNP in the snps dataset in
the bdinfo list</p>
</td></tr>
<tr><td><code id="getsnp_+3A_dosageonly">dosageonly</code></td>
<td>
<p>Indicator to return the dosages only or the
dosages allowing with the genetic probabilities.
Default value is TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with either the dosages or the dosages and the genetic
probabilities.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get the information about the file
vcf1abdfile &lt;- system.file("extdata", "vcf1a.bdose", package = "BinaryDosage")
bdinfo &lt;- getbdinfo(bdfiles = vcf1abdfile)

# Read the first SNP
getsnp(bdinfo, 1, FALSE)
</code></pre>

<hr>
<h2 id='getvcfinfo'>Get information about a vcf file</h2><span id='topic+getvcfinfo'></span>

<h3>Description</h3>

<p>Routine to return information about a vcf file.
This information is used by other routines to
allow for quicker extraction of values from the
file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getvcfinfo(vcffiles, gz = FALSE, index = TRUE, snpidformat = 0L)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getvcfinfo_+3A_vcffiles">vcffiles</code></td>
<td>
<p>A vector of file names.
The first is the name of the vcf file. The
second is name of the file that contains information
about the imputation of the SNPs. This file is produced
by minimac 3 and 4.</p>
</td></tr>
<tr><td><code id="getvcfinfo_+3A_gz">gz</code></td>
<td>
<p>Indicator if VCF file is compressed using gzip.
Default value is FALSE.</p>
</td></tr>
<tr><td><code id="getvcfinfo_+3A_index">index</code></td>
<td>
<p>Indicator if file should be indexed. This
allows for faster reading of the file. Indexing a gzipped
file is not supported.
Default value is TRUE.</p>
</td></tr>
<tr><td><code id="getvcfinfo_+3A_snpidformat">snpidformat</code></td>
<td>
<p>The format that the SNP ID will be saved as.
0 - same as in the VCF file
1 - chromosome:location
2 - chromosome:location:referenceallele:alternateallele
If snpidformat is 1 and the VCF file uses format 2, an error is
generated. Default value is 0.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List containing information about the VCF file
to include file name, subject IDs, and information about
the SNPs. Indices for faster reading will be included
if index is set to TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get file names of th vcf and infromation file
vcf1afile &lt;- system.file("extdata", "set1a.vcf", package = "BinaryDosage")
vcf1ainfo &lt;- system.file("extdata", "set1a.info", package = "BinaryDosage")

# Get the information about the vcf file
vcf1ainfo &lt;- getvcfinfo(vcffiles = c(vcf1afile, vcf1ainfo))
</code></pre>

<hr>
<h2 id='vcfapply'>Apply a function to each SNP in a vcf file</h2><span id='topic+vcfapply'></span>

<h3>Description</h3>

<p>A routine that reads in the SNP data serially
from a vcf file and applies a user specified
function to the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcfapply(vcfinfo, func, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vcfapply_+3A_vcfinfo">vcfinfo</code></td>
<td>
<p>List with information about the
vcf file returned from getvcfinfo</p>
</td></tr>
<tr><td><code id="vcfapply_+3A_func">func</code></td>
<td>
<p>A user supplied function to apply
to the data for each snp. The function must be
provide with the following parameters, dosage,
p0, p1, and p2, where dosage is the dosage values
for each subject and p0, p1, and p2 are the
probabilities that a subject has zero, one,
and two copies of the alternate allele,
respectively.</p>
</td></tr>
<tr><td><code id="vcfapply_+3A_...">...</code></td>
<td>
<p>Additional parameters needed by the
user supplied function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with length equal to the number
of SNPs in the vcf file. Each element
of the list is the value returned by the user
supplied function
</p>


<h3>See Also</h3>

<p>Other Iterating functions: 
<code><a href="#topic+bdapply">bdapply</a>()</code>,
<code><a href="#topic+genapply">genapply</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get information about a vcf file

vcf1afile &lt;- system.file("extdata", "set1a.vcf", package = "BinaryDosage")
vcfinfo &lt;- getvcfinfo(vcffiles = vcf1afile)

# Apply the getaaf, get alternate allele frequency, function
# to all the SNPs in the vcf file

aaf &lt;- vcfapply(vcfinfo = vcfinfo,
                func = BinaryDosage:::getaaf)
</code></pre>

<hr>
<h2 id='vcftobd'>Convert a VCF file to a binary dosage file</h2><span id='topic+vcftobd'></span>

<h3>Description</h3>

<p>Routine to read information from a VCF file and create
a binary dosage file. The function is designed to use
files return from the Michigan Imputation Server but will
run on other VCF files if they contain dosage and genetic
probabilities. Note: This routine can take a long time to
run if the VCF file is large.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcftobd(
  vcffiles,
  gz = FALSE,
  bdfiles,
  format = 4L,
  subformat = 0L,
  snpidformat = 0,
  bdoptions = character(0)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vcftobd_+3A_vcffiles">vcffiles</code></td>
<td>
<p>A vector of file names.
The first is the name of the vcf file. The
second is name of the file that contains information
about the imputation of the SNPs. This file is produced
by minimac 3 and 4.</p>
</td></tr>
<tr><td><code id="vcftobd_+3A_gz">gz</code></td>
<td>
<p>Indicator if VCF file is compressed using gzip.
Default value is FALSE.</p>
</td></tr>
<tr><td><code id="vcftobd_+3A_bdfiles">bdfiles</code></td>
<td>
<p>Vector of names of the output files.
The binary dosage file name is first. The family and
map files follow. For format 4, no family and map file
names are needed.</p>
</td></tr>
<tr><td><code id="vcftobd_+3A_format">format</code></td>
<td>
<p>The format of the output binary dosage file.
Allowed values are 1, 2, 3, and 4. The default value is 4.
Using the default value is recommended.</p>
</td></tr>
<tr><td><code id="vcftobd_+3A_subformat">subformat</code></td>
<td>
<p>The subformat of the format of the output
binary dosage file. A value of 1 or 3 indicates that only the
dosage value is saved. A value of 2 or 4 indicates
the dosage and genetic probabilities will be output. Values
of 3 or 4 are only allowed with formats 3 and 4. If a value
of zero if provided, and genetic probabilities are in the vcf
file, subformat 2 will be used for formats 1 and 2, and
subformat 4 will be used for formats 3 and 4. If the vcf file
does not contain genetic probabilities, subformat 1 will be
used for formats 1 and 2, and subformat 3 will be used for
formats 3 and 4. The default value is 0.</p>
</td></tr>
<tr><td><code id="vcftobd_+3A_snpidformat">snpidformat</code></td>
<td>
<p>The format that the SNP ID will be saved as.
-1 SNP ID not written
0 - same as in the VCF file
1 - chromosome:location
2 - chromosome:location:reference_allele:alternate_allele
If snpidformat is 1 and the VCF file uses format 2, an error is
generated. Default value is 0.</p>
</td></tr>
<tr><td><code id="vcftobd_+3A_bdoptions">bdoptions</code></td>
<td>
<p>Character array containing any of the following
value, &quot;aaf&quot;, &quot;maf&quot;, &quot;rsq&quot;. The presence of any of these
values indicates that the specified values should be
calculates and stored in the binary dosage file. These values only
apply to format 4.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Find the vcf file names
vcf1afile &lt;- system.file("extdata", "set1a.vcf", package = "BinaryDosage")
vcf1ainfo &lt;- system.file("extdata", "set1a.info", package = "BinaryDosage")
bdfiles &lt;- tempfile()
# Convert the file
vcftobd(vcffiles = c(vcf1afile, vcf1ainfo), bdfiles = bdfiles)
# Verify the file was written correctly
bdinfo &lt;- getbdinfo(bdfiles)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
