<!DOCTYPE html><html lang="en"><head><title>Help for package biocompute</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {biocompute}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#biocompute-package'><p>biocompute: Create and Manipulate BioCompute Objects</p></a></li>
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#compose_description_v1.4.2'><p>Compose BioCompute Object - Description Domain (v1.4.2)</p></a></li>
<li><a href='#compose_error_v1.4.2'><p>Compose BioCompute Object - Error Domain (v1.4.2)</p></a></li>
<li><a href='#compose_execution_v1.4.2'><p>Compose BioCompute Object - Execution Domain (v1.4.2)</p></a></li>
<li><a href='#compose_extension_v1.4.2'><p>Compose BioCompute Object - Extension Domain (v1.4.2)</p></a></li>
<li><a href='#compose_fhir_v1.4.2'><p>Compose BioCompute Object - FHIR Extension (v1.4.2)</p></a></li>
<li><a href='#compose_io_v1.4.2'><p>Compose BioCompute Object - Input and Output Domain (v1.4.2)</p></a></li>
<li><a href='#compose_parametric_v1.4.2'><p>Compose BioCompute Object - Parametric Domain (v1.4.2)</p></a></li>
<li><a href='#compose_provenance_v1.4.2'><p>Compose BioCompute Object - Provenance Domain (v1.4.2)</p></a></li>
<li><a href='#compose_scm_v1.4.2'><p>Compose BioCompute Object - SCM Extension (v1.4.2)</p></a></li>
<li><a href='#compose_tlf_v1.4.2'><p>Compose BioCompute Object - Top Level Fields (v1.4.2)</p></a></li>
<li><a href='#compose_usability_v1.4.2'><p>Compose BioCompute Object - Usability Domain (v1.4.2)</p></a></li>
<li><a href='#compose_v1.4.2'><p>Compose BioCompute Object (v1.4.2)</p></a></li>
<li><a href='#convert_json'><p>Convert BioCompute Object or domain to JSON string</p></a></li>
<li><a href='#convert_yaml'><p>Convert BioCompute Object or domain to YAML string</p></a></li>
<li><a href='#export_html'><p>Export BioCompute Object as HTML</p></a></li>
<li><a href='#export_json'><p>Export BioCompute Object as JSON</p></a></li>
<li><a href='#export_pdf'><p>Export BioCompute Object as PDF</p></a></li>
<li><a href='#export_sevenbridges'><p>Export BioCompute Object to Seven Bridges Platforms</p></a></li>
<li><a href='#export_word'><p>Export BioCompute Object as Word document</p></a></li>
<li><a href='#generate_example'><p>Generate example BioCompute Objects</p></a></li>
<li><a href='#generate_id'><p>Generate ID for the BioCompute Object</p></a></li>
<li><a href='#is_bco'><p>Is this a BCO object?</p></a></li>
<li><a href='#is_domain'><p>Is this a domain object?</p></a></li>
<li><a href='#read_bco'><p>Parse Biocompute Object From JSON File to R Object</p></a></li>
<li><a href='#validate_checksum_v1.4.2'><p>BioCompute Objects checksum validator (v1.4.2)</p></a></li>
<li><a href='#validate_schema_v1.4.2'><p>BioCompute Objects schema validator (v1.4.2)</p></a></li>
<li><a href='#versions'><p>BioCompute Object specification versions</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Create and Manipulate BioCompute Objects</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Soner Koc &lt;soner.koc@sevenbridges.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Tools to create, validate, and export BioCompute Objects
    described in King et al. (2019) &lt;<a href="https://doi.org/10.17605%2Fosf.io%2Fh59uh">doi:10.17605/osf.io/h59uh</a>&gt;.
    Users can encode information in data frames, and compose
    BioCompute Objects from the domains defined by the standard.
    A checksum validator and a JSON schema validator are provided.
    This package also supports exporting BioCompute Objects as JSON,
    PDF, HTML, or 'Word' documents, and exporting to cloud-based platforms.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/AGPL-3">AGPL-3</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://sbg.github.io/biocompute/">https://sbg.github.io/biocompute/</a>,
<a href="https://github.com/sbg/biocompute">https://github.com/sbg/biocompute</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/sbg/biocompute/issues">https://github.com/sbg/biocompute/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, jsonlite, yaml, digest, uuid, jsonvalidate, httr,
curl, crayon, cli, stringr, magrittr, rmarkdown</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-04-01 15:14:46 UTC; c02ck1a7lvdm</td>
</tr>
<tr>
<td>Author:</td>
<td>Soner Koc <a href="https://orcid.org/0000-0002-0772-6600"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Jeffrey Grover <a href="https://orcid.org/0000-0001-6246-1767"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Nan Xiao <a href="https://orcid.org/0000-0002-0250-5673"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Dennis Dean <a href="https://orcid.org/0000-0002-7621-9717"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Seven Bridges Genomics [cph, fnd]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-03 07:00:27 UTC</td>
</tr>
</table>
<hr>
<h2 id='biocompute-package'>biocompute: Create and Manipulate BioCompute Objects</h2><span id='topic+biocompute'></span><span id='topic+biocompute-package'></span>

<h3>Description</h3>

<p>Tools to create, validate, and export BioCompute Objects described in King et al. (2019) &lt;doi:10.17605/osf.io/h59uh&gt;. Users can encode information in data frames, and compose BioCompute Objects from the domains defined by the standard. A checksum validator and a JSON schema validator are provided. This package also supports exporting BioCompute Objects as JSON, PDF, HTML, or 'Word' documents, and exporting to cloud-based platforms.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Soner Koc <a href="mailto:soner.koc@sevenbridges.com">soner.koc@sevenbridges.com</a> (<a href="https://orcid.org/0000-0002-0772-6600">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Jeffrey Grover <a href="mailto:jeffrey.grover@sevenbridges.com">jeffrey.grover@sevenbridges.com</a> (<a href="https://orcid.org/0000-0001-6246-1767">ORCID</a>)
</p>
</li>
<li><p> Nan Xiao (<a href="https://orcid.org/0000-0002-0250-5673">ORCID</a>)
</p>
</li>
<li><p> Dennis Dean <a href="mailto:dennis.dean@sevenbridges.com">dennis.dean@sevenbridges.com</a> (<a href="https://orcid.org/0000-0002-7621-9717">ORCID</a>)
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Seven Bridges Genomics [copyright holder, funder]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://sbg.github.io/biocompute/">https://sbg.github.io/biocompute/</a>
</p>
</li>
<li> <p><a href="https://github.com/sbg/biocompute">https://github.com/sbg/biocompute</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/sbg/biocompute/issues">https://github.com/sbg/biocompute/issues</a>
</p>
</li></ul>


<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>

<hr>
<h2 id='compose_description_v1.4.2'>Compose BioCompute Object - Description Domain (v1.4.2)</h2><span id='topic+compose_description_v1.4.2'></span><span id='topic+compose_description'></span>

<h3>Description</h3>

<p>Compose BioCompute Object - Description Domain (v1.4.2)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compose_description_v1.4.2(
  keywords = NULL,
  xref = NULL,
  platform = list("Seven Bridges Platform"),
  pipeline_meta = NULL,
  pipeline_prerequisite = NULL,
  pipeline_input = NULL,
  pipeline_output = NULL
)

compose_description(
  keywords = NULL,
  xref = NULL,
  platform = list("Seven Bridges Platform"),
  pipeline_meta = NULL,
  pipeline_prerequisite = NULL,
  pipeline_input = NULL,
  pipeline_output = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compose_description_v1.4.2_+3A_keywords">keywords</code></td>
<td>
<p>Character vector. A list of keywords to aid in
searchability and description of the experiment.</p>
</td></tr>
<tr><td><code id="compose_description_v1.4.2_+3A_xref">xref</code></td>
<td>
<p>Data frame. A list of the databases and/or ontology IDs
that are cross-referenced in the BCO.</p>
</td></tr>
<tr><td><code id="compose_description_v1.4.2_+3A_platform">platform</code></td>
<td>
<p>Character string or list. Reference to a particular deployment
of an existing platform where this BCO can be reproduced.</p>
</td></tr>
<tr><td><code id="compose_description_v1.4.2_+3A_pipeline_meta">pipeline_meta</code></td>
<td>
<p>Data frame. Pipeline metadata.
Variables include <code>step_number</code>, <code>name</code>, <code>description</code>,
and <code>version</code>.</p>
</td></tr>
<tr><td><code id="compose_description_v1.4.2_+3A_pipeline_prerequisite">pipeline_prerequisite</code></td>
<td>
<p>Data frame. Packages or prerequisites
for running the tools used. Variables include <code>step_number</code>,
<code>name</code>, <code>uri</code>, and <code>access_time</code>.</p>
</td></tr>
<tr><td><code id="compose_description_v1.4.2_+3A_pipeline_input">pipeline_input</code></td>
<td>
<p>Data frame. Input files for the tools.
Variables include <code>step_number</code>, <code>uri</code>, and <code>access_time</code>.</p>
</td></tr>
<tr><td><code id="compose_description_v1.4.2_+3A_pipeline_output">pipeline_output</code></td>
<td>
<p>Data frame. Output files for the tools.
Variables include <code>step_number</code>, <code>uri</code>, and <code>access_time</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of class <code>bco.domain</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>keywords &lt;- c("HCV1a", "Ledipasvir", "antiviral resistance", "SNP", "amino acid substitutions")
xref &lt;- data.frame(
  "namespace" = c("pubchem.compound", "pubmed", "so", "taxonomy"),
  "name" = c("PubChem-compound", "PubMed", "Sequence Ontology", "Taxonomy"),
  "ids" = I(list(
    "67505836",
    "26508693",
    c("SO:000002", "SO:0000694", "SO:0000667", "SO:0000045"),
    "31646"
  )),
  "access_time" = c(
    as.POSIXct("2017-01-20T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-21T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-22T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-23T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST")
  ),
  stringsAsFactors = FALSE
)

platform &lt;- "Seven Bridges Platform"

pipeline_meta &lt;- data.frame(
  "step_number" = c("1"),
  "name" = c("HIVE-hexagon"),
  "description" = c("Alignment of reads to a set of references"),
  "version" = c("1.3"),
  stringsAsFactors = FALSE
)

pipeline_prerequisite &lt;- data.frame(
  "step_number" = rep("1", 5),
  "name" = c(
    "Hepatitis C virus genotype 1",
    "Hepatitis C virus type 1b complete genome",
    "Hepatitis C virus (isolate JFH-1) genomic RNA",
    "Hepatitis C virus clone J8CF, complete genome",
    "Hepatitis C virus S52 polyprotein gene"
  ),
  "uri" = c(
    "https://www.ncbi.nlm.nih.gov/nuccore/22129792",
    "https://www.ncbi.nlm.nih.gov/nuccore/5420376",
    "https://www.ncbi.nlm.nih.gov/nuccore/13122261",
    "https://www.ncbi.nlm.nih.gov/nuccore/386646758",
    "https://www.ncbi.nlm.nih.gov/nuccore/295311559"
  ),
  "access_time" = c(
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST")
  ),
  stringsAsFactors = FALSE
)

pipeline_input &lt;- data.frame(
  "step_number" = rep("1", 2),
  "uri" = c(
    "https://example.com/dna.cgi?cmd=objFile&amp;ids=514683",
    "https://example.com/dna.cgi?cmd=objFile&amp;ids=514682"
  ),
  "access_time" = c(
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST")
  ),
  stringsAsFactors = FALSE
)

pipeline_output &lt;- data.frame(
  "step_number" = rep("1", 2),
  "uri" = c(
    "https://example.com/data/514769/allCount-aligned.csv",
    "https://example.com/data/514801/SNPProfile*.csv"
  ),
  "access_time" = c(
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST")
  ),
  stringsAsFactors = FALSE
)

compose_description(
  keywords, xref, platform,
  pipeline_meta, pipeline_prerequisite, pipeline_input, pipeline_output
) %&gt;% convert_json()
</code></pre>

<hr>
<h2 id='compose_error_v1.4.2'>Compose BioCompute Object - Error Domain (v1.4.2)</h2><span id='topic+compose_error_v1.4.2'></span><span id='topic+compose_error'></span>

<h3>Description</h3>

<p>The error domain can be used to determine what range of input
returns outputs that are within the tolerance level defined
in this subdomain and therefore can be used to optimize algorithm
(<a href="https://github.com/biocompute-objects/BCO_Specification/blob/1.4.2/docs/error-domain.md">domain definition</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compose_error_v1.4.2(empirical = NULL, algorithmic = NULL)

compose_error(empirical = NULL, algorithmic = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compose_error_v1.4.2_+3A_empirical">empirical</code></td>
<td>
<p>Data frame. Variables include <code>key</code> and <code>value</code>.
Each row is one item in the empirical error subdomain.</p>
</td></tr>
<tr><td><code id="compose_error_v1.4.2_+3A_algorithmic">algorithmic</code></td>
<td>
<p>Data frame. Variables include <code>key</code> and <code>value</code>.
Each row is one item in the algorithmic subdomain.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of class <code>bco.domain</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>empirical &lt;- data.frame(
  "key" = c("false_negative_alignment_hits", "false_discovery"),
  "value" = c("&lt;0.0010", "&lt;0.05"),
  stringsAsFactors = FALSE
)

algorithmic &lt;- data.frame(
  "key" = c("false_positive_mutation_calls", "false_discovery"),
  "value" = c("&lt;0.00005", "0.005"),
  stringsAsFactors = FALSE
)

compose_error(empirical, algorithmic) %&gt;% convert_json()
</code></pre>

<hr>
<h2 id='compose_execution_v1.4.2'>Compose BioCompute Object - Execution Domain (v1.4.2)</h2><span id='topic+compose_execution_v1.4.2'></span><span id='topic+compose_execution'></span>

<h3>Description</h3>

<p>Compose BioCompute Object - Execution Domain (v1.4.2)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compose_execution_v1.4.2(
  script = NULL,
  script_driver = NULL,
  software_prerequisites = NULL,
  external_data_endpoints = NULL,
  environment_variables = NULL
)

compose_execution(
  script = NULL,
  script_driver = NULL,
  software_prerequisites = NULL,
  external_data_endpoints = NULL,
  environment_variables = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compose_execution_v1.4.2_+3A_script">script</code></td>
<td>
<p>Character string or list. Points to internal or external
references to an object that was used to perform computations
for this BCO instance.</p>
</td></tr>
<tr><td><code id="compose_execution_v1.4.2_+3A_script_driver">script_driver</code></td>
<td>
<p>Character string. Indicate what kind of
executable can be launched in order to perform a sequence of
commands described in the script in order to run the pipeline.</p>
</td></tr>
<tr><td><code id="compose_execution_v1.4.2_+3A_software_prerequisites">software_prerequisites</code></td>
<td>
<p>Data frame. The minimal necessary
prerequisites, library, and tool versions needed to successfully
run the script to produce BCO. Variables include
<code>name</code>, <code>version</code>, <code>uri</code>, <code>access_time</code>,
and <code>sha1_chksum</code>.
Each row is one item in the output subdomain.</p>
</td></tr>
<tr><td><code id="compose_execution_v1.4.2_+3A_external_data_endpoints">external_data_endpoints</code></td>
<td>
<p>Data frame. The minimal necessary
domain-specific external data source access to successfully
run the script to produce the BCO. Variables include
<code>mediatype</code>, <code>name</code>, and <code>url</code>.
Each row is one item in the output subdomain.</p>
</td></tr>
<tr><td><code id="compose_execution_v1.4.2_+3A_environment_variables">environment_variables</code></td>
<td>
<p>Data frame. Key-value pairs
useful to configure the execution environment on the
target platform. Variables include <code>key</code> and <code>value</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of class <code>bco.domain</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>script &lt;- "https://example.com/workflows/antiviral_resistance_detection_hive.py"
script_driver &lt;- "shell"
software_prerequisites &lt;- data.frame(
  "name" = c("HIVE-hexagon", "HIVE-heptagon"),
  "version" = c("babajanian.1", "albinoni.2"),
  "uri" = c(
    "https://example.com/dna.cgi?cmd=dna-hexagon&amp;cmdMode=-",
    "https://example.com/dna.cgi?cmd=dna-heptagon&amp;cmdMode=-"
  ),
  "access_time" = c(
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST")
  ),
  "sha1_chksum" = c("d60f506cddac09e9e816531e7905ca1ca6641e3c", NA),
  stringsAsFactors = FALSE
)
external_data_endpoints &lt;- data.frame(
  "name" = c("generic name", "access to ftp server", "access to e-utils web service"),
  "url" = c(
    "protocol://domain:port/application/path",
    "ftp://data.example.com:21/",
    "https://eutils.ncbi.nlm.nih.gov/entrez/eutils"
  ),
  stringsAsFactors = FALSE
)
environment_variables &lt;- data.frame(
  "key" = c("HOSTTYPE", "EDITOR"),
  "value" = c("x86_64-linux", "vim")
)

compose_execution(
  script, script_driver, software_prerequisites, external_data_endpoints, environment_variables
) %&gt;% convert_json()
</code></pre>

<hr>
<h2 id='compose_extension_v1.4.2'>Compose BioCompute Object - Extension Domain (v1.4.2)</h2><span id='topic+compose_extension_v1.4.2'></span><span id='topic+compose_extension'></span>

<h3>Description</h3>

<p>Compose BioCompute Object - Extension Domain (v1.4.2)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compose_extension_v1.4.2(fhir = NULL, scm = NULL)

compose_extension(fhir = NULL, scm = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compose_extension_v1.4.2_+3A_fhir">fhir</code></td>
<td>
<p>FHIR extension domain composed by <code><a href="#topic+compose_fhir">compose_fhir</a></code>.</p>
</td></tr>
<tr><td><code id="compose_extension_v1.4.2_+3A_scm">scm</code></td>
<td>
<p>SCM extension domain composed by <code><a href="#topic+compose_scm">compose_scm</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of class <code>bco.domain</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fhir_endpoint &lt;- "https://fhirtest.uhn.ca/baseDstu3"
fhir_version &lt;- "3"
fhir_resources &lt;- data.frame(
  "id" = c("21376", "6288583", "25544", "92440", "4588936"),
  "resource" = c(
    "Sequence", "DiagnosticReport", "ProcedureRequest",
    "Observation", "FamilyMemberHistory"
  ),
  stringsAsFactors = FALSE
)

fhir &lt;- compose_fhir(fhir_endpoint, fhir_version, fhir_resources)

scm_repository &lt;- "https://github.com/example/repo"
scm_type &lt;- "git"
scm_commit &lt;- "c9ffea0b60fa3bcf8e138af7c99ca141a6b8fb21"
scm_path &lt;- "workflow/hive-viral-mutation-detection.cwl"
scm_preview &lt;- "https://github.com/example/repo/blob/master/mutation-detection.cwl"

scm &lt;- compose_scm(scm_repository, scm_type, scm_commit, scm_path, scm_preview)

compose_extension(fhir, scm) %&gt;% convert_json()
</code></pre>

<hr>
<h2 id='compose_fhir_v1.4.2'>Compose BioCompute Object - FHIR Extension (v1.4.2)</h2><span id='topic+compose_fhir_v1.4.2'></span><span id='topic+compose_fhir'></span>

<h3>Description</h3>

<p>Compose BioCompute Object - FHIR Extension (v1.4.2)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compose_fhir_v1.4.2(endpoint = NULL, version = NULL, resources = NULL)

compose_fhir(endpoint = NULL, version = NULL, resources = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compose_fhir_v1.4.2_+3A_endpoint">endpoint</code></td>
<td>
<p>Character string. The URL of the endpoint of the
FHIR server containing the resource.</p>
</td></tr>
<tr><td><code id="compose_fhir_v1.4.2_+3A_version">version</code></td>
<td>
<p>Character string. The FHIR version used.</p>
</td></tr>
<tr><td><code id="compose_fhir_v1.4.2_+3A_resources">resources</code></td>
<td>
<p>Data frame with two variables: <code>id</code> and <code>resource</code>.
Each row is one item of resources to fetch from the endpoint.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of class <code>bco.domain</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fhir_endpoint &lt;- "https://fhirtest.uhn.ca/baseDstu3"
fhir_version &lt;- "3"
fhir_resources &lt;- data.frame(
  "id" = c("21376", "6288583", "25544", "92440", "4588936"),
  "resource" = c(
    "Sequence", "DiagnosticReport", "ProcedureRequest",
    "Observation", "FamilyMemberHistory"
  ),
  stringsAsFactors = FALSE
)

compose_fhir(fhir_endpoint, fhir_version, fhir_resources) %&gt;% convert_json()
</code></pre>

<hr>
<h2 id='compose_io_v1.4.2'>Compose BioCompute Object - Input and Output Domain (v1.4.2)</h2><span id='topic+compose_io_v1.4.2'></span><span id='topic+compose_io'></span>

<h3>Description</h3>

<p>This domain contains the list of global input and output files
created by the computational workflow, excluding the intermediate files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compose_io_v1.4.2(input = NULL, output = NULL)

compose_io(input = NULL, output = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compose_io_v1.4.2_+3A_input">input</code></td>
<td>
<p>Data frame. Variables include
<code>filename</code>, <code>uri</code>, and <code>access_time</code>.
Each row is one item in the input subdomain.</p>
</td></tr>
<tr><td><code id="compose_io_v1.4.2_+3A_output">output</code></td>
<td>
<p>Data frame. Variables include
<code>mediatype</code>, <code>uri</code>, and <code>access_time</code>.
Each row is one item in the output subdomain.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of class <code>bco.domain</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>input_subdomain &lt;- data.frame(
  "filename" = c(
    "Hepatitis C virus genotype 1",
    "Hepatitis C virus type 1b complete genome"
  ),
  "uri" = c(
    "https://www.ncbi.nlm.nih.gov/nuccore/22129792",
    "https://www.ncbi.nlm.nih.gov/nuccore/5420376"
  ),
  "access_time" = c(
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST")
  ),
  stringsAsFactors = FALSE
)

output_subdomain &lt;- data.frame(
  "mediatype" = c("text/csv", "text/csv"),
  "uri" = c(
    "https://example.com/data/514769/dnaAccessionBased.csv",
    "https://example.com/data/514801/SNPProfile*.csv"
  ),
  "access_time" = c(
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST")
  ),
  stringsAsFactors = FALSE
)

compose_io(input_subdomain, output_subdomain) %&gt;% convert_json()
</code></pre>

<hr>
<h2 id='compose_parametric_v1.4.2'>Compose BioCompute Object - Parametric Domain (v1.4.2)</h2><span id='topic+compose_parametric_v1.4.2'></span><span id='topic+compose_parametric'></span>

<h3>Description</h3>

<p>Non-default parameters customizing the computational flow
which can affect the output of the calculations
(<a href="https://github.com/biocompute-objects/BCO_Specification/blob/1.4.2/docs/parametric-domain.md">domain definition</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compose_parametric_v1.4.2(df = NULL)

compose_parametric(df = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compose_parametric_v1.4.2_+3A_df">df</code></td>
<td>
<p>Data frame. Variables include <code>param</code> (parameter names),
<code>value</code> (value of the parameters), and <code>step</code>
(step number for each parameter).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of class <code>bco.domain</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df_parametric &lt;- data.frame(
  "param" = c(
    "seed", "minimum_match_len",
    "divergence_threshold_percent",
    "minimum_coverage", "freq_cutoff"
  ),
  "value" = c("14", "66", "0.30", "15", "0.10"),
  "step" = c(1, 1, 1, 2, 2)
)

compose_parametric(df_parametric) %&gt;% convert_json()
</code></pre>

<hr>
<h2 id='compose_provenance_v1.4.2'>Compose BioCompute Object - Provenance Domain (v1.4.2)</h2><span id='topic+compose_provenance_v1.4.2'></span><span id='topic+compose_provenance'></span>

<h3>Description</h3>

<p>Compose BioCompute Object - Provenance Domain (v1.4.2)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compose_provenance_v1.4.2(
  name = NULL,
  version = NULL,
  review = NULL,
  derived_from = NULL,
  obsolete_after = NULL,
  embargo = NULL,
  created = NULL,
  modified = NULL,
  contributors = NULL,
  license = NULL
)

compose_provenance(
  name = NULL,
  version = NULL,
  review = NULL,
  derived_from = NULL,
  obsolete_after = NULL,
  embargo = NULL,
  created = NULL,
  modified = NULL,
  contributors = NULL,
  license = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compose_provenance_v1.4.2_+3A_name">name</code></td>
<td>
<p>Character string. Name for the BCO.</p>
</td></tr>
<tr><td><code id="compose_provenance_v1.4.2_+3A_version">version</code></td>
<td>
<p>Character string. Version of this BCO instance object.
Should follow the Semantic Versioning format (MAJOR.MINOR.PATCH).</p>
</td></tr>
<tr><td><code id="compose_provenance_v1.4.2_+3A_review">review</code></td>
<td>
<p>Data frame. Reviewer identifiers and descriptions
of the status of an object in the review process.</p>
</td></tr>
<tr><td><code id="compose_provenance_v1.4.2_+3A_derived_from">derived_from</code></td>
<td>
<p>Character string. Inheritance/derivation description.</p>
</td></tr>
<tr><td><code id="compose_provenance_v1.4.2_+3A_obsolete_after">obsolete_after</code></td>
<td>
<p>Date-time object. Expiration date of the object (optional).</p>
</td></tr>
<tr><td><code id="compose_provenance_v1.4.2_+3A_embargo">embargo</code></td>
<td>
<p>Vector of date-time objects <code>start_time</code> and <code>end_time</code>.
If the object has a period of time that it is not public,
that range can be specified with this.</p>
</td></tr>
<tr><td><code id="compose_provenance_v1.4.2_+3A_created">created</code></td>
<td>
<p>Date-time object. Initial creation time of the object.</p>
</td></tr>
<tr><td><code id="compose_provenance_v1.4.2_+3A_modified">modified</code></td>
<td>
<p>Date-time object. The most recent modification time of the object.</p>
</td></tr>
<tr><td><code id="compose_provenance_v1.4.2_+3A_contributors">contributors</code></td>
<td>
<p>Data frame. Contributor identifiers and descriptions
of their contribution types.</p>
</td></tr>
<tr><td><code id="compose_provenance_v1.4.2_+3A_license">license</code></td>
<td>
<p>Character string. Licence URL or other licence information (text).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of class <code>bco.domain</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>name &lt;- "HCV1a ledipasvir resistance SNP detection"
version &lt;- "1.0.0"
review &lt;- data.frame(
  "status" = c("approved", "approved"),
  "reviewer_comment" = c(
    "Approved by [company name] staff. Waiting for approval from FDA Reviewer",
    "The revised BCO looks fine"
  ),
  "date" = c(
    as.POSIXct("2017-11-12T12:30:48", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-12-12T12:30:48", format = "%Y-%m-%dT%H:%M:%S", tz = "America/Los_Angeles")
  ),
  "reviewer_name" = c("Jane Doe", "John Doe"),
  "reviewer_affiliation" = c("Seven Bridges Genomics", "U.S. Food and Drug Administration"),
  "reviewer_email" = c("example@sevenbridges.com", "example@fda.gov"),
  "reviewer_contribution" = c("curatedBy", "curatedBy"),
  "reviewer_orcid" = c("https://orcid.org/0000-0000-0000-0000", NA),
  stringsAsFactors = FALSE
)

derived_from &lt;- "https://github.com/biocompute-objects/BCO_Specification/blob/1.2.1-beta/HCV1a.json"
obsolete_after &lt;- as.POSIXct("2018-11-12T12:30:48", format = "%Y-%m-%dT%H:%M:%S", tz = "EST")

embargo &lt;- c(
  "start_time" = as.POSIXct("2017-10-12T12:30:48", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
  "end_time" = as.POSIXct("2017-11-12T12:30:48", format = "%Y-%m-%dT%H:%M:%S", tz = "EST")
)

created &lt;- as.POSIXct("2017-01-20T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST")

modified &lt;- as.POSIXct("2019-05-10T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST")

contributors &lt;- data.frame(
  "name" = c("Jane Doe", "John Doe"),
  "affiliation" = c("Seven Bridges Genomics", "U.S. Food and Drug Administration"),
  "email" = c("example@sevenbridges.com", "example@fda.gov"),
  "contribution" = I(list(c("createdBy", "curatedBy"), c("authoredBy"))),
  "orcid" = c("https://orcid.org/0000-0000-0000-0000", NA),
  stringsAsFactors = FALSE
)

license &lt;- "https://creativecommons.org/licenses/by/4.0/"

compose_provenance(
  name, version, review, derived_from, obsolete_after,
  embargo, created, modified, contributors, license
) %&gt;% convert_json()
</code></pre>

<hr>
<h2 id='compose_scm_v1.4.2'>Compose BioCompute Object - SCM Extension (v1.4.2)</h2><span id='topic+compose_scm_v1.4.2'></span><span id='topic+compose_scm'></span>

<h3>Description</h3>

<p>Compose BioCompute Object - SCM Extension (v1.4.2)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compose_scm_v1.4.2(
  scm_repository = NULL,
  scm_type = c("git", "svn", "hg", "other"),
  scm_commit = NULL,
  scm_path = NULL,
  scm_preview = NULL
)

compose_scm(
  scm_repository = NULL,
  scm_type = c("git", "svn", "hg", "other"),
  scm_commit = NULL,
  scm_path = NULL,
  scm_preview = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compose_scm_v1.4.2_+3A_scm_repository">scm_repository</code></td>
<td>
<p>Character string. Base URL of the SCM repository.</p>
</td></tr>
<tr><td><code id="compose_scm_v1.4.2_+3A_scm_type">scm_type</code></td>
<td>
<p>Character string. Type of SCM database. Must be one of
<code>"git"</code>, <code>"svn"</code>, <code>"hg"</code>, or <code>"other"</code>.</p>
</td></tr>
<tr><td><code id="compose_scm_v1.4.2_+3A_scm_commit">scm_commit</code></td>
<td>
<p>Character string. Revision within the SCM repository.
Should be a repository-wide commit identifier or name of a tag,
but may be a name of a branch.</p>
</td></tr>
<tr><td><code id="compose_scm_v1.4.2_+3A_scm_path">scm_path</code></td>
<td>
<p>Character string. Path from the repository to the
source code referenced. Should not start with <code>/</code>.</p>
</td></tr>
<tr><td><code id="compose_scm_v1.4.2_+3A_scm_preview">scm_preview</code></td>
<td>
<p>Character string. The full URI for the source code
referenced by the BioCompute Object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of class <code>bco.domain</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>scm_repository &lt;- "https://github.com/example/repo"
scm_type &lt;- "git"
scm_commit &lt;- "c9ffea0b60fa3bcf8e138af7c99ca141a6b8fb21"
scm_path &lt;- "workflow/hive-viral-mutation-detection.cwl"
scm_preview &lt;- "https://github.com/example/repo/blob/master/mutation-detection.cwl"

compose_scm(scm_repository, scm_type, scm_commit, scm_path, scm_preview) %&gt;% convert_json()
</code></pre>

<hr>
<h2 id='compose_tlf_v1.4.2'>Compose BioCompute Object - Top Level Fields (v1.4.2)</h2><span id='topic+compose_tlf_v1.4.2'></span><span id='topic+compose_tlf'></span>

<h3>Description</h3>

<p>Compose BioCompute Object - Top Level Fields (v1.4.2)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compose_tlf_v1.4.2(
  provenance,
  usability,
  extension,
  description,
  execution,
  parametric,
  io,
  error,
  object_id = NULL
)

compose_tlf(
  provenance,
  usability,
  extension,
  description,
  execution,
  parametric,
  io,
  error,
  object_id = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compose_tlf_v1.4.2_+3A_provenance">provenance</code></td>
<td>
<p>Provenance domain</p>
</td></tr>
<tr><td><code id="compose_tlf_v1.4.2_+3A_usability">usability</code></td>
<td>
<p>Usability domain</p>
</td></tr>
<tr><td><code id="compose_tlf_v1.4.2_+3A_extension">extension</code></td>
<td>
<p>Extension domain</p>
</td></tr>
<tr><td><code id="compose_tlf_v1.4.2_+3A_description">description</code></td>
<td>
<p>Description domain</p>
</td></tr>
<tr><td><code id="compose_tlf_v1.4.2_+3A_execution">execution</code></td>
<td>
<p>Execution domain</p>
</td></tr>
<tr><td><code id="compose_tlf_v1.4.2_+3A_parametric">parametric</code></td>
<td>
<p>Parametric domain</p>
</td></tr>
<tr><td><code id="compose_tlf_v1.4.2_+3A_io">io</code></td>
<td>
<p>I/O domain</p>
</td></tr>
<tr><td><code id="compose_tlf_v1.4.2_+3A_error">error</code></td>
<td>
<p>Error domain</p>
</td></tr>
<tr><td><code id="compose_tlf_v1.4.2_+3A_object_id">object_id</code></td>
<td>
<p>BioCompute Object identifier
(<a href="https://github.com/biocompute-objects/BCO_Specification/blob/1.4.2/docs/top-level.md#202-biocompute-object-identifier-object_id">definition</a>).
If <code>NULL</code>, will use a UUID generated by <code><a href="#topic+generate_id">generate_id</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of top level fields
</p>


<h3>Examples</h3>

<pre><code class='language-R'>compose_tlf(
  compose_provenance(), compose_usability(), compose_extension(),
  compose_description(), compose_execution(), compose_parametric(),
  compose_io(), compose_error()
) %&gt;% convert_json()
</code></pre>

<hr>
<h2 id='compose_usability_v1.4.2'>Compose BioCompute Object - Usability Domain (v1.4.2)</h2><span id='topic+compose_usability_v1.4.2'></span><span id='topic+compose_usability'></span>

<h3>Description</h3>

<p>The usability domain
(<a href="https://github.com/biocompute-objects/BCO_Specification/blob/1.4.2/docs/usability-domain.md">domain definition</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compose_usability_v1.4.2(text = NULL)

compose_usability(text = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compose_usability_v1.4.2_+3A_text">text</code></td>
<td>
<p>A character vector of free text values that could
improves search-ability, provide specific scientific use cases,
and a description of the function of the object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of class <code>bco.domain</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>text &lt;- c(
  paste(
    "Identify baseline single nucleotide polymorphisms (SNPs)[SO:0000694]",
    "(insertions)[SO:0000667], and (deletions)[SO:0000045] that correlate",
    "with reduced (ledipasvir)[pubchem.compound:67505836] antiviral drug",
    "efficacy in (Hepatitis C virus subtype 1)[taxonomy:31646]"
  ),
  paste(
    "Identify treatment emergent amino acid (substitutions)[SO:1000002]",
    "that correlate with antiviral drug treatment failure"
  ),
  paste(
    "Determine whether the treatment emergent amino acid",
    "(substitutions)[SO:1000002] identified correlate with treatment",
    "failure involving other drugs against the same virus"
  )
)

text %&gt;%
  compose_usability() %&gt;%
  convert_json()
</code></pre>

<hr>
<h2 id='compose_v1.4.2'>Compose BioCompute Object (v1.4.2)</h2><span id='topic+compose_v1.4.2'></span><span id='topic+compose'></span>

<h3>Description</h3>

<p>Compose BioCompute Object (v1.4.2)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compose_v1.4.2(
  tlf,
  provenance,
  usability,
  extension,
  description,
  execution,
  parametric,
  io,
  error
)

compose(
  tlf,
  provenance,
  usability,
  extension,
  description,
  execution,
  parametric,
  io,
  error
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compose_v1.4.2_+3A_tlf">tlf</code></td>
<td>
<p>Top level fields</p>
</td></tr>
<tr><td><code id="compose_v1.4.2_+3A_provenance">provenance</code></td>
<td>
<p>Provenance domain</p>
</td></tr>
<tr><td><code id="compose_v1.4.2_+3A_usability">usability</code></td>
<td>
<p>Usability domain</p>
</td></tr>
<tr><td><code id="compose_v1.4.2_+3A_extension">extension</code></td>
<td>
<p>Extension domain</p>
</td></tr>
<tr><td><code id="compose_v1.4.2_+3A_description">description</code></td>
<td>
<p>Description domain</p>
</td></tr>
<tr><td><code id="compose_v1.4.2_+3A_execution">execution</code></td>
<td>
<p>Execution domain</p>
</td></tr>
<tr><td><code id="compose_v1.4.2_+3A_parametric">parametric</code></td>
<td>
<p>Parametric domain</p>
</td></tr>
<tr><td><code id="compose_v1.4.2_+3A_io">io</code></td>
<td>
<p>I/O domain</p>
</td></tr>
<tr><td><code id="compose_v1.4.2_+3A_error">error</code></td>
<td>
<p>Error domain</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of class <code>bco</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tlf &lt;- compose_tlf(
  compose_provenance(), compose_usability(), compose_extension(),
  compose_description(), compose_execution(), compose_parametric(),
  compose_io(), compose_error()
)
biocompute::compose(
  tlf,
  compose_provenance(), compose_usability(), compose_extension(),
  compose_description(), compose_execution(), compose_parametric(),
  compose_io(), compose_error()
) %&gt;% convert_json()
</code></pre>

<hr>
<h2 id='convert_json'>Convert BioCompute Object or domain to JSON string</h2><span id='topic+convert_json'></span>

<h3>Description</h3>

<p>Convert BioCompute Object or domain to JSON string
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_json(x, pretty = TRUE, auto_unbox = TRUE, na = "string", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_json_+3A_x">x</code></td>
<td>
<p>BioCompute Object or domain</p>
</td></tr>
<tr><td><code id="convert_json_+3A_pretty">pretty</code></td>
<td>
<p>Prettify the JSON string? Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="convert_json_+3A_auto_unbox">auto_unbox</code></td>
<td>
<p>Unbox all atomic vectors of length 1? Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="convert_json_+3A_na">na</code></td>
<td>
<p>How to represent NA values: must be <code>"null"</code> or <code>"string"</code>.
Default is <code>"string"</code>.</p>
</td></tr>
<tr><td><code id="convert_json_+3A_...">...</code></td>
<td>
<p>Additional parameters for <code><a href="jsonlite.html#topic+toJSON">toJSON</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>JSON string of the BioCompute Object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>compose_description() %&gt;% convert_json()
generate_example("minimal") %&gt;% convert_json()
</code></pre>

<hr>
<h2 id='convert_yaml'>Convert BioCompute Object or domain to YAML string</h2><span id='topic+convert_yaml'></span>

<h3>Description</h3>

<p>Convert BioCompute Object or domain to YAML string
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_yaml(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_yaml_+3A_x">x</code></td>
<td>
<p>BioCompute Object or domain</p>
</td></tr>
<tr><td><code id="convert_yaml_+3A_...">...</code></td>
<td>
<p>Additional parameters for <code><a href="yaml.html#topic+as.yaml">as.yaml</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>YAML string of the BioCompute Object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>compose_description() %&gt;%
  convert_yaml() %&gt;%
  cat()
generate_example("minimal") %&gt;%
  convert_yaml() %&gt;%
  cat()
</code></pre>

<hr>
<h2 id='export_html'>Export BioCompute Object as HTML</h2><span id='topic+export_html'></span>

<h3>Description</h3>

<p>Export BioCompute Object as HTML
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_html(x, file, wrap = FALSE, linewidth = 80, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export_html_+3A_x">x</code></td>
<td>
<p>BioCompute Object JSON string from <code><a href="#topic+convert_json">convert_json</a></code></p>
</td></tr>
<tr><td><code id="export_html_+3A_file">file</code></td>
<td>
<p>HTML output file path</p>
</td></tr>
<tr><td><code id="export_html_+3A_wrap">wrap</code></td>
<td>
<p>Should the long lines be wrapped?</p>
</td></tr>
<tr><td><code id="export_html_+3A_linewidth">linewidth</code></td>
<td>
<p>Maximum linewidth when <code>wrap</code> is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="export_html_+3A_...">...</code></td>
<td>
<p>Additional parameters for <code><a href="rmarkdown.html#topic+render">render</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Path to the output file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
file_html &lt;- tempfile(fileext = ".html")
generate_example("HCV1a") %&gt;%
  convert_json() %&gt;%
  export_html(file_html)

## End(Not run)
</code></pre>

<hr>
<h2 id='export_json'>Export BioCompute Object as JSON</h2><span id='topic+export_json'></span>

<h3>Description</h3>

<p>Export BioCompute Object as JSON
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_json(x, file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export_json_+3A_x">x</code></td>
<td>
<p>BioCompute Object JSON string from <code><a href="#topic+convert_json">convert_json</a></code></p>
</td></tr>
<tr><td><code id="export_json_+3A_file">file</code></td>
<td>
<p>JSON file path</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Path to the output file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_json &lt;- tempfile(fileext = ".json")
generate_example("HCV1a") %&gt;%
  convert_json() %&gt;%
  export_json(file_json)
cat(paste(readLines(file_json), collapse = "\n"))
</code></pre>

<hr>
<h2 id='export_pdf'>Export BioCompute Object as PDF</h2><span id='topic+export_pdf'></span>

<h3>Description</h3>

<p>Export BioCompute Object as PDF
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_pdf(x, file, wrap = FALSE, linewidth = 80, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export_pdf_+3A_x">x</code></td>
<td>
<p>BioCompute Object JSON string from <code><a href="#topic+convert_json">convert_json</a></code></p>
</td></tr>
<tr><td><code id="export_pdf_+3A_file">file</code></td>
<td>
<p>PDF output file path</p>
</td></tr>
<tr><td><code id="export_pdf_+3A_wrap">wrap</code></td>
<td>
<p>Should the long lines be wrapped?</p>
</td></tr>
<tr><td><code id="export_pdf_+3A_linewidth">linewidth</code></td>
<td>
<p>Maximum linewidth when <code>wrap</code> is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="export_pdf_+3A_...">...</code></td>
<td>
<p>Additional parameters for <code><a href="rmarkdown.html#topic+render">render</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Path to the output file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
file_pdf &lt;- tempfile(fileext = ".pdf")
generate_example("HCV1a") %&gt;%
  convert_json() %&gt;%
  export_pdf(file_pdf)

## End(Not run)
</code></pre>

<hr>
<h2 id='export_sevenbridges'>Export BioCompute Object to Seven Bridges Platforms</h2><span id='topic+export_sevenbridges'></span>

<h3>Description</h3>

<p>Export BioCompute Object to Seven Bridges Platforms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_sevenbridges(
  file,
  name = NULL,
  project = NULL,
  token = NULL,
  base_url = "https://api.sbgenomics.com/v2/",
  overwrite = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export_sevenbridges_+3A_file">file</code></td>
<td>
<p>Path to the BCO file.</p>
</td></tr>
<tr><td><code id="export_sevenbridges_+3A_name">name</code></td>
<td>
<p>Name of the BCO file to create on the platform.
Defaults to the name of the input file.</p>
</td></tr>
<tr><td><code id="export_sevenbridges_+3A_project">project</code></td>
<td>
<p>Project to upload (export) the BCO file to.
Format: <code>"username/project"</code>.</p>
</td></tr>
<tr><td><code id="export_sevenbridges_+3A_token">token</code></td>
<td>
<p>API auth token for the platform.
Generate the token from the platform's Developer Dashboard.</p>
</td></tr>
<tr><td><code id="export_sevenbridges_+3A_base_url">base_url</code></td>
<td>
<p>API base URL.
Get the base URL from the platform's Developer Dashboard.</p>
</td></tr>
<tr><td><code id="export_sevenbridges_+3A_overwrite">overwrite</code></td>
<td>
<p>If <code>TRUE</code>, will overwrite the existing
BCO file with the same name in that project (if any).
If <code>FALSE</code>, will not overwrite.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Response of the file upload request
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
file_json &lt;- tempfile(fileext = ".json")
generate_example("HCV1a") %&gt;%
  convert_json() %&gt;%
  export_json(file_json)

try(
  export_sevenbridges(
    file_json,
    project = "rosalind_franklin/project_name",
    token = "your_api_auth_token",
    base_url = "https://cgc-api.sbgenomics.com/v2/"
  )
)

## End(Not run)
</code></pre>

<hr>
<h2 id='export_word'>Export BioCompute Object as Word document</h2><span id='topic+export_word'></span>

<h3>Description</h3>

<p>Export BioCompute Object as Word document
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_word(x, file, wrap = FALSE, linewidth = 80, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export_word_+3A_x">x</code></td>
<td>
<p>BioCompute Object JSON string from <code><a href="#topic+convert_json">convert_json</a></code></p>
</td></tr>
<tr><td><code id="export_word_+3A_file">file</code></td>
<td>
<p>Word (docx) output file path</p>
</td></tr>
<tr><td><code id="export_word_+3A_wrap">wrap</code></td>
<td>
<p>Should the long lines be wrapped?</p>
</td></tr>
<tr><td><code id="export_word_+3A_linewidth">linewidth</code></td>
<td>
<p>Maximum linewidth when <code>wrap</code> is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="export_word_+3A_...">...</code></td>
<td>
<p>Additional parameters for <code><a href="rmarkdown.html#topic+render">render</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Path to the output file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
file_docx &lt;- tempfile(fileext = ".docx")
generate_example("HCV1a") %&gt;%
  convert_json() %&gt;%
  export_word(file_docx)

## End(Not run)
</code></pre>

<hr>
<h2 id='generate_example'>Generate example BioCompute Objects</h2><span id='topic+generate_example'></span>

<h3>Description</h3>

<p>Generate example BioCompute Objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_example(type = c("minimal", "HCV1a"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="generate_example_+3A_type">type</code></td>
<td>
<p>Example type. Default is <code>"minimal"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Example BioCompute Object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>generate_example("minimal") %&gt;% convert_json()
</code></pre>

<hr>
<h2 id='generate_id'>Generate ID for the BioCompute Object</h2><span id='topic+generate_id'></span>

<h3>Description</h3>

<p>Generate ID for the BioCompute Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_id(platform = c("sevenbridges"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="generate_id_+3A_platform">platform</code></td>
<td>
<p>Platform. Default is <code>"sevenbridges"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>BioCompute Object ID
</p>


<h3>Examples</h3>

<pre><code class='language-R'>generate_id()
</code></pre>

<hr>
<h2 id='is_bco'>Is this a BCO object?</h2><span id='topic+is_bco'></span>

<h3>Description</h3>

<p>Is this a BCO object?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_bco(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is_bco_+3A_x">x</code></td>
<td>
<p>any object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Logical. <code>TRUE</code> if it is a BCO object, <code>FALSE</code> if not.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>generate_example("minimal") %&gt;% is_bco()
</code></pre>

<hr>
<h2 id='is_domain'>Is this a domain object?</h2><span id='topic+is_domain'></span>

<h3>Description</h3>

<p>Is this a domain object?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_domain(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is_domain_+3A_x">x</code></td>
<td>
<p>any object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Logical. <code>TRUE</code> if it is a domain object, <code>FALSE</code> if not.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>is_domain(compose_description())
</code></pre>

<hr>
<h2 id='read_bco'>Parse Biocompute Object From JSON File to R Object</h2><span id='topic+read_bco'></span>

<h3>Description</h3>

<p>Parse Biocompute Object From JSON File to R Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_bco(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_bco_+3A_x">x</code></td>
<td>
<p>BioCompute Object .json file</p>
</td></tr>
<tr><td><code id="read_bco_+3A_...">...</code></td>
<td>
<p>Additional parameters for <code><a href="jsonlite.html#topic+fromJSON">fromJSON</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of class <code>bco</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bco &lt;- tempfile(fileext = ".json")
bco &lt;- generate_example("HCV1a") %&gt;%
  convert_json() %&gt;%
  export_json(bco)
bco %&gt;% read_bco()
</code></pre>

<hr>
<h2 id='validate_checksum_v1.4.2'>BioCompute Objects checksum validator (v1.4.2)</h2><span id='topic+validate_checksum_v1.4.2'></span><span id='topic+validate_checksum'></span>

<h3>Description</h3>

<p>BioCompute Objects checksum validator (v1.4.2)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate_checksum_v1.4.2(file)

validate_checksum(file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate_checksum_v1.4.2_+3A_file">file</code></td>
<td>
<p>Path to the BCO JSON file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Logical. <code>TRUE</code> if the checksum matched, <code>FALSE</code> if not.
</p>


<h3>Note</h3>

<p>An SHA-256 checksum is
<a href="https://github.com/biocompute-objects/BCO_Specification/blob/1.4.2/docs/top-level.md#203-etag-etag">calculated and stored</a>
in the top level fields when a BioCompute Object is created. In reality,
due to the delicate differences in how the data in JSON is represented,
parsed, and handled in different languages, there could be false positives
in the validation results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bco &lt;- tempfile(fileext = ".json")
generate_example("HCV1a") %&gt;%
  convert_json() %&gt;%
  export_json(bco)
bco %&gt;% validate_checksum()
</code></pre>

<hr>
<h2 id='validate_schema_v1.4.2'>BioCompute Objects schema validator (v1.4.2)</h2><span id='topic+validate_schema_v1.4.2'></span><span id='topic+validate_schema'></span>

<h3>Description</h3>

<p>BioCompute Objects schema validator (v1.4.2)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate_schema_v1.4.2(file)

validate_schema(file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate_schema_v1.4.2_+3A_file">file</code></td>
<td>
<p>Path to the BCO JSON file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>Note</h3>

<p>JSON schema validators for BCO domains and complete BCO based on
jsonvalidate. Refer to the
<a href="https://github.com/biocompute-objects/BCO_Specification/tree/1.4.2/ieee-2791-schema">BioCompute Objects Schema</a>
for specific JSON schemas.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bco &lt;- tempfile(fileext = ".json")
generate_example("HCV1a") %&gt;%
  convert_json() %&gt;%
  export_json(bco)
bco %&gt;% validate_schema()
</code></pre>

<hr>
<h2 id='versions'>BioCompute Object specification versions</h2><span id='topic+versions'></span>

<h3>Description</h3>

<p>BioCompute Object specification versions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>versions()
</code></pre>


<h3>Value</h3>

<p>List of current and all available BioCompute Object
specification versions supported by the package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>versions()
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
