<!DOCTYPE html><html><head><title>Help for package BiocGenerics</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BiocGenerics}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#annotation'><p>Accessing annotation information</p></a></li>
<li><a href='#aperm'><p>Transposing an array-like object</p></a></li>
<li><a href='#append'><p>Append elements to a vector-like object</p></a></li>
<li><a href='#as.data.frame'><p>Coerce to a data frame</p></a></li>
<li><a href='#as.list'><p>Coerce to a list</p></a></li>
<li><a href='#as.vector'><p>Coerce an object into a vector</p></a></li>
<li><a href='#BiocGenerics-package'><p>S4 generic functions for Bioconductor</p></a></li>
<li><a href='#boxplot'><p>Box plots</p></a></li>
<li><a href='#cbind'><p>Combine objects by rows or columns</p></a></li>
<li><a href='#combine'><p>Combining or merging different Bioconductor data structures</p></a></li>
<li><a href='#dbconn'><p>Accessing SQLite DB information</p></a></li>
<li><a href='#density'><p>Kernel density estimation</p></a></li>
<li><a href='#dge'><p>Accessors and generic functions used in the context of count datasets</p></a></li>
<li><a href='#dims'><p>Get the dimensions of each element of a list-like object</p></a></li>
<li><a href='#do.call'><p>Execute a function call</p></a></li>
<li><a href='#duplicated'><p>Determine duplicate elements</p></a></li>
<li><a href='#eval'><p>Evaluate an (unevaluated) expression</p></a></li>
<li><a href='#evalq'><p>Evaluate an (unevaluated) expression</p></a></li>
<li><a href='#Extremes'><p>Maxima and minima</p></a></li>
<li><a href='#fileName'><p>Accessing the file name of an object</p></a></li>
<li><a href='#format'><p>Format an R object for pretty printing</p></a></li>
<li><a href='#funprog'><p>Common higher-order functions in functional programming languages</p></a></li>
<li><a href='#get'><p>Return the value of a named object</p></a></li>
<li><a href='#grep'><p>Pattern Matching and Replacement</p></a></li>
<li><a href='#image'><p>Display a color image</p></a></li>
<li><a href='#IQR'><p>The Interquartile Range</p></a></li>
<li><a href='#is.unsorted'><p>Test if a vector-like object is not sorted</p></a></li>
<li><a href='#lapply'><p>Apply a function over a list-like or vector-like object</p></a></li>
<li><a href='#mad'><p>Median Absolute Deviation</p></a></li>
<li><a href='#mapply'><p>Apply a function to multiple list-like or vector-like arguments</p></a></li>
<li><a href='#match'><p>Value matching</p></a></li>
<li><a href='#mean'><p>Arithmetic Mean</p></a></li>
<li><a href='#normalize'><p>Normalize an object</p></a></li>
<li><a href='#nrow'><p>The number of rows/columns of an array-like object</p></a></li>
<li><a href='#Ontology'><p>Generic Ontology getter</p></a></li>
<li><a href='#order'><p>Ordering permutation</p></a></li>
<li><a href='#organism_species'><p>Organism and species accessors</p></a></li>
<li><a href='#paste'><p>Concatenate strings</p></a></li>
<li><a href='#path'><p>Accessing the path of an object</p></a></li>
<li><a href='#plotMA'><p>MA-plot: plot differences versus averages for high-throughput data</p></a></li>
<li><a href='#plotPCA'><p>PCA-plot: Principal Component Analysis plot</p></a></li>
<li><a href='#rank'><p>Ranks the values in a vector-like object</p></a></li>
<li><a href='#relist'><p>Re-listing an unlist()ed object</p></a></li>
<li><a href='#rep'><p>Replicate elements of a vector-like object</p></a></li>
<li><a href='#residuals'><p>Extract model residuals</p></a></li>
<li><a href='#row+colnames'><p>Row and column names</p></a></li>
<li><a href='#S3-classes-as-S4-classes'><p>S3 classes as S4 classes</p></a></li>
<li><a href='#score'><p>Score accessor</p></a></li>
<li><a href='#sets'><p>Set operations</p></a></li>
<li><a href='#sort'><p>Sorting a vector-like object</p></a></li>
<li><a href='#start'><p>The start(), end(), width(), and pos() generic getters and setters</p></a></li>
<li><a href='#strand'><p>Accessing strand information</p></a></li>
<li><a href='#subset'><p>Subsetting vector-like, matrix-like and data-frame-like objects</p></a></li>
<li><a href='#t'><p>Matrix Transpose</p></a></li>
<li><a href='#table'><p>Cross tabulation and table creation</p></a></li>
<li><a href='#tapply'><p>Apply a function over a ragged array</p></a></li>
<li><a href='#testPackage'><p>Run RUnit package unit tests</p></a></li>
<li><a href='#toTable'><p>An alternative to as.data.frame()</p></a></li>
<li><a href='#type'><p>Accessing the type of an object</p></a></li>
<li><a href='#unique'><p>Extract unique elements</p></a></li>
<li><a href='#unlist'><p>Flatten list-like objects</p></a></li>
<li><a href='#unsplit'><p>Unsplit a list-like object</p></a></li>
<li><a href='#updateObject'><p>Update an object to its current class definition</p></a></li>
<li><a href='#var'><p>Variance and Standard Deviation</p></a></li>
<li><a href='#weights'><p>Extract model weights</p></a></li>
<li><a href='#which'><p>Which values in an object are considered TRUE?</p></a></li>
<li><a href='#which.min'><p>What's the index of the first min or max value in an object?</p></a></li>
<li><a href='#xtabs'><p>Cross tabulation</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>S4 generic functions used in Bioconductor</td>
</tr>
<tr>
<td>Description:</td>
<td>The package defines many S4 generic functions used in Bioconductor.</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Infrastructure</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://bioconductor.org/packages/BiocGenerics">https://bioconductor.org/packages/BiocGenerics</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Bioconductor/BiocGenerics/issues">https://github.com/Bioconductor/BiocGenerics/issues</a></td>
</tr>
<tr>
<td>Version:</td>
<td>0.49.1</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Author:</td>
<td>The Bioconductor Dev Team</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hervé Pagès &lt;hpages.on.github@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0), methods, utils, graphics, stats</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, utils, graphics, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>Biobase, S4Vectors, IRanges, GenomicRanges, DelayedArray,
Biostrings, Rsamtools, AnnotationDbi, affy, affyPLM, DESeq2,
flowClust, MSnbase, annotate, RUnit</td>
</tr>
<tr>
<td>Collate:</td>
<td>S3-classes-as-S4-classes.R utils.R normarg-utils.R
replaceSlots.R aperm.R append.R as.data.frame.R as.list.R
as.vector.R cbind.R do.call.R duplicated.R eval.R Extremes.R
format.R funprog.R get.R grep.R is.unsorted.R lapply.R mapply.R
match.R mean.R nrow.R order.R paste.R rank.R rep.R
row_colnames.R sets.R sort.R start.R subset.R t.R table.R
tapply.R unique.R unlist.R unsplit.R relist.R var.R which.R
which.min.R boxplot.R image.R density.R IQR.R mad.R residuals.R
weights.R xtabs.R annotation.R combine.R dbconn.R dge.R dims.R
fileName.R normalize.R Ontology.R organism_species.R path.R
plotMA.R plotPCA.R score.R strand.R toTable.R type.R
updateObject.R testPackage.R zzz.R</td>
</tr>
<tr>
<td>git_url:</td>
<td><a href="https://git.bioconductor.org/packages/BiocGenerics">https://git.bioconductor.org/packages/BiocGenerics</a></td>
</tr>
<tr>
<td>git_branch:</td>
<td>devel</td>
</tr>
<tr>
<td>git_last_commit:</td>
<td>2e1a2a6</td>
</tr>
<tr>
<td>git_last_commit_date:</td>
<td>2023-10-31</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-01</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-01 20:01:09 UTC; biocbuild</td>
</tr>
<tr>
<td>Built:</td>
<td>R 4.4.0; ; 2024-01-02 07:13:08 UTC; unix</td>
</tr>
</table>
<hr>
<h2 id='annotation'>Accessing annotation information</h2><span id='topic+annotation'></span><span id='topic+annotation+3C-'></span>

<h3>Description</h3>

<p>Get or set the annotation information contained in an object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annotation(object, ...)
annotation(object, ...) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annotation_+3A_object">object</code></td>
<td>

<p>An object containing annotation information.
</p>
</td></tr>
<tr><td><code id="annotation_+3A_...">...</code></td>
<td>

<p>Additional arguments, for use in specific methods.
</p>
</td></tr>
<tr><td><code id="annotation_+3A_value">value</code></td>
<td>

<p>The annotation information to set on <code>object</code>.
</p>
</td></tr>
</table>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="Biobase.html#topic+annotation+2CeSet-method">annotation,eSet-method</a> in the <span class="pkg">Biobase</span> package
for an example of a specific <code>annotation</code> method (defined for
<a href="Biobase.html#topic+eSet">eSet</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>annotation
showMethods("annotation")

library(Biobase)
showMethods("annotation")
selectMethod("annotation", "eSet")
</code></pre>

<hr>
<h2 id='aperm'>Transposing an array-like object</h2><span id='topic+aperm'></span>

<h3>Description</h3>

<p>Transpose an array-like object by permuting its dimensions.
</p>
<p>This is a multidimensional generalization of the
<code><a href="#topic+t">t</a>()</code> operator used for 2D-transposition.
</p>
<p>NOTE: This man page is for the <code>aperm</code>
<em>S4 generic function</em> defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+aperm">aperm</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aperm(a, perm, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aperm_+3A_a">a</code></td>
<td>

<p>An array-like object.
</p>
</td></tr>
<tr><td><code id="aperm_+3A_perm">perm</code>, <code id="aperm_+3A_...">...</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+aperm">aperm</a></code> for a description of
these arguments.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A transposed version of array-like object <code>a</code>, with subscripts
permuted as indicated by the <code>perm</code> vector.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+aperm">aperm</a></code> for the default
<code>aperm</code> method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="DelayedArray.html#topic+aperm+2CDelayedArray-method">aperm,DelayedArray-method</a> in the
<span class="pkg">DelayedArray</span> package for an example of a specific
<code>aperm</code> method (defined for <a href="DelayedArray.html#topic+DelayedArray">DelayedArray</a>
objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>aperm  # note the dispatch on the 'a' arg only
showMethods("aperm")
selectMethod("aperm", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='append'>Append elements to a vector-like object</h2><span id='topic+append'></span>

<h3>Description</h3>

<p>Append (or insert) elements to (in) a vector-like object.
</p>
<p>NOTE: This man page is for the <code>append</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+append">append</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
(typically vector-like or data-frame-like) not supported by the
default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>append(x, values, after=length(x))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="append_+3A_x">x</code></td>
<td>

<p>The vector-like object to be modified.
</p>
</td></tr>
<tr><td><code id="append_+3A_values">values</code></td>
<td>

<p>The vector-like object containing the values to be appended to <code>x</code>.
<code>values</code> would typically be of the same class as <code>x</code>, but not
necessarily.
</p>
</td></tr>
<tr><td><code id="append_+3A_after">after</code></td>
<td>

<p>A subscript, after which the values are to be appended.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+append">append</a></code> for the value returned by the
default method.
</p>
<p>Specific methods defined in Bioconductor packages will typically
return an object of the same class as <code>x</code> and of length
<code>length(x) + length(values)</code>.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+append">append</a></code> for the default <code>append</code>
method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+append+2CVector+2CVector-method">append,Vector,Vector-method</a> in the <span class="pkg">S4Vectors</span>
package for an example of a specific <code>append</code> method (defined
for <a href="S4Vectors.html#topic+Vector">Vector</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>append  # note the dispatch on the 'x' and 'values' args only
showMethods("append")
selectMethod("append", c("ANY", "ANY"))  # the default method
</code></pre>

<hr>
<h2 id='as.data.frame'>Coerce to a data frame</h2><span id='topic+as.data.frame'></span>

<h3>Description</h3>

<p>Generic function to coerce to a data frame, if possible.
</p>
<p>NOTE: This man page is for the <code>as.data.frame</code>
<em>S4 generic function</em> defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+as.data.frame">as.data.frame</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.data.frame(x, row.names=NULL, optional=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.data.frame_+3A_x">x</code></td>
<td>

<p>The object to coerce.
</p>
</td></tr>
<tr><td><code id="as.data.frame_+3A_row.names">row.names</code>, <code id="as.data.frame_+3A_optional">optional</code>, <code id="as.data.frame_+3A_...">...</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+as.data.frame">as.data.frame</a></code> for a description of
these arguments.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An ordinary data frame.
</p>
<p>See <code>?base::<a href="base.html#topic+as.data.frame">as.data.frame</a></code> for the value returned
by the default method.
</p>
<p>Specific methods defined in Bioconductor packages should
behave as consistently as possible with the default method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+as.data.frame">as.data.frame</a></code> for the default
<code>as.data.frame</code> method.
</p>
</li>
<li> <p><code><a href="#topic+toTable">toTable</a></code> for an alternative to
<code>as.data.frame</code>.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+as.data.frame+2CDataFrame-method">as.data.frame,DataFrame-method</a> in the
<span class="pkg">S4Vectors</span> package, and
<a href="IRanges.html#topic+as.data.frame+2CIntegerRanges-method">as.data.frame,IntegerRanges-method</a> in the
<span class="pkg">IRanges</span> package, for examples of specific <code>as.data.frame</code>
methods (defined for <a href="S4Vectors.html#topic+DataFrame">DataFrame</a> and
<a href="IRanges.html#topic+IntegerRanges">IntegerRanges</a> objects, respectively).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>as.data.frame  # note the dispatch on the 'x' arg only
showMethods("as.data.frame")
selectMethod("as.data.frame", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='as.list'>Coerce to a list</h2><span id='topic+as.list'></span>

<h3>Description</h3>

<p>Generic function to coerce to a list, if possible.
</p>
<p>NOTE: This man page is for the <code>as.list</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+as.list">as.list</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.list(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.list_+3A_x">x</code></td>
<td>

<p>The object to coerce.
</p>
</td></tr>
<tr><td><code id="as.list_+3A_...">...</code></td>
<td>

<p>Additional arguments, for use in specific methods.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An ordinary list.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+as.list">as.list</a></code> for the default <code>as.list</code>
method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+as.list+2CList-method">as.list,List-method</a> in the <span class="pkg">S4Vectors</span>
package for an example of a specific <code>as.list</code> method
(defined for <a href="S4Vectors.html#topic+List">List</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>as.list
showMethods("as.list")
selectMethod("as.list", "ANY")  # the default method

library(S4Vectors)
showMethods("as.list")
## The as.list() method for List objects:
selectMethod("as.list", "List")
</code></pre>

<hr>
<h2 id='as.vector'>Coerce an object into a vector</h2><span id='topic+as.vector'></span>

<h3>Description</h3>

<p>Attempt to coerce an object into a vector of the specified mode.
If the mode is not specified, attempt to coerce to whichever vector
mode is considered more appropriate for the class of the supplied
object.
</p>
<p>NOTE: This man page is for the <code>as.vector</code>
<em>S4 generic function</em> defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+as.vector">as.vector</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.vector(x, mode="any")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.vector_+3A_x">x</code></td>
<td>

<p>The object to coerce.
</p>
</td></tr>
<tr><td><code id="as.vector_+3A_mode">mode</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+as.vector">as.vector</a></code> for a description of
this argument.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector.
</p>
<p>See <code>?base::<a href="base.html#topic+as.vector">as.vector</a></code> for the value returned
by the default method.
</p>
<p>Specific methods defined in Bioconductor packages should
behave as consistently as possible with the default method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+as.vector">as.vector</a></code> for the default
<code>as.vector</code> method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+as.vector+2CRle-method">as.vector,Rle-method</a> in the <span class="pkg">S4Vectors</span>
package, and <a href="IRanges.html#topic+as.vector+2CAtomicList-method">as.vector,AtomicList-method</a> in
the <span class="pkg">IRanges</span> packages, for examples of specific
<code>as.vector</code> methods (defined for <a href="S4Vectors.html#topic+Rle">Rle</a>
and <a href="IRanges.html#topic+AtomicList">AtomicList</a> objects, respectively).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>as.vector  # note the dispatch on the 'x' arg only
showMethods("as.vector")
selectMethod("as.vector", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='BiocGenerics-package'>S4 generic functions for Bioconductor</h2><span id='topic+BiocGenerics-package'></span><span id='topic+BiocGenerics'></span>

<h3>Description</h3>

<p>S4 generic functions needed by many Bioconductor packages.
</p>


<h3>Details</h3>

<p>We divide the generic functions defined in the <span class="pkg">BiocGenerics</span> package
in 2 categories: (1) functions already defined in base R and explicitly
promoted to generics in BiocGenerics, and (2) Bioconductor specific
generics.
</p>


<h4>(1) Functions defined in base R and explicitly promoted to
generics in the <span class="pkg">BiocGenerics</span> package</h4>

<p>Generics for functions defined in package <span class="pkg">base</span>:
</p>

<ul>
<li> <p><code>BiocGenerics::<a href="#topic+aperm">aperm</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+append">append</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+as.data.frame">as.data.frame</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+as.list">as.list</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+as.vector">as.vector</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+rbind">rbind</a></code>,
<code>BiocGenerics::<a href="#topic+cbind">cbind</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+do.call">do.call</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+duplicated">duplicated</a></code>,
<code>BiocGenerics::<a href="#topic+anyDuplicated">anyDuplicated</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+eval">eval</a></code>
</p>
</li>
<li><p> Extremes:
<code>BiocGenerics::<a href="#topic+pmax">pmax</a></code>,
<code>BiocGenerics::<a href="#topic+pmin">pmin</a></code>,
<code>BiocGenerics::<a href="#topic+pmax.int">pmax.int</a></code>,
<code>BiocGenerics::<a href="#topic+pmin.int">pmin.int</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+format">format</a></code>
</p>
</li>
<li><p> funprog:
<code>BiocGenerics::<a href="#topic+Reduce">Reduce</a></code>,
<code>BiocGenerics::<a href="#topic+Filter">Filter</a></code>,
<code>BiocGenerics::<a href="#topic+Find">Find</a></code>,
<code>BiocGenerics::<a href="#topic+Map">Map</a></code>,
<code>BiocGenerics::<a href="#topic+Position">Position</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+get">get</a></code>,
<code>BiocGenerics::<a href="#topic+mget">mget</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+grep">grep</a></code>,
<code>BiocGenerics::<a href="#topic+grepl">grepl</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+is.unsorted">is.unsorted</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+lapply">lapply</a></code>,
<code>BiocGenerics::<a href="#topic+sapply">sapply</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+mapply">mapply</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+match">match</a></code>,
<code>BiocGenerics::<a href="#topic++25in+25">%in%</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+nrow">nrow</a></code>,
<code>BiocGenerics::<a href="#topic+ncol">ncol</a></code>,
<code>BiocGenerics::<a href="#topic+NROW">NROW</a></code>,
<code>BiocGenerics::<a href="#topic+NCOL">NCOL</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+order">order</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+paste">paste</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+rank">rank</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+rep.int">rep.int</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+rownames">rownames</a></code>,
<code>BiocGenerics::<a href="#topic+rownames+3C-">rownames&lt;-</a></code>,
<code>BiocGenerics::<a href="#topic+colnames">colnames</a></code>,
<code>BiocGenerics::<a href="#topic+colnames+3C-">colnames&lt;-</a></code>
</p>
</li>
<li><p> sets:
<code>BiocGenerics::<a href="#topic+union">union</a></code>,
<code>BiocGenerics::<a href="#topic+intersect">intersect</a></code>,
<code>BiocGenerics::<a href="#topic+setdiff">setdiff</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+sort">sort</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+start">start</a></code>,
<code>BiocGenerics::<a href="#topic+start+3C-">start&lt;-</a></code>,
<code>BiocGenerics::<a href="#topic+end">end</a></code>,
<code>BiocGenerics::<a href="#topic+end+3C-">end&lt;-</a></code>,
<code>BiocGenerics::<a href="#topic+width">width</a></code>,
<code>BiocGenerics::<a href="#topic+width+3C-">width&lt;-</a></code>,
<code>BiocGenerics::<a href="#topic+pos">pos</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+subset">subset</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+t">t</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+table">table</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+tapply">tapply</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+unique">unique</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+unlist">unlist</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+unsplit">unsplit</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+which">which</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+which.min">which.min</a></code>,
<code>BiocGenerics::<a href="#topic+which.max">which.max</a></code>
</p>
</li></ul>

<p>Generics for functions defined in package <span class="pkg">utils</span>:
</p>

<ul>
<li> <p><code>BiocGenerics::<a href="#topic+relist">relist</a></code>
</p>
</li></ul>

<p>Generics for functions defined in package <span class="pkg">graphics</span>:
</p>

<ul>
<li> <p><code>BiocGenerics::<a href="#topic+boxplot">boxplot</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+image">image</a></code>
</p>
</li></ul>

<p>Generics for functions defined in package <span class="pkg">stats</span>:
</p>

<ul>
<li> <p><code>BiocGenerics::<a href="#topic+density">density</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+residuals">residuals</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+weights">weights</a></code>
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+xtabs">xtabs</a></code>
</p>
</li></ul>




<h4>(2) Bioconductor specific generics</h4>


<ul>
<li> <p><code><a href="#topic+annotation">annotation</a></code>,
<code><a href="#topic+annotation+3C-">annotation&lt;-</a></code>
</p>
</li>
<li> <p><code><a href="#topic+combine">combine</a></code>
</p>
</li>
<li> <p><code><a href="#topic+dbconn">dbconn</a></code>,
<code><a href="#topic+dbfile">dbfile</a></code>
</p>
</li>
<li> <p><code><a href="#topic+counts">counts</a></code>,
<code><a href="#topic+counts+3C-">counts&lt;-</a></code>,
<code><a href="#topic+design">design</a></code>,
<code><a href="#topic+design+3C-">design&lt;-</a></code>,
<code><a href="#topic+dispTable">dispTable</a></code>,
<code><a href="#topic+dispTable+3C-">dispTable&lt;-</a></code>,
<code><a href="#topic+sizeFactors">sizeFactors</a></code>,
<code><a href="#topic+sizeFactors+3C-">sizeFactors&lt;-</a></code>,
<code><a href="#topic+conditions">conditions</a></code>,
<code><a href="#topic+conditions+3C-">conditions&lt;-</a></code>,
<code><a href="#topic+estimateSizeFactors">estimateSizeFactors</a></code>,
<code><a href="#topic+estimateDispersions">estimateDispersions</a></code>,
<code><a href="#topic+plotDispEsts">plotDispEsts</a></code>
</p>
</li>
<li> <p><code><a href="#topic+dims">dims</a></code>,
<code><a href="#topic+nrows">nrows</a></code>,
<code><a href="#topic+ncols">ncols</a></code>,
</p>
</li>
<li> <p><code><a href="#topic+fileName">fileName</a></code>
</p>
</li>
<li> <p><code><a href="#topic+normalize">normalize</a></code>
</p>
</li>
<li> <p><code><a href="#topic+Ontology">Ontology</a></code>
</p>
</li>
<li> <p><code><a href="#topic+organism">organism</a></code>,
<code><a href="#topic+organism+3C-">organism&lt;-</a></code>,
<code><a href="#topic+species">species</a></code>,
<code><a href="#topic+species+3C-">species&lt;-</a></code>
</p>
</li>
<li> <p><code><a href="#topic+path">path</a></code>,
<code><a href="#topic+path+3C-">path&lt;-</a></code>,
<code><a href="#topic+basename">basename</a></code>,
<code><a href="#topic+basename+3C-">basename&lt;-</a></code>,
<code><a href="#topic+dirname">dirname</a></code>,
<code><a href="#topic+dirname+3C-">dirname&lt;-</a></code>
</p>
</li>
<li> <p><code><a href="#topic+plotMA">plotMA</a></code>
</p>
</li>
<li> <p><code><a href="#topic+plotPCA">plotPCA</a></code>
</p>
</li>
<li> <p><code><a href="#topic+score">score</a></code>,
<code><a href="#topic+score+3C-">score&lt;-</a></code>
</p>
</li>
<li> <p><code><a href="#topic+strand">strand</a></code>,
<code><a href="#topic+strand+3C-">strand&lt;-</a></code>,
<code><a href="#topic+invertStrand">invertStrand</a></code>
</p>
</li>
<li> <p><code><a href="#topic+toTable">toTable</a></code>
</p>
</li>
<li> <p><code><a href="#topic+type">type</a></code>,
<code><a href="#topic+type+3C-">type&lt;-</a></code>
</p>
</li>
<li> <p><code><a href="#topic+updateObject">updateObject</a></code>
</p>
</li></ul>




<h3>Note</h3>

<p>More generics can be added on request by sending an email to the
Bioc-devel mailing list:
</p>
<p><a href="http://bioconductor.org/help/mailing-list/">http://bioconductor.org/help/mailing-list/</a>
</p>
<p>Things that should NOT be added to the <span class="pkg">BiocGenerics</span> package:
</p>

<ul>
<li><p> Internal generic primitive functions like <code><a href="base.html#topic+length">length</a></code>,
<code><a href="base.html#topic+dim">dim</a></code>, <code>`<a href="base.html#topic+dim+3C-">dim&lt;-</a>`</code>, etc...
See <code>?<a href="base.html#topic+InternalMethods">InternalMethods</a></code> for the complete list.
There are a few exceptions though, that is, the <span class="pkg">BiocGenerics</span>
package may actually redefine a few of those internal generic
primitive functions as S4 generics when for example the signature
of the internal generic primitive is not appropriate (this is the
case for <code>BiocGenerics::<a href="#topic+cbind">cbind</a></code>).
</p>
</li>
<li><p> S3 and S4 group generic functions like <code><a href="methods.html#topic+Math">Math</a></code>,
<code><a href="methods.html#topic+Ops">Ops</a></code>, etc... See <code>?<a href="base.html#topic+groupGeneric">groupGeneric</a></code> and
<code>?<a href="methods.html#topic+S4groupGeneric">S4groupGeneric</a></code> for the complete list.
</p>
</li>
<li><p> Generics already defined in the <span class="pkg">stats4</span> package.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>The Bioconductor Dev Team</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+setGeneric">setGeneric</a></code> and
<code><a href="methods.html#topic+setMethod">setMethod</a></code> for defining generics and methods.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## List all the symbols defined in this package:
ls('package:BiocGenerics')
</code></pre>

<hr>
<h2 id='boxplot'>Box plots</h2><span id='topic+boxplot'></span>

<h3>Description</h3>

<p>Produce box-and-whisker plot(s) of the given (grouped) values.
</p>
<p>NOTE: This man page is for the <code>boxplot</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?graphics::<a href="graphics.html#topic+boxplot">boxplot</a></code> for the default method
(defined in the <span class="pkg">graphics</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>boxplot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="boxplot_+3A_x">x</code>, <code id="boxplot_+3A_...">...</code></td>
<td>

<p>See <code>?graphics::<a href="graphics.html#topic+boxplot">boxplot</a></code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?graphics::<a href="graphics.html#topic+boxplot">boxplot</a></code> for the value returned by the
default method.
</p>
<p>Specific methods defined in Bioconductor packages should
behave as consistently as possible with the default method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>graphics::<a href="graphics.html#topic+boxplot">boxplot</a></code> for the default
<code>boxplot</code> method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="affy.html#topic+boxplot+2CAffyBatch-method">boxplot,AffyBatch-method</a> in the <span class="pkg">affy</span> package
for an example of a specific <code>boxplot</code> method (defined for
<a href="affy.html#topic+AffyBatch">AffyBatch</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>boxplot
showMethods("boxplot")
selectMethod("boxplot", "ANY")  # the default method

library(affy)
showMethods("boxplot")
## The boxplot() method for AffyBatch objects:
selectMethod("boxplot", "AffyBatch")
</code></pre>

<hr>
<h2 id='cbind'>Combine objects by rows or columns</h2><span id='topic+rbind'></span><span id='topic+cbind'></span>

<h3>Description</h3>

<p><code>rbind</code> and <code>cbind</code> take one or more objects and combine them
by columns or rows, respectively.
</p>
<p>NOTE: This man page is for the <code>rbind</code> and <code>cbind</code>
<em>S4 generic functions</em> defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+cbind">cbind</a></code> for the default methods
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
(typically vector-like or matrix-like) not supported by the default
methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rbind(..., deparse.level=1)
cbind(..., deparse.level=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cbind_+3A_...">...</code></td>
<td>

<p>One or more vector-like or matrix-like objects. These can be given as
named arguments.
</p>
</td></tr>
<tr><td><code id="cbind_+3A_deparse.level">deparse.level</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+cbind">cbind</a></code> for a description of
this argument.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+cbind">cbind</a></code> for the value returned by the
default methods.
</p>
<p>Specific methods defined in Bioconductor packages will typically
return an object of the same class as the input objects.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+cbind">cbind</a></code> for the default <code>rbind</code> and
<code>cbind</code> methods.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+rbind+2CRectangularData-method">rbind,RectangularData-method</a> and
<a href="S4Vectors.html#topic+cbind+2CDataFrame-method">cbind,DataFrame-method</a> in the <span class="pkg">S4Vectors</span>
package for examples of specific <code>rbind</code> and <code>cbind</code>
methods (defined for <a href="S4Vectors.html#topic+RectangularData">RectangularData</a> derivatives
and <a href="S4Vectors.html#topic+DataFrame">DataFrame</a> objects, respectively).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>rbind  # note the dispatch on the '...' arg only
showMethods("rbind")
selectMethod("rbind", "ANY")  # the default method

cbind  # note the dispatch on the '...' arg only
showMethods("cbind")
selectMethod("cbind", "ANY")  # the default method

library(S4Vectors)
showMethods("rbind")
## The rbind() method for RectangularData derivatives:
selectMethod("rbind", "RectangularData")
## The cbind() method for DataFrame objects:
selectMethod("cbind", "DataFrame")
</code></pre>

<hr>
<h2 id='combine'>Combining or merging different Bioconductor data structures</h2><span id='topic+combine'></span><span id='topic+combine+2CANY+2Cmissing-method'></span><span id='topic+combine+2Cdata.frame+2Cdata.frame-method'></span><span id='topic+combine+2Cmatrix+2Cmatrix-method'></span>

<h3>Description</h3>

<p>The <code>combine</code> generic function handles methods for combining
or merging different Bioconductor data structures.
It should, given an arbitrary number of arguments of the same class
(possibly by inheritance), combine them into a single instance in
a sensible way (some methods may only combine 2 objects,
ignoring <code>...</code> in the argument list; because Bioconductor
data structures are complicated, check carefully that <code>combine</code>
does as you intend).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine(x, y, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combine_+3A_x">x</code></td>
<td>
<p>One of the values.</p>
</td></tr>
<tr><td><code id="combine_+3A_y">y</code></td>
<td>
<p>A second value.</p>
</td></tr>
<tr><td><code id="combine_+3A_...">...</code></td>
<td>
<p>Any other objects of the same class as <code>x</code> and <code>y</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are two basic combine strategies. One is an intersection
strategy. The returned value should only have rows (or columns) that
are found in all input data objects. The union strategy says that the
return value will have all rows (or columns) found in any one of the
input data objects (in which case some indication of what to use for
missing values will need to be provided).
</p>
<p>These functions and methods are currently under construction. Please
let us know if there are features that you require.
</p>


<h3>Value</h3>

<p>A single value of the same class as the most specific common ancestor
(in class terms) of the input values. This will contain the
appropriate combination of the data in the input values.
</p>


<h3>Methods</h3>

<p>The following methods are defined in the <span class="pkg">BiocGenerics</span> package:
</p>

<dl>
<dt><code>combine(x=ANY, missing)</code></dt><dd><p>Return the first (x) argument
unchanged.</p>
</dd>
<dt><code>combine(data.frame, data.frame)</code></dt><dd><p>Combines two
<code>data.frame</code> objects so that the resulting <code>data.frame</code>
contains all rows and columns of the original objects. Rows and
columns in the returned value are unique, that is, a row or column
represented in both arguments is represented only once in the
result. To perform this operation, <code>combine</code> makes sure that data
in shared rows and columns are identical in the two
data.frames. Data differences in shared rows and columns usually cause an
error. <code>combine</code> issues a warning when a column is a
<code><a href="base.html#topic+factor">factor</a></code> and the levels of the factor in the two
data.frames are different.</p>
</dd>
<dt><code>combine(matrix, matrix)</code></dt><dd><p>Combined two <code>matrix</code>
objects so that the resulting <code>matrix</code> contains all rows and
columns of the original objects. Both matricies must have
<code>dimnames</code>. Rows and columns in the returned
value are unique, that is, a row or column represented in both
arguments is represented only once in the result. To perform this
operation, <code>combine</code> makes sure that data in shared rows and
columns are all equal in the two matricies.</p>
</dd>
</dl>

<p>Additional <code>combine</code> methods are defined in the <span class="pkg">Biobase</span> package
for <a href="Biobase.html#topic+AnnotatedDataFrame">AnnotatedDataFrame</a>,
<a href="Biobase.html#topic+AssayData">AssayData</a>, <a href="Biobase.html#topic+MIAME">MIAME</a>,
and <a href="Biobase.html#topic+eSet">eSet</a> objects.
</p>


<h3>Author(s)</h3>

<p>Biocore</p>


<h3>See Also</h3>


<ul>
<li> <p><a href="Biobase.html#topic+combine+2CAnnotatedDataFrame+2CAnnotatedDataFrame-method">combine,AnnotatedDataFrame,AnnotatedDataFrame-method</a>,
<a href="Biobase.html#topic+combine+2CAssayData+2CAssayData-method">combine,AssayData,AssayData-method</a>,
<a href="Biobase.html#topic+combine+2CMIAME+2CMIAME-method">combine,MIAME,MIAME-method</a>,
and <a href="Biobase.html#topic+combine+2CeSet+2CeSet-method">combine,eSet,eSet-method</a> in the <span class="pkg">Biobase</span>
package for additional <code>combine</code> methods.
</p>
</li>
<li> <p><code><a href="base.html#topic+merge">merge</a></code> for merging two data frames (or data-frame-like)
objects.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>combine
showMethods("combine")
selectMethod("combine", c("ANY", "missing"))
selectMethod("combine", c("data.frame", "data.frame"))
selectMethod("combine", c("matrix", "matrix"))

## ---------------------------------------------------------------------
## COMBINING TWO DATA FRAMES
## ---------------------------------------------------------------------
x &lt;- data.frame(x=1:5,
        y=factor(letters[1:5], levels=letters[1:8]),
        row.names=letters[1:5])
y &lt;- data.frame(z=3:7,
        y=factor(letters[3:7], levels=letters[1:8]),
        row.names=letters[3:7])
combine(x,y)

w &lt;- data.frame(w=4:8,
       y=factor(letters[4:8], levels=letters[1:8]),
       row.names=letters[4:8])
combine(w, x, y)

# y is converted to 'factor' with different levels
df1 &lt;- data.frame(x=1:5,y=letters[1:5], row.names=letters[1:5])
df2 &lt;- data.frame(z=3:7,y=letters[3:7], row.names=letters[3:7])
try(combine(df1, df2)) # fails
# solution 1: ensure identical levels
y1 &lt;- factor(letters[1:5], levels=letters[1:7])
y2 &lt;- factor(letters[3:7], levels=letters[1:7])
df1 &lt;- data.frame(x=1:5,y=y1, row.names=letters[1:5])
df2 &lt;- data.frame(z=3:7,y=y2, row.names=letters[3:7])
combine(df1, df2)
# solution 2: force column to be 'character'
df1 &lt;- data.frame(x=1:5,y=I(letters[1:5]), row.names=letters[1:5])
df2 &lt;- data.frame(z=3:7,y=I(letters[3:7]), row.names=letters[3:7])
combine(df1, df2)

## ---------------------------------------------------------------------
## COMBINING TWO MATRICES
## ---------------------------------------------------------------------
m &lt;- matrix(1:20, nrow=5, dimnames=list(LETTERS[1:5], letters[1:4]))
combine(m[1:3,], m[4:5,])
combine(m[1:3, 1:3], m[3:5, 3:4]) # overlap
</code></pre>

<hr>
<h2 id='dbconn'>Accessing SQLite DB information</h2><span id='topic+dbconn'></span><span id='topic+dbfile'></span>

<h3>Description</h3>

<p>Get a connection object or file path for a SQLite DB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbconn(x)
dbfile(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dbconn_+3A_x">x</code></td>
<td>

<p>An object with a SQLite connection.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>dbconn</code> returns a connection object to the SQLite DB containing
<code>x</code>'s data.
</p>
<p><code>dbfile</code> returns a path (character string) to the SQLite DB (file)
containing <code>x</code>'s data.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="AnnotationDbi.html#topic+dbconn+2CAnnotationDb-method">dbconn,AnnotationDb-method</a> in the
<span class="pkg">AnnotationDbi</span> package for an example of a specific
<code>dbconn</code> method (defined for <a href="AnnotationDbi.html#topic+dbconn">dbconn</a>
objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>dbconn
showMethods("dbconn")
dbfile
showMethods("dbfile")

library(AnnotationDbi)
showMethods("dbconn")
selectMethod("dbconn", "AnnotationDb")
</code></pre>

<hr>
<h2 id='density'>Kernel density estimation</h2><span id='topic+density'></span>

<h3>Description</h3>

<p>The generic function <code>density</code> computes kernel density
estimates.
</p>
<p>NOTE: This man page is for the <code>density</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?stats::<a href="stats.html#topic+density">density</a></code> for the default method
(defined in the <span class="pkg">stats</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>density(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="density_+3A_x">x</code>, <code id="density_+3A_...">...</code></td>
<td>

<p>See <code>?stats::<a href="stats.html#topic+density">density</a></code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?stats::<a href="stats.html#topic+density">density</a></code> for the value returned by the
default method.
</p>
<p>Specific methods defined in Bioconductor packages should
behave as consistently as possible with the default method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>stats::<a href="stats.html#topic+density">density</a></code> for the default <code>density</code>
method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="flowClust.html#topic+density+2CflowClust-method">density,flowClust-method</a> in the <span class="pkg">flowClust</span>
package for an example of a specific <code>density</code> method (defined
for <a href="flowClust.html#topic+flowClust">flowClust</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>density
showMethods("density")
selectMethod("density", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='dge'>Accessors and generic functions used in the context of count datasets</h2><span id='topic+dge'></span><span id='topic+counts'></span><span id='topic+counts+3C-'></span><span id='topic+design'></span><span id='topic+design+3C-'></span><span id='topic+dispTable'></span><span id='topic+dispTable+3C-'></span><span id='topic+sizeFactors'></span><span id='topic+sizeFactors+3C-'></span><span id='topic+conditions'></span><span id='topic+conditions+3C-'></span><span id='topic+estimateSizeFactors'></span><span id='topic+estimateDispersions'></span><span id='topic+plotDispEsts'></span>

<h3>Description</h3>

<p>These generic functions provide basic interfaces to operations on
and data access to count datasets.</p>


<h3>Usage</h3>

<pre><code class='language-R'>counts(object, ...)
counts(object, ...) &lt;- value
design(object, ...)
design(object, ...) &lt;- value
dispTable(object, ...)
dispTable(object, ...)  &lt;- value
sizeFactors(object, ...)
sizeFactors(object, ...) &lt;- value
conditions(object, ...)
conditions(object, ...) &lt;- value
estimateSizeFactors(object, ...)
estimateDispersions(object, ...)
plotDispEsts(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dge_+3A_object">object</code></td>
<td>
<p>Object of class for which methods are defined, e.g.,
<code>CountDataSet</code>, <code>DESeqSummarizedExperiment</code> or <code>ExonCountSet</code>.</p>
</td></tr>
<tr><td><code id="dge_+3A_value">value</code></td>
<td>
<p>Value to be assigned to corresponding components of
<code>object</code>; supported types depend on method implementation.</p>
</td></tr>
<tr><td><code id="dge_+3A_...">...</code></td>
<td>
<p>Further arguments, perhaps used by metohds</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For the details, please consult the manual pages of the
methods in the <span class="pkg">DESeq</span>, <span class="pkg">DESeq2</span>, and <span class="pkg">DEXSeq</span> packages and the package
vignettes.</p>


<h3>Author(s)</h3>

<p>W. Huber, S. Anders</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>


<hr>
<h2 id='dims'>Get the dimensions of each element of a list-like object</h2><span id='topic+dims'></span><span id='topic+nrows'></span><span id='topic+ncols'></span>

<h3>Description</h3>

<p>Get the dimensions, number of rows, or number of columns, of each element
of a list-like object.
</p>
<p>Note that these functions are the <em>vectorized versions</em> of
corresponding functions <code>dim()</code>, <code>nrow()</code>, and <code>ncol()</code>,
in the same fashion that <code>lengths()</code> is the <em>vectorized version</em>
of <code>length</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dims(x, use.names=TRUE)
nrows(x, use.names=TRUE)
ncols(x, use.names=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dims_+3A_x">x</code></td>
<td>

<p>List-like object (or environment) where all the list elements
are expected to be array-like objects with the <em>same number
of dimensions</em>.
</p>
</td></tr>
<tr><td><code id="dims_+3A_use.names">use.names</code></td>
<td>

<p>Logical indicating if the names on <code>x</code> should be propagated to
the returned matrix (as its rownames) or vector (as its names).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>For <code>dims()</code>: Typically a numeric matrix with one row per list element
in <code>x</code> and one column per dimension in these list elements (they're
all expected to have the same number of dimensions). The i-th row in the
returned matrix is a vector containing the dimensions of the i-th list
element in <code>x</code>. More formally:
</p>
<pre>  dims(x)[i, ] is dim(x[[i]])</pre>
<p>for any valid <code>i</code>.
By default the names on <code>x</code>, if any, are propagated as the rownames
of the returned matrix, unless <code>use.names</code> is set to <code>FALSE</code>.
</p>
<p>For <code>nrows()</code> or <code>ncols()</code>: A numeric vector with one element
per list element in <code>x</code>. The i-th element in the returned vector is
the number of rows (or columns) of the i-th list element in <code>x</code>.
More formally:
</p>
<pre>  nrows(x)[i] is nrow(x[[i]]) and ncols(x)[i] is ncol(x[[i]])</pre>
<p>for any valid <code>i</code>.
By default the names on <code>x</code>, if any, are propagated on the returned
vector, unless <code>use.names</code> is set to <code>FALSE</code>.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="IRanges.html#topic+dims+2CDataFrameList-method">dims,DataFrameList-method</a> in the
<span class="pkg">IRanges</span> package for an example of a specific
<code>dims</code> method (defined for <a href="IRanges.html#topic+DataFrameList">DataFrameList</a>
objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>dims
showMethods("dims")

library(IRanges)
showMethods("dims")
selectMethod("dims", "DataFrameList")
</code></pre>

<hr>
<h2 id='do.call'>Execute a function call</h2><span id='topic+do.call'></span>

<h3>Description</h3>

<p><code>do.call</code> constructs and executes a function call from a name or a
function and a list of arguments to be passed to it.
</p>
<p>NOTE: This man page is for the <code>do.call</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+do.call">do.call</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do.call(what, args, quote=FALSE, envir=parent.frame())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="do.call_+3A_what">what</code></td>
<td>

<p>The default method expects either a function or a non-empty character
string naming the function to be called.
See <code>?base::<a href="base.html#topic+do.call">do.call</a></code> for the details.
</p>
<p>Specific methods can support other objects.
Please refer to the documentation of a particular method for the details.
</p>
</td></tr>
<tr><td><code id="do.call_+3A_args">args</code></td>
<td>

<p>The default method expects a <em>list</em> of arguments to the function
call (the <code>names</code> attribute of <code>args</code> gives the argument names).
See <code>?base::<a href="base.html#topic+do.call">do.call</a></code> for the details.
</p>
<p>Specific methods can support other objects.
Please refer to the documentation of a particular method for the details.
</p>
</td></tr>
<tr><td><code id="do.call_+3A_quote">quote</code>, <code id="do.call_+3A_envir">envir</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+do.call">do.call</a></code> for a description of
these arguments.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of the (evaluated) function call.
</p>
<p>Specific methods defined in Bioconductor packages should 
behave as consistently as possible with the default method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+do.call">do.call</a></code> for the default <code>do.call</code>
method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>do.call  # note the dispatch on the 'what' and 'args' args only
showMethods("do.call")
selectMethod("do.call", c("ANY", "ANY"))  # the default method
</code></pre>

<hr>
<h2 id='duplicated'>Determine duplicate elements</h2><span id='topic+duplicated'></span><span id='topic+anyDuplicated'></span>

<h3>Description</h3>

<p>Determines which elements of a vector-like or data-frame-like object
are duplicates of elements with smaller subscripts, and returns a
logical vector indicating which elements (rows) are duplicates.
</p>
<p>NOTE: This man page is for the <code>duplicated</code> and <code>anyDuplicated</code>
<em>S4 generic functions</em> defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+duplicated">duplicated</a></code> for the default methods
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
(typically vector-like or data-frame-like) not supported by the
default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>duplicated(x, incomparables=FALSE, ...)
anyDuplicated(x, incomparables=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="duplicated_+3A_x">x</code></td>
<td>

<p>A vector-like or data-frame-like object.
</p>
</td></tr>
<tr><td><code id="duplicated_+3A_incomparables">incomparables</code>, <code id="duplicated_+3A_...">...</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+duplicated">duplicated</a></code> for a description of
these arguments.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The default <code>duplicated</code> method (see
<code>?base::<a href="base.html#topic+duplicated">duplicated</a></code>) returns a logical vector
of length N where N is:
</p>

<ul>
<li> <p><code>length(x)</code> when <code>x</code> is a vector;
</p>
</li>
<li> <p><code>nrow(x)</code> when <code>x</code> is a data frame.
</p>
</li></ul>

<p>Specific <code>duplicated</code> methods defined in Bioconductor
packages must also return a logical vector of the same length
as <code>x</code> when <code>x</code> is a vector-like object, and a logical
vector with one element for each row when <code>x</code> is a
data-frame-like object.
</p>
<p>The default <code>anyDuplicated</code> method (see
<code>?base::<a href="base.html#topic+duplicated">duplicated</a></code>) returns a single
non-negative integer and so must the specific <code>anyDuplicated</code>
methods defined in Bioconductor packages.
</p>
<p><code>anyDuplicated</code> should always behave consistently with
<code>duplicated</code>.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+duplicated">duplicated</a></code> for the default
<code>duplicated</code> and <code>anyDuplicated</code> methods.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+duplicated+2CRle-method">duplicated,Rle-method</a> in the <span class="pkg">S4Vectors</span>
package for an example of a specific <code>duplicated</code> method
(defined for <a href="S4Vectors.html#topic+Rle">Rle</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>duplicated
showMethods("duplicated")
selectMethod("duplicated", "ANY")  # the default method

anyDuplicated
showMethods("anyDuplicated")
selectMethod("anyDuplicated", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='eval'>Evaluate an (unevaluated) expression</h2><span id='topic+eval'></span>

<h3>Description</h3>

<p><code>eval</code> evaluates an R expression in a specified environment.
</p>
<p>NOTE: This man page is for the <code>eval</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+eval">eval</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eval(expr, envir=parent.frame(),
           enclos=if (is.list(envir) || is.pairlist(envir))
                  parent.frame() else baseenv())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="eval_+3A_expr">expr</code></td>
<td>

<p>An object to be evaluated.
May be any object supported by the default method
(see <code>?base::<a href="base.html#topic+eval">eval</a></code>) or by the additional
methods defined in Bioconductor packages.
</p>
</td></tr>
<tr><td><code id="eval_+3A_envir">envir</code></td>
<td>

<p>The <em>environment</em> in which <code>expr</code> is to be evaluated.
May be any object supported by the default method
(see <code>?base::<a href="base.html#topic+eval">eval</a></code>) or by the additional
methods defined in Bioconductor packages.
</p>
</td></tr>
<tr><td><code id="eval_+3A_enclos">enclos</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+eval">eval</a></code> for a description of
this argument.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+eval">eval</a></code> for the value returned by the
default method.
</p>
<p>Specific methods defined in Bioconductor packages should
behave as consistently as possible with the default method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+eval">eval</a></code> for the default <code>eval</code> method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="IRanges.html#topic+eval+2Cexpression+2CVector-method">eval,expression,Vector-method</a> in the <span class="pkg">IRanges</span>
package for an example of a specific <code>eval</code> method (defined
for when the <code>expr</code> and <code>envir</code> arguments are an
<a href="base.html#topic+expression">expression</a> and a <a href="S4Vectors.html#topic+Vector">Vector</a> object,
respectively).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>eval  # note the dispatch on 'expr' and 'envir' args only
showMethods("eval")
selectMethod("eval", c("ANY", "ANY"))  # the default method
</code></pre>

<hr>
<h2 id='evalq'>Evaluate an (unevaluated) expression</h2><span id='topic+evalq'></span>

<h3>Description</h3>

<p><code>evalq</code> evaluates an R expression (the quoted form of its first
argument) in a specified environment.
</p>
<p>NOTE: This man page is for the <code>evalq</code> wrapper defined in the
<span class="pkg">BiocGenerics</span> package.  See <code>?base::<a href="base.html#topic+evalq">evalq</a></code> for the
function defined in the <span class="pkg">base</span> package. This wrapper correctly delegates
to the <code>eval</code> generic, rather than
<code>base::<a href="base.html#topic+eval">eval</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evalq(expr, envir=parent.frame(),
           enclos=if (is.list(envir) || is.pairlist(envir))
                  parent.frame() else baseenv())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="evalq_+3A_expr">expr</code></td>
<td>

<p>Quoted to form the expression that is evaluated.
</p>
</td></tr>
<tr><td><code id="evalq_+3A_envir">envir</code></td>
<td>

<p>The <em>environment</em> in which <code>expr</code> is to be evaluated.
May be any object supported by methods on the <code><a href="base.html#topic+eval">eval</a></code>
generic.
</p>
</td></tr>
<tr><td><code id="evalq_+3A_enclos">enclos</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+evalq">evalq</a></code> for a description of
this argument.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+evalq">evalq</a></code>.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+evalq">evalq</a></code> for the base <code>evalq</code> function.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>evalq  # note just a copy of the original evalq
</code></pre>

<hr>
<h2 id='Extremes'>Maxima and minima</h2><span id='topic+Extremes'></span><span id='topic+pmax'></span><span id='topic+pmin'></span><span id='topic+pmax.int'></span><span id='topic+pmin.int'></span>

<h3>Description</h3>

<p><code>pmax</code>, <code>pmin</code>, <code>pmax.int</code> and <code>pmin.int</code> return the
parallel maxima and minima of the input values.
</p>
<p>NOTE: This man page is for the <code>pmax</code>, <code>pmin</code>, <code>pmax.int</code>
and <code>pmin.int</code> <em>S4 generic functions</em> defined in the
<span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+pmax">pmax</a></code> for the default methods
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
(typically vector-like or matrix-like) not supported by the default
methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pmax(..., na.rm=FALSE)
pmin(..., na.rm=FALSE)
     
pmax.int(..., na.rm=FALSE)
pmin.int(..., na.rm=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Extremes_+3A_...">...</code></td>
<td>

<p>One or more vector-like or matrix-like objects.
</p>
</td></tr>
<tr><td><code id="Extremes_+3A_na.rm">na.rm</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+pmax">pmax</a></code> for a description of
this argument.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+pmax">pmax</a></code> for the value returned by the
default methods.
</p>
<p>Specific methods defined in Bioconductor packages will typically
return an object of the same class as the input objects.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+pmax">pmax</a></code> for the default <code>pmax</code>,
<code>pmin</code>, <code>pmax.int</code> and <code>pmin.int</code> methods.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+pmax+2CRle-method">pmax,Rle-method</a> in the <span class="pkg">S4Vectors</span> package
for an example of a specific <code>pmax</code> method (defined for
<a href="S4Vectors.html#topic+Rle">Rle</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>pmax
showMethods("pmax")
selectMethod("pmax", "ANY")  # the default method

pmin
showMethods("pmin")
selectMethod("pmin", "ANY")  # the default method

pmax.int
showMethods("pmax.int")
selectMethod("pmax.int", "ANY")  # the default method

pmin.int
showMethods("pmin.int")
selectMethod("pmin.int", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='fileName'>Accessing the file name of an object</h2><span id='topic+fileName'></span>

<h3>Description</h3>

<p>Get the file name of an object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fileName(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fileName_+3A_object">object</code></td>
<td>

<p>An object with a file name.
</p>
</td></tr>
<tr><td><code id="fileName_+3A_...">...</code></td>
<td>

<p>Additional arguments, for use in specific methods.
</p>
</td></tr>
</table>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="MSnbase.html#topic+fileName+2CMSmap-method">fileName,MSmap-method</a> in the
<span class="pkg">MSnbase</span> package for an example of a specific
<code>fileName</code> method (defined for <a href="MSnbase.html#topic+MSmap">MSmap</a>
objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>fileName
showMethods("fileName")

library(MSnbase)
showMethods("fileName")
selectMethod("fileName", "MSmap")
</code></pre>

<hr>
<h2 id='format'>Format an R object for pretty printing</h2><span id='topic+format'></span><span id='topic+format.list'></span>

<h3>Description</h3>

<p>Turn an R object into a character vector used for pretty printing.
</p>
<p>NOTE: This man page is for the <code>format</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+format">format</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>format(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="format_+3A_x">x</code></td>
<td>

<p>The object to format.
</p>
</td></tr>
<tr><td><code id="format_+3A_...">...</code></td>
<td>

<p>Additional arguments, for use in specific methods.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector that provides a &quot;compact representation&quot; of <code>x</code>.
This character vector is typically used by <code>print.data.frame</code> to
display the columns of a data.frame object.
See <code>?base::<a href="base.html#topic+print.data.frame">print.data.frame</a></code> for more information.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+format">format</a></code> for the default <code>format</code>
method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>format
showMethods("format")
selectMethod("format", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='funprog'>Common higher-order functions in functional programming languages</h2><span id='topic+funprog'></span><span id='topic+Reduce'></span><span id='topic+Filter'></span><span id='topic+Find'></span><span id='topic+Map'></span><span id='topic+Position'></span>

<h3>Description</h3>

<p><code>Reduce</code> uses a binary function to successively combine the
elements of a given list-like or vector-like object and a possibly
given initial value.
<code>Filter</code> extracts the elements of a list-like or vector-like
object for which a predicate (logical) function gives true.
<code>Find</code> and <code>Position</code> give the first or last such element
and its position in the object, respectively.
<code>Map</code> applies a function to the corresponding elements of given
list-like or vector-like objects.
</p>
<p>NOTE: This man page is for the <code>Reduce</code>, <code>Filter</code>,
<code>Find</code>, <code>Map</code> and <code>Position</code> <em>S4 generic functions</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+Reduce">Reduce</a></code> for the default methods
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
(typically list-like or vector-like) not supported by the
default methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Reduce(f, x, init, right=FALSE, accumulate=FALSE)
Filter(f, x)
Find(f, x, right=FALSE, nomatch=NULL)
Map(f, ...)
Position(f, x, right=FALSE, nomatch=NA_integer_)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="funprog_+3A_f">f</code>, <code id="funprog_+3A_init">init</code>, <code id="funprog_+3A_right">right</code>, <code id="funprog_+3A_accumulate">accumulate</code>, <code id="funprog_+3A_nomatch">nomatch</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+Reduce">Reduce</a></code> for a description of
these arguments.
</p>
</td></tr>
<tr><td><code id="funprog_+3A_x">x</code></td>
<td>

<p>A list-like or vector-like object.
</p>
</td></tr>
<tr><td><code id="funprog_+3A_...">...</code></td>
<td>

<p>One or more list-like or vector-like objects.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+Reduce">Reduce</a></code> for the value returned by the
default methods.
</p>
<p>Specific methods defined in Bioconductor packages should
behave as consistently as possible with the default methods.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+Reduce">Reduce</a></code> for the default <code>Reduce</code>,
<code>Filter</code>, <code>Find</code>, <code>Map</code> and <code>Position</code> methods.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+Reduce+2CList-method">Reduce,List-method</a> in the <span class="pkg">S4Vectors</span> package
for an example of a specific <code>Reduce</code> method (defined for
<a href="S4Vectors.html#topic+List">List</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>Reduce  # note the dispatch on the 'x' arg only
showMethods("Reduce")
selectMethod("Reduce", "ANY")  # the default method

Filter  # note the dispatch on the 'x' arg only
showMethods("Filter")
selectMethod("Filter", "ANY")  # the default method

Find  # note the dispatch on the 'x' arg only
showMethods("Find")
selectMethod("Find", "ANY")  # the default method

Map  # note the dispatch on the '...' arg only
showMethods("Map")
selectMethod("Map", "ANY")  # the default method

Position  # note the dispatch on the 'x' arg only
showMethods("Position")
selectMethod("Position", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='get'>Return the value of a named object</h2><span id='topic+get'></span><span id='topic+mget'></span>

<h3>Description</h3>

<p>Search for an object with a given name and return it.
</p>
<p>NOTE: This man page is for the <code>get</code> and <code>mget</code> <em>S4 generic
functions</em> defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+get">get</a></code> for the default methods
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
(list-like or environment-like) not supported by the default methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get(x, pos=-1, envir=as.environment(pos), mode="any", inherits=TRUE)
mget(x, envir, mode="any", ifnotfound, inherits=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_+3A_x">x</code></td>
<td>

<p>For <code>get</code>: A variable name (or, more generally speaking,
a <em>key</em>), given as a single string.
</p>
<p>For <code>mget</code>: A vector of variable names (or <em>keys</em>).
</p>
</td></tr>
<tr><td><code id="get_+3A_envir">envir</code></td>
<td>

<p>Where to look for the key(s). Typically a list-like or environment-like
object.
</p>
</td></tr>
<tr><td><code id="get_+3A_pos">pos</code>, <code id="get_+3A_mode">mode</code>, <code id="get_+3A_inherits">inherits</code>, <code id="get_+3A_ifnotfound">ifnotfound</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+get">get</a></code> for a description of
these arguments.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See <code>?base::<a href="base.html#topic+get">get</a></code> for details about the default methods.
</p>


<h3>Value</h3>

<p>For <code>get</code>: The value corresponding to the specified key.
</p>
<p>For <code>mget</code>: The list of values corresponding to the specified keys.
The returned list must have one element per key, and in the same order
as in <code>x</code>.
</p>
<p>See <code>?base::<a href="base.html#topic+get">get</a></code> for the value returned by the
default methods.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+get">get</a></code> for the default <code>get</code> and
<code>mget</code> methods.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="AnnotationDbi.html#topic+get+2CANY+2CBimap+2Cmissing-method">get,ANY,Bimap,missing-method</a> in the
<span class="pkg">AnnotationDbi</span> package for an example of a specific <code>get</code>
method (defined for <a href="AnnotationDbi.html#topic+Bimap">Bimap</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>get  # note the dispatch on the 'x', 'pos' and 'envir' args only
showMethods("get")
selectMethod("get", c("ANY", "ANY", "ANY"))  # the default method

mget  # note the dispatch on the 'x' and 'envir' args only
showMethods("mget")
selectMethod("mget", c("ANY", "ANY"))  # the default method
</code></pre>

<hr>
<h2 id='grep'>Pattern Matching and Replacement</h2><span id='topic+grep'></span><span id='topic+grepl'></span>

<h3>Description</h3>

<p>Search for matches to argument 'pattern' within each element of a character
vector.
</p>
<p>NOTE: This man page is for the <code>grep</code> and <code>grepl</code>
<em>S4 generic functions</em> defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+grep">grep</a></code> for the default methods
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grep(pattern, x, ignore.case = FALSE, perl = FALSE, value = FALSE,
          fixed = FALSE, useBytes = FALSE, invert = FALSE)
grepl(pattern, x, ignore.case = FALSE, perl = FALSE,
          fixed = FALSE, useBytes = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="grep_+3A_pattern">pattern</code></td>
<td>

<p>The pattern for searching in <code>x</code>, such as a regular expression.
</p>
</td></tr>
<tr><td><code id="grep_+3A_x">x</code></td>
<td>

<p>The character vector (in the general sense) to search.
</p>
</td></tr>
<tr><td><code id="grep_+3A_ignore.case">ignore.case</code>, <code id="grep_+3A_perl">perl</code>, <code id="grep_+3A_value">value</code>, <code id="grep_+3A_fixed">fixed</code>, <code id="grep_+3A_usebytes">useBytes</code>, <code id="grep_+3A_invert">invert</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+grep">grep</a></code> for a description of
these arguments.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+grep">grep</a></code> for the value returned
by the default method.
</p>
<p>Specific methods defined in Bioconductor packages should
behave as consistently as possible with the default method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+grep">grep</a></code> for the default
<code>grep</code> and <code>grepl</code> methods.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>grep  # note the dispatch on 'pattern' and 'x' args only
showMethods("grep")
selectMethod("grep", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='image'>Display a color image</h2><span id='topic+image'></span>

<h3>Description</h3>

<p>Creates a grid of colored or gray-scale rectangles with colors
corresponding to the values in <code>z</code>.
This can be used to display three-dimensional or spatial data aka
<em>images</em>.
</p>
<p>NOTE: This man page is for the <code>image</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?graphics::<a href="graphics.html#topic+image">image</a></code> for the default method
(defined in the <span class="pkg">graphics</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>image(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="image_+3A_x">x</code>, <code id="image_+3A_...">...</code></td>
<td>

<p>See <code>?graphics::<a href="graphics.html#topic+image">image</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See <code>?graphics::<a href="graphics.html#topic+image">image</a></code> for the details.
</p>
<p>Specific methods defined in Bioconductor packages should
behave as consistently as possible with the default method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>graphics::<a href="graphics.html#topic+image">image</a></code> for the default <code>image</code>
method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="affy.html#topic+image+2CAffyBatch-method">image,AffyBatch-method</a> in the <span class="pkg">affy</span> package
for an example of a specific <code>image</code> method (defined for
<a href="affy.html#topic+AffyBatch">AffyBatch</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>image
showMethods("image")
selectMethod("image", "ANY")  # the default method

library(affy)
showMethods("image")
## The image() method for AffyBatch objects:
selectMethod("image", "AffyBatch")
</code></pre>

<hr>
<h2 id='IQR'>The Interquartile Range</h2><span id='topic+IQR'></span>

<h3>Description</h3>

<p>Compute the interquartile range for a vector.
</p>
<p>NOTE: This man page is for the <code>IQR</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?stats::<a href="stats.html#topic+IQR">IQR</a></code> for the default method
(defined in the <span class="pkg">stats</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IQR(x, na.rm = FALSE, type = 7)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IQR_+3A_x">x</code>, <code id="IQR_+3A_na.rm">na.rm</code>, <code id="IQR_+3A_type">type</code></td>
<td>

<p>See <code>?stats::<a href="stats.html#topic+IQR">IQR</a></code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?stats::<a href="stats.html#topic+IQR">IQR</a></code> for the value returned by the
default method.
</p>
<p>Specific methods defined in Bioconductor packages should
behave as consistently as possible with the default method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>stats::<a href="stats.html#topic+IQR">IQR</a></code> for the default
<code>IQR</code> method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>IQR
showMethods("IQR")
selectMethod("IQR", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='is.unsorted'>Test if a vector-like object is not sorted</h2><span id='topic+is.unsorted'></span>

<h3>Description</h3>

<p>Test if a vector-like object is not sorted, without the cost of sorting it.
</p>
<p>NOTE: This man page is for the <code>is.unsorted</code>
<em>S4 generic function</em> defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+is.unsorted">is.unsorted</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
(typically vector-like) not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.unsorted(x, na.rm=FALSE, strictly=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.unsorted_+3A_x">x</code></td>
<td>

<p>A vector-like object.
</p>
</td></tr>
<tr><td><code id="is.unsorted_+3A_na.rm">na.rm</code>, <code id="is.unsorted_+3A_strictly">strictly</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+is.unsorted">is.unsorted</a></code> for a description of
these arguments.
</p>
</td></tr>
<tr><td><code id="is.unsorted_+3A_...">...</code></td>
<td>

<p>Additional arguments, for use in specific methods.
</p>
<p>Note that <code>base::<a href="base.html#topic+is.unsorted">is.unsorted</a></code> (the default method) only
takes the <code>x</code>, <code>na.rm</code>, and <code>strictly</code> arguments.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+is.unsorted">is.unsorted</a></code> for the value returned
by the default method.
</p>
<p>Specific methods defined in Bioconductor packages should
behave as consistently as possible with the default method.
</p>


<h3>Note</h3>

<p>TO DEVELOPERS:
</p>
<p>The <code>is.unsorted</code> method for specific vector-like objects should
adhere to the same underlying order used by the
<code><a href="#topic+order">order</a></code>, <code><a href="#topic+sort">sort</a></code>,
and <code><a href="#topic+rank">rank</a></code> methods for the same objects.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+is.unsorted">is.unsorted</a></code> for the default
<code>is.unsorted</code> method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="GenomicRanges.html#topic+is.unsorted+2CGenomicRanges-method">is.unsorted,GenomicRanges-method</a> in
the <span class="pkg">GenomicRanges</span> package for an example of a specific
<code>is.unsorted</code> method (defined for
<a href="GenomicRanges.html#topic+GenomicRanges">GenomicRanges</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>is.unsorted  # note the dispatch on the 'x' arg only
showMethods("is.unsorted")
selectMethod("is.unsorted", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='lapply'>Apply a function over a list-like or vector-like object</h2><span id='topic+lapply'></span><span id='topic+sapply'></span>

<h3>Description</h3>

<p><code>lapply</code> returns a list of the same length as <code>X</code>, each
element of which is the result of applying <code>FUN</code> to the
corresponding element of <code>X</code>.
</p>
<p><code>sapply</code> is a user-friendly version and wrapper of <code>lapply</code>
by default returning a vector, matrix or, if <code>simplify="array"</code>,
an array if appropriate, by applying <code>simplify2array()</code>.
<code>sapply(x, f, simplify=FALSE, USE.NAMES=FALSE)</code> is the same
as <code>lapply(x, f)</code>.
</p>
<p>NOTE: This man page is for the <code>lapply</code> and <code>sapply</code>
<em>S4 generic functions</em> defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+lapply">lapply</a></code> for the default methods
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
(typically list-like or vector-like) not supported by the
default methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lapply(X, FUN, ...)
sapply(X, FUN, ..., simplify=TRUE, USE.NAMES=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lapply_+3A_x">X</code></td>
<td>

<p>A list-like or vector-like object.
</p>
</td></tr>
<tr><td><code id="lapply_+3A_fun">FUN</code>, <code id="lapply_+3A_...">...</code>, <code id="lapply_+3A_simplify">simplify</code>, <code id="lapply_+3A_use.names">USE.NAMES</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+lapply">lapply</a></code> for a description of
these arguments.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+lapply">lapply</a></code> for the value returned by the
default methods.
</p>
<p>Specific methods defined in Bioconductor packages should 
behave as consistently as possible with the default methods.
In particular, <code>lapply</code> and <code>sapply(simplify=FALSE)</code>
should always return a list.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+lapply">lapply</a></code> for the default <code>lapply</code> and
<code>sapply</code> methods.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+lapply+2CList-method">lapply,List-method</a> in the <span class="pkg">S4Vectors</span> package
for an example of a specific <code>lapply</code> method (defined for
<a href="S4Vectors.html#topic+List">List</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>lapply  # note the dispatch on the 'X' arg only
showMethods("lapply")
selectMethod("lapply", "ANY")  # the default method

sapply  # note the dispatch on the 'X' arg only
showMethods("sapply")
selectMethod("sapply", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='mad'>Median Absolute Deviation</h2><span id='topic+mad'></span>

<h3>Description</h3>

<p>Compute the median absolute deviation for a vector, dispatching only
on the first argument, <code>x</code>.
</p>
<p>NOTE: This man page is for the <code>mad</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?stats::<a href="stats.html#topic+mad">mad</a></code> for the default method
(defined in the <span class="pkg">stats</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mad(x, center = median(x), constant = 1.4826,
    na.rm = FALSE, low = FALSE, high = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mad_+3A_x">x</code>, <code id="mad_+3A_center">center</code>, <code id="mad_+3A_constant">constant</code>, <code id="mad_+3A_na.rm">na.rm</code>, <code id="mad_+3A_low">low</code>, <code id="mad_+3A_high">high</code></td>
<td>

<p>See <code>?stats::<a href="stats.html#topic+mad">mad</a></code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?stats::<a href="stats.html#topic+mad">mad</a></code> for the value returned by the
default method.
</p>
<p>Specific methods defined in Bioconductor packages should
behave as consistently as possible with the default method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>stats::<a href="stats.html#topic+mad">mad</a></code> for the default
<code>mad</code> method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>mad
showMethods("mad")
selectMethod("mad", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='mapply'>Apply a function to multiple list-like or vector-like arguments</h2><span id='topic+mapply'></span>

<h3>Description</h3>

<p><code>mapply</code> is a multivariate version of
<code><a href="#topic+sapply">sapply</a></code>. <code>mapply</code> applies
<code>FUN</code> to the first elements of each <code>...</code> argument,
the second elements, the third elements, and so on. Arguments
are recycled if necessary.
</p>
<p>NOTE: This man page is for the <code>mapply</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+mapply">mapply</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
(typically list-like or vector-like) not supported by the
default methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapply(FUN, ..., MoreArgs=NULL, SIMPLIFY=TRUE, USE.NAMES=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapply_+3A_fun">FUN</code>, <code id="mapply_+3A_moreargs">MoreArgs</code>, <code id="mapply_+3A_simplify">SIMPLIFY</code>, <code id="mapply_+3A_use.names">USE.NAMES</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+mapply">mapply</a></code> for a description of
these arguments.
</p>
</td></tr>
<tr><td><code id="mapply_+3A_...">...</code></td>
<td>

<p>One or more list-like or vector-like objects of strictly
positive length, or all of zero length.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+mapply">mapply</a></code> for the value returned by the
default method.
</p>
<p>Specific methods defined in Bioconductor packages should 
behave as consistently as possible with the default method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+mapply">mapply</a></code> for the default <code>mapply</code>
method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>mapply  # note the dispatch on the '...' arg only
showMethods("mapply")
selectMethod("mapply", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='match'>Value matching</h2><span id='topic+match'></span><span id='topic++25in+25'></span>

<h3>Description</h3>

<p><code>match</code> returns a vector of the positions of (first) matches of
its first argument in its second.
</p>
<p><code>%in%</code> is a binary operator that returns a logical vector of the
length of its left operand indicating if the elements in it have a match
or not.
</p>
<p>NOTE: This man page is for the <code>match</code> and <code>%in%</code> <em>S4
generic functions</em> defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+match">match</a></code> for the default methods
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
(typically vector-like) not supported by the default methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>match(x, table, nomatch=NA_integer_, incomparables=NULL, ...)

x %in% table
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="match_+3A_x">x</code>, <code id="match_+3A_table">table</code></td>
<td>

<p>Vector-like objects (typically of the same class, but not necessarily).
</p>
</td></tr>
<tr><td><code id="match_+3A_nomatch">nomatch</code>, <code id="match_+3A_incomparables">incomparables</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+match">match</a></code> for a description of
these arguments.
</p>
</td></tr>
<tr><td><code id="match_+3A_...">...</code></td>
<td>

<p>Additional arguments, for use in specific methods.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The same as the default methods (see <code>?base::<a href="base.html#topic+match">match</a></code> for
the value returned by the default methods).
</p>
<p>Specific methods defined in Bioconductor packages should
behave as consistently as possible with the default methods.
</p>


<h3>Note</h3>

<p>The default <code>base::<a href="base.html#topic+match">match</a></code> method (defined in the <span class="pkg">base</span>
package) doesn't have the <code>...</code> argument. We've added it to the generic
function defined in the <span class="pkg">BiocGenerics</span> package in order to allow specific
methods to support additional arguments if needed.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+match">match</a></code> for the default <code>match</code> and
<code>%in%</code> methods.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+match+2CHits+2CHits-method">match,Hits,Hits-method</a> and
<a href="S4Vectors.html#topic++25in+25+2CRle+2CANY-method">%in%,Rle,ANY-method</a> in the <span class="pkg">S4Vectors</span>
package for examples of specific <code>match</code> and <code>%in%</code>
methods (defined for <a href="S4Vectors.html#topic+Hits">Hits</a> and
<a href="S4Vectors.html#topic+Rle">Rle</a> objects, respectively).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>match  # note the dispatch on the 'x' and 'table' args only
showMethods("match")
selectMethod("match", c("ANY", "ANY"))  # the default method

`%in%`
showMethods("%in%")
selectMethod("%in%", c("ANY", "ANY"))  # the default method
</code></pre>

<hr>
<h2 id='mean'>Arithmetic Mean</h2><span id='topic+mean'></span>

<h3>Description</h3>

<p>Generic function for the (trimmed) arithmetic mean.
</p>
<p>NOTE: This man page is for the <code>mean</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+mean">mean</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
(typically vector-like) not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mean(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mean_+3A_x">x</code></td>
<td>

<p>typically a vector-like object
</p>
</td></tr>
<tr><td><code id="mean_+3A_...">...</code></td>
<td>

<p>see <code><a href="base.html#topic+mean">mean</a></code>
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+mean">mean</a></code> for the value returned by the
default method.
</p>
<p>Specific methods defined in Bioconductor packages will typically
return an object of the same class as the input object.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+mean">mean</a></code> for the default <code>mean</code> method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+mean+2CRle-method">mean,Rle-method</a> in the <span class="pkg">S4Vectors</span> package
for an example of a specific <code>mean</code> method (defined for
<a href="S4Vectors.html#topic+Rle">Rle</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>mean
showMethods("mean")
selectMethod("mean", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='normalize'>Normalize an object</h2><span id='topic+normalize'></span>

<h3>Description</h3>

<p>A generic function which normalizes an object containing microarray data
or other data.
Normalization is intended to remove from the intensity measures any
systematic trends which arise from the microarray technology rather than
from differences between the probes or between the target RNA samples
hybridized to the arrays.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalize(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalize_+3A_object">object</code></td>
<td>

<p>A data object, typically containing microarray data.
</p>
</td></tr>
<tr><td><code id="normalize_+3A_...">...</code></td>
<td>

<p>Additional arguments, for use in specific methods.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object containing the normalized data.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="affy.html#topic+normalize+2CAffyBatch-method">normalize,AffyBatch-method</a> in the <span class="pkg">affy</span>
package and <a href="MSnbase.html#topic+normalize+2CMSnExp-method">normalize,MSnExp-method</a> in the
<span class="pkg">MSnbase</span> package for examples of specific <code>normalize</code>
methods (defined for <a href="affy.html#topic+AffyBatch">AffyBatch</a> and
<a href="MSnbase.html#topic+MSnExp">MSnExp</a> objects, respectively).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>normalize
showMethods("normalize")

library(affy)
showMethods("normalize")
selectMethod("normalize", "AffyBatch")
</code></pre>

<hr>
<h2 id='nrow'>The number of rows/columns of an array-like object</h2><span id='topic+nrow'></span><span id='topic+ncol'></span><span id='topic+NROW'></span><span id='topic+NCOL'></span>

<h3>Description</h3>

<p>Return the number of rows or columns present in an array-like object.
</p>
<p>NOTE: This man page is for the <code>nrow</code>, <code>ncol</code>, <code>NROW</code> and
<code>NCOL</code> <em>S4 generic functions</em> defined in the <span class="pkg">BiocGenerics</span>
package.
See <code>?base::<a href="base.html#topic+nrow">nrow</a></code> for the default methods (defined in
the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
(typically matrix- or array-like) not supported by the default methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nrow(x)
ncol(x)
NROW(x)
NCOL(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nrow_+3A_x">x</code></td>
<td>

<p>A matrix- or array-like object.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single integer or <code>NULL</code>.
</p>
<p>Specific methods defined in Bioconductor packages should 
behave as consistently as possible with the default methods.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+nrow">nrow</a></code> for the default <code>nrow</code>,
<code>ncol</code>, <code>NROW</code> and <code>NCOL</code> methods.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+nrow+2CDataFrame-method">nrow,DataFrame-method</a> in the <span class="pkg">S4Vectors</span>
package for an example of a specific <code>nrow</code> method (defined
for <a href="S4Vectors.html#topic+DataFrame">DataFrame</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>nrow
showMethods("nrow")
selectMethod("nrow", "ANY")  # the default method

ncol
showMethods("ncol")
selectMethod("ncol", "ANY")  # the default method

NROW
showMethods("NROW")
selectMethod("NROW", "ANY")  # the default method

NCOL
showMethods("NCOL")
selectMethod("NCOL", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='Ontology'>Generic Ontology getter</h2><span id='topic+Ontology'></span>

<h3>Description</h3>

<p>Get the Ontology of an object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ontology(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Ontology_+3A_object">object</code></td>
<td>

<p>An object with an Ontology.
</p>
</td></tr>
</table>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="AnnotationDbi.html#topic+Ontology+2CGOTerms-method">Ontology,GOTerms-method</a> in the
<span class="pkg">AnnotationDbi</span> package for an example of a specific
<code>Ontology</code> method (defined for <a href="AnnotationDbi.html#topic+GOTerms">GOTerms</a>
objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>Ontology
showMethods("Ontology")

library(AnnotationDbi)
showMethods("Ontology")
selectMethod("Ontology", "GOTerms")
</code></pre>

<hr>
<h2 id='order'>Ordering permutation</h2><span id='topic+order'></span>

<h3>Description</h3>

<p><code>order</code> returns a permutation which rearranges its first argument
into ascending or descending order, breaking ties by further
arguments.
</p>
<p>NOTE: This man page is for the <code>order</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+order">order</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
(typically vector-like) not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>order(..., na.last=TRUE, decreasing=FALSE, method=c("auto", "shell", "radix"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="order_+3A_...">...</code></td>
<td>

<p>One or more vector-like objects, all of the same length.
</p>
</td></tr>
<tr><td><code id="order_+3A_na.last">na.last</code>, <code id="order_+3A_decreasing">decreasing</code>, <code id="order_+3A_method">method</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+order">order</a></code> for a description of
these arguments.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The default method (see <code>?base::<a href="base.html#topic+order">order</a></code>) returns
an integer vector of length N where N is the common length of the
input objects. This integer vector represents a permutation of N
elements and can be used to rearrange the first argument in
<code>...</code> into ascending or descending order (by subsetting it).
</p>
<p>Specific methods defined in Bioconductor packages should also
return an integer vector representing a permutation of N elements.
</p>


<h3>Note</h3>

<p>TO DEVELOPERS:
</p>
<p>Specific <code>order</code> methods should preferably be made &quot;stable&quot; for
consistent behavior across platforms and consistency with
<code>base::order()</code>. Note that C qsort() is <em>not</em> &quot;stable&quot; so
<code>order</code> methods that use qsort() at the C-level need to ultimately
break ties by position, which can easily be done by adding a little
extra code at the end of the comparison function passed to qsort().
</p>
<p><code>order(x, decreasing=TRUE)</code> is <em>not</em> always equivalent to
<code>rev(order(x))</code>.
</p>
<p><code><a href="#topic+order">order</a></code>, <code><a href="#topic+sort">sort</a></code>,
and <code><a href="#topic+rank">rank</a></code> methods for specific vector-like
objects should adhere to the same underlying order that should be
conceptually defined as a binary relation on the set of all possible
vector values. For completeness, this binary relation should also be
incarnated by a <a href="base.html#topic++3C+3D">&lt;=</a> method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+order">order</a></code> for the default <code>order</code> method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="IRanges.html#topic+order+2CIntegerRanges-method">order,IntegerRanges-method</a> in the <span class="pkg">IRanges</span>
package for an example of a specific <code>order</code> method (defined
for <a href="IRanges.html#topic+IntegerRanges">IntegerRanges</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>order
showMethods("order")
selectMethod("order", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='organism_species'>Organism and species accessors</h2><span id='topic+organism_species'></span><span id='topic+organism'></span><span id='topic+organism+3C-'></span><span id='topic+species'></span><span id='topic+species+3C-'></span>

<h3>Description</h3>

<p>Get or set the organism and/or species of an object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>organism(object)
organism(object) &lt;- value

species(object)
species(object) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="organism_species_+3A_object">object</code></td>
<td>

<p>An object to get or set the organism or species of.
</p>
</td></tr>
<tr><td><code id="organism_species_+3A_value">value</code></td>
<td>

<p>The organism or species to set on <code>object</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>organism</code> should return the <em>scientific name</em> (i.e. genus and
species, or genus and species and subspecies) of the organism. Preferably
in the format <code>"Genus species"</code> (e.g. <code>"Homo sapiens"</code>)
or <code>"Genus species subspecies"</code> (e.g.
<code>"Homo sapiens neanderthalensis"</code>).
</p>
<p><code>species</code> should of course return the species of the organism.
Unfortunately there is a long history of misuse of this accessor in
Bioconductor so its usage is now discouraged (starting with BioC 3.1).
</p>


<h3>Note</h3>

<p>TO DEVELOPERS:
</p>
<p><code>species</code> has been historically misused in many places in Bioconductor
and is redundant with <code>organism</code>. So implementing the <code>species</code>
accessor is now discouraged (starting with BioC 3.1). The <code>organism</code>
accessor (returning the <em>scientific name</em>) should be implemented
instead.
</p>


<h3>See Also</h3>


<ul>
<li> <p><a href="http://bioconductor.org/packages/release/BiocViews.html#___Organism">http://bioconductor.org/packages/release/BiocViews.html#___Organism</a>
for browsing the annotation packages currently available in
Bioconductor by organism.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="annotate.html#topic+organism+2Ccharacter-method">organism,character-method</a> and
<a href="annotate.html#topic+organism+2CchromLocation-method">organism,chromLocation-method</a> in the
<span class="pkg">annotate</span> package for examples of specific <code>organism</code>
methods (defined for character and <a href="annotate.html#topic+chromLocation">chromLocation</a>
objects).
</p>
</li>
<li> <p><a href="AnnotationDbi.html#topic+species+2CAnnotationDb-method">species,AnnotationDb-method</a> in the
<span class="pkg">AnnotationDbi</span> package for an example of a specific
<code>species</code> method (defined for <a href="AnnotationDbi.html#topic+AnnotationDb">AnnotationDb</a>
objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## organism() getter:
organism
showMethods("organism")

library(annotate)
showMethods("organism")
selectMethod("organism", "character")
selectMethod("organism", "chromLocation")

## organism() setter:
`organism&lt;-`
showMethods("organism&lt;-")

## species() getter:
species
showMethods("species")

library(AnnotationDbi)
selectMethod("species", "AnnotationDb")

## species() setter:
`species&lt;-`
showMethods("species&lt;-")
</code></pre>

<hr>
<h2 id='paste'>Concatenate strings</h2><span id='topic+paste'></span>

<h3>Description</h3>

<p><code>paste</code> concatenates vectors of strings or vector-like
objects containing strings.
</p>
<p>NOTE: This man page is for the <code>paste</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+paste">paste</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
(typically vector-like objects containing strings) not supported by
the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>paste(..., sep=" ", collapse=NULL, recycle0=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="paste_+3A_...">...</code></td>
<td>

<p>One or more vector-like objects containing strings.
</p>
</td></tr>
<tr><td><code id="paste_+3A_sep">sep</code>, <code id="paste_+3A_collapse">collapse</code>, <code id="paste_+3A_recycle0">recycle0</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+paste">paste</a></code> for a description of
these arguments.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+paste">paste</a></code> for the value returned by the
default method.
</p>
<p>Specific methods defined in Bioconductor packages will typically
return an object of the same class as the input objects.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+paste">paste</a></code> for the default <code>paste</code> method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+paste+2CRle-method">paste,Rle-method</a> in the <span class="pkg">S4Vectors</span> package
for an example of a specific <code>paste</code> method (defined for
<a href="S4Vectors.html#topic+Rle">Rle</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>paste
showMethods("paste")
selectMethod("paste", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='path'>Accessing the path of an object</h2><span id='topic+path'></span><span id='topic+path+3C-'></span><span id='topic+basename'></span><span id='topic+basename+3C-'></span><span id='topic+dirname'></span><span id='topic+dirname+3C-'></span><span id='topic+basename+2CANY-method'></span><span id='topic+dirname+2CANY-method'></span><span id='topic+basename+3C-+2Ccharacter-method'></span><span id='topic+basename+3C-+2CANY-method'></span><span id='topic+dirname+3C-+2Ccharacter-method'></span><span id='topic+dirname+3C-+2CANY-method'></span>

<h3>Description</h3>

<p>Get or set the path of an object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>path(object, ...)
path(object, ...) &lt;- value

basename(path, ...)
basename(path, ...) &lt;- value

dirname(path, ...)
dirname(path, ...) &lt;- value

## The purpose of the following methods is to make the basename() and
## dirname() getters work out-of-the-box on any object for which the
## path() getter works.

## S4 method for signature 'ANY'
basename(path, ...)

## S4 method for signature 'ANY'
dirname(path, ...)

## The purpose of the following replacement methods is to make the
## basename() and dirname() setters work out-of-the-box on any object
## for which the path() getter and setter work.

## S4 replacement method for signature 'character'
basename(path, ...) &lt;- value

## S4 replacement method for signature 'ANY'
basename(path, ...) &lt;- value

## S4 replacement method for signature 'character'
dirname(path, ...) &lt;- value

## S4 replacement method for signature 'ANY'
dirname(path, ...) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="path_+3A_object">object</code></td>
<td>

<p>An object containing paths. Even though it will typically contain
a single path, <code>object</code> can actually contain an arbitrary
number of paths.
</p>
</td></tr>
<tr><td><code id="path_+3A_...">...</code></td>
<td>

<p>Additional arguments, for use in specific methods.
</p>
</td></tr>
<tr><td><code id="path_+3A_value">value</code></td>
<td>

<p>For <code>path&lt;-</code>, the paths to set on <code>object</code>.
</p>
<p>For <code>basename&lt;-</code> or <code>dirname&lt;-</code>, the basenames or dirnames
to set on <code>path</code>.
</p>
</td></tr>
<tr><td><code id="path_+3A_path">path</code></td>
<td>

<p>A character vector <em>or an object containing paths</em>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector for <code>path(object)</code>, <code>basename(path)</code>,
and <code>dirname(path)</code>. Typically of length 1 but not necessarily.
Possibly with names on it for <code>path(object)</code>.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+basename">basename</a></code> for the functions the
<code>basename</code> and <code>dirname</code> generics are based on.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="Rsamtools.html#topic+path+2CRsamtoolsFile-method">path,RsamtoolsFile-method</a> in the
<span class="pkg">Rsamtools</span> package for an example of a specific
<code>path</code> method (defined for <a href="Rsamtools.html#topic+RsamtoolsFile">RsamtoolsFile</a>
objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## ---------------------------------------------------------------------
## GENERIC FUNCTIONS AND DEFAULT METHODS
## ---------------------------------------------------------------------

path
showMethods("path")

`path&lt;-`
showMethods("path&lt;-")

basename
showMethods("basename")

`basename&lt;-`
showMethods("basename&lt;-")

dirname
showMethods("dirname")

`dirname`
showMethods("dirname&lt;-")


## Default basename() and dirname() getters:
selectMethod("basename", "ANY")
selectMethod("dirname", "ANY")

## Default basename() and dirname() setters:
selectMethod("basename&lt;-", "character")
selectMethod("basename&lt;-", "ANY")
selectMethod("dirname&lt;-", "character")
selectMethod("dirname&lt;-", "ANY")

## ---------------------------------------------------------------------
## OBJECTS CONTAINING PATHS
## ---------------------------------------------------------------------

## Let's define a simple class to represent objects that contain paths:
setClass("A", slots=c(stuff="ANY", path="character"))

a &lt;- new("A", stuff=runif(5),
              path=c(one="path/to/file1", two="path/to/file2"))

## path() getter:
setMethod("path", "A", function(object) object@path)

path(a)

## Because the path() getter works on 'a', now the basename() and
## dirname() getters also work:
basename(a)
dirname(a)

## path() setter:
setReplaceMethod("path", "A",
    function(object, ..., value)
    {
        if (length(list(...)) != 0L) {
            dots &lt;- match.call(expand.dots=FALSE)[[3L]]
            stop(BiocGenerics:::unused_arguments_msg(dots))
        }
        object@path &lt;- value
        object
    }
)

a &lt;- new("A", stuff=runif(5))
path(a) &lt;- c(one="path/to/file1", two="path/to/file2")
path(a)

## Because the path() getter and setter work on 'a', now the basename()
## and dirname() setters also work:
basename(a) &lt;- toupper(basename(a))
path(a)
dirname(a) &lt;- "~/MyDataFiles"
path(a)
</code></pre>

<hr>
<h2 id='plotMA'>MA-plot: plot differences versus averages for high-throughput data</h2><span id='topic+plotMA'></span><span id='topic+plotMA+2CANY-method'></span>

<h3>Description</h3>

<p>A generic function which produces an MA-plot for an object containing
microarray, RNA-Seq or other data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotMA(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotMA_+3A_object">object</code></td>
<td>

<p>A data object, typically containing count values from an RNA-Seq
experiment or microarray intensity values.
</p>
</td></tr>
<tr><td><code id="plotMA_+3A_...">...</code></td>
<td>

<p>Additional arguments, for use in specific methods.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Undefined. The function exists for its side effect, producing a plot.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><code><a href="limma.html#topic+plotMA">plotMA</a></code> in the <span class="pkg">limma</span> package
for a function with the same name that is not dispatched through
this generic function.
</p>
</li>
<li> <p><code><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a></code> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>showMethods("plotMA")

suppressWarnings(
  if(require("DESeq2"))
    example("plotMA", package="DESeq2", local=TRUE)
)
</code></pre>

<hr>
<h2 id='plotPCA'>PCA-plot: Principal Component Analysis plot</h2><span id='topic+plotPCA'></span>

<h3>Description</h3>

<p>A generic function which produces a PCA-plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPCA(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotPCA_+3A_object">object</code></td>
<td>

<p>A data object, typically containing gene expression information.
</p>
</td></tr>
<tr><td><code id="plotPCA_+3A_...">...</code></td>
<td>

<p>Additional arguments, for use in specific methods.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Undefined. The function exists for its side effect, producing a plot.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><code><a href="DESeq2.html#topic+plotPCA">plotPCA</a></code> in the <span class="pkg">DESeq2</span> package
for an example method that uses this generic.
</p>
</li>
<li> <p><code><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a></code> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>showMethods("plotPCA")

suppressWarnings(
  if(require("DESeq2"))
    example("plotPCA", package="DESeq2", local=TRUE)
)
</code></pre>

<hr>
<h2 id='rank'>Ranks the values in a vector-like object</h2><span id='topic+rank'></span>

<h3>Description</h3>

<p>Returns the ranks of the values in a vector-like object.
Ties (i.e., equal values) and missing values can be handled in several ways.
</p>
<p>NOTE: This man page is for the <code>rank</code>
<em>S4 generic function</em> defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+rank">rank</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rank(x, na.last=TRUE,
     ties.method=c("average", "first", "last", "random", "max", "min"),
     ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rank_+3A_x">x</code></td>
<td>

<p>A vector-like object.
</p>
</td></tr>
<tr><td><code id="rank_+3A_na.last">na.last</code>, <code id="rank_+3A_ties.method">ties.method</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+rank">rank</a></code> for a description of
these arguments.
</p>
</td></tr>
<tr><td><code id="rank_+3A_...">...</code></td>
<td>

<p>Additional arguments, for use in specific methods.
</p>
<p>Note that <code>base::<a href="base.html#topic+rank">rank</a></code> (the default method) only
takes the <code>x</code>, <code>na.last</code>, and <code>ties.method</code> arguments.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+rank">rank</a></code> for the value returned
by the default method.
</p>
<p>Specific methods defined in Bioconductor packages should
behave as consistently as possible with the default method.
</p>


<h3>Note</h3>

<p>TO DEVELOPERS:
</p>
<p>See note in <code>?BiocGenerics::<a href="#topic+order">order</a></code> 
about &quot;stable&quot; order.
</p>
<p><code><a href="#topic+order">order</a></code>, <code><a href="#topic+sort">sort</a></code>,
and <code><a href="#topic+rank">rank</a></code> methods for specific vector-like
objects should adhere to the same underlying order that should be
conceptually defined as a binary relation on the set of all possible
vector values. For completeness, this binary relation should also be
incarnated by a <a href="base.html#topic++3C+3D">&lt;=</a> method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+rank">rank</a></code> for the default <code>rank</code> method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+rank+2CVector-method">rank,Vector-method</a> in the <span class="pkg">S4Vectors</span> package
for an example of a specific <code>rank</code> method (defined for
<a href="S4Vectors.html#topic+Vector">Vector</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>rank  # note the dispatch on the 'x' arg only
showMethods("rank")
selectMethod("rank", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='relist'>Re-listing an unlist()ed object</h2><span id='topic+relist'></span>

<h3>Description</h3>

<p><code>relist</code> is a generic function with a few methods in order to allow
easy inversion of <code>unlist(x)</code>.
</p>
<p>NOTE: This man page is for the <code>relist</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?utils::<a href="utils.html#topic+relist">relist</a></code> for the default method
(defined in the <span class="pkg">utils</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relist(flesh, skeleton)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="relist_+3A_flesh">flesh</code></td>
<td>

<p>A vector-like object.
</p>
</td></tr>
<tr><td><code id="relist_+3A_skeleton">skeleton</code></td>
<td>

<p>A list-like object. Only the &quot;shape&quot; (i.e. the lengths of the individual
list elements) of <code>skeleton</code> matters. Its exact content is ignored.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list-like object with the same &quot;shape&quot; as <code>skeleton</code> and that would
give <code>flesh</code> back if unlist()ed.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>utils::<a href="utils.html#topic+relist">relist</a></code> for the default <code>relist</code>
method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="IRanges.html#topic+relist+2CANY+2CList-method">relist,ANY,List-method</a> in the <span class="pkg">IRanges</span>
package for an example of a specific <code>relist</code> method (defined
for when <code>skeleton</code> is a <a href="S4Vectors.html#topic+List">List</a> object).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>relist
showMethods("relist")
selectMethod("relist", c("ANY", "ANY"))  # the default method
</code></pre>

<hr>
<h2 id='rep'>Replicate elements of a vector-like object</h2><span id='topic+rep.int'></span>

<h3>Description</h3>

<p><code>rep.int</code> replicates the elements in <code>x</code>.
</p>
<p>NOTE: This man page is for the <code>rep.int</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+rep.int">rep.int</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
(typically vector-like) not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rep.int(x, times)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rep_+3A_x">x</code></td>
<td>

<p>The object to replicate (typically vector-like).
</p>
</td></tr>
<tr><td><code id="rep_+3A_times">times</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+rep.int">rep.int</a></code> for a description of
this argument.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+rep.int">rep.int</a></code> for the value returned by the
default method.
</p>
<p>Specific methods defined in Bioconductor packages will typically
return an object of the same class as the input object.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+rep.int">rep.int</a></code> for the default <code>rep.int</code>,
<code>intersect</code>, and <code>setdiff</code> methods.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+rep.int+2CRle-method">rep.int,Rle-method</a> in the <span class="pkg">S4Vectors</span> package
for an example of a specific <code>rep.int</code> method (defined for
<a href="S4Vectors.html#topic+Rle">Rle</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>rep.int
showMethods("rep.int")
selectMethod("rep.int", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='residuals'>Extract model residuals</h2><span id='topic+residuals'></span>

<h3>Description</h3>

<p><code>residuals</code> is a generic function which extracts model residuals
from objects returned by modeling functions.
</p>
<p>NOTE: This man page is for the <code>residuals</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?stats::<a href="stats.html#topic+residuals">residuals</a></code> for the default method
(defined in the <span class="pkg">stats</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>residuals(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="residuals_+3A_object">object</code>, <code id="residuals_+3A_...">...</code></td>
<td>

<p>See <code>?stats::<a href="stats.html#topic+residuals">residuals</a></code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Residuals extracted from the object <code>object</code>.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>stats::<a href="stats.html#topic+residuals">residuals</a></code> for the default
<code>residuals</code> method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="affyPLM.html#topic+residuals+2CPLMset-method">residuals,PLMset-method</a> in the <span class="pkg">affyPLM</span> package
for an example of a specific <code>residuals</code> method (defined for
<a href="affyPLM.html#topic+PLMset">PLMset</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>residuals
showMethods("residuals")
selectMethod("residuals", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='row+2Bcolnames'>Row and column names</h2><span id='topic+row+2Bcolnames'></span><span id='topic+rownames'></span><span id='topic+rownames+3C-'></span><span id='topic+colnames'></span><span id='topic+colnames+3C-'></span>

<h3>Description</h3>

<p>Get or set the row or column names of a matrix-like object.
</p>
<p>NOTE: This man page is for the <code>rownames</code>, <code>`rownames&lt;-`</code>,
<code>colnames</code>, and <code>`colnames&lt;-`</code> <em>S4 generic functions</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+rownames">rownames</a></code> for the default methods
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
(typically matrix-like) not supported by the default methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rownames(x, do.NULL=TRUE, prefix="row")
rownames(x) &lt;- value

colnames(x, do.NULL=TRUE, prefix="col")
colnames(x) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="row+2B2Bcolnames_+3A_x">x</code></td>
<td>

<p>A matrix-like object.
</p>
</td></tr>
<tr><td><code id="row+2B2Bcolnames_+3A_do.null">do.NULL</code>, <code id="row+2B2Bcolnames_+3A_prefix">prefix</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+rownames">rownames</a></code> for a description of
these arguments.
</p>
</td></tr>
<tr><td><code id="row+2B2Bcolnames_+3A_value">value</code></td>
<td>

<p>Either <code>NULL</code> or a character vector equal of length equal to the
appropriate dimension.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The getters will return <code>NULL</code> or a character vector of length
<code><a href="base.html#topic+nrow">nrow</a>(x)</code> for <code>rownames</code> and length <code><a href="base.html#topic+ncol">ncol</a>(x)</code>
for <code>colnames(x)</code>.
</p>
<p>See <code>?base::<a href="base.html#topic+rownames">rownames</a></code> for more information about the
default methods, including how the setters are expected to behave.
</p>
<p>Specific methods defined in Bioconductor packages should 
behave as consistently as possible with the default methods.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+rownames">rownames</a></code> for the default <code>rownames</code>,
<code>`rownames&lt;-`</code>, <code>colnames</code>, and <code>`colnames&lt;-`</code>
methods.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+rownames+2CDataFrame-method">rownames,DataFrame-method</a> in the <span class="pkg">S4Vectors</span>
package for an example of a specific <code>rownames</code> method (defined
for <a href="S4Vectors.html#topic+DataFrame">DataFrame</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## rownames() getter:
rownames  # note the dispatch on the 'x' arg only
showMethods("rownames")
selectMethod("rownames", "ANY")  # the default method

## rownames() setter:
`rownames&lt;-`
showMethods("rownames&lt;-")
selectMethod("rownames&lt;-", "ANY")  # the default method

## colnames() getter:
colnames  # note the dispatch on the 'x' arg only
showMethods("colnames")
selectMethod("colnames", "ANY")  # the default method

## colnames() setter:
`colnames&lt;-`
showMethods("colnames&lt;-")
selectMethod("colnames&lt;-", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='S3-classes-as-S4-classes'>S3 classes as S4 classes</h2><span id='topic+S3-classes-as-S4-classes'></span><span id='topic+connection-class'></span><span id='topic+file-class'></span><span id='topic+url-class'></span><span id='topic+gzfile-class'></span><span id='topic+bzfile-class'></span><span id='topic+unz-class'></span><span id='topic+pipe-class'></span><span id='topic+fifo-class'></span><span id='topic+sockconn-class'></span><span id='topic+terminal-class'></span><span id='topic+textConnection-class'></span><span id='topic+gzcon-class'></span><span id='topic+character_OR_connection-class'></span><span id='topic+AsIs-class'></span><span id='topic+dist-class'></span>

<h3>Description</h3>

<p>Some old-style (aka S3) classes are turned into formally defined (aka S4)
classes by the <span class="pkg">BiocGenerics</span> package. This allows S4 methods defined in
Bioconductor packages to use them in their signatures.
</p>


<h3>Details</h3>

<p>S3 classes currently turned into S4 classes:
</p>

<ul>
<li><p> connection class and subclasses:
<a href="base.html#topic+connection">connection</a>,
file, url, gzfile, bzfile, unz, pipe,
fifo, sockconn, terminal, textConnection,
gzcon.
Addtitionally the character_OR_connection S4 class is
defined as the union of classes character and connection.
</p>
</li>
<li><p> others: <a href="base.html#topic+AsIs">AsIs</a>, <a href="stats.html#topic+dist">dist</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="methods.html#topic+setOldClass">setOldClass</a> and <a href="methods.html#topic+setClassUnion">setClassUnion</a> in the <span class="pkg">methods</span> package.
</p>

<hr>
<h2 id='score'>Score accessor</h2><span id='topic+score'></span><span id='topic+score+3C-'></span>

<h3>Description</h3>

<p>Get or set the score value contained in an object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>score(x, ...)
score(x, ...) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="score_+3A_x">x</code></td>
<td>

<p>An object to get or set the score value of.
</p>
</td></tr>
<tr><td><code id="score_+3A_...">...</code></td>
<td>

<p>Additional arguments, for use in specific methods.
</p>
</td></tr>
<tr><td><code id="score_+3A_value">value</code></td>
<td>

<p>The score value to set on <code>x</code>.
</p>
</td></tr>
</table>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="GenomicRanges.html#topic+score+2CGenomicRanges-method">score,GenomicRanges-method</a> in the
<span class="pkg">GenomicRanges</span> package for an example of a specific
<code>score</code> method (defined for <a href="GenomicRanges.html#topic+GenomicRanges">GenomicRanges</a>
objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>score
showMethods("score")

`score&lt;-`
showMethods("score&lt;-")

library(GenomicRanges)

showMethods("score")
selectMethod("score", "GenomicRanges")

showMethods("score&lt;-")
selectMethod("score&lt;-", "GenomicRanges")
</code></pre>

<hr>
<h2 id='sets'>Set operations</h2><span id='topic+sets'></span><span id='topic+union'></span><span id='topic+intersect'></span><span id='topic+setdiff'></span>

<h3>Description</h3>

<p>Performs <em>set</em> union, intersection and (asymmetric!) difference
on two vector-like objects.
</p>
<p>NOTE: This man page is for the <code>union</code>, <code>intersect</code> and
<code>setdiff</code> <em>S4 generic functions</em> defined in the <span class="pkg">BiocGenerics</span>
package.
See <code>?base::<a href="base.html#topic+union">union</a></code> for the default methods
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
(typically vector-like) not supported by the default methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>union(x, y, ...)
intersect(x, y, ...)
setdiff(x, y, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sets_+3A_x">x</code>, <code id="sets_+3A_y">y</code></td>
<td>

<p>Vector-like objects (typically of the same class, but not necessarily).
</p>
</td></tr>
<tr><td><code id="sets_+3A_...">...</code></td>
<td>

<p>Additional arguments, for use in specific methods.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+union">union</a></code> for the value returned by the
default methods.
</p>
<p>Specific methods defined in Bioconductor packages will typically
return an object of the same class as the input objects.
</p>


<h3>Note</h3>

<p>The default methods (defined in the <span class="pkg">base</span> package) only take 2
arguments. We've added the <code>...</code> argument to the generic functions
defined in the <span class="pkg">BiocGenerics</span> package so they can be called with an
arbitrary number of effective arguments.
For <code>union</code> or <code>intersect</code>, this typically allows Bioconductor
packages to define methods that compute the union or intersection of more
than 2 objects. However, for <code>setdiff</code>, which is conceptually a
binary operation, this typically allows methods to add extra arguments
for controlling/altering the behavior of the operation.
Like for example the <code>ignore.strand</code> argument supported by the
<code>setdiff</code> method for <a href="GenomicRanges.html#topic+GenomicRanges">GenomicRanges</a> objects
(defined in the <span class="pkg">GenomicRanges</span> package). (Note that the <code>union</code>
and <code>intersect</code> methods for those objects also support the
<code>ignore.strand</code> argument.)
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+union">union</a></code> for the default <code>union</code>,
<code>intersect</code>, and <code>setdiff</code> methods.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="GenomicRanges.html#topic+union+2CGenomicRanges+2CGenomicRanges-method">union,GenomicRanges,GenomicRanges-method</a> in
the <span class="pkg">GenomicRanges</span> package for examples of specific
<code>union</code>, <code>intersect</code>, and <code>setdiff</code> methods (defined
for <a href="GenomicRanges.html#topic+GenomicRanges">GenomicRanges</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>union
showMethods("union")
selectMethod("union", c("ANY", "ANY"))  # the default method

intersect
showMethods("intersect")
selectMethod("intersect", c("ANY", "ANY"))  # the default method

setdiff
showMethods("setdiff")
selectMethod("setdiff", c("ANY", "ANY"))  # the default method
</code></pre>

<hr>
<h2 id='sort'>Sorting a vector-like object</h2><span id='topic+sort'></span>

<h3>Description</h3>

<p>Sort a vector-like object into ascending or descending order.
</p>
<p>NOTE: This man page is for the <code>sort</code>
<em>S4 generic function</em> defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+sort">sort</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sort(x, decreasing=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sort_+3A_x">x</code></td>
<td>

<p>A vector-like object.
</p>
</td></tr>
<tr><td><code id="sort_+3A_decreasing">decreasing</code>, <code id="sort_+3A_...">...</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+sort">sort</a></code> for a description of
these arguments.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+sort">sort</a></code> for the value returned
by the default method.
</p>
<p>Specific methods defined in Bioconductor packages should
behave as consistently as possible with the default method.
</p>


<h3>Note</h3>

<p>TO DEVELOPERS:
</p>
<p>See note in <code>?BiocGenerics::<a href="#topic+order">order</a></code>
about &quot;stable&quot; order.
</p>
<p><code><a href="#topic+order">order</a></code>, <code><a href="#topic+sort">sort</a></code>,
and <code><a href="#topic+rank">rank</a></code> methods for specific vector-like
objects should adhere to the same underlying order that should be
conceptually defined as a binary relation on the set of all possible
vector values. For completeness, this binary relation should also be
incarnated by a <a href="base.html#topic++3C+3D">&lt;=</a> method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+sort">sort</a></code> for the default
<code>sort</code> method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+sort+2CVector-method">sort,Vector-method</a> in the <span class="pkg">S4Vectors</span> package
for an example of a specific <code>sort</code> method (defined for
<a href="S4Vectors.html#topic+Vector">Vector</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>sort  # note the dispatch on the 'x' arg only
showMethods("sort")
selectMethod("sort", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='start'>The start(), end(), width(), and pos() generic getters and setters</h2><span id='topic+start'></span><span id='topic+start+3C-'></span><span id='topic+end'></span><span id='topic+end+3C-'></span><span id='topic+width'></span><span id='topic+width+3C-'></span><span id='topic+pos'></span>

<h3>Description</h3>

<p>Get or set the start, end, width, or single positions stored in an object.
</p>
<p>NOTE: This man page is for the <code>start</code>, <code>`start&lt;-`</code>,
<code>end</code>, <code>`end&lt;-`</code>, <code>width</code>, <code>`width&lt;-`</code>, and
<code>pos</code> <em>S4 generic functions</em> defined in the <span class="pkg">BiocGenerics</span>
package.
See <code>?stats::<a href="stats.html#topic+start">start</a></code> for the <code>start</code> and <code>end</code>
S3 generics defined in the <span class="pkg">stats</span> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>start(x, ...)
start(x, ...) &lt;- value

end(x, ...)
end(x, ...) &lt;- value

width(x)
width(x, ...) &lt;- value

pos(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="start_+3A_x">x</code></td>
<td>

<p>For the <code>start()</code>, <code>end()</code>, and <code>width()</code> getters/setters:
an object containing start, end, and width values.
</p>
<p>For the <code>pos{}</code> getter: an object containing single positions.
</p>
</td></tr>
<tr><td><code id="start_+3A_...">...</code></td>
<td>

<p>Additional arguments, for use in specific methods.
</p>
</td></tr>
<tr><td><code id="start_+3A_value">value</code></td>
<td>

<p>The start, end, or width values to set on <code>x</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See specific methods defined in Bioconductor packages.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>stats::<a href="stats.html#topic+start">start</a></code> in the <span class="pkg">stats</span> package
for the <code>start</code> and <code>end</code> S3 generics.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="IRanges.html#topic+start+2CIRanges-method">start,IRanges-method</a> in the <span class="pkg">IRanges</span>
package for examples of specific <code>start</code>, <code>end</code>, and
<code>width</code> methods (defined for <a href="IRanges.html#topic+IRanges">IRanges</a> objects).
</p>
</li>
<li> <p><a href="IRanges.html#topic+pos+2CUnstitchedIPos-method">pos,UnstitchedIPos-method</a> in the <span class="pkg">IRanges</span>
package for an example of a specific <code>pos</code> method
(defined for <a href="IRanges.html#topic+UnstitchedIPos">UnstitchedIPos</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## start() getter:
start
showMethods("start")

library(IRanges)
showMethods("start")
selectMethod("start", "IRanges")  # start() getter for IRanges objects

## start() setter:
`start&lt;-`
showMethods("start&lt;-")
selectMethod("start&lt;-", "IRanges")  # start() setter for IRanges objects

## end() getter:
end
showMethods("end")
selectMethod("end", "IRanges")  # end() getter for IRanges objects

## end() setter:
`end&lt;-`
showMethods("end&lt;-")
selectMethod("end&lt;-", "IRanges")  # end() setter for IRanges objects

## width() getter:
width
showMethods("width")
selectMethod("width", "IRanges")  # width() getter for IRanges objects

## width() setter:
`width&lt;-`
showMethods("width&lt;-")
selectMethod("width&lt;-", "IRanges")  # width() setter for IRanges objects

## pos() getter:
pos
showMethods("pos")
selectMethod("pos", "UnstitchedIPos")  # pos() getter for UnstitchedIPos
                                       # objects
</code></pre>

<hr>
<h2 id='strand'>Accessing strand information</h2><span id='topic+strand'></span><span id='topic+strand+3C-'></span><span id='topic+unstrand'></span><span id='topic+invertStrand'></span><span id='topic+invertStrand+2CANY-method'></span>

<h3>Description</h3>

<p>Get or set the strand information contained in an object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>strand(x, ...)
strand(x, ...) &lt;- value

unstrand(x)

invertStrand(x)
## S4 method for signature 'ANY'
invertStrand(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="strand_+3A_x">x</code></td>
<td>

<p>An object containing strand information.
</p>
</td></tr>
<tr><td><code id="strand_+3A_...">...</code></td>
<td>

<p>Additional arguments, for use in specific methods.
</p>
</td></tr>
<tr><td><code id="strand_+3A_value">value</code></td>
<td>

<p>The strand information to set on <code>x</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All the <code>strand</code> methods defined in the <span class="pkg">GenomicRanges</span> package
use the same set of 3 values (called the &quot;standard strand levels&quot;) to
specify the strand of a genomic location: <code>+</code>, <code>-</code>, and <code>*</code>.
<code>*</code> is used when the exact strand of the location is unknown,
or irrelevant, or when the &quot;feature&quot; at that location belongs to
both strands.
</p>
<p>Note that <code>unstrand</code> is not a generic function, just a convenience
wrapper to the generic <code>strand()</code> setter (<code>strand&lt;-</code>) that does:
</p>
<pre>  strand(x) &lt;- "*"
  x
</pre>
<p>The default method for <code>invertStrand</code> does:
</p>
<pre>  strand(x) &lt;- invertStrand(strand(x))
  x
</pre>


<h3>Value</h3>

<p>If <code>x</code> is a vector-like object, <code>strand(x)</code> will typically
return a vector-like object <em>parallel</em> to <code>x</code>, that is, an
object of the same length as <code>x</code> where the i-th element describes
the strand of the i-th element in <code>x</code>.
</p>
<p><code>unstrand(x)</code> and <code>invertStrand(x)</code> return a copy of <code>x</code>
with the strand set to <code>"*"</code> for <code>unstrand</code> or inverted for
<code>invertStrand</code> (i.e. <code>"+"</code> and <code>"-"</code> switched, and
<code>"*"</code> untouched).
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="GenomicRanges.html#topic+strand+2CGRanges-method">strand,GRanges-method</a> in the
<span class="pkg">GenomicRanges</span> package for an example of a specific
<code>strand</code> method (defined for <a href="GenomicRanges.html#topic+GRanges">GRanges</a>
objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>strand
showMethods("strand")

`strand&lt;-`
showMethods("strand&lt;-")

unstrand

invertStrand
showMethods("invertStrand")
selectMethod("invertStrand", "ANY")  # the default method

library(GenomicRanges)

showMethods("strand")
selectMethod("strand", "missing")
strand()

showMethods("strand&lt;-")
</code></pre>

<hr>
<h2 id='subset'>Subsetting vector-like, matrix-like and data-frame-like objects</h2><span id='topic+subset'></span>

<h3>Description</h3>

<p>Return subsets of vector-like, matrix-like or data-frame-like objects
which meet conditions.
</p>
<p>NOTE: This man page is for the <code>subset</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+subset">subset</a></code> for the <code>subset</code> S3 generic
defined in the <span class="pkg">base</span> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subset(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset_+3A_x">x</code></td>
<td>

<p>A vector-like, matrix-like or data-frame-like object to be subsetted.
</p>
</td></tr>
<tr><td><code id="subset_+3A_...">...</code></td>
<td>

<p>Additional arguments (e.g. <code>subset</code>, <code>select</code>, <code>drop</code>),
for use in specific methods. See <code>?base::<a href="base.html#topic+subset">subset</a></code> for
more information.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object similar to <code>x</code> containing just the selected elements (for a
vector-like object), or the selected rows and columns (for a matrix-like or
data-frame-like object).
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+subset">subset</a></code> in the <span class="pkg">base</span> package
for the <code>subset</code> S3 generic.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+subset+2CRectangularData-method">subset,RectangularData-method</a> in the <span class="pkg">S4Vectors</span>
package for an example of a specific <code>subset</code> method (defined
for <a href="S4Vectors.html#topic+RectangularData">RectangularData</a> derivatives).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>subset
showMethods("subset")
selectMethod("subset", "ANY")  # the default method

library(S4Vectors)
showMethods("subset")
## The subset() method for RectangularData derivatives:
selectMethod("subset", "RectangularData")
</code></pre>

<hr>
<h2 id='t'>Matrix Transpose</h2><span id='topic+t'></span>

<h3>Description</h3>

<p>Given a rectangular object <code>x</code>, <code>t</code> returns the
transpose of <code>x</code>.
</p>
<p>NOTE: This man page is for the <code>t</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+t">t</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
(typically array-like) not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>t(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="t_+3A_x">x</code></td>
<td>

<p>A matrix-like or other rectangular object.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+t">t</a></code> for the value returned by the
default method.
</p>
<p>Specific methods defined in Bioconductor packages will typically
return an object of the same class as the input object.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+t">t</a></code> for the default <code>t</code> method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+t+2CHits-method">t,Hits-method</a> in the <span class="pkg">S4Vectors</span> package
for an example of a specific <code>t</code> method (defined for
<a href="S4Vectors.html#topic+Hits">Hits</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>t
showMethods("t")
selectMethod("t", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='table'>Cross tabulation and table creation</h2><span id='topic+table'></span>

<h3>Description</h3>

<p><code>table</code> uses the cross-classifying factors to build a contingency
table of the counts at each combination of factor levels.
</p>
<p>NOTE: This man page is for the <code>table</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+table">table</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>table(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="table_+3A_...">...</code></td>
<td>

<p>One or more objects which can be interpreted as factors
(including character strings), or a list (or data frame)
whose components can be so interpreted.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+table">table</a></code> for the value returned by the
default method.
</p>
<p>Specific methods defined in Bioconductor packages should also
return the type of object returned by the default method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+table">table</a></code> for the default <code>table</code> method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+table+2CRle-method">table,Rle-method</a> in the <span class="pkg">S4Vectors</span> package
for an example of a specific <code>table</code> method (defined for
<a href="S4Vectors.html#topic+Rle">Rle</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>table
showMethods("table")
selectMethod("table", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='tapply'>Apply a function over a ragged array</h2><span id='topic+tapply'></span>

<h3>Description</h3>

<p><code>tapply</code> applies a function to each cell of a ragged array,
that is to each (non-empty) group of values given by a unique
combination of the levels of certain factors.
</p>
<p>NOTE: This man page is for the <code>tapply</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+tapply">tapply</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
(typically list-like or vector-like) not supported by the
default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tapply(X, INDEX, FUN=NULL, ..., default=NA, simplify=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tapply_+3A_x">X</code></td>
<td>

<p>The default method expects an atomic object, typically a vector.
See <code>?base::<a href="base.html#topic+tapply">tapply</a></code> for the details.
</p>
<p>Specific methods can support other objects (typically list-like
or vector-like).
Please refer to the documentation of a particular method for the details.
</p>
</td></tr>
<tr><td><code id="tapply_+3A_index">INDEX</code></td>
<td>

<p>The default method expects a list of one or more factors, each of same
length as <code>X</code>.
See <code>?base::<a href="base.html#topic+tapply">tapply</a></code> for the details.
</p>
<p>Specific methods can support other objects (typically list-like).
Please refer to the documentation of a particular method for the details.
</p>
</td></tr>
<tr><td><code id="tapply_+3A_fun">FUN</code>, <code id="tapply_+3A_...">...</code>, <code id="tapply_+3A_default">default</code>, <code id="tapply_+3A_simplify">simplify</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+tapply">tapply</a></code> for a description of
these arguments.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+tapply">tapply</a></code> for the value returned by the
default method.
</p>
<p>Specific methods defined in Bioconductor packages should 
behave as consistently as possible with the default method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+tapply">tapply</a></code> for the default <code>tapply</code>
method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="IRanges.html#topic+tapply+2CVector+2CANY-method">tapply,Vector,ANY-method</a> in the <span class="pkg">IRanges</span>
package for an example of a specific <code>tapply</code> method (defined
for <a href="S4Vectors.html#topic+Vector">Vector</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>tapply  # note the dispatch on the 'X' and 'INDEX' args only
showMethods("tapply")
selectMethod("tapply", c("ANY", "ANY"))  # the default method
</code></pre>

<hr>
<h2 id='testPackage'>Run RUnit package unit tests</h2><span id='topic+testPackage'></span>

<h3>Description</h3>

<p><code>testPackage</code> helps developers implement unit tests using the
<span class="pkg">RUnit</span> testing conventions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>testPackage(pkgname=NULL, subdir="unitTests", pattern="^test_.*\\.R$",
    path=getwd())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="testPackage_+3A_pkgname">pkgname</code></td>
<td>

<p>The name of the package whose installed unit tests are to be run. A
missing or NULL value implies that the <code>testPackage</code> command
will look for tests within the package source directory indicated by
<code>path</code>.
</p>
</td></tr>
<tr><td><code id="testPackage_+3A_subdir">subdir</code></td>
<td>

<p>A character(1) vector providing the subdirectory in which unit tests
are located. The directory is searched first in the (installed or
source) package root, or in a subdirectory <code>inst/</code> below the
root.
</p>
</td></tr>
<tr><td><code id="testPackage_+3A_pattern">pattern</code></td>
<td>

<p>A character(1) regular expression describing the file names to be
evaluated; typically used to restrict tests to a subset of all test
files.
</p>
</td></tr>
<tr><td><code id="testPackage_+3A_path">path</code></td>
<td>

<p>A character(1) directory path indicating, when <code>pkgname</code> is
missing or NULL, where unit tests will be searched. <code>path</code> can
be any location at or below the package root.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is not exported from the package namespace, and must be
invoked using triple colons, <code>BiocGenerics:::testPackage()</code>; it
is provided primarily for the convenience of developers.
</p>
<p>When invoked with missing or NULL <code>pkgname</code> argument, the
function assumes that it has been invoked from within the package
source tree (or that the source tree is located above <code>path</code>),
and finds unit tests in <code>subdir="unitTests"</code> in either the base
or <code>inst/</code> directories at the root of the package source
tree. This mode is useful when developing unit tests, since the
package does not have to be re-installed to run an updated test.
</p>
<p>When invoked with <code>pkgname</code> set to the name of an installed
package, unit tests are searched for in the installed package
directory.
</p>


<h3>Value</h3>

<p>The function returns the result of <code>RUnit::runTestSuite</code> invoked
on the unit tests specified in the function call.
</p>


<h3>See Also</h3>

<p><a href="http://bioconductor.org/developers/how-to/unitTesting-guidelines/">http://bioconductor.org/developers/how-to/unitTesting-guidelines/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Run unit tests found in the library location where
## BiocGenerics is installed
BiocGenerics:::testPackage("BiocGenerics")
## Not run: ## Run unit tests for the package whose source tree implied
## by getwd()
BiocGenerics:::testPackage()

## End(Not run)
</code></pre>

<hr>
<h2 id='toTable'>An alternative to as.data.frame()</h2><span id='topic+toTable'></span>

<h3>Description</h3>

<p><code>toTable()</code> is an <em>S4 generic function</em> provided as an
alternative to <code><a href="#topic+as.data.frame">as.data.frame</a>()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>toTable(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="toTable_+3A_x">x</code></td>
<td>

<p>The object to turn into a data frame.
</p>
</td></tr>
<tr><td><code id="toTable_+3A_...">...</code></td>
<td>

<p>Additional arguments, for use in specific methods.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame.
</p>


<h3>See Also</h3>


<ul>
<li><p> The <code><a href="#topic+as.data.frame">as.data.frame</a></code> <em>S4 generic</em>
defined in the <span class="pkg">BiocGenerics</span> package.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="AnnotationDbi.html#topic+toTable+2CBimap-method">toTable,Bimap-method</a> in the
<span class="pkg">AnnotationDbi</span> package for an example of a specific
<code>toTable</code> method (defined for <a href="AnnotationDbi.html#topic+Bimap">Bimap</a>
objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>toTable
showMethods("toTable")

library(AnnotationDbi)
showMethods("toTable")
selectMethod("toTable", "Bimap")
</code></pre>

<hr>
<h2 id='type'>Accessing the type of an object</h2><span id='topic+type'></span><span id='topic+type+2Cvector-method'></span><span id='topic+type+2Carray-method'></span><span id='topic+type+2Cfactor-method'></span><span id='topic+type+2Cdata.frame-method'></span><span id='topic+type+3C-'></span><span id='topic+type+3C-+2Cvector-method'></span><span id='topic+type+3C-+2Carray-method'></span>

<h3>Description</h3>

<p>Get or set the <em>type</em> of an object.
</p>
<p>Note that <code>type</code> and <code>type&lt;-</code> are defined as <em>S4 generic
functions</em> and what <em>type</em> means exactly (and what <code>type()</code>
returns) depends on the objects for which the <code>type</code> and/or
<code>type&lt;-</code> methods are defined.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>type(x)
type(x) &lt;- value

## Methods defined in the BiocGenerics package:

## S4 method for signature 'vector'
type(x)
## S4 method for signature 'array'
type(x)
## S4 method for signature 'factor'
type(x)  # returns "character"
## S4 method for signature 'data.frame'
type(x)

## S4 replacement method for signature 'vector'
type(x) &lt;- value
## S4 replacement method for signature 'array'
type(x) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="type_+3A_x">x</code></td>
<td>

<p>Any object for which the <code>type()</code> getter or setter is defined.
Note that objects will either: not support the getter or setter at all,
or support only the getter, or support both the getter and setter.
</p>
</td></tr>
<tr><td><code id="type_+3A_value">value</code></td>
<td>

<p>The type to set on <code>x</code> (assuming <code>x</code> supports the <code>type()</code>
setter). <code>value</code> is typically (but not necessarily) expected to be
a single string (i.e. a character vector of length 1).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>On an ordinary vector, matrix, or array <code>x</code>, <code>type(x)</code> returns
<code>typeof(x)</code>.
</p>
<p>On a data frame <code>x</code> where all the columns are ordinary vectors
or factors, <code>type(x)</code> is <em>semantically equivalent</em> to
<code>typeof(as.matrix(x))</code>. However, the actual implementation is
careful to avoid turning the full data frame <code>x</code> into a matrix,
as this would tend to be very inefficient in general.
</p>
<p>Note that for a matrix-like or array-like object, <code>type(x)</code>
returns the type of the <em>elements</em> in the object.
See <code>?S4Arrays::<a href="S4Arrays.html#topic+type">type</a></code> for more information.
</p>


<h3>Value</h3>

<p><code>type(x)</code> is expected to return the type of <code>x</code> as a single
string i.e. as a character vector of length 1.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="DelayedArray.html#topic+type+2CANY-method">type,ANY-method</a> in the <span class="pkg">DelayedArray</span>
package for the default <code>type</code> method.
</p>
</li>
<li> <p><a href="S4Arrays.html#topic+type+2CDataFrame-method">type,DataFrame-method</a> in the <span class="pkg">S4Arrays</span>
package, and <a href="Biostrings.html#topic+type+2CPairwiseAlignments-method">type,PairwiseAlignments-method</a> in
the <span class="pkg">Biostrings</span> package, for examples of specific <code>type</code>
methods (defined for <a href="S4Vectors.html#topic+DataFrame">DataFrame</a> and
<a href="Biostrings.html#topic+PairwiseAlignments">PairwiseAlignments</a> objects, respectively).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>type
showMethods("type")

`type&lt;-`
showMethods("type&lt;-")

## The BiocGenerics package defines methods for ordinary vectors, arrays,
## and data frames:
m &lt;- matrix(11:22, nrow=3)
type(m)           # equivalent to 'typeof(m)' or 'storage.mode(m)'
type(m) &lt;- "raw"  # equivalent to 'storage.mode(m) &lt;- "raw"'
m
type(m)

selectMethod("type", "array")

selectMethod("type&lt;-", "array")

df &lt;- data.frame(a=44:49, b=letters[1:6], c=c(TRUE, FALSE))
stopifnot(identical(type(df), typeof(as.matrix(df))))

## Examples of methods defined in other packages:

library(DelayedArray)
showMethods("type")
selectMethod("type", "ANY")  # the default "type" method

library(Biostrings)
showMethods("type")
## The type() method for PairwiseAlignments objects:
selectMethod("type", "PairwiseAlignments")
</code></pre>

<hr>
<h2 id='unique'>Extract unique elements</h2><span id='topic+unique'></span>

<h3>Description</h3>

<p><code>unique</code> returns an object of the same class as <code>x</code> (typically
a vector-like, data-frame-like, or array-like object) but with duplicate
elements/rows removed.
</p>
<p>NOTE: This man page is for the <code>unique</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+unique">unique</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
(typically vector-like or data-frame-like) not supported by the
default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unique(x, incomparables=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unique_+3A_x">x</code></td>
<td>

<p>A vector-like, data-frame-like, or array-like object.
</p>
</td></tr>
<tr><td><code id="unique_+3A_incomparables">incomparables</code>, <code id="unique_+3A_...">...</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+unique">unique</a></code> for a description of
these arguments.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+unique">unique</a></code> for the value returned by the
default method.
</p>
<p>Specific methods defined in Bioconductor packages will typically
return an object of the same class as the input object.
</p>
<p><code>unique</code> should always behave consistently with
<code>BiocGenerics::<a href="#topic+duplicated">duplicated</a></code>.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+unique">unique</a></code> for the default <code>unique</code>
method.
</p>
</li>
<li> <p><code>BiocGenerics::<a href="#topic+duplicated">duplicated</a></code> for determining
duplicate elements.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+unique+2CRle-method">unique,Rle-method</a> in the <span class="pkg">S4Vectors</span> package
for an example of a specific <code>unique</code> method (defined for
<a href="S4Vectors.html#topic+Rle">Rle</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>unique
showMethods("unique")
selectMethod("unique", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='unlist'>Flatten list-like objects</h2><span id='topic+unlist'></span>

<h3>Description</h3>

<p>Given a list-like object <code>x</code>, <code>unlist</code> produces a vector-like
object obtained by concatenating (conceptually thru <code><a href="base.html#topic+c">c</a></code>) all
the top-level elements in <code>x</code> (each of them being expected to be
a vector-like object, typically).
</p>
<p>NOTE: This man page is for the <code>unlist</code>
<em>S4 generic function</em> defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+unlist">unlist</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unlist(x, recursive=TRUE, use.names=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unlist_+3A_x">x</code></td>
<td>

<p>A list-like object.
</p>
</td></tr>
<tr><td><code id="unlist_+3A_recursive">recursive</code>, <code id="unlist_+3A_use.names">use.names</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+unlist">unlist</a></code> for a description of
these arguments.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+unlist">unlist</a></code> for the value returned
by the default method.
</p>
<p>Specific methods defined in Bioconductor packages should
behave as consistently as possible with the default method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+unlist">unlist</a></code> for the default
<code>unlist</code> method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+unlist+2CList-method">unlist,List-method</a> in the <span class="pkg">S4Vectors</span> package
for an example of a specific <code>unlist</code> method (defined for
<a href="S4Vectors.html#topic+List">List</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>unlist  # note the dispatch on the 'x' arg only
showMethods("unlist")
selectMethod("unlist", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='unsplit'>Unsplit a list-like object</h2><span id='topic+unsplit'></span>

<h3>Description</h3>

<p>Given a list-like object <code>value</code> and grouping <code>f</code>,
<code>unsplit</code> produces a vector-like object <code>x</code> by conceptually
reversing the split operation <code>value &lt;- split(x, f)</code>.
</p>
<p>NOTE: This man page is for the <code>unsplit</code>
<em>S4 generic function</em> defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+unsplit">unsplit</a></code> for the default method
(defined in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unsplit(value, f, drop=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unsplit_+3A_value">value</code></td>
<td>

<p>A list-like object.
</p>
</td></tr>
<tr><td><code id="unsplit_+3A_f">f</code></td>
<td>

<p>A factor or other grouping object that corresponds to the <code>f</code>
symbol in <code>value &lt;- split(x, f)</code>.
</p>
</td></tr>
<tr><td><code id="unsplit_+3A_drop">drop</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+unsplit">unsplit</a></code> for a description of
this argument.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+unsplit">unsplit</a></code> for the value returned
by the default method.
</p>
<p>Specific methods defined in Bioconductor packages should
behave as consistently as possible with the default method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+unsplit">unsplit</a></code> for the default
<code>unsplit</code> method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="IRanges.html#topic+unsplit+2CList-method">unsplit,List-method</a> in the <span class="pkg">IRanges</span> package
for an example of a specific <code>unsplit</code> method (defined for
<a href="S4Vectors.html#topic+List">List</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>unsplit  # note the dispatch on the 'value' and 'f' args only
showMethods("unsplit")
selectMethod("unsplit", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='updateObject'>Update an object to its current class definition</h2><span id='topic+updateObject'></span><span id='topic+updateObject+2CANY-method'></span><span id='topic+updateObject+2Clist-method'></span><span id='topic+updateObject+2Cenvironment-method'></span><span id='topic+updateObject+2Cformula-method'></span><span id='topic+updateObject+2CenvRefClass-method'></span><span id='topic+updateObjectFromSlots'></span><span id='topic+getObjectSlots'></span>

<h3>Description</h3>

<p><code>updateObject</code> is a generic function that returns an instance
of <code>object</code> updated to its current class definition.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>updateObject(object, ..., verbose=FALSE)

## Related utilities:
updateObjectFromSlots(object, objclass=class(object)[[1L]], ...,
                      verbose=FALSE)
getObjectSlots(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="updateObject_+3A_object">object</code></td>
<td>

<p>Object to be updated for <code>updateObject</code> and
<code>updateObjectFromSlots</code>.
</p>
<p>Object for slot information to be extracted from for
<code>getObjectSlots</code>.
</p>
</td></tr>
<tr><td><code id="updateObject_+3A_...">...</code></td>
<td>

<p>Additional arguments, for use in specific <code>updateObject</code>
methods.
</p>
</td></tr>
<tr><td><code id="updateObject_+3A_verbose">verbose</code></td>
<td>

<p><code>TRUE</code> or <code>FALSE</code>, indicating whether information about
the update should be reported. Use <code><a href="base.html#topic+message">message</a></code> to
report this information.
</p>
</td></tr>
<tr><td><code id="updateObject_+3A_objclass">objclass</code></td>
<td>

<p>Optional character string naming the class of the object
to be created.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Updating objects is primarily useful when an object has been
serialized (e.g., stored to disk) for some time (e.g., months), and the
class definition has in the mean time changed. Because of the changed
class definition, the serialized instance is no longer valid.
</p>
<p><code>updateObject</code> requires that the class of the returned object be
the same as the class of the argument <code>object</code>, and that the
object is valid (see <code><a href="methods.html#topic+validObject">validObject</a></code>). By default,
<code>updateObject</code> has the following behaviors:
</p>

<dl>
<dt><code>updateObject(ANY, ..., verbose=FALSE)</code></dt><dd>
<p>By default, <code>updateObject</code> uses heuristic methods to determine
whether the object should be the &lsquo;new&rsquo; S4 type (introduced in R 2.4.0),
but is not. If the heuristics indicate an update is required,
the <code>updateObjectFromSlots</code> function tries to update the
object. The default method returns the original S4 object or the
successfully updated object, or issues an error if an update is
required but not possible.
The optional named argument <code>verbose</code> causes a message to be
printed describing the action.
Arguments <code>...</code> are passed to <code>updateObjectFromSlots</code>.
</p>
</dd>
<dt><code>updateObject(list, ..., verbose=FALSE)</code></dt><dd>
<p>Visit each element in <code>list</code>, applying
<code>updateObject(list[[elt]], ..., verbose=verbose)</code>.
</p>
</dd>
<dt><code>updateObject(environment, ..., verbose=FALSE)</code></dt><dd>
<p>Visit each element in <code>environment</code>, applying
<code>updateObject(environment[[elt]], ..., verbose=verbose)</code>
</p>
</dd>
<dt><code>updateObject(formula, ..., verbose=FALSE)</code></dt><dd>
<p>Do nothing; the environment of the formula may be too general
(e.g., <code>R_GlobalEnv</code>) to attempt an update.
</p>
</dd>
<dt><code>updateObject(envRefClass, ..., verbose=FALSE)</code></dt><dd>
<p>Attempt to update objects from fields using a strategy like
<code>updateObjectFromSlots</code> Method 1.
</p>
</dd>
</dl>

<p><code>updateObjectFromSlots(object, objclass=class(object), ...,
  verbose=FALSE)</code> is a utility function that identifies the intersection
of slots defined in the <code>object</code> instance and <code>objclass</code>
definition. Under Method 1, the corresponding elements in
<code>object</code> are then updated (with <code>updateObject(elt, ...,
  verbose=verbose)</code>) and used as arguments to a call to <code>new(class,
  ...)</code>, with <code>...</code> replaced by slots from the original
object. If this fails, then Method 2 tries <code>new(class)</code> and
assigns slots of <code>object</code> to the newly created instance.
</p>
<p><code>getObjectSlots(object)</code> extracts the slot names and contents from
<code>object</code>. This is useful when <code>object</code> was created by a class
definition that is no longer current, and hence the contents of
<code>object</code> cannot be determined by accessing known slots.
</p>


<h3>Value</h3>

<p><code>updateObject</code> returns a valid instance of <code>object</code>.
</p>
<p><code>updateObjectFromSlots</code> returns an instance of class
<code>objclass</code>.
</p>
<p><code>getObjectSlots</code> returns a list of named elements, with each
element corresponding to a slot in <code>object</code>.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="Biobase.html#topic+updateObjectTo">updateObjectTo</a></code> in the <span class="pkg">Biobase</span> package
for updating an object to the class definition of a template (might
be useful for updating a virtual superclass).
</p>
</li>
<li> <p><code><a href="methods.html#topic+validObject">validObject</a></code> for testing the validity of an
object.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>updateObject
showMethods("updateObject")
selectMethod("updateObject", "ANY")  # the default method

library(Biobase)
## update object, same class
data(sample.ExpressionSet)
obj &lt;- updateObject(sample.ExpressionSet)

setClass("UpdtA", representation(x="numeric"), contains="data.frame")
setMethod("updateObject", "UpdtA",
    function(object, ..., verbose=FALSE)
    {
        if (verbose)
            message("updateObject object = 'A'")
        object &lt;- callNextMethod()
        object@x &lt;- -object@x
        object
    }
)

a &lt;- new("UpdtA", x=1:10)
## See steps involved
updateObject(a)

removeMethod("updateObject", "UpdtA")
removeClass("UpdtA")
</code></pre>

<hr>
<h2 id='var'>Variance and Standard Deviation</h2><span id='topic+var'></span><span id='topic+sd'></span>

<h3>Description</h3>

<p><code>var</code> and <code>sd</code> compute the variance and standard deviation
of a vector <code>x</code>.
</p>
<p>NOTE: This man page is for the <code>var</code> and <code>sd</code>,
<em>S4 generic functions</em> defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?stats::<a href="stats.html#topic+var">var</a></code> and <code>?stats::<a href="stats.html#topic+sd">sd</a></code>
for the default methods (defined in the <span class="pkg">stats</span> package).
Bioconductor packages can define specific methods for objects
(typically array-like) not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>     var(x, y = NULL, na.rm = FALSE, use)
     sd(x, na.rm = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="var_+3A_x">x</code></td>
<td>

<p>a vector-like object
</p>
</td></tr>
<tr><td><code id="var_+3A_y">y</code></td>
<td>

<p>a vector-like object, or <code>NULL</code>
</p>
</td></tr>
<tr><td><code id="var_+3A_na.rm">na.rm</code>, <code id="var_+3A_use">use</code></td>
<td>
<p>see <a href="stats.html#topic+var">var</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?stats::<a href="stats.html#topic+var">var</a></code> and <code>?stats::<a href="stats.html#topic+sd">sd</a></code>
for the value returned by the default methods.
</p>
<p>Specific methods defined in Bioconductor packages will typically
return an object of the same class as the input object.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>stats::<a href="stats.html#topic+var">var</a></code> and <code>stats::<a href="stats.html#topic+sd">sd</a></code>
for the default methods.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>var
showMethods("var")
selectMethod("var", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='weights'>Extract model weights</h2><span id='topic+weights'></span>

<h3>Description</h3>

<p><code>weights</code> is a generic function which extracts fitting weights
from objects returned by modeling functions.
</p>
<p>NOTE: This man page is for the <code>weights</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?stats::<a href="stats.html#topic+weights">weights</a></code> for the default method
(defined in the <span class="pkg">stats</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>weights(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="weights_+3A_object">object</code>, <code id="weights_+3A_...">...</code></td>
<td>

<p>See <code>?stats::<a href="stats.html#topic+weights">weights</a></code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Weights extracted from the object <code>object</code>.
</p>
<p>See <code>?stats::<a href="stats.html#topic+weights">weights</a></code> for the value returned by the
default method.
</p>
<p>Specific methods defined in Bioconductor packages should
behave as consistently as possible with the default method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>stats::<a href="stats.html#topic+weights">weights</a></code> for the default <code>weights</code>
method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="affyPLM.html#topic+weights+2CPLMset-method">weights,PLMset-method</a> in the <span class="pkg">affyPLM</span> package
for an example of a specific <code>weights</code> method (defined for
<a href="affyPLM.html#topic+PLMset">PLMset</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>weights
showMethods("weights")
selectMethod("weights", "ANY")  # the default method
</code></pre>

<hr>
<h2 id='which'>Which values in an object are considered TRUE?</h2><span id='topic+which'></span>

<h3>Description</h3>

<p>Give the indices of the values in a vector-, array-, or list-like object
that are considered <code>TRUE</code>, allowing for array indices in the case
of an array-like object.
</p>
<p>NOTE: This man page is for the <code>which</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+which">which</a></code> for the default method (defined
in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects (typically
vector-, array-, or list-like) not supported by the default methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>which(x, arr.ind=FALSE, useNames=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="which_+3A_x">x</code></td>
<td>

<p>An object, typically with a vector-, array-, or list-like semantic.
</p>
</td></tr>
<tr><td><code id="which_+3A_arr.ind">arr.ind</code>, <code id="which_+3A_usenames">useNames</code></td>
<td>

<p>See <code>?base::<a href="base.html#topic+which">which</a></code> for a description of
these arguments.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+which">which</a></code> for the value returned
by the default method.
</p>
<p>Specific methods defined in Bioconductor packages should
behave as consistently as possible with the default method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+which">which</a></code> for the default <code>which</code> method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="DelayedArray.html#topic+which+2CDelayedArray-method">which,DelayedArray-method</a> in the
<span class="pkg">DelayedArray</span> package for an example of a specific
<code>which</code> method (defined for <a href="DelayedArray.html#topic+DelayedArray">DelayedArray</a>
objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>which
showMethods("which")
selectMethod("which", "ANY")  # the default method

library(DelayedArray)
showMethods("which")
## The which() method for DelayedArray objects:
selectMethod("which", "DelayedArray")
</code></pre>

<hr>
<h2 id='which.min'>What's the index of the first min or max value in an object?</h2><span id='topic+which.min'></span><span id='topic+which.max'></span>

<h3>Description</h3>

<p>Determines the location (i.e. index) of the (first) minimum or
maximum value in an object.
</p>
<p>NOTE: This man page is for the <code>which.min</code> and <code>which.max</code>
<em>S4 generic functions</em> defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?base::<a href="base.html#topic+which.min">which.min</a></code> for the default methods (defined
in the <span class="pkg">base</span> package).
Bioconductor packages can define specific methods for objects (typically
vector-, array-, or list-like) not supported by the default methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>which.min(x, ...)
which.max(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="which.min_+3A_x">x</code></td>
<td>

<p>An object, typically with a vector-, array-, or list-like semantic.
</p>
</td></tr>
<tr><td><code id="which.min_+3A_...">...</code></td>
<td>

<p>Additional arguments, for use in specific methods.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?base::<a href="base.html#topic+which.min">which.min</a></code> for the value returned by the
default methods.
</p>
<p>Specific methods defined in Bioconductor packages should
behave as consistently as possible with the default methods.
</p>


<h3>Note</h3>

<p>The default methods (defined in the <span class="pkg">base</span> package) only take a
single argument.  We've added the <code>...</code> argument to the generic
functions defined in the <span class="pkg">BiocGenerics</span> package so they
can be called with an arbitrary number of effective arguments.
This typically allows methods to add extra arguments for
controlling/altering the behavior of the operation.  Like for
example the <code>global</code> argument supported by the <code>which.max</code>
method for <a href="IRanges.html#topic+NumericList">NumericList</a> objects (defined in the
<span class="pkg">IRanges</span> package).
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>base::<a href="base.html#topic+which.min">which.min</a></code> for the default <code>which.min</code>
and <code>which.max</code> methods.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="IRanges.html#topic+which.max+2CNumericList-method">which.max,NumericList-method</a> in the <span class="pkg">IRanges</span>
package for an example of a specific <code>which.max</code> method
(defined for <a href="IRanges.html#topic+NumericList">NumericList</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>which.min
showMethods("which.min")
selectMethod("which.min", "ANY")  # the default method

which.max
showMethods("which.max")
selectMethod("which.max", "ANY")  # the default method

library(IRanges)
showMethods("which.max")
## The which.max() method for NumericList objects:
selectMethod("which.max", "NumericList")
</code></pre>

<hr>
<h2 id='xtabs'>Cross tabulation</h2><span id='topic+xtabs'></span>

<h3>Description</h3>

<p><code>xtabs</code> creates a contingency table (optionally a sparse matrix) from
cross-classifying factors, usually contained in a data-frame-like object,
using a formula interface.
</p>
<p>NOTE: This man page is for the <code>xtabs</code> <em>S4 generic function</em>
defined in the <span class="pkg">BiocGenerics</span> package.
See <code>?stats::<a href="stats.html#topic+xtabs">xtabs</a></code> for the default method
(defined in the <span class="pkg">stats</span> package).
Bioconductor packages can define specific methods for objects
not supported by the default method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xtabs(formula=~., data=parent.frame(), subset, sparse=FALSE, 
      na.action, addNA=FALSE, exclude=if(!addNA)c(NA, NaN),
      drop.unused.levels=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xtabs_+3A_formula">formula</code>, <code id="xtabs_+3A_subset">subset</code>, <code id="xtabs_+3A_sparse">sparse</code>, <code id="xtabs_+3A_na.action">na.action</code>, <code id="xtabs_+3A_addna">addNA</code>, <code id="xtabs_+3A_exclude">exclude</code>, <code id="xtabs_+3A_drop.unused.levels">drop.unused.levels</code></td>
<td>

<p>See <code>?stats::<a href="stats.html#topic+xtabs">xtabs</a></code> for a description of
these arguments.
</p>
</td></tr>
<tr><td><code id="xtabs_+3A_data">data</code></td>
<td>

<p>A data-frame-like object.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See <code>?stats::<a href="stats.html#topic+xtabs">xtabs</a></code> for the value returned by the
default method.
</p>
<p>Specific methods defined in Bioconductor packages should also
return the type of object returned by the default method.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>stats::<a href="stats.html#topic+xtabs">xtabs</a></code> for the default <code>xtabs</code>
method.
</p>
</li>
<li> <p><code><a href="methods.html#topic+showMethods">showMethods</a></code> for displaying a summary of the
methods defined for a given generic function.
</p>
</li>
<li> <p><code><a href="methods.html#topic+selectMethod">selectMethod</a></code> for getting the definition of
a specific method.
</p>
</li>
<li> <p><a href="S4Vectors.html#topic+xtabs+2CDataFrame-method">xtabs,DataFrame-method</a> in the <span class="pkg">S4Vectors</span>
package for an example of a specific <code>xtabs</code> method (defined
for <a href="S4Vectors.html#topic+DataFrame">DataFrame</a> objects).
</p>
</li>
<li> <p><a href="BiocGenerics.html#topic+BiocGenerics">BiocGenerics</a> for a summary of all the generics defined
in the <span class="pkg">BiocGenerics</span> package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>xtabs  # note the dispatch on the 'data' arg only
showMethods("xtabs")
selectMethod("xtabs", "ANY")  # the default method

library(S4Vectors)
showMethods("xtabs")
## The xtabs() method for DataFrame objects:
selectMethod("xtabs", "DataFrame")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
