<!DOCTYPE html><html><head><title>Help for package BioCircos</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BioCircos}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BioCircos'><p>BioCircos</p></a></li>
<li><a href='#BioCircos-shiny'><p>Shiny bindings for BioCircos</p></a></li>
<li><a href='#BioCircosArcTrack'><p>Create a track with arcs to be added to a BioCircos tracklist</p></a></li>
<li><a href='#BioCircosBackgroundTrack'><p>Create a background track to be added to a BioCircos tracklist</p></a></li>
<li><a href='#BioCircosBarTrack'><p>Create a track with a bar plot to be added to a BioCircos tracklist</p></a></li>
<li><a href='#BioCircosCNVTrack'><p>Create a track with concentric arcs to be added to a BioCircos tracklist</p></a></li>
<li><a href='#BioCircosHeatmapTrack'><p>Create a heatmap track to be added to a BioCircos tracklist</p></a></li>
<li><a href='#BioCircosLineTrack'><p>Create a track with lines to be added to a BioCircos tracklist</p></a></li>
<li><a href='#BioCircosLinkTrack'><p>Create an inner track with links to be added to a BioCircos tracklist</p></a></li>
<li><a href='#BioCircosSNPTrack'><p>Create a track with SNPs to be added to a BioCircos tracklist</p></a></li>
<li><a href='#BioCircosTextTrack'><p>Create a Text track to be added to a BioCircos tracklist</p></a></li>
<li><a href='#BioCircosTracklist'><p>Create a list of BioCircos tracks</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Interactive Circular Visualization of Genomic Data using
'htmlwidgets' and 'BioCircos.js'</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.4</td>
</tr>
<tr>
<td>Description:</td>
<td>Implement in 'R' interactive Circos-like visualizations of genomic data, to map information
	such as genetic variants, genomic fusions and aberrations to a circular genome, as proposed by the
	'JavaScript' library 'BioCircos.js', based on the 'JQuery' and 'D3' technologies. The output is by 
	default displayed in stand-alone HTML documents or in the 'RStudio' viewer pane. Moreover it can be
	integrated in 'R Markdown' documents and 'Shiny' applications.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/lvulliard/BioCircos.R">https://github.com/lvulliard/BioCircos.R</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>RColorBrewer, htmlwidgets, jsonlite, plyr, grDevices</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-03-12 15:03:38 UTC; lvulliard</td>
</tr>
<tr>
<td>Author:</td>
<td>Loan Vulliard [trl, cre],
  Xiaowei Chen [aut],
  Ya Cui [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Loan Vulliard &lt;lvulliard@cemm.at&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-03-19 16:23:26 UTC</td>
</tr>
</table>
<hr>
<h2 id='BioCircos'>BioCircos</h2><span id='topic+BioCircos'></span>

<h3>Description</h3>

<p>Interactive circular visualization of genomic data using ‘htmlwidgets’ and ‘BioCircos.js’
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BioCircos(tracklist = BioCircosTracklist(), genome = "hg19",
  yChr = TRUE, genomeFillColor = "Spectral", chrPad = 0.04,
  displayGenomeBorder = TRUE, genomeBorderColor = "#000",
  genomeBorderSize = 0.5, genomeTicksDisplay = TRUE,
  genomeTicksLen = 5, genomeTicksColor = "#000",
  genomeTicksTextSize = "0.6em", genomeTicksTextColor = "#000",
  genomeTicksScale = 3e+07, genomeLabelDisplay = TRUE,
  genomeLabelTextSize = "10pt", genomeLabelTextColor = "#000",
  genomeLabelDx = 0, genomeLabelDy = 10, genomeLabelOrientation = 0,
  zoom = TRUE, TEXTModuleDragEvent = FALSE,
  SNPMouseOverDisplay = TRUE, SNPMouseOverColor = "#FF0000",
  SNPMouseOverCircleSize = 3, SNPMouseOverCircleOpacity = 0.9,
  SNPMouseOutDisplay = TRUE, SNPMouseOutColor = "none",
  SNPMouseOverTooltipsHtml01 = "Chromosome: ",
  SNPMouseOverTooltipsHtml02 = "&lt;br/&gt;Position: ",
  SNPMouseOverTooltipsHtml03 = "&lt;br/&gt;Value: ",
  SNPMouseOverTooltipsHtml04 = "&lt;br/&gt;",
  SNPMouseOverTooltipsHtml05 = "",
  SNPMouseOverTooltipsBorderWidth = "1px", ARCMouseOverDisplay = TRUE,
  ARCMouseOverColor = "#FF0000", ARCMouseOverArcOpacity = 0.9,
  ARCMouseOutDisplay = TRUE, ARCMouseOutColor = "none",
  ARCMouseOverTooltipsHtml01 = "Chromosome: ",
  ARCMouseOverTooltipsHtml02 = "&lt;br/&gt;Start: ",
  ARCMouseOverTooltipsHtml03 = "&lt;br/&gt;End: ",
  ARCMouseOverTooltipsHtml04 = "&lt;br/&gt;",
  ARCMouseOverTooltipsHtml05 = "",
  ARCMouseOverTooltipsBorderWidth = "1px", LINKMouseOverDisplay = TRUE,
  LINKMouseOverStrokeColor = "#FF00FF", LINKMouseOverOpacity = 0.9,
  LINKMouseOutDisplay = TRUE, LINKMouseOutStrokeColor = "none",
  LINKMouseOverTooltipsHtml01 = "Fusion: ",
  LINKMouseOverTooltipsHtml02 = "",
  LINKMouseOverTooltipsBorderWidth = "1px",
  LINKMouseOverStrokeWidth = 5, LINKMouseOutStrokeWidth = "none",
  BARMouseOutDisplay = TRUE, BARMouseOutColor = "none",
  BARMouseOverDisplay = TRUE, BARMouseOverColor = "#FF0000",
  BARMouseOverOpacity = 0.9,
  BARMouseOverTooltipsHtml01 = "Chromosome: ",
  BARMouseOverTooltipsHtml02 = "&lt;br/&gt;Start: ",
  BARMouseOverTooltipsHtml03 = " End: ",
  BARMouseOverTooltipsHtml04 = "&lt;br/&gt;",
  BARMouseOverTooltipsHtml05 = "&lt;br/&gt;Value: ",
  BARMouseOverTooltipsHtml06 = "",
  BARMouseOverTooltipsBorderWidth = "1px",
  HEATMAPMouseOutDisplay = TRUE, HEATMAPMouseOutColor = "none",
  HEATMAPMouseOverDisplay = TRUE, HEATMAPMouseOverColor = "#FF0000",
  HEATMAPMouseOverOpacity = 0.9,
  HEATMAPMouseOverTooltipsHtml01 = "Chromosome: ",
  HEATMAPMouseOverTooltipsHtml02 = "&lt;br/&gt;Start: ",
  HEATMAPMouseOverTooltipsHtml03 = " End: ",
  HEATMAPMouseOverTooltipsHtml04 = "&lt;br/&gt;",
  HEATMAPMouseOverTooltipsHtml05 = "&lt;br/&gt;Value: ",
  HEATMAPMouseOverTooltipsHtml06 = "",
  HEATMAPMouseOverTooltipsBorderWidth = "1px",
  LINEMouseOutDisplay = TRUE, LINEMouseOutLineOpacity = "none",
  LINEMouseOutLineStrokeColor = "none",
  LINEMouseOutLineStrokeWidth = "none", LINEMouseOverDisplay = T,
  LINEMouseOverLineOpacity = 1,
  LINEMouseOverLineStrokeColor = "#FF0000",
  LINEMouseOverLineStrokeWidth = "none",
  LINEMouseOverTooltipsHtml01 = "Line",
  LINEMouseOverTooltipsBorderWidth = 0, CNVMouseOutDisplay = TRUE,
  CNVMouseOutColor = "none", CNVMouseOutArcOpacity = 1,
  CNVMouseOutArcStrokeColor = "none", CNVMouseOutArcStrokeWidth = 0,
  CNVMouseOverDisplay = TRUE, CNVMouseOverColor = "#FF0000",
  CNVMouseOverArcOpacity = 0.9, CNVMouseOverArcStrokeColor = "#F26223",
  CNVMouseOverArcStrokeWidth = 3,
  CNVMouseOverTooltipsHtml01 = "Chromosome: ",
  CNVMouseOverTooltipsHtml02 = "&lt;br&gt;Start: ",
  CNVMouseOverTooltipsHtml03 = "&lt;br&gt;End: ",
  CNVMouseOverTooltipsHtml04 = "&lt;br&gt;Value: ",
  CNVMouseOverTooltipsHtml05 = "",
  CNVMouseOverTooltipsBorderWidth = "1px", width = NULL,
  height = NULL, elementId = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BioCircos_+3A_tracklist">tracklist</code></td>
<td>
<p>A list of tracks to display.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_genome">genome</code></td>
<td>
<p>A list of chromosome lengths to be used as reference for the visualization or 'hg19' to use
the chromosomes 1 to 22 and the sexual chromosomes according to the hg19 reference.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_ychr">yChr</code></td>
<td>
<p>A logical stating if the Y chromosome should be displayed. Used only when genome is set to 'hg19'.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_genomefillcolor">genomeFillColor</code></td>
<td>
<p>The color to display in each chromosome. Can be a RColorBrewer palette name used to
generate one color per chromosome, or a character object or vector of character objects stating RGB values in hexadecimal
format or base R colors. If the vector is shorter than the reference genome, values will be repeated.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_chrpad">chrPad</code></td>
<td>
<p>Distance between chromosomes.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_displaygenomeborder">displayGenomeBorder</code>, <code id="BioCircos_+3A_genomebordercolor">genomeBorderColor</code>, <code id="BioCircos_+3A_genomebordersize">genomeBorderSize</code></td>
<td>
<p>Should the reference genome have borders?
If yes specify the color, in RGB hexadecimal format, and the thickness.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_genometicksdisplay">genomeTicksDisplay</code>, <code id="BioCircos_+3A_genometickslen">genomeTicksLen</code>, <code id="BioCircos_+3A_genometickscolor">genomeTicksColor</code>, <code id="BioCircos_+3A_genometickstextsize">genomeTicksTextSize</code>, <code id="BioCircos_+3A_genometickstextcolor">genomeTicksTextColor</code>, <code id="BioCircos_+3A_genometicksscale">genomeTicksScale</code></td>
<td>
<p>Should the refence genome have ticks, of which length, color (in hexadecimal RGB format), with labels in which font
size and color, and spaced by how many bases?</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_genomelabeldisplay">genomeLabelDisplay</code>, <code id="BioCircos_+3A_genomelabeltextsize">genomeLabelTextSize</code>, <code id="BioCircos_+3A_genomelabeltextcolor">genomeLabelTextColor</code>, <code id="BioCircos_+3A_genomelabeldx">genomeLabelDx</code>, <code id="BioCircos_+3A_genomelabeldy">genomeLabelDy</code>, <code id="BioCircos_+3A_genomelabelorientation">genomeLabelOrientation</code></td>
<td>
<p>Should the reference genome have labels on each chromosome, in which font size and color? Moreover rotation
and radius shifts for the label texts can be added, and the angle between the radius and the label changed.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_zoom">zoom</code></td>
<td>
<p>Is zooming and moving in the visualization allowed?</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_textmoduledragevent">TEXTModuleDragEvent</code></td>
<td>
<p>Are text annotations draggable?</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_snpmouseoverdisplay">SNPMouseOverDisplay</code></td>
<td>
<p>Display the tooltip when mouse hover on a SNP point.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_snpmouseovercolor">SNPMouseOverColor</code></td>
<td>
<p>Color of the SNP point when hovered by the mouse, in hexadecimal RGB format.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_snpmouseovercirclesize">SNPMouseOverCircleSize</code></td>
<td>
<p>Size of the SNP point when hovered by the mouse.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_snpmouseovercircleopacity">SNPMouseOverCircleOpacity</code></td>
<td>
<p>Opacity of the SNP point when hovered by the mouse.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_snpmouseoutdisplay">SNPMouseOutDisplay</code></td>
<td>
<p>Hide tooltip when mouse is not hovering a SNP point anymore.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_snpmouseoutcolor">SNPMouseOutColor</code></td>
<td>
<p>Color of the SNP point when mouse is not hovering a SNP point anymore, in hexadecimal
RGB format. To revert back to original color, use the value &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_snpmouseovertooltipshtml01">SNPMouseOverTooltipsHtml01</code></td>
<td>
<p>Label displayed in tooltip in first position, before chromosome number.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_snpmouseovertooltipshtml02">SNPMouseOverTooltipsHtml02</code></td>
<td>
<p>Label displayed in tooltip in second position, before genomic position.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_snpmouseovertooltipshtml03">SNPMouseOverTooltipsHtml03</code></td>
<td>
<p>Label displayed in tooltip in third position, before value.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_snpmouseovertooltipshtml04">SNPMouseOverTooltipsHtml04</code></td>
<td>
<p>Label displayed in tooltip in fourth position, before SNP labels if any.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_snpmouseovertooltipshtml05">SNPMouseOverTooltipsHtml05</code></td>
<td>
<p>Label displayed in tooltip in fifth position, after SNP labels if any.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_snpmouseovertooltipsborderwidth">SNPMouseOverTooltipsBorderWidth</code></td>
<td>
<p>The thickness of the tooltip borders, with units specified (such as em or px).</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_arcmouseoverdisplay">ARCMouseOverDisplay</code></td>
<td>
<p>Display the tooltip when mouse hover on an arc.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_arcmouseovercolor">ARCMouseOverColor</code></td>
<td>
<p>Color of the arc when hovered by the mouse, in hexadecimal RGB format.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_arcmouseoverarcopacity">ARCMouseOverArcOpacity</code></td>
<td>
<p>Opacity of the arc when hovered by the mouse.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_arcmouseoutdisplay">ARCMouseOutDisplay</code></td>
<td>
<p>Hide tooltip when mouse is not hovering an arc anymore.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_arcmouseoutcolor">ARCMouseOutColor</code></td>
<td>
<p>Color of the arc when mouse is not hovering an arc anymore, in hexadecimal
RGB format. To revert back to original color, use the value &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_arcmouseovertooltipshtml01">ARCMouseOverTooltipsHtml01</code></td>
<td>
<p>Label displayed in tooltip in first position, before chromosome number.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_arcmouseovertooltipshtml02">ARCMouseOverTooltipsHtml02</code></td>
<td>
<p>Label displayed in tooltip in second position, before genomic position.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_arcmouseovertooltipshtml03">ARCMouseOverTooltipsHtml03</code></td>
<td>
<p>Label displayed in tooltip in third position, before value.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_arcmouseovertooltipshtml04">ARCMouseOverTooltipsHtml04</code></td>
<td>
<p>Label displayed in tooltip in fourth position, before ARC labels if any.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_arcmouseovertooltipshtml05">ARCMouseOverTooltipsHtml05</code></td>
<td>
<p>Label displayed in tooltip in fifth position, after ARC labels if any.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_arcmouseovertooltipsborderwidth">ARCMouseOverTooltipsBorderWidth</code></td>
<td>
<p>The thickness of the tooltip borders, with units specified (such as em or px).</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_linkmouseoverdisplay">LINKMouseOverDisplay</code></td>
<td>
<p>Display the tooltip when mouse hover on a link.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_linkmouseoverstrokecolor">LINKMouseOverStrokeColor</code></td>
<td>
<p>Color of the link when hovered.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_linkmouseoveropacity">LINKMouseOverOpacity</code></td>
<td>
<p>Opacity of the link when hovered.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_linkmouseoutdisplay">LINKMouseOutDisplay</code></td>
<td>
<p>Hide tooltip when mouse is not hovering a link anymore.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_linkmouseoutstrokecolor">LINKMouseOutStrokeColor</code></td>
<td>
<p>Color of the link when mouse is not hovering anymore, in hexadecimal
RGB format. To revert back to original color, use the value &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_linkmouseovertooltipshtml01">LINKMouseOverTooltipsHtml01</code></td>
<td>
<p>Label displayed in tooltip in first position, before label.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_linkmouseovertooltipshtml02">LINKMouseOverTooltipsHtml02</code></td>
<td>
<p>Label displayed in tooltip in second position, after label.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_linkmouseovertooltipsborderwidth">LINKMouseOverTooltipsBorderWidth</code></td>
<td>
<p>The thickness of the tooltip borders, with units specified (such as em or px).</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_linkmouseoverstrokewidth">LINKMouseOverStrokeWidth</code></td>
<td>
<p>Thickness of the link when hovered.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_linkmouseoutstrokewidth">LINKMouseOutStrokeWidth</code></td>
<td>
<p>Thickness of the link when mouse is not hovering a link anymore.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_barmouseoutdisplay">BARMouseOutDisplay</code></td>
<td>
<p>Hide tooltip when mouse is not hovering a bar anymore.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_barmouseoutcolor">BARMouseOutColor</code></td>
<td>
<p>Color of the bar when mouse is not hovering anymore, in hexadecimal
RGB format. To revert back to original color, use the value &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_barmouseoverdisplay">BARMouseOverDisplay</code></td>
<td>
<p>Display the tooltip when mouse hover on a bar.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_barmouseovercolor">BARMouseOverColor</code></td>
<td>
<p>Color of the bar when hovered.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_barmouseoveropacity">BARMouseOverOpacity</code></td>
<td>
<p>Opacity of the bar when hovered.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_barmouseovertooltipshtml01">BARMouseOverTooltipsHtml01</code></td>
<td>
<p>Label displayed in tooltip in first position, before chromosome number.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_barmouseovertooltipshtml02">BARMouseOverTooltipsHtml02</code></td>
<td>
<p>Label displayed in tooltip in second position, before start position.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_barmouseovertooltipshtml03">BARMouseOverTooltipsHtml03</code></td>
<td>
<p>Label displayed in tooltip in second position, before end position.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_barmouseovertooltipshtml04">BARMouseOverTooltipsHtml04</code></td>
<td>
<p>Label displayed in tooltip in third position, before labels if any.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_barmouseovertooltipshtml05">BARMouseOverTooltipsHtml05</code></td>
<td>
<p>Label displayed in tooltip in fourth position, before values.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_barmouseovertooltipshtml06">BARMouseOverTooltipsHtml06</code></td>
<td>
<p>Label displayed in tooltip in fifth position, after values.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_barmouseovertooltipsborderwidth">BARMouseOverTooltipsBorderWidth</code></td>
<td>
<p>The thickness of the tooltip borders, with units specified (such as em or px).</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_heatmapmouseoutdisplay">HEATMAPMouseOutDisplay</code></td>
<td>
<p>Hide tooltip when mouse is not hovering a box anymore.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_heatmapmouseoutcolor">HEATMAPMouseOutColor</code></td>
<td>
<p>Color of the box when mouse is not hovering anymore, in hexadecimal
RGB format. To revert back to original color, use the value &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_heatmapmouseoverdisplay">HEATMAPMouseOverDisplay</code></td>
<td>
<p>Display the tooltip when mouse hover on a box.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_heatmapmouseovercolor">HEATMAPMouseOverColor</code></td>
<td>
<p>Color of the box when hovered.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_heatmapmouseoveropacity">HEATMAPMouseOverOpacity</code></td>
<td>
<p>Opacity of the box when hovered.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_heatmapmouseovertooltipshtml01">HEATMAPMouseOverTooltipsHtml01</code></td>
<td>
<p>Label displayed in tooltip in first position, before chromosome number.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_heatmapmouseovertooltipshtml02">HEATMAPMouseOverTooltipsHtml02</code></td>
<td>
<p>Label displayed in tooltip in second position, before start position.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_heatmapmouseovertooltipshtml03">HEATMAPMouseOverTooltipsHtml03</code></td>
<td>
<p>Label displayed in tooltip in second position, before end position.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_heatmapmouseovertooltipshtml04">HEATMAPMouseOverTooltipsHtml04</code></td>
<td>
<p>Label displayed in tooltip in third position, before labels if any.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_heatmapmouseovertooltipshtml05">HEATMAPMouseOverTooltipsHtml05</code></td>
<td>
<p>Label displayed in tooltip in fourth position, before values.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_heatmapmouseovertooltipshtml06">HEATMAPMouseOverTooltipsHtml06</code></td>
<td>
<p>Label displayed in tooltip in fifth position, after values.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_heatmapmouseovertooltipsborderwidth">HEATMAPMouseOverTooltipsBorderWidth</code></td>
<td>
<p>The thickness of the tooltip borders, with units specified (such as em or px).</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_linemouseoutdisplay">LINEMouseOutDisplay</code></td>
<td>
<p>Hide tooltip when mouse is not hovering a line anymore.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_linemouseoutlineopacity">LINEMouseOutLineOpacity</code></td>
<td>
<p>Opacity of the line when mouse is not hovering a link anymore.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_linemouseoutlinestrokecolor">LINEMouseOutLineStrokeColor</code></td>
<td>
<p>Color of the line when mouse is not hovering anymore, in hexadecimal
RGB format. To revert back to original color, use the value &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_linemouseoutlinestrokewidth">LINEMouseOutLineStrokeWidth</code></td>
<td>
<p>Thickness of the line when mouse is not hovering a link anymore.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_linemouseoverdisplay">LINEMouseOverDisplay</code></td>
<td>
<p>Display the tooltip when mouse hover on a line.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_linemouseoverlineopacity">LINEMouseOverLineOpacity</code></td>
<td>
<p>Opacity of the line when hovered by the mouse, in hexadecimal RGB format.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_linemouseoverlinestrokecolor">LINEMouseOverLineStrokeColor</code></td>
<td>
<p>Color of the line when hovered by the mouse, in hexadecimal RGB format.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_linemouseoverlinestrokewidth">LINEMouseOverLineStrokeWidth</code></td>
<td>
<p>Width of the line when hovered by the mouse, in hexadecimal RGB format.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_linemouseovertooltipshtml01">LINEMouseOverTooltipsHtml01</code></td>
<td>
<p>Label displayed in tooltip.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_linemouseovertooltipsborderwidth">LINEMouseOverTooltipsBorderWidth</code></td>
<td>
<p>The thickness of the tooltip borders, with units specified (such as em or px).</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_cnvmouseoutdisplay">CNVMouseOutDisplay</code></td>
<td>
<p>Hide tooltip when mouse is not hovering an arc anymore.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_cnvmouseoutcolor">CNVMouseOutColor</code></td>
<td>
<p>Color of the line when mouse is not hovering anymore, in hexadecimal
RGB format. To revert back to original color, use the value &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_cnvmouseoutarcopacity">CNVMouseOutArcOpacity</code></td>
<td>
<p>Opacity of the arc when not hovered by the mouse anymore.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_cnvmouseoutarcstrokecolor">CNVMouseOutArcStrokeColor</code></td>
<td>
<p>Color of the arc's stroke when not hovered by the mouse anymore.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_cnvmouseoutarcstrokewidth">CNVMouseOutArcStrokeWidth</code></td>
<td>
<p>Width of the arc's stroke when not hovered by the mouse anymore.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_cnvmouseoverdisplay">CNVMouseOverDisplay</code></td>
<td>
<p>Display the tooltip when mouse hover on an arc.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_cnvmouseovercolor">CNVMouseOverColor</code></td>
<td>
<p>Color of the arc when hovered by the mouse, in hexadecimal RGB format.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_cnvmouseoverarcopacity">CNVMouseOverArcOpacity</code></td>
<td>
<p>Opacity of the arc when hovered by the mouse.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_cnvmouseoverarcstrokecolor">CNVMouseOverArcStrokeColor</code></td>
<td>
<p>Color of the arc's stroke when hovered by the mouse, in hexadecimal RGB format.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_cnvmouseoverarcstrokewidth">CNVMouseOverArcStrokeWidth</code></td>
<td>
<p>Width of the arc's stroke when hovered by the mouse.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_cnvmouseovertooltipshtml01">CNVMouseOverTooltipsHtml01</code></td>
<td>
<p>Label displayed in tooltip in first position, before chromosome number.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_cnvmouseovertooltipshtml02">CNVMouseOverTooltipsHtml02</code></td>
<td>
<p>Label displayed in tooltip in second position, before starting position.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_cnvmouseovertooltipshtml03">CNVMouseOverTooltipsHtml03</code></td>
<td>
<p>Label displayed in tooltip in second position, before ending position.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_cnvmouseovertooltipshtml04">CNVMouseOverTooltipsHtml04</code></td>
<td>
<p>Label displayed in tooltip in third position, before value.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_cnvmouseovertooltipshtml05">CNVMouseOverTooltipsHtml05</code></td>
<td>
<p>Label displayed in tooltip in third position, after value.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_cnvmouseovertooltipsborderwidth">CNVMouseOverTooltipsBorderWidth</code></td>
<td>
<p>The thickness of the tooltip borders, with units specified (such as em or px).</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_width">width</code>, <code id="BioCircos_+3A_height">height</code></td>
<td>
<p>Must be a valid CSS unit (like <code>'100%'</code>,
<code>'400px'</code>, <code>'auto'</code>) or a number, which will be coerced to a
string and have <code>'px'</code> appended.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_elementid">elementId</code></td>
<td>
<p>the name of the HTML id to be used to contain the visualization.</p>
</td></tr>
<tr><td><code id="BioCircos_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>BioCircos(yChr = FALSE, chrPad = 0, genomeFillColor = "Blues")

</code></pre>

<hr>
<h2 id='BioCircos-shiny'>Shiny bindings for BioCircos</h2><span id='topic+BioCircos-shiny'></span><span id='topic+BioCircosOutput'></span><span id='topic+renderBioCircos'></span>

<h3>Description</h3>

<p>Output and render functions for using BioCircos within Shiny
applications and interactive Rmd documents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BioCircosOutput(outputId, width = "100%", height = "400px")

renderBioCircos(expr, env = parent.frame(), quoted = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BioCircos-shiny_+3A_outputid">outputId</code></td>
<td>
<p>output variable to read from</p>
</td></tr>
<tr><td><code id="BioCircos-shiny_+3A_width">width</code>, <code id="BioCircos-shiny_+3A_height">height</code></td>
<td>
<p>Must be a valid CSS unit (like <code>'100%'</code>,
<code>'400px'</code>, <code>'auto'</code>) or a number, which will be coerced to a
string and have <code>'px'</code> appended.</p>
</td></tr>
<tr><td><code id="BioCircos-shiny_+3A_expr">expr</code></td>
<td>
<p>An expression that generates a BioCircos</p>
</td></tr>
<tr><td><code id="BioCircos-shiny_+3A_env">env</code></td>
<td>
<p>The environment in which to evaluate <code>expr</code>.</p>
</td></tr>
<tr><td><code id="BioCircos-shiny_+3A_quoted">quoted</code></td>
<td>
<p>Is <code>expr</code> a quoted expression (with <code>quote()</code>)? This
is useful if you want to save an expression in a variable.</p>
</td></tr>
</table>

<hr>
<h2 id='BioCircosArcTrack'>Create a track with arcs to be added to a BioCircos tracklist</h2><span id='topic+BioCircosArcTrack'></span>

<h3>Description</h3>

<p>Arcs are defined by beginning and ending genomic coordinates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BioCircosArcTrack(trackname, chromosomes, starts, ends,
  colors = "#40B9D4", labels = "", opacities = 1, maxRadius = 0.9,
  minRadius = 0.5, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BioCircosArcTrack_+3A_trackname">trackname</code></td>
<td>
<p>The name of the new track.</p>
</td></tr>
<tr><td><code id="BioCircosArcTrack_+3A_chromosomes">chromosomes</code></td>
<td>
<p>A vector containing the chromosomes on which each arc is found.
Values should match the chromosome names given in the genome parameter of the BioCircos function.</p>
</td></tr>
<tr><td><code id="BioCircosArcTrack_+3A_starts">starts</code>, <code id="BioCircosArcTrack_+3A_ends">ends</code></td>
<td>
<p>Vectors containing the coordinates on which each arc begins or ends.
Values should be inferior to the chromosome lengths given in the genome parameter of the BioCircos function.</p>
</td></tr>
<tr><td><code id="BioCircosArcTrack_+3A_colors">colors</code></td>
<td>
<p>The colors for each arc. Can be a RColorBrewer palette name used to
generate one color per arc, or a character object or vector of character objects stating RGB values in hexadecimal
format or base R colors. If the vector is shorter than the number of arcs, values will be repeated.</p>
</td></tr>
<tr><td><code id="BioCircosArcTrack_+3A_labels">labels</code></td>
<td>
<p>One or multiple character objects to label each arc.</p>
</td></tr>
<tr><td><code id="BioCircosArcTrack_+3A_opacities">opacities</code></td>
<td>
<p>One or multiple opacity values for the arcs, between 0 and 1.</p>
</td></tr>
<tr><td><code id="BioCircosArcTrack_+3A_minradius">minRadius</code>, <code id="BioCircosArcTrack_+3A_maxradius">maxRadius</code></td>
<td>
<p>Where the track should begin and end, in proportion of the inner radius of the plot.</p>
</td></tr>
<tr><td><code id="BioCircosArcTrack_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>BioCircos(BioCircosArcTrack('ArcTrack', chromosomes = 1:5, starts = 2e+7*1:5, ends = 2.5e+7*2:6))

</code></pre>

<hr>
<h2 id='BioCircosBackgroundTrack'>Create a background track to be added to a BioCircos tracklist</h2><span id='topic+BioCircosBackgroundTrack'></span>

<h3>Description</h3>

<p>Simple background to display behind another track
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BioCircosBackgroundTrack(trackname, fillColors = "#EEEEFF",
  borderColors = "#000000", maxRadius = 0.9, minRadius = 0.5,
  borderSize = 0.3, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BioCircosBackgroundTrack_+3A_trackname">trackname</code></td>
<td>
<p>The name of the new track.</p>
</td></tr>
<tr><td><code id="BioCircosBackgroundTrack_+3A_fillcolors">fillColors</code></td>
<td>
<p>The color of the background element, in hexadecimal RGB format.</p>
</td></tr>
<tr><td><code id="BioCircosBackgroundTrack_+3A_bordercolors">borderColors</code></td>
<td>
<p>The color of the background borders, in hexadecimal RGB format.</p>
</td></tr>
<tr><td><code id="BioCircosBackgroundTrack_+3A_minradius">minRadius</code>, <code id="BioCircosBackgroundTrack_+3A_maxradius">maxRadius</code></td>
<td>
<p>Where the track should begin and end, in proportion of the inner radius of the plot.</p>
</td></tr>
<tr><td><code id="BioCircosBackgroundTrack_+3A_bordersize">borderSize</code></td>
<td>
<p>The thickness of the background borders.</p>
</td></tr>
<tr><td><code id="BioCircosBackgroundTrack_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>BioCircos(BioCircosBackgroundTrack('bgTrack', fillColors="#FFEEEE", borderSize = 1))

</code></pre>

<hr>
<h2 id='BioCircosBarTrack'>Create a track with a bar plot to be added to a BioCircos tracklist</h2><span id='topic+BioCircosBarTrack'></span>

<h3>Description</h3>

<p>Bins are defined by a genomic range and associated with a numerical value
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BioCircosBarTrack(trackname, chromosomes, starts, ends, values,
  labels = "", maxRadius = 0.9, minRadius = 0.5, color = "#40B9D4",
  range = 0, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BioCircosBarTrack_+3A_trackname">trackname</code></td>
<td>
<p>The name of the new track.</p>
</td></tr>
<tr><td><code id="BioCircosBarTrack_+3A_chromosomes">chromosomes</code></td>
<td>
<p>A vector containing the chromosomes on which each bar is found.
Values should match the chromosome names given in the genome parameter of the BioCircos function.</p>
</td></tr>
<tr><td><code id="BioCircosBarTrack_+3A_starts">starts</code>, <code id="BioCircosBarTrack_+3A_ends">ends</code></td>
<td>
<p>Vectors containing the coordinates on which each bin begins or ends.</p>
</td></tr>
<tr><td><code id="BioCircosBarTrack_+3A_values">values</code></td>
<td>
<p>A vector of numerical values associated with each bin, used to determine the 
height of each bar on the track.</p>
</td></tr>
<tr><td><code id="BioCircosBarTrack_+3A_labels">labels</code></td>
<td>
<p>One or multiple character objects to label each bar.</p>
</td></tr>
<tr><td><code id="BioCircosBarTrack_+3A_minradius">minRadius</code>, <code id="BioCircosBarTrack_+3A_maxradius">maxRadius</code></td>
<td>
<p>Where the track should begin and end, in proportion of the inner radius of the plot.</p>
</td></tr>
<tr><td><code id="BioCircosBarTrack_+3A_color">color</code></td>
<td>
<p>The color for the bars, in hexadecimal RGB format.</p>
</td></tr>
<tr><td><code id="BioCircosBarTrack_+3A_range">range</code></td>
<td>
<p>Vector of values to be mapped to the minimum and maximum radii of the track.
Default to 0, mapping the minimal and maximal values input in the values parameter.</p>
</td></tr>
<tr><td><code id="BioCircosBarTrack_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>BioCircos(BioCircosBarTrack('BarTrack', chromosomes = 1:3, starts = 1e+7*2:4, ends = 2.5e+7*2:4, 
  values = 1:3, labels = c('A ', 'B ', 'C '), range = c(0,4)) + BioCircosBackgroundTrack('BGTrack'))

</code></pre>

<hr>
<h2 id='BioCircosCNVTrack'>Create a track with concentric arcs to be added to a BioCircos tracklist</h2><span id='topic+BioCircosCNVTrack'></span>

<h3>Description</h3>

<p>Arcs are defined by a genomic range and radially associated with a numerical value
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BioCircosCNVTrack(trackname, chromosomes, starts, ends, values,
  maxRadius = 0.9, minRadius = 0.5, width = 1, color = "#40B9D4",
  range = 0, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BioCircosCNVTrack_+3A_trackname">trackname</code></td>
<td>
<p>The name of the new track.</p>
</td></tr>
<tr><td><code id="BioCircosCNVTrack_+3A_chromosomes">chromosomes</code></td>
<td>
<p>A vector containing the chromosomes on which each arc is found.
Values should match the chromosome names given in the genome parameter of the BioCircos function.</p>
</td></tr>
<tr><td><code id="BioCircosCNVTrack_+3A_starts">starts</code>, <code id="BioCircosCNVTrack_+3A_ends">ends</code></td>
<td>
<p>Vectors containing the coordinates on which each arc begins or ends.</p>
</td></tr>
<tr><td><code id="BioCircosCNVTrack_+3A_values">values</code></td>
<td>
<p>A vector of numerical values associated with each bin, used to determine the 
height of each bar on the track.</p>
</td></tr>
<tr><td><code id="BioCircosCNVTrack_+3A_minradius">minRadius</code>, <code id="BioCircosCNVTrack_+3A_maxradius">maxRadius</code></td>
<td>
<p>Where the track should begin and end, in proportion of the inner radius of the plot.</p>
</td></tr>
<tr><td><code id="BioCircosCNVTrack_+3A_width">width</code></td>
<td>
<p>The thickness of the arc</p>
</td></tr>
<tr><td><code id="BioCircosCNVTrack_+3A_color">color</code></td>
<td>
<p>The color for the arcs, in hexadecimal RGB format.</p>
</td></tr>
<tr><td><code id="BioCircosCNVTrack_+3A_range">range</code></td>
<td>
<p>Vector of values to be mapped to the minimum and maximum radii of the track.
Default to 0, mapping the minimal and maximal values input in the values parameter.</p>
</td></tr>
<tr><td><code id="BioCircosCNVTrack_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>BioCircos(BioCircosCNVTrack('BarTrack', chromosomes = 1:3, starts = 1e+7*2:4, ends = 2.5e+7*2:4, 
  values = 1:3, color = "#BB0000", maxRadius = 0.85, minRadius = 0.55) 
  + BioCircosBackgroundTrack('BGTrack'))
</code></pre>

<hr>
<h2 id='BioCircosHeatmapTrack'>Create a heatmap track to be added to a BioCircos tracklist</h2><span id='topic+BioCircosHeatmapTrack'></span>

<h3>Description</h3>

<p>Heatmaps are defined by the genomic range and the color-associated numerical value
of each box of the heatmap layer
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BioCircosHeatmapTrack(trackname, chromosomes, starts, ends, values,
  labels = "", maxRadius = 0.9, minRadius = 0.5,
  color = c("#40B9D4", "#F8B100"), range = 0, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BioCircosHeatmapTrack_+3A_trackname">trackname</code></td>
<td>
<p>The name of the new track.</p>
</td></tr>
<tr><td><code id="BioCircosHeatmapTrack_+3A_chromosomes">chromosomes</code></td>
<td>
<p>A vector containing the chromosomes on which each box is found.
Values should match the chromosome names given in the genome parameter of the BioCircos function.</p>
</td></tr>
<tr><td><code id="BioCircosHeatmapTrack_+3A_starts">starts</code>, <code id="BioCircosHeatmapTrack_+3A_ends">ends</code></td>
<td>
<p>Vectors containing the coordinates on which each box begins or ends.</p>
</td></tr>
<tr><td><code id="BioCircosHeatmapTrack_+3A_values">values</code></td>
<td>
<p>A vector of numerical values associated with each box, used to determine the 
height of each bar on the track.</p>
</td></tr>
<tr><td><code id="BioCircosHeatmapTrack_+3A_labels">labels</code></td>
<td>
<p>One or multiple character objects to label each bar.</p>
</td></tr>
<tr><td><code id="BioCircosHeatmapTrack_+3A_minradius">minRadius</code>, <code id="BioCircosHeatmapTrack_+3A_maxradius">maxRadius</code></td>
<td>
<p>Where the track should begin and end, in proportion of the inner radius of the plot.</p>
</td></tr>
<tr><td><code id="BioCircosHeatmapTrack_+3A_color">color</code></td>
<td>
<p>a vector of the colors in hexadecimal RGB format to be mapped to the minimum and 
maximum values of the track.
Colors of intermediate values will be linearly interpolated between this two colors.</p>
</td></tr>
<tr><td><code id="BioCircosHeatmapTrack_+3A_range">range</code></td>
<td>
<p>a vector of the values to be mapped to the minimum and maximum colors of the track.
Default to 0, mapping the minimal and maximal values input in the values parameter.</p>
</td></tr>
<tr><td><code id="BioCircosHeatmapTrack_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>BioCircos(BioCircosHeatmapTrack('HmTrack', chromosomes = 1:3, starts = 1e+7*2:4, ends = 2.5e+7*2:4, 
  values = 1:3, labels = c('A ', 'B ', 'C ')))

</code></pre>

<hr>
<h2 id='BioCircosLineTrack'>Create a track with lines to be added to a BioCircos tracklist</h2><span id='topic+BioCircosLineTrack'></span>

<h3>Description</h3>

<p>Lines are defined by genomic coordinates and values of an ordered set of points,
that will define the edges of the segments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BioCircosLineTrack(trackname, chromosomes, positions, values,
  color = "#40B9D4", width = 2, maxRadius = 0.9, minRadius = 0.5,
  range = 0, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BioCircosLineTrack_+3A_trackname">trackname</code></td>
<td>
<p>The name of the new track.</p>
</td></tr>
<tr><td><code id="BioCircosLineTrack_+3A_chromosomes">chromosomes</code></td>
<td>
<p>A vector containing the chromosomes on which each vertex is found.
Values should match the chromosome names given in the genome parameter of the BioCircos function.</p>
</td></tr>
<tr><td><code id="BioCircosLineTrack_+3A_positions">positions</code></td>
<td>
<p>A vector containing the coordinates on which each vertex are found.
Values should be inferior to the chromosome lengths given in the genome parameter of the BioCircos function.</p>
</td></tr>
<tr><td><code id="BioCircosLineTrack_+3A_values">values</code></td>
<td>
<p>A vector of numerical values associated with each vertex, used to determine the 
radial coordinate of each vertex on the visualization.</p>
</td></tr>
<tr><td><code id="BioCircosLineTrack_+3A_color">color</code></td>
<td>
<p>The color of the line in hexadecimal RGB format.</p>
</td></tr>
<tr><td><code id="BioCircosLineTrack_+3A_width">width</code></td>
<td>
<p>The line width.</p>
</td></tr>
<tr><td><code id="BioCircosLineTrack_+3A_minradius">minRadius</code>, <code id="BioCircosLineTrack_+3A_maxradius">maxRadius</code></td>
<td>
<p>Where the track should begin and end, in proportion of the inner radius of the plot.</p>
</td></tr>
<tr><td><code id="BioCircosLineTrack_+3A_range">range</code></td>
<td>
<p>Vector of values to be mapped to the minimum and maximum radii of the track.
Default to 0, mapping the minimal and maximal values input in the values parameter.</p>
</td></tr>
<tr><td><code id="BioCircosLineTrack_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>BioCircos(BioCircosLineTrack('LnId', rep(1,30), 2e+6*(1:100), log(1:100)) 
  + BioCircosBackgroundTrack('BGId'))
</code></pre>

<hr>
<h2 id='BioCircosLinkTrack'>Create an inner track with links to be added to a BioCircos tracklist</h2><span id='topic+BioCircosLinkTrack'></span>

<h3>Description</h3>

<p>Links are defined by beginning and ending genomic coordinates of the 2 regions to linked,
such as the positions linked in genomic fusions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BioCircosLinkTrack(trackname, gene1Chromosomes, gene1Starts, gene1Ends,
  gene2Chromosomes, gene2Starts, gene2Ends, color = "#40B9D4",
  labels = "", maxRadius = 0.4, width = "0.1em", gene1Names = "",
  gene2Names = "", displayAxis = TRUE, axisColor = "#B8B8B8",
  axisWidth = 0.5, axisPadding = 0, displayLabel = TRUE,
  labelColor = "#000000", labelSize = "1em", labelPadding = 3, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BioCircosLinkTrack_+3A_trackname">trackname</code></td>
<td>
<p>The name of the new track.</p>
</td></tr>
<tr><td><code id="BioCircosLinkTrack_+3A_gene1chromosomes">gene1Chromosomes</code>, <code id="BioCircosLinkTrack_+3A_gene1starts">gene1Starts</code>, <code id="BioCircosLinkTrack_+3A_gene1ends">gene1Ends</code>, <code id="BioCircosLinkTrack_+3A_gene1names">gene1Names</code>, <code id="BioCircosLinkTrack_+3A_gene2chromosomes">gene2Chromosomes</code>, <code id="BioCircosLinkTrack_+3A_gene2starts">gene2Starts</code>, <code id="BioCircosLinkTrack_+3A_gene2ends">gene2Ends</code>, <code id="BioCircosLinkTrack_+3A_gene2names">gene2Names</code></td>
<td>
<p>Vectors with the chromosomes, genomic coordinates of beginning and end, and names of both genes to link.
Chromosomes and positions should respect the chromosome names and lengths given in the genome parameter of
the BioCircos function.</p>
</td></tr>
<tr><td><code id="BioCircosLinkTrack_+3A_color">color</code></td>
<td>
<p>The color for the links, in hexadecimal RGB format.</p>
</td></tr>
<tr><td><code id="BioCircosLinkTrack_+3A_labels">labels</code></td>
<td>
<p>A vector of character objects to label each link.</p>
</td></tr>
<tr><td><code id="BioCircosLinkTrack_+3A_maxradius">maxRadius</code></td>
<td>
<p>Where the track should end, in proportion of the inner radius of the plot.</p>
</td></tr>
<tr><td><code id="BioCircosLinkTrack_+3A_width">width</code></td>
<td>
<p>The thickness of the links.</p>
</td></tr>
<tr><td><code id="BioCircosLinkTrack_+3A_displayaxis">displayAxis</code></td>
<td>
<p>Display additional axis (i.e. circle) around the track.</p>
</td></tr>
<tr><td><code id="BioCircosLinkTrack_+3A_axiscolor">axisColor</code>, <code id="BioCircosLinkTrack_+3A_axiswidth">axisWidth</code>, <code id="BioCircosLinkTrack_+3A_axispadding">axisPadding</code></td>
<td>
<p>Color, thickness and padding of the additional axis.</p>
</td></tr>
<tr><td><code id="BioCircosLinkTrack_+3A_displaylabel">displayLabel</code></td>
<td>
<p>Display labels of the track.</p>
</td></tr>
<tr><td><code id="BioCircosLinkTrack_+3A_labelcolor">labelColor</code>, <code id="BioCircosLinkTrack_+3A_labelsize">labelSize</code>, <code id="BioCircosLinkTrack_+3A_labelpadding">labelPadding</code></td>
<td>
<p>Color, font size and padding of the labels around the track.</p>
</td></tr>
<tr><td><code id="BioCircosLinkTrack_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>start_chromosomes &lt;- 1:5
end_chromosomes &lt;- 2*10:6
start_pos &lt;- 2.5e+7*2:6
end_pos &lt;- 2e+7*1:5
BioCircos(BioCircosLinkTrack('LinkTrack', start_chromosomes, start_pos, start_pos+1,
  end_chromosomes, end_pos, end_pos+1, color = '#FF00FF'))

</code></pre>

<hr>
<h2 id='BioCircosSNPTrack'>Create a track with SNPs to be added to a BioCircos tracklist</h2><span id='topic+BioCircosSNPTrack'></span>

<h3>Description</h3>

<p>SNPs are defined by genomic coordinates and associated with a numerical value
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BioCircosSNPTrack(trackname, chromosomes, positions, values,
  colors = "#40B9D4", labels = "", size = 2, shape = "circle",
  opacities = 1, maxRadius = 0.9, minRadius = 0.5, range = 0, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BioCircosSNPTrack_+3A_trackname">trackname</code></td>
<td>
<p>The name of the new track.</p>
</td></tr>
<tr><td><code id="BioCircosSNPTrack_+3A_chromosomes">chromosomes</code></td>
<td>
<p>A vector containing the chromosomes on which each SNP are found.
Values should match the chromosome names given in the genome parameter of the BioCircos function.</p>
</td></tr>
<tr><td><code id="BioCircosSNPTrack_+3A_positions">positions</code></td>
<td>
<p>A vector containing the coordinates on which each SNP are found.
Values should be inferior to the chromosome lengths given in the genome parameter of the BioCircos function.</p>
</td></tr>
<tr><td><code id="BioCircosSNPTrack_+3A_values">values</code></td>
<td>
<p>A vector of numerical values associated with each SNPs, used to determine the 
radial coordinates of each point on the visualization.</p>
</td></tr>
<tr><td><code id="BioCircosSNPTrack_+3A_colors">colors</code></td>
<td>
<p>The colors for each point. Can be a RColorBrewer palette name used to
generate one color per point, or a character object or vector of character objects stating RGB values in hexadecimal
format or base R colors. If the vector is shorter than the number of points, values will be repeated.</p>
</td></tr>
<tr><td><code id="BioCircosSNPTrack_+3A_labels">labels</code></td>
<td>
<p>One or multiple character objects to label each point.</p>
</td></tr>
<tr><td><code id="BioCircosSNPTrack_+3A_size">size</code></td>
<td>
<p>The size of each point.</p>
</td></tr>
<tr><td><code id="BioCircosSNPTrack_+3A_shape">shape</code></td>
<td>
<p>Shape of the points. Can be &quot;circle&quot; or &quot;rect&quot;.</p>
</td></tr>
<tr><td><code id="BioCircosSNPTrack_+3A_opacities">opacities</code></td>
<td>
<p>One or multiple opacity values for the points, between 0 and 1.</p>
</td></tr>
<tr><td><code id="BioCircosSNPTrack_+3A_minradius">minRadius</code>, <code id="BioCircosSNPTrack_+3A_maxradius">maxRadius</code></td>
<td>
<p>Where the track should begin and end, in proportion of the inner radius of the plot.</p>
</td></tr>
<tr><td><code id="BioCircosSNPTrack_+3A_range">range</code></td>
<td>
<p>Vector of values to be mapped to the minimum and maximum radii of the track.
Default to 0, mapping the minimal and maximal values input in the values parameter.</p>
</td></tr>
<tr><td><code id="BioCircosSNPTrack_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>BioCircos(BioCircosSNPTrack('SNPTrack', chromosomes = 1:3, positions = 1e+7*2:4, 
  values = 1:3, colors = "Accent", labels = c('A', 'B', 'C')) + BioCircosBackgroundTrack('BGTrack'))

</code></pre>

<hr>
<h2 id='BioCircosTextTrack'>Create a Text track to be added to a BioCircos tracklist</h2><span id='topic+BioCircosTextTrack'></span>

<h3>Description</h3>

<p>Simple text annotation displayed in the visualization
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BioCircosTextTrack(trackname, text, x = -0.15, y = 0, size = "1.2em",
  weight = "bold", opacity = 1, color = "#000000", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BioCircosTextTrack_+3A_trackname">trackname</code></td>
<td>
<p>The name of the new track.</p>
</td></tr>
<tr><td><code id="BioCircosTextTrack_+3A_text">text</code></td>
<td>
<p>The text to be displayed.</p>
</td></tr>
<tr><td><code id="BioCircosTextTrack_+3A_x">x</code>, <code id="BioCircosTextTrack_+3A_y">y</code></td>
<td>
<p>Coordinates of the lower left corner of the annotation, in proportion of the inner radius of the plot.</p>
</td></tr>
<tr><td><code id="BioCircosTextTrack_+3A_size">size</code></td>
<td>
<p>Font size, with units specified (such as em or px).</p>
</td></tr>
<tr><td><code id="BioCircosTextTrack_+3A_weight">weight</code></td>
<td>
<p>Font weight. Can be &quot;normal&quot;, &quot;bold&quot;, &quot;bolder&quot; or &quot;lighter&quot;.</p>
</td></tr>
<tr><td><code id="BioCircosTextTrack_+3A_opacity">opacity</code></td>
<td>
<p>Font opacity.</p>
</td></tr>
<tr><td><code id="BioCircosTextTrack_+3A_color">color</code></td>
<td>
<p>Font color, in hexadecimal RGB format.</p>
</td></tr>
<tr><td><code id="BioCircosTextTrack_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>BioCircos(BioCircosTextTrack('textTrack', 'Annotation', color = '#DD2222', x = -0.3))

</code></pre>

<hr>
<h2 id='BioCircosTracklist'>Create a list of BioCircos tracks</h2><span id='topic+BioCircosTracklist'></span><span id='topic++2B.BioCircosTracklist'></span><span id='topic+-.BioCircosTracklist'></span>

<h3>Description</h3>

<p>This allows the use of the '+' and '-' operator on these lists
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BioCircosTracklist()

## S3 method for class 'BioCircosTracklist'
x + ...

## S3 method for class 'BioCircosTracklist'
x - ...
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BioCircosTracklist_+3A_x">x</code></td>
<td>
<p>The tracklist on which other tracks should be added or removed.</p>
</td></tr>
<tr><td><code id="BioCircosTracklist_+3A_...">...</code></td>
<td>
<p>The tracks to add (as tracklists) or to remove (as track names).</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
