<!DOCTYPE html><html><head><title>Help for package biodosetools</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {biodosetools}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#%&gt;%'><p>Pipe operator</p></a></li>
<li><a href='#AIC_from_data'><p>Calculate AIC (Akaike's 'An Information Criterion')</p></a></li>
<li><a href='#biodosetools'><p><code>biodosetools</code> package</p></a></li>
<li><a href='#calculate_aberr'><p>Aberration calculation functions</p></a></li>
<li><a href='#calculate_aberr_table'><p>Calculate aberrations table</p></a></li>
<li><a href='#calculate_genome_factor'><p>Calculate genomic conversion factor</p></a></li>
<li><a href='#calculate_model_stats'><p>Calculate model statistics</p></a></li>
<li><a href='#calculate_trans_rate_manual'><p>Calculate manual translocation rate</p></a></li>
<li><a href='#calculate_trans_rate_sigurdson'><p>Calculate Sigurdson's translocation rate</p></a></li>
<li><a href='#calculate_yield'><p>Calculate yield from dose</p></a></li>
<li><a href='#calculate_yield_infimum'><p>Calculate theoretical yield infimum</p></a></li>
<li><a href='#correct_boundary'><p>Correct boundary of irradiated fractions to be bounded by 0 and 1</p></a></li>
<li><a href='#correct_conf_int'><p>Correct yield confidence interval</p></a></li>
<li><a href='#correct_negative_vals'><p>Correct negative values</p></a></li>
<li><a href='#correct_yield'><p>Correct yields if they are below the curve</p></a></li>
<li><a href='#dna_content_fractions_ihgsc'><p>DNA Content Fractions of Human Chromosomes (IHGSC)</p></a></li>
<li><a href='#dna_content_fractions_morton'><p>DNA Content Fractions of Human Chromosomes (Morton 1991)</p></a></li>
<li><a href='#estimate_hetero_mixed_poisson'><p>Heterogeneous dose estimation (Mixed Poisson model)</p></a></li>
<li><a href='#estimate_partial_body_dolphin'><p>Partial-body dose estimation (Dolphin's method)</p></a></li>
<li><a href='#estimate_whole_body_delta'><p>Whole-body dose estimation (delta method)</p></a></li>
<li><a href='#estimate_whole_body_merkle'><p>Whole-body dose estimation (Merkle's method)</p></a></li>
<li><a href='#fit'><p>Perform dose-effect fitting algorithm</p></a></li>
<li><a href='#fit_glm_method'><p>Perform GLM (Generalised Linear Model) fitting</p></a></li>
<li><a href='#fit_maxlik_method'><p>Perform max-likelihood optimization fitting</p></a></li>
<li><a href='#get_deltamethod_std_err'><p>Get standard errors using delta method</p></a></li>
<li><a href='#include_help'><p>Include Markdown help</p></a></li>
<li><a href='#load_rmd_report'><p>Load RMarkdown report</p></a></li>
<li><a href='#plot_estimated_dose_curve'><p>Plot dose estimation curve</p></a></li>
<li><a href='#plot_fit_dose_curve'><p>Plot fit dose curve</p></a></li>
<li><a href='#prepare_maxlik_count_data'><p>Prepare count data for max-likelihood optimization fitting</p></a></li>
<li><a href='#project_yield'><p>Project yield into dose-effect fitting curve</p></a></li>
<li><a href='#protracted_g_function'><p>Calculate protracted function <code class="reqn">G(x)</code></p></a></li>
<li><a href='#R_factor'><p>Calculate R regression confidence factor</p></a></li>
<li><a href='#run_app'><p>Run the Shiny Application</p></a></li>
<li><a href='#yield_error_fun'><p>Calculate yield error</p></a></li>
<li><a href='#yield_fun'><p>Calculate yield</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>An R Shiny Application for Biological Dosimetry</td>
</tr>
<tr>
<td>Version:</td>
<td>3.6.1</td>
</tr>
<tr>
<td>Description:</td>
<td>A tool to perform all different statistical tests and calculations
    needed by Biological Dosimetry Laboratories.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://biodosetools-team.github.io/biodosetools/">https://biodosetools-team.github.io/biodosetools/</a>,
<a href="https://github.com/biodosetools-team/biodosetools/">https://github.com/biodosetools-team/biodosetools/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/biodosetools-team/biodosetools/issues/">https://github.com/biodosetools-team/biodosetools/issues/</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>bsplus, dplyr (&ge; 1.0.0), ggplot2, magrittr, MASS, maxLik,
mixtools, msm, rhandsontable, rlang (&ge; 0.4.0), rmarkdown,
shiny, shinydashboard, shinyWidgets (&ge; 0.5.0), tidyr (&ge;
1.0.0), config, golem, cli</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), covr, knitr, kableExtra, markdown,
pander, tinytex, xtable</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-11-16 15:26:28 UTC; aldomann</td>
</tr>
<tr>
<td>Author:</td>
<td>Alfredo Hernández <a href="https://orcid.org/0000-0002-2660-4545"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  David Endesfelder [aut],
  Pere Puig <a href="https://orcid.org/0000-0002-6607-9642"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alfredo Hernández &lt;aldomann.designs@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-11-16 16:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling 'rhs(lhs)'.
</p>

<hr>
<h2 id='AIC_from_data'>Calculate AIC (Akaike's 'An Information Criterion')</h2><span id='topic+AIC_from_data'></span>

<h3>Description</h3>

<p>Calculate AIC (Akaike's 'An Information Criterion')
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AIC_from_data(
  general_fit_coeffs,
  data,
  dose_var = "dose",
  yield_var = "yield",
  fit_link = "identity"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AIC_from_data_+3A_general_fit_coeffs">general_fit_coeffs</code></td>
<td>
<p>Generalised fit coefficients matrix.</p>
</td></tr>
<tr><td><code id="AIC_from_data_+3A_data">data</code></td>
<td>
<p>Data (dose, yield) to calculate AIC from.</p>
</td></tr>
<tr><td><code id="AIC_from_data_+3A_dose_var">dose_var</code></td>
<td>
<p>Name of the dose variable (enquoted).</p>
</td></tr>
<tr><td><code id="AIC_from_data_+3A_yield_var">yield_var</code></td>
<td>
<p>Name of the yield variable (enquoted).</p>
</td></tr>
<tr><td><code id="AIC_from_data_+3A_fit_link">fit_link</code></td>
<td>
<p>A specification for the model link function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value of AIC.
</p>

<hr>
<h2 id='biodosetools'><code>biodosetools</code> package</h2><span id='topic+biodosetools'></span>

<h3>Description</h3>

<p>Shiny App To Be Used By Biological Dosimetry Laboratories
</p>


<h3>Details</h3>

<p>See the README on
<a href="https://github.com/biodosetools-team/biodosetools/">GitHub</a>
</p>

<hr>
<h2 id='calculate_aberr'>Aberration calculation functions</h2><span id='topic+calculate_aberr'></span><span id='topic+calculate_aberr_power'></span><span id='topic+calculate_aberr_mean'></span><span id='topic+calculate_aberr_var'></span><span id='topic+calculate_aberr_disp_index'></span><span id='topic+calculate_aberr_u_value'></span><span id='topic+init_aberr_table'></span>

<h3>Description</h3>

<p>Aberration calculation functions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_aberr_power(data, aberr_prefix = "C", power = 1)

calculate_aberr_mean(X, N)

calculate_aberr_var(X, X2, N)

calculate_aberr_disp_index(mean, var)

calculate_aberr_u_value(X, N, mean, var, assessment_u = 1)

init_aberr_table(
  data,
  type = c("count", "case"),
  aberr_module = c("dicentrics", "translocations", "micronuclei")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_aberr_+3A_data">data</code></td>
<td>
<p>Count or case data.</p>
</td></tr>
<tr><td><code id="calculate_aberr_+3A_aberr_prefix">aberr_prefix</code></td>
<td>
<p>Prefix of the aberrations in the data.</p>
</td></tr>
<tr><td><code id="calculate_aberr_+3A_power">power</code></td>
<td>
<p>Power of aberration.</p>
</td></tr>
<tr><td><code id="calculate_aberr_+3A_x">X</code></td>
<td>
<p>Sum of detected aberrations.</p>
</td></tr>
<tr><td><code id="calculate_aberr_+3A_n">N</code></td>
<td>
<p>Number of cells analysed.</p>
</td></tr>
<tr><td><code id="calculate_aberr_+3A_x2">X2</code></td>
<td>
<p>Quadratic sum of detected aberrations.</p>
</td></tr>
<tr><td><code id="calculate_aberr_+3A_mean">mean</code></td>
<td>
<p>Mean.</p>
</td></tr>
<tr><td><code id="calculate_aberr_+3A_var">var</code></td>
<td>
<p>Variance.</p>
</td></tr>
<tr><td><code id="calculate_aberr_+3A_assessment_u">assessment_u</code></td>
<td>
<p>Expected <code class="reqn">u</code>-value of the assessment. For a Poisson distribution this should be unity.</p>
</td></tr>
<tr><td><code id="calculate_aberr_+3A_type">type</code></td>
<td>
<p>Type of input data. Either &quot;count&quot; and &quot;case&quot;.</p>
</td></tr>
<tr><td><code id="calculate_aberr_+3A_aberr_module">aberr_module</code></td>
<td>
<p>Aberration module.</p>
</td></tr>
</table>

<hr>
<h2 id='calculate_aberr_table'>Calculate aberrations table</h2><span id='topic+calculate_aberr_table'></span>

<h3>Description</h3>

<p>Calculate aberrations table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_aberr_table(
  data,
  type = c("count", "case"),
  aberr_module = c("dicentrics", "translocations", "micronuclei"),
  assessment_u = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_aberr_table_+3A_data">data</code></td>
<td>
<p>Count or case data.</p>
</td></tr>
<tr><td><code id="calculate_aberr_table_+3A_type">type</code></td>
<td>
<p>Type of input data. Either &quot;count&quot; and &quot;case&quot;.</p>
</td></tr>
<tr><td><code id="calculate_aberr_table_+3A_aberr_module">aberr_module</code></td>
<td>
<p>Aberration module, required for <code>type = "case"</code>.</p>
</td></tr>
<tr><td><code id="calculate_aberr_table_+3A_assessment_u">assessment_u</code></td>
<td>
<p>Expected <code class="reqn">u</code>-value of the assessment. For a Poisson distribution this should be unity.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame containing cell count (<code class="reqn">N</code>), aberrations (<code class="reqn">X</code>),
and other coefficients (dispersion index, <code class="reqn">u</code>-value, ...), as well as
raw count or case <code>data</code>.
</p>

<hr>
<h2 id='calculate_genome_factor'>Calculate genomic conversion factor</h2><span id='topic+calculate_genome_factor'></span>

<h3>Description</h3>

<p>Method based on the paper by Lucas, J. N. et al. (1992). Rapid Translocation
Frequency Analysis in Humans Decades after Exposure to Ionizing Radiation.
International Journal of Radiation Biology, 62(1), 53-63.
&lt;doi:10.1080/09553009214551821&gt;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_genome_factor(dna_table, chromosomes, colors, sex)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_genome_factor_+3A_dna_table">dna_table</code></td>
<td>
<p>DNA content fractions table. Can be <code>dna_content_fractions_morton</code> or <code>dna_content_table_ihgsc</code>.</p>
</td></tr>
<tr><td><code id="calculate_genome_factor_+3A_chromosomes">chromosomes</code></td>
<td>
<p>Vector of stained chromosomes.</p>
</td></tr>
<tr><td><code id="calculate_genome_factor_+3A_colors">colors</code></td>
<td>
<p>Vector of colors of the stains.</p>
</td></tr>
<tr><td><code id="calculate_genome_factor_+3A_sex">sex</code></td>
<td>
<p>Sex of the individual.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value of genomic conversion factor.
</p>

<hr>
<h2 id='calculate_model_stats'>Calculate model statistics</h2><span id='topic+calculate_model_stats'></span>

<h3>Description</h3>

<p>Calculate model statistics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_model_stats(
  model_data,
  fit_coeffs_vec,
  glm_results = NULL,
  fit_algorithm = NULL,
  response = "yield",
  link = c("identity", "log"),
  type = c("theory", "raw"),
  Y = NULL,
  mu = NULL,
  n = NULL,
  npar = NULL,
  genome_factor = NULL,
  calc_type = c("fitting", "estimation")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_model_stats_+3A_model_data">model_data</code></td>
<td>
<p>Data of the model.</p>
</td></tr>
<tr><td><code id="calculate_model_stats_+3A_fit_coeffs_vec">fit_coeffs_vec</code></td>
<td>
<p>Vector of fitting coefficients.</p>
</td></tr>
<tr><td><code id="calculate_model_stats_+3A_glm_results">glm_results</code></td>
<td>
<p>Results of glm.</p>
</td></tr>
<tr><td><code id="calculate_model_stats_+3A_fit_algorithm">fit_algorithm</code></td>
<td>
<p>String of the algorithm used.</p>
</td></tr>
<tr><td><code id="calculate_model_stats_+3A_response">response</code></td>
<td>
<p>Type of response.</p>
</td></tr>
<tr><td><code id="calculate_model_stats_+3A_link">link</code></td>
<td>
<p>Fit link.</p>
</td></tr>
<tr><td><code id="calculate_model_stats_+3A_type">type</code></td>
<td>
<p>Theoretical or raw glm model statistics.</p>
</td></tr>
<tr><td><code id="calculate_model_stats_+3A_y">Y</code></td>
<td>
<p>Y response (required in constraint-maxlik-optimization).</p>
</td></tr>
<tr><td><code id="calculate_model_stats_+3A_mu">mu</code></td>
<td>
<p>mu response required in constraint-maxlik-optimization).</p>
</td></tr>
<tr><td><code id="calculate_model_stats_+3A_n">n</code></td>
<td>
<p>number of parameters (required in constraint-maxlik-optimization).</p>
</td></tr>
<tr><td><code id="calculate_model_stats_+3A_npar">npar</code></td>
<td>
<p>number of parameters (required in constraint-maxlik-optimization).</p>
</td></tr>
<tr><td><code id="calculate_model_stats_+3A_genome_factor">genome_factor</code></td>
<td>
<p>Genomic conversion factor used in translocations.</p>
</td></tr>
<tr><td><code id="calculate_model_stats_+3A_calc_type">calc_type</code></td>
<td>
<p>Calculation type, either &quot;fitting&quot; or &quot;estimation&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame of model statistics.
</p>

<hr>
<h2 id='calculate_trans_rate_manual'>Calculate manual translocation rate</h2><span id='topic+calculate_trans_rate_manual'></span>

<h3>Description</h3>

<p>Calculate manual translocation rate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_trans_rate_manual(cells, genome_factor, expected_aberr_value)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_trans_rate_manual_+3A_cells">cells</code></td>
<td>
<p>Number of cells <code>N</code>.</p>
</td></tr>
<tr><td><code id="calculate_trans_rate_manual_+3A_genome_factor">genome_factor</code></td>
<td>
<p>Genomic conversion factor.</p>
</td></tr>
<tr><td><code id="calculate_trans_rate_manual_+3A_expected_aberr_value">expected_aberr_value</code></td>
<td>
<p>Expected aberrations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value of translocation rate.
</p>

<hr>
<h2 id='calculate_trans_rate_sigurdson'>Calculate Sigurdson's translocation rate</h2><span id='topic+calculate_trans_rate_sigurdson'></span>

<h3>Description</h3>

<p>Method based on the paper by Sigurdson, A. J. et al. (2008). International
study of factors affecting human chromosome translocations. Mutation
Research/Genetic Toxicology and Environmental Mutagenesis, 652(2), 112-121.
&lt;doi:10.1016/j.mrgentox.2008.01.005&gt;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_trans_rate_sigurdson(
  cells,
  genome_factor,
  age_value,
  sex_bool = FALSE,
  sex_value = "none",
  smoker_bool = FALSE,
  ethnicity_value = "none",
  region_value = "none"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_trans_rate_sigurdson_+3A_cells">cells</code></td>
<td>
<p>Number of cells <code>N</code>.</p>
</td></tr>
<tr><td><code id="calculate_trans_rate_sigurdson_+3A_genome_factor">genome_factor</code></td>
<td>
<p>Genomic conversion factor.</p>
</td></tr>
<tr><td><code id="calculate_trans_rate_sigurdson_+3A_age_value">age_value</code></td>
<td>
<p>Age of the individual.</p>
</td></tr>
<tr><td><code id="calculate_trans_rate_sigurdson_+3A_sex_bool">sex_bool</code></td>
<td>
<p>If <code>TRUE</code>, <code>sex_value</code> will be used.</p>
</td></tr>
<tr><td><code id="calculate_trans_rate_sigurdson_+3A_sex_value">sex_value</code></td>
<td>
<p>Sex of the individual, either &quot;male&quot; of &quot;female&quot;.</p>
</td></tr>
<tr><td><code id="calculate_trans_rate_sigurdson_+3A_smoker_bool">smoker_bool</code></td>
<td>
<p>Whether the individual smokes or not.</p>
</td></tr>
<tr><td><code id="calculate_trans_rate_sigurdson_+3A_ethnicity_value">ethnicity_value</code></td>
<td>
<p>Ethnicity of the individual.</p>
</td></tr>
<tr><td><code id="calculate_trans_rate_sigurdson_+3A_region_value">region_value</code></td>
<td>
<p>Region of the individual.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value of translocation rate.
</p>

<hr>
<h2 id='calculate_yield'>Calculate yield from dose</h2><span id='topic+calculate_yield'></span>

<h3>Description</h3>

<p>Calculate yield from dose
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_yield(
  dose,
  type = c("estimate", "lower", "upper"),
  general_fit_coeffs,
  general_fit_var_cov_mat = NULL,
  protracted_g_value = 1,
  conf_int = 0.95
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_yield_+3A_dose">dose</code></td>
<td>
<p>Numeric value of dose.</p>
</td></tr>
<tr><td><code id="calculate_yield_+3A_type">type</code></td>
<td>
<p>Type of yield calculation. Can be &quot;estimate&quot;, &quot;lower&quot;, or &quot;upper&quot;.</p>
</td></tr>
<tr><td><code id="calculate_yield_+3A_general_fit_coeffs">general_fit_coeffs</code></td>
<td>
<p>Generalised fit coefficients matrix.</p>
</td></tr>
<tr><td><code id="calculate_yield_+3A_general_fit_var_cov_mat">general_fit_var_cov_mat</code></td>
<td>
<p>Generalised variance-covariance matrix.</p>
</td></tr>
<tr><td><code id="calculate_yield_+3A_protracted_g_value">protracted_g_value</code></td>
<td>
<p>Protracted <code class="reqn">G(x)</code> value.</p>
</td></tr>
<tr><td><code id="calculate_yield_+3A_conf_int">conf_int</code></td>
<td>
<p>Curve confidence interval, 95% by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value of yield.
</p>

<hr>
<h2 id='calculate_yield_infimum'>Calculate theoretical yield infimum</h2><span id='topic+calculate_yield_infimum'></span>

<h3>Description</h3>

<p>Calculate theoretical yield infimum
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_yield_infimum(
  type = c("estimate", "lower", "upper"),
  general_fit_coeffs,
  general_fit_var_cov_mat = NULL,
  conf_int = 0.95
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_yield_infimum_+3A_type">type</code></td>
<td>
<p>Type of yield calculation. Can be &quot;estimate&quot;, &quot;lower&quot;, or &quot;upper&quot;.</p>
</td></tr>
<tr><td><code id="calculate_yield_infimum_+3A_general_fit_coeffs">general_fit_coeffs</code></td>
<td>
<p>Generalised fit coefficients matrix.</p>
</td></tr>
<tr><td><code id="calculate_yield_infimum_+3A_general_fit_var_cov_mat">general_fit_var_cov_mat</code></td>
<td>
<p>Generalised variance-covariance matrix.</p>
</td></tr>
<tr><td><code id="calculate_yield_infimum_+3A_conf_int">conf_int</code></td>
<td>
<p>Curve confidence interval, 95% by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value of yield infimum.
</p>

<hr>
<h2 id='correct_boundary'>Correct boundary of irradiated fractions to be bounded by 0 and 1</h2><span id='topic+correct_boundary'></span>

<h3>Description</h3>

<p>Correct boundary of irradiated fractions to be bounded by 0 and 1
</p>


<h3>Usage</h3>

<pre><code class='language-R'>correct_boundary(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="correct_boundary_+3A_x">x</code></td>
<td>
<p>Numeric value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value in [0, 1] range.
</p>

<hr>
<h2 id='correct_conf_int'>Correct yield confidence interval</h2><span id='topic+correct_conf_int'></span>

<h3>Description</h3>

<p>Correct yield confidence interval if simple method is required.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>correct_conf_int(
  conf_int,
  general_fit_var_cov_mat,
  protracted_g_value = 1,
  type,
  dose = seq(0, 10, 0.2)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="correct_conf_int_+3A_conf_int">conf_int</code></td>
<td>
<p>Confidence interval.</p>
</td></tr>
<tr><td><code id="correct_conf_int_+3A_general_fit_var_cov_mat">general_fit_var_cov_mat</code></td>
<td>
<p>Generalised variance-covariance matrix.</p>
</td></tr>
<tr><td><code id="correct_conf_int_+3A_protracted_g_value">protracted_g_value</code></td>
<td>
<p>Protracted <code class="reqn">G(x)</code> value.</p>
</td></tr>
<tr><td><code id="correct_conf_int_+3A_type">type</code></td>
<td>
<p>Type of yield calculation. Can be &quot;estimate&quot;, &quot;lower&quot;, or &quot;upper&quot;.</p>
</td></tr>
<tr><td><code id="correct_conf_int_+3A_dose">dose</code></td>
<td>
<p>Numeric value of dose.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value of corrected confidence interval.
</p>

<hr>
<h2 id='correct_negative_vals'>Correct negative values</h2><span id='topic+correct_negative_vals'></span>

<h3>Description</h3>

<p>Correct negative values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>correct_negative_vals(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="correct_negative_vals_+3A_x">x</code></td>
<td>
<p>Numeric value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value corrected to zero if negative.
</p>

<hr>
<h2 id='correct_yield'>Correct yields if they are below the curve</h2><span id='topic+correct_yield'></span>

<h3>Description</h3>

<p>Correct yields if they are below the curve
</p>


<h3>Usage</h3>

<pre><code class='language-R'>correct_yield(
  yield,
  type = "estimate",
  general_fit_coeffs,
  general_fit_var_cov_mat,
  conf_int
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="correct_yield_+3A_yield">yield</code></td>
<td>
<p>Numeric value of yield.</p>
</td></tr>
<tr><td><code id="correct_yield_+3A_type">type</code></td>
<td>
<p>Type of yield calculation. Can be &quot;estimate&quot;, &quot;lower&quot;, or &quot;upper&quot;.</p>
</td></tr>
<tr><td><code id="correct_yield_+3A_general_fit_coeffs">general_fit_coeffs</code></td>
<td>
<p>Generalised fit coefficients matrix.</p>
</td></tr>
<tr><td><code id="correct_yield_+3A_general_fit_var_cov_mat">general_fit_var_cov_mat</code></td>
<td>
<p>Generalised variance-covariance matrix.</p>
</td></tr>
<tr><td><code id="correct_yield_+3A_conf_int">conf_int</code></td>
<td>
<p>Curve confidence interval.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value of corrected yield.
</p>

<hr>
<h2 id='dna_content_fractions_ihgsc'>DNA Content Fractions of Human Chromosomes (IHGSC)</h2><span id='topic+dna_content_fractions_ihgsc'></span>

<h3>Description</h3>

<p>Normalised DNA Content of Human Chromosomes from
the International Human Genome Sequencing Consortium.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dna_content_fractions_ihgsc
</code></pre>


<h3>Format</h3>

<p>A data frame with 24 rows and 3 variables:
</p>

<dl>
<dt>chromosome</dt><dd><p>Chromosome.</p>
</dd>
<dt>fraction_male</dt><dd><p>Normalised content of megabases on male human DNA.</p>
</dd>
<dt>fraction_female</dt><dd><p>Normalised content of megabases on female human DNA.</p>
</dd>
</dl>



<h3>Details</h3>

<p>Last accessed in July 2020.
</p>


<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/grc/human/data">https://www.ncbi.nlm.nih.gov/grc/human/data</a>
</p>

<hr>
<h2 id='dna_content_fractions_morton'>DNA Content Fractions of Human Chromosomes (Morton 1991)</h2><span id='topic+dna_content_fractions_morton'></span>

<h3>Description</h3>

<p>Normalised DNA Content of Human Chromosomes from
Morton, N. E. (1991). Parameters of the human genome. Proceedings of the National Academy of Sciences, 88(17), 7474-7476.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dna_content_fractions_morton
</code></pre>


<h3>Format</h3>

<p>A data frame with 24 rows and 3 variables:
</p>

<dl>
<dt>chromosome</dt><dd><p>Chromosome.</p>
</dd>
<dt>fraction_male</dt><dd><p>Normalised content of megabases on male human DNA.</p>
</dd>
<dt>fraction_female</dt><dd><p>Normalised content of megabases on female human DNA.</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://doi.org/10.1073/pnas.88.17.7474">doi:10.1073/pnas.88.17.7474</a>
</p>

<hr>
<h2 id='estimate_hetero_mixed_poisson'>Heterogeneous dose estimation (Mixed Poisson model)</h2><span id='topic+estimate_hetero_mixed_poisson'></span>

<h3>Description</h3>

<p>Method based on the paper by Pujol, M. et al. (2016). A New Model for
Biological Dose Assessment in Cases of Heterogeneous Exposures to Ionizing
Radiation. Radiation Research, 185(2), 151-162. &lt;doi:10.1667/RR14145.1&gt;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_hetero_mixed_poisson(
  case_data,
  fit_coeffs,
  fit_var_cov_mat,
  conf_int = 0.95,
  protracted_g_value = 1,
  gamma,
  gamma_error
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_hetero_mixed_poisson_+3A_case_data">case_data</code></td>
<td>
<p>Case data in data frame form.</p>
</td></tr>
<tr><td><code id="estimate_hetero_mixed_poisson_+3A_fit_coeffs">fit_coeffs</code></td>
<td>
<p>Fitting coefficients matrix.</p>
</td></tr>
<tr><td><code id="estimate_hetero_mixed_poisson_+3A_fit_var_cov_mat">fit_var_cov_mat</code></td>
<td>
<p>Fitting variance-covariance matrix.</p>
</td></tr>
<tr><td><code id="estimate_hetero_mixed_poisson_+3A_conf_int">conf_int</code></td>
<td>
<p>Confidence interval, 95% by default.</p>
</td></tr>
<tr><td><code id="estimate_hetero_mixed_poisson_+3A_protracted_g_value">protracted_g_value</code></td>
<td>
<p>Protracted <code class="reqn">G(x)</code> value.</p>
</td></tr>
<tr><td><code id="estimate_hetero_mixed_poisson_+3A_gamma">gamma</code></td>
<td>
<p>Survival coefficient of irradiated cells.</p>
</td></tr>
<tr><td><code id="estimate_hetero_mixed_poisson_+3A_gamma_error">gamma_error</code></td>
<td>
<p>Error of the survival coefficient of irradiated cells.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List containing estimated mixing proportions data frame, estimated yields data
frame, estimated doses data frame, estimated fraction of irradiated blood data frame,
AIC, and <code>conf_int_*</code> used.
</p>

<hr>
<h2 id='estimate_partial_body_dolphin'>Partial-body dose estimation (Dolphin's method)</h2><span id='topic+estimate_partial_body_dolphin'></span>

<h3>Description</h3>

<p>Method based on the paper by Dolphin, G. W. (1969). Biological Dosimetry with
Particular Reference to Chromosome Aberration Analysis: A Review of Methods.
International Atomic Energy Agency (IAEA) Retrieved from
<a href="https://inis.iaea.org/search/search.aspx?orig_q=RN:45029080">https://inis.iaea.org/search/search.aspx?orig_q=RN:45029080</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_partial_body_dolphin(
  case_data,
  fit_coeffs,
  fit_var_cov_mat,
  conf_int = 0.95,
  protracted_g_value = 1,
  genome_factor = 1,
  gamma,
  aberr_module = c("dicentrics", "translocations", "micronuclei")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_partial_body_dolphin_+3A_case_data">case_data</code></td>
<td>
<p>Case data in data frame form.</p>
</td></tr>
<tr><td><code id="estimate_partial_body_dolphin_+3A_fit_coeffs">fit_coeffs</code></td>
<td>
<p>Fitting coefficients matrix.</p>
</td></tr>
<tr><td><code id="estimate_partial_body_dolphin_+3A_fit_var_cov_mat">fit_var_cov_mat</code></td>
<td>
<p>Fitting variance-covariance matrix.</p>
</td></tr>
<tr><td><code id="estimate_partial_body_dolphin_+3A_conf_int">conf_int</code></td>
<td>
<p>Confidence interval, 95% by default.</p>
</td></tr>
<tr><td><code id="estimate_partial_body_dolphin_+3A_protracted_g_value">protracted_g_value</code></td>
<td>
<p>Protracted <code class="reqn">G(x)</code> value.</p>
</td></tr>
<tr><td><code id="estimate_partial_body_dolphin_+3A_genome_factor">genome_factor</code></td>
<td>
<p>Genomic conversion factor used in translocations, else 1.</p>
</td></tr>
<tr><td><code id="estimate_partial_body_dolphin_+3A_gamma">gamma</code></td>
<td>
<p>Survival coefficient of irradiated cells.</p>
</td></tr>
<tr><td><code id="estimate_partial_body_dolphin_+3A_aberr_module">aberr_module</code></td>
<td>
<p>Aberration module.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List containing estimated doses data frame, observed fraction of cells scored
which were irradiated, estimated fraction of irradiated blood data frame, AIC, and
<code>conf_int_*</code> used.
</p>

<hr>
<h2 id='estimate_whole_body_delta'>Whole-body dose estimation (delta method)</h2><span id='topic+estimate_whole_body_delta'></span>

<h3>Description</h3>

<p>Method based on 2001 manual by the International Atomic Energy Agency (IAEA).
Cytogenetic Analysis for Radiation Dose Assessment, Technical Reports Series
(2001). Retrieved from <a href="https://www.iaea.org/publications/6303/cytogenetic-analysis-for-radiation-dose-assessment">https://www.iaea.org/publications/6303/cytogenetic-analysis-for-radiation-dose-assessment</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_whole_body_delta(
  case_data,
  fit_coeffs,
  fit_var_cov_mat,
  conf_int = 0.95,
  protracted_g_value = 1,
  aberr_module = c("dicentrics", "translocations", "micronuclei")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_whole_body_delta_+3A_case_data">case_data</code></td>
<td>
<p>Case data in data frame form.</p>
</td></tr>
<tr><td><code id="estimate_whole_body_delta_+3A_fit_coeffs">fit_coeffs</code></td>
<td>
<p>Fitting coefficients matrix.</p>
</td></tr>
<tr><td><code id="estimate_whole_body_delta_+3A_fit_var_cov_mat">fit_var_cov_mat</code></td>
<td>
<p>Fitting variance-covariance matrix.</p>
</td></tr>
<tr><td><code id="estimate_whole_body_delta_+3A_conf_int">conf_int</code></td>
<td>
<p>Confidence interval, 95% by default.</p>
</td></tr>
<tr><td><code id="estimate_whole_body_delta_+3A_protracted_g_value">protracted_g_value</code></td>
<td>
<p>Protracted <code class="reqn">G(x)</code> value.</p>
</td></tr>
<tr><td><code id="estimate_whole_body_delta_+3A_aberr_module">aberr_module</code></td>
<td>
<p>Aberration module.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List containing estimated doses data frame, AIC, and <code>conf_int</code> used.
</p>

<hr>
<h2 id='estimate_whole_body_merkle'>Whole-body dose estimation (Merkle's method)</h2><span id='topic+estimate_whole_body_merkle'></span>

<h3>Description</h3>

<p>Method based on the paper by Merkle, W. (1983). Statistical methods in
regression and calibration analysis of chromosome aberration data. Radiation
and Environmental Biophysics, 21(3), 217-233. &lt;doi:10.1007/BF01323412&gt;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_whole_body_merkle(
  case_data,
  fit_coeffs,
  fit_var_cov_mat,
  conf_int_yield = 0.83,
  conf_int_curve = 0.83,
  protracted_g_value = 1,
  genome_factor = 1,
  aberr_module = c("dicentrics", "translocations", "micronuclei")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_whole_body_merkle_+3A_case_data">case_data</code></td>
<td>
<p>Case data in data frame form.</p>
</td></tr>
<tr><td><code id="estimate_whole_body_merkle_+3A_fit_coeffs">fit_coeffs</code></td>
<td>
<p>Fitting coefficients matrix.</p>
</td></tr>
<tr><td><code id="estimate_whole_body_merkle_+3A_fit_var_cov_mat">fit_var_cov_mat</code></td>
<td>
<p>Fitting variance-covariance matrix.</p>
</td></tr>
<tr><td><code id="estimate_whole_body_merkle_+3A_conf_int_yield">conf_int_yield</code></td>
<td>
<p>Confidence interval of the yield, 83% by default.</p>
</td></tr>
<tr><td><code id="estimate_whole_body_merkle_+3A_conf_int_curve">conf_int_curve</code></td>
<td>
<p>Confidence interval of the curve, 83% by default.</p>
</td></tr>
<tr><td><code id="estimate_whole_body_merkle_+3A_protracted_g_value">protracted_g_value</code></td>
<td>
<p>Protracted <code class="reqn">G(x)</code> value.</p>
</td></tr>
<tr><td><code id="estimate_whole_body_merkle_+3A_genome_factor">genome_factor</code></td>
<td>
<p>Genomic conversion factor used in translocations, else 1.</p>
</td></tr>
<tr><td><code id="estimate_whole_body_merkle_+3A_aberr_module">aberr_module</code></td>
<td>
<p>Aberration module.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List containing estimated doses data frame, AIC, and <code>conf_int_*</code> used.
</p>

<hr>
<h2 id='fit'>Perform dose-effect fitting algorithm</h2><span id='topic+fit'></span>

<h3>Description</h3>

<p>Perform dose-effect fitting. A generalized linear model (GLM) is used by
default, with a maximum likelihood estimation (MLE) as a fallback method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit(
  count_data,
  model_formula,
  model_family,
  fit_link = "identity",
  aberr_module = c("dicentrics", "translocations", "micronuclei"),
  algorithm = c("glm", "maxlik")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_+3A_count_data">count_data</code></td>
<td>
<p>Count data in data frame form.</p>
</td></tr>
<tr><td><code id="fit_+3A_model_formula">model_formula</code></td>
<td>
<p>Model formula.</p>
</td></tr>
<tr><td><code id="fit_+3A_model_family">model_family</code></td>
<td>
<p>Model family.</p>
</td></tr>
<tr><td><code id="fit_+3A_fit_link">fit_link</code></td>
<td>
<p>Family link.</p>
</td></tr>
<tr><td><code id="fit_+3A_aberr_module">aberr_module</code></td>
<td>
<p>Aberration module.</p>
</td></tr>
<tr><td><code id="fit_+3A_algorithm">algorithm</code></td>
<td>
<p>Optional selection of algorithm to be used, either &quot;glm&quot; (for GLM) or &quot;maxlik&quot; (for MLE). By default, &quot;glm&quot; is used, with &quot;maxlik&quot; as a fallback method.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The GLM method is based on the paper by Edwards, A. A. et al. (1979).
Radiation induced chromosome aberrations and the Poisson distribution.
Radiation and Environmental Biophysics, 16(2), 89-100.
&lt;doi:10.1007/BF01323216&gt;.
</p>
<p>The MLE method is based on the paperby Oliveira, M. et al. (2016).
Zero-inflated regression models for radiation-induced chromosome aberration
data: A comparative study. Biometrical Journal, 58(2), 259-279.
&lt;doi:10.1002/bimj.201400233&gt;.
</p>


<h3>Value</h3>

<p>List object containing fit results either using GLM or maxLik optimization.
</p>

<hr>
<h2 id='fit_glm_method'>Perform GLM (Generalised Linear Model) fitting</h2><span id='topic+fit_glm_method'></span>

<h3>Description</h3>

<p>Method based on the paper by Edwards, A. A. et al. (1979). Radiation induced
chromosome aberrations and the Poisson distribution. Radiation and
Environmental Biophysics, 16(2), 89-100. &lt;doi:10.1007/BF01323216&gt;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_glm_method(
  count_data,
  model_formula,
  model_family = c("automatic", "poisson", "quasipoisson", "nb2"),
  fit_link = "identity",
  aberr_module = c("dicentrics", "translocations", "micronuclei")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_glm_method_+3A_count_data">count_data</code></td>
<td>
<p>Count data in data frame form.</p>
</td></tr>
<tr><td><code id="fit_glm_method_+3A_model_formula">model_formula</code></td>
<td>
<p>Model formula.</p>
</td></tr>
<tr><td><code id="fit_glm_method_+3A_model_family">model_family</code></td>
<td>
<p>Model family.</p>
</td></tr>
<tr><td><code id="fit_glm_method_+3A_fit_link">fit_link</code></td>
<td>
<p>Family link.</p>
</td></tr>
<tr><td><code id="fit_glm_method_+3A_aberr_module">aberr_module</code></td>
<td>
<p>Aberration module.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List object containing GLM fit results.
</p>

<hr>
<h2 id='fit_maxlik_method'>Perform max-likelihood optimization fitting</h2><span id='topic+fit_maxlik_method'></span>

<h3>Description</h3>

<p>Method based on the paper by Oliveira, M. et al. (2016). Zero-inflated
regression models for radiation-induced chromosome aberration data:
A comparative study. Biometrical Journal, 58(2), 259-279.
&lt;doi:10.1002/bimj.201400233&gt;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_maxlik_method(
  data,
  model_formula,
  model_family = c("automatic", "poisson", "quasipoisson", "nb2"),
  fit_link,
  aberr_module = c("dicentrics", "translocations", "micronuclei")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_maxlik_method_+3A_data">data</code></td>
<td>
<p>Count data.</p>
</td></tr>
<tr><td><code id="fit_maxlik_method_+3A_model_formula">model_formula</code></td>
<td>
<p>Model formula.</p>
</td></tr>
<tr><td><code id="fit_maxlik_method_+3A_model_family">model_family</code></td>
<td>
<p>Model family.</p>
</td></tr>
<tr><td><code id="fit_maxlik_method_+3A_fit_link">fit_link</code></td>
<td>
<p>Family link.</p>
</td></tr>
<tr><td><code id="fit_maxlik_method_+3A_aberr_module">aberr_module</code></td>
<td>
<p>Aberration module.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List object containing maxLik fit results.
</p>

<hr>
<h2 id='get_deltamethod_std_err'>Get standard errors using delta method</h2><span id='topic+get_deltamethod_std_err'></span>

<h3>Description</h3>

<p>Delta method for approximating the standard error of a transformation <code class="reqn">g(X)</code> of a random variable <code class="reqn">X = (x1, x2, ...)</code>, given estimates of the mean and covariance matrix of <code class="reqn">X</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_deltamethod_std_err(
  fit_is_lq,
  variable = c("dose", "fraction_partial", "fraction_hetero"),
  mean_estimate,
  cov_estimate,
  protracted_g_value = NA,
  d0 = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_deltamethod_std_err_+3A_fit_is_lq">fit_is_lq</code></td>
<td>
<p>Whether the fit is linear quadratic (<code>TRUE</code>) or linear (<code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="get_deltamethod_std_err_+3A_variable">variable</code></td>
<td>
<p>Variable resulting of the transformation <code class="reqn">g(X)</code>.</p>
</td></tr>
<tr><td><code id="get_deltamethod_std_err_+3A_mean_estimate">mean_estimate</code></td>
<td>
<p>The estimated mean of <code class="reqn">X</code>.</p>
</td></tr>
<tr><td><code id="get_deltamethod_std_err_+3A_cov_estimate">cov_estimate</code></td>
<td>
<p>The estimated covariance matrix of <code class="reqn">X</code>.</p>
</td></tr>
<tr><td><code id="get_deltamethod_std_err_+3A_protracted_g_value">protracted_g_value</code></td>
<td>
<p>Protracted <code class="reqn">G(x)</code> value.</p>
</td></tr>
<tr><td><code id="get_deltamethod_std_err_+3A_d0">d0</code></td>
<td>
<p>Survival coefficient of irradiated cells.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value containing the standard error of the dose estimate.
</p>

<hr>
<h2 id='include_help'>Include Markdown help</h2><span id='topic+include_help'></span>

<h3>Description</h3>

<p>Include Markdown help
</p>


<h3>Usage</h3>

<pre><code class='language-R'>include_help(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="include_help_+3A_...">...</code></td>
<td>
<p>Character vector specifying directory and or file to
point to inside the current package.</p>
</td></tr>
</table>

<hr>
<h2 id='load_rmd_report'>Load RMarkdown report</h2><span id='topic+load_rmd_report'></span>

<h3>Description</h3>

<p>Load RMarkdown report
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_rmd_report(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_rmd_report_+3A_...">...</code></td>
<td>
<p>Character vector specifying directory and or file to
point to inside the current package.</p>
</td></tr>
</table>

<hr>
<h2 id='plot_estimated_dose_curve'>Plot dose estimation curve</h2><span id='topic+plot_estimated_dose_curve'></span>

<h3>Description</h3>

<p>Plot dose estimation curve
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_estimated_dose_curve(
  est_doses,
  fit_coeffs,
  fit_var_cov_mat,
  protracted_g_value = 1,
  conf_int_curve,
  aberr_name
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_estimated_dose_curve_+3A_est_doses">est_doses</code></td>
<td>
<p>List of dose estimations results from <code>estimate_*()</code> family of functions.</p>
</td></tr>
<tr><td><code id="plot_estimated_dose_curve_+3A_fit_coeffs">fit_coeffs</code></td>
<td>
<p>Fitting coefficients matrix.</p>
</td></tr>
<tr><td><code id="plot_estimated_dose_curve_+3A_fit_var_cov_mat">fit_var_cov_mat</code></td>
<td>
<p>Fitting variance-covariance matrix.</p>
</td></tr>
<tr><td><code id="plot_estimated_dose_curve_+3A_protracted_g_value">protracted_g_value</code></td>
<td>
<p>Protracted <code class="reqn">G(x)</code> value.</p>
</td></tr>
<tr><td><code id="plot_estimated_dose_curve_+3A_conf_int_curve">conf_int_curve</code></td>
<td>
<p>Confidence interval of the curve.</p>
</td></tr>
<tr><td><code id="plot_estimated_dose_curve_+3A_aberr_name">aberr_name</code></td>
<td>
<p>Name of the aberration to use in the y-axis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>ggplot2</code> object.
</p>

<hr>
<h2 id='plot_fit_dose_curve'>Plot fit dose curve</h2><span id='topic+plot_fit_dose_curve'></span>

<h3>Description</h3>

<p>Plot fit dose curve
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_fit_dose_curve(fit_results_list, aberr_name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_fit_dose_curve_+3A_fit_results_list">fit_results_list</code></td>
<td>
<p>List of fit results.</p>
</td></tr>
<tr><td><code id="plot_fit_dose_curve_+3A_aberr_name">aberr_name</code></td>
<td>
<p>Name of the aberration to use in the y-axis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>ggplot2</code> object.
</p>

<hr>
<h2 id='prepare_maxlik_count_data'>Prepare count data for max-likelihood optimization fitting</h2><span id='topic+prepare_maxlik_count_data'></span>

<h3>Description</h3>

<p>Prepare count data for max-likelihood optimization fitting
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_maxlik_count_data(
  count_data,
  model_formula,
  aberr_module = c("dicentrics", "translocations", "micronuclei")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare_maxlik_count_data_+3A_count_data">count_data</code></td>
<td>
<p>Count data in data frame form.</p>
</td></tr>
<tr><td><code id="prepare_maxlik_count_data_+3A_model_formula">model_formula</code></td>
<td>
<p>Model formula.</p>
</td></tr>
<tr><td><code id="prepare_maxlik_count_data_+3A_aberr_module">aberr_module</code></td>
<td>
<p>Aberration module.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame of parsed count data.
</p>

<hr>
<h2 id='project_yield'>Project yield into dose-effect fitting curve</h2><span id='topic+project_yield'></span>

<h3>Description</h3>

<p>Project yield into dose-effect fitting curve
</p>


<h3>Usage</h3>

<pre><code class='language-R'>project_yield(
  yield,
  type = "estimate",
  general_fit_coeffs,
  general_fit_var_cov_mat = NULL,
  protracted_g_value = 1,
  conf_int = 0.95
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="project_yield_+3A_yield">yield</code></td>
<td>
<p>Yield to be projected.</p>
</td></tr>
<tr><td><code id="project_yield_+3A_type">type</code></td>
<td>
<p>Type of yield calculation. Can be &quot;estimate&quot;, &quot;lower&quot;, or &quot;upper&quot;.</p>
</td></tr>
<tr><td><code id="project_yield_+3A_general_fit_coeffs">general_fit_coeffs</code></td>
<td>
<p>Generalised fit coefficients matrix.</p>
</td></tr>
<tr><td><code id="project_yield_+3A_general_fit_var_cov_mat">general_fit_var_cov_mat</code></td>
<td>
<p>Generalised variance-covariance matrix.</p>
</td></tr>
<tr><td><code id="project_yield_+3A_protracted_g_value">protracted_g_value</code></td>
<td>
<p>Protracted <code class="reqn">G(x)</code> value.</p>
</td></tr>
<tr><td><code id="project_yield_+3A_conf_int">conf_int</code></td>
<td>
<p>Curve confidence interval, 95% by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value of projected dose.
</p>

<hr>
<h2 id='protracted_g_function'>Calculate protracted function <code class="reqn">G(x)</code></h2><span id='topic+protracted_g_function'></span>

<h3>Description</h3>

<p>Calculation based on the paper by Lea, D. E. &amp; Catcheside, D. G. (1942). The
mechanism of the induction by radiation of chromosome aberrations
in<em>Tradescantia</em>. Journal of Genetics, 44(2-3), 216-245.
&lt;doi:10.1007/BF02982830&gt;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>protracted_g_function(time, time_0 = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="protracted_g_function_+3A_time">time</code></td>
<td>
<p>Time over which the irradiation occurred.</p>
</td></tr>
<tr><td><code id="protracted_g_function_+3A_time_0">time_0</code></td>
<td>
<p>The mean lifetime of the breaks, which has been shown to be on the order of ~ 2 hours (default value).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value of <code class="reqn">G(x)</code>.
</p>

<hr>
<h2 id='R_factor'>Calculate R regression confidence factor</h2><span id='topic+R_factor'></span>

<h3>Description</h3>

<p>Calculate R regression confidence factor depending on selected confidence interval and type of fit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>R_factor(general_fit_coeffs, conf_int = 0.95)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="R_factor_+3A_general_fit_coeffs">general_fit_coeffs</code></td>
<td>
<p>Generalised fit coefficients matrix.</p>
</td></tr>
<tr><td><code id="R_factor_+3A_conf_int">conf_int</code></td>
<td>
<p>Confidence interval, 95% by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value of R regression confidence factor.
</p>

<hr>
<h2 id='run_app'>Run the Shiny Application</h2><span id='topic+run_app'></span>

<h3>Description</h3>

<p>Run the Shiny Application
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_app(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_app_+3A_...">...</code></td>
<td>
<p>A series of options to be used inside the app.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Used for side-effect.
</p>

<hr>
<h2 id='yield_error_fun'>Calculate yield error</h2><span id='topic+yield_error_fun'></span>

<h3>Description</h3>

<p>Calculate yield error using Merkle's method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>yield_error_fun(dose, general_fit_var_cov_mat = NULL, protracted_g_value = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="yield_error_fun_+3A_dose">dose</code></td>
<td>
<p>Numeric value of dose.</p>
</td></tr>
<tr><td><code id="yield_error_fun_+3A_general_fit_var_cov_mat">general_fit_var_cov_mat</code></td>
<td>
<p>Generalised variance-covariance matrix.</p>
</td></tr>
<tr><td><code id="yield_error_fun_+3A_protracted_g_value">protracted_g_value</code></td>
<td>
<p>Protracted <code class="reqn">G(x)</code> value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value of yield error.
</p>

<hr>
<h2 id='yield_fun'>Calculate yield</h2><span id='topic+yield_fun'></span>

<h3>Description</h3>

<p>Calculate yield
</p>


<h3>Usage</h3>

<pre><code class='language-R'>yield_fun(dose, general_fit_coeffs, protracted_g_value = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="yield_fun_+3A_dose">dose</code></td>
<td>
<p>Numeric value of dose.</p>
</td></tr>
<tr><td><code id="yield_fun_+3A_general_fit_coeffs">general_fit_coeffs</code></td>
<td>
<p>Generalised fit coefficients matrix.</p>
</td></tr>
<tr><td><code id="yield_fun_+3A_protracted_g_value">protracted_g_value</code></td>
<td>
<p>Protracted <code class="reqn">G(x)</code> value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value of yield.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
