<!DOCTYPE html><html lang="en"><head><title>Help for package BioInsight</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BioInsight}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#describeRNA'><p>describeRNA</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Filter and Plot RNA Biotypes</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Analyze and plot the abundance of different RNA biotypes present in a count matrix, this evaluation can be useful if you want to test different strategies of normalization or analyze a particular biotype in a differential gene expression analysis.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>knitr, wordcloud, RColorBrewer</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), edgeR, limma</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), biomaRt</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-06-20 06:55:27 UTC; justb</td>
</tr>
<tr>
<td>Author:</td>
<td>Luiz Antonio de Jesus Rocha [aut, cre, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Luiz Antonio de Jesus Rocha &lt;luiz.rocha@unifesp.br&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-06-20 07:20:02 UTC</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
</table>
<hr>
<h2 id='describeRNA'>describeRNA</h2><span id='topic+describeRNA'></span>

<h3>Description</h3>

<p>This function provide filterByExpr with two customized options. See 'Arguments'. Different strategies can be useful either if you are trying to compare different approaches of normalization or if you want to analyze a particular biotype where a different variation of expression is expected under certain conditions. A BioInsight data frame will return with your new count matrix where you can proceed with your Differential Expression Analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>describeRNA(
  counts,
  biotypes,
  groups,
  report = FALSE,
  verbose = FALSE,
  filter = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="describeRNA_+3A_counts">counts</code></td>
<td>
<p>data.frame where you have gene counts</p>
</td></tr>
<tr><td><code id="describeRNA_+3A_biotypes">biotypes</code></td>
<td>
<p>data.frame where you have a gene_biotype column</p>
</td></tr>
<tr><td><code id="describeRNA_+3A_groups">groups</code></td>
<td>
<p>factor with groups and samples (see Examples)</p>
</td></tr>
<tr><td><code id="describeRNA_+3A_report">report</code></td>
<td>
<p>if TRUE will generate a .pdf file in /tmp/ folder ('wordcloud' and 'RColorBrewer' are necessary)</p>
</td></tr>
<tr><td><code id="describeRNA_+3A_verbose">verbose</code></td>
<td>
<p>if TRUE will print your result in your console ('knitr' is necessary)</p>
</td></tr>
<tr><td><code id="describeRNA_+3A_filter">filter</code></td>
<td>
<p>Your threshold (see edgeR::filterByExpr)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For filter you may want to use &quot;1&quot; if you want the default option from filterByExpr. If 2 &quot;Slightly&quot; above the default will be applied. If &quot;3&quot; A more restrictive option will be applied.
</p>


<h3>Note</h3>

<p>You can use trace(describeRNA, edit=T) to set different values as threshold for &quot;filter&quot; option.
</p>


<h3>References</h3>

<p>Robinson MD, McCarthy DJ, Smyth GK. edgeR: a Bioconductor package for differential expression analysis of digital gene expression data. Bioinformatics. 2010;26(1):139-140. doi:10.1093/bioinformatics/btp616
</p>


<h3>Examples</h3>

<pre><code class='language-R'>counts = system.file("extdata", "count_matrix.tsv", package="BioInsight")
counts = read.table(counts, row.names=1, header=TRUE)
biotypes = system.file("extdata", "Rattus_Norvegicus_biomart.tsv", package="BioInsight")
biotypes = read.table(biotypes, row.names=1, header=TRUE)

groups = rep(as.factor(c("1","2")), each=5)

describeRNA(counts=counts,
            biotypes=biotypes,
            groups=groups,
            filter=2)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
