<!DOCTYPE html><html><head><title>Help for package biomartr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {biomartr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#biomartr-package'><p>Genomic Data Retrieval</p></a></li>
<li><a href='#biomart'><p>Main BioMart Query Function</p></a></li>
<li><a href='#cachedir'><p>Get directory to store back end files like kingdom summaries etc</p></a></li>
<li><a href='#cachedir_set'><p>Set directory to store back end files like kingdom summaries etc</p></a></li>
<li><a href='#check_annotation_biomartr'><p>Check whether an annotation file contains outlier lines</p></a></li>
<li><a href='#download.database'><p>Download a NCBI Database to Your Local Hard Drive</p></a></li>
<li><a href='#download.database.all'><p>Download all elements of an NCBI databse</p></a></li>
<li><a href='#ensembl_divisions'><p>List all available ENSEMBL divisions</p></a></li>
<li><a href='#get.ensembl.info'><p>Helper function to retrieve species information from the ENSEMBL API</p></a></li>
<li><a href='#getAssemblyStats'><p>Genome Assembly Stats Retrieval</p></a></li>
<li><a href='#getAttributes'><p>Retrieve All Available Attributes for a Specific Dataset</p></a></li>
<li><a href='#getBio'><p>A wrapper to all bio getters, selected with 'type' argument</p></a></li>
<li><a href='#getBioSet'><p>Generic Bio data set extractor</p></a></li>
<li><a href='#getCDS'><p>Coding Sequence Retrieval</p></a></li>
<li><a href='#getCDSSet'><p>CDS retrieval of multiple species</p></a></li>
<li><a href='#getCollection'><p>Retrieve a Collection: Genome, Proteome, CDS, RNA, GFF, Repeat Masker, AssemblyStats</p></a></li>
<li><a href='#getCollectionSet'><p>Retrieve a Collection: Genome, Proteome, CDS, RNA, GFF, Repeat Masker, AssemblyStats of multiple species</p></a></li>
<li><a href='#getDatasets'><p>Retrieve All Available Datasets for a BioMart Database</p></a></li>
<li><a href='#getENSEMBL'><p>Download sequence or annotation from ENSEMBL</p></a></li>
<li><a href='#getENSEMBL.gtf'><p>Helper function for retrieving gtf files from ENSEMBL</p></a></li>
<li><a href='#getENSEMBL.Seq'><p>Helper function for retrieving biological sequence files from ENSEMBL</p></a></li>
<li><a href='#getENSEMBLGENOMESInfo'><p>Retrieve ENSEMBLGENOMES info file</p></a></li>
<li><a href='#getENSEMBLInfo'><p>Retrieve ENSEMBL info file</p></a></li>
<li><a href='#getFilters'><p>Retrieve All Available Filters for a Specific Dataset</p></a></li>
<li><a href='#getGenome'><p>Genome Retrieval</p></a></li>
<li><a href='#getGENOMEREPORT'><p>Retrieve NCBI GENOME_REPORTS file</p></a></li>
<li><a href='#getGenomeSet'><p>Genome Retrieval of multiple species</p></a></li>
<li><a href='#getGFF'><p>Genome Annotation Retrieval (GFF3)</p></a></li>
<li><a href='#getGFFSet'><p>GFF retrieval of multiple species</p></a></li>
<li><a href='#getGO'><p>Gene Ontology Query</p></a></li>
<li><a href='#getGroups'><p>Retrieve available groups for a kingdom of life (only available for NCBI RefSeq and NCBI Genbank)</p></a></li>
<li><a href='#getGTF'><p>Genome Annotation Retrieval (GTF)</p></a></li>
<li><a href='#getKingdomAssemblySummary'><p>Retrieve and summarise the assembly_summary.txt files from</p>
NCBI for all kingdoms</a></li>
<li><a href='#getKingdoms'><p>Retrieve available kingdoms of life</p></a></li>
<li><a href='#getMarts'><p>Retrieve information about available Ensembl Biomart databases</p></a></li>
<li><a href='#getMetaGenomeAnnotations'><p>Retrieve annotation *.gff files for metagenomes from NCBI Genbank</p></a></li>
<li><a href='#getMetaGenomes'><p>Retrieve metagenomes from NCBI Genbank</p></a></li>
<li><a href='#getMetaGenomeSummary'><p>Retrieve the assembly_summary.txt file from NCBI genbank metagenomes</p></a></li>
<li><a href='#getProteome'><p>Proteome Retrieval</p></a></li>
<li><a href='#getProteomeSet'><p>Proteome retrieval of multiple species</p></a></li>
<li><a href='#getReleases'><p>Retrieve available database releases or versions</p>
of ENSEMBL</a></li>
<li><a href='#getRepeatMasker'><p>Repeat Masker Retrieval</p></a></li>
<li><a href='#getRNA'><p>RNA Sequence Retrieval</p></a></li>
<li><a href='#getRNASet'><p>RNA Retrieval of multiple species</p></a></li>
<li><a href='#getSummaryFile'><p>Helper function to retrieve the assembly_summary.txt file from NCBI</p></a></li>
<li><a href='#getUniProtInfo'><p>Get uniprot info from organism</p></a></li>
<li><a href='#getUniProtSTATS'><p>Retrieve UniProt Database Information File (STATS)</p></a></li>
<li><a href='#is.genome.available'><p>Check Genome Availability</p></a></li>
<li><a href='#listDatabases'><p>Retrieve a List of Available NCBI Databases for Download</p></a></li>
<li><a href='#listGenomes'><p>List All Available Genomes either by kingdom, group, or subgroup</p></a></li>
<li><a href='#listGroups'><p>List number of available genomes in each taxonomic group</p></a></li>
<li><a href='#listKingdoms'><p>List number of available genomes in each kingdom of life</p></a></li>
<li><a href='#listMetaGenomes'><p>List available metagenomes on NCBI Genbank</p></a></li>
<li><a href='#meta.retrieval'><p>Perform Meta-Genome Retrieval</p></a></li>
<li><a href='#meta.retrieval.all'><p>Perform Meta-Genome Retrieval of all organisms in all kingdoms of life</p></a></li>
<li><a href='#organismAttributes'><p>Retrieve Ensembl Biomart attributes for a query organism</p></a></li>
<li><a href='#organismBM'><p>Retrieve Ensembl Biomart marts and datasets for a query organism</p></a></li>
<li><a href='#organismFilters'><p>Retrieve Ensembl Biomart filters for a query organism</p></a></li>
<li><a href='#read_assemblystats'><p>Import Genome Assembly Stats File</p></a></li>
<li><a href='#read_cds'><p>Import CDS as Biostrings or data.table object</p></a></li>
<li><a href='#read_genome'><p>Import Genome Assembly as Biostrings or data.table object</p></a></li>
<li><a href='#read_gff'><p>Import GFF File</p></a></li>
<li><a href='#read_proteome'><p>Import Proteome as Biostrings or data.table object</p></a></li>
<li><a href='#read_rm'><p>Import Repeat Masker output file</p></a></li>
<li><a href='#read_rna'><p>Import RNA as Biostrings or data.table object</p></a></li>
<li><a href='#refseqOrganisms'><p>Retrieve All Organism Names Stored on refseq</p></a></li>
<li><a href='#summary_cds'><p>Retrieve summary statistics for a coding sequence (CDS) file</p></a></li>
<li><a href='#summary_genome'><p>Retrieve summary statistics for a genome assembly file</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Genomic Data Retrieval</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.7</td>
</tr>
<tr>
<td>Description:</td>
<td>Perform large scale genomic data retrieval and functional annotation retrieval. This package aims to provide users with a standardized
                way to automate genome, proteome, 'RNA', coding sequence ('CDS'), 'GFF', and metagenome
                retrieval from 'NCBI RefSeq', 'NCBI Genbank', 'ENSEMBL', 
                and 'UniProt' databases. Furthermore, an interface to the 'BioMart' database
                (Smedley et al. (2009) &lt;<a href="https://doi.org/10.1186%2F1471-2164-10-22">doi:10.1186/1471-2164-10-22</a>&gt;) allows users to retrieve
                functional annotation for genomic loci. In addition, users can download entire databases such
                as 'NCBI RefSeq' (Pruitt et al. (2007) &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgkl842">doi:10.1093/nar/gkl842</a>&gt;), 'NCBI nr',
                'NCBI nt', 'NCBI Genbank' (Benson et al. (2013) &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgks1195">doi:10.1093/nar/gks1195</a>&gt;), etc. with only one command.</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>biomaRt, Biostrings, curl, tibble, jsonlite, data.table (&ge;
1.9.4), dplyr (&ge; 0.3.0), readr (&ge; 1.4.0), downloader (&ge;
0.3), RCurl (&ge; 1.95-4.5), XML (&ge; 3.98-1.1), httr (&ge; 0.6.1),
stringr (&ge; 0.6.2), purrr, R.utils, philentropy, withr, fs</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr (&ge; 1.6), rmarkdown (&ge; 0.3.3), devtools (&ge; 1.6.1),
testthat, seqinr, magrittr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://docs.ropensci.org/biomartr/">https://docs.ropensci.org/biomartr/</a>,
<a href="https://github.com/ropensci/biomartr">https://github.com/ropensci/biomartr</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ropensci/biomartr/issues">https://github.com/ropensci/biomartr/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>X-schema.org-keywords:</td>
<td>BioMart, genomic-data-retrieval,
annotation-retrieval, database-retrieval, NCBI, ENSEMBL,
biological-data-retrieval</td>
</tr>
<tr>
<td>X-schema.org-applicationCategory:</td>
<td>Data Access</td>
</tr>
<tr>
<td>X-schema.org-isPartof:</td>
<td>"ropensci.org"</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-12-02 14:46:32 UTC; hdrost</td>
</tr>
<tr>
<td>Author:</td>
<td>Hajk-Georg Drost <a href="https://orcid.org/0000-0002-1567-306X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Haakon Tjeldnes [aut, ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hajk-Georg Drost &lt;hajk-georg.drost@tuebingen.mpg.de&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-12-02 17:40:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='biomartr-package'>Genomic Data Retrieval</h2><span id='topic+biomartr-package'></span><span id='topic+biomartr'></span>

<h3>Description</h3>

<p>This package interacts with a suite of web Application
Programming Interfaces and FTP sites to perform automated genomic data
retieval and annotation information retrieval.
</p>


<h3>About</h3>

<p>To automate the retrieval process on a meta-genomic scale, this package
provides useful interface functions for genomic sequence retrieval and
functional annotation retrieval.
The major aim of <code>biomartr</code> is to facilitate computational
reproducibility and large-scale handling of genomic data for
(meta-)genomic analyses.
</p>
<p>In detail, <code>biomartr</code> aims to provide users with an easy to use
framework to obtain genome, proteome, CDS, GFF (annotation), genome
assembly quality, and metagenome project data. Furthermore, an interface to
the Ensembl Biomart database allows users to retrieve functional annotation
for genomic loci.
Users can download entire databases
such as
</p>

<ul>
<li> <p><code>NCBI RefSeq</code>
</p>
</li>
<li> <p><code>NCBI nr</code>
</p>
</li>
<li> <p><code>NCBI nt</code>
</p>
</li>
<li> <p><code>NCBI Genbank</code>
</p>
</li>
<li> <p><code>NCBI nt</code>
</p>
</li>
<li> <p><code>Ensembl</code>
</p>
</li>
<li> <p><code>Ensembl Genomes</code>
</p>
</li>
<li> <p><code>UniProt</code>
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Hajk-Georg Drost <a href="mailto:hajk-georg.drost@tuebingen.mpg.de">hajk-georg.drost@tuebingen.mpg.de</a>
</p>

<hr>
<h2 id='biomart'>Main BioMart Query Function</h2><span id='topic+biomart'></span>

<h3>Description</h3>

<p>This function takes a set of gene ids and the biomart
specifications and performs a biomart query for the given set of gene ids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>biomart(genes, mart, dataset, attributes, filters, mute_citation = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="biomart_+3A_genes">genes</code></td>
<td>
<p>a character vector storing the gene ids of a organisms
of interest to be queried against BioMart.</p>
</td></tr>
<tr><td><code id="biomart_+3A_mart">mart</code></td>
<td>
<p>a character string specifying the mart to be used. Users
can obtain available marts using <code><a href="#topic+getMarts">getMarts</a></code>.</p>
</td></tr>
<tr><td><code id="biomart_+3A_dataset">dataset</code></td>
<td>
<p>a character string specifying the dataset within the mart to
be used, e.g. <code>dataset</code> = <code>"hsapiens_gene_ensembl"</code>.</p>
</td></tr>
<tr><td><code id="biomart_+3A_attributes">attributes</code></td>
<td>
<p>a character vector specifying the attributes that shall be
used, e.g. <code>attributes</code> =
<code>c("start_position","end_position","description")</code>.</p>
</td></tr>
<tr><td><code id="biomart_+3A_filters">filters</code></td>
<td>
<p>a character vector specifying the filter (query key) for the
BioMart query, e.g. <code>filter</code> = <code>"ensembl_gene_id"</code>.</p>
</td></tr>
<tr><td><code id="biomart_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical value indicating whether citation message should be muted.</p>
</td></tr>
<tr><td><code id="biomart_+3A_...">...</code></td>
<td>
<p>additional parameters for the
<code><a href="biomaRt.html#topic+getBM">getBM</a></code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is the main query function of the biomartr package.
</p>
<p>It enables to fastly access annotations of a given gene set based
on the <span class="pkg">biomaRt</span> package
implemented by Steffen Durinck et al.
</p>


<h3>Value</h3>

<p>A data.table storing the initial query gene vector in
the first column, the output
gene vector in the second column, and all attributes in
the following columns.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other biomaRt: 
<code><a href="#topic+getAttributes">getAttributes</a>()</code>,
<code><a href="#topic+getDatasets">getDatasets</a>()</code>,
<code><a href="#topic+getMarts">getMarts</a>()</code>,
<code><a href="#topic+organismBM">organismBM</a>()</code>,
<code><a href="#topic+organismFilters">organismFilters</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# 1) select a mart
getMarts()

# we will select mart 'plants_mart' and search for available datasets
getDatasets(mart = "plants_mart")

# we choose dataset 'athaliana_eg_gene' and run biomart()
# using mart: 'plants_mart', dataset: "athaliana_eg_gene"
# attributes: c("start_position","end_position","description")
# for an example gene set of Arabidopsis thaliana:
# c("AT1G06090", "AT1G06100", "AT1G06110", "AT1G06120",
#    "AT1G06130", "AT1G06200")

biomart(genes      = c("AT1G06090", "AT1G06100",
                       "AT1G06110", "AT1G06120",
                       "AT1G06130", "AT1G06200"),
        mart       = "plants_mart",
        dataset    = "athaliana_eg_gene",
        attributes = c("start_position","end_position","description"),
        filters    = "ensembl_gene_id")

## End(Not run)

</code></pre>

<hr>
<h2 id='cachedir'>Get directory to store back end files like kingdom summaries etc</h2><span id='topic+cachedir'></span>

<h3>Description</h3>

<p>Get directory to store back end files like kingdom summaries etc
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cachedir(non_temp_cache = "~/.biomartr_cache_dir.rds")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cachedir_+3A_non_temp_cache">non_temp_cache</code></td>
<td>
<p>&quot;~/.biomartr_cache_dir.rds&quot;,</p>
</td></tr>
</table>


<h3>Value</h3>

<p>reads the rds file, and returns the path for local cache, if not
existing, use tempdir().
</p>


<h3>See Also</h3>

<p>Other cachedir: 
<code><a href="#topic+cachedir_set">cachedir_set</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cachedir()
</code></pre>

<hr>
<h2 id='cachedir_set'>Set directory to store back end files like kingdom summaries etc</h2><span id='topic+cachedir_set'></span>

<h3>Description</h3>

<p>Set directory to store back end files like kingdom summaries etc
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cachedir_set(path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cachedir_set_+3A_path">path</code></td>
<td>
<p>the path to cache dir, example &quot;~/Bio_data/biomartr_cache/&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>invisible(NULL), only save the file to path location
</p>


<h3>See Also</h3>

<p>Other cachedir: 
<code><a href="#topic+cachedir">cachedir</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># By default it is tempdir()
cachedir()
# cachedir_set("~/Bio_data/biomartr_cache/")
cachedir()
</code></pre>

<hr>
<h2 id='check_annotation_biomartr'>Check whether an annotation file contains outlier lines</h2><span id='topic+check_annotation_biomartr'></span>

<h3>Description</h3>

<p>Some annotation files include lines with character lengths greater than 65000. 
This causes problems when trying to import such annotation files into R using <code>import</code>.
To overcome this issue, this function screens for such lines
in a given annotation file and removes these lines so that
<code>import</code> can handle the file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_annotation_biomartr(annotation_file, remove_annotation_outliers = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_annotation_biomartr_+3A_annotation_file">annotation_file</code></td>
<td>
<p>a file path to the annotation file.</p>
</td></tr>
<tr><td><code id="check_annotation_biomartr_+3A_remove_annotation_outliers">remove_annotation_outliers</code></td>
<td>
<p>shall outlier lines be removed from the input <code>annotation_file</code>? 
If yes, then the initial <code>annotation_file</code> will be overwritten and the removed outlier lines will be stored at <code><a href="base.html#topic+tempdir">tempdir</a></code>
for further exploration.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# download an example annotation file from NCBI RefSeq
Ath_path &lt;- biomartr::getGFF(organism = "Arabidopsis thaliana")
# run annotation file check on the downloaded file
biomartr::check_annotation_biomartr(Ath_path)
# several outlier lines were detected, thus we re-run the
# function using 'remove_annotation_outliers = TRUE'
# to remove the outliers and overwrite the file
biomartr::check_annotation_biomartr(Ath_path, remove_annotation_outliers = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='download.database'>Download a NCBI Database to Your Local Hard Drive</h2><span id='topic+download.database'></span>

<h3>Description</h3>

<p>This function allows users to download a database selected by
<code><a href="#topic+listDatabases">listDatabases</a></code> to their local hard drive.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>download.database(db, path = "database")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="download.database_+3A_db">db</code></td>
<td>
<p>a character string specifying the database that shall be downloaded
(selected from <code><a href="#topic+listDatabases">listDatabases</a></code>).</p>
</td></tr>
<tr><td><code id="download.database_+3A_path">path</code></td>
<td>
<p>a character string specifying the location (a folder) in
which the corresponding database shall be stored.
Default is <code>path = "database"</code>.
In case this folder does not exist yet, it will be created.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function downloads large databases to your hard drive.
For this purpose a folder
named <code>database</code> (default) is created and the correspondning
database then stored in this folder.
</p>


<h3>Value</h3>

<p>File path to the downloaded database file.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+download.database.all">download.database.all</a></code>, <code><a href="#topic+listDatabases">listDatabases</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  # search for available NCBI nr databases
  listNCBIDatabases(db = "nr")
  # select NCBI nr version 27 =  "nr.27.tar.gz"
  # and download it to your hard drive
  # -&gt; please note that large databases take some time for download!
  download.database(db = "nr.27.tar.gz")

## End(Not run)
</code></pre>

<hr>
<h2 id='download.database.all'>Download all elements of an NCBI databse</h2><span id='topic+download.database.all'></span>

<h3>Description</h3>

<p>The <code><a href="#topic+download.database">download.database</a></code> functions allows users to
retrieve individual packages of a NCBI database. This function is designed to
retrieve the entire database selected by the users (hence all packages 
corresponding to this database).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>download.database.all(db, path = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="download.database.all_+3A_db">db</code></td>
<td>
<p>a character string specifying the database that shall be downloaded
(selected from <code><a href="#topic+listDatabases">listDatabases</a></code>).</p>
</td></tr>
<tr><td><code id="download.database.all_+3A_path">path</code></td>
<td>
<p>a character string specifying the location (a folder) in which
the corresponding
database shall be stored. In case this folder does not exist yet,
it will be created.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector storing the file paths of the downloaded databases.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+download.database">download.database</a></code>, <code><a href="#topic+listNCBIDatabases">listNCBIDatabases</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# search for available NCBI databases
  listNCBIDatabases(db = "all")
# choose database NCBI nr and download compelete database
  download.database.all(db = "nr", path = "nr")

## End(Not run)
</code></pre>

<hr>
<h2 id='ensembl_divisions'>List all available ENSEMBL divisions</h2><span id='topic+ensembl_divisions'></span>

<h3>Description</h3>

<p>Retrieve a list of available databases on ENSEMBL for which <code>get.ensembl.info</code> can be retrieved.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ensembl_divisions()
</code></pre>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ensembl_divisions()
</code></pre>

<hr>
<h2 id='get.ensembl.info'>Helper function to retrieve species information from the ENSEMBL API</h2><span id='topic+get.ensembl.info'></span>

<h3>Description</h3>

<p>This function interfaces with the ENSEMBL API
(https://rest.ensembl.org/info/species?content-type=application/json)
and internally stores the output to use this information for subsequent
retrieval function calls.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.ensembl.info(update = FALSE, division)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get.ensembl.info_+3A_update">update</code></td>
<td>
<p>logical, default FALSE. If TRUE, force re-download of info.</p>
</td></tr>
<tr><td><code id="get.ensembl.info_+3A_division">division</code></td>
<td>
<p>the ENSEMBL database (division) for which information shall
be retrieved (available options can be obtained with <code><a href="#topic+ensembl_divisions">ensembl_divisions</a></code>).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ensembl_divisions">ensembl_divisions</a></code>, <code><a href="#topic+getKingdomAssemblySummary">getKingdomAssemblySummary</a></code>, <code><a href="#topic+getENSEMBLInfo">getENSEMBLInfo</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Look at available ENSEMBL division options
ensembl_divisions()
# Retrieve available information for EnsemblVertebrates
example &lt;- get.ensembl.info(division = "EnsemblVertebrates")
example
# Update information file stored in the tempdir() folder.
example_update &lt;- get.ensembl.info(division = "EnsemblVertebrates", update = TRUE)
example_update

## End(Not run)
</code></pre>

<hr>
<h2 id='getAssemblyStats'>Genome Assembly Stats Retrieval</h2><span id='topic+getAssemblyStats'></span>

<h3>Description</h3>

<p>Main genome assembly stats retrieval function for an organism
of interest. By specifying the scientific name of an organism of interest the
corresponding  genome assembly stats file storing the assembly statistics of
the organism of interest can be downloaded and stored locally.
Genome assembly stats files can be retrieved from several databases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAssemblyStats(
  db = "refseq",
  organism,
  reference = FALSE,
  skip_bacteria = TRUE,
  release = NULL,
  type = "download",
  path = file.path("_ncbi_downloads", "genomeassembly_stats"),
  mute_citation = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAssemblyStats_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="getAssemblyStats_+3A_organism">organism</code></td>
<td>
<p>a character string specifying the scientific name of the
organism of interest, e.g. <code>organism = "Homo sapiens"</code>.</p>
</td></tr>
<tr><td><code id="getAssemblyStats_+3A_reference">reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database as either a reference genome or a representative genome.</p>
</td></tr>
<tr><td><code id="getAssemblyStats_+3A_skip_bacteria">skip_bacteria</code></td>
<td>
<p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p>
</td></tr>
<tr><td><code id="getAssemblyStats_+3A_release">release</code></td>
<td>
<p>most recent database version is used. release = 75 would for human would give the stable GRCh37 release in ensembl. 
Value must be &gt; 46, since ensembl did not structure their data if the standard format before that.</p>
</td></tr>
<tr><td><code id="getAssemblyStats_+3A_type">type</code></td>
<td>
<p>shall only the file be retrieved (default)
<code>type = "download"</code> or should the corresponding file be downloaded and
subsequently be imported <code>type = "import"</code>.</p>
</td></tr>
<tr><td><code id="getAssemblyStats_+3A_path">path</code></td>
<td>
<p>a character string specifying the location (a folder) in
which the corresponding file shall be stored. Default is
<code>path</code> = <code>file.path("_ncbi_downloads","genomeassembly_stats")</code>.</p>
</td></tr>
<tr><td><code id="getAssemblyStats_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical value indicating whether citation message should be muted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI:
</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/
</p>
<p>to retrieve available scientific names of organisms and creates a directory
'_ncbi_downloads/genomeassembly_stats' to store
the Genome Assembly Stats of interest as text file for future processing.
In case the corresponding fasta file already exists within the
'_ncbi_downloads/genomeassembly_stats' folder and is
accessible within the workspace, no download process will be performed.
</p>
<p>An example genome assembly stats file can be found here:
ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/
GCF_000001405.36_GRCh38.p10/GCF_000001405.36_GRCh38.p10_assembly_stats.txt.
</p>


<h3>Value</h3>

<p>File path to downloaded genome assembly stats file.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getGenome">getGenome</a></code>, <code><a href="#topic+getProteome">getProteome</a></code>, <code><a href="#topic+getCDS">getCDS</a></code>,
<code><a href="#topic+getGFF">getGFF</a></code>, <code><a href="#topic+getRNA">getRNA</a></code>, <code><a href="#topic+getCollection">getCollection</a></code>, <code><a href="#topic+meta.retrieval">meta.retrieval</a></code>,
<code><a href="#topic+read_assemblystats">read_assemblystats</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# download the genome assembly stats file of Saccharomyces cerevisiae
# from NCBI RefSeq
# and store the corresponding genome file in
# '_ncbi_downloads/genomeassembly_stats'
file_path &lt;- getAssemblyStats( db = "refseq",
                 organism = "Saccharomyces cerevisiae",
                 path = file.path("_ncbi_downloads","genomeassembly_stats"))
# import the raw file as it is downloaded
Scerevisiae.stats &lt;- read_assemblystats(file_path, type = "raw")

# download the genome assembly stats file of Saccharomyces cerevisiae
# from NCBI RefSeq
# and import overall statistics of the genome assembly
Scerevisiae.stats.import &lt;- getAssemblyStats( db = "refseq",
                 organism = "Saccharomyces cerevisiae",
                 type = "import",
                 path = file.path("_ncbi_downloads","genomeassembly_stats"))

## End(Not run)

</code></pre>

<hr>
<h2 id='getAttributes'>Retrieve All Available Attributes for a Specific Dataset</h2><span id='topic+getAttributes'></span>

<h3>Description</h3>

<p>This function queries the BioMart Interface and returns a table
storing all available attributes for a specific dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAttributes(mart, dataset, mute_citation = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAttributes_+3A_mart">mart</code></td>
<td>
<p>a character string specifying the database (mart)
for which datasets shall be listed.</p>
</td></tr>
<tr><td><code id="getAttributes_+3A_dataset">dataset</code></td>
<td>
<p>a character string specifying the dataset for which
attributes shall be listed.</p>
</td></tr>
<tr><td><code id="getAttributes_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical value indicating whether citation message should be muted.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other biomaRt: 
<code><a href="#topic+biomart">biomart</a>()</code>,
<code><a href="#topic+getDatasets">getDatasets</a>()</code>,
<code><a href="#topic+getMarts">getMarts</a>()</code>,
<code><a href="#topic+organismBM">organismBM</a>()</code>,
<code><a href="#topic+organismFilters">organismFilters</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# search for available datasets
getMarts()

# choose database (mart): ENSEMBL_MART_ENSEMBL
# and get a table of all available datasets from this BioMart database
head(getDatasets(mart = "ENSEMBL_MART_ENSEMBL"), 10)

# choose dataset: "hsapiens_gene_ensembl"
head(getAttributes(mart = "ENSEMBL_MART_ENSEMBL",
                   dataset = "hsapiens_gene_ensembl") , 5)

## End(Not run)
</code></pre>

<hr>
<h2 id='getBio'>A wrapper to all bio getters, selected with 'type' argument</h2><span id='topic+getBio'></span>

<h3>Description</h3>

<p>A wrapper to all bio getters, selected with 'type' argument
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getBio(
  db = "refseq",
  organism,
  type,
  reference = FALSE,
  release = NULL,
  gunzip = FALSE,
  update = FALSE,
  skip_bacteria = TRUE,
  path = paste0("set_", toupper(type)),
  remove_annotation_outliers = FALSE,
  analyse_genome = FALSE,
  assembly_type = "toplevel",
  format = "gff3",
  mute_citation = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getBio_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="getBio_+3A_organism">organism</code></td>
<td>
<p>Organism selector id,
there are three options to characterize an organism:
</p>

<ul>
<li><p> by <code>scientific name</code>: e.g. <code>organism = "Homo sapiens"</code>
</p>
</li>
<li><p> by <code>database specific accession identifier</code>: e.g. <code>organism = "GCF_000001405.37"</code> (= NCBI RefSeq identifier for <code>Homo sapiens</code>)
</p>
</li>
<li><p> by <code>taxonomic identifier from NCBI Taxonomy</code>: e.g. <code>organism = "9606"</code> (= taxid of <code>Homo sapiens</code>)
</p>
</li></ul>
</td></tr>
<tr><td><code id="getBio_+3A_type">type</code></td>
<td>
<p>biological sequence type. (alternatives are: genome, gff, cds,
rna, proteome, assembly_stats, repeat_masker, collection (all the others))</p>
</td></tr>
<tr><td><code id="getBio_+3A_reference">reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database as either a reference genome or a representative genome.</p>
</td></tr>
<tr><td><code id="getBio_+3A_release">release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td></tr>
<tr><td><code id="getBio_+3A_gunzip">gunzip</code></td>
<td>
<p>a logical, indicating whether or not files should be unzipped.</p>
</td></tr>
<tr><td><code id="getBio_+3A_update">update</code></td>
<td>
<p>logical, default FALSE. Updated backend cached files needed.
Usually keep this false, to make ut run much faster. Only set to TRUE, if you
believe you cache is outdated (Species only exist in newest release etc)</p>
</td></tr>
<tr><td><code id="getBio_+3A_skip_bacteria">skip_bacteria</code></td>
<td>
<p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p>
</td></tr>
<tr><td><code id="getBio_+3A_path">path</code></td>
<td>
<p>character, default location is paste0(&quot;set_&quot;, toupper(type))</p>
</td></tr>
<tr><td><code id="getBio_+3A_remove_annotation_outliers">remove_annotation_outliers</code></td>
<td>
<p>shall outlier lines be removed from the input <code>annotation_file</code>?
If yes, then the initial <code>annotation_file</code> will be overwritten and the removed outlier lines will be stored at <code><a href="base.html#topic+tempdir">tempdir</a></code>
for further exploration.</p>
</td></tr>
<tr><td><code id="getBio_+3A_analyse_genome">analyse_genome</code></td>
<td>
<p>logical, default FALSE. If TRUE, get general genome statistics like
gc content etc. For more details, see ?summary_genome</p>
</td></tr>
<tr><td><code id="getBio_+3A_assembly_type">assembly_type</code></td>
<td>
<p>character, default c(&quot;primary_assembly&quot;, &quot;toplevel&quot;). Used for ensembl only,
specifies the genome assembly type. Searches for both primary and toplevel, and if both are found, uses the
first by order (so primary is prioritized by default).
The Primary assembly should usually be used if it exists.
The &quot;primary assembly&quot; contains all the top-level sequence regions,
excluding alternative haplotypes and patches.
If the primary assembly file is not present for a species
(only defined for standard model organisms),
that indicates that there were no haplotype/patch regions,
and in such cases, the 'toplevel file is used.
For more details see:
<a href="https://grch37.ensembl.org/info/genome/genebuild/assembly.html">ensembl tutorial</a></p>
</td></tr>
<tr><td><code id="getBio_+3A_format">format</code></td>
<td>
<p>&quot;gff3&quot;, alternative &quot;gtf&quot; for ensembl.</p>
</td></tr>
<tr><td><code id="getBio_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical, default FALSE, indicating whether citation message should be muted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI:
</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/
</p>
<p>and creates a directory relative to file type, if you get fasta genomes it will be
_ncbi_downloads/genomes'.
In case the corresponding fasta file already exists within the
'_ncbi_downloads/genomes' folder and is accessible within the workspace,
no download process will be performed.
For other file types the same rule applies.
</p>


<h3>Value</h3>

<p>File path to downloaded genome.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other getBio: 
<code><a href="#topic+getCDS">getCDS</a>()</code>,
<code><a href="#topic+getCollection">getCollection</a>()</code>,
<code><a href="#topic+getGFF">getGFF</a>()</code>,
<code><a href="#topic+getGenome">getGenome</a>()</code>,
<code><a href="#topic+getProteome">getProteome</a>()</code>,
<code><a href="#topic+getRNA">getRNA</a>()</code>
</p>

<hr>
<h2 id='getBioSet'>Generic Bio data set extractor</h2><span id='topic+getBioSet'></span>

<h3>Description</h3>

<p>Usually you want to use one of the specific set extractors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getBioSet(
  db = "refseq",
  organisms,
  set_type,
  reference = FALSE,
  release = NULL,
  gunzip = TRUE,
  update = FALSE,
  skip_bacteria = TRUE,
  path = paste0("set_", toupper(set_type)),
  remove_annotation_outliers = FALSE,
  assembly_type = "toplevel",
  format = "gff3",
  mute_citation = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getBioSet_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="getBioSet_+3A_organisms">organisms</code></td>
<td>
<p>a character vector storing the names of the organisms than shall be retrieved.
There are three available options to characterize an organism:</p>
</td></tr>
<tr><td><code id="getBioSet_+3A_set_type">set_type</code></td>
<td>
<p>the biological sequence type that shall be retrieved.
Available options are
</p>

<ul>
<li> <p><code>set_type = "genome"</code>
</p>
</li>
<li> <p><code>set_type = "proteome"</code>
</p>
</li>
<li> <p><code>set_type = "cds"</code>
</p>
</li>
<li> <p><code>set_type = "gff"</code>
</p>
</li>
<li> <p><code>set_type = "rna"</code>
</p>
</li>
<li> <p><code>set_type = "assembly_stats"</code>
</p>
</li>
<li> <p><code>set_type = "repeat_masker"</code>
</p>
</li>
<li> <p><code>set_type = "collection"</code> (all the others)
</p>
</li></ul>
</td></tr>
<tr><td><code id="getBioSet_+3A_reference">reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database as either a reference genome or a representative genome.</p>
</td></tr>
<tr><td><code id="getBioSet_+3A_release">release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td></tr>
<tr><td><code id="getBioSet_+3A_gunzip">gunzip</code></td>
<td>
<p>a logical, indicating whether or not files should be unzipped.</p>
</td></tr>
<tr><td><code id="getBioSet_+3A_update">update</code></td>
<td>
<p>logical, default FALSE. Updated backend cached files needed.
Usually keep this false, to make ut run much faster. Only set to TRUE, if you
believe you cache is outdated (Species only exist in newest release etc)</p>
</td></tr>
<tr><td><code id="getBioSet_+3A_skip_bacteria">skip_bacteria</code></td>
<td>
<p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p>
</td></tr>
<tr><td><code id="getBioSet_+3A_path">path</code></td>
<td>
<p>character, default location is paste0(&quot;set_&quot;, toupper(set_type))</p>
</td></tr>
<tr><td><code id="getBioSet_+3A_remove_annotation_outliers">remove_annotation_outliers</code></td>
<td>
<p>shall outlier lines be removed from the input <code>annotation_file</code>?
If yes, then the initial <code>annotation_file</code> will be overwritten and the removed outlier lines will be stored at <code><a href="base.html#topic+tempdir">tempdir</a></code>
for further exploration.</p>
</td></tr>
<tr><td><code id="getBioSet_+3A_assembly_type">assembly_type</code></td>
<td>
<p>character, default c(&quot;primary_assembly&quot;, &quot;toplevel&quot;). Used for ensembl only,
specifies the genome assembly type. Searches for both primary and toplevel, and if both are found, uses the
first by order (so primary is prioritized by default).
The Primary assembly should usually be used if it exists.
The &quot;primary assembly&quot; contains all the top-level sequence regions,
excluding alternative haplotypes and patches.
If the primary assembly file is not present for a species
(only defined for standard model organisms),
that indicates that there were no haplotype/patch regions,
and in such cases, the 'toplevel file is used.
For more details see:
<a href="https://grch37.ensembl.org/info/genome/genebuild/assembly.html">ensembl tutorial</a></p>
</td></tr>
<tr><td><code id="getBioSet_+3A_format">format</code></td>
<td>
<p>&quot;gff3&quot;, alternative &quot;gtf&quot; for ensembl.</p>
</td></tr>
<tr><td><code id="getBioSet_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical, default FALSE, indicating whether citation message should be muted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI:
</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/
</p>
<p>and creates a directory 'set_CDSs' to store
the CDSs of interest as fasta files for future processing.
In case the corresponding fasta file already exists within the
'set_CDSs' folder and is accessible within the workspace,
no download process will be performed.
</p>


<h3>Value</h3>

<p>File path to downloaded genomes
(names are identifiers: 'new' (file was downloaded now),
'old' files did already exist)
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getBio">getBio</a></code>
</p>
<p>Other getBioSet: 
<code><a href="#topic+getCDSSet">getCDSSet</a>()</code>,
<code><a href="#topic+getCollectionSet">getCollectionSet</a>()</code>,
<code><a href="#topic+getGFFSet">getGFFSet</a>()</code>,
<code><a href="#topic+getGenomeSet">getGenomeSet</a>()</code>,
<code><a href="#topic+getProteomeSet">getProteomeSet</a>()</code>,
<code><a href="#topic+getRNASet">getRNASet</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getBioSet("refseq", organisms = c("Arabidopsis thaliana",
                                  "Arabidopsis lyrata",
                                  "Capsella rubella"),
                                  set_type = "cds")

## End(Not run)
</code></pre>

<hr>
<h2 id='getCDS'>Coding Sequence Retrieval</h2><span id='topic+getCDS'></span>

<h3>Description</h3>

<p>Main retrieval function for coding sequences (CDS)
of an organism of interest.
By specifying the scientific name of an organism of interest the
corresponding fasta-file storing the CDS information for the organism
of interest can be downloaded and stored locally. CDS files can be retrieved
from several databases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCDS(
  db = "refseq",
  organism,
  reference = FALSE,
  skip_bacteria = TRUE,
  release = NULL,
  gunzip = FALSE,
  path = file.path("_ncbi_downloads", "CDS"),
  mute_citation = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCDS_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="getCDS_+3A_organism">organism</code></td>
<td>
<p>Organism selector id,
there are three options to characterize an organism:
</p>

<ul>
<li><p> by <code>scientific name</code>: e.g. <code>organism = "Homo sapiens"</code>
</p>
</li>
<li><p> by <code>database specific accession identifier</code>: e.g. <code>organism = "GCF_000001405.37"</code> (= NCBI RefSeq identifier for <code>Homo sapiens</code>)
</p>
</li>
<li><p> by <code>taxonomic identifier from NCBI Taxonomy</code>: e.g. <code>organism = "9606"</code> (= taxid of <code>Homo sapiens</code>)
</p>
</li></ul>
</td></tr>
<tr><td><code id="getCDS_+3A_reference">reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database as either a reference genome or a representative genome.</p>
</td></tr>
<tr><td><code id="getCDS_+3A_skip_bacteria">skip_bacteria</code></td>
<td>
<p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p>
</td></tr>
<tr><td><code id="getCDS_+3A_release">release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td></tr>
<tr><td><code id="getCDS_+3A_gunzip">gunzip</code></td>
<td>
<p>a logical, indicating whether or not files should be unzipped.</p>
</td></tr>
<tr><td><code id="getCDS_+3A_path">path</code></td>
<td>
<p>a character string specifying the location (a folder)
in which the corresponding CDS file shall be stored.
Default is <code>path</code> = <code>file.path("_ncbi_downloads","CDS")</code>.</p>
</td></tr>
<tr><td><code id="getCDS_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical, default FALSE, indicating whether citation message should be muted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI:
</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/
</p>
<p>and creates a directory relative to file type, if you get fasta genomes it will be
_ncbi_downloads/genomes'.
In case the corresponding fasta file already exists within the
'_ncbi_downloads/genomes' folder and is accessible within the workspace,
no download process will be performed.
For other file types the same rule applies.
</p>


<h3>Value</h3>

<p>File path to downloaded genome.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other getBio: 
<code><a href="#topic+getBio">getBio</a>()</code>,
<code><a href="#topic+getCollection">getCollection</a>()</code>,
<code><a href="#topic+getGFF">getGFF</a>()</code>,
<code><a href="#topic+getGenome">getGenome</a>()</code>,
<code><a href="#topic+getProteome">getProteome</a>()</code>,
<code><a href="#topic+getRNA">getRNA</a>()</code>
</p>
<p>Other cds: 
<code><a href="#topic+getCDSSet">getCDSSet</a>()</code>,
<code><a href="#topic+read_cds">read_cds</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# download the genome of Arabidopsis thaliana from refseq
# and store the corresponding genome CDS file in '_ncbi_downloads/CDS'
file_path &lt;- getCDS( db       = "refseq",
             organism = "Arabidopsis thaliana",
             path     = file.path("_ncbi_downloads","CDS"))

Ath_CDS &lt;- read_cds(file_path, format = "fasta")


## End(Not run)
</code></pre>

<hr>
<h2 id='getCDSSet'>CDS retrieval of multiple species</h2><span id='topic+getCDSSet'></span>

<h3>Description</h3>

<p>Main CDS retrieval function for a set of organism of interest.
By specifying the scientific names of the organisms of interest the corresponding fasta-files storing the CDS of the organisms of interest
will be downloaded and stored locally. CDS files can be retrieved from several databases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCDSSet(
  db = "refseq",
  organisms,
  reference = FALSE,
  release = NULL,
  gunzip = TRUE,
  update = FALSE,
  path = "set_CDS"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCDSSet_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="getCDSSet_+3A_organisms">organisms</code></td>
<td>
<p>a character vector storing the names of the organisms than shall be retrieved.
There are three available options to characterize an organism:</p>
</td></tr>
<tr><td><code id="getCDSSet_+3A_reference">reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database as either a reference genome or a representative genome.</p>
</td></tr>
<tr><td><code id="getCDSSet_+3A_release">release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td></tr>
<tr><td><code id="getCDSSet_+3A_gunzip">gunzip</code></td>
<td>
<p>a logical, indicating whether or not files should be unzipped.</p>
</td></tr>
<tr><td><code id="getCDSSet_+3A_update">update</code></td>
<td>
<p>logical, default FALSE. Updated backend cached files needed.
Usually keep this false, to make ut run much faster. Only set to TRUE, if you
believe you cache is outdated (Species only exist in newest release etc)</p>
</td></tr>
<tr><td><code id="getCDSSet_+3A_path">path</code></td>
<td>
<p>character, default location is paste0(&quot;set_&quot;, toupper(set_type))</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI:
</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/
</p>
<p>and creates a directory 'set_CDSs' to store
the CDSs of interest as fasta files for future processing.
In case the corresponding fasta file already exists within the
'set_CDSs' folder and is accessible within the workspace,
no download process will be performed.
</p>


<h3>Value</h3>

<p>File path to downloaded genomes
(names are identifiers: 'new' (file was downloaded now),
'old' files did already exist)
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other getBioSet: 
<code><a href="#topic+getBioSet">getBioSet</a>()</code>,
<code><a href="#topic+getCollectionSet">getCollectionSet</a>()</code>,
<code><a href="#topic+getGFFSet">getGFFSet</a>()</code>,
<code><a href="#topic+getGenomeSet">getGenomeSet</a>()</code>,
<code><a href="#topic+getProteomeSet">getProteomeSet</a>()</code>,
<code><a href="#topic+getRNASet">getRNASet</a>()</code>
</p>
<p>Other cds: 
<code><a href="#topic+getCDS">getCDS</a>()</code>,
<code><a href="#topic+read_cds">read_cds</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getBioSet("refseq", organisms = c("Arabidopsis thaliana",
                                  "Arabidopsis lyrata",
                                  "Capsella rubella"),
                                  set_type = "cds")

## End(Not run)
</code></pre>

<hr>
<h2 id='getCollection'>Retrieve a Collection: Genome, Proteome, CDS, RNA, GFF, Repeat Masker, AssemblyStats</h2><span id='topic+getCollection'></span>

<h3>Description</h3>

<p>Main collection retrieval function for an organism of interest.
By specifying the scientific name of an organism of interest a collection consisting of
the genome file, proteome file, CDS file, RNA file, GFF file, Repeat Masker file, AssemblyStats
file of the organism of interest
can be downloaded and stored locally. Collections can be retrieved from
several databases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCollection(
  db = "refseq",
  organism,
  reference = TRUE,
  skip_bacteria = TRUE,
  release = NULL,
  assembly_type = "toplevel",
  analyse_genome = FALSE,
  remove_annotation_outliers = FALSE,
  gunzip = FALSE,
  path = file.path("_db_downloads", "collections"),
  mute_citation = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCollection_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="getCollection_+3A_organism">organism</code></td>
<td>
<p>Organism selector id,
there are three options to characterize an organism:
</p>

<ul>
<li><p> by <code>scientific name</code>: e.g. <code>organism = "Homo sapiens"</code>
</p>
</li>
<li><p> by <code>database specific accession identifier</code>: e.g. <code>organism = "GCF_000001405.37"</code> (= NCBI RefSeq identifier for <code>Homo sapiens</code>)
</p>
</li>
<li><p> by <code>taxonomic identifier from NCBI Taxonomy</code>: e.g. <code>organism = "9606"</code> (= taxid of <code>Homo sapiens</code>)
</p>
</li></ul>
</td></tr>
<tr><td><code id="getCollection_+3A_reference">reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database as either a reference genome or a representative genome.</p>
</td></tr>
<tr><td><code id="getCollection_+3A_skip_bacteria">skip_bacteria</code></td>
<td>
<p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p>
</td></tr>
<tr><td><code id="getCollection_+3A_release">release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td></tr>
<tr><td><code id="getCollection_+3A_assembly_type">assembly_type</code></td>
<td>
<p>character, default c(&quot;primary_assembly&quot;, &quot;toplevel&quot;). Used for ensembl only,
specifies the genome assembly type. Searches for both primary and toplevel, and if both are found, uses the
first by order (so primary is prioritized by default).
The Primary assembly should usually be used if it exists.
The &quot;primary assembly&quot; contains all the top-level sequence regions,
excluding alternative haplotypes and patches.
If the primary assembly file is not present for a species
(only defined for standard model organisms),
that indicates that there were no haplotype/patch regions,
and in such cases, the 'toplevel file is used.
For more details see:
<a href="https://grch37.ensembl.org/info/genome/genebuild/assembly.html">ensembl tutorial</a></p>
</td></tr>
<tr><td><code id="getCollection_+3A_analyse_genome">analyse_genome</code></td>
<td>
<p>logical, default FALSE. If TRUE, get general genome statistics like
gc content etc. For more details, see ?summary_genome</p>
</td></tr>
<tr><td><code id="getCollection_+3A_remove_annotation_outliers">remove_annotation_outliers</code></td>
<td>
<p>shall outlier lines be removed from the input <code>annotation_file</code>?
If yes, then the initial <code>annotation_file</code> will be overwritten and the removed outlier lines will be stored at <code><a href="base.html#topic+tempdir">tempdir</a></code>
for further exploration.</p>
</td></tr>
<tr><td><code id="getCollection_+3A_gunzip">gunzip</code></td>
<td>
<p>a logical, indicating whether or not files should be unzipped.</p>
</td></tr>
<tr><td><code id="getCollection_+3A_path">path</code></td>
<td>
<p>a character string specifying the location (a folder) in which
the corresponding collection shall be stored. Default is
<code>path</code> = <code>file.path("_db_downloads","collections")</code>.</p>
</td></tr>
<tr><td><code id="getCollection_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical, default FALSE, indicating whether citation message should be muted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI:
</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/
</p>
<p>and creates a directory relative to file type, if you get fasta genomes it will be
_ncbi_downloads/genomes'.
In case the corresponding fasta file already exists within the
'_ncbi_downloads/genomes' folder and is accessible within the workspace,
no download process will be performed.
For other file types the same rule applies.
</p>


<h3>Value</h3>

<p>File path to downloaded genome.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other getBio: 
<code><a href="#topic+getBio">getBio</a>()</code>,
<code><a href="#topic+getCDS">getCDS</a>()</code>,
<code><a href="#topic+getGFF">getGFF</a>()</code>,
<code><a href="#topic+getGenome">getGenome</a>()</code>,
<code><a href="#topic+getProteome">getProteome</a>()</code>,
<code><a href="#topic+getRNA">getRNA</a>()</code>
</p>
<p>Other collection: 
<code><a href="#topic+getCollectionSet">getCollectionSet</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# download the collection of Homo sapiens from refseq
# and store the corresponding genome file in '_ncbi_downloads/collection'
 Hsap_collection &lt;- getCollection( db       = "refseq",
             organism = "Homo sapiens",
             path = file.path("_db_downloads","collections"))
# download the collection of Homo sapiens from genbank
# and store the corresponding genome file in '_ncbi_downloads/collection'
 Hsap_collection &lt;- getCollection( db       = "genbank",
             organism = "Homo sapiens",
             path = file.path("_db_downloads","collections"))
# download the collection of Homo sapiens from ensembl
# and store the corresponding genome file in '_ncbi_downloads/collection'
 Hsap_collection &lt;- getCollection( db       = "ensembl",
             organism = "Homo sapiens",
             path = file.path("_db_downloads","collections"))

## End(Not run)

</code></pre>

<hr>
<h2 id='getCollectionSet'>Retrieve a Collection: Genome, Proteome, CDS, RNA, GFF, Repeat Masker, AssemblyStats of multiple species</h2><span id='topic+getCollectionSet'></span>

<h3>Description</h3>

<p>Main collection retrieval function for an organism of interest.
By specifying the scientific name of an organism of interest a collection consisting of
the genome file, proteome file, CDS file, RNA file, GFF file, Repeat Masker file, AssemblyStats
file of the organism of interest
can be downloaded and stored locally. Collections can be retrieved from
several databases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCollectionSet(
  db = "refseq",
  organisms,
  reference = FALSE,
  release = NULL,
  skip_bacteria = TRUE,
  gunzip = TRUE,
  update = FALSE,
  remove_annotation_outliers = TRUE,
  path = "set_collections",
  mute_citation = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCollectionSet_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="getCollectionSet_+3A_organisms">organisms</code></td>
<td>
<p>a character vector storing the names of the organisms than shall be retrieved.
There are three available options to characterize an organism:</p>
</td></tr>
<tr><td><code id="getCollectionSet_+3A_reference">reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database as either a reference genome or a representative genome.</p>
</td></tr>
<tr><td><code id="getCollectionSet_+3A_release">release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td></tr>
<tr><td><code id="getCollectionSet_+3A_skip_bacteria">skip_bacteria</code></td>
<td>
<p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p>
</td></tr>
<tr><td><code id="getCollectionSet_+3A_gunzip">gunzip</code></td>
<td>
<p>a logical, indicating whether or not files should be unzipped.</p>
</td></tr>
<tr><td><code id="getCollectionSet_+3A_update">update</code></td>
<td>
<p>logical, default FALSE. Updated backend cached files needed.
Usually keep this false, to make ut run much faster. Only set to TRUE, if you
believe you cache is outdated (Species only exist in newest release etc)</p>
</td></tr>
<tr><td><code id="getCollectionSet_+3A_remove_annotation_outliers">remove_annotation_outliers</code></td>
<td>
<p>shall outlier lines be removed from the input <code>annotation_file</code>?
If yes, then the initial <code>annotation_file</code> will be overwritten and the removed outlier lines will be stored at <code><a href="base.html#topic+tempdir">tempdir</a></code>
for further exploration.</p>
</td></tr>
<tr><td><code id="getCollectionSet_+3A_path">path</code></td>
<td>
<p>a character string specifying the location (a folder) in which
the corresponding collection shall be stored. Default is
<code>path</code> = <code>file.path("_db_downloads","collections")</code>.</p>
</td></tr>
<tr><td><code id="getCollectionSet_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical, default FALSE, indicating whether citation message should be muted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI:
</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/
</p>
<p>and creates a directory 'set_CDSs' to store
the CDSs of interest as fasta files for future processing.
In case the corresponding fasta file already exists within the
'set_CDSs' folder and is accessible within the workspace,
no download process will be performed.
</p>


<h3>Value</h3>

<p>File path to downloaded genomes
(names are identifiers: 'new' (file was downloaded now),
'old' files did already exist)
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other getBioSet: 
<code><a href="#topic+getBioSet">getBioSet</a>()</code>,
<code><a href="#topic+getCDSSet">getCDSSet</a>()</code>,
<code><a href="#topic+getGFFSet">getGFFSet</a>()</code>,
<code><a href="#topic+getGenomeSet">getGenomeSet</a>()</code>,
<code><a href="#topic+getProteomeSet">getProteomeSet</a>()</code>,
<code><a href="#topic+getRNASet">getRNASet</a>()</code>
</p>
<p>Other collection: 
<code><a href="#topic+getCollection">getCollection</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getBioSet("refseq", organisms = c("Arabidopsis thaliana",
                                  "Arabidopsis lyrata",
                                  "Capsella rubella"),
                                  set_type = "cds")

## End(Not run)
</code></pre>

<hr>
<h2 id='getDatasets'>Retrieve All Available Datasets for a BioMart Database</h2><span id='topic+getDatasets'></span>

<h3>Description</h3>

<p>This funcion queries the BioMart API and returns a table
storing all available datasets for a selected BioMart databases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDatasets(mart, mute_citation = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getDatasets_+3A_mart">mart</code></td>
<td>
<p>a character string specifying the database (mart) for
which datasets shall be listed.</p>
</td></tr>
<tr><td><code id="getDatasets_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical value indicating whether citation message should be muted.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other biomaRt: 
<code><a href="#topic+biomart">biomart</a>()</code>,
<code><a href="#topic+getAttributes">getAttributes</a>()</code>,
<code><a href="#topic+getMarts">getMarts</a>()</code>,
<code><a href="#topic+organismBM">organismBM</a>()</code>,
<code><a href="#topic+organismFilters">organismFilters</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# search for available datasets
# getMarts()
# choose database: "ENSEMBL_MART_ENSEMBL"
head(getDatasets("ENSEMBL_MART_ENSEMBL"), 10)

## End(Not run)
</code></pre>

<hr>
<h2 id='getENSEMBL'>Download sequence or annotation from ENSEMBL</h2><span id='topic+getENSEMBL'></span>

<h3>Description</h3>

<p>Backend function for retrieving files sequence and annotation
files from the ENSEMBL ftp server
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getENSEMBL(
  organism,
  type = "dna",
  id.type = "toplevel",
  release = NULL,
  path,
  format
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getENSEMBL_+3A_organism">organism</code></td>
<td>
<p>Organism selector id,
there are three options to characterize an organism:
</p>

<ul>
<li><p> by <code>scientific name</code>: e.g. <code>organism = "Homo sapiens"</code>
</p>
</li>
<li><p> by <code>database specific accession identifier</code>: e.g. <code>organism = "GCF_000001405.37"</code> (= NCBI RefSeq identifier for <code>Homo sapiens</code>)
</p>
</li>
<li><p> by <code>taxonomic identifier from NCBI Taxonomy</code>: e.g. <code>organism = "9606"</code> (= taxid of <code>Homo sapiens</code>)
</p>
</li></ul>
</td></tr>
<tr><td><code id="getENSEMBL_+3A_type">type</code></td>
<td>
<p>character, biological sequence type (e.g. &quot;dna&quot;, &quot;cds&quot;)</p>
</td></tr>
<tr><td><code id="getENSEMBL_+3A_id.type">id.type</code></td>
<td>
<p>a character, default &quot;toplevel&quot;.
id type of assembly, either &quot;toplevel&quot; or &quot;primary_assembly&quot; for genomes.
Can be other strings, for non genome objects.</p>
</td></tr>
<tr><td><code id="getENSEMBL_+3A_release">release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td></tr>
<tr><td><code id="getENSEMBL_+3A_path">path</code></td>
<td>
<p>location where file shall be stored.</p>
</td></tr>
<tr><td><code id="getENSEMBL_+3A_format">format</code></td>
<td>
<p>&quot;gff3&quot;, alternative &quot;gtf&quot; for ensembl.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>either a character path to downloaded file, or a logical FALSE, specifying failure.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>

<hr>
<h2 id='getENSEMBL.gtf'>Helper function for retrieving gtf files from ENSEMBL</h2><span id='topic+getENSEMBL.gtf'></span>

<h3>Description</h3>

<p>This function downloads gff
files of query organisms from ENSEMBL.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getENSEMBL.gtf(organism, type = "dna", path, release = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getENSEMBL.gtf_+3A_organism">organism</code></td>
<td>
<p>Organism selector id,
there are three options to characterize an organism:
</p>

<ul>
<li><p> by <code>scientific name</code>: e.g. <code>organism = "Homo sapiens"</code>
</p>
</li>
<li><p> by <code>database specific accession identifier</code>: e.g. <code>organism = "GCF_000001405.37"</code> (= NCBI RefSeq identifier for <code>Homo sapiens</code>)
</p>
</li>
<li><p> by <code>taxonomic identifier from NCBI Taxonomy</code>: e.g. <code>organism = "9606"</code> (= taxid of <code>Homo sapiens</code>)
</p>
</li></ul>
</td></tr>
<tr><td><code id="getENSEMBL.gtf_+3A_type">type</code></td>
<td>
<p>character, biological sequence type (e.g. &quot;dna&quot;, &quot;cds&quot;)</p>
</td></tr>
<tr><td><code id="getENSEMBL.gtf_+3A_path">path</code></td>
<td>
<p>location where file shall be stored.</p>
</td></tr>
<tr><td><code id="getENSEMBL.gtf_+3A_release">release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character filepath to download file, returns FALSE if failed.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>

<hr>
<h2 id='getENSEMBL.Seq'>Helper function for retrieving biological sequence files from ENSEMBL</h2><span id='topic+getENSEMBL.Seq'></span>

<h3>Description</h3>

<p>This function downloads gff files of query
organisms from ENSEMBL.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getENSEMBL.Seq(
  organism,
  type = "dna",
  id.type = "toplevel",
  release = NULL,
  path
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getENSEMBL.Seq_+3A_organism">organism</code></td>
<td>
<p>Organism selector id,
there are three options to characterize an organism:
</p>

<ul>
<li><p> by <code>scientific name</code>: e.g. <code>organism = "Homo sapiens"</code>
</p>
</li>
<li><p> by <code>database specific accession identifier</code>: e.g. <code>organism = "GCF_000001405.37"</code> (= NCBI RefSeq identifier for <code>Homo sapiens</code>)
</p>
</li>
<li><p> by <code>taxonomic identifier from NCBI Taxonomy</code>: e.g. <code>organism = "9606"</code> (= taxid of <code>Homo sapiens</code>)
</p>
</li></ul>
</td></tr>
<tr><td><code id="getENSEMBL.Seq_+3A_type">type</code></td>
<td>
<p>character, biological sequence type (e.g. &quot;dna&quot;, &quot;cds&quot;)</p>
</td></tr>
<tr><td><code id="getENSEMBL.Seq_+3A_id.type">id.type</code></td>
<td>
<p>a character, default &quot;toplevel&quot;.
id type of assembly, either &quot;toplevel&quot; or &quot;primary_assembly&quot; for genomes.
Can be other strings, for non genome objects.</p>
</td></tr>
<tr><td><code id="getENSEMBL.Seq_+3A_release">release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td></tr>
<tr><td><code id="getENSEMBL.Seq_+3A_path">path</code></td>
<td>
<p>location where file shall be stored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>either a character path to downloaded file, or a logical FALSE, specifying failure.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>

<hr>
<h2 id='getENSEMBLGENOMESInfo'>Retrieve ENSEMBLGENOMES info file</h2><span id='topic+getENSEMBLGENOMESInfo'></span>

<h3>Description</h3>

<p>Retrieve species and genome information from
http://rest.ensemblgenomes.org/info/species?content-type=application/json/.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getENSEMBLGENOMESInfo()
</code></pre>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
info.file &lt;- getENSEMBLGENOMESInfo()
info.file

## End(Not run)
</code></pre>

<hr>
<h2 id='getENSEMBLInfo'>Retrieve ENSEMBL info file</h2><span id='topic+getENSEMBLInfo'></span>

<h3>Description</h3>

<p>Retrieve species and genome information from
http://rest.ensembl.org/info/species?content-type=application/json/.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getENSEMBLInfo(update = FALSE, divisions = ensembl_divisions())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getENSEMBLInfo_+3A_update">update</code></td>
<td>
<p>logical, default FALSE. If TRUE, update cached list,
if FALSE use existing cache (if it exists). For cache location see
<code>cachedir()</code></p>
</td></tr>
<tr><td><code id="getENSEMBLInfo_+3A_divisions">divisions</code></td>
<td>
<p>character, name of divisions to check, default is all from
<code>ensembl_divisions()</code>. If NULL, also all is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a tibble table storing info for all available ENSEMBL divisions.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ensembl_divisions">ensembl_divisions</a></code>, <code><a href="#topic+get.ensembl.info">get.ensembl.info</a></code>, <code><a href="#topic+getKingdomAssemblySummary">getKingdomAssemblySummary</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# look at available divisions
ensembl_divisions()
# retrieve information for all ENSEMBL divisions at once
test &lt;- getENSEMBLInfo()
test
# retrieve information for a particular ENSEMBL division (e.g. EnsemblVertebrates)
test_vertebrates &lt;- get.ensembl.info(update = TRUE, division = "EnsemblVertebrates")
test_vertebrates

## End(Not run)
</code></pre>

<hr>
<h2 id='getFilters'>Retrieve All Available Filters for a Specific Dataset</h2><span id='topic+getFilters'></span>

<h3>Description</h3>

<p>This funcion queries the BioMart API and returns a table
storing all available filters for a specific dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getFilters(mart, dataset, mute_citation = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getFilters_+3A_mart">mart</code></td>
<td>
<p>a character string specifying the database (mart) for which
datasets shall be listed.</p>
</td></tr>
<tr><td><code id="getFilters_+3A_dataset">dataset</code></td>
<td>
<p>a character string specifying the dataset for which filters
shall be listed.</p>
</td></tr>
<tr><td><code id="getFilters_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical value indicating whether citation message should be muted.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getMarts">getMarts</a></code>, <code><a href="#topic+getDatasets">getDatasets</a></code>,
<code><a href="#topic+getAttributes">getAttributes</a></code>, <code><a href="#topic+organismBM">organismBM</a></code>,
<code><a href="#topic+organismFilters">organismFilters</a></code>, <code><a href="#topic+organismAttributes">organismAttributes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# search for available datasets
# getMarts()
# choose database (mart): "ENSEMBL_MART_ENSEMBL"
# head(getDatasets(mart = "ENSEMBL_MART_ENSEMBL"), 10)
# choose dataset: "hsapiens_gene_ensembl"
head(getFilters(mart = "ENSEMBL_MART_ENSEMBL",
                dataset = "hsapiens_gene_ensembl") , 5)

## End(Not run)
</code></pre>

<hr>
<h2 id='getGenome'>Genome Retrieval</h2><span id='topic+getGenome'></span>

<h3>Description</h3>

<p>Main genome retrieval function for an organism of interest.
By specifying the scientific name of an organism of interest the
corresponding fasta-file storing the genome of the organism of interest
can be downloaded and stored locally. Genome files can be retrieved from
several databases. In addition, the genome summary statistics for the
retrieved species is stored locally to provide users with
insights regarding the genome assembly quality (see <code><a href="#topic+summary_genome">summary_genome</a></code> for details).
This is useful when comparing genomes with large difference in genome assembly qualities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGenome(
  db = "refseq",
  organism,
  reference = FALSE,
  skip_bacteria = TRUE,
  release = NULL,
  gunzip = FALSE,
  path = file.path("_ncbi_downloads", "genomes"),
  assembly_type = "toplevel",
  mute_citation = FALSE,
  analyse_genome = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGenome_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="getGenome_+3A_organism">organism</code></td>
<td>
<p>Organism selector id,
there are three options to characterize an organism:
</p>

<ul>
<li><p> by <code>scientific name</code>: e.g. <code>organism = "Homo sapiens"</code>
</p>
</li>
<li><p> by <code>database specific accession identifier</code>: e.g. <code>organism = "GCF_000001405.37"</code> (= NCBI RefSeq identifier for <code>Homo sapiens</code>)
</p>
</li>
<li><p> by <code>taxonomic identifier from NCBI Taxonomy</code>: e.g. <code>organism = "9606"</code> (= taxid of <code>Homo sapiens</code>)
</p>
</li></ul>
</td></tr>
<tr><td><code id="getGenome_+3A_reference">reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database as either a reference genome or a representative genome.</p>
</td></tr>
<tr><td><code id="getGenome_+3A_skip_bacteria">skip_bacteria</code></td>
<td>
<p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p>
</td></tr>
<tr><td><code id="getGenome_+3A_release">release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td></tr>
<tr><td><code id="getGenome_+3A_gunzip">gunzip</code></td>
<td>
<p>a logical, indicating whether or not files should be unzipped.</p>
</td></tr>
<tr><td><code id="getGenome_+3A_path">path</code></td>
<td>
<p>character, default location is paste0(&quot;set_&quot;, toupper(type))</p>
</td></tr>
<tr><td><code id="getGenome_+3A_assembly_type">assembly_type</code></td>
<td>
<p>character, default c(&quot;primary_assembly&quot;, &quot;toplevel&quot;). Used for ensembl only,
specifies the genome assembly type. Searches for both primary and toplevel, and if both are found, uses the
first by order (so primary is prioritized by default).
The Primary assembly should usually be used if it exists.
The &quot;primary assembly&quot; contains all the top-level sequence regions,
excluding alternative haplotypes and patches.
If the primary assembly file is not present for a species
(only defined for standard model organisms),
that indicates that there were no haplotype/patch regions,
and in such cases, the 'toplevel file is used.
For more details see:
<a href="https://grch37.ensembl.org/info/genome/genebuild/assembly.html">ensembl tutorial</a></p>
</td></tr>
<tr><td><code id="getGenome_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical, default FALSE, indicating whether citation message should be muted.</p>
</td></tr>
<tr><td><code id="getGenome_+3A_analyse_genome">analyse_genome</code></td>
<td>
<p>logical, default FALSE. If TRUE, get general genome statistics like
gc content etc. For more details, see ?summary_genome</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI:
</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/
</p>
<p>and creates a directory relative to file type, if you get fasta genomes it will be
_ncbi_downloads/genomes'.
In case the corresponding fasta file already exists within the
'_ncbi_downloads/genomes' folder and is accessible within the workspace,
no download process will be performed.
For other file types the same rule applies.
</p>


<h3>Value</h3>

<p>File path to downloaded genome.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other getBio: 
<code><a href="#topic+getBio">getBio</a>()</code>,
<code><a href="#topic+getCDS">getCDS</a>()</code>,
<code><a href="#topic+getCollection">getCollection</a>()</code>,
<code><a href="#topic+getGFF">getGFF</a>()</code>,
<code><a href="#topic+getProteome">getProteome</a>()</code>,
<code><a href="#topic+getRNA">getRNA</a>()</code>
</p>
<p>Other genome: 
<code><a href="#topic+getGenomeSet">getGenomeSet</a>()</code>,
<code><a href="#topic+read_genome">read_genome</a>()</code>
</p>

<hr>
<h2 id='getGENOMEREPORT'>Retrieve NCBI GENOME_REPORTS file</h2><span id='topic+getGENOMEREPORT'></span>

<h3>Description</h3>

<p>Retrieves NCBI GENOME_REPORTS file from
ftp://ftp.ncbi.nlm.nih.gov/genomes/GENOME_REPORTS/overview.txt.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGENOMEREPORT(
  local_file = file.path(cachedir(), "_ncbi_downloads", "overview.txt")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGENOMEREPORT_+3A_local_file">local_file</code></td>
<td>
<p>character, file path, default: file.path(cachedir(), &quot;_ncbi_downloads&quot;, &quot;overview.txt&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a tibble object with the report
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
report &lt;- getGENOMEREPORT()
report

## End(Not run)
</code></pre>

<hr>
<h2 id='getGenomeSet'>Genome Retrieval of multiple species</h2><span id='topic+getGenomeSet'></span>

<h3>Description</h3>

<p>Main genome retrieval function for a set of organism of interest.
By specifying the scientific names of the organisms of interest the corresponding fasta-files storing the genome of the organisms of interest
will be downloaded and stored locally. Genome files can be retrieved from several databases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGenomeSet(
  db = "refseq",
  organisms,
  reference = FALSE,
  release = NULL,
  skip_bacteria = TRUE,
  gunzip = TRUE,
  update = FALSE,
  path = "set_genomes",
  assembly_type = "toplevel",
  mute_citation = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGenomeSet_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="getGenomeSet_+3A_organisms">organisms</code></td>
<td>
<p>a character vector storing the names of the organisms than shall be retrieved.
There are three available options to characterize an organism:</p>
</td></tr>
<tr><td><code id="getGenomeSet_+3A_reference">reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database as either a reference genome or a representative genome.</p>
</td></tr>
<tr><td><code id="getGenomeSet_+3A_release">release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td></tr>
<tr><td><code id="getGenomeSet_+3A_skip_bacteria">skip_bacteria</code></td>
<td>
<p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p>
</td></tr>
<tr><td><code id="getGenomeSet_+3A_gunzip">gunzip</code></td>
<td>
<p>a logical, indicating whether or not files should be unzipped.</p>
</td></tr>
<tr><td><code id="getGenomeSet_+3A_update">update</code></td>
<td>
<p>logical, default FALSE. Updated backend cached files needed.
Usually keep this false, to make ut run much faster. Only set to TRUE, if you
believe you cache is outdated (Species only exist in newest release etc)</p>
</td></tr>
<tr><td><code id="getGenomeSet_+3A_path">path</code></td>
<td>
<p>a character string specifying the location (a folder) in which
the corresponding genomes shall be stored. Default is
<code>path</code> = <code>"set_genomes"</code>.</p>
</td></tr>
<tr><td><code id="getGenomeSet_+3A_assembly_type">assembly_type</code></td>
<td>
<p>character, default c(&quot;primary_assembly&quot;, &quot;toplevel&quot;). Used for ensembl only,
specifies the genome assembly type. Searches for both primary and toplevel, and if both are found, uses the
first by order (so primary is prioritized by default).
The Primary assembly should usually be used if it exists.
The &quot;primary assembly&quot; contains all the top-level sequence regions,
excluding alternative haplotypes and patches.
If the primary assembly file is not present for a species
(only defined for standard model organisms),
that indicates that there were no haplotype/patch regions,
and in such cases, the 'toplevel file is used.
For more details see:
<a href="https://grch37.ensembl.org/info/genome/genebuild/assembly.html">ensembl tutorial</a></p>
</td></tr>
<tr><td><code id="getGenomeSet_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical, default FALSE, indicating whether citation message should be muted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI:
</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/
</p>
<p>and creates a directory 'set_CDSs' to store
the CDSs of interest as fasta files for future processing.
In case the corresponding fasta file already exists within the
'set_CDSs' folder and is accessible within the workspace,
no download process will be performed.
</p>


<h3>Value</h3>

<p>File path to downloaded genomes
(names are identifiers: 'new' (file was downloaded now),
'old' files did already exist)
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other getBioSet: 
<code><a href="#topic+getBioSet">getBioSet</a>()</code>,
<code><a href="#topic+getCDSSet">getCDSSet</a>()</code>,
<code><a href="#topic+getCollectionSet">getCollectionSet</a>()</code>,
<code><a href="#topic+getGFFSet">getGFFSet</a>()</code>,
<code><a href="#topic+getProteomeSet">getProteomeSet</a>()</code>,
<code><a href="#topic+getRNASet">getRNASet</a>()</code>
</p>
<p>Other genome: 
<code><a href="#topic+getGenome">getGenome</a>()</code>,
<code><a href="#topic+read_genome">read_genome</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getBioSet("refseq", organisms = c("Arabidopsis thaliana",
                                  "Arabidopsis lyrata",
                                  "Capsella rubella"),
                                  set_type = "cds")

## End(Not run)
</code></pre>

<hr>
<h2 id='getGFF'>Genome Annotation Retrieval (GFF3)</h2><span id='topic+getGFF'></span>

<h3>Description</h3>

<p>Main retrieval function for GFF files of an
organism of interest. By specifying the scientific name of an organism of
interest the corresponding gff file storing the annotation  for the organism
of interest can be downloaded and stored locally. GFF files can be retrieved
from several databases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGFF(
  db = "refseq",
  organism,
  reference = FALSE,
  skip_bacteria = TRUE,
  release = NULL,
  gunzip = FALSE,
  remove_annotation_outliers = FALSE,
  path = file.path("_ncbi_downloads", "annotation"),
  mute_citation = FALSE,
  format = "gff3"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGFF_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="getGFF_+3A_organism">organism</code></td>
<td>
<p>Organism selector id,
there are three options to characterize an organism:
</p>

<ul>
<li><p> by <code>scientific name</code>: e.g. <code>organism = "Homo sapiens"</code>
</p>
</li>
<li><p> by <code>database specific accession identifier</code>: e.g. <code>organism = "GCF_000001405.37"</code> (= NCBI RefSeq identifier for <code>Homo sapiens</code>)
</p>
</li>
<li><p> by <code>taxonomic identifier from NCBI Taxonomy</code>: e.g. <code>organism = "9606"</code> (= taxid of <code>Homo sapiens</code>)
</p>
</li></ul>
</td></tr>
<tr><td><code id="getGFF_+3A_reference">reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database as either a reference genome or a representative genome.</p>
</td></tr>
<tr><td><code id="getGFF_+3A_skip_bacteria">skip_bacteria</code></td>
<td>
<p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p>
</td></tr>
<tr><td><code id="getGFF_+3A_release">release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td></tr>
<tr><td><code id="getGFF_+3A_gunzip">gunzip</code></td>
<td>
<p>a logical, indicating whether or not files should be unzipped.</p>
</td></tr>
<tr><td><code id="getGFF_+3A_remove_annotation_outliers">remove_annotation_outliers</code></td>
<td>
<p>shall outlier lines be removed from the input <code>annotation_file</code>?
If yes, then the initial <code>annotation_file</code> will be overwritten and the removed outlier lines will be stored at <code><a href="base.html#topic+tempdir">tempdir</a></code>
for further exploration.</p>
</td></tr>
<tr><td><code id="getGFF_+3A_path">path</code></td>
<td>
<p>a character string specifying the location (a folder) in which
the corresponding annotation file shall be stored. Default is
<code>path = file.path("_ncbi_downloads","annotation")</code>.</p>
</td></tr>
<tr><td><code id="getGFF_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical, default FALSE, indicating whether citation message should be muted.</p>
</td></tr>
<tr><td><code id="getGFF_+3A_format">format</code></td>
<td>
<p>&quot;gff3&quot;, alternative &quot;gtf&quot; for ensembl.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI:
</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/
</p>
<p>and creates a directory relative to file type, if you get fasta genomes it will be
_ncbi_downloads/genomes'.
In case the corresponding fasta file already exists within the
'_ncbi_downloads/genomes' folder and is accessible within the workspace,
no download process will be performed.
For other file types the same rule applies.
</p>


<h3>Value</h3>

<p>File path to downloaded genome.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other getBio: 
<code><a href="#topic+getBio">getBio</a>()</code>,
<code><a href="#topic+getCDS">getCDS</a>()</code>,
<code><a href="#topic+getCollection">getCollection</a>()</code>,
<code><a href="#topic+getGenome">getGenome</a>()</code>,
<code><a href="#topic+getProteome">getProteome</a>()</code>,
<code><a href="#topic+getRNA">getRNA</a>()</code>
</p>
<p>Other gff: 
<code><a href="#topic+getGFFSet">getGFFSet</a>()</code>,
<code><a href="#topic+read_gff">read_gff</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# download the annotation of Arabidopsis thaliana from refseq
# and store the corresponding genome file in '_ncbi_downloads/annotation'
Athal_gff &lt;- getGFF( db       = "refseq",
               organism = "Arabidopsis thaliana",
               path = file.path("_ncbi_downloads","annotation"),
               remove_annotation_outliers = TRUE)
Athal_gff_import &lt;- read_gff(Athal_gff)


# download the genome of Arabidopsis thaliana from genbank
# and store the corresponding genome file in '_ncbi_downloads/annotation'
Athal_gff &lt;- getGFF( db       = "genbank",
               organism = "Arabidopsis thaliana",
               path = file.path("_ncbi_downloads","annotation"),
               remove_annotation_outliers = TRUE)
Athal_gff_import &lt;- read_gff(Athal_gff)

# download the genome of Homo sapiens from ensembl
# and store the corresponding genome file in '_ncbi_downloads/annotation'
Hsap_gff &lt;- getGFF( db       = "ensembl",
               organism = "Homo sapiens",
               path = file.path("_ncbi_downloads","annotation"),
               remove_annotation_outliers = TRUE)
Hsap_gff_import &lt;- read_gff(Hsap_gff)


## End(Not run)

</code></pre>

<hr>
<h2 id='getGFFSet'>GFF retrieval of multiple species</h2><span id='topic+getGFFSet'></span>

<h3>Description</h3>

<p>Main GFF retrieval function for a set of organism of interest.
By specifying the scientific names of the organisms of interest the corresponding fasta-files storing the GFF of the organisms of interest
will be downloaded and stored locally. GFF files can be retrieved from several databases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGFFSet(
  db = "refseq",
  organisms,
  reference = FALSE,
  release = NULL,
  skip_bacteria = TRUE,
  gunzip = TRUE,
  remove_annotation_outliers = FALSE,
  update = FALSE,
  format = "gff3",
  path = "set_GFF",
  mute_citation = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGFFSet_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="getGFFSet_+3A_organisms">organisms</code></td>
<td>
<p>a character vector storing the names of the organisms than shall be retrieved.
There are three available options to characterize an organism:</p>
</td></tr>
<tr><td><code id="getGFFSet_+3A_reference">reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database as either a reference genome or a representative genome.</p>
</td></tr>
<tr><td><code id="getGFFSet_+3A_release">release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td></tr>
<tr><td><code id="getGFFSet_+3A_skip_bacteria">skip_bacteria</code></td>
<td>
<p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p>
</td></tr>
<tr><td><code id="getGFFSet_+3A_gunzip">gunzip</code></td>
<td>
<p>a logical, indicating whether or not files should be unzipped.</p>
</td></tr>
<tr><td><code id="getGFFSet_+3A_remove_annotation_outliers">remove_annotation_outliers</code></td>
<td>
<p>shall outlier lines be removed from the input <code>annotation_file</code>?
If yes, then the initial <code>annotation_file</code> will be overwritten and the removed outlier lines will be stored at <code><a href="base.html#topic+tempdir">tempdir</a></code>
for further exploration.</p>
</td></tr>
<tr><td><code id="getGFFSet_+3A_update">update</code></td>
<td>
<p>logical, default FALSE. Updated backend cached files needed.
Usually keep this false, to make ut run much faster. Only set to TRUE, if you
believe you cache is outdated (Species only exist in newest release etc)</p>
</td></tr>
<tr><td><code id="getGFFSet_+3A_format">format</code></td>
<td>
<p>&quot;gff3&quot;, alternative &quot;gtf&quot; for ensembl.</p>
</td></tr>
<tr><td><code id="getGFFSet_+3A_path">path</code></td>
<td>
<p>character, default location is paste0(&quot;set_&quot;, toupper(set_type))</p>
</td></tr>
<tr><td><code id="getGFFSet_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical, default FALSE, indicating whether citation message should be muted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI:
</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/
</p>
<p>and creates a directory 'set_CDSs' to store
the CDSs of interest as fasta files for future processing.
In case the corresponding fasta file already exists within the
'set_CDSs' folder and is accessible within the workspace,
no download process will be performed.
</p>


<h3>Value</h3>

<p>File path to downloaded genomes
(names are identifiers: 'new' (file was downloaded now),
'old' files did already exist)
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other getBioSet: 
<code><a href="#topic+getBioSet">getBioSet</a>()</code>,
<code><a href="#topic+getCDSSet">getCDSSet</a>()</code>,
<code><a href="#topic+getCollectionSet">getCollectionSet</a>()</code>,
<code><a href="#topic+getGenomeSet">getGenomeSet</a>()</code>,
<code><a href="#topic+getProteomeSet">getProteomeSet</a>()</code>,
<code><a href="#topic+getRNASet">getRNASet</a>()</code>
</p>
<p>Other gff: 
<code><a href="#topic+getGFF">getGFF</a>()</code>,
<code><a href="#topic+read_gff">read_gff</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getBioSet("refseq", organisms = c("Arabidopsis thaliana",
                                  "Arabidopsis lyrata",
                                  "Capsella rubella"),
                                  set_type = "cds")

## End(Not run)
</code></pre>

<hr>
<h2 id='getGO'>Gene Ontology Query</h2><span id='topic+getGO'></span>

<h3>Description</h3>

<p>This function takes a gene id as character vector from a given 
query organism and returns the corresponding GO terms and additional GO 
information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGO(organism, genes, filters, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGO_+3A_organism">organism</code></td>
<td>
<p>a character string specifying the scientific name 
of a query organism.</p>
</td></tr>
<tr><td><code id="getGO_+3A_genes">genes</code></td>
<td>
<p>a character vector storing the gene ids of a organisms 
of interest to be queried against Ensembl Biomart.</p>
</td></tr>
<tr><td><code id="getGO_+3A_filters">filters</code></td>
<td>
<p>a character vector specifying the filter (query key) for 
the Ensembl Biomart query, e.g. <code>filter</code> = <code>"ensembl_gene_id"</code>.</p>
</td></tr>
<tr><td><code id="getGO_+3A_...">...</code></td>
<td>
<p>additional parameters that can be passed to the 
<code><a href="#topic+biomart">biomart</a></code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes the scientific name of a query organism, a set of genes 
for which GO terms and additional information shall be retrieved, and a 
filter argument that specifies the attribute for the query genes.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+biomart">biomart</a></code>, <code><a href="#topic+organismFilters">organismFilters</a></code>, 
<code><a href="#topic+organismBM">organismBM</a></code>, <code><a href="biomaRt.html#topic+getBM">getBM</a></code>, 
<code><a href="#topic+getMarts">getMarts</a></code>, <code><a href="#topic+getDatasets">getDatasets</a></code>, 
<code><a href="#topic+getFilters">getFilters</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run:  
GO_tbl &lt;- getGO(organism = "Arabidopsis thaliana", 
                genes    = c("AT1G06090", "AT1G06100"),
                filters  = "ensembl_gene_id")

# look at the result
head(GO_tbl)

## End(Not run)
</code></pre>

<hr>
<h2 id='getGroups'>Retrieve available groups for a kingdom of life (only available for NCBI RefSeq and NCBI Genbank)</h2><span id='topic+getGroups'></span>

<h3>Description</h3>

<p>A short list of available groups for a kingdom of life.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGroups(db = "refseq", kingdom)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGroups_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome 
shall be retrieved: 
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li></ul>

<p>Default is <code>db = "refseq"</code>.</p>
</td></tr>
<tr><td><code id="getGroups_+3A_kingdom">kingdom</code></td>
<td>
<p>a character string specifying for which kingdom of life 
groups shall be retrieved. See <code><a href="#topic+getKingdoms">getKingdoms</a></code> for details.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+meta.retrieval">meta.retrieval</a></code>, <code><a href="#topic+getGenome">getGenome</a></code>, 
<code><a href="#topic+getProteome">getProteome</a></code>, <code><a href="#topic+getCDS">getCDS</a></code>, <code><a href="#topic+getKingdoms">getKingdoms</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># get possible kigdom names
getKingdoms(db = "refseq")
## Not run: 
# retrieve subgroups for vertebrate_mammalian available from refseq
getGroups(db = "refseq", kingdom = "vertebrate_mammalian")

# get possible kigdom names
getKingdoms(db = "genbank")
# retrieve subgroups for vertebrate_mammalian available from genbank
getGroups(db = "genbank", kingdom = "vertebrate_mammalian")

## End(Not run)
</code></pre>

<hr>
<h2 id='getGTF'>Genome Annotation Retrieval (GTF)</h2><span id='topic+getGTF'></span>

<h3>Description</h3>

<p>Main retrieval function for GTF files of an
organism of interest. By specifying the scientific name of an organism of
interest the corresponding GTF file storing the annotation  for the organism
of interest can be downloaded and stored locally. GTF files can be retrieved
from several databases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGTF(
  db = "ensembl",
  organism,
  remove_annotation_outliers = FALSE,
  path = file.path("ensembl", "annotation"),
  release = NULL,
  mute_citation = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGTF_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="getGTF_+3A_organism">organism</code></td>
<td>
<p>Organism selector id,
there are three options to characterize an organism:
</p>

<ul>
<li><p> by <code>scientific name</code>: e.g. <code>organism = "Homo sapiens"</code>
</p>
</li>
<li><p> by <code>database specific accession identifier</code>: e.g. <code>organism = "GCF_000001405.37"</code> (= NCBI RefSeq identifier for <code>Homo sapiens</code>)
</p>
</li>
<li><p> by <code>taxonomic identifier from NCBI Taxonomy</code>: e.g. <code>organism = "9606"</code> (= taxid of <code>Homo sapiens</code>)
</p>
</li></ul>
</td></tr>
<tr><td><code id="getGTF_+3A_remove_annotation_outliers">remove_annotation_outliers</code></td>
<td>
<p>shall outlier lines be removed from the input <code>annotation_file</code>?
If yes, then the initial <code>annotation_file</code> will be overwritten and the removed outlier lines will be stored at <code><a href="base.html#topic+tempdir">tempdir</a></code>
for further exploration.</p>
</td></tr>
<tr><td><code id="getGTF_+3A_path">path</code></td>
<td>
<p>a character string specifying the location (a folder) in which
the corresponding annotation file shall be stored. Default is
<code>path = file.path("_ncbi_downloads","annotation")</code>.</p>
</td></tr>
<tr><td><code id="getGTF_+3A_release">release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td></tr>
<tr><td><code id="getGTF_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical, default FALSE, indicating whether citation message should be muted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI:
</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/
</p>
<p>and creates a directory relative to file type, if you get fasta genomes it will be
_ncbi_downloads/genomes'.
In case the corresponding fasta file already exists within the
'_ncbi_downloads/genomes' folder and is accessible within the workspace,
no download process will be performed.
For other file types the same rule applies.
</p>


<h3>Value</h3>

<p>File path to downloaded genome.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other getBio: 
<code><a href="#topic+getBio">getBio</a>()</code>,
<code><a href="#topic+getCDS">getCDS</a>()</code>,
<code><a href="#topic+getCollection">getCollection</a>()</code>,
<code><a href="#topic+getGenome">getGenome</a>()</code>,
<code><a href="#topic+getProteome">getProteome</a>()</code>,
<code><a href="#topic+getRNA">getRNA</a>()</code>
</p>
<p>Other gff: 
<code><a href="#topic+getGFFSet">getGFFSet</a>()</code>,
<code><a href="#topic+read_gff">read_gff</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# download the annotation of Homo sapiens from ensembl
# and store the corresponding genome file in 'ensembl/annotation'
getGTF(db            = "ensembl",
       organism      = "Homo sapiens",
       path          = file.path("ensembl","annotation"))

getGTF(db            = "ensembl",
       organism      = "Homo sapiens",
       path          = file.path("ensembl","annotation"),
       assembly_type = "primary_assembly")


## End(Not run)

</code></pre>

<hr>
<h2 id='getKingdomAssemblySummary'>Retrieve and summarise the assembly_summary.txt files from
NCBI for all kingdoms</h2><span id='topic+getKingdomAssemblySummary'></span>

<h3>Description</h3>

<p>Retrieval function of the assembly_summary.txt file
from NCBI for all kingdoms.
The assembly_summary.txt files store available species on NCBI.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getKingdomAssemblySummary(
  db,
  skip_bacteria = TRUE,
  file = assemblies_info_path(db)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getKingdomAssemblySummary_+3A_db">db</code></td>
<td>
<p>database name. E.g. <code>refseq</code> or <code>genbank</code>.</p>
</td></tr>
<tr><td><code id="getKingdomAssemblySummary_+3A_skip_bacteria">skip_bacteria</code></td>
<td>
<p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p>
</td></tr>
<tr><td><code id="getKingdomAssemblySummary_+3A_file">file</code></td>
<td>
<p>path, local path to total summary file, default is in tmp folder.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getSummaryFile">getSummaryFile</a></code>, <code><a href="#topic+getMetaGenomeSummary">getMetaGenomeSummary</a></code>, <code><a href="#topic+get.ensembl.info">get.ensembl.info</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# This example will run the default version of this function
# whereby information for Bacteria are not downloaded
test &lt;- getKingdomAssemblySummary(db = "genbank", skip_bacteria = TRUE)
test
# Users can then retrieve information for Bacteria by skip_bacteria = FALSE
test2 &lt;- getKingdomAssemblySummary(db = "genbank", skip_bacteria = FALSE)
test2

## End(Not run)
</code></pre>

<hr>
<h2 id='getKingdoms'>Retrieve available kingdoms of life</h2><span id='topic+getKingdoms'></span>

<h3>Description</h3>

<p>A short list of available kingdoms of life
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getKingdoms(db = "refseq")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getKingdoms_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved: <code>db = "refseq"</code>, <code>db = "genbank"</code>,
<code>db = "ensembl"</code>, <code>db = "ensemblgenomes"</code>.
Default is <code>db = "refseq"</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+meta.retrieval">meta.retrieval</a></code>, <code><a href="#topic+getGenome">getGenome</a></code>,
<code><a href="#topic+getProteome">getProteome</a></code>, <code><a href="#topic+getCDS">getCDS</a></code>, <code><a href="#topic+getGroups">getGroups</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># retrieve kingdoms available from refseq
getKingdoms(db = "refseq")

# retrieve kingdoms available from genbank
getKingdoms(db = "genbank")
</code></pre>

<hr>
<h2 id='getMarts'>Retrieve information about available Ensembl Biomart databases</h2><span id='topic+getMarts'></span>

<h3>Description</h3>

<p>This funcion queries the Ensembl Biomart API and returns a table
storing information about all available Ensembl Biomart databases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMarts(update = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getMarts_+3A_update">update</code></td>
<td>
<p>logical, default FALSE. If FALSE, use cached file if it exists.
Set to TRUE to force new update</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other biomaRt: 
<code><a href="#topic+biomart">biomart</a>()</code>,
<code><a href="#topic+getAttributes">getAttributes</a>()</code>,
<code><a href="#topic+getDatasets">getDatasets</a>()</code>,
<code><a href="#topic+organismBM">organismBM</a>()</code>,
<code><a href="#topic+organismFilters">organismFilters</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# get a table of all available databases from Ensembl Biomart
getMarts()
 
## End(Not run)
</code></pre>

<hr>
<h2 id='getMetaGenomeAnnotations'>Retrieve annotation *.gff files for metagenomes from NCBI Genbank</h2><span id='topic+getMetaGenomeAnnotations'></span>

<h3>Description</h3>

<p>Retrieve available annotation *.gff files for metagenomes 
from NCBI Genbank. NCBI Genbank allows users
to download entire metagenomes and their annotations of several metagenome 
projects. This function downloads available metagenomes that can then be 
downloaded via <code><a href="#topic+getMetaGenomes">getMetaGenomes</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMetaGenomeAnnotations(
  name,
  path = file.path("_ncbi_downloads", "metagenome", "annotations")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getMetaGenomeAnnotations_+3A_name">name</code></td>
<td>
<p>metagenome name retrieved by <code><a href="#topic+listMetaGenomes">listMetaGenomes</a></code>.</p>
</td></tr>
<tr><td><code id="getMetaGenomeAnnotations_+3A_path">path</code></td>
<td>
<p>a character string specifying the location (a folder) 
in which the corresponding metagenome annotations shall be stored. 
Default is
<code>path</code> = <code>file.path("_ncbi_downloads","metagenome","annotations")</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getMetaGenomes">getMetaGenomes</a></code>, <code><a href="#topic+listMetaGenomes">listMetaGenomes</a></code>, 
<code><a href="#topic+getGFF">getGFF</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Frist, retrieve a list of available metagenomes
listMetaGenomes()

# Now, retrieve the 'human gut metagenome'
getMetaGenomeAnnotations(name = "human gut metagenome")

## End(Not run)
</code></pre>

<hr>
<h2 id='getMetaGenomes'>Retrieve metagenomes from NCBI Genbank</h2><span id='topic+getMetaGenomes'></span>

<h3>Description</h3>

<p>Retrieve available metagenomes from NCBI Genbank. 
NCBI Genbank allows users to download entire metagenomes of several 
metagenome projects. This function downloads available metagenomes that can 
then be downloaded via <code><a href="#topic+getMetaGenomes">getMetaGenomes</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMetaGenomes(name, path = file.path("_ncbi_downloads", "metagenome"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getMetaGenomes_+3A_name">name</code></td>
<td>
<p>metagenome name retrieved by <code><a href="#topic+listMetaGenomes">listMetaGenomes</a></code>.</p>
</td></tr>
<tr><td><code id="getMetaGenomes_+3A_path">path</code></td>
<td>
<p>a character string specifying the location (a folder) in 
which the corresponding metagenome shall be stored. 
Default is <code>path</code> = <code>file.path("_ncbi_downloads","metagenome")</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getMetaGenomeAnnotations">getMetaGenomeAnnotations</a></code>, 
<code><a href="#topic+listMetaGenomes">listMetaGenomes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Frist, retrieve a list of available metagenomes
listMetaGenomes()

# Now, retrieve the 'human gut metagenome'
getMetaGenomes(name = "human gut metagenome")

## End(Not run)
</code></pre>

<hr>
<h2 id='getMetaGenomeSummary'>Retrieve the assembly_summary.txt file from NCBI genbank metagenomes</h2><span id='topic+getMetaGenomeSummary'></span>

<h3>Description</h3>

<p>Retrieval function of the assembly_summary.txt file
from NCBI genbank metagenomes.
This files stores all available metagenome projects on NCBI Genbank.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMetaGenomeSummary(
  local_file = file.path(cachedir(), "assembly_summary_metagenomes_genbank.txt")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getMetaGenomeSummary_+3A_local_file">local_file</code></td>
<td>
<p>where to store this backend file, default:
file.path(cachedir(), &quot;assembly_summary_metagenomes_genbank.txt&quot;)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getKingdomAssemblySummary">getKingdomAssemblySummary</a></code>,
<code><a href="#topic+getSummaryFile">getSummaryFile</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
meta.summary &lt;- getMetaGenomeSummary()
meta.summary

## End(Not run)
</code></pre>

<hr>
<h2 id='getProteome'>Proteome Retrieval</h2><span id='topic+getProteome'></span>

<h3>Description</h3>

<p>Main proteome retrieval function for an organism of interest.
By specifying the scientific name of an organism of interest the
corresponding fasta-file storing the proteome of the organism of interest
can be downloaded and stored locally. Proteome files can be retrieved from
several databases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getProteome(
  db = "refseq",
  organism,
  reference = TRUE,
  skip_bacteria = TRUE,
  release = NULL,
  gunzip = FALSE,
  update = TRUE,
  path = file.path("_ncbi_downloads", "proteomes"),
  mute_citation = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getProteome_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="getProteome_+3A_organism">organism</code></td>
<td>
<p>Organism selector id,
there are three options to characterize an organism:
</p>

<ul>
<li><p> by <code>scientific name</code>: e.g. <code>organism = "Homo sapiens"</code>
</p>
</li>
<li><p> by <code>database specific accession identifier</code>: e.g. <code>organism = "GCF_000001405.37"</code> (= NCBI RefSeq identifier for <code>Homo sapiens</code>)
</p>
</li>
<li><p> by <code>taxonomic identifier from NCBI Taxonomy</code>: e.g. <code>organism = "9606"</code> (= taxid of <code>Homo sapiens</code>)
</p>
</li></ul>
</td></tr>
<tr><td><code id="getProteome_+3A_reference">reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database as either a reference genome or a representative genome.</p>
</td></tr>
<tr><td><code id="getProteome_+3A_skip_bacteria">skip_bacteria</code></td>
<td>
<p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p>
</td></tr>
<tr><td><code id="getProteome_+3A_release">release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td></tr>
<tr><td><code id="getProteome_+3A_gunzip">gunzip</code></td>
<td>
<p>a logical, indicating whether or not files should be unzipped.</p>
</td></tr>
<tr><td><code id="getProteome_+3A_update">update</code></td>
<td>
<p>logical, default TRUE. (Uniprot only for now!)
If species info file exists already,
do not re download, makes it faster but the file can be old, i.e. no longer
as complete as it could be.</p>
</td></tr>
<tr><td><code id="getProteome_+3A_path">path</code></td>
<td>
<p>a character string specifying the location (a folder) in which
the corresponding proteome shall be stored. Default is
<code>path</code> = <code>file.path("_ncbi_downloads","proteomes")</code>.</p>
</td></tr>
<tr><td><code id="getProteome_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical, default FALSE, indicating whether citation message should be muted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI:
</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/
</p>
<p>and creates a directory '_ncbi_downloads/proteomes' to store
the proteome of interest as fasta file for future processing.
</p>


<h3>Value</h3>

<p>File path to downloaded proteome.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other getBio: 
<code><a href="#topic+getBio">getBio</a>()</code>,
<code><a href="#topic+getCDS">getCDS</a>()</code>,
<code><a href="#topic+getCollection">getCollection</a>()</code>,
<code><a href="#topic+getGFF">getGFF</a>()</code>,
<code><a href="#topic+getGenome">getGenome</a>()</code>,
<code><a href="#topic+getRNA">getRNA</a>()</code>
</p>
<p>Other proteome: 
<code><a href="#topic+getProteomeSet">getProteomeSet</a>()</code>,
<code><a href="#topic+read_proteome">read_proteome</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# download the proteome of Arabidopsis thaliana from NCBI RefSeq
# and store the corresponding proteome file in '_ncbi_downloads/refseq/proteomes'
file_path &lt;- getProteome( db       = "refseq",
             organism = "Arabidopsis thaliana",
             path     = file.path("_ncbi_downloads","refseq","proteomes") )
# import proteome into R session
Ath_proteome &lt;- read_proteome(file_path, format = "fasta")

# download the proteome of Arabidopsis thaliana from NCBI Genbank
# and store the corresponding proteome file in '_ncbi_downloads/genbank/proteomes'
file_path &lt;- getProteome( db       = "genbank",
             organism = "Arabidopsis thaliana",
             path     = file.path("_ncbi_downloads","genbank","proteomes") )
# import proteome into R session
Ath_proteome &lt;- read_proteome(file_path, format = "fasta")

# and store the corresponding proteome file in '_downloads/uniprot/proteomes'
file_path &lt;- getProteome( db       = "uniprot",
             organism = "Arabidopsis thaliana",
             path     = file.path("_downloads","uniprot","proteomes") )
# import proteome into R session
Ath_proteome &lt;- read_proteome(file_path, format = "fasta")

# download the proteome of Arabidopsis thaliana from ENSEMBL
# and store the corresponding proteome file in '_downloads/ensembl/proteomes'
file_path &lt;- getProteome( db       = "ensembl",
             organism = "Arabidopsis thaliana",
             path     = file.path("_downloads","ensembl","proteomes") )
# import proteome into R session
Ath_proteome &lt;- read_proteome(file_path, format = "fasta")

## End(Not run)
</code></pre>

<hr>
<h2 id='getProteomeSet'>Proteome retrieval of multiple species</h2><span id='topic+getProteomeSet'></span>

<h3>Description</h3>

<p>Main proteome retrieval function for a set of organism of interest.
By specifying the scientific names of the organisms of interest the corresponding fasta-files storing the proteome of the organisms of interest
will be downloaded and stored locally. proteome files can be retrieved from several databases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getProteomeSet(
  db = "refseq",
  organisms,
  reference = FALSE,
  release = NULL,
  skip_bacteria = TRUE,
  gunzip = TRUE,
  update = FALSE,
  path = "set_proteomes",
  mute_citation = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getProteomeSet_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="getProteomeSet_+3A_organisms">organisms</code></td>
<td>
<p>a character vector storing the names of the organisms than shall be retrieved.
There are three available options to characterize an organism:</p>
</td></tr>
<tr><td><code id="getProteomeSet_+3A_reference">reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database as either a reference genome or a representative genome.</p>
</td></tr>
<tr><td><code id="getProteomeSet_+3A_release">release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td></tr>
<tr><td><code id="getProteomeSet_+3A_skip_bacteria">skip_bacteria</code></td>
<td>
<p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p>
</td></tr>
<tr><td><code id="getProteomeSet_+3A_gunzip">gunzip</code></td>
<td>
<p>a logical, indicating whether or not files should be unzipped.</p>
</td></tr>
<tr><td><code id="getProteomeSet_+3A_update">update</code></td>
<td>
<p>logical, default FALSE. Updated backend cached files needed.
Usually keep this false, to make ut run much faster. Only set to TRUE, if you
believe you cache is outdated (Species only exist in newest release etc)</p>
</td></tr>
<tr><td><code id="getProteomeSet_+3A_path">path</code></td>
<td>
<p>a character string specifying the location (a folder) in which
the corresponding proteomes shall be stored. Default is <code>path</code> = <code>"set_proteomes"</code>.</p>
</td></tr>
<tr><td><code id="getProteomeSet_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical, default FALSE, indicating whether citation message should be muted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI:
</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/
</p>
<p>and creates a directory 'set_CDSs' to store
the CDSs of interest as fasta files for future processing.
In case the corresponding fasta file already exists within the
'set_CDSs' folder and is accessible within the workspace,
no download process will be performed.
</p>


<h3>Value</h3>

<p>File path to downloaded genomes
(names are identifiers: 'new' (file was downloaded now),
'old' files did already exist)
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other getBioSet: 
<code><a href="#topic+getBioSet">getBioSet</a>()</code>,
<code><a href="#topic+getCDSSet">getCDSSet</a>()</code>,
<code><a href="#topic+getCollectionSet">getCollectionSet</a>()</code>,
<code><a href="#topic+getGFFSet">getGFFSet</a>()</code>,
<code><a href="#topic+getGenomeSet">getGenomeSet</a>()</code>,
<code><a href="#topic+getRNASet">getRNASet</a>()</code>
</p>
<p>Other proteome: 
<code><a href="#topic+getProteome">getProteome</a>()</code>,
<code><a href="#topic+read_proteome">read_proteome</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# download the proteomes of three different species at the same time
#### Database: NCBI RefSeq
file_paths &lt;- getProteomeSet(db = "refseq", organisms = c("Arabidopsis thaliana",
                                      "Arabidopsis lyrata",
                                       "Capsella rubella"))
# look at file paths
file_paths

#### Database: NCBI Genbank
file_paths &lt;- getProteomeSet(db = "genbank", organisms = c("Arabidopsis thaliana",
                                      "Arabidopsis lyrata",
                                       "Capsella rubella"))
# look at file paths
file_paths

# download the proteomes of three different species at the same time
#### Database: ENSEMBL
file_paths &lt;- getProteomeSet(db = "ensembl", organisms = c("Homo sapiens",
                                      "Mus musculus",
                                       "Caenorhabditis elegans"))
# look at file paths
file_paths

# download the proteomes of three different species at the same time
#### Database: UniProt
file_paths &lt;- getProteomeSet(db = "uniprot", organisms = c("Homo sapiens",
                                      "Mus musculus",
                                       "Caenorhabditis elegans"))
# look at file paths
file_paths

## End(Not run)

</code></pre>

<hr>
<h2 id='getReleases'>Retrieve available database releases or versions
of ENSEMBL</h2><span id='topic+getReleases'></span>

<h3>Description</h3>

<p>Retrieve available database releases or versions of
ENSEMBL.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getReleases(db = "ensembl")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getReleases_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which
available resease versions shall be retrieved:
</p>

<ul>
<li> <p><code>db = "ensembl"</code>
</p>
</li></ul>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# retrieve available resease versions of ENSEMBL
getReleases("ensembl")

## End(Not run)
</code></pre>

<hr>
<h2 id='getRepeatMasker'>Repeat Masker Retrieval</h2><span id='topic+getRepeatMasker'></span>

<h3>Description</h3>

<p>Main Repeat Masker output retrieval function for an
organism of interest.
By specifying the scientific name of an organism of interest the
corresponding Repeat Masker file storing the genome of the organism of
interest can be downloaded and stored locally.
Repeat Masker files can be retrieved from several databases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getRepeatMasker(
  db = "refseq",
  organism,
  reference = FALSE,
  skip_bacteria = TRUE,
  release = NULL,
  gunzip = FALSE,
  path = file.path("_ncbi_downloads", "repeatmasker"),
  mute_citation = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getRepeatMasker_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="getRepeatMasker_+3A_organism">organism</code></td>
<td>
<p>a character string specifying the scientific name of the
organism of interest, e.g. <code>organism = "Homo sapiens"</code>.</p>
</td></tr>
<tr><td><code id="getRepeatMasker_+3A_reference">reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database as either a reference genome or a representative genome.</p>
</td></tr>
<tr><td><code id="getRepeatMasker_+3A_skip_bacteria">skip_bacteria</code></td>
<td>
<p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p>
</td></tr>
<tr><td><code id="getRepeatMasker_+3A_release">release</code></td>
<td>
<p>most recent database version is used. release = 75 would for human would give the stable GRCh37 release in ensembl. 
Value must be &gt; 46, since ensembl did not structure their data if the standard format before that.</p>
</td></tr>
<tr><td><code id="getRepeatMasker_+3A_gunzip">gunzip</code></td>
<td>
<p>a logical, indicating whether or not files should be unzipped.</p>
</td></tr>
<tr><td><code id="getRepeatMasker_+3A_path">path</code></td>
<td>
<p>a character string specifying the location (a folder) in which
the corresponding file shall be stored. Default is
<code>path</code> = <code>file.path("_ncbi_downloads","repeatmasker")</code>.</p>
</td></tr>
<tr><td><code id="getRepeatMasker_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical value indicating whether citation message should be muted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI:
</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/
</p>
<p>and creates a directory '_ncbi_downloads/repeatmasker' to store
the files of interest as fasta file for future processing.
In case the corresponding fasta file already exists within the
'_ncbi_downloads/repeatmasker' folder and is accessible within the workspace,
no download process will be performed.
</p>


<h3>Value</h3>

<p>File path to downloaded Repeat Masker output file.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getGenome">getGenome</a></code>, <code><a href="#topic+getProteome">getProteome</a></code>, <code><a href="#topic+getCDS">getCDS</a></code>,
<code><a href="#topic+getGFF">getGFF</a></code>, <code><a href="#topic+getRNA">getRNA</a></code>, <code><a href="#topic+getCollection">getCollection</a></code>, <code><a href="#topic+meta.retrieval">meta.retrieval</a></code>,
<code><a href="#topic+read_rm">read_rm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# download the Repeat Masker output file of Homo sapiens from refseq
# and store the corresponding genome file in '_ncbi_downloads/genomes'
file_path &lt;- getRepeatMasker( db       = "refseq",
             organism = "Homo sapiens",
             path = file.path("_ncbi_downloads","repeatmasker"))

Hsap_repeatmasker &lt;- read_rm(file_path)


## End(Not run)

</code></pre>

<hr>
<h2 id='getRNA'>RNA Sequence Retrieval</h2><span id='topic+getRNA'></span>

<h3>Description</h3>

<p>Main retrieval function for RNA sequences of an organism
of interest. By specifying the scientific name of an organism of interest the
corresponding fasta-file storing the RNA information for the organism
of interest can be downloaded and stored locally.
RNA files can be retrieved from several databases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getRNA(
  db = "refseq",
  organism,
  reference = FALSE,
  skip_bacteria = TRUE,
  release = NULL,
  assembly_type = "toplevel",
  path = file.path("_ncbi_downloads", "RNA"),
  gunzip = FALSE,
  mute_citation = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getRNA_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="getRNA_+3A_organism">organism</code></td>
<td>
<p>Organism selector id,
there are three options to characterize an organism:
</p>

<ul>
<li><p> by <code>scientific name</code>: e.g. <code>organism = "Homo sapiens"</code>
</p>
</li>
<li><p> by <code>database specific accession identifier</code>: e.g. <code>organism = "GCF_000001405.37"</code> (= NCBI RefSeq identifier for <code>Homo sapiens</code>)
</p>
</li>
<li><p> by <code>taxonomic identifier from NCBI Taxonomy</code>: e.g. <code>organism = "9606"</code> (= taxid of <code>Homo sapiens</code>)
</p>
</li></ul>
</td></tr>
<tr><td><code id="getRNA_+3A_reference">reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database as either a reference genome or a representative genome.</p>
</td></tr>
<tr><td><code id="getRNA_+3A_skip_bacteria">skip_bacteria</code></td>
<td>
<p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p>
</td></tr>
<tr><td><code id="getRNA_+3A_release">release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td></tr>
<tr><td><code id="getRNA_+3A_assembly_type">assembly_type</code></td>
<td>
<p>character, default c(&quot;primary_assembly&quot;, &quot;toplevel&quot;). Used for ensembl only,
specifies the genome assembly type. Searches for both primary and toplevel, and if both are found, uses the
first by order (so primary is prioritized by default).
The Primary assembly should usually be used if it exists.
The &quot;primary assembly&quot; contains all the top-level sequence regions,
excluding alternative haplotypes and patches.
If the primary assembly file is not present for a species
(only defined for standard model organisms),
that indicates that there were no haplotype/patch regions,
and in such cases, the 'toplevel file is used.
For more details see:
<a href="https://grch37.ensembl.org/info/genome/genebuild/assembly.html">ensembl tutorial</a></p>
</td></tr>
<tr><td><code id="getRNA_+3A_path">path</code></td>
<td>
<p>a character string specifying the location (a folder) in which
the corresponding
CDS file shall be stored. Default is
<code>path</code> = <code>file.path("_ncbi_downloads","RNA")</code>.</p>
</td></tr>
<tr><td><code id="getRNA_+3A_gunzip">gunzip</code></td>
<td>
<p>a logical, indicating whether or not files should be unzipped.</p>
</td></tr>
<tr><td><code id="getRNA_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical, default FALSE, indicating whether citation message should be muted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI:
</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/
</p>
<p>and creates a directory relative to file type, if you get fasta genomes it will be
_ncbi_downloads/genomes'.
In case the corresponding fasta file already exists within the
'_ncbi_downloads/genomes' folder and is accessible within the workspace,
no download process will be performed.
For other file types the same rule applies.
</p>


<h3>Value</h3>

<p>File path to downloaded genome.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other getBio: 
<code><a href="#topic+getBio">getBio</a>()</code>,
<code><a href="#topic+getCDS">getCDS</a>()</code>,
<code><a href="#topic+getCollection">getCollection</a>()</code>,
<code><a href="#topic+getGFF">getGFF</a>()</code>,
<code><a href="#topic+getGenome">getGenome</a>()</code>,
<code><a href="#topic+getProteome">getProteome</a>()</code>
</p>
<p>Other rna: 
<code><a href="#topic+getRNASet">getRNASet</a>()</code>,
<code><a href="#topic+read_rna">read_rna</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# download the RNA of Arabidopsis thaliana from refseq
# and store the corresponding RNA file in '_ncbi_downloads/RNA'
file_path &lt;- getRNA( db       = "refseq",
             organism = "Arabidopsis thaliana",
             path     = file.path("_ncbi_downloads","RNA"))

Ath_RNA &lt;- read_rna(file_path, format = "fasta")

## End(Not run)
</code></pre>

<hr>
<h2 id='getRNASet'>RNA Retrieval of multiple species</h2><span id='topic+getRNASet'></span>

<h3>Description</h3>

<p>Main RNA retrieval function for a set of organism of interest.
By specifying the scientific names of the organisms of interest the corresponding fasta-files storing the RNA of the organisms of interest
will be downloaded and stored locally. RNA files can be retrieved from several databases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getRNASet(
  db = "refseq",
  organisms,
  reference = FALSE,
  release = NULL,
  skip_bacteria = TRUE,
  gunzip = TRUE,
  update = FALSE,
  path = "set_RNAs",
  mute_citation = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getRNASet_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="getRNASet_+3A_organisms">organisms</code></td>
<td>
<p>a character vector storing the names of the organisms than shall be retrieved.
There are three available options to characterize an organism:</p>
</td></tr>
<tr><td><code id="getRNASet_+3A_reference">reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database as either a reference genome or a representative genome.</p>
</td></tr>
<tr><td><code id="getRNASet_+3A_release">release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td></tr>
<tr><td><code id="getRNASet_+3A_skip_bacteria">skip_bacteria</code></td>
<td>
<p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p>
</td></tr>
<tr><td><code id="getRNASet_+3A_gunzip">gunzip</code></td>
<td>
<p>a logical, indicating whether or not files should be unzipped.</p>
</td></tr>
<tr><td><code id="getRNASet_+3A_update">update</code></td>
<td>
<p>logical, default FALSE. Updated backend cached files needed.
Usually keep this false, to make ut run much faster. Only set to TRUE, if you
believe you cache is outdated (Species only exist in newest release etc)</p>
</td></tr>
<tr><td><code id="getRNASet_+3A_path">path</code></td>
<td>
<p>a character string specifying the location (a folder) in which
the corresponding RNAs shall be stored. Default is <code>path</code> = <code>"set_RNAs"</code>.</p>
</td></tr>
<tr><td><code id="getRNASet_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical, default FALSE, indicating whether citation message should be muted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI:
</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/
</p>
<p>and creates a directory 'set_CDSs' to store
the CDSs of interest as fasta files for future processing.
In case the corresponding fasta file already exists within the
'set_CDSs' folder and is accessible within the workspace,
no download process will be performed.
</p>


<h3>Value</h3>

<p>File path to downloaded genomes
(names are identifiers: 'new' (file was downloaded now),
'old' files did already exist)
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other getBioSet: 
<code><a href="#topic+getBioSet">getBioSet</a>()</code>,
<code><a href="#topic+getCDSSet">getCDSSet</a>()</code>,
<code><a href="#topic+getCollectionSet">getCollectionSet</a>()</code>,
<code><a href="#topic+getGFFSet">getGFFSet</a>()</code>,
<code><a href="#topic+getGenomeSet">getGenomeSet</a>()</code>,
<code><a href="#topic+getProteomeSet">getProteomeSet</a>()</code>
</p>
<p>Other rna: 
<code><a href="#topic+getRNA">getRNA</a>()</code>,
<code><a href="#topic+read_rna">read_rna</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getBioSet("refseq", organisms = c("Arabidopsis thaliana",
                                  "Arabidopsis lyrata",
                                  "Capsella rubella"),
                                  set_type = "cds")

## End(Not run)
</code></pre>

<hr>
<h2 id='getSummaryFile'>Helper function to retrieve the assembly_summary.txt file from NCBI</h2><span id='topic+getSummaryFile'></span>

<h3>Description</h3>

<p>Retrieval function of the assembly_summary.txt file from NCBI.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSummaryFile(db, kingdom, file = assemblies_info_path(db, kingdom))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSummaryFile_+3A_db">db</code></td>
<td>
<p>database name. E.g. <code>refseq</code> or <code>genbank</code>.</p>
</td></tr>
<tr><td><code id="getSummaryFile_+3A_kingdom">kingdom</code></td>
<td>
<p>kingdom for which assembly_summary.txt file shall be
retrieved. See also <code><a href="#topic+getKingdoms">getKingdoms</a></code>.</p>
</td></tr>
<tr><td><code id="getSummaryFile_+3A_file">file</code></td>
<td>
<p>path, local path to total summary file, default is in tmp folder.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getKingdomAssemblySummary">getKingdomAssemblySummary</a></code>,
<code><a href="#topic+getMetaGenomeSummary">getMetaGenomeSummary</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
test &lt;- getSummaryFile("refseq","plant")
test

## End(Not run)
</code></pre>

<hr>
<h2 id='getUniProtInfo'>Get uniprot info from organism</h2><span id='topic+getUniProtInfo'></span>

<h3>Description</h3>

<p>Get uniprot info from organism
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getUniProtInfo(organism, path = cachedir(), update = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getUniProtInfo_+3A_organism">organism</code></td>
<td>
<p>character, name of organism</p>
</td></tr>
<tr><td><code id="getUniProtInfo_+3A_path">path</code></td>
<td>
<p>path at which the info file shall be stored locally.</p>
</td></tr>
<tr><td><code id="getUniProtInfo_+3A_update">update</code></td>
<td>
<p>shall the internal <code><a href="#topic+cachedir">cachedir</a></code> file be deleted and the info
file freshly downloaded from the UniProt API?</p>
</td></tr>
</table>

<hr>
<h2 id='getUniProtSTATS'>Retrieve UniProt Database Information File (STATS)</h2><span id='topic+getUniProtSTATS'></span>

<h3>Description</h3>

<p>The UniProt stores a <code>STATS</code> file to summarise all available information for their reference proteomes.
Users can now download this file and process it with <code>biomartr</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getUniProtSTATS(update = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getUniProtSTATS_+3A_update">update</code></td>
<td>
<p>shall the internal <code><a href="#topic+cachedir">cachedir</a></code> file be deleted and the <code>STATS</code>
file freshly downloaded from the UniProt FTP servers?</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# retrieve STATS file from UniProt
uniprot_info &lt;- getUniProtSTATS(update = TRUE)
# look at results
uniprot_info

## End(Not run)
</code></pre>

<hr>
<h2 id='is.genome.available'>Check Genome Availability</h2><span id='topic+is.genome.available'></span>

<h3>Description</h3>

<p>This function checks the availability of a given genome on the
NBCI servers specified as scientific name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.genome.available(
  db = "refseq",
  organism,
  skip_bacteria = TRUE,
  details = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.genome.available_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li>
<li> <p><code>db = "uniprot"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="is.genome.available_+3A_organism">organism</code></td>
<td>
<p>there are three options to characterize an organism:
</p>

<ul>
<li><p> by <code>scientific name</code>: e.g. <code>organism = "Homo sapiens"</code>
</p>
</li>
<li><p> by <code>database specific accession identifier</code>: e.g. <code>organism = "GCF_000001405.37"</code> (= NCBI RefSeq identifier for <code>Homo sapiens</code>)
</p>
</li>
<li><p> by <code>taxonomic identifier from NCBI Taxonomy</code>: e.g. <code>organism = "9606"</code> (= taxid of <code>Homo sapiens</code>)
</p>
</li></ul>
</td></tr>
<tr><td><code id="is.genome.available_+3A_skip_bacteria">skip_bacteria</code></td>
<td>
<p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p>
</td></tr>
<tr><td><code id="is.genome.available_+3A_details">details</code></td>
<td>
<p>a logical value specifying whether or not details on genome
size, kingdom, etc. shall be printed to the console intead of a
boolean value.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally this function calls the <code><a href="#topic+listGenomes">listGenomes</a></code> function to
detect all available genomes and checks whether or not the specified organism
is available for download.
</p>


<h3>Value</h3>

<p>a logical value specifing whether or not the genome of the input
organism is available. In case <code>details</code> = <code>TRUE</code> only a character
string specifying the genome details is being returned.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# checking whether the Homo sapiens genome is stored on NCBI
is.genome.available(organism = "Homo sapiens", db = "refseq")

# and printing details
is.genome.available(organism = "Homo sapiens", db = "refseq", details = TRUE)

# checking whether the Homo sapiens genome is stored on ENSEMBL
is.genome.available(organism = "Homo sapiens", db = "ensembl")

# and printing details
is.genome.available(organism = "Homo sapiens",
                    details = TRUE,
                    db = "ensembl")

## End(Not run)
</code></pre>

<hr>
<h2 id='listDatabases'>Retrieve a List of Available NCBI Databases for Download</h2><span id='topic+listDatabases'></span><span id='topic+listNCBIDatabases'></span>

<h3>Description</h3>

<p>This function allows you to retrieve a list of database names 
and versions that can be downloaded from correspondning servers.
</p>
<p>Database retrieval is crucial for most biological studies and analyses.
There is a vast diversity of databases that can be accessed remotely or that 
can be downloaded to your local machine. This function provides an interface 
to databases that can be downloaded from NCBI servers and lists all available
databases and their database version to be able to select an appropriate 
database for download with <code><a href="#topic+download.database">download.database</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>listDatabases(db = "nr", update = FALSE)

listNCBIDatabases(db = "nr", update = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="listDatabases_+3A_db">db</code></td>
<td>
<p>a character string specifying the name of the database that shall 
be searched for.</p>
</td></tr>
<tr><td><code id="listDatabases_+3A_update">update</code></td>
<td>
<p>a logical value specifying whether or not the local 
listDatabases.txt file shall be updated by remote access to NCBI.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+download.database">download.database</a></code>, <code><a href="#topic+download.database.all">download.database.all</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# retrieve all versions of the NCBI 'nr' database that can be downloaded
listNCBIDatabases(db = "nr")

# analogous:
# listNCBIDatabases(db = "cdd")
# listNCBIDatabases(db = "nt")
# listNCBIDatabases(db = "gss")
# listNCBIDatabases(db = "refseq_protein")

## End(Not run)
</code></pre>

<hr>
<h2 id='listGenomes'>List All Available Genomes either by kingdom, group, or subgroup</h2><span id='topic+listGenomes'></span>

<h3>Description</h3>

<p>This function retrieves the names of all genomes available on
the NCBI ftp:// server and stores the results in a file named 'overview.txt'
inside the directory _ncbi_downloads' that is built inside the workspace.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>listGenomes(
  db = "refseq",
  type = "all",
  subset = NULL,
  details = FALSE,
  update = FALSE,
  skip_bacteria = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="listGenomes_+3A_db">db</code></td>
<td>
<p>a character string specifying the database for which genome
availability shall be checked. Available options are:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="listGenomes_+3A_type">type</code></td>
<td>
<p>a character string specifying a potential filter of available
genomes. Available options are:
</p>

<ul>
<li> <p><code>type = "all", no subset</code>
</p>
</li>
<li> <p><code>type = "kingdom", subset on kingdom</code>
</p>
</li>
<li> <p><code>type = "group", subset on group</code>
</p>
</li>
<li> <p><code>type = "subgroup", subset on subgroup</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="listGenomes_+3A_subset">subset</code></td>
<td>
<p>a character string or character vector specifying a subset of
<code>type</code>. E.g. if users are interested in retrieving all
<code>Eukaryota</code> species, they can specify: <code>type = "kingdom"</code> and
<code>subset = "Eukaryota"</code>.</p>
</td></tr>
<tr><td><code id="listGenomes_+3A_details">details</code></td>
<td>
<p>a boolean value specifying whether only the scientific names
of stored genomes shall be returned (details = FALSE) or all information such
as
</p>

<ul>
<li> <p><code>organism_name</code>
</p>
</li>
<li> <p><code>kingdoms</code>
</p>
</li>
<li> <p><code>group</code>
</p>
</li>
<li> <p><code>subgroup</code>
</p>
</li>
<li> <p><code>file_size_MB</code>, etc.
</p>
</li></ul>
</td></tr>
<tr><td><code id="listGenomes_+3A_update">update</code></td>
<td>
<p>logical, default FALSE. If TRUE, update cached list,
if FALSE use existing cache (if it exists). For cache location see
<code>cachedir()</code></p>
</td></tr>
<tr><td><code id="listGenomes_+3A_skip_bacteria">skip_bacteria</code></td>
<td>
<p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI
and creates a directory '_ncbi_downloads' in the <code>temdir()</code>
folder to store the overview.txt file for future processing. In case the
overview.txt file already exists within the '_ncbi_downloads' folder and is
accessible within the workspace, no download process will be performed again.
</p>


<h3>Note</h3>

<p>Please note that the ftp:// connection relies on the NCBI or ENSEMBL server
and cannot be accurately accessed via a proxy.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# print details for refseq
listGenomes(db = "refseq")
# print details for all plants in refseq
listGenomes(db = "refseq", type = "kingdom")
# print details for all plant groups in refseq
listGenomes(db = "refseq", type = "group")
# print details for all plant subgroups in refseq
listGenomes(db = "refseq", type = "subgroup")
# Ensembl
listGenomes(db = "ensembl", type = "kingdom", subset = "EnsemblVertebrates")

## End(Not run)
</code></pre>

<hr>
<h2 id='listGroups'>List number of available genomes in each taxonomic group</h2><span id='topic+listGroups'></span>

<h3>Description</h3>

<p>Users can retrieve the available number of sequenced 
genomes per group. Only available for <code>db = "refseq"</code> and 
<code>db = "genbank"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>listGroups(db = "refseq", kingdom = "all", details = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="listGroups_+3A_db">db</code></td>
<td>
<p>a character string specifying the database for which genome 
availability shall be checked. Available options are:
</p>

<ul>
<li> <p><code>db = "refseq"</code> 
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="listGroups_+3A_kingdom">kingdom</code></td>
<td>
<p>a kingdom specification retrieved by 
<code><a href="#topic+getKingdoms">getKingdoms</a></code>.</p>
</td></tr>
<tr><td><code id="listGroups_+3A_details">details</code></td>
<td>
<p>shall all species corresponding to the specified 
<code>kingdom</code> be returned? Default is <code>details = FALSE</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+listGenomes">listGenomes</a></code>, <code><a href="#topic+is.genome.available">is.genome.available</a></code>, 
<code><a href="#topic+listKingdoms">listKingdoms</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# example for refseq
listGroups(db = "refseq")
# example for genbank
listGroups(db = "genbank")
### in case groups should be specified by kingdom
# first, retrieve available kingdom names
listKingdoms()
# now we choose kingdom "bacteria"
listGroups(db = "refseq", kingdom = "bacteria")
# or
listGroups(db = "genbank", kingdom = "bacteria")

## End(Not run)
</code></pre>

<hr>
<h2 id='listKingdoms'>List number of available genomes in each kingdom of life</h2><span id='topic+listKingdoms'></span>

<h3>Description</h3>

<p>Users can retrieve the available number of sequenced 
genomes per kingdom.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>listKingdoms(db = "refseq")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="listKingdoms_+3A_db">db</code></td>
<td>
<p>a character string specifying the database for which genome 
availability shall be checked, 
e.g. <code>db = "refseq"</code>, <code>db = "genbank"</code>, <code>db = "ensembl"</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+listGenomes">listGenomes</a></code>, <code><a href="#topic+is.genome.available">is.genome.available</a></code>, 
<code><a href="#topic+listGroups">listGroups</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# list number of available genomes in refseq for each kingdom of life
listKingdoms(db = "refseq")
# example for genbank
listKingdoms(db = "genbank")
# example for ensembl
listKingdoms(db = "ensembl")

## End(Not run)
</code></pre>

<hr>
<h2 id='listMetaGenomes'>List available metagenomes on NCBI Genbank</h2><span id='topic+listMetaGenomes'></span>

<h3>Description</h3>

<p>List available metagenomes on NCBI genbank. NCBI genbank 
allows users to download entire metagenomes of several metagenome projects. 
This function lists all available metagenomes that can then be downloaded via
<code><a href="#topic+getMetaGenomes">getMetaGenomes</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>listMetaGenomes(details = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="listMetaGenomes_+3A_details">details</code></td>
<td>
<p>a boolean value specifying whether only the scientific names 
of stored metagenomes shall be returned (<code>details = FALSE</code>) or all 
information such as &quot;organism_name&quot;,&quot;bioproject&quot;, 
etc (<code>details = TRUE</code>).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getMetaGenomes">getMetaGenomes</a></code>, <code><a href="#topic+getMetaGenomeSummary">getMetaGenomeSummary</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# retrieve available metagenome projects at NCBI Genbank
listMetaGenomes()
# retrieve detailed information on available metagenome projects 
# at NCBI Genbank
listMetaGenomes(details = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='meta.retrieval'>Perform Meta-Genome Retrieval</h2><span id='topic+meta.retrieval'></span>

<h3>Description</h3>

<p>Download genomes, proteomes, cds, gff, rna, or assembly stats
files of all species within a kingdom of life.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meta.retrieval(
  db = "refseq",
  kingdom,
  group = NULL,
  type = "genome",
  restart_at_last = TRUE,
  reference = FALSE,
  combine = FALSE,
  path = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meta.retrieval_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "emsembl"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="meta.retrieval_+3A_kingdom">kingdom</code></td>
<td>
<p>a character string specifying the kingdom of the organisms
of interest, e.g.
</p>

<ul>
<li><p> For <code>NCBI RefSeq</code>:
</p>

<ul>
<li> <p><code>kingdom = "archaea"</code>
</p>
</li>
<li> <p><code>kingdom = "bacteria"</code>
</p>
</li>
<li> <p><code>kingdom = "fungi"</code>
</p>
</li>
<li> <p><code>kingdom = "invertebrate"</code>
</p>
</li>
<li> <p><code>kingdom = "plant"</code>
</p>
</li>
<li> <p><code>kingdom = "protozoa"</code>
</p>
</li>
<li> <p><code>kingdom = "viral"</code>
</p>
</li>
<li> <p><code>kingdom = "vertebrate_mammalian"</code>
</p>
</li>
<li> <p><code>kingdom = "vertebrate_other"</code>
</p>
</li></ul>

</li>
<li><p> For <code>NCBI Genbank</code>:
</p>

<ul>
<li> <p><code>kingdom = "archaea"</code>
</p>
</li>
<li> <p><code>kingdom = "bacteria"</code>
</p>
</li>
<li> <p><code>kingdom = "fungi"</code>
</p>
</li>
<li> <p><code>kingdom = "invertebrate"</code>
</p>
</li>
<li> <p><code>kingdom = "plant"</code>
</p>
</li>
<li> <p><code>kingdom = "protozoa"</code>
</p>
</li>
<li> <p><code>kingdom = "vertebrate_mammalian"</code>
</p>
</li>
<li> <p><code>kingdom = "vertebrate_other"</code>
</p>
</li></ul>

</li>
<li><p> For <code>ENSEMBL</code>:
</p>

<ul>
<li> <p><code>kingdom = "EnsemblVertebrates"</code>
</p>
</li>
<li> <p><code>kingdom = "EnsemblPlants"</code>
</p>
</li>
<li> <p><code>kingdom = "EnsemblFungi"</code>
</p>
</li>
<li> <p><code>kingdom = "EnsemblMetazoa"</code>
</p>
</li>
<li> <p><code>kingdom = "EnsemblBacteria"</code>
</p>
</li>
<li> <p><code>kingdom = "EnsemblProtists"</code>
</p>
</li></ul>

</li></ul>

<p>Available kingdoms can be retrieved with <code><a href="#topic+getKingdoms">getKingdoms</a></code>.</p>
</td></tr>
<tr><td><code id="meta.retrieval_+3A_group">group</code></td>
<td>
<p>only species belonging to this subgroup will be downloaded.
Groups can be retrieved with <code><a href="#topic+getGroups">getGroups</a></code>.</p>
</td></tr>
<tr><td><code id="meta.retrieval_+3A_type">type</code></td>
<td>
<p>type of sequences that shall be retrieved. Options are:
</p>

<ul>
<li> <p><code>type = "genome"</code> :
(for genome assembly retrieval; see also <code><a href="#topic+getGenome">getGenome</a></code>),
</p>
</li>
<li> <p><code>type = "proteome"</code> :
(for proteome retrieval; see also <code><a href="#topic+getProteome">getProteome</a></code>),
</p>
</li>
<li> <p><code>type = "cds"</code> :
(for coding sequence retrieval; see also <code><a href="#topic+getCDS">getCDS</a></code>),
</p>
</li>
<li> <p><code>type = "gff"</code> :
(for annotation file retrieval in gff format; see also <code><a href="#topic+getGFF">getGFF</a></code>),
</p>
</li>
<li> <p><code>type = "gtf"</code> :
(for annotation file retrieval in gtf format (only for ensembl and
ensemblgenomes); see also <code><a href="#topic+getGTF">getGTF</a></code>)
</p>
</li>
<li> <p><code>type = "rna"</code> :
(for RNA file retrieval in fasta format; see also <code><a href="#topic+getRNA">getRNA</a></code>),
</p>
</li>
<li> <p><code>type = "rm"</code> :
(for Repeat Masker output file retrieval; see also
<code><a href="#topic+getRepeatMasker">getRepeatMasker</a></code>),
</p>
</li>
<li> <p><code>type = "assemblystats"</code> :
(for genome assembly quality stats file retrieval;
see also <code><a href="#topic+getAssemblyStats">getAssemblyStats</a></code>).
</p>
</li></ul>
</td></tr>
<tr><td><code id="meta.retrieval_+3A_restart_at_last">restart_at_last</code></td>
<td>
<p>a logical value indicating whether or not <code>meta.retrieval</code> should pick up at the last species when re-running the function.
</p>

<ul>
<li><p> If <code>restart_at_last = TRUE</code> (Default) then <code>meta.retrieval</code> will skip all organisms that are already present in the folder
and will start downloading all remaining species. However, this way <code>meta.wretrieval</code> will not be able to check whether
already downloaded organism files are corrupted or not by checking the md5 checksum.
</p>
</li>
<li><p> If <code>restart_at_last = FALSE</code> then <code>meta.retrieval</code> will start from the beginning and crawl through already downloaded
organism files and check whether already downloaded organism files are corrupted or not by checking the md5 checksum.
After checking existing files the function will start downloading all remaining organisms.
</p>
</li></ul>
</td></tr>
<tr><td><code id="meta.retrieval_+3A_reference">reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database
as either a reference genome or a representative genome. Options are:
</p>

<ul>
<li> <p><code>reference = FALSE</code> (Default): all organisms (reference, representative, and non-representative genomes) are downloaded.
</p>
</li>
<li> <p><code>reference = TRUE</code>: organisms that are downloaded must be either a reference or representative genome. Thus, most genomes which are usually non-reference genomes
will not be downloaded.
</p>
</li></ul>
</td></tr>
<tr><td><code id="meta.retrieval_+3A_combine">combine</code></td>
<td>
<p>just in case <code>type = "assemblystats"</code> is specified, shall
assemby stats of individual species be imported and combined to a
<code><a href="base.html#topic+data.frame">data.frame</a></code>?</p>
</td></tr>
<tr><td><code id="meta.retrieval_+3A_path">path</code></td>
<td>
<p>path to the folder in which downloaded genomes shall be stored.
By default the kingdom name is used to name the output folder.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function aims to perform bulk retrieval of the genomes,
proteomes, cds, etc. of species that belong to the same kingdom of life or
to the same subgroup.
</p>


<h3>Value</h3>

<p>a character vector storing the file paths of the retrieved files.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other meta_retrival: 
<code><a href="#topic+meta.retrieval.all">meta.retrieval.all</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# get all available kingdoms for refseq
getKingdoms(db = "refseq")
# download all vertebrate genomes from refseq
meta.retrieval(kingdom = "vertebrate_mammalian",
               db = "refseq",
               type = "genome")

# get all available kingdoms for genbank
getKingdoms(db = "genbank")
# download all vertebrate genomes from genbank
meta.retrieval(kingdom = "vertebrate_mammalian",
               db = "genbank",
               type = "genome")


# In case users do not wish to retrieve genomes from an entire kingdom,
# but rather from a subgoup (e.g. from species belonging to the
# Gammaproteobacteria class, a subgroup of the bacteria kingdom),
# they can use the following workflow"
# First, users can again consult the getKingdoms() function to retrieve
# kingdom information.
getKingdoms(db = "refseq")

# In this example, we will choose the bacteria kingdom.
# Now, the getGroups() function allows users to obtain available
# subgroups of the bacteria kingdom.
getGroups(db = "refseq", kingdom = "bacteria")

# Now we choose the group Gammaproteobacteria and specify
# the group argument in the meta.retrieval() function
meta.retrieval(kingdom = "bacteria",
   roup = "Gammaproteobacteria",
   db = "refseq",
   type = "genome")

## End(Not run)
</code></pre>

<hr>
<h2 id='meta.retrieval.all'>Perform Meta-Genome Retrieval of all organisms in all kingdoms of life</h2><span id='topic+meta.retrieval.all'></span>

<h3>Description</h3>

<p>Download genomes, proteomes, cds, gff, rna, or assembly stats
files of individual species of all kingdoms of life.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meta.retrieval.all(db = "refseq", type = "genome", reference = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meta.retrieval.all_+3A_db">db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "emsembl"</code>
</p>
</li>
<li> <p><code>db = "ensemblgenomes"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="meta.retrieval.all_+3A_type">type</code></td>
<td>
<p>type of sequences that shall be retrieved. Options are:
</p>

<ul>
<li> <p><code>type = "genome"</code> :
for genome assembly retrieval; see also <code><a href="#topic+getGenome">getGenome</a></code>),
</p>
</li>
<li> <p><code>type = "proteome"</code> :
(for proteome retrieval; see also <code><a href="#topic+getProteome">getProteome</a></code>),
</p>
</li>
<li> <p><code>type = "cds"</code> :
(for coding sequence retrieval; see also <code><a href="#topic+getCDS">getCDS</a></code>),
</p>
</li>
<li> <p><code>type = "gff"</code> :
(for annotation file retrieval in gff format; see also <code><a href="#topic+getGFF">getGFF</a></code>),
</p>
</li>
<li> <p><code>type = "gtf"</code> :
(for annotation file retrieval in gtf format
(only for ensembl and ensemblgenomes); see also <code><a href="#topic+getGTF">getGTF</a></code>),
</p>
</li>
<li> <p><code>type = "rna"</code> :
(for RNA file retrieval in fasta format; see also <code><a href="#topic+getRNA">getRNA</a></code>),
</p>
</li>
<li> <p><code>type = "rm"</code> :
(for Repeat Masker output file retrieval; see also
<code><a href="#topic+getRepeatMasker">getRepeatMasker</a></code>),
</p>
</li>
<li> <p><code>type = "assemblystats"</code> (for genome assembly quality stats
file retrieval; see also <code><a href="#topic+getAssemblyStats">getAssemblyStats</a></code>).
</p>
</li></ul>
</td></tr>
<tr><td><code id="meta.retrieval.all_+3A_reference">reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database
as either a reference genome or a representative genome. Options are:
</p>

<ul>
<li> <p><code>reference = FALSE</code> (Default): all organisms (reference, representative, and non-representative genomes) are downloaded.
</p>
</li>
<li> <p><code>reference = TRUE</code>: organisms that are downloaded must be either a reference or representative genome. Thus, most genomes which are usually non-reference genomes
will not be downloaded.
</p>
</li></ul>
</td></tr>
</table>


<h3>Details</h3>

<p>This function aims to perform bulk retrieval of all genomes
of species for all kingdoms of life.
</p>


<h3>Value</h3>

<p>a character vector storing the file paths of the retrieved files.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other meta_retrival: 
<code><a href="#topic+meta.retrieval">meta.retrieval</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# download all genomes from refseq
meta.retrieval.all(db = "refseq", type = "genome")
# download all vertebrate genomes from genbank
meta.retrieval.all(db = "genbank", type = "genome")
# download all vertebrate genomes from ensemblgenomes
meta.retrieval.all(db = "genbank", type = "ensemblgenomes")

## End(Not run)
</code></pre>

<hr>
<h2 id='organismAttributes'>Retrieve Ensembl Biomart attributes for a query organism</h2><span id='topic+organismAttributes'></span>

<h3>Description</h3>

<p>In addition to the <code><a href="#topic+organismBM">organismBM</a></code> function, 
this function returns all available attributes that can be accessed through 
different marts and datasets for a given query organism.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>organismAttributes(organism, update = FALSE, topic = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="organismAttributes_+3A_organism">organism</code></td>
<td>
<p>a character string specifying the scientific name 
of a query organism.</p>
</td></tr>
<tr><td><code id="organismAttributes_+3A_update">update</code></td>
<td>
<p>a logical value specifying whether or not the local 
listMart.txt, listDatasets.txt, and listAttributes_organism.txt files shall 
be updated by remote access to BioMart.</p>
</td></tr>
<tr><td><code id="organismAttributes_+3A_topic">topic</code></td>
<td>
<p>a character string specifying a topic (category) of attributes, 
e.g. <code>topic</code> = <code>"id"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For a given query organism, this function retrieves all available attributes 
that can be accessed through different marts and datasets.
</p>
<p>Sometimes the same attribute names correspond to different datasets and 
marts causing problems when using <code><a href="#topic+getMarts">getMarts</a></code>. The approach 
introduced by this function provides (again) a organism centric way of 
accessing organism specific attributes.
</p>
<p>The <code>topic</code> argument allows the user to search for specific attribute 
topics/categories for faster filtering.
</p>


<h3>Value</h3>

<p>a data.frame storing corresponding attribute names, description, 
datasets, and marts.
</p>


<h3>Note</h3>

<p>When you run this function for the first time, the data retrieval procedure 
will take some time, due to the remote access to BioMart. The corresponding 
result is then saved in a *.txt file within the <code><a href="base.html#topic+tempdir">tempdir</a></code>
directory named &quot;_biomart/listMarts.txt&quot;,&quot;_biomart/listDatasets.txt&quot;, and 
&quot;_biomart/listAttributes_organism.txt&quot;, allowing subsequent queries to 
perform much faster.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p><a href="http://biomart.org/">http://biomart.org/</a>
</p>
<p>Mapping identifiers for the integration of genomic datasets with the
R/Bioconductor package biomaRt. Steffen Durinck, Paul T. Spellman, Ewan
Birney and Wolfgang Huber, Nature Protocols 4, 1184-1191 (2009).
</p>
<p>BioMart and Bioconductor: a powerful link between biological databases and
microarray data analysis. Steffen Durinck, Yves Moreau, Arek Kasprzyk, Sean
Davis, Bart De Moor, Alvis Brazma and Wolfgang Huber, Bioinformatics 21,
3439-3440 (2005).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+organismFilters">organismFilters</a></code>, <code><a href="#topic+organismBM">organismBM</a></code>, 
<code><a href="#topic+biomart">biomart</a></code>, <code><a href="biomaRt.html#topic+listAttributes">listAttributes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run:  
# search for attribute topic id
head(organismAttributes("Homo sapiens", topic = "id"), 20)

## End(Not run)
</code></pre>

<hr>
<h2 id='organismBM'>Retrieve Ensembl Biomart marts and datasets for a query organism</h2><span id='topic+organismBM'></span>

<h3>Description</h3>

<p>This function returns either all available biomart connections
for all available organisms for which biomart access is possible, or
(when specified) returns all organism specific biomart connections.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>organismBM(organism = NULL, update = FALSE, mute_citation = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="organismBM_+3A_organism">organism</code></td>
<td>
<p>a character string specifying the scientific name of a
query organism. Default is <code>organism</code> = <code>NULL</code>. In this case all
available biomart connections are returned.</p>
</td></tr>
<tr><td><code id="organismBM_+3A_update">update</code></td>
<td>
<p>a logical value specifying whether or not the local
listMart.txt and listDatasets.txt files shall be updated by remote access
to BioMart.</p>
</td></tr>
<tr><td><code id="organismBM_+3A_mute_citation">mute_citation</code></td>
<td>
<p>logical value indicating whether citation message should be muted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function collects all available biomart connections and returns a table
storing the organism for which biomart connections are available as well as
the corresponding mart and database.
</p>


<h3>Note</h3>

<p>When you run this function for the first time, the data retrieval procedure
will take some time, due to the remote access to BioMart. The corresponding
result is then saved in a *.txt file named &quot;_biomart/listDatasets.txt&quot; in the
<code><a href="base.html#topic+tempdir">tempdir</a></code> directory, allowing subsequent queries to perform
much faster.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p><a href="http://biomart.org/">http://biomart.org/</a>
</p>
<p>Mapping identifiers for the integration of genomic datasets with the
R/Bioconductor package biomaRt. Steffen Durinck, Paul T. Spellman, Ewan
Birney and Wolfgang Huber, Nature Protocols 4, 1184-1191 (2009).
</p>
<p>BioMart and Bioconductor: a powerful link between biological databases and
microarray data analysis. Steffen Durinck, Yves Moreau, Arek Kasprzyk, Sean
Davis, Bart De Moor, Alvis Brazma and Wolfgang Huber, Bioinformatics 21,
3439-3440 (2005).
</p>


<h3>See Also</h3>

<p>Other biomaRt: 
<code><a href="#topic+biomart">biomart</a>()</code>,
<code><a href="#topic+getAttributes">getAttributes</a>()</code>,
<code><a href="#topic+getDatasets">getDatasets</a>()</code>,
<code><a href="#topic+getMarts">getMarts</a>()</code>,
<code><a href="#topic+organismFilters">organismFilters</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# returning all available biomart connections
head(organismBM(), 20)
# retrieving all available datasets and biomart connections for
# a specific query organism (scientific name)
organismBM(organism = "Homo sapiens")
# you can also update the downloaded version using
# the "update = TRUE" argument
head(organismBM(update = TRUE), 20)

## End(Not run)
</code></pre>

<hr>
<h2 id='organismFilters'>Retrieve Ensembl Biomart filters for a query organism</h2><span id='topic+organismFilters'></span>

<h3>Description</h3>

<p>In addition to the <code><a href="#topic+organismBM">organismBM</a></code> and
<code><a href="#topic+organismAttributes">organismAttributes</a></code> functions, this function
returns all available filters that can be accessed through different marts
and datasets for a given query organism.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>organismFilters(organism, update = FALSE, topic = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="organismFilters_+3A_organism">organism</code></td>
<td>
<p>a character string specifying the scientific name of a
query organism.</p>
</td></tr>
<tr><td><code id="organismFilters_+3A_update">update</code></td>
<td>
<p>a logical value specifying whether or not the local
listMart.txt, listDatasets.txt, and listFilters_organism.txt files shall be
updated by remote access to BioMart.</p>
</td></tr>
<tr><td><code id="organismFilters_+3A_topic">topic</code></td>
<td>
<p>a character string specifying a topic (category) of filters,
e.g. <code>topic</code> = <code>"id"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For a given query organism, this function retrieves all available
filters that can be accessed through different marts and datasets.
</p>
<p>Sometimes the same filter names correspond to different datasets and
marts causing problems when using <code><a href="#topic+getMarts">getMarts</a></code>.
The approach introduced by this function provides (again) a organism centric
way of accessing organism specific filters.
</p>
<p>The <code>topic</code> argument allows the user to search for specific filters
topics/categories for faster selection.
</p>


<h3>Value</h3>

<p>a data.frame storing corresponding filter names, description,
datasets, and marts.
</p>


<h3>Note</h3>

<p>When you run this function for the first time, the data retrieval procedure
will take some time, due to the remote access to BioMart. The corresponding
result is then saved in a *.txt file within the <code><a href="base.html#topic+tempdir">tempdir</a></code>
directory  named &quot;_biomart/listMarts.txt&quot;,&quot;_biomart/listDatasets.txt&quot;, and
&quot;_biomart/listFilters_organism.txt&quot;, allowing subsequent queries to perform
much faster.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p><a href="http://biomart.org/">http://biomart.org/</a>
</p>
<p>Mapping identifiers for the integration of genomic datasets with the
R/Bioconductor package biomaRt. Steffen Durinck, Paul T. Spellman, Ewan
Birney and Wolfgang Huber, Nature Protocols 4, 1184-1191 (2009).
</p>
<p>BioMart and Bioconductor: a powerful link between biological databases and
microarray data analysis. Steffen Durinck, Yves Moreau, Arek Kasprzyk, Sean
Davis, Bart De Moor, Alvis Brazma and Wolfgang Huber, Bioinformatics 21,
3439-3440 (2005).
</p>


<h3>See Also</h3>

<p>Other biomaRt: 
<code><a href="#topic+biomart">biomart</a>()</code>,
<code><a href="#topic+getAttributes">getAttributes</a>()</code>,
<code><a href="#topic+getDatasets">getDatasets</a>()</code>,
<code><a href="#topic+getMarts">getMarts</a>()</code>,
<code><a href="#topic+organismBM">organismBM</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# search for filter topic "id"
head(organismFilters("Homo sapiens", topic = "id"), 20)

## End(Not run)
</code></pre>

<hr>
<h2 id='read_assemblystats'>Import Genome Assembly Stats File</h2><span id='topic+read_assemblystats'></span>

<h3>Description</h3>

<p>This function reads an organism specific Genome Assembly
Stats file that was retrieved with <code><a href="#topic+getAssemblyStats">getAssemblyStats</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_assemblystats(file, type = "raw")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_assemblystats_+3A_file">file</code></td>
<td>
<p>a character string specifying the path to the file storing
the Genome Assembly Stats file.</p>
</td></tr>
<tr><td><code id="read_assemblystats_+3A_type">type</code></td>
<td>
<p>a tibble object, either <code>type = "raw"</code> to import the entire genome assembly
stats file or <code>type = "stats"</code> to import overall statistics including
all chromosomes, mitochondria and plastids.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a string specifying the path to the Genome
Assembly Stats file of interest (e.g. the path returned by
<code><a href="#topic+getAssemblyStats">getAssemblyStats</a></code>) and imports it.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getAssemblyStats">getAssemblyStats</a></code>, <code><a href="#topic+read_genome">read_genome</a></code>,
<code><a href="#topic+read_proteome">read_proteome</a></code>, <code><a href="#topic+read_cds">read_cds</a></code>, <code><a href="#topic+read_gff">read_gff</a></code>
</p>

<hr>
<h2 id='read_cds'>Import CDS as Biostrings or data.table object</h2><span id='topic+read_cds'></span>

<h3>Description</h3>

<p>This function reads an organism specific CDS stored in a
defined file format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_cds(
  file,
  format = "fasta",
  obj.type = "Biostrings",
  delete_corrupt = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_cds_+3A_file">file</code></td>
<td>
<p>a character string specifying the path to the file storing
the CDS.</p>
</td></tr>
<tr><td><code id="read_cds_+3A_format">format</code></td>
<td>
<p>a character string specifying the file format used to store
the genome, e.g. <code>format = "fasta"</code> (default) or <code>format = "gbk"</code>.</p>
</td></tr>
<tr><td><code id="read_cds_+3A_obj.type">obj.type</code></td>
<td>
<p>a character string specifying the object stype in which the
genomic sequence shall be represented.
Either as <code>obj.type = "Biostrings"</code> (default) or as
<code>obj.type = "data.table"</code>.</p>
</td></tr>
<tr><td><code id="read_cds_+3A_delete_corrupt">delete_corrupt</code></td>
<td>
<p>a logical value specifying whether potential CDS
sequences that cannot be divided by 3 shall be
be excluded from the the dataset. Default is <code>delete_corrupt = FALSE</code>.</p>
</td></tr>
<tr><td><code id="read_cds_+3A_...">...</code></td>
<td>
<p>additional arguments that are used by
<code><a href="seqinr.html#topic+read.fasta">read.fasta</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>read.cds</code> function takes a string specifying the path
to the cds file of interest as first argument.
</p>
<p>It is possible to read in different proteome file standards such as
<em>fasta</em> or <em>genebank</em>.
</p>
<p>CDS stored in fasta files can be downloaded from
http://www.ensembl.org/info/data/ftp/index.html.
</p>


<h3>Value</h3>

<p>A data.table storing the gene id in the first column and the
corresponding sequence as string in the second column.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other cds: 
<code><a href="#topic+getCDSSet">getCDSSet</a>()</code>,
<code><a href="#topic+getCDS">getCDS</a>()</code>
</p>
<p>Other readers: 
<code><a href="#topic+read_genome">read_genome</a>()</code>,
<code><a href="#topic+read_gff">read_gff</a>()</code>,
<code><a href="#topic+read_proteome">read_proteome</a>()</code>,
<code><a href="#topic+read_rna">read_rna</a>()</code>
</p>

<hr>
<h2 id='read_genome'>Import Genome Assembly as Biostrings or data.table object</h2><span id='topic+read_genome'></span>

<h3>Description</h3>

<p>This function reads an organism specific genome stored in a
defined file format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_genome(file, format = "fasta", obj.type = "Biostrings", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_genome_+3A_file">file</code></td>
<td>
<p>a character string specifying the path to the file
storing the genome.</p>
</td></tr>
<tr><td><code id="read_genome_+3A_format">format</code></td>
<td>
<p>a character string specifying the file format used to store
the genome, e.g. <code>format = "fasta"</code> (default) or <code>format = "gbk"</code>.</p>
</td></tr>
<tr><td><code id="read_genome_+3A_obj.type">obj.type</code></td>
<td>
<p>a character string specifying the object stype in which
the genomic sequence shall be represented. Either as
<code>obj.type = "Biostrings"</code> (default) or
as <code>obj.type = "data.table"</code>.</p>
</td></tr>
<tr><td><code id="read_genome_+3A_...">...</code></td>
<td>
<p>additional arguments that are used by the
<code><a href="seqinr.html#topic+read.fasta">read.fasta</a></code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a string specifying the path to the genome file
of interest as first argument (e.g. the path returned by
<code><a href="#topic+getGenome">getGenome</a></code>).
</p>


<h3>Value</h3>

<p>Either a <code>Biostrings</code> or <code>data.table</code> object.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other genome: 
<code><a href="#topic+getGenomeSet">getGenomeSet</a>()</code>,
<code><a href="#topic+getGenome">getGenome</a>()</code>
</p>
<p>Other readers: 
<code><a href="#topic+read_cds">read_cds</a>()</code>,
<code><a href="#topic+read_gff">read_gff</a>()</code>,
<code><a href="#topic+read_proteome">read_proteome</a>()</code>,
<code><a href="#topic+read_rna">read_rna</a>()</code>
</p>

<hr>
<h2 id='read_gff'>Import GFF File</h2><span id='topic+read_gff'></span>

<h3>Description</h3>

<p>This function reads an organism specific CDS stored
in a defined file format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_gff(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_gff_+3A_file">file</code></td>
<td>
<p>a character string specifying the path to the file
storing the CDS.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a string specifying the path to the GFF file
of interest (e.g. the path returned by <code><a href="#topic+getGFF">getGFF</a></code>).
</p>


<h3>Value</h3>

<p>Either a <code>Biostrings</code> or <code>data.table</code> object.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other gff: 
<code><a href="#topic+getGFFSet">getGFFSet</a>()</code>,
<code><a href="#topic+getGFF">getGFF</a>()</code>
</p>
<p>Other readers: 
<code><a href="#topic+read_cds">read_cds</a>()</code>,
<code><a href="#topic+read_genome">read_genome</a>()</code>,
<code><a href="#topic+read_proteome">read_proteome</a>()</code>,
<code><a href="#topic+read_rna">read_rna</a>()</code>
</p>

<hr>
<h2 id='read_proteome'>Import Proteome as Biostrings or data.table object</h2><span id='topic+read_proteome'></span>

<h3>Description</h3>

<p>This function reads an organism specific proteome stored in a
defined file format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_proteome(file, format = "fasta", obj.type = "Biostrings", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_proteome_+3A_file">file</code></td>
<td>
<p>a character string specifying the path to the file storing
the proteome.</p>
</td></tr>
<tr><td><code id="read_proteome_+3A_format">format</code></td>
<td>
<p>a character string specifying the file format used to store the
genome, e.g. <code>format = "fasta"</code> (default) or <code>format = "gbk"</code>.</p>
</td></tr>
<tr><td><code id="read_proteome_+3A_obj.type">obj.type</code></td>
<td>
<p>a character string specifying the object stype in which the
genomic sequence shall be represented.
Either as <code>obj.type = "Biostrings"</code> (default) or as
<code>obj.type = "data.table"</code>.</p>
</td></tr>
<tr><td><code id="read_proteome_+3A_...">...</code></td>
<td>
<p>additional arguments that are used by
<code><a href="seqinr.html#topic+read.fasta">read.fasta</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a string specifying the path to the
proteome file of interest as first argument.
</p>
<p>It is possible to read in different proteome file standards such as
<em>fasta</em> or <em>genebank</em>.
</p>


<h3>Value</h3>

<p>Either a <code>Biostrings</code> or <code>data.table</code> object.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other readers: 
<code><a href="#topic+read_cds">read_cds</a>()</code>,
<code><a href="#topic+read_genome">read_genome</a>()</code>,
<code><a href="#topic+read_gff">read_gff</a>()</code>,
<code><a href="#topic+read_rna">read_rna</a>()</code>
</p>
<p>Other proteome: 
<code><a href="#topic+getProteomeSet">getProteomeSet</a>()</code>,
<code><a href="#topic+getProteome">getProteome</a>()</code>
</p>

<hr>
<h2 id='read_rm'>Import Repeat Masker output file</h2><span id='topic+read_rm'></span>

<h3>Description</h3>

<p>This function reads an organism specific 
Repeat Masker output file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_rm(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_rm_+3A_file">file</code></td>
<td>
<p>a character string specifying the path to the file 
storing the Repeat Masker output (e.g. retrieved with <code><a href="#topic+getRepeatMasker">getRepeatMasker</a></code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a string specifying the path to the 
Repeat Masker output file of interest as first argument.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getRepeatMasker">getRepeatMasker</a></code>, <code><a href="#topic+read_genome">read_genome</a></code>, 
<code><a href="#topic+read_proteome">read_proteome</a></code>, <code><a href="#topic+read_gff">read_gff</a></code>, <code><a href="#topic+read_rna">read_rna</a></code>
</p>

<hr>
<h2 id='read_rna'>Import RNA as Biostrings or data.table object</h2><span id='topic+read_rna'></span>

<h3>Description</h3>

<p>This function reads an organism specific RNA stored in a
defined file format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_rna(file, format = "fasta", obj.type = "Biostrings", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_rna_+3A_file">file</code></td>
<td>
<p>a character string specifying the path to the file
storing the RNA.</p>
</td></tr>
<tr><td><code id="read_rna_+3A_format">format</code></td>
<td>
<p>a character string specifying the file format used to store the
genome, e.g. <code>format = "fasta"</code> (default) or <code>format = "gbk"</code>.</p>
</td></tr>
<tr><td><code id="read_rna_+3A_obj.type">obj.type</code></td>
<td>
<p>a character string specifying the object stype in which the
genomic sequence shall be represented.
Either as <code>obj.type = "Biostrings"</code> (default) or
as <code>obj.type = "data.table"</code>.</p>
</td></tr>
<tr><td><code id="read_rna_+3A_...">...</code></td>
<td>
<p>additional arguments that are used by
<code><a href="seqinr.html#topic+read.fasta">read.fasta</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a string specifying the path to the RNA file
of interest as first argument. It is possible to read in different proteome
file standards such as <em>fasta</em> or <em>genebank</em>.
</p>


<h3>Value</h3>

<p>A data.table storing the gene id in the first column and the
corresponding sequence as string in the second column.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other rna: 
<code><a href="#topic+getRNASet">getRNASet</a>()</code>,
<code><a href="#topic+getRNA">getRNA</a>()</code>
</p>
<p>Other readers: 
<code><a href="#topic+read_cds">read_cds</a>()</code>,
<code><a href="#topic+read_genome">read_genome</a>()</code>,
<code><a href="#topic+read_gff">read_gff</a>()</code>,
<code><a href="#topic+read_proteome">read_proteome</a>()</code>
</p>

<hr>
<h2 id='refseqOrganisms'>Retrieve All Organism Names Stored on refseq</h2><span id='topic+refseqOrganisms'></span>

<h3>Description</h3>

<p>This function extracts all organism names (scientific names)
for which genomes, proteomes, and CDS files are stored on 
the NCBI refseq server.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refseqOrganisms()
</code></pre>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>

<hr>
<h2 id='summary_cds'>Retrieve summary statistics for a coding sequence (CDS) file</h2><span id='topic+summary_cds'></span>

<h3>Description</h3>

<p>A summary statistics of specific CDS features is returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summary_cds(file, organism)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary_cds_+3A_file">file</code></td>
<td>
<p>file path to a CDS file in <code>fasta</code> format.</p>
</td></tr>
<tr><td><code id="summary_cds_+3A_organism">organism</code></td>
<td>
<p>character string specifying the organism at hand.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The summary statistics include:
</p>

<ul>
<li> <p><code>total_seqs</code>: 
</p>
</li>
<li> <p><code>nnn_abs</code>: The total number of NNN's 
(over all chromosomes/scaffolds/contigs) in all coding sequences combined
</p>
</li>
<li> <p><code>nnn_perc</code>: The percentage (relative frequency) of NNN's 
(over all chromosomes/scaffolds/contigs) compared to the total number of 
nucleotides of all coding sequences
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getCollection">getCollection</a></code>, <code><a href="#topic+getCDS">getCDS</a></code>, <code><a href="#topic+read_cds">read_cds</a></code>, <code><a href="#topic+summary_genome">summary_genome</a></code>
</p>

<hr>
<h2 id='summary_genome'>Retrieve summary statistics for a genome assembly file</h2><span id='topic+summary_genome'></span>

<h3>Description</h3>

<p>A summary statistics of specific genome features is generated.
These statistics are useful to assess the genome quality of retrieved genome assemblies
when performing comparative genomics tasks. This way, users can assess whether or not
patterns found based on genome comparisons aren't just a technical artifact of
differences in genome assembly quality.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summary_genome(file, organism)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary_genome_+3A_file">file</code></td>
<td>
<p>file path to a genome assembly file in <code>fasta</code> format.</p>
</td></tr>
<tr><td><code id="summary_genome_+3A_organism">organism</code></td>
<td>
<p>character string specifying the organism at hand.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The summary statistics include:
</p>

<ul>
<li> <p><code>genome_size_mbp</code>: Genome size in mega base pairs
</p>
</li>
<li> <p><code>n50_mbp</code>: The N50 contig size of the genome assembly in mega base pairs
</p>
</li>
<li> <p><code>n_seqs</code>: The number of chromosomes/scaffolds/contigs of the genome assembly file
</p>
</li>
<li> <p><code>n_nnn</code>: The absolute number of NNNs (over all chromosomes or scaffolds or contigs) in the genome assembly file
</p>
</li>
<li> <p><code>rel_nnn</code>: The percentage (relative frequency) of NNNs (over all chromosomes or scaffolds or contigs) compared to the total number of
nucleotides in the genome assembly file
</p>
</li>
<li> <p><code>genome_entropy</code>: The <code>Shannon Entropy</code> of the genome assembly file (median entropy over all individual chromosome entropies)
</p>
</li>
<li> <p><code>n_gc</code>: The total number of GCs
(over all chromosomes or scaffolds or contigs) in the genome assembly file
</p>
</li>
<li> <p><code>rel_gc</code>: The (relative frequency) of GCs
(over all chromosomes or scaffolds or contigs) compared to the total number of
nucleotides in the genome assembly file
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summary_cds">summary_cds</a></code>, <code><a href="#topic+getCollection">getCollection</a></code>, <code><a href="#topic+getGenome">getGenome</a></code>, <code><a href="#topic+read_genome">read_genome</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# retrieve genome from NCBI RefSeq
Sc &lt;- biomartr::getGenome(db = "refseq", organism = "Saccharomyces cerevisiae")
# compute genome assembly summary statistics
Sc_genome_summary &lt;- summary_genome(file = Sc, organism = "Saccharomyces cerevisiae")
# look at results
Sc_genome_summary

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
