<!DOCTYPE html><html><head><title>Help for package BioM2</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BioM2}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AddUnmapped'><p>Add unmapped probe</p></a></li>
<li><a href='#baseModel'><p>Prediction by Machine Learning</p></a></li>
<li><a href='#BioM2'><p>Biologically Explainable Machine Learning Framework</p></a></li>
<li><a href='#FindParaModule'><p>Find suitable parameters for partitioning pathways modules</p></a></li>
<li><a href='#GO_Ancestor'><p>Pathways in the GO database and their Ancestor</p></a></li>
<li><a href='#GO_Ancestor_exact'><p>Pathways in the GO database and their Ancestor</p></a></li>
<li><a href='#GO2ALLEGS_BP'><p>An example about pathlistDB</p></a></li>
<li><a href='#HybaseModel'><p>Selection of the optimal base model</p></a></li>
<li><a href='#MethylAnno'><p>An example about FeatureAnno for methylation data</p></a></li>
<li><a href='#MethylData_Test'><p>An example about TrainData/TestData for methylation data</p></a></li>
<li><a href='#PathwaysModule'><p>Delineate differential pathway modules with high biological interpretability</p></a></li>
<li><a href='#PlotCorModule'><p>Correlalogram for Biological Differences Modules</p></a></li>
<li><a href='#PlotPathFearture'><p>Visualisation of significant pathway-level features</p></a></li>
<li><a href='#PlotPathInner'><p>Visualisation Original features that make up the pathway</p></a></li>
<li><a href='#PlotPathNet'><p>Network diagram of pathways-level features</p></a></li>
<li><a href='#ShowModule'><p>Display biological information within each pathway module</p></a></li>
<li><a href='#Stage1_FeartureSelection'><p>Stage 1 Fearture Selection</p></a></li>
<li><a href='#Stage2_FeartureSelection'><p>Stage 2 Fearture Selection</p></a></li>
<li><a href='#TransAnno'><p>An example about FeatureAnno for gene expression</p></a></li>
<li><a href='#TransData_Test'><p>An example about TrainData/TestData for gene expression</p></a></li>
<li><a href='#VisMultiModule'><p>Visualisation of the results of the analysis of the pathway modules</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Biologically Explainable Machine Learning Framework</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.7</td>
</tr>
<tr>
<td>Author:</td>
<td>Shunjie Zhang and Junfang Chen</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Shunjie Zhang &lt;zhang.shunjie@qq.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Biologically Explainable Machine Learning Framework for Phenotype Prediction using omics data described in Chen and Schwarz (2017) &lt;<a href="https://doi.org/10.48550%2FarXiv.1712.00336">doi:10.48550/arXiv.1712.00336</a>&gt;.Identifying reproducible and interpretable biological patterns from high-dimensional omics data is a critical factor in understanding the risk mechanism of complex disease. As such, explainable machine learning can offer biological insight in addition to personalized risk scoring.In this process, a feature space of biological pathways will be generated, and the feature space can also be subsequently analyzed using WGCNA (Described in Horvath and Zhang (2005) &lt;<a href="https://doi.org/10.2202%2F1544-6115.1128">doi:10.2202/1544-6115.1128</a>&gt; and Langfelder and Horvath (2008) &lt;<a href="https://doi.org/10.1186%2F1471-2105-9-559">doi:10.1186/1471-2105-9-559</a>&gt; ) methods.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>WGCNA, mlr3, CMplot, ggsci, ROCR, caret, ggplot2, ggpubr,
viridis, ggthemes, ggstatsplot, htmlwidgets, jiebaR, mlr3verse,
parallel, uwot, webshot, wordcloud2,ggforce, igraph, ggnetwork</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-06-09 03:45:21 UTC; Shedom</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-06-09 08:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='AddUnmapped'>Add unmapped probe</h2><span id='topic+AddUnmapped'></span>

<h3>Description</h3>

<p>Add unmapped probe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AddUnmapped(
  train = NULL,
  test = NULL,
  Unmapped_num = NULL,
  Add_FeartureSelection_Method = "wilcox.test",
  anno = NULL,
  len = NULL,
  verbose = TRUE,
  cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AddUnmapped_+3A_train">train</code></td>
<td>
<p>The input training dataset. The first column
is the label or the output. For binary classes,
0 and 1 are used to indicate the class member.</p>
</td></tr>
<tr><td><code id="AddUnmapped_+3A_test">test</code></td>
<td>
<p>The input test dataset. The first column
is the label or the output. For binary classes,
0 and 1 are used to indicate the class member.</p>
</td></tr>
<tr><td><code id="AddUnmapped_+3A_unmapped_num">Unmapped_num</code></td>
<td>
<p>The number of unmapped probes.</p>
</td></tr>
<tr><td><code id="AddUnmapped_+3A_add_feartureselection_method">Add_FeartureSelection_Method</code></td>
<td>
<p>Feature selection methods. Available options are
c('cor', 'wilcox.test').</p>
</td></tr>
<tr><td><code id="AddUnmapped_+3A_anno">anno</code></td>
<td>
<p>The annotation data stored in a data.frame for probe
mapping. It must have at least two columns named 'ID' and 'entrezID'.
(For details, please refer to data( data(&quot;MethylAnno&quot;) )</p>
</td></tr>
<tr><td><code id="AddUnmapped_+3A_len">len</code></td>
<td>
<p>The number of unmapped probes</p>
</td></tr>
<tr><td><code id="AddUnmapped_+3A_verbose">verbose</code></td>
<td>
<p>Whether to print running process information to the console</p>
</td></tr>
<tr><td><code id="AddUnmapped_+3A_cores">cores</code></td>
<td>
<p>The number of cores used for computation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of unmapped probes
</p>

<hr>
<h2 id='baseModel'>Prediction by Machine Learning</h2><span id='topic+baseModel'></span>

<h3>Description</h3>

<p>Prediction by Machine Learning with different learners ( From 'mlr3' )
</p>


<h3>Usage</h3>

<pre><code class='language-R'>baseModel(
  trainData,
  testData,
  predMode = "probability",
  classifier,
  paramlist = NULL,
  inner_folds = 10
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="baseModel_+3A_traindata">trainData</code></td>
<td>
<p>The input training dataset. The first column
is the label or the output. For binary classes,
0 and 1 are used to indicate the class member.</p>
</td></tr>
<tr><td><code id="baseModel_+3A_testdata">testData</code></td>
<td>
<p>The input test dataset. The first column
is the label or the output. For binary classes,
0 and 1 are used to indicate the class member.</p>
</td></tr>
<tr><td><code id="baseModel_+3A_predmode">predMode</code></td>
<td>
<p>The prediction mode. Available options are
c('probability', 'classification').</p>
</td></tr>
<tr><td><code id="baseModel_+3A_classifier">classifier</code></td>
<td>
<p>Learners in mlr3</p>
</td></tr>
<tr><td><code id="baseModel_+3A_paramlist">paramlist</code></td>
<td>
<p>Learner parameters</p>
</td></tr>
<tr><td><code id="baseModel_+3A_inner_folds">inner_folds</code></td>
<td>
<p>k-fold cross validation ( Only supported when testData = NULL )</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The predicted output for the test data.
</p>


<h3>Author(s)</h3>

<p>Shunjie Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(mlr3verse)
library(caret)
library(BioM2)
data=MethylData_Test
set.seed(1)
part=unlist(createDataPartition(data$label,p=0.8))#Split data
predict=baseModel(trainData=data[part,1:10],
                 testData=data[-part,1:10],
                 classifier = 'svm')#Use 10 features to make predictions,Learner uses svm


</code></pre>

<hr>
<h2 id='BioM2'>Biologically Explainable Machine Learning Framework</h2><span id='topic+BioM2'></span>

<h3>Description</h3>

<p>Biologically Explainable Machine Learning Framework
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BioM2(
  TrainData = NULL,
  TestData = NULL,
  pathlistDB = NULL,
  FeatureAnno = NULL,
  resampling = NULL,
  nfolds = 5,
  classifier = "liblinear",
  predMode = "probability",
  PathwaySizeUp = 200,
  PathwaySizeDown = 20,
  MinfeatureNum_pathways = 10,
  Add_UnMapped = TRUE,
  Unmapped_num = 300,
  Add_FeartureSelection_Method = "wilcox.test",
  Inner_CV = TRUE,
  inner_folds = 10,
  Stage1_FeartureSelection_Method = "cor",
  cutoff = 0.3,
  Stage2_FeartureSelection_Method = "RemoveHighcor",
  cutoff2 = 0.95,
  classifier2 = NULL,
  target = "predict",
  p.adjust.method = "fdr",
  save_pathways_matrix = FALSE,
  cores = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BioM2_+3A_traindata">TrainData</code></td>
<td>
<p>The input training dataset. The first column
is the label or the output. For binary classes,
0 and 1 are used to indicate the class member.</p>
</td></tr>
<tr><td><code id="BioM2_+3A_testdata">TestData</code></td>
<td>
<p>The input test dataset. The first column
is the label or the output. For binary classes,
0 and 1 are used to indicate the class member.</p>
</td></tr>
<tr><td><code id="BioM2_+3A_pathlistdb">pathlistDB</code></td>
<td>
<p>A list of pathways with pathway IDs and their
corresponding genes ('entrezID' is used).
For details, please refer to ( data(&quot;GO2ALLEGS_BP&quot;) )</p>
</td></tr>
<tr><td><code id="BioM2_+3A_featureanno">FeatureAnno</code></td>
<td>
<p>The annotation data stored in a data.frame for probe
mapping. It must have at least two columns named 'ID' and 'entrezID'.
(For details, please refer to data( data(&quot;MethylAnno&quot;) )</p>
</td></tr>
<tr><td><code id="BioM2_+3A_resampling">resampling</code></td>
<td>
<p>Resampling in mlr3verse.</p>
</td></tr>
<tr><td><code id="BioM2_+3A_nfolds">nfolds</code></td>
<td>
<p>k-fold cross validation ( Only supported when TestData = NULL )</p>
</td></tr>
<tr><td><code id="BioM2_+3A_classifier">classifier</code></td>
<td>
<p>Learners in mlr3</p>
</td></tr>
<tr><td><code id="BioM2_+3A_predmode">predMode</code></td>
<td>
<p>The prediction mode. Available options are
c('probability', 'classification').</p>
</td></tr>
<tr><td><code id="BioM2_+3A_pathwaysizeup">PathwaySizeUp</code></td>
<td>
<p>The upper-bound of the number of genes in each
biological pathways.</p>
</td></tr>
<tr><td><code id="BioM2_+3A_pathwaysizedown">PathwaySizeDown</code></td>
<td>
<p>The lower-bound of the number of genes in each
biological pathways.</p>
</td></tr>
<tr><td><code id="BioM2_+3A_minfeaturenum_pathways">MinfeatureNum_pathways</code></td>
<td>
<p>The minimal defined pathway size after mapping your
own data to pathlistDB(KEGG database/GO database).</p>
</td></tr>
<tr><td><code id="BioM2_+3A_add_unmapped">Add_UnMapped</code></td>
<td>
<p>Whether to add unmapped probes for prediction</p>
</td></tr>
<tr><td><code id="BioM2_+3A_unmapped_num">Unmapped_num</code></td>
<td>
<p>The number of unmapped probes</p>
</td></tr>
<tr><td><code id="BioM2_+3A_add_feartureselection_method">Add_FeartureSelection_Method</code></td>
<td>
<p>Feature selection methods.</p>
</td></tr>
<tr><td><code id="BioM2_+3A_inner_cv">Inner_CV</code></td>
<td>
<p>Whether to perform a k-fold verification on the training set.</p>
</td></tr>
<tr><td><code id="BioM2_+3A_inner_folds">inner_folds</code></td>
<td>
<p>k-fold verification on the training set.</p>
</td></tr>
<tr><td><code id="BioM2_+3A_stage1_feartureselection_method">Stage1_FeartureSelection_Method</code></td>
<td>
<p>Feature selection methods.</p>
</td></tr>
<tr><td><code id="BioM2_+3A_cutoff">cutoff</code></td>
<td>
<p>The cutoff used for feature selection threshold. It can be any value
between 0 and 1.</p>
</td></tr>
<tr><td><code id="BioM2_+3A_stage2_feartureselection_method">Stage2_FeartureSelection_Method</code></td>
<td>
<p>Feature selection methods.</p>
</td></tr>
<tr><td><code id="BioM2_+3A_cutoff2">cutoff2</code></td>
<td>
<p>The cutoff used for feature selection threshold. It can be any value
between 0 and 1.</p>
</td></tr>
<tr><td><code id="BioM2_+3A_classifier2">classifier2</code></td>
<td>
<p>Learner for stage 2 prediction(if classifier2==NULL,then it is the same as the learner in stage 1.)</p>
</td></tr>
<tr><td><code id="BioM2_+3A_target">target</code></td>
<td>
<p>Is it used to predict or explore potential biological mechanisms?
Available options are c('predict', 'pathways').</p>
</td></tr>
<tr><td><code id="BioM2_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>p-value adjustment method.(holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;,</p>
</td></tr>
<tr><td><code id="BioM2_+3A_save_pathways_matrix">save_pathways_matrix</code></td>
<td>
<p>Whether to output the path matrix file</p>
</td></tr>
<tr><td><code id="BioM2_+3A_cores">cores</code></td>
<td>
<p>The number of cores used for computation.</p>
</td></tr>
<tr><td><code id="BioM2_+3A_verbose">verbose</code></td>
<td>
<p>Whether to print running process information to the console</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing prediction results and prediction result evaluation
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


library(mlr3verse)
library(caret)
library(parallel)
library(BioM2)
data=MethylData_Test
set.seed(1)
part=unlist(createDataPartition(data$label,p=0.8))
Train=data[part,]
Test=data[-part,]
pathlistDB=GO2ALLEGS_BP
FeatureAnno=MethylAnno


pred=BioM2(TrainData = Train,TestData = Test,
           pathlistDB=pathlistDB,FeatureAnno=FeatureAnno,
           classifier='svm',nfolds=5,
           PathwaySizeUp=25,PathwaySizeDown=20,MinfeatureNum_pathways=10,
           Add_UnMapped='Yes',Unmapped_num=300,
           Inner_CV='None',inner_folds=5,
           Stage1_FeartureSelection_Method='cor',cutoff=0.3,
           Stage2_FeartureSelection_Method='None',
           target='predict',cores=1
)#(To explore biological mechanisms, set target=‘pathways’)



</code></pre>

<hr>
<h2 id='FindParaModule'>Find suitable parameters for partitioning pathways modules</h2><span id='topic+FindParaModule'></span>

<h3>Description</h3>

<p>Find suitable parameters for partitioning pathways modules
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FindParaModule(
  pathways_matrix = NULL,
  control_label = NULL,
  minModuleSize = seq(10, 20, 5),
  mergeCutHeight = seq(0, 0.3, 0.1),
  minModuleNum = 20,
  power = NULL,
  exact = TRUE,
  ancestor_anno = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FindParaModule_+3A_pathways_matrix">pathways_matrix</code></td>
<td>
<p>A pathway matrix generated by the BioM2( target='pathways') function.</p>
</td></tr>
<tr><td><code id="FindParaModule_+3A_control_label">control_label</code></td>
<td>
<p>The label of the control group ( A single number, factor, or character )</p>
</td></tr>
<tr><td><code id="FindParaModule_+3A_minmodulesize">minModuleSize</code></td>
<td>
<p>minimum module size for module detection. Detail for WGCNA::blockwiseModules()</p>
</td></tr>
<tr><td><code id="FindParaModule_+3A_mergecutheight">mergeCutHeight</code></td>
<td>
<p>dendrogram cut height for module merging. Detail for WGCNA::blockwiseModules()</p>
</td></tr>
<tr><td><code id="FindParaModule_+3A_minmodulenum">minModuleNum</code></td>
<td>
<p>Minimum total number of modules detected</p>
</td></tr>
<tr><td><code id="FindParaModule_+3A_power">power</code></td>
<td>
<p>soft-thresholding power for network construction. Detail for WGCNA::blockwiseModules()</p>
</td></tr>
<tr><td><code id="FindParaModule_+3A_exact">exact</code></td>
<td>
<p>Whether to divide GO pathways more accurately (work when ancestor_anno=NULL)</p>
</td></tr>
<tr><td><code id="FindParaModule_+3A_ancestor_anno">ancestor_anno</code></td>
<td>
<p>Annotations for ancestral relationships (like data('GO_Ancestor') )</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing recommended parameters
</p>

<hr>
<h2 id='GO_Ancestor'>Pathways in the GO database and their Ancestor</h2><span id='topic+GO_Ancestor'></span>

<h3>Description</h3>

<p>Inclusion relationships between pathways
</p>


<h3>Format</h3>

<p>A data frame :
</p>

<p>...

</p>


<h3>Details</h3>

<p>In the GO database, each pathway will have its own ancestor pathway.
Map pathways in GO database to about 20 common ancestor pathways.
</p>


<h3>Source</h3>

<p>From GO.db
</p>

<hr>
<h2 id='GO_Ancestor_exact'>Pathways in the GO database and their Ancestor</h2><span id='topic+GO_Ancestor_exact'></span>

<h3>Description</h3>

<p>Inclusion relationships between pathways
</p>


<h3>Format</h3>

<p>A data frame :
</p>

<p>...

</p>


<h3>Details</h3>

<p>In the GO database, each pathway will have its own ancestor pathway.
Map pathways in GO database to about 400 common ancestor pathways.
</p>


<h3>Source</h3>

<p>From GO.db
</p>

<hr>
<h2 id='GO2ALLEGS_BP'>An example about pathlistDB</h2><span id='topic+GO2ALLEGS_BP'></span>

<h3>Description</h3>

<p>An example about pathlistDB
</p>


<h3>Format</h3>

<p>A list :
</p>

<p>...

</p>


<h3>Details</h3>

<p>A list of pathways with pathway IDs and their
corresponding genes ('entrezID' is used).
</p>

<hr>
<h2 id='HybaseModel'>Selection of the optimal base model</h2><span id='topic+HybaseModel'></span>

<h3>Description</h3>

<p>Selection of the optimal base model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HybaseModel(
  data = NULL,
  pathlistDB = NULL,
  FeatureAnno = NULL,
  resampling = NULL,
  nfolds = 5,
  classifiers = "liblinear",
  predMode = "probability",
  PathwaySizeUp = 200,
  PathwaySizeDown = 20,
  MinfeatureNum_pathways = 10,
  Add_UnMapped = TRUE,
  Unmapped_num = 300,
  Add_FeartureSelection_Method = "wilcox.test",
  Inner_CV = TRUE,
  inner_folds = 10,
  Stage1_FeartureSelection_Method = "cor",
  cutoff = 0.3,
  Stage2_FeartureSelection_Method = "RemoveHighcor",
  cutoff2 = 0.95,
  cores = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HybaseModel_+3A_data">data</code></td>
<td>
<p>The input training dataset. The first column
is the label or the output. For binary classes,
0 and 1 are used to indicate the class member.</p>
</td></tr>
<tr><td><code id="HybaseModel_+3A_pathlistdb">pathlistDB</code></td>
<td>
<p>A list of pathways with pathway IDs and their
corresponding genes ('entrezID' is used).
For details, please refer to ( data(&quot;GO2ALLEGS_BP&quot;) )</p>
</td></tr>
<tr><td><code id="HybaseModel_+3A_featureanno">FeatureAnno</code></td>
<td>
<p>The annotation data stored in a data.frame for probe
mapping. It must have at least two columns named 'ID' and 'entrezID'.
(For details, please refer to data( data(&quot;MethylAnno&quot;) )</p>
</td></tr>
<tr><td><code id="HybaseModel_+3A_resampling">resampling</code></td>
<td>
<p>Resampling in mlr3verse.</p>
</td></tr>
<tr><td><code id="HybaseModel_+3A_nfolds">nfolds</code></td>
<td>
<p>k-fold cross validation</p>
</td></tr>
<tr><td><code id="HybaseModel_+3A_classifiers">classifiers</code></td>
<td>
<p>A string of character vectors(Learners in mlr3)</p>
</td></tr>
<tr><td><code id="HybaseModel_+3A_predmode">predMode</code></td>
<td>
<p>The prediction mode. Available options are
c('probability', 'classification').</p>
</td></tr>
<tr><td><code id="HybaseModel_+3A_pathwaysizeup">PathwaySizeUp</code></td>
<td>
<p>The upper-bound of the number of genes in each
biological pathways.</p>
</td></tr>
<tr><td><code id="HybaseModel_+3A_pathwaysizedown">PathwaySizeDown</code></td>
<td>
<p>The lower-bound of the number of genes in each
biological pathways.</p>
</td></tr>
<tr><td><code id="HybaseModel_+3A_minfeaturenum_pathways">MinfeatureNum_pathways</code></td>
<td>
<p>The minimal defined pathway size after mapping your
own data to pathlistDB(KEGG database/GO database).</p>
</td></tr>
<tr><td><code id="HybaseModel_+3A_add_unmapped">Add_UnMapped</code></td>
<td>
<p>Whether to add unmapped probes for prediction</p>
</td></tr>
<tr><td><code id="HybaseModel_+3A_unmapped_num">Unmapped_num</code></td>
<td>
<p>The number of unmapped probes</p>
</td></tr>
<tr><td><code id="HybaseModel_+3A_add_feartureselection_method">Add_FeartureSelection_Method</code></td>
<td>
<p>Feature selection methods.</p>
</td></tr>
<tr><td><code id="HybaseModel_+3A_inner_cv">Inner_CV</code></td>
<td>
<p>Whether to perform a k-fold verification on the training set.</p>
</td></tr>
<tr><td><code id="HybaseModel_+3A_inner_folds">inner_folds</code></td>
<td>
<p>k-fold verification on the training set.</p>
</td></tr>
<tr><td><code id="HybaseModel_+3A_stage1_feartureselection_method">Stage1_FeartureSelection_Method</code></td>
<td>
<p>Feature selection methods.</p>
</td></tr>
<tr><td><code id="HybaseModel_+3A_cutoff">cutoff</code></td>
<td>
<p>The cutoff used for feature selection threshold. It can be any value
between 0 and 1.</p>
</td></tr>
<tr><td><code id="HybaseModel_+3A_stage2_feartureselection_method">Stage2_FeartureSelection_Method</code></td>
<td>
<p>Feature selection methods.</p>
</td></tr>
<tr><td><code id="HybaseModel_+3A_cutoff2">cutoff2</code></td>
<td>
<p>The cutoff used for feature selection threshold. It can be any value
between 0 and 1.</p>
</td></tr>
<tr><td><code id="HybaseModel_+3A_cores">cores</code></td>
<td>
<p>The number of cores used for computation.</p>
</td></tr>
<tr><td><code id="HybaseModel_+3A_verbose">verbose</code></td>
<td>
<p>Whether to print running process information to the console</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing the predictive performance of each basemodel
</p>

<hr>
<h2 id='MethylAnno'>An example about FeatureAnno for methylation data</h2><span id='topic+MethylAnno'></span>

<h3>Description</h3>

<p>An example about FeatureAnno for methylation data
</p>


<h3>Format</h3>

<p>A data frame :
</p>

<p>...

</p>


<h3>Details</h3>

<p>The annotation data stored in a data.frame for probe
mapping. It must have at least two columns named 'ID' and 'entrezID'.
</p>

<hr>
<h2 id='MethylData_Test'>An example about TrainData/TestData for methylation data</h2><span id='topic+MethylData_Test'></span>

<h3>Description</h3>

<p>An example about TrainData/TestData for methylation data
MethylData_Test.
</p>


<h3>Format</h3>

<p>A data frame :
</p>

<p>...

</p>


<h3>Details</h3>

<p>The first column
is the label or the output. For binary classes,
0 and 1 are used to indicate the class member.
</p>

<hr>
<h2 id='PathwaysModule'>Delineate differential pathway modules with high biological interpretability</h2><span id='topic+PathwaysModule'></span>

<h3>Description</h3>

<p>Delineate differential pathway modules with high biological interpretability
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PathwaysModule(
  pathways_matrix = NULL,
  control_label = NULL,
  power = NULL,
  minModuleSize = NULL,
  mergeCutHeight = NULL,
  cutoff = 70,
  MinNumPathways = 5,
  p.adjust.method = "fdr",
  exact = TRUE,
  ancestor_anno = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PathwaysModule_+3A_pathways_matrix">pathways_matrix</code></td>
<td>
<p>A pathway matrix generated by the BioM2( target='pathways') function.</p>
</td></tr>
<tr><td><code id="PathwaysModule_+3A_control_label">control_label</code></td>
<td>
<p>The label of the control group ( A single number, factor, or character )</p>
</td></tr>
<tr><td><code id="PathwaysModule_+3A_power">power</code></td>
<td>
<p>soft-thresholding power for network construction. Detail for WGCNA::blockwiseModules()</p>
</td></tr>
<tr><td><code id="PathwaysModule_+3A_minmodulesize">minModuleSize</code></td>
<td>
<p>minimum module size for module detection. Detail for WGCNA::blockwiseModules()</p>
</td></tr>
<tr><td><code id="PathwaysModule_+3A_mergecutheight">mergeCutHeight</code></td>
<td>
<p>dendrogram cut height for module merging. Detail for WGCNA::blockwiseModules()</p>
</td></tr>
<tr><td><code id="PathwaysModule_+3A_cutoff">cutoff</code></td>
<td>
<p>Thresholds for Biological Interpretability Difference Modules</p>
</td></tr>
<tr><td><code id="PathwaysModule_+3A_minnumpathways">MinNumPathways</code></td>
<td>
<p>Minimum number of pathways included in the biologically interpretable difference module</p>
</td></tr>
<tr><td><code id="PathwaysModule_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>p-value adjustment method.(holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;,</p>
</td></tr>
<tr><td><code id="PathwaysModule_+3A_exact">exact</code></td>
<td>
<p>Whether to divide GO pathways more accurately (work when ancestor_anno=NULL)</p>
</td></tr>
<tr><td><code id="PathwaysModule_+3A_ancestor_anno">ancestor_anno</code></td>
<td>
<p>Annotations for ancestral relationships (like data('GO_Ancestor') )</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing differential module results that are highly biologically interpretable
</p>

<hr>
<h2 id='PlotCorModule'>Correlalogram for Biological Differences Modules</h2><span id='topic+PlotCorModule'></span>

<h3>Description</h3>

<p>Correlalogram for Biological Differences Modules
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotCorModule(
  PathwaysModule_obj = NULL,
  alpha = 0.7,
  begin = 0.2,
  end = 0.9,
  option = "C",
  family = "serif"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotCorModule_+3A_pathwaysmodule_obj">PathwaysModule_obj</code></td>
<td>
<p>Results produced by PathwaysModule()</p>
</td></tr>
<tr><td><code id="PlotCorModule_+3A_alpha">alpha</code></td>
<td>
<p>The alpha transparency, a number in (0,1). Detail for scale_fill_viridis()</p>
</td></tr>
<tr><td><code id="PlotCorModule_+3A_begin">begin</code></td>
<td>
<p>The (corrected) hue in (0,1) at which the color map begins. Detail for scale_fill_viridis().</p>
</td></tr>
<tr><td><code id="PlotCorModule_+3A_end">end</code></td>
<td>
<p>The (corrected) hue in (0,1) at which the color map ends. Detail for scale_fill_viridis()</p>
</td></tr>
<tr><td><code id="PlotCorModule_+3A_option">option</code></td>
<td>
<p>A character string indicating the color map option to use. Detail for scale_fill_viridis()</p>
</td></tr>
<tr><td><code id="PlotCorModule_+3A_family">family</code></td>
<td>
<p>calligraphic style</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot object
</p>

<hr>
<h2 id='PlotPathFearture'>Visualisation of significant pathway-level features</h2><span id='topic+PlotPathFearture'></span>

<h3>Description</h3>

<p>Visualisation of significant pathway-level features
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotPathFearture(
  BioM2_pathways_obj = NULL,
  pathlistDB = NULL,
  top = 10,
  p.adjust.method = "none",
  begin = 0.1,
  end = 0.9,
  alpha = 0.9,
  option = "C",
  seq = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotPathFearture_+3A_biom2_pathways_obj">BioM2_pathways_obj</code></td>
<td>
<p>Results produced by BioM2(,target='pathways')</p>
</td></tr>
<tr><td><code id="PlotPathFearture_+3A_pathlistdb">pathlistDB</code></td>
<td>
<p>A list of pathways with pathway IDs and their corresponding genes ('entrezID' is used).
For details, please refer to ( data(&quot;GO2ALLEGS_BP&quot;) )</p>
</td></tr>
<tr><td><code id="PlotPathFearture_+3A_top">top</code></td>
<td>
<p>Number of significant pathway-level features visualised</p>
</td></tr>
<tr><td><code id="PlotPathFearture_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>p-value adjustment method.(holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;,
&quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;,&quot;fdr&quot;,&quot;none&quot;)</p>
</td></tr>
<tr><td><code id="PlotPathFearture_+3A_begin">begin</code></td>
<td>
<p>The (corrected) hue in (0,1) at which the color map begins. Detail for scale_fill_viridis().</p>
</td></tr>
<tr><td><code id="PlotPathFearture_+3A_end">end</code></td>
<td>
<p>The (corrected) hue in (0,1) at which the color map ends. Detail for scale_fill_viridis()</p>
</td></tr>
<tr><td><code id="PlotPathFearture_+3A_alpha">alpha</code></td>
<td>
<p>The alpha transparency, a number in (0,1). Detail for scale_fill_viridis()</p>
</td></tr>
<tr><td><code id="PlotPathFearture_+3A_option">option</code></td>
<td>
<p>A character string indicating the color map option to use. Detail for scale_fill_viridis()</p>
</td></tr>
<tr><td><code id="PlotPathFearture_+3A_seq">seq</code></td>
<td>
<p>Interval of x-coordinate</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot2 object
</p>

<hr>
<h2 id='PlotPathInner'>Visualisation Original features that make up the pathway</h2><span id='topic+PlotPathInner'></span>

<h3>Description</h3>

<p>Visualisation Original features that make up the pathway
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotPathInner(
  data = NULL,
  pathlistDB = NULL,
  FeatureAnno = NULL,
  PathNames = NULL,
  p.adjust.method = "none",
  save_pdf = FALSE,
  alpha = 1,
  cols = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotPathInner_+3A_data">data</code></td>
<td>
<p>The input omics data</p>
</td></tr>
<tr><td><code id="PlotPathInner_+3A_pathlistdb">pathlistDB</code></td>
<td>
<p>A list of pathways with pathway IDs and their corresponding genes ('entrezID' is used).
For details, please refer to ( data(&quot;GO2ALLEGS_BP&quot;) )</p>
</td></tr>
<tr><td><code id="PlotPathInner_+3A_featureanno">FeatureAnno</code></td>
<td>
<p>The annotation data stored in a data.frame for probe mapping.
It must have at least two columns named 'ID' and 'entrezID'.
(For details, please refer to data( data(&quot;MethylAnno&quot;) )</p>
</td></tr>
<tr><td><code id="PlotPathInner_+3A_pathnames">PathNames</code></td>
<td>
<p>A vector.A vector containing the names of pathways</p>
</td></tr>
<tr><td><code id="PlotPathInner_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>p-value adjustment method.(holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;,
&quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;,&quot;fdr&quot;,&quot;none&quot;)</p>
</td></tr>
<tr><td><code id="PlotPathInner_+3A_save_pdf">save_pdf</code></td>
<td>
<p>Whether to save images in PDF format</p>
</td></tr>
<tr><td><code id="PlotPathInner_+3A_alpha">alpha</code></td>
<td>
<p>The alpha transparency, a number in (0,1).</p>
</td></tr>
<tr><td><code id="PlotPathInner_+3A_cols">cols</code></td>
<td>
<p>palette (vector of colour names)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a plot object
</p>

<hr>
<h2 id='PlotPathNet'>Network diagram of pathways-level features</h2><span id='topic+PlotPathNet'></span>

<h3>Description</h3>

<p>Network diagram of pathways-level features
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotPathNet(
  data = NULL,
  BioM2_pathways_obj = NULL,
  FeatureAnno = NULL,
  pathlistDB = NULL,
  PathNames = NULL,
  cutoff = 0.2,
  num = 10
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotPathNet_+3A_data">data</code></td>
<td>
<p>The input omics data</p>
</td></tr>
<tr><td><code id="PlotPathNet_+3A_biom2_pathways_obj">BioM2_pathways_obj</code></td>
<td>
<p>Results produced by BioM2()</p>
</td></tr>
<tr><td><code id="PlotPathNet_+3A_featureanno">FeatureAnno</code></td>
<td>
<p>The annotation data stored in a data.frame for probe mapping.
It must have at least two columns named 'ID' and 'entrezID'.
(For details, please refer to data( data(&quot;MethylAnno&quot;) )</p>
</td></tr>
<tr><td><code id="PlotPathNet_+3A_pathlistdb">pathlistDB</code></td>
<td>
<p>A list of pathways with pathway IDs and their corresponding genes ('entrezID' is used).
For details, please refer to ( data(&quot;GO2ALLEGS_BP&quot;) )</p>
</td></tr>
<tr><td><code id="PlotPathNet_+3A_pathnames">PathNames</code></td>
<td>
<p>A vector.A vector containing the names of pathways</p>
</td></tr>
<tr><td><code id="PlotPathNet_+3A_cutoff">cutoff</code></td>
<td>
<p>Threshold for correlation between features within a pathway</p>
</td></tr>
<tr><td><code id="PlotPathNet_+3A_num">num</code></td>
<td>
<p>The first few internal features of each pathway that are most relevant to the phenotype</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot object
</p>

<hr>
<h2 id='ShowModule'>Display biological information within each pathway module</h2><span id='topic+ShowModule'></span>

<h3>Description</h3>

<p>Display biological information within each pathway module
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ShowModule(obj = NULL, ID_Module = NULL, exact = TRUE, ancestor_anno = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ShowModule_+3A_obj">obj</code></td>
<td>
<p>Results produced by PathwaysModule()</p>
</td></tr>
<tr><td><code id="ShowModule_+3A_id_module">ID_Module</code></td>
<td>
<p>ID of the diff module</p>
</td></tr>
<tr><td><code id="ShowModule_+3A_exact">exact</code></td>
<td>
<p>Whether to divide GO pathways more accurately (work when ancestor_anno=NULL)</p>
</td></tr>
<tr><td><code id="ShowModule_+3A_ancestor_anno">ancestor_anno</code></td>
<td>
<p>Annotations for ancestral relationships (like data('GO_Ancestor') )</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List containing biologically specific information within the module
</p>

<hr>
<h2 id='Stage1_FeartureSelection'>Stage 1 Fearture Selection</h2><span id='topic+Stage1_FeartureSelection'></span>

<h3>Description</h3>

<p>Stage 1 Fearture Selection
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Stage1_FeartureSelection(
  Stage1_FeartureSelection_Method = "cor",
  data = NULL,
  cutoff = NULL,
  featureAnno = NULL,
  pathlistDB_sub = NULL,
  cores = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Stage1_FeartureSelection_+3A_stage1_feartureselection_method">Stage1_FeartureSelection_Method</code></td>
<td>
<p>Feature selection methods. Available options are
c(NULL, 'cor', 'wilcox.test', 'cor_rank', 'wilcox.test_rank').</p>
</td></tr>
<tr><td><code id="Stage1_FeartureSelection_+3A_data">data</code></td>
<td>
<p>The input training dataset. The first column is the label.</p>
</td></tr>
<tr><td><code id="Stage1_FeartureSelection_+3A_cutoff">cutoff</code></td>
<td>
<p>The cutoff used for feature selection threshold. It can be any value
between 0 and 1. Commonly used cutoffs are c(0.5, 0.1, 0.05, 0.01, etc.).</p>
</td></tr>
<tr><td><code id="Stage1_FeartureSelection_+3A_featureanno">featureAnno</code></td>
<td>
<p>The annotation data stored in a data.frame for probe
mapping. It must have at least two columns named 'ID' and 'entrezID'.
(For details, please refer to data( data(&quot;MethylAnno&quot;) )</p>
</td></tr>
<tr><td><code id="Stage1_FeartureSelection_+3A_pathlistdb_sub">pathlistDB_sub</code></td>
<td>
<p>A list of pathways with pathway IDs and their
corresponding genes ('entrezID' is used).
For details, please refer to ( data(&quot;GO2ALLEGS_BP&quot;) )</p>
</td></tr>
<tr><td><code id="Stage1_FeartureSelection_+3A_cores">cores</code></td>
<td>
<p>The number of cores used for computation.</p>
</td></tr>
<tr><td><code id="Stage1_FeartureSelection_+3A_verbose">verbose</code></td>
<td>
<p>Whether to print running process information to the console</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of matrices with pathway IDs as the associated list member
names.
</p>


<h3>Author(s)</h3>

<p>Shunjie Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(parallel)
data=MethylData_Test
feature_pathways=Stage1_FeartureSelection(Stage1_FeartureSelection_Method='cor',
                 data=data,cutoff=0,
                 featureAnno=MethylAnno,pathlistDB_sub=GO2ALLEGS_BP,cores=1)

</code></pre>

<hr>
<h2 id='Stage2_FeartureSelection'>Stage 2 Fearture Selection</h2><span id='topic+Stage2_FeartureSelection'></span>

<h3>Description</h3>

<p>Stage 2 Fearture Selection
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Stage2_FeartureSelection(
  Stage2_FeartureSelection_Method = "RemoveHighcor",
  data = NULL,
  label = NULL,
  cutoff = NULL,
  preMode = NULL,
  classifier = NULL,
  verbose = TRUE,
  cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Stage2_FeartureSelection_+3A_stage2_feartureselection_method">Stage2_FeartureSelection_Method</code></td>
<td>
<p>Feature selection methods. Available options are
c(NULL, 'cor', 'wilcox.test', 'RemoveHighcor', 'RemoveLinear').</p>
</td></tr>
<tr><td><code id="Stage2_FeartureSelection_+3A_data">data</code></td>
<td>
<p>The input training dataset. The first column is the label.</p>
</td></tr>
<tr><td><code id="Stage2_FeartureSelection_+3A_label">label</code></td>
<td>
<p>The label of dataset</p>
</td></tr>
<tr><td><code id="Stage2_FeartureSelection_+3A_cutoff">cutoff</code></td>
<td>
<p>The cutoff used for feature selection threshold. It can be any value
between 0 and 1.</p>
</td></tr>
<tr><td><code id="Stage2_FeartureSelection_+3A_premode">preMode</code></td>
<td>
<p>The prediction mode. Available options are
c('probability', 'classification').</p>
</td></tr>
<tr><td><code id="Stage2_FeartureSelection_+3A_classifier">classifier</code></td>
<td>
<p>Learners in mlr3</p>
</td></tr>
<tr><td><code id="Stage2_FeartureSelection_+3A_verbose">verbose</code></td>
<td>
<p>Whether to print running process information to the console</p>
</td></tr>
<tr><td><code id="Stage2_FeartureSelection_+3A_cores">cores</code></td>
<td>
<p>The number of cores used for computation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Column index of feature
</p>


<h3>Author(s)</h3>

<p>Shunjie Zhang
</p>

<hr>
<h2 id='TransAnno'>An example about FeatureAnno for gene expression</h2><span id='topic+TransAnno'></span>

<h3>Description</h3>

<p>An example about FeatureAnno for gene expression
</p>


<h3>Format</h3>

<p>A data frame :
</p>

<p>...

</p>


<h3>Details</h3>

<p>The annotation data stored in a data.frame for probe
mapping. It must have at least two columns named 'ID' and 'entrezID'.
</p>

<hr>
<h2 id='TransData_Test'>An example about TrainData/TestData for gene expression</h2><span id='topic+TransData_Test'></span>

<h3>Description</h3>

<p>An example about TrainData/TestData for gene expression
MethylData_Test.
</p>


<h3>Format</h3>

<p>A data frame :
</p>

<p>...

</p>


<h3>Details</h3>

<p>The first column
is the label or the output. For binary classes,
0 and 1 are used to indicate the class member.
</p>

<hr>
<h2 id='VisMultiModule'>Visualisation of the results of the analysis of the pathway modules</h2><span id='topic+VisMultiModule'></span>

<h3>Description</h3>

<p>Visualisation of the results of the analysis of the pathway modules
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VisMultiModule(
  BioM2_pathways_obj = NULL,
  FindParaModule_obj = NULL,
  ShowModule_obj = NULL,
  PathwaysModule_obj = NULL,
  exact = TRUE,
  ancestor_anno = NULL,
  type_text_table = FALSE,
  text_table_theme = ttheme("mOrange"),
  volin = FALSE,
  control_label = 0,
  module = NULL,
  cols = NULL,
  n_neighbors = 8,
  spread = 1,
  min_dist = 2,
  target_weight = 0.5,
  size = 1.5,
  alpha = 1,
  ellipse = TRUE,
  ellipse.alpha = 0.2,
  theme = ggthemes::theme_base(base_family = "serif"),
  save_pdf = FALSE,
  width = 7,
  height = 7
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VisMultiModule_+3A_biom2_pathways_obj">BioM2_pathways_obj</code></td>
<td>
<p>Results produced by BioM2(,target='pathways')</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_findparamodule_obj">FindParaModule_obj</code></td>
<td>
<p>Results produced by FindParaModule()</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_showmodule_obj">ShowModule_obj</code></td>
<td>
<p>Results produced by ShowModule()</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_pathwaysmodule_obj">PathwaysModule_obj</code></td>
<td>
<p>Results produced by PathwaysModule()</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_exact">exact</code></td>
<td>
<p>Whether to divide GO pathways more accurately (work when ancestor_anno=NULL)</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_ancestor_anno">ancestor_anno</code></td>
<td>
<p>Annotations for ancestral relationships (like data('GO_Ancestor') )</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_type_text_table">type_text_table</code></td>
<td>
<p>Whether to display it in a table</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_text_table_theme">text_table_theme</code></td>
<td>
<p>The topic of this table.Detail for ggtexttable()</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_volin">volin</code></td>
<td>
<p>Can only be used when PathwaysModule_obj exists. ( Violin diagram )</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_control_label">control_label</code></td>
<td>
<p>Can only be used when PathwaysModule_obj exists. ( Control group label )</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_module">module</code></td>
<td>
<p>Can only be used when PathwaysModule_obj exists.( PathwaysModule ID )</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_cols">cols</code></td>
<td>
<p>palette (vector of colour names)</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_n_neighbors">n_neighbors</code></td>
<td>
<p>The size of local neighborhood (in terms of number of neighboring sample points) used for manifold approximation.
Larger values result in more global views of the manifold, while smaller values result in more local data being preserved.
In general values should be in the range 2 to 100.</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_spread">spread</code></td>
<td>
<p>The effective scale of embedded points. In combination with min_dist, this determines how clustered/clumped the embedded points are.</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_min_dist">min_dist</code></td>
<td>
<p>The effective minimum distance between embedded points.
Smaller values will result in a more clustered/clumped embedding where nearby points on the manifold are drawn closer together,
while larger values will result on a more even dispersal of points.
The value should be set relative to the spread value,
which determines the scale at which embedded points will be spread out.</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_target_weight">target_weight</code></td>
<td>
<p>Weighting factor between data topology and target topology.
A value of 0.0 weights entirely on data, a value of 1.0 weights entirely on target.
The default of 0.5 balances the weighting equally between data and target.
Only applies if y is non-NULL.</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_size">size</code></td>
<td>
<p>Scatter plot point size</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_alpha">alpha</code></td>
<td>
<p>Alpha for ellipse specifying the transparency level of fill color. Use alpha = 0 for no fill color.</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_ellipse">ellipse</code></td>
<td>
<p>logical value. If TRUE, draws ellipses around points.</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_ellipse.alpha">ellipse.alpha</code></td>
<td>
<p>Alpha for ellipse specifying the transparency level of fill color. Use alpha = 0 for no fill color.</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_theme">theme</code></td>
<td>
<p>Default:theme_base(base_family = &quot;serif&quot;)</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_save_pdf">save_pdf</code></td>
<td>
<p>Whether to save images in PDF format</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_width">width</code></td>
<td>
<p>image width</p>
</td></tr>
<tr><td><code id="VisMultiModule_+3A_height">height</code></td>
<td>
<p>image height</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot2 object
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
