<!DOCTYPE html><html><head><title>Help for package bios2mds</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {bios2mds}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bios2mds-package'>
<p>From BIOlogical Sequences to MultiDimensional Scaling</p></a></li>
<li><a href='#col.group'><p>Links elements in a mmds object to specific groups and colors</p></a></li>
<li><a href='#dif'>
<p>Difference score</p></a></li>
<li><a href='#dis'>
<p>Dissimilarity score</p></a></li>
<li><a href='#export.fasta'>
<p>Writes a MSA file in FASTA format</p></a></li>
<li><a href='#extract.cluster'>
<p>Extraction of clusters alignments</p></a></li>
<li><a href='#gpcr'>
<p>Pre-analyzed G-Protein-Coupled Receptor (GPCR) data set</p></a></li>
<li><a href='#import.fasta'>
<p>Reads a file in FASTA format</p></a></li>
<li><a href='#import.msf'>
<p>Reads a multiple sequence alignment file in MSF format</p></a></li>
<li><a href='#kmeans.run'>
<p>Multiple runs of K-means analysis</p></a></li>
<li><a href='#mat.dif'>
<p>Matrices of difference scores between sequences</p></a></li>
<li><a href='#mat.dis'>
<p>Matrices of dissimilarity scores between amino acid sequences</p></a></li>
<li><a href='#mmds'>
<p>Metric multidimensional scaling</p></a></li>
<li><a href='#mmds.2D.multi'>
<p>Plots the mmds coordinates onto a 2D space</p></a></li>
<li><a href='#mmds.2D.plot'>
<p>Plots the mmds coordinates onto a 2D space</p></a></li>
<li><a href='#mmds.3D.plot'>
<p>Display the mmds coordinates onto a 3D space</p></a></li>
<li><a href='#mmds.plot'>
<p>Plots a summary of the mmds results</p></a></li>
<li><a href='#mmds.project'>
<p>Metric multidimensional Scaling Projection</p></a></li>
<li><a href='#random.msa'>
<p>Random Alignment</p></a></li>
<li><a href='#scree.plot'>
<p>Plots the eingenvalues of an MDS analysis</p></a></li>
<li><a href='#sil.score'>
<p>Silhouette score</p></a></li>
<li><a href='#sub.mat'>
<p>Amino acid substitution matrices</p></a></li>
<li><a href='#write.mmds.pdb'>
<p>Writes PDB file</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>From Biological Sequences to Multidimensional Scaling</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-04-02</td>
</tr>
<tr>
<td>Author:</td>
<td>Julien Pele with Jean-Michel Becu, Rym Ben Boubaker, Herve Abdi, and Marie
        Chabbert</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Marie Chabbert &lt;marie.chabbert@univ-angers.fr&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0), amap, e1071, scales, cluster, rgl</td>
</tr>
<tr>
<td>Description:</td>
<td>Utilities dedicated to the analysis of
        biological sequences by metric MultiDimensional Scaling with
        projection of supplementary data. It contains functions for
        reading multiple sequence alignment files, calculating distance
        matrices, performing metric multidimensional scaling and
        visualizing results.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-04-06 18:44:35 UTC; rym</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-04-07 09:50:09 UTC</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
</table>
<hr>
<h2 id='bios2mds-package'>
From BIOlogical Sequences to MultiDimensional Scaling
</h2><span id='topic+bios2mds-package'></span><span id='topic+bios2mds'></span>

<h3>Description</h3>

<p>The <code>bios2mds</code> package is developed in the Bioinformatics team at Integrated
Neurovascular Biology Laboratory, UMR CNRS 6214 / INSERM 771, University of Angers - FRANCE.
</p>
<p>This package is dedicated to the analysis of biological sequences by metric MultiDimensional Scaling (MDS) with projection of supplementary data. It contains functions for reading multiple sequence alignment (MSA) files, calculating distance matrices from MSA files, performing MDS analysis and visualizing results. The MDS analysis and visualization tools can be applied to any kind of data.
</p>
<p>The main functionalities of <code>bios2mds</code> are summarized below:
</p>

<dl>
<dt>(1) BUILDING DISTANCE MATRICES FROM MULTIPLE SEQUENCE ALIGNMENTS : </dt><dd>
<p>Several functions allow users to read multiple sequence alignment files and to compute matrices of distances between these sequences:
</p>

<ul>
<li> <p><code><a href="#topic+import.fasta">import.fasta</a></code>: reads a multiple sequence alignment in FASTA format.
</p>
</li>
<li> <p><code><a href="#topic+import.msf">import.msf</a></code>: reads a multiple sequence alignment in MSF format.
</p>
</li>
<li> <p><code><a href="#topic+mat.dif">mat.dif</a></code>: computes a matrix of pairwise distances between sequences based on sequence difference.
</p>
</li>
<li> <p><code><a href="#topic+mat.dis">mat.dis</a></code>: computes a matrix of pairwise distances between sequences based on sequence dissimilarity.
</p>
</li></ul>

</dd>
<dt>(2) MULTIDIMENSIONAL SCALING : </dt><dd>
<p>A function performs metric MDS analysis of a distance matrix between active elements with the option of projecting supplementary elements onto the active space.
</p>

<ul>
<li> <p><code><a href="#topic+mmds">mmds</a></code>: performs metric multidimensional scaling.
</p>
</li>
<li> <p><code><a href="#topic+mmds.project">mmds.project</a></code>: performs projection of supplementary elements onto the active space.
</p>
</li></ul>

</dd>
<dt>(3) GRAPHICAL TOOLS : </dt><dd>
<p>Several functions are proposed to visualize results of metric MDS analysis:
</p>

<ul>
<li> <p><code><a href="#topic+scree.plot">scree.plot</a></code>: draws the scree plot of eigenvalues.
</p>
</li>
<li> <p><code><a href="#topic+mmds.2D.plot">mmds.2D.plot</a></code>: draws a scatter plot of the MDS coordinates. 
</p>
</li>
<li> <p><code><a href="#topic+mmds.2D.multi">mmds.2D.multi</a></code>: draws a scatter plot of the MDS coordinates with projection of multiple groups of supplementary element. 
</p>
</li>
<li> <p><code><a href="#topic+mmds.3D.plot">mmds.3D.plot</a></code>: Displays a 3D plot of the MDS coordinates. 
</p>
</li>
<li> <p><code><a href="#topic+mmds.plot">mmds.plot</a></code>: wrapper function that draws the scree plot and three scatter plots of MDS coordinates.
</p>
</li>
<li> <p><code><a href="#topic+col.group">col.group</a></code>: colors scatter plots with user provided groupings and colors.
</p>
</li>
<li> <p><code><a href="#topic+write.mmds.pdb">write.mmds.pdb</a></code>: writes MDS coordinates in a PDB formatted file for 3D visualisation.
</p>
</li></ul>

</dd>
<dt>(4) CLUSTER ANALYSIS : </dt><dd>
<p>Several functions allow users to perform data clustering and to assess the clustering robustness:
</p>

<ul>
<li> <p><code><a href="#topic+kmeans.run">kmeans.run</a></code>: performs multiple runs of K-means clustering and analyzes clusters
</p>
</li>
<li> <p><code><a href="#topic+sil.score">sil.score</a></code>: calculates the silhouette score from multiple K-means runs to determine the optimal number of clusters.
</p>
</li>
<li> <p><code><a href="#topic+extract.cluster">extract.cluster</a></code>: extraction of clusters alignments
</p>
</li></ul>

</dd>
<dt>(5) DATASETS : </dt><dd>
<p>Two raw datasets are proposed to test the functionalities of <code>bios2mds</code>. They correspond to the multiple sequence alignments of GPCRs from <em>H. sapiens</em> and <em>D. melanogaster</em> in .msf and .fa formats (msa/human_gpcr.* and msa/drome_gpcr.*). They are based on the non-redundant sets of non-olfactory class A GPCRs, prepared and analyzed in Deville et al. (2009) and updated with the July 2009 release of Uniprot <a href="http://www.uniprot.org">http://www.uniprot.org</a>. 
Each MSA file is related to a .csv file that assigns a group and a color to each sequence of the alignment (csv/human_gpcr_group.csv and csv/drome_gpcr_group.csv). 
</p>
<p>Pre-analyzed data from these two MSA files are in <code>gpcr</code>. Moreover, <code>gpcr</code> contain projection of GPCRs from <em>N. vectensis</em> and from <em>C. intestinalis</em> onto the active space of human GPCRs calculated by MDS analysis.
</p>
</dd>
</dl>

<p>For an index of functions, use library(help = <code>bios2mds</code>).
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package </td><td style="text-align: left;"> bios2mds</td>
</tr>
<tr>
 <td style="text-align: left;">
Type </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version </td><td style="text-align: left;"> 1.2.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Repository </td><td style="text-align: left;"> CRAN</td>
</tr>
<tr>
 <td style="text-align: left;">
Date </td><td style="text-align: left;"> 2012-06-01</td>
</tr>
<tr>
 <td style="text-align: left;">
License </td><td style="text-align: left;"> GPL version 2 or newer</td>
</tr>
<tr>
 <td style="text-align: left;">
Collate	</td><td style="text-align: left;"> Other useful packages can be found in the CRAN task view.</td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> See <a href="https://mirror.its.sfu.ca/mirror/CRAN/web/views/Multivariate.html">https://mirror.its.sfu.ca/mirror/CRAN/web/views/Multivariate.html</a></td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> and <a href="https://mirror.its.sfu.ca/mirror/CRAN/web/views/Cluster.html">https://mirror.its.sfu.ca/mirror/CRAN/web/views/Cluster.html</a></td>
</tr>

</table>


<h3>Author(s)</h3>

<p>Julien Pele &lt;julien.pele@yahoo.fr&gt; with Jean-Michel Becu &lt;jean-michel.becu@etu.univ-rouen.fr&gt;, 
Herve Abdi &lt;herve@utdallas.edu&gt; and Marie Chabbert &lt;marie.chabbert@univ-angers.fr&gt;.
</p>
<p>Maintainer: Marie Chabbert &lt;marie.chabbert@univ-angers.fr&gt;
</p>


<h3>References</h3>

<p>citation('bios2mds')
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# The MSA files provided with the package correspond to the sequence 
# alignment of non-olfactory class A G-protein-coupled receptors from
# H. sapiens and D. melanogaster prepared by Deville et al. (2009).  
# loading GPCR data
wd &lt;- tempdir()
file &lt;- file.path(wd,"R.pdb")
data(gpcr)

# building distance matrices between the aligned GPCR sequences from
# H. sapiens and D. melanogaster
human &lt;- import.fasta(system.file("msa/human_gpcr.fa", package = "bios2mds"))
drome &lt;- import.fasta(system.file("msa/drome_gpcr.fa", package = "bios2mds"))

#active &lt;- mat.dif(human, human)
# or
active &lt;- gpcr$dif$sapiens.sapiens

#sup &lt;- mat.dif(drome, human)
# or
sup &lt;- gpcr$dif$melanogaster.sapiens

# performing MDS analysis of the GPCR sequences from H. sapiens 
mmds_active &lt;- mmds(active, group.file=system.file(
"csv/human_gpcr_group.csv",package = "bios2mds"))

# performing MDS analysis of the GPCR sequences from H. sapiens 
# with projection of GPCRs from D. melanogaster 
# as supplementary elements onto the space of human GPCRs
mmds_sup &lt;- mmds.project(mmds_active, sup,system.file(
"csv/drome_gpcr_group.csv",package = "bios2mds"))

# displaying MDS coordinates 
layout(matrix(1:6, 2, 3))

scree.plot(mmds_active$eigen.perc, lab = TRUE, title = "Scree plot of metric MDS")

mmds.2D &lt;- mmds.2D.plot(mmds_active, title = "Sequence space of human GPCRs")

mmds.2D.plot(mmds_active,mmds_sup, title = "Projection of GPCRs from D. melanogaster
 onto the space space of human GPCRs ", active.alpha = 0.3)
   
# writing PDB files for 3D visualization of MDS coordinates
write.mmds.pdb(mmds_active,file.pdb=file)

</code></pre>

<hr>
<h2 id='col.group'>Links elements in a mmds object to specific groups and colors</h2><span id='topic+col.group'></span>

<h3>Description</h3>

<p>Links elements in a <code>mmds</code> object or <code>mmds.project</code> object to user-provided groups and colors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>col.group(x,file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="col.group_+3A_x">x</code></td>
<td>

<p>a 'mmds' or 'project' object obtained from <code><a href="#topic+mmds">mmds</a></code> or <code><a href="#topic+mmds.project">mmds.project</a></code> function
</p>
</td></tr>
<tr><td><code id="col.group_+3A_file">file</code></td>
<td>

<p>a string of characters to indicate the file name assigning groups and colors to each active OR each supplementary element of the mmds object.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>col.group</code> assigns each element of the mmds object to user-provided groupings and colors for coloring and labeling mmds scatter plots. 
</p>
<p><code>col.group</code> requires a formatted file. See &quot;csv/human_gpcr_group.csv&quot; for an example. Each line corresponds to one element of the mmds object and must contain three parameters separated by &quot;,&quot;. <br />
The first parameter is the element name, as given in the multiple sequence alignment file. <br />
The second parameter is the group name. Groupings must be provided by the user.<br />
The third parameter is the group color in full letters (example : &quot;black&quot;,&quot;green&quot;). Two or more groups can have the same color, but elements within the same group must have the same color. The group is colored by the first color encountered. 
</p>


<h3>Value</h3>

<p>Adds data to a mmds object in order to color and label mmds scatter plots with user-provided groupings and colors.
</p>


<h3>Author(s)</h3>

<p>Jean-Michel Becu
</p>


<h3>See Also</h3>

<p>See <code>colors</code> function (default R package).<br />
See <code>getcol</code> in <code>made4</code> package for special color palette developed to maximize the contrast between colors.<br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'># performing metric MDS on human GPCRs with projection of 
# GPCRs from D. melanogaster as supplementary data:
data(gpcr)
active &lt;- gpcr$dif$sapiens.sapiens
mmds_active &lt;- mmds(active)
mmds_active&lt;-col.group(mmds_active,system.file("csv/human_gpcr_group.csv"
,package = "bios2mds"))
</code></pre>

<hr>
<h2 id='dif'>
Difference score
</h2><span id='topic+dif'></span>

<h3>Description</h3>

<p>Measures the difference score between two aligned amino acid or nucleotide sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dif(seq1, seq2, gap = FALSE, aa.strict = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dif_+3A_seq1">seq1</code></td>
<td>

<p>a character vector representing a first sequence.
</p>
</td></tr>
<tr><td><code id="dif_+3A_seq2">seq2</code></td>
<td>

<p>a character vector representing a second sequence.
</p>
</td></tr>
<tr><td><code id="dif_+3A_gap">gap</code></td>
<td>

<p>a boolean indicating whether the gap character should be taken as a supplementary symbol
(TRUE) or not (FALSE). Default is FALSE.
</p>
</td></tr>
<tr><td><code id="dif_+3A_aa.strict">aa.strict</code></td>
<td>

<p>a boolean indicating whether only strict amino acids should be taken into account (TRUE)
or not (FALSE). Default is FALSE. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The difference score between two aligned sequences is given by the proportion of sites that differs and is equivalent to <code class="reqn">1 - {PID}</code> (percent identity). 
<code>dif</code> is given by the number of aligned positions (sites) whose symbols differ, divided by the number of aligned positions. <code>dif</code> is equivalent to the <em>p</em> distance defined by Nei and Zhang (2006).
In <code>dif</code>, positions with at least one gap can be excluded (gap = FALSE). When gaps are taken as a supplementary symbol (gap = TRUE), sites with gaps in both sequences are excluded.
</p>
<p>From Nei and Zhang (2006), the <em>p</em> distance, which is the proportion of sites that differ between
two sequences, is estimated by:
</p>
<p style="text-align: center;"><code class="reqn">{p} = \frac{n_d}{n},</code>
</p>

<p>where <em>n</em> is the number of sites and <em><code class="reqn">n_d</code></em> is the number of sites with different symbols.
</p>
<p>The difference score ranges from 0, for identical sequences, to 1, for completely 
different sequences.
</p>


<h3>Value</h3>

<p>A single numeric value representing the difference score.
</p>


<h3>Author(s)</h3>

<p>Julien Pele
</p>


<h3>References</h3>

<p>May AC (2004) Percent sequence identity: the need to be explicit. <em>Structure</em>
<b>12</b>:737-738.
</p>
<p>Nei M and Zhang J (2006) Evolutionary Distance: Estimation.
<em>Encyclopedia of Life Sciences</em>.
</p>
<p>Nei M and Kumar S (2000) Molecular Evolution and Phylogenetics.
<em>Oxford University Press</em>, New York.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># calculating the difference score between the sequences 
# of CLTR1_HUMAN and CLTR2_HUMAN:
aln &lt;- import.fasta(system.file("msa/human_gpcr.fa", package = "bios2mds"))
dif &lt;- dif(aln$CLTR1_HUMAN, aln$CLTR2_HUMAN)
dif
</code></pre>

<hr>
<h2 id='dis'>
Dissimilarity score
</h2><span id='topic+dis'></span>

<h3>Description</h3>

<p>Computes the dissimilarity score between two aligned amino acid sequences, based on substitution matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dis(seq1, seq2, sub.mat.id = "PAM250", gap = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dis_+3A_seq1">seq1</code></td>
<td>

<p>a character vector representing a first amino acid sequence.
</p>
</td></tr>
<tr><td><code id="dis_+3A_seq2">seq2</code></td>
<td>

<p>a character vector representing a second amino acid sequence.
</p>
</td></tr>
<tr><td><code id="dis_+3A_sub.mat.id">sub.mat.id</code></td>
<td>

<p>a string of characters indicating the amino acid substitution matrix to be taken into 
account for calculation. This should be one of &quot;PAM40&quot;, &quot;PAM80&quot;, &quot;PAM120&quot;, &quot;PAM160&quot;, &quot;PAM250&quot;, &quot;BLOSUM30&quot;, &quot;BLOSUM45&quot;, &quot;BLOSUM62&quot;, &quot;BLOSUM80&quot;, &quot;GONNET&quot;, &quot;JTT&quot;, &quot;JTT_TM&quot; and &quot;PHAT&quot;. Default is PAM250. See <code><a href="#topic+sub.mat">sub.mat</a></code>. 
</p>
</td></tr>
<tr><td><code id="dis_+3A_gap">gap</code></td>
<td>

<p>a numeric vector of length 2, indicating the penalty for initiating and extending a strand of two gaps (gap ignored, default).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Grishin and Grishin (2002) developped a method to calculate the similarity score with amino acid
substitution matrices.
</p>
<p>Let <b><em>s</em></b> be an amino acid substitution matrix with elements <em>s(a, b)</em>,
let <b><em>A</em></b> be an alignment of <em>n</em> sequences, <em><code class="reqn">A_{ik}</code></em> is a symbol (amino acid
or gap: '-') in the site <em>k</em> of the sequence <em>i</em>. For each pair of sequences <em>i</em> and
<em>j</em> from <b><em>A</em></b>, the following equations<b>(1)</b>, <b>(2)</b>, <b>(3)</b> and <b>(4)</b> are calculated as follows:
</p>

<dl>
<dt>(1)</dt><dd>
<p><em><code class="reqn">S_{ij}</code></em>, called score per site, is obtained as:
</p>
<p style="text-align: center;"><code class="reqn">S_{ij} = \sum_{k \in K_{ij}}s(A_{ik}, A_{jk})/l(K_{ij})</code>
</p>

<p>where <code class="reqn">K_{ik}</code> is the set of sites <em>k</em> such that <code class="reqn">A_{ik}</code> <code class="reqn">!=</code> '-' and <code class="reqn">A_{jk}</code> <code class="reqn">!=</code> '-'
and <em><code class="reqn">l(K_{ij})</code></em> is the number of elements in <code class="reqn">K_{ij}</code>.
</p>
</dd>
<dt>(2)</dt><dd>
<p><em><code class="reqn">T_{ij}</code></em>, called average upper limit of the score per site, is obtained as:
</p>
<p style="text-align: center;"><code class="reqn">T_{ij} = 0.5 \sum_{k \in K_{ij}}(s(A_{ik}, A_{ik}) + s(A_{jk}, A_{jk}))/l(K_{ij})</code>
</p>

</dd>
<dt>(3)</dt><dd>
<p><em><code class="reqn">S^{rand}_{ij}</code></em>, called score per site expected from random sequences, is obtained as:
</p>
<p style="text-align: center;"><code class="reqn">S^{rand}_{ij} = \sum^{20}_{a=1}\sum^{20}_{b=1}f^i_j(a)f^j_i(b)s(a, b)</code>
</p>

<p>where <code class="reqn">f^i_j(a)</code> is the frequency of amino acid '<em>a</em>' in <code class="reqn">i^{th}</code> protein sequence of <b><em>A</em></b>
over all sites in <em><code class="reqn">K_{ij}</code></em>.
</p>
</dd>
<dt>(4)</dt><dd>
<p><em><code class="reqn">V_{ij}</code></em>, called normalized score (Feng and Doolittle, 1997), is obtained as:
</p>
<p style="text-align: center;"><code class="reqn">V_{ij} = \frac{S_{ij} - S^{rand}_{ij}}{T_{ij} - S^{rand}_{ij}}</code>
</p>

</dd>
</dl>
<p>The normalized score <em><code class="reqn">V_{ij}</code></em> ranges from 0 (for random sequences) to 1 (for identical
sequences). However, for very divergent sequences, <em><code class="reqn">V_{ij}</code></em> can become negative due to
statistical errors. In this case, <code>dis</code> attributes 0 to negative scores.
</p>
<p>The dissimilarity score <em><code class="reqn">D_{ij}</code></em> between sequences <em>i</em> and <em>j</em> is obtained from the similarity score as:
</p>
<p style="text-align: center;"><code class="reqn">D_{ij} = V_{ij} - 1</code>
</p>




<h3>Value</h3>

<p>A single numeric value representing the dissimilarity score.
</p>


<h3>Author(s)</h3>

<p>Julien Pele
</p>


<h3>References</h3>

<p>Grishin VN and Grishin NV (2002) Euclidian space and grouping of biological objects.
<em>Bioinformatics</em> <b>18</b>:1523-1534.
</p>
<p>Feng DF and Doolittle RF (1997) Converting amino acid alignment scores into measures of
evolutionary time: a simulation study of various relationships. <em>J Mol Evol</em>  
<b>44</b>:361-370.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># calculating dis between the sequences of CLTR1_HUMAN and CLTR2_HUMAN:
aln &lt;- import.fasta(system.file("msa/human_gpcr.fa", package = "bios2mds"))
dis &lt;- dis(aln$CLTR1_HUMAN, aln$CLTR2_HUMAN)
dis
</code></pre>

<hr>
<h2 id='export.fasta'>
Writes a MSA file in FASTA format
</h2><span id='topic+export.fasta'></span>

<h3>Description</h3>

<p>Writes a multiple sequence alignment (MSA) file in FASTA format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export.fasta(x, outfile = "alignment.fa", ncol = 60, open = "w")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="export.fasta_+3A_x">x</code></td>
<td>

<p>an object of class 'align', obtained from <code><a href="#topic+import.fasta">import.fasta</a></code> or <code><a href="#topic+import.msf">import.msf</a></code>, or an element list from the <code><a href="#topic+extract.cluster">extract.cluster</a></code> function return.
</p>
</td></tr>
<tr><td><code id="export.fasta_+3A_outfile">outfile</code></td>
<td>

<p>a string of characters or string vector to indicate the name of the MSA file(s) to be written. If x is an object of class 'align', default is &quot;alignment.fa&quot;. If x is an element list, for each element in x, default is the element name, followed by the &quot;.alignement.fa&quot; extension.
</p>
</td></tr>
<tr><td><code id="export.fasta_+3A_ncol">ncol</code></td>
<td>

<p>an integer value indicating the number of characters per line for the sequences in outfile.Default is 60. 
</p>
</td></tr>
<tr><td><code id="export.fasta_+3A_open">open</code></td>
<td>

<p>a character value indicating the opening mode for outfile. This should be either of &quot;w&quot; to write into
a new file, or &quot;a&quot; to append at the end of an already existing file. Default is &quot;w&quot;.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Initially, FASTA (for FAST-ALL) was the input format of the FASTA program, used for protein comparison and searching in databases.
Presently, FASTA format is a standard format for biological sequences.
</p>
<p>The FASTA formatted file of a single sequence displays:
</p>

<ul>
<li><p> a single-line description beginning with a greater-than (&gt;) symbol. The following word is the identifier.
</p>
</li>
<li><p> followed by any number of lines, representing biological sequence.
</p>
</li></ul>

<p>For multiple alignments, the FASTA formatted sequences are concatenated to create a multiple
FASTA format.
</p>


<h3>Value</h3>

<p>Produces a FASTA file for an 'align' object or a FASTA file for each cluster in list.
</p>


<h3>Note</h3>

<p>For further information about FASTA format, see: <a href="http://www.ncbi.nlm.nih.gov/BLAST/fasta.shtml">http://www.ncbi.nlm.nih.gov/BLAST/fasta.shtml</a>
</p>


<h3>Author(s)</h3>

<p>Jean-Michel Becu
</p>


<h3>See Also</h3>

<p><code>write.fasta</code> function from <code>seqinr</code> package.<br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'># reading of the multiple sequence alignment of human GPCRS in FASTA format:
wd &lt;- tempdir()
#wd &lt;- getwd()
file1 &lt;- file.path(wd,"alignment.fa")
aln &lt;- import.fasta(system.file("msa/human_gpcr.fa", package = "bios2mds"))
export.fasta(aln,file1)
</code></pre>

<hr>
<h2 id='extract.cluster'>
Extraction of clusters alignments
</h2><span id='topic+extract.cluster'></span>

<h3>Description</h3>

<p>Extracts the multiple sequence alignement of each cluster after K-means clustering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract.cluster(x, align)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extract.cluster_+3A_x">x</code></td>
<td>

<p>an object of class 'kmean', obtained from <code><a href="#topic+kmeans.run">kmeans.run</a></code> function.
</p>
</td></tr>
<tr><td><code id="extract.cluster_+3A_align">align</code></td>
<td>

<p>an object of class 'align', obtained from <code><a href="#topic+import.fasta">import.fasta</a></code> or <code><a href="#topic+import.msf">import.msf</a></code> function.  
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Extraction of the MSA of each cluster.
</p>


<h3>Value</h3>

<p>A named list of 'align' objects.
</p>


<h3>Author(s)</h3>

<p>Jean-Michel Becu
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Clustering human GPCRs in 4 groups with 100 runs of K-means 
# and extraction of the alignment of each cluster
aln &lt;- import.fasta(system.file("msa/human_gpcr.fa", package = "bios2mds"))
data(gpcr)
kmeans &lt;- kmeans.run(gpcr$mmds$sapiens.active$coord, nb.clus = 4, nb.run = 100)
clusAlign &lt;- extract.cluster(kmeans,aln)

</code></pre>

<hr>
<h2 id='gpcr'>
Pre-analyzed G-Protein-Coupled Receptor (GPCR) data set
</h2><span id='topic+gpcr'></span>

<h3>Description</h3>

<p>This data set was obtained by the <code>bios2mds</code> analysis of the two multiple sequence alignment files provided in /msa. The MSA files were prepared as previously described by Deville et al. (2009) and updated with the July 2009 release of Uniprot (<a href="http://www.uniprot.org">http://www.uniprot.org</a>). They correspond to non-redundant sets of non-olfactory class A G-protein-coupled receptors (GPCRs) from <em>H. sapiens</em> and <em>D. melanogaster</em> (283 and 59 sequences, respectively). The data sets from <em>H. sapiens</em> and <em>D. melanogaster</em> constitute the active and supplementary data sets, respectively. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gpcr)</code></pre>


<h3>Format</h3>

<p><code>gpcr</code> is a named list of three elements:
</p>

<dl>
<dt>dif</dt><dd>
<p>a named list containing two distance matrices calculated from the <code><a href="#topic+mat.dif">mat.dif</a></code> function (distances based on difference scores) with default parameters:
</p>

<dl>
<dt>sapiens.sapiens</dt><dd><p> a 283 by 283 matrix of difference scores between the 283 aligned sequences of human GPCRs.</p>
</dd>
<dt>melanogaster.sapiens</dt><dd><p> a 59 by 283 matrix of difference scores between the aligned sequences of GPCRs from <em>H. sapiens</em> and <em>D. melanogaster</em> (283 and 59 sequences, respectively)..</p>
</dd>
</dl>

</dd>
</dl>


<dl>
<dt>dis</dt><dd>
<p>a named list containing sixteen distance matrices calculated from the <code><a href="#topic+mat.dis">mat.dis</a></code>
function (distances based on dissimilarity scores) for the eight substitution matrices in <code><a href="#topic+sub.mat">sub.mat</a></code> (other parameters by default):
</p>

<dl>
<dt>sapiens.sapiens$*</dt><dd><p> are 283 by 283 matrices of dissimilarity scores between the 283 aligned sequences of human GPCRs.
</p>

<dl>
<dt>sapiens.sapiens$PAM40</dt><dd><p> is calculated with PAM40.</p>
</dd>
<dt>sapiens.sapiens$PAM80</dt><dd><p> is calculated with PAM80.</p>
</dd>
<dt>sapiens.sapiens$PAM120</dt><dd><p> is calculated with PAM120.</p>
</dd>
<dt>sapiens.sapiens$PAM250</dt><dd><p> is calculated with PAM250.</p>
</dd>
<dt>sapiens.sapiens$BLOSUM30</dt><dd><p> is calculated with BLOSUM30.</p>
</dd>
<dt>sapiens.sapiens$BLOSUM45</dt><dd><p> is calculated with BLOSUM45.</p>
</dd>
<dt>sapiens.sapiens$BLOSUM62</dt><dd><p> is calculated with BLOSUM62.</p>
</dd>
<dt>sapiens.sapiens$BLOSUM80</dt><dd><p> is calculated with BLOSUM80.</p>
</dd>
<dt>sapiens.sapiens$GONNET</dt><dd><p> is calculated with GONNET.</p>
</dd>
<dt>sapiens.sapiens$JTT</dt><dd><p> is calculated with JTT.</p>
</dd>
<dt>sapiens.sapiens$JTT_TM</dt><dd><p> is calculated with JTT_TM.</p>
</dd>
<dt>sapiens.sapiens$PHAT</dt><dd><p> is calculated with PHAT.</p>
</dd>
</dl>

</dd>
<dt>melanogaster.sapiens$*</dt><dd><p> are 59 by 283 matrices of dissimilarity scores between the 283 aligned sequences from <em>H. sapiens</em> and <em>D. melanogaster</em>
</p>

<dl>
<dt>melanogaster.sapiens$PAM40</dt><dd><p> is calculated with PAM40.</p>
</dd>
<dt>melanogaster.sapiens$PAM80</dt><dd><p> is calculated with PAM80.</p>
</dd>
<dt>melanogaster.sapiens$PAM120</dt><dd><p> is calculated with PAM120.</p>
</dd>
<dt>melanogaster.sapiens$PAM250</dt><dd><p> is calculated with PAM250.</p>
</dd>
<dt>melanogaster.sapiens$BLOSUM30</dt><dd><p> is calculated with BLOSUM30.</p>
</dd>
<dt>melanogaster.sapiens$BLOSUM45</dt><dd><p> is calculated with BLOSUM45.</p>
</dd>
<dt>melanogaster.sapiens$BLOSUM62</dt><dd><p> is calculated with BLOSUM62.</p>
</dd>
<dt>melanogaster.sapiens$BLOSUM80</dt><dd><p> is calculated with BLOSUM80.</p>
</dd>
<dt>melanogaster.sapiens$GONNET</dt><dd><p> is calculated with GONNET.</p>
</dd>
<dt>melanogaster.sapiens$JTT</dt><dd><p> is calculated with JTT.</p>
</dd>
<dt>melanogaster.sapiens$JTT_TM</dt><dd><p> is calculated with JTT_TM.</p>
</dd>
<dt>melanogaster.sapiens$PHAT</dt><dd><p> is calculated with PHAT.</p>
</dd>
</dl>

</dd>
</dl>

</dd>
</dl>


<dl>
<dt>mmds</dt><dd>
<p>a typical example of metric MDS analysis with the <code><a href="#topic+mmds">mmds</a></code> function of <code>bios2mds</code> using GPCRs from <em>H. sapiens</em> as active data and GPCRs from <em>D. melanogaster</em>, <em>N. vectensis</em> and <em>C. intestinalis</em> for supplementary data.
</p>

<dl>
<dt>sapiens.active</dt><dd><p> metric MDS analysis of active data from <em>H. sapiens</em>.</p>
</dd>
<dt>melanogaster.project</dt><dd><p> projection of supplementary data from <em>D. melanogaster</em> onto the human active space.</p>
</dd>
<dt>vectensis.project</dt><dd><p>projection of supplementary data from <em>N. vectensis</em> onto active the human space.</p>
</dd>
<dt>intestinalis.project</dt><dd><p>projection of supplementary data from <em>C. intestinalis</em> onto the human active space.</p>
</dd>
</dl>

</dd>
</dl>



<h3>Source</h3>

<p>Deville J, Rey J and Chabbert M (2009) An indel in transmembrane helix 2 helps to trace
the molecular evolution of class A G-protein-coupled receptors. <em>J Mol Evol</em> <b>68</b>: 475-
489.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># loading gpcr
data(gpcr)

# displaying the matrix of differences scores between GPCRs sequences
# from H. sapiens
gpcr$dif$sapiens.sapiens

# displaying the matrix of dissimilarity scores between GPCRs sequences
# from H. sapiens
gpcr$dis$sapiens.sapiens$PAM250

# displaying the matrix of dissimilarity scores between GPCRs sequences
# from H. sapiens and D. Melanogaster calculated with the BLOSUM45 matrix
gpcr$dis$melanogaster.sapiens$BLOSUM45

# displaying mmds analysis of the MSA of GPCRs from H. sapiens
# and D. Melanogaster
gpcr$mmds$sapiens.active
gpcr$mmds$melanogaster.project
</code></pre>

<hr>
<h2 id='import.fasta'>
Reads a file in FASTA format
</h2><span id='topic+import.fasta'></span>

<h3>Description</h3>

<p>Reads a Multiple Sequence Alignment (MSA) file in FASTA format (.fasta or .fa extension).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import.fasta(file, aa.to.upper = TRUE, gap.to.dash = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import.fasta_+3A_file">file</code></td>
<td>

<p>a string of characters to indicate the name of the MSA file to be read.
</p>
</td></tr>
<tr><td><code id="import.fasta_+3A_aa.to.upper">aa.to.upper</code></td>
<td>

<p>a logical value indicating whether amino acids should be converted to upper case (TRUE) or not (FALSE). Default is TRUE. 
</p>
</td></tr>
<tr><td><code id="import.fasta_+3A_gap.to.dash">gap.to.dash</code></td>
<td>

<p>a logical value indicating whether the dot (.) and tilde (<code class="reqn">\sim</code>) gap symbols should be converted
to dash (-) character (TRUE) or not (FALSE). Default is TRUE.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Initially, FASTA (for FAST-ALL) was the input format of the FASTA program, used for protein comparison and searching in databases.
Presently, FASTA format is a standard format for biological sequences.
</p>
<p>The FASTA formatted file of a single sequence displays:
</p>

<ul>
<li><p> a single-line description beginning with a greater-than (&gt;) symbol. The following word is the identifier.
</p>
</li>
<li><p> followed by any number of lines, representing biological sequence.
</p>
</li></ul>

<p>For multiple alignments, the FASTA formatted sequences are concatenated to create a multiple
FASTA format.
</p>


<h3>Value</h3>

<p>A object of class 'align', which is a named list whose elements correspond to sequences, in the form of character vectors.
</p>


<h3>Note</h3>

<p>For further information about FASTA format, see: <a href="http://www.ncbi.nlm.nih.gov/BLAST/fasta.shtml">http://www.ncbi.nlm.nih.gov/BLAST/fasta.shtml</a>
</p>


<h3>Author(s)</h3>

<p>Julien Pele
</p>


<h3>References</h3>

<p>Pearson WR and Lipman DJ (1988) Improved tools for biological sequence comparison.
<em>Proc Natl Acad Sci U S A</em> <b>27</b>:2444-2448.
</p>


<h3>See Also</h3>

<p><code>read.fasta</code> function from <code>bio3d</code> package.<br />
<code>read.fasta</code> function from <code>seqinr</code> package.<br />
<code>read.FASTA</code> function from <code>aaMI</code> package (archived).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># reading of the multiple sequence alignment of human GPCRS in FASTA format:
aln &lt;- import.fasta(system.file("msa/human_gpcr.fa", package = "bios2mds"))
</code></pre>

<hr>
<h2 id='import.msf'>
Reads a multiple sequence alignment file in MSF format
</h2><span id='topic+import.msf'></span>

<h3>Description</h3>

<p>Reads a Multiple Sequence Alignment (MSA) file in MSF format (.msf extension).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import.msf(file, aa.to.upper = TRUE, gap.to.dash = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import.msf_+3A_file">file</code></td>
<td>

<p>a string of characters to indicate the name of the MSA file to be read.
</p>
</td></tr>
<tr><td><code id="import.msf_+3A_aa.to.upper">aa.to.upper</code></td>
<td>

<p>a logical value indicating whether amino acids should be converted to upper case (TRUE) or not (FALSE). Default is TRUE.
</p>
</td></tr>
<tr><td><code id="import.msf_+3A_gap.to.dash">gap.to.dash</code></td>
<td>

<p>a logical value indicating whether the dot (.) and tilde (<code class="reqn">\sim</code>) gap symbols should be converted
to the dash (-) character (TRUE) or not (FALSE). Default is TRUE.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Initially, Multiple Sequence Format (MSF) was the multiple sequence alignment format of
the Wisconsin Package (WP) or GCG (Genetic Computer Group). This package is a suite of over 130
sequence analysis programs for database searching, secondary structure prediction
or sequence alignment. Presently, numerous multiple sequence alignment editors
(Jalview and GeneDoc for example) can read and write MSF files.
</p>
<p>MSF file displays several specificities:
</p>

<ul>
<li><p> a header containing sequence identifiers and characteristics (length, check and weight).
</p>
</li>
<li><p> a separator symbolized by 2 slashes (//).
</p>
</li>
<li><p> sequences of identifiers, displayed by consecutive blocks.
</p>
</li></ul>



<h3>Value</h3>

<p>A object of class 'align', which is a named list whose elements correspond to sequences, in the form of character vectors.
</p>


<h3>Note</h3>

<p><code>import.msf</code> checks the presence of duplicated identifiers in header. Sequences whose
identifiers are missing in header are ignored.
</p>


<h3>Author(s)</h3>

<p>Julien Pele
</p>


<h3>See Also</h3>

<p><code>read.alignment</code> function from <code>seqinr</code> package.<br />
<code>read.GDoc</code> function from <code>aaMI</code> package (archived).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># reading of the multiple sequence alignment of human GPCRs in MSF format:
aln &lt;- import.msf(system.file("msa/human_gpcr.msf", package = "bios2mds"))
</code></pre>

<hr>
<h2 id='kmeans.run'>
Multiple runs of K-means analysis
</h2><span id='topic+kmeans.run'></span>

<h3>Description</h3>

<p>Performs multiple runs of K-means clustering and analyzes data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kmeans.run(mat, nb.clus = 2, nb.run = 1000, iter.max = 10000,
method = "euclidean")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kmeans.run_+3A_mat">mat</code></td>
<td>

<p>a numeric matrix representing the coordinates of the elements after metric MDS analysis.
</p>
</td></tr>
<tr><td><code id="kmeans.run_+3A_nb.clus">nb.clus</code></td>
<td>

<p>a numeric value indicating the number of clusters. Default is 2. 
</p>
</td></tr>
<tr><td><code id="kmeans.run_+3A_nb.run">nb.run</code></td>
<td>

<p>a numeric value indicating the number of runs. Default is 1000.
</p>
</td></tr>
<tr><td><code id="kmeans.run_+3A_iter.max">iter.max</code></td>
<td>

<p>a numeric value indicating the maximum number of iterations for K-means. Default is 10000. 
</p>
</td></tr>
<tr><td><code id="kmeans.run_+3A_method">method</code></td>
<td>

<p>a string of characters to determine the distance to be used. This should be one of &quot;euclidean&quot;, &quot;maximum&quot;,
&quot;manhattan&quot;, &quot;canberra&quot;, &quot;binary&quot;, &quot;pearson&quot;, &quot;correlation&quot;, &quot;spearman&quot; or &quot;kendall&quot;. Default is &quot;euclidean&quot;.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The aim of K-means clustering is the partition of elements into a user-provided number of clusters. Several runs of K-means analysis on the same data may return different cluster assignments because the K-means procedure attributes random initial centroids for each run. The robustness of an assignment depends on its reproducibility. 
</p>
<p>The function <code>matchClasses</code> from the <code>e1071</code> package is used to compare the cluster assignments of the different runs and returns a score of agreement between them. The most frequent clustering solution is selected and used as a reference to assess the reproducibility of the analysis.
</p>
<p><code>kmeans.run</code> returns two lists. In either list, the clusters refer to those observed in the most frequent solution. The first list provides, for each element, the relative ratio of its assignment to each cluster in the different runs. The second list provides, for each cluster, the list of the assigned elements along with the relative assignment to this cluster in the different runs. 
</p>


<h3>Value</h3>

<p>A object of class 'kmean', which is a named list of two elements
</p>
<table>
<tr><td><code>elements</code></td>
<td>

<p>a named list of elements with the relative assignment of each element to each cluster.
</p>
</td></tr>
<tr><td><code>clusters</code></td>
<td>

<p>a named list of clusters with the elements assigned to this cluster in the most frequent solution and their relative assignment to this cluster in multiple runs. 
</p>
</td></tr>
</table>


<h3>Note</h3>

<p>During the K-means procedure, an empty cluster can be obtained if no objects are allocated to the
cluster. In <code>kmeans.run</code>, runs with empty clusters are discarded. 
</p>
<p><code>kmeans.run</code> requires <code>Kmeans</code> and <code>matchClasses</code> functions from <code>amap</code>
and <code>e1071</code> packages, respectively.
</p>


<h3>Author(s)</h3>

<p>Julien Pele
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Clustering human GPCRs in 4 groups with 100 runs of K-means
data(gpcr)
coord &lt;- gpcr$mmds$sapiens.active$coord
kmeans.run1 &lt;- kmeans.run(coord, nb.clus = 4, nb.run = 100)
kmeans.run1$clusters
kmeans.run1$elements
</code></pre>

<hr>
<h2 id='mat.dif'>
Matrices of difference scores between sequences 
</h2><span id='topic+mat.dif'></span>

<h3>Description</h3>

<p>Computes a matrix providing the distances based on the difference scores between sequences from two multiple sequence alignments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mat.dif(align1, align2, gap = FALSE, aa.strict = FALSE, sqrt = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mat.dif_+3A_align1">align1</code></td>
<td>

<p>a list of character vectors representing a first multiple sequence aligment.
</p>
</td></tr>
<tr><td><code id="mat.dif_+3A_align2">align2</code></td>
<td>

<p>a list of character vectors representing a second multiple sequence aligment.
</p>
</td></tr>
<tr><td><code id="mat.dif_+3A_gap">gap</code></td>
<td>

<p>a logical value indicating whether gap character should be taken as supplementary symbol
(TRUE) or not (FALSE). Default is FALSE.
</p>
</td></tr>
<tr><td><code id="mat.dif_+3A_aa.strict">aa.strict</code></td>
<td>

<p>a logical value indicating whether only strict amino acids should be taken into account (TRUE)
or not (FALSE). To be used only for amino acid sequences. Default is FALSE. 
</p>
</td></tr>
<tr><td><code id="mat.dif_+3A_sqrt">sqrt</code></td>
<td>

<p>a logical value indicating whether the distance should be equal to the squared root of the difference score (TRUE) or not
(FALSE). Default is FALSE.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>align1</code> and <code>align2</code> are identical, <code>mat.dif</code> computes the symetrical matrix of distances between each sequence of the alignment.
</p>
<p>Before using <code>mat.dif</code>, users must check the alignment of sequences within <code>align1</code> and <code>align2</code> and between <code>align1</code> and <code>align2</code>.
</p>


<h3>Value</h3>

<p>A named numeric matrix providing the difference-based distances between each pair of sequences from <code>align1</code> and <code>align2</code>. The number of rows and columns is identical to the number of sequences in <code>align1</code> and <code>align2</code>, respectively.
</p>


<h3>Author(s)</h3>

<p>Julien Pele and Jean-Michel Becu
</p>


<h3>See Also</h3>

<p><code>identity</code> function from <code>bio3d</code> package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># calculating the matrix of distances based on the difference scores
# between GPCRs sample from H. sapiens and D. melanogaster:
aln_human &lt;- import.fasta(system.file("msa/human_gpcr.fa", package = "bios2mds"))
aln_drome &lt;- import.fasta(system.file("msa/drome_gpcr.fa", package = "bios2mds"))
mat.dif1 &lt;- mat.dif(aln_human[1:5], aln_drome[1:5])
mat.dif1
</code></pre>

<hr>
<h2 id='mat.dis'>
Matrices of dissimilarity scores between amino acid sequences
</h2><span id='topic+mat.dis'></span>

<h3>Description</h3>

<p>Computes a matrix providing the distances based on dissimilarity scores between sequences from two multiple sequence alignments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mat.dis(align1, align2, sub.mat.id = "PAM250", sqrt=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mat.dis_+3A_align1">align1</code></td>
<td>

<p>a list of character vectors representing a first multiple sequence aligment.
</p>
</td></tr>
<tr><td><code id="mat.dis_+3A_align2">align2</code></td>
<td>

<p>a list of character vectors representing a second multiple sequence aligment.
</p>
</td></tr>
<tr><td><code id="mat.dis_+3A_sub.mat.id">sub.mat.id</code></td>
<td>

<p>a string of characters indicating the amino acid substitution matrix used for calculation 
of the dissimilarity score. This should be one of &quot;PAM40&quot;, &quot;PAM80&quot;, &quot;PAM120&quot;, &quot;PAM160&quot;, &quot;PAM250&quot;, &quot;BLOSUM30&quot;, &quot;BLOSUM45&quot;, &quot;BLOSUM62&quot;, &quot;BLOSUM80&quot;, &quot;GONNET&quot;, &quot;JTT&quot;, &quot;JTT_TM&quot; and &quot;PHAT&quot;. 
The supported substitution matrices are in <code><a href="#topic+sub.mat">sub.mat</a></code>. Default is PAM250.
</p>
</td></tr>
<tr><td><code id="mat.dis_+3A_sqrt">sqrt</code></td>
<td>

<p>a logical value indicating whether the distance should be equal to the squared root of the difference score (TRUE) or not (FALSE). Default is FALSE.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The dissimilarity score between a sequence <em>i</em> from <code>align1</code> and a sequence <em>j</em> from <code>align2</code> is calculated with an amino acid substitution matrix from <code>sub.mat</code>. 
</p>
<p>If <code>align1</code> and <code>align2</code> are identical, <code>mat.dis</code> computes the symetrical matrix of distances between each sequence of the alignment.
</p>
<p>Before using <code>mat.dis</code>, users must check the alignment of sequences within <code>align1</code> and <code>align2</code> and between <code>align1</code> and <code>align2</code>.
</p>


<h3>Value</h3>

<p>A named numeric matrix providing the dissimilarity-based distances between each pair of sequences from <code>align1</code> and <code>align2</code>, based on the substitution matrix <code>sub.mat.id</code>. The number of rows and columns is identical to the number of sequences in <code>align1</code> and <code>align2</code>, respectively.
</p>


<h3>Author(s)</h3>

<p>Julien Pele and Jean-Michel Becu
</p>


<h3>Examples</h3>

<pre><code class='language-R'># calculating dissimilarity distances between GPCR sequences sample from 
#H. sapiens and D. melanogaster, based on the PAM250 matrix:
aln_human &lt;- import.fasta(system.file("msa/human_gpcr.fa", package = "bios2mds"))
aln_drome &lt;- import.fasta(system.file("msa/drome_gpcr.fa", package = "bios2mds"))
mat.dis1 &lt;- mat.dis(aln_human[1:5], aln_drome[1:5])
mat.dis1

# calculating dissimilarity distances between GPCRs sequences sample from 
#H. sapiens and D. melanogaster, based on the BLOSUM45 matrix:
aln_human &lt;- import.fasta(system.file("msa/human_gpcr.fa", package = "bios2mds"))
aln_drome &lt;- import.fasta(system.file("msa/drome_gpcr.fa", package = "bios2mds"))
mat.dis1 &lt;- mat.dis(aln_human[1:5], aln_drome[1:5], sub.mat.id = "BLOSUM45")
mat.dis1
</code></pre>

<hr>
<h2 id='mmds'>
Metric multidimensional scaling
</h2><span id='topic+mmds'></span>

<h3>Description</h3>

<p>Performs metric MultiDimensional Scaling (MDS) analysis of active elements.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mmds(active, pc = 3, group.file = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mmds_+3A_active">active</code></td>
<td>

<p>a numeric matrix of distances between active elements.
</p>
</td></tr>
<tr><td><code id="mmds_+3A_pc">pc</code></td>
<td>

<p>a numeric value indicating the number of principal components to be saved. Default is 3.
</p>
</td></tr>
<tr><td><code id="mmds_+3A_group.file">group.file</code></td>
<td>

<p>a string of characters to indicate the file name assigning groups and colors to each active element.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Metric multidimensional scaling is a statistical analysis technique aimed at analyzing a matrix of distances between 'active' elements. MDS maps the elements onto a low dimensional space (usually 2D or 3D),  In this space, elements are represented by points whose respective distances best approximate the initial distance. 
</p>
<p><code>active</code> must have some characteristics:
</p>

<ul>
<li> <p><code>active</code> represents the matrix of pairwise distances between active elements. The active matrix must be symmetric (square and equals to its transpose). The distances on the main diagonal must be equal to 0. The distances do not have to be Euclidean. They can just express a difference or dissimilarity score, with a 0 value between same elements and a positive value between different elements.
</p>
</li></ul>

<p>The method for the computation of metric MDS is described by Abdi (2007). Briefly, if <em>N</em> is the number of active sequences and <em>D</em> is the <em>N</em> by <em>N</em> matrix of the squared distances computed from the active matrix, the <code>mmds</code> function performs the following steps:
</p>

<dl>
<dt>(1)</dt><dd>
<p>Transforms <b>D</b> into a cross-product matrix <b>S</b>:
</p>
<p style="text-align: center;"><code class="reqn">\bold{S} = -0.5 [\bold{I} - \bold{m}^{T}\bold{1}] \times \bold{D} \times [\bold{I} - \bold{m}^{T}\bold{1}]^{T}</code>
</p>

<p>where <b>I</b> is the <em>N</em> by <em>N</em> identity matrix, <b>m</b> is the <em>N</em> by <em>1</em> matrix mass, where each mass equal to <em><code class="reqn">\frac{1}{N}</code></em> and <b>1</b> is an <em>N</em> by <em>N</em> matrix of ones.
</p>
</dd>
<dt>(2)</dt><dd>
<p>Transforms <b><em>S</em></b> into a factor matrix <b><em>F</em></b>:
</p>
<p style="text-align: center;"><code class="reqn">\bold{F} = \bold{M}^{\frac{-1}{2}}\bold{U\Lambda}^{\frac{1}{2}}</code>
</p>

<p>where <b>M</b> is the diag{<b>m</b>}, <b><em>U</em></b> is the eigenvector matrix and <em><code class="reqn">\bold{\Lambda}</code></em> is the diagonal matrix of the eigenvalues, such as <b><em>S</em></b> = <em><code class="reqn">\bold{{U{\Lambda}U}^T}</code></em>, where <em><code class="reqn">\bold{^T}</code></em> denotes the transposition operation.
</p>
</dd>
</dl>

<p>The eigenvectors of <b><em>S</em></b>, also called principal components (whose number is smaller or
equal to <b><em>N</em></b>), form the active space. <b><em>F</em></b> gives the coordinates of the
active elements in this space.
</p>


<h3>Value</h3>

<p>A object of class 'mmds', which is a named list of five elements:
</p>
<table>
<tr><td><code>eigen</code></td>
<td>

<p>a numeric vector of the eigenvalues.
</p>
</td></tr>
<tr><td><code>eigen.perc</code></td>
<td>

<p>a numeric vector of the relative eigenvalues (eigenvalues divided by the sum of the absolute eigenvalues).
</p>
</td></tr>
<tr><td><code>coord</code></td>
<td>

<p>a named numeric matrix representing the coordinates of active elements.
</p>
</td></tr>
<tr><td><code>group</code></td>
<td>

<p>a named string matrix representing the differents groups of elements and the associate color (Default is 'NoGroup' and 'black').
</p>
</td></tr>
<tr><td><code>col</code></td>
<td>

<p>a named string matrix representing, foreach named elements, the associate group and color (Default is 'NoGroup' and 'black').
</p>
</td></tr>
<tr><td><code>source</code></td>
<td>

<p>a named list with 2 elements, the matrix (<b>D</b>) of squared distance and the vector of mass <b>m</b>, that will used for projection of supplementary elements, if required, with the <code>mmds.project</code> function.
</p>
</td></tr>
</table>


<h3>Note</h3>

<p>If <code>active</code> do not contain names:<br />
A tag &quot;A&quot; followed by an incremented number names the rows and the columns of <code>active</code>.<br /> 
</p>


<h3>Author(s)</h3>

<p>Julien Pele and Jean-Michel Becu
</p>


<h3>References</h3>

<p>Abdi H (2007) Metric multidimensional scaling. In N.J. Salkind (Ed.):
<em>Encyclopedia of Measurement and Statistics.</em> Thousand Oaks (CA): Sage. pp. 598-605.
</p>
<p>For further information on multidimentional scaling:
</p>
<p>Takane Y, Jung S and Oshima-Takane Y (2009) Multidimensional scaling, in <em>Handbook
of quantitative methods in psychology</em>, eds Millsap R, Maydeu-Olivares A (Sage
Publications, London) pp. 219-242.
</p>
<p>Borg I and Groenen PJF (2005) Modern multidimensional scaling.
New York : Springer.
</p>
<p>Gower JC (1967) A comparison of some methods of cluster analysis.
<em>Biometrics</em> <b>23</b>:623-637.
</p>
<p>ToRgerson WS (1958) Theory and methods of scaling.
New York : Wiley.
</p>


<h3>See Also</h3>

<p><code>cmdscale</code> function from <code>stats</code> package.<br />
<code>dudi.pco</code>, <code>suprow</code> and <code>supcol</code> functions from <code>ade4</code> package.<br />
<code>PCA</code> function from <code>FactoMineR</code> package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># performing metric MDS of human GPCRs with projection of
# GPCRs from D. melanogaster as supplementary elements:
data(gpcr)
active &lt;- gpcr$dif$sapiens.sapiens
mmds1 &lt;- mmds(active = active)
mmds1$active.coord
</code></pre>

<hr>
<h2 id='mmds.2D.multi'>
Plots the mmds coordinates onto a 2D space
</h2><span id='topic+mmds.2D.multi'></span>

<h3>Description</h3>

<p>Displays a scatter plot of the active elements and the barycenter of supplementary elements, or of groups of supplementary elements, after a metric MDS analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mmds.2D.multi(x,project, title = NULL, axis = c(1, 2), xlim = NULL,
ylim = NULL, outfile.type = NULL,bary="p",
outfile.name = "mmds",new.plot = TRUE, active.col = x$col[,3],
active.alpha = 0.6, active.pch = 20, sup.pch = NULL , active.cex = 2,
sup.cex = 2, active.legend.cex = 2, sup.legend.cex = 2,
active.legend.lwd = 1, sup.legend.lwd = 2, active.lwd = 1, sup.lwd = 3,
legend = TRUE, active.legend.pos = "bottomleft",
sup.legend.pos = "bottomright", group.name = NULL,
ensemble.legend.name = "", group.col = NULL, outfile.width = NULL, outfile.height = NULL,
box.lwd = 1, cex.axis = 1, sup.legend.text = 1,
active.legend.text = 1, legend.axis = TRUE, grid = TRUE, axes = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mmds.2D.multi_+3A_x">x</code></td>
<td>

<p>an object of class 'mmds', obtained from <code><a href="#topic+mmds">mmds</a></code> function.
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_project">project</code></td>
<td>

<p>an object of class 'project' or a list of objects of class 'project', obtained from the <code><a href="#topic+mmds.project">mmds.project</a></code> function. In the list the order of 'project' objects is very important because if the <code>bary</code> option equal to &quot;l&quot;, the barycenters are linked in the same order as the list with the active barycenter linked as the last object.
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_title">title</code></td>
<td>

<p>a string of characters representing the title of the plot. Default is &quot;Metric MDS&quot;.  
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_bary">bary</code></td>
<td>

<p>a character to specify the representation of group barycenter with &quot;p&quot; for representation of group barycenter foreach 'project' and active space by dots and &quot;l&quot; for representation of barycenter linked foreach group in order of appearence in project list and with the barycenter of active group in last position. Default is &quot;p&quot;.
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_axis">axis</code></td>
<td>

<p>a numeric vector of length two representing the principal components displayed on the plot. Default is c(1, 2). 
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_xlim">xlim</code></td>
<td>

<p>a numeric vector representing the range for the x values. Default is full range.
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_ylim">ylim</code></td>
<td>

<p>a numeric vector representing the range for the y values. Default is full range.
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_outfile.type">outfile.type</code></td>
<td>

<p>a string indicating the extension type of the graph outfile. Default is NULL. If not NULL, this should be one of &quot;pdf&quot;, &quot;tiff&quot;, &quot;png&quot; or &quot;postscript&quot;. In this case, the parameter outfile.name and the forthcoming parameters are activated.
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_outfile.name">outfile.name</code></td>
<td>

<p>a string (&quot;mmds&quot;, default) indicating the name and directory of pdf graph outfile. The extension file is added automaticaly. See <code>outfile.type</code>.
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_new.plot">new.plot</code></td>
<td>

<p>a boolean indicating whether a new graphical device should be created (TRUE) or not (FALSE). Default is TRUE.
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_active.pch">active.pch</code></td>
<td>

<p>an integer indicating the symbol of active elements. Default is 20, corresponding to dots.
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_sup.pch">sup.pch</code></td>
<td>

<p>an integer vector representing the symbol for project barycenter. Default is NULL. 
If sup.pch is NULL then symbol are choose randomly.
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_active.col">active.col</code></td>
<td>

<p>a string of characters or character vector representing the color(s) of the active elements. Default is x$col[,3]. 
It corresponds either to the user-provided colors if the <code>col.group</code> function has been used previously, 
or to black (filled automaticaly with the <code>mmds</code> function).    
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_active.alpha">active.alpha</code></td>
<td>

<p>a numeric value indicating the alpha value for opacity of active objects. This
value must range from 0 (invisible) to 1 (full opacity). Default is 0.6.
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_active.cex">active.cex</code></td>
<td>

<p>a numeric value indicating the size of the active symbols. Default is 2. 
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_sup.cex">sup.cex</code></td>
<td>

<p>a numeric value indicating the size of the supplementary symbols. Default is 2.  
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_active.legend.cex">active.legend.cex</code></td>
<td>

<p>a numeric value indicating the size of active symbols in legend. Default is 2. 
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_sup.legend.cex">sup.legend.cex</code></td>
<td>

<p>a numeric value indicating the size of supplementary symbols in legend. Default is 2.
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_active.lwd">active.lwd</code></td>
<td>

<p>a numeric value indicating the width of active symbols. Default is 1.  
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_sup.lwd">sup.lwd</code></td>
<td>

<p>a numeric value indicating the width of supplementary symbols. Default is 3. 
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_active.legend.lwd">active.legend.lwd</code></td>
<td>

<p>a numeric value indicating the width of active symbols in legend. Default is 1.  
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_sup.legend.lwd">sup.legend.lwd</code></td>
<td>

<p>a numeric value indicating the width of supplementary objects in legend. Default is 4. 
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_legend">legend</code></td>
<td>

<p>a boolean indicating whether the legend should be displayed (TRUE) or not (FALSE). Default is TRUE.
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_active.legend.pos">active.legend.pos</code></td>
<td>

<p>a string indicating the position of the legend for active elements. Default is &quot;topleft&quot;. 
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_sup.legend.pos">sup.legend.pos</code></td>
<td>

<p>a string indicating the position of the legend for supplementary elements. Default is &quot;topright&quot;.  
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_group.name">group.name</code></td>
<td>

<p>a string vector indicating the names of the used groups. NULL vector allow to use all groups present in the 'mmds' object (x) and in all 'mmds.project' object. It corresponds either to the user-provided groups if the <code>col.group</code> function has been used previously, or to &quot;NoGroup&quot; (filled automaticaly with the <code>mmds</code> ans <code>mmds.project</code> functions). Each group in this vector must be presents in 'mmds' or 'mmds.project' objects and only group present in this vector will draw in the plot. Default is NULL.
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_ensemble.legend.name">ensemble.legend.name</code></td>
<td>

<p>a string vector indicating the names of the <code>project</code> and the name of active space. Default is &quot;&quot;.
It corresponds either to the names of project object ans in the last position of vector the name of the active space.
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_group.col">group.col</code></td>
<td>

<p>a string vector indicating the colors of appearance of the used groups. This vector must have the same length of the group.name vector. NULL vector associate automaticaly the color of groups in 'mmds' and 'mmds.project' object to the group in the group.name parameter. It corresponds either to the user-provided colors if the <code>col.group</code> function has been used previously, or to black (filled automaticaly with the <code>mmds</code> and <code>project.mmds</code> functions). Default is NULL.
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_outfile.width">outfile.width</code></td>
<td>

<p>a numeric value in inches indicating the width of graph outfile. Default differs by outfile.type. See <code>pdf</code>, <code>png</code>, <code>postscript</code>.  
The resolution for tiff and png figures is 150 dpi.  
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_outfile.height">outfile.height</code></td>
<td>

<p>a numeric value in inches indicating the height of graph outfile. Default differs by outfile.type. The resolution for tiff and png figures is 150 dpi. See <code>pdf</code>, codetiff, <code>png</code>, <code>postscript</code>.
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_box.lwd">box.lwd</code></td>
<td>

<p>a numeric value indicating the border width of graph box and legend box. Default is 1. 
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_cex.axis">cex.axis</code></td>
<td>

<p>a numeric value indicating the character size for the x and y axes. Default is 1.  
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_sup.legend.text">sup.legend.text</code></td>
<td>

<p>a numeric value indicating the character size of the supplementary tag in legend. Default is 1. 
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_active.legend.text">active.legend.text</code></td>
<td>

<p>a numeric value indicating the character size of active tag in legend. Default is 1.  
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_legend.axis">legend.axis</code></td>
<td>

<p>a boolean indicating whether axis name should be displayed (TRUE) or not (FALSE). Default is TRUE.
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_grid">grid</code></td>
<td>

<p>a boolean indicating whether grid should be displayed (TRUE) or not (FALSE). Default is TRUE.
</p>
</td></tr>
<tr><td><code id="mmds.2D.multi_+3A_axes">axes</code></td>
<td>

<p>a boolean indicating whether x and y axes should be displayed (TRUE) or not (FALSE). Default is TRUE.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>mmds.2D.plot</code> is used after the <code>col.group</code> function, 
the elements are colored by the color scheme provided in the .csv file (see <code>col.group</code> for details).
If the <code>col.group</code> function has not been used, the default colors are black and magenta for active and supplementary elements.
</p>
<p><code>mmds.2D.plot</code> helps identify patterns in data and compare active and supplementary elements.
</p>
<p><code>active.alpha</code> argument is helpful for visualization of supplementary elements because it allows the symbols of supplementary elements to be in the foreground as compared to active elements.
</p>


<h3>Value</h3>

<p>Produces a scatter plot on the active graphical device.
</p>


<h3>Note</h3>

<p><code>mmds.2D.multi</code> requires <code>alpha</code> function from <code>scales</code> package.
</p>


<h3>Author(s)</h3>

<p>Jean-Michel Becu
</p>


<h3>See Also</h3>

<p><code>plot.PCA</code> function from <code>FactoMineR</code> package.<br />
<code>png</code>, <code>pdf</code>, <code>postscript</code> functions (default R package).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># scatter plot of human GPCRs onto the first two axes obtained
# from MDS analysis with projection of GPCRs from N.vectensis 
# and C.intestinalis as supplementary elements:
data(gpcr)

mmds_human &lt;- gpcr$mmds$sapiens.active
project_vectensis &lt;-gpcr$mmds$vectensis.project
project_intestinalis &lt;-gpcr$mmds$intestinalis.project

mmds.2D.multi(mmds_human,project=list(project_vectensis,project_intestinalis),
bary='l',cex.axis=0.01,active.cex = 1, sup.cex = 1,active.lwd=1.5,sup.lwd=3,
ensemble.legend.name=c('nemve','inte','human'),legend=FALSE,title='multi_human')

# with selected group

mmds.2D.multi(mmds_human,project=list(project_vectensis,project_intestinalis),
bary='l',cex.axis=0.01,active.cex = 1, sup.cex = 1,active.lwd=1.5,sup.lwd=3,
ensemble.legend.name=c('nemve','inte','human'),legend=FALSE,title='multi_human',
group.name=c('SO','PEP','OPN','ADENO'),group.col=c("red","forestgreen","orange","maroon"))

</code></pre>

<hr>
<h2 id='mmds.2D.plot'>
Plots the mmds coordinates onto a 2D space
</h2><span id='topic+mmds.2D.plot'></span>

<h3>Description</h3>

<p>Displays a scatter plot of the active elements and, if present, of supplementary elements, after a metric MDS analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mmds.2D.plot(x,project = NULL, title = NULL, axis = c(1, 2), xlim = NULL,
ylim = NULL, outfile.type = NULL, outfile.name = "mmds",new.plot = TRUE,
active.col = x$col[,3], active.alpha = 1, sup.col = project$col[,3], 
active.pch = 20, sup.pch = 3, active.lab = FALSE, sup.lab = FALSE, 
active.cex = 2, sup.cex = 2, active.legend.cex = 2, sup.legend.cex = 2,
active.legend.lwd = 1, sup.legend.lwd = 2, active.lwd = 1, sup.lwd = 4,
legend = TRUE, active.legend.pos = "bottomleft",
sup.legend.pos = "bottomright", active.legend.name = x$group[,1],
sup.legend.name = project$group[,1], active.legend.col = x$group[,2],
sup.legend.col = project$group[,2], outfile.width = NULL, outfile.height = NULL,
box.lwd = 1, cex.axis = 1, cex.lab = 1, sup.legend.text = 1,
active.legend.text = 1, legend.axis = TRUE, grid = TRUE, axes = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mmds.2D.plot_+3A_x">x</code></td>
<td>

<p>an object of class 'mmds', obtained from <code><a href="#topic+mmds">mmds</a></code> function.
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_project">project</code></td>
<td>

<p>an object of class 'project', obtained from <code><a href="#topic+mmds.project">mmds.project</a></code> function.
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_title">title</code></td>
<td>

<p>a string of characters representing the title of the plot. Default is &quot;Metric MDS&quot;.  
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_axis">axis</code></td>
<td>

<p>a numeric vector of length two representing the principal components displayed on the plot. Default is c(1, 2). 
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_xlim">xlim</code></td>
<td>

<p>a numeric vector representing the range for the x values. Default is full range.
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_ylim">ylim</code></td>
<td>

<p>a numeric vector representing the range for the y values. Default is full range.
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_outfile.type">outfile.type</code></td>
<td>

<p>a string indicating the extension type of the graph outfile. Default is NULL. If not NULL, this should be one of &quot;pdf&quot;, &quot;tiff&quot;, &quot;png&quot; or &quot;postscript&quot;. In this case, the parameter outfile.name and the forthcoming parameters are activated.
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_outfile.name">outfile.name</code></td>
<td>

<p>a string (&quot;mmds&quot;, default) indicating the name and directory of pdf graph outfile. The extension file is added automaticaly. See <code>outfile.type</code>.
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_new.plot">new.plot</code></td>
<td>

<p>a boolean indicating whether a new graphical device should be created (TRUE) or not (FALSE). Default is TRUE.
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_active.pch">active.pch</code></td>
<td>

<p>an integer indicating the symbol of active elements. Default is 20, corresponding to dots.
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_sup.pch">sup.pch</code></td>
<td>

<p>an integer indicating the symbol of supplementary elements. Default is 3, corresponding to crosses.
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_active.col">active.col</code></td>
<td>

<p>a string of characters or character vector representing the color(s) of the active elements. Default is x$col[,3]. 
It corresponds either to the user-provided colors if the <code>col.group</code> function has been used previously, 
or to black (filled automaticaly with the <code>mmds</code> function).    
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_active.alpha">active.alpha</code></td>
<td>

<p>a numeric value indicating the alpha value for opacity of active objects. This
value must range from 0 (invisible) to 1 (full opacity). Default is 1.
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_sup.col">sup.col</code></td>
<td>

<p>a string or character vector representing the color(s) of supplementary elements. Default is project$col[,3]. 
It corresponds either to the user-provided colors if the <code>col.group</code> function has been used previously, 
or to magenta (filled automaticaly with the <code>mmds</code> function). 
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_active.lab">active.lab</code></td>
<td>

<p>a boolean indicating whether labels of active elements should be displayed (TRUE) or not (FALSE). Default is FALSE.
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_sup.lab">sup.lab</code></td>
<td>

<p>a boolean indicating whether labels of supplementary elements should be displayed (TRUE) or not (FALSE). Default is FALSE.
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_active.cex">active.cex</code></td>
<td>

<p>a numeric value indicating the size of the active symbols. Default is 2. 
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_sup.cex">sup.cex</code></td>
<td>

<p>a numeric value indicating the size of the supplementary symbols. Default is 2.  
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_active.legend.cex">active.legend.cex</code></td>
<td>

<p>a numeric value indicating the size of active symbols in legend. Default is 2. 
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_sup.legend.cex">sup.legend.cex</code></td>
<td>

<p>a numeric value indicating the size of supplementary symbols in legend. Default is 2.
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_active.lwd">active.lwd</code></td>
<td>

<p>a numeric value indicating the width of active symbols. Default is 1.  
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_sup.lwd">sup.lwd</code></td>
<td>

<p>a numeric value indicating the width of supplementary symbols. Default is 4. 
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_active.legend.lwd">active.legend.lwd</code></td>
<td>

<p>a numeric value indicating the width of active symbols in legend. Default is 1.  
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_sup.legend.lwd">sup.legend.lwd</code></td>
<td>

<p>a numeric value indicating the width of supplementary objects in legend. Default is 4. 
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_legend">legend</code></td>
<td>

<p>a boolean indicating whether the legend should be displayed (TRUE) or not (FALSE). Default is TRUE.
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_active.legend.pos">active.legend.pos</code></td>
<td>

<p>a string indicating the position of the legend for active elements. Default is &quot;topleft&quot;. 
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_sup.legend.pos">sup.legend.pos</code></td>
<td>

<p>a string indicating the position of the legend for supplementary elements. Default is &quot;topright&quot;.  
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_active.legend.name">active.legend.name</code></td>
<td>

<p>a string vector indicating the names of the <code>active</code> groups. Default is x$group[,1]. 
It corresponds either to the user-provided groups if the <code>col.group</code> function has been used previously, 
or to &quot;NoGroup&quot; (filled automaticaly with the <code>mmds</code> function).
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_sup.legend.name">sup.legend.name</code></td>
<td>

<p>a string vector indicating the names of the <code>sup</code> groups. Default is project$group[,1]. 
It corresponds either to the user-provided groups if the <code>col.group</code> function has been used previously, 
or to &quot;NoGroup&quot; (filled automaticaly with the <code>mmds</code> function).
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_active.legend.col">active.legend.col</code></td>
<td>

<p>a string vector indicating the colors of the different <code>active</code> groups. Default is x$group[,2]. It corresponds either to the user-provided colors if the <code>col.group</code> function has been used previously, 
or to black (filled automaticaly with the <code>mmds</code> function).
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_sup.legend.col">sup.legend.col</code></td>
<td>

<p>a string vector indicating the colors of the different <code>sup</code> groups. Default is project$group[,2]. It corresponds either to the user-provided colors if the <code>col.group</code> function has been used previously, 
or to magenta (filled automaticaly with the <code>mmds</code> function).
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_outfile.width">outfile.width</code></td>
<td>

<p>a numeric value in inches indicating the width of graph outfile. Default differs by outfile.type. See <code>pdf</code>, <code>png</code>, <code>postscript</code>.  
The resolution for tiff and png figures is 150 dpi.  
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_outfile.height">outfile.height</code></td>
<td>

<p>a numeric value in inches indicating the height of graph outfile. Default differs by outfile.type. The resolution for tiff and png figures is 150 dpi. See <code>pdf</code>, codetiff, <code>png</code>, <code>postscript</code>.
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_box.lwd">box.lwd</code></td>
<td>

<p>a numeric value indicating the border width of graph box and legend box. Default is 1. 
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_cex.axis">cex.axis</code></td>
<td>

<p>a numeric value indicating the character size for the x and y axes. Default is 1.  
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_cex.lab">cex.lab</code></td>
<td>

<p>a numeric value indicating the character size for the labels of the x and y axes. Default is 1.  
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_sup.legend.text">sup.legend.text</code></td>
<td>

<p>a numeric value indicating the character size of the supplementary tag in legend. Default is 1. 
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_active.legend.text">active.legend.text</code></td>
<td>

<p>a numeric value indicating the character size of active tag in legend. Default is 1.  
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_legend.axis">legend.axis</code></td>
<td>

<p>a boolean indicating whether axis name should be displayed (TRUE) or not (FALSE). Default is TRUE.
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_grid">grid</code></td>
<td>

<p>a boolean indicating whether grid should be displayed (TRUE) or not (FALSE). Default is TRUE.
</p>
</td></tr>
<tr><td><code id="mmds.2D.plot_+3A_axes">axes</code></td>
<td>

<p>a boolean indicating whether x and y axes should be displayed (TRUE) or not (FALSE). Default is TRUE.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>mmds.2D.plot</code> is used after the <code>col.group</code> function, 
the elements are colored by the color scheme provided in the .csv file (see <code>col.group</code> for details).
If the <code>col.group</code> function has not been used, the default colors are black and magenta for active and supplementary elements.
</p>
<p><code>mmds.2D.plot</code> helps identify patterns in data and compare active and supplementary elements.
</p>
<p><code>active.alpha</code> argument is helpful for visualization of supplementary elements because it allows the symbols of supplementary elements to be in the foreground as compared to active elements.
</p>


<h3>Value</h3>

<p>Produces a scatter plot on the active graphical device.
</p>


<h3>Note</h3>

<p><code>mmds.2D.plot</code> requires <code>alpha</code> function from <code>scales</code> package.
</p>


<h3>Author(s)</h3>

<p>Julien Pele and Jean-Michel Becu
</p>


<h3>See Also</h3>

<p><code>plot.PCA</code> function from <code>FactoMineR</code> package.<br />
<code>png</code>, <code>pdf</code>, <code>postscript</code> functions (default R package).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># scatter plot of human GPCRs onto the first two axes obtained from MDS analysis
# with projection of GPCRs from D. melanogaster as supplementary elements:
data(gpcr)
active &lt;- gpcr$dif$sapiens.sapiens
mmds_active &lt;- mmds(active,group.file=system.file(
"csv/human_gpcr_group.csv",package = "bios2mds"))
mmds.2D.plot(mmds_active, active.alpha = 0.5, active.lab = TRUE)

# with group information
sup &lt;- gpcr$dif$melanogaster.sapiens
mmds_sup &lt;- mmds.project(mmds_active, sup,group.file=system.file(
"csv/drome_gpcr_group.csv",package = "bios2mds"))
mmds.2D.plot(mmds_active,mmds_sup)
</code></pre>

<hr>
<h2 id='mmds.3D.plot'>
Display the mmds coordinates onto a 3D space
</h2><span id='topic+mmds.3D.plot'></span>

<h3>Description</h3>

<p>Displays a 3D plot of the active elements and, if present, of supplementary elements, after a metric MDS analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mmds.3D.plot(x, project = NULL, title = NULL, axis = c(1:3), 
active.type = "s", sup.type = "p", active.size = 2, radius = 0.005,
sup.size = 10, active.col = x$col[,3], sup.col = project$col[,3], 
box = TRUE, axes = TRUE, new.plot = TRUE, label = TRUE,
xlim = NULL, ylim = NULL, zlim = NULL, box.lwd = 2,
box.antialias = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mmds.3D.plot_+3A_x">x</code></td>
<td>

<p>an object of class 'mmds', obtained from <code><a href="#topic+mmds">mmds</a></code> function.
</p>
</td></tr>
<tr><td><code id="mmds.3D.plot_+3A_title">title</code></td>
<td>

<p>a string of characters representing the title of the plot. Default is &quot;Metric MDS&quot;.  
</p>
</td></tr>
<tr><td><code id="mmds.3D.plot_+3A_project">project</code></td>
<td>

<p>an object of class 'project', obtained from <code><a href="#topic+mmds.project">mmds.project</a></code> function, or a list of object of class 'project'.
</p>
</td></tr>
<tr><td><code id="mmds.3D.plot_+3A_axis">axis</code></td>
<td>

<p>a numeric vector of length three representing the principal components displayed on the plot. Default is c(1:3). 
</p>
</td></tr>
<tr><td><code id="mmds.3D.plot_+3A_active.type">active.type</code></td>
<td>

<p>an character indicating the symbol of active elements. This should be one of &quot;s&quot; for spheres, &quot;p&quot; for points, &quot;l&quot; for lines, &quot;h&quot; for line segments from z=0 and &quot;n&quot; for none. Default is &quot;s&quot;, corresponding to spheres.
</p>
</td></tr>
<tr><td><code id="mmds.3D.plot_+3A_sup.type">sup.type</code></td>
<td>

<p>an character indicating the symbol of supplementary elements. This should be one of &quot;s&quot; for spheres, &quot;p&quot; for points, &quot;l&quot; for lines, &quot;h&quot; for line segments from z=0 and &quot;n&quot; for none. Default is &quot;p&quot;, corresponding to points.
</p>
</td></tr>
<tr><td><code id="mmds.3D.plot_+3A_active.col">active.col</code></td>
<td>

<p>a string of characters or character vector representing the color(s) of the active elements. Default is x$col[,3]. 
It corresponds either to the user-provided colors if the <code>col.group</code> function has been used previously, 
or to black (filled automaticaly with the <code>mmds</code> function). 
</p>
</td></tr>
<tr><td><code id="mmds.3D.plot_+3A_sup.col">sup.col</code></td>
<td>

<p>a string or character vector representing the color(s) of supplementary elements. Default is project$col[,3]. 
It corresponds either to the user-provided colors if the <code>col.group</code> function has been used previously, 
or to magenta (filled automaticaly with the <code>mmds</code> function)
</p>
</td></tr>
<tr><td><code id="mmds.3D.plot_+3A_active.size">active.size</code></td>
<td>

<p>a numeric value indicating the size of active symbols. Default is 2.
</p>
</td></tr>
<tr><td><code id="mmds.3D.plot_+3A_sup.size">sup.size</code></td>
<td>

<p>a numeric value indicating the size of supplementary symbols. Default is 20. 
</p>
</td></tr>
<tr><td><code id="mmds.3D.plot_+3A_box">box</code></td>
<td>

<p>a boolean indicating whether the box should be displayed (TRUE) or not (FALSE). Default is TRUE.
</p>
</td></tr>
<tr><td><code id="mmds.3D.plot_+3A_axes">axes</code></td>
<td>

<p>a boolean indicating whether axes should be displayed (TRUE) or not (FALSE). Default is TRUE. 
</p>
</td></tr>
<tr><td><code id="mmds.3D.plot_+3A_radius">radius</code></td>
<td>

<p>a numeric value indicating the radius of spheres symbols only. If x.type equal to &quot;s&quot; the x.size parameter was inactivated. Default is 0.01.
</p>
</td></tr>
<tr><td><code id="mmds.3D.plot_+3A_new.plot">new.plot</code></td>
<td>

<p>a boolean indicating whether a new 3D plot create/replace active 3D device (TRUE) or not to insert in it (FALSE). Default is TRUE.
</p>
</td></tr>
<tr><td><code id="mmds.3D.plot_+3A_label">label</code></td>
<td>

<p>a boolean indicating whether the label axes should be displayed (TRUE) or not (FALSE). Default is TRUE.
</p>
</td></tr>
<tr><td><code id="mmds.3D.plot_+3A_xlim">xlim</code></td>
<td>

<p>a numeric vector representing the range for the x values. Default is full range.
</p>
</td></tr>
<tr><td><code id="mmds.3D.plot_+3A_ylim">ylim</code></td>
<td>

<p>a numeric vector representing the range for the y values. Default is full range.
</p>
</td></tr>
<tr><td><code id="mmds.3D.plot_+3A_zlim">zlim</code></td>
<td>

<p>a numeric vector representing the range for the z values. Default is full range.
</p>
</td></tr>
<tr><td><code id="mmds.3D.plot_+3A_box.lwd">box.lwd</code></td>
<td>

<p>a numeric value indicating the width of box and axes lines. Default is 2.
</p>
</td></tr>
<tr><td><code id="mmds.3D.plot_+3A_box.antialias">box.antialias</code></td>
<td>

<p>a boolean specifying if box and axes lines should be antiliased. Default is TRUE.
</p>
</td></tr>
<tr><td><code id="mmds.3D.plot_+3A_...">...</code></td>
<td>

<p>additional parameters which will be passed to <code>par3d</code>, <code>material3d</code> and <code>decorate3d</code> of <code>rgl</code> package.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>mmds.3D.plot</code> is used after the <code>col.group</code> function, 
the elements are colored by the color scheme provided in the .csv file (see <code>col.group</code> for details).
If the <code>col.group</code> function has not been used, the default colors are black and magenta for active and supplementary elements.
</p>
<p><code>mmds.3D.plot</code> helps identify patterns in data and compare active and supplementary elements.
</p>


<h3>Value</h3>

<p>Produces a 3D plot on graphical device.
</p>


<h3>Note</h3>

<p><code>mmds.3D.plot</code> requires <code>plot3D</code> function from <code>rgl</code> package.
See <code>rgl</code> documentation to supplementary function like <code>snapshot3D</code> to save image file of 3D device.
</p>


<h3>Author(s)</h3>

<p>Jean-Michel Becu
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 3D plot of human GPCRs onto the first three axes obained from MDS analysis
# with projection of GPCRs from D. melanogaster as supplementary elements:
data(gpcr)
mmds.3D.plot(gpcr$mmds$sapiens.active,active.type="p",label=FALSE,lit=FALSE,
point_antialias=TRUE,box.lwd=3,sup.size=4.3,active.size=4.3)
bbox3d(shininess=0.5)

</code></pre>

<hr>
<h2 id='mmds.plot'>
Plots a summary of the mmds results
</h2><span id='topic+mmds.plot'></span>

<h3>Description</h3>

<p>Displays one scree plot and three scatter plots of <code><a href="#topic+mmds">mmds</a></code> results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mmds.plot(x, project = NULL, new.plot = TRUE, pdf.file = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mmds.plot_+3A_x">x</code></td>
<td>

<p>an object of class 'mmds', obtained from <code><a href="#topic+mmds">mmds</a></code>.
</p>
</td></tr>
<tr><td><code id="mmds.plot_+3A_project">project</code></td>
<td>

<p>an object of class 'project', obtained from <code><a href="#topic+mmds.project">mmds.project</a></code> function.
</p>
</td></tr>
<tr><td><code id="mmds.plot_+3A_new.plot">new.plot</code></td>
<td>

<p>a boolean indicating whether a new graphical device should be created (TRUE) or not (FALSE). Default is TRUE.
</p>
</td></tr>
<tr><td><code id="mmds.plot_+3A_pdf.file">pdf.file</code></td>
<td>

<p>a string indicating the name and directory of the pdf graph outfile. Default is NULL. If this parameter is not
NULL, the parameter <code>new.plot</code> is inactivated.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>mmds.plot</code> is a wrapper calling of both <code><a href="#topic+scree.plot">scree.plot</a></code> and <code><a href="#topic+mmds.2D.plot">mmds.2D.plot</a></code>.
It produces a 2x2 plot with one scree plot of the relative eigenvalues, in the upper left, and three scatter plots. The three scatter plots are generated as follows:
</p>

<ul>
<li><p> scatter plot of the elements on the first and second components in the upper right.
</p>
</li>
<li><p> scatter plot of the elements on the first and third components in the lower left.
</p>
</li>
<li><p> scatter plot of the elements on the second and third components in the lower right.
</p>
</li></ul>

<p>If object <code>x</code> contains supplementary elements, they are also projected onto the three scatter plots.
The active and supplementary elements are represented by dots and crosses, respectively.
The <code>color.group</code> function may be used before calling <code>mmds.plot</code> to color elements by user-provided groups. 
</p>


<h3>Value</h3>

<p>Produces a summary plot of the MDS analysis on the same active graphical device.
</p>


<h3>Note</h3>

<p>The scatter plots can display supplementary objects if their coordinates are present in <code>x</code> input.
</p>


<h3>Author(s)</h3>

<p>Julien Pele and Jean-Michel Becu
</p>


<h3>See Also</h3>

<p><code>plot.pca</code> function in <code>bio3d</code> package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># summary plot of the MDS analysis of human GPCRs with projection of GPCRs
# from D. melanogaster as supplementary elements:
data(gpcr)
mmds.plot(gpcr$mmds$sapiens.active,gpcr$mmds$melanogaster.project)
</code></pre>

<hr>
<h2 id='mmds.project'>
Metric multidimensional Scaling Projection
</h2><span id='topic+mmds.project'></span>

<h3>Description</h3>

<p>Performs metric MultiDimensional Scaling (MDS) analysis of active elements and projects supplementary elements onto the active space defined by active elements.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mmds.project(mmds, sup, pc = 3, group.file = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mmds.project_+3A_mmds">mmds</code></td>
<td>

<p>an object of class 'mmds', obtained from <code><a href="#topic+mmds">mmds</a></code>, providing the active elements.
</p>
</td></tr>
<tr><td><code id="mmds.project_+3A_sup">sup</code></td>
<td>

<p>a numeric matrix of distances between supplementary and active elements.
</p>
</td></tr>
<tr><td><code id="mmds.project_+3A_pc">pc</code></td>
<td>

<p>a numeric value indicating the number of principal components to be saved. Default is 3.
</p>
</td></tr>
<tr><td><code id="mmds.project_+3A_group.file">group.file</code></td>
<td>

<p>a string of characters to indicate the file name assigning groups and colors to each supplementary element of the sup matrix.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Metric multidimensional scaling is a statistical analysis technique aimed at analyzing a matrix of distances between 'active' elements. MDS maps the elements onto a low dimensional space (usually 2D or 3D).  In this space, elements are represented by points whose respective distances best approximate the initial distance. 
In addition, after the metric MDS analysis of active elements, the <code>mmds.project</code> function allows projecting supplementary elements onto the active space in the context of the R environment. The active space is defined only by the MDS analysis of the active elements. The position of supplementary elements onto the active space depends only on their distances to active elements. 
</p>
<p><code>sup</code> must have some characteristics:
</p>

<ul>
<li> <p><code>sup</code> represents the matrix of pairwise distances between supplementary (rows) and active (columns) elements and does not have to be symmetric. The number of supplementary elements may be lower or higher than the number of active objects. The names of supplementary elements must be placed in the left column of <code>sup</code>.
</p>
</li></ul>

<p>The method for the computation of metric MDS projection of supplementary data is described by Abdi (2007). Briefly, if <em>N</em> is the number of active sequences and <em>D</em> is the <em>N</em> by <em>N</em> matrix of the squared distances computed from the active matrix, the <code>mmds</code> function performs the following steps:
</p>

<dl>
<dt>(1)</dt><dd>
<p>Transforms <b>D</b> into a cross-product matrix <b>S</b>:
</p>
<p style="text-align: center;"><code class="reqn">\bold{S} = -0.5 [\bold{I} - \bold{m}^{T}\bold{1}] \times \bold{D} \times [\bold{I} - \bold{m}^{T}\bold{1}]^{T}</code>
</p>

<p>where <b>I</b> is the <em>N</em> by <em>N</em> identity matrix, <b>m</b> is the <em>N</em> by <em>1</em> matrix mass, where each mass equal to <em><code class="reqn">\frac{1}{N}</code></em> and <b>1</b> is an <em>N</em> by <em>N</em> matrix of ones.
</p>
</dd>
<dt>(2)</dt><dd>
<p>Transforms <b><em>S</em></b> into a factor matrix <b><em>F</em></b>:
</p>
<p style="text-align: center;"><code class="reqn">\bold{F} = \bold{M}^{\frac{-1}{2}}\bold{U\Lambda}^{\frac{1}{2}}</code>
</p>

<p>where <b>M</b> is the diag{<b>m</b>}, <b><em>U</em></b> is the eigenvector matrix and <em><code class="reqn">\bold{\Lambda}</code></em> is the diagonal matrix of the eigenvalues, such as <b><em>S</em></b> = <em><code class="reqn">\bold{{U{\Lambda}U}^T}</code></em>, where <em><code class="reqn">\bold{^T}</code></em> denotes the transposition operation.
</p>
</dd>
</dl>

<p>The eigenvectors of <b><em>S</em></b>, also called principal components (whose number is smaller or
equal to <b><em>N</em></b>), form the active space. <b><em>F</em></b> gives the coordinates of the
active elements in this space.
</p>
<p>The supplementary elements are projected onto the active space as described below.
If <b><code class="reqn">N_{sup}</code></b> is the number of supplementary sequences and if <b><code class="reqn">D_{sup}</code></b> is the <b><code class="reqn">N_{sup}</code></b> by <em>N</em>
matrix of the squared distances, the <code>mmds.project</code> function performs the following steps :
</p>

<dl>
<dt>(3)</dt><dd>
<p>Transforms <code class="reqn">\bold{D_{sup}}</code> into a cross-product matrix <code class="reqn">\bold{S_{sup}}</code>:
</p>
<p style="text-align: center;"><code class="reqn">\bold{S_{sup}} = -0.5[\bold{I} - \bold{m}^{T}\bold{1}] \times [\bold{{D_{sup}}^{T}} - \bold{D}\bold{m}\bold{1_{sup}^{T}}],</code>
</p>

<p>where <code class="reqn">\bold{1_{sup}}</code> is an <code class="reqn">\bold{N_{sup}}</code> by <b>N</b> matrix of ones.
</p>
</dd>
<dt>(4)</dt><dd>
<p>Transforms <code class="reqn">\bold{S_{sup}}</code> into a factor matrix <code class="reqn">\bold{F_{sup}}</code>:
</p>
<p style="text-align: center;"><code class="reqn">\bold{F_{sup}} = \bold{{S_{sup}}^T}\bold{F}\bold{\Lambda}^{-1}.</code>
</p>

<p><code class="reqn">\bold{F_{sup}}</code> gives the coordinates of the supplementary elements in the active space.
</p>
</dd>
</dl>



<h3>Value</h3>

<p>A object of class 'project', which is a named list of three elements:
</p>
<table>
<tr><td><code>coord</code></td>
<td>

<p>a named numeric matrix representing the coordinates of active elements.
</p>
</td></tr>
<tr><td><code>group</code></td>
<td>

<p>a named string matrix representing the differents groups of elements and the associate color (Default is 'NoGroup' and 'black').
</p>
</td></tr>
<tr><td><code>col</code></td>
<td>

<p>a named string matrix representing, foreach named elements, the associate group and color (Default is 'NoGroup' and 'black).
</p>
</td></tr>
</table>


<h3>Note</h3>

<p>If <code>sup</code> do not contain names:<br />
A tag &quot;S&quot; followed by an incremented number names the rows of <code>sup</code>.<br />
The columns of <code>sup</code> are named as the rows of <code>mmds$D</code>. 
</p>


<h3>Author(s)</h3>

<p>Julien Pele and Jean-Michel Becu
</p>


<h3>References</h3>

<p>Abdi H (2007) Metric multidimensional scaling. In N.J. Salkind (Ed.):
<em>Encyclopedia of Measurement and Statistics.</em> Thousand Oaks (CA): Sage. pp. 598-605.
</p>
<p>For further reading on projection of supplementary elements:
</p>
<p>Gower JC (1968) adding a point to vector diagrams in multivariate analysis. <em>Biometrika</em> <b>55</b>:582-585.
</p>
<p>Trosset MW and Pribe CE (2008) The out-of-sample problem for classical multidimensional scaling. <em>Computational statistics &amp; Data analysis</em> <b>52</b>:4635-4642.
</p>
<p>Pele J, Abdi H, Moreau M, Thybert D and Chabbert M (2011) Multidimensional scaling reveals the main evolutionary pathways of class A G-protein-coupled receptors. <em>PloS ONE</em> <b>6</b>:e19094.
</p>


<h3>See Also</h3>

<p><code>cmdscale</code> function from <code>stats</code> package.<br />
<code>dudi.pco</code>, <code>suprow</code> and <code>supcol</code> functions from <code>ade4</code> package.<br />
<code>PCA</code> function from <code>FactoMineR</code> package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># performing metric MDS of human GPCRs with projection of
# GPCRs from D. melanogaster as supplementary elements:
data(gpcr)
active &lt;- gpcr$dif$sapiens.sapiens
sup &lt;- gpcr$dif$melanogaster.sapiens
mmds_active&lt;-mmds(active)
mmds_sup &lt;- mmds.project(mmds_active,sup)
</code></pre>

<hr>
<h2 id='random.msa'>
Random Alignment
</h2><span id='topic+random.msa'></span>

<h3>Description</h3>

<p>Builds a multiple sequence alignment (MSA) of random sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>random.msa(nb.seq = 100, id = "SEQ", nb.pos = 100, gap = FALSE,
aa.strict = FALSE, align = NULL, align.replace = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="random.msa_+3A_nb.seq">nb.seq</code></td>
<td>

<p>a numeric value indicating the number of sequences in the random MSA. Default is 100. 
</p>
</td></tr>
<tr><td><code id="random.msa_+3A_id">id</code></td>
<td>

<p>a string of characters used to tag each sequence name. Default is &quot;SEQ&quot;. An
incremented number is attached to this tag to name each sequence.
</p>
</td></tr>
<tr><td><code id="random.msa_+3A_nb.pos">nb.pos</code></td>
<td>

<p>a numeric value indicating the length of each sequence in the random MSA. Default is 100.
</p>
</td></tr>
<tr><td><code id="random.msa_+3A_gap">gap</code></td>
<td>

<p>a logical value indicating whether the gap character should be considered as a supplementary symbol
(TRUE) or not (FALSE). Default is FALSE. 
</p>
</td></tr>
<tr><td><code id="random.msa_+3A_aa.strict">aa.strict</code></td>
<td>

<p>a logical value indicating whether only strict amino acids should be taken into account (TRUE)
or not (FALSE). Default is FALSE.
</p>
</td></tr>
<tr><td><code id="random.msa_+3A_align">align</code></td>
<td>

<p>an object of class 'align', obtained from <code><a href="#topic+import.fasta">import.fasta</a></code> or <code><a href="#topic+import.msf">import.msf</a></code> function.
If this parameter is not NULL, the composition of the output sequences is based on the composition of the input sequences. Default is NULL.
</p>
</td></tr>
<tr><td><code id="random.msa_+3A_align.replace">align.replace</code></td>
<td>

<p>a logical value indicating random drawing with replacement (TRUE) or without replacement (FALSE) of characters present in <code>align</code>. Default is FALSE.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>random.msa</code> may be used to compare a reference MSA to a random MSA. 
The random MSA must have the same characteristics as the reference MSA (same number of sequences of same length).
</p>
<p>A <code><a href="#topic+mmds">mmds</a></code> procedure can be applied to the random MSA to assess the amount of variance due to
random mutations in the reference MSA. 
</p>


<h3>Value</h3>

<p>A named list whose objects correspond to random sequences.
</p>


<h3>Note</h3>

<p>The <code>subset</code> function is used for random selection of the amino acids. If a truly random
procedure is needed, see <code>random</code> package.
</p>


<h3>Author(s)</h3>

<p>Julien Pele
</p>


<h3>References</h3>

<p>For an application of random MSA see : 
</p>
<p>Pele J, Abdi H, Moreau M, Thybert D and Chabbert M (2011) Multidimensional scaling reveals the main evolutionary pathways of class A G-protein-coupled receptors. <em>PLoS ONE</em> <b>6</b>: e19094. doi:10.1371.
</p>


<h3>See Also</h3>

<p><code>permutation</code> and <code>synsequence</code> functions from <code>seqinr</code> package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># generating a random sequence alignment with the same characterics
# as human GPCRs:
aln &lt;- import.fasta(system.file("msa/human_gpcr.fa", package = "bios2mds"))
nb.seq &lt;- length(aln)
nb.pos &lt;- length(aln[[1]])
aln.random &lt;- random.msa(nb.seq = nb.seq, nb.pos = nb.pos)
</code></pre>

<hr>
<h2 id='scree.plot'>
Plots the eingenvalues of an MDS analysis
</h2><span id='topic+scree.plot'></span>

<h3>Description</h3>

<p>Displays a bar plot of the eigenvalues obtained by MDS
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scree.plot(x, lab = FALSE, title = NULL, xlim = NULL,
    ylim = NULL, new.plot = TRUE, pdf.file = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scree.plot_+3A_x">x</code></td>
<td>

<p>a numeric vector representing the raw or the relative eigenvalues of each component.
</p>
</td></tr>
<tr><td><code id="scree.plot_+3A_lab">lab</code></td>
<td>

<p>a boolean indicating whether bar labels should be displayed (TRUE) or not (FALSE). Default is FALSE.
</p>
</td></tr>
<tr><td><code id="scree.plot_+3A_title">title</code></td>
<td>

<p>a character string representing the title of the plot. Default is &quot;Scree plot&quot;.  
</p>
</td></tr>
<tr><td><code id="scree.plot_+3A_xlim">xlim</code></td>
<td>

<p>a numeric vector representing the range of the x values. Default is full range.  
</p>
</td></tr>
<tr><td><code id="scree.plot_+3A_ylim">ylim</code></td>
<td>

<p>a numeric vector representing the range of the y values. Default is full range.  
</p>
</td></tr>
<tr><td><code id="scree.plot_+3A_new.plot">new.plot</code></td>
<td>

<p>a boolean indicating whether a new graphical device should be created (TRUE) or not (FALSE). Default is TRUE.
</p>
</td></tr>
<tr><td><code id="scree.plot_+3A_pdf.file">pdf.file</code></td>
<td>

<p>a string indicating the name and directory of the pdf graph outfile. Default is NULL. If this parameter is not NULL, the parameter <code>new.plot</code> is inactivated.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A scree plot is a method for determining the optimal number of components useful to describe the data in the context of metric MultiDimensional Scaling (MDS). The scree plot is an histogram showing the eigenvalues of each component. The relative eigenvalues express the ratio of each eigenvalue to the sum of the eigenvalues. The relative eigenvalue of a component gives the proportion of the data variance explained by this component.
</p>
<p>The aim is to evaluate the number of components required to capture most information contained in the data. In a scree plot, the relative eigenvalues decrease when the component number increases. The 'elbow' of the plot determines the optimal number of components to describe the data (usually the components before the 'elbow').
</p>


<h3>Value</h3>

<p>Produces a bar plot on the active graphical device.
</p>


<h3>Note</h3>

<p>The scree plot is not an exclusive method to determine the optimal number of components. 
A shepard plot, which is a scatterplot of the scaled MDS distances against the original distance data, can be
another solution. See <code>shepard</code> function from <code>MASS</code> package.
</p>


<h3>Author(s)</h3>

<p>Julien Pele
</p>


<h3>See Also</h3>

<p><code>plot.pca.scree</code> function from <code>bio3d</code> package.<br />
<code>goodness.metaMDS</code> function from <code>vegan</code> package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># displaying the scree plot of the MDS analysis of human GPCRs
data(gpcr)
active &lt;- gpcr$dif$sapiens.sapiens
mmds1 &lt;- mmds(active, pc = 5)
scree.plot(mmds1$eigen.perc, lab = TRUE, title = "Scree plot of metric MDS")
</code></pre>

<hr>
<h2 id='sil.score'>
Silhouette score
</h2><span id='topic+sil.score'></span>

<h3>Description</h3>

<p>Computes silhouette scores for multiple runs of K-means clustering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sil.score(mat, nb.clus = c(2:13), nb.run = 100, iter.max = 1000,
method = "euclidean")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sil.score_+3A_mat">mat</code></td>
<td>

<p>a numeric matrix representing the coordinates of the elements.
</p>
</td></tr>
<tr><td><code id="sil.score_+3A_nb.clus">nb.clus</code></td>
<td>

<p>a numeric vector indicating the range of the numbers of clusters. Default is c(2:13). 
</p>
</td></tr>
<tr><td><code id="sil.score_+3A_nb.run">nb.run</code></td>
<td>

<p>a numeric value indicating the number of runs. Default is 100. 
</p>
</td></tr>
<tr><td><code id="sil.score_+3A_iter.max">iter.max</code></td>
<td>

<p>a numeric value indicating the maximum number of iterations for K-means clustering. Default is 1000. 
</p>
</td></tr>
<tr><td><code id="sil.score_+3A_method">method</code></td>
<td>

<p>a string of characters to determine the distance measure. This should be one of &quot;euclidean&quot; , &quot;maximum&quot;,
&quot;manhattan&quot;, &quot;canberra&quot; or &quot;binary&quot;. Default is &quot;euclidean&quot;. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Silhouettes are a general graphical aid for interpretation and validation of cluster analysis.
This technique is available through the <code>silhouette</code> function (<code>cluster</code> package). In order to
calculate silhouettes, two types of data are needed:
</p>

<ul>
<li><p> the collection of all distances between objects. These distances are obtained from
application of <code>dist</code> function on the coordinates of the elements in <code>mat</code> with argument <code>method</code>.
</p>
</li>
<li><p> the partition obtained by the application of a clustering technique. In <code>sil.score</code>
context, the partition is obtained from the <code>Kmeans</code> function (<code>amap</code> package) with argument
<code>method</code> which indicates the cluster to which each element is assigned.
</p>
</li></ul>

<p>For each element, a silhouette value is calculated and evaluates the degree of confidence in the assignment of the element:
</p>

<ul>
<li><p> well-clustered elements have a score near 1,
</p>
</li>
<li><p> poorly-clustered elements have a score near -1.
</p>
</li></ul>

<p>Thus, silhouettes indicates the objects that are well or poorly clustered. To summarize the results, for each cluster, the silhouettes values can be displayed as an <b>average silhouette width</b>, which is the mean of silhouettes for all the elements assigned to this cluster. Finally, the <b>overall average silhouette</b> width is the mean of average silhouette widths of the different clusters.
</p>
<p>Silhouette values offer the advantage that they depend only on the partition of the elements. As a consequence, silhouettes can be used to compare the output of the same clustering algorithm applied
to the same data but for different numbers of clusters. A range of numbers of clusters can be tested, with the <code>nb.clus</code> argument. The optimal number of clusters is reached for the maximum of the overall
silhouette width. This means that the clustering algorithm reaches a strong clustering structure.
However, for a given number of clusters, the cluster assignment obtained by different K-means runs can be different because the K-means procedure assigns random initial centroids for each run. It may be necessary to run the K-means procedure several times, with the nb.run argument, to evaluate the uncertainty of the results. In that case, for each number of clusters, the mean of the overall average silhouettes for <code>nb.run</code> K-means runs is calculated. The maximum of this core gives the optimal number of clusters.
</p>


<h3>Value</h3>

<p>A named numeric vector representing the silhouette scores for each number of clusters.
</p>


<h3>Note</h3>

<p><code>sil.score</code> requires <code>Kmeans</code> and <code>silhouette</code> functions from <code>amap</code> and
<code>cluster</code> packages, respectively.
</p>


<h3>Author(s)</h3>

<p>Julien Pele
</p>


<h3>References</h3>

<p>Rousseeuw PJ (1987) Silhouettes: A Graphical Aid to the Interpretation and Validation of
Cluster Analysis. <em>Journal of Computational and Applied Mathematics</em>, <b>20</b>:53-65.
</p>
<p>Lovmar L, Ahlford A, Jonsson M and Syvanen AC (2005) Silhouette scores for assessment
of SNP genotype clusters. <em>BMC Genomics</em>, <b>6</b>:35.
</p>
<p>Guy B, Vasyl P, Susmita D and Somnath D (2008) clValid: An R Package for Cluster Validation.
<em>Journal of Statistical Software</em>, <b>25</b>.
</p>


<h3>See Also</h3>

<p><code>connectivity</code> and <code>dunn</code> functions from <code>clValid</code> package.<br />
<code>silhouette</code> function from <code>cluster</code> package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># calculating silhouette scores for K-means clustering of human GPCRs:
data(gpcr)
active &lt;- gpcr$dif$sapiens.sapiens
mds &lt;- mmds(active)
sil.score1 &lt;- sil.score(mds$coord, nb.clus = c(2:10),
 nb.run = 100, iter.max = 100)
barplot(sil.score1)
</code></pre>

<hr>
<h2 id='sub.mat'>
Amino acid substitution matrices
</h2><span id='topic+sub.mat'></span>

<h3>Description</h3>

<p>Contains eight amino acid substitution matrices, imported from version 9.1 of the aaindex2 database
and PAM matrix calculator of Wageningen Bioinformatics Webportal.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sub.mat)</code></pre>


<h3>Format</h3>

<p>A named list with eight elements corresponding to a 20 by 20 named matrix. Rows and columns names
correspond to the twenty strict amino acids.
</p>


<h3>Details</h3>


<dl>
<dt>PAM40</dt><dd>
<p>matrix was produced by &quot;pam&quot; Version 1.0.7 
</p>
</dd>
<dt>PAM120</dt><dd>
<p>matrix was produced by &quot;pam&quot; Version 1.0.7
</p>
</dd>
<dt>PAM160</dt><dd>
<p>matrix was produced by &quot;pam&quot; Version 1.0.7 
</p>
</dd>
<dt>PAM250</dt><dd>
<p>log odds matrix for 250 PAMs (Dayhoff et al., 1978)
</p>
</dd>
<dt>BLOSUM30</dt><dd>
<p>substitution matrix (Henikoff-Henikoff, 1992)
</p>
</dd>
<dt>BLOSUM45</dt><dd>
<p>substitution matrix (Henikoff-Henikoff, 1992)
</p>
</dd>
<dt>BLOSUM62</dt><dd>
<p>substitution matrix (Henikoff-Henikoff, 1992)
</p>
</dd>
<dt>BLOSUM80</dt><dd>
<p>substitution matrix (Henikoff-Henikoff, 1992)
</p>
</dd>
<dt>GONNET</dt><dd>
<p>substitution matrix (GONNET et al., 1992)
</p>
</dd>
<dt>JTT</dt><dd>
<p>substitution matrix (Jones et al., 1992)
</p>
</dd>
<dt>JTT_TM</dt><dd>
<p>transmembrane protein exchange matrix (Jones et al., 1994)
</p>
</dd>
<dt>PHAT</dt><dd>
<p>substitution matrix built from hydrophobic and transmembrane regions of the Blocks database (Ng et al., 2000)
</p>
</dd>
</dl>



<h3>Source</h3>

<p>The matrices were downloaded from the AAindex database at
<a href="http://www.genome.jp/aaindex">http://www.genome.jp/aaindex</a> or were calculated on the PAM server at <a href="http://www.bioinformatics.nl/tools/pam.html">http://www.bioinformatics.nl/tools/pam.html</a>.
</p>


<h3>References</h3>

<p>Kawashima S and Kanehisa M (2000) AAindex: amino acid index 
database. <em>Nucleic Acids Res</em> <b>28</b>:374.
</p>
<p>Dayhoff MO, Schwartz R and Orcutt BC (1978) A model of Evolutionary Change in Proteins. 
Atlas of protein sequence and structure (volume 5, supplement 3 ed.). Nat. Biomed. Res. Found.. pp. 345-358. 
</p>
<p>Henikoff S and Henikoff JG (1992) Amino acid substitution matrices from protein blocks.
<em>Proc Natl Acad Sci U S A</em> <b>89</b>:10915-9. 
</p>
<p>Jones DT, Taylor WR and Thornton JM (1992) The rapid generation of mutation data matrices
from protein sequences. <em>Comput Appl. Biosci</em> <b>8</b>:275-282. 
</p>
<p>Jones DT, Taylor WR and Thornton JM (1994) A mutation data matrix for transmembrane
proteins. <em>FEBS Lett</em> <b>339</b>:269-75. 
</p>
<p>Ng PC, Henikoff JG and Henikoff S (2000) PHAT: a transmembrane-specific substitution
matrix. Predicted hydrophobic and transmembrane. <em>Bioinformatics</em> <b>16</b>:760-6.
</p>
<p>Gonnet GH, Cohen MA and Benner SA (1992) Exhaustive matching of the entire protein sequence database.
<em>Science</em> <b>256</b>:1443-1445.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># loading
data(sub.mat)

# displaying PAM40:
sub.mat$PAM40
</code></pre>

<hr>
<h2 id='write.mmds.pdb'>
Writes PDB file</h2><span id='topic+write.mmds.pdb'></span>

<h3>Description</h3>

<p>Writes MDS coordinates in the Protein Data Bank format for visualization with a molecular graphics viewer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.mmds.pdb(x,project = NULL, axis = c(1, 2, 3), file.pdb = "R.pdb", file.pml=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.mmds.pdb_+3A_x">x</code></td>
<td>

<p>an object of class 'mmds', obtained from <code><a href="#topic+mmds">mmds</a></code> function.
</p>
</td></tr>
<tr><td><code id="write.mmds.pdb_+3A_project">project</code></td>
<td>

<p>an object of class 'project', obtained from <code><a href="#topic+mmds.project">mmds.project</a></code> function, or a list of object of class 'project'.
</p>
</td></tr>
<tr><td><code id="write.mmds.pdb_+3A_axis">axis</code></td>
<td>

<p>a numeric vector of length three the principal components to be displayed. Default is c(1, 2, 3). 
</p>
</td></tr>
<tr><td><code id="write.mmds.pdb_+3A_file.pdb">file.pdb</code></td>
<td>

<p>a string of characters indicating the output PDB file name. Default is &quot;R.pdb&quot;.
</p>
</td></tr>
<tr><td><code id="write.mmds.pdb_+3A_file.pml">file.pml</code></td>
<td>

<p>a string of characters indicating the output pml file name for visualization with Pymol. If this parameter is not NULL, the pml file will be written. Default is NULL.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The elements can be visualized in three dimensions (3D) with a molecular viewer as Pymol or Rasmol. 
If <code>x</code> contains active and supplementary elements, the active and supplementary elements are numbered from 1 and from 5001, respectively. If group is not NULL, the assignment of an element to a group is indicated by the chain name from <em>A</em> for the first group to <em>Z</em> when the maximum number of groups, 26, is reached. 
</p>
<p>The pml file allows a fancy visualization of the PDB file with the Pymol molecular viewer. The user must first open the PDB file with Pymol, then run the pml file. The active and inactive elements will be displayed as spheres and crosses, respectively, with coloring based on the user-provided colors with the <code>col.group</code> function.
</p>


<h3>Value</h3>

<p>Produces a PDB file from the MDS coordinates, with the elements numbered in the order of the MSA file and the groups corresponding to the chain numbers. Optionnaly, produces a pml file to add color and group selection in pymol with the pdb file.
</p>


<h3>Author(s)</h3>

<p>Julien Pele and Jean-Michel Becu
</p>


<h3>References</h3>

<p><a href="http://www.wwpdb.org/docs.html">http://www.wwpdb.org/docs.html</a>
</p>


<h3>See Also</h3>

<p><code>write.pdb</code> function from <code>bio3d</code> package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># writing the first three MDS coordinates of human GPCRs in a PDB file 
wd &lt;- tempdir()
#wd &lt;- getwd()
file1 &lt;- file.path(wd,"sapiens.pdb")
file2 &lt;- file.path(wd,"sapiens.pml")
data(gpcr)
write.mmds.pdb(gpcr$mmds$sapiens.active,file.pdb=file1,file.pml=file2)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
