<!DOCTYPE html><html><head><title>Help for package biospear</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {biospear}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BMsel'><p>Biomarker selection in a Cox regression model</p></a></li>
<li><a href='#Breast'><p>Early breast cancer data</p></a></li>
<li><a href='#expSurv'><p>Computation of expected survival based on a prediction model</p></a></li>
<li><a href='#predRes'><p>Evaluation of the prediction accuracy of a prediction model</p></a></li>
<li><a href='#selRes'><p>Evaluation of the selection accuracy of a prediction model</p></a></li>
<li><a href='#simdata'><p>Generation of data sets with survival outcome</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Biomarker Selection in Penalized Regression Models</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Author:</td>
<td>Nils Ternes [aut],
  Federico Rotolo [aut], Stefan Michiels [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Stefan Michiels &lt;stefan.michiels@gustaveroussy.fr&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), pkgconfig</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides some tools for developing and validating prediction models, estimate expected survival of patients and visualize them graphically. Most of the implemented methods are based on penalized regressions such as: the lasso (Tibshirani R (1996)), the elastic net (Zou H et al. (2005) &lt;<a href="https://doi.org/10.1111%2Fj.1467-9868.2005.00503.x">doi:10.1111/j.1467-9868.2005.00503.x</a>&gt;), the adaptive lasso (Zou H (2006) &lt;<a href="https://doi.org/10.1198%2F016214506000000735">doi:10.1198/016214506000000735</a>&gt;), the stability selection (Meinshausen N et al. (2010) &lt;<a href="https://doi.org/10.1111%2Fj.1467-9868.2010.00740.x">doi:10.1111/j.1467-9868.2010.00740.x</a>&gt;), some extensions of the lasso (Ternes et al. (2016) &lt;<a href="https://doi.org/10.1002%2Fsim.6927">doi:10.1002/sim.6927</a>&gt;), some methods for the interaction setting (Ternes N et al. (2016) &lt;<a href="https://doi.org/10.1002%2Fbimj.201500234">doi:10.1002/bimj.201500234</a>&gt;), or others. A function generating simulated survival data set is also provided.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>cobs, corpcor, devtools, glmnet, graphics, grplasso, MASS,
Matrix, mboost, parallel, plsRcox, pROC, PRROC, RCurl, stats,
survAUC, survival</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-12-04 09:16:42 UTC; s_michiels</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-12-04 10:20:19 UTC</td>
</tr>
</table>
<hr>
<h2 id='BMsel'>Biomarker selection in a Cox regression model</h2><span id='topic+BMsel'></span><span id='topic+summary.resBMsel'></span>

<h3>Description</h3>

<p>This function enables to fit a Cox regression model for a prognostic
or a biomarker-by-treatment interaction setting subject to a selection procedure to perform variable selection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BMsel(data, x, y, z, tt, inter, std.x = TRUE, std.i = FALSE, std.tt = TRUE, 
  method = c('alassoL', 'alassoR', 'alassoU', 'enet', 'gboost', 
    'glasso', 'lasso', 'lasso-1se', 'lasso-AIC', 'lasso-BIC', 
    'lasso-HQIC', 'lasso-pct', 'lasso-pcvl','lasso-RIC', 'modCov',
    'PCAlasso', 'PLSlasso', 'ridge', 'ridgelasso', 'stabSel', 'uniFDR'), 
  folds = 5, uni.fdr = 0.05, uni.test = 1, ss.rando = F, ss.nsub = 100,
  ss.fsub = 0.5, ss.fwer = 1, ss.thr = 0.6, dfmax = ncol(data) + 1, 
  pct.rep = 1, pct.qtl = 0.95, showWarn = TRUE, trace = TRUE)

## S3 method for class 'resBMsel'
summary(object, show = TRUE, keep = c('tt', 'z', 'x', 'xt'), 
  add.ridge = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BMsel_+3A_data">data</code></td>
<td>
<p>input <code><a href="base.html#topic+data.frame">data.frame</a></code>. Each row is an observation.</p>
</td></tr>
<tr><td><code id="BMsel_+3A_x">x</code></td>
<td>
<p>colnames or position of the biomarkers in <code>data</code>.</p>
</td></tr>
<tr><td><code id="BMsel_+3A_y">y</code></td>
<td>
<p>colnames or position of the survival outcome in <code>data</code>. The
first column must be the time and the second must be the indicator (0/1).</p>
</td></tr>
<tr><td><code id="BMsel_+3A_z">z</code></td>
<td>
<p>colnames or position of the clinical covariates in <code>data</code>,
if any.</p>
</td></tr>
<tr><td><code id="BMsel_+3A_tt">tt</code></td>
<td>
<p>colname or position of the treatment in <code>data</code>, if any.</p>
</td></tr>
<tr><td><code id="BMsel_+3A_inter">inter</code></td>
<td>
<p>logical parameter indicating if biomarker-by-treatment
interactions should be computed.</p>
</td></tr>
<tr><td><code id="BMsel_+3A_std.x">std.x</code></td>
<td>
<p>logical parameter indicating if the biomarkers should be
standardized (i.e. substracting by the mean and dividing by the standard
deviation of each biomarker).</p>
</td></tr>
<tr><td><code id="BMsel_+3A_std.i">std.i</code></td>
<td>
<p>logical parameter indicating if the biomarker-by-treatment
interactions should be standardized (i.e. substracting by the mean and
dividing by the standard deviation of each interaction).</p>
</td></tr>
<tr><td><code id="BMsel_+3A_std.tt">std.tt</code></td>
<td>
<p>logical parameter indicating if the treatment should be recoded as +/-0.5.</p>
</td></tr>
<tr><td><code id="BMsel_+3A_method">method</code></td>
<td>
<p>methods computed to perform variable selection and to estimate
the regression coefficients. See the Details section to understand all the implemented methods.</p>
</td></tr>
<tr><td><code id="BMsel_+3A_folds">folds</code></td>
<td>
<p>number of folds. <code>folds</code> must be either a value between 3 and the sample size (leave-one-out CV, but not
recommended for large datasets), or a vector (same length as the sample size) indicating the fold assignment group of each observation.</p>
</td></tr>
<tr><td><code id="BMsel_+3A_uni.fdr">uni.fdr</code>, <code id="BMsel_+3A_uni.test">uni.test</code></td>
<td>
<p>specific parameters for the univariate procedure. <code>uni.fdr</code>: threshold false discovery rate (FDR) to control for multiple testing (Benjamini and Hochberg, 1995), <code>uni.test</code>: model comparison approach. 1: p-value of the biomarker effect (i.e. main effect for the prognostic setting, or main effect + interaction for the interaction setting), 2: p-value of the interaction (only available for the interaction setting).</p>
</td></tr>
<tr><td><code id="BMsel_+3A_ss.fsub">ss.fsub</code>, <code id="BMsel_+3A_ss.fwer">ss.fwer</code>, <code id="BMsel_+3A_ss.nsub">ss.nsub</code>, <code id="BMsel_+3A_ss.rando">ss.rando</code>, <code id="BMsel_+3A_ss.thr">ss.thr</code></td>
<td>
<p>specific parameters for the stability selection. <code>ss.fsub</code>: fraction of samples to use in the sampling process, <code>ss.fwer</code>: parameter to control for the family-wise error rate (FWER, i.e. number of noise variables), <code>ss.nsub</code>: number of subsampling, <code>ss.rando</code>: logical parameter indicating if random weights should be added in the lasso penalty, <code>ss.thr</code>: threshold of the stability probability for filtering variable.</p>
</td></tr>
<tr><td><code id="BMsel_+3A_dfmax">dfmax</code></td>
<td>
<p>limit the maximum number of variables in the model. Useful for very large number of covariates to limit the time computation.</p>
</td></tr>
<tr><td><code id="BMsel_+3A_pct.rep">pct.rep</code>, <code id="BMsel_+3A_pct.qtl">pct.qtl</code></td>
<td>
<p>specific parameters for the percentile lasso.
<code>pct.rep</code>: number of replicates, <code>pct.qtl</code>: percentile used to estimate the lambda among its empirical distribution.</p>
</td></tr>
<tr><td><code id="BMsel_+3A_showwarn">showWarn</code></td>
<td>
<p>logical parameter indicating if warnings should be printed.</p>
</td></tr>
<tr><td><code id="BMsel_+3A_trace">trace</code></td>
<td>
<p>logical parameter indicating if messages should be printed.</p>
</td></tr>
<tr><td><code id="BMsel_+3A_object">object</code></td>
<td>
<p>object of class '<code>resBMsel</code>' returned by <code>BMsel</code>.</p>
</td></tr>
<tr><td><code id="BMsel_+3A_show">show</code></td>
<td>
<p>parameter for the <code>summary()</code> indicating if the result should be printed.</p>
</td></tr>
<tr><td><code id="BMsel_+3A_keep">keep</code></td>
<td>
<p>parameter for the <code>summary()</code> indicating the type of covariates that should be kept for the summary (<code>tt</code>: treatment covariate, <code>z</code>: clinical covariates, <code>x</code>: biomarker main effects and <code>xt</code>: biomarker-by-treatment interactions).</p>
</td></tr>
<tr><td><code id="BMsel_+3A_add.ridge">add.ridge</code></td>
<td>
<p>parameter for the <code>summary()</code> indicating if the ridge penalty should be kept for the summary as no selection is performed.</p>
</td></tr>
<tr><td><code id="BMsel_+3A_...">...</code></td>
<td>
<p>other paramaters for <a href="graphics.html#topic+plot">plot</a> or <a href="base.html#topic+summary">summary</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The objects <code>x</code>, <code>y</code>, <code>z</code> (if any) and <code>tt</code> (if any) are mandatory for non-simulated data sets.<br />
The <code>method</code> parameter specifies the approaches for model selection. Most of these selection methods are based on the lasso penalty (Tibshirani, 1996). The tuning parameter is usually chosen though the cross-validated log-likelihood criterion (cvl), except for the empirical extensions of the <code>lasso</code> in which additional penalties to the cvl (given with a suffix, e.g. <code>lasso-pcvl</code>) are used to estimate the tuning parameter. Other methods based on the lasso are also implemented such as the adaptive lasso (<code>alassoL</code>, <code>alassoR</code> and <code>alassoU</code> for which the last letter indicates the procedure used to estimate the preliminary weights: &quot;<code>L</code>&quot; for lasso, &quot;<code>R</code>&quot; for ridge and &quot;<code>U</code>&quot; for univariate), the elastic-net (<code>enet</code>) or the stability selection (<code>stabSel</code>). For the interaction setting, specific methods were implemented: to reduce/control the main effects matrix (i.e. ridge (<code>ridgelasso</code>) or dimension reduction (<code>PCAlasso</code> or <code>PLSlasso</code>)), to select or discard main effects and interactions simultaneously (i.e. group-lasso (<code>glasso</code>)), or to include only the interaction part in the model (i.e. <code>modCov</code>). Some selection methods not based on penalized regression are also proposed: univariate selection (<code>uniFDR</code>), gradient boosting (<code>gboost</code>). The <code>ridge</code> penalty without selection can also be applied. 
</p>
<p>For all methods but the <code>uniFDR</code>, tuning parameters are chosen by maximizing the cross-validated log-likelihood (max-cvl). For the elastic-net, the &quot;alpha&quot; parameter (trade-off between ridge and lasso) is investigated among a predefined grid of values (as suggested by the authors, Zou et al. 2005) and the &quot;lambda&quot; is estimated by maximizing the above-mentioned cvl criterion for each of the &quot;alpha&quot; parameter. The combination (alpha; lambda) that maximizes the cvl is finally retained. For the gradient boosting, the number of steps is also estimated by the max-cvl. For the univariate selection, the tuning parameter is the FDR threshold defined by the user to control for multiple testing (using the parameter <code>uni.fdr</code>).
</p>
<p>We have included the possibility to adjust for clinical covariates (<code>z</code>) for all methods. For penalized regressions, these covariates are considered as unpenalized. For the gradient boosting, a model with clinical covariates is preliminary implemented and regression coefficients are fixed as offset in the boosting approach. For the univariate selection, clinical covariates are forced as adjustment variables in the model and the FDR is calculated on the Wald p-values of the coefficient associated with the biomarker in such models.
</p>


<h3>Value</h3>

<p>An object of class '<code>resBMsel</code>' containing the list of the selected biomarkers and their estimated regression coefficients for the chosen methods.
</p>


<h3>Author(s)</h3>

<p>Nils Ternes, Federico Rotolo, and Stefan Michiels<br />
Maintainer: Nils Ternes <a href="mailto:nils.ternes@yahoo.com">nils.ternes@yahoo.com</a>
</p>


<h3>References</h3>

<p>Ternes N, Rotolo F and Michiels S.
Empirical extensions of the lasso penalty to reduce
the false discovery rate in high-dimensional Cox regression models.
<em>Statistics in Medicine</em> 2016;35(15):2561-2573.
doi:<a href="http://onlinelibrary.wiley.com/doi/10.1002/sim.6927/abstract">10.1002/sim.6927</a><br />
Ternes N, Rotolo F, Heinze G and Michiels S.
Identification of biomarker-by-treatment interactions in randomized
clinical trials with survival outcomes and high-dimensional spaces.
<em>Biometrical journal</em>. In press.
doi:<a href="http://onlinelibrary.wiley.com/doi/10.1002/bimj.201500234/abstract">10.1002/bimj.201500234</a><br />
Tibshirani R.
Regression shrinkage and selection via the lasso.
<em>Journal of the Royal Statistical Society, Ser B</em> 1996;58:267-288.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>########################################
# Simulated data set
########################################

## Low calculation time
  set.seed(654321)
  sdata &lt;- simdata(
    n = 500, p = 20, q.main = 3, q.inter = 0,
    prob.tt = 0.5, alpha.tt = 0,
    beta.main = -0.8,
    b.corr = 0.6, b.corr.by = 4,
    m0 = 5, wei.shape = 1, recr = 4, fu = 2,
    timefactor = 1)

  resBM &lt;- BMsel(
    data = sdata, 
    method = c("lasso", "lasso-pcvl"), 
    inter = FALSE, 
    folds = 5)
  
  summary(resBM)

## Not run: 
## Moderate calculation time
  set.seed(123456)
  sdata &lt;- simdata(
    n = 500, p = 100, q.main = 5, q.inter = 5,
    prob.tt = 0.5, alpha.tt = -0.5,
    beta.main = c(-0.5, -0.2), beta.inter = c(-0.7, -0.4),
    b.corr = 0.6, b.corr.by = 10,
    m0 = 5, wei.shape = 1, recr = 4, fu = 2,
    timefactor = 1,
    active.inter = c("bm003", "bm021", "bm044", "bm049", "bm097"))

  resBM &lt;- BMsel(
    data = sdata, 
    method = c("lasso", "lasso-pcvl"), 
    inter = TRUE, 
    folds = 5)
  
  summary(resBM)
  summary(resBM, keep = "xt")

## End(Not run)  

########################################
# Breast cancer data set
########################################

## Not run: 
  data(Breast)
  dim(Breast)

  set.seed(123456)
  resBM &lt;-  BMsel(
    data = Breast,
    x = 4:ncol(Breast),
    y = 2:1,
    tt = 3,
    inter = FALSE,
    std.x = TRUE,
    folds = 5,
    method = c("lasso", "lasso-pcvl"))

  summary(resBM)

## End(Not run)

########################################
########################################
</code></pre>

<hr>
<h2 id='Breast'>Early breast cancer data</h2><span id='topic+Breast'></span>

<h3>Description</h3>

<p><code>Breast</code> contains clinical and genomic data of 614 early breast cancer patients.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Breast)</code></pre>


<h3>Format</h3>

<p>A dataframe with variables:
</p>

<dl>
<dt>time</dt><dd><p>Distant-relapse free survival time (in years).</p>
</dd>
<dt>status</dt><dd><p>Distant-relapse free survival indicator (0 = censored, 1 = event).</p>
</dd>
<dt>treat</dt><dd><p>Treatment arm (Anthracycline-based adjuvant chemotherapy with (treat = +0.5) or without (treat = -0.5) taxane).</p>
</dd>
<dt>...</dt><dd><p>All other covariates (p=1689) are gene expression values.</p>
</dd>
</dl>



<h3>References</h3>

<p>Desmedt C, Di Leo A, de Azambuja E, Larsimont D, Haibe-Kains B et al.
Multifactorial approach to predicting resistance to anthracyclines
<em>Journal of Clinical Oncology</em> 2011;29:1578-86.
doi:<a href="http://ascopubs.org/doi/full/10.1200/jco.2010.31.2231">10.1200/JCO.2010.31.2231</a><br />
Hatzis C, Pusztai L, Valero V, Booser DJ, Esserman L et al.
A genomic predictor of response and survival following
taxane-anthracycline chemotherapy for invasive breast cancer.
<em>Journal of the American Medical Association</em> 2011;305:1873-1881.
doi:<a href="http://jamanetwork.com/journals/jama/fullarticle/899864">10.1001/jama.2011.593</a><br />
Ternes N, Rotolo F, Heinze G and Michiels S.
Identification of biomarker-by-treatment interactions in randomized
clinical trials with survival outcomes and high-dimensional spaces.
<em>Biometrical journal</em>. In press.
doi:<a href="http://onlinelibrary.wiley.com/doi/10.1002/bimj.201500234/abstract">10.1002/bimj.201500234</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  library(survival)
  data(Breast)
  dim(Breast)
  
  km &lt;- survfit(Surv(time, status) ~ treat, data = Breast)
  km
  plot(km, col = c("black", "red"), lwd = 2, xlim = c(0, 5), xaxt = "n", yaxt = "n")
  legend("bottomleft", legend = c("Control", "Experimental"), col = 1:2, 
    lty = 1, lwd = 2, cex = 1.5)
  axis(1, cex.axis = 1.3)
  axis(2, las = 2, cex.axis = 1.3)
  mtext("Distant-recurrence free survival", side = 2, line = 3.2, cex = 1.5)
  mtext("Time (in years)", side = 1, line = 2.5, cex = 1.5)
</code></pre>

<hr>
<h2 id='expSurv'>Computation of expected survival based on a prediction model</h2><span id='topic+expSurv'></span><span id='topic+predict.resexpSurv'></span><span id='topic+plot.resexpSurv'></span>

<h3>Description</h3>

<p>Based on a prediction model, this function computes expected survival for patients with associated confidence intervals. The returned object can be plotted to obtain a meaningful graphical visualization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expSurv(res, traindata, method, ci.level = .95, boot = FALSE, nboot, smooth = TRUE,
  pct.group = 4, time, trace = TRUE, ncores = 1)

## S3 method for class 'resexpSurv'
predict(object, newdata, ...)

## S3 method for class 'resexpSurv'
plot(x, method, pr.group, print.ci = TRUE, 
  xlim, ylim, xlab, ylab, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="expSurv_+3A_res">res</code></td>
<td>
<p>an object of class '<code>resBMsel</code>' generated by <code><a href="#topic+BMsel">BMsel</a></code>.</p>
</td></tr>
<tr><td><code id="expSurv_+3A_traindata">traindata</code></td>
<td>
<p>the <code>data.frame</code> used to compute the <code>res</code> object (training set).</p>
</td></tr>
<tr><td><code id="expSurv_+3A_method">method</code></td>
<td>
<p>selection method to compute. If missing, all methods contained in <code>res</code> are computed.</p>
</td></tr>
<tr><td><code id="expSurv_+3A_ci.level">ci.level</code></td>
<td>
<p>the nominal level for the two-sided confidence interval (CI) of the survival probability.</p>
</td></tr>
<tr><td><code id="expSurv_+3A_boot">boot</code></td>
<td>
<p>logical value: <code>TRUE</code> = boostraped CI, <code>FALSE</code> = analytical CI.</p>
</td></tr>
<tr><td><code id="expSurv_+3A_nboot">nboot</code></td>
<td>
<p>number of bootstrap replicates (only used when <code>boot</code>=<code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="expSurv_+3A_smooth">smooth</code></td>
<td>
<p>logical value indicating if smoothed B-splines should be computed.</p>
</td></tr>
<tr><td><code id="expSurv_+3A_pct.group">pct.group</code></td>
<td>
<p>number or percentile of the prognostic-risk groups. If a single number is provided, all the groups must be defined according to Cox (1957). If percentiles are provided, the sum must be 1 (e.g. 0.164, 0.336, 0.336, 0.164).</p>
</td></tr>
<tr><td><code id="expSurv_+3A_time">time</code></td>
<td>
<p>single time point to estimate the expected survival probabilities.</p>
</td></tr>
<tr><td><code id="expSurv_+3A_trace">trace</code></td>
<td>
<p>logical parameter indicating if messages should be printed.</p>
</td></tr>
<tr><td><code id="expSurv_+3A_ncores">ncores</code></td>
<td>
<p>number of CPUs used (for the bootstrap CI).</p>
</td></tr>
<tr><td><code id="expSurv_+3A_object">object</code>, <code id="expSurv_+3A_x">x</code></td>
<td>
<p>an object of class '<code>resexpSurv</code>' generated by <code><a href="#topic+expSurv">expSurv</a></code>.</p>
</td></tr>
<tr><td><code id="expSurv_+3A_newdata">newdata</code></td>
<td>
<p><code><a href="base.html#topic+data.frame">data.frame</a></code> containing new patients data. <code>newdata</code> must have the same variables as <code>traindata</code>.</p>
</td></tr>
<tr><td><code id="expSurv_+3A_pr.group">pr.group</code></td>
<td>
<p>parameter for the <code>plot()</code> indicating the number of the prognostic-risk group for which the plot will be printed.</p>
</td></tr>
<tr><td><code id="expSurv_+3A_print.ci">print.ci</code></td>
<td>
<p>logical parameter for the <code>plot()</code> indicating if CI will be printed.</p>
</td></tr>
<tr><td><code id="expSurv_+3A_xlim">xlim</code>, <code id="expSurv_+3A_ylim">ylim</code>, <code id="expSurv_+3A_xlab">xlab</code>, <code id="expSurv_+3A_ylab">ylab</code></td>
<td>
<p>usual parameters for <a href="graphics.html#topic+plot">plot</a>.</p>
</td></tr>
<tr><td><code id="expSurv_+3A_...">...</code></td>
<td>
<p>other paramaters for <a href="stats.html#topic+predict">predict</a> or <a href="graphics.html#topic+plot">plot</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Using an object of class '<code>resBMsel</code>' generated by <code><a href="#topic+BMsel">BMsel</a></code>, <code>expSurv</code> computes expected survival at a given <code>time</code> and constructs confidence intervals thereof either with an analytical (<code>boot</code> = <code>FALSE</code>) or non-parametric bootstrap approach (<code>boot</code> = <code>TRUE</code>). Smoothed B-splines (logical option <code>smooth</code>) and categorization of the prognostic score into risk groups (using the option <code>pct.group</code>) may be used to obtain a meaningful graphical visualization. Predictions for new patients (<code>newdata</code> data frame) can be computed using <code>predict()</code>. Graphical visualization can be obtained using <code>plot()</code>.
</p>


<h3>Value</h3>

<p>A <code>list</code> of length three containing the expected survival (<code>surv</code>) and their corresponding confidence intervals (<code>lower</code> and <code>upper</code>). Each element of the list contains a <code>matrix</code> of dimension number of patients x number of implemented methods. 
</p>


<h3>Author(s)</h3>

<p>Nils Ternes, Federico Rotolo, and Stefan Michiels<br />
Maintainer: Nils Ternes <a href="mailto:nils.ternes@yahoo.com">nils.ternes@yahoo.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>########################################
# Simulated data set
########################################

## Low calculation time
  set.seed(654321)
  sdata &lt;- simdata(
    n = 500, p = 20, q.main = 3, q.inter = 0,
    prob.tt = 0.5, alpha.tt = 0,
    beta.main = -0.8,
    b.corr = 0.6, b.corr.by = 4,
    m0 = 5, wei.shape = 1, recr = 4, fu = 2,
    timefactor = 1)

  resBM &lt;- BMsel(
    data = sdata, 
    method = c("lasso", "lasso-pcvl"), 
    inter = FALSE, 
    folds = 5)
  
  esurv &lt;- expSurv(
    res = resBM,
    traindata = sdata,
    boot = FALSE,
    time = 5,
    trace = TRUE)
  plot(esurv, method = "lasso-pcvl")

## Not run: 
## Moderate calculation time
  set.seed(123456)
  sdata &lt;- simdata(
    n = 500, p = 100, q.main = 5, q.inter = 5,
    prob.tt = 0.5, alpha.tt = -0.5,
    beta.main = c(-0.5, -0.2), beta.inter = c(-0.7, -0.4),
    b.corr = 0.6, b.corr.by = 10,
    m0 = 5, wei.shape = 1, recr = 4, fu = 2,
    timefactor = 1,
    active.inter = c("bm003", "bm021", "bm044", "bm049", "bm097"))

  resBM &lt;- BMsel(
    data = sdata, 
    method = c("lasso", "lasso-pcvl"), 
    inter = TRUE, 
    folds = 5)
  
  esurv &lt;- expSurv(
    res = resBM,
    traindata = sdata,
    boot = TRUE,
    nboot = 100,
    smooth = TRUE,
    pct.group = 4,
    time = 5,
    ncores = 5)
  plot(esurv, method = "lasso", pr.group = 3)

## End(Not run)  

########################################
# Breast cancer data set
########################################

## Not run: 
  data(Breast)
  dim(Breast)

  set.seed(123456)
  resBM &lt;-  BMsel(
    data = Breast,
    x = 4:ncol(Breast),
    y = 2:1,
    tt = 3,
    inter = FALSE,
    std.x = TRUE,
    folds = 5,
    method = c("lasso", "lasso-pcvl"))

  esurv &lt;- expSurv(
    res = resBM,
    traindata = Breast,
    boot = FALSE,
    smooth = TRUE,
    time = 4,
    trace = TRUE
  )
  plot(esurv, method = "lasso")


## End(Not run)

########################################
########################################
</code></pre>

<hr>
<h2 id='predRes'>Evaluation of the prediction accuracy of a prediction model</h2><span id='topic+predRes'></span><span id='topic+plot.predRes'></span>

<h3>Description</h3>

<p>This function computes several criteria to assess the prediction
accuracy of a prediction model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predRes(res, method, traindata, newdata, int.cv, int.cv.nfold = 5, time,
  trace = TRUE, ncores = 1)

## S3 method for class 'predRes'
plot(x, method, crit = c("C", "PE", "dC"),
  xlim, ylim, xlab, ylab, col,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predRes_+3A_res">res</code></td>
<td>
<p>an object of class '<code>resBMsel</code>' generated by <code><a href="#topic+BMsel">BMsel</a></code>.</p>
</td></tr>
<tr><td><code id="predRes_+3A_method">method</code></td>
<td>
<p>methods for which prediction criteria are computed. If missing, all methods contained in <code>res</code> are computed.</p>
</td></tr>
<tr><td><code id="predRes_+3A_traindata">traindata</code></td>
<td>
<p>input <code><a href="base.html#topic+data.frame">data.frame</a></code> used to compute the
<code>res</code> object. This object is mandatory.</p>
</td></tr>
<tr><td><code id="predRes_+3A_newdata">newdata</code></td>
<td>
<p>input <code><a href="base.html#topic+data.frame">data.frame</a></code> not used to compute the
<code>res</code> object. This object is not mandatory (see Details section).</p>
</td></tr>
<tr><td><code id="predRes_+3A_int.cv">int.cv</code></td>
<td>
<p>logical parameter indicating if a double
cross-validation process (2CV) should be performed to mimick an external
validation set.</p>
</td></tr>
<tr><td><code id="predRes_+3A_int.cv.nfold">int.cv.nfold</code></td>
<td>
<p>number of folds for the double cross-validation. Considering a large value for <code>int.cv.nfold</code> should provide extremely large computation time. <code>int.cv.nfold</code> must not be considered when <code>int.cv = FALSE</code>.</p>
</td></tr>
<tr><td><code id="predRes_+3A_time">time</code></td>
<td>
<p>time points to compute the prediction criteria.</p>
</td></tr>
<tr><td><code id="predRes_+3A_trace">trace</code></td>
<td>
<p>logical parameter indicating if messages should be printed.</p>
</td></tr>
<tr><td><code id="predRes_+3A_ncores">ncores</code></td>
<td>
<p>number of CPUs used (for the double cross-validation).</p>
</td></tr>
<tr><td><code id="predRes_+3A_x">x</code></td>
<td>
<p>an object of class '<code>predRes</code>' generated from <code><a href="#topic+predRes">predRes</a></code>.</p>
</td></tr>
<tr><td><code id="predRes_+3A_crit">crit</code></td>
<td>
<p>parameter indicating the criterion for which the results will be printed (<code>C</code>: concordance via Uno's C-statistic, <code>PE</code>: prediction error via integrated Brier score and <code>dC</code>: delta Uno's C-statistic (for the interaction setting only)).</p>
</td></tr>
<tr><td><code id="predRes_+3A_xlim">xlim</code>, <code id="predRes_+3A_ylim">ylim</code>, <code id="predRes_+3A_xlab">xlab</code>, <code id="predRes_+3A_ylab">ylab</code>, <code id="predRes_+3A_col">col</code></td>
<td>
<p>usual parameters for <a href="graphics.html#topic+plot">plot</a>.</p>
</td></tr>
<tr><td><code id="predRes_+3A_...">...</code></td>
<td>
<p>other paramaters for <a href="graphics.html#topic+plot">plot</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To evaluate the accuracy of the selected models, three predictive accuracy measures are implemented: <br />
- the integrated Brier score (<code>PE</code>) to measure the overall prediction error of the prediction model. The time-dependent Brier score is a quadratic score based on the predicted time-dependent survival probability.<br />
- the Uno's C-statistic (<code>C</code>) to evaluate the discrimination of the prediction model. It's one of the least biased concordance statistic estimator in the presence of censoring (Uno et al., 2011).<br />
- the absolute difference of the treatment-specific Uno's C-statistics (<code>dC</code>) to evaluate the interaction strength of the prediction model (Ternes et al., 2016).<br />
For simulated datasets, the predictive accuracy metrics are also computed for the &quot;oracle model&quot; that is the unpenalized Cox proportional hazards model fitted to the active biomarkers only.
</p>


<h3>Value</h3>

<p>A <code>list</code> of the same length of the <code>time</code> considered. Each element of the list contains between 1 and 3 sublists depending on the chosen validation (i.e. training set [always computed], internal validation through double cross-validation (2CV) [if <code>int.cv</code> = <code>TRUE</code>] and/or external validation [if <code>newdata</code> is provided]). Each sublist is a <code>matrix</code> containing the predictive accuracy metrics of the implemented methods.
</p>


<h3>Author(s)</h3>

<p>Nils Ternes, Federico Rotolo, and Stefan Michiels<br />
Maintainer: Nils Ternes <a href="mailto:nils.ternes@yahoo.com">nils.ternes@yahoo.com</a>
</p>


<h3>References</h3>

<p>Ternes N, Rotolo F and Michiels S.
Empirical extensions of the lasso penalty to reduce
the false discovery rate in high-dimensional Cox regression models.
<em>Statistics in Medicine</em> 2016;35(15):2561-2573.
doi:<a href="http://onlinelibrary.wiley.com/doi/10.1002/sim.6927/abstract">10.1002/sim.6927</a><br />
Ternes N, Rotolo F, Heinze G and Michiels S.
Identification of biomarker-by-treatment interactions in randomized
clinical trials with survival outcomes and high-dimensional spaces.
<em>Biometrical journal</em>. In press.
doi:<a href="http://onlinelibrary.wiley.com/doi/10.1002/bimj.201500234/abstract">10.1002/bimj.201500234</a><br />
Uno H, Cai T, Pencina MJ, DAgostino RB and Wei LJ.
On the C-statistics for evaluating overall adequacy 
of risk prediction procedures with censored survival data.
<em>Statistics in Medicine</em> 2011;30:1105-1117.
doi:<a href="http://onlinelibrary.wiley.com/doi/10.1002/sim.4154/abstract">10.1002/sim.4154</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>########################################
# Simulated data set
########################################

## Low calculation time
  set.seed(654321)
  sdata &lt;- simdata(
    n = 500, p = 20, q.main = 3, q.inter = 0,
    prob.tt = 0.5, alpha.tt = 0,
    beta.main = -0.8,
    b.corr = 0.6, b.corr.by = 4,
    m0 = 5, wei.shape = 1, recr = 4, fu = 2,
    timefactor = 1)
  
  newdata &lt;- simdataV(
    traindata = sdata,
    Nvalid = 500
  )
   
  resBM &lt;- BMsel(
    data = sdata, 
    method = c("lasso", "lasso-pcvl"), 
    inter = FALSE, 
    folds = 5)
  
  predAcc &lt;- predRes(
    res = resBM,
    traindata = sdata,
    newdata = newdata,
    time = 1:5)
    
  plot(predAcc, crit = "C")

## Not run: 
## Moderate calculation time
  set.seed(123456)
  sdata &lt;- simdata(
    n = 500, p = 100, q.main = 5, q.inter = 5,
    prob.tt = 0.5, alpha.tt = -0.5,
    beta.main = c(-0.5, -0.2), beta.inter = c(-0.7, -0.4),
    b.corr = 0.6, b.corr.by = 10,
    m0 = 5, wei.shape = 1, recr = 4, fu = 2,
    timefactor = 1,
    active.inter = c("bm003", "bm021", "bm044", "bm049", "bm097"))

  resBM &lt;- BMsel(
    data = sdata, 
    method = c("lasso", "lasso-pcvl"), 
    inter = TRUE, 
    folds = 5)
  
  predAcc &lt;- predRes(
    res = resBM,
    traindata = sdata, 
    int.cv = TRUE, 
    time = 1:5, 
    ncores = 5)
  plot(predAcc, crit = "dC")

## End(Not run)

########################################
# Breast cancer data set
########################################

## Not run: 
  data(Breast)
  dim(Breast)
  
  set.seed(123456)
  resBM &lt;-  BMsel(
    data = Breast,
    x = 4:ncol(Breast),
    y = 2:1,
    tt = 3,
    inter = FALSE,
    std.x = TRUE,
    folds = 5,
    method = c("lasso", "lasso-pcvl"))

  summary(resBM)

  predAcc &lt;- predRes(
    res = resBM,
    traindata = Breast,
    time = 1:4,
    trace = TRUE)
  plot(predAcc, crit = "C")

## End(Not run)

########################################
########################################
</code></pre>

<hr>
<h2 id='selRes'>Evaluation of the selection accuracy of a prediction model</h2><span id='topic+selRes'></span>

<h3>Description</h3>

<p>This function computes several criteria to assess the selection
accuracy of a prediction model. Of note, this function is only
available for simulated data sets for which true biomarkers are known.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>selRes(res)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="selRes_+3A_res">res</code></td>
<td>
<p>an object of class '<code>resBMsel</code>' generated by <code><a href="#topic+BMsel">BMsel</a></code> with data simulated using <code><a href="#topic+simdata">simdata</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Based on the 2x2 contingency table (active vs. inactive / selected vs. unselected),
four selection criteria are provided: <br />
- the false discovery rate (<code>FDR</code>) that is the proportion of inactive biomarkers
among the selected ones, <br />
- the false non-discovery rate (<code>FNDR</code>) that is the proportion of active biomarkers
among the unselected ones, <br />
- the false negative rate (<code>FNR</code>) that is the proportion of unselected
biomarkers among the active ones,<br />
- the false positive rate (<code>FPR</code>) that is the proportion of selected
biomarkers among the inactive ones.<br />
These four criteria are between 0 and 1, and must be minimized.<br /> 
We also provided two discrimination criteria, translating the ability to discard inactive
biomarkers more likely than active ones independently of the tuning parameters: <br />
- the area under the ROC curve (<code>AUC</code>) depending on the sensitivity [1 - FNR] and specificity [1 - FPR],<br />
- the area under the precision-recall curve (<code>AUPRC</code>) depending on the FNR and FDR (Davis and Goadrich, 2006).<br />
Of note, the AUPRC is more meaningful than the AUC when there are many more inactive than active biomarkers. 
These two criteria are between 0 and 1, and must be maximized.
</p>


<h3>Value</h3>

<p>A <code>matrix</code> of dimension 6 x the number of methods used to fit <code>res</code>.
</p>


<h3>Author(s)</h3>

<p>Nils Ternes, Federico Rotolo, and Stefan Michiels<br />
Maintainer: Nils Ternes <a href="mailto:nils.ternes@yahoo.com">nils.ternes@yahoo.com</a>
</p>


<h3>References</h3>

<p>Davis J and Goadrich M.
The relationship between Precision-Recall and ROC curves.
<em>Proceedings of the 23rd International Conference on Machine Learning</em>.
ACM, Pittsburgh PA, 233-240.<br />
Ternes N, Rotolo F and Michiels S.
Empirical extensions of the lasso penalty to reduce
the false discovery rate in high-dimensional Cox regression models.
<em>Statistics in Medicine</em> 2016;35(15):2561-2573.
doi:<a href="http://onlinelibrary.wiley.com/doi/10.1002/sim.6927/abstract">10.1002/sim.6927</a><br />
Ternes N, Rotolo F, Heinze G and Michiels S.
Identification of biomarker-by-treatment interactions in randomized
clinical trials with survival outcomes and high-dimensional spaces.
<em>Biometrical journal</em>. In press.
doi:<a href="http://onlinelibrary.wiley.com/doi/10.1002/bimj.201500234/abstract">10.1002/bimj.201500234</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>########################################
# Simulated data set
########################################

## Low calculation time
  set.seed(654321)
  sdata &lt;- simdata(
    n = 500, p = 20, q.main = 3, q.inter = 0,
    prob.tt = 0.5, alpha.tt = 0,
    beta.main = -0.8,
    b.corr = 0.6, b.corr.by = 4,
    m0 = 5, wei.shape = 1, recr = 4, fu = 2,
    timefactor = 1)

  resBM &lt;- BMsel(
    data = sdata, 
    method = c("lasso", "lasso-pcvl"), 
    inter = FALSE, 
    folds = 5)
  
  selAcc &lt;- selRes(resBM)

## Not run: 
## Moderate calculation time
  set.seed(123456)
  sdata &lt;- simdata(
    n = 500, p = 100, q.main = 5, q.inter = 5,
    prob.tt = 0.5, alpha.tt = -0.5,
    beta.main = c(-0.5, -0.2), beta.inter = c(-0.7, -0.4),
    b.corr = 0.6, b.corr.by = 10,
    m0 = 5, wei.shape = 1, recr = 4, fu = 2,
    timefactor = 1,
    active.inter = c("bm003", "bm021", "bm044", "bm049", "bm097"))

  resBM &lt;- BMsel(
    data = sdata, 
    method = c("lasso", "lasso-pcvl"), 
    inter = TRUE, 
    folds = 5)
  
  selAcc &lt;- selRes(resBM)

## End(Not run)
</code></pre>

<hr>
<h2 id='simdata'>Generation of data sets with survival outcome</h2><span id='topic+simdata'></span><span id='topic+simdataV'></span>

<h3>Description</h3>

<p>This function simulates a data set with survival outcome with given active biomarkers (prognostic and/or interacting with the treatment).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simdata(n, p, q.main, q.inter, prob.tt, m0, alpha.tt, beta.main,
  beta.inter, b.corr, b.corr.by, wei.shape, recr, fu, timefactor,
  active.main, active.inter)

simdataV(traindata, Nvalid)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simdata_+3A_n">n</code></td>
<td>
<p>the sample size.</p>
</td></tr>
<tr><td><code id="simdata_+3A_p">p</code></td>
<td>
<p>the number of biomarkers.</p>
</td></tr>
<tr><td><code id="simdata_+3A_q.main">q.main</code></td>
<td>
<p>the number of true prognostic biomarkers.</p>
</td></tr>
<tr><td><code id="simdata_+3A_q.inter">q.inter</code></td>
<td>
<p>the number of true biomarkers interacting with the
treatement.</p>
</td></tr>
<tr><td><code id="simdata_+3A_prob.tt">prob.tt</code></td>
<td>
<p>the treatement assignement probability.</p>
</td></tr>
<tr><td><code id="simdata_+3A_m0">m0</code></td>
<td>
<p>the baseline median survival time.</p>
</td></tr>
<tr><td><code id="simdata_+3A_alpha.tt">alpha.tt</code></td>
<td>
<p>the effect of the treatment (in log-scale).</p>
</td></tr>
<tr><td><code id="simdata_+3A_beta.main">beta.main</code></td>
<td>
<p>the effect of the prognostic biomarkers (in log-scale).</p>
</td></tr>
<tr><td><code id="simdata_+3A_beta.inter">beta.inter</code></td>
<td>
<p>the effect of the biomarkers interacting with the
treatment (in log-scale).</p>
</td></tr>
<tr><td><code id="simdata_+3A_b.corr">b.corr</code></td>
<td>
<p>the correlation between biomarker blocks.</p>
</td></tr>
<tr><td><code id="simdata_+3A_b.corr.by">b.corr.by</code></td>
<td>
<p>the size of the blocks of correlated biomarkers.</p>
</td></tr>
<tr><td><code id="simdata_+3A_wei.shape">wei.shape</code></td>
<td>
<p>the shape parameter of the Weibull distribution.</p>
</td></tr>
<tr><td><code id="simdata_+3A_recr">recr</code></td>
<td>
<p>the recruitment period duration.</p>
</td></tr>
<tr><td><code id="simdata_+3A_fu">fu</code></td>
<td>
<p>the follow-up period duration.</p>
</td></tr>
<tr><td><code id="simdata_+3A_timefactor">timefactor</code></td>
<td>
<p>the scale multiplicative factor for times (i.e. 1 = times in years).</p>
</td></tr>
<tr><td><code id="simdata_+3A_active.main">active.main</code></td>
<td>
<p>the list of the prognostic biomarkers (not
mandatory).</p>
</td></tr>
<tr><td><code id="simdata_+3A_active.inter">active.inter</code></td>
<td>
<p>the list of the biomarkers interacting with the
treatment (not mandatory).</p>
</td></tr>
<tr><td><code id="simdata_+3A_traindata">traindata</code></td>
<td>
<p>the training set returned by <code>simdata</code>,
used to generate the new validation data set with the same characteristics.</p>
</td></tr>
<tr><td><code id="simdata_+3A_nvalid">Nvalid</code></td>
<td>
<p>the sample size of the new validation data set.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>simdata</code> function generates <code>p</code> Gaussian unit-variance (<code class="reqn">\sigma</code> = 1) biomarkers including autoregressive correlation (<code class="reqn">\sigma</code>_ij = <code>b.corr</code>^|i-j|) within <code>b.corr.by</code>-biomarker blocks. The number of active biomarkers and their effect sizes (in log-scale) can be specified using <code>q.main</code> and <code>beta.main</code> for the true prognostic biomarkers and using <code>q.inter</code> and <code>beta.inter</code> for the true treatment-effect modifiers. A total of <code>n</code> patients is generated and randomly assigned to the experimental (coded as +0.5, with probability <code>prob.tt</code>) and control treatment (coded as -0.5). The treatment effect is specified using <code>alpha.tt</code> (in log-scale). Survival times are generated using a Weibull with shape <code>wei.shape</code> (i.e. 1 = exponential distribution) and patient-specific scale depending on the baseline median survival time <code>m0</code> and the biomarkers values of the patient.
Censor status is generated by considering independant censoring from a U(<code>fu</code>, <code>fu</code> + <code>recr</code>) distribution, reflecting a trial with <code>recr</code> years of accrual and <code>fu</code> years of follow-up.
Another data set with the same characteristics as the one generated by <code>simdata</code> can be obtained by using the <code>simdataV</code> function.
</p>


<h3>Value</h3>

<p>A simulated <code>data.frame</code> object.
</p>


<h3>Author(s)</h3>

<p>Nils Ternes, Federico Rotolo, and Stefan Michiels<br />
Maintainer: Nils Ternes <a href="mailto:nils.ternes@yahoo.com">nils.ternes@yahoo.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  set.seed(123456)
  sdata &lt;- simdata(
    n = 500, p = 100, q.main = 5, q.inter = 5,
    prob.tt = 0.5, alpha.tt = -0.5,
    beta.main = c(-0.5, -0.2), beta.inter = c(-0.7, -0.4),
    b.corr = 0.6, b.corr.by = 10,
    m0 = 5, wei.shape = 1, recr = 4, fu = 2,
    timefactor = 1,
    active.inter = c("bm003", "bm021", "bm044", "bm049", "bm097"))

  newdata &lt;- simdataV(
    traindata = sdata,
    Nvalid = 500)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
