<!DOCTYPE html><html lang="en"><head><title>Help for package biotools</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {biotools}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#biotools-package'><p> Tools for Biometry and Applied Statistics in Agricultural Science</p></a></li>
<li><a href='#aer'><p> Apparent Error Rate</p></a></li>
<li><a href='#boxM'><p> Box's M-test</p></a></li>
<li><a href='#brazil'>
<p>Brazil Grid</p></a></li>
<li><a href='#confusionmatrix'><p> Confusion Matrix</p></a></li>
<li><a href='#cov2pcov'><p> Partial Covariance Matrix</p></a></li>
<li><a href='#creategroups'><p> Creating Homogeneous Groups</p></a></li>
<li><a href='#D2.disc'><p>Discriminant Analysis Based on Mahalanobis Distance</p></a></li>
<li><a href='#D2.dist'><p> Pairwise Squared Generalized Mahalanobis Distances</p></a></li>
<li><a href='#distClust'><p> Cluster Distance Matrix</p></a></li>
<li><a href='#findSubsample'><p> Finding an Optimized Subsample</p></a></li>
<li><a href='#fitplotsize'><p> Parameter Estimation of the Plot Size Model</p></a></li>
<li><a href='#garlicdist'><p> Distances Between Garlic Cultivars</p></a></li>
<li><a href='#gencovtest'>
<p>Testing Genetic Covariance</p></a></li>
<li><a href='#germinationcount.test'><p> Germination Count Range Test for Seed Sample Heterogeneity</p></a></li>
<li><a href='#maize'>
<p>Maize Data</p></a></li>
<li><a href='#mantelPower'><p>Power of Mantel's Test</p></a></li>
<li><a href='#mantelTest'><p> Mantel's Permutation Test</p></a></li>
<li><a href='#moco'>
<p>Moco Cotton Data</p></a></li>
<li><a href='#multcor.test'><p> Pairwise Correlation t-Test</p></a></li>
<li><a href='#mvpaircomp'>
<p>Multivariate Pairwise Comparisons</p></a></li>
<li><a href='#optimumplotsize'><p> Maximum Curvature Point for Optimum Plot Size</p></a></li>
<li><a href='#pathanalysis'><p> Path Analysis, Simple and Under Collinearity</p></a></li>
<li><a href='#peppercorr'><p> Correlations Between Pepper Variables</p></a></li>
<li><a href='#raise.matrix'><p> Raising a Square Matrix to a Power</p></a></li>
<li><a href='#samplesize'><p> Minimum Sample Size</p></a></li>
<li><a href='#sHe'>
<p>Spatial Analysis of Gene Diversity</p></a></li>
<li><a href='#singh'><p> Importance of Variables According to the Singh (1981) Criterion</p></a></li>
<li><a href='#spatialpred'>
<p>Spatial Predictions Based on the Circular Variable-Radius Moving Window Method</p></a></li>
<li><a href='#tocher'><p> Tocher's Clustering</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Tools for Biometry and Applied Statistics in Agricultural
Science</td>
</tr>
<tr>
<td>Version:</td>
<td>4.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-08-06</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>Author:</td>
<td>Anderson Rodrigo da Silva
    <a href="https://orcid.org/0000-0003-2518-542X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>MASS</td>
</tr>
<tr>
<td>Imports:</td>
<td>utils, stats, graphics, boot, grDevices, datasets</td>
</tr>
<tr>
<td>Suggests:</td>
<td>soilphysics, tkrplot, tcltk, rpanel, lattice, SpatialEpi,
knitr, rmarkdown</td>
</tr>
<tr>
<td>Description:</td>
<td>Tools designed to perform and evaluate cluster analysis (including Tocher's algorithm), 
	discriminant analysis and path analysis (standard and under collinearity), as well as some 
	useful miscellaneous tools for dealing with sample size and optimum plot size calculations. 
	A test for seed sample heterogeneity is now available. Mantel's permutation test can be found in this package. 
	A new approach for calculating its power is implemented. biotools also contains tests for genetic covariance components.
	Heuristic approaches for performing non-parametric spatial predictions of generic response variables and 
	spatial gene diversity are implemented.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://arsilva87.github.io/biotools/">https://arsilva87.github.io/biotools/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/arsilva87/biotools/issues">https://github.com/arsilva87/biotools/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-08-06 21:47:06 UTC; ander</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-08-07 04:40:14 UTC</td>
</tr>
</table>
<hr>
<h2 id='biotools-package'> Tools for Biometry and Applied Statistics in Agricultural Science </h2><span id='topic+biotools-package'></span><span id='topic+biotools'></span>

<h3>Description</h3>

<p>Tools designed to perform and evaluate cluster analysis (including Tocher's algorithm), 
discriminant analysis and path analysis (standard and under collinearity), as well as some 
useful miscellaneous tools for dealing with sample size and optimum plot size calculations. 
A test for seed sample heterogeneity is now available. Mantel's permutation test can be found in this package. 
A new approach for calculating its power is implemented. biotools also contains tests for genetic covariance components.
Heuristic approaches for performing non-parametric spatial predictions of generic response variables and 
spatial gene diversity are implemented.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> biotools</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 4.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2021-08-06</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Note</h3>

<p><em>biotools</em> is an ongoing project.
Any and all criticism, comments and suggestions are welcomed.
</p>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva
</p>
<p>Maintainer: Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>References</h3>

<p>Rao, R.C. (1952) <em>Advanced statistical methods in biometric research</em>. New York: John Wiley &amp; Sons.
</p>
<p>Sharma, J.R. (2006) <em>Statistical and biometrical techniques in plant breeding</em>. Delhi: New Age International.
</p>
<p>Da Silva, A.R.; Malafaia, G.; Menezes, I.P.P. (2017) biotools: an R function to predict 
spatial gene diversity via an individual-based approach. <em>Genetics and Molecular Research</em>, 
16: gmr16029655. 
</p>
<p>Da Silva, A.R., Silva, A.P.A., Tiago-Neto, L.J. (2020) A new local stochastic method for predicting data with spatial heterogeneity. 
<em>ACTA SCIENTIARUM-AGRONOMY</em>, <b>43</b>: e49947.
</p>
<p>Silva, A.R. &amp; Dias, C.T.S. (2013) A cophenetic correlation coefficient for Tocher's method. 
<em>Pesquisa Agropecuaria Brasileira</em>, 48: 589-596.
</p>
<p>Da Silva, A.R. (2020). On testing for seed sample heterogeneity with the exact probability distribution of the germination count range. 
<em>Seed Science Research</em>, <b>30</b>(1): 59-63.
</p>

<hr>
<h2 id='aer'> Apparent Error Rate </h2><span id='topic+aer'></span>

<h3>Description</h3>

<p>A function to calculate the apparent error rate of two classification
vectors, i.e., the proportion of observed cases incorrectly predicted.
It can be useful for evaluating discriminant analysis or other classification systems.
</p>
<p style="text-align: center;"><code class="reqn">aer = \frac{1}{n} \sum_{i=1}^{n} I(y_i \neq \hat{y}_i)</code>
</p>



<h3>Usage</h3>

<pre><code class='language-R'>aer(obs, predict)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="aer_+3A_obs">obs</code></td>
<td>
<p> a vector containing the observed classes. </p>
</td></tr>
<tr><td><code id="aer_+3A_predict">predict</code></td>
<td>
<p> a vector with the same length of <code>obs</code> containing the predicted classes. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>The apparent error rate, a number between 0 (no agreement) and 1 (thorough agreement).
</p>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+confusionmatrix">confusionmatrix</a></code>, <code><a href="MASS.html#topic+lda">lda</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
da &lt;- lda(Species ~ ., data = iris)
pred &lt;- predict(da, dimen = 1)
aer(iris$Species, pred$class)

# End (not run)
</code></pre>

<hr>
<h2 id='boxM'> Box's M-test </h2><span id='topic+boxM'></span>

<h3>Description</h3>

<p>It performs the Box's M-test for homogeneity of covariance matrices
obtained from multivariate normal data according to one classification
factor. The test is based on the chi-square approximation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>boxM(data, grouping)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="boxM_+3A_data">data</code></td>
<td>
<p> a numeric data.frame or matrix containing <em>n</em> observations of <em>p</em> variables;
it is expected that <em>n &gt; p</em>. </p>
</td></tr>
<tr><td><code id="boxM_+3A_grouping">grouping</code></td>
<td>
<p> a vector of length n containing the class of each observation;
it is usualy a factor. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with class &quot;htest&quot; containing the following components:
</p>
<table role = "presentation">
<tr><td><code>statistic</code></td>
<td>
<p>an approximated value of the chi-square distribution.</p>
</td></tr>
<tr><td><code>parameter</code></td>
<td>
<p>the degrees of freedom related of the test statistic in this case that it follows a Chi-square distribution.</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>the p-value of the test.</p>
</td></tr>
<tr><td><code>cov</code></td>
<td>
<p>a list containing the within covariance matrix for each level of <code>grouping</code>.</p>
</td></tr>
<tr><td><code>pooled</code></td>
<td>
<p>the pooled covariance matrix.</p>
</td></tr>
<tr><td><code>logDet</code></td>
<td>
<p>a vector containing the natural logarithm of each matrix in <code>cov</code>.</p>
</td></tr>
<tr><td><code>data.name</code></td>
<td>
<p>a character string giving the names of the data.</p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p>the character string &quot;Box's M-test for Homogeneity of Covariance Matrices&quot;.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>References</h3>

<p>Morrison, D.F. (1976) <em>Multivariate Statistical Methods</em>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
boxM(iris[, -5], iris[, 5])

# End (not run)
</code></pre>

<hr>
<h2 id='brazil'>
Brazil Grid
</h2><span id='topic+brazil'></span>

<h3>Description</h3>

<p>Lat/Long coordinates within Brazil's limits.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("brazil")</code></pre>


<h3>Format</h3>

<p>A data frame with 17141 observations on the following 2 variables.
</p>

<dl>
<dt><code>x</code></dt><dd><p>a numeric vector (longitude)</p>
</dd>
<dt><code>y</code></dt><dd><p>a numeric vector (latitude)</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(brazil)
plot(brazil, cex = 0.1, col = "gray")
</code></pre>

<hr>
<h2 id='confusionmatrix'> Confusion Matrix </h2><span id='topic+confusionmatrix'></span>

<h3>Description</h3>

<p>A function to compute the confusion matrix of two classification
vectors. It can be useful for evaluating discriminant analysis or
other classification systems.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>confusionmatrix(obs, predict)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="confusionmatrix_+3A_obs">obs</code></td>
<td>
<p> a vector containing the observed classes.</p>
</td></tr>
<tr><td><code id="confusionmatrix_+3A_predict">predict</code></td>
<td>
<p> a vector with the same length of <code>obs</code> containing the predicted classes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A square matrix containing the number of objects in each class,
observed (rows) and predicted (columns). Diagonal elements
refers to agreement of <code>obs</code> and <code>predict</code>.
</p>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+aer">aer</a></code>, <code><a href="MASS.html#topic+lda">lda</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
da &lt;- lda(Species ~ ., data = iris)
pred &lt;- predict(da, dimen = 1)
confusionmatrix(iris$Species, pred$class)

# End (not run)
</code></pre>

<hr>
<h2 id='cov2pcov'> Partial Covariance Matrix </h2><span id='topic+cov2pcov'></span>

<h3>Description</h3>

<p>Compute a matrix of partial (co)variances for a group of variables with respect to another.
</p>
<p>Take <code class="reqn">\Sigma</code> as the covariance matrix of dimension <em>p</em>. Now consider dividing <code class="reqn">\Sigma</code> into two groups
of variables. The partial covariance matrices are calculate by:
</p>
<p style="text-align: center;"><code class="reqn"> \Sigma_{11.2} = \Sigma_{11} - \Sigma_{12} \Sigma_{22}^{-1} \Sigma_{21} </code>
</p>

<p style="text-align: center;"><code class="reqn"> \Sigma_{22.1} = \Sigma_{22} - \Sigma_{21} \Sigma_{11}^{-1} \Sigma_{12} </code>
</p>



<h3>Usage</h3>

<pre><code class='language-R'>cov2pcov(m, vars1, vars2 = seq(1, ncol(m))[-vars1])
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cov2pcov_+3A_m">m</code></td>
<td>
<p> a square numeric matrix.</p>
</td></tr>
<tr><td><code id="cov2pcov_+3A_vars1">vars1</code></td>
<td>
<p> a numeric vector indicating the position (rows or columns in <code>m</code>) of the set of variables 
at which to compute the partial covariance matrix.</p>
</td></tr>
<tr><td><code id="cov2pcov_+3A_vars2">vars2</code></td>
<td>
<p> a numeric vector indicating the position (rows or columns in <code>m</code>) of the set of variables 
at which to adjust the partial covariance matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p> A square numeric matrix.</p>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro at hotmail.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+cov">cov</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>(Cl &lt;- cov(longley))
cov2pcov(Cl, 1:2)

# End (Not run)
</code></pre>

<hr>
<h2 id='creategroups'> Creating Homogeneous Groups </h2><span id='topic+creategroups'></span>

<h3>Description</h3>

<p>A function to create homogeneous groups of named objects according
to an objective function evaluated at a covariate. It can be useful
to design experiments which contain a fixed covariate factor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>creategroups(x, ngroups, sizes, fun = mean, tol = 0.01, maxit = 200)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="creategroups_+3A_x">x</code></td>
<td>
<p> a numeric vector of a covariate at which to evaluate the objective function.</p>
</td></tr>
<tr><td><code id="creategroups_+3A_ngroups">ngroups</code></td>
<td>
<p> the number of groups to create.</p>
</td></tr>
<tr><td><code id="creategroups_+3A_sizes">sizes</code></td>
<td>
<p> a numeric vector of length equal to <code>ngroups</code> containing the group sizes. </p>
</td></tr>
<tr><td><code id="creategroups_+3A_fun">fun</code></td>
<td>
<p> the objective function, i.e., to create groups with similar <code>fun</code>; default is <code>mean</code>. </p>
</td></tr>
<tr><td><code id="creategroups_+3A_tol">tol</code></td>
<td>
<p> the tolerance level to define the groups as homogenenous; see details.</p>
</td></tr>
<tr><td><code id="creategroups_+3A_maxit">maxit</code></td>
<td>
<p> the maximum number of iterations; default is 200. </p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>creategroups</code> uses a <code>tol</code> value to evaluate the following statistic:
<code class="reqn">h = \sum_{j}^{ngroups} abs( t_{j+1} - t_j ) / ngroups</code>, where <code class="reqn">t_j = fun(group_j)</code>.
If <code class="reqn">h \leq tol</code>, the groups are considered homogeneous.
</p>


<h3>Value</h3>

<p>A list of
</p>
<table role = "presentation">
<tr><td><code>covar</code></td>
<td>
<p>a character indicating the name of the covariate.</p>
</td></tr>
<tr><td><code>func</code></td>
<td>
<p>a character indicating the name of the objective function.</p>
</td></tr>
<tr><td><code>val.func</code></td>
<td>
<p>a numeric vector containing the values evaluated by <code>func</code> on each group.</p>
</td></tr>
<tr><td><code>niter</code></td>
<td>
<p>the number of iteration require to achieve convergence.</p>
</td></tr>
<tr><td><code>labels</code></td>
<td>
<p>a list containing the labels of the objects in each group.</p>
</td></tr>
<tr><td><code>groups</code></td>
<td>
<p>a list of named vectors containing the values for the groups</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- rnorm(10, 1, 0.5)
names(x) &lt;- letters[1:10]
creategroups(x, ngroups = 2, sizes = c(5, 5))
creategroups(x, ngroups = 3, sizes = c(3, 4, 3), tol = 0.05)

# End (not run)
</code></pre>

<hr>
<h2 id='D2.disc'>Discriminant Analysis Based on Mahalanobis Distance</h2><span id='topic+D2.disc'></span><span id='topic+D2.disc.default'></span><span id='topic+print.D2.disc'></span><span id='topic+predict.D2.disc'></span>

<h3>Description</h3>

<p>A function to perform discriminant analysis based on the squared 
generalized Mahalanobis distance (D2) of the observations to the center of the groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
D2.disc(data, grouping, pooled.cov = NULL)
## S3 method for class 'D2.disc'
print(x, ...)
## S3 method for class 'D2.disc'
predict(object, newdata = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="D2.disc_+3A_data">data</code></td>
<td>
<p> a numeric <code>data.frame</code> or <code>matrix</code> (<em>n x p</em>).</p>
</td></tr>
<tr><td><code id="D2.disc_+3A_grouping">grouping</code></td>
<td>
<p> a vector of length <em>n</em> containing the class of each observation (row) in <code>data</code>.</p>
</td></tr>
<tr><td><code id="D2.disc_+3A_pooled.cov">pooled.cov</code></td>
<td>
<p>a <code>grouping</code>-pooled covariance matrix (<em>p x p</em>). If <code>NULL</code>
(default), <code>D2.disc</code> will automatically compute a pooled covariance matrix.</p>
</td></tr>
<tr><td><code id="D2.disc_+3A_x">x</code>, <code id="D2.disc_+3A_object">object</code></td>
<td>
<p> an object of class <code>"D2.disc"</code>.</p>
</td></tr>
<tr><td><code id="D2.disc_+3A_newdata">newdata</code></td>
<td>
<p> numeric <code>data.frame</code> or <code>matrix</code> of observations to be classified. 
If <code>NULL</code> (default), the input data used as argument in <code>D2.disc</code> will be used.</p>
</td></tr>
<tr><td><code id="D2.disc_+3A_...">...</code></td>
<td>
<p> further arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of
</p>
<table role = "presentation">
<tr><td><code>call</code></td>
<td>
<p>the call which produced the result.</p>
</td></tr>
<tr><td><code>data</code></td>
<td>
<p> numeric matrix; the input data.</p>
</td></tr>
<tr><td><code>D2</code></td>
<td>
<p>a matrix containing the Mahalanobis distances between each row of <code>data</code>
and the center of each class of <code>grouping</code>. In addition, the original
and the predicted (lowest distance) class are displayed, as well as a
chacater vector indicating where the misclassification has occured.</p>
</td></tr>
<tr><td><code>means</code></td>
<td>
<p>a matrix containing the vector of means of each class in <code>grouping</code>.</p>
</td></tr>
<tr><td><code>pooled</code></td>
<td>
<p>the pooled covariance matrix.</p>
</td></tr>
<tr><td><code>confusion.matrix</code></td>
<td>
<p>an object of class <code><a href="#topic+confusionmatrix">confusionmatrix</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;</p>


<h3>References</h3>

<p>Manly, B.F.J. (2004) <em>Multivariate statistical methods</em>: a primer. CRC Press. (p. 105-106).
</p>
<p>Mahalanobis, P.C. (1936) On the generalized distance in statistics. 
<em>Proceedings of The National Institute of Sciences of India</em>, 12:49-55.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+D2.dist">D2.dist</a></code>, <code><a href="#topic+confusionmatrix">confusionmatrix</a></code>, <code><a href="MASS.html#topic+lda">lda</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
(disc &lt;- D2.disc(iris[, -5], iris[, 5]))
first10 &lt;- iris[1:10, -5]
predict(disc, first10)
predict(disc, iris[, -5])$class

# End (not run)
</code></pre>

<hr>
<h2 id='D2.dist'> Pairwise Squared Generalized Mahalanobis Distances </h2><span id='topic+D2.dist'></span>

<h3>Description</h3>

<p>Function to calculate the squared generalized Mahalanobis distance between all pairs of rows
in a data frame with respect to a covariance matrix. The element of the <em>i</em>-th row and
<em>j</em>-th column of the distance matrix is defined as
</p>
<p style="text-align: center;"><code class="reqn">D_{ij}^2 = (\bold{x}_i - \bold{x}_j)' \bold{\Sigma}^{-1} (\bold{x}_i - \bold{x}_j)</code>
</p>



<h3>Usage</h3>

<pre><code class='language-R'>D2.dist(data, cov, inverted = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="D2.dist_+3A_data">data</code></td>
<td>
<p> a data frame or matrix of data (<em>n x p</em>).</p>
</td></tr>
<tr><td><code id="D2.dist_+3A_cov">cov</code></td>
<td>
<p> a variance-covariance matrix (<em>p x p</em>).</p>
</td></tr>
<tr><td><code id="D2.dist_+3A_inverted">inverted</code></td>
<td>
<p> logical. If <code>FALSE</code> (default), <code>cov</code> is supposed to be a variance-covariance matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class &quot;dist&quot;.
</p>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>References</h3>

<p>Mahalanobis, P. C. (1936) On the generalized distance in statistics.
<em>Proceedings of The National Institute of Sciences of India</em>, 12:49-55.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+dist">dist</a></code>, <code><a href="#topic+singh">singh</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Manly (2004, p.65-66)
x1 &lt;- c(131.37, 132.37, 134.47, 135.50, 136.17)
x2 &lt;- c(133.60, 132.70, 133.80, 132.30, 130.33)
x3 &lt;- c(99.17, 99.07, 96.03, 94.53, 93.50)
x4 &lt;- c(50.53, 50.23, 50.57, 51.97, 51.37)
x &lt;- cbind(x1, x2, x3, x4)
Cov &lt;- matrix(c(21.112,0.038,0.078,2.01, 0.038,23.486,5.2,2.844,
	0.078,5.2,24.18,1.134, 2.01,2.844,1.134,10.154), 4, 4)
D2.dist(x, Cov)

# End (not run)
</code></pre>

<hr>
<h2 id='distClust'> Cluster Distance Matrix </h2><span id='topic+distClust'></span>

<h3>Description</h3>

<p>Function to compute a matrix of average distances within and between clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distClust(d, nobj.cluster, id.cluster)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="distClust_+3A_d">d</code></td>
<td>
<p> an object of class &quot;dist&quot; containing the distances between objects.</p>
</td></tr>
<tr><td><code id="distClust_+3A_nobj.cluster">nobj.cluster</code></td>
<td>
<p> a numeric vector containing the numbers of objects per cluster. </p>
</td></tr>
<tr><td><code id="distClust_+3A_id.cluster">id.cluster</code></td>
<td>
<p> a numeric vector for identification of the objects per cluster. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A squared matrix containing distances within (diagonal) and between
(off-diagonal) clusters.
</p>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+tocher">tocher</a></code>, <code><a href="stats.html#topic+dist">dist</a></code>
</p>

<hr>
<h2 id='findSubsample'> Finding an Optimized Subsample</h2><span id='topic+findSubsample'></span>

<h3>Description</h3>

<p> It allows one to find an optimized (minimized or maximized) numeric subsample according to
a statistic of interest. For example, it might be of interest to determine a subsample whose
standard deviation is the lowest among all of those obtained from all possible subsamples of the same size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findSubsample(x, size, fun = sd, minimize = TRUE, niter = 10000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="findSubsample_+3A_x">x</code></td>
<td>
<p> a numeric vector.</p>
</td></tr>
<tr><td><code id="findSubsample_+3A_size">size</code></td>
<td>
<p> an integer; the size of the subsample.</p>
</td></tr>
<tr><td><code id="findSubsample_+3A_fun">fun</code></td>
<td>
<p> an object of class <code>function</code>; the statistic at which to evaluate the subsample.</p>
</td></tr>
<tr><td><code id="findSubsample_+3A_minimize">minimize</code></td>
<td>
<p> logical; if TRUE (default) <code>findSubsample</code> will find a subsample that minimizes <code>stat</code>.</p>
</td></tr>
<tr><td><code id="findSubsample_+3A_niter">niter</code></td>
<td>
<p> an integer indicating the number of iterations, i.e., the number of subsamples to be selected
(without replacement) from the original sample, <code>x</code>. The larger is this number, the more optimized is
the subsample to be found, but this also implies in time-consuming.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of
</p>
<table role = "presentation">
<tr><td><code>dataname</code></td>
<td>
<p>a <code>character</code>.</p>
</td></tr>
<tr><td><code>niter</code></td>
<td>
<p>the number of iterations.</p>
</td></tr>
<tr><td><code>fun</code></td>
<td>
<p>the objective function.</p>
</td></tr>
<tr><td><code>stat</code></td>
<td>
<p>the achieved statistic for the optimized subsample.</p>
</td></tr>
<tr><td><code>criterion</code></td>
<td>
<p>a <code>character</code> indicating the type of optimization.</p>
</td></tr>
<tr><td><code>subsample</code></td>
<td>
<p>a numeric vector; the optimized subsample.</p>
</td></tr>
<tr><td><code>labels</code></td>
<td>
<p> a string containg the labels of the subsample values.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt; </p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+sample">sample</a></code>, <code><a href="#topic+creategroups">creategroups</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1
y &lt;- rnorm(40, 5, 2)
findSubsample(x = y, size = 6)

# Example 2
f &lt;- function(x) diff(range(x)) # max(x) - min(x)
findSubsample(x = y, size = 6, fun = f, minimize = FALSE, niter = 20000)

# End (not run)
</code></pre>

<hr>
<h2 id='fitplotsize'> Parameter Estimation of the Plot Size Model </h2><span id='topic+fitplotsize'></span>

<h3>Description</h3>

<p>Function to estimate the parameters of the nonlinear Lessman &amp; Atkins (1963) model
for determining the optimum plot size as a function of the experimental
coefficient of variation (CV) or as a function of the residual standard error.
</p>
<p style="text-align: center;"><code class="reqn">CV = a * plotsize ^ {-b}.</code>
</p>

<p>It creates initial estimates of the parameters <em>a</em> and <em>b</em> by log-linearization
and uses them to provide its least-squares estimates via <code><a href="stats.html#topic+nls">nls</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitplotsize(plotsize, CV)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fitplotsize_+3A_plotsize">plotsize</code></td>
<td>
<p> a numeric vector containing estimates of plot size.</p>
</td></tr>
<tr><td><code id="fitplotsize_+3A_cv">CV</code></td>
<td>
<p> a numeric vector of experimental coefficient of variation or residual standard error. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="stats.html#topic+nls">nls</a></code> output.
</p>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>References</h3>

<p>Lessman, K. J. &amp; Atkins, R. E. (1963) Optimum plot size and relative
efficiency of lattice designs for grain sorghum yield tests.
<em>Crop Sci</em>., 3:477-481.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+optimumplotsize">optimumplotsize</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ps &lt;- c(1, 2, 3, 4, 6, 8, 12)
cv &lt;- c(35.6, 29, 27.1, 25.6, 24.4, 23.3, 21.6)
out &lt;- fitplotsize(plotsize = ps, CV = cv)
predict(out) # fitted.values
plot(cv ~ ps)
curve(coef(out)[1] * x^(-coef(out)[2]), add = TRUE)

# End (not run)
</code></pre>

<hr>
<h2 id='garlicdist'> Distances Between Garlic Cultivars </h2><span id='topic+garlicdist'></span>

<h3>Description</h3>

<p>The data give the squared generalized Mahalanobis distances between 17 garlic cultivars.
The data are taken from the article published by Silva &amp; Dias (2013).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(garlicdist)</code></pre>


<h3>Format</h3>

<p>An object of class &quot;dist&quot; based on 17 objects.
</p>


<h3>Source</h3>

<p>Silva, A.R. &amp; Dias, C.T.S. (2013) A cophenetic correlation coefficient for
Tocher's method. <em>Pesquisa Agropecuaria Brasileira</em>, 48:589-596. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(garlicdist)
tocher(garlicdist)

# End (not run)
</code></pre>

<hr>
<h2 id='gencovtest'>
Testing Genetic Covariance
</h2><span id='topic+gencovtest'></span><span id='topic+gencovtest.manova'></span><span id='topic+print.gencovtest'></span><span id='topic+plot.gencovtest'></span>

<h3>Description</h3>

<p><code>gencovtest()</code> tests genetic covariance components from a MANOVA model. Two different approaches can
be used: (I) a test statistic that takes into account the genetic and environmental effects and (II) a test 
statistic that only considers the genetic information. The first type refers to tests based on the mean 
cross-products ratio, whose distribution is obtained via Monte Carlo simulation of Wishart matrices. The 
second way of testing genetic covariance refers to tests based upon an adaptation of Wilks' and Pillai's 
statistics for evaluating independence of two sets of variables. All these tests are described by Silva (2015).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'manova'
gencovtest(obj, geneticFactor, gcov = NULL, 
	residualFactor = NULL, adjNrep = 1, 
	test = c("MCPR", "Wilks", "Pillai"), 
	nsim = 9999, 
	alternative = c("two.sided", "less", "greater"))
## S3 method for class 'gencovtest'
print(x, digits = 4, ...)
## S3 method for class 'gencovtest'
plot(x, var1, var2, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gencovtest_+3A_obj">obj</code></td>
<td>
<p> an object of class <code>"manova"</code>.</p>
</td></tr>
<tr><td><code id="gencovtest_+3A_geneticfactor">geneticFactor</code></td>
<td>
<p> a character indicating the genetic factor from which to test covariance components.
It must be declared as a factor in the manova object.</p>
</td></tr>
<tr><td><code id="gencovtest_+3A_gcov">gcov</code></td>
<td>
<p> optional; a matrix containing estimates of genetic covariances to be tested. If 
<code>NULL</code> (default), an estimate is obtained via method of moments.</p>
</td></tr>
<tr><td><code id="gencovtest_+3A_residualfactor">residualFactor</code></td>
<td>
<p> optional; a character indicating a source in the manova model to be used as
error term. If <code>NULL</code> (default), the usual term &quot;Residuals&quot; will be used.</p>
</td></tr>
<tr><td><code id="gencovtest_+3A_adjnrep">adjNrep</code></td>
<td>
<p> a correction index for dealing with unbalanced data. See details.</p>
</td></tr>
<tr><td><code id="gencovtest_+3A_test">test</code></td>
<td>
<p> a character indicating the test. It must be on of the following:
<code>"MCPR"</code> - the empirical type-I test based on Mean Cross-Products Ratios via Wishart simulation,
<code>"Wilks"</code> - a type-II test based on the partial Wilks' Lambda,
<code>"Pillai"</code> - a type-II test based on the partial Pillai's statistic.</p>
</td></tr>
<tr><td><code id="gencovtest_+3A_nsim">nsim</code></td>
<td>
<p> the number of Monte Carlo simulations. Used only if <code>test = "MCPR"</code>.</p>
</td></tr>
<tr><td><code id="gencovtest_+3A_alternative">alternative</code></td>
<td>
<p> the type of alternative hypothesis. Used only if <code>test = "MCPR"</code>. So far,
only the option <code>"two.sided"</code> is implemented.</p>
</td></tr>
<tr><td><code id="gencovtest_+3A_x">x</code></td>
<td>
<p> an object of class <code>"gencovtest"</code>.</p>
</td></tr>
<tr><td><code id="gencovtest_+3A_digits">digits</code></td>
<td>
<p> the number of digits to be displayed by the print method.</p>
</td></tr>
<tr><td><code id="gencovtest_+3A_var1">var1</code></td>
<td>
<p> a character of integer indicating one of the two response variable or its position.</p>
</td></tr>
<tr><td><code id="gencovtest_+3A_var2">var2</code></td>
<td>
<p> a character of integer indicating one of the two response variable or its position.</p>
</td></tr>
<tr><td><code id="gencovtest_+3A_...">...</code></td>
<td>
<p> further arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The genetic covariance matrix is currently estimated via method of moments, following the equation:
</p>
<p style="text-align: center;"><code class="reqn">G = (Mg - Me) / (nrep * adjNrep)</code>
</p>

<p>where <code class="reqn">Mg</code> and <code class="reqn">Me</code> are the matrices of mean cross-products associated with the genetic factor and 
the residuals, respectively; <code class="reqn">nrep</code> is the number of replications, calculated as the ratio between the 
total number of observations and the number of levels of the genetic factor; <code class="reqn">adjNrep</code> is supposed to 
adjust nrep, specially when estimating <code class="reqn">G</code> from unbalanced data.
</p>


<h3>Value</h3>

<p>An object of class <code>gencovtest</code>, a list of
</p>
<table role = "presentation">
<tr><td><code>gcov</code></td>
<td>
<p> a p-dimensional square matrix containing estimates of the genetic covariances.</p>
</td></tr>
<tr><td><code>gcor</code></td>
<td>
<p> a p-dimensional square matrix containing estimates of the genetic correlations.</p>
</td></tr>
<tr><td><code>test</code></td>
<td>
<p> the test (as input).</p>
</td></tr>
<tr><td><code>statistics</code></td>
<td>
<p> a p-dimensional square matrix containing the <code>test</code> statistics. 
If <code>test = "MCPR"</code> the mean cross-products ratios are computed; if <code>test = "Wilks"</code>
the Wilks' Lambda is; and <code>test = "Pillai"</code> results on Pillai's <code class="reqn">Tn</code>.</p>
</td></tr>
<tr><td><code>p.values</code></td>
<td>
<p> a p-dimensional square matrix containing the associated p-values.</p>
</td></tr>
<tr><td><code>alternative</code></td>
<td>
<p> the type of alternative hypothesis (as input).</p>
</td></tr>
<tr><td><code>X2</code></td>
<td>
<p> a p-dimensional square matrix containing the Chi-square (D.f. = 1) approximation for Wilks's 
and Pillai's statistics. Stored only if one of these two tests is chosen.</p>
</td></tr>
<tr><td><code>simRatio</code></td>
<td>
<p> an array consisting of <code>nsim</code> p-dimensional matrices containing the simulated mean 
cross-products ratios.</p>
</td></tr>
<tr><td><code>dfg</code></td>
<td>
<p> the number of degrees of freedom associated with the genetic factor.</p>
</td></tr>
<tr><td><code>dfe</code></td>
<td>
<p> the number of degrees of freedom associated with the residual term.</p>
</td></tr>
</table>


<h3>Warning</h3>

<p>When using the MCPR test, be aware that <code>dfg</code> should be equal or greater than the number of variables (p). 
Otherwise the simulation of Wishart matrices may not be done.
</p>
<p>A collinearity diagnosis is carried out using the condition number (CN), for the inferences may be affected by the 
quality of <code class="reqn">G</code>. Thus, if CN &gt; 100, a warning message is displayed.
</p>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>References</h3>

<p>Silva, A.R. (2015) <em>On Testing Genetic Covariance</em>. LAP Lambert Academic Publishing. ISBN 3659716553
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+manova">manova</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># MANOVA
data(maize)
M &lt;- manova(cbind(NKPR, ED, CD, PH) ~ family + env, data = maize)
summary(M)

# Example 1 - MCPR
t1 &lt;- gencovtest(obj = M, geneticFactor = "family")
print(t1)
plot(t1, "ED", "PH")

# Example 2 - Pillai
t2 &lt;- gencovtest(obj = M, geneticFactor = "family", test = "Pillai")
print(t2)
plot(t2, "ED", "PH")

# End (not run)
</code></pre>

<hr>
<h2 id='germinationcount.test'> Germination Count Range Test for Seed Sample Heterogeneity </h2><span id='topic+germinationcount.test'></span>

<h3>Description</h3>

<p>A test based on the exact probability distribution of the germination count range, i.e, the difference between germination count of seed samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>germinationcount.test(r, nsamples, n, N, K)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="germinationcount.test_+3A_r">r</code></td>
<td>
<p> an integer representing the germination count difference between seed samples.</p>
</td></tr> 
<tr><td><code id="germinationcount.test_+3A_nsamples">nsamples</code></td>
<td>
<p> an integer representing the number of seed samples.</p>
</td></tr>
<tr><td><code id="germinationcount.test_+3A_n">n</code></td>
<td>
<p> an integer representing the number of seeds per sample.</p>
</td></tr>
<tr><td><code id="germinationcount.test_+3A_n">N</code></td>
<td>
<p> an integer representing the size (number of seeds) of the seed lot.</p>
</td></tr>
<tr><td><code id="germinationcount.test_+3A_k">K</code></td>
<td>
<p> an integer representing the number of germinating seeds in the seed lot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of
</p>
<table role = "presentation">
<tr><td><code>R.value</code></td>
<td>
<p> integer; the input R-value (<code>r</code>).</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p> numeric; the exact p-value.</p>
</td></tr>
<tr><td><code>germination.rate</code></td>
<td>
<p> numeric; the germination rate of the seed lot, calculated as the ration of <code>K</code> and <code>N</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>References</h3>

<p>Da Silva, A.R. (2020). On testing for seed sample heterogeneity with the exact probability distribution of the germination count range. 
<em>Seed Science Research</em>, <b>30</b>(1): 59&ndash;63. doi:10.1017/S0960258520000112
</p>


<h3>Examples</h3>

<pre><code class='language-R'>germinationcount.test(r = 6, nsamples = 4, n = 50, N = 2000, K = 1700)

# End (Not run)
</code></pre>

<hr>
<h2 id='maize'>
Maize Data
</h2><span id='topic+maize'></span>

<h3>Description</h3>

<p>Data from and experiment with five maize families carried out in randomized block design, 
with four replications (environments).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("maize")</code></pre>


<h3>Format</h3>

<p>A data frame with 20 observations on the following 6 variables.
</p>

<dl>
<dt><code>NKPR</code></dt><dd><p>a numeric vector containing values of Number of Kernels Per cob Row.</p>
</dd>
<dt><code>ED</code></dt><dd><p>a numeric vector containing values of Ear Diameter (in cm).</p>
</dd>
<dt><code>CD</code></dt><dd><p>a numeric vector containing values of Cob Diameter (in cm).</p>
</dd>
<dt><code>PH</code></dt><dd><p>a numeric vector containing values of Plant Heigth (in m).</p>
</dd>
<dt><code>family</code></dt><dd><p>a factor with levels <code>1</code> <code>2</code> <code>3</code> <code>4</code> <code>5</code></p>
</dd>
<dt><code>env</code></dt><dd><p>a factor with levels <code>1</code> <code>2</code> <code>3</code> <code>4</code></p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(maize)
str(maize)
summary(maize)
</code></pre>

<hr>
<h2 id='mantelPower'>Power of Mantel's Test</h2><span id='topic+mantelPower'></span>

<h3>Description</h3>

<p>Power calculation of Mantel's permutation test.</p>


<h3>Usage</h3>

<pre><code class='language-R'>mantelPower(obj, effect.size = seq(0, 1, length.out = 50), alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mantelPower_+3A_obj">obj</code></td>
<td>
<p> an object of class &quot;mantelTest&quot;. See <code><a href="#topic+mantelTest">mantelTest</a></code>. </p>
</td></tr>
<tr><td><code id="mantelPower_+3A_effect.size">effect.size</code></td>
<td>
<p> numeric; the effect size specifying the alternative hypothesis.</p>
</td></tr>
<tr><td><code id="mantelPower_+3A_alpha">alpha</code></td>
<td>
<p> numeric; the significance level at which to compute the power level.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing the effect size and its respective power level.
</p>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>References</h3>

<p>Silva, A.R.; Dias, C.T.S.; Cecon, P.R.; Rego, E.R. (2015). An alternative procedure for performing a 
power analysis of Mantel's test. <em>Journal of Applied Statistics</em>, <b>42</b>(9): 1984-1992.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mantelTest">mantelTest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Mantel test
data(garlicdist)
garlic &lt;- tocher(garlicdist)
coph &lt;- cophenetic(garlic)
mt1 &lt;- mantelTest(garlicdist, coph, xlim = c(-1, 1))

# Power calculation, H1: rho = 0.3
mantelPower(mt1, effect.size = 0.3)

# Power calculation, multiple H1s and different alphas
p01 &lt;- mantelPower(mt1, alpha = 0.01)
p05 &lt;- mantelPower(mt1, alpha = 0.05)
p10 &lt;- mantelPower(mt1, alpha = 0.10)
plot(p01, type = "l", col = 4)
lines(p05, lty = 2, col = 4)
lines(p10, lty = 3, col = 4)
legend("bottomright", c("0.10", "0.05", "0.01"), 
	title = expression(alpha), col = 4, lty = 3:1, cex = 0.8)

# End (Not run)
</code></pre>

<hr>
<h2 id='mantelTest'> Mantel's Permutation Test </h2><span id='topic+mantelTest'></span>

<h3>Description</h3>

<p>Mantel's permutation test based on Pearson's correlation coefficient 
to evaluate the association between two distance square matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mantelTest(m1, m2, nperm = 999, alternative = "greater", 
	graph = TRUE, main = "Mantel's test", xlab = "Correlation", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mantelTest_+3A_m1">m1</code></td>
<td>
<p> an object of class &quot;matrix&quot; or &quot;dist&quot;, containing distances among <em>n</em> individuals.</p>
</td></tr> 
<tr><td><code id="mantelTest_+3A_m2">m2</code></td>
<td>
<p> an object of class &quot;matrix&quot; or &quot;dist&quot;, containing distances among <em>n</em> individuals.</p>
</td></tr>
<tr><td><code id="mantelTest_+3A_nperm">nperm</code></td>
<td>
<p> the number of matrix permutations.</p>
</td></tr>
<tr><td><code id="mantelTest_+3A_alternative">alternative</code></td>
<td>
<p> a character specifying the alternative hypothesis. It must be one of &quot;greater&quot; (default),
&quot;two.sided&quot; or &quot;less&quot;.</p>
</td></tr>
<tr><td><code id="mantelTest_+3A_graph">graph</code></td>
<td>
<p> logical; if TRUE (default), the empirical distribution is plotted.</p>
</td></tr>
<tr><td><code id="mantelTest_+3A_main">main</code></td>
<td>
<p> opitional; a character describing the title of the graphic.</p>
</td></tr>
<tr><td><code id="mantelTest_+3A_xlab">xlab</code></td>
<td>
<p>opitional; a character describing the <em>x</em>-axis label.</p>
</td></tr>
<tr><td><code id="mantelTest_+3A_...">...</code></td>
<td>
<p> further graphical arguments. See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of
</p>
<table role = "presentation">
<tr><td><code>correlation</code></td>
<td>
<p> numeric; the observed Pearson's correlation between <code>m1</code> and <code>m2</code>.</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p> numeric; the empirical p-value of the permutation test.</p>
</td></tr>
<tr><td><code>alternative</code></td>
<td>
<p> character; the alternative hypothesis used to compute <code>p.value</code>.</p>
</td></tr>
<tr><td><code>nullcor</code></td>
<td>
<p> numeric vector containing randomized values of correlation, i.e., under the null hypothesis
that the true correlation is equal to zero.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>References</h3>

<p>Mantel, N. (1967). The detection of disease clustering and a generalized regression approach. 
<em>Cancer Research</em>, 27:209&ndash;220.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mantelPower">mantelPower</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Distances between garlic cultivars
data(garlicdist)
garlicdist

# Tocher's clustering
garlic &lt;- tocher(garlicdist)
garlic

# Cophenetic distances
coph &lt;- cophenetic(garlic)
coph

# Mantel's test
mantelTest(garlicdist, coph, 
	xlim = c(-1, 1))

# End (Not run)
</code></pre>

<hr>
<h2 id='moco'>
Moco Cotton Data
</h2><span id='topic+moco'></span>

<h3>Description</h3>

<p>Data set of...
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("moco")</code></pre>


<h3>Format</h3>

<p>A data frame with 206 observations (sampling points) on the following 20 variables (coordinates and markers).
</p>

<dl>
<dt><code>Lon</code></dt><dd><p>a numeric vector containing values of longitude</p>
</dd>
<dt><code>Lat</code></dt><dd><p>a numeric vector containing values of latitude</p>
</dd>
<dt><code>BNL1434.1</code></dt><dd><p>a numeric vector (marker)</p>
</dd>
<dt><code>BNL1434.2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BNL840.1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BNL840.2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BNL2496.1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BNL2496.2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BNL1421.1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BNL1421.2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BNL1551.1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BNL1551.2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CIR249.1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CIR249.2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BNL3103.1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BNL3103.2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CIR311.1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CIR311.2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CIR246.1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CIR246.2</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>...
</p>


<h3>References</h3>

<p>...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(moco)
str(moco)
</code></pre>

<hr>
<h2 id='multcor.test'> Pairwise Correlation t-Test </h2><span id='topic+multcor.test'></span>

<h3>Description</h3>

<p>It performs multiple correlation t-tests from a correlation matrix based
on the statistic:
</p>
<p style="text-align: center;"><code class="reqn">t = r * \sqrt(df / (1 - r^2))</code>
</p>

<p>where, in general, <code class="reqn">df = n - 2</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multcor.test(x, n = NULL, Df = NULL,
	alternative = c("two.sided", "less", "greater"), adjust = "none")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multcor.test_+3A_x">x</code></td>
<td>
<p> a correlation matrix.</p>
</td></tr>
<tr><td><code id="multcor.test_+3A_n">n</code></td>
<td>
<p> the number of observations; if <code>NULL</code> (default), the argument 
<code>Df</code> must be passed.</p>
</td></tr>
<tr><td><code id="multcor.test_+3A_df">Df</code></td>
<td>
<p> the number of degrees of freedom of the t statistic; if <code>NULL</code> (default),
the argument <code>n</code> must be passed and, in this case, <code>multcor.test</code>
considers <code class="reqn">Df = n - 2</code>.</p>
</td></tr>
<tr><td><code id="multcor.test_+3A_alternative">alternative</code></td>
<td>
<p> the alternative hypothesis. It must be one of &quot;two.sided&quot;,
&quot;greater&quot; or &quot;less&quot;. You can specify just the initial letter.
&quot;greater&quot; corresponds to positive association,&quot;less&quot; to negative association.
The default is &quot;two.sided&quot;.</p>
</td></tr>
<tr><td><code id="multcor.test_+3A_adjust">adjust</code></td>
<td>
<p> The adjustment method for multiple tests. It must be one of
&quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;,
&quot;none&quot; (default). For more information, see <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with class &quot;multcor.test&quot; containing the following components: 
</p>
<table role = "presentation">
<tr><td><code>t.values</code></td>
<td>
<p>the t-value calculated for each correlation.</p>
</td></tr>
<tr><td><code>p.values</code></td>
<td>
<p>the p.value for each t-test, adjusted for multiple tests.</p>
</td></tr>
<tr><td><code>p.check</code></td>
<td>
<p>a matrix containing the <code>p.values</code> for each t-test (lower triangular)
and a symbol indicating the significance level at which one can to reject the null hypothesis
(upper triangular).</p>
</td></tr>
<tr><td><code>adjustemnt</code></td>
<td>
<p>a character indicating the p-value adjustment method.</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>the degrees of freedom of the tests.</p>
</td></tr>
<tr><td><code>alternative</code></td>
<td>
<p>a character indicating the type of alternative hypothesis.</p>
</td></tr>
<tr><td><code>data.name</code></td>
<td>
<p>a character string giving the name of the data.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+cor">cor</a></code>, <code><a href="stats.html#topic+cor.test">cor.test</a></code>, <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(peppercorr)
multcor.test(peppercorr, n = 20)

# End (not run)
</code></pre>

<hr>
<h2 id='mvpaircomp'>
Multivariate Pairwise Comparisons
</h2><span id='topic+mvpaircomp'></span><span id='topic+print.mvpaircomp'></span>

<h3>Description</h3>

<p>Performs pairwise comparisons of multivariate mean vectors of factor levels, overall or nested.
The tests are run in the same spirt of <code>summary.manova()</code>, based on multivariate statistics such as Pillai's trace 
and Wilks' lambda, which can be applied to test multivariate contrasts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mvpaircomp(model, factor1, nesting.factor = NULL, 
   test = "Pillai", adjust = "none", SSPerror = NULL, DFerror = NULL) 

## S3 method for class 'mvpaircomp'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mvpaircomp_+3A_model">model</code></td>
<td>
<p> a multivariate analysis of variance (MANOVA) model, fitted using <code>lm()</code> or <code>manova()</code>; an object of class <code>"mlm"</code>.</p>
</td></tr>
<tr><td><code id="mvpaircomp_+3A_factor1">factor1</code></td>
<td>
<p> a character string indicating a factor declared in the <code>model</code>, whose levels will be compared in pairs.</p>
</td></tr>
<tr><td><code id="mvpaircomp_+3A_nesting.factor">nesting.factor</code></td>
<td>
<p> optional; a character string indicating a factor also declared in <code>model</code> whose levels will nest the contrasts performed
with <code>factor1</code>. <code>factor1</code> and <code>nesting.factor</code> can have pretty much any form or relationship, that is,
not necessarily nested one another.</p>
</td></tr>
<tr><td><code id="mvpaircomp_+3A_test">test</code></td>
<td>
<p> a character string indicating the type of multivariate statistics to be calculated to perform the
F-test approximation. Default is <code>"Pillai"</code>. Other options are: <code>"Wilks"</code>, <code>"Hotelling-Lawley"</code> and <code>"Roy"</code>.
But they use to give very close results.</p>
</td></tr>
<tr><td><code id="mvpaircomp_+3A_adjust">adjust</code></td>
<td>
<p> a character string indicating the p-value adjustment method for multiple comparisons. Default is <code>"none"</code>.
See <code><a href="stats.html#topic+p.adjust">p.adjust</a></code></p>
</td></tr>
<tr><td><code id="mvpaircomp_+3A_ssperror">SSPerror</code></td>
<td>
<p> optional; a numeric matrix representing the residual sum of squares and cross-products, to be used to compute the
multivariate statistics.</p>
</td></tr>
<tr><td><code id="mvpaircomp_+3A_dferror">DFerror</code></td>
<td>
<p> optional; a numeric value representing the residual degrees of freedom, to be used to compute the
multivariate statistics.</p>
</td></tr>
<tr><td><code id="mvpaircomp_+3A_x">x</code></td>
<td>
<p> an object of class <code>mvpaircomp</code>.</p>
</td></tr>
<tr><td><code id="mvpaircomp_+3A_...">...</code></td>
<td>
<p> further arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>mvpaircomp</code>, a list of
</p>
<table role = "presentation">
<tr><td><code>st</code></td>
<td>
<p> an array containing the summary of the multivariate tests.</p>
</td></tr>
<tr><td><code>SSPcontrast</code></td>
<td>
<p> an array containing p-dimensional square matrices of sum of squares and cross-products of the contrasts.</p>
</td></tr>
<tr><td><code>adjust</code></td>
<td>
<p> a character string indicating the p-value adjustment method used.</p>
</td></tr>
<tr><td><code>fac1</code></td>
<td>
<p> a character string indicating the factor being tested.</p>
</td></tr>
<tr><td><code>fac2</code></td>
<td>
<p> a character string indicating the nesting factor.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>References</h3>

<p>Krzanowski, W. J. (1988) <em>Principles of Multivariate Analysis</em>. A User's Perspective. Oxford. 
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+manova">manova</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1
data(maize)
M &lt;- lm(cbind(NKPR, ED, CD, PH) ~ family + env, data = maize)
anova(M)  # MANOVA table
mvpaircomp(M, factor1 = "family", adjust = "bonferroni")

# Example 2 (with nesting factor)
# Data on producing plastic film from Krzanowski (1998, p. 381)
tear &lt;- c(6.5, 6.2, 5.8, 6.5, 6.5, 6.9, 7.2, 6.9, 6.1, 6.3,
          6.7, 6.6, 7.2, 7.1, 6.8, 7.1, 7.0, 7.2, 7.5, 7.6)
gloss &lt;- c(9.5, 9.9, 9.6, 9.6, 9.2, 9.1, 10.0, 9.9, 9.5, 9.4,
           9.1, 9.3, 8.3, 8.4, 8.5, 9.2, 8.8, 9.7, 10.1, 9.2)
opacity &lt;- c(4.4, 6.4, 3.0, 4.1, 0.8, 5.7, 2.0, 3.9, 1.9, 5.7,
             2.8, 4.1, 3.8, 1.6, 3.4, 8.4, 5.2, 6.9, 2.7, 1.9)
Y &lt;- cbind(tear, gloss, opacity)
rate     &lt;- gl(2, 10, labels = c("Low", "High"))
additive &lt;- gl(2, 5, length = 20, labels = c("Low", "High"))

fit &lt;- manova(Y ~ rate * additive)
summary(fit, test = "Wilks")  # MANOVA table
mvpaircomp(fit, factor1 = "rate", nesting.factor = "additive", test = "Wilks")
mvpaircomp(fit, factor1 = "additive", nesting.factor = "rate", test = "Wilks")

# End (not run)
</code></pre>

<hr>
<h2 id='optimumplotsize'> Maximum Curvature Point for Optimum Plot Size </h2><span id='topic+optimumplotsize'></span>

<h3>Description</h3>

<p>The Meier &amp; Lessman (1971) method to determine the maximum curvature point
for optimum plot size as a function of the experimental coefficient of variation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>optimumplotsize(a, b)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="optimumplotsize_+3A_a">a</code></td>
<td>
<p> a parameter estimate of the plot size model; see <code><a href="#topic+fitplotsize">fitplotsize</a></code>. </p>
</td></tr>
<tr><td><code id="optimumplotsize_+3A_b">b</code></td>
<td>
<p> a parameter estimate of the plot size model; see <code><a href="#topic+fitplotsize">fitplotsize</a></code>. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>The (approximated) optimum plot size value.
</p>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>References</h3>

<p>Meier, V. D. &amp; Lessman, K. J. (1971) Estimation of optimum field plot
shape and size for testing yield in Crambe abyssinia Hochst.
<em>Crop Sci</em>., 11:648-650.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fitplotsize">fitplotsize</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ps &lt;- c(1, 2, 3, 4, 6, 8, 12)
cv &lt;- c(35.6, 29, 27.1, 25.6, 24.4, 23.3, 21.6)
out &lt;- fitplotsize(plotsize = ps, CV = cv)
plot(cv ~ ps)
curve(coef(out)[1] * x^(-coef(out)[2]), add = TRUE)
optimumplotsize(a = coef(out)[1], b = coef(out)[2])

# End (not run)
</code></pre>

<hr>
<h2 id='pathanalysis'> Path Analysis, Simple and Under Collinearity </h2><span id='topic+pathanalysis'></span>

<h3>Description</h3>

<p>Function to perform the simple path analysis and the path analysis
under collinearity (sometimes called <em>ridge path analysis</em>). It computes
the direct (diagonal) and indirect (off-diagonal) effects of each explanatory
variable over a response one.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathanalysis(corMatrix, resp.col, collinearity = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pathanalysis_+3A_cormatrix">corMatrix</code></td>
<td>
<p> a correlation matrix. </p>
</td></tr>
<tr><td><code id="pathanalysis_+3A_resp.col">resp.col</code></td>
<td>
<p> an integer value indicating the column in <code>corMatrix</code> that corresponds to
the response variable. </p>
</td></tr>
<tr><td><code id="pathanalysis_+3A_collinearity">collinearity</code></td>
<td>
<p> logical; if <code>TRUE</code>, an external interactive display is
used to pass a value, say k, at which to evaluate the system:
(<b>X'X</b> + <b>I</b>k)<b>B</b> = <b>X'Y</b>, being <b>X'X</b> the correlation matrix between explanatory variables,
<b>X'Y</b> the correlation vector between all explanatory variables and the response variable,
<b>B</b> is the vector of path coefficients and <em>k</em> is a value between 0 and 1; default is <code>FALSE</code>, i.e., k = 0. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of
</p>
<table role = "presentation">
<tr><td><code>coef</code></td>
<td>
<p> a matrix containing the direct (diagonal) and indirect (off-diagonal)
effects of each variable.</p>
</td></tr>
<tr><td><code>Rsq</code></td>
<td>
<p> the coefficient of determination.</p>
</td></tr>
<tr><td><code>ResidualEffect</code></td>
<td>
<p> the residual effect.</p>
</td></tr>
<tr><td><code>VIF</code></td>
<td>
<p> a vector containing the variance inflation factors.</p>
</td></tr>
<tr><td><code>CN</code></td>
<td>
<p> the condition number.</p>
</td></tr>
</table>


<h3>Side Effects </h3>

<p>If <code>collinearity = TRUE</code>, an interactive graphic is displayed for dealing with collinearity.
</p>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>References</h3>

<p>Carvalho, S.P. (1995) <em>Metodos alternativos de estimacao de coeficientes de
trilha e indices de selecao, sob multicolinearidade</em>. Ph.D. Thesis, 
Federal University of Vicosa (UFV), Vicosa, MG, Brazil.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(peppercorr)
pathanalysis(peppercorr, 6, collinearity = FALSE)

# End (not run)
</code></pre>

<hr>
<h2 id='peppercorr'> Correlations Between Pepper Variables</h2><span id='topic+peppercorr'></span>

<h3>Description</h3>

<p>The data give the correlations between 6 pepper variables. The data are taken from the
article published by Silva et al. (2013).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(peppercorr)</code></pre>


<h3>Format</h3>

<p>An object of class &quot;matrix&quot;.
</p>


<h3>Source</h3>

<p>Silva et al. (2013) Path analysis in multicollinearity for fruit traits of pepper.
<em>Idesia</em>, 31:55-60.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(peppercorr)
print(peppercorr)

# End (not run)
</code></pre>

<hr>
<h2 id='raise.matrix'> Raising a Square Matrix to a Power </h2><span id='topic+raise.matrix'></span>

<h3>Description</h3>

<p><code>raise.matrix</code> raises a square matrix to a power by using
spectral decomposition.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>raise.matrix(x, power = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="raise.matrix_+3A_x">x</code></td>
<td>
<p> a square matrix. </p>
</td></tr>
<tr><td><code id="raise.matrix_+3A_power">power</code></td>
<td>
<p> numeric; default is 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class &quot;matrix&quot;.
</p>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+eigen">eigen</a></code>, <code><a href="base.html#topic+svd">svd</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- matrix(c(1, -2, -2, 4), 2, 2)
raise.matrix(m)
raise.matrix(m, 2)

# End (not run)
</code></pre>

<hr>
<h2 id='samplesize'> Minimum Sample Size </h2><span id='topic+samplesize'></span>

<h3>Description</h3>

<p>Function to determine the minimum sample size for calculating a
statistic based on its the confidence interval.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>samplesize(x, fun, sizes = NULL, lcl = NULL, ucl = NULL,
	nboot = 200, conf.level = 0.95, nrep = 500, graph = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="samplesize_+3A_x">x</code></td>
<td>
<p> a numeric vector. </p>
</td></tr>
<tr><td><code id="samplesize_+3A_fun">fun</code></td>
<td>
<p> an objective function at which to evaluate the sample size; see details.</p>
</td></tr>
<tr><td><code id="samplesize_+3A_sizes">sizes</code></td>
<td>
<p> a numeric vector containing sample sizes; if <code>NULL</code> (default), <code>samplesize</code> 
creates a vector ranging from 2 to <em>n</em>-1. </p>
</td></tr>
<tr><td><code id="samplesize_+3A_lcl">lcl</code></td>
<td>
<p> the lower confidence limit for the statistic defined in <code>fun</code>; 
if <code>NULL</code> (default), <code>samplesize</code> estimates <code>lcl</code> based on bootstrap 
percentile interval. </p>
</td></tr>
<tr><td><code id="samplesize_+3A_ucl">ucl</code></td>
<td>
<p> the upper confidence limit for the statistic defined in <code>fun</code>; 
if <code>NULL</code> (default), <code>samplesize</code> estimates <code>ucl</code> based on bootstrap 
percentile interval. </p>
</td></tr>
<tr><td><code id="samplesize_+3A_nboot">nboot</code></td>
<td>
<p> the number of bootstrap samples; it is used only if <code>lcl</code> or <code>ucl</code> is <code>NULL</code>. </p>
</td></tr>
<tr><td><code id="samplesize_+3A_conf.level">conf.level</code></td>
<td>
<p> the confidence level for calculating the <code>lcl</code> and <code>ucl</code>; 
it is used only if <code>lcl</code> or <code>ucl</code> is <code>NULL</code>. </p>
</td></tr>
<tr><td><code id="samplesize_+3A_nrep">nrep</code></td>
<td>
<p> the resampling (with replacement) number for each sample size in <code>sizes</code>; 
default is 500. </p>
</td></tr>
<tr><td><code id="samplesize_+3A_graph">graph</code></td>
<td>
<p> logical; default is <code>TRUE</code>. </p>
</td></tr>
<tr><td><code id="samplesize_+3A_...">...</code></td>
<td>
<p> further graphical arguments. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>ucl</code> or <code>lcl</code> is <code>NULL</code>, <code>fun</code> must be defined as in <code><a href="boot.html#topic+boot">boot</a></code>, i.e.,
the first argument passed will always be the original data and the second will be a vector of indices,
frequencies or weights which define the bootstrap sample. By now, <code>samplesize</code> considers the
second argument only as index.
</p>


<h3>Value</h3>

<p>A list of
</p>
<table role = "presentation">
<tr><td><code>CI</code></td>
<td>
<p>a vector containing the lower and the upper confidence limit for the statistic evaluated.</p>
</td></tr>
<tr><td><code>pointsOut</code></td>
<td>
<p>a data frame containing the sample sizes (in <code>sizes</code>), the number of 
points outside the CI (<code>n.out</code>) and the proportion of this number (<code>prop</code>). </p>
</td></tr>
</table>


<h3>Side Effects </h3>

<p>If <code>graph = TRUE</code>, a graphic with the dispersion of the estimates for each sample size, 
as well as the graphic containing the number of points outside the confidence interval for 
the reference sample.
</p>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cv &lt;- function(x, i) sd(x[i]) / mean(x[i]) # coefficient of variation
x = rnorm(20, 15, 2)
cv(x)
samplesize(x, cv)

par(mfrow = c(1, 3), cex = 0.7, las = 1)
samplesize(x, cv, lcl = 0.05, ucl = 0.20)
abline(h = 0.05 * 500, col = "blue") # sample sizes with 5% (or less) out CI

# End (not run)
</code></pre>

<hr>
<h2 id='sHe'>
Spatial Analysis of Gene Diversity
</h2><span id='topic+sHe'></span>

<h3>Description</h3>

<p>Estimate spatial gene diversity (expected heterozygozity - <em>He</em>) through the individual-centred approach by Manel et al. (2007).
<code>sHe()</code> calculates the unbiased estimate of <em>He</em> based on the information of allele frequency obtained from codominant or
dominant markers in individuals within a circular moving windows of known radius over the sampling area.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sHe(x, coord.cols = 1:2, marker.cols = 3:4, 
	marker.type = c("codominant", "dominant"), 
	grid = NULL, latlong2km = TRUE, radius, nmin = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sHe_+3A_x">x</code></td>
<td>
<p> a data frame or numeric matrix containing columns with coordinates of individuals and marker genotyping</p>
</td></tr>
<tr><td><code id="sHe_+3A_coord.cols">coord.cols</code></td>
<td>
<p> a vector of integer giving the columns of coordinates in <code>x</code></p>
</td></tr>
<tr><td><code id="sHe_+3A_marker.cols">marker.cols</code></td>
<td>
<p> a vector of integer giving the columns of markers in <code>x</code></p>
</td></tr>
<tr><td><code id="sHe_+3A_marker.type">marker.type</code></td>
<td>
<p> a character; the type of molecular marker</p>
</td></tr>
<tr><td><code id="sHe_+3A_grid">grid</code></td>
<td>
<p> optional; a two-column matrix containing coordinates over which to predict <em>He</em></p>
</td></tr>
<tr><td><code id="sHe_+3A_latlong2km">latlong2km</code></td>
<td>
<p> logical; should coordinates be converted from lat/long format into kilometer-grid based?</p>
</td></tr>
<tr><td><code id="sHe_+3A_radius">radius</code></td>
<td>
<p> the radius of the moving window. It must be in the same format as sampling coordinates</p>
</td></tr>
<tr><td><code id="sHe_+3A_nmin">nmin</code></td>
<td>
<p> optional; a numeric value indicating the minimum number of individuals used to calculate <em>He</em>. If is
the number of individuals in a certain location is less then <code>nmin</code>, <code>sHe</code> will consider <em>He</em> as zero.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The unbiased estimate of expected heterogygozity (Nei, 1978) is given by:
</p>
<p style="text-align: center;"><code class="reqn"> He = (1 - \sum_{i=1}^{n} p_{i}^{2}) \frac{2n}{2n - 1} </code>
</p>

<p>where <code class="reqn">p_{i}</code> is the frequency of the i-th allele per locus considering the <code class="reqn">n</code> individuals in a certain location.
</p>


<h3>Value</h3>

<p>A list of
</p>
<table role = "presentation">
<tr><td><code>diversity</code></td>
<td>
<p> a data frame with the following columns: <em>coord.x</em> - the x-axis coordinates of the predicion grid,
<em>coord.y</em> - the y-axis coordinates of the predicion grid, <em>n</em> - the number of individuals in a certain points in the grid,
<em>MaxDist</em> - the maximum observed distance among these individuals, <em>uHe</em> - the unbiased estimate
of gene diversity (as expressed above), and <em>SE</em> - the standard error of <em>uHe</em>. </p>
</td></tr>
<tr><td><code>mHe</code></td>
<td>
<p> a matrix containing the estimates of <em>He</em> for every marker, on each point of the <code>grid</code>.</p>
</td></tr>
<tr><td><code>locations</code></td>
<td>
<p> a numeric matrix containing the sampling coordinates, as provides as input.</p>
</td></tr>
</table>


<h3>Warning </h3>

<p>Depending on the dimension of <code>x</code> and/or <code>grid</code>, <code>sHe()</code> can be time demanding.
</p>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>
<p>Ivandilson Pessoa Pinto de Menezes &lt;ivan.menezes@ifgoiano.edu.br&gt;
</p>


<h3>References</h3>

<p>da Silva, A.R.; Malafaia, G.; Menezes, I.P.P. (2017) biotools: an R function to predict 
spatial gene diversity via an individual-based approach. <em>Genetics and Molecular Research</em>, 
<b>16</b>: gmr16029655.
</p>
<p>Manel, S., Berthoud, F., Bellemain, E., Gaudeul, M., Luikart, G., Swenson, J.E., Waits, L.P., 
Taberlet, P.; Intrabiodiv Consortium. (2007) A new individual-based spatial approach for 
identifying genetic discontinuities in natural populations. <em>Molecular Ecology</em>, <b>16</b>:2031-2043.
</p>
<p>Nei, M. (1978) Estimation of average heterozygozity and genetic distance from a small number of individuals. 
<em>Genetics</em>, <b>89</b>: 583-590.
</p>


<h3>See Also</h3>

<p><code><a href="lattice.html#topic+levelplot">levelplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(moco)
data(brazil)

# check points
plot(brazil, cex = 0.1, col = "gray")
points(Lat ~ Lon, data = moco, col = "blue", pch = 20)

# using a retangular grid (not passed as input!)
# ex &lt;- sHe(x = moco, coord.cols = 1:2,
#	marker.cols = 3:20, marker.type = "codominant",
#	grid = NULL, radius = 150)
#ex
# plot(ex, xlab = "Lon", ylab = "Lat")

# A FANCIER PLOT...
# using Brazil's coordinates as prediction grid
# ex2 &lt;- sHe(x = moco, coord.cols = 1:2,
#	marker.cols = 3:20, marker.type = "codominant",
#	grid = brazil, radius = 150)
# ex2
#
# library(maps)
# borders &lt;- data.frame(x = map("world", "brazil")$x, 
#	y = map("world", "brazil")$y)
#
# library(latticeExtra)
# plot(ex2, xlab = "Lon", ylab = "Lat", 
#	xlim = c(-75, -30), ylim = c(-35, 10), aspect = "iso") +
#   latticeExtra::as.layer(xyplot(y ~ x, data = borders, type = "l")) +
#   latticeExtra::as.layer(xyplot(Lat ~ Lon, data = moco))

# End (not run)
</code></pre>

<hr>
<h2 id='singh'> Importance of Variables According to the Singh (1981) Criterion </h2><span id='topic+singh'></span><span id='topic+singh.default'></span><span id='topic+plot.singh'></span>

<h3>Description</h3>

<p>A function to calculate the Singh (1981) criterion for importance
of variables based on the squared generalized Mahalanobis distance.
</p>
<p style="text-align: center;"><code class="reqn">S_{.j} = \sum_{i=1}^{n-1} \sum_{i'&gt;i}^{n} (x_{ij} - x_{i'j}) * (\bold{x}_i - \bold{x}_{i'})' * \bold{\Sigma}_{j}^{-1} </code>
</p>



<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
singh(data, cov, inverted = FALSE)
## S3 method for class 'singh'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="singh_+3A_data">data</code></td>
<td>
<p> a data frame or matrix of data (<em>n x p</em>). </p>
</td></tr>
<tr><td><code id="singh_+3A_cov">cov</code></td>
<td>
<p> a variance-covariance matrix (<em>p x p</em>). </p>
</td></tr>
<tr><td><code id="singh_+3A_inverted">inverted</code></td>
<td>
<p> logical. If <code>FALSE</code> (default), <code>cov</code> is supposed to be a variance-covariance matrix. </p>
</td></tr>
<tr><td><code id="singh_+3A_x">x</code></td>
<td>
<p> an object of class <code>"singh"</code>. </p>
</td></tr>
<tr><td><code id="singh_+3A_...">...</code></td>
<td>
<p> further graphical arguments. </p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>singh</code> returns a matrix containing the Singh statistic, the
importance proportion and the cummulative proprtion of each
variable (column) in data.
</p>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>References</h3>

<p>Singh, D. (1981) The relative importance of characters affecting genetic
divergence. <em>Indian Journal Genetics &amp; Plant Breeding</em>, 41:237-245.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+D2.dist">D2.dist</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Manly (2004, p.65-66)
x1 &lt;- c(131.37, 132.37, 134.47, 135.50, 136.17)
x2 &lt;- c(133.60, 132.70, 133.80, 132.30, 130.33)
x3 &lt;- c(99.17, 99.07, 96.03, 94.53, 93.50)
x4 &lt;- c(50.53, 50.23, 50.57, 51.97, 51.37)
x &lt;- cbind(x1, x2, x3, x4)
Cov &lt;- matrix(c(21.112,0.038,0.078,2.01, 0.038,23.486,5.2,2.844,
	0.078,5.2,24.18,1.134, 2.01,2.844,1.134,10.154), 4, 4)
(s &lt;- singh(x, Cov))
plot(s)

# End (not run)
</code></pre>

<hr>
<h2 id='spatialpred'>
Spatial Predictions Based on the Circular Variable-Radius Moving Window Method
</h2><span id='topic+spatialpred'></span>

<h3>Description</h3>

<p>A heuristic method to perform spatial predictions. The method consists of a local interpolator with stochastic features. 
It allows to build effective detailed maps and to estimate the spatial dependence without any assumptions on the spatial process.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spatialpred(coords, data, grid)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="spatialpred_+3A_coords">coords</code></td>
<td>
<p> a data frame or numeric matrix containing columns with geographic coordinates</p>
</td></tr>
<tr><td><code id="spatialpred_+3A_data">data</code></td>
<td>
<p> a numeric vector of compatible dimension with <code>coords</code> containing the response variable data to be predicted at each <code>grid</code> point </p>
</td></tr>
<tr><td><code id="spatialpred_+3A_grid">grid</code></td>
<td>
<p> a data frame or numeric matrix containing columns with geographic coordinates where <code>data</code> is to be predicted </p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>grid</code> receives the same input as <code>coords</code>, <code>spatialpred</code> will calculate the Percenntual Absolute Mean Error (PAME) of predictions.
</p>


<h3>Value</h3>

<p>A data.frame containing spatial predictions, standard errors, the radius and the number of observations used in each prediction over the grid.
</p>


<h3>Warning </h3>

<p>Depending on the dimension of <code>coords</code> and/or <code>grid</code>, <code>spatialpred()</code> can be time demanding.
</p>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>References</h3>

<p>Da Silva, A.R., Silva, A.P.A., Tiago-Neto, L.J. (2020) A new local stochastic method for predicting data with spatial heterogeneity. 
<em>ACTA SCIENTIARUM-AGRONOMY</em>, <b>43</b>:e49947.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sHe">sHe</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># data(moco)
# p &lt;- spatialpred(coords = moco[, 1:2], data = rnorm(206), grid = moco[, 1:2]) 
# note: using coords as grid to calculate PAME
# head(p)
# lattice::levelplot(pred ~ Lat*Lon, data = p)

# End (not run)
</code></pre>

<hr>
<h2 id='tocher'> Tocher's Clustering </h2><span id='topic+tocher'></span><span id='topic+tocher.dist'></span><span id='topic+print.tocher'></span><span id='topic+cophenetic.tocher'></span><span id='topic+coph.tocher'></span>

<h3>Description</h3>

<p><code>tocher</code> performs the Tocher (Rao, 1952) optimization clustering from a distance matrix.
The cophenetic distance matrix for a Tocher's clustering can also be computed using the methodology proposed
by Silva \&amp; Dias (2013).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dist'
tocher(d, algorithm = c("original", "sequential"))
## S3 method for class 'tocher'
print(x, ...)
## S3 method for class 'tocher'
cophenetic(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tocher_+3A_d">d</code></td>
<td>
<p> an object of class <code>"dist"</code>.</p>
</td></tr>
<tr><td><code id="tocher_+3A_algorithm">algorithm</code></td>
<td>
<p> a character indicating the algorithm to be used for clustering objects.
It must be one of the two: <code>"original"</code> (default) or <code>"sequential"</code>. The latter is 
the method proposed by Vasconcelos et al. (2007), and sometimes called &quot;modified&quot; Tocher.</p>
</td></tr>
<tr><td><code id="tocher_+3A_x">x</code></td>
<td>
<p> an object of class <code>"tocher"</code>.</p>
</td></tr>
<tr><td><code id="tocher_+3A_...">...</code></td>
<td>
<p> optional further arguments from <code>print</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>tocher</code>. A list of 
</p>
<table role = "presentation">
<tr><td><code>call</code></td>
<td>
<p>the call which produced the result.</p>
</td></tr>
<tr><td><code>algorithm</code></td>
<td>
<p>character; the algorithm that has been used as input.</p>
</td></tr>
<tr><td><code>clusters</code></td>
<td>
<p>a list of length <em>k</em> (the number of clusters),
containing the labels of the objects in <code>d</code> for each cluster.</p>
</td></tr>
<tr><td><code>class</code></td>
<td>
<p>a numeric vector indicating the class (the cluster) of each object in <code>d</code>.</p>
</td></tr>
<tr><td><code>criterion</code></td>
<td>
<p>a numeric vector containing the clustering criteria - the greatest amongst 
the smallest distances involving each object in <code>d</code>. If <code>algorithm = "original"</code>,
this vector contains an unique value, i.e., the same criterion is used for every clustering step.</p>
</td></tr>
<tr><td><code>distClust</code></td>
<td>
<p>a matrix of distances within (diagonal) and between (off-diagonal) clusters.</p>
</td></tr>
<tr><td><code>d</code></td>
<td>
<p>the input object.</p>
</td></tr>
</table>


<h3>Warning </h3>

<p>Clustering a large number of objects (say 300 or more) can be time demanding.</p>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>


<h3>References</h3>

<p>Cruz, C.D.; Ferreira, F.M.; Pessoni, L.A. (2011) <em>Biometria aplicada ao estudo
da diversidade genetica</em>. Visconde do Rio Branco: Suprema.
</p>
<p>Rao, R.C. (1952) <em>Advanced statistical methods in biometric research</em>.
New York: John Wiley &amp; Sons.
</p>
<p>Sharma, J.R. (2006) <em>Statistical and biometrical techniques in plant breeding</em>.
Delhi: New Age International.
</p>
<p>Silva, A.R. &amp; Dias, C.T.S. (2013) A cophenetic correlation coefficient for
Tocher's method. <em>Pesquisa Agropecuaria Brasileira</em>, 48:589-596. 
</p>
<p>Vasconcelos, E.S.; Cruz, C.D.; Bhering, L.L.; Resende Junior, M.F.R. (2007) Alternative 
methodology for the cluster analysis. <em>Pesquisa Agropecuaria Brasileira</em>, 42:1421-1428.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+dist">dist</a></code>, <code><a href="#topic+D2.dist">D2.dist</a></code>, <code><a href="stats.html#topic+cophenetic">cophenetic</a></code>, <code><a href="#topic+distClust">distClust</a></code>, <code><a href="stats.html#topic+hclust">hclust</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># example 1
data(garlicdist)
(garlic &lt;- tocher(garlicdist))
garlic$distClust  # cluster distances

# example 2
data(USArrests)
(usa &lt;- tocher(dist(USArrests)))
usa$distClust

# cophenetic correlation
cophUS &lt;- cophenetic(usa)
cor(cophUS, dist(USArrests))

# using the sequential algorithm
(usa2 &lt;- tocher(dist(USArrests), algorithm = "sequential"))
usa2$criterion

# example 3
data(eurodist)
(euro &lt;- tocher(eurodist))
euro$distClust

# End (not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
