<!DOCTYPE html><html lang="en-US"><head><title>Help for package BioVizSeq</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BioVizSeq}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#biovizseq'><p>BioVizSeq shiny app start function.</p></a></li>
<li><a href='#cdd_plot'><p>cdd_plot</p></a></li>
<li><a href='#cdd_to_loc'><p>cdd_to_loc</p></a></li>
<li><a href='#combi_p'><p>combi_p</p></a></li>
<li><a href='#fastaleng'><p>fastaleng</p></a></li>
<li><a href='#geom_rrect'><p>geom_rrect</p></a></li>
<li><a href='#gff_plot'><p>gff_plot</p></a></li>
<li><a href='#gff_to_loc'><p>gff_to_loc</p></a></li>
<li><a href='#globalVariables'><p>Global Variables Declaration</p></a></li>
<li><a href='#meme_plot'><p>meme_plot</p></a></li>
<li><a href='#meme_seq'><p>meme_seq</p></a></li>
<li><a href='#meme_to_loc'><p>meme_to_loc</p></a></li>
<li><a href='#motif_plot'><p>motif_plot</p></a></li>
<li><a href='#pfam_plot'><p>pfam_plot</p></a></li>
<li><a href='#pfam_to_loc'><p>pfam_to_loc</p></a></li>
<li><a href='#plantcare_classify'><p>plantcare_classify</p></a></li>
<li><a href='#plantcare_plot'><p>plantcare_plot</p></a></li>
<li><a href='#plantcare_statistic1'><p>plantcare_statistic1</p></a></li>
<li><a href='#plantcare_statistic2'><p>plantcare_statistic2</p></a></li>
<li><a href='#plantcare_to_loc'><p>plantcare_to_loc</p></a></li>
<li><a href='#smart_plot'><p>smart_plot</p></a></li>
<li><a href='#smart_to_loc'><p>smart_to_loc</p></a></li>
<li><a href='#upload_fa_to_plantcare'><p>upload_fa_to_plantcare</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Visualizing the Elements Within Bio-Sequences</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Shiqi Zhao &lt;zhaosq89@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Visualizing the types and distribution of elements within
  bio-sequences. At the same time, We have developed a geom layer,
  geom_rrect(), that can generate rounded rectangles. No external references
  are used in the development of this package.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, ggplot2, httr, RColorBrewer, seqinr, stringr, tidyr,
magrittr, ggtree, treeio, shiny</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-20 08:09:35 UTC; zhaoshiqi</td>
</tr>
<tr>
<td>Author:</td>
<td>Shiqi Zhao <a href="https://orcid.org/0000-0002-2360-956X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre, cph]</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-20 08:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='biovizseq'>BioVizSeq shiny app start function.</h2><span id='topic+biovizseq'></span>

<h3>Description</h3>

<p>BioVizSeq shiny app start function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>biovizseq()
</code></pre>


<h3>Value</h3>

<p>Shinyapp: BioVizSeq shiny app.
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library BioVizSeq package
library(BioVizSeq)

</code></pre>

<hr>
<h2 id='cdd_plot'>cdd_plot</h2><span id='topic+cdd_plot'></span>

<h3>Description</h3>

<p>Visualization of domain in CDD file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cdd_plot(
  cdd_file,
  fasta_file,
  the_order = NULL,
  domain_select = NULL,
  shape = "RoundRect",
  r = 0.3,
  legend_size = 15,
  domain_color = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cdd_plot_+3A_cdd_file">cdd_file</code></td>
<td>
<p>The path of cdd file.</p>
</td></tr>
<tr><td><code id="cdd_plot_+3A_fasta_file">fasta_file</code></td>
<td>
<p>The path of fasta file.</p>
</td></tr>
<tr><td><code id="cdd_plot_+3A_the_order">the_order</code></td>
<td>
<p>The path of order file. A List of Gene ID , One ID Per Line.</p>
</td></tr>
<tr><td><code id="cdd_plot_+3A_domain_select">domain_select</code></td>
<td>
<p>The domain ID which you want to align with.</p>
</td></tr>
<tr><td><code id="cdd_plot_+3A_shape">shape</code></td>
<td>
<p>RoundRect or Rect.</p>
</td></tr>
<tr><td><code id="cdd_plot_+3A_r">r</code></td>
<td>
<p>The radius of rounded corners.</p>
</td></tr>
<tr><td><code id="cdd_plot_+3A_legend_size">legend_size</code></td>
<td>
<p>The size of legend.</p>
</td></tr>
<tr><td><code id="cdd_plot_+3A_domain_color">domain_color</code></td>
<td>
<p>The color set of domain.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>p
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>hitdata_path &lt;- system.file("extdata", "hitdata.txt", package = "BioVizSeq")
fa_path &lt;- system.file("extdata", "idpep.fa", package = "BioVizSeq")
cdd_plot(hitdata_path, fa_path)

order_path &lt;- system.file("extdata", "order.csv", package = "BioVizSeq")
cdd_plot(hitdata_path, fa_path, the_order = order_path)
</code></pre>

<hr>
<h2 id='cdd_to_loc'>cdd_to_loc</h2><span id='topic+cdd_to_loc'></span>

<h3>Description</h3>

<p>Extract the location information of domain from cdd file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cdd_to_loc(cdd_file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cdd_to_loc_+3A_cdd_file">cdd_file</code></td>
<td>
<p>CDD file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>hitdata_path &lt;- system.file("extdata", "hitdata.txt", package = "BioVizSeq")
cdd_file &lt;- readLines(hitdata_path)
domain_loc &lt;- cdd_to_loc(cdd_file)
</code></pre>

<hr>
<h2 id='combi_p'>combi_p</h2><span id='topic+combi_p'></span>

<h3>Description</h3>

<p>Get ggplot2 files to facilitate free combination in patchwork
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combi_p(
  tree_path,
  gff_path = NULL,
  meme_path = NULL,
  pfam_path = NULL,
  cdd_path = NULL,
  fa_path = NULL,
  smart_path = FALSE,
  plantcare_path = NULL,
  promoter_length = NULL,
  renamefile = NULL,
  groupfile = NULL,
  shape = "RoundRect",
  r = 0.3,
  legend_size = 6
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="combi_p_+3A_tree_path">tree_path</code></td>
<td>
<p>The path of tree file (.newick).</p>
</td></tr>
<tr><td><code id="combi_p_+3A_gff_path">gff_path</code></td>
<td>
<p>The path of .gff/gtf file.</p>
</td></tr>
<tr><td><code id="combi_p_+3A_meme_path">meme_path</code></td>
<td>
<p>The path of .meme/mast file.</p>
</td></tr>
<tr><td><code id="combi_p_+3A_pfam_path">pfam_path</code></td>
<td>
<p>The path of pfam result file (.tsv).</p>
</td></tr>
<tr><td><code id="combi_p_+3A_cdd_path">cdd_path</code></td>
<td>
<p>The path of cdd result file (.txt).</p>
</td></tr>
<tr><td><code id="combi_p_+3A_fa_path">fa_path</code></td>
<td>
<p>The path of protein file (.fa/fasta).</p>
</td></tr>
<tr><td><code id="combi_p_+3A_smart_path">smart_path</code></td>
<td>
<p>Do SMART or not. (TRUE or FALSE)</p>
</td></tr>
<tr><td><code id="combi_p_+3A_plantcare_path">plantcare_path</code></td>
<td>
<p>The path of plantcare file (.tab).</p>
</td></tr>
<tr><td><code id="combi_p_+3A_promoter_length">promoter_length</code></td>
<td>
<p>The length of promoter.</p>
</td></tr>
<tr><td><code id="combi_p_+3A_renamefile">renamefile</code></td>
<td>
<p>Rename file. Two cols: new_name and old_name.</p>
</td></tr>
<tr><td><code id="combi_p_+3A_groupfile">groupfile</code></td>
<td>
<p>Group information. Two cols: label and Group.</p>
</td></tr>
<tr><td><code id="combi_p_+3A_shape">shape</code></td>
<td>
<p>RoundRect or Rect.</p>
</td></tr>
<tr><td><code id="combi_p_+3A_r">r</code></td>
<td>
<p>The radius of rounded corners.</p>
</td></tr>
<tr><td><code id="combi_p_+3A_legend_size">legend_size</code></td>
<td>
<p>The size of legend.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tree_path &lt;- system.file("extdata", "idpep.nwk", package = "BioVizSeq")
plot_file &lt;- combi_p(tree_path)
</code></pre>

<hr>
<h2 id='fastaleng'>fastaleng</h2><span id='topic+fastaleng'></span>

<h3>Description</h3>

<p>Statistical sequence length
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fastaleng(fasta_file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fastaleng_+3A_fasta_file">fasta_file</code></td>
<td>
<p>The path of protein fasta file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fasta_path &lt;- system.file("extdata", "idpep.fa", package = "BioVizSeq") 
fastaleng(fasta_path)

</code></pre>

<hr>
<h2 id='geom_rrect'>geom_rrect</h2><span id='topic+geom_rrect'></span>

<h3>Description</h3>

<p>Rounded rectangle
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geom_rrect(
  mapping = NULL,
  data = NULL,
  stat = "identity",
  position = "identity",
  r = 0.2,
  ...,
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geom_rrect_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="geom_rrect_+3A_data">data</code></td>
<td>
<p>A data.frame, or other object, will override the plot data. All objects will be
fortified to produce a data frame.</p>
</td></tr>
<tr><td><code id="geom_rrect_+3A_stat">stat</code></td>
<td>
<p>Name of stat to modify data.</p>
</td></tr>
<tr><td><code id="geom_rrect_+3A_position">position</code></td>
<td>
<p>The position adjustment to use for overlapping points on this layer.</p>
</td></tr>
<tr><td><code id="geom_rrect_+3A_r">r</code></td>
<td>
<p>The radius of rounded corners.</p>
</td></tr>
<tr><td><code id="geom_rrect_+3A_...">...</code></td>
<td>
<p>additional parameter, e.g. color, linewidth, alpha.</p>
</td></tr>
<tr><td><code id="geom_rrect_+3A_na.rm">na.rm</code></td>
<td>
<p>If &quot;FALSE&quot; (default), missing values are removed with a warning. If &quot;TRUE&quot;, missing values are silently removed, logical.</p>
</td></tr>
<tr><td><code id="geom_rrect_+3A_show.legend">show.legend</code></td>
<td>
<p>Whether to show legend, logical.</p>
</td></tr>
<tr><td><code id="geom_rrect_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>Whether to inherit aesthetic mappings, logical, defaults to &quot;TRUE&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>draws rounded rectangle by using the locations
of the four corners (xmin, xmax, ymin and ymax) like geom_rect().
</p>


<h3>Value</h3>

<p>ggplot object
</p>


<h3>Aesthetics</h3>

<p><code>geom_rrect()</code> understands the following aesthetics (required aesthetics are in bold):
</p>

<ul>
<li> <p><strong><code><a href="ggplot2.html#topic+xmin">xmin</a></code></strong>
</p>
</li>
<li> <p><strong><code><a href="ggplot2.html#topic+xmax">xmax</a></code></strong>
</p>
</li>
<li> <p><strong><code><a href="ggplot2.html#topic+ymin">ymin</a></code></strong>
</p>
</li>
<li> <p><strong><code><a href="ggplot2.html#topic+ymax">ymax</a></code></strong>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+alpha">alpha</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+colour">colour</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+fill">fill</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+group">group</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+linetype">linetype</a></code>
</p>
</li></ul>

<p>Learn more about setting these aesthetics in <code>vignette("ggplot2-specs")</code>.
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)
df &lt;- data.frame(
 xmin = c(1, 2, 3),
 xmax = c(2, 3, 4),
 ymin = c(1, 2, 3),
 ymax = c(2, 3, 4),
 category = c("A", "B", "C")
 )
 
 p &lt;- ggplot(df) +
   geom_rrect(aes(xmin = xmin, xmax = xmax, 
              ymin = ymin, ymax = ymax, fill = category), 
             r = 0.4, linewidth = 1, colour = "black") 
 
 print(p)

</code></pre>

<hr>
<h2 id='gff_plot'>gff_plot</h2><span id='topic+gff_plot'></span>

<h3>Description</h3>

<p>Visualization of element in gff or gtf file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gff_plot(
  gff_file,
  the_order = NULL,
  shape = "Rect",
  r = 0.3,
  legend_size = 15,
  element_color = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gff_plot_+3A_gff_file">gff_file</code></td>
<td>
<p>The path of gff file.</p>
</td></tr>
<tr><td><code id="gff_plot_+3A_the_order">the_order</code></td>
<td>
<p>The path of order of mRNA. It is also the mRNA you want to showcase. A List of Gene ID , One ID Per Line.</p>
</td></tr>
<tr><td><code id="gff_plot_+3A_shape">shape</code></td>
<td>
<p>RoundRect or Rect.</p>
</td></tr>
<tr><td><code id="gff_plot_+3A_r">r</code></td>
<td>
<p>The radius of rounded corners.</p>
</td></tr>
<tr><td><code id="gff_plot_+3A_legend_size">legend_size</code></td>
<td>
<p>The size of legend.</p>
</td></tr>
<tr><td><code id="gff_plot_+3A_element_color">element_color</code></td>
<td>
<p>The color set of element.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>p
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gff_path &lt;- system.file("extdata", "test.gff", package = "BioVizSeq")
gff_plot(gff_path)
</code></pre>

<hr>
<h2 id='gff_to_loc'>gff_to_loc</h2><span id='topic+gff_to_loc'></span>

<h3>Description</h3>

<p>Extract the location information of element from gff or gtf file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gff_to_loc(gff_data, mRNA_ID = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gff_to_loc_+3A_gff_data">gff_data</code></td>
<td>
<p>gff file.</p>
</td></tr>
<tr><td><code id="gff_to_loc_+3A_mrna_id">mRNA_ID</code></td>
<td>
<p>The mRNA you selected. If NULL, it means selecting all mRNAs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
gff_path &lt;- system.file("extdata", "test.gff", package = "BioVizSeq")
gff_data &lt;- read.table(gff_path, header = FALSE, sep = '\t')
gff_loc &lt;- gff_to_loc(gff_data)

ID_path &lt;- system.file("extdata", "ID_select.csv", package = "BioVizSeq")
mRNA_ID &lt;- readLines(ID_path) 
gff_loc &lt;- gff_to_loc(gff_data, mRNA_ID=mRNA_ID)
</code></pre>

<hr>
<h2 id='globalVariables'>Global Variables Declaration</h2><span id='topic+globalVariables'></span>

<h3>Description</h3>

<p>Global Variables
</p>


<h3>Details</h3>

<p>This block declares global variables to avoid R CMD check warnings.
</p>

<hr>
<h2 id='meme_plot'>meme_plot</h2><span id='topic+meme_plot'></span>

<h3>Description</h3>

<p>Visualization of motif in meme file or mast file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meme_plot(
  meme_file,
  the_order = NULL,
  motif_select = NULL,
  shape = "RoundRect",
  show_motif_id = FALSE,
  r = 0.3,
  legend_size = 15,
  motif_color = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="meme_plot_+3A_meme_file">meme_file</code></td>
<td>
<p>The path of meme file or mast file.</p>
</td></tr>
<tr><td><code id="meme_plot_+3A_the_order">the_order</code></td>
<td>
<p>The path of order file. A List of Gene ID , One ID Per Line.</p>
</td></tr>
<tr><td><code id="meme_plot_+3A_motif_select">motif_select</code></td>
<td>
<p>The motif ID which you want to align with.</p>
</td></tr>
<tr><td><code id="meme_plot_+3A_shape">shape</code></td>
<td>
<p>RoundRect or Rect.</p>
</td></tr>
<tr><td><code id="meme_plot_+3A_show_motif_id">show_motif_id</code></td>
<td>
<p>Display the name of the motif.</p>
</td></tr>
<tr><td><code id="meme_plot_+3A_r">r</code></td>
<td>
<p>The radius of rounded corners.</p>
</td></tr>
<tr><td><code id="meme_plot_+3A_legend_size">legend_size</code></td>
<td>
<p>The size of legend.</p>
</td></tr>
<tr><td><code id="meme_plot_+3A_motif_color">motif_color</code></td>
<td>
<p>The color set of motif.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>p
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>meme_path &lt;- system.file("extdata", "meme.xml", package = "BioVizSeq")
meme_plot(meme_path)

mast_path &lt;- system.file("extdata", "mast.xml", package = "BioVizSeq")
meme_plot(mast_path)

meme_plot(meme_path, motif_select="1", show_motif_id = TRUE)

order_path &lt;- system.file("extdata", "order.csv", package = "BioVizSeq")
meme_plot(meme_path, the_order=order_path, motif_select="1")
</code></pre>

<hr>
<h2 id='meme_seq'>meme_seq</h2><span id='topic+meme_seq'></span>

<h3>Description</h3>

<p>Get motif sequence from meme file or mast file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meme_seq(meme_file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="meme_seq_+3A_meme_file">meme_file</code></td>
<td>
<p>The path of meme file or mast file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>meme_path &lt;- system.file("extdata", "meme.xml", package = "BioVizSeq")
meme_file &lt;- readLines(meme_path)
motifseq&lt;- meme_seq(meme_file)

mast_path &lt;- system.file("extdata", "mast.xml", package = "BioVizSeq")
mast_file &lt;- readLines(mast_path)
motifseq&lt;- meme_seq(mast_file)

</code></pre>

<hr>
<h2 id='meme_to_loc'>meme_to_loc</h2><span id='topic+meme_to_loc'></span>

<h3>Description</h3>

<p>Extract the location information of motif from mast or meme file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meme_to_loc(motif_file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="meme_to_loc_+3A_motif_file">motif_file</code></td>
<td>
<p>The motif data of mast or meme file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>meme_path &lt;- system.file("extdata", "meme.xml", package = "BioVizSeq") 
meme_file &lt;- readLines(meme_path)
motif_loc &lt;- meme_to_loc(meme_file)

mast_path &lt;- system.file("extdata", "mast.xml", package = "BioVizSeq") 
mast_file &lt;- readLines(mast_path)
motif_loc &lt;- meme_to_loc(mast_file)

</code></pre>

<hr>
<h2 id='motif_plot'>motif_plot</h2><span id='topic+motif_plot'></span>

<h3>Description</h3>

<p>Draws multiple rounded rectangle.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>motif_plot(
  motif_loc,
  gene_length,
  the_order = NULL,
  motif_select = NULL,
  shape = "RoundRect",
  show_motif_id = FALSE,
  r = 0.3,
  legend_size = 15,
  motif_color = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="motif_plot_+3A_motif_loc">motif_loc</code></td>
<td>
<p>A data.frame contains for columuns: ID, motif, start, end.</p>
</td></tr>
<tr><td><code id="motif_plot_+3A_gene_length">gene_length</code></td>
<td>
<p>A data.fram of the length of biosequences. Two columns: ID, length.</p>
</td></tr>
<tr><td><code id="motif_plot_+3A_the_order">the_order</code></td>
<td>
<p>A List of Gene ID , One ID Per Line.</p>
</td></tr>
<tr><td><code id="motif_plot_+3A_motif_select">motif_select</code></td>
<td>
<p>The motif ID which you want to align with.</p>
</td></tr>
<tr><td><code id="motif_plot_+3A_shape">shape</code></td>
<td>
<p>RoundRect or Rect.</p>
</td></tr>
<tr><td><code id="motif_plot_+3A_show_motif_id">show_motif_id</code></td>
<td>
<p>Display the name of the motif.</p>
</td></tr>
<tr><td><code id="motif_plot_+3A_r">r</code></td>
<td>
<p>The radius of rounded corners.</p>
</td></tr>
<tr><td><code id="motif_plot_+3A_legend_size">legend_size</code></td>
<td>
<p>The size of legend.</p>
</td></tr>
<tr><td><code id="motif_plot_+3A_motif_color">motif_color</code></td>
<td>
<p>The color set of motif.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>motif_plot() draws multiple rounded rectangle to represent the above elements of biosequences,
but not limited to biosequences
</p>


<h3>Value</h3>

<p>p
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df &lt;- data.frame(
 ID = rep(c("geneA", "geneB", "geneC"), each = 3),
 motif = rep(c("1", "2", "3"), times = 3),
 start = c(1, 3, 6, 1, 6, 10, 10, 7, 17),
 end = c(3, 5, 11, 3, 8, 15, 12, 9, 22)
 )
 
 length_data &lt;- data.frame(
 ID = c("geneA", "geneB", "geneC"),
 length = c(15, 27, 30)
 )
 
 order_data &lt;- c("geneB", "geneA", "geneC")
 
 motif_plot(df, length_data)
 motif_plot(df, length_data, the_order = order_data)

</code></pre>

<hr>
<h2 id='pfam_plot'>pfam_plot</h2><span id='topic+pfam_plot'></span>

<h3>Description</h3>

<p>Visualization of domain in pfam result file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pfam_plot(
  pfam_file,
  the_order = NULL,
  domain_select = NULL,
  shape = "RoundRect",
  r = 0.3,
  legend_size = 15,
  domain_color = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pfam_plot_+3A_pfam_file">pfam_file</code></td>
<td>
<p>The path of meme file or mast file.</p>
</td></tr>
<tr><td><code id="pfam_plot_+3A_the_order">the_order</code></td>
<td>
<p>The path of order file. A List of Gene ID , One ID Per Line.</p>
</td></tr>
<tr><td><code id="pfam_plot_+3A_domain_select">domain_select</code></td>
<td>
<p>The domain ID which you want to align with.</p>
</td></tr>
<tr><td><code id="pfam_plot_+3A_shape">shape</code></td>
<td>
<p>RoundRect or Rect.</p>
</td></tr>
<tr><td><code id="pfam_plot_+3A_r">r</code></td>
<td>
<p>The radius of rounded corners.</p>
</td></tr>
<tr><td><code id="pfam_plot_+3A_legend_size">legend_size</code></td>
<td>
<p>The size of legend.</p>
</td></tr>
<tr><td><code id="pfam_plot_+3A_domain_color">domain_color</code></td>
<td>
<p>The color set of domain.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>p
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pfam_path &lt;- system.file("extdata", "iprscan.tsv", package = "BioVizSeq")
order_path &lt;- system.file("extdata", "order.csv", package = "BioVizSeq")
pfam_plot(pfam_path)
pfam_plot(pfam_path, the_order=order_path)

</code></pre>

<hr>
<h2 id='pfam_to_loc'>pfam_to_loc</h2><span id='topic+pfam_to_loc'></span>

<h3>Description</h3>

<p>Extract the location information of domain from pfam result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pfam_to_loc(pfam_data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pfam_to_loc_+3A_pfam_data">pfam_data</code></td>
<td>
<p>The result file (.tsv) of pfam (via InterPro).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pfam_path &lt;- system.file("extdata", "iprscan.tsv", package = "BioVizSeq")
pfam_file &lt;- read.table(pfam_path, sep='\t', header = FALSE)
motif_loc &lt;- pfam_to_loc(pfam_file)
</code></pre>

<hr>
<h2 id='plantcare_classify'>plantcare_classify</h2><span id='topic+plantcare_classify'></span>

<h3>Description</h3>

<p>Classify the functions of cis element from Plantcare
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plantcare_classify(plantcare_file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plantcare_classify_+3A_plantcare_file">plantcare_file</code></td>
<td>
<p>The result file (.tab) of Plantcare.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plantcare_path &lt;- system.file("extdata", "plantCARE_output.tab", package = "BioVizSeq") 
plantcare_file &lt;- read.table(plantcare_path, header = FALSE, sep = '\t', quote="")
plantcare_data &lt;- plantcare_classify(plantcare_file)
</code></pre>

<hr>
<h2 id='plantcare_plot'>plantcare_plot</h2><span id='topic+plantcare_plot'></span>

<h3>Description</h3>

<p>Visualization of cis-element in plantcare result file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plantcare_plot(
  plantcare_file,
  promoter_length = 2000,
  the_order = NULL,
  shape = "Rect",
  r = 6,
  legend_size = 15,
  element_color = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plantcare_plot_+3A_plantcare_file">plantcare_file</code></td>
<td>
<p>The path of plantcare result file (.tab).</p>
</td></tr>
<tr><td><code id="plantcare_plot_+3A_promoter_length">promoter_length</code></td>
<td>
<p>The promoter length.</p>
</td></tr>
<tr><td><code id="plantcare_plot_+3A_the_order">the_order</code></td>
<td>
<p>The path of order file. A List of Gene ID , One ID Per Line.</p>
</td></tr>
<tr><td><code id="plantcare_plot_+3A_shape">shape</code></td>
<td>
<p>RoundRect or Rect.</p>
</td></tr>
<tr><td><code id="plantcare_plot_+3A_r">r</code></td>
<td>
<p>The radius of rounded corners.</p>
</td></tr>
<tr><td><code id="plantcare_plot_+3A_legend_size">legend_size</code></td>
<td>
<p>The size of legend.</p>
</td></tr>
<tr><td><code id="plantcare_plot_+3A_element_color">element_color</code></td>
<td>
<p>The color set of cis-element.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>p
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plantcare_path &lt;- system.file("extdata", "plantCARE_output.tab", package = "BioVizSeq") 
plantcare_plot(plantcare_path, promoter_length = 2000)
</code></pre>

<hr>
<h2 id='plantcare_statistic1'>plantcare_statistic1</h2><span id='topic+plantcare_statistic1'></span>

<h3>Description</h3>

<p>Count the number of cis element from Plantcare for heatmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plantcare_statistic1(plantcare_data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plantcare_statistic1_+3A_plantcare_data">plantcare_data</code></td>
<td>
<p>The result of plantcare_classify().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plantcare_path &lt;- system.file("extdata", "plantCARE_output.tab", package = "BioVizSeq") 
plantcare_file &lt;- read.table(plantcare_path, header = FALSE, sep = '\t', quote="")
plantcare_data &lt;- plantcare_classify(plantcare_file)
statistic_data1 &lt;- plantcare_statistic1(plantcare_data)

</code></pre>

<hr>
<h2 id='plantcare_statistic2'>plantcare_statistic2</h2><span id='topic+plantcare_statistic2'></span>

<h3>Description</h3>

<p>Count the number of cis element from Plantcare for Bar chart
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plantcare_statistic2(plantcare_data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plantcare_statistic2_+3A_plantcare_data">plantcare_data</code></td>
<td>
<p>The result of plantcare_classify().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plantcare_path &lt;- system.file("extdata", "plantCARE_output.tab", package = "BioVizSeq") 
plantcare_file &lt;- read.table(plantcare_path, header = FALSE, sep = '\t', quote="")
plantcare_data &lt;- plantcare_classify(plantcare_file)
statistic_data2 &lt;- plantcare_statistic2(plantcare_data)
</code></pre>

<hr>
<h2 id='plantcare_to_loc'>plantcare_to_loc</h2><span id='topic+plantcare_to_loc'></span>

<h3>Description</h3>

<p>Extract the location information of cis-element from Plantcare
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plantcare_to_loc(plantcare_data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plantcare_to_loc_+3A_plantcare_data">plantcare_data</code></td>
<td>
<p>The result of plantcare_classify().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plantcare_path &lt;- system.file("extdata", "plantCARE_output.tab", package = "BioVizSeq") 
plantcare_file &lt;- read.table(plantcare_path, header = FALSE, sep = '\t', quote="")
plantcare_data &lt;- plantcare_classify(plantcare_file)
plantcare_loc &lt;- plantcare_to_loc(plantcare_data)
</code></pre>

<hr>
<h2 id='smart_plot'>smart_plot</h2><span id='topic+smart_plot'></span>

<h3>Description</h3>

<p>Visualization of domain in SMART result file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>smart_plot(
  fasta_file,
  the_order = NULL,
  domain_select = NULL,
  shape = "RoundRect",
  r = 0.3,
  legend_size = 15,
  domain_color = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="smart_plot_+3A_fasta_file">fasta_file</code></td>
<td>
<p>The path of protein fasta file.</p>
</td></tr>
<tr><td><code id="smart_plot_+3A_the_order">the_order</code></td>
<td>
<p>The path of order file. A List of Gene ID , One ID Per Line.</p>
</td></tr>
<tr><td><code id="smart_plot_+3A_domain_select">domain_select</code></td>
<td>
<p>The domain ID which you want to align with.</p>
</td></tr>
<tr><td><code id="smart_plot_+3A_shape">shape</code></td>
<td>
<p>RoundRect or Rect.</p>
</td></tr>
<tr><td><code id="smart_plot_+3A_r">r</code></td>
<td>
<p>The radius of rounded corners.</p>
</td></tr>
<tr><td><code id="smart_plot_+3A_legend_size">legend_size</code></td>
<td>
<p>The size of legend.</p>
</td></tr>
<tr><td><code id="smart_plot_+3A_domain_color">domain_color</code></td>
<td>
<p>The color set of domain.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>p
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>

<hr>
<h2 id='smart_to_loc'>smart_to_loc</h2><span id='topic+smart_to_loc'></span>

<h3>Description</h3>

<p>Extract the location information of domain from SMART result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>smart_to_loc(input_file, do_pfam = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="smart_to_loc_+3A_input_file">input_file</code></td>
<td>
<p>The path of potein fasta file.</p>
</td></tr>
<tr><td><code id="smart_to_loc_+3A_do_pfam">do_pfam</code></td>
<td>
<p>Include the pfam domain or not.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>

<hr>
<h2 id='upload_fa_to_plantcare'>upload_fa_to_plantcare</h2><span id='topic+upload_fa_to_plantcare'></span>

<h3>Description</h3>

<p>Upload the promoter file to Plantcare database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>upload_fa_to_plantcare(fasta_file, email)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="upload_fa_to_plantcare_+3A_fasta_file">fasta_file</code></td>
<td>
<p>The path of promoter file.</p>
</td></tr>
<tr><td><code id="upload_fa_to_plantcare_+3A_email">email</code></td>
<td>
<p>e-mail address.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Due to the file size limitation of plantcare on fasta,
upload_fa_to_plantcare() first splits fasta file.
Then uploads the splited fasta files to the plantcare database,
and automatically returns the results to the email provided by the user.
</p>
<p>upload_fa_to_plantcare(&quot;the path/test.fasta&quot;, &quot;your e-mail address&quot;)
</p>


<h3>Value</h3>

<p>plantcare_result
</p>


<h3>Author(s)</h3>

<p>Shiqi Zhao
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
