<!DOCTYPE html><html><head><title>Help for package BiSEp</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BiSEp}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BiSEp-package'>
<p>BiSEp: Bimodality in gene expression to dissect tumours and reveal synthetic lethal drug targets and biomarkers</p></a></li>
<li><a href='#BEEM'><p>BEEM: Bimodal Expression Exclusive with Mutation</p></a></li>
<li><a href='#BIGEE'><p>BIGEE: Bimodal Gene Expression Exclusivity.</p></a></li>
<li><a href='#BISEP'><p>BISEP: Bimodality in Gene Expression data.</p></a></li>
<li><a href='#BISEP_dat'>
<p>A list object containing 3 data frames.</p></a></li>
<li><a href='#BISEP_data'>
<p>A list object containing 3 data frames.</p></a></li>
<li><a href='#expressionPlot'><p>expressionPlot: Create visualisations from BIGEE output</p></a></li>
<li><a href='#FURE'><p>FURE: Functional redundancy between synthetic lethal gene pairs</p></a></li>
<li><a href='#FURE_data'>
<p>Output matrix from the BIGEE tool</p></a></li>
<li><a href='#INPUT_data'>
<p>A Log2 Gene Expression matrix</p></a></li>
<li><a href='#MUT_data'>
<p>A matrix containing discreet mutation calls</p></a></li>
<li><a href='#waterfallPlot'><p>waterfallPlot: Create visualisations from BEEM output</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Toolkit to Identify Candidate Synthetic Lethality</td>
</tr>
<tr>
<td>Version:</td>
<td>2.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-05-04</td>
</tr>
<tr>
<td>Author:</td>
<td>Mark Wappett</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mark Wappett &lt;m.wappett@qub.ac.uk&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>AnnotationDbi, mclust, GO.db, org.Hs.eg.db, GOSemSim</td>
</tr>
<tr>
<td>Description:</td>
<td>Enables the user to infer potential synthetic lethal relationships
    by analysing relationships between bimodally distributed gene pairs in big
    gene expression datasets.  Enables the user to visualise these candidate
    synthetic lethal relationships.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-05-03 11:12:38 UTC; 3055680</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-05-04 23:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='BiSEp-package'>
BiSEp: Bimodality in gene expression to dissect tumours and reveal synthetic lethal drug targets and biomarkers
</h2><span id='topic+BiSEp-package'></span><span id='topic+BiSEp'></span>

<h3>Description</h3>

<p>A set of tools that enable the user to accurately identify bimodality and non-normality in gene expression data and stratify samples as high or low expression for bimodal genes.  Enables identification of candidate synthetic lethal gene pairs. Enables the user to assess and visualise functional redundancy between candidate synthetic lethal gene pairs.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> BiSEp</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2014-10-21</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>This package has a mixture of CRAN and bioconductor packages listed as dependancies.  Please ensure that you have Bioconductor installed.
</p>


<h3>Author(s)</h3>

<p>Author: Mark Wappett
</p>
<p>Maintainer: Mark Wappett &lt;mark.wappett@astrazeneca.com&gt;
</p>

<hr>
<h2 id='BEEM'>BEEM: Bimodal Expression Exclusive with Mutation</h2><span id='topic+BEEM'></span>

<h3>Description</h3>

<p>Takes the output from the function BISEP and a discreet mutation matrix as input.  The mutation matrix samples (columns) must mirror or overlap with the gene expression matrix.  The data in the mutation matrix must be a discreet 'WT' or 'MUT' call based on the status of each gene with each sample. Detects mutations of genes enriched in either the high or low gene expression modes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BEEM(
	bisepData=data, 
	mutData=mutData, 
	sampleType=c("cell_line", "cell_line_low", "patient", "patient_low"), 
	minMut=10
	)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BEEM_+3A_bisepdata">bisepData</code></td>
<td>
<p>This should be the output from the BISEP function.
</p>
</td></tr>
<tr><td><code id="BEEM_+3A_mutdata">mutData</code></td>
<td>
<p>This should be a matrix with genes rownames and samples as column names.  All cells should be made up of a discreet 'WT' or 'MUT' call.  There should be overlap (by sample) with the gene expression matrix.
</p>
</td></tr>
<tr><td><code id="BEEM_+3A_sampletype">sampleType</code></td>
<td>
<p>The type of sample being analysed.  Select 'cell_line' or 'patient' for datasets with greater than ~200 samples.  For datasets with less than ~200 samples, use 'cell_line_low' or 'patient_low'.
</p>
</td></tr>
<tr><td><code id="BEEM_+3A_minmut">minMut</code></td>
<td>
<p>The minimum number of mutations you for a gene would consider for analysis.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Lower sample numbers have more stringent bimodality hurdles to clear in order to keep the false positive rate lower.  The tool returns a percentage complete text window so the user can observe the status of the job. 
</p>


<h3>Value</h3>

<p>A matrix containing 10 columns. Column 1 contains the bimodal genes from the expression data (gene 1) and column 2 contains the mutated candidate synthetic lethal gene pair (gene 2).  Columns 3 and 4 contain the number of mutations of gene 2 in the low and high expression modes of gene 1.  Column 5 contains the fishers p value that evaluates enrichment of mutation in either the high or low mode (indicated by column 10).  Columns 6 and 7 contain the percentage of samples in the low and high expression modes of gene 1 that are mutated for gene 2.  Columns 8 and 9 contain information on the overall size (in terms of sample) of the low and high expression modes of gene 1. 
</p>


<h3>Author(s)</h3>

<p>Mark Wappett
</p>

<hr>
<h2 id='BIGEE'>BIGEE: Bimodal Gene Expression Exclusivity.
</h2><span id='topic+BIGEE'></span>

<h3>Description</h3>

<p>Part of the Synthetic Lethality detection in Genomics toolkit.  Detects bimodality and non-normality in all genes across the dataset.  Compares all pairwise combinations of bimodal genes and searches for mutually exclusive low expression as evidence of potential synthetic lethality.  Scores gene-pairs based on the presence of mutual exclusive bimodality and the distribution of signal intensity across the rest of the dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BIGEE(
	bisepData=data, 
	sampleType=c("cell_line", "cell_line_low", "patient", "patient_low")
	)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BIGEE_+3A_bisepdata">bisepData</code></td>
<td>
<p>This should be the output from the BISEP function.
</p>
</td></tr>
<tr><td><code id="BIGEE_+3A_sampletype">sampleType</code></td>
<td>
<p>The type of sample being analysed.  Select 'cell_line' or 'patient' for datasets with greater than ~200 samples.  For datasets with less than ~200 samples, use 'cell_line_low' or 'patient_low'.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Lower sample numbers have more stringent bimodality hurdles to clear in order to keep the false positive rate lower.  The tool returns a percentage complete text window so the user can observe the status of the job. 
</p>


<h3>Value</h3>

<p>A matrix containing three columns. Columns 1 and 2 are the gene symbols that make up the candidate synthetic lethal gene pairs.  Column 3 is the score calculated the tool to rank the statistical significance of the gene pairs.
</p>


<h3>Author(s)</h3>

<p>Mark Wappett
</p>

<hr>
<h2 id='BISEP'>BISEP: Bimodality in Gene Expression data.
</h2><span id='topic+BISEP'></span>

<h3>Description</h3>

<p>Detects bimodality and non-normality in all genes across the dataset.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BISEP(
	data = data
	)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BISEP_+3A_data">data</code></td>
<td>
<p>This should be a log2 gene expression matrix with genes as rownames and samples as column names.  Suitable for gene expression data from any platform - NGS datasets should be RPKM or RSEM values.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The lower confidence calls will dramatically affect the number of gene pairs that the tool produces and increase the false positive rate.  The tool will take approximately 10 minutes to run a 5,000 row and 200 column input matrix using a 'medium,' confidence interval.
</p>


<h3>Value</h3>

<p>A list containing three matrices.  Matrix 1 contains the output of the BISEP algorithm - including the midpoint of the bimodal distribution and the associated p value. Matrix 2 contains the output from the BI algorithm - including the delta, pi and BI values.  Matrix 3 contains the input matrix.
</p>


<h3>Author(s)</h3>

<p>Mark Wappett
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(INPUT_data)
outputBISEP &lt;- BISEP(data=INPUT_data)
</code></pre>

<hr>
<h2 id='BISEP_dat'>
A list object containing 3 data frames.
</h2><span id='topic+BISEP_dat'></span>

<h3>Description</h3>

<p>Matrix 1 contains the output of the BISEP algorithm - including the midpoint of the bimodal distribution and the associated p value. Matrix 2 contains the output from the BI algorithm - including the delta, pi and BI values.  Matrix 3 contains the input matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(BISEP_dat)</code></pre>


<h3>Format</h3>

<p>13 observations across 100 variables. 
</p>

<hr>
<h2 id='BISEP_data'>
A list object containing 3 data frames.
</h2><span id='topic+BISEP_data'></span>

<h3>Description</h3>

<p>Matrix 1 contains the output of the BISEP algorithm - including the midpoint of the bimodal distribution and the associated p value. Matrix 2 contains the output from the BI algorithm - including the delta, pi and BI values.  Matrix 3 contains the input matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(BISEP_data)</code></pre>


<h3>Format</h3>

<p>13 observations across 442 variables. 
</p>

<hr>
<h2 id='expressionPlot'>expressionPlot: Create visualisations from BIGEE output</h2><span id='topic+expressionPlot'></span>

<h3>Description</h3>

<p>Takes the output from the function BISEP and two gene names that correspond to a relevant gene pair.  Gene names must be available in the input BISEP object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expressionPlot(
	bisepData=data,
	gene1,
	gene2
	)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="expressionPlot_+3A_bisepdata">bisepData</code></td>
<td>
<p>This should be the output from the BISEP function.
</p>
</td></tr>
<tr><td><code id="expressionPlot_+3A_gene1">gene1</code></td>
<td>
<p>The first gene whose expression you would like to plot.
</p>
</td></tr>
<tr><td><code id="expressionPlot_+3A_gene2">gene2</code></td>
<td>
<p>The second gene whose expression you would like to plot.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function will return an error if any of the input information is incorrect or missing.  The resulting plot will be returned in real time.
</p>


<h3>Value</h3>

<p>A scatter plot of the two genes you have identified as bimodal.  The red lines correspond to the mid-points of the bimodal distribution for these two genes.  Ideally the lower left quadrant would be empty when observing a candidate SL interaction.
</p>


<h3>Author(s)</h3>

<p>Mark Wappett
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BISEP_data)
data(MUT_data)
expressionOut &lt;- expressionPlot(BISEP_data, gene1="SMARCA1", gene2="SMARCA4")
</code></pre>

<hr>
<h2 id='FURE'>FURE: Functional redundancy between synthetic lethal gene pairs</h2><span id='topic+FURE'></span>

<h3>Description</h3>

<p>Utilises gene ontology information from the GO database bioconductor package.  Assesses gene pairs output from the SLinG and BEEM tools for gene ontology functional redundancy.  Performs semantic similarity scoring utilising the GOSemSim bioconductor package</p>


<h3>Usage</h3>

<pre><code class='language-R'>FURE(
	data=data, 
	inputType=inputType)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FURE_+3A_data">data</code></td>
<td>
<p>This should be the output matrix (or similar) from the SLinG and BEEM tools.  Columns 1 and 2 should be gene symbols.
</p>
</td></tr>
<tr><td><code id="FURE_+3A_inputtype">inputType</code></td>
<td>
<p>Either 'BIGEE' or 'BEEM' based on origin of the input matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of matrices containing gene pairs with associated synthetic lethal statistical significance values + gene ontology annotation/ scores. 
</p>


<h3>Author(s)</h3>

<p>Mark Wappett
</p>

<hr>
<h2 id='FURE_data'>
Output matrix from the BIGEE tool
</h2><span id='topic+FURE_data'></span>

<h3>Description</h3>

<p>Output matrix from the BIGEE tool.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(FURE_data)</code></pre>


<h3>Format</h3>

<p>A data frame with 1 observation across 3 variables.
</p>

<hr>
<h2 id='INPUT_data'>
A Log2 Gene Expression matrix
</h2><span id='topic+INPUT_data'></span>

<h3>Description</h3>

<p>A Log2 Gene Expression matrix where rownames are genes and colnames are samples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(INPUT_data)</code></pre>


<h3>Format</h3>

<p>13 observations across 442 variables. 
</p>

<hr>
<h2 id='MUT_data'>
A matrix containing discreet mutation calls
</h2><span id='topic+MUT_data'></span>

<h3>Description</h3>

<p>A matrix containing discreet mutation calls of either 'WT' or 'MUT' where rownames are genes and column names are samples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(MUT_data)</code></pre>


<h3>Format</h3>

<p>4 observations across 442 variables. 
</p>

<hr>
<h2 id='waterfallPlot'>waterfallPlot: Create visualisations from BEEM output</h2><span id='topic+waterfallPlot'></span>

<h3>Description</h3>

<p>Takes the output from the function BISEP and a discreet mutation matrix as input.  The mutation matrix samples (columns) must mirror or overlap with the gene expression matrix.  The data in the mutation matrix must be a discreet 'WT' or 'MUT' call based on the status of each gene with each sample. Gene names must be available in the input matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>waterfallPlot(
	bisepData=data, 
	mutData=mutData, 
	expressionGene, 
	mutationGene
	)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="waterfallPlot_+3A_bisepdata">bisepData</code></td>
<td>
<p>This should be the output from the BISEP function.
</p>
</td></tr>
<tr><td><code id="waterfallPlot_+3A_mutdata">mutData</code></td>
<td>
<p>This should be a matrix with genes rownames and samples as column names.  All cells should be made up of a discreet 'WT' or 'MUT' call.  There should be overlap (by sample) with the gene expression matrix.
</p>
</td></tr>
<tr><td><code id="waterfallPlot_+3A_expressiongene">expressionGene</code></td>
<td>
<p>The gene whose expression you would like to plot.
</p>
</td></tr>
<tr><td><code id="waterfallPlot_+3A_mutationgene">mutationGene</code></td>
<td>
<p>The gene whose mutation status you would like to overlap with the expression gene.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function will return an error if any of the input information is incorrect or missing.  The resulting plot will be returned in real time.
</p>


<h3>Value</h3>

<p>A waterfall plot.  The plot is made up of two panels: the left panel is a density distribution of the expression gene provided and the right panel is a bar-chart of the gene expression level coloured by mutation status.
</p>


<h3>Author(s)</h3>

<p>Mark Wappett
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BISEP_data)
data(MUT_data)
waterfallOut &lt;- waterfallPlot(BISEP_data, MUT_data, expressionGene="micb", mutationGene="PBRM1")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
