<!DOCTYPE html><html><head><title>Help for package BisRNA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BisRNA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BisRNA-package'><p>Analysis of RNA Cytosine-5 Methylation from bisulfite sequencing experiments</p></a></li>
<li><a href='#Bisdata'><p>Three synthetic samples of RNA bisulfite sequencing for examples and tests</p></a></li>
<li><a href='#Bisdata1'><p>Synthetic sample 1 of RNA bisulfite sequencing</p></a></li>
<li><a href='#Bisdata2'><p>Synthetic sample 2 of RNA bisulfite sequencing</p></a></li>
<li><a href='#Bisdata3'><p>Synthetic sample 3 of RNA bisulfite sequencing</p></a></li>
<li><a href='#class.BisXP'><p>Cast bisulfite experiment data into a BisXP object</p></a></li>
<li><a href='#fisher.method'><p>Function implementing Fisher's method to combine independent p-values</p></a></li>
<li><a href='#intersectMatrix'><p>Take intersection of 2 tables</p></a></li>
<li><a href='#read.BisXP'><p>Read RNA bisulfite experiment data and cast it into a BisXP object</p></a></li>
<li><a href='#RNAmeth.poisson.par'><p>Find ratio (Poisson parameter / coverage) for one sample of bisulfite-converted RNA</p></a></li>
<li><a href='#RNAmeth.poisson.test'><p>Test RNA Cytosine methylation based on Poisson model distribution</p></a></li>
<li><a href='#samples.combine'><p>Combine samples p-values and ratios</p></a></li>
<li><a href='#testMeth'><p>Apply poisson.test to BS coverage and non-conversion ratio.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Analysis of RNA Cytosine-5 Methylation</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.2</td>
</tr>
<tr>
<td>Author:</td>
<td>C Legrand</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Carine Legrand &lt;c.legrand@dkfz.de&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Bisulfite-treated RNA non-conversion in a set of samples is
  analysed as follows : each sample's non-conversion distribution is
  identified to a Poisson distribution. P-values adjusted for multiple
  testing are calculated in each sample. Combined non-conversion P-values
  and standard errors are calculated on the intersection of the set of
  samples. For further details, see C Legrand, F Tuorto, M Hartmann, 
  R Liebers, D Jakob, M Helm and F Lyko (2017) &lt;<a href="https://doi.org/10.1101%2Fgr.210666.116">doi:10.1101/gr.210666.116</a>&gt;.</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-12-15</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.2)</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS, graphics, stats, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>IHW, testthat</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-12-15 12:47:58 UTC; legrand</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-12-15 13:33:54 UTC</td>
</tr>
</table>
<hr>
<h2 id='BisRNA-package'>Analysis of RNA Cytosine-5 Methylation from bisulfite sequencing experiments</h2><span id='topic+BisRNA'></span><span id='topic+BisRNA-package'></span>

<h3>Description</h3>

<p>BisRNA analyses non-conversion of bisulfite-treated RNA
of a set of samples, in order to distinguish actual 
cytosine-5 methylation marks from artifacts. 
Calculations are done as follows:
</p>

<ul>
<li><p> each sample's ratio (Poisson parameter / coverage) is determined, assuming a null Poisson distribution (<code><a href="#topic+RNAmeth.poisson.par">RNAmeth.poisson.par</a></code>),
</p>
</li>
<li><p> non-conversion p-values are calculated for each sample (<code><a href="#topic+RNAmeth.poisson.test">RNAmeth.poisson.test</a></code>),
</p>
</li>
<li><p> combined non-conversion p-values and standard error of the non-conversion ratios
are calculated on the intersection of the set of samples <code><a href="#topic+samples.combine">samples.combine</a></code>.
</p>
</li></ul>

<p>A small non-conversion p-value points to methylation of a RNA cytosine, or
another event blocking bisulfite conversion.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 ## Load data
 data(Bisdata,package="BisRNA")

 ## Ratio (Poisson parameter / coverage), derived p-values, 
 #  adjusted for multiple testing using either BH (here), or IHW if available.
 lambda1 &lt;- RNAmeth.poisson.par(Bisdata1)$estimate
 BisXP1  &lt;- RNAmeth.poisson.test(Bisdata1,lambda1,method="BH")
 #
 lambda2 &lt;- RNAmeth.poisson.par(Bisdata2)$estimate
 BisXP2  &lt;- RNAmeth.poisson.test(Bisdata2,lambda2,method="BH")
 #
 lambda3 &lt;- RNAmeth.poisson.par(Bisdata3)$estimate
 BisXP3  &lt;- RNAmeth.poisson.test(Bisdata3,lambda3,method="BH")

 ## Combined p-values ; median and standard error of
 ## bisulfite non-conversion ratio
 BisXP.combined &lt;- samples.combine(BisXP1,BisXP2,BisXP3)
</code></pre>

<hr>
<h2 id='Bisdata'>Three synthetic samples of RNA bisulfite sequencing for examples and tests</h2><span id='topic+Bisdata'></span>

<h3>Description</h3>

<p>A dataset containing the RNA names, C position, coverage and 
non-conversion ratio for three synthetic samples of a RNA
bisulfite sequencing experiment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Bisdata)
</code></pre>


<h3>Format</h3>

<p>Three data frames with 330 to 345 rows and the 4 following variables:
</p>

<dl>
<dt>RNA</dt><dd><p>RNA name or identifier</p>
</dd>
<dt>Cpos</dt><dd><p>Cytosine position in RNA</p>
</dd>
<dt>coverage</dt><dd><p>Number of available reads at Cpos</p>
</dd>
<dt>ncratio</dt><dd><p>Bisulfite non-conversion ratio at Cpos</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>data(Bisdata)
ls(pattern="Bisdata")
head(Bisdata1)

</code></pre>

<hr>
<h2 id='Bisdata1'>Synthetic sample 1 of RNA bisulfite sequencing</h2><span id='topic+Bisdata1'></span>

<h3>Description</h3>

<p>A dataset containing the RNA names, C position, coverage and 
non-conversion ratio for synthetic sample 1 of a RNA
bisulfite sequencing experiment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Bisdata)
</code></pre>


<h3>Format</h3>

<p>Data frame with 330 to 345 rows and the 4 following variables:
</p>

<dl>
<dt>RNA</dt><dd><p>RNA name or identifier</p>
</dd>
<dt>Cpos</dt><dd><p>Cytosine position in RNA</p>
</dd>
<dt>coverage</dt><dd><p>Number of available reads at Cpos</p>
</dd>
<dt>ncratio</dt><dd><p>Bisulfite non-conversion ratio at Cpos</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>data(Bisdata)
ls(pattern="Bisdata")
head(Bisdata1)

</code></pre>

<hr>
<h2 id='Bisdata2'>Synthetic sample 2 of RNA bisulfite sequencing</h2><span id='topic+Bisdata2'></span>

<h3>Description</h3>

<p>A dataset containing the RNA names, C position, coverage and 
non-conversion ratio for synthetic sample 2 of a RNA
bisulfite sequencing experiment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Bisdata)
</code></pre>


<h3>Format</h3>

<p>Data frame with 330 to 345 rows and the 4 following variables:
</p>

<dl>
<dt>RNA</dt><dd><p>RNA name or identifier</p>
</dd>
<dt>Cpos</dt><dd><p>Cytosine position in RNA</p>
</dd>
<dt>coverage</dt><dd><p>Number of available reads at Cpos</p>
</dd>
<dt>ncratio</dt><dd><p>Bisulfite non-conversion ratio at Cpos</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>data(Bisdata)
ls(pattern="Bisdata")
head(Bisdata2)

</code></pre>

<hr>
<h2 id='Bisdata3'>Synthetic sample 3 of RNA bisulfite sequencing</h2><span id='topic+Bisdata3'></span>

<h3>Description</h3>

<p>A dataset containing the RNA names, C position, coverage and 
non-conversion ratio for synthetic sample 3 of a RNA
bisulfite sequencing experiment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Bisdata)
</code></pre>


<h3>Format</h3>

<p>Data frame with 330 to 345 rows and the 4 following variables:
</p>

<dl>
<dt>RNA</dt><dd><p>RNA name or identifier</p>
</dd>
<dt>Cpos</dt><dd><p>Cytosine position in RNA</p>
</dd>
<dt>coverage</dt><dd><p>Number of available reads at Cpos</p>
</dd>
<dt>ncratio</dt><dd><p>Bisulfite non-conversion ratio at Cpos</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>data(Bisdata)
ls(pattern="Bisdata")
head(Bisdata3)

</code></pre>

<hr>
<h2 id='class.BisXP'>Cast bisulfite experiment data into a BisXP object</h2><span id='topic+class.BisXP'></span>

<h3>Description</h3>

<p><code>class.BisXP</code> creates a BisXP object from a table containing RNA name,
C position, bisulfite non-conversion ratio and adjusted p-value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>class.BisXP(BisData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="class.BisXP_+3A_bisdata">BisData</code></td>
<td>
<p>A data frame with 4 columns:
</p>

<ul>
<li><p> RNA name
</p>
</li>
<li><p> C position (integer, in [1,+Inf])
</p>
</li>
<li><p> bisulfite non-conversion ratio      (numeric, in [0,1])
</p>
</li>
<li><p> pvalue.adj (numeric, in [0,1])</p>
</li></ul>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes bisulfite experiment data as an input table, performs checks, and casts the table
into a BisXP object, with rows labelled after a RNA_C.position pattern.
</p>


<h3>Value</h3>

<p>If the input table is correct, then the output will be a BisXP object
corresponding to a consolidated data frame with rows labelled after a
RNA_C.position pattern.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> RNA     &lt;- c("NM_00001","NM_00001","NM_00002")
 Cpos     &lt;- as.integer(c(1,5,1))
 ncratio  &lt;- c(0.1,0.5,0.3)
 pv.adj   &lt;- c(0.001,0.1,0.3)
 BSdata   &lt;- data.frame(RNA, Cpos, ncratio, pv.adj, stringsAsFactors = FALSE)
 bsXP     &lt;- class.BisXP(BSdata)
</code></pre>

<hr>
<h2 id='fisher.method'>Function implementing Fisher's method to combine independent p-values</h2><span id='topic+fisher.method'></span>

<h3>Description</h3>

<p><code>fisher.method</code> takes a list of independent p-values and combines them
using Fisher's method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fisher.method(pvalues)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fisher.method_+3A_pvalues">pvalues</code></td>
<td>
<p>A list of p-values</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a list of p-values as input, determines the number of
degrees of freedom (2 * number of p-values), combines the p-values using
Fisher's method and returns the combined p-value in output.
</p>


<h3>Value</h3>

<p>The p-value combined using Fisher's method.
</p>


<h3>References</h3>

<p>Fisher RA (1925) Statistical Methods for Research Workers. Edinburg: Oliver and Boyd.
</p>
<p>Fisher RA (1948) Questions and Answers #14. In: Mosteller F, Fisher RA (1948) The American Statistician, 2:30-31 <a href="http://www.jstor.org/stable/2681650">http://www.jstor.org/stable/2681650</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>list.pv    &lt;- c(0.0001,0.0142,0.0150)
p.combined &lt;- fisher.method(list.pv)

</code></pre>

<hr>
<h2 id='intersectMatrix'>Take intersection of 2 tables</h2><span id='topic+intersectMatrix'></span>

<h3>Description</h3>

<p><code>intersectMatrix</code> takes 2 matrices and outputs their
intersection based on common row.names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intersectMatrix(Tab1, Tab2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="intersectMatrix_+3A_tab1">Tab1</code></td>
<td>
<p>A matrix or data frame with defined row.names</p>
</td></tr>
<tr><td><code id="intersectMatrix_+3A_tab2">Tab2</code></td>
<td>
<p>A matrix or data frame with defined row.names</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes 2 matrices as input, determines the intersection of
their row names, and returns a single matrix containing the rows in the
intersection and concatenated columns of the initial matrices.
</p>


<h3>Value</h3>

<p>A matrix with rows common to both Tab1 and Tab2,
and concatenated columns.
</p>

<hr>
<h2 id='read.BisXP'>Read RNA bisulfite experiment data and cast it into a BisXP object</h2><span id='topic+read.BisXP'></span>

<h3>Description</h3>

<p><code>read.BisXP</code> reads a table containing RNA name, C position,
bisulfite non-conversion ratio and adjusted p-value from a file
and casts it into a BisXP object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.BisXP(filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.BisXP_+3A_filename">filename</code></td>
<td>
<p>Address of the file containing data from a bisulfite experiment in 4 columns separated by a tabulation, with header on the first line:
</p>

<ul>
<li><p> RNA name
</p>
</li>
<li><p> C position (integer, in [1,+Inf])
</p>
</li>
<li><p> bisulfite non-conversion ratio      (numeric, in [0,1])
</p>
</li>
<li><p> pvalue.adj (numeric, in [0,1])</p>
</li></ul>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a file name as input, reads the bisulfite data table
contained in this file, performs checks, and casts the data into a BisXP object,
which contain input data with rows labelled after a RNA_C.position pattern.
</p>


<h3>Value</h3>

<p>If the input table is correct, then the output will be a BisXP object
corresponding to a consolidated data frame with rows labelled after a
RNA_C.position pattern.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> RNAs     &lt;- c("NM_00001","NM_00001","NM_00002")
 Cpos     &lt;- c(1,5,1)
 ncratio  &lt;- c(0.1,0.5,0.3)
 pv.adj   &lt;- c(0.001,0.1,0.3)
 BSdata &lt;- data.frame(RNAs, Cpos, ncratio, pv.adj)
</code></pre>

<hr>
<h2 id='RNAmeth.poisson.par'>Find ratio (Poisson parameter / coverage) for one sample of bisulfite-converted RNA</h2><span id='topic+RNAmeth.poisson.par'></span>

<h3>Description</h3>

<p><code>RNAmeth.poisson.par</code> determines the ratio (Poisson parameter / coverage),
based on the assumption that bisulfite-treated RNA's non-conversion ratios
follow a Poisson distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RNAmeth.poisson.par(BSrna)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RNAmeth.poisson.par_+3A_bsrna">BSrna</code></td>
<td>
<p>A dataframe containing RNA name, C position, coverage and
non-conversion ratio, in this order.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes RNA bisulfite sequencing data from one sample as input, restricts the dataset to RNAs for which coverage is at least 10,
and divides data into coverage bins. Then, the Poisson parameter is estimated at each coverage bin using fitdistr from package MASS.
The ratio (Poisson parameter / coverage) (median and 95% confidence interval) is returned in output.
</p>


<h3>Value</h3>

<p>This function returns a list containing:
</p>

<dl>
<dt>estimate</dt><dd><p>The ratio (Poisson parameter / coverage) (median taken over coverage bins)</p>
</dd>
<dt>bca.ci</dt><dd><p>A confidence interval for the ratio (Poisson parameter / coverage) (bootstrap
confidence interval of the median, type &quot;bca&quot;)</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>
 ## Load data, find out ratio (Poisson parameter / coverage).
 data(Bisdata,package="BisRNA")
 lambda1 &lt;- RNAmeth.poisson.par(Bisdata1)$estimate
 
 ## P-values adjusted for multiple testing, using either BH (here) or IHW method.
 BisXP1  &lt;- RNAmeth.poisson.test(Bisdata1,lambda1,method="BH")

 ## Print BisXP1
 BisXP1

 ## Display BisXP1 as a data frame
 BisXP1.df &lt;- data.frame(BisXP1$nonconv.ratio, BisXP1$pv.adj, row.names=BisXP1$RNA.pos)
 BisXP1.df

</code></pre>

<hr>
<h2 id='RNAmeth.poisson.test'>Test RNA Cytosine methylation based on Poisson model distribution</h2><span id='topic+RNAmeth.poisson.test'></span>

<h3>Description</h3>

<p><code>RNAmeth.poisson.test</code> tests RNA bisulfite sequencing non-conversion
based on a sample and the ratio (Poisson parameter / coverage) for this sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RNAmeth.poisson.test(BisRNA, lambda, method="BH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RNAmeth.poisson.test_+3A_bisrna">BisRNA</code></td>
<td>
<p>    A dataframe containing RNA name, C position, coverage and
non-conversion ratio</p>
</td></tr>
<tr><td><code id="RNAmeth.poisson.test_+3A_lambda">lambda</code></td>
<td>
<p>    Ratio (Poisson parameter / coverage)</p>
</td></tr>
<tr><td><code id="RNAmeth.poisson.test_+3A_method">method</code></td>
<td>
<p>    Adjustment method for multiple testing, either
&quot;BH&quot; (Benjamini-Hochberg) or &quot;IHW&quot; (Independent
Hypothesis Weighting, from R package IHW)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes RNA bisulfite sequencing data from one sample
and the ratio (Poisson parameter / coverage) as inputs. Then, the
dataset is restricted to those C positions where non-conversion 
ratio is larger than (Poisson parameter / coverage). 
Finally, it carries out poisson.test and a correction for multiple 
testing. The output of function RNAmeth.poisson.test is a BisXP 
object which contains RNA names, C positions, non-conversion ratios 
and adjusted p-values. The formatting into a BisXP object guarantees 
that some consistency checks are satisfied.
</p>


<h3>Value</h3>

<p>This function returns a BisXP object whose elements correspond to
RNAs where non-conversion ratio is higher than 
lambda =  ratio (Poisson parameter / coverage).
The variables contained in this object are the 
non-conversion ratio and the adjusted p-value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 ## Load data, find out ratio Poison parameter / coverage,
 ## and produce corresponding adjusted p-values.
 data(Bisdata,package="BisRNA")
 lambda1 &lt;- RNAmeth.poisson.par(Bisdata1)$estimate
 BisXP1  &lt;- RNAmeth.poisson.test(Bisdata1,lambda1,method="BH")

 ## Print BisXP1
 BisXP1

 ## Display BisXP1 as a data frame
 BisXP1.df &lt;- data.frame(BisXP1$nonconv.ratio, BisXP1$pv.adj, row.names=BisXP1$RNA.pos)
 BisXP1.df

</code></pre>

<hr>
<h2 id='samples.combine'>Combine samples p-values and ratios</h2><span id='topic+samples.combine'></span>

<h3>Description</h3>

<p><code>samples.combine</code> takes RNA bisulfite sequencing samples as input,
combines their p-values using Fisher's method, and produces median and
standard error of the bisulfite non-conversion ratios.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>samples.combine(BisXP1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="samples.combine_+3A_bisxp1">BisXP1</code></td>
<td>
<p>    A BisXP object containing non-conversion ratio and p-value</p>
</td></tr>
<tr><td><code id="samples.combine_+3A_...">...</code></td>
<td>
<p>       One or more additional samples, in the form of BisXP objects</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes several bisulfite sequencing samples, in form of BisXP
objects, as inputs. It is recommended to provide at least 3 samples and
in any case all available, relevant samples. Using RNA and C positions
present in all samples, the adjusted p-values of each sample are combined
using Fisher's method. Median and standard error of the non-conversion
ratio are also given in output.
</p>


<h3>Value</h3>

<p>This function returns a data frame whose row names correspond to the
RNA and C position which are present in all samples, and the following
variables:
</p>

<dl>
<dt>p.adj.combined</dt><dd><p>p-value adjusted (done in the preparation of
the BisXP object) and combined (done here)</p>
</dd>
<dt>nonconv.ratio.med</dt><dd><p>Median of bisulfite non-conversion ratio for a
specific RNA and C positions</p>
</dd>
<dt>nonconv.ratio.se</dt><dd><p>Standard error of bisulfite non-conversion ratio
for a specific RNA and C positions</p>
</dd>
</dl>



<h3>References</h3>

<p>Fisher RA (1925) Statistical Methods for Research Workers. Edinburg: Oliver and Boyd.
</p>
<p>Fisher RA (1948) Questions and Answers #14. In: Mosteller F, Fisher RA (1948) The American Statistician, 2:30-31
<a href="http://www.jstor.org/stable/2681650">http://www.jstor.org/stable/2681650</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 ## Load data
 data(Bisdata,package="BisRNA")

 ## Obtain the ratio (Poisson parameter / coverage), and p-values
 ## adjusted for multiple testing using BH (here) or IHW method.
 lambda1 &lt;- RNAmeth.poisson.par(Bisdata1)$estimate
 BisXP1  &lt;- RNAmeth.poisson.test(Bisdata1,lambda1,method="BH")
 lambda2 &lt;- RNAmeth.poisson.par(Bisdata2)$estimate
 BisXP2  &lt;- RNAmeth.poisson.test(Bisdata2,lambda2,method="BH")
 lambda3 &lt;- RNAmeth.poisson.par(Bisdata3)$estimate
 BisXP3  &lt;- RNAmeth.poisson.test(Bisdata3,lambda3,method="BH")

 ## Combine samples
 BisXP.combined &lt;- samples.combine(BisXP1,BisXP2,BisXP3)

</code></pre>

<hr>
<h2 id='testMeth'>Apply poisson.test to BS coverage and non-conversion ratio.</h2><span id='topic+testMeth'></span>

<h3>Description</h3>

<p><code>testMeth</code> applies poisson.test to a sample from bisulfite-treated RNA experiment taking into account the ratio (Poisson parameter / coverage) characteristic of this sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>testMeth(X, lambda)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="testMeth_+3A_x">X</code></td>
<td>
<p>A list containing coverage as 1st element and non-conversion
ratio as 2nd element, for one C position.</p>
</td></tr>
<tr><td><code id="testMeth_+3A_lambda">lambda</code></td>
<td>
<p>ratio (Poisson parameter / coverage)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes RNA bisulfite sequencing coverage and non-conversion
ratio, applies poisson.test and outputs the p-value.
</p>


<h3>Value</h3>

<p>This function returns the p-value from poisson.test.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
