<!DOCTYPE html><html><head><title>Help for package BiTrinA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BiTrinA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BASCResult-class'><p>Class &quot;BASCResult&quot;</p></a></li>
<li><a href='#binarizationExample'>
<p>An artificial data set consisting of ten artificial feature vectors.</p></a></li>
<li><a href='#BinarizationResult-class'><p>Class &quot;BinarizationResult&quot;</p></a></li>
<li><a href='#binarize.BASC'>
<p>Binarization Across Multiple Scales</p></a></li>
<li><a href='#binarize.kMeans'>
<p>k-means Binarization</p></a></li>
<li><a href='#binarizeMatrix'>
<p>Utility function to binarize a matrix of measurements</p></a></li>
<li><a href='#plot,BinarizationResult-method'>
<p>Visualization of binarization results.</p></a></li>
<li><a href='#plot,TrinarizationResult-method'>
<p>Visualization of trinarization results.</p></a></li>
<li><a href='#plotStepFunctions'>
<p>Plot all step functions for BASC or TASC</p></a></li>
<li><a href='#TASC'>
<p>Trinarization Across Multiple Scales</p></a></li>
<li><a href='#TASCResult-class'><p>Class &quot;TASCResult&quot;</p></a></li>
<li><a href='#trinarizationExample'>
<p>An artificial data set consisting of ten artificial feature vectors.</p></a></li>
<li><a href='#TrinarizationResult-class'><p>Class &quot;TrinarizationResult&quot;</p></a></li>
<li><a href='#trinarize.kMeans'>
<p>k-means Trinarization</p></a></li>
<li><a href='#trinarizeMatrix'>
<p>Utility function to trinarize a matrix of measurements</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Binarization and Trinarization of One-Dimensional Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-10-02</td>
</tr>
<tr>
<td>Author:</td>
<td>Stefan Mundus, Christoph Müssel, Florian Schmid, Ludwig Lausser, Tamara J. Blätte, Martin Hopfensitz, Hans A. Kestler</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hans Kestler &lt;hans.kestler@uni-ulm.de&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides methods for the binarization and trinarization of one-dimensional data and some visualization functions.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Imports:</td>
<td>graphics, stats</td>
</tr>
<tr>
<td>Depends:</td>
<td>methods, diptest</td>
</tr>
<tr>
<td>Suggests:</td>
<td>BoolNet</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-02 12:58:39 UTC; julian_schwab</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-02 13:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='BASCResult-class'>Class &quot;BASCResult&quot;</h2><span id='topic+BASCResult-class'></span><span id='topic+plotStepFunctions+2CBASCResult-method'></span><span id='topic+print+2CBASCResult-method'></span><span id='topic+show+2CBASCResult-method'></span>

<h3>Description</h3>

<p>A specialized class storing the results of a call to <code><a href="#topic+binarize.BASC">binarize.BASC</a></code>.
</p>


<h3>Objects of this class</h3>

<p>Objects of this class shouldn't be created directly. They are created implicitly by a call to <code><a href="#topic+binarize.BASC">binarize.BASC</a></code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>p.value</code>:</dt><dd><p>The p-value of the statistical test for reliability of the binarization.</p>
</dd>
<dt><code>intermediateSteps</code>:</dt><dd><p>A matrix specifying the optimal step functions from which the binarization was calculated. The number of rows corresponds to the number of step functions, and the number of columns is determined by the length of the input vector minus 2 (that is, the length of the step function corresponding to the input vector). From the first to the last row, the number of steps increases. The non-zero entries of the matrix represent the locations of the steps. Step functions with fewer steps than the input step function have entries set to zero.</p>
</dd>
<dt><code>intermediateHeights</code>:</dt><dd><p>A matrix giving the jump heights of the steps supplied in <code>intermediateSteps</code>.</p>
</dd>
<dt><code>intermediateStrongestSteps</code>:</dt><dd><p>A vector with one entry for each step function (row) in <code>intermediateSteps</code>. The entries specify the location of the strongest step for each of the functions.</p>
</dd>
<dt><code>originalMeasurements</code>:</dt><dd><p>A numeric vector storing the input measurements.</p>
</dd>
<dt><code>binarizedMeasurements</code>:</dt><dd><p>An integer vector of binarized values (0 or 1) corresponding to the original measurements.</p>
</dd>
<dt><code>threshold</code>:</dt><dd><p>The threshold that separates 0 and 1.</p>
</dd>
<dt><code>method</code>:</dt><dd><p>A string describing the binarization method that yielded the result.</p>
</dd>
</dl>



<h3>Extends</h3>

<p>Class <code>"<a href="#topic+BinarizationResult-class">BinarizationResult</a>"</code>, directly.
</p>


<h3>Methods</h3>


<dl>
<dt><a href="#topic+plotStepFunctions">plotStepFunctions</a></dt><dd><p><code>signature(x = "BASCResult")</code>: Plot the intermediate optimal step functions used to determine the threshold.</p>
</dd>
<dt>print</dt><dd><p><code>signature(x = "BASCResult")</code>: Print a summary of the binarization.</p>
</dd>
<dt>show</dt><dd><p><code>signature(object = "BASCResult")</code>: ... </p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+binarize.BASC">binarize.BASC</a></code>, 
<code><a href="#topic+BinarizationResult-class">BinarizationResult</a></code>
</p>

<hr>
<h2 id='binarizationExample'>
An artificial data set consisting of ten artificial feature vectors.
</h2><span id='topic+binarizationExample'></span>

<h3>Description</h3>

<p>An artificial data set consisting of ten artificial feature vectors that are used to illustrate the binarization methods in the package vignette. Each row of the matrix <code>binarizationExample</code> corresponds to one feature vector, of which 10 measurements are drawn from a normal distribution N(0,1). The remaining 10 measurements are drawn from a normal distribution N(m,1), with <code>m=10:1</code> decreasing from the first to the last row.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(binarizationExample)</code></pre>


<h3>Format</h3>

<p>The data is a matrix with 20 columns and 10 rows.
</p>

<hr>
<h2 id='BinarizationResult-class'>Class &quot;BinarizationResult&quot;</h2><span id='topic+BinarizationResult-class'></span><span id='topic+plotBinarization+2CBinarizationResult-method'></span><span id='topic+print+2CBinarizationResult-method'></span><span id='topic+show+2CBinarizationResult-method'></span>

<h3>Description</h3>

<p>This is the base class for objects that store the results of a binarization algorithm. It defines the slots and methods that the results of all algorithms share.
</p>


<h3>Objects of this class</h3>

<p>Objects of this class shouldn't be created directly. They are created implicitly by a call to one of the binarizeation algorithms.
</p>


<h3>Slots</h3>


<dl>
<dt><code>originalMeasurements</code>:</dt><dd><p>A numeric vector storing the input measurements.</p>
</dd>
<dt><code>binarizedMeasurements</code>:</dt><dd><p>An integer vector of binarized values (0 or 1) corresponding to the original measurements.</p>
</dd>
<dt><code>threshold</code>:</dt><dd><p>The threshold that separates 0 and 1.</p>
</dd>
<dt><code>method</code>:</dt><dd><p>A string describing the binarization method that yielded the result.</p>
</dd>
<dt><code>p.value</code>:</dt><dd><p>The p-value obtained by a test for validity of the binarization (e.g. BASC bootstrap test, Hartigan's dip test for k-means binarization, scan statistic p-value for best window. If no test was performed, this is <code>NA</code>.</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt><a href="#topic+plot+2CBinarizationResult-method">plot</a></dt><dd><p><code>signature(x = "BinarizationResult")</code>: Plot the binarization and the threshold.</p>
</dd>
<dt>print</dt><dd><p><code>signature(x = "BinarizationResult")</code>: Print a summary of the binarization.</p>
</dd>
<dt>show</dt><dd><p><code>signature(object = "BinarizationResult")</code>: ... </p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+binarize.BASC">binarize.BASC</a></code>, 
<code><a href="#topic+binarize.kMeans">binarize.kMeans</a></code>, 
<code><a href="#topic+BASCResult-class">BASCResult</a></code>, 
</p>

<hr>
<h2 id='binarize.BASC'>
Binarization Across Multiple Scales
</h2><span id='topic+binarize.BASC'></span>

<h3>Description</h3>

<p>Binarizes real-valued data using the multiscale BASC methods.</p>


<h3>Usage</h3>

<pre><code class='language-R'>binarize.BASC(vect, 
              method = c("A","B"), 
              tau = 0.01, 
              numberOfSamples = 999, 
              sigma = seq(0.1, 20, by=.1),
              na.rm=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binarize.BASC_+3A_method">method</code></td>
<td>

<p>Chooses the BASC method to use (see details), i.e. either  &quot;A&quot; or &quot;B&quot;. 
</p>
</td></tr>
<tr><td><code id="binarize.BASC_+3A_vect">vect</code></td>
<td>

<p>A real-valued vector of data to binarize.
</p>
</td></tr>
<tr><td><code id="binarize.BASC_+3A_tau">tau</code></td>
<td>

<p>This parameter adjusts the sensitivity and the specificity of the statistical testing procedure that rates the quality of the binarization. Defaults to 0.01.
</p>
</td></tr>
<tr><td><code id="binarize.BASC_+3A_numberofsamples">numberOfSamples</code></td>
<td>

<p>The number of samples for the bootstrap test. Defaults to 999.
</p>
</td></tr>
<tr><td><code id="binarize.BASC_+3A_sigma">sigma</code></td>
<td>

<p>If <code>method="B"</code>, this specifies a vector of different sigma values for the convolutions with the Bessel function. Ignored for <code>method="A"</code>.
</p>
</td></tr>
<tr><td><code id="binarize.BASC_+3A_na.rm">na.rm</code></td>
<td>

<p>If set to <code>TRUE</code>, <code>NA</code> values are removed from the input. Otherwise, binarization will fail in the presence of <code>NA</code> values.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The two BASC methods can be subdivided into three steps:
</p>
	
<dl>
<dt>Compute a series of step functions:</dt><dd><p>An initial step function is obtained by rearranging the original time series measurements in increasing order. Then, step functions with fewer discontinuities are calculated. BASC A calculates these step functions in such a way that each minimizes the Euclidean distance to the initial step function. BASC B obtains step functions from smoothened versions of the input function in a scale-space manner.</p>
</dd>
<dt>Find strongest discontinuity in each step function:</dt><dd><p>A strong discontinuity is a high jump size (derivative) in combination with a low approximation error.</p>
</dd>
<dt>Estimate location and variation of the strongest discontinuities:</dt><dd><p>Based on these estimates, data values can be excluded from further analyses.
</p>
</dd>
</dl>



<h3>Value</h3>

<p>Returns an object of class <code><a href="#topic+BASCResult-class">BASCResult</a></code>.
</p>


<h3>References</h3>

<p>M. Hopfensitz, C. Müssel, C. Wawra, M. Maucher, M. Kuehl, H. Neumann, and H. A. Kestler. Multiscale Binarization of Gene Expression Data for Reconstructing Boolean Networks. IEEE/ACM Transactions on Computational Biology and Bioinformatics 9(2):487-498, 2012.).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BinarizationResult-class">BinarizationResult</a></code>, 
<code><a href="#topic+BASCResult-class">BASCResult</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>par(mfrow=c(2,1))
result &lt;- binarize.BASC(iris[,"Petal.Length"], method="A", tau=0.15)
print(result)
plot(result)

result &lt;- binarize.BASC(iris[,"Petal.Length"], method="B", tau=0.15)
print(result)
plot(result)
</code></pre>

<hr>
<h2 id='binarize.kMeans'>
k-means Binarization
</h2><span id='topic+binarize.kMeans'></span>

<h3>Description</h3>

<p>Binarizes a vector of real-valued data using the k-means clustering algorithm. The data is first split into 2 clusters.The values belonging to the cluster with the smaller centroid are set to 0, and the values belonging to the greater centroid are set to 1.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binarize.kMeans(vect, 
                nstart=1, 
                iter.max=10,
                dip.test=TRUE,
                na.rm=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binarize.kMeans_+3A_vect">vect</code></td>
<td>

<p>A real-valued vector to be binarized (at least 3 measurements).
</p>
</td></tr>
<tr><td><code id="binarize.kMeans_+3A_nstart">nstart</code></td>
<td>

<p>The number of restarts for k-means. See <code><a href="stats.html#topic+kmeans">kmeans</a></code> for details.</p>
</td></tr>
<tr><td><code id="binarize.kMeans_+3A_iter.max">iter.max</code></td>
<td>

<p>The maximum number of iterations for k-means. See <code><a href="stats.html#topic+kmeans">kmeans</a></code> for details.</p>
</td></tr>
<tr><td><code id="binarize.kMeans_+3A_dip.test">dip.test</code></td>
<td>
<p>If set to <code>TRUE</code>, Hartigan's dip test for unimodality is performed on <code>vect</code>, and its p-value is returned in the <code>pvalue</code> slot of the result. An insignificant test indicates that the data may not be binarizeable.</p>
</td></tr>
<tr><td><code id="binarize.kMeans_+3A_na.rm">na.rm</code></td>
<td>

<p>If set to <code>TRUE</code>, <code>NA</code> values are removed from the input. Otherwise, binarization will fail in the presence of <code>NA</code> values.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class <code><a href="#topic+BinarizationResult-class">BinarizationResult</a></code>. 
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+kmeans">kmeans</a></code>, 
<code><a href="#topic+BinarizationResult-class">BinarizationResult</a></code>, 
<code><a href="BoolNet.html#topic+binarizeTimeSeries">BoolNet::binarizeTimeSeries()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>result &lt;- binarize.kMeans(iris[,"Petal.Length"])

print(result)
plot(result, twoDimensional=TRUE)
</code></pre>

<hr>
<h2 id='binarizeMatrix'>
Utility function to binarize a matrix of measurements
</h2><span id='topic+binarizeMatrix'></span>

<h3>Description</h3>

<p>Binarizes a matrix of measurements all at once, and returns the binarized vectors as well as the binarization thresholds and the p-values. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binarizeMatrix(mat, 
               method = c("BASCA", "BASCB", "kMeans"), 
               adjustment = "none", 
               ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binarizeMatrix_+3A_mat">mat</code></td>
<td>

<p>A n x m matrix comprising m raw measurements of n features.
</p>
</td></tr>
<tr><td><code id="binarizeMatrix_+3A_method">method</code></td>
<td>

<p>The binarization algorithm to be used. <code>method="BASCA"</code> calls <code><a href="#topic+binarize.BASC">binarize.BASC</a></code> with <code>method="A"</code>. <code>method="BASCB"</code> calls <code><a href="#topic+binarize.BASC">binarize.BASC</a></code> with <code>method="B"</code>. <code>method="kMeans"</code> calls <code><a href="#topic+binarize.kMeans">binarize.kMeans</a></code>.
</p>
</td></tr>
<tr><td><code id="binarizeMatrix_+3A_adjustment">adjustment</code></td>
<td>

<p>Specifies an optional adjustment for multiple testing that is applied to the p-values (see <code><a href="stats.html#topic+p.adjust">p.adjust</a></code> for possible values). By default, no adjustment is applied.
</p>
</td></tr>
<tr><td><code id="binarizeMatrix_+3A_...">...</code></td>
<td>

<p>Further parameters that are passed to the respective binarization methods (<code><a href="#topic+binarize.BASC">binarize.BASC</a></code>  or <code>method="kMeans"</code>).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A n x (m+2) matrix of binarized measurements. Here, the first m columns correspond to the binarized measurements. The m+1-st column comprises the binarization thresholds for the features, and the m+2-nd column contains the p-values.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+binarize.BASC">binarize.BASC</a></code>, <code><a href="#topic+binarize.kMeans">binarize.kMeans</a></code>, <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bin &lt;- binarizeMatrix(t(iris[,1:4]))
print(bin)
</code></pre>

<hr>
<h2 id='plot+2CBinarizationResult-method'>
Visualization of binarization results.
</h2><span id='topic+plot+2CBinarizationResult-method'></span><span id='topic+plot+2CBinarizationResult+2CANY-method'></span><span id='topic+plot+2Cnumeric+2CBinarizationResult-method'></span>

<h3>Description</h3>

<p>Visualizes a binarization as a ray or a two-dimensional plot. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'BinarizationResult,ANY'
plot(x,
     twoDimensional=FALSE, 
     showLegend=TRUE, 
     showThreshold=TRUE, 
     ...)
## S4 method for signature 'numeric,BinarizationResult'
plot(x, 
                                            y, 
                                            showLegend=TRUE, 
                                            showThreshold=TRUE, 
                                            ...)     
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2CBinarizationResult-method_+3A_x">x</code></td>
<td>

<p>If <code>y</code> is supplied, this is a vector of x coordinates for the binarization values in<code>y</code>, which are plotted on the y axis. If <code>y</code> is not supplied, this is object of class <code><a href="#topic+BinarizationResult-class">BinarizationResult</a></code> containing the binarized values to visualize.
</p>
</td></tr>
<tr><td><code id="plot+2B2CBinarizationResult-method_+3A_y">y</code></td>
<td>

<p>If <code>x</code> is a vector of x coordinates, this is object of class <code><a href="#topic+BinarizationResult-class">BinarizationResult</a></code> containing the binarized values to visualize.    
</p>
</td></tr>    
<tr><td><code id="plot+2B2CBinarizationResult-method_+3A_twodimensional">twoDimensional</code></td>
<td>

<p>Specifies whether the binarization is depicted as a ray or as a two-dimensional curve (see details).
</p>
</td></tr>
<tr><td><code id="plot+2B2CBinarizationResult-method_+3A_showlegend">showLegend</code></td>
<td>

<p>If set to <code>true</code>, a legend is included in the plot.
</p>
</td></tr>
<tr><td><code id="plot+2B2CBinarizationResult-method_+3A_showthreshold">showThreshold</code></td>
<td>

<p>If set to <code>true</code>, the binarization threshold is depicted as a horizontal or vertical line (depending on <code>twoDimensional</code>).
</p>
</td></tr>
<tr><td><code id="plot+2B2CBinarizationResult-method_+3A_...">...</code></td>
<td>

<p>Further graphical parameters to be passed to <code><a href="base.html#topic+plot">plot</a></code>. The parameters <code>col</code> and <code>pch</code> can be supplied in different ways: If supplied as vectors of size 2, the first value corresponds to a 0 in the binarization, and the second value corresponds to a 1 in the binarization. <code>col</code> can also have length 3, in which case the third entry is the color of the threshold line. 
If <code>col</code> or <code>pch</code> have the size of the input vector, the corresponding colors and symbols are assigned to the data points. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function comprises two different plots: If <code>twoDimensional = TRUE</code>, the positions in the input vector are aligned with the x axis, and the y axis corresponds to the values. The binarization threshold is shown as a horizontal line, and the binarization is indicated by two different symbols. 
</p>
<p>If <code>twoDimensional = FALSE</code>, the binarized values are aligned with a one-dimensional ray, and the separating threshold is depicted as a vertical line. 
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+plot">plot</a></code>, 
<code><a href="#topic+BinarizationResult-class">BinarizationResult</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># plot a binarization in one and two dimensions
res &lt;- binarize.BASC(iris[,"Petal.Length"], method="A")
plot(res)
plot(res, twoDimensional = TRUE)
plot(res, twoDimensional = TRUE, 
     pch = c("x", "+"), 
     col = c("red", "black", "royalblue"), 
     lty = 4, lwd = 2)
</code></pre>

<hr>
<h2 id='plot+2CTrinarizationResult-method'>
Visualization of trinarization results.
</h2><span id='topic+plot+2CTrinarizationResult-method'></span><span id='topic+plot+2CTrinarizationResult+2CANY-method'></span><span id='topic+plot+2Cnumeric+2CTrinarizationResult-method'></span>

<h3>Description</h3>

<p>Visualizes a trinarization as a ray or a two-dimensional plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'TrinarizationResult,ANY'
plot(x,
	twoDimensional=FALSE,
	showLegend=TRUE,
	showThreshold=TRUE,
	...)
## S4 method for signature 'numeric,TrinarizationResult'
plot(x,
	y,
	showLegend=TRUE,
	showThreshold=TRUE,
	...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2CTrinarizationResult-method_+3A_x">x</code></td>
<td>

<p>If <code>y</code> is supplied, this is a vector of x coordinates for the trinarization values in<code>y</code>, which are plotted on the y axis. If <code>y</code> is not supplied, this is object of class <code><a href="#topic+TrinarizationResult-class">TrinarizationResult</a></code> containing the trinarized values to visualize.
</p>
</td></tr>
<tr><td><code id="plot+2B2CTrinarizationResult-method_+3A_y">y</code></td>
<td>

<p>If <code>x</code> is a vector of x coordinates, this is object of class <code><a href="#topic+TrinarizationResult-class">TrinarizationResult</a></code> containing the trinarized values to visualize.
</p>
</td></tr>
<tr><td><code id="plot+2B2CTrinarizationResult-method_+3A_twodimensional">twoDimensional</code></td>
<td>

<p>Specifies whether the trinarization is depicted as a ray or as a two-dimensional curve (see details).
</p>
</td></tr>
<tr><td><code id="plot+2B2CTrinarizationResult-method_+3A_showlegend">showLegend</code></td>
<td>

<p>If set to <code>true</code>, a legend is included in the plot.
</p>
</td></tr>
<tr><td><code id="plot+2B2CTrinarizationResult-method_+3A_showthreshold">showThreshold</code></td>
<td>

<p>If set to <code>true</code>, the trinarization thresholds are depicted as a horizontal or vertical lines (depending on <code>twoDimensional</code>).
</p>
</td></tr>
<tr><td><code id="plot+2B2CTrinarizationResult-method_+3A_...">...</code></td>
<td>

<p>Further graphical parameters to be passed to <code><a href="base.html#topic+plot">plot</a></code>. The parameters <code>col</code> and <code>pch</code> can be supplied in different ways: If supplied as vectors of size 3, the first value corresponds to a 0 in the trinarization, the second value corresponds to a 1, and the third value corresponds to a 2. <code>col</code> can also have length 4, in which case the fourth entry is the color of the threshold line.
If <code>col</code> or <code>pch</code> have the size of the input vector, the corresponding colors and symbols are assigned to the data points. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function comprises two different plots: If <code>twoDimensional = TRUE</code>, the positions in the input vector are aligned with the x axis, and the y axis corresponds to the values. The trinarization thresholds are shown as a horizontal lines, and the trinarization is indicated by three different symbols.
</p>
<p>If <code>twoDimensional = FALSE</code>, the trinarized values are aligned with a one-dimensional ray, and the separating thresholds are depicted as a vertical lines.
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+plot">plot</a></code>, 
<code><a href="#topic+TrinarizationResult-class">TrinarizationResult</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># plot a binarization in one and two dimensions
res &lt;- TASC(iris[,"Petal.Length"])
plot(res)
plot(res, twoDimensional = TRUE)
plot(res, twoDimensional = TRUE, 
	 pch = c("x", "+"), 
	 col = c("red", "black", "royalblue", "green"), 
	 lty = 4, lwd = 2)
</code></pre>

<hr>
<h2 id='plotStepFunctions'>
Plot all step functions for BASC or TASC
</h2><span id='topic+plotStepFunctions'></span>

<h3>Description</h3>

<p>A specialized visualization that plots all the optimal step functions computed by the BASC algorithms or TASC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotStepFunctions(x, 
                  showLegend=TRUE, 
                  connected=FALSE, 
                  withOriginal=TRUE, 
                  ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotStepFunctions_+3A_x">x</code></td>
<td>

<p>A binarization (or trinarisation) result object of class <code><a href="#topic+BASCResult-class">BASCResult</a></code> (<code><a href="#topic+TASCResult-class">TASCResult</a></code>).
</p>
</td></tr>
<tr><td><code id="plotStepFunctions_+3A_showlegend">showLegend</code></td>
<td>

<p>If <code>TRUE</code>, a legend is included in the plot.
</p>
</td></tr>
<tr><td><code id="plotStepFunctions_+3A_connected">connected</code></td>
<td>

<p>If <code>TRUE</code>, the single steps of the step functions are connected by lines.
</p>
</td></tr>
<tr><td><code id="plotStepFunctions_+3A_withoriginal">withOriginal</code></td>
<td>

<p>If set to <code>TRUE</code>, the original step function (i.e. the sorted input vector) is included in the plot.
</p>
</td></tr>
<tr><td><code id="plotStepFunctions_+3A_...">...</code></td>
<td>

<p>Additional graphical parameters to be passed to <code><a href="base.html#topic+plot">plot</a></code>.
</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+BASCResult-class">BASCResult</a></code>,
<code><a href="#topic+binarize.BASC">binarize.BASC</a></code>,
<code><a href="#topic+TASCResult-class">TASCResult</a></code>,
<code><a href="#topic+TASC">TASC</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>result &lt;- binarize.BASC(iris[,"Petal.Width"], 
                        method="B")
plotStepFunctions(result)

result &lt;- TASC(iris[,"Petal.Width"])
plotStepFunctions(result)
</code></pre>

<hr>
<h2 id='TASC'>
Trinarization Across Multiple Scales
</h2><span id='topic+TASC'></span>

<h3>Description</h3>

<p>Trinarizes real-valued data using the multiscale TASC method.</p>


<h3>Usage</h3>

<pre><code class='language-R'>TASC(vect, 
	method = c("A","B"), 
	tau = 0.01, 
	numberOfSamples = 999, 
	sigma = seq(0.1, 20, by=.1),
	na.rm=FALSE,
	error = c("mean", "min"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TASC_+3A_method">method</code></td>
<td>

<p>Chooses the TASC method to use (see details), i.e. either  &quot;A&quot; or &quot;B&quot;. 
</p>
</td></tr>
<tr><td><code id="TASC_+3A_vect">vect</code></td>
<td>

<p>A real-valued vector of data to trinarize.
</p>
</td></tr>
<tr><td><code id="TASC_+3A_tau">tau</code></td>
<td>

<p>This parameter adjusts the sensitivity and the specificity of the statistical testing procedure that rates the quality of the trinarization. Defaults to 0.01.
</p>
</td></tr>
<tr><td><code id="TASC_+3A_numberofsamples">numberOfSamples</code></td>
<td>

<p>The number of samples for the bootstrap test. Defaults to 999.
</p>
</td></tr>
<tr><td><code id="TASC_+3A_sigma">sigma</code></td>
<td>

<p>If <code>method="B"</code>, this specifies a vector of different sigma values for the convolutions with the Bessel function. Ignored for <code>method="A"</code>.
</p>
</td></tr>
<tr><td><code id="TASC_+3A_na.rm">na.rm</code></td>
<td>

<p>If set to <code>TRUE</code>, <code>NA</code> values are removed from the input. Otherwise, trinarization will fail in the presence of <code>NA</code> values.
</p>
</td></tr>
<tr><td><code id="TASC_+3A_error">error</code></td>
<td>

<p>Determines which error should be used for the data points between two thresholds, the &quot;mean&quot; error (default) to the thresholds or the &quot;min&quot; error.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The two TASC methods can be subdivided into three steps:
</p>
	
<dl>
<dt>Compute a series of step functions:</dt><dd><p>An initial step function is obtained by rearranging the original time series measurements in increasing order. Then, step functions with fewer discontinuities are calculated. TASC A calculates these step functions in such a way that each minimizes the Euclidean distance to the initial step function. TASC B obtains step functions from smoothened versions of the input function in a scale-space manner.</p>
</dd>
<dt>Find strongest discontinuities in each step function:</dt><dd><p>A strong discontinuity is a high jump size (derivative) in combination with a low approximation error. For TASC a pair of strongest discontinuities is determined.</p>
</dd>
<dt>Estimate location and variation of the strongest discontinuities:</dt><dd><p>Based on these estimates, data values can be excluded from further analyses.
</p>
</dd>
</dl>



<h3>Value</h3>

<p>Returns an object of class <code><a href="#topic+TASCResult-class">TASCResult</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+TrinarizationResult-class">TrinarizationResult</a></code>, 
<code><a href="#topic+TASCResult-class">TASCResult</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>par(mfrow=c(2,1))
result &lt;- TASC(iris[,"Petal.Width"], method="A", tau=0.15)
print(result)
plot(result)

result &lt;- TASC(iris[,"Petal.Width"], method="B", tau=0.15)
print(result)
plot(result)
</code></pre>

<hr>
<h2 id='TASCResult-class'>Class &quot;TASCResult&quot;</h2><span id='topic+TASCResult-class'></span><span id='topic+plotStepFunctions+2CTASCResult-method'></span><span id='topic+print+2CTASCResult-method'></span><span id='topic+show+2CTASCResult-method'></span>

<h3>Description</h3>

<p>A specialized class storing the results of a call to <code><a href="#topic+TASC">TASC</a></code>.
</p>


<h3>Objects of this class</h3>

<p>Objects of this class shouldn't be created directly. They are created implicitly by a call to <code><a href="#topic+TASC">TASC</a></code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>p.value</code>:</dt><dd><p>The p-value of the statistical test for reliability of the trinarization.</p>
</dd>
<dt><code>intermediateSteps</code>:</dt><dd><p>A matrix specifying the optimal step functions from which the trinarization was calculated. The number of rows corresponds to the number of step functions, and the number of columns is determined by the length of the input vector minus 2 (that is, the length of the step function corresponding to the input vector). From the first to the last row, the number of steps increases. The non-zero entries of the matrix represent the locations of the steps. Step functions with fewer steps than the input step function have entries set to zero.</p>
</dd>
<dt><code>intermediateHeights1</code>:</dt><dd><p>A matrix giving the jump heights of the steps supplied in <code>intermediateSteps</code> for the first threshold.</p>
</dd>
<dt><code>intermediateHeights2</code>:</dt><dd><p>A matrix giving the jump heights of the steps supplied in <code>intermediateSteps</code> for the second threshold.</p>
</dd>
<dt><code>intermediateStrongestSteps</code>:</dt><dd><p>A matrix with one row for each step function (row) in <code>intermediateSteps</code>. The entries specify the location of the two strongest steps for each of the functions.</p>
</dd>
<dt><code>originalMeasurements</code>:</dt><dd><p>A numeric vector storing the input measurements.</p>
</dd>
<dt><code>trinarizedMeasurements</code>:</dt><dd><p>An integer vector of trinarized values (0, 1 or 2) corresponding to the original measurements.</p>
</dd>
<dt><code>threshold1</code>:</dt><dd><p>The threshold that separates 0 from 1.</p>
</dd>
<dt><code>threshold2</code>:</dt><dd><p>The threshold that separates 1 from 2.</p>
</dd>
<dt><code>method</code>:</dt><dd><p>A string describing the trinarization method that yielded the result.</p>
</dd>
</dl>



<h3>Extends</h3>

<p>Class <code>"<a href="#topic+TrinarizationResult-class">TrinarizationResult</a>"</code>, directly.
</p>


<h3>Methods</h3>


<dl>
<dt><a href="#topic+plotStepFunctions">plotStepFunctions</a></dt><dd><p><code>signature(x = "TASCResult")</code>: Plot the intermediate optimal step functions used to determine the thresholds.</p>
</dd>
<dt>print</dt><dd><p><code>signature(x = "TASCResult")</code>: Print a summary of the trinarization.</p>
</dd>
<dt>show</dt><dd><p><code>signature(object = "TASCResult")</code>: ... </p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+TASC">TASC</a></code>, 
<code><a href="#topic+TrinarizationResult-class">TrinarizationResult</a></code>
</p>

<hr>
<h2 id='trinarizationExample'>
An artificial data set consisting of ten artificial feature vectors.
</h2><span id='topic+trinarizationExample'></span>

<h3>Description</h3>

<p>An artificial data set consisting of 100 artificial feature vectors that are used to illustrate the trinarization methods in the package vignette. Each row of the matrix <code>trinarizationExample</code> corresponds to one feature vector, of which 5 measurements are drawn from a normal distribution N(0,1). The remaining 10 measurements are drawn from two normal distributions N(m,1), with <code>m=10:1</code> and <code>m=seq(20,2,by=-2)</code> (5 meansurements per distribution).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(trinarizationExample)</code></pre>


<h3>Format</h3>

<p>The data is a matrix with 15 columns and 100 rows.
</p>

<hr>
<h2 id='TrinarizationResult-class'>Class &quot;TrinarizationResult&quot;</h2><span id='topic+TrinarizationResult-class'></span><span id='topic+plotTrinarization+2CTrinarizationResult-method'></span><span id='topic+print+2CTrinarizationResult-method'></span><span id='topic+show+2CTrinarizationResult-method'></span>

<h3>Description</h3>

<p>This is the base class for objects that store the results of a trinarization algorithm. It defines the slots and methods that the results of all algorithms share.
</p>


<h3>Objects of this class</h3>

<p>Objects of this class shouldn't be created directly. They are created implicitly by a call to one of the trinarizeation algorithms.
</p>


<h3>Slots</h3>


<dl>
<dt><code>originalMeasurements</code>:</dt><dd><p>A numeric vector storing the input measurements.</p>
</dd>
<dt><code>trinarizedMeasurements</code>:</dt><dd><p>An integer vector of trinarized values (0 or 1 or 2) corresponding to the original measurements.</p>
</dd>
<dt><code>threshold1</code>:</dt><dd><p>The threshold that separates 0 and 1.</p>
</dd>
<dt><code>threshold2</code>:</dt><dd><p>The threshold that separates 1 and 2.</p>
</dd>
<dt><code>method</code>:</dt><dd><p>A string describing the trinarization method that yielded the result.</p>
</dd>
<dt><code>p.value</code>:</dt><dd><p>The p-value obtained by a test for validity of the trinarization (e.g. TASC bootstrap test). If no test was performed, this is <code>NA</code>.</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt><a href="#topic+plot+2CBinarizationResult-method">plot</a></dt><dd><p><code>signature(x = "TrinarizationResult")</code>: Plot the trinarization and the thresholds.</p>
</dd>
<dt>print</dt><dd><p><code>signature(x = "TrinarizationResult")</code>: Print a summary of the trinarization.</p>
</dd>
<dt>show</dt><dd><p><code>signature(object = "TrinarizationResult")</code>: ... </p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+TASC">TASC</a></code>, 
<code><a href="#topic+TASCResult-class">TASCResult</a></code>, 
</p>

<hr>
<h2 id='trinarize.kMeans'>
k-means Trinarization
</h2><span id='topic+trinarize.kMeans'></span>

<h3>Description</h3>

<p>Trinarizes a vector of real-valued data using the k-means clustering algorithm. The data is first split into 3 clusters.The values belonging to the cluster with the smallest centroid are set to 0, the values belonging to the greater centroid are set to 1, and the values belonging to the greatest centroid are set to 2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trinarize.kMeans(vect, 
                nstart=1, 
                iter.max=10,
                dip.test=TRUE,
                na.rm=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trinarize.kMeans_+3A_vect">vect</code></td>
<td>

<p>A real-valued vector to be trinarized.
</p>
</td></tr>
<tr><td><code id="trinarize.kMeans_+3A_nstart">nstart</code></td>
<td>

<p>The number of restarts for k-means. See <code><a href="stats.html#topic+kmeans">kmeans</a></code> for details.</p>
</td></tr>
<tr><td><code id="trinarize.kMeans_+3A_iter.max">iter.max</code></td>
<td>

<p>The maximum number of iterations for k-means. See <code><a href="stats.html#topic+kmeans">kmeans</a></code> for details.</p>
</td></tr>
<tr><td><code id="trinarize.kMeans_+3A_dip.test">dip.test</code></td>
<td>
<p>If set to <code>TRUE</code>, Hartigan's dip test for unimodality is performed on <code>vect</code>, and its p-value is returned in the <code>pvalue</code> slot of the result. An insignificant test indicates that the data may not be trinarizeable.</p>
</td></tr>
<tr><td><code id="trinarize.kMeans_+3A_na.rm">na.rm</code></td>
<td>

<p>If set to <code>TRUE</code>, <code>NA</code> values are removed from the input. Otherwise, trinarization will fail in the presence of <code>NA</code> values.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class <code><a href="#topic+TrinarizationResult-class">TrinarizationResult</a></code>. 
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+kmeans">kmeans</a></code>, 
<code><a href="#topic+TrinarizationResult-class">TrinarizationResult</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>result &lt;- trinarize.kMeans(iris[,"Petal.Length"])

print(result)
plot(result, twoDimensional=TRUE)
</code></pre>

<hr>
<h2 id='trinarizeMatrix'>
Utility function to trinarize a matrix of measurements
</h2><span id='topic+trinarizeMatrix'></span>

<h3>Description</h3>

<p>Trinarizes a matrix of measurements all at once, and returns the trinarized vectors as well as the trinarization thresholds and the p-values. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trinarizeMatrix(mat, 
               method = c("TASCA", "TASCB","kMeans"), 
               adjustment = "none", 
               ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trinarizeMatrix_+3A_mat">mat</code></td>
<td>

<p>A n x m matrix comprising m raw measurements of n features.
</p>
</td></tr>
<tr><td><code id="trinarizeMatrix_+3A_method">method</code></td>
<td>

<p>The trinarization algorithm to be used. <code>method="TASCA"</code> calls <code><a href="#topic+TASC">TASC</a></code> with <code>method="A"</code>. <code>method="TASCB"</code> calls <code><a href="#topic+TASC">TASC</a></code> with <code>method="B"</code>. <code>method="kMeans"</code> calls <code><a href="#topic+trinarize.kMeans">trinarize.kMeans</a></code>.
</p>
</td></tr>
<tr><td><code id="trinarizeMatrix_+3A_adjustment">adjustment</code></td>
<td>

<p>Specifies an optional adjustment for multiple testing that is applied to the p-values (see <code><a href="stats.html#topic+p.adjust">p.adjust</a></code> for possible values). By default, no adjustment is applied.
</p>
</td></tr>
<tr><td><code id="trinarizeMatrix_+3A_...">...</code></td>
<td>

<p>Further parameters that are passed to the respective trinarization methods (<code><a href="#topic+TASC">TASC</a></code>, <code><a href="#topic+trinarize.kMeans">trinarize.kMeans</a></code>).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A n x (m+3) matrix of trinarized measurements. Here, the first m columns correspond to the trinarized measurements. The m+1-st and the m+2-st column comprises the trinarization thresholds for the features, and the m+3-nd column contains the p-values.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+TASC">TASC</a></code>, <code><a href="#topic+trinarize.kMeans">trinarize.kMeans</a></code>, <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tri &lt;- trinarizeMatrix(t(iris[,1:4]))
print(tri)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
