<!DOCTYPE html><html><head><title>Help for package blaster</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {blaster}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#blaster-package'><p>blaster: Native R Implementation of an Efficient BLAST-Like Algorithm</p></a></li>
<li><a href='#blast'><p>Runs BLAST sequence comparison algorithm.</p></a></li>
<li><a href='#read_fasta'><p>Reads the contents of nucleotide or protein FASTA file into a dataframe.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Native R Implementation of an Efficient BLAST-Like Algorithm</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-08-22</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Manu Tamminen &lt;mavatam@utu.fi&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implementation of an efficient BLAST-like sequence
             comparison algorithm, written in 'C++11' and using
             native R datatypes. Blaster is based on 'nsearch' -
             Schmid et al (2018) &lt;<a href="https://doi.org/10.1101%2F399782">doi:10.1101/399782</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/BSD-3-Clause">BSD_3_clause</a> + file LICENSE</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 1.0.5)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>C++</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/tamminenlab/blaster">https://github.com/tamminenlab/blaster</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-22 13:12:19 UTC; mavatam</td>
</tr>
<tr>
<td>Author:</td>
<td>Manu Tamminen <a href="https://orcid.org/0000-0001-5891-7653"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Timothy Julian <a href="https://orcid.org/0000-0003-1000-0306"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Aditya Jeevennavar
    <a href="https://orcid.org/0000-0002-0737-7316"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Steven Schmid [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-22 14:40:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='blaster-package'>blaster: Native R Implementation of an Efficient BLAST-Like Algorithm</h2><span id='topic+blaster'></span><span id='topic+blaster-package'></span>

<h3>Description</h3>

<p>Implementation of an efficient BLAST-like sequence comparison algorithm, written in 'C++11' and using native R datatypes. Blaster is based on 'nsearch' - Schmid et al (2018) <a href="https://doi.org/10.1101/399782">doi:10.1101/399782</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Manu Tamminen <a href="mailto:mavatam@utu.fi">mavatam@utu.fi</a> (<a href="https://orcid.org/0000-0001-5891-7653">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Timothy Julian <a href="mailto:tim.julian@eawag.ch">tim.julian@eawag.ch</a> (<a href="https://orcid.org/0000-0003-1000-0306">ORCID</a>)
</p>
</li>
<li><p> Aditya Jeevennavar <a href="mailto:aditya.a.jeevannavar@utu.fi">aditya.a.jeevannavar@utu.fi</a> (<a href="https://orcid.org/0000-0002-0737-7316">ORCID</a>)
</p>
</li>
<li><p> Steven Schmid <a href="mailto:stevschmid@gmail.com">stevschmid@gmail.com</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/tamminenlab/blaster">https://github.com/tamminenlab/blaster</a>
</p>
</li></ul>


<hr>
<h2 id='blast'>Runs BLAST sequence comparison algorithm.</h2><span id='topic+blast'></span>

<h3>Description</h3>

<p>Runs BLAST sequence comparison algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>blast(
  query,
  db,
  maxAccepts = 1,
  maxRejects = 16,
  minIdentity = 0.75,
  alphabet = "nucleotide",
  strand = "both",
  output_to_file = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="blast_+3A_query">query</code></td>
<td>
<p>A dataframe of the query sequences (containing Id and Seq columns)
or a string specifying the FASTA file of the query sequences.</p>
</td></tr>
<tr><td><code id="blast_+3A_db">db</code></td>
<td>
<p>A dataframe of the database sequences (containing Id and Seq columns)
or a string specifying the FASTA file of the database sequences.</p>
</td></tr>
<tr><td><code id="blast_+3A_maxaccepts">maxAccepts</code></td>
<td>
<p>A number specifying the maximum accepted hits.</p>
</td></tr>
<tr><td><code id="blast_+3A_maxrejects">maxRejects</code></td>
<td>
<p>A number specifying the maximum rejected hits.</p>
</td></tr>
<tr><td><code id="blast_+3A_minidentity">minIdentity</code></td>
<td>
<p>A number specifying the minimal accepted sequence
similarity between the query and hit sequences.</p>
</td></tr>
<tr><td><code id="blast_+3A_alphabet">alphabet</code></td>
<td>
<p>A string specifying the query and database alphabet:
'nucleotide' or 'protein'. Defaults to 'nucleotide'.</p>
</td></tr>
<tr><td><code id="blast_+3A_strand">strand</code></td>
<td>
<p>A string specifying the strand to search: 'plus', 'minus' or
'both'. Defaults to 'both'. Only affects nucleotide searches.</p>
</td></tr>
<tr><td><code id="blast_+3A_output_to_file">output_to_file</code></td>
<td>
<p>A boolean specifying the output type. If TRUE, the
results are written into a temporary file a string
containing the file name and location is returned.
Otherwise a dataframe of the results is returned.
Defaults to FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe or a string. A dataframe is returned by default, containing
the BLAST output in columns QueryId, TargetId, QueryMatchStart, QueryMatchEnd,
TargetMatchStart, TargetMatchEnd, QueryMatchSeq, TargetMatchSeq, NumColumns,
NumMatches, NumMismatches, NumGaps, Identity and Alignment. A string is returned
if 'output_to_file' is set to TRUE. This string points to the file
containing the output table.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
query &lt;- system.file("extdata", "query.fasta", package = "blaster")
db &lt;- system.file("extdata", "db.fasta", package = "blaster")

blast_table &lt;- blast(query = query, db = db)

query &lt;- read_fasta(filename = query)
db &lt;- read_fasta(filename = db)
blast_table &lt;- blast(query = query, db = db)

prot &lt;- system.file("extdata", "prot.fasta", package = "blaster")
prot_blast_table &lt;- blast(query = prot, db = prot, alphabet = "protein")

</code></pre>

<hr>
<h2 id='read_fasta'>Reads the contents of nucleotide or protein FASTA file into a dataframe.</h2><span id='topic+read_fasta'></span>

<h3>Description</h3>

<p>Reads the contents of nucleotide or protein FASTA file into a dataframe.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_fasta(
  filename,
  filter = "",
  non_standard_chars = "error",
  alphabet = "nucleotide"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_fasta_+3A_filename">filename</code></td>
<td>
<p>A string specifying the name of the FASTA file to be imported.</p>
</td></tr>
<tr><td><code id="read_fasta_+3A_filter">filter</code></td>
<td>
<p>An optional string specifying a sequence motif for sequence filtering.
Only keeps those sequences containing this motif. Also splits the
matched sequences and provides the split parts in two additional columns.</p>
</td></tr>
<tr><td><code id="read_fasta_+3A_non_standard_chars">non_standard_chars</code></td>
<td>
<p>A string specifying instructions for handling non-standard
nucleotide or amino acid characters. Options include 'remove',
'ignore' or throw an 'error'. Defaults to 'error'.</p>
</td></tr>
<tr><td><code id="read_fasta_+3A_alphabet">alphabet</code></td>
<td>
<p>A string specifying the query and database alphabet:
'nucleotide' or 'protein'. Defaults to 'nucleotide'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe containing FASTA ids (Id column) and sequences (Seq column).
If 'filter' is specified, the split sequences are stored in additional columns
Part1 and Part2.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
query &lt;- system.file("extdata", "query.fasta", package = "blaster")

query &lt;- read_fasta(filename = query)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
