<!DOCTYPE html><html lang="en"><head><title>Help for package Blend</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Blend}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Blend-package'><p>Robust Bayesian Longitudinal Regularized Semiparametric Mixed Model</p></a></li>
<li><a href='#Blend'><p>fit a robust Bayesian longitudinal regularized semi-parametric mixed model</p></a></li>
<li><a href='#Coverage'><p>95% coverage for a Blend object with structural identification</p></a></li>
<li><a href='#data'><p>simulated data for demonstrating the features of Blend</p></a></li>
<li><a href='#plot_Blend'><p>plot a Blend object</p></a></li>
<li><a href='#selection'><p>Variable selection for a Blend object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Robust Bayesian Longitudinal Regularized Semiparametric Mixed
Models</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2025-01-29</td>
</tr>
<tr>
<td>Description:</td>
<td>Our recently developed fully robust Bayesian semiparametric mixed-effect model for high-dimensional longitudinal studies with heterogeneous observations can be implemented through this package. This model can distinguish between time-varying interactions and constant-effect-only cases to avoid model misspecifications. Facilitated by spike-and-slab priors, this model leads to superior performance in estimation, identification and statistical inference. In particular, robust Bayesian inferences in terms of valid Bayesian credible intervals on both parametric and nonparametric effects can be validated on finite samples. The Markov chain Monte Carlo algorithms of the proposed and alternative models are efficiently implemented in 'C++'.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.2.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/kunfa/Blend">https://github.com/kunfa/Blend</a></td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, splines, stats, ggplot2</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-01-29 21:22:57 UTC; kunfan</td>
</tr>
<tr>
<td>Author:</td>
<td>Kun Fan [aut, cre],
  Cen Wu [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kun Fan &lt;kfan@ksu.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-29 21:40:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='Blend-package'>Robust Bayesian Longitudinal Regularized Semiparametric Mixed Model</h2><span id='topic+Blend-package'></span>

<h3>Description</h3>

<p>In this package, we further extend the sparse robust Bayesian mixed models to nonlinear longitudinal interactions. Specifically, the proposed Bayesian semiparametric model is robust not only to outliers and heavy‐tailed distributions of the response variable, but also to the misspecification of interaction effect in the forms other than non-linear interactions. We have developed the Gibbs sampler with the spike‐and‐slab priors to promote sparse identification of appropriate forms of main and interaction effects.
In addition to the default method, users can also choose different selection structures for separation of constant and varying effects or not, methods without spike&ndash;and&ndash;slab priors and non-robust methods. In total, <em>Blend</em> provides 8 different methods (4 robust and 4 non-robust) under the random intercept and slope model. All the methods in this package are developed for the first time. Please read the Details below for how to configure the method used.
</p>


<h3>Details</h3>

<p>The user friendly, integrated interface <strong>Blend()</strong> allows users to flexibly choose the fitting methods by specifying the following parameter:
</p>

<table>
<tr>
 <td style="text-align: right;">
robust: </td><td style="text-align: left;"> whether to use robust methods for modelling.</td>
</tr>
<tr>
 <td style="text-align: right;"></td>
</tr>
<tr>
 <td style="text-align: right;">
structural: </td><td style="text-align: left;"> whether to incorporate structural identification(separation of constant and varying effects) .</td>
</tr>
<tr>
 <td style="text-align: right;"></td>
</tr>
<tr>
 <td style="text-align: right;">
sparse: </td><td style="text-align: left;"> whether to use the spike-and-slab priors to impose sparsity.
</td>
</tr>

</table>

<p>The function Blend() returns a Blend object that contains the posterior estimates of each coefficients and other useful information for selection().
S3 generic functions selection() and print() are implemented for Blend objects.
selection() takes a Blend object and returns the variable selection results.
</p>


<h3>References</h3>

<p>Fan, K., Ren, J., Ma, Shuangge and Wu, C. (2025+). Robust Bayesian Regularized Semiparametric Mixed Models in Longitudinal Studies. (submitted)
</p>
<p>Fan, K., Subedi, S., Yang, G., Lu, X., Ren, J., and Wu, C. (2024). Is Seeing Believing? A Practitioner’s Perspective on High-Dimensional Statistical Inference in Cancer Genomics Studies. Entropy, 26(9), 794.
</p>
<p>Ren, J., Zhou, F., Li, X., Ma, S., Jiang, Y. and Wu, C. (2023). Robust Bayesian variable selection for gene-environment interactions.
<em>Biometrics</em>, 79(2), 684-694  <a href="https://doi.org/10.1111/biom.13670">doi:10.1111/biom.13670</a>
</p>
<p>Zhou, F., Ren, J., Ma, S. and Wu, C. (2023). The Bayesian regularized quantile varying coefficient model. Computational Statistics &amp; Data Analysis, 187, 107808.
</p>
<p>Zhou, F., Lu, X., Ren, J., Fan, K., Ma, S., &amp; Wu, C. (2022). Sparse group variable selection for gene–environment interactions in the longitudinal study. Genetic epidemiology, 46(5-6), 317-340.
</p>
<p>Zhou, F., Ren, J.,  Li, G., Jiang, Y., Li, X., Wang, W. and Wu, C. (2019). Penalized Variable Selection for Lipid-Environment Interactions in a Longitudinal Lipidomics Study.
<em>Genes</em>, 10(12), 1002 <a href="https://doi.org/10.3390/genes10121002">doi:10.3390/genes10121002</a>
</p>
<p>Ren, J., Zhou, F., Li, X., Ma, S., Jiang, Y. and Wu, C. (2020). roben: Robust Bayesian Variable Selection for Gene-Environment Interactions.
R package version 0.1.1. <a href="https://CRAN.R-project.org/package=roben">https://CRAN.R-project.org/package=roben</a>
</p>
<p>Zhou, F., Ren, J., Lu, X., Ma, S. and Wu, C. (2021). Gene–Environment Interaction: a Variable Selection Perspective.
<em>Epistasis. Methods in Molecular Biology.</em> 2212:191–223 <a href="https://doi.org/10.1007/978-1-0716-0947-7_13">doi:10.1007/978-1-0716-0947-7_13</a>
</p>
<p>Ren, J., Zhou, F., Li, X., Chen, Q., Zhang, H., Ma, S., Jiang, Y. and Wu, C. (2020) Semi-parametric Bayesian variable selection for gene-environment interactions.
<em>Statistics in Medicine</em>, 39: 617– 638 <a href="https://doi.org/10.1002/sim.8434">doi:10.1002/sim.8434</a>
</p>
<p>Ren, J., Zhou, F., Li, X., Wu, C. and Jiang, Y. (2019) spinBayes: Semi-Parametric Gene-Environment Interaction via Bayesian Variable Selection.
R package version 0.1.0. <a href="https://CRAN.R-project.org/package=spinBayes">https://CRAN.R-project.org/package=spinBayes</a>
</p>
<p>Wu, C., Jiang, Y., Ren, J., Cui, Y. and Ma, S. (2018). Dissecting gene-environment interactions: A penalized robust approach accounting for hierarchical structures.
<em>Statistics in Medicine</em>, 37:437–456 <a href="https://doi.org/10.1002/sim.7518">doi:10.1002/sim.7518</a>
</p>
<p>Wu, C., Cui, Y., and Ma, S. (2014). Integrative analysis of gene–environment interactions under a multi–response partially linear varying coefficient model.
<em>Statistics in Medicine</em>, 33(28), 4988–4998 <a href="https://doi.org/10.1002/sim.6287">doi:10.1002/sim.6287</a>
</p>
<p>Wu, C., Zhong, P.S. and Cui, Y. (2013). High dimensional variable selection for gene-environment interactions.
<em>Technical Report. Michigan State University.</em>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Blend">Blend</a></code>
</p>

<hr>
<h2 id='Blend'>fit a robust Bayesian longitudinal regularized semi-parametric mixed model</h2><span id='topic+Blend'></span>

<h3>Description</h3>

<p>fit a robust Bayesian longitudinal regularized semi-parametric mixed model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Blend(
  y,
  x,
  t,
  J,
  kn,
  degree,
  iterations = 10000,
  burn.in = NULL,
  robust = TRUE,
  sparse = "TRUE",
  structural = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Blend_+3A_y">y</code></td>
<td>
<p>the vector of repeated - measured response variable. The current version of mixed only supports continuous response.</p>
</td></tr>
<tr><td><code id="Blend_+3A_x">x</code></td>
<td>
<p>the matrix of repeated - measured predictors (genetic factors) with intercept. Each row should be an observation vector for each measurement.</p>
</td></tr>
<tr><td><code id="Blend_+3A_t">t</code></td>
<td>
<p>the vector of scheduled time points.</p>
</td></tr>
<tr><td><code id="Blend_+3A_j">J</code></td>
<td>
<p>the vector of number of repeated measurement for each subject.</p>
</td></tr>
<tr><td><code id="Blend_+3A_kn">kn</code></td>
<td>
<p>the number of interior knots for B-spline.</p>
</td></tr>
<tr><td><code id="Blend_+3A_degree">degree</code></td>
<td>
<p>the degree of B spline basis.</p>
</td></tr>
<tr><td><code id="Blend_+3A_iterations">iterations</code></td>
<td>
<p>the number of MCMC iterations.</p>
</td></tr>
<tr><td><code id="Blend_+3A_burn.in">burn.in</code></td>
<td>
<p>the number of iterations for burn-in.</p>
</td></tr>
<tr><td><code id="Blend_+3A_robust">robust</code></td>
<td>
<p>logical flag. If TRUE, robust methods will be used.</p>
</td></tr>
<tr><td><code id="Blend_+3A_sparse">sparse</code></td>
<td>
<p>logical flag. If TRUE, spike-and-slab priors will be used to shrink coefficients of irrelevant covariates to zero exactly.</p>
</td></tr>
<tr><td><code id="Blend_+3A_structural">structural</code></td>
<td>
<p>logical flag. If TRUE, the coefficient functions with varying effects and constant effects will be penalized separately.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Consider the data model described in &quot;<code><a href="#topic+data">data</a></code>&quot;:
</p>
<p style="text-align: center;"><code class="reqn">Y_{ij} = \alpha_0(t_{ij})+\sum_{k=1}^{m}\beta_{k}(t_{ij})X_{ijk}+\boldsymbol{Z^\top_{ij}}\boldsymbol{\zeta_{i}}+\epsilon_{ij}.</code>
</p>

<p>The basis expansion and changing of basis with B splines will be done automatically:
</p>
<p style="text-align: center;"><code class="reqn">\beta_{k}(\cdot)\approx \gamma_{k1} + \sum_{u=2}^{q}{B}_{ku}(\cdot)\gamma_{ku}</code>
</p>

<p>where <code class="reqn">B_{ku}(\cdot)</code> represents B spline basis. <code class="reqn">\gamma_{k1}</code> and <code class="reqn">(\gamma_{k2}, \ldots, \gamma_{kq})^\top</code> correspond to the constant and varying parts of the coefficient functional, respectively.
q=kn+degree+1 is the number of basis functions. By default, kn=degree=2. User can change the values of kn and degree to any other positive integers.
When 'structural=TRUE'(default), the coefficient functions with varying effects and constant effects will be penalized separately. Otherwise, the coefficient functions with varying effects and constant effects will be penalized together.
</p>
<p>When 'sparse=&quot;TRUE&quot;' (default), spike-and-slab priors are imposed on individual and/or group levels to identify important constant and varying effects. Otherwise, Laplacian shrinkage will be used.
</p>
<p>When 'robust=TRUE' (default), the distribution of <code class="reqn">\epsilon_{ij}</code> is defined as a Laplace distribution with density.
</p>
<p><code class="reqn">
f(\epsilon_{ij}|\theta,\tau) = \theta(1-\theta)\exp\left\{-\tau\rho_{\theta}(\epsilon_{ij})\right\}
</code>, (<code class="reqn">i=1,\dots,n,j=1,\dots,J_{i} </code>), where <code class="reqn">\theta = 0.5</code>. If 'robust=FALSE', <code class="reqn">\epsilon_{ij}</code> follows a normal distribution.
<br />
</p>
<p>Please check the references for more details about the prior distributions.
</p>


<h3>Value</h3>

<p>an object of class &lsquo;Blend&rsquo; is returned, which is a list with component:
</p>
<table role = "presentation">
<tr><td><code>posterior</code></td>
<td>
<p>the posteriors of coefficients.</p>
</td></tr>
<tr><td><code>coefficient</code></td>
<td>
<p>the estimated coefficients.</p>
</td></tr>
<tr><td><code>burn.in</code></td>
<td>
<p>the total number of burn-ins.</p>
</td></tr>
<tr><td><code>iterations</code></td>
<td>
<p>the total number of iterations.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+data">data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dat)

## default method
fit = Blend(y,x,t,J,kn,degree)
fit$coefficient


## alternative: robust non-structural
fit = Blend(y,x,t,J,kn,degree, structural=FALSE)
fit$coefficient

## alternative: non-robust structural
fit = Blend(y,x,t,J,kn,degree, robust=FALSE)
fit$coefficient

## alternative: non-robust non-structural
fit = Blend(y,x,t,J,kn,degree, robust=FALSE, structural=FALSE)
fit$coefficient



</code></pre>

<hr>
<h2 id='Coverage'>95% coverage for a Blend object with structural identification</h2><span id='topic+Coverage'></span>

<h3>Description</h3>

<p>calculate 95% coverage for varying effects and constant effects under example data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Coverage(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Coverage_+3A_x">x</code></td>
<td>
<p>Blend object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>coverage
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Blend">Blend</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dat)
fit = Blend(y,x,t,J,kn,degree)
Coverage(fit)

</code></pre>

<hr>
<h2 id='data'>simulated data for demonstrating the features of Blend</h2><span id='topic+data'></span><span id='topic+y'></span><span id='topic+x'></span><span id='topic+t'></span><span id='topic+J'></span><span id='topic+kn'></span><span id='topic+degree'></span>

<h3>Description</h3>

<p>Simulated gene expression data for demonstrating the features of Blend.
</p>


<h3>Format</h3>

<p>The data object consists of 8 components: y, x, t, J, kn and degree.
</p>


<h3>Details</h3>

<p><strong>The data and model setting</strong>
</p>
<p>Consider a longitudinal study on <code class="reqn">n</code> subjects with <code class="reqn">J_i</code> repeated measurements for each subject. Let <code class="reqn">Y_{ij}</code> be the measurement for the <code class="reqn">i</code>-th subject at each time point <code class="reqn">t_{ij}</code>, <code class="reqn">(1 \leq i \leq n, 1 \leq j \leq J_i)</code>. We use an <code class="reqn">m</code>-dimensional vector <code class="reqn">X_{ij}</code> to denote the genetic factors, where <code class="reqn">X_{ij} = (X_{ij1},...,X_{ijm})^\top</code>. <code class="reqn">Z_{ij}</code> is a <code class="reqn">2 \times 1</code> covariate associated with random effects and <code class="reqn">\zeta_{i}</code> is a <code class="reqn">2 \times 1</code> vector of random effects corresponding to the random intercept and slope model. We have the following semi-parametric quantile mixed-effects model:
</p>
<p><code class="reqn">Y_{ij} = \alpha_0(t_{ij}) + \sum_{k=1}^{m} \beta_{k}(t_{ij}) X_{ijk} + Z_{ij}^\top \zeta_{i} + \epsilon_{ij}, \zeta_{i} \sim N(0, \Lambda)</code>
</p>
<p>where the fixed effects include: (a) the varying intercept <code class="reqn">\alpha_0(t_{ij})</code>, and (b) the varying coefficients <code class="reqn">\beta(t_{ij})</code>.
</p>
<p>The varying intercept and the varying coefficients for the genetic factors can be further expressed as <code class="reqn">\alpha_0(t_{ij})</code> and <code class="reqn">\beta(t_{ij}) = (\beta_{1}(t_{ij}), ..., \beta_{m}(t_{ij}))^\top</code>.
</p>
<p>For the random intercept and slope model, <code class="reqn">Z_{ij}^\top = (1, j)</code> and <code class="reqn">\zeta_{i} = (\zeta_{i1}, \zeta_{i2})^\top</code>.
</p>
<p>Furthermore, <code class="reqn">Z_{ij}^\top \zeta_{i}</code> can be expressed as <code class="reqn">(b_i^\top \otimes Z^\top_{ij}) J_2 \delta</code>,
where <code class="reqn">\zeta_{i} = \Delta b_i</code>, <code class="reqn">\Lambda = \Delta \Delta^\top</code>, and
</p>
<p><code class="reqn">b_i^\top \otimes Z^\top_{ij} = (b_{i1} Z_{ij1}, b_{i1} Z_{ij2}, b_{i2}Z_{ij1}, b_{i2} Z_{ij2})^\top</code>.
</p>
<p>In the simulated data,
</p>
<p style="text-align: center;"><code class="reqn">Y = \alpha_{0}(t)+\beta_{1}(t)X_{1} + \beta_{2}(t)X_{2} + \beta_{3}(t)X_{3}+ \beta_{4}(t)X_{4}+0.8X_{5} -1.2 X_{6} + 0.7X_{7}-1.1 X_{8}+\epsilon</code>
</p>

<p>where <code class="reqn">\epsilon\sim N(0,1)</code>, <code class="reqn">\alpha_{0}(t)=2+\sin(2\pi t)</code>, <code class="reqn">\beta_{1}(t)=2.5\exp(2.5t-1) </code>,<code class="reqn">\beta_{2}(t)=3t^2-2t+2</code>,<code class="reqn">\beta_{3}(t)=-4t^3+3</code> and <code class="reqn">\beta_{4}(t)=3-2t</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Blend">Blend</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dat)
length(y)
dim(x)
length(t)
length(J)
print(t)
print(J)
print(kn)
print(degree)
</code></pre>

<hr>
<h2 id='plot_Blend'>plot a Blend object</h2><span id='topic+plot_Blend'></span>

<h3>Description</h3>

<p>plot the identified varying effects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_Blend(x, sparse, prob=0.95)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_Blend_+3A_x">x</code></td>
<td>
<p>Blend object.</p>
</td></tr>
<tr><td><code id="plot_Blend_+3A_sparse">sparse</code></td>
<td>
<p>sparsity.</p>
</td></tr>
<tr><td><code id="plot_Blend_+3A_prob">prob</code></td>
<td>
<p>probability for credible interval, between 0 and 1. e.g. prob=0.95 leads to 95% credible interval</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Blend">Blend</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dat)
fit = Blend(y,x,t,J,kn,degree)
plot_Blend(fit,sparse=TRUE)

</code></pre>

<hr>
<h2 id='selection'>Variable selection for a Blend object</h2><span id='topic+selection'></span>

<h3>Description</h3>

<p>Variable selection for a Blend object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>selection(obj, sparse)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="selection_+3A_obj">obj</code></td>
<td>
<p>Blend object.</p>
</td></tr>
<tr><td><code id="selection_+3A_sparse">sparse</code></td>
<td>
<p>logical flag. If TRUE, spike-and-slab priors will be used to shrink coefficients of irrelevant covariates to zero exactly.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If sparse, the median probability model (MPM) (Barbieri and Berger, 2004) is used to identify predictors that are significantly associated
with the response variable. Otherwise, variable selection is based on 95% credible interval.
Please check the references for more details about the variable selection.
</p>


<h3>Value</h3>

<p>an object of class &lsquo;selection&rsquo; is returned, which is a list with component:
</p>
<table role = "presentation">
<tr><td><code>method</code></td>
<td>
<p>posterior samples from the MCMC</p>
</td></tr>
<tr><td><code>indices</code></td>
<td>
<p>a list of indices and names of selected variables</p>
</td></tr>
<tr><td><code>summary</code></td>
<td>
<p>a summary of selected variables</p>
</td></tr>
</table>


<h3>References</h3>

<p>Ren, J., Zhou, F., Li, X., Ma, S., Jiang, Y. and Wu, C. (2023). Robust Bayesian variable selection for gene-environment interactions.
<em>Biometrics</em>, 79(2), 684-694  <a href="https://doi.org/10.1111/biom.13670">doi:10.1111/biom.13670</a>
</p>
<p>Barbieri, M.M. and Berger, J.O. (2004). Optimal predictive model selection. <em>Ann. Statist</em>, 32(3):870–897
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Blend">Blend</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dat)
## sparse
fit = Blend(y,x,t,J,kn,degree)
selected=selection(fit,sparse=TRUE)
selected


## non-sparse
fit = Blend(y,x,t,J,kn,degree,sparse="FALSE")
selected=selection(fit,sparse=FALSE)
selected


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
