<!DOCTYPE html><html lang="en"><head><title>Help for package blit</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {blit}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#blit-package'><p>blit: Bioinformatics Library for Integrated Tools</p></a></li>
<li><a href='#allele_counter'><p>Run alleleCount</p></a></li>
<li><a href='#arg'><p>Deliver arguments of command</p></a></li>
<li><a href='#cellranger'><p>Run cellranger</p></a></li>
<li><a href='#cmd_run'><p>Execute command</p></a></li>
<li><a href='#cmd_wd'><p>Define the environment when running the command</p></a></li>
<li><a href='#Command'><p>R6 Class to prepare command parameters</p></a></li>
<li><a href='#exec'><p>Invoke a System Command</p></a></li>
<li><a href='#fastq_pair'><p>FASTQ PAIR</p></a></li>
<li><a href='#gistic2'><p>Run GISTIC2</p></a></li>
<li><a href='#kraken_tools'><p>KrakenTools is a suite of scripts to be used alongside the Kraken,</p>
KrakenUniq, Kraken 2, or Bracken programs.</a></li>
<li><a href='#kraken2'><p>Running Kraken2</p></a></li>
<li><a href='#make_command'><p>Helper function to create new command</p></a></li>
<li><a href='#perl'><p>Perl is a highly capable, feature-rich programming language with over 36</p>
years of development.</a></li>
<li><a href='#pyscenic'><p>Run pyscenic</p></a></li>
<li><a href='#python'><p>Python is a programming language that lets you work quickly and integrate</p>
systems more effectively.</a></li>
<li><a href='#seqkit'><p>Run seqkit</p></a></li>
<li><a href='#trust4'><p>TRUST4: immune repertoire reconstruction from bulk and single-cell RNA-seq</p>
data</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Bioinformatics Library for Integrated Tools</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>An all-encompassing R toolkit designed to streamline the
    process of calling various bioinformatics software and then performing data
    analysis and visualization in R. With 'blit', users can easily integrate a
    wide array of bioinformatics command line tools into their workflows,
    leveraging the power of R for sophisticated data manipulation and graphical
    representation.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>cli, data.table, R6, rlang (&ge; 1.1.0), sys, utils, withr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>Unix System</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/WangLabCSU/blit">https://github.com/WangLabCSU/blit</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/WangLabCSU/blit/issues">https://github.com/WangLabCSU/blit/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-26 08:32:43 UTC; yun</td>
</tr>
<tr>
<td>Author:</td>
<td>Yun Peng <a href="https://orcid.org/0000-0003-2801-3332"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Shixiang Wang <a href="https://orcid.org/0000-0001-9855-7357"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Jennifer Lu [cph] (Author of the included scripts from Kraken2 and
    KrakenTools libraries),
  Li Song [cph] (Author of included scripts from TRUST4 library),
  X. Shirley Liu [cph] (Author of included scripts from TRUST4 library)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yun Peng &lt;yunyunp96@163.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-27 16:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='blit-package'>blit: Bioinformatics Library for Integrated Tools</h2><span id='topic+blit'></span><span id='topic+blit-package'></span>

<h3>Description</h3>

<p><img src="../help/figures/logo.png" style='float: right' alt='logo' width='120' />
</p>
<p>An all-encompassing R toolkit designed to streamline the process of calling various bioinformatics software and then performing data analysis and visualization in R. With 'blit', users can easily integrate a wide array of bioinformatics command line tools into their workflows, leveraging the power of R for sophisticated data manipulation and graphical representation.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Yun Peng <a href="mailto:yunyunp96@163.com">yunyunp96@163.com</a> (<a href="https://orcid.org/0000-0003-2801-3332">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Shixiang Wang <a href="mailto:w_shixiang@163.com">w_shixiang@163.com</a> (<a href="https://orcid.org/0000-0001-9855-7357">ORCID</a>)
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Jennifer Lu <a href="mailto:jennifer.lu717@gmail.com">jennifer.lu717@gmail.com</a> (Author of the included scripts from Kraken2 and KrakenTools libraries) [copyright holder]
</p>
</li>
<li><p> Li Song <a href="mailto:Li.Song@dartmouth.edu">Li.Song@dartmouth.edu</a> (Author of included scripts from TRUST4 library) [copyright holder]
</p>
</li>
<li><p> X. Shirley Liu <a href="mailto:xsliu@ds.dfci.harvard.edu">xsliu@ds.dfci.harvard.edu</a> (Author of included scripts from TRUST4 library) [copyright holder]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/WangLabCSU/blit">https://github.com/WangLabCSU/blit</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/WangLabCSU/blit/issues">https://github.com/WangLabCSU/blit/issues</a>
</p>
</li></ul>


<hr>
<h2 id='allele_counter'>Run alleleCount</h2><span id='topic+allele_counter'></span>

<h3>Description</h3>

<p>The <code>alleleCount</code> program primarily exists to prevent code duplication
between some other projects, specifically <code>AscatNGS</code> and <code>Battenberg</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>allele_counter(
  hts_file,
  loci_file,
  ofile,
  ...,
  odir = getwd(),
  alleleCounter = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="allele_counter_+3A_hts_file">hts_file</code></td>
<td>
<p>A string of path to sample HTS file.</p>
</td></tr>
<tr><td><code id="allele_counter_+3A_loci_file">loci_file</code></td>
<td>
<p>A string of path to loci file.</p>
</td></tr>
<tr><td><code id="allele_counter_+3A_ofile">ofile</code></td>
<td>
<p>A string of path to the output file.</p>
</td></tr>
<tr><td><code id="allele_counter_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dynamic dots</a>&gt; Additional arguments passed to <code>alleleCounter</code> command. Empty arguments are automatically trimmed. If a single argument, such as a file path, contains spaces, it must be quoted, for example using <code><a href="base.html#topic+shQuote">shQuote</a></code>. Details see: <code>cmd_help(alleleCounter())</code>.</p>
</td></tr>
<tr><td><code id="allele_counter_+3A_odir">odir</code></td>
<td>
<p>A string of path to the output directory.</p>
</td></tr>
<tr><td><code id="allele_counter_+3A_allelecounter">alleleCounter</code></td>
<td>
<p>A string of path to <code>alleleCounter</code> command.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>command</code> object.
</p>


<h3>See Also</h3>

<p><a href="https://github.com/cancerit/alleleCount">https://github.com/cancerit/alleleCount</a>
</p>

<hr>
<h2 id='arg'>Deliver arguments of command</h2><span id='topic+arg'></span>

<h3>Description</h3>

<p>Deliver arguments of command
</p>


<h3>Usage</h3>

<pre><code class='language-R'>arg(tag, value, indicator = FALSE, lgl2int = FALSE, format = "%s", sep = " ")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="arg_+3A_tag">tag</code></td>
<td>
<p>A string specifying argument tag, like &quot;-i&quot;, &quot;-o&quot;.</p>
</td></tr>
<tr><td><code id="arg_+3A_value">value</code></td>
<td>
<p>Value passed to the argument.</p>
</td></tr>
<tr><td><code id="arg_+3A_indicator">indicator</code></td>
<td>
<p>A logical value specifying whether value should be an
indicator of tag. If TRUE, logical value will explain the set or unset of
tag.</p>
</td></tr>
<tr><td><code id="arg_+3A_lgl2int">lgl2int</code></td>
<td>
<p>A logical value indicates whether transfrom value <code>TRUE</code> to
<code>1</code> or <code>FALSE</code> to <code>0</code>. If <code>TRUE</code>, format will always be set to <code>"%d"</code>.</p>
</td></tr>
<tr><td><code id="arg_+3A_format">format</code></td>
<td>
<p>The format of the value, details see <code><a href="base.html#topic+sprintf">sprintf</a></code>.</p>
</td></tr>
<tr><td><code id="arg_+3A_sep">sep</code></td>
<td>
<p>A character string used to separate &quot;tag&quot; and &quot;value&quot;, usually <code>" "</code> or <code>"="</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A string.
</p>

<hr>
<h2 id='cellranger'>Run cellranger</h2><span id='topic+cellranger'></span>

<h3>Description</h3>

<p>Run cellranger
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cellranger(subcmd = NULL, ..., cellranger = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cellranger_+3A_subcmd">subcmd</code></td>
<td>
<p>Sub-Command of cellranger.</p>
</td></tr>
<tr><td><code id="cellranger_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dynamic dots</a>&gt; Additional arguments passed to <code style="white-space: pre;">&#8288;cellranger subcmd&#8288;</code> command. Empty arguments are automatically trimmed. If a single argument, such as a file path, contains spaces, it must be quoted, for example using <code><a href="base.html#topic+shQuote">shQuote</a></code>. Details see: <code style="white-space: pre;">&#8288;cmd_help(cellranger subcmd())&#8288;</code>.</p>
</td></tr>
<tr><td><code id="cellranger_+3A_cellranger">cellranger</code></td>
<td>
<p>A string of path to <code>cellranger</code> command.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>command</code> object.
</p>

<hr>
<h2 id='cmd_run'>Execute command</h2><span id='topic+cmd_run'></span><span id='topic+cmd_background'></span><span id='topic+cmd_help'></span>

<h3>Description</h3>


<ul>
<li> <p><code>cmd_run</code>: Run the command.
</p>
</li>
<li> <p><code>cmd_background</code>: Run the command in the background.
</p>
</li>
<li> <p><code>cmd_help</code>: Print the help document for this command.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>cmd_run(
  command,
  stdout = TRUE,
  stderr = TRUE,
  stdin = NULL,
  timeout = NULL,
  verbose = TRUE
)

cmd_background(
  command,
  stdout = NULL,
  stderr = NULL,
  stdin = NULL,
  verbose = TRUE
)

cmd_help(command, stdout = NULL, stderr = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cmd_run_+3A_command">command</code></td>
<td>
<p>A <code>command</code> object.</p>
</td></tr>
<tr><td><code id="cmd_run_+3A_stdout">stdout</code>, <code id="cmd_run_+3A_stderr">stderr</code></td>
<td>
<p>How output streams of the child process are processed.
Possible values are:
</p>

<ul>
<li> <p><code>TRUE</code>: print the child output in R console.
</p>
</li>
<li> <p><code>FALSE</code>: suppress output stream
</p>
</li>
<li> <p><strong>string</strong>: name or path of file to redirect output
</p>
</li>
<li> <p><code>connection</code>: a writable R <code><a href="base.html#topic+connection">connection</a></code> object
</p>
</li>
<li> <p><code>function</code>: a callback function (including purrr-like lambda syntax) with
one argument accepting a raw vector (use <code><a href="sys.html#topic+as_text">as_text()</a></code> to
convert to text).
</p>
</li></ul>

<p>For <code>cmd_background()</code>, only a string (file path), or a single boolean value
can be used.
</p>
<p>For <code>cmd_help()</code>, only a string (file path), <code>connection</code>, or <code>function</code> can
be used.</p>
</td></tr>
<tr><td><code id="cmd_run_+3A_stdin">stdin</code></td>
<td>
<p>should the input be diverted? A character string naming a file.</p>
</td></tr>
<tr><td><code id="cmd_run_+3A_timeout">timeout</code></td>
<td>
<p>Timeout in seconds. This is a limit for the elapsed time
running command in the separate process.</p>
</td></tr>
<tr><td><code id="cmd_run_+3A_verbose">verbose</code></td>
<td>
<p>A single boolean value indicating whether the command
execution should be verbose.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code>cmd_run</code>: Exit status invisiblely.
</p>
</li></ul>


<ul>
<li> <p><code>cmd_background</code>: Returns the process ID, which can be terminated manually
using <code><a href="tools.html#topic+pskill">tools::pskill()</a></code>. You can also use <code><a href="sys.html#topic+exec">sys::exec_status()</a></code> to check
the process's exit status.
</p>
</li></ul>


<ul>
<li> <p><code>cmd_help</code>: the input <code>command</code> invisiblely.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+cmd_wd">cmd_wd()</a></code>/<code><a href="#topic+cmd_envvar">cmd_envvar()</a></code>/<code><a href="#topic+cmd_envpath">cmd_envpath()</a></code>
</p>

<hr>
<h2 id='cmd_wd'>Define the environment when running the command</h2><span id='topic+cmd_wd'></span><span id='topic+cmd_envvar'></span><span id='topic+cmd_envpath'></span>

<h3>Description</h3>


<ul>
<li> <p><code>cmd_wd</code>: define the working directory.
</p>
</li>
<li> <p><code>cmd_envvar</code>: define the environment variables.
</p>
</li>
<li> <p><code>cmd_envpath</code>: define the <code>PATH</code>-like environment variables.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>cmd_wd(command, wd = NULL)

cmd_envvar(command, ..., action = "replace", sep = " ")

cmd_envpath(command, ..., action = "prefix", name = "PATH")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cmd_wd_+3A_command">command</code></td>
<td>
<p>A <code>command</code> object.</p>
</td></tr>
<tr><td><code id="cmd_wd_+3A_wd">wd</code></td>
<td>
<p>A string or <code>NULL</code> define the working directory of the command.</p>
</td></tr>
<tr><td><code id="cmd_wd_+3A_...">...</code></td>
<td>

<ul>
<li> <p><code>cmd_envvar</code>: Named character define the environment variables.
</p>
</li>
<li> <p><code>cmd_envpath</code>: Unnamed character to define the <code>PATH</code>-like environment
variables <code>name</code>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="cmd_wd_+3A_action">action</code></td>
<td>
<p>Should the new values <code>"replace"</code>, <code>"prefix"</code> or <code>"suffix"</code>
existing environment variables?</p>
</td></tr>
<tr><td><code id="cmd_wd_+3A_sep">sep</code></td>
<td>
<p>A string to separate new and old value when <code>action</code> is <code>"prefix"</code>
or <code>"suffix"</code>.</p>
</td></tr>
<tr><td><code id="cmd_wd_+3A_name">name</code></td>
<td>
<p>A string define the PATH environment variable name. You
can use this to define other <code>PATH</code>-like environment variable such as
<code>PYTHONPATH</code>.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code>cmd_wd</code>: The <code>command</code> object itself, with working directory updated.
</p>
</li>
<li> <p><code>cmd_envvar</code>: The <code>command</code> object itself, with running environment
variable updated.
</p>
</li>
<li> <p><code>cmd_envpath</code>: The <code>command</code> object self, with running environment variable
<code>name</code> updated.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+cmd_run">cmd_run()</a></code>/<code><a href="#topic+cmd_background">cmd_background()</a></code>/<code><a href="#topic+cmd_help">cmd_help()</a></code>
</p>

<hr>
<h2 id='Command'>R6 Class to prepare command parameters</h2><span id='topic+Command'></span>

<h3>Description</h3>

<p>R6 Class to prepare command parameters
</p>
<p>R6 Class to prepare command parameters
</p>


<h3>Methods</h3>



<h4>Public methods</h4>


<ul>
<li> <p><a href="#method-Command-new"><code>Command$new()</code></a>
</p>
</li>
<li> <p><a href="#method-Command-evaluate"><code>Command$evaluate()</code></a>
</p>
</li>
<li> <p><a href="#method-Command-build"><code>Command$build()</code></a>
</p>
</li>
<li> <p><a href="#method-Command-print"><code>Command$print()</code></a>
</p>
</li>
<li> <p><a href="#method-Command-clone"><code>Command$clone()</code></a>
</p>
</li></ul>


<hr>
<a id="method-Command-new"></a>



<h4>Method <code>new()</code></h4>

<p>Create a new <code>Command</code> object.
</p>


<h5>Usage</h5>

<div class="r"><pre>Command$new(..., .subcmd = NULL)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>...</code></dt><dd><p>Additional argument passed into command.</p>
</dd>
<dt><code>.subcmd</code></dt><dd><p>Sub-command string.</p>
</dd>
</dl>

</div>


<hr>
<a id="method-Command-evaluate"></a>



<h4>Method <code>evaluate()</code></h4>

<p>Evaluate the parameters to execute command.
</p>


<h5>Usage</h5>

<div class="r"><pre>Command$evaluate()</pre></div>



<h5>Returns</h5>

<p>The object itself.
</p>


<hr>
<a id="method-Command-build"></a>



<h4>Method <code>build()</code></h4>

<p>Build parameters to run command.
</p>


<h5>Usage</h5>

<div class="r"><pre>Command$build(help = FALSE, verbose = TRUE, envir = caller_env())</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>help</code></dt><dd><p>A boolean value indicating whether to build parameters
for help document or not.</p>
</dd>
<dt><code>verbose</code></dt><dd><p>A boolean value indicating whether the command
execution should be verbose.</p>
</dd>
<dt><code>envir</code></dt><dd><p>An environment used to Execute command.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>An atomic character combine the command and parameters.
</p>


<hr>
<a id="method-Command-print"></a>



<h4>Method <code>print()</code></h4>

<p>Build parameters to run command.
</p>


<h5>Usage</h5>

<div class="r"><pre>Command$print(indent = NULL)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>indent</code></dt><dd><p>A single integer number giving the space of indent.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>The object itself.
</p>


<hr>
<a id="method-Command-clone"></a>



<h4>Method <code>clone()</code></h4>

<p>The objects of this class are cloneable with this method.
</p>


<h5>Usage</h5>

<div class="r"><pre>Command$clone(deep = FALSE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>deep</code></dt><dd><p>Whether to make a deep clone.</p>
</dd>
</dl>

</div>



<hr>
<h2 id='exec'>Invoke a System Command</h2><span id='topic+exec'></span>

<h3>Description</h3>

<p>Invoke a System Command
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exec(cmd, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="exec_+3A_cmd">cmd</code></td>
<td>
<p>Command to be invoked, as a character string.</p>
</td></tr>
<tr><td><code id="exec_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dynamic dots</a>&gt; Additional arguments passed to <code>cmd</code> command. Empty arguments are automatically trimmed. If a single argument, such as a file path, contains spaces, it must be quoted, for example using <code><a href="base.html#topic+shQuote">shQuote</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>command</code> object.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+cmd_wd">cmd_wd()</a></code>/<code><a href="#topic+cmd_envvar">cmd_envvar()</a></code>/<code><a href="#topic+cmd_envpath">cmd_envpath()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+cmd_run">cmd_run()</a></code>/<code><a href="#topic+cmd_background">cmd_background()</a></code>/<code><a href="#topic+cmd_help">cmd_help()</a></code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>cmd_run(exec("echo", "$PATH"))
</code></pre>

<hr>
<h2 id='fastq_pair'>FASTQ PAIR</h2><span id='topic+fastq_pair'></span><span id='topic+fastq_read_pair'></span>

<h3>Description</h3>

<p>Rewrite paired end fastq files to make sure that all reads have a mate and to
separate out singletons.
</p>
<p>Usually when you get paired end read files you have two files with a /1
sequence in one and a /2 sequence in the other (or a /f and /r or just two
reads with the same ID). However, often when working with files from a third
party source (e.g. the SRA) there are different numbers of reads in each file
(because some reads fail QC). Spades, bowtie2 and other tools break because
they demand paired end files have the same number of reads.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fastq_pair(
  fq1,
  fq2,
  ...,
  hash_table_size = NULL,
  max_hash_table_size = NULL,
  fastq_pair = NULL
)

fastq_read_pair(fastq_files)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fastq_pair_+3A_fq1">fq1</code>, <code id="fastq_pair_+3A_fq2">fq2</code></td>
<td>
<p>A string of fastq file path.</p>
</td></tr>
<tr><td><code id="fastq_pair_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dynamic dots</a>&gt; Additional arguments passed to <code>fastq_pair</code> command. Empty arguments are automatically trimmed. If a single argument, such as a file path, contains spaces, it must be quoted, for example using <code><a href="base.html#topic+shQuote">shQuote</a></code>. Details see: <code>cmd_help(fastq_pair())</code>.</p>
</td></tr>
<tr><td><code id="fastq_pair_+3A_hash_table_size">hash_table_size</code></td>
<td>
<p>Size of hash table to use.</p>
</td></tr>
<tr><td><code id="fastq_pair_+3A_max_hash_table_size">max_hash_table_size</code></td>
<td>
<p>Maximal hash table size to use.</p>
</td></tr>
<tr><td><code id="fastq_pair_+3A_fastq_pair">fastq_pair</code></td>
<td>
<p>A string of path to <code>fastq_pair</code> command.</p>
</td></tr>
<tr><td><code id="fastq_pair_+3A_fastq_files">fastq_files</code></td>
<td>
<p>A character of the fastq file paths.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>command</code> object.
</p>


<h3>See Also</h3>

<p><a href="https://github.com/linsalrob/fastq-pair">https://github.com/linsalrob/fastq-pair</a>
</p>

<hr>
<h2 id='gistic2'>Run GISTIC2</h2><span id='topic+gistic2'></span>

<h3>Description</h3>

<p>The GISTIC module identifies regions of the genome that are
significantly amplified or deleted across a set of samples. Each aberration
is assigned a G-score that considers the amplitude of the aberration as well
as the frequency of its occurrence across samples. False Discovery Rate
q-values are then calculated for the aberrant regions, and regions with
q-values below a user-defined threshold are considered significant. For each
significant region, a &quot;peak region&quot; is identified, which is the part of the
aberrant region with greatest amplitude and frequency of alteration. In
addition, a &quot;wide peak&quot; is determined using a leave-one-out algorithm to
allow for errors in the boundaries in a single sample. The &quot;wide peak&quot;
boundaries are more robust for identifying the most likely gene targets in
the region. Each significantly aberrant region is also tested to determine
whether it results primarily from broad events (longer than half a chromosome
arm), focal events, or significant levels of both. The GISTIC module reports
the genomic locations and calculated q-values for the aberrant regions. It
identifies the samples that exhibit each significant amplification or
deletion, and it lists genes found in each &quot;wide peak&quot; region.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gistic2(seg, refgene, ..., odir = getwd(), gistic2 = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gistic2_+3A_seg">seg</code></td>
<td>
<p>A data.frame of segmented data.</p>
</td></tr>
<tr><td><code id="gistic2_+3A_refgene">refgene</code></td>
<td>
<p>Path to reference genome data input file (REQUIRED, see below
for file description).</p>
</td></tr>
<tr><td><code id="gistic2_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dynamic dots</a>&gt; Additional arguments passed to <code>gistic2</code> command. Empty arguments are automatically trimmed. If a single argument, such as a file path, contains spaces, it must be quoted, for example using <code><a href="base.html#topic+shQuote">shQuote</a></code>. Details see: <code>cmd_help(gistic2())</code>.</p>
</td></tr>
<tr><td><code id="gistic2_+3A_odir">odir</code></td>
<td>
<p>A string of path to the output directory.</p>
</td></tr>
<tr><td><code id="gistic2_+3A_gistic2">gistic2</code></td>
<td>
<p>A string of path to <code>gistic2</code> command.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>command</code> object.
</p>


<h3>See Also</h3>

<p><a href="https://broadinstitute.github.io/gistic2/">https://broadinstitute.github.io/gistic2/</a>
</p>

<hr>
<h2 id='kraken_tools'>KrakenTools is a suite of scripts to be used alongside the Kraken,
KrakenUniq, Kraken 2, or Bracken programs.</h2><span id='topic+kraken_tools'></span>

<h3>Description</h3>

<p>These scripts are designed to help Kraken users with downstream
analysis of Kraken results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kraken_tools(script, ..., python = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="kraken_tools_+3A_script">script</code></td>
<td>
<p>Name of the kraken2 script. One of
<code>"combine_kreports"</code>, <code>"combine_mpa"</code>, <code>"extract_kraken_reads"</code>, <code>"filter_bracken_out"</code>, <code>"fix_unmapped"</code>, <code>"kreport2krona"</code>, <code>"kreport2mpa"</code>, <code>"make_kreport"</code>, and <code>"make_ktaxonomy"</code>.</p>
</td></tr>
<tr><td><code id="kraken_tools_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dynamic dots</a>&gt; Additional arguments passed to <code>kraken_tools</code> command. Empty arguments are automatically trimmed. If a single argument, such as a file path, contains spaces, it must be quoted, for example using <code><a href="base.html#topic+shQuote">shQuote</a></code>. Details see: <code>cmd_help(kraken_tools())</code>.</p>
</td></tr>
<tr><td><code id="kraken_tools_+3A_python">python</code></td>
<td>
<p>A string of path to <code>python</code> command.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>command</code> object.
</p>


<h3>See Also</h3>

<p><a href="https://github.com/jenniferlu717/KrakenTools">https://github.com/jenniferlu717/KrakenTools</a>
</p>

<hr>
<h2 id='kraken2'>Running Kraken2</h2><span id='topic+kraken2'></span>

<h3>Description</h3>

<p>Kraken is a taxonomic sequence classifier that assigns taxonomic labels to
DNA sequences. Kraken examines the k-mers within a query sequence and uses
the information within those k-mers to query a database. That database maps
k-mers to the lowest common ancestor (LCA) of all genomes known to contain a
given k-mer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kraken2(
  fq1,
  ...,
  fq2 = NULL,
  ofile = "kraken_output.txt",
  report = "kraken_report.txt",
  classified_out = NULL,
  unclassified_out = NULL,
  odir = getwd(),
  kraken2 = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="kraken2_+3A_fq1">fq1</code>, <code id="kraken2_+3A_fq2">fq2</code></td>
<td>
<p>A string of fastq file path.</p>
</td></tr>
<tr><td><code id="kraken2_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dynamic dots</a>&gt; Additional arguments passed to <code>kraken2</code> command. Empty arguments are automatically trimmed. If a single argument, such as a file path, contains spaces, it must be quoted, for example using <code><a href="base.html#topic+shQuote">shQuote</a></code>. Details see: <code>cmd_help(kraken2())</code>.</p>
</td></tr>
<tr><td><code id="kraken2_+3A_ofile">ofile</code></td>
<td>
<p>A string of path to save kraken2 output.</p>
</td></tr>
<tr><td><code id="kraken2_+3A_report">report</code></td>
<td>
<p>A string of path to save kraken2 report.</p>
</td></tr>
<tr><td><code id="kraken2_+3A_classified_out">classified_out</code></td>
<td>
<p>A string of path to save classified sequences, which
should be a fastq file.</p>
</td></tr>
<tr><td><code id="kraken2_+3A_unclassified_out">unclassified_out</code></td>
<td>
<p>A string of path to save unclassified sequences,
which should be a fastq file.</p>
</td></tr>
<tr><td><code id="kraken2_+3A_odir">odir</code></td>
<td>
<p>A string of path to the output directory.</p>
</td></tr>
<tr><td><code id="kraken2_+3A_kraken2">kraken2</code></td>
<td>
<p>A string of path to <code>kraken2</code> command.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>command</code> object.
</p>


<h3>See Also</h3>


<ul>
<li> <p><a href="https://github.com/DerrickWood/kraken2/wiki/Manual">https://github.com/DerrickWood/kraken2/wiki/Manual</a>
</p>
</li>
<li> <p><a href="https://benlangmead.github.io/aws-indexes/k2">https://benlangmead.github.io/aws-indexes/k2</a>
</p>
</li></ul>


<hr>
<h2 id='make_command'>Helper function to create new command</h2><span id='topic+make_command'></span>

<h3>Description</h3>

<p>Helper function to create new command
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_command(name, fun, envir = caller_env())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="make_command_+3A_name">name</code></td>
<td>
<p>A string of the function name.</p>
</td></tr>
<tr><td><code id="make_command_+3A_fun">fun</code></td>
<td>
<p>A function used to initialize the <code>Command</code> object.</p>
</td></tr>
<tr><td><code id="make_command_+3A_envir">envir</code></td>
<td>
<p>A environment used to bind the created function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A function.
</p>

<hr>
<h2 id='perl'>Perl is a highly capable, feature-rich programming language with over 36
years of development.</h2><span id='topic+perl'></span>

<h3>Description</h3>

<p>Perl is a highly capable, feature-rich programming language with over 36
years of development.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perl(..., perl = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="perl_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dynamic dots</a>&gt; Additional arguments passed to <code>perl</code> command. Empty arguments are automatically trimmed. If a single argument, such as a file path, contains spaces, it must be quoted, for example using <code><a href="base.html#topic+shQuote">shQuote</a></code>. Details see: <code>cmd_help(perl())</code>.</p>
</td></tr>
<tr><td><code id="perl_+3A_perl">perl</code></td>
<td>
<p>A string of path to <code>perl</code> command.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>command</code> object.
</p>


<h3>See Also</h3>

<p><a href="https://www.perl.org/">https://www.perl.org/</a>
</p>

<hr>
<h2 id='pyscenic'>Run pyscenic</h2><span id='topic+pyscenic'></span>

<h3>Description</h3>

<p>Run pyscenic
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pyscenic(subcmd = NULL, ..., pyscenic = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pyscenic_+3A_subcmd">subcmd</code></td>
<td>
<p>Sub-Command of pyscenic.</p>
</td></tr>
<tr><td><code id="pyscenic_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dynamic dots</a>&gt; Additional arguments passed to <code style="white-space: pre;">&#8288;pyscenic subcmd&#8288;</code> command. Empty arguments are automatically trimmed. If a single argument, such as a file path, contains spaces, it must be quoted, for example using <code><a href="base.html#topic+shQuote">shQuote</a></code>. Details see: <code style="white-space: pre;">&#8288;cmd_help(pyscenic subcmd())&#8288;</code>.</p>
</td></tr>
<tr><td><code id="pyscenic_+3A_pyscenic">pyscenic</code></td>
<td>
<p>A string of path to <code>pyscenic</code> command.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>command</code> object.
</p>


<h3>References</h3>

<p><a href="https://github.com/aertslab/pySCENIC">https://github.com/aertslab/pySCENIC</a>
</p>

<hr>
<h2 id='python'>Python is a programming language that lets you work quickly and integrate
systems more effectively.</h2><span id='topic+python'></span>

<h3>Description</h3>

<p>Python is a programming language that lets you work quickly and integrate
systems more effectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>python(..., python = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="python_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dynamic dots</a>&gt; Additional arguments passed to <code>python</code> command. Empty arguments are automatically trimmed. If a single argument, such as a file path, contains spaces, it must be quoted, for example using <code><a href="base.html#topic+shQuote">shQuote</a></code>. Details see: <code>cmd_help(python())</code>.</p>
</td></tr>
<tr><td><code id="python_+3A_python">python</code></td>
<td>
<p>A string of path to <code>python</code> command.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>command</code> object.
</p>


<h3>See Also</h3>

<p><a href="https://www.python.org/">https://www.python.org/</a>
</p>

<hr>
<h2 id='seqkit'>Run seqkit</h2><span id='topic+seqkit'></span>

<h3>Description</h3>

<p>Run seqkit
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seqkit(subcmd = NULL, ..., seqkit = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="seqkit_+3A_subcmd">subcmd</code></td>
<td>
<p>Sub-Command of seqkit.</p>
</td></tr>
<tr><td><code id="seqkit_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dynamic dots</a>&gt; Additional arguments passed to <code style="white-space: pre;">&#8288;seqkit subcmd&#8288;</code> command. Empty arguments are automatically trimmed. If a single argument, such as a file path, contains spaces, it must be quoted, for example using <code><a href="base.html#topic+shQuote">shQuote</a></code>. Details see: <code style="white-space: pre;">&#8288;cmd_help(seqkit subcmd())&#8288;</code>.</p>
</td></tr>
<tr><td><code id="seqkit_+3A_seqkit">seqkit</code></td>
<td>
<p>A string of path to <code>seqkit</code> command.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>command</code> object.
</p>


<h3>See Also</h3>

<p><a href="https://bioinf.shenwei.me/seqkit/">https://bioinf.shenwei.me/seqkit/</a>
</p>

<hr>
<h2 id='trust4'>TRUST4: immune repertoire reconstruction from bulk and single-cell RNA-seq
data</h2><span id='topic+trust4'></span><span id='topic+trust4_imgt_annot'></span><span id='topic+trust4_gene_names'></span>

<h3>Description</h3>

<p>TRUST4: immune repertoire reconstruction from bulk and single-cell RNA-seq
data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trust4(
  file1,
  ref_coordinate,
  ...,
  file2 = NULL,
  mode = NULL,
  ref_annot = NULL,
  ofile = NULL,
  odir = getwd(),
  trust4 = NULL
)

trust4_imgt_annot(
  species = "Homo_sapien",
  ...,
  ofile = "IMGT+C.fa",
  odir = getwd(),
  perl = NULL
)

trust4_gene_names(imgt_annot, ofile = "bcr_tcr_gene_name.txt", odir = getwd())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="trust4_+3A_file1">file1</code></td>
<td>
<p>Path to bam file or fastq file.</p>
</td></tr>
<tr><td><code id="trust4_+3A_ref_coordinate">ref_coordinate</code></td>
<td>
<p>Path to the fasta file coordinate and sequence of
V/D/J/C genes.</p>
</td></tr>
<tr><td><code id="trust4_+3A_...">...</code></td>
<td>

<ul>
<li> <p><code>trust4</code>: &lt;<a href="rlang.html#topic+dyn-dots">dynamic dots</a>&gt; Additional arguments passed to <code>run-trust4</code> command. Empty arguments are automatically trimmed. If a single argument, such as a file path, contains spaces, it must be quoted, for example using <code><a href="base.html#topic+shQuote">shQuote</a></code>. Details see: <code>cmd_help(run-trust4())</code>.
</p>
</li>
<li> <p><code>trust4_imgt_annot</code>: &lt;<a href="rlang.html#topic+dyn-dots">dynamic dots</a>&gt; Additional arguments passed to <code>trust4_imgt_annot</code> command. Empty arguments are automatically trimmed. If a single argument, such as a file path, contains spaces, it must be quoted, for example using <code><a href="base.html#topic+shQuote">shQuote</a></code>. Details see: <code>cmd_help(trust4_imgt_annot())</code>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="trust4_+3A_file2">file2</code></td>
<td>
<p>Path to the second paired-end read fastq file, only used for
<code>mode = "fastq"</code>.</p>
</td></tr>
<tr><td><code id="trust4_+3A_mode">mode</code></td>
<td>
<p>One of &quot;bam&quot; or &quot;fastq&quot;. If <code>NULL</code>, will be inferred from
<code>file1</code>.</p>
</td></tr>
<tr><td><code id="trust4_+3A_ref_annot">ref_annot</code></td>
<td>
<p>Path to detailed V/D/J/C gene reference file, such as from
IMGT database. (default: not used). (recommended).</p>
</td></tr>
<tr><td><code id="trust4_+3A_ofile">ofile</code></td>
<td>

<ul>
<li> <p><code>trust4</code>: Prefix of output files. (default: inferred from file prefix).
</p>
</li>
<li> <p><code>trust4_imgt_annot</code>: Output file name.
</p>
</li>
<li> <p><code>trust4_gene_names</code>: Output file name.
</p>
</li></ul>
</td></tr>
<tr><td><code id="trust4_+3A_odir">odir</code></td>
<td>
<p>A string of path to the output directory.</p>
</td></tr>
<tr><td><code id="trust4_+3A_trust4">trust4</code></td>
<td>
<p>A string of path to <code>run-trust4</code> command.</p>
</td></tr>
<tr><td><code id="trust4_+3A_species">species</code></td>
<td>
<p>Species to extract IMGT annotation, details see
<a href="https://www.imgt.org//download/V-QUEST/IMGT_V-QUEST_reference_directory/">https://www.imgt.org//download/V-QUEST/IMGT_V-QUEST_reference_directory/</a>.</p>
</td></tr>
<tr><td><code id="trust4_+3A_perl">perl</code></td>
<td>
<p>A string of path to <code>perl</code> command.</p>
</td></tr>
<tr><td><code id="trust4_+3A_imgt_annot">imgt_annot</code></td>
<td>
<p>Path of IMGT annotation file, created via
<code>trust4_imgt_annot</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>command</code> object.
</p>


<h3>See Also</h3>

<p><a href="https://github.com/liulab-dfci/TRUST4">https://github.com/liulab-dfci/TRUST4</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
