<!DOCTYPE html><html><head><title>Help for package bmass</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {bmass}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bmass'><p>Bayesian multivariate analysis of summary statistics</p>
(<code>bmass</code>)</a></li>
<li><a href='#bmass_SimulatedData1'><p>bmass Simulated Dataset 1</p></a></li>
<li><a href='#bmass_SimulatedData2'><p>bmass Simulated Dataset 2</p></a></li>
<li><a href='#bmass_SimulatedSigSNPs'><p>bmass Simulated GWAS SNPs</p></a></li>
<li><a href='#bmass_TestData1'><p>bmass Test Dataset 1</p></a></li>
<li><a href='#bmass_TestData2'><p>bmass Test Dataset 2</p></a></li>
<li><a href='#bmass_TestSigSNPs'><p>bmass Test GWAS SNPs</p></a></li>
<li><a href='#GetMarginalPosteriors'><p>Get Marginal {<b>U</b>,<b>D</b>,<b>I</b>} Posteriors</p></a></li>
<li><a href='#GetModelPriorMatrix'><p>Get Model Prior Matrix</p></a></li>
<li><a href='#GetTopModelsPerSNPViaPosteriors'><p>Get Top Multivariate Models</p></a></li>
<li><a href='#GlobalLipids2013.GWASsnps'><p>GlobalLipids2013 GWAS SNPs</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-05-15</td>
</tr>
<tr>
<td>Title:</td>
<td>Bayesian Multivariate Analysis of Summary Statistics</td>
</tr>
<tr>
<td>Description:</td>
<td>Multivariate tool for analyzing genome-wide association
    study results in the form of univariate summary statistics. The 
    goal of 'bmass' is to comprehensively test all possible multivariate
    models given the phenotypes and datasets provided. Multivariate
    models are determined by assigning each phenotype to being either
    Unassociated (U), Directly associated (D) or Indirectly associated
    (I) with the genetic variant of interest. Test results for each model 
    are presented in the form of Bayes factors, thereby allowing direct
    comparisons between models. The underlying framework implemented
    here is based on the modeling developed in "A Unified Framework 
    for Association Analysis with Multiple Related Phenotypes",
    M. Stephens (2013) &lt;<a href="https://doi.org/10.1371%2Fjournal.pone.0065245">doi:10.1371/journal.pone.0065245</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/mturchin20/bmass">https://github.com/mturchin20/bmass</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/mturchin20/bmass/issues">https://github.com/mturchin20/bmass/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>utils, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-05-16 16:23:07 UTC; mturchin20</td>
</tr>
<tr>
<td>Author:</td>
<td>Michael Turchin [aut, cre],
  Matthew Stephens [aut],
  Peter Carbonetto [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Michael Turchin &lt;mturchin20@uchicago.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-05-17 07:20:12 UTC</td>
</tr>
</table>
<hr>
<h2 id='bmass'>Bayesian multivariate analysis of summary statistics 
(<code>bmass</code>)</h2><span id='topic+bmass'></span>

<h3>Description</h3>

<p>Run <code>bmass</code> on a set of phenotypes that each have
univariate GWAS statistics on the same set of SNPs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bmass(DataSources, GWASsnps = NULL,
  SNPMarginalUnivariateThreshold = 1e-06,
  SNPMarginalMultivariateThreshold = 1e-06, GWASThreshFlag = TRUE,
  GWASThreshValue = 5e-08, NminThreshold = 0,
  PrintMergedData = FALSE, PrintProgress = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bmass_+3A_datasources">DataSources</code></td>
<td>
<p>A string indicating the variable names of the
input datafiles and phenotypes. No default value.</p>
</td></tr>
<tr><td><code id="bmass_+3A_gwassnps">GWASsnps</code></td>
<td>
<p>A data.table containing rows of SNPs that were
univariate genome-wide significant in the phenotypes being used for
analysis; <code>GWASsnps</code> input file should have two columns, one 
for chromosome and another for basepair position (with column headers
of <code>Chr</code> and <code>BP</code>). No default value.</p>
</td></tr>
<tr><td><code id="bmass_+3A_snpmarginalunivariatethreshold">SNPMarginalUnivariateThreshold</code></td>
<td>
<p>A numerical value indicating
the univariate p-value threshold to use when collecting marginally
significant SNPs for final <code>bmass</code> analysis. Default is 
<code>1e-6</code>.</p>
</td></tr>
<tr><td><code id="bmass_+3A_snpmarginalmultivariatethreshold">SNPMarginalMultivariateThreshold</code></td>
<td>
<p>A numerical value
indicating the basic multivariate p-value threshold to use when
collecting marginally significant SNPs for final <code>bmass</code> 
analysis. Default is <code>1e-6</code>.</p>
</td></tr>
<tr><td><code id="bmass_+3A_gwasthreshflag">GWASThreshFlag</code></td>
<td>
<p>A logical <code>TRUE</code>/<code>FALSE</code> flag that 
indicates whether to threshold input <code>GWASsnps</code> list by a 
univariate GWAS p-value or not (eg the input <code>GWASsnps</code> list 
contains variants that are significant from discovery + replication 
data, but the input summary statistics are just from the discovery 
cohort). Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="bmass_+3A_gwasthreshvalue">GWASThreshValue</code></td>
<td>
<p>A numerical value indicating the univariate
p-value threshold to use in conjunction with the <code>GWASThreshFlag</code>.
Default is <code>5e-8</code>.</p>
</td></tr>
<tr><td><code id="bmass_+3A_nminthreshold">NminThreshold</code></td>
<td>
<p>A numerical value that indicates a sample size
threshold to use where SNPs below which are removed. Default is
<code>0</code>.</p>
</td></tr>
<tr><td><code id="bmass_+3A_printmergeddata">PrintMergedData</code></td>
<td>
<p>A logical <code>TRUE</code>/<code>FALSE</code> flag that 
indicates whether the intermediary 'merged datafile' should be included 
in the final <code>bmass</code> output; this file combines all the phenotypes 
for every SNP provided just prior to thresholding for marginally
significant SNPs. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="bmass_+3A_printprogress">PrintProgress</code></td>
<td>
<p>A logical <code>TRUE</code>/<code>FALSE</code> flag that 
indicates whether progress statements should be printed to 
<code>stderr</code> during the course of running <code>bmass</code> or not.
Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="bmass_+3A_...">...</code></td>
<td>
<p>Additional optional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing model, SNP, and posterior information for
both the previously significant univariate SNPs (<code>PreviousSNPs</code>)
and the newly significant multivariate SNPs (<code>NewSNPs</code>). For a 
full breakdown of the <code>bmass</code> output list structure, please see 
the associated vignettes.
</p>


<h3>Other Examples</h3>

<p><code>bmass(c("HDL","LDL","TG","TC"), GWASsnps, NminThreshold = 50000)</code>
<code>bmass(c("HDL","LDL","TG","TC"), GWASsnps, GWASThreshValue = 1e-8,
  NminThreshold = 50000, PrintProgress = TRUE)</code> 
<code>bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps, GWASThreshFlag = FALSE,
  SNPMarginalUnivariateThreshold = 1e-4,
  SNPMarginalMultivariateThreshold = 1e-4,
  PrintMergedData = TRUE)</code> 
<code>bmassOutput &lt;- bmass(c("HDL","LDL","TG","TC"),
  GWASsnps, NminThreshold = 50000)</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Phenotypes &lt;- c("bmass_SimulatedData1", "bmass_SimulatedData2")
bmassOutput &lt;- bmass(Phenotypes, bmass_SimulatedSigSNPs)
summary(bmassOutput)
bmassOutput$NewSNPs$SNPs

</code></pre>

<hr>
<h2 id='bmass_SimulatedData1'>bmass Simulated Dataset 1</h2><span id='topic+bmass_SimulatedData1'></span>

<h3>Description</h3>

<p>A manually created sample dataset for use in Roxygen2
documents and vignettes.
</p>


<h3>Format</h3>

<p>A data frame with 11 rows and 9 variables:
</p>

<dl>
<dt>Chr</dt><dd><p>chromosome</p>
</dd>
<dt>BP</dt><dd><p>basepair position</p>
</dd>
<dt>Marker</dt><dd><p>rsID# or other identifier</p>
</dd>
<dt>MAF</dt><dd><p>Minor Allele Frequency</p>
</dd>
<dt>A1</dt><dd><p>reference allele</p>
</dd>
<dt>A2</dt><dd><p>alternative allele</p>
</dd>
<dt>Direction</dt><dd><p>direction of association
effect size, + or -</p>
</dd>
<dt>pValue</dt><dd><p>p-Value of GWAS association</p>
</dd>
<dt>N</dt><dd><p>sample size</p>
</dd>
</dl>


<h3>Source</h3>

<p>Manually created
</p>

<hr>
<h2 id='bmass_SimulatedData2'>bmass Simulated Dataset 2</h2><span id='topic+bmass_SimulatedData2'></span>

<h3>Description</h3>

<p>A manually created sample dataset for use in Roxygen2
documents and vignettes.
</p>


<h3>Format</h3>

<p>A data frame with 11 rows and 9 variables:
</p>

<dl>
<dt>Chr</dt><dd><p>chromosome</p>
</dd>
<dt>BP</dt><dd><p>basepair position</p>
</dd>
<dt>Marker</dt><dd><p>rsID# or other identifier</p>
</dd>
<dt>MAF</dt><dd><p>Minor Allele Frequency</p>
</dd>
<dt>A1</dt><dd><p>reference allele</p>
</dd>
<dt>A2</dt><dd><p>alternative allele</p>
</dd>
<dt>Direction</dt><dd><p>direction of association
effect size, + or -</p>
</dd>
<dt>pValue</dt><dd><p>p-Value of GWAS association</p>
</dd>
<dt>N</dt><dd><p>sample size</p>
</dd>
</dl>


<h3>Source</h3>

<p>Manually created
</p>

<hr>
<h2 id='bmass_SimulatedSigSNPs'>bmass Simulated GWAS SNPs</h2><span id='topic+bmass_SimulatedSigSNPs'></span>

<h3>Description</h3>

<p>A manually created list of GWAS significant SNPs to be
used in conjunction with 'bmass_SimulatedData1' and 'bmass_SimulatedData2'.
</p>


<h3>Format</h3>

<p>A data frame with 2 rows and 2 variables:
</p>

<dl>
<dt>Chr</dt><dd><p>chromosome</p>
</dd>
<dt>BP</dt><dd><p>basepair position</p>
</dd>
</dl>


<h3>Source</h3>

<p>Manually created
</p>

<hr>
<h2 id='bmass_TestData1'>bmass Test Dataset 1</h2><span id='topic+bmass_TestData1'></span>

<h3>Description</h3>

<p>A manually created sample dataset for use in unit
tests.
</p>


<h3>Format</h3>

<p>A data frame with 11 rows and 9 variables:
</p>

<dl>
<dt>Chr</dt><dd><p>chromosome</p>
</dd>
<dt>BP</dt><dd><p>basepair position</p>
</dd>
<dt>Marker</dt><dd><p>rsID# or other identifier</p>
</dd>
<dt>MAF</dt><dd><p>Minor Allele Frequency</p>
</dd>
<dt>A1</dt><dd><p>reference allele</p>
</dd>
<dt>A2</dt><dd><p>alternative allele</p>
</dd>
<dt>Direction</dt><dd><p>direction of association
effect size, + or -</p>
</dd>
<dt>pValue</dt><dd><p>p-Value of GWAS association</p>
</dd>
<dt>N</dt><dd><p>sample size</p>
</dd>
</dl>


<h3>Source</h3>

<p>Manually created
</p>

<hr>
<h2 id='bmass_TestData2'>bmass Test Dataset 2</h2><span id='topic+bmass_TestData2'></span>

<h3>Description</h3>

<p>A manually created sample dataset for use in unit
tests.
</p>


<h3>Format</h3>

<p>A data frame with 11 rows and 9 variables:
</p>

<dl>
<dt>Chr</dt><dd><p>chromosome</p>
</dd>
<dt>BP</dt><dd><p>basepair position</p>
</dd>
<dt>Marker</dt><dd><p>rsID# or other identifier</p>
</dd>
<dt>MAF</dt><dd><p>Minor Allele Frequency</p>
</dd>
<dt>A1</dt><dd><p>reference allele</p>
</dd>
<dt>A2</dt><dd><p>alternative allele</p>
</dd>
<dt>Direction</dt><dd><p>direction of association
effect size, + or -</p>
</dd>
<dt>pValue</dt><dd><p>p-Value of GWAS association</p>
</dd>
<dt>N</dt><dd><p>sample size</p>
</dd>
</dl>


<h3>Source</h3>

<p>Manually created
</p>

<hr>
<h2 id='bmass_TestSigSNPs'>bmass Test GWAS SNPs</h2><span id='topic+bmass_TestSigSNPs'></span>

<h3>Description</h3>

<p>A manually created list of GWAS significant SNPs to be
used in conjunction with 'bmass_TestData1' and 'bmass_TestData2'.
</p>


<h3>Format</h3>

<p>A data frame with 2 rows and 2 variables:
</p>

<dl>
<dt>Chr</dt><dd><p>chromosome</p>
</dd>
<dt>BP</dt><dd><p>basepair position</p>
</dd>
</dl>


<h3>Source</h3>

<p>Manually created
</p>

<hr>
<h2 id='GetMarginalPosteriors'>Get Marginal {<b>U</b>,<b>D</b>,<b>I</b>} Posteriors</h2><span id='topic+GetMarginalPosteriors'></span>

<h3>Description</h3>

<p>Get marginal posteriors for how much every individual
phenotype belongs to categories {<b>U</b>,<b>D</b>,<b>I</b>} 
across each SNP
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetMarginalPosteriors(DataSources, ListSNPs, Models, LogFile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetMarginalPosteriors_+3A_datasources">DataSources</code></td>
<td>
<p>A string indicating the variable names of the
input datafiles and phenotypes.</p>
</td></tr>
<tr><td><code id="GetMarginalPosteriors_+3A_listsnps">ListSNPs</code></td>
<td>
<p>A list produced from running <code><a href="#topic+bmass">bmass</a></code> 
containing the SNPs of interest to get marginal posteriors for.</p>
</td></tr>
<tr><td><code id="GetMarginalPosteriors_+3A_models">Models</code></td>
<td>
<p>A matrix describing the models being explored
(default output from running <code><a href="#topic+bmass">bmass</a></code>).</p>
</td></tr>
<tr><td><code id="GetMarginalPosteriors_+3A_logfile">LogFile</code></td>
<td>
<p>A matrix of string outputs for function logging
purposes (default output from running <code><a href="#topic+bmass">bmass</a></code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing three matrices of SNPs x Phenotypes
marginal posteriors for each category {<b>U</b>,<b>D</b>,<b>I</b>}; 
this list is appended to the input ListSNPs as a new object, 
<code>Marginals</code> (the full returned object is a list containing the 
input ListSNPs and the input LogFile).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Phenotypes &lt;- c("bmass_SimulatedData1", "bmass_SimulatedData2")
bmassOutput &lt;- bmass(Phenotypes, bmass_SimulatedSigSNPs)
bmassOutput[c("PreviousSNPs", "LogFile")] &lt;-
GetMarginalPosteriors(Phenotypes, bmassOutput$PreviousSNPs,
bmassOutput$Models, bmassOutput$LogFile)
bmassOutput$PreviousSNPs$Marginals

</code></pre>

<hr>
<h2 id='GetModelPriorMatrix'>Get Model Prior Matrix</h2><span id='topic+GetModelPriorMatrix'></span>

<h3>Description</h3>

<p>Creates a matrix containing the model descriptions and
their associated priors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetModelPriorMatrix(DataSources, Models, ModelPriors, LogFile,
  SigmaAlphas = c(0.005, 0.0075, 0.01, 0.015, 0.02, 0.03, 0.04, 0.05,
  0.06, 0.07, 0.08, 0.09, 0.1, 0.15))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetModelPriorMatrix_+3A_datasources">DataSources</code></td>
<td>
<p>A string indicating the variable names of the
input datafiles and phenotypes.</p>
</td></tr>
<tr><td><code id="GetModelPriorMatrix_+3A_models">Models</code></td>
<td>
<p>A matrix describing the models being explored
(default output from running <code><a href="#topic+bmass">bmass</a></code>).</p>
</td></tr>
<tr><td><code id="GetModelPriorMatrix_+3A_modelpriors">ModelPriors</code></td>
<td>
<p>A vector containing the priors on each model
across each tranche of sigma alpha (default output from running
<code><a href="#topic+bmass">bmass</a></code>; length is number of models times number of 
sigma alphas).</p>
</td></tr>
<tr><td><code id="GetModelPriorMatrix_+3A_logfile">LogFile</code></td>
<td>
<p>A matrix of string outputs for function logging
purposes (default output from running <code><a href="#topic+bmass">bmass</a></code>).</p>
</td></tr>
<tr><td><code id="GetModelPriorMatrix_+3A_sigmaalphas">SigmaAlphas</code></td>
<td>
<p>A vector containing the different values
traversed for this 'effect size controlling' hyperparameter (see
&quot;Prior on Sigma_Alpha&quot; in Stephens 2013 PLoS ONE, 
<a href="https://doi.org/10.1371/journal.pone.0065245">https://doi.org/10.1371/journal.pone.0065245</a>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix containing the original description of each model
sort by prior, each model's trained prior, the cummulative prior
distribution, and the model's original order position.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Phenotypes &lt;- c("bmass_SimulatedData1", "bmass_SimulatedData2")
bmassOutput &lt;- bmass(Phenotypes,bmass_SimulatedSigSNPs)
bmassOutput[c("ModelPriorMatrix", "LogFile")] &lt;- 
  GetModelPriorMatrix(Phenotypes, bmassOutput$Models,
  bmassOutput$ModelPriors, bmassOutput$LogFile)
head(bmassOutput$ModelPriorMatrix)

</code></pre>

<hr>
<h2 id='GetTopModelsPerSNPViaPosteriors'>Get Top Multivariate Models</h2><span id='topic+GetTopModelsPerSNPViaPosteriors'></span>

<h3>Description</h3>

<p>Get a summary of the top models per SNP across all
multivariate {<b>U</b>,<b>D</b>,<b>I</b>} combinations based on 
posterior probabilities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetTopModelsPerSNPViaPosteriors(DataSources, ListSNPs, ModelPriorMatrix,
  LogFile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetTopModelsPerSNPViaPosteriors_+3A_datasources">DataSources</code></td>
<td>
<p>A string indicating the variable names of the
input datafiles and phenotypes.</p>
</td></tr>
<tr><td><code id="GetTopModelsPerSNPViaPosteriors_+3A_listsnps">ListSNPs</code></td>
<td>
<p>A list produced from running <code><a href="#topic+bmass">bmass</a></code> 
containing the SNPs of interest to get marginal posteriors for.</p>
</td></tr>
<tr><td><code id="GetTopModelsPerSNPViaPosteriors_+3A_modelpriormatrix">ModelPriorMatrix</code></td>
<td>
<p>A matrix detailing the models being
explored and their associated priors (obtained by running
<code><a href="#topic+GetModelPriorMatrix">GetModelPriorMatrix</a></code>)</p>
</td></tr>
<tr><td><code id="GetTopModelsPerSNPViaPosteriors_+3A_logfile">LogFile</code></td>
<td>
<p>A matrix of string outputs for function logging
purposes (default output from running <code><a href="#topic+bmass">bmass</a></code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix containing each model that was a SNP's top model
at least once, along with related information; this matrix is
appended to the input ListSNPs as a new object, <code>TopModels</code> 
(the full returned object is a list containing the input ListSNPs and
the input LogFile).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Phenotypes &lt;- c("bmass_SimulatedData1", "bmass_SimulatedData2")
bmassOutput &lt;- bmass(Phenotypes, bmass_SimulatedSigSNPs)
bmassOutput[c("ModelPriorMatrix", "LogFile")] &lt;- 
  GetModelPriorMatrix(Phenotypes, bmassOutput$Models,
  bmassOutput$ModelPriors, bmassOutput$LogFile)
bmassOutput[c("PreviousSNPs", "LogFile")] &lt;-
  GetTopModelsPerSNPViaPosteriors(Phenotypes,
  bmassOutput$PreviousSNPs, bmassOutput$ModelPriorMatrix, bmassOutput$LogFile)
head(bmassOutput$PreviousSNPs$TopModels)

</code></pre>

<hr>
<h2 id='GlobalLipids2013.GWASsnps'>GlobalLipids2013 GWAS SNPs</h2><span id='topic+GlobalLipids2013.GWASsnps'></span>

<h3>Description</h3>

<p>A list of the univariate GWAS significant SNPs from
the GlobalLipids2013 dataset to be used in the second introductory 
bmass vignette.
</p>


<h3>Format</h3>

<p>A data frame with 157 rows and 2 variables:
</p>

<dl>
<dt>Chr</dt><dd><p>chromosome</p>
</dd>
<dt>BP</dt><dd><p>basepair position</p>
</dd>
</dl>


<h3>Source</h3>

<p>Supplementary Tables 2 and 3 from
<a href="https://doi.org/10.1038/ng.2797">https://doi.org/10.1038/ng.2797</a>.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
