<!DOCTYPE html><html><head><title>Help for package BNrich</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BNrich}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BN_struct'><p>Construct Bayesian networks structures</p></a></li>
<li><a href='#BNrich'><p>Analysis of significant final BNs</p></a></li>
<li><a href='#esti_par'><p>Estimate parameters of BNs in control and disease states</p></a></li>
<li><a href='#fetch_data_file'><p>Download data file</p></a></li>
<li><a href='#LASSO_BN'><p>LASSO regression</p></a></li>
<li><a href='#parm_Ttest'><p>Testing the equality regression coefficients</p></a></li>
<li><a href='#unify_path'><p>Simplification networks &ndash; applied to unifying nodes</p></a></li>
<li><a href='#var_mat'><p>Estimate variance-covariance matrixes for any parameters of BNs</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Pathway Enrichment Analysis Based on Bayesian Network</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Samaneh Maleknia &lt;maleknias@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Maleknia et al. (2020) &lt;<a href="https://doi.org/10.1101%2F2020.01.13.905448">doi:10.1101/2020.01.13.905448</a>&gt;. A novel pathway enrichment analysis package based on Bayesian network to investigate the topology features of the pathways. firstly, 187 kyoto encyclopedia of genes and genomes (KEGG) human non-metabolic pathways which their cycles were eliminated by biological approach, enter in analysis as Bayesian network structures. The constructed Bayesian network were optimized by the Least Absolute Shrinkage Selector Operator (lasso) and the parameters were learned based on gene expression data. Finally, the impacted pathways were enriched by Fisherâ€™s Exact Test on significant parameters.</td>
</tr>
<tr>
<td>biocViews:</td>
<td>NetworkEnrichment, GeneExpression, Pathways, Bayesian, KEGG</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.0</td>
</tr>
<tr>
<td>Imports:</td>
<td>bnlearn, corpcor, glmnet, graph, stats, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/Samaneh-Bioinformatics/BNrich">https://github.com/Samaneh-Bioinformatics/BNrich</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Samaneh-Bioinformatics/BNrich/issues">https://github.com/Samaneh-Bioinformatics/BNrich/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-04-04 07:17:44 UTC; Shawshank</td>
</tr>
<tr>
<td>Author:</td>
<td>Samaneh Maleknia [aut, cre],
  Mohsen Namazi [aut],
  Kaveh Kavousi [ctb],
  Ali Sharifi-Zarchi [ctb],
  Vahid Rezaei-Tabar [ctb]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-04-04 08:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='BN_struct'>Construct Bayesian networks structures</h2><span id='topic+BN_struct'></span>

<h3>Description</h3>

<p>Construct BNs structures using unified signaling pathways
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BN_struct(mapkG1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BN_struct_+3A_mapkg1">mapkG1</code></td>
<td>
<p>A list contains unified signaling pathways</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list contains Bayesian networks structures
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#All the 187 preprocessed signaling pathways can be entered in analysis by fetch_data_file().
#But here you enter a subset of those pathways to see how this package works.
files &lt;- system.file("extdata", "test_files_to_start.RData", package = "BNrich", mustWork = TRUE)
load(files)
Data &lt;- system.file("extdata", "Test_DATA.RData", package = "BNrich", mustWork = TRUE)
load(Data)
uni_Result &lt;- unify_path(dataH, dataD, MapkG = sub_mapkG, Pathway.id = path.id)
M1 &lt;- uni_Result$mapkG1
BN &lt;- BN_struct(M1)
</code></pre>

<hr>
<h2 id='BNrich'>Analysis of significant final BNs</h2><span id='topic+BNrich'></span>

<h3>Description</h3>

<p>Fisher's exact test applied to PEA on final BNs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BNrich(Ttest_Results, Pathway.id1, PathName_Final, fdr.value = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BNrich_+3A_ttest_results">Ttest_Results</code></td>
<td>
<p>A data frame contains T-test results for all parameters</p>
</td></tr>
<tr><td><code id="BNrich_+3A_pathway.id1">Pathway.id1</code></td>
<td>
<p>A vector contains modified KEEG pathway IDs</p>
</td></tr>
<tr><td><code id="BNrich_+3A_pathname_final">PathName_Final</code></td>
<td>
<p>A data frame contains is IDs and names of KEEG pathways</p>
</td></tr>
<tr><td><code id="BNrich_+3A_fdr.value">fdr.value</code></td>
<td>
<p>A numeric threshold to determine significant parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame contains fisher test results for any final pathways
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#All the 187 preprocessed signaling pathways can be entered in analysis by fetch_data_file().
#But here you enter a subset of those pathways to see how this package works.
files &lt;- system.file("extdata", "test_files_to_start.RData", package = "BNrich", mustWork = TRUE)
load(files)
Data &lt;- system.file("extdata", "Test_DATA.RData", package = "BNrich", mustWork = TRUE)
load(Data)
uni_Result &lt;- unify_path(dataH, dataD, MapkG = sub_mapkG, Pathway.id = path.id)
M1 &lt;- uni_Result$mapkG1
BN &lt;- BN_struct(M1)
data_h1 &lt;- uni_Result$data_h
data_d1 &lt;- uni_Result$data_d
LASSO_Result &lt;- LASSO_BN(BN = BN , data_h = data_h1 , data_d = data_d1)
BN_h1 &lt;- LASSO_Result$BN_h
BN_d1 &lt;- LASSO_Result$BN_d
esti_result &lt;- esti_par(BN_H = BN_h1, BN_D = BN_d1, data_h = data_h1, data_d = data_d1)
BNs_H &lt;- esti_result$BNs_h
BNs_D &lt;- esti_result$BNs_d
coef_h &lt;- esti_result$coef_h
coef_d &lt;- esti_result$coef_d
var_result &lt;- var_mat(data_h1, coef_h,  BNs_H, data_d1, coef_d, BNs_D)
Var_H = var_result$var_mat_Bh
Var_D = var_result$var_mat_Bd
path.id1 &lt;- uni_Result$pathway.id1
Ttest_result &lt;- parm_Ttest(data_h1, coef_h, BNs_H, data_d1, coef_d, BNs_D, Var_H, Var_D, path.id1)
BNrich_result &lt;- BNrich(Ttest_result, path.id1, Path.Name)
</code></pre>

<hr>
<h2 id='esti_par'>Estimate parameters of BNs in control and disease states</h2><span id='topic+esti_par'></span>

<h3>Description</h3>

<p>Estimate parameters of BNs in control and disease states
</p>


<h3>Usage</h3>

<pre><code class='language-R'>esti_par(BN_H, BN_D, data_h, data_d)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="esti_par_+3A_bn_h">BN_H</code></td>
<td>
<p>A list contains simplified BNs structures for control objects</p>
</td></tr>
<tr><td><code id="esti_par_+3A_bn_d">BN_D</code></td>
<td>
<p>A list contains simplified BNs structures for disease objects</p>
</td></tr>
<tr><td><code id="esti_par_+3A_data_h">data_h</code></td>
<td>
<p>A list contains data frames related to control objects for any BN</p>
</td></tr>
<tr><td><code id="esti_par_+3A_data_d">data_d</code></td>
<td>
<p>A list contains data frames related to disease objects for any BN</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A listcontains four lists BNs_h, BNs_d, coef_h and coef_d
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#All the 187 preprocessed signaling pathways can be entered in analysis by fetch_data_file().
#But here you enter a subset of those pathways to see how this package works.
files &lt;- system.file("extdata", "test_files_to_start.RData", package = "BNrich", mustWork = TRUE)
load(files)
Data &lt;- system.file("extdata", "Test_DATA.RData", package = "BNrich", mustWork = TRUE)
load(Data)
uni_Result &lt;- unify_path(dataH, dataD, MapkG = sub_mapkG, Pathway.id = path.id)
M1 &lt;- uni_Result$mapkG1
BN &lt;- BN_struct(M1)
data_h1 &lt;- uni_Result$data_h
data_d1 &lt;- uni_Result$data_d
LASSO_Result &lt;- LASSO_BN(BN = BN , data_h = data_h1 , data_d = data_d1)
BN_h1 &lt;- LASSO_Result$BN_h
BN_d1 &lt;- LASSO_Result$BN_d
esti_result &lt;- esti_par(BN_H = BN_h1, BN_D = BN_d1, data_h = data_h1, data_d = data_d1)
</code></pre>

<hr>
<h2 id='fetch_data_file'>Download data file</h2><span id='topic+fetch_data_file'></span>

<h3>Description</h3>

<p>Download necessary data file to start BNrich
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fetch_data_file(destfile, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fetch_data_file_+3A_destfile">destfile</code></td>
<td>
<p>A directory in user's own computer for save preprocessed data file</p>
</td></tr>
<tr><td><code id="fetch_data_file_+3A_verbose">verbose</code></td>
<td>
<p>A logical argument to show verbose results</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list contain mapkG, PathName_final and pathway.id. The mapkG is a list contains imported 187 preprocessed signaling pathways, PathName_final is a data.frame includes names and IDs of all 187 pathways and pathway.id is a character vector of pathways IDs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
destfile = tempfile("files", fileext = ".rda")
files &lt;- fetch_data_file()
load(destfile)

</code></pre>

<hr>
<h2 id='LASSO_BN'>LASSO regression</h2><span id='topic+LASSO_BN'></span>

<h3>Description</h3>

<p>LASSO regression â€“ second step of simplification of BNs structures
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LASSO_BN(BN, data_h, data_d)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LASSO_BN_+3A_bn">BN</code></td>
<td>
<p>A list of Bayesian networks achieved by BN_struct function</p>
</td></tr>
<tr><td><code id="LASSO_BN_+3A_data_h">data_h</code></td>
<td>
<p>A list contains data frames related to control objects</p>
</td></tr>
<tr><td><code id="LASSO_BN_+3A_data_d">data_d</code></td>
<td>
<p>A list contains data frames related to disease objects</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list contains two lists.BN_h and BN_d are simplified BNs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#All the 187 preprocessed signaling pathways can be entered in analysis by fetch_data_file().
#But here you enter a subset of those pathways to see how this package works.
files &lt;- system.file("extdata", "test_files_to_start.RData", package = "BNrich", mustWork = TRUE)
load(files)
Data &lt;- system.file("extdata", "Test_DATA.RData", package = "BNrich", mustWork = TRUE)
load(Data)
uni_Result &lt;- unify_path(dataH, dataD, MapkG = sub_mapkG, Pathway.id = path.id)
M1 &lt;- uni_Result$mapkG1
BN &lt;- BN_struct(M1)
data_h1 &lt;- uni_Result$data_h
data_d1 &lt;- uni_Result$data_d
LASSO_Result &lt;- LASSO_BN(BN = BN , data_h = data_h1 , data_d = data_d1)
</code></pre>

<hr>
<h2 id='parm_Ttest'>Testing the equality regression coefficients</h2><span id='topic+parm_Ttest'></span>

<h3>Description</h3>

<p>t-test for equality the corresponging parameters in any BN
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parm_Ttest(
  Data_h,
  coef_H,
  BNs_H,
  Data_d,
  coef_D,
  BNs_D,
  Var_mat_Bh,
  Var_mat_Bd,
  Pathway.id1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parm_Ttest_+3A_data_h">Data_h</code></td>
<td>
<p>A list contains data frames related to control objects for any BN</p>
</td></tr>
<tr><td><code id="parm_Ttest_+3A_coef_h">coef_H</code></td>
<td>
<p>A list contains parameters of BN_h</p>
</td></tr>
<tr><td><code id="parm_Ttest_+3A_bns_h">BNs_H</code></td>
<td>
<p>A list contains BNs learned by control objects data</p>
</td></tr>
<tr><td><code id="parm_Ttest_+3A_data_d">Data_d</code></td>
<td>
<p>A list contains data frames related to disease objects for any BN</p>
</td></tr>
<tr><td><code id="parm_Ttest_+3A_coef_d">coef_D</code></td>
<td>
<p>A list contains parameters of BN_d</p>
</td></tr>
<tr><td><code id="parm_Ttest_+3A_bns_d">BNs_D</code></td>
<td>
<p>A list contains BNs learned by disease objects data</p>
</td></tr>
<tr><td><code id="parm_Ttest_+3A_var_mat_bh">Var_mat_Bh</code></td>
<td>
<p>A list contains covariance matrixes for any node of BN_h</p>
</td></tr>
<tr><td><code id="parm_Ttest_+3A_var_mat_bd">Var_mat_Bd</code></td>
<td>
<p>A list contains covariance matrixes for any node of BN_d</p>
</td></tr>
<tr><td><code id="parm_Ttest_+3A_pathway.id1">Pathway.id1</code></td>
<td>
<p>A vector contains modified KEEG pathway IDs</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame contains T-test results for all parameters in final BNs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#All the 187 preprocessed signaling pathways can be entered in analysis by fetch_data_file().
#But here you enter a subset of those pathways to see how this package works.
files &lt;- system.file("extdata", "test_files_to_start.RData", package = "BNrich", mustWork = TRUE)
load(files)
Data &lt;- system.file("extdata", "Test_DATA.RData", package = "BNrich", mustWork = TRUE)
load(Data)
uni_Result &lt;- unify_path(dataH, dataD, MapkG = sub_mapkG, Pathway.id = path.id)
M1 &lt;- uni_Result$mapkG1
BN &lt;- BN_struct(M1)
data_h1 &lt;- uni_Result$data_h
data_d1 &lt;- uni_Result$data_d
LASSO_Result &lt;- LASSO_BN(BN = BN , data_h = data_h1 , data_d = data_d1)
BN_h1 &lt;- LASSO_Result$BN_h
BN_d1 &lt;- LASSO_Result$BN_d
esti_result &lt;- esti_par(BN_H = BN_h1, BN_D = BN_d1, data_h = data_h1, data_d = data_d1)
BNs_H &lt;- esti_result$BNs_h
BNs_D &lt;- esti_result$BNs_d
coef_h &lt;- esti_result$coef_h
coef_d &lt;- esti_result$coef_d
var_result &lt;- var_mat(data_h1, coef_h,  BNs_H, data_d1, coef_d, BNs_D)
Var_H = var_result$var_mat_Bh
Var_D = var_result$var_mat_Bd
path.id1 &lt;- uni_Result$pathway.id1
Ttest_result &lt;- parm_Ttest(data_h1, coef_h, BNs_H, data_d1, coef_d, BNs_D, Var_H, Var_D, path.id1)
</code></pre>

<hr>
<h2 id='unify_path'>Simplification networks &ndash; applied to unifying nodes</h2><span id='topic+unify_path'></span>

<h3>Description</h3>

<p>Unifying nodes based imported signaling pathways and GE data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unify_path(dataH, dataD, MapkG, Pathway.id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unify_path_+3A_datah">dataH</code></td>
<td>
<p>A data frame contains (healthy) control objects data</p>
</td></tr>
<tr><td><code id="unify_path_+3A_datad">dataD</code></td>
<td>
<p>A data frame contains disease objects data</p>
</td></tr>
<tr><td><code id="unify_path_+3A_mapkg">MapkG</code></td>
<td>
<p>A list contains imported 187 signaling pathways</p>
</td></tr>
<tr><td><code id="unify_path_+3A_pathway.id">Pathway.id</code></td>
<td>
<p>A vector contains 187 KEEG pathway IDs</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list contain data_h,data_d,mapkG1 and pathway.id1
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#All the 187 preprocessed signaling pathways can be entered in analysis by fetch_data_file().
#But here you enter a subset of those pathways to see how this package works.
files &lt;- system.file("extdata", "test_files_to_start.RData", package = "BNrich", mustWork = TRUE)
load(files)
Data &lt;- system.file("extdata", "Test_DATA.RData", package = "BNrich", mustWork = TRUE)
load(Data)
uni_Result &lt;- unify_path(dataH, dataD, MapkG = sub_mapkG, Pathway.id = path.id)
</code></pre>

<hr>
<h2 id='var_mat'>Estimate variance-covariance matrixes for any parameters of BNs</h2><span id='topic+var_mat'></span>

<h3>Description</h3>

<p>Estimate variance-covariance matrixes for any parameters of
</p>


<h3>Usage</h3>

<pre><code class='language-R'>var_mat(Data_h, coef_H, BNs_H, Data_d, coef_D, BNs_D)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="var_mat_+3A_data_h">Data_h</code></td>
<td>
<p>A list contains data frames related to control objects for any BN</p>
</td></tr>
<tr><td><code id="var_mat_+3A_coef_h">coef_H</code></td>
<td>
<p>A lists of parameters of BN_h achieved</p>
</td></tr>
<tr><td><code id="var_mat_+3A_bns_h">BNs_H</code></td>
<td>
<p>A list of BNs learned by control objects data</p>
</td></tr>
<tr><td><code id="var_mat_+3A_data_d">Data_d</code></td>
<td>
<p>A list contains data frames related to disease objects for any BN</p>
</td></tr>
<tr><td><code id="var_mat_+3A_coef_d">coef_D</code></td>
<td>
<p>A lists of parameters of BN_d</p>
</td></tr>
<tr><td><code id="var_mat_+3A_bns_d">BNs_D</code></td>
<td>
<p>A list of BNs learned by disease objects data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A listcontains two lists var_mat_Bh and var_mat_Bd
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#All the 187 preprocessed signaling pathways can be entered in analysis by fetch_data_file().
#But here you enter a subset of those pathways to see how this package works.
files &lt;- system.file("extdata", "test_files_to_start.RData", package = "BNrich", mustWork = TRUE)
load(files)
Data &lt;- system.file("extdata", "Test_DATA.RData", package = "BNrich", mustWork = TRUE)
load(Data)
uni_Result &lt;- unify_path(dataH, dataD, MapkG = sub_mapkG, Pathway.id = path.id)
M1 &lt;- uni_Result$mapkG1
BN &lt;- BN_struct(M1)
data_h1 &lt;- uni_Result$data_h
data_d1 &lt;- uni_Result$data_d
LASSO_Result &lt;- LASSO_BN(BN = BN , data_h = data_h1 , data_d = data_d1)
BN_h1 &lt;- LASSO_Result$BN_h
BN_d1 &lt;- LASSO_Result$BN_d
esti_result &lt;- esti_par(BN_H = BN_h1, BN_D = BN_d1, data_h = data_h1, data_d = data_d1)
BNs_H &lt;- esti_result$BNs_h
BNs_D &lt;- esti_result$BNs_d
coef_h &lt;- esti_result$coef_h
coef_d &lt;- esti_result$coef_d
var_result &lt;- var_mat(data_h1, coef_h,  BNs_H, data_d1, coef_d, BNs_D)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
