<!DOCTYPE html><html><head><title>Help for package bold</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {bold}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bold-package'><p>bold</p></a></li>
<li><a href='#b_sepFasta'><p>Seperate sequences (fasta) from <code>bold_seqspec</code> output.</p></a></li>
<li><a href='#bold_filter'><p>Filter BOLD specimen + sequence data (output of bold_seqspec)</p></a></li>
<li><a href='#bold_get_attr'><p>Get the error messages and parameters used for a request from a bold output.</p></a></li>
<li><a href='#bold_identify'><p>Search for matches to sequences against the BOLD COI database.</p></a></li>
<li><a href='#bold_identify_parents-deprecated'><p>Add taxonomic parent names to a data.frame</p></a></li>
<li><a href='#bold_identify_taxonomy'><p>Add taxonomic parent names to a data set containing the process IDs of identified sequences.</p></a></li>
<li><a href='#bold_seq'><p>Search BOLD for sequences.</p></a></li>
<li><a href='#bold_seqspec'><p>Get BOLD specimen + sequence data.</p></a></li>
<li><a href='#bold_specimens'><p>Search BOLD for specimens.</p></a></li>
<li><a href='#bold_stats'><p>Get BOLD stats</p></a></li>
<li><a href='#bold_tax_id-deprecated'><p>Search BOLD for taxonomy data by BOLD ID.</p></a></li>
<li><a href='#bold_tax_id2'><p>Search BOLD for taxonomy data by BOLD ID.</p></a></li>
<li><a href='#bold_tax_name'><p>Search BOLD for taxonomy data by taxonomic name</p></a></li>
<li><a href='#bold_trace'><p>Get BOLD trace files</p></a></li>
<li><a href='#bold-deprecated'><p>Deprecated functions in package <span class="pkg">bold</span>.</p></a></li>
<li><a href='#sequences'><p>List of 3 nucleotide sequences to use in examples for the</p>
<code>bold_identify</code> function</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.3.0</td>
</tr>
<tr>
<td>Title:</td>
<td>Interface to Bold Systems API</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>sangerseqR, testthat, vcr (&ge; 0.5.4), tibble</td>
</tr>
<tr>
<td>Imports:</td>
<td>xml2, crul (&ge; 0.3.8), stringi, jsonlite, data.table, methods</td>
</tr>
<tr>
<td>Description:</td>
<td>A programmatic interface to the Web Service methods provided by
    Bold Systems (<a href="http://www.boldsystems.org/">http://www.boldsystems.org/</a>) for genetic 'barcode' data.
    Functions include methods for searching by sequences by taxonomic names,
    ids, collectors, and institutions; as well as a function for searching
    for specimens, and downloading trace files.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://docs.ropensci.org/bold/">https://docs.ropensci.org/bold/</a> (documentation),
<a href="https://github.com/ropensci/bold">https://github.com/ropensci/bold</a> (source)</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ropensci/bold/issues">https://github.com/ropensci/bold/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>X-schema.org-applicationCategory:</td>
<td>Data Access</td>
</tr>
<tr>
<td>X-schema.org-keywords:</td>
<td>biodiversity, barcode, DNA, sequences, fasta</td>
</tr>
<tr>
<td>X-schema.org-isPartOf:</td>
<td>https://ropensci.org</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-05-01 19:56:58 UTC; salix</td>
</tr>
<tr>
<td>Author:</td>
<td>Salix Dubois [aut, cre],
  Scott Chamberlain <a href="https://orcid.org/0000-0003-1444-9135"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Salix Dubois &lt;salixdubois+bold@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-05-02 17:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='bold-package'>bold</h2><span id='topic+bold-package'></span><span id='topic+bold'></span>

<h3>Description</h3>

<p>bold: A programmatic interface to the Barcode of Life data
</p>


<h3>About</h3>

<p>This package gives you access to data from BOLD System
http://www.boldsystems.org/ via their API
(http://v4.boldsystems.org/index.php/api_home)
</p>


<h3>Functions</h3>


<ul>
<li> <p><code><a href="#topic+bold_specimens">bold_specimens</a></code> - Search for specimen data
</p>
</li>
<li> <p><code><a href="#topic+bold_seq">bold_seq</a></code> - Search for and retrieve sequences
</p>
</li>
<li> <p><code><a href="#topic+bold_seqspec">bold_seqspec</a></code> - Get sequence and specimen data together
</p>
</li>
<li> <p><code><a href="#topic+bold_trace">bold_trace</a></code> - Get trace files - saves to disk
</p>
</li>
<li> <p><code><a href="#topic+read_trace">read_trace</a></code> - Read trace files into R
</p>
</li>
<li> <p><code><a href="#topic+bold_tax_name">bold_tax_name</a></code> - Get taxonomic names via input names
</p>
</li>
<li> <p><code><a href="#topic+bold_tax_id">bold_tax_id</a></code> - Get taxonomic names via BOLD identifiers (Deprecated)
</p>
</li>
<li> <p><code><a href="#topic+bold_tax_id2">bold_tax_id2</a></code> - Get taxonomic names via BOLD identifiers (improved)
</p>
</li>
<li> <p><code><a href="#topic+bold_identify">bold_identify</a></code> - Search for match given a COI sequence
</p>
</li>
<li> <p><code><a href="#topic+bold_identify_parents">bold_identify_parents</a></code> - Adds guessed parent ranks (Deprecated)
</p>
</li>
<li> <p><code><a href="#topic+bold_identify_taxonomy">bold_identify_taxonomy</a></code> - Adds real parent ranks.
</p>
</li></ul>

<p>Interestingly, they provide xml and tsv format data for the specimen data,
while they provide fasta data format for the sequence data. So for the
specimen data you can get back raw XML, or a data frame parsed from the
tsv data, while for sequence data you get back a list (b/c sequences are
quite long and would make a data frame unwieldy).
</p>

<hr>
<h2 id='b_sepFasta'>Seperate sequences (fasta) from <code>bold_seqspec</code> output.</h2><span id='topic+b_sepFasta'></span>

<h3>Description</h3>

<p>Seperate sequences (fasta) from <code>bold_seqspec</code> output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>b_sepFasta(x, format = "tsv")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="b_sepFasta_+3A_x">x</code></td>
<td>
<p>(object) The output of a <code>bold_seqspec</code> call.</p>
</td></tr>
<tr><td><code id="b_sepFasta_+3A_format">format</code></td>
<td>
<p>(character) The format used in the <code>bold_seqspec</code> call. One of 'tsv' (default) or 'xml'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of length two : the specimen data and the sequences list.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
res &lt;- bold_seqspec(taxon='Osmia')
res &lt;- b_sepFasta(res)
# (same as bold_seqspec(taxon='Osmia', sepFasta = TRUE))

## End(Not run)
</code></pre>

<hr>
<h2 id='bold_filter'>Filter BOLD specimen + sequence data (output of bold_seqspec)</h2><span id='topic+bold_filter'></span>

<h3>Description</h3>

<p>Picks either shortest or longest sequences, for a given grouping variable
(e.g., species name)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bold_filter(x, by, how = "max", returnTibble = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bold_filter_+3A_x">x</code></td>
<td>
<p>(data.frame) a data.frame, as returned from
<code><a href="#topic+bold_seqspec">bold_seqspec</a></code>. Note that some combinations of parameters
in <code><a href="#topic+bold_seqspec">bold_seqspec</a></code> don't return a data.frame. Stops with
error message if this is not a data.frame. Required.</p>
</td></tr>
<tr><td><code id="bold_filter_+3A_by">by</code></td>
<td>
<p>(character) the column by which to group. For example,
if you want the longest sequence for each unique species name, then
pass <strong>species_name</strong>. If the column doesn't exist, error
with message saying so. Required.</p>
</td></tr>
<tr><td><code id="bold_filter_+3A_how">how</code></td>
<td>
<p>(character) one of &quot;max&quot; or &quot;min&quot;, which get used as
<code>which.max</code> or <code>which.min</code> to get the longest or shortest
sequence, respectively. Note that we remove gap/alignment characters
(<code>-</code>)</p>
</td></tr>
<tr><td><code id="bold_filter_+3A_returntibble">returnTibble</code></td>
<td>
<p>Whether the output should be a tibble or
a data.frame. Default is TRUE, but verifies that the <code>tibble</code> package is
installed, if it's not, it will be returned as data.frame.
Since this package is only used in this function, doing this so it can be
moved to suggested instead of dependency without breaking old scripts.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
res &lt;- bold_seqspec(taxon = 'Osmia')
maxx &lt;- bold_filter(res, by = "species_name")
minn &lt;- bold_filter(res, by = "species_name", how = "min")

vapply(maxx$nucleotides, nchar, 1, USE.NAMES = FALSE)
vapply(minn$nucleotides, nchar, 1, USE.NAMES = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='bold_get_attr'>Get the error messages and parameters used for a request from a bold output.</h2><span id='topic+bold_get_attr'></span><span id='topic+bold_get_errors'></span><span id='topic+bold_get_params'></span>

<h3>Description</h3>

<p>Get the error messages and parameters used for a request from a bold output.
</p>
<p>Get the error messages from a bold output.
</p>
<p>Get the parameters used for a request from a bold output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bold_get_attr(x)

bold_get_errors(x)

bold_get_params(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bold_get_attr_+3A_x">x</code></td>
<td>
<p>Any object with an attribute &quot;params&quot;. Usually the output of <code><a href="#topic+bold_tax_name">bold_tax_name</a></code> or <code><a href="#topic+bold_tax_id2">bold_tax_id2</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of the attributes 'errors' and 'params' of the object.
</p>
<p>The 'errors' attribute of the object.
</p>
<p>The 'params' attribute of the object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
x &lt;- bold_tax_name(name=c("Apis","Felis","Pinus"), tax_division = "Animalia")
bold_get_errors(x)
y &lt;- bold_tax_id2(id = c(88899999, 125295, NA_integer_), dataTypes = c("basic", "stats"))
bold_get_errors(y)

## End(Not run)
## Not run: 
x &lt;- bold_tax_name(name=c("Apis","Felis","Pinus"), tax_division = "Animalia")
bold_get_errors(x)
y &lt;- bold_tax_id2(id = c(88899999, 125295, NA_integer_), dataTypes = c("basic", "stats"))
bold_get_errors(y)

## End(Not run)
## Not run: 
x &lt;- bold_tax_name(name=c("Apis","Felis","Pinus"), tax_division = "Animalia")
bold_get_params(x)
y &lt;- bold_tax_id2(id = c(88899999, 125295, NA_integer_), dataTypes = c("basic", "stats"))
bold_get_params(y)

## End(Not run)
</code></pre>

<hr>
<h2 id='bold_identify'>Search for matches to sequences against the BOLD COI database.</h2><span id='topic+bold_identify'></span>

<h3>Description</h3>

<p>Search for matches to sequences against the BOLD COI database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bold_identify(
  sequences,
  db = c("COX1", "COX1_SPECIES", "COX1_SPECIES_PUBLIC", "COX1_L640bp"),
  response = FALSE,
  keepSeq = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bold_identify_+3A_sequences">sequences</code></td>
<td>
<p>(character) A vector or list of sequences to identify.
Required. See Details.</p>
</td></tr>
<tr><td><code id="bold_identify_+3A_db">db</code></td>
<td>
<p>(character) The database to match against, one of COX1 (default),
COX1_SPECIES, COX1_SPECIES_PUBLIC, OR COX1_L640bp. See Details for
more information.</p>
</td></tr>
<tr><td><code id="bold_identify_+3A_response">response</code></td>
<td>
<p>(logical) Note that response is the object that returns
from the Curl call, useful for debugging, and getting detailed info on
the API call.</p>
</td></tr>
<tr><td><code id="bold_identify_+3A_keepseq">keepSeq</code></td>
<td>
<p>(logical) If TRUE (default), returns each data.frame
with an attribute 'sequence' containing sequence used to get those results.</p>
</td></tr>
<tr><td><code id="bold_identify_+3A_...">...</code></td>
<td>
<p>Further args passed on to <code><a href="crul.html#topic+verb-GET">verb-GET</a></code>, main
purpose being curl debugging</p>
</td></tr>
</table>


<h3>Details</h3>

<p>BOLD only allows one sequences per query. We internally
<code><a href="base.html#topic+lapply">lapply</a></code> over the input values given to the sequences'
parameter to search with one sequences per query. Remember this if you have a
lot of sequences - you are doing a separate query for each one, so it can
take a long time - if you run into errors let us know.
</p>


<h3>Value</h3>

<p>A data.frame or  list of (one per sequences) with the top specimen
matches (up to 100) and their details. If the query fails, returns <code>NULL</code>.
Each data.frame has the attributes <code>sequence</code> with the provided
sequence to match (unless keepSeq is set to FALSE) and <code>errors</code> with the
error message given from a failed request.
</p>


<h3>db parmeter options</h3>


<ul>
<li><p> COX1 Every COI barcode record on BOLD with a minimum sequences
length of 500bp (warning: unvalidated library and includes records without
species level identification). This includes many species represented by
only one or two specimens as well as all species with interim taxonomy. This
search only returns a list of the nearest matches and does not provide a
probability of placement to a taxon.
</p>
</li>
<li><p> COX1_SPECIES Every COI barcode record with a species level
identification and a minimum sequences length of 500bp. This includes
many species represented by only one or two specimens as well as  all
species with interim taxonomy.
Note : Sometimes it does return matches that don't have a species level
identification. Will be checking with BOLD.
</p>
</li>
<li><p> COX1_SPECIES_PUBLIC All published COI records from BOLD and GenBank
with a minimum sequences length of 500bp. This library is a collection of
records from the published projects section of BOLD.
</p>
</li>
<li><p> OR COX1_L640bp Subset of the Species library with a minimum sequences
length of 640bp and containing both public and private records. This library
is intended for short sequences identification as it provides maximum overlap
with short reads from the barcode region of COI.
</p>
</li></ul>



<h3>Named outputs</h3>

<p>For a named output list, make sure to pass in a named list or vector to the
<code>sequences</code> parameter. You can use <code><a href="base.html#topic+names+3C-">names&lt;-</a></code> or
<code><a href="stats.html#topic+setNames">setNames</a></code> to set names on a list or vector of sequences.
</p>


<h3>References</h3>

<p>http://v4.boldsystems.org/index.php/resources/api?type=idengine
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bold_identify_parents">bold_identify_parents</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
seq &lt;- sequences$seq1
res &lt;- bold_identify(sequences=seq)
head(res[[1]])
head(bold_identify(sequences=seq, db='COX1_SPECIES')[[1]])

## End(Not run)

</code></pre>

<hr>
<h2 id='bold_identify_parents-deprecated'>Add taxonomic parent names to a data.frame</h2><span id='topic+bold_identify_parents-deprecated'></span>

<h3>Description</h3>

<p>Add taxonomic parent names to a data.frame
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="bold_identify_parents-deprecated_+3A_x">x</code></td>
<td>
<p>(data.frame/list) list of data.frames - the output from a call to
<code><a href="#topic+bold_identify">bold_identify</a></code>. or a single data.frame from the output from same.
required.</p>
</td></tr>
<tr><td><code id="bold_identify_parents-deprecated_+3A_wide">wide</code></td>
<td>
<p>(logical) output in long or wide format. See Details.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="bold_identify_parents-deprecated_+3A_taxid">taxid</code></td>
<td>
<p>(character) A taxid name. Optional. See <code>Filtering</code> below.</p>
</td></tr>
<tr><td><code id="bold_identify_parents-deprecated_+3A_taxon">taxon</code></td>
<td>
<p>(character) A taxon name. Optional. See <code>Filtering</code> below.</p>
</td></tr>
<tr><td><code id="bold_identify_parents-deprecated_+3A_tax_rank">tax_rank</code></td>
<td>
<p>(character) A tax_rank name. Optional. See <code>Filtering</code>
below.</p>
</td></tr>
<tr><td><code id="bold_identify_parents-deprecated_+3A_tax_division">tax_division</code></td>
<td>
<p>(character) A tax_division name. Optional. See
<code>Filtering</code> below.</p>
</td></tr>
<tr><td><code id="bold_identify_parents-deprecated_+3A_parentid">parentid</code></td>
<td>
<p>(character) A parentid name. Optional. See <code>Filtering</code>
below.</p>
</td></tr>
<tr><td><code id="bold_identify_parents-deprecated_+3A_parentname">parentname</code></td>
<td>
<p>(character) A parentname name. Optional. See <code>Filtering</code>
below.</p>
</td></tr>
<tr><td><code id="bold_identify_parents-deprecated_+3A_taxonrep">taxonrep</code></td>
<td>
<p>(character) A taxonrep name. Optional. See <code>Filtering</code>
below.</p>
</td></tr>
<tr><td><code id="bold_identify_parents-deprecated_+3A_specimenrecords">specimenrecords</code></td>
<td>
<p>(character) A specimenrecords name. Optional.
See <code>Filtering</code> below.</p>
</td></tr>
<tr><td><code id="bold_identify_parents-deprecated_+3A_...">...</code></td>
<td>
<p>Further args passed on to <code><a href="crul.html#topic+verb-GET">verb-GET</a></code>, main
purpose being curl debugging</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+bold-deprecated">bold-deprecated</a></code>
</p>

<hr>
<h2 id='bold_identify_taxonomy'>Add taxonomic parent names to a data set containing the process IDs of identified sequences.</h2><span id='topic+bold_identify_taxonomy'></span><span id='topic+bold_identify_taxonomy+2Clist-method'></span><span id='topic+bold_identify_taxonomy+2Cmatrix-method'></span><span id='topic+bold_identify_taxonomy+2Cdata.frame-method'></span><span id='topic+bold_identify_taxonomy+2Cmissing-method'></span>

<h3>Description</h3>

<p>Add taxonomic parent names to a data set containing the process IDs of identified sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bold_identify_taxonomy(x, taxOnly = TRUE)

## S4 method for signature 'list'
bold_identify_taxonomy(x, taxOnly = TRUE)

## S4 method for signature 'matrix'
bold_identify_taxonomy(x, taxOnly = TRUE)

## S4 method for signature 'data.frame'
bold_identify_taxonomy(x, taxOnly = TRUE)

## S4 method for signature 'missing'
bold_identify_taxonomy(x, taxOnly = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bold_identify_taxonomy_+3A_x">x</code></td>
<td>
<p>A single data.frame or matrix, or a list of. Usually the output from a call to <code><a href="#topic+bold_identify">bold_identify</a></code>. Required.</p>
</td></tr>
<tr><td><code id="bold_identify_taxonomy_+3A_taxonly">taxOnly</code></td>
<td>
<p>(logical) If TRUE (Default), only the taxonomic names and ids are added (equivalent format to the results of <code><a href="#topic+bold_identify_parents">bold_identify_parents</a></code> when <code>wide</code>is set to TRUE. If FALSE, also joins the rest of the data returned by <code><a href="#topic+bold_specimens">bold_specimens</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function gets the process ids from the
input data.frame(s) (ID column), then queries <code><a href="#topic+bold_specimens">bold_specimens</a></code>
to get the sample information and adds it to the input data.frame(s).
</p>
<p>Records in the input data that do not have matches for parent names
simply get NA values in the added columns.
</p>


<h3>Value</h3>

<p>a data.frame or a list of data.frames with added taxonomic
classification.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
seqs &lt;- bold_identify(sequences = bold::sequences$seq2)

seqs_tax &lt;- bold_identify_taxonomy(seqs)
head(seqs_tax[[1]])

x &lt;- bold_seq(taxon = "Satyrium")
seqs &lt;- bold_identify(x$sequence[1:2])
seqs_tax &lt;- bold_identify_taxonomy(seqs)
seqs_tax

## End(Not run)

</code></pre>

<hr>
<h2 id='bold_seq'>Search BOLD for sequences.</h2><span id='topic+bold_seq'></span>

<h3>Description</h3>

<p>Get sequences for a taxonomic name, id, bin, container, institution,
researcher, geographic, place, or gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bold_seq(
  taxon = NULL,
  ids = NULL,
  bin = NULL,
  container = NULL,
  institutions = NULL,
  researchers = NULL,
  geo = NULL,
  marker = NULL,
  response = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bold_seq_+3A_taxon">taxon</code></td>
<td>
<p>(character) One or more taxonomic name. Optional.</p>
</td></tr>
<tr><td><code id="bold_seq_+3A_ids">ids</code></td>
<td>
<p>(character|integer|numeric) One or more IDs. Optional. IDs
include Sample IDs, Process IDs, Museum IDs and Field IDs.</p>
</td></tr>
<tr><td><code id="bold_seq_+3A_bin">bin</code></td>
<td>
<p>(character) One or more Barcode Index Number URI. Optional.</p>
</td></tr>
<tr><td><code id="bold_seq_+3A_container">container</code></td>
<td>
<p>(character) One or more project codes or dataset codes.
Optional.</p>
</td></tr>
<tr><td><code id="bold_seq_+3A_institutions">institutions</code></td>
<td>
<p>(character) One or more institution's name. Optional.
Institutions are the Specimen Storing Site.</p>
</td></tr>
<tr><td><code id="bold_seq_+3A_researchers">researchers</code></td>
<td>
<p>(character) One or more researcher names. Optional.
Include collectors and specimen identifiers.</p>
</td></tr>
<tr><td><code id="bold_seq_+3A_geo">geo</code></td>
<td>
<p>(character) One or more geographic sites.
Includes countries and province/states.</p>
</td></tr>
<tr><td><code id="bold_seq_+3A_marker">marker</code></td>
<td>
<p>(character) Returns all records containing matching
marker codes.</p>
</td></tr>
<tr><td><code id="bold_seq_+3A_response">response</code></td>
<td>
<p>(logical) Default : FALSE. If TRUE, returns the object from
the Curl call. Useful for debugging and getting more detailed info on the
API call.</p>
</td></tr>
<tr><td><code id="bold_seq_+3A_...">...</code></td>
<td>
<p>Further args passed on to
<code><a href="crul.html#topic+verb-GET">crul::verb-GET</a></code>, main purpose being curl
debugging</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with each element as row and 5 columns for processid, identification,
marker, accession, and sequence.
</p>


<h3>Large requests</h3>

<p>Some requests can lead to errors. These often have to do with requesting
data for a rank that is quite high in the tree, such as an Order,
for example, Coleoptera. If your request is taking a long time,
it's likely that something will go wrong on the BOLD server side,
or we'll not be able to parse the result here in R because
R can only process strings of a certain length. <code>bold</code>
users have reported errors in which the resulting response from
BOLD is so large that we could not parse it.
</p>
<p>A good strategy for when you want data for a high rank is to
do many separate requests for lower ranks within your target
rank. You can do this manually, or use the function
<code>taxize::downstream</code> to get all the names of a lower
rank within a target rank. There's an example in the README
(https://docs.ropensci.org/bold/#large-data)
</p>


<h3>If a request times out</h3>

<p>This is likely because you're request was for a large number of
sequences and the BOLD service timed out. You still should get
some output, those sequences that were retrieved before the time
out happened. As above, see the README
(https://docs.ropensci.org/bold/#large-data) for an example of
dealing with large data problems with this function.
</p>


<h3>Marker</h3>

<p>Notes from BOLD on the <code>marker</code> param:
&quot;All markers for a specimen matching the search string will be returned.
ie. A record with COI-5P and ITS will return sequence data for both
markers even if only COI-5P was specified.&quot;
</p>
<p>You will likely end up with data with markers that you did not request -
just be sure to filter those out as needed.
</p>


<h3>Note</h3>

<p>If using the <code>taxon</code> parameter with another parameter, if the <code>taxon</code> isn't found in the public database, it will act as if no <code>taxon</code> was specified and try to return all the data for the other specified parameter. You can make sure that the <code>taxon</code> you're looking up has public records with <code><a href="#topic+bold_stats">bold_stats</a></code>.
</p>


<h3>References</h3>

<p>http://v4.boldsystems.org/index.php/resources/api?type=webservices
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
bold_seq(taxon='Coelioxys')
bold_seq(taxon='Aglae')
bold_seq(taxon=c('Coelioxys','Osmia'))
bold_seq(ids='ACRJP618-11')
bold_seq(ids=c('ACRJP618-11','ACRJP619-11'))
bold_seq(bin='BOLD:AAA5125')
bold_seq(container='ACRJP')
bold_seq(researchers='Thibaud Decaens')
bold_seq(geo='Ireland')
bold_seq(geo=c('Ireland','Denmark'))

# Return the http response object for detailed Curl call response details
res &lt;- bold_seq(taxon='Coelioxys', response=TRUE)
res$url
res$status_code
res$response_headers

## curl debugging
### You can do many things, including get verbose output on the curl
### call, and set a timeout
bold_seq(taxon='Coelioxys', verbose = TRUE)[1:2]

## End(Not run)

</code></pre>

<hr>
<h2 id='bold_seqspec'>Get BOLD specimen + sequence data.</h2><span id='topic+bold_seqspec'></span>

<h3>Description</h3>

<p>Get BOLD specimen + sequence data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bold_seqspec(
  taxon = NULL,
  ids = NULL,
  bin = NULL,
  container = NULL,
  institutions = NULL,
  researchers = NULL,
  geo = NULL,
  marker = NULL,
  response = FALSE,
  format = "tsv",
  sepfasta = FALSE,
  cleanData = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bold_seqspec_+3A_taxon">taxon</code></td>
<td>
<p>(character) One or more taxonomic name. Optional.</p>
</td></tr>
<tr><td><code id="bold_seqspec_+3A_ids">ids</code></td>
<td>
<p>(character|integer|numeric) One or more IDs. Optional. IDs
include Sample IDs, Process IDs, Museum IDs and Field IDs.</p>
</td></tr>
<tr><td><code id="bold_seqspec_+3A_bin">bin</code></td>
<td>
<p>(character) One or more Barcode Index Number URI. Optional.</p>
</td></tr>
<tr><td><code id="bold_seqspec_+3A_container">container</code></td>
<td>
<p>(character) One or more project codes or dataset codes.
Optional.</p>
</td></tr>
<tr><td><code id="bold_seqspec_+3A_institutions">institutions</code></td>
<td>
<p>(character) One or more institution's name. Optional.
Institutions are the Specimen Storing Site.</p>
</td></tr>
<tr><td><code id="bold_seqspec_+3A_researchers">researchers</code></td>
<td>
<p>(character) One or more researcher names. Optional.
Include collectors and specimen identifiers.</p>
</td></tr>
<tr><td><code id="bold_seqspec_+3A_geo">geo</code></td>
<td>
<p>(character) One or more geographic sites.
Includes countries and province/states.</p>
</td></tr>
<tr><td><code id="bold_seqspec_+3A_marker">marker</code></td>
<td>
<p>(character) Returns all records containing matching marker
codes. See Details.</p>
</td></tr>
<tr><td><code id="bold_seqspec_+3A_response">response</code></td>
<td>
<p>(logical) Default : FALSE. If TRUE, returns the object from
the Curl call. Useful for debugging and getting more detailed info on the
API call.</p>
</td></tr>
<tr><td><code id="bold_seqspec_+3A_format">format</code></td>
<td>
<p>(character) One of xml or tsv (default). tsv format gives
back a data.frame object. xml gives back parsed xml as a list.</p>
</td></tr>
<tr><td><code id="bold_seqspec_+3A_sepfasta">sepfasta</code></td>
<td>
<p>(logical) If <code>TRUE</code>, the fasta data is separated into
a list with names matching the processid's for each records. Works with both 'tsv' and 'xml' format.
Note: This means multiple sequences can have the same name if a process id has multiple sequences.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="bold_seqspec_+3A_cleandata">cleanData</code></td>
<td>
<p>(logical) If <code>TRUE</code>, the cell values containing only duplicated values (ex : &quot;COI-5P|COI-5P|COI-5P&quot;) will be reduce to one value (&quot;COI-5P&quot;) and empty string will be change to NA. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="bold_seqspec_+3A_...">...</code></td>
<td>
<p>Further args passed on to
<code><a href="crul.html#topic+verb-GET">crul::verb-GET</a></code>, main purpose being curl
debugging</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Either a data.frame, parsed xml, a http response object, or a list
of length two (data: a data.frame w/o nucleotide column, and fasta: a list
of nucleotides with the processid as name)
</p>


<h3>Large requests</h3>

<p>Some requests can lead to errors. These often have to do with requesting
data for a rank that is quite high in the tree, such as an Order,
for example, Coleoptera. If your request is taking a long time,
it's likely that something will go wrong on the BOLD server side,
or we'll not be able to parse the result here in R because
R can only process strings of a certain length. <code>bold</code>
users have reported errors in which the resulting response from
BOLD is so large that we could not parse it.
</p>
<p>A good strategy for when you want data for a high rank is to
do many separate requests for lower ranks within your target
rank. You can do this manually, or use the function
<code>taxize::downstream</code> to get all the names of a lower
rank within a target rank. There's an example in the README
(https://docs.ropensci.org/bold/#large-data)
</p>


<h3>If a request times out</h3>

<p>This is likely because you're request was for a large number of
sequences and the BOLD service timed out. You still should get
some output, those sequences that were retrieved before the time
out happened. As above, see the README
(https://docs.ropensci.org/bold/#large-data) for an example of
dealing with large data problems with this function.
</p>


<h3>Marker</h3>

<p>Notes from BOLD on the <code>marker</code> param:
&quot;All markers for a specimen matching the search string will be returned.
ie. A record with COI-5P and ITS will return sequence data for both
markers even if only COI-5P was specified.&quot;
</p>
<p>You will likely end up with data with markers that you did not request -
just be sure to filter those out as needed.
</p>


<h3>Note</h3>

<p>If using the <code>taxon</code> parameter with another parameter, if the <code>taxon</code> isn't found in the public database, it will act as if no <code>taxon</code> was specified and try to return all the data for the other specified parameter. You can make sure that the <code>taxon</code> you're looking up has public records with <code><a href="#topic+bold_stats">bold_stats</a></code>.
</p>


<h3>References</h3>

<p>http://v4.boldsystems.org/index.php/resources/api?type=webservices
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
bold_seqspec(taxon='Osmia')
bold_seqspec(taxon='Osmia', format='xml')
bold_seqspec(taxon='Osmia', response=TRUE)
res &lt;- bold_seqspec(taxon='Osmia', sepfasta=TRUE)
res$fasta[1:2]
res$fasta['GBAH0293-06']

# records that match a marker name
res &lt;- bold_seqspec(taxon="Melanogrammus aeglefinus", marker="COI-5P")

# records that match a geographic locality
res &lt;- bold_seqspec(taxon="Melanogrammus aeglefinus", geo="Canada")

## curl debugging
### You can do many things, including get verbose output on the curl call,
### and set a timeout
head(bold_seqspec(taxon='Osmia', verbose = TRUE))
head(bold_seqspec(taxon='Osmia', timeout_ms = 1))

## End(Not run)

</code></pre>

<hr>
<h2 id='bold_specimens'>Search BOLD for specimens.</h2><span id='topic+bold_specimens'></span>

<h3>Description</h3>

<p>Search BOLD for specimens.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bold_specimens(
  taxon = NULL,
  ids = NULL,
  bin = NULL,
  container = NULL,
  institutions = NULL,
  researchers = NULL,
  geo = NULL,
  response = FALSE,
  format = "tsv",
  cleanData = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bold_specimens_+3A_taxon">taxon</code></td>
<td>
<p>(character) One or more taxonomic name. Optional.</p>
</td></tr>
<tr><td><code id="bold_specimens_+3A_ids">ids</code></td>
<td>
<p>(character|integer|numeric) One or more IDs. Optional. IDs
include Sample IDs, Process IDs, Museum IDs and Field IDs.</p>
</td></tr>
<tr><td><code id="bold_specimens_+3A_bin">bin</code></td>
<td>
<p>(character) One or more Barcode Index Number URI. Optional.</p>
</td></tr>
<tr><td><code id="bold_specimens_+3A_container">container</code></td>
<td>
<p>(character) One or more project codes or dataset codes.
Optional.</p>
</td></tr>
<tr><td><code id="bold_specimens_+3A_institutions">institutions</code></td>
<td>
<p>(character) One or more institution's name. Optional.
Institutions are the Specimen Storing Site.</p>
</td></tr>
<tr><td><code id="bold_specimens_+3A_researchers">researchers</code></td>
<td>
<p>(character) One or more researcher names. Optional.
Include collectors and specimen identifiers.</p>
</td></tr>
<tr><td><code id="bold_specimens_+3A_geo">geo</code></td>
<td>
<p>(character) One or more geographic sites.
Includes countries and province/states.</p>
</td></tr>
<tr><td><code id="bold_specimens_+3A_response">response</code></td>
<td>
<p>(logical) Default : FALSE. If TRUE, returns the object from
the Curl call. Useful for debugging and getting more detailed info on the
API call.</p>
</td></tr>
<tr><td><code id="bold_specimens_+3A_format">format</code></td>
<td>
<p>(character) One of xml, json, tsv (default). tsv format gives
back a data.frame object. xml gives back parsed XML as <code>xml_document</code>
object. 'json' (JavaScript Object Notation) and 'dwc' (Darwin Core Archive)
are supported in theory, but the JSON can be malformed, so we don't support
that here, and the DWC option actually returns TSV.</p>
</td></tr>
<tr><td><code id="bold_specimens_+3A_cleandata">cleanData</code></td>
<td>
<p>(logical) If <code>TRUE</code>, the cell values containing only duplicated values (ex : &quot;COI-5P|COI-5P|COI-5P&quot;) will be reduce to one value (&quot;COI-5P&quot;) and empty string will be change to NA. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="bold_specimens_+3A_...">...</code></td>
<td>
<p>Further args passed on to
<code><a href="crul.html#topic+verb-GET">crul::verb-GET</a></code>, main purpose being curl
debugging</p>
</td></tr>
</table>


<h3>Note</h3>

<p>If using the <code>taxon</code> parameter with another parameter, if the <code>taxon</code> isn't found in the public database, it will act as if no <code>taxon</code> was specified and try to return all the data for the other specified parameter. You can make sure that the <code>taxon</code> you're looking up has public records with <code><a href="#topic+bold_stats">bold_stats</a></code>.
</p>


<h3>References</h3>

<p>http://v4.boldsystems.org/index.php/resources/api?type=webservices
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
bold_specimens(taxon='Osmia')
bold_specimens(taxon='Osmia', format='xml')
bold_specimens(taxon='Osmia', response=TRUE)
res &lt;- bold_specimens(taxon='Osmia', format='xml', response=TRUE)
res$url
res$status_code
res$response_headers

# More than 1 can be given for all search parameters
bold_specimens(taxon=c('Coelioxys','Osmia'))

## curl debugging
### These examples below take a long time, so you can set a timeout so that
### it stops by X sec
head(bold_specimens(taxon='Osmia', verbose = TRUE))
# head(bold_specimens(geo='Costa Rica', timeout_ms = 6))

## End(Not run)

</code></pre>

<hr>
<h2 id='bold_stats'>Get BOLD stats</h2><span id='topic+bold_stats'></span>

<h3>Description</h3>

<p>Get BOLD stats
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bold_stats(
  taxon = NULL,
  ids = NULL,
  bin = NULL,
  container = NULL,
  institutions = NULL,
  researchers = NULL,
  geo = NULL,
  dataType = "drill_down",
  response = FALSE,
  simplify = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bold_stats_+3A_taxon">taxon</code></td>
<td>
<p>(character) One or more taxonomic name. Optional.</p>
</td></tr>
<tr><td><code id="bold_stats_+3A_ids">ids</code></td>
<td>
<p>(character|integer|numeric) One or more IDs. Optional. IDs
include Sample IDs, Process IDs, Museum IDs and Field IDs.</p>
</td></tr>
<tr><td><code id="bold_stats_+3A_bin">bin</code></td>
<td>
<p>(character) One or more Barcode Index Number URI. Optional.</p>
</td></tr>
<tr><td><code id="bold_stats_+3A_container">container</code></td>
<td>
<p>(character) One or more project codes or dataset codes.
Optional.</p>
</td></tr>
<tr><td><code id="bold_stats_+3A_institutions">institutions</code></td>
<td>
<p>(character) One or more institution's name. Optional.
Institutions are the Specimen Storing Site.</p>
</td></tr>
<tr><td><code id="bold_stats_+3A_researchers">researchers</code></td>
<td>
<p>(character) One or more researcher names. Optional.
Include collectors and specimen identifiers.</p>
</td></tr>
<tr><td><code id="bold_stats_+3A_geo">geo</code></td>
<td>
<p>(character) One or more geographic sites.
Includes countries and province/states.</p>
</td></tr>
<tr><td><code id="bold_stats_+3A_datatype">dataType</code></td>
<td>
<p>(character) one of &quot;drill_down&quot;(default) or &quot;overview&quot;.
&quot;drill_down&quot;: a detailed summary of information which provides record counts
by BINs, Countries, Storing Institutions, Orders, Families, Genus, Species.
&quot;overview&quot;: the total record counts of BINs, Countries, Storing Institutions,
Orders, Families, Genus, Species. The record counts include all gene markers, not only COI. To see the drill down of markers use <code><a href="#topic+bold_tax_id2">bold_tax_id2</a></code> with &quot;stats&quot; as <code>dataTypes</code>.</p>
</td></tr>
<tr><td><code id="bold_stats_+3A_response">response</code></td>
<td>
<p>(logical) Default : FALSE. If TRUE, returns the object from
the Curl call. Useful for debugging and getting more detailed info on the
API call.</p>
</td></tr>
<tr><td><code id="bold_stats_+3A_simplify">simplify</code></td>
<td>
<p>(logical) whether the returned list should be simplified to a data.frame. See Details.</p>
</td></tr>
<tr><td><code id="bold_stats_+3A_...">...</code></td>
<td>
<p>Further args passed on to
<code><a href="crul.html#topic+verb-GET">crul::verb-GET</a></code>, main purpose being curl
debugging</p>
</td></tr>
</table>


<h3>Value</h3>

<p>By default, returns a nested list with the number of total records, the number of records with a species name, then for each of bins, countries, depositories, order, family, genus and species, the total count and the drill down of the records by up to 10 entities of that category. If <code>simplify</code> is set to TRUE, returns a list of length 2 : the overview data (number of total records, the number of records with a species name, and the total counts) simplified as a data.frame of 1 row and 9 columns and the drill_down data simplified to a one level list of data.frame. When <code>dataType</code> is set to &quot;overview&quot;, returns a nested list with the number of total records, the number of records with a species name, and the total count for each of bins, countries, depositories, order, family, genus and species. If <code>simplify</code> is set to TRUE, returns a data.frame of 1 row and 9 columns.
</p>


<h3>References</h3>

<p>http://v4.boldsystems.org/index.php/resources/api?type=webservices
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
x &lt;- bold_stats(taxon='Osmia')
x$total_records
x$records_with_species_name
x$bins
x$countries
x$depositories
x$order
x$family
x$genus
x$species

# just get all counts
lapply(Filter(is.list, x), `[[`, "count")

bold_stats(taxon='Osmia', dataType = "overview", simplified = TRUE)

x &lt;- bold_stats(taxon='Osmia', simplified = TRUE)
x$overview
x$drill_down

res &lt;- bold_stats(taxon='Osmia', response=TRUE)
res$url
res$status_code
res$response_headers

# More than 1 can be given for all search parameters

## curl debugging
### These examples below take a long time, so you can set a timeout so that
### it stops by X sec
bold_stats(taxon='Osmia', verbose = TRUE)
# bold_stats(geo='Costa Rica', timeout_ms = 6)

## End(Not run)

</code></pre>

<hr>
<h2 id='bold_tax_id-deprecated'>Search BOLD for taxonomy data by BOLD ID.</h2><span id='topic+bold_tax_id-deprecated'></span>

<h3>Description</h3>

<p>Search BOLD for taxonomy data by BOLD ID.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="bold_tax_id-deprecated_+3A_id">id</code></td>
<td>
<p>(integer) One or more BOLD taxonomic identifiers. required.</p>
</td></tr>
<tr><td><code id="bold_tax_id-deprecated_+3A_datatypes">dataTypes</code></td>
<td>
<p>(character) Specifies the datatypes that will be
returned. 'all' returns all data. 'basic' returns basic taxon information.
'images' returns specimen images.</p>
</td></tr>
<tr><td><code id="bold_tax_id-deprecated_+3A_includetree">includeTree</code></td>
<td>
<p>(logical) If <code>TRUE</code> (default: <code>FALSE</code>), returns a list
containing information for parent taxa as well as the specified taxon.</p>
</td></tr>
<tr><td><code id="bold_tax_id-deprecated_+3A_response">response</code></td>
<td>
<p>(logical) Default : FALSE. If TRUE, returns the object from
the Curl call. Useful for debugging and getting more detailed info on the
API call.</p>
</td></tr>
<tr><td><code id="bold_tax_id-deprecated_+3A_...">...</code></td>
<td>
<p>Further args passed on to
<code><a href="crul.html#topic+verb-GET">crul::verb-GET</a></code>, main purpose being curl
debugging</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+bold-deprecated">bold-deprecated</a></code>
</p>

<hr>
<h2 id='bold_tax_id2'>Search BOLD for taxonomy data by BOLD ID.</h2><span id='topic+bold_tax_id2'></span>

<h3>Description</h3>

<p>Search BOLD for taxonomy data by BOLD ID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bold_tax_id2(
  id,
  dataTypes = "basic",
  includeTree = FALSE,
  response = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bold_tax_id2_+3A_id">id</code></td>
<td>
<p>(integer|numeric|character) One or more BOLD taxonomic identifiers. required.</p>
</td></tr>
<tr><td><code id="bold_tax_id2_+3A_datatypes">dataTypes</code></td>
<td>
<p>(character) One or more BOLD data types: 'basic', 'stats', 'geo', 'images'/'img', 'sequencinglabs'/'labs', 'depository'/'depo', 'thirdparty'/'wiki' or 'all' (default: 'basic'). Specifies the information that will be returned, see details.</p>
</td></tr>
<tr><td><code id="bold_tax_id2_+3A_includetree">includeTree</code></td>
<td>
<p>(logical) If <code>TRUE</code> (default: <code>FALSE</code>), returns the information for parent taxa as well as the specified taxon, see details.</p>
</td></tr>
<tr><td><code id="bold_tax_id2_+3A_response">response</code></td>
<td>
<p>(logical) Default : FALSE. If TRUE, returns the object from
the Curl call. Useful for debugging and getting more detailed info on the
API call.</p>
</td></tr>
<tr><td><code id="bold_tax_id2_+3A_...">...</code></td>
<td>
<p>Further args passed on to
<code><a href="crul.html#topic+verb-GET">crul::verb-GET</a></code>, main purpose being curl
debugging</p>
</td></tr>
</table>


<h3>dataTypes</h3>

<p>&quot;basic&quot; returns basic taxonomy info: includes taxid, taxon name, tax rank, tax division, parent taxid, parent taxon name.
&quot;stats&quot; returns specimen and sequence statistics: includes public species count, public BIN count, public marker counts, public record count, specimen count, sequenced specimen count, barcode specimen count, species count, barcode species count.
&quot;geo&quot; returns collection site information: includes country, collection site map.
&quot;images&quot; returns specimen images: includes copyright information, image URL, image metadata.
&quot;sequencinglabs&quot; returns sequencing labs: includes lab name, record count.
&quot;depository&quot; returns specimen depositories: includes depository name, record count.
&quot;thirdparty&quot; returns information from third parties: includes wikipedia_summary summary, wikipedia_summary URL.
&quot;all&quot; returns all information: identical to specifying all data types at once.
</p>


<h3>includeTree</h3>

<p>When <code>includeTree</code> is set to true, for the <code>dataTypes</code> other than 'basic' the information of the parent taxa are identified by their taxonomic id only. To get their ranks and names too, make sure 'basic' is in <code>dataTypes</code>.
</p>


<h3>References</h3>

<p>http://v4.boldsystems.org/index.php/resources/api?type=taxonomy
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bold_tax_name">bold_tax_name</a></code>, <code><a href="#topic+bold_get_attr">bold_get_attr</a></code>, <code><a href="#topic+bold_get_errors">bold_get_errors</a></code>, <code><a href="#topic+bold_get_params">bold_get_params</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
bold_tax_id2(id = 88899)
bold_tax_id2(id = 88899, includeTree = TRUE)
bold_tax_id2(id = 88899, includeTree = TRUE, dataTypes = "stats")
bold_tax_id2(id = c(88899, 125295))

## dataTypes parameter
bold_tax_id2(id = 88899, dataTypes = "basic")
bold_tax_id2(id = 88899, dataTypes = "stats")
bold_tax_id2(id = 88899, dataTypes = "images")
bold_tax_id2(id = 88899, dataTypes = "geo")
bold_tax_id2(id = 88899, dataTypes = "sequencinglabs")
bold_tax_id2(id = 88899, dataTypes = "depository")
bold_tax_id2(id = 88899, dataTypes = "thirdparty")
bold_tax_id2(id = 88899, dataTypes = "all")
bold_tax_id2(id = c(88899, 125295), dataTypes = c("basic", "geo"))
bold_tax_id2(id = c(88899, 125295), dataTypes = c("basic", "stats", "images"))

## Passing in NA
bold_tax_id2(id = NA)
bold_tax_id2(id = c(88899, 125295, NA))

## get http response object only
bold_tax_id2(id = 88899, response = TRUE)
bold_tax_id2(id = c(88899, 125295), response = TRUE)

## curl debugging
bold_tax_id2(id = 88899, verbose = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='bold_tax_name'>Search BOLD for taxonomy data by taxonomic name</h2><span id='topic+bold_tax_name'></span>

<h3>Description</h3>

<p>Search BOLD for taxonomy data by taxonomic name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bold_tax_name(
  name,
  fuzzy = FALSE,
  response = FALSE,
  tax_division = NULL,
  tax_rank = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bold_tax_name_+3A_name">name</code></td>
<td>
<p>(character) One or more scientific names. required.</p>
</td></tr>
<tr><td><code id="bold_tax_name_+3A_fuzzy">fuzzy</code></td>
<td>
<p>(logical) Whether to use fuzzy search or not (default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="bold_tax_name_+3A_response">response</code></td>
<td>
<p>(logical) Default : FALSE. If TRUE, returns the object from
the Curl call. Useful for debugging and getting more detailed info on the
API call.</p>
</td></tr>
<tr><td><code id="bold_tax_name_+3A_tax_division">tax_division</code></td>
<td>
<p>(character) Taxonomic division to filter the results.</p>
</td></tr>
<tr><td><code id="bold_tax_name_+3A_tax_rank">tax_rank</code></td>
<td>
<p>(character) Taxonomic rank to filter the results.</p>
</td></tr>
<tr><td><code id="bold_tax_name_+3A_...">...</code></td>
<td>
<p>Further args passed on to
<code><a href="crul.html#topic+verb-GET">crul::verb-GET</a></code>, main purpose being curl
debugging</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>dataTypes</code> parameter is not supported in this function.
If you want to use that parameter, get an ID from this function and pass
it into <code>bold_tax_id</code>, and then use the <code>dataTypes</code> parameter.
</p>


<h3>Note</h3>

<p>The column 'specimenrecords' in the returned object represents the number of species records in BOLD's <em>Taxonomy Browser</em>, not the number of records in the <em>Public Data Portal</em>. To know the amount of public records available, use <code><a href="#topic+bold_stats">bold_stats</a></code>.
</p>


<h3>References</h3>

<p>Taxonomy API: http://v4.boldsystems.org/index.php/resources/api?type=taxonomy
Taxonomy Browser: https://www.boldsystems.org/index.php/TaxBrowser_Home
Public Data Portal: https://www.boldsystems.org/index.php/Public_BINSearch?searchtype=records
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bold_tax_id">bold_tax_id</a></code>, <code><a href="#topic+bold_get_attr">bold_get_attr</a></code>, <code><a href="#topic+bold_get_errors">bold_get_errors</a></code>, <code><a href="#topic+bold_get_params">bold_get_params</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
bold_tax_name(name='Diplura')
bold_tax_name(name='Osmia')
bold_tax_name(name=c('Diplura','Osmia'))
bold_tax_name(name=c("Apis","Puma concolor","Pinus concolor"))
bold_tax_name(name='Diplur', fuzzy=TRUE)
bold_tax_name(name='Osm', fuzzy=TRUE)

## get http response object only
bold_tax_name(name='Diplura', response=TRUE)
bold_tax_name(name=c('Diplura','Osmia'), response=TRUE)

## Names with no data in BOLD database
bold_tax_name("Nasiaeshna pentacantha")
bold_tax_name(name = "Cordulegaster erronea")
bold_tax_name(name = "Cordulegaster erronea", response=TRUE)

## curl debugging
bold_tax_name(name='Diplura', verbose = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='bold_trace'>Get BOLD trace files</h2><span id='topic+bold_trace'></span><span id='topic+print.boldtrace'></span><span id='topic+read_trace'></span><span id='topic+bold_read_trace'></span>

<h3>Description</h3>

<p>Get BOLD trace files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bold_trace(
  taxon = NULL,
  ids = NULL,
  bin = NULL,
  container = NULL,
  institutions = NULL,
  researchers = NULL,
  geo = NULL,
  marker = NULL,
  dest = NULL,
  overwrite = TRUE,
  progress = TRUE,
  ...
)

## S3 method for class 'boldtrace'
print(x, ...)

read_trace(x)

bold_read_trace(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bold_trace_+3A_taxon">taxon</code></td>
<td>
<p>(character) One or more taxonomic name. Optional.</p>
</td></tr>
<tr><td><code id="bold_trace_+3A_ids">ids</code></td>
<td>
<p>(character|integer|numeric) One or more IDs. Optional. IDs
include Sample IDs, Process IDs, Museum IDs and Field IDs.</p>
</td></tr>
<tr><td><code id="bold_trace_+3A_bin">bin</code></td>
<td>
<p>(character) One or more Barcode Index Number URI. Optional.</p>
</td></tr>
<tr><td><code id="bold_trace_+3A_container">container</code></td>
<td>
<p>(character) One or more project codes or dataset codes.
Optional.</p>
</td></tr>
<tr><td><code id="bold_trace_+3A_institutions">institutions</code></td>
<td>
<p>(character) One or more institution's name. Optional.
Institutions are the Specimen Storing Site.</p>
</td></tr>
<tr><td><code id="bold_trace_+3A_researchers">researchers</code></td>
<td>
<p>(character) One or more researcher names. Optional.
Include collectors and specimen identifiers.</p>
</td></tr>
<tr><td><code id="bold_trace_+3A_geo">geo</code></td>
<td>
<p>(character) One or more geographic sites.
Includes countries and province/states.</p>
</td></tr>
<tr><td><code id="bold_trace_+3A_marker">marker</code></td>
<td>
<p>(character) Returns all records containing matching
marker codes.</p>
</td></tr>
<tr><td><code id="bold_trace_+3A_dest">dest</code></td>
<td>
<p>(character) A directory to write the files to</p>
</td></tr>
<tr><td><code id="bold_trace_+3A_overwrite">overwrite</code></td>
<td>
<p>(logical) Overwrite existing directory and file?</p>
</td></tr>
<tr><td><code id="bold_trace_+3A_progress">progress</code></td>
<td>
<p>(logical) Print progress or not. NOT AVAILABLE FOR NOW.
HOPEFULLY WILL RETURN SOON.</p>
</td></tr>
<tr><td><code id="bold_trace_+3A_...">...</code></td>
<td>
<p>Further args passed on to <code><a href="crul.html#topic+verb-GET">verb-GET</a></code></p>
</td></tr>
<tr><td><code id="bold_trace_+3A_x">x</code></td>
<td>
<p>(list or character) Either the boldtrace object returned from
<code><a href="#topic+bold_trace">bold_trace</a></code> or the path(s) of bold trace file(s).</p>
</td></tr>
</table>


<h3>References</h3>

<p>http://v4.boldsystems.org/index.php/resources/api?type=webservices
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Use a specific destination directory
bold_trace(taxon = "Bombus ignitus", geo = "Japan", dest = "~/mytarfiles")

# Another example
# bold_trace(ids='ACRJP618-11', dest="~/mytarfiles")
# bold_trace(ids=c('ACRJP618-11','ACRJP619-11'), dest="~/mytarfiles")

# read file in
x &lt;- bold_trace(ids=c('ACRJP618-11','ACRJP619-11'), dest="~/mytarfiles")
(res &lt;- read_trace(x$ab1[2]))
# read all files in
(res &lt;- read_trace(x))

# The progress dialog is pretty verbose, so quiet=TRUE is a nice touch,
# but not by default
# Beware, this one take a while
# x &lt;- bold_trace(taxon='Osmia', quiet=TRUE)

if (requireNamespace("sangerseqR", quietly = TRUE)) {
 library("sangerseqR")
 primarySeq(res)
 secondarySeq(res)
 head(traceMatrix(res))
}

## End(Not run)
</code></pre>

<hr>
<h2 id='bold-deprecated'>Deprecated functions in package <span class="pkg">bold</span>.</h2><span id='topic+bold-deprecated'></span><span id='topic+bold_identify_parents'></span><span id='topic+bold_tax_id'></span>

<h3>Description</h3>

<p>The functions listed below are deprecated and will be defunct in
the near future. When possible, alternative functions with similar
functionality are also mentioned. Help pages for deprecated functions are
available at <code>help("&lt;function&gt;-deprecated")</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bold_identify_parents(
  x,
  wide = FALSE,
  taxid = NULL,
  taxon = NULL,
  tax_rank = NULL,
  tax_division = NULL,
  parentid = NULL,
  parentname = NULL,
  taxonrep = NULL,
  specimenrecords = NULL,
  ...
)

bold_tax_id(
  id,
  dataTypes = "basic",
  includeTree = FALSE,
  response = FALSE,
  ...
)
</code></pre>


<h3>Details</h3>

<p>DEPRECATED. See <code><a href="#topic+bold_identify_taxonomy">bold_identify_taxonomy</a></code>. It's faster and gets the accurate  taxonomy directly from the record of the sequence.
</p>
<p>This function gets unique set of taxonomic names from the input
data.frame, then queries <code><a href="#topic+bold_tax_name">bold_tax_name</a></code> to get the
taxonomic ID, passing it to <code><a href="#topic+bold_tax_id">bold_tax_id</a></code> to get the parent
names, then attaches those to the input data.
</p>
<p>Records in the input data that do not have matches for parent names
simply get NA values in the added columns.
</p>


<h3>Value</h3>

<p>a list of the same length as the input
</p>


<h3><code>bold_identify_parents</code></h3>

<p>For <code>bold_identify_parents</code>, use <code><a href="#topic+bold_identify_taxonomy">bold_identify_taxonomy</a></code>.
</p>


<h3>Filtering</h3>

<p>The parameters <code>taxid</code>, <code>taxon</code>, <code>tax_rank</code>, <code>tax_division</code>,
<code>parentid</code>, <code>parentname</code>,<code>taxonrep</code>, and <code>specimenrecords</code> are not used
in the search sent to BOLD, but are used in filtering the data down
to a subset that is closer to the target you want. For all these
parameters, you can use regex strings since we use <code><a href="base.html#topic+grep">grep</a></code>
internally to match. Filtering narrows down to the set that matches your
query, and removes the rest. The data.frame that we filter on with these
parameters internally is the result of a call to the
<code><a href="#topic+bold_tax_name">bold_tax_name</a></code> function.
</p>


<h3>wide vs long format</h3>

<p>When <code>wide = FALSE</code> you get many rows for each record. Essentially,
we <code>cbind</code> the taxonomic classification onto the one row from the
result of <code><a href="#topic+bold_identify">bold_identify</a></code>, giving as many rows as there are
taxa in the taxonomic classification.
</p>
<p>When <code>wide = TRUE</code> you get one row for each record - thus the
dimensions of the input data stay the same. For this option, we take just
the rows for taxonomic ID and name for each taxon in the taxonomic
classification, and name the columns by the taxon rank, so you get
<code>phylum</code> and <code>phylum_id</code>, and so on.
</p>


<h3><code>bold_tax_id</code></h3>

<p>For <code>bold_tax_id</code>, use <code><a href="#topic+bold_tax_id2">bold_tax_id2</a></code>.
</p>


<h3>References</h3>

<p>http://v4.boldsystems.org/index.php/resources/api?type=taxonomy
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bold_tax_name">bold_tax_name()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
df &lt;- bold_identify(sequences = sequences$seq2)

# long format
out &lt;- bold_identify_parents(df)
str(out)
head(out[[1]])

# wide format
out &lt;- bold_identify_parents(df, wide = TRUE)
str(out)
head(out[[1]])

x &lt;- bold_seq(taxon = "Satyrium")
out &lt;- bold_identify(c(x[[1]]$sequence, x[[13]]$sequence))
res &lt;- bold_identify_parents(out)
res

x &lt;- bold_seq(taxon = 'Diplura')
out &lt;- bold_identify(vapply(x, "[[", "", "sequence")[1:20])
res &lt;- bold_identify_parents(out)

## End(Not run)
## Not run: 
bold_tax_id(id = 88899)
bold_tax_id(id = 88899, includeTree = TRUE)
bold_tax_id(id = 88899, includeTree = TRUE, dataTypes = "stats")
bold_tax_id(id = c(88899,125295))

## dataTypes parameters
bold_tax_id(id = 88899, dataTypes = "basic")
bold_tax_id(id = 88899, dataTypes = "stats")
bold_tax_id(id = 88899, dataTypes = "images")
bold_tax_id(id = 88899, dataTypes = "geo")
bold_tax_id(id = 88899, dataTypes = "sequencinglabs")
bold_tax_id(id = 88899, dataTypes = "depository")
bold_tax_id(id = c(88899, 125295), dataTypes = "geo")
bold_tax_id(id = c(88899, 125295), dataTypes = "images")

## Passing in NA
bold_tax_id(id = NA)
bold_tax_id(id = c(88899, 125295, NA))

## get http response object only
bold_tax_id(id = 88899, response=TRUE)
bold_tax_id(id = c(88899, 125295), response=TRUE)

## curl debugging
bold_tax_id(id = 88899, verbose = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='sequences'>List of 3 nucleotide sequences to use in examples for the
<code><a href="#topic+bold_identify">bold_identify</a></code> function</h2><span id='topic+sequences'></span>

<h3>Description</h3>

<p>List of 3 nucleotide sequences to use in examples for the
<code><a href="#topic+bold_identify">bold_identify</a></code> function
</p>


<h3>Format</h3>

<p>list of length 3
</p>


<h3>Details</h3>

<p>Each sequence is a character string, of lengths 410, 600, and 696.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
