<!DOCTYPE html><html lang="en"><head><title>Help for package BoolFilter</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BoolFilter}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BoolFilter-package'>
<p>Optimal Estimation of Partially-Observed Boolean Dynamical Systems</p></a></li>
<li><a href='#BKF'>
<p>Boolean Kalman Filter</p></a></li>
<li><a href='#BKS'>
<p>Boolean Kalman Smoother</p></a></li>
<li><a href='#melanoma'>
<p>Melanoma Regulatory Network</p></a></li>
<li><a href='#MMAE'>
<p>Multiple Model Adaptive Estimation</p></a></li>
<li><a href='#p53net_DNAdsb0'>
<p>p53 Negative-Feedback Gene Regulatory Boolean Network</p></a></li>
<li><a href='#p53net_DNAdsb1'>
<p>p53 Negative-Feedback Gene Regulatory Boolean Network</p></a></li>
<li><a href='#plotTrajectory'>
<p>Plot state variables of Boolean Regulatory Systems</p></a></li>
<li><a href='#simulateNetwork'>
<p>Simulate Boolean Network</p></a></li>
<li><a href='#SIR_BKF'>
<p>Particle Filter</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Optimal Estimation of Partially Observed Boolean Dynamical
Systems</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-01-08</td>
</tr>
<tr>
<td>Author:</td>
<td>Levi McClenny, Mahdi Imani, Ulisses Braga-Neto</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Levi McClenny &lt;levimcclenny@tamu.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.15.0), Rlab, BoolNet</td>
</tr>
<tr>
<td>BuildManual:</td>
<td>yes</td>
</tr>
<tr>
<td>Description:</td>
<td>Tools for optimal and approximate state estimation as well as
    network inference of Partially-Observed Boolean Dynamical Systems.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-01-09 15:52:01 UTC; levimcclenny</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-01-09 17:11:26</td>
</tr>
</table>
<hr>
<h2 id='BoolFilter-package'>
Optimal Estimation of Partially-Observed Boolean Dynamical Systems
</h2><span id='topic+BoolFilter-package'></span><span id='topic+BoolFilter'></span>

<h3>Description</h3>

<p>Tools for optimal and approximate state estimation as well as network inference of Partially-Observed Boolean Dynamical Systems.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> BoolFilter</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2017-01-08</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> Artistic-2.0</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Estimation of Partially-Observed Boolean Dynamical Systems (POBDS) is handled by various algorithms, revolving primarily around the Boolean Kalman Filter (<cite><a href="#topic+BKF">BKF</a></cite>). Other similar algorithms have been developed and documented in the BoolFilter package, including <cite><a href="#topic+BKS">BKS</a></cite> for batch datasets and the <cite><a href="#topic+SIR_BKF">SIR_BKF</a></cite> for large POBDS. 
</p>
<p>These algorithms can be run on data with various types of noise including:
</p>

<ul>
<li><p>Bernoulli
</p>
</li>
<li><p>Gaussian
</p>
</li>
<li><p>Poisson
</p>
</li>
<li><p>Negative-Binomial
</p>
</li></ul>

<p>These types of noise models are included to handle different potential observation noise that could be encountered in real applications. 
</p>
<p>These observation noise models can be simulated in the <cite><a href="#topic+simulateNetwork">simulateNetwork</a></cite> function, which can generate both state and observation trajectories of user defined length and observation model preference. 
</p>
<p>Additionally, the <cite><a href="#topic+plotTrajectory">plotTrajectory</a></cite> function can be implemented to visualize the trajectories of state variables.
</p>


<h3>Author(s)</h3>

<p>Levi McClenny, Mahdi Imani, Ulisses Braga-Neto
</p>
<p>Maintainer: Levi McClenny &lt;levimcclenny@tamu.edu&gt;
</p>


<h3>References</h3>

<p>Braga-Neto U. Optimal state estimation for Boolean dynamical systems. In 2011 Conference Record of the Forty Fifth Asilomar Conference on Signals, Systems and Computers (ASILOMAR) 2011 Nov 6 (pp. 1050-1054). IEEE.
<br />
<br />
Imani, M., &amp; Braga-Neto, U. Maximum-likelihood adaptive filter for partially-observed boolean dynamical systems. IEEE transaction on Signal Processing, 65:359-371, 2017.
</p>
<p>Further references are listed in the corresponding reference sections
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(p53net_DNAdsb0)
 
#Simulate data from a Bernoulli observation model
data &lt;- simulateNetwork(p53net_DNAdsb0, n.data = 100, p = 0.02,
                        obsModel = list(type = "Bernoulli",
                                          p = 0.02))
                            
#Derive an estimate of the network using a BKF approach
Results &lt;- BKF(data$Y, p53net_DNAdsb0, .02,
                        obsModel = list(type = "Bernoulli",
                                          p = 0.02))
                        
#View network approximation vs. correct trajectory
plotTrajectory(Results$Xhat,
                labels = p53net_DNAdsb0$genes,
                dataset2 = data$X,
                compare = TRUE)

</code></pre>

<hr>
<h2 id='BKF'>
Boolean Kalman Filter
</h2><span id='topic+BKF'></span>

<h3>Description</h3>

<p>Implements the Boolean Kalman Filter to derive the optimal MMSE estimate of state of a Partially Observed Boolean Dynamical System
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BKF(Y, net, p, obsModel)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BKF_+3A_y">Y</code></td>
<td>

<p>Time series of noisy observations of the Boolean regulatory network. <br />
Each row and column correspond to a specific Boolean variable and time point respectively.
</p>
</td></tr>
<tr><td><code id="BKF_+3A_net">net</code></td>
<td>

<p>A Boolean Network object (specified in BoolNet vernacular) that the time series of observations presented in <code>Y</code> is based on
</p>
</td></tr>
<tr><td><code id="BKF_+3A_p">p</code></td>
<td>

<p>Intensity of Bernoulli process noise
</p>
</td></tr>
<tr><td><code id="BKF_+3A_obsmodel">obsModel</code></td>
<td>

<p>Parameters for the chosen observation model.<br />
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A novel state-space signal model has been proposed for stochastic Boolean dynamical systems observed through noisy measurements, of which the Boolean Kalman Filter is the optimal MMSE estimator. State evolution is governed by Boolean functions (i.e. logic gates) and noise. The current system state, in which transitions between Boolean states can be perturbed by some process noise, creating an ergodic state transition matrix, is observed through an arbitrary noisy measurement. The optimal recursive MMSE estimator for this model is called the Boolean Kalman Filter (BKF), and an efficient algorithm has been proposed for its exact computation. This algorithm is presented here.
</p>
<p>The Boolean Kalman Filtering algorithm can handle various observation models, including Bernoulli, Gaussian, Poisson, and Negative-Binomial, based on the input to the <code>obsModel</code> parameters. 
</p>

<ul>
<li><p>Bernoulli observation model requires only one parameter, aside from declaring the type, e.g. <br /> <br />
<code>obsModel = list(type = 'Bernoulli', q = 0.05)</code>
<br />
</p>
</li>
<li> <p>Gaussian observation model requires a vector of the observation parameters, which include the mean and standard deviation of Boolean variables in inactivated and activated states. This will be defined as a vector, e.g. <br /> <br />
<code>mu0 = 1</code> <br />
<code>sigma0 = 2</code><br />
<code>mu1 = 5</code><br />
<code>sigma1 = 2</code><br />
<code>obsModel = list(type = 'Gaussian', model = c(mu0, sigma0, mu1, sigma1))</code>
<br />
</p>
</li>
<li><p>The Poisson observation model requires a list of parameters. This list will have 3 entries in addition to the type definition, for a total of 4 entries:
</p>

<ul>
<li><p>Sequencing depth <code>s</code>
</p>
</li>
<li><p>Baseline espression in inactivated state, referred to as <code>mu</code>
</p>
</li>
<li><p>The differential expression, referred to as <code>delta</code>, which must be input as a vector of the same length as the number of Boolean variables in the network.
</p>
</li></ul>

<p>In this way, the user can define the exact observation parameter for each individual gene. For a 4-gene network, a potential <code>obsModel</code> parameter for a Poisson distribution could be defined as:<br /> <br />
<code>obsModel = list(type = 'Poisson', s = 10.875, mu = 0.01, delta = c(2, 2, 2, 2))</code>
<br />
</p>
</li>
<li>
<p>Negative-Binomial observation model also requires a list of parameters. This list will have 4 entries in addition to the type definition, for a total of 5 entries:
</p>

<ul>
<li><p>Sequencing depth <code>s</code>
</p>
</li>
<li><p>Baseline espression in inactivated state, referred to as <code>mu</code>
</p>
</li>
<li><p>Differential expression, referred to as <code>delta</code>,  which must be input as a vector of the same length as the number of Boolean variables in the network.
</p>
</li>
<li><p>Inverse Dispersion, referred to as <code>phi</code>, which must also be input as a vector of the same length as the number of Boolean variables in the network.
</p>
</li></ul>

<p>For a 4-gene network, a potential <code>obsModel</code> parameter for a Negative-Binomial observation model could be defined as:<br /> <br />
<code>delta = c(2, 2, 2, 2)</code> <br />
<code>phi = c(3, 3, 3, 3)</code> <br />
<code>obsModel = list(type = 'NB', s = 10.875, mu = 0.01, delta, phi)</code>
<br />
</p>
</li></ul>



<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Xhat</code></td>
<td>
<p>Estimate the state of of the Partially-Observed Boolean Dynamical System based on the recursive BKF algorithm</p>
</td></tr>
<tr><td><code>MSE</code></td>
<td>
<p>Mean Squared Error of the estimate returned by the BKF algorithm for each time instance</p>
</td></tr>
<tr><td><code>Beta</code></td>
<td>
<p>Normalized PDV for each time instance</p>
</td></tr>
</table>


<h3>Source</h3>

<p>Braga-Neto, U. (2011, November). Optimal state estimation for Boolean dynamical systems. In Signals, Systems and Computers (ASILOMAR), 2011 Conference Record of the Forty Fifth Asilomar Conference on (pp. 1050-1054). IEEE.
<br />
<br />
Imani, M., &amp; Braga-Neto, U. Maximum-likelihood adaptive filter for partially-observed boolean dynamical systems. IEEE transaction on Signal Processing, 65:359-371, 2017.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(p53net_DNAdsb0)

obsModel = list(type = 'NB', 
                   s = 10.875, 
                  mu = 0.01, 
               delta = c(2, 2, 2, 2), 
                 phi = c(3, 3, 3, 3))

#Simulate a network with Negative-Binomial observation model
data &lt;- simulateNetwork(p53net_DNAdsb0, n.data = 100, p = 0.02, obsModel)
          
#Derive the optimal estimate of the network using a BKF approach
Results &lt;- BKF(data$Y, p53net_DNAdsb0, p = 0.02, obsModel)

</code></pre>

<hr>
<h2 id='BKS'>
Boolean Kalman Smoother
</h2><span id='topic+BKS'></span>

<h3>Description</h3>

<p>Genereates the optimal MMSE estimate of the state of a Partially-Observed Boolean Dynamical System by implementing a Boolean Kalman Smoother to batch data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BKS(Y, net, p, obsModel)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BKS_+3A_y">Y</code></td>
<td>

<p>Time series of noisy observations of the Boolean regulatory network. <br />
Each row and column correspond to a specific Boolean variable and time point respectively.
</p>
</td></tr>
<tr><td><code id="BKS_+3A_net">net</code></td>
<td>

<p>A Boolean Network object (specified in BoolNet vernacular) that the time series of observations presented in <code>Y</code> is based on
</p>
</td></tr>
<tr><td><code id="BKS_+3A_p">p</code></td>
<td>

<p>Intensity of Bernoulli process noise
</p>
</td></tr>
<tr><td><code id="BKS_+3A_obsmodel">obsModel</code></td>
<td>

<p>Parameters for the chosen observation model.<br />
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In the event that a sequence of measurements is available offline, the BKS can be used for computation of the optimal MMSE of smoothed trajectory.
</p>
<p>The Boolean Kalman Smoother algorithm can handle various observation models, including Bernoulli, Gaussian, Poisson, and Negative-Binomial, based on the input to the <code>obsModel</code> parameter.
</p>
<p>The obsModel parameter is defined the same as the <code>Boolean Kalman Filter</code> and <code>simulateNetwork</code> functions, reference the documentation for <cite><a href="#topic+BKF">BKF</a></cite> or <cite><a href="#topic+simulateNetwork">simulateNetwork</a></cite> for details.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Xhat</code></td>
<td>
<p>Estimate of the sate of the Partially-Observed Boolean Dynamical System based on the BKS algorithm</p>
</td></tr>
<tr><td><code>MSE</code></td>
<td>
<p>Mean Squared Error of the estimate returned by the BKS algorithm for each time instance</p>
</td></tr>
</table>


<h3>Source</h3>

<p>Imani, M., &amp; Braga-Neto, U. (2015, December). Optimal state estimation for boolean dynamical systems using a boolean Kalman smoother. In 2015 IEEE Global Conference on Signal and Information Processing (GlobalSIP) (pp. 972-976). IEEE.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(p53net_DNAdsb0)

obsModel = list(type = 'Bernoulli', q = 0.02)

#Simulate a network with Bernoulli observation noise
data &lt;- simulateNetwork(p53net_DNAdsb0, n.data = 100, p = 0.02, obsModel)
          
#Derive the optimal estimate of state of the network using the BKS algorithm
Results &lt;- BKS(data$Y, p53net_DNAdsb0, p = 0.02, obsModel)

</code></pre>

<hr>
<h2 id='melanoma'>
Melanoma Regulatory Network
</h2><span id='topic+melanoma'></span>

<h3>Description</h3>

<p>Melanoma Boolean regulatory network as described by sM.R. Yousefi, et al.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(melanoma)</code></pre>


<h3>Details</h3>

<p>The data consists of activities of 10 genes in melanoma regulatory network. The 10 genes are as follows: WNT5A, pirin, S100P, RET1, MMP3, PHOC, MART1, HADHB, synuclein, and STC2.
</p>


<h3>Source</h3>

<p>Yousefi, Mohammadmahdi R., and Edward R. Dougherty. &quot;Intervention in gene regulatory networks with maximal phenotype alteration.&quot;&quot; Bioinformatics 29.14 (2013): 1758-1767.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(melanoma)

data &lt;- simulateNetwork(melanoma, n.data = 100, p = .02,
                      obsModel = list(type = 'Bernoulli', 
                                      q = 0.05))
</code></pre>

<hr>
<h2 id='MMAE'>
Multiple Model Adaptive Estimation
</h2><span id='topic+MMAE'></span>

<h3>Description</h3>

<p>This function implements the Multiple Model Adaptive Estimation (MMAE) algorithm, which implements a bank of Boolean Kalman Filters running in parallel on a dataset in order to estimate model parameters. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MMAE(data, net, p, threshold, Prior = NA, obsModel = NA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MMAE_+3A_data">data</code></td>
<td>

<p>Data from which the MMAE algorithm should attempt to estimate parameters
</p>
</td></tr>
<tr><td><code id="MMAE_+3A_net">net</code></td>
<td>

<p>A vector of potential networks, defined in the BoolNet <code>loadNetwork</code> vernacular, from which the MMAE algorithm should choose the most probable network for the given data in <code>Y</code>
</p>
</td></tr>
<tr><td><code id="MMAE_+3A_p">p</code></td>
<td>

<p>A vector containing different possible intensities of Bernoulli process noise, from which the MMAE algorithm should choose the most likely
</p>
</td></tr>
<tr><td><code id="MMAE_+3A_threshold">threshold</code></td>
<td>

<p>A posterior probability threshold value (between 0 and 1) for which the MMAE should stop running
</p>
</td></tr>
<tr><td><code id="MMAE_+3A_prior">Prior</code></td>
<td>

<p>A vector of the prior probabilities of the possible combinations of the various networks and process noise parameters input to the algorithm. The vector must be of size <code>length(net)*length(p)</code>. The entries into the vector will follow the pattern:
<br />
<br />
net1, p1<br />
net1, p2<br />
...<br />
netm, pn<br />
<br />
where m is the total number of potential networks and n is the total number of the possible <code>p</code> values.
The default prior ovse models is uniform.
</p>
</td></tr>
<tr><td><code id="MMAE_+3A_obsmodel">obsModel</code></td>
<td>

<p>Parameters for the chosen observation model.<br />
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The MMAE algorithm will return which network and process noise value is most likely to have generated the given dataset supplied in <code>Y</code>. 
<br />
<br />
If the posterior probability for any of the combinations of networks and process noise parameters fails to surpass the value defined in the <code>threshold</code> variable, the algorithm will output that no decision could be made using the given data. 
</p>


<h3>Source</h3>

<p>Imani, M., &amp; Braga-Neto, U. (2015, November). Optimal gene regulatory network inference using the boolean kalman filter and multiple model adaptive estimation. In 2015 49th Asilomar Conference on Signals, Systems and Computers (pp. 423-427). IEEE.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#load potential networks
data(p53net_DNAdsb0)
data(p53net_DNAdsb1)

net1 &lt;- p53net_DNAdsb0
net2 &lt;- p53net_DNAdsb1

#define observation model
observation = list(type = 'NB', s = 10.875, mu = 0.01, delta = c(2, 2, 2, 2), phi = c(3, 3, 3, 3))

#simulate data using one of the networks and a given 'p'
data &lt;- simulateNetwork(net1, n.data = 100, p = 0.02, obsModel = observation)
       
#run MMAE to determine model selection and parameter estimation
MMAE(data, net=c("net1","net2"), p=c(0.02,0.1,0.15), threshold=0.8, obsModel = observation)


</code></pre>

<hr>
<h2 id='p53net_DNAdsb0'>
p53 Negative-Feedback Gene Regulatory Boolean Network
</h2><span id='topic+p53net_DNAdsb0'></span>

<h3>Description</h3>

<p>The data consists of activities of 4 genes in the well-known p53-MDM2 negative-feedback gene regulatory network when the external input (called dna_dsb) to this system is ON or OFF. The p53 gene codes for the tumor suppressor protein p53 in humans, and its activation plays a critical role in cellular responses to various stress signals that might cause genome instability. The four genes are as follows: ATM, p53, Wip1, and MDM2. The input 'dna_dsb' indicates the presence of DNA double strand breaks. In this dataset, <b>dna_dsb is 0</b>, indicating no external input.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(p53net_DNAdsb0)</code></pre>


<h3>Details</h3>

<p>The p53-MDM2 negative-feedback gene regulatory network described by E. Batchelor et al.
</p>


<h3>Source</h3>

<p>E. Batchelor, A. Loewer, and G. Lahav, &quot;The ups and downs of p53: Understanding protein dynamics in single cells,&quot; Nature Rev. Cancer, vol. 9, no. 5, pp. 371-377, 2009.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(p53net_DNAdsb0)

data &lt;- simulateNetwork(p53net_DNAdsb0, n.data = 100, p = .02,
                      obsModel = list(type = 'Bernoulli', 
                                      q = 0.05))
</code></pre>

<hr>
<h2 id='p53net_DNAdsb1'>
p53 Negative-Feedback Gene Regulatory Boolean Network
</h2><span id='topic+p53net_DNAdsb1'></span>

<h3>Description</h3>

<p>The data consists of activities of 4 genes in the well-known p53-MDM2 negative-feedback gene regulatory network when the external input (called dna_dsb) to this system is ON or OFF. The p53 gene codes for the tumor suppressor protein p53 in humans, and its activation plays a critical role in cellular responses to various stress signals that might cause genome instability. The four genes are as follows: ATM, p53, Wip1, and MDM2. The input 'dna_dsb' indicates the presence of DNA double strand breaks. In this dataset, <b>dna_dsb is 1</b>, indicating no external input.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data('p53net_DNAdsb1')</code></pre>


<h3>Details</h3>

<p>The p53-MDM2 negative-feedback gene regulatory network described by E. Batchelor et al.
</p>


<h3>Source</h3>

<p>E. Batchelor, A. Loewer, and G. Lahav, &quot;The ups and downs of p53: Understanding protein dynamics in single cells,&quot; Nature Rev. Cancer, vol. 9, no. 5, pp. 371-377, 2009.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(p53net_DNAdsb1)

data &lt;- simulateNetwork(p53net_DNAdsb1, n.data = 100, p = .02,
                      obsModel = list(type = 'Bernoulli', 
                                      q = 0.05))
</code></pre>

<hr>
<h2 id='plotTrajectory'>
Plot state variables of Boolean Regulatory Systems
</h2><span id='topic+plotTrajectory'></span>

<h3>Description</h3>

<p>Allows for visualization of Boolean state variables. If <code>compare = TRUE</code>, the trajectory in <b>BLACK</b> is dataset1 and the trajectory in <b>RED</b> is dataset2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotTrajectory(dataset1, 
               labels = NA, 
               dataset2 = NA, 
               compare = FALSE, 
               byrow = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotTrajectory_+3A_dataset1">dataset1</code></td>
<td>

<p>Trajectory to be viewed. Input is limited to 4 variables to allow for concise viewing. Shown in <b>BLACK</b>.
</p>
</td></tr>
<tr><td><code id="plotTrajectory_+3A_labels">labels</code></td>
<td>

<p>Vector of labels to apply to plots
</p>
</td></tr>
<tr><td><code id="plotTrajectory_+3A_dataset2">dataset2</code></td>
<td>

<p>If <code>compare = TRUE</code>, another dataset to be overlayed on dataset1. This is, again, limited to 4 variables. Shown in <b>DASHED RED</b> if <code>compare = TRUE</code>.
</p>
</td></tr>
<tr><td><code id="plotTrajectory_+3A_compare">compare</code></td>
<td>

<p>Set <code>compare = TRUE</code> if an overlay is desired
</p>
</td></tr>
<tr><td><code id="plotTrajectory_+3A_byrow">byrow</code></td>
<td>

<p>Set <code>byrow = FALSE</code> if the input data is not in the format of a single state variable corresponding to a row of the dataset.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(p53net_DNAdsb1)

data &lt;- simulateNetwork(p53net_DNAdsb1, n.data = 100, p = 0.02,
                        obsModel = list(type = 'Bernoulli',
                                        q = 0.05))


plotTrajectory(data$X,              
              labels = p53net_DNAdsb1$genes)
                        
                        
#View both (original state trajectory and observation) datasets overlayed
plotTrajectory(data$X,              
              labels = p53net_DNAdsb1$genes,
              dataset2 = data$Y,
              compare = TRUE)

</code></pre>

<hr>
<h2 id='simulateNetwork'>
Simulate Boolean Network
</h2><span id='topic+simulateNetwork'></span>

<h3>Description</h3>

<p>Simulates a Boolean network and generates noise of user-defined model preference and paramters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulateNetwork(net, n.data, p, obsModel)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simulateNetwork_+3A_net">net</code></td>
<td>

<p>A boolean Network object (specified in BoolNet vernacular) to be simulated
</p>
</td></tr>
<tr><td><code id="simulateNetwork_+3A_n.data">n.data</code></td>
<td>

<p>Length of time-series to simulate
</p>
</td></tr>
<tr><td><code id="simulateNetwork_+3A_p">p</code></td>
<td>

<p>Process noise to build into the simulation. Should be 0&lt;p&lt;0.5.
</p>
</td></tr>
<tr><td><code id="simulateNetwork_+3A_obsmodel">obsModel</code></td>
<td>

<p>Parameters for the chosen observation model.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>simulateNetwork</code> function can simulate many observation models to create observational data. An overvoew of how to use the various types of observation models present in <code>BoolFilter</code> is given below:
</p>

<ul>
<li><p>Bernoulli observation model requires only one parameter, aside from declaring the type, e.g. <br /> <br />
<code>obsModel = list(type = 'Bernoulli', q = 0.05)</code>
<br />
</p>
</li>
<li> <p>Gaussian observation model requires a vector of the observation parameters, which include the mean and standard deviation of Boolean variables in inactivated and activated states. This will be defined as a vector, e.g. <br /> <br />
<code>mu0 = 1</code> <br />
<code>sigma0 = 2</code><br />
<code>mu1 = 5</code><br />
<code>sigma1 = 2</code><br />
<code>obsModel = list(type = 'Gaussian', model = c(mu0, sigma0, mu1, sigma1))</code>
<br />
</p>
</li>
<li><p>Poisson observation model requires a list of parameters. This list will have 3 entries in addition to the type definition, for a total of 4 entries:
</p>

<ul>
<li><p>Sequencing depth <code>s</code>
</p>
</li>
<li><p>Baseline espression in inactivated state, referred to as <code>mu</code>
</p>
</li>
<li><p>The differential expression, referred to as <code>delta</code>, which must be input as a vector of the same length as the number of genes in the network.
</p>
</li></ul>

<p>In this way, the user can define the exact observation parameter for each individual gene. For a 4-gene network, a potential <code>obsModel</code> parameter for a Poisson distribution could be defined as:<br /> <br />
<code>obsModel = list(type = 'Poisson', s = 10.875, mu = 0.01, delta = c(2, 2, 2, 2))</code>
<br />
</p>
</li>
<li>
<p>Negative-Binomial observation models also require a list of parameters. This list will have 4 entries in addition to the type definition, for a total of 5 entries:
</p>

<ul>
<li><p>Sequencing depth <code>s</code>
</p>
</li>
<li><p>Baseline espression in inactivated state, referred to as <code>mu</code>
</p>
</li>
<li><p>Differential expression, referred to as <code>delta</code>,  which must be input as a vector of the same length as the number of genes in the network.
</p>
</li>
<li><p>Inverse Dispersion, referred to as <code>phi</code>, which must also be input as a vector of the same length as the number of genes in the network.
</p>
</li></ul>

<p>For a 4-gene network, a potential <code>obsModel</code> parameter for a Negative-Binomial observation model could be defined as:<br /> <br />
<code>delta = c(2, 2, 2, 2)</code> <br />
<code>phi = c(3, 3, 3, 3)</code> <br />
<code>obsModel = list(type = 'NB', s = 10.875, mu = 0.01, delta, phi)</code>
<br />
</p>
</li></ul>



<h3>Value</h3>

<table role = "presentation">
<tr><td><code>X</code></td>
<td>
<p>Original Boolean state trajectory, <b>without</b> observation noise</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>Observation trajectory</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(p53net_DNAdsb1)

#generate data from poisson observation model
dataPoisson &lt;- simulateNetwork(p53net_DNAdsb1, n.data = 100, p = 0.02, 
                          obsModel = list(type = 'Poisson',
                                          s = 10.875, 
                                          mu = 0.01, 
                                          delta = c(2,2,2,2)))

#generate data from Bernoulli observation model
dataBernoulli &lt;- simulateNetwork(p53net_DNAdsb1, n.data = 100, p = 0.02, 
                          obsModel = list(type = 'Bernoulli',
                                          q = 0.05))
</code></pre>

<hr>
<h2 id='SIR_BKF'>
Particle Filter
</h2><span id='topic+SIR_BKF'></span>

<h3>Description</h3>

<p>Performs the SIR-BKF algorithm approach in order to approximate the optimal MMSE estimate of state variables of a Partially-Observed Boolean Dynamical System. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SIR_BKF(Y, N, alpha, net, p, obsModel)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SIR_BKF_+3A_y">Y</code></td>
<td>

<p>Time series of noisy observations of the boolean regulatory network. <br />
Each row and column correspond to a specific Boolean variable and time point respectively.
</p>
</td></tr>
<tr><td><code id="SIR_BKF_+3A_n">N</code></td>
<td>

<p>Number of particles used in approximation
</p>
</td></tr>
<tr><td><code id="SIR_BKF_+3A_alpha">alpha</code></td>
<td>

<p>Parameter denoting the cutoff threshold for resampling
</p>
</td></tr>
<tr><td><code id="SIR_BKF_+3A_net">net</code></td>
<td>

<p>A Boolean Network object (specified in BoolNet vernacular) that the time series of observations presented in <code>Y</code> is based on
</p>
</td></tr>
<tr><td><code id="SIR_BKF_+3A_p">p</code></td>
<td>

<p>Intensity of Bernoulli process noise
</p>
</td></tr>
<tr><td><code id="SIR_BKF_+3A_obsmodel">obsModel</code></td>
<td>

<p>Parameters for the chosen observation model.<br />
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Exact optimal state estimation for partially-observed Boolean dynamical systems may become impractical computationally if system dimensionality is large. This opens the floor for an approximation algorithm to handle cases of high-dimensional systems. The SIR-BKF Particle Filtering algorithm contained in BoolFilter handles situations in which the Boolean Kalman Filter proves too computationally inefficient. 
</p>
<p>The Particle Filtering approximation, like the Boolean Kalman Filter, can handle various observation models, including Bernoulli, Gaussian, Poisson, and Negative-Binomial, based on the input to the <code>obsModel</code> paramter. 
</p>
<p>The obsModel parameter is defined the same as the <code>Boolean Kalman Filter</code> and <code>simulateNetwork</code> functions, reference the documentation for <cite><a href="#topic+BKF">BKF</a></cite> or <cite><a href="#topic+simulateNetwork">simulateNetwork</a></cite> for details.
</p>


<h3>Source</h3>

<p>Braga-Neto U. Particle filtering approach to state estimation in Boolean dynamical systems. In Global Conference on Signal and Information Processing (GlobalSIP), 2013 IEEE 2013 Dec 3 (pp. 81-84). IEEE.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(cellcycle)

obsModel = list(type = 'Gaussian', 
               model = c(mu0 = 1, sigma0 = 2, mu1 = 5, sigma1 = 2))

#generate data from Negative Binomial observation model for the
#10-gene Mammalian Cell Cycle Network
data &lt;- simulateNetwork(cellcycle, n.data = 100, p = 0.02, obsModel)

#perform SIR-BKF algorithm
Results &lt;- SIR_BKF(data$Y, N = 1000, alpha = 0.95, cellcycle, p = 0.02, obsModel)
                                          
                                        

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
