<!DOCTYPE html><html><head><title>Help for package BootMRMR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BootMRMR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bmrmr.pval.cutoff'><p>Selection of informative geneset based on statistical significance value using Bootstrap-MRMR technique</p></a></li>
<li><a href='#bmrmr.weight.cutoff'><p>Selection of informative geneset using gene weights obtained from the Bootstrap-MRMR technique</p></a></li>
<li><a href='#bootmr.weight'><p>Computation of weights for informative genes/ geneset selection using Bootstrap-MRMR technique</p></a></li>
<li><a href='#geneslect.f'><p>Informative gene set selection using F-score</p></a></li>
<li><a href='#mbmr.pval.cutoff'><p>Selection of informative geneset based on statistical significance value using Modified Bootstrap MRMR technique</p></a></li>
<li><a href='#mbmr.weight.cutoff'><p>Identification of informative geneset based on weights obtained from Modified Bootstrap-MRMR technique</p></a></li>
<li><a href='#mrmr.cutoff'><p>Informative geneset selection using MRMR weights</p></a></li>
<li><a href='#pval.bmrmr'><p>Compuation of statistical significance values for genes using Bootstrap-MRMR technique</p></a></li>
<li><a href='#pval.mbmr'><p>Computation of statistical significance values for genes using Modified Bootstrap MRMR technique for a particular trait/condition</p></a></li>
<li><a href='#rice_salt'>
<p>A gene expression dataset of rice under salinity stress</p></a></li>
<li><a href='#topsis.meth'><p>Selection of optimal gene selection method(s)/method(s) through multi-criteria decision analysis</p></a></li>
<li><a href='#weight.mbmr'><p>Computation of weights for informative gene selection using Modified Bootstrap MRMR technique</p></a></li>
<li><a href='#Weights.mrmr'><p>Computation of MRMR weights for gene selection</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Bootstrap-MRMR Technique for Informative Gene Selection</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-08-30</td>
</tr>
<tr>
<td>Author:</td>
<td>Samarendra Das &lt;samarendra4849@gmail.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Samarendra Das &lt;samarendra4849@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.1)</td>
</tr>
<tr>
<td>Description:</td>
<td>Selection of informative features like genes, transcripts, RNA seq, etc. using Bootstrap Maximum Relevance and Minimum Redundancy technique from a given high dimensional genomic dataset. Informative gene selection involves identification of relevant genes and removal of redundant genes as much as possible from a large gene space. Main applications in high-dimensional expression data analysis (e.g. microarray data, NGS expression data and other genomics and proteomics applications).</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>Yes</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-09-12 06:47:39 UTC; samarendra</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-09-12 15:08:51</td>
</tr>
</table>
<hr>
<h2 id='bmrmr.pval.cutoff'>Selection of informative geneset based on statistical significance value using Bootstrap-MRMR technique</h2><span id='topic+bmrmr.pval.cutoff'></span>

<h3>Description</h3>

<p>The informative geneset which has maximum relevance with target class/trait and minimum redundancy among genes based on statistical significance values computed from the Bootstrap-MRMR technique.</p>


<h3>Usage</h3>

<pre><code class='language-R'>bmrmr.pval.cutoff(x, y, s, Q, n)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bmrmr.pval.cutoff_+3A_x">x</code></td>
<td>
<p>x is a N by p data frame of gene expression values where rows represent genes and columns represent samples or subjects or time point. Each cell entry represents the expression level of a gene in a sample/subject (row names of x as gene names or gene ids).</p>
</td></tr>
<tr><td><code id="bmrmr.pval.cutoff_+3A_y">y</code></td>
<td>
<p>y is a p by 1 numeric vector with entries 1 or -1 representing sample labels, where, 1/-1 represents the sample label of subjects/ samples for stress/control condition(for two class problems).</p>
</td></tr>
<tr><td><code id="bmrmr.pval.cutoff_+3A_s">s</code></td>
<td>
<p>s is a scalar representing the number of bootstrap generated, s must be sufficiently large (i.e. number of times  bootstrap samples are generated.</p>
</td></tr>
<tr><td><code id="bmrmr.pval.cutoff_+3A_q">Q</code></td>
<td>
<p>Q is a scalar representing the quartile value of the rankscores of genes (lies within 1/N to 1), usually the second quartile, i.e. 0.5 or third quartile i.e. 0.75 may be taken.</p>
</td></tr>
<tr><td><code id="bmrmr.pval.cutoff_+3A_n">n</code></td>
<td>
<p>n is a scalar representing the size of the informative geneset to be obtained.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a list of the genes/informative geneset which are highly relevant to the particular trait/condition under investigation and minimal redundant among themselves.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rice_salt)
x=as.data.frame(rice_salt[-1,])
y=as.numeric(rice_salt[1,])
s=80
Q=0.5
n=20
bmrmr.pval.cutoff(x, y, s, Q, n)
</code></pre>

<hr>
<h2 id='bmrmr.weight.cutoff'>Selection of informative geneset using gene weights obtained from the Bootstrap-MRMR technique</h2><span id='topic+bmrmr.weight.cutoff'></span>

<h3>Description</h3>

<p>The function enables to find set of informative genes which are obtained based on weights computed from Bootstrap-MRMR technique.</p>


<h3>Usage</h3>

<pre><code class='language-R'>bmrmr.weight.cutoff(x, y, s, n)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bmrmr.weight.cutoff_+3A_x">x</code></td>
<td>
<p>x is a N by p dataframe of gene expression, where rows are genes and columns are as samples or subjects (gene names are taken as row names). Each cell or entry represents the expression level of a gene for a sample or subject.</p>
</td></tr>   
<tr><td><code id="bmrmr.weight.cutoff_+3A_y">y</code></td>
<td>
<p>y is a p by 1 numeric vector having elements as 1 and -1 representing the sample labels of samples or subjects (for two class problems, i.e. stress or control respectively).</p>
</td></tr>
<tr><td><code id="bmrmr.weight.cutoff_+3A_s">s</code></td>
<td>
<p>s is a numeric constant representing the number of bootstrap samples drawn (s must be sufficiently large)</p>
</td></tr>
<tr><td><code id="bmrmr.weight.cutoff_+3A_n">n</code></td>
<td>
<p>n must be a numeric constant representing the number of informative genes to be selected from the large gene space.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a set of genes, which are highly informative to the trait or condition under consideration based on the computed weights form Bootstrap-MRMR technique.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rice_salt)
x=as.data.frame(rice_salt[-1,])
y=as.numeric(rice_salt[1,])
s=80
n=20
bmrmr.weight.cutoff(x, y, s, n)
</code></pre>

<hr>
<h2 id='bootmr.weight'>Computation of weights for informative genes/ geneset selection using Bootstrap-MRMR technique</h2><span id='topic+bootmr.weight'></span>

<h3>Description</h3>

<p>The function computes the weights associated with each genes for a given dataset using Bootstrap-MRMR technique.</p>


<h3>Usage</h3>

<pre><code class='language-R'>bootmr.weight(x, y, s, plot)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bootmr.weight_+3A_x">x</code></td>
<td>
<p>x is a N by p dataframe of gene expression, where rows are genes and columns are as samples/subjects (gene names are taken as row names). Each cell/entry represents the expression level of a gene in a sample/subject.</p>
</td></tr>   
<tr><td><code id="bootmr.weight_+3A_y">y</code></td>
<td>
<p>y is a p by 1 numeric vector having elements as 1/-1 representing the sample labels of samples/subjects (for two class problems, i.e. stress/control)</p>
</td></tr>
<tr><td><code id="bootmr.weight_+3A_s">s</code></td>
<td>
<p>s is a numeric constant representing the number of bootstrap samples drawn (s must be sufficiently large)</p>
</td></tr>
<tr><td><code id="bootmr.weight_+3A_plot">plot</code></td>
<td>
<p>plot is a character string must either take logical value TRUE/FALSE representing whether the plot of the gene weights of all genes in the dataset needs to be constructed or not.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function returns a vector of weights associated with each genes computed from Bootstrap-MRMR technique for a given dataset.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(rice_salt)
x=as.data.frame(rice_salt[-1,])
y=as.numeric(rice_salt[1,])
s=80
bootmr.weight(x, y, s, plot=FALSE)
</code></pre>

<hr>
<h2 id='geneslect.f'>Informative gene set selection using F-score</h2><span id='topic+geneslect.f'></span>

<h3>Description</h3>

<p>The function returns geneset which is informative for a particular trait/condition using F-score as the gene selection criterion.</p>


<h3>Usage</h3>

<pre><code class='language-R'>geneslect.f(x, y, s)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geneslect.f_+3A_x">x</code></td>
<td>
<p>x is a N by p dataframe of gene expression, where, rows represent as genes and columns as samples/subjects (with row names as gene names/ids).</p>
</td></tr>
<tr><td><code id="geneslect.f_+3A_y">y</code></td>
<td>
<p>y is a p by 1 numeric vector of 1 and -1, where 1/-1 indicates the class label of the samples/subjects either of two classes (e.g. stress and control).</p>
</td></tr>
<tr><td><code id="geneslect.f_+3A_s">s</code></td>
<td>
<p>s is a numeric constant (&lt; N) representing the number of genes to be selected from the large gene space.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function identifies the genes/ geneset which is informative for the particular trait/condition using F-score as a criterion.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rice_salt)
x=as.data.frame(rice_salt[-1,])
y=as.numeric(rice_salt[1,])
s=20
geneslect.f (x, y, s)
</code></pre>

<hr>
<h2 id='mbmr.pval.cutoff'>Selection of informative geneset based on statistical significance value using Modified Bootstrap MRMR technique</h2><span id='topic+mbmr.pval.cutoff'></span>

<h3>Description</h3>

<p>The informative geneset which has maximum relevance with target class/trait and minimum redundancy among genes are selected based on p-values obtained from Modified Bootstrap MRMR technique.</p>


<h3>Usage</h3>

<pre><code class='language-R'>mbmr.pval.cutoff(x, y, m, s, Q, n)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mbmr.pval.cutoff_+3A_x">x</code></td>
<td>
<p>x is a N by p data frame of gene expression values where rows represent genes and columns represent samples/subject/time point. Each cell entry represents the expression level of a gene in a sample/subject (row names of x as gene names/gene ids).</p>
</td></tr>
<tr><td><code id="mbmr.pval.cutoff_+3A_y">y</code></td>
<td>
<p>y is a p by 1 numeric vector with entries 1/-1 representing sample labels, where 1/-1 represents the sample label of subjects/ samples for stress/control condition  (for two class problems).</p>
</td></tr>
<tr><td><code id="mbmr.pval.cutoff_+3A_m">m</code></td>
<td>
<p>m is a scalar representing the size of the Modified Bootstrap Sample (i.e. Out of p samples/subjects, m samples/subjects are randomly drawn with replacement, which constitutes one Modified Bootstrap Sample).</p>
</td></tr>
<tr><td><code id="mbmr.pval.cutoff_+3A_s">s</code></td>
<td>
<p>s is a scalar representing the number of Modified Bootstrap samples (i.e. number of times each of the m samples/subjects will be resampled from p samples/subjects). </p>
</td></tr>
<tr><td><code id="mbmr.pval.cutoff_+3A_q">Q</code></td>
<td>
<p>Q is a scalar representing the quartile value of the gene rankscores (lies within 1/N to 1), usually the second quartile, i.e. 0.5 or third quartile i.e. 0.75</p>
</td></tr>
<tr><td><code id="mbmr.pval.cutoff_+3A_n">n</code></td>
<td>
<p>n is a scalar representing the size of the informative gene set to be obtained.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a list of the genes/ geneset which are highly informative to the particular trait/condition under investigation using Modified Bootstrap MRMR technique.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rice_salt)
x=as.data.frame(rice_salt[-1,])
y=as.numeric(rice_salt[1,])
m=36
s=80
Q=0.5
n=20
mbmr.pval.cutoff(x, y, m, s, Q, n)
</code></pre>

<hr>
<h2 id='mbmr.weight.cutoff'>Identification of informative geneset based on weights obtained from Modified Bootstrap-MRMR technique</h2><span id='topic+mbmr.weight.cutoff'></span>

<h3>Description</h3>

<p>The function enables to find set of informative genes based on weights which are obtained by maximising the relevancy of genes with classes/condition/trait and minimising the redundancy among genes using Modified Bootstrap-MRMR technique</p>


<h3>Usage</h3>

<pre><code class='language-R'>mbmr.weight.cutoff(x, y, m, s, n)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mbmr.weight.cutoff_+3A_x">x</code></td>
<td>
<p>x is a N by p data frame of gene expression values where rows represent genes and columns represent samples/subject/time point. Each cell entry represents the expression level of a gene in a sample/subject (row names of x as gene names/gene ids).</p>
</td></tr>
<tr><td><code id="mbmr.weight.cutoff_+3A_y">y</code></td>
<td>
<p>y is a p by 1 numeric vector with entries 1/-1 representing sample labels, where 1/-1 represents the sample label of subjects/ samples for stress/control condition  (for two class problems).</p>
</td></tr>
<tr><td><code id="mbmr.weight.cutoff_+3A_m">m</code></td>
<td>
<p>m is a scalar representing the size of the Modified Bootstrap Sample (i.e. Out of p samples/subjects, m samples/subjects are randomly drawn with replacement, which constitutes one Modified Bootstrap Sample).</p>
</td></tr>
<tr><td><code id="mbmr.weight.cutoff_+3A_s">s</code></td>
<td>
<p>s is a scalar representing the number of Modified Bootstrap samples (i.e. number of times each of the m samples/subjects will be resampled from p samples/subjects). </p>
</td></tr>
<tr><td><code id="mbmr.weight.cutoff_+3A_n">n</code></td>
<td>
<p>n is a numeric constant representing the number of informative genes to be selected from the large gene space.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a set of genes, which are highly informative to the trait or condition under consideration based Modified Bootstrap-MRMR weights.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rice_salt)
x=as.data.frame(rice_salt[-1,])
y=as.numeric(rice_salt[1,])
m=36
s=80
n=20
mbmr.weight.cutoff(x, y, m, s, n)
</code></pre>

<hr>
<h2 id='mrmr.cutoff'>Informative geneset selection using MRMR weights</h2><span id='topic+mrmr.cutoff'></span>

<h3>Description</h3>

<p>The function returns the informative genes/ geneset for the particular trait/condition under investigation using Maximum Relevance and Minimum Redundancy (MRMR) technique.</p>


<h3>Usage</h3>

<pre><code class='language-R'>mrmr.cutoff(x, y, n)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mrmr.cutoff_+3A_x">x</code></td>
<td>
<p>x is a N by p data frame of gene expression values where rows represent genes and columns represent samples/subject/time point. Each cell entry represents the expression level of a gene in a sample/subject (row names of x as gene names/gene ids).</p>
</td></tr>
<tr><td><code id="mrmr.cutoff_+3A_y">y</code></td>
<td>
<p>y is a p by 1 numeric vector with entries 1 and -1 representing sample labels, where 1 and -1 represents the sample label of subjects/ samples for stress and control condition respectively.</p>
</td></tr>
<tr><td><code id="mrmr.cutoff_+3A_n">n</code></td>
<td>
<p>n is a numeric constant represents the number of informative genes to be selected.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An informative geneset is obtained, which is relevanit to the particular trait/condition and the genes within the selected geneset are minimum redundant using MRMR technique.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>References</h3>

<p>Ding, C and Peng, H (2005). Minimum redundancy feature selection from microarray gene expression data. J. Bioinformatics Comput Biol 3(2):185-205.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rice_salt)
x=as.data.frame(rice_salt[-1,])
y=as.numeric(rice_salt[1,])
n=20
mrmr.cutoff(x, y, n)
</code></pre>

<hr>
<h2 id='pval.bmrmr'>Compuation of statistical significance values for genes using Bootstrap-MRMR technique</h2><span id='topic+pval.bmrmr'></span>

<h3>Description</h3>

<p>The function computes ths statisical significance values for the genes from the non-parametric test &quot;H0: i-th gene is not informative against H1: i-th gene is informative&quot; for selection of informative genes using Bootstrap-MRMR technique</p>


<h3>Usage</h3>

<pre><code class='language-R'>pval.bmrmr(x, y, s, Q, plot)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pval.bmrmr_+3A_x">x</code></td>
<td>
<p>x is a N by p data frame of gene expression values where rows represent genes and columns represent samples/subject/time point. Each cell entry represents the expression level of a gene in a sample/subject (row names of x as gene names/gene ids).</p>
</td></tr>
<tr><td><code id="pval.bmrmr_+3A_y">y</code></td>
<td>
<p>y is a p by 1 numeric vector with entries 1 and -1 representing sample labels, where 1 and -1 represents the sample label of subjects/ samples for stress and control condition respectively.</p>
</td></tr>
<tr><td><code id="pval.bmrmr_+3A_s">s</code></td>
<td>
<p>s is a scalar representing the number of bootstraps generated, s must be sufficiently large (i.e. number of times  bootstrap samples are generated</p>
</td></tr>
<tr><td><code id="pval.bmrmr_+3A_q">Q</code></td>
<td>
<p>Q is a scalar representing the quartile value of the gene rankscores (lies within 1/N to 1), usually the second quartile (Q2), i.e. 0.5 or third quartile (Q3) i.e. 0.75 is taken.</p>
</td></tr>
<tr><td><code id="pval.bmrmr_+3A_plot">plot</code></td>
<td>
<p>plot is a character string must either take logical value TRUE/FALSE representing whether to plot the statistical significance values of genes in the dataset.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a vector of p-values for all the genes from the given statistical test in the dataset using Bootstrap-MRMR technique.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rice_salt)
x=as.data.frame(rice_salt[-1,])
y=as.numeric(rice_salt[1,])
s=80
Q=0.5
pval.bmrmr(x, y, s, Q, plot=FALSE)
</code></pre>

<hr>
<h2 id='pval.mbmr'>Computation of statistical significance values for genes using Modified Bootstrap MRMR technique for a particular trait/condition</h2><span id='topic+pval.mbmr'></span>

<h3>Description</h3>

<p>The statisical significance values (p-values) will be computed for all the genes in the dataset from the non-parametric test &quot;H0: i-th gene is not informative against H1: i-th gene is informative&quot; for selection of informative genes using Modified Bootstrap MRMR technique.</p>


<h3>Usage</h3>

<pre><code class='language-R'>pval.mbmr(x, y, m, s, Q, plot)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pval.mbmr_+3A_x">x</code></td>
<td>
<p>x is a N by p data frame of gene expression values where rows represent genes and columns represent samples/subject/time point. Each cell entry represents the expression level of a gene in a sample/subject (row names of x as gene names/gene ids).</p>
</td></tr>
<tr><td><code id="pval.mbmr_+3A_y">y</code></td>
<td>
<p>y is a p by 1 numeric vector with entries 1 and -1 representing sample labels, where 1 and -1 represents the sample label of subjects/ samples for stress and control condition respectively.</p>
</td></tr>
<tr><td><code id="pval.mbmr_+3A_m">m</code></td>
<td>
<p>m is a scalar representing the size of the Modified Bootstrap Sample (i.e. Out of p samples/subjects, m samples/subjects are randomly drawn with replacement, which constitutes one Modified Bootstrap Sample).</p>
</td></tr>
<tr><td><code id="pval.mbmr_+3A_s">s</code></td>
<td>
<p>s is a scalar representing the number of Modified Bootstrap samples (i.e. number of times each of the m samples/subjects will be resampled from p samples/subjects). </p>
</td></tr>
<tr><td><code id="pval.mbmr_+3A_q">Q</code></td>
<td>
<p>Q is a scalar representing the quartile value of the gene rankscores (lies within 1/N to 1), usually the second quartile, i.e. 0.5 or third quartile i.e. 0.75.</p>
</td></tr>
<tr><td><code id="pval.mbmr_+3A_plot">plot</code></td>
<td>
<p>plot is a character string must either take logical value TRUE/FALSE representing whether to plot the statistical significance values of genes in the dataset.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a vector of p-values for all the genes from the given statistical test in the gene space/dataset using Modified Bootstrap MRMR technique.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rice_salt)
x=as.data.frame(rice_salt[-1,])
y=as.numeric(rice_salt[1,])
m=36
s=80
Q=0.5
pval.mbmr(x, y, m, s, Q, plot=FALSE)
</code></pre>

<hr>
<h2 id='rice_salt'>
A gene expression dataset of rice under salinity stress
</h2><span id='topic+rice_salt'></span>

<h3>Description</h3>

<p>This data has gene expression values of 200 genes over 40 samples/subjects for a salinity vs. control study in rice. These 40 samples belong to either of salinity stress or control condition (two class problem). This gene expression data is balanced type as the first 20 samples are under salinity stress and the later 20 samples are under control condition.
The first row of the data contains the samples/subjects labels with entries are 1 and -1, where the label '1' and '-1' represent samples generated under salinity stress and control condition respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("rice_salt")</code></pre>


<h3>Format</h3>

<p>A data frame with 200 rows as genes with 40 columns as samples/subjects. Each column (sample) represent the gene expression values of genes. Each column as microarray samples with labels -1 or 1 represents control or salinity stress respectively. 
</p>


<h3>Details</h3>

<p>The data is created by taking 200 genes from the large number of genes from NCBI GEO database. The rows are the genes and columns are the samples/subjects. The first half of the samples/subjects are generated under salinity stress condition and other half under control condition.The first row of the data contains the samples/subjects albels with entries are 1 and -1, where th label '1' and '-1' represent samples generated under salinity stress and control condition respectively.</p>


<h3>Source</h3>

<p>Gene Expression Omnibus: NCBI gene expression and hybridization array data repository.ncbi.nlm.nih.gov/geo/.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rice_salt)
</code></pre>

<hr>
<h2 id='topsis.meth'>Selection of optimal gene selection method(s)/method(s) through multi-criteria decision analysis</h2><span id='topic+topsis.meth'></span>

<h3>Description</h3>

<p>The function enables to rank gene selection methods/method(s) under a multi-criteria decision making set up and further selection of optimum gene selection method using Technique for Order of Preference by Similarity to Ideal Solution (TOPSIS) approach.</p>


<h3>Usage</h3>

<pre><code class='language-R'>topsis.meth(x)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="topsis.meth_+3A_x">x</code></td>
<td>
<p>x is a M by C data frame representing the performance value of the methods under different criteria, where rows are the methods and columns are the criteria. The performance of the methods are adjudged based on mangitude of the criteria (i.e. higher the criteria value better is the method).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a data frame consisting rows as method names and different columns with distance score for positive, negative ideal solution, TOPSIS score and ranks for respective methods.</p>


<h3>Author(s)</h3>

<p>Samarendra Das </p>


<h3>References</h3>

<p>Ahn BS (2011) Compatible weighting method with rank order centroid: Maximum entropy ordered weighted averaging approach. Eur J Oper Res 212: 552-559.</p>


<h3>Examples</h3>

<pre><code class='language-R'>
x=matrix(runif(150), 10, 15)
rownames(x)=paste("Method",1:nrow(x), sep="")
colnames(x)=paste("C",1:ncol(x), sep="")
x=as.data.frame(x)
topsis.meth(x)
</code></pre>

<hr>
<h2 id='weight.mbmr'>Computation of weights for informative gene selection using Modified Bootstrap MRMR technique</h2><span id='topic+weight.mbmr'></span>

<h3>Description</h3>

<p>Weights associated with genes in a dataset computed from the Modified Bootstrap MRMR technique will provide a reliable measure for informative gene selection.</p>


<h3>Usage</h3>

<pre><code class='language-R'>weight.mbmr(x, y, m, s, plot)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="weight.mbmr_+3A_x">x</code></td>
<td>
<p>x is a N by p dataframe of gene expression, where rows are genes and columns are as samples/subjects (gene names are taken as row names).</p>
</td></tr>   
<tr><td><code id="weight.mbmr_+3A_y">y</code></td>
<td>
<p>y is a p by 1 numeric vector with entries 1 and -1 representing sample labels, where 1 and -1 represents the sample label of subjects/ samples for stress and control condition respectively.</p>
</td></tr>
<tr><td><code id="weight.mbmr_+3A_m">m</code></td>
<td>
<p>m is a scalar representing the size of the Modified Bootstrap Sample (i.e. Out of p samples/subjects, m samples/subjects are randomly drawn with replacement, which constitutes one Modified Bootstrap Sample).</p>
</td></tr>
<tr><td><code id="weight.mbmr_+3A_s">s</code></td>
<td>
<p>s is a scalar representing the number of Modified Bootstrap samples (i.e. number of times each of the m samples/subjects will be resampled from p samples/subjects). </p>
</td></tr>
<tr><td><code id="weight.mbmr_+3A_plot">plot</code></td>
<td>
<p>plot is a character string must either take logical value TRUE/FALSE representing whether to plot the weights of genes in the dataset.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function returns a vector of weights associated with each genes in the dataset using Modified Bootstrap MRMR technique.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>References</h3>

<p>Wang J, Chen L, Wang Y, Zhang J, Liang Y, Xu D (2013) A Computational systems biology study for understanding salt tolerance mechanism in Rice. PLoS one 8(6): e64929.</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(rice_salt)
x=as.data.frame(rice_salt[-1,])
y=as.numeric(rice_salt[1,])
m=36
s=80
weight.mbmr(x, y, m, s, plot=FALSE)
</code></pre>

<hr>
<h2 id='Weights.mrmr'>Computation of MRMR weights for gene selection</h2><span id='topic+Weights.mrmr'></span>

<h3>Description</h3>

<p>The MRMR weights associated with each gene in the dataset are computed by using the MRMR technique for informative gene selection.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Weights.mrmr(x, y)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Weights.mrmr_+3A_x">x</code></td>
<td>
<p>x is a N by p dataframe of gene expression, where, rows as genes and columns as samples (with row names as gene names/ids)</p>
</td></tr>
<tr><td><code id="Weights.mrmr_+3A_y">y</code></td>
<td>
<p>y is a p by 1 numeric vector with entries 1 and -1 representing sample labels, where 1 and -1 represents the sample label of subjects/ samples for stress and control condition respectively.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function returns a vector of MRMR weights for all genes in the dataset.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>References</h3>

<p>Ding, C and Peng, H (2005). Minimum redundancy feature selection from microarray gene expression data. J. Bioinformatics Comput Biol 3(2):185-205.</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(rice_salt)
x=as.data.frame(rice_salt[-1,])
y=as.numeric(rice_salt[1,])
Weights.mrmr(x, y)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
