<!DOCTYPE html><html><head><title>Help for package BoSSA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BoSSA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BoSSA-package'>
<p>A Bunch of Structure and Sequence Analysis</p></a></li>
<li><a href='#circular_tree'>
<p>Plot an inside-out circular tree</p></a></li>
<li><a href='#plot.pplace'>
<p>Plot a pplace or jplace object</p></a></li>
<li><a href='#pplace'>
<p>A placement object as obtained with the read_sqlite function</p></a></li>
<li><a href='#pplace_to_matrix'>
<p>Pplace to contingency matrix</p></a></li>
<li><a href='#pplace_to_table'>
<p>Merge the multiclass and the placement table of pplace object</p></a></li>
<li><a href='#pplace_to_taxonomy'>
<p>Convert a pplace object to a taxonomy table</p></a></li>
<li><a href='#print.pplace'>
<p>Compact display of pplace and jplace objects</p></a></li>
<li><a href='#print.protdb'>
<p>Compact display of protdb object</p></a></li>
<li><a href='#read_jplace'>
<p>Read a jplace file</p></a></li>
<li><a href='#read_protdb'>
<p>Read Protein Data Bank (PDB) file</p></a></li>
<li><a href='#read_sqlite'>
<p>Read a pplacer/guppy sqlite file</p></a></li>
<li><a href='#refpkg'>
<p>Summary data and plots for reference packages</p></a></li>
<li><a href='#sub_pplace'>
<p>Subsets a pplace object</p></a></li>
<li><a href='#write_jplace'>
<p>Write a jplace or pplace object to the disk</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Bunch of Structure and Sequence Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>3.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-10-16</td>
</tr>
<tr>
<td>Author:</td>
<td>Pierre Lefeuvre</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Pierre Lefeuvre &lt;pierre.lefeuvre@cirad.fr&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, RSQLite, jsonlite, phangorn, plotrix</td>
</tr>
<tr>
<td>Suggests:</td>
<td>prettydoc, knitr, rmarkdown, XML, rentrez, httr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Description:</td>
<td>Reads and plots phylogenetic placements.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-10-20 04:28:26 UTC; lefeuvre</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-10-20 07:20:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='BoSSA-package'>
A Bunch of Structure and Sequence Analysis
</h2><span id='topic+BoSSA-package'></span><span id='topic+BoSSA'></span>

<h3>Description</h3>

<p>Reads and plots phylogenetic placements.
</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
</p>

<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> BoSSA</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Title: </td><td style="text-align: left;"> A Bunch of Structure and Sequence Analysis</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 3.7</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2020-10-16</td>
</tr>
<tr>
 <td style="text-align: left;">
Author: </td><td style="text-align: left;"> Pierre Lefeuvre</td>
</tr>
<tr>
 <td style="text-align: left;">
Maintainer: </td><td style="text-align: left;"> Pierre Lefeuvre &lt;pierre.lefeuvre@cirad.fr&gt;</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> R (&gt;= 3.3.0)</td>
</tr>
<tr>
 <td style="text-align: left;">
Imports: </td><td style="text-align: left;"> ape, RSQLite, jsonlite, phangorn, plotrix</td>
</tr>
<tr>
 <td style="text-align: left;">
Suggests: </td><td style="text-align: left;"> prettydoc, knitr, rmarkdown, XML, rentrez, httr</td>
</tr>
<tr>
 <td style="text-align: left;">
VignetteBuilder: </td><td style="text-align: left;"> knitr</td>
</tr>
<tr>
 <td style="text-align: left;">
Description: </td><td style="text-align: left;"> Reads and plots phylogenetic placements.</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>


<p>Index of help topics:
</p>
<pre>
BoSSA-package           A Bunch of Structure and Sequence Analysis
circular_tree           Plot an inside-out circular tree
plot.pplace             Plot a pplace or jplace object
pplace                  A placement object as obtained with the
                        read_sqlite function
pplace_to_matrix        Pplace to contingency matrix
pplace_to_table         Merge the multiclass and the placement table of
                        pplace object
pplace_to_taxonomy      Convert a pplace object to a taxonomy table
print.pplace            Compact display of pplace and jplace objects
print.protdb            Compact display of protdb object
read_jplace             Read a jplace file
read_protdb             Read Protein Data Bank (PDB) file
read_sqlite             Read a pplacer/guppy sqlite file
refpkg                  Summary data and plots for reference packages
sub_pplace              Subsets a pplace object
write_jplace            Write a jplace or pplace object to the disk
</pre>
<p>BoSSA contains functions to read and plot phylogenetic placement files obtained using softwares such as pplacer, guppy, EPA and RAPPAS.
</p>


<h3>Author(s)</h3>

<p>Pierre Lefeuvre
Maintainer: Pierre Lefeuvre &lt;pierre.lefeuvre@cirad.fr&gt;
</p>


<h3>References</h3>

<p>- pplacer and guppy
http://matsen.fhcrc.org/pplacer/
http://matsen.github.io/pplacer/
- EPA
https://sco.h-its.org/exelixis/web/software/epa/index.html
- RAPPAS
https://github.com/benclaff/RAPPAS
- Common file format
http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0031009
</p>

<hr>
<h2 id='circular_tree'>
Plot an inside-out circular tree
</h2><span id='topic+circular_tree'></span>

<h3>Description</h3>

<p>Plot a tree in a circular manner with the tips pointing inward
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circular_tree(phy,ratio=0.5,def=1000,pos_out=FALSE,tip_labels=TRUE,cex_tips=0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="circular_tree_+3A_phy">phy</code></td>
<td>

<p>a class phylo object
</p>
</td></tr>
<tr><td><code id="circular_tree_+3A_ratio">ratio</code></td>
<td>

<p>the ratio of the tree size compared to the plot size
</p>
</td></tr>
<tr><td><code id="circular_tree_+3A_def">def</code></td>
<td>

<p>the def parameter controls the granularity of the curves
</p>
</td></tr>
<tr><td><code id="circular_tree_+3A_pos_out">pos_out</code></td>
<td>

<p>a matrix with the x and y coordinates of the branches extremities (i.e. nodes and tips) is outputed when set to TRUE
</p>
</td></tr>
<tr><td><code id="circular_tree_+3A_tip_labels">tip_labels</code></td>
<td>

<p>whether or not the tiplabels should be plotted
</p>
</td></tr>
<tr><td><code id="circular_tree_+3A_cex_tips">cex_tips</code></td>
<td>

<p>the size of the tiplabels
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function plot a tree in a circular manner. Note that the tree will produce a correct output only if there is no topology modifications after reading the original tree using the ape read.tree function.
</p>


<h3>Value</h3>

<p>a plot
</p>


<h3>Author(s)</h3>

<p>pierre lefeuvre
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(ape)

test_tree &lt;- rtree(20)

circular_tree(test_tree)

</code></pre>

<hr>
<h2 id='plot.pplace'>
Plot a pplace or jplace object
</h2><span id='topic+plot.pplace'></span><span id='topic+plot.jplace'></span>

<h3>Description</h3>

<p>Plot the tree and placements from a pplace or a jplace object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pplace'
plot(x,type="precise",simplify=FALSE,
		main="",N=NULL,transfo=NULL,legend=TRUE,stl=FALSE,
		asb=FALSE,edge.width=1,max_width=10,cex.number=0.5,
		cex.text=0.8,transp=80,add=FALSE,color=NULL,discrete_col=FALSE,
		pch=16,run_id=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.pplace_+3A_x">x</code></td>
<td>

<p>A pplace or jplace object
</p>
</td></tr>
<tr><td><code id="plot.pplace_+3A_type">type</code></td>
<td>

<p>The type of ploting desired with either, &quot;precise&quot;, &quot;color&quot;, &quot;fattree&quot; or &quot;number&quot;. For each option, placement sizes represent the multiplication of the N value with the placement ML ratio.
</p>
</td></tr>
<tr><td><code id="plot.pplace_+3A_simplify">simplify</code></td>
<td>

<p>If set to TRUE, only plot the best position for each placement. default is FALSE.
</p>
</td></tr>
<tr><td><code id="plot.pplace_+3A_main">main</code></td>
<td>

<p>An optionnal title to plot along the tree
</p>
</td></tr>
<tr><td><code id="plot.pplace_+3A_n">N</code></td>
<td>

<p>An optionnal vector of the weight of each placement. Must be of the same length and order as placements in the multiclass table. Note that the placement mass (potentially) available from the original files are imported into R but aren't use in the analysis. The N parameter should be used instead.
</p>
</td></tr>
<tr><td><code id="plot.pplace_+3A_transfo">transfo</code></td>
<td>

<p>An optionnal function to transform the placement size when type set to &quot;precise&quot;. Beware that it is also applied to the legend so that it does not anymore correspond to the placement size but to the transform dot size
</p>
</td></tr>
<tr><td><code id="plot.pplace_+3A_legend">legend</code></td>
<td>

<p>Plot a legend. Not available for type &quot;number&quot; or &quot;fattree&quot;
</p>
</td></tr>
<tr><td><code id="plot.pplace_+3A_stl">stl</code></td>
<td>

<p>Show tip labels
</p>
</td></tr>
<tr><td><code id="plot.pplace_+3A_asb">asb</code></td>
<td>

<p>Add scale bar
</p>
</td></tr>
<tr><td><code id="plot.pplace_+3A_edge.width">edge.width</code></td>
<td>

<p>The tree edge width
</p>
</td></tr>
<tr><td><code id="plot.pplace_+3A_max_width">max_width</code></td>
<td>

<p>The maximum edge width when type is set to &quot;fattree&quot;
</p>
</td></tr>
<tr><td><code id="plot.pplace_+3A_cex.number">cex.number</code></td>
<td>

<p>Control the size of the text when type is set to &quot;number&quot;
</p>
</td></tr>
<tr><td><code id="plot.pplace_+3A_cex.text">cex.text</code></td>
<td>

<p>Control the size of the main
</p>
</td></tr>
<tr><td><code id="plot.pplace_+3A_transp">transp</code></td>
<td>

<p>Control the transparency of the placement when type is &quot;precise&quot; and the transparency of the branch without placement when type is set to &quot;color&quot;. Encoded in hexadecimal scale (i.e. range from &quot;00&quot; to &quot;FF&quot;)
</p>
</td></tr>
<tr><td><code id="plot.pplace_+3A_add">add</code></td>
<td>

<p>Add placement to an existing plot when type is set to precise. Default is FALSE. If it was drawn, the legend won't be updated. Beware to use the same value for the &quot;transfo&quot; option in each plot. Dots color scale won't be accurate when using the &quot;add&quot; option. It is highly recommanded to use a single color.
</p>
</td></tr>
<tr><td><code id="plot.pplace_+3A_color">color</code></td>
<td>

<p>The colors used for pendant branch length scale when type is set to &quot;precise&quot;. Default is a color ramp with &quot;blue&quot;, &quot;green&quot;, &quot;yellow&quot; and &quot;red&quot;
</p>
</td></tr>
<tr><td><code id="plot.pplace_+3A_discrete_col">discrete_col</code></td>
<td>

<p>Discretise the color scale for pendant branch length
</p>
</td></tr>
<tr><td><code id="plot.pplace_+3A_pch">pch</code></td>
<td>

<p>The dot style used for placements when type is set to &quot;precise&quot;
</p>
</td></tr>
<tr><td><code id="plot.pplace_+3A_run_id">run_id</code></td>
<td>

<p>A vector of run_id to subset
</p>
</td></tr>
<tr><td><code id="plot.pplace_+3A_...">...</code></td>
<td>

<p>Further arguments passed to or from other methods.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>pierre lefeuvre
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(pplace)

### number type
plot(pplace,type="number",main="number")

### color type without and with legend
plot(pplace,type="color",main="color without legend",legend=FALSE)
plot(pplace,type="color",main="color with legend",legend=TRUE)

### fattree type
plot(pplace,type="fattree",main="fattree")

### precise type
plot(pplace,type="precise",main="precise vanilla")
plot(pplace,type="precise",simplify=TRUE,main="precise simplify")

# using the read number information encoded here in the name (if available)
Npplace &lt;- sample(1:100,nrow(pplace$multiclass),replace=TRUE)
# in the following exemple, the dots are too large...
plot(pplace,type="precise",main="precise N",legend=TRUE,N=Npplace,simplify=TRUE)

# using the transfo option to modify dot sizes
# note that placements sizes inferior to 1 won't
# behave properly with log10 as a transformation function.
# In this case, you rather use simplify (all the placement
# will corresponds to at least one sequence).
# Beware that when using the transfo option, 
# the legend does not anymore correspond to the actual placement
# size but to the transform placement size
# (i.e. the transform function applied to the dot size).
# we will use the the log10 function
plot(pplace,type="precise",main="precise log10",
	legend=TRUE,N=Npplace,transfo=log10)
# or without simplify, you can use a custom function
# as transfo that will produce positive sized dots
plot(pplace,type="precise",main="precise custom"
	,legend=TRUE,N=Npplace,transfo=function(X){log10(X+1)})
</code></pre>

<hr>
<h2 id='pplace'>
A placement object as obtained with the read_sqlite function
</h2><span id='topic+pplace'></span>

<h3>Description</h3>

<p>A placement object as obtained with the read_sqlite function. In this example, a set of 100 sequence reads are placed over a 16S phylogeny. This example is a subset of those available for download at http://fhcrc.github.io/microbiome-demo/
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("pplace")</code></pre>


<h3>References</h3>

<p>http://fhcrc.github.io/microbiome-demo/
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pplace)
str(pplace)
</code></pre>

<hr>
<h2 id='pplace_to_matrix'>
Pplace to contingency matrix
</h2><span id='topic+pplace_to_matrix'></span>

<h3>Description</h3>

<p>Convert the pplace object into a contingency matrix OTUs / sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pplace_to_matrix(pplace, sample_info, N = NULL, tax_name = FALSE
				 ,run_id=NULL,round_type=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pplace_to_matrix_+3A_pplace">pplace</code></td>
<td>

<p>A pplace object
</p>
</td></tr>
<tr><td><code id="pplace_to_matrix_+3A_sample_info">sample_info</code></td>
<td>

<p>A vector or list specifying the association between placement (in the multiclass table) and sample. In the case of a list, multiple sample can be associated with a single placement.
</p>
</td></tr>
<tr><td><code id="pplace_to_matrix_+3A_n">N</code></td>
<td>

<p>An optionnal vector or list with a number of occurence (or weight) associated to each placed sequence. If &quot;sample_info&quot; is a list, &quot;N&quot; must also be a list. Note that the placement mass (potentially) available from the original files are imported into R but aren't use in the analysis. The N parameter should be used instead.
</p>
</td></tr>
<tr><td><code id="pplace_to_matrix_+3A_tax_name">tax_name</code></td>
<td>

<p>Either the tax ids (when set to FALSE, default) or the tax names (when set to TRUE) are used as column names. The tax names are obtained form the &quot;taxo&quot; table of the pplace object.
</p>
</td></tr>
<tr><td><code id="pplace_to_matrix_+3A_run_id">run_id</code></td>
<td>

<p>A vector of run_id to subset
</p>
</td></tr>
<tr><td><code id="pplace_to_matrix_+3A_round_type">round_type</code></td>
<td>

<p>The name of the rounding fonction to apply to the product of the number of individuals classified in a given category and the likelihood ratio of this classification.
Should be set to NULL (no rounding) or one of &quot;trunc&quot;, &quot;round&quot;, &quot;ceiling&quot; or &quot;floor&quot;.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A contingency matrix with OTUs / species in rows and samples in columns.
</p>


<h3>Author(s)</h3>

<p>pierre lefeuvre
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(pplace)

### simple example
pplace_to_matrix(pplace,c(rep("sample1",27),rep("sample2",50),rep("sample3",23)))

### using the N option to specify the number of sequence each placement represents
Npplace &lt;- sample(1:20,100,replace=TRUE)
pplace_to_matrix(pplace,c(rep("sample1",27),rep("sample2",50),rep("sample3",23)),N=Npplace)

### with tax_name=TRUE
pplace_to_matrix(pplace,c(rep("sample1",27),rep("sample2",50),rep("sample3",23)),tax_name=TRUE)
</code></pre>

<hr>
<h2 id='pplace_to_table'>
Merge the multiclass and the placement table of pplace object
</h2><span id='topic+pplace_to_table'></span>

<h3>Description</h3>

<p>Merge the multiclass and the placement table of pplace object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pplace_to_table(pplace, type = "full",run_id=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pplace_to_table_+3A_pplace">pplace</code></td>
<td>

<p>a pplace object
</p>
</td></tr>
<tr><td><code id="pplace_to_table_+3A_type">type</code></td>
<td>

<p>the placement type to consider
</p>
</td></tr>
<tr><td><code id="pplace_to_table_+3A_run_id">run_id</code></td>
<td>

<p>A vector of run_id to subset
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For the type argument, either &quot;full&quot; or &quot;best&quot; are accepted. Whereas for the &quot;full&quot; type, all the placements are considered, only the best placement for each sequence is considered for the &quot;best&quot; type.
</p>


<h3>Value</h3>

<p>a data frame with the same column names as the mutliclass and placements tables
</p>


<h3>Author(s)</h3>

<p>pierre lefeuvre
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(pplace)

### with every placement
pplace_to_table(pplace)

### keeping only the best placement for each sequence
pplace_to_table(pplace,type="best")
</code></pre>

<hr>
<h2 id='pplace_to_taxonomy'>
Convert a pplace object to a taxonomy table
</h2><span id='topic+pplace_to_taxonomy'></span>

<h3>Description</h3>

<p>Convert a pplace object to a taxonomy table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pplace_to_taxonomy(pplace,taxonomy,
rank=c("phylum","class","order","family","genus","species"),
type="all",tax_name=TRUE,run_id=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pplace_to_taxonomy_+3A_pplace">pplace</code></td>
<td>

<p>A pplace object
</p>
</td></tr>
<tr><td><code id="pplace_to_taxonomy_+3A_taxonomy">taxonomy</code></td>
<td>

<p>The taxonomy table as obtained using the refseq fonction with type set to taxonomy
</p>
</td></tr>
<tr><td><code id="pplace_to_taxonomy_+3A_rank">rank</code></td>
<td>

<p>The desired rank for the taxonomy table
</p>
</td></tr>
<tr><td><code id="pplace_to_taxonomy_+3A_type">type</code></td>
<td>

<p>Wether all the possible classification available in the multiclass table are outputed (type=&quot;all&quot;) or only the best (type=&quot;best&quot;)
</p>
</td></tr>
<tr><td><code id="pplace_to_taxonomy_+3A_tax_name">tax_name</code></td>
<td>

<p>Wether to use taxonomy names (default) or tax_id number
</p>
</td></tr>
<tr><td><code id="pplace_to_taxonomy_+3A_run_id">run_id</code></td>
<td>

<p>A vector of run_id to subset
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with taxonomic ranks for each sequence
</p>


<h3>Author(s)</h3>

<p>pierre lefeuvre
</p>

<hr>
<h2 id='print.pplace'>
Compact display of pplace and jplace objects
</h2><span id='topic+print.pplace'></span><span id='topic+print.jplace'></span>

<h3>Description</h3>

<p>Compact display of pplace and jplace objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pplace'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.pplace_+3A_x">x</code></td>
<td>

<p>a pplace or jplace object
</p>
</td></tr>
<tr><td><code id="print.pplace_+3A_...">...</code></td>
<td>

<p>further arguments passed to or from other methods
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>pierre lefeuvre
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pplace)
print(pplace)
</code></pre>

<hr>
<h2 id='print.protdb'>
Compact display of protdb object
</h2><span id='topic+print.protdb'></span>

<h3>Description</h3>

<p>Function to print the header section of the protdb object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'protdb'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.protdb_+3A_x">x</code></td>
<td>

<p>a protdb class object
</p>
</td></tr>
<tr><td><code id="print.protdb_+3A_...">...</code></td>
<td>

<p>further arguments passed to or from other methods
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>pierre lefeuvre
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pdb_file &lt;- system.file("extdata", "1L2M.pdb", package = "BoSSA")
pdb &lt;- read_protdb(pdb_file)
print(pdb)
</code></pre>

<hr>
<h2 id='read_jplace'>
Read a jplace file
</h2><span id='topic+read_jplace'></span>

<h3>Description</h3>

<p>Read a jplace file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_jplace(jplace_file, full = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_jplace_+3A_jplace_file">jplace_file</code></td>
<td>

<p>A jplace file name
</p>
</td></tr>
<tr><td><code id="read_jplace_+3A_full">full</code></td>
<td>

<p>If set to FALSE, only the tree is read from the jplace file
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When the jplace or sqlite files are imported into R, the node numbering available in the original
file is converted to the class &quot;phylo&quot; numbering. The class phylo is defined in the &quot;ape&quot; package.
</p>


<h3>Value</h3>

<p>A list with
</p>
<table>
<tr><td><code>arbre</code></td>
<td>
<p>The tree in class &quot;phylo&quot; over wich placements are performed</p>
</td></tr>
<tr><td><code>placement</code></td>
<td>
<p>The placement table</p>
</td></tr>
<tr><td><code>multiclass</code></td>
<td>
<p>The multiclass table</p>
</td></tr>
<tr><td><code>run</code></td>
<td>
<p>The command line used to obtained the jplace file</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>pierre lefeuvre
</p>


<h3>See Also</h3>

<p>read_sqlite
</p>

<hr>
<h2 id='read_protdb'>
Read Protein Data Bank (PDB) file 
</h2><span id='topic+read_protdb'></span>

<h3>Description</h3>

<p>Read Protein Data Bank (PDB) file 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_protdb(X)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_protdb_+3A_x">X</code></td>
<td>

<p>The path/name of a pdb file.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a list of objects	
</p>
<table>
<tr><td><code>header</code></td>
<td>
<p>The header of the pdb file</p>
</td></tr>
<tr><td><code>compound</code></td>
<td>
<p>A data frame summarizing the CMPND part of the pdb file. This include the molecule ID, the molecule name and the chain ID</p>
</td></tr>
<tr><td><code>atom</code></td>
<td>
<p>A data frame with the atom type, the amino acid, the amino acid number, the chain and the euclidian X, Y, Z coordinates of the atoms</p>
</td></tr>
<tr><td><code>sequence</code></td>
<td>
<p>A list with the numbering of the amino acid and the amino acid sequence for each chain</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>pierre lefeuvre
</p>


<h3>References</h3>

<p>http://www.rcsb.org/pdb/home/home.do
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pdb_file &lt;- system.file("extdata", "1L2M.pdb", package = "BoSSA")
pdb &lt;- read_protdb(pdb_file)
pdb
</code></pre>

<hr>
<h2 id='read_sqlite'>
Read a pplacer/guppy sqlite file
</h2><span id='topic+read_sqlite'></span>

<h3>Description</h3>

<p>Read a pplacer/guppy sqlite file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_sqlite(sqlite_file,jplace_file=gsub("sqlite$","jplace",sqlite_file),
rank="species")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_sqlite_+3A_sqlite_file">sqlite_file</code></td>
<td>

<p>A pplacer/guppy sqlite path/file name
</p>
</td></tr>
<tr><td><code id="read_sqlite_+3A_jplace_file">jplace_file</code></td>
<td>

<p>An optionnal jplace file name. By default, the sqlite file name with the suffix changed from &quot;sqlite&quot; to jplace&quot; is used. If different, the jplace path/name must be specified.
</p>
</td></tr>
<tr><td><code id="read_sqlite_+3A_rank">rank</code></td>
<td>

<p>The desired taxonomic assignation rank to extract. default is &quot;species&quot;.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As the tree informations are not available in the sqlite file, the jplace file is also required.
When the jplace or sqlite files are import into R, the node numbering available in the original
file is converted to the class &quot;phylo&quot; numbering.
</p>


<h3>Value</h3>

<p>A list with
</p>
<table>
<tr><td><code>runs</code></td>
<td>
<p>The command line used to obtained the sqlite file</p>
</td></tr>
<tr><td><code>taxa</code></td>
<td>
<p>The taxonomic information table</p>
</td></tr>
<tr><td><code>multiclass</code></td>
<td>
<p>The multiclass table</p>
</td></tr>
<tr><td><code>placement_positions</code></td>
<td>
<p>A data frame with the posiotn of each placement in the reference tree</p>
</td></tr>
<tr><td><code>arbre</code></td>
<td>
<p>The tree in class &quot;phylo&quot; over wich placements are performed</p>
</td></tr>
<tr><td><code>edge_key</code></td>
<td>
<p>A matrix with correspondance of node numbering between the original tree in the jplace file and the class phylo tree of the &quot;arbre&quot; component</p>
</td></tr>
<tr><td><code>original_tree</code></td>
<td>
<p>The tree string from the jplace file</p>
</td></tr>
</table>
<p>For details on the other components (i.e. &quot;placements, &quot;placement_classifications&quot;, &quot;placement_evidence&quot;, &quot;placement_median_identities&quot;, &quot;placement_names&quot;, &quot;placement_nbc&quot;, &quot;placements&quot;, &quot;ranks&quot; and &quot;sqlite_sequence&quot;, please, refer to http://erick.matsen.org/pplacer/generated_rst/guppy_classify.html)
</p>


<h3>Author(s)</h3>

<p>pierre lefeuvre
</p>


<h3>References</h3>

<p>http://erick.matsen.org/pplacer/generated_rst/guppy_classify.html
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### the path to the sqlite and jplace files
sqlite_file &lt;- system.file("extdata", "example.sqlite", package = "BoSSA")
jplace_file &lt;- system.file("extdata", "example.jplace", package = "BoSSA")
pplace &lt;- read_sqlite(sqlite_file,jplace_file)
</code></pre>

<hr>
<h2 id='refpkg'>
Summary data and plots for reference packages
</h2><span id='topic+refpkg'></span>

<h3>Description</h3>

<p>Summary data and plots for reference packages
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refpkg(refpkg_path,type="summary",rank_tree="species",
rank_pie=c("phylum","class","order","family","genus"),
scale_pie=TRUE,alpha_order=TRUE,cex.text=0.7,
cex.legend=1,asb=TRUE,rotate_label=TRUE,
out_krona="for_krona.txt",text2krona=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refpkg_+3A_refpkg_path">refpkg_path</code></td>
<td>

<p>The path of the reference package directory
</p>
</td></tr>
<tr><td><code id="refpkg_+3A_type">type</code></td>
<td>

<p>The type of summary to perform with &quot;summary&quot;, &quot;taxonomy&quot;, &quot;info&quot;, &quot;tree&quot;, &quot;pie&quot; or &quot;krona&quot; available
</p>
</td></tr>
<tr><td><code id="refpkg_+3A_rank_tree">rank_tree</code></td>
<td>

<p>The desired rank for tree coloring
</p>
</td></tr>
<tr><td><code id="refpkg_+3A_rank_pie">rank_pie</code></td>
<td>

<p>The ranks to be plot for the taxonomy pie chart
</p>
</td></tr>
<tr><td><code id="refpkg_+3A_scale_pie">scale_pie</code></td>
<td>

<p>Wether or not to take into account the number of sequences available within the reference package for the pie chart
</p>
</td></tr>
<tr><td><code id="refpkg_+3A_alpha_order">alpha_order</code></td>
<td>

<p>Wether or not the color should follow taxa alpahabetic order when type set to &quot;tree&quot;
</p>
</td></tr>
<tr><td><code id="refpkg_+3A_cex.text">cex.text</code></td>
<td>

<p>The tip labels cex parameter when type is set to &quot;tree&quot; and the text cex parameter when type is set to &quot;pie&quot;
</p>
</td></tr>
<tr><td><code id="refpkg_+3A_cex.legend">cex.legend</code></td>
<td>

<p>The size of the legend when type set to &quot;tree&quot;
</p>
</td></tr>
<tr><td><code id="refpkg_+3A_asb">asb</code></td>
<td>

<p>Add a scale bar on the tree
</p>
</td></tr>
<tr><td><code id="refpkg_+3A_rotate_label">rotate_label</code></td>
<td>

<p>Rotates the pie slice labels
</p>
</td></tr>
<tr><td><code id="refpkg_+3A_out_krona">out_krona</code></td>
<td>

<p>The name of the output file when type is set to &quot;krona&quot;.
</p>
</td></tr>
<tr><td><code id="refpkg_+3A_text2krona">text2krona</code></td>
<td>

<p>The full path to the krona &quot;ImportText.pl&quot; script when KronaTools is installed and you wish to directly produce the html krona file.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A summary print on screen when type set to &quot;summary&quot;.
A data frame when type set to &quot;taxonomy&quot; or &quot;info&quot;.
A file written to the disk when type is set to &quot;krona&quot;.
A plot otherwise.
</p>


<h3>Author(s)</h3>

<p>pierre lefeuvre
</p>


<h3>References</h3>

<p>https://github.com/marbl/Krona/wiki/KronaTools
http://fhcrc.github.io/taxtastic/
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
refpkg_path &lt;- paste(find.package("BoSSA"),"/extdata/example.refpkg",sep="")

### summary
refpkg(refpkg_path)

### taxonomy
taxonomy &lt;- refpkg(refpkg_path,type="taxonomy")
head(taxonomy)

### info
refpkg(refpkg_path,type="info")

### tree
refpkg(refpkg_path,type="tree",rank_tree="order",cex.text=0.5)

### pie
refpkg(refpkg_path,type="pie",rank_pie=c("class","order","family"),cex.text=0.6)

### krona
# it will produce a flat text file
# this file can be use as input for the the "ImportText.pl" krona script
# see https://github.com/marbl/Krona/wiki/KronaTools for more details on krona
## Not run: 
refpkg(refpkg_path,type="krona",out_krona="for_krona.txt")

## End(Not run)
</code></pre>

<hr>
<h2 id='sub_pplace'>
Subsets a pplace object
</h2><span id='topic+sub_pplace'></span>

<h3>Description</h3>

<p>Subsets a pplace or jplace object based on the placement_id, the name of the placement or a regular expression of the name of the placement
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sub_pplace(x, placement_id = NULL, ech_id = NULL, ech_regexp = NULL, run_id = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sub_pplace_+3A_x">x</code></td>
<td>

<p>The pplace or jplace object to subset
</p>
</td></tr>
<tr><td><code id="sub_pplace_+3A_placement_id">placement_id</code></td>
<td>

<p>A vector of the placement_id to subset
</p>
</td></tr>
<tr><td><code id="sub_pplace_+3A_ech_id">ech_id</code></td>
<td>

<p>A vector of the names of the placement to subset
</p>
</td></tr>
<tr><td><code id="sub_pplace_+3A_ech_regexp">ech_regexp</code></td>
<td>

<p>A regular expression of the name of the placement to subset
</p>
</td></tr>
<tr><td><code id="sub_pplace_+3A_run_id">run_id</code></td>
<td>

<p>A vector of run_id to subset
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When using placement_id, the subset is performed based on the placement_id column of the multiclass, placements, placement_positions, placement_names, placement_classifications, placement_evidence, placement_median_identities and placement_nbc data frames.
When using ech_id and ech_regexp, the subset is performed from the multiclass$name column.
When using run_id, the subset is performed based on the placements$run_id column.
</p>


<h3>Value</h3>

<p>A pplace object
</p>


<h3>Author(s)</h3>

<p>pierre lefeuvre
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(pplace)

### subsetting using placement ids. Here placements 1 to 5
sub1 &lt;- sub_pplace(pplace,placement_id=1:5)
sub1

### subsetting using sequenes ids
id &lt;- c("GWZHISEQ01:514:HMCLFBCXX:2:1108:1739:60356_90",
"GWZHISEQ01:514:HMCLFBCXX:2:1114:13665:31277_80")
sub2 &lt;- sub_pplace(pplace,ech_id=id)
sub2

### subsetting using a regular expression of sequence ids
sub3 &lt;- sub_pplace(pplace,ech_regexp="^HWI")
sub3
</code></pre>

<hr>
<h2 id='write_jplace'>
Write a jplace or pplace object to the disk
</h2><span id='topic+write_jplace'></span>

<h3>Description</h3>

<p>Write a jplace or pplace object to the disk in the jplace JSON format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_jplace(x,outfile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write_jplace_+3A_x">x</code></td>
<td>

<p>A pplace or jplace object
</p>
</td></tr>
<tr><td><code id="write_jplace_+3A_outfile">outfile</code></td>
<td>

<p>The name of the output file
</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Note that the placement mass (potentially) available from the original files are imported into R but aren't use in the analysis. Anyway, the write_jplace function takes into account possible weight/mass information available in the the &quot;nm&quot; column of the multiclass table for jplace objects and in the &quot;mass&quot; column from the placement_names table for the pplace objects. The values in these column can be edited before writing the jplace file if one want to use distinct mass/weight in downtstream analysis (e.g. using the guppy program functionalities).
</p>


<h3>Author(s)</h3>

<p>pierre lefeuvre
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(pplace)
## Not run: 
write_jplace(pplace,"test.jplace")

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
