<!DOCTYPE html><html><head><title>Help for package bossR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {bossR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#getbeta'><p>Get regression coefficient</p></a></li>
<li><a href='#getOC'><p>Select Optimal cutoff for a biomarker</p></a></li>
<li><a href='#getpvalue'><p>Computes the distribution function of the multivariate normal distribution</p></a></li>
<li><a href='#myGene'><p>clinical dataset</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Biomarker Optimal Segmentation System</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.4</td>
</tr>
<tr>
<td>Description:</td>
<td>The Biomarker Optimal Segmentation System R package, 'bossR', is designed for precision medicine, helping to identify individual traits using biomarkers. It focuses on determining the most effective cutoff value for a continuous biomarker, which is crucial for categorizing patients into two groups with distinctly different clinical outcomes. The package simultaneously finds the optimal cutoff from given candidate values and tests its significance. Simulation studies demonstrate that 'bossR' offers statistical power and false positive control non-inferior to the permutation approach (considered the gold standard in this field), while being hundreds of times faster. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>mvtnorm,survival,stats</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10),</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-12 19:24:16 UTC; xzhang</td>
</tr>
<tr>
<td>Author:</td>
<td>Liuyi Lan [aut],
  Xing Li <a href="https://orcid.org/0000-0002-4186-7909"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Xuanjin Cheng [aut],
  Xuekui Zhang <a href="https://orcid.org/0000-0003-4728-2343"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xuekui Zhang &lt;ubcxzhang@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-15 16:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='getbeta'>Get regression coefficient</h2><span id='topic+getbeta'></span>

<h3>Description</h3>

<p>Computes the regression coefficient of certain regression based on certain cutoff.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getbeta(data, point, type = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getbeta_+3A_data">data</code></td>
<td>
<p>A data frame which contains 3 columns for cox regression : survival time, censor status, Biomarker values.
2 columns for linear regression :  Y, X.</p>
</td></tr>
<tr><td><code id="getbeta_+3A_point">point</code></td>
<td>
<p>A number to cut biomarker or X.</p>
</td></tr>
<tr><td><code id="getbeta_+3A_type">type</code></td>
<td>
<p>A number; if =1, will perform linear regression; if =2(default) will perform cox regression.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object with 3 class: Coefficient beta, number of samples of which the biomarker is greater than the point, standard error of coefficient estimation.
</p>

<hr>
<h2 id='getOC'>Select Optimal cutoff for a biomarker</h2><span id='topic+getOC'></span>

<h3>Description</h3>

<p>Given a set of data including survival time ,censor status and Biomarker values,
return the Optimal cutoff for the biomarker.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getOC(data, cutoff, type = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getOC_+3A_data">data</code></td>
<td>
<p>A data frame which contains 3 columns for cox regression : survival time, censor status, Biomarker values.
2 columns for linear regression :  Y, X.</p>
</td></tr>
<tr><td><code id="getOC_+3A_cutoff">cutoff</code></td>
<td>
<p>Numeric vector of candidate cutoffs.</p>
</td></tr>
<tr><td><code id="getOC_+3A_type">type</code></td>
<td>
<p>A number; if =1, will perform linear regression;if =2(default) will perform cox regerssion.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Optimal cutoff for the biomarker, the FWER of the model
</p>


<h3>References</h3>

<p>BOSS - Biomarker Optimal Segmentation System
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cutoff=c(56,112,167,223,278,334,389,445)
    data(myGene)
    getOC(data=myGene,cutoff)
</code></pre>

<hr>
<h2 id='getpvalue'>Computes the distribution function of the multivariate normal distribution</h2><span id='topic+getpvalue'></span>

<h3>Description</h3>

<p>Computes the distribution function of the multivariate normal distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getpvalue(threshold, mu, n, Sigma)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getpvalue_+3A_threshold">threshold</code></td>
<td>
<p>A number.</p>
</td></tr>
<tr><td><code id="getpvalue_+3A_mu">mu</code></td>
<td>
<p>The mean vector of length n.</p>
</td></tr>
<tr><td><code id="getpvalue_+3A_n">n</code></td>
<td>
<p>A number indicates dimension.</p>
</td></tr>
<tr><td><code id="getpvalue_+3A_sigma">Sigma</code></td>
<td>
<p>The correlation matrix of dimension n.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The evaluated distribution function
</p>

<hr>
<h2 id='myGene'>clinical dataset</h2><span id='topic+myGene'></span>

<h3>Description</h3>

<p>This data set gives the expression levels of gene data, overall survival time, and censoring
status from 500 lung adenocarcinoma cases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>myGene</code></pre>


<h3>Format</h3>

<p>A dataframe containing 500 observations of 3 variables.</p>


<h3>Source</h3>

<p> raw survival data come from https://tau.cmmt.ubc.ca/cSurvival/project_data/TCGA-LUAD/df_survival_o.csv and raw gene expression data come from  https://tau.cmmt.ubc.ca/cSurvival/project_data/TCGA-LUAD/df_gene.csv</p>


<h3>References</h3>

<p>Xuanjin Cheng, Yongxing Liu, Jiahe Wang, Yujie Chen, Andrew Gordon Robertson, Xuekui Zhang, Steven J M
Jones, and Stefan Taubert. (2022) <em>cSurvival: a web resource for biomarker interactions in cancer outcomes and in cell lines</em>. Briefings in Bioinformatics.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
