<!DOCTYPE html><html><head><title>Help for package BPM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BPM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#annotGeneNames'><p>gene names of probes in 450K array dat</p></a></li>
<li><a href='#ApiGetDMCs'><p>Get TOPK=500 DMCs and non-DMCs using moderated-t test</p></a></li>
<li><a href='#BayPM'><p>Bayesian Purity Model (BPM) Main functions.</p></a></li>
<li><a href='#BPM'><p>BPM software package</p></a></li>
<li><a href='#estimateNu'><p>Estimate noise intensity (nv) for non-DMCs,</p>
using maximum likelihood estmiation.</a></li>
<li><a href='#fullSampler'><p>Sampling xi and alpha (tumor purity)</p></a></li>
<li><a href='#goodProbes'><p>good probes in packages</p></a></li>
<li><a href='#simUCEC'><p>Simulated data to illustrate datasets in packages</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Bayesian Purity Model to Estimate Tumor Purity</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Jianzhao Gao, Linghao Shen, Xiaodan Fan</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jianzhao Gao and Xiaodan Fan &lt;gaojz@nankai.edu.cn&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Bayesian purity model to estimate tumor purity using methylation 
    array data (DNA methylation Infinium 450K array data) without reference samples. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, limma</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-10-31 07:23:58 UTC; Cliff</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-11-03 16:10:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='annotGeneNames'>gene names of probes in 450K array dat</h2><span id='topic+annotGeneNames'></span>

<h3>Description</h3>

<p>gene names of probes in 450K array dat
</p>


<h3>Format</h3>

<p>A vector with length 480457</p>

<hr>
<h2 id='ApiGetDMCs'>Get TOPK=500 DMCs and non-DMCs using moderated-t test</h2><span id='topic+ApiGetDMCs'></span>

<h3>Description</h3>

<p>Get TOPK=500 DMCs and non-DMCs using moderated-t test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ApiGetDMCs(betaValue, TOPK = 500, tumorNum = NULL,
  filterProbes = FALSE, userProbes = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ApiGetDMCs_+3A_betavalue">betaValue</code></td>
<td>
<p>A matrix from TCGA array data</p>
</td></tr>
<tr><td><code id="ApiGetDMCs_+3A_topk">TOPK</code></td>
<td>
<p>An integer number, default 500. Number of DMCs/non-DMCs.</p>
</td></tr>
<tr><td><code id="ApiGetDMCs_+3A_tumornum">tumorNum</code></td>
<td>
<p>A postive number, First tumorNum columns in betaValue
are tumor samples. If tumorNum is NULL, first half of
columns are considered as tumor samples,</p>
</td></tr>
<tr><td><code id="ApiGetDMCs_+3A_filterprobes">filterProbes</code></td>
<td>
<p>Logistic. defalut is FALSE. The code use all probes in betaValue.
If TRUE,
you can use default good probes provided in our code.
you can also provide your good probes in userProbes.</p>
</td></tr>
<tr><td><code id="ApiGetDMCs_+3A_userprobes">userProbes</code></td>
<td>
<p>A number list. The row numbers in betaValue.
These rows are considered as good probes.
return DMCs        (TOPK DMCs and TOPK non-DMCs row index in betaValue)</p>
</td></tr>
</table>


<h3>Note</h3>

<p>User can provide the good probes indexes (row number)
to filter the probes.
A global variable goodProbes are used in this function.
goodProbes: probes with SNPs at the CpG or single base
extension sites, and corss-reative probes are removed.
More details see the reference paper.
</p>

<hr>
<h2 id='BayPM'>Bayesian Purity Model (BPM) Main functions.</h2><span id='topic+BayPM'></span>

<h3>Description</h3>

<p>Bayesian Purity Model (BPM) Main functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BayPM(betaValue, TOPK = 500, tumorNum = NULL, filterProbes = FALSE,
  userProbes = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BayPM_+3A_betavalue">betaValue</code></td>
<td>
<p>A matrix,TCGA methlation array data. Each row: loci,
Tumor1,Tumor2,...,Normal1,Nomral2,...</p>
</td></tr>
<tr><td><code id="BayPM_+3A_topk">TOPK</code></td>
<td>
<p>A number. Number of DMCs/nonDMCs selected</p>
</td></tr>
<tr><td><code id="BayPM_+3A_tumornum">tumorNum</code></td>
<td>
<p>The number of tumor samples.
if NULL, the default number is half of column number of dataset.</p>
</td></tr>
<tr><td><code id="BayPM_+3A_filterprobes">filterProbes</code></td>
<td>
<p>Logistic. defalut is FALSE. The code use all probes in betaValue.
If TRUE,
you can use default good probes provided in our code.
you can also provide your good probes in userProbes.</p>
</td></tr>
<tr><td><code id="BayPM_+3A_userprobes">userProbes</code></td>
<td>
<p>A number list. The row numbers in betaValue.
These rows are considered as good probes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tumor purity estimation of tumor samples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### need to install package "limma"
### source("https://bioconductor.org/biocLite.R");biocLite("limma");
BayPM(simUCEC,20,2);

</code></pre>

<hr>
<h2 id='BPM'>BPM software package</h2><span id='topic+BPM'></span><span id='topic+BPM-package'></span>

<h3>Description</h3>

<p>Bayesian model for purity estimation using DNA
methylation data
</p>


<h3>Details</h3>

<p>The main function is <code><a href="#topic+BayPM">BayPM</a></code>
</p>


<h3>Author(s)</h3>

<p>Jianzhao Gao(gaojz@nankai.edu.cn), Linghao Shen Xiaodan Fan (xfan@cuhk.edu.hk)
</p>


<h3>References</h3>

<p>Jianzhao Gao, Linghao Shen, and Xiaodan Fan,
Bayesian model for purity estimation using DNA methylation data.(submitted)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### need to install package "limma"
### source("https://bioconductor.org/biocLite.R");biocLite("limma");
library(BPM);
BayPM(simUCEC,20,2);
</code></pre>

<hr>
<h2 id='estimateNu'>Estimate noise intensity (nv) for non-DMCs,
using maximum likelihood estmiation.</h2><span id='topic+estimateNu'></span>

<h3>Description</h3>

<p>Estimate noise intensity (nv) for non-DMCs,
using maximum likelihood estmiation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimateNu(z, phi, maxit = 50, beginP = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimateNu_+3A_z">z</code></td>
<td>
<p>A matrix. Observated mixed turmor samples.</p>
</td></tr>
<tr><td><code id="estimateNu_+3A_phi">phi</code></td>
<td>
<p>mode of beta-values of each row in pure nomral samples y.</p>
</td></tr>
<tr><td><code id="estimateNu_+3A_maxit">maxit</code></td>
<td>
<p>A postive integer. The iteration number used in maximum likelihood.</p>
</td></tr>
<tr><td><code id="estimateNu_+3A_beginp">beginP</code></td>
<td>
<p>A number, where the method start to search from for root.
</p>
<p>return  estimated nv (noise intensity)</p>
</td></tr>
</table>

<hr>
<h2 id='fullSampler'>Sampling xi and alpha (tumor purity)</h2><span id='topic+fullSampler'></span>

<h3>Description</h3>

<p>Sampling xi and alpha (tumor purity)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fullSampler(y, z, mstates, xprior = NULL, maxit = 1000,
  burnin = maxit, xpar = FALSE, n_ab0 = NULL, alp0 = NULL,
  xbar0 = NULL, trace = FALSE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fullSampler_+3A_y">y</code></td>
<td>
<p>A matrix, observed pure normal samples</p>
</td></tr>
<tr><td><code id="fullSampler_+3A_z">z</code></td>
<td>
<p>A matrix, observed mixed tumor samples</p>
</td></tr>
<tr><td><code id="fullSampler_+3A_mstates">mstates</code></td>
<td>
<p>A matrix, hyper/hypo of dataset</p>
</td></tr>
<tr><td><code id="fullSampler_+3A_xprior">xprior</code></td>
<td>
<p>A matrix, prior knowledge about purity</p>
</td></tr>
<tr><td><code id="fullSampler_+3A_maxit">maxit</code></td>
<td>
<p>A number, maximum iteraction</p>
</td></tr>
<tr><td><code id="fullSampler_+3A_burnin">burnin</code></td>
<td>
<p>A number, &quot;burn-in&quot; sample</p>
</td></tr>
<tr><td><code id="fullSampler_+3A_xpar">xpar</code></td>
<td>
<p>Logistic, default is FALSE</p>
</td></tr>
<tr><td><code id="fullSampler_+3A_n_ab0">n_ab0</code></td>
<td>
<p>initial value of n_ab</p>
</td></tr>
<tr><td><code id="fullSampler_+3A_alp0">alp0</code></td>
<td>
<p>initial value of alpha</p>
</td></tr>
<tr><td><code id="fullSampler_+3A_xbar0">xbar0</code></td>
<td>
<p>initial value of xbar</p>
</td></tr>
<tr><td><code id="fullSampler_+3A_trace">trace</code></td>
<td>
<p>Logisitc, check the values in code, default is FALSE</p>
</td></tr>
<tr><td><code id="fullSampler_+3A_verbose">verbose</code></td>
<td>
<p>Logistic, output the message,default is FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>x_bar    x_mode,
x_last   x2
x_sample x_sample
xpar     xprior2,
nab      n_ab2,
alp      alp2
</p>

<hr>
<h2 id='goodProbes'>good probes in packages</h2><span id='topic+goodProbes'></span>

<h3>Description</h3>

<p>good probes removed Y chrome.
</p>


<h3>Format</h3>

<p>A vector with length 425698</p>

<hr>
<h2 id='simUCEC'>Simulated data to illustrate datasets in packages</h2><span id='topic+simUCEC'></span>

<h3>Description</h3>

<p>A dataset containing 100 gene and 4 smaples,
first two columns are tumor1 tumor2
last  two columns are normal1 normal2
</p>

<ul>
<li><p> x. the genes
</p>
</li>
<li><p> y. two tumor samples; two normal samples;
</p>
</li></ul>



<h3>Format</h3>

<p>A matrix with 100 rows and 4 columns</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
