<!DOCTYPE html><html lang="en"><head><title>Help for package BRACoD.R</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BRACoD.R}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#convergence_tests'><p>Perform convergence tests on the p and beta variables</p></a></li>
<li><a href='#install_bracod'><p>Install BRACoD in python</p></a></li>
<li><a href='#obesity'><p>Example microbiome data</p></a></li>
<li><a href='#remove_null'><p>Remove NULL values in your OTU and environmental variable</p></a></li>
<li><a href='#run_bracod'><p>Run the main BRACoD algorithm</p></a></li>
<li><a href='#scale_counts'><p>Normalize OTU counts and add a pseudo count</p></a></li>
<li><a href='#score'><p>Score the results of BRACoD</p></a></li>
<li><a href='#simulate_microbiome_counts'><p>Simulate microbiome counts</p></a></li>
<li><a href='#summarize_trace'><p>Summarize the results of BRACoD</p></a></li>
<li><a href='#threshold_count_data'><p>Threshold your microbiome counts data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>BRACoD: Bayesian Regression Analysis of Compositional Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.2.0</td>
</tr>
<tr>
<td>Description:</td>
<td>The goal of this method is to identify associations between bacteria and an environmental variable in 16S or other compositional data. The environmental variable is any variable which is measure for each microbiome sample, for example, a butyrate measurement paired with every sample in the data. Microbiome data is compositional, meaning that the total abundance of each sample sums to 1, and this introduces severe statistical distortions. This method takes a Bayesian approach to correcting for these statistical distortions, in which the total abundance is treated as an unknown variable. This package runs the python implementation using reticulate.</td>
</tr>
<tr>
<td>Imports:</td>
<td>reticulate</td>
</tr>
<tr>
<td>Config/reticulate:</td>
<td>list( packages = list( list(package = "BRACoD") ) )</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1.9001</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-03-17 14:28:55 UTC; averster</td>
</tr>
<tr>
<td>Author:</td>
<td>Adrian Verster [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Adrian Verster &lt;adrian.verster@hc-sc.gc.ca&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-03-24 15:10:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='convergence_tests'>Perform convergence tests on the p and beta variables</h2><span id='topic+convergence_tests'></span>

<h3>Description</h3>

<p>You may get errors are divergence of some variables after pymc3 samples the posterior.
We are not overly concerned about some of the variables, such as the variance, rather
we are really interested in the inclusion probabilities (p) and contribution coefficients
(beta). The convergence tests that are included here focus on evaluating those two variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convergence_tests(trace, df_relab)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convergence_tests_+3A_trace">trace</code></td>
<td>
<p>the output of run_bracod()</p>
</td></tr>
<tr><td><code id="convergence_tests_+3A_df_relab">df_relab</code></td>
<td>
<p>the microbiome relative abundance</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return value
</p>

<hr>
<h2 id='install_bracod'>Install BRACoD in python</h2><span id='topic+install_bracod'></span>

<h3>Description</h3>

<p>Uses pip to install the latest BRACoD release in python. You might need
to specify a python environment with either reticulate::use_virtualenv or
reticulate::use_condaenv.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>install_bracod(method = "auto", conda = "auto")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="install_bracod_+3A_method">method</code></td>
<td>
<p>passed to reticulate::py_install</p>
</td></tr>
<tr><td><code id="install_bracod_+3A_conda">conda</code></td>
<td>
<p>passed to reticulate::py_install</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return value
</p>

<hr>
<h2 id='obesity'>Example microbiome data</h2><span id='topic+obesity'></span><span id='topic+df_counts_obesity'></span><span id='topic+df_counts_obesity+2Cdf_scfa'></span><span id='topic+df_scfa'></span>

<h3>Description</h3>

<p>This data is mouse stool microbiome data from a study of obesity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(obesity)

df_scfa
</code></pre>


<h3>Format</h3>

<p>a DataFrame of 16S microbiome counts, and a dataframe with corresponding butyrate measurements
</p>
<p>An object of class <code>data.frame</code> with 119 rows and 1 columns.
</p>

<hr>
<h2 id='remove_null'>Remove NULL values in your OTU and environmental variable</h2><span id='topic+remove_null'></span>

<h3>Description</h3>

<p>This will remove samples that are NULL in the environmental variable, as well as
the corresponding samples in your relative abundance data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_null(df_relab, Y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="remove_null_+3A_df_relab">df_relab</code></td>
<td>
<p>microbiome relative abundance data in a dataframe</p>
</td></tr>
<tr><td><code id="remove_null_+3A_y">Y</code></td>
<td>
<p>values of the environmental variable</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing 1) the relative abundance data and 2) the Y values
</p>

<hr>
<h2 id='run_bracod'>Run the main BRACoD algorithm</h2><span id='topic+run_bracod'></span>

<h3>Description</h3>

<p>Uses pymc3 to sample the posterior of the model to determine bacteria that are
associated with your environmental variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_bracod(df_relab, env_var, n_sample = 1000, n_burn = 1000, njobs = 4)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_bracod_+3A_df_relab">df_relab</code></td>
<td>
<p>A dataframe of relative microbiome abundances. Samples are rows and bacteria are columns.</p>
</td></tr>
<tr><td><code id="run_bracod_+3A_env_var">env_var</code></td>
<td>
<p>the environmental variable you are evaluating. You need 1 measurement associated with each sample.</p>
</td></tr>
<tr><td><code id="run_bracod_+3A_n_sample">n_sample</code></td>
<td>
<p>number of posterior samples.</p>
</td></tr>
<tr><td><code id="run_bracod_+3A_n_burn">n_burn</code></td>
<td>
<p>number of burn-in steps before actual sampling stops.</p>
</td></tr>
<tr><td><code id="run_bracod_+3A_njobs">njobs</code></td>
<td>
<p>number of parallel MCMC chains to run.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the pymc trace object which holds the samples of the posterior distribution
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(obesity)
r &lt;- simulate_microbiome_counts(obesity)
sim_counts &lt;- r[[1]]
sim_y &lt;- r[[2]]
contributions &lt;- r[[3]]
sim_relab &lt;- scale_counts(sim_counts)
trace &lt;- run_bracod(sim_relab, sim_y, n_sample = 1000, n_burn=1000, njobs=4)

## End(Not run)
</code></pre>

<hr>
<h2 id='scale_counts'>Normalize OTU counts and add a pseudo count</h2><span id='topic+scale_counts'></span>

<h3>Description</h3>

<p>BRACoD requires relative abundance and cannot handle zeros, so this function
adds a small pseudo count (1/10th the smallest non-zero value).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scale_counts(df_counts)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scale_counts_+3A_df_counts">df_counts</code></td>
<td>
<p>A dataframe of OTU counts. Samples are rows and bacteria are columns.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of relative abundance data
</p>

<hr>
<h2 id='score'>Score the results of BRACoD</h2><span id='topic+score'></span>

<h3>Description</h3>

<p>This calculate the precision, recall and F1 of your BRACoD results if you know
the ground truth, ie. if this is simulated data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>score(taxon_identified, taxon_actual)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="score_+3A_taxon_identified">taxon_identified</code></td>
<td>
<p>a list of integers corresponding to the indicies of the taxon you identified with BRACoD</p>
</td></tr>
<tr><td><code id="score_+3A_taxon_actual">taxon_actual</code></td>
<td>
<p>a list of integers corresponding to the indicies of the taxon that truely contribute to butyrate levels</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing 1) the precision 2) the recall 3) the f1 metric
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
df_summary &lt;- summarize_trace(trace, colnames(sim_relab))
taxon_identified &lt;- df_summary$taxon
taxon_actual &lt;- which(contributions != 0)

r &lt;- score(taxon_identified, taxon_actual)

precision &lt;- r[[1]]
recall &lt;- r[[2]]
f1 &lt;- r[[3]]

print(sprintf("Precision: %.2f, Recall: %.2f, F1: %.2f",precision, recall, f1))

## End(Not run)
</code></pre>

<hr>
<h2 id='simulate_microbiome_counts'>Simulate microbiome counts</h2><span id='topic+simulate_microbiome_counts'></span>

<h3>Description</h3>

<p>Each bacteria's absolute abundance is simulated from a lognormal distribution.
Then, convert each sample to relative abundance, and simulate sequencing counts
using a multinomial distribution, based on the desired number of reads and the
simulated relative abundances. This also simulates an environmental variable that
is produced by some of the bacteria.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulate_microbiome_counts(
  df,
  n_contributors = 20,
  coeff_contributor = 0,
  min_ab_contributor = -9,
  sd_Y = 1,
  n_reads = 1e+05,
  var_contributor = 5,
  use_uniform = TRUE,
  n_samples_use = NULL,
  corr_value = NULL,
  return_absolute = FALSE,
  seed = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simulate_microbiome_counts_+3A_df">df</code></td>
<td>
<p>A dataframe of OTU counts that is a model for data simulation. Samples are rows and bacteria are columns.</p>
</td></tr>
<tr><td><code id="simulate_microbiome_counts_+3A_n_contributors">n_contributors</code></td>
<td>
<p>the number of bacteria that are to contribute to your environmental variable.</p>
</td></tr>
<tr><td><code id="simulate_microbiome_counts_+3A_coeff_contributor">coeff_contributor</code></td>
<td>
<p>the average of the distribution used to simulate the contribution coefficient.</p>
</td></tr>
<tr><td><code id="simulate_microbiome_counts_+3A_min_ab_contributor">min_ab_contributor</code></td>
<td>
<p>The minimum log relative abundance, averaged across samples, to include a bacteria</p>
</td></tr>
<tr><td><code id="simulate_microbiome_counts_+3A_sd_y">sd_Y</code></td>
<td>
<p>the standard deviation of the simulated environmental variable</p>
</td></tr>
<tr><td><code id="simulate_microbiome_counts_+3A_n_reads">n_reads</code></td>
<td>
<p>the number of reads to be simulated per sample</p>
</td></tr>
<tr><td><code id="simulate_microbiome_counts_+3A_var_contributor">var_contributor</code></td>
<td>
<p>If you use a uniform distribution, this is the range of the distribution, with a normal distribution it is the variance used to simulate the contribution coefficient.</p>
</td></tr>
<tr><td><code id="simulate_microbiome_counts_+3A_use_uniform">use_uniform</code></td>
<td>
<p>use a uniform distribution to simulate the contribution coefficient. Alternative is the normal distribution.</p>
</td></tr>
<tr><td><code id="simulate_microbiome_counts_+3A_n_samples_use">n_samples_use</code></td>
<td>
<p>number of microbiome samples to simulate. If NULL, uses the same number of samples as in your dataframe</p>
</td></tr>
<tr><td><code id="simulate_microbiome_counts_+3A_corr_value">corr_value</code></td>
<td>
<p>the bacteria-bacteria correlation value you want to include in the simulation</p>
</td></tr>
<tr><td><code id="simulate_microbiome_counts_+3A_return_absolute">return_absolute</code></td>
<td>
<p>returns the abosulte abundance values instead of the simulated microbiome counts</p>
</td></tr>
<tr><td><code id="simulate_microbiome_counts_+3A_seed">seed</code></td>
<td>
<p>random seed for reproducibility</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing 1) the simulated count data 2) the simulated environmental variable and 3) the simulated contribution coefficients
</p>

<hr>
<h2 id='summarize_trace'>Summarize the results of BRACoD</h2><span id='topic+summarize_trace'></span>

<h3>Description</h3>

<p>This summarizes the trace object that run_bracod() returns. It returns a dataframe
that contains two parameters of interest, the average inclusion (p) and the average
coefficient (beta), telling you the association between that bacteria and the environmental
variable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summarize_trace(trace, taxon_names = NULL, cutoff = 0.3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summarize_trace_+3A_trace">trace</code></td>
<td>
<p>the pymc3 object that is the output of run_bracod()</p>
</td></tr>
<tr><td><code id="summarize_trace_+3A_taxon_names">taxon_names</code></td>
<td>
<p>optional, a list of names of the bacteria to include in the results</p>
</td></tr>
<tr><td><code id="summarize_trace_+3A_cutoff">cutoff</code></td>
<td>
<p>this is the cutoff on the average inclusion for inclusion. We reccomend a value of 0.3, but you can lower the value to include less confident taxon or raise the cutoff to exclude them.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe with information about the bacteria that BRACoD identified
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
trace &lt;- run_bracod(sim_relab, sim_y, n_sample = 1000, n_burn=1000, njobs=4)
df_summary &lt;- summarize_trace(trace, colnames(sim_relab))

## End(Not run)
</code></pre>

<hr>
<h2 id='threshold_count_data'>Threshold your microbiome counts data</h2><span id='topic+threshold_count_data'></span>

<h3>Description</h3>

<p>This function removes samples below a minimum counts and bacteria below a minimum log abundance.
Run this before running BRACoD because the algorithm does not perform well when there are many
low abundance bacteria that are only present in a few samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>threshold_count_data(df_counts, min_counts = 1000, min_ab = 1e-04)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="threshold_count_data_+3A_df_counts">df_counts</code></td>
<td>
<p>A dataframe of OTU counts. Samples are rows and bacteria are columns.</p>
</td></tr>
<tr><td><code id="threshold_count_data_+3A_min_counts">min_counts</code></td>
<td>
<p>threshold samples with fewer than this many counts</p>
</td></tr>
<tr><td><code id="threshold_count_data_+3A_min_ab">min_ab</code></td>
<td>
<p>threshold bacteria whose average log abundance is below this</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of microbiome counts
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
