<!DOCTYPE html><html><head><title>Help for package breakaway</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {breakaway}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#%&gt;%'><p>Pipe operator</p></a></li>
<li><a href='#alpha_estimate'><p>alpha_estimate</p></a></li>
<li><a href='#alpha_estimates'><p>alpha_estimates</p></a></li>
<li><a href='#apples'><p>(Data) Frequency count table of soil microbes in an apples orchard.</p></a></li>
<li><a href='#betta'><p>Modelling total diversity with betta</p></a></li>
<li><a href='#betta_lincom'><p>Confidence intervals and testing for linear combinations of fixed effects estimated via betta() or betta_random()</p></a></li>
<li><a href='#betta_pic'><p>function for plotting total diversity</p></a></li>
<li><a href='#betta_random'><p>modelling total diversity with random effects</p></a></li>
<li><a href='#breakaway'><p>Species richness estimation with breakaway</p></a></li>
<li><a href='#breakaway_nof1'><p>species richness estimation without singletons</p></a></li>
<li><a href='#build_frequency_count_tables'><p>Build frequency count tables from an OTU table</p></a></li>
<li><a href='#chao_bunge'><p>Chao-Bunge species richness estimator</p></a></li>
<li><a href='#chao_shen'><p>The Chao-Shen estimate of Shannon diversity</p></a></li>
<li><a href='#chao1'><p>Chao1 species richness estimator</p></a></li>
<li><a href='#chao1_bc'><p>Bias-corrected Chao1 species richness estimator</p></a></li>
<li><a href='#check_format'><p>Run some basic checks on a possible frequency count table</p></a></li>
<li><a href='#convert'><p>convert between different inputs for alpha-diversity estimates</p></a></li>
<li><a href='#cutoff_wrap'><p>Find a cut-off for estimates relying on contiguous counts</p></a></li>
<li><a href='#F_test'><p>Conduct F test of null hypothesis LB = 0 using output from betta()</p>
or betta_random()</a></li>
<li><a href='#get_F_stat'><p>Calculate F statistic under null hypothesis LB = 0 using output from betta()</p>
or betta_random()</a></li>
<li><a href='#good_turing'><p>The Good-Turing estimate of species richness</p></a></li>
<li><a href='#hawaii'><p>(Data) Frequency count table of soil microbes in Hawaii.</p></a></li>
<li><a href='#kemp'><p>Species richness estimation with Kemp-type models</p></a></li>
<li><a href='#make_design_matrix'><p>Make design matrix</p></a></li>
<li><a href='#make_frequency_count_table'><p>Draw frequency count subtables from an OTU table</p></a></li>
<li><a href='#objective_bayes_geometric'><p>Estimate species richness with an objective Bayes method using a geometric model</p></a></li>
<li><a href='#objective_bayes_mixedgeo'><p>Objective Bayes species richness estimate with the mixed-geometric model</p></a></li>
<li><a href='#objective_bayes_negbin'><p>Objective Bayes species richness estimate with the Negative Binomial model</p></a></li>
<li><a href='#objective_bayes_poisson'><p>Objective Bayes species richness estimate with the Poisson model</p></a></li>
<li><a href='#pasolli_et_al'><p>(Data) Data frame of covariate information about pasolli_et_al.</p></a></li>
<li><a href='#physeq_wrap'><p>Wrapper for phyloseq objects</p></a></li>
<li><a href='#plot.alpha_estimates'><p>Plot function for alpha_estimates class</p></a></li>
<li><a href='#poisson_model'><p>PoissonModel</p></a></li>
<li><a href='#poisson_model_nof1'><p>PoissonModelNof1</p></a></li>
<li><a href='#proportions_instead'><p>OTU table to relative abundances</p></a></li>
<li><a href='#rnbinomtable'><p>Negative binomially distributed frequency count tables.</p></a></li>
<li><a href='#rztnbinomtable'><p>beta version: Zero-truncated negative binomially distributed frequency count</p>
tables.</a></li>
<li><a href='#sample_inverse_simpson'><p>Plug-in Inverse Simpson diversity</p></a></li>
<li><a href='#sample_richness'><p>Sample richness estimator</p></a></li>
<li><a href='#sample_shannon'><p>Plug-in Shannon diversity</p></a></li>
<li><a href='#sample_shannon_e'><p>Plug-in Shannon's E (&quot;Equitability&quot;)</p></a></li>
<li><a href='#sample_simpson'><p>Plug-in Simpson diversity</p></a></li>
<li><a href='#sample_size_estimate'><p>Estimate the sample size needed to do an unpaired one-way test using betta</p></a></li>
<li><a href='#sample_size_figure'><p>Plot the power obtained with sample size</p></a></li>
<li><a href='#simulate_betta'><p>Simulate from a fitted betta model</p></a></li>
<li><a href='#simulate_betta_random'><p>Simulate from a fitted betta_random model</p></a></li>
<li><a href='#soil_phylo'><p>(Data) Data frame of soil data from whitman_et_al.</p></a></li>
<li><a href='#test_submodel'><p>Conduct F test of null hypothesis LB = 0 using output from betta()</p>
or betta_random()</a></li>
<li><a href='#toy_metadata'><p>(Data) Data frame of covariate information about toy_otu_table.</p></a></li>
<li><a href='#toy_otu_table'><p>(Data) A toy OTU table.</p></a></li>
<li><a href='#toy_taxonomy'><p>(Data) The taxonomy of the OTUs in 'toy_otu_table'.</p></a></li>
<li><a href='#true_gini'><p>Calculate the true Gini-Simpson index</p></a></li>
<li><a href='#true_hill'><p>Calculate the true Hill numbers</p></a></li>
<li><a href='#true_inverse_simpson'><p>Calculate the true Inverse Simpson index</p></a></li>
<li><a href='#true_shannon'><p>Calculate the true Shannon index based on proportions</p></a></li>
<li><a href='#true_shannon_e'><p>Calculate the true Shannon's equitability index</p></a></li>
<li><a href='#true_simpson'><p>Calculate the true Simpson index</p></a></li>
<li><a href='#wlrm_transformed'><p>The transformed weighted linear regression estimator for species richness estimation</p></a></li>
<li><a href='#wlrm_untransformed'><p>The untransformed weighted linear regression estimator for species richness estimation</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Species Richness Estimation and Modeling</td>
</tr>
<tr>
<td>Version:</td>
<td>4.8.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-11-17</td>
</tr>
<tr>
<td>Description:</td>
<td>Understanding the drivers of microbial diversity is an important frontier of microbial ecology, and investigating the diversity of samples from microbial ecosystems is a common step in any microbiome analysis. 'breakaway' is the premier package for statistical analysis of microbial diversity. 'breakaway' implements the latest and greatest estimates of species richness, described in Willis and Bunge (2015) &lt;<a href="https://doi.org/10.1111%2Fbiom.12332">doi:10.1111/biom.12332</a>&gt;, Willis et al. (2017) &lt;<a href="https://doi.org/10.1111%2Frssc.12206">doi:10.1111/rssc.12206</a>&gt;, and Willis (2016) &lt;<a href="https://arxiv.org/abs/1604.02598">arXiv:1604.02598</a>&gt;, as well as the most commonly used estimates, including the objective Bayes approach described in Barger and Bunge (2010) &lt;<a href="https://doi.org/10.1214%2F10-BA527">doi:10.1214/10-BA527</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/adw96/breakaway/issues">https://github.com/adw96/breakaway/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, graphics, lme4, magrittr, MASS, phyloseq, stats,
tibble, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>devtools, knitr, plyr, RCurl, rmarkdown, testthat, dplyr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://adw96.github.io/breakaway/">https://adw96.github.io/breakaway/</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-11-21 18:52:52 UTC; sarahteichman</td>
</tr>
<tr>
<td>Author:</td>
<td>Amy D Willis [aut, cre],
  Bryan D Martin [aut],
  Pauline Trinh [aut],
  Sarah Teichman [aut],
  David Clausen [aut],
  Kathryn Barger [aut],
  John Bunge [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Amy D Willis &lt;adwillis@uw.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-11-22 09:10:27 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling 'rhs(lhs)'.
</p>

<hr>
<h2 id='alpha_estimate'>alpha_estimate</h2><span id='topic+alpha_estimate'></span>

<h3>Description</h3>

<p>Build objects of class alpha_estimate from their components. <code>alpha_estimate()</code> is a constructor method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alpha_estimate(
  estimate = NULL,
  error = NULL,
  estimand = NULL,
  name = NULL,
  interval = NULL,
  interval_type = NULL,
  type = NULL,
  model = NULL,
  warnings = NULL,
  frequentist = NULL,
  parametric = NULL,
  plot = NULL,
  reasonable = NULL,
  other = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alpha_estimate_+3A_estimate">estimate</code></td>
<td>
<p>The estimate</p>
</td></tr>
<tr><td><code id="alpha_estimate_+3A_error">error</code></td>
<td>
<p>The standard error in the estimate</p>
</td></tr>
<tr><td><code id="alpha_estimate_+3A_estimand">estimand</code></td>
<td>
<p>What is the estimate trying to estimate? (richness, Shannon...)</p>
</td></tr>
<tr><td><code id="alpha_estimate_+3A_name">name</code></td>
<td>
<p>The name of the method</p>
</td></tr>
<tr><td><code id="alpha_estimate_+3A_interval">interval</code></td>
<td>
<p>An interval estimate</p>
</td></tr>
<tr><td><code id="alpha_estimate_+3A_interval_type">interval_type</code></td>
<td>
<p>Type of interval estimate</p>
</td></tr>
<tr><td><code id="alpha_estimate_+3A_type">type</code></td>
<td>
<p>TODO(Amy): Deprecate?</p>
</td></tr>
<tr><td><code id="alpha_estimate_+3A_model">model</code></td>
<td>
<p>What model is fit</p>
</td></tr>
<tr><td><code id="alpha_estimate_+3A_warnings">warnings</code></td>
<td>
<p>Any warnings?</p>
</td></tr>
<tr><td><code id="alpha_estimate_+3A_frequentist">frequentist</code></td>
<td>
<p>Logical. Frequentist or Bayesian?</p>
</td></tr>
<tr><td><code id="alpha_estimate_+3A_parametric">parametric</code></td>
<td>
<p>Logical. Parametric or not?</p>
</td></tr>
<tr><td><code id="alpha_estimate_+3A_plot">plot</code></td>
<td>
<p>A ggplot associated with the estimate</p>
</td></tr>
<tr><td><code id="alpha_estimate_+3A_reasonable">reasonable</code></td>
<td>
<p>Is the estimate likely to be reasonable?</p>
</td></tr>
<tr><td><code id="alpha_estimate_+3A_other">other</code></td>
<td>
<p>Any other relevant objects</p>
</td></tr>
<tr><td><code id="alpha_estimate_+3A_...">...</code></td>
<td>
<p>Any other objects</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class alpha_estimate
</p>

<hr>
<h2 id='alpha_estimates'>alpha_estimates</h2><span id='topic+alpha_estimates'></span>

<h3>Description</h3>

<p>Build objects of class alpha_estimates from their components. <code>alpha_estimates()</code> is a constructor method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alpha_estimates(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alpha_estimates_+3A_...">...</code></td>
<td>
<p>Objects of class alpha_estimate, or a list of objects of class alpha_estimate</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class alpha_estimates
</p>

<hr>
<h2 id='apples'>(Data) Frequency count table of soil microbes in an apples orchard.</h2><span id='topic+apples'></span>

<h3>Description</h3>

<p>(Data) Frequency count table of soil microbes in an apples orchard.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>apples
</code></pre>


<h3>Format</h3>

<p>A data frame with 88 rows and 2 variables:
</p>

<dl>
<dt>index</dt><dd><p>an index variable</p>
</dd>
<dt>frequency</dt><dd><p>number of taxa that were observed with this frequency</p>
</dd>
</dl>
<p>...

</p>


<h3>References</h3>

<p>Willis, A. and Bunge, J. (2015). Estimating diversity via
frequency ratios. <em>Biometrics</em>, <b>71</b>(4), 1042&ndash;1049.
<a href="https://doi.org/10.1111/biom.12332">doi:10.1111/biom.12332</a>
</p>
<p>Walsh, F. et al. (2014). (2014). Restricted streptomycin use in apple 
orchards did not adversely alter the soil bacteria communities. 
<em>Frontiers in Microbiology</em> <b>4</b>, 383.
</p>

<hr>
<h2 id='betta'>Modelling total diversity with betta</h2><span id='topic+betta'></span>

<h3>Description</h3>

<p>This function tests for heterogeneity of total diversity (observed plus
unobserved) across multiple sites. It can account or test for fixed effects
that may explain diversity. It returns the significance of the covariates in
explaining diversity and a hypothesis test for heterogeneity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>betta(
  chats = NULL,
  ses,
  X = NULL,
  initial_est = NULL,
  formula = NULL,
  data = NULL,
  p.digits = 3
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="betta_+3A_chats">chats</code></td>
<td>
<p>A vector of estimates of total diversity at different sampling
locations. &lsquo;<span class="samp">&#8288;breakaway&#8288;</span>&rsquo; estimates are suggested in the high-diversity
case but not enforced.</p>
</td></tr>
<tr><td><code id="betta_+3A_ses">ses</code></td>
<td>
<p>The standard errors in <code>chats</code>, the diversity estimates. This
can either be a vector of standard errors (with the arguments <code>chats</code> and
<code>X</code>), or the name of the variable in the dataframe <code>data</code> that contains
the standard errors (with the arguments <code>formula</code> and <code>data</code>).</p>
</td></tr>
<tr><td><code id="betta_+3A_x">X</code></td>
<td>
<p>A numeric matrix of covariates. If not supplied, an intercept-only
model will be fit. This is optional with the <code>chats</code> argument.</p>
</td></tr>
<tr><td><code id="betta_+3A_initial_est">initial_est</code></td>
<td>
<p>(Optional) A vector of length 1 + ncol(X) giving the starting values for the likelihood maximisation search. The first element is the starting estimate for sigma^2_u, and the remaining elements are the starting elements for beta. Defaults to NULL, in which case the starting values outlined in the paper are used.</p>
</td></tr>
<tr><td><code id="betta_+3A_formula">formula</code></td>
<td>
<p>A formula object of the form <code class="reqn">y ~ x | group</code>. Required with
the <code>data</code> argument.</p>
</td></tr>
<tr><td><code id="betta_+3A_data">data</code></td>
<td>
<p>A dataframe containing the response, response standard errors, covariates,
and grouping variable. Required with the <code>formula</code> argument.</p>
</td></tr>
<tr><td><code id="betta_+3A_p.digits">p.digits</code></td>
<td>
<p>(Optional) A number that specifies the number of digits to which p-values will be rounded. The default value is 3 digits.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>table</code></td>
<td>
<p> A coefficient table for the model parameters. The
columns give the parameter estimates, standard errors, and p-values,
respectively. This model is only as effective as your diversity estimation
procedure; for this reason please confirm that your estimates are
appropriate and that your model is not misspecified. betta_pic may be
useful for this purpose.  </p>
</td></tr> <tr><td><code>cov</code></td>
<td>
<p> Estimated covariance matrix of the
parameter estimates.  </p>
</td></tr> <tr><td><code>ssq_u</code></td>
<td>
<p> The estimate of the heterogeneity
variance.  </p>
</td></tr> <tr><td><code>homogeneity</code></td>
<td>
<p> The test statistic and p-value for the test
of homogeneity.  </p>
</td></tr> <tr><td><code>global</code></td>
<td>
<p> The test statistic and p-value for the
test of model explanatory power.  </p>
</td></tr> <tr><td><code>blups</code></td>
<td>
<p> The conditional expected
values of the diversity estimates (conditional on the random effects). The
authors propose that if the practitioner believes that information from one
diversity estimator may inform the others, then using the &lsquo;<span class="samp">&#8288;condfits&#8288;</span>&rsquo; as
estimators of total diversity rather than &lsquo;<span class="samp">&#8288;Chats&#8288;</span>&rsquo; may reduce variance
of diversity estimates by &ldquo;sharing strength&rdquo; across the samples.  </p>
</td></tr>
<tr><td><code>blupses</code></td>
<td>
<p> The estimated standard deviation (standard errors) in the
blups.  </p>
</td></tr>
<tr><td><code>loglikelihood</code></td>
<td>
<p> The log likelihood of the fitted model.  </p>
</td></tr>
<tr><td><code>aic</code></td>
<td>
<p> The Akaike information criterion for the fitted model. </p>
</td></tr>
<tr><td><code>aicc</code></td>
<td>
<p> The finite sample correction of the Akaike information criterion for the fitted model.  </p>
</td></tr>
<tr><td><code>r_squared_wls</code></td>
<td>
<p>  The weighted R^2 statistic, appropriate for heteroskedastic linear models. </p>
</td></tr>
<tr><td><code>function.args</code></td>
<td>
<p>A list containing
values initially passed to betta_random.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Ecologists who are interested in the way species richness varies with
covariate information often run a regression-type analysis on the observed
diversity using their covariate information as predictors. However, in many
settings (especially microbial), rare and unobserved taxa play a hugely
important role in explaining the subtleties of the ecosystem, however, a
regression analysis on the observed diversity level fails to account for
these unobserved taxa. By predicting the total level of diversity (for
example, via <code><a href="#topic+breakaway">breakaway</a></code>) and estimating the standard error in
the estimate, one can take account of these unobserved, but important, taxa.
In order to account for the estimated nature of the response, a mixed model
approach is taken, whereby the varying levels of confidence in the estimates
contributes to a diagonal but heteroscedastic covariance matrix. Given
covariates constitute the fixed effects in the mixed model, and significance
of the random effect term &ldquo;sigsq_u&rdquo; reflects heterogeneity in the sample,
that is, variability that cannot be explained by only the covariates. The
authors believe this to be the first attempt at modelling total diversity in
a way that accounts for its estimated nature.
</p>


<h3>Author(s)</h3>

<p>Amy Willis
</p>


<h3>References</h3>

<p>Willis, A., Bunge, J., and Whitman, T. (2015). Inference for
changes in biodiversity. <em>arXiv preprint.</em>
</p>
<p>Willis, A. and Bunge, J. (2015). Estimating diversity via frequency ratios.
<em>Biometrics.</em>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+breakaway">breakaway</a></code>; <code><a href="#topic+breakaway_nof1">breakaway_nof1</a></code>;
<code><a href="#topic+apples">apples</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
















df &lt;- data.frame(chats = c(2000, 3000, 4000, 3000), ses = c(100, 200, 150, 180),
                 Cont_var = c(100, 150, 100, 50))

# formula notation
betta(formula = chats ~ Cont_var, ses = ses, data = df)

# direct input
betta(c(2000, 3000, 4000, 3000), c(100, 200, 150, 180), cbind(1, c(100, 150, 100,
    50)))

## handles missing data
betta(c(2000, 3000, 4000, 3000), c(100, 200, 150, NA))

## A test for heterogeneity of apples diversity estimates vs butterfly estimates
betta(c(1552, 1500, 884), c(305, 675, 205), cbind(1, c(0, 0, 1)))






















</code></pre>

<hr>
<h2 id='betta_lincom'>Confidence intervals and testing for linear combinations of fixed effects estimated via betta() or betta_random()</h2><span id='topic+betta_lincom'></span>

<h3>Description</h3>

<p>This function provides point estimates, standard errors, and equal-tailed confidence intervals for
linear combinations of fixed effects estimated via betta() or betta_random(). A p-value for a Wald
test of the null that the linear combination of effects is equal to zero (against a general alternative)
is also returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>betta_lincom(fitted_betta, linear_com, signif_cutoff = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="betta_lincom_+3A_fitted_betta">fitted_betta</code></td>
<td>
<p>A fitted betta object &ndash; i.e., the output of either betta() or
betta_random() &ndash; containing fixed effect estimates of interest.</p>
</td></tr>
<tr><td><code id="betta_lincom_+3A_linear_com">linear_com</code></td>
<td>
<p>The linear combination of fixed effects for which a point estimate,
confidence interval, and hypothesis test are to be produced.</p>
</td></tr>
<tr><td><code id="betta_lincom_+3A_signif_cutoff">signif_cutoff</code></td>
<td>
<p>The type-I significance threshold for confidence intervals. Defaults
to 0.05.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>table</code></td>
<td>
<p> A table containing a point estimate, standard error, lower and upper confidence bounds,
and a p-value for the linear combination of fixed effects specified in input. The p-value is generated via a two-sided
Wald test of the null that the linear combination of fixed effects is equal to zero.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>David Clausen
</p>


<h3>References</h3>

<p>Willis, A., Bunge, J., and Whitman, T. (2015). Inference for
changes in biodiversity. <em>arXiv preprint.</em>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+betta">betta</a></code>;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# generate example data
df &lt;- data.frame(chats = c(2000, 3000, 4000, 3000), ses = c(100, 200, 150, 180),
                 Cont_var = c(100, 150, 100, 50))

# fit betta()
example_fit &lt;- betta(formula = chats ~ Cont_var, ses = ses, data = df)

# generate point estimate and 95% CI for mean richness at Cont_var = 125

betta_lincom(fitted_betta = example_fit,
linear_com = c(1, 125)) # this tells betta_lincom to estimate value of beta_0 + 125*beta_1,
# where beta_0 is the intercept, and beta_1 is the (true value of the) coefficient on Cont_var


</code></pre>

<hr>
<h2 id='betta_pic'>function for plotting total diversity</h2><span id='topic+betta_pic'></span>

<h3>Description</h3>

<p>A simple plotting interface for comparing total diversity across samples or
a covariate gradient.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>betta_pic(
  y,
  se,
  x = 1:length(y),
  ylimu = NULL,
  myy = NULL,
  mymain = NULL,
  mycol = NULL,
  labs = NULL,
  mypch = NULL,
  myxlim = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="betta_pic_+3A_y">y</code></td>
<td>
<p>A vector of estimates of total diversity. Other parameter estimates
are accessible; this method may be used for plotting any parameter
estimates..</p>
</td></tr>
<tr><td><code id="betta_pic_+3A_se">se</code></td>
<td>
<p>The standard errors in &lsquo;<span class="samp">&#8288;y&#8288;</span>&rsquo;, the diversity (or other
parameter's) estimates.</p>
</td></tr>
<tr><td><code id="betta_pic_+3A_x">x</code></td>
<td>
<p>A vector of covariates to form the x-coordinates of the intervals.
If no argument is given, defaults to the order.</p>
</td></tr>
<tr><td><code id="betta_pic_+3A_ylimu">ylimu</code></td>
<td>
<p>The upper endpoint of the y-axis.</p>
</td></tr>
<tr><td><code id="betta_pic_+3A_myy">myy</code></td>
<td>
<p>Deprecated, for backwards compatibility</p>
</td></tr>
<tr><td><code id="betta_pic_+3A_mymain">mymain</code></td>
<td>
<p>Deprecated, for backwards compatibility</p>
</td></tr>
<tr><td><code id="betta_pic_+3A_mycol">mycol</code></td>
<td>
<p>Deprecated, for backwards compatibility</p>
</td></tr>
<tr><td><code id="betta_pic_+3A_labs">labs</code></td>
<td>
<p>Deprecated, for backwards compatibility</p>
</td></tr>
<tr><td><code id="betta_pic_+3A_mypch">mypch</code></td>
<td>
<p>Deprecated, for backwards compatibility</p>
</td></tr>
<tr><td><code id="betta_pic_+3A_myxlim">myxlim</code></td>
<td>
<p>Deprecated, for backwards compatibility</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object.
</p>


<h3>Author(s)</h3>

<p>Amy Willis
</p>


<h3>References</h3>

<p>Willis, A., Bunge, J., and Whitman, T. (2015). Inference for
changes in biodiversity. arXiv preprint.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+betta">betta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>betta_pic(c(1552, 1500, 884), c(305, 675, 205), mymain = "Example title")

</code></pre>

<hr>
<h2 id='betta_random'>modelling total diversity with random effects</h2><span id='topic+betta_random'></span>

<h3>Description</h3>

<p>This function extends betta() to permit random effects modelling.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>betta_random(
  chats = NULL,
  ses,
  X = NULL,
  groups = NULL,
  formula = NULL,
  data = NULL,
  p.digits = 3
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="betta_random_+3A_chats">chats</code></td>
<td>
<p>A vector of estimates of total diversity at different sampling
locations. Required with the <code>groups</code> argument, and optionally with the
<code>X</code> argument.</p>
</td></tr>
<tr><td><code id="betta_random_+3A_ses">ses</code></td>
<td>
<p>The standard errors in <code>chats</code>, the diversity estimates. This
can either be a vector of standard errors (with the arguments <code>chats</code> and
<code>X</code>), or the name of the variable in the dataframe <code>data</code> that contains
the standard errors (with the arguments <code>formula</code> and <code>data</code>).</p>
</td></tr>
<tr><td><code id="betta_random_+3A_x">X</code></td>
<td>
<p>A numeric matrix of covariates corresponding to fixed effects. If
not supplied, an intercept-only model will be fit. Optional with the <code>chats</code>
and <code>groups</code> arguments.</p>
</td></tr>
<tr><td><code id="betta_random_+3A_groups">groups</code></td>
<td>
<p>A categorical variable representing the random-effects groups
that each of the estimates belong to. Required with the <code>chats</code> argument and
optionally with the <code>X</code> argument.</p>
</td></tr>
<tr><td><code id="betta_random_+3A_formula">formula</code></td>
<td>
<p>A formula object of the form <code class="reqn">y ~ x | group</code>. Required with
the <code>data</code> argument.</p>
</td></tr>
<tr><td><code id="betta_random_+3A_data">data</code></td>
<td>
<p>A dataframe containing the response, response standard errors, covariates,
and grouping variable. Required with the <code>formula</code> argument.</p>
</td></tr>
<tr><td><code id="betta_random_+3A_p.digits">p.digits</code></td>
<td>
<p>(Optional) A number that specifies the number of digits to which p-values will be rounded. The default value is 3 digits.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>table</code></td>
<td>
<p> A coefficient table for the model parameters. The
columns give the parameter estimates, standard errors, and p-values,
respectively. This model is only as effective as your diversity estimation
procedure; for this reason please confirm that your estimates are
appropriate and that your model is not misspecified. betta_pic may be
useful for this purpose.  </p>
</td></tr> <tr><td><code>cov</code></td>
<td>
<p> Estimated covariance matrix of the
parameter estimates.  </p>
</td></tr> <tr><td><code>ssq_u</code></td>
<td>
<p> The estimate of the heterogeneity
variance.  </p>
</td></tr> <tr><td><code>ssq_g</code></td>
<td>
<p> Estimates of within-group variance. The estimate
will be zero for groups with only one observation.  </p>
</td></tr> <tr><td><code>homogeneity</code></td>
<td>

<p>The test statistic and p-value for the test of homogeneity.  </p>
</td></tr>
<tr><td><code>global</code></td>
<td>
<p> The test statistic and p-value for the test of model
explanatory power.  </p>
</td></tr> <tr><td><code>blups</code></td>
<td>
<p> The conditional expected values of the
diversity estimates (conditional on the random effects). Estimates of
variability for the random effects case are unavailable at this time; please
contact the maintainer if needed.</p>
</td></tr> <tr><td><code>function.args</code></td>
<td>
<p>A list containing
values initially passed to betta_random.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Amy Willis
</p>


<h3>References</h3>

<p>Willis, A., Bunge, J., and Whitman, T. (2015). Inference for
changes in biodiversity. <em>arXiv preprint.</em>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+betta">betta</a></code>;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
df &lt;- data.frame(chats = c(2000, 3000, 4000, 3000),
ses = c(100, 200, 150, 180),
Cont_var = c(100, 150, 100, 50),
groups = c("a", "a", "a", "b"))

# formula notation
betta_random(formula = chats ~ Cont_var| groups,
ses = ses,
 data = df)

# direct input
betta_random(c(2000, 3000, 4000, 3000), c(100, 200, 150, 180),
X = cbind(Int = 1, Cont_var = c(100, 150, 100, 50)),
groups = c("a", "a", "a", "b"))

## handles missing data
betta_random(c(2000, 3000, 4000, 3000), c(100, 200, 150, NA),
groups = c("a", NA,
    "b", "b"))

</code></pre>

<hr>
<h2 id='breakaway'>Species richness estimation with breakaway</h2><span id='topic+breakaway'></span>

<h3>Description</h3>

<p>breakaway is a wrapper for modern species richness estimation
for modern datasets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>breakaway(
  input_data,
  cutoff = NA,
  output = NULL,
  plot = NULL,
  answers = NULL,
  print = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="breakaway_+3A_input_data">input_data</code></td>
<td>
<p>An input type that can be processed by <code>convert()</code></p>
</td></tr>
<tr><td><code id="breakaway_+3A_cutoff">cutoff</code></td>
<td>
<p>The maximum frequency count to use for model fitting</p>
</td></tr>
<tr><td><code id="breakaway_+3A_output">output</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
<tr><td><code id="breakaway_+3A_plot">plot</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
<tr><td><code id="breakaway_+3A_answers">answers</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
<tr><td><code id="breakaway_+3A_print">print</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
<tr><td><code id="breakaway_+3A_...">...</code></td>
<td>
<p>Additional arguments will be ignored; this is for backward compatibility</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>alpha_estimate</code>
</p>


<h3>Note</h3>

<p>&lsquo;<span class="samp">&#8288;breakaway&#8288;</span>&rsquo; presents an estimator of species richness that is
well-suited to the high-diversity/microbial setting. However, many microbial
datasets display more diversity than the Kemp-type models can permit. In
this case, the log-transformed WLRM diversity estimator of Rocchetti et. al.
(2011) is returned. The authors' experience suggests that some datasets that
require the log-transformed WLRM contain &ldquo;false&rdquo; diversity, that is,
diversity attributable to sequencing errors (via an inflated singleton
count). The authors encourage judicious use of diversity estimators when the
dataset may contain these errors, and recommend the use of
<code><a href="#topic+breakaway_nof1">breakaway_nof1</a></code> as an exploratory tool in this case.
</p>


<h3>Author(s)</h3>

<p>Amy Willis
</p>


<h3>References</h3>

<p>Willis, A. and Bunge, J. (2015). Estimating diversity via
frequency ratios. <em>Biometrics</em>, <b>71</b>(4), 1042&ndash;1049.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+breakaway_nof1">breakaway_nof1</a></code>; <code><a href="#topic+kemp">kemp</a></code>; <code><a href="#topic+apples">apples</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
breakaway(apples)
breakaway(apples, plot = FALSE, output = FALSE, answers = TRUE)

</code></pre>

<hr>
<h2 id='breakaway_nof1'>species richness estimation without singletons</h2><span id='topic+breakaway_nof1'></span>

<h3>Description</h3>

<p>This function permits estimation of total diversity based on a sample
frequency count table. Unlike <code><a href="#topic+breakaway">breakaway</a></code>, it does not require
an input for the number of species observed once, making it an excellent
exploratory tool for microbial ecologists who believe that their sample may
contain spurious singletons. The underlying estimation procedure is similar
to that of <code><a href="#topic+breakaway">breakaway</a></code> and is outlined in Willis &amp; Bunge (2014).
The diversity estimate, standard error, estimated model coefficients and
plot of the fitted model are returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>breakaway_nof1(
  input_data,
  output = NULL,
  plot = NULL,
  answers = NULL,
  print = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="breakaway_nof1_+3A_input_data">input_data</code></td>
<td>
<p>An input type that can be processed by <code>convert()</code></p>
</td></tr>
<tr><td><code id="breakaway_nof1_+3A_output">output</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
<tr><td><code id="breakaway_nof1_+3A_plot">plot</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
<tr><td><code id="breakaway_nof1_+3A_answers">answers</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
<tr><td><code id="breakaway_nof1_+3A_print">print</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>alpha_estimate</code>  </p>
<table>
<tr><td><code>code</code></td>
<td>
<p> A category representing algorithm behaviour.
&lsquo;<span class="samp">&#8288;code=1&#8288;</span>&rsquo; indicates no nonlinear models converged and the transformed
WLRM diversity estimate of Rocchetti et. al. (2011) is returned.
&lsquo;<span class="samp">&#8288;code=2&#8288;</span>&rsquo; indicates that the iteratively reweighted model converged and
was returned. &lsquo;<span class="samp">&#8288;code=3&#8288;</span>&rsquo; indicates that iterative reweighting did not
converge but a model based on a simplified variance structure was returned
(in this case, the variance of the frequency ratios is assumed to be
proportional to the denominator frequency index). Please peruse your fitted
model before using your diversity estimate.  </p>
</td></tr> <tr><td><code>name</code></td>
<td>
<p> The &ldquo;name&rdquo; of
the selected model. The first integer represents the numerator polynomial
degree and the second integer represents the denominator polynomial degree.
See Willis &amp; Bunge (2014) for details.  </p>
</td></tr> <tr><td><code>para</code></td>
<td>
<p> Estimated model
parameters and standard errors.  </p>
</td></tr> <tr><td><code>est</code></td>
<td>
<p> The estimate of total
(observed plus unobserved) diversity.  </p>
</td></tr> <tr><td><code>seest</code></td>
<td>
<p> The standard error in
the diversity estimate.  </p>
</td></tr> <tr><td><code>full</code></td>
<td>
<p> The chosen nonlinear model for
frequency ratios.  </p>
</td></tr>
</table>


<h3>Note</h3>

<p>It is common for microbial ecologists to believe that their dataset
contains false diversity. This often arises because sequencing errors result
in classifying already observed organisms as new organisms.
&lsquo;<span class="samp">&#8288;breakaway_nof1&#8288;</span>&rsquo; was developed as an exploratory tool in this case.
Practitioners can run &lsquo;<span class="samp">&#8288;breakaway&#8288;</span>&rsquo; on their dataset including the
singletons, and &lsquo;<span class="samp">&#8288;breakaway_nof1&#8288;</span>&rsquo; on their dataset excluding the
singletons, and assess if the estimated levels of diversity are very
different. Great disparity may provide evidence of an inflated singleton
count, or at the very least, that &lsquo;<span class="samp">&#8288;breakaway&#8288;</span>&rsquo; is especially sensitive
to the number of rare species observed. Note that &lsquo;<span class="samp">&#8288;breakaway_nof1&#8288;</span>&rsquo; may
be less stable than &lsquo;<span class="samp">&#8288;breakaway&#8288;</span>&rsquo; due to predicting based on a reduced
dataset, and have greater standard errors.
</p>


<h3>Author(s)</h3>

<p>Amy Willis
</p>


<h3>References</h3>

<p>Willis, A. (2015). Species richness estimation with high
diversity but spurious singletons. <em>arXiv.</em>
</p>
<p>Willis, A. and Bunge, J. (2015). Estimating diversity via frequency ratios.
<em>Biometrics.</em>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+breakaway">breakaway</a></code>; <code><a href="#topic+apples">apples</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
breakaway_nof1(apples[-1, ])
breakaway_nof1(apples[-1, ], plot = FALSE, output = FALSE, answers = TRUE)

</code></pre>

<hr>
<h2 id='build_frequency_count_tables'>Build frequency count tables from an OTU table</h2><span id='topic+build_frequency_count_tables'></span>

<h3>Description</h3>

<p>Build frequency count tables from an OTU table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_frequency_count_tables(the_table)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="build_frequency_count_tables_+3A_the_table">the_table</code></td>
<td>
<p>An OTU table as a data frame or a matrix. Columns are the
samples and rows give the taxa.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of frequency count tables corresponding to the columns.
</p>

<hr>
<h2 id='chao_bunge'>Chao-Bunge species richness estimator</h2><span id='topic+chao_bunge'></span>

<h3>Description</h3>

<p>This function implements the species richness estimation procedure outlined
in Chao &amp; Bunge (2002).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chao_bunge(input_data, cutoff = 10, output = NULL, answers = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chao_bunge_+3A_input_data">input_data</code></td>
<td>
<p>An input type that can be processed by <code>convert()</code> or a <code>phyloseq</code> object</p>
</td></tr>
<tr><td><code id="chao_bunge_+3A_cutoff">cutoff</code></td>
<td>
<p>The maximum frequency to use in fitting.</p>
</td></tr>
<tr><td><code id="chao_bunge_+3A_output">output</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
<tr><td><code id="chao_bunge_+3A_answers">answers</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>alpha_estimate</code>, or <code>alpha_estimates</code> for <code>phyloseq</code> objects
</p>


<h3>Author(s)</h3>

<p>Amy Willis
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
chao_bunge(apples)

</code></pre>

<hr>
<h2 id='chao_shen'>The Chao-Shen estimate of Shannon diversity</h2><span id='topic+chao_shen'></span>

<h3>Description</h3>

<p>The Chao-Shen estimate of Shannon diversity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chao_shen(input_data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chao_shen_+3A_input_data">input_data</code></td>
<td>
<p>An input type that can be processed by <code>convert()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>alpha_estimate</code>
</p>

<hr>
<h2 id='chao1'>Chao1 species richness estimator</h2><span id='topic+chao1'></span>

<h3>Description</h3>

<p>This function implements the Chao1 richness estimate, which is often
mistakenly referred to as an index.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chao1(input_data, output = NULL, answers = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chao1_+3A_input_data">input_data</code></td>
<td>
<p>An input type that can be processed by <code>convert()</code> or a <code>phyloseq</code> object</p>
</td></tr>
<tr><td><code id="chao1_+3A_output">output</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
<tr><td><code id="chao1_+3A_answers">answers</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>alpha_estimate</code>, or <code>alpha_estimates</code> for <code>phyloseq</code> objects
</p>


<h3>Note</h3>

<p>The authors of this package strongly discourage the use of this
estimator.  It is only valid when you wish to assume that every taxa has
equal probability of being observed. You don't really think that's possible,
do you?
</p>


<h3>Author(s)</h3>

<p>Amy Willis
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

chao1(apples)


</code></pre>

<hr>
<h2 id='chao1_bc'>Bias-corrected Chao1 species richness estimator</h2><span id='topic+chao1_bc'></span>

<h3>Description</h3>

<p>This function implements the bias-corrected Chao1 richness estimate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chao1_bc(input_data, output = NULL, answers = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chao1_bc_+3A_input_data">input_data</code></td>
<td>
<p>An input type that can be processed by <code>convert()</code> or a <code>phyloseq</code> object</p>
</td></tr>
<tr><td><code id="chao1_bc_+3A_output">output</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
<tr><td><code id="chao1_bc_+3A_answers">answers</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>alpha_estimate</code>, or <code>alpha_estimates</code> for <code>phyloseq</code> objects
</p>


<h3>Note</h3>

<p>The authors of this package strongly discourage the use of this
estimator.  It is only valid when you wish to assume that every taxa has
equal probability of being observed. You don't really think that's possible,
do you?
</p>


<h3>Author(s)</h3>

<p>Amy Willis
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

chao1_bc(apples)


</code></pre>

<hr>
<h2 id='check_format'>Run some basic checks on a possible frequency count table</h2><span id='topic+check_format'></span>

<h3>Description</h3>

<p>Run some basic checks on a possible frequency count table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_format(output_data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_format_+3A_output_data">output_data</code></td>
<td>
<p>A matrix to test</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A checked frequency count table
</p>

<hr>
<h2 id='convert'>convert between different inputs for alpha-diversity estimates</h2><span id='topic+convert'></span>

<h3>Description</h3>

<p>Inputs slated for development include phyloseq and otu_table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert(input_data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert_+3A_input_data">input_data</code></td>
<td>
<p>Supported types include filenames, data frames, matrices, vectors...</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Frequency count able
</p>

<hr>
<h2 id='cutoff_wrap'>Find a cut-off for estimates relying on contiguous counts</h2><span id='topic+cutoff_wrap'></span>

<h3>Description</h3>

<p>Find a cut-off for estimates relying on contiguous counts
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cutoff_wrap(my_data, requested = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cutoff_wrap_+3A_my_data">my_data</code></td>
<td>
<p>Frequency count table</p>
</td></tr>
<tr><td><code id="cutoff_wrap_+3A_requested">requested</code></td>
<td>
<p>The user-requested cutoff</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Cutoff value
</p>

<hr>
<h2 id='F_test'>Conduct F test of null hypothesis LB = 0 using output from betta()
or betta_random()</h2><span id='topic+F_test'></span>

<h3>Description</h3>

<p>This function performs an F-test of a null hypothesis LB = 0 where B is a
vector of p fixed effects returned by betta() or betta_random() and L is an
m x p matrix with linearly independent rows.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>F_test(fitted_betta, L, method = "bootstrap", nboot = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="F_test_+3A_fitted_betta">fitted_betta</code></td>
<td>
<p>A fitted betta object &ndash; i.e., the output of either betta() or
betta_random() &ndash; containing fixed effect estimates of interest.</p>
</td></tr>
<tr><td><code id="F_test_+3A_l">L</code></td>
<td>
<p>An m x p matrix defining the null LB = 0. L must have full row rank.</p>
</td></tr>
<tr><td><code id="F_test_+3A_method">method</code></td>
<td>
<p>A character variable indicating which method should be used to
estimate the distribution of the test statistic under the null.</p>
</td></tr>
<tr><td><code id="F_test_+3A_nboot">nboot</code></td>
<td>
<p>Number of bootstrap samples to use if method = &quot;bootstrap&quot;.
Ignored if method = &quot;asymptotic&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing
</p>
<table>
<tr><td><code>pval</code></td>
<td>
<p>The p-value</p>
</td></tr>
<tr><td><code>F_stat</code></td>
<td>
<p>The calculated F statistic</p>
</td></tr>
<tr><td><code>boot_F</code></td>
<td>
<p>A vector of bootstrapped F statistics if bootstrap has been used.
Otherwise NULL.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>David Clausen
</p>


<h3>References</h3>

<p>Willis, A., Bunge, J., and Whitman, T. (2015). Inference for
changes in biodiversity. <em>arXiv preprint.</em>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+betta">betta</a></code>;
</p>
<p><code><a href="#topic+betta_random">betta_random</a></code>;
</p>
<p><code><a href="#topic+betta">betta</a></code>;
</p>
<p><code><a href="#topic+get_F_stat">get_F_stat</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# generate example data
df &lt;- data.frame(chats = c(2000, 3000, 4000, 3000,
2000, 3000, 4000, 3000), ses = c(100, 200, 150, 180,
100, 200, 150, 180),
                 Cont_var = c(100, 150, 100, 50,
                 100, 150, 100, 50),
                 Cont_var_2 = c(50,200,25,125,
                 50,200,25,125))

# fit betta()
example_fit &lt;- betta(formula = chats ~ Cont_var + Cont_var_2, ses = ses, data = df)


# construct L for hypothesis that B_cont_var = B_cont_var_2 = 0
L &lt;- rbind(c(0,1,0),
           c(0,0,1))

F_test_results &lt;- F_test(example_fit, L, nboot = 100)

</code></pre>

<hr>
<h2 id='get_F_stat'>Calculate F statistic under null hypothesis LB = 0 using output from betta()
or betta_random()</h2><span id='topic+get_F_stat'></span>

<h3>Description</h3>

<p>This function calculates an F statistic for a test of null hypothesis LB = 0
(against an unrestricted alternative) where B is a
vector of p fixed effects returned by betta() or betta_random() and L is an
m x p matrix with linearly independent rows.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_F_stat(fitted_betta, L)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_F_stat_+3A_fitted_betta">fitted_betta</code></td>
<td>
<p>A fitted betta object &ndash; i.e., the output of either betta() or
betta_random() &ndash; containing fixed effect estimates of interest.</p>
</td></tr>
<tr><td><code id="get_F_stat_+3A_l">L</code></td>
<td>
<p>An m x p matrix defining the null LB = 0. L must have full row rank.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing
</p>
<table>
<tr><td><code>F_stat</code></td>
<td>
<p>The calculated F statistic</p>
</td></tr>
</table>

<hr>
<h2 id='good_turing'>The Good-Turing estimate of species richness</h2><span id='topic+good_turing'></span>

<h3>Description</h3>

<p>The Good-Turing estimate of species richness
</p>


<h3>Usage</h3>

<pre><code class='language-R'>good_turing(input_data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="good_turing_+3A_input_data">input_data</code></td>
<td>
<p>An input type that can be processed by <code>convert()</code> or a <code>phyloseq</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>alpha_estimate</code>, or <code>alpha_estimates</code> for <code>phyloseq</code> objects
</p>

<hr>
<h2 id='hawaii'>(Data) Frequency count table of soil microbes in Hawaii.</h2><span id='topic+hawaii'></span>

<h3>Description</h3>

<p>(Data) Frequency count table of soil microbes in Hawaii.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hawaii
</code></pre>


<h3>Format</h3>

<p>A data frame with 198 rows and 2 variables:
</p>

<dl>
<dt>index</dt><dd><p>an index variable</p>
</dd>
<dt>frequency</dt><dd><p>number of taxa that were observed with this frequency</p>
</dd>
</dl>
<p>...

</p>


<h3>References</h3>

<p>Willis, A. and Bunge, J. (2015). Estimating diversity via
frequency ratios. <em>Biometrics</em>, <b>71</b>(4), 1042&ndash;1049.
<a href="https://doi.org/10.1111/biom.12332">doi:10.1111/biom.12332</a>
</p>

<hr>
<h2 id='kemp'>Species richness estimation with Kemp-type models</h2><span id='topic+kemp'></span>

<h3>Description</h3>

<p>This function implements the species richness estimation procedure outlined
in Willis &amp; Bunge (2015). The diversity estimate, standard error, estimated
model coefficients, model details and plot of the fitted model are returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kemp(
  input_data,
  cutoff = NA,
  output = NULL,
  plot = NULL,
  answers = NULL,
  print = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kemp_+3A_input_data">input_data</code></td>
<td>
<p>An input type that can be processed by <code>convert()</code></p>
</td></tr>
<tr><td><code id="kemp_+3A_cutoff">cutoff</code></td>
<td>
<p>The maximum frequency count to use for model fitting</p>
</td></tr>
<tr><td><code id="kemp_+3A_output">output</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
<tr><td><code id="kemp_+3A_plot">plot</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
<tr><td><code id="kemp_+3A_answers">answers</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
<tr><td><code id="kemp_+3A_print">print</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
<tr><td><code id="kemp_+3A_...">...</code></td>
<td>
<p>Additional arguments will be ignored; this is for backward compatibility</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>alpha_estimate</code> </p>
<table>
<tr><td><code>code</code></td>
<td>
<p> A category representing algorithm behaviour.
&lsquo;<span class="samp">&#8288;code=1&#8288;</span>&rsquo; indicates no nonlinear models converged and the transformed
WLRM diversity estimate of Rocchetti et. al. (2011) is returned.
&lsquo;<span class="samp">&#8288;code=2&#8288;</span>&rsquo; indicates that the iteratively reweighted model converged and
was returned. &lsquo;<span class="samp">&#8288;code=3&#8288;</span>&rsquo; indicates that iterative reweighting did not
converge but a model based on a simplified variance structure was returned
(in this case, the variance of the frequency ratios is assumed to be
proportional to the denominator frequency index). Please peruse your fitted
model before using your diversity estimate.  </p>
</td></tr> <tr><td><code>name</code></td>
<td>
<p> The &ldquo;name&rdquo; of
the selected model. The first integer represents the numerator polynomial
degree and the second integer represents the denominator polynomial degree
of the model for the frequency ratios. See Willis &amp; Bunge (2015) for
details.  </p>
</td></tr> <tr><td><code>para</code></td>
<td>
<p> Estimated model parameters and standard errors.  </p>
</td></tr>
<tr><td><code>est</code></td>
<td>
<p> The estimate of total (observed plus unobserved) diversity.  </p>
</td></tr>
<tr><td><code>seest</code></td>
<td>
<p> The standard error in the diversity estimate.  </p>
</td></tr> <tr><td><code>full</code></td>
<td>

<p>The chosen nonlinear model for frequency ratios.  </p>
</td></tr> <tr><td><code>ci</code></td>
<td>
<p> An asymmetric
95% confidence interval for diversity.  </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Amy Willis
</p>


<h3>References</h3>

<p>Willis, A. and Bunge, J. (2015). Estimating diversity via
frequency ratios. <em>Biometrics</em>, <b>71</b>(4), 1042&ndash;1049.
</p>
<p>Rocchetti, I., Bunge, J. and Bohning, D. (2011). Population size estimation
based upon ratios of recapture probabilities. <em>Annals of Applied
Statistics</em>, <b>5</b>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+breakaway">breakaway</a></code>; <code><a href="#topic+breakaway_nof1">breakaway_nof1</a></code>; <code><a href="#topic+apples">apples</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
kemp(apples)
kemp(apples, plot = FALSE, output = FALSE, answers = TRUE)

</code></pre>

<hr>
<h2 id='make_design_matrix'>Make design matrix</h2><span id='topic+make_design_matrix'></span>

<h3>Description</h3>

<p>Make design matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_design_matrix(phyloseq_object, variables)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make_design_matrix_+3A_phyloseq_object">phyloseq_object</code></td>
<td>
<p>A phyloseq object</p>
</td></tr>
<tr><td><code id="make_design_matrix_+3A_variables">variables</code></td>
<td>
<p>variable names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix object giving the design matrix from the desired variables.
</p>

<hr>
<h2 id='make_frequency_count_table'>Draw frequency count subtables from an OTU table</h2><span id='topic+make_frequency_count_table'></span>

<h3>Description</h3>

<p>Draw frequency count subtables from an OTU table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_frequency_count_table(labels)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make_frequency_count_table_+3A_labels">labels</code></td>
<td>
<p>A vector of counts of the taxa; i.e. a vector giving the number of times each taxon was observed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A frequency count table.
</p>

<hr>
<h2 id='objective_bayes_geometric'>Estimate species richness with an objective Bayes method using a geometric model</h2><span id='topic+objective_bayes_geometric'></span>

<h3>Description</h3>

<p>Estimate species richness with an objective Bayes method using a geometric model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>objective_bayes_geometric(
  data,
  output = TRUE,
  plot = TRUE,
  answers = FALSE,
  tau = 10,
  burn.in = 100,
  iterations = 2500,
  Metropolis.stdev.N = 75,
  Metropolis.start.theta = 1,
  Metropolis.stdev.theta = 0.3
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="objective_bayes_geometric_+3A_data">data</code></td>
<td>
<p>TODO(Kathryn)(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_geometric_+3A_output">output</code></td>
<td>
<p>TODO(Kathryn)(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_geometric_+3A_plot">plot</code></td>
<td>
<p>TODO(Kathryn)(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_geometric_+3A_answers">answers</code></td>
<td>
<p>TODO(Kathryn)(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_geometric_+3A_tau">tau</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_geometric_+3A_burn.in">burn.in</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_geometric_+3A_iterations">iterations</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_geometric_+3A_metropolis.stdev.n">Metropolis.stdev.N</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_geometric_+3A_metropolis.start.theta">Metropolis.start.theta</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_geometric_+3A_metropolis.stdev.theta">Metropolis.stdev.theta</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of results, including </p>
<table>
<tr><td><code>est</code></td>
<td>
<p>the median of estimates of N</p>
</td></tr></table>
<p>, </p>
<table>
<tr><td><code>ci</code></td>
<td>
<p>a confidence interval for N</p>
</td></tr></table>
<p>,
</p>
<table>
<tr><td><code>mean</code></td>
<td>
<p>the mean of estimates of N</p>
</td></tr></table>
<p>, </p>
<table>
<tr><td><code>semeanest</code></td>
<td>
<p>the standard error of mean estimates</p>
</td></tr></table>
<p>,
</p>
<table>
<tr><td><code>dic</code></td>
<td>
<p>the DIC of the model</p>
</td></tr></table>
<p>, </p>
<table>
<tr><td><code>fits</code></td>
<td>
<p>fitted values</p>
</td></tr></table>
<p>, and </p>
<table>
<tr><td><code>diagnostics</code></td>
<td>
<p>model diagonstics</p>
</td></tr></table>
<p>.
</p>

<hr>
<h2 id='objective_bayes_mixedgeo'>Objective Bayes species richness estimate with the mixed-geometric model</h2><span id='topic+objective_bayes_mixedgeo'></span>

<h3>Description</h3>

<p>Objective Bayes species richness estimate with the mixed-geometric model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>objective_bayes_mixedgeo(
  data,
  output = TRUE,
  plot = TRUE,
  answers = FALSE,
  tau = 10,
  burn.in = 100,
  iterations = 2500,
  Metropolis.stdev.N = 100,
  Metropolis.start.T1 = 1,
  Metropolis.stdev.T1 = 2,
  Metropolis.start.T2 = 3,
  Metropolis.stdev.T2 = 2,
  bars = 3
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="objective_bayes_mixedgeo_+3A_data">data</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_mixedgeo_+3A_output">output</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_mixedgeo_+3A_plot">plot</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_mixedgeo_+3A_answers">answers</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_mixedgeo_+3A_tau">tau</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_mixedgeo_+3A_burn.in">burn.in</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_mixedgeo_+3A_iterations">iterations</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_mixedgeo_+3A_metropolis.stdev.n">Metropolis.stdev.N</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_mixedgeo_+3A_metropolis.start.t1">Metropolis.start.T1</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_mixedgeo_+3A_metropolis.stdev.t1">Metropolis.stdev.T1</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_mixedgeo_+3A_metropolis.start.t2">Metropolis.start.T2</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_mixedgeo_+3A_metropolis.stdev.t2">Metropolis.stdev.T2</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_mixedgeo_+3A_bars">bars</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of results, including </p>
<table>
<tr><td><code>est</code></td>
<td>
<p>the median of estimates of N</p>
</td></tr></table>
<p>, </p>
<table>
<tr><td><code>ci</code></td>
<td>
<p>a confidence interval for N</p>
</td></tr></table>
<p>,
</p>
<table>
<tr><td><code>mean</code></td>
<td>
<p>the mean of estimates of N</p>
</td></tr></table>
<p>, </p>
<table>
<tr><td><code>semeanest</code></td>
<td>
<p>the standard error of mean estimates</p>
</td></tr></table>
<p>,
</p>
<table>
<tr><td><code>dic</code></td>
<td>
<p>the DIC of the model</p>
</td></tr></table>
<p>, </p>
<table>
<tr><td><code>fits</code></td>
<td>
<p>fitted values</p>
</td></tr></table>
<p>, and </p>
<table>
<tr><td><code>diagnostics</code></td>
<td>
<p>model diagonstics</p>
</td></tr></table>
<p>.
</p>

<hr>
<h2 id='objective_bayes_negbin'>Objective Bayes species richness estimate with the Negative Binomial model</h2><span id='topic+objective_bayes_negbin'></span>

<h3>Description</h3>

<p>Objective Bayes species richness estimate with the Negative Binomial model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>objective_bayes_negbin(
  data,
  output = TRUE,
  plot = TRUE,
  answers = FALSE,
  tau = 10,
  burn.in = 1000,
  iterations = 5000,
  Metropolis.stdev.N = 100,
  Metropolis.start.T1 = -0.8,
  Metropolis.stdev.T1 = 0.01,
  Metropolis.start.T2 = 0.8,
  Metropolis.stdev.T2 = 0.01,
  bars = 5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="objective_bayes_negbin_+3A_data">data</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_negbin_+3A_output">output</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_negbin_+3A_plot">plot</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_negbin_+3A_answers">answers</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_negbin_+3A_tau">tau</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_negbin_+3A_burn.in">burn.in</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_negbin_+3A_iterations">iterations</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_negbin_+3A_metropolis.stdev.n">Metropolis.stdev.N</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_negbin_+3A_metropolis.start.t1">Metropolis.start.T1</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_negbin_+3A_metropolis.stdev.t1">Metropolis.stdev.T1</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_negbin_+3A_metropolis.start.t2">Metropolis.start.T2</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_negbin_+3A_metropolis.stdev.t2">Metropolis.stdev.T2</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_negbin_+3A_bars">bars</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of results, including </p>
<table>
<tr><td><code>est</code></td>
<td>
<p>the median of estimates of N</p>
</td></tr></table>
<p>, </p>
<table>
<tr><td><code>ci</code></td>
<td>
<p>a confidence interval for N</p>
</td></tr></table>
<p>,
</p>
<table>
<tr><td><code>mean</code></td>
<td>
<p>the mean of estimates of N</p>
</td></tr></table>
<p>, </p>
<table>
<tr><td><code>semeanest</code></td>
<td>
<p>the standard error of mean estimates</p>
</td></tr></table>
<p>,
</p>
<table>
<tr><td><code>dic</code></td>
<td>
<p>the DIC of the model</p>
</td></tr></table>
<p>, </p>
<table>
<tr><td><code>fits</code></td>
<td>
<p>fitted values</p>
</td></tr></table>
<p>, and </p>
<table>
<tr><td><code>diagnostics</code></td>
<td>
<p>model diagonstics</p>
</td></tr></table>
<p>.
</p>

<hr>
<h2 id='objective_bayes_poisson'>Objective Bayes species richness estimate with the Poisson model</h2><span id='topic+objective_bayes_poisson'></span>

<h3>Description</h3>

<p>Objective Bayes species richness estimate with the Poisson model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>objective_bayes_poisson(
  data,
  output = TRUE,
  plot = TRUE,
  answers = FALSE,
  tau = 10,
  burn.in = 100,
  iterations = 2500,
  Metropolis.stdev.N = 75,
  Metropolis.start.lambda = 1,
  Metropolis.stdev.lambda = 0.3,
  bars = 5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="objective_bayes_poisson_+3A_data">data</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_poisson_+3A_output">output</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_poisson_+3A_plot">plot</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_poisson_+3A_answers">answers</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_poisson_+3A_tau">tau</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_poisson_+3A_burn.in">burn.in</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_poisson_+3A_iterations">iterations</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_poisson_+3A_metropolis.stdev.n">Metropolis.stdev.N</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_poisson_+3A_metropolis.start.lambda">Metropolis.start.lambda</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_poisson_+3A_metropolis.stdev.lambda">Metropolis.stdev.lambda</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
<tr><td><code id="objective_bayes_poisson_+3A_bars">bars</code></td>
<td>
<p>TODO(Kathryn)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of results, including </p>
<table>
<tr><td><code>est</code></td>
<td>
<p>the median of estimates of N</p>
</td></tr></table>
<p>, </p>
<table>
<tr><td><code>ci</code></td>
<td>
<p>a confidence interval for N</p>
</td></tr></table>
<p>,
</p>
<table>
<tr><td><code>mean</code></td>
<td>
<p>the mean of estimates of N</p>
</td></tr></table>
<p>, </p>
<table>
<tr><td><code>semeanest</code></td>
<td>
<p>the standard error of mean estimates</p>
</td></tr></table>
<p>,
</p>
<table>
<tr><td><code>dic</code></td>
<td>
<p>the DIC of the model</p>
</td></tr></table>
<p>, </p>
<table>
<tr><td><code>fits</code></td>
<td>
<p>fitted values</p>
</td></tr></table>
<p>, and </p>
<table>
<tr><td><code>diagnostics</code></td>
<td>
<p>model diagonstics</p>
</td></tr></table>
<p>.
</p>

<hr>
<h2 id='pasolli_et_al'>(Data) Data frame of covariate information about pasolli_et_al.</h2><span id='topic+pasolli_et_al'></span>

<h3>Description</h3>

<p>(Data) Data frame of covariate information about pasolli_et_al.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pasolli_et_al
</code></pre>


<h3>Format</h3>

<p>A data frame with 4930 rows and 9 variables:
</p>

<dl>
<dt>SGB.ID</dt><dd><p>sample ID</p>
</dd>
<dt>AverageAbundance</dt><dd><p>average abundance in the sample</p>
</dd>
<dt>std</dt><dd><p>standard deviation</p>
</dd>
<dt>q1</dt><dd><p>first quartile</p>
</dd>
<dt>MedianAbundance</dt><dd><p>median abundance</p>
</dd>
<dt>q3</dt><dd><p>third quartile</p>
</dd>
<dt>min</dt><dd><p>minimum</p>
</dd>
<dt>max</dt><dd><p>maximum</p>
</dd>
<dt>#.Samples</dt><dd><p>number of samples</p>
</dd>
</dl>
<p>...

</p>

<hr>
<h2 id='physeq_wrap'>Wrapper for phyloseq objects</h2><span id='topic+physeq_wrap'></span>

<h3>Description</h3>

<p>Wrapper for phyloseq objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>physeq_wrap(fn, physeq, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="physeq_wrap_+3A_fn">fn</code></td>
<td>
<p>alpha diversity estimator function with <code>breakaway</code> to be applied to <code>physeq</code></p>
</td></tr>
<tr><td><code id="physeq_wrap_+3A_physeq">physeq</code></td>
<td>
<p>A <code>phyloseq</code> object, or an object of class <code>otu_table</code></p>
</td></tr>
<tr><td><code id="physeq_wrap_+3A_...">...</code></td>
<td>
<p>Additional arguments for fn</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Object of class <code>alpha_estimates</code>
</p>

<hr>
<h2 id='plot.alpha_estimates'>Plot function for alpha_estimates class</h2><span id='topic+plot.alpha_estimates'></span>

<h3>Description</h3>

<p>Plot function for alpha_estimates class
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'alpha_estimates'
plot(
  x,
  physeq = NULL,
  measure = NULL,
  color = NULL,
  shape = NULL,
  title = NULL,
  trim_plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.alpha_estimates_+3A_x">x</code></td>
<td>
<p>Object of class <code>alpha_estimates</code>.</p>
</td></tr>
<tr><td><code id="plot.alpha_estimates_+3A_physeq">physeq</code></td>
<td>
<p>(Optional). Default <code>NULL</code>. Required object of class <code>phyloseq</code> if including a <code>sample_data</code> variable for <code>color</code> or <code>shape</code>.</p>
</td></tr>
<tr><td><code id="plot.alpha_estimates_+3A_measure">measure</code></td>
<td>
<p>(Optional). If there are multiple richness measures included in <code>x</code>, this can be set to the the desired measure to be plotted. Defaults to the measure of the first alpha diversity estimate.</p>
</td></tr>
<tr><td><code id="plot.alpha_estimates_+3A_color">color</code></td>
<td>
<p>(Optional). Default <code>NULL</code>. The sample variable to map to different colors. Can be a single character string of the variable name in <code>sample_data</code> or a custom supplied vector with length equal to the number of samples.</p>
</td></tr>
<tr><td><code id="plot.alpha_estimates_+3A_shape">shape</code></td>
<td>
<p>(Optional). Default <code>NULL</code>. The sample variable to map to different shapes. Can be a single character string of the variable name in <code>sample_data</code> or a custom supplied vector with length equal to the number of samples.</p>
</td></tr>
<tr><td><code id="plot.alpha_estimates_+3A_title">title</code></td>
<td>
<p>(Optional). Default NULL. Character string. The main title for the graphic.</p>
</td></tr>
<tr><td><code id="plot.alpha_estimates_+3A_trim_plot">trim_plot</code></td>
<td>
<p>(Optional). Default <code>FALSE</code>. Boolean indicator for whether you want the plot to include the full confidence intervals.</p>
</td></tr>
<tr><td><code id="plot.alpha_estimates_+3A_...">...</code></td>
<td>
<p>See details</p>
</td></tr>
</table>


<h3>Details</h3>

<p>... does not currently have any implemented options. Optional arguments currently include &quot;trim_plot&quot;, a  Optional
</p>


<h3>Value</h3>

<p>A ggplot object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(phyloseq)
data(GlobalPatterns)
alphas &lt;- breakaway(GlobalPatterns)
plot(alphas)

</code></pre>

<hr>
<h2 id='poisson_model'>PoissonModel</h2><span id='topic+poisson_model'></span>

<h3>Description</h3>

<p>A model to estimate the number of missing taxa under a Poisson Model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>poisson_model(input_data, cutoff = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="poisson_model_+3A_input_data">input_data</code></td>
<td>
<p>A frequency count table</p>
</td></tr>
<tr><td><code id="poisson_model_+3A_cutoff">cutoff</code></td>
<td>
<p>The largest frequency to use for predicting f0. Defaults to 10.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>alpha_estimate</code>.
</p>

<hr>
<h2 id='poisson_model_nof1'>PoissonModelNof1</h2><span id='topic+poisson_model_nof1'></span>

<h3>Description</h3>

<p>A model to estimate the number of missing taxa under a zero- and one-truncated Poisson Model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>poisson_model_nof1(input_data, cutoff = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="poisson_model_nof1_+3A_input_data">input_data</code></td>
<td>
<p>A frequency count table</p>
</td></tr>
<tr><td><code id="poisson_model_nof1_+3A_cutoff">cutoff</code></td>
<td>
<p>The largest frequency to use for predicting f0 and f1. Defaults to 10.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>alpha_estimate</code>.
</p>

<hr>
<h2 id='proportions_instead'>OTU table to relative abundances</h2><span id='topic+proportions_instead'></span>

<h3>Description</h3>

<p>OTU table to relative abundances
</p>


<h3>Usage</h3>

<pre><code class='language-R'>proportions_instead(the_table)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="proportions_instead_+3A_the_table">the_table</code></td>
<td>
<p>An OTU table</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A proportion table or vector.
</p>

<hr>
<h2 id='rnbinomtable'>Negative binomially distributed frequency count tables.</h2><span id='topic+rnbinomtable'></span>

<h3>Description</h3>

<p>Simulate a frequency count table based on a negative binomial model.
Zero-truncated, obviously.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rnbinomtable(C, size, probability)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rnbinomtable_+3A_c">C</code></td>
<td>
<p>species richness</p>
</td></tr>
<tr><td><code id="rnbinomtable_+3A_size">size</code></td>
<td>
<p>size parameter for the negative binomial distribution</p>
</td></tr>
<tr><td><code id="rnbinomtable_+3A_probability">probability</code></td>
<td>
<p>probability parameter for the negative binomial
distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A simulated frequency count table.
</p>


<h3>Author(s)</h3>

<p>Amy Willis
</p>

<hr>
<h2 id='rztnbinomtable'>beta version: Zero-truncated negative binomially distributed frequency count
tables.</h2><span id='topic+rztnbinomtable'></span>

<h3>Description</h3>

<p>Simulate a frequency count table based on a negative binomial model.
Zero-truncated, obviously.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rztnbinomtable(C, size, probability)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rztnbinomtable_+3A_c">C</code></td>
<td>
<p>species richness</p>
</td></tr>
<tr><td><code id="rztnbinomtable_+3A_size">size</code></td>
<td>
<p>size parameter for the negative binomial distribution</p>
</td></tr>
<tr><td><code id="rztnbinomtable_+3A_probability">probability</code></td>
<td>
<p>probability parameter for the negative binomial
distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A simulated frequency count table.
</p>


<h3>Author(s)</h3>

<p>Amy Willis
</p>

<hr>
<h2 id='sample_inverse_simpson'>Plug-in Inverse Simpson diversity</h2><span id='topic+sample_inverse_simpson'></span>

<h3>Description</h3>

<p>This function implements the plug-in Inverse Simpson diversity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_inverse_simpson(input_data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_inverse_simpson_+3A_input_data">input_data</code></td>
<td>
<p>An input type that can be processed by <code>convert()</code> or a <code>phyloseq</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>alpha_estimate</code>, or <code>alpha_estimates</code> for <code>phyloseq</code> objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sample_inverse_simpson(apples)

</code></pre>

<hr>
<h2 id='sample_richness'>Sample richness estimator</h2><span id='topic+sample_richness'></span>

<h3>Description</h3>

<p>This function implements the sample richness estimate, which is the number of non-zero taxa per sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_richness(input_data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_richness_+3A_input_data">input_data</code></td>
<td>
<p>An input type that can be processed by <code>convert()</code> or a <code>phyloseq</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>alpha_estimate</code>, or <code>alpha_estimates</code> for <code>phyloseq</code> objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

sample_richness(apples)


</code></pre>

<hr>
<h2 id='sample_shannon'>Plug-in Shannon diversity</h2><span id='topic+sample_shannon'></span>

<h3>Description</h3>

<p>This function implements the plug-in Shannon diversity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_shannon(input_data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_shannon_+3A_input_data">input_data</code></td>
<td>
<p>An input type that can be processed by <code>convert()</code> or a <code>phyloseq</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>alpha_estimate</code>, or <code>alpha_estimates</code> for <code>phyloseq</code> objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sample_shannon(apples)

</code></pre>

<hr>
<h2 id='sample_shannon_e'>Plug-in Shannon's E (&quot;Equitability&quot;)</h2><span id='topic+sample_shannon_e'></span>

<h3>Description</h3>

<p>This function implements the plug-in Shannon's E
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_shannon_e(input_data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_shannon_e_+3A_input_data">input_data</code></td>
<td>
<p>An input type that can be processed by <code>convert()</code> or a <code>phyloseq</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>alpha_estimate</code>, or <code>alpha_estimates</code> for <code>phyloseq</code> objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sample_shannon_e(apples)

</code></pre>

<hr>
<h2 id='sample_simpson'>Plug-in Simpson diversity</h2><span id='topic+sample_simpson'></span>

<h3>Description</h3>

<p>This function implements the plug-in Simpson diversity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_simpson(input_data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_simpson_+3A_input_data">input_data</code></td>
<td>
<p>An input type that can be processed by <code>convert()</code> or a <code>phyloseq</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>alpha_estimate</code>, or <code>alpha_estimates</code> for <code>phyloseq</code> objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sample_simpson(apples)

</code></pre>

<hr>
<h2 id='sample_size_estimate'>Estimate the sample size needed to do an unpaired one-way test using betta</h2><span id='topic+sample_size_estimate'></span>

<h3>Description</h3>

<p>Estimate the sample size needed to do an unpaired one-way test using betta
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_size_estimate(
  control_group_est,
  se_est,
  diff = 5,
  alpha = 0.05,
  prop = 0.8,
  samples = 20,
  precision = 5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_size_estimate_+3A_control_group_est">control_group_est</code></td>
<td>
<p>An estimate of the alpha diversity parameter for
the control group</p>
</td></tr>
<tr><td><code id="sample_size_estimate_+3A_se_est">se_est</code></td>
<td>
<p>An estimate of the (common) standard deviation</p>
</td></tr>
<tr><td><code id="sample_size_estimate_+3A_diff">diff</code></td>
<td>
<p>An estimate of the difference between the control and
non-control groups</p>
</td></tr>
<tr><td><code id="sample_size_estimate_+3A_alpha">alpha</code></td>
<td>
<p>Minimum significance level desired</p>
</td></tr>
<tr><td><code id="sample_size_estimate_+3A_prop">prop</code></td>
<td>
<p>Desired power</p>
</td></tr>
<tr><td><code id="sample_size_estimate_+3A_samples">samples</code></td>
<td>
<p>Number of bootstrap resamples used to estimate the sample
size. Increase for a more accurate estimate.</p>
</td></tr>
<tr><td><code id="sample_size_estimate_+3A_precision">precision</code></td>
<td>
<p>How much to increment the sample size as we try to increase
the power</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An estimate of the necessary sample size and some details
</p>

<hr>
<h2 id='sample_size_figure'>Plot the power obtained with sample size</h2><span id='topic+sample_size_figure'></span>

<h3>Description</h3>

<p>Plot the power obtained with sample size
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_size_figure(control_group_est, se_est, diff = 5, samples = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_size_figure_+3A_control_group_est">control_group_est</code></td>
<td>
<p>An estimate of the alpha diversity parameter for
the control group</p>
</td></tr>
<tr><td><code id="sample_size_figure_+3A_se_est">se_est</code></td>
<td>
<p>An estimate of the (common) standard deviation</p>
</td></tr>
<tr><td><code id="sample_size_figure_+3A_diff">diff</code></td>
<td>
<p>An estimate of the difference between the control and
non-control groups</p>
</td></tr>
<tr><td><code id="sample_size_figure_+3A_samples">samples</code></td>
<td>
<p>Number of bootstrap resamples used to estimate the sample
size. Increase for a more accurate estimate.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot of the power with the sample size
</p>

<hr>
<h2 id='simulate_betta'>Simulate from a fitted betta model</h2><span id='topic+simulate_betta'></span>

<h3>Description</h3>

<p>Simulate from a fitted betta model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulate_betta(fitted_betta, nsim)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulate_betta_+3A_fitted_betta">fitted_betta</code></td>
<td>
<p>A fitted betta object</p>
</td></tr>
<tr><td><code id="simulate_betta_+3A_nsim">nsim</code></td>
<td>
<p>Number of times to simulate</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of length nsim, each element of which is a vector of
simulated Y-values under the fitted betta model
</p>

<hr>
<h2 id='simulate_betta_random'>Simulate from a fitted betta_random model</h2><span id='topic+simulate_betta_random'></span>

<h3>Description</h3>

<p>Simulate from a fitted betta_random model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulate_betta_random(fitted_betta, nsim)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulate_betta_random_+3A_fitted_betta">fitted_betta</code></td>
<td>
<p>A fitted betta_random object</p>
</td></tr>
<tr><td><code id="simulate_betta_random_+3A_nsim">nsim</code></td>
<td>
<p>Number of times to simulate</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of length nsim, each element of which is a vector of
simulated Y-values under the fitted betta model
</p>

<hr>
<h2 id='soil_phylo'>(Data) Data frame of soil data from whitman_et_al.</h2><span id='topic+soil_phylo'></span>

<h3>Description</h3>

<p>A phyloseq object with an OTU table and sample data from a soil microbiome study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>soil_phylo
</code></pre>


<h3>Format</h3>

<p>A phyloseq-class experiment-level object with an OTU table and sample data.
</p>

<dl>
<dt>otu_table</dt><dd><p>OTU table with 7,770 taxa and 119 samples</p>
</dd>
<dt>tax_table</dt><dd><p>taxonomy table</p>
</dd>
<dt>sam_data</dt><dd><p>sample data with the following covariates:
</p>

<ul>
<li> <p><code>Plants</code>, values <code>0</code> and <code>1</code>. Index for different plants
</p>
</li>
<li> <p><code>Day</code>, values <code>0</code> (initial sampling point), <code>1</code> (12 days after treatment additions), and <code>2</code> (82 days after treatment additions). Index for different days of measurement
</p>
</li>
<li> <p><code>Amdmt</code>, values <code>0</code> (no additions), <code>1</code> (biochar additions), and <code>2</code> (fresh biomass additions). Index for different soil additives.
</p>
</li>
<li> <p><code>DayAmdmt</code>, values <code>00</code>, <code>01</code>, <code>02</code>, <code>10</code>, <code>11</code>, <code>12</code>, <code>20</code>, <code>21</code>, and <code>22</code>. A single index for the combination of <code>Day</code> and <code>Amdmt</code> with <code>Day</code> as the first digit and <code>Amdmt</code> as the second digit.
</p>
</li>
<li> <p><code>ID</code>, values <code>A</code>, <code>B</code>, <code>C</code>, <code>D</code>,  and <code>F</code>. Index for different soil plots.
</p>
</li></ul>
</dd>
</dl>
<p>...

</p>


<h3>References</h3>

<p>Whitman, T., Pepe-Ranney, C., Enders, A., Koechli, C., Campbell, A.,  Buckley, D. H., Lehmann, J. (2016). <em>Dynamics of microbial community composi-tion and soil organic carbon mineralization in soil following addition of pyrogenic andfresh organic matter</em>. The ISME journal, 10(12):2918. &lt;doi: 10.1038/ismej.2016.68&gt;.
</p>

<hr>
<h2 id='test_submodel'>Conduct F test of null hypothesis LB = 0 using output from betta()
or betta_random()</h2><span id='topic+test_submodel'></span>

<h3>Description</h3>

<p>This function performs an F-test of a null hypothesis LB = 0 where B is a
vector of p fixed effects returned by betta() or betta_random() and L is an
m x p matrix with linearly independent rows.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test_submodel(
  fitted_betta,
  submodel_formula,
  method = "bootstrap",
  nboot = 1000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="test_submodel_+3A_fitted_betta">fitted_betta</code></td>
<td>
<p>A fitted betta object &ndash; i.e., the output of either betta() or
betta_random() &ndash; containing fixed effect estimates of interest.</p>
</td></tr>
<tr><td><code id="test_submodel_+3A_submodel_formula">submodel_formula</code></td>
<td>
<p>A formula defining which submodel to treat as the null.
It is not necessary to include random effects in this formula (they will be ignored
if included &ndash; the submodel will be fit with the same random effect structure
as the full model regardless of input.)</p>
</td></tr>
<tr><td><code id="test_submodel_+3A_method">method</code></td>
<td>
<p>A character variable indicating which method should be used to
estimate the distribution of the test statistic under the null.</p>
</td></tr>
<tr><td><code id="test_submodel_+3A_nboot">nboot</code></td>
<td>
<p>Number of bootstrap samples to use if method = &quot;bootstrap&quot;.
Ignored if method = &quot;asymptotic&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing
</p>
<table>
<tr><td><code>pval</code></td>
<td>
<p>The p-value</p>
</td></tr>
<tr><td><code>F_stat</code></td>
<td>
<p>The calculated F statistic</p>
</td></tr>
<tr><td><code>boot_F</code></td>
<td>
<p>A vector of bootstrapped F statistics if bootstrap has been used.
Otherwise NULL.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>David Clausen
</p>


<h3>References</h3>

<p>Willis, A., Bunge, J., and Whitman, T. (2015). Inference for
changes in biodiversity. <em>arXiv preprint.</em>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+betta">betta</a></code>;
</p>
<p><code><a href="#topic+betta_random">betta_random</a></code>;
</p>
<p><code><a href="#topic+betta">betta</a></code>;
</p>
<p><code><a href="#topic+F_test">F_test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# generate example data
df &lt;- data.frame(chats = c(2000, 3000, 4000, 3000,
2000, 3000, 4000, 3000), ses = c(100, 200, 150, 180,
100, 200, 150, 180),
                 Cont_var = c(100, 150, 100, 50,
                 100, 150, 100, 50),
                 Cont_var_2 = c(50,200,25,125,
                 50,200,25,125))

# fit betta()
example_fit &lt;- betta(formula = chats ~ Cont_var + Cont_var_2, ses = ses, data = df)


# construct L for hypothesis that B_cont_var = B_cont_var_2 = 0
L &lt;- rbind(c(0,1,0),
           c(0,0,1))

F_test_results &lt;- F_test(example_fit, L, nboot = 100)

</code></pre>

<hr>
<h2 id='toy_metadata'>(Data) Data frame of covariate information about toy_otu_table.</h2><span id='topic+toy_metadata'></span>

<h3>Description</h3>

<p>(Data) Data frame of covariate information about toy_otu_table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>toy_metadata
</code></pre>


<h3>Format</h3>

<p>A data frame with 143 rows and 4 variables:
</p>

<dl>
<dt>Years</dt><dd><p>Year of sampling</p>
</dd>
<dt>bloom2</dt><dd><p>Did the sample correspond to a bloom event?</p>
</dd>
<dt>Period</dt><dd><p>What season was sampled?</p>
</dd>
<dt>Site</dt><dd><p>Where was the sample taken from?</p>
</dd>
</dl>
<p>...

</p>

<hr>
<h2 id='toy_otu_table'>(Data) A toy OTU table.</h2><span id='topic+toy_otu_table'></span>

<h3>Description</h3>

<p>Covariate info available in 'toy_metadata'. A data frame with 448 rows and 143 columns.
Rows give the abundance of each taxon; columns give the samples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>toy_otu_table
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 448 rows and 143 columns.
</p>

<hr>
<h2 id='toy_taxonomy'>(Data) The taxonomy of the OTUs in 'toy_otu_table'.</h2><span id='topic+toy_taxonomy'></span>

<h3>Description</h3>

<p>(Data) The taxonomy of the OTUs in 'toy_otu_table'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>toy_taxonomy
</code></pre>


<h3>Format</h3>

<p>An object of class <code>factor</code> of length 448.
</p>

<hr>
<h2 id='true_gini'>Calculate the true Gini-Simpson index</h2><span id='topic+true_gini'></span>

<h3>Description</h3>

<p>Calculate the true Gini-Simpson index
</p>


<h3>Usage</h3>

<pre><code class='language-R'>true_gini(input)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="true_gini_+3A_input">input</code></td>
<td>
<p>A vector of proportions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The Gini-Simpson index of the population given by input.
</p>


<h3>Note</h3>

<p>This function is intended for population-level data. If you are
dealing with a microbial sample, use DivNet instead.
</p>

<hr>
<h2 id='true_hill'>Calculate the true Hill numbers</h2><span id='topic+true_hill'></span>

<h3>Description</h3>

<p>Calculate the true Hill numbers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>true_hill(input, q)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="true_hill_+3A_input">input</code></td>
<td>
<p>A vector of proportions.</p>
</td></tr>
<tr><td><code id="true_hill_+3A_q">q</code></td>
<td>
<p>The Hill number of interest. q = 0 corresponds to species richness, q = 2 corresponds to inverse Simpson, etc.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The Hill number of the population given by input.
</p>

<hr>
<h2 id='true_inverse_simpson'>Calculate the true Inverse Simpson index</h2><span id='topic+true_inverse_simpson'></span>

<h3>Description</h3>

<p>Calculate the true Inverse Simpson index
</p>


<h3>Usage</h3>

<pre><code class='language-R'>true_inverse_simpson(input)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="true_inverse_simpson_+3A_input">input</code></td>
<td>
<p>A vector of proportions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The inverse-Simpson index of the population given by input.
</p>


<h3>Note</h3>

<p>This function is intended for population-level data. If you are
dealing with a microbial sample, use DivNet instead.
</p>

<hr>
<h2 id='true_shannon'>Calculate the true Shannon index based on proportions</h2><span id='topic+true_shannon'></span>

<h3>Description</h3>

<p>Calculate the true Shannon index based on proportions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>true_shannon(input)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="true_shannon_+3A_input">input</code></td>
<td>
<p>A vector of proportions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The Shannon index of the population given by input.
</p>


<h3>Note</h3>

<p>This function is intended for population-level data. If you are
dealing with a microbial sample, use DivNet instead.
</p>

<hr>
<h2 id='true_shannon_e'>Calculate the true Shannon's equitability index</h2><span id='topic+true_shannon_e'></span>

<h3>Description</h3>

<p>Calculate the true Shannon's equitability index
</p>


<h3>Usage</h3>

<pre><code class='language-R'>true_shannon_e(input)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="true_shannon_e_+3A_input">input</code></td>
<td>
<p>A vector of proportions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The Shannon E's of the population given by input.
</p>


<h3>Note</h3>

<p>This function is intended for population-level data. If you are
dealing with a microbial sample, use DivNet instead.
</p>

<hr>
<h2 id='true_simpson'>Calculate the true Simpson index</h2><span id='topic+true_simpson'></span>

<h3>Description</h3>

<p>Calculate the true Simpson index
</p>


<h3>Usage</h3>

<pre><code class='language-R'>true_simpson(input)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="true_simpson_+3A_input">input</code></td>
<td>
<p>A vector of proportions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The Simpson index of the population given by input.
</p>


<h3>Note</h3>

<p>This function is intended for population-level data. If you are
dealing with a microbial sample, use DivNet instead.
</p>

<hr>
<h2 id='wlrm_transformed'>The transformed weighted linear regression estimator for species richness estimation</h2><span id='topic+wlrm_transformed'></span>

<h3>Description</h3>

<p>This function implements the transformed version of the species richness
estimation procedure outlined in Rocchetti, Bunge and Bohning (2011).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wlrm_transformed(
  input_data,
  cutoff = NA,
  print = NULL,
  plot = NULL,
  answers = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wlrm_transformed_+3A_input_data">input_data</code></td>
<td>
<p>An input type that can be processed by <code>convert()</code> or a <code>phyloseq</code> object</p>
</td></tr>
<tr><td><code id="wlrm_transformed_+3A_cutoff">cutoff</code></td>
<td>
<p>Maximum frequency count to use</p>
</td></tr>
<tr><td><code id="wlrm_transformed_+3A_print">print</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
<tr><td><code id="wlrm_transformed_+3A_plot">plot</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
<tr><td><code id="wlrm_transformed_+3A_answers">answers</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>alpha_estimate</code>, or <code>alpha_estimates</code> for <code>phyloseq</code> objects
</p>


<h3>Note</h3>

<p>While robust to many different structures, model is almost always
misspecified. The result is usually implausible diversity estimates with
artificially low standard errors. Extreme caution is advised.
</p>


<h3>Author(s)</h3>

<p>Amy Willis
</p>


<h3>References</h3>

<p>Rocchetti, I., Bunge, J. and Bohning, D. (2011). Population size
estimation based upon ratios of recapture probabilities. <em>Annals of
Applied Statistics</em>, <b>5</b>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+breakaway">breakaway</a></code>; <code><a href="#topic+apples">apples</a></code>;
<code><a href="#topic+wlrm_untransformed">wlrm_untransformed</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
wlrm_transformed(apples)
wlrm_transformed(apples, plot = FALSE, print = FALSE, answers = TRUE)

</code></pre>

<hr>
<h2 id='wlrm_untransformed'>The untransformed weighted linear regression estimator for species richness estimation</h2><span id='topic+wlrm_untransformed'></span>

<h3>Description</h3>

<p>This function implements the untransformed version of the species richness
estimation procedure outlined in Rocchetti, Bunge and Bohning (2011).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wlrm_untransformed(
  input_data,
  cutoff = NA,
  print = NULL,
  plot = NULL,
  answers = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wlrm_untransformed_+3A_input_data">input_data</code></td>
<td>
<p>An input type that can be processed by <code>convert()</code> or a <code>phyloseq</code> object</p>
</td></tr>
<tr><td><code id="wlrm_untransformed_+3A_cutoff">cutoff</code></td>
<td>
<p>Maximum frequency count to use</p>
</td></tr>
<tr><td><code id="wlrm_untransformed_+3A_print">print</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
<tr><td><code id="wlrm_untransformed_+3A_plot">plot</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
<tr><td><code id="wlrm_untransformed_+3A_answers">answers</code></td>
<td>
<p>Deprecated; only for backwards compatibility</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>alpha_estimate</code>, or <code>alpha_estimates</code> for <code>phyloseq</code> objects
</p>


<h3>Note</h3>

<p>This estimator is based on the negative binomial model and for that
reason generally produces poor fits to microbial data. The result is usually
artificially low standard errors. Caution is advised.
</p>


<h3>Author(s)</h3>

<p>Amy Willis
</p>


<h3>References</h3>

<p>Rocchetti, I., Bunge, J. and Bohning, D. (2011). Population size
estimation based upon ratios of recapture probabilities. <em>Annals of
Applied Statistics</em>, <b>5</b>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+breakaway">breakaway</a></code>; <code><a href="#topic+apples">apples</a></code>;
<code><a href="#topic+wlrm_transformed">wlrm_transformed</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
wlrm_untransformed(apples)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
