<!DOCTYPE html><html lang="en"><head><title>Help for package breakpoint</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {breakpoint}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#breakpoint-package'><p>Multiple Break-Point Detection via the Cross-Entropy Method</p></a></li>
<li><a href='#CE.NB'><p>Multiple Break-point Detection via the CE Method with Negative Binomial Distribution</p></a></li>
<li><a href='#CE.NB.Init'><p>Multiple Break-point Detection via the CE Method with Negative Binomial Distribution with initial locations</p></a></li>
<li><a href='#CE.Normal.Init.Mean'>
<p>Multiple break-point detection via the CE method for continuous data with initial locations (mean levels)</p></a></li>
<li><a href='#CE.Normal.Init.MeanVar'>
<p>Multiple break-point detection via the CE method for continuous data with initial locations (both mean and variance changes)</p></a></li>
<li><a href='#CE.Normal.Mean'>
<p>Multiple Break-point Detection via the CE Method for Continuous Data (Mean levels)</p></a></li>
<li><a href='#CE.Normal.MeanVar'>
<p>Multiple break-point detection via the CE method for continuous data (both mean and variance changes)</p></a></li>
<li><a href='#CE.ZINB'>
<p>Multiple Break-point Detection via the CE Method with Zero-Inflated Negative Binomial Distribution</p></a></li>
<li><a href='#CE.ZINB.Init'>
<p>Multiple Break-point Detection via the CE Method with Zero-Inflated Negative Binomial Distribution with initial locations</p></a></li>
<li><a href='#ch1.GM03563'>
<p>Fibroblast cell line (GM03563) data</p></a></li>
<li><a href='#profilePlot'>
<p>Mean profile plot</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>An R Package for Multiple Break-Point Detection via the
Cross-Entropy Method</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-01-11</td>
</tr>
<tr>
<td>Author:</td>
<td>Priyadarshana W.J.R.M. and Georgy Sofronov</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Priyadarshana W.J.R.M. &lt;mjayawardana@swin.edu.au&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements the Cross-Entropy (CE) method, which is a model based stochastic optimization technique to estimate both the number and their corresponding locations of break-points in continuous and discrete measurements (Priyadarshana and Sofronov (2015), Priyadarshana and Sofronov (2012a), Priyadarshana and Sofronov (2012b)).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.5)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2 (&ge; 1.0.0), MASS, msm (&ge; 1.0.1), foreach (&ge; 1.2.0),
parallel, doParallel (&ge; 1.0.10)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/madawaweer/breakpoint">https://github.com/madawaweer/breakpoint</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-01-13 10:45:29 UTC; MKMM</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-01-13 14:39:45</td>
</tr>
</table>
<hr>
<h2 id='breakpoint-package'>Multiple Break-Point Detection via the Cross-Entropy Method</h2><span id='topic+breakpoint-package'></span><span id='topic+breakpoint'></span>

<h3>Description</h3>

<p>The breakpoint package implements variants of the Cross-Entropy (CE) method proposed in Priyadarshana and Sofronov (2015, 2012a and 2012b) to estimate both the number and the corresponding locations of break-points in biological sequences of continuous and discrete measurements. The proposed method primarily built to detect multiple break-points in genomic sequences. However, it can be easily extended and applied to other problems.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> breakpoint</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-01-11</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL 2.0</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>&quot;breakpoint&quot;&quot; package provides estimates on both the number as well as the corresponding locations of break-points. The algorithms utilize the Cross-Entropy (CE) method, which is a model-based stochastic optimization procedure to obtain the estimates on locations. Model selection procedures are used to obtain the number of break-points. Current implementation of the methodology works as an exact search method in estimating the number of break-points. However, it supports calculations if the initial locations are provided. A parallel implementation of the procedures can be carried-out in Unix/Linux/MAC OSX and WINDOWS OS with the use of &quot;parallel&quot; and &quot;doParallel&quot; packages. 
</p>


<h3>Author(s)</h3>

<p>Priyadarshana, W.J.R.M. and Sofronov, G.
</p>
<p>Maintainer: Priyadarshana, W.J.R.M. &lt;mjayawardana@swin.edu.au&gt;
</p>


<h3>References</h3>

<p>Priyadarshana, W. J. R. M., Sofronov G. (2015). Multiple Break-Points Detection in Array CGH Data via the Cross-Entropy Method, IEEE/ACM Transactions on Computational Biology and Bioinformatics, 12 (2), pp.487-498.
</p>
<p>Priyadarshana, W. J. R. M. and Sofronov, G. (2012a). A Modified Cross- Entropy Method for Detecting Multiple Change-Points in DNA Count Data. In Proc. of the IEEE Conference on Evolutionary Computation (CEC), 1020-1027, DOI: 10.1109/CEC.2012.6256470.
</p>
<p>Priyadarshana, W. J. R. M. and Sofronov, G. (2012b). The Cross-Entropy Method and Multiple Change-Points Detection in Zero-Inflated DNA read count data. In: Y. T. Gu, S. C. Saha (Eds.) The 4th International Conference on Computational Methods (ICCM2012), 1-8, ISBN 978-1-921897-54-2.
</p>
<p>Rubinstein, R., and Kroese, D. (2004) The Cross-Entropy Method: A Unified Approach to Combinatorial Optimization, Monte-Carlo Simulation and Machine Learning. Springer-Verlag, New York.
</p>
<p>Zhang, N.R., and Siegmund, D.O. (2007) A modified Bayes information criterion with applications to the analysis of comparative genomic hybridization data. Biometrics, 63, 22-32.
</p>

<hr>
<h2 id='CE.NB'>Multiple Break-point Detection via the CE Method with Negative Binomial Distribution</h2><span id='topic+CE.NB'></span>

<h3>Description</h3>

<p>Performs calculations to estimate both the number of break-points and their corresponding locations of discrete measurements with the CE method. Negative binomial distribution is used to model the over-dispersed discrete (count) data. This function supports for the simulation of break-point locations in the CE algorithm based on the four parameter beta distribution or truncated normal distribution. User can select either BIC or AIC to select the optimal number of break-points. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CE.NB(data, Nmax = 10, eps = 0.01, rho = 0.05, M = 200, h = 5, a = 0.8, b = 0.8, 
distyp = 1, penalty = "BIC", parallel = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CE.NB_+3A_data">data</code></td>
<td>

<p>data to be analysed. A single column array or a dataframe.
</p>
</td></tr>
<tr><td><code id="CE.NB_+3A_nmax">Nmax</code></td>
<td>

<p>maximum number of break-points. Default value is 10. 
</p>
</td></tr>
<tr><td><code id="CE.NB_+3A_eps">eps</code></td>
<td>

<p>the cut-off value for the stopping criterion in the CE method. Default value is 0.01.
</p>
</td></tr>
<tr><td><code id="CE.NB_+3A_rho">rho</code></td>
<td>

<p>the fraction which is used to obtain the best performing set of sample solutions (i.e., elite sample). Default value is 0.05.
</p>
</td></tr>
<tr><td><code id="CE.NB_+3A_m">M</code></td>
<td>

<p>sample size to be used in simulating the locations of break-points. Default value is 200.
</p>
</td></tr>
<tr><td><code id="CE.NB_+3A_h">h</code></td>
<td>

<p>minimum aberration width. Default is 5.
</p>
</td></tr>
<tr><td><code id="CE.NB_+3A_a">a</code></td>
<td>

<p>a smoothing parameter value. It is used in the four parameter beta distribution to smooth both shape parameters. When simulating from the truncated normal distribution, this value is used to smooth the estimates of the mean values. Default is 0.8.
</p>
</td></tr>
<tr><td><code id="CE.NB_+3A_b">b</code></td>
<td>

<p>a smoothing parameter value. It is used in the truncated normal distribution to smooth the estimates of the standard deviation. Default is 0.8.
</p>
</td></tr>
<tr><td><code id="CE.NB_+3A_distyp">distyp</code></td>
<td>

<p>distribution to simulate break-point locations. Options: 1 = four parameter beta distribution, 2 = truncated normal distribution. Default is 1. 
</p>
</td></tr>
<tr><td><code id="CE.NB_+3A_penalty">penalty</code></td>
<td>

<p>User can select either BIC or AIC to obtain the number of break-points. Options: &quot;BIC&quot;, &quot;AIC&quot;. Default is &quot;BIC&quot;.
</p>
</td></tr>
<tr><td><code id="CE.NB_+3A_parallel">parallel</code></td>
<td>

<p>A logical argument specifying if parallel computation should be carried-out (TRUE) or not (FALSE). By default it is set as &lsquo;FALSE&rsquo;. In WINDOWS OS systems &quot;snow&quot; functionalities are used, whereas in Unix/Linux/MAC OSX &quot;multicore&quot; functionalities are used to carryout parallel computations with the maximum number of cores available.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The negative binomial (NB) distribution is used to model the discrete (count) data. NB model is preferred over the Poission model when over-dispersion is observed in the count data. A performance function score (BIC or AIC) is calculated for each of the solutions generated by the statistical distribution (four parameter beta distribution or truncated normal distribution), which is used to simulate break-points from no break-point to the user provided maximum number of break-points (default is 10). The solution that minimizes the BIC/AIC with respect to the number of break-points is reported as the optimal solution. Finally, a list containing a vector of break-point locations, number of break-points,  BIC/AIC values and log-likelihood value is returned in the console. 
</p>


<h3>Value</h3>

<p>A list is returned with following items:
</p>
<table role = "presentation">
<tr><td><code>No.BPs</code></td>
<td>
<p>The number of break-points in the data that is estimated by the CE method</p>
</td></tr>
<tr><td><code>BP.Loc</code></td>
<td>
<p>A vector of break-point locations</p>
</td></tr>
<tr><td><code>BIC/AIC</code></td>
<td>
<p>BIC/AIC value</p>
</td></tr>
<tr><td><code>ll</code></td>
<td>
<p>Loglikelihood of the optimal solution</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Priyadarshana, W.J.R.M. &lt;mjayawardana@swin.edu.au&gt;
</p>


<h3>References</h3>

<p>Priyadarshana, W. J. R. M. and Sofronov, G. (2012a) A Modified Cross- Entropy Method for Detecting Multiple Change-Points in DNA Count Data, In Proc. of the IEEE Conference on Evolutionary Computation (CEC), 1020-1027, DOI: 10.1109/CEC.2012.6256470.
</p>
<p>Priyadarshana, W. J. R. M. and Sofronov, G. (2012b) The Cross-Entropy Method and Multiple Change-Points Detection in Zero-Inflated DNA read count data, In: Y. T. Gu, S. C. Saha (Eds.) The 4th International Conference on Computational Methods (ICCM2012), 1-8, ISBN 978-1-921897-54-2.
</p>
<p>Rubinstein, R., and Kroese, D. (2004) The Cross-Entropy Method: A Unified Approach to Combinatorial Optimization, Monte-Carlo Simulation and Machine Learning. Springer-Verlag, New York.
</p>
<p>Schwarz, G. (1978) Estimating the dimension of a model, The Annals of Statistics, 6(2), 461-464.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CE.NB.Init">CE.NB.Init</a></code> for CE with Negative binomial with initial locations,
</p>
<p><code><a href="#topic+CE.ZINB">CE.ZINB</a></code> for CE with zero-inflated negative binomial,
</p>
<p><code><a href="#topic+CE.ZINB.Init">CE.ZINB.Init</a></code> for CE with zero-inflated negative binomial with initial locations,
</p>
<p><code><a href="#topic+profilePlot">profilePlot</a></code> to obtain mean profile plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#### Simulated data example ###
segs &lt;- 6 # Number of segements
M &lt;- c(1500, 2200, 800, 2500, 1000, 2000) # Segment width
#true.locations &lt;- c(1501, 3701, 4501, 7001, 8001)  # True break-point locations 
seg &lt;- NULL 
p &lt;- c(0.45, 0.25, 0.4, 0.2, 0.3, 0.6) # Specification of p's for each segment
for(j in 1:segs){
  seg &lt;- c(seg, rnbinom(M[j], size =10, prob = p[j]))
}
simdata &lt;- as.data.frame(seg)
rm(p, M, seg, segs, j)
#plot(data[, 1])

## Not run: 
## CE with the four parameter beta distribution with BIC as the selection criterion ##

obj1 &lt;- CE.NB(simdata, distyp = 1, penalty = BIC, parallel = TRUE) # Parallel computation
obj1

profilePlot(obj1, simdata) # To obtain the mean profile plot

## CE with truncated normal distribution with BIC as the selection criterion ##

obj2 &lt;- CE.NB(simdata, distyp = 2, penalty = BIC, parallel = TRUE) # Parallel computation
obj2

profilePlot(obj1, simdata) # To obtain the mean profile plot

## End(Not run)
</code></pre>

<hr>
<h2 id='CE.NB.Init'>Multiple Break-point Detection via the CE Method with Negative Binomial Distribution with initial locations</h2><span id='topic+CE.NB.Init'></span>

<h3>Description</h3>

<p>Performs calculations to estimate the break-point locations when their initial values are given. Negative binomial distribution is used to model the over-dispersed discrete (count) data. This function supports for the simulation of break-point locations in the CE algorithm based on the four parameter beta distribution or truncated normal distribution. User can select either BIC or AIC to select the optimal number of break-points. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CE.NB.Init(data, init.locs, eps = 0.01, rho = 0.05, M = 200, h = 5, a = 0.8, b = 0.8,
distyp = 1, penalty = "BIC", var.init = 1e+05, parallel = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CE.NB.Init_+3A_data">data</code></td>
<td>

<p>data to be analysed. A single column array or a dataframe.
</p>
</td></tr>
<tr><td><code id="CE.NB.Init_+3A_init.locs">init.locs</code></td>
<td>

<p>Initial break-point locations.
</p>
</td></tr>
<tr><td><code id="CE.NB.Init_+3A_eps">eps</code></td>
<td>

<p>the cut-off value for the stopping criterion in the CE method. Default value is 0.01.
</p>
</td></tr>
<tr><td><code id="CE.NB.Init_+3A_rho">rho</code></td>
<td>

<p>the fraction which is used to obtain the best performing set of sample solutions (i.e., elite sample). Default value is 0.05.
</p>
</td></tr>
<tr><td><code id="CE.NB.Init_+3A_m">M</code></td>
<td>

<p>sample size to be used in simulating the locations of break-points. Default value is 200.
</p>
</td></tr>
<tr><td><code id="CE.NB.Init_+3A_h">h</code></td>
<td>

<p>minimum aberration width. Default is 5.
</p>
</td></tr>
<tr><td><code id="CE.NB.Init_+3A_a">a</code></td>
<td>

<p>a smoothing parameter value. It is used in the four parameter beta distribution to smooth both shape parameters. When simulating from the truncated normal distribution, this value is used to smooth the estimates of the mean values. Default is 0.8.
</p>
</td></tr>
<tr><td><code id="CE.NB.Init_+3A_b">b</code></td>
<td>

<p>a smoothing parameter value. It is used in the truncated normal distribution to smooth the estimates of the standard deviation. Default is 0.8.
</p>
</td></tr>
<tr><td><code id="CE.NB.Init_+3A_distyp">distyp</code></td>
<td>

<p>distribution to simulate break-point locations. Options: 1 = four parameter beta distribution, 2 = truncated normal distribution. Default is 1. 
</p>
</td></tr>
<tr><td><code id="CE.NB.Init_+3A_penalty">penalty</code></td>
<td>

<p>User can select either BIC or AIC to obtain the number of break-points. Options: &quot;BIC&quot;, &quot;AIC&quot;. Default is &quot;BIC&quot;.
</p>
</td></tr>
<tr><td><code id="CE.NB.Init_+3A_var.init">var.init</code></td>
<td>

<p>Initial variance value to facilitate the search process. Default is 100000.
</p>
</td></tr>
<tr><td><code id="CE.NB.Init_+3A_parallel">parallel</code></td>
<td>

<p>A logical argument specifying if parallel computation should be carried-out (TRUE) or not (FALSE). By default it is set as &lsquo;FALSE&rsquo;. In WINDOWS OS systems &quot;snow&quot; functionalities are used, whereas in Unix/Linux/MAC OSX &quot;multicore&quot; functionalities are used to carryout parallel computations with the maximum number of cores available.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The negative binomial (NB) distribution is used to model the discrete (count) data. NB model is preferred over the Poission model when over-dispersion is observed in the count data. A performance function score (BIC or AIC) is calculated for each of the solutions generated by the statistical distribution (four parameter beta distribution or truncated normal distribution) with respect to the user provided initial locations. Finally, a list containing a vector of break-point locations, number of break-points,  BIC/AIC values and log-likelihood value is returned in the console. 
</p>


<h3>Value</h3>

<p>A list is returned with following items:
</p>
<table role = "presentation">
<tr><td><code>No.BPs</code></td>
<td>
<p>The number of break-points</p>
</td></tr>
<tr><td><code>BP.Loc</code></td>
<td>
<p>A vector of break-point locations</p>
</td></tr>
<tr><td><code>BIC/AIC</code></td>
<td>
<p>BIC/AIC value</p>
</td></tr>
<tr><td><code>ll</code></td>
<td>
<p>Loglikelihood of the optimal solution</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Priyadarshana, W.J.R.M. &lt;mjayawardana@swin.edu.au&gt;
</p>


<h3>References</h3>

<p>Priyadarshana, W. J. R. M. and Sofronov, G. (2012a) A Modified Cross- Entropy Method for Detecting Multiple Change-Points in DNA Count Data, In Proc. of the IEEE Conference on Evolutionary Computation (CEC), 1020-1027, DOI: 10.1109/CEC.2012.6256470.
</p>
<p>Priyadarshana, W. J. R. M. and Sofronov, G. (2012b) The Cross-Entropy Method and Multiple Change-Points Detection in Zero-Inflated DNA read count data, In: Y. T. Gu, S. C. Saha (Eds.) The 4th International Conference on Computational Methods (ICCM2012), 1-8, ISBN 978-1-921897-54-2.
</p>
<p>Rubinstein, R., and Kroese, D. (2004) The Cross-Entropy Method: A Unified Approach to Combinatorial Optimization, Monte-Carlo Simulation and Machine Learning. Springer-Verlag, New York.
</p>
<p>Schwarz, G. (1978) Estimating the dimension of a model, The Annals of Statistics, 6(2), 461-464.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CE.NB">CE.NB</a></code> for CE with Negative binomial,
</p>
<p><code><a href="#topic+CE.ZINB">CE.ZINB</a></code> for CE with zero-inflated negative binomial,
</p>
<p><code><a href="#topic+CE.ZINB.Init">CE.ZINB.Init</a></code> for CE with zero-inflated negative binomial with initial locations,
</p>
<p><code><a href="#topic+profilePlot">profilePlot</a></code> to obtain mean profile plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#### Simulated data example ###
segs &lt;- 6 # Number of segements
M &lt;- c(1500, 2200, 800, 2500, 1000, 2000) # Segment width
#true.locations &lt;- c(1501, 3701, 4501, 7001, 8001)  # True break-point locations 
seg &lt;- NULL 
p &lt;- c(0.45, 0.25, 0.4, 0.2, 0.3, 0.6) # Specification of p's for each segment
for(j in 1:segs){
  seg &lt;- c(seg, rnbinom(M[j], size =10, prob = p[j]))
}
simdata &lt;- as.data.frame(seg)
rm(p, M, seg, segs, j)
#plot(data[, 1])

## Not run: 
## CE with the four parameter beta distribution with BIC as the selection criterion ##

##Specification of initial locations
init.locations &lt;- c(1400, 3400, 4650, 7100, 8200)

obj1 &lt;- CE.NB.Init(simdata, init.locs = init.locations, distyp = 1, penalty = BIC, parallel = TRUE)
obj1

profilePlot(obj1, simdata) # To obtain the mean profile plot

## CE with truncated normal distribution with BIC as the selection criterion ##

obj2 &lt;- CE.NB.Init(simdata, init.locs = init.locations, distyp = 2, penalty = BIC, parallel = TRUE)
obj2

profilePlot(obj1, simdata) # To obtain the mean profile plot

## End(Not run)
</code></pre>

<hr>
<h2 id='CE.Normal.Init.Mean'>
Multiple break-point detection via the CE method for continuous data with initial locations (mean levels)
</h2><span id='topic+CE.Normal.Init.Mean'></span>

<h3>Description</h3>

<p>Performs calculations to estimate the break-point locations when their initial values are given. Normal distribution is used to model the observed continous data. Accross the segments standard deviation is assumed to be the same. This function supports for the simulation of break-point locations based on the four parameter beta distribution or truncated normal distribution. User can select from the modified BIC (mBIC) proposed by Zhang and Siegmund (2007), BIC or AIC to obtain the optimal number of break-points. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CE.Normal.Init.Mean(data, init.locs, eps = 0.01, rho = 0.05, M = 200, h = 5, a = 0.8,
b = 0.8, distyp = 1, penalty = "mBIC", var.init = 1e+05, parallel = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CE.Normal.Init.Mean_+3A_data">data</code></td>
<td>

<p>data to be analysed. A single column array or a dataframe.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.Mean_+3A_init.locs">init.locs</code></td>
<td>

<p>Initial break-point locations.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.Mean_+3A_eps">eps</code></td>
<td>

<p>the cut-off value for the stopping criterion in the CE method. Default value is 0.01.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.Mean_+3A_rho">rho</code></td>
<td>

<p>the fraction which is used to obtain the best performing set of sample solutions (i.e., elite sample). Default value is 0.05.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.Mean_+3A_m">M</code></td>
<td>

<p>sample size to be used in simulating the locations of break-points. Default value is 200.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.Mean_+3A_h">h</code></td>
<td>

<p>minimum aberration width. Default is 5.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.Mean_+3A_a">a</code></td>
<td>

<p>a smoothing parameter value. It is used in the four parameter beta distribution to smooth both shape parameters. When simulating from the truncated normal distribution, this value is used to smooth the estimates of the mean values. Default is 0.8.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.Mean_+3A_b">b</code></td>
<td>

<p>a smoothing parameter value. It is used in the truncated normal distribution to smooth the estimates of the standard deviation. Default is 0.8.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.Mean_+3A_distyp">distyp</code></td>
<td>

<p>distribution to simulate break-point locations. Options: 1 = four parameter beta distribution, 2 = truncated normal distribution. Default is 1. 
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.Mean_+3A_penalty">penalty</code></td>
<td>

<p>User can select from mBIC, BIC or AIC to obtain the optimal number of break-points. Options: &quot;mBIC&quot;, &quot;BIC&quot; and &quot;AIC&quot;. Default is &quot;mBIC&quot;.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.Mean_+3A_var.init">var.init</code></td>
<td>

<p>Initial variance value to facilitate the search process. Default is 100000.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.Mean_+3A_parallel">parallel</code></td>
<td>

<p>A logical argument specifying if parallel computation should be carried-out (TRUE) or not (FALSE). By default it is set as &lsquo;FALSE&rsquo;. In WINDOWS OS systems &quot;snow&quot; functionalities are used, whereas in Unix/Linux/MAC OSX &quot;multicore&quot; functionalities are used to carryout parallel computations with the maximum number of cores available.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The normal distribution is used to model the continuous data. A performance function score (mBIC/BIC/AIC) is calculated for each of the solutions generated by the statistical distribution (four parameter beta distribution or truncated normal distribution), which is used to simulate break-points from the user provided initial locations. The solution that maximizes the selection criteria with respect to the number of break-points is reported as the optimal solution. Finally, a list containing a vector of break-point locations, number of break-points,  mBIC/BIC/AIC values and log-likelihood value is returned in the console.
</p>


<h3>Value</h3>

<p>A list is returned with following items:
</p>
<table role = "presentation">
<tr><td><code>No.BPs</code></td>
<td>
<p>The number of break-points</p>
</td></tr>
<tr><td><code>BP.Loc</code></td>
<td>
<p>A vector of break-point locations</p>
</td></tr>
<tr><td><code>mBIC/BIC/AIC</code></td>
<td>
<p>mBIC/BIC/AIC value</p>
</td></tr>
<tr><td><code>ll</code></td>
<td>
<p>Loglikelihood of the optimal solution</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Priyadarshana, W.J.R.M. &lt;mjayawardana@swin.edu.au&gt;
</p>


<h3>References</h3>

<p>Priyadarshana, W. J. R. M., Sofronov G. (2015). Multiple Break-Points Detection in Array CGH Data via the Cross-Entropy Method, IEEE/ACM Transactions on Computational Biology and Bioinformatics, 12 (2), pp.487-498.
</p>
<p>Priyadarshana, W. J. R. M. and Sofronov, G. (2012) A Modified Cross- Entropy Method for Detecting Multiple Change-Points in DNA Count Data, In Proc. of the IEEE Conference on Evolutionary Computation (CEC), 1020-1027, DOI: 10.1109/CEC.2012.6256470.
</p>
<p>Rubinstein, R., and Kroese, D. (2004) The Cross-Entropy Method: A Unified Approach to Combinatorial Optimization, Monte-Carlo Simulation and Machine Learning. Springer-Verlag, New York.
</p>
<p>Zhang, N.R., and Siegmund, D.O. (2007) A modified Bayes information criterion with applications to the analysis of comparative genomic hybridization data. Biometrics, 63, 22-32.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CE.Normal.Mean">CE.Normal.Mean</a></code> for CE with normal,
</p>
<p><code><a href="#topic+CE.Normal.MeanVar">CE.Normal.MeanVar</a></code> for CE with normal to detect break-points in both mean and variance,
</p>
<p><code><a href="#topic+CE.Normal.Init.MeanVar">CE.Normal.Init.MeanVar</a></code> for CE with normal to detect break-points in both mean and variance with initial locations,
</p>
<p><code><a href="#topic+profilePlot">profilePlot</a></code> to obtain mean profile plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

simdata &lt;- as.data.frame(c(rnorm(200,100,5),rnorm(100,300,5),rnorm(300,150,5)))

## CE with four parameter beta distribution with mBIC as the selection criterion  ##
obj1 &lt;- CE.Normal.Init.Mean(simdata, init.locs = c(150, 380), distyp = 1, parallel =TRUE)
profilePlot(obj1, simdata)

## CE with truncated normal distribution with mBIC as the selection criterion  ##
obj2 &lt;- CE.Normal.Init.Mean(simdata, init.locs = c(150, 380), distyp = 2, parallel =TRUE)
profilePlot(obj2, simdata)

## End(Not run)
</code></pre>

<hr>
<h2 id='CE.Normal.Init.MeanVar'>
Multiple break-point detection via the CE method for continuous data with initial locations (both mean and variance changes)
</h2><span id='topic+CE.Normal.Init.MeanVar'></span>

<h3>Description</h3>

<p>Performs calculations to estimate the break-point locations when their initial values are given. The normal distribution is used to model the observed continous data. Both changes in mean and variance are estimated. This function supports for the simulation of break-point locations based on the four parameter beta distribution or truncated normal distribution. User can select either from the general BIC or AIC to obtain the optimal number of break-points. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CE.Normal.Init.MeanVar(data, init.locs, eps = 0.01, rho = 0.05, M = 200, h = 5, a = 0.8,
b = 0.8, distyp = 1, penalty = "BIC", var.init = 1e+05, parallel = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CE.Normal.Init.MeanVar_+3A_data">data</code></td>
<td>

<p>data to be analysed. A single column array or a dataframe.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.MeanVar_+3A_init.locs">init.locs</code></td>
<td>

<p>Initial break-point locations.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.MeanVar_+3A_eps">eps</code></td>
<td>

<p>the cut-off value for the stopping criterion in the CE method. Default value is 0.01.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.MeanVar_+3A_rho">rho</code></td>
<td>

<p>the fraction which is used to obtain the best performing set of sample solutions (i.e., elite sample). Default value is 0.05.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.MeanVar_+3A_m">M</code></td>
<td>

<p>sample size to be used in simulating the locations of break-points. Default value is 200.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.MeanVar_+3A_h">h</code></td>
<td>

<p>minimum aberration width. Default is 5.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.MeanVar_+3A_a">a</code></td>
<td>

<p>a smoothing parameter value. It is used in the four parameter beta distribution to smooth both shape parameters. When simulating from the truncated normal distribution, this value is used to smooth the estimates of the mean values. Default is 0.8.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.MeanVar_+3A_b">b</code></td>
<td>

<p>a smoothing parameter value. It is used in the truncated normal distribution to smooth the estimates of the standard deviation. Default is 0.8.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.MeanVar_+3A_distyp">distyp</code></td>
<td>

<p>distribution to simulate break-point locations. Options: 1 = four parameter beta distribution, 2 = truncated normal distribution. Default is 1. 
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.MeanVar_+3A_penalty">penalty</code></td>
<td>

<p>User can select either from BIC or AIC to obtain the optimal number of break-points. Options: &quot;BIC&quot; and &quot;AIC&quot;. Default is &quot;BIC&quot;.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.MeanVar_+3A_var.init">var.init</code></td>
<td>

<p>Initial variance value to facilitate the search process. Default is 100000.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Init.MeanVar_+3A_parallel">parallel</code></td>
<td>

<p>A logical argument specifying if parallel computation should be carried-out (TRUE) or not (FALSE). By default it is set as &lsquo;FALSE&rsquo;. In WINDOWS OS systems &quot;snow&quot; functionalities are used, whereas in Unix/Linux/MAC OSX &quot;multicore&quot; functionalities are used to carryout parallel computations with the maximum number of cores available.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The normal distribution is used to model the continuous data. A performance function score (BIC/AIC) is calculated for each of the solutions generated by the statistical distribution (four parameter beta distribution or truncated normal distribution), which is used to simulate break-points from the user provided initial locations. Changes in both mean and variances are estimated. The solution that maximizes the selection criteria with respect to the number of break-points is reported as the optimal solution. Finally, a list containing a vector of break-point locations, number of break-points,  BIC/AIC values and log-likelihood value is returned in the console.
</p>


<h3>Value</h3>

<p>A list is returned with following items:
</p>
<table role = "presentation">
<tr><td><code>No.BPs</code></td>
<td>
<p>The number of break-points</p>
</td></tr>
<tr><td><code>BP.Loc</code></td>
<td>
<p>A vector of break-point locations</p>
</td></tr>
<tr><td><code>BIC/AIC</code></td>
<td>
<p>BIC/AIC value</p>
</td></tr>
<tr><td><code>ll</code></td>
<td>
<p>Loglikelihood of the optimal solution</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Priyadarshana, W.J.R.M. &lt;mjayawardana@swin.edu.au&gt;
</p>


<h3>References</h3>

<p>Priyadarshana, W. J. R. M., Sofronov G. (2015). Multiple Break-Points Detection in Array CGH Data via the Cross-Entropy Method, IEEE/ACM Transactions on Computational Biology and Bioinformatics, 12 (2), pp.487-498.
</p>
<p>Priyadarshana, W. J. R. M. and Sofronov, G. (2012) A Modified Cross- Entropy Method for Detecting Multiple Change-Points in DNA Count Data, In Proc. of the IEEE Conference on Evolutionary Computation (CEC), 1020-1027, DOI: 10.1109/CEC.2012.6256470.
</p>
<p>Rubinstein, R., and Kroese, D. (2004) The Cross-Entropy Method: A Unified Approach to Combinatorial Optimization, Monte-Carlo Simulation and Machine Learning. Springer-Verlag, New York.
</p>
<p>Zhang, N.R., and Siegmund, D.O. (2007) A modified Bayes information criterion with applications to the analysis of comparative genomic hybridization data. Biometrics, 63, 22-32.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CE.Normal.Init.Mean">CE.Normal.Init.Mean</a></code> for CE with normal with initial locations,
</p>
<p><code><a href="#topic+CE.Normal.Mean">CE.Normal.Mean</a></code> for CE with normal to detect break-points in mean levels,
</p>
<p><code><a href="#topic+CE.Normal.MeanVar">CE.Normal.MeanVar</a></code> for CE with normal to detect break-points in both mean and variance,
</p>
<p><code><a href="#topic+profilePlot">profilePlot</a></code> to obtain mean profile plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

simdata &lt;- as.data.frame(c(rnorm(200,100,5),rnorm(1000,160,8),rnorm(300,120,10)))
initial.locs &lt;- c(225, 1300)

## CE with four parameter beta distribution with BIC as the selection criterion  ##
obj1 &lt;- CE.Normal.Init.MeanVar(simdata, init.locs = initial.locs, distyp = 1, parallel =TRUE)
profilePlot(obj1, simdata)

## CE with truncated normal distribution with BIC as the selection criterion  ##
obj2 &lt;- CE.Normal.Init.MeanVar(simdata, init.locs = initial.locs, distyp = 2, parallel =TRUE)
profilePlot(obj2, simdata)

## End(Not run)
</code></pre>

<hr>
<h2 id='CE.Normal.Mean'>
Multiple Break-point Detection via the CE Method for Continuous Data (Mean levels)
</h2><span id='topic+CE.Normal.Mean'></span>

<h3>Description</h3>

<p>This function performs calculations to estimate both the number of break-points and their corresponding locations of continuous measurements with the CE method. The normal distribution is used to model the observed continous data. Accross the segments standard deviation is assumed to be the same. This function supports for the simulation of break-point locations based on the four parameter beta distribution or truncated normal distribution. User can select from the modified BIC (mBIC) proposed by Zhang and Siegmund (2007), BIC or AIC to obtain the optimal number of break-points. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CE.Normal.Mean(data, Nmax = 10, eps = 0.01, rho = 0.05, M = 200, h = 5, a = 0.8, b = 0.8,
distyp = 1, penalty = "mBIC", parallel = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CE.Normal.Mean_+3A_data">data</code></td>
<td>

<p>data to be analysed. A single column array or a dataframe.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Mean_+3A_nmax">Nmax</code></td>
<td>

<p>maximum number of break-points. Default value is 10. 
</p>
</td></tr>
<tr><td><code id="CE.Normal.Mean_+3A_eps">eps</code></td>
<td>

<p>the cut-off value for the stopping criterion in the CE method. Default value is 0.01.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Mean_+3A_rho">rho</code></td>
<td>

<p>the fraction which is used to obtain the best performing set of sample solutions (i.e., elite sample). Default value is 0.05.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Mean_+3A_m">M</code></td>
<td>

<p>sample size to be used in simulating the locations of break-points. Default value is 200.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Mean_+3A_h">h</code></td>
<td>

<p>minimum aberration width. Default is 5.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Mean_+3A_a">a</code></td>
<td>

<p>a smoothing parameter value. It is used in the four parameter beta distribution to smooth both shape parameters. When simulating from the truncated normal distribution, this value is used to smooth the estimates of the mean values. Default is 0.8.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Mean_+3A_b">b</code></td>
<td>

<p>a smoothing parameter value. It is used in the truncated normal distribution to smooth the estimates of the standard deviation. Default is 0.8.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Mean_+3A_distyp">distyp</code></td>
<td>

<p>distributions to simulate break-point locations. Options: 1 = four parameter beta distribution, 2 = truncated normal distribution. Default is 1. 
</p>
</td></tr>
<tr><td><code id="CE.Normal.Mean_+3A_penalty">penalty</code></td>
<td>

<p>User can select from mBIC, BIC or AIC to obtain the optimal number of break-points. Options: &quot;mBIC&quot;, &quot;BIC&quot; and &quot;AIC&quot;. Default is &quot;mBIC&quot;.
</p>
</td></tr>
<tr><td><code id="CE.Normal.Mean_+3A_parallel">parallel</code></td>
<td>

<p>A logical argument specifying if parallel computation should be carried-out (TRUE) or not (FALSE). By default it is set as &lsquo;FALSE&rsquo;. In WINDOWS OS systems &quot;snow&quot; functionalities are used, whereas in Unix/Linux/MAC OSX &quot;multicore&quot; functionalities are used to carryout parallel computations with the maximum number of cores available.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The normal distribution is used to model the continuous data. A performance function score (mBIC/BIC/AIC) is calculated for each of the solutions generated by the statistical distribution (four parameter beta distribution or truncated normal distribution), which is used to simulate break-points from no break-point to the user provided maximum number of break-points. The solution that maximizes the selection criteria with respect to the number of break-points is reported as the optimal solution. Finally, a list containing a vector of break-point locations, number of break-points,  mBIC/BIC/AIC values and log-likelihood value is returned in the console.
</p>


<h3>Value</h3>

<p>A list is returned with following items:
</p>
<table role = "presentation">
<tr><td><code>No.BPs</code></td>
<td>
<p>The number of break-points</p>
</td></tr>
<tr><td><code>BP.Loc</code></td>
<td>
<p>A vector of break-point locations</p>
</td></tr>
<tr><td><code>mBIC/BIC/AIC</code></td>
<td>
<p>mBIC/BIC/AIC value</p>
</td></tr>
<tr><td><code>ll</code></td>
<td>
<p>Loglikelihood of the optimal solution</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Priyadarshana, W.J.R.M. &lt;mjayawardana@swin.edu.au&gt;
</p>


<h3>References</h3>

<p>Priyadarshana, W. J. R. M., Sofronov G. (2015). Multiple Break-Points Detection in Array CGH Data via the Cross-Entropy Method, IEEE/ACM Transactions on Computational Biology and Bioinformatics, 12 (2), pp.487-498.
</p>
<p>Priyadarshana, W. J. R. M. and Sofronov, G. (2012) A Modified Cross- Entropy Method for Detecting Multiple Change-Points in DNA Count Data, In Proc. of the IEEE Conference on Evolutionary Computation (CEC), 1020-1027, DOI: 10.1109/CEC.2012.6256470.
</p>
<p>Rubinstein, R., and Kroese, D. (2004) The Cross-Entropy Method: A Unified Approach to Combinatorial Optimization, Monte-Carlo Simulation and Machine Learning. Springer-Verlag, New York.
</p>
<p>Zhang, N.R., and Siegmund, D.O. (2007) A modified Bayes information criterion with applications to the analysis of comparative genomic hybridization data. Biometrics, 63, 22-32.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CE.Normal.Init.Mean">CE.Normal.Init.Mean</a></code> for CE with normal with initial locations,
</p>
<p><code><a href="#topic+CE.Normal.MeanVar">CE.Normal.MeanVar</a></code> for CE with normal to detect break-points in both mean and variance,
</p>
<p><code><a href="#topic+CE.Normal.Init.MeanVar">CE.Normal.Init.MeanVar</a></code> for CE with normal to detect break-points in both mean and variance with initial locations,
</p>
<p><code><a href="#topic+profilePlot">profilePlot</a></code> to obtain mean profile plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ch1.GM03563)
## Not run: 
## CE with four parameter beta distribution with mBIC as the selection criterion  ##
obj1 &lt;- CE.Normal.Mean(ch1.GM03563, distyp = 1, penalty = "mBIC", parallel =TRUE)
profilePlot(obj1, simdata)

## CE with truncated normal distribution with mBIC as the selection criterion  ##
obj2 &lt;- CE.Normal.Mean(ch1.GM03563, distyp = 2, penalty = "mBIC", parallel =TRUE)
profilePlot(obj2, simdata)

## End(Not run)
</code></pre>

<hr>
<h2 id='CE.Normal.MeanVar'>
Multiple break-point detection via the CE method for continuous data (both mean and variance changes)
</h2><span id='topic+CE.Normal.MeanVar'></span>

<h3>Description</h3>

<p>This function performs calculations to estimate both the number of break-points and their corresponding locations of continuous measurements with the CE method. The normal distribution is used to model the observed continous data. This function supports for the simulation of break-point locations based on the four parameter beta distribution or truncated normal distribution. User can select either from the genral BIC or AIC to obtain the optimal number of break-points. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CE.Normal.MeanVar(data, Nmax = 10, eps = 0.01, rho = 0.05, M = 200, h = 5, a = 0.8,
b = 0.8, distyp = 1, penalty = "BIC", parallel = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CE.Normal.MeanVar_+3A_data">data</code></td>
<td>

<p>data to be analysed. A single column array or a dataframe.
</p>
</td></tr>
<tr><td><code id="CE.Normal.MeanVar_+3A_nmax">Nmax</code></td>
<td>

<p>maximum number of break-points. Default value is 10. 
</p>
</td></tr>
<tr><td><code id="CE.Normal.MeanVar_+3A_eps">eps</code></td>
<td>

<p>the cut-off value for the stopping criterion in the CE method. Default value is 0.01.
</p>
</td></tr>
<tr><td><code id="CE.Normal.MeanVar_+3A_rho">rho</code></td>
<td>

<p>the fraction which is used to obtain the best performing set of sample solutions (i.e., elite sample). Default value is 0.05.
</p>
</td></tr>
<tr><td><code id="CE.Normal.MeanVar_+3A_m">M</code></td>
<td>

<p>sample size to be used in simulating the locations of break-points. Default value is 200.
</p>
</td></tr>
<tr><td><code id="CE.Normal.MeanVar_+3A_h">h</code></td>
<td>

<p>minimum aberration width. Default is 5.
</p>
</td></tr>
<tr><td><code id="CE.Normal.MeanVar_+3A_a">a</code></td>
<td>

<p>a smoothing parameter value. It is used in the four parameter beta distribution to smooth both shape parameters. When simulating from the truncated normal distribution, this value is used to smooth the estimates of the mean values. Default is 0.8.
</p>
</td></tr>
<tr><td><code id="CE.Normal.MeanVar_+3A_b">b</code></td>
<td>

<p>a smoothing parameter value. It is used in the truncated normal distribution to smooth the estimates of the standard deviation. Default is 0.8.
</p>
</td></tr>
<tr><td><code id="CE.Normal.MeanVar_+3A_distyp">distyp</code></td>
<td>

<p>distributions to simulate break-point locations. Options: 1 = four parameter beta distribution, 2 = truncated normal distribution. Default is 1. 
</p>
</td></tr>
<tr><td><code id="CE.Normal.MeanVar_+3A_penalty">penalty</code></td>
<td>

<p>User can select from BIC or AIC to obtain the optimal number of break-points. Options: &quot;BIC&quot; and &quot;AIC&quot;. Default is &quot;BIC&quot;.
</p>
</td></tr>
<tr><td><code id="CE.Normal.MeanVar_+3A_parallel">parallel</code></td>
<td>

<p>A logical argument specifying if parallel computation should be carried-out (TRUE) or not (FALSE). By default it is set as &lsquo;FALSE&rsquo;. In WINDOWS OS systems &quot;snow&quot; functionalities are used, whereas in Unix/Linux/MAC OSX &quot;multicore&quot; functionalities are used to carryout parallel computations with the maximum number of cores available.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The normal distribution is used to model the continuous data. A performance function score (BIC/AIC) is calculated for each of the solutions generated by the statistical distribution (four parameter beta distribution or truncated normal distribution), which is used to simulate break-points from no break-point to the user provided maximum number of break-points. Changes in both mean and variance are estimated. The solution that maximizes the selection criteria with respect to the number of break-points is reported as the optimal solution. Finally, a list containing a vector of break-point locations, number of break-points,  BIC/AIC values and log-likelihood value is returned in the console.
</p>


<h3>Value</h3>

<p>A list is returned with following items:
</p>
<table role = "presentation">
<tr><td><code>No.BPs</code></td>
<td>
<p>The number of break-points</p>
</td></tr>
<tr><td><code>BP.Loc</code></td>
<td>
<p>A vector of break-point locations</p>
</td></tr>
<tr><td><code>BIC/AIC</code></td>
<td>
<p>BIC/AIC value</p>
</td></tr>
<tr><td><code>ll</code></td>
<td>
<p>Loglikelihood of the optimal solution</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Priyadarshana, W.J.R.M. &lt;mjayawardana@swin.edu.au&gt;
</p>


<h3>References</h3>

<p>Priyadarshana, W. J. R. M., Sofronov G. (2015). Multiple Break-Points Detection in Array CGH Data via the Cross-Entropy Method, IEEE/ACM Transactions on Computational Biology and Bioinformatics, 12 (2), pp.487-498.
</p>
<p>Priyadarshana, W. J. R. M. and Sofronov, G. (2012) A Modified Cross- Entropy Method for Detecting Multiple Change-Points in DNA Count Data, In Proc. of the IEEE Conference on Evolutionary Computation (CEC), 1020-1027, DOI: 10.1109/CEC.2012.6256470.
</p>
<p>Rubinstein, R., and Kroese, D. (2004) The Cross-Entropy Method: A Unified Approach to Combinatorial Optimization, Monte-Carlo Simulation and Machine Learning. Springer-Verlag, New York.
</p>
<p>Zhang, N.R., and Siegmund, D.O. (2007) A modified Bayes information criterion with applications to the analysis of comparative genomic hybridization data. Biometrics, 63, 22-32.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CE.Normal.Init.Mean">CE.Normal.Init.Mean</a></code> for CE with normal with initial locations,
</p>
<p><code><a href="#topic+CE.Normal.Mean">CE.Normal.Mean</a></code> for CE with normal to detect break-points in mean levels,
</p>
<p><code><a href="#topic+CE.Normal.Init.MeanVar">CE.Normal.Init.MeanVar</a></code> for CE with normal to detect break-points in both mean and variance with initial locations,
</p>
<p><code><a href="#topic+profilePlot">profilePlot</a></code> to obtain mean profile plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

simdata &lt;- as.data.frame(c(rnorm(200,100,5),rnorm(1000,160,8),rnorm(300,120,10)))

## CE with four parameter beta distribution with BIC as the selection criterion  ##
obj1 &lt;- CE.Normal.MeanVar(simdata, distyp = 1, penalty = "BIC", parallel =TRUE)
profilePlot(obj1, simdata)

## CE with truncated normal distribution with BIC as the selection criterion  ##
obj2 &lt;- CE.Normal.MeanVar(simdata, distyp = 2, penalty = "BIC", parallel =TRUE)
profilePlot(obj2, simdata)

## End(Not run)
</code></pre>

<hr>
<h2 id='CE.ZINB'>
Multiple Break-point Detection via the CE Method with Zero-Inflated Negative Binomial Distribution</h2><span id='topic+CE.ZINB'></span>

<h3>Description</h3>

<p>Performs calculations to estimate both the number of break-points and their corresponding locations of discrete measurements with the CE method. Zero-inflated negative binomial distribution is used to model the excess zero observations and to model over-dispersesion in the oberved discrete (count) data. This function supports for the simulation of break-point locations in the CE algorithm based on the four parameter beta distribution and truncated normal distribution. The general BIC or AIC can be used to select the optimal number of break-points. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CE.ZINB(data, Nmax = 10, eps = 0.01, rho = 0.05, M = 200, h = 5, a = 0.8, b = 0.8,
distyp = 1, penalty = "BIC", parallel = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CE.ZINB_+3A_data">data</code></td>
<td>

<p>data to be analysed. A single column array or a dataframe.
</p>
</td></tr>
<tr><td><code id="CE.ZINB_+3A_nmax">Nmax</code></td>
<td>

<p>maximum number of break-points. Default value is 10. 
</p>
</td></tr>
<tr><td><code id="CE.ZINB_+3A_eps">eps</code></td>
<td>

<p>the cut-off value for the stopping criterion in the CE method. Default value is 0.01.
</p>
</td></tr>
<tr><td><code id="CE.ZINB_+3A_rho">rho</code></td>
<td>

<p>the fraction which is used to obtain the best performing set of sample solutions (i.e., elite sample). Default value is 0.05.
</p>
</td></tr>
<tr><td><code id="CE.ZINB_+3A_m">M</code></td>
<td>

<p>sample size to be used in simulating the locations of break-points. Default value is 200.
</p>
</td></tr>
<tr><td><code id="CE.ZINB_+3A_h">h</code></td>
<td>

<p>minimum aberration width. Default is 5.
</p>
</td></tr>
<tr><td><code id="CE.ZINB_+3A_a">a</code></td>
<td>

<p>a smoothing parameter value. It is used in the four parameter beta distribution to smooth both shape parameters. When simulating from the truncated normal distribution, this value is used to smooth the estimates of the mean values. Default is 0.8.
</p>
</td></tr>
<tr><td><code id="CE.ZINB_+3A_b">b</code></td>
<td>

<p>a smoothing parameter value. It is used in the truncated normal distribution to smooth the estimates of the standard deviation. Default is 0.8.
</p>
</td></tr>
<tr><td><code id="CE.ZINB_+3A_distyp">distyp</code></td>
<td>

<p>distribution to simulate break-point locations. Options: 1 = four parameter beta distribution, 2 = truncated normal distribution. Default is 1. 
</p>
</td></tr>
<tr><td><code id="CE.ZINB_+3A_penalty">penalty</code></td>
<td>

<p>User can select either BIC or AIC to obtain the number of break-points. Options: &quot;BIC&quot;, &quot;AIC&quot;. Default is &quot;BIC&quot;.
</p>
</td></tr>
<tr><td><code id="CE.ZINB_+3A_parallel">parallel</code></td>
<td>

<p>A logical argument specifying if parallel computation should be carried-out (TRUE) or not (FALSE). By default it is set as &lsquo;FALSE&rsquo;. In WINDOWS OS systems &quot;snow&quot; functionalities are used, whereas in Unix/Linux/MAC OSX &quot;multicore&quot; functionalities are used to carryout parallel computations with the maximum number of cores available.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Zero-inflated negative binomial (ZINB) distribution is used to model the discrete (count) data. ZINB model is preferred over the NB model when both excess zero values and over-dispersion observed in the count data. A performance function score (BIC) is calculated for each of the solutions generated by the statistical distribution (four parameter beta distribution or truncated normal distribution), which is used to simulate break-points from no break-point to the user provided maximum number of break-points. The solution that minimizes the BIC/AIC with respect to the number of break-points is reported as the optimal solution. Finally, a list containing a vector of break-point, BIC/AIC values and log-likelihood value is returned in the console. 
</p>


<h3>Value</h3>

<p>A list is returned with following items:
</p>
<table role = "presentation">
<tr><td><code>No.BPs</code></td>
<td>
<p>The number of break-points</p>
</td></tr>
<tr><td><code>BP.Loc</code></td>
<td>
<p>A vector of break-point locations</p>
</td></tr>
<tr><td><code>BIC/AIC</code></td>
<td>
<p>BIC/AIC value</p>
</td></tr>
<tr><td><code>ll</code></td>
<td>
<p>Loglikelihood of the optimal solution</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Priyadarshana, W.J.R.M. &lt;mjayawardana@swin.edu.au&gt;
</p>


<h3>References</h3>

<p>Priyadarshana, W. J. R. M. and Sofronov, G. (2012a) A Modified Cross- Entropy Method for Detecting Multiple Change-Points in DNA Count Data, In Proc. of the IEEE Conference on Evolutionary Computation (CEC), 1020-1027, DOI: 10.1109/CEC.2012.6256470.
</p>
<p>Priyadarshana, W. J. R. M. and Sofronov, G. (2012b) The Cross-Entropy Method and Multiple Change-Points Detection in Zero-Inflated DNA read count data, In: Y. T. Gu, S. C. Saha (Eds.) The 4th International Conference on Computational Methods (ICCM2012), 1-8, ISBN 978-1-921897-54-2.
</p>
<p>Rubinstein, R., and Kroese, D. (2004) The Cross-Entropy Method: A Unified Approach to Combinatorial Optimization, Monte-Carlo Simulation and Machine Learning. Springer-Verlag, New York.
</p>
<p>Schwarz, G. (1978) Estimating the dimension of a model, The Annals of Statistics, 6(2), 461-464.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CE.NB">CE.NB</a></code> for CE with negative binomial,
</p>
<p><code><a href="#topic+CE.NB.Init">CE.NB.Init</a></code> for CE with negative binomial with initial locations,
</p>
<p><code><a href="#topic+CE.ZINB.Init">CE.ZINB.Init</a></code> for CE with zero-inflated negative binomial with initial locations,
</p>
<p><code><a href="#topic+profilePlot">profilePlot</a></code> to obtain mean profile plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#### Simulated data example ###
# gamlss R package is used to simulate data from the ZINB.

## Not run: 
library(gamlss)
segs &lt;- 6 # Number of segements
M &lt;- c(1500, 2200, 800, 2500, 1000, 2000) # Segment width
#true.locations &lt;- c(1501, 3701, 4501, 7001, 8001)  # True break-point locations 
seg &lt;- NULL
p &lt;- c(0.6, 0.1, 0.3, 0.05, 0.2, 0.4) # Specification of p's on each segment'
sigma.val &lt;- c(1,2,3,4,5,6) # Specification of sigma vlaues 

for(j in 1:segs){
  seg &lt;- c(seg, rZINBI(M[j], mu = 300, sigma = sigma.val[j], nu = p[j]))
}

simdata &lt;- as.data.frame(seg)
rm(p, M, seg, segs, j, sigma.val)
#plot(data[, 1])

## CE with the four parameter beta distribution with BIC as the selection criterion  ##

obj1 &lt;- CE.ZINB(simdata, distyp = 1, penalty = BIC, parallel = TRUE) # Parallel computation
obj1

profilePlot(obj1, simdata) # To obtain the mean profile plot

## CE with truncated normal distribution with BIC as the selection criterion  ##

obj2 &lt;- CE.ZINB(simdata, distyp = 2, penalty = BIC, parallel = TRUE) # Parallel computation
obj2

profilePlot(obj2, simdata) # To obtain the mean profile plot

## End(Not run)
</code></pre>

<hr>
<h2 id='CE.ZINB.Init'>
Multiple Break-point Detection via the CE Method with Zero-Inflated Negative Binomial Distribution with initial locations
</h2><span id='topic+CE.ZINB.Init'></span>

<h3>Description</h3>

<p>Performs calculations to estimate the break-point locations when their initial values are given. Zero-inflated negative binomial distribution is used to model the excess zero observations and to model over-dispersesion in the oberved discrete (count) data. This function supports for the simulation of break-point locations in the CE algorithm based on the four parameter beta distribution and truncated normal distribution. The general BIC or AIC can be used to select the optimal number of break-points. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CE.ZINB.Init(data, init.locs, eps = 0.01, rho = 0.05, M = 200, h = 5, a = 0.8, b = 0.8,
distyp = 1, penalty = "BIC", var.init = 1e+05, parallel = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CE.ZINB.Init_+3A_data">data</code></td>
<td>

<p>data to be analysed. A single column array or a dataframe.
</p>
</td></tr>
<tr><td><code id="CE.ZINB.Init_+3A_init.locs">init.locs</code></td>
<td>

<p>Initial break-point locations.
</p>
</td></tr>
<tr><td><code id="CE.ZINB.Init_+3A_eps">eps</code></td>
<td>

<p>the cut-off value for the stopping criterion in the CE method. Default value is 0.01.
</p>
</td></tr>
<tr><td><code id="CE.ZINB.Init_+3A_rho">rho</code></td>
<td>

<p>the fraction which is used to obtain the best performing set of sample solutions (i.e., elite sample). Default value is 0.05.
</p>
</td></tr>
<tr><td><code id="CE.ZINB.Init_+3A_m">M</code></td>
<td>

<p>sample size to be used in simulating the locations of break-points. Default value is 200.
</p>
</td></tr>
<tr><td><code id="CE.ZINB.Init_+3A_h">h</code></td>
<td>

<p>minimum aberration width. Default is 5.
</p>
</td></tr>
<tr><td><code id="CE.ZINB.Init_+3A_a">a</code></td>
<td>

<p>a smoothing parameter value. It is used in the four parameter beta distribution to smooth both shape parameters. When simulating from the truncated normal distribution, this value is used to smooth the estimates of the mean values. Default is 0.8.
</p>
</td></tr>
<tr><td><code id="CE.ZINB.Init_+3A_b">b</code></td>
<td>

<p>a smoothing parameter value. It is used in the truncated normal distribution to smooth the estimates of the standard deviation. Default is 0.8.
</p>
</td></tr>
<tr><td><code id="CE.ZINB.Init_+3A_distyp">distyp</code></td>
<td>

<p>distribution to simulate break-point locations. Options: 1 = four parameter beta distribution, 2 = truncated normal distribution. Default is 1. 
</p>
</td></tr>
<tr><td><code id="CE.ZINB.Init_+3A_penalty">penalty</code></td>
<td>

<p>User can select either BIC or AIC to obtain the number of break-points. Options: &quot;BIC&quot;, &quot;AIC&quot;. Default is &quot;BIC&quot;.
</p>
</td></tr>
<tr><td><code id="CE.ZINB.Init_+3A_var.init">var.init</code></td>
<td>

<p>Initial variance value to facilitate the search process. Default is 100000.
</p>
</td></tr>
<tr><td><code id="CE.ZINB.Init_+3A_parallel">parallel</code></td>
<td>

<p>A logical argument specifying if parallel computation should be carried-out (TRUE) or not (FALSE). By default it is set as &lsquo;FALSE&rsquo;. In WINDOWS OS systems &quot;snow&quot; functionalities are used, whereas in Unix/Linux/MAC OSX &quot;multicore&quot; functionalities are used to carryout parallel computations with the maximum number of cores available.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Zero-inflated negative binomial (ZINB) distribution is used to model the discrete (count) data. ZINB model is preferred over the NB model when both excess zero values and over-dispersion observed in the count data. A performance function score (BIC) is calculated for each of the solutions generated by the statistical distribution (four parameter beta distribution or truncated normal distribution), which is used to simulate break-points when the initial locations are provided. Finally, a list containing a vector of break-point locations, number of break-points,  BIC/AIC values and log-likelihood value is returned in the console. 
</p>


<h3>Value</h3>

<p>A list is returned with following items:
</p>
<table role = "presentation">
<tr><td><code>No.BPs</code></td>
<td>
<p>The number of break-points</p>
</td></tr>
<tr><td><code>BP.Loc</code></td>
<td>
<p>A vector of break-point locations</p>
</td></tr>
<tr><td><code>BIC/AIC</code></td>
<td>
<p>BIC/AIC value</p>
</td></tr>
<tr><td><code>ll</code></td>
<td>
<p>Loglikelihood of the optimal solution</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Priyadarshana, W.J.R.M. &lt;mjayawardana@swin.edu.au&gt;
</p>


<h3>References</h3>

<p>Priyadarshana, W. J. R. M. and Sofronov, G. (2012a) A Modified Cross- Entropy Method for Detecting Multiple Change-Points in DNA Count Data, In Proc. of the IEEE Conference on Evolutionary Computation (CEC), 1020-1027, DOI: 10.1109/CEC.2012.6256470.
</p>
<p>Priyadarshana, W. J. R. M. and Sofronov, G. (2012b) The Cross-Entropy Method and Multiple Change-Points Detection in Zero-Inflated DNA read count data, In: Y. T. Gu, S. C. Saha (Eds.) The 4th International Conference on Computational Methods (ICCM2012), 1-8, ISBN 978-1-921897-54-2.
</p>
<p>Rubinstein, R., and Kroese, D. (2004) The Cross-Entropy Method: A Unified Approach to Combinatorial Optimization, Monte-Carlo Simulation and Machine Learning. Springer-Verlag, New York.
</p>
<p>Schwarz, G. (1978) Estimating the dimension of a model, The Annals of Statistics, 6(2), 461-464.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CE.NB">CE.NB</a></code> for CE with negative binomial,
</p>
<p><code><a href="#topic+CE.NB.Init">CE.NB.Init</a></code> for CE with negative binomial with initial locations,
</p>
<p><code><a href="#topic+CE.ZINB">CE.ZINB</a></code> for CE with zero-inflated negative binomial,
</p>
<p><code><a href="#topic+profilePlot">profilePlot</a></code> to obtain mean profile plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#### Simulated data example ###
# gamlss R package is used to simulate data from the ZINB.

## Not run: 
library(gamlss)
segs &lt;- 6 # Number of segements
M &lt;- c(1500, 2200, 800, 2500, 1000, 2000) # Segment width
#true.locations &lt;- c(1501, 3701, 4501, 7001, 8001)  # True break-point locations 
seg &lt;- NULL
p &lt;- c(0.6, 0.1, 0.3, 0.05, 0.2, 0.4) # Specification of p's on each segment'
sigma.val &lt;- c(1,2,3,4,5,6) # Specification of sigma vlaues 

for(j in 1:segs){
  seg &lt;- c(seg, rZINBI(M[j], mu = 300, sigma = sigma.val[j], nu = p[j]))
}

simdata &lt;- as.data.frame(seg)
rm(p, M, seg, segs, j, sigma.val)
#plot(data[, 1])

## CE with the four parameter beta distribution with BIC as the selection criterion  ##

init.loci &lt;- c(1400, 3400, 4650, 7100, 8200)

obj1 &lt;- CE.ZINB.Init(simdata, init.locs = init.loci, distyp = 1, penalty = BIC, parallel = TRUE)
obj1

profilePlot(obj1, simdata) # To obtain the mean profile plot

## CE with truncated normal distribution with BIC as the selection criterion  ##

obj2 &lt;- CE.ZINB.Init(simdata, init.locs = init.loci, distyp = 2, penalty = BIC, parallel = TRUE)
obj2

profilePlot(obj2, simdata) # To obtain the mean profile plot

## End(Not run)
</code></pre>

<hr>
<h2 id='ch1.GM03563'>
Fibroblast cell line (GM03563) data
</h2><span id='topic+ch1.GM03563'></span>

<h3>Description</h3>

<p>Chromosome 1 of cell line GM03563
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("ch1.GM03563")</code></pre>


<h3>Format</h3>

<p>A single column data frame with 135 observations corresponds to chromosome 1 of cell line GM03563.
</p>

<dl>
<dt><code>log2ratio</code></dt><dd><p>normalized average of the log base 2 test over reference ratio data</p>
</dd>
</dl>



<h3>Details</h3>

<p>This data set is extracted from a single experiments on 15 fibroblast cell lines with each array containing over 2000 (mapped) BACs spotted in triplicate discussed in Snijders et al.(2001). Data corresponds to the chromosome 1 of cell line GM03563.
</p>


<h3>References</h3>

<p>Snijders,A.M. et al. (2001) Assembly of microarrays for genome-wide measurement of DNA copy number. Nature Genetics, 29, 263-26.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ch1.GM03563)
## Not run: 
## CE with four parameter beta distribution ##
obj1 &lt;- CE.Normal.Mean(ch1.GM03563, distyp = 1, parallel =TRUE)
profilePlot(obj1, ch1.GM03563)

## CE with truncated normal distribution ##
obj2 &lt;- CE.Normal.Mean(ch1.GM03563, distyp = 2, parallel =TRUE)
profilePlot(obj2, ch1.GM03563)

## End(Not run)</code></pre>

<hr>
<h2 id='profilePlot'>
Mean profile plot</h2><span id='topic+profilePlot'></span>

<h3>Description</h3>

<p>Plotting function to obtain mean profile plot of the testing dataset based on the estimates of the break-points. An R object created from the CE.Normal, CE.NB ot CE.ZINB is required. User can alter the axis names. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>profilePlot(obj, data, x.label = "Data Sequence", y.label = "Value")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="profilePlot_+3A_obj">obj</code></td>
<td>

<p>R object created from CE.Normal, CE.NB or CE.ZINB.
</p>
</td></tr>
<tr><td><code id="profilePlot_+3A_data">data</code></td>
<td>

<p>data to be analysed. A single column array or a dataframe.
</p>
</td></tr>
<tr><td><code id="profilePlot_+3A_x.label">x.label</code></td>
<td>

<p>x axis label. Default is &quot;Data Sequence&quot;.
</p>
</td></tr>
<tr><td><code id="profilePlot_+3A_y.label">y.label</code></td>
<td>

<p>y axis label. Default is &quot;Value&quot;.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Priyadarshana, W.J.R.M. &lt;mjayawardana@swin.edu.au&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CE.Normal.Mean">CE.Normal.Mean</a></code>, 
</p>
<p><code><a href="#topic+CE.NB">CE.NB</a></code>, 
</p>
<p><code><a href="#topic+CE.ZINB">CE.ZINB</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ch1.GM03563)
## Not run: 
## CE with four parameter beta distribution ##
obj1 &lt;- CE.Normal.Mean(ch1.GM03563, distyp = 1, penalty = "mBIC", parallel =TRUE)
profilePlot(obj1)

## CE with truncated normal distribution ##
obj2 &lt;- CE.Normal.Mean(ch1.GM03563, distyp = 2, penalty = "mBIC", parallel =TRUE)
profilePlot(obj2)

## End(Not run)</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
