<!DOCTYPE html><html lang="en"><head><title>Help for package BRETIGEA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {BRETIGEA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#aba_marker_expression'><p>Normalized FPKM expression data from a subset of the Allen Brain Atlas Aging, Dementia, and TBI study.</p></a></li>
<li><a href='#aba_pheno_data'><p>Phenotype data from brain samples in the Allen Brain Atlas Aging, Dementia, and TBI study.</p></a></li>
<li><a href='#adjustBrainCells'><p>Estimate and adjust for brain cell type proportions in bulk expression data.</p></a></li>
<li><a href='#adjustCells'><p>Adjust for estimated cell type proportions in bulk gene expression data.</p></a></li>
<li><a href='#brainCells'><p>Estimate brain cell type proportions in bulk expression data with marker genes.</p></a></li>
<li><a href='#BRETIGEA'><p>BRETIGEA: BRain cEll Type specIfic Gene Expression Analysis</p></a></li>
<li><a href='#findCells'><p>Find cell type proportions from bulk gene expression data using marker genes.</p></a></li>
<li><a href='#kelley_df_brain'><p>Marker genes estimated from</p></a></li>
<li><a href='#markers_df_brain'><p>Marker genes estimated from a meta-analysis of brain cell gene expression data from both humans and mice.</p></a></li>
<li><a href='#markers_df_human_brain'><p>Marker genes estimated from a meta-analysis of brain cell gene expression data from humans only.</p></a></li>
<li><a href='#markers_df_mouse_brain'><p>Marker genes estimated from a meta-analysis of brain cell gene expression data from mice only.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Brain Cell Type Specific Gene Expression Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.3</td>
</tr>
<tr>
<td>Author:</td>
<td>Andrew McKenzie [aut, cre],
    Minghui Wang [aut],
    Bin Zhang [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Andrew McKenzie &lt;amckenz@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Analysis of relative cell type proportions in bulk gene expression data. Provides a well-validated set of brain cell type-specific marker genes derived from multiple types of experiments, as described in McKenzie (2018) &lt;<a href="https://doi.org/10.1038%2Fs41598-018-27293-5">doi:10.1038/s41598-018-27293-5</a>&gt;. For brain tissue data sets, there are marker genes available for astrocytes, endothelial cells, microglia, neurons, oligodendrocytes, and oligodendrocyte precursor cells, derived from each of human, mice, and combination human/mouse data sets. However, if you have access to your own marker genes, the functions can be applied to bulk gene expression data from any tissue. Also implements multiple options for relative cell type proportion estimation using these marker genes, adapting and expanding on approaches from the 'CellCODE' R package described in Chikina (2015) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtv015">doi:10.1093/bioinformatics/btv015</a>&gt;. The number of cell type marker genes used in a given analysis can be increased or decreased based on your preferences and the data set. Finally, provides functions to use the estimates to adjust for variability in the relative proportion of cell types across samples prior to downstream analyses.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, stats, utils, knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.0</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-05-04 23:13:00 UTC; mckena01</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-05-05 04:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='aba_marker_expression'>Normalized FPKM expression data from a subset of the Allen Brain Atlas Aging, Dementia, and TBI study.</h2><span id='topic+aba_marker_expression'></span>

<h3>Description</h3>

<p>Filtered data set to only contain brain marker genes as well as ~100 other genes. The total data set can be downloaded by following the links in the original paper.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aba_marker_expression
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 395 rows and 377 columns.
</p>


<h3>References</h3>

<p>© 2016 Allen Institute for Cell Science. Aging, Dementia and TBI. Available from: http://aging.brain-map.org/
</p>

<hr>
<h2 id='aba_pheno_data'>Phenotype data from brain samples in the Allen Brain Atlas Aging, Dementia, and TBI study.</h2><span id='topic+aba_pheno_data'></span>

<h3>Description</h3>

<p>A subset of phenotype data to be used for correlations with the expression data in the tests and vignette, to validate that the estimated cell type proportions correspond to useful quantities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aba_pheno_data
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 377 rows and 4 columns.
</p>


<h3>References</h3>

<p>© 2016 Allen Institute for Cell Science. Aging, Dementia and TBI. Available from: http://aging.brain-map.org/
</p>

<hr>
<h2 id='adjustBrainCells'>Estimate and adjust for brain cell type proportions in bulk expression data.</h2><span id='topic+adjustBrainCells'></span>

<h3>Description</h3>

<p>This function uses a linear model to adjust each row of gene expression for cell types. Other covariates can be included as well.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjustBrainCells(
  inputMat,
  nMarker = 50,
  species = "combined",
  celltypes = c("ast", "end", "mic", "neu", "oli", "opc"),
  addMeans = FALSE,
  formula = NULL,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="adjustBrainCells_+3A_inputmat">inputMat</code></td>
<td>
<p>Input gene expression data, with rows as features (e.g. genes) and samples as columns.</p>
</td></tr>
<tr><td><code id="adjustBrainCells_+3A_nmarker">nMarker</code></td>
<td>
<p>The number of marker genes (that are present in your expression data set) to use in estimating the surrogate cell type proportion variable for each cell type.</p>
</td></tr>
<tr><td><code id="adjustBrainCells_+3A_species">species</code></td>
<td>
<p>By default, this function uses markers from combined human and mouse measurements, which are the most robust and reliable, as the gene expression patterns are very conserved between these two species. Other options are &quot;human&quot; and &quot;mouse&quot; for data specific to those species. Note that OPCs only have 500 gene symbols in this case, and are taken from only the Darmanis et al or Tasic et al data sets, respectively.</p>
</td></tr>
<tr><td><code id="adjustBrainCells_+3A_celltypes">celltypes</code></td>
<td>
<p>Character vector of which cell types to estimate and adjust for.</p>
</td></tr>
<tr><td><code id="adjustBrainCells_+3A_addmeans">addMeans</code></td>
<td>
<p>Whether the mean should be added to the residuals in the resulting adjusted gene expression table.</p>
</td></tr>
<tr><td><code id="adjustBrainCells_+3A_formula">formula</code></td>
<td>
<p>If you want to add additional covariates to be adjusted for, then you can supply them by adding an (optional) formula function here. The format should be &quot;expression_data ~ $cov1 + factor($cov2) +&quot;, where $cov1 is a numeric covariate present in the current environment, and $cov2 is a factor variable also defined in the current environment.</p>
</td></tr>
<tr><td><code id="adjustBrainCells_+3A_verbose">verbose</code></td>
<td>
<p>Whether to report the formula used for adjustment of each row.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing both a matrix of estimate surrogate proportion variables (SPVs), as well as a matrix of adjusted gene expression values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
brain_cells_adjusted = adjustBrainCells(aba_marker_expression,
  nMarker = 50, species = "combined")
expression_data_adj = brain_cells_adjusted$expression
cor_mic_unadj = cor.test(as.numeric(aba_marker_expression["AIF1", ]),
  as.numeric(aba_pheno_data$ihc_iba1_ffpe), method = "spearman")
cor_mic_adj = cor.test(expression_data_adj["AIF1", ],
  as.numeric(aba_pheno_data$ihc_iba1_ffpe), method = "spearman")

</code></pre>

<hr>
<h2 id='adjustCells'>Adjust for estimated cell type proportions in bulk gene expression data.</h2><span id='topic+adjustCells'></span>

<h3>Description</h3>

<p>This function uses a linear model to adjust each row of gene expression for cell types, taking the residuals from the linear model for downstream analysis. Other covariates can be included as well, if they are defined in the current environment, through the use of the formula argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjustCells(
  inputMat,
  cellSPV,
  celltypes = NULL,
  addMeans = FALSE,
  formula = NULL,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="adjustCells_+3A_inputmat">inputMat</code></td>
<td>
<p>Input gene expression data, with rows as features (e.g. genes) and samples as columns.</p>
</td></tr>
<tr><td><code id="adjustCells_+3A_cellspv">cellSPV</code></td>
<td>
<p>Estimated matrix of surrogate proportion variables for each cell type, with samples as rows and columns as cell types.</p>
</td></tr>
<tr><td><code id="adjustCells_+3A_celltypes">celltypes</code></td>
<td>
<p>Character vector of which cell types to use. Must correspond to column names in the cellSPV data frame. If NULL, all of the columns of SPV will be used for adjustment.</p>
</td></tr>
<tr><td><code id="adjustCells_+3A_addmeans">addMeans</code></td>
<td>
<p>Whether the mean should be added to the residuals in the resulting adjusted gene expression table.</p>
</td></tr>
<tr><td><code id="adjustCells_+3A_formula">formula</code></td>
<td>
<p>If you want to add additional covariates to be adjusted for, then you can supply them by adding an (optional) formula function here. The format should be &quot;expression_data ~ $cov1 + factor($cov2) +&quot;, where $cov1 is a numeric covariate present in the current environment, and $cov2 is a factor variable also defined in the current environment.</p>
</td></tr>
<tr><td><code id="adjustCells_+3A_verbose">verbose</code></td>
<td>
<p>Whether to print out the formula used for adjustment of each row.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of adjusted gene expression values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
svp_res = brainCells(inputMat = aba_marker_expression, nMarker = 10,
  species = "human", celltypes = c("ast", "neu", "oli"))
str(svp_res)
adjust_res = adjustCells(inputMat = aba_marker_expression,
  cellSPV = svp_res, addMeans = FALSE)
str(adjust_res)

</code></pre>

<hr>
<h2 id='brainCells'>Estimate brain cell type proportions in bulk expression data with marker genes.</h2><span id='topic+brainCells'></span>

<h3>Description</h3>

<p>This function uses marker genes estimated in a meta-analysis of brain cell type-associated RNA expression data sets, and uses them as input for the findCells cell type proportion estimation procedure pipeline.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>brainCells(
  inputMat,
  nMarker = 50,
  species = "combined",
  data_set = "mckenzie",
  celltypes = c("ast", "end", "mic", "neu", "oli", "opc"),
  method = "SVD",
  scale = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="brainCells_+3A_inputmat">inputMat</code></td>
<td>
<p>Gene expression data frame or matrix, with rownames corresponding to gene names, some of which are marker genes, and columns corresponding to samples.</p>
</td></tr>
<tr><td><code id="brainCells_+3A_nmarker">nMarker</code></td>
<td>
<p>The number of marker genes (that are present in your expression data set) to use in estimating the surrogate cell type proportion variable for each cell type.</p>
</td></tr>
<tr><td><code id="brainCells_+3A_species">species</code></td>
<td>
<p>By default, this function uses markers from combined human and mouse measurements, which are the most robust and reliable, as the gene expression patterns are very conserved between these two species. Other options are &quot;human&quot; and &quot;mouse&quot; for data specific to those species. Note that OPCs only have 500 gene symbols in this case, and are taken from only the Darmanis et al or Tasic et al data sets, respectively.</p>
</td></tr>
<tr><td><code id="brainCells_+3A_data_set">data_set</code></td>
<td>
<p>Which data set the data should be derived from. Options are &quot;mckenzie&quot; (default), &quot;kelley&quot;. Note that the &quot;kelley&quot; data set will ignore the &quot;species&quot; argument.</p>
</td></tr>
<tr><td><code id="brainCells_+3A_celltypes">celltypes</code></td>
<td>
<p>Character vector of which cell types to estimate.</p>
</td></tr>
<tr><td><code id="brainCells_+3A_method">method</code></td>
<td>
<p>To estimate the cell type proportions, can either use &quot;PCA&quot; or &quot;SVD&quot;.</p>
</td></tr>
<tr><td><code id="brainCells_+3A_scale">scale</code></td>
<td>
<p>Whether or not to scale the gene expression data from each marker gene prior to using it as an input for dimension reduction.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A sample-by-cell type matrix of estimate cell type proportion variables.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ct_res = brainCells(aba_marker_expression, nMarker = 50, species = "combined")
cor.test(ct_res[, "mic"], as.numeric(aba_pheno_data$ihc_iba1_ffpe), method = "spearman")

</code></pre>

<hr>
<h2 id='BRETIGEA'>BRETIGEA: BRain cEll Type specIfic Gene Expression Analysis</h2><span id='topic+BRETIGEA'></span>

<h3>Description</h3>

<p>This package provides two major functions:
1) A function to estimate cell type surrogate proportion variables based on marker genes.
2) A function to adjust bulk gene expression data for estimated cell type proportions, as covariates.
It also provides data containing estimated marker genes for six major brain cell types, i.e. astrocytes, endothelial cells, microglia, neurons, oligodendrocytes, and oligodendrocyte precursor cells (OPCs), as well as wrapper functions to use this data on the two major package functions.
</p>

<hr>
<h2 id='findCells'>Find cell type proportions from bulk gene expression data using marker genes.</h2><span id='topic+findCells'></span>

<h3>Description</h3>

<p>Input a gene expression matrix and your own data frame of marker genes, and this function will estimate cell type proportions in your data set using one of the SVD or PCA dimension reduction approaches.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findCells(inputMat, markers, nMarker = 50, method = "SVD", scale = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="findCells_+3A_inputmat">inputMat</code></td>
<td>
<p>Numeric gene expression data frame or matrix, with rownames corresponding to gene names, some of which are marker genes, and columns corresponding to samples.</p>
</td></tr>
<tr><td><code id="findCells_+3A_markers">markers</code></td>
<td>
<p>Data frame with marker genes in one column (named &quot;marker&quot;) and the cell type that that gene symbol corresponds to in another column (named &quot;cell&quot;).</p>
</td></tr>
<tr><td><code id="findCells_+3A_nmarker">nMarker</code></td>
<td>
<p>The number of marker genes (that are present in your expression data set) to use in estimating the surrogate cell type proportion variable for each cell type.</p>
</td></tr>
<tr><td><code id="findCells_+3A_method">method</code></td>
<td>
<p>To estimate the cell type proportions, can either use PCA or SVD.</p>
</td></tr>
<tr><td><code id="findCells_+3A_scale">scale</code></td>
<td>
<p>Whether or not to scale the gene expression data from each marker gene prior to using it as an input for dimension reduction.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A sample-by-cell type matrix of estimate cell type proportion variables.
</p>


<h3>References</h3>

<p>Chikina M, Zaslavsky E, Sealfon SC. CellCODE: a robust latent variable approach to differential expression analysis for heterogeneous cell populations. Bioinformatics. 2015;31(10):1584-91.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
cell_type_proportions = findCells(aba_marker_expression,
 markers = markers_df_brain, nMarker = 10)
str(cell_type_proportions)

</code></pre>

<hr>
<h2 id='kelley_df_brain'>Marker genes estimated from</h2><span id='topic+kelley_df_brain'></span>

<h3>Description</h3>

<p>Top 1000 marker genes from each of four major brain cell types (ie astrocytes, microglia, neurons, and oligodendrocytes) estimated via correlation-based analysis of bulk brain tissue. Contains marker data averaged across all brain regions as well as markers derived from individual brain regions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kelley_df_brain
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 84000 rows and 2 columns.
</p>


<h3>References</h3>

<p>Kelley KW, Nakao-inoue H, Molofsky AV, Oldham MC. Variation among intact tissue samples reveals the core transcriptional features of human CNS cell classes. Nat Neurosci. 2018;21(9):1171-1184.
</p>

<hr>
<h2 id='markers_df_brain'>Marker genes estimated from a meta-analysis of brain cell gene expression data from both humans and mice.</h2><span id='topic+markers_df_brain'></span>

<h3>Description</h3>

<p>Top 1000 marker genes from each of the six major brain cell types (ie astrocytes, endothelial cells, microglia, neurons, oligodendrocytes, and OPCs) estimated from a meta-analysis of brain cell gene expression data from both humans and mice.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>markers_df_brain
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 6000 rows and 2 columns.
</p>


<h3>References</h3>

<p>Mckenzie AT, Wang M, Hauberg ME, et al. Brain Cell Type Specific Gene Expression and Co-expression Network Architectures. Sci Rep. 2018;8(1):8868.
</p>

<hr>
<h2 id='markers_df_human_brain'>Marker genes estimated from a meta-analysis of brain cell gene expression data from humans only.</h2><span id='topic+markers_df_human_brain'></span>

<h3>Description</h3>

<p>Top 1000 marker genes from each of the six major brain cell types (ie astrocytes, endothelial cells, microglia, neurons, oligodendrocytes, and OPCs) estimated from a meta-analysis of brain cell gene expression data from humans only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>markers_df_human_brain
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 5500 rows and 2 columns.
</p>


<h3>References</h3>

<p>Mckenzie AT, Wang M, Hauberg ME, et al. Brain Cell Type Specific Gene Expression and Co-expression Network Architectures. Sci Rep. 2018;8(1):8868.
</p>

<hr>
<h2 id='markers_df_mouse_brain'>Marker genes estimated from a meta-analysis of brain cell gene expression data from mice only.</h2><span id='topic+markers_df_mouse_brain'></span>

<h3>Description</h3>

<p>Top 1000 marker genes from each of the six major brain cell types (ie astrocytes, endothelial cells, microglia, neurons, oligodendrocytes, and OPCs) estimated from a meta-analysis of brain cell gene expression data from mice only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>markers_df_mouse_brain
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 5430 rows and 2 columns.
</p>


<h3>References</h3>

<p>Mckenzie AT, Wang M, Hauberg ME, et al. Brain Cell Type Specific Gene Expression and Co-expression Network Architectures. Sci Rep. 2018;8(1):8868.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
