<!DOCTYPE html><html><head><title>Help for package bridger2</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {bridger2}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BridgeRCore'><p>BridgeR basic function for calculating RNA half-life from BRIC-seq data</p></a></li>
<li><a href='#BridgeRDatasetChecker'><p>BRIC-seq Dataset checker</p></a></li>
<li><a href='#BridgeRDataSetFromMatrix'><p>Calculate relative RPKM expression from data.table format.</p></a></li>
<li><a href='#BridgeRDataSetFromRaw'><p>Calculate relative RPKM expression from raw data.</p></a></li>
<li><a href='#BridgeReport'><p>Shinyapp reporting for drawing RNA decay curve.</p></a></li>
<li><a href='#BridgeRGrubbsTest'><p>RNA half-life Grubbs test.</p></a></li>
<li><a href='#BridgeRHalfLifeCalc3models'><p>Calculate RNA half-life for each gene using 3model method.</p></a></li>
<li><a href='#BridgeRHalfLifeCalcR2Select'><p>Calculate RNA half-life for each gene using R2 selection method.</p></a></li>
<li><a href='#BridgeRNormalization'><p>Calculate the normalized RPKM for BRIC-seq dataset.</p></a></li>
<li><a href='#BridgeRNormalizationFactors'><p>Calculate normalization factors for BRIC-seq datasets.</p></a></li>
<li><a href='#BridgeRNormalizationFactorsHK'><p>Calculate normalization factors from house-keeping genes.</p></a></li>
<li><a href='#BridgeRPvalueEvaluation'><p>Calculate Fold-change of RNA half-life and p-value.</p></a></li>
<li><a href='#BridgeRResultChecker'><p>BRIC-seq result checker</p></a></li>
<li><a href='#CalcHalflifeDeviation'><p>Calculate RNA half-life SD.</p></a></li>
<li><a href='#halflife_table'><p>BRIC-seq result dataset for p-value estimation using grubbs test</p></a></li>
<li><a href='#RNA_halflife_comparison'><p>test BRIC-seq dataset for RNA half-life comparison</p></a></li>
<li><a href='#RNA_halflife_comparison_HK'><p>test BRIC-seq dataset for RNA half-life comparison using House-keeping genes.</p></a></li>
<li><a href='#RNA_halflife_grubbs_test'><p>test BRIC-seq dataset for p-value estimation using grubbs test</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Genome-Wide RNA Degradation Analysis Using BRIC-Seq Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>BRIC-seq is a genome-wide approach for determining RNA stability in mammalian cells.
    This package provides a series of functions for performing quality check of your BRIC-seq data,
    calculation of RNA half-life for each transcript and comparison of RNA half-lives between two conditions.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, ggplot2, shiny, shinydashboard, plotly, BSDA,
outliers</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-12-05 11:08:16 UTC; Naoto</td>
</tr>
<tr>
<td>Author:</td>
<td>Naoto Imamachi [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Naoto Imamachi &lt;naoto.imamachi@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-12-05 14:49:24</td>
</tr>
</table>
<hr>
<h2 id='BridgeRCore'>BridgeR basic function for calculating RNA half-life from BRIC-seq data</h2><span id='topic+BridgeRCore'></span>

<h3>Description</h3>

<p><code>BridgeRCore</code> is a basic function for calculating RNA half-life
BRIC-seq data and a wrapper of the other individual bridger2 functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BridgeRCore(inputFile, inforColumn = 4, group = c("Control", "Knockdown"),
  hour = c(0, 1, 2, 4, 8, 12), RPKMcutoff = 0.1, cutoffBelow = 0.1,
  YMin = -2, YMax = 2, downsamplingFig = 0.2, makeFig = FALSE,
  cutoffQuantile = 0.975, inforHKGenesRow = "symbol", HKGenes = c("GAPDH",
  "PGK1", "PPIA", "ENO1", "ATP5B", "ALDOA"), CutoffTimePointNumber = 4,
  R2_criteria = 0.9, TimePointRemoval1 = c(1, 2), TimePointRemoval2 = c(8,
  12), ThresholdHalfLife1 = 3, ThresholdHalfLife2 = 12, save = TRUE,
  outputPrefix = "BridgeR", normalization = "default", method = "default")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BridgeRCore_+3A_inputfile">inputFile</code></td>
<td>
<p>The vector of tab-delimited matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_inforcolumn">inforColumn</code></td>
<td>
<p>The number of information columns.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_group">group</code></td>
<td>
<p>The vector of group names.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_hour">hour</code></td>
<td>
<p>The vector of time course about BRIC-seq experiment.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_rpkmcutoff">RPKMcutoff</code></td>
<td>
<p>Cutoff value of RPKM at 0hr.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_cutoffbelow">cutoffBelow</code></td>
<td>
<p>Cutoff value of RPKM at all time points.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_ymin">YMin</code></td>
<td>
<p>Y-axis min.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_ymax">YMax</code></td>
<td>
<p>Y-axis max.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_downsamplingfig">downsamplingFig</code></td>
<td>
<p>the factor for downsampling.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_makefig">makeFig</code></td>
<td>
<p>Whether to save the figure of normalization factor.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_cutoffquantile">cutoffQuantile</code></td>
<td>
<p>cutoff value of quantile.#' @param save Whether to save the output matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_inforhkgenesrow">inforHKGenesRow</code></td>
<td>
<p>The column number of house-keeping gene information.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_hkgenes">HKGenes</code></td>
<td>
<p>The vector of house-keeping genes.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_cutofftimepointnumber">CutoffTimePointNumber</code></td>
<td>
<p>The number of minimum time points for calc.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_r2_criteria">R2_criteria</code></td>
<td>
<p>The cutoff of R2 for R2 selection.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_timepointremoval1">TimePointRemoval1</code></td>
<td>
<p>The candicate_1 of time point removal.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_timepointremoval2">TimePointRemoval2</code></td>
<td>
<p>The candicate_2 of time point removal.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_thresholdhalflife1">ThresholdHalfLife1</code></td>
<td>
<p>The cutoff of TimePointRemoval1.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_thresholdhalflife2">ThresholdHalfLife2</code></td>
<td>
<p>The cutoff of TimePointRemoval2.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_save">save</code></td>
<td>
<p>Whether to save the output matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_outputprefix">outputPrefix</code></td>
<td>
<p>The prefix for the name of the output.</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_normalization">normalization</code></td>
<td>
<p>select &quot;default&quot; (percentile method) or &quot;house_keeping_genes&quot;</p>
</td></tr>
<tr><td><code id="BridgeRCore_+3A_method">method</code></td>
<td>
<p>select &quot;default&quot; (R2 selection/1st-order) or &quot;3models&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table object including RNA half-life, R2 and the selected fitting model.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>halflife_table &lt;- BridgeRCore(RNA_halflife_comparison[1:30,],
                              save = FALSE)
halflife_table &lt;- BridgeRCore(RNA_halflife_comparison_HK[177:206],
                              save = FALSE,
                              normalization = "house_keeping_genes",
                              method = "3models")

</code></pre>

<hr>
<h2 id='BridgeRDatasetChecker'>BRIC-seq Dataset checker</h2><span id='topic+BridgeRDatasetChecker'></span>

<h3>Description</h3>

<p><code>BridgeRDatasetChecker</code> returns several BRIC-seq dataset information.
This function is used for checking your BRIC-seq dataset quality.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BridgeRDatasetChecker(inputFile, group = c("Control", "Knockdown"),
  hour = c(0, 1, 2, 4, 8, 12), inforColumn = 4, percentile = c(0.99, 0.95,
  0.9, 0.8, 0.7, 0.6, 0.5, 0.4, 0.3, 0.2, 0.1, 0.05), save = T,
  outputPrefix = "BridgeR_2_raw")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BridgeRDatasetChecker_+3A_inputfile">inputFile</code></td>
<td>
<p>Input matrix object.</p>
</td></tr>
<tr><td><code id="BridgeRDatasetChecker_+3A_group">group</code></td>
<td>
<p>The vector of group names.</p>
</td></tr>
<tr><td><code id="BridgeRDatasetChecker_+3A_hour">hour</code></td>
<td>
<p>The vector of time course about BRIC-seq experiment.</p>
</td></tr>
<tr><td><code id="BridgeRDatasetChecker_+3A_inforcolumn">inforColumn</code></td>
<td>
<p>The number of information columns.</p>
</td></tr>
<tr><td><code id="BridgeRDatasetChecker_+3A_percentile">percentile</code></td>
<td>
<p>Percentile numbers.</p>
</td></tr>
<tr><td><code id="BridgeRDatasetChecker_+3A_save">save</code></td>
<td>
<p>Whether to save the output fig file.</p>
</td></tr>
<tr><td><code id="BridgeRDatasetChecker_+3A_outputprefix">outputPrefix</code></td>
<td>
<p>The prefix for the name of the output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list object about ggplot2 fig data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(data.table)
normalized_table &lt;- data.table(gr_id = c(8, 9, 14),
                               symbol = c("AAAS", "AACS", "AADAT"),
                               accession_id = c("NM_015665", "NM_023928", "NM_182662"),
                               locus = c("chr12", "chr12", "chr4"),
                               CTRL_1_0h = c(1.00, 1.00, 1.00),
                               CTRL_1_1h = c(1.00, 0.86, 0.96),
                               CTRL_1_2h = c(1.00, 0.96, 0.88),
                               CTRL_1_4h = c(1.00, 0.74, 0.85),
                               CTRL_1_8h = c(1.00, 0.86, 0.68),
                               CTRL_1_12h = c(1.01, 0.65, 0.60),
                               gr_id = c(8, 9, 14),
                               symbol = c("AAAS", "AACS", "AADAT"),
                               accession_id = c("NM_015665", "NM_023928", "NM_182662"),
                               locus = c("chr12", "chr12", "chr4"),
                               KD_1_0h = c(1.00, 1.00, 1.00),
                               KD_1_1h = c(1.01, 0.73, 0.71),
                               KD_1_2h = c(1.01, 0.77, 0.69),
                               KD_1_4h = c(1.01, 0.72, 0.67),
                               KD_1_8h = c(1.01, 0.64, 0.38),
                               KD_1_12h = c(1.00, 0.89, 0.63))
group &lt;- c("Control", "Knockdown")
hour &lt;- c(0, 1, 2, 4, 8, 12)
fig_list_norm &lt;- BridgeRDatasetChecker(inputFile = normalized_table,
                                       save = FALSE)

</code></pre>

<hr>
<h2 id='BridgeRDataSetFromMatrix'>Calculate relative RPKM expression from data.table format.</h2><span id='topic+BridgeRDataSetFromMatrix'></span>

<h3>Description</h3>

<p><code>BridgeRDataSetFromMatrix</code> calculates the relative RPKM values compared with 0hr,
importing data.table format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BridgeRDataSetFromMatrix(inputFile, group = c("Control", "Knockdown"),
  hour = c(0, 1, 2, 4, 8, 12), cutoff = 0.1, cutoffBelow = 0.1,
  inforColumn = 4, save = T, outputPrefix = "BridgeR_1")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BridgeRDataSetFromMatrix_+3A_inputfile">inputFile</code></td>
<td>
<p>The vector of tab-delimited matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRDataSetFromMatrix_+3A_group">group</code></td>
<td>
<p>The vector of group names.</p>
</td></tr>
<tr><td><code id="BridgeRDataSetFromMatrix_+3A_hour">hour</code></td>
<td>
<p>The vector of time course about BRIC-seq experiment.</p>
</td></tr>
<tr><td><code id="BridgeRDataSetFromMatrix_+3A_cutoff">cutoff</code></td>
<td>
<p>Cutoff value of RPKM at 0hr.</p>
</td></tr>
<tr><td><code id="BridgeRDataSetFromMatrix_+3A_cutoffbelow">cutoffBelow</code></td>
<td>
<p>Cutoff value of RPKM at all time points.</p>
</td></tr>
<tr><td><code id="BridgeRDataSetFromMatrix_+3A_inforcolumn">inforColumn</code></td>
<td>
<p>The number of information columns.</p>
</td></tr>
<tr><td><code id="BridgeRDataSetFromMatrix_+3A_save">save</code></td>
<td>
<p>Whether to save the output matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRDataSetFromMatrix_+3A_outputprefix">outputPrefix</code></td>
<td>
<p>The prefix for the name of the output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table object about relative RPKM values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(data.table)
rpkm_matrix &lt;- data.table(gr_id = c(8, 9, 14),
                          symbol = c("AAAS", "AACS", "AADAT"),
                          accession_id = c("NM_015665", "NM_023928", "NM_182662"),
                          locus = c("chr12", "chr12", "chr4"),
                          CTRL_1_0h = c(41, 5, 5),
                          CTRL_1_1h = c(48, 7, 6),
                          CTRL_1_2h = c(56, 10, 6),
                          CTRL_1_4h = c(87, 12, 10),
                          CTRL_1_8h = c(124, 20, 11),
                          CTRL_1_12h = c(185, 22, 15),
                          gr_id = c(8, 9, 14),
                          symbol = c("AAAS", "AACS", "AADAT"),
                          accession_id = c("NM_015665", "NM_023928", "NM_182662"),
                          locus = c("chr12", "chr12", "chr4"),
                          KD_1_0h = c(21, 10, 3),
                          KD_1_1h = c(33, 11, 3),
                          KD_1_2h = c(42, 15, 4),
                          KD_1_4h = c(60, 20, 5),
                          KD_1_8h = c(65, 37, 6),
                          KD_1_12h = c(70, 42, 6))
group &lt;- c("Control", "Knockdown")
hour &lt;- c(0, 1, 2, 4, 8, 12)
test_table &lt;- BridgeRDataSetFromMatrix(inputFile = rpkm_matrix,
                                       group = group,
                                       hour = hour,
                                       cutoff = 0.1,
                                       inforColumn = 4,
                                       save = FALSE)

</code></pre>

<hr>
<h2 id='BridgeRDataSetFromRaw'>Calculate relative RPKM expression from raw data.</h2><span id='topic+BridgeRDataSetFromRaw'></span>

<h3>Description</h3>

<p><code>BridgeRDataSetFromRaw</code> calculates the relative RPKM values
compared with 0hr, importing tab-delimited txt file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BridgeRDataSetFromRaw(inputFile, group = c("Control", "Knockdown"),
  hour = c(0, 1, 2, 4, 8, 12), cutoff = 0.1, cutoffBelow = 0.1,
  inforColumn = 4, save = T, outputPrefix = "BridgeR_1")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BridgeRDataSetFromRaw_+3A_inputfile">inputFile</code></td>
<td>
<p>The vector of tab-delimited matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRDataSetFromRaw_+3A_group">group</code></td>
<td>
<p>The vector of group names.</p>
</td></tr>
<tr><td><code id="BridgeRDataSetFromRaw_+3A_hour">hour</code></td>
<td>
<p>The vector of time course about BRIC-seq experiment.</p>
</td></tr>
<tr><td><code id="BridgeRDataSetFromRaw_+3A_cutoff">cutoff</code></td>
<td>
<p>Cutoff value of RPKM at 0hr.</p>
</td></tr>
<tr><td><code id="BridgeRDataSetFromRaw_+3A_cutoffbelow">cutoffBelow</code></td>
<td>
<p>Cutoff value of RPKM at all time points.</p>
</td></tr>
<tr><td><code id="BridgeRDataSetFromRaw_+3A_inforcolumn">inforColumn</code></td>
<td>
<p>The number of information columns.</p>
</td></tr>
<tr><td><code id="BridgeRDataSetFromRaw_+3A_save">save</code></td>
<td>
<p>Whether to save the output matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRDataSetFromRaw_+3A_outputprefix">outputPrefix</code></td>
<td>
<p>The prefix for the name of the output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table object about relative RPKM values.
</p>

<hr>
<h2 id='BridgeReport'>Shinyapp reporting for drawing RNA decay curve.</h2><span id='topic+BridgeReport'></span>

<h3>Description</h3>

<p><code>BridgeReport</code> returns a shinyapp object to draw RNA decay curve.
You can easily check RNA half-life and RNA decay fitting curve on
your web browser.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BridgeReport(inputFile, group = c("Control", "Knockdown"), hour = c(0, 1, 2,
  4, 8, 12), comparisonFile = c("Control", "Knockdown"),
  searchRowName = "symbol", inforColumn = 4, color = c("black", "red"),
  TimePointRemoval1 = c(1, 2), TimePointRemoval2 = c(8, 12))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BridgeReport_+3A_inputfile">inputFile</code></td>
<td>
<p>The vector of tab-delimited matrix file.</p>
</td></tr>
<tr><td><code id="BridgeReport_+3A_group">group</code></td>
<td>
<p>The vector of group names.</p>
</td></tr>
<tr><td><code id="BridgeReport_+3A_hour">hour</code></td>
<td>
<p>The vector of time course about BRIC-seq experiment.</p>
</td></tr>
<tr><td><code id="BridgeReport_+3A_comparisonfile">comparisonFile</code></td>
<td>
<p>The vector of group names.</p>
</td></tr>
<tr><td><code id="BridgeReport_+3A_searchrowname">searchRowName</code></td>
<td>
<p>Row name for searching.</p>
</td></tr>
<tr><td><code id="BridgeReport_+3A_inforcolumn">inforColumn</code></td>
<td>
<p>The number of information columns.</p>
</td></tr>
<tr><td><code id="BridgeReport_+3A_color">color</code></td>
<td>
<p>color of line graph for two decay curve.</p>
</td></tr>
<tr><td><code id="BridgeReport_+3A_timepointremoval1">TimePointRemoval1</code></td>
<td>
<p>The candicate_1 of time point removal.</p>
</td></tr>
<tr><td><code id="BridgeReport_+3A_timepointremoval2">TimePointRemoval2</code></td>
<td>
<p>The candicate_2 of time point removal.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>shiny.appobj object for searching and showing RNA decay curve for each gene.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(data.table)
normalized_rpkm_matrix &lt;- data.table(gr_id = c(8, 9, 14),
                                     symbol = c("AAAS", "AACS", "AADAT"),
                                     accession_id = c("NM_015665", "NM_023928", "NM_182662"),
                                     locus = c("chr12", "chr12", "chr4"),
                                     CTRL_1_0h = c(1.00, 1.00, 1.00),
                                     CTRL_1_1h = c(1.00, 0.86, 0.96),
                                     CTRL_1_2h = c(1.00, 0.96, 0.88),
                                     CTRL_1_4h = c(1.00, 0.74, 0.85),
                                     CTRL_1_8h = c(1.00, 0.86, 0.68),
                                     CTRL_1_12h = c(1.01, 0.65, 0.60),
                                     gr_id = c(8, 9, 14),
                                     symbol = c("AAAS", "AACS", "AADAT"),
                                     accession_id = c("NM_015665", "NM_023928", "NM_182662"),
                                     locus = c("chr12", "chr12", "chr4"),
                                     KD_1_0h = c(1.00, 1.00, 1.00),
                                     KD_1_1h = c(1.01, 0.73, 0.71),
                                     KD_1_2h = c(1.01, 0.77, 0.69),
                                     KD_1_4h = c(1.01, 0.72, 0.67),
                                     KD_1_8h = c(1.01, 0.64, 0.38),
                                     KD_1_12h = c(1.00, 0.89, 0.63))
group &lt;- c("Control", "Knockdown")
hour &lt;- c(0, 1, 2, 4, 8, 12)
halflife_table &lt;- BridgeRHalfLifeCalcR2Select(normalized_rpkm_matrix,
                                              group = group,
                                              hour = hour,
                                              save = FALSE)
pvalue_table &lt;- BridgeRPvalueEvaluation(halflife_table,
                                        save = FALSE)
shiny_test &lt;- BridgeReport(pvalue_table)

</code></pre>

<hr>
<h2 id='BridgeRGrubbsTest'>RNA half-life Grubbs test.</h2><span id='topic+BridgeRGrubbsTest'></span>

<h3>Description</h3>

<p><code>BridgeRGrubbsTest</code> calculates the p-value for each gene
using grubbs test. The estimation is based on the standard deviation of
RNA half-lives in control conditions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BridgeRGrubbsTest(controlFile, compFile, hour = c(0, 1, 2, 4, 8, 12),
  controlGroup = c("CTRL_PUM1", "CTRL_PUM2", "CTRL_DKD"), inforColumn = 4,
  compIndex = 2, save = T, outputPrefix = "BridgeR_8")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BridgeRGrubbsTest_+3A_controlfile">controlFile</code></td>
<td>
<p>The dataframe of halflife table.</p>
</td></tr>
<tr><td><code id="BridgeRGrubbsTest_+3A_compfile">compFile</code></td>
<td>
<p>The dataframe of RPKM table.</p>
</td></tr>
<tr><td><code id="BridgeRGrubbsTest_+3A_hour">hour</code></td>
<td>
<p>The vector of time course about BRIC-seq experiment.</p>
</td></tr>
<tr><td><code id="BridgeRGrubbsTest_+3A_controlgroup">controlGroup</code></td>
<td>
<p>The vector of group names.</p>
</td></tr>
<tr><td><code id="BridgeRGrubbsTest_+3A_inforcolumn">inforColumn</code></td>
<td>
<p>The number of information columns.</p>
</td></tr>
<tr><td><code id="BridgeRGrubbsTest_+3A_compindex">compIndex</code></td>
<td>
<p>The number of information columns.</p>
</td></tr>
<tr><td><code id="BridgeRGrubbsTest_+3A_save">save</code></td>
<td>
<p>Whether to save the output matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRGrubbsTest_+3A_outputprefix">outputPrefix</code></td>
<td>
<p>The prefix for the name of the output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table object about Grabbs test result.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>group &lt;- c("Control", "Knockdown")
hour &lt;- c(0, 1, 2, 4, 8, 12)
half_sd_table &lt;- CalcHalflifeDeviation(halflife_table,
                                       RNA_halflife_grubbs_test,
                                       group = c("CTRL_1",
                                                 "CTRL_2",
                                                 "CTRL_3"),
                                       save = FALSE)
grubbs_table &lt;- BridgeRGrubbsTest(half_sd_table,
                                  halflife_table,
                                  compIndex = 4,
                                  controlGroup = c("CTRL_1",
                                                   "CTRL_2",
                                                   "CTRL_3"),
                                  save = FALSE)

</code></pre>

<hr>
<h2 id='BridgeRHalfLifeCalc3models'>Calculate RNA half-life for each gene using 3model method.</h2><span id='topic+BridgeRHalfLifeCalc3models'></span>

<h3>Description</h3>

<p><code>BridgeRHalfLifeCalc3models</code> calculates RNA half-life for each gene
using 3 models methods (older version).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BridgeRHalfLifeCalc3models(inputFile, group = c("Control", "Knockdown"),
  hour = c(0, 1, 2, 4, 8, 12), inforColumn = 4, CutoffTimePointNumber = 4,
  save = T, outputPrefix = "BridgeR_5")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BridgeRHalfLifeCalc3models_+3A_inputfile">inputFile</code></td>
<td>
<p>The vector of tab-delimited matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRHalfLifeCalc3models_+3A_group">group</code></td>
<td>
<p>The vector of group names.</p>
</td></tr>
<tr><td><code id="BridgeRHalfLifeCalc3models_+3A_hour">hour</code></td>
<td>
<p>The vector of time course about BRIC-seq experiment.</p>
</td></tr>
<tr><td><code id="BridgeRHalfLifeCalc3models_+3A_inforcolumn">inforColumn</code></td>
<td>
<p>The number of information columns.</p>
</td></tr>
<tr><td><code id="BridgeRHalfLifeCalc3models_+3A_cutofftimepointnumber">CutoffTimePointNumber</code></td>
<td>
<p>The number of minimum time points for calc.</p>
</td></tr>
<tr><td><code id="BridgeRHalfLifeCalc3models_+3A_save">save</code></td>
<td>
<p>Whether to save the output matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRHalfLifeCalc3models_+3A_outputprefix">outputPrefix</code></td>
<td>
<p>The prefix for the name of the output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table object about RNA half-life, R2 and fitting model.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(data.table)
normalized_rpkm_matrix &lt;- data.table(gr_id = c(8, 9, 14),
                                     symbol = c("AAAS", "AACS", "AADAT"),
                                     accession_id = c("NM_015665", "NM_023928", "NM_182662"),
                                     locus = c("chr12", "chr12", "chr4"),
                                     CTRL_1_0h = c(1.00, 1.00, 1.00),
                                     CTRL_1_1h = c(1.00, 0.86, 0.96),
                                     CTRL_1_2h = c(1.00, 0.96, 0.88),
                                     CTRL_1_4h = c(1.00, 0.74, 0.85),
                                     CTRL_1_8h = c(1.00, 0.86, 0.68),
                                     CTRL_1_12h = c(1.01, 0.65, 0.60),
                                     gr_id = c(8, 9, 14),
                                     symbol = c("AAAS", "AACS", "AADAT"),
                                     accession_id = c("NM_015665", "NM_023928", "NM_182662"),
                                     locus = c("chr12", "chr12", "chr4"),
                                     KD_1_0h = c(1.00, 1.00, 1.00),
                                     KD_1_1h = c(1.01, 0.73, 0.71),
                                     KD_1_2h = c(1.01, 0.77, 0.69),
                                     KD_1_4h = c(1.01, 0.72, 0.67),
                                     KD_1_8h = c(1.01, 0.64, 0.38),
                                     KD_1_12h = c(1.00, 0.89, 0.63))
group &lt;- c("Control", "Knockdown")
hour &lt;- c(0, 1, 2, 4, 8, 12)
halflife_table &lt;- BridgeRHalfLifeCalc3models(normalized_rpkm_matrix,
                                             group = group,
                                             hour = hour,
                                             save = FALSE)

</code></pre>

<hr>
<h2 id='BridgeRHalfLifeCalcR2Select'>Calculate RNA half-life for each gene using R2 selection method.</h2><span id='topic+BridgeRHalfLifeCalcR2Select'></span>

<h3>Description</h3>

<p><code>BridgeRHalfLifeCalcR2Select</code> calculates RNA half-life for each gene
using R2 selection method (default version).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BridgeRHalfLifeCalcR2Select(inputFile, group = c("Control", "Knockdown"),
  hour = c(0, 1, 2, 4, 8, 12), inforColumn = 4, CutoffTimePointNumber = 4,
  R2_criteria = 0.9, TimePointRemoval1 = c(1, 2), TimePointRemoval2 = c(8,
  12), ThresholdHalfLife1 = 3, ThresholdHalfLife2 = 12, save = T,
  outputPrefix = "BridgeR_5")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BridgeRHalfLifeCalcR2Select_+3A_inputfile">inputFile</code></td>
<td>
<p>The vector of tab-delimited matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRHalfLifeCalcR2Select_+3A_group">group</code></td>
<td>
<p>The vector of group names.</p>
</td></tr>
<tr><td><code id="BridgeRHalfLifeCalcR2Select_+3A_hour">hour</code></td>
<td>
<p>The vector of time course about BRIC-seq experiment.</p>
</td></tr>
<tr><td><code id="BridgeRHalfLifeCalcR2Select_+3A_inforcolumn">inforColumn</code></td>
<td>
<p>The number of information columns.</p>
</td></tr>
<tr><td><code id="BridgeRHalfLifeCalcR2Select_+3A_cutofftimepointnumber">CutoffTimePointNumber</code></td>
<td>
<p>The number of minimum time points for calc.</p>
</td></tr>
<tr><td><code id="BridgeRHalfLifeCalcR2Select_+3A_r2_criteria">R2_criteria</code></td>
<td>
<p>The cutoff of R2 for R2 selection.</p>
</td></tr>
<tr><td><code id="BridgeRHalfLifeCalcR2Select_+3A_timepointremoval1">TimePointRemoval1</code></td>
<td>
<p>The candicate_1 of time point removal.</p>
</td></tr>
<tr><td><code id="BridgeRHalfLifeCalcR2Select_+3A_timepointremoval2">TimePointRemoval2</code></td>
<td>
<p>The candicate_2 of time point removal.</p>
</td></tr>
<tr><td><code id="BridgeRHalfLifeCalcR2Select_+3A_thresholdhalflife1">ThresholdHalfLife1</code></td>
<td>
<p>The cutoff of TimePointRemoval1.</p>
</td></tr>
<tr><td><code id="BridgeRHalfLifeCalcR2Select_+3A_thresholdhalflife2">ThresholdHalfLife2</code></td>
<td>
<p>The cutoff of TimePointRemoval2.</p>
</td></tr>
<tr><td><code id="BridgeRHalfLifeCalcR2Select_+3A_save">save</code></td>
<td>
<p>Whether to save the output matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRHalfLifeCalcR2Select_+3A_outputprefix">outputPrefix</code></td>
<td>
<p>The prefix for the name of the output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table object about RNA half-life, R2 and fitting model.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(data.table)
normalized_rpkm_matrix &lt;- data.table(gr_id = c(8, 9, 14),
                                     symbol = c("AAAS", "AACS", "AADAT"),
                                     accession_id = c("NM_015665", "NM_023928", "NM_182662"),
                                     locus = c("chr12", "chr12", "chr4"),
                                     CTRL_1_0h = c(1.00, 1.00, 1.00),
                                     CTRL_1_1h = c(1.00, 0.86, 0.96),
                                     CTRL_1_2h = c(1.00, 0.96, 0.88),
                                     CTRL_1_4h = c(1.00, 0.74, 0.85),
                                     CTRL_1_8h = c(1.00, 0.86, 0.68),
                                     CTRL_1_12h = c(1.01, 0.65, 0.60),
                                     gr_id = c(8, 9, 14),
                                     symbol = c("AAAS", "AACS", "AADAT"),
                                     accession_id = c("NM_015665", "NM_023928", "NM_182662"),
                                     locus = c("chr12", "chr12", "chr4"),
                                     KD_1_0h = c(1.00, 1.00, 1.00),
                                     KD_1_1h = c(1.01, 0.73, 0.71),
                                     KD_1_2h = c(1.01, 0.77, 0.69),
                                     KD_1_4h = c(1.01, 0.72, 0.67),
                                     KD_1_8h = c(1.01, 0.64, 0.38),
                                     KD_1_12h = c(1.00, 0.89, 0.63))
group &lt;- c("Control", "Knockdown")
hour &lt;- c(0, 1, 2, 4, 8, 12)
halflife_table &lt;- BridgeRHalfLifeCalcR2Select(normalized_rpkm_matrix,
                                              group = group,
                                              hour = hour,
                                              save = FALSE)

</code></pre>

<hr>
<h2 id='BridgeRNormalization'>Calculate the normalized RPKM for BRIC-seq dataset.</h2><span id='topic+BridgeRNormalization'></span>

<h3>Description</h3>

<p><code>BridgeRNormalization</code> calculates the normalized RPKM values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BridgeRNormalization(inputFile, normFactorFile, group = c("Control",
  "Knockdown"), hour = c(0, 1, 2, 4, 8, 12), inforColumn = 4, save = T,
  outputPrefix = "BridgeR_4")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BridgeRNormalization_+3A_inputfile">inputFile</code></td>
<td>
<p>The vector of tab-delimited matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRNormalization_+3A_normfactorfile">normFactorFile</code></td>
<td>
<p>The vector of tab-delimited normalization factor file.</p>
</td></tr>
<tr><td><code id="BridgeRNormalization_+3A_group">group</code></td>
<td>
<p>The vector of group names.</p>
</td></tr>
<tr><td><code id="BridgeRNormalization_+3A_hour">hour</code></td>
<td>
<p>The vector of time course about BRIC-seq experiment.</p>
</td></tr>
<tr><td><code id="BridgeRNormalization_+3A_inforcolumn">inforColumn</code></td>
<td>
<p>The number of information columns.</p>
</td></tr>
<tr><td><code id="BridgeRNormalization_+3A_save">save</code></td>
<td>
<p>Whether to save the output matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRNormalization_+3A_outputprefix">outputPrefix</code></td>
<td>
<p>The prefix for the name of the output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table object about normalized RPKM values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(data.table)
rpkm_matrix &lt;- data.table(gr_id = c(8, 9, 14),
                          symbol = c("AAAS", "AACS", "AADAT"),
                          accession_id = c("NM_015665", "NM_023928", "NM_182662"),
                          locus = c("chr12", "chr12", "chr4"),
                          CTRL_1_0h = c(41, 5, 5),
                          CTRL_1_1h = c(48, 7, 6),
                          CTRL_1_2h = c(56, 10, 6),
                          CTRL_1_4h = c(87, 12, 10),
                          CTRL_1_8h = c(124, 20, 11),
                          CTRL_1_12h = c(185, 22, 15),
                          gr_id = c(8, 9, 14),
                          symbol = c("AAAS", "AACS", "AADAT"),
                          accession_id = c("NM_015665", "NM_023928", "NM_182662"),
                          locus = c("chr12", "chr12", "chr4"),
                          KD_1_0h = c(21, 10, 3),
                          KD_1_1h = c(33, 11, 3),
                          KD_1_2h = c(42, 15, 4),
                          KD_1_4h = c(60, 20, 5),
                          KD_1_8h = c(65, 37, 6),
                          KD_1_12h = c(70, 42, 6))
group &lt;- c("Control", "Knockdown")
hour &lt;- c(0, 1, 2, 4, 8, 12)
rpkm_list &lt;- BridgeRDataSetFromMatrix(inputFile = rpkm_matrix,
                                      group = group,
                                      hour = hour,
                                      cutoff = 0.1,
                                      inforColumn = 4,
                                      save = FALSE)
raw_table &lt;- rpkm_list[[1]]
test_table &lt;- rpkm_list[[2]]
factor_table &lt;- BridgeRNormalizationFactors(test_table,
                                            save = FALSE)
normalized_table &lt;- BridgeRNormalization(test_table,
                                         factor_table,
                                         save = FALSE)

</code></pre>

<hr>
<h2 id='BridgeRNormalizationFactors'>Calculate normalization factors for BRIC-seq datasets.</h2><span id='topic+BridgeRNormalizationFactors'></span>

<h3>Description</h3>

<p><code>BridgeRNormalizationFactors</code> calculates the normalization factors
for BRIC-seq datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BridgeRNormalizationFactors(inputFile, group = c("Control", "Knockdown"),
  hour = c(0, 1, 2, 4, 8, 12), inforColumn = 4, save = T, YMin = -2,
  YMax = 2, downsamplingFig = 0.2, makeFig = FALSE,
  cutoffQuantile = 0.975, figOutputPrefix = "BridgeR_3_fig",
  factorOutputPrefix = "BridgeR_3")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BridgeRNormalizationFactors_+3A_inputfile">inputFile</code></td>
<td>
<p>The vector of tab-delimited matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRNormalizationFactors_+3A_group">group</code></td>
<td>
<p>The vector of group names.</p>
</td></tr>
<tr><td><code id="BridgeRNormalizationFactors_+3A_hour">hour</code></td>
<td>
<p>The vector of time course about BRIC-seq experiment.</p>
</td></tr>
<tr><td><code id="BridgeRNormalizationFactors_+3A_inforcolumn">inforColumn</code></td>
<td>
<p>The number of information columns.</p>
</td></tr>
<tr><td><code id="BridgeRNormalizationFactors_+3A_save">save</code></td>
<td>
<p>Whether to save the output matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRNormalizationFactors_+3A_ymin">YMin</code></td>
<td>
<p>Y-axis min.</p>
</td></tr>
<tr><td><code id="BridgeRNormalizationFactors_+3A_ymax">YMax</code></td>
<td>
<p>Y-axis max.</p>
</td></tr>
<tr><td><code id="BridgeRNormalizationFactors_+3A_downsamplingfig">downsamplingFig</code></td>
<td>
<p>the factor for downsampling.</p>
</td></tr>
<tr><td><code id="BridgeRNormalizationFactors_+3A_makefig">makeFig</code></td>
<td>
<p>Whether to save the figure of normalization factor.</p>
</td></tr>
<tr><td><code id="BridgeRNormalizationFactors_+3A_cutoffquantile">cutoffQuantile</code></td>
<td>
<p>cutoff value of quantile.</p>
</td></tr>
<tr><td><code id="BridgeRNormalizationFactors_+3A_figoutputprefix">figOutputPrefix</code></td>
<td>
<p>The prefix for the name of figure output.</p>
</td></tr>
<tr><td><code id="BridgeRNormalizationFactors_+3A_factoroutputprefix">factorOutputPrefix</code></td>
<td>
<p>The prefix for the name of factor output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table object about normalization factors calculated by quantile method.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(data.table)
rpkm_matrix &lt;- data.table(gr_id = c(8, 9, 14),
                          symbol = c("AAAS", "AACS", "AADAT"),
                          accession_id = c("NM_015665", "NM_023928", "NM_182662"),
                          locus = c("chr12", "chr12", "chr4"),
                          CTRL_1_0h = c(41, 5, 5),
                          CTRL_1_1h = c(48, 7, 6),
                          CTRL_1_2h = c(56, 10, 6),
                          CTRL_1_4h = c(87, 12, 10),
                          CTRL_1_8h = c(124, 20, 11),
                          CTRL_1_12h = c(185, 22, 15),
                          gr_id = c(8, 9, 14),
                          symbol = c("AAAS", "AACS", "AADAT"),
                          accession_id = c("NM_015665", "NM_023928", "NM_182662"),
                          locus = c("chr12", "chr12", "chr4"),
                          KD_1_0h = c(21, 10, 3),
                          KD_1_1h = c(33, 11, 3),
                          KD_1_2h = c(42, 15, 4),
                          KD_1_4h = c(60, 20, 5),
                          KD_1_8h = c(65, 37, 6),
                          KD_1_12h = c(70, 42, 6))
group &lt;- c("Control", "Knockdown")
hour &lt;- c(0, 1, 2, 4, 8, 12)
rpkm_list &lt;- BridgeRDataSetFromMatrix(inputFile = rpkm_matrix,
                                      group = group,
                                      hour = hour,
                                      cutoff = 0.1,
                                      inforColumn = 4,
                                      save = FALSE)
raw_table &lt;- rpkm_list[[1]]
test_table &lt;- rpkm_list[[2]]
factor_table &lt;- BridgeRNormalizationFactors(test_table,
                                            save = FALSE)

</code></pre>

<hr>
<h2 id='BridgeRNormalizationFactorsHK'>Calculate normalization factors from house-keeping genes.</h2><span id='topic+BridgeRNormalizationFactorsHK'></span>

<h3>Description</h3>

<p><code>BridgeRNormalizationFactorsHK</code> calculates the normalization factors
from house-keeping genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BridgeRNormalizationFactorsHK(inputFile, group = c("Control", "Knockdown"),
  hour = c(0, 1, 2, 4, 8, 12), inforColumn = 4,
  inforHKGenesRow = "symbol", HKGenes = c("GAPDH", "PGK1", "PPIA", "ENO1",
  "ATP5B", "ALDOA"), save = T, factorOutputPrefix = "BridgeR_3")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BridgeRNormalizationFactorsHK_+3A_inputfile">inputFile</code></td>
<td>
<p>The vector of tab-delimited matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRNormalizationFactorsHK_+3A_group">group</code></td>
<td>
<p>The vector of group names.</p>
</td></tr>
<tr><td><code id="BridgeRNormalizationFactorsHK_+3A_hour">hour</code></td>
<td>
<p>The vector of time course about BRIC-seq experiment.</p>
</td></tr>
<tr><td><code id="BridgeRNormalizationFactorsHK_+3A_inforcolumn">inforColumn</code></td>
<td>
<p>The number of information columns.</p>
</td></tr>
<tr><td><code id="BridgeRNormalizationFactorsHK_+3A_inforhkgenesrow">inforHKGenesRow</code></td>
<td>
<p>The column number of house-keeping gene information.</p>
</td></tr>
<tr><td><code id="BridgeRNormalizationFactorsHK_+3A_hkgenes">HKGenes</code></td>
<td>
<p>The vector of house-keeping genes.</p>
</td></tr>
<tr><td><code id="BridgeRNormalizationFactorsHK_+3A_save">save</code></td>
<td>
<p>Whether to save the output matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRNormalizationFactorsHK_+3A_factoroutputprefix">factorOutputPrefix</code></td>
<td>
<p>The prefix for the name of factor output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table object about normalization factor calculated by house-keeping genes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(data.table)
rpkm_matrix &lt;- data.table(gr_id = c(8, 9, 14),
                          symbol = c("AAAS", "AACS", "AADAT"),
                          accession_id = c("NM_015665", "NM_023928", "NM_182662"),
                          locus = c("chr12", "chr12", "chr4"),
                          CTRL_1_0h = c(41, 5, 5),
                          CTRL_1_1h = c(48, 7, 6),
                          CTRL_1_2h = c(56, 10, 6),
                          CTRL_1_4h = c(87, 12, 10),
                          CTRL_1_8h = c(124, 20, 11),
                          CTRL_1_12h = c(185, 22, 15),
                          gr_id = c(8, 9, 14),
                          symbol = c("AAAS", "AACS", "AADAT"),
                          accession_id = c("NM_015665", "NM_023928", "NM_182662"),
                          locus = c("chr12", "chr12", "chr4"),
                          KD_1_0h = c(21, 10, 3),
                          KD_1_1h = c(33, 11, 3),
                          KD_1_2h = c(42, 15, 4),
                          KD_1_4h = c(60, 20, 5),
                          KD_1_8h = c(65, 37, 6),
                          KD_1_12h = c(70, 42, 6))
group &lt;- c("Control", "Knockdown")
hour &lt;- c(0, 1, 2, 4, 8, 12)
rpkm_list &lt;- BridgeRDataSetFromMatrix(inputFile = rpkm_matrix,
                                      group = group,
                                      hour = hour,
                                      cutoff = 0.1,
                                      inforColumn = 4,
                                      save = FALSE)
raw_table &lt;- rpkm_list[[1]]
test_table &lt;- rpkm_list[[2]]
factor_table &lt;- BridgeRNormalizationFactorsHK(test_table,
                                              save = FALSE)

</code></pre>

<hr>
<h2 id='BridgeRPvalueEvaluation'>Calculate Fold-change of RNA half-life and p-value.</h2><span id='topic+BridgeRPvalueEvaluation'></span>

<h3>Description</h3>

<p><code>BridgeRPvalueEvaluation</code> calculates the fold-change of RNA half-life
and p-value between two conditions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BridgeRPvalueEvaluation(inputFile, group = c("Control", "Knockdown"),
  hour = c(0, 1, 2, 4, 8, 12), comparisonFile = c("Control", "Knockdown"),
  inforColumn = 4, CutoffTimePointNumber = 4, calibration = FALSE,
  save = TRUE, outputPrefix = "BridgeR_6")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BridgeRPvalueEvaluation_+3A_inputfile">inputFile</code></td>
<td>
<p>The vector of tab-delimited matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRPvalueEvaluation_+3A_group">group</code></td>
<td>
<p>The vector of group names.</p>
</td></tr>
<tr><td><code id="BridgeRPvalueEvaluation_+3A_hour">hour</code></td>
<td>
<p>The vector of time course about BRIC-seq experiment.</p>
</td></tr>
<tr><td><code id="BridgeRPvalueEvaluation_+3A_comparisonfile">comparisonFile</code></td>
<td>
<p>The vector of group names.</p>
</td></tr>
<tr><td><code id="BridgeRPvalueEvaluation_+3A_inforcolumn">inforColumn</code></td>
<td>
<p>The number of information columns.</p>
</td></tr>
<tr><td><code id="BridgeRPvalueEvaluation_+3A_cutofftimepointnumber">CutoffTimePointNumber</code></td>
<td>
<p>The number of minimum time points for calc.</p>
</td></tr>
<tr><td><code id="BridgeRPvalueEvaluation_+3A_calibration">calibration</code></td>
<td>
<p>Calibration of RNA half-life.</p>
</td></tr>
<tr><td><code id="BridgeRPvalueEvaluation_+3A_save">save</code></td>
<td>
<p>Whether to save the output matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRPvalueEvaluation_+3A_outputprefix">outputPrefix</code></td>
<td>
<p>The prefix for the name of the output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table object about Fold-change of RNA half-lives, p-value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(data.table)
normalized_rpkm_matrix &lt;- data.table(gr_id = c(8, 9, 14),
                                     symbol = c("AAAS", "AACS", "AADAT"),
                                     accession_id = c("NM_015665", "NM_023928", "NM_182662"),
                                     locus = c("chr12", "chr12", "chr4"),
                                     CTRL_1_0h = c(1.00, 1.00, 1.00),
                                     CTRL_1_1h = c(1.00, 0.86, 0.96),
                                     CTRL_1_2h = c(1.00, 0.96, 0.88),
                                     CTRL_1_4h = c(1.00, 0.74, 0.85),
                                     CTRL_1_8h = c(1.00, 0.86, 0.68),
                                     CTRL_1_12h = c(1.01, 0.65, 0.60),
                                     gr_id = c(8, 9, 14),
                                     symbol = c("AAAS", "AACS", "AADAT"),
                                     accession_id = c("NM_015665", "NM_023928", "NM_182662"),
                                     locus = c("chr12", "chr12", "chr4"),
                                     KD_1_0h = c(1.00, 1.00, 1.00),
                                     KD_1_1h = c(1.01, 0.73, 0.71),
                                     KD_1_2h = c(1.01, 0.77, 0.69),
                                     KD_1_4h = c(1.01, 0.72, 0.67),
                                     KD_1_8h = c(1.01, 0.64, 0.38),
                                     KD_1_12h = c(1.00, 0.89, 0.63))
group &lt;- c("Control", "Knockdown")
hour &lt;- c(0, 1, 2, 4, 8, 12)
halflife_table &lt;- BridgeRHalfLifeCalcR2Select(normalized_rpkm_matrix,
                                              group = group,
                                              hour = hour,
                                              save = FALSE)
pvalue_table &lt;- BridgeRPvalueEvaluation(halflife_table,
                                        group = group,
                                        hour = hour,
                                        save = FALSE)

</code></pre>

<hr>
<h2 id='BridgeRResultChecker'>BRIC-seq result checker</h2><span id='topic+BridgeRResultChecker'></span>

<h3>Description</h3>

<p><code>BridgeRResultChecker</code> returns several BRIC-seq result information.
This function is used for checking the distribution of genome-wide RNA half-lives.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BridgeRResultChecker(inputFile, group = c("Control", "Knockdown"),
  hour = c(0, 1, 2, 4, 8, 12), inforColumn = 4, save = T,
  outputPrefix = "BridgeR_9")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BridgeRResultChecker_+3A_inputfile">inputFile</code></td>
<td>
<p>The vector of tab-delimited matrix file.</p>
</td></tr>
<tr><td><code id="BridgeRResultChecker_+3A_group">group</code></td>
<td>
<p>The vector of group names.</p>
</td></tr>
<tr><td><code id="BridgeRResultChecker_+3A_hour">hour</code></td>
<td>
<p>The vector of time course about BRIC-seq experiment.</p>
</td></tr>
<tr><td><code id="BridgeRResultChecker_+3A_inforcolumn">inforColumn</code></td>
<td>
<p>The number of information columns.</p>
</td></tr>
<tr><td><code id="BridgeRResultChecker_+3A_save">save</code></td>
<td>
<p>Whether to save the output fig file.</p>
</td></tr>
<tr><td><code id="BridgeRResultChecker_+3A_outputprefix">outputPrefix</code></td>
<td>
<p>The prefix for the name of the output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list object about ggplot2 fig data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(data.table)
normalized_table &lt;- data.table(gr_id = c(8, 9, 14),
                               symbol = c("AAAS", "AACS", "AADAT"),
                               accession_id = c("NM_015665", "NM_023928", "NM_182662"),
                               locus = c("chr12", "chr12", "chr4"),
                               CTRL_1_0h = c(1.00, 1.00, 1.00),
                               CTRL_1_1h = c(1.00, 0.86, 0.96),
                               CTRL_1_2h = c(1.00, 0.96, 0.88),
                               CTRL_1_4h = c(1.00, 0.74, 0.85),
                               CTRL_1_8h = c(1.00, 0.86, 0.68),
                               CTRL_1_12h = c(1.01, 0.65, 0.60),
                               gr_id = c(8, 9, 14),
                               symbol = c("AAAS", "AACS", "AADAT"),
                               accession_id = c("NM_015665", "NM_023928", "NM_182662"),
                               locus = c("chr12", "chr12", "chr4"),
                               KD_1_0h = c(1.00, 1.00, 1.00),
                               KD_1_1h = c(1.01, 0.73, 0.71),
                               KD_1_2h = c(1.01, 0.77, 0.69),
                               KD_1_4h = c(1.01, 0.72, 0.67),
                               KD_1_8h = c(1.01, 0.64, 0.38),
                               KD_1_12h = c(1.00, 0.89, 0.63))
group &lt;- c("Control", "Knockdown")
hour &lt;- c(0, 1, 2, 4, 8, 12)
halflife_table &lt;- BridgeRHalfLifeCalcR2Select(normalized_table,
                                              group = group,
                                              hour = hour,
                                              save = FALSE)
pvalue_table &lt;- BridgeRPvalueEvaluation(halflife_table,
                                        group = group,
                                        hour = hour,
                                        save = FALSE)
result_fig &lt;- BridgeRResultChecker(pvalue_table,
                                   save = FALSE)

</code></pre>

<hr>
<h2 id='CalcHalflifeDeviation'>Calculate RNA half-life SD.</h2><span id='topic+CalcHalflifeDeviation'></span>

<h3>Description</h3>

<p><code>BridgeRHalfLifeCalcR2Select</code> calculates RPKM SD and RNA half-life SD
for each gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CalcHalflifeDeviation(inputFile, rawFile, group = c("CTRL_PUM1", "CTRL_PUM2",
  "CTRL_DKD"), hour = c(0, 1, 2, 4, 8, 12), save = T, figSave = F,
  inforColumn = 4, outputPrefix = "BridgeR_7")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CalcHalflifeDeviation_+3A_inputfile">inputFile</code></td>
<td>
<p>The dataframe of halflife table.</p>
</td></tr>
<tr><td><code id="CalcHalflifeDeviation_+3A_rawfile">rawFile</code></td>
<td>
<p>The dataframe of RPKM table.</p>
</td></tr>
<tr><td><code id="CalcHalflifeDeviation_+3A_group">group</code></td>
<td>
<p>The vector of group names.</p>
</td></tr>
<tr><td><code id="CalcHalflifeDeviation_+3A_hour">hour</code></td>
<td>
<p>The vector of time course about BRIC-seq experiment.</p>
</td></tr>
<tr><td><code id="CalcHalflifeDeviation_+3A_save">save</code></td>
<td>
<p>Whether to save the output matrix file.</p>
</td></tr>
<tr><td><code id="CalcHalflifeDeviation_+3A_figsave">figSave</code></td>
<td>
<p>Whether to save the output fig file.</p>
</td></tr>
<tr><td><code id="CalcHalflifeDeviation_+3A_inforcolumn">inforColumn</code></td>
<td>
<p>The number of information columns.</p>
</td></tr>
<tr><td><code id="CalcHalflifeDeviation_+3A_outputprefix">outputPrefix</code></td>
<td>
<p>The prefix for the name of the output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table object about RNA half-life SD.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>group &lt;- c("Control", "Knockdown")
hour &lt;- c(0, 1, 2, 4, 8, 12)
half_sd_table &lt;- CalcHalflifeDeviation(halflife_table,
                                       RNA_halflife_grubbs_test,
                                       group = c("CTRL_1",
                                                 "CTRL_2",
                                                 "CTRL_3"),
                                       save = FALSE)

</code></pre>

<hr>
<h2 id='halflife_table'>BRIC-seq result dataset for p-value estimation using grubbs test</h2><span id='topic+halflife_table'></span>

<h3>Description</h3>

<p>A dataset containing the RPKM for each time point, information column,
RNA half-life, R2 and fitting model about 200 genes.
The variables are as follows:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>halflife_table
</code></pre>


<h3>Format</h3>

<p>A data frame with 200 rows and 52 variables:
</p>

<dl>
<dt>gr_id</dt><dd><p>Group id</p>
</dd>
<dt>symbol</dt><dd><p>Gene symbol</p>
</dd>
<dt>accession_id</dt><dd><p>Gene accession id (RefSeq)</p>
</dd>
<dt>locus</dt><dd><p>Genome locus</p>
</dd>
<dt>T00_1</dt><dd><p>RPKM value at 0h in control condition</p>
</dd>
<dt>T01_1</dt><dd><p>RPKM value at 1h in control condition</p>
</dd>
<dt>T02_1</dt><dd><p>RPKM value at 2h in control condition</p>
</dd>
<dt>T04_1</dt><dd><p>RPKM value at 4h in control condition</p>
</dd>
<dt>T08_1</dt><dd><p>RPKM value at 8h in control condition</p>
</dd>
<dt>T12_1</dt><dd><p>RPKM value at 12h in control condition</p>
</dd>
<dt>Model</dt><dd><p>RNA decay fitting model</p>
</dd>
<dt>R2</dt><dd><p>R2 for fitting curve</p>
</dd>
<dt>half_life</dt><dd><p>RNA half-life</p>
</dd>
<dt>gr_id</dt><dd><p>Group id</p>
</dd>
<dt>symbol</dt><dd><p>Gene symbol</p>
</dd>
<dt>accession_id</dt><dd><p>Gene accession id (RefSeq)</p>
</dd>
<dt>locus</dt><dd><p>Genome locus</p>
</dd>
<dt>T00_2</dt><dd><p>RPKM value at 0h in control condition</p>
</dd>
<dt>T01_2</dt><dd><p>RPKM value at 1h in control condition</p>
</dd>
<dt>T02_2</dt><dd><p>RPKM value at 2h in control condition</p>
</dd>
<dt>T04_2</dt><dd><p>RPKM value at 4h in control condition</p>
</dd>
<dt>T08_2</dt><dd><p>RPKM value at 8h in control condition</p>
</dd>
<dt>T12_2</dt><dd><p>RPKM value at 12h in control condition</p>
</dd>
<dt>Model</dt><dd><p>RNA decay fitting model</p>
</dd>
<dt>R2</dt><dd><p>R2 for fitting curve</p>
</dd>
<dt>half_life</dt><dd><p>RNA half-life</p>
</dd>
<dt>gr_id</dt><dd><p>Group id</p>
</dd>
<dt>symbol</dt><dd><p>Gene symbol</p>
</dd>
<dt>accession_id</dt><dd><p>Gene accession id (RefSeq)</p>
</dd>
<dt>locus</dt><dd><p>Genome locus</p>
</dd>
<dt>T00_3</dt><dd><p>RPKM value at 0h in control condition</p>
</dd>
<dt>T01_3</dt><dd><p>RPKM value at 1h in control condition</p>
</dd>
<dt>T02_3</dt><dd><p>RPKM value at 2h in control condition</p>
</dd>
<dt>T04_3</dt><dd><p>RPKM value at 4h in control condition</p>
</dd>
<dt>T08_3</dt><dd><p>RPKM value at 8h in control condition</p>
</dd>
<dt>T12_3</dt><dd><p>RPKM value at 12h in control condition</p>
</dd>
<dt>Model</dt><dd><p>RNA decay fitting model</p>
</dd>
<dt>R2</dt><dd><p>R2 for fitting curve</p>
</dd>
<dt>half_life</dt><dd><p>RNA half-life</p>
</dd>
<dt>gr_id</dt><dd><p>Group id</p>
</dd>
<dt>symbol</dt><dd><p>Gene symbol</p>
</dd>
<dt>accession_id</dt><dd><p>Gene accession id (RefSeq)</p>
</dd>
<dt>locus</dt><dd><p>Genome locus</p>
</dd>
<dt>T00_4</dt><dd><p>RPKM value at 0h in knockdown condition</p>
</dd>
<dt>T01_4</dt><dd><p>RPKM value at 1h in knockdown condition</p>
</dd>
<dt>T02_4</dt><dd><p>RPKM value at 2h in knockdown condition</p>
</dd>
<dt>T04_4</dt><dd><p>RPKM value at 4h in knockdown condition</p>
</dd>
<dt>T08_4</dt><dd><p>RPKM value at 8h in knockdown condition</p>
</dd>
<dt>T12_4</dt><dd><p>RPKM value at 12h in knockdown condition</p>
</dd>
<dt>Model</dt><dd><p>RNA decay fitting model</p>
</dd>
<dt>R2</dt><dd><p>R2 for fitting curve</p>
</dd>
<dt>half_life</dt><dd><p>RNA half-life</p>
</dd>
</dl>

<hr>
<h2 id='RNA_halflife_comparison'>test BRIC-seq dataset for RNA half-life comparison</h2><span id='topic+RNA_halflife_comparison'></span>

<h3>Description</h3>

<p>A dataset containing the RPKM for each time point and information column
about 200 genes. The variables are as follows:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RNA_halflife_comparison
</code></pre>


<h3>Format</h3>

<p>A data frame with 200 rows and 20 variables:
</p>

<dl>
<dt>gr_id</dt><dd><p>Group id</p>
</dd>
<dt>symbol</dt><dd><p>Gene symbol</p>
</dd>
<dt>accession_id</dt><dd><p>Gene accession id (RefSeq)</p>
</dd>
<dt>locus</dt><dd><p>Genome locus</p>
</dd>
<dt>CTRL_1_0h</dt><dd><p>RPKM value at 0h in control condition</p>
</dd>
<dt>CTRL_1_1h</dt><dd><p>RPKM value at 1h in control condition</p>
</dd>
<dt>CTRL_1_2h</dt><dd><p>RPKM value at 2h in control condition</p>
</dd>
<dt>CTRL_1_4h</dt><dd><p>RPKM value at 4h in control condition</p>
</dd>
<dt>CTRL_1_8h</dt><dd><p>RPKM value at 8h in control condition</p>
</dd>
<dt>CTRL_1_12h</dt><dd><p>RPKM value at 12h in control condition</p>
</dd>
<dt>gr_id</dt><dd><p>Group id</p>
</dd>
<dt>symbol</dt><dd><p>Gene symbol</p>
</dd>
<dt>accession_id</dt><dd><p>Gene accession id (RefSeq)</p>
</dd>
<dt>locus</dt><dd><p>Genome locus</p>
</dd>
<dt>KD_1_0h</dt><dd><p>RPKM value at 0h in knockdown condition</p>
</dd>
<dt>KD_1_1h</dt><dd><p>RPKM value at 1h in knockdown condition</p>
</dd>
<dt>KD_1_2h</dt><dd><p>RPKM value at 2h in knockdown condition</p>
</dd>
<dt>KD_1_4h</dt><dd><p>RPKM value at 4h in knockdown condition</p>
</dd>
<dt>KD_1_8h</dt><dd><p>RPKM value at 8h in knockdown condition</p>
</dd>
<dt>KD_1_12h</dt><dd><p>RPKM value at 12h in knockdown condition</p>
</dd>
</dl>

<hr>
<h2 id='RNA_halflife_comparison_HK'>test BRIC-seq dataset for RNA half-life comparison using House-keeping genes.</h2><span id='topic+RNA_halflife_comparison_HK'></span>

<h3>Description</h3>

<p>A dataset containing the RPKM for each time point and information column
about 200 genes + house-keeping genes. The variables are as follows:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RNA_halflife_comparison_HK
</code></pre>


<h3>Format</h3>

<p>A data frame with 200 rows and 20 variables:
</p>

<dl>
<dt>gr_id</dt><dd><p>Group id</p>
</dd>
<dt>symbol</dt><dd><p>Gene symbol</p>
</dd>
<dt>accession_id</dt><dd><p>Gene accession id (RefSeq)</p>
</dd>
<dt>locus</dt><dd><p>Genome locus</p>
</dd>
<dt>CTRL_1_0h</dt><dd><p>RPKM value at 0h in control condition</p>
</dd>
<dt>CTRL_1_1h</dt><dd><p>RPKM value at 1h in control condition</p>
</dd>
<dt>CTRL_1_2h</dt><dd><p>RPKM value at 2h in control condition</p>
</dd>
<dt>CTRL_1_4h</dt><dd><p>RPKM value at 4h in control condition</p>
</dd>
<dt>CTRL_1_8h</dt><dd><p>RPKM value at 8h in control condition</p>
</dd>
<dt>CTRL_1_12h</dt><dd><p>RPKM value at 12h in control condition</p>
</dd>
<dt>gr_id</dt><dd><p>Group id</p>
</dd>
<dt>symbol</dt><dd><p>Gene symbol</p>
</dd>
<dt>accession_id</dt><dd><p>Gene accession id (RefSeq)</p>
</dd>
<dt>locus</dt><dd><p>Genome locus</p>
</dd>
<dt>KD_1_0h</dt><dd><p>RPKM value at 0h in knockdown condition</p>
</dd>
<dt>KD_1_1h</dt><dd><p>RPKM value at 1h in knockdown condition</p>
</dd>
<dt>KD_1_2h</dt><dd><p>RPKM value at 2h in knockdown condition</p>
</dd>
<dt>KD_1_4h</dt><dd><p>RPKM value at 4h in knockdown condition</p>
</dd>
<dt>KD_1_8h</dt><dd><p>RPKM value at 8h in knockdown condition</p>
</dd>
<dt>KD_1_12h</dt><dd><p>RPKM value at 12h in knockdown condition</p>
</dd>
</dl>

<hr>
<h2 id='RNA_halflife_grubbs_test'>test BRIC-seq dataset for p-value estimation using grubbs test</h2><span id='topic+RNA_halflife_grubbs_test'></span>

<h3>Description</h3>

<p>A dataset containing the RPKM for each time point and information column
about 200 genes. The variables are as follows:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RNA_halflife_grubbs_test
</code></pre>


<h3>Format</h3>

<p>A data frame with 200 rows and 40 variables:
</p>

<dl>
<dt>gr_id</dt><dd><p>Group id</p>
</dd>
<dt>symbol</dt><dd><p>Gene symbol</p>
</dd>
<dt>accession_id</dt><dd><p>Gene accession id (RefSeq)</p>
</dd>
<dt>locus</dt><dd><p>Genome locus</p>
</dd>
<dt>CTRL_1_0h</dt><dd><p>RPKM value at 0h in control condition</p>
</dd>
<dt>CTRL_1_1h</dt><dd><p>RPKM value at 1h in control condition</p>
</dd>
<dt>CTRL_1_2h</dt><dd><p>RPKM value at 2h in control condition</p>
</dd>
<dt>CTRL_1_4h</dt><dd><p>RPKM value at 4h in control condition</p>
</dd>
<dt>CTRL_1_8h</dt><dd><p>RPKM value at 8h in control condition</p>
</dd>
<dt>CTRL_1_12h</dt><dd><p>RPKM value at 12h in control condition</p>
</dd>
<dt>gr_id</dt><dd><p>Group id</p>
</dd>
<dt>symbol</dt><dd><p>Gene symbol</p>
</dd>
<dt>accession_id</dt><dd><p>Gene accession id (RefSeq)</p>
</dd>
<dt>locus</dt><dd><p>Genome locus</p>
</dd>
<dt>CTRL_2_0h</dt><dd><p>RPKM value at 0h in control condition</p>
</dd>
<dt>CTRL_2_1h</dt><dd><p>RPKM value at 1h in control condition</p>
</dd>
<dt>CTRL_2_2h</dt><dd><p>RPKM value at 2h in control condition</p>
</dd>
<dt>CTRL_2_4h</dt><dd><p>RPKM value at 4h in control condition</p>
</dd>
<dt>CTRL_2_8h</dt><dd><p>RPKM value at 8h in control condition</p>
</dd>
<dt>CTRL_2_12h</dt><dd><p>RPKM value at 12h in control condition</p>
</dd>
<dt>gr_id</dt><dd><p>Group id</p>
</dd>
<dt>symbol</dt><dd><p>Gene symbol</p>
</dd>
<dt>accession_id</dt><dd><p>Gene accession id (RefSeq)</p>
</dd>
<dt>locus</dt><dd><p>Genome locus</p>
</dd>
<dt>CTRL_3_0h</dt><dd><p>RPKM value at 0h in control condition</p>
</dd>
<dt>CTRL_3_1h</dt><dd><p>RPKM value at 1h in control condition</p>
</dd>
<dt>CTRL_3_2h</dt><dd><p>RPKM value at 2h in control condition</p>
</dd>
<dt>CTRL_3_4h</dt><dd><p>RPKM value at 4h in control condition</p>
</dd>
<dt>CTRL_3_8h</dt><dd><p>RPKM value at 8h in control condition</p>
</dd>
<dt>CTRL_3_12h</dt><dd><p>RPKM value at 12h in control condition</p>
</dd>
<dt>gr_id</dt><dd><p>Group id</p>
</dd>
<dt>symbol</dt><dd><p>Gene symbol</p>
</dd>
<dt>accession_id</dt><dd><p>Gene accession id (RefSeq)</p>
</dd>
<dt>locus</dt><dd><p>Genome locus</p>
</dd>
<dt>KD_1_0h</dt><dd><p>RPKM value at 0h in knockdown condition</p>
</dd>
<dt>KD_1_1h</dt><dd><p>RPKM value at 1h in knockdown condition</p>
</dd>
<dt>KD_1_2h</dt><dd><p>RPKM value at 2h in knockdown condition</p>
</dd>
<dt>KD_1_4h</dt><dd><p>RPKM value at 4h in knockdown condition</p>
</dd>
<dt>KD_1_8h</dt><dd><p>RPKM value at 8h in knockdown condition</p>
</dd>
<dt>KD_1_12h</dt><dd><p>RPKM value at 12h in knockdown condition</p>
</dd>
</dl>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
