<!DOCTYPE html><html lang="en"><head><title>Help for package c3net</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {c3net}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#c3'><p>C3NET second step: Selecting maximum valued elements</p></a></li>
<li><a href='#c3net'><p>All in one function to infer network with C3NET</p></a></li>
<li><a href='#checknet'><p>Validating the inferred network</p></a></li>
<li><a href='#copula'><p>Copula transformation of data set</p></a></li>
<li><a href='#expdata'><p>Example data set</p></a></li>
<li><a href='#makemim'><p>Constructing mutual information matrix</p></a></li>
<li><a href='#netplot'><p>Plotting the inferred network</p></a></li>
<li><a href='#sigtestMTC'><p>Elimination of nonsignificant edges by application of a procedure for multiple testing correction</p></a></li>
<li><a href='#sigtestp'><p>Significance test for elimination of nonsignificant edges</p></a></li>
<li><a href='#truenet'><p>Reference, e.g. true, network of the example data set</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>1.1.1.1</td>
</tr>
<tr>
<td>Title:</td>
<td>Inferring Large-Scale Gene Networks with C3NET</td>
</tr>
<tr>
<td>Author:</td>
<td>Gokmen Altay, Frank Emmert-Streib</td>
</tr>
<tr>
<td>Description:</td>
<td>Allows inferring gene regulatory networks
        with direct physical interactions from microarray expression
        data using C3NET.</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Gokmen Altay &lt;altayscience@gmail.com&gt;</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.12.1), igraph</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Repository/R-Forge/Project:</td>
<td>c3net</td>
</tr>
<tr>
<td>Repository/R-Forge/Revision:</td>
<td>23</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-06-24 12:29:23 UTC</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-24 12:13:53 UTC; hornik</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
</table>
<hr>
<h2 id='c3'>C3NET second step: Selecting maximum valued elements</h2><span id='topic+c3'></span>

<h3>Description</h3>

<p><code>c3</code> takes the mutual information matrix as input and implements the second step of C3NET - see details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c3( mim, sym = TRUE )</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="c3_+3A_mim">mim</code></td>
<td>
<p>A symmetric square mutual information matrix, where the elements (i,j) correspond to the mutual information <code class="reqn">I(i,j)</code> or <code class="reqn">I(j,i)</code> between variables i and j.</p>
</td></tr>
<tr><td><code id="c3_+3A_sym">sym</code></td>
<td>
<p>Decides the output matrix to be symmetric or not.</p>
</td></tr>  
</table>


<h3>Details</h3>

<p>The C3NET algorithm consists of two main steps. The first step is the same as for relevance networks (RELNET), where all the non-significant mutual information values in the matrix are eliminated if statistically not significant. The second step of C3NET keeps all maximum valued mutual information values for each row in the matrix and sets the rest of the elements in the matrix zero (the diagonal of the matrix is ignored). The output is normally symmetric matrix but if the argument sym is set to FALSE then the output becomes non-symmetric.  
</p>


<h3>Value</h3>

<p><code>c3</code> implements the second step of the C3NET algorithm returning a symmetric mutual information matrix. Specifically, the non-zero elements in the returned matrix represent undirected links between variables which are statistically significant (tested in the first step of C3NET). 
</p>


<h3>References</h3>

<p>G. Altay, F. Emmert-Streib, &quot;Inferring the conservative causal core of gene regulatory networks&quot;, BMC Systems Biology (2010) 4:132.    
</p>


<h3>See Also</h3>

<p><code><a href="#topic+makemim">makemim</a></code>, <code><a href="#topic+copula">copula</a></code>, <code><a href="#topic+sigtestp">sigtestp</a></code>, <code><a href="#topic+sigtestMTC">sigtestMTC</a></code>, <code><a href="#topic+netplot">netplot</a></code>, <code><a href="#topic+c3net">c3net</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expdata)

expdata &lt;- copula(expdata)

mim &lt;- makemim(expdata)

Ic &lt;- mean(mim[upper.tri(mim)])  #Example cut-off for the first step of C3NET

# Ic &lt;- 2 can be set for the example.

mim[mim &lt; Ic] &lt;-0  #nonsignificant values eliminated wrt C3NET step 1.

net &lt;- c3(mim)  # regulatory network inferred (non zero elements stand for links of
		 #the predicted network)
</code></pre>

<hr>
<h2 id='c3net'>All in one function to infer network with C3NET</h2><span id='topic+c3net'></span>

<h3>Description</h3>

<p><code>c3net</code> takes a data set as input and computes the inferred network using C3NET - see details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c3net(dataset, cop=TRUE, alpha=0.01, methodstep1="cutoff", cutoffMI= 0,
	      MTCmethod="BH", itnum=5, network=FALSE)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="c3net_+3A_dataset">dataset</code></td>
<td>
<p>Data set where rows are variables (e.g. genes) and columns are samples.</p>
</td></tr>
<tr><td><code id="c3net_+3A_cop">cop</code></td>
<td>
<p>The data set is copula transformed by default. If not wanted, then &quot;cop&quot; is set to FALSE.</p>
</td></tr>
<tr><td><code id="c3net_+3A_alpha">alpha</code></td>
<td>
<p>Statistical significance threshold</p>
</td></tr>
<tr><td><code id="c3net_+3A_itnum">itnum</code></td>
<td>
<p>Number of iterations to resample data to get sampling distribution.</p>
</td></tr>
<tr><td><code id="c3net_+3A_methodstep1">methodstep1</code></td>
<td>
<p>If it is &quot;cutoff&quot; then the input cutoffMI is used as threshold for elimination in step 1. If it is &quot;MTC&quot; then MTCmethod input is used as the specific multiple testing correction method to employ. If it is &quot;justp&quot; then only significance threshold alpha is used to compute significance threshold MI.</p>
</td></tr>
<tr><td><code id="c3net_+3A_mtcmethod">MTCmethod</code></td>
<td>
<p>Available option are &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;.</p>
</td></tr>
<tr><td><code id="c3net_+3A_cutoffmi">cutoffMI</code></td>
<td>
<p>If methodstep1 is set to &quot;cutoff&quot; then this value is used for MI threshold. If this is set to 0 (or by default) it uses mean MI for threshold.</p>
</td></tr>
<tr><td><code id="c3net_+3A_network">network</code></td>
<td>
<p>If TRUE, the network is plotted in the end.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For Step 1 of C3NET, there are three options that can be used. If methodstep1=&quot;cutoff&quot; then cutoffMI input is taken as the significance threshold for step 1. In case cutoffMI is set to 0 then by default mean MI is taken as cutoffMI. If methodstep1 is set to &quot;MTC&quot; then the method from input MTCmethod, itnum for iteration number to get null distribution and alpha for statistical significance is taken for computations. If methodstep1 is set to &quot;justp&quot; or anything else then just alpha and itnum is used for computations. 
</p>


<h3>Value</h3>

<p><code>c3net</code> returns a symmetric mutual information matrix, which is obtained after implementing C3NET. Specifically, non-zero elements in the returned matrix represents undirected link between variables. The inferred network may also be plotted if the argument network is set TRUE. 
</p>


<h3>References</h3>

<p>G. Altay, F. Emmert-Streib, &quot;Inferring the conservative causal core of gene regulatory networks&quot;, BMC Systems Biology (2010) 4:132.    
</p>


<h3>See Also</h3>

<p><code><a href="#topic+makemim">makemim</a></code>, <code><a href="#topic+copula">copula</a></code>, <code><a href="#topic+c3">c3</a></code>, <code><a href="#topic+sigtestp">sigtestp</a></code>, <code><a href="#topic+sigtestMTC">sigtestMTC</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expdata)
data(truenet)

net &lt;- c3net(expdata, network=TRUE)
scores &lt;- checknet(net,truenet)
</code></pre>

<hr>
<h2 id='checknet'>Validating the inferred network</h2><span id='topic+checknet'></span>

<h3>Description</h3>

<p><code>checknet</code> takes the inferred mutual information matrix and the true network for validation. - see details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checknet( finalrelationmatrix, realrelationmatrix)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checknet_+3A_finalrelationmatrix">finalrelationmatrix</code></td>
<td>
<p>The inferred symmetric mutual information matrix, where i,j th element is the mutual information <code class="reqn">I(i,j)</code> or <code class="reqn">I(j,i)</code> between two variables i and j. The diagonal is set to zero.</p>
</td></tr>
<tr><td><code id="checknet_+3A_realrelationmatrix">realrelationmatrix</code></td>
<td>
<p>A symmetric reference connection matrix, where a 1 at i,j th element defines the connection between variables i and j  and non-connection is represented by 0. The diagonal is all zero.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>checknet</code> returns a vector with 6 elements, that contains the parameters as output &lt;- c(precision, Fscore, recall, TP, FP, FN). 
</p>


<h3>References</h3>

<p>G. Altay, F. Emmert-Streib, &quot;Inferring the conservative causal core of gene regulatory networks&quot;, BMC Systems Biology, (2010) 4:132.    
</p>


<h3>See Also</h3>

<p><code><a href="#topic+makemim">makemim</a></code>, <code><a href="#topic+copula">copula</a></code>, <code><a href="#topic+c3">c3</a></code>, <code><a href="#topic+c3net">c3net</a></code>, <code><a href="#topic+c3">c3</a></code>, <code><a href="#topic+makemim">makemim</a></code>, <code><a href="#topic+copula">copula</a></code>, <code><a href="#topic+sigtestp">sigtestp</a></code>, <code><a href="#topic+sigtestMTC">sigtestMTC</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expdata)

data(truenet)

expdata &lt;- copula(expdata)

mim &lt;- makemim(expdata)

Ic &lt;- 2  #Example cut-off value for mutual information for the first step of C3NET

mim[mim &lt; Ic] &lt;-0  #nonsignificant values eliminated wrt C3NET step 1.

net &lt;- c3(mim)  # regulatory network inferred (non zero elements stand for links of
		# the predicted network)

scores &lt;- checknet(net,truenet)
</code></pre>

<hr>
<h2 id='copula'>Copula transformation of data set</h2><span id='topic+copula'></span>

<h3>Description</h3>

<p><code>copula</code> takes a data set as input and applies a copula transformation - see details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>copula( expdata )</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="copula_+3A_expdata">expdata</code></td>
<td>
<p>Data set; rows correspond to variables (e.g. genes) and columns to samples.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The data set can be copula transformed for more stable estimations of the mutual information matrix.  Each row of the data set (gene samples of each gene) is copula transformed and obtained rdata. Here, the formula for conversion is copuladata =(rdata- 0.5)/numberofsamples. 
</p>


<h3>Value</h3>

<p><code>copula</code> returns a copula transformed data set. 
</p>


<h3>References</h3>

<p>G. Altay, F. Emmert-Streib, &quot;Inferring the conservative causal core of gene regulatory networks&quot;, BMC Systems Biology (2010) 4:132.    
</p>


<h3>See Also</h3>

<p><code><a href="#topic+c3">c3</a></code>, <code><a href="#topic+c3net">c3net</a></code>, <code><a href="#topic+makemim">makemim</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expdata)

expdata &lt;- copula(expdata)
</code></pre>

<hr>
<h2 id='expdata'>Example data set</h2><span id='topic+expdata'></span>

<h3>Description</h3>

<p><code>expdata</code> This data set has been generated by SynTRen. The source network corresponds to a subnetwork of E.coli (file name &quot;EColi_full_HongWu_Ma_NAR2004.sif&quot; from Ma HW, et. al. Nuc. Ac. Res. 2004). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data( expdata )</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="expdata_+3A_expdata">expdata</code></td>
<td>
<p>Example data; rows correspond to variables (e.g. genes) and columns to samples.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The example is also given to provide a sample data set so that one can take it as reference while formatting the new data set as entry to c3net. Data sets must be complete and no empty or non-numerical character allowed.  
</p>


<h3>Value</h3>

<p><code>expdata</code> returns an example data set where rows are variables (e.g. genes) and columns are samples. It consists of 400 genes, 898 edges, and 800 steady-state samples. 
</p>


<h3>References</h3>

<p>Ma HW, Kumar B, Ditges U, Gunzer F, Buer J, Zeng AP:  An extended transcriptional regulatory network of Escherichia coli and analysis of its hierarchical structure and network motifs. Nucleic Acids Res 2004 , 32:6643-6649
</p>
<p>Bulcke, T.V., Leemput, K.V., Naudts, B., et. al. (2006) Syntren: a generator of synthetic gene expression data for design and analysis of structure learning algorithms. BMC Bioinformatics, 7(1), 43.
</p>
<p>G. Altay, F. Emmert-Streib, &quot;Inferring the conservative causal core of gene regulatory networks&quot;, BMC Systems Biology (2010) 4:132.    
</p>


<h3>See Also</h3>

<p><code><a href="#topic+c3">c3</a></code>, <code><a href="#topic+c3net">c3net</a></code>, <code><a href="#topic+makemim">makemim</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expdata)
</code></pre>

<hr>
<h2 id='makemim'>Constructing mutual information matrix </h2><span id='topic+makemim'></span>

<h3>Description</h3>

<p><code>makemim</code> takes the data set as input and computes mutual information values for each pair of variables (e.g. gene). - see details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makemim( expdata )</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="makemim_+3A_expdata">expdata</code></td>
<td>
<p>Data matrix where rows correspond to variables (e.g. genes) and columns to samples.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A mutual information matrix is generated from a data set using an empirical Gaussian estimator.   
</p>


<h3>Value</h3>

<p><code>makemim</code> returns a symmetric mutual information matrix using empirical Gaussian estimator. 
</p>


<h3>References</h3>

<p>G. A. Darbellay, I. Vajda, &quot;Estimation of the information by an adaptive partitioning of the observation space&quot;, IEEE Trans. Inf. Theory vol.45 (1999)	
</p>
<p>G. Altay, F. Emmert-Streib, &quot;Inferring the conservative causal core of gene regulatory networks&quot;, BMC Systems Biology, (2010) 4:132.    
</p>


<h3>See Also</h3>

<p><code><a href="#topic+c3">c3</a></code>,<code><a href="#topic+c3net">c3net</a></code>, <code><a href="#topic+copula">copula</a></code>, <code><a href="#topic+checknet">checknet</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expdata)

expdata &lt;- copula(expdata)

mim &lt;- makemim(expdata)

</code></pre>

<hr>
<h2 id='netplot'>Plotting the inferred network</h2><span id='topic+netplot'></span>

<h3>Description</h3>

<p><code>netplot</code> takes the inferred mutual information matrix and plots a network wrt nonzero elements of the matrix - see details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>netplot(gnet)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="netplot_+3A_gnet">gnet</code></td>
<td>
<p>A symmetric squared mutual information matrix that is inferred as final connection matrix, where i,j th element is the mutual information <code class="reqn">I(i,j)</code> or <code class="reqn">I(j,i)</code> between variables i and j. The nonzero elements may also be 1. Diagonal is all zero.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A labelled undirected network is plotted wrt the row (column) names of the input matrix. When plotted, select &quot;Layout&quot; and then &quot;Fruchterman-Reingold&quot; and &quot;OK&quot; to get a better view. To remove labels, select &quot;View&quot; and deselect &quot;Labels&quot;. To change the colors or size of nodes and edges, from &quot;Select&quot;, click on &quot;all edges&quot; or &quot;vertices&quot; and on the network right click on nodes and change color or size.     
</p>


<h3>Value</h3>

<p><code>netplot</code> returns an undirected network. 
</p>


<h3>References</h3>

<p>G. Altay, F. Emmert-Streib, &quot;Inferring the conservative causal core of gene regulatory networks&quot;, BMC Systems Biology (2010) 4:132.    
</p>
<p>CSARDI G, NEPUSZ T.: igraph-package for R, 2008
</p>


<h3>See Also</h3>

<p><code><a href="#topic+makemim">makemim</a></code>, <code><a href="#topic+copula">copula</a></code>, <code><a href="#topic+c3">c3</a></code>, <code><a href="#topic+sigtestp">sigtestp</a></code>, <code><a href="#topic+sigtestMTC">sigtestMTC</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expdata)

data(truenet)

expdata &lt;- copula(expdata)

mim &lt;- makemim(expdata)

Ic &lt;- 2  #Example cut-off for the first step of C3NET

mim[mim &lt; Ic] &lt;-0  #nonsignificant values eliminated wrt C3NET step 1.

net &lt;- c3(mim)  # regulatory network inferred (non zero elements stand for links of
		# the predicted network)

netplot(net)
</code></pre>

<hr>
<h2 id='sigtestMTC'>Elimination of nonsignificant edges by application of a procedure for multiple testing correction</h2><span id='topic+sigtestMTC'></span>

<h3>Description</h3>

<p><code>sigtestMTC</code> takes a data set as input and computes the mutual information matrix in which the nonsignificant elements are eliminated by a multiple testing correction - see details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sigtestMTC( data, alpha, itnum, methodsig="BH")</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sigtestMTC_+3A_data">data</code></td>
<td>
<p>Data set where rows correspond to variables (e.g. genes) and columns to samples.</p>
</td></tr>
<tr><td><code id="sigtestMTC_+3A_alpha">alpha</code></td>
<td>
<p>Significance level</p>
</td></tr>
<tr><td><code id="sigtestMTC_+3A_itnum">itnum</code></td>
<td>
<p>Number of iterations to resample data to estimate sampling distribution.</p>
</td></tr>
<tr><td><code id="sigtestMTC_+3A_methodsig">methodsig</code></td>
<td>
<p>A procedure to perform a multiple testing correction, either controling the false discovery rate (FDR) or the family-wise error. Available option are &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The data set is resampled for each iteration and at the end the sampling distribution is obtained.   
</p>


<h3>Value</h3>

<p><code>sigtestMTC</code> returns an environment res that contains the new MI matrix, res$Inew, that is obtained after the elimination of nonsignificant elements with a MTC procedure. For advanced users, who are able to write code in R, we provided the variables obtained in the function so that one might want to make further analysis without running the function again. Please see the source code for the additional return values of the environment res. 
</p>


<h3>References</h3>

<p>G. Altay, F. Emmert-Streib, &quot;Inferring the conservative causal core of gene regulatory networks&quot;, BMC Systems Biology (2010) 4:132.    
</p>


<h3>See Also</h3>

<p><code><a href="#topic+makemim">makemim</a></code>, <code><a href="#topic+copula">copula</a></code>, <code><a href="#topic+c3">c3</a></code>, <code><a href="#topic+sigtestp">sigtestp</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expdata)

data(truenet)

alpha &lt;- 0.001

itnum &lt;-2

res &lt;- sigtestMTC( expdata, alpha, itnum, methodsig="BH")

net &lt;- c3(res$Inew)  # regulatory network inferred (non zero elements stand for links of
		# the predicted network)

scores &lt;- checknet(net,truenet)
</code></pre>

<hr>
<h2 id='sigtestp'>Significance test for elimination of nonsignificant edges</h2><span id='topic+sigtestp'></span>

<h3>Description</h3>

<p><code>sigtestp</code> takes the data set as input and computes the mutual information matrix in which the nonsignificant elements are eliminated by using a significance threshold only (no MTC) - see details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sigtestp( data, alpha, itnum)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sigtestp_+3A_data">data</code></td>
<td>
<p>Data set where rows are variables (e.g. genes) and columns are samples.</p>
</td></tr>
<tr><td><code id="sigtestp_+3A_alpha">alpha</code></td>
<td>
<p>Statistical significance threshold</p>
</td></tr>
<tr><td><code id="sigtestp_+3A_itnum">itnum</code></td>
<td>
<p>Number of iterations to resample data to get sampling distribution.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The data set is resampled completely  at each iteration and at the end the sampling distribution is obtained for using in significance test.   
</p>


<h3>Value</h3>

<p><code>sigtestp</code> returns environment res that contains the obtained threshold value res$I0. For advanced users, who are able to write code in R, we provided the variables obtained in the function so that one might want to make further analysis without running the function again. Please see the source code for the additional return values of the environment res. 
</p>


<h3>References</h3>

<p>G. Altay, F. Emmert-Streib, &quot;Inferring the conservative causal core of gene regulatory networks&quot;, BMC Systems Biology (2010) 4:132.    
</p>


<h3>See Also</h3>

<p><code><a href="#topic+makemim">makemim</a></code>, <code><a href="#topic+copula">copula</a></code>, <code><a href="#topic+c3">c3</a></code>,<code><a href="#topic+c3">c3</a></code>, <code><a href="#topic+sigtestMTC">sigtestMTC</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expdata)

data(truenet)

alpha &lt;- 0.001

itnum &lt;- 2

res &lt;- sigtestp( expdata, alpha, itnum)

net &lt;- c3(res$Inew)  # regulatory network inferred (non zero elements stand for links of
		# the predicted network)

scores &lt;- checknet(net,truenet)
</code></pre>

<hr>
<h2 id='truenet'>Reference, e.g. true, network of the example data set</h2><span id='topic+truenet'></span>

<h3>Description</h3>

<p><code>truenet</code> A symmetric matrix, where non-zero elements correspond to true connections. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data( truenet )</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="truenet_+3A_truenet">truenet</code></td>
<td>
<p>Reference (true) network of the example data set.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A symmetric matrix, where non-zero elements correspond to true connections.  
</p>


<h3>Value</h3>

<p><code>truenet</code> Reference (true) network of the example data set. 
</p>


<h3>References</h3>

<p>G. Altay, F. Emmert-Streib, &quot;Inferring the conservative causal core of gene regulatory networks&quot;, BMC Systems Biology (2010) 4:132.    
</p>


<h3>See Also</h3>

<p><code><a href="#topic+c3">c3</a></code>, <code><a href="#topic+c3net">c3net</a></code>, <code><a href="#topic+makemim">makemim</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(truenet)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
