<!DOCTYPE html><html><head><title>Help for package CACIMAR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CACIMAR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CACIMAR_cols'><p>CACIMAR colors palette</p></a></li>
<li><a href='#Format_Markers_Frac'><p>Format marker genes for plotting</p></a></li>
<li><a href='#Heatmap_Cor'><p>plot the heatmap of marker genes across different species</p></a></li>
<li><a href='#Identify_CellType'><p>Identify cell type of each cluster</p></a></li>
<li><a href='#Identify_ConservedCellTypes'><p>Identify conserved cell types based on power of genes and orthologs database</p></a></li>
<li><a href='#Identify_ConservedMarkers'><p>Identify orthologs marker genes for two species</p></a></li>
<li><a href='#Identify_ConservedNetworks'><p>Identify conserved regulatory networks</p></a></li>
<li><a href='#Identify_Markers'><p>Identify markers of each cluster</p></a></li>
<li><a href='#OrthG_Hs_Ch'><p>Orthologs genes database for homo sapiens and zebrafish</p></a></li>
<li><a href='#OrthG_Hs_Mm'><p>Orthologs genes database for homo sapiens and mus musculus</p></a></li>
<li><a href='#OrthG_Hs_Zf'><p>Orthologs genes database for homo sapiens and zebrafish</p></a></li>
<li><a href='#OrthG_Mm_Ch'><p>Orthologs genes database for mus musculus and chicken</p></a></li>
<li><a href='#OrthG_Mm_Zf'><p>Orthologs genes database for mus musculus and zebrafish</p></a></li>
<li><a href='#OrthG_Zf_Ch'><p>Orthologs genes database for mus zebrafish and chicken</p></a></li>
<li><a href='#Plot_MarkersHeatmap'><p>Plot Markers in each cell type</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Cross-Species Analysis of Cell Identities, Markers and
Regulations</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Junyao Jiang &lt;jiangjunyao789@163.com&gt;</td>
</tr>
<tr>
<td>Author:</td>
<td>Junyao Jiang &lt;jiangjunyao789@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A toolkit to perform cross-species analysis based on scRNA-seq data. This package contains 5 main features. (1) identify Markers in each cluster. (2) Cell type annotation (3) identify conserved markers. (4) identify conserved cell types. (5) identify conserved modules of regulatory networks.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>pheatmap, pbapply, psych, ROCR, reshape2, dplyr, grDevices,
stats, methods, viridisLite</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0), Seurat</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/jiang-junyao/CACIMAR">https://github.com/jiang-junyao/CACIMAR</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/jiang-junyao/CACIMAR/issues">https://github.com/jiang-junyao/CACIMAR/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-17 12:49:03 UTC; jiangjunyao</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-18 08:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='CACIMAR_cols'>CACIMAR colors palette</h2><span id='topic+CACIMAR_cols'></span>

<h3>Description</h3>

<p>CACIMAR colors palette
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CACIMAR_cols(color_number)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CACIMAR_cols_+3A_color_number">color_number</code></td>
<td>
<p>numeric, indicating used colors number</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of colors
</p>


<h3>Examples</h3>

<pre><code class='language-R'>CACIMAR_cols(10)
CACIMAR_cols(20)
</code></pre>

<hr>
<h2 id='Format_Markers_Frac'>Format marker genes for plotting</h2><span id='topic+Format_Markers_Frac'></span>

<h3>Description</h3>

<p>Order the gene expression in each cluster to make the heatmap
look better
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Format_Markers_Frac(Marker_genes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Format_Markers_Frac_+3A_marker_genes">Marker_genes</code></td>
<td>
<p>data.frame, generated by <code><a href="#topic+Identify_Markers">Identify_Markers</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Markers corresponding to certain cluster
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("pbmc_small")
all.markers &lt;- Identify_Markers(pbmc_small)
all.markers2 &lt;- Format_Markers_Frac(all.markers)
</code></pre>

<hr>
<h2 id='Heatmap_Cor'>plot the heatmap of marker genes across different species</h2><span id='topic+Heatmap_Cor'></span>

<h3>Description</h3>

<p>plot the heatmap of marker genes across different species
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Heatmap_Cor(
  RNA1,
  RowType1 = "",
  ColType1 = "",
  cluster_cols = TRUE,
  cluster_rows = FALSE,
  Color1 = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Heatmap_Cor_+3A_rna1">RNA1</code></td>
<td>
<p>correlation of expression in each cell type</p>
</td></tr>
<tr><td><code id="Heatmap_Cor_+3A_rowtype1">RowType1</code></td>
<td>
<p>character, indicating the cell types that you want to show
on the row in heatmap. RowType1=&rdquo; means show all cell types</p>
</td></tr>
<tr><td><code id="Heatmap_Cor_+3A_coltype1">ColType1</code></td>
<td>
<p>character, indicating the cell types that you want to show
on the column in heatmap. RowType1=&rdquo; means show all cell types</p>
</td></tr>
<tr><td><code id="Heatmap_Cor_+3A_cluster_cols">cluster_cols</code></td>
<td>
<p>boolean values determining if columns should be clustered
or hclust object</p>
</td></tr>
<tr><td><code id="Heatmap_Cor_+3A_cluster_rows">cluster_rows</code></td>
<td>
<p>boolean values determining if rows should be clustered or
</p>
<p>hclust object</p>
</td></tr>
<tr><td><code id="Heatmap_Cor_+3A_color1">Color1</code></td>
<td>
<p>vector of colors used in heatmap</p>
</td></tr>
<tr><td><code id="Heatmap_Cor_+3A_...">...</code></td>
<td>
<p>parameter in pheatmap</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pheatmap object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load(system.file("extdata", "network_example.rda", package = "CACIMAR"))
n1 &lt;- Identify_ConservedNetworks(OrthG_Mm_Zf,mmNetwork,zfNetwork,'mm','zf')
Heatmap_Cor(n1[[2]],cluster_cols=TRUE, cluster_rows=FALSE)
</code></pre>

<hr>
<h2 id='Identify_CellType'>Identify cell type of each cluster</h2><span id='topic+Identify_CellType'></span>

<h3>Description</h3>

<p>This function has three steps to identify cell type of each cluster.
(1) Calculate the power of each known marker based on AUC
(area under the receiver operating characteristic curve of gene expression)
which indicates the capability of marker i from cell type m to distinguish
cluster j and the other clusters. (2) Calculate the united power (UP)
for cell type m across each cluster j. (3) For each cluster j we determine
the cell type according to UP. Generally, the cluster beongs to the cell
type which have the highest united power or higher than the threshold of
the united power (for example &gt; 0.9 power).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Identify_CellType(seurat_object, Marker_gene_table)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Identify_CellType_+3A_seurat_object">seurat_object</code></td>
<td>
<p>seurat object</p>
</td></tr>
<tr><td><code id="Identify_CellType_+3A_marker_gene_table">Marker_gene_table</code></td>
<td>
<p>data.frame, indicating marker gene and its
corresponding cell type. Marker_gene_table should contain two columns: 'CellType'
represent correseponding cell types of each marker and 'Marker' represent Markers</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Cell type with the highest power in each cluster
</p>


<h3>Examples</h3>

<pre><code class='language-R'>KnownMarker=data.frame(c('AIF1','BID','CCL5','CD79A','CD79B','MS4A6A'),c('a','a','a','b','b','b'))
data("pbmc_small")
colnames(KnownMarker)=c('Marker','CellType')
CT &lt;- Identify_CellType(pbmc_small,KnownMarker)
</code></pre>

<hr>
<h2 id='Identify_ConservedCellTypes'>Identify conserved cell types based on power of genes and orthologs database</h2><span id='topic+Identify_ConservedCellTypes'></span>

<h3>Description</h3>

<p>Identify conserved cell types based on power of genes and orthologs database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Identify_ConservedCellTypes(
  OrthG,
  Species1_Marker_table,
  Species2_Marker_table,
  Species_name1,
  Species_name2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Identify_ConservedCellTypes_+3A_orthg">OrthG</code></td>
<td>
<p>ortholog genes database</p>
</td></tr>
<tr><td><code id="Identify_ConservedCellTypes_+3A_species1_marker_table">Species1_Marker_table</code></td>
<td>
<p>data.frame of species 1, should contain three column:
'gene', 'cluster' and 'power'</p>
</td></tr>
<tr><td><code id="Identify_ConservedCellTypes_+3A_species2_marker_table">Species2_Marker_table</code></td>
<td>
<p>data.frame of species 2, should contain three column:
'gene', 'cluster' and 'power'</p>
</td></tr>
<tr><td><code id="Identify_ConservedCellTypes_+3A_species_name1">Species_name1</code></td>
<td>
<p>character, indicating the species names of Species1_Marker_table</p>
</td></tr>
<tr><td><code id="Identify_ConservedCellTypes_+3A_species_name2">Species_name2</code></td>
<td>
<p>character, indicating the species names of Species2_Marker_table</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list contains two elements: first one is details of conserved cell types,
second one is matrix of cell types conserved score
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load(system.file("extdata", "CellTypeAllMarkers.rda", package = "CACIMAR"))
expression &lt;- Identify_ConservedCellTypes(OrthG_Mm_Zf,mm_Marker[1:30,],zf_Marker[1:30,],'mm','zf')
</code></pre>

<hr>
<h2 id='Identify_ConservedMarkers'>Identify orthologs marker genes for two species</h2><span id='topic+Identify_ConservedMarkers'></span>

<h3>Description</h3>

<p>Identify orthologs marker genes for two species based on orthologs database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Identify_ConservedMarkers(
  OrthG,
  Species1_Marker_table,
  Species2_Marker_table,
  Species_name1,
  Species_name2,
  match_cell_name = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Identify_ConservedMarkers_+3A_orthg">OrthG</code></td>
<td>
<p>ortholog genes database</p>
</td></tr>
<tr><td><code id="Identify_ConservedMarkers_+3A_species1_marker_table">Species1_Marker_table</code></td>
<td>
<p>data.frame of species 1, first column should be gene name,
second column should be Clusters corresponding to marker gene</p>
</td></tr>
<tr><td><code id="Identify_ConservedMarkers_+3A_species2_marker_table">Species2_Marker_table</code></td>
<td>
<p>data.frame of species 2, first column should be gene name,
second column should be Clusters corresponding to marker gene
of marker genes.</p>
</td></tr>
<tr><td><code id="Identify_ConservedMarkers_+3A_species_name1">Species_name1</code></td>
<td>
<p>character, indicating the species names of Species1_Marker_table.</p>
</td></tr>
<tr><td><code id="Identify_ConservedMarkers_+3A_species_name2">Species_name2</code></td>
<td>
<p>character, indicating the species names of Species2_Marker_table</p>
</td></tr>
<tr><td><code id="Identify_ConservedMarkers_+3A_match_cell_name">match_cell_name</code></td>
<td>
<p>characters contained in both cell names
to match similar cell types</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame of conserved markers
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load(system.file("extdata", "CellMarkers.rda", package = "CACIMAR"))
o1 &lt;- Identify_ConservedMarkers(OrthG_Mm_Zf,Mm_marker_cell_type,
Zf_marker_cell_type,Species_name1 = 'mm',Species_name2 = 'zf')
o2 &lt;- Identify_ConservedMarkers(OrthG_Zf_Ch,Ch_marker_cell_type,
Zf_marker_cell_type,Species_name1 = 'ch',Species_name2 = 'zf')
</code></pre>

<hr>
<h2 id='Identify_ConservedNetworks'>Identify conserved regulatory networks</h2><span id='topic+Identify_ConservedNetworks'></span>

<h3>Description</h3>

<p>Use Score of Conserved network to identify conserved regulatory
network modules based on homologous genes databased and topology of networks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Identify_ConservedNetworks(
  OrthG,
  Species1_GRN,
  Species2_GRN,
  Species_name1,
  Species_name2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Identify_ConservedNetworks_+3A_orthg">OrthG</code></td>
<td>
<p>ortholog genes database</p>
</td></tr>
<tr><td><code id="Identify_ConservedNetworks_+3A_species1_grn">Species1_GRN</code></td>
<td>
<p>gene regulatory network of species 1</p>
</td></tr>
<tr><td><code id="Identify_ConservedNetworks_+3A_species2_grn">Species2_GRN</code></td>
<td>
<p>gene regulatory network of species 2</p>
</td></tr>
<tr><td><code id="Identify_ConservedNetworks_+3A_species_name1">Species_name1</code></td>
<td>
<p>character, indicating the species names of Species1_GRN</p>
</td></tr>
<tr><td><code id="Identify_ConservedNetworks_+3A_species_name2">Species_name2</code></td>
<td>
<p>character, indicating the species names of Species2_GRN</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list contains two df. First df contains details of conserved regulatory
network, second df contains NCS between module pairs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load(system.file("extdata", "gene_network.rda", package = "CACIMAR"))
n1 &lt;- Identify_ConservedNetworks(OrthG_Mm_Zf,mm_gene_network,zf_gene_network,'mm','zf')
</code></pre>

<hr>
<h2 id='Identify_Markers'>Identify markers of each cluster</h2><span id='topic+Identify_Markers'></span>

<h3>Description</h3>

<p>This function first identify marker genes in each cluster
with Roc threshold &gt; RocThr. Then, based on marker genes identified above,
this function calculates the difference and power of marker genes in each
cluster, and marker genes with Difference threshold &gt; DiffThr will be retained.
Next, gene with the largest power in which cluster will be the marker
gene in this cluster. Eventually, make fisher test for power of each cluster,
cluster with p.value &lt; 0.05 will be retained as the final cluster for marker gene
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Identify_Markers(
  Seurat_object,
  PowerCutoff = 0.4,
  DifferenceCutoff = 0,
  PvalueCutoff = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Identify_Markers_+3A_seurat_object">Seurat_object</code></td>
<td>
<p>Seurat object, should contain cluster information</p>
</td></tr>
<tr><td><code id="Identify_Markers_+3A_powercutoff">PowerCutoff</code></td>
<td>
<p>numeric, indicating the cutoff of gene power to refine marker genes</p>
</td></tr>
<tr><td><code id="Identify_Markers_+3A_differencecutoff">DifferenceCutoff</code></td>
<td>
<p>numeric, indicating the cutoff of difference in marker genes between
clusters to refine marker genes</p>
</td></tr>
<tr><td><code id="Identify_Markers_+3A_pvaluecutoff">PvalueCutoff</code></td>
<td>
<p>numeric, indicating the p.value cutoff of chi-square test to refine marker genes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame of conserved markers
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("pbmc_small")
all.markers &lt;- Identify_Markers(pbmc_small)
</code></pre>

<hr>
<h2 id='OrthG_Hs_Ch'>Orthologs genes database for homo sapiens and zebrafish</h2><span id='topic+OrthG_Hs_Ch'></span>

<h3>Description</h3>

<p>Orthologs genes database for homo sapiens and zebrafish
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OrthG_Hs_Ch
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 12219 rows and 5 columns.
</p>

<hr>
<h2 id='OrthG_Hs_Mm'>Orthologs genes database for homo sapiens and mus musculus</h2><span id='topic+OrthG_Hs_Mm'></span>

<h3>Description</h3>

<p>Orthologs genes database for homo sapiens and mus musculus
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OrthG_Hs_Mm
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 16754 rows and 5 columns.
</p>

<hr>
<h2 id='OrthG_Hs_Zf'>Orthologs genes database for homo sapiens and zebrafish</h2><span id='topic+OrthG_Hs_Zf'></span>

<h3>Description</h3>

<p>Orthologs genes database for homo sapiens and zebrafish
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OrthG_Hs_Zf
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 12017 rows and 5 columns.
</p>

<hr>
<h2 id='OrthG_Mm_Ch'>Orthologs genes database for mus musculus and chicken</h2><span id='topic+OrthG_Mm_Ch'></span>

<h3>Description</h3>

<p>Orthologs genes database for mus musculus and chicken
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OrthG_Mm_Ch
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 62661 rows and 5 columns.
</p>

<hr>
<h2 id='OrthG_Mm_Zf'>Orthologs genes database for mus musculus and zebrafish</h2><span id='topic+OrthG_Mm_Zf'></span>

<h3>Description</h3>

<p>Orthologs genes database for mus musculus and zebrafish
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OrthG_Mm_Zf
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 65631 rows and 5 columns.
</p>

<hr>
<h2 id='OrthG_Zf_Ch'>Orthologs genes database for mus zebrafish and chicken</h2><span id='topic+OrthG_Zf_Ch'></span>

<h3>Description</h3>

<p>Orthologs genes database for mus zebrafish and chicken
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OrthG_Zf_Ch
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 38394 rows and 5 columns.
</p>

<hr>
<h2 id='Plot_MarkersHeatmap'>Plot Markers in each cell type</h2><span id='topic+Plot_MarkersHeatmap'></span>

<h3>Description</h3>

<p>This function integrate R package pheatmap to plot markers in each
cell type
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Plot_MarkersHeatmap(
  ConservedMarker,
  start_col = 2,
  module_colors = NA,
  heatmap_colors = NA,
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  show_rownames = FALSE,
  show_colnames = FALSE,
  cellwidth = NA,
  cellheight = NA,
  legend = FALSE,
  annotation_legend = FALSE,
  annotation_names_row = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Plot_MarkersHeatmap_+3A_conservedmarker">ConservedMarker</code></td>
<td>
<p>Markers table</p>
</td></tr>
<tr><td><code id="Plot_MarkersHeatmap_+3A_start_col">start_col</code></td>
<td>
<p>numeric, indicating the start column of marker power in each
cell type</p>
</td></tr>
<tr><td><code id="Plot_MarkersHeatmap_+3A_module_colors">module_colors</code></td>
<td>
<p>vector, indicating colors of modules (annotation_colors)</p>
</td></tr>
<tr><td><code id="Plot_MarkersHeatmap_+3A_heatmap_colors">heatmap_colors</code></td>
<td>
<p>vector, indicating colors used in heatmap</p>
</td></tr>
<tr><td><code id="Plot_MarkersHeatmap_+3A_cluster_rows">cluster_rows</code></td>
<td>
<p>boolean values determining if rows should be clustered or
hclust object</p>
</td></tr>
<tr><td><code id="Plot_MarkersHeatmap_+3A_cluster_cols">cluster_cols</code></td>
<td>
<p>boolean values determining if columns should be clustered
or hclust object</p>
</td></tr>
<tr><td><code id="Plot_MarkersHeatmap_+3A_show_rownames">show_rownames</code></td>
<td>
<p>boolean specifying if column names are be shown</p>
</td></tr>
<tr><td><code id="Plot_MarkersHeatmap_+3A_show_colnames">show_colnames</code></td>
<td>
<p>boolean specifying if column names are be shown</p>
</td></tr>
<tr><td><code id="Plot_MarkersHeatmap_+3A_cellwidth">cellwidth</code></td>
<td>
<p>individual cell width in points. If left as NA, then the
values depend on the size of plotting window</p>
</td></tr>
<tr><td><code id="Plot_MarkersHeatmap_+3A_cellheight">cellheight</code></td>
<td>
<p>individual cell height in points. If left as NA, then the
values depend on the size of plotting window</p>
</td></tr>
<tr><td><code id="Plot_MarkersHeatmap_+3A_legend">legend</code></td>
<td>
<p>logical to determine if legend should be drawn or not</p>
</td></tr>
<tr><td><code id="Plot_MarkersHeatmap_+3A_annotation_legend">annotation_legend</code></td>
<td>
<p>boolean value showing if the legend for annotation
tracks should be drawn</p>
</td></tr>
<tr><td><code id="Plot_MarkersHeatmap_+3A_annotation_names_row">annotation_names_row</code></td>
<td>
<p>boolean value showing if the names for row
annotation tracks should be drawn</p>
</td></tr>
<tr><td><code id="Plot_MarkersHeatmap_+3A_...">...</code></td>
<td>
<p>parameter in pheatmap</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pheatmap object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("pbmc_small")
all.markers &lt;- Identify_Markers(pbmc_small)
all.markers &lt;- Format_Markers_Frac(all.markers)
Plot_MarkersHeatmap(all.markers[,c(2,6,7,8)])
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
