<!DOCTYPE html><html lang="en-US"><head><title>Help for package CALANGO</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CALANGO}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#install_bioc_dependencies'><p>Install Bioconductor dependencies</p></a></li>
<li><a href='#make_report'><p>Prepare and render the HTML5 report</p></a></li>
<li><a href='#retrieve_calanguize_genomes'><p>Retrieve calanguize_genomes script from the Github repository</p></a></li>
<li><a href='#retrieve_data_files'><p>Retrieve data files from the Github repository</p></a></li>
<li><a href='#run_CALANGO'><p>Run the CALANGO pipeline</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Comparative Analysis with Annotation-Based Genomic Components</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.20</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-09-03</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Felipe Campelo &lt;fcampelo@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A first-principle, phylogeny-aware comparative genomics tool for 
             investigating associations between terms used to annotate genomic
             components (e.g., Pfam IDs, Gene Ontology terms,) with quantitative 
             or rank variables such as number of cell types, genome size, or 
             density of specific genomic elements. See the project website for 
             more information, documentation and examples, and 
             &lt;<a href="https://doi.org/10.1016%2Fj.patter.2023.100728">doi:10.1016/j.patter.2023.100728</a>&gt; for the full paper.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>assertthat (&ge; 0.2.1), pbmcapply (&ge; 1.5.0), ape (&ge; 5.3.0),
rmarkdown (&ge; 2.1.0), nlme (&ge; 3.1.0), BiocManager (&ge;
1.30.10), dendextend (&ge; 1.15.2), heatmaply (&ge; 1.1.0), ggplot2
(&ge; 2.3.2), plotly (&ge; 4.9.2), DT (&ge; 0.13), htmltools (&ge;
0.5.0), htmlwidgets (&ge; 1.5.1), pkgdown (&ge; 1.5.1), knitr (&ge;
1.28)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>AnnotationDbi, KEGGREST, GO.db</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://labpackages.github.io/CALANGO/">https://labpackages.github.io/CALANGO/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/fcampelo/CALANGO/issues/">https://github.com/fcampelo/CALANGO/issues/</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-09-05 07:36:40 UTC; me24832</td>
</tr>
<tr>
<td>Author:</td>
<td>Francisco Lobo [aut],
  Felipe Campelo [aut, cre],
  Jorge Augusto Hongo [aut],
  Giovanni Marques de Castro [aut],
  Gabriel Almeida [sad, dnc]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-05 14:40:01 UTC</td>
</tr>
</table>
<hr>
<h2 id='install_bioc_dependencies'>Install Bioconductor dependencies</h2><span id='topic+install_bioc_dependencies'></span>

<h3>Description</h3>

<p>This function installs the latest versions of all Bioconductor packages
required for the report generation, namely:
</p>

<ul>
<li><p> AnnotationDbi
</p>
</li>
<li><p> KEGGREST
</p>
</li>
<li><p> GO.db
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>install_bioc_dependencies(bioc.args = list(), force = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="install_bioc_dependencies_+3A_bioc.args">bioc.args</code></td>
<td>
<p>list containing further arguments to be passed
down to 'BiocManager::install()'.</p>
</td></tr>
<tr><td><code id="install_bioc_dependencies_+3A_force">force</code></td>
<td>
<p>logical: reinstall already-installed packages?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is essential that 
these Bioconductor packages be installed for CALANGO to work properly.
It uses 'BiocManager::install()' for installing Bioconductor
packages. Further arguments to this function are passed as a list.
</p>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  install_bioc_dependencies()

## End(Not run)

</code></pre>

<hr>
<h2 id='make_report'>Prepare and render the HTML5 report</h2><span id='topic+make_report'></span>

<h3>Description</h3>

<p>This script generates the HTML5 report based on an enriched
'CALANGO'-type list output by 'run_CALANGO()'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_report(defs, render.report = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="make_report_+3A_defs">defs</code></td>
<td>
<p>an enriched 'CALANGO'-type list generated by 'run_CALANGO()'</p>
</td></tr>
<tr><td><code id="make_report_+3A_render.report">render.report</code></td>
<td>
<p>logical: should the HTML5 report be generated (for
internal use only)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Updated 'defs' list, containing:
</p>

<ul>
<li><p> All input parameters originally passed (see [run_CALANGO()] for 
details).
</p>
</li>
<li><p> Derived fields calculated for displaying the results, including 
several statistical summaries of the data (including correlations, 
contrasts, covariances, p-values).
</p>
</li></ul>

<p>This function is mainly called for its side effect, namely the generation of 
an HTML5 report of the analysis, which is saved to the folder indicated in 
'defs$output.dir'.
</p>

<hr>
<h2 id='retrieve_calanguize_genomes'>Retrieve calanguize_genomes script from the Github repository</h2><span id='topic+retrieve_calanguize_genomes'></span>

<h3>Description</h3>

<p>This script downloads the *calanguize_genomes.pl* Perl script from the 
repository, together with associated README instructions for using the 
script, managing dependencies, etc. It will extract the data into a folder 
containing everything that is needed for preparing data for using 
CALANGO.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>retrieve_calanguize_genomes(
  target.dir,
  method = "auto",
  unzip = getOption("unzip"),
  overwrite = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="retrieve_calanguize_genomes_+3A_target.dir">target.dir</code></td>
<td>
<p>path to the folder where the files will be saved (
accepts relative and absolute paths)</p>
</td></tr>
<tr><td><code id="retrieve_calanguize_genomes_+3A_method">method</code></td>
<td>
<p>Method to be used for downloading files. Current download
methods are &quot;internal&quot;, &quot;wininet&quot; (Windows only) &quot;libcurl&quot;, &quot;wget&quot; and
&quot;curl&quot;, and there is a value &quot;auto&quot;: see _Details_ and _Note_ in the
documentation of <code>utils::download.file()</code>.</p>
</td></tr>
<tr><td><code id="retrieve_calanguize_genomes_+3A_unzip">unzip</code></td>
<td>
<p>The unzip method to be used. See the documentation of
<code>utils::unzip()</code> for details.</p>
</td></tr>
<tr><td><code id="retrieve_calanguize_genomes_+3A_overwrite">overwrite</code></td>
<td>
<p>If TRUE, overwrite existing files in target the folder 
(the equivalent of unzip -o), otherwise ignore such files (the equivalent of 
unzip -n).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the 'target.dir' provided does not exist it is created
(recursively) by the function.
</p>


<h3>Value</h3>

<p>No return value, called for side effects (see Description).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  CALANGO::retrieve_calanguize_genomes(target.dir = "./data")

## End(Not run)

</code></pre>

<hr>
<h2 id='retrieve_data_files'>Retrieve data files from the Github repository</h2><span id='topic+retrieve_data_files'></span>

<h3>Description</h3>

<p>This script downloads relevant data files from the CALANGO project
repository. It will extract the data into a folder containing
directories related to dictionary files, Gene Ontology annotation
files, tree files, etc. Note: you may need to edit the file paths in the 
example scripts contained under the 'parameters' subfolder of 'target.dir', 
or pass an appropriate base path using parameter 'basedir' in [run_CALANGO()].
</p>


<h3>Usage</h3>

<pre><code class='language-R'>retrieve_data_files(
  target.dir,
  method = "auto",
  unzip = getOption("unzip"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="retrieve_data_files_+3A_target.dir">target.dir</code></td>
<td>
<p>path to the folder where the files will be saved (
accepts relative and absolute paths)</p>
</td></tr>
<tr><td><code id="retrieve_data_files_+3A_method">method</code></td>
<td>
<p>Method to be used for downloading files. Current download
methods are &quot;internal&quot;, &quot;wininet&quot; (Windows only) &quot;libcurl&quot;, &quot;wget&quot; and
&quot;curl&quot;, and there is a value &quot;auto&quot;: see _Details_ and _Note_ in the
documentation of <code>utils::download.file()</code>.</p>
</td></tr>
<tr><td><code id="retrieve_data_files_+3A_unzip">unzip</code></td>
<td>
<p>The unzip method to be used. See the documentation of
<code>utils::unzip()</code> for details.</p>
</td></tr>
<tr><td><code id="retrieve_data_files_+3A_...">...</code></td>
<td>
<p>additional attributes (currently ignored)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the 'target.dir' provided does not exist it is created
(recursively) by the function.
</p>


<h3>Value</h3>

<p>No return value, called for side effects (see Description).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  CALANGO::retrieve_data_files(target.dir = "./data")

## End(Not run)

</code></pre>

<hr>
<h2 id='run_CALANGO'>Run the CALANGO pipeline</h2><span id='topic+run_CALANGO'></span>

<h3>Description</h3>

<p>This function runs the complete workflow of CALANGO and generates the
HTML5 output pages and export files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_CALANGO(
  defs,
  type = "correlation",
  cores = NULL,
  render.report = TRUE,
  basedir = ""
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_CALANGO_+3A_defs">defs</code></td>
<td>
<p>either a CALANGO-type list object or
a path to a text file containing the required definitions (see Details).</p>
</td></tr>
<tr><td><code id="run_CALANGO_+3A_type">type</code></td>
<td>
<p>type of analysis to perform. Currently only &quot;correlation&quot; is
supported.</p>
</td></tr>
<tr><td><code id="run_CALANGO_+3A_cores">cores</code></td>
<td>
<p>positive integer, how many CPU cores to use. Setting
this parameter overrides any 'cores' field from 'defs'.</p>
</td></tr>
<tr><td><code id="run_CALANGO_+3A_render.report">render.report</code></td>
<td>
<p>logical: should a HTML5 report be generated?</p>
</td></tr>
<tr><td><code id="run_CALANGO_+3A_basedir">basedir</code></td>
<td>
<p>path to base folder to which all relative paths in 'defs' 
refer to.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The script expects a 'CALANGO'-type list, passed either as an actual list
object or as a file path. In the latter case, notice that
the file must be a text file with a 'field = value' format.
Blank likes and lines starting with '#' are ignored. The function expects the
input list to have the following fields:
</p>

<ul>
<li><p><code>annotation.files.dir</code> (required, string) - Folder where
annotation files are located.
</p>
</li>
<li><p><code>output.dir</code> (required, string) - output folder for results
</p>
</li>
<li><p><code>dataset.info</code> (required, string) - genome metadata file, it
should contain at least:
</p>

<ul>
<li><p> File names. Please notice this information should be the first
column in metadata file;
</p>
</li>
<li><p> Phenotype data (numeric, this is the value CALANGO uses to rank
species when searching for associations)
</p>
</li>
<li><p> Normalization data (numeric, this is the value CALANGO uses as a
denominator to compute annotation term frequencies to remove potential
biases caused by, for instance, over annotation of model organisms or
large differences in the counts of genomic elements). Please notice that
CALANGO does not require normalization data for GO, as it computes the
total number of GO terms per species and uses it as a normalizing factor.
</p>
</li></ul>

</li>
<li><p><code>x.column</code> (required, numeric) - which column in &quot;dataset.info&quot;
contains the phenotype data?
</p>
</li>
<li><p><code>ontology</code> (required, string)  - which dictionary data type to
use? Possible values are &quot;GO&quot; and &quot;other&quot;. For GO, CALANGO can compute
normalization data.
</p>
</li>
<li><p><code>dict.path</code> (required, string)  - file for dictionary file
(two-column file containing annotation IDs and their descriptions. Not
needed for GO.
</p>
</li>
<li><p><code>column</code> (required, string)  - which column in annotation files
should be used (column name)
</p>
</li>
<li><p><code>denominator.column</code> (optional, numeric) - which column contains
normalization data (column number)
</p>
</li>
<li><p><code>tree.path</code> (required, string)  - path for tree file in either
newick or nexus format
</p>
</li>
<li><p><code>tree.type</code> (required, string) - tree file type (either &quot;nexus&quot;
or &quot;newick&quot;)
</p>
</li>
<li><p><code>cores</code> (optional, numeric) - how many cores to use? If not
provided the function defaults to 1.
</p>
</li>
<li><p><code>linear.model.cutoff</code> (required, numeric) - parameter that
regulates how much graphical output is produced. We configure it to
generate plots only for annotation terms with corrected q-values for
phylogenetically independent contrasts (standard: smaller than 0.5).
</p>
</li>
<li><p><code>MHT.method</code> (optional, string) - type of multiple hypothesis
correction to be used. Accepts all methods listed by
'stats::p.adjust.methods()'. If not provided the function defaults to
&quot;BH&quot;.
</p>
</li></ul>



<h3>Value</h3>

<p>Updated 'defs' list, containing:
</p>

<ul>
<li><p> All input parameters originally passed or read from a 'defs' file 
(see **Details**).
</p>
</li>
<li><p> Derived fields loaded and preprocessed from the files indicated in 
'defs'.
</p>
</li>
<li><p> Several statistical summaries of the data (used to render the 
report), including correlations, contrasts, covariances, p-values and 
other summary statistics.
</p>
</li></ul>

<p>Results are also saved to files under 'defs$output.dir'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

## Install any missing BioConductor packages for report generation 
## (only needs to be done once)
# CALANGO::install_bioc_dependencies()

# Retrieve example files
basedir &lt;- tempdir()
retrieve_data_files(target.dir = paste0(basedir, "/data"))
defs &lt;- paste0(basedir, "/data/parameters/parameters_domain2GO_count_less_phages.txt")

# Run CALANGO
res &lt;- run_CALANGO(defs, cores = 2)

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
