<!DOCTYPE html><html><head><title>Help for package CAMAN</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CAMAN}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='# vem_grad '><p> VEM algorithm for univariate data, for bivariate data and for meta data</p></a></li>
<li><a href='#anova.CAMAN.object'><p>ANOVA for finite mixture models</p></a></li>
<li><a href='#aspirin'><p>Aspirin use and breast cancer risk</p></a></li>
<li><a href='#betaplasma'><p>Determinants of Beta-Carotene Levels</p></a></li>
<li><a href='#bivariate.EM'><p>EM-algorithm for bivariate normally distributed data</p></a></li>
<li><a href='#bivariate.mixalg'><p> EM algorithm and classification for univariate data, for bivariate data and for meta data</p></a></li>
<li><a href='#bivariate.VEM'><p>VEM algorithm for univariate data, for bivariate data and for meta data</p></a></li>
<li><a href='#CAMANboot'><p> Parametric bootstrap</p></a></li>
<li><a href='#CT'><p>Meta-anaysis: noninvasive coronary angiography using computed tomography (CT)</p></a></li>
<li><a href='#getFDR'>
<p>Compute false discovery rates and related statistics</p></a></li>
<li><a href='#golubMerge'><p>Data from the Golub et al (1999) Paper</p></a></li>
<li><a href='#heage'>
<p>Northeast Thailand malnourishment data.</p></a></li>
<li><a href='#hepab'>
<p>Hepatitis B data from Schlattmann and B\&quot;ohning (1993).</p></a></li>
<li><a href='#hist.CAMAN.object'><p>Histograms for finite mixture models</p></a></li>
<li><a href='#leukDat'><p>Childhood leukemia in the former GDR</p></a></li>
<li><a href='#mixalg'><p>Fitting Finite Mixture Models</p></a></li>
<li><a href='#mixalg.boot'><p>bootstrap replication / validation of finite mixture models</p></a></li>
<li><a href='#mixalg.EM'><p>EM algorithm</p></a></li>
<li><a href='#mixalg.VEM'><p>VEM algorithm</p></a></li>
<li><a href='#mixcov'><p>Fitting mixture models with covariates</p></a></li>
<li><a href='#NoP'><p>Ames test data: Mutagenicity of 4NoP</p></a></li>
<li><a href='#PCT'><p>Procalcitonin as diagnostic marker for sepsis</p></a></li>
<li><a href='#plot.CAMAN.BIMIXALG.object'><p>Plot ellipses</p></a></li>
<li><a href='#rs12363681'><p>Gene calling</p></a></li>
<li><a href='#thai_cohort'><p>Cohort study in north east Thailand</p></a></li>
<li><a href='#vitamin'><p>Meta-analysis if vitamin A supplementation of childhood mortality</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>0.78</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-09-19</td>
</tr>
<tr>
<td>Title:</td>
<td>Finite Mixture Models and Meta-Analysis Tools - Based on C.A.MAN</td>
</tr>
<tr>
<td>Author:</td>
<td>Peter Schlattmann &lt;peter.schlattmann@mti.uni-jena.de&gt;, Johannes
        Hoehne, Maryna Verba</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Philipp Doebler &lt;philipp.doebler@googlemail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Tools for the analysis of finite semiparametric mixtures.
        These are useful when data is heterogeneous, e.g. in
        pharmacokinetics or meta-analysis. The NPMLE and VEM algorithms
        (flexible support size) and EM algorithms (fixed support size)
        are provided for univariate (Bohning et al., 1992; 
        &lt;<a href="https://doi.org/10.2307%2F2532756">doi:10.2307/2532756</a>&gt;) and bivariate data (Schlattmann et 
        al., 2015; &lt;<a href="https://doi.org/10.1016%2Fj.jclinepi.2014.08.013">doi:10.1016/j.jclinepi.2014.08.013</a>&gt;).</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), methods, sp, mvtnorm</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-19 09:11:33 UTC; doebler</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-22 15:10:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='+20vem_grad+20'> VEM algorithm for univariate data, for bivariate data and for meta data</h2><span id='topic+vem_grad'></span>

<h3>Description</h3>

<p>This function implements a VEM algorithm.</p>


<h3>Usage</h3>

<pre><code class='language-R'>vem_grad(obs1, obs2, type, data, var1, var2, 
         lambda1, lambda2, p, startk, 
         numiter=5000, acc=1.e-7)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B20vem_grad+2B20_+3A_obs1">obs1</code></td>
<td>
<p>the first column of the observations</p>
</td></tr></table>
<p><br />
</p>
<table>
<tr><td><code id="+2B20vem_grad+2B20_+3A_obs2">obs2</code></td>
<td>
<p>the second column of the observations</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="+2B20vem_grad+2B20_+3A_type">type</code></td>
<td>
<p>kind of data</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="+2B20vem_grad+2B20_+3A_data">data</code></td>
<td>
<p>an optional data frame</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="+2B20vem_grad+2B20_+3A_lambda1">lambda1</code></td>
<td>
<p>Means of the first column of the observations</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="+2B20vem_grad+2B20_+3A_lambda2">lambda2</code></td>
<td>
<p>Means of the second column of the observations</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="+2B20vem_grad+2B20_+3A_p">p</code></td>
<td>
<p>Mixing weight</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="+2B20vem_grad+2B20_+3A_var1">var1</code></td>
<td>
<p>Variance of the first column of the observations(only for  meta-analysis)</p>
</td></tr> </table>
<p><br /> 
</p>
<table>
<tr><td><code id="+2B20vem_grad+2B20_+3A_var2">var2</code></td>
<td>
<p>Variance of the second column of the observations (only for meta-analysis)</p>
</td></tr> </table>
<p><br /> 
</p>
<table>
<tr><td><code id="+2B20vem_grad+2B20_+3A_startk">startk</code></td>
<td>
<p> starting/maximal number of components. This number will be used to compute the grid in the VEM. Default is 20.)</p>
</td></tr> </table>
<p><br /> 
</p>
<table>
<tr><td><code id="+2B20vem_grad+2B20_+3A_numiter">numiter</code></td>
<td>
<p>parameter to control the maximal number of iterations in the VEM and EM loops. Default is 5000.</p>
</td></tr> </table>
<p><br />
</p>
<table>
<tr><td><code id="+2B20vem_grad+2B20_+3A_acc">acc</code></td>
<td>
<p>convergence criterion. Default is 1.e-7</p>
</td></tr> </table>
<p><br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#1.	VEM-algorithm  for bivariate normally distributed data 
#Examples
data(rs12363681)
vem_grad(obs1=x,obs2=y,type="bi", data=rs12363681,startk=20)
#2.VEM for metadata
data(CT)
vem_grad(obs1=logitTPR, obs2= logitTNR, 
         var1= varlogitTPR, var2= varlogitTNR,
         type="meta", data=CT, startk=20)

## End(Not run)
</code></pre>

<hr>
<h2 id='anova.CAMAN.object'>ANOVA for finite mixture models</h2><span id='topic+anova.CAMAN.object'></span>

<h3>Description</h3>

<p>A common problem in the estimation of mixture models is to determine the
number of components. This may be done using a parametric bootstrap.
This function simulates from a mixture model under the null hypothesis
with <code>k0</code> components. A mixture model with k0 and usually k0 +1 components 
is fit to the data and then the likelihood ratio statistic (LRS) is computed. <br />
</p>
<p>Based on the bootstrap the distribution of the LRS is obtained which allows to obtain
an approximation to the achieved level of significance corresponding to
the value of <code class="reqn">-2 \log \xi</code> obtained from the original sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CAMAN.object'
anova(object, object1, nboot=2500, limit=0.01, acc=10^(-7), 
     numiter=5000, giveBootstrapData=FALSE, giveLikelihood=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="anova.CAMAN.object_+3A_object">object</code></td>
<td>

<p>A CAMAN-object which quantifies a finite mixture model under null hypothesis.
</p>
</td></tr>
<tr><td><code id="anova.CAMAN.object_+3A_object1">object1</code></td>
<td>

<p>A CAMAN-object which quantifies another finite mixture model under the alterative hypothesis.
</p>
</td></tr>
<tr><td><code id="anova.CAMAN.object_+3A_nboot">nboot</code></td>
<td>

<p>Number of bootstrap samples.
</p>
</td></tr>
<tr><td><code id="anova.CAMAN.object_+3A_limit">limit</code></td>
<td>

<p>parameter to control the limit of union several components. Default is 0.01.
</p>
</td></tr>
<tr><td><code id="anova.CAMAN.object_+3A_acc">acc</code></td>
<td>

<p>convergence criterion. VEM and EM loops stop when deltaLL&lt;acc. Default is 10^(-7). 
</p>
</td></tr>
<tr><td><code id="anova.CAMAN.object_+3A_numiter">numiter</code></td>
<td>

<p>parameter to control the maximal number of iterations in the VEM and EM loops. Default is 5000. 
</p>
</td></tr>
<tr><td><code id="anova.CAMAN.object_+3A_givebootstrapdata">giveBootstrapData</code></td>
<td>

<p>A Boolean that indicates whether the bootstrapped data should be returned or not
</p>
</td></tr>
<tr><td><code id="anova.CAMAN.object_+3A_givelikelihood">giveLikelihood</code></td>
<td>

<p>Return the likelihood-values of both models for each generated dataset.
</p>
</td></tr>
<tr><td><code id="anova.CAMAN.object_+3A_...">...</code></td>
<td>

<p>Arguments to be passed on to other methods; currently none.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The parameters <code>limit, acc</code> and <code>numiter</code> are used for the VEM algorithm in each bootstrap sample.
</p>


<h3>Value</h3>

<p>The function returns a list with components
</p>

<ul>
<li> <p><code>overview</code>: comparison of the models, including BIC, LL and LL-ratio <br />
</p>
</li>
<li> <p><code>`LL ratios in bootstrap-data`</code>: 90, 95, 97.5 and 99 percentiles of LL-ratios <br />
</p>
</li>
<li> <p><code>`simulated p-value`</code>: p-value, quantifying the null model  <br />
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Peter Schlattmann and Johannes Hoehne   
</p>


<h3>References</h3>

<p>McLachlan, G. and Peel, D. (2000). <em>Finite Mixture Models</em>, Chichester: Wiley. <br />
</p>
<p>Schlattmann, P. (2009). <em>Medical Applications of Finite Mixture Models</em>. Berlin: Springer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(thai_cohort)
mix0 &lt;- mixalg(obs="counts", weights="frequency", family="poisson", data=thai_cohort,
               numiter=18000, acc=0.00001,startk=25)
em0&lt;-mixalg.EM(mix0,p=c(1),t=c(1))
em1&lt;-mixalg.EM(mix0,p=c(0.7,0.3),t=c(2,9))
## Not run: ll&lt;-anova(em0,em1,nboot=250) #might take some minutes
</code></pre>

<hr>
<h2 id='aspirin'>Aspirin use and breast cancer risk</h2><span id='topic+aspirin'></span>

<h3>Description</h3>

<p>This data set can be used to examine the recent epidemiological studies on aspirin use and breast cancer risk published from 2001 to 2007 within a meta-analysis, and to investigate reasons for heterogeneity between the individual studies. <br />
</p>
<p>We systematically searched for cohort-studies and case-control-studies from 2001-2007, which evaluated the association between aspirin and breast cancer risk. A total of 15 studies (seven cohort studies and eight case-control studies) met the inclusion criteria.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("aspirin")</code></pre>


<h3>Format</h3>

<p>A data frame consisting of 15 data sets (rows) and 11 attributes (columns)</p>


<h3>References</h3>

<p>Schlattmann, P.(2009) <em>Medical Applications of Finite Mixture Models.</em> Berlin: Springer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Example
#Homogeneous Metaregression adjusting for study type and year of publication

data(aspirin)
wgt &lt;- 1/aspirin$var# calculate weights
m0 &lt;- mixcov(dep="logrr", fixed=c("type","yearc"), random="", weight=wgt,
             k=1,family="gaussian",data=aspirin)

</code></pre>

<hr>
<h2 id='betaplasma'>Determinants of Beta-Carotene Levels</h2><span id='topic+betaplasma'></span>

<h3>Description</h3>

<p>Observational studies have suggested that low dietary intake or low plasma 
concentrations of retinol, beta-carotene, or other carotenoids might be associated 
with increased risk of developing certain types of cancer.  However, relatively few 
studies have investigated the determinants of plasma concentrations of these 
micronutrients. Nierenberg et al. designed a cross-sectional study to investigate the relationship between personal characteristics and dietary factors, and plasma concentrations of retinol, beta-carotene and other carotenoids. Study subjects (N = 315) were patients 
who had an elective surgical procedure during a three-year period to biopsy or 
remove a lesion of the lung, colon, breast, skin, ovary or uterus that was found to be non-cancerous. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("betaplasma")</code></pre>


<h3>Format</h3>

<p>datafile contains 315 observations on 15 variables.  This data set can be used to demonstrate multiple regression, transformations, categorical variables, outliers, pooled tests of significance and model building strategies. 
</p>

<p>Variable Names in order from left to right:
</p>
<dl>
<dt>item</dt><dd><p>participant id</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>age</dt><dd><p>Age (years)</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>sex</dt><dd><p>Sex, Factor with 2 levels (M=Male, F=Female).</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>smokestat</dt><dd><p>Smoking status, Factor with 3 levels (Never, Former, Current Smoker)</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>bmi</dt><dd><p>body mass index (weight/(height^2))</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>vituse</dt><dd><p>Vitamin Use, Factor with 3 levels (Yes (fairly often), Yes (not often), No)</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>calories</dt><dd><p>Number of calories consumed per day.</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>fat</dt><dd><p>Grams of fat consumed per day.</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>fiber</dt><dd><p>Grams of fiber consumed per day.</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>alcohol</dt><dd><p>Number of alcoholic drinks consumed per week.</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>chol</dt><dd><p>Cholesterol consumed (mg per day).</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>betadiet</dt><dd><p>Dietary beta-carotene consumed (mcg per day).</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>retdiet</dt><dd><p>Dietary retinol consumed (mcg per day)</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>betacaro</dt><dd><p>Plasma beta-carotene (ng/ml)</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>retplasma</dt><dd><p>Plasma Retinol (ng/ml)</p>
</dd> </dl>
<p><br />

</p>


<h3>References</h3>

<p>Schlattmann, P.(2009) <em>Medical Applications of Finite Mixture Models.</em> Berlin: Springer. 
</p>
<p>These data have not been published yet but a related reference is
</p>
<p>Nierenberg DW, Stukel TA, Baron JA, Dain BJ, Greenberg ER (1989) &ldquo;Determinants of plasma levels of beta-carotene and retinol.&rdquo;  <em>American Journal of Epidemiology</em>, <b>130</b>, 511&ndash;521.
</p>
<p>The use of the data is kindly permitted Dr. Therese Stukel, Dartmouth Hitchcock Medical Center, USA
</p>
<p>The data can also be found at StatLib: http://lib.stat.cmu.edu/index.php
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(betaplasma)
beta4 &lt;- mixcov(dep="betacaro", fixed=c("chol","sex","bmi"), random="betadiet",
                data=betaplasma,k=4,family="gaussian")

</code></pre>

<hr>
<h2 id='bivariate.EM'>EM-algorithm for bivariate normally distributed data</h2><span id='topic+bivariate.EM'></span>

<h3>Description</h3>

<p>Function</p>


<h3>Usage</h3>

<pre><code class='language-R'>bivariate.EM(obs1, obs2, type, data = NULL,
             var1, var2, 
             corr, lambda1, lambda2, p, 
             numiter=5000,acc=1.e-7,class)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bivariate.EM_+3A_obs1">obs1</code></td>
<td>
<p>the first column of the observations</p>
</td></tr></table>
<p><br />
</p>
<table>
<tr><td><code id="bivariate.EM_+3A_obs2">obs2</code></td>
<td>
<p>the second column of the observations</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.EM_+3A_type">type</code></td>
<td>
<p>kind of data</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.EM_+3A_data">data</code></td>
<td>
<p>an optional data frame. If not <code>NULL</code>, <code>obs1</code>, <code>obs2</code>, <code>var1</code>, <code>var2</code> and <code>corr</code> will be looked for in <code>data</code></p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.EM_+3A_var1">var1</code></td>
<td>
<p>Variance of the first column of the observations(except meta-analysis)</p>
</td></tr> </table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.EM_+3A_var2">var2</code></td>
<td>
<p>Variance of the second column of the observations (except meta-analysis)</p>
</td></tr> </table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.EM_+3A_corr">corr</code></td>
<td>
<p>correlation coefficient</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.EM_+3A_lambda1">lambda1</code></td>
<td>
<p>Means of the first column of the observations</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.EM_+3A_lambda2">lambda2</code></td>
<td>
<p>Means of the second column of the observations</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.EM_+3A_p">p</code></td>
<td>
<p>Mixing weight</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.EM_+3A_numiter">numiter</code></td>
<td>
<p>parameter to control the maximal number of iterations in the  EM loops. Default is 5000.</p>
</td></tr> </table>
<p><br />
</p>
<table>
<tr><td><code id="bivariate.EM_+3A_acc">acc</code></td>
<td>
<p>convergence criterion. Default is 1.e-7</p>
</td></tr> </table>
<p><br />
</p>
<table>
<tr><td><code id="bivariate.EM_+3A_class">class</code></td>
<td>
<p>classification of studies?</p>
</td></tr> </table>
<p><br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# 1.EM and classification for bivariate data with starting values
data(rs12363681)
lambda1&lt;-c(1540.97, 837.12, 945.40, 1053.69)
lambda2&lt;-c(906.66, 1371.81 ,1106.01,973.11)
p&lt;-c(0.05,0.15,0.6,0.2)
test&lt;-bivariate.EM(obs1=x, obs2=y, type="bi", lambda1=lambda1,lambda2=lambda2,
                   p=p,data=rs12363681,class="TRUE")
# scatter plot with ellipse
plot(test, ellipse=TRUE)
# scatter plot without ellipse
plot(test, ellipse=FALSE)

## End(Not run)
# 2. EM-algorithm for a diagnostic meta-analysis with bivariate 
#    normally distributed data and study specific fixed variances
data(CT)
p2&lt;-c(0.4,0.6)
lamlog12&lt;-c(2.93,3.22)
lamlog22&lt;-c(2.5,1.5)
ct.m1 &lt;- bivariate.EM(obs1=logitTPR, obs2=logitTNR, 
                      var1=varlogitTPR, var2=varlogitTNR,
                      type="meta", lambda1=lamlog12, lambda2=lamlog22,
                      p=p2,data=CT,class="TRUE")
</code></pre>

<hr>
<h2 id='bivariate.mixalg'> EM algorithm and classification for univariate data, for bivariate data and for meta data </h2><span id='topic+bivariate.mixalg'></span>

<h3>Description</h3>

<p>Function</p>


<h3>Usage</h3>

<pre><code class='language-R'>bivariate.mixalg(obs1, obs2, type, data = NULL, 
                 var1, var2, corr, lambda1, lambda2,
                 p,startk, numiter=5000, acc=1.e-7, class)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bivariate.mixalg_+3A_obs1">obs1</code></td>
<td>
<p>the first column of the observations</p>
</td></tr></table>
<p><br />
</p>
<table>
<tr><td><code id="bivariate.mixalg_+3A_obs2">obs2</code></td>
<td>
<p>the second column of the observations</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.mixalg_+3A_type">type</code></td>
<td>
<p>kind of data</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.mixalg_+3A_data">data</code></td>
<td>
<p>an optional data frame</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.mixalg_+3A_var1">var1</code></td>
<td>
<p>Variance of the first column of the observations(except meta-analysis)</p>
</td></tr> </table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.mixalg_+3A_var2">var2</code></td>
<td>
<p>Variance of the second column of the observations (except meta-analysis)</p>
</td></tr> </table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.mixalg_+3A_corr">corr</code></td>
<td>
<p>correlation coefficient</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.mixalg_+3A_lambda1">lambda1</code></td>
<td>
<p>Means of the first column of the observations</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.mixalg_+3A_lambda2">lambda2</code></td>
<td>
<p>Means of the second column of the observations</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.mixalg_+3A_p">p</code></td>
<td>
<p>Probability</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.mixalg_+3A_startk">startk</code></td>
<td>
<p> starting/maximal number of components. This number will be used to compute the grid in the VEM. Default is 20.</p>
</td></tr> </table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.mixalg_+3A_numiter">numiter</code></td>
<td>
<p>parameter to control the maximal number of iterations in the VEM and EM loops. Default is 5000.</p>
</td></tr> </table>
<p><br />
</p>
<table>
<tr><td><code id="bivariate.mixalg_+3A_acc">acc</code></td>
<td>
<p>convergence criterion. Default is 1.e-7</p>
</td></tr> </table>
<p><br />
</p>
<table>
<tr><td><code id="bivariate.mixalg_+3A_class">class</code></td>
<td>
<p> classification of studies </p>
</td></tr> </table>
<p><br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#1.EM and classification for bivariate data
#Examples
data(rs12363681)
test &lt;- bivariate.mixalg(obs1=x, obs2=y, type="bi", 
                         lambda1=0, lambda2=0, p=0, 
                         data=rs12363681, startk=20, class="TRUE")
#scatter plot with ellipse
plot(test)
#scatter plot without ellipse
plot(test, ellipse = FALSE)
#2.EM and classification for meta data
#Examples
data(CT)
bivariate.mixalg(obs1=logitTPR, obs2=logitTNR, 
                 var1=varlogitTPR, var2=varlogitTNR,
                 type="meta", lambda1=0, lambda2=0,
                 p=0,data=CT,startk=20,class="TRUE")

## End(Not run)
</code></pre>

<hr>
<h2 id='bivariate.VEM'>VEM algorithm for univariate data, for bivariate data and for meta data</h2><span id='topic+bivariate.VEM'></span>

<h3>Description</h3>

<p>VEM algorithm for univariate data, for bivariate data and for meta data</p>


<h3>Usage</h3>

<pre><code class='language-R'>bivariate.VEM(obs1, obs2, type, data = NULL, var1, var2, 
             lambda1, lambda2, p, startk, numiter=5000,
             acc=1.e-7)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bivariate.VEM_+3A_obs1">obs1</code></td>
<td>
<p>the first column of the observations</p>
</td></tr></table>
<p><br />
</p>
<table>
<tr><td><code id="bivariate.VEM_+3A_obs2">obs2</code></td>
<td>
<p>the second column of the observations</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.VEM_+3A_type">type</code></td>
<td>
<p>kind of data</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.VEM_+3A_data">data</code></td>
<td>
<p>an optional data frame. If not <code>NULL</code>, <code>obs1</code>, <code>obs2</code>, <code>var1</code> and <code>var2</code> will be looked for in <code>data</code></p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.VEM_+3A_lambda1">lambda1</code></td>
<td>
<p>Means of the first column of the observations</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.VEM_+3A_lambda2">lambda2</code></td>
<td>
<p>Means of the second column of the observations</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.VEM_+3A_p">p</code></td>
<td>
<p>Mixing weight</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.VEM_+3A_var1">var1</code></td>
<td>
<p>Variance of the first column of the observations(only for meta-analysis)</p>
</td></tr> </table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.VEM_+3A_var2">var2</code></td>
<td>
<p>Variance of the second column of the observations (only for meta-analysis)</p>
</td></tr> </table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.VEM_+3A_startk">startk</code></td>
<td>
<p> starting/maximal number of components. This number will be used to compute the grid in the VEM. Default is 20.)</p>
</td></tr> </table>
<p><br /> 
</p>
<table>
<tr><td><code id="bivariate.VEM_+3A_numiter">numiter</code></td>
<td>
<p>parameter to control the maximal number of iterations in the VEM and EM loops. Default is 5000.</p>
</td></tr> </table>
<p><br />
</p>
<table>
<tr><td><code id="bivariate.VEM_+3A_acc">acc</code></td>
<td>
<p>convergence criterion. Default is 1.e-7</p>
</td></tr> </table>
<p><br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# 1.	VEM-algorithm  for bivariate normally distributed data 
#Examples
data(rs12363681)
bivariate.VEM(obs1=x,obs2=y,type="bi", data=rs12363681,startk=20)
# 2.VEM for metadata
data(CT)
bivariate.VEM(obs1=logitTPR, obs2=logitTNR, 
              var1=varlogitTPR, var2= varlogitTNR,
              type="meta", data=CT, startk=20)

## End(Not run)
</code></pre>

<hr>
<h2 id='CAMANboot'> Parametric bootstrap </h2><span id='topic+CAMANboot'></span>

<h3>Description</h3>

<p>Parametric bootstrap for bivariate normally distributed data</p>


<h3>Usage</h3>

<pre><code class='language-R'>CAMANboot(obs1, obs2, var1, var2, lambda11, lambda12, 
     prob1, lambda21, lambda22, prob2, rep, 
     data,numiter=10000,acc=1.e-7)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CAMANboot_+3A_obs1">obs1</code></td>
<td>
<p>the first column of the observations</p>
</td></tr></table>
<p><br />
</p>
<table>
<tr><td><code id="CAMANboot_+3A_obs2">obs2</code></td>
<td>
<p>the second column of the observations</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="CAMANboot_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="CAMANboot_+3A_var1">var1</code></td>
<td>
<p>Variance of the first column of the observations(except meta-analysis)</p>
</td></tr> </table>
<p><br /> 
</p>
<table>
<tr><td><code id="CAMANboot_+3A_var2">var2</code></td>
<td>
<p>Variance of the second column of the observations (except meta-analysis)</p>
</td></tr> </table>
<p><br /> 
</p>
<table>
<tr><td><code id="CAMANboot_+3A_lambda11">lambda11</code></td>
<td>
<p>first means of the first column of the observations</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="CAMANboot_+3A_lambda12">lambda12</code></td>
<td>
<p>first means of the second column of the observations</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="CAMANboot_+3A_prob1">prob1</code></td>
<td>
<p>first mixing weight</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="CAMANboot_+3A_lambda21">lambda21</code></td>
<td>
<p>second means of the first column of the
observations</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="CAMANboot_+3A_lambda22">lambda22</code></td>
<td>
<p>second means of the second column of the observations</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="CAMANboot_+3A_prob2">prob2</code></td>
<td>
<p>second mixing weight</p>
</td></tr></table>
<p><br /> 
</p>
<table>
<tr><td><code id="CAMANboot_+3A_numiter">numiter</code></td>
<td>
<p>parameter to control the maximal number of iterations in the VEM and EM loops. Default is 5000.</p>
</td></tr> </table>
<p><br />
</p>
<table>
<tr><td><code id="CAMANboot_+3A_acc">acc</code></td>
<td>
<p>convergence criterion. Default is 1.e-7</p>
</td></tr> </table>
<p><br />
</p>
<table>
<tr><td><code id="CAMANboot_+3A_rep">rep</code></td>
<td>
<p>number of repetitions</p>
</td></tr></table>
<p><br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Parametric bootstrap for bivariate normally distributed data
data(CT)
library(mvtnorm)
hom1&lt;-c(3.142442)
hom2&lt;-c(-1.842393)
p1&lt;-c(1)
start1&lt;-c(2.961984,3.226141)
start2&lt;-c(-2.578836, -1.500823)
pvem&lt;-c(0.317,0.683)
CAMANboot(obs1=logitTPR, obs2=logitTNR, var1=varlogitTPR, var2=varlogitTNR,
          lambda11=hom1, lambda12=hom2, prob1=p1, 
          lambda21=start1, lambda22=start2, prob2=pvem,rep=3,data=CT)
</code></pre>

<hr>
<h2 id='CT'>Meta-anaysis: noninvasive coronary angiography using computed tomography (CT)</h2><span id='topic+CT'></span>

<h3>Description</h3>

<p> CT  for ruling out clinically significant coronary
artery disease (CAD) in adults with suspected or known CAD. The accuracy and clinical value of CT  was assessed in this meta-analysis.
</p>
<p>MEDLINE, EMBASE, and ISI Web of Science searches from inception through 2 June 2009 and bibliographies of reviews.
Prospective English- or German-language studies that compared CT or MRI with conventional coronary angiography in all patients and included
sufficient data for compilation of 2 x 2 tables. Two investigators independently extracted patient and study characteristics; differences were resolved by consensus.
89 studies comprising 7516 assessed the diagnostic value of CT. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("CT")</code></pre>


<h3>Format</h3>

<p>A data frame consisting of 91 data sets (rows) and 10 attributes (columns)
</p>

<p>Variable Names in order from left to right:
</p>
<dl>
<dt>Author</dt><dd><p>Author</p>
</dd> </dl>
<p><br /> 
</p>
<dl>
<dt>Year</dt><dd><p>Year </p>
</dd> </dl>
<p><br /> 
</p>
<dl>
<dt>TP</dt><dd><p>true positive</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>FP</dt><dd><p>False positive</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>FN</dt><dd><p>False negative </p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>TN</dt><dd><p>True negative</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>logitTPR</dt><dd><p>logit-true positive rate</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>logitTNR</dt><dd><p>logit-true negative rate</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>varlogitTPR</dt><dd><p>Variance of logit TPR</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>varlogitTNR</dt><dd><p>Variance of logit TPR</p>
</dd> </dl>
<p><br />
</p>



<h3>References</h3>

<p>Schuetz GM, Zacharopoulou NM, Schlattmann P, Dewey M. Meta-analysis:
noninvasive coronary angiography using computed tomography versus magnetic
resonance imaging. Ann Intern Med. 2010 Feb 2;152(3):167-77. doi:
10.7326/0003-4819-152-3-201002020-00008.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Use the EM-algorithm for a diagnostic meta-analysis based on a mixture 
#of bivariate  normal densities.
#Here fixed study specific variances are calculated based on logit 
#transformed sensitivity and specificity. 
data(CT)
p2 &lt;- c(0.4,0.6)
lamlog12 &lt;- c(2.93,3.22)
lamlog22 &lt;- c(2.5,1.5)

m0 &lt;- bivariate.EM(obs1=logitTPR,obs2=logitTNR,
                   var1=varlogitTPR,var2=varlogitTNR,
                   type="meta",lambda1=lamlog12,lambda2=lamlog22,
                   p=p2,data=CT,class="FALSE")
</code></pre>

<hr>
<h2 id='getFDR'>
Compute false discovery rates and related statistics
</h2><span id='topic+getFDR'></span>

<h3>Description</h3>

<p>This function is especially useful in the context of genetic data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getFDR(dat, threshold = 0.7, idxNotDiff = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getFDR_+3A_dat">dat</code></td>
<td>

<p>A mixture, i.e. a <code>CAMAN.object</code>.
</p>
</td></tr>
<tr><td><code id="getFDR_+3A_threshold">threshold</code></td>
<td>

<p>numeric, should be between 0 and 1 to be meaningful.
</p>
</td></tr>
<tr><td><code id="getFDR_+3A_idxnotdiff">idxNotDiff</code></td>
<td>

<p>integer, index of component for which computations are required.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See Schlattmann (2009) for more details.
</p>


<h3>Value</h3>

<p>A list with components <code>FDR</code>, <code>FNDR</code>, <code>FPR</code> and <code>FNR</code>.</p>


<h3>Author(s)</h3>

<p>Peter Schlattmann and Johannes Hoehne
</p>


<h3>References</h3>

<p>Schlattmann, P. (2009). <em>Medical Applications of Finite Mixture Models.</em> Berlin: Springer.
</p>

<hr>
<h2 id='golubMerge'>Data from the Golub et al (1999) Paper</h2><span id='topic+golubMerge'></span><span id='topic+golubMerge.exprs'></span><span id='topic+sample.labels'></span>

<h3>Description</h3>

<p>The data are from Golub et al. These are the combined training samples and test samples. There are 47 patients with acute lymphoblastic leukemia (ALL) and 25 patients with acute myeloid leukemia (AML). The samples were assayed using Affymetrix Hgu6800 chips and data on the expression of 7129 genes (Affymetrix probes) are available. The data were obtained from the Web site listed below and transformed slightly. Two objects are in the workspace when the data is loaded: <code>golubMerge.exprs</code> and <code>sample.labels</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("golubMerge")
</code></pre>


<h3>Format</h3>

<p>A matrix with 7129 rows (for the genes) and 72 columns (for the patients).
</p>


<h3>Note</h3>

<p>The data also appear in the Bioconductor package <code>golubEsets</code> in a different format. See Schlattmann (2009) for details on how to handle this type of data.</p>


<h3>Source</h3>

<p>This data is a variant of the data from the Bioconductor <code>golubEsets</code> package.</p>


<h3>References</h3>

<p>Molecular Classification of Cancer: Class Discovery and Class Prediction by Gene Expression Monitoring, Science, 531-537, 1999, T. R. Golub and D. K. Slonim and P. Tamayo and C. Huard and M. Gaasenbeek and J. P. Mesirov and H. Coller and M.L. Loh and J. R. Downing and M. A. Caligiuri and C. D. Bloomfield and E. S. Lander
</p>
<p>Schlattmann, P. (2009). <em>Medical Applications of Finite Mixture Models.</em> Berlin: Springer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## microarray analysis example
data(golubMerge)
idxALL &lt;- which(sample.labels== "ALL")
idxAML &lt;- which(sample.labels == "AML")
pvals &lt;- apply(golubMerge.exprs, 1, function(x){t.test(x[idxAML],x[idxALL])[[3]]})
zvals &lt;- qnorm(1-pvals)
hist(zvals,100)
### Z-values are gaussian distributed, mix identifies a mixture of gaussians. 
mix &lt;- mixalg(obs=zvals, family="gaussian", startk=25)
hist(mix)

### get False discovery rate (Not-differential expressed genes are in component 1)
getFDR(mix, threshold=.4)

## End(Not run)
</code></pre>

<hr>
<h2 id='heage'>
Northeast Thailand malnourishment data.
</h2><span id='topic+heage'></span>

<h3>Description</h3>

<p>When screening for subclinical malnourishment standardized scores like the HE/AGE (based on an international reference population) are recommended. HE/AGE is computed as the difference of the childs height minus the median height in the reference population divided by the standadard deviation. Measurements of 708 preschool children in  northeast Thailand are contained in this data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(heage)</code></pre>


<h3>Format</h3>

<p>A data frame with 708 observations on the following variable.
</p>

<dl>
<dt><code>value</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>References</h3>

<p>B\&quot;ohning, D.  and Schlattmann, P. and Lindsay, B. (1992). &ldquo;Computer-Assisted Analysis of Mixtures (C.A.MAN): Statistical Algorithms&rdquo;. <em>Biometrics</em>. <b>48</b>, 283&ndash;303.
</p>

<hr>
<h2 id='hepab'>
Hepatitis B data from Schlattmann and B\&quot;ohning (1993).
</h2><span id='topic+hepab'></span>

<h3>Description</h3>

<p>This data set appears as an example for an application of mixture modelling in disease mapping. The data is on hepatitis B cases in Berlin 1989.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hepab)</code></pre>


<h3>Format</h3>

<p>A data frame with 23 observations on the following 2 variables.
</p>

<dl>
<dt><code>observations</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>expected</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>References</h3>

<p>Schlattmann, P. and B\&quot;ohning, D. (1993). &ldquo;Mixture models and disease mapping&rdquo;. <em>Statistics in Medicine</em>. <b>12</b>, 1943&ndash;1950.
</p>

<hr>
<h2 id='hist.CAMAN.object'>Histograms for finite mixture models</h2><span id='topic+hist.CAMAN.object'></span>

<h3>Description</h3>

<p>Graphical display of the data and the mixture. Intended for larger data sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CAMAN.object'
hist(x, nbreaks=NULL, mixdens=TRUE, mixdens.col="red", 
                return.mixdens=FALSE, data.plot=NULL, singleDistr=TRUE, 
                main="", xlab="", plotlegend=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hist.CAMAN.object_+3A_x">x</code></td>
<td>
<p>A <code>CAMAN.object</code>.</p>
</td></tr>
<tr><td><code id="hist.CAMAN.object_+3A_nbreaks">nbreaks</code></td>
<td>
<p>Number of breaks in histogram. Defaults to <code>NULL</code>. In this case it is set to <code>min(60,max(30,x@num.obs))</code>. Passed on to <code><a href="graphics.html#topic+hist">hist</a></code>.</p>
</td></tr>
<tr><td><code id="hist.CAMAN.object_+3A_mixdens">mixdens</code></td>
<td>
<p>logical; plot mixture density?</p>
</td></tr>
<tr><td><code id="hist.CAMAN.object_+3A_mixdens.col">mixdens.col</code></td>
<td>
<p>color of mixture density.</p>
</td></tr>
<tr><td><code id="hist.CAMAN.object_+3A_return.mixdens">return.mixdens</code></td>
<td>
<p>logical; return the mixture density?</p>
</td></tr>
<tr><td><code id="hist.CAMAN.object_+3A_data.plot">data.plot</code></td>
<td>
<p>data to plot. Defaults to <code>NULL</code>. In this case it is taken from the data of the mixture density.</p>
</td></tr>
<tr><td><code id="hist.CAMAN.object_+3A_singledistr">singleDistr</code></td>
<td>
<p>logical; should the distributions of the single components be plotted?</p>
</td></tr>
<tr><td><code id="hist.CAMAN.object_+3A_main">main</code></td>
<td>
<p>character; heading of the plot.</p>
</td></tr>
<tr><td><code id="hist.CAMAN.object_+3A_xlab">xlab</code></td>
<td>
<p>character; label of x-axis.</p>
</td></tr>
<tr><td><code id="hist.CAMAN.object_+3A_plotlegend">plotlegend</code></td>
<td>
<p>logical; should a legend be plotted.</p>
</td></tr>
<tr><td><code id="hist.CAMAN.object_+3A_...">...</code></td>
<td>
<p>arguments to be passed on to <code><a href="graphics.html#topic+hist">hist</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function plots a histogram with the mixture density and its components overlaid. If <code>return.mixdens</code> is set to <code>TRUE</code>, the density at the breaks will also be returned.
</p>


<h3>Author(s)</h3>

<p>Peter Schlattmann and Johannes Hoehne   
</p>


<h3>References</h3>

<p>Schlattmann, P. (2009). <em>Medical Applications of Finite Mixture Models</em>. Berlin: Springer.
</p>

<hr>
<h2 id='leukDat'>Childhood leukemia in the former GDR</h2><span id='topic+leukDat'></span><span id='topic+GDRmap'></span>

<h3>Description</h3>

<p>In this example we present data from the former East Germany within the time
period from 1980 to 1989. The data are taken from the cancer atlas of the former
East Germany (Moehner et al. 1994). The data provided  the regional distribution of childhood leukemia in the former East Germany.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("leukDat")</code></pre>


<h3>References</h3>

<p>Schlattmann, P.(2009) <em>Medical Applications of Finite Mixture Models.</em> Berlin: Springer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## disease mapping example
data(leukDat)
data(GDRmap) # this is the map of the former GDR
mix.leuk &lt;- mixalg(obs = "oleuk", pop.at.risk = "eleuk",
                   data = leukDat, family = "poisson")
mix.leuk
plot(GDRmap, col = mix.leuk@classification)          
</code></pre>

<hr>
<h2 id='mixalg'>Fitting Finite Mixture Models</h2><span id='topic+mixalg'></span>

<h3>Description</h3>

<p>This hybrid mixture algorithm  combines the VEM algorithm  for flexible
support size and the EM algorithm for a fixed number of components.
The solution of the VEM algorithm provides starting values for the EM
algorithm. By the NPMLE theorem the EM algorithm
thus starts very close to the global maximum and proper convergence of the
EM algorithm to a global maximum is ensured.
</p>
<p>The algorithm proceeds as follows
</p>
<p>Step 1: Define an approximating grid <code>lambda[1], ..., lambda[L]</code>
</p>
<p>Step 2: Use the VEM algorithm to maximize L(P) in the simplex
<code class="reqn">\Omega</code> and identify grid points with positive support. 
Here positive support is defined as <code>p[j] &gt;= epsilon</code> (often <code>epsilon = 10^-2</code>).<br />
This gives an initial estimate of <code>k</code>.<br />
</p>
<p>Step 3: Use these <code>k</code> points and corresponding mixing weights
p[j] as starting values for the EM algorithm
</p>
<p>Step 4: Collapse identical components if <code>| lambda[j]- lambda[i] | &lt; delta </code> (often delta=0.05) for i != j
</p>
<p>Step 5: Obtain the final number of components <code>k</code>
</p>
<p>This sequential algorithm leads to an initial estimate of the  NPMLE
and a proper solution for the subsequent EM algorithm. Crucial points are
the definitions of <code class="reqn">\delta</code> and <code class="reqn">\epsilon</code>. Depending on these settings
different solutions could result from this algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mixalg(obs, weights=NULL, family="gaussian", data=NULL, pop.at.risk=NULL, 
       var.lnOR=NULL, limit=0.01, acc=10^(-7), numiter=5000, startk=50)
</code></pre>


<h3>Arguments</h3>

   
<table>
<tr><td><code id="mixalg_+3A_obs">obs</code></td>
<td>

<p>observed / dependent variable. Vector or colname of <code>data</code>. Must be specified!
</p>
</td></tr>
<tr><td><code id="mixalg_+3A_weights">weights</code></td>
<td>

<p>weights of the data. Vector or colname of <code>data</code>. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="mixalg_+3A_family">family</code></td>
<td>

<p>the underlying type density function as a character (&quot;gaussian&quot;, &quot;poisson&quot; or &quot;binomial&quot;)!
</p>
</td></tr>
<tr><td><code id="mixalg_+3A_data">data</code></td>
<td>

<p>an optional data frame. <code>obs</code>, <code>weights</code>, <code>pop.at.risk</code> and <code>var.lnOR</code> can be specified as column name of the data frame. 
</p>
</td></tr>      
<tr><td><code id="mixalg_+3A_pop.at.risk">pop.at.risk</code></td>
<td>

<p>population at risk: These data could be used to determine a mixture model for Poisson data. Vector or colname of <code>data</code>. Default is<code>NULL</code>. 
</p>
</td></tr>
<tr><td><code id="mixalg_+3A_var.lnor">var.lnOR</code></td>
<td>

<p>variances of the data: These variances might be given when working with meta analyses! Vector or colname of <code>data</code>. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="mixalg_+3A_limit">limit</code></td>
<td>

<p>parameter to control the limit of union several components. Default is 0.01.
</p>
</td></tr>
<tr><td><code id="mixalg_+3A_acc">acc</code></td>
<td>

<p>convergence criterion. VEM and EM loops stop when deltaLL&lt;acc. Default is 10^(-7). 
</p>
</td></tr>
<tr><td><code id="mixalg_+3A_numiter">numiter</code></td>
<td>

<p>parameter to control the maximal number of iterations in the VEM and EM loops. Default is 5000. 
</p>
</td></tr>
<tr><td><code id="mixalg_+3A_startk">startk</code></td>
<td>

<p>starting/maximal number of components. This number will be used to compute the grid in the VEM. Default is 50.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The documentation of <code><a href="#topic+leukDat">leukDat</a></code> contains a disease mapping example using <code>mixalg</code> and the documentation of <code><a href="#topic+golubMerge">golubMerge</a></code> contains a microarray analysis example.
</p>


<h3>Value</h3>

<p>The function returns a CAMAN.object, describing a finite mixture model.
The main information about the mixture model is printed by just typing the &lt;object&gt;. Additional information is given in <code>summary(object)</code> (summary.CAMAN.object). 
Single attributes can be accessed using the <code>@</code>, e.g. mix@LL.
<br /> 
</p>
<table>
<tr><td><code>dat</code></td>
<td>
<p>(input) data </p>
</td></tr></table>
<p><br />
</p>
<table>
<tr><td><code>family</code></td>
<td>
<p>underlying type density function  </p>
</td></tr>
<tr><td><code>LL</code></td>
<td>
<p>Likelihood of the final (best) iteration </p>
</td></tr>
<tr><td><code>BIC</code></td>
<td>
<p>Likelihood of the final (best) iteration </p>
</td></tr>
<tr><td><code>num.k</code></td>
<td>
<p>number of components obtained </p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>probability of each component</p>
</td></tr>
<tr><td><code>t</code></td>
<td>
<p>parameter of distribution (normal distr. -&gt; mean, poisson distr. -&gt; lambda, binomial distr. -&gt; prob) </p>
</td></tr>
<tr><td><code>component.var</code></td>
<td>
<p>variance of each component (ONLY if <code>family</code> == &quot;gaussian&quot;) </p>
</td></tr>
<tr><td><code>prob</code></td>
<td>
<p>probabilies, belonging to each component </p>
</td></tr>
<tr><td><code>classification</code></td>
<td>
<p>classification labels for each observation (<code>which.max</code> of <code>@prob</code>). </p>
</td></tr>
<tr><td><code>steps</code></td>
<td>
<p>number of steps performed (EM, VEM). </p>
</td></tr>
<tr><td><code>VEM_result</code></td>
<td>
<p>result of VEM algorithm. </p>
</td></tr>
<tr><td><code>cl</code></td>
<td>
<p>the matched call. </p>
</td></tr>
<tr><td><code>is_metaAnalysis</code></td>
<td>
<p>parameter specifying, whether a meta analysis was performed. </p>
</td></tr> 
<tr><td><code>VEM_result</code></td>
<td>
<p>Outcome of the VEM-algorithm, which was run before the EM.</p>
</td></tr>
<tr><td><code>finalacc</code></td>
<td>
<p>deltaLL of the final iteration (for VEM and EM)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Peter Schlattmann and Johannes Hoehne  
</p>


<h3>References</h3>

<p>D. B\&quot;ohning, P. Schlattmann, B.G. Lindsay: C.A.MAN - Computer Assisted
Analysis of Mixtures: Statistical Algorithms.<em>Biometrics</em>, 1992, 48, 283-303
</p>
<p>P. Schlattmann: On bootstrapping the unknown number of components in
finite mixtures of Poisson distributions. <em>Statistics and Computing</em>, 2005, 15, 179-188
</p>
<p>Schlattmann, P. (2009). <em>Medical Applications of Finite Mixture Models.</em> Berlin: Springer.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mixalg.EM">mixalg.EM</a></code>, <code><a href="#topic+mixalg.VEM">mixalg.VEM</a></code>, <code><a href="#topic+anova.CAMAN.object">anova.CAMAN.object</a></code>, <code><a href="#topic+mixcov">mixcov</a></code>, <code><a href="#topic+mixalg.boot">mixalg.boot</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### POISSON data with weights: thai_cohort
data(thai_cohort)
mix &lt;- mixalg(obs="counts", weights="frequency", family="poisson", 
              data=thai_cohort, numiter=18000, acc=0.00001, startk=25)


# meta analysis
data(aspirin)
mix &lt;- mixalg(obs="logrr", var.lnOR="var", data=aspirin)


## See the documentation of golub.Merge for a
## microarray analysis example using mixalg

## See the documentation of leukDat for a disease
## mapping example using mixalg

</code></pre>

<hr>
<h2 id='mixalg.boot'>bootstrap replication / validation of finite mixture models</h2><span id='topic+mixalg.boot'></span><span id='topic+mixalg.Boot'></span><span id='topic+mixboot'></span>

<h3>Description</h3>

<p>This function may be used to estimate the number of components based on a
nonparametric bootstrap approach.A bootstrap sample is obtained from the
original sample  with replacement. Corresponding to the bootstrap data set we
obtain an estimate of the number of components <code>k</code> applying a combination
of the VEM- and EM algorithm.
The bootstrap algorithm involves drawing <code>B</code> independent bootstrap samples
and estimating <code>k</code> using the hybrid mixture algorithm. The result is
the bootstrap distribution of the number of components <code>k</code>. 
The mode of this distribution is taken as an estimate of the number of components.   
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mixalg.boot(mix, nboot=500, limit=0.01, acc=10^(-5), numiter=5000, 
            startk=50, returnBootstrapRep=FALSE)    
</code></pre>


<h3>Arguments</h3>

<p>the parameters <code>limit, acc, numiter</code> and <code>startk</code> are used for the VEM algorithm in each bootstrap sample.
</p>
<table>
<tr><td><code id="mixalg.boot_+3A_mix">mix</code></td>
<td>

<p>A CAMAN-object which quantifies a finite mixture model.
</p>
</td></tr>
<tr><td><code id="mixalg.boot_+3A_nboot">nboot</code></td>
<td>

<p>number of bootstrap replications</p>
</td></tr>
<tr><td><code id="mixalg.boot_+3A_limit">limit</code></td>
<td>

<p>parameter to control the limit of union several components. Default is 0.01.
</p>
</td></tr>
<tr><td><code id="mixalg.boot_+3A_acc">acc</code></td>
<td>

<p>convergence criterion. VEM and EM loops stop when deltaLL&lt;acc. Default is 10^(-7). 
</p>
</td></tr>
<tr><td><code id="mixalg.boot_+3A_numiter">numiter</code></td>
<td>

<p>parameter to control the maximal number of iterations in the VEM and EM loops. Default is 5000. 
</p>
</td></tr>
<tr><td><code id="mixalg.boot_+3A_startk">startk</code></td>
<td>

<p>starting/maximal number of components for the VEM algorithm in each bootstrap sample. This number will be used to compute the grid in the VEM. Default is 50.
</p>
</td></tr>
<tr><td><code id="mixalg.boot_+3A_returnbootstraprep">returnBootstrapRep</code></td>
<td>

<p>A Boolean that indicates whether the bootstrapped data should be returned or not
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a list, describing the bootstrap replications: <br /> 
</p>
<table>
<tr><td><code>$dat.bootstrap</code></td>
<td>
<p> (used) sampled data </p>
</td></tr>
<tr><td><code>$LL</code></td>
<td>
<p> Likelihood of the final solutions of each bootstrap replication </p>
</td></tr>
<tr><td><code>$LL_k1</code></td>
<td>
<p> vector with LL for each bootstrap replication using a homogeneous model (k=1) </p>
</td></tr>
<tr><td><code>$numk.boot</code></td>
<td>
<p> number of components obtained in replications.</p>
</td></tr>
</table>


<h3>Note</h3>

<p><code>mixalg.Boot</code> and <code>mixboot</code>  are deprecated names for the <code>mixalg.boot</code> function.
</p>


<h3>Author(s)</h3>

<p>Peter Schlattmann and Johannes Hoehne   
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### POISSON data with weights: thai_cohort
data(thai_cohort)
m.thai &lt;- mixalg(obs="counts", weights="frequency",
                 family="poisson", data=thai_cohort,
                 acc = 0.00003)
## Not run: boot &lt;- mixalg.boot(m.thai, nboot=1000) #may take a few minutes

### POISSON data with observed and expected data: hepab
data(hepab)
mix &lt;- mixalg(obs="observations",pop.at.risk="expected",family= "poisson",data=hepab)
## Not run: boot &lt;- mixalg.boot(mix, nboot=250) #may take some time 
table(boot$numk.boot)
## End(Not run)
</code></pre>

<hr>
<h2 id='mixalg.EM'>EM algorithm</h2><span id='topic+mixalg.EM'></span>

<h3>Description</h3>

<p>In the fixed support size case the number of components <code>k</code> is assumed to
be known. Here the unknown parameters are the mixing weights <code>p[j]</code> and the
parameters <code>lambda[j]</code> of the subpopulation.<br />
</p>
<p>Estimation of these models' parameters is usually achieved by application
of the EM algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mixalg.EM (mix = NULL, p, t, obs = NULL, weights = NULL, family = "gaussian", 
    data = NULL, pop.at.risk = NULL, var.lnOR = NULL, limit = 0.01, 
    acc = 10^(-7), numiter = 5000)    
</code></pre>


<h3>Arguments</h3>

  
<table>
<tr><td><code id="mixalg.EM_+3A_mix">mix</code></td>
<td>

<p>A CAMAN-object which quantifies a finite mixture model and the input data.
</p>
</td></tr>
<tr><td><code id="mixalg.EM_+3A_p">p</code></td>
<td>
<p> vector containing the starting values for p</p>
</td></tr>
<tr><td><code id="mixalg.EM_+3A_t">t</code></td>
<td>
<p> vector containing the starting values for the distribution specific parameters (poisson-&gt;<code class="reqn">\lambda</code>, gaussian-&gt; <code class="reqn">\mu</code>)</p>
</td></tr>
<tr><td><code id="mixalg.EM_+3A_obs">obs</code></td>
<td>

<p>observed / dependent variable. Vector or colname of <code>data</code>. Must be specified!
</p>
</td></tr>
<tr><td><code id="mixalg.EM_+3A_weights">weights</code></td>
<td>

<p>weights of the data. Vector or colname of <code>data</code>. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="mixalg.EM_+3A_family">family</code></td>
<td>

<p>the underlying type density function as a character (&quot;gaussian&quot;, &quot;poisson&quot; or &quot;binomial&quot;)!
</p>
</td></tr>
<tr><td><code id="mixalg.EM_+3A_data">data</code></td>
<td>

<p>an optional data frame. <code>obs</code>, <code>weights</code>, <code>pop.at.risk</code> and <code>var.lnOR</code> can be specified as column name of the data frame. 
</p>
</td></tr>      
<tr><td><code id="mixalg.EM_+3A_pop.at.risk">pop.at.risk</code></td>
<td>

<p>population at risk: These data could be used to determine a mixture model for Poisson data. Vector or colname of <code>data</code>. Default is<code>NULL</code>. 
</p>
</td></tr>
<tr><td><code id="mixalg.EM_+3A_var.lnor">var.lnOR</code></td>
<td>

<p>variances of the data: These variances might be given when working with meta analyses! Vector or colname of <code>data</code>. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="mixalg.EM_+3A_limit">limit</code></td>
<td>

<p>parameter to control the limit of union several components. Default is 0.01.
</p>
</td></tr>
<tr><td><code id="mixalg.EM_+3A_acc">acc</code></td>
<td>

<p>convergence criterion. VEM and EM loops stop when deltaLL&lt;acc. Default is 10^(-7). 
</p>
</td></tr>
<tr><td><code id="mixalg.EM_+3A_numiter">numiter</code></td>
<td>

<p>parameter to control the maximal number of iterations in the VEM and EM loops. Default is 5000. 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a <code>CAMAN.object</code> object, thus the same as mixalg!
</p>


<h3>Author(s)</h3>

<p>Peter Schlattmann and Johannes Hoehne   
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(vitA)
m1&lt;-mixalg.EM(obs="logrr",var.lnOR="var" ,family="gaussian",
              p=c(1),t=c(0),data=vitA)
m2&lt;-mixalg.EM(obs="logrr",var.lnOR="var" ,family="gaussian",
              p=c(0.5,0.5),t=c(-0.3,0.2),data=vitA)

# apply EM-algorithm on an existing CAMAN.object: 
data(thai_cohort)
mix0 &lt;- mixalg(obs="counts", weights="frequency", family="poisson", 
               data=thai_cohort, numiter=18000, acc=0.00001,startk=25)
em0&lt;-mixalg.EM(mix0,p=c(1),t=c(1))
em1&lt;-mixalg.EM(mix0,p=c(0.7,0.3),t=c(2,9))

</code></pre>

<hr>
<h2 id='mixalg.VEM'>VEM algorithm</h2><span id='topic+mixalg.VEM'></span>

<h3>Description</h3>

<p>When fitting finite mixture models two cases must be distinguished. The
flexible support size case, where no assumption about the
number of components <code>k</code> is made in advance and the fixed support size
case. For the flexible support size case the VEM-algorithm can be used.
<br /> <br />
The algorithm proceeds as follows:<br />
Step 1: Define an approximating grid lambda[1], ..., lambda[L] <br />
Step 2: Use the VEM algorithm to maximize <code>L(P)</code> in the simplex 
<code class="reqn">\Omega</code> and identify grid points with positive support. <br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mixalg.VEM(mix = NULL, obs=NULL, weights=NULL, data=NULL, pop.at.risk=NULL, 
           var.lnOR=NULL, family="gaussian", limit=0.01, acc=10^(-7), 
           numiter=5000, startk=50)    
</code></pre>


<h3>Arguments</h3>

  
<table>
<tr><td><code id="mixalg.VEM_+3A_mix">mix</code></td>
<td>

<p>A CAMAN-object which quantifies a finite mixture model and the input data.
</p>
</td></tr>
<tr><td><code id="mixalg.VEM_+3A_obs">obs</code></td>
<td>

<p>observed / dependent variable. Vector or colname of <code>data</code>. Must be specified!
</p>
</td></tr>
<tr><td><code id="mixalg.VEM_+3A_weights">weights</code></td>
<td>

<p>weights of the data. Vector or colname of <code>data</code>. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="mixalg.VEM_+3A_family">family</code></td>
<td>

<p>the underlying type density function as a character (&quot;gaussian&quot;, &quot;poisson&quot; or &quot;binomial&quot;)!
</p>
</td></tr>
<tr><td><code id="mixalg.VEM_+3A_data">data</code></td>
<td>

<p>an optional data frame. <code>obs</code>, <code>weights</code>, <code>pop.at.risk</code> and <code>var.lnOR</code> can be specified as column name of the data frame. 
</p>
</td></tr>      
<tr><td><code id="mixalg.VEM_+3A_pop.at.risk">pop.at.risk</code></td>
<td>

<p>population at risk: These data could be used to determine a mixture model for Poisson data. Vector or colname of <code>data</code>. Default is<code>NULL</code>. 
</p>
</td></tr>
<tr><td><code id="mixalg.VEM_+3A_var.lnor">var.lnOR</code></td>
<td>

<p>variances of the data: These variances might be given when working with meta analyses! Vector or colname of <code>data</code>. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="mixalg.VEM_+3A_limit">limit</code></td>
<td>

<p>parameter to control the limit of union several components. Default is 0.01.
</p>
</td></tr>
<tr><td><code id="mixalg.VEM_+3A_acc">acc</code></td>
<td>

<p>convergence criterion. VEM and EM loops stop when deltaLL&lt;acc. Default is 10^(-7). 
</p>
</td></tr>
<tr><td><code id="mixalg.VEM_+3A_numiter">numiter</code></td>
<td>

<p>parameter to control the maximal number of iterations in the VEM and EM loops. Default is 5000. 
</p>
</td></tr>
<tr><td><code id="mixalg.VEM_+3A_startk">startk</code></td>
<td>

<p>starting/maximal number of components. This number will be used to compute the grid in the VEM. Default is 50.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a <code>CAMAN.VEM.object</code> object.
</p>


<h3>Author(s)</h3>

<p>Peter Schlattmann and Johannes Hoehne   
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(vitA)
m0&lt;-mixalg.VEM(obs="logrr",var.lnOR="var",family="gaussian", data=vitA,startk=20)
plot(m0@totalgrid[,2],m0@totalgrid[,3],type="l",xlab="parameter",ylab="gradient")
m1&lt;-mixalg.EM(obs="logrr",var.lnOR="var" ,family="gaussian",p=c(1),t=c(0),data=vitA)
</code></pre>

<hr>
<h2 id='mixcov'>Fitting mixture models with covariates</h2><span id='topic+mixcov'></span>

<h3>Description</h3>

<p>The function mixcov can be used to estimate the parameters of covariate
adjusted mixture models. This is done using the EM algorithm.
The function first performs the necessary data augmentation and then
applies the EM-algorithm. Covariates may be included as fixed and random
effects into the model. <br />
</p>
<p>Thus, the EM algorithm for covariate adjusted mixture models implies to
perform  first the necessary data augmentation, and then  based on
starting values for <code>p[j]</code> and <code>beta[j]</code> the computation of posterior
probabilities <code>e[ij]</code>.  This is the E-step. In the M-step new mixing
weights <code>p[j]</code> and regression coefficients <code>beta[j]</code> are computed. 
<br />
<br />
Then the algorithm performs as follows:<br /> \
Step 0: Let <code>P</code> be any vector of  starting values <br />
Step 1: Compute the E-step, that is estimate the probability of component 
membership for each observation <br />
Step 2: Compute the M-step, that is compute new mixing weights and model
coefficients <br />
Proceed until convergence is met.           
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mixcov(dep, fixed, random="", data, k, weight=NULL, pop.at.risk=NULL, 
       var.lnOR=NULL, family="gaussian", maxiter=50, 
       acc=10^-7, returnHomogeneousModel = FALSE) 
    </code></pre>


<h3>Arguments</h3>

 
<table>
<tr><td><code id="mixcov_+3A_dep">dep</code></td>
<td>

<p>dependent variable. Vector or colname of <code>data</code>. Must be specified!
</p>
</td></tr>
<tr><td><code id="mixcov_+3A_fixed">fixed</code></td>
<td>

<p>fixed effects. Vector or colname of <code>data</code>. Must be specified!
</p>
</td></tr>
<tr><td><code id="mixcov_+3A_random">random</code></td>
<td>

<p>random effects. Vector or colname of <code>data</code>. Must be specified!
</p>
</td></tr>
<tr><td><code id="mixcov_+3A_k">k</code></td>
<td>

<p>number of components.
</p>
</td></tr> 
<tr><td><code id="mixcov_+3A_weight">weight</code></td>
<td>

<p>weights of the data. Vector or colname of <code>data</code>. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="mixcov_+3A_family">family</code></td>
<td>

<p>the underlying type density function as a character (&quot;gaussian&quot; or &quot;poisson&quot;)!
</p>
</td></tr>
<tr><td><code id="mixcov_+3A_data">data</code></td>
<td>

<p>an optional data frame. <code>obs</code>, <code>weights</code>, <code>pop.at.risk</code> and <code>var.lnOR</code> can be specified as column name of the data frame. 
</p>
</td></tr>      
<tr><td><code id="mixcov_+3A_pop.at.risk">pop.at.risk</code></td>
<td>

<p>population at risk: An offset that could be used to determine a mixture model for Poisson data from unequally large populations at risk. Vector or colname of <code>data</code>. Default is<code>NULL</code>. 
</p>
</td></tr>
<tr><td><code id="mixcov_+3A_var.lnor">var.lnOR</code></td>
<td>

<p>variances of the data: These variances might be given when working with meta analyses! Vector or colname of <code>data</code>. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="mixcov_+3A_maxiter">maxiter</code></td>
<td>

<p>parameter to control the maximal number of iterations in the VEM and EM loops. Default is 5000. 
</p>
</td></tr>
<tr><td><code id="mixcov_+3A_acc">acc</code></td>
<td>

<p>convergence criterion. VEM and EM loops stop when deltaLL&lt;acc. Default is 10^(-7). 
</p>
</td></tr>
<tr><td><code id="mixcov_+3A_returnhomogeneousmodel">returnHomogeneousModel</code></td>
<td>

<p>boolean to indicate whether the homogeneous model (simple glm) should be returned too. Default is FALSE.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a CAMAN.glm.object (S4-class), describing a mixture model with covariates.
The main information about the mixture model is printed by just typing the &lt;object&gt;. Additional information is given in <code>summary(object)</code>. 
Single attributes can be accessed using the <code>@</code>, e.g. mix@LL.
</p>
<table>
<tr><td><code>dat</code></td>
<td>
<p>(input) data </p>
</td></tr></table>
<p><br />
</p>
<table>
<tr><td><code>family</code></td>
<td>
<p>underlying type density function  </p>
</td></tr>
<tr><td><code>LL</code></td>
<td>
<p>Likelihood of the final (best) iteration </p>
</td></tr>
<tr><td><code>BIC</code></td>
<td>
<p>Likelihood of the final (best) iteration </p>
</td></tr>
<tr><td><code>num.k</code></td>
<td>
<p>number of components obtained </p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>probability of each component</p>
</td></tr>
<tr><td><code>t</code></td>
<td>
<p>parameter of distribution (normal distr. -&gt; mean, poisson distr. -&gt; lambda, binomial distr. -&gt; prob) </p>
</td></tr>
<tr><td><code>coefMatrix</code></td>
<td>
<p>complete coefficient matrix </p>
</td></tr>
<tr><td><code>prob</code></td>
<td>
<p>probabilies, belonging to each component </p>
</td></tr>
<tr><td><code>steps</code></td>
<td>
<p>number of steps performed (EM). </p>
</td></tr>
<tr><td><code>commonEffect</code></td>
<td>
<p>common effects</p>
</td></tr>
<tr><td><code>hetvar</code></td>
<td>
<p>heterogeneity variance</p>
</td></tr>
<tr><td><code>commonEffect</code></td>
<td>
<p>common effects</p>
</td></tr>
<tr><td><code>classification</code></td>
<td>
<p>classification labels for each observation (<code>which.max</code> of prob). </p>
</td></tr>
<tr><td><code>cl</code></td>
<td>
<p>the matched call. </p>
</td></tr>
<tr><td><code>fittedObs</code></td>
<td>
<p>predictions of the fitted model, given the observations</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Peter Schlattmann and Johannes Hoehne  
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Toy data: simulate subjects with a different relationship between age and salariy
  grps = sample(1:3,70, replace=TRUE) #assign each person to one group 
  salary=NULL
  age = round(runif(70) * 47 + 18)
  #random effects: age has a different influence (slope) on the salary
  salary[grps == 1] = 2000 + 12 * age[grps==1]
  salary[grps == 2] = 4000 + 4 * age[grps==2]
  salary[grps == 3] = 3200 + (-15) * age[grps==3]
  salary = salary + rnorm(70)*30  #some noise
  sex =sample(c("m","w"), 70, replace=TRUE)
  salary[sex=="m"] = salary[sex=="m"] * 1.2 #men earn 20 percent more than women
  salaryData = data.frame(salary=salary, age=age, sex=sex)
  tstSalary &lt;- mixcov(dep="salary", fixed="sex", random="age" ,data=salaryData,
                      k=3,family="gaussian", acc=10^-3)
 

### POISSON data:
data(NoP)
ames3 &lt;- mixcov(dep="count",fixed=c("dose", "logd"),random="",data=NoP,
                k=3,family="poisson")

### Gaussian data
data(betaplasma)
beta4 &lt;- mixcov(dep="betacaro", fixed=c("chol","sex","bmi"), random="betadiet",
                data=betaplasma, k=4, family="gaussian")
</code></pre>

<hr>
<h2 id='NoP'>Ames test data: Mutagenicity of 4NoP</h2><span id='topic+NoP'></span>

<h3>Description</h3>

<p>Ames test data where a strain of SalmonellaTA98 was activated with a homogenate of rat liver cells and exposed to 4-nitro-ophenylenediamine (4NoP) with various doses. 4NoP is a nitrated aromatic amine.  This chemical is a component of both semipermanent and permanent hair dye formulations. The substance is frequently used as a comparator in mutagenicity tests. The are  taken from Margolin et al. (1989) and denote the bacteria count for various doses of4NoP. The primary question of interest is whether 4NoP acts mutagenically in the Ames test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("NoP")</code></pre>


<h3>Format</h3>

<p>A data frame with 100 rows and 3 columns (dose, count, logd)</p>


<h3>References</h3>

<p>Margolin, B. H., B. S. Kim, and K. J. Risko. (1989). &ldquo;The Ames Salmonella Microsome Mutagenicity Assay: Issues of Inference and Validation.&rdquo; <em>Journal of the American Statistical Association</em> <b>84</b>, 651&ndash;661.
</p>
<p>Schlattmann, P.(2009). <em>Medical Applications of Finite Mixture Models.</em> Berlin: Springer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(NoP)
ames3 &lt;- mixcov(dep="count",fixed=c("dose", "logd"),random="",data=NoP,
                k=3,family="poisson")
</code></pre>

<hr>
<h2 id='PCT'>Procalcitonin as diagnostic marker for sepsis</h2><span id='topic+PCT'></span>

<h3>Description</h3>

<p>Procalcitonin is a promising marker for identification of bacterial infections. The accuracy and clinical value of procalcitonin for
diagnosis of sepsis in critically ill patients was assessed in this meta-analysis.
</p>
<p>Medline, Embase, ISI Web of Knowledge, the Cochrane Library, Scopus, BioMed Central, and Science Direct were searched, from inception to Feb 21, 2012,
and reference lists of identified primary studies. We included articles written in English, German, or French that investigated procalcitonin
for differentiation of septic patients&ndash;those with sepsis, severe sepsis, or septic shock&ndash;from those with a systemic inflammatory response syndrome of 
non-infectious origin. Studies of healthy people, patients without probable infection, and children younger than 28 days were excluded. 
Two independent investigators extracted patient and study characteristics; discrepancies were resolved by consensus. 
</p>
<p>This search returned 3487 reports, of which 30 fulfilled the inclusion criteria, accounting for 3244 patients. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("PCT")</code></pre>


<h3>Format</h3>

<p>A data frame consisting of 35 data sets (rows) and 10 attributes (columns)
</p>

<p>Variable Names in order from left to right:
</p>
<dl>
<dt>Study</dt><dd><p>Study</p>
</dd> </dl>
<p><br /> 
</p>
<dl>
<dt>Year</dt><dd><p>Year</p>
</dd> </dl>
<p><br /> 
</p>
<dl>
<dt>TP</dt><dd><p>True positive</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>FP</dt><dd><p>False positive</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>FN</dt><dd><p>False negative </p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>TN</dt><dd><p> True negative</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>logitTPR</dt><dd><p>logit-true positive rate</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>logitTNR</dt><dd><p>logit-true negative rate</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>varlogitTPR</dt><dd><p>Variance of logit TPR</p>
</dd> </dl>
<p><br />
</p>
<dl>
<dt>varlogitTNR</dt><dd><p>Variance of logit TPR</p>
</dd> </dl>
<p><br />
</p>



<h3>References</h3>

<p>Wacker C, Prkno A, Brunkhorst FM, Schlattmann P. Procalcitonin as a diagnostic
marker for sepsis: a systematic review and meta-analysis. Lancet Infect Dis. 2013
May;13(5):426-35. doi: 10.1016/S1473-3099(12)70323-7
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Use the VEM-algorithm for a diagnostic meta-analysis based on a mixture 
#of bivariate  normal densities.
#Study specific fixed variances are based on logit transformed 
#sensitivity and specificity. 

data(PCT)
names(PCT)
   
m0 &lt;- bivariate.VEM(obs1 = logitTPR, obs2 = logitTNR,
                    var1 = varlogitTPR, var2 = varlogitTNR,
                    type = "meta", data=PCT, startk=20)
</code></pre>

<hr>
<h2 id='plot.CAMAN.BIMIXALG.object'>Plot ellipses</h2><span id='topic+plot.CAMAN.BIMIXALG.object'></span><span id='topic+plot.CAMAN.BIEM.object'></span>

<h3>Description</h3>

<p>Graphical display of the EM algorithm and bivariate MIXALG</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CAMAN.BIMIXALG.object'
plot(x, ellipse, ...)
## S3 method for class 'CAMAN.BIEM.object'
plot(x, ellipse, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.CAMAN.BIMIXALG.object_+3A_x">x</code></td>
<td>
<p>a <code>CAMAN.BIEM.object</code> or a <code>CAMAN.BIMIXALG.object</code></p>
</td></tr>
<tr><td><code id="plot.CAMAN.BIMIXALG.object_+3A_ellipse">ellipse</code></td>
<td>
<p>logical. Plot ellipse? Defaults is <code>TRUE</code></p>
</td></tr>
<tr><td><code id="plot.CAMAN.BIMIXALG.object_+3A_...">...</code></td>
<td>
<p>further arguments. Currently ignored.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Peter Schlattmann 
</p>


<h3>References</h3>

<p>Schlattmann, P. (2009). <em>Medical Applications of Finite Mixture Models</em>. Berlin: Springer.
</p>

<hr>
<h2 id='rs12363681'>Gene calling</h2><span id='topic+rs12363681'></span>

<h3>Description</h3>

<p>This data set can be used to perform a cluster analyis of bivariate data.  <br />
</p>
<p>This data set contains SNP of 3680 individuala 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("rs12363681")</code></pre>


<h3>Format</h3>

<p>A data frame consisting of 3680 data sets (rows) and 2 attributes (columns)</p>


<h3>References</h3>

<p>Schlattmann, P.(2009) <em>Medical Applications of Finite Mixture Models.</em> Berlin: Springer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Example
# EM and classification for bivariate data
data(rs12363681)
test &lt;- bivariate.mixalg(obs1=x, obs2=y, type="bi", 
                         lambda1=0, lambda2=0, p=0, 
                         data=rs12363681, startk=20, class="TRUE")
# scatter plot with ellipse
plot(test)
# scatter plot without ellipse
plot(test, ellipse = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='thai_cohort'>Cohort study in north east Thailand</h2><span id='topic+thai_cohort'></span>

<h3>Description</h3>

<p>In a cohort study in northeast Thailand the health status of 602 preschool children
was checked every 2 weeks from June 1982 until September 1985 (Schelp et al.
1990). In this time period it was recorded how often the children showed symptoms
of fever, cough, running nose, or these symptoms together. The frequencies of these
illness spells are given in the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("thai_cohort")</code></pre>


<h3>References</h3>

<p>Schelp, F., P. Vivatanasept, P. Sitaputra, S. Sornmani, P. Pongpaew, N. Vudhivai, S. Egormaiphol, and D. B\&quot;ohning.  &ldquo;Relationship of the morbidity of under-fives to anthropometric measurements and community health intervention.&rdquo; <em>Trop Med Parasitol</em>, 1990, 41(2), 121&ndash;126.
</p>
<p>Schlattmann, P. (2009) <em>Medical Applications of Finite Mixture Models</em>. Berlin: Springer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("thai_cohort")
mix &lt;- mixalg(obs="counts", weights="frequency", family="poisson", data=thai_cohort,
              numiter=18000, acc=0.00001, startk=25)
</code></pre>

<hr>
<h2 id='vitamin'>Meta-analysis if vitamin A supplementation of childhood mortality</h2><span id='topic+vitamin'></span><span id='topic+vitA'></span><span id='topic+vitA2'></span>

<h3>Description</h3>

<p>Supplementation of vitamins is not only supposed to be beneficial for the prevention
of cancer as discussed in Sect. 2.1. Fawzi et al. (1993) study the effect of vitamin
A supplementation and childhood mortality in preschool children. They concluded
that vitamin A supplements are associated with a significant reduction in mortality
when given periodically to children at the community level.  All studies were
community randomized trials from South Asia or Southeast Asia, except the second
study, which was from northern Sudan. This data set contains the log relative risk and corresponding variance. The data is supplied in three different formats
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("vitamin")
data("vitA")
data("vitA2")
</code></pre>


<h3>References</h3>

<p>Schlattmann, P. (2009) <em>Medical Applications of Finite Mixture Models.</em> Berlin: Springer.
</p>
<p>Fawzi,W.W., T. C. Chalmers, M. G. Herrera, and F. Mosteller (1993). &ldquo;Vitamin A supplementation and child mortality. A meta-analysis.&rdquo; <em>JAMA</em> <b>269</b>(7), 898&ndash;903.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
