<!DOCTYPE html><html><head><title>Help for package CAMML</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CAMML}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BuildGeneSets'><p>Build Gene Sets for reference data to be applied to CAMML</p></a></li>
<li><a href='#CAMML'><p>Cell-typing using variance Adjusted Mahalanobis distances with Multi-Labeling (CAMML)</p></a></li>
<li><a href='#ChIMP'><p>CAMML with the Integration of Marker Proteins (ChIMP)</p></a></li>
<li><a href='#GetCAMMLLabels'><p>Single- and Multi-Labelling for CAMML</p></a></li>
<li><a href='#GetGeneSets'><p>Get Pre-Built Gene Sets to be applied to CAMML</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Cell-Typing using Variance Adjusted Mahalanobis Distances with
Multi-Labeling</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Courtney Schiebout &lt;courtney.t.schiebout.gr@dartmouth.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Creates multi-label cell-types for single-cell RNA-sequencing data based on weighted VAM scoring of cell-type specific gene sets. Schiebout, Frost (2022) <a href="https://psb.stanford.edu/psb-online/proceedings/psb22/schiebout.pdf">https://psb.stanford.edu/psb-online/proceedings/psb22/schiebout.pdf</a>.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Copyright:</td>
<td>Dartmouth College</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>VAM, Seurat (&ge; 4.0.0), MASS, Matrix (&ge; 1.3.3), utils,
BiocManager, org.Hs.eg.db, org.Dr.eg.db, org.Mm.eg.db,
AnnotationDbi, SeuratObject (&ge; 4.0.0), methods, edgeR</td>
</tr>
<tr>
<td>Suggests:</td>
<td>sctransform (&ge; 0.3.2)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-06 22:25:52 UTC; ctschiebout</td>
</tr>
<tr>
<td>Author:</td>
<td>H. Robert Frost [aut],
  Courtney Schiebout
    <a href="https://orcid.org/0000-0002-4830-8237"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-13 15:03:24 UTC</td>
</tr>
</table>
<hr>
<h2 id='BuildGeneSets'>Build Gene Sets for reference data to be applied to CAMML</h2><span id='topic+BuildGeneSets'></span>

<h3>Description</h3>

<p>BuildGeneSets takes in an expression matrix, either as a Seurat Object or as a simple matrix (where cells are the columns, and genes the rows), and labels for each of the cells. EdgeR differential expression analysis is then run within the function and gene sets are built based on a log fold change (FC) cut-off (the default is 2). Cut-offs can also be set by FC and -log10(p-value). Of note, when more than one cut-off is given, genes must meet ALL criteria. Gene weights are recorded for each gene in the gene set as the log2(FC), FC, or -log10(p-value) (the default is log2(FC)). Each gene in the gene set is converted into its corresponding Ensembl ID, necessitating that users also provide the species of interest. Currently humans &quot;Hs&quot;, mice &quot;Mm&quot;, and zebrafish &quot;Dr&quot; are accepted (the default is humans).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    BuildGeneSets(exp.data, labels = as.character(Idents(exp.data)), 
    cutoff.type = "logfc", cutoff = 2, species = "Hs", weight.type = "logfc")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BuildGeneSets_+3A_exp.data">exp.data</code></td>
<td>
<p>A Seurat Object or expression matrix for the reference data that has previously been normalized and scaled.</p>
</td></tr> 
<tr><td><code id="BuildGeneSets_+3A_labels">labels</code></td>
<td>
<p>A vector of the cell type labels for each cell in the expression matrix. This must have a label for each cell and be in the order the cells appear as columns in the expression matrix. The default will be the Idents of the Seurat Object.</p>
</td></tr> 
<tr><td><code id="BuildGeneSets_+3A_cutoff.type">cutoff.type</code></td>
<td>
<p>One or more of the following: &quot;logfc&quot;, &quot;fc&quot;, and &quot;-logp&quot;. This value will determine what value(s) genes must have to be included in a given gene set. &quot;logfc&quot; and &quot;fc&quot; cut-off genes based on their log fold change and fold change values respectively. &quot;-logp&quot; will cut-off genes based on the significance of their differential expression according to the -log10(p-value). When more than one cut-off is given, genes must meet ALL criteria.</p>
</td></tr>
<tr><td><code id="BuildGeneSets_+3A_cutoff">cutoff</code></td>
<td>
<p>A number or vector of numbers that correspond to the cutoff.types. The value should be greater than 0 if cutoff.type is &quot;logfc&quot;, greater than 1 if cutoff.type is &quot;fc&quot;, and greater than 1 if cutoff.type is &quot;-logp&quot;. The number of values given should match the number of cutoff types listed and should be in the same order as the cutoff types vector.</p>
</td></tr>
<tr><td><code id="BuildGeneSets_+3A_species">species</code></td>
<td>
<p>Either &quot;Hs&quot;, &quot;Mm&quot;, or &quot;Dr&quot; for human, mouse, and zebrafish respectively. Used to convert gene symbols into Ensembl IDs.</p>
</td></tr>
<tr><td><code id="BuildGeneSets_+3A_weight.type">weight.type</code></td>
<td>
<p>Can be one of the following: &quot;logfc&quot;,&quot;fc&quot;, or &quot;-logp&quot;. The former two options will assign gene weights by their log2FC or FC respectively in differential expression analysis. The final option will assign weight by the negative log10 of the p-values for each gene from differential expression analysis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> with the cell type, gene name, ensembl ID, and weight for each gene in each gene set.
</p>


<h3>See Also</h3>

<p><code>edgeR,org.Hs.eg.db,org.Mm.eg.db</code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#Only run code if Seurat package is available
if (requireNamespace("Seurat", quietly=TRUE) &amp; requireNamespace("SeuratObject", quietly=TRUE)) {
  #See vignettes for more examples
  BuildGeneSets(exp.data=SeuratObject::pbmc_small, 
  labels = c(rep(1,40),rep(2,40)), cutoff.type = "logfc", 
  cutoff = 2, species = "Hs", weight.type = "logfc")
}
</code></pre>

<hr>
<h2 id='CAMML'>Cell-typing using variance Adjusted Mahalanobis distances with Multi-Labeling (CAMML)</h2><span id='topic+CAMML'></span>

<h3>Description</h3>

<p>Multi-label cell-typing method for single-cell RNA-sequencing data. CAMML takes in cell type-specific gene sets, with weights for each gene, and builds weighted Variance-Adjusted Mahalanobis (VAM) scores for each of them. CAMML then outputs a Seurat Object with an assay for CAMML that has the weighted VAM score for each cell type in each cell. CAMML takes in several arguments: <code>seurat</code>: a Seurat Object of the scRNA-seq data, <code>gene.set.df</code>: a data frame with a row for each gene and the following required columns: &quot;ensemb.id&quot; and &quot;cell.type&quot; and optional columns of &quot;gene.weight&quot; and &quot;gene.symbol&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    CAMML(seurat, gene.set.df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CAMML_+3A_seurat">seurat</code></td>
<td>
<p>A Seurat Object that has previously been normalized and scaled.</p>
</td></tr> 
<tr><td><code id="CAMML_+3A_gene.set.df">gene.set.df</code></td>
<td>
<p>A list of lists of genes in each gene set, with each gene set list named for the cell type it represents.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>SeuratObject</code> with a CAMML assay with the scores for each cell type in each cell. This will be in the form of a matrix with columns for each cell and rows for each cell type that was scored.
</p>


<h3>See Also</h3>

<p><code><a href="VAM.html#topic+vamForSeurat">vamForSeurat</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># Only run example code if Seurat package is available
if (requireNamespace("Seurat", quietly=TRUE) &amp; requireNamespace("SeuratObject", quietly=TRUE)) {
  # See vignettes for more examples
  seurat &lt;- CAMML(seurat=SeuratObject::pbmc_small,
  gene.set.df = data.frame(cbind(ensembl.id = c("ENSG00000172005",
  "ENSG00000173114","ENSG00000139187"),
  cell.type = c("T cell","T cell","T cell"))))
  seurat@assays$CAMML@data
}
</code></pre>

<hr>
<h2 id='ChIMP'>CAMML with the Integration of Marker Proteins (ChIMP)</h2><span id='topic+ChIMP'></span>

<h3>Description</h3>

<p>ChIMP takes in the output of CAMML and a list of the CITE-seq markers designated for each cell type. For each marker, a k=2 means clustering will be applied to discretize their presence, resulting in a 0 in cells where the marker expression is in the lower value cluster and a 1 in cells where the marker expression is in the higher value cluster. Additionally, if a quantile cutoff is desired instead, this method can be designated and a cutoff can be given (the default is .5). These discretized scores are then multiplied by the CAMML score for each cell type in each cell. The function also takes in a vector of booleans the length of the number of cell types being evaluated that designates whether each cell type is required to have all markers score 1 or any marker score a 1 in order for the CAMML score to be maintained. If the boolean is true, ChIMP will weight CAMML by the maximum marker score for each cell type. For example, if both CD4 and CD8 are listed markers for T cells and either marker scoring a 1 is sufficient, the boolean will be true. If it is false, all markers designated for a cell type need to be in the higher value cluster for a given cell. ChIMP can also use the absence of a CITE-seq marker as support for a cell type by designating it &quot;FALSE&quot; with the greater argument. For example, if one is looking to identify non-immune cell types, CD45 can be used with greater = FALSE to support cell-type scores for a non-immune cell type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    ChIMP(seurat, citelist, method = "k", cutoff = .5, 
    anyMP = rep(T, length(rownames(seurat))),
    greater = rep(T, length(unlist(citelist))))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ChIMP_+3A_seurat">seurat</code></td>
<td>
<p>A Seurat Object that has previously been run on CAMML.</p>
</td></tr> 
<tr><td><code id="ChIMP_+3A_citelist">citelist</code></td>
<td>
<p>A list of all the surface markers for each cell type, named by their cell type.</p>
</td></tr>
<tr><td><code id="ChIMP_+3A_method">method</code></td>
<td>
<p>Either a &quot;k&quot; or a &quot;q&quot; to designate the desired method. &quot;k&quot; will use a k=2 k-means clustering method for discretization. &quot;q&quot; will use a quantile cutoff method.</p>
</td></tr> 
<tr><td><code id="ChIMP_+3A_cutoff">cutoff</code></td>
<td>
<p>A value between 0 and 1 designating the cutoff to be used if the quantile method is selected.</p>
</td></tr> 
<tr><td><code id="ChIMP_+3A_anymp">anyMP</code></td>
<td>
<p>A vector of booleans regarding whether the CITE-seq weighting will take any positive marker protein score (TRUE) or requires all positive marker scores (FALSE)</p>
</td></tr>
<tr><td><code id="ChIMP_+3A_greater">greater</code></td>
<td>
<p>A vector of booleans for every CITE-seq marker designating whether to evaluate it as present (TRUE) in a cell type or absent (FALSE) in a cell type.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>SeuratObject</code> with a ChIMP assay with the scores for each cell type in each cell, weighted by their CITE-seq score. This will be in the form of a matrix with columns for each cell and rows for each cell type that was scored.
</p>


<h3>See Also</h3>

<p><code><a href="VAM.html#topic+vamForSeurat">vamForSeurat</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># Only run example code if Seurat package is available
if (requireNamespace("Seurat", quietly=TRUE) &amp;&amp; 
requireNamespace("SeuratObject", quietly=TRUE)) {
  # See vignettes for more examples
  seurat &lt;- CAMML(seurat=SeuratObject::pbmc_small,
  gene.set.df = data.frame(cbind(ensembl.id = c("ENSG00000172005",
  "ENSG00000173114","ENSG00000139187"),
  cell.type = c("T cell","T cell","T cell"))))
  cite &lt;- matrix(c(rnorm(40), rnorm(40,2,1)),
  nrow = length(rownames(seurat@assays$CAMML)),
  ncol = length(colnames(seurat@assays$CAMML)))
  rownames(cite) &lt;- "marker"
  colnames(cite) &lt;- colnames(seurat)
  assay &lt;- SeuratObject::CreateAssayObject(counts = cite)
  seurat[["ADT"]] &lt;- assay
  citelist &lt;- list()
  citelist[[1]] = "marker"
  names(citelist) = "T cell"
  seurat &lt;- ChIMP(seurat, citelist)
  seurat@assays$ChIMP@data
}
</code></pre>

<hr>
<h2 id='GetCAMMLLabels'>Single- and Multi-Labelling for CAMML</h2><span id='topic+GetCAMMLLabels'></span>

<h3>Description</h3>

<p>This function takes in the Seurat Object output from the CAMML function and returns one of four labelling options. &quot;top1&quot; will return the top cell type for each cell. &quot;top2&quot; will return the top two highest scoring cell types for each cell. &quot;top10p&quot; will return the top scoring cell type and all other cell types with 10% of that score for each cell. &quot;2xmean&quot; will return all cell types with scores greater than twice the mean of all scores for a given cell.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetCAMMLLabels(seurat, labels = "top1")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetCAMMLLabels_+3A_seurat">seurat</code></td>
<td>
<p>A Seurat Object with a CAMML assay with weighted VAM scores for each cell type in each query cell. This is the output from the CAMML function.</p>
</td></tr>
<tr><td><code id="GetCAMMLLabels_+3A_labels">labels</code></td>
<td>
<p>One of the following: &quot;top1&quot;, &quot;top2&quot;, &quot;top10p&quot;, or &quot;top2xmean&quot;. 
&quot;top1&quot; will return the single-label for the top-scoring cell type for each cell.
&quot;top2&quot; will return the labels for the two top-scoring cell types for each cell.
&quot;top10p&quot; will return the top scoring cell type and any other cell types with scores within 10% of the top score for each cell.
&quot;top2xmean&quot; will return any cell types with scores two times the average of all cell type scores for each cell.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the labels designated by the &quot;labels&quot; argument.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CAMML">CAMML</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># Only run example code if Seurat and CAMML packages are available
if (requireNamespace("Seurat", quietly=TRUE) &amp; 
requireNamespace("SeuratObject", quietly=TRUE) &amp; 
requireNamespace("CAMML", quietly=TRUE)) {
  # See vignettes for more examples
  seurat &lt;- CAMML(seurat=SeuratObject::pbmc_small,
  gene.set.df=data.frame(cbind(ensembl.id = c("ENSG00000172005",
  "ENSG00000173114","ENSG00000139187"),
  cell.type = c("T cell","T cell","T cell"))))
  GetCAMMLLabels(seurat, labels = "top1")
}
</code></pre>

<hr>
<h2 id='GetGeneSets'>Get Pre-Built Gene Sets to be applied to CAMML</h2><span id='topic+GetGeneSets'></span>

<h3>Description</h3>

<p>GetGeneSets takes in a one of the following: &quot;immune.cells&quot;,&quot;skin.immune.cells&quot;,&quot;T.subset.cells&quot;, or &quot;mouse.cells&quot; and a Seurat Object that will be cell-typed using CAMML. The function will then build a gene.set.collection and a list of gene.weights based on one of the pre-built gene sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    GetGeneSets(data = "immune.cells")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetGeneSets_+3A_data">data</code></td>
<td>
<p>One of the following: &quot;immune.cells&quot;,&quot;skin.immune.cells&quot;,&quot;T.subset.cells&quot;, or &quot;mouse.cells&quot;. 
</p>

<ul>
<li><p> &quot;immune.cells&quot; provides gene sets for 5 human immune cells: B, T, NK, Monocytes, and HSCs. 
</p>
</li>
<li><p> &quot;skin.immune.cells&quot; provides gene sets for 6 human cell types: B, T, NK, Endothelial, Fibroblast, and Monocytes.
</p>
</li>
<li><p> &quot;T.subset.cells&quot; provides 6 gene sets for 5 human T cell subtypes: naive CD4, CD4, naive CD8, CD8, memory, and regulatory T cells.
</p>
</li>
<li><p> &quot;mouse.cells&quot; provides gene sets for 7 mouse cell types: B, T, NK, DC, Endothelial, Fibroblasts, and Macrophages.
</p>
</li></ul>

<p>All datasets were built using differential expression of data in the package celldex using the package EdgeR.</p>
</td></tr> 
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> with the cell type, gene name, ensembl ID, and weight for each gene in each gene set.
</p>


<h3>See Also</h3>

<p><code>org.Hs.eg.db,org.Mm.eg.db</code></p>


<h3>Examples</h3>

<pre><code class='language-R'>GetGeneSets("immune.cells")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
