<!DOCTYPE html><html lang="en"><head><title>Help for package Canopy</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Canopy}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#addsamptree'>
<p>To determine whether the sampled tree will be accepted</p></a></li>
<li><a href='#AML43'>
<p>SNA input for primary tumor and relapse genome of leukemia patient from</p>
Ding et al. Nature 2012.</a></li>
<li><a href='#canopy.BIC'>
<p>To get BIC as a model selection criterion</p></a></li>
<li><a href='#canopy.cluster'>
<p>EM algorithm for multivariate clustering of SNAs</p></a></li>
<li><a href='#canopy.cluster.Estep'>
<p>E-step of EM algorithm for multivariate clustering of SNAs</p></a></li>
<li><a href='#canopy.cluster.Mstep'>
<p>M-step of EM algorithm for multivariate clustering of SNAs</p></a></li>
<li><a href='#canopy.output'>
<p>To generate a posterior tree</p></a></li>
<li><a href='#canopy.plottree'>
<p>To plot tree inferred by Canopy</p></a></li>
<li><a href='#canopy.post'>
<p>Posterior evaluation of MCMC sampled trees</p></a></li>
<li><a href='#canopy.sample'>
<p>MCMC sampling in tree space</p></a></li>
<li><a href='#canopy.sample.cluster'>
<p>MCMC sampling in tree space with pre-clustering of SNAs</p></a></li>
<li><a href='#canopy.sample.cluster.nocna'>
<p>MCMC sampling in tree space with pre-clustering of SNAs</p></a></li>
<li><a href='#canopy.sample.nocna'>
<p>MCMC sampling in tree space</p></a></li>
<li><a href='#getclonalcomposition'>
<p>To get clonal composition</p></a></li>
<li><a href='#getCMCm'>
<p>To get major and minor copy per clone</p></a></li>
<li><a href='#getCZ'>
<p>To get CNA genotyping matrix CZ</p></a></li>
<li><a href='#getlikelihood'>
<p>To get likelihood of the tree</p></a></li>
<li><a href='#getlikelihood.sna'>
<p>To get SNA likelihood of the tree</p></a></li>
<li><a href='#getQ'>
<p>To get SNA-CNA genotyping matrix</p></a></li>
<li><a href='#getVAF'>
<p>To get variant allele frequency (VAF)</p></a></li>
<li><a href='#getZ'>
<p>To get SNA genotyping matrix <code class="reqn">Z</code></p></a></li>
<li><a href='#initialcna'>
<p>To initialize positions of CNAs</p></a></li>
<li><a href='#initialcnacopy'>
<p>To initialize major and minor copies of CNAs</p></a></li>
<li><a href='#initialP'>
<p>To initialize clonal frequency matrix</p></a></li>
<li><a href='#initialsna'>
<p>To initialize positions of SNAs</p></a></li>
<li><a href='#MDA231'>
<p>Dataset for project MDA231</p></a></li>
<li><a href='#MDA231_sampchain'>
<p>List of pre-sampled trees</p></a></li>
<li><a href='#MDA231_tree'>
<p>Most likely tree from project MDA231</p></a></li>
<li><a href='#sampcna'>
<p>To sample CNA positions</p></a></li>
<li><a href='#sampcnacopy'>
<p>To sample major and minor copies of CNAs</p></a></li>
<li><a href='#sampP'>
<p>To sample clonal frequency</p></a></li>
<li><a href='#sampsna'>
<p>To sample SNA positions</p></a></li>
<li><a href='#sampsna.cluster'>
<p>To sample positions of SNA clusters</p></a></li>
<li><a href='#sortcna'>
<p>To sort identified overlapping CNAs.</p></a></li>
<li><a href='#toy'>
<p>Toy dataset for Canopy</p></a></li>
<li><a href='#toy2'>
<p>Toy dataset 2 for Canopy</p></a></li>
<li><a href='#toy3'>
<p>Toy dataset 3 for Canopy</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Accessing Intra-Tumor Heterogeneity and Tracking Longitudinal
and Spatial Clonal Evolutionary History by Next-Generation
Sequencing</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Yuchao Jiang, Nancy R. Zhang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yuchao Jiang &lt;yuchaoj@email.unc.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A statistical framework and computational procedure for identifying
  the sub-populations within a tumor, determining the mutation profiles of each 
  subpopulation, and inferring the tumor's phylogenetic history. The input are 
  variant allele frequencies (VAFs) of somatic single nucleotide alterations 
  (SNAs) along with allele-specific coverage ratios between the tumor and matched
  normal sample for somatic copy number alterations (CNAs). These quantities can
  be directly taken from the output of existing software. Canopy provides a 
  general mathematical framework for pooling data across samples and sites to 
  infer the underlying parameters. For SNAs that fall within CNA regions, Canopy
  infers their temporal ordering and resolves their phase.  When there are 
  multiple evolutionary configurations consistent with the data, Canopy outputs 
  all configurations along with their confidence assessment.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4), ape, fields, pheatmap, scatterplot3d</td>
</tr>
<tr>
<td>Imports:</td>
<td>grDevices, graphics, stats, utils</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/yuchaojiang/Canopy">https://github.com/yuchaojiang/Canopy</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-12-18 16:00:51 UTC; yuchaojiang</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-12-18 19:12:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='addsamptree'>
To determine whether the sampled tree will be accepted
</h2><span id='topic+addsamptree'></span>

<h3>Description</h3>

<p>To determine whether the sampled tree will be accepted by comparing the
likelihood, used in <code><a href="#topic+canopy.sample">canopy.sample</a>.</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addsamptree(tree,tree.new)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="addsamptree_+3A_tree">tree</code></td>
<td>

<p>input tree (current)</p>
</td></tr>
<tr><td><code id="addsamptree_+3A_tree.new">tree.new</code></td>
<td>

<p>input tree (newly sampled)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returned tree (either retain the old tree or accept the new tree).
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231)
data(MDA231_tree)
sna.name = MDA231$sna.name
Y = MDA231$Y
C = MDA231$C
R = MDA231$R
X = MDA231$X
WM = MDA231$WM
Wm = MDA231$Wm
epsilonM = MDA231$epsilonM
epsilonm = MDA231$epsilonm
# sampling location of SNAs
tree.new = MDA231_tree
tree.new$sna = sampsna(MDA231_tree)
tree.new$Z = getZ(tree.new, sna.name)
tree.new$Q = getQ(tree.new, Y, C)
tree.new$H = tree.new$Q
tree.new$VAF = getVAF(tree.new, Y)
tree.new$likelihood = getlikelihood(tree.new, R, X, WM, Wm, epsilonM, epsilonm)
tree = addsamptree(MDA231_tree,tree.new)
</code></pre>

<hr>
<h2 id='AML43'>
SNA input for primary tumor and relapse genome of leukemia patient from
Ding et al. Nature 2012.
</h2><span id='topic+AML43'></span>

<h3>Description</h3>

<p>1242 SNAs from sequencing of leukemia patient at two timepoints. All SNAs
are filtered to be from copy-number-neutral region.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(AML43)</code></pre>


<h3>Value</h3>

<p>List of simulated SNA input data for Canopy.
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    data(AML43)
</code></pre>

<hr>
<h2 id='canopy.BIC'>
To get BIC as a model selection criterion
</h2><span id='topic+canopy.BIC'></span>

<h3>Description</h3>

<p>To get BIC as a model selection criterion from MCMC sampling results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>canopy.BIC(sampchain,projectname,K,numchain,burnin,thin,pdf)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="canopy.BIC_+3A_sampchain">sampchain</code></td>
<td>

<p>list of sampled trees returned by <code><a href="#topic+canopy.sample">canopy.sample</a></code></p>
</td></tr>
<tr><td><code id="canopy.BIC_+3A_projectname">projectname</code></td>
<td>

<p>name of project</p>
</td></tr>
<tr><td><code id="canopy.BIC_+3A_k">K</code></td>
<td>

<p>number of subclones (vector)</p>
</td></tr>
<tr><td><code id="canopy.BIC_+3A_numchain">numchain</code></td>
<td>

<p>number of MCMC chains with random initiations</p>
</td></tr>
<tr><td><code id="canopy.BIC_+3A_burnin">burnin</code></td>
<td>

<p>burnin of MCMC chains</p>
</td></tr>
<tr><td><code id="canopy.BIC_+3A_thin">thin</code></td>
<td>

<p>MCMC chains thinning</p>
</td></tr>
<tr><td><code id="canopy.BIC_+3A_pdf">pdf</code></td>
<td>

<p>whether a pdf plot of BIC should be generated, default to be TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>BIC values (vector) for model selection with plot generated (pdf format).
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_sampchain)
sampchain = MDA231_sampchain
projectname = 'MD231'
K = 3:6
numchain = 20
burnin = 150
thin = 5
bic = canopy.BIC(sampchain = sampchain, projectname = projectname, K = K,
                 numchain = numchain, burnin = burnin, thin = thin)
</code></pre>

<hr>
<h2 id='canopy.cluster'>
EM algorithm for multivariate clustering of SNAs
</h2><span id='topic+canopy.cluster'></span>

<h3>Description</h3>

<p>EM algorithm for multivariate clustering of SNAs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    canopy.cluster(R, X, num_cluster, num_run, Mu.init = NULL, Tau_Kplus1 = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="canopy.cluster_+3A_r">R</code></td>
<td>

<p>alternative allele read depth matrix</p>
</td></tr>
<tr><td><code id="canopy.cluster_+3A_x">X</code></td>
<td>

<p>total read depth matrix</p>
</td></tr>
<tr><td><code id="canopy.cluster_+3A_num_cluster">num_cluster</code></td>
<td>

<p>number of mutation clusters (BIC as model selection metric)</p>
</td></tr>
<tr><td><code id="canopy.cluster_+3A_num_run">num_run</code></td>
<td>

<p>number of EM runs for estimation for each specific number of clusters 
(to avoid EM being stuck in local optima)</p>
</td></tr>
<tr><td><code id="canopy.cluster_+3A_mu.init">Mu.init</code></td>
<td>

<p>(optional) initial value of the VAF centroid for each mutation cluster
in each sample</p>
</td></tr>
<tr><td><code id="canopy.cluster_+3A_tau_kplus1">Tau_Kplus1</code></td>
<td>

<p>(optional) pre-specified proportion of noise component in clustering,
uniformly distributed between 0 and 1
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of posterior probability of cluster assignment for each mutation.</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    data(AML43)
    R = AML43$R
    X = AML43$X
    Mu = AML43$Mu
    Tau = AML43$Tau
    pG = canopy.cluster.Estep(Tau, Mu, R, X)
</code></pre>

<hr>
<h2 id='canopy.cluster.Estep'>
E-step of EM algorithm for multivariate clustering of SNAs
</h2><span id='topic+canopy.cluster.Estep'></span>

<h3>Description</h3>

<p>E-step of EM algorithm for multivariate clustering of SNAs. Used in
<code><a href="#topic+canopy.cluster">canopy.cluster</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    canopy.cluster.Estep(Tau, Mu, R, X)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="canopy.cluster.Estep_+3A_tau">Tau</code></td>
<td>

<p>prior for proportions of mutation clusters</p>
</td></tr>
<tr><td><code id="canopy.cluster.Estep_+3A_mu">Mu</code></td>
<td>

<p>MAF centroid for each mutation cluster in each sample</p>
</td></tr>
<tr><td><code id="canopy.cluster.Estep_+3A_r">R</code></td>
<td>

<p>alternative allele read depth matrix</p>
</td></tr>
<tr><td><code id="canopy.cluster.Estep_+3A_x">X</code></td>
<td>

<p>total read depth matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of posterior probability of cluster assignment for each mutation.</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    data(AML43)
    R = AML43$R
    X = AML43$X
    Mu = AML43$Mu
    Tau = AML43$Tau
    pG = canopy.cluster.Estep(Tau, Mu, R, X)
</code></pre>

<hr>
<h2 id='canopy.cluster.Mstep'>
M-step of EM algorithm for multivariate clustering of SNAs
</h2><span id='topic+canopy.cluster.Mstep'></span>

<h3>Description</h3>

<p>M-step of EM algorithm for multivariate clustering of SNAs. Used in
<code><a href="#topic+canopy.cluster">canopy.cluster</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    canopy.cluster.Mstep(pG, R, X, Tau_Kplus1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="canopy.cluster.Mstep_+3A_pg">pG</code></td>
<td>

<p>matrix of posterior probability of cluster assignment for each mutation</p>
</td></tr>
<tr><td><code id="canopy.cluster.Mstep_+3A_r">R</code></td>
<td>

<p>alternative allele read depth matrix</p>
</td></tr>
<tr><td><code id="canopy.cluster.Mstep_+3A_x">X</code></td>
<td>

<p>total read depth matrix</p>
</td></tr>
<tr><td><code id="canopy.cluster.Mstep_+3A_tau_kplus1">Tau_Kplus1</code></td>
<td>

<p>proportion mutation cluster that is uniformly distributed to capture noise</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of bic, converged Mu, Tau, and SNA cluster assignment.</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    data(AML43)
    R = AML43$R; X = AML43$X
    num_cluster = 4 # Range of number of clusters to run
    num_run = 6 # How many EM runs per clustering step
    Tau_Kplus1=0.05 # Proportion of noise component
    Mu.init=cbind(c(0.01,0.15,0.25,0.45),c(0.2,0.2,0.01,0.2)) # initial value
                                                              # of centroid
    canopy.cluster=canopy.cluster(R = R, X = X, num_cluster = num_cluster,
                                  num_run = num_run, Mu.init = Mu.init,
                                  Tau_Kplus1=Tau_Kplus1)
</code></pre>

<hr>
<h2 id='canopy.output'>
To generate a posterior tree
</h2><span id='topic+canopy.output'></span>

<h3>Description</h3>

<p>To generate a posterior tree from the sub-tree space of trees with the same
configurations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>canopy.output(post, config.i, C)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="canopy.output_+3A_post">post</code></td>
<td>

<p>list returned by <code><a href="#topic+canopy.post">canopy.post</a></code></p>
</td></tr>
<tr><td><code id="canopy.output_+3A_config.i">config.i</code></td>
<td>

<p>configuration of sub-tree space to be output</p>
</td></tr>
<tr><td><code id="canopy.output_+3A_c">C</code></td>
<td>

<p>CNA and CNA-region overlapping matrix, only needed if overlapping CNAs are
used as input</p>
</td></tr>
</table>


<h3>Value</h3>

<p>posterior tree from the sub-tree space of trees with the same configurations.
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_sampchain)
data(MDA231)
sampchain = MDA231_sampchain
projectname = 'MD231'
K = 3:6
numchain = 20
burnin = 150
thin = 5
optK = 4
C = MDA231$C
post = canopy.post(sampchain = sampchain, projectname = projectname, K = K,
                   numchain = numchain, burnin = burnin, thin = thin, 
                   optK = optK, C = C)
config.i = 3
output.tree = canopy.output(post = post, config.i = config.i, C = C)
</code></pre>

<hr>
<h2 id='canopy.plottree'>
To plot tree inferred by Canopy
</h2><span id='topic+canopy.plottree'></span>

<h3>Description</h3>

<p>To plot Canopy's reconstructed phylogeny. Major plotting function of Canopy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>canopy.plottree(tree, pdf, pdf.name, txt, txt.name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="canopy.plottree_+3A_tree">tree</code></td>
<td>

<p>input tree to be plotted</p>
</td></tr>
<tr><td><code id="canopy.plottree_+3A_pdf">pdf</code></td>
<td>

<p>whether a pdf plot should be generated, default to be FALSE</p>
</td></tr>
<tr><td><code id="canopy.plottree_+3A_pdf.name">pdf.name</code></td>
<td>

<p>name of pdf to be generated, has to be provided if pdf is to be generated</p>
</td></tr>
<tr><td><code id="canopy.plottree_+3A_txt">txt</code></td>
<td>

<p>whether a txt file should be generated with information on mutations along 
the tree branches, default to be FALSE</p>
</td></tr>
<tr><td><code id="canopy.plottree_+3A_txt.name">txt.name</code></td>
<td>

<p>name of txt to be generated, has to be provided if txt is to be generated</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot of tree structure, clonal frequency and mutation legends (pdf format).
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_tree)
canopy.plottree(MDA231_tree, pdf = TRUE, pdf.name = 'MDA231_tree.pdf')
</code></pre>

<hr>
<h2 id='canopy.post'>
Posterior evaluation of MCMC sampled trees
</h2><span id='topic+canopy.post'></span>

<h3>Description</h3>

<p>Burnin, thinning, and posterior evaluation of MCMC sampled trees.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>canopy.post(sampchain, projectname, K, numchain, burnin, thin, optK,
            C, post.config.cutoff)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="canopy.post_+3A_sampchain">sampchain</code></td>
<td>

<p>list of sampled trees returned by <code><a href="#topic+canopy.sample">canopy.sample</a></code></p>
</td></tr>
<tr><td><code id="canopy.post_+3A_projectname">projectname</code></td>
<td>

<p>name of project</p>
</td></tr>
<tr><td><code id="canopy.post_+3A_k">K</code></td>
<td>

<p>number of subclones (vector)</p>
</td></tr>
<tr><td><code id="canopy.post_+3A_numchain">numchain</code></td>
<td>

<p>number of MCMC chains with random initiations</p>
</td></tr>
<tr><td><code id="canopy.post_+3A_burnin">burnin</code></td>
<td>

<p>burnin of MCMC chains</p>
</td></tr>
<tr><td><code id="canopy.post_+3A_thin">thin</code></td>
<td>

<p>MCMC chain thinning.</p>
</td></tr>
<tr><td><code id="canopy.post_+3A_optk">optK</code></td>
<td>

<p>optimal number of subclones determined by <code><a href="#topic+canopy.BIC">canopy.BIC</a></code></p>
</td></tr>
<tr><td><code id="canopy.post_+3A_c">C</code></td>
<td>

<p>CNA and CNA-region overlapping matrix, only needed if overlapping CNAs are
used as input</p>
</td></tr>
<tr><td><code id="canopy.post_+3A_post.config.cutoff">post.config.cutoff</code></td>
<td>

<p>cutoff value for posterior probabilities of tree configurations,
default is set to be 0.05 (only tree configurations with greater
than 0.05 posterior probabilities will be reported by Canopy)</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>samptreethin</code></td>
<td>

<p>list of sampled posterior trees</p>
</td></tr>
<tr><td><code>samptreethin.lik</code></td>
<td>

<p>vector of likelihood of sampled posterior trees</p>
</td></tr>
<tr><td><code>config</code></td>
<td>

<p>vector of configuration of sampled posterior trees (integer values)</p>
</td></tr>
<tr><td><code>config.summary</code></td>
<td>

<p>summary of configurations of sampled posterior trees</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_sampchain)
data(MDA231)
sampchain = MDA231_sampchain
projectname = 'MD231'
K = 3:6
numchain = 20
burnin = 150
thin = 5
optK = 4
C = MDA231$C
post = canopy.post(sampchain = sampchain, projectname = projectname, K = K,
                   numchain = numchain, burnin = burnin, thin = thin, 
                   optK = optK, C = C)
</code></pre>

<hr>
<h2 id='canopy.sample'>
MCMC sampling in tree space
</h2><span id='topic+canopy.sample'></span>

<h3>Description</h3>

<p>To sample the posterior trees. Major function of Canopy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>canopy.sample(R, X, WM, Wm, epsilonM, epsilonm, C=NULL,
              Y, K, numchain, max.simrun, min.simrun, writeskip, projectname,
              cell.line=NULL, plot.likelihood=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="canopy.sample_+3A_r">R</code></td>
<td>

<p>alternative allele read depth matrix</p>
</td></tr>
<tr><td><code id="canopy.sample_+3A_x">X</code></td>
<td>

<p>total read depth matrix</p>
</td></tr>
<tr><td><code id="canopy.sample_+3A_wm">WM</code></td>
<td>

<p>observed major copy number matrix</p>
</td></tr>
<tr><td><code id="canopy.sample_+3A_wm">Wm</code></td>
<td>

<p>observed minor copy number matrix</p>
</td></tr>
<tr><td><code id="canopy.sample_+3A_epsilonm">epsilonM</code></td>
<td>

<p>observed standard deviation of major copy number (scalar input is
transformed into matrix)</p>
</td></tr>
<tr><td><code id="canopy.sample_+3A_epsilonm">epsilonm</code></td>
<td>

<p>observed standard deviation of minor copy number (scalar input is
transformed into matrix)</p>
</td></tr>
<tr><td><code id="canopy.sample_+3A_c">C</code></td>
<td>

<p>CNA and CNA-region overlapping matrix, only needed if overlapping CNAs are
used as input</p>
</td></tr>
<tr><td><code id="canopy.sample_+3A_y">Y</code></td>
<td>

<p>SNA and CNA-region overlapping matrix</p>
</td></tr>
<tr><td><code id="canopy.sample_+3A_k">K</code></td>
<td>

<p>number of subclones (vector)</p>
</td></tr>
<tr><td><code id="canopy.sample_+3A_numchain">numchain</code></td>
<td>

<p>number of MCMC chains with random initiations</p>
</td></tr>
<tr><td><code id="canopy.sample_+3A_max.simrun">max.simrun</code></td>
<td>

<p>maximum number of simutation iterations for each chain</p>
</td></tr>
<tr><td><code id="canopy.sample_+3A_min.simrun">min.simrun</code></td>
<td>

<p>minimum number of simutation iterations for each chain</p>
</td></tr>
<tr><td><code id="canopy.sample_+3A_writeskip">writeskip</code></td>
<td>

<p>interval to store sampled trees</p>
</td></tr>
<tr><td><code id="canopy.sample_+3A_projectname">projectname</code></td>
<td>

<p>name of project</p>
</td></tr>
<tr><td><code id="canopy.sample_+3A_cell.line">cell.line</code></td>
<td>

<p>default to be FALSE, TRUE if input sample is cell line (no normal cell
contamination)</p>
</td></tr>
<tr><td><code id="canopy.sample_+3A_plot.likelihood">plot.likelihood</code></td>
<td>

<p>default to be TRUE, posterior likelihood plot generated for check of
convergence and selection of burnin and thinning in
<code><a href="#topic+canopy.post">canopy.post</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of sampleed trees in subtree space with different number of subclones;
plot of posterior likelihoods in each subtree space generated (pdf format).</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231)
R = MDA231$R; X = MDA231$X
WM = MDA231$WM; Wm = MDA231$Wm
epsilonM = MDA231$epsilonM; epsilonm = MDA231$epsilonm
C = MDA231$C
Y = MDA231$Y
K = 3:6
numchain = 20
projectname = 'MDA231'
# sampchain = canopy.sample(R = R, X = X, WM = WM, Wm = Wm, epsilonM = epsilonM, 
#             epsilonm = epsilonm, C = C, Y = Y, K = K, numchain = numchain, 
#             max.simrun = 50000, min.simrun = 10000, writeskip = 200, 
#             projectname = projectname, cell.line = TRUE, plot.likelihood = TRUE)
</code></pre>

<hr>
<h2 id='canopy.sample.cluster'>
MCMC sampling in tree space with pre-clustering of SNAs
</h2><span id='topic+canopy.sample.cluster'></span>

<h3>Description</h3>

<p>To sample the posterior trees with pre-clustering step of SNAs. Major
function of Canopy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    canopy.sample.cluster(R, X, sna_cluster, WM, Wm, epsilonM, epsilonm, C=NULL,
                  Y, K, numchain, max.simrun, min.simrun, writeskip, projectname,
                  cell.line=NULL, plot.likelihood=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="canopy.sample.cluster_+3A_r">R</code></td>
<td>

<p>alternative allele read depth matrix</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster_+3A_x">X</code></td>
<td>

<p>total read depth matrix</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster_+3A_sna_cluster">sna_cluster</code></td>
<td>

<p>cluster assignment for each mutation from the EM Binomial clustering
algorithm
</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster_+3A_wm">WM</code></td>
<td>

<p>observed major copy number matrix</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster_+3A_wm">Wm</code></td>
<td>

<p>observed minor copy number matrix</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster_+3A_epsilonm">epsilonM</code></td>
<td>

<p>observed standard deviation of major copy number (scalar input is
transformed into matrix)</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster_+3A_epsilonm">epsilonm</code></td>
<td>

<p>observed standard deviation of minor copy number (scalar input is
transformed into matrix)</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster_+3A_c">C</code></td>
<td>

<p>CNA and CNA-region overlapping matrix, only needed if overlapping CNAs are
used as input</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster_+3A_y">Y</code></td>
<td>

<p>SNA and CNA-region overlapping matrix</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster_+3A_k">K</code></td>
<td>

<p>number of subclones (vector)</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster_+3A_numchain">numchain</code></td>
<td>

<p>number of MCMC chains with random initiations</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster_+3A_max.simrun">max.simrun</code></td>
<td>

<p>maximum number of simutation iterations for each chain</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster_+3A_min.simrun">min.simrun</code></td>
<td>

<p>minimum number of simutation iterations for each chain</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster_+3A_writeskip">writeskip</code></td>
<td>

<p>interval to store sampled trees</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster_+3A_projectname">projectname</code></td>
<td>

<p>name of project</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster_+3A_cell.line">cell.line</code></td>
<td>

<p>default to be FALSE, TRUE if input sample is cell line (no normal cell
contamination)</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster_+3A_plot.likelihood">plot.likelihood</code></td>
<td>

<p>default to be TRUE, posterior likelihood plot generated for check of
convergence and selection of burnin and thinning in
<code><a href="#topic+canopy.post">canopy.post</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of sampleed trees in subtree space with different number of subclones;
plot of posterior likelihoods in each subtree space generated (pdf format).</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    data(MDA231)
    R = MDA231$R; X = MDA231$X
    WM = MDA231$WM; Wm = MDA231$Wm
    epsilonM = MDA231$epsilonM; epsilonm = MDA231$epsilonm
    C = MDA231$C
    Y = MDA231$Y
    K = 3:6
    numchain = 20
    projectname = 'MDA231'
    # sampchain = canopy.sample.cluster(R = R, X = X, sna_cluster=c(1,2,3,4),
    #             WM = WM, Wm = Wm, epsilonM = epsilonM, 
    #             epsilonm = epsilonm, C = C, Y = Y, K = K, numchain = numchain, 
    #             max.simrun = 50000, min.simrun = 10000, writeskip = 200, 
    #             projectname = projectname, cell.line = TRUE, plot.likelihood = TRUE)
</code></pre>

<hr>
<h2 id='canopy.sample.cluster.nocna'>
MCMC sampling in tree space with pre-clustering of SNAs
</h2><span id='topic+canopy.sample.cluster.nocna'></span>

<h3>Description</h3>

<p>To sample the posterior trees with pre-clustering step of SNAs. Major
function of Canopy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    canopy.sample.cluster.nocna(R, X, sna_cluster, K, numchain, 
                                max.simrun, min.simrun, writeskip, projectname,
                                cell.line=NULL, plot.likelihood=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="canopy.sample.cluster.nocna_+3A_r">R</code></td>
<td>

<p>alternative allele read depth matrix</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster.nocna_+3A_x">X</code></td>
<td>

<p>total read depth matrix</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster.nocna_+3A_sna_cluster">sna_cluster</code></td>
<td>

<p>cluster assignment for each mutation from the EM Binomial clustering
algorithm
</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster.nocna_+3A_k">K</code></td>
<td>

<p>number of subclones (vector)</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster.nocna_+3A_numchain">numchain</code></td>
<td>

<p>number of MCMC chains with random initiations</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster.nocna_+3A_max.simrun">max.simrun</code></td>
<td>

<p>maximum number of simutation iterations for each chain</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster.nocna_+3A_min.simrun">min.simrun</code></td>
<td>

<p>minimum number of simutation iterations for each chain</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster.nocna_+3A_writeskip">writeskip</code></td>
<td>

<p>interval to store sampled trees</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster.nocna_+3A_projectname">projectname</code></td>
<td>

<p>name of project</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster.nocna_+3A_cell.line">cell.line</code></td>
<td>

<p>default to be FALSE, TRUE if input sample is cell line (no normal cell
contamination)</p>
</td></tr>
<tr><td><code id="canopy.sample.cluster.nocna_+3A_plot.likelihood">plot.likelihood</code></td>
<td>

<p>default to be TRUE, posterior likelihood plot generated for check of
convergence and selection of burnin and thinning in
<code><a href="#topic+canopy.post">canopy.post</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of sampleed trees in subtree space with different number of subclones;
plot of posterior likelihoods in each subtree space generated (pdf format).</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    data(toy3)
    R = toy3$R; X = toy3$X
    sna_cluster = toy3$sna_cluster
    K = 3:5
    numchain = 10
    projectname = 'toy3'
    # sampchain = canopy.sample.cluster.nocna(R = R, X = X, 
    #             sna_cluster=sna_cluster, K = K, numchain = numchain, 
    #             max.simrun = 40000, min.simrun = 10000, writeskip = 200, 
    #             projectname = projectname,
    #             cell.line = TRUE, plot.likelihood = TRUE)
</code></pre>

<hr>
<h2 id='canopy.sample.nocna'>
MCMC sampling in tree space
</h2><span id='topic+canopy.sample.nocna'></span>

<h3>Description</h3>

<p>To sample the posterior trees without CNA input. Major function of Canopy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    canopy.sample.nocna(R, X, K, numchain, max.simrun, min.simrun, writeskip, 
                  projectname, cell.line=NULL, plot.likelihood=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="canopy.sample.nocna_+3A_r">R</code></td>
<td>

<p>alternative allele read depth matrix</p>
</td></tr>
<tr><td><code id="canopy.sample.nocna_+3A_x">X</code></td>
<td>

<p>total read depth matrix</p>
</td></tr>
<tr><td><code id="canopy.sample.nocna_+3A_k">K</code></td>
<td>

<p>number of subclones (vector)</p>
</td></tr>
<tr><td><code id="canopy.sample.nocna_+3A_numchain">numchain</code></td>
<td>

<p>number of MCMC chains with random initiations</p>
</td></tr>
<tr><td><code id="canopy.sample.nocna_+3A_max.simrun">max.simrun</code></td>
<td>

<p>maximum number of simutation iterations for each chain</p>
</td></tr>
<tr><td><code id="canopy.sample.nocna_+3A_min.simrun">min.simrun</code></td>
<td>

<p>minimum number of simutation iterations for each chain</p>
</td></tr>
<tr><td><code id="canopy.sample.nocna_+3A_writeskip">writeskip</code></td>
<td>

<p>interval to store sampled trees</p>
</td></tr>
<tr><td><code id="canopy.sample.nocna_+3A_projectname">projectname</code></td>
<td>

<p>name of project</p>
</td></tr>
<tr><td><code id="canopy.sample.nocna_+3A_cell.line">cell.line</code></td>
<td>

<p>default to be FALSE, TRUE if input sample is cell line (no normal cell
contamination)</p>
</td></tr>
<tr><td><code id="canopy.sample.nocna_+3A_plot.likelihood">plot.likelihood</code></td>
<td>

<p>default to be TRUE, posterior likelihood plot generated for check of
convergence and selection of burnin and thinning in
<code><a href="#topic+canopy.post">canopy.post</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of sampleed trees in subtree space with different number of subclones;
plot of posterior likelihoods in each subtree space generated (pdf format).</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    data(toy3)
    R = toy3$R; X = toy3$X
    K = 3:5
    numchain = 10
    projectname = 'toy3'
    # sampchain = canopy.sample.nocna(R = R, X = X, K = K, numchain = numchain, 
    #             max.simrun = 50000, min.simrun = 10000, writeskip = 200, 
    #             projectname = projectname,
    #             cell.line = TRUE, plot.likelihood = TRUE)
</code></pre>

<hr>
<h2 id='getclonalcomposition'>
To get clonal composition
</h2><span id='topic+getclonalcomposition'></span>

<h3>Description</h3>

<p>To get clonal composition (mutational profile of each clone) of tree. Used in
<code><a href="#topic+canopy.post">canopy.post</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getclonalcomposition(tree)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getclonalcomposition_+3A_tree">tree</code></td>
<td>

<p>input tree</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of each clone's mutational profile.
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_tree)
getclonalcomposition(MDA231_tree)
</code></pre>

<hr>
<h2 id='getCMCm'>
To get major and minor copy per clone
</h2><span id='topic+getCMCm'></span>

<h3>Description</h3>

<p>To get major and minor copy per clone. Used in <code><a href="#topic+canopy.sample">canopy.sample</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCMCm(tree, C)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getCMCm_+3A_tree">tree</code></td>
<td>

<p>input tree</p>
</td></tr>
<tr><td><code id="getCMCm_+3A_c">C</code></td>
<td>

<p>CNA regions and CNA overlapping matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>CM</code></td>
<td>

<p>Matrix of major copy per clone.</p>
</td></tr>
<tr><td><code>Cm</code></td>
<td>

<p>Matrix of minor copy per clone.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_tree)
data(MDA231)
C = MDA231$C
getCMCm(MDA231_tree, C)
</code></pre>

<hr>
<h2 id='getCZ'>
To get CNA genotyping matrix CZ
</h2><span id='topic+getCZ'></span>

<h3>Description</h3>

<p>To get CNA genotyping matrix CZ from location of CNAs on the tree. Used in
<code><a href="#topic+canopy.sample">canopy.sample</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCZ(tree)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getCZ_+3A_tree">tree</code></td>
<td>

<p>input tree</p>
</td></tr>
</table>


<h3>Value</h3>

<p>CNA genotyping matrix CZ.
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_tree)
getCZ(MDA231_tree)
</code></pre>

<hr>
<h2 id='getlikelihood'>
To get likelihood of the tree
</h2><span id='topic+getlikelihood'></span>

<h3>Description</h3>

<p>To get likelihood of the tree given tree struture and data input. Used in
<code><a href="#topic+canopy.sample">canopy.sample</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getlikelihood(tree,R,X,WM,Wm,epsilonM,epsilonm)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getlikelihood_+3A_tree">tree</code></td>
<td>

<p>input tree</p>
</td></tr>
<tr><td><code id="getlikelihood_+3A_r">R</code></td>
<td>

<p>alternative allele read depth matrix</p>
</td></tr>
<tr><td><code id="getlikelihood_+3A_x">X</code></td>
<td>

<p>total read depth matrix</p>
</td></tr>
<tr><td><code id="getlikelihood_+3A_wm">WM</code></td>
<td>

<p>observed major copy number matrix</p>
</td></tr>
<tr><td><code id="getlikelihood_+3A_wm">Wm</code></td>
<td>

<p>observed minor copy number matrix</p>
</td></tr>
<tr><td><code id="getlikelihood_+3A_epsilonm">epsilonM</code></td>
<td>

<p>observed standard deviation of major copy number (scalar input is
transformed into matrix)</p>
</td></tr>
<tr><td><code id="getlikelihood_+3A_epsilonm">epsilonm</code></td>
<td>

<p>observed standard deviation of minor copy number (scalar input is
transformed into matrix)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Likelihood of sampled tree.</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231)
data(MDA231_tree)
R = MDA231$R
X = MDA231$X
WM = MDA231$WM
Wm = MDA231$Wm
epsilonM = MDA231$epsilonM
epsilonm = MDA231$epsilonm
getlikelihood(MDA231_tree, R, X, WM, Wm, epsilonM, epsilonm)
</code></pre>

<hr>
<h2 id='getlikelihood.sna'>
To get SNA likelihood of the tree
</h2><span id='topic+getlikelihood.sna'></span>

<h3>Description</h3>

<p>To get SNA likelihood of the tree given tree struture and data input. Used in
<code><a href="#topic+canopy.sample.nocna">canopy.sample.nocna</a></code> and <code><a href="#topic+canopy.sample.cluster.nocna">canopy.sample.cluster.nocna</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    getlikelihood.sna(tree, R, X)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getlikelihood.sna_+3A_tree">tree</code></td>
<td>

<p>input tree</p>
</td></tr>
<tr><td><code id="getlikelihood.sna_+3A_r">R</code></td>
<td>

<p>alternative allele read depth matrix</p>
</td></tr>
<tr><td><code id="getlikelihood.sna_+3A_x">X</code></td>
<td>

<p>total read depth matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Likelihood of sampled tree.</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    data(MDA231)
    data(MDA231_tree)
    R = MDA231$R
    X = MDA231$X
    getlikelihood.sna(MDA231_tree, R, X)
</code></pre>

<hr>
<h2 id='getQ'>
To get SNA-CNA genotyping matrix
</h2><span id='topic+getQ'></span>

<h3>Description</h3>

<p>To get SNA-CNA genotyping matrix <code class="reqn">Q</code>, which specifies whether an SNA
precedes a CNA. Used in <code><a href="#topic+canopy.sample">canopy.sample</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getQ(tree, Y, C)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getQ_+3A_tree">tree</code></td>
<td>

<p>input tree</p>
</td></tr>
<tr><td><code id="getQ_+3A_y">Y</code></td>
<td>

<p>SNA CNA overlapping matrix</p>
</td></tr>
<tr><td><code id="getQ_+3A_c">C</code></td>
<td>

<p>CNA and CNA region overlapping matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Genotyping matrix <code class="reqn">Q</code>.
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_tree)
data(MDA231)
Y = MDA231$Y
C = MDA231$C
getQ(MDA231_tree, Y, C)
</code></pre>

<hr>
<h2 id='getVAF'>
To get variant allele frequency (VAF)
</h2><span id='topic+getVAF'></span>

<h3>Description</h3>

<p>To get variant allele frequency (VAF) matrix, which contains percentage of 
mutant SNA alleles across samples. Used in <code><a href="#topic+canopy.sample">canopy.sample</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getVAF(tree,Y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getVAF_+3A_tree">tree</code></td>
<td>

<p>input tree</p>
</td></tr>
<tr><td><code id="getVAF_+3A_y">Y</code></td>
<td>

<p>SNA CNA overlapping matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Variant allele frequency matrix VAF.
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_tree)
data(MDA231)
Y = MDA231$Y
getVAF(MDA231_tree, Y)
</code></pre>

<hr>
<h2 id='getZ'>
To get SNA genotyping matrix <code class="reqn">Z</code>
</h2><span id='topic+getZ'></span>

<h3>Description</h3>

<p>To get SNA genotyping matrix <code class="reqn">Z</code> from location of SNAs on the tree. Used in
<code><a href="#topic+canopy.sample">canopy.sample</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getZ(tree, sna.name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getZ_+3A_tree">tree</code></td>
<td>

<p>input tree</p>
</td></tr>
<tr><td><code id="getZ_+3A_sna.name">sna.name</code></td>
<td>

<p>vector of SNA names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Genotyping matrix <code class="reqn">Z</code>.
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_tree)
data(MDA231)
sna.name = rownames(MDA231$R)
getZ(MDA231_tree, sna.name)
</code></pre>

<hr>
<h2 id='initialcna'>
To initialize positions of CNAs
</h2><span id='topic+initialcna'></span>

<h3>Description</h3>

<p>To initialize positions of CNAs on the tree. Used in initialization step of
<code><a href="#topic+canopy.sample">canopy.sample</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initialcna(tree,cna.name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="initialcna_+3A_tree">tree</code></td>
<td>

<p>input tree</p>
</td></tr>
<tr><td><code id="initialcna_+3A_cna.name">cna.name</code></td>
<td>

<p>vector of input CNA names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix specifying positions of CNAs (start and end node).
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_tree)
data(MDA231)
cna.name = rownames(MDA231$WM)
initialcna(MDA231_tree, cna.name)
</code></pre>

<hr>
<h2 id='initialcnacopy'>
To initialize major and minor copies of CNAs
</h2><span id='topic+initialcnacopy'></span>

<h3>Description</h3>

<p>To initialize major and minor copies of CNAs. Used in initialization step of
<code><a href="#topic+canopy.sample">canopy.sample</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initialcnacopy(tree)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="initialcnacopy_+3A_tree">tree</code></td>
<td>

<p>input tree</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix specifying major and minor copies of CNAs.
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_tree)
initialcnacopy(MDA231_tree)
</code></pre>

<hr>
<h2 id='initialP'>
To initialize clonal frequency matrix
</h2><span id='topic+initialP'></span>

<h3>Description</h3>

<p>To initialize clonal frequency matris <code class="reqn">P</code>. Used in initialization step of
<code><a href="#topic+canopy.sample">canopy.sample</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initialP(tree,sampname,cell.line)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="initialP_+3A_tree">tree</code></td>
<td>

<p>input tree</p>
</td></tr>
<tr><td><code id="initialP_+3A_sampname">sampname</code></td>
<td>

<p>vector of input sample names</p>
</td></tr>
<tr><td><code id="initialP_+3A_cell.line">cell.line</code></td>
<td>

<p>default to be FALSE, TRUE if input sample is cell line (no normal cell
contamination)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Clonal frequency matrix <code class="reqn">P</code>.
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_tree)
data(MDA231)
sampname = colnames(MDA231$R)
initialP(MDA231_tree, sampname, cell.line = TRUE)
</code></pre>

<hr>
<h2 id='initialsna'>
To initialize positions of SNAs
</h2><span id='topic+initialsna'></span>

<h3>Description</h3>

<p>To initialize positions of SNAs on the tree. Used in initialization step of
<code><a href="#topic+canopy.sample">canopy.sample</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initialsna(tree,sna.name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="initialsna_+3A_tree">tree</code></td>
<td>

<p>input tree</p>
</td></tr>
<tr><td><code id="initialsna_+3A_sna.name">sna.name</code></td>
<td>

<p>vector of input SNA names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix specifying positions of SNAs (start and end node).
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_tree)
data(MDA231)
sna.name = rownames(MDA231$R)
initialsna(MDA231_tree, sna.name)
</code></pre>

<hr>
<h2 id='MDA231'>
Dataset for project MDA231
</h2><span id='topic+MDA231'></span>

<h3>Description</h3>

<p>Pre-stored dataset for project MDA231. A transplantable metastasis model
system was derived from a heterogeneous human breast cancer cell line 
MDA-MB-231. Cancer cells from the parental line MDA-MB-231 were engrafted into
mouse hosts leading to organ-specific metastasis. Mixed cell populations (MCPs)
were in vivo selected from either bone or lung metastasis and grew into 
phenotypically stable and metastatically competent cancer cell lines. The
parental line as well as the MCP sublines were whole-exome sequenced with 
somatic SNAs and CNAs profiled.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(MDA231)</code></pre>


<h3>Value</h3>

<p>List of input data for Canopy from project MDA231.
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231)
</code></pre>

<hr>
<h2 id='MDA231_sampchain'>
List of pre-sampled trees
</h2><span id='topic+MDA231_sampchain'></span>

<h3>Description</h3>

<p>List of sampleed trees in subtree space with different number of subclones for
project MDA231.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(MDA231_sampchain)</code></pre>


<h3>Value</h3>

<p>List of sampled trees from different subtree space
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_sampchain)
</code></pre>

<hr>
<h2 id='MDA231_tree'>
Most likely tree from project MDA231
</h2><span id='topic+MDA231_tree'></span>

<h3>Description</h3>

<p>Most likely tree from project MDA231 as a tree example.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(MDA231_tree)</code></pre>


<h3>Value</h3>

<p>Most likely tree from project MDA231
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_tree)
</code></pre>

<hr>
<h2 id='sampcna'>
To sample CNA positions
</h2><span id='topic+sampcna'></span>

<h3>Description</h3>

<p>To sample CNA positions along the tree. Used in <code><a href="#topic+canopy.sample">canopy.sample</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampcna(tree)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sampcna_+3A_tree">tree</code></td>
<td>

<p>input tree</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Newly sampled matrix specifying positions of CNAs (start and end node).
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_tree)
sampcna(MDA231_tree)
</code></pre>

<hr>
<h2 id='sampcnacopy'>
To sample major and minor copies of CNAs
</h2><span id='topic+sampcnacopy'></span>

<h3>Description</h3>

<p>To sample major and minor copies of CNAs. Used in <code><a href="#topic+canopy.sample">canopy.sample</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampcnacopy(tree)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sampcnacopy_+3A_tree">tree</code></td>
<td>

<p>input tree</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Newly sampled matrix specifying major and minor copies of CNAs.
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_tree)
sampcnacopy(MDA231_tree)
</code></pre>

<hr>
<h2 id='sampP'>
To sample clonal frequency
</h2><span id='topic+sampP'></span>

<h3>Description</h3>

<p>To sample clonal frequency matrix <code class="reqn">P</code>. Used in <code><a href="#topic+canopy.sample">canopy.sample</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampP(tree, cell.line)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sampP_+3A_tree">tree</code></td>
<td>

<p>input tree</p>
</td></tr>
<tr><td><code id="sampP_+3A_cell.line">cell.line</code></td>
<td>

<p>default to be FALSE, TRUE if input sample is cell line (no normal cell
contamination)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Newly sampled clonal frequency matrix <code class="reqn">P</code>.
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_tree)
sampP(MDA231_tree, cell.line = TRUE)
</code></pre>

<hr>
<h2 id='sampsna'>
To sample SNA positions
</h2><span id='topic+sampsna'></span>

<h3>Description</h3>

<p>To sample SNA positions along the tree. Used in <code><a href="#topic+canopy.sample">canopy.sample</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampsna(tree)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sampsna_+3A_tree">tree</code></td>
<td>

<p>input tree</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Newly sampled matrix specifying positions of SNAs (start and end node).
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_tree)
sampsna(MDA231_tree)
</code></pre>

<hr>
<h2 id='sampsna.cluster'>
To sample positions of SNA clusters
</h2><span id='topic+sampsna.cluster'></span>

<h3>Description</h3>

<p>To sample SNA cluster positions along the tree. Used in 
<code><a href="#topic+canopy.sample.cluster">canopy.sample.cluster</a></code> and <code><a href="#topic+canopy.sample.cluster.nocna">canopy.sample.cluster.nocna</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    sampsna.cluster(tree)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sampsna.cluster_+3A_tree">tree</code></td>
<td>
<p>input tree</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Newly sampled matrix specifying positions of SNA clusters (start and end node).
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    data(MDA231_tree)
    MDA231_tree$sna.cluster=initialsna(MDA231_tree,paste('cluster',1:4,sep=''))
    sampsna.cluster(MDA231_tree)
</code></pre>

<hr>
<h2 id='sortcna'>
To sort identified overlapping CNAs.
</h2><span id='topic+sortcna'></span>

<h3>Description</h3>

<p>To sort identified overlapping CNAs by their major and minor copy numbers.
Used in <code><a href="#topic+canopy.post">canopy.post</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sortcna(tree,C)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sortcna_+3A_tree">tree</code></td>
<td>

<p>input tree</p>
</td></tr>
<tr><td><code id="sortcna_+3A_c">C</code></td>
<td>

<p>CNA and CNA-region overlapping matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Tree whose overlapping CNAs are sorted by major and minor copy numbers.
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MDA231_tree)
data(MDA231)
C = MDA231$C
sortcna(MDA231_tree, C)
</code></pre>

<hr>
<h2 id='toy'>
Toy dataset for Canopy
</h2><span id='topic+toy'></span>

<h3>Description</h3>

<p>Pre-stored simulated toy dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(toy)</code></pre>


<h3>Value</h3>

<p>List of simulated input data for Canopy.
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(toy)
</code></pre>

<hr>
<h2 id='toy2'>
Toy dataset 2 for Canopy
</h2><span id='topic+toy2'></span>

<h3>Description</h3>

<p>Pre-stored simulated toy dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(toy2)</code></pre>


<h3>Value</h3>

<p>List of simulated input data for Canopy.
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(toy2)
</code></pre>

<hr>
<h2 id='toy3'>
Toy dataset 3 for Canopy
</h2><span id='topic+toy3'></span>

<h3>Description</h3>

<p>Pre-stored simulated toy dataset. 200 simulated SNAs from a tree with 4
branches. No CNA events at play.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(toy3)</code></pre>


<h3>Value</h3>

<p>List of simulated SNA input data for Canopy.
</p>


<h3>Author(s)</h3>

<p>Yuchao Jiang <a href="mailto:yuchaoj@wharton.upenn.edu">yuchaoj@wharton.upenn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    data(toy3)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
