<!DOCTYPE html><html><head><title>Help for package canprot</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {canprot}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#canprot-package'><p>Chemical Metrics of Differentially Expressed Proteins</p></a></li>
<li><a href='#check_IDs'><p>Check UniProt IDs</p></a></li>
<li><a href='#cleanup'><p>Clean Up Data</p></a></li>
<li><a href='#CLES'><p>Common Language Effect Size</p></a></li>
<li><a href='#diffplot'><p>Plot Differences of Chemical Metrics</p></a></li>
<li><a href='#get_comptab'><p>Calculate Differences of Chemical Metrics</p></a></li>
<li><a href='#human'><p>Amino Acid Compositions of Human Proteins</p></a></li>
<li><a href='#metrics'><p>Calculate Chemical Metrics for Proteins</p></a></li>
<li><a href='#mkvig'><p>Compile and View Vignettes from the Command Line</p></a></li>
<li><a href='#pdat_'><p>Get Protein Expression Data</p></a></li>
<li><a href='#protcomp'><p>Amino Acid Compositions</p></a></li>
<li><a href='#qdist'><p>Quantile Distributions for One Dataset</p></a></li>
<li><a href='#xsummary'><p>Summarize Chemical Differences</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Date:</td>
<td>2022-01-17</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.2</td>
</tr>
<tr>
<td>Title:</td>
<td>Chemical Metrics of Differentially Expressed Proteins</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jeffrey Dick &lt;j3ffdick@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0), CHNOSZ (&ge; 1.3.2)</td>
</tr>
<tr>
<td>Imports:</td>
<td>xtable, MASS, rmarkdown</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, testthat</td>
</tr>
<tr>
<td>Description:</td>
<td>Chemical metrics of differentially expressed proteins in cancer
  and cell culture proteomics experiments. Data files in the package have amino
  acid compositions of proteins obtained from UniProt and &gt;250 published lists of
  up- and down-regulated proteins in different cancer types and laboratory
  experiments. Functions are provided to calculate chemical metrics including
  protein length, grand average of hydropathicity (GRAVY), isoelectric point
  (pI), carbon oxidation state, and stoichiometric hydration state; the latter
  two are described in Dick et al. (2020) &lt;<a href="https://doi.org/10.5194%2Fbg-17-6145-2020">doi:10.5194/bg-17-6145-2020</a>&gt;. The
  vignettes visualize differences of chemical metrics between up- and
  down-regulated proteins and list literature references for all datasets.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>BuildResaveData:</td>
<td>no</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/jedick/canprot">https://github.com/jedick/canprot</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-01-17 03:55:34 UTC; jedick</td>
</tr>
<tr>
<td>Author:</td>
<td>Jeffrey Dick <a href="https://orcid.org/0000-0002-0687-5890"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Ben Bolker <a href="https://orcid.org/0000-0002-2127-0443"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-01-17 08:22:51 UTC</td>
</tr>
</table>
<hr>
<h2 id='canprot-package'>Chemical Metrics of Differentially Expressed Proteins</h2><span id='topic+canprot-package'></span>

<h3>Description</h3>

<p><span class="pkg">canprot</span> is a package for computing chemical metrics of proteins from their amino acid compositions.
The package has datasets for differentially expressed proteins in cancer and cell culture conditions from over 250 studies.
</p>


<h3>Overview</h3>

<p>This package includes datasets for differential expression of proteins in six cancer types (breast, colorectal, liver, lung, pancreatic, prostate), and four cell culture conditions (hypoxia, hyperosmotic stress, secreted proteins in hypoxia, and 3D compared to 2D growth conditions).
The hyperosmotic stress data are divided into bacteria, archaea (both high- and low-salt experiments) and eukaryotes; the latter are further divided into salt and glucose experiments.
Nearly all datasets use UniProt IDs; if not given in the original publications they have been added using the UniProt mapping tool (<a href="https://www.uniprot.org/mapping/">https://www.uniprot.org/mapping/</a>).
</p>
<p>The analysis vignettes have plots for each cancer type and cell culture condition and references for all data sources used.
Because of their size, pre-built vignette HTML files are not included with the package; use <code><a href="#topic+mkvig">mkvig</a></code> to compile and view any of the vignettes.
</p>
<p>The functions in this package were originally based on code for the papers of Dick (2016 and 2017).
Updated data compilations and revised vignettes were developed by Dick et al. (2020) and Dick (2021).
</p>


<h3>References</h3>

<p>Dick, J. M. (2016) Proteomic indicators of oxidation and hydration state in colorectal cancer. <em>PeerJ</em> <b>4</b>, e2238. doi: <a href="https://doi.org/10.7717/peerj.2238">10.7717/peerj.2238</a>
</p>
<p>Dick, J. M. (2017) Chemical composition and the potential for proteomic transformation in cancer, hypoxia, and hyperosmotic stress. <em>PeerJ</em> <b>5</b>, e3421. doi: <a href="https://doi.org/10.7717/peerj.3421">10.7717/peerj.3421</a>
</p>
<p>Dick, J. M., Yu, M. and Tan, J. (2020) Uncovering chemical signatures of salinity gradients through compositional analysis of protein sequences. <em>Biogeosciences</em> <b>17</b>, 6145&ndash;6162. doi: <a href="https://doi.org/10.5194/bg-17-6145-2020">10.5194/bg-17-6145-2020</a>
</p>
<p>Dick, J. M. (2021) Water as a reactant in the differential expression of proteins in cancer. <em>Comp. Sys. Onco.</em> <b>1</b>:e1007. doi: <a href="https://doi.org/10.1002/cso2.1007">10.1002/cso2.1007</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># List the data files for all studies
# (one study can have more than one dataset)
exprdata &lt;- system.file("extdata/expression", package="canprot")
datafiles &lt;- dir(exprdata, recursive=TRUE)
print(datafiles)
# Show the number of data files for each condition
table(dirname(datafiles))
</code></pre>

<hr>
<h2 id='check_IDs'>Check UniProt IDs</h2><span id='topic+check_IDs'></span>

<h3>Description</h3>

<p>Find the first ID for each protein that matches a known UniProt ID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  check_IDs(dat, IDcol, aa_file = NULL, updates_file = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_IDs_+3A_dat">dat</code></td>
<td>
<p>data frame, protein expression data</p>
</td></tr>
<tr><td><code id="check_IDs_+3A_idcol">IDcol</code></td>
<td>
<p>character, name of column that has the UniProt IDs</p>
</td></tr>
<tr><td><code id="check_IDs_+3A_aa_file">aa_file</code></td>
<td>
<p>character, name of file with additional amino acid compositions</p>
</td></tr>
<tr><td><code id="check_IDs_+3A_updates_file">updates_file</code></td>
<td>
<p>character, name of file with old to new ID mappings</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>check_IDs</code> is used to check for known UniProt IDs and to update obsolete IDs.
The source IDs should be provided in the <code>IDcol</code> column of <code>dat</code>; multiple IDs for one protein can be separated by a semicolon.
</p>
<p>The function keeps the first &ldquo;known&rdquo; ID for each protein, which must be present in one of these groups:
</p>

<ul>
<li><p> The <code><a href="#topic+human_aa">human_aa</a></code> dataset of amino acid compositions.
</p>
</li>
<li><p> Old UniProt IDs that are mapped to new UniProt IDs in <code><a href="#topic+uniprot_updates">uniprot_updates</a></code> or in <code>updates_file</code> if specified.
</p>
</li>
<li><p> IDs of proteins in <code>aa_file</code>, which lists amino acid compositions in the format described for <code><a href="#topic+human_aa">human_aa</a></code> (see <code>extdata/protein/human_extra.csv</code> for an example and <code><a href="CHNOSZ.html#topic+thermo">thermo</a>$protein</code> for more details).
</p>
</li></ul>



<h3>Value</h3>

<p><code>dat</code> is returned with possibly changed values in the column designated by <code>IDcol</code>; old IDs are replaced with new ones, the first known ID for each protein is kept, then proteins with no known IDs are assigned <code>NA</code>.
</p>


<h3>See Also</h3>

<p>This function is used by the <code><a href="#topic+pdat_">pdat_</a></code> functions, where it is called before <code><a href="#topic+cleanup">cleanup</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Make up some data for this example
ID &lt;- c("P61247;PXXXXX", "PYYYYY;P46777;P60174", "PZZZZZ")
dat &lt;- data.frame(ID = ID, stringsAsFactors = FALSE)
# Get the first known ID for each protein; the third one is NA
check_IDs(dat, "ID")

# Update an old ID
dat &lt;- data.frame(Entry = "P50224", stringsAsFactors = FALSE)
check_IDs(dat, "Entry")
</code></pre>

<hr>
<h2 id='cleanup'>Clean Up Data</h2><span id='topic+cleanup'></span>

<h3>Description</h3>

<p>Remove proteins with unavailable IDs, ambiguous expression ratios, and duplicated IDs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  cleanup(dat, IDcol, up2 = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cleanup_+3A_dat">dat</code></td>
<td>
<p>data frame, protein expression data</p>
</td></tr>
<tr><td><code id="cleanup_+3A_idcol">IDcol</code></td>
<td>
<p>character, name of column that has the UniProt IDs</p>
</td></tr>
<tr><td><code id="cleanup_+3A_up2">up2</code></td>
<td>
<p>logical, TRUE for up-regulated proteins, FALSE for down-regulated proteins</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>cleanup</code> is used in the <code><a href="#topic+pdat_">pdat_</a></code> functions to clean up the dataset given in <code>dat</code>.
<code>IDcol</code> is the name of the column that has the UniProt IDs, and <code>up2</code> indicates the expression change for each protein.
The function removes proteins with unavailable (NA or &quot;&quot;) or duplicated IDs.
If <code>up2</code> is provided, the function also removes unquantified proteins (those that have NA values of <code>up2</code>) and those with ambiguous expression ratios (up and down for the same ID).
For each operation, a message is printed describing the number of proteins that are &lsquo;<span class="samp">&#8288;unavailable&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;unquantified&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;ambiguous&#8288;</span>&rsquo;, or &lsquo;<span class="samp">&#8288;duplicated&#8288;</span>&rsquo;.
</p>
<p>Alternatively, if <code>IDcol</code> is a logical value, it selects proteins to be unconditionally removed.
</p>


<h3>See Also</h3>

<p>This function is used extensively by the <code><a href="#topic+pdat_">pdat_</a></code> functions, where it is called after <code><a href="#topic+check_IDs">check_IDs</a></code> (if needed).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Set up a simple workflow
extdatadir &lt;- system.file("extdata", package="canprot")
datadir &lt;- paste0(extdatadir, "/expression/pancreatic/")
dataset &lt;- "CYD+05"
dat &lt;- read.csv(paste0(datadir, dataset, ".csv.xz"), as.is = TRUE)
up2 &lt;- dat$Ratio..cancer.normal. &gt; 1
# Remove two unavailable and one duplicated proteins
dat &lt;- cleanup(dat, "Entry", up2)
# Now we can retrieve the amino acid compositions
pcomp &lt;- protcomp(dat$Entry)

# Read another data file
datadir &lt;- paste0(system.file("extdata", package="canprot"), "/expression/colorectal/")
dataset &lt;- "STK+15"
dat &lt;- read.csv(paste0(datadir, "STK+15.csv.xz"), as.is = TRUE)
# Remove unavailable proteins
dat &lt;- cleanup(dat, "uniprot")
# Remove proteins that have less than 2-fold expression ratio
dat &lt;- cleanup(dat, abs(log2(dat$invratio)) &lt; 1)
</code></pre>

<hr>
<h2 id='CLES'>Common Language Effect Size</h2><span id='topic+CLES'></span>

<h3>Description</h3>

<p>Calculate the common language effect size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  CLES(x, y, distribution = "normal")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CLES_+3A_x">x</code></td>
<td>
<p>numeric, data</p>
</td></tr>
<tr><td><code id="CLES_+3A_y">y</code></td>
<td>
<p>numeric, data</p>
</td></tr>
<tr><td><code id="CLES_+3A_distribution">distribution</code></td>
<td>
<p>&lsquo;<span class="samp">&#8288;normal&#8288;</span>&rsquo; to use probabilities calculated for a normal distribution, or NA for empirical probabilities</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The common language statistic is defined for continuous data as &ldquo;the probability that a score sampled at random from one distribution will be greater than a score sampled from some other distribution&rdquo; (McGraw and Wong, 1992).
</p>
<p>Given the default value of <code>distribution</code> (&lsquo;<span class="samp">&#8288;normal&#8288;</span>&rsquo;), this function uses <code><a href="stats.html#topic+pnorm">pnorm</a></code> to calculate the probability that a random sample from the unit normal distribution is greater than the Z score (i.e. (the mean of &lsquo;<span class="samp">&#8288;y&#8288;</span>&rsquo; minus the mean of &lsquo;<span class="samp">&#8288;x&#8288;</span>&rsquo;) / square root of (variance of &lsquo;<span class="samp">&#8288;x&#8288;</span>&rsquo; plus variance of &lsquo;<span class="samp">&#8288;y&#8288;</span>&rsquo;)).
</p>
<p>If <code>distribution</code> is NA, this function calculates the empirical probability that the difference is positive, that is, the fraction of all possible pairings between elements of <code>x</code> and <code>y</code> where the difference (&lsquo;<span class="samp">&#8288;y&#8288;</span>&rsquo; value - &lsquo;<span class="samp">&#8288;x&#8288;</span>&rsquo; value) is positive.
It may not be possible to calculate the empirical probability for very large samples because of memory limits.
</p>
<p>The examples use <em>simulated data for normal distributions</em>, given the sample size, mean, and standard deviation of datasets cited by McGraw and Wong, 1992.
Therefore, the empirical probability in the examples approaches the normal curve probability.
However, the empirical probability for <em>nonnormal</em> distributions is distinct from the normal curve probability, as discussed on p. 364-365 of McGraw and Wong, 1992.
</p>


<h3>References</h3>

<p>McGraw, Kenneth O. and Wong, S. P. (1992) A common language effect size statistic. <em>Psychological Bulletin</em> <b>11</b>, 361&ndash;365. doi: <a href="https://doi.org/10.1037/0033-2909.111.2.361">10.1037/0033-2909.111.2.361</a>
</p>
<p>National Center for Health Statistics (1987) <em>Anthropometric Reference Data and Prevalence of Overweight: United States, 1976-1980.</em> Data from the National Health Survey, Series 11, No. 238. DHHS Publication (PHS) No. 87-1688. U.S. Government Printing Office, Washington, DC. <a href="https://www.cdc.gov/nchs/data/series/sr_11/sr11_238.pdf">https://www.cdc.gov/nchs/data/series/sr_11/sr11_238.pdf</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1: Height differences between males and females
# a) Use statistics quoted by McGraw and Wong, 1992 from NCHS, 1987
# for heights in inches of 18-24 year-old males and females
# Table 14: number, mean height, and standard deviation of height of females
n1 &lt;- 1066
M1 &lt;- 64.3
SD1 &lt;- 2.8
# Table 13: number, mean height, and standard deviation of height of males
n2 &lt;- 988
M2 &lt;- 69.7
SD2 &lt;- 2.6
# b) Simulate data from a normal distribution with exact mean and SD
# use rnorm2 function from Ben Bolker's answer to
# https://stackoverflow.com/questions/18919091/generate-random-numbers-with-fixed-mean-and-sd
rnorm2 &lt;- function(n, mean, sd) { mean + sd * scale(rnorm(n)) }
set.seed(1234)
height_female &lt;- rnorm2(n1, M1, SD1)
height_male &lt;- rnorm2(n2, M2, SD2)
# c) Calculate the CLES using the normal distribution and empirical probability
CLES_normal &lt;- CLES(height_female, height_male)
CLES_empirical &lt;- CLES(height_female, height_male, distribution = NA)
# d) Test numerical equivalence of the results
# The CLES is approximately 0.92 (McGraw and Wong, 1992)
# (note: becasue we used rnorm2, this doesn't depend on the seed)
stopifnot(all.equal(CLES_normal, 0.92, tol = 0.01))
# With this seed, the difference between the normal curve probability
# and empirical probability is less than 1%
stopifnot(all.equal(CLES_normal, CLES_empirical, tol = 0.01))

# Example 1.5: Use multiple simulated datasets to show approach
# of empirical probability to normal curve probability
CLES_empirical_n &lt;- sapply(1:100, function(x) {
  height_female &lt;- rnorm2(n1, M1, SD1)
  height_male &lt;- rnorm2(n2, M2, SD2)
  CLES(height_female, height_male, distribution = NA)
})
CLES_empirical &lt;- mean(CLES_empirical_n)
# now we're even closer to the normal curve probability
stopifnot(all.equal(CLES_normal, CLES_empirical, tol = 0.0001))

# Example 2: Multiple datasets in Table 2 of McGraw and Wong, 1992
# Sample statistics for females
n1 &lt;- c(638, 672, 3139, 420740, 19274, 104263, 207, 394, 1066, 982, 108, 108)
M1 &lt;- c(103, 15, 103, 18.9, 30, 16.1, 6.9, 13.3, 64.3, 134, 45, 94)
SD1 &lt;- sqrt(c(908, 74, 219, 27, 110, 59, 15, 164, 6.8, 688, 310, 1971))
# Sample statistics for males
n2 &lt;- c(354, 359, 3028, 356704, 21768, 133882, 199, 469, 988, 988, 443, 443)
M2 &lt;- c(112, 23, 100, 17.9, 33, 18.6, 9.3, 21.8, 69.7, 163, 86, 212)
SD2 &lt;- sqrt(c(1096, 96, 202, 29, 110, 61, 15, 133, 7.8, 784, 818, 5852))
# A function to calculate the effect size using simulated data
CLESfun &lt;- function(n1, M1, SD1, n2, M2, SD2, distribution) {
  rnorm2 &lt;- function(n, mean, sd) { mean + sd * scale(rnorm(n)) }
  set.seed(1234)
  x &lt;- rnorm2(n1, M1, SD1)
  y &lt;- rnorm2(n2, M2, SD2)
  CLES(x, y, distribution)
}
# Calculate 100 * CL for the normal curve probabilities
CLnorm &lt;- sapply(1:12, function(i) {
  CL &lt;- CLESfun(n1[i], M1[i], SD1[i], n2[i], M2[i], SD2[i], "normal")
  round(100 * CL)
})
# Calculate 100 * CL for empirical probabilities
CLemp &lt;- sapply(1:12, function(i) {
  # skip very large samples: not enough memory
  if(n1[i] &gt; 5000 | n2[i] &gt; 5000) NA else {
    CL &lt;- CLESfun(n1[i], M1[i], SD1[i], n2[i], M2[i], SD2[i], NA)
    round(100 * CL)
  }
})
# The difference between the empirical and normal curve
# probabilities is not more than 1 percent
stopifnot(max(abs(CLemp - CLnorm), na.rm = TRUE) &lt;= 1)
# TODO: Why are some of the calculated values different from
# Table 2 of McGraw and Wong, 1992?
CLref &lt;- c(54, 74, 44, 45, 56, 63, 67, 65, 92, 78, 89, 91)
# Differences range from -4 to 4
range(CLnorm - CLref) 
#stopifnot(max(abs(CLnorm - CLref)) == 0)
</code></pre>

<hr>
<h2 id='diffplot'>Plot Differences of Chemical Metrics</h2><span id='topic+diffplot'></span><span id='topic+cplab'></span>

<h3>Description</h3>

<p>Make a plot showing differences of selected chemical metrics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  diffplot(comptab, vars = c("ZC", "nH2O"), col = "black", plot.rect = FALSE,
           pt.text = c(letters, LETTERS), cex.text = 0.85, oldstyle = FALSE,
           pch = 1, cex = 2.1, contour = TRUE, col.contour = par("fg"),
           probs = 0.5, add = FALSE, labtext = NULL, ...)
  cplab
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diffplot_+3A_comptab">comptab</code></td>
<td>
<p>list or data frame, chemical differences generated by <code><a href="#topic+get_comptab">get_comptab</a></code></p>
</td></tr>
<tr><td><code id="diffplot_+3A_vars">vars</code></td>
<td>
<p>character, which variables to plot</p>
</td></tr>
<tr><td><code id="diffplot_+3A_col">col</code></td>
<td>
<p>character or numeric, color(s) for the points</p>
</td></tr>
<tr><td><code id="diffplot_+3A_plot.rect">plot.rect</code></td>
<td>
<p>logical, plot a reference rectangle?</p>
</td></tr>
<tr><td><code id="diffplot_+3A_pt.text">pt.text</code></td>
<td>
<p>character, text labels for the points</p>
</td></tr>
<tr><td><code id="diffplot_+3A_cex.text">cex.text</code></td>
<td>
<p>numeric, size of text labels</p>
</td></tr>
<tr><td><code id="diffplot_+3A_oldstyle">oldstyle</code></td>
<td>
<p>logical, use old style plot?</p>
</td></tr>
<tr><td><code id="diffplot_+3A_pch">pch</code></td>
<td>
<p>numeric, point symbol</p>
</td></tr>
<tr><td><code id="diffplot_+3A_cex">cex</code></td>
<td>
<p>numeric, point size</p>
</td></tr>
<tr><td><code id="diffplot_+3A_contour">contour</code></td>
<td>
<p>logical, add contour lines?</p>
</td></tr>
<tr><td><code id="diffplot_+3A_col.contour">col.contour</code></td>
<td>
<p>character or numeric, color of contour lines</p>
</td></tr>
<tr><td><code id="diffplot_+3A_probs">probs</code></td>
<td>
<p>numeric, probability level(s) for contours</p>
</td></tr>
<tr><td><code id="diffplot_+3A_add">add</code></td>
<td>
<p>logical, add to an existing plot?</p>
</td></tr>
<tr><td><code id="diffplot_+3A_labtext">labtext</code></td>
<td>
<p>character, text to add to axis labels</p>
</td></tr>
<tr><td><code id="diffplot_+3A_...">...</code></td>
<td>
<p>other argumenents passed to <code><a href="graphics.html#topic+plot">plot</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>A plot is created with points showing the differences between up- and down-regulated proteins for two chemical metrics, as calculated by <code><a href="#topic+get_comptab">get_comptab</a></code>.
The default setting of <code>vars</code> refers to average oxidation state of carbon (<I>Z</I><sub>C</sub>) as the x-variable and stoichiometric hydration state (<i>n</i><sub>H<sub>2</sub>O</sub>) as the y-variable.
</p>
<p>The colors of the points are controlled by <code>col</code>, which is recycled to be equal to the number of comparisons in <code>comptab</code>.
</p>
<p>If <code>plot.rect</code> is TRUE, a shaded <code><a href="graphics.html#topic+rect">rect</a></code>angle is drawn with coordinates -0.01, -0.01, 0.01, 0.01.
This is useful for visualizing the different scales of multi-panel plots.
</p>
<p>If <code>pt.text</code> is not NA or FALSE, <code><a href="graphics.html#topic+text">text</a></code> labels are added with size controlled by <code>cex.text</code>.
The default value produces labels that are taken sequentially from the 26 lowercase Roman letters in alphabetical order (<code><a href="base.html#topic+letters">letters</a></code>), followed by the set of uppercase letters (<code><a href="base.html#topic+LETTERS">LETTERS</a></code>).
</p>
<p>For <code>labtext</code> = NULL, descriptive text (&ldquo;median difference&rdquo; or &ldquo;mean difference&rdquo;) is added to the axis labels in parentheses.
This text can be changed by giving a value in <code>labtext</code> (for both axes), two values (for each axis), or NA to suppress the text.
</p>
<p><code>cplab</code> is a list of formatted labels used by <code>diffplot</code>.
It is an exported object, available to the user and other packages.
</p>


<h3>Plot style</h3>

<p>The overall style of the plot is controlled by <code>oldstyle</code>.
</p>

<dl>
<dt><code>oldstyle = FALSE</code></dt><dd>
<p>This is the current default style.
Use <code>pch</code> and <code>cex</code> to control the point symbol and size.
Contours are added for confidence regions of highest probability density, computed using a 2-D kernel density estimate (<code><a href="MASS.html#topic+kde2d">kde2d</a></code>).
<code>probs</code> gives the probability level(s) and <code>col.contour</code> sets the color(s) of the contour lines.
<code>contour</code> can be a logical vector, indicating which points to include; set it to FALSE to omit the contour lines.
</p>
<p>The code to calculate the contour levels is modified from <code>HPDregionplot</code> in the <span class="pkg">emdbook</span> package by Ben Bolker (<a href="https://cran.r-project.org/package=emdbook">https://cran.r-project.org/package=emdbook</a>).
</p>
</dd>
<dt><code>oldstyle = TRUE</code></dt><dd>
<p>This style was used for the historical (2017) vignettes, which have been moved to the &lsquo;<span class="samp">&#8288;extdata/cpcp&#8288;</span>&rsquo; directory in <span class="pkg">JMDplots</span> (<a href="https://github.com/jedick/JMDplots">https://github.com/jedick/JMDplots</a>).
For each dataset, the point symbol is a filled square if the <em>p</em>-values of both the x-variable and y-variable are less than 0.05, a filled circle if the <em>p</em>-value of one of the x- or y-variables is less than 0.05, and an open circle otherwise.
A solid line is drawn from the point to the corresponding axis if the rounded, absolute value of (<code><a href="#topic+CLES">CLES</a></code> in percent - 50) of the x- or y-variable is greater than or equal 10.
Otherwise, a dashed line is drawn from the point to the corresponding axis if the <em>p</em>-value of the x- or y-variable is less than 0.05.
Otherwise, no line is drawn.
</p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+qdist">qdist</a></code> to plot quantile distributions for a single dataset.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(CHNOSZ)
# Make an old-style plot for two datasets
comptab &lt;- lapply(c("JKMF10", "WDO+15_C.N"), function(dataset) {
  pdat &lt;- pdat_colorectal(dataset)
  get_comptab(pdat, oldstyle = TRUE)
})
diffplot(comptab, oldstyle = TRUE)
</code></pre>

<hr>
<h2 id='get_comptab'>Calculate Differences of Chemical Metrics</h2><span id='topic+get_comptab'></span>

<h3>Description</h3>

<p>Compute differences of carbon oxidation state, stoichiometric hydration state and other chemical metrics between groups of up- and down-regulated proteins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  get_comptab(pdat, var1 = "ZC", var2 = "nH2O", plot.it = FALSE,
    mfun = "median", oldstyle = FALSE, basis = getOption("basis"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_comptab_+3A_pdat">pdat</code></td>
<td>
<p>list, data object generated by a <code><a href="#topic+pdat_">pdat_</a></code> function</p>
</td></tr>
<tr><td><code id="get_comptab_+3A_var1">var1</code></td>
<td>
<p>character, the first variable</p>
</td></tr>
<tr><td><code id="get_comptab_+3A_var2">var2</code></td>
<td>
<p>character, the second variable</p>
</td></tr>
<tr><td><code id="get_comptab_+3A_plot.it">plot.it</code></td>
<td>
<p>logical, make a scatterplot?</p>
</td></tr>
<tr><td><code id="get_comptab_+3A_mfun">mfun</code></td>
<td>
<p>character, either &lsquo;<span class="samp">&#8288;median&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;mean&#8288;</span>&rsquo;</p>
</td></tr>
<tr><td><code id="get_comptab_+3A_oldstyle">oldstyle</code></td>
<td>
<p>logical, also calculate <code><a href="#topic+CLES">CLES</a></code> and <em>p</em>-values?</p>
</td></tr>
<tr><td><code id="get_comptab_+3A_basis">basis</code></td>
<td>
<p>character, keyword for basis species to use</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The available variables are:
</p>

<table>
<tr>
 <td style="text-align: left;">
  &lsquo;<span class="samp">&#8288;ZC&#8288;</span>&rsquo;   </td><td style="text-align: left;"> average oxidation state of carbon (<I>Z</I><sub>C</sub>; see <code><a href="#topic+ZCAA">ZCAA</a></code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
  &lsquo;<span class="samp">&#8288;nH2O&#8288;</span>&rsquo; </td><td style="text-align: left;"> stoichiometric hydration state per residue (<i>n</i><sub>H<sub>2</sub>O</sub>; see <code><a href="#topic+H2OAA">H2OAA</a></code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
  &lsquo;<span class="samp">&#8288;nO2&#8288;</span>&rsquo; </td><td style="text-align: left;"> stoichiometric oxidation state per residue (<i>n</i><sub>O<sub>2</sub></sub>; see <code><a href="#topic+O2AA">O2AA</a></code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
  &lsquo;<span class="samp">&#8288;V0&#8288;</span>&rsquo;   </td><td style="text-align: left;"> standard molal volume per residue </td>
</tr>
<tr>
 <td style="text-align: left;">
  &lsquo;<span class="samp">&#8288;nAA&#8288;</span>&rsquo;  </td><td style="text-align: left;"> protein length (number of amino acids) </td>
</tr>
<tr>
 <td style="text-align: left;">
  &lsquo;<span class="samp">&#8288;GRAVY&#8288;</span>&rsquo;</td><td style="text-align: left;"> grand average of hydropathicity (see <code><a href="#topic+GRAVY">GRAVY</a></code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
  &lsquo;<span class="samp">&#8288;pI&#8288;</span>&rsquo;   </td><td style="text-align: left;"> isoelectric point (see <code><a href="#topic+pI">pI</a></code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
  &lsquo;<span class="samp">&#8288;MW&#8288;</span>&rsquo;   </td><td style="text-align: left;"> molecular weight per residue </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Differentially expressed proteins are identified by the value of <code>pdat$up2</code> (TRUE for up-regulated proteins and FALSE for down-regulated proteins).
The differences are calculated as (median for up-regulated proteins) - (median for down-regulated proteins); if <code>mfun</code> is &lsquo;<span class="samp">&#8288;mean&#8288;</span>&rsquo;, means of the groups are used instead.
If <code>oldstyle</code> is TRUE, the function also calculates the common language effect size (<code><a href="#topic+CLES">CLES</a></code>, in percent) and <em>p</em>-value for each variable.
</p>
<p>The <code>basis</code> argument is used to select the basis species, which are used for the calculation of <i>n</i><sub>H<sub>2</sub>O</sub> and <i>n</i><sub>O<sub>2</sub></sub>.
The default for <code>getOption("basis")</code> is to use the &lsquo;<span class="samp">&#8288;QEC&#8288;</span>&rsquo; basis species (see <code><a href="#topic+metrics">metrics</a></code>).
</p>
<p>Volume is calculated using amino acid group additivity as described by Dick et al. (2006).
</p>
<p>Set <code>plot.it</code> to <code>TRUE</code> to make a scatterplot.
Open red squares and filled blue circles stand for up-regulated and down-regulated proteins, respectively.
</p>


<h3>Value</h3>

<p>A data frame is returned invisibly containing the columns &lsquo;<span class="samp">&#8288;dataset&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;description&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;n1&#8288;</span>&rsquo; (number of down-regulated proteins), &lsquo;<span class="samp">&#8288;n2&#8288;</span>&rsquo; (number of up-regulated proteins), followed two sets of columns for the variables.
These are denoted generically as (&lsquo;<span class="samp">&#8288;var.mfun1&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;var.mfun2&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;var.diff&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;var.CLES&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;var.p.value&#8288;</span>&rsquo;), where &lsquo;<span class="samp">&#8288;var&#8288;</span>&rsquo; is replaced by the name of <code>var1</code> or <code>var2</code>, and &lsquo;<span class="samp">&#8288;mfun&#8288;</span>&rsquo; is replaced by the value of <code>mfun</code>.
For example, &lsquo;<span class="samp">&#8288;ZC.median1&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;ZC.median2&#8288;</span>&rsquo; are the median <I>Z</I><sub>C</sub> of the down- and up-regulated proteins, respectively.
</p>


<h3>References</h3>

<p>Dick, J. M., LaRowe, D. E. and Helgeson, H. C. (2006) Temperature, pressure, and electrochemical constraints on protein speciation: Group additivity calculation of the standard molal thermodynamic properties of ionized unfolded proteins. <em>Biogeosciences</em> <b>3</b>, 311&ndash;336. doi: <a href="https://doi.org/10.5194/bg-3-311-2006">10.5194/bg-3-311-2006</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pd &lt;- pdat_colorectal("JKMF10")
# default variables: ZC and nH2O
get_comptab(pd, plot.it = TRUE)
# protein length and per-residue volume
get_comptab(pd, "nAA", "V0", plot.it = TRUE)
</code></pre>

<hr>
<h2 id='human'>Amino Acid Compositions of Human Proteins</h2><span id='topic+human'></span><span id='topic+human_aa'></span><span id='topic+human_base'></span><span id='topic+human_additional'></span><span id='topic+human_extra'></span><span id='topic+uniprot_updates'></span>

<h3>Description</h3>

<p>Data for amino acid compositions of proteins and conversion from old to new UniProt IDs.
</p>


<h3>Format</h3>

<p><code>human_aa</code> is a data frame with 25 columns in the format used for amino acid compositions in <span class="pkg">CHNOSZ</span> (see <code><a href="CHNOSZ.html#topic+thermo">thermo</a></code>):
</p>

<table>
<tr>
 <td style="text-align: left;">
  <code>protein</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Identification of protein</td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>organism</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Identification of organism</td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>ref</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Reference key for source of sequence data</td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>abbrv</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Abbreviation or other ID for protein (e.g. gene name)</td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>chains</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Number of polypeptide chains in the protein</td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>Ala</code>...<code>Tyr</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Number of each amino acid in the protein
</td>
</tr>

</table>

<p>The <code>protein</code> column contains UniProt IDs in the format <code>database|accession-isoform</code>, where <code>database</code> is most often &lsquo;<span class="samp">&#8288;sp&#8288;</span>&rsquo; (Swiss-Prot) or &lsquo;<span class="samp">&#8288;tr&#8288;</span>&rsquo; (TrEMBL), and <code>isoform</code> is an optional suffix indicating the isoform of the protein (particularly in the <code>human_additional</code> file).
</p>


<h3>Details</h3>

<p>The amino acid compositions of human proteins are stored in three files under <code>extdata/protein</code>.
</p>

<ul>
<li> <p><code>human_base.rds</code> contains amino acid compositions of canonical isoforms of manually reviewed proteins in the <a href="https://www.uniprot.org/">UniProt</a> reference human proteome (computed from sequences in <code>UP000005640_9606.fasta.gz</code>, dated 2016-04-03).
</p>
</li>
<li> <p><code>human_additional.rds</code> contains amino acid compositions of additional proteins (<code>UP000005640_9606_additional.fasta.gz</code>) including isoforms and unreviewed sequences. In version 0.1.5, this file was trimmed to include only those proteins that are used in any of the datasets in the package.
</p>
</li>
<li> <p><code>human_extra.csv</code> contains amino acid compositions of other (&ldquo;extra&rdquo;) proteins used in a dataset but not listed in one of the files above. These proteins may include obsolete, unreviewed, or newer additions to the UniProt database. Most, but not all, sequences here are HUMAN (see the <code>organism</code> column and the <code>ref</code> column for the reference keys).
</p>
</li></ul>

<p>On loading the package, the individual data files are read and combined, and the result is assigned to the <code>human_aa</code> object in the <code>human</code> environment.
</p>
<p>As an aid for processing datasets that list old (obsolete) UniProt IDs, the corresponding new (current) IDs are are stored in <code>uniprot_updates</code>.
These ID mappings have been manually added as needed for individual datasets, and include proteins from humans as well as other organisms.
<code><a href="#topic+check_IDs">check_IDs</a></code> performs the conversion of old to new IDs.
</p>


<h3>See Also</h3>

<p>Amino acid compositions of non-human proteins are stored under <code>extdata/aa</code> in directories <code>archaea</code>, <code>bacteria</code>, <code>cow</code>, <code>dog</code>, <code>mouse</code>, <code>rat</code>, and <code>yeast</code>.
These files can be loaded in <code><a href="#topic+protcomp">protcomp</a></code> via the <code>aa_file</code> argument, which is used e.g. in <code><a href="#topic+pdat_osmotic_bact">pdat_osmotic_bact</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># The number of proteins
nrow(get("human_aa", human))
# The number of old to new ID mappings
nrow(get("uniprot_updates", human))
</code></pre>

<hr>
<h2 id='metrics'>Calculate Chemical Metrics for Proteins</h2><span id='topic+metrics'></span><span id='topic+H2OAA'></span><span id='topic+O2AA'></span><span id='topic+ZCAA'></span><span id='topic+GRAVY'></span><span id='topic+pI'></span><span id='topic+MWAA'></span><span id='topic+basis.text'></span>

<h3>Description</h3>

<p>These functions calculate chemical metrics of proteins given a data frame of amino acid compositions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  ZCAA(AAcomp, nothing = NULL)
  H2OAA(AAcomp, basis = getOption("basis"))
  O2AA(AAcomp, basis = getOption("basis"))
  GRAVY(AAcomp)
  pI(AAcomp)
  MWAA(AAcomp)
  basis.text(basis)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metrics_+3A_aacomp">AAcomp</code></td>
<td>
<p>data frame, amino acid compositions</p>
</td></tr>
<tr><td><code id="metrics_+3A_nothing">nothing</code></td>
<td>
<p>dummy argument</p>
</td></tr>
<tr><td><code id="metrics_+3A_basis">basis</code></td>
<td>
<p>character, basis species</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Columns in <code>AAcomp</code> should be named with the three-letter abbreviations for the amino acids (&lsquo;<span class="samp">&#8288;Ala&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;Arg&#8288;</span>&rsquo;, ...).
Abbreviations are matched without regard to case (e.g. &lsquo;<span class="samp">&#8288;ALA&#8288;</span>&rsquo; is the same as &lsquo;<span class="samp">&#8288;ala&#8288;</span>&rsquo;).
</p>
<p>The metrics are described below:
</p>

<dl>
<dt><code>ZCAA</code></dt><dd><p>Average oxidation state of carbon (<I>Z</I><sub>C</sub>) (Dick, 2014).
<code>nothing</code> is an extra argument that does nothing.
It is provided so that <code><a href="base.html#topic+do.call">do.call</a></code> can be used to run <code>ZCAA</code> or <code>H2OAA</code> with the same number of arguments.
</p>
<p>This metric is independent of the choice of basis species.
</p>
</dd>
<dt><code>H2OAA</code></dt><dd><p>Stoichiometric hydration state (<i>n</i><sub>H<sub>2</sub>O</sub>) per residue.
The available <code>basis</code> species are:
</p>

<ul>
<li><p>&lsquo;<span class="samp">&#8288;QEC&#8288;</span>&rsquo; - glutamine, glutamic acid, cysteine, H<sub>2</sub>O, O<sub>2</sub> (Dick et al., 2020) (this is the default for <code>getOption("basis")</code>)
</p>
</li>
<li><p>&lsquo;<span class="samp">&#8288;QCa&#8288;</span>&rsquo; - glutamine, cysteine, acetic acid, H<sub>2</sub>O, O<sub>2</sub>
</p>
</li>
<li><p>Any other valid basis specification for <code><a href="CHNOSZ.html#topic+basis">basis</a></code>, such as &lsquo;<span class="samp">&#8288;CHNOS&#8288;</span>&rsquo; for CO<sub>2</sub>, NH<sub>3</sub>, H<sub>2</sub>S, H<sub>2</sub>O, and O<sub>2</sub>
</p>
</li></ul>

</dd>
<dt><code>O2AA</code></dt><dd><p>Stoichiometric oxidation state (<i>n</i><sub>O<sub>2</sub></sub>) per residue.
The basis species also affect this calculation.
</p>
</dd>
<dt><code>GRAVY</code></dt><dd><p>Grand average of hydropathicity.
Values of the hydropathy index for individual amino acids are from Kyte and Doolittle (1982).
</p>
</dd>
<dt><code>pI</code></dt><dd><p>Isoelectric point.
The net charge for each ionizable group was pre-calculated from pH 0 to 14 at intervals of 0.01.
The isoelectric point is found as the pH where the sum of charges of all groups in the protein is closest to zero.
The p<i>K</i> values for the terminal groups and sidechains are taken from Bjellqvist et al. (1993) and Bjellqvist et al. (1994); note that the calculation does not implement position-specific adjustments described in the latter paper.
The number of N- and C-terminal groups is taken to be one, unless a value for <code>chains</code> (number of polypeptide chains) is given in <code>AAcomp</code>.
</p>
</dd>
<dt><code>MWAA</code></dt><dd><p>Molecular weight per residue.</p>
</dd>
</dl>

<p>Note that <I>Z</I><sub>C</sub> is a per-carbon average, but <i>n</i><sub>H<sub>2</sub>O</sub> is a per-residue average.
The contribution of H<sub>2</sub>O from the terminal groups of proteins is counted, so shorter proteins have slightly greater <i>n</i><sub>H<sub>2</sub>O</sub>.
</p>
<p>Tests for a few proteins (see examples) indicate that <code>GRAVY</code> and <code>pI</code> are equal those calculated with the ProtParam tool (<a href="https://web.expasy.org/protparam/">https://web.expasy.org/protparam/</a>; Gasteiger et al., 2005).
</p>
<p><code>basis.text</code> is used in the vignettes to generate a textual description of the names of the basis species, except H<sub>2</sub>O and O<sub>2</sub>, for one of the keywords &lsquo;<span class="samp">&#8288;QEC&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;QCa&#8288;</span>&rsquo;.
</p>


<h3>References</h3>

<p>Bjellqvist, B., Hughes, G. J., Pasquali, C., Paquet, N., Ravier, F., Sanchez, J.-C., Frutiger, S. and Hochstrasser, D. (1993) The focusing positions of polypeptides in immobilized pH gradients can be predicted from their amino acid sequences. <em>Electrophoresis</em> <b>14</b>, 1023&ndash;1031. doi: <a href="https://doi.org/10.1002/elps.11501401163">10.1002/elps.11501401163</a>
</p>
<p>Bjellqvist, B. and Basse, B. and Olsen, E. and Celis, J. E. (1994) Reference points for comparisons of two-dimensional maps of proteins from different human cell types defined in a pH scale where isoelectric points correlate with polypeptide compositions. <em>Electrophoresis</em> <b>15</b>, 529&ndash;539. doi: <a href="https://doi.org/10.1002/elps.1150150171">10.1002/elps.1150150171</a>
</p>
<p>Dick, J. M. (2014) Average oxidation state of carbon in proteins. <em>J. R. Soc. Interface</em> <b>11</b>, 20131095. doi: <a href="https://doi.org/10.1098/rsif.2013.1095">10.1098/rsif.2013.1095</a>
</p>
<p>Dick, J. M., Yu, M. and Tan, J. (2020) Uncovering chemical signatures of salinity gradients through compositional analysis of protein sequences. <em>Biogeosciences</em> <b>17</b>, 6145&ndash;6162. doi: <a href="https://doi.org/10.5194/bg-17-6145-2020">10.5194/bg-17-6145-2020</a>
</p>
<p>Gasteiger, E., Hoogland, C., Gattiker, A., Duvaud, S., Wilkins, M. R., Appel, R. D. and Bairoch, A. (2005) Protein identification and analysis tools on the ExPASy server. In J. M. Walker (Ed.), <em>The Proteomics Protocols Handbook</em> (pp. 571&ndash;607). Totowa, NJ: Humana Press Inc. doi: <a href="https://doi.org/10.1385/1-59259-890-0:571">10.1385/1-59259-890-0:571</a>
</p>
<p>Kyte, J. and Doolittle, R. F. (1982) A simple method for displaying the hydropathic character of a protein. <em>J. Mol. Biol.</em> <b>157</b>, 105&ndash;132. doi: <a href="https://doi.org/10.1016/0022-2836(82)90515-0">10.1016/0022-2836(82)90515-0</a>
</p>


<h3>See Also</h3>

<p>For calculation of <I>Z</I><sub>C</sub> from a chemical formula instead of amino acid composition, see the <code><a href="CHNOSZ.html#topic+ZC">ZC</a></code> function in <span class="pkg">CHNOSZ</span>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># we need CHNOSZ for these examples
require(CHNOSZ)

# for reference, compute ZC of alanine and glycine "by hand"
ZC.Gly &lt;- ZC("C2H5NO2")
ZC.Ala &lt;- ZC("C3H7NO2")
# define the composition of a Gly-Ala-Gly tripeptide
AAcomp &lt;- data.frame(Gly = 2, Ala = 1)
# calculate the ZC of the tripeptide (value: 0.571)
ZC.GAG &lt;- ZCAA(AAcomp)
# this is equal to the carbon-number-weighted average of the amino acids
nC.Gly &lt;- 2 * 2
nC.Ala &lt;- 1 * 3
ZC.average &lt;- (nC.Gly * ZC.Gly + nC.Ala * ZC.Ala) / (nC.Ala + nC.Gly)
stopifnot(all.equal(ZC.GAG, ZC.average))

# compute the per-residue nH2O of Gly-Ala-Gly
basis("QEC")
nH2O.GAG &lt;- species("Gly-Ala-Gly")$H2O
# divide by the length to get residue average (we keep the terminal H-OH)
nH2O.residue &lt;- nH2O.GAG / 3
# compare with the value calculated by H2OAA() (-0.2)
nH2O.H2OAA &lt;- H2OAA(AAcomp, "QEC")
stopifnot(all.equal(nH2O.residue, nH2O.H2OAA))

# calculate GRAVY for a few proteins
# first get the protein index in CHNOSZ's list of proteins
iprotein &lt;- pinfo(c("LYSC_CHICK", "RNAS1_BOVIN", "AMYA_PYRFU"))
# then get the amino acid compositions
AAcomp &lt;- pinfo(iprotein)
# then calculate GRAVY
Gcalc &lt;- as.numeric(GRAVY(AAcomp))
# these are equal to values obtained with ProtParam on uniprot.org
# https://web.expasy.org/cgi-bin/protparam/protparam1?P00698@19-147@
# https://web.expasy.org/cgi-bin/protparam/protparam1?P61823@27-150@
# https://web.expasy.org/cgi-bin/protparam/protparam1?P49067@2-649@
Gref &lt;- c(-0.472, -0.663, -0.325)
stopifnot(all.equal(round(Gcalc, 3), Gref))
# also calculate molecular weight of the proteins
MWcalc &lt;- as.numeric(MWAA(AAcomp)) * protein.length(iprotein)
MWref &lt;- c(14313.14, 13690.29, 76178.25)
stopifnot(all.equal(round(MWcalc, 2), MWref))

# calculate pI for a few proteins
iprotein &lt;- pinfo(c("LYSC_CHICK", "RNAS1_BOVIN", "AMYA_PYRFU", "CSG_HALJP"))
AAcomp &lt;- pinfo(iprotein)
pI_calc &lt;- pI(AAcomp)
# reference values calculated with ProtParam on uniprot.org
# LYSC_CHICK: residues 19-147 (sequence v1)
# RNAS1_BOVIN: residues 27-150 (sequence v1)
# AMYA_PYRFU: residues 2-649 (sequence v2)
# CSG_HALJP: residues 35-862 (sequence v1)
pI_ref &lt;- c(9.32, 8.64, 5.46, 3.37)
stopifnot(all.equal(as.numeric(pI_calc), pI_ref))
</code></pre>

<hr>
<h2 id='mkvig'>Compile and View Vignettes from the Command Line</h2><span id='topic+mkvig'></span>

<h3>Description</h3>

<p>Compile the indicated vignette and open it in the browser.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  mkvig(vig = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mkvig_+3A_vig">vig</code></td>
<td>
<p>character, name of a vignette without &lsquo;<span class="samp">&#8288;.Rmd&#8288;</span>&rsquo; extension</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In order to reduce package space and check time, pre-built vignettes are not included in the package.
<code>mkvig</code> is a convenience function to compile the vignettes on demand and view them in a browser.
</p>
<p>The available vignettes for <code>mkvig</code> are listed here:
</p>

<ul>
<li> <p><em>Cell culture</em> &ndash; &lsquo;<span class="samp">&#8288;hypoxia&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;secreted&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;osmotic_bact&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;osmotic_euk&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;osmotic_halo&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;glucose&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;3D&#8288;</span>&rsquo;
</p>
</li>
<li> <p><em>Cancer</em> &ndash; &lsquo;<span class="samp">&#8288;breast&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;colorectal&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;liver&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;lung&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;pancreatic&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;prostate&#8288;</span>&rsquo;
</p>
</li></ul>

<p>Note that pandoc (including pandoc-citeproc), as a system dependency of <span class="pkg">rmarkdown</span>, is required to build the vignettes.
</p>


<h3>See Also</h3>

<p>The vignettes can also be run using e.g. <code><a href="utils.html#topic+demo">demo</a>("glucose")</code>, and through the interactive help system (<code><a href="utils.html#topic+help.start">help.start</a></code> &gt; Packages &gt; canprot &gt; Code demos).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
mkvig("colorectal")

## End(Not run)
</code></pre>

<hr>
<h2 id='pdat_'>Get Protein Expression Data</h2><span id='topic+pdat_'></span><span id='topic+pdat_breast'></span><span id='topic+pdat_colorectal'></span><span id='topic+pdat_lung'></span><span id='topic+pdat_liver'></span><span id='topic+pdat_pancreatic'></span><span id='topic+pdat_prostate'></span><span id='topic+pdat_hypoxia'></span><span id='topic+pdat_secreted'></span><span id='topic+pdat_3D'></span><span id='topic+pdat_glucose'></span><span id='topic+pdat_osmotic_bact'></span><span id='topic+pdat_osmotic_euk'></span><span id='topic+pdat_osmotic_halo'></span><span id='topic+.pdat_multi'></span><span id='topic+.pdat_osmotic'></span>

<h3>Description</h3>

<p>Get differentially expressed proteins and amino acid compositions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  pdat_breast(dataset = 2020)
  pdat_colorectal(dataset = 2020)
  pdat_liver(dataset = 2020)
  pdat_lung(dataset = 2020)
  pdat_pancreatic(dataset = 2020)
  pdat_prostate(dataset = 2020)
  pdat_hypoxia(dataset = 2020)
  pdat_secreted(dataset = 2020)
  pdat_3D(dataset = 2020)
  pdat_glucose(dataset = 2020)
  pdat_osmotic_bact(dataset = 2020)
  pdat_osmotic_euk(dataset = 2020)
  pdat_osmotic_halo(dataset = 2020)
  .pdat_multi(dataset = 2020)
  .pdat_osmotic(dataset = 2017)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pdat__+3A_dataset">dataset</code></td>
<td>
<p>character, dataset name</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>pdat_</code> functions assemble lists of up- and down-regulated proteins and retrieve their amino acid compositions using <code><a href="#topic+protcomp">protcomp</a></code>.
The result can be used with <code><a href="#topic+get_comptab">get_comptab</a></code> to make a table of chemical metrics that can then be plotted with <code><a href="#topic+diffplot">diffplot</a></code>.
</p>
<p>If <code>dataset</code> is &lsquo;<span class="samp">&#8288;2020&#8288;</span>&rsquo; (the default) or &lsquo;<span class="samp">&#8288;2017&#8288;</span>&rsquo;, the function returns the names of all datasets in the compilation for the respective year.
</p>
<p>Each dataset name starts with a reference key indicating the study (i.e. paper or other publication) where the data were reported.
The reference keys are made by combining the first characters of the authors' family names with the 2-digit year of publication.
</p>
<p>If a study has more than one dataset, the reference key is followed by an underscore and an identifier for the particular dataset.
This identifier is saved in the variable named <code>stage</code> in the functions, but can be any descriptive text.
</p>
<p>To retrieve the data, provide a single dataset name in the <code>dataset</code> argument.
Protein expression data is read from the CSV files stored in <code>extdata/expression/</code>, under the subdirectory corresponding to the name of the <code>pdat_</code> function.
Some of the functions also read amino acid compositions (e.g. for non-human proteins) from the files in <code>extdata/aa/</code>.
</p>
<p>Descriptions for each function:
</p>

<ul>
<li> <p><code>pdat_colorectal</code>, <code>pdat_pancreatic</code>, <code>pdat_breast</code>, <code>pdat_lung</code>, <code>pdat_prostate</code>, and <code>pdat_liver</code> retrieve data for protein expression in different cancer types.
</p>
</li>
<li> <p><code>pdat_hypoxia</code> gets data for cellular extracts in hypoxia and <code>pdat_secreted</code> gets data for secreted proteins (e.g. exosomes) in hypoxia.
</p>
</li>
<li> <p><code>pdat_3D</code> retrieves data for 3D (e.g. tumor spheroids and aggregates) compared to 2D (monolayer) cell culture.
</p>
</li>
<li> <p><code>.pdat_osmotic</code> retrieves data for hyperosmotic stress, for the 2017 compilation only.
In 2020, this compilation was expanded and split into <code>pdat_osmotic_bact</code> (bacteria), <code>pdat_osmotic_euk</code> (eukaryotic cells) and <code>pdat_osmotic_halo</code> (halophilic bacteria and archaea).
</p>
</li>
<li> <p><code>pdat_glucose</code> gets data for high-glucose experiments in eukaryotic cells.
</p>
</li>
<li> <p><code>.pdat_multi</code> retrieves data for studies that have multiple types of datasets (e.g. both cellular and secreted proteins in hypoxia), and is used internally by the specific functions (e.g. <code>pdat_hypoxia</code> and <code>pdat_secreted</code>).
</p>
</li></ul>



<h3>Value</h3>

<p>A list consisting of:
</p>

<dl>
<dt><code>dataset</code></dt><dd><p>Name of the dataset</p>
</dd>
<dt><code>description</code></dt><dd><p>Descriptive text for the dataset, used for making the tables in the vignettes (see <code><a href="#topic+mkvig">mkvig</a></code>)</p>
</dd>
<dt><code>pcomp</code></dt><dd><p>UniProt IDs together with amino acid compositions obtained using <code><a href="#topic+protcomp">protcomp</a></code></p>
</dd>
<dt><code>up2</code></dt><dd><p>Logical vector with length equal to the number of proteins; TRUE for up-regulated proteins and FALSE for down-regulated proteins</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'># List datasets in the 2017 complilation for colorectal cancer
pdat_colorectal(2017)
# Get proteins and amino acid compositions for one dataset
pdat_colorectal("JKMF10")
</code></pre>

<hr>
<h2 id='protcomp'>Amino Acid Compositions</h2><span id='topic+protcomp'></span>

<h3>Description</h3>

<p>Get amino acid compositions of proteins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  protcomp(uniprot = NULL, aa = NULL, aa_file = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="protcomp_+3A_uniprot">uniprot</code></td>
<td>
<p>character, UniProt IDs of proteins</p>
</td></tr>
<tr><td><code id="protcomp_+3A_aa">aa</code></td>
<td>
<p>data frame, amino acid compositions</p>
</td></tr>
<tr><td><code id="protcomp_+3A_aa_file">aa_file</code></td>
<td>
<p>character, file name</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function retrieves the amino acid compositions of one or more proteins specified by <code>uniprot</code>.
</p>
<p>This function depends on the amino acid compositions of human proteins, which are stored in the <code><a href="#topic+human">human</a></code> environment when the package is attached.
If <code>aa_file</code> is specified, additional amino acid compositions to be considered are read from this file, which should be in the same format as e.g. <code><a href="#topic+human_extra">human_extra</a>.csv</code> (see also <code><a href="CHNOSZ.html#topic+thermo">thermo</a>$protein</code>).
Alternatively, the amino acid compositions can be given in <code>aa</code>, bypassing the search step.
</p>


<h3>Value</h3>

<p>The function returns a list with elements <code>uniprot</code> (UniProt IDs as given in the arguments) and <code>aa</code> (amino acid compositions of the proteins).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cleanup">cleanup</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>protcomp("P24298")
</code></pre>

<hr>
<h2 id='qdist'>Quantile Distributions for One Dataset</h2><span id='topic+qdist'></span>

<h3>Description</h3>

<p>Make a plot showing quantile distributions for up- and down-regulated proteins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  qdist(pdat, vars = c("ZC", "nH2O"), show.steps = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qdist_+3A_pdat">pdat</code></td>
<td>
<p>list, output of a <code><a href="#topic+pdat_">pdat_</a></code> function for a single dataset</p>
</td></tr>
<tr><td><code id="qdist_+3A_vars">vars</code></td>
<td>
<p>character, which variables to plot</p>
</td></tr>
<tr><td><code id="qdist_+3A_show.steps">show.steps</code></td>
<td>
<p>logical, show the steps using <code><a href="stats.html#topic+plot.ecdf">plot.ecdf</a></code>?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function makes a quantile distribution plot with lines for both up- and down-regulated proteins.
The variable (<code>var</code>) can be &lsquo;<span class="samp">&#8288;ZC&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;H2O&#8288;</span>&rsquo;, or both (two plots are made for the latter).
The horizontal axis is the variable and the vertical axis is the quantile point.
A solid black line is drawn for the down-regulated proteins, and a dashed red line for the up-regulated proteins.
The median difference is shown by a gray horizontal line drawn between the distributions at the 0.5 quantile point.
</p>


<h3>References</h3>

<p>Jimenez, C. R. and Knol, J. C. and Meijer, G. A. and Fijneman, R. J. A. (2010) Proteomics of colorectal cancer: Overview of discovery studies and identification of commonly identified cancer-associated proteins and candidate CRC serum markers. <em>J. Proteomics</em> <b>73</b>, 1873&ndash;1895. doi: <a href="https://doi.org/10.1016/j.jprot.2010.06.004">10.1016/j.jprot.2010.06.004</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+diffplot">diffplot</a></code> to plot median differences for multiple datasets.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Plot the data of Jimenez et al., 2010 for colorectal cancer
pdat &lt;- pdat_colorectal("JKMF10")
qdist()
</code></pre>

<hr>
<h2 id='xsummary'>Summarize Chemical Differences</h2><span id='topic+xsummary'></span><span id='topic+xsummary2'></span><span id='topic+xsummary3'></span>

<h3>Description</h3>

<p>Make an HTML table summarizing chemical differences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  xsummary(comptab, vars = c("ZC", "nH2O"))
  xsummary2(comptab1, comptab2)
  xsummary3(comptab1, comptab2, comptab3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xsummary_+3A_comptab">comptab</code></td>
<td>
<p>list or data frame, summary of comparisons generated by <code><a href="#topic+get_comptab">get_comptab</a></code></p>
</td></tr>
<tr><td><code id="xsummary_+3A_vars">vars</code></td>
<td>
<p>character, two variables to tabulate</p>
</td></tr>
<tr><td><code id="xsummary_+3A_comptab1">comptab1</code></td>
<td>
<p>list, output of <code>get_comptab</code></p>
</td></tr>
<tr><td><code id="xsummary_+3A_comptab2">comptab2</code></td>
<td>
<p>list, output of <code>get_comptab</code></p>
</td></tr>
<tr><td><code id="xsummary_+3A_comptab3">comptab3</code></td>
<td>
<p>list, output of <code>get_comptab</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>xsummary</code> makes an HTML table (using <code><a href="xtable.html#topic+xtable">xtable</a></code>) and adds bold and underline formatting to highlight significant chemical differences.
The <em>p</em>-value is bolded if it is less than 0.05, and the percent common language effect size (<code><a href="#topic+CLES">CLES</a></code>) is bolded if it is &lt;= 40 or &gt;= 60.
The mean (or median) difference is [underlined / bolded] if [only one of / both] the <em>p</em>-value and CLES pass these cutoffs.
</p>
<p>The generated table is written to the console, and can be used in a vignette using the <code>results = "asis"</code> chunk option.
The function also returns (invisibly) the data frame used to make the table; this data frame differs from <code>comptab</code> by having row names added (alphabetical one-letter IDs for the datasets).
</p>
<p><code>xsummary2</code> is an updated version that is used in the current vignettes in the package.
It shows negative numbers in bold (<em>p</em>-value and CLES are not shown).
<code>xsummary3</code> is a further revision that shows GRAVY and pI; it is used in the &lsquo;<span class="samp">&#8288;osmotic_bact&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;osmotic_halo&#8288;</span>&rsquo; vignettes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>comptab &lt;- lapply(c("JKMF10", "WDO+15_C.N"), function(dataset) {
  pdat &lt;- pdat_colorectal(dataset)
  get_comptab(pdat, oldstyle = TRUE)
})
xsummary(comptab)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
