<!DOCTYPE html><html><head><title>Help for package canprot</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {canprot}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#add_cld'><p>Compact letter display</p></a></li>
<li><a href='#add_hull'><p>Add convex hull</p></a></li>
<li><a href='#calc_metrics'><p>Calculate one or more chemical metrics</p></a></li>
<li><a href='#canprot-package'><p>Chemical analysis of proteins</p></a></li>
<li><a href='#CLES'><p>Common language effect size</p></a></li>
<li><a href='#human'><p>Amino acid compositions of human proteins</p></a></li>
<li><a href='#human_aa'><p>Get amino acid compositions of human proteins</p></a></li>
<li><a href='#metrics'><p>Calculate chemical metrics for proteins</p></a></li>
<li><a href='#read_fasta'><p>Functions for reading FASTA files</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Date:</td>
<td>2024-03-28</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.0</td>
</tr>
<tr>
<td>Title:</td>
<td>Chemical Analysis of Proteins</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jeffrey Dick &lt;j3ffdick@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stringi, multcompView</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, tinytest, CHNOSZ (&ge; 1.3.4)</td>
</tr>
<tr>
<td>Description:</td>
<td>Chemical analysis of proteins based on their amino acid
  compositions. Amino acid compositions can be read from FASTA files and used to
  calculate chemical metrics including carbon oxidation state and stoichiometric
  hydration state, as described in Dick et al. (2020) &lt;<a href="https://doi.org/10.5194%2Fbg-17-6145-2020">doi:10.5194/bg-17-6145-2020</a>&gt;. 
  Other properties that can be calculated include protein length, grand average of
  hydropathy (GRAVY), isoelectric point (pI), molecular weight (MW), standard
  molal volume (V0), and metabolic costs (Akashi and Gojobori, 2002
  &lt;<a href="https://doi.org/10.1073%2Fpnas.062526999">doi:10.1073/pnas.062526999</a>&gt;; Wagner, 2005 &lt;<a href="https://doi.org/10.1093%2Fmolbev%2Fmsi126">doi:10.1093/molbev/msi126</a>&gt;;
  Zhang et al., 2018 &lt;<a href="https://doi.org/10.1038%2Fs41467-018-06461-1">doi:10.1038/s41467-018-06461-1</a>&gt;). A database of amino acid
  compositions of human proteins derived from UniProt is provided.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>BuildResaveData:</td>
<td>no</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/jedick/canprot">https://github.com/jedick/canprot</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-28 06:11:03 UTC; jedick</td>
</tr>
<tr>
<td>Author:</td>
<td>Jeffrey Dick <a href="https://orcid.org/0000-0002-0687-5890"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-28 16:00:13 UTC</td>
</tr>
</table>
<hr>
<h2 id='add_cld'>Compact letter display</h2><span id='topic+add_cld'></span>

<h3>Description</h3>

<p>Adds compact letter display (significant difference letters) to a boxplot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  add_cld(datlist, bp, dx = NULL, dy = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_cld_+3A_datlist">datlist</code></td>
<td>
<p>list, list of data values for different groups</p>
</td></tr>
<tr><td><code id="add_cld_+3A_bp">bp</code></td>
<td>
<p>list, output of <code><a href="graphics.html#topic+boxplot">boxplot</a></code></p>
</td></tr>
<tr><td><code id="add_cld_+3A_dx">dx</code></td>
<td>
<p>numeric, offset for letters in the x direction</p>
</td></tr>
<tr><td><code id="add_cld_+3A_dy">dy</code></td>
<td>
<p>numeric, offset for letters in the y direction</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function adds a compact letter display (cld) to an existing boxplot.
It calculates a one-way ANOVA with <code><a href="stats.html#topic+aov">aov</a></code> followed by Tukey's Honest Significant Differences with <code><a href="stats.html#topic+TukeyHSD">TukeyHSD</a></code>, then obtains the cld with <code><a href="multcompView.html#topic+multcompLetters4">multcompLetters4</a></code>.
The letters are added to the plot at the upper right sides of the bars.
Default values for <code>dx</code> and <code>dy</code> are computed from the current plot dimensions; these values can be adjusted if needed.
</p>


<h3>Value</h3>

<p>Invisibly returns a list with <code>dx</code>, <code>dy</code>, and <code>letters</code> (letters used for the cld, in the same order as the groups in <code>datlist</code>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Are there significant differences of nH2O among human proteins with different Zc?
aa &lt;- get("human.aa", canprot)
# Remove extremely short sequences
aa &lt;- aa[!plength(aa) &lt; 20, ]
Zc &lt;- Zc(aa)
ilo &lt;- Zc &lt; -0.15
ihi &lt;- Zc &gt; -0.10
imid &lt;- !ilo &amp; !ihi
nH2O &lt;- nH2O(aa)
nH2Olist &lt;- list(lo.Zc = nH2O[ilo], mid.Zc = nH2O[imid], hi.Zc = nH2O[ihi])
bp &lt;- boxplot(nH2Olist, ylab = cplab$nH2O)
add_cld(nH2Olist, bp)
# Yes, higher Zc is associated with lower nH2O
</code></pre>

<hr>
<h2 id='add_hull'>Add convex hull</h2><span id='topic+add_hull'></span>

<h3>Description</h3>

<p>Adds a convex hull around the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  add_hull(x, y = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_hull_+3A_x">x</code></td>
<td>
<p>x values</p>
</td></tr>
<tr><td><code id="add_hull_+3A_y">y</code></td>
<td>
<p>y values</p>
</td></tr>
<tr><td><code id="add_hull_+3A_...">...</code></td>
<td>
<p>arguments for <code><a href="graphics.html#topic+polygon">polygon</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>add_hull</code> draws a convex hull around the points given in <code>x</code>, <code>y</code>.
This function is a wrapper for <code><a href="grDevices.html#topic+chull">chull</a></code> and <code><a href="graphics.html#topic+polygon">polygon</a></code>.
</p>


<h3>Value</h3>

<p>Invisibly returns the result from <code>chull</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat &lt;- iris[, 1:2]
plot(dat)
add_hull(dat)
</code></pre>

<hr>
<h2 id='calc_metrics'>Calculate one or more chemical metrics</h2><span id='topic+calc_metrics'></span>

<h3>Description</h3>

<p>Calculates selected chemical metrics from amino acid composition(s) of proteins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  calc_metrics(AAcomp, metrics = c("Zc", "nO2", "nH2O"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_metrics_+3A_aacomp">AAcomp</code></td>
<td>
<p>data frame with amino acid compositions</p>
</td></tr>
<tr><td><code id="calc_metrics_+3A_metrics">metrics</code></td>
<td>
<p>character, chemical metrics to calculate</p>
</td></tr>
<tr><td><code id="calc_metrics_+3A_...">...</code></td>
<td>
<p>additional arguments passed to individual functions</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is a wrapper function for the functions described at <code><a href="#topic+metrics">metrics</a></code>.
This wrapper implements case-insensitve matching to the function names (e.g., <code>zc</code> matches <code><a href="#topic+Zc">Zc</a></code>).
The following additional shortcuts are defined (e.g., <code>length</code> can be used in place of the actual function name, <code><a href="#topic+plength">plength</a></code>):
</p>

<table>
<tr>
 <td style="text-align: left;">
  <code>length</code> </td><td style="text-align: left;"> <code><a href="#topic+plength">plength</a></code> </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>H/C</code>, <code>H_C</code> </td><td style="text-align: left;"> <code><a href="#topic+HC">HC</a></code> </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>N/C</code>, <code>N_C</code> </td><td style="text-align: left;"> <code><a href="#topic+NC">NC</a></code> </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>O/C</code>, <code>O_C</code> </td><td style="text-align: left;"> <code><a href="#topic+OC">OC</a></code> </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>S/C</code>, <code>S_C</code> </td><td style="text-align: left;"> <code><a href="#topic+SC">SC</a></code> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Value</h3>

<p>A data frame with the same number of rows as <code>AAcomp</code> and one column of numeric values for each of the <code>metrics</code>.
An error is produced if any of the <code>metrics</code> is not available for calculation.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metrics">metrics</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Define the acid composition of alanylglycine
AG &lt;- data.frame(Ala = 1, Gly = 1)
# Calculate default metrics (Zc, nO2, nH2O)
calc_metrics(AG)
# Calculate selected metrics
calc_metrics(AG, c("H/C", "O/C", "Length"))
</code></pre>

<hr>
<h2 id='canprot-package'>Chemical analysis of proteins</h2><span id='topic+canprot-package'></span>

<h3>Description</h3>

<p>Chemical metrics of proteins are important for understanding biomolecular adaptation to environments.
<span class="pkg">canprot</span> computes chemical metrics of proteins from their amino acid compositions.
</p>


<h3>Details</h3>


<ul>
<li> <p><code><a href="#topic+read_fasta">read_fasta</a></code> &ndash; start here to read amino acid compositions from FASTA files.
</p>
</li>
<li> <p><code><a href="#topic+metrics">metrics</a></code> &ndash; use these functions to calculate chemical metrics from amino acid compositions.
</p>
</li></ul>

<p>Chemical metrics include carbon oxidation state (<I>Z</I><sub>C</sub>) stoichiometric oxidation and hydration state (<i>n</i><sub>O<sub>2</sub></sub> and <i>n</i><sub>H<sub>2</sub>O</sub>) as described in Dick et al. (2020).
Other variables that can be calculated are protein length, average molecular weight of amino acid residues, grand average of hydropathy (GRAVY), and isoelectric point (pI).
</p>
<p>Differential expression datasets that were in the package up to version 1.1.2, mainly for the paper by Dick (2021), have been moved to <a href="https://github.com/jedick/JMDplots">JMDplots</a>.
</p>
<p>Three demos are available:
</p>

<ul>
<li> <p><code>demo("thermophiles")</code>: 
Specific entropy vs <I>Z</I><sub>C</sub> for methanogen genomes and Nitrososphaeria MAGs.
Amino acid compositions and optimal growth temperatures for methanogens were obtained from Dick et al. (2023).
Accession numbers and data for Nitrososphaeria MAGs were taken from Tables S1 and S3 Luo et al. (2024) and are saved in &lsquo;<span class="file">extdata/aa/nitrososphaeria_MAGs.csv</span>&rsquo;.
Sequences of MAGs were downloaded from NCBI and processed to obtain amino acid compositions using the script in &lsquo;<span class="file">extdata/aa/prepare.R</span>&rsquo;
</p>
</li>
<li> <p><code>demo("locations")</code>:
<I>Z</I><sub>C</sub> and pI for human proteins with different subcellular locations.
The localization data was taken from Thul et al. (2017) and is stored in &lsquo;<span class="file">extdata/protein/TAW+17_Table_S6_Validated.csv</span>&rsquo;.
</p>
</li>
<li> <p><code>demo("redoxins")</code>: 
<I>Z</I><sub>C</sub> vs the midpoint reduction potentials of ferredoxin and thioredoxin in spinach and glutaredoxin and thioredoxin in <em>E. coli</em>.
The sequences were obtained from UniProt and is stored in &lsquo;<span class="file">extdata/fasta/redoxin.fasta</span>&rsquo;.
The reduction potential data was taken from Åslund et al. (1997) and Hirasawa et al. (1999) and is stored in &lsquo;<span class="file">extdata/fasta/redoxin.csv</span>&rsquo;.
This file also lists UniProt IDs and start and stop positions for the protein chains excluding initiator methionines and signal peptides.
</p>
</li></ul>



<h3>References</h3>

<p>Åslund F, Berndt KD, Holmgren A. 1997. Redox potentials of glutaredoxins and other thiol-disulfide oxidoreductases of the thioredoxin superfamily determined by direct protein-protein redox equilibria. <em>Journal of Biological Chemistry</em> <b>272</b>(49): 30780–30786. <a href="https://doi.org/10.1074/jbc.272.49.30780">doi:10.1074/jbc.272.49.30780</a>
</p>
<p>Dick JM. 2021. Water as a reactant in the differential expression of proteins in cancer. <em>Computational and Systems Oncology</em> <b>1</b>(1): e1007. <a href="https://doi.org/10.1002/cso2.1007">doi:10.1002/cso2.1007</a>
</p>
<p>Dick JM, Yu M, Tan J. 2020. Uncovering chemical signatures of salinity gradients through compositional analysis of protein sequences. <em>Biogeosciences</em> <b>17</b>(23): 6145&ndash;6162. <a href="https://doi.org/10.5194/bg-17-6145-2020">doi:10.5194/bg-17-6145-2020</a>
</p>
<p>Dick JM, Boyer GM, Canovas PA, Shock EL. 2023. Using thermodynamics to obtain geochemical information from genomes. <em>Geobiology</em> <b>21</b>(2): 262–273. <a href="https://doi.org/10.1111/gbi.12532">doi:10.1111/gbi.12532</a>
</p>
<p>Hirasawa M, Schürmann P, Jacquot J-P, Manieri W, Jacquot P, Keryer E, Hartman FC, Knaff DB. 1999. Oxidation-reduction properties of chloroplast thioredoxins, ferredoxin:thioredoxin reductase, and thioredoxin <em>f</em>-regulated enzymes. <em>Biochemistry</em> <b>38</b>(16): 5200–5205. <a href="https://doi.org/10.1021/bi982783v">doi:10.1021/bi982783v</a>
</p>
<p>Luo Z-H, Li Q, Xie Y-G, Lv A-P, Qi Y-L, Li M-M, Qu Y-N, Liu Z-T, Li Y-X, Rao Y-Z, et al. 2024 Jan. Temperature, pH, and oxygen availability contributed to the functional differentiation of ancient <em>Nitrososphaeria</em>. <em>The ISME Journal</em> <b>18</b>(1): wrad031. <a href="https://doi.org/10.1093/ismejo/wrad031">doi:10.1093/ismejo/wrad031</a>
</p>
<p>Thul PJ, Åkesson L, Wiking M, Mahdessian D, Geladaki A, Blal HA, Alm T, Asplund A, Björk L, Breckels LM, et al. 2017. A subcellular map of the human proteome. <em>Science</em> <b>356</b>(6340): eaal3321. <a href="https://doi.org/10.1126/science.aal3321">doi:10.1126/science.aal3321</a>
</p>

<hr>
<h2 id='CLES'>Common language effect size</h2><span id='topic+CLES'></span>

<h3>Description</h3>

<p>Calculate the common language effect size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  CLES(x, y, distribution = "normal")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CLES_+3A_x">x</code></td>
<td>
<p>numeric, data</p>
</td></tr>
<tr><td><code id="CLES_+3A_y">y</code></td>
<td>
<p>numeric, data</p>
</td></tr>
<tr><td><code id="CLES_+3A_distribution">distribution</code></td>
<td>
<p>&lsquo;<span class="samp">&#8288;normal&#8288;</span>&rsquo; to use probabilities calculated for a normal distribution, or NA for empirical probabilities</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The common language statistic is defined for continuous data as &ldquo;the probability that a score sampled at random from one distribution will be greater than a score sampled from some other distribution&rdquo; (McGraw and Wong, 1992).
</p>
<p>Given the default value of <code>distribution</code> (&lsquo;<span class="samp">&#8288;normal&#8288;</span>&rsquo;), this function uses <code><a href="stats.html#topic+pnorm">pnorm</a></code> to calculate the probability that a random sample from the unit normal distribution is greater than the Z score (i.e. (the mean of &lsquo;<span class="samp">&#8288;y&#8288;</span>&rsquo; minus the mean of &lsquo;<span class="samp">&#8288;x&#8288;</span>&rsquo;) / square root of (variance of &lsquo;<span class="samp">&#8288;x&#8288;</span>&rsquo; plus variance of &lsquo;<span class="samp">&#8288;y&#8288;</span>&rsquo;)).
</p>
<p>If <code>distribution</code> is NA, this function calculates the empirical probability that the difference is positive, that is, the fraction of all possible pairings between elements of <code>x</code> and <code>y</code> where the difference (&lsquo;<span class="samp">&#8288;y&#8288;</span>&rsquo; value - &lsquo;<span class="samp">&#8288;x&#8288;</span>&rsquo; value) is positive.
It may not be possible to calculate the empirical probability for very large samples because of memory limits.
</p>
<p>The examples use <em>simulated data for normal distributions</em>, given the sample size, mean, and standard deviation of datasets cited by McGraw and Wong, 1992.
Therefore, the empirical probability in the examples approaches the normal curve probability.
However, the empirical probability for <em>nonnormal</em> distributions is distinct from the normal curve probability, as discussed on p. 364-365 of McGraw and Wong, 1992.
</p>


<h3>References</h3>

<p>McGraw KO, Wong SP. 1992. A common language effect size statistic. <em>Psychological Bulletin</em> <b>111</b>(2): 361&ndash;365. <a href="https://doi.org/10.1037/0033-2909.111.2.361">doi:10.1037/0033-2909.111.2.361</a>
</p>
<p>National Center for Health Statistics. 1987. Anthropometric Reference Data and Prevalence of Overweight: United States, 1976-1980. U.S. Department of Health and Human Services. Data from the National Health Survey, Series 11, No. 238.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1: Height differences between males and females
# a) Use statistics quoted by McGraw and Wong, 1992 from NCHS, 1987
# for heights in inches of 18-24 year-old males and females
# Table 14: number, mean height, and standard deviation of height of females
n1 &lt;- 1066
M1 &lt;- 64.3
SD1 &lt;- 2.8
# Table 13: number, mean height, and standard deviation of height of males
n2 &lt;- 988
M2 &lt;- 69.7
SD2 &lt;- 2.6
# b) Simulate data from a normal distribution with exact mean and SD
# use rnorm2 function from Ben Bolker's answer to
# https://stackoverflow.com/questions/18919091/generate-random-numbers-with-fixed-mean-and-sd
rnorm2 &lt;- function(n, mean, sd) { mean + sd * scale(rnorm(n)) }
set.seed(1234)
height_female &lt;- rnorm2(n1, M1, SD1)
height_male &lt;- rnorm2(n2, M2, SD2)
# c) Calculate the CLES using the normal distribution and empirical probability
CLES_normal &lt;- CLES(height_female, height_male)
CLES_empirical &lt;- CLES(height_female, height_male, distribution = NA)
# d) Test numerical equivalence of the results
# The CLES is approximately 0.92 (McGraw and Wong, 1992)
# (note: becasue we used rnorm2, this doesn't depend on the seed)
stopifnot(all.equal(CLES_normal, 0.92, tol = 0.01))
# With this seed, the difference between the normal curve probability
# and empirical probability is less than 1%
stopifnot(all.equal(CLES_normal, CLES_empirical, tol = 0.01))

# Example 1.5: Use multiple simulated datasets to show approach
# of empirical probability to normal curve probability
CLES_empirical_n &lt;- sapply(1:100, function(x) {
  height_female &lt;- rnorm2(n1, M1, SD1)
  height_male &lt;- rnorm2(n2, M2, SD2)
  CLES(height_female, height_male, distribution = NA)
})
CLES_empirical &lt;- mean(CLES_empirical_n)
# now we're even closer to the normal curve probability
stopifnot(all.equal(CLES_normal, CLES_empirical, tol = 0.0001))

# Example 2: Multiple datasets in Table 2 of McGraw and Wong, 1992
# Sample statistics for females
n1 &lt;- c(638, 672, 3139, 420740, 19274, 104263, 207, 394, 1066, 982, 108, 108)
M1 &lt;- c(103, 15, 103, 18.9, 30, 16.1, 6.9, 13.3, 64.3, 134, 45, 94)
SD1 &lt;- sqrt(c(908, 74, 219, 27, 110, 59, 15, 164, 6.8, 688, 310, 1971))
# Sample statistics for males
n2 &lt;- c(354, 359, 3028, 356704, 21768, 133882, 199, 469, 988, 988, 443, 443)
M2 &lt;- c(112, 23, 100, 17.9, 33, 18.6, 9.3, 21.8, 69.7, 163, 86, 212)
SD2 &lt;- sqrt(c(1096, 96, 202, 29, 110, 61, 15, 133, 7.8, 784, 818, 5852))
# A function to calculate the effect size using simulated data
CLESfun &lt;- function(n1, M1, SD1, n2, M2, SD2, distribution) {
  rnorm2 &lt;- function(n, mean, sd) { mean + sd * scale(rnorm(n)) }
  set.seed(1234)
  x &lt;- rnorm2(n1, M1, SD1)
  y &lt;- rnorm2(n2, M2, SD2)
  CLES(x, y, distribution)
}
# Calculate 100 * CL for the normal curve probabilities
CLnorm &lt;- sapply(1:12, function(i) {
  CL &lt;- CLESfun(n1[i], M1[i], SD1[i], n2[i], M2[i], SD2[i], "normal")
  round(100 * CL)
})
# Calculate 100 * CL for empirical probabilities
CLemp &lt;- sapply(1:12, function(i) {
  # skip very large samples: not enough memory
  if(n1[i] &gt; 5000 | n2[i] &gt; 5000) NA else {
    CL &lt;- CLESfun(n1[i], M1[i], SD1[i], n2[i], M2[i], SD2[i], NA)
    round(100 * CL)
  }
})
# The difference between the empirical and normal curve
# probabilities is not more than 1 percent
stopifnot(max(abs(CLemp - CLnorm), na.rm = TRUE) &lt;= 1)
# TODO: Why are some of the calculated values different from
# Table 2 of McGraw and Wong, 1992?
CLref &lt;- c(54, 74, 44, 45, 56, 63, 67, 65, 92, 78, 89, 91)
# Differences range from -4 to 4
range(CLnorm - CLref) 
#stopifnot(max(abs(CLnorm - CLref)) == 0)
</code></pre>

<hr>
<h2 id='human'>Amino acid compositions of human proteins</h2><span id='topic+canprot'></span><span id='topic+human'></span><span id='topic+human.aa'></span><span id='topic+human.base'></span><span id='topic+human.additional'></span><span id='topic+human.extra'></span>

<h3>Description</h3>

<p>Amino acid compositions of human proteins derived from UniProt.
</p>


<h3>Format</h3>

<p><code>human.aa</code> is a data frame with 25 columns in the format used for amino acid compositions in <span class="pkg">CHNOSZ</span> (see <code><a href="CHNOSZ.html#topic+thermo">thermo</a></code>):
</p>

<table>
<tr>
 <td style="text-align: left;">
  <code>protein</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Identification of protein</td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>organism</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Identification of organism</td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>ref</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Reference key for source of sequence data</td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>abbrv</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Abbreviation or other ID for protein (e.g. gene name)</td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>chains</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Number of polypeptide chains in the protein</td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>Ala</code>...<code>Tyr</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Number of each amino acid in the protein
</td>
</tr>

</table>

<p>The <code>protein</code> column contains UniProt IDs in the format <code>database|accession-isoform</code>, where <code>database</code> is most often &lsquo;<span class="samp">&#8288;sp&#8288;</span>&rsquo; (Swiss-Prot) or &lsquo;<span class="samp">&#8288;tr&#8288;</span>&rsquo; (TrEMBL), and <code>isoform</code> is an optional suffix indicating the isoform of the protein (particularly in the <code>human.additional</code> file).
</p>


<h3>Details</h3>

<p>The amino acid compositions of human proteins are stored in three files under <code>extdata/protein</code>.
</p>

<ul>
<li> <p><code>human.base.rds</code> contains amino acid compositions of canonical isoforms of manually reviewed proteins in the <a href="https://www.uniprot.org/">UniProt</a> reference human proteome (computed from sequences in <code>UP000005640_9606.fasta.gz</code>, dated 2016-04-03).
</p>
</li>
<li> <p><code>human.additional.rds</code> contains amino acid compositions of additional proteins (<code>UP000005640_9606_additional.fasta.gz</code>) including isoforms and unreviewed sequences. In version 0.1.5, this file was trimmed to include only those proteins that are used in any of the datasets in the package.
</p>
</li>
<li> <p><code>human.extra.csv</code> contains amino acid compositions of other (&ldquo;extra&rdquo;) proteins used in a dataset but not listed in one of the files above. These proteins may include obsolete, unreviewed, or newer additions to the UniProt database. Most, but not all, sequences here are HUMAN (see the <code>organism</code> column and the <code>ref</code> column for the reference keys).
</p>
</li></ul>

<p>On loading the package, the individual data files are read and combined, and the result is assigned to the <code>human.aa</code> object in the <code>canprot</code> environment.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+human_aa">human_aa</a></code> gets amino acid compositions for human proteins specified by their UniProt IDs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># The number of proteins
nrow(get("human.aa", canprot))
</code></pre>

<hr>
<h2 id='human_aa'>Get amino acid compositions of human proteins</h2><span id='topic+human_aa'></span>

<h3>Description</h3>

<p>Get amino acid compositions of human proteins from their UniProt IDs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  human_aa(uniprot = NULL, aa_file = NULL,
    stop_if_missing = FALSE, warn_if_duplicated = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="human_aa_+3A_uniprot">uniprot</code></td>
<td>
<p>character, UniProt IDs of proteins</p>
</td></tr>
<tr><td><code id="human_aa_+3A_aa_file">aa_file</code></td>
<td>
<p>character, file name</p>
</td></tr>
<tr><td><code id="human_aa_+3A_stop_if_missing">stop_if_missing</code></td>
<td>
<p>logical, stop with an error if there are UniProt IDs that can't be found?</p>
</td></tr>
<tr><td><code id="human_aa_+3A_warn_if_duplicated">warn_if_duplicated</code></td>
<td>
<p>logical, emit a warning if duplicate UniProt IDs are detected?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function retrieves the amino acid compositions of one or more proteins specified by <code>uniprot</code>.
</p>
<p>This function depends on the amino acid compositions of human proteins, which are stored in the <code><a href="#topic+canprot">canprot</a></code> environment when the package is attached.
If <code>aa_file</code> is specified, additional amino acid compositions are read from this file.
This file should be in the same format as <code><a href="#topic+human.extra">human.extra</a>.csv</code> in the installation directory of the package.
</p>


<h3>Value</h3>

<p>The function returns a data frame with amino acid compositions of proteins.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>human_aa("P24298")
</code></pre>

<hr>
<h2 id='metrics'>Calculate chemical metrics for proteins</h2><span id='topic+metrics'></span><span id='topic+Zc'></span><span id='topic+nO2'></span><span id='topic+nH2O'></span><span id='topic+GRAVY'></span><span id='topic+pI'></span><span id='topic+MW'></span><span id='topic+pMW'></span><span id='topic+V0'></span><span id='topic+pV0'></span><span id='topic+V0g'></span><span id='topic+Density'></span><span id='topic+S0'></span><span id='topic+pS0'></span><span id='topic+S0g'></span><span id='topic+SV'></span><span id='topic+Zcg'></span><span id='topic+nO2g'></span><span id='topic+nH2Og'></span><span id='topic+HC'></span><span id='topic+NC'></span><span id='topic+OC'></span><span id='topic+SC'></span><span id='topic+nC'></span><span id='topic+pnC'></span><span id='topic+plength'></span><span id='topic+Cost'></span><span id='topic+RespiratoryCost'></span><span id='topic+FermentativeCost'></span><span id='topic+B20Cost'></span><span id='topic+Y20Cost'></span><span id='topic+H11Cost'></span><span id='topic+cplab'></span>

<h3>Description</h3>

<p>Calculate chemical metrics for proteins from their amino acid compositions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  Zc(AAcomp, ...)
  nO2(AAcomp, basis = "QEC", ...)
  nH2O(AAcomp, basis = "QEC", terminal_H2O = 0)
  GRAVY(AAcomp, ...)
  pI(AAcomp, terminal_H2O = 1, ...)
  MW(AAcomp, terminal_H2O = 0, ...)
  pMW(AAcomp, terminal_H2O = 1, ...)
  V0(AAcomp, terminal_H2O = 0, ...)
  pV0(AAcomp, terminal_H2O = 1, ...)
  V0g(AAcomp, ...)
  Density(AAcomp, ...)
  S0(AAcomp, terminal_H2O = 0, ...)
  pS0(AAcomp, terminal_H2O = 1, ...)
  S0g(AAcomp, ...)
  SV(AAcomp, ...)
  Zcg(AAcomp, ...)
  nH2Og(AAcomp, ...)
  nO2g(AAcomp, ...)
  HC(AAcomp, ...)
  NC(AAcomp, ...)
  OC(AAcomp, ...)
  SC(AAcomp, ...)
  nC(AAcomp, ...)
  pnC(AAcomp, ...)
  plength(AAcomp, ...)
  Cost(AAcomp, ...)
  RespiratoryCost(AAcomp, ...)
  FermentativeCost(AAcomp, ...)
  B20Cost(AAcomp, ...)
  Y20Cost(AAcomp, ...)
  H11Cost(AAcomp, ...)
  cplab
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metrics_+3A_aacomp">AAcomp</code></td>
<td>
<p>data frame, amino acid compositions</p>
</td></tr>
<tr><td><code id="metrics_+3A_...">...</code></td>
<td>
<p>ignored additional arguments</p>
</td></tr>
<tr><td><code id="metrics_+3A_basis">basis</code></td>
<td>
<p>character, set of basis species</p>
</td></tr>
<tr><td><code id="metrics_+3A_terminal_h2o">terminal_H2O</code></td>
<td>
<p>numeric, number of pairs of terminal groups</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Columns in <code>AAcomp</code> should be named with the three-letter abbreviations for the amino acids.
Case-insensitive matching matching of the abbreviations is used; e.g., &lsquo;<span class="samp">&#8288;Ala&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;ALA&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;ala&#8288;</span>&rsquo; all refer to alanine.
</p>
<p>Metrics are normalized per amino acid residue except for <code>Zc</code>, <code>pI</code>, <code>Density</code>, <code>plength</code>, and other functions starting with <code>p</code> (for protein).
The contribution of protein terminal groups (-H and -OH) to residue-normalized metrics is turned off by default.
Set <code>terminal_H2O</code> to 1 (or to the number of polypeptide chains, if greater than one) to include their contribution.
</p>
<p>The metrics are described below:
</p>

<dl>
<dt><code>Zc</code></dt><dd><p>Average oxidation state of carbon (<I>Z</I><sub>C</sub>) (Dick, 2014).
This metric is independent of the choice of basis species.
Note that <I>Z</I><sub>C</sub> is normalized by number of carbon atoms, not by number of residues.
</p>
</dd>
<dt><code>nO2</code></dt><dd><p>Stoichiometric oxidation state (<i>n</i><sub>O<sub>2</sub></sub> per residue).
The available <code>basis</code> species are:
</p>

<ul>
<li><p>&lsquo;<span class="samp">&#8288;QEC&#8288;</span>&rsquo; - glutamine, glutamic acid, cysteine, H<sub>2</sub>O, O<sub>2</sub> (Dick et al., 2020)
</p>
</li>
<li><p>&lsquo;<span class="samp">&#8288;QCa&#8288;</span>&rsquo; - glutamine, cysteine, acetic acid, H<sub>2</sub>O, O<sub>2</sub>
</p>
</li></ul>

</dd>
<dt><code>nH2O</code></dt><dd><p>Stoichiometric hydration state (<i>n</i><sub>H<sub>2</sub>O</sub> per residue).
The basis species also affect this calculation.
</p>
</dd>
<dt><code>GRAVY</code></dt><dd><p>Grand average of hydropathy.
Values of the hydropathy index for individual amino acids are from Kyte and Doolittle (1982).
</p>
</dd>
<dt><code>pI</code></dt><dd><p>Isoelectric point.
The net charge for each ionizable group was pre-calculated from pH 0 to 14 at intervals of 0.01.
The isoelectric point is found as the pH where the sum of charges of all groups in the protein is closest to zero.
The p<i>K</i> values for the terminal groups and sidechains are taken from Bjellqvist et al. (1993) and Bjellqvist et al. (1994); note that the calculation does not implement position-specific adjustments described in the latter paper.
The number of N- and C-terminal groups is taken from <code>terminal_H2O</code>.
</p>
</dd>
<dt><code>MW</code></dt><dd><p>Molecular weight.</p>
</dd>
<dt><code>pMW</code></dt><dd><p>Molecular weight per protein.</p>
</dd>
<dt><code>V0</code></dt><dd><p>Standard molal volume.
The values are derived from group contributions of amino acid sidechains and protein backbones (Dick et al., 2006).
</p>
</dd>
<dt><code>pV0</code></dt><dd><p>Standard molal volume per protein.</p>
</dd>
<dt><code>V0g</code></dt><dd><p>Specific volume (reciprocal density).</p>
</dd>
<dt><code>Density</code></dt><dd><p>Density (MW / V0).</p>
</dd>
<dt><code>S0</code></dt><dd><p>Standard molal entropy.
The values are derived from group contributions of amino acid sidechains and protein backbones (Dick et al., 2006).
</p>
</dd>
<dt><code>pS0</code></dt><dd><p>Standard molal entropy per protein.</p>
</dd>
<dt><code>S0g</code></dt><dd><p>Specific entropy.</p>
</dd>
<dt><code>SV</code></dt><dd><p>Entropy density.</p>
</dd>
<dt><code>Zcg</code></dt><dd><p>Carbon oxidation state per gram.</p>
</dd>
<dt><code>nO2g</code></dt><dd><p>Stoichiometric oxidation state per gram.</p>
</dd>
<dt><code>nH2Og</code></dt><dd><p>Stoichiometric hydration state per gram.</p>
</dd>
<dt><code>HC</code></dt><dd><p>H/C ratio (not counting terminal -H and -OH groups).</p>
</dd>
<dt><code>NC</code></dt><dd><p>N/C ratio.</p>
</dd>
<dt><code>OC</code></dt><dd><p>O/C ratio (not counting terminal -H and -OH groups).</p>
</dd>
<dt><code>SC</code></dt><dd><p>S/C ratio.</p>
</dd>
<dt><code>nC</code></dt><dd><p>Number of carbon atoms per residue.</p>
</dd>
<dt><code>pnC</code></dt><dd><p>Number of carbon atoms per protein.</p>
</dd>
<dt><code>plength</code></dt><dd><p>Protein length (number of amino acid residues).</p>
</dd>
<dt><code>Cost</code></dt><dd><p>Metabolic cost (Akashi and Gojobori, 2002).</p>
</dd>
<dt><code>RespiratoryCost</code></dt><dd><p>Respiratory cost (Wagner, 2005).</p>
</dd>
<dt><code>FermentativeCost</code></dt><dd><p>Fermentative cost (Wagner, 2005).</p>
</dd>
<dt><code>B20Cost</code></dt><dd><p>Biosynthetic cost in bacteria (Zhang et al., 2018).</p>
</dd>
<dt><code>Y20Cost</code></dt><dd><p>Biosynthetic cost in yeast (Zhang et al., 2018).</p>
</dd>
<dt><code>H11Cost</code></dt><dd><p>Biosynthetic cost in humans (Zhang et al., 2018).</p>
</dd>
</dl>

<p><code>...</code> is provided to permit <code><a href="base.html#topic+get">get</a></code> or <code><a href="base.html#topic+do.call">do.call</a></code> constructions with the same arguments for all metrics.
For instance, a <code>terminal_H2O</code> argument can be suppled to either <code>Zc</code> or <code>nH2O</code>, but it only has an effect on the latter.
</p>
<p><code>cplab</code> is a list of formatted labels for each of the chemical metrics listed here.
A check in the code ensures that the names of the functions for calculating metrics and the names for labels listed <code>cplab</code> are identical.
</p>


<h3>References</h3>

<p>Akashi H, Gojobori T. 2002. Metabolic efficiency and amino acid composition in the proteomes of <em>Escherichia coli</em> and <em>Bacillus subtilis</em>. <em>Proceedings of the National Academy of Sciences</em> <b>99</b>(6): 3695–3700. <a href="https://doi.org/10.1073/pnas.062526999">doi:10.1073/pnas.062526999</a>
</p>
<p>Bjellqvist B, Hughes GJ, Pasquali C, Paquet N, Ravier F, Sanchez J-C, Frutiger S, Hochstrasser D. 1993. The focusing positions of polypeptides in immobilized pH gradients can be predicted from their amino acid sequences. <em>Electrophoresis</em> <b>14</b>: 1023&ndash;1031. <a href="https://doi.org/10.1002/elps.11501401163">doi:10.1002/elps.11501401163</a>
</p>
<p>Bjellqvist B, Basse B, Olsen E, Celis JE. 1994. Reference points for comparisons of two-dimensional maps of proteins from different human cell types defined in a pH scale where isoelectric points correlate with polypeptide compositions. <em>Electrophoresis</em> <b>15</b>: 529&ndash;539. <a href="https://doi.org/10.1002/elps.1150150171">doi:10.1002/elps.1150150171</a>
</p>
<p>Dick JM, LaRowe DE, Helgeson HC. 2006. Temperature, pressure, and electrochemical constraints on protein speciation: Group additivity calculation of the standard molal thermodynamic properties of ionized unfolded proteins. <em>Biogeosciences</em> <b>3</b>(3): 311&ndash;336. <a href="https://doi.org/10.5194/bg-3-311-2006">doi:10.5194/bg-3-311-2006</a>
</p>
<p>Dick JM. 2014. Average oxidation state of carbon in proteins. <em>J. R. Soc. Interface</em> <b>11</b>: 20131095. <a href="https://doi.org/10.1098/rsif.2013.1095">doi:10.1098/rsif.2013.1095</a>
</p>
<p>Dick JM, Yu M, Tan J. 2020. Uncovering chemical signatures of salinity gradients through compositional analysis of protein sequences. <em>Biogeosciences</em> <b>17</b>: 6145&ndash;6162. <a href="https://doi.org/10.5194/bg-17-6145-2020">doi:10.5194/bg-17-6145-2020</a>
</p>
<p>Kyte J, Doolittle RF. 1982. A simple method for displaying the hydropathic character of a protein. <em>J. Mol. Biol.</em> <b>157</b>: 105&ndash;132. <a href="https://doi.org/10.1016/0022-2836%2882%2990515-0">doi:10.1016/0022-2836(82)90515-0</a>
</p>
<p>Wagner A. 2005. Energy constraints on the evolution of gene expression. <em>Molecular Biology and Evolution</em> <b>22</b>(6): 1365–1374. <a href="https://doi.org/10.1093/molbev/msi126">doi:10.1093/molbev/msi126</a>
</p>
<p>Zhang H, Wang Y, Li J, Chen H, He X, Zhang H, Liang H, Lu J. 2018. Biosynthetic energy cost for amino acids decreases in cancer evolution. <em>Nature Communications</em> <b>9</b>(1): 4124. <a href="https://doi.org/10.1038/s41467-018-06461-1">doi:10.1038/s41467-018-06461-1</a>
</p>


<h3>See Also</h3>

<p>For calculation of <I>Z</I><sub>C</sub> from an elemental formula (instead of amino acid composition), see the <code><a href="CHNOSZ.html#topic+ZC">ZC</a></code> function in <span class="pkg">CHNOSZ</span>.
<code><a href="#topic+calc_metrics">calc_metrics</a></code> is a wrapper to calculate one or more metrics specified in an argument.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Amino acid composition of a tripeptide (Gly-Ala-Gly)
aa &lt;- data.frame(Ala = 1, Gly = 2)
# Calculate Zc, nH2O, and length
Zc(aa)
nH2O(aa)
plength(aa)

# Make a plot with formatted labels
plot(Zc(aa), nH2O(aa), xlab = cplab$Zc, ylab = cplab$nH2O)
</code></pre>

<hr>
<h2 id='read_fasta'>Functions for reading FASTA files</h2><span id='topic+read_fasta'></span><span id='topic+count_aa'></span><span id='topic+sum_aa'></span>

<h3>Description</h3>

<p>Read protein amino acid composition or sequences from a file and count numbers of amino acids in given sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  read_fasta(file, iseq = NULL, type = "count", lines = NULL, 
    ihead = NULL, start = NULL, stop = NULL, molecule = "protein", id = NULL)
  count_aa(sequence, start = NULL, stop = NULL, molecule = "protein")
  sum_aa(AAcomp, abundance = 1, average = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_fasta_+3A_file">file</code></td>
<td>
<p>character, path to FASTA file</p>
</td></tr>
<tr><td><code id="read_fasta_+3A_iseq">iseq</code></td>
<td>
<p>numeric, which sequences to read from the file</p>
</td></tr>
<tr><td><code id="read_fasta_+3A_type">type</code></td>
<td>
<p>character, type of return value (&lsquo;<span class="samp">&#8288;count&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;sequence&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;lines&#8288;</span>&rsquo;, or &lsquo;<span class="samp">&#8288;headers&#8288;</span>&rsquo;)</p>
</td></tr>
<tr><td><code id="read_fasta_+3A_lines">lines</code></td>
<td>
<p>list of character, supply the lines here instead of reading them from file</p>
</td></tr>
<tr><td><code id="read_fasta_+3A_ihead">ihead</code></td>
<td>
<p>numeric, which lines are headers</p>
</td></tr>
<tr><td><code id="read_fasta_+3A_start">start</code></td>
<td>
<p>numeric, position in sequence to start counting</p>
</td></tr>
<tr><td><code id="read_fasta_+3A_stop">stop</code></td>
<td>
<p>numeric, position in sequence to stop counting</p>
</td></tr>
<tr><td><code id="read_fasta_+3A_molecule">molecule</code></td>
<td>
<p>character, type of molecule (&lsquo;<span class="samp">&#8288;protein&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;DNA&#8288;</span>&rsquo;, or &lsquo;<span class="samp">&#8288;RNA&#8288;</span>&rsquo;)</p>
</td></tr>
<tr><td><code id="read_fasta_+3A_id">id</code></td>
<td>
<p>character, value to be used for <code>protein</code> in output table</p>
</td></tr>
<tr><td><code id="read_fasta_+3A_sequence">sequence</code></td>
<td>
<p>character, one or more sequences</p>
</td></tr>
<tr><td><code id="read_fasta_+3A_aacomp">AAcomp</code></td>
<td>
<p>data frame, amino acid composition(s) of proteins</p>
</td></tr>
<tr><td><code id="read_fasta_+3A_abundance">abundance</code></td>
<td>
<p>numeric, abundances of proteins</p>
</td></tr>
<tr><td><code id="read_fasta_+3A_average">average</code></td>
<td>
<p>logical, return the weighted average of amino acid counts?</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>read_fasta</code> is used to retrieve entries from a FASTA file.
Use <code>iseq</code> to select the sequences to read (the default is all sequences).
</p>
<p>The function returns various data formats depending on the value of <code>type</code>:
</p>

<dl>
<dt>&lsquo;<span class="samp">&#8288;count&#8288;</span>&rsquo;</dt><dd><p>data frame of amino acid counts</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;sequence&#8288;</span>&rsquo;</dt><dd><p>list of sequences</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;lines&#8288;</span>&rsquo;</dt><dd><p>list of lines from the FASTA file (including headers)</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;headers&#8288;</span>&rsquo;</dt><dd><p>list of header lines from the FASTA file</p>
</dd>
</dl>

<p>When <code>type</code> is &lsquo;<span class="samp">&#8288;count&#8288;</span>&rsquo;, the header lines of the file are parsed to obtain protein names that are put into the <code>protein</code> column in the result.
Furthermore, if a UniProt FASTA header is detected (using the regular expression <code>"\|......\|.*_"</code>), the information there (accession, name, organism) is split into the <code>protein</code>, <code>abbrv</code>, and <code>organism</code> columns of the resulting data frame.
this behavior (which may take a while for large files) can be suppressed by supplying protein names in <code>id</code>.
</p>
<p>To speed up processing, if the line numbers of the header lines were previously determined, they can be supplied in <code>ihead</code>. 
Optionally, the lines of a previously read file may be supplied in <code>lines</code> (in this case no file is needed so <code>file</code> should be set to &quot;&quot;).
</p>
<p><code>count_aa</code> is the underlying function that counts the numbers of each amino acid or nucleic-acid base in one or more sequences.
The matching of letters is case-insensitive.
A message is generated if any character in <code>sequence</code>, excluding spaces, is not one of the single-letter amino acid or nucleobase abbreviations.
<code>start</code> and/or <code>stop</code> can be provided to process a fragment of the sequence.
If only one of <code>start</code> or <code>stop</code> is present, the other defaults to 1 (<code>start</code>) or the length of the respective sequence (<code>stop</code>).
</p>
<p><code>sum_aa</code> sums the amino acid compositions in the input <code>AAcomp</code> data frame.
It only applies to columns with the three-letter abbreviations of amino acids and to a column named <code>chains</code> (if present).
The values in these columns are multiplied by the indicated <code>abundance</code> after recycling to the number of proteins.
The values in these columns are then summed; if <code>average</code> is TRUE then the sum is divided by the number of proteins.
Proteins with missing values (NA) of amino acid composition or abundance are omitted from the calculation.
The output has one row and the same number of columns as the input; the value in the non-amino acid columns is taken from the first row of the input.
</p>


<h3>Value</h3>

<p><code>count_aa</code> returns a data frame with these columns (for proteins): <code>Ala</code>, <code>Cys</code>, <code>Asp</code>, <code>Glu</code>, <code>Phe</code>, <code>Gly</code>, <code>His</code>, <code>Ile</code>, <code>Lys</code>, <code>Leu</code>, <code>Met</code>, <code>Asn</code>, <code>Pro</code>, <code>Gln</code>, <code>Arg</code>, <code>Ser</code>, <code>Thr</code>, <code>Val</code>, <code>Trp</code>, <code>Tyr</code>.
For &lsquo;<span class="samp">&#8288;DNA&#8288;</span>&rsquo;, the columns are changed to <code>A</code>, <code>C</code>, <code>G</code>, <code>T</code>, and for &lsquo;<span class="samp">&#8288;RNA&#8288;</span>&rsquo;, the columns are changed to <code>A</code>, <code>C</code>, <code>G</code>, <code>U</code>.
</p>
<p><code>read_fasta</code> returns a list of sequences (for <code>type</code> equal to &lsquo;<span class="samp">&#8288;sequence&#8288;</span>&rsquo;) or a list of lines (for <code>type</code> equal to &lsquo;<span class="samp">&#8288;lines&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;headers&#8288;</span>&rsquo;).
Otherwise, (for <code>type</code> equal to &lsquo;<span class="samp">&#8288;count&#8288;</span>&rsquo;) a data frame with these columns: <code>protein</code>, <code>organism</code>, <code>ref</code>, <code>abbrv</code>, <code>chains</code>, and the columns described above for <code>count_aa</code>.
</p>
<p><code>sum_aa</code> returns a one-row data frame.
</p>


<h3>See Also</h3>

<p>Pass the output of <code>read_fasta</code> to <code><a href="CHNOSZ.html#topic+add.protein">add.protein</a></code> in the CHNOSZ package to set up thermodynamic calculations for proteins.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Reading a protein FASTA file
# The path to the file
file &lt;- system.file("extdata/fasta/KHAB17.fasta", package = "canprot")
# Read the sequences, and print the first one
read_fasta(file, type = "seq")[[1]]
# Count the amino acids in the sequences
aa &lt;- read_fasta(file)
# Calculate protein length (number of amino acids in each protein)
plength(aa)
# Sum the amino acid compositions
sum_aa(aa)

# Count amino acids in a sequence
count_aa("GGSGG")
# A message is issued for unrecognized characters
count_aa("AAAXXX")
# Count nucleobases in a sequence
bases &lt;- count_aa("ACCGGGTTT", molecule = "DNA")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
