<!DOCTYPE html><html><head><title>Help for package care</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {care}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#care-internal'><p>Internal care Functions</p></a></li>
<li><a href='#care-package'><p>The care Package</p></a></li>
<li><a href='#carscore'><p>Estimate CAR Scores and Marginal Correlations</p></a></li>
<li><a href='#efron2004'><p>Diabetes Data from Efron et al. (2004)</p></a></li>
<li><a href='#lu2004'><p>Brain Aging Study of Lu et al. (2004)</p></a></li>
<li><a href='#slm'><p>Shrinkage Estimation of Regression Coefficients</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.1.11</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-11-21</td>
</tr>
<tr>
<td>Title:</td>
<td>High-Dimensional Regression and CAR Score Variable Selection</td>
</tr>
<tr>
<td>Author:</td>
<td>Verena Zuber and Korbinian Strimmer.</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Korbinian Strimmer &lt;strimmerlab@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.2), corpcor (&ge; 1.6.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>crossval</td>
</tr>
<tr>
<td>Enhances:</td>
<td>sda</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements the regression approach 
  of Zuber and Strimmer (2011) "High-dimensional regression and variable 
  selection using CAR scores" SAGMB 10: 34, &lt;<a href="https://doi.org/10.2202%2F1544-6115.1730">doi:10.2202/1544-6115.1730</a>&gt;.
  CAR scores measure the correlation between the response and the 
  Mahalanobis-decorrelated  predictors.  The squared CAR score is a 
  natural measure of variable importance and provides a canonical 
  ordering of variables. This package provides functions for estimating 
  CAR scores, for variable selection using CAR scores, and for estimating 
  corresponding regression coefficients. Both shrinkage as well as 
  empirical estimators are available.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://strimmerlab.github.io/software/care/">https://strimmerlab.github.io/software/care/</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-11-21 16:04:28 UTC; strimmer</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-11-21 16:40:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='care-internal'>Internal care Functions</h2><span id='topic+care-internal'></span>

<h3>Description</h3>

<p>Internal care functions.
</p>


<h3>Note</h3>

<p>These are not to be called by the user (or in some cases are just
waiting for proper documentation to be written).
</p>

<hr>
<h2 id='care-package'>The care Package</h2><span id='topic+care-package'></span>

<h3>Description</h3>

<p>The &quot;care&quot; package implements the CAR regression approach 
described in Zuber and Strimmer (2011).  CAR scores measure the 
correlations between the response and the Mahalanobis-decorrelated 
predictors.  The squared CAR score is a natural measure of variable 
importance and provides a canonical ordering of variables - see
Zuber and Strimmer (2011) for details.
</p>
<p>This package provides functions for estimating CAR scores, for
variable selection using CAR scores, and for estimating corresponding
regression coefficients. Both shrinkage as well as empirical estimators 
are available.
</p>
<p>The name of the package refers to <b>CAR</b> 
<b>e</b>stimation  and <b>CAR</b> <b>re</b>gression.
</p>


<h3>Author(s)</h3>

<p>Verena Zuber and Korbinian Strimmer (<a href="https://strimmerlab.github.io/">https://strimmerlab.github.io/</a>)</p>


<h3>References</h3>

<p>Zuber, V., and K. Strimmer. 2011.  High-dimensional regression and 
variable selection using CAR scores.  Statist. Appl. Genet. Mol. Biol. 10: 34.
&lt;DOI:10.2202/1544-6115.1730&gt;
</p>
<p>Website: <a href="https://strimmerlab.github.io/software/care/">https://strimmerlab.github.io/software/care/</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+carscore">carscore</a></code>, <code><a href="#topic+slm">slm</a></code>, 
<code><a href="#topic+efron2004">efron2004</a></code>, <code><a href="#topic+lu2004">lu2004</a></code>.
</p>

<hr>
<h2 id='carscore'>Estimate CAR Scores and Marginal Correlations</h2><span id='topic+carscore'></span>

<h3>Description</h3>

<p><code>carscore</code> estimates the vector of CAR scores, either using the standard
empirical estimator of the correlation matrix, or a shrinkage estimator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carscore(Xtrain, Ytrain, lambda, diagonal=FALSE, verbose=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="carscore_+3A_xtrain">Xtrain</code></td>
<td>
<p>Matrix of predictors (columns correspond to variables).</p>
</td></tr>
<tr><td><code id="carscore_+3A_ytrain">Ytrain</code></td>
<td>
<p>Univariate response variable.</p>
</td></tr>
<tr><td><code id="carscore_+3A_lambda">lambda</code></td>
<td>
<p>The correlation shrinkage intensity (range 0-1).  
If not specified (the default) it is estimated using an
analytic formula from Sch\&quot;afer and Strimmer (2005).  For <code>lambda=0</code>
the empirical correlations are used.</p>
</td></tr>
<tr><td><code id="carscore_+3A_diagonal">diagonal</code></td>
<td>
<p>For <code>diagonal=FALSE</code> (the default) CAR scores are computed;
otherwise with <code>diagonal=TRUE</code> marginal correlations.   
</p>
</td></tr>
<tr><td><code id="carscore_+3A_verbose">verbose</code></td>
<td>
<p>If <code>verbose=TRUE</code> then the shrinkage intensity 
used in estimating the shrinkage correlation matrix is reported.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The CAR scores are the correlations between the response and the 
Mahalanobis-decorrelated predictors.  CAR score is an abbreviation
for Correlation-Adjusted (marginal) coRelation, where the first 
correlation matrix refers dependencies among predictors.
</p>
<p>In Zuber and Strimmer (2011) it
is argued that squared CAR scores are a natural measure for variable 
importance and it is shown that variable selection based on CAR scores
is highly efficient compared to competing approaches such as elastic net
lasso, or boosting. 
</p>
<p>If the response is binary (or descrete) the corresponding quantity
are CAT scores (see <code><a href="sda.html#topic+catscore">catscore</a></code>).
</p>


<h3>Value</h3>

<p><code>carscore</code> returns
a vector containing the CAR scores (or marginal correlations for <code>diagonal=TRUE</code>).
</p>


<h3>Author(s)</h3>

<p>Verena Zuber and Korbinian Strimmer (<a href="https://strimmerlab.github.io">https://strimmerlab.github.io</a>).
</p>


<h3>References</h3>

<p>Zuber, V., and K. Strimmer. 2011.  High-dimensional regression and 
variable selection using CAR scores.  Statist. Appl. Genet. Mol. Biol. 10: 34.
&lt;DOI:10.2202/1544-6115.1730&gt;
</p>


<h3>See Also</h3>

<p><code><a href="sda.html#topic+catscore">catscore</a></code>.</p>


<h3>Examples</h3>

<pre><code class='language-R'># load care library
library("care")

######

# empirical CAR scores for diabetes data
data(efron2004)
xnames = colnames(efron2004$x)
n = dim(efron2004$x)[1]

car = carscore(efron2004$x, efron2004$y, lambda=0)
car

# compare orderings

# variables ordered by squared CAR scores
xnames[order(car^2, decreasing=TRUE)]
# "bmi" "s5"  "bp"  "s3"  "s4"  "s6"  "sex" "age" "s2"  "s1" 

# compare with ordering by t-scores / partial correlations 
pcor = pcor.shrink(cbind(efron2004$y,efron2004$x), lambda=0, verbose=FALSE)[-1,1]
xnames[order(pcor^2, decreasing=TRUE)]
# "bmi" "bp"  "s5"  "sex" "s1"  "s2"  "s4"  "s6"  "s3"  "age"

# compare with ordering by marginal correlations 
mcor = cor(efron2004$y,efron2004$x)
#mcor = carscore(efron2004$x, efron2004$y, diagonal=TRUE, lambda=0)
xnames[order(mcor^2, decreasing=TRUE)]
# "bmi" "s5"  "bp"  "s4"  "s3"  "s6"  "s1"  "age" "s2"  "sex"

# decomposition of R^2
sum(car^2)
slm(efron2004$x, efron2004$y, lambda=0, lambda.var=0)$R2

# pvalues for empirical CAR scores
pval = 1-pbeta(car^2, shape1=1/2, shape2=(n-2)/2)
pval &lt;= 0.05

######

# shrinkage CAR scores for Lu et al. (2004) data
data(lu2004)
dim(lu2004$x)    # 30 403

# compute shrinkage car scores
car = carscore(lu2004$x, lu2004$y)

# most important genes
order(car^2, decreasing=TRUE)[1:10]

# compare with empirical marginal correlations
mcor = cor(lu2004$y, lu2004$x)
order(mcor^2, decreasing=TRUE)[1:10]

# decomposition of R^2
sum(car^2) 
slm(lu2004$x, lu2004$y)$R2

</code></pre>

<hr>
<h2 id='efron2004'>Diabetes Data from Efron et al. (2004)</h2><span id='topic+efron2004'></span>

<h3>Description</h3>

<p>Diabetes data (10 variables, 442 measurements) as used in 
the study of Efron et al. (2004).  The data is standardized such that
the means of all variables are zero, and all variances are equal to one.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(efron2004)
</code></pre>


<h3>Format</h3>

<p><code>efron2004$x</code> is a 422 x 10 matrix containing the measurements
of the explanatory variables (age, sex, body mass, etc.).
The rows contain the samples and the columns the variables.
</p>
<p><code>efron2004$y</code> contains the response.
</p>


<h3>Source</h3>

<p>The orginal data are available in the lars R package, 
see <a href="https://cran.r-project.org/package=lars">https://cran.r-project.org/package=lars</a>. Note that this uses a slightly different standardization.
</p>


<h3>References</h3>

<p>Efron, B., et al. 2004. Least angle regression (with discussion). Ann. Statist. 32:407&ndash;499. &lt;DOI:10.1214/009053604000000067&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load care library
library("care")

# load Efron et al. (2004) diabetes data set
data(efron2004)
dim(efron2004$x) # 442 10
colnames(efron2004$x)
length(efron2004$y) # 442
</code></pre>

<hr>
<h2 id='lu2004'>Brain Aging Study of Lu et al. (2004)</h2><span id='topic+lu2004'></span>

<h3>Description</h3>

<p>Gene expression data (403 genes for 30 samples) from the
microarray study of Lu et al. (2004).  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(lu2004)
</code></pre>


<h3>Format</h3>

<p><code>lu2004$x</code> is a 30 x 403 matrix containing the log expression levels.
The rows contain the samples and the columns the genes.
</p>
<p><code>lu2004$y</code> is the age of for each sample.
</p>


<h3>Details</h3>

<p> This data set contains measurements of the gene
expression of 403 genes from 30 human brain samples.  In addition,
the age of each patient is provided.  
</p>


<h3>Source</h3>

<p>The original data are available from the GEO public functional genomics database at URL <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE1572">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE1572</a> and are described in Lu et al. (2004).
The selected 403 genes result from prescreening and preprocessing as
described in Zuber and Strimmer (2011). 
</p>


<h3>References</h3>

<p>Lu, T., et al. 2004. Gene regulation and DNA damage in the ageing human brain.
Nature 429:883&ndash;891. &lt;DOI:10.1038/nature02661&gt;
</p>
<p>Zuber, V., and K. Strimmer. 2011.  High-dimensional regression and 
variable selection using CAR scores.  Statist. Appl. Genet. Mol. Biol. 10: 34.
&lt;DOI:10.2202/1544-6115.1730&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load care library
library("care")

# load Lu et al. (2004) data set
data(lu2004)
dim(lu2004$x) # 30 403
hist(lu2004$x)
length(lu2004$y) # 30
lu2004$y # age
</code></pre>

<hr>
<h2 id='slm'>Shrinkage Estimation of Regression Coefficients</h2><span id='topic+slm'></span><span id='topic+predict.slm'></span><span id='topic+make.predlist'></span>

<h3>Description</h3>

<p><code>slm</code> fits a linear model and computes
(standardized) regression coefficients by plugin of shrinkage estimates of correlations and variances.
Using the argument <code>predlist</code> several models can be fitted on the same data set.
</p>
<p><code>make.predlist</code> constructs a <code>predlist</code> argument for use with <code>slm</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>slm(Xtrain, Ytrain, predlist, lambda, lambda.var, diagonal=FALSE, verbose=TRUE)
## S3 method for class 'slm'
predict(object, Xtest, verbose=TRUE, ...)
make.predlist(ordering, numpred, name="SIZE")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="slm_+3A_xtrain">Xtrain</code></td>
<td>
<p>Matrix of predictors (columns correspond to variables).</p>
</td></tr>
<tr><td><code id="slm_+3A_ytrain">Ytrain</code></td>
<td>
<p>Univariate continous response variable.</p>
</td></tr>
<tr><td><code id="slm_+3A_predlist">predlist</code></td>
<td>
<p>A list specifying the predictors to be included when fitting the linear regression.  Each entry in the list is a vector containing the indices of variables used per model. If left unspecified single full-sized model using all variables in Xtrain is assumed. For a given ordering of covariables a suitable <code>predlist</code> can be generated  using the helper function <code><a href="#topic+make.predlist">make.predlist</a></code> - see examples below.
</p>
</td></tr>
<tr><td><code id="slm_+3A_lambda">lambda</code></td>
<td>
<p>The correlation shrinkage intensity (range 0-1).  
If not specified (the default) it is estimated using an
analytic formula from Sch\&quot;afer and Strimmer (2005).  For <code>lambda=0</code>
the empirical correlations are used.</p>
</td></tr>
<tr><td><code id="slm_+3A_lambda.var">lambda.var</code></td>
<td>
<p>The variance shrinkage intensity (range 0-1).  If
not specified (the default) it is estimated
using an analytic formula from Opgen-Rhein and Strimmer
(2007).  For <code>lambda.var=0</code> the empirical
variances are used.</p>
</td></tr>
<tr><td><code id="slm_+3A_diagonal">diagonal</code></td>
<td>
<p>If <code>diagonal=FALSE</code> (the default) then the correlation
among predictor veriables assumed to be non-zero and is estimated
from data. If <code>diagonal=TRUE</code> 
then it is assumed that the correlation among predictors vanishes
and is set to zero.</p>
</td></tr>
<tr><td><code id="slm_+3A_verbose">verbose</code></td>
<td>
<p>If <code>verbose=TRUE</code> then the estimated shrinkage intensities are reported.
</p>
</td></tr>
<tr><td><code id="slm_+3A_object">object</code></td>
<td>
<p>An <code>slm</code> fit object obtained from the function <code>slm</code>.</p>
</td></tr>
<tr><td><code id="slm_+3A_xtest">Xtest</code></td>
<td>
<p>A matrix containing the test data set. Note that 
the rows correspond to observations and the columns
to variables.</p>
</td></tr>
<tr><td><code id="slm_+3A_...">...</code></td>
<td>
<p>Additional arguments for generic predict.</p>
</td></tr>
<tr><td><code id="slm_+3A_ordering">ordering</code></td>
<td>
<p>The ordering of the predictors (most important predictors are first).
</p>
</td></tr>
<tr><td><code id="slm_+3A_numpred">numpred</code></td>
<td>
<p>The number of included predictors (may be a scalar or a vector). The predictors
are included in the order specified by <code>ordering</code>.
</p>
</td></tr>
<tr><td><code id="slm_+3A_name">name</code></td>
<td>
<p>The name assigned to each model is <code>name</code> plus &quot;.&quot; and the number of included predictors.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The regression coefficients are obtained by estimating the joint
joint covariance matrix of the response and the predictors,
and subsequently computing the the regression coefficients by
inversion of this matrix - see  Opgen-Rhein and Strimmer (2007).
As estimators for the covariance matrix either the standard empirical estimator
or a Stein-type shrinkage estimator is employed.  The use of the empirical covariance
leads to the OLS estimates of the regression coefficients, whereas otherwise
shrinkage estimates are obtained. 
</p>


<h3>Value</h3>

<p><code>slm</code> returns a list with the following components:
</p>
<p><code>regularization</code>: The shrinkage intensities used for estimating correlations and variances.
</p>
<p><code>std.coefficients</code>: The standardized regression coefficients, i.e. the regression coefficients
computed from centered and standardized input data.  Thus, by construction the intercept is zero.
Furthermore, for <code>diagonal=TRUE</code> the standardized regression coefficient for each predictor is
identical to the respective marginal correlation.
</p>
<p><code>coefficients</code>: Regression coefficients.
</p>
<p><code>numpred</code>: The number of predictors used in each investigated model.
</p>
<p><code>R2</code>: For <code>diagonal=TRUE</code> this is the multiple correlation coefficient 
between the response and the predictor, or the proportion of explained variance, with range
from 0 to 1.
For <code>diagonal=TRUE</code> this equals the sum of squared marginal
correlations.  Note that this sum may be larger than 1!
</p>
<p><code>sd.resid</code>: The residual unexplained error.
</p>
<p><code>predict.slm</code> returns the means predicted for each sample and model as well as the corresponding
predictive standard deviations (attached as attribute &quot;sd&quot;).
</p>


<h3>Author(s)</h3>

<p>Korbinian Strimmer (<a href="https://strimmerlab.github.io">https://strimmerlab.github.io</a>).
</p>


<h3>References</h3>

<p>Opgen-Rhein, R., and K. Strimmer. 2007.
From correlation to causation networks: a simple approximate learning 
algorithm and its   application to high-dimensional plant gene expression data.
BMC Syst. Biol. 1: 37.  &lt;DOI:10.1186/1752-0509-1-37&gt;
</p>
<p>Sch\&quot;afer, J., and K. Strimmer. 2005.  A shrinkage approach to
large-scale covariance estimation and implications for functional
genomics.  Statist. Appl. Genet. Mol. Biol. 4: 32.  &lt;DOI:10.2202/1544-6115.1175&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+carscore">carscore</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'># load care library
library("care")

## example with large number of samples and small dimension
## (using empirical estimates of regression coefficients)

# diabetes data
data(efron2004)
x = efron2004$x
y = efron2004$y
n = dim(x)[1]
d = dim(x)[2]
xnames = colnames(x)

# empirical regression coefficients
fit = slm(x, y, lambda=0, lambda.var=0)
fit
# note that in this example the regression coefficients
# and the standardized regression coefficients are identical
# as the input data have been standardized to mean zero and variance one

# compute corresponding t scores / partial correlations
df = n-d-1
pcor = pcor.shrink(cbind(y,x), lambda=0)[-1,1] 
t = pcor * sqrt(df/(1-pcor^2))
t.pval = 2 - 2 * pt(abs(t), df)
b = fit$coefficients[1,-1]
cbind(b, pcor, t, t.pval)

# compare results with those from lm function
lm.out = lm(y ~ x)
summary(lm.out)

# prediction of fitted values at the position of the training data
lm.out$fitted.values
mu.hat = predict(fit, x) # precticted means
mu.hat
attr(mu.hat, "sd") # predictive error
sd(y-mu.hat)


# ordering of the variables using squared empirical CAR score
car = carscore(x, y, lambda=0)
ocar = order(car^2, decreasing=TRUE)
xnames[ocar]

# CAR regression models with 5, 7, 9 included predictors
car.predlist = make.predlist(ocar, numpred = c(5,7,9), name="CAR")
car.predlist
slm(x, y, car.predlist, lambda=0, lambda.var=0)


# plot regression coefficients for all possible CAR models

p=ncol(x)
car.predlist = make.predlist(ocar, numpred = 1:p, name="CAR")
cm = slm(x, y, car.predlist, lambda=0, lambda.var=0)
bmat = cm$coefficients[,-1]
bmat

par(mfrow=c(2,1))

plot(1:p, bmat[,1], type="l", 
  ylab="estimated regression coefficients", 
  xlab="number of included predictors", 
  main="CAR Regression Models for Diabetes Data", 
  xlim=c(1,p+1), ylim=c(min(bmat), max(bmat)))

for (i in 2:p) lines(1:p, bmat[,i], col=i, lty=i)
for (i in 1:p) points(1:p, bmat[,i], col=i)
for (i in 1:p) text(p+0.5, bmat[p,i], xnames[i])

plot(1:p, cm$R2, type="l", 
  ylab="estimated R2",
  xlab="number of included predictors",
  main="Proportion of Explained Variance",
  ylim=c(0,0.6))
R2max = max(cm$R2)
lines(c(1,p), c(R2max, R2max), col=2)

par(mfrow=c(1,1))


## example with small number of samples and large dimension
## (using shrinkage estimates of regression coefficients)

data(lu2004)
dim(lu2004$x)    # 30 403

fit = slm(lu2004$x, lu2004$y)
fit

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
