<!DOCTYPE html><html><head><title>Help for package CARlasso</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CARlasso}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bGlasso'><p>Gibbs sampler for Bayesian Graphical LASSO and extensions</p></a></li>
<li><a href='#CARlasso'><p>Gibbs sampler for Conditional Autoregressive LASSO and extensions</p></a></li>
<li><a href='#horseshoe'><p>Horseshoe method for graphical structure inference</p></a></li>
<li><a href='#mgp154'><p>Gut microbiota in the Irish Elderly</p></a></li>
<li><a href='#mgp2592'><p>Hofmockel Soil Aggregate COB KBASE</p></a></li>
<li><a href='#plot.bglasso_out'><p>plot the graph estimated by graphical lasso with threshold method using ggraph</p></a></li>
<li><a href='#plot.carlasso_out'><p>plot the chain graph estimated by CAR-LASSO with threshold or horseshoe method using ggraph</p></a></li>
<li><a href='#rCARAlasso_'><p>Block Gibbs sampler for adaptive CAR-LASSO</p></a></li>
<li><a href='#rCARlasso_'><p>Block Gibbs sampler for CAR-LASSO</p></a></li>
<li><a href='#simu_AR1'><p>Simulate a simple AR1 model with specific predictor</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Conditional Autoregressive LASSO</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.2</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yunyi Shen &lt;yshen99@wisc.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Algorithms to fit Bayesian Conditional Autoregressive LASSO with automatic and adaptive shrinkage described in Shen and Solis-Lemus (2020) &lt;<a href="https://arxiv.org/abs/2012.08397">arXiv:2012.08397</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo, RcppProgress</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, coda, Matrix, igraph, ggraph, ggplot2, MASS, methods</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>C++11</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/YunyiShen/CAR-LASSO">https://github.com/YunyiShen/CAR-LASSO</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/YunyiShen/CAR-LASSO/issues">https://github.com/YunyiShen/CAR-LASSO/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-08-11 18:23:15 UTC; yunyi</td>
</tr>
<tr>
<td>Author:</td>
<td>Yunyi Shen <a href="https://orcid.org/0000-0003-2779-6507"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Claudia Solis-Lemus
    <a href="https://orcid.org/0000-0002-9789-8915"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-08-11 19:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='bGlasso'>Gibbs sampler for Bayesian Graphical LASSO and extensions</h2><span id='topic+bGlasso'></span>

<h3>Description</h3>

<p>Main sampling algorithm of Glasso model, note that the mean is in CAR parameterization
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bGlasso(
  data,
  link = "identity",
  r_Omega = 1,
  delta_Omega = 0.01,
  n_iter = 2000,
  n_burn_in = 1000,
  thin_by = 10,
  ns = 1000,
  m = 20,
  emax = 64,
  progress = TRUE,
  verbos = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bGlasso_+3A_data">data</code></td>
<td>
<p>A data.frame with all response, row as observations</p>
</td></tr>
<tr><td><code id="bGlasso_+3A_link">link</code></td>
<td>
<p>String name of link function? Currently can be &quot;identity&quot; for normal response, &quot;probit&quot; for binary, &quot;log&quot; for counting, &quot;logit&quot; for compositional. Note that when use &quot;logit&quot;, the last response will be used as reference.</p>
</td></tr>
<tr><td><code id="bGlasso_+3A_r_omega">r_Omega</code></td>
<td>
<p>Hyper-parameter for precision matrix, shape parameter of Gamma. Should be a scalar</p>
</td></tr>
<tr><td><code id="bGlasso_+3A_delta_omega">delta_Omega</code></td>
<td>
<p>Hyper-parameter for precision matrix, rate parameter of Gamma. Shoule be a scalar</p>
</td></tr>
<tr><td><code id="bGlasso_+3A_n_iter">n_iter</code></td>
<td>
<p>Number of sampling iterations (i.e. after burn in) for the Gibbs sampler</p>
</td></tr>
<tr><td><code id="bGlasso_+3A_n_burn_in">n_burn_in</code></td>
<td>
<p>Number of burn in iterations for the Gibbs sampler</p>
</td></tr>
<tr><td><code id="bGlasso_+3A_thin_by">thin_by</code></td>
<td>
<p>Final sample was thin by this number</p>
</td></tr>
<tr><td><code id="bGlasso_+3A_ns">ns</code></td>
<td>
<p>parameter for ARS, maximum number of hulls, only used when link is &quot;log&quot; and &quot;logit&quot;</p>
</td></tr>
<tr><td><code id="bGlasso_+3A_m">m</code></td>
<td>
<p>parameter for ARS, initial number of hulls, only used when link is &quot;log&quot; and &quot;logit&quot;</p>
</td></tr>
<tr><td><code id="bGlasso_+3A_emax">emax</code></td>
<td>
<p>parameter for ARS, tolerance for small values being 0, larger meaning we tolerate smaller values, only used when link is &quot;log&quot; and &quot;logit&quot;</p>
</td></tr>
<tr><td><code id="bGlasso_+3A_progress">progress</code></td>
<td>
<p>Bool, whether report progress from C++</p>
</td></tr>
<tr><td><code id="bGlasso_+3A_verbos">verbos</code></td>
<td>
<p>Bool, whether show warnings and messages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>bglasso_out</code> object with elements:
</p>

<ul>
<li><p><code style="white-space: pre;">&#8288;$point_est&#8288;</code>
</p>

<ul>
<li><p><code style="white-space: pre;">&#8288;$Omega&#8288;</code>: Posterior mean of precision matrix
</p>
</li></ul>


</li>
<li><p><code style="white-space: pre;">&#8288;$nodes&#8288;</code>
</p>

<ul>
<li><p><code style="white-space: pre;">&#8288;$responses&#8288;</code>: node name of responses
</p>
</li></ul>


</li>
<li><p><code style="white-space: pre;">&#8288;$data&#8288;</code>
</p>

<ul>
<li><p><code style="white-space: pre;">&#8288;$response&#8288;</code>: response matrix
</p>
</li></ul>


</li>
<li><p><code style="white-space: pre;">&#8288;$settings&#8288;</code>: all settings sent to the algorithm, exclude data
</p>
</li>
<li><p><code style="white-space: pre;">&#8288;$MCMC_output&#8288;</code>
</p>

<ul>
<li><p><code style="white-space: pre;">&#8288;$mu&#8288;</code>: A coda::mcmc object, each row was an MCMC sample of the mean vector
</p>
</li>
<li><p><code style="white-space: pre;">&#8288;$Omega&#8288;</code>: A coda::mcmc object, each row was an MCMC sample of the upper triangular part (with diagonal) of precision matrix Omega
</p>
</li>
<li><p><code style="white-space: pre;">&#8288;$lambda&#8288;</code>:  A coda::mcmc object, first column was the shrinkage parameter lambda for regression coefficient and the second column was shrinkage parameter lambda for precision matrix
</p>
</li></ul>


</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>set.seed(42)
dt &lt;- simu_AR1()
glassores &lt;- bGlasso(data = dt[,1:5])
plot(glassores) 

</code></pre>

<hr>
<h2 id='CARlasso'>Gibbs sampler for Conditional Autoregressive LASSO and extensions</h2><span id='topic+CARlasso'></span>

<h3>Description</h3>

<p>Main sampling algorithm of CAR-LASSO model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CARlasso(
  formula,
  data,
  link = "identity",
  adaptive = FALSE,
  r_beta = ifelse(adaptive, 0.01, 1),
  delta_beta = ifelse(adaptive, 1e-06, 0.01),
  r_Omega = ifelse(adaptive, 0.01, 1),
  delta_Omega = ifelse(adaptive, 1e-06, 0.01),
  lambda_diag = 0,
  n_iter = 2000,
  n_burn_in = 1000,
  thin_by = 10,
  ns = 1000,
  m = 20,
  emax = 64,
  progress = TRUE,
  verbos = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CARlasso_+3A_formula">formula</code></td>
<td>
<p>A double sided formula with response at left hand side and predictors at right hand side</p>
</td></tr>
<tr><td><code id="CARlasso_+3A_data">data</code></td>
<td>
<p>A data.frame with all response and predictors, row as observations</p>
</td></tr>
<tr><td><code id="CARlasso_+3A_link">link</code></td>
<td>
<p>String name of link function? Currently can be &quot;identity&quot; for normal response, &quot;probit&quot; for binary, &quot;log&quot; for counting, &quot;logit&quot; for compositional. Note that when use &quot;logit&quot;, the last response will be used as reference.</p>
</td></tr>
<tr><td><code id="CARlasso_+3A_adaptive">adaptive</code></td>
<td>
<p>Bool, whether run the adaptive version of the model</p>
</td></tr>
<tr><td><code id="CARlasso_+3A_r_beta">r_beta</code></td>
<td>
<p>Hyper-parameter for regression coefficient, shape parameter of Gamma, if adaptive, should have row number same as number of predictors while column number of responses</p>
</td></tr>
<tr><td><code id="CARlasso_+3A_delta_beta">delta_beta</code></td>
<td>
<p>Hyper-parameter for regression coefficient, rate parameter of Gamma, if adaptive, should have row number same as number of predictors while column number of responses</p>
</td></tr>
<tr><td><code id="CARlasso_+3A_r_omega">r_Omega</code></td>
<td>
<p>Hyper-parameter for precision matrix, shape parameter of Gamma. If adaptive, can be a matrix with same size as precision matrix, if this is the case, only upper triangular part without diagonal will be used, or can be a vector whose size was the upper triangular part of precision matrix, if non-adaptive, a scalar.</p>
</td></tr>
<tr><td><code id="CARlasso_+3A_delta_omega">delta_Omega</code></td>
<td>
<p>Hyper-parameter for precision matrix, rate parameter of Gamma, If adaptive, can be a matrix with same size as precision matrix, if this is the case, only upper triangular part without diagonal will be used, or can be a vector whose size was the upper triangular part of precision matrix, if non-adaptive, a scalar.</p>
</td></tr>
<tr><td><code id="CARlasso_+3A_lambda_diag">lambda_diag</code></td>
<td>
<p>adaptive only hyper-parameter for penalties on diagonal entries of Omega, should have dimension k and non-negative</p>
</td></tr>
<tr><td><code id="CARlasso_+3A_n_iter">n_iter</code></td>
<td>
<p>Number of sampling iterations (i.e. after burn in) for the Gibbs sampler</p>
</td></tr>
<tr><td><code id="CARlasso_+3A_n_burn_in">n_burn_in</code></td>
<td>
<p>Number of burn in iterations for the Gibbs sampler</p>
</td></tr>
<tr><td><code id="CARlasso_+3A_thin_by">thin_by</code></td>
<td>
<p>Final sample was thin by this number</p>
</td></tr>
<tr><td><code id="CARlasso_+3A_ns">ns</code></td>
<td>
<p>parameter for ARS, maximum number of hulls, only used when link is &quot;log&quot; and &quot;logit&quot;</p>
</td></tr>
<tr><td><code id="CARlasso_+3A_m">m</code></td>
<td>
<p>parameter for ARS, initial number of hulls, only used when link is &quot;log&quot; and &quot;logit&quot;</p>
</td></tr>
<tr><td><code id="CARlasso_+3A_emax">emax</code></td>
<td>
<p>parameter for ARS, tolerance for small values being 0, larger meaning we tolerate smaller values, only used when link is &quot;log&quot; and &quot;logit&quot;</p>
</td></tr>
<tr><td><code id="CARlasso_+3A_progress">progress</code></td>
<td>
<p>Bool, whether report progress from C++</p>
</td></tr>
<tr><td><code id="CARlasso_+3A_verbos">verbos</code></td>
<td>
<p>Bool, whether show warnings and messages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>carlasso_out</code> object with elements:
</p>

<ul>
<li><p><code style="white-space: pre;">&#8288;$point_est&#8288;</code>
</p>

<ul>
<li><p><code style="white-space: pre;">&#8288;$Omega&#8288;</code>: Posterior mean of precision matrix
</p>
</li>
<li><p><code style="white-space: pre;">&#8288;$beta&#8288;</code>: Posterior mean of regression coefficient
</p>
</li>
<li><p><code style="white-space: pre;">&#8288;$CAR&#8288;</code>
</p>

<ul>
<li><p><code style="white-space: pre;">&#8288;$C&#8288;</code>: The conditional regression coefficients among responses
</p>
</li>
<li><p><code style="white-space: pre;">&#8288;$B&#8288;</code>: The conditional regression coefficients between response and predictors
</p>
</li>
<li><p><code style="white-space: pre;">&#8288;$M&#8288;</code>: The conditional variance
</p>
</li></ul>


</li></ul>


</li>
<li><p><code style="white-space: pre;">&#8288;$nodes&#8288;</code>
</p>

<ul>
<li><p><code style="white-space: pre;">&#8288;$responses&#8288;</code>: node name of responses
</p>
</li>
<li><p><code style="white-space: pre;">&#8288;$predictors&#8288;</code>: node name of predictors
</p>
</li></ul>


</li>
<li><p><code style="white-space: pre;">&#8288;$data&#8288;</code>
</p>

<ul>
<li><p><code style="white-space: pre;">&#8288;$response&#8288;</code>: response matrix
</p>
</li>
<li><p><code style="white-space: pre;">&#8288;$design&#8288;</code>: design matrix
</p>
</li></ul>


</li>
<li><p><code style="white-space: pre;">&#8288;$settings&#8288;</code>: all settings sent to the algorithm, exclude data
</p>
</li>
<li><p><code style="white-space: pre;">&#8288;$MCMC_output&#8288;</code>
</p>

<ul>
<li><p><code style="white-space: pre;">&#8288;$beta&#8288;</code>: A coda::mcmc object, each row was an MCMC sample of the (column) vectorization of regression coefficient B
</p>
</li>
<li><p><code style="white-space: pre;">&#8288;$mu&#8288;</code>: A coda::mcmc object, each row was an MCMC sample of the mean vector
</p>
</li>
<li><p><code style="white-space: pre;">&#8288;$Omega&#8288;</code>: A coda::mcmc object, each row was an MCMC sample of the upper triangular part (with diagonal) of precision matrix Omega
</p>
</li>
<li><p><code style="white-space: pre;">&#8288;$lambda&#8288;</code>: <strong>Non-adaptive only</strong>, A coda::mcmc object, first column was the shrinkage parameter lambda for regression coefficient and the second column was shrinkage parameter lambda for precision matrix
</p>
</li>
<li><p><code style="white-space: pre;">&#8288;$lambda_beta&#8288;</code>: <strong>Adaptive only</strong>, A coda::mcmc object, each row was an MCMC sample of the (column) vectorization of shrinkage parameter for regression coefficient B
</p>
</li>
<li><p><code style="white-space: pre;">&#8288;$lambda_Omega&#8288;</code>: <strong>Adaptive only</strong>, A coda::mcmc object, each row was an MCMC sample of the shrinkage parameter for the upper triangular part (without diagonal) of precision matrix Omega
</p>
</li></ul>


</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>set.seed(42)
dt &lt;- simu_AR1()
car_res &lt;- CARlasso(y1+y2+y3+y4+y5~x1+x2+x3+x4+x5, data = dt, adaptive = TRUE)
plot(car_res,tol = 0.05)
# with horseshoe inference
car_res &lt;- horseshoe(car_res)
plot(car_res)


</code></pre>

<hr>
<h2 id='horseshoe'>Horseshoe method for graphical structure inference</h2><span id='topic+horseshoe'></span>

<h3>Description</h3>

<p>Horseshoe method for graphical structure inference
</p>


<h3>Usage</h3>

<pre><code class='language-R'>horseshoe(obj, Bbar = NULL, A = NULL, nu = 3, V = NULL, thr = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="horseshoe_+3A_obj">obj</code></td>
<td>
<p>The carlasso_out object from CARlasso</p>
</td></tr>
<tr><td><code id="horseshoe_+3A_bbar">Bbar</code></td>
<td>
<p>Prior mean of regression coefficients, default all 0s</p>
</td></tr>
<tr><td><code id="horseshoe_+3A_a">A</code></td>
<td>
<p>Prior precision of regression coefficients, default 1e-8</p>
</td></tr>
<tr><td><code id="horseshoe_+3A_nu">nu</code></td>
<td>
<p>Prior degree of freedom of the Wishart on precision matrix</p>
</td></tr>
<tr><td><code id="horseshoe_+3A_v">V</code></td>
<td>
<p>prior covariance matrix of the Wishart on precision matrix</p>
</td></tr>
<tr><td><code id="horseshoe_+3A_thr">thr</code></td>
<td>
<p>threshold for horseshoe inference, default 0.5</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This method fits a linear regression with less informative prior on any parameters and compare the posterior mean with the LASSO result. If LASSO is comparably less than result without sparsity prior, we argue that the edge should be absent
</p>


<h3>Value</h3>

<p>A <code>carlasso_out</code> object with learned binary adjacency matrix and multi-response linear regression MCMC out put
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(42)
dt &lt;- simu_AR1()
car_res &lt;- CARlasso(y1+y2+y3+y4+y5~x1+x2+x3+x4+x5, data = dt, adaptive = TRUE)
car_res &lt;- horseshoe(car_res)
plot(car_res)
</code></pre>

<hr>
<h2 id='mgp154'>Gut microbiota in the Irish Elderly</h2><span id='topic+mgp154'></span>

<h3>Description</h3>

<p>This study is based on pyrosequencing of 16S rDNA amplicons from faecal samples collected from 178 elderly Irish citizens and 13 healthy young control subjects. A subset of these samples were also subjected to shotgun sequencing using Illumina HiSeq 2000 2x91bp reads. Antibiotic treatment was an exclusion criterion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(mgp154)
</code></pre>


<h3>Format</h3>

<p>An data.frame with genus and predictors.
</p>


<h3>Source</h3>

<p><a href="https://www.mg-rast.org/mgmain.html?mgpage=project&amp;project=mgp154">MG-RAST-mgp154</a>
</p>


<h3>References</h3>

<p>Claesson, Marcus J., et al. &quot;Gut microbiota composition correlates with diet and health in the elderly.&quot; Nature 488.7410 (2012): 178-184.
</p>

<hr>
<h2 id='mgp2592'>Hofmockel Soil Aggregate COB KBASE</h2><span id='topic+mgp2592'></span>

<h3>Description</h3>

<p>This study is to examine soil microbial community composition and structure of both bacteria and fungi at a microbially-relevant scale. The researchers isolated soil aggregates from three land management systems in central Iowa to test if the aggregate-level microbial responses are related to plant community and management practices. The clean dataset has 120 samples with 17 genus under consideration.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(mgp2592)
</code></pre>


<h3>Format</h3>

<p>An data.frame with genus and predictors.
</p>


<h3>Source</h3>

<p><a href="https://www.mg-rast.org/mgmain.html?mgpage=project&amp;project=mgp2592">MG-RAST-mgp2592</a>
</p>


<h3>References</h3>

<p>Bach, Elizabeth M., et al. &quot;Greatest soil microbial diversity found in micro-habitats.&quot; Soil biology and Biochemistry 118 (2018): 217-226.
</p>

<hr>
<h2 id='plot.bglasso_out'>plot the graph estimated by graphical lasso with threshold method using ggraph</h2><span id='topic+plot.bglasso_out'></span>

<h3>Description</h3>

<p>plot the graph estimated by graphical lasso with threshold method using ggraph
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'bglasso_out'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.bglasso_out_+3A_x">x</code></td>
<td>
<p>The bglasso_out</p>
</td></tr>
<tr><td><code id="plot.bglasso_out_+3A_...">...</code></td>
<td>

<ul>
<li><p><code>tol</code>: threshold for ploting default 0.01, if horseshoed, then horseshoe result is used
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot</code> xect
</p>

<hr>
<h2 id='plot.carlasso_out'>plot the chain graph estimated by CAR-LASSO with threshold or horseshoe method using ggraph</h2><span id='topic+plot.carlasso_out'></span>

<h3>Description</h3>

<p>plot the chain graph estimated by CAR-LASSO with threshold or horseshoe method using ggraph
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'carlasso_out'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.carlasso_out_+3A_x">x</code></td>
<td>
<p>The carlasso_out xect</p>
</td></tr>
<tr><td><code id="plot.carlasso_out_+3A_...">...</code></td>
<td>

<ul>
<li><p><code>tol</code>: threshold for ploting default 0.01, if horseshoed, then horseshoe result is used
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot</code> xect
</p>

<hr>
<h2 id='rCARAlasso_'>Block Gibbs sampler for adaptive CAR-LASSO</h2><span id='topic+rCARAlasso_'></span>

<h3>Description</h3>

<p><strong>This function is for advanced users to build their own sampler use adaptive CARlasso as core.</strong> It will execute one round of Gibbs sampler of adaptive CAR-LASSO model. Be aware that the function is a <code>void</code> function implemented in C++, and all updated parameters e.g. Omega will be manipulate directly in memory to save space. Users should manage to do their own work to save the state. Also be aware that R uses shallow copy by default, which means one cannot save the state by simply give it to another object e.g. first <code>Omega_old &lt;- Omega_curr</code> then update <code>Omega_curr</code>, <code>Omega_old</code> will also change. <strong>This function will NOT check dimensions of input.</strong> Below we assume n samples, k responses and p predictors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rCARAlasso_(
  Z_curr,
  design,
  lambda2_beta,
  tau2_curr,
  beta_curr,
  lambda_Omega,
  Omega_curr,
  mu_curr,
  r_beta,
  delta_beta,
  r_Omega,
  delta_Omega,
  lambda_diag,
  k,
  p,
  n
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rCARAlasso__+3A_z_curr">Z_curr</code></td>
<td>
<p>the current (latent) normal Z_curr, should be n*k. Will not be changed</p>
</td></tr>
<tr><td><code id="rCARAlasso__+3A_design">design</code></td>
<td>
<p>the design matrix, should be n*p. Will not be changed</p>
</td></tr>
<tr><td><code id="rCARAlasso__+3A_lambda2_beta">lambda2_beta</code></td>
<td>
<p>the current shrinkage parameter of regression coefficients, should be a vector with p*k entries. Will be updated</p>
</td></tr>
<tr><td><code id="rCARAlasso__+3A_tau2_curr">tau2_curr</code></td>
<td>
<p>the current latent scale parameter in the normal mixture representation of Laplace, for regression coefficients, should be a vector with p*k entries. Will be updated.</p>
</td></tr>
<tr><td><code id="rCARAlasso__+3A_beta_curr">beta_curr</code></td>
<td>
<p>the current regression coefficients, should be a matrix sized p*k (p row and k columns). Will be updated.</p>
</td></tr>
<tr><td><code id="rCARAlasso__+3A_lambda_omega">lambda_Omega</code></td>
<td>
<p>the current shrinkage parameter for Omega, should be a vector with k*(k-1)/2 entries. Will be updated.</p>
</td></tr>
<tr><td><code id="rCARAlasso__+3A_omega_curr">Omega_curr</code></td>
<td>
<p>the current Omega matrix, should be a matrix of size k*k. Will be updated.</p>
</td></tr>
<tr><td><code id="rCARAlasso__+3A_mu_curr">mu_curr</code></td>
<td>
<p>the current mu, intercept, should be a vector of size k. Will be updated.</p>
</td></tr>
<tr><td><code id="rCARAlasso__+3A_r_beta">r_beta</code></td>
<td>
<p>hyperprior's parameter of shrinkage for regression coefficients, should be a scalar of type 'double' and positive. Will not be updated.</p>
</td></tr>
<tr><td><code id="rCARAlasso__+3A_delta_beta">delta_beta</code></td>
<td>
<p>hyperprior's parameter of shrinkage for regression coefficients, should be a scalar of type 'double' and positive. Will not be updated.</p>
</td></tr>
<tr><td><code id="rCARAlasso__+3A_r_omega">r_Omega</code></td>
<td>
<p>hyperprior's parameter of shrinkage for precision Omega, should be a scalar of type 'double' and positive. Will not be updated.</p>
</td></tr>
<tr><td><code id="rCARAlasso__+3A_delta_omega">delta_Omega</code></td>
<td>
<p>hyperprior's parameter of shrinkage for rprecision Omega, should be a scalar of type 'double' and positive. Will not be updated.</p>
</td></tr>
<tr><td><code id="rCARAlasso__+3A_lambda_diag">lambda_diag</code></td>
<td>
<p>shrinkage parameter of the diagonal of Omega, should be a vector of size k, should be non-negative. Will not be updated</p>
</td></tr>
<tr><td><code id="rCARAlasso__+3A_k">k</code></td>
<td>
<p>integer, number of responses</p>
</td></tr>
<tr><td><code id="rCARAlasso__+3A_p">p</code></td>
<td>
<p>integer, number of predictors</p>
</td></tr>
<tr><td><code id="rCARAlasso__+3A_n">n</code></td>
<td>
<p>integer, number of Z_curr points</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Again this is a <code>void</code> function and will not return anything. All update happened in memory directly.
</p>

<hr>
<h2 id='rCARlasso_'>Block Gibbs sampler for CAR-LASSO</h2><span id='topic+rCARlasso_'></span>

<h3>Description</h3>

<p><strong>This function is for advanced users to build their own sampler use CARlasso as core.</strong> It will execute one round of Gibbs sampler of CAR-LASSO model. Be aware that the function is a <code>void</code> function implemented in C++, and all updated parameters e.g. Omega will be manipulate directly in memory to save space. Users should manage to do their own work to save the state. Also be aware that R uses shallow copy by default, which means one cannot save the state by simply give it to another object e.g. first <code>Omega_old &lt;- Omega_curr</code> then update <code>Omega_curr</code>, <code>Omega_old</code> will also change. <strong>This function will NOT check dimensions of input.</strong> Below we assume n samples, k responses and p predictors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rCARlasso_(
  Z_curr,
  design,
  lambda2_beta,
  tau2_curr,
  beta_curr,
  lambda_Omega,
  Omega_curr,
  mu_curr,
  r_beta,
  delta_beta,
  r_Omega,
  delta_Omega,
  k,
  p,
  n
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rCARlasso__+3A_z_curr">Z_curr</code></td>
<td>
<p>the current (latent) normal data, should be n*k. Will not be changed</p>
</td></tr>
<tr><td><code id="rCARlasso__+3A_design">design</code></td>
<td>
<p>the design matrix, should be n*p. Will not be changed</p>
</td></tr>
<tr><td><code id="rCARlasso__+3A_lambda2_beta">lambda2_beta</code></td>
<td>
<p>the current shrinkage parameter of regression coefficients, should be a scalar of type <code>double</code>. Will be updated</p>
</td></tr>
<tr><td><code id="rCARlasso__+3A_tau2_curr">tau2_curr</code></td>
<td>
<p>the current latent scale parameter in the normal mixture representation of Laplace, for regression coefficients, should be a vector with p*k entries. Will be updated.</p>
</td></tr>
<tr><td><code id="rCARlasso__+3A_beta_curr">beta_curr</code></td>
<td>
<p>the current regression coefficients, should be a matrix sized p*k (p row and k columns). Will be updated.</p>
</td></tr>
<tr><td><code id="rCARlasso__+3A_lambda_omega">lambda_Omega</code></td>
<td>
<p>the current shrinkage parameter for Omega, should be a scalar of tyoe <code>double</code>. Will be updated.</p>
</td></tr>
<tr><td><code id="rCARlasso__+3A_omega_curr">Omega_curr</code></td>
<td>
<p>the current Omega matrix, should be a matrix of size k*k. Will be updated.</p>
</td></tr>
<tr><td><code id="rCARlasso__+3A_mu_curr">mu_curr</code></td>
<td>
<p>the current mu, intercept, should be a vector of size k. Will be updated.</p>
</td></tr>
<tr><td><code id="rCARlasso__+3A_r_beta">r_beta</code></td>
<td>
<p>hyperprior's parameter of shrinkage for regression coefficients, should be a scalar of type 'double' and positive. Will not be updated.</p>
</td></tr>
<tr><td><code id="rCARlasso__+3A_delta_beta">delta_beta</code></td>
<td>
<p>hyperprior's parameter of shrinkage for regression coefficients, should be a scalar of type 'double' and positive. Will not be updated.</p>
</td></tr>
<tr><td><code id="rCARlasso__+3A_r_omega">r_Omega</code></td>
<td>
<p>hyperprior's parameter of shrinkage for precision Omega, should be a scalar of type 'double' and positive. Will not be updated.</p>
</td></tr>
<tr><td><code id="rCARlasso__+3A_delta_omega">delta_Omega</code></td>
<td>
<p>hyperprior's parameter of shrinkage for rprecision Omega, should be a scalar of type 'double' and positive. Will not be updated.</p>
</td></tr>
<tr><td><code id="rCARlasso__+3A_k">k</code></td>
<td>
<p>integer, number of responses</p>
</td></tr>
<tr><td><code id="rCARlasso__+3A_p">p</code></td>
<td>
<p>integer, number of predictors</p>
</td></tr>
<tr><td><code id="rCARlasso__+3A_n">n</code></td>
<td>
<p>integer, number of data points</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Again this is a <code>void</code> function and will not return anything. All update happened in memory directly.
</p>

<hr>
<h2 id='simu_AR1'>Simulate a simple AR1 model with specific predictor</h2><span id='topic+simu_AR1'></span>

<h3>Description</h3>

<p>Simulate a simple AR1 model with specific predictor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simu_AR1(n = 100, k = 5, rho = 0.7)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simu_AR1_+3A_n">n</code></td>
<td>
<p>sample size</p>
</td></tr>
<tr><td><code id="simu_AR1_+3A_k">k</code></td>
<td>
<p>number of responses</p>
</td></tr>
<tr><td><code id="simu_AR1_+3A_rho">rho</code></td>
<td>
<p>partial correlation in AR1</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simulate a simple AR1 model with k responses and k predictors, each predictor has effect on exact one response node
</p>


<h3>Value</h3>

<p>a dataframe, with y1 to yk as responses and x1 to xk as predictors
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
