<!DOCTYPE html><html><head><title>Help for package CascadeData</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CascadeData}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CascadeData-package'><p>Experimental Data of Cascade Experiments in Genomics</p></a></li>
<li><a href='#micro_S'><p>Stimulated dataset</p></a></li>
<li><a href='#micro_US'><p>Unstimulated control dataset</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Experimental Data of Cascade Experiments in Genomics</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-11-28</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Author:</td>
<td>Frederic Bertrand <a href="https://orcid.org/0000-0002-0837-8281"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [cre, aut],
  Myriam Maumy-Bertrand
    <a href="https://orcid.org/0000-0002-4615-1512"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Laurent Vallat [ctb],
  Nicolas Jung [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Frederic Bertrand &lt;frederic.bertrand@utt.fr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>These experimental expression data (5 leukemic 'CLL' B-lymphocyte of aggressive form from 'GSE39411', &lt;<a href="https://doi.org/10.1073%2Fpnas.1211130110">doi:10.1073/pnas.1211130110</a>&gt;), after B-cell receptor stimulation, are used as examples by packages such as the 'Cascade' one, a modeling tool allowing gene selection, reverse engineering, and prediction in cascade networks. Jung, N., Bertrand, F., Bahram, S., Vallat, L., and Maumy-Bertrand, M. (2014) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtt705">doi:10.1093/bioinformatics/btt705</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://fbertran.github.io/CascadeData/">https://fbertran.github.io/CascadeData/</a>,
<a href="https://github.com/fbertran/CascadeData/">https://github.com/fbertran/CascadeData/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/fbertran/CascadeData/issues/">https://github.com/fbertran/CascadeData/issues/</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-11-28 10:13:43 UTC; fbertran</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-11-28 12:20:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='CascadeData-package'>Experimental Data of Cascade Experiments in Genomics</h2><span id='topic+CascadeData-package'></span><span id='topic+CascadeData'></span>

<h3>Description</h3>

<p>These are the data from the agressive subject group the from GSE39411
dataset, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE39411">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE39411</a>,
Vallat, L., Kemper, C. A., Jung, N., Maumy-Bertrand, M., Bertrand, F.,
..., Bahram, S. (2013). Reverse-engineering the genetic circuitry of a
cancer cell with predicted intervention in chronic lymphocytic leukemia.
Proceedings of the National Academy of Sciences, 110(2), 459-464,
<a href="https://doi.org/10.1073/pnas.1211130110">doi:10.1073/pnas.1211130110</a>
</p>


<h3>Details</h3>

<p>5 leukemic CLL B-lymphocyte of aggressive form were stimulated in vitro with
an anti-IgM antibody, activating the B-cell receptor (BCR). We analyzed the
gene expression at 4 time points (60, 90, 210 and 390 minutes). Each gene
expression measurement is performed both in stimulated cells and in control
unstimulated cells.
</p>
<p>The data were normalized and are ready to use.
</p>
<p>These Experimental Data are used as examples by packages such as the Cascade
one <a href="https://doi.org/10.1093/bioinformatics/btt705">doi:10.1093/bioinformatics/btt705</a> in one of its vignettes. The Cascade
package is a modeling tool allowing gene selection, reverse engineering, and
prediction in Cascade networks.
</p>
<p>Data were collected on HG-U133_Plus_2, Affymetrix Human Genome U133 Plus 2.0
Array.
</p>


<h3>Author(s)</h3>

<p>This package has been written by Frederic Bertrand, Myriam
Maumy-Bertrand and Nicolas Jung with biological insights from Laurent
Vallat.
</p>
<p>Maintainer: Frederic Bertrand &lt;frederic.bertrand@utt.fr&gt;
</p>


<h3>References</h3>

<p>Jung, N., Bertrand, F., Bahram, S., Vallat, L., and
Maumy-Bertrand, M. (2013). Cascade: a R-package to study, predict and
simulate the diffusion of a signal through a temporal gene network.
<em>Bioinformatics</em>, btt705.
</p>
<p>Vallat, L., Kemper, C. A., Jung, N., Maumy-Bertrand, M., Bertrand, F., ... &amp;
Bahram, S. (2013). Reverse-engineering the genetic circuitry of a cancer
cell with predicted intervention in chronic lymphocytic leukemia.
<em>Proceedings of the National Academy of Sciences</em>, 110(2), 459-464.
</p>

<hr>
<h2 id='micro_S'>Stimulated dataset</h2><span id='topic+micro_S'></span>

<h3>Description</h3>

<p>This is the stimulated data part of the GSE39411 dataset,
<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE39411">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE39411</a>. Data were
normalized and are ready to use.
</p>


<h3>Format</h3>

<p>A data frame with 54613 probesets measured 6 times throught 4 time
points.
</p>


<h3>Details</h3>

<p>5 leukemic CLL B-lymphocyte of aggressive form were stimulated in vitro with
an anti-IgM antibody, activating the B-cell receptor (BCR). We analyzed the
gene expression at 4 time points (60, 90, 210 and 390 minutes). Each gene
expression measurement is performed both in stimulated cells and in control
unstimulated cells. This is the stimulated cells dataset.
</p>
<p>Data were collected on HG-U133_Plus_2, Affymetrix Human Genome U133 Plus 2.0
Array.
</p>


<h3>References</h3>

<p>Vallat, L., Kemper, C. A., Jung, N., Maumy-Bertrand, M.,
Bertrand, F., ..., Bahram, S. (2013). Reverse-engineering the genetic
circuitry of a cancer cell with predicted intervention in chronic
lymphocytic leukemia. Proceedings of the National Academy of Sciences,
110(2), 459-464, <a href="https://doi.org/10.1073/pnas.1211130110">doi:10.1073/pnas.1211130110</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(micro_S)

</code></pre>

<hr>
<h2 id='micro_US'>Unstimulated control dataset</h2><span id='topic+micro_US'></span>

<h3>Description</h3>

<p>This is the unstimulated data part of the GSE39411 dataset,
<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE39411">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE39411</a>. Data were
normalized and are ready to use.
</p>


<h3>Format</h3>

<p>A data frame with 54613 probesets measured 6 times throught 4 time
points.
</p>


<h3>Details</h3>

<p>5 leukemic CLL B-lymphocyte of aggressive form were stimulated in vitro with
an anti-IgM antibody, activating the B-cell receptor (BCR). We analyzed the
gene expression at 4 time points (60, 90, 210 and 390 minutes). Each gene
expression measurement is performed both in stimulated cells and in control
unstimulated cells. This is the unstimulated cells dataset.
</p>
<p>Data were collected on HG-U133_Plus_2, Affymetrix Human Genome U133 Plus 2.0
Array.
</p>


<h3>References</h3>

<p>Vallat, L., Kemper, C. A., Jung, N., Maumy-Bertrand, M.,
Bertrand, F., ..., Bahram, S. (2013). Reverse-engineering the genetic
circuitry of a cancer cell with predicted intervention in chronic
lymphocytic leukemia. Proceedings of the National Academy of Sciences,
110(2), 459-464, <a href="https://doi.org/10.1073/pnas.1211130110">doi:10.1073/pnas.1211130110</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(micro_US)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
