<!DOCTYPE html><html><head><title>Help for package CASMAP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CASMAP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CASMAP-class'><p>Constructor for CASMAP class object.</p></a></li>
<li><a href='#CASMAPenv'><p>Global variables environment</p></a></li>
<li><a href='#checkIsBoolean'><p>Check if a variable is boolean or not</p></a></li>
<li><a href='#getExampleCovariatesFilename'><p>Get the path to the example covariates file for regionGWAS mode</p></a></li>
<li><a href='#getExampleDataFilename'><p>Get the path to the example data file for regionGWAS mode</p></a></li>
<li><a href='#getExampleLabelsFilename'><p>Get the path to the example labels file for regionGWAS mode</p></a></li>
<li><a href='#getExampleSignificantRegionsFilename'><p>Get the path to the example significant intervals file</p></a></li>
<li><a href='#getHigherOrderEpistasisString'><p>Gets the higherOrderEpistasis string</p></a></li>
<li><a href='#getMinModeLength'><p>Gets the minModeLength</p></a></li>
<li><a href='#getParentFunctionName'><p>Get the function name</p></a></li>
<li><a href='#getRegionGWASString'><p>Gets the regionGWAS string</p></a></li>
<li><a href='#isHigherOrderEpistasisString'><p>Checks if substring is part of higherOrderEpistasis</p></a></li>
<li><a href='#isInOpenInterval'><p>A method to check value is numeric and in open interval</p></a></li>
<li><a href='#isRegionGWASString'><p>Checks if substring is part of regionGWAS</p></a></li>
<li><a href='#lib_delete_search_chi'><p>Internal function</p></a></li>
<li><a href='#lib_delete_search_e'><p>Internal function</p></a></li>
<li><a href='#lib_delete_search_facs'><p>Internal function</p></a></li>
<li><a href='#lib_delete_search_fastcmh'><p>Internal function</p></a></li>
<li><a href='#lib_execute_int'><p>Internal function</p></a></li>
<li><a href='#lib_execute_iset'><p>Internal function</p></a></li>
<li><a href='#lib_filter_intervals_write_to_file'><p>Internal function</p></a></li>
<li><a href='#lib_get_filtered_intervals'><p>Internal function</p></a></li>
<li><a href='#lib_get_result_facs'><p>Internal function</p></a></li>
<li><a href='#lib_get_result_fais'><p>Internal function</p></a></li>
<li><a href='#lib_get_result_int'><p>Internal function</p></a></li>
<li><a href='#lib_get_result_iset'><p>Internal function</p></a></li>
<li><a href='#lib_get_significant_intervals'><p>Internal function</p></a></li>
<li><a href='#lib_get_significant_itemsets'><p>Internal function</p></a></li>
<li><a href='#lib_new_search_chi'><p>Internal function</p></a></li>
<li><a href='#lib_new_search_e'><p>Internal function</p></a></li>
<li><a href='#lib_new_search_facs'><p>Internal function</p></a></li>
<li><a href='#lib_new_search_fastcmh'><p>Internal function</p></a></li>
<li><a href='#lib_profiler_write_to_file'><p>Internal function</p></a></li>
<li><a href='#lib_pvals_significant_ints_write_to_file'><p>Internal function</p></a></li>
<li><a href='#lib_pvals_significant_isets_write_to_file'><p>Internal function</p></a></li>
<li><a href='#lib_pvals_testable_ints_write_to_file'><p>Internal function</p></a></li>
<li><a href='#lib_pvals_testable_isets_write_to_file'><p>Internal function</p></a></li>
<li><a href='#lib_read_covariates_file_facs'><p>Internal function</p></a></li>
<li><a href='#lib_read_covariates_file_fastcmh'><p>Internal function</p></a></li>
<li><a href='#lib_read_eth_files'><p>Internal function</p></a></li>
<li><a href='#lib_read_eth_files_with_cov_facs'><p>Internal function</p></a></li>
<li><a href='#lib_read_eth_files_with_cov_fastcmh'><p>Internal function</p></a></li>
<li><a href='#lib_read_plink_files'><p>Internal function</p></a></li>
<li><a href='#lib_read_plink_files_with_cov_facs'><p>Internal function</p></a></li>
<li><a href='#lib_read_plink_files_with_cov_fastcmh'><p>Internal function</p></a></li>
<li><a href='#lib_summary_write_to_file_facs'><p>Internal function</p></a></li>
<li><a href='#lib_summary_write_to_file_fais'><p>Internal function</p></a></li>
<li><a href='#lib_summary_write_to_file_fastcmh'><p>Internal function</p></a></li>
<li><a href='#lib_write_eth_files_int'><p>Internal function</p></a></li>
<li><a href='#lib_write_eth_files_iset'><p>Internal function</p></a></li>
<li><a href='#lib_write_eth_files_with_cov_facs'><p>Internal function</p></a></li>
<li><a href='#lib_write_eth_files_with_cov_fastcmh'><p>Internal function</p></a></li>
<li><a href='#m_SignificantFeaturesSearch-class'><p>Internal class</p></a></li>
<li><a href='#m_SignificantFeaturesSearchWithCovariates-class'><p>Internal class</p></a></li>
<li><a href='#m_SignificantIntervalSearch-class'><p>Internal class</p></a></li>
<li><a href='#m_SignificantIntervalSearchFais-class'><p>Internal class</p></a></li>
<li><a href='#m_SignificantItemsetSearch-class'><p>Internal class</p></a></li>
<li><a href='#modeErrorMessage'><p>Error message for mode</p></a></li>
<li><a href='#modeLengthErrorMessage'><p>Error message for mode, if too short</p></a></li>
<li><a href='#modeNeedsMoreChars'><p>Checks mode string is long enough</p></a></li>
<li><a href='#SignificantIntervalSearchChi-class'><p>Approximate fast significant interval search</p></a></li>
<li><a href='#SignificantIntervalSearchExact-class'><p>Internal class for search for significant regions</p></a></li>
<li><a href='#SignificantIntervalSearchFastCmh-class'><p>Fast significant interval search with categorical covariates</p></a></li>
<li><a href='#SignificantItemsetSearchFacs-class'><p>Significant itemsets search with categorical covariates</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Detection of Statistically Significant Combinations of SNPs in
Association Mapping</td>
</tr>
<tr>
<td>Version:</td>
<td>0.6.1</td>
</tr>
<tr>
<td>Description:</td>
<td>A significant pattern mining-based toolbox for region-based genome-wide association studies and higher-order epistasis analyses, implementing the methods described in Llinares-López et al. (2017) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtx071">doi:10.1093/bioinformatics/btx071</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.2)</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, Rcpp</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>C++11</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown</td>
</tr>
<tr>
<td>Author:</td>
<td>Felipe Llinares-López [aut, cph],
  Laetitia Papaxanthos [aut, cph],
  Damian Roqueiro [aut, cph],
  Matthew Baker [ctr],
  Mikołaj Rybiński [ctr],
  Uwe Schmitt [ctr],
  Dean Bodenham [aut, cre, cph],
  Karsten Borgwardt [aut, fnd, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dean Bodenham &lt;deanbodenhambsse@gmail.com&gt;</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-05-04 18:15:36 UTC; dean</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-05-05 18:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='CASMAP-class'>Constructor for CASMAP class object.</h2><span id='topic+CASMAP-class'></span><span id='topic+CASMAP'></span>

<h3>Description</h3>

<p>Constructor for CASMAP class object.
</p>


<h3>Details</h3>

<p>Constructor for CASMAP class object, which needs the <code>mode</code>
parameter to be set by the user. Please see the examples.
</p>


<h3>Fields</h3>


<dl>
<dt><code>mode</code></dt><dd><p>Either <code>'regionGWAS'</code> or <code>'higherOrderEpistasis'</code>.</p>
</dd>
<dt><code>alpha</code></dt><dd><p>A numeric value setting the Family-wise Error Rate (FWER).
Must be strictly between <code>0</code> and <code>1</code>. Default
value is <code>0.05</code>.</p>
</dd>
<dt><code>max_comb_size</code></dt><dd><p>A numeric specifying the maximum length of 
combinations. For example, if set to <code>4</code>, 
then only combinations of size between <code>1</code>
and <code>4</code> (inclusive) will be considered.
To consider combinations of arbitrary (maximal)
length, use value <code>0</code>, which is the default 
value.</p>
</dd>
</dl>


<h3>Base method, for both modes</h3>


<dl>
<dt><code>readFiles</code></dt><dd><p>Read the data, label and possibly covariates 
files. Parameters are <code>genotype_file</code>, 
for the data, <code>phenotype_file</code> for the
labels and (optional) <code>covariates_file</code>
for the covariates. The option 
<code>plink_file_root</code> is not supported
in the current version, but will be supported
in future versions.</p>
</dd>
<dt><code>setMode</code></dt><dd><p>Can set/change the mode, but note that any
data files will need to read in again using
the <code>readFiles</code> command.</p>
</dd>
<dt><code>setTargetFWER</code></dt><dd><p>Can set/change the Family-wise 
Error Rate (FWER). Takes a numeric
parameter <code>alpha</code>, strictly between
<code>0</code> and <code>1</code>.</p>
</dd>
<dt><code>execute</code></dt><dd><p>Once the data files have been read, can execute the
algorithm. Please note that, depending on the size
of the data files, this could take a long time.</p>
</dd>
<dt><code>getSummary</code></dt><dd><p>Returns a data frame with a summary of the 
results from the execution, but not any
significant regions/itemsets. See 
<code>getSignificantRegions</code>, 
<code>getSignificantInteractions</code>, and
<code>getSignificantClusterRepresentatives</code>. </p>
</dd>
<dt><code>writeSummary</code></dt><dd><p>Directly write the information
from <code>getSummary</code> to file.</p>
</dd>
</dl>



<h3><code>regionGWAS</code> Methods</h3>


<dl>
<dt><code>getSignificantRegions</code></dt><dd><p>Returns a data frame with the 
the significant regions. Only valid when
<code>mode='regionGWAS'</code>.</p>
</dd>
<dt><code>getSignificantClusterRepresentatives</code></dt><dd><p>Returns a data 
frame with the 
the representatives of the significant 
clusters. This will be a subset of the regions
returned from <code>getSignificantRegions</code>. 
Only valid when <code>mode='regionGWAS'</code>.</p>
</dd>
<dt><code>writeSignificantRegions</code></dt><dd><p>Writes the data from 
<code>getSignificantRegions</code> to file, which
must be specified in the parameter 
<code>path</code>.
Only valid when <code>mode='regionGWAS'</code>.</p>
</dd>
<dt><code>writeSignificantClusterRepresentatives</code></dt><dd><p>Writes the data 
from 
<code>getSignificantClusterRepresentatives</code> to 
file, which must be specified in the parameter 
<code>path</code>.
Only valid when <code>mode='regionGWAS'</code>.</p>
</dd>
</dl>



<h3><code>higherOrderEpistasis</code> Methods</h3>


<dl>
<dt><code>getSignificantInteractions</code></dt><dd><p>Returns the frame 
from <code>getSignificantInteractions</code> to 
file, which must be specified in the parameter 
<code>path</code>. Only valid 
when <code>mode='higherOrderEpistasis'</code>.</p>
</dd>
<dt><code>writeSignificantInteractions</code></dt><dd><p>Writes a data frame with  
the significant interactions. Only valid 
when <code>mode='higherOrderEpistasis'</code>.</p>
</dd>
</dl>



<h3>References</h3>

<p>A. Terada, M. Okada-Hatakeyama, K. Tsuda and J. Sese
<em>Statistical significance of combinatorial regulations</em>,
Proceedings of the National Academy of Sciences (2013) 110 
(32): 12996-13001
</p>
<p>F. Llinares-Lopez, D. G. Grimm, D. Bodenham,
U. Gieraths, M. Sugiyama, B. Rowan and K. Borgwardt,
<em>Genome-wide detection of intervals of genetic heterogeneity 
associated with complex traits</em>,
ISMB 2015, Bioinformatics (2015) 31 (12): i240-i249
</p>
<p>L. Papaxanthos, F. Llinares-Lopez, D. Bodenham,
K .Borgwardt,
<em>Finding significant combinations of features in the 
presence of categorical covariates</em>, Advances
in Neural Information Processing Systems 29 (NIPS 2016), 2271-2279.
</p>
<p>F. Llinares-Lopez, L. Papaxanthos, D. Bodenham,
D. Roqueiro and K .Borgwardt,
<em>Genome-wide genetic heterogeneity discovery 
with categorical covariates</em>.
Bioinformatics 2017, 33 (12): 1820-1828.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## An example using the "regionGWAS" mode
fastcmh &lt;- CASMAP(mode="regionGWAS")      # initialise object

datafile &lt;- getExampleDataFilename()      # file name of example data
labelsfile &lt;- getExampleLabelsFilename()  # file name of example labels
covfile &lt;- getExampleCovariatesFilename() # file name of example covariates 

# read the data, labels and covariate files
fastcmh$readFiles(genotype_file=getExampleDataFilename(),
                  phenotype_file=getExampleLabelsFilename(), 
                  covariate_file=getExampleCovariatesFilename() )

# execute the algorithm (this may take some time)
fastcmh$execute()

#get the summary results
summary_results &lt;- fastcmh$getSummary()

#get the significant regions
sig_regions &lt;- fastcmh$getSignificantRegions()

#get the clustered representatives for the significant regions
sig_cluster_rep &lt;- fastcmh$getSignificantClusterRepresentatives()


## Another example of regionGWAS
fais &lt;- CASMAP(mode="regionGWAS")      # initialise object

# read the data and labels, but no covariates
fastcmh$readFiles(genotype_file=getExampleDataFilename(),
                  phenotype_file=getExampleLabelsFilename())


## Another example, doing higher order epistasis search
facs &lt;- CASMAP(mode="higherOrderEpistasis")      # initialise object

</code></pre>

<hr>
<h2 id='CASMAPenv'>Global variables environment</h2><span id='topic+CASMAPenv'></span>

<h3>Description</h3>

<p>An environment to store a few global variables. Internal.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CASMAPenv
</code></pre>


<h3>Format</h3>

<p>An object of class <code>environment</code> of length 3.</p>

<hr>
<h2 id='checkIsBoolean'>Check if a variable is boolean or not</h2><span id='topic+checkIsBoolean'></span>

<h3>Description</h3>

<p>Checks if a variable is boolean, if not throws an error, otherwise
returns boolean.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkIsBoolean(var, name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkIsBoolean_+3A_var">var</code></td>
<td>
<p>The variable to be checked (if boolean).</p>
</td></tr>
<tr><td><code id="checkIsBoolean_+3A_name">name</code></td>
<td>
<p>The name of the variable to appear in any error message.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If not boolean (or <code>NA</code>), throws error.
If <code>NA</code>, return <code>FALSE</code>. Otherwise return 
boolean value of <code>var</code>.
</p>

<hr>
<h2 id='getExampleCovariatesFilename'>Get the path to the example covariates file for regionGWAS mode</h2><span id='topic+getExampleCovariatesFilename'></span>

<h3>Description</h3>

<p>Path to <code>CASMAP_example_covariates_1.txt</code> in <code>inst/extdata</code>.
The covariates categories for the data set  
<code>CASMAP_example_data_1.txt</code>, the path to which is given by 
<code>getExampleDataFilename</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getExampleCovariatesFilename()
</code></pre>


<h3>Format</h3>

<p>A single column vector of 100 labels, each of which
is <code>0</code> or <code>1</code> (same format as labels file).</p>


<h3>Details</h3>

<p>Path to the file containing the labels, for reading in to
CASMAP object using the <code>readFiles</code> function.
</p>


<h3>See Also</h3>

<p><code>getExampleDataFilename</code>, 
<code>getExampleLabelsFilename</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>covfile &lt;- getExampleCovariatesFilename()
</code></pre>

<hr>
<h2 id='getExampleDataFilename'>Get the path to the example data file for regionGWAS mode</h2><span id='topic+getExampleDataFilename'></span>

<h3>Description</h3>

<p>Path to <code>CASMAP_example_data_1.txt</code> in <code>inst/extdata</code>.
A dataset containing binary samples for the regionGWAS method.
There are accompanying labels and covariates dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getExampleDataFilename()
</code></pre>


<h3>Format</h3>

<p>A matrix of <code>0</code>s and <code>1</code>s, with 1000 rows (features) 
and 100 columns
(samples). In other words, each column is a sample, and each sample
has 1000 binary features.</p>


<h3>Details</h3>

<p>Path to the file containing the data, for reading in to
CASMAP object using the <code>readFiles</code> function.
Note that the significant region is <code>[99, 102]</code>.
</p>


<h3>See Also</h3>

<p><code>getExampleLabelsFilename</code>, 
<code>getExampleCovariatesFilename</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>datafile &lt;- getExampleDataFilename()
</code></pre>

<hr>
<h2 id='getExampleLabelsFilename'>Get the path to the example labels file for regionGWAS mode</h2><span id='topic+getExampleLabelsFilename'></span>

<h3>Description</h3>

<p>Path to <code>CASMAP_example_labels_1.txt</code> in <code>inst/extdata</code>.
A dataset containing the binary labels for the data in the file 
<code>CASMAP_example_data_1.txt</code>, the path to which is given by 
<code>getExampleDataFilename</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getExampleLabelsFilename()
</code></pre>


<h3>Format</h3>

<p>A single column of 100 labels, each of which is either <code>0</code>
or <code>1</code>.</p>


<h3>Details</h3>

<p>Path to the file containing the labels, for reading in to
CASMAP object using the <code>readFiles</code> function.
</p>


<h3>See Also</h3>

<p><code>getExampleDataFilename</code>, 
<code>getExampleCovariatesFilename</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>labelsfile &lt;- getExampleLabelsFilename()
</code></pre>

<hr>
<h2 id='getExampleSignificantRegionsFilename'>Get the path to the example significant intervals file</h2><span id='topic+getExampleSignificantRegionsFilename'></span>

<h3>Description</h3>

<p>Path to <code>CASMAP_example_covariates_1.txt</code> in 
<code>inst/extdata</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getExampleSignificantRegionsFilename()
</code></pre>


<h3>Examples</h3>

<pre><code class='language-R'>sigregfile &lt;- getExampleSignificantRegionsFilename()
</code></pre>

<hr>
<h2 id='getHigherOrderEpistasisString'>Gets the higherOrderEpistasis string</h2><span id='topic+getHigherOrderEpistasisString'></span>

<h3>Description</h3>

<p>A getter for the global <code>higherOrderEpistasis</code> value, a string
for the mode parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getHigherOrderEpistasisString()
</code></pre>

<hr>
<h2 id='getMinModeLength'>Gets the minModeLength</h2><span id='topic+getMinModeLength'></span><span id='topic+getMinModeLength'></span>

<h3>Description</h3>

<p>A getter for the global <code>minModeLength</code> value, a string
for the mode parameter.
</p>
<p>Gets the minimum mode character length (should be 3)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMinModeLength()

getMinModeLength()
</code></pre>

<hr>
<h2 id='getParentFunctionName'>Get the function name</h2><span id='topic+getParentFunctionName'></span>

<h3>Description</h3>

<p>Uses <code>match.call</code> and <code>as.character</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getParentFunctionName()
</code></pre>

<hr>
<h2 id='getRegionGWASString'>Gets the regionGWAS string</h2><span id='topic+getRegionGWASString'></span>

<h3>Description</h3>

<p>A getter for the global <code>regionGWAS</code> value, a string
for the mode parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getRegionGWASString()
</code></pre>

<hr>
<h2 id='isHigherOrderEpistasisString'>Checks if substring is part of higherOrderEpistasis</h2><span id='topic+isHigherOrderEpistasisString'></span>

<h3>Description</h3>

<p>Using grep to search through vector of strings
</p>


<h3>Usage</h3>

<pre><code class='language-R'>isHigherOrderEpistasisString(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="isHigherOrderEpistasisString_+3A_x">x</code></td>
<td>
<p>The string which will be compared to 'higherOrderEpistasis'</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Uses <code>grep</code> to search for exact match.
</p>


<h3>Value</h3>

<p><code>TRUE</code> if the string is a substring of 'higherOrderEpistasis',
otherwise returns <code>FALSE</code>.
</p>

<hr>
<h2 id='isInOpenInterval'>A method to check value is numeric and in open interval</h2><span id='topic+isInOpenInterval'></span>

<h3>Description</h3>

<p>Checks if a value is numeric and strictly between two other values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>isInOpenInterval(x, lower = 0, upper = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="isInOpenInterval_+3A_x">x</code></td>
<td>
<p>Value to be checked. Needs to be numeric.</p>
</td></tr>
<tr><td><code id="isInOpenInterval_+3A_lower">lower</code></td>
<td>
<p>Lower bound. Default value is <code>0</code>.</p>
</td></tr>
<tr><td><code id="isInOpenInterval_+3A_upper">upper</code></td>
<td>
<p>Upper bound. Default value is <code>1</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If numeric, and  strictly greater than <code>lower</code> and 
strictly smaller than <code>upper</code>, then return <code>TRUE</code>. 
Else return <code>FALSE</code>.
</p>

<hr>
<h2 id='isRegionGWASString'>Checks if substring is part of regionGWAS</h2><span id='topic+isRegionGWASString'></span>

<h3>Description</h3>

<p>Using<code>grepl</code> to compare strings, ignoring case.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>isRegionGWASString(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="isRegionGWASString_+3A_x">x</code></td>
<td>
<p>The string which will be compared to 'regionGWAS'</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Uses <code>grepl</code> to search for exact match. Case will be ignored.
</p>


<h3>Value</h3>

<p><code>TRUE</code> if the string is a substring of 'regionGWAS',
otherwise returns <code>FALSE</code>.
</p>

<hr>
<h2 id='lib_delete_search_chi'>Internal function</h2><span id='topic+lib_delete_search_chi'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_delete_search_chi(inst)
</code></pre>

<hr>
<h2 id='lib_delete_search_e'>Internal function</h2><span id='topic+lib_delete_search_e'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_delete_search_e(inst)
</code></pre>

<hr>
<h2 id='lib_delete_search_facs'>Internal function</h2><span id='topic+lib_delete_search_facs'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_delete_search_facs(inst)
</code></pre>

<hr>
<h2 id='lib_delete_search_fastcmh'>Internal function</h2><span id='topic+lib_delete_search_fastcmh'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_delete_search_fastcmh(inst)
</code></pre>

<hr>
<h2 id='lib_execute_int'>Internal function</h2><span id='topic+lib_execute_int'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_execute_int(inst, alpha, l_max)
</code></pre>

<hr>
<h2 id='lib_execute_iset'>Internal function</h2><span id='topic+lib_execute_iset'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_execute_iset(inst, alpha, l_max)
</code></pre>

<hr>
<h2 id='lib_filter_intervals_write_to_file'>Internal function</h2><span id='topic+lib_filter_intervals_write_to_file'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_filter_intervals_write_to_file(inst, output_file)
</code></pre>

<hr>
<h2 id='lib_get_filtered_intervals'>Internal function</h2><span id='topic+lib_get_filtered_intervals'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_get_filtered_intervals(inst)
</code></pre>

<hr>
<h2 id='lib_get_result_facs'>Internal function</h2><span id='topic+lib_get_result_facs'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_get_result_facs(inst)
</code></pre>

<hr>
<h2 id='lib_get_result_fais'>Internal function</h2><span id='topic+lib_get_result_fais'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_get_result_fais(inst)
</code></pre>

<hr>
<h2 id='lib_get_result_int'>Internal function</h2><span id='topic+lib_get_result_int'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_get_result_int(inst)
</code></pre>

<hr>
<h2 id='lib_get_result_iset'>Internal function</h2><span id='topic+lib_get_result_iset'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_get_result_iset(inst)
</code></pre>

<hr>
<h2 id='lib_get_significant_intervals'>Internal function</h2><span id='topic+lib_get_significant_intervals'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_get_significant_intervals(inst)
</code></pre>

<hr>
<h2 id='lib_get_significant_itemsets'>Internal function</h2><span id='topic+lib_get_significant_itemsets'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_get_significant_itemsets(inst)
</code></pre>

<hr>
<h2 id='lib_new_search_chi'>Internal function</h2><span id='topic+lib_new_search_chi'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_new_search_chi()
</code></pre>

<hr>
<h2 id='lib_new_search_e'>Internal function</h2><span id='topic+lib_new_search_e'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_new_search_e()
</code></pre>

<hr>
<h2 id='lib_new_search_facs'>Internal function</h2><span id='topic+lib_new_search_facs'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_new_search_facs()
</code></pre>

<hr>
<h2 id='lib_new_search_fastcmh'>Internal function</h2><span id='topic+lib_new_search_fastcmh'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_new_search_fastcmh()
</code></pre>

<hr>
<h2 id='lib_profiler_write_to_file'>Internal function</h2><span id='topic+lib_profiler_write_to_file'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_profiler_write_to_file(inst, output_file)
</code></pre>

<hr>
<h2 id='lib_pvals_significant_ints_write_to_file'>Internal function</h2><span id='topic+lib_pvals_significant_ints_write_to_file'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_pvals_significant_ints_write_to_file(inst, output_file)
</code></pre>

<hr>
<h2 id='lib_pvals_significant_isets_write_to_file'>Internal function</h2><span id='topic+lib_pvals_significant_isets_write_to_file'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_pvals_significant_isets_write_to_file(inst, output_file)
</code></pre>

<hr>
<h2 id='lib_pvals_testable_ints_write_to_file'>Internal function</h2><span id='topic+lib_pvals_testable_ints_write_to_file'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_pvals_testable_ints_write_to_file(inst, output_file)
</code></pre>

<hr>
<h2 id='lib_pvals_testable_isets_write_to_file'>Internal function</h2><span id='topic+lib_pvals_testable_isets_write_to_file'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_pvals_testable_isets_write_to_file(inst, output_file)
</code></pre>

<hr>
<h2 id='lib_read_covariates_file_facs'>Internal function</h2><span id='topic+lib_read_covariates_file_facs'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_read_covariates_file_facs(inst, cov_filename)
</code></pre>

<hr>
<h2 id='lib_read_covariates_file_fastcmh'>Internal function</h2><span id='topic+lib_read_covariates_file_fastcmh'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_read_covariates_file_fastcmh(inst, cov_filename)
</code></pre>

<hr>
<h2 id='lib_read_eth_files'>Internal function</h2><span id='topic+lib_read_eth_files'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_read_eth_files(inst, x_filename, y_filename, encoding)
</code></pre>

<hr>
<h2 id='lib_read_eth_files_with_cov_facs'>Internal function</h2><span id='topic+lib_read_eth_files_with_cov_facs'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_read_eth_files_with_cov_facs(inst, x_filename, y_filename, covfilename,
  encoding)
</code></pre>

<hr>
<h2 id='lib_read_eth_files_with_cov_fastcmh'>Internal function</h2><span id='topic+lib_read_eth_files_with_cov_fastcmh'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_read_eth_files_with_cov_fastcmh(inst, x_filename, y_filename, covfilename,
  encoding)
</code></pre>

<hr>
<h2 id='lib_read_plink_files'>Internal function</h2><span id='topic+lib_read_plink_files'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_read_plink_files(inst, base_filename, encoding)
</code></pre>

<hr>
<h2 id='lib_read_plink_files_with_cov_facs'>Internal function</h2><span id='topic+lib_read_plink_files_with_cov_facs'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_read_plink_files_with_cov_facs(inst, base_filename, covfilename, encoding)
</code></pre>

<hr>
<h2 id='lib_read_plink_files_with_cov_fastcmh'>Internal function</h2><span id='topic+lib_read_plink_files_with_cov_fastcmh'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_read_plink_files_with_cov_fastcmh(inst, base_filename, covfilename,
  encoding)
</code></pre>

<hr>
<h2 id='lib_summary_write_to_file_facs'>Internal function</h2><span id='topic+lib_summary_write_to_file_facs'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_summary_write_to_file_facs(inst, output_file)
</code></pre>

<hr>
<h2 id='lib_summary_write_to_file_fais'>Internal function</h2><span id='topic+lib_summary_write_to_file_fais'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_summary_write_to_file_fais(inst, output_file)
</code></pre>

<hr>
<h2 id='lib_summary_write_to_file_fastcmh'>Internal function</h2><span id='topic+lib_summary_write_to_file_fastcmh'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_summary_write_to_file_fastcmh(inst, output_file)
</code></pre>

<hr>
<h2 id='lib_write_eth_files_int'>Internal function</h2><span id='topic+lib_write_eth_files_int'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_write_eth_files_int(inst, x_filename, y_filename)
</code></pre>

<hr>
<h2 id='lib_write_eth_files_iset'>Internal function</h2><span id='topic+lib_write_eth_files_iset'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_write_eth_files_iset(inst, x_filename, y_filename)
</code></pre>

<hr>
<h2 id='lib_write_eth_files_with_cov_facs'>Internal function</h2><span id='topic+lib_write_eth_files_with_cov_facs'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_write_eth_files_with_cov_facs(inst, x_filename, y_filename, covfilename)
</code></pre>

<hr>
<h2 id='lib_write_eth_files_with_cov_fastcmh'>Internal function</h2><span id='topic+lib_write_eth_files_with_cov_fastcmh'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lib_write_eth_files_with_cov_fastcmh(inst, x_filename, y_filename, covfilename)
</code></pre>

<hr>
<h2 id='m_SignificantFeaturesSearch-class'>Internal class</h2><span id='topic+m_SignificantFeaturesSearch-class'></span><span id='topic+m_SignificantFeaturesSearch'></span>

<h3>Description</h3>

<p>in internal class
</p>

<hr>
<h2 id='m_SignificantFeaturesSearchWithCovariates-class'>Internal class</h2><span id='topic+m_SignificantFeaturesSearchWithCovariates-class'></span><span id='topic+m_SignificantFeaturesSearchWithCovariates'></span>

<h3>Description</h3>

<p>Internal class
</p>

<hr>
<h2 id='m_SignificantIntervalSearch-class'>Internal class</h2><span id='topic+m_SignificantIntervalSearch-class'></span><span id='topic+m_SignificantIntervalSearch'></span>

<h3>Description</h3>

<p>An internal class
</p>

<hr>
<h2 id='m_SignificantIntervalSearchFais-class'>Internal class</h2><span id='topic+m_SignificantIntervalSearchFais-class'></span><span id='topic+m_SignificantIntervalSearchFais'></span>

<h3>Description</h3>

<p>An internal class.
</p>

<hr>
<h2 id='m_SignificantItemsetSearch-class'>Internal class</h2><span id='topic+m_SignificantItemsetSearch-class'></span><span id='topic+m_SignificantItemsetSearch'></span>

<h3>Description</h3>

<p>Internal class
</p>

<hr>
<h2 id='modeErrorMessage'>Error message for mode</h2><span id='topic+modeErrorMessage'></span>

<h3>Description</h3>

<p>Return the appropriate error message for incorrect mode input
</p>


<h3>Usage</h3>

<pre><code class='language-R'>modeErrorMessage()
</code></pre>

<hr>
<h2 id='modeLengthErrorMessage'>Error message for mode, if too short</h2><span id='topic+modeLengthErrorMessage'></span>

<h3>Description</h3>

<p>Return the appropriate error message for incorrect mode input
</p>


<h3>Usage</h3>

<pre><code class='language-R'>modeLengthErrorMessage()
</code></pre>

<hr>
<h2 id='modeNeedsMoreChars'>Checks mode string is long enough</h2><span id='topic+modeNeedsMoreChars'></span>

<h3>Description</h3>

<p>Checks mode string is at least minimum length
</p>


<h3>Usage</h3>

<pre><code class='language-R'>modeNeedsMoreChars(mode)
</code></pre>

<hr>
<h2 id='SignificantIntervalSearchChi-class'>Approximate fast significant interval search</h2><span id='topic+SignificantIntervalSearchChi-class'></span><span id='topic+SignificantIntervalSearchChi'></span>

<h3>Description</h3>

<p>Class for approximate significant intervals search with Tarone correction for
bounding intermediate FWERs.
</p>

<hr>
<h2 id='SignificantIntervalSearchExact-class'>Internal class for search for significant regions</h2><span id='topic+SignificantIntervalSearchExact-class'></span><span id='topic+SignificantIntervalSearchExact'></span>

<h3>Description</h3>

<p>Please use the <code>CASMAP</code> constructor.
</p>

<hr>
<h2 id='SignificantIntervalSearchFastCmh-class'>Fast significant interval search with categorical covariates</h2><span id='topic+SignificantIntervalSearchFastCmh-class'></span><span id='topic+SignificantIntervalSearchFastCmh'></span>

<h3>Description</h3>

<p>Internal class, please use <code>CASMAP</code> constructor.
</p>

<hr>
<h2 id='SignificantItemsetSearchFacs-class'>Significant itemsets search with categorical covariates</h2><span id='topic+SignificantItemsetSearchFacs-class'></span><span id='topic+SignificantItemsetSearchFacs'></span>

<h3>Description</h3>

<p>Internal class, please use <code>CASMAP</code> constructor.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
