<!DOCTYPE html><html><head><title>Help for package CB2</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CB2}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calc_mappability'><p>A function to calculate the mappabilities of each NGS sample.</p></a></li>
<li><a href='#Evers_CRISPRn_RT112'><p>A benchmark CRISPRn pooled screen data from Evers et al.</p></a></li>
<li><a href='#fit_ab'><p>A C++ function to perform a parameter estimation for the sgRNA-level test.</p>
It will estimate two different parameters 'phat' and 'vhat,'
and we assume input count data follows the beta-binomial distribution.
Dr. Keith Baggerly initially implemented this code in Matlab,
and it has been rewritten it in C++ for the speed-up.</a></li>
<li><a href='#get_CPM'><p>A function to normalize sgRNA read counts.</p></a></li>
<li><a href='#join_count_and_design'><p>A function to join a count table and a design table.</p></a></li>
<li><a href='#measure_gene_stats'><p>A function to perform gene-level test using a sgRNA-level statistics.</p></a></li>
<li><a href='#measure_sgrna_stats'><p>A function to perform a statistical test at a sgRNA-level</p></a></li>
<li><a href='#plot_corr_heatmap'><p>A function to show a heatmap sgRNA-level corrleations of the NGS samples.</p></a></li>
<li><a href='#plot_count_distribution'><p>A function to plot read count distribution.</p></a></li>
<li><a href='#plot_dotplot'><p>A function to visualize dot plots for a gene.</p></a></li>
<li><a href='#plot_PCA'><p>A function to plot the first two principal components of samples.</p></a></li>
<li><a href='#quant'><p>A C++ function to quantify sgRNA abundance from NGS samples.</p></a></li>
<li><a href='#run_estimation'><p>A function to perform a statistical test at a sgRNA-level, deprecated.</p></a></li>
<li><a href='#run_sgrna_quant'><p>A function to run a sgRNA quantification algorithm from NGS sample</p></a></li>
<li><a href='#Sanson_CRISPRn_A375'><p>A benchmark CRISPRn pooled screen data from Sanson et al.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>CRISPR Pooled Screen Analysis using Beta-Binomial Test</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-07-23</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides functions for hit gene identification and quantification of sgRNA (single-guided RNA) abundances for CRISPR (Clustered Regularly Interspaced Short Palindromic Repeats) pooled screen data analysis. 
  Details are in Jeong et al. (2019) &lt;<a href="https://doi.org/10.1101%2Fgr.245571.118">doi:10.1101/gr.245571.118</a>&gt; and Baggerly et al. (2003) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtg173">doi:10.1093/bioinformatics/btg173</a>&gt;. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.12.16), metap, magrittr, dplyr, tibble, stringr,
ggplot2, tidyr, glue, pheatmap, tools, readr, parallel, R.utils</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-07-23 15:58:35 UTC; hwan</td>
</tr>
<tr>
<td>Author:</td>
<td>Hyun-Hwan Jeong [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hyun-Hwan Jeong &lt;jeong.hyunhwan@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-07-24 09:42:24 UTC</td>
</tr>
</table>
<hr>
<h2 id='calc_mappability'>A function to calculate the mappabilities of each NGS sample.</h2><span id='topic+calc_mappability'></span>

<h3>Description</h3>

<p>A function to calculate the mappabilities of each NGS sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_mappability(count_obj, df_design)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_mappability_+3A_count_obj">count_obj</code></td>
<td>
<p>A list object is created by 'run_sgrna_quant'.</p>
</td></tr>
<tr><td><code id="calc_mappability_+3A_df_design">df_design</code></td>
<td>
<p>The table contains a study design.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>library(CB2)
library(magrittr)
library(tibble)
library(dplyr)
library(glue)
FASTA &lt;- system.file("extdata", "toydata", "small_sample.fasta", package = "CB2")
ex_path &lt;- system.file("extdata", "toydata", package = "CB2")

df_design &lt;- tribble(
  ~group, ~sample_name,
  "Base", "Base1",  
  "Base", "Base2", 
  "High", "High1",
  "High", "High2") %&gt;% 
    mutate(fastq_path = glue("{ex_path}/{sample_name}.fastq"))

cb2_count &lt;- run_sgrna_quant(FASTA, df_design)
calc_mappability(cb2_count, df_design)

</code></pre>

<hr>
<h2 id='Evers_CRISPRn_RT112'>A benchmark CRISPRn pooled screen data from Evers et al.</h2><span id='topic+Evers_CRISPRn_RT112'></span>

<h3>Description</h3>

<p>A benchmark CRISPRn pooled screen data from Evers et al.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Evers_CRISPRn_RT112)
</code></pre>


<h3>Format</h3>

<p>The data object is a list and contains below information:
</p>

<dl>
<dt>count</dt><dd><p>The count matrix from Evers et al.'s paper and contains the CRISPRn screening result using RT112 cell-line.
It contains three different replicates for T0 (before) and contains different three replicates for T1 (after).</p>
</dd>
<dt>egenes</dt><dd><p>The list of 46 essential genes used in Evers et al.'s study.</p>
</dd>
<dt>ngenes</dt><dd><p>The list of 47 non-essential genes used in Evers et al.'s study.</p>
</dd>
<dt>design</dt><dd><p>The data.frame contains study design.</p>
</dd>
<dt>sg_stat</dt><dd><p>The data.frame contains the sgRNA-level statistics.</p>
</dd>
<dt>gene_stat</dt><dd><p>The data.frame contains the gene-level statistics.</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/pubmed/27111720">https://www.ncbi.nlm.nih.gov/pubmed/27111720</a>
</p>

<hr>
<h2 id='fit_ab'>A C++ function to perform a parameter estimation for the sgRNA-level test. 
It will estimate two different parameters 'phat' and 'vhat,'
and we assume input count data follows the beta-binomial distribution.
Dr. Keith Baggerly initially implemented this code in Matlab,
and it has been rewritten it in C++ for the speed-up.</h2><span id='topic+fit_ab'></span>

<h3>Description</h3>

<p>A C++ function to perform a parameter estimation for the sgRNA-level test. 
It will estimate two different parameters 'phat' and 'vhat,'
and we assume input count data follows the beta-binomial distribution.
Dr. Keith Baggerly initially implemented this code in Matlab,
and it has been rewritten it in C++ for the speed-up.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_ab(xvec, nvec)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_ab_+3A_xvec">xvec</code></td>
<td>
<p>a matrix contains sgRNA read counts.</p>
</td></tr>
<tr><td><code id="fit_ab_+3A_nvec">nvec</code></td>
<td>
<p>a vector contains the library size.</p>
</td></tr>
</table>

<hr>
<h2 id='get_CPM'>A function to normalize sgRNA read counts.</h2><span id='topic+get_CPM'></span>

<h3>Description</h3>

<p>A function to normalize sgRNA read counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_CPM(sgcount)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_CPM_+3A_sgcount">sgcount</code></td>
<td>
<p>The input table contains read counts of sgRNAs for each sample
</p>
<p>A function to calculate the CPM (Counts Per Million) (required)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a normalized CPM table will be returned
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(CB2)
data(Evers_CRISPRn_RT112)
get_CPM(Evers_CRISPRn_RT112$count)

</code></pre>

<hr>
<h2 id='join_count_and_design'>A function to join a count table and a design table.</h2><span id='topic+join_count_and_design'></span>

<h3>Description</h3>

<p>A function to join a count table and a design table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>join_count_and_design(sgcount, df_design)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="join_count_and_design_+3A_sgcount">sgcount</code></td>
<td>
<p>The input matrix contains read counts of sgRNAs for each sample.</p>
</td></tr>
<tr><td><code id="join_count_and_design_+3A_df_design">df_design</code></td>
<td>
<p>The table contains a study design.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tall-thin and combined table of the sgRNA read counts and study design will be returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(CB2)
data(Evers_CRISPRn_RT112) 
head(join_count_and_design(Evers_CRISPRn_RT112$count, Evers_CRISPRn_RT112$design))

</code></pre>

<hr>
<h2 id='measure_gene_stats'>A function to perform gene-level test using a sgRNA-level statistics.</h2><span id='topic+measure_gene_stats'></span>

<h3>Description</h3>

<p>A function to perform gene-level test using a sgRNA-level statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>measure_gene_stats(sgrna_stat, logFC_level = "sgRNA")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="measure_gene_stats_+3A_sgrna_stat">sgrna_stat</code></td>
<td>
<p>A data frame created by &lsquo;measure_sgrna_stats&rsquo;</p>
</td></tr>
<tr><td><code id="measure_gene_stats_+3A_logfc_level">logFC_level</code></td>
<td>
<p>The level of &lsquo;logFC&rsquo; value. It can be &lsquo;gene&rsquo; or &lsquo;sgRNA&rsquo;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table contains the gene-level test result, and the table contains these columns: 
</p>

<ul>
<li><p> &lsquo;gene&rsquo;: Theg gene name to be tested.
</p>
</li>
<li><p> &lsquo;n_sgrna&rsquo;: The number of sgRNA targets the gene in the library.
</p>
</li>
<li><p> &lsquo;cpm_a&rsquo;: The mean of CPM of sgRNAs within the first group.
</p>
</li>
<li><p> &lsquo;cpm_b&rsquo;: The mean of CPM of sgRNAs within the second group.
</p>
</li>
<li><p> &lsquo;logFC&rsquo;: The log fold change of the gene between two groups. Taking the mean of sgRNA &lsquo;logFC&rsquo;s is default, and &lsquo;logFC' is calculated by 'log2(cpm_b+1) - log2(cpm_a+1)&rsquo; if &lsquo;logFC_level&rsquo; parameter is set to &lsquo;gene&rsquo;.
</p>
</li>
<li><p> &lsquo;p_ts&rsquo;: The p-value indicates a difference between the two groups at the gene-level.
</p>
</li>
<li><p> &lsquo;p_pa&rsquo;: The p-value indicates enrichment of the first group at the gene-level.
</p>
</li>
<li><p> &lsquo;p_pb&rsquo;: The p-value indicates enrichment of the second group at the gene-level.
</p>
</li>
<li><p> &lsquo;fdr_ts&rsquo;: The adjusted P-value of &lsquo;p_ts&rsquo;.
</p>
</li>
<li><p> &lsquo;fdr_pa&rsquo;: The adjusted P-value of &lsquo;p_pa&rsquo;.
</p>
</li>
<li><p> &lsquo;fdr_pb&rsquo;: The adjusted P-value of &lsquo;p_pb&rsquo;.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>data(Evers_CRISPRn_RT112)
measure_gene_stats(Evers_CRISPRn_RT112$sg_stat)

</code></pre>

<hr>
<h2 id='measure_sgrna_stats'>A function to perform a statistical test at a sgRNA-level</h2><span id='topic+measure_sgrna_stats'></span>

<h3>Description</h3>

<p>A function to perform a statistical test at a sgRNA-level
</p>


<h3>Usage</h3>

<pre><code class='language-R'>measure_sgrna_stats(
  sgcount,
  design,
  group_a,
  group_b,
  delim = "_",
  ge_id = NULL,
  sg_id = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="measure_sgrna_stats_+3A_sgcount">sgcount</code></td>
<td>
<p>This data frame contains read counts of sgRNAs for the samples.</p>
</td></tr>
<tr><td><code id="measure_sgrna_stats_+3A_design">design</code></td>
<td>
<p>This table contains study design. It has to contain 'group.'</p>
</td></tr>
<tr><td><code id="measure_sgrna_stats_+3A_group_a">group_a</code></td>
<td>
<p>The first group to be tested.</p>
</td></tr>
<tr><td><code id="measure_sgrna_stats_+3A_group_b">group_b</code></td>
<td>
<p>The second group to be tested.</p>
</td></tr>
<tr><td><code id="measure_sgrna_stats_+3A_delim">delim</code></td>
<td>
<p>The delimiter between a gene name and a sgRNA ID. It will be used if only rownames contains sgRNA ID.</p>
</td></tr>
<tr><td><code id="measure_sgrna_stats_+3A_ge_id">ge_id</code></td>
<td>
<p>The column name of the gene column.</p>
</td></tr>
<tr><td><code id="measure_sgrna_stats_+3A_sg_id">sg_id</code></td>
<td>
<p>The column/columns of sgRNA identifiers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table contains the sgRNA-level test result, and the table contains these columns: 
</p>

<ul>
<li><p> &lsquo;sgRNA&rsquo;: The sgRNA identifier.
</p>
</li>
<li><p> &lsquo;gene&rsquo;: The gene is the target of the sgRNA 
</p>
</li>
<li><p> &lsquo;n_a&rsquo;: The number of replicates of the first group.
</p>
</li>
<li><p> &lsquo;n_b&rsquo;: The number of replicates of the second group.
</p>
</li>
<li><p> &lsquo;phat_a&rsquo;: The proportion value of the sgRNA for the first group.
</p>
</li>
<li><p> &lsquo;phat_b&rsquo;: The proportion value of the sgRNA for the second group.
</p>
</li>
<li><p> &lsquo;vhat_a&rsquo;: The variance of the sgRNA for the first group.
</p>
</li>
<li><p> &lsquo;vhat_b&rsquo;: The variance of the sgRNA for the second group.
</p>
</li>
<li><p> &lsquo;cpm_a&rsquo;: The mean CPM of the sgRNA within the first group.
</p>
</li>
<li><p> &lsquo;cpm_b&rsquo;: The mean CPM of the sgRNA within the second group.
</p>
</li>
<li><p> &lsquo;logFC&rsquo;: The log fold change of sgRNA between two groups.
</p>
</li>
<li><p> &lsquo;t_value&rsquo;: The value for the t-statistics.
</p>
</li>
<li><p> &lsquo;df&rsquo;: The value of the degree of freedom, and will be used to calculate the p-value of the sgRNA.
</p>
</li>
<li><p> &lsquo;p_ts&rsquo;: The p-value indicates a difference between the two groups.
</p>
</li>
<li><p> &lsquo;p_pa&rsquo;: The p-value indicates enrichment of the first group.
</p>
</li>
<li><p> &lsquo;p_pb&rsquo;: The p-value indicates enrichment of the second group.
</p>
</li>
<li><p> &lsquo;fdr_ts&rsquo;: The adjusted P-value of &lsquo;p_ts&rsquo;.
</p>
</li>
<li><p> &lsquo;fdr_pa&rsquo;: The adjusted P-value of &lsquo;p_pa&rsquo;.
</p>
</li>
<li><p> &lsquo;fdr_pb&rsquo;: The adjusted P-value of &lsquo;p_pb&rsquo;.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>library(CB2)
data(Evers_CRISPRn_RT112)
measure_sgrna_stats(Evers_CRISPRn_RT112$count, Evers_CRISPRn_RT112$design, "before", "after")

</code></pre>

<hr>
<h2 id='plot_corr_heatmap'>A function to show a heatmap sgRNA-level corrleations of the NGS samples.</h2><span id='topic+plot_corr_heatmap'></span>

<h3>Description</h3>

<p>A function to show a heatmap sgRNA-level corrleations of the NGS samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_corr_heatmap(sgcount, df_design, cor_method = "pearson")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_corr_heatmap_+3A_sgcount">sgcount</code></td>
<td>
<p>The input matrix contains read counts of sgRNAs for each sample.</p>
</td></tr>
<tr><td><code id="plot_corr_heatmap_+3A_df_design">df_design</code></td>
<td>
<p>The table contains a study design.</p>
</td></tr>
<tr><td><code id="plot_corr_heatmap_+3A_cor_method">cor_method</code></td>
<td>
<p>A string parameter of the correlation measure. One of the three - &quot;pearson&quot;, &quot;kendall&quot;, or &quot;spearman&quot; will be the string.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A pheatmap object contains the correlation heatmap
</p>
<p>library(CB2)
data(Evers_CRISPRn_RT112)
plot_corr_heatmap(Evers_CRISPRn_RT112$count, Evers_CRISPRn_RT112$design)
</p>

<hr>
<h2 id='plot_count_distribution'>A function to plot read count distribution.</h2><span id='topic+plot_count_distribution'></span>

<h3>Description</h3>

<p>A function to plot read count distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_count_distribution(sgcount, df_design, add_dots = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_count_distribution_+3A_sgcount">sgcount</code></td>
<td>
<p>The input matrix contains read counts of sgRNAs for each sample.</p>
</td></tr>
<tr><td><code id="plot_count_distribution_+3A_df_design">df_design</code></td>
<td>
<p>The table contains a study design.</p>
</td></tr>
<tr><td><code id="plot_count_distribution_+3A_add_dots">add_dots</code></td>
<td>
<p>The function will display dots of sgRNA counts if it is set to 'TRUE'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object contains a read count distribution plot for 'sgcount'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(CB2)
data(Evers_CRISPRn_RT112)
cpm &lt;- get_CPM(Evers_CRISPRn_RT112$count)
plot_count_distribution(cpm, Evers_CRISPRn_RT112$design)

</code></pre>

<hr>
<h2 id='plot_dotplot'>A function to visualize dot plots for a gene.</h2><span id='topic+plot_dotplot'></span>

<h3>Description</h3>

<p>A function to visualize dot plots for a gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_dotplot(sgcount, df_design, gene, ge_id = NULL, sg_id = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_dotplot_+3A_sgcount">sgcount</code></td>
<td>
<p>The input matrix contains read counts of sgRNAs for each sample.</p>
</td></tr>
<tr><td><code id="plot_dotplot_+3A_df_design">df_design</code></td>
<td>
<p>The table contains a study design.</p>
</td></tr>
<tr><td><code id="plot_dotplot_+3A_gene">gene</code></td>
<td>
<p>The gene to be shown.</p>
</td></tr>
<tr><td><code id="plot_dotplot_+3A_ge_id">ge_id</code></td>
<td>
<p>A name of the column contains gene names.</p>
</td></tr>
<tr><td><code id="plot_dotplot_+3A_sg_id">sg_id</code></td>
<td>
<p>A name of the column contains sgRNA IDs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object contains dot plots of sgRNA read counts for a gene.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(CB2)
data(Evers_CRISPRn_RT112)
plot_dotplot(get_CPM(Evers_CRISPRn_RT112$count), Evers_CRISPRn_RT112$design, "RPS7")

</code></pre>

<hr>
<h2 id='plot_PCA'>A function to plot the first two principal components of samples.</h2><span id='topic+plot_PCA'></span>

<h3>Description</h3>

<p>This function will perform a principal component analysis, and it returns a ggplot object of the PCA plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_PCA(sgcount, df_design)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_PCA_+3A_sgcount">sgcount</code></td>
<td>
<p>The input matrix contains read counts of sgRNAs for each sample.</p>
</td></tr>
<tr><td><code id="plot_PCA_+3A_df_design">df_design</code></td>
<td>
<p>The table contains a study design.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object contains a PCA plot for the input.
</p>
<p>library(CB2)
data(Evers_CRISPRn_RT112)
plot_PCA(Evers_CRISPRn_RT112$count, Evers_CRISPRn_RT112$design)
</p>

<hr>
<h2 id='quant'>A C++ function to quantify sgRNA abundance from NGS samples.</h2><span id='topic+quant'></span>

<h3>Description</h3>

<p>A C++ function to quantify sgRNA abundance from NGS samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>quant(ref_path, fastq_path, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="quant_+3A_ref_path">ref_path</code></td>
<td>
<p>the path of the annotation file and it has to be a FASTA formatted file.</p>
</td></tr>
<tr><td><code id="quant_+3A_fastq_path">fastq_path</code></td>
<td>
<p>a list of the FASTQ files.</p>
</td></tr>
<tr><td><code id="quant_+3A_verbose">verbose</code></td>
<td>
<p>Display some logs during the quantification if it is set to 'true'.</p>
</td></tr>
</table>

<hr>
<h2 id='run_estimation'>A function to perform a statistical test at a sgRNA-level, deprecated.</h2><span id='topic+run_estimation'></span>

<h3>Description</h3>

<p>A function to perform a statistical test at a sgRNA-level, deprecated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_estimation(
  sgcount,
  design,
  group_a,
  group_b,
  delim = "_",
  ge_id = NULL,
  sg_id = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_estimation_+3A_sgcount">sgcount</code></td>
<td>
<p>This data frame contains read counts of sgRNAs for the samples.</p>
</td></tr>
<tr><td><code id="run_estimation_+3A_design">design</code></td>
<td>
<p>This table contains study design. It has to contain 'group.'</p>
</td></tr>
<tr><td><code id="run_estimation_+3A_group_a">group_a</code></td>
<td>
<p>The first group to be tested.</p>
</td></tr>
<tr><td><code id="run_estimation_+3A_group_b">group_b</code></td>
<td>
<p>The second group to be tested.</p>
</td></tr>
<tr><td><code id="run_estimation_+3A_delim">delim</code></td>
<td>
<p>The delimiter between a gene name and a sgRNA ID. It will be used if only rownames contains sgRNA ID.</p>
</td></tr>
<tr><td><code id="run_estimation_+3A_ge_id">ge_id</code></td>
<td>
<p>The column name of the gene column.</p>
</td></tr>
<tr><td><code id="run_estimation_+3A_sg_id">sg_id</code></td>
<td>
<p>The column/columns of sgRNA identifiers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table contains the sgRNA-level test result, and the table contains these columns: 
</p>

<ul>
<li><p> &lsquo;sgRNA&rsquo;: The sgRNA identifier.
</p>
</li>
<li><p> &lsquo;gene&rsquo;: The gene is the target of the sgRNA 
</p>
</li>
<li><p> &lsquo;n_a&rsquo;: The number of replicates of the first group.
</p>
</li>
<li><p> &lsquo;n_b&rsquo;: The number of replicates of the second group.
</p>
</li>
<li><p> &lsquo;phat_a&rsquo;: The proportion value of the sgRNA for the first group.
</p>
</li>
<li><p> &lsquo;phat_b&rsquo;: The proportion value of the sgRNA for the second group.
</p>
</li>
<li><p> &lsquo;vhat_a&rsquo;: The variance of the sgRNA for the first group.
</p>
</li>
<li><p> &lsquo;vhat_b&rsquo;: The variance of the sgRNA for the second group.
</p>
</li>
<li><p> &lsquo;cpm_a&rsquo;: The mean CPM of the sgRNA within the first group.
</p>
</li>
<li><p> &lsquo;cpm_b&rsquo;: The mean CPM of the sgRNA within the second group.
</p>
</li>
<li><p> &lsquo;logFC&rsquo;: The log fold change of sgRNA between two groups.
</p>
</li>
<li><p> &lsquo;t_value&rsquo;: The value for the t-statistics.
</p>
</li>
<li><p> &lsquo;df&rsquo;: The value of the degree of freedom, and will be used to calculate the p-value of the sgRNA.
</p>
</li>
<li><p> &lsquo;p_ts&rsquo;: The p-value indicates a difference between the two groups.
</p>
</li>
<li><p> &lsquo;p_pa&rsquo;: The p-value indicates enrichment of the first group.
</p>
</li>
<li><p> &lsquo;p_pb&rsquo;: The p-value indicates enrichment of the second group.
</p>
</li>
<li><p> &lsquo;fdr_ts&rsquo;: The adjusted P-value of &lsquo;p_ts&rsquo;.
</p>
</li>
<li><p> &lsquo;fdr_pa&rsquo;: The adjusted P-value of &lsquo;p_pa&rsquo;.
</p>
</li>
<li><p> &lsquo;fdr_pb&rsquo;: The adjusted P-value of &lsquo;p_pb&rsquo;.
</p>
</li></ul>


<hr>
<h2 id='run_sgrna_quant'>A function to run a sgRNA quantification algorithm from NGS sample</h2><span id='topic+run_sgrna_quant'></span>

<h3>Description</h3>

<p>A function to run a sgRNA quantification algorithm from NGS sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_sgrna_quant(lib_path, design, map_path = NULL, ncores = 1, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_sgrna_quant_+3A_lib_path">lib_path</code></td>
<td>
<p>The path of the FASTA file.</p>
</td></tr>
<tr><td><code id="run_sgrna_quant_+3A_design">design</code></td>
<td>
<p>A table contains the study design. It must contain 'fastq_path' and 'sample_name.'</p>
</td></tr>
<tr><td><code id="run_sgrna_quant_+3A_map_path">map_path</code></td>
<td>
<p>The path of file contains gene-sgRNA mapping.</p>
</td></tr>
<tr><td><code id="run_sgrna_quant_+3A_ncores">ncores</code></td>
<td>
<p>The number that indicates how many processors will be used with a parallelization. 
The parallelization will be enabled if users do not set the parameter as '-1&ldquo; 
(it means the full physical cores will be used) or greater than '1'.</p>
</td></tr>
<tr><td><code id="run_sgrna_quant_+3A_verbose">verbose</code></td>
<td>
<p>Display some logs during the quantification if it is set to 'TRUE'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It will return a list, and the list contains three elements. 
The first element (&lsquo;count&rsquo;) is a data frame contains the result of the quantification for each sample. 
The second element (&lsquo;total&rsquo;) is a numeric vector contains the total number of reads of each sample.
The last element (&lsquo;sequence&rsquo;) a data frame contains the sequence of each sgRNA in the library.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(CB2)
library(magrittr)
library(tibble)
library(dplyr)
library(glue)
FASTA &lt;- system.file("extdata", "toydata", "small_sample.fasta", package = "CB2")
ex_path &lt;- system.file("extdata", "toydata", package = "CB2")

df_design &lt;- tribble(
  ~group, ~sample_name,
  "Base", "Base1",  
  "Base", "Base2", 
  "High", "High1",
  "High", "High2") %&gt;% 
    mutate(fastq_path = glue("{ex_path}/{sample_name}.fastq"))

cb2_count &lt;- run_sgrna_quant(FASTA, df_design)

</code></pre>

<hr>
<h2 id='Sanson_CRISPRn_A375'>A benchmark CRISPRn pooled screen data from Sanson et al.</h2><span id='topic+Sanson_CRISPRn_A375'></span>

<h3>Description</h3>

<p>A benchmark CRISPRn pooled screen data from Sanson et al.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Sanson_CRISPRn_A375)
</code></pre>


<h3>Format</h3>

<p>The data object is a list and contains below information:
</p>

<dl>
<dt>count</dt><dd><p>The count matrix from Sanson et al.'s paper and contains the CRISPRn screening result using A375 cell-line. 
It contains a sample of plasimd, and three biological replicates after three weeks.</p>
</dd>
<dt>egenes</dt><dd><p>The list of 1,580 essential genes used in Sanson et al.'s study.</p>
</dd>
<dt>ngenes</dt><dd><p>The list of 927 non-essential genes used in Sanson et al.'s study.</p>
</dd>
<dt>design</dt><dd><p>The data.frame contains study design.</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/pubmed/30575746">https://www.ncbi.nlm.nih.gov/pubmed/30575746</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
