<!DOCTYPE html><html><head><title>Help for package cbioportalR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {cbioportalR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cbioportalR-package'><p>cbioportalR: Browse and Query Clinical and Genomic Data from cBioPortal</p></a></li>
<li><a href='#.get_clinical_by_list_item'><p>Get clinical data by attribute, study ID and sample ID</p></a></li>
<li><a href='#.get_clinical_pat_by_list_item'><p>Get clinical data by attribute, study ID and patient ID</p></a></li>
<li><a href='#.get_data_by_sample'><p>Internal Function to Get Mutations/CNA/Fusion By Sample ID</p></a></li>
<li><a href='#.get_data_by_study'><p>Internal Function to Get Mutations/CNA/Fusion By Study ID</p></a></li>
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#available_clinical_attributes'><p>Get all available clinical attribute IDs for a study</p></a></li>
<li><a href='#available_gene_panels'><p>Get Available Gene Panels For a Database</p></a></li>
<li><a href='#available_patients'><p>Get All Patient IDs in a Study</p></a></li>
<li><a href='#available_profiles'><p>Get Available Genomic Profiles For a Study or Database</p></a></li>
<li><a href='#available_sample_lists'><p>Get All Sample Lists Available For a Study</p></a></li>
<li><a href='#available_samples'><p>Get All Sample IDs in a Study</p></a></li>
<li><a href='#available_studies'><p>Get Metadata on All Available Studies in a Database</p></a></li>
<li><a href='#cbp_api'><p>Core function to retrieve data from cBioPortal API.</p></a></li>
<li><a href='#get_alias'><p>Get Gene Name Alias for a Given Hugo Symbol</p></a></li>
<li><a href='#get_cbioportal_token'><p>Get cBioPortal Access Token</p></a></li>
<li><a href='#get_clinical_by_patient'><p>Get clinical data by attribute, study ID and patient ID</p></a></li>
<li><a href='#get_clinical_by_sample'><p>Get clinical data by attribute, study ID and sample ID</p></a></li>
<li><a href='#get_clinical_by_study'><p>Get all available clinical data for a specified study</p></a></li>
<li><a href='#get_cna_by_sample'><p>Get CNA By Sample ID</p></a></li>
<li><a href='#get_cna_by_study'><p>Get CNA By Study</p></a></li>
<li><a href='#get_entrez_id'><p>Get Entrez Gene ID for a given set of Hugo Symbols</p></a></li>
<li><a href='#get_fusions_by_sample'><p>Get Fusions By Sample ID</p></a></li>
<li><a href='#get_fusions_by_study'><p>Get Fusions By Study</p></a></li>
<li><a href='#get_gene_panel'><p>Retrieve Genes Included For a Specified Panel ID</p></a></li>
<li><a href='#get_genes'><p>Get A List of Genes for a Specified Database</p></a></li>
<li><a href='#get_genetics_by_sample'><p>Get All Genomic Information By Sample IDs</p></a></li>
<li><a href='#get_genetics_by_study'><p>Get All Genomic Information By Study</p></a></li>
<li><a href='#get_hugo_symbol'><p>Get Hugo Symbol for a given set of Entrez IDs</p></a></li>
<li><a href='#get_mutations_by_sample'><p>Get Mutations By Sample ID</p></a></li>
<li><a href='#get_mutations_by_study'><p>Get Mutations By Study ID</p></a></li>
<li><a href='#get_panel_by_sample'><p>Get Gene Panel by study ID and sample ID</p></a></li>
<li><a href='#get_samples_by_patient'><p>Get sample IDs for a given set of patient IDs</p></a></li>
<li><a href='#get_segments_by_sample'><p>Get Copy Number Segmentation Data By Sample ID</p></a></li>
<li><a href='#get_segments_by_study'><p>Get Copy Number Segmentation Data By Study</p></a></li>
<li><a href='#get_study_info'><p>Get Metadata on All Available Studies in Database or a Specified Study</p></a></li>
<li><a href='#impact_gene_info'><p>IMPACT Gene Meta Data</p></a></li>
<li><a href='#null-default'><p>Default value for <code>NULL</code>.</p></a></li>
<li><a href='#set_cbioportal_db'><p>Connect to cBioPortal DB</p></a></li>
<li><a href='#test_cbioportal_db'><p>Test the Database Connection Anytime During your R Session</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Browse and Query Clinical and Genomic Data from cBioPortal</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides R users with direct access to genomic and clinical data from the 'cBioPortal' web resource via user-friendly
    functions that wrap 'cBioPortal's' existing API endpoints <a href="https://www.cbioportal.org/api/swagger-ui/index.html">https://www.cbioportal.org/api/swagger-ui/index.html</a>. Users can browse and query genomic data on mutations,
    copy number alterations and fusions, as well as data on tumor mutational burden ('TMB'),
    microsatellite instability status ('MSI'), 'FACETS' and select clinical data points (depending on the study). 
    See <a href="https://www.cbioportal.org/">https://www.cbioportal.org/</a> and Gao et al., (2013) &lt;<a href="https://doi.org/10.1126%2Fscisignal.2004088">doi:10.1126/scisignal.2004088</a>&gt; for more information on the cBioPortal web resource.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.1.4), knitr (&ge; 1.39), rmarkdown (&ge; 2.14),
covr (&ge; 3.5.1), spelling (&ge; 2.2)</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>httr (&ge; 1.4.3), tibble (&ge; 3.1.7), purrr (&ge; 0.3.4), magrittr
(&ge; 2.0.3), rlang (&ge; 1.0.3), glue (&ge; 1.6.2), jsonlite (&ge;
1.8.0), tidyr (&ge; 1.2.0), dplyr (&ge; 1.0.9), stringr (&ge; 1.4.0),
cli (&ge; 3.3.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/karissawhiting/cbioportalR">https://github.com/karissawhiting/cbioportalR</a>,
<a href="https://www.karissawhiting.com/cbioportalR/">https://www.karissawhiting.com/cbioportalR/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/karissawhiting/cbioportalR/issues">https://github.com/karissawhiting/cbioportalR/issues</a></td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-05 15:52:47 UTC; kwhiting</td>
</tr>
<tr>
<td>Author:</td>
<td>Karissa Whiting <a href="https://orcid.org/0000-0002-4683-1868"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Karissa Whiting &lt;karissa.whiting@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-05 18:50:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='cbioportalR-package'>cbioportalR: Browse and Query Clinical and Genomic Data from cBioPortal</h2><span id='topic+cbioportalR'></span><span id='topic+cbioportalR-package'></span>

<h3>Description</h3>

<p><img src="../help/figures/logo.png" style='float: right' alt='logo' width='120' />
</p>
<p>Provides R users with direct access to genomic and clinical data from the 'cBioPortal' web resource via user-friendly functions that wrap 'cBioPortal's' existing API endpoints <a href="https://www.cbioportal.org/api/swagger-ui/index.html">https://www.cbioportal.org/api/swagger-ui/index.html</a>. Users can browse and query genomic data on mutations, copy number alterations and fusions, as well as data on tumor mutational burden ('TMB'), microsatellite instability status ('MSI'), 'FACETS' and select clinical data points (depending on the study). See <a href="https://www.cbioportal.org/">https://www.cbioportal.org/</a> and Gao et al., (2013) <a href="https://doi.org/10.1126/scisignal.2004088">doi:10.1126/scisignal.2004088</a> for more information on the cBioPortal web resource.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Karissa Whiting <a href="mailto:karissa.whiting@gmail.com">karissa.whiting@gmail.com</a> (<a href="https://orcid.org/0000-0002-4683-1868">ORCID</a>)
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/karissawhiting/cbioportalR">https://github.com/karissawhiting/cbioportalR</a>
</p>
</li>
<li> <p><a href="https://www.karissawhiting.com/cbioportalR/">https://www.karissawhiting.com/cbioportalR/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/karissawhiting/cbioportalR/issues">https://github.com/karissawhiting/cbioportalR/issues</a>
</p>
</li></ul>


<hr>
<h2 id='.get_clinical_by_list_item'>Get clinical data by attribute, study ID and sample ID</h2><span id='topic+.get_clinical_by_list_item'></span>

<h3>Description</h3>

<p>Get clinical data by attribute, study ID and sample ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.get_clinical_by_list_item(
  study_id = NULL,
  sample_id = NULL,
  clinical_attribute = NULL,
  base_url = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".get_clinical_by_list_item_+3A_study_id">study_id</code></td>
<td>
<p>A string indicating the study ID from which to pull data. If no study ID, will
guess the study ID based on your URL and inform. Only 1 study ID can be passed. If mutations/cna from
more than 1 study needed, see <code>sample_study_pairs</code></p>
</td></tr>
<tr><td><code id=".get_clinical_by_list_item_+3A_sample_id">sample_id</code></td>
<td>
<p>a vector of sample IDs (character)</p>
</td></tr>
<tr><td><code id=".get_clinical_by_list_item_+3A_clinical_attribute">clinical_attribute</code></td>
<td>
<p>one or more clinical attributes for your study.
If none provided, will return all attributes available for studies</p>
</td></tr>
<tr><td><code id=".get_clinical_by_list_item_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of a specific clinical attribute
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
.get_clinical_by_list_item(study_id = "acc_tcga", sample_id = "TCGA-OR-A5J2-01",
 clinical_attribute = "CANCER_TYPE", base_url = 'www.cbioportal.org/api')
 
## End(Not run)

</code></pre>

<hr>
<h2 id='.get_clinical_pat_by_list_item'>Get clinical data by attribute, study ID and patient ID</h2><span id='topic+.get_clinical_pat_by_list_item'></span>

<h3>Description</h3>

<p>Get clinical data by attribute, study ID and patient ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.get_clinical_pat_by_list_item(
  study_id = NULL,
  patient_id = NULL,
  clinical_attribute = NULL,
  base_url = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".get_clinical_pat_by_list_item_+3A_study_id">study_id</code></td>
<td>
<p>A string indicating the study ID from which to pull data. If no study ID, will
guess the study ID based on your URL and inform. Only 1 study ID can be passed. If mutations/cna from
more than 1 study needed, see <code>sample_study_pairs</code></p>
</td></tr>
<tr><td><code id=".get_clinical_pat_by_list_item_+3A_patient_id">patient_id</code></td>
<td>
<p>a cBioPortal patient_id</p>
</td></tr>
<tr><td><code id=".get_clinical_pat_by_list_item_+3A_clinical_attribute">clinical_attribute</code></td>
<td>
<p>one or more clinical attributes for your study.
If none provided, will return all attributes available for studies</p>
</td></tr>
<tr><td><code id=".get_clinical_pat_by_list_item_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of a specific clinical attribute
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
.get_clinical_pat_by_list_item(study_id = "msk_impact_2017",
 patient_id = "P-0001453",
  base_url = 'www.cbioportal.org/api')
  
## End(Not run)

</code></pre>

<hr>
<h2 id='.get_data_by_sample'>Internal Function to Get Mutations/CNA/Fusion By Sample ID</h2><span id='topic+.get_data_by_sample'></span>

<h3>Description</h3>

<p>Internal Function to Get Mutations/CNA/Fusion By Sample ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.get_data_by_sample(
  sample_id = NULL,
  study_id = NULL,
  molecular_profile_id = NULL,
  sample_study_pairs = NULL,
  data_type = c("mutation", "cna", "fusion", "structural_variant", "segment"),
  genes = NULL,
  panel = NULL,
  add_hugo = TRUE,
  base_url = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".get_data_by_sample_+3A_sample_id">sample_id</code></td>
<td>
<p>a vector of sample IDs (character)</p>
</td></tr>
<tr><td><code id=".get_data_by_sample_+3A_study_id">study_id</code></td>
<td>
<p>A string indicating the study ID from which to pull data. If no study ID, will
guess the study ID based on your URL and inform. Only 1 study ID can be passed. If mutations/cna from
more than 1 study needed, see <code>sample_study_pairs</code></p>
</td></tr>
<tr><td><code id=".get_data_by_sample_+3A_molecular_profile_id">molecular_profile_id</code></td>
<td>
<p>A string indicating the molecular profile ID from which to pull data. If ID supplied, will
guess the molecular profile ID based on the study ID. Only 1 molecular profile ID can be passed. If mutations from
more than 1 study needed, see <code>sample_study_pairs</code></p>
</td></tr>
<tr><td><code id=".get_data_by_sample_+3A_sample_study_pairs">sample_study_pairs</code></td>
<td>
<p>A dataframe with columns: <code>sample_id</code>, <code>study_id</code> and <code>molecular_profile_id</code> (optional). Variations in capitalization of column names are accepted.
This can be used in place of <code>sample_id</code>, <code>study_id</code>, <code>molecular_profile_id</code> arguments above if you
need to pull samples from several different studies at once. If passed this will take overwrite <code>sample_id</code>, <code>study_id</code>, <code>molecular_profile_id</code> if also passed.</p>
</td></tr>
<tr><td><code id=".get_data_by_sample_+3A_data_type">data_type</code></td>
<td>
<p>specify what type of data to return. Options are<code>mutations</code>, <code>cna</code>, <code>fusion</code> or <code>structural_variant</code> (same as <code>fusion</code>).</p>
</td></tr>
<tr><td><code id=".get_data_by_sample_+3A_genes">genes</code></td>
<td>
<p>A vector of Entrez ids or Hugo symbols. If Hugo symbols are supplied, they will be converted to entrez ids using the <code>get_entrez_id()</code> function.
If <code>panel</code> and <code>genes</code> are both supplied, genes from both arguments will be returned. If both are NULL (default), it will return gene results for all available genomic data for that sample.</p>
</td></tr>
<tr><td><code id=".get_data_by_sample_+3A_panel">panel</code></td>
<td>
<p>One or more panel IDs to query (e.g. 'IMPACT468').
If <code>panel</code> and <code>genes</code> are both supplied, genes from both arguments will be returned. If both are NULL (default), it will return gene results for all available genomic data for that sample.</p>
</td></tr>
<tr><td><code id=".get_data_by_sample_+3A_add_hugo">add_hugo</code></td>
<td>
<p>Logical indicating whether <code>HugoGeneSymbol</code> should be added to your resulting data frame, if not already present in raw API results.
Argument is <code>TRUE</code> by default. If <code>FALSE</code>, results will be returned as is (i.e. any existing Hugo Symbol columns in raw results will not be removed).</p>
</td></tr>
<tr><td><code id=".get_data_by_sample_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of mutations, CNAs or fusions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set_cbioportal_db("public")
.get_data_by_sample(sample_id = c("TCGA-OR-A5J2-01","TCGA-OR-A5J6-01"),
 study_id = "acc_tcga", data_type = "mutation")

.get_data_by_sample(sample_id = c("DS-sig-010-P2"),
 molecular_profile_id = "blca_plasmacytoid_mskcc_2016_cna", data_type = "cna")


.get_data_by_sample(sample_id = c("P-0002146-T01-IM3"),
 study_id = "blca_plasmacytoid_mskcc_2016", data_type = "fusion")

df_pairs &lt;- data.frame(
"sample_id" = c("s_C_36924L_P001_d",
"s_C_03LNU8_P001_d"),
"study_id" = c("prad_msk_2019"))

.get_data_by_sample(sample_study_pairs = df_pairs, data_type = "mutation")
.get_data_by_sample(sample_study_pairs = df_pairs, genes = 7157, data_type = "mutation")
.get_data_by_sample(sample_study_pairs = df_pairs, data_type = "cna")
.get_data_by_sample(sample_study_pairs = df_pairs, data_type = "fusion")

df_pairs2 &lt;- data.frame(
"sample_id" = c("P-0002146-T01-IM3", "s_C_CAUWT7_P001_d"),
 "study_id" = c("blca_plasmacytoid_mskcc_2016", "prad_msk_2019"))

.get_data_by_sample(sample_study_pairs = df_pairs2, data_type = "mutation")
.get_data_by_sample(sample_study_pairs = df_pairs2, genes = 7157)
.get_data_by_sample(sample_study_pairs = df_pairs2, data_type = "cna")
.get_data_by_sample(sample_study_pairs = df_pairs2, data_type = "fusion")

## End(Not run)

</code></pre>

<hr>
<h2 id='.get_data_by_study'>Internal Function to Get Mutations/CNA/Fusion By Study ID</h2><span id='topic+.get_data_by_study'></span>

<h3>Description</h3>

<p>Endpoints for retrieving mutation and cna data are structurally similar.
This internal function allows you to pull data from either endpoint. It has
logic for sensible default guesses at <code>study_id</code> and <code>molecular_profile_id</code> when those are <code>NULL</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.get_data_by_study(
  study_id = NULL,
  molecular_profile_id = NULL,
  data_type = c("mutation", "cna", "fusion", "structural_variant", "segment"),
  base_url = NULL,
  add_hugo = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".get_data_by_study_+3A_study_id">study_id</code></td>
<td>
<p>A study ID to query mutations. If NULL, guesses study ID based
on molecular_profile_id.</p>
</td></tr>
<tr><td><code id=".get_data_by_study_+3A_molecular_profile_id">molecular_profile_id</code></td>
<td>
<p>a molecular profile to query mutations.
If NULL, guesses molecular_profile_id based on study ID.</p>
</td></tr>
<tr><td><code id=".get_data_by_study_+3A_data_type">data_type</code></td>
<td>
<p>specify what type of data to return. Options are<code>mutation</code>, <code>cna</code>, <code>fusion</code>, or<code>structural_variant</code> (same as <code>fusion</code>),
and <code>segment</code> (copy number segmentation data)..</p>
</td></tr>
<tr><td><code id=".get_data_by_study_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
<tr><td><code id=".get_data_by_study_+3A_add_hugo">add_hugo</code></td>
<td>
<p>Logical indicating whether <code>HugoGeneSymbol</code> should be added to your resulting data frame, if not already present in raw API results.
Argument is <code>TRUE</code> by default. If <code>FALSE</code>, results will be returned as is (i.e. any existing Hugo Symbol columns in raw results will not be removed).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of mutations, CNAs or structural variants
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set_cbioportal_db("public")
.get_data_by_study(study_id = "prad_msk_2019", data_type = "cna")
.get_data_by_study(study_id = "prad_msk_2019", data_type = "mutation")
.get_data_by_study(study_id = "prad_msk_2019", data_type = "fusion")

.get_data_by_study(molecular_profile_id = "prad_msk_2019_cna", data_type = "cna")
.get_data_by_study(molecular_profile_id = "prad_msk_2019_mutations", data_type = "mutation")
.get_data_by_study(molecular_profile_id = "prad_msk_2019_structural_variants", data_type = "fusion")

## End(Not run)

</code></pre>

<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling <code>rhs(lhs)</code>.
</p>

<hr>
<h2 id='available_clinical_attributes'>Get all available clinical attribute IDs for a study</h2><span id='topic+available_clinical_attributes'></span>

<h3>Description</h3>

<p>Get all available clinical attribute IDs for a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>available_clinical_attributes(study_id = NULL, base_url = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="available_clinical_attributes_+3A_study_id">study_id</code></td>
<td>
<p>cbioportal study ID</p>
</td></tr>
<tr><td><code id="available_clinical_attributes_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query.
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame of available clinical attributes for that study
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
available_clinical_attributes("acc_tcga", base_url = 'www.cbioportal.org/api')

## End(Not run)

</code></pre>

<hr>
<h2 id='available_gene_panels'>Get Available Gene Panels For a Database</h2><span id='topic+available_gene_panels'></span>

<h3>Description</h3>

<p>Get Available Gene Panels For a Database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>available_gene_panels(base_url = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="available_gene_panels_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query.
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of metadata regarding each available panel
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set_cbioportal_db("public")
available_gene_panels()

## End(Not run)

</code></pre>

<hr>
<h2 id='available_patients'>Get All Patient IDs in a Study</h2><span id='topic+available_patients'></span>

<h3>Description</h3>

<p>Get All Patient IDs in a Study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>available_patients(study_id = NULL, base_url = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="available_patients_+3A_study_id">study_id</code></td>
<td>
<p>A character string indicating which study ID should be searched.
Only 1 study ID allowed.</p>
</td></tr>
<tr><td><code id="available_patients_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of patient_ids in a given study
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set_cbioportal_db("public")
available_samples(study_id = "acc_tcga")

## End(Not run)

</code></pre>

<hr>
<h2 id='available_profiles'>Get Available Genomic Profiles For a Study or Database</h2><span id='topic+available_profiles'></span>

<h3>Description</h3>

<p>Get Available Genomic Profiles For a Study or Database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>available_profiles(study_id = NULL, base_url = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="available_profiles_+3A_study_id">study_id</code></td>
<td>
<p>A character vector of length 1 indicating study_id.
See <code>get_studies()</code> to see all available studies for your URL. If <code>NULL</code>, it will
return all profiles for your current database url</p>
</td></tr>
<tr><td><code id="available_profiles_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query.
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of available genetic profiles and their profile ids
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set_cbioportal_db("public")
available_profiles()
available_profiles(study_id = "acc_tcga")

## End(Not run)
</code></pre>

<hr>
<h2 id='available_sample_lists'>Get All Sample Lists Available For a Study</h2><span id='topic+available_sample_lists'></span>

<h3>Description</h3>

<p>Get All Sample Lists Available For a Study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>available_sample_lists(study_id = NULL, base_url = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="available_sample_lists_+3A_study_id">study_id</code></td>
<td>
<p>A character string indicating which study ID should be searched.
Only 1 study ID allowed.</p>
</td></tr>
<tr><td><code id="available_sample_lists_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of patient_ids in a given study
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set_cbioportal_db("public")
available_sample_lists(study_id = "acc_tcga")

## End(Not run)

</code></pre>

<hr>
<h2 id='available_samples'>Get All Sample IDs in a Study</h2><span id='topic+available_samples'></span>

<h3>Description</h3>

<p>Pulls all available sample IDs for a given study ID or sample list ID.
Either a study ID or sample list ID must be passed. If both <code>sample_list</code> and <code>study_id</code> are not <code>NULL</code>,
<code>sample_list</code> ID will be searched and <code>study_id</code> will be ignored.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>available_samples(study_id = NULL, sample_list_id = NULL, base_url = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="available_samples_+3A_study_id">study_id</code></td>
<td>
<p>A character string indicating which study ID should be searched.
Only 1 study ID allowed.</p>
</td></tr>
<tr><td><code id="available_samples_+3A_sample_list_id">sample_list_id</code></td>
<td>
<p>A character string indicating which sample list ID should be searched.
Only 1 sample list ID allowed.</p>
</td></tr>
<tr><td><code id="available_samples_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of sample_ids in a given study
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set_cbioportal_db("public")
available_samples(study_id = "acc_tcga")
available_samples(sample_list_id = "acc_tcga_cna")

## End(Not run)

</code></pre>

<hr>
<h2 id='available_studies'>Get Metadata on All Available Studies in a Database</h2><span id='topic+available_studies'></span>

<h3>Description</h3>

<p>Get Metadata on All Available Studies in a Database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>available_studies(base_url = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="available_studies_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query.
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of available studies and their metadata
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set_cbioportal_db("public")
available_studies()

## End(Not run)
</code></pre>

<hr>
<h2 id='cbp_api'>Core function to retrieve data from cBioPortal API.</h2><span id='topic+cbp_api'></span>

<h3>Description</h3>

<p>Core function to retrieve data from cBioPortal API.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cbp_api(
  url_path,
  method = NULL,
  token = get_cbioportal_token(),
  body = NULL,
  extra_box = FALSE,
  quiet = TRUE,
  base_url = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cbp_api_+3A_url_path">url_path</code></td>
<td>
<p>The url path for API call</p>
</td></tr>
<tr><td><code id="cbp_api_+3A_method">method</code></td>
<td>
<p>Which API method to use. Must be &quot;get&quot; or &quot;post&quot;. Default is &quot;get&quot;.</p>
</td></tr>
<tr><td><code id="cbp_api_+3A_token">token</code></td>
<td>
<p>Authentication token, if needed. Default is <code>get_cbioportal_token()</code></p>
</td></tr>
<tr><td><code id="cbp_api_+3A_body">body</code></td>
<td>
<p>Arguments passed to API call (e.g. sample ID or gene IDs)</p>
</td></tr>
<tr><td><code id="cbp_api_+3A_extra_box">extra_box</code></td>
<td>
<p>Some functions require an additional list() wrapping around body idk why</p>
</td></tr>
<tr><td><code id="cbp_api_+3A_quiet">quiet</code></td>
<td>
<p>Returns queried URL. Default is TRUE</p>
</td></tr>
<tr><td><code id="cbp_api_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A parsed API response
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
cbp_api(url_path = "genes/TP53", base_url = "public")

## End(Not run)

</code></pre>

<hr>
<h2 id='get_alias'>Get Gene Name Alias for a Given Hugo Symbol</h2><span id='topic+get_alias'></span>

<h3>Description</h3>

<p>This function grabs known gene aliases for a given Hugo Symbol.
You may notice that genes -alias pairs are not always consistent. For example
get_alias(&quot;KMT2D&quot;) will return &quot;MLL2&quot; but get_alias(&quot;MLL2&quot;) will not return &quot;KMT2D&quot;
This function relies on the existing cBioPortal API which controls this database
of aliases. Therefore, this is a convenience function but
you may want to consider a more carefully curated alias list like <code>cbioportalR::impact_gene_info</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_alias(hugo_symbol = NULL, base_url = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_alias_+3A_hugo_symbol">hugo_symbol</code></td>
<td>
<p>a hugo symbol for which to return aliases</p>
</td></tr>
<tr><td><code id="get_alias_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string with all aliases
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

get_alias(hugo_symbol = "FGFR3", base_url = 'www.cbioportal.org/api')
get_alias(hugo_symbol = c("FGFR3", "TP53"), base_url = 'www.cbioportal.org/api')

## End(Not run)

</code></pre>

<hr>
<h2 id='get_cbioportal_token'>Get cBioPortal Access Token</h2><span id='topic+get_cbioportal_token'></span>

<h3>Description</h3>

<p>Convenience function that retrieves cBioPortal token System Environment variable &quot;CBIOPORTAL_TOKEN&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_cbioportal_token()
</code></pre>


<h3>Value</h3>

<p>Returns a string with cBioPortal token if successfully authenticated, or
a warning that token could not be found.
</p>


<h3>Author(s)</h3>

<p>Karissa Whiting, Daniel D. Sjoberg
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_cbioportal_token()

## End(Not run)

</code></pre>

<hr>
<h2 id='get_clinical_by_patient'>Get clinical data by attribute, study ID and patient ID</h2><span id='topic+get_clinical_by_patient'></span>

<h3>Description</h3>

<p>Get clinical data by attribute, study ID and patient ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_clinical_by_patient(
  study_id = NULL,
  patient_id = NULL,
  patient_study_pairs = NULL,
  clinical_attribute = NULL,
  base_url = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_clinical_by_patient_+3A_study_id">study_id</code></td>
<td>
<p>A string indicating the study ID from which to pull data. If no study ID, will
guess the study ID based on your URL and inform. Only 1 study ID can be passed. If mutations/cna from
more than 1 study needed, see <code>sample_study_pairs</code></p>
</td></tr>
<tr><td><code id="get_clinical_by_patient_+3A_patient_id">patient_id</code></td>
<td>
<p>a cBioPortal patient_id</p>
</td></tr>
<tr><td><code id="get_clinical_by_patient_+3A_patient_study_pairs">patient_study_pairs</code></td>
<td>
<p>A dataframe with columns: <code>patient_id</code>, <code>study_id</code>.
Variations in capitalization of column names are accepted.
This can be used in place of <code>patient_id</code>, <code>study_id</code>, arguments above if you
need to pull samples from several different studies at once. If passed, this will take overwrite <code>patient_id</code> and <code>study_id</code> if they are also passed.</p>
</td></tr>
<tr><td><code id="get_clinical_by_patient_+3A_clinical_attribute">clinical_attribute</code></td>
<td>
<p>one or more clinical attributes for your study.
If none provided, will return all attributes available for studies</p>
</td></tr>
<tr><td><code id="get_clinical_by_patient_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of a specific clinical attribute
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

ex &lt;- tibble::tribble(
~patientID, ~study_id,
"P-0001453", "blca_nmibc_2017",
"P-0002166", "blca_nmibc_2017",
"P-0003238", "blca_nmibc_2017",
"P-0000004", "msk_impact_2017",
"P-0000023", "msk_impact_2017")

x &lt;- get_clinical_by_patient(patient_study_pairs = ex,
 clinical_attribute = NULL, base_url = 'www.cbioportal.org/api')
 
## End(Not run)

</code></pre>

<hr>
<h2 id='get_clinical_by_sample'>Get clinical data by attribute, study ID and sample ID</h2><span id='topic+get_clinical_by_sample'></span>

<h3>Description</h3>

<p>Get clinical data by attribute, study ID and sample ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_clinical_by_sample(
  study_id = NULL,
  sample_id = NULL,
  sample_study_pairs = NULL,
  clinical_attribute = NULL,
  base_url = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_clinical_by_sample_+3A_study_id">study_id</code></td>
<td>
<p>A string indicating the study ID from which to pull data. If no study ID, will
guess the study ID based on your URL and inform. Only 1 study ID can be passed. If mutations/cna from
more than 1 study needed, see <code>sample_study_pairs</code></p>
</td></tr>
<tr><td><code id="get_clinical_by_sample_+3A_sample_id">sample_id</code></td>
<td>
<p>a vector of sample IDs (character)</p>
</td></tr>
<tr><td><code id="get_clinical_by_sample_+3A_sample_study_pairs">sample_study_pairs</code></td>
<td>
<p>A dataframe with columns: <code>sample_id</code>, <code>study_id</code> and <code>molecular_profile_id</code> (optional). Variations in capitalization of column names are accepted.
This can be used in place of <code>sample_id</code>, <code>study_id</code>, <code>molecular_profile_id</code> arguments above if you
need to pull samples from several different studies at once. If passed this will take overwrite <code>sample_id</code>, <code>study_id</code>, <code>molecular_profile_id</code> if also passed.</p>
</td></tr>
<tr><td><code id="get_clinical_by_sample_+3A_clinical_attribute">clinical_attribute</code></td>
<td>
<p>one or more clinical attributes for your study.
If none provided, will return all attributes available for studies</p>
</td></tr>
<tr><td><code id="get_clinical_by_sample_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of a specific clinical attribute
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_clinical_by_sample(study_id = "acc_tcga", sample_id = "TCGA-OR-A5J2-01",
 clinical_attribute = "CANCER_TYPE", base_url = 'www.cbioportal.org/api')

ex &lt;- tibble::tribble(
~sample_id, ~study_id,
"P-0001453-T01-IM3", "blca_nmibc_2017",
"P-0002166-T01-IM3", "blca_nmibc_2017",
"P-0003238-T01-IM5", "blca_nmibc_2017",
"P-0000004-T01-IM3", "msk_impact_2017",
"P-0000023-T01-IM3", "msk_impact_2017")

x &lt;- get_clinical_by_sample(sample_study_pairs = ex,
 clinical_attribute = NULL, base_url = 'www.cbioportal.org/api')
 
## End(Not run)
</code></pre>

<hr>
<h2 id='get_clinical_by_study'>Get all available clinical data for a specified study</h2><span id='topic+get_clinical_by_study'></span>

<h3>Description</h3>

<p>Returns all sample-level and patient-level clinical data for a given study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_clinical_by_study(
  study_id = NULL,
  clinical_attribute = NULL,
  base_url = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_clinical_by_study_+3A_study_id">study_id</code></td>
<td>
<p>study ID</p>
</td></tr>
<tr><td><code id="get_clinical_by_study_+3A_clinical_attribute">clinical_attribute</code></td>
<td>
<p>one or more clinical attributes for your study.
If none provided, will return all attributes available for
that study (<code style="white-space: pre;">&#8288;available_clinical_attributes(&lt;study_id&gt;)&#8288;</code>)</p>
</td></tr>
<tr><td><code id="get_clinical_by_study_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query.
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of all available clinical attributes and their values
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_clinical_by_study(study_id = "acc_tcga",
 clinical_attribute = "CANCER_TYPE", base_url = 'www.cbioportal.org/api')

get_clinical_by_study(study_id = "acc_tcga", base_url = 'www.cbioportal.org/api')

## End(Not run)

</code></pre>

<hr>
<h2 id='get_cna_by_sample'>Get CNA By Sample ID</h2><span id='topic+get_cna_by_sample'></span>

<h3>Description</h3>

<p>Get CNA By Sample ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_cna_by_sample(
  sample_id = NULL,
  study_id = NULL,
  molecular_profile_id = NULL,
  sample_study_pairs = NULL,
  genes = NULL,
  panel = NULL,
  add_hugo = TRUE,
  base_url = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_cna_by_sample_+3A_sample_id">sample_id</code></td>
<td>
<p>a vector of sample IDs (character)</p>
</td></tr>
<tr><td><code id="get_cna_by_sample_+3A_study_id">study_id</code></td>
<td>
<p>A string indicating the study ID from which to pull data. If no study ID, will
guess the study ID based on your URL and inform. Only 1 study ID can be passed. If mutations/cna from
more than 1 study needed, see <code>sample_study_pairs</code></p>
</td></tr>
<tr><td><code id="get_cna_by_sample_+3A_molecular_profile_id">molecular_profile_id</code></td>
<td>
<p>A string indicating the molecular profile ID from which to pull data. If ID supplied, will
guess the molecular profile ID based on the study ID. Only 1 molecular profile ID can be passed. If mutations from
more than 1 study needed, see <code>sample_study_pairs</code></p>
</td></tr>
<tr><td><code id="get_cna_by_sample_+3A_sample_study_pairs">sample_study_pairs</code></td>
<td>
<p>A dataframe with columns: <code>sample_id</code>, <code>study_id</code> and <code>molecular_profile_id</code> (optional). Variations in capitalization of column names are accepted.
This can be used in place of <code>sample_id</code>, <code>study_id</code>, <code>molecular_profile_id</code> arguments above if you
need to pull samples from several different studies at once. If passed this will take overwrite <code>sample_id</code>, <code>study_id</code>, <code>molecular_profile_id</code> if also passed.</p>
</td></tr>
<tr><td><code id="get_cna_by_sample_+3A_genes">genes</code></td>
<td>
<p>A vector of Entrez ids or Hugo symbols. If Hugo symbols are supplied, they will be converted to entrez ids using the <code>get_entrez_id()</code> function.
If <code>panel</code> and <code>genes</code> are both supplied, genes from both arguments will be returned. If both are NULL (default), it will return gene results for all available genomic data for that sample.</p>
</td></tr>
<tr><td><code id="get_cna_by_sample_+3A_panel">panel</code></td>
<td>
<p>One or more panel IDs to query (e.g. 'IMPACT468').
If <code>panel</code> and <code>genes</code> are both supplied, genes from both arguments will be returned. If both are NULL (default), it will return gene results for all available genomic data for that sample.</p>
</td></tr>
<tr><td><code id="get_cna_by_sample_+3A_add_hugo">add_hugo</code></td>
<td>
<p>Logical indicating whether <code>HugoGeneSymbol</code> should be added to your resulting data frame, if not already present in raw API results.
Argument is <code>TRUE</code> by default. If <code>FALSE</code>, results will be returned as is (i.e. any existing Hugo Symbol columns in raw results will not be removed).</p>
</td></tr>
<tr><td><code id="get_cna_by_sample_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of CNAs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set_cbioportal_db("public")
get_cna_by_sample(sample_id = c("s_C_36924L_P001_d"),
                 study_id = "prad_msk_2019")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_cna_by_study'>Get CNA By Study</h2><span id='topic+get_cna_by_study'></span>

<h3>Description</h3>

<p>Get CNA By Study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_cna_by_study(
  study_id = NULL,
  molecular_profile_id = NULL,
  add_hugo = TRUE,
  base_url = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_cna_by_study_+3A_study_id">study_id</code></td>
<td>
<p>A study ID to query mutations. If NULL, guesses study ID based
on molecular_profile_id.</p>
</td></tr>
<tr><td><code id="get_cna_by_study_+3A_molecular_profile_id">molecular_profile_id</code></td>
<td>
<p>a molecular profile to query mutations.
If NULL, guesses molecular_profile_id based on study ID.</p>
</td></tr>
<tr><td><code id="get_cna_by_study_+3A_add_hugo">add_hugo</code></td>
<td>
<p>Logical indicating whether <code>HugoGeneSymbol</code> should be added to your resulting data frame, if not already present in raw API results.
Argument is <code>TRUE</code> by default. If <code>FALSE</code>, results will be returned as is (i.e. any existing Hugo Symbol columns in raw results will not be removed).</p>
</td></tr>
<tr><td><code id="get_cna_by_study_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of CNAs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_cna_by_study(study_id = "prad_msk_2019")
get_cna_by_study(molecular_profile_id = "prad_msk_2019_cna")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_entrez_id'>Get Entrez Gene ID for a given set of Hugo Symbols</h2><span id='topic+get_entrez_id'></span>

<h3>Description</h3>

<p>Get Entrez Gene ID for a given set of Hugo Symbols
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_entrez_id(hugo_symbol = NULL, base_url = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_entrez_id_+3A_hugo_symbol">hugo_symbol</code></td>
<td>
<p>a character vector of Hugo Symbols</p>
</td></tr>
<tr><td><code id="get_entrez_id_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with Entrez Gene IDs and Hugo Symbols
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_entrez_id(hugo_symbol = "TAP1", base_url = 'www.cbioportal.org/api')
get_entrez_id(hugo_symbol = c("FGFR1", "TP53") , base_url = 'www.cbioportal.org/api')

## End(Not run)
</code></pre>

<hr>
<h2 id='get_fusions_by_sample'>Get Fusions By Sample ID</h2><span id='topic+get_fusions_by_sample'></span><span id='topic+get_structural_variants_by_sample'></span>

<h3>Description</h3>

<p>Get Fusions By Sample ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_fusions_by_sample(
  sample_id = NULL,
  study_id = NULL,
  molecular_profile_id = NULL,
  sample_study_pairs = NULL,
  genes = NULL,
  panel = NULL,
  base_url = NULL
)

get_structural_variants_by_sample(
  sample_id = NULL,
  study_id = NULL,
  molecular_profile_id = NULL,
  sample_study_pairs = NULL,
  genes = NULL,
  panel = NULL,
  base_url = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_fusions_by_sample_+3A_sample_id">sample_id</code></td>
<td>
<p>a vector of sample IDs (character)</p>
</td></tr>
<tr><td><code id="get_fusions_by_sample_+3A_study_id">study_id</code></td>
<td>
<p>A string indicating the study ID from which to pull data. If no study ID, will
guess the study ID based on your URL and inform. Only 1 study ID can be passed. If mutations/cna from
more than 1 study needed, see <code>sample_study_pairs</code></p>
</td></tr>
<tr><td><code id="get_fusions_by_sample_+3A_molecular_profile_id">molecular_profile_id</code></td>
<td>
<p>A string indicating the molecular profile ID from which to pull data. If ID supplied, will
guess the molecular profile ID based on the study ID. Only 1 molecular profile ID can be passed. If mutations from
more than 1 study needed, see <code>sample_study_pairs</code></p>
</td></tr>
<tr><td><code id="get_fusions_by_sample_+3A_sample_study_pairs">sample_study_pairs</code></td>
<td>
<p>A dataframe with columns: <code>sample_id</code>, <code>study_id</code> and <code>molecular_profile_id</code> (optional). Variations in capitalization of column names are accepted.
This can be used in place of <code>sample_id</code>, <code>study_id</code>, <code>molecular_profile_id</code> arguments above if you
need to pull samples from several different studies at once. If passed this will take overwrite <code>sample_id</code>, <code>study_id</code>, <code>molecular_profile_id</code> if also passed.</p>
</td></tr>
<tr><td><code id="get_fusions_by_sample_+3A_genes">genes</code></td>
<td>
<p>A vector of Entrez ids or Hugo symbols. If Hugo symbols are supplied, they will be converted to entrez ids using the <code>get_entrez_id()</code> function.
If <code>panel</code> and <code>genes</code> are both supplied, genes from both arguments will be returned. If both are NULL (default), it will return gene results for all available genomic data for that sample.</p>
</td></tr>
<tr><td><code id="get_fusions_by_sample_+3A_panel">panel</code></td>
<td>
<p>One or more panel IDs to query (e.g. 'IMPACT468').
If <code>panel</code> and <code>genes</code> are both supplied, genes from both arguments will be returned. If both are NULL (default), it will return gene results for all available genomic data for that sample.</p>
</td></tr>
<tr><td><code id="get_fusions_by_sample_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of Fusions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set_cbioportal_db("public")

#' # These return the same results
get_fusions_by_sample(sample_id = c("s_C_CAUWT7_P001_d"),
                 study_id = "prad_msk_2019")
get_structural_variants_by_sample(sample_id = c("s_C_CAUWT7_P001_d"),
                 study_id = "prad_msk_2019")
                 
## End(Not run)
</code></pre>

<hr>
<h2 id='get_fusions_by_study'>Get Fusions By Study</h2><span id='topic+get_fusions_by_study'></span><span id='topic+get_structural_variants_by_study'></span>

<h3>Description</h3>

<p>Get Fusions By Study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_fusions_by_study(
  study_id = NULL,
  molecular_profile_id = NULL,
  base_url = NULL
)

get_structural_variants_by_study(
  study_id = NULL,
  molecular_profile_id = NULL,
  base_url = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_fusions_by_study_+3A_study_id">study_id</code></td>
<td>
<p>A study ID to query mutations. If NULL, guesses study ID based
on molecular_profile_id.</p>
</td></tr>
<tr><td><code id="get_fusions_by_study_+3A_molecular_profile_id">molecular_profile_id</code></td>
<td>
<p>a molecular profile to query mutations.
If NULL, guesses molecular_profile_id based on study ID.</p>
</td></tr>
<tr><td><code id="get_fusions_by_study_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of fusions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# These return the same results
get_fusions_by_study(molecular_profile_id = "prad_msk_2019_structural_variants")

get_structural_variants_by_study(molecular_profile_id =
       "prad_msk_2019_structural_variants")
       
## End(Not run)
</code></pre>

<hr>
<h2 id='get_gene_panel'>Retrieve Genes Included For a Specified Panel ID</h2><span id='topic+get_gene_panel'></span>

<h3>Description</h3>

<p>Retrieve Genes Included For a Specified Panel ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gene_panel(panel_id = NULL, base_url = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_gene_panel_+3A_panel_id">panel_id</code></td>
<td>
<p>name of panel. See <code>available_gene_panels()</code> to get panel ID</p>
</td></tr>
<tr><td><code id="get_gene_panel_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query.
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of genes in a specified panel
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_gene_panel(panel_id = "IMPACT468", base_url = 'www.cbioportal.org/api')

## End(Not run)

</code></pre>

<hr>
<h2 id='get_genes'>Get A List of Genes for a Specified Database</h2><span id='topic+get_genes'></span>

<h3>Description</h3>

<p>Get A List of Genes for a Specified Database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_genes(base_url = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_genes_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of gene ids, hugo symbols, and gene types
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_genes(base_url = 'www.cbioportal.org/api')

## End(Not run)
</code></pre>

<hr>
<h2 id='get_genetics_by_sample'>Get All Genomic Information By Sample IDs</h2><span id='topic+get_genetics_by_sample'></span>

<h3>Description</h3>

<p>Get All Genomic Information By Sample IDs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_genetics_by_sample(
  sample_id = NULL,
  study_id = NULL,
  sample_study_pairs = NULL,
  genes = NULL,
  panel = NULL,
  add_hugo = TRUE,
  base_url = NULL,
  return_segments = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_genetics_by_sample_+3A_sample_id">sample_id</code></td>
<td>
<p>a vector of sample IDs (character)</p>
</td></tr>
<tr><td><code id="get_genetics_by_sample_+3A_study_id">study_id</code></td>
<td>
<p>A string indicating the study ID from which to pull data. If no study ID, will
guess the study ID based on your URL and inform. Only 1 study ID can be passed. If mutations/cna from
more than 1 study needed, see <code>sample_study_pairs</code></p>
</td></tr>
<tr><td><code id="get_genetics_by_sample_+3A_sample_study_pairs">sample_study_pairs</code></td>
<td>
<p>A dataframe with columns: <code>sample_id</code>, <code>study_id</code> and <code>molecular_profile_id</code> (optional). Variations in capitalization of column names are accepted.
This can be used in place of <code>sample_id</code>, <code>study_id</code>, <code>molecular_profile_id</code> arguments above if you
need to pull samples from several different studies at once. If passed this will take overwrite <code>sample_id</code>, <code>study_id</code>, <code>molecular_profile_id</code> if also passed.</p>
</td></tr>
<tr><td><code id="get_genetics_by_sample_+3A_genes">genes</code></td>
<td>
<p>A vector of Entrez ids or Hugo symbols. If Hugo symbols are supplied, they will be converted to entrez ids using the <code>get_entrez_id()</code> function.
If <code>panel</code> and <code>genes</code> are both supplied, genes from both arguments will be returned. If both are NULL (default), it will return gene results for all available genomic data for that sample.</p>
</td></tr>
<tr><td><code id="get_genetics_by_sample_+3A_panel">panel</code></td>
<td>
<p>One or more panel IDs to query (e.g. 'IMPACT468').
If <code>panel</code> and <code>genes</code> are both supplied, genes from both arguments will be returned. If both are NULL (default), it will return gene results for all available genomic data for that sample.</p>
</td></tr>
<tr><td><code id="get_genetics_by_sample_+3A_add_hugo">add_hugo</code></td>
<td>
<p>Logical indicating whether <code>HugoGeneSymbol</code> should be added to your resulting data frame, if not already present in raw API results.
Argument is <code>TRUE</code> by default. If <code>FALSE</code>, results will be returned as is (i.e. any existing Hugo Symbol columns in raw results will not be removed).</p>
</td></tr>
<tr><td><code id="get_genetics_by_sample_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
<tr><td><code id="get_genetics_by_sample_+3A_return_segments">return_segments</code></td>
<td>
<p>Default is <code>FALSE</code> where copy number segmentation data won't be returned in addition to the mutation, cna and structural variant data.
<code>TRUE</code> will return any available segmentation data with results.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of mutations, cna and structural variants (including fusions), if available. Will also return copy number segmentation data if <code>return_segments = TRUE</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_genetics_by_sample(sample_id = c("TCGA-OR-A5J2-01","TCGA-OR-A5J6-01"),
 study_id = "acc_tcga",
 return_segments = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_genetics_by_study'>Get All Genomic Information By Study</h2><span id='topic+get_genetics_by_study'></span>

<h3>Description</h3>

<p>Get All Genomic Information By Study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_genetics_by_study(
  study_id = NULL,
  add_hugo = TRUE,
  base_url = NULL,
  return_segments = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_genetics_by_study_+3A_study_id">study_id</code></td>
<td>
<p>A study ID to query mutations. If NULL, guesses study ID based
on molecular_profile_id.</p>
</td></tr>
<tr><td><code id="get_genetics_by_study_+3A_add_hugo">add_hugo</code></td>
<td>
<p>Logical indicating whether <code>HugoGeneSymbol</code> should be added to your resulting data frame, if not already present in raw API results.
Argument is <code>TRUE</code> by default. If <code>FALSE</code>, results will be returned as is (i.e. any existing Hugo Symbol columns in raw results will not be removed).</p>
</td></tr>
<tr><td><code id="get_genetics_by_study_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
<tr><td><code id="get_genetics_by_study_+3A_return_segments">return_segments</code></td>
<td>
<p>Default is <code>FALSE</code> where copy number segmentation data won't be returned in addition to the mutation, cna and structural variant data.
<code>TRUE</code> will return any available segmentation data with results.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of mutations, cna and structural variants (including fusions), if available. Will also return copy number segmentation data if <code>return_segments = TRUE</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_genetics_by_study(study_id = "prad_msk_2019")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_hugo_symbol'>Get Hugo Symbol for a given set of Entrez IDs</h2><span id='topic+get_hugo_symbol'></span>

<h3>Description</h3>

<p>Get Hugo Symbol for a given set of Entrez IDs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_hugo_symbol(entrez_id = NULL, base_url = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_hugo_symbol_+3A_entrez_id">entrez_id</code></td>
<td>
<p>a character or numeric vector of Entrez gene IDs</p>
</td></tr>
<tr><td><code id="get_hugo_symbol_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with Entrez Gene IDs and Hugo Symbols
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_hugo_symbol(entrez_id = 2261, base_url = 'www.cbioportal.org/api')
get_hugo_symbol(entrez_id = c(2261, 7157) , base_url = 'www.cbioportal.org/api')

## End(Not run)
</code></pre>

<hr>
<h2 id='get_mutations_by_sample'>Get Mutations By Sample ID</h2><span id='topic+get_mutations_by_sample'></span>

<h3>Description</h3>

<p>Get Mutations By Sample ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mutations_by_sample(
  sample_id = NULL,
  study_id = NULL,
  molecular_profile_id = NULL,
  sample_study_pairs = NULL,
  genes = NULL,
  panel = NULL,
  add_hugo = TRUE,
  base_url = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_mutations_by_sample_+3A_sample_id">sample_id</code></td>
<td>
<p>a vector of sample IDs (character)</p>
</td></tr>
<tr><td><code id="get_mutations_by_sample_+3A_study_id">study_id</code></td>
<td>
<p>A string indicating the study ID from which to pull data. If no study ID, will
guess the study ID based on your URL and inform. Only 1 study ID can be passed. If mutations/cna from
more than 1 study needed, see <code>sample_study_pairs</code></p>
</td></tr>
<tr><td><code id="get_mutations_by_sample_+3A_molecular_profile_id">molecular_profile_id</code></td>
<td>
<p>A string indicating the molecular profile ID from which to pull data. If ID supplied, will
guess the molecular profile ID based on the study ID. Only 1 molecular profile ID can be passed. If mutations from
more than 1 study needed, see <code>sample_study_pairs</code></p>
</td></tr>
<tr><td><code id="get_mutations_by_sample_+3A_sample_study_pairs">sample_study_pairs</code></td>
<td>
<p>A dataframe with columns: <code>sample_id</code>, <code>study_id</code> and <code>molecular_profile_id</code> (optional). Variations in capitalization of column names are accepted.
This can be used in place of <code>sample_id</code>, <code>study_id</code>, <code>molecular_profile_id</code> arguments above if you
need to pull samples from several different studies at once. If passed this will take overwrite <code>sample_id</code>, <code>study_id</code>, <code>molecular_profile_id</code> if also passed.</p>
</td></tr>
<tr><td><code id="get_mutations_by_sample_+3A_genes">genes</code></td>
<td>
<p>A vector of Entrez ids or Hugo symbols. If Hugo symbols are supplied, they will be converted to entrez ids using the <code>get_entrez_id()</code> function.
If <code>panel</code> and <code>genes</code> are both supplied, genes from both arguments will be returned. If both are NULL (default), it will return gene results for all available genomic data for that sample.</p>
</td></tr>
<tr><td><code id="get_mutations_by_sample_+3A_panel">panel</code></td>
<td>
<p>One or more panel IDs to query (e.g. 'IMPACT468').
If <code>panel</code> and <code>genes</code> are both supplied, genes from both arguments will be returned. If both are NULL (default), it will return gene results for all available genomic data for that sample.</p>
</td></tr>
<tr><td><code id="get_mutations_by_sample_+3A_add_hugo">add_hugo</code></td>
<td>
<p>Logical indicating whether <code>HugoGeneSymbol</code> should be added to your resulting data frame, if not already present in raw API results.
Argument is <code>TRUE</code> by default. If <code>FALSE</code>, results will be returned as is (i.e. any existing Hugo Symbol columns in raw results will not be removed).</p>
</td></tr>
<tr><td><code id="get_mutations_by_sample_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of mutations (maf file format)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_mutations_by_sample(sample_id = c("TCGA-OR-A5J2-01","TCGA-OR-A5J6-01"),
study_id = "acc_tcga",
base_url = "public")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_mutations_by_study'>Get Mutations By Study ID</h2><span id='topic+get_mutations_by_study'></span>

<h3>Description</h3>

<p>Get Mutations By Study ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mutations_by_study(
  study_id = NULL,
  molecular_profile_id = NULL,
  add_hugo = TRUE,
  base_url = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_mutations_by_study_+3A_study_id">study_id</code></td>
<td>
<p>A study ID to query mutations. If NULL, guesses study ID based
on molecular_profile_id.</p>
</td></tr>
<tr><td><code id="get_mutations_by_study_+3A_molecular_profile_id">molecular_profile_id</code></td>
<td>
<p>a molecular profile to query mutations.
If NULL, guesses molecular_profile_id based on study ID.</p>
</td></tr>
<tr><td><code id="get_mutations_by_study_+3A_add_hugo">add_hugo</code></td>
<td>
<p>Logical indicating whether <code>HugoGeneSymbol</code> should be added to your resulting data frame, if not already present in raw API results.
Argument is <code>TRUE</code> by default. If <code>FALSE</code>, results will be returned as is (i.e. any existing Hugo Symbol columns in raw results will not be removed).</p>
</td></tr>
<tr><td><code id="get_mutations_by_study_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of mutations (maf file format)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_mutations_by_study(study_id = "prad_msk_2019")
get_mutations_by_study(molecular_profile_id = "prad_msk_2019_mutations")

## End(Not run)

</code></pre>

<hr>
<h2 id='get_panel_by_sample'>Get Gene Panel by study ID and sample ID</h2><span id='topic+get_panel_by_sample'></span>

<h3>Description</h3>

<p>Get Gene Panel by study ID and sample ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_panel_by_sample(
  study_id = NULL,
  sample_id = NULL,
  sample_study_pairs = NULL,
  base_url = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_panel_by_sample_+3A_study_id">study_id</code></td>
<td>
<p>A string indicating the study ID from which to pull data. If no study ID, will
guess the study ID based on your URL and inform. Only 1 study ID can be passed. If mutations/cna from
more than 1 study needed, see <code>sample_study_pairs</code></p>
</td></tr>
<tr><td><code id="get_panel_by_sample_+3A_sample_id">sample_id</code></td>
<td>
<p>a vector of sample IDs (character)</p>
</td></tr>
<tr><td><code id="get_panel_by_sample_+3A_sample_study_pairs">sample_study_pairs</code></td>
<td>
<p>A dataframe with columns: <code>sample_id</code>, <code>study_id</code> and <code>molecular_profile_id</code> (optional). Variations in capitalization of column names are accepted.
This can be used in place of <code>sample_id</code>, <code>study_id</code>, <code>molecular_profile_id</code> arguments above if you
need to pull samples from several different studies at once. If passed this will take overwrite <code>sample_id</code>, <code>study_id</code>, <code>molecular_profile_id</code> if also passed.</p>
</td></tr>
<tr><td><code id="get_panel_by_sample_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of a specific clinical attribute
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_panel_by_sample(study_id = "blca_plasmacytoid_mskcc_2016",
 sample_id = "DS-sig-010-P2",
 base_url = 'www.cbioportal.org/api')
 
## End(Not run)

</code></pre>

<hr>
<h2 id='get_samples_by_patient'>Get sample IDs for a given set of patient IDs</h2><span id='topic+get_samples_by_patient'></span>

<h3>Description</h3>

<p>Get sample IDs for a given set of patient IDs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_samples_by_patient(patient_id = NULL, study_id = NULL, base_url = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_samples_by_patient_+3A_patient_id">patient_id</code></td>
<td>
<p>A character string of sample IDs to query</p>
</td></tr>
<tr><td><code id="get_samples_by_patient_+3A_study_id">study_id</code></td>
<td>
<p>A character string indicating which study ID should be searched.
Only 1 study allowed. If NULL, we will guess a default study ID based on your database URL.</p>
</td></tr>
<tr><td><code id="get_samples_by_patient_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of patient IDs and corresponding sample IDs. If patient
has multiple samples, there will be multiple rows per patient.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_samples_by_patient(patient_id = c("P-0000034", "P-0000036"))

## End(Not run)

</code></pre>

<hr>
<h2 id='get_segments_by_sample'>Get Copy Number Segmentation Data By Sample ID</h2><span id='topic+get_segments_by_sample'></span>

<h3>Description</h3>

<p>Get Copy Number Segmentation Data By Sample ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_segments_by_sample(
  sample_id = NULL,
  study_id = NULL,
  sample_study_pairs = NULL,
  base_url = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_segments_by_sample_+3A_sample_id">sample_id</code></td>
<td>
<p>a vector of sample IDs (character)</p>
</td></tr>
<tr><td><code id="get_segments_by_sample_+3A_study_id">study_id</code></td>
<td>
<p>A string indicating the study ID from which to pull data. If no study ID, will
guess the study ID based on your URL and inform. Only 1 study ID can be passed. If mutations/cna from
more than 1 study needed, see <code>sample_study_pairs</code></p>
</td></tr>
<tr><td><code id="get_segments_by_sample_+3A_sample_study_pairs">sample_study_pairs</code></td>
<td>
<p>A dataframe with columns: <code>sample_id</code>, <code>study_id</code> and <code>molecular_profile_id</code> (optional). Variations in capitalization of column names are accepted.
This can be used in place of <code>sample_id</code>, <code>study_id</code>, <code>molecular_profile_id</code> arguments above if you
need to pull samples from several different studies at once. If passed this will take overwrite <code>sample_id</code>, <code>study_id</code>, <code>molecular_profile_id</code> if also passed.</p>
</td></tr>
<tr><td><code id="get_segments_by_sample_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of CNA segments
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set_cbioportal_db("public")

get_segments_by_sample(sample_id = c("s_C_CAUWT7_P001_d"),
                 study_id = "prad_msk_2019")
                 
## End(Not run)
</code></pre>

<hr>
<h2 id='get_segments_by_study'>Get Copy Number Segmentation Data By Study</h2><span id='topic+get_segments_by_study'></span>

<h3>Description</h3>

<p>Get Copy Number Segmentation Data By Study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_segments_by_study(study_id = NULL, add_hugo = TRUE, base_url = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_segments_by_study_+3A_study_id">study_id</code></td>
<td>
<p>A study ID to query mutations. If NULL, guesses study ID based
on molecular_profile_id.</p>
</td></tr>
<tr><td><code id="get_segments_by_study_+3A_add_hugo">add_hugo</code></td>
<td>
<p>Logical indicating whether <code>HugoGeneSymbol</code> should be added to your resulting data frame, if not already present in raw API results.
Argument is <code>TRUE</code> by default. If <code>FALSE</code>, results will be returned as is (i.e. any existing Hugo Symbol columns in raw results will not be removed).</p>
</td></tr>
<tr><td><code id="get_segments_by_study_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of CNA segments
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_segments_by_study(study_id = "prad_msk_2019")
get_segments_by_study(molecular_profile_id = "prad_msk_2019_cna")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_study_info'>Get Metadata on All Available Studies in Database or a Specified Study</h2><span id='topic+get_study_info'></span>

<h3>Description</h3>

<p>Get Metadata on All Available Studies in Database or a Specified Study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_study_info(study_id = NULL, base_url = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_study_info_+3A_study_id">study_id</code></td>
<td>
<p>one or more study IDs (see <code>available_studies()</code> to lookup IDs)</p>
</td></tr>
<tr><td><code id="get_study_info_+3A_base_url">base_url</code></td>
<td>
<p>The database URL to query.
If <code>NULL</code> will default to URL set with <code style="white-space: pre;">&#8288;set_cbioportal_db(&lt;your_db&gt;)&#8288;</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of study metadata
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set_cbioportal_db("public")
get_study_info("acc_tcga")

## End(Not run)
</code></pre>

<hr>
<h2 id='impact_gene_info'>IMPACT Gene Meta Data</h2><span id='topic+impact_gene_info'></span>

<h3>Description</h3>

<p>Dataframe labeling all genes included in IMPACT panels along with their corresponding
platform ID and Entrez ID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>impact_gene_info
</code></pre>


<h3>Format</h3>

<p>A data frame with 470 genes
</p>

<dl>
<dt>hugo_symbol</dt><dd><p>Factor w/ 574 levels,
Column containing all HUGO symbols genes included in IMPACT</p>
</dd>
<dt>entrez_id</dt><dd><p>Integer, contains all Entrez IDs for genes included in IMPACT</p>
</dd>
<dt>platform_341</dt><dd><p>Character, indicates whether each gene was included in
IMPACT platform 341. Options are <code>included</code> and <code style="white-space: pre;">&#8288;not included&#8288;</code></p>
</dd>
<dt>platform_410</dt><dd><p>Character, indicates whether each gene was included in
IMPACT platform 410. Options are <code>included</code> and <code style="white-space: pre;">&#8288;not included&#8288;</code></p>
</dd>
<dt>platform_468</dt><dd><p>Character, indicates whether each gene was included in
IMPACT platform 468. Options are <code>included</code> and <code style="white-space: pre;">&#8288;not included&#8288;</code></p>
</dd>
<dt>alias</dt><dd><p>A nested dataframe of aliases for each gene and corresponding entrez gene ids for aliases if they exist</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="http://www.cbioportal.org/">http://www.cbioportal.org/</a>
</p>

<hr>
<h2 id='null-default'>Default value for <code>NULL</code>.</h2><span id='topic+null-default'></span><span id='topic++25+7C+7C+25'></span>

<h3>Description</h3>

<p>Default value for <code>NULL</code>.
</p>

<hr>
<h2 id='set_cbioportal_db'>Connect to cBioPortal DB</h2><span id='topic+set_cbioportal_db'></span>

<h3>Description</h3>

<p>This function sets a base cBioPortal URL
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_cbioportal_db(db = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_cbioportal_db_+3A_db">db</code></td>
<td>
<p>The database URL to use as base URL for calls, or &quot;public&quot; for https://www.cbioportal.org/</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects.
Will display an alert notifying if the user has successfully authenticated
to cBioPortal.
</p>


<h3>Author(s)</h3>

<p>Karissa Whiting, Daniel D. Sjoberg
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set_cbioportal_db(db = "public")

## End(Not run)

</code></pre>

<hr>
<h2 id='test_cbioportal_db'>Test the Database Connection Anytime During your R Session</h2><span id='topic+test_cbioportal_db'></span>

<h3>Description</h3>

<p>Helps troubleshoot API issues during an R session
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test_cbioportal_db()
</code></pre>


<h3>Value</h3>

<p>No return value, called for side effects.
Will display an alert notifying if the user has successfully authenticated
to cBioPortal
</p>


<h3>Author(s)</h3>

<p>Karissa Whiting, Daniel D. Sjoberg
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set_cbioportal_db("public")
test_cbioportal_db()

## End(Not run)


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
