<!DOCTYPE html><html><head><title>Help for package ccRemover</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ccRemover}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bootstrap_diff'><p>Calculates the difference in the loading score for cell-cycle and control</p>
genes</a></li>
<li><a href='#ccRemover'><p>Removes the effect of the cell-cycle</p></a></li>
<li><a href='#dat'><p> A simulated scRNA-Seq data.</p></a></li>
<li><a href='#gene_indexer'><p>Identifies genes annotated to the cell-cycle</p></a></li>
<li><a href='#get_diff'><p>Calculates the average load difference between the cell-cycle genes and</p>
control genes for each component.</a></li>
<li><a href='#hello'><p>Hello, World!</p></a></li>
<li><a href='#HScc_genes'><p>Homo Sapien genes which are annotated to the cell-cycle</p></a></li>
<li><a href='#MMcc_genes'><p>Mus Musculus genes which are annotated to the cell-cycle</p></a></li>
<li><a href='#t.cell_data'><p> Single-Cell RNA-Seq data from differentiating T-helper cells</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Removes the Cell-Cycle Effect from Single-Cell RNA-Sequencing
Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.4</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements a method for identifying and removing
				the cell-cycle effect from scRNA-Seq data. The description of the 
				method is in Barron M. and Li J. (2016) &lt;<a href="https://doi.org/10.1038%2Fsrep33892">doi:10.1038/srep33892</a>&gt;. Identifying and removing 
				the cell-cycle effect from single-cell RNA-Sequencing data. Submitted. 
				Different from previous methods, ccRemover implements a mechanism that
				formally tests whether a component is cell-cycle related or not, and thus
				while it often thoroughly removes the cell-cycle effect, it preserves
				other features/signals of interest in the data.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, utils</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-08-17 15:33:13 UTC; Administrator</td>
</tr>
<tr>
<td>Author:</td>
<td>Jun Li [aut, cre],
  Martin Barron [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jun Li &lt;jun.li@nd.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-08-19 12:23:32 UTC</td>
</tr>
</table>
<hr>
<h2 id='bootstrap_diff'>Calculates the difference in the loading score for cell-cycle and control
genes</h2><span id='topic+bootstrap_diff'></span>

<h3>Description</h3>

<p>This function is only used internally inside ccRemover. The function
calcualtes the average load difference on the cell-cycle and control genes.
Bootstrap resampling is then used to provide a score for each component.
Please see the original manuscript for the mathematical details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bootstrap_diff(xy, xn, nboot = 200, bar = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bootstrap_diff_+3A_xy">xy</code></td>
<td>
<p>The data for the genes which are annotated to the cell-cycle, i.e.
those genes for which &quot;if_cc&quot; is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="bootstrap_diff_+3A_xn">xn</code></td>
<td>
<p>The data for the genes which are not annotated to the cell-cycle,
control genes, genes for which &quot;if_cc&quot; is <code>FALSE</code></p>
</td></tr>
<tr><td><code id="bootstrap_diff_+3A_nboot">nboot</code></td>
<td>
<p>The number of bootstrap repititions to be carried out on each
iteration to determine the significance of each component.</p>
</td></tr>
<tr><td><code id="bootstrap_diff_+3A_bar">bar</code></td>
<td>
<p>Whether to display a progress bar or not. The progress bar will
not work in R-markdown enviornments so this option may be turned off. The
default value is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing the loadings for each component on the
cell-cycle and control genes as well as the difference between the loadings
and the bootstrapped statistic for each loading.
</p>

<hr>
<h2 id='ccRemover'>Removes the effect of the cell-cycle</h2><span id='topic+ccRemover'></span>

<h3>Description</h3>

<p><code>ccRemover</code> returns a data matrix with the effects of the cell-cycle
removed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ccRemover(dat, cutoff = 3, max_it = 4, nboot = 200, ntop = 10,
  bar = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ccRemover_+3A_dat">dat</code></td>
<td>
<p>A list containing a data frame , <code>x</code>, that contains gene expression
measurements with each column representing a sample and each row
representing a gene and a logical vector, <code>if_cc</code>, that indicates
which of the genes/rows are related to the cell-cycle or factor of interest.
</p>
<p>It is recommended that the elements of x are log-transformed and centered
for each gene. For example if <code>x</code> contains TPM measurements then we
suggest the following two-steps:
Step 1: <code>dat$x &lt;- log(dat$x + 1)</code>
Step 2: <code>dat$x</code> - rowMeans(dat$x)
ccRemover requires that the samples have been properly normalized for
sequencing depth and we recommend doing so prior to applying the above steps.
</p>
<p>The <code>if_cc</code> vector must be the same length as the number of rows in
<code>x</code> and have elements equal to <code>TRUE</code> for genes which are related
to the cell-cycle and and elements equal to <code>FALSE</code> for genes which
are unrelated to the cell-cycle.</p>
</td></tr>
<tr><td><code id="ccRemover_+3A_cutoff">cutoff</code></td>
<td>
<p>The significance cutoff for identifying sources of variation
related to the cell-cycle. The default value is 3, which roughly corresponds
to a p-value of 0.01.</p>
</td></tr>
<tr><td><code id="ccRemover_+3A_max_it">max_it</code></td>
<td>
<p>The maximum number of iterations for the algorithm. The
default value is 4.</p>
</td></tr>
<tr><td><code id="ccRemover_+3A_nboot">nboot</code></td>
<td>
<p>The number of bootstrap repititions to be carried out on each
iteration to determine the significance of each component.</p>
</td></tr>
<tr><td><code id="ccRemover_+3A_ntop">ntop</code></td>
<td>
<p>The number of components considered tested at each iteration as
cell-cycle effects. The default value if 10</p>
</td></tr>
<tr><td><code id="ccRemover_+3A_bar">bar</code></td>
<td>
<p>Whether to display a progress bar or not. The progress bar will
not work in R-markdown enviornments so this option may be turned off. The
default value is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Implements the algorithm described in Barron, M. &amp; Li, J.
&quot;Identifying and removing the cell-cycle effect from scRNA-Sequencing data&quot;
(2016), Scientific Reports. This function takes a normalized,
log-transformed and centered matrix of scRNA-seq  expression data
and a list of genes which are known to be related to the cell-cycle effect.
It then captures the main sources of variation in the data and determines
which of these are related to the cell-cycle before removing those that are.
Please see the original manuscript for further details.
</p>


<h3>Value</h3>

<p>A data matrix with the effects of the cell-cycle removed.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(10)
# Load in example data
data(t.cell_data)
head(t.cell_data[,1:5])
# Center data and select small sample for example
t_cell_data_cen &lt;- t(scale(t(t.cell_data[,1:20]), center=TRUE, scale=FALSE))
# Extract gene names
gene_names &lt;- rownames(t_cell_data_cen)
# Determine which genes are annotated to the cell-cycle
cell_cycle_gene_indices &lt;- gene_indexer(gene_names,
species = "mouse", name_type = "symbol")
# Create "if_cc" vector
if_cc &lt;- rep(FALSE,nrow(t_cell_data_cen))
if_cc[cell_cycle_gene_indices] &lt;- TRUE
# Move data into list
dat &lt;- list(x=t_cell_data_cen, if_cc=if_cc)
# Run ccRemover
## Not run: 
 xhat &lt;- ccRemover(dat, cutoff = 3, max_it = 4, nboot = 200, ntop = 10)

## End(Not run)
# Run ccRemover with reduced bootstrap repetitions for example only
xhat &lt;- ccRemover(dat, cutoff = 3, max_it = 4, nboot = 20, ntop = 10)
head(xhat[,1:5])
# Run ccRemover with more compoents considered
## Not run: 
xhat &lt;- ccRemover(dat, cutoff = 3, max_it = 4, nboot = 200, ntop = 15)
 
## End(Not run)

</code></pre>

<hr>
<h2 id='dat'> A simulated scRNA-Seq data. </h2><span id='topic+dat'></span>

<h3>Description</h3>

<p>This data contains expression levels (log-transformed and centered) for 50 cells and 2000 genes.
The 50 cells are randomly assigned to two cell types and three cell-cycle stages.
400 genes are assigned as cell-cycle genes, and the other 1600 genes are control genes.
For descriptions of how we generated this data, please refer to the paper.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(dat)</code></pre>


<h3>Format</h3>

<p>A list that contains the following attributes (only <code>x</code> and <code>if.cc</code> are used by ccRemover.main.)
</p>

<dl>
<dt><code>x</code></dt><dd><p>the data matrix. rows are genes, and columns are cells. These should be treated as log-transformed and
centered (each row has mean 0) expression levels.</p>
</dd>
<dt><code>if.cc</code></dt><dd><p>a vector of values FALSE's or TRUE's, denoting whether the genes are cell-cycle related or control.</p>
</dd>
<dt><code>n</code></dt><dd><p>the number of cells. n=ncol(x).</p>
</dd>
<dt><code>p</code></dt><dd><p>the number of genes. p=nrow(x).</p>
</dd>
<dt><code>pc</code></dt><dd><p>the number of cell-cycle genes. pc=sum(if.cc).</p>
</dd>
<dt><code>ct</code></dt><dd><p>cell types. a vector of values 1 and 2.</p>
</dd>
<dt><code>cc</code></dt><dd><p>cell-cycle stages. a vector of values 1, 2, or 3.</p>
</dd>
</dl>



<h3>Value</h3>

<p>A simulated dataset used to demonstrate the application of 
ccRemover</p>

<hr>
<h2 id='gene_indexer'>Identifies genes annotated to the cell-cycle</h2><span id='topic+gene_indexer'></span>

<h3>Description</h3>

<p>Determines which of the genes contained in the dataset are annotated ti the
cell-cycle. This is a preprocessing function for ccRemover. Genes can be
either mouse or human and either official gene symbols, Ensembl, Entrez or
Unigene IDs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_indexer(gene_names, species = NULL, name_type = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gene_indexer_+3A_gene_names">gene_names</code></td>
<td>
<p>A vector containing the gene names for the dataset.</p>
</td></tr>
<tr><td><code id="gene_indexer_+3A_species">species</code></td>
<td>
<p>The species which the gene names are from. Either
<code>"human"</code> or <code>"mouse"</code>.</p>
</td></tr>
<tr><td><code id="gene_indexer_+3A_name_type">name_type</code></td>
<td>
<p>The type of gene name considered either, Ensembl gene IDS
(<code>"ensembl"</code>), offical gene symbols (<code>symbol</code>), Entrez gene IDS
(<code>"entrez"</code>), or Unigene IDS (<code>unigene</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector containg the indices of genes which are annotated to the cell-cycle
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(10)
# Load in example data
data(t.cell_data)
head(t.cell_data[,1:5])
# Center example data
t_cell_data_cen &lt;- t(scale(t(t.cell_data), center=TRUE, scale=FALSE))
# Extract gene names
gene_names &lt;- rownames(t_cell_data_cen)
# Determine which genes are annotated to the cell-cycle
cell_cycle_gene_indices &lt;- gene_indexer(gene_names = gene_names,
species = "mouse", name_type = "symbol")
# Create "if_cc" vector
if_cc &lt;- rep(FALSE,nrow(t_cell_data_cen))
if_cc[cell_cycle_gene_indices] &lt;- TRUE

# Can allow the function to automatically detect the name type
cell_cycle_gene_indices &lt;- gene_indexer(gene_names = gene_names,
species = NULL, name_type = NULL)

</code></pre>

<hr>
<h2 id='get_diff'>Calculates the average load difference between the cell-cycle genes and
control genes for each component.</h2><span id='topic+get_diff'></span>

<h3>Description</h3>

<p>This is an interal function for use by &quot;bootstrap_diff&quot; only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_diff(xy, xn)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_diff_+3A_xy">xy</code></td>
<td>
<p>The data for the genes which are annotated to the cell-cycle, i.e.
those genes for which &quot;if_cc&quot; is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="get_diff_+3A_xn">xn</code></td>
<td>
<p>The data for the genes which are not annotated to the cell-cycle,
control genes, genes for which &quot;if_cc&quot; is <code>FALSE</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing the loadings for each component on the
cell-cycle and control genes.
</p>

<hr>
<h2 id='hello'>Hello, World!</h2><span id='topic+hello'></span>

<h3>Description</h3>

<p>Prints 'Hello, world!'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hello()
</code></pre>


<h3>Examples</h3>

<pre><code class='language-R'>hello()
</code></pre>

<hr>
<h2 id='HScc_genes'>Homo Sapien genes which are annotated to the cell-cycle</h2><span id='topic+human_cell_cycle_genes'></span>

<h3>Description</h3>

<p>This data set contains Homo Sapien genes which are annotated to the cell-cycle.
These genes were retrieved from biomart and are intended for use with the 
&quot;gene_indexer&quot; function. The data set contains the gene names in four different 
formats, Ensemble Gene IDs (1838 values), HGNC symbols (1740 values), 
Entrez Gene IDs (1744 values) and Unigene IDs (1339).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("HScc_genes")</code></pre>


<h3>Format</h3>

<p>A data set that contains with the following attributes
</p>

<dl>
<dt><code>human_cell_cycle_genes</code></dt><dd><p>A data frame with four columns corresponding 
to each of the different ID formats.</p>
</dd>
</dl>



<h3>Value</h3>

<p>A data set containing genes annotated to the cell-cycle in different ID formats</p>

<hr>
<h2 id='MMcc_genes'>Mus Musculus genes which are annotated to the cell-cycle</h2><span id='topic+mouse_cell_cycle_genes'></span>

<h3>Description</h3>

<p>This data set contains Mus Musculus genes which are annotated to the cell-cycle.
These genes were retrieved from biomart and are intended for use with the 
&quot;gene_indexer&quot; function. The data set contains the gene names in three different 
formats, Ensemble Gene IDs (1433 values), MGI symbols (1422 values),
Entrez Gene IDs (1435 values) and Unigene IDs (1102 values).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("MMcc_genes")</code></pre>


<h3>Format</h3>

<p>A data set that contains with the following attributes
</p>

<dl>
<dt><code>mouse_cell_cycle_genes</code></dt><dd><p>A data frame with four columns corresponding 
to each of the different ID formats.</p>
</dd>
</dl>



<h3>Value</h3>

<p>A dataset containing genes annotated to the cell-cycle in different ID formats</p>

<hr>
<h2 id='t.cell_data'> Single-Cell RNA-Seq data from differentiating T-helper cells</h2><span id='topic+t.cell_data'></span>

<h3>Description</h3>

<p>This data contains expression levels (log-transformed normalized count values)
for 81 cells and 14,147 genes. The data was normalized using ERCC spike-ins. This
data was generated by Mahata, B. et al (2014). The processed data was retrieved from
the supplementary material of Buettner et al. (2015), for descriptions of how the data was
processed, please refer to their paper.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(t.cell_data)</code></pre>


<h3>Format</h3>

<p>A data set that contains with the following attributes
</p>

<dl>
<dt><code>t.cell_data</code></dt><dd><p>the data matrix. rows are cells, and columns are genes. These should be treated as log-transformed and normalized</p>
</dd>
</dl>



<h3>Value</h3>

<p>A scRNA-Seq dataset with gene expression levels for 187 T-helper cells</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
