<!DOCTYPE html><html><head><title>Help for package cellOrigins</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {cellOrigins}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BDGP_insitu_dmel_embryo'>

<p>Patterns of gene expression in Drosophila melanogaster embryos</p></a></li>
<li><a href='#cellOrigins-package'>
<p>Finding the most likely originating tissue(s) and developmental stage(s) of RNASeq data</p></a></li>
<li><a href='#diagnosticPlots'>

<p>Diagnostic plots to explore seqVsInsitu results</p></a></li>
<li><a href='#discovery_probability'>
<p>In situ discovery probability as a function of FPKM</p>
</a></li>
<li><a href='#discovery.log'>

<p>Calculates discovery probability by RNA in situ hybridisation given a sequencing signal</p></a></li>
<li><a href='#iterating_seqVsInsitu'>

<p>Faster comparisons between mixed tissue-specific RNA sequencing data and high-throughput RNA in situ hybridisation</p></a></li>
<li><a href='#prior.temporal_proximity_is_good'>

<p>Assign a prior probability to a combination of anatomical terms</p></a></li>
<li><a href='#seqVsInsitu'>
<p>Determine the most likely source(s) of a tissue-specific RNAseq dataset</p>
</a></li>
<li><a href='#vncMedianCoverage.tsv'>

<p>Drosophila melanogaster embryo ventral nerve cord RNASeq coverage</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Finds RNASeq Source Tissues Using In Situ Hybridisation Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.3</td>
</tr>
<tr>
<td>Author:</td>
<td>David Molnar</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>David Molnar &lt;dmolnar100@icloud.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Finds the most likely originating tissue(s) and developmental stage(s) of tissue-specific RNA sequencing data. The package identifies both pure transcriptomes and mixtures of transcriptomes. The most likely identity is found through comparisons of the sequencing data with high-throughput in situ hybridisation patterns. Typical uses are the identification of cancer cell origins, validation of cell culture strain identities, validation of single-cell transcriptomes, and validation of identity and purity of flow-sorting and dissection sequencing products.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://creativecommons.org/licenses/by-nc-sa/4.0">CC BY-NC-SA 4.0</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>iterpc</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-06-02 12:25:49 UTC; dm</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-06-05 09:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='BDGP_insitu_dmel_embryo'>

Patterns of gene expression in Drosophila melanogaster embryos
</h2><span id='topic+BDGP_insitu_dmel_embryo'></span>

<h3>Description</h3>


<p>High-confidence dataset of embryonic <em>Drosophila melanogaster</em> RNA expression patterns at 6 developmental stages. This dataset was generated by filtering the &quot;BDGP insitu&quot; high-throughput RNA in situ hybridisation data set (Tomancak, Genome Biol. 2007;8(7):R145) for high-confidence results. Only genes useful for tissue identification were retained, and they thus represent gene expression fingerprints of organs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("BDGP_insitu_dmel_embryo")</code></pre>


<h3>Format</h3>

<p>The format is:
num [1:2395, 1:337] 1 0 0 0 1 1 0 1 1 1 ...
- attr(*, &quot;dimnames&quot;)=List of 2
..$ : chr [1:2395] &quot;LD11379&quot; &quot;LD11394&quot; &quot;LD12611&quot; &quot;LD12613&quot; ...
..$ : chr [1:337] &quot;1|maternal&quot; &quot;2|pole cell&quot; &quot;3|pole cell&quot; &quot;4|germ cell&quot; ...
</p>


<h3>Details</h3>


<p>The expression data are collated in a matrix. The columns in the matrix are labeled stage|domain (e.g. &quot;6|midgut&quot;). The expression domains are denoted using the BDGP insitu controlled anatomical vocabulary.The rows are labeled with transcripts/probe names according to the BDGP insitu data set. The hybridisation probe genomic coordinates (Drosophila melanogaster genome release 5) are supplied as an additional file in this package.
</p>
<p>The data set characterises the expression of 2395 RNA species. This is the differentially expressed, high-confidence subset of BDGP insitu. The starting point for dataset preparation was the published SQL database dump with annotations (http://insitu.fruitfly.org/insitu-mysql-dump/insitu.sql.gz). All in situ hybridisations for wild type <em>Drosophila melanogaster</em> embryos were extracted from this source. The reporter construct annotations were not used.
</p>
<p>Only high-confidence expression patterns were retained. The gene expression in the BDGP insitu database was annotated by human curators from microscopic images. Depending on the quality of images and staining some expression patterns were easier to discern than others. The curators expressed their confidence in their expression call together with the annotation data of each gene. The filtering criteria for including a probe's exression pattern were that
</p>

<ol>
<li><p> the final call of the annotators was 'acceptable',
</p>
</li>
<li><p> there was no remark about staining intensity (pointing to substandard quality),
</p>
</li>
<li><p> the microscopic image was not excluded by quality control,
</p>
</li>
<li><p> the annotation was displayed on the database's website,
</p>
</li>
<li><p> the probe/staining was not flagged for repeating or for giving up, and
</p>
</li>
<li><p> the final word of the annotators (a free text field) did not contain negative remarks like &quot;weak&quot;, &quot;nonspecific&quot;, &quot;muddy&quot;, &quot;poor&quot;, &quot;dull&quot;, &quot;spillover&quot; or &quot;suspicious&quot; staining; lack of staining penetration; a call to repeat the staining; signs of doubt (e.g. &quot;might&quot;, &quot;perhaps&quot;, &quot;maybe&quot;, &quot;could&quot;, &quot;not sure&quot;, &quot;not confirmed&quot;, &quot;unconvincing&quot;, &quot;conflicting&quot;, &quot;can't say&quot;, &quot;failure&quot;, &quot;wrong&quot;, &quot;junk&quot;); on camera problems; artefacts or transposons.
</p>
</li>
<li><p> there was no annotation with &quot;no staining&quot; to avoid false negatives.
</p>
</li></ol>

<p>Genes with known ubiquitous expression (including faint-ubiquitous) at any stage were excluded.
</p>
<p>Genes for which there was no published probe sequence (approximately 300) were excluded. Most of the RNA in situ hybridisation probes originated from the Drosophila Gold Collection (http://www.fruitfly.org/EST/gold_collection.shtml) and the Drosophila Gene Collection (http://www.fruitfly.org/DGC/index.html).
</p>
<p>Annotated gene expression in each anatomical unit was propagated to all its anatomical subunits. For example &quot;5|Malpighian tubule primordium&quot; expression was propagated to &quot;5|Malpighian tubule main body primordium&quot; and &quot;5|Malpighian tubule tip cell primordium&quot;. Only this made both the presence and the absence of staining meaningful. In the original data set gene expression was usually only annotated to the largest unit of expression, but not to its subunits. For instance if there was expression in the whole foregut, there was by necessity also expression in its pharynx subunit. However, in such a case expression in the pharynx was not commonly denoted in the original data set. Consequently some anatomic units had very few expressed genes associated. These genes were those that were exclusively expressed in those anatomical units and in no superior units.
</p>


<h3>Source</h3>


<p>Tomancak, Genome Biol. 2007;8(7):R145
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BDGP_insitu_dmel_embryo)
</code></pre>

<hr>
<h2 id='cellOrigins-package'>
Finding the most likely originating tissue(s) and developmental stage(s) of RNASeq data
</h2><span id='topic+cellOrigins-package'></span><span id='topic+cellOrigins'></span>

<h3>Description</h3>

<p><img src="../help/figures/delphi.jpg" alt="Logo: the Pythia at Delphi." />
</p>
<p><code>cellOrigins</code> compares RNASeq read coverages with in high-throughput RNA in situ hybridisation patterns for transcriptome source identification and verification. The package can identify both pure transcriptomes and mixtures of transcriptomes. Typical uses are the identification of cancer cell origins, validation of cell culture strain identities, validation of single-cell transcriptomes, and validation of identity and purity of flow-sorting and dissection sequencing products.
</p>
<p>The comparison of quantitative RNA sequencing coverage with thresholded, qualitative staining patterns is probabilistic. First, given the sequenced transcriptome, a prediction is made how likely each sequenced transcript would lead to a positive signal in a high-throughput in situ hybridisation experiment. The probability of staining increases with the logarithm of the sequencing coverage. This relationship was empirically found through a comparison between <em>Drosophila</em> embryo transcriptomes and RNA in situ staining results. Then, using Bayes's theorem all the genes in the simulated and observed hybridisation patterns are compared. The pattern (or linear combination of patterns) with the highest posterior probability is identified as the most likely source.
</p>
<p>Batteries included: the package contains a filtered high-confidence expression pattern dataset for <em>Drosophila melanogaster</em> embryos (based on BDGP insitu).
</p>
<p>Typical use:
</p>
<p>I GENERATE INPUT
</p>
<p>Input is RNASeq mean FPKM (fragments per kilobase per million reads). Whole-gene FPKM may  be used (as output by e.g. cufflinks/cuffquant), however assignment difficulties at overlapping transcripts and transcript isoforms reduce prediction quality. For best results use FPKM values calculated for the targets of the in situ hybridisation probes as described below:
</p>
<p>Step 1) Generate masking bed file &ndash; this file is included for BDGP insitu in the extdata folder. For other species align probe sequences to the target genome using BLAT (https://genome.ucsc.edu/FAQ/FAQblat.html). Convert the best-scoring alignments to a masking bed file with <code>psl_to_bed_best_score.pl</code> (https://gist.github.com/davetang/7314846).
Then sort with <code>bedtools sort</code> (http://bedtools.readthedocs.org/).
</p>
<p>Step 2) Get coverages. Use Bedtools with the masking bed file to extract the mean sequencing covereage from wig files in the in situ probed regions:
</p>
<p><code>bedtools map -a sorted_probes.bed -b sequenced.wig -o max -c 4 &gt;insitu_high_confidence.tsv</code>
</p>
<p>Use the output tab separated values file as input for the function <code><a href="#topic+seqVsInsitu">seqVsInsitu</a></code>.
</p>
<p>II SOURCE IDENTIFICATION
</p>
<p><code><a href="#topic+seqVsInsitu">seqVsInsitu</a></code> and <code><a href="#topic+iterating_seqVsInsitu">iterating_seqVsInsitu</a></code> calculate the probability for each in situ expression pattern that it is produced by the same gene expression patterns as the sequencing data. If you believe you have a mixed input, allow combined patterns from several target tissues. This is computationally expensive for more than two tissues. <code>iterating_seqVsInsitu</code> is faster thorugh calculating all combinations for n==2 and then using only the top tissues for n==3. The top tissues of n==3 is then are used for n==4 etc.
</p>
<p>III INTERPRETATION
</p>
<p><code>seqVsInsitu</code> and <code>iterating_seqVsInsitu</code> return the terms or term combinations together with a log2 probability score for each. They also produce two diagnostic graphs. If multiple tissues contribute to the sample, the scatterplot should show a number of clusters at low n. As n increases, the clusters should merge into just two clusters at the ideal value of n. The line graph shows the log2 probability distribution.
</p>
<p><code><a href="#topic+discovery_probability">discovery_probability</a></code> if RNASeq and in situ hybridisation data from the same tissue are paired, then with increasing FPKM the probability of RNA in situ discovery should increase logarithmically. If the tissue sources do not match, no such relationship should be visible. Using this function, if the tissue combination in the argument is a match, there should by a nearly linearly increasing relationship in the log-plot, with saturation at very high FPKM values only.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> cellOrigins</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2015-03-18</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>David Molnar
</p>
<p>Maintainer: David Molnar &lt;dmolnar100@icloud.com&gt;
</p>


<h3>References</h3>

<p>Molnar, D 2015, 'Single embryo-single organ transcriptomics of Drosophila embryos', PhD thesis, University of Cambridge. <br />
BDGP insitu: Tomancak, Genome Biol. 2007;8(7):R145. <br />
BDGP insitu homepage: insitu.fruitfly.org/cgi-bin/ex/insitu.pl
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pmoracle &lt;- seqVsInsitu(transcriptomeMatrix)
rownames(pmoracle)[1:3]
diagnosticPlots(pmoracle)

## End(Not run)
##loading the BDGP insitu probe coordinates if not
##copied directly from the package extdata folder
system.file("extdata", "BDGP_insitu_probes.bed", package = "cellOrigins")
</code></pre>

<hr>
<h2 id='diagnosticPlots'>

Diagnostic plots to explore seqVsInsitu results
</h2><span id='topic+diagnosticPlots'></span><span id='topic+diagnosticPlots.list'></span><span id='topic+diagnosticPlots.matrix'></span>

<h3>Description</h3>


<p>Accepts the result of <code><a href="#topic+seqVsInsitu">seqVsInsitu</a></code> and <code><a href="#topic+iterating_seqVsInsitu">iterating_seqVsInsitu</a></code> and produces diagnostic plots.
If the sequencing data fits to one or more terms or combinations of terms, then the scatterplot will cluster into foci.
As the number of combined terms is increased the foci merge into fewer groups. A diagonal in the scatterplot is a sign of error.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diagnosticPlots(seqVsInsitu_results)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diagnosticPlots_+3A_seqvsinsitu_results">seqVsInsitu_results</code></td>
<td>


<p>Value of <code><a href="#topic+seqVsInsitu">seqVsInsitu</a></code> or <code><a href="#topic+iterating_seqVsInsitu">iterating_seqVsInsitu</a></code>.
</p>
</td></tr>
</table>


<h3>Value</h3>






<p>None.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fpath &lt;- system.file("extdata", "vncMedianCoverage.tsv", package="cellOrigins")
vncExpression &lt;- read.delim(file = fpath, header=FALSE, as.is=TRUE)

expression &lt;- vncExpression$V2
names(expression) &lt;- vncExpression$V1

result &lt;- seqVsInsitu(expression, depth=1)
diagnosticPlots(result)

## Not run: 
oracleResponse &lt;- iterating_seqVsInsitu(expression, 3)
diagnosticPlots(oracleResponse)

## End(Not run)
</code></pre>

<hr>
<h2 id='discovery_probability'>
In situ discovery probability as a function of FPKM

</h2><span id='topic+discovery_probability'></span>

<h3>Description</h3>


<p>Groups transcripts by expression strength and calculates for each such group the percentage of genes that gave a positive staining signal in the in situ hybridisation.
</p>
<p>If the sequenced material matches the in situ hybridisation tissue, then weakly expressed genes in the sequenced material should be rearely in the in situ staining set of genes. Strongly expressed genes should correspondingly often also stain during hybridisation. Overall, if the match is not spurious, there should be a logarithmic dose-response relationship between sequencing read coverage and staining probability. In a plot of discovery probability against log(coverage) this shows as an approximately straight line (see example).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>discovery_probability(seq_signature, terms, cut.points,
    insitu=cellOrigins::BDGP_insitu_dmel_embryo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="discovery_probability_+3A_seq_signature">seq_signature</code></td>
<td>


<p>A named vector containing FPKM RNAseq data. Each element name must correspond to the names used in the <code>insitu</code> argument. NAs are permitted.
</p>
</td></tr>
<tr><td><code id="discovery_probability_+3A_terms">terms</code></td>
<td>


<p>A vector of anatomical terms which together are assumed to be the origin of the RNAseq data.
</p>
</td></tr>
<tr><td><code id="discovery_probability_+3A_cut.points">cut.points</code></td>
<td>


<p>A vector of cut points for grouping of values. E.g. 0:3 denotes the bins 0&lt;=x&lt;1, 1&lt;=x&lt;2, 2&lt;=x&lt;3.
</p>
</td></tr>
<tr><td><code id="discovery_probability_+3A_insitu">insitu</code></td>
<td>


<p>Matrix with in situ hybridisation data. Rows are transcript names (same names as used for <code>seq_signature</code>) and coloumns are anatomical terms (possibly combined with developmental stages). 1 denotes staining of a particular transcript in a particular tissue, 0 denotes no staining.
Defaults to <code><a href="#topic+BDGP_insitu_dmel_embryo">BDGP_insitu_dmel_embryo</a></code>, a staining dataset for <em>Drosophila melanogaster</em> embryos.
</p>
</td></tr>
</table>


<h3>Value</h3>






<p>A matrix with a row for each bin and three coloumns. The first coloumn is the probability of discovery, the second the number of transcripts in the expression bin that were discovered by in situ hybridisation. The third coloumn is the total number of transcripts in the bin.</p>


<h3>See Also</h3>


<p><code><a href="#topic+iterating_seqVsInsitu">iterating_seqVsInsitu</a></code>, <code><a href="#topic+BDGP_insitu_dmel_embryo">BDGP_insitu_dmel_embryo</a></code>, <code><a href="#topic+discovery.log">discovery.log</a></code>, <code><a href="#topic+discovery.linear">discovery.linear</a></code>, <code><a href="#topic+discovery.identic">discovery.identic</a></code>, <code><a href="#topic+prior.temporal_proximity_is_good">prior.temporal_proximity_is_good</a></code>, <code><a href="#topic+prior.all_equal">prior.all_equal</a></code>, <code><a href="#topic+diagnosticPlots">diagnosticPlots</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fpath &lt;- system.file("extdata", "vncMedianCoverage.tsv", package="cellOrigins")
vncExpression &lt;- read.delim(file = fpath, header=FALSE, as.is=TRUE)

expression &lt;- vncExpression$V2
names(expression) &lt;- vncExpression$V1

p &lt;- discovery_probability(expression,
  "6|ventral nerve cord", c(0, 2^(0:10)))

plot(x=-1:9, y=p[,1], type="l",
  xlab="log2(FPKM)", ylab="p(discovery in situ)")

</code></pre>

<hr>
<h2 id='discovery.log'>

Calculates discovery probability by RNA in situ hybridisation given a sequencing signal
</h2><span id='topic+discovery.log'></span><span id='topic+discovery.linear'></span><span id='topic+discovery.identic'></span>

<h3>Description</h3>


<p>A set of functions with different assumptions on the probability of RNA in situ staining, given a sequencing coverage.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>discovery.log(seq, saturate = 60, bias = 0.01)
discovery.linear(seq, saturate = 60, bias = 0.01)
discovery.identic(seq, saturate=Inf, bias=0)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="discovery.log_+3A_seq">seq</code></td>
<td>


<p>A vector of sequencing FPKMs.
</p>
</td></tr>
<tr><td><code id="discovery.log_+3A_saturate">saturate</code></td>
<td>


<p>FPKM value from which on maximum discovery probability (=0.99) is assumed (i.e. almost certain true positives).
Value of 60 is default, may need adjustment to sequencing coverage.
</p>
</td></tr>
<tr><td><code id="discovery.log_+3A_bias">bias</code></td>
<td>


<p>Positive staining probability of 0 FPKM transcripts (i.e. false positives). Must be &gt;0.
Default is 0.01, an empirically determined value.
</p>
</td></tr>
</table>


<h3>Details</h3>



<ol>
<li> <p><strong>discovery.log</strong> Uses a logarithmic saturation function for discovery probabilities. This relationship was empirically determined from sequencing and hybridisation data.
</p>
</li>
<li> <p><strong>discovery.linear</strong> Linear saturation function for discovery probabilities.
</p>
</li>
<li> <p><strong>discovery.identic</strong> Passes input through. Useful for comparing RNASeq Vs. RNASeq data. Also for cases when the discovery probability for each transcript has been already determined in some other way.
</p>
</li></ol>



<h3>Value</h3>






<p>A vector of probabilities. Element names are preserved.
</p>


<h3>See Also</h3>


<p><code><a href="#topic+seqVsInsitu">seqVsInsitu</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(0:80, discovery.log(0:80),
  ylim=c(0,1.1), type="l",
  xlab="FPKM", ylab="p(discovery insitu hybridization)")

plot(0:80, discovery.linear(0:80),
  ylim=c(0,1.1), type="l",
  xlab="FPKM", ylab="p(discovery insitu hybridization)")

</code></pre>

<hr>
<h2 id='iterating_seqVsInsitu'>

Faster comparisons between mixed tissue-specific RNA sequencing data and high-throughput RNA in situ hybridisation
</h2><span id='topic+iterating_seqVsInsitu'></span>

<h3>Description</h3>


<p>The same functionality as <code><a href="#topic+seqVsInsitu">seqVsInsitu</a></code> but computationally less expensive if combinations of anatomical terms are tested.
</p>
<p>The number of term combinations to test increases rapidly in <code><a href="#topic+seqVsInsitu">seqVsInsitu</a></code>. For example with 350 anatomical terms there are 61425 combinations of 2 terms and 7207200 combinations of 3 terms. This makes the exhaustive search of <code>seqVsInsitu</code> costly with depth&gt;2.
</p>
<p><code>iterating_seqVsInsitu</code> reduces the computational cost by initially testing the  combinations of only a few terms. Then in each iteration the cardinality of the combinations is increased by one, but only the top anatomical terms of the previous iteration are used to reduce the number of tested combinations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iterating_seqVsInsitu(seq_signature, upto_depth, use_topN = 50,
  start_depth = 2, insitu = cellOrigins::BDGP_insitu_dmel_embryo,
  insitu_discovery_function = discovery.log, saturate = 500,
  prior = prior.temporal_proximity_is_good)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="iterating_seqVsInsitu_+3A_seq_signature">seq_signature</code></td>
<td>


<p>A named vector containing FPKM RNAseq data. Each element name must correspond to the names used in the <code>insitu</code> argument. NAs are permitted.
</p>
</td></tr>
<tr><td><code id="iterating_seqVsInsitu_+3A_upto_depth">upto_depth</code></td>
<td>


<p>Number of terms to combine in the final iteration.
</p>
</td></tr>
<tr><td><code id="iterating_seqVsInsitu_+3A_use_topn">use_topN</code></td>
<td>


<p>How many of the top results from the previous iteration to use to find the terms for the current iteration.
</p>
</td></tr>
<tr><td><code id="iterating_seqVsInsitu_+3A_start_depth">start_depth</code></td>
<td>


<p>Number of terms to combine in the first iteration. All combinations of all terms are tested at this step.
</p>
</td></tr>
<tr><td><code id="iterating_seqVsInsitu_+3A_insitu">insitu</code></td>
<td>


<p>Matrix with RNA in situ hybridisation data. Rows are transcript names (queried by probes: same names as used for <code>seq_signature</code>) and coloumns are anatomical terms (possibly combined with developmental stages). If a probe stains in a particular tissue, the value is 1, otherwise 0.
Defaults to <code><a href="#topic+BDGP_insitu_dmel_embryo">BDGP_insitu_dmel_embryo</a></code>, a staining dataset for fruit fly embryos.
</p>
</td></tr>
<tr><td><code id="iterating_seqVsInsitu_+3A_insitu_discovery_function">insitu_discovery_function</code></td>
<td>


<p>A function that converts FPKM values to the probability of discovery by RNA in situ hybridisation. Values must be ]0..1[, 0 and 1 are not permitted.
Defaults to <code><a href="#topic+discovery.log">discovery.log</a></code>, an approximation of empirically determined discovery probabilities. Other available functions are <code><a href="#topic+discovery.linear">discovery.linear</a></code> and <code><a href="#topic+discovery.identic">discovery.identic</a></code>.
</p>
</td></tr>
<tr><td><code id="iterating_seqVsInsitu_+3A_saturate">saturate</code></td>
<td>


<p>Will be passed on to the <code>insitu_discovery_function</code>. The data set dependent maximum value at which the discovery probability should saturate.
Defaults to 500 (FPKM).
</p>
</td></tr>
<tr><td><code id="iterating_seqVsInsitu_+3A_prior">prior</code></td>
<td>


<p>A function that evaluates to the log2 prior probability of each anatomic term or combination of terms.
Defaults to <code><a href="#topic+prior.temporal_proximity_is_good">prior.temporal_proximity_is_good</a></code>, which works well with <code><a href="#topic+BDGP_insitu_dmel_embryo">BDGP_insitu_dmel_embryo</a></code>. <code><a href="#topic+prior.all_equal">prior.all_equal</a></code> assumes equal probability of all terms.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named list that contains a matrix for each iteration like those produced by <code><a href="#topic+seqVsInsitu">seqVsInsitu</a></code>.





</p>


<h3>See Also</h3>


<p><code><a href="#topic+seqVsInsitu">seqVsInsitu</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
fpath &lt;- system.file("extdata", "vncMedianCoverage.tsv", package="cellOrigins")
vncExpression &lt;- read.delim(file = fpath, header=FALSE, as.is=TRUE)

expression &lt;- vncExpression$V2
names(expression) &lt;- vncExpression$V1

oracleResponse &lt;- iterating_seqVsInsitu(expression, 3)
head(oracleResponse[[1]])
head(oracleResponse[[2]])
diagnosticPlots(oracleResponse)

## End(Not run)
</code></pre>

<hr>
<h2 id='prior.temporal_proximity_is_good'>

Assign a prior probability to a combination of anatomical terms
</h2><span id='topic+prior.temporal_proximity_is_good'></span><span id='topic+prior.all_equal'></span>

<h3>Description</h3>


<p>Accepts one or more anatomical terms and assigns to them a prior probability in the Bayesian sense.
<code>prior.all_equal</code> assumes all terms and combinations to be equally probable.
<code>prior.temporal_proximity_is_good</code> is meant mainly for use with <code><a href="#topic+BDGP_insitu_dmel_embryo">BDGP_insitu_dmel_embryo</a></code> if working with single or staged embryos.
With this function the prior probability increases if the developmental stages in the tested terms are close together.
The magnitude of the prior is scaled to the number of tested genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prior.temporal_proximity_is_good(term_pairs, insitu_signature)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prior.temporal_proximity_is_good_+3A_term_pairs">term_pairs</code></td>
<td>


<p>A vector with anatomical terms that are tested in combination.
</p>
</td></tr>
<tr><td><code id="prior.temporal_proximity_is_good_+3A_insitu_signature">insitu_signature</code></td>
<td>


<p>The RNA in situ hybridisation data set as produced by fusion of the expression patterns in <code>term_pairs</code>, and as it will be used for calculating the posterior probability in <code><a href="#topic+seqVsInsitu">seqVsInsitu</a></code>.
</p>
</td></tr>
</table>

<hr>
<h2 id='seqVsInsitu'>
Determine the most likely source(s) of a tissue-specific RNAseq dataset

</h2><span id='topic+seqVsInsitu'></span>

<h3>Description</h3>


<p>Compares tissue-specific RNA sequencing coverage with high-throughput RNA in situ hybridisation patterns of gene expression. All pattern combinations are tested in an exhaustive search.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seqVsInsitu(seq_signature, depth = 2, insitu = cellOrigins::BDGP_insitu_dmel_embryo,
  insitu_discovery_function = discovery.log, saturate = 500,
  prior = prior.temporal_proximity_is_good)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="seqVsInsitu_+3A_seq_signature">seq_signature</code></td>
<td>


<p>A named vector containing FPKM RNAseq data. Each element name must correspond to the names used in the <code>insitu</code> argument. NAs are permitted.
</p>
</td></tr>
<tr><td><code id="seqVsInsitu_+3A_depth">depth</code></td>
<td>


<p>Number of RNA in situ expression patterns to combine to identify mixed populations. If 1, the expression patterns as given are used. Otherwise all combinations of <code>depth</code> expression patterns are tried. Each term combined with itself is also tested i.e. pure populations will still be identified if depth&gt;1.
Defaults to 2. <code>seqVsInsitu</code> Depths &gt; 2 can be slow. <code><a href="#topic+iterating_seqVsInsitu">iterating_seqVsInsitu</a></code> is much faster in these cases.
</p>
</td></tr>
<tr><td><code id="seqVsInsitu_+3A_insitu">insitu</code></td>
<td>


<p>Matrix with RNA in situ hybridisation results. Rows are transcript names (same names as used for <code>seq_signature</code>) and coloumns are anatomical terms (possibly combined with developmental stages). 1 denotes staining of a particular transcript in a particular tissue, 0 denotes no staining.
Defaults to <code><a href="#topic+BDGP_insitu_dmel_embryo">BDGP_insitu_dmel_embryo</a></code>, a staining dataset for <em>Drosophila melanogaster</em> embryos.
</p>
</td></tr>
<tr><td><code id="seqVsInsitu_+3A_insitu_discovery_function">insitu_discovery_function</code></td>
<td>


<p>A function that converts FPKM values to the probability of discovery by RNA in situ hybridisation. Probabilities must be ]0..1[, the values 0 and 1 are not permitted.
Defaults to <code><a href="#topic+discovery.log">discovery.log</a></code>, an approximation of empirically determined discovery probabilities. Other available functions are <code><a href="#topic+discovery.linear">discovery.linear</a></code> and <code><a href="#topic+discovery.identic">discovery.identic</a></code>.
</p>
</td></tr>
<tr><td><code id="seqVsInsitu_+3A_saturate">saturate</code></td>
<td>


<p>Will be passed on to the <code>insitu_discovery_function</code>. The data set dependent maximum value at which discovery probability should saturate.
Defaults to 500 (FPKM).
</p>
</td></tr>
<tr><td><code id="seqVsInsitu_+3A_prior">prior</code></td>
<td>


<p>A function that returns the log2 prior probability of each anatomic term or combination of terms.
Defaults to <code><a href="#topic+prior.temporal_proximity_is_good">prior.temporal_proximity_is_good</a></code>, which works well with <code><a href="#topic+BDGP_insitu_dmel_embryo">BDGP_insitu_dmel_embryo</a></code>. <code><a href="#topic+prior.all_equal">prior.all_equal</a></code> assumes that all terms are equally probable.
</p>
</td></tr>
</table>


<h3>Details</h3>


<p>First, the function calculates for each sequenced transcript how likely it is that it would produce an RNA in situ signal, given its expresion strength. Using these staining probabilities and Bayes's rule the function then calculates the probability score for each of the given RNA in situ hybridisation patterns that it was produced by the same gene expression pattern as the sequenced transcriptome.
</p>
<p>If <code>depth</code>&gt;1 then the function identifies the origins of not pure sequenced material. For that it merges multiple RNA in situ hybridisation patterns for comparison with the sequenced data. This simulates the outcome of cell populations mixing.
</p>
<p><code>seq_signature</code> is best generated by taking the mean coverage of the regions which are actually tested with the RNA in situ hybridisation probes. This circumvents problems from misannotation, overlapping transcripts and faulty quantitation of individual transcripts from sequencing data. A protocol for generating such datasets is given in the package reference.
</p>


<h3>Value</h3>






<p>A matrix with a row for each anatomical term (or combination of terms) and at least four columns. The terms are sorted by the posterior value and the top term is the most likely source of the RNAseq transcriptome.
</p>
<table>
<tr><td><code>posterior</code></td>
<td>
<p>A log2 posterior probability score. The highest value is given to the most likely tissue of origin. The value is only meaningful in comparison with other values within the same result set.</p>
</td></tr>
<tr><td><code>prior</code></td>
<td>
<p>Prior probability of the anatomical term(s), as given by the function <code>prior</code>.</p>
</td></tr>
<tr><td><code>likelihood.from.absence.insitu</code></td>
<td>
<p>Probability score from all the genes where RNA in situ hybridisation did not report staining.</p>
</td></tr>
<tr><td><code>likelihood.from.presence.insitu</code></td>
<td>
<p>Probability score from all the genes where in situ hybridisation reported staining.</p>
</td></tr>
<tr><td><code>remaining coloumns</code></td>
<td>
<p>Number of additional expressed genes added to the in situ signature with each term in the tested combination. Sometimes additional terms add only very few or no new genes at all. Such tissue contributions are meaningless artefacts.</p>
</td></tr>
</table>
<p>The posterior column is the sum of the other three named columns. The scores are proportional to the (unknown) probabilities of identity.
</p>


<h3>See Also</h3>


<p><code><a href="#topic+iterating_seqVsInsitu">iterating_seqVsInsitu</a></code>, <code><a href="#topic+BDGP_insitu_dmel_embryo">BDGP_insitu_dmel_embryo</a></code>, <code><a href="#topic+discovery.log">discovery.log</a></code>, <code><a href="#topic+discovery.linear">discovery.linear</a></code>, <code><a href="#topic+discovery.identic">discovery.identic</a></code>, <code><a href="#topic+prior.temporal_proximity_is_good">prior.temporal_proximity_is_good</a></code>, <code><a href="#topic+prior.all_equal">prior.all_equal</a></code>, <code><a href="#topic+diagnosticPlots">diagnosticPlots</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fpath &lt;- system.file("extdata", "vncMedianCoverage.tsv", package="cellOrigins")
vncExpression &lt;- read.delim(file = fpath, header=FALSE, as.is=TRUE)

expression &lt;- vncExpression$V2
names(expression) &lt;- vncExpression$V1

result &lt;- seqVsInsitu(expression, depth=1)
</code></pre>

<hr>
<h2 id='vncMedianCoverage.tsv'>

Drosophila melanogaster embryo ventral nerve cord RNASeq coverage
</h2><span id='topic+vncMedianCoverage'></span>

<h3>Description</h3>


<p>Median RNAseq read coverages from 3 dissected embryonic (stage 11) fruit fly ventral nerve cords. The sequencing coverages are measured within the probing intervals of <a href="#topic+BDGP_insitu_dmel_embryo">high-confidence BDGP insitu probes</a>, as described in <a href="#topic+cellOrigins-package">cellOrigins-package</a>.
</p>


<h3>Format</h3>

<p>The format is: probe name, coverage, chromosome, probe beginn, probe end, strand.
</p>


<h3>Source</h3>


<p>Molnar, D 2015, 'Single embryo-single organ transcriptomics of Drosophila embryos', PhD thesis, University of Cambridge.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  fpath &lt;- system.file("extdata", "vncMedianCoverage.tsv", package="cellOrigins")
vncExpression &lt;- read.delim(file = fpath, header=FALSE, as.is=TRUE)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
