<!DOCTYPE html><html lang="en"><head><title>Help for package CePa</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CePa}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CePa-package'>
<p>Centrality-based pathway enrichment</p></a></li>
<li><a href='#cepa'>
<p>Apply CePa algorithm on a single pathway</p></a></li>
<li><a href='#cepa.all'>
<p>Apply CePa algorithm on a list of pathways under multiple centralities</p></a></li>
<li><a href='#cepa.all.parallel'>
<p>use CePa package through parallel computing</p></a></li>
<li><a href='#cepa.ora'>
<p>Apply centrality-extended ORA on a single pathway</p></a></li>
<li><a href='#cepa.ora.all'>
<p>Apply centrality-extented ORA on a list of pathways</p></a></li>
<li><a href='#cepa.univariate'>
<p>Apply centrality-extended GSA on a single pathway</p></a></li>
<li><a href='#cepa.univariate.all'>
<p>Apply centrality-extented GSA on a list of pathways</p></a></li>
<li><a href='#gene.list'>
<p>Differential gene list and background gene list</p></a></li>
<li><a href='#generate.pathway'>
<p>Generate igraph object from edge list</p></a></li>
<li><a href='#get.cepa'>
<p>get single cepa object from cepa.all object</p></a></li>
<li><a href='#p.table'>
<p>Table of p-values of pathways</p></a></li>
<li><a href='#pathway.nodes'>
<p>names of the pathway nodes</p></a></li>
<li><a href='#PID.db'>
<p>pathway catalogues from Pathway Interaction Database(PID)</p></a></li>
<li><a href='#plot.cepa'>
<p>Plot the cepa object</p></a></li>
<li><a href='#plot.cepa.all'>
<p>plot the cepa.all object</p></a></li>
<li><a href='#plot.pathway.catalogue'>
<p>plot pathway.catalogue object</p></a></li>
<li><a href='#plotGraph'>
<p>Plot graph for the pathway network</p></a></li>
<li><a href='#plotNull'>
<p>Plot the null distribution of the pathway score</p></a></li>
<li><a href='#print.cepa'>
<p>print the cepa object</p></a></li>
<li><a href='#print.cepa.all'>
<p>print the cepa.all object</p></a></li>
<li><a href='#print.pathway.catalogue'>
<p>print pathway.catalogue object</p></a></li>
<li><a href='#radiality'>
<p>Calculate radiality centrality</p></a></li>
<li><a href='#reach'>
<p>Calculate largest reach centrality</p></a></li>
<li><a href='#read.cls'>
<p>Read CLS file which stores the phenotype data</p></a></li>
<li><a href='#read.gct'>
<p>Read GCT format file which stores the expression values</p></a></li>
<li><a href='#report'>
<p>Generate report for CePa analysis</p></a></li>
<li><a href='#sampleLabel'>
<p>Generate data structure of sample labels</p></a></li>
<li><a href='#set.pathway.catalogue'>
<p>store pathway data and pre-processing</p></a></li>
<li><a href='#spread'>
<p>Calculate radiality centrality</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Centrality-Based Pathway Enrichment</td>
</tr>
<tr>
<td>Version:</td>
<td>0.8.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-10-07</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>igraph (&ge; 0.6), stats, graphics, methods, grDevices,
parallel, Rgraphviz, graph</td>
</tr>
<tr>
<td>Description:</td>
<td>It aims to find significant pathways through 
    network topology information. It has several advantages compared 
    with current pathway enrichment tools. First, pathway node instead 
    of single gene is taken as the basic unit when analysing networks 
    to meet the fact that genes must be constructed into complexes to 
    hold normal functions. Second, multiple network centrality measures are 
    applied simultaneously to measure importance of nodes from different 
    aspects to make a full view on the biological system. CePa extends 
    standard pathway enrichment methods, which include both over-representation 
    analysis procedure and gene-set analysis procedure. 
    &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtt008">doi:10.1093/bioinformatics/btt008</a>&gt;.</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/jokergoo/CePa">https://github.com/jokergoo/CePa</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-10-08 09:43:06 UTC; guz</td>
</tr>
<tr>
<td>Author:</td>
<td>Zuguang Gu <a href="https://orcid.org/0000-0002-7395-8709"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zuguang Gu &lt;z.gu@dkfz.de&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-10-08 10:10:27 UTC</td>
</tr>
</table>
<hr>
<h2 id='CePa-package'>
Centrality-based pathway enrichment
</h2><span id='topic+CePa-package'></span>

<h3>Description</h3>

<p>Centrality-based pathway enrichment
</p>


<h3>Details</h3>

<p>Gene set enrichment analysis is broadly used in microarray data analysis
aimed to find which biological functions are affected by a group of 
related genes behind the massive information. A lot of methods have been
developed under the framework of over-represented analysis (ORA) such
as <code>GOstats</code> and <code>GSEABase</code>. For a specific
form of gene sets, biological pathways are collections of correlated genes/proteins, 
RNAs and compounds that work together to regulate specific biological
processes. Instead of just being a list of genes, a pathway contains 
the most important information that is how the member genes interact 
with each other. Thus network structure information is necessary for
the intepretation of the importance of the pathways.
</p>
<p>In this package, the original pathway enrichment method
(ORA) is extended by introducing network centralities as the weight 
of nodes which have been mapped from differentially expressed genes 
in pathways. There are two advantages compared to former work.
First, for the diversity of genes' characters and the difficulties of 
covering the importance of genes from all aspects, we do not design a 
fixed measurement for each gene but set it as an optional parameter in the model. 
Researchers can select from candidate choices where different measurement 
reflects different aspect of the importance of genes. 
In our model, network centralities are used to measure the importance of genes in pathways. 
Different centrality measurements assign the importance to nodes from different aspects. 
For example, degree centrality measures the amount of neighbours that 
a node directly connects to, and betweenness centrality measures how many 
information streams must pass through a certain node. Generally speaking, 
nodes having large centrality values are central nodes in the network. 
It's observed that nodes represented as metabolites, proteins or genes 
with high centralities are essential to keep the steady state of biological networks. 
Moreover, different centrality measurements may relate to different biological functions. 
The selection of centralities for researchers depends on what kind of genes 
they think important. Second, we use nodes as the basic units of pathways 
instead of genes. We observe that nodes in the pathways include different 
types of molecules, such as single gene, complex and protein families. 
Assuming a complex or family contains ten differentially expressed member genes, 
in traditional ORA, these ten genes behave as the same position as other
genes represented as single nodes, and thus they have effect of ten. 
It is not proper because these ten genes stay in a same node in the 
pathway and make functions with the effect of one node. Also, 
a same gene may locate in different complexes in a pathway and if 
taking the gene with effect of one, it would greatly decrease the importance 
of the gene. Therefore a mapping procedure from genes to pathway nodes 
is applied in our model. What's more, the nodes in pathways also include 
none-gene nodes such as microRNAs and compounds. These nodes also 
contribute to the topology of the pathway. So, when analyzing pathways, 
all types of nodes are retained.
</p>
<p>The core function of the package is <code><a href="#topic+cepa.all">cepa.all</a></code>. There is also a parallel version
<code><a href="#topic+cepa.all.parallel">cepa.all.parallel</a></code>. User can refer to the vignette to find
how to use it (<code>vignette("CePa")</code>).
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>References</h3>

<p>Gu Z, Liu J, Cao K, Zhang J, Wang J. Centrality-based pathway enrichment: a systematic 
approach for finding significant pathways dominated by key genes. BMC Syst Biol. 2012 Jun 6;6(1):56.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# load the pathway database
data(PID.db)

# if you only have a differential gene list or other genes of interest
# in form of a list, you can apply the centrality-extended ORA method
res = cepa.all(dif = dif, bk = bk, pc = PID.db$NCI)
# in the above code, dif is your differential gene list, bk is your background
# gene list which always be whole set of genes on a certain microarray. If you 
# do not have a background gene list, do not set it and the function would use
# the whole human genome genes as default. pc is the pathway catalogue which in
# this example is the NCI catalogue gathered from PID database.

# after about 20 min, you can obtain a detailed report of the analysis
report(res)

# if you have a complete gene expression data, you can apply the centrality-extended
# GSA methods
res = cepa.all(mat = mat, label = label, pc = PID.db$NCI)
# mat is your expression value matrix, label is the design of the microarray experiment.
# By default, we use absolute value of t-value as the statistic for each gene and
# use the mean value of genes' statistics as the pathway statistic.

# after about 50 min, you can obtain a detailed report of the analysis
report(res)


## End(Not run)
</code></pre>

<hr>
<h2 id='cepa'>
Apply CePa algorithm on a single pathway
</h2><span id='topic+cepa'></span>

<h3>Description</h3>

<p>Apply CePa algorithm on a single pathway
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cepa(dif = NULL, bk = NULL, mat = NULL, label = NULL, pc, pathway = NULL,
    id = NULL, cen = "equal.weight",
    cen.name = if(is.function(cen)) deparse(substitute(cen))
    else if(mode(cen) == "name") deparse(cen) else cen,
    nlevel = "tvalue_abs", plevel = "mean", iter = 1000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cepa_+3A_dif">dif</code></td>
<td>
<p>differential gene list</p>
</td></tr>
<tr><td><code id="cepa_+3A_bk">bk</code></td>
<td>
<p>background gene list. If background gene list are not specified, use whole human genes</p>
</td></tr>
<tr><td><code id="cepa_+3A_mat">mat</code></td>
<td>
<p>expression matrix in which rows are genes and columns are samples</p>
</td></tr>
<tr><td><code id="cepa_+3A_label">label</code></td>
<td>
<p>a <code><a href="#topic+sampleLabel">sampleLabel</a></code> object identify the design of the microarray experiment</p>
</td></tr>
<tr><td><code id="cepa_+3A_pc">pc</code></td>
<td>
<p>a <code>pathway.catalogue</code> object storing information of pathways</p>
</td></tr>
<tr><td><code id="cepa_+3A_pathway">pathway</code></td>
<td>
<p>an <code><a href="igraph.html#topic+aaa-igraph-package">igraph</a></code> object or edge list</p>
</td></tr>
<tr><td><code id="cepa_+3A_id">id</code></td>
<td>
<p>identify which pathway should be analysis in the pathway catalogue</p>
</td></tr>
<tr><td><code id="cepa_+3A_cen">cen</code></td>
<td>
<p>centrality measuments, it can ce a string, or function has been quote</p>
</td></tr>
<tr><td><code id="cepa_+3A_cen.name">cen.name</code></td>
<td>
<p>centrality measurement names. This argument should be set if the <code>cen</code> is a function.</p>
</td></tr>
<tr><td><code id="cepa_+3A_nlevel">nlevel</code></td>
<td>
<p>node level transformation, should be one of &quot;tvalue&quot;, &quot;tvalue_sq&quot;, &quot;tvalue_abs&quot;. Also self-defined functions are allowed, see <code><a href="#topic+cepa.univariate">cepa.univariate</a></code> for detail.</p>
</td></tr>
<tr><td><code id="cepa_+3A_plevel">plevel</code></td>
<td>
<p>pathway level transformation, should be one of &quot;max&quot;, &quot;min&quot;, &quot;median&quot;, &quot;sum&quot;, &quot;mean&quot;, &quot;rank&quot;. Also, self-defined functions are allowed, see <code><a href="#topic+cepa.univariate">cepa.univariate</a></code> for detail.</p>
</td></tr>
<tr><td><code id="cepa_+3A_iter">iter</code></td>
<td>
<p>number of simulations</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is a wrapper of <code><a href="#topic+cepa.ora">cepa.ora</a></code> and <code><a href="#topic+cepa.univariate">cepa.univariate</a></code>.
Selection of which function depends on the arguments specified.
</p>
<p>If <code>dif</code>, <code>bk</code>, <code>pc</code>, <code>pathway</code>, <code>id</code>, <code>cen</code>, <code>cen.name</code> and <code>iter</code>
are specified, the arguments are passed to <code><a href="#topic+cepa.ora">cepa.ora</a></code>. The centrality-extension 
of over-representation analysis (ORA) will be applied on the list of differential genes.
</p>
<p>If <code>mat</code>, <code>label</code>, <code>pc</code>, <code>pathway</code>, <code>id</code>, <code>cen</code>, <code>cen.name</code>, <code>nlevel</code>,
<code>plevel</code> and <code>iter</code> are specified, the arguments are passed to <code><a href="#topic+cepa.univariate">cepa.univariate</a></code>.
The centrality-extension of gene-set analysis (GSA) will be applied on the whole gene expressions.
</p>
<p>This function is always called by <code><a href="#topic+cepa.all">cepa.all</a></code>. But you can still use it
if you want to analysis a single pathway under a specific centrality.
</p>


<h3>Value</h3>

<p>A <code><a href="#topic+cepa">cepa</a></code> class object
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cepa.all">cepa.all</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

data(PID.db)

# ORA extension
data(gene.list)
# will spend about 20 min
res.ora = cepa(dif = gene.list$dif, bk = gene.list$bk, pc = PID.db$NCI, id = 2)

# GSA extension
# P53_symbol.gct and P53_cls can be downloaded from
# http://mcube.nju.edu.cn/jwang/lab/soft/cepa/
eset = read.gct("P53_symbol.gct")
label = read.cls("P53.cls", treatment="MUT", control="WT")
# will take about 45 min
res.gsa = cepa(mat = eset, label = label, pc = PID.db$NCI, id = 2)

## End(Not run)
</code></pre>

<hr>
<h2 id='cepa.all'>
Apply CePa algorithm on a list of pathways under multiple centralities
</h2><span id='topic+cepa.all'></span>

<h3>Description</h3>

<p>Apply CePa algorithm on a list of pathways under multiple centralities
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cepa.all(dif = NULL, bk = NULL, mat = NULL, label = NULL, pc, cen = default.centralities,
    cen.name = sapply(cen, function(x) ifelse(mode(x) == "name", deparse(x), x)),
    nlevel = "tvalue_abs", plevel = "mean", iter = 1000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cepa.all_+3A_dif">dif</code></td>
<td>
<p>differential gene list</p>
</td></tr>
<tr><td><code id="cepa.all_+3A_bk">bk</code></td>
<td>
<p>background gene list. If background gene list are not specified, use whole human genes</p>
</td></tr>
<tr><td><code id="cepa.all_+3A_mat">mat</code></td>
<td>
<p>expression matrix in which rows are genes and columns are samples</p>
</td></tr>
<tr><td><code id="cepa.all_+3A_label">label</code></td>
<td>
<p>a <code><a href="#topic+sampleLabel">sampleLabel</a></code> object identify the design of the microarray experiment</p>
</td></tr>
<tr><td><code id="cepa.all_+3A_pc">pc</code></td>
<td>
<p>a <code>pathway.catalogue</code> object storing information of pathways</p>
</td></tr>
<tr><td><code id="cepa.all_+3A_cen">cen</code></td>
<td>
<p>centrality measuments, it can ce a string, or a function</p>
</td></tr>
<tr><td><code id="cepa.all_+3A_cen.name">cen.name</code></td>
<td>
<p>centrality measurement names. By default it is parsed from <code>cen</code> argument</p>
</td></tr>
<tr><td><code id="cepa.all_+3A_nlevel">nlevel</code></td>
<td>
<p>node level transformation, should be one of &quot;tvalue&quot;, &quot;tvalue_sq&quot;, &quot;tvalue_abs&quot;. Also self-defined functions are allowed, see <code><a href="#topic+cepa.univariate.all">cepa.univariate.all</a></code> for detail.</p>
</td></tr>
<tr><td><code id="cepa.all_+3A_plevel">plevel</code></td>
<td>
<p>pathway level transformation, should be one of &quot;max&quot;, &quot;min&quot;, &quot;median&quot;, &quot;sum&quot;, &quot;mean&quot;, &quot;rank&quot;. Also, self-defined functions are allowed, see <code><a href="#topic+cepa.univariate.all">cepa.univariate.all</a></code> for detail.</p>
</td></tr>
<tr><td><code id="cepa.all_+3A_iter">iter</code></td>
<td>
<p>number of simulations</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All the calculation can be achieved by this function. The function is wrapper of 
both ORA extension and GSA extension. It chooses corresponding procedure according 
to the arguments specified. If the arguments contain gene lists, then the calculation 
is sent to functions doing ORA extension. While if the arguments contain an expression 
matrix and a phenotype label, the GSA extension is evoked.
</p>
<p>The function is a wrapper of <code><a href="#topic+cepa.ora.all">cepa.ora.all</a></code> and <code><a href="#topic+cepa.univariate.all">cepa.univariate.all</a></code>.
</p>
<p>This is the core function of the package. User can refer to the vignette to find
how to use it (<code>vignette("CePa")</code>).
</p>
<p>If <code>dif</code>, <code>bk</code>, <code>pc</code>, <code>cen</code>, <code>cen.name</code> and <code>iter</code>
are specified, the arguments are passed to <code>cepa.ora.all</code>. The centrality-extension 
of over-representation analysis (ORA) will be applied on the list of differential genes.
</p>
<p>If <code>mat</code>, <code>label</code>, <code>pc</code>, <code>cen</code>, <code>cen.name</code>, <code>nlevel</code>,
<code>plevel</code> and <code>iter</code> are specified, the arguments are passed to <code>cepa.univariate.all</code>.
The centrality-extension of gene-set analysis (GSA) will be applied on the whole gene expressions.
</p>
<p>There is a parallel version of the function: <code><a href="#topic+cepa.all.parallel">cepa.all.parallel</a></code>.
</p>


<h3>Value</h3>

<p>A <code><a href="#topic+cepa.all">cepa.all</a></code> class object
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>References</h3>

<p>Gu Z, Liu J, Cao K, Zhang J, Wang J. Centrality-based pathway enrichment: a systematic 
approach for finding significant pathways dominated by key genes. BMC Syst Biol. 2012 Jun 6;6(1):56.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cepa">cepa</a></code>, <code><a href="#topic+cepa.ora.all">cepa.ora.all</a></code>, <code><a href="#topic+cepa.univariate.all">cepa.univariate.all</a></code>, <code><a href="#topic+cepa.all.parallel">cepa.all.parallel</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

data(PID.db)

# ORA extension
data(gene.list)
# will spend about 20 min
res.ora = cepa.all(dif = gene.list$dif, bk = gene.list$bk, pc = PID.db$NCI)

# GSA extension
# P53_symbol.gct and P53_cls can be downloaded from
# http://mcube.nju.edu.cn/jwang/lab/soft/cepa/
eset = read.gct("http://mcube.nju.edu.cn/jwang/lab/soft/cepa/P53_symbol.gct")
label = read.cls("http://mcube.nju.edu.cn/jwang/lab/soft/cepa/P53.cls", 
    treatment="MUT", control="WT")
# will spend about 45 min
res.gsa = cepa.all(mat = eset, label = label, pc = PID.db$NCI)

## End(Not run)
</code></pre>

<hr>
<h2 id='cepa.all.parallel'>
use CePa package through parallel computing
</h2><span id='topic+cepa.all.parallel'></span>

<h3>Description</h3>

<p>use CePa package through parallel computing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cepa.all.parallel(dif = NULL, bk = NULL, mat = NULL, label = NULL,
    pc, cen = default.centralities,
    cen.name = sapply(cen, function(x) ifelse(mode(x) == "name", deparse(x), x)),
    nlevel = "tvalue_abs", plevel = "mean", iter = 1000, ncores = 2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cepa.all.parallel_+3A_dif">dif</code></td>
<td>
<p>differential gene list</p>
</td></tr>
<tr><td><code id="cepa.all.parallel_+3A_bk">bk</code></td>
<td>
<p>background gene list. If background gene list are not specified, use whole human genes</p>
</td></tr>
<tr><td><code id="cepa.all.parallel_+3A_mat">mat</code></td>
<td>
<p>expression matrix in which rows are genes and columns are samples</p>
</td></tr>
<tr><td><code id="cepa.all.parallel_+3A_label">label</code></td>
<td>
<p>a <code><a href="#topic+sampleLabel">sampleLabel</a></code> object identify the design of the microarray experiment</p>
</td></tr>
<tr><td><code id="cepa.all.parallel_+3A_pc">pc</code></td>
<td>
<p>a <code>pathway.catalogue</code> object storing information of pathways</p>
</td></tr>
<tr><td><code id="cepa.all.parallel_+3A_cen">cen</code></td>
<td>
<p>centrality measuments, it can ce a string, or a function</p>
</td></tr>
<tr><td><code id="cepa.all.parallel_+3A_cen.name">cen.name</code></td>
<td>
<p>centrality measurement names. By default it is parsed from <code>cen</code> argument</p>
</td></tr>
<tr><td><code id="cepa.all.parallel_+3A_nlevel">nlevel</code></td>
<td>
<p>node level transformation, should be one of &quot;tvalue&quot;, &quot;tvalue_sq&quot;, &quot;tvalue_abs&quot;. Also self-defined functions are allowed, see <code><a href="#topic+cepa.univariate.all">cepa.univariate.all</a></code> for detail.</p>
</td></tr>
<tr><td><code id="cepa.all.parallel_+3A_plevel">plevel</code></td>
<td>
<p>pathway level transformation, should be one of &quot;max&quot;, &quot;min&quot;, &quot;median&quot;, &quot;sum&quot;, &quot;mean&quot;, &quot;rank&quot;. Also, self-defined functions are allowed, see <code><a href="#topic+cepa.univariate.all">cepa.univariate.all</a></code> for detail.</p>
</td></tr>
<tr><td><code id="cepa.all.parallel_+3A_iter">iter</code></td>
<td>
<p>number of simulations</p>
</td></tr>
<tr><td><code id="cepa.all.parallel_+3A_ncores">ncores</code></td>
<td>
<p>number of cores for parallel computing</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function divides the pathway list into several parts and each part is sent to a core for 
parallel computing.
</p>
<p>The package for parallel computing is <code>snow</code>.
</p>
<p>Note: there may be warnings saying connections not closed. In fact I have closed
connections after the parallel computing is done. I don't know why this
happens. Maybe you breaked the computing ahead manually. However it does not matter 
unless you have obsessive compulsive disorder.
</p>


<h3>Value</h3>

<p>A <code><a href="#topic+cepa.all">cepa.all</a></code> class object
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>References</h3>

<p>Gu Z, Liu J, Cao K, Zhang J, Wang J. Centrality-based pathway enrichment: a systematic 
approach for finding significant pathways dominated by key genes. BMC Syst Biol. 2012 Jun 6;6(1):56.
</p>


<h3>See Also</h3>

<p>cepa.all
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(PID.db)
# ORA extension
data(gene.list)
res.ora = cepa.all.parallel(dif = gene.list$dif, bk = gene.list$bk, pc = PID.db$NCI, ncores = 4)
# GSA extension
# P53_symbol.gct and P53_cls can be downloaded from
# http://mcube.nju.edu.cn/jwang/lab/soft/cepa/
eset = read.gct("http://mcube.nju.edu.cn/jwang/lab/soft/cepa/P53_symbol.gct")
label = read.cls("http://mcube.nju.edu.cn/jwang/lab/soft/cepa/P53.cls", 
    treatment="MUT", control="WT")
res.gsa = cepa.all.parallel(mat = eset, label = label, pc = PID.db$NCI, ncores = 4)

## End(Not run)
</code></pre>

<hr>
<h2 id='cepa.ora'>
Apply centrality-extended ORA on a single pathway
</h2><span id='topic+cepa.ora'></span>

<h3>Description</h3>

<p>Apply centrality-extended ORA on a single pathway
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cepa.ora(dif, pc, bk = NULL, pathway = NULL, id = NULL, cen = "equal.weight",
    cen.name = if(is.function(cen)) deparse(substitute(cen))
    else if(mode(cen) == "name") deparse(cen)
    else cen,
    iter = 1000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cepa.ora_+3A_dif">dif</code></td>
<td>
<p>differential gene list</p>
</td></tr>
<tr><td><code id="cepa.ora_+3A_pc">pc</code></td>
<td>
<p>a <code>pathway.catalogue</code> class object</p>
</td></tr>
<tr><td><code id="cepa.ora_+3A_bk">bk</code></td>
<td>
<p>background gene list. If background gene list are not specified, use whole human genes</p>
</td></tr>
<tr><td><code id="cepa.ora_+3A_pathway">pathway</code></td>
<td>
<p><code><a href="igraph.html#topic+aaa-igraph-package">igraph</a></code> object or edge list</p>
</td></tr>
<tr><td><code id="cepa.ora_+3A_id">id</code></td>
<td>
<p>identify which pathway in the catalogue</p>
</td></tr>
<tr><td><code id="cepa.ora_+3A_cen">cen</code></td>
<td>
<p>centrality measuments, it can ce a string, function, or function that has been quoted</p>
</td></tr>
<tr><td><code id="cepa.ora_+3A_cen.name">cen.name</code></td>
<td>
<p>centrality measurement names. This argument should be set if the <code>cen</code> is a function.</p>
</td></tr>
<tr><td><code id="cepa.ora_+3A_iter">iter</code></td>
<td>
<p>number of simulations</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is always called by <code><a href="#topic+cepa.ora.all">cepa.ora.all</a></code>. But you can still
use it if you realy want to analysis just one pathway under one centrality.
</p>


<h3>Value</h3>

<p>A <code>cepa</code> class object
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cepa.all">cepa.all</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(PID.db)

# ORA extension
data(gene.list)
# will spend about 20 min
res.ora = cepa(dif = gene.list$dif, bk = gene.list$bk, pc = PID.db$NCI, id = 2)

## End(Not run)
</code></pre>

<hr>
<h2 id='cepa.ora.all'>
Apply centrality-extented ORA on a list of pathways
</h2><span id='topic+cepa.ora.all'></span>

<h3>Description</h3>

<p>Apply centrality-extented ORA on a list of pathways
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cepa.ora.all(dif, pc, bk = NULL, cen = default.centralities,
    cen.name = sapply(cen, function(x) ifelse(mode(x) == "name", deparse(x), x)),
    iter = 1000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cepa.ora.all_+3A_dif">dif</code></td>
<td>
<p>differential gene list</p>
</td></tr>
<tr><td><code id="cepa.ora.all_+3A_pc">pc</code></td>
<td>
<p>a <code>pathway.catalogue</code> class object</p>
</td></tr>
<tr><td><code id="cepa.ora.all_+3A_bk">bk</code></td>
<td>
<p>background gene list. If background gene list are not specified, use whole human genes</p>
</td></tr>
<tr><td><code id="cepa.ora.all_+3A_cen">cen</code></td>
<td>
<p>centrality measuments, it can ce a string, or a function</p>
</td></tr>
<tr><td><code id="cepa.ora.all_+3A_cen.name">cen.name</code></td>
<td>
<p>centrality measurement names. By default it is parsed from <code>cen</code> argument</p>
</td></tr>
<tr><td><code id="cepa.ora.all_+3A_iter">iter</code></td>
<td>
<p>number of simulations</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The traditional over-representation analysis (ORA) to find significant pathways 
uses a 2x2 contingency table to test the independency of genes belonging to a 
functional category and these genes being differentially expressed, usually by 
Fisher's exact test. The ORA only consider the number of genes and the function
extend traditional ORA with network centralities.
</p>
<p>The differential gene list and the background gene list should be indicated
with the same identifiers (e.g. gene symbol or refseq ID). All genes in
the differential gene list should exist in the background gene list. If users 
use the <code><a href="#topic+PID.db">PID.db</a></code> data, all genes should be formatted in gene symbol.
</p>
<p>If the centrality measurement is set as a string, only pre-defined &quot;equal.weight&quot;,
&quot;in.degree&quot;, &quot;out.degree&quot;, &quot;degree&quot;, &quot;betweenness&quot;, &quot;in.reach&quot;, &quot;out.reach&quot;,
&quot;reach&quot;, &quot;in.spread&quot;, &quot;out.spread&quot; and &quot;spread&quot; are allowed. More centrality
measurements can be used by setting it as a function (such as closeness,
cluster coefficient). In the function, we recommand users choose
at least two centrality measurements. The default centralities are &quot;equal.weight&quot;,
&quot;in.degree&quot;, &quot;out.degree&quot;, &quot;betweenness&quot;, &quot;in.reach&quot; and &quot;out.reach&quot;.
</p>
<p>However, in most circumstance, the function is called by <code><a href="#topic+cepa.all">cepa.all</a></code>.
</p>


<h3>Value</h3>

<p>A <code><a href="#topic+cepa.all">cepa.all</a></code> class object
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(PID.db)
# ORA extension
data(gene.list)
# will spend about 20 min
res.ora = cepa.ora.all(dif = gene.list$dif, bk = gene.list$bk, pc = PID.db$NCI)

## End(Not run)
</code></pre>

<hr>
<h2 id='cepa.univariate'>
Apply centrality-extended GSA on a single pathway
</h2><span id='topic+cepa.univariate'></span>

<h3>Description</h3>

<p>Apply centrality-extended GSA on a single pathway
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cepa.univariate(mat, label, pc, pathway = NULL, id = NULL, cen = "equal.weight",
    cen.name = if(is.function(cen)) deparse(substitute(cen))
    else if(mode(cen) == "name") deparse(cen)
    else cen,
    iter = 1000, nlevel = "tvalue_abs", plevel = "mean",
    node.level.from.expr = NULL, node.level.t.value = NULL,
    r.node.level.from.expr = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cepa.univariate_+3A_mat">mat</code></td>
<td>
<p>expression matrix in which rows are genes and columns are samples</p>
</td></tr>
<tr><td><code id="cepa.univariate_+3A_label">label</code></td>
<td>
<p>a <code><a href="#topic+sampleLabel">sampleLabel</a></code> object identify the design of the microarray experiment</p>
</td></tr>
<tr><td><code id="cepa.univariate_+3A_pc">pc</code></td>
<td>
<p>a <code>pathway.catalogue</code> object storing information of pathways</p>
</td></tr>
<tr><td><code id="cepa.univariate_+3A_pathway">pathway</code></td>
<td>
<p><code><a href="igraph.html#topic+aaa-igraph-package">igraph</a></code> object or edge list</p>
</td></tr>
<tr><td><code id="cepa.univariate_+3A_id">id</code></td>
<td>
<p>identify the number of the pathway in the catalogue</p>
</td></tr>
<tr><td><code id="cepa.univariate_+3A_cen">cen</code></td>
<td>
<p>centrality measuments, it can ce a string, or function has been quote</p>
</td></tr>
<tr><td><code id="cepa.univariate_+3A_cen.name">cen.name</code></td>
<td>
<p>centrality measurement names</p>
</td></tr>
<tr><td><code id="cepa.univariate_+3A_nlevel">nlevel</code></td>
<td>
<p>node level transformation, should be one of &quot;tvalue&quot;, &quot;tvalue_sq&quot;, &quot;tvalue_abs&quot;. Also self-defined functions are allowed, see <code><a href="#topic+cepa.univariate.all">cepa.univariate.all</a></code> for detail.</p>
</td></tr>
<tr><td><code id="cepa.univariate_+3A_plevel">plevel</code></td>
<td>
<p>pathway level transformation, should be one of &quot;max&quot;, &quot;min&quot;, &quot;median&quot;, &quot;sum&quot;, &quot;mean&quot;, &quot;rank&quot;. Also, self-defined functions are allowed, see <code><a href="#topic+cepa.univariate.all">cepa.univariate.all</a></code> for detail.</p>
</td></tr>
<tr><td><code id="cepa.univariate_+3A_node.level.from.expr">node.level.from.expr</code></td>
<td>
<p>for simplicity of computing</p>
</td></tr>
<tr><td><code id="cepa.univariate_+3A_node.level.t.value">node.level.t.value</code></td>
<td>
<p>for simplicity of computing</p>
</td></tr>
<tr><td><code id="cepa.univariate_+3A_r.node.level.from.expr">r.node.level.from.expr</code></td>
<td>
<p>for simplicity of computing</p>
</td></tr>
<tr><td><code id="cepa.univariate_+3A_iter">iter</code></td>
<td>
<p>number of simulations</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is always called by <code><a href="#topic+cepa.univariate.all">cepa.univariate.all</a></code>. But you can still
use it if you realy want to analysis just one pathway under one centrality.
</p>


<h3>Value</h3>

<p>A <code><a href="#topic+cepa">cepa</a></code> class object
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

data(PID.db)

# GSA extension
# P53_symbol.gct and P53_cls can be downloaded from
# http://mcube.nju.edu.cn/jwang/lab/soft/cepa/
eset = read.gct("P53_symbol.gct")
label = read.cls("P53.cls", treatment="MUT", control="WT")
# will spend about 45 min
res.gsa = cepa.univariate(mat = eset, label = label, pc = PID.db$NCI, id = 2)

## End(Not run)
</code></pre>

<hr>
<h2 id='cepa.univariate.all'>
Apply centrality-extented GSA on a list of pathways
</h2><span id='topic+cepa.univariate.all'></span>

<h3>Description</h3>

<p>Apply centrality-extented GSA on a list of pathways
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cepa.univariate.all(mat, label, pc, cen = default.centralities,
    cen.name = sapply(cen, function(x) ifelse(mode(x) == "name", deparse(x), x)),
    nlevel = "tvalue_abs", plevel = "mean", iter = 1000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cepa.univariate.all_+3A_mat">mat</code></td>
<td>
<p>expression matrix in which rows are genes and columns are samples</p>
</td></tr>
<tr><td><code id="cepa.univariate.all_+3A_label">label</code></td>
<td>
<p>a <code><a href="#topic+sampleLabel">sampleLabel</a></code> object identify the design of the microarray experiment</p>
</td></tr>
<tr><td><code id="cepa.univariate.all_+3A_pc">pc</code></td>
<td>
<p>a <code>pathway.catalogue</code> object storing information of pathways</p>
</td></tr>
<tr><td><code id="cepa.univariate.all_+3A_cen">cen</code></td>
<td>
<p>centrality measuments, it can ce a string, or a function</p>
</td></tr>
<tr><td><code id="cepa.univariate.all_+3A_cen.name">cen.name</code></td>
<td>
<p>centrality measurement names. By default it is parsed from <code>cen</code> argument</p>
</td></tr>
<tr><td><code id="cepa.univariate.all_+3A_nlevel">nlevel</code></td>
<td>
<p>node level transformation, should be one of &quot;tvalue&quot;, &quot;tvalue_sq&quot;, &quot;tvalue_abs&quot;. Also self-defined functions are allowed</p>
</td></tr>
<tr><td><code id="cepa.univariate.all_+3A_plevel">plevel</code></td>
<td>
<p>pathway level transformation, should be one of &quot;max&quot;, &quot;min&quot;, &quot;median&quot;, &quot;sum&quot;, &quot;mean&quot;, &quot;rank&quot;. Also, self-defined functions are allowed</p>
</td></tr>
<tr><td><code id="cepa.univariate.all_+3A_iter">iter</code></td>
<td>
<p>number of simulations</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The traditional gene-set analysis (GSA) to find significant pathways 
uses the whole expression matrix. GSA methods are implemented via either a univariate 
or a multivariate procedure. In univariate analysis, node level statistics are 
initially calculated from fold changes or statistical tests (e.g., t-test). 
These statistics are then combined into a pathway level statistic by summation or 
averaging. Multivariate analysis considers the correlations between genes in the 
pathway and calculates the pathway level statistic directly from the expression 
value matrix using Hotelling's T^2 test or MANOVA models. The function implement
univariate procedure of GSA with network centralities.
</p>
<p>If users use the <code><a href="#topic+PID.db">PID.db</a></code> data, all genes should be formatted in gene symbol.
</p>
<p>If the centrality measurement is set as a string, only pre-defined &quot;equal.weight&quot;,
&quot;in.degree&quot;, &quot;out.degree&quot;, &quot;degree&quot;, &quot;betweenness&quot;, &quot;in.reach&quot;, &quot;out.reach&quot;,
&quot;reach&quot;, &quot;in.spread&quot;, &quot;out.spread&quot; and &quot;spread&quot; are allowed. More centrality
measurements can be used by setting it as a function (such as closeness,
cluster coefficient). In the function, we recommand users choose
at least two centrality measurements. Note that the self-defined function should
only contain one argument which is an igraph object. The default centralities are &quot;equal.weight&quot;,
&quot;in.degree&quot;, &quot;out.degree&quot;, &quot;betweenness&quot;, &quot;in.reach&quot; and &quot;out.reach&quot;.
</p>
<p>The node level statistic can be self-defined. The self-defined function should contain
two arguments: a vector for expression value in treatment class and a vector for
expression value in control class.
</p>
<p>The pathway level statistic can be self-defined. The self-defined function should
only contain one argument: the vector of node-level statistic.
</p>
<p>However, in most circumstance, the function is called by <code><a href="#topic+cepa.all">cepa.all</a></code>.
</p>
<p>We are sorry that only the univariate procedures in GSA are extended. We are still
trying to figure out the extension for the multivariate procedures in GSA.
</p>


<h3>Value</h3>

<p>A <code><a href="#topic+cepa.all">cepa.all</a></code> class object
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cepa">cepa</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(PID.db)
# GSA extension
# P53_symbol.gct and P53.cls can be downloaded from
# http://mcube.nju.edu.cn/jwang/lab/soft/cepa/
eset = read.gct("http://mcube.nju.edu.cn/jwang/lab/soft/cepa/P53_symbol.gct")
label = read.cls("http://mcube.nju.edu.cn/jwang/lab/soft/cepa/P53.cls", 
    treatment="MUT", control="WT")
# will spend about 45 min
res.gsa = cepa.univariate.all(mat = eset, label = label, pc = PID.db$NCI)

## End(Not run)
</code></pre>

<hr>
<h2 id='gene.list'>
Differential gene list and background gene list
</h2><span id='topic+gene.list'></span>

<h3>Description</h3>

<p>Differential gene list and background gene list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gene.list)
</code></pre>


<h3>Details</h3>

<p>Differential gene list and background gene list was extracted from
microarray data from GEO database. The accession number for the data set
is GSE22058. The t-test was applied to find differentially expressed genes.
Top 2000 genes were selected as the gene list.
</p>


<h3>Value</h3>

<p>A list containing two componets:
</p>

<dl>
<dt><code>bk</code></dt><dd><p>background gene list, gene symbol</p>
</dd>
<dt><code>dif</code></dt><dd><p>differentially expressed gene list, gene symbol</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(gene.list)
names(gene.list)
</code></pre>

<hr>
<h2 id='generate.pathway'>
Generate igraph object from edge list
</h2><span id='topic+generate.pathway'></span>

<h3>Description</h3>

<p>Generate igraph object from edge list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate.pathway(el)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="generate.pathway_+3A_el">el</code></td>
<td>
<p>edge list, matrix with two columns. The first column is the input node and the second column is the output node.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is a wrapper of <code><a href="igraph.html#topic+graph.edgelist">graph.edgelist</a></code> and it generates
a directed graph.
</p>
<p>In the function, repeated edged for two nodes will be eliminated.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cepa">cepa</a></code>, <code><a href="igraph.html#topic+graph.edgelist">graph.edgelist</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>edgelist = rbind(c("a", "b"), c("a", "b"), c("a", "c"))
g = generate.pathway(edgelist)
</code></pre>

<hr>
<h2 id='get.cepa'>
get single cepa object from cepa.all object
</h2><span id='topic+get.cepa'></span>

<h3>Description</h3>

<p>get single cepa object from cepa.all object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.cepa(x, id = NULL, cen = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.cepa_+3A_x">x</code></td>
<td>
<p>a <code><a href="#topic+cepa.all">cepa.all</a></code> object</p>
</td></tr>
<tr><td><code id="get.cepa_+3A_id">id</code></td>
<td>
<p>index or the name of the pathway</p>
</td></tr>
<tr><td><code id="get.cepa_+3A_cen">cen</code></td>
<td>
<p>index or the name of the centrality</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 'cepa.all object contains the result for pathways under several centrality
measurements. In <code><a href="#topic+cepa.all">cepa.all</a></code> object, each pathway under a specific centrality
is a single <code><a href="#topic+cepa">cepa</a></code> object. The <code><a href="#topic+get.cepa">get.cepa</a></code> function is used to get the <code><a href="#topic+cepa">cepa</a></code>
object from the <code><a href="#topic+cepa.all">cepa.all</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cepa">cepa</a></code>, <code><a href="#topic+cepa.all">cepa.all</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(PID.db)

# ORA extension
data(gene.list)
# will spend about 20 min
res.ora = cepa.all(dif = gene.list$dif, bk = gene.list$bk, pc = PID.db$NCI)
ora = get.cepa(res.ora, id = 5, cen = 3)

# GSA extension
# P53_symbol.gct and P53_cls can be downloaded from
# http://mcube.nju.edu.cn/jwang/lab/soft/cepa/
eset = read.gct("P53_symbol.gct")
label = read.cls("P53.cls", treatment="MUT", control="WT")
# will spend about 45 min
res.gsa = cepa.all(mat = eset, label = label, pc = PID.db$NCI)
gsa = get.cepa(res.gsa, id = 5, cen = 3)

## End(Not run)
</code></pre>

<hr>
<h2 id='p.table'>
Table of p-values of pathways
</h2><span id='topic+p.table'></span>

<h3>Description</h3>

<p>Table of p-values of pathways
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p.table(x, adj.method = NA, cutoff = ifelse(adj.method == "none", 0.01, 0.05))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p.table_+3A_x">x</code></td>
<td>
<p>a <code><a href="#topic+cepa.all">cepa.all</a></code> object</p>
</td></tr>
<tr><td><code id="p.table_+3A_adj.method">adj.method</code></td>
<td>
<p>methods in <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>, available methods are &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;</p>
</td></tr>
<tr><td><code id="p.table_+3A_cutoff">cutoff</code></td>
<td>
<p>cutoff for significance</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Since the p-values for each pathway are calculated for several centralities, the
whole p-values are represented as a table.
</p>
<p>Also it can extract significant pathways only.
</p>


<h3>Value</h3>

<p>A data matrix where rows are pathways and columns are centralities.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cepa.all">cepa.all</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(PID.db)

# ORA extension
data(gene.list)
# will spend about 20 min
res.ora = cepa.all(dif = gene.list$dif, bk = gene.list$bk, pc = PID.db$NCI)
p.table(res.ora)
p.table(res.ora, adj.method = "BH")

# GSA extension
# P53_symbol.gct and P53_cls can be downloaded from
# http://mcube.nju.edu.cn/jwang/lab/soft/cepa/
eset = read.gct("P53_symbol.gct")
label = read.cls("P53.cls", treatment="MUT", control="WT")
# will spend about 45 min
res.gsa = cepa.all(mat = eset, label = label, pc = PID.db$NCI)
p.table(res.gsa)

## End(Not run)
</code></pre>

<hr>
<h2 id='pathway.nodes'>
names of the pathway nodes
</h2><span id='topic+pathway.nodes'></span>

<h3>Description</h3>

<p>names of the pathway nodes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathway.nodes(pathway)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pathway.nodes_+3A_pathway">pathway</code></td>
<td>
<p>an <code><a href="igraph.html#topic+aaa-igraph-package">igraph</a></code> object</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If nodes in the pathway have names, then it returns a vector of nodes
names. If nodes in the pathway have no name, it just returns the index
of nodes (start from 1, after igraph version 0.6).
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>interaction = rbind(c("a", "b"),
                    c("a", "c"))
g = generate.pathway(interaction)
pathway.nodes(g)
</code></pre>

<hr>
<h2 id='PID.db'>
pathway catalogues from Pathway Interaction Database(PID)
</h2><span id='topic+PID.db'></span>

<h3>Description</h3>

<p>pathway catalogues from Pathway Interaction Database(PID)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(PID.db)
</code></pre>


<h3>Details</h3>

<p>The pathway data is parsed from XML format file provided by PID FTP site.
</p>
<p>There are four pathway catalogues which are NCI_Nature, BioCarta, KEGG
and Reactome.
</p>
<p>Each catalogue contains at least three members: pathway list (pathList), 
interaction list (interactionList) and mappings from node id to gene id (mapping). 
The pathway list contains a list of pathways in which each pathway is represented
by a list of interaction id. The interactions can be queried from the interaction list
by the interaction id. The interaction list data is represented as a data frame
in which the first column is the interaction id, the second column is the input
node id and the third column is the output node id. In real biological pathways,
a node in the pathway can be proteins, complex, families and none-gene nodes, so the 
mapping from node ids to gene ids is also provided. It must be noted that in this
package, gene symbol is selected as the primary gene id, so if users apply the PID.db
data, they should pay attension to the gene ids they transform.
</p>
<p>Besides, in each catalogue, there also a node.name, node.type and version data.
The node.name provides the custumed name for each node. The node.type provides the 
type for each node (e.g. it is a complex or compound). The version provides the version
of the catalogue data.
</p>
<p>Data has been updated to the lastest version by the day the package released (at 2012_07_19 09:34::20).
</p>
<p>Note only part of pathways in the XML file are listed on the PID website. Also, 
we have set the minimum and maximum connected nodes when extracting pathways 
from PID, so not all the pathways listed on the PID website are in PID.db.
</p>


<h3>Value</h3>

<p>A list containing four component:
</p>

<dl>
<dt>NCI</dt><dd><p>NCI_Nature-curated pathway catalogue</p>
</dd>
<dt>BioCarta</dt><dd><p>BioCarta pathway catalogue</p>
</dd>
<dt>KEGG</dt><dd><p>KEGG pathway catalogue</p>
</dd>
<dt>Reactome</dt><dd><p>Reactome pathway catalogue</p>
</dd>
</dl>

<p>Each pathway catalogue is a <code>pathway.catalogue</code> class object. Each pathway
catalogue can be used directly in <code><a href="#topic+cepa.all">cepa.all</a></code> and <code><a href="#topic+cepa">cepa</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PID.db)
names(PID.db)
PID.db$NCI
plot(PID.db$NCI)
</code></pre>

<hr>
<h2 id='plot.cepa'>
Plot the cepa object
</h2><span id='topic+plot.cepa'></span>

<h3>Description</h3>

<p>Plot the cepa object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cepa'
plot(x, type = c("graph", "null"), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.cepa_+3A_x">x</code></td>
<td>
<p>a <code><a href="#topic+cepa">cepa</a></code> object</p>
</td></tr>
<tr><td><code id="plot.cepa_+3A_type">type</code></td>
<td>
<p>identify the type for the plot</p>
</td></tr>
<tr><td><code id="plot.cepa_+3A_...">...</code></td>
<td>
<p>arguments passed to <code><a href="#topic+plotGraph">plotGraph</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is wrapper of <code><a href="#topic+plotGraph">plotGraph</a></code> and <code><a href="#topic+plotNull">plotNull</a></code>.
If type is specified to &quot;graph&quot;, the graph of the network will be plotted (see <code><a href="#topic+plotGraph">plotGraph</a></code> for details).
If type is specified to &quot;null&quot;, the null distribution of the pathway score
in the pathway will be plotted (see <code><a href="#topic+plotNull">plotNull</a></code> for details).
</p>


<h3>Value</h3>

<p>if type is set to &quot;graph&quot;, the function will return a <code><a href="igraph.html#topic+aaa-igraph-package">igraph</a></code> object or a <code>graphML</code> object of the pathway. Else it is NULL.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cepa">cepa</a></code>, <code><a href="#topic+plotNull">plotNull</a></code>, <code><a href="#topic+plotGraph">plotGraph</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

data(PID.db)

# ORA extension
data(gene.list)
# will spend about 20 min
res.ora = cepa(dif = gene.list$dif, bk = gene.list$bk, pc = PID.db$NCI, id = 2)
plot(res.ora)
plot(res.ora, type = "null")

# GSA extension
# P53_symbol.gct and P53_cls can be downloaded from
# http://mcube.nju.edu.cn/jwang/lab/soft/cepa/
eset = read.gct("P53_symbol.gct")
label = read.cls("P53.cls", treatment="MUT", control="WT")
# will spend about 45 min
res.gsa = cepa(mat = eset, label = label, pc = PID.db$NCI, id = 2)
plot(res.gsa, type = "null")

## End(Not run)
</code></pre>

<hr>
<h2 id='plot.cepa.all'>
plot the cepa.all object
</h2><span id='topic+plot.cepa.all'></span>

<h3>Description</h3>

<p>plot the cepa.all object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cepa.all'
plot(x, id = NULL, cen = 1, type = c("graph", "null"), tool = c("igraph", "Rgraphviz"),
    node.name = NULL, node.type = NULL,
    adj.method = "none", only.sig = FALSE,
    cutoff = ifelse(adj.method == "none", 0.01, 0.05), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.cepa.all_+3A_x">x</code></td>
<td>
<p>a <code><a href="#topic+cepa.all">cepa.all</a></code> object</p>
</td></tr>
<tr><td><code id="plot.cepa.all_+3A_id">id</code></td>
<td>
<p>index or the name for the pathway</p>
</td></tr>
<tr><td><code id="plot.cepa.all_+3A_cen">cen</code></td>
<td>
<p>index or the name for the centrality</p>
</td></tr>
<tr><td><code id="plot.cepa.all_+3A_type">type</code></td>
<td>
<p>If the aim is to plot single pathway, then this argument is to identify the kind of the plotting.</p>
</td></tr>
<tr><td><code id="plot.cepa.all_+3A_tool">tool</code></td>
<td>
<p>Use which tool to visualize the graph. Choices are 'igraph' and 'Rgraphviz'</p>
</td></tr>
<tr><td><code id="plot.cepa.all_+3A_node.name">node.name</code></td>
<td>
<p>node.name for each node</p>
</td></tr>
<tr><td><code id="plot.cepa.all_+3A_node.type">node.type</code></td>
<td>
<p>node.type for each node</p>
</td></tr>
<tr><td><code id="plot.cepa.all_+3A_adj.method">adj.method</code></td>
<td>
<p>method of <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>, available methods are &quot;holm&quot;, &quot;hochberg&quot;,  &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;</p>
</td></tr>
<tr><td><code id="plot.cepa.all_+3A_only.sig">only.sig</code></td>
<td>
<p>whether to show all pathways. If just show significant pathways,  the names for each significant pathway will be draw.</p>
</td></tr>
<tr><td><code id="plot.cepa.all_+3A_cutoff">cutoff</code></td>
<td>
<p>cutoff for significance</p>
</td></tr>
<tr><td><code id="plot.cepa.all_+3A_...">...</code></td>
<td>
<p>other arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function has two applications. First, it can draw heatmaps of p-values
of all pathways under different centrality measurements. To do it, users should set
<code>x</code>, <code>adj.method</code>, <code>only.sig</code>, <code>cutoff</code> arguments.
</p>
<p>Second, it can draw figures of single
pathway under specific centrality measurement. Under this circumstance, 
this function is just a wrapper of <code><a href="#topic+plot.cepa">plot.cepa</a></code>. To do it, 
users should set <code>x</code>, <code>id</code>, <code>cen</code>, <code>type</code>, <code>tool</code>, <code>node.name</code> and <code>node.type</code> arguments. The
<code>id</code> and <code>cen</code> arguments are used to get single <code><a href="#topic+cepa">cepa</a></code> object that sent to the 
plot function.
</p>
<p>It must be noted that these two kinds of arguments should not be mixed.
</p>
<p>There is also another popular method <code>qvalue</code> to adjust p-values. However, errors
may occur when adjusting some kind of p-value list by <code>qvalue</code>.
So <code>qvalue</code> was not implemented into CePa. But still users can override the default
p.adjust to support qvalue by themselves, see the vignette.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cepa.all">cepa.all</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(PID.db)

# ORA extension
data(gene.list)
# will spend about 20 min
res.ora = cepa.all(dif = gene.list$dif, bk = gene.list$bk, pc = PID.db$NCI)
plot(res.ora)
plot(res.ora, id = 3)
plot(res.ora, id = 3, type = "null")

# GSA extension
# P53_symbol.gct and P53_cls can be downloaded from
# http://mcube.nju.edu.cn/jwang/lab/soft/cepa/
eset = read.gct("P53_symbol.gct")
label = read.cls("P53.cls", treatment="MUT", control="WT")
# will spend about 45 min
res.gsa = cepa.all(mat = eset, label = label, pc = PID.db$NCI)
plot(res.gsa)
plot(res.gsa, id = 3, cen = 2)
plot(res.gsa, id = 3, cen = 2, type = "null")

## End(Not run)
</code></pre>

<hr>
<h2 id='plot.pathway.catalogue'>
plot pathway.catalogue object
</h2><span id='topic+plot.pathway.catalogue'></span>

<h3>Description</h3>

<p>plot pathway.catalogue object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pathway.catalogue'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.pathway.catalogue_+3A_x">x</code></td>
<td>
<p>a <code>pathway.catalogue</code> object</p>
</td></tr>
<tr><td><code id="plot.pathway.catalogue_+3A_...">...</code></td>
<td>
<p>other arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are three fugures: A) Distribution of the number of member genes in each node; 
B) Distribution of the number of nodes in which a single gene resides; C) Relationship 
between node count and gene count in biological pathways.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set.pathway.catalogue">set.pathway.catalogue</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PID.db)
NCI = PID.db$NCI
plot(NCI)
</code></pre>

<hr>
<h2 id='plotGraph'>
Plot graph for the pathway network
</h2><span id='topic+plotGraph'></span>

<h3>Description</h3>

<p>Plot graph for the pathway network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotGraph(x, node.name = NULL, node.type = NULL, draw = TRUE,
    tool = c("igraph", "Rgraphviz"), graph.node.max.size = 20,
    graph.node.min.size = 3, graph.layout.method = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotGraph_+3A_x">x</code></td>
<td>
<p>a <code><a href="#topic+cepa">cepa</a></code> object</p>
</td></tr>
<tr><td><code id="plotGraph_+3A_node.name">node.name</code></td>
<td>
<p>node.name for each node</p>
</td></tr>
<tr><td><code id="plotGraph_+3A_node.type">node.type</code></td>
<td>
<p>node.type for each node</p>
</td></tr>
<tr><td><code id="plotGraph_+3A_draw">draw</code></td>
<td>
<p>Whether to draw the graph</p>
</td></tr>
<tr><td><code id="plotGraph_+3A_tool">tool</code></td>
<td>
<p>Use which tool to visualize the graph. Choices are 'igraph' and 'Rgraphviz'</p>
</td></tr>
<tr><td><code id="plotGraph_+3A_graph.node.max.size">graph.node.max.size</code></td>
<td>
<p>max size of the node in the graph</p>
</td></tr>
<tr><td><code id="plotGraph_+3A_graph.node.min.size">graph.node.min.size</code></td>
<td>
<p>min size of the node in the graph</p>
</td></tr>
<tr><td><code id="plotGraph_+3A_graph.layout.method">graph.layout.method</code></td>
<td>
<p>function of the layout method. For the list of available methods, see <code><a href="igraph.html#topic+layout_">layout</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Graph view of the pathway where the size of node is proportional to centrality 
value of the node.
</p>
<p>By default, the layout for the pathway tree-like. If the number of pathway nodes
is large, the layout would be a random layout.
</p>
<p>Two packages can be selected to visualize the graph: <code>igraph</code> and <code>Rgraphviz</code>.
Default package is <code>igraph</code> (in fact, this package just uses the data generated from
the layout function in <code><a href="igraph.html#topic+aaa-igraph-package">igraph</a></code> package, which are the coordinate of nodes and edges.
And the I re-wrote the plotting function to generate the graph). From my personal view,
<code>Rgraphviz</code> package generated more beautiful graphs.
</p>
<p>If the <code>tool</code> is set as <code>igraph</code>, the function returns a <code><a href="igraph.html#topic+aaa-igraph-package">igraph</a></code> object. And
if the <code>tool</code> is set as <code>Rgraphviz</code>, the function returns a <code>graphAM</code> class object.
So if users don't satisfy, they can draw graphs of the network with their
own settings.
</p>
<p>The function is always called through <code><a href="#topic+plot.cepa.all">plot.cepa.all</a></code> and <code><a href="#topic+plot.cepa">plot.cepa</a></code>.
</p>


<h3>Value</h3>

<p>A <code><a href="igraph.html#topic+aaa-igraph-package">igraph</a></code> object of the pathway
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(PID.db)
# ORA extension
data(gene.list)
# will spend about 20 min
res.ora = cepa.all(dif = gene.list$dif, bk = gene.list$bk, pc = PID.db$NCI)
ora = get.cepa(res.ora, id = 5, cen = 3)
plotGraph(ora)
# GSA extension
# P53_symbol.gct and P53_cls can be downloaded from
# http://mcube.nju.edu.cn/jwang/lab/soft/cepa/
eset = read.gct("P53_symbol.gct")
label = read.cls("P53.cls", treatment="MUT", control="WT")
# will spend about 45 min
res.gsa = cepa.all(mat = eset, label = label, pc = PID.db$NCI)
gsa = get.cepa(res.gsa, id = 5, cen = 3)
plotGraph(gsa)

## End(Not run)
</code></pre>

<hr>
<h2 id='plotNull'>
Plot the null distribution of the pathway score
</h2><span id='topic+plotNull'></span>

<h3>Description</h3>

<p>Plot the null distribution of the pathway score
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotNull(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotNull_+3A_x">x</code></td>
<td>
<p>a <code><a href="#topic+cepa">cepa</a></code> object</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are two figures in the plotting.
</p>

<ul>
<li><p> A) Distribution of node score in the pathway under simulation. Since a pathway contains a list of nodes. The distribution of node score for the pathway in each simulation is measures by maximum value,  the 75th quantile, median value and minimum value. The distribution of node score for  the pathway in the real data is highlighted.
</p>
</li></ul>


<ul>
<li><p> B) Histogram of simulated pathway scores. 
</p>
</li></ul>

<p>The function is always called through <code><a href="#topic+plot.cepa.all">plot.cepa.all</a></code> and <code><a href="#topic+plot.cepa">plot.cepa</a></code>.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cepa">cepa</a></code>, <code><a href="#topic+plot.cepa">plot.cepa</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(PID.db)

# ORA extension
data(gene.list)
# will spend about 20 min
res.ora = cepa.all(dif = gene.list$dif, bk = gene.list$bk, pc = PID.db$NCI)
ora = get.cepa(res.ora, id = 5, cen = 3)
plotNull(ora)

# GSA extension
# P53_symbol.gct and P53_cls can be downloaded from
# http://mcube.nju.edu.cn/jwang/lab/soft/cepa/
eset = read.gct("P53_symbol.gct")
label = read.cls("P53.cls", treatment="MUT", control="WT")
# will spend about 45 min
res.gsa = cepa.all(mat = eset, label = label, pc = PID.db$NCI)
gsa = get.cepa(res.gsa, id = 5, cen = 3)
plotNull(gsa)

## End(Not run)
</code></pre>

<hr>
<h2 id='print.cepa'>
print the cepa object
</h2><span id='topic+print.cepa'></span>

<h3>Description</h3>

<p>print the cepa object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cepa'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.cepa_+3A_x">x</code></td>
<td>
<p>a <code><a href="#topic+cepa">cepa</a></code> object</p>
</td></tr>
<tr><td><code id="print.cepa_+3A_...">...</code></td>
<td>
<p>other arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function print procedure of the analysis, the centrality and the p-value for the pathway.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cepa">cepa</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

data(PID.db)

# ORA extension
data(gene.list)
# will spend about 20 min
res.ora = cepa(dif = gene.list$dif, bk = gene.list$bk, pc = PID.db$NCI, id = 2)
res.ora

# GSA extension
# P53_symbol.gct and P53_cls can be downloaded from
# http://mcube.nju.edu.cn/jwang/lab/soft/cepa/
eset = read.gct("P53_symbol.gct")
label = read.cls("P53.cls", treatment="MUT", control="WT")
# will spend about 45 min
res.gsa = cepa(mat = eset, label = label, pc = PID.db$NCI, id = 2)
res.gsa

## End(Not run)
</code></pre>

<hr>
<h2 id='print.cepa.all'>
print the cepa.all object
</h2><span id='topic+print.cepa.all'></span>

<h3>Description</h3>

<p>print the cepa.all object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cepa.all'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.cepa.all_+3A_x">x</code></td>
<td>
<p>a <code><a href="#topic+cepa.all">cepa.all</a></code> object</p>
</td></tr>
<tr><td><code id="print.cepa.all_+3A_...">...</code></td>
<td>
<p>other arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function print the number of significant pathways under various
centrality measures at p-value &lt;= 0.01.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cepa.all">cepa.all</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(PID.db)

# ORA extension
data(gene.list)
# will spend about 20 min
res.ora = cepa.all(dif = gene.list$dif, bk = gene.list$bk, pc = PID.db$NCI)
res.ora

# GSA extension
# P53_symbol.gct and P53_cls can be downloaded from
# http://mcube.nju.edu.cn/jwang/lab/soft/cepa/
eset = read.gct("P53_symbol.gct")
label = read.cls("P53.cls", treatment="MUT", control="WT")
# will spend about 45 min
res.gsa = cepa.all(mat = eset, label = label, pc = PID.db$NCI)
res.gsa

## End(Not run)
</code></pre>

<hr>
<h2 id='print.pathway.catalogue'>
print pathway.catalogue object
</h2><span id='topic+print.pathway.catalogue'></span>

<h3>Description</h3>

<p>print pathway.catalogue object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pathway.catalogue'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.pathway.catalogue_+3A_x">x</code></td>
<td>
<p>a <code>pathway.catalogue</code> object</p>
</td></tr>
<tr><td><code id="print.pathway.catalogue_+3A_...">...</code></td>
<td>
<p>other arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simply print the number of pathways in the catalogue
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+set.pathway.catalogue">set.pathway.catalogue</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PID.db)
NCI = PID.db$NCI
NCI
</code></pre>

<hr>
<h2 id='radiality'>
Calculate radiality centrality
</h2><span id='topic+radiality'></span>

<h3>Description</h3>

<p>Calculate radiality centrality
</p>


<h3>Usage</h3>

<pre><code class='language-R'>radiality(graph, mode = c("all", "in", "out"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="radiality_+3A_graph">graph</code></td>
<td>
<p>an <code><a href="igraph.html#topic+aaa-igraph-package">igraph</a></code> object</p>
</td></tr>
<tr><td><code id="radiality_+3A_mode">mode</code></td>
<td>
<p>mode of the centrality</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The radiality is defined as <code>sum(d_G + 1 - d(v, w))/(n - 1)</code>. where <code>d(w, v)</code> is the
length of the shortest path from node <code>w</code> to node <code>v</code>, <code>d_G</code> is the diameter of the network,
n is the size of the network.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(igraph)
pathway = barabasi.game(200)
radiality(pathway)
</code></pre>

<hr>
<h2 id='reach'>
Calculate largest reach centrality
</h2><span id='topic+reach'></span>

<h3>Description</h3>

<p>Calculate largest reach centrality
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reach(graph, weights=E(graph)$weight, mode=c("all", "in", "out"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reach_+3A_graph">graph</code></td>
<td>
<p>an <code><a href="igraph.html#topic+aaa-igraph-package">igraph</a></code> object</p>
</td></tr>
<tr><td><code id="reach_+3A_mode">mode</code></td>
<td>
<p>mode of the centrality</p>
</td></tr>
<tr><td><code id="reach_+3A_weights">weights</code></td>
<td>
<p>If the edges in the graph have weight, then by default, the weight is used to calculate the length of the shortest path. Set it to NULL to supress the weight.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The largest reach centrality measures how far a node can send or receive the information in the network.
It is defined as the largest length of the shortest path from all the other nodes in the network.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='read.cls'>
Read CLS file which stores the phenotype data
</h2><span id='topic+read.cls'></span>

<h3>Description</h3>

<p>Read CLS file which stores the phenotype data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.cls(file, treatment, control)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read.cls_+3A_file">file</code></td>
<td>
<p>cls file path</p>
</td></tr>
<tr><td><code id="read.cls_+3A_treatment">treatment</code></td>
<td>
<p>string of treatment label in cls file</p>
</td></tr>
<tr><td><code id="read.cls_+3A_control">control</code></td>
<td>
<p>string of control label in cls file</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The CLS file format defines the phenotype data of microarray experiments.
The first line is the number of samples, number of classes and the third number always be 1.
These three numbers are seperated by spaces or tabs.
The second line begins with #. The next two strings usually are the label of the phenotype. 
The third line is the label of each samples where same label represents 
the same class.
</p>
<p>The first and the second line is ignored by this function 
and class labels are taken from the factor of the vector parsed from the third line.
</p>


<h3>Value</h3>

<p>A <code>sampleLabel</code> class object
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.gct">read.gct</a></code>, <code><a href="#topic+sampleLabel">sampleLabel</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# P53.cls can be downloaded from
# http://mcube.nju.edu.cn/jwang/lab/soft/cepa/
label = read.cls("http://mcube.nju.edu.cn/jwang/lab/soft/cepa/P53.cls", 
    treatment="MUT", control="WT")

## End(Not run)
</code></pre>

<hr>
<h2 id='read.gct'>
Read GCT format file which stores the expression values
</h2><span id='topic+read.gct'></span>

<h3>Description</h3>

<p>Read GCT format file which stores the expression values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.gct(file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read.gct_+3A_file">file</code></td>
<td>
<p>gct file path</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The GCT format is a tab delimited file format that stores the expression value matrix.
The first line of the file is the version number which always be #1.2.
The second line is the number of the size of genes and samples, 
seperated by space, usually for the initiation of reading the expression matrix.
The third line contains a list of identifiers for the samples associated 
with each of the columns in the remainder of the file.
From the fourth line will be the expression value of each gene.
</p>
<p>GCT file is used together with CLS file.
</p>


<h3>Value</h3>

<p>A matrix of the expression values, with rows correponding to genes and cols to
samples.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# expression data stored in a gct format file
# P53_symbol.gct can be downloaded from
# http://mcube.nju.edu.cn/jwang/lab/soft/cepa/
eset = read.gct("http://mcube.nju.edu.cn/jwang/lab/soft/cepa/P53_symbol.gct")
head(eset)

## End(Not run)
</code></pre>

<hr>
<h2 id='report'>
Generate report for CePa analysis
</h2><span id='topic+report'></span>

<h3>Description</h3>

<p>Generate report for CePa analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>report(x, adj.method = "none", cutoff = ifelse(adj.method == "none", 0.01, 0.05),
    template.file = system.file(package = "CePa", "extdata", "cepa.template"),
    only.sig = TRUE, dir.path = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="report_+3A_x">x</code></td>
<td>
<p>a <code><a href="#topic+cepa.all">cepa.all</a></code> object</p>
</td></tr>
<tr><td><code id="report_+3A_adj.method">adj.method</code></td>
<td>
<p>methods in <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>, available methods are &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;</p>
</td></tr>
<tr><td><code id="report_+3A_cutoff">cutoff</code></td>
<td>
<p>cutoff for significance</p>
</td></tr>
<tr><td><code id="report_+3A_template.file">template.file</code></td>
<td>
<p>path of the template file</p>
</td></tr>
<tr><td><code id="report_+3A_only.sig">only.sig</code></td>
<td>
<p>whether to generate detailed report for every pathway. If it is set to FALSE, the page for every pathway under every centrality would be generated (there would be so many images!).</p>
</td></tr>
<tr><td><code id="report_+3A_dir.path">dir.path</code></td>
<td>
<p>dir name</p>
</td></tr>
<tr><td><code id="report_+3A_...">...</code></td>
<td>
<p>other arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The report is in HTML format that you can view it in you web browser. Networks
for pathways can be visualized interactively (by using Cytoscape Web, in which 
you can drag the network, zoom in and zoom out the network). To load Flash Player
successful in you browser, you may need to set the Flash security settings on your
machine.
</p>
<p>The report would locate at the current working directory. View the report
by clicking <code>index.html</code> in the report directory.
</p>
<p>There is also another popular method qvalue to adjust p-values. Turn to <code><a href="#topic+plot.cepa.all">plot.cepa.all</a></code>
to find out how to use qvalue.
</p>


<h3>Source</h3>

<p><a href="https://cytoscapeweb.cytoscape.org/">https://cytoscapeweb.cytoscape.org/</a></p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cepa.all">cepa.all</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(PID.db)

# ORA extension
data(gene.list)
# will spend about 20 min
res.ora = cepa.all(dif = gene.list$dif, bk = gene.list$bk, pc = PID.db$NCI)
report(res.ora)

## End(Not run)
</code></pre>

<hr>
<h2 id='sampleLabel'>
Generate data structure of sample labels
</h2><span id='topic+sampleLabel'></span>

<h3>Description</h3>

<p>Generate data structure of sample labels
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampleLabel(label, treatment, control)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sampleLabel_+3A_label">label</code></td>
<td>
<p>sample label vector</p>
</td></tr>
<tr><td><code id="sampleLabel_+3A_treatment">treatment</code></td>
<td>
<p>treatment label</p>
</td></tr>
<tr><td><code id="sampleLabel_+3A_control">control</code></td>
<td>
<p>control label</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Since sample label will not be modified in the analysis, this function is used to
integrate all the label information in one single data object.
</p>


<h3>Value</h3>

<p>A <code>sampleLabel</code> class object
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sampleLabel(c("A", "B", "B", "A", "A", "A", "B", "B"), treatment = "A", control = "B")
</code></pre>

<hr>
<h2 id='set.pathway.catalogue'>
store pathway data and pre-processing
</h2><span id='topic+set.pathway.catalogue'></span>

<h3>Description</h3>

<p>store pathway data and pre-processing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set.pathway.catalogue(pathList, interactionList, mapping,
    min.node = 5, max.node = 500, min.gene = min.node, max.gene = max.node, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set.pathway.catalogue_+3A_pathlist">pathList</code></td>
<td>
<p>list of pathways</p>
</td></tr>
<tr><td><code id="set.pathway.catalogue_+3A_interactionlist">interactionList</code></td>
<td>
<p>list of interactions</p>
</td></tr>
<tr><td><code id="set.pathway.catalogue_+3A_mapping">mapping</code></td>
<td>
<p>a data frame or matrix providing mappings from gene id to pathway node id.  The first column is node id and the second column is gene id.</p>
</td></tr>
<tr><td><code id="set.pathway.catalogue_+3A_min.node">min.node</code></td>
<td>
<p>minimum number of connected nodes in each pathway</p>
</td></tr>
<tr><td><code id="set.pathway.catalogue_+3A_max.node">max.node</code></td>
<td>
<p>maximum number of connected nodes in each pathway</p>
</td></tr>
<tr><td><code id="set.pathway.catalogue_+3A_min.gene">min.gene</code></td>
<td>
<p>minimum number of genes in each pathway</p>
</td></tr>
<tr><td><code id="set.pathway.catalogue_+3A_max.gene">max.gene</code></td>
<td>
<p>maximum number of genes in each pathway</p>
</td></tr>
<tr><td><code id="set.pathway.catalogue_+3A_...">...</code></td>
<td>
<p>other arguments, should have names, these data will be stored as a list member in the returned value from the function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The pathway data will not be changed in the analysis, so the pathway data is integrated
in one single data object by this function. Also, the function will do a little 
preprocess of the pathway data.
</p>
<p>Basicly, a pathway contains a list of interactions. The pathList argument
is a list where elements in the list is the vector of interaction IDs 
in the pathway. The interactions in the pathway can be got from a interaction
list pool represented as interactionList argument. The interactionList argument
stores the total interaction list in the pathway catalogue. It represents
as a three columns data frame or matrix where the first column is the interaction id,
the second column is the input node id and the third column is the output node id.
</p>
<p>The mapping data frame provide the mapping from node id to gene id. The 
first column is the node id and the second column is the gene id.
</p>
<p>Besides the pathList, interactionList and mapping arguments, more arguments can
be added to the function. These new data will be stored as the member of the list
that returned by the function. E.g., in the <code><a href="#topic+PID.db">PID.db</a></code> data, each catalogue
is a <code>pathway.catalogue</code> object. Except the pathList, interactionList and mapping arguments,
there are also node.name, node.type and version arguments.
</p>
<p>The summary can be visualized by <code><a href="#topic+plot.pathway.catalogue">plot.pathway.catalogue</a></code>.
</p>


<h3>Value</h3>

<p>A <code>pathway.catalogue</code> class object
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cepa.all">cepa.all</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(PID.db)
catalogue = set.pathway.catalogue(pathList = PID.db$NCI$pathList[1:20],
            interactionList = PID.db$NCI$intertionList, mapping = PID.db$NCI$mapping)

## End(Not run)
</code></pre>

<hr>
<h2 id='spread'>
Calculate radiality centrality
</h2><span id='topic+spread'></span>

<h3>Description</h3>

<p>Calculate radiality centrality
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spread(graph, mode = c("all", "in", "out"),
    weights = E(graph)$weight, f = function(x) 1/x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="spread_+3A_graph">graph</code></td>
<td>
<p>an <code><a href="igraph.html#topic+aaa-igraph-package">igraph</a></code> object</p>
</td></tr>
<tr><td><code id="spread_+3A_mode">mode</code></td>
<td>
<p>mode of the centrality</p>
</td></tr>
<tr><td><code id="spread_+3A_weights">weights</code></td>
<td>
<p>If edges in the graph have weight, then by default, the weight is used to calculate the length of the shortest path. Set it to NULL to supress the weight</p>
</td></tr>
<tr><td><code id="spread_+3A_f">f</code></td>
<td>
<p>function for the weaken rate</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The spread centrality measures how wide the node can send or receive the information in the network.
Like the water wave, the effect would be weakened with the increase of the distance to other nodes.
</p>
<p>If the weaken function is defined as <code>1/x</code>, then the spread centrality is calculated as
<code>sum(1/d(w, v))</code> where <code>d(w, v)</code> is the length of the shortest path of node <code>w</code> and node <code>v</code>.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+reach">reach</a></code>, <code><a href="#topic+radiality">radiality</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(igraph)
pathway = barabasi.game(200)
spread(pathway)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
