<!DOCTYPE html><html><head><title>Help for package CeRNASeek</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CeRNASeek}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CeRNASeek-package'>
<p>CeRNASeek</p></a></li>
<li><a href='#ceRNA.app'>
<p>identify the miRNAs that co-regulate the gene pairs of interest.</p></a></li>
<li><a href='#ceRNA.basic'>
<p>Identifying miRNA sponge interactions using ceRNA.basic function</p></a></li>
<li><a href='#ceRNA.cernia'>
<p>identifying miRNA sponge interactions using ceRNA.cernia function.</p></a></li>
<li><a href='#ceRNA.cmi'>
<p>identifying miRNA sponge interactions using ceRNA.cmi function.</p></a></li>
<li><a href='#ceRNA.cor'>
<p>identifying miRNA sponge interactions using ceRNA.cor function</p></a></li>
<li><a href='#ceRNA.enrich'>
<p>Enrich the biological functions of interest.</p></a></li>
<li><a href='#ceRNA.Net'>
<p>visualize and analyze the identified ceRNA-ceRNA network using ceRNA.Net function</p></a></li>
<li><a href='#ceRNA.surv'>
<p>survival analysis of ceRNA ternary relationship pairs</p></a></li>
<li><a href='#dataset'>
<p>Data for Examples</p></a></li>
<li><a href='#surv.plot'>
<p>Draw survival curve</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Identification and Analysis of ceRNA Regulation</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-04-20</td>
</tr>
<tr>
<td>Author:</td>
<td>Mengying Zhang,Yongsheng Li,Juan Xu*,Xia Li*</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mengying Zhang &lt;zhangmengying@hrbmu.edu.cn&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides several functions to identify and analyse miRNA sponge, including
             popular methods for identifying miRNA sponge interactions, two types 
             of global ceRNA regulation prediction methods and four types of context-specific 
             prediction methods( Li Y et al.(2017) &lt;<a href="https://doi.org/10.1093%2Fbib%2Fbbx137">doi:10.1093/bib/bbx137</a>&gt;), which are based 
             on miRNA-messenger RNA regulation alone, or by integrating heterogeneous data, 
             respectively. In addition, For predictive ceRNA relationship pairs, this package 
             provides several downstream analysis algorithms, including regulatory network 
             analysis and functional annotation analysis, as well as survival prognosis analysis
             based on expression of ceRNA ternary pair. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>biocViews:</td>
<td>competing endogenous RNA (ceRNA),
GeneExpression,triplet,function,Software</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>gtools ,igraph,survival,parallel</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-04-29 07:16:20 UTC; dell</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-04-29 08:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='CeRNASeek-package'>
CeRNASeek
</h2><span id='topic+CeRNASeek-package'></span><span id='topic+CeRNASeek'></span>

<h3>Description</h3>

<p>identifying miRNA sponge interactions .
</p>


<h3>Details</h3>

<p>CeRNASeek considers five method to identify ceRNA crosstalk, including two types of global 
ceRNA regulation prediction methods(ratio based, we termed ratio and hypergeometric test
based, termed HyperT) and three types of context-specific prediction methods (hypergeometric 
test plus coexpression, termed HyperC, sensitivity correlation-based method (SC)and conditional 
mutual information (CMI)-based methods and cernia method).
Ratio-based prediction:
this method ranked the candidate genes by the proportion of common miRNAs,For instance, we need to identify the 
ceRNA partners for gene i from all candidate gene sets S, and the ratio is calculated as the intersection of miRNAi 
and miRNAj divided by miRNAj Where miRNAi is the miRNA set that regulated gene i and miRNAj is the miRNA set that regulated gene j.
</p>
<p>Hypergeometric test-based prediction-HyperT:
it is usually used the hypergeometric to evaluate whether two genes were coregulated by miRNAs,This statistic 
test computed the significance of common miRNAs for each RNA pairs.
</p>
<p>Hypergeometric test combined with coexpression-based prediction-HyperC: 
To discover the active ceRNA-ceRNA regulatory pairs in a specific context, the commonly used method is to using the
coexpression principle to filter the ceRNA-ceRNA regulation identified based on the above two global methods,this
method integrated context-specific gene expression profile data sets. The Pearson correlation coefficient (R) of 
each candidate ceRNA regulatory pairs identified was calculated.
</p>
<p>SC-based prediction:
Another method introduced the expression profile data of shared miRNAs, and uses partial correlation to calculate ceRNA 
interaction pairs. Then, the Sensitivity correlation(SC) of miRNA-M, for the corresponding candidate ceRNA pair is 
calculated. 
</p>
<p>CMI-based methods:
CMI is widely used to identify the RNA-RNA correlations, given the value of miRNAs. Hermes is a widely used method, 
which predicts ceRNA interactions from expression profiles of candidate RNAs and their common miRNA regulators using CMI,
as in Sumazin et al.Firstly computed the significance of difference of I(miRNAi;T2|T1) and I(miRNAi;T2),where miRNAi represents the ith 
miRNA in the miRNA shared by the two target genes. Then Shuffled condition target's expression in 1000 times. The final 
significance based on Fisher’s method was calculated.For more information, please refer to the article by Sumazin P et al.
</p>
<p>cernia method
cernia method is  implemented based on the following seven scores: 
1. The fraction of common miRNAs;
2. The density of the MREs for all shared miRNAs;
3. The distribution of MREs of the putative ceRNAs;
4. The relation between the overall number of MREs for a putative ceRNA, compared with the number of miRNAs that yield
these MREs;
5. The density of the hybridization energies related to MREs for
all shared miRNAs;
6. The DT-Hybrid recommendation scores; and
7. The pairwise Pearson correlation between putative ceRNA
expressions from selected tissue.
</p>


<h3>References</h3>

<p>Xu J , Feng L , Han Z , et al. Extensive ceRNA–ceRNA interaction networks mediated by miRNAs regulate
development in multiple rhesus tissues[J]. Nucleic Acids Research, 2016:gkw587.
</p>
<p>Sumazin P , Yang X , Chiu H S , et al. An Extensive MicroRNA-Mediated Network of RNA-RNA Interactions 
Regulates Established Oncogenic Pathways in Glioblastoma[J]. Cell, 2011, 147(2):0-381.
</p>
<p>Paci P , Colombo T , Farina L . Computational analysis identifies a sponge interaction network between
long non-coding RNAs and messenger RNAs in human breast cancer[J]. BMC Systems Biology, 2014, 8(1):83.
</p>
<p>Zhang Y , Xu Y , Feng L , et al. Comprehensive characterization of lncRNA-mRNA related ceRNA network
across 12 major cancers[J]. Oncotarget, 2014, 7(39):64148-64167.
</p>
<p>Sardina D S , Alaimo S , Ferro A , et al. A novel computational method for inferring competing endogenous
interactions[J]. Briefings in Bioinformatics, 2016:bbw084.
</p>

<hr>
<h2 id='ceRNA.app'>
identify the miRNAs that co-regulate the gene pairs of interest.
</h2><span id='topic+ceRNA.app'></span>

<h3>Description</h3>

<p>The ceRNA function is used to identify the miRNAs that co-regulate the gene pairs of interest.
The users can input the genes of interest. Otherwise, all the paired genes were calculated the
number of co-regulating miRNAs. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ceRNA.app(miRtar, targetce = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ceRNA.app_+3A_mirtar">miRtar</code></td>
<td>

<p>A data frame representing the relationship between miRNA and target. The data frame contains the name of 
the miRNA and target regulatory relationship.
</p>
</td></tr>
<tr><td><code id="ceRNA.app_+3A_targetce">targetce</code></td>
<td>

<p>a character string (vector) specifying candidate target name to analyse (default (targetce = NULL)).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note:All the arguments without default value must be assigned.
</p>


<h3>Value</h3>

<p>A list of identified  miRNAs that co-regulate the gene pairs containing following components:
</p>

<ul>
<li> <p><code>ceRNA</code>  identify the miRNAs that co-regulate the gene pairs of interest,List all possible 
ceRNA interactions,a 4 columns dataframe as following:
</p>

<ul>
<li> <p><code>targetce</code> represented target names,respectively.
</p>
</li>
<li> <p><code>anotherce</code> names of modulators that another possible target(modulators) constitutes a ceRNA interaction relation.
</p>
</li>
<li> <p><code>miRNAs</code> names of miRNA shared by two targets.
</p>
</li>
<li> <p><code>miRNAs_num</code> number of miRNAs shared by two targets.
</p>
</li></ul>

</li>
<li> <p><code>miR_l</code> Number of miRNAs interacting with each target in the input file.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>##Here we take the regulatory relationship between six genes and 71 miRNAs.

ceRNA.app(dataset[["miRtar"]],targetce=NULL)

</code></pre>

<hr>
<h2 id='ceRNA.basic'>
Identifying miRNA sponge interactions using ceRNA.basic function 
</h2><span id='topic+ceRNA.basic'></span>

<h3>Description</h3>

<p>This function predicts ceRNA interactions by ratio or hypergeometric test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ceRNA.basic(miRtar, targetce = NULL, method = "ratio", numMIR = 1,
 cutoff = ifelse(method == "ratio", 1/3, 0.05))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ceRNA.basic_+3A_mirtar">miRtar</code></td>
<td>

<p>A data frame representing the relationship between miRNA and target. The data frame contains the name of 
the miRNA and target regulatory relationship.
</p>
</td></tr>
<tr><td><code id="ceRNA.basic_+3A_targetce">targetce</code></td>
<td>

<p>a character string (vector) specifying candidate target name to analyse (default (targetce = NULL)).
</p>
</td></tr>
<tr><td><code id="ceRNA.basic_+3A_method">method</code></td>
<td>

<p>a character string (default &quot;ratio&quot;) indicating which statistical method to choose to calculate the ceRNA 
interaction relationship. One of &quot;ratio&quot; (default), or &quot;hypergeometric&quot;, can be abbreviated.
</p>
</td></tr>
<tr><td><code id="ceRNA.basic_+3A_nummir">numMIR</code></td>
<td>

<p>a numeric vector that specify the minimum number of 2 gene-shared miRNAs
</p>
</td></tr>
<tr><td><code id="ceRNA.basic_+3A_cutoff">cutoff</code></td>
<td>

<p>a numeric vector of the  method(&quot;ratio&quot;,&quot;hypergeometric&quot;) specifying threshold between ceRNA interactions .(default (1/3)).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note:All the arguments without default value must be assigned.
</p>


<h3>Value</h3>

<p>A list of identified miRNA sponge interactions containing following components:
</p>

<ul>
<li> <p><code>cesig</code> predicted  significant triplets and related information,a 5 columns dataframe as following:
</p>

<ul>
<li> <p><code>targetce</code> represented target names,respectively.
</p>
</li>
<li> <p><code>anotherce</code> names of modulators that another target(modulators) constitutes a ceRNA interaction relation.
</p>
</li>
<li> <p><code>miRNAs</code> names of miRNA shared by two targets.
</p>
</li>
<li> <p><code>miRNAs_num</code> number of miRNAs shared by two targets.
</p>
</li>
<li> <p><code>ratio/pvalue</code> The ratio/pvalue(optional for method(&quot;ratio&quot;,&quot;hypergeometric&quot;)) of the identified ceRNA interaction relation.
</p>
</li></ul>

</li>
<li> <p><code>cenotsig</code> predicted  not significant triplets and related information,a 5 columns dataframe as following:
</p>

<ul>
<li> <p><code>targetce</code> same as <code>targetce</code> in <code>cesig</code>
</p>
</li>
<li> <p><code>anotherce</code> same as <code>anotherce</code> in <code>cesig</code>
</p>
</li>
<li> <p><code>miRNAs</code> same as <code>miRNAs</code> in <code>cesig</code>
</p>
</li>
<li> <p><code>miRNAs_num</code> same as <code>miRNAs_num</code> in <code>cesig</code>
</p>
</li>
<li> <p><code>pvalue</code> same as <code>pvalue</code> in <code>cesig</code>
</p>
</li></ul>

</li></ul>



<h3>References</h3>

<p>Xu J , Feng L , Han Z , et al. Extensive ceRNA–ceRNA interaction networks mediated by miRNAs regulate
development in multiple rhesus tissues[J]. Nucleic Acids Research, 2016:gkw587.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##identifying miRNA sponge interactions
##Here we take six candidate targets(modulators) for example
ceRNA.basic(miRtar=dataset[["miRtar"]],targetce=NULL,method="ratio",numMIR = 1,cutoff =1/3)
</code></pre>

<hr>
<h2 id='ceRNA.cernia'>
identifying miRNA sponge interactions using ceRNA.cernia function.
</h2><span id='topic+ceRNA.cernia'></span>

<h3>Description</h3>

<p>identifying miRNA sponge interactions using ceRNA.cernia.In this function We
implement cernia score to identify miRNA sponge interactions.
The fraction of common miRNAs;
1. The fraction of common miRNAs;
2. The density of the MREs for all shared miRNAs;
3. The distribution of MREs of the putative ceRNAs;
4. The relation between the overall number of MREs for a putative ceRNA, 
compared with the number of miRNAs that yield these MREs;
5. The density of the hybridization energies related to MREs for
all shared miRNAs;
6. The DT-Hybrid recommendation scores; 
7. The pairwise Pearson correlation between putative ceRNA
expressions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ceRNA.cernia(miRtar, targetce = NULL, geneexp, miRexp, mres, numMIR = 3, cor_cutoff = 0,
             s_cutoff = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ceRNA.cernia_+3A_mirtar">miRtar</code></td>
<td>

<p>A data frame representing the relationship between miRNA and target. The data frame contains the name of 
the miRNA and target regulatory relationship.
</p>
</td></tr>
<tr><td><code id="ceRNA.cernia_+3A_targetce">targetce</code></td>
<td>

<p>a character string (vector) specifying candidate target name to analyse (default (targetce = NULL)).
</p>
</td></tr>
<tr><td><code id="ceRNA.cernia_+3A_geneexp">geneexp</code></td>
<td>

<p>An input target expression data frame, the columns are genes and
the rows are samples.The expression value may be  gene expression ,non-coding RNA expression  or 
expression values of circRNAs and so on.
</p>
</td></tr>
<tr><td><code id="ceRNA.cernia_+3A_mirexp">miRexp</code></td>
<td>

<p>An input miRNA expression data frame, the columns are miRNA and
the rows are samples.
</p>
</td></tr>
<tr><td><code id="ceRNA.cernia_+3A_mres">mres</code></td>
<td>

<p>miRNA response elements (mres) data frame, each row
contains five elements: mirna, target, energy, gap_l, gap_r. 
</p>
</td></tr>
<tr><td><code id="ceRNA.cernia_+3A_nummir">numMIR</code></td>
<td>

<p>a numeric vector that specify the minimum number of 2 gene-shared miRNAs.
</p>
</td></tr>
<tr><td><code id="ceRNA.cernia_+3A_cor_cutoff">cor_cutoff</code></td>
<td>

<p>a numeric vector of the form method specifying threshold between
ceRNA interactions correlation,default(cor_cutoff=0).
</p>
</td></tr>
<tr><td><code id="ceRNA.cernia_+3A_s_cutoff">s_cutoff</code></td>
<td>

<p>the threshold of seven comprehensive scores. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note:All the arguments without default value must be assigned.The miRNA 
in the file of the target-miRNA regulatory relationship pair should also be 
present in the expression profile data file.
Internal functions (parMM, graphWeights, recommendation, dtHybrid) of cernia method are from
the website: https://github.com/dsardina/cernia Copyright 2016 Rosalba Giugno Licensed under
the Apache License, Version 2.0 (the 'License')
</p>


<h3>Value</h3>

<p>A list of identified miRNA sponge interactions containing following components:
</p>

<ul>
<li> <p><code>targetce</code> represented target names,respectively.
</p>
</li>
<li> <p><code>anotherce</code> names of modulators that another target(modulators) constitutes a ceRNA interaction relation.
</p>
</li>
<li> <p><code>miRNAs_num</code> names of miRNA in the triplet.
</p>
</li>
<li> <p><code>Score 1</code> The fraction of common miRNAs;
</p>
</li>
<li> <p><code>Score 2</code> The density of the MREs for all shared miRNAs;
</p>
</li>
<li> <p><code>Score 3</code> The distribution of MREs of the putative ceRNAs;
</p>
</li>
<li> <p><code>Score 4</code> The relation between the overall number of MREs for a putative ceRNA, 
compared with the number of miRNAs that yield these MREs;
</p>
</li>
<li> <p><code>Score 5</code> The density of the hybridization energies related to MREs for
all shared miRNAs;
</p>
</li>
<li> <p><code>Score 6</code> The DT-Hybrid recommendation scores; 
</p>
</li>
<li> <p><code>Score 7</code> The pairwise Pearson correlation between putative ceRNA
expressions.
</p>
</li>
<li> <p><code>Normalized score</code> combine the sum of 7 scores and standardize by z-score.
</p>
</li></ul>



<h3>References</h3>

<p>Sardina D S , Alaimo S , Ferro A , et al. A novel computational method for inferring 
competing endogenous interactions[J]. Briefings in Bioinformatics, 2016:bbw084.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##identifying miRNA sponge interactions.
##Here we take six candidate targets(modulators) and corresponding expression 
##data for example,Specify target(PTEN) to predict ceRNA interaction.
ceRNA.cernia(miRtar=dataset[["miRtar"]], targetce = "PTEN", geneexp=dataset[["geneexp"]],
          numMIR = 1, miRexp=dataset[["miRexp"]], mres=dataset[["mres"]], 
		  cor_cutoff = 0.2, s_cutoff = 0.5) 
</code></pre>

<hr>
<h2 id='ceRNA.cmi'>
identifying miRNA sponge interactions using ceRNA.cmi function. 
</h2><span id='topic+ceRNA.cmi'></span>

<h3>Description</h3>

<p>identifying miRNA sponge interactions using ceRNA.cmi.In this function We
implement CMI methods to identify miRNA sponge interactions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ceRNA.cmi(miRtar, targetce = NULL, geneexp, miRexp, numMIR = 1, num_perm = 100, 
cutoff = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ceRNA.cmi_+3A_mirtar">miRtar</code></td>
<td>

<p>A data frame representing the relationship between miRNA and target. The data frame contains the name of 
the miRNA and target regulatory relationship.
</p>
</td></tr>
<tr><td><code id="ceRNA.cmi_+3A_targetce">targetce</code></td>
<td>

<p>a character string (vector) specifying candidate target name to analyse (default (targetce = NULL)).
</p>
</td></tr>
<tr><td><code id="ceRNA.cmi_+3A_geneexp">geneexp</code></td>
<td>

<p>An input target expression data frame, the columns are genes and
the rows are samples.The expression value may be  gene expression ,non-coding RNA expression  or 
expression values of circRNAs and so on.
</p>
</td></tr>
<tr><td><code id="ceRNA.cmi_+3A_mirexp">miRexp</code></td>
<td>

<p>An input miRNA expression data frame, the columns are miRNA and
the rows are samples.
</p>
</td></tr>
<tr><td><code id="ceRNA.cmi_+3A_nummir">numMIR</code></td>
<td>

<p>a numeric vector that specify the minimum number of 2 gene-shared miRNAs
</p>
</td></tr>
<tr><td><code id="ceRNA.cmi_+3A_num_perm">num_perm</code></td>
<td>

<p>The number of random default(num_perm=100).
</p>
</td></tr>
<tr><td><code id="ceRNA.cmi_+3A_cutoff">cutoff</code></td>
<td>

<p>a numeric vector of the form method specifying threshold between
ceRNA interactions default(cutoff=0.05).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note:All the arguments without default value must be assigned.The miRNA 
in the file of the target-miRNA regulatory relationship pair should also be 
present in the expression profile data file.
</p>


<h3>Value</h3>

<p>A list of identified miRNA sponge interactions containing following components:
</p>

<ul>
<li> <p><code>ceRNA_cmi</code> predicted triplets and related information,a 5 columns dataframe as following:
</p>

<ul>
<li> <p><code>targetce</code> represented target names,respectively.
</p>
</li>
<li> <p><code>miRNA</code> names of miRNA in the triplet.
</p>
</li>
<li> <p><code>anotherce</code> names of modulators that another target(modulators) constitutes a ceRNA interaction relation.
</p>
</li>
<li> <p><code>cmi</code> Conditional mutual information(CMI) of triplets calculated using expression values.
</p>
</li>
<li> <p><code>pvalue</code> The p value of the identified ceRNA interaction relation by CMI.
</p>
</li></ul>

</li>
<li> <p><code>ceRNA_comP</code> Number of miRNAs interacting with each target in the input file.
</p>

<ul>
<li> <p><code>targetce</code> represented target names,respectively.
</p>
</li>
<li> <p><code>anotherce</code> names of predicted another target(modulators) constitutes a ceRNA interaction relation.
</p>
</li>
<li> <p><code>commonMiR</code> names of miRNA shared by predicted ceRNA.
</p>
</li>
<li> <p><code>xsq</code> Chi-square value of p-value <code>pvalue</code> in conditional mutual information .
</p>
</li>
<li> <p><code>comP</code> The p value calculated by incomplete gamma function .
</p>
</li></ul>

</li></ul>



<h3>References</h3>

<p>Sumazin P , Yang X , Chiu H S , et al. An Extensive MicroRNA-Mediated Network of RNA-RNA Interactions 
Regulates Established Oncogenic Pathways in Glioblastoma[J]. Cell, 2011, 147(2):0-381.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##identifying miRNA sponge interactions.
##Here we take six candidate targets(modulators) and corresponding expression 
##data for example,Specify target(PTEN) to predict ceRNA interaction.
ceRNA.cmi(miRtar=dataset[["miRtar"]], targetce = "PTEN", geneexp=dataset[["geneexp"]],
          numMIR = 1, miRexp=dataset[["miRexp"]],num_perm=50)
</code></pre>

<hr>
<h2 id='ceRNA.cor'>
identifying miRNA sponge interactions using ceRNA.cor function 
</h2><span id='topic+ceRNA.cor'></span>

<h3>Description</h3>

<p>identifying miRNA sponge interactions using ceRNA.cor.In this function We
implement several popular linear methods ( HyperC, SC) to identify miRNA sponge.
interactions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ceRNA.cor(miRtar, targetce = NULL, geneexp, miRexp, numMIR = 1, method = "pearson", 
numrandom = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ceRNA.cor_+3A_mirtar">miRtar</code></td>
<td>

<p>A data frame representing the relationship between miRNA and target. The data frame contains the name of 
the miRNA and target regulatory relationship.Required option for method &quot;pearson&quot; and &quot;partial correlation&quot;.
</p>
</td></tr>
<tr><td><code id="ceRNA.cor_+3A_targetce">targetce</code></td>
<td>

<p>a character string (vector) specifying candidate target name to analyse (default (targetce = NULL)).
</p>
</td></tr>
<tr><td><code id="ceRNA.cor_+3A_geneexp">geneexp</code></td>
<td>

<p>An input target expression data frame, the columns are genes and
the rows are samples.The expression value may be  gene expression ,non-coding RNA expression  or 
expression values of circRNAs and so on. Required option for method pearson&quot; and &quot;partial correlation&quot;.
</p>
</td></tr>
<tr><td><code id="ceRNA.cor_+3A_mirexp">miRexp</code></td>
<td>

<p>An input miRNA expression data frame, the columns are miRNA and
the rows are samples. Required option for method pearson&quot; and &quot;partial correlation&quot;.
</p>
</td></tr>
<tr><td><code id="ceRNA.cor_+3A_nummir">numMIR</code></td>
<td>

<p>a numeric vector that specify the minimum number of 2 gene-shared miRNAs.
</p>
</td></tr>
<tr><td><code id="ceRNA.cor_+3A_method">method</code></td>
<td>

<p>a character string (default &quot;pearson&quot;) indicating which statistical method to choose to calculate the ceRNA 
interaction relationship. One of &quot;pearson&quot; (default), or &quot;partial correlation&quot;, can be abbreviated.
</p>
</td></tr>
<tr><td><code id="ceRNA.cor_+3A_numrandom">numrandom</code></td>
<td>

<p>The number of random. Required option for method &quot;partial correlation&quot;,default
(numrandom = 100).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note:All the arguments without default value must be assigned.
</p>


<h3>Value</h3>

<p>A list of identified miRNA sponge interactions containing following components:
</p>

<ul>
<li> <p><code>ceRNA</code> predicted triplets and related information,a 5 columns dataframe as following:
</p>

<ul>
<li> <p><code>targetce</code> represented target names,respectively.
</p>
</li>
<li> <p><code>anotherce</code> names of modulators that another target(modulators) constitutes a ceRNA interaction relation.
</p>
</li>
<li> <p><code>miRNAs</code> names of miRNA shared by two targets.
</p>
</li>
<li> <p><code>miRNAs_num</code> number of miRNAs shared by two targets.
</p>
</li>
<li> <p><code>correlation</code> The correlation of the identified ceRNA interaction relation.
</p>
</li>
<li> <p><code>pvalue</code> The p value of the identified ceRNA interaction relation.
</p>
</li></ul>

</li>
<li> <p><code>miR_l</code> Number of miRNAs interacting with each target in the input file.
</p>
</li></ul>



<h3>References</h3>

<p>Paci P , Colombo T , Farina L . Computational analysis identifies a sponge interaction network between
long non-coding RNAs and messenger RNAs in human breast cancer[J]. BMC Systems Biology, 2014, 8(1):83.
Zhang Y , Xu Y , Feng L , et al. Comprehensive characterization of lncRNA-mRNA related ceRNA network
across 12 major cancers[J]. Oncotarget, 2014, 7(39):64148-64167.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##identifying miRNA sponge interactions
##Here we take the regulatory relationship between six genes and 71 miRNAs  and corresponding
##expression profilesas an example.
ceRNA.cor(miRtar=dataset[["miRtar"]], targetce = NULL, geneexp=dataset[["geneexp"]],
          miRexp=dataset[["miRexp"]], numMIR = 1,method = "pearson", numrandom = 100)
</code></pre>

<hr>
<h2 id='ceRNA.enrich'>
Enrich the biological functions of interest. 
</h2><span id='topic+ceRNA.enrich'></span>

<h3>Description</h3>

<p>Downstream analysis function of ceRNA,This function can be used to identify 
biological functions of interest, and users can enrich the functions of 
interest by ceRNA.enrich.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ceRNA.enrich(data, GOterms, background, threshold = 2, correction = "BH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ceRNA.enrich_+3A_data">data</code></td>
<td>

<p>a dataframe that the ceRNA relationship is the data, and the prediction result data obtained 
according to the ceRNA prediction algorithm.Such as ceRNA.cmi prediction result file.
</p>
</td></tr>
<tr><td><code id="ceRNA.enrich_+3A_goterms">GOterms</code></td>
<td>

<p>a list that goterm of interest and all gene sets in the term.
</p>
</td></tr>
<tr><td><code id="ceRNA.enrich_+3A_background">background</code></td>
<td>

<p>a vector containing a gene set in which GOterm annotated genes must be.Its id style must be 
consistent with the id format in <em>GOterms</em>.
</p>
</td></tr>
<tr><td><code id="ceRNA.enrich_+3A_threshold">threshold</code></td>
<td>

<p>a numeric (default 2) representing min number of intersection between a modulator's targets and a GOterms genes.
</p>
</td></tr>
<tr><td><code id="ceRNA.enrich_+3A_correction">correction</code></td>
<td>

<p>correction method (default &quot;BH&quot;) in one of <code><a href="stats.html#topic+p.adjust.methods">p.adjust.methods</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note:All the arguments without default value must be assigned.
</p>


<h3>Value</h3>

<p>A list of identified miRNA sponge interactions containing following components:
</p>

<ul>
<li> <p><code>target</code> represented target names,respectively.
</p>
</li>
<li> <p><code>GOterm</code> the GOterm name.
</p>
</li>
<li> <p><code>target_num</code> names of represented target in the triplet.
</p>
</li>
<li> <p><code>GOtermnum</code> the gene number of a GOterm;
</p>
</li>
<li> <p><code>term_tar</code> the number of intersected factor between a GOterm genes and a modulator targets;
</p>
</li>
<li> <p><code>P_value</code> the p value of the significance enrichment;
</p>
</li>
<li> <p><code>fdr</code> corrected P_value by the assigned method;
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.

## The function is currently defined as
ceRNA.enrich(data=dataset[["Pre.ceRNA"]],GOterms=dataset[["GOterms"]],
             background=dataset[["background"]],threshold=1,correction="BH")
</code></pre>

<hr>
<h2 id='ceRNA.Net'>
visualize and analyze the identified ceRNA-ceRNA network using ceRNA.Net function 
</h2><span id='topic+ceRNA.Net'></span>

<h3>Description</h3>

<p>A downstream analysis function of ceRNAseekvisualize and analyze the identified ceRNA-ceRNA network, the network can
be defined as weighted or un-weighted network.and the basic topological features (such as degree, 
closeness, betweenness and centrality) of the ceRNAs can be output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ceRNA.Net(data, net_direct = TRUE, vertex_size = 20, v.label = TRUE,
         node_shape = "circle",n_color = "orange",E_weight = TRUE, ity = 1,
         label_cex = 2, label_color = "black",edge_color = "gray", 
         n_frame.color = "gray")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ceRNA.Net_+3A_data">data</code></td>
<td>

<p>A matrix of ceRNA interaction pairs identified by statistical identification methods selected by the user .
</p>
</td></tr>
<tr><td><code id="ceRNA.Net_+3A_net_direct">net_direct</code></td>
<td>

<p>A logical variable specifies a directed or undirected network.default (net_direct = TRUE).
</p>
</td></tr>
<tr><td><code id="ceRNA.Net_+3A_vertex_size">vertex_size</code></td>
<td>

<p>a numeric vector to adjust the node size,default (vertex_size = 20).
</p>
</td></tr>
<tr><td><code id="ceRNA.Net_+3A_v.label">v.label</code></td>
<td>

<p>Whether to display the label of the node.default (v.label = TRUE).
</p>
</td></tr>
<tr><td><code id="ceRNA.Net_+3A_node_shape">node_shape</code></td>
<td>

<p>A character vector is used to adjust the shape of the node,The selectable shape parameters are
&quot;circle&quot;,&quot;square&quot;,&quot;csquare&quot;,&quot;rectangle&quot;,&quot;crectangle&quot;,&quot;vrectangle&quot;,&quot;pie&quot;,
&quot;sphere&quot;,&quot;none&quot;.default (node_shape = &quot;circle&quot;).
</p>
</td></tr>
<tr><td><code id="ceRNA.Net_+3A_n_color">n_color</code></td>
<td>

<p>The character vector used to define the fill color of the node.
</p>
</td></tr>
<tr><td><code id="ceRNA.Net_+3A_e_weight">E_weight</code></td>
<td>

<p>A logical vector represents whether the network is a weighted network,default (E_weight = TRUE).
</p>
</td></tr>
<tr><td><code id="ceRNA.Net_+3A_ity">ity</code></td>
<td>

<p>A numeric vector defines whether the edge is a solid line or a dotted line,the possible values of the vector are c(1, 2),
default (ity = 1).
</p>
</td></tr>
<tr><td><code id="ceRNA.Net_+3A_label_cex">label_cex</code></td>
<td>

<p>Specify the size of the node label font.
</p>
</td></tr>
<tr><td><code id="ceRNA.Net_+3A_label_color">label_color</code></td>
<td>

<p>Specify the label color of the node.
</p>
</td></tr>
<tr><td><code id="ceRNA.Net_+3A_edge_color">edge_color</code></td>
<td>

<p>Specify the color of the network side.
</p>
</td></tr>
<tr><td><code id="ceRNA.Net_+3A_n_frame.color">n_frame.color</code></td>
<td>

<p>The character vector used to define the border color of the node.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calls the igraph package. For specific parameter settings, please refer to the igrap help documentation.
Note:All the arguments without default value must be assigned.
</p>


<h3>Value</h3>

<p>The output includes two parts:
the network diagram of ceRNA interaction and the topology attribute information of the network.
</p>

<p>Network topology attributes include 5 types of information:
</p>

<ul>
<li> <p><code>degree</code> degree refers to the number of edges in the network directly connected to the node
</p>
</li>
<li> <p><code>closeness</code> An indicator that describes the average distance of a node to all other nodes in the network.
</p>
</li>
<li> <p><code>betweenness</code> The proportion of this node that appears in the shortest path between other nodes.
</p>
</li>
<li> <p><code>cluster coefficient</code> Representing the dense connection nature between some nodes in the network
</p>
</li>
<li> <p><code>Eigenvector centrality</code> Representing the characteristic vector centrality of the network.
</p>
</li></ul>




<h3>Examples</h3>

<pre><code class='language-R'>##Display ceRNA interactions in a network format and output network topology attributes.
##The input file can be a list of [ceRNA] of the ceRNA.Lin result file, a list of [cesig] 
##for the result file identified by ceRNA.basic, or a list of [ceRNA_comP] in the result
##file identified by the ceRNA.cmi function.
##Here we apply the ceRNA list in the example file for CMI identification to 
##display the network and analyze the network topology properties.
ceRNA.Net(as.matrix(dataset[["Pre.ceRNA"]]),net_direct=TRUE,vertex_size=20,v.label = TRUE,
         node_shape="circle",n_color = "orange",E_weight=TRUE,ity=1,label_cex=2,
         label_color="black",edge_color="gray",n_frame.color = "gray")
</code></pre>

<hr>
<h2 id='ceRNA.surv'>
survival analysis of ceRNA ternary relationship pairs
</h2><span id='topic+ceRNA.surv'></span>

<h3>Description</h3>

<p>It is used to predict the survival of ternary relationship pairs 
and to support the survival prognosis of training sets and test sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ceRNA.surv(ceRNA, exp.sur, train = NULL, test = NULL, index)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ceRNA.surv_+3A_cerna">ceRNA</code></td>
<td>

<p>a dataframe that the ceRNA relationship is the data, and the prediction result data obtained 
according to the ceRNA prediction algorithm.Such as ceRNA.cmi prediction result file.
</p>
</td></tr>
<tr><td><code id="ceRNA.surv_+3A_exp.sur">exp.sur</code></td>
<td>

<p>dataframe specifying expression and survival information.Its rownames are sample names.Its colnames
are names in triplets and survival information (see example data in details).
</p>
</td></tr>
<tr><td><code id="ceRNA.surv_+3A_train">train</code></td>
<td>

<p>a character string vector specifying train sample names.
</p>
</td></tr>
<tr><td><code id="ceRNA.surv_+3A_test">test</code></td>
<td>

<p>a character string vector specifying test sample names.
</p>
</td></tr>
<tr><td><code id="ceRNA.surv_+3A_index">index</code></td>
<td>

<p>a numeric vector (default 1) reprsenting rowindex of triplets analyed.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of identified miRNA sponge interactions containing following components:
</p>

<ul>
<li> <p><code>targetce</code> represented target names,respectively.
</p>
</li>
<li> <p><code>anotherce</code> names of modulators that another target(modulators) constitutes a ceRNA interaction relation.
</p>
</li>
<li> <p><code>coef_targetce</code> the <em>coxph</em> coefficient of targetce;
</p>
</li>
<li> <p><code>p_targetce</code> the <em>coxph</em> significance of targetce;
</p>
</li>
<li> <p><code>coef_anotherce</code>  the <em>coxph</em> coefficient of anotherce;
</p>
</li>
<li> <p><code>p_anotherce</code> the <em>coxph</em> significance of anotherce;
</p>
</li>
<li> <p><code>N_low</code> The genes were ranked according to expression, and the number of samples 
expressed in the bottom 25
</p>
</li>
<li> <p><code>N_high</code> The genes were ranked according to expression, and the number of samples 
expressed in the top 25
</p>
</li>
<li> <p><code>HR</code> Risk score of ceRNA ternary pair.
</p>
</li>
<li> <p><code>p</code> Survival significance of ceRNA ternary pairs.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.

## The function is currently defined as
ceRNA.surv(ceRNA=dataset[["Pre.ceRNA"]],exp.sur=dataset[["exp.sur"]],train=NULL,test=NULL,index=5)
</code></pre>

<hr>
<h2 id='dataset'>
Data for Examples
</h2><span id='topic+dataset'></span>

<h3>Description</h3>

<p>This object contains data for examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(dataset)</code></pre>


<h3>Format</h3>

<p>A list with 4 variables:
</p>

<dl>
<dt>miRtar</dt><dd>
<p>MiRNA–target regulation,assembled genome-wide miRNA-gene regulation by TargetScan,It includes 44 rows standing for miRNA-gene regulation pairs in glioblastoma and 2 columns (the first is dataframe header).
</p>
</dd><dt>geneexp</dt><dd>
<p>a dataframe representing target regulations expression ,all of which are gene expression value,in glioblastoma.It includes 6 rows standing for 6 gene targets and 541 columns standing for 541 samples.Its rownames are gene symbols.
</p>
</dd><dt>miRexp</dt><dd>
<p>a dataframe representing miRNA expression profile in glioblastoma.It includes 57 rows standing for 57 miRNA and 541 columns standing for 541 samples.Its rownames are miRNA symbols.
</p>
</dd><dt>Pre.ceRNA</dt><dd>
<p>a data frame representing the ceRNA identified in the pre-experiment using the method provided by the software to draw a network map.
</p>
</dd><dt>mres</dt><dd>
<p>a data frame representing the MRE.
</p>
</dd><dt>GOterms</dt><dd>
<p>a list representing that goterm of interest and all gene sets in the term.
</p>
</dd><dt>background</dt><dd>
<p>a character representing a gene set in which GOterm annotated genes must be.
</p>
</dd><dt>exp.sur</dt><dd>
<p>a data frame representing specifying expression and survival information.Its rownames are sample names.Its colnames
are names in triplets and survival information
</p>
</dd><dt>train</dt><dd>
<p>a data frame representing specifying train sample names
</p>
</dd><dt>test</dt><dd>
<p>a data frame representing specifying test sample names.
</p>
</dd>
</dl>



<h3>Details</h3>

<p>All epression data is from a study about glioblastoma.The miRNA-gene regulation information is from TargetScan.
</p>


<h3>Source</h3>

<p>Network, Atlas T C G. et al.
</p>


<h3>References</h3>

<p>Network, Atlas T C G . Corrigendum: Comprehensive genomic characterization defines human glioblastoma genes and core pathways[J]. Nature, 2013, 494(7438):506-506.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dataset)
## maybe str(dataset) ; ...
</code></pre>

<hr>
<h2 id='surv.plot'>
Draw survival curve
</h2><span id='topic+surv.plot'></span>

<h3>Description</h3>

<p>It is used to draw the survival curve of ternary relationship pairs 
and to support the survival prognosis of training sets and test sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>surv.plot(ceRNA, exp.sur, train = NULL, test = NULL, index)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="surv.plot_+3A_cerna">ceRNA</code></td>
<td>

<p>a dataframe that the ceRNA relationship is the data, and the prediction result data obtained 
according to the ceRNA prediction algorithm.Such as ceRNA.cmi prediction result file.
</p>
</td></tr>
<tr><td><code id="surv.plot_+3A_exp.sur">exp.sur</code></td>
<td>

<p>dataframe specifying expression and survival information.Its rownames are sample names.Its colnames
are names in triplets and survival information (see example data in details).
</p>
</td></tr>
<tr><td><code id="surv.plot_+3A_train">train</code></td>
<td>

<p>a character string vector specifying train sample names.
</p>
</td></tr>
<tr><td><code id="surv.plot_+3A_test">test</code></td>
<td>

<p>a character string vector specifying test sample names.
</p>
</td></tr>
<tr><td><code id="surv.plot_+3A_index">index</code></td>
<td>

<p>a numeric vector (default 1) reprsenting rowindex of triplets analyed.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Survival curve of user-selected ceRNA ternary pairs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.

## The function is currently defined as
surv.plot(ceRNA=dataset[["Pre.ceRNA"]],exp.sur=dataset[["exp.sur"]],train=NULL,test=NULL,index=5)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
