<!DOCTYPE html><html><head><title>Help for package cglasso</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {cglasso}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cglasso-package'><p>Conditional Graphical LASSO for Gaussian Graphical Models with Censored and Missing Values</p></a></li>
<li><a href='#AIC.cglasso'><p>Akaike Information Criterion</p></a></li>
<li><a href='#BIC.cglasso'><p>Bayesian Information Criterion</p></a></li>
<li><a href='#cggm'><p>Post-Hoc Maximum Likelihood Refitting of a Conditional Graphical Lasso</p></a></li>
<li><a href='#cglasso'><p>Conditional Graphical Lasso Estimator</p></a></li>
<li><a href='#cglasso-internal'><p>Internal Functions</p></a></li>
<li><a href='#coef'><p>Extract Model Coefficients</p></a></li>
<li><a href='#ColMeans+20+2B+20ColMeans'><p>Calculate Column Means and Vars of a &ldquo;datacggm&rdquo; Object</p></a></li>
<li><a href='#datacggm'><p>Create a Dataset from a Conditional Gaussian Graphical Model with Censored and/or Missing Values</p></a></li>
<li><a href='#dim.datacggm'><p>Dimensions of a &ldquo;datacggm&rdquo; Object</p></a></li>
<li><a href='#dimnames.datacggm'><p>Dimnames of a &ldquo;datacggm&rdquo; Object</p></a></li>
<li><a href='#event'><p>Status Indicator Matrix from a &lsquo;<code>datacggm</code>&rsquo; Object</p></a></li>
<li><a href='#Example'>
<p>Simulated data for the cglasso vignette</p></a></li>
<li><a href='#fitted'><p>Extract Model Fitted Values</p></a></li>
<li><a href='#getGraph'><p>Retrieve Graphs from a &lsquo;<code>cglasso2igraph</code>&rsquo; Object</p></a></li>
<li><a href='#getMatrix'><p>Retrieve Matrices &lsquo;<code>Y</code>&rsquo; and &lsquo;<code>X</code>&rsquo; from a &lsquo;<code>datacggm</code>&rsquo; Object</p></a></li>
<li><a href='#hist.datacggm'><p>Histogram for a <code>datacggm</code> Object</p></a></li>
<li><a href='#impute'><p>Imputation of Missing and Censored Values</p></a></li>
<li><a href='#is.cglasso2igraph'><p>Is an Object of Class &lsquo;cglasso2igraph&rsquo;?</p></a></li>
<li><a href='#is.datacggm'><p>Is an Object of Class &lsquo;datacggm&rsquo;?</p></a></li>
<li><a href='#lower+20+2B+20upper'><p>Lower and Upper Limits from a &ldquo;datacggm&rdquo; Object</p></a></li>
<li><a href='#MKMEP'>
<p>Megakaryocyte-Erythroid Progenitors</p></a></li>
<li><a href='#MM'>
<p>The Rule of miRNA in Multiple Myeloma</p></a></li>
<li><a href='#nobs+20+2B+20nresp+20+2B+20npred'><p>Extract the Number of Observations/Responses/Predictors from a datacggm Object</p></a></li>
<li><a href='#plot.cggm'><p>Plot Method for a &lsquo;<code>cggm</code>&rsquo; Object</p></a></li>
<li><a href='#plot.cglasso'><p>Plot Method for &lsquo;<code>cglasso</code>&rsquo; Object</p></a></li>
<li><a href='#plot.cglasso2igraph'><p>Plot Method for a cglasso2igraph Object&quot;</p></a></li>
<li><a href='#plot.GoF'><p>Plot for &lsquo;<code>GoF</code>&rsquo; Object</p></a></li>
<li><a href='#predict'><p>Predict Method for cglasso and cggm Fits</p></a></li>
<li><a href='#QFun'><p>Extract Q-Function</p></a></li>
<li><a href='#qqcnorm'><p>Quantile-Quantile Plots for a <code>datacggm</code> Object</p></a></li>
<li><a href='#rcggm'><p>Simulate Data from a Conditional Gaussian Graphical Model with Censored and/or Missing Values</p></a></li>
<li><a href='#residuals'><p>Extract Model Residuals</p></a></li>
<li><a href='#rowNames+20+2B+20colNames'><p>Row and Column Names of a &ldquo;datacggm&rdquo; Object</p></a></li>
<li><a href='#select_cglasso'><p>Model Selection for the Conditional Graphical Lasso Estimator</p></a></li>
<li><a href='#ShowStructure'><p>Show Package Structure</p></a></li>
<li><a href='#summary.cglasso'><p>Summarizing cglasso and cggm Fits</p></a></li>
<li><a href='#summary.datacggm'><p>Summarizing Objects of Class &lsquo;<code>datacggm</code>&rsquo;</p></a></li>
<li><a href='#to_graph'><p>Create Graphs from cglasso or cggm Objects</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>2.0.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-02-10</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Conditional Graphical LASSO for Gaussian Graphical Models with
Censored and Missing Values</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0), igraph</td>
</tr>
<tr>
<td>Description:</td>
<td>Conditional graphical lasso estimator is an extension of the graphical lasso proposed to estimate the conditional dependence structure of a set of p response variables given q predictors. This package provides suitable extensions developed to study datasets with censored and/or missing values. Standard conditional graphical lasso is available as a special case. Furthermore, the package provides an integrated set of core routines for visualization, analysis, and simulation of datasets with censored and/or missing values drawn from a Gaussian graphical model. Details about the implemented models can be found in Augugliaro et al. (2023) &lt;<a href="https://doi.org/10.18637%2Fjss.v105.i01">doi:10.18637/jss.v105.i01</a>&gt;, Augugliaro et al. (2020b) &lt;<a href="https://doi.org/10.1007%2Fs11222-020-09945-7">doi:10.1007/s11222-020-09945-7</a>&gt;, Augugliaro et al. (2020a) &lt;<a href="https://doi.org/10.1093%2Fbiostatistics%2Fkxy043">doi:10.1093/biostatistics/kxy043</a>&gt;, Yin et al. (2001) &lt;<a href="https://doi.org/10.1214%2F11-AOAS494">doi:10.1214/11-AOAS494</a>&gt; and Stadler et al. (2012) &lt;<a href="https://doi.org/10.1007%2Fs11222-010-9219-7">doi:10.1007/s11222-010-9219-7</a>&gt;.</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, MASS</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Author:</td>
<td>Luigi Augugliaro <a href="https://orcid.org/0000-0002-4603-7541"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Gianluca Sottile <a href="https://orcid.org/0000-0001-9347-7251"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Ernst C. Wit <a href="https://orcid.org/0000-0002-3671-9610"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Veronica Vinciotti
    <a href="https://orcid.org/0000-0002-2625-7977"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Luigi Augugliaro &lt;luigi.augugliaro@unipa.it&gt;</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-12 08:26:37 UTC; luigiaugugliaro</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-12 08:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='cglasso-package'>Conditional Graphical LASSO for Gaussian Graphical Models with Censored and Missing Values</h2><span id='topic+cglasso-package'></span>

<h3>Description</h3>

<p>Conditional graphical lasso (cglasso) estimator (Yin <em>and other</em>, 2011) is an extension of the graphical lasso (Yuan <em>and other</em>, 2007) proposed to estimate the conditional dependence structure of a set of p response variables given q predictors. This package provides suitable extensions developed to study datasets with censored and/or missing values (Augugliaro <em>and other</em>, 2020a and Augugliaro <em>and other</em>, 2020b). Standard conditional graphical lasso is available as a special case. Furthermore, the cglasso package provides an integrated set of core routines for visualization, analysis, and simulation of datasets with censored and/or missing values drawn from a Gaussian graphical model.</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> cglasso</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.0.6</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2023-01-17</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;=2)
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Luigi Augugliaro [aut, cre],<br />
Gianluca Sottile [aut]<br />
Ernst C. Wit [aut]<br />
Veronica Vinciotti [aut]<br />
Maintainer: Luigi Augugliaro &lt;<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>&gt;
</p>


<h3>References</h3>

<p>Augugliaro L., Sottile G., Wit E.C., and Vinciotti V. (2023) &lt;<a href="https://doi.org/10.18637/jss.v105.i01">doi:10.18637/jss.v105.i01</a>&gt;.
cglasso: An R Package for Conditional Graphical Lasso Inference with Censored and Missing Values.
<em>Journal of Statistical Software</em> <b>105</b>(1), 1&ndash;58.
</p>
<p>Augugliaro, L., Sottile, G., and Vinciotti, V. (2020a) &lt;<a href="https://doi.org/10.1007/s11222-020-09945-7">doi:10.1007/s11222-020-09945-7</a>&gt;.
The conditional censored graphical lasso estimator.
<em>Statistics and Computing</em> <b>30</b>, 1273&ndash;1289.
</p>
<p>Augugliaro, L., Abbruzzo, A., and Vinciotti, V. (2020b) &lt;<a href="https://doi.org/10.1093/biostatistics/kxy043">doi:10.1093/biostatistics/kxy043</a>&gt;.
<code class="reqn">\ell_1</code>-Penalized censored Gaussian graphical model.
<em>Biostatistics</em> <b>21</b>, e1&ndash;e16.
</p>
<p>Friedman, J.H., Hastie, T., and Tibshirani, R. (2008) &lt;<a href="https://doi.org/10.1093/biostatistics/kxm045">doi:10.1093/biostatistics/kxm045</a>&gt;.
Sparse inverse covariance estimation with the graphical lasso.
<em>Biostatistics</em> <b>9</b>, 432&ndash;441.
</p>
<p>Yin, J. and Li, H. (2001) &lt;<a href="https://doi.org/10.1214/11-AOAS494">doi:10.1214/11-AOAS494</a>&gt;.
A sparse conditional Gaussian graphical model for analysis of genetical genomics data.
<em>The Annals of Applied Statistics</em> <b>5</b>(4), 2630&ndash;2650.
</p>
<p>Yuan, M., and Lin, Y. (2007) &lt;<a href="https://doi.org/10.1093/biomet/asm018">doi:10.1093/biomet/asm018</a>&gt;.
Model selection and estimation in the Gaussian graphical model.
<em>Biometrika</em> <b>94</b>, 19&ndash;35.
</p>

<hr>
<h2 id='AIC.cglasso'>Akaike Information Criterion</h2><span id='topic+AIC'></span><span id='topic+AIC.cglasso'></span>

<h3>Description</h3>

<p>&lsquo;<code>AIC</code>&rsquo; computes the &lsquo;Akaike Information Criterion&rsquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cglasso'
AIC(object, k = 2, mle, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AIC.cglasso_+3A_object">object</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object inheriting class &lsquo;<code>cglasso</code>&rsquo;, that is, the output of the model-fitting functions <code><a href="#topic+cglasso">cglasso</a></code> and <code><a href="#topic+cggm">cggm</a></code>.</p>
</td></tr>
<tr><td><code id="AIC.cglasso_+3A_k">k</code></td>
<td>
<p>the <em>penalty</em> parameter to be used; the default <code>k = 2</code> is the classical AIC.</p>
</td></tr>
<tr><td><code id="AIC.cglasso_+3A_mle">mle</code></td>
<td>
<p>logical. TRUE if the measure of goodness-of-fit should be computed using the maximum likelihood estimates. Default depends on the class of the argument <code>object</code>: <code>mle = FALSE</code> for objects of class <code>cglasso</code> and <code>mle = TRUE</code> for objects of class <code>cggm</code>.</p>
</td></tr>
<tr><td><code id="AIC.cglasso_+3A_...">...</code></td>
<td>
<p>further arguments passed to <code><a href="#topic+cggm">cggm</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>&lsquo;<code>AIC</code>&rsquo; computes the following measure of goodness-of-fit (Ibrahim <em>and other</em>, 2008): </p>
<p style="text-align: center;"><code class="reqn">-2\,\mbox{Q-function} + k\,\mbox{df},</code>
</p>
<p> where <code class="reqn">k</code> is the <em>penalty</em> parameter and <code class="reqn">\mbox{df}</code> represents the number of unique non-zero parameters in the fitted model.
</p>
<p>The values of the Q-function function are computed using <code><a href="#topic+QFun">QFun</a></code>. By default, for an object of class <code>cglasso</code> these values are computed using the penalized estimates whereas maximum likelihood estimates are used if the <code>object</code> is of class <code>cggm</code> (see argument &lsquo;<code>mle</code>&rsquo; in <code><a href="#topic+QFun">QFun</a></code>).
</p>
<p>The Akaike Information Criterion (AIC) is returned by letting <code class="reqn">k = 2</code> (default value of the function <code>AIC</code>) whereas the &lsquo;Bayesian Information Criterion&rsquo; (BIC) is returned by letting <code class="reqn">k = \log(n)</code>, where <code class="reqn">n</code> is the sample size.
</p>
<p>Function <code>AIC</code> can be passed to the functions <code><a href="#topic+select_cglasso">select_cglasso</a></code> and <code><a href="#topic+summary.cglasso">summary.cglasso</a></code> to select and print the optimal fitted model, respectively.
</p>
<p>The function <code><a href="#topic+plot.GoF">plot.GoF</a></code> can be used to graphically evaluate the behaviour of the fitted models in terms of goodness-of-fit.
</p>


<h3>Value</h3>

<p>&lsquo;<code>AIC</code>&rsquo; return an <span class="rlang"><b>R</b></span> object of S3 class &ldquo;<code>GoF</code>&rdquo;, i.e., a named list containing the following components:
</p>
<table>
<tr><td><code>value_gof</code></td>
<td>
<p>a matrix storing the values of the measure of goodness-of-fit used to evaluate the fitted models.</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>a matrix storing the number of the estimated non-zero parameters.</p>
</td></tr>
<tr><td><code>dfB</code></td>
<td>
<p>a matrix storing the number of estimated non-zero regression coefficients.</p>
</td></tr>
<tr><td><code>dfTht</code></td>
<td>
<p>a matrix storing the number of estimated non-zero partial correlation coefficients.</p>
</td></tr>
<tr><td><code>value</code></td>
<td>
<p>a matrix storing the values of the Q-function.</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>the sample size.</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>the number of response variables.</p>
</td></tr>
<tr><td><code>q</code></td>
<td>
<p>the number of columns of the design matrix <code>X</code> used to fit the model.</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>the <code class="reqn">\lambda</code>-values used to fit the model.</p>
</td></tr>
<tr><td><code>nlambda</code></td>
<td>
<p>the number of <code class="reqn">\lambda</code>-values used.</p>
</td></tr>
<tr><td><code>rho</code></td>
<td>
<p>the <code class="reqn">\rho</code>-values used to fit the model.</p>
</td></tr>
<tr><td><code>nrho</code></td>
<td>
<p>the number of <code class="reqn">\rho</code>-values used.</p>
</td></tr>
<tr><td><code>type</code></td>
<td>
<p>a description of the computed measure of goodness-of-fit.</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>a description of the fitted model passed through the argument <code>object</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>References</h3>

<p>Ibrahim, J.G., Zhu, H. and Tang, N. (2008) &lt;<a href="https://doi.org/10.1198/016214508000001057">doi:10.1198/016214508000001057</a>&gt;.
Model selection criteria for missing-data problems using the EM algorithm.
<em>Journal of the American Statistical Association</em> <b>103</b>, 1648&ndash;1658.
</p>
<p>Sakamoto, Y., Ishiguro, M., and Kitagawa, G. (1986).
<em>Akaike Information Criterion Statistics</em>.
D. Reidel Publishing Company.
</p>
<p>Wit, E., Heuvel, E. V. D., &amp; Romeijn, J. W. (2012) &lt;<a href="https://doi.org/10.1111/j.1467-9574.2012.00530.x">doi:10.1111/j.1467-9574.2012.00530.x</a>&gt;.
All models are wrong...?: an introduction to model uncertainty.
<em>Statistica Neerlandica</em>, 66(3), 217-236.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>, <code><a href="#topic+cglasso">cglasso</a></code>, <code><a href="#topic+cggm">cggm</a></code>, <code><a href="#topic+QFun">QFun</a></code>, <code><a href="#topic+plot.GoF">plot.GoF</a></code> and <code><a href="#topic+summary.cglasso">summary.cglasso</a></code>  
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)

# Y ~ N(b0 + XB, Sigma) and probability of left/right censored values equal to 0.05
n &lt;- 100L
p &lt;- 3L
q &lt;- 2L
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probl = 0.05, probr = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
AIC(out)

out.mle &lt;- cggm(out, lambda.id = 3L, rho.id = 3L)
AIC(out.mle)
</code></pre>

<hr>
<h2 id='BIC.cglasso'>Bayesian Information Criterion</h2><span id='topic+BIC'></span><span id='topic+BIC.cglasso'></span>

<h3>Description</h3>

<p>&lsquo;<code>BIC</code>&rsquo; computes the Bayesian Information Criterion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cglasso'
BIC(object, g = 0, type, mle, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BIC.cglasso_+3A_object">object</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object inheriting class &lsquo;<code>cglasso</code>&rsquo;, that is, the output of the model-fitting functions <code><a href="#topic+cglasso">cglasso</a></code> and <code><a href="#topic+cggm">cggm</a></code>.</p>
</td></tr>
<tr><td><code id="BIC.cglasso_+3A_g">g</code></td>
<td>
<p>a value belonging to the interval <code class="reqn">[0, 1]</code>. Classical BIC is returned by letting <code>g = 0</code> (default value), whereas extended BIC corresponds to the case <code>g = 0.5</code>.</p>
</td></tr>
<tr><td><code id="BIC.cglasso_+3A_type">type</code></td>
<td>
<p>character; if <code>g</code> is not zero then the measure proposed in Foygel <em>and other</em> (2010) is returned by setting <code>type = "FD"</code>, otherwise (<code>type = "CC"</code>) returns the measure proposed in Chen <em>and other</em> (2008, 2012). See section &lsquo;Details&rsquo; for more details.</p>
</td></tr>
<tr><td><code id="BIC.cglasso_+3A_mle">mle</code></td>
<td>
<p>logical. TRUE if the measure of goodness-of-fit should be computed using the maximum likelihood estimates. Default depends on the class of the argument <code>object</code>: <code>mle = FALSE</code> for objects of class <code>cglasso</code> and <code>mle = TRUE</code> for objects of class <code>cggm</code>.</p>
</td></tr>
<tr><td><code id="BIC.cglasso_+3A_...">...</code></td>
<td>
<p>further arguments passed to the model-fitting function <code><a href="#topic+cggm">cggm</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>&lsquo;<code>BIC</code>&rsquo; computes the Bayesian Information Criterion (BIC) for models fitted by <code><a href="#topic+cglasso">cglasso</a></code> or <code><a href="#topic+cggm">cggm</a></code>. As proposed in Ibrahim <em>and other</em> (2008), <code>BIC</code> computes the measure of goodness-of-fit by replacing the log-likelihood function with the Q-function, that is, the function maximized in the M-Step of the EM-algorithm. The values of the Q-function are computed using <code><a href="#topic+QFun">QFun</a></code>. By default, for an object of class <code>cglasso</code> these values are computed using the penalized estimates whereas, if the <code>object</code> has class <code>cggm</code>, maximum likelihood estimates are used (see argument &lsquo;<code>mle</code>&rsquo; in <code><a href="#topic+QFun">QFun</a></code>).
</p>
<p>By default, <code>BIC</code> computes the standard BIC measure (<code class="reqn">\gamma = 0</code>): </p>
<p style="text-align: center;"><code class="reqn">-2\,\mbox{Q-function} + \log(n)\,\mbox{df},</code>
</p>
<p> where <code class="reqn">n</code> is the sample size and <code class="reqn">\mbox{df}</code> represents the number of unique non-zero parameters in the fitted model.
</p>
<p>If <code class="reqn">\gamma \ne 0</code>, the default depends on the number of predictors (<code class="reqn">q</code>).
</p>
<p>If <code class="reqn">q = 0</code>, <code>BIC</code> computes the measure of goodness-of-fit proposed in Foygel <em>and other</em> (2010) (<code>type = "FD"</code>): 
</p>
<p style="text-align: center;"><code class="reqn">\mbox{eBIC} = -2\,\mbox{QFun} + (\log n + 4 \, \gamma \, log \, p)\,\mbox{df},</code>
</p>
 
<p>where <code class="reqn">\gamma</code> is a value belonging to the interval <code class="reqn">[0, 1]</code> and indexing the measure of goodness-of-fit.
</p>
<p>If <code class="reqn">q \ne 0</code> , <code>BIC</code> computes the measure of goodness-of-fit proposed in Chen <em>and other</em> (2008, 2012) (<code>type = "CC"</code>): 
</p>
<p style="text-align: center;"><code class="reqn">\mbox{eBIC} = -2\,\mbox{QFun} + (\log n + 2 \, \gamma \, log \, q)\,\mbox{df},</code>
</p>
 
<p><code>BIC</code> can be passed to the functions  <code><a href="#topic+select_cglasso">select_cglasso</a></code> and <code><a href="#topic+summary.cglasso">summary.cglasso</a></code> to select and print the optimal fitted model, respectively.
</p>
<p>The function <code><a href="#topic+plot.GoF">plot.GoF</a></code> can be used to graphically evaluate the behaviour of the fitted models in terms of goodness-of-fit.
</p>


<h3>Value</h3>

<p>&lsquo;<code>BIC</code>&rsquo; returns an <span class="rlang"><b>R</b></span> object of S3 class &ldquo;<code>GoF</code>&rdquo;, i.e. a named list containing the following components:
</p>
<table>
<tr><td><code>value_gof</code></td>
<td>
<p>a matrix storing the values of the measure of goodness-of-fit used to evaluate the fitted models.</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>a matrix storing the number of the estimated non-zero parameters.</p>
</td></tr>
<tr><td><code>dfB</code></td>
<td>
<p>a matrix storing the number of estimated non-zero regression coefficients.</p>
</td></tr>
<tr><td><code>dfTht</code></td>
<td>
<p>a matrix storing the number of estimated non-zero partial correlation coefficients.</p>
</td></tr>
<tr><td><code>value</code></td>
<td>
<p>a matrix storing the values of the Q-function.</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>the sample size.</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>the number of response variables.</p>
</td></tr>
<tr><td><code>q</code></td>
<td>
<p>the number of columns of the design matrix <code>X</code> used to fit the model.</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>the <code class="reqn">\lambda</code>-values used to fit the model.</p>
</td></tr>
<tr><td><code>nlambda</code></td>
<td>
<p>the number of <code class="reqn">\lambda</code>-values used.</p>
</td></tr>
<tr><td><code>rho</code></td>
<td>
<p>the <code class="reqn">\rho</code>-values used to fit the model.</p>
</td></tr>
<tr><td><code>nrho</code></td>
<td>
<p>the number of <code class="reqn">\rho</code>-values used.</p>
</td></tr>
<tr><td><code>type</code></td>
<td>
<p>a description of the computed measure of goodness-of-fit.</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>a description of the fitted model passed through the argument <code>object</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>References</h3>

<p>Foygel, R. and Drton, M. (2010).
Extended Bayesian Information Criteria for Gaussian Graphical Models.
In: Lafferty, J., Williams, C., Shawe-taylor, J., Zemel, R.s. and Culott, A. (editors),
<em>Advances in Neural Information Processing Systems 23</em>. pp. 604&ndash;612.
</p>
<p>Chen, J. and Chen, Z. (2008) &lt;<a href="https://doi.org/10.1093/biomet/asn034">doi:10.1093/biomet/asn034</a>&gt;.
Extended Bayesian information criteria for model selection with large model spaces.
<em>Biometrika</em>, Vol. 95(2), pp. 759&ndash;771.
</p>
<p>Chen, J. and Chen, Z. (2012) &lt;<a href="https://doi.org/10.5705/ss.2010.216">doi:10.5705/ss.2010.216</a>&gt;.
Extended BIC for small-n-large-p sparse GLM.
<em>Statistica Sinica</em>, Vol. 22, pp. 555&ndash;574.
</p>
<p>Wit, E., Heuvel, E. V. D., &amp; Romeijn, J. W. (2012) &lt;<a href="https://doi.org/10.1111/j.1467-9574.2012.00530.x">doi:10.1111/j.1467-9574.2012.00530.x</a>&gt;.
All models are wrong...?: an introduction to model uncertainty.
<em>Statistica Neerlandica</em>, 66(3), 217-236.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cglasso">cglasso</a></code>, <code><a href="#topic+cggm">cggm</a></code>, <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code>, <code><a href="#topic+QFun">QFun</a></code>, <code><a href="#topic+plot.GoF">plot.GoF</a></code> and <code><a href="#topic+summary.cglasso">summary.cglasso</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)

# Y ~ N(0, Sigma) and probability of left/right censored values equal to 0.05
n &lt;- 100L
p &lt;- 3L
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, Sigma = Sigma, probl = 0.05, probr = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
BIC(out)                                    # standard BIC measure
BIC(out, mle = TRUE, g = 0.5, type = "FD")  # eBIC proposed in Foygel and other (2010)

# Y ~ N(b0 + XB, Sigma)  and probability of left/right censored values equal to 0.05
n &lt;- 100L
p &lt;- 3L
q &lt;- 2
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probl = 0.05, probr = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
BIC(out)                                    # standard BIC measure
BIC(out, mle = TRUE, g = 0.5, type = "FD")  # eBIC proposed in Foygel and other (2010)
BIC(out, mle = TRUE, g = 0.5, type = "CC")  # eBIC proposed in Chen and other (2008, 2010)
</code></pre>

<hr>
<h2 id='cggm'>Post-Hoc Maximum Likelihood Refitting of a Conditional Graphical Lasso</h2><span id='topic+cggm'></span>

<h3>Description</h3>

<p>&lsquo;<code>cggm</code>&rsquo; is used to perform post-hoc maximum likelihood refitting of a selected conditional graphical lasso model with censored and/or missing values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cggm(object, GoF = AIC, lambda.id, rho.id, tp.min = 1.0E-6, ntp = 100L,
     maxit.em = 1.0E+4, thr.em = 1.0E-3, maxit.bcd = 1.0E+5, thr.bcd = 1.0E-4, 
     trace = 0L, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cggm_+3A_object">object</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object of S3 class &lsquo;<code>cglasso</code>&rsquo;, that is, the output of the function <code><a href="#topic+cglasso">cglasso</a></code>.</p>
</td></tr>
<tr><td><code id="cggm_+3A_gof">GoF</code></td>
<td>
<p>a valid goodness-of-fit function, such as <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code> or <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>, or an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>GoF</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="cggm_+3A_lambda.id">lambda.id</code></td>
<td>
<p>an optional integer used to identify a <code class="reqn">\lambda</code>-value stored in <code>object</code>. See section &lsquo;Details&rsquo; for more details.</p>
</td></tr>
<tr><td><code id="cggm_+3A_rho.id">rho.id</code></td>
<td>
<p>an optional integer used to identify a <code class="reqn">\rho</code>-value stored in <code>object</code>. See section &lsquo;Details&rsquo; for more details.</p>
</td></tr>
<tr><td><code id="cggm_+3A_tp.min">tp.min</code></td>
<td>
<p>the smallest <code class="reqn">\lambda</code> and <code class="reqn">\rho</code> value. See section &lsquo;Details&rsquo; for more details.</p>
</td></tr>
<tr><td><code id="cggm_+3A_ntp">ntp</code></td>
<td>
<p>integer; number of <code class="reqn">\lambda</code>- and <code class="reqn">\rho</code>-values used to compute the coefficients path. See section &lsquo;Details&rsquo; for more details.</p>
</td></tr>
<tr><td><code id="cggm_+3A_maxit.em">maxit.em</code></td>
<td>
<p>maximum number of iterations of the EM algorithm. Default is <code>1.0E+4</code>.</p>
</td></tr>
<tr><td><code id="cggm_+3A_thr.em">thr.em</code></td>
<td>
<p>threshold for the convergence of the EM algorithm. Default value is <code>1.0E-4</code>.</p>
</td></tr>
<tr><td><code id="cggm_+3A_maxit.bcd">maxit.bcd</code></td>
<td>
<p>maximum number of iterations of the glasso algorithm. Default is <code>1.0E+5</code>.</p>
</td></tr>
<tr><td><code id="cggm_+3A_thr.bcd">thr.bcd</code></td>
<td>
<p>threshold for the convergence of the glasso algorithm. Default is <code>1.0E-4</code>.</p>
</td></tr>
<tr><td><code id="cggm_+3A_trace">trace</code></td>
<td>
<p>integer for printing information out as iterations proceed: <code>trace = 0</code> no information is printed out on screen; <code>trace = 1</code> minimal information is printed; <code>trace = 2</code> detailed information is printed on screen.</p>
</td></tr>
<tr><td><code id="cggm_+3A_...">...</code></td>
<td>
<p>the <em>penalty</em> parameter passed to the goodness-of-fit function (argument &lsquo;<code>GoF</code>&rsquo;).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The model-fitting function <code>cggm</code> is used to obtain the maximum likelihood estimates of a censored Gaussian graphical model whose structure was found by penalised cglasso inference. That is, given a fitted cglasso model, identified along the path via a goodness-of-fit function (passed by <code>GoF</code>) or via the optional arguments <code>lambda.id</code> and <code>rho.id</code>), <code>cggm</code> computes the maximum likelihood estimates of the parameters with zero constraints associated to the given structure.
</p>
<p>Maximum likelihood estimates are computed using <code>cglasso</code> as workhorse function, that is, <code>cggm</code> fits a sequence of cglasso models, with length equal to <code>ntp</code>, reducing <code class="reqn">\lambda</code> and <code class="reqn">\rho</code> until they are equal to the value &lsquo;<code>min(tp.min, lambda.min, rho.min)</code>&rsquo;, where <code>lambda.min</code> and <code>rho.min</code> are the smallest <code class="reqn">\lambda</code>- and <code class="reqn">\rho</code>-values stored in <code>object</code>. Maximum likelihood estimates are obtained from the last fitted cglasso model.
</p>
<p>The model-fitting function <code>cggm</code> returns an <span class="rlang"><b>R</b></span> object inheriting the S3 class &lsquo;<code>cglasso</code>&rsquo;, for which all printing and plotting functions, designed for a <code>cglasso</code> object, can be used (see section &lsquo;<code>See Also</code>&rsquo;). Function <code><a href="#topic+ShowStructure">ShowStructure</a></code> can be used to show the structure of the package.
</p>
<p>NOTE: if the user tries a problem when the sample size is not large enough, then increase the number of fitted cglasso models (argument <code>ntp</code>).
</p>


<h3>Value</h3>

<p><code>cggm</code> returns an object of S3 class &ldquo;<code>cggm</code>&rdquo;, i.e., a list containing the
following components:
</p>
<table>
<tr><td><code>call</code></td>
<td>
<p>the call that produced this object.</p>
</td></tr>
<tr><td><code>Yipt</code></td>
<td>
<p>an array storing the &lsquo;working response matrix&rsquo;.</p>
</td></tr>
<tr><td><code>B</code></td>
<td>
<p>the maximum likelihood estimate of the regression coefficient matrix.</p>
</td></tr>
<tr><td><code>mu</code></td>
<td>
<p>the fitted expected values.</p>
</td></tr>
<tr><td><code>R</code></td>
<td>
<p>the &lsquo;working residuals&rsquo; matrix.</p>
</td></tr>
<tr><td><code>S</code></td>
<td>
<p>the &lsquo;working empirical covariance matrix&rsquo;.</p>
</td></tr>
<tr><td><code>Sgm</code></td>
<td>
<p>the maximum likelihood estimate of the covariance matrix.</p>
</td></tr>
<tr><td><code>Tht</code></td>
<td>
<p>the maximum likelihood estimate of the precision matrix.</p>
</td></tr>
<tr><td><code>dfB</code></td>
<td>
<p>the number of estimated non-zero regression coefficients. Only for internal purpose.</p>
</td></tr>
<tr><td><code>dfTht</code></td>
<td>
<p>the number of estimated non-zero (off-diagonal) partial correlation coefficients. Only for internal purpose.</p>
</td></tr>
<tr><td><code>InfoStructure</code></td>
<td>
<p>a named list whose elements are used to store the information about the estimated networks. Only for internal purpose.</p>
</td></tr>
<tr><td><code>nit</code></td>
<td>
<p>the number of EM steps.</p>
</td></tr>
<tr><td><code>Z</code></td>
<td>
<p>the &lsquo;<code>datacggm</code>&rsquo; object used to compute the censored graphical lasso estimator.</p>
</td></tr>
<tr><td><code>nlambda</code></td>
<td>
<p>Only for internal purpose.</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>the <code class="reqn">\lambda</code>-value of the selected cglasso model.</p>
</td></tr>
<tr><td><code>nrho</code></td>
<td>
<p>Only for internal purpose.</p>
</td></tr>
<tr><td><code>rho</code></td>
<td>
<p>the <code class="reqn">\rho</code>-value of the selected cglasso model.</p>
</td></tr>
<tr><td><code>maxit.em</code></td>
<td>
<p>maximum number of iterations of the EM algorithm.</p>
</td></tr>
<tr><td><code>thr.em</code></td>
<td>
<p>threshold for the convergence of the EM algorithm.</p>
</td></tr>
<tr><td><code>maxit.bcd</code></td>
<td>
<p>maximum number of iterations of the glasso algorithm.</p>
</td></tr>
<tr><td><code>thr.bcd</code></td>
<td>
<p>threshold for the convergence of the glasso algorithm.</p>
</td></tr>
<tr><td><code>conv</code></td>
<td>
<p>a description of the error that has occurred.</p>
</td></tr>
<tr><td><code>subrout</code></td>
<td>
<p>the name of the Fortran subroutine where the error has occurred (for internal debug only).</p>
</td></tr>
<tr><td><code>trace</code></td>
<td>
<p>the integer used for printing information on screen.</p>
</td></tr>
<tr><td><code>nobs</code></td>
<td>
<p>the sample size</p>
</td></tr>
<tr><td><code>nresp</code></td>
<td>
<p>the number of response variables used to fit the model.</p>
</td></tr>
<tr><td><code>npred</code></td>
<td>
<p>the number of predictors used to fit the model.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>See Also</h3>

<p><code><a href="#topic+cglasso">cglasso</a></code>, <code><a href="#topic+coef.cglasso">coef.cglasso</a></code>, <code><a href="#topic+fitted.cglasso">fitted.cglasso</a></code>, <code><a href="#topic+residuals.cglasso">residuals.cglasso</a></code>, <code><a href="#topic+predict.cggm">predict.cggm</a></code>, <code><a href="#topic+impute">impute</a></code>, <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code>, <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>, <code><a href="#topic+summary.cglasso">summary.cglasso</a></code>, <code><a href="#topic+to_graph">to_graph</a></code>, <code><a href="#topic+plot.cglasso2igraph">plot.cglasso2igraph</a></code> and <code><a href="#topic+ShowStructure">ShowStructure</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
# Y ~ N(XB, Sigma)  and
# 1. probability of left/right censored values equal to 0.05
# 2. probability of missing-at-random values equal to 0.05
n &lt;- 100L
p &lt;- 3L
q &lt;- 2L
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probl = 0.05, probr = 0.5, probna = 0.05)
out &lt;- cglasso(. ~ ., data = Z)

# MLE of the censored Gaussian graphical model identified by 'BIC'
out.mle &lt;- cggm(out, GoF = BIC)
out.mle

# accessor functions
coef(out.mle, drop = TRUE)
fitted(out.mle, drop = TRUE)
residuals(out.mle, type = "working", drop = TRUE)
impute(out.mle, type = "both")

# goodness-of-fit functions
AIC(out.mle)
BIC(out.mle)
summary(out.mle)

# network analysis
out.graph &lt;- plot(out.mle)
out.graph
</code></pre>

<hr>
<h2 id='cglasso'>Conditional Graphical Lasso Estimator</h2><span id='topic+cglasso'></span>

<h3>Description</h3>

<p>&lsquo;<code>cglasso</code>&rsquo; fits the conditional graphical lasso model to datasets with censored and/or missing values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cglasso(formula, data, subset, contrasts = NULL, diagonal = FALSE,
        weights.B = NULL, weights.Tht = NULL, nlambda, lambda.min.ratio,
        lambda, nrho, rho.min.ratio, rho, maxit.em = 1.0E+4, thr.em = 1.0E-3,
        maxit.bcd = 1.0E+5, thr.bcd = 1.0E-4, trace = 0L)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cglasso_+3A_formula">formula</code></td>
<td>
<p>an object of class &lsquo;<code>formula</code>&rsquo; (or one that can be coerced to that class): a symbolic description of the model to be fitted.</p>
</td></tr>
<tr><td><code id="cglasso_+3A_data">data</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object of S3 class &lsquo;<code>datacggm</code>&rsquo;, that is, the output of the function <code><a href="#topic+datacggm">datacggm</a></code>. See section &lsquo;Description&rsquo; for more details.</p>
</td></tr>
<tr><td><code id="cglasso_+3A_subset">subset</code></td>
<td>
<p>an optional vector specifying a subset of observations to be used in the fitting process.</p>
</td></tr>
<tr><td><code id="cglasso_+3A_contrasts">contrasts</code></td>
<td>
<p>an optional list. See the <code>contrasts.arg</code> of <code><a href="stats.html#topic+model.matrix.default">model.matrix.default</a></code>.</p>
</td></tr>
<tr><td><code id="cglasso_+3A_diagonal">diagonal</code></td>
<td>
<p>logical. Should diagonal entries of the concentration matrix be penalized? Default is &lsquo;<code>diagonal = FALSE</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="cglasso_+3A_weights.b">weights.B</code></td>
<td>
<p>an optional <code class="reqn">q\times p</code> dimensional matrix of non-negative weights used to penalize the regression coefficients (intercepts are unpenalized). This matrix can be also used to specify the unpenalized regression coefficients (&lsquo;<code>weights.B[i, j] = 0</code>&rsquo;) or the structural zeros (&lsquo;<code>weights.B[i, j] = +Inf</code>&rsquo;). By default, all weights are set equal to 1, meaning that the penalized regression coefficients are unweighted.</p>
</td></tr>
<tr><td><code id="cglasso_+3A_weights.tht">weights.Tht</code></td>
<td>
<p>an optional symmetric matrix of non-negative weights used to penalize the partial regression coefficients. This matrix can be used to specify the unpenalized partial correlation coefficients (&lsquo;<code>weights.Tht[i, j] = 0</code>&rsquo;) or the structural zeros in the precision matrix (&lsquo;<code>weights.Tht[i, j] = +Inf</code>&rsquo;). By default, the off-diagonal entries of the matrix &lsquo;<code>weights.Tht</code>&rsquo;  are set equal to 1, meaning that the partial correlation coefficients are unweighted, whereas the diagonal entries are equal to 0, that is the diagonal entries of the precision matrix are unpenalized.</p>
</td></tr>
<tr><td><code id="cglasso_+3A_nlambda">nlambda</code></td>
<td>
<p>integer. The number of <code class="reqn">\lambda</code>-values used to penalize the regression coefficients. By default &lsquo;<code>nlambda = 10</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="cglasso_+3A_lambda.min.ratio">lambda.min.ratio</code></td>
<td>
<p>the smallest <code class="reqn">\lambda</code>-value is defined as a fraction of &lsquo;<code>lambda.max</code>&rsquo; (i.e., the smallest <code class="reqn">\lambda</code>-value for which all the estimated regression coefficients are equal to zero). The default depends on the sample size &lsquo;<code class="reqn">n</code>&rsquo; relative to the number of predictors &lsquo;<code class="reqn">q</code>&rsquo;. If &lsquo;<code class="reqn">q &lt; n</code>&rsquo;, default is &lsquo;1.0E-6&rsquo; otherwise the value &lsquo;1.0E-2&rsquo; is used as default. A very small value of &lsquo;<code>lambda.min.ratio</code>&rsquo; will lead to a saturated fitted model in the &lsquo;<code class="reqn">q &lt; n</code>&rsquo; case.</p>
</td></tr>
<tr><td><code id="cglasso_+3A_lambda">lambda</code></td>
<td>
<p>an optional user-supplied decreasing sequence of <code class="reqn">\lambda</code>-values. By default <code>cglasso</code> computes a sequence of <code class="reqn">\lambda</code>-values using <code>nlambda</code> and <code>lambda.min.ratio</code>. If <code>lambda</code> is supplied then <code>nlambda</code> and <code>lambda.min.ratio</code> are overwritten. WARNING: use with care and avoid supplying a single <code class="reqn">\lambda</code>-value; supply instead a decreasing sequence.</p>
</td></tr>
<tr><td><code id="cglasso_+3A_nrho">nrho</code></td>
<td>
<p>integer. The number of <code class="reqn">\rho</code>-values used to penalize the partial correlation coefficients. By default &lsquo;<code>nrho = 10</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="cglasso_+3A_rho.min.ratio">rho.min.ratio</code></td>
<td>
<p>the smallest <code class="reqn">\rho</code>-value is defined as a fraction of &lsquo;<code>rho.max</code>&rsquo; (i.e., the smallest <code class="reqn">\rho</code>-value for which all the estimated partial correlation coefficients are equal to zero). The default depends on the sample size &lsquo;<code class="reqn">n</code>&rsquo; relative to the number of response variables &lsquo;<code class="reqn">p</code>&rsquo;. If &lsquo;<code class="reqn">p &lt; n</code>&rsquo;, the default is &lsquo;1.0E-6&rsquo; otherwise the value &lsquo;1.0E-2&rsquo; is used as default. A very small value of &lsquo;<code>rho.min.ratio</code>&rsquo; will lead to a saturated fitted model in the &lsquo;<code class="reqn">p &lt; n</code>&rsquo; case.</p>
</td></tr>
<tr><td><code id="cglasso_+3A_rho">rho</code></td>
<td>
<p>an optional user supplied decreasing sequence of <code class="reqn">\rho</code>-values. By default <code>cglasso</code> computes a sequence of <code class="reqn">\rho</code>-values using <code>nrho</code> and <code>rho.min.ratio</code>. If <code>rho</code> is supplied then <code>nrho</code> and <code>rho.min.ratio</code> are overwritten. WARNING: use with care and avoid supplying a single <code class="reqn">\rho</code>-value; supply instead a decreasing sequence.</p>
</td></tr>
<tr><td><code id="cglasso_+3A_maxit.em">maxit.em</code></td>
<td>
<p>maximum number of iterations of the EM algorithm. Default is <code>1.0E+4</code>.</p>
</td></tr>
<tr><td><code id="cglasso_+3A_thr.em">thr.em</code></td>
<td>
<p>threshold for the convergence of the EM algorithm. Default value is <code>1.0E-4</code>.</p>
</td></tr>
<tr><td><code id="cglasso_+3A_maxit.bcd">maxit.bcd</code></td>
<td>
<p>maximum number of iterations of the glasso algorithm. Default is <code>1.0E+5</code>.</p>
</td></tr>
<tr><td><code id="cglasso_+3A_thr.bcd">thr.bcd</code></td>
<td>
<p>threshold for the convergence of the glasso algorithm. Default is <code>1.0E-4</code>.</p>
</td></tr>
<tr><td><code id="cglasso_+3A_trace">trace</code></td>
<td>
<p>integer for printing information out as iterations proceed: <code>trace = 0</code> no information is printed out; <code>trace = 1</code> minimal information is printed on screen; <code>trace = 2</code> detailed information is printed on screen.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>cglasso</code> is the main model-fitting function and can be used to fit a broad range of extensions of the glasso estimator (Friedman <em>and other</em>, 2008). It is specifically proposed to study datasets with censored and/or missing response values. To help the user, the <code>cglasso</code> function has been designed to automatically select the most suitable extension by using the information stored in the &lsquo;<code><a href="#topic+datacggm">datacggm</a></code>&rsquo; object passed through <code>Z</code>. 
</p>
<p>Below we sum up the available extenions:
</p>

<ul>
<li><p> if only left/right-censored are observed in the response matrix <code>Y</code> (without missing values) and no predictor matrix is stored in <code>Z</code>, then <code>cglasso</code> computes the censored glasso estimator (Augugliaro <em>and other</em>, 2020a);
</p>
</li>
<li><p> if only left/right-censored are observed in the response matrix <code>Y</code> (without missing values) and and a predictor matrix <code>X</code> is stored in <code>Z</code>, then <code>cglasso</code> computes the conditional censored glasso estimator (Augugliaro <em>and other</em>, 2020b);
</p>
</li>
<li><p> if only missing values are stored in the response matrix <code>Y</code> (without censored values), then <code>cglasso</code> computes the missglasso estimator (Stadler <em>and other</em>, 2012);
</p>
</li>
<li><p> starting with version 2.0.0, <code>cglasso</code> can also handle datasets with both missing and censored response values.
</p>
</li></ul>

<p>See section &lsquo;Examples&rsquo; for some example.
</p>
<p>The model-fitting function <code>cglasso</code> returns an <span class="rlang"><b>R</b></span> object of S3 class &lsquo;<code>cglasso</code>&rsquo; for which there are available a set of accessor functions, a set of functions designed to evaluate the goodness-of-fit of the fitted models and, finally, a set of functions developed to analyze the selected network. The function <code><a href="#topic+ShowStructure">ShowStructure</a></code> can be used to show the structure of the package.
</p>
<p>The accessor functions <code><a href="#topic+coef.cglasso">coef.cglasso</a></code>, <code><a href="#topic+fitted.cglasso">fitted.cglasso</a></code>, <code><a href="#topic+residuals.cglasso">residuals.cglasso</a></code>, <code><a href="#topic+predict.cglasso">predict.cglasso</a></code> and <code><a href="#topic+impute">impute</a></code> can be used to extract various useful features of the object fitted by <code>cglasso</code>.
</p>
<p>For an <span class="rlang"><b>R</b></span> object returned by <code>cglasso</code>, the functions <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code> and <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code> can be used to evaluate the goodness-of-fit of the fitted models. Usually, these functions are used together with the function <code><a href="#topic+summary.cglasso">summary.cglasso</a></code>, which gives more information about the sequence of fitted models. The plotting function <code><a href="#topic+plot.GoF">plot.GoF</a></code> can be used to graphically identify the optimal pair of the tuning parameters.
</p>
<p>Given a pair of the tuning paremeters, the functions <code><a href="#topic+to_graph">to_graph</a></code> and <code><a href="#topic+plot.cglasso2igraph">plot.cglasso2igraph</a></code> can be used to analyze and show the selected network. Finally, the function <code><a href="#topic+cggm">cggm</a></code> can be used to produce post-hoc maximum likelihood refitting of the selected graphical model.  
</p>


<h3>Value</h3>

<p><code>cglasso</code> returns an object of S3 class &ldquo;<code>cglasso</code>&rdquo;, i.e., a named list containing the
following components:
</p>
<table>
<tr><td><code>call</code></td>
<td>
<p>the call that produced this object.</p>
</td></tr>
<tr><td><code>Yipt</code></td>
<td>
<p>an array of dimension &lsquo;<code>n x p x nlambda x nrho</code>&rsquo; storing the &lsquo;working response matrices&rsquo;, that is, <code>Yipt[, , i, j]</code> is used as response matrix to compute the multilasso estimator (Augugliaro <em>and other</em>, 2020b). The accessor function &lsquo;<code><a href="#topic+impute">impute</a></code>&rsquo; can be used to extract the desidered imputed matrix.</p>
</td></tr>
<tr><td><code>B</code></td>
<td>
<p>an array of dimension &lsquo;<code>(q + 1) x p x nlambda x nrho</code>&rsquo; storing the penalized estimate of the regression coefficient matrices. The accessor function &lsquo;<code><a href="#topic+coef.cglasso">coef.cglasso</a></code>&rsquo; can be used to extract the desired estimates.</p>
</td></tr>
<tr><td><code>mu</code></td>
<td>
<p>an array of dimension &lsquo;<code>n x p x nlambda x nrho</code>&rsquo; storing the fitted values. The accessor function &lsquo;<code><a href="#topic+fitted.cglasso">fitted.cglasso</a></code>&rsquo; can be used to extract the desired fitted values.</p>
</td></tr>
<tr><td><code>R</code></td>
<td>
<p>an array of dimension &lsquo;<code>n x p x nlambda x nrho</code>&rsquo; storing the &lsquo;working residuals&rsquo;, that is, <code>R[, , i, j]</code> is defined as the difference between <code>Yipt[, , i, j]</code> and <code>mu[, , i, j]</code>. The accessor function &lsquo;<code><a href="#topic+residuals.cglasso">residuals.cglasso</a></code>&rsquo; can be used to extract the desidered residual matrix.</p>
</td></tr>
<tr><td><code>S</code></td>
<td>
<p>an array of dimension &lsquo;<code>p x p x nlambda x nrho</code>&rsquo; storing the &lsquo;working empirical covariance matrices&rsquo;, that is, &lsquo;<code>S[, , i, j]</code>&rsquo; is used to compute the glasso estimator.</p>
</td></tr>
<tr><td><code>Sgm</code></td>
<td>
<p>an array of dimension &lsquo;<code>p x p x nlambda x nrho</code>&rsquo; storing the estimated covariance matrices. The accessor function  &lsquo;<code><a href="#topic+coef">coef</a></code>&rsquo; can be used to extract the desired estimates.</p>
</td></tr>
<tr><td><code>Tht</code></td>
<td>
<p>an array of dimension &lsquo;<code>p x p x nlambda x nrho</code>&rsquo; storing the estimated precision matrices. The accessor funtion &lsquo;<code><a href="#topic+coef">coef</a></code>&rsquo; can be used to extract the desired estimates.</p>
</td></tr>
<tr><td><code>dfB</code></td>
<td>
<p>a matrix of dimension &lsquo;<code>(p + 1) x nlambda x nrho</code>&rsquo; storing the number of estimated non-zero regression coefficients. Only for internal purpose.</p>
</td></tr>
<tr><td><code>dfTht</code></td>
<td>
<p>a matrix of dimension &lsquo;<code>nlambda x nrho</code>&rsquo; storing the number of estimated non-zero (off-diagonal) partial correlation coefficients. Only for internal purpose.</p>
</td></tr>
<tr><td><code>InfoStructure</code></td>
<td>
<p>a named list whose elements contain information about the estimated networks. Only for internal purpose.</p>
</td></tr>
<tr><td><code>nit</code></td>
<td>
<p>an array of dimension &lsquo;<code>2 x nlambda x nrho</code>&rsquo; storing the number of EM steps.</p>
</td></tr>
<tr><td><code>Z</code></td>
<td>
<p>the &lsquo;<code>datacggm</code>&rsquo; object used to compute the censored graphical lasso estimator.</p>
</td></tr>
<tr><td><code>diagonal</code></td>
<td>
<p>the flag used to specify if the diagonal entries of the precision matrix are penalized.</p>
</td></tr>
<tr><td><code>weights.B</code></td>
<td>
<p>the matrix of non-negative weights used for the regression coefficients.</p>
</td></tr>
<tr><td><code>weights.Tht</code></td>
<td>
<p>the matrix of non-negative weights used for the precision matrix.</p>
</td></tr>
<tr><td><code>nlambda</code></td>
<td>
<p>the number of <code class="reqn">\lambda</code>-values used.</p>
</td></tr>
<tr><td><code>lambda.min.ratio</code></td>
<td>
<p>the value used to compute the smallest <code class="reqn">\lambda</code>-value.</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>the sequence of <code class="reqn">\lambda</code>-values used to fit the model.</p>
</td></tr>
<tr><td><code>nrho</code></td>
<td>
<p>the number of <code class="reqn">\rho</code>-values used.</p>
</td></tr>
<tr><td><code>rho.min.ratio</code></td>
<td>
<p>the value used to compute the smallest <code class="reqn">\rho</code>-value.</p>
</td></tr>
<tr><td><code>rho</code></td>
<td>
<p>the sequence of <code class="reqn">\rho</code>-values used to fit the model.</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>a description of the fitted model.</p>
</td></tr>
<tr><td><code>maxit.em</code></td>
<td>
<p>maximum number of iterations of the EM algorithm.</p>
</td></tr>
<tr><td><code>thr.em</code></td>
<td>
<p>threshold for the convergence of the EM algorithm.</p>
</td></tr>
<tr><td><code>maxit.bcd</code></td>
<td>
<p>maximum number of iterations of the glasso algorithm.</p>
</td></tr>
<tr><td><code>thr.bcd</code></td>
<td>
<p>threshold for the convergence of the glasso algorithm.</p>
</td></tr>
<tr><td><code>conv</code></td>
<td>
<p>a description of the error that has occurred.</p>
</td></tr>
<tr><td><code>subrout</code></td>
<td>
<p>the name of the Fortran subroutine where the error has occurred (for internal debug only).</p>
</td></tr>
<tr><td><code>trace</code></td>
<td>
<p>the integer used for printing information on screen.</p>
</td></tr>
<tr><td><code>nobs</code></td>
<td>
<p>the sample size</p>
</td></tr>
<tr><td><code>nresp</code></td>
<td>
<p>the number of response variables used to fit the model.</p>
</td></tr>
<tr><td><code>npred</code></td>
<td>
<p>the number of predictors used to fit the model.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>References</h3>

<p>Augugliaro L., Sottile G., Wit E.C., and Vinciotti V. (2023) &lt;<a href="https://doi.org/10.18637/jss.v105.i01">doi:10.18637/jss.v105.i01</a>&gt;.
cglasso: An R Package for Conditional Graphical Lasso Inference with Censored and Missing Values.
<em>Journal of Statistical Software</em> <b>105</b>(1), 1&ndash;58.
</p>
<p>Augugliaro, L., Abbruzzo, A., and Vinciotti, V. (2020a) &lt;<a href="https://doi.org/10.1093/biostatistics/kxy043">doi:10.1093/biostatistics/kxy043</a>&gt;.
<code class="reqn">\ell_1</code>-Penalized censored Gaussian graphical model.
<em>Biostatistics</em> <b>21</b>, e1&ndash;e16.
</p>
<p>Augugliaro, L., Sottile, G., and Vinciotti, V. (2020b) &lt;<a href="https://doi.org/10.1007/s11222-020-09945-7">doi:10.1007/s11222-020-09945-7</a>&gt;.
The conditional censored graphical lasso estimator.
<em>Statistics and Computing</em> <b>30</b>, 1273&ndash;1289.
</p>
<p>Friedman, J.H., Hastie, T., and Tibshirani, R. (2008) &lt;<a href="https://doi.org/10.1093/biostatistics/kxm045">doi:10.1093/biostatistics/kxm045</a>&gt;.
Sparse inverse covariance estimation with the graphical lasso.
<em>Biostatistics</em> <b>9</b>, 432&ndash;441.
</p>
<p>Stadler, N. and Buhlmann, P. (2012) &lt;<a href="https://doi.org/10.1007/s11222-010-9219-7">doi:10.1007/s11222-010-9219-7</a>&gt;.
Missing values: sparse inverse covariance estimation and an extension to sparse regression.
<em>Statistics and Computing</em> <b>22</b>, 219&ndash;235.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+datacggm">datacggm</a></code>, <code><a href="#topic+coef.cglasso">coef.cglasso</a></code>, <code><a href="#topic+fitted.cglasso">fitted.cglasso</a></code>, <code><a href="#topic+residuals.cglasso">residuals.cglasso</a></code>, <code><a href="#topic+predict.cglasso">predict.cglasso</a></code>, <code><a href="#topic+impute">impute</a></code>, <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code>, <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>, <code><a href="#topic+summary.cglasso">summary.cglasso</a></code>, <code><a href="#topic+select_cglasso">select_cglasso</a></code>, <code><a href="#topic+plot.GoF">plot.GoF</a></code>, <code><a href="#topic+to_graph">to_graph</a></code>, <code><a href="#topic+plot.cglasso2igraph">plot.cglasso2igraph</a></code>, <code><a href="#topic+cggm">cggm</a></code> and <code><a href="#topic+ShowStructure">ShowStructure</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)

# Model 1: censored glasso estimator (Augugliaro \emph{and other}, 2020a)
# Y ~ N(0, Sigma) and probability of left/right censored values equal to 0.05
n &lt;- 1000L
p &lt;- 3L
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, Sigma = Sigma, probl = 0.05, probr = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
out

# Model 2: conditional censored glasso estimator (Augugliaro \emph{and other}, 2020b)
# Y ~ N(b0 + XB, Sigma)  and probability of left/right censored values equal to 0.05
n &lt;- 1000L
p &lt;- 3L
q &lt;- 2L
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probl = 0.05, probr = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
out

# Model 3: missglasso estimator (Stadler \emph{and other}, 2012)
# Y ~ N(b0 + XB, Sigma)  and probability of missing-at-random values equal to 0.05
n &lt;- 1000L
p &lt;- 3L
q &lt;- 2L
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probna = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
out

# Model 4: mixed estimator
# Y ~ N(b0 + XB, Sigma)  and
# 1. probability of left/right censored values equal to 0.05
# 2. probability of missing-at-random values equal to 0.05
n &lt;- 1000L
p &lt;- 3L
q &lt;- 2L
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, X = X, b0 = b0, B = B, Sigma = Sigma, probl = 0.05, probr = 0.05,
           probna = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
out
</code></pre>

<hr>
<h2 id='cglasso-internal'>Internal Functions</h2><span id='topic+print.datacggm'></span><span id='topic+as.character.datacggm'></span><span id='topic+fit_matrix'></span><span id='topic+fitMarginalCGM'></span><span id='topic+Math.datacggm'></span><span id='topic+Ops.datacggm'></span><span id='topic+print.cglasso'></span><span id='topic+cglasso.fit'></span><span id='topic+print.QFun'></span><span id='topic+print.GoF'></span><span id='topic+cggm.fit'></span><span id='topic+print.cggm'></span><span id='topic+print.cglasso2igraph'></span><span id='topic+function2xy'></span>

<h3>Description</h3>

<p>Internal cglasso functions.</p>


<h3>Details</h3>

<p>These functions are not intended for users.</p>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>

<hr>
<h2 id='coef'>Extract Model Coefficients</h2><span id='topic+coef'></span><span id='topic+coef.cglasso'></span>

<h3>Description</h3>

<p>The accessor function <code>coef</code> extracts model coefficients from an <span class="rlang"><b>R</b></span> object inheriting class &lsquo;<code>cglasso</code>&rsquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cglasso'
coef(object, type = c("all", "B", "Sigma", "Theta"), lambda.id, rho.id, 
     drop = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coef_+3A_object">object</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object inheriting class &lsquo;<code>cglasso</code>&rsquo;, that is, the output of the model-fitting functions <code><a href="#topic+cglasso">cglasso</a></code> and <code><a href="#topic+cggm">cggm</a></code>.</p>
</td></tr>
<tr><td><code id="coef_+3A_type">type</code></td>
<td>
<p>a description of the desired estimates.</p>
</td></tr>
<tr><td><code id="coef_+3A_lambda.id">lambda.id</code></td>
<td>
<p>an optional vector of integers used to specify the <code class="reqn">\lambda</code>-values.</p>
</td></tr>
<tr><td><code id="coef_+3A_rho.id">rho.id</code></td>
<td>
<p>an optional vector of integers used to specify the <code class="reqn">\rho</code>-values.</p>
</td></tr>
<tr><td><code id="coef_+3A_drop">drop</code></td>
<td>
<p>logical. Dimensions of the required objects can only be dropped if their extent is one.</p>
</td></tr>
<tr><td><code id="coef_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Coefficients extracted from &lsquo;<code>object</code>&rsquo; are returned. By default, a named list storing all the estimated parameters is returned.
</p>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>See Also</h3>

<p>Model-fitting functions <code><a href="#topic+cglasso">cglasso</a></code>, <code><a href="#topic+cggm">cggm</a></code> and the accessor functions <code><a href="#topic+fitted.cglasso">fitted.cglasso</a></code>, <code><a href="#topic+residuals.cglasso">residuals.cglasso</a></code>, <code><a href="#topic+predict.cglasso">predict.cglasso</a></code> and <code><a href="#topic+impute">impute</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
# Y ~ N(0, Sigma) and probability of left/right censored values equal to 0.05
n &lt;- 1000L
p &lt;- 3L
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, Sigma = Sigma, probl = 0.05, probr = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
coef(out, type = "Theta", rho.id = 1:4)
coef(out, type = "Theta", rho.id = 3, drop = TRUE)

# Y ~ N(b0 + XB, Sigma)  and probability of left/right censored values equal to 0.05
n &lt;- 1000L
p &lt;- 3L
q &lt;- 2
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probl = 0.05, probr = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
coef(out, type = "B", lambda.id = 3, rho.id = 1:4)
coef(out, type = "B", lambda.id = 3, rho.id = 3, drop = TRUE)
</code></pre>

<hr>
<h2 id='ColMeans+20+2B+20ColMeans'>Calculate Column Means and Vars of a &ldquo;datacggm&rdquo; Object</h2><span id='topic+ColMeans'></span><span id='topic+ColVars'></span>

<h3>Description</h3>

<p>Retrieve column means and column variances of a &ldquo;datacggm&rdquo; object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ColMeans(x)
ColVars(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ColMeans+2B20+2B2B+2B20ColMeans_+3A_x">x</code></td>
<td>
<p>an object of class &lsquo;<code>datacggm</code>&rsquo;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For an <span class="rlang"><b>R</b></span> object <code>x</code> of class &lsquo;<code>datacggm</code>&rsquo;, <code>ColMeans</code> (<code>ColVars</code>) retrieves the column means (variances) of the matrices obtained by <code>getMatrix(x, "Y")</code> and <code>getMatrix(x, "X")</code>. For the response variables, marginal means and variances are estimated using a EM-algorithm under the assumption that the <code class="reqn">p</code> response variables are marginally normally distributed (see also Details section in <code><a href="#topic+datacggm">datacggm</a></code>). For the numeric predictor variables, marginal means and variances are computed by <code><a href="base.html#topic+mean">mean</a></code> and <code><a href="stats.html#topic+var">var</a></code>, whereas, for categorical data, <code>ColMeans</code> (<code>ColVars</code>) retrieves the statistical mode and the Gini-Simpson Index, respectively.
</p>


<h3>Value</h3>

<p><code>ColMeans</code> (<code>ColVars</code>) returns a named list with the columns means (variances).
</p>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>See Also</h3>

<p><code><a href="#topic+datacggm">datacggm</a></code>, <code><a href="#topic+rcggm">rcggm</a></code>, <code><a href="#topic+qqcnorm">qqcnorm</a></code> and <code><a href="#topic+hist.datacggm">hist.datacggm</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
n &lt;- 1000L
p &lt;- 3L
b0 &lt;- rep(0, p)
Z &lt;- rcggm(n = n, b0 = b0, probl = 0.05, probr = 0.05)
ColMeans(Z)
ColVars(Z)

n &lt;- 1000L
p &lt;- 3L
q &lt;- 2
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, probl = 0.05, probr = 0.05)
ColMeans(Z)
ColVars(Z)
</code></pre>

<hr>
<h2 id='datacggm'>Create a Dataset from a Conditional Gaussian Graphical Model with Censored and/or Missing Values</h2><span id='topic+datacggm'></span>

<h3>Description</h3>

<p>&lsquo;The <code>datacggm</code>&rsquo; function is used to create a dataset from a conditional Gaussian graphical model with censored and/or missing values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>datacggm(Y, lo = -Inf, up =  +Inf, X = NULL, control = list(maxit = 1.0E+4,
         thr = 1.0E-4))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="datacggm_+3A_y">Y</code></td>
<td>
<p>a <code class="reqn">(n\times p)</code>-dimensional matrix; each row is an observation from a conditional Gaussian graphical model with censoring vectors <code>lo</code> and <code>up</code>. Missing-at-random values are recorded as &lsquo;<code>NA</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="datacggm_+3A_lo">lo</code></td>
<td>
<p>the lower censoring vector; <code>lo[j]</code> is used to specify the lower censoring value for the random variable <code class="reqn">Y_j</code>.</p>
</td></tr>
<tr><td><code id="datacggm_+3A_up">up</code></td>
<td>
<p>the upper censoring vector; <code>up[j]</code> is used to specify the upper censoring value for the random variable <code class="reqn">Y_j</code>.</p>
</td></tr>
<tr><td><code id="datacggm_+3A_x">X</code></td>
<td>
<p>an optional <code class="reqn">(n\times q)</code>-dimensional data frame of predictors. If missing (default), a dataset from a Gaussian graphical model is returned otherwise a dataset from a conditional Gaussian graphical model is returned.</p>
</td></tr>
<tr><td><code id="datacggm_+3A_control">control</code></td>
<td>
<p>a named list used to pass the arguments to the EM algorithm (see below for more details). The components are:
</p>

<ul>
<li> <p><code>maxit</code>: maximum number of iterations. Default is <code>1.0E+4</code>.
</p>
</li>
<li> <p><code>thr</code>: threshold for the convergence. Default value is <code>1.0E-4</code>.</p>
</li></ul>

</td></tr>
</table>


<h3>Details</h3>

<p>The function &lsquo;<code>datacggm</code>&rsquo; returns an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>datacggm</code>&rsquo;, that is a named list containing the elements needed to fit a conditional graphical LASSO (cglasso) model to datasets with censored and/or missing values.
</p>
<p>A set of specific method functions are developed to decsribe data with censored/missing values. For example, the method function &lsquo;<code>print.datacggm</code>&rsquo;  prints out the left and right-censored values using the following rules: a right-censored value is labeled adding the symbol &lsquo;<code>+</code>&rsquo; at the end of the value, whereas the symbol &lsquo;<code>-</code>&rsquo; is used for the left-censored values (see examples below). The summary statistics can be obtained using the method function &lsquo;<code><a href="#topic+summary.datacggm">summary.datacggm</a></code>&rsquo;. The matrices <code>Y</code> and <code>X</code> are extracted from a <code>datacggm</code> object using the function &lsquo;<code><a href="#topic+getMatrix">getMatrix</a></code>&rsquo;.
</p>
<p>For each column of the matrix &lsquo;<code>Y</code>&rsquo;, mean and variance are estimated using a standard EM-algorithm based on the assumption of a Gaussian distribution. &lsquo;<code>maxit</code>&rsquo; and &lsquo;<code>thr</code>&rsquo; are used to set the number of iterations and the threshold for convergence, respectively. Marginal means and variances can be extracted using the accessor functions &lsquo;<code><a href="#topic+ColMeans">ColMeans</a></code>&rsquo; and &lsquo;<code><a href="#topic+ColVars">ColVars</a></code>&rsquo;, respectively. Furthermore, the plotting functions &lsquo;<code><a href="#topic+hist.datacggm">hist.datacggm</a></code>&rsquo; and &lsquo;<code><a href="#topic+qqcnorm">qqcnorm</a></code>&rsquo; can be used to inspect the marginal distribution of each column of the matrix &lsquo;<code>Y</code>&rsquo;.
</p>
<p>The status indicator matrix, denoted by <code>R</code>, can be extracted by using the function <code><a href="#topic+event">event</a></code>. The entries of this matrix specify the status of an observation using the following code:
</p>

<ul>
<li> <p>&lsquo;<code>R[i, j] =  0</code>&rsquo; means that the <code class="reqn">y_{ij}</code> is inside the open interval <code>(lo[j], up[j])</code>;
</p>
</li>
<li> <p>&lsquo;<code>R[i, j] = -1</code>&rsquo; means that the <code class="reqn">y_{ij}</code> is a left-censored value;
</p>
</li>
<li> <p>&lsquo;<code>R[i, j] = +1</code>&rsquo; means that the <code class="reqn">y_{ij}</code> is a right-censored value;
</p>
</li>
<li> <p>&lsquo;<code>R[i, j] = +9</code>&rsquo; means that the <code class="reqn">y_{ij}</code> is a missing value.
</p>
</li></ul>

<p>See below for the other functions related to an object of class &lsquo;<code>datacggm</code>&rsquo;.
</p>


<h3>Value</h3>

<p>&lsquo;<code>datacggm</code>&rsquo; returns an <span class="rlang"><b>R</b></span> object of S3 class &ldquo;<code>datacggm</code>&rdquo;, that is, a nested named list containing the
following components:
</p>
<table>
<tr><td><code>Y</code></td>
<td>
<p>the <code class="reqn">(n\times p)</code>-dimensional matrix <code>Y</code>.</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>the <code class="reqn">(n\times q)</code>-dimensional data frame <code>X</code>.</p>
</td></tr>
<tr><td><code>Info</code></td>
<td>


<ul>
<li> <p><code>lo:</code> the lower censoring vector;
</p>
</li>
<li> <p><code>up:</code> the upper censoring vector;
</p>
</li>
<li> <p><code>R:</code> the status indicator matrix encoding the censored/missing values (mainly for internal purposes);
</p>
</li>
<li> <p><code>order:</code> an integer vector used for the ordering of the matrices <code>Y</code> and <code>X</code> (for internal purposes only);
</p>
</li>
<li> <p><code>Pattern:</code> a matrix encoding the information about the the patterns of censored/missing values (for internal purposes only);
</p>
</li>
<li> <p><code>ym:</code> the estimated marginal means of the random variables <code class="reqn">Y_j</code>;
</p>
</li>
<li> <p><code>yv:</code> the estimated marginal variances of the random variables <code class="reqn">Y_j</code>;
</p>
</li>
<li> <p><code>n:</code> the sample size;
</p>
</li>
<li> <p><code>p:</code> the number of response variables;
</p>
</li>
<li> <p><code>q:</code> the number of columns of the data frame <code>X</code>.
</p>
</li></ul>

</td></tr>
</table>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>References</h3>

<p>Augugliaro L., Sottile G., Wit E.C., and Vinciotti V. (2023) &lt;<a href="https://doi.org/10.18637/jss.v105.i01">doi:10.18637/jss.v105.i01</a>&gt;.
cglasso: An R Package for Conditional Graphical Lasso Inference with Censored and Missing Values.
<em>Journal of Statistical Software</em> <b>105</b>(1), 1&ndash;58.
</p>
<p>Augugliaro, L., Sottile, G., and Vinciotti, V. (2020a) &lt;<a href="https://doi.org/10.1007/s11222-020-09945-7">doi:10.1007/s11222-020-09945-7</a>&gt;.
The conditional censored graphical lasso estimator.
<em>Statistics and Computing</em> <b>30</b>, 1273&ndash;1289.
</p>
<p>Augugliaro, L., Abbruzzo, A., and Vinciotti, V. (2020b) &lt;<a href="https://doi.org/10.1093/biostatistics/kxy043">doi:10.1093/biostatistics/kxy043</a>&gt;.
<code class="reqn">\ell_1</code>-Penalized censored Gaussian graphical model.
<em>Biostatistics</em> <b>21</b>, e1&ndash;e16.
</p>


<h3>See Also</h3>

<p>Related to the <span class="rlang"><b>R</b></span> objects of class &ldquo;<code>datacggm</code>&rdquo; there are the accessor functions, <code><a href="#topic+rowNames">rowNames</a></code>, <code><a href="#topic+colNames">colNames</a></code>, <code><a href="#topic+getMatrix">getMatrix</a></code>, <code><a href="#topic+ColMeans">ColMeans</a></code>, <code><a href="#topic+ColVars">ColVars</a></code>, <code><a href="#topic+upper">upper</a></code>, <code><a href="#topic+lower">lower</a></code>, <code><a href="#topic+event">event</a></code>, <code><a href="#topic+qqcnorm">qqcnorm</a></code> and the method functions <code><a href="#topic+is.datacggm">is.datacggm</a></code>, <code><a href="#topic+dim.datacggm">dim.datacggm</a></code>, <code><a href="#topic+summary.datacggm">summary.datacggm</a></code> and <code><a href="#topic+hist.datacggm">hist.datacggm</a></code>. The function <code><a href="#topic+rcggm">rcggm</a></code> can be used to simulate a dataset from a conditional Gaussian graphical model whereas <code><a href="#topic+cglasso">cglasso</a></code> is the model fitting function devoted to the l1-penalized censored Gaussian graphical model.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)

# a dataset from a right-censored Gaussian graphical model
n &lt;- 100L
p &lt;- 3L
Y &lt;- matrix(rnorm(n * p), n, p)
up &lt;- 1
Y[Y &gt;= up] &lt;- up
Z &lt;- datacggm(Y = Y, up = up)
Z

# a dataset from a  conditional censored Gaussian graphical model
n &lt;- 100L
p &lt;- 3L
q &lt;- 2
Y &lt;- matrix(rnorm(n * p), n, p)
up &lt;- 1
lo &lt;- -1
Y[Y &gt;= up] &lt;- up
Y[Y &lt;= lo] &lt;- lo
X &lt;- matrix(rnorm(n * q), n, q)
Z &lt;- datacggm(Y = Y, lo = lo, up = up, X = X)
Z

# a dataset from a  conditional censored Gaussian graphical model 
# and with missing-at-random values
n &lt;- 100L
p &lt;- 3L
q &lt;- 2
Y &lt;- matrix(rnorm(n * p), n, p)
NA.id &lt;- matrix(rbinom(n * p, 1L, 0.01), n, p)
Y[NA.id == 1L] &lt;- NA
up &lt;- 1
lo &lt;- -1
Y[Y &gt;= up] &lt;- up
Y[Y &lt;= lo] &lt;- lo
X &lt;- matrix(rnorm(n * q), n, q)
Z &lt;- datacggm(Y = Y, lo = lo, up = up, X = X)
Z
</code></pre>

<hr>
<h2 id='dim.datacggm'>Dimensions of a &ldquo;datacggm&rdquo; Object</h2><span id='topic+dim'></span><span id='topic+dim.datacggm'></span>

<h3>Description</h3>

<p>Retrieve the dimension of an <span class="rlang"><b>R</b></span> object of class &ldquo;datacggm&rdquo;, that is, the sample size (<code class="reqn">n</code>), the number of response variables (<code class="reqn">p</code>) and the number of predictors (<code class="reqn">q</code>). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'datacggm'
dim(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dim.datacggm_+3A_x">x</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>datacggm</code>&rsquo;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>datacggm</code>&rsquo;, <code>dim</code> retrieves a list with elements named <code>Y</code> and <code>X</code>. The first component is the &ldquo;<code>dim</code>&rdquo; attribute of the matrix <code>Y</code> whereas the second component is the &ldquo;<code>dim</code>&rdquo; attribute of the matrix <code>X</code>. If <code>X</code> is missing, the second component is <code>NULL</code>.
</p>


<h3>Value</h3>

<p>For an object of class &lsquo;<code>datacggm</code>&rsquo;, <code>dim</code> retrieves a named list with components <code>Y</code> and <code>X</code> which are the &ldquo;<code>dim</code>&rdquo; attribute of the two matrices, respectively. If <code>X</code> is missing then the last component is <code>NULL</code>.
</p>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>See Also</h3>

<p><code><a href="#topic+datacggm">datacggm</a></code>, <code><a href="#topic+rcggm">rcggm</a></code>,<code><a href="#topic+nobs">nobs</a></code>, <code><a href="#topic+nresp">nresp</a></code> and <code><a href="#topic+npred">npred</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
# a dataset from a censored Gaussian graphical model
n &lt;- 100L
p &lt;- 3L
b0 &lt;- rep(0, p)
Z &lt;- rcggm(n = n, b0 = b0, probl = 0.05, probr = 0.05)
dim(Z)

# a dataset from a  conditional censored Gaussian graphical model
n &lt;- 100L
p &lt;- 3L
q &lt;- 2
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, probl = 0.05, probr = 0.05)
dim(Z)
</code></pre>

<hr>
<h2 id='dimnames.datacggm'>Dimnames of a &ldquo;datacggm&rdquo; Object</h2><span id='topic+dimnames'></span><span id='topic+dimnames.datacggm'></span><span id='topic+dimnames+3C-.datacggm'></span>

<h3>Description</h3>

<p>Retrieve or set the dimnames of a &ldquo;datacggm&rdquo; object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'datacggm'
dimnames(x)

## S3 replacement method for class 'datacggm'
dimnames(x) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dimnames.datacggm_+3A_x">x</code></td>
<td>
<p>an object of class &lsquo;<code>datacggm</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="dimnames.datacggm_+3A_value">value</code></td>
<td>
<p>a nested list with names &lsquo;<code>X</code>&rsquo; and &lsquo;<code>Y</code>&rsquo;. See below for mode details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For an <span class="rlang"><b>R</b></span> object &lsquo;<code>x</code>&rsquo; of class &lsquo;<code>datacggm</code>&rsquo;, <code>dimnames</code> retrieves or sets the <code>dimnames</code> attribute for the matrices <code>Y</code> and <code>X</code> (see &lsquo;<code><a href="#topic+datacggm">datacggm</a></code>&rsquo; for more details). When setting the <code>dimnames</code> attribute,  <code>value$Y</code> can be <code>NULL</code> (which is not stored) or a list of length two. In the last case, <code>value$Y</code> is passed to the method function  <code><a href="base.html#topic+dimnames">dimnames</a></code> for setting the <code>dimnames</code> attributes of the matrix retrieved by <code>getMatrix(x, "Y")</code>. In the same way, <code>value$X</code> can be used for setting the <code>dimnames</code> attributes of the matrix retrieved by <code>getMatrix(x, "X")</code>.
</p>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>See Also</h3>

<p><code><a href="#topic+datacggm">datacggm</a></code>, <code><a href="#topic+rcggm">rcggm</a></code>, <code><a href="#topic+getMatrix">getMatrix</a></code>, <code><a href="#topic+rowNames">rowNames</a></code> and <code><a href="#topic+colNames">colNames</a></code>. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
# a dataset from a censored Gaussian graphical model
n &lt;- 100L
p &lt;- 3L
b0 &lt;- rep(0, p)
Z &lt;- rcggm(n = n, b0 = b0, probl = 0.05, probr = 0.05)
dimnames(Z)

dimnames(Z) &lt;- list(Y = list(paste0("i", seq_len(n)), paste("Y", 1:p, sep = ":")))
dimnames(Z)

# the same as
# dimnames(Z)$Y &lt;- list(paste0("i", seq_len(n)), paste("Y", 1:p, sep = ":"))
# dimnames(Z)


# a dataset from a  conditional censored Gaussian graphical model
n &lt;- 100L
p &lt;- 3L
q &lt;- 2
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, probl = 0.05, probr = 0.05)
dimnames(Z)

dimnames(Z) &lt;- list(Y = list(NULL, paste("Y", 1:p, sep = ":")), 
                    X = list(NULL, paste("X", 1:q, sep = ":")))
dimnames(Z)

# the same as
# dimnames(Z)$Y &lt;- list(NULL, paste("Y", 1:p, sep = ":"))
# dimnames(Z)$X &lt;- list(NULL, paste("X", 1:q, sep = ":"))
# dimnames(Z)
</code></pre>

<hr>
<h2 id='event'>Status Indicator Matrix from a &lsquo;<code>datacggm</code>&rsquo; Object</h2><span id='topic+event'></span>

<h3>Description</h3>

<p>The &lsquo;<code>event</code>&rsquo; function retrieves the status indicator matrix from an object of class &lsquo;<code>datacggm</code>&rsquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>event(x, ordered = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="event_+3A_x">x</code></td>
<td>
<p>an object of class &lsquo;<code>datacggm</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="event_+3A_ordered">ordered</code></td>
<td>
<p>logical value used to specify if the rows of the status indicator matrix should be ordered according to the patterns of censored/missing values. Default <code>ordered = FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The &lsquo;<code>event</code>&rsquo; function is used to retrieve the status indicator matrix, denoted by <code>R</code>, from an object of class &lsquo;<code>datacggm</code>&rsquo;. The entries of the matrix are used to specify the status of the response variable:
</p>

<ul>
<li> <p>&lsquo;<code>R[i, j] =  0</code>&rsquo; means that <code class="reqn">y_{ij}</code> is inside the open interval <code>(lo[j], up[j])</code>;
</p>
</li>
<li> <p>&lsquo;<code>R[i, j] = -1</code>&rsquo; means that <code class="reqn">y_{ij}</code> is a left-censored value;
</p>
</li>
<li> <p>&lsquo;<code>R[i, j] = +1</code>&rsquo; means that <code class="reqn">y_{ij}</code> is a right-censored value;
</p>
</li>
<li> <p>&lsquo;<code>R[i, j] = +9</code>&rsquo; means that <code class="reqn">y_{ij}</code> is a missing value.
</p>
</li></ul>



<h3>Value</h3>

<p><code>event</code> returns a <code class="reqn">(n\times p)</code>-dimensional matrix.
</p>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>References</h3>

<p>Augugliaro L., Sottile G., Wit E.C., and Vinciotti V. (2023) &lt;<a href="https://doi.org/10.18637/jss.v105.i01">doi:10.18637/jss.v105.i01</a>&gt;.
cglasso: An R Package for Conditional Graphical Lasso Inference with Censored and Missing Values.
<em>Journal of Statistical Software</em> <b>105</b>(1), 1&ndash;58.
</p>
<p>Augugliaro, L., Sottile, G., and Vinciotti, V. (2020) &lt;<a href="https://doi.org/10.1007/s11222-020-09945-7">doi:10.1007/s11222-020-09945-7</a>&gt;.
The conditional censored graphical lasso estimator.
<em>Statistics and Computing</em> <b>30</b>, 1273&ndash;1289.
</p>
<p>Augugliaro, L., Abbruzzo, A., and Vinciotti, V. (2020) &lt;<a href="https://doi.org/10.1093/biostatistics/kxy043">doi:10.1093/biostatistics/kxy043</a>&gt;.
<code class="reqn">\ell_1</code>-Penalized censored Gaussian graphical model.
<em>Biostatistics</em> <b>21</b>, e1&ndash;e16.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+datacggm">datacggm</a></code> and <code><a href="#topic+rcggm">rcggm</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)

# Y ~ N(b0 + XB, Sigma) and
# 1. probability of left/right censored values equal to 0.05
# 2. probability of missing-at-random euqals to 0.05
n &lt;- 100L
p &lt;- 3L
q &lt;- 2
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probl = 0.05, 
           probr = 0.05, probna = 0.05)

# status indicator matrix
event(Z)

# in this case the status indicator matrix is returned with
# rows ordered according to the patterns of missing data
event(Z, ordered = TRUE)
</code></pre>

<hr>
<h2 id='Example'>
Simulated data for the cglasso vignette
</h2><span id='topic+Example'></span><span id='topic+MKMEP.Sim'></span><span id='topic+MM.Sim'></span>

<h3>Description</h3>

<p>Simulated data, used to demonstrate the features of datacggm class. It contains two simulated datasets that mimic the structural characteristics of the two real datasets MKMEP and MM. 
</p>

<ol>
<li> <p><code>MKMEP.Sim</code>: is a reponse matrix containing 50 samples of 10 genes;
</p>
</li>
<li> <p><code>MM.Sim</code>: is a list containing two components;
</p>

<ul>
<li> <p><code>Y</code>: is the response matrix containing 50 measurements of 10 miRNAs
</p>
</li>
<li> <p><code>X</code>: is the predictor matrix containing 50 measurements of 5 variables (four numerical and one categorical)
</p>
</li></ul>

</li></ol>

<p>In both datasets, the response matrices are right-censored with an upper limit of detection fixed to 40.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Example")</code></pre>


<h3>See Also</h3>

<p><code><a href="#topic+cglasso">cglasso</a></code>, <code><a href="#topic+to_graph">to_graph</a></code>, and the method functions <code><a href="base.html#topic+summary">summary</a></code>, <code><a href="#topic+coef">coef</a></code>, <code><a href="base.html#topic+plot">plot</a></code>, <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code>, <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>, <code><a href="#topic+MKMEP">MKMEP</a></code> and <code><a href="#topic+MM">MM</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("Example")

MM.Sim &lt;- datacggm(Y = MM.Sim$Y, up = 40, X = MM.Sim$X)
ColMeans(MM.Sim)
ColVars(MM.Sim)
summary(MM.Sim)

MKMEP.Sim &lt;- datacggm(Y = MKMEP.Sim, up = 40)
ColMeans(MKMEP.Sim)
ColVars(MKMEP.Sim)
summary(MKMEP.Sim)
</code></pre>

<hr>
<h2 id='fitted'>Extract Model Fitted Values</h2><span id='topic+fitted'></span><span id='topic+fitted.cglasso'></span>

<h3>Description</h3>

<p>The accessor function <code>fitted</code> extracts model fitted values from an <span class="rlang"><b>R</b></span> object inheriting class &lsquo;<code>cglasso</code>&rsquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cglasso'
fitted(object, lambda.id, rho.id, drop = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fitted_+3A_object">object</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object inheriting class &lsquo;<code>cglasso</code>&rsquo;, that is, the output of the model-fitting functions <code><a href="#topic+cglasso">cglasso</a></code> or <code><a href="#topic+cggm">cggm</a></code>.</p>
</td></tr>
<tr><td><code id="fitted_+3A_lambda.id">lambda.id</code></td>
<td>
<p>a vector of integers used to specify the <code class="reqn">\lambda</code>-values.</p>
</td></tr>
<tr><td><code id="fitted_+3A_rho.id">rho.id</code></td>
<td>
<p>a vector of integers used to specify the <code class="reqn">\rho</code>-values.</p>
</td></tr>
<tr><td><code id="fitted_+3A_drop">drop</code></td>
<td>
<p>logical. Dimensions can only be dropped if their extent is one.</p>
</td></tr>
<tr><td><code id="fitted_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Fitted values extracted from &lsquo;<code>object</code>&rsquo; are returned.
</p>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>See Also</h3>

<p>Model-fitting functions <code><a href="#topic+cglasso">cglasso</a></code>, <code><a href="#topic+cggm">cggm</a></code> and the accessor functions <code><a href="#topic+coef.cglasso">coef.cglasso</a></code>, <code><a href="#topic+residuals.cglasso">residuals.cglasso</a></code>, <code><a href="#topic+predict.cglasso">predict.cglasso</a></code> and <code><a href="#topic+impute">impute</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
# Y ~ N(0, Sigma) and probability of left/right censored values equal to 0.05
n &lt;- 1000L
p &lt;- 3L
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, Sigma = Sigma, probl = 0.05, probr = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
fitted(out, rho.id = 3L, drop = TRUE)

# Y ~ N(b0 + XB, Sigma)  and probability of left/right censored values equal to 0.05
n &lt;- 1000L
p &lt;- 3L
q &lt;- 2
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probl = 0.05, probr = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
fitted(out, lambda.id = 3L, rho.id = 3L, drop = TRUE)
</code></pre>

<hr>
<h2 id='getGraph'>Retrieve Graphs from a &lsquo;<code>cglasso2igraph</code>&rsquo; Object</h2><span id='topic+getGraph'></span>

<h3>Description</h3>

<p>&lsquo;<code>getGraph</code>&rsquo; retrieves graphs from an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>cglasso2igraph</code>&rsquo;. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGraph(x, type = c("Gyy", "Gxy", "both"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGraph_+3A_x">x</code></td>
<td>
<p>an object of class &lsquo;<code>cglasso2igraph</code>&rsquo; (see also <code><a href="#topic+to_graph">to_graph</a></code>).</p>
</td></tr>
<tr><td><code id="getGraph_+3A_type">type</code></td>
<td>
<p>a description of the required graph. Default is &lsquo;<code>Gyy</code>&rsquo;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>getGraph</code> retrieves an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>igraph</code>&rsquo; representing the graph required by the argument <code>type.</code>
</p>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>See Also</h3>

<p><code><a href="#topic+to_graph">to_graph</a></code>, <code><a href="#topic+is.cglasso2igraph">is.cglasso2igraph</a></code> and <code><a href="#topic+plot.cglasso2igraph">plot.cglasso2igraph</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
# Y ~ N(0, Sigma) and probability of left/right censored values equal to 0.05
n &lt;- 100L
p &lt;- 3L
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, Sigma = Sigma, probl = 0.05, probr = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
out.graph &lt;- to_graph(out)
getGraph(out.graph)


# Y ~ N(b0 + XB, Sigma)  and probability of left/right censored values equal to 0.05
n &lt;- 100L
p &lt;- 3L
q &lt;- 2L
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probl = 0.05, probr = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
out.graph &lt;- to_graph(out, lambda.id = 3, rho.id = 3, weighted = TRUE)
getGraph(out.graph, type = "Gyy")
getGraph(out.graph, type = "Gxy")
getGraph(out.graph, type = "both")
</code></pre>

<hr>
<h2 id='getMatrix'>Retrieve Matrices &lsquo;<code>Y</code>&rsquo; and &lsquo;<code>X</code>&rsquo; from a &lsquo;<code>datacggm</code>&rsquo; Object</h2><span id='topic+getMatrix'></span>

<h3>Description</h3>

<p>&lsquo;<code>getMatrix</code>&rsquo; retrieves matrices &lsquo;<code>Y</code>&rsquo; and/or &lsquo;<code>X</code>&rsquo; from an object of class &lsquo;<code>datacggm</code>&rsquo;. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMatrix(x, name = c("Y", "X", "both"), ordered = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getMatrix_+3A_x">x</code></td>
<td>
<p>an object of class &lsquo;<code><a href="#topic+datacggm">datacggm</a></code>&rsquo;.</p>
</td></tr>
<tr><td><code id="getMatrix_+3A_name">name</code></td>
<td>
<p>the name of the required matrix.</p>
</td></tr>
<tr><td><code id="getMatrix_+3A_ordered">ordered</code></td>
<td>
<p>logical value used to specify if the required matrix should be retrieved with rows ordered according to the patterns of censored values. See below for some example.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>getMatrix</code> retrieves the matrix specified by &lsquo;<code>name</code>&rsquo; and with row ordering specified by &lsquo;<code>ordered</code>&rsquo;. A named list returned if <code>name</code> is <code>"both"</code>.
</p>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>References</h3>

<p>Augugliaro L., Sottile G., Wit E.C., and Vinciotti V. (2023) &lt;<a href="https://doi.org/10.18637/jss.v105.i01">doi:10.18637/jss.v105.i01</a>&gt;.
cglasso: An R Package for Conditional Graphical Lasso Inference with Censored and Missing Values.
<em>Journal of Statistical Software</em> <b>105</b>(1), 1&ndash;58.
</p>
<p>Augugliaro, L., Sottile, G., and Vinciotti, V. (2020) &lt;<a href="https://doi.org/10.1007/s11222-020-09945-7">doi:10.1007/s11222-020-09945-7</a>&gt;.
The conditional censored graphical lasso estimator.
<em>Statistics and Computing</em> <b>30</b>, 1273&ndash;1289.
</p>
<p>Augugliaro, L., Abbruzzo, A., and Vinciotti, V. (2020) &lt;<a href="https://doi.org/10.1093/biostatistics/kxy043">doi:10.1093/biostatistics/kxy043</a>&gt;.
<code class="reqn">\ell_1</code>-Penalized censored Gaussian graphical model.
<em>Biostatistics</em> <b>21</b>, e1&ndash;e16.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+datacggm">datacggm</a></code>, <code><a href="#topic+rcggm">rcggm</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)

# a dataset from a  conditional censored Gaussian graphical model
n &lt;- 100L
p &lt;- 3L
q &lt;- 2
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, probl = 0.05, probr = 0.05)
getMatrix(Z, name = "Y")

# in the following examples 'Y' and 'X' is returned with rows ordered 
# according to the  patterns of censored data
getMatrix(Z, name = "Y", ordered = TRUE)

getMatrix(Z, name = "X")
getMatrix(Z, name = "X", ordered = TRUE)

getMatrix(Z, name = "both")
getMatrix(Z, name = "both", ordered = TRUE)
</code></pre>

<hr>
<h2 id='hist.datacggm'>Histogram for a <code>datacggm</code> Object</h2><span id='topic+hist'></span><span id='topic+hist.datacggm'></span>

<h3>Description</h3>

<p>Creates histograms with a normal density as background and two segments corresponding to left and/or right censored values (if any) in a conditional censored Gaussian graphical model. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'datacggm'
hist(x, breaks = "Sturges", include.lowest = TRUE, right = TRUE, nclass = NULL,
     which, max.hist = 1L, save.hist = FALSE, grdev = pdf, grdev.arg, 
     polygon.col = adjustcolor("grey", alpha.f = 0.25), polygon.border = NA, 
     segments.lwd = 4L, segments.lty = 2L, segments.col = "gray40", 
     points.pch = c(4L, 1L), points.cex = 1.8, points.col = rep("black", 2L), 
     legend = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hist.datacggm_+3A_x">x</code></td>
<td>
<p>an object of class &lsquo;<code>datacggm</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="hist.datacggm_+3A_breaks">breaks</code></td>
<td>
<p>one of:
</p>

<ul>
<li><p>a vector giving the breakpoints between histogram cells,
</p>
</li>
<li><p>a function to compute the vector of breakpoints,
</p>
</li>
<li><p>a single number giving the number of cells for the histogram,
</p>
</li>
<li><p>a character string naming an algorithm to compute the number of cells,
</p>
</li>
<li><p>a function to compute the number of cells.
</p>
</li></ul>

<p>See <code><a href="#topic+hist">hist</a></code> for more details.
</p>
</td></tr>
<tr><td><code id="hist.datacggm_+3A_include.lowest">include.lowest</code></td>
<td>
<p>logical; if <code>TRUE</code>, an <code>x[i]</code> equal to the breaks value will be included in the first (or last, for <code>right = FALSE</code>) bar. This will be ignored (with a warning) unless <code>breaks</code> is a vector.</p>
</td></tr>
<tr><td><code id="hist.datacggm_+3A_right">right</code></td>
<td>
<p>logical; if <code>TRUE</code>, the histogram cells are right-closed (left open) intervals.</p>
</td></tr>
<tr><td><code id="hist.datacggm_+3A_nclass">nclass</code></td>
<td>
<p>numeric (integer). For S(-PLUS) compatibility only, <code>nclass</code> is equivalent to <code>breaks</code> for a scalar or character argument.</p>
</td></tr>
<tr><td><code id="hist.datacggm_+3A_which">which</code></td>
<td>
<p>a vector of integers used to specify the response variables for which the histogram is required.</p>
</td></tr>
<tr><td><code id="hist.datacggm_+3A_max.hist">max.hist</code></td>
<td>
<p>the maximum number of histograms drawn in a single figure.</p>
</td></tr>
<tr><td><code id="hist.datacggm_+3A_save.hist">save.hist</code></td>
<td>
<p>a logical variable or a string specifying the path of the directory where plots will be saved. Letting &lsquo;<code>save.hist = TRUE</code>&rsquo;, the required plots will be saved as external files inside the current working directory. User can save these files on a specific working directory passing the absolute path through the argument <code>save.hist</code>. On exit, working directory will be always set the previous one.</p>
</td></tr>
<tr><td><code id="hist.datacggm_+3A_grdev">grdev</code></td>
<td>
<p>the graphics device used to save the required histograms on external files. See &lsquo;<code><a href="grDevices.html#topic+device">device</a></code>&rsquo; for more details.</p>
</td></tr>
<tr><td><code id="hist.datacggm_+3A_grdev.arg">grdev.arg</code></td>
<td>
<p>additional parameters passed to the graphics device specified by &lsquo;<code>grdev</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="hist.datacggm_+3A_polygon.col">polygon.col</code></td>
<td>
<p>graphical parameter; the colour for filling the area underlying the Gaussian distribution.</p>
</td></tr>
<tr><td><code id="hist.datacggm_+3A_polygon.border">polygon.border</code></td>
<td>
<p>graphical parameter; the colour of the border of the Gaussian distribution.</p>
</td></tr>
<tr><td><code id="hist.datacggm_+3A_segments.lwd">segments.lwd</code></td>
<td>
<p>graphical parameter; the line width used in plotting the segments associated to the symbols specified by &lsquo;<code>points.pch</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="hist.datacggm_+3A_segments.lty">segments.lty</code></td>
<td>
<p>graphical parameter; the line type used in plotting the segments associated to the symbols specified by &lsquo;<code>points.pch</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="hist.datacggm_+3A_segments.col">segments.col</code></td>
<td>
<p>graphical parameter; the colour used in plotting the segments associated to the symbols specified by &lsquo;<code>points.pch</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="hist.datacggm_+3A_points.pch">points.pch</code></td>
<td>
<p>graphical parameter; a pair of integers specifying the symbols used to plot the proportion of censored response variables (default &lsquo;<code>points.pch = 4L</code>&rsquo;) and the probability to observe a censored value (default &lsquo;<code>points.pch = 1L</code>&rsquo;), respectively.</p>
</td></tr>
<tr><td><code id="hist.datacggm_+3A_points.cex">points.cex</code></td>
<td>
<p>graphical parameter; a numerical value giving the amount by which the symbols, specified by &lsquo;<code>points.pch</code>&rsquo;, are magnified relative to the default.</p>
</td></tr>
<tr><td><code id="hist.datacggm_+3A_points.col">points.col</code></td>
<td>
<p>graphical parameter; the colours used to plot the symbols specified by &lsquo;<code>points.pch</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="hist.datacggm_+3A_legend">legend</code></td>
<td>
<p>logical; if &lsquo;<code>TRUE</code>&rsquo; legend is added to the plot.</p>
</td></tr>
<tr><td><code id="hist.datacggm_+3A_...">...</code></td>
<td>
<p>additional graphical parameters passed to &lsquo;<code><a href="graphics.html#topic+plot.histogram">plot.histogram</a></code>&rsquo;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each response variable, the method function &lsquo;<code>hist.datacggm</code>&rsquo; plots a histogram using only the observed values. Densities, plotted on the <code>y</code>-axis, are computed in such a way that the sum of all densities plus the proportion of censored values is equal to one. 
</p>
<p>To evaluate the distributional assumption underlying the censored Gaussian distribution, on each histogram the area underlying the Gaussian density function is also shown in the background (marginal parameters are estimated as described in &lsquo;<code><a href="#topic+datacggm">datacggm</a></code>&rsquo; and &lsquo;<code><a href="#topic+ColMeans">ColMeans</a></code>&rsquo;). Furthermore, on each plot, the proportions of left/right censored values are graphically compared with the corresponding Gaussian tail probabilities. If the assumption about the censoring mechanism is satisfied, then the proportion of censored values and the tail probability are approximately equals to each other.
</p>


<h3>Author(s)</h3>

<p>Gianluca Sottile (<a href="mailto:gianluca.sottile@unipa.it">gianluca.sottile@unipa.it</a>)</p>


<h3>See Also</h3>

<p><code><a href="#topic+datacggm">datacggm</a></code>, <code><a href="#topic+rcggm">rcggm</a></code>, <code><a href="#topic+ColMeans">ColMeans</a></code>, <code><a href="#topic+ColVars">ColVars</a></code> and <code><a href="#topic+qqcnorm">qqcnorm</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)

# a dataset from a right-censored Gaussian graphical model
n &lt;- 1000L
p &lt;- 10L
Y &lt;- matrix(rnorm(n * p), n, p)
up &lt;- 1
Y[Y &gt;= up] &lt;- up
Z &lt;- datacggm(Y = Y, up = up)
hist(Z, max.hist = 4L)
</code></pre>

<hr>
<h2 id='impute'>Imputation of Missing and Censored Values</h2><span id='topic+impute'></span>

<h3>Description</h3>

<p>Imputes multivariate missing and censored values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>impute(object, type = c("mar", "censored", "both"), lambda.new, rho.new)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="impute_+3A_object">object</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object inheriting class &lsquo;<code>cglasso</code>&rsquo;, that is, the output of the model-fitting functions <code><a href="#topic+cglasso">cglasso</a></code> and <code><a href="#topic+cggm">cggm</a></code>.</p>
</td></tr>
<tr><td><code id="impute_+3A_type">type</code></td>
<td>
<p>a description of the imputation required (see section &lsquo;Description&rsquo;). By default only the missing-at-random values are imputed.</p>
</td></tr>
<tr><td><code id="impute_+3A_lambda.new">lambda.new</code></td>
<td>
<p>value of the tuning parameter <code class="reqn">\lambda</code> at which the imputations are required.</p>
</td></tr>
<tr><td><code id="impute_+3A_rho.new">rho.new</code></td>
<td>
<p>value of the tuning parameter <code class="reqn">\rho</code> at which the imputations are required.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>impute</code> function returns the response matrix with the imputed missing-at-random values (&lsquo;<code>type = "mar"</code>&rsquo;), the imputed censored values (&lsquo;<code>type = "censored"</code>&rsquo;), or both (&lsquo;<code>type = "both"</code>&rsquo;).
</p>
<p>If &lsquo;<code>type = "mar"</code>&rsquo; then, for each row, the missing response values are replaced with the expected values of a multivariate normal distribution conditioned on the observed response values.
</p>
<p>If &lsquo;<code>type = "censored"</code>&rsquo; then, for each row, the censored response values are imputed using the expected values of a multivariate truncated normal distribution conditioned on the observed response values.
</p>
<p>If &lsquo;<code>type = "both"</code>&rsquo; then missing-at-random and censored values are imputed. In this case the returned matrix corresponds to the working response matrix computed during the E-Step.
</p>


<h3>Value</h3>

<p>The <code>impute</code> function returns a response matrix with the required imputed data.
</p>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>See Also</h3>

<p>Model-fitting functions <code><a href="#topic+cglasso">cglasso</a></code>, <code><a href="#topic+cggm">cggm</a></code> and the accessor functions <code><a href="#topic+coef.cglasso">coef.cglasso</a></code>, <code><a href="#topic+fitted.cglasso">fitted.cglasso</a></code>, <code><a href="#topic+residuals.cglasso">residuals.cglasso</a></code> and <code><a href="#topic+predict.cglasso">predict.cglasso</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
# Y ~ N(0, Sigma) and 
# 1. probability of left/right censored values equal to 0.05
# 2. probability of missing-at-random values equal to 0.05
n &lt;- 100L
p &lt;- 3L
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, Sigma = Sigma, probl = 0.05, probr = 0.05, probna = 0.05)

out &lt;- cglasso(. ~ ., data = Z)
rho.new &lt;- mean(out$rho)

# imputing missing values
Y.impute &lt;- impute(out, type = "mar", rho.new = rho.new)

# imputing censored values
Y.impute &lt;- impute(out, type = "censored", rho.new = rho.new)

# imputing missing and censored values
Y.impute &lt;- impute(out, type = "both", rho.new = rho.new)


# Y ~ N(XB, Sigma) and 
# 1. probability of left/right censored values equal to 0.05
# 2. probability of missing-at-random values equal to 0.05
n &lt;- 100L
p &lt;- 3L
q &lt;- 2
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probl = 0.05, probr = 0.05, 
           probna = 0.05)

out &lt;- cglasso(. ~ ., data = Z)
lambda.new &lt;- mean(out$lambda)
rho.new &lt;- mean(out$rho)

# imputing missing values
Y.impute &lt;- impute(out, type = "mar", lambda.new = lambda.new, rho.new = rho.new)

# imputing censored values
Y.impute &lt;- impute(out, type = "censored", lambda.new = lambda.new, rho.new = rho.new)

# imputing missing and censored values
Y.impute &lt;- impute(out, type = "both", lambda.new = lambda.new, rho.new = rho.new)
</code></pre>

<hr>
<h2 id='is.cglasso2igraph'>Is an Object of Class &lsquo;cglasso2igraph&rsquo;?</h2><span id='topic+is.cglasso2igraph'></span>

<h3>Description</h3>

<p><code>is.cglasso2igraph</code> tests if its argument is an object of class &lsquo;cglasso2igraph&rsquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.cglasso2igraph(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.cglasso2igraph_+3A_x">x</code></td>
<td>
<p>object to be tested.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>See Also</h3>

<p><code><a href="#topic+to_graph">to_graph</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
n &lt;- 100L
p &lt;- 3L
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, Sigma = Sigma, probl = 0.05, probr = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
out.graph &lt;- to_graph(out)
is.cglasso2igraph(out.graph )
</code></pre>

<hr>
<h2 id='is.datacggm'>Is an Object of Class &lsquo;datacggm&rsquo;?</h2><span id='topic+is.datacggm'></span>

<h3>Description</h3>

<p>&lsquo;<code>is.datacggm</code>&rsquo; tests if its argument is an object of class &lsquo;datacggm&rsquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.datacggm(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.datacggm_+3A_x">x</code></td>
<td>
<p>object to be tested.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>References</h3>

<p>Augugliaro L., Sottile G., Wit E.C., and Vinciotti V. (2023) &lt;<a href="https://doi.org/10.18637/jss.v105.i01">doi:10.18637/jss.v105.i01</a>&gt;.
cglasso: An R Package for Conditional Graphical Lasso Inference with Censored and Missing Values.
<em>Journal of Statistical Software</em> <b>105</b>(1), 1&ndash;58.
</p>
<p>Augugliaro, L., Sottile, G., and Vinciotti, V. (2020a) &lt;<a href="https://doi.org/10.1007/s11222-020-09945-7">doi:10.1007/s11222-020-09945-7</a>&gt;.
The conditional censored graphical lasso estimator.
<em>Statistics and Computing</em> <b>30</b>, 1273&ndash;1289.
</p>
<p>Augugliaro, L., Abbruzzo, A., and Vinciotti, V. (2020b) &lt;<a href="https://doi.org/10.1093/biostatistics/kxy043">doi:10.1093/biostatistics/kxy043</a>&gt;.
<code class="reqn">\ell_1</code>-Penalized censored Gaussian graphical model.
<em>Biostatistics</em> <b>21</b>, e1&ndash;e16.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+datacggm">datacggm</a></code> and <code><a href="#topic+rcggm">rcggm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
n &lt;- 100L
p &lt;- 3L
b0 &lt;- rep(0, p)
Z &lt;- rcggm(n = n, b0 = b0, probl = 0.05, probr = 0.05)
is.datacggm(Z) # TRUE
</code></pre>

<hr>
<h2 id='lower+20+2B+20upper'>Lower and Upper Limits from a &ldquo;datacggm&rdquo; Object</h2><span id='topic+lower'></span><span id='topic+upper'></span>

<h3>Description</h3>

<p>Functions &lsquo;<code>lower</code>&rsquo; and &lsquo;<code>upper</code>&rsquo; retrieve the lower and upper censoring values from an object of class &ldquo;datacggm&rdquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lower(x)
upper(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lower+2B20+2B2B+2B20upper_+3A_x">x</code></td>
<td>
<p>an object of class &lsquo;<code>datacggm</code>&rsquo;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For an <span class="rlang"><b>R</b></span> object <code>x</code> of class &lsquo;<code>datacggm</code>&rsquo;, <code>lower</code> (<code>upper</code>) retrieves the lower (upper) censoring values of the response variables.
</p>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>See Also</h3>

<p><code><a href="#topic+datacggm">datacggm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
n &lt;- 100L
p &lt;- 3L
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))

Z &lt;- rcggm(n = n, Sigma = Sigma, probr = 0.05)
lower(Z)
upper(Z)

Z &lt;- rcggm(n = n, Sigma = Sigma, probl = 0.05)
lower(Z)
upper(Z)

Z &lt;- rcggm(n = n, Sigma = Sigma, probl = 0.05, probr = 0.05)
lower(Z)
upper(Z)
</code></pre>

<hr>
<h2 id='MKMEP'>
Megakaryocyte-Erythroid Progenitors
</h2><span id='topic+MKMEP'></span>

<h3>Description</h3>

<p>In a study about the formation of blood cells, Psaila <em>and others</em> (2016) have recently identified three distinct sub-populations of cells, which are all derived from hematopoietic stem cells through cell differentiation. One of these sub-populations, denoted by MK-MEP, is a previously unknown, rare population of cells that are bipotent but primarily generate megakaryocytic progeny.
</p>
<p>Multiplex RT-qPCR has been used to profile 63 genes and 48 single human MK-MEP cells. RT-qPCR data are typically right-censored with a limit of detection fixed by the manufacturer to 40. Raw data have been mean normalized using the method proposed in Pipelers <em>and others</em> (2017). See Section 5 in Augugliaro <em>and others</em> (2018) for more details.
</p>
<p>&lsquo;<code>MKMEP</code>&rsquo; is a matrix containing a subset of the data available from Psaila <em>and others</em> (2016).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("MKMEP")</code></pre>


<h3>References</h3>

<p>Augugliaro L., Sottile G., Wit E.C., and Vinciotti V. (2023) &lt;<a href="https://doi.org/10.18637/jss.v105.i01">doi:10.18637/jss.v105.i01</a>&gt;.
cglasso: An R Package for Conditional Graphical Lasso Inference with Censored and Missing Values.
<em>Journal of Statistical Software</em> <b>105</b>(1), 1&ndash;58.
</p>
<p>Augugliaro, L., Sottile, G., and Vinciotti, V. (2020) &lt;<a href="https://doi.org/10.1007/s11222-020-09945-7">doi:10.1007/s11222-020-09945-7</a>&gt;.
The conditional censored graphical lasso estimator.
<em>Statistics and Computing</em> <b>30</b>, 1273&ndash;1289.
</p>
<p>Augugliaro, L., Abbruzzo, A., and Vinciotti, V. (2020) &lt;<a href="https://doi.org/10.1093/biostatistics/kxy043">doi:10.1093/biostatistics/kxy043</a>&gt;.
<code class="reqn">\ell_1</code>-Penalized censored Gaussian graphical model.
<em>Biostatistics</em> <b>21</b>, e1&ndash;e16.
</p>
<p>Pipelers, P., Clement, L., Vynck, M., Hellemans, J., Vandesompele, J. and Thas, O. (2017) &lt;doi: 10.1371/journal.pone.0182832&gt;.
A unified censored normal regression model for qPCR differential gene expression analysis.
<em>PLoS One</em> <b>12</b>, e0182832.
</p>
<p>Psaila, B., Barkas, N., Iskander, D., Roy, A., Anderson, S., Ashley, N., Caputo, V. S., Lichtenberg, J., Loaiza, S., Bodine, D. M. <em>and others</em>. (2016) &lt;doi: 10.1186/s13059-016-0939-7&gt;
Single-cell profiling of human megakaryocyte-erythroid progenitors identifies distinct megakaryocyte and erythroid differentiation pathways.
<em>Genome Biology</em> <b>17</b>, 83&ndash;102.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cglasso">cglasso</a></code>, <code><a href="#topic+to_graph">to_graph</a></code>, and the method functions <code><a href="base.html#topic+summary">summary</a></code>, <code><a href="#topic+coef">coef</a></code>, <code><a href="base.html#topic+plot">plot</a></code>, <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code> and <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("MKMEP")
is.matrix(MKMEP)
Z &lt;- datacggm(Y = MKMEP, up = 40)
str(Z)
is.datacggm(Z)
print(Z, width = 80L)
</code></pre>

<hr>
<h2 id='MM'>
The Rule of miRNA in Multiple Myeloma
</h2><span id='topic+MM'></span>

<h3>Description</h3>

<p>MicroRNAs (miRNAs) are endogenous small non-coding RNAs, approximately 22 nucleotides in length, that play a regulatory role in gene expression by mediating mRNA cleavage or translation expression. Several studies have shown that a deregulation of the miRNAs can cause a disruption in the gene regulation mechanisms of the cell and that this might even lead to cancerous phenotypes.
</p>
<p>Gutierrez <em>and others</em> (2010) investigated the expression level of a set of 141 miRNAs, measured by RT-qPCR technology on a sample of 64 patients with multiple myeloma (MM). Patients were selected to represent the most relevant recurrent genetic abnormalities in MM. RT-qPCR data are typically right-censored and, in this study, the upper limit of detection was fixed to 40 cycles.
</p>
<p>Formally, <code>MM</code> is a list with containing a subset of the date studied in Gutierrez <em>and others</em> (2010). Its components are:
</p>

<ul>
<li> <p><code>Y</code>: the response matrix containing the cycle-threshold of the measured 49 miRNAs;
</p>
</li>
<li> <p><code>X</code>: the predictor matrix containing the expression levels of 15 endogenous internal reference genes, called housekeeping, and a factor encoding the cytogenetic abnormalities.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>data("MM")</code></pre>


<h3>References</h3>

<p>Augugliaro L., Sottile G., Wit E.C., and Vinciotti V. (2023) &lt;<a href="https://doi.org/10.18637/jss.v105.i01">doi:10.18637/jss.v105.i01</a>&gt;.
cglasso: An R Package for Conditional Graphical Lasso Inference with Censored and Missing Values.
<em>Journal of Statistical Software</em> <b>105</b>(1), 1&ndash;58.
</p>
<p>Augugliaro, L., Sottile, G., and Vinciotti, V. (2020) &lt;<a href="https://doi.org/10.1007/s11222-020-09945-7">doi:10.1007/s11222-020-09945-7</a>&gt;.
The conditional censored graphical lasso estimator.
<em>Statistics and Computing</em> <b>30</b>, 1273&ndash;1289.
</p>
<p>Augugliaro, L., Abbruzzo, A., and Vinciotti, V. (2020) &lt;<a href="https://doi.org/10.1093/biostatistics/kxy043">doi:10.1093/biostatistics/kxy043</a>&gt;.
<code class="reqn">\ell_1</code>-Penalized censored Gaussian graphical model.
<em>Biostatistics</em> <b>21</b>, e1&ndash;e16.
</p>
<p>Gutierrez, N. Sarasquete, M., Misiewicz-Krzeminska, I., Delgado, M. De Las Rivas, J., Ticona, F.V., Ferminan, E., Martin-Jimenez, P., Chillon, C., Risueno, A., Hernandez, J.M., Garcia-Sanz, R., Gonzalez, M. and San Miguel, J.F. (2010) &lt;<a href="https://doi.org/10.1038/leu.2009.274">doi:10.1038/leu.2009.274</a>&gt;.
Deregulation of microRNA expression in the different genetic subtypes of multiple myeloma and correlation with gene expression profiling.
<em>Leukemia</em> <b>24</b>(3), 629&ndash;637.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cglasso">cglasso</a></code>, <code><a href="#topic+to_graph">to_graph</a></code>, and the method functions <code><a href="base.html#topic+summary">summary</a></code>, <code><a href="#topic+coef">coef</a></code>, <code><a href="base.html#topic+plot">plot</a></code>, <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code> and <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("MM")
Z &lt;- datacggm(Y = MM$Y, X = MM$X, up = 40)
print(Z, width = 80L)
</code></pre>

<hr>
<h2 id='nobs+20+2B+20nresp+20+2B+20npred'>Extract the Number of Observations/Responses/Predictors from a datacggm Object</h2><span id='topic+nobs'></span><span id='topic+nobs.datacggm'></span><span id='topic+nresp'></span><span id='topic+nresp.datacggm'></span><span id='topic+npred'></span><span id='topic+npred.datacggm'></span>

<h3>Description</h3>

<p>Extract the number of observations, response variables and predictors from an object of class <code>datacggm</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'datacggm'
nobs(object, ...)
## S3 method for class 'datacggm'
nresp(object, ...)
## S3 method for class 'datacggm'
npred(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nobs+2B20+2B2B+2B20nresp+2B20+2B2B+2B20npred_+3A_object">object</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object of class <code>datacggm</code>.</p>
</td></tr>
<tr><td><code id="nobs+2B20+2B2B+2B20nresp+2B20+2B2B+2B20npred_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to methods.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>See Also</h3>

<p><code><a href="#topic+datacggm">datacggm</a></code>, <code><a href="#topic+rcggm">rcggm</a></code> and <code><a href="#topic+dim.datacggm">dim.datacggm</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
n &lt;- 100
p &lt;- 3
q &lt;- 2
b0 &lt;- rep(1, p)
X &lt;- matrix(rnorm(n * q), n, q)
B &lt;- matrix(rnorm(q * p), q, p)
Sigma &lt;- outer(1:p, 1:p, function(i, j) 0.3^abs(i - j))
probl &lt;- 0.05
probr &lt;- 0.05
probna &lt;- 0.05

Z &lt;- rcggm(n = n, b0 = b0, Sigma = Sigma, probl = probl, probr = probr, 
           probna = probna)
nobs(Z)
nresp(Z)
npred(Z)

Z &lt;- rcggm(b0 = b0, X = X, B = B, Sigma = Sigma, probl = probl, probr = probr, 
           probna = probna)
nobs(Z)
nresp(Z)
npred(Z)
</code></pre>

<hr>
<h2 id='plot.cggm'>Plot Method for a &lsquo;<code>cggm</code>&rsquo; Object</h2><span id='topic+plot.cggm'></span>

<h3>Description</h3>

<p>The <code>plot.cggm</code> produces graphs from an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>cggm</code>&rsquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cggm'
plot(x, type, weighted = FALSE, simplify = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.cggm_+3A_x">x</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>cggm</code>&rsquo;, that is, the output of the function <code><a href="#topic+cggm">cggm</a></code>.</p>
</td></tr>
<tr><td><code id="plot.cggm_+3A_type">type</code></td>
<td>
<p>a description of the required graph. Default depends on the type of fitted model, that is, <code>type = both</code> if a conditional censored graphical lasso estimator is fitted, otherwise <code>type</code> is equal to <code>Gyy</code> (see <code><a href="#topic+getGraph">getGraph</a></code>) for more details.</p>
</td></tr>
<tr><td><code id="plot.cggm_+3A_weighted">weighted</code></td>
<td>
<p>logical. Should weighted graphs be created? Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plot.cggm_+3A_simplify">simplify</code></td>
<td>
<p>logical. Should isolated vertices be removed from the graph? Default is <code>TRUE</code>, i.e., isolated vertices are removed.</p>
</td></tr>
<tr><td><code id="plot.cggm_+3A_...">...</code></td>
<td>
<p>additional graphical arguments passed to the functions <code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>See Also</h3>

<p><code><a href="#topic+cggm">cggm</a></code>, <code><a href="#topic+to_graph">to_graph</a></code>, <code><a href="#topic+getGraph">getGraph</a></code> and <code><a href="#topic+plot.cglasso2igraph">plot.cglasso2igraph</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
# Y ~ N(XB, Sigma)  and
# 1. probability of left/right censored values equal to 0.05
# 2. probability of missing-at-random values equal to 0.05
n &lt;- 100L
p &lt;- 3L
q &lt;- 2L
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probl = 0.05, probr = 0.5, 
           probna = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
out.mle &lt;- cggm(out, lambda.id = 3L, rho.id = 3L)
plot(out.mle, type = "Gyy")
plot(out.mle, type = "Gxy")
plot(out.mle, type = "both")
</code></pre>

<hr>
<h2 id='plot.cglasso'>Plot Method for &lsquo;<code>cglasso</code>&rsquo; Object</h2><span id='topic+plot.cglasso'></span>

<h3>Description</h3>

<p>&lsquo;The <code>plot.cglasso</code>&rsquo; function produces plots to study the coefficient paths of fitted cglasso models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cglasso'
plot(x, what = c("Theta", "diag(Theta)", "b0", "B"), 
     penalty = ifelse(x$nrho &gt;= x$nlambda, "rho", "lambda"), given = NULL,
     GoF = AIC, add.labels, matplot.arg1, matplot.arg2, labels.arg, abline.arg,
     mtext.arg, save.plot, grdev = pdf, grdev.arg, digits = 4L, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.cglasso_+3A_x">x</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>cglasso</code>&rsquo;, that is, the output of the fitting function <code><a href="#topic+cglasso">cglasso</a></code>.</p>
</td></tr>
<tr><td><code id="plot.cglasso_+3A_what">what</code></td>
<td>
<p>a character or a formula specifying the required plot. If <code>what</code> is a character, then it is used to specify the conditional coefficient path. Allowed descriptors are: &lsquo;<code>Theta</code>&rsquo;, the off-diagonal entries of the precision matrix, &lsquo;<code>diag(Theta)</code>&rsquo;, the diagonal entries of the precision matrix, &lsquo;<code>b0</code>&rsquo;, the intercepts of the conditional models, or equivalently the expected values of the response variables, and &lsquo;<code>B</code>&rsquo;, the regression coefficients. Optionally, &lsquo;<code>what</code>&rsquo; can be a model formula (see sections Description and Examples for more details).</p>
</td></tr>
<tr><td><code id="plot.cglasso_+3A_penalty">penalty</code></td>
<td>
<p>optional character argument used to specify the tuning parameter needed to plot the conditional coefficient path. Allowed descriptors are &lsquo;<code>rho</code>&rsquo; and &lsquo;<code>lambda</code>&rsquo;. This argument can be omitted if <code>what</code> is a model formula.</p>
</td></tr>
<tr><td><code id="plot.cglasso_+3A_given">given</code></td>
<td>
<p>an optional vector of integers identifying the conditioning values of the second tuning parameter to be used for the  coefficient profile plots of the parameters identified by the input &lsquo;<code>what</code>&rsquo; across their corresponding tuning parameter. This argument is required only if &lsquo;<code>what</code>&rsquo; is a character, otherwise it can be omitted.</p>
</td></tr>
<tr><td><code id="plot.cglasso_+3A_gof">GoF</code></td>
<td>
<p>a valid goodness-of-fit function, such as <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code> and <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>, or an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>GoF</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="plot.cglasso_+3A_add.labels">add.labels</code></td>
<td>
<p>logical value. Should labels be added to the plot?</p>
</td></tr>
<tr><td><code id="plot.cglasso_+3A_matplot.arg1">matplot.arg1</code></td>
<td>
<p>a named list with the graphical parameters used to plot the paths of the estimates identified by &lsquo;<code>GoF</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="plot.cglasso_+3A_matplot.arg2">matplot.arg2</code></td>
<td>
<p>a named list with the graphical parameters used to plot the paths of the remaining estimates.</p>
</td></tr>
<tr><td><code id="plot.cglasso_+3A_labels.arg">labels.arg</code></td>
<td>
<p>a named list with the graphical parameters used to plot the labels.</p>
</td></tr>
<tr><td><code id="plot.cglasso_+3A_abline.arg">abline.arg</code></td>
<td>
<p>a named list with the graphical parameters used to plot the line identifying the optimal tuning parameter value.</p>
</td></tr>
<tr><td><code id="plot.cglasso_+3A_mtext.arg">mtext.arg</code></td>
<td>
<p>a named list with the graphical parameters used to plot the text reported on the third axis.</p>
</td></tr>
<tr><td><code id="plot.cglasso_+3A_save.plot">save.plot</code></td>
<td>
<p>a logical variable or a string specifying the path of the directory where plots will be saved. Letting &lsquo;<code>save.plot = TRUE</code>&rsquo;, the required plots will be saved as external files inside the current working directory. User can save these files on a specific working directory passing the absolute path through the argument <code>save.plot</code>. On exit, working directory will be always set the previous one.</p>
</td></tr>
<tr><td><code id="plot.cglasso_+3A_grdev">grdev</code></td>
<td>
<p>the graphics device used to save the required histograms on external files. See &lsquo;<code><a href="grDevices.html#topic+device">device</a></code>&rsquo; for more details.</p>
</td></tr>
<tr><td><code id="plot.cglasso_+3A_grdev.arg">grdev.arg</code></td>
<td>
<p>additional parameters passed to the graphics device specified by &lsquo;<code>grdev</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="plot.cglasso_+3A_digits">digits</code></td>
<td>
<p>number of digits used to print the value of the second tuning parameter identified by &lsquo;<code>given</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="plot.cglasso_+3A_...">...</code></td>
<td>
<p>further arguments passed to the chosen goodness-of-fit function, such as &lsquo;<code>AIC</code>&rsquo; or &lsquo;<code>BIC</code>&rsquo;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>plot.cglasso</code> produces the coefficient profile plot. The output depends both on the type of fitted model and on the setting of the three main arguments, <code>what</code>, <code>penalty</code> and <code>given</code>. Below we give more details.
</p>
<p>If the model fitting function <code><a href="#topic+cglasso">cglasso</a></code> is used to fit an l1-penalized censored Gaussian graphical model (see first part in Section Example), then the user can specify only the main argument &lsquo;<code>what</code>&rsquo;, whereas &lsquo;<code>penalty</code>&rsquo; and &lsquo;<code>given</code>&rsquo; can be omitted. If &lsquo;<code>penalty</code>&rsquo; is specified, then it must be equal to &ldquo;<code>rho</code>&rdquo;. The main argument &lsquo;<code>what</code>&rsquo; is used to specify the estimator that is plotted on the <code class="reqn">y</code>-axis. In this case, it must be equal to one of the following descriptors:
</p>

<ul>
<li> <p>&ldquo;<code>Theta</code>&rdquo;: the path of the estimated partial correlation coefficients is returned;
</p>
</li>
<li> <p>&ldquo;<code>diag(Theta)</code>&rdquo;: the path of the diagonal entries of the estimated  precision matrix is returned;
</p>
</li>
<li> <p>&ldquo;<code>b0</code>&rdquo;: the path of the estimated  expected values of the response variables is returned;
</p>
</li></ul>

<p>If a l1-penalized conditional censored Gaussian graphical model is fitted (see second part in Section Example), then all the main arguments can be used. In this case:
</p>

<ol>
<li> <p>&lsquo;<code>what</code>&rsquo; must be equal to one of the following descriptors:
</p>

<ul>
<li> <p>&ldquo;<code>Theta</code>&rdquo;: the path of the off-diagonal entries of the estimated precision matrix is returned;
</p>
</li>
<li> <p>&ldquo;<code>diag(Theta)</code>&rdquo;: the path of the estimated partial correlation coefficients is returned;
</p>
</li>
<li> <p>&ldquo;<code>b0</code>&rdquo;: the path of the estimated intercepts is returned;
</p>
</li>
<li> <p>&ldquo;<code>B</code>&rdquo;: the path of the estimated regression coefficients is returned for each response variable;
</p>
</li></ul>

</li>
<li> <p>&lsquo;<code>penalty</code>&rsquo; is used to specify the tuning parameter that is reported on the <code class="reqn">x</code>-axis and it must be equal to one of the following descriptors:
</p>

<ul>
<li> <p>&ldquo;<code>rho</code>&rdquo;: the tuning parameter used to penalize the precision matrix;
</p>
</li>
<li> <p>&ldquo;<code>lambda</code>&rdquo;: the tuning parameter used to penalize the regression coefficient matrix;
</p>
</li></ul>

</li>
<li> <p>&lsquo;<code>given</code>&rsquo; is an optional vector of integers used to specify the values of the second tuning parameter. For instance, letting <code>what = "Theta"</code>, <code>penalty = "rho"</code> and <code>given = 1</code> then <code>plot.cglasso</code> returns the plot of the estimated partial correlation coefficients versus the <code class="reqn">\rho</code>-values and given the first value of the used <code class="reqn">\lambda</code>-values. If <code>given</code> is left unspecified then a sequence of conditional profile plot is returned, a plot for each value of the second tuning parameter. 
</p>
</li></ol>

<p>Optionally, the user can specify a conditional profile plot using a model formula with the following template:
</p>
<p>&lsquo;<code>what ~ penalty | given</code>&rsquo;
</p>
<p>For instance, the previous plot can be specified using the following model formula
</p>
<p>&lsquo;<code>Theta ~ rho | 1</code>&rsquo;
</p>
<p>In this case, the arguments <code>penalty</code> and <code>given</code> must be left unspecified whereas <code>what</code> is used for the model formula (see also examples below).
</p>
<p>The optional argument <code>GoF</code> can be used to identify the non-zero estimates by the output of a goodness-of-fit function, such as <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code> and <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>. In this case, a vertical red dashed line is used the identify the optimal value of the tuning parameter reported on the <code class="reqn">x</code>-axis, whereas the path of the identified non-zero estimates is drawn using a solid black line; the remaining paths are drawn using gray dashed lines. This option is disabled if we let &lsquo;<code>GoF = NULL</code>&rsquo;, when all the paths are drawn usign solid black lines.
</p>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>References</h3>

<p>Augugliaro L., Sottile G., Wit E.C., and Vinciotti V. (2023) &lt;<a href="https://doi.org/10.18637/jss.v105.i01">doi:10.18637/jss.v105.i01</a>&gt;.
cglasso: An R Package for Conditional Graphical Lasso Inference with Censored and Missing Values.
<em>Journal of Statistical Software</em> <b>105</b>(1), 1&ndash;58.
</p>
<p>Augugliaro, L., Abbruzzo, A., and Vinciotti, V. (2020a) &lt;<a href="https://doi.org/10.1093/biostatistics/kxy043">doi:10.1093/biostatistics/kxy043</a>&gt;.
<code class="reqn">\ell_1</code>-Penalized censored Gaussian graphical model.
<em>Biostatistics</em> <b>21</b>, e1&ndash;e16.
</p>
<p>Augugliaro, L., Sottile, G., and Vinciotti, V. (2020b) &lt;<a href="https://doi.org/10.1007/s11222-020-09945-7">doi:10.1007/s11222-020-09945-7</a>&gt;.
The conditional censored graphical lasso estimator.
<em>Statistics and Computing</em> <b>30</b>, 1273&ndash;1289.
</p>
<p>Friedman, J.H., Hastie, T., and Tibshirani, R. (2008) &lt;<a href="https://doi.org/10.1093/biostatistics/kxm045">doi:10.1093/biostatistics/kxm045</a>&gt;.
Sparse inverse covariance estimation with the graphical lasso.
<em>Biostatistics</em> <b>9</b>, 432&ndash;441.
</p>
<p>Stadler, N. and Buhlmann, P. (2012) &lt;<a href="https://doi.org/10.1007/s11222-010-9219-7">doi:10.1007/s11222-010-9219-7</a>&gt;.
Missing values: sparse inverse covariance estimation and an extension to sparse regression.
<em>Statistics and Computing</em> <b>22</b>, 219&ndash;235.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cglasso">cglasso</a></code>, <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code> and <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)

################################################################################
# Model 1: censored glasso estimator (Augugliaro \emph{and other}, 2020a)
# Y ~ N(0, Sigma) and probability of left/right censored values equal to 0.05
################################################################################
n &lt;- 100L
p &lt;- 10L
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, Sigma = Sigma, probl = 0.05, probr = 0.05)
out &lt;- cglasso(. ~ ., data = Z)

# coefficient profile plot of the off-diagonal entries of the precision matrix
plot(out, what = "Theta")
# the same as
#plot(out, what = "Theta", penalty = "rho")

# the output of a goodness-of-fit function can be used to identify the 
# non-zero estimates
plot(out, what = "Theta", GoF = BIC)

# letting 'GoF = NULL' the previous option is disabled
plot(out, what = "Theta", GoF = NULL)

# coefficient profile plot of the diagonal entries of the precision matrix
plot(out, what = "diag(Theta)")

# coefficient profile plot of the expected values of the response variables
plot(out, what = "b0")

################################################################################
# Model 2: conditional censored glasso estimator (Augugliaro \emph{and other}, 2020b)
# Y ~ N(b0 + XB, Sigma)  and probability of left/right censored values equal to 0.05
################################################################################
n &lt;- 100L
p &lt;- 10L
q &lt;- 5L
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probl = 0.05, probr = 0.05)
out &lt;- cglasso(. ~ ., data = Z)

# conditional profile plot of the estimater partial correlation coefficients versus
# the used values of the tunin parameter rho and given the first lambda-value
plot(out, what = "Theta", penalty = "rho", given = 1L)
out$lambda[1L]

lambda.id &lt;- c(2, 4)
plot(out, what = "Theta", penalty = "rho", given = lambda.id)
out$lambda[lambda.id]

# in this case a sequence of ten conditional profile plots is returned, tha is a 
# plot for each lambda-value.
plot(out, what = "Theta", penalty = "rho")

# optionally, the user can specify the conditional profile plots using the model
# formula
plot(out, what = Theta ~ rho | lambda.id)
lambda.id

plot(out, what = Theta ~ rho)

# the output of a goodness-of-fit function can be used to identify the 
# non-zero estimates
plot(out, what = Theta ~ rho | 10, GoF = BIC)

# letting 'GoF = NULL' the previous option is disabled
plot(out, what = Theta ~ rho | 10, GoF = NULL)
</code></pre>

<hr>
<h2 id='plot.cglasso2igraph'>Plot Method for a cglasso2igraph Object&quot;</h2><span id='topic+plot.cglasso2igraph'></span>

<h3>Description</h3>

<p><code>plot.cglasso2igraph</code> produces graphs from an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>cglasso2igraph</code>&rsquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cglasso2igraph'
plot(x, type, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.cglasso2igraph_+3A_x">x</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>cglasso2igraph</code>&rsquo;, that is, the output of the function <code><a href="#topic+to_graph">to_graph</a></code>.</p>
</td></tr>
<tr><td><code id="plot.cglasso2igraph_+3A_type">type</code></td>
<td>
<p>a description of the required graph. Default is &lsquo;<code>both</code>&rsquo; in a conditional glasso estimator or &lsquo;<code>Gyy</code>&rsquo; in a glasso estimator (see <code><a href="#topic+getGraph">getGraph</a></code>).</p>
</td></tr>
<tr><td><code id="plot.cglasso2igraph_+3A_...">...</code></td>
<td>
<p>additional graphical arguments passed to the functions <code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>See Also</h3>

<p><code><a href="#topic+to_graph">to_graph</a></code> and <code><a href="#topic+getGraph">getGraph</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
# Y ~ N(0, Sigma) and probability of left/right censored values equal to 0.05
n &lt;- 100L
p &lt;- 3L
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, Sigma = Sigma, probl = 0.05, probr = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
out.graph &lt;- to_graph(out)
plot(out.graph, type = "Gyy")

out.graph &lt;- to_graph(out, weighted = TRUE)
plot(out.graph,  type = "Gyy")


# Y ~ N(b0 +XB, Sigma)  and probability of left/right censored values equal to 0.05
n &lt;- 100L
p &lt;- 3L
q &lt;- 2L
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probl = 0.05, probr = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
out.graph &lt;- to_graph(out, lambda.id = 3, rho.id = 3, weighted = TRUE)
plot(out.graph, type = "Gyy")
plot(out.graph, type = "Gxy")
plot(out.graph, type = "both")
</code></pre>

<hr>
<h2 id='plot.GoF'>Plot for &lsquo;<code>GoF</code>&rsquo; Object</h2><span id='topic+plot.GoF'></span>

<h3>Description</h3>

<p>&lsquo;The <code>plot.GoF</code>&rsquo; function produces plots to study the sequence of fitted models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GoF'
plot(x, add.line = TRUE, arg.line = list(lty = 2L, lwd = 2L, col = "red"), 
      add.text = FALSE, arg.text = list(side = 3L), arg.points = list(pch = 2L),
      ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.GoF_+3A_x">x</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>GoF</code>&rsquo;, that is, the output of a goodness-of-fit function such as <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code> or <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>.</p>
</td></tr>
<tr><td><code id="plot.GoF_+3A_add.line">add.line</code></td>
<td>
<p>logical; if &lsquo;<code>add.line = TRUE</code>&rsquo; then a line is added to identify the optimal value of the tuning parameter.</p>
</td></tr>
<tr><td><code id="plot.GoF_+3A_arg.line">arg.line</code></td>
<td>
<p>a named list of graphical parameters passed to the function <code><a href="graphics.html#topic+abline">abline</a></code> (see also <code><a href="graphics.html#topic+par">par</a></code>).</p>
</td></tr>
<tr><td><code id="plot.GoF_+3A_add.text">add.text</code></td>
<td>
<p>logical; if &lsquo;<code>add.text = TRUE</code>&rsquo; then a text is added to the line used to identify the optimal value of the tuning parameter.</p>
</td></tr>
<tr><td><code id="plot.GoF_+3A_arg.text">arg.text</code></td>
<td>
<p>a list of further parameters passed to the function <code><a href="graphics.html#topic+mtext">mtext</a></code> (only if &lsquo;<code>add.text = TRUE</code>&rsquo;).</p>
</td></tr>
<tr><td><code id="plot.GoF_+3A_arg.points">arg.points</code></td>
<td>
<p>a named list of graphical parameters passed to the function <code><a href="graphics.html#topic+points">points</a></code>.</p>
</td></tr>
<tr><td><code id="plot.GoF_+3A_...">...</code></td>
<td>
<p>additional graphical arguments passed to the functions <code><a href="graphics.html#topic+plot">plot</a></code>,  <code><a href="graphics.html#topic+contour">contour</a></code> or <code><a href="graphics.html#topic+filled.contour">filled.contour</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>plot.GoF</code> is the plotting method function of an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>GoF</code>&rsquo;, that is, the output of a goodness-of-fit function (see <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code>, or <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>). This function produces a plot aimed both to evaluate the sequence of fitted models in terms of goodness-of-fit and to identify the optimal values of the tuning parameters.
</p>
<p>If a tuning parameter is held fixed, then <code>plot.GoF</code> produces a plot showing the chosen measure of goodness-of-fit as a function of the remaining tuning parameter. In this case, the optimal value is identified by a vertical dashed line. The degrees-of-freedom of the selected fitted model are also shown.
</p>
<p>If the cglasso model is fitted using both a sequence of <code class="reqn">\rho</code> and <code class="reqn">\lambda</code> values, then <code>plot.GoF</code> produces a contour plot and a triangle is used to identify the optimal pair of the two tuning parameters.
</p>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>See Also</h3>

<p><code><a href="#topic+cglasso">cglasso</a></code>, <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code>, <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>, <code><a href="#topic+summary.cglasso">summary.cglasso</a></code> and <code><a href="#topic+select_cglasso">select_cglasso</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
n &lt;- 1000L
p &lt;- 3L
q &lt;- 2
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probl = 0.05, probr = 0.05)

out &lt;- cglasso(. ~ ., data = Z, nlambda = 1L)
plot(AIC(out))
plot(BIC(out))

out &lt;- cglasso(. ~ ., data = Z, nrho = 1L)
plot(AIC(out))
plot(BIC(out))

out &lt;- cglasso(. ~ ., data = Z)
plot(AIC(out))
plot(BIC(out))
</code></pre>

<hr>
<h2 id='predict'>Predict Method for cglasso and cggm Fits</h2><span id='topic+predict'></span><span id='topic+predict.cglasso'></span><span id='topic+predict.cggm'></span>

<h3>Description</h3>

<p>Obtains predictions from an <span class="rlang"><b>R</b></span> object inheriting class &lsquo;<code>cglasso</code>&rsquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cglasso'
predict(object, type = c("B", "mu", "Sigma", "Theta"), X.new, lambda.new, rho.new,
        ...)
        
## S3 method for class 'cggm'
predict(object, X.new, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict_+3A_object">object</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object inheriting class &lsquo;<code>cglasso</code>&rsquo;, that is, the output of the model-fitting function &lsquo;<code>cglasso</code>&rsquo; or &lsquo;<code>cggm</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="predict_+3A_type">type</code></td>
<td>
<p>a description of prediction required.</p>
</td></tr>
<tr><td><code id="predict_+3A_x.new">X.new</code></td>
<td>
<p>matrix of new values for <code>X</code> at which predictions are to be made. This argument is used only if &lsquo;<code>type = "mu"</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="predict_+3A_lambda.new">lambda.new</code></td>
<td>
<p>value of the tuning parameter <code class="reqn">\lambda</code> at which predictions are required.</p>
</td></tr>
<tr><td><code id="predict_+3A_rho.new">rho.new</code></td>
<td>
<p>value of the tuning parameter <code class="reqn">\rho</code> at which predictions are required.</p>
</td></tr>
<tr><td><code id="predict_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>object</code> has S3 class &lsquo;<code>cglasso</code>&rsquo;, then for a new pair of the tuning parameters <code class="reqn">\lambda</code> and <code class="reqn">\rho</code>, the <code>predict</code> function can be used to predict the estimate of the regression coefficient matrix (&lsquo;<code>type = "B"</code>&rsquo;), the estimate of the covariance matrix (&lsquo;<code>type = "Sigma"</code>&rsquo;) or the estimate of the precision matrix (&lsquo;<code>type = "Theta"</code>&rsquo;). If <code>X.new</code> is missing and &lsquo;<code>type = "mu"</code>&rsquo;, then the <code>predict</code> function returns the predicted values using the matrix of predictors <code>X</code>, otherwise the predicted fitted values are computed using the matrix <code>X.new</code>.
</p>
<p>For a new pair of the tuning parameters <code class="reqn">\lambda</code> and <code class="reqn">\rho</code>, the predicted values are computed using a bilinear interpolation.
</p>
<p>If the <code>object</code> has S3 class &lsquo;<code>cggm</code>&rsquo;, then the <code>predict</code> function returns only the predicted fitted values using the argument <code>X.new</code>. 
</p>


<h3>Value</h3>

<p>The matrix of predicted values.
</p>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>See Also</h3>

<p>Model-fitting function <code><a href="#topic+cglasso">cglasso</a></code> and the other accessor functions <code><a href="#topic+coef.cglasso">coef.cglasso</a></code>, <code><a href="#topic+fitted.cglasso">fitted.cglasso</a></code>, <code><a href="#topic+residuals.cglasso">residuals.cglasso</a></code> and <code><a href="#topic+impute">impute</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
# Y ~ N(0, Sigma) and probability of left/right censored values equal to 0.05
n &lt;- 100L
p &lt;- 3L
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, Sigma = Sigma, probl = 0.05, probr = 0.05)

out &lt;- cglasso(. ~ ., data = Z)
rho.new &lt;- mean(out$rho)
Theta.pred &lt;- predict(out, rho.new = rho.new, type = "Theta")
Theta.pred

# Y ~ N(b0 + XB, Sigma)  and probability of left/right censored values equal to 0.05
n &lt;- 100L
p &lt;- 3L
q &lt;- 2
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probl = 0.05, probr = 0.05)

out &lt;- cglasso(. ~ ., data = Z)
rho.new &lt;- mean(out$rho)
lambda.new &lt;- mean(out$lambda)
Theta.pred &lt;- predict(out, lambda.new = lambda.new, rho.new = rho.new, type = "Theta")
Theta.pred
</code></pre>

<hr>
<h2 id='QFun'>Extract Q-Function</h2><span id='topic+QFun'></span>

<h3>Description</h3>

<p>&lsquo;<code>QFun</code>&rsquo; extracts the values of the Q-function from an <span class="rlang"><b>R</b></span> object inheriting class &lsquo;<code>cglasso</code>&rsquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QFun(object, mle, verbose = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QFun_+3A_object">object</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object inheriting class &lsquo;<code>cglasso</code>&rsquo;, that is, the output of the model-fitting functions <code><a href="#topic+cglasso">cglasso</a></code> and <code><a href="#topic+cggm">cggm</a></code>.</p>
</td></tr>
<tr><td><code id="QFun_+3A_mle">mle</code></td>
<td>
<p>logical. Should Q-values be computed using the maximum likelihood estimates? Default depends on the class of the argument <code>object</code>: <code>mle = FALSE</code> for objects with class <code>cglasso</code> and <code>mle = TRUE</code> for objects with class <code>cggm</code>.</p>
</td></tr>
<tr><td><code id="QFun_+3A_verbose">verbose</code></td>
<td>
<p>logical for printing out a progress bar on the <span class="rlang"><b>R</b></span> console. Default is <code>verbose = FALSE</code>.</p>
</td></tr>
<tr><td><code id="QFun_+3A_...">...</code></td>
<td>
<p>further arguments passed to <code><a href="#topic+cggm">cggm</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>&lsquo;<code>QFun</code>&rsquo; returns the value of the Q-function, i.e., the value of the function maximised in the M-step of the EM algorithm. The Q-function is defined as follows: </p>
<p style="text-align: center;"><code class="reqn">\frac{n}{2}\left\{\log det\Theta - tr(S\Theta) - p\log(2\pi)\right\},</code>
</p>
<p> where <code class="reqn">S</code> is the &lsquo;working&rsquo; empirical covariance matrix computed during the E-step.
</p>
<p><code>QFun</code> is used as a workhorse function to compute the measures of goodness-of-fit returned by the functions <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code> and <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>.
</p>
<p>The function &lsquo;<code>print.QFun</code>&rsquo; is used the improve the readability of the results.
</p>


<h3>Value</h3>

<p>&lsquo;<code>QFun</code>&rsquo; returns an <span class="rlang"><b>R</b></span> object of S3 class &ldquo;<code>QFun</code>&rdquo;, i.e., a named list containing the following components:
</p>
<table>
<tr><td><code>value</code></td>
<td>
<p>a matrix with the values of the Q-function.</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>a matrix with the number of estimated non-zero parameters.</p>
</td></tr>
<tr><td><code>dfB</code></td>
<td>
<p>a matrix with the number of estimated non-zero regression coefficients.</p>
</td></tr>
<tr><td><code>dfTht</code></td>
<td>
<p>a matrix with the number of estimated non-zero partial correlation coefficients.</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>the sample size.</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>the number of response variables.</p>
</td></tr>
<tr><td><code>q</code></td>
<td>
<p>the number of columns of the design matrix <code>X</code> used to fit the model.</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>the <code class="reqn">\lambda</code>-values used to fit the model.</p>
</td></tr>
<tr><td><code>nlambda</code></td>
<td>
<p>the number of <code class="reqn">\lambda</code>-values.</p>
</td></tr>
<tr><td><code>rho</code></td>
<td>
<p>the <code class="reqn">\rho</code>-values used to fit the model.</p>
</td></tr>
<tr><td><code>nrho</code></td>
<td>
<p>the number of <code class="reqn">\rho</code>-values.</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>a description of the fitted model passed through the argument <code>object</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>See Also</h3>

<p><code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code>, <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>, <code><a href="#topic+cglasso">cglasso</a></code>, <code><a href="#topic+cggm">cggm</a></code>, <code><a href="#topic+summary.cglasso">summary.cglasso</a></code>, <code><a href="#topic+select_cglasso">select_cglasso</a></code> and <code><a href="#topic+to_graph">to_graph</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
# Y ~ N(b0+ XB, Sigma)  and
# 1. probability of left/right censored values equal to 0.05
# 2. probability of missing-at-random values equal to 0.05
n &lt;- 100L
p &lt;- 3L
q &lt;- 2L
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1:p, 1:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probl = 0.05, probr = 0.5, 
           probna = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
QFun(out)

out.mle &lt;- cggm(out, lambda.id = 3L, rho.id = 3L)
QFun(out.mle)
</code></pre>

<hr>
<h2 id='qqcnorm'>Quantile-Quantile Plots for a <code>datacggm</code> Object</h2><span id='topic+qqcnorm'></span>

<h3>Description</h3>

<p>Creates a quantile-quantile plot for a censored Gaussian distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qqcnorm(x, which, max.plot = 1L, save.plot = FALSE, grdev = pdf, grdev.arg,
        main = "Censored Normal Q-Q Plot", xlab = "Theoretical Quantiles",
        ylab = "Sample Quantiles", plot.it = TRUE, plot.pch = c(2L, 20L),
        plot.col = c(2L, 1L), plot.cex = c(2L, 1L), abline = FALSE,
        line.col = "gray50", line.lwd = 1L, line.lty = 2L, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qqcnorm_+3A_x">x</code></td>
<td>
<p>an object of class &lsquo;<code>datacggm</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="qqcnorm_+3A_which">which</code></td>
<td>
<p>a vector of integers used to specify the response variables for which the histogram is required.</p>
</td></tr>
<tr><td><code id="qqcnorm_+3A_max.plot">max.plot</code></td>
<td>
<p>the maximum number of plots drawn in a single figure.</p>
</td></tr>
<tr><td><code id="qqcnorm_+3A_save.plot">save.plot</code></td>
<td>
<p>a logical variable or a string specifying the path of the directory where plots will be saved. Letting &lsquo;<code>save.plot = TRUE</code>&rsquo;, the required plots will be saved as external files inside the current working directory. User can save these files on a specific working directory passing the absolute path through the argument <code>save.plot</code>. On exit, working directory will be always set the previous one.</p>
</td></tr>
<tr><td><code id="qqcnorm_+3A_grdev">grdev</code></td>
<td>
<p>the graphics device used to save the required histograms on external files. See &lsquo;<code><a href="grDevices.html#topic+device">device</a></code>&rsquo; for more details.</p>
</td></tr>
<tr><td><code id="qqcnorm_+3A_grdev.arg">grdev.arg</code></td>
<td>
<p>additional parameters passed to the graphics device specified by &lsquo;<code>grdev</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="qqcnorm_+3A_main">main</code>, <code id="qqcnorm_+3A_xlab">xlab</code>, <code id="qqcnorm_+3A_ylab">ylab</code></td>
<td>
<p>plot labels.</p>
</td></tr>
<tr><td><code id="qqcnorm_+3A_plot.it">plot.it</code></td>
<td>
<p>logical. Should the result be plotted?</p>
</td></tr>
<tr><td><code id="qqcnorm_+3A_plot.pch">plot.pch</code></td>
<td>
<p>a pair of graphical parameters. The first entry specifies the symbol used for plotting the points associated to the censoring values <code class="reqn">lo</code> and <code class="reqn">up</code> (by default a triangle), whereas the second entry specifies the symbol used for the remaining points (by default a black point).</p>
</td></tr>
<tr><td><code id="qqcnorm_+3A_plot.col">plot.col</code></td>
<td>
<p>a pair of graphical parameters. The first entry specifies the colour used for plotting the points associated to the censoring values <code class="reqn">lo</code> and <code class="reqn">up</code> (by default a red triangle), whereas the second entry specifies the colour used for the remaining points (by default a black point).</p>
</td></tr>
<tr><td><code id="qqcnorm_+3A_plot.cex">plot.cex</code></td>
<td>
<p>a pair of graphical parameters. The first entry specifies the size of the symbol used for plotting the points associated to the censoring values <code class="reqn">lo</code> and <code class="reqn">up</code>, whereas the second entry specifies the size of the symbol used for the remaining points.</p>
</td></tr>
<tr><td><code id="qqcnorm_+3A_abline">abline</code></td>
<td>
<p>logical. Should the line <code class="reqn">y = x</code> be plotted?</p>
</td></tr>
<tr><td><code id="qqcnorm_+3A_line.col">line.col</code></td>
<td>
<p>graphical parameter. If &lsquo;<code>abline = TRUE</code>&rsquo;, then this argument specifies the colour of the line <code class="reqn">y = x</code>.</p>
</td></tr>
<tr><td><code id="qqcnorm_+3A_line.lwd">line.lwd</code></td>
<td>
<p>graphical parameter. If &lsquo;<code>abline = TRUE</code>&rsquo;, then this argument specifies the line width of the line <code class="reqn">y = x</code>.</p>
</td></tr>
<tr><td><code id="qqcnorm_+3A_line.lty">line.lty</code></td>
<td>
<p>graphical parameter. If &lsquo;<code>abline = TRUE</code>&rsquo;, then this argument specifies the line type of the line <code class="reqn">y = x</code>.</p>
</td></tr>
<tr><td><code id="qqcnorm_+3A_...">...</code></td>
<td>
<p>additional graphical parameter passed to &lsquo;<code>plot</code>&rsquo;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>&lsquo;<code>qqcnorm</code>&rsquo; produces a censored normal QQ plot, that is, a graphical method for comparing the empirical distribution of a given response variable (specified by the argument <code>which</code>) to the censored Gaussian distribution, which is defined as:
</p>

<table>
<tr>
 <td style="text-align: left;">
<code class="reqn">\Phi((lo - \mu)/\sigma)</code> </td><td style="text-align: center;"> if </td><td style="text-align: left;"> <code class="reqn">y\le lo</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code class="reqn">\phi((y - \mu)/\sigma) / \sigma</code> </td><td style="text-align: center;"> if </td><td style="text-align: left;"> <code class="reqn">lo &lt; y &lt; up</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code class="reqn">1 - \Phi((up - \mu)/\sigma)</code> </td><td style="text-align: center;"> if </td><td style="text-align: left;"> <code class="reqn">y\ge up</code>
</td>
</tr>

</table>

<p>where <code class="reqn">\phi</code> and <code class="reqn">\Phi</code> are the probability density function and the cumulative ditribution of the standard normal distribution, respectively, whereas <code class="reqn">lo</code> and <code class="reqn">up</code> are the lower and upper censoring values, respectively.
</p>
<p>The comparison is done by plotting the empirical quantiles (<code class="reqn">y</code>-coordinate) against the theoretical quantiles (<code class="reqn">x</code>-coordinate) of the censored Gaussian distribution, which are defined as follows:
</p>

<table>
<tr>
 <td style="text-align: left;">
<code class="reqn">lo</code> </td><td style="text-align: center;"> if </td><td style="text-align: left;"> <code class="reqn">p \le \Phi((lo - \mu)/\sigma)</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code class="reqn">\mu + \sigma \Phi^{-1}(p)</code> </td><td style="text-align: center;"> if </td><td style="text-align: left;"> <code class="reqn">\Phi((lo - \mu)/\sigma) &lt; p &lt; 1 - \Phi((up - \mu)/\sigma)</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code class="reqn">up</code> </td><td style="text-align: center;"> if </td><td style="text-align: left;"> <code class="reqn">p \ge 1 - \Phi((up - \mu)/\sigma)</code> 
</td>
</tr>

</table>

<p>where <code class="reqn">p\in(0, 1)</code>. If the two distributions are similar, the points will approximately lie on the line <code class="reqn">y = x</code>. If the distributions are linearly related, the points will approximately lie on a line, but not necessarily on the line <code class="reqn">y = x</code>. In order to evaluate if the proportions of left/right-censored values are similar to the Gaussian tail probabilities, points corresponding to the censored values are plotted using a  specific symbol (see argument &lsquo;<code>plot.pch</code>&rsquo;), colour (see argument &lsquo;<code>plot.col</code>&rsquo;) and size (see argument &lsquo;<code>plot.cex</code>&rsquo;).
</p>
<p>Finally, maximum likelihood estimates of the marginal parameters <code class="reqn">\mu</code> and <code class="reqn">\sigma</code> are computed as described in &lsquo;<code><a href="#topic+datacggm">datacggm</a></code>&rsquo; and can be extracted from an <span class="rlang"><b>R</b></span> of class &lsquo;<code>datacggm</code>&rsquo; by using the functions &lsquo;<code><a href="#topic+ColMeans">ColMeans</a></code>&rsquo; and &lsquo;<code><a href="#topic+ColVars">ColVars</a></code>&rsquo;, respectively.
</p>


<h3>Value</h3>

<p>A named list is silently returned. Each element of the list contains a two-columns matrix; first columns (named &lsquo;<code>x</code>&rsquo;) contains the theoretical quantiles whereas second columns (named &lsquo;<code>y</code>&rsquo;) contains the empirical quantiles.
</p>


<h3>Author(s)</h3>

<p>Gianluca Sottile (<a href="mailto:gianluca.sottile@unipa.it">gianluca.sottile@unipa.it</a>)</p>


<h3>References</h3>

<p>Augugliaro L., Sottile G., Wit E.C., and Vinciotti V. (2023) &lt;<a href="https://doi.org/10.18637/jss.v105.i01">doi:10.18637/jss.v105.i01</a>&gt;.
cglasso: An R Package for Conditional Graphical Lasso Inference with Censored and Missing Values.
<em>Journal of Statistical Software</em> <b>105</b>(1), 1&ndash;58.
</p>
<p>Augugliaro, L., Sottile, G., and Vinciotti, V. (2020) &lt;<a href="https://doi.org/10.1007/s11222-020-09945-7">doi:10.1007/s11222-020-09945-7</a>&gt;.
The conditional censored graphical lasso estimator.
<em>Statistics and Computing</em> <b>30</b>, 1273&ndash;1289.
</p>
<p>Augugliaro, L., Abbruzzo, A., and Vinciotti, V. (2020) &lt;<a href="https://doi.org/10.1093/biostatistics/kxy043">doi:10.1093/biostatistics/kxy043</a>&gt;.
<code class="reqn">\ell_1</code>-Penalized censored Gaussian graphical model.
<em>Biostatistics</em> <b>21</b>, e1&ndash;e16.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+datacggm">datacggm</a></code>, <code><a href="#topic+rcggm">rcggm</a></code>, <code><a href="#topic+ColMeans">ColMeans</a></code>, <code><a href="#topic+ColVars">ColVars</a></code> and <code><a href="#topic+hist.datacggm">hist.datacggm</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)

# a dataset from a right-censored Gaussian graphical model
n &lt;- 1000L
p &lt;- 10L
Y &lt;- matrix(rnorm(n * p), n, p)
up &lt;- 1
Y[Y &gt;= up] &lt;- up
Z &lt;- datacggm(Y = Y, up = up)
qqcnorm(Z, max.plot = 4L)

# a dataset from a  conditional censored Gaussian graphical model
n &lt;- 1000L
p &lt;- 10L
q &lt;- 2
Y &lt;- matrix(rnorm(n * p), n, p)
up &lt;- 1
lo &lt;- -1
Y[Y &gt;= up] &lt;- up
Y[Y &lt;= lo] &lt;- lo
X &lt;- matrix(rnorm(n * q), n, q)
Z &lt;- datacggm(Y = Y, lo = lo, up = up, X = X)
qqcnorm(Z, max.plot = 4L)
</code></pre>

<hr>
<h2 id='rcggm'>Simulate Data from a Conditional Gaussian Graphical Model with Censored and/or Missing Values</h2><span id='topic+rcggm'></span>

<h3>Description</h3>

<p>&lsquo;<code>rcggm</code>&rsquo; function is used to produce one or more samples from a conditional Gaussian graphical model with censored and/or missing values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rcggm(n, p, b0, X, B, Sigma, probl, probr, probna, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rcggm_+3A_n">n</code></td>
<td>
<p>the number of samples required (optional, see below for a description).</p>
</td></tr>
<tr><td><code id="rcggm_+3A_p">p</code></td>
<td>
<p>the number of response variables (optional, see below for a description).</p>
</td></tr>
<tr><td><code id="rcggm_+3A_b0">b0</code></td>
<td>
<p>a vector of length <code>p</code> used to specify the intercepts. Default is a zero vector of length <code>p</code> (optional, see below for a description).</p>
</td></tr>
<tr><td><code id="rcggm_+3A_x">X</code></td>
<td>
<p>a matrix of dimension <code class="reqn">n\times q</code> used to model the expected values of the response variables (optional, see below for a description).</p>
</td></tr>
<tr><td><code id="rcggm_+3A_b">B</code></td>
<td>
<p>a matrix of dimension <code class="reqn">q\times p</code> used to specify the regression coefficients. If <code>X</code> is missing then <code>B</code> is set equal to <code>NULL</code> (optional, see below for a description).</p>
</td></tr>
<tr><td><code id="rcggm_+3A_sigma">Sigma</code></td>
<td>
<p>a positive-definite symmetric matrix specifying the covariance matrix of the response variables. Default is the identity matrix (optional, see below for a description).</p>
</td></tr>
<tr><td><code id="rcggm_+3A_probl">probl</code></td>
<td>
<p>a vector giving the probabilities that the response variables are left-censored.</p>
</td></tr>
<tr><td><code id="rcggm_+3A_probr">probr</code></td>
<td>
<p>a vector giving the probabilities that the response variables are right-censored.</p>
</td></tr>
<tr><td><code id="rcggm_+3A_probna">probna</code></td>
<td>
<p>the probability that a response value is missing-at-random. By default &lsquo;<code>probna</code>&rsquo; is set equal to zero.</p>
</td></tr>
<tr><td><code id="rcggm_+3A_...">...</code></td>
<td>
<p>further arguments passed to the function &lsquo;<code>mvrnorm</code>&rsquo;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>&lsquo;The <code>rcggm</code>&rsquo; function simulates a dataset from a conditional Gaussian graphical model with censored or missing values and returns an object of class &lsquo;<code>datacggm</code>&rsquo;. Censoring values are implicitly specified using arguments <code>probl</code> and <code>probr</code>, that is, <code>lo</code> and <code>up</code> are computed in such a way that the average probabilities of left and right censoring are equal to <code>probl</code> and <code>probr</code>, respectively. Missing-at-random values are simulated using a Bernoulli distribution with probability <code>probna</code>.
</p>
<p>The dataset is simulated through the following steps:
</p>

<ol>
<li><p> lower and upper censoring values (<code>lo</code> and <code>up</code>) are computed according to the arguments <code>probl</code> and <code>probr</code>;
</p>
</li>
<li><p> The function <code><a href="MASS.html#topic+mvrnorm">mvrnorm</a></code> is used to simulate one or more samples from the multivariate Gaussian distribution specified by the arguments <code>b0</code>, <code>X</code>, <code>B</code> and <code>Sigma</code>;
</p>
</li>
<li><p> The response values that are outside of the interval <code>[lo, up]</code> are replaced with the corresponding censoring values;
</p>
</li>
<li><p> if <code>probna</code> is greater than zero, then missing-at-random values are simulated using a Bernoulli distribution with probability <code>probna</code>.
</p>
</li></ol>
 

<table>
<tr>
 <td style="text-align: center;">
Model </td><td style="text-align: center;"> <code>n</code> </td><td style="text-align: center;"> <code>p</code> </td><td style="text-align: center;"> <code>b0</code> </td><td style="text-align: center;"> <code>X</code> </td><td style="text-align: center;"> <code>B</code> </td><td style="text-align: center;"> <code>Sigma</code> </td><td style="text-align: left;"> Gaussian distribution </td>
</tr>
<tr>
 <td style="text-align: center;">
1 </td><td style="text-align: center;"> <code>x</code> </td><td style="text-align: center;"> <code>x</code> </td><td style="text-align: center;">           </td><td style="text-align: center;">          </td><td style="text-align: center;">          </td><td style="text-align: center;">              </td><td style="text-align: left;"> <code class="reqn">Y\sim N(0, I)</code> </td>
</tr>
<tr>
 <td style="text-align: center;">
2 </td><td style="text-align: center;"> <code>x</code> </td><td style="text-align: center;">          </td><td style="text-align: center;">           </td><td style="text-align: center;">          </td><td style="text-align: center;">          </td><td style="text-align: center;"> <code>x</code>     </td><td style="text-align: left;"> <code class="reqn">Y\sim N(0, \Sigma)</code> </td>
</tr>
<tr>
 <td style="text-align: center;">
3 </td><td style="text-align: center;"> <code>x</code> </td><td style="text-align: center;">          </td><td style="text-align: center;"> <code>x</code>  </td><td style="text-align: center;">          </td><td style="text-align: center;">          </td><td style="text-align: center;">              </td><td style="text-align: left;"> <code class="reqn">Y\sim N(b0, I)</code> </td>
</tr>
<tr>
 <td style="text-align: center;">
4 </td><td style="text-align: center;"> <code>x</code> </td><td style="text-align: center;">          </td><td style="text-align: center;"> <code>x</code>  </td><td style="text-align: center;">          </td><td style="text-align: center;">          </td><td style="text-align: center;"> <code>x</code>     </td><td style="text-align: left;"> <code class="reqn">Y\sim N(b0, \Sigma)</code> </td>
</tr>
<tr>
 <td style="text-align: center;">
5 </td><td style="text-align: center;">          </td><td style="text-align: center;">          </td><td style="text-align: center;">           </td><td style="text-align: center;"><code>x</code>  </td><td style="text-align: center;"> <code>x</code> </td><td style="text-align: center;">              </td><td style="text-align: left;"> <code class="reqn">Y\sim N(XB, I)</code> </td>
</tr>
<tr>
 <td style="text-align: center;">
6 </td><td style="text-align: center;">          </td><td style="text-align: center;">          </td><td style="text-align: center;">           </td><td style="text-align: center;"><code>x</code>  </td><td style="text-align: center;"> <code>x</code> </td><td style="text-align: center;"> <code>x</code>     </td><td style="text-align: left;"> <code class="reqn">Y\sim N(XB, \Sigma)</code> </td>
</tr>
<tr>
 <td style="text-align: center;">
7 </td><td style="text-align: center;">          </td><td style="text-align: center;">          </td><td style="text-align: center;"> <code>x</code>  </td><td style="text-align: center;"><code>x</code>  </td><td style="text-align: center;"> <code>x</code> </td><td style="text-align: center;">              </td><td style="text-align: left;"> <code class="reqn">Y\sim N(b0 + XB, I)</code> </td>
</tr>
<tr>
 <td style="text-align: center;">
8 </td><td style="text-align: center;">          </td><td style="text-align: center;">          </td><td style="text-align: center;"> <code>x</code>  </td><td style="text-align: center;"><code>x</code>  </td><td style="text-align: center;"> <code>x</code> </td><td style="text-align: center;"> <code>x</code>     </td><td style="text-align: left;"> <code class="reqn">Y\sim N(b0 + XB, \Sigma)</code>
</td>
</tr>

</table>

<p>The previous table sums up the default setting of the multivariate Gaussian distribution used in step 2 (specified arguments are marked by the symbol &lsquo;<code>x</code>&rsquo;). See also below for some examples.
</p>


<h3>Value</h3>

<p><code>rcggm</code> returns an object of class &lsquo;<code>datacggm</code>&rsquo;. See <code><a href="#topic+datacggm">datacggm</a></code> for further details.
</p>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>References</h3>

<p>Augugliaro L., Sottile G., Wit E.C., and Vinciotti V. (2023) &lt;<a href="https://doi.org/10.18637/jss.v105.i01">doi:10.18637/jss.v105.i01</a>&gt;.
cglasso: An R Package for Conditional Graphical Lasso Inference with Censored and Missing Values.
<em>Journal of Statistical Software</em> <b>105</b>(1), 1&ndash;58.
</p>
<p>Augugliaro, L., Sottile, G., and Vinciotti, V. (2020a) &lt;<a href="https://doi.org/10.1007/s11222-020-09945-7">doi:10.1007/s11222-020-09945-7</a>&gt;.
The conditional censored graphical lasso estimator.
<em>Statistics and Computing</em> <b>30</b>, 1273&ndash;1289.
</p>
<p>Augugliaro, L., Abbruzzo, A., and Vinciotti, V. (2020b) &lt;<a href="https://doi.org/10.1093/biostatistics/kxy043">doi:10.1093/biostatistics/kxy043</a>&gt;.
<code class="reqn">\ell_1</code>-Penalized censored Gaussian graphical model.
<em>Biostatistics</em> <b>21</b>, e1&ndash;e16.</p>


<h3>See Also</h3>

<p><code><a href="#topic+datacggm">datacggm</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)

n &lt;- 100
p &lt;- 3
q &lt;- 2
b0 &lt;- rep(1, p)
X &lt;- matrix(rnorm(n * q), n, q)
B &lt;- matrix(rnorm(q * p), q, p)
Sigma &lt;- outer(1:p, 1:p, function(i, j) 0.3^abs(i - j))
probl &lt;- 0.05
probr &lt;- 0.05
probna &lt;- 0.05

# Model 1: Y ~ N(0, I)
Z &lt;- rcggm(n = n, p = p, probl = probl, probr = probr, probna = probna)
summary(Z)

# Model 2: Y ~ N(0, Sigma)
Z &lt;- rcggm(n = n, Sigma = Sigma, probl = probl, probr = probr, probna = probna)
summary(Z)

# Model 3: Y ~ N(b0, I)
Z &lt;- rcggm(n = n, b0 = b0, probl = probl, probr = probr, probna = probna)
summary(Z)

# Model 4: Y ~ N(b0, Sigma)
Z &lt;- rcggm(n = n, b0 = b0, Sigma = Sigma, probl = probl, probr = probr, 
           probna = probna)
summary(Z)

# Model 5: Y ~ N(XB, I)
Z &lt;- rcggm(X = X, B = B, probl = probl, probr = probr, probna = probna)
summary(Z)

# Model 6: Y ~ N(XB, Sigma)
Z &lt;- rcggm(X = X, B = B, Sigma = Sigma, probl = probl, probr = probr, 
           probna = probna)
summary(Z)

# Model 7: Y ~ N(b0 + XB, I)
Z &lt;- rcggm(b0 = b0, X = X, B = B, probl = probl, probr = probr, probna = probna)
summary(Z)

# Model 8: Y ~ N(b0 + XB, Sigma)
Z &lt;- rcggm(b0 = b0, X = X, B = B, Sigma = Sigma, probl = probl, probr = probr, 
           probna = probna)
summary(Z)
</code></pre>

<hr>
<h2 id='residuals'>Extract Model Residuals</h2><span id='topic+residuals'></span><span id='topic+residuals.cglasso'></span>

<h3>Description</h3>

<p>Extracts model residuals from an <span class="rlang"><b>R</b></span> object inheriting class &lsquo;<code>cglasso</code>&rsquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cglasso'
residuals(object, type = c("observed", "working"), lambda.id, rho.id,
          drop = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="residuals_+3A_object">object</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object inheriting class &lsquo;<code>cglasso</code>&rsquo;, that is, the output of the model-fitting functions <code><a href="#topic+cglasso">cglasso</a></code> and <code><a href="#topic+cggm">cggm</a></code>.</p>
</td></tr>
<tr><td><code id="residuals_+3A_type">type</code></td>
<td>
<p>a description of the desired residuals (see section &lsquo;Details&rsquo;).</p>
</td></tr>
<tr><td><code id="residuals_+3A_lambda.id">lambda.id</code></td>
<td>
<p>a vector of integers used to specify the <code class="reqn">\lambda</code>-values.</p>
</td></tr>
<tr><td><code id="residuals_+3A_rho.id">rho.id</code></td>
<td>
<p>a vector of integers used to specify the <code class="reqn">\rho</code>-values.</p>
</td></tr>
<tr><td><code id="residuals_+3A_drop">drop</code></td>
<td>
<p>logical. Dimensions can only be dropped if their extent is one.</p>
</td></tr>
<tr><td><code id="residuals_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The accessor function &lsquo;<code>residuals</code>&rsquo; returns an array storing the &lsquo;observed&rsquo; or &lsquo;working&rsquo; residuals, depending on the argument <code>type</code>.
</p>
<p>The &lsquo;observed&rsquo; residuals are defined as the difference between the observed response values and the corresponding fitted expected values. For missing and censored response values, the &lsquo;observed&rsquo; residuals are set equal to &lsquo;<code>NA</code>&rsquo;.
</p>
<p>The &lsquo;working&rsquo; residuals are obtained as a byproduct of the EM-algorithm used to fit the model. In the E-step, the algorithm computes the &lsquo;working&rsquo; response matrix, that is, the response matrix with missing values replaced by the conditional expected values of the multivariate normal distribution and censored values replaced by the conditional expected values of the multivariate truncated normal distribution. The &lsquo;working&rsquo; residuals are defined as the difference between &lsquo;working&rsquo; responses and fitted expected values. 
</p>


<h3>Value</h3>

<p>The accessor function &lsquo;<code>residuals</code>&rsquo; returns an array storing the desired residuals.
</p>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>See Also</h3>

<p>Model-fitting functions <code><a href="#topic+cglasso">cglasso</a></code>, <code><a href="#topic+cggm">cggm</a></code> and the accessor functions <code><a href="#topic+coef.cglasso">coef.cglasso</a></code>, <code><a href="#topic+fitted.cglasso">fitted.cglasso</a></code>, <code><a href="#topic+predict.cglasso">predict.cglasso</a></code> and <code><a href="#topic+impute">impute</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
# Y ~ N(0, Sigma) and probability of left/right censored values equal to 0.05
n &lt;- 1000L
p &lt;- 3L
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, Sigma = Sigma, probl = 0.05, probr = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
residuals(out, type = "observed", rho.id = 3L, drop = TRUE)
residuals(out, type = "working", rho.id = 3L, drop = TRUE)

# Y ~ N(b0 + XB, Sigma)  and probability of left/right censored values equal to 0.05
n &lt;- 1000L
p &lt;- 3L
q &lt;- 2
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probl = 0.05, probr = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
residuals(out, type = "observed", lambda.id = 3L, rho.id = 3L, drop = TRUE)
residuals(out, type = "working", lambda.id = 3L, rho.id = 3L, drop = TRUE)
</code></pre>

<hr>
<h2 id='rowNames+20+2B+20colNames'>Row and Column Names of a &ldquo;datacggm&rdquo; Object</h2><span id='topic+rowNames'></span><span id='topic+colNames'></span><span id='topic+rowNames+3C-'></span><span id='topic+colNames+3C-'></span>

<h3>Description</h3>

<p>Retrieve or set the row or column names of a &ldquo;datacggm&rdquo; object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rowNames(x)
rowNames(x) &lt;- value

colNames(x)
colNames(x) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rowNames+2B20+2B2B+2B20colNames_+3A_x">x</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>datacggm</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="rowNames+2B20+2B2B+2B20colNames_+3A_value">value</code></td>
<td>
<p>a named list with elements &lsquo;<code>X</code>&rsquo; and &lsquo;<code>Y</code>&rsquo;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>datacggm</code>&rsquo;, <code>rowNames</code> (<code>colNames</code>) retrieves or set <code>rownames</code> (<code>colnames</code>) for matrices <code>Y</code> and <code>X</code>. See below for some examples.
</p>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>See Also</h3>

<p><code><a href="#topic+datacggm">datacggm</a></code>, <code><a href="#topic+rcggm">rcggm</a></code> and the method function <code><a href="#topic+dimnames">dimnames</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
# a dataset from a censored Gaussian graphical model
n &lt;- 100L
p &lt;- 3L
Z &lt;- rcggm(n = n, p = p, probl = 0.05, probr = 0.05)

rowNames(Z)
rowNames(Z) &lt;- list(Y = paste0("i", seq_len(n))) 
# the same as rowNames(Z)$Y &lt;- paste0("i", seq_len(n))
rowNames(Z)

colNames(Z)
colNames(Z) &lt;- list(Y = paste("Y", 1:p, sep = ":"))
# the same as colNames(Z)$Y &lt;- paste("Y", 1:p, sep = ":")
colNames(Z)

# a dataset from a  conditional censored Gaussian graphical model
n &lt;- 100L
p &lt;- 3L
q &lt;- 2
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, probl = 0.05, probr = 0.05)

rowNames(Z)$Y &lt;- paste0("i", seq_len(n))
rowNames(Z)$X &lt;- paste0("i", seq_len(n))
dimnames(Z)

colNames(Z)$Y &lt;- paste("Y", 1:p, sep = ":")
colNames(Z)$X &lt;- paste("X", 1:q, sep = ":")
dimnames(Z)
</code></pre>

<hr>
<h2 id='select_cglasso'>Model Selection for the Conditional Graphical Lasso Estimator</h2><span id='topic+select_cglasso'></span>

<h3>Description</h3>

<p>&lsquo;<code>select_cglasso</code>&rsquo; returns the optimal fitted model selected by a chosen measure of goodness-of-fit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select_cglasso(object, GoF = AIC, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="select_cglasso_+3A_object">object</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object of class <code>cglasso</code>.</p>
</td></tr>
<tr><td><code id="select_cglasso_+3A_gof">GoF</code></td>
<td>
<p>a valid goodness-of-fit function, such as <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code> or <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>, or an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>GoF</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="select_cglasso_+3A_...">...</code></td>
<td>
<p>further arguments passed to the chosen goodness-of-fit function (argument &lsquo;<code>GoF</code>&rsquo;).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>select_cglasso</code> evaluates the goodness-of-fit of the models fitted by <code>cglasso</code> and extracts the selected model.
</p>
<p>Model evaluation can be made in two ways. The easiest way is to use a valid goodness-of-fit function, such as <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code> or <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>. In this case, further arguments are passed to these functions by &lsquo;...&rsquo;. The second way consists on passing the output of a goodness-of-fit function, that is, an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>GoF</code>&rsquo;. Usually, this approach is preferable when the computation of the  chosen goodness-of-fit measure is time-consuming, such as when the sample size is small relative to the number of parameters and the <code>AIC</code> or <code>BIC</code> functions are used to evaluate a long sequence of fitted models. In these cases, we suggest the computation of several measures of goodness-of-fit in a preliminary step and then the use of the <code>select_cglasso</code> function in a subsequent step to select the optimal fitted model.
</p>


<h3>Value</h3>

<p>&lsquo;<code>select_cglasso</code>&rsquo; returns the optimal fitted model.
</p>


<h3>Author(s)</h3>

<p>Gianluca Sottile (<a href="mailto:gianluca.sottile@unipa.it">gianluca.sottile@unipa.it</a>)</p>


<h3>See Also</h3>

<p><code><a href="#topic+cglasso">cglasso</a></code>, <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code> and <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
# Y ~ N(0, Sigma) and probability of left/right censored values equal to 0.05
n &lt;- 100L
p &lt;- 3L
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, Sigma = Sigma, probl = 0.05, probr = 0.05)

out &lt;- cglasso(. ~ ., data = Z)
select_cglasso(out)                                    # models selection by AIC
select_cglasso(out, GoF = BIC)                         # models selection by BIC
select_cglasso(out, GoF = BIC, mle = TRUE, g = 0.5)    # models selection by eBIC

# Y ~ N(b0 + XB, Sigma)  and probability of left/right censored values equal to 0.05
n &lt;- 100L
p &lt;- 3L
q &lt;- 2
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probl = 0.05, probr = 0.05)

out &lt;- cglasso(. ~ ., data = Z, lambda = 0.01)
select_cglasso(out)                                    # models selection by AIC
select_cglasso(out, GoF = BIC)                         # models selection by BIC
select_cglasso(out, GoF = BIC, mle = TRUE, g = 0.5)    # models selection by eBIC

out &lt;- cglasso(. ~ ., data = Z, rho = 0.01)
select_cglasso(out)                                    # models selection by AIC
select_cglasso(out, GoF = BIC)                         # models selection by BIC
select_cglasso(out, GoF = BIC, mle = TRUE, g = 0.5)    # models selection by eBIC

out &lt;- cglasso(. ~ ., data = Z)
select_cglasso(out)                                    # models selection by AIC
select_cglasso(out, GoF = BIC)                         # models selection by BIC
select_cglasso(out, GoF = BIC, mle = TRUE, g = 0.5)    # models selection by eBIC
</code></pre>

<hr>
<h2 id='ShowStructure'>Show Package Structure</h2><span id='topic+ShowStructure'></span>

<h3>Description</h3>

<p><code>ShowStructure</code> gives information about the package structure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ShowStructure(module = c("ALL", "DM", "MF", "MS", "NA"), description = TRUE,
              plot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ShowStructure_+3A_module">module</code></td>
<td>
<p>a description of the module for which the description is required.</p>
</td></tr>
<tr><td><code id="ShowStructure_+3A_plot">plot</code></td>
<td>
<p>logical. If <code>TRUE</code> a graph showing the package structure is plotted.</p>
</td></tr>
<tr><td><code id="ShowStructure_+3A_description">description</code></td>
<td>
<p>logical. If <code>TRUE</code> a description of the available function is printed on screen.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>ShowStructure</code> silently returns a named list with components:
</p>
<table>
<tr><td><code>description</code></td>
<td>
<p>a named list containing the description of the available functions.</p>
</td></tr>
<tr><td><code>graph</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>igraph</code>&rsquo; describing the package structure.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>

<hr>
<h2 id='summary.cglasso'>Summarizing cglasso and cggm Fits</h2><span id='topic+summary.cglasso'></span>

<h3>Description</h3>

<p>&lsquo;<code>summary</code>&rsquo; produces a summary of the models fitted using <code>cglasso</code> or <code>cggm</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cglasso'
summary(object, GoF = AIC, print.all = TRUE, digits = 3L, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.cglasso_+3A_object">object</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object inheriting class <code>cglasso</code>, that is, the output of the model-fitting functions <code><a href="#topic+cglasso">cglasso</a></code> and <code><a href="#topic+cggm">cggm</a></code>.</p>
</td></tr>
<tr><td><code id="summary.cglasso_+3A_gof">GoF</code></td>
<td>
<p>a valid goodness-of-fit function, such as <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code> or <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>, or an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>GoF</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="summary.cglasso_+3A_print.all">print.all</code></td>
<td>
<p>logical. Should all summary statistics be printed?</p>
</td></tr>
<tr><td><code id="summary.cglasso_+3A_digits">digits</code></td>
<td>
<p>the minimum number of significant digits to be used: see &lsquo;<a href="base.html#topic+print.default">print.default</a>&rsquo;.</p>
</td></tr>
<tr><td><code id="summary.cglasso_+3A_...">...</code></td>
<td>
<p>the <em>penalty</em> parameter passed to the goodness-of-fit function (argument &lsquo;<code>GoF</code>&rsquo;) or further arguments passed to &lsquo;<a href="base.html#topic+print.data.frame">print.data.frame</a>&rsquo; and &lsquo;<a href="base.html#topic+print.listof">print.listof</a>&rsquo;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>summary.cglasso</code> computes the summary statistics needed to evaluate the goodness-of-fit of the models fitted by <code>cglasso</code> or <code>cggm</code>. 
</p>
<p>Model evaluation can be made in two ways. The easiest way is to use a valid goodness-of-fit function, such as <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code> or <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>. In this case, further arguments can be passed to these functions by the argument &lsquo;...&rsquo;. The second way consists in passing the output of a goodness-of-fit function, that is, an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>GoF</code>&rsquo;. Usually, this approach is preferable when the computation of  the chosen goodness-of-fit measure is time-consuming, such as when the sample size is small relative to the number of parameters and the <code>AIC</code> or <code>BIC</code> functions are used to evaluate a sequence of fitted models. In these cases, we suggest the computation of several measures of goodness-of-fit in a preliminary step and then the use of <code>summary.cglasso</code> in a subsequent step to evaluate the sequence of fitted models.
</p>
<p>To improve the readability of the results, the output is divided into two sections. 
</p>
<p>The first section is structured in a sequence of tables showing, for each combination of the two tuning parameters &lsquo;<code>lambda</code>&rsquo; and &lsquo;<code>rho</code>&rsquo;, the number of estimated non-zero regression coefficients (&lsquo;<code>df.B</code>&rsquo;), the number of estimated non-zero partial correlation coefficients (&lsquo;<code>df.Tht</code>&rsquo;), the degrees-of-freedom (&lsquo;<code>df</code>&rsquo;) and the value of the goodness-of-fit measure used to evaluate the fitted models. To help the user with the identification of the optimal fitted model, the last column of each table reports the ranking of the models (where the optimal model is marked with the symbol &lsquo;<code>&lt;-</code>&rsquo;).
</p>
<p>The second section reports the summary statistics of the selected optimal model.
</p>


<h3>Value</h3>

<p>A named list with the following elements is silently returned:
</p>
<table>
<tr><td><code>table</code></td>
<td>
<p>data.frame containing the summary statistics used to evaluate the sequence of fitted models.</p>
</td></tr>
<tr><td><code>lambda.id</code></td>
<td>
<p>position of the optimal <code class="reqn">\lambda</code>-value identified by the chosen goodness-of-fit function.</p>
</td></tr>
<tr><td><code>rho.id</code></td>
<td>
<p>position of the optimal <code class="reqn">\rho</code>-value identified by the chosen goodness-of-fit function.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>See Also</h3>

<p><code><a href="#topic+cglasso">cglasso</a></code>, <code><a href="#topic+cggm">cggm</a></code>, <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code> and <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)

# Y ~ N(0, Sigma) and probability of left/right censored values equal to 0.05
n &lt;- 100L
p &lt;- 3L
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, Sigma = Sigma, probl = 0.05, probr = 0.05)

out &lt;- cglasso(. ~ ., data = Z)
summary(out)                                   # models evaluation by AIC
summary(out, GoF = BIC)                        # models evaluation by BIC
summary(out, GoF = BIC, mle = TRUE, g = 0.5)   # models evaluation by eBIC

# Y ~ N(b0 + XB, Sigma)  and probability of left/right censored values equal to 0.05
n &lt;- 100L
p &lt;- 3L
q &lt;- 2
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probl = 0.05, probr = 0.05)

out &lt;- cglasso(. ~ ., data = Z, lambda = 0.01)
summary(out)                                   # models evaluation by AIC
summary(out, GoF = BIC)                        # models evaluation by BIC
summary(out, GoF = BIC, mle = TRUE, g = 0.5)   # models evaluation by eBIC

out &lt;- cglasso(. ~ ., data = Z, rho = 0.01)
summary(out)                                   # models evaluation by AIC
summary(out, GoF = BIC)                        # models evaluation by BIC
summary(out, GoF = BIC, mle = TRUE, g = 0.5)   # models evaluation by eBIC

out &lt;- cglasso(. ~ ., data = Z)
summary(out)                                   # models evaluation by AIC
summary(out, GoF = BIC)                        # models evaluation by BIC
summary(out, GoF = BIC, mle = TRUE, g = 0.5)   # models evaluation by eBIC
</code></pre>

<hr>
<h2 id='summary.datacggm'>Summarizing Objects of Class &lsquo;<code>datacggm</code>&rsquo;</h2><span id='topic+summary.datacggm'></span>

<h3>Description</h3>

<p>The function &lsquo;<code>summary.datacggm</code>&rsquo; produces summaries of an object of class &lsquo;<code>datacggm</code>&rsquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'datacggm'
summary(object, n, quantile.type = 7L, digits = 3L, quote = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.datacggm_+3A_object">object</code></td>
<td>
<p>an object of class &lsquo;<code>datacggm</code>&rsquo; for which a summary is desired.</p>
</td></tr>
<tr><td><code id="summary.datacggm_+3A_n">n</code></td>
<td>
<p>maximum number of rows printed on video. By default all rows are printed.</p>
</td></tr>
<tr><td><code id="summary.datacggm_+3A_quantile.type">quantile.type</code></td>
<td>
<p>an integer between 1 and 9 selecting one of the nine quantile algorithms (see also <code><a href="stats.html#topic+quantile">quantile</a></code> function).</p>
</td></tr>
<tr><td><code id="summary.datacggm_+3A_digits">digits</code></td>
<td>
<p>integer, used for number formatting with &lsquo;<code><a href="base.html#topic+format">format</a>()</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="summary.datacggm_+3A_quote">quote</code></td>
<td>
<p>logical, indicating whether or not strings should be printed with surrounding quotes.</p>
</td></tr>
<tr><td><code id="summary.datacggm_+3A_...">...</code></td>
<td>
<p>further arguments passed to <code><a href="base.html#topic+print.listof">print.listof</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function &lsquo;<code>summary.datacggm</code>&rsquo; extends the results given by &lsquo;<code><a href="base.html#topic+summary.matrix">summary.matrix</a>()</code>&rsquo; taking into account the specific structure of an object of class &lsquo;<code>datacggm</code>&rsquo;. Summary statistics are printed out in two sections: the first section reports the summary statistics for the response matrix &lsquo;<code>Y</code>&rsquo;, whereas the second section is devoted to summarising the matrix of predictors &lsquo;<code>X</code>&rsquo;. 
</p>
<p>For each response variable, the first section reports the standard summary statistics computed using only the observed values, that is the entries of the response matrix whose status indicators are equals to zero (see &lsquo;<code><a href="#topic+event">event</a></code>&rsquo; for more details), together with the information about the lower and upper censoring values (columns named <code>Lower</code> and <code>Upper</code>) and the percentage of missing-at-random, left-censored and right-censored response values (columns named <code>NA%</code>, <code>LC%</code> and <code>RC%</code>, respectively).
</p>
<p>The second section is divided into two subsections reporting the main summary statistics computed for numeric and categorical predictors, respectively.
</p>
<p>The two sections are printed out in such a way that the readability of the summary statistics is improved in a high-dimensional setting, that is, when the number of predictors and response variables exceeds the sample size. 
</p>


<h3>Value</h3>

<p>&lsquo;<code>summary.dataccgm</code>&rsquo; returns a named list with components
</p>
<table>
<tr><td><code>Y</code></td>
<td>
<p>a matrix with class &lsquo;<code>table</code>&rsquo; obtained by computing the summary measures on the response variables.</p>
</td></tr>
<tr><td><code>X.numeric</code></td>
<td>
<p>a matrix with class &lsquo;<code>table</code>&rsquo; obtained by computing the summary measures on the numeric predictors.</p>
</td></tr>
<tr><td><code>X.categorical</code></td>
<td>
<p>a list containing the summary measures on the categorical predictors.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>References</h3>

<p>Augugliaro L., Sottile G., Wit E.C., and Vinciotti V. (2023) &lt;<a href="https://doi.org/10.18637/jss.v105.i01">doi:10.18637/jss.v105.i01</a>&gt;.
cglasso: An R Package for Conditional Graphical Lasso Inference with Censored and Missing Values.
<em>Journal of Statistical Software</em> <b>105</b>(1), 1&ndash;58.
</p>
<p>Augugliaro, L., Sottile, G., and Vinciotti, V. (2020a) &lt;<a href="https://doi.org/10.1007/s11222-020-09945-7">doi:10.1007/s11222-020-09945-7</a>&gt;.
The conditional censored graphical lasso estimator.
<em>Statistics and Computing</em> <b>30</b>, 1273&ndash;1289.
</p>
<p>Augugliaro, L., Abbruzzo, A., and Vinciotti, V. (2020b) &lt;<a href="https://doi.org/10.1093/biostatistics/kxy043">doi:10.1093/biostatistics/kxy043</a>&gt;.
<code class="reqn">\ell_1</code>-Penalized censored Gaussian graphical model.
<em>Biostatistics</em> <b>21</b>, e1&ndash;e16.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+datacggm">datacggm</a></code>, <code><a href="#topic+rcggm">rcggm</a></code> and <code><a href="#topic+event">event</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)

# case 1: Y ~ N(0, Sigma) 
# 1. probability of left/right censored values equal to 0.05
# 2. probability of missing-at-random equals to 0.05
n &lt;- 50L
p &lt;- 100L
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, Sigma = Sigma, probl = 0.05, probr = 0.05, probna = 0.05)
summary(Z, max = n * p)

# case 2: Y ~ N(b0 + XB, Sigma) and
# 1. probability of left/right censored values equal to 0.05
# 2. probability of missing-at-random equals to 0.05
n &lt;- 50L
p &lt;- 100L
q &lt;- 100L
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probl = 0.05, probr = 0.05, 
           probna = 0.05)
summary(Z)
summary(Z, max = n * p)
</code></pre>

<hr>
<h2 id='to_graph'>Create Graphs from cglasso or cggm Objects</h2><span id='topic+to_graph'></span>

<h3>Description</h3>

<p>&lsquo;<code>to_graph</code>&rsquo; returns a named list of graphs using the results of an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>cglasso</code>&rsquo; or &lsquo;<code>cggm</code>&rsquo;. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>to_graph(object, GoF = AIC, lambda.id, rho.id, weighted = FALSE, simplify = TRUE,
         ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="to_graph_+3A_object">object</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object inheriting class &lsquo;<code>cglasso</code>&rsquo;, that is, the output of the model-fitting functions <code><a href="#topic+cglasso">cglasso</a></code> and <code><a href="#topic+cggm">cggm</a></code>.</p>
</td></tr>
<tr><td><code id="to_graph_+3A_gof">GoF</code></td>
<td>
<p>a valid goodness-of-fit function, such as <code><a href="#topic+AIC.cglasso">AIC.cglasso</a></code> or <code><a href="#topic+BIC.cglasso">BIC.cglasso</a></code>, or an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>GoF</code>&rsquo;.</p>
</td></tr>
<tr><td><code id="to_graph_+3A_lambda.id">lambda.id</code></td>
<td>
<p>optional. If <code>object</code> has class &lsquo;<code>cglasso</code>&rsquo;, this argument is an integer used to identify a specific fitted cglasso model, otherwise (i.e. if the <code>object</code> has class &lsquo;<code>cggm</code>&rsquo;) this argument can be omitted. See section &lsquo;Details&rsquo; for more details.</p>
</td></tr>
<tr><td><code id="to_graph_+3A_rho.id">rho.id</code></td>
<td>
<p>optional. If <code>object</code> has class &lsquo;<code>cglasso</code>&rsquo;, this argument is an integer used to identify a specific fitted cglasso model, otherwise (i.e. if <code>object</code> has class &lsquo;<code>cggm</code>&rsquo;) this argument can be omitted. See section &lsquo;Details&rsquo; for more details.</p>
</td></tr>
<tr><td><code id="to_graph_+3A_weighted">weighted</code></td>
<td>
<p>logical. Should weighted graphs be created? Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="to_graph_+3A_simplify">simplify</code></td>
<td>
<p>logical. Should isolated vertices be removed from the graph? Default is <code>TRUE</code>, i.e., isolated vertices are removed.</p>
</td></tr>
<tr><td><code id="to_graph_+3A_...">...</code></td>
<td>
<p>further arguments passed to the chosen goodness-of-fit function (argument &lsquo;<code>GoF</code>&rsquo;).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>&lsquo;<code>to_graph</code>&rsquo; returns a named list of graphs using the results of an <span class="rlang"><b>R</b></span> object of class &lsquo;<code>cglasso</code>&rsquo; or &lsquo;<code>cggm</code>&rsquo;. 
</p>
<p>If <code>object</code> has class &lsquo;<code>cglasso</code>&rsquo;, then the goodness-of-fit function passed through the argument <code>GoF</code> is used to identify the adjacency matrix (<code>object$InfoStructure$Adj_yy</code>) describing the undirected edges among the <code class="reqn">p</code> response variables. If the model is fitted using <code class="reqn">q</code> predictors, then the matrix  describing the effects of the predictors onto the response variables (see object$InfoStructure$Adj_xy) is also returned. Finally, these matrices are used to return an undirected and directed graph. Opionally, the user can identify a specific fitted model using the arguments <code>lambda.id</code> and <code>rho.id</code>.
</p>
<p>If <code>object</code> has class &lsquo;<code>cggm</code>&rsquo;, then <code>GoF</code>, <code>lambda.id</code> and <code>rho.id</code> can be omitted.
</p>
<p>If argument weighted is set equal to &lsquo;<code>TRUE</code>&rsquo;, then the estimated precision matrix and, if available, the estimated regression coefficient matrix are used to return weighted graphs. In this case, edges associated with positive estimates are shown using a solid line. Otherwise, a dashed line is used.
</p>


<h3>Value</h3>

<p>&lsquo;<code>to_graph</code>&rsquo; returns an <span class="rlang"><b>R</b></span> object of S3 class &ldquo;<code>cglasso2igraph</code>&rdquo;, i.e., a named list containing the
following components:
</p>
<table>
<tr><td><code>Gyy</code></td>
<td>
<p>an undirected graph representing the conditional dependence structure among the <code class="reqn">p</code> response variables.</p>
</td></tr>
<tr><td><code>Gxy</code></td>
<td>
<p>a directed graph representing the effetcs of the <code class="reqn">q</code> predictors onto the <code class="reqn">p</code> response variables.</p>
</td></tr>
</table>
<p>Each component is an <span class="rlang"><b>R</b></span> object of class <code><a href="igraph.html#topic+igraph">igraph</a></code>.
</p>


<h3>Author(s)</h3>

<p>Luigi Augugliaro (<a href="mailto:luigi.augugliaro@unipa.it">luigi.augugliaro@unipa.it</a>)</p>


<h3>See Also</h3>

<p><code><a href="#topic+cglasso">cglasso</a></code>, <code><a href="#topic+cggm">cggm</a></code> and <code><a href="#topic+plot.cglasso2igraph">plot.cglasso2igraph</a></code>. For more details about the object of class &lsquo;<code>igraph</code>&rsquo;, the interested reader is referred to the package <span class="pkg"><a href="igraph.html#topic+igraph">igraph</a></span>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
# Y ~ N(0, Sigma) and probability of left/right censored values equal to 0.05
n &lt;- 100L
p &lt;- 3L
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, Sigma = Sigma, probl = 0.05, probr = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
out.graph &lt;- to_graph(out)
out.graph

# Y ~ N(b0 + XB, Sigma)  and probability of left/right censored values equal to 0.05
n &lt;- 100L
p &lt;- 3L
q &lt;- 2L
b0 &lt;- runif(p)
B &lt;- matrix(runif(q * p), nrow = q, ncol = p)
X &lt;- matrix(rnorm(n * q), nrow = n, ncol = q)
rho &lt;- 0.3
Sigma &lt;- outer(1L:p, 1L:p, function(i, j) rho^abs(i - j))
Z &lt;- rcggm(n = n, b0 = b0, X = X, B = B, Sigma = Sigma, probl = 0.05, probr = 0.05)
out &lt;- cglasso(. ~ ., data = Z)
out.graph &lt;- to_graph(out, lambda.id = 3, rho.id = 3, weighted = TRUE)
out.graph
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
