<!DOCTYPE html><html lang="en"><head><title>Help for package changepointGA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {changepointGA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AMOCcrossover'><p>Average crossover operator to produce offspring for AMOC problem</p></a></li>
<li><a href='#AMOCmutation'><p>Jump mutation operator to produce offspring for AMOC problem</p></a></li>
<li><a href='#AMOCpopulation'><p>Random population initialization for AMOC problem</p></a></li>
<li><a href='#AMOCselection'><p>The parents selection genetic algorithm operator for AMOC problem</p></a></li>
<li><a href='#ARIMA.BIC.Order'><p>Calculating BIC for Multiple changepoint detection with model order selection</p></a></li>
<li><a href='#arrayaCGH'><p>Glioblastoma array aCGH data</p></a></li>
<li><a href='#AtlantaTemperature'><p>Atlanta airport yearly average temperatures data</p></a></li>
<li><a href='#BinSearch.BIC'><p>Example function: Calculating BIC for AR(1) model</p></a></li>
<li><a href='#cptDist'><p>Comparing multiple changepoint configurations by pairwise distance</p></a></li>
<li><a href='#GA'><p>Genetic algorithm</p></a></li>
<li><a href='#GA_param'>
<p>GA_param</p></a></li>
<li><a href='#IslandGA'><p>Island model based genetic algorithm</p></a></li>
<li><a href='#IslandGA_param'>
<p>IslandGA_param</p></a></li>
<li><a href='#mutation'><p>The default mutation operator in genetic algorithm</p></a></li>
<li><a href='#operators'>
<p>operators</p></a></li>
<li><a href='#random_population'><p>Random population initialization</p></a></li>
<li><a href='#selection_linearrank'><p>The default parents selection genetic algorithm operator</p></a></li>
<li><a href='#selectTau'><p>Randomly select the chromosome</p></a></li>
<li><a href='#ts.sim'><p>Time series simulation with changepoint effects</p></a></li>
<li><a href='#TsPlotCheck'><p>Plot the simulated time series</p></a></li>
<li><a href='#uniformcrossover'><p>Uniform crossover to produce offsprings</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Changepoint Detection via Modified Genetic Algorithm</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2025-01-21</td>
</tr>
<tr>
<td>Description:</td>
<td>The Genetic Algorithm (GA) is used to perform changepoint analysis in time series data. The package also includes an extended island version of GA, as described in Lu, Lund, and Lee (2010, &lt;<a href="https://doi.org/10.1214%2F09-AOAS289">doi:10.1214/09-AOAS289</a>&gt;). By mimicking the principles of natural selection and evolution, GA provides a powerful stochastic search technique for solving combinatorial optimization problems. In 'changepointGA', each chromosome represents a changepoint configuration, including the number and locations of changepoints, hyperparameters, and model parameters. The package employs genetic operators—selection, crossover, and mutation—to iteratively improve solutions based on the given fitness (objective) function. Key features of 'changepointGA' include encoding changepoint configurations in an integer format, enabling dynamic and simultaneous estimation of model hyperparameters, changepoint configurations, and associated parameters. The detailed algorithmic implementation can be found in the package vignettes and in the paper of Li (2024, &lt;<a href="https://doi.org/10.48550%2FarXiv.2410.15571">doi:10.48550/arXiv.2410.15571</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>foreach, doParallel, Rcpp, RcppArmadillo, clue, stats</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/mli171/changepointGA">https://github.com/mli171/changepointGA</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/mli171/changepointGA/issues">https://github.com/mli171/changepointGA/issues</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-13 21:49:58 UTC; mo</td>
</tr>
<tr>
<td>Author:</td>
<td>Mo Li [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mo Li &lt;mo.li@louisiana.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-17 11:30:16 UTC</td>
</tr>
</table>
<hr>
<h2 id='AMOCcrossover'>Average crossover operator to produce offspring for AMOC problem</h2><span id='topic+AMOCcrossover'></span>

<h3>Description</h3>

<p>In this crossover operator designed for AMOC problem, the new child is
produced by taking the average of the changepoint locations from <code>dad</code>
and <code>mom</code> and round to an integer. Note, every chromosome has at most
one candidate changepoint location.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AMOCcrossover(mom, dad, p.range = NULL, minDist, lmax, N)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AMOCcrossover_+3A_mom">mom</code></td>
<td>
<p>Among two selected individuals, <code>mom</code> represents the selected
chromosome representation with lower fitness function value.</p>
</td></tr>
<tr><td><code id="AMOCcrossover_+3A_dad">dad</code></td>
<td>
<p>Among two selected individuals, <code>dad</code> represents the selected
chromosome representation with larger fitness function value.</p>
</td></tr>
<tr><td><code id="AMOCcrossover_+3A_p.range">p.range</code></td>
<td>
<p>The default value is <code>NULL</code>. If there is no requirement
on model order selection, such an auxiliary argument is needed for <code>GA</code>
and <code>IslandGA</code> functions.</p>
</td></tr>
<tr><td><code id="AMOCcrossover_+3A_mindist">minDist</code></td>
<td>
<p>The minimum length between two adjacent changepoints.</p>
</td></tr>
<tr><td><code id="AMOCcrossover_+3A_lmax">lmax</code></td>
<td>
<p>The maximum possible length of the chromosome representation.</p>
</td></tr>
<tr><td><code id="AMOCcrossover_+3A_n">N</code></td>
<td>
<p>The length of time series.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The child chromosome that produced from <code>mom</code> and <code>dad</code> for
next generation.
</p>

<hr>
<h2 id='AMOCmutation'>Jump mutation operator to produce offspring for AMOC problem</h2><span id='topic+AMOCmutation'></span>

<h3>Description</h3>

<p>In a certain probability, the <code>mutation</code> genetic operator can be applied
to generate a new <code>child</code>. In this AMOC mutation operator, the new child
changepoint location can be down via a &quot;jump&quot; method. The child changepoint
location will jump <code>minDist</code> time units either to the left or right to
produce the changepoint location for the mutated child. The jump direction is
randomly decided with 0.5 probability.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AMOCmutation(
  child,
  p.range = NULL,
  minDist,
  Pchangepoint = NULL,
  lmax = NULL,
  mmax = NULL,
  N = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AMOCmutation_+3A_child">child</code></td>
<td>
<p>The child chromosome resulting from the <code>crossover</code> genetic
operator.</p>
</td></tr>
<tr><td><code id="AMOCmutation_+3A_p.range">p.range</code></td>
<td>
<p>The default value is <code>NULL</code>. If there is no requirement
on model order selection, such an auxiliary argument is needed for <code>GA</code>
and <code>IslandGA</code> functions.</p>
</td></tr>
<tr><td><code id="AMOCmutation_+3A_mindist">minDist</code></td>
<td>
<p>The minimum length between two adjacent changepoints in
<code><a href="#topic+AMOCselection">AMOCselection</a></code> operator, which is also the jump magnitude in the
<code>AMOCmutation</code> operator.</p>
</td></tr>
<tr><td><code id="AMOCmutation_+3A_pchangepoint">Pchangepoint</code></td>
<td>
<p>An auxiliary argument is needed for <code>GA</code>
and <code>IslandGA</code> functions.</p>
</td></tr>
<tr><td><code id="AMOCmutation_+3A_lmax">lmax</code></td>
<td>
<p>An auxiliary argument is needed for <code>GA</code> and <code>IslandGA</code>
functions.</p>
</td></tr>
<tr><td><code id="AMOCmutation_+3A_mmax">mmax</code></td>
<td>
<p>An auxiliary argument is needed for <code>GA</code> and <code>IslandGA</code>
functions.</p>
</td></tr>
<tr><td><code id="AMOCmutation_+3A_n">N</code></td>
<td>
<p>An auxiliary argument is needed for <code>GA</code> and <code>IslandGA</code>
functions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The resulting child chromosome representation.
</p>

<hr>
<h2 id='AMOCpopulation'>Random population initialization for AMOC problem</h2><span id='topic+AMOCpopulation'></span>

<h3>Description</h3>

<p>Randomly generate the individuals' chromosomes (changepoint configurations)
to construct the first generation population for the at most one changepoint
(AMOC) problem.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AMOCpopulation(popsize, p.range, N, minDist, Pchangepoint, mmax, lmax)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AMOCpopulation_+3A_popsize">popsize</code></td>
<td>
<p>An integer represents the number of individual in each
population for GA (or subpopulation for IslandGA).</p>
</td></tr>
<tr><td><code id="AMOCpopulation_+3A_p.range">p.range</code></td>
<td>
<p>Default is <code>NULL</code> for only changepoint detection. If
<code>p.range</code> is specified as a list object, which contains the range of
each model order parameters for order selection (integers). The number of
order parameters must be equal to the length of <code>p.range</code>.</p>
</td></tr>
<tr><td><code id="AMOCpopulation_+3A_n">N</code></td>
<td>
<p>The length of time series.</p>
</td></tr>
<tr><td><code id="AMOCpopulation_+3A_mindist">minDist</code></td>
<td>
<p>The minimum length between two adjacent changepoints.</p>
</td></tr>
<tr><td><code id="AMOCpopulation_+3A_pchangepoint">Pchangepoint</code></td>
<td>
<p>The probability that a changepoint can occur.</p>
</td></tr>
<tr><td><code id="AMOCpopulation_+3A_mmax">mmax</code></td>
<td>
<p>The maximum possible number of changepoints in the data set.</p>
</td></tr>
<tr><td><code id="AMOCpopulation_+3A_lmax">lmax</code></td>
<td>
<p>The maximum possible length of the chromosome representation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Given the possible candidate changepoint location set, each chromosome in the
first generation population can be obtained by randomly sampling one location
from the candidate set. The first element of every chromosome represent the
number of changepoints and the last non-zero element always equal to the
length of time series plus one (N+1).
</p>


<h3>Value</h3>

<p>A matrix that contains each individual's chromosome.
</p>

<hr>
<h2 id='AMOCselection'>The parents selection genetic algorithm operator for AMOC problem</h2><span id='topic+AMOCselection'></span>

<h3>Description</h3>

<p>The genetic algorithm require to select a pair of chromosomes, representing
<code>dad</code> and <code>mom</code>, for the <code>crossover</code> operator to
produce offspring (individual for next generation). Here, the same linear
ranking method in <code><a href="#topic+selection_linearrank">selection_linearrank</a></code> is used to select a pair
of chromosomes for <code>dad</code> and <code>mom</code> in the at most one changepoint
(AMOC) problem. By default, the dad has better fit/smaller fitness function
value/larger rank than <code>mom</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AMOCselection(pop, popFit)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AMOCselection_+3A_pop">pop</code></td>
<td>
<p>A matrix contains the chromosomes for all individuals. The number of
rows is equal to <code>lmax</code> and the number of columns is equal to the
<code>popsize</code>.</p>
</td></tr>
<tr><td><code id="AMOCselection_+3A_popfit">popFit</code></td>
<td>
<p>A vector contains the objective function value (population fit)
being associated to each individual chromosome from above.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list contains the chromosomes for <code>dad</code> and <code>mom</code>.
</p>

<hr>
<h2 id='ARIMA.BIC.Order'>Calculating BIC for Multiple changepoint detection with model order selection</h2><span id='topic+ARIMA.BIC.Order'></span>

<h3>Description</h3>

<p>The objective function for changepoint search in Autoregressive
moving average with model order selection via Bayesian Information Criterion
(BIC).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ARIMA.BIC.Order(chromosome, plen = 2, XMat, Xt)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ARIMA.BIC.Order_+3A_chromosome">chromosome</code></td>
<td>
<p>The chromosome consists of the number of changepoints, the
order of AR part, the order of MA part, the changepoint locations, and a
value of time series length plus 1 (N+1) indicating the end of the chromosome.</p>
</td></tr>
<tr><td><code id="ARIMA.BIC.Order_+3A_plen">plen</code></td>
<td>
<p>The number of model order parameters that need to be selected.
If model order selection needs to be performed simultaneously with the
changepoint detection task, <code>plen</code> should be nonzero.</p>
</td></tr>
<tr><td><code id="ARIMA.BIC.Order_+3A_xmat">XMat</code></td>
<td>
<p>A matrix contains the covariates, but not includes changepoint
effects, for time series regression.</p>
</td></tr>
<tr><td><code id="ARIMA.BIC.Order_+3A_xt">Xt</code></td>
<td>
<p>The simulated ARMA time series from <code>ts.sim</code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The BIC value of the objective function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>N = 1000
XMatT = matrix(1, nrow=N, ncol=1)
Xt = ts.sim(beta=0.5, XMat=XMatT, sigma=1, phi=0.5, theta=0.8,
            Delta=c(2, -2), CpLoc=c(250, 750), seed=1234)

# one chromosome representation
chromosome = c(2, 1, 1, 250, 750, 1001)
ARIMA.BIC.Order(chromosome, plen=2, XMat=XMatT, Xt=Xt)
</code></pre>

<hr>
<h2 id='arrayaCGH'>Glioblastoma array aCGH data</h2><span id='topic+arrayaCGH'></span>

<h3>Description</h3>

<p>Glioblastoma multiforme (GBM) can cause chromosomal aberrations during DNA
replication. Array comparative genomic hybridization (aCGH) experiments analyze
these aberrations using fluorescence intensity ratios, which indicate DNA copy
number variations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(arrayaCGH)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 193 rows and 5 columns.
</p>


<h3>References</h3>

<p>Lai, W. R., Johnson, M. D., Kucherlapati, R., &amp; Park, P. J. (2005).
Comparative analysis of algorithms for identifying amplifications and deletions
in array CGH data. <em>Bioinformatics</em>, 21(19), 3763-3770.
</p>
<p>Killick, R., &amp; Eckley, I. A. (2014). changepoint: An R package for
changepoint analysis. <em>Journal of statistical software</em>, 58, 1-19.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(arrayaCGH)
</code></pre>

<hr>
<h2 id='AtlantaTemperature'>Atlanta airport yearly average temperatures data</h2><span id='topic+AtlantaTemperature'></span>

<h3>Description</h3>

<p>The annual mean temperature data from 1879 to 2013 at Atlanta, Georgia’s
Hartsfield International Airport weather station.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(AtlantaTemperature)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 134 rows and 2 columns.
</p>


<h3>References</h3>

<p>Lund, R. B., Beaulieu, C., Killick, R., Lu, Q., &amp; Shi, X. (2023).
Good practices and common pitfalls in climate time series changepoint techniques:
A review. <em>Journal of Climate</em>, 36(23), 8041-8057.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AtlantaTemperature)
</code></pre>

<hr>
<h2 id='BinSearch.BIC'>Example function: Calculating BIC for AR(1) model</h2><span id='topic+BinSearch.BIC'></span>

<h3>Description</h3>

<p>The example objective function for changepoint search via Bayesian
Information Criterion (BIC) for simple AR(1) model from Shi et al. (2022).
The model is expressed as <code class="reqn">X_{t}=\kappa_{t}+\epsilon_{t}</code>, where
<code class="reqn">\epsilon_{t}</code> follows a stationary AR(1) process, and <code class="reqn">\kappa_{t}</code>
denotes the regimen mean.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BinSearch.BIC(chromosome, plen = 0, Xt)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BinSearch.BIC_+3A_chromosome">chromosome</code></td>
<td>
<p>The chromosome consists of the number of changepoints, their
locations, and a value of time series length plus 1 (N+1) indicating the end
of the chromosome.</p>
</td></tr>
<tr><td><code id="BinSearch.BIC_+3A_plen">plen</code></td>
<td>
<p>The number of model order parameters that need to be selected.
Since we don't need model order selection in this example, <code>plen</code> equals
to 0.</p>
</td></tr>
<tr><td><code id="BinSearch.BIC_+3A_xt">Xt</code></td>
<td>
<p>The simulated AR(1) time series from <code>ts.sim</code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The BIC value of the objective function.
</p>


<h3>References</h3>


<p>Shi, X., Gallagher, C., Lund, R., &amp; Killick, R. (2022).
A comparison of single and multiple changepoint techniques for time series data.
<em>Computational Statistics &amp; Data Analysis</em>, 170, 107433.

</p>


<h3>Examples</h3>

<pre><code class='language-R'>Ts = 1000
betaT = c(0.5) # intercept
XMatT = matrix(1, nrow=Ts, ncol=1)
colnames(XMatT) = "intercept"
sigmaT = 1
phiT = c(0.5)
thetaT = NULL
DeltaT = c(2, -2)
Cp.prop = c(1/4, 3/4)
CpLocT = floor(Ts*Cp.prop)

myts = ts.sim(beta=betaT, XMat=XMatT, sigma=sigmaT, phi=phiT, theta=thetaT,
              Delta=DeltaT, CpLoc=CpLocT, seed=1234)

# candidate changepoint configuration
chromosome = c(2, 250, 750, 1001)
BinSearch.BIC(chromosome, Xt=myts)
</code></pre>

<hr>
<h2 id='cptDist'>Comparing multiple changepoint configurations by pairwise distance</h2><span id='topic+cptDist'></span>

<h3>Description</h3>

<p>This function is to calculate the distance metric (Shi et al., 2022) between
two changepoint configurations, <code class="reqn">C_{1}=\{\tau_{1}, \ldots, \tau_{m}\}</code>
and <code class="reqn">C_{2}=\{\eta_{1}, \ldots, \eta_{k}\}</code>, where <code class="reqn">\tau</code>'s are the
changepoint locations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cptDist(tau1, tau2, N)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cptDist_+3A_tau1">tau1</code></td>
<td>
<p>A vector contains the changepoint locations for <code class="reqn">C_{1}</code>
for comparison. A value <code>NULL</code> is required if there is no changepoint detected.</p>
</td></tr>
<tr><td><code id="cptDist_+3A_tau2">tau2</code></td>
<td>
<p>A vector contains the changepoint locations for  <code class="reqn">C_{2}</code>
for comparison. A value <code>NULL</code> is required if there is no changepoint detected.</p>
</td></tr>
<tr><td><code id="cptDist_+3A_n">N</code></td>
<td>
<p>The simulated time series sample size. Two changepoint configurations
should have the same <code class="reqn">n</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The pairwise distance was proposed by Shi et al. (2022),
</p>
<p style="text-align: center;"><code class="reqn">d(C_{1}, C_{2})=|m-k|+ \min(A(C_{1}, C_{2})),</code>
</p>

<p>where <code class="reqn">m</code> is the number of changepoints in configuration <code class="reqn">C_{1}</code> and
<code class="reqn">k</code> is the number of changepoints in configuration <code class="reqn">C_{2}</code>.
The term <code class="reqn">\min(A(C_{1}, C_{2}))</code> reflects the cost of matching
changepoint locations between <code class="reqn">C_{1}</code> and <code class="reqn">C_{2}</code> and can be calculated
using the linear assignment method. Details can be found in Shi et al. (2022).
Note: if one configuration doesn't contain any changepoints (valued <code>NULL</code>),
the distance is defined as <code class="reqn">|m-k|</code>. The function can also be used
to examine changepoint detection performance in simulation studies. Given the
true changepoint configuration, <code class="reqn">C_{true}</code>, used in generating the time
series, the calculated distance between the estimated multiple changepoint
configuration, <code class="reqn">C_{est}=\{\eta_{1}, \ldots, \eta_{k}\}</code>, and <code class="reqn">C_{true}</code>
can be used to evaluate the performance of the detection algorithm.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>dist</code></td>
<td>
<p>The calculated distance.</p>
</td></tr>
</table>


<h3>References</h3>


<p>Shi, X., Gallagher, C., Lund, R., &amp; Killick, R. (2022).
A comparison of single and multiple changepoint techniques for time series data.
<em>Computational Statistics &amp; Data Analysis</em>, 170, 107433.

</p>


<h3>Examples</h3>

<pre><code class='language-R'>N = 100

# both tau1 and tau2 has detected changepoints
tau2 = c(25, 50, 75)
tau1 = c(20, 35, 70, 80, 90)
cptDist(tau1=tau1, tau2=tau2, N=N)

# either tau1 or tau2 has zero detected changepoints
cptDist(tau1=tau1, tau2=NULL, N=N)
cptDist(tau1=NULL, tau2=tau2, N=N)

# both tau1 and tau2 has zero detected changepoints
cptDist(tau1=NULL, tau2=NULL, N=N)
</code></pre>

<hr>
<h2 id='GA'>Genetic algorithm</h2><span id='topic+GA'></span>

<h3>Description</h3>

<p>Perform the modified genetic algorithm for changepoint detection.
This involves the minimization of an objective function using a genetic algorithm (GA).
The algorithm can be run sequentially or with explicit parallelization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GA(
  ObjFunc,
  N,
  GA_param = .default.GA_param,
  GA_operators = .default.operators,
  p.range = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GA_+3A_objfunc">ObjFunc</code></td>
<td>
<p>The fitness function to be minimized. Users can specify any R or Rcpp
function as the fitness function, setting the input as the potential solution to
the optimization problem and returning a numerical value as the output/fitness.
Depending on the user-specified chromosome representation, the optimization task
can be changepoint detection only or changepoint detection plus model order selection,
which can be specified via the <code>option</code> parameter in <code><a href="#topic+GA_param">GA_param</a></code>. When
<code>option="both"</code>, the list <code>p.range</code> must be specified to give the range
of model orders.</p>
</td></tr>
<tr><td><code id="GA_+3A_n">N</code></td>
<td>
<p>The sample size of the time series.</p>
</td></tr>
<tr><td><code id="GA_+3A_ga_param">GA_param</code></td>
<td>
<p>A list contains the hyper-parameters for genetic algorithm.
The default values for these hyper-parameters are included in <code>.default.GA_param</code>.
See <code><a href="#topic+GA_param">GA_param</a></code> for the details.</p>
</td></tr>
<tr><td><code id="GA_+3A_ga_operators">GA_operators</code></td>
<td>
<p>A list includes the functions for population initialization,
new individual selection, and genetic operator of crossover and mutation.
See <code><a href="#topic+operators">operators</a></code> for the details and default functions.</p>
</td></tr>
<tr><td><code id="GA_+3A_p.range">p.range</code></td>
<td>
<p>Default is <code>NULL</code> for only changepoint detection. If
<code>p.range</code> is specified as a list object, which contains the range of
each model order parameters for order selection (integers). The number of
order parameters must be equal to the length of <code>p.range</code>.</p>
</td></tr>
<tr><td><code id="GA_+3A_...">...</code></td>
<td>
<p>additional arguments that will be passed to the fitness function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list that has components:
</p>
<table role = "presentation">
<tr><td><code>overbestfit</code></td>
<td>
<p>The obtained minimum value of the objective function after
the final iteration.</p>
</td></tr>
<tr><td><code>overbestchrom</code></td>
<td>
<p>The locations of the detected changepoints associating
with the <code>overbestfit</code> the after the final iteration.</p>
</td></tr>
<tr><td><code>bestfit</code></td>
<td>
<p>The minimized fitness function values at each iteration.</p>
</td></tr>
<tr><td><code>bestchrom</code></td>
<td>
<p>The detected changepoints at each iteration.</p>
</td></tr>
<tr><td><code>count</code></td>
<td>
<p>The number of iterations undertaken by the genetic algorithm.</p>
</td></tr>
<tr><td><code>convg</code></td>
<td>
<p>An integer code.
</p>

<ul>
<li><p>0 indicates the algorithm successful completion.
</p>
</li>
<li><p>1 indicates the the total number of generations exceeds the
prespecified <code>maxgen</code> limit.
</p>
</li></ul>

</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
N = 1000
XMatT = matrix(1, nrow=N, ncol=1)
Xt = ts.sim(beta=0.5, XMat=XMatT, sigma=1, phi=0.5, theta=NULL,
            Delta=c(2, -2), CpLoc=c(250, 750), seed=1234)
TsPlotCheck(X=1:N, Xat=seq(from=1, to=N, length=10), Y=Xt, tau=c(250, 750))

GA.res = GA(ObjFunc=BinSearch.BIC, N=N, Xt=Xt)
GA.res$overbestfit
GA.res$overbestchrom

</code></pre>

<hr>
<h2 id='GA_param'>
GA_param
</h2><span id='topic+GA_param'></span>

<h3>Description</h3>

<p>A list object contains the hyperparameters for GA running.
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GA_param_+3A_popsize">popsize</code></td>
<td>
<p>An integer represents the number of individuals in each population.</p>
</td></tr>
<tr><td><code id="GA_param_+3A_pcrossover">Pcrossover</code></td>
<td>
<p>The probability that the crossover operator applies on two individual chromosomes.</p>
</td></tr>
<tr><td><code id="GA_param_+3A_pmutation">Pmutation</code></td>
<td>
<p>The probability that the mutation operator applies on one individual chromosome.</p>
</td></tr>
<tr><td><code id="GA_param_+3A_pchangepoint">Pchangepoint</code></td>
<td>
<p>The probability that a changepoint has occurred. User
could change this probability based on domain knowledge and the time series
length.</p>
</td></tr>
<tr><td><code id="GA_param_+3A_mindist">minDist</code></td>
<td>
<p>The minimum length between two adjacent changepoints.</p>
</td></tr>
<tr><td><code id="GA_param_+3A_mmax">mmax</code></td>
<td>
<p>The maximum possible number of changepoints in the data set.
For a time series of length 1000, the default value is 499. The suggested
value should be based on the length of the time series. For instance, if a
time series has length N, the recommended <code>mmax</code> should be N/2-1.</p>
</td></tr>
<tr><td><code id="GA_param_+3A_lmax">lmax</code></td>
<td>
<p>The maximum possible length of the chromosome representation. For
a time series of length 1000, the default value is 501. The suggested value
should be based on the length of the time series. For instance, if a time
series has length N, the recommended <code>lmax</code> should be 2+N/2-1.</p>
</td></tr>
<tr><td><code id="GA_param_+3A_maxgen">maxgen</code></td>
<td>
<p>The maximum number of generation that the GA can last.</p>
</td></tr>
<tr><td><code id="GA_param_+3A_maxconv">maxconv</code></td>
<td>
<p>If the overall best fitted value doesn't change after <code>maxconv</code> consecutive migrations, the GA algorithm stops.</p>
</td></tr>
<tr><td><code id="GA_param_+3A_option">option</code></td>
<td>
<p>A string controls the optimization task. &ldquo;cp&rdquo; indicates the task is changepoint detection only. &ldquo;both&rdquo; indicates the task will include both changepoint detection and model order selection.</p>
</td></tr>
<tr><td><code id="GA_param_+3A_monitoring">monitoring</code></td>
<td>
<p>A binary interger 0 or 1, indicating whether print out middle results for each iterations of GA.</p>
</td></tr>
<tr><td><code id="GA_param_+3A_parallel">parallel</code></td>
<td>
<p>Whether use multiple threads to parallel compute the individual fittness function values.</p>
</td></tr>
<tr><td><code id="GA_param_+3A_ncore">nCore</code></td>
<td>
<p>An integer represents the number of cores used in parallel computing.</p>
</td></tr>
<tr><td><code id="GA_param_+3A_tol">tol</code></td>
<td>
<p>The tolerance level for deciding GA to stop.</p>
</td></tr>
<tr><td><code id="GA_param_+3A_seed">seed</code></td>
<td>
<p>An single integer allows function produce reproducible results.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mo Li</p>


<h3>Examples</h3>

<pre><code class='language-R'># time series length
N = 1000

GA_param = list(
  popsize      = 40,
  Pcrossover   = 0.95,
  Pmutation    = 0.1,
  Pchangepoint = 0.06,
  minDist      = 3,
  mmax         = N/2 - 1,
  lmax         = 2 + N/2 - 1,
  maxgen       = 50000,
  maxconv      = 5000,
  option       = "cp",
  monitoring   = FALSE,
  parallel     = FALSE,
  nCore        = NULL,
  tol          = 1e-5,
  seed         = NULL
)
</code></pre>

<hr>
<h2 id='IslandGA'>Island model based genetic algorithm</h2><span id='topic+IslandGA'></span>

<h3>Description</h3>

<p>Perform the modified island-based genetic algorithm (IslandGA) for multiple changepoint detection.
Minimization of an objective function using genetic algorithm (GA).
The algorithm can be run sequentially or in explicit parallelisation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IslandGA(
  ObjFunc,
  N,
  IslandGA_param = .default.IslandGA_param,
  IslandGA_operators = .default.operators,
  p.range = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="IslandGA_+3A_objfunc">ObjFunc</code></td>
<td>
<p>The fitness function to be minimized. Users can specify any R or Rcpp
functions as the fitness function with setting input as potential solution to
the optimization problem and returning a numerical value as the output/fitness.</p>
</td></tr>
<tr><td><code id="IslandGA_+3A_n">N</code></td>
<td>
<p>The sample size of the time series.</p>
</td></tr>
<tr><td><code id="IslandGA_+3A_islandga_param">IslandGA_param</code></td>
<td>
<p>A list contains the hyper-parameters for IslandGA.
The default values for these hyper-parameters are included in <code>.default.IslandGA_param</code>.
See <code><a href="#topic+IslandGA_param">IslandGA_param</a></code> for the details.</p>
</td></tr>
<tr><td><code id="IslandGA_+3A_islandga_operators">IslandGA_operators</code></td>
<td>
<p>A list includes the functions for population initialization,
new individual selection, and genetic operator of crossover and mutation.
See <code><a href="#topic+operators">operators</a></code> for the details and default functions.</p>
</td></tr>
<tr><td><code id="IslandGA_+3A_p.range">p.range</code></td>
<td>
<p>Default is <code>NULL</code> for only changepoint detection. If
<code>p.range</code> is specified as a list object, which contains the range of
each model order parameters for order selection (integers). The number of
order parameters must be equal to the length of <code>p.range</code>.</p>
</td></tr>
<tr><td><code id="IslandGA_+3A_...">...</code></td>
<td>
<p>additional arguments that will be passed to the fitness function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list that has the following components.
</p>
<table role = "presentation">
<tr><td><code>overbestfit</code></td>
<td>
<p>The obtained minimum value of the objective function after
the final iteration.</p>
</td></tr>
<tr><td><code>overbestchrom</code></td>
<td>
<p>The locations of the detected changepoints associating
with the <code>overbestfit</code> the after the final iteration.</p>
</td></tr>
<tr><td><code>bestfit</code></td>
<td>
<p>The minimized fitness function values at each iteration.</p>
</td></tr>
<tr><td><code>bestchrom</code></td>
<td>
<p>The detected changepoints at each iteration.</p>
</td></tr>
<tr><td><code>countMig</code></td>
<td>
<p>The number of migrations undertaken by the IslandGA.</p>
</td></tr>
<tr><td><code>count</code></td>
<td>
<p>The number of iterations (generations) undertaken by the island
genetic algorithm model.</p>
</td></tr>
<tr><td><code>convg</code></td>
<td>
<p>An integer code.
</p>

<ul>
<li><p>0 indicates the algorithm successful completion.
</p>
</li>
<li><p>1 indicates the the total number of generations exceeds the
prespecified <code>maxgen</code> limit.
</p>
</li></ul>

</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>N = 1000
XMatT = matrix(1, nrow=N, ncol=1)
Xt = ts.sim(beta=0.5, XMat=XMatT, sigma=1, phi=0.5, theta=NULL,
            Delta=c(2, -2), CpLoc=c(250, 750), seed=1234)
TsPlotCheck(X=1:N, Xat=seq(from=1, to=N, length=10), Y=Xt, tau=c(250, 750))

IslandGA.res = IslandGA(ObjFunc=BinSearch.BIC, N=N, Xt=Xt)
IslandGA.res$overbestfit
IslandGA.res$overbestchrom

</code></pre>

<hr>
<h2 id='IslandGA_param'>
IslandGA_param
</h2><span id='topic+IslandGA_param'></span>

<h3>Description</h3>

<p>A list object contains the hyperparameters for island-based GA running.
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="IslandGA_param_+3A_subpopsize">subpopsize</code></td>
<td>
<p>An integer represents the number of individual in each sub-population (island).</p>
</td></tr>
<tr><td><code id="IslandGA_param_+3A_islandsize">Islandsize</code></td>
<td>
<p>The number of subpopulation (island).</p>
</td></tr>
<tr><td><code id="IslandGA_param_+3A_pcrossover">Pcrossover</code></td>
<td>
<p>The probability that the crossover operator applies on two individual chromosomes.</p>
</td></tr>
<tr><td><code id="IslandGA_param_+3A_pmutation">Pmutation</code></td>
<td>
<p>The probability that the mutation operator applies on one individual chromosome.</p>
</td></tr>
<tr><td><code id="IslandGA_param_+3A_pchangepoint">Pchangepoint</code></td>
<td>
<p>The probability that a changepoint has occurred.</p>
</td></tr>
<tr><td><code id="IslandGA_param_+3A_mindist">minDist</code></td>
<td>
<p>The minimum length between two adjacent changepoints.</p>
</td></tr>
<tr><td><code id="IslandGA_param_+3A_mmax">mmax</code></td>
<td>
<p>The maximum possible number of changepoints in the data set.</p>
</td></tr>
<tr><td><code id="IslandGA_param_+3A_lmax">lmax</code></td>
<td>
<p>The maximum possible length of the chromosome representation.</p>
</td></tr>
<tr><td><code id="IslandGA_param_+3A_maxmig">maxMig</code></td>
<td>
<p>The maximum number of migrations. After <code>maxMig</code> migrations, the island-based GA algorithm stops.</p>
</td></tr>
<tr><td><code id="IslandGA_param_+3A_maxgen">maxgen</code></td>
<td>
<p>The maximum number of generations that each subpopulation (island) has. The migration will apply after <code>maxgen</code> generations for each subpopulation(island).</p>
</td></tr>
<tr><td><code id="IslandGA_param_+3A_maxconv">maxconv</code></td>
<td>
<p>If the overall best fitted value doesn't change after  <code>maxconv</code> consecutive migrations, the island-based GA algorithm stops.</p>
</td></tr>
<tr><td><code id="IslandGA_param_+3A_option">option</code></td>
<td>
<p>A string controls the optimization task. &ldquo;cp&rdquo; indicates the task is changepoint detection only. &ldquo;both&rdquo; indicates the task will include both changepoint detection and model order selection.</p>
</td></tr>
<tr><td><code id="IslandGA_param_+3A_monitoring">monitoring</code></td>
<td>
<p>A binary interger 0 or 1, indicating whether print out middle results for each iterations of GA.</p>
</td></tr>
<tr><td><code id="IslandGA_param_+3A_parallel">parallel</code></td>
<td>
<p>Whether use multiple threads to parallel compute the individual fittness function values.</p>
</td></tr>
<tr><td><code id="IslandGA_param_+3A_ncore">nCore</code></td>
<td>
<p>An integer represents the number of cores used in parallel computing.</p>
</td></tr>
<tr><td><code id="IslandGA_param_+3A_tol">tol</code></td>
<td>
<p>The tolerance level for deciding GA to stop.</p>
</td></tr>
<tr><td><code id="IslandGA_param_+3A_seed">seed</code></td>
<td>
<p>An single integer allows function produce reproducible results.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mo Li</p>


<h3>Examples</h3>

<pre><code class='language-R'># time series length
N = 1000

IslandGA_param = list(
  subpopsize   = 40,
  Islandsize   = 5,
  Pcrossover   = 0.95,
  Pmutation    = 0.15,
  Pchangepoint = 0.1,
  minDist      = 2,
  mmax         = N/2 - 1,
  lmax         = 2 + N/2 - 1,
  maxMig       = 1000,
  maxgen       = 50,
  maxconv      = 100,
  option       = "cp",
  monitoring   = FALSE,
  parallel     = FALSE, ###
  nCore        = NULL,
  tol          = 1e-5,
  seed         = NULL
)
</code></pre>

<hr>
<h2 id='mutation'>The default mutation operator in genetic algorithm</h2><span id='topic+mutation'></span>

<h3>Description</h3>

<p>In a certain probability, the <code>mutation</code> genetic operator can be applied
to generate a new <code>child</code>. By default, the new child selection can be
down by the similar individual selection method in population initialization,
<code><a href="#topic+selectTau">selectTau</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mutation(child, p.range = NULL, minDist, Pb, lmax, mmax, N)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mutation_+3A_child">child</code></td>
<td>
<p>The child chromosome resulting from the <code>crossover</code> genetic
operator.</p>
</td></tr>
<tr><td><code id="mutation_+3A_p.range">p.range</code></td>
<td>
<p>Default is <code>NULL</code> for only changepoint detection. If
<code>p.range</code> is specified as a list object, which contains the range of
each model order parameters for order selection (integers). The number of
order parameters must be equal to the length of <code>p.range</code>.</p>
</td></tr>
<tr><td><code id="mutation_+3A_mindist">minDist</code></td>
<td>
<p>The required minimum distance between two adjacent changepoints.</p>
</td></tr>
<tr><td><code id="mutation_+3A_pb">Pb</code></td>
<td>
<p>The probability of changepoints for every time series.</p>
</td></tr>
<tr><td><code id="mutation_+3A_lmax">lmax</code></td>
<td>
<p>The user specified maximum number of changepoints, by default,
as <code>N/2 - 1</code>.</p>
</td></tr>
<tr><td><code id="mutation_+3A_mmax">mmax</code></td>
<td>
<p>The user specified maximum length of individual chromosome,
by default, as <code>2+N/2 + 1</code>.</p>
</td></tr>
<tr><td><code id="mutation_+3A_n">N</code></td>
<td>
<p>The sample size of the time series.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A function can apply mutation to the produced child with the specified
probability <code>Pmutation</code> in <code><a href="#topic+GA_param">GA_param</a></code> and
<code><a href="#topic+IslandGA_param">IslandGA_param</a></code>. If order selection is not requested
(<code>option = "cp"</code> in <code>GA_param</code> and <code>Island_GA</code>), the default
<code><a href="#topic+mutation">mutation</a></code> operator function uses <code>selectTau</code> to select
a completely new individual with a new chromosome as the mutated child.
For details, see <code><a href="#topic+selectTau">selectTau</a></code>. If order selection is needed
(<code>option = "both"</code> in <code>GA_param</code> and <code>Island_GA</code>), we first
decide whether to keep the produced child's model order with a probability
of 0.5. If the child's model order is retained, the <code>selectTau</code>
function is used to select a completely new individual with a new chromosome
as the mutated child. If a new model order is selected from the candidate
model order set, there is a 0.5 probability to either select a completely new
individual with new changepoint locations or retain the original child's
changepoint locations for the mutated child. Note that the current model
orders in the child's chromosome are excluded from the set to avoid redundant
objective function evaluation. Finally, the function returns a vector
containing the modified chromosomes for mutated <code>child</code>.
</p>


<h3>Value</h3>

<p>The resulting child chromosome representation.
</p>

<hr>
<h2 id='operators'>
operators
</h2><span id='topic+operators'></span>

<h3>Description</h3>

<p>A list object contains the GA or IslandGA operators function names.
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="operators_+3A_population">population</code></td>
<td>
<p>It could be a function or a matrix. The function should be designed for initializing a population. The default population initialization is random initialization with some imposed constraints. See <code><a href="#topic+random_population">random_population</a></code> for example. The function returned object is a matrix, <code>pop</code>. The users can specified their own <code>population</code> function. It could also be a matrix object, which contain the user specified chromosome. By default, each column represents one individual chromosome.</p>
</td></tr>
<tr><td><code id="operators_+3A_selection">selection</code></td>
<td>
<p>A function can help select <code>mom</code> and <code>dad</code> from current generation population, where <code>dad</code> is set to have better fit (smaller fitness function values). The default for selection uses the linear rank selection method. See <code><a href="#topic+selection_linearrank">selection_linearrank</a></code> for example. The function returned object is a list contain the chromosomes for <code>mom</code> and <code>dad</code>.</p>
</td></tr>
<tr><td><code id="operators_+3A_crossover">crossover</code></td>
<td>
<p>A function can apply crossover to the chosen parents to produce child for next generation with specified probability. The default for crossover uses the uniform crossover method. See <code><a href="#topic+uniformcrossover">uniformcrossover</a></code> for details in the default crossover operator. The function returned object is a vector contain the chromosomes for <code>child</code>.</p>
</td></tr>
<tr><td><code id="operators_+3A_mutation">mutation</code></td>
<td>
<p>A function can apply mutation to the produced child with the specified probability <code>Pmutation</code> in <code><a href="#topic+GA_param">GA_param</a></code> and <code><a href="#topic+IslandGA_param">IslandGA_param</a></code>. See <code><a href="#topic+mutation">mutation</a></code> for details in the default mutation operator.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Lu, Q., Lund, R., &amp; Lee, T. C. (2010). An MDL approach to the climate segmentation problem.
<em>Ann. Appl. Stat.</em> 4 (1) 299 - 319.</p>


<h3>See Also</h3>

<p>See Also as <code><a href="#topic+GA">GA</a></code>.</p>


<h3>Examples</h3>

<pre><code class='language-R'>operators = list(population = "random_population",
                 selection  = "selection_linearrank",
                 crossover  = "uniformcrossover",
                 mutation   = "mutation")
</code></pre>

<hr>
<h2 id='random_population'>Random population initialization</h2><span id='topic+random_population'></span>

<h3>Description</h3>

<p>Randomly generate the individuals' chromosomes (changepoint confirgurations)
to construct the first generation population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>random_population(popsize, prange, N, minDist, Pb, mmax, lmax)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="random_population_+3A_popsize">popsize</code></td>
<td>
<p>An integer represents the number of individual in each
population for GA (or subpopulation for IslandGA).</p>
</td></tr>
<tr><td><code id="random_population_+3A_prange">prange</code></td>
<td>
<p>Default is <code>NULL</code> for only changepoint detection. If
<code>prange</code> is specified as a list object, which contains the range of
each model order parameters for order selection (integers). The number of
order parameters must be equal to the length of <code>prange</code>.</p>
</td></tr>
<tr><td><code id="random_population_+3A_n">N</code></td>
<td>
<p>The length of time series.</p>
</td></tr>
<tr><td><code id="random_population_+3A_mindist">minDist</code></td>
<td>
<p>The minimum length between two adjacent changepoints.</p>
</td></tr>
<tr><td><code id="random_population_+3A_pb">Pb</code></td>
<td>
<p>Same as <code>Pchangepoint</code>, the probability that a changepoint has occurred.</p>
</td></tr>
<tr><td><code id="random_population_+3A_mmax">mmax</code></td>
<td>
<p>The maximum possible number of changepoints in the data set.</p>
</td></tr>
<tr><td><code id="random_population_+3A_lmax">lmax</code></td>
<td>
<p>The maximum possible length of the chromosome representation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The default population initialization uses <code><a href="#topic+selectTau">selectTau</a></code> to
select the chromosome for the first generation population. Each column from
the produced population matrix represent an chromosome of an individual.
The first element of every chromosome represent the number of changepoints
and the last non-zero element always equal to the length of time series
plus one (N+1).
</p>


<h3>Value</h3>

<p>A matrix that contains each individual's chromosome.
</p>

<hr>
<h2 id='selection_linearrank'>The default parents selection genetic algorithm operator</h2><span id='topic+selection_linearrank'></span>

<h3>Description</h3>

<p>The genetic algorithm require to select a pair of chromosomes, representing
<code>dad</code> and <code>mom</code>, for the <code>crossover</code> operator to
produce offspring (individual for next generation). The parents chromosomes
are randomly selectd from the initialized population by a linear ranking
method according to each individual's fittness in the input argument
<code>popFit</code>. By default, the dad has better fit/smaller fitness function
value/larger rank than <code>mom</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>selection_linearrank(pop, popFit)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="selection_linearrank_+3A_pop">pop</code></td>
<td>
<p>A matrix contains the chromosomes for all individuals. The number of
rows is equal to <code>lmax</code> and the number of columns is equal to the
<code>popsize</code>.</p>
</td></tr>
<tr><td><code id="selection_linearrank_+3A_popfit">popFit</code></td>
<td>
<p>A vector contains the objective function value (population fit)
being associated to each individual chromosome from above.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list contains the chromosomes for <code>dad</code> and <code>mom</code>.
</p>

<hr>
<h2 id='selectTau'>Randomly select the chromosome</h2><span id='topic+selectTau'></span>

<h3>Description</h3>

<p>Randomly select the changepoint configuration for population initialization.
The selected changepoint configuration represents a changepoint chromosome.
The first element of the chromosome represent the number of changepoints
and the last non-zero element always equal to the length of time series + 1.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>selectTau(N, prange, minDist, Pb, mmax, lmax)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="selectTau_+3A_n">N</code></td>
<td>
<p>The length of time series.</p>
</td></tr>
<tr><td><code id="selectTau_+3A_prange">prange</code></td>
<td>
<p>A list object containing the possible range for other
pre-defined model parameters, i.e. AR/MA order of ARMA models.</p>
</td></tr>
<tr><td><code id="selectTau_+3A_mindist">minDist</code></td>
<td>
<p>The minimum length between two adjacent changepoints.</p>
</td></tr>
<tr><td><code id="selectTau_+3A_pb">Pb</code></td>
<td>
<p>Same as <code>Pchangepoint</code>, the probability that a changepoint has occurred.</p>
</td></tr>
<tr><td><code id="selectTau_+3A_mmax">mmax</code></td>
<td>
<p>The maximum possible number of changepoints in the data set.</p>
</td></tr>
<tr><td><code id="selectTau_+3A_lmax">lmax</code></td>
<td>
<p>The maximum possible length of the chromosome representation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single changepoint configuration format as above.
</p>

<hr>
<h2 id='ts.sim'>Time series simulation with changepoint effects</h2><span id='topic+ts.sim'></span>

<h3>Description</h3>

<p>This is a function to simulate time series with changepoint effects. See
details below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ts.sim(
  beta,
  XMat,
  sigma,
  phi = NULL,
  theta = NULL,
  Delta = NULL,
  CpLoc = NULL,
  seed = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ts.sim_+3A_beta">beta</code></td>
<td>
<p>A parameter vector contains other mean function parameters without changepoint parameters.</p>
</td></tr>
<tr><td><code id="ts.sim_+3A_xmat">XMat</code></td>
<td>
<p>The covairates for time series mean function without changepoint indicators.</p>
</td></tr>
<tr><td><code id="ts.sim_+3A_sigma">sigma</code></td>
<td>
<p>The standard deviation for time series residuals <code class="reqn">\epsilon_{t}</code>.</p>
</td></tr>
<tr><td><code id="ts.sim_+3A_phi">phi</code></td>
<td>
<p>A vector for the autoregressive (AR) parameters for AR part.</p>
</td></tr>
<tr><td><code id="ts.sim_+3A_theta">theta</code></td>
<td>
<p>A vector for the moving average (MA) parameters for MA part.</p>
</td></tr>
<tr><td><code id="ts.sim_+3A_delta">Delta</code></td>
<td>
<p>The parameter vector contains the changepoint parameters for time series mean function.</p>
</td></tr>
<tr><td><code id="ts.sim_+3A_cploc">CpLoc</code></td>
<td>
<p>A vector contains the changepoint locations range from <code class="reqn">1\leq\tau\leq T_{s}</code>.</p>
</td></tr>
<tr><td><code id="ts.sim_+3A_seed">seed</code></td>
<td>
<p>The random seed for simulation reproducibility.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The simulated time series <code class="reqn">Z_{t}, t=1,\ldots,T_{s}</code> is from a class of models,
</p>
<p style="text-align: center;"><code class="reqn">Z_{t}=\mu_{t}+e_{t}.</code>
</p>


<ul>
<li><p>Time series observations are IID <br /> <code class="reqn">\mu_{t}</code> is a constant
and <code class="reqn">e_{t}</code>'s are independent and identically distributed as <code class="reqn">N(0,\sigma)</code>.
</p>
</li>
<li><p>ARMA(p,q) model with constant mean <br /> <code class="reqn">\mu_{t}</code> is a constant,
and <code class="reqn">e_{t}</code> follows an ARMA(p,q) process.
</p>
</li>
<li><p>ARMA(p,q) model with seasonality <br /> <code class="reqn">\mu_{t}</code> is not a constant,
</p>
<p style="text-align: center;"><code class="reqn">\mu_{t} = ASin(\frac{2\pi t}{S})+BSin(\frac{2\pi t}{S}),</code>
</p>

<p>and <code class="reqn">e_{t}</code> follows an ARMA(p,q) process.
</p>
</li>
<li><p>ARMA(p,q) model with seasonality and trend <br /> <code class="reqn">\mu_{t}</code> is not a constant,
</p>
<p style="text-align: center;"><code class="reqn">\mu_{t} = ASin(\frac{2\pi t}{S})+BSin(\frac{2\pi t}{S}) + \alpha t,</code>
</p>

<p>and <code class="reqn">e_{t}</code> follows an ARMA(p,q) process.
The changepoint effects could be introduced through the <code class="reqn">\mu_{t}</code> as
</p>
<p style="text-align: center;"><code class="reqn">\mu_{t} = \Delta_{1}I_{t&gt;\tau_{1}} + \ldots + \Delta_{m}I_{t&gt;\tau_{m}},</code>
</p>

<p>where <code class="reqn">1\leq\tau_{1}&lt;\ldots&lt;\tau_{m}\leq T_{s}</code> are the changepoint locations and
<code class="reqn">\Delta_{1},\ldots,\Delta_{m}</code> are the changepoint parameter that need to be estimated.
</p>
</li></ul>



<h3>Value</h3>

<p>The simulated time series with attributes:
</p>
<table role = "presentation">
<tr><td><code>Z</code></td>
<td>
<p>The simulated time series.</p>
</td></tr>
<tr><td><code>Attributes</code></td>
<td>


<ul>
<li><p><code>DesignX</code> The covariates include all changepoint indicators.
</p>
</li>
<li><p><code>mu</code> A vector includes the mean values for simulated time series sequences.
</p>
</li>
<li><p><code>theta</code> The true parameter vector (including changepoint effects).
</p>
</li>
<li><p><code>CpEff</code> The true changepoint magnitudes.
</p>
</li>
<li><p><code>CpLoc</code> The true changepoint locations where we introduce the mean changing effects.
</p>
</li>
<li><p><code>arEff</code> A vector givies the AR coefficients.
</p>
</li>
<li><p><code>maEff</code> A vector givies the MA coefficients.
</p>
</li>
<li><p><code>seed</code> The random seed used for this simulation.
</p>
</li></ul>

</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>##### M1: Time series observations are IID
Ts = 1000
betaT = c(0.5) # intercept
XMatT = matrix(1, nrow=Ts, ncol=1)
colnames(XMatT) = "intercept"
sigmaT = 1
DeltaT = c(2, -2)
Cp.prop = c(1/4, 3/4)
CpLocT = floor(Ts*Cp.prop)

myts = ts.sim(beta=betaT, XMat=XMatT, sigma=sigmaT,
              Delta=DeltaT, CpLoc=CpLocT, seed=1234)
TsPlotCheck(Y=myts, tau=CpLocT)

##### M2: ARMA(2,1) model with constant mean
Ts = 1000
betaT = c(0.5) # intercept
XMatT = matrix(1, nrow=Ts, ncol=1)
colnames(XMatT) = "intercept"
sigmaT = 1
phiT = c(0.5, -0.5)
thetaT = c(0.8)
DeltaT = c(2, -2)
Cp.prop = c(1/4, 3/4)
CpLocT = floor(Ts*Cp.prop)

myts = ts.sim(beta=betaT, XMat=XMatT, sigma=sigmaT,
              phi=phiT, theta=thetaT, Delta=DeltaT, CpLoc=CpLocT, seed=1234)
TsPlotCheck(Y=myts, tau=CpLocT)

##### M3: ARMA(2,1) model with seasonality
Ts = 1000
betaT = c(0.5, -0.5, 0.3) # intercept, B, D
period = 30
XMatT = cbind(rep(1, Ts), cos(2*pi*(1:Ts)/period), sin(2*pi*(1:Ts)/period))
colnames(XMatT) = c("intercept", "Bvalue", "DValue")
sigmaT = 1
phiT = c(0.5, -0.5)
thetaT = c(0.8)
DeltaT = c(2, -2)
Cp.prop = c(1/4, 3/4)
CpLocT = floor(Ts*Cp.prop)

myts = ts.sim(beta=betaT, XMat=XMatT, sigma=sigmaT,
              phi=phiT, theta=thetaT, Delta=DeltaT, CpLoc=CpLocT, seed=1234)
TsPlotCheck(Y=myts, tau=CpLocT)


##### M4: ARMA(2,1) model with seasonality and trend
# scaled trend if large number of sample size
Ts = 1000
betaT = c(0.5, -0.5, 0.3, 0.01) # intercept, B, D, alpha
period = 30
XMatT = cbind(rep(1, Ts), cos(2*pi*(1:Ts)/period), sin(2*pi*(1:Ts)/period), 1:Ts)
colnames(XMatT) = c("intercept", "Bvalue", "DValue", "trend")
sigmaT = 1
phiT = c(0.5, -0.5)
thetaT = c(0.8)
DeltaT = c(2, -2)
Cp.prop = c(1/4, 3/4)
CpLocT = floor(Ts*Cp.prop)

myts = ts.sim(beta=betaT, XMat=XMatT, sigma=sigmaT,
              phi=phiT, theta=thetaT, Delta=DeltaT, CpLoc=CpLocT, seed=1234)
TsPlotCheck(Y=myts, tau=CpLocT)
</code></pre>

<hr>
<h2 id='TsPlotCheck'>Plot the simulated time series</h2><span id='topic+TsPlotCheck'></span>

<h3>Description</h3>

<p>This is a function to plot the simulated time series with segmentation visualization
by provided changepoint locations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TsPlotCheck(
  X = NULL,
  Xat = NULL,
  Y,
  tau = NULL,
  mu = NULL,
  XLAB = NULL,
  YLAB = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="TsPlotCheck_+3A_x">X</code></td>
<td>
<p>The time series time index, which could be specified as years, months, days, or others.
The default value is NULL and the vector from 1 to the time series length will be applied.</p>
</td></tr>
<tr><td><code id="TsPlotCheck_+3A_xat">Xat</code></td>
<td>
<p>The values from <code>X</code> that will be used as the X axis tick marks.</p>
</td></tr>
<tr><td><code id="TsPlotCheck_+3A_y">Y</code></td>
<td>
<p>The time series data.</p>
</td></tr>
<tr><td><code id="TsPlotCheck_+3A_tau">tau</code></td>
<td>
<p>The provided changepoint locations.</p>
</td></tr>
<tr><td><code id="TsPlotCheck_+3A_mu">mu</code></td>
<td>
<p>The provided meam values for each time <code class="reqn">t</code>.</p>
</td></tr>
<tr><td><code id="TsPlotCheck_+3A_xlab">XLAB</code></td>
<td>
<p>A descriptive label for X axis.</p>
</td></tr>
<tr><td><code id="TsPlotCheck_+3A_ylab">YLAB</code></td>
<td>
<p>A descriptive label for Y axis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Ts = 1000
betaT = c(0.5) # intercept
XMatT = matrix(1, nrow=Ts, ncol=1)
colnames(XMatT) = "intercept"
sigmaT = 1
DeltaT = c(2, -2)
Cp.prop = c(1/4, 3/4)
CpLocT = floor(Ts*Cp.prop)

myts = ts.sim(beta=betaT, XMat=XMatT, sigma=sigmaT, Delta=DeltaT, CpLoc=CpLocT, seed=1234)
TsPlotCheck(X=1:Ts, Xat=seq(from=1, to=Ts, length=10), Y=myts, tau=CpLocT)
</code></pre>

<hr>
<h2 id='uniformcrossover'>Uniform crossover to produce offsprings</h2><span id='topic+uniformcrossover'></span>

<h3>Description</h3>

<p>In uniform crossover, typically, each bit is chosen from either parent with
equal probability. Other mixing ratios are sometimes used, resulting in
offspring which inherit more genetic information from one parent than the
other. In a uniform crossover, we don’t divide the chromosome into segments,
rather we treat each gene separately. In this, we essentially flip a coin
for each chromosome to decide whether or not it will be included in the
off-spring. If model order selection is requested, each child's model order
has the equal probability (0.5) from <code>dad</code> and <code>mom</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>uniformcrossover(mom, dad, prange, minDist, lmax, N)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="uniformcrossover_+3A_mom">mom</code></td>
<td>
<p>Among two selected individuals, <code>mom</code> represents the selected
chromosome representation with lower fitness function value.</p>
</td></tr>
<tr><td><code id="uniformcrossover_+3A_dad">dad</code></td>
<td>
<p>Among two selected individuals, <code>dad</code> represents the selected
chromosome representation with larger fitness function value.</p>
</td></tr>
<tr><td><code id="uniformcrossover_+3A_prange">prange</code></td>
<td>
<p>Default value is <code>NULL</code> for only changepoint detection. If
<code>prange</code> is specified as a list object, which contains the range of
each model order parameters for order selection (integers). The number of
order parameters must be equal to the length of <code>prange</code>.</p>
</td></tr>
<tr><td><code id="uniformcrossover_+3A_mindist">minDist</code></td>
<td>
<p>The required minimum distance between two adjacent changepoints.</p>
</td></tr>
<tr><td><code id="uniformcrossover_+3A_lmax">lmax</code></td>
<td>
<p>The user specified maximum number of changepoints, by default,
as <code>N/2 - 1</code>.</p>
</td></tr>
<tr><td><code id="uniformcrossover_+3A_n">N</code></td>
<td>
<p>The length of time series.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The child chromosome that produced from <code>mom</code> and <code>dad</code> for
next generation.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
