<!DOCTYPE html><html><head><title>Help for package chem16S</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {chem16S}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#chem16S-package'><p>Chemical metrics for microbial communities</p></a></li>
<li><a href='#chem16S-data'><p>'phyloseq-class' objects generated using the DADA2 Pipeline Tutorial</p></a></li>
<li><a href='#chemlab'><p>Formatted labels for chemical metrics</p></a></li>
<li><a href='#get_metadata'><p>Template function for metadata</p></a></li>
<li><a href='#get_metrics'><p>Calculate chemical metrics of community reference proteomes</p></a></li>
<li><a href='#map_taxa'><p>Map taxonomic names to NCBI or GTDB taxonomy</p></a></li>
<li><a href='#physeq'><p>Process 'phyloseq-class' objects to calculate chemical metrics</p></a></li>
<li><a href='#plot_metrics'><p>Plot chemical metrics of community reference proteomes</p></a></li>
<li><a href='#read_RDP'><p>Read and filter RDP Classifier output</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Date:</td>
<td>2024-07-01</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Title:</td>
<td>Chemical Metrics for Microbial Communities</td>
</tr>
<tr>
<td>Author:</td>
<td>Jeffrey Dick <a href="https://orcid.org/0000-0002-0687-5890"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jeffrey Dick &lt;j3ffdick@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>plyr, ggplot2, rlang, reshape2, phyloseq, canprot (&ge; 2.0.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>tinytest, knitr, patchwork, ggpmisc, rmarkdown</td>
</tr>
<tr>
<td>Description:</td>
<td>Combines taxonomic classifications of high-throughput 16S rRNA
  gene sequences with reference proteomes of archaeal and bacterial taxa to
  generate amino acid compositions of community reference proteomes. Calculates
  chemical metrics including carbon oxidation state ('Zc'), stoichiometric
  oxidation and hydration state ('nO2' and 'nH2O'), H/C, N/C, O/C, and S/C
  ratios, grand average of hydropathicity ('GRAVY'), isoelectric point ('pI'),
  protein length, and average molecular weight of amino acid residues. Uses
  precomputed reference proteomes for archaea and bacteria derived from the
  Genome Taxonomy Database ('GTDB'). Also includes reference proteomes derived
  from the NCBI Reference Sequence ('RefSeq') database and manual mapping from
  the 'RDP Classifier' training set to 'RefSeq' taxonomy as described by Dick and
  Tan (2023) &lt;<a href="https://doi.org/10.1007%2Fs00248-022-01988-9">doi:10.1007/s00248-022-01988-9</a>&gt;. Processes taxonomic
  classifications in 'RDP Classifier' format or OTU tables in 'phyloseq-class'
  objects from the Bioconductor package 'phyloseq'.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/jedick/chem16S">https://github.com/jedick/chem16S</a>,
<a href="https://doi.org/10.5281/zenodo.6793059">https://doi.org/10.5281/zenodo.6793059</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-07-01 13:07:33 UTC; jedick</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-07-01 14:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='chem16S-package'>Chemical metrics for microbial communities</h2><span id='topic+chem16S-package'></span><span id='topic+chem16S'></span>

<h3>Description</h3>

<p>Functions and data to calculate chemical metrics for reference proteomes for microbial (archaeal and bacterial) communities.
Amino acid compositions of community reference proteomes are generated by combining reference proteomes of taxa (derived from GTDB or RefSeq) with taxonomic classifications of 16S rRNA gene sequences.
</p>


<h3>Details</h3>


<ul>
<li> <p><code><a href="#topic+read_RDP">read_RDP</a></code> - Read and filter an RDP Classifier table
</p>
</li>
<li> <p><code><a href="#topic+map_taxa">map_taxa</a></code> - Map RDP Classifier assignments to the NCBI taxonomy
</p>
</li>
<li> <p><code><a href="#topic+get_metrics">get_metrics</a></code> - Get chemical metrics for community reference proteomes
</p>
</li>
<li> <p><code><a href="#topic+get_metadata">get_metadata</a></code> - Example function for retrieving sample metadata
</p>
</li>
<li> <p><code><a href="#topic+plot_metrics">plot_metrics</a></code> - Plot metrics with symbols and colors based on metadata
</p>
</li>
<li> <p><code><a href="#topic+physeq">physeq</a></code> - Functions designed to analyze <code><a href="phyloseq.html#topic+phyloseq-class">phyloseq-class</a></code> objects
</p>
</li></ul>

<p>History:
</p>

<ol>
<li><p> Work begun in 2021. The combination of RefSeq reference proteomes with taxonomic abundances to compute community-level chemical metrics was described by Dick and Tan (2023). <span class="pkg">chem16S</span> originated as code in the <span class="pkg">JMDplots</span> package (<a href="https://github.com/jedick/JMDplots">https://github.com/jedick/JMDplots</a>).
</p>
</li>
<li><p> Development in 2022. Dick and Meng (2023) compared community <I>Z</I><sub>C</sub> with redox potential measurements from local to global scales. The term &ldquo;community reference proteomes&rdquo; was first applied, and <span class="pkg">chem16S</span> was split into a separate package.
</p>
</li>
<li><p> Late 2022. GTDB r207 was added as a reference database.
</p>
</li>
<li><p> June&ndash;July 2023. Integration with <span class="pkg">phyloseq</span> and addition of vignettes: <em>Chemical metrics of reference proteomes</em>, <em>Integration of chem16S with phyloseq</em>, and <em>Plotting two chemical metrics</em>. Default reference database changed to GTDB r207.
</p>
</li>
<li><p> April 2024. Updated to GTDB r214.
</p>
</li></ol>



<h3>Options set in package chem16S</h3>

<p><span class="pkg">chem16S</span> sets an option using the global <code><a href="base.html#topic+options">options</a></code> mechanism in R.
This option will be set when package <span class="pkg">chem16S</span> (or its namespace) is loaded if not already set.
</p>

<dl>
<dt><code>manual_mappings</code></dt><dd>
<p>A data frame of mappings between RDP and NCBI (RefSeq) taxonomies, which is read from &lsquo;<span class="file">extdata/manual_mappings.csv</span>&rsquo;.
The columns include <code>RDP.rank</code>, <code>RDP.name</code>, <code>NCBI.rank</code>, <code>NCBI.name</code>, and <code>notes</code>.
This option is made available so the user can modify the manual mappings used by <code><a href="#topic+map_taxa">map_taxa</a></code> at runtime.
</p>
</dd>
</dl>



<h3>Files in RefDB/RefSeq_206</h3>

<p>NOTE: None of the &lsquo;<span class="file">*.R</span>&rsquo; files in the &lsquo;<span class="file">extdata</span>&rsquo; directories are included in the package submitted to CRAN; see GitHub or Zenodo for these files.
</p>
<p>This directory contains two sets of files:
1) scripts to process source RefSeq sequence files to generate amino acid compositions of species-level reference proteomes and taxonomic names;
2) script and output for amino acid compositions of higher-level taxa.
The files are based on RefSeq release 206 of 2021-05-21 (O'Leary et al., 2016).
</p>

<dl>
<dt>&lsquo;<span class="file">README.txt</span>&rsquo;</dt><dd><p>Description of steps to generate reference proteomes of species-level taxa (including downloads and shell commands).</p>
</dd>
<dt>&lsquo;<span class="file">gencat.sh</span>&rsquo;</dt><dd><p>Helper script to extract microbial protein records from the RefSeq catalog.</p>
</dd>
<dt>&lsquo;<span class="file">genome_AA.R</span>&rsquo;</dt><dd>
<p>R code to sum the amino acid compositions of all proteins for each bacterial, archaeal, and viral species in the NCBI Reference Sequence database.
NOTE: To save space in this package, the output file (&lsquo;<span class="file">genome_AA.csv</span>&rsquo;) is stored in the <code>RefDB/RefSeq_206</code> directory of the JMDplots package on GitHub (<a href="https://github.com/jedick/JMDplots">https://github.com/jedick/JMDplots</a>.
The first five columns are: <code>protein</code> (&ldquo;refseq&rdquo;), <code>organism</code> (taxonomic id), <code>ref</code> (organism name), <code>abbrv</code> (empty), <code>chains</code> (number of protein sequences for this organism).
Columns 6 to 25 have the counts of amino acids.
</p>
</dd>
<dt>&lsquo;<span class="file">taxonomy.R</span>&rsquo;</dt><dd>
<p>R code for processing taxonomic IDs; the output file is &lsquo;<span class="file">taxonomy.csv</span>&rsquo;.
The columns are NCBI taxonomic ID (taxid), and names at different taxonomic rank (species, genus, family, order, class, phylum, superkingdom).
</p>
</dd>
<dt>&lsquo;<span class="file">taxon_AA.R</span>&rsquo;</dt><dd><p>Functions to create the files listed below:</p>
</dd>
<dt>&lsquo;<span class="file">taxon_AA.csv.xz</span>&rsquo;</dt><dd><p>Average amino acid composition of reference proteomes for all species in each genus, family, order, class, phylum, and superkingdom.</p>
</dd>
</dl>



<h3>Files in RefDB/GTDB_214</h3>


<dl>
<dt>&lsquo;<span class="file">taxon_AA.R</span>&rsquo;</dt><dd><p>Functions to process GTDB source files (Parks et al., 2022) and produce the following output file:</p>
</dd>
<dt>&lsquo;<span class="file">taxon_AA.csv.xz</span>&rsquo;</dt><dd>
<p>Average amino acid composition of reference proteomes for all species in each genus, family, order, class, phylum, and domain.
In both this file and the corresponding file for RefSeq (see above), the <code>protein</code>, <code>organism</code>, <code>ref</code>, and <code>abbrv</code> columns contain the rank, taxon name, number of species used to generate the amino acid composition of this taxon, and parent taxon.
<code>chains</code> is <code>1</code>, denoting a single polypeptide chain, so the amino acid composition represents the average per-protein amino acid composition in this taxon, and the sum of amino acid counts is the average protein length.
</p>
</dd>
</dl>



<h3>Files in extdata/metadata</h3>


<dl>
<dt>&lsquo;<span class="file">BGPF13.csv</span>&rsquo;</dt><dd><p>Metadata for Heart Lake Geyser Basin, Yellowstone (Bowen De León et al., 2012).</p>
</dd>
<dt>&lsquo;<span class="file">HLA+16.csv</span>&rsquo;</dt><dd><p>Metadata for the Baltic Sea (Herlemann et al., 2016).</p>
</dd>
<dt>&lsquo;<span class="file">SMS+12.csv</span>&rsquo;</dt><dd><p>Metadata for Bison Pool, Yellowstone (Swingley et al., 2012).</p>
</dd>
</dl>



<h3>Files in extdata/RDP</h3>

<p>Output of RDP Classifier with the default training set.
</p>

<dl>
<dt>&lsquo;<span class="file">pipeline.R</span>&rsquo;</dt><dd>
<p>Pipeline for sequence data processing (uses external programs fastq-dump, vsearch, seqtk, RDP Classifier).
This was used to make the files in both &lsquo;<span class="file">RDP</span>&rsquo; and &lsquo;<span class="file">RDP-GTDB_207</span>&rsquo; (the latter with <code>GTDB = TRUE</code> in the script).
</p>
</dd>
<dt>&lsquo;<span class="file">BGPF13.tab.xz</span>&rsquo;</dt><dd><p>Heart Lake Geyser Basin.</p>
</dd>
<dt>&lsquo;<span class="file">HLA+16.tab.xz</span>&rsquo;</dt><dd><p>Baltic Sea.</p>
</dd>
<dt>&lsquo;<span class="file">SMS+12.tab.xz</span>&rsquo;</dt><dd><p>Bison Pool.</p>
</dd>
</dl>



<h3>Files in extdata/RDP-GTDB_207</h3>

<p>Output of RDP Classifer using the GTDB r207 SSU training set (<a href="https://doi.org/10.5281/zenodo.7633100">doi:10.5281/zenodo.7633100</a>).
</p>

<dl>
<dt>&lsquo;<span class="file">BGPF13.tab.xz</span>&rsquo;</dt><dd><p>Heart Lake Geyser Basin.</p>
</dd>
<dt>&lsquo;<span class="file">HLA+16.tab.xz</span>&rsquo;</dt><dd><p>Baltic Sea.</p>
</dd>
<dt>&lsquo;<span class="file">SMS+12.tab.xz</span>&rsquo;</dt><dd><p>Bison Pool.</p>
</dd>
</dl>



<h3>Files in extdata/DADA2-GTDB_214</h3>

<p>Identification and taxonomic classification of sequences using DADA2 with GTDB r214.
</p>

<dl>
<dt>&lsquo;<span class="file">FEN+22</span>&rsquo;</dt><dd>
<p>Analysis of data from Fonseca et al. (2022) for marine sediment from the Humboldt Sulfuretum.
&lsquo;<span class="file">pipeline.R</span>&rsquo; has the commands used to process the 16S rRNA gene sequence data and was adapted by Jeffrey Dick from the DADA2 pipeline tutorial (Callahan, 2020).
&lsquo;<span class="file">SraRunInfo.csv</span>&rsquo; was obtained from the NCBI Sequence Read Archive (SRA) (<a href="https://www.ncbi.nlm.nih.gov/sra/?term=PRJNA251688">https://www.ncbi.nlm.nih.gov/sra/?term=PRJNA251688</a>).
&lsquo;<span class="file">sample_data.csv</span>&rsquo; has data obtained from NCBI BioSample records for BioProject PRJNA251688.
&lsquo;<span class="file">*.png</span>&rsquo; are several plots created while running the DADA2 pipeline.
&lsquo;<span class="file">ps_FEN+22.rds</span>&rsquo; contains the phyloseq object with (including <code>otu_table</code>, <code>sample_data</code>, and <code>refseq</code> objects) created at the end of the DADA2 pipeline.
</p>
</dd>
</dl>


<dl>
<dt>&lsquo;<span class="file">ZFZ+23</span>&rsquo;</dt><dd>
<p>Analysis of data from Zhang et al. (2023) for hot springs in the Qinghai-Tibet Plateau.
&lsquo;<span class="file">pipeline.R</span>&rsquo; has the commands used to process the 16S rRNA gene sequence data and was adapted by Jeffrey Dick from the DADA2 pipeline tutorial (Callahan, 2020).
&lsquo;<span class="file">SraRunInfo.csv</span>&rsquo; was obtained from the NCBI Sequence Read Archive (SRA) (<a href="https://www.ncbi.nlm.nih.gov/sra/?term=PRJNA860942">https://www.ncbi.nlm.nih.gov/sra/?term=PRJNA860942</a>).
&lsquo;<span class="file">sample_data.csv</span>&rsquo; has data obtained from NCBI BioSample records for BioProject PRJNA860942.
&lsquo;<span class="file">*.png</span>&rsquo; are several plots created while running the DADA2 pipeline.
&lsquo;<span class="file">ps_ZFZ+23.rds</span>&rsquo; contains the phyloseq object with (including <code>otu_table</code>, <code>sample_data</code>, and <code>refseq</code> objects) created at the end of the DADA2 pipeline.
</p>
</dd>
</dl>



<h3>References</h3>

<p>Bowen De León K, Gerlach R, Peyton BM, Fields MW. 2013. Archaeal and bacterial communities in three alkaline hot springs in Heart Lake Geyser Basin, Yellowstone National Park. <em>Frontiers in Microbiology</em> <b>4</b>: 330. <a href="https://doi.org/10.3389/fmicb.2013.00330">doi:10.3389/fmicb.2013.00330</a>
</p>
<p>Callahan B. 2020. DADA2 Pipeline Tutorial (1.16). <a href="https://benjjneb.github.io/dada2/tutorial.html">https://benjjneb.github.io/dada2/tutorial.html</a>, accessed on 2023-06-14.
</p>
<p>Dick JM, Meng D. 2023. Community- and genome-based evidence for a shaping influence of redox potential on bacterial protein evolution. <em>mSystems</em> <b>8</b>(3): e00014-23. <a href="https://doi.org/10.1128/msystems.00014-23">doi:10.1128/msystems.00014-23</a>
</p>
<p>Dick JM, Tan J. 2023. Chemical links between redox conditions and estimated community proteomes from 16S rRNA and reference protein sequences. <em>Microbial Ecology</em> <b>85</b>: 1338&ndash;1355. <a href="https://doi.org/10.1007/s00248-022-01988-9">doi:10.1007/s00248-022-01988-9</a>
</p>
<p>Fonseca A, Espinoza C, Nielsen LP, Marshall IPG, Gallardo VA. 2022. Bacterial community of sediments under the Eastern Boundary Current System shows high microdiversity and a latitudinal spatial pattern. <em>Frontiers in Microbiology</em> <b>13</b>: 1016418. <a href="https://doi.org/10.3389/fmicb.2022.1016418">doi:10.3389/fmicb.2022.1016418</a>
</p>
<p>Herlemann DPR, Lundin D, Andersson AF, Labrenz M, Jürgens K. 2016. Phylogenetic signals of salinity and season in bacterial community composition across the salinity gradient of the Baltic Sea. <em>Frontiers in Microbiology</em> <b>7</b>: 1883. <a href="https://doi.org/10.3389/fmicb.2016.01883">doi:10.3389/fmicb.2016.01883</a>
</p>
<p>O'Leary NA et al. 2016. Reference sequence (RefSeq) database at NCBI: current status, taxonomic expansion, and functional annotation. <em>Nucleic Acids Research</em> <b>44</b>: D733-D745. <a href="https://doi.org/10.1093/nar/gkv1189">doi:10.1093/nar/gkv1189</a>
</p>
<p>Parks DH, Chuvochina M, Rinke C, Mussig AJ, Chaumeil P-A, Hugenholtz P. 2022. GTDB: an ongoing census of bacterial and archaeal diversity through a phylogenetically consistent, rank normalized and complete genome-based taxonomy. <em>Nucleic Acids Research</em> <b>50</b>: D785&ndash;D794. <a href="https://doi.org/10.1093/nar/gkab776">doi:10.1093/nar/gkab776</a>
</p>
<p>Swingley WD, Meyer-Dombard DR, Shock EL, Alsop EB, Falenski HD, Havig JR, Raymond J. 2012. Coordinating environmental genomics and geochemistry reveals metabolic transitions in a hot spring ecosystem. <em>PLOS One</em> <b>7</b>(6): e38108. <a href="https://doi.org/10.1371/journal.pone.0038108">doi:10.1371/journal.pone.0038108</a>
</p>
<p>Zhang H-S, Feng Q-D, Zhang D-Y, Zhu G-L, Yang L. 2023. Bacterial community structure in geothermal springs on the northern edge of Qinghai-Tibet plateau. <em>Frontiers in Microbiology</em> <b>13</b>: 994179. <a href="https://doi.org/10.3389/fmicb.2022.994179">doi:10.3389/fmicb.2022.994179</a>
</p>

<hr>
<h2 id='chem16S-data'>'phyloseq-class' objects generated using the DADA2 Pipeline Tutorial</h2><span id='topic+mouse.silva'></span><span id='topic+mouse.RDP'></span><span id='topic+mouse.GTDB_214'></span>

<h3>Description</h3>

<p>Objects in <code><a href="phyloseq.html#topic+phyloseq-class">phyloseq-class</a></code> format created by following the DADA2 Pipeline Tutorial (Callahan, 2020).
</p>


<h3>Details</h3>

<p>The example dataset for gut communities in a single mouse was taken from the <a href="https://mothur.org/wiki/miseq_sop/">mothur MiSeq SOP</a>.
It is an extract of the complete dataset reported by Schloss et al. (2012).
The output files were generated by using three different training sets to assign taxonomy to genus level:
</p>

<dl>
<dt>&lsquo;<span class="file">mouse.silva</span>&rsquo;</dt><dd><p>silva_nr99_v138.1: <a href="https://zenodo.org/record/4587955">Silva 138.1 prokaryotic SSU taxonomic training data formatted for DADA2</a></p>
</dd>
<dt>&lsquo;<span class="file">mouse.RDP</span>&rsquo;</dt><dd><p>rdp_train_set_18: <a href="https://zenodo.org/record/4310151">RDP taxonomic training data formatted for DADA2 (RDP trainset 18/release 11.5)</a></p>
</dd>
<dt>&lsquo;<span class="file">mouse.GTDB_214</span>&rsquo;</dt><dd><p>GTDB_bac120_arc53_ssu_r214: <a href="https://zenodo.org/record/10403693">GTDB: DADA2 formatted 16S rRNA gene sequences for both bacteria &amp; archaea</a></p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Jeffrey M. Dick, following the DADA2 Pipeline Tutorial by Callahan (2020).
</p>


<h3>References</h3>

<p>Callahan B (2020). DADA2 Pipeline Tutorial (1.16). <a href="https://benjjneb.github.io/dada2/tutorial.html">https://benjjneb.github.io/dada2/tutorial.html</a>, accessed on 2023-06-14. Publication year taken from <a href="https://code.bioconductor.org/browse/dada2/blob/RELEASE_3_11/DESCRIPTION">https://code.bioconductor.org/browse/dada2/blob/RELEASE_3_11/DESCRIPTION</a>.
</p>
<p>Schloss PD, Schubert AM, Zackular JP, Iverson KD, Young VB, Petrosino JF. 2012. Stabilization of the murine gut microbiome following weaning. <em>Gut Microbes</em> <b>3</b>(4): 383–393. <a href="https://doi.org/10.4161/gmic.21008">doi:10.4161/gmic.21008</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load one data set
data(mouse.silva)
# Summarize the object (phyloseq-class)
mouse.silva
</code></pre>

<hr>
<h2 id='chemlab'>Formatted labels for chemical metrics</h2><span id='topic+chemlab'></span>

<h3>Description</h3>

<p>Formatted labels for indicating chemical metrics on plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  chemlab(varname)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chemlab_+3A_varname">varname</code></td>
<td>
<p>character, the name of a variable</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function provides formatted labels for chemical metrics and related variables.
<code>varname</code> can be any one of the chemical metrics described in <code><a href="canprot.html#topic+calc_metrics">calc_metrics</a></code>.
</p>


<h3>Value</h3>

<p>For most values of <code>varname</code>, a language object suitable for plotting (see <code><a href="grDevices.html#topic+plotmath">plotmath</a></code>).
For labels where no special mathematical notation is required, a character object.
If <code>varname</code> is not among the chemical metrics described in <code><a href="canprot.html#topic+calc_metrics">calc_metrics</a></code>, the value of <code>varname</code> itself.
</p>


<h3>See Also</h3>

<p><code><a href="canprot.html#topic+cplab">cplab</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>chemlab("Zc")
</code></pre>

<hr>
<h2 id='get_metadata'>Template function for metadata</h2><span id='topic+get_metadata'></span>

<h3>Description</h3>

<p>Simple function that processes metadata files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  get_metadata(file, metrics)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_metadata_+3A_file">file</code></td>
<td>
<p>character, metadata file (CSV format)</p>
</td></tr>
<tr><td><code id="get_metadata_+3A_metrics">metrics</code></td>
<td>
<p>data frame, output of <code>get_metrics</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is provided as a simple example of reading metadata about 16S rRNA datasets.
The basic metadata for all datasets include &lsquo;<span class="samp">&#8288;name&#8288;</span>&rsquo; (study name), &lsquo;<span class="samp">&#8288;Run&#8288;</span>&rsquo; (SRA or other run ID), &lsquo;<span class="samp">&#8288;BioSample&#8288;</span>&rsquo; (SRA BioSample), and &lsquo;<span class="samp">&#8288;Sample&#8288;</span>&rsquo; (sample description).
Each data set contains environmental data, but the specific variables differ between datasets.
Therefore, this function processes the environmental data for different datasets and appends &lsquo;<span class="samp">&#8288;pch&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;col&#8288;</span>&rsquo; columns that can be used for visualization and identifying trends between groups of samples.
</p>
<p>Samples are identified by the <code>Run</code> column (case-sensitive) in both <code>file</code> and <code>metrics</code>.
It is possible that metrics for one or more samples are not available (for example, if the number of RDP classifications is below <code>mincount</code> in <code><a href="#topic+get_metrics">get_metrics</a></code>).
The function therefore removes metadata for unavailable samples.
Then, the samples in <code>metrics</code> are placed in the same order as they appear in metadata.
Ordering the samples is important because the columns in RDP Classifier output are not necessarily in the same order as the metadata.
</p>


<h3>Value</h3>

<p>If <code>metrics</code> is NULL, a data frame read from <code>file</code> with columns &lsquo;<span class="samp">&#8288;pch&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;col&#8288;</span>&rsquo; appended.
Otherwise, a list with two components, <code>metadata</code> and <code>metrics</code>.
In this list, <code>metrics</code> is the value from the <code>metrics</code> argument and <code>metadata</code> is a data frame read from <code>file</code> with columns &lsquo;<span class="samp">&#8288;pch&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;col&#8288;</span>&rsquo; appended and excluding any rows that correspond to unavailable samples in <code>metrics</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get metrics for the Bison Pool dataset (Swingley et al., 2012)
RDPfile &lt;- system.file("extdata/RDP/SMS+12.tab.xz", package = "chem16S")
RDP &lt;- read_RDP(RDPfile)
map &lt;- map_taxa(RDP, refdb = "RefSeq_206")
metrics &lt;- get_metrics(RDP, map, refdb = "RefSeq_206")
# Read the metadata file
mdatfile &lt;- system.file("extdata/metadata/SMS+12.csv", package = "chem16S")
mdat &lt;- get_metadata(mdatfile, metrics)

# Print sample names
mdat$metadata$Sample # 1 2 3 4 5
# Plot Zc of community reference proteomes at sites 1-5
Site &lt;- mdat$metadata$Sample
Zc &lt;- mdat$metrics$Zc
plot(Site, Zc, ylab = "Carbon oxidation state", type = "b")

# Find the columns of the RDP file for samples 1 to 5
match(mdat$metadata$Run, colnames(RDP)) # 6 5 7 8 9
# NOTE: for this dataset, samples 1 and 2 are in columns 6 and 5, respectively
</code></pre>

<hr>
<h2 id='get_metrics'>Calculate chemical metrics of community reference proteomes</h2><span id='topic+get_metrics'></span>

<h3>Description</h3>

<p>Combines taxonomic classifications with reference proteomes for taxa to get amino acid compositions of community reference proteomes.
Amino acid compositions are used to calculate chemical metrics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  get_metrics(RDP, map, refdb = "GTDB_214", taxon_AA = NULL, groups = NULL,
    return_AA = FALSE, zero_AA = NULL, metrics = c("Zc", "nO2", "nH2O"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_metrics_+3A_rdp">RDP</code></td>
<td>
<p>data frame, taxonomic abundances produced by <code><a href="#topic+read_RDP">read_RDP</a></code> or <code><a href="#topic+ps_taxacounts">ps_taxacounts</a></code></p>
</td></tr>
<tr><td><code id="get_metrics_+3A_map">map</code></td>
<td>
<p>data frame, taxonomic mapping produced by <code><a href="#topic+map_taxa">map_taxa</a></code></p>
</td></tr>
<tr><td><code id="get_metrics_+3A_refdb">refdb</code></td>
<td>
<p>character, name of reference database (&lsquo;<span class="samp">&#8288;GTDB_214&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;RefSeq_206&#8288;</span>&rsquo;)</p>
</td></tr>
<tr><td><code id="get_metrics_+3A_taxon_aa">taxon_AA</code></td>
<td>
<p>data frame, amino acid compositions of taxa, used to bypass <code>refdb</code> specification</p>
</td></tr>
<tr><td><code id="get_metrics_+3A_groups">groups</code></td>
<td>
<p>list of indexing vectors, samples to be aggregated into groups</p>
</td></tr>
<tr><td><code id="get_metrics_+3A_return_aa">return_AA</code></td>
<td>
<p>logical, return the amino acid composition for each sample instead of the chemical metrics?</p>
</td></tr>
<tr><td><code id="get_metrics_+3A_zero_aa">zero_AA</code></td>
<td>
<p>character, three-letter abbreviation(s) of amino acid(s) to assign zero counts for calculating chemical metrics</p>
</td></tr>
<tr><td><code id="get_metrics_+3A_metrics">metrics</code></td>
<td>
<p>character, the chemical metrics to calculate</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>get_metrics</code> calculates selected chemical metrics for the community reference proteome in each sample.
The community reference proteome is computed from the amino acid compositions of reference proteomes for taxa (obtained from the reference database in <code>refdb</code>), multiplied by taxonomic abundances given in <code>RDP</code>.
<code>RDP</code> may include results from the RDP Classifier (read using <code><a href="#topic+read_RDP">read_RDP</a></code>) or derived from the OTU table of a <code><a href="phyloseq.html#topic+phyloseq-class">phyloseq-class</a></code> object (see <code><a href="#topic+ps_taxacounts">ps_taxacounts</a></code>).
<code>map</code> defines the taxonomic mapping between <code>RDP</code> and <code>refdb</code>.
Then, chemical metrics are calculated from the amino acid composition of the community reference proteome.
The default chemical metrics are carbon oxidation state (<I>Z</I><sub>C</sub>), stoichiometric oxidation state (<i>n</i><sub>O<sub>2</sub></sub>), and stoichiometric hydration state (<i>n</i><sub>H<sub>2</sub>O</sub>).
See <code><a href="canprot.html#topic+calc_metrics">calc_metrics</a></code> for other available metrics.
</p>
<p><code>groups</code>, if given, is a list of one or more indexing vectors (with logical or numeric values) corresponding to samples whose taxonomic classifications are aggregated into groups before calculating amino acid compositions and chemical metrics.
</p>


<h3>Value</h3>

<p>A data frame with one row for each sample, corresponding to columns 5 and above of <code>RDP</code>.
The sample names are in the first column, which is named <code>Run</code> by default, or <code>group</code> if the <code>groups</code> argument is provided.
The remaining columns have numeric values and are named for each of the calculated <code>metrics</code>.
</p>


<h3>See Also</h3>

<p><code><a href="canprot.html#topic+calc_metrics">calc_metrics</a></code>, <code><a href="#topic+plot_metrics">plot_metrics</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## First two examples are for RDP Classifier with default training set
## and mapping to NCBI taxonomy with RefSeq reference proteomes

# Get chemical metrics for all samples in a dataset
RDPfile &lt;- system.file("extdata/RDP/BGPF13.tab.xz", package = "chem16S")
RDP &lt;- read_RDP(RDPfile)
map &lt;- map_taxa(RDP, refdb = "RefSeq_206")
# This is a data frame with 14 rows and Run, Zc, nO2, and nH2O columns
(metrics &lt;- get_metrics(RDP, map, refdb = "RefSeq_206"))

# Read the metadata file
mdatfile &lt;- system.file("extdata/metadata/BGPF13.csv", package = "chem16S")
# Create list with metadata and metrics in same sample order
mdat &lt;- get_metadata(mdatfile, metrics)
# Calculate metrics for aggregated samples of Archaea and Bacteria
groups &lt;- list(A = mdat$metadata$domain == "Archaea",
  B = mdat$metadata$domain == "Bacteria")
# This is a data frame with 2 rows and group, Zc, nO2, and nH2O columns
get_metrics(RDP, map, refdb = "RefSeq_206", groups = groups)

## For the next example, classifications were made using the RDP Classifer
## with 16S rRNA sequences from GTDB r207, but
## reference proteomes are based on GTDB r214
## (this uses the default option of refdb = "GTDB_214").
RDPfile.GTDB &lt;- system.file("extdata/RDP-GTDB_207/BGPF13.tab.xz", package = "chem16S")
RDP.GTDB &lt;- read_RDP(RDPfile.GTDB)
map.GTDB &lt;- map_taxa(RDP.GTDB)
metrics.GTDB &lt;- get_metrics(RDP.GTDB, map.GTDB)

# Plot Zc from GTDB vs RefSeq
xylim &lt;- range(metrics$Zc, metrics.GTDB$Zc)
plot(metrics$Zc, metrics.GTDB$Zc, xlim = xylim, ylim = xylim, type = "n")
lines(xylim, xylim, lty = 2, col = 8)
points(metrics$Zc, metrics.GTDB$Zc, pch = mdat$metadata$pch, bg = mdat$metadata$col)
md.leg &lt;- mdat$metadata[1:2, ]
legend("bottomright", md.leg$domain, pch = md.leg$pch, pt.bg = md.leg$col)
title(quote(italic(Z)[C]~"from GTDB vs RefSeq"))

# To exclude tryptophan, tyrosine, and phenylalanine
# from the calculation of chemical metrics
get_metrics(RDP.GTDB, map.GTDB, zero_AA = c("Trp", "Tyr", "Phe"))
</code></pre>

<hr>
<h2 id='map_taxa'>Map taxonomic names to NCBI or GTDB taxonomy</h2><span id='topic+map_taxa'></span>

<h3>Description</h3>

<p>Maps taxonomic names to NCBI (RefSeq) or GTDB taxonomy by automatic matching of taxonomic names, with manual mappings for some groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  map_taxa(taxacounts = NULL, refdb = "GTDB_214", taxon_AA = NULL, quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="map_taxa_+3A_taxacounts">taxacounts</code></td>
<td>
<p>data frame with taxonomic name and abundances</p>
</td></tr>
<tr><td><code id="map_taxa_+3A_refdb">refdb</code></td>
<td>
<p>character, name of reference database (&lsquo;<span class="samp">&#8288;GTDB_214&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;RefSeq_206&#8288;</span>&rsquo;)</p>
</td></tr>
<tr><td><code id="map_taxa_+3A_taxon_aa">taxon_AA</code></td>
<td>
<p>data frame, amino acid compositions of taxa, used to bypass <code>refdb</code> specification</p>
</td></tr>
<tr><td><code id="map_taxa_+3A_quiet">quiet</code></td>
<td>
<p>logical, suppress printed messages?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function maps taxonomic names to the NCBI (RefSeq) or GTDB taxonomy.
<code>taxacounts</code> should be a data frame generated by either <code><a href="#topic+read_RDP">read_RDP</a></code> or <code><a href="#topic+ps_taxacounts">ps_taxacounts</a></code>.
Input names are made by combining the taxonomic rank and name with an underscore separator (e.g. &lsquo;<span class="samp">&#8288;genus_ Escherichia/Shigella&#8288;</span>&rsquo;).
Input names are then matched to the taxa listed in &lsquo;<span class="file">taxon_AA.csv.xz</span>&rsquo; found under &lsquo;<span class="file">RefDB/RefSeq_206</span>&rsquo; or &lsquo;<span class="file">RefDB/GTDB_214</span>&rsquo;.
The <code>protein</code> and <code>organism</code> columns in these files hold the rank and taxon name extracted from the RefSeq or GTDB database.
Only exactly matching names are automatically mapped.
</p>
<p>For mapping to the NCBI (RefSeq) taxonomy, some group names are manually mapped as follows (see Dick and Tan, 2023):
</p>

<table>
<tr>
 <td style="text-align: left;">
    <strong>RDP training set</strong> </td><td style="text-align: left;"> <strong>NCBI</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
    genus_Escherichia/Shigella </td><td style="text-align: left;"> genus_Escherichia </td>
</tr>
<tr>
 <td style="text-align: left;">
    phylum_Cyanobacteria/Chloroplast </td><td style="text-align: left;"> phylum_Cyanobacteria </td>
</tr>
<tr>
 <td style="text-align: left;">
    genus_Marinimicrobia_genera_incertae_sedis </td><td style="text-align: left;"> species_Candidatus Marinimicrobia bacterium </td>
</tr>
<tr>
 <td style="text-align: left;">
    class_Cyanobacteria </td><td style="text-align: left;"> phylum_Cyanobacteria </td>
</tr>
<tr>
 <td style="text-align: left;">
    genus_Spartobacteria_genera_incertae_sedis </td><td style="text-align: left;"> species_Spartobacteria bacterium LR76 </td>
</tr>
<tr>
 <td style="text-align: left;">
    class_Planctomycetacia </td><td style="text-align: left;"> class_Planctomycetia </td>
</tr>
<tr>
 <td style="text-align: left;">
    class_Actinobacteria </td><td style="text-align: left;"> phylum_Actinobacteria </td>
</tr>
<tr>
 <td style="text-align: left;">
    order_Rhizobiales </td><td style="text-align: left;"> order_Hyphomicrobiales </td>
</tr>
<tr>
 <td style="text-align: left;">
    genus_Gp1 </td><td style="text-align: left;"> genus_Acidobacterium </td>
</tr>
<tr>
 <td style="text-align: left;">
    genus_Gp6 </td><td style="text-align: left;"> genus_Luteitalea </td>
</tr>
<tr>
 <td style="text-align: left;">
    genus_GpI </td><td style="text-align: left;"> genus_Nostoc </td>
</tr>
<tr>
 <td style="text-align: left;">
    genus_GpIIa </td><td style="text-align: left;"> genus_Synechococcus </td>
</tr>
<tr>
 <td style="text-align: left;">
    genus_GpVI </td><td style="text-align: left;"> genus_Pseudanabaena </td>
</tr>
<tr>
 <td style="text-align: left;">
    family_Family II </td><td style="text-align: left;"> family_Synechococcaceae </td>
</tr>
<tr>
 <td style="text-align: left;">
    genus_Subdivision3_genera_incertae_sedis </td><td style="text-align: left;"> family_Verrucomicrobia subdivision 3 </td>
</tr>
<tr>
 <td style="text-align: left;">
    order_Clostridiales </td><td style="text-align: left;"> order_Eubacteriales </td>
</tr>
<tr>
 <td style="text-align: left;">
    family_Ruminococcaceae </td><td style="text-align: left;"> family_Oscillospiraceae
</td>
</tr>

</table>

<p>To avoid manual mapping, GTDB can be used for both taxonomic assignemnts and reference proteomes.
Taxonomic assignments based on 16S rRNA sequences from GTDB can be made using training files for the RDP Classifier (<a href="https://doi.org/10.5281/zenodo.7633099">doi:10.5281/zenodo.7633099</a>) or <span class="pkg">dada2</span> (<a href="https://doi.org/10.5281/zenodo.2541238">doi:10.5281/zenodo.2541238</a>) (make sure to choose the appropriate GTDB version).
Example files created using the RDP Classifier are provided under &lsquo;<span class="file">extdata/RDP-GTDB_207</span>&rsquo;.
An example dataset created with DADA2 is <code>data(<a href="#topic+mouse.GTDB_214">mouse.GTDB_214</a>)</code>; this is a <code><a href="phyloseq.html#topic+phyloseq-class">phyloseq-class</a></code> object that can be processed with functions described at <code><a href="#topic+physeq">physeq</a></code>.
</p>
<p>Change <code>quiet</code> to TRUE to suppress printing of messages about manual mappings, most abundant unmapped groups, and overall percentage of mapped names.
</p>


<h3>Value</h3>

<p>Integer vector with length equal to number of rows of <code>taxacounts</code>.
Values are rownumbers in the data frame generated by reading <code>taxon_AA.csv.xz</code>, or NA for no matching taxon.
Attributes <code>unmapped_groups</code> and <code>unmapped_percent</code> have the input names of unmapped groups and their percentage of the total classification count.
</p>


<h3>References</h3>

<p>Dick JM, Tan J. 2023. Chemical links between redox conditions and estimated community proteomes from 16S rRNA and reference protein sequences. <em>Microbial Ecology</em> <b>85</b>: 1338&ndash;1355. <a href="https://doi.org/10.1007/s00248-022-01988-9">doi:10.1007/s00248-022-01988-9</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Mapping taxonomic classifications from RDP training set to NCBI taxonomy
file &lt;- system.file("extdata/RDP/SMS+12.tab.xz", package = "chem16S")
RDP &lt;- read_RDP(file)
map &lt;- map_taxa(RDP, refdb = "RefSeq_206")
# About 24% of classifications are unmapped
sum(attributes(map)$unmapped_percent)

# Mapping from GTDB training set to GTDB taxonomy
file &lt;- system.file("extdata/RDP-GTDB_207/SMS+12.tab.xz", package = "chem16S")
RDP.GTDB &lt;- read_RDP(file)
map.GTDB &lt;- map_taxa(RDP.GTDB)
# The classifications were made with GTDB r207, but the reference database 
# in chem16S has been updated to GTDB r214, so the mapping rate is less than 100 %
sum(attributes(map.GTDB)$unmapped_percent)
</code></pre>

<hr>
<h2 id='physeq'>Process 'phyloseq-class' objects to calculate chemical metrics</h2><span id='topic+physeq'></span><span id='topic+ps_taxacounts'></span><span id='topic+ps_metrics'></span><span id='topic+plot_ps_metrics'></span><span id='topic+plot_ps_metrics2'></span>

<h3>Description</h3>

<p>Calculate chemical metrics of community reference proteomes from taxonomic abundances in <code><a href="phyloseq.html#topic+phyloseq-class">phyloseq-class</a></code> objects and make plots of individual metrics or two metrics against each other.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  ps_taxacounts(physeq, split = TRUE)
  ps_metrics(physeq, split = TRUE, refdb = "GTDB_214", quiet = FALSE, ...)
  plot_ps_metrics(physeq, metrics = c("Zc", "nO2", "nH2O"), x = "samples",
    color = NULL, shape = NULL, title = NULL,
    scales = "free_y", nrow = 1, sortby = NULL, ...)
  plot_ps_metrics2(physeq, metrics = c("Zc", "nH2O"), color = NULL,
    shape = NULL, title = NULL, refdb = "GTDB_214", quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="physeq_+3A_physeq">physeq</code></td>
<td>
<p>'phyloseq-class', a <span class="pkg">phyloseq</span> object containing an 'otu_table' and 'tax_table'</p>
</td></tr>
<tr><td><code id="physeq_+3A_split">split</code></td>
<td>
<p>logical, calculate metrics for each sample (TRUE) or the entire set (FALSE)</p>
</td></tr>
<tr><td><code id="physeq_+3A_refdb">refdb</code></td>
<td>
<p>character, name of reference database (passed to <code><a href="#topic+get_metrics">get_metrics</a></code>)</p>
</td></tr>
<tr><td><code id="physeq_+3A_quiet">quiet</code></td>
<td>
<p>logical, suppress printed messages? (passed to <code><a href="#topic+map_taxa">map_taxa</a></code>)</p>
</td></tr>
<tr><td><code id="physeq_+3A_...">...</code></td>
<td>
<p>additional arguments passed to <code>ps_metrics</code> and thence to <code><a href="#topic+get_metrics">get_metrics</a></code></p>
</td></tr>
<tr><td><code id="physeq_+3A_metrics">metrics</code></td>
<td>
<p>character, chemical metrics to calculate</p>
</td></tr>
<tr><td><code id="physeq_+3A_x">x</code></td>
<td>
<p>map this variable to the horizontal axis</p>
</td></tr>
<tr><td><code id="physeq_+3A_color">color</code></td>
<td>
<p>map this variable to colors</p>
</td></tr>
<tr><td><code id="physeq_+3A_shape">shape</code></td>
<td>
<p>map this variable to shapes</p>
</td></tr>
<tr><td><code id="physeq_+3A_title">title</code></td>
<td>
<p>main title for the plot</p>
</td></tr>
<tr><td><code id="physeq_+3A_scales">scales</code></td>
<td>
<p>&lsquo;<span class="samp">&#8288;free_y&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;fixed&#8288;</span>&rsquo; for different vertical scales or same scale in each panel</p>
</td></tr>
<tr><td><code id="physeq_+3A_nrow">nrow</code></td>
<td>
<p>number of rows of panels (passed to <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>)</p>
</td></tr>
<tr><td><code id="physeq_+3A_sortby">sortby</code></td>
<td>
<p>subset of <code>metrics</code> argument, sort x-index by the mean values of these metrics</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>ps_taxacounts</code> returns a data frame with lowest-level classifications (from genus to phylum) and abundances for each OTU in the <code>physeq</code> object.
</p>
<p><code>ps_metrics</code> calculates chemical metrics of community reference proteomes from the lowest-level classifications after taxonomic mapping with <code><a href="#topic+map_taxa">map_taxa</a></code>.
</p>
<p><code>plot_ps_metrics</code> plots individual chemical metrics or multiple chemical metrics, each in their own panel.
This function is modified from <code><a href="phyloseq.html#topic+plot_richness">plot_richness</a></code>; see that help page for further details about graphics-related arguments.
</p>
<p><code>plot_ps_metrics2</code> plot two chemical metrics against each other.
</p>
<p>The <code>metrics</code> argument is passed to <code><a href="#topic+get_metrics">get_metrics</a></code> and selects the metrics to calculate.
<code>split</code>, <code>refdb</code>, and <code>quiet</code> can also be used as additional arguments in <code>plot_ps_metrics</code> and <code>plot_ps_metrics2</code>.
</p>


<h3>Value</h3>

<p>For <code>ps_taxacounts</code>, a data frame with samples in rows, and columns <code>taxid</code> (sample name taken from the OTU table obtained from <code><a href="phyloseq.html#topic+otu_table">otu_table</a>(physeq)</code>), <code>lineage</code> (all taxonomic names from the taxonomy table obtained from <code><a href="phyloseq.html#topic+tax_table">tax_table</a>(physeq)</code>, separated by semicolons), <code>name</code> (lowest-level taxonomic name), <code>rank</code> (rank of lowest-level taxon), followed by the abundances from the OTU table.
</p>
<p>For <code>ps_metrics</code>, a data frame with samples in rows, and columns corresponding to each of the <code>metrics</code>.
</p>
<p>For <code>plot_ps_metrics</code> and <code>plot_ps_metrics2</code>, ggplot objects.
</p>


<h3>Author(s)</h3>

<p>The author of this Rd file and the <code>ps_taxacounts</code> and <code>ps_metrics</code> functions is Jeffrey Dick.
<code>plot_ps_metrics</code> is based on the <code><a href="phyloseq.html#topic+plot_richness">plot_richness</a></code> function by Paul J. McMurdie and was modified from that function by Jeffrey Dick.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1: Humboldt Sulfuretum (Fonseca et al., 2022)
file &lt;- system.file("extdata/DADA2-GTDB_214/FEN+22/ps_FEN+22.rds", package = "chem16S")
physeq &lt;- readRDS(file)
# Get lowest-level (to genus) classification for each OTU
taxacounts &lt;- ps_taxacounts(physeq)
# Show numbers of assignments at each taxonomic level
table(taxacounts$rank)
# Map taxonomic names
map &lt;- map_taxa(taxacounts, refdb = "GTDB_214")
# Taxonomy and reference proteomes are both from GTDB,
# so there are no unmapped classifications
sum(attr(map, "unmapped_percent")) # == 0
# Calculate chemical metrics
ps_metrics(physeq, refdb = "GTDB_214")

# Example 2: GlobalPatterns dataset from phyloseq
data(GlobalPatterns, package = "phyloseq")
# Plot metrics grouped by sample type and sorted by mean Zc;
# refdb = "RefSeq" uses manual mappings from the RDP to NCBI taxonomy
p &lt;- plot_ps_metrics(GlobalPatterns, x = "SampleType",
                     sortby = "Zc", refdb = "RefSeq_206")
# Change orientation of x-axis labels
p + ggplot2::theme(axis.text.x = ggplot2::element_text(
  angle = 45, vjust = 1, hjust = 1))
</code></pre>

<hr>
<h2 id='plot_metrics'>Plot chemical metrics of community reference proteomes</h2><span id='topic+plot_metrics'></span>

<h3>Description</h3>

<p>Functions to plot chemical metrics of community reference proteomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  plot_metrics(mdat,
    xvar = "Zc", yvar = "nH2O",
    xlim = NULL, ylim = NULL,
    xlab = chemlab(xvar), ylab = chemlab(yvar),
    plot.it = TRUE, add = FALSE, identify = FALSE,
    points = TRUE, lines = FALSE, title = TRUE,
    cex = 1, pt.open.col = 1, pch1 = 1, pch2 = 21,
    return = "data", extracolumn = NULL
  )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_metrics_+3A_mdat">mdat</code></td>
<td>
<p>list, output by <code><a href="#topic+get_metadata">get_metadata</a></code></p>
</td></tr>
<tr><td><code id="plot_metrics_+3A_xvar">xvar</code></td>
<td>
<p>character, name of x variable</p>
</td></tr>
<tr><td><code id="plot_metrics_+3A_yvar">yvar</code></td>
<td>
<p>character, name of y variable</p>
</td></tr>
<tr><td><code id="plot_metrics_+3A_xlim">xlim</code></td>
<td>
<p>numeric, x axis limits</p>
</td></tr>
<tr><td><code id="plot_metrics_+3A_ylim">ylim</code></td>
<td>
<p>numeric, y axis limits</p>
</td></tr>
<tr><td><code id="plot_metrics_+3A_xlab">xlab</code></td>
<td>
<p>x axis label</p>
</td></tr>
<tr><td><code id="plot_metrics_+3A_ylab">ylab</code></td>
<td>
<p>y axis label</p>
</td></tr>
<tr><td><code id="plot_metrics_+3A_plot.it">plot.it</code></td>
<td>
<p>logical, make a plot?</p>
</td></tr>
<tr><td><code id="plot_metrics_+3A_add">add</code></td>
<td>
<p>logical, add to existing plot?</p>
</td></tr>
<tr><td><code id="plot_metrics_+3A_identify">identify</code></td>
<td>
<p>logical, run <code><a href="graphics.html#topic+identify">identify</a></code> for interactive identification of points?</p>
</td></tr>
<tr><td><code id="plot_metrics_+3A_points">points</code></td>
<td>
<p>logical, plot points?</p>
</td></tr>
<tr><td><code id="plot_metrics_+3A_lines">lines</code></td>
<td>
<p>logical, plot lines?</p>
</td></tr>
<tr><td><code id="plot_metrics_+3A_title">title</code></td>
<td>
<p>character, plot title</p>
</td></tr>
<tr><td><code id="plot_metrics_+3A_cex">cex</code></td>
<td>
<p>numeric, point size</p>
</td></tr>
<tr><td><code id="plot_metrics_+3A_pt.open.col">pt.open.col</code></td>
<td>
<p>color of border for open point symbols (<code>pch &gt; 20</code>)</p>
</td></tr>
<tr><td><code id="plot_metrics_+3A_pch1">pch1</code></td>
<td>
<p>numeric, symbol for samples in group 1</p>
</td></tr>
<tr><td><code id="plot_metrics_+3A_pch2">pch2</code></td>
<td>
<p>numeric, symbol for samples in group 2</p>
</td></tr>
<tr><td><code id="plot_metrics_+3A_return">return</code></td>
<td>
<p>character, indicates whether to return &lsquo;<span class="samp">&#8288;data&#8288;</span>&rsquo; values or group &lsquo;<span class="samp">&#8288;means&#8288;</span>&rsquo;</p>
</td></tr>
<tr><td><code id="plot_metrics_+3A_extracolumn">extracolumn</code></td>
<td>
<p>character, the name of one or more extra columns (from <code><a href="#topic+get_metadata">get_metadata</a></code>) to include in the output</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>plot_metrics</code> plots the values of <I>Z</I><sub>C</sub> and <i>n</i><sub>H<sub>2</sub>O</sub> (or other variables indicated by <code>xvar</code> and <code>yvar</code>) provided in <code>mdat$metrics</code>.
Symbol shape and color (<code>pch</code> and <code>col</code>) are taken from <code>mdat$metadata</code>.
</p>
<p>If <code>pch1</code> and <code>pch2</code> are provided, then samples are classified into two groups according to the value of <code>mdat$metadata$pch</code>.
Mean values of the chemical metrics for each group are plotted with star-shaped symbols.
</p>


<h3>Value</h3>

<p>For <code>plot_metrics</code>, a data frame with columns for study name and Run IDs (&lsquo;<span class="samp">&#8288;name&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;Run&#8288;</span>&rsquo;), chemical metrics (taken from <code>mdat$metrics</code>), and symbols and colors for plotting points (&lsquo;<span class="samp">&#8288;pch&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;col&#8288;</span>&rsquo;).
</p>


<h3>References</h3>

<p>Herlemann, D. P. R., Lundin, D., Andersson, A. F., Labrenz, M. and Jürgens, K. (2016) Phylogenetic signals of salinity and season in bacterial community composition across the salinity gradient of the Baltic Sea. <em>Front. Microbiol.</em> <b>7</b>, 1883. <a href="https://doi.org/10.3389/fmicb.2016.01883">doi:10.3389/fmicb.2016.01883</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Make a plot for the Baltic Sea salinity gradient
# (data from Herlemann et al., 2016)
RDPfile &lt;- system.file("extdata/RDP/HLA+16.tab.xz", package = "chem16S")
RDP &lt;- read_RDP(RDPfile)
map &lt;- map_taxa(RDP, refdb = "RefSeq_206")
metrics &lt;- get_metrics(RDP, map, refdb = "RefSeq_206")
mdatfile &lt;- system.file("extdata/metadata/HLA+16.csv", package = "chem16S")
mdat &lt;- get_metadata(mdatfile, metrics)
pm &lt;- plot_metrics(mdat)
# Add a legend
legend &lt;- c("&lt; 6 PSU", "6-20 PSU", "&gt; 20 PSU")
pch &lt;- c(24, 20, 21)
pt.bg &lt;- c(3, NA, 4)
legend("bottomright", legend, pch = pch, col = 1, pt.bg = pt.bg, bg = "white")
# Classify samples with low and high salinity
ilo &lt;- mdat$metadata$salinity &lt; 6
ihi &lt;- mdat$metadata$salinity &gt; 20
# Add convex hulls
canprot::add_hull(pm$Zc[ilo], pm$nH2O[ilo],
  col = adjustcolor("green3", alpha.f = 0.3), border = NA)
canprot::add_hull(pm$Zc[ihi], pm$nH2O[ihi],
  col = adjustcolor("blue", alpha.f = 0.3), border = NA)

# Show points for all samples and larger star-shaped points
# for mean values of high- and low-salinity samples
plot_metrics(mdat, pch1 = 21, pch2 = 24)

# Plot nO2 instead of Zc
plot_metrics(mdat, xvar = "nO2")

# Make a plot for only Proteobacteria
RDP &lt;- read_RDP(RDPfile, lineage = "Proteobacteria")
map &lt;- map_taxa(RDP, refdb = "RefSeq_206")
metrics &lt;- get_metrics(RDP, map, refdb = "RefSeq_206")
mdatfile &lt;- system.file("extdata/metadata/HLA+16.csv", package = "chem16S")
mdat &lt;- get_metadata(mdatfile, metrics)
mdat$metadata$name &lt;- paste(mdat$metadata$name, "(Proteobacteria)")
plot_metrics(mdat)
</code></pre>

<hr>
<h2 id='read_RDP'>Read and filter RDP Classifier output</h2><span id='topic+read_RDP'></span>

<h3>Description</h3>

<p>Reads RDP Classifier output and optionally extracts lineages, removes classifications below a certain count, or truncates classifications to lowest taxonomic level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  read_RDP(file, lineage = NULL, mincount = 200,
    lowest.level = NULL, quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_RDP_+3A_file">file</code></td>
<td>
<p>character, file name</p>
</td></tr>
<tr><td><code id="read_RDP_+3A_lineage">lineage</code></td>
<td>
<p>character, regular expression for filtering lineages</p>
</td></tr>
<tr><td><code id="read_RDP_+3A_mincount">mincount</code></td>
<td>
<p>integer, samples with less than this number of RDP classifications are excluded</p>
</td></tr>
<tr><td><code id="read_RDP_+3A_lowest.level">lowest.level</code></td>
<td>
<p>character, truncate classifications at this taxonomic level</p>
</td></tr>
<tr><td><code id="read_RDP_+3A_quiet">quiet</code></td>
<td>
<p>logical, suppress printed messages?</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>read_RDP</code> reads and filters RDP results for all samples in a study.
The input file should be created by RDP Classifer using the -h option to create a hierarchical listing.
Classifications for multiple samples can be combined into a single file using the merge-count command of RDP Classifier.
</p>
<p>Only rows (lineages) with count &gt; 0 for at least one sample are retained.
Sequences with classification results at only the root or domain level (Root, Archaea, or Bacteria) are omitted because they provide poor taxonomic resolution.
Sequences classified to the class Chloroplast or genera Chlorophyta or Bacillariophyta are also omitted because they have little correspondence with the NCBI taxonomy.
Then, only columns (samples) with classification count &gt;= <code>mincount</code> are retained.
All remaining sequences (those classified to genus or higher levels) can be used for mapping to the NCBI taxonomy.
</p>
<p>The lineage text of the RDP Classifier looks like &ldquo;Root;rootrank;Archaea;domain;Diapherotrites; phylum;Diapherotrites Incertae Sedis AR10;genus;&rdquo;, so you can use <code>lineage = "Archaea"</code> to select the archaeal classifications or <code>lineage = "genus"</code> to select genus-level classifications.
</p>
<p>Use the <code>lowest.level</code> argument to truncate the classifications to a level higher than genus.
This argument does not reduce the number of classifications, but only trims the RDP lineages to the specified level.
This may create duplicate lineages, for which the classification counts are summed, and only unique lineages are present in the returned data frame.
</p>
<p>Change <code>quiet</code> to TRUE to suppress printing of messages about percentage classification to genus level, omitted sequences, and final range of total counts among all samples.
</p>


<h3>Value</h3>

<p>Data frame with columns inherited from <code>file</code> (from RDP output files: <code>taxid</code>, <code>rank</code>, <code>lineage</code>, <code>name</code>, and one column of classification counts for each sample).
If any sample has less than <code>mincount</code> total counts, that column is omitted from the result.
The rows in the result are subset from those in <code>file</code> by filtering operations.
</p>
<p>NB: taxids in RDP files are not NCBI taxids.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># An example file
file &lt;- system.file("extdata/RDP/SMS+12.tab.xz", package = "chem16S")

# Default settings
r0 &lt;- read_RDP(file)

# Suppress messages
r1 &lt;- read_RDP(file, quiet = TRUE)

# Increase minimum count threshold
r2 &lt;- read_RDP(file, mincount = 500)
# This should be 3 (i.e., 3 samples have less than 500 counts)
ncol(r1) - ncol(r2)

# Keep only Archaea
r3 &lt;- read_RDP(file, lineage = "Archaea")
# This should be TRUE
all(grepl("Archaea", r3$lineage))

# Truncate taxonomy at phylum
r4 &lt;- read_RDP(file, lowest.level = "phylum")
# This should be TRUE
all(sapply(strsplit(r4$lineage, ";"), tail, 1) == "phylum")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
