<!DOCTYPE html><html lang="en"><head><title>Help for package ChemoSpec</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ChemoSpec}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ChemoSpec-package'><p>Exploratory Chemometrics for Spectroscopy</p></a></li>
<li><a href='#aov_pcaSpectra'><p>ANOVA-PCA Analysis of Spectra Data</p></a></li>
<li><a href='#aovPCAloadings'><p>Plot aovPCAscores Loadings of a Spectra Object</p></a></li>
<li><a href='#aovPCAscores'><p>Plot ANOVA-PCA Scores from a Spectra Object</p></a></li>
<li><a href='#averageReplicates'><p>Average Replicates in a Spectra Object</p></a></li>
<li><a href='#baselineSpectra'><p>Baseline Correction of a Spectra Object</p></a></li>
<li><a href='#binSpectra'><p>Bin or Bucket a Spectra Object</p></a></li>
<li><a href='#c_pcaSpectra'><p>Classical PCA of Spectra Objects</p></a></li>
<li><a href='#check4Gaps'><p>Check for Discontinuities (Gaps) in a Vector &amp; Optionally Make a Plot</p></a></li>
<li><a href='#chkGraphicsOpt'><p>Check the Graphics Output Option/Mode</p></a></li>
<li><a href='#chkSpectra'><p>Verify the Integrity of a Spectra or Spectra2D Object</p></a></li>
<li><a href='#clupaSpectra'><p>Hierarchical Cluster-Based Peak Alignment on a Spectra Object</p></a></li>
<li><a href='#colorSymbol'><p>Color and Symbols in ChemoSpec and ChemoSpec2D</p></a></li>
<li><a href='#conColScheme'><p>Change the Color Scheme of a Spectra or Spectra2D Object</p></a></li>
<li><a href='#cv_pcaSpectra'><p>Cross-Validation of Classical PCA Results for a Spectra Object</p></a></li>
<li><a href='#evalClusters'><p>Evaluate or Compare the Quality of Clusters Quantitatively</p></a></li>
<li><a href='#files2SpectraObject'><p>Import Data into a Spectra Object</p></a></li>
<li><a href='#hcaScores'><p>HCA on PCA/MIA/PARAFAC scores from a Spectra or Spectra2D Object</p></a></li>
<li><a href='#hcaSpectra'><p>Plot HCA Results of a Spectra Object</p></a></li>
<li><a href='#hmapSpectra'><p>Seriated Heat Map for a Spectra Object</p></a></li>
<li><a href='#hypTestScores'><p>Conduct MANOVA using PCA Scores and Factors in a Spectra Object</p></a></li>
<li><a href='#irlba_pcaSpectra'><p>IRLBA PCA of Spectra Objects</p></a></li>
<li><a href='#mclust3dSpectra'><p>mclust Analysis of a Spectra Object in 3D</p></a></li>
<li><a href='#mclustSpectra'><p>mclust Analysis of a Spectra Object PCA Results</p></a></li>
<li><a href='#metMUD1'><p>Made Up NMR Data Sets</p></a></li>
<li><a href='#normSpectra'><p>Normalize a Spectra Object</p></a></li>
<li><a href='#pcaDiag'><p>Outlier Diagnostic Plots for PCA of a Spectra Object</p></a></li>
<li><a href='#plot2Loadings'><p>Plot PCA Loadings from a Spectra Object Against Each Other</p></a></li>
<li><a href='#plot3dScores'><p>3D PCA Score Plot for a Spectra Object</p></a></li>
<li><a href='#plotLoadings'><p>Plot PCA Loadings for a Spectra Object</p></a></li>
<li><a href='#plotScores'><p>Plot Scores from PCA, MIA or PARAFAC Analysis of a Spectra or Spectra2D Object</p></a></li>
<li><a href='#plotScree'><p>Scree Plots from PCA or MIA Analysis of a Spectra or Spectra2D Object</p></a></li>
<li><a href='#plotSpectra'><p>Plot Spectra Object</p></a></li>
<li><a href='#plotSpectraDist'><p>Plot the Distance Between Spectra and a Reference Spectrum in a Spectra Object</p></a></li>
<li><a href='#plotSpectraJS'><p>Plot a Spectra Object Interactively</p></a></li>
<li><a href='#r_pcaSpectra'><p>Robust PCA of a Spectra Object</p></a></li>
<li><a href='#removeFreq'><p>Remove Frequencies from a Spectra or Spectra2D Object</p></a></li>
<li><a href='#removeGroup'><p>Remove Groups from a Spectra or Spectra2D Object</p></a></li>
<li><a href='#removeSample'><p>Remove Samples from a Spectra or Spectra2D Object</p></a></li>
<li><a href='#reviewAllSpectra'><p>Review All the Spectra in a Spectra Object</p></a></li>
<li><a href='#rowDist'><p>Compute Distance Between Rows of a Matrix</p></a></li>
<li><a href='#s_pcaSpectra'><p>Sparse PCA of Spectra Objects</p></a></li>
<li><a href='#sampleDist'><p>Compute the Distances Between Samples in a Spectra or Spectra2D Object</p></a></li>
<li><a href='#sgfSpectra'><p>Apply Savitzky-Golay filters to a Spectra object</p></a></li>
<li><a href='#Spectra'><p>Spectra Objects</p></a></li>
<li><a href='#splitSpectraGroups'><p>Create New Groups from an Existing Spectra Object</p></a></li>
<li><a href='#sPlotSpectra'><p>s-Plot of Spectra Data (Post PCA)</p></a></li>
<li><a href='#SrE.IR'><p>IR and NMR Spectra of Serenoa repens (Saw Palmetto) Oil Extracts and</p>
Reference Oils</a></li>
<li><a href='#sumGroups'><p>Summarize the Group Membership of a Spectra or Spectra2D Object</p></a></li>
<li><a href='#sumSpectra'><p>Summarize a Spectra or Spectra2D Object</p></a></li>
<li><a href='#surveySpectra'><p>Plot Measures of Central Tendency and Spread for a Spectra Object</p></a></li>
<li><a href='#updateGroups'><p>Update Group Names in a Spectra or Spectra2D Object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Exploratory Chemometrics for Spectroscopy</td>
</tr>
<tr>
<td>Version:</td>
<td>6.1.10</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-02-03</td>
</tr>
<tr>
<td>Description:</td>
<td>A collection of functions for top-down exploratory data analysis
    of spectral data including nuclear magnetic resonance (NMR), infrared (IR),
    Raman, X-ray fluorescence (XRF) and other similar types of spectroscopy.
    Includes functions for plotting and inspecting spectra, peak alignment,
    hierarchical cluster analysis (HCA), principal components analysis (PCA) and
    model-based clustering. Robust methods appropriate for this type of
    high-dimensional data are available. ChemoSpec is designed for structured
    experiments, such as metabolomics investigations, where the samples fall into
    treatment and control groups. Graphical output is formatted consistently for
    publication quality plots. ChemoSpec is intended to be very user friendly and
    to help you get usable results quickly. A vignette covering typical operations
    is available.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5), ChemoSpecUtils (&ge; 1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, utils, grDevices, reshape2, readJDX (&ge; 0.6),
patchwork, ggplot2, plotly, magrittr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>IDPmisc, knitr, js, NbClust, clusterCrit, lattice, baseline,
mclust, pls, R.utils, RColorBrewer, seriation, MASS, grid,
pcaPP, jsonlite, signal, speaq, tinytest, elasticnet, irlba,
rmarkdown, bookdown, chemometrics, hyperSpec, amap, gsubfn,
roxut</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/bryanhanson/ChemoSpec/issues">https://github.com/bryanhanson/ChemoSpec/issues</a></td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>TRUE</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-03 16:42:07 UTC; bryanh</td>
</tr>
<tr>
<td>Author:</td>
<td>Bryan A. Hanson <a href="https://orcid.org/0000-0003-3536-8246"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Mike Bostock [cph, ctb] (author of the d3.js library used by
    plotSpectraJS, http://d3js.org),
  Matt Keinsley [ctb] (author of initial AOV-PCA code),
  Tejasvi Gupta [ctb] (author of ggplot2 and plotly graphics)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Bryan A. Hanson &lt;hanson@depauw.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-03 17:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ChemoSpec-package'>Exploratory Chemometrics for Spectroscopy</h2><span id='topic+ChemoSpec-package'></span><span id='topic+ChemoSpec'></span>

<h3>Description</h3>

<p>A collection of functions for top-down exploratory data analysis of spectral
data obtained via nuclear magnetic resonance (NMR), infrared (IR) or Raman
spectroscopy.  Includes functions for plotting and inspecting spectra, peak
alignment, hierarchical cluster analysis (HCA), principal components
analysis (PCA) and model-based clustering. Robust methods appropriate for
this type of high-dimensional data are available.  ChemoSpec is designed
with metabolomics data sets in mind, where the samples fall into groups such
as treatment and control.  Graphical output is formatted consistently for
publication quality plots.  ChemoSpec is intended to be very user friendly
and help you get usable results quickly.  A vignette covering typical
operations is available.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University), Tejasvi Gupta &amp; Matthew J. Keinsley.
</p>
<p>Maintainer: Bryan A. Hanson <a href="mailto:hanson@depauw.edu">hanson@depauw.edu</a>
</p>


<h3>See Also</h3>

<p>Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>

<hr>
<h2 id='aov_pcaSpectra'>ANOVA-PCA Analysis of Spectra Data</h2><span id='topic+aov_pcaSpectra'></span>

<h3>Description</h3>

<p>ANOVA-PCA is a combination of both methods developed by Harrington.  The
data is partitioned into submatrices corresponding to each experimental
factor, which are then subjected to PCA separately after adding the residual
error back.  If the effect of a factor is large compared to the residual
error, separation along the 1st PC in the score plot should be evident.
With this method, the significance of a factor can be visually determined
(ANOVA-PCA is not blind to group membership). ANOVA-PCA with only one factor
is the same as standard PCA and gives no additional separation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aov_pcaSpectra(spectra, fac, type = "class", choice = NULL, showNames = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="aov_pcaSpectra_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="aov_pcaSpectra_+3A_fac">fac</code></td>
<td>
<p>A vector of character strings giving the factors to be used in
the analysis.  These should be elements of <code><a href="#topic+Spectra">Spectra</a></code>.  Note that
there should be 2 or more factors, because ANOVA-PCA on one factor is the
same as standard PCA.  See the example.</p>
</td></tr>
<tr><td><code id="aov_pcaSpectra_+3A_type">type</code></td>
<td>
<p>Either classical (&quot;cls&quot;) or robust (&quot;rob&quot;); Results in either
<code><a href="#topic+c_pcaSpectra">c_pcaSpectra</a></code> or <code><a href="#topic+r_pcaSpectra">r_pcaSpectra</a></code> being called on the
<code><a href="#topic+Spectra">Spectra</a></code> object.</p>
</td></tr>
<tr><td><code id="aov_pcaSpectra_+3A_choice">choice</code></td>
<td>
<p>The type of scaling to be performed.  See
<code><a href="#topic+c_pcaSpectra">c_pcaSpectra</a></code> and <code><a href="#topic+r_pcaSpectra">r_pcaSpectra</a></code> for details.</p>
</td></tr>
<tr><td><code id="aov_pcaSpectra_+3A_shownames">showNames</code></td>
<td>
<p>Logical.  Show the names of the submatrices in the console.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of PCA results, one for each computed submatrix.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University), Matthew J. Keinsley.
</p>


<h3>References</h3>

<p>Pinto, Bosc, Nocairi, Barros, and Rutledge. &quot;Using ANOVA-PCA for
Discriminant Analysis: ...&quot; Analytica Chimica Acta 629.1-2 (2008): 47-55.
</p>
<p>Harrington, Vieira, Espinoza, Nien, Romero, and Yergey. &quot;Analysis of
Variance&ndash;Principal Component Analysis: ...&quot; Analytica Chimica Acta 544.1-2
(2005): 118-27.
</p>


<h3>See Also</h3>

<p>The output of this function is used in
used in <code><a href="#topic+aovPCAscores">aovPCAscores</a></code> and <code><a href="#topic+aovPCAloadings">aovPCAloadings</a></code>.
Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
# This example assumes the graphics output is set to ggplot2 (see ?GraphicsOptions).
library("ggplot2")
data(metMUD2)

# Original factor encoding:
levels(metMUD2$groups)

# Split those original levels into 2 new ones (re-code them)
new.grps &lt;- list(geneBb = c("B", "b"), geneCc = c("C", "c"))
mM3 &lt;- splitSpectraGroups(metMUD2, new.grps)

# run aov_pcaSpectra
PCAs &lt;- aov_pcaSpectra(mM3, fac = c("geneBb", "geneCc"))

p1 &lt;- aovPCAscores(mM3, PCAs, submat = 1, ellipse = "cls")
p1 &lt;- p1 + ggtitle("aovPCA: B vs b")
p1

p2 &lt;- aovPCAscores(mM3, PCAs, submat = 2)
p2 &lt;- p2 + ggtitle("aovPCA: C vs c")
p2

p3 &lt;- aovPCAscores(mM3, PCAs, submat = 3)
p3 &lt;- p3 + ggtitle("aovPCA: Interaction Term")
p3

p4 &lt;- aovPCAloadings(spectra = mM3, PCA = PCAs)
p4 &lt;- p4 + ggtitle("aov_pcaSpectra: Bb Loadings")
p4

## End(Not run)
</code></pre>

<hr>
<h2 id='aovPCAloadings'>Plot aovPCAscores Loadings of a Spectra Object</h2><span id='topic+aovPCAloadings'></span>

<h3>Description</h3>

<p>Uses the results from <code><a href="#topic+aovPCAscores">aovPCAscores</a></code> to plot the corresponding
loadings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aovPCAloadings(spectra, PCA, submat = 1, loads = 1, ref = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="aovPCAloadings_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="aovPCAloadings_+3A_pca">PCA</code></td>
<td>
<p>List of pca results created by <code><a href="#topic+aov_pcaSpectra">aov_pcaSpectra</a></code>.</p>
</td></tr>
<tr><td><code id="aovPCAloadings_+3A_submat">submat</code></td>
<td>
<p>Integer.  Selects list element <code>submat</code> from <code>PCA</code>
which is a list of PCA results, each corresponding to the computation
in <code><a href="#topic+aov_pcaSpectra">aov_pcaSpectra</a></code>.</p>
</td></tr>
<tr><td><code id="aovPCAloadings_+3A_loads">loads</code></td>
<td>
<p>An integer vector giving the loadings to plot.</p>
</td></tr>
<tr><td><code id="aovPCAloadings_+3A_ref">ref</code></td>
<td>
<p>An integer specifying the reference spectrum to plot, which
appears at the bottom of the plot.</p>
</td></tr>
<tr><td><code id="aovPCAloadings_+3A_...">...</code></td>
<td>
<p>Parameters to be passed to the plotting routines. <em>Applies to base graphics only</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The returned value depends on the graphics option selected (see <code>GraphicsOptions()</code>).
</p>

<ul>
<li> <p><code>base</code>: None.  Side effect is a plot.
</p>
</li>
<li> <p><code>ggplot2</code>: The plot is displayed, and a <code>ggplot2</code> object is returned if the value is assigned.  The plot can be modified in the usual <code>ggplot2</code> manner.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University), Matthew J. Keinsley.
</p>


<h3>References</h3>

<p>Pinto, Bosc, Nocairi, Barros, and Rutledge. &quot;Using ANOVA-PCA for
Discriminant Analysis: ...&quot; Analytica Chimica Acta 629.1-2 (2008): 47-55.
</p>
<p>Harrington, Vieira, Espinoza, Nien, Romero, and Yergey. &quot;Analysis of
Variance&ndash;Principal Component Analysis: ...&quot; Analytica Chimica Acta 544.1-2
(2005): 118-27.
</p>


<h3>See Also</h3>

<p>An example using this function can be seen in
<code><a href="#topic+aov_pcaSpectra">aov_pcaSpectra</a></code>.  See also <code><a href="#topic+plotLoadings">plotLoadings</a></code>.
Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>

<hr>
<h2 id='aovPCAscores'>Plot ANOVA-PCA Scores from a Spectra Object</h2><span id='topic+aovPCAscores'></span>

<h3>Description</h3>

<p>Uses the results from <code><a href="#topic+aov_pcaSpectra">aov_pcaSpectra</a></code> to plot the scores.
Argument <code>submat</code> is used to select PCA results from among those
stored in argument <code>PCA</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aovPCAscores(
  spectra,
  so,
  submat = 1,
  ellipse = "none",
  tol = "none",
  use.sym = FALSE,
  leg.loc = "topright",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="aovPCAscores_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="aovPCAscores_+3A_so">so</code></td>
<td>
<p><em>List</em> of pca results created by <code><a href="#topic+aov_pcaSpectra">aov_pcaSpectra</a></code>.</p>
</td></tr>
<tr><td><code id="aovPCAscores_+3A_submat">submat</code></td>
<td>
<p>Integer.  Selects list element <code>submat</code> from <code>PCA</code>
which is a list of PCA results, each corresponding to the computation
in <code><a href="#topic+aov_pcaSpectra">aov_pcaSpectra</a></code>.</p>
</td></tr>
<tr><td><code id="aovPCAscores_+3A_ellipse">ellipse</code></td>
<td>
<p>A character vector specifying the type of ellipses to be
plotted.  One of <code>c("both"</code>, <code>"none"</code>, <code>"cls"</code>, <code>"rob")</code>.  <code>cls</code>
specifies classical confidence ellipses, <code>rob</code> specifies robust
confidence ellipses.  An ellipse is drawn for each group unless there
are three or fewer samples in the group.</p>
</td></tr>
<tr><td><code id="aovPCAscores_+3A_tol">tol</code></td>
<td>
<p>A number describing the fraction of points to be labeled. <code>tol = 1.0</code> labels all the points; <code>tol = 0.05</code> labels <em>approximately</em> the most extreme 5 percent. Set to <code>'none'</code> to completely suppress labels. Note that a simple approach based upon quantiles is used, assumes that both x and y are each normally distributed, and treats x and y separately.  Thus, this is not a formal treatment of outliers, just a means of labeling points. Groups are lumped together for the computation.</p>
</td></tr>
<tr><td><code id="aovPCAscores_+3A_use.sym">use.sym</code></td>
<td>
<p>A logical; if TRUE, the color scheme is set to black and the
points plotted with symbols.  Applies only to <code><a href="#topic+Spectra">Spectra</a></code> objects.</p>
</td></tr>
<tr><td><code id="aovPCAscores_+3A_leg.loc">leg.loc</code></td>
<td>
<p>Character; if <code>"none"</code> no legend will be drawn.
Otherwise, any string acceptable to <code><a href="graphics.html#topic+legend">legend</a></code>.</p>
</td></tr>
<tr><td><code id="aovPCAscores_+3A_...">...</code></td>
<td>
<p>Additional parameters to be passed to <code><a href="#topic+plotScores">plotScores</a></code>.
For example, you can plot confidence ellipses this way.  Note that ellipses
are drawn based on the groups in <code>spectra$groups</code>, but the separation
done by <code>aov_pcaSpectra</code> is based on argument <code>fac</code>.  These may
not correspond, but you can edit <code>spectra$groups</code> to match if necessary.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The returned value depends on the graphics option selected (see <code><a href="ChemoSpecUtils.html#topic+GraphicsOptions">ChemoSpecUtils::GraphicsOptions()</a></code>).
</p>

<ul>
<li> <p><strong>base</strong> A data frame or list containing the data plotted.  Assign the value and run <code>str()</code> or <code>names()</code> on it to see what it contains. Side effect is a plot.
</p>
</li>
<li> <p><strong>ggplot2</strong> The plot is displayed, and a <code>ggplot2</code> plot object is returned if the value is assigned. The plot can be modified in the usual <code>ggplot2</code> manner. If you want the plotted values, you can access them via the base graphics mode.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University), Matthew J. Keinsley.
</p>


<h3>References</h3>

<p>Pinto, Bosc, Nocairi, Barros, and Rutledge. &quot;Using ANOVA-PCA for
Discriminant Analysis: ...&quot; Analytica Chimica Acta 629.1-2 (2008): 47-55.
</p>
<p>Harrington, Vieira, Espinoza, Nien, Romero, and Yergey. &quot;Analysis of
Variance&ndash;Principal Component Analysis: ...&quot; Analytica Chimica Acta 544.1-2
(2005): 118-27.
</p>


<h3>See Also</h3>

<p>The use of this function can be seen in
<code><a href="#topic+aov_pcaSpectra">aov_pcaSpectra</a></code>.  See also <code><a href="#topic+plotScores">plotScores</a></code>.
Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>

<hr>
<h2 id='averageReplicates'>Average Replicates in a Spectra Object</h2><span id='topic+averageReplicates'></span>

<h3>Description</h3>

<p>Average the replicates in a <code><a href="#topic+Spectra">Spectra</a></code> object and return a new Spectra object
with fewer samples.  One should probably not do this until each individual sample has been
visualized for quality control, in case it is a potential outlier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>averageReplicates(spectra, uniq)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="averageReplicates_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="averageReplicates_+3A_uniq">uniq</code></td>
<td>
<p>Character.  A character vector containing strings representing unique sample
identifiers.  The sample names will be searched for these strings, and all samples
matching a given string will be averaged and put into a new Spectra object.  For
example, consider the case where samples are named
<code>S_1_01, S_1_02, ..., S_2_01, S_2_02, ...</code>
where <code>_01</code> and so forth signifies replicates of a particular sample.
With <code>uniq = c("S_1", "S_2")</code> all <code>S_1</code> replicates will be averaged and
all <code>S_2</code> replicates will be averaged.  N.B. the strings will be used as
regex pattern and grepped.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of S3 class <code><a href="#topic+Spectra">Spectra</a></code>.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SrE.IR)
averaged &lt;- averageReplicates(SrE.IR, uniq = c("EPO", "OO", "adSrE", "pSrE"))
sumSpectra(SrE.IR)
sumSpectra(averaged)

</code></pre>

<hr>
<h2 id='baselineSpectra'>Baseline Correction of a Spectra Object</h2><span id='topic+baselineSpectra'></span>

<h3>Description</h3>

<p>This function mostly wraps functions in package <span class="pkg">baseline</span> which
carries out a variety of baseline correction routines.  A simple linear
correction method is also available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>baselineSpectra(spectra, int = TRUE, retC = FALSE, show = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="baselineSpectra_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="baselineSpectra_+3A_int">int</code></td>
<td>
<p>Logical; if <code>TRUE</code>, do the correction interactively using
widgets.  No results are saved.  Use this for inspection and exploration
only.  Automatically overriden to <code>FALSE</code> if <code>interactive()</code> returns
<code>FALSE</code>.  This is necessary so that plots appear in vignettes etc.</p>
</td></tr>
<tr><td><code id="baselineSpectra_+3A_retc">retC</code></td>
<td>
<p>Logical: shall the baseline-corrected spectra be returned in the
<code>Spectra</code> object?</p>
</td></tr>
<tr><td><code id="baselineSpectra_+3A_show">show</code></td>
<td>
<p>Integer. A (single) sample number for which you wish
to see the results of the baseline correction.  By &quot;sample number&quot;
we mean the rows in the <code>spectra$data</code> matrix.  To find a specific
sample type <code>spectra$names</code> to see which row contains that sample.</p>
</td></tr>
<tr><td><code id="baselineSpectra_+3A_...">...</code></td>
<td>
<p>Other arguments passed downstream.  The relevant ones can be
found in <code><a href="baseline.html#topic+baseline">baseline</a></code>.  Be sure to pay attention to
argument <code>method</code> as you will probably want to use it.  You can also
use <code>method = "linear"</code> for a simple linear fit, see Details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In plots using methods from the baseline package, the x axis ticks give the
data point index, not the original values from your data. Note that you
cannot zoom the non-interactive display of corrected spectra because the
underlying function hardwires the display.  Try the interactive version
instead (<code>int = TRUE</code>), or use <code><a href="#topic+plotSpectra">plotSpectra</a></code> on the
corrected data.
In addition to the methods provided by <code>baseline</code>, you can also use
<code>method = "linear"</code>.  This correction is handled locally, and is very
simple: a line is drawn from the first data point to the last, and this
becomes the new baseline.  This is most suitable for cases in which the
baseline rises or falls steadily, as is often seen in chromatograms.
</p>


<h3>Value</h3>

<p>If <code>int = TRUE</code>, an interactive plot is created.  If <code>int
= FALSE</code> and <code>retC = FALSE</code>, an object of class <code>baseline</code> is
returned (see <code><a href="baseline.html#topic+baseline-class">baseline-class</a></code>).  If <code>int =
FALSE</code> and <code>retC = TRUE</code>, a <code>Spectra</code> object containing the
corrected spectra is returned.  In these latter two cases plots are also
drawn.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>See Also</h3>

<p>Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># You need to install package "baseline" for this example
if (requireNamespace("baseline", quietly = TRUE)) {
  data(SrE.IR)
  temp &lt;- baselineSpectra(SrE.IR, int = FALSE, method = "modpolyfit")
}

</code></pre>

<hr>
<h2 id='binSpectra'>Bin or Bucket a Spectra Object</h2><span id='topic+binSpectra'></span>

<h3>Description</h3>

<p>This function will bin a <code><a href="#topic+Spectra">Spectra</a></code> object by averaging every
<code>bin.ratio</code> frequency values, and summing the corresponding intensity
values.  The net effect is a smoothed and smaller data set.  If there are
gaps in the frequency axis, each data chunk is processed separately.  Note:
some folks refer to binning as bucketing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binSpectra(spectra, bin.ratio)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="binSpectra_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="binSpectra_+3A_bin.ratio">bin.ratio</code></td>
<td>
<p>An integer giving the binning ratio, that is, the number of
points to be grouped together into one subset of data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the frequency range is not divisible by bin.ratio to give a whole number,
data points are removed from the beginning of the frequency data until it
is, and the number of data points removed is reported at the console.  If
there are gaps in the data where frequencies have been removed, each
continuous piece is sent out and binned separately (by
<code><a href="#topic+binSpectra">binSpectra</a></code>).
</p>


<h3>Value</h3>

<p>An object of S3 class <code><a href="#topic+Spectra">Spectra</a></code>.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>See Also</h3>

<p>Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(metMUD1)
sumSpectra(metMUD1)
res &lt;- binSpectra(metMUD1, bin.ratio = 4)
sumSpectra(res)

</code></pre>

<hr>
<h2 id='c_pcaSpectra'>Classical PCA of Spectra Objects</h2><span id='topic+c_pcaSpectra'></span>

<h3>Description</h3>

<p>A wrapper which carries out classical PCA analysis on a
<code><a href="#topic+Spectra">Spectra</a></code> object.  The user can select various options for
scaling.  There is no normalization by rows - do this manually using
<code><a href="#topic+normSpectra">normSpectra</a></code>.  There is an option to control centering, but
this is mainly for compatibility with the <code><a href="#topic+aov_pcaSpectra">aov_pcaSpectra</a></code>
series of functions.  Centering the data should always be done in PCA and it
is the default here.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c_pcaSpectra(spectra, choice = "noscale", cent = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="c_pcaSpectra_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="c_pcaSpectra_+3A_choice">choice</code></td>
<td>
<p>A character string indicating the choice of scaling.  One of
<code>c("noscale"</code>, <code>"autoscale"</code>, <code>"Pareto")</code>. <code>"autoscale"</code>
is called &quot;standard normal variate&quot; or &quot;correlation matrix PCA&quot; in some literature.</p>
</td></tr>
<tr><td><code id="c_pcaSpectra_+3A_cent">cent</code></td>
<td>
<p>Logical: whether or not to center the data.  Always center the
data unless you know it to be already centered.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The scale choice <code>autoscale</code> scales the columns by their standard
deviation.  <code>Pareto</code> scales by the square root of the standard
deviation.
</p>


<h3>Value</h3>

<p>An object of class <code><a href="stats.html#topic+prcomp">prcomp</a></code>, modified to include a list
element called <code>$method</code>, a character string describing the
pre-processing carried out and the type of PCA performed (used to annotate
plots).
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>References</h3>

<p>K. Varmuza and P. Filzmoser <em>Introduction to Multivariate
Statistical Analysis in Chemometrics</em>, CRC Press, 2009.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+prcomp">prcomp</a></code> for the underlying function,
<code><a href="#topic+s_pcaSpectra">s_pcaSpectra</a></code> for sparse PCA calculations,
<code><a href="#topic+r_pcaSpectra">r_pcaSpectra</a></code> for robust PCA calculations,
<code><a href="#topic+irlba_pcaSpectra">irlba_pcaSpectra</a></code> for PCA via the IRLBA algorithm.
Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>
<p>For displaying the results, <code><a href="ChemoSpecUtils.html#topic+plotScree">ChemoSpecUtils::plotScree()</a></code>, <code><a href="ChemoSpecUtils.html#topic+plotScores">ChemoSpecUtils::plotScores()</a></code>, <code><a href="#topic+plotLoadings">plotLoadings()</a></code>, <code><a href="#topic+plot2Loadings">plot2Loadings()</a></code>, <code><a href="#topic+sPlotSpectra">sPlotSpectra()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# This example assumes the graphics output is set to ggplot2 (see ?GraphicsOptions).
library("ggplot2")
data(metMUD1)
pca &lt;- c_pcaSpectra(metMUD1)

p1 &lt;- plotScree(pca)
p1

p2 &lt;- plotScores(metMUD1, pca, pcs = c(1, 2), ellipse = "cls", tol = 0.05)
p2 &lt;- p2 + ggtitle("Scores: metMUD1 NMR Data")
p2

p3 &lt;- plotLoadings(metMUD1, pca, loads = 1:2, ref = 1)
p3 &lt;- p3 + ggtitle("Loadings: metMUD1 NMR Data")
p3

## End(Not run)

</code></pre>

<hr>
<h2 id='check4Gaps'>Check for Discontinuities (Gaps) in a Vector &amp; Optionally Make a Plot</h2><span id='topic+check4Gaps'></span>

<h3>Description</h3>

<p>This function is used by <code>ChemoSpec</code> and <code>ChemoSpec2D</code>, but is
formally part of <code>ChemoSpecUtils</code>.  You can access
full documentation via <code><a href="ChemoSpecUtils.html#topic+check4Gaps">check4Gaps</a></code>.
</p>

<hr>
<h2 id='chkGraphicsOpt'>Check the Graphics Output Option/Mode</h2><span id='topic+chkGraphicsOpt'></span>

<h3>Description</h3>

<p>This function is used by <code>ChemoSpec</code> and <code>ChemoSpec2D</code>, but is
formally part of <code>ChemoSpecUtils</code>.  You can access
full documentation via <code><a href="ChemoSpecUtils.html#topic+chkGraphicsOpt">chkGraphicsOpt</a></code>.
</p>

<hr>
<h2 id='chkSpectra'>Verify the Integrity of a Spectra or Spectra2D Object</h2><span id='topic+chkSpectra'></span>

<h3>Description</h3>

<p>This function is used by <code>ChemoSpec</code> and <code>ChemoSpec2D</code>, but is
formally part of <code>ChemoSpecUtils</code>.  You can access
full documentation via <code><a href="ChemoSpecUtils.html#topic+chkSpectra">chkSpectra</a></code>.
</p>

<hr>
<h2 id='clupaSpectra'>Hierarchical Cluster-Based Peak Alignment on a Spectra Object</h2><span id='topic+clupaSpectra'></span>

<h3>Description</h3>

<p>This function is a wrapper to several functions in the <span class="pkg">speaq</span> package.
It implements the CluPA algorithm described in the reference.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clupaSpectra(spectra, bT = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clupaSpectra_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="clupaSpectra_+3A_bt">bT</code></td>
<td>
<p>Numeric.  The baseline threshold. Defaults to five percent of the
range of the data, in <code>spectra$data</code>.  Passed to
<code>detectSpecPeaks</code>.</p>
</td></tr>
<tr><td><code id="clupaSpectra_+3A_...">...</code></td>
<td>
<p>Other arguments to be passed to the underlying functions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A modifed <code><a href="#topic+Spectra">Spectra</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>References</h3>

<p>Vu TN, Valkenborg D, Smets K, Verwaest KA, Dommisse R, Lemiere F, Verschoren
A, Goethals B, Laukens K. &quot;An integrated workflow for robust alignment and
simplified quantitative analysis of NMR spectrometry data&quot; BMC
Bioinformatics vol. 12 pg. 405 (2011).
</p>


<h3>See Also</h3>

<p>Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># You need to install package speaq for this example
# This example assumes the graphics output is set to ggplot2 (see ?GraphicsOptions).
if (requireNamespace("speaq", quietly = TRUE)) {
  library("ggplot2")
  data(alignMUD)

  p1 &lt;- plotSpectra(alignMUD, which = 1:20, lab.pos = 4.5, offset = 0.1,
    yrange = c(0, 5000), amp = 500)
  p1 &lt;- p1 + ggtitle("Misaligned NMR Spectra") +
    coord_cartesian(xlim = c(1.5, 1.8), ylim = c(0, 1900))
  p1

  aMUD &lt;- clupaSpectra(alignMUD)

  p2 &lt;- plotSpectra(aMUD, which = 1:20, lab.pos = 4.5, offset = 0.1,
    yrange = c(0, 5000), amp = 500)
  p2 &lt;- p2 +  ggtitle("Aligned NMR Spectra") +
    coord_cartesian(xlim = c(1.5, 1.8), ylim = c(0, 1900))
  p2
}

</code></pre>

<hr>
<h2 id='colorSymbol'>Color and Symbols in ChemoSpec and ChemoSpec2D</h2><span id='topic+colorSymbol'></span>

<h3>Description</h3>

<p>This help page serves both <code>ChemoSpec</code> and <code>ChemoSpec2D</code>, but is
formally part of <code>ChemoSpecUtils</code>. You can access
full documentation via <code><a href="ChemoSpecUtils.html#topic+colorSymbol">colorSymbol</a></code>.
</p>

<hr>
<h2 id='conColScheme'>Change the Color Scheme of a Spectra or Spectra2D Object</h2><span id='topic+conColScheme'></span>

<h3>Description</h3>

<p>This help page serves both <code>ChemoSpec</code> and <code>ChemoSpec2D</code>, but is
formally part of <code>ChemoSpecUtils</code>. You can access
full documentation via <code><a href="ChemoSpecUtils.html#topic+conColScheme">conColScheme</a></code>.
</p>

<hr>
<h2 id='cv_pcaSpectra'>Cross-Validation of Classical PCA Results for a Spectra Object</h2><span id='topic+cv_pcaSpectra'></span>

<h3>Description</h3>

<p>This function carries out classical PCA on the data in a
<code><a href="#topic+Spectra">Spectra</a></code> object using a cross-validation method.  A simple
re-write of Peter Filzmoser's <code><a href="chemometrics.html#topic+pcaCV">pcaCV</a></code> method
with some small plotting changes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cv_pcaSpectra(
  spectra,
  pcs,
  choice = "noscale",
  repl = 50,
  segments = 4,
  segment.type = c("random", "consecutive", "interleaved"),
  length.seg,
  trace = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cv_pcaSpectra_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="cv_pcaSpectra_+3A_pcs">pcs</code></td>
<td>
<p>As per <code><a href="chemometrics.html#topic+pcaCV">pcaCV</a></code> where it is called amax;
an integer giving the number of PC scores to include.</p>
</td></tr>
<tr><td><code id="cv_pcaSpectra_+3A_choice">choice</code></td>
<td>
<p>A character string indicating the choice of scaling.  One of
<code>c("noscale"</code>, <code>"autoscale"</code>, <code>"Pareto")</code>.</p>
</td></tr>
<tr><td><code id="cv_pcaSpectra_+3A_repl">repl</code></td>
<td>
<p>As per <code><a href="chemometrics.html#topic+pcaCV">pcaCV</a></code>; the number of
replicates to perform.</p>
</td></tr>
<tr><td><code id="cv_pcaSpectra_+3A_segments">segments</code></td>
<td>
<p>As per <code><a href="chemometrics.html#topic+pcaCV">pcaCV</a></code>.</p>
</td></tr>
<tr><td><code id="cv_pcaSpectra_+3A_segment.type">segment.type</code></td>
<td>
<p>As per <code><a href="chemometrics.html#topic+pcaCV">pcaCV</a></code>.</p>
</td></tr>
<tr><td><code id="cv_pcaSpectra_+3A_length.seg">length.seg</code></td>
<td>
<p>As per <code><a href="chemometrics.html#topic+pcaCV">pcaCV</a></code>.</p>
</td></tr>
<tr><td><code id="cv_pcaSpectra_+3A_trace">trace</code></td>
<td>
<p>As per <code><a href="chemometrics.html#topic+pcaCV">pcaCV</a></code>.</p>
</td></tr>
<tr><td><code id="cv_pcaSpectra_+3A_...">...</code></td>
<td>
<p>Parameters to be passed to the plotting routines.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly, a list as described in <code><a href="chemometrics.html#topic+pcaCV">pcaCV</a></code>.
Side effect is a plot.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson, DePauw University. Derived from <code><a href="chemometrics.html#topic+pcaCV">pcaCV</a></code>.
</p>


<h3>References</h3>

<p>K. Varmuza and P. Filzmoser <em>Introduction to Multivariate
Statistical Analysis in Chemometrics</em>, CRC Press, 2009.
</p>


<h3>See Also</h3>

<p><code><a href="chemometrics.html#topic+pcaCV">pcaCV</a></code> for the underlying function.
Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># You need to install package "pls" for this example
if (requireNamespace("pls", quietly = TRUE)) {
  data(SrE.IR)
  pca &lt;- cv_pcaSpectra(SrE.IR, pcs = 5)
}

</code></pre>

<hr>
<h2 id='evalClusters'>Evaluate or Compare the Quality of Clusters Quantitatively</h2><span id='topic+evalClusters'></span>

<h3>Description</h3>

<p>This function is a wrapper to two functions: <code>intCriteria</code> function in
package <span class="pkg">clusterCrit</span>, and <code>NbClust</code> in package <span class="pkg">NbClust</span>. It
can be used to quantitatively compare different clustering options.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evalClusters(
  spectra,
  pkg = "NbClust",
  hclst = NULL,
  k = NULL,
  h = NULL,
  crit = "Dunn",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="evalClusters_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra</a></code>.</p>
</td></tr>
<tr><td><code id="evalClusters_+3A_pkg">pkg</code></td>
<td>
<p>Character.  One of <code>c("NbClust", "clusterCrit")</code>.  The
package to use for comparing clusters.</p>
</td></tr>
<tr><td><code id="evalClusters_+3A_hclst">hclst</code></td>
<td>
<p>An object of S3 class <code>hclust</code>.  Only applies to
<code>pkg = "clusterCrit"</code>.</p>
</td></tr>
<tr><td><code id="evalClusters_+3A_k">k</code></td>
<td>
<p>Integer.  The number of groups in which to cut the tree
(<code>hclust</code>).  Only applies to <code>pkg = "clusterCrit"</code>.</p>
</td></tr>
<tr><td><code id="evalClusters_+3A_h">h</code></td>
<td>
<p>Numeric.  The height at which to cut the tree.  Either <code>k</code> or
<code>h</code> must be given, with <code>k</code> taking precedence.  See
<code><a href="stats.html#topic+cutree">cutree</a></code>.  Only applies to <code>pkg = "clusterCrit"</code>.</p>
</td></tr>
<tr><td><code id="evalClusters_+3A_crit">crit</code></td>
<td>
<p>String.  A string giving the criteria to be used in evaluating
the quality of the cluster.  See <code>liintCriteria</code>.  Only applies to
<code>pkg = "clusterCrit"</code>.</p>
</td></tr>
<tr><td><code id="evalClusters_+3A_...">...</code></td>
<td>
<p>Other parameters to be passed to the functions. In particular,
the default <code>NbClust</code> package will need some parameters.  See the
example.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Both of the packages used here compute very similar quantities.  For
details, see the publication and respective vignettes.  Package
<span class="pkg">clusterCrit</span> takes the approach in which you cluster in a separate
step using whatever parameters you like, then the tree is cut either at a
given height or in such a way as to produce a fixed number of groups.  One
or more indices are then computed.  Then, you repeat this process with
different clustering criteria, and compare.  Package <span class="pkg">NbClust</span> allows
one to specify a range of possible number of clusters and a few other
parameters and will return indices corresponding to each set options, which
is somewhat more automated.
</p>


<h3>Value</h3>

<p>A list giving the results, as described in <code><a href="clusterCrit.html#topic+intCriteria">intCriteria</a></code> or
<code><a href="NbClust.html#topic+NbClust">NbClust</a></code>.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>References</h3>

<p>M. Charrad et. al.  &quot;NbClust: An R Package for Determining the
Relevant Number of Clusters in a Data Set.&quot;  J. Stat. Soft. vol. 61 no. 6
October 2014.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+hclust">hclust</a></code> for the underlying base function.
<code><a href="#topic+hcaSpectra">hcaSpectra</a></code> for HCA analysis of a <code><a href="#topic+Spectra">Spectra</a></code> object.
<code><a href="#topic+hcaScores">hcaScores</a></code> for HCA analysis of PCA scores from a
<code><a href="#topic+Spectra">Spectra</a></code> object. Additional documentation at
<a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
data(metMUD2)

# Using clusterCrit
res1 &lt;- hcaSpectra(metMUD2) # default clustering and distance methods
res2 &lt;- hcaSpectra(metMUD2, d.method = "cosine")
# The return value from hcaSpectra is a list with hclust as the first element.
crit1 &lt;- evalClusters(metMUD2, pkg = "clusterCrit", hclst = res1[[1]], k = 2)
crit2 &lt;- evalClusters(metMUD2, pkg = "clusterCrit", hclst = res2[[1]], k = 2)
# crit1 and crit2 can now be compared.

# Using NbClust
res3 &lt;- evalClusters(metMUD2, min.nc = 2, max.nc = 5, method = "average", index = "kl")

## End(Not run)

</code></pre>

<hr>
<h2 id='files2SpectraObject'>Import Data into a Spectra Object</h2><span id='topic+files2SpectraObject'></span><span id='topic+matrix2SpectraObject'></span>

<h3>Description</h3>

<p>These functions import data into a <code><a href="#topic+Spectra">Spectra</a></code> object.  For &quot;csv-like&quot; files they use
<code><a href="utils.html#topic+read.table">read.table</a></code>, so they are very flexible in regard to file formatting.
<span class="pkg">Be sure to see the ... argument below for important details you need to provide.</span>
<code>files2SpectraObject</code> can also read JCAMP-DX files and will do so if <code>fileExt</code> is
any of <code>"dx"</code>, <code>"DX"</code>, <code>"jdx"</code> or <code>"JDX"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>files2SpectraObject(
  gr.crit = NULL,
  gr.cols = "auto",
  freq.unit = "no frequency unit provided",
  int.unit = "no intensity unit provided",
  descrip = "no description provided",
  fileExt = "\\.(csv|CSV)$",
  out.file = "mydata",
  debug = FALSE,
  ...
)

matrix2SpectraObject(
  gr.crit = NULL,
  gr.cols = c("auto"),
  freq.unit = "no frequency unit provided",
  int.unit = "no intensity unit provided",
  descrip = "no description provided",
  in.file = NULL,
  out.file = "mydata",
  chk = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="files2SpectraObject_+3A_gr.crit">gr.crit</code></td>
<td>
<p>Group Criteria.  A vector of character strings which will be
searched for among the file/sample names in order to assign an individual
spectrum to group membership. This is done using grep, so characters
like &quot;.&quot; (period/dot) do not have their literal meaning (see below).
Warnings are issued if there are file/sample names that don't match entries in
<code>gr.crit</code> or there are entries in <code>gr.crit</code> that don't match any file names.</p>
</td></tr>
<tr><td><code id="files2SpectraObject_+3A_gr.cols">gr.cols</code></td>
<td>
<p>Group Colors.  See <code><a href="#topic+colorSymbol">colorSymbol</a></code> for some options. One of the following:
</p>

<ul>
<li><p> Legacy behavior and the default: The word <code>"auto"</code>, in which case up to 8 colors will
be automatically assigned from package <code>RColorBrewer Set1</code>.
</p>
</li>
<li> <p><code>"Col7"</code>. A unique set of up to 7 colorblind-friendly colors is used.
</p>
</li>
<li> <p><code>"Col8"</code>. A unique set of up to 8 colors is used.
</p>
</li>
<li> <p><code>"Col12"</code>. A mostly paired set of up to 12 colors is used.
</p>
</li>
<li><p> A vector of acceptable color designations with the same length as <code>gr.crit</code>.
</p>
</li></ul>

<p>Colors will be assigned one for one, so the first element of
<code>gr.crit</code> is assigned the first element of <code>gr.col</code> and so forth.  For <code>Col12</code>
you should pay careful attention to the order of <code>gr.crit</code> in order to match up colors.</p>
</td></tr>
<tr><td><code id="files2SpectraObject_+3A_freq.unit">freq.unit</code></td>
<td>
<p>A character string giving the units of the x-axis
(frequency or wavelength).</p>
</td></tr>
<tr><td><code id="files2SpectraObject_+3A_int.unit">int.unit</code></td>
<td>
<p>A character string giving the units of the y-axis (some sort
of intensity).</p>
</td></tr>
<tr><td><code id="files2SpectraObject_+3A_descrip">descrip</code></td>
<td>
<p>A character string describing the data set that will be stored.  This string is used
in some plots so it is recommended that its length be less than about 40 characters.</p>
</td></tr>
<tr><td><code id="files2SpectraObject_+3A_fileext">fileExt</code></td>
<td>
<p>A character string giving the extension of the files to be
processed. <code>regex</code> strings can be used.  For instance, the default
finds files with either <code>".csv"</code> or <code>".CSV"</code> as the extension.
Matching is done via a grep process, which is greedy.  See also the
&quot;Advanced Tricks&quot; section.</p>
</td></tr>
<tr><td><code id="files2SpectraObject_+3A_out.file">out.file</code></td>
<td>
<p>A file name.  The completed object of S3 class <code><a href="#topic+Spectra">Spectra</a></code> will be written
to this file.</p>
</td></tr>
<tr><td><code id="files2SpectraObject_+3A_debug">debug</code></td>
<td>
<p>Logical. Applies to <code>files2SpectraObject</code> only.
Set to <code>TRUE</code> for troubleshooting when an error
is thrown during import.  In addition, values of 1-5 will work
when importing a JCAMP-DX file via <code>fileExt = "\.jdx"</code> etc.  These
will be passed through to the <code><a href="readJDX.html#topic+readJDX">readJDX</a></code> function.
See there for much more info on importing JCAMP-DX files.</p>
</td></tr>
<tr><td><code id="files2SpectraObject_+3A_...">...</code></td>
<td>
<p>Arguments to be passed to <code><a href="utils.html#topic+read.table">read.table</a></code>,
<code><a href="base.html#topic+list.files">list.files</a></code> or <code>readJDX</code>; see the &quot;Advanced Tricks&quot; section.
For <code>read.table</code>, <span class="pkg">You MUST supply values for <code>sep</code>, <code>dec</code> and <code>header</code> consistent
with your file structure, unless they are the same as the defaults for <code><a href="utils.html#topic+read.table">read.table</a></code></span>.</p>
</td></tr>
<tr><td><code id="files2SpectraObject_+3A_in.file">in.file</code></td>
<td>
<p>Character.  Applies to <code>matrix2SpectraObject</code> only.
Input file name, including extension.  Can be a vector of file names.</p>
</td></tr>
<tr><td><code id="files2SpectraObject_+3A_chk">chk</code></td>
<td>
<p>Logical. Applies to <code>matrix2SpectraObject</code> only.
Should the <code>Spectra</code> object be checked for integrity?  If you are having trouble
importing your data, set this to <code>FALSE</code> and do <code>str(your object)</code> to troubleshoot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A object of class <code><a href="#topic+Spectra">Spectra</a></code>.  An <em>unnamed</em> object
of S3 class <code><a href="#topic+Spectra">Spectra</a></code> is also written to <code>out.file</code>.  To
read it back into the workspace, use <code>new.name &lt;- loadObject(out.file)</code>
(<code>loadObject</code> is package <span class="pkg">R.utils</span>).
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>files2SpectraObject()</code>: Import data from separate csv files
</p>
</li>
<li> <p><code>matrix2SpectraObject()</code>: Import a matrix of data
</p>
</li></ul>


<h3>files2SpectraObject</h3>

<p><code>files2SpectraObject</code> acts on all files in the current working
directory with the specified <code>fileExt</code> so there should be no
extra files of that type hanging around (except see next paragraph).
The first column should contain the frequency values and the second column the intensity values. The
files may have a header or not (supply <code>header = TRUE/FALSE</code> as
necessary).  The frequency column is assumed to be the same in all files.
</p>
<p>If <code>fileExt</code> contains any of <code>"dx"</code>, <code>"DX"</code>, <code>"jdx"</code> or
<code>"JDX"</code>, then the files will be processed by <code><a href="readJDX.html#topic+readJDX">readJDX</a></code>.
Consider setting <code>debug = TRUE</code>, or <code>debug = 1</code> etc for this format, as there are many
options for JCAMP, and many are untested. See <code><a href="readJDX.html#topic+readJDX">readJDX</a></code> for options and
known limitations.
</p>


<h3>matrix2SpectraObject</h3>

<p>This function takes one or more csv-like files, containing frequencies in the first
column, and samples in additional columns, and processes it into a
<code><a href="#topic+Spectra">Spectra</a></code> object.  The file MUST have a header row which includes
the sample names.  There need not be a header for the first (frequency)
column.  If more than one file given, they must all have the same frequency entries.
</p>


<h3>gr.crit and Sample Name Gotchas</h3>

<p>The matching of <code>gr.crit</code> against the sample file names
(in <code>files2SpectraObject</code>) or column headers/sample names
(in <code>matrix2SpectraObject</code>) is done one at
a time, in order, using grep.  While powerful, this has the potential to lead
to some &quot;gotchas&quot; in certain cases, noted below.
</p>
<p>Your file system may allow file/sample names which <code>R</code> will not like, and will
cause confusing behavior.  File/sample names become variables in <code>ChemoSpec</code>, and <code>R</code>
does not like things like &quot;-&quot; (minus sign or hyphen) in file/sample names.  A hyphen
is converted to a period (&quot;.&quot;) if found, which is fine for a variable name.
However, a period in <code>gr.crit</code> is interpreted from the grep point of view,
namely a period matches any single character.  At this point, things may behave
very differently than one might hope.  See <code><a href="base.html#topic+make.names">make.names</a></code> for allowed
characters in <code>R</code> variables and make sure your file/sample names comply.
</p>
<p>The entries in <code>gr.crit</code> must be
mutually exclusive.  For example, if you have files with names like
&quot;Control_1&quot; and &quot;Sample_1&quot; and use <code>gr.crit = c("Control", "Sample")</code>
groups will be assigned as you would expect.  But, if you have file names
like &quot;Control_1_Shade&quot; and &quot;Sample_1_Sun&quot; you can't use <code>gr.crit =
c("Control",</code> <code>"Sample"</code>, <code>"Sun"</code>, <code>"Shade")</code> because each criteria is grepped in
order, and the &quot;Sun/Shade&quot; phrases, being last, will form the basis for your
groups.  Because this is a grep process, you can get around this by using
regular expressions in your <code>gr.crit</code> argument to specify the desired
groups in a mutually exclusive manner.  In this second example, you could
use <code>gr.crit = c("Control(.*)Sun"</code>, <code>"Control(.*)Shade"</code>, <code>"Sample(.*)Sun"</code>,
<code>"Sample(.*)Shade")</code> to have your groups assigned based upon both phrases in
the file names.
</p>
<p>To summarize, <code>gr.crit</code> is used as a grep pattern, and the file/sample names
are the target.  Make sure your file/sample names comply with <code><a href="base.html#topic+make.names">make.names</a></code>.
</p>
<p>Finally, samples whose names are not matched using <code>gr.crit</code> are still
incorporated into the <code><a href="#topic+Spectra">Spectra</a></code> object, but they are not
assigned a group or color. Therefore they don't plot, but they do take up space in a
plot!  A warning is issued in these cases, since one wouldn't normally want
a spectrum to be orphaned this way.
</p>
<p>All these problems can generally be identified by running <code><a href="#topic+sumSpectra">sumSpectra</a></code>
once the data is imported.
</p>


<h3>Advanced Tricks</h3>

<p>The ... argument can be used to pass any argument to <code>read.table</code> or <code>list.files</code>.
This includes the possibility of passing arguments that will cause trouble later, for instance
<code>na.strings</code> in <code>read.table</code>.  While one might successfully read in data with <code>NA</code>,
it will eventually cause problems.  The intent of this feature is to allow one to recurse
a directory tree containing the data, and/or to specify a starting point other than the current
working directory.  So for instance if the current working directory is not the directory containing
the data files, you can use <code>path = "my_path"</code> to point to the desired top-level
directory, and <code>recursive = TRUE</code> to work your way through a set of subdirectories.  In addition,
if you are reading in JCAMP-DX files, you can pass arguments to <code>readJDX</code> via ..., e.g. <code>SOFC = FALSE</code>.
Finally, while argument <code>fileExt</code> appears to be a file extension (from its
name and the description elsewhere), it's actually just a grep pattern that you can apply
to any part of the file name if you know how to construct the proper pattern.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>See Also</h3>

<p>Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>,
as well as <code><a href="#topic+updateGroups">updateGroups</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# This example assumes the graphics output is set to ggplot2 (see ?GraphicsOptions).
library("ggplot2")

wd &lt;- getwd() # save current location
setwd(tempdir())

# Grab an included file &amp; move to a temporary directory
tf &lt;- system.file("extdata/PCRF.jdx", package = "ChemoSpec")
chk &lt;- file.copy(from = tf, to = basename(tf))

# Now read in the file, summarize and plot
spec &lt;- files2SpectraObject(
  gr.crit = "PCRF", freq.unit = "ppm", int.unit = "intensity",
  descrip = "test import", fileExt = "\\.jdx")
sumSpectra(spec)
p &lt;- plotSpectra(spec, lab.pos = 3.5, main = "Reduced Fat Potato Chip")
p &lt;- p + ggtitle("Reduced Fat Potato Chip")
p

setwd(wd) # restore working directory

## End(Not run)

</code></pre>

<hr>
<h2 id='hcaScores'>HCA on PCA/MIA/PARAFAC scores from a Spectra or Spectra2D Object</h2><span id='topic+hcaScores'></span>

<h3>Description</h3>

<p>This function is used by <code>ChemoSpec</code> and <code>ChemoSpec2D</code>, but is
formally part of <code>ChemoSpecUtils</code>.  You can access
full documentation via <code><a href="ChemoSpecUtils.html#topic+hcaScores">hcaScores</a></code>.
</p>

<hr>
<h2 id='hcaSpectra'>Plot HCA Results of a Spectra Object</h2><span id='topic+hcaSpectra'></span>

<h3>Description</h3>

<p>A wrapper which carries out HCA and plots a dendrogram colored by the
information in a <code><a href="#topic+Spectra">Spectra</a></code> object.  Many methods for computing
the clusters and distances are available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hcaSpectra(
  spectra,
  c.method = "complete",
  d.method = "euclidean",
  use.sym = FALSE,
  leg.loc = "topright",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hcaSpectra_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="hcaSpectra_+3A_c.method">c.method</code></td>
<td>
<p>A character string describing the clustering method; must be
acceptable to <code><a href="stats.html#topic+hclust">hclust</a></code>.</p>
</td></tr>
<tr><td><code id="hcaSpectra_+3A_d.method">d.method</code></td>
<td>
<p>A character string describing the distance calculation
method; must be acceptable as a method in <code><a href="#topic+rowDist">rowDist</a></code>.</p>
</td></tr>
<tr><td><code id="hcaSpectra_+3A_use.sym">use.sym</code></td>
<td>
<p>A logical; if true, use no color and use lower-case letters
to indicate group membership.</p>
</td></tr>
<tr><td><code id="hcaSpectra_+3A_leg.loc">leg.loc</code></td>
<td>
<p>Character; if <code>"none"</code> no legend will be drawn.
Otherwise, any string acceptable to <code><a href="graphics.html#topic+legend">legend</a></code>.</p>
</td></tr>
<tr><td><code id="hcaSpectra_+3A_...">...</code></td>
<td>
<p>Other parameters to be passed to the plotting functions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list, containing an object of class <code><a href="stats.html#topic+hclust">hclust</a></code> and an
object of class <code><a href="stats.html#topic+dendrogram">dendrogram</a></code>.  The side effect is a plot.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+hclust">hclust</a></code> for the underlying function.
<code><a href="#topic+hcaScores">hcaScores</a></code> for similar analysis of PCA scores from a
<code><a href="#topic+Spectra">Spectra</a></code> object.  Additional documentation at
<a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># You need to install package "amap" for this example
if (requireNamespace("amap", quietly = TRUE)) {
  data(SrE.IR)
  myt &lt;- expression(bolditalic(Serenoa) ~ bolditalic(repens) ~ bold(IR ~ Spectra))
  res &lt;- hcaSpectra(SrE.IR, main = myt)
}

</code></pre>

<hr>
<h2 id='hmapSpectra'>Seriated Heat Map for a Spectra Object</h2><span id='topic+hmapSpectra'></span>

<h3>Description</h3>

<p>Creates a heat map with marginal dendrograms using seriation procedures.
Heirchical cluster analysis is followed by re-ordering the clusters in a coordinated way
across each dimension (controlled by argument <code>method</code>, see <code><a href="seriation.html#topic+hmap">hmap</a></code>).
The vignette for package <span class="pkg">seriation</span> has more details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hmapSpectra(spectra, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hmapSpectra_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="hmapSpectra_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed downstream.  A great deal of
control is available - check <code><a href="seriation.html#topic+hmap">hmap</a></code> for details.  Most of the control
actually derives from the <code>heatmap</code> function in package <span class="pkg">stats</span>.
See the examples.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list as described in <code><a href="seriation.html#topic+hmap">hmap</a></code>. Side effect is a plot.
</p>


<h3>Note</h3>

<p>The underlying <code>stats::heatmap</code> function does certain things automatically for the user.
For instance, if you pass a vector of names to argument <code>row_labels</code> it is automatically
reordered using the <code>rowInd</code> vector that is computed.  The user does not need to
do the reordering.  Another example is the labeling of the columns.  The labels are
automatically turned 90 degrees, and not every column is labeled to keep things readible.
</p>


<h3>Interpretation</h3>

<p>Looking at the 2nd <code>hmapSpectra</code> example, and keeping in mind the nature of the sample
(see <code><a href="#topic+SrE.IR">SrE.IR</a></code>), the most similar samples based on the ester peaks (~1740),  are in
the lower right corner of the heatmap.  These are the two outlier samples, composed of triglycerides
which contain the ester functional group (and no detectable carboxylic acid).  The most similar
samples based on the carboxylic acid peaks (~1710) are in the upper left corner.  These samples are
mostly from the &quot;pure&quot; extract group, according to the manufacturer's label.  These samples have a modest
to low amount of the ester functional group, which indicates dilution (or adulteration if you like).
In fact, the first two samples (NP_adSrE and NR_pSrE) are the two samples with the smallest ester
peaks (see first plot in the examples).  This suggests that NP_adSrE was diluted only a little with added
olive oil.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>See Also</h3>

<p><code><a href="seriation.html#topic+hmap">hmap</a></code> which will get you to the package
(there is no package index page); the vignette is a good place to begin
(<code>browseVignettes("seriation")</code>).  Additional documentation at
<a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># You need to install package "seriation" for this example
if (requireNamespace("seriation", quietly = TRUE)) {
  data(SrE.IR)

  # Let's look just at the carbonyl region
  IR &lt;- removeFreq(SrE.IR, rem.freq = SrE.IR$freq &gt; 1775 | SrE.IR$freq &lt; 1660)
  p &lt;- plotSpectra(IR, which = 1:16, lab.pos = 1800)

  # Defaults, except for color scheme:
  res &lt;- hmapSpectra(IR, col = heat.colors(5))

  # Label samples and frequencies by passing arguments to stats:heatmap
  # Also make a few other nice plot adjustments
  res &lt;- hmapSpectra(IR,
    col = heat.colors(5),
    row_labels = IR$names, col_labels = as.character(round(IR$freq)),
    margins = c(4, 6)
    )
}

</code></pre>

<hr>
<h2 id='hypTestScores'>Conduct MANOVA using PCA Scores and Factors in a Spectra Object</h2><span id='topic+hypTestScores'></span>

<h3>Description</h3>

<p>This function provides a convenient interface for carrying out manova using
the scores from PCA and the factors (groups) stored in a
<code><a href="#topic+Spectra">Spectra</a></code> object.  The function will do anova as well, if you
only provide one vector of scores, though this is probably of limited use.
A <code><a href="#topic+Spectra">Spectra</a></code> object contains group information stored in its
<code>spectra$groups</code> element, but you can also use
<code><a href="#topic+splitSpectraGroups">splitSpectraGroups</a></code> to generate additional groups/factors that
might be more useful than the original.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hypTestScores(spectra, pca, pcs = 1:3, fac = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hypTestScores_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="hypTestScores_+3A_pca">pca</code></td>
<td>
<p>An object of class <code><a href="stats.html#topic+prcomp">prcomp</a></code>.</p>
</td></tr>
<tr><td><code id="hypTestScores_+3A_pcs">pcs</code></td>
<td>
<p>An integer vector giving the PCA scores to use as the response in
the manova analysis.</p>
</td></tr>
<tr><td><code id="hypTestScores_+3A_fac">fac</code></td>
<td>
<p>A character vector giving the factors to be used in the manova.
They will be searched for within the <code><a href="#topic+Spectra">Spectra</a></code> object.</p>
</td></tr>
<tr><td><code id="hypTestScores_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed downstream, in this case to
<code>aov</code>.  Untested.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is an extraordinarily thin wrapper which helps the user to
avoid writing a very tedious <code>formula</code> specification.
</p>


<h3>Value</h3>

<p>The results of the analysis print to the console unless assigned.
If assigned, the object class is one of several described in
<code><a href="stats.html#topic+aov">aov</a></code> depending upon the data passed to it.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+splitSpectraGroups">splitSpectraGroups</a></code> which can be used to create
additional factor elements in the <code><a href="#topic+Spectra">Spectra</a></code> object, which can then be
used with this function.  Additional documentation at
<a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(metMUD2)

# Original factor encoding:
levels(metMUD2$groups)

# Split those original levels into 2 new ones (re-code them)
new.grps &lt;- list(geneBb = c("B", "b"), geneCc = c("C", "c"))
mM3 &lt;- splitSpectraGroups(metMUD2, new.grps)

# Now do the PCA and anova, with 3 ways to see the results
pca &lt;- c_pcaSpectra(mM3)
res &lt;- hypTestScores(mM3, pca, fac = c("geneBb", "geneCc"))
res
summary(res)
summary.aov(res)

# You can also call this function on the existing groups:
res &lt;- hypTestScores(metMUD2, pca, fac = "groups")

</code></pre>

<hr>
<h2 id='irlba_pcaSpectra'>IRLBA PCA of Spectra Objects</h2><span id='topic+irlba_pcaSpectra'></span>

<h3>Description</h3>

<p>A wrapper which carries out IRLBA PCA analysis on a
<code><a href="#topic+Spectra">Spectra</a></code> object.  The user can select various options for
scaling.  There is no normalization by rows - do this manually using
<code><a href="#topic+normSpectra">normSpectra</a></code>. The data can be supplied already centered
if desired.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>irlba_pcaSpectra(spectra, choice = "noscale", n = 3, center = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="irlba_pcaSpectra_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="irlba_pcaSpectra_+3A_choice">choice</code></td>
<td>
<p>A character string indicating the choice of scaling.  One of
<code>c("noscale"</code>, <code>"autoscale"</code>, <code>"Pareto")</code>. <code>"autoscale"</code>
is called &quot;standard normal variate&quot; or &quot;correlation matrix PCA&quot; in some literature.</p>
</td></tr>
<tr><td><code id="irlba_pcaSpectra_+3A_n">n</code></td>
<td>
<p>Integer. The number of components desired.</p>
</td></tr>
<tr><td><code id="irlba_pcaSpectra_+3A_center">center</code></td>
<td>
<p>Logical.  Should the data be centered?  Data must be centered
for PCA, either before arriving here or via this argument.</p>
</td></tr>
<tr><td><code id="irlba_pcaSpectra_+3A_...">...</code></td>
<td>
<p>Other parameters to be passed to <code><a href="irlba.html#topic+irlba">irlba</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The scale choice <code>autoscale</code> scales the columns by their standard
deviation.  <code>Pareto</code> scales by the square root of the standard
deviation.
</p>


<h3>Value</h3>

<p>A modified object of class <code>prcomp</code> and <code>computed_via_irlba</code>,
which includes a list element called <code>$method</code>, a character string describing the
pre-processing carried out and the type of PCA performed (used to annotate
plots).
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>References</h3>

<p>J. Baglama and L. Reichel, &quot;Augmented Implicitly Restarted Lanczos
Bidiagonalization Methods&quot;  <em>SIAM J. Sci. Comput.</em> (2005).
</p>


<h3>See Also</h3>

<p><code><a href="irlba.html#topic+prcomp_irlba">prcomp_irlba</a></code> for the underlying function,
<code><a href="#topic+c_pcaSpectra">c_pcaSpectra</a></code> for classical PCA calculations,
<code><a href="#topic+r_pcaSpectra">r_pcaSpectra</a></code> for robust PCA calculations,
<code><a href="#topic+s_pcaSpectra">s_pcaSpectra</a></code> for sparse PCA calculations.
Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>
<p>For displaying the results, <code><a href="ChemoSpecUtils.html#topic+plotScree">ChemoSpecUtils::plotScree()</a></code>, <code><a href="ChemoSpecUtils.html#topic+plotScores">ChemoSpecUtils::plotScores()</a></code>, <code><a href="#topic+plotLoadings">plotLoadings()</a></code>, <code><a href="#topic+plot2Loadings">plot2Loadings()</a></code>, <code><a href="#topic+sPlotSpectra">sPlotSpectra()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# This example assumes the graphics output is set to ggplot2 (see ?GraphicsOptions).
library("ggplot2")
data(SrE.NMR)
pca &lt;- irlba_pcaSpectra(SrE.NMR)

p1 &lt;- plotScree(pca)
p1

p2 &lt;- plotScores(SrE.NMR, pca, pcs = c(1, 2), ellipse = "cls", tol = 0.05)
p2 &lt;- p2 + ggtitle("Scores: SrE NMR Data")
p2

p3 &lt;- plotLoadings(SrE.NMR, pca, loads = 1:2, ref = 1)
p3 &lt;- p3 + ggtitle("Loadings: SrE NMR Data")
p3

## End(Not run)

</code></pre>

<hr>
<h2 id='mclust3dSpectra'>mclust Analysis of a Spectra Object in 3D</h2><span id='topic+mclust3dSpectra'></span>

<h3>Description</h3>

<p>This function conducts an mclust analysis of the PCA results of a
<code><a href="#topic+Spectra">Spectra</a></code> object and displays the results in 3D.  Classical or
robust confidence ellipses can be added if desired. Improperly classified
data points can be marked.  The interactive plot is made via <code>plotly</code>
and appears in a browser window. Note that the confidence ellipses computed
here are generated independently of the <code>Mclust</code> results - they do not
correspond to the ellipses seen in 2D plots from <code>Mclust</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mclust3dSpectra(
  spectra,
  pca,
  pcs = 1:3,
  ellipse = TRUE,
  rob = FALSE,
  cl = 0.95,
  frac.pts.used = 0.8,
  truth = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mclust3dSpectra_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="mclust3dSpectra_+3A_pca">pca</code></td>
<td>
<p>An object of class <code><a href="stats.html#topic+prcomp">prcomp</a></code>.</p>
</td></tr>
<tr><td><code id="mclust3dSpectra_+3A_pcs">pcs</code></td>
<td>
<p>An integer vector describing which PCs to use.</p>
</td></tr>
<tr><td><code id="mclust3dSpectra_+3A_ellipse">ellipse</code></td>
<td>
<p>Logical indicating if confidence ellipses should be drawn.</p>
</td></tr>
<tr><td><code id="mclust3dSpectra_+3A_rob">rob</code></td>
<td>
<p>Logical; if <code>ellipse = TRUE</code>, indicates that robust
confidence ellipses should be drawn.  If <code>FALSE</code>, classical confidence
ellipses are drawn.</p>
</td></tr>
<tr><td><code id="mclust3dSpectra_+3A_cl">cl</code></td>
<td>
<p>A number indicating the confidence interval for the ellipse.</p>
</td></tr>
<tr><td><code id="mclust3dSpectra_+3A_frac.pts.used">frac.pts.used</code></td>
<td>
<p>If <code>ellipse = TRUE</code> and <code>rob = TRUE</code>, a
number indicating the fraction of the data points to be considered &quot;good&quot;
and thus used to compute the robust confidence ellipse.</p>
</td></tr>
<tr><td><code id="mclust3dSpectra_+3A_truth">truth</code></td>
<td>
<p>A character vector indicating the known group membership for
reach row of the PC scores.  Generally this would be <code>spectra$groups</code>.</p>
</td></tr>
<tr><td><code id="mclust3dSpectra_+3A_...">...</code></td>
<td>
<p>Arguments to be passed to <code>mclust</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The mclust model is returned invisibly, and a plot is produced.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>See Also</h3>

<p><code><a href="mclust.html#topic+Mclust">Mclust</a></code> for background on the method.
Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(mclust)
data(metMUD1)
pca &lt;- c_pcaSpectra(metMUD1)
mclust3dSpectra(metMUD1, pca)

# show mis-classified points
mclust3dSpectra(metMUD1, pca, truth = metMUD1$groups)

## End(Not run)


</code></pre>

<hr>
<h2 id='mclustSpectra'>mclust Analysis of a Spectra Object PCA Results</h2><span id='topic+mclustSpectra'></span>

<h3>Description</h3>

<p>This function is a wrapper for the <code>Mclust</code> function and associated
plotting functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mclustSpectra(
  spectra,
  pca,
  pcs = c(1:3),
  dims = c(1, 2),
  plot = c("BIC", "proj", "errors"),
  use.sym = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mclustSpectra_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="mclustSpectra_+3A_pca">pca</code></td>
<td>
<p>An object of class <code><a href="stats.html#topic+prcomp">prcomp</a></code>.</p>
</td></tr>
<tr><td><code id="mclustSpectra_+3A_pcs">pcs</code></td>
<td>
<p>An integer vector describing which PCs to use.</p>
</td></tr>
<tr><td><code id="mclustSpectra_+3A_dims">dims</code></td>
<td>
<p>A integer vector giving the PCA dimensions to use.</p>
</td></tr>
<tr><td><code id="mclustSpectra_+3A_plot">plot</code></td>
<td>
<p>A character string indicating what plot to make.  Options are
<code>c("BIC", "proj", "error")</code>; see <code>Mclust</code> for details.</p>
</td></tr>
<tr><td><code id="mclustSpectra_+3A_use.sym">use.sym</code></td>
<td>
<p>Logical; if true, the color scheme is changed to black and
symbols are used for plotting.</p>
</td></tr>
<tr><td><code id="mclustSpectra_+3A_...">...</code></td>
<td>
<p>Other parameters to be passed downstream.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The <code>Mclust</code> model is returned invisibly, and a plot is made.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>See Also</h3>

<p><code><a href="mclust.html#topic+Mclust">Mclust</a></code> for background on the method.
Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require("mclust")
data(metMUD1)
cls &lt;- c_pcaSpectra(metMUD1, choice = "autoscale")

p &lt;- plotScores(metMUD1, cls)

mclustSpectra(metMUD1, cls, plot = "BIC")
mclustSpectra(metMUD1, cls, plot = "proj")
mclustSpectra(metMUD1, cls, plot = "errors", truth = metMUD1$groups)

## End(Not run)

</code></pre>

<hr>
<h2 id='metMUD1'>Made Up NMR Data Sets</h2><span id='topic+metMUD1'></span><span id='topic+alignMUD'></span><span id='topic+metMUD2'></span>

<h3>Description</h3>

<p>These data sets are simulated 300 MHz NMR spectra.  They are designed mainly
to illustrate certain chemometric methods and are small enough that they
process quickly.
</p>


<h3>Format</h3>

<p>The data is stored as a <code><a href="#topic+Spectra">Spectra</a></code> object.
</p>


<h3>Details</h3>

<p><code>alignMUD</code> is a series of mis-aligned spectra of a single small organic
molecule.
</p>
<p><code>metMUD1</code> is composed of 20 samples, each a mixture of four typical
small organic compounds (we'll leave it to the reader as an exercise to
deduce the spin systems!).  These compounds are present in varying random
amounts.  Ten of the samples are control samples, and ten are treatment
samples.  Thus you can run PCA and other methods on this data set, and
expect to see a separation.  This data set is normalized.
</p>
<p><code>metMUD2</code> also consists of 20 samples of mixtures of the same four
compounds.  However, the concentrations of some of the compounds are
correlated with other compounds, both positively and negatively, and some
concentrations are random. <code>metMUD2</code> is divided into different sample
groups which correspond conceptually to two genes, each active or knocked
out.  This data set is designed to be similar to a metabolomics data set in
which the concentrations of some compounds co-vary, and others are
independent. This data set is normalized.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>Source</h3>

<p>Created using various tools.  Contact the author for a script if
interested.
</p>


<h3>See Also</h3>

<p>Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(metMUD1)
sumSpectra(metMUD1)
#
data(metMUD2)
sumSpectra(metMUD2)
</code></pre>

<hr>
<h2 id='normSpectra'>Normalize a Spectra Object</h2><span id='topic+normSpectra'></span>

<h3>Description</h3>

<p>This function carries out normalization of the spectra in a
<code><a href="#topic+Spectra">Spectra</a></code> object.  There are currently four options:
</p>

<ul>
<li> <p><code>"PQN"</code> carries out &quot;Probabalistic Quotient Normalization&quot; as described
in the reference.  This is probably the best option for many data sets.  However, please
be careful if your sample has protein in it, PQN is potentially biased.
See the references.
</p>
</li>
<li> <p><code>"TotInt"</code> normalizes by total intensity.  In this
case, the y-data of a <code><a href="#topic+Spectra">Spectra</a></code> object is normalized by dividing
each y-value by the sum of the y-values in a given spectrum.  Thus each
spectrum sums to 1.  This method assumes that the total concentration of
all substances giving peaks does not vary across samples which may not be true.
</p>
</li>
<li> <p><code>"Range"</code> allows one to do something similar to <code>"TotInt"</code> but rather than using the
sum of the entire spectrum as the denominator, only the sum of the given
range is used.  This would be appropriate if there was an internal standard
in the spectrum which was free of interferance, and one wanted to normalize
relative to it.
</p>
</li>
<li> <p><code>"zero2one"</code> scales each spectrum separately to a <code style="white-space: pre;">&#8288;[0...1]&#8288;</code> scale.
This is sometimes useful for visual comparison of chromatograms but is
inappropriate for spectral data sets.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>normSpectra(spectra, method = "PQN", RangeExpress = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="normSpectra_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="normSpectra_+3A_method">method</code></td>
<td>
<p>One of <code>c("PQN", "TotInt", "Range", "zero2one")</code> giving
the method for normalization.</p>
</td></tr>
<tr><td><code id="normSpectra_+3A_rangeexpress">RangeExpress</code></td>
<td>
<p>A vector of
logicals (must be of <code>length(Spectra$freq)</code>).  This vector should be <code>TRUE</code> for
the frequency range you want to serve as the basis for norming, and <code>FALSE</code> otherwise.
The entire spectrum will be divided by the sum of the <code>TRUE</code> range.  See the examples.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>References</h3>


<ul>
<li> <p>Probabalistic Quotient Normalization is reported in F. Dieterle
et al. Analytical Chemistry vol. 78 pages 4281-4290 (2006).
</p>
</li>
<li> <p>The exact same mathematics are called &quot;median fold change normalization&quot;
by Nicholson's group, reported in K. A. Veselkov et. al. Analytical
Chemistry vol. 83 pages 5864-5872 (2011).
</p>
</li>
<li> <p>Corriea et al. &quot;1H NMR Signals from Urine Excreted Protein are
a Source of Bias in Probablistic Quotient Normalization&quot; Analytical Chemistry
vol. 94 pages 6919-6923 (2022).
</p>
</li></ul>



<h3>See Also</h3>

<p>Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># This example assumes the graphics output is set to ggplot2 (see ?GraphicsOptions).
library("ggplot2")
data(SrE.IR)

# Reference spectrum before normalization
p1 &lt;- plotSpectra(SrE.IR) + ggtitle("Original Spectrum")
p1

# Default PQN normalization
res1 &lt;- normSpectra(SrE.IR)
p2 &lt;- plotSpectra(res1) + ggtitle("PQN Normalization")
p2

# Norm over carbonyl region
RE &lt;- SrE.IR$freq &gt; 1650 &amp; SrE.IR$freq &lt; 1800
res2 &lt;- normSpectra(SrE.IR, method = "Range", RangeExpress = RE)
p3 &lt;- plotSpectra(res2) + ggtitle("Normalized to Carbonyl Peaks")
p3

# Check numerically
rowSums(res2$data[, RE]) # compare to rowSums(SrE.IR$data[,RE])

</code></pre>

<hr>
<h2 id='pcaDiag'>Outlier Diagnostic Plots for PCA of a Spectra Object</h2><span id='topic+pcaDiag'></span>

<h3>Description</h3>

<p>A function to carry diagnostics on the PCA results for a
<code><a href="#topic+Spectra">Spectra</a></code> object.  Basically a wrapper to Filzmoser's
<code><a href="chemometrics.html#topic+pcaDiagplot">pcaDiagplot</a></code> which colors everything according to
the scheme stored in the <code><a href="#topic+Spectra">Spectra</a></code> object.  Works with PCA
results of either class <code>prcomp</code> or class <code>princomp</code>.  Works
with either classical or robust PCA results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pcaDiag(
  spectra,
  pca,
  pcs = 3,
  quantile = 0.975,
  plot = c("OD", "SD"),
  use.sym = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pcaDiag_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="pcaDiag_+3A_pca">pca</code></td>
<td>
<p>An object of class <code><a href="stats.html#topic+prcomp">prcomp</a></code>
modified to include a character string (<code>$method</code>) describing the
pre-processing carried out and the type of PCA performed.</p>
</td></tr>
<tr><td><code id="pcaDiag_+3A_pcs">pcs</code></td>
<td>
<p>As per <code><a href="chemometrics.html#topic+pcaDiagplot">pcaDiagplot</a></code>.  The number of
principal components to include.</p>
</td></tr>
<tr><td><code id="pcaDiag_+3A_quantile">quantile</code></td>
<td>
<p>As per <code><a href="chemometrics.html#topic+pcaDiagplot">pcaDiagplot</a></code>.  The
significance criteria to use as a cutoff.</p>
</td></tr>
<tr><td><code id="pcaDiag_+3A_plot">plot</code></td>
<td>
<p>A character string, indicating whether to plot the score
distances or orthogonal distances, or both.  Options are <code>c("OD",
"SD")</code>.</p>
</td></tr>
<tr><td><code id="pcaDiag_+3A_use.sym">use.sym</code></td>
<td>
<p>logical; if true, the color scheme is change to black and
symbols are used for plotting.</p>
</td></tr>
<tr><td><code id="pcaDiag_+3A_...">...</code></td>
<td>
<p>Parameters to be passed to the plotting routines. <em>Applies to base graphics only</em>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If both plots are desired, the output should be directed to a file rather
than the screen.  Otherwise, the 2nd plot overwrites the 1st in the active
graphics window.  Alternatively, just call the function twice, once
specifying <code>OD</code> and once specifying <code>SD</code>.
</p>


<h3>Value</h3>

<p>The returned value depends on the graphics option selected (see <code><a href="ChemoSpecUtils.html#topic+GraphicsOptions">ChemoSpecUtils::GraphicsOptions()</a></code>).
</p>

<ul>
<li> <p><strong>base</strong> A data frame or list containing the data plotted.  Assign the value and run <code>str()</code> or <code>names()</code> on it to see what it contains. Side effect is a plot.
</p>
</li>
<li> <p><strong>ggplot2</strong> The plot is displayed, and a <code>ggplot2</code> plot object is returned if the value is assigned. The plot can be modified in the usual <code>ggplot2</code> manner. If you want the plotted values, you can access them via the base graphics mode.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University), Tejasvi Gupta.
</p>


<h3>References</h3>

<p>K. Varmuza and P. Filzmoser <em>Introduction to Multivariate
Statistical Analysis in Chemometrics</em>, CRC Press, 2009.
</p>


<h3>See Also</h3>

<p><code><a href="chemometrics.html#topic+pcaDiagplot">pcaDiagplot</a></code> in package
<code>chemometrics</code> for the underlying function. Additional documentation at
<a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># This example assumes the graphics output is set to ggplot2 (see ?GraphicsOptions).
library("ggplot2")
data(SrE.IR)
pca &lt;- c_pcaSpectra(SrE.IR, choice = "noscale")
p1 &lt;- pcaDiag(SrE.IR, pca, pcs = 2, plot = "OD") + ggtitle("OD Plot")
p1
p2 &lt;- pcaDiag(SrE.IR, pca, pcs = 2, plot = "SD") + ggtitle("SD Plot")
p2

</code></pre>

<hr>
<h2 id='plot2Loadings'>Plot PCA Loadings from a Spectra Object Against Each Other</h2><span id='topic+plot2Loadings'></span>

<h3>Description</h3>

<p>Plots two PCA loadings specified by the user, and labels selected (extreme)
points.  Typically used to determine which variables (frequencies) are
co-varying, although in spectroscopy most peaks are represented by several
variables and hence there is a lot of co-varying going on.  Also useful to
determine which variables are contributing the most to the clustering on a
score plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot2Loadings(spectra, pca, loads = c(1, 2), tol = 0.05, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot2Loadings_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="plot2Loadings_+3A_pca">pca</code></td>
<td>
<p>An object of class <code><a href="stats.html#topic+prcomp">prcomp</a></code>, modified to include a
list element called <code>$method</code>, a character string describing the
pre-processing carried out and the type of PCA performed (it appears on the
plot).  This is automatically provided if <code>ChemoSpec</code> functions
<code><a href="#topic+c_pcaSpectra">c_pcaSpectra</a></code> or <code><a href="#topic+r_pcaSpectra">r_pcaSpectra</a></code> were used to create
<code>pca</code>.</p>
</td></tr>
<tr><td><code id="plot2Loadings_+3A_loads">loads</code></td>
<td>
<p>A vector of two integers specifying which loading vectors to
plot.</p>
</td></tr>
<tr><td><code id="plot2Loadings_+3A_tol">tol</code></td>
<td>
<p>A number describing the fraction of points to be labeled. <code>tol = 1.0</code> labels all the points; <code>tol = 0.05</code> labels <em>approximately</em> the most extreme 5 percent. Set to <code>'none'</code> to completely suppress labels. Note that a simple approach based upon quantiles is used, assumes that both x and y are each normally distributed, and treats x and y separately.  Thus, this is not a formal treatment of outliers, just a means of labeling points. Groups are lumped together for the computation.</p>
</td></tr>
<tr><td><code id="plot2Loadings_+3A_...">...</code></td>
<td>
<p>Parameters to be passed to the plotting routines. <em>Applies to base graphics only</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The returned value depends on the graphics option selected (see <code>GraphicsOptions()</code>).
</p>

<ul>
<li> <p><code>base</code>: None.  Side effect is a plot.
</p>
</li>
<li> <p><code>ggplot2</code>: The plot is displayed, and a <code>ggplot2</code> object is returned if the value is assigned.  The plot can be modified in the usual <code>ggplot2</code> manner.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University), Tejasvi Gupta.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+plotLoadings">plotLoadings</a></code> to plot one loading against the
original variable (frequency) axis.  See <code><a href="#topic+sPlotSpectra">sPlotSpectra</a></code> for
a different approach.  Additional documentation at
<a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# This example assumes the graphics output is set to ggplot2 (see ?GraphicsOptions).
library("ggplot2")
data(SrE.IR)
pca &lt;- c_pcaSpectra(SrE.IR)
myt &lt;- expression(bolditalic(Serenoa) ~ bolditalic(repens) ~ bold(IR ~ Spectra))
p &lt;- res &lt;- plot2Loadings(SrE.IR, pca, loads = c(1, 2), tol = 0.001)
p &lt;- p + ggtitle(myt)
p

</code></pre>

<hr>
<h2 id='plot3dScores'>3D PCA Score Plot for a Spectra Object</h2><span id='topic+plot3dScores'></span>

<h3>Description</h3>

<p>Creates an interactive 3D plot of PCA scores from the analysis of a
<code><a href="#topic+Spectra">Spectra</a></code> object, color coded according the to scheme stored in
the object.  The plot is created by <code>plotly</code> and appears in a browser window.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot3dScores(
  spectra,
  pca,
  pcs = c(1:3),
  ellipse = TRUE,
  rob = FALSE,
  cl = 0.95,
  frac.pts.used = 0.8
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot3dScores_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="plot3dScores_+3A_pca">pca</code></td>
<td>
<p>An object of class <code><a href="stats.html#topic+prcomp">prcomp</a></code>.</p>
</td></tr>
<tr><td><code id="plot3dScores_+3A_pcs">pcs</code></td>
<td>
<p>A vector of three integers specifying the PCA scores to plot.</p>
</td></tr>
<tr><td><code id="plot3dScores_+3A_ellipse">ellipse</code></td>
<td>
<p>Logical indicating if confidence ellipses should be drawn.</p>
</td></tr>
<tr><td><code id="plot3dScores_+3A_rob">rob</code></td>
<td>
<p>Logical; if <code>ellipse = TRUE</code>, indicates that robust
confidence ellipses should be drawn.  If <code>FALSE</code>, classical confidence
ellipses are drawn.</p>
</td></tr>
<tr><td><code id="plot3dScores_+3A_cl">cl</code></td>
<td>
<p>A number indicating the confidence interval for the ellipse.</p>
</td></tr>
<tr><td><code id="plot3dScores_+3A_frac.pts.used">frac.pts.used</code></td>
<td>
<p>If <code>ellipse = TRUE</code> and <code>rob = TRUE</code>, a
number indicating the fraction of the data points to be considered &quot;good&quot;
and thus used to compute the robust confidence ellipse.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None.  Side effect is a plot in a browser window.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>See Also</h3>

<p>Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  data(metMUD1)
  pca &lt;- c_pcaSpectra(metMUD1, choice = "noscale")
  p &lt;- plot3dScores(metMUD1, pca)
  p
}

</code></pre>

<hr>
<h2 id='plotLoadings'>Plot PCA Loadings for a Spectra Object</h2><span id='topic+plotLoadings'></span>

<h3>Description</h3>

<p>Creates a multi-panel plot of loadings along with a reference spectrum.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotLoadings(spectra, pca, loads = c(1), ref = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotLoadings_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="plotLoadings_+3A_pca">pca</code></td>
<td>
<p>An object of class <code><a href="stats.html#topic+prcomp">prcomp</a></code>, modified to include a
list element called <code>$method</code>, a character string describing the
pre-processing carried out and the type of PCA performed (it appears on the
plot).  This is automatically provided if <code>ChemoSpec</code> functions
<code><a href="#topic+c_pcaSpectra">c_pcaSpectra</a></code> or <code><a href="#topic+r_pcaSpectra">r_pcaSpectra</a></code> were used to create
<code>pca</code>.</p>
</td></tr>
<tr><td><code id="plotLoadings_+3A_loads">loads</code></td>
<td>
<p>An integer vector giving the loadings to plot.  More than 3
loadings creates a useless plot using the default graphics window.</p>
</td></tr>
<tr><td><code id="plotLoadings_+3A_ref">ref</code></td>
<td>
<p>An integer specifying the reference spectrum to plot, which
appears at the bottom of the plot.</p>
</td></tr>
<tr><td><code id="plotLoadings_+3A_...">...</code></td>
<td>
<p>Parameters to be passed to the plotting routines. <em>Applies to base graphics only</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The returned value depends on the graphics option selected (see <code>GraphicsOptions()</code>).
</p>

<ul>
<li> <p><code>base</code>: None.  Side effect is a plot.
</p>
</li>
<li> <p><code>ggplot2</code>: The plot is displayed, and a <code>ggplot2</code> object is returned if the value is assigned.  The plot can be modified in the usual <code>ggplot2</code> manner.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University), Tejasvi Gupta.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+c_pcaSpectra">c_pcaSpectra</a></code> for an example.  See <code><a href="#topic+plot2Loadings">plot2Loadings</a></code>
to plot two loadings against each
other, and <code><a href="#topic+sPlotSpectra">sPlotSpectra</a></code> for an alternative approach.
Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>

<hr>
<h2 id='plotScores'>Plot Scores from PCA, MIA or PARAFAC Analysis of a Spectra or Spectra2D Object</h2><span id='topic+plotScores'></span>

<h3>Description</h3>

<p>This function is used by <code>ChemoSpec</code> and <code>ChemoSpec2D</code>, but is
formally part of <code>ChemoSpecUtils</code>.  You can access
full documentation via <code><a href="ChemoSpecUtils.html#topic+plotScores">plotScores</a></code>.
</p>

<hr>
<h2 id='plotScree'>Scree Plots from PCA or MIA Analysis of a Spectra or Spectra2D Object</h2><span id='topic+plotScree'></span>

<h3>Description</h3>

<p>This function is used by <code>ChemoSpec</code> and <code>ChemoSpec2D</code>, but is
formally part of <code>ChemoSpecUtils</code>.  You can access
full documentation via <code><a href="ChemoSpecUtils.html#topic+plotScree">plotScree</a></code>.
</p>

<hr>
<h2 id='plotSpectra'>Plot Spectra Object</h2><span id='topic+plotSpectra'></span>

<h3>Description</h3>

<p>Plots the spectra stored in a <code><a href="#topic+Spectra">Spectra</a></code> object. Spectra may be plotted
offset or stacked.  The vertical scale is controlled by a combination of
several parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSpectra(
  spectra,
  which = c(1),
  yrange = range(spectra$data),
  offset = 0,
  amplify = 1,
  lab.pos = mean(spectra$freq),
  showGrid = TRUE,
  leg.loc = "none",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotSpectra_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="plotSpectra_+3A_which">which</code></td>
<td>
<p>An integer vector specifying which spectra to plot, and the
order.</p>
</td></tr>
<tr><td><code id="plotSpectra_+3A_yrange">yrange</code></td>
<td>
<p>A vector giving the limits of the y axis desired, for instance
<code>c(0, 15)</code>.  This parameter depends upon the range of values in the
stored spectra and defaults to the height of the largest peak in the data
set.  Interacts with the next two arguments, as well as the number of
spectra to be plotted as given in <code>which</code>.  Trial and error is used to
adjust all these arguments to produce the desired plot.</p>
</td></tr>
<tr><td><code id="plotSpectra_+3A_offset">offset</code></td>
<td>
<p>A number specifying the vertical offset between spectra if
more than one is plotted.  Set to 0.0 to overlay the spectra.</p>
</td></tr>
<tr><td><code id="plotSpectra_+3A_amplify">amplify</code></td>
<td>
<p>A number specifying an amplification factor to be applied to
all spectra.  Useful for magnifying spectra so small features show up
(though large peaks will then be clipped, unless you zoom on the x axis).</p>
</td></tr>
<tr><td><code id="plotSpectra_+3A_lab.pos">lab.pos</code></td>
<td>
<p>A number (in frequency units) giving the location of a label for each spectrum.
Generally, pick an area that is clear in all spectra plotted.  If no label
is desired, set <code>lab.pos = "none"</code>.</p>
</td></tr>
<tr><td><code id="plotSpectra_+3A_showgrid">showGrid</code></td>
<td>
<p>Logical.  Places light gray vertical lines at each tick mark
if <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plotSpectra_+3A_leg.loc">leg.loc</code></td>
<td>
<p>Either a list with elements <code>x</code> and <code>y</code>, or a string like <code>'topright'</code>.  Values in a list should be on <code style="white-space: pre;">&#8288;[0,1]&#8288;</code>, i.e. the lower left of the plot area is <code style="white-space: pre;">&#8288;0,0&#8288;</code> and the upper right is <code style="white-space: pre;">&#8288;1,1&#8288;</code>.  Allowed string values are those described in <code><a href="graphics.html#topic+legend">graphics::legend()</a></code> under 'Details'. A value of <code>'none'</code> is acceptable as well.</p>
</td></tr>
<tr><td><code id="plotSpectra_+3A_...">...</code></td>
<td>
<p>Parameters to be passed to the plotting routines. <em>Applies to base graphics only</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The returned value depends on the graphics option selected (see <code>GraphicsOptions()</code>).
</p>

<ul>
<li> <p><code>base</code>: None.  Side effect is a plot.
</p>
</li>
<li> <p><code>ggplot2</code>: The plot is displayed, and a <code>ggplot2</code> object is returned if the value is assigned.  The plot can be modified in the usual <code>ggplot2</code> manner.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University), Tejasvi Gupta.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotSpectraJS">plotSpectraJS</a></code> for the interactive version. See <code>GraphicsOptions</code>
for more information about the graphics options. Additional documentation at
<a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># This example assumes the graphics output is set to ggplot2 (see ?GraphicsOptions).
library("ggplot2")
data(metMUD1)

p1 &lt;- plotSpectra(metMUD1, which = c(10, 11), yrange = c(0, 1.5),
  offset = 0.06, amplify = 10, lab.pos = 0.5)
p1 &lt;- p1 + ggtitle("metMUD1 NMR Data")
p1

# Add a legend at x, y coords
p2 &lt;- plotSpectra(metMUD1, which = c(10, 11), yrange = c(0, 1.5),
  offset = 0.06, amplify = 10, lab.pos = 0.5, leg.loc = list(x = 0.8, y = 0.8))
p2 &lt;- p2 + ggtitle("metMUD1 NMR Data")
p2

</code></pre>

<hr>
<h2 id='plotSpectraDist'>Plot the Distance Between Spectra and a Reference Spectrum in a Spectra Object</h2><span id='topic+plotSpectraDist'></span>

<h3>Description</h3>

<p>This function plots the distance between a reference spectrum and all other
spectra in a <code><a href="#topic+Spectra">Spectra</a></code> object.  Distance can be defined in a number of
ways (see Arguments).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSpectraDist(spectra, method = "pearson", ref = 1, labels = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotSpectraDist_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="plotSpectraDist_+3A_method">method</code></td>
<td>
<p>Character.  Any method acceptable to <code><a href="#topic+rowDist">rowDist</a></code>.</p>
</td></tr>
<tr><td><code id="plotSpectraDist_+3A_ref">ref</code></td>
<td>
<p>Integer.  The spectrum to be used as a reference.</p>
</td></tr>
<tr><td><code id="plotSpectraDist_+3A_labels">labels</code></td>
<td>
<p>Logical.  Shall the points be labeled?</p>
</td></tr>
<tr><td><code id="plotSpectraDist_+3A_...">...</code></td>
<td>
<p>Parameters to be passed to the plotting routines. <em>Applies to base graphics only</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The returned value depends on the graphics option selected (see <code><a href="ChemoSpecUtils.html#topic+GraphicsOptions">ChemoSpecUtils::GraphicsOptions()</a></code>).
</p>

<ul>
<li> <p><strong>base</strong> A data frame or list containing the data plotted.  Assign the value and run <code>str()</code> or <code>names()</code> on it to see what it contains. Side effect is a plot.
</p>
</li>
<li> <p><strong>ggplot2</strong> The plot is displayed, and a <code>ggplot2</code> plot object is returned if the value is assigned. The plot can be modified in the usual <code>ggplot2</code> manner. If you want the plotted values, you can access them via the base graphics mode.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University), Tejasvi Gupta.
</p>


<h3>See Also</h3>

<p>To compare all spectra simultaneously in a heatmap, see
<code><a href="ChemoSpecUtils.html#topic+sampleDist">sampleDist</a></code>.  Additional documentation
at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# You need to install package "amap" to use this function
if (requireNamespace("amap", quietly = TRUE)) {
  # This example assumes the graphics output is set to ggplot2 (see ?GraphicsOptions).
  library("ggplot2")
  data(SrE.NMR)
  txt1 &lt;- paste("Distance from", SrE.NMR$names[1])
  txt2 &lt;- paste("Rank Distance from", SrE.NMR$names[1])
  p &lt;- plotSpectraDist(SrE.NMR)
  p &lt;- p + labs(title = txt1, xlab = txt2, ylab = txt2) +
           coord_cartesian(ylim = c(0, 1.1), xlim = c(0, 16))
p
}

</code></pre>

<hr>
<h2 id='plotSpectraJS'>Plot a Spectra Object Interactively</h2><span id='topic+plotSpectraJS'></span>

<h3>Description</h3>

<p>This function uses the d3.js JavaScript library by Mike Bostock to plot a <code><a href="#topic+Spectra">Spectra</a></code>
object interactively.  This is most useful for data exploration.  For high
quality plots, consider <code><a href="#topic+plotSpectra">plotSpectra</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSpectraJS(spectra, which = NULL, browser = NULL, minify = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotSpectraJS_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="plotSpectraJS_+3A_which">which</code></td>
<td>
<p>Integer.  If not <code>NULL</code>, specifies by number which spectra to plot.
If greater control is needed, use <code><a href="#topic+removeSample">removeSample</a></code> which is more flexible
before calling this function.</p>
</td></tr>
<tr><td><code id="plotSpectraJS_+3A_browser">browser</code></td>
<td>
<p>Character.  Something that will make sense to your OS.  Only
necessary if you want to override your system specified browser as
understood by <code>R</code>.  See below for further details.</p>
</td></tr>
<tr><td><code id="plotSpectraJS_+3A_minify">minify</code></td>
<td>
<p>Logical.  Shall the JavaScript be minified?  This improves
performance.  However, it requires package <code>js</code> which in turn requires
package <code>V8</code>.  The latter is not available on all platforms.  Details
may be available at <a href="https://github.com/jeroen/V8">https://github.com/jeroen/V8</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The spectral data are incorporated into the web page. Keep in mind that very large
data sets, like NMR spectra with 32K points, will bog down the browser.
In these cases, you may need to limit the number of samples in passed to this function.
See <code><a href="#topic+removeSample">removeSample</a></code> or use argument <code>which</code>.
</p>


<h3>Value</h3>

<p>None; side effect is an interactive web page.  The temporary
directory containing the files that drive the web page is written to the
console in case you wish to use those files.  This directory is deleted when
you quit R.  If you wish to read the file, don't minify the code, it will be
unreadable.
</p>


<h3>Browser Choice</h3>

<p>The browser is called by
<code><a href="utils.html#topic+browseURL">browseURL</a></code>, which
in turn uses <code>options("browser")</code>.  Exactly how this is handled
is OS dependent.
</p>


<h3>RStudio Viewer</h3>

<p>If browser is <code>NULL</code>, you are using RStudio, and a
viewer is specified, this will be called.  You can stop this by with
<code>options(viewer = NULL)</code>.
</p>


<h3>Browser Choice (Mac)</h3>

<p>On a Mac, the default browser is called
by <code>/bin/sh/open</code>
which in turn looks at which browser you have set in the system settings.  You can
override your default with
<code>browser = "/usr/bin/open -a 'Google Chrome'"</code> for example.
</p>


<h3>Browser Choice &amp; Performance</h3>

<p>You can check the performance of
your browser at peacekeeper.futuremark.com  The most relevant score
is the rendering category.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotSpectra">plotSpectra</a></code> for non-interactive plotting.  Details
about <code>d3.js</code> are at <a href="https://d3js.org">https://d3js.org</a>. Additional documentation at
<a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (interactive()) {
  require("jsonlite")
  require("js")
  data(metMUD2)
  plotSpectraJS(metMUD2)
}
</code></pre>

<hr>
<h2 id='r_pcaSpectra'>Robust PCA of a Spectra Object</h2><span id='topic+r_pcaSpectra'></span>

<h3>Description</h3>

<p>A wrapper which carries out robust PCA analysis on a <code><a href="#topic+Spectra">Spectra</a></code>
object.  The data are row- and column-centered, and the user can select
various options for scaling.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>r_pcaSpectra(spectra, choice = "noscale")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="r_pcaSpectra_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="r_pcaSpectra_+3A_choice">choice</code></td>
<td>
<p>A character vector describing the type of scaling to be
carried out.  One of <code>c("noscale", "mad")</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of classes <code>converted_from_princomp</code> and <code>prcomp</code>.
It includes a list element called <code>$method</code>, a
character string describing the pre-processing carried out and the type of
PCA performed (used to annotate plots).
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>References</h3>

<p>K. Varmuza and P. Filzmoser <em>Introduction to Multivariate
Statistical Analysis in Chemometrics</em>, CRC Press, 2009.
</p>


<h3>See Also</h3>

<p><code><a href="pcaPP.html#topic+PCAgrid">PCAgrid</a></code> for the underlying function,
<code><a href="#topic+c_pcaSpectra">c_pcaSpectra</a></code> for classical PCA calculations,
<code><a href="#topic+s_pcaSpectra">s_pcaSpectra</a></code> for sparse PCA calculations,
<code><a href="#topic+irlba_pcaSpectra">irlba_pcaSpectra</a></code> for PCA via the IRLBA algorithm.
Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>
<p>For displaying the results, <code><a href="ChemoSpecUtils.html#topic+plotScree">ChemoSpecUtils::plotScree()</a></code>, <code><a href="ChemoSpecUtils.html#topic+plotScores">ChemoSpecUtils::plotScores()</a></code>, <code><a href="#topic+plotLoadings">plotLoadings()</a></code>, <code><a href="#topic+plot2Loadings">plot2Loadings()</a></code>, <code><a href="#topic+sPlotSpectra">sPlotSpectra()</a></code>.
</p>
<p><a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
# This example assumes the graphics output is set to ggplot2 (see ?GraphicsOptions).
library("ggplot2")
data(metMUD1)
pca &lt;- r_pcaSpectra(metMUD1)

p1 &lt;- plotScree(pca)
p1

p2 &lt;- plotScores(metMUD1, pca, pcs = c(1, 2), ellipse = "cls", tol = 0.05)
p2 &lt;- p2 + ggtitle("Scores: metMUD1 NMR Data")
p2

p3 &lt;- plotLoadings(metMUD1, pca, loads = 1:2, ref = 1)
p3 &lt;- p3 + ggtitle("Loadings: metMUD1 NMR Data")
p3

## End(Not run)
</code></pre>

<hr>
<h2 id='removeFreq'>Remove Frequencies from a Spectra or Spectra2D Object</h2><span id='topic+removeFreq'></span>

<h3>Description</h3>

<p>This function is used by <code>ChemoSpec</code> and <code>ChemoSpec2D</code>, but is
formally part of <code>ChemoSpecUtils</code>.  You can access
full documentation via <code><a href="ChemoSpecUtils.html#topic+removeFreq">removeFreq</a></code>.
</p>

<hr>
<h2 id='removeGroup'>Remove Groups from a Spectra or Spectra2D Object</h2><span id='topic+removeGroup'></span>

<h3>Description</h3>

<p>This function is used by <code>ChemoSpec</code> and <code>ChemoSpec2D</code>, but is
formally part of <code>ChemoSpecUtils</code>.  You can access
full documentation via <code><a href="ChemoSpecUtils.html#topic+removeGroup">removeGroup</a></code>.
</p>

<hr>
<h2 id='removeSample'>Remove Samples from a Spectra or Spectra2D Object</h2><span id='topic+removeSample'></span>

<h3>Description</h3>

<p>This function is used by <code>ChemoSpec</code> and <code>ChemoSpec2D</code>, but is
formally part of <code>ChemoSpecUtils</code>.  You can access
full documentation via <code><a href="ChemoSpecUtils.html#topic+removeSample">removeSample</a></code>.
</p>

<hr>
<h2 id='reviewAllSpectra'>Review All the Spectra in a Spectra Object</h2><span id='topic+reviewAllSpectra'></span>

<h3>Description</h3>

<p>Utility to review all spectra in a <code>Spectra</code> object.  Output depends upon the graphics
output choice.
</p>

<dl>
<dt>base:</dt><dd><p>Plots each spectrum one at a time, and waits for a return in the console before
plotting the next spectrum.  Use <code>ESC</code> to get out of the loop.</p>
</dd>
<dt>ggplot2:</dt><dd><p>All the spectra are plotted in a single column.</p>
</dd>
</dl>



<h3>Usage</h3>

<pre><code class='language-R'>reviewAllSpectra(spectra, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reviewAllSpectra_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="reviewAllSpectra_+3A_...">...</code></td>
<td>
<p>Parameters to be passed to the plotting routines. <em>Applies to base graphics only</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The returned value depends on the graphics option selected (see <code>GraphicsOptions()</code>).
</p>

<ul>
<li> <p><code>base</code>: None.  Side effect is a plot.
</p>
</li>
<li> <p><code>ggplot2</code>: The plot is displayed, and a <code>ggplot2</code> object is returned if the value is assigned.  The plot can be modified in the usual <code>ggplot2</code> manner.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University), Tejasvi Gupta.
</p>


<h3>See Also</h3>

<p>See <code>GraphicsOptions</code>
for more information about the graphics options. Additional documentation at
<a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Because there are 16 spectra in this data set, you probably want to
# expand the height of the graphics device to see the spectra clearly.
# This example assumes the graphics output is set to ggplot2 or plotly (see ?GraphicsOptions).
# If you do options(ChemoSpecGraphics == "plotly") you'll get the results
# in a web page, which is particularly convenient.
library("ggplot2")
data(metMUD1)
p &lt;- reviewAllSpectra(metMUD1)
p


</code></pre>

<hr>
<h2 id='rowDist'>Compute Distance Between Rows of a Matrix</h2><span id='topic+rowDist'></span>

<h3>Description</h3>

<p>This function is used by <code>ChemoSpec</code> and <code>ChemoSpec2D</code>, but is
formally part of <code>ChemoSpecUtils</code>.  You can access
full documentation via <code><a href="ChemoSpecUtils.html#topic+rowDist">rowDist</a></code>.
</p>

<hr>
<h2 id='s_pcaSpectra'>Sparse PCA of Spectra Objects</h2><span id='topic+s_pcaSpectra'></span>

<h3>Description</h3>

<p>A wrapper which carries out sparse PCA analysis on a
<code><a href="#topic+Spectra">Spectra</a></code> object.  The user can select various options for
scaling.  There is no normalization by rows - do this manually using
<code><a href="#topic+normSpectra">normSpectra</a></code>. The data will be centered, as is required by PCA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>s_pcaSpectra(spectra, choice = "noscale", K = 3, para = rep(0.5, K), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="s_pcaSpectra_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="s_pcaSpectra_+3A_choice">choice</code></td>
<td>
<p>A character string indicating the choice of scaling.  One of
<code>c("noscale"</code>, <code>"autoscale"</code>, <code>"Pareto")</code>. <code>"autoscale"</code>
is called &quot;standard normal variate&quot; or &quot;correlation matrix PCA&quot; in some literature.</p>
</td></tr>
<tr><td><code id="s_pcaSpectra_+3A_k">K</code></td>
<td>
<p>Integer. The number of components desired.</p>
</td></tr>
<tr><td><code id="s_pcaSpectra_+3A_para">para</code></td>
<td>
<p>A vector of <code>length(K)</code> giving the tuning parameters.</p>
</td></tr>
<tr><td><code id="s_pcaSpectra_+3A_...">...</code></td>
<td>
<p>Other parameters to be passed to <code><a href="elasticnet.html#topic+arrayspc">arrayspc</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The scale choice <code>autoscale</code> scales the columns by their standard
deviation.  <code>Pareto</code> scales by the square root of the standard
deviation.
</p>


<h3>Value</h3>

<p>An object of class <code>prcomp</code> and <code>converted_from_arrayspc</code>,
which includes a list
element called <code>$method</code>, a character string describing the
pre-processing carried out and the type of PCA performed (used to annotate
plots).  A check is carried out to see if the computation was successful
and a warning issued if it failed.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>References</h3>

<p>H. Zou, T. Hastie and R. Tibshirani &quot;Sparse Principal Components Analysis&quot;
<em>J. Comp. Stat. Graphics</em> vol. 15 no. 2 pgs. 265-286 (2006).
</p>


<h3>See Also</h3>

<p><code><a href="elasticnet.html#topic+arrayspc">arrayspc</a></code> for the underlying function,
<code><a href="#topic+c_pcaSpectra">c_pcaSpectra</a></code> for classical PCA calculations,
<code><a href="#topic+r_pcaSpectra">r_pcaSpectra</a></code> for robust PCA calculations,
<code><a href="#topic+irlba_pcaSpectra">irlba_pcaSpectra</a></code> for PCA via the IRLBA algorithm.
Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>
<p>For displaying the results, <code><a href="ChemoSpecUtils.html#topic+plotScree">ChemoSpecUtils::plotScree()</a></code>, <code><a href="ChemoSpecUtils.html#topic+plotScores">ChemoSpecUtils::plotScores()</a></code>, <code><a href="#topic+plotLoadings">plotLoadings()</a></code>, <code><a href="#topic+plot2Loadings">plot2Loadings()</a></code>, <code><a href="#topic+sPlotSpectra">sPlotSpectra()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# This example assumes the graphics output is set to ggplot2 (see ?GraphicsOptions).
library("ggplot2")
data(SrE.NMR)
pca &lt;- s_pcaSpectra(SrE.NMR)

p1 &lt;- plotScree(pca)
p1

p2 &lt;- plotScores(SrE.NMR, pca, pcs = c(1, 2), ellipse = "cls", tol = 0.05)
p2 &lt;- p2 + ggtitle("Scores: SrE NMR Data")
p2

p3 &lt;- plotLoadings(SrE.NMR, pca, loads = 1:2, ref = 1)
p3 &lt;- p3 + ggtitle("Loadings: SrE NMR Data")
p3

## End(Not run)

</code></pre>

<hr>
<h2 id='sampleDist'>Compute the Distances Between Samples in a Spectra or Spectra2D Object</h2><span id='topic+sampleDist'></span>

<h3>Description</h3>

<p>This function is used by <code>ChemoSpec</code> and <code>ChemoSpec2D</code>, but is
formally part of <code>ChemoSpecUtils</code>.  You can access
full documentation via <code><a href="ChemoSpecUtils.html#topic+sampleDist">sampleDist</a></code>.
</p>

<hr>
<h2 id='sgfSpectra'>Apply Savitzky-Golay filters to a Spectra object</h2><span id='topic+sgfSpectra'></span>

<h3>Description</h3>

<p>This function is a simple wrapper around the function
<code><a href="signal.html#topic+sgolayfilt">sgolayfilt</a></code>.  It allows one to apply Savitzky-Golay
filters to a <code><a href="#topic+Spectra">Spectra</a></code> object in a convenient way.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sgfSpectra(spectra, m = 0, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sgfSpectra_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="sgfSpectra_+3A_m">m</code></td>
<td>
<p>The desired m-th derivative.  <code>m = 0</code> smooths the data (i.ei a rolling
average), <code>m = 1</code> gives the first derivative etc.</p>
</td></tr>
<tr><td><code id="sgfSpectra_+3A_...">...</code></td>
<td>
<p>Other parameters to be passed to
<code><a href="signal.html#topic+sgolayfilt">sgolayfilt</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A object of class <code><a href="#topic+Spectra">Spectra</a></code>.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>See Also</h3>

<p>Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># You need to install package "signal" for this example
# This example assumes the graphics output is set to ggplot2 (see ?GraphicsOptions).
if (requireNamespace("signal", quietly = TRUE)) {
  library("ggplot2")
  library("patchwork")
  data(SrE.IR)
  myt1 &lt;- expression(bolditalic(Serenoa) ~ bolditalic(repens) ~ bold(IR ~ Spectra))
  myt2 &lt;- expression(bolditalic(Serenoa) ~ bolditalic(repens) ~ bold(IR ~ Spectra ~ (Smoothed)))

  p1 &lt;- plotSpectra(SrE.IR)
  p1 &lt;- p1 + ggtitle(myt1) + coord_cartesian(xlim = c(1900, 2100), ylim = c(0.0, 0.03))

  sgf &lt;- sgfSpectra(SrE.IR)

  p2 &lt;- plotSpectra(sgf)
  p2 &lt;- p2 + ggtitle(myt2) + coord_cartesian(xlim = c(1900, 2100), ylim = c(0.0, 0.03))

  p3 &lt;- p1/p2
  p3
}

</code></pre>

<hr>
<h2 id='Spectra'>Spectra Objects</h2><span id='topic+Spectra'></span>

<h3>Description</h3>

<p>In <code>ChemoSpec</code>, spectral data sets are stored in an S3 class called
<code>Spectra</code>, which contains a variety of information in addition to the
spectra themselves.  <code>Spectra</code> objects are created by
<code><a href="#topic+files2SpectraObject">files2SpectraObject</a></code> or <code><a href="#topic+matrix2SpectraObject">matrix2SpectraObject</a></code>.
</p>


<h3>Structure</h3>

<p>The structure of a <code>Spectra</code> object is a list of 9
elements and an attribute as follows:
</p>

<ul>
<li> <p><strong>$freq</strong> (numeric) A common frequency (or wavelength) axis for all the spectra.
</p>
</li>
<li> <p><strong>$data</strong> (numeric) The intensities for the spectra.  A matrix of dimension no. samples x no. frequency points.
</p>
</li>
<li> <p><strong>$names</strong> (character) The sample names for the spectra; length must be no. samples.
</p>
</li>
<li> <p><strong>$groups</strong> (factor) The group classification of the samples; length must be no. samples.
</p>
</li>
<li> <p><strong>$colors</strong> (character) The colors for each sample; length must be no. samples. Groups and colors correspond.
</p>
</li>
<li> <p><strong>$sym</strong> (integer) As for <code>colors</code>, but symbols for plotting (if b/w is desired).
</p>
</li>
<li> <p><strong>$alt.sym</strong> (character) Lower-case letters as alternate symbols for plotting.
</p>
</li>
<li> <p><strong>$unit</strong> (character) Two entries, the first giving the x axis unit, the second the y axis unit.
</p>
</li>
<li> <p><strong>$desc</strong> (character) A character string describing the data set.  This appears on plots and therefore should probably be kept to 40 characters or less.
</p>
</li>
<li><p> attribute: character with value &quot;Spectra&quot; The S3 class designation.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sumSpectra">sumSpectra</a></code> to summarize a <code>Spectra</code> object.
<code><a href="#topic+sumGroups">sumGroups</a></code> to summarize group membership of a <code>Spectra</code>
object. <code><a href="#topic+chkSpectra">chkSpectra</a></code> to verify the integrity of a
<code>Spectra</code> object. <code><a href="#topic+colorSymbol">colorSymbol</a></code> for a discussion of color
options. Finally, additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>

<hr>
<h2 id='splitSpectraGroups'>Create New Groups from an Existing Spectra Object</h2><span id='topic+splitSpectraGroups'></span>

<h3>Description</h3>

<p>This function takes an existing <code><a href="#topic+Spectra">Spectra</a></code> object and uses your
instructions to split the existing <code>spectra$groups</code> into new groups.
The new groups are added to the existing <code><a href="#topic+Spectra">Spectra</a></code> object (a
list) as new elements.  This allows one to use different combinations of
factors than were originally encoded in the <code><a href="#topic+Spectra">Spectra</a></code> object.
The option also exists to replace the color scheme with one which
corresponds to the new factors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>splitSpectraGroups(spectra, inst = NULL, rep.cols = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="splitSpectraGroups_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="splitSpectraGroups_+3A_inst">inst</code></td>
<td>
<p>A list giving the name of the new element to be created from a
set of target strings given in a character vector.  See the example for the
syntax.</p>
</td></tr>
<tr><td><code id="splitSpectraGroups_+3A_rep.cols">rep.cols</code></td>
<td>
<p>Optional.  A vector giving new colors which correspond to
the levels of <code>inst</code>.  Only possible if <code>inst</code> has only one
element, as the possible combinations of levels and colors may get
complicated.</p>
</td></tr>
<tr><td><code id="splitSpectraGroups_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed downstream. Currently not
used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The items in the character vector are grepped among the existing
<code>spectra$groups</code> entries; when found, they are placed in a new element
of <code><a href="#topic+Spectra">Spectra</a></code>.  In the example, all <code>spectra$groups</code> entries
containing &quot;G&quot; are coded as &quot;G&quot; in a new element called <code>spectra$env</code>,
and any entries containing &quot;T&quot; are handled likewise.  This amounts to a sort
of recoding of factors (the example demonstrates this).  Every entry in
<code>spectra$groups</code> should be matched by one of the entries in the
character vector.  If not, you will get <code>NA</code> entries.  Also, if the targets
in the character vector are not unique, your results will reflect the order
of the levels.  Since this is a grep process, you can pass any valid grep
string as the target.
</p>
<p>If <code>rep.cols</code> is provided, these colors are mapped one for one onto the
levels of the the first element of <code>inst</code>.  This provides a different
means of changing the sample color encoding than <code><a href="#topic+conColScheme">conColScheme</a></code>.
</p>


<h3>Value</h3>

<p>An object of S3 class <code><a href="#topic+Spectra">Spectra</a></code>, modified to have
additional elements as specified by <code>inst</code>.
</p>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+conColScheme">conColScheme</a></code>
Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(metMUD2)
levels(metMUD2$groups) # original factor encoding

# Split those original levels into 2 new ones (re-code them)
new.grps &lt;- list(geneBb = c("B", "b"), geneCc = c("C", "c"))
res &lt;- splitSpectraGroups(metMUD2, new.grps)
str(res) # note two new elements, "geneBb" and "geneCc"
sumSpectra(res) # reports on extra elements

# Note that if you want to use a newly created group in
# plotScores and other functions to drive the color scheme
# and labeling, you'll have to update the groups element:
res$groups &lt;- as.factor(paste(res$geneBb, res$geneCc, sep = ""))

</code></pre>

<hr>
<h2 id='sPlotSpectra'>s-Plot of Spectra Data (Post PCA)</h2><span id='topic+sPlotSpectra'></span>

<h3>Description</h3>

<p>Produces a scatter plot of the correlation of the
variables against their covariance for a chosen principal component.  It
allows visual identification of variables driving the separation and thus is
a useful adjunct to traditional loading plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sPlotSpectra(spectra, pca, pc = 1, tol = 0.05, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sPlotSpectra_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="sPlotSpectra_+3A_pca">pca</code></td>
<td>
<p>The result of a pca calculation on <code><a href="#topic+Spectra">Spectra</a></code> (i.e.
the output from <code><a href="#topic+c_pcaSpectra">c_pcaSpectra</a></code> or <code><a href="#topic+r_pcaSpectra">r_pcaSpectra</a></code>).</p>
</td></tr>
<tr><td><code id="sPlotSpectra_+3A_pc">pc</code></td>
<td>
<p>An integer specifying the desired pc plot.</p>
</td></tr>
<tr><td><code id="sPlotSpectra_+3A_tol">tol</code></td>
<td>
<p>A number describing the fraction of points to be labeled. <code>tol = 1.0</code> labels all the points; <code>tol = 0.05</code> labels <em>approximately</em> the most extreme 5 percent. Set to <code>'none'</code> to completely suppress labels. Note that a simple approach based upon quantiles is used, assumes that both x and y are each normally distributed, and treats x and y separately.  Thus, this is not a formal treatment of outliers, just a means of labeling points. Groups are lumped together for the computation.</p>
</td></tr>
<tr><td><code id="sPlotSpectra_+3A_...">...</code></td>
<td>
<p>Parameters to be passed to the plotting routines. <em>Applies to base graphics only</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The returned value depends on the graphics option selected (see <code>GraphicsOptions()</code>).
</p>

<ul>
<li> <p><code>base</code>: None.  Side effect is a plot.
</p>
</li>
<li> <p><code>ggplot2</code>: The plot is displayed, and a <code>ggplot2</code> object is returned if the value is assigned.  The plot can be modified in the usual <code>ggplot2</code> manner.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University), Tejasvi Gupta, Matthew J. Keinsley.
</p>


<h3>References</h3>

<p>Wiklund, Johansson, Sjostrom, Mellerowicz, Edlund, Shockcor,
Gottfries, Moritz, and Trygg. &quot;Visualization of GC/TOF-MS-Based
Metabololomics Data for Identification of Biochemically Interesting
Compounds Usings OPLS Class Models&quot; Analytical Chemistry Vol.80 no.1 pgs.
115-122 (2008).
</p>


<h3>See Also</h3>

<p>Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# This example assumes the graphics output is set to ggplot2 (see ?GraphicsOptions).
library("ggplot2")
data(SrE.IR)
pca &lt;- c_pcaSpectra(SrE.IR)
myt &lt;- expression(bolditalic(Serenoa) ~ bolditalic(repens) ~ bold(IR ~ Spectra))
p &lt;- sPlotSpectra(spectra = SrE.IR, pca = pca, pc = 1, tol = 0.001)
p &lt;- p + ggtitle(myt)
p

</code></pre>

<hr>
<h2 id='SrE.IR'>IR and NMR Spectra of Serenoa repens (Saw Palmetto) Oil Extracts and
Reference Oils</h2><span id='topic+SrE.IR'></span><span id='topic+SrE.NMR'></span>

<h3>Description</h3>

<p>A collection of 14 IR and NMR spectra of essential oil extracted from the
palm <em>Serenoa repens</em> or Saw Palmetto, which is commonly used to treat
BPH in men.  The 14 spectra are of different retail samples, and are divided
into two categories based upon the label description: adSrE, adulterated
extract, and pSrE, pure extract.  The adulterated samples typically have
olive oil added to them, which is inactive towards BPH.  There are two
additional spectra included as references/outliers: evening primrose oil,
labeled EPO in the data set, and olive oil, labeled OO.  These latter two
oils are mixtures of triglycerides for the most part, while the SrE samples
are largely fatty acids.  As a result, the spectra of these two groups are
subtly different.
</p>


<h3>Format</h3>

<p>The data are stored as a <code><a href="#topic+Spectra">Spectra</a></code> object.
</p>


<h3>Source</h3>

<p>IR data collected in the author's laboratory.  NMR data collected at
Purdue University with the generosity and assistance of Prof. Dan Raftery
and Mr. Tao Ye.
</p>


<h3>See Also</h3>

<p>Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(SrE.IR)
sumSpectra(SrE.IR)
data(SrE.NMR)
sumSpectra(SrE.NMR)
</code></pre>

<hr>
<h2 id='sumGroups'>Summarize the Group Membership of a Spectra or Spectra2D Object</h2><span id='topic+sumGroups'></span>

<h3>Description</h3>

<p>This function is used by <code>ChemoSpec</code> and <code>ChemoSpec2D</code>, but is
formally part of <code>ChemoSpecUtils</code>.  You can access
full documentation via <code><a href="ChemoSpecUtils.html#topic+sumGroups">sumGroups</a></code>.
</p>

<hr>
<h2 id='sumSpectra'>Summarize a Spectra or Spectra2D Object</h2><span id='topic+sumSpectra'></span>

<h3>Description</h3>

<p>This function is used by <code>ChemoSpec</code> and <code>ChemoSpec2D</code>, but is
formally part of <code>ChemoSpecUtils</code>.  You can access
full documentation via <code><a href="ChemoSpecUtils.html#topic+sumSpectra">sumSpectra</a></code>.
</p>

<hr>
<h2 id='surveySpectra'>Plot Measures of Central Tendency and Spread for a Spectra Object</h2><span id='topic+surveySpectra'></span><span id='topic+surveySpectra2'></span>

<h3>Description</h3>

<p>Compute and plot various measures of central tendency and
spread for a <code><a href="#topic+Spectra">Spectra</a></code> object.  Several different measures/spreads
are available.  These are useful as an overview of where a data set varies
the most.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>surveySpectra(
  spectra,
  method = c("sd", "sem", "sem95", "mad", "iqr"),
  by.gr = TRUE,
  ...
)

surveySpectra2(
  spectra,
  method = c("sd", "sem", "sem95", "mad", "iqr"),
  lab.pos = 0.9 * max(spectra$freq),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="surveySpectra_+3A_spectra">spectra</code></td>
<td>
<p>An object of S3 class <code><a href="#topic+Spectra">Spectra()</a></code>.</p>
</td></tr>
<tr><td><code id="surveySpectra_+3A_method">method</code></td>
<td>
<p>Character.  One of <code>c("sd", "sem", "sem95", "mad",
"iqr")</code>.</p>
</td></tr>
<tr><td><code id="surveySpectra_+3A_by.gr">by.gr</code></td>
<td>
<p>Logical, indicating if the analysis is to be done by group or
not. Applies to <code>surveySpectra</code> only.</p>
</td></tr>
<tr><td><code id="surveySpectra_+3A_...">...</code></td>
<td>
<p>Parameters to be passed to the plotting routines. <em>Applies to base graphics only</em>.</p>
</td></tr>
<tr><td><code id="surveySpectra_+3A_lab.pos">lab.pos</code></td>
<td>
<p>Numeric, giving the frequency where the label should be drawn.
Applies to <code>surveySpectra2</code> only.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For <code>surveySpectra</code> the method choice works as follows: <code>sd</code> plots
the mean spectrum +/- the standard deviation, <code>sem</code> plots the mean
spectrum +/- the standard error of the mean, <code>sem95</code> plots the mean
spectrum +/- the standard error at the 95 percent confidence interval,
<code>mad</code> plots the median spectrum +/- the median absolute deviation, and
finally, <code>iqr</code> plots the median spectrum + the upper hinge and - the
lower hinge.
</p>
<p>For <code>surveySpectra2</code>, the spectra are mean centered and plotted.  Below
that, the relative summary statistic is plotted, offset, but on the same
scale.
</p>


<h3>Value</h3>

<p>The returned value depends on the graphics option selected (see <code>GraphicsOptions()</code>).
</p>

<ul>
<li> <p><code>base</code>: None.  Side effect is a plot.
</p>
</li>
<li> <p><code>ggplot2</code>: The plot is displayed, and a <code>ggplot2</code> object is returned if the value is assigned.  The plot can be modified in the usual <code>ggplot2</code> manner.
</p>
</li></ul>



<h3>Functions</h3>


<ul>
<li> <p><code>surveySpectra()</code>: Spectral survey emphasizing mean or median spectrum, optionally by group.
</p>
</li>
<li> <p><code>surveySpectra2()</code>: Spectral survey emphasizing variation among spectra.
</p>
</li></ul>


<h3>Author(s)</h3>

<p>Bryan A. Hanson (DePauw University), Tejasvi Gupta.
</p>


<h3>See Also</h3>

<p>Additional documentation at <a href="https://bryanhanson.github.io/ChemoSpec/">https://bryanhanson.github.io/ChemoSpec/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# This example assumes the graphics output is set to ggplot2 (see ?GraphicsOptions).
library("ggplot2")
data(SrE.IR)
myt &lt;- expression(bolditalic(Serenoa) ~ bolditalic(repens) ~ bold(Extract ~ IR ~ Spectra))

p1 &lt;- surveySpectra(SrE.IR, method = "iqr")
p1 &lt;- p1 + ggtitle(myt)
p1

p2 &lt;- surveySpectra2(SrE.IR, method = "iqr")
p2 &lt;- p2 + ggtitle(myt)
p2

</code></pre>

<hr>
<h2 id='updateGroups'>Update Group Names in a Spectra or Spectra2D Object</h2><span id='topic+updateGroups'></span>

<h3>Description</h3>

<p>This function is used by <code>ChemoSpec</code> and <code>ChemoSpec2D</code>, but is
formally part of <code>ChemoSpecUtils</code>.  You can access
full documentation via <code><a href="ChemoSpecUtils.html#topic+updateGroups">updateGroups</a></code>.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
