<!DOCTYPE html><html><head><title>Help for package chemodiv</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {chemodiv}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#alpinaCompData'><p>Arabis alpina floral scent compounds</p></a></li>
<li><a href='#alpinaCompDis'><p>Arabis alpina floral scent compound dissimilarity matrix</p></a></li>
<li><a href='#alpinaMolNet'><p>Arabis alpina floral scent molecular network</p></a></li>
<li><a href='#alpinaNPCTable'><p>Arabis alpina floral scent NPClassifier table</p></a></li>
<li><a href='#alpinaPopData'><p>Arabis alpina populations</p></a></li>
<li><a href='#alpinaSampData'><p>Arabis alpina floral scent data</p></a></li>
<li><a href='#alpinaSampDis'><p>Arabis alpina floral scent sample dissimilarity matrix</p></a></li>
<li><a href='#calcBetaDiv'><p>Calculate beta diversity</p></a></li>
<li><a href='#calcDiv'><p>Calculate various diversity measures</p></a></li>
<li><a href='#calcDivProf'><p>Calculate a diversity profile</p></a></li>
<li><a href='#chemodiv'><p>chemodiv: A package for analysing phytochemical diversity</p></a></li>
<li><a href='#chemoDivCheck'><p>Check data formatting</p></a></li>
<li><a href='#chemoDivPlot'><p>Plot chemodiversity</p></a></li>
<li><a href='#compDis'><p>Calculate compound dissimilarities</p></a></li>
<li><a href='#minimalCompData'><p>Minimal compound dataset</p></a></li>
<li><a href='#minimalCompDis'><p>Minimal compound dissimilarity matrix</p></a></li>
<li><a href='#minimalMolNet'><p>Minimal molecular network</p></a></li>
<li><a href='#minimalNPCTable'><p>Minimal NPClassifier table</p></a></li>
<li><a href='#minimalSampData'><p>Minimal sample dataset</p></a></li>
<li><a href='#minimalSampDis'><p>Minimal sample dissimilarity matrix</p></a></li>
<li><a href='#molNet'><p>Generate a molecular network with some properties</p></a></li>
<li><a href='#molNetPlot'><p>Plot molecular networks</p></a></li>
<li><a href='#NPCTable'><p>Generate NPClassifier classification</p></a></li>
<li><a href='#quickChemoDiv'><p>Quickly calculate or plot chemodiversity</p></a></li>
<li><a href='#sampDis'><p>Calculate sample dissimilarities</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Analysing Chemodiversity of Phytochemical Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Quantify and visualise various measures of chemical diversity and
    dissimilarity, for phytochemical compounds and other sets of chemical
    composition data. Importantly, these measures can incorporate biosynthetic
    and/or structural properties of the chemical compounds, resulting in a more
    comprehensive quantification of diversity and dissimilarity. For details,
    see Petrén, Köllner and Junker (2023) &lt;<a href="https://doi.org/10.1111%2Fnph.18685">doi:10.1111/nph.18685</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>jsonlite, httr, vegan, webchem, fmcsR, ChemmineR, hillR, ape,
GUniFrac, tidygraph, igraph, ggraph, ggplot2, gridExtra,
ggdendro, tidyr, rlang, curl</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/hpetren/chemodiv">https://github.com/hpetren/chemodiv</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/hpetren/chemodiv/issues">https://github.com/hpetren/chemodiv/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-17 12:51:39 UTC; Hampus</td>
</tr>
<tr>
<td>Author:</td>
<td>Hampus Petrén <a href="https://orcid.org/0000-0001-6490-4517"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Tobias G. Köllner <a href="https://orcid.org/0000-0002-7037-904X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Robert R. Junker <a href="https://orcid.org/0000-0002-7919-9678"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hampus Petrén &lt;hampus.petren@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-17 17:52:33 UTC</td>
</tr>
</table>
<hr>
<h2 id='alpinaCompData'>Arabis alpina floral scent compounds</h2><span id='topic+alpinaCompData'></span>

<h3>Description</h3>

<p>A dataset listing the compounds in <code><a href="#topic+alpinaSampData">alpinaSampData</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alpinaCompData
</code></pre>


<h3>Format</h3>

<p>A data frame with 15 rows and 3 columns. Each row is a compound.
First column is a common name of the compound, second column is the
SMILES (Simplified Molecular-Input Line-Entry System) specification,
third column is the InChIKey (International Chemical Identifier).
</p>


<h3>Source</h3>

<p>Petren H, Torang P, Agren J, Friberg M. 2021.
Evolution of floral scent in relation to self-incompatibility and capacity
for autonomous self-pollination in the perennial herb <em>Arabis alpina</em>.
Annals of Botany 127: 737-747.
</p>

<hr>
<h2 id='alpinaCompDis'>Arabis alpina floral scent compound dissimilarity matrix</h2><span id='topic+alpinaCompDis'></span>

<h3>Description</h3>

<p>A matrix with compound dissimilarities calculated using
<code><a href="#topic+compDis">compDis</a></code> with <code>type = "PubChemFingerprint"</code>,
for the compounds in <code><a href="#topic+alpinaCompData">alpinaCompData</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alpinaCompDis
</code></pre>


<h3>Format</h3>

<p>A 15x15 compound dissimilarity matrix.
</p>

<hr>
<h2 id='alpinaMolNet'>Arabis alpina floral scent molecular network</h2><span id='topic+alpinaMolNet'></span>

<h3>Description</h3>

<p>A molecular network. Generated by the <code><a href="#topic+molNet">molNet</a></code> function
used on the <code><a href="#topic+alpinaCompDis">alpinaCompDis</a></code> dataset, with
<code>cutOff = 0.75</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alpinaMolNet
</code></pre>


<h3>Format</h3>

<p>A <code>tbl_graph</code> object with 15 nodes and 56 edges.
</p>

<hr>
<h2 id='alpinaNPCTable'>Arabis alpina floral scent NPClassifier table</h2><span id='topic+alpinaNPCTable'></span>

<h3>Description</h3>

<p>A table with the NPClassifier pathways, superclasses and classes,
along with the compound names, smiles and inchikey. Generated by
the <code><a href="#topic+NPCTable">NPCTable</a></code> function used on the
<code><a href="#topic+alpinaCompData">alpinaCompData</a></code> dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alpinaNPCTable
</code></pre>


<h3>Format</h3>

<p>A dataframe with 15 compounds and their NPClassifier classifications.
</p>

<hr>
<h2 id='alpinaPopData'>Arabis alpina populations</h2><span id='topic+alpinaPopData'></span>

<h3>Description</h3>

<p>A dataset listing what population each sample
in <code><a href="#topic+alpinaSampData">alpinaSampData</a></code> comes from.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alpinaPopData
</code></pre>


<h3>Format</h3>

<p>A data frame with 87 rows and 1 column. Each row represents
the population each sample in <code><a href="#topic+alpinaSampData">alpinaSampData</a></code> comes from
(It8, S1 or G1).
</p>


<h3>Source</h3>

<p>Petren H, Torang P, Agren J, Friberg M. 2021.
Evolution of floral scent in relation to self-incompatibility and capacity
for autonomous self-pollination in the perennial herb <em>Arabis alpina</em>.
Annals of Botany 127: 737-747.
</p>

<hr>
<h2 id='alpinaSampData'>Arabis alpina floral scent data</h2><span id='topic+alpinaSampData'></span>

<h3>Description</h3>

<p>A dataset with proportional floral scent data from three populations of
the plant Arabis alpina.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alpinaSampData
</code></pre>


<h3>Format</h3>

<p>A data frame with 87 rows and 15 columns. Each row is a sample,
each column is a floral scent compound.
</p>


<h3>Source</h3>

<p>Petren H, Torang P, Agren J, Friberg M. 2021.
Evolution of floral scent in relation to self-incompatibility and capacity
for autonomous self-pollination in the perennial herb <em>Arabis alpina</em>.
Annals of Botany 127: 737-747.
</p>

<hr>
<h2 id='alpinaSampDis'>Arabis alpina floral scent sample dissimilarity matrix</h2><span id='topic+alpinaSampDis'></span>

<h3>Description</h3>

<p>A matrix with sample dissimilarities calculated from
<code><a href="#topic+alpinaSampData">alpinaSampData</a></code> and <code><a href="#topic+alpinaCompDis">alpinaCompDis</a></code>
using <code><a href="#topic+sampDis">sampDis</a></code> with <code>type = "GenUniFrac"</code> and
<code>alpha = 0.5</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alpinaSampDis
</code></pre>


<h3>Format</h3>

<p>A 87x87 sample dissimilarity matrix.
</p>

<hr>
<h2 id='calcBetaDiv'>Calculate beta diversity</h2><span id='topic+calcBetaDiv'></span>

<h3>Description</h3>

<p>Function to calculate beta diversity in the Hill diversity framework.
This can be calculated as Hill beta diversity or
Functional Hill beta diversity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcBetaDiv(sampleData, compDisMat = NULL, type = "HillDiv", q = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calcBetaDiv_+3A_sampledata">sampleData</code></td>
<td>
<p>Data frame with the relative concentration of each
compound (column) in every sample (row).</p>
</td></tr>
<tr><td><code id="calcBetaDiv_+3A_compdismat">compDisMat</code></td>
<td>
<p>Compound dissimilarity matrix, as calculated by
<code><a href="#topic+compDis">compDis</a></code>. Has to be supplied for calculations of
Functional Hill beta diversity.</p>
</td></tr>
<tr><td><code id="calcBetaDiv_+3A_type">type</code></td>
<td>
<p>Type(s) of Hill beta diversity to calculate. <code>"HillDiv"</code>
and/or <code>"FuncHillDiv"</code>.</p>
</td></tr>
<tr><td><code id="calcBetaDiv_+3A_q">q</code></td>
<td>
<p>Diversity order to use for the calculation of beta diversity.
See <code><a href="#topic+calcDiv">calcDiv</a></code> for further details on <em>q</em>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates a single beta diversity value for the supplied
<code>sampleData</code>. This is calculated as <em>beta = gamma / alpha</em>. Gamma
diversity represents the diversity of the pooled data set, alpha diversity
represents the mean diversity across individual samples, and
beta diversity represents turnover or variability among samples.
With <code>type = "HillDiv"</code> and <code>q = 0</code> the calculated beta diversity
is equal to the well-known and most simple measure of beta diversity
introduced by Whittaker 1960, where <em>beta = gamma / alpha</em>, based only
on the number of species (here compounds).
</p>


<h3>Value</h3>

<p>Data frame with type of Hill beta diversity calculated, <em>q</em>, and
values for gamma diversity, mean alpha diversity and beta diversity.
</p>


<h3>References</h3>

<p>Chao A, Chiu C-H, Jost L. 2014. Unifying Species Diversity,
Phylogenetic Diversity, Functional Diversity, and Related Similarity and
Differentiation Measures Through Hill Numbers.
Annual Review of Ecology, Evolution, and Systematics 45: 297-324.
</p>
<p>Jost L. 2007. Partitioning diversity into independent alpha and
beta components. Ecology 88: 2427-2439.
</p>
<p>Whittaker RH. 1960. Vegetation of the Siskiyou Mountains, Oregon
and California. Ecological Monographs 30: 279-338.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(minimalSampData)
data(minimalCompDis)
calcBetaDiv(sampleData = minimalSampData)
calcBetaDiv(sampleData = minimalSampData, compDisMat = minimalCompDis,
type = c("HillDiv", "FuncHillDiv"), q = 2)

data(alpinaSampData)
data(alpinaCompDis)
calcBetaDiv(sampleData = alpinaSampData, compDisMat = alpinaCompDis,
type = "FuncHillDiv")
</code></pre>

<hr>
<h2 id='calcDiv'>Calculate various diversity measures</h2><span id='topic+calcDiv'></span>

<h3>Description</h3>

<p>Function to calculate different common measures of diversity, and
components (richness, evenness, disparity) thereof. Types of measures that
can be calculated includes Hill diversity, Functional Hill diversity,
Mean Pairwise Dissimilarity (MPD), Shannon's diversity, Simpson diversity,
Rao's Q, Pielou's evenness and Hill evenness.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcDiv(sampleData, compDisMat = NULL, type = "HillDiv", q = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calcDiv_+3A_sampledata">sampleData</code></td>
<td>
<p>Data frame with the relative concentration of each
compound (column) in every sample (row).</p>
</td></tr>
<tr><td><code id="calcDiv_+3A_compdismat">compDisMat</code></td>
<td>
<p>Compound dissimilarity matrix, as calculated by
<code><a href="#topic+compDis">compDis</a></code>. Has to be supplied for calculations of
Functional Hill Diversity and Rao's Q.</p>
</td></tr>
<tr><td><code id="calcDiv_+3A_type">type</code></td>
<td>
<p>Type(s) of diversity or component(s) thereof to calculate.
Any of <code>"Shannon", "Simpson", "HillDiv", "FuncHillDiv", "MPD",
"RaoQ", "PielouEven", "HillEven"</code>.</p>
</td></tr>
<tr><td><code id="calcDiv_+3A_q">q</code></td>
<td>
<p>Diversity order to use for Hill diversity, Functional Hill
Diversity and Hill Evenness. <em>q</em> should be equal to or larger than zero.
This parameter determines the sensitivity of the (Functional) Hill Diversity
measure to the relative frequencies of compounds. Commonly set to 0, 1 or 2,
although any value &gt; 0 may be used. For <code>q = 0</code> compound proportions
are not taken into account. For <code>q = 1</code> (default) compounds are
weighed according to their proportion in the sample. For <code>q = 2</code>,
more weight is put on compounds with high proportions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates diversity and/or components thereof for each
sample in <code>sampleData</code>. It can calculate the following indices:
</p>

<ul>
<li> <p><code>Shannon</code>. Shannon's Diversity.
</p>
</li>
<li> <p><code>Simpson</code>. Simpson Diversity, often referred to as
the Inverse Simpson Index.
</p>
</li>
<li> <p><code>HillDiv</code>. Hill Diversity. Equation 4a/4b in Chao et al. 2014.
Also referred to as the Hill number or the effective number of
species (here compounds). The parameter <em>q</em> determines the sensitivity
of the measure to the relative frequencies of
compounds (see above for details). For <code>q = 0</code>, this equals the
number of compounds in a sample, i.e. the richness. For <code>q = 1</code>,
this equals the exponential of Shannon's Diversity. For <code>q = 2</code>,
this equals the Simpson Diversity.
</p>
</li>
<li> <p><code>FuncHillDiv</code>. Functional Hill Diversity. There are
different versions of this. Here, the most common version is calculated.
This is the FD(Q), called &quot;total functional diversity&quot;, calculated in
equation 4b/6b in Chiu &amp; Chao 2014. This measure quantifies the
effective total dissimilarity between compounds in the sample.
Calculation of Functional Hill Diversity requires a compound
dissimilarity matrix. The parameter <em>q</em> determines the sensitivity of
the measure to the relative frequencies of compounds
(see above for details). For <code>q = 1</code>, this is a measure
sensitive to compound richness, evenness and dissimilarity, and is therefore
the most comprehensive measure of diversity. For <code>q = 0</code>, this is
equal to Functional Attribute Diversity (FAD) which is the sum of all
dissimilarities in the dissimilarity matrix.
</p>
</li>
<li> <p><code>MPD</code>. Mean Pairwise Dissimilarity. As the name suggests, this
is equal to the mean of the pairwise dissimilarities in the compound
dissimilarity  matrix (excluding the 0 values in the diagonal). Therefore,
in contrast to FAD (see above) this measure is not dependent on the
number of compounds, and hence represents the disparity component
of diversity. Practically, MPD is calculated as Functional Hill Diversity
at <code>q = 0</code>, divided by <em>n(n-1)</em>, where <em>n</em> is the number of compounds,
i.e. Hill Diversity at <code>q = 0</code>.
</p>
</li>
<li> <p><code>RaoQ</code>. Rao's quadratic entropy index Q.
The perhaps most common measure of functional diversity.
Requires a compound dissimilarity matrix. Rao's Q represents the
average dissimilarity of two randomly selected (weighed by
their proportions) compounds in the sample.
</p>
</li>
<li> <p><code>PielouEven</code>. Pielou's Evenness, also referred to as
Shannon's equitability. This is perhaps the most common evenness
measure. Equal to the Shannon's Diversity divided by the natural
logarithm of the number of compounds. In other words, it expresses evenness
with the observed Shannon's diversity as a proportion of the maximum
Shannon's diversity where all compounds are equally abundant. Therefore,
this is a relative measure with a minimum value of 0 and a maximum value
of 1. This measure of evenness is not replication invariant.
</p>
</li>
<li> <p><code>HillEven</code>. Hill Evenness, as defined by equation 8 in
Tuomisto 2012. This is equal to the Hill Diversity, at a given value
of <em>q</em>, divided by the number of compounds, and therefore
has a minimum value of 1 / number of compounds and maximum value of 1.
This measure of evenness is replication invariant. This measure can be
normalized to range from 0 to 1 (equation 13 in Tuomisto 2012).
</p>
</li></ul>



<h3>Value</h3>

<p>Data frame with calculated diversity values for each sample.
</p>


<h3>References</h3>

<p>Chao A, Chiu C-H, Jost L. 2014. Unifying Species Diversity,
Phylogenetic Diversity, Functional Diversity, and Related Similarity and
Differentiation Measures Through Hill Numbers.
Annual Review of Ecology, Evolution, and Systematics 45: 297-324.
</p>
<p>Chiu C-H, Chao A. 2014. Distance-Based Functional Diversity Measures
and Their Decomposition: A Framework Based on Hill Numbers.
PLoS ONE 9: e100014.
</p>
<p>Hill MO. 1973. Diversity and Evenness: A Unifying Notation
and Its Consequences. Ecology 54: 427-432.
</p>
<p>Petren H, Koellner TG, Junker RR. 2023. Quantifying chemodiversity
considering biochemical and structural properties of compounds with the
R package <em>chemodiv</em>. New Phytologist 237: 2478-2492.
</p>
<p>Petren H, Anaia RA, Aragam KS, Braeutigam A, Eckert S, Heinen R,
Jakobs R, Ojeda L, Popp M, Sasidharan R, Schnitzler J-P, Steppuhn A,
Thon F, Tschikin S, Unsicker SB, van Dam NM, Weisser WW, Wittmann MJ,
Yepes S, Ziaja D, Meuller C, Junker RR. 2023. Understanding the
phytochemical diversity of plants: Quantification, variation and
ecological function. bioRxiv doi: 10.1101/2023.03.23.533415.
</p>
<p>Tuomisto H. 2012. An updated consumer's guide to evenness
and related indices. Oikos 121: 1203-1218
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(minimalSampData)
data(minimalCompDis)
calcDiv(sampleData = minimalSampData)
calcDiv(sampleData = minimalSampData, type = c("HillDiv", "HillEven"))
calcDiv(sampleData = minimalSampData, compDisMat = minimalCompDis,
type = "FuncHillDiv", q = 2)

data(alpinaSampData)
data(alpinaCompDis)
calcDiv(sampleData = alpinaSampData, compDisMat = alpinaCompDis,
type = "FuncHillDiv")
</code></pre>

<hr>
<h2 id='calcDivProf'>Calculate a diversity profile</h2><span id='topic+calcDivProf'></span>

<h3>Description</h3>

<p>Function to calculate a diversity profile, i.e. calculate Hill diversity
or Functional Hill Diversity for a range of <em>q</em> values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcDivProf(
  sampleData,
  compDisMat = NULL,
  type = "HillDiv",
  qMin = 0,
  qMax = 3,
  step = 0.1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calcDivProf_+3A_sampledata">sampleData</code></td>
<td>
<p>Data frame with the relative concentration of each
compound (column) in every sample (row).</p>
</td></tr>
<tr><td><code id="calcDivProf_+3A_compdismat">compDisMat</code></td>
<td>
<p>Compound distance matrix, as calculated by
<code><a href="#topic+compDis">compDis</a></code>. Has to be supplied for calculations of
Functional Hill diversity.</p>
</td></tr>
<tr><td><code id="calcDivProf_+3A_type">type</code></td>
<td>
<p>Type of Hill Diversity to calculate for the diversity profile.
<code>"HillDiv"</code> or <code>"FuncHillDiv"</code>.</p>
</td></tr>
<tr><td><code id="calcDivProf_+3A_qmin">qMin</code></td>
<td>
<p>Minimum value of <em>q</em>.</p>
</td></tr>
<tr><td><code id="calcDivProf_+3A_qmax">qMax</code></td>
<td>
<p>Maximum value of <em>q</em>.</p>
</td></tr>
<tr><td><code id="calcDivProf_+3A_step">step</code></td>
<td>
<p>Increment by which <em>q</em> will be calculated between <code>qMin</code>
and <code>qMax</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates a diversity profile for each sample
in <code>sampleData</code>. A diversity profile is a calculation of
Hill Diversity or Functional Hill Diversity for a range of
different values of <em>q</em>. This function performs the calculations,
while <code><a href="#topic+chemoDivPlot">chemoDivPlot</a></code> can be used to conveniently
create the diversity profile plot, where Hill Diversity is
plotted as a function of <em>q</em> within the chosen range.
The shape of the diversity profile curve reflects the evenness
of compound proportions in the sample. For a perfectly even sample
the curve is flat. The more uneven the compound proportions are,
the more steep is the decline of the curve. A common range,
used as default, of <em>q</em> values is between <code>qMin = 0</code> and
<code>qMax = 3</code>, as diversity should change little beyond <code>qMax = 3</code>.
See <code><a href="#topic+calcDiv">calcDiv</a></code> for further details on <em>q</em>.
</p>


<h3>Value</h3>

<p>List with a diversity profile data frame with samples as rows
and the Hill diversity or Functional Hill diversity for different <em>q</em>
values as columns; and values for <code>type</code>, <code>qMin</code>,
<code>qMax</code> and <code>step</code>.
</p>


<h3>References</h3>

<p>Chao A, Chiu C-H, Jost L. 2014. Unifying Species Diversity,
Phylogenetic Diversity, Functional Diversity, and Related Similarity and
Differentiation Measures Through Hill Numbers.
Annual Review of Ecology, Evolution, and Systematics 45: 297-324.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(minimalSampData)
data(minimalCompDis)
calcDivProf(sampleData = minimalSampData)
calcDivProf(sampleData = minimalSampData, compDisMat = minimalCompDis,
type = "FuncHillDiv")

data(alpinaCompData)
data(alpinaCompDis)
calcDivProf(sampleData = alpinaSampData, compDisMat = alpinaCompDis,
type = "FuncHillDiv", qMin = 1, qMax = 2, step = 0.2)
</code></pre>

<hr>
<h2 id='chemodiv'>chemodiv: A package for analysing phytochemical diversity</h2><span id='topic+chemodiv'></span>

<h3>Description</h3>

<p><em>chemodiv</em> is an R package for analysing the chemodiversity of
phytochemical data. The package includes a number of functions that enables
quantification and visualization of phytochemical diversity and
dissimilarity for any type of phytochemical (and similar) samples, such as
herbivore defence compounds, volatiles and similar. Importantly,
calculations of diversity and dissimilarity can incorporate biosynthetic
and/or structural properties of the phytochemical compounds, resulting
in more comprehensive quantifications of diversity and dissimilarity.
Functions in the R-package will work best for sets of data, commonly
generated by chemical ecologists using GC-MS, LC-MS or similar, where all
or most compounds in the samples have been confidently identified.
See Petren et al. 2023a for a detailed description of the package,
and Petren et al. 2023b for a more in-depth discussion and review
of plant chemodiversity.
</p>


<h3>Details</h3>

<p>Two datasets are needed to use the full set of analyses
included in the package.
</p>
<p>The first dataset should contain data on the relative
abundance/concentration (i.e. proportion) of different compounds (columns)
in different samples (rows). See the included
dataset <code><a href="#topic+minimalSampData">minimalSampData</a></code> for a basic example.
Note that all calculations of diversity, and most calculations of
dissimilarity, are only performed on relative, rather than absolute, values.
</p>
<p>The second dataset should contain, in each of three columns in a data frame,
the compound name, SMILES and InChIKey IDs of all the compounds
present in the first dataset. See the included dataset
<code><a href="#topic+minimalCompData">minimalCompData</a></code> for a basic example. SMILES and InChIKey
are chemical identifiers that are easily obtained for each compound
by searching for it in PubChem <a href="https://pubchem.ncbi.nlm.nih.gov/">https://pubchem.ncbi.nlm.nih.gov/</a>.
Here, a search with a common name will bring up the compound's
record in the database, where the (isomeric/canonical) SMILES and
InChIKey are included. Various automated tools such as
the PubChem Identifier Exchange Service
<a href="https://pubchem.ncbi.nlm.nih.gov/idexchange/idexchange.cgi">https://pubchem.ncbi.nlm.nih.gov/idexchange/idexchange.cgi</a> or
The Chemical Translation Service <a href="https://cts.fiehnlab.ucdavis.edu/">https://cts.fiehnlab.ucdavis.edu/</a>
can also be used. The user is intentionally required to compile the
chemical identifiers manually to ensure these are correct,
as lists of compounds very often contain compounds wrongly named,
wrongly formatted, under various synonyms etc. which prevents easy
automatic translation of compound names to SMILES and InChIKey.
Note that SMILES IDs might contain the character combination <code>"\C"</code>.
If SMILES are entered manually directly in R, this is interpreted as an
unrecognized escape and results in an error. In this case, an extra
backslash has to be added: <code>"\\C"</code>. If the dataset is instead
imported into R as a csv-file or txt-file (recommended), this is done
automatically and no manual edits has to be done.
</p>
<p>The second dataset with the chemical IDs is primarily used to construct
one or more dissimilarity matrices with pairwise dissimilarities between
chemical compounds, which can then be used in calculations of phytochemical
diversity and dissimilarity. As noted above, to do this, the compounds
in the samples have to be identified and their chemical IDs listed.
If some compounds in a dataset are unknown, these can be handled in
different ways decided by the user, see <code><a href="#topic+compDis">compDis</a></code> for details.
If many or all compounds are unknown, as is common for more metabolomic
type datasets, phytochemical diversity and dissimilarity can still be
calculated using indices that do not consider compound dissimilarities.
Alternatively, other ways to calculate compound dissimilarities,
not based on knowing compound identities, can be used.
For example, cosine dissimilarities between tandem (MS/MS) mass spectra of
metabolomic features can be calculated in the GNPS
framework <a href="https://gnps.ucsd.edu">https://gnps.ucsd.edu</a> (Wang et al. 2016).
A dissimilarity matrix of such dissimilarities can then be used
for the <code>compDisMat</code> argument in various functions in the package,
thereby enabling comprehensive quantification of phytochemical diversity
and dissimilarity also for datasets consisting of unidentified compounds.
</p>
<p>Once the dataset with samples and the dataset with compounds are prepared,
these should be imported/constructed as separate data frames in R,
and all analyses in the package can then be performed, in largely the
same order as they appear in the list below.
</p>


<h3>Data format checks</h3>

<p><code><a href="#topic+chemoDivCheck">chemoDivCheck</a></code>
</p>


<h3>Compound classification and dissimilarity</h3>

<p><code><a href="#topic+NPCTable">NPCTable</a></code>
<code><a href="#topic+compDis">compDis</a></code>
</p>


<h3>Diversity calculations</h3>

<p><code><a href="#topic+calcDiv">calcDiv</a></code>
<code><a href="#topic+calcBetaDiv">calcBetaDiv</a></code>
<code><a href="#topic+calcDivProf">calcDivProf</a></code>
</p>


<h3>Sample dissimilarities</h3>

<p><code><a href="#topic+sampDis">sampDis</a></code>
</p>


<h3>Molecular network and properties</h3>

<p><code><a href="#topic+molNet">molNet</a></code>
</p>


<h3>Chemodiversity and network plots</h3>

<p><code><a href="#topic+molNetPlot">molNetPlot</a></code>
<code><a href="#topic+chemoDivPlot">chemoDivPlot</a></code>
</p>


<h3>Shortcut function</h3>

<p><code><a href="#topic+quickChemoDiv">quickChemoDiv</a></code>
</p>


<h3>Author(s)</h3>

<p>Hampus Petren, Tobias G. Koellner, Robert R. Junker
</p>


<h3>References</h3>

<p>Petren H, Koellner TG, Junker RR. 2023a. Quantifying chemodiversity
considering biochemical and structural properties of compounds with the
R package <em>chemodiv</em>. New Phytologist 237: 2478-2492.
</p>
<p>Petren H, Anaia RA, Aragam KS, Braeutigam A, Eckert S, Heinen R,
Jakobs R, Ojeda L, Popp M, Sasidharan R, Schnitzler J-P, Steppuhn A,
Thon F, Tschikin S, Unsicker SB, van Dam NM, Weisser WW, Wittmann MJ,
Yepes S, Ziaja D, Meuller C, Junker RR. 2023b. Understanding the
phytochemical diversity of plants: Quantification, variation and
ecological function. bioRxiv doi: 10.1101/2023.03.23.533415.
</p>
<p>Wang M, Carver JJ, Phelan VV, et al. 2016. Sharing and community
curation of mass spectrometry data with Global Natural Products
Social Molecular Networking. Nature Biotechnology 34: 828-837.
</p>


<h3>See Also</h3>

<p><a href="https://github.com/hpetren/chemodiv">https://github.com/hpetren/chemodiv</a>
</p>

<hr>
<h2 id='chemoDivCheck'>Check data formatting</h2><span id='topic+chemoDivCheck'></span>

<h3>Description</h3>

<p>Function to check that the datasets used by other functions in
the <em>chemodiv</em> package are correctly formatted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chemoDivCheck(sampleData, compoundData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chemoDivCheck_+3A_sampledata">sampleData</code></td>
<td>
<p>Data frame with the relative concentration of each
compound (column) in every sample (row).</p>
</td></tr>
<tr><td><code id="chemoDivCheck_+3A_compounddata">compoundData</code></td>
<td>
<p>Data frame with the compounds in <code>sampleData</code>
as rows. Should have a column named &quot;compound&quot; with common names of
the compounds, a column named &quot;smiles&quot; with SMILES IDs of the compounds,
and a column named &quot;inchikey&quot; with the InChIKey IDs for the compounds.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function performs a number of checks on the two main datasets used
as input data, to make sure datasets are formatted in a way suitable
for the other functions in the package. This should make it easier for
users to correctly construct datasets before starting with analyses.
</p>
<p>Two datasets are needed to use the full set of analyses included in
the package, and these can be checked for formatting issues.
The first dataset should contain data on the proportions
of different compounds (columns) in different samples (rows).
Note that all calculations of diversity, and most calculations of
dissimilarity, are only performed on relative, rather than absolute,
values. The second dataset should contain, in each of three
columns in a data frame, the compound name, SMILES and InChIKey IDs of
all the compounds present in the first dataset. See
<code><a href="#topic+chemodiv">chemodiv</a></code> for details on obtaining SMILES and InChIKey IDs.
Avoid including Greek letters or other special characters in the
compound names.
</p>


<h3>Value</h3>

<p>One or several messages pointing out problems with data formatting,
or a message informing that the datasets appear to be correctly formatted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(minimalSampData)
data(minimalCompData)
chemoDivCheck(minimalSampData, minimalCompData) # Correct format
chemoDivCheck(minimalSampData, minimalCompData[c(2,3,1),]) # Incorrect format

data(alpinaSampData)
data(alpinaCompData)
chemoDivCheck(sampleData = alpinaSampData, compoundData = alpinaCompData)
</code></pre>

<hr>
<h2 id='chemoDivPlot'>Plot chemodiversity</h2><span id='topic+chemoDivPlot'></span>

<h3>Description</h3>

<p>Function to conveniently create basic plots of the different types of
chemodiversity measurements calculated by functions in the package.
This function exists to provide an easy way to make basic chemodiversity
plots. As functions in the package output data in standard formats,
customized plots are easily created with <code><a href="ggplot2.html#topic+ggplot2">ggplot2</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chemoDivPlot(
  compDisMat = NULL,
  divData = NULL,
  divProfData = NULL,
  sampDisMat = NULL,
  groupData = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chemoDivPlot_+3A_compdismat">compDisMat</code></td>
<td>
<p>Compound dissimilarity matrix, generated by
the <code><a href="#topic+compDis">compDis</a></code> function. Note that only a single matrix
should be supplied, and not the whole list.</p>
</td></tr>
<tr><td><code id="chemoDivPlot_+3A_divdata">divData</code></td>
<td>
<p>Diversity/evenness data frame,
generated by the <code><a href="#topic+calcDiv">calcDiv</a></code> function. This data frame can
contain a single or multiple columns with diversity/evenness measures.</p>
</td></tr>
<tr><td><code id="chemoDivPlot_+3A_divprofdata">divProfData</code></td>
<td>
<p>Diversity profile, generated by
the <code><a href="#topic+calcDivProf">calcDivProf</a></code> function. Note that the whole list
outputted by the <code><a href="#topic+calcDivProf">calcDivProf</a></code> function should be supplied.</p>
</td></tr>
<tr><td><code id="chemoDivPlot_+3A_sampdismat">sampDisMat</code></td>
<td>
<p>Sample dissimilarity matrix, generated by
the <code><a href="#topic+sampDis">sampDis</a></code> function. This can be either the list of one or
both matrices outputted by the function, or a single matrix directly.</p>
</td></tr>
<tr><td><code id="chemoDivPlot_+3A_groupdata">groupData</code></td>
<td>
<p>Grouping data. Should be either a vector or a data frame
with a single column.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function can create four different types of plots,
(using <code><a href="ggplot2.html#topic+ggplot2">ggplot2</a></code>) depending on which input data
is supplied:
</p>

<ul>
<li><p> Function argument <code>compDisMat</code>. A compound dissimilarity matrix
will be plotted as a dendrogram visualizing how
structurally/biosynthetically similar different compounds are to each other.
</p>
</li>
<li><p> Function argument <code>divData</code>. Diversity/evenness values will
be plotted as a boxplot.
</p>
</li>
<li><p> Function argument <code>divProfData</code>. A diversity profile, plotting
(Functional) Hill diversity at different values of q will be plotted
as a line plot.
</p>
</li>
<li><p> Function argument <code>sampDisMat</code>. A sample dissimilarity matrix
will be plotted as an NMDS plot.
</p>
</li>
<li><p> Function argument <code>groupData</code>. Grouping data (e.g. population,
species etc.) may be supplied, to plot each group in different
boxes/lines/colours.
</p>
</li></ul>

<p>Note that this function can take any combination of the four arguments
as input, and argument names should always be specified to ensure
each dataset is correctly plotted. If including the function
argument <code>sampDisMat</code>, a Nonmetric Multidimensional Scaling (NMDS)
will be performed, which may take time for larger datasets.
</p>


<h3>Value</h3>

<p>The specified chemodiversity plots.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>minimalDiv &lt;- calcDiv(minimalSampData, minimalCompDis, type = "FuncHillDiv")
groups &lt;- c("A", "A", "B", "B")
chemoDivPlot(divData = minimalDiv, groupData = groups)

data(alpinaCompDis)
data(alpinaSampDis)
data(alpinaPopData)
alpinaDiv &lt;- calcDiv(sampleData = alpinaSampData, compDisMat = alpinaCompDis,
type = "FuncHillDiv")
alpinaDivProf &lt;- calcDivProf(sampleData = alpinaSampData,
compDisMat = alpinaCompDis, type = "FuncHillDiv",
qMin = 0, qMax = 2, step = 0.2)
chemoDivPlot(compDisMat = alpinaCompDis, divData = alpinaDiv,
divProfData = alpinaDivProf, sampDisMat = alpinaSampDis,
groupData = alpinaPopData)
</code></pre>

<hr>
<h2 id='compDis'>Calculate compound dissimilarities</h2><span id='topic+compDis'></span>

<h3>Description</h3>

<p>Function to quantify dissimilarities between phytochemical compounds.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compDis(
  compoundData,
  type = "PubChemFingerprint",
  npcTable = NULL,
  unknownCompoundsMean = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compDis_+3A_compounddata">compoundData</code></td>
<td>
<p>Data frame with the chemical compounds of interest,
usually the compounds found in the sample dataset.
Should have a column named &quot;compound&quot; with common names of
the compounds, a column named &quot;smiles&quot; with SMILES IDs of the compounds,
and a column named &quot;inchikey&quot; with the InChIKey IDs for the compounds.</p>
</td></tr>
<tr><td><code id="compDis_+3A_type">type</code></td>
<td>
<p>Type of data compound dissimilarity calculations will be
based on: <code>NPClassifier</code>, <code>PubChemFingerprint</code> or <code>fMCS</code>.
If more than one is chosen, a matrix with mean values of the other
matrices will also calculated.</p>
</td></tr>
<tr><td><code id="compDis_+3A_npctable">npcTable</code></td>
<td>
<p>A data frame already generated by <code><a href="#topic+NPCTable">NPCTable</a></code>
can optionally be supplied, if compound dissimilarities are to be
calculated using <code>type = "NPClassifier"</code>.</p>
</td></tr>
<tr><td><code id="compDis_+3A_unknowncompoundsmean">unknownCompoundsMean</code></td>
<td>
<p>If unknown compounds, i.e. ones without SMILES
or InChIKey, should be given mean dissimilarity values. If not, these
will have dissimilarity 1 to all other compounds.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates matrices with pairwise dissimilarities between
the chemical compounds in <code>compoundData</code>, to quantify how
different the molecules are to each other. It does so in three
different ways, based on the biosynthetic classification or
molecular structure of the molecules:
</p>

<ol>
<li><p> Using the classification from the <em>NPClassifier</em> tool,
<code>type = "NPClassifier"</code>. <em>NPClassifier</em> (Kim et al. 2021) is a
deep-learning tool that automatically classifies natural products
(i.e. phytochemical compounds) into a hierarchical classification of
three levels: pathway, superclass and class. This classification largely
corresponds to the biosynthetic groups/pathways the compounds
are produced in. Classifications are downloaded from
<a href="https://npclassifier.ucsd.edu/">https://npclassifier.ucsd.edu/</a>. <em>NPClassifier</em> does not always
manage to classify every compound into all three hierarchical levels. In
such cases, it might be beneficial to first run <code><a href="#topic+NPCTable">NPCTable</a></code>,
manually edit the resulting data frame with probable classifications if
possible (with help from the Supporting Information in Kim et al. 2021),
and then supply this classification to the <code>compDis</code> function
with the <code>npcTable</code> argument. This will ensure that compound
dissimilarities are computed optimally.
</p>
</li>
<li><p> Using PubChem Fingerprints, <code>type = "PubChemFingerprint"</code>.
This is a binary substructure fingerprint with 881 binary
variables describing the chemical structure of a compound.
With this method, compounds are therefore compared
based on how structurally dissimilar the molecules are.
See <a href="https://pubchem.ncbi.nlm.nih.gov/docs/data-specification">https://pubchem.ncbi.nlm.nih.gov/docs/data-specification</a>
for more information. (There are many other types of fingerprints,
and ways of calculating compound dissimilarities based on them, see
e.g. packages <code>fingerprint</code> and <code>rcdk</code>). Fingerprint data for
molecules is downloaded from PubChem. In association with this,
there might be a Warning message about closing unused connections,
which is not important.
</p>
</li>
<li><p> fMCS, flexible Maximum Common Substructure,
<code>type = "fMCS"</code>. This is a pairwise graph matching concept.
The fMCS of two compounds is the largest substructure that occurs in both
compounds allowing for atom and/or bond mismatches (Wang et al 2013).
As with the fingerprints, compounds are compared based on how
structurally dissimilar the molecules are. While potentially a very
accurate similarity measure, fMCS is much more computationally demanding
than the other methods, and will take a significant amount of time for
larger data sets. Data on molecules is downloaded from PubChem.
In association with this, there might be a Warning message about closing
unused connections, which is not important.
</p>
</li></ol>

<p>Dissimilarities using NPClassifier and PubChem Fingerprints
are generated by calculating Jaccard (Tanimoto) dissimilarities from a
0/1 table with compounds as rows and group (NPClassifier) or binary
fingerprint variable (PubChem Fingerprints) as columns. fMCS generates
dissimilarity values by calculating Jaccard dissimilarities based on the
number of atoms in the maximum common substructure, allowing for one
atom and one bond mismatch. Dissimilarities are outputted as
dissimilarity matrices.
</p>
<p>If dissimilarities are calculated with more than one method,
the function will output additional dissimilarity matrices.
This always includes a matrix with the mean dissimilarity values of the
selected methods. If <code>"NPClassifier"</code> is included in <code>type</code>,
a matrix of &quot;mix&quot; values is also calculated. The values in this matrix
are the dissimilarities from NPClassifier when these are &gt; 0.
For pairs of compounds where dissimilarities from NPClassifier
equals 0 (i.e. when the compounds belong to the same pathway, superclass
and class), values are equal to half of the (mean) value(s) of the
structural dissimilarity/-ies from PubChem Fingerprints and/or fMCS.
With this method, compound dissimilarities are primarily based on
NPClassifier, but instead of compounds with identical classification having
0 dissimilarity, these have a dissimilarity based on PubChem Fingerprints
and/or fMCS, scaled to always be less (&lt; 0.5) than compounds being in the
same pathway and superclass, but different class.
</p>
<p>If there are unknown compounds, which do not have a
corresponding SMILES or InChIKey, this can be handled in three
different ways. First, these can be completely removed from the list
of compounds and the sample data set, and hence excluded from all analyses.
Second, if <code>unknownCompoundsMean = FALSE</code>, unknown compounds will
be given a dissimilarity value of 1 to all other compounds. Third, if
<code>unknownCompoundsMean = TRUE</code>, unknown compounds will be given
a dissimilarity value to all other compounds which equals the mean
dissimilarity value between all known compounds. See <code><a href="#topic+chemodiv">chemodiv</a></code>
for alternative methods that can be used when most or all compounds
are unknown.
</p>


<h3>Value</h3>

<p>List with compound dissimilarity matrices. A list is always
outputted, even if only one matrix is calculated. Downstream functions,
including <code><a href="#topic+calcDiv">calcDiv</a></code>, <code><a href="#topic+calcBetaDiv">calcBetaDiv</a></code>,
<code><a href="#topic+calcDivProf">calcDivProf</a></code>, <code><a href="#topic+sampDis">sampDis</a></code>, <code><a href="#topic+molNet">molNet</a></code>
and <code><a href="#topic+chemoDivPlot">chemoDivPlot</a></code> require only the matrix as
input (e.g. as <code>fullList$specificMatrix</code>) rather than the whole list.
</p>


<h3>References</h3>

<p>Kim HW, Wang M, Leber CA, Nothias L-F, Reher R, Kang KB,
van der Hooft JJJ, Dorrestein PC, Gerwick WH, Cottrell GW. 2021.
NPClassifier: A Deep Neural Network-Based Structural Classification
Tool for Natural Products. Journal of Natural Products 84: 2795-2807.
</p>
<p>Wang Y, Backman TWH, Horan K, Girke T. 2013.
fmcsR: mismatch tolerant maximum common substructure searching in R.
Bioinformatics 29: 2792-2794.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(minimalCompData)
data(minimalNPCTable)
compDis(minimalCompData, type = "NPClassifier",
npcTable = minimalNPCTable) # Dissimilarity based on NPClassifier

## Not run: compDis(minimalCompData) # Dissimilarity based on Fingerprints

data(alpinaCompData)
data(alpinaNPCTable)
compDis(compoundData = alpinaCompData, type = "NPClassifier",
npcTable = alpinaNPCTable) # Dissimilarity based on NPClassifier
</code></pre>

<hr>
<h2 id='minimalCompData'>Minimal compound dataset</h2><span id='topic+minimalCompData'></span>

<h3>Description</h3>

<p>A small dataset with three phytochemical compounds.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>minimalCompData
</code></pre>


<h3>Format</h3>

<p>A data frame with 3 rows and 3 columns. Each row is a phytochemical
compound. First column is a common name of the compound, second column is
the SMILES (Simplified Molecular-Input Line-Entry System) specification,
third column is the InChIKey (International Chemical Identifier).
</p>

<hr>
<h2 id='minimalCompDis'>Minimal compound dissimilarity matrix</h2><span id='topic+minimalCompDis'></span>

<h3>Description</h3>

<p>A matrix with compound dissimilarities calculated using
<code><a href="#topic+compDis">compDis</a></code> with <code>type = "PubChemFingerprint"</code>,
for the compounds in <code><a href="#topic+minimalCompData">minimalCompData</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>minimalCompDis
</code></pre>


<h3>Format</h3>

<p>A 3x3 compound dissimilarity matrix.
</p>

<hr>
<h2 id='minimalMolNet'>Minimal molecular network</h2><span id='topic+minimalMolNet'></span>

<h3>Description</h3>

<p>A molecular network. Generated by the <code><a href="#topic+molNet">molNet</a></code> function
used on the <code><a href="#topic+minimalCompDis">minimalCompDis</a></code> dataset, with
<code>cutOff = "median"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>minimalMolNet
</code></pre>


<h3>Format</h3>

<p>A <code>tbl_graph</code> object with 3 nodes and 4 edges.
</p>

<hr>
<h2 id='minimalNPCTable'>Minimal NPClassifier table</h2><span id='topic+minimalNPCTable'></span>

<h3>Description</h3>

<p>A table with the NPClassifier pathways, superclasses and classes,
along with the compound names, smiles and inchikey. Generated by
the <code><a href="#topic+NPCTable">NPCTable</a></code> function used on the
<code><a href="#topic+minimalCompData">minimalCompData</a></code> dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>minimalNPCTable
</code></pre>


<h3>Format</h3>

<p>A dataframe with 3 compounds and their NPClassifier classifications.
</p>

<hr>
<h2 id='minimalSampData'>Minimal sample dataset</h2><span id='topic+minimalSampData'></span>

<h3>Description</h3>

<p>A small made up dataset with phytochemical data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>minimalSampData
</code></pre>


<h3>Format</h3>

<p>A data frame with 4 rows and 3 columns. Each row is a sample,
each column is a phytochemical compound.
</p>

<hr>
<h2 id='minimalSampDis'>Minimal sample dissimilarity matrix</h2><span id='topic+minimalSampDis'></span>

<h3>Description</h3>

<p>A matrix with sample dissimilarities calculated from
<code><a href="#topic+minimalSampData">minimalSampData</a></code> and <code><a href="#topic+minimalCompDis">minimalCompDis</a></code>
using <code><a href="#topic+sampDis">sampDis</a></code> with <code>type = "GenUniFrac"</code> and
<code>alpha = 0.5</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>minimalSampDis
</code></pre>


<h3>Format</h3>

<p>A 4x4 sample dissimilarity matrix.
</p>

<hr>
<h2 id='molNet'>Generate a molecular network with some properties</h2><span id='topic+molNet'></span>

<h3>Description</h3>

<p>Function to generate a molecular network object, and some
basic properties of the network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>molNet(compDisMat, npcTable = NULL, cutOff = "median")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="molNet_+3A_compdismat">compDisMat</code></td>
<td>
<p>Compound dissimilarity matrix, as calculated by
<code><a href="#topic+compDis">compDis</a></code>. Note that the supplied dissimilarity matrix
is transformed into a similarity matrix, and this is what <code>cutOff</code>
values are set for. Note also that <code><a href="#topic+compDis">compDis</a></code> always outputs a
list of one or more matrices, while <code>molNet</code> requires a single matrix
as input. Therefore, a specific matrix has to be selected from this list,
as <code>compDisOutput$matrix</code>.</p>
</td></tr>
<tr><td><code id="molNet_+3A_npctable">npcTable</code></td>
<td>
<p>A data frame generated by <code><a href="#topic+NPCTable">NPCTable</a></code> can
be supplied for calculations of the number of NPC pathways and
network modularity.</p>
</td></tr>
<tr><td><code id="molNet_+3A_cutoff">cutOff</code></td>
<td>
<p>Cut-off value for compound similarities. Any similarity
lower than this value will be set to zero when the network is generated,
which strongly affects the look of the network. The value can be set
manually to any value between 0 and 1; to the <code>median</code> similarity
value from the compDisMat; or, if an NPCTable is supplied,
to <code>minPathway</code>, the lowest within-pathway similarity
(which allows all within-NPC-pathway similarities to be kept).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Molecular networks can be used to illustrate the biosynthetic/structural
similarity of phytochemical compounds in a sample, while
simultaneously visualizing their relative concentrations. <code>molNet</code>
creates the network, and <code><a href="#topic+molNetPlot">molNetPlot</a></code> can subsequently be
used to create a plot of the network.
</p>


<h3>Value</h3>

<p>List with a (tbl_graph) graph object, the number of compounds,
number of NPC pathways and a measure of the modularity of the network
(see <code><a href="igraph.html#topic+modularity">modularity</a></code>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(minimalCompDis)
data(minimalNPCTable)
molNet(minimalCompDis, minimalNPCTable, cutOff = 0)

data(alpinaCompDis)
molNet(compDisMat = alpinaCompDis, cutOff = 0.75)
</code></pre>

<hr>
<h2 id='molNetPlot'>Plot molecular networks</h2><span id='topic+molNetPlot'></span>

<h3>Description</h3>

<p>Function to conveniently create a basic plot of the molecular network
created by the <code><a href="#topic+molNet">molNet</a></code> function. Molecular networks can be
used to illustrate the biosynthetic/structural similarity of
phytochemical compounds in a sample, while simultaneously visualizing
their relative concentrations. In the network, nodes are compounds,
with node sizes or node colours representing the relative concentrations
of compounds. Edges connects nodes, with edge widths representing
compound similarity. This function exists to provide an easy way to
make basic molecular network plots. Customized network plots can be
created with e.g. the <code><a href="ggraph.html#topic+ggraph">ggraph</a></code> package or the
Cytoscape software platform.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>molNetPlot(
  sampleData,
  networkObject,
  groupData = NULL,
  npcTable = NULL,
  plotNames = FALSE,
  layout = "kk"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="molNetPlot_+3A_sampledata">sampleData</code></td>
<td>
<p>Data frame with the relative concentration of each
compound (column) in every sample (row).</p>
</td></tr>
<tr><td><code id="molNetPlot_+3A_networkobject">networkObject</code></td>
<td>
<p>A network object, as created by the
<code><a href="#topic+molNet">molNet</a></code> function. Note that this is only the network object,
which is one of the elements in the list outputted by <code><a href="#topic+molNet">molNet</a></code>.
The network is extracted as <code>molNetOutput$networkObject</code>.</p>
</td></tr>
<tr><td><code id="molNetPlot_+3A_groupdata">groupData</code></td>
<td>
<p>Grouping data (e.g. population, species etc.).
If supplied, a separate network will be created for each group.
Should be either a vector, or a data frame with a single column.</p>
</td></tr>
<tr><td><code id="molNetPlot_+3A_npctable">npcTable</code></td>
<td>
<p>It is optional but recommended to supply an
<code><a href="#topic+NPCTable">NPCTable</a></code>. This will result in network nodes being
coloured by their NPC pathway classification.</p>
</td></tr>
<tr><td><code id="molNetPlot_+3A_plotnames">plotNames</code></td>
<td>
<p>Indicates if compounds names should be included
in the molecular network plot.</p>
</td></tr>
<tr><td><code id="molNetPlot_+3A_layout">layout</code></td>
<td>
<p>Layout used by <code><a href="ggraph.html#topic+ggraph">ggraph</a></code> when creating
the network. The default chosen here, <code>"kk"</code>, is the the Kamada-Kawai
layout algorithm which in most cases should produce a visually
pleasing network. Another useful option is <code>"circle"</code>, which puts all
nodes in a circle, for easier comparisons between different networks.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The network object from <code><a href="#topic+molNet">molNet</a></code> and <code>sampleData</code> have to
be supplied. In addition, <code>groupData</code> and/or an <code><a href="#topic+NPCTable">NPCTable</a></code>
can be supplied. If an NPCTable is supplied, which is recommended,
node colours will represent NPC pathways, and node sizes the relative
concentration of the compounds. Edge widths represent compound similarity,
and only edges with similarity values above the <code>cutOff</code> value
in the <code><a href="#topic+molNet">molNet</a></code> function will be plotted.
If <code>groupData</code> is supplied, one network will be created for each group.
When <code>groupData</code> but not an <code><a href="#topic+NPCTable">NPCTable</a></code> is supplied,
compounds missing (i.e. having a mean of 0) from
specific groups are plotted as triangles. When <code>groupData</code>
and an <code><a href="#topic+NPCTable">NPCTable</a></code> is supplied, compounds missing from
specific groups have a white fill. Additionally, in both cases, edges
connecting to missing compounds are lighter coloured. These graphical
styles are done so that networks are more easy to compare across groups.
</p>


<h3>Value</h3>

<p>A plot with one or more molecular networks.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(minimalSampData)
data(minimalNPCTable)
data(minimalMolNet)
groups &lt;- c("A", "A", "B", "B")
molNetPlot(minimalSampData, minimalMolNet)
molNetPlot(minimalSampData, minimalMolNet, groups)
molNetPlot(minimalSampData, minimalMolNet, plotNames = TRUE)

data(alpinaSampData)
data(alpinaPopData)
data(alpinaMolNet)
data(alpinaNPCTable)
molNetPlot(sampleData = alpinaSampData, networkObject = alpinaMolNet,
npcTable = alpinaNPCTable)
</code></pre>

<hr>
<h2 id='NPCTable'>Generate NPClassifier classification</h2><span id='topic+NPCTable'></span>

<h3>Description</h3>

<p>Function to classify compounds with <em>NPClassifier</em>, and put the results in
a data frame containing the pathway, superclass and class for each compound.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NPCTable(compoundData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NPCTable_+3A_compounddata">compoundData</code></td>
<td>
<p>Data frame with the chemical compounds of interest,
usually the compounds found in the sample dataset.
Should include a column named &quot;compound&quot; with common names of
the compounds and a column named &quot;smiles&quot; with SMILES IDs of the compounds.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><em>NPClassifier</em> (Kim et al. 2021) is a deep-learning tool that automatically
classifies natural products (i.e. phytochemical compounds) into a
hierarchical classification of three levels: pathway, superclass and class.
This classification largely corresponds to the biosynthetic groups/pathways
the compounds are produced in. The <code>NPCTable</code> function
conveniently performs this classification directly in R on the
compounds in <code>compoundData</code>, by accessing the tool
at <a href="https://npclassifier.ucsd.edu/">https://npclassifier.ucsd.edu/</a> and downloading the classifications.
</p>


<h3>Value</h3>

<p>Data frame with the NPClassifier classification for each compound
as pathway, superclass and class. Note that compounds may be classified
in more than one group, or no group, at each level of classification.
</p>


<h3>References</h3>

<p>Kim HW, Wang M, Leber CA, Nothias L-F, Reher R, Kang KB,
van der Hooft JJJ, Dorrestein PC, Gerwick WH, Cottrell GW. 2021.
NPClassifier: A Deep Neural Network-Based Structural Classification
Tool for Natural Products. Journal of Natural Products 84: 2795-2807.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(minimalCompData)
NPCTable(minimalCompData)

data(alpinaCompData)
NPCTable(compoundData = alpinaCompData[1:3,]) # First three compounds only
</code></pre>

<hr>
<h2 id='quickChemoDiv'>Quickly calculate or plot chemodiversity</h2><span id='topic+quickChemoDiv'></span>

<h3>Description</h3>

<p>This function is a shortcut that makes use of many of the other functions
in the package. In one simple step chemodiversity is calculated, and if
requested also plotted, for users wanting to quickly explore their
data using standard parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>quickChemoDiv(
  sampleData,
  compoundData = NULL,
  groupData = NULL,
  outputType = "plots"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="quickChemoDiv_+3A_sampledata">sampleData</code></td>
<td>
<p>Data frame with the relative concentration of each
compound (column) in every sample (row).</p>
</td></tr>
<tr><td><code id="quickChemoDiv_+3A_compounddata">compoundData</code></td>
<td>
<p>Data frame with the compounds in <code>sampleData</code>
as rows. Should have a column named &quot;compound&quot; with common names of
the compounds, a column named &quot;smiles&quot; with SMILES IDs of the compounds,
and a column named &quot;inchikey&quot; with the InChIKey IDs for the compounds.
See <code><a href="#topic+chemodiv">chemodiv</a></code> for details on obtaining SMILES and InChIKey IDs.</p>
</td></tr>
<tr><td><code id="quickChemoDiv_+3A_groupdata">groupData</code></td>
<td>
<p>Grouping data (e.g. population, species etc.).
Should be either a vector, or a data frame with a single column.</p>
</td></tr>
<tr><td><code id="quickChemoDiv_+3A_outputtype">outputType</code></td>
<td>
<p>Type of output that should be returned: either
<code>data</code> to output a list with different types of chemodiversity data,
or <code>plots</code> to instead produce standard plots of this data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function requires sample data as input, and can also include
compound data. <code><a href="#topic+chemoDivCheck">chemoDivCheck</a></code> can be used to ensure these
datasets are correctly formatted, see <code><a href="#topic+chemodiv">chemodiv</a></code> for further
details on data formatting. If only sample data is supplied,
phytochemical diversity and dissimilarity will be calculated
as Hill diversity and Bray-Curtis dissimilarity, respectively.
If sample data and compound data is supplied, phytochemical diversity
and dissimilarity will be calculated as Functional Hill diversity
and Generalized UniFrac dissimilarity, respectively.
This function then uses the following other functions in the package:
</p>

<ul>
<li> <p><code><a href="#topic+compDis">compDis</a></code> is used to calculate compound dissimilarities
using PubChem Fingerprints, if compound data is supplied.
</p>
</li>
<li> <p><code><a href="#topic+calcDiv">calcDiv</a></code> is used to calculate (Functional) Hill
Diversity for q = 1.
</p>
</li>
<li> <p><code><a href="#topic+calcDivProf">calcDivProf</a></code> is used to calculate a diversity profile
with (Functional) Hill Diversity for <em>q</em> = 0-3.
</p>
</li>
<li> <p><code><a href="#topic+sampDis">sampDis</a></code> is used to calculate Bray-Curtis or
Generalized UniFrac dissimilarities between samples.
</p>
</li>
<li> <p><code><a href="#topic+chemoDivPlot">chemoDivPlot</a></code> is used to create different
chemodiversity plots if requested.
</p>
</li></ul>

<p><code>quickChemoDiv</code> is designed to provide an easy way to calculate and
visualize the most central measures of phytochemical diversity. It uses
default parameters to do so, which should be reasonable in most cases.
However, for detailed analyses it is recommended to use the separate
functions to allow for full control of function input, arguments and output.
</p>


<h3>Value</h3>

<p>Different types of chemodiversity measures, either as elements in
a list or as separate plots. If <code>outputType = "data"</code>, function returns
a compound dissimilarity matrix (if compound data was supplied),
a data frame with (Functional) Hill Diversity at <em>q</em> = 1,
a data frame with a (Functional) Hill Diversity profile for <em>q</em> = 0-3,
and a sample dissimilarity matrix. If <code>outputType = "plots"</code>,
these data sets are plotted as a dendrogram (if compound data was supplied),
a boxplot, a diversity profile plot and an NMDS plot, respectively.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(minimalCompData)
data(minimalSampData)
groups &lt;- c("A", "A", "B", "B")
quickChemoDiv(sampleData = minimalSampData, groupData = groups,
outputType = "data") # Without compound data

data(alpinaSampData)
data(alpinaPopData)
quickChemoDiv(sampleData = alpinaSampData, outputType = "plots",
groupData = alpinaPopData) # Without compound data
</code></pre>

<hr>
<h2 id='sampDis'>Calculate sample dissimilarities</h2><span id='topic+sampDis'></span>

<h3>Description</h3>

<p>Function to calculate dissimilarities between samples.
Either Bray-Curtis dissimilarities and/or Generalized UniFrac
dissimilarities are calculated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampDis(sampleData, compDisMat = NULL, type = "BrayCurtis", alpha = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sampDis_+3A_sampledata">sampleData</code></td>
<td>
<p>Data frame with the relative concentration of each
compound (column) in every sample (row).</p>
</td></tr>
<tr><td><code id="sampDis_+3A_compdismat">compDisMat</code></td>
<td>
<p>Compound dissimilarity matrix, as calculated by
<code><a href="#topic+compDis">compDis</a></code>. If this is supplied, Generalized UniFrac
dissimilarities can be calculated.</p>
</td></tr>
<tr><td><code id="sampDis_+3A_type">type</code></td>
<td>
<p>Type of sample dissimilarities to be calculated. This is
Bray-Curtis dissimilarities, <code>type = "BrayCurtis"</code>, and/or
Generalized UniFrac dissimilarities, <code>type = "GenUniFrac"</code>.</p>
</td></tr>
<tr><td><code id="sampDis_+3A_alpha">alpha</code></td>
<td>
<p>Parameter used in calculations of Generalized UniFrac
dissimilarities. alpha can be set between 0 and 1.
With <code>alpha = 0</code>, equal weight is put on every
branch in the dendrogram. With <code>alpha = 1</code>, branches are
weighted by their abundance, and hence more emphasis is put on high
abundance branches. <code>alpha = 0.5</code> strikes a balance between the two.
alpha 0.5 or 1 is recommended, with <code>alpha = 1</code> as default.
See Chen et al. 2012 for details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates a dissimilarity matrix for all the samples
in <code>sampleData</code>, for the given dissimilarity index/indices.
Bray-Curtis dissimilarities are calculated using only
the <code>sampleData</code>. This is the most commonly calculated dissimilarity
index used for phytochemical data (other types of dissimilarities are
easily calculated using the <code><a href="vegan.html#topic+vegdist">vegdist</a></code> function in
the <code>vegan</code> package).
</p>
<p>If a compound dissimilarity matrix, <code>compDisMat</code>, is supplied,
Generalized UniFrac dissimilarities can be calculated, which also
use the compound dissimilarity matrix for the sample dissimilarity
calculations. For the calculation of Generalized UniFrac
dissimilarities (Chen et al. 2012), the compound dissimilarity matrix is
transformed into a dendrogram using hierarchical clustering (with the
UPGMA method). Calculations of UniFrac dissimilarities quantifies the
fraction of the total branch length of the dendrogram that leads to
compounds present in either sample, but not both. The (weighted) Generalized
UniFrac dissimilarities implemented here additionally take compound
abundances into account. In this way, both the relative proportions of
compounds and the biosynthetic/structural dissimilarities of the compounds
are accounted for in the calculations of sample dissimilarities, such that
two samples containing more biosynthetically/structurally different
compounds have a higher pairwise dissimilarity than two samples
containing more biosynthetically/structurally similar compounds.
As with Bray-Curtis dissimilarities, Generalized UniFrac dissimilarities
range in value from 0 to 1.
</p>


<h3>Value</h3>

<p>List with sample dissimilarity matrices. A list is always
outputted, even if only one matrix is calculated.
</p>


<h3>References</h3>

<p>Bray JR, Curtis JT. 1957. An Ordination of the Upland Forest Communities
of Southern Wisconsin. Ecological Monographs 27: 325-349.
</p>
<p>Chen J, Bittinger K, Charlson ES, et al. 2012. Associating microbiome
composition with environmental covariates using generalized
UniFrac distances. Bioinformatics 28: 2106-2113.
</p>
<p>Lozupone C, Knight R. 2005. UniFrac: a New Phylogenetic Method for Comparing
Microbial Communities. Applied and Environmental Microbiology 71: 8228-8235.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(minimalSampData)
data(minimalCompDis)
sampDis(minimalSampData)
sampDis(sampleData = minimalSampData, compDisMat = minimalCompDis,
type = c("BrayCurtis", "GenUniFrac"), alpha = 0.5)

data(alpinaSampData)
data(alpinaCompDis)
sampDis(sampleData = alpinaSampData, compDisMat = alpinaCompDis,
type = "GenUniFrac")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
