<!DOCTYPE html><html><head><title>Help for package chicane</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {chicane}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#add.covariates'><p>add.covariates</p></a></li>
<li><a href='#add.fragment.coordinates'><p>add.fragment.coordinates</p></a></li>
<li><a href='#bedtools.installed'><p>bedtools.installed</p></a></li>
<li><a href='#bre80'><p>Bre80 Cell Line</p></a></li>
<li><a href='#check.model.numerical.fit'><p>check.model.identifiability</p></a></li>
<li><a href='#check.split.data.numerical.fit'><p>check.split.data.numerical.fit</p></a></li>
<li><a href='#chicane'><p>chicane</p></a></li>
<li><a href='#combine.replicates'><p>combine.replicates</p></a></li>
<li><a href='#compare.replicates'><p>compare.replicates</p></a></li>
<li><a href='#convert.bam'><p>convert.bam</p></a></li>
<li><a href='#convert.hicup.digest.bed'><p>convert.hicup.digest.bed</p></a></li>
<li><a href='#convert.standard.format'><p>convert.standard.format</p></a></li>
<li><a href='#convert.to.one.based'><p>convert.to.one.based</p></a></li>
<li><a href='#create.locus.plot'><p>create.locus.plot</p></a></li>
<li><a href='#create.modelfit.plot'><p>create.modelfit.plot</p></a></li>
<li><a href='#distance.bin'><p>distance.bin</p></a></li>
<li><a href='#distance.split'><p>distance.split</p></a></li>
<li><a href='#fill.in.zeros'><p>fill.in.zeros</p></a></li>
<li><a href='#filter.fragments'><p>filter.fragments</p></a></li>
<li><a href='#fit.glm'><p>fit.glm</p></a></li>
<li><a href='#fit.model'><p>fit.model</p></a></li>
<li><a href='#get.combination.count'><p>get.combination.count</p></a></li>
<li><a href='#get.components'><p>get.components</p></a></li>
<li><a href='#get.distance'><p>get.distance</p></a></li>
<li><a href='#get.id.components'><p>get.id.components</p></a></li>
<li><a href='#get.interaction.id'><p>get.interaction.id</p></a></li>
<li><a href='#get.trans.counts'><p>get.trans.counts</p></a></li>
<li><a href='#is.glm.nb.maxiter.warning'><p>is.glm.nb.maxiter.warning</p></a></li>
<li><a href='#is.glm.nb.theta.error'><p>check.glm.nb.theta.error</p></a></li>
<li><a href='#is.glm.nb.theta.warning'><p>is.glm.nb.theta.warning</p></a></li>
<li><a href='#model.rows.sanity.check'><p>model.rows.sanity.check</p></a></li>
<li><a href='#model.try.catch'><p>model.try.catch</p></a></li>
<li><a href='#multiple.testing.correct'><p>multiple.testing.correct</p></a></li>
<li><a href='#prepare.data'><p>prepare.data</p></a></li>
<li><a href='#read.bed'><p>read.bed</p></a></li>
<li><a href='#run.model.fitting'><p>run.model.fitting</p></a></li>
<li><a href='#smart.split'><p>smart.split</p></a></li>
<li><a href='#stratified.enrichment.sample'><p>stratified.enrichment.sample</p></a></li>
<li><a href='#test.enrichment'><p>test.enrichment</p></a></li>
<li><a href='#verify.interaction.data'><p>verify.interaction.data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Capture Hi-C Analysis Engine</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.8</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-11-03</td>
</tr>
<tr>
<td>Author:</td>
<td>Erle Holgersen [aut],
  Olivia Leavy [aut],
  Olivia Fletcher [aut],
  Frank Dudbridge [aut],
  Syed Haider [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Syed Haider &lt;Syed.Haider@icr.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Toolkit for processing and calling interactions in capture Hi-C data. Converts BAM files into counts of reads linking restriction fragments, and identifies pairs of fragments that interact more than expected by chance. Significant interactions are identified by comparing the observed read count to the expected background rate from a count regression model.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), gamlss.tr, gamlss, data.table</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS, stats, foreach, doParallel, iterators, bedr, knitr,
rmarkdown</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>bedtools</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, GenomicInteractions, GenomicRanges, Gviz, countreg</td>
</tr>
<tr>
<td>Additional_repositories:</td>
<td><a href="http://R-Forge.R-project.org/">http://R-Forge.R-project.org/</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-11-03 18:11:43 UTC; shaider</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-11-06 15:00:16 UTC</td>
</tr>
</table>
<hr>
<h2 id='add.covariates'>add.covariates</h2><span id='topic+add.covariates'></span>

<h3>Description</h3>

<p>Add model covariates (trans counts and distance) to an interactions data table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add.covariates(interaction.data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add.covariates_+3A_interaction.data">interaction.data</code></td>
<td>
<p>data.table with interaction data. Must contain columns 
<code>bait.id</code>, <code>target.id</code>, <code>bait.chr</code>, <code>bait.start</code>, 
<code>bait.end</code>, <code>target.chr</code>, <code>target.start</code>, <code>target.end</code> and <code>count</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Updated data table with new columns
</p>
<table>
<tr><td><code>bait.trans.count</code></td>
<td>
<p>number of trans interactions of bait fragment</p>
</td></tr>
<tr><td><code>target.trans.count</code></td>
<td>
<p>number of trans interactions of target fragment</p>
</td></tr>
<tr><td><code>distance</code></td>
<td>
<p>distance between bait and target fragment, or NA if trans</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Erle Holgersen &lt;Erle.Holgersen@icr.ac.uk&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	data(bre80);
input.cols &lt;- c('bait.id', 'target.id', 'bait.chr', 'bait.start', 
	'bait.end', 'target.chr', 'target.start', 'target.end', 'count');
output &lt;- add.covariates(bre80[, input.cols, with = FALSE]);

</code></pre>

<hr>
<h2 id='add.fragment.coordinates'>add.fragment.coordinates</h2><span id='topic+add.fragment.coordinates'></span>

<h3>Description</h3>

<p>Expand target and bait IDs of the form chrN:start-end to separate coordinate columns in the data table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add.fragment.coordinates(id.data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add.fragment.coordinates_+3A_id.data">id.data</code></td>
<td>
<p>data table containing columns <code>target.id</code> and/or <code>bait.id</code> to be expanded</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data table with added coordinate columns for target and bait (as applicable).
</p>


<h3>Author(s)</h3>

<p>Erle Holgersen &lt;Erle.Holgersen@icr.ac.uk&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bre80);
add.fragment.coordinates(bre80[, .(bait.id, target.id)]);

</code></pre>

<hr>
<h2 id='bedtools.installed'>bedtools.installed</h2><span id='topic+bedtools.installed'></span>

<h3>Description</h3>

<p>Check if bedtools exists in PATH
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bedtools.installed()
</code></pre>


<h3>Value</h3>

<p>Logical indicating if bedtools was found in PATH
</p>


<h3>Author(s)</h3>

<p>Erle Holgersen &lt;Erle.Holgersen@icr.ac.uk&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bedtools.installed();

</code></pre>

<hr>
<h2 id='bre80'>Bre80 Cell Line</h2><span id='topic+bre80'></span>

<h3>Description</h3>

<p>A dataset containing processed data from a capture Hi-C experiment in the Bre80 
normal epithelial breast tissue cell line. The experiment targeted several breast 
cancer risk loci, and reads that mapped to the 2q35 SNPs rs13387042 and rs16857609
are included in the dataset. 
</p>
<p>Data was prepared using the <code>prepare.data</code> function. Coordinates are GRCh38.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(bre80)
</code></pre>


<h3>Format</h3>

<p>A data table object with 47,766 rows and 13 columns.
</p>
<p>The variables are as follows:
</p>

<ul>
<li><p> target.id String in chrN:start-end format identifying target fragment
</p>
</li>
<li><p> bait.id String in chrN:start-end format identifying bait fragment
</p>
</li>
<li><p> target.chr Chromosome of target fragment
</p>
</li>
<li><p> target.start Start coordinate of target fragment (zero-based)
</p>
</li>
<li><p> target.end End coordinate of target fragment
</p>
</li>
<li><p> bait.chr Chromosome of bait fragment
</p>
</li>
<li><p> bait.start Start coordinate of bait fragment (zero-based)
</p>
</li>
<li><p> bait.end End coordinate of bait fragment
</p>
</li>
<li><p> bait.to.bait Boolean indicating if the interaction is bait-to-bait (i.e. the fragment listed as target is also a bait)
</p>
</li>
<li><p> bait.trans.count The number of reads linking the bait to fragments in trans (a measure of &quot;interactibility&quot;)
</p>
</li>
<li><p> target.trans.count The number of reads linking the target to fragments in trans (a measure of &quot;interactibility&quot;)
</p>
</li>
<li><p> distance Distance between the midpoints of the bait and target fragments (basepairs). NA for trans interactions
</p>
</li>
<li><p> count The number of reads linking the two fragments
</p>
</li></ul>



<h3>References</h3>

<p>Baxter, Joseph S., et al. &quot;Capture Hi-C identifies putative target genes at 33 breast cancer risk loci.&quot; Nature Communications 9.1 (2018): 1028.
</p>

<hr>
<h2 id='check.model.numerical.fit'>check.model.identifiability</h2><span id='topic+check.model.numerical.fit'></span>

<h3>Description</h3>

<p>Check if chicane model can be fit on a given dataset. 
<code>glm.nb</code> does not work when all responses are constant, or there are only two unique values and a covariate is a perfect predictor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check.model.numerical.fit(interaction.data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check.model.numerical.fit_+3A_interaction.data">interaction.data</code></td>
<td>
<p>Data table of interaction data on which model is to be fit</p>
</td></tr>
</table>


<h3>Value</h3>

<p>boolean indicating if model can be fit
</p>

<hr>
<h2 id='check.split.data.numerical.fit'>check.split.data.numerical.fit</h2><span id='topic+check.split.data.numerical.fit'></span>

<h3>Description</h3>

<p>Helper function to check if the chicane model can be fit on each element of a split data list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check.split.data.numerical.fit(split.data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check.split.data.numerical.fit_+3A_split.data">split.data</code></td>
<td>
<p>List of data.table objects with fragment interaction data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Logical indicating if the model can be fit
</p>

<hr>
<h2 id='chicane'>chicane</h2><span id='topic+chicane'></span>

<h3>Description</h3>

<p>Run full method for detecting significant interactions in capture Hi-C experiments, starting 
either from a BAM file or preprocessed data from <code>prepare.data</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chicane(
  bam = NULL,
  baits = NULL,
  fragments = NULL,
  interactions = NULL,
  replicate.merging.method = "sum",
  distribution = "negative-binomial",
  include.zeros = "none",
  bait.filters = c(0, 1),
  target.filters = c(0, 1),
  distance.bins = NULL,
  multiple.testing.correction = c("bait-level", "global"),
  adjustment.terms = NULL,
  remove.adjacent = FALSE,
  temp.directory = NULL,
  keep.files = FALSE,
  maxit = 100,
  epsilon = 1e-08,
  cores = 1,
  trace = FALSE,
  verbose = FALSE,
  interim.data.dir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chicane_+3A_bam">bam</code></td>
<td>
<p>Path to a BAM file</p>
</td></tr>
<tr><td><code id="chicane_+3A_baits">baits</code></td>
<td>
<p>Path to a BED file containing the baits</p>
</td></tr>
<tr><td><code id="chicane_+3A_fragments">fragments</code></td>
<td>
<p>Path to a BED file containing all restriction fragments in the genome</p>
</td></tr>
<tr><td><code id="chicane_+3A_interactions">interactions</code></td>
<td>
<p>Data table or path to a text file detailing fragment interactions, typically from <code>prepare.data</code>. 
Can be used instead of bam/baits/fragments specification if the text files have already been prepared.</p>
</td></tr>
<tr><td><code id="chicane_+3A_replicate.merging.method">replicate.merging.method</code></td>
<td>
<p>Method that should be used for merging replicates, if applicable</p>
</td></tr>
<tr><td><code id="chicane_+3A_distribution">distribution</code></td>
<td>
<p>Name of distribution of the counts. Options are 'negative-binomial', 
'poisson', 'truncated-poisson', and 'truncated-negative-binomial'</p>
</td></tr>
<tr><td><code id="chicane_+3A_include.zeros">include.zeros</code></td>
<td>
<p>String specifying what zero counts to include. Options are none (default), cis, and all.</p>
</td></tr>
<tr><td><code id="chicane_+3A_bait.filters">bait.filters</code></td>
<td>
<p>Vector of length two, where the first element corresponds to the lower-end filter and the second to the upper-end filter.
When global multiple testing correction is performed, altering the bait filtering settings may affect the number of significant results.</p>
</td></tr>
<tr><td><code id="chicane_+3A_target.filters">target.filters</code></td>
<td>
<p>Vector of length two, giving lower and higher filter, respectively. 
Changing this filtering setting may affect multiple testing correction by altering the number of tests performed.</p>
</td></tr>
<tr><td><code id="chicane_+3A_distance.bins">distance.bins</code></td>
<td>
<p>Number of bins to split distance into. Models are fit separately in each bin.</p>
</td></tr>
<tr><td><code id="chicane_+3A_multiple.testing.correction">multiple.testing.correction</code></td>
<td>
<p>String specifying how multiple testing correction should be performed, by bait or globally.</p>
</td></tr>
<tr><td><code id="chicane_+3A_adjustment.terms">adjustment.terms</code></td>
<td>
<p>Character vector of extra terms to adjust for in the model fit.</p>
</td></tr>
<tr><td><code id="chicane_+3A_remove.adjacent">remove.adjacent</code></td>
<td>
<p>Logical indicating whether to remove all reads mapping to adjacent restriction fragments.</p>
</td></tr>
<tr><td><code id="chicane_+3A_temp.directory">temp.directory</code></td>
<td>
<p>Directory where temporary files should be stored. Defaults to current directory.</p>
</td></tr>
<tr><td><code id="chicane_+3A_keep.files">keep.files</code></td>
<td>
<p>Logical indicating whether to keep temporary files</p>
</td></tr>
<tr><td><code id="chicane_+3A_maxit">maxit</code></td>
<td>
<p>Maximum number of IWLS iterations for fitting the model (passed to <code>glm.control</code>)</p>
</td></tr>
<tr><td><code id="chicane_+3A_epsilon">epsilon</code></td>
<td>
<p>Positive convergence tolerance for Poisson and negative binomial models. Passed to <code>glm.control</code></p>
</td></tr>
<tr><td><code id="chicane_+3A_cores">cores</code></td>
<td>
<p>Integer value specifying how many cores to use to fit model for cis-interactions.</p>
</td></tr>
<tr><td><code id="chicane_+3A_trace">trace</code></td>
<td>
<p>Logical indicating if output should be produced for each of model fitting procedure. Passed to <code>glm.control</code> or <code>gamlss.control</code></p>
</td></tr>
<tr><td><code id="chicane_+3A_verbose">verbose</code></td>
<td>
<p>Logical indicating whether to print progress reports.</p>
</td></tr>
<tr><td><code id="chicane_+3A_interim.data.dir">interim.data.dir</code></td>
<td>
<p>Path to directory to store intermediate QC data and plots. NULL indicate skip intermediate results.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data table with columns
</p>
<table>
<tr><td><code>target.id</code></td>
<td>
<p>String in chrN:start-end format identifying target fragment</p>
</td></tr>
<tr><td><code>bait.id</code></td>
<td>
<p>String in chrN:start-end format identifying bait fragment</p>
</td></tr>
<tr><td><code>target.chr</code></td>
<td>
<p>Chromosome of target fragment</p>
</td></tr>
<tr><td><code>target.start</code></td>
<td>
<p>Start coordinate of target fragment (zero-based)</p>
</td></tr>
<tr><td><code>target.end</code></td>
<td>
<p>End coordinate of target fragment</p>
</td></tr>
<tr><td><code>bait.chr</code></td>
<td>
<p>Chromosome of bait fragment</p>
</td></tr>
<tr><td><code>bait.start</code></td>
<td>
<p>Start coordinate of bait fragment (zero-based)</p>
</td></tr>
<tr><td><code>bait.end</code></td>
<td>
<p>End coordinate of bait fragment</p>
</td></tr>
<tr><td><code>bait.to.bait</code></td>
<td>
<p>Boolean indicating if the interaction is bait-to-bait (i.e. the fragment listed as target is also a bait)</p>
</td></tr>
<tr><td><code>bait.trans.count</code></td>
<td>
<p>The number of reads linking the bait to fragments in trans (a measure of &quot;interactibility&quot;)</p>
</td></tr>
<tr><td><code>target.trans.count</code></td>
<td>
<p>The number of reads linking the target to fragments in trans (a measure of &quot;interactibility&quot;)</p>
</td></tr>
<tr><td><code>distance</code></td>
<td>
<p>Distance between the midpoints of the bait and target fragments (basepairs). NA for trans interactions</p>
</td></tr>
<tr><td><code>count</code></td>
<td>
<p>The number of reads linking the two fragments</p>
</td></tr>
<tr><td><code>expected</code></td>
<td>
<p>The expected number of reads linking the two fragments under the fitted model</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>P-value for test of the observed number of reads significantly exceeding the expected count</p>
</td></tr>
<tr><td><code>q.value</code></td>
<td>
<p>FDR-corrected p-value</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Erle Holgersen &lt;Erle.Holgersen@icr.ac.uk&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if( bedtools.installed() ) {
  # start from BAM file
  bam &lt;- system.file('extdata', 'Bre80_2q35.bam', package = 'chicane');
  baits &lt;- system.file('extdata', '2q35.bed', package = 'chicane');
  fragments &lt;- system.file('extdata', 'GRCh38_HindIII_chr2.bed.gz', package = 'chicane');
  results &lt;- chicane(
	bam = bam, 
	baits = baits, 
	fragments = fragments
	);
}

# start from pre-processed data
data(bre80); 
results &lt;- chicane(interactions = bre80);


</code></pre>

<hr>
<h2 id='combine.replicates'>combine.replicates</h2><span id='topic+combine.replicates'></span>

<h3>Description</h3>

<p>Merge biological replicates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine.replicates(replicates, method = c("sum", "weighted-sum"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combine.replicates_+3A_replicates">replicates</code></td>
<td>
<p>list of data table objects from <code>prepare.data</code></p>
</td></tr>
<tr><td><code id="combine.replicates_+3A_method">method</code></td>
<td>
<p>string specifying the method for merging replicates. Options are 'sum' and 'weighted-sum'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The parameter <code>method</code> determines which method is used for merging 
replicates. Available options are weighted-sum and sum.
</p>
<p>'weighted-sum' implements the size factor scaling approach used in DEseq, 
rounded to the closest integer. See Anders and Huber 2010 for details.
</p>
<p>'sum' is the naive sum of counts across biological replicates.
</p>


<h3>Value</h3>

<p>Data table object containing merged data, where counts are stored in colums
</p>
<table>
<tr><td><code>count.i</code></td>
<td>
<p>count of interaction in ith replicate</p>
</td></tr>
<tr><td><code>count</code></td>
<td>
<p>count after merging replicates</p>
</td></tr>
</table>


<h3>References</h3>

<p>Anders, Simon, and Wolfgang Huber. &quot;Differential expression analysis for sequence count data.&quot; <em>Genome biology</em> 11.10 (2010): R106.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if( bedtools.installed() ) {
	  # preprocess data
	  bam &lt;- system.file('extdata', 'Bre80_2q35.bam', package = 'chicane');
	  baits &lt;- system.file('extdata', '2q35.bed', package = 'chicane');
	  fragments &lt;- system.file('extdata', 'GRCh38_HindIII_chr2.bed.gz', package = 'chicane');
	  input.data &lt;- prepare.data(
	  bam = bam, 
	  baits = baits, 
	  fragments = fragments
	  );

	  # combined two datasets into one
  merged &lt;- combine.replicates(list(input.data, input.data));
 }


</code></pre>

<hr>
<h2 id='compare.replicates'>compare.replicates</h2><span id='topic+compare.replicates'></span>

<h3>Description</h3>

<p>Compare replicates in a pairwise manner and further stratified by distance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compare.replicates(interaction.data = NULL, output.directory = "./")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compare.replicates_+3A_interaction.data">interaction.data</code></td>
<td>
<p>A named vector specifying paths to files created using {prepare.data()}</p>
</td></tr>
<tr><td><code id="compare.replicates_+3A_output.directory">output.directory</code></td>
<td>
<p>Path to the output directory where pairwise plots are generated</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if pairwise plots were successfully created
</p>


<h3>Author(s)</h3>

<p>Syed Haider
</p>


<h3>Examples</h3>

<pre><code class='language-R'># TODO

</code></pre>

<hr>
<h2 id='convert.bam'>convert.bam</h2><span id='topic+convert.bam'></span>

<h3>Description</h3>

<p>Convert a BAM file to a format that can be used for replicate merging. 
</p>
<p><strong>Note:</strong> This function does not process data enough to be used for interaction calling. Use <code>prepare.data</code> for full preprocessing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert.bam(bam, baits, fragments, temp.directory = NULL, keep.files = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert.bam_+3A_bam">bam</code></td>
<td>
<p>Path to a BAM file</p>
</td></tr>
<tr><td><code id="convert.bam_+3A_baits">baits</code></td>
<td>
<p>Path to a BED file containing the baits</p>
</td></tr>
<tr><td><code id="convert.bam_+3A_fragments">fragments</code></td>
<td>
<p>Path to a BED file containing all restriction fragments in the genome</p>
</td></tr>
<tr><td><code id="convert.bam_+3A_temp.directory">temp.directory</code></td>
<td>
<p>Directory where temporary files should be stored. Defaults to current directory.</p>
</td></tr>
<tr><td><code id="convert.bam_+3A_keep.files">keep.files</code></td>
<td>
<p>Logical indicating whether to keep temporary files</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Erle Holgersen &lt;Erle.Holgersen@icr.ac.uk&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+prepare.data">prepare.data</a></code>
</p>

<hr>
<h2 id='convert.hicup.digest.bed'>convert.hicup.digest.bed</h2><span id='topic+convert.hicup.digest.bed'></span>

<h3>Description</h3>

<p>Convert a HiCUP digest file to BED format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert.hicup.digest.bed(hicup.digest, file.name = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert.hicup.digest.bed_+3A_hicup.digest">hicup.digest</code></td>
<td>
<p>Path to HiCUP digest</p>
</td></tr>
<tr><td><code id="convert.hicup.digest.bed_+3A_file.name">file.name</code></td>
<td>
<p>Path to output file. A blank string indicates output to the console.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>hicup.digest &lt;- system.file('extdata', 'HiCUP_digest_example.txt', package = 'chicane');
	convert.hicup.digest.bed(hicup.digest);

</code></pre>

<hr>
<h2 id='convert.standard.format'>convert.standard.format</h2><span id='topic+convert.standard.format'></span>

<h3>Description</h3>

<p>Create a file in standard format for cross compatability including with 
WashU Epigenome Browser.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert.standard.format(chicane.results, file.name = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert.standard.format_+3A_chicane.results">chicane.results</code></td>
<td>
<p>Path to CHiCANE interaction calls file</p>
</td></tr>
<tr><td><code id="convert.standard.format_+3A_file.name">file.name</code></td>
<td>
<p>Path to output file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if output files are created successfully
</p>


<h3>Author(s)</h3>

<p>Andrea Gillespie, Syed Haider
</p>


<h3>Examples</h3>

<pre><code class='language-R'>chicane.results &lt;- system.file(
   'extdata', 'T47D_2q35_filtered_chicane_calls.txt', 
   package = 'chicane'
   );
 output.file = file.path(tempdir(), 'temp_standard_format.txt');
	convert.standard.format(chicane.results, file.name = output.file);

</code></pre>

<hr>
<h2 id='convert.to.one.based'>convert.to.one.based</h2><span id='topic+convert.to.one.based'></span>

<h3>Description</h3>

<p>Convert zero-based region in format chr:start-end to 1-based
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert.to.one.based(id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert.to.one.based_+3A_id">id</code></td>
<td>
<p>string in format chr:start-end</p>
</td></tr>
</table>


<h3>Value</h3>

<p>one-converted ID
</p>

<hr>
<h2 id='create.locus.plot'>create.locus.plot</h2><span id='topic+create.locus.plot'></span>

<h3>Description</h3>

<p>Create a file compatible with WashU Epigenome Browser from CHiCANE interaction calls.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create.locus.plot(
  genome = "hg38",
  chr = NULL,
  start = NULL,
  end = NULL,
  gene.data = NULL,
  genomic.features = NULL,
  feature.name = NULL,
  fdr.filter = 0.05,
  interaction.data = NULL,
  file.name = NULL,
  height = 5.5,
  width = 8.5,
  track.heights = c(0.2, 0.5, 0.8, 0.5, 1.5, 2),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create.locus.plot_+3A_genome">genome</code></td>
<td>
<p>Name of genome build (e.g. 'hg38' or 'hg37')</p>
</td></tr>
<tr><td><code id="create.locus.plot_+3A_chr">chr</code></td>
<td>
<p>Chromosome number for desired locus including 'chr' (e.g. 'chr1')</p>
</td></tr>
<tr><td><code id="create.locus.plot_+3A_start">start</code></td>
<td>
<p>Start coordinate of desired locus</p>
</td></tr>
<tr><td><code id="create.locus.plot_+3A_end">end</code></td>
<td>
<p>End coordinate of desired locus</p>
</td></tr>
<tr><td><code id="create.locus.plot_+3A_gene.data">gene.data</code></td>
<td>
<p>Path to chosen genome annotation file in .gtf format</p>
</td></tr>
<tr><td><code id="create.locus.plot_+3A_genomic.features">genomic.features</code></td>
<td>
<p>Path to BED file with coordinates of desired feature track</p>
</td></tr>
<tr><td><code id="create.locus.plot_+3A_feature.name">feature.name</code></td>
<td>
<p>Title to appear above genomic features</p>
</td></tr>
<tr><td><code id="create.locus.plot_+3A_fdr.filter">fdr.filter</code></td>
<td>
<p>Q-value filter threshold for interaction calls to be included</p>
</td></tr>
<tr><td><code id="create.locus.plot_+3A_interaction.data">interaction.data</code></td>
<td>
<p>Path to unfiltered CHiCANE calls output</p>
</td></tr>
<tr><td><code id="create.locus.plot_+3A_file.name">file.name</code></td>
<td>
<p>Path to output file</p>
</td></tr>
<tr><td><code id="create.locus.plot_+3A_height">height</code></td>
<td>
<p>Height in inches for desired plot</p>
</td></tr>
<tr><td><code id="create.locus.plot_+3A_width">width</code></td>
<td>
<p>Width in inches of desired plot</p>
</td></tr>
<tr><td><code id="create.locus.plot_+3A_track.heights">track.heights</code></td>
<td>
<p>Vector of length 6 indicating desired height of individual tracks</p>
</td></tr>
<tr><td><code id="create.locus.plot_+3A_...">...</code></td>
<td>
<p>Any additional parameters to <code>Gviz::plotTracks</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if plot was successfully created
</p>


<h3>Author(s)</h3>

<p>Andrea Gillespie, Syed Haider
</p>


<h3>Examples</h3>

<pre><code class='language-R'># In order to conserve memory only significant interactions are included in example 
# interaction.data file. However, in order to show raw counts, unfiltered calls should be 
# included and only significant interactions (as set by fdr.filter) wil be displayed

gene.data &lt;- system.file('extdata', 'gencode_2q35.gtf', package = 'chicane');
genomic.features &lt;- system.file('extdata', '2q35.bed', package = 'chicane');
interaction.data &lt;- system.file(
  'extdata', 'T47D_2q35_filtered_chicane_calls.txt', 
  package = 'chicane'
  );
file.name &lt;- file.path(tempdir(), "chr2_interactions.pdf");


create.locus.plot(
   genome = 'hg38', 
   chr = 'chr2', 
   start = 216600000,
   end = 217200000,
   gene.data = gene.data,
   genomic.features = genomic.features,
   feature.name = 'baits',
   interaction.data = interaction.data,
   file.name = file.name,
   collapseTranscripts = TRUE,
   shape = "arrow"
   );
 

</code></pre>

<hr>
<h2 id='create.modelfit.plot'>create.modelfit.plot</h2><span id='topic+create.modelfit.plot'></span>

<h3>Description</h3>

<p>create a plot representing model's fit
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create.modelfit.plot(model, file.name = NULL, resolution = 300)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create.modelfit.plot_+3A_model">model</code></td>
<td>
<p>An object of fitted model</p>
</td></tr>
<tr><td><code id="create.modelfit.plot_+3A_file.name">file.name</code></td>
<td>
<p>A string specifying plotting file name</p>
</td></tr>
<tr><td><code id="create.modelfit.plot_+3A_resolution">resolution</code></td>
<td>
<p>A numeric specifying plot's resolution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if plot was successfully created
</p>


<h3>Author(s)</h3>

<p>Syed Haider
</p>

<hr>
<h2 id='distance.bin'>distance.bin</h2><span id='topic+distance.bin'></span>

<h3>Description</h3>

<p>Assign distances to a meaningful category
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distance.bin(distance)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="distance.bin_+3A_distance">distance</code></td>
<td>
<p>Vector of distances that should be mapped to a distance bin</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of same length as <code>distance</code> containing assigned distance bins
</p>

<hr>
<h2 id='distance.split'>distance.split</h2><span id='topic+distance.split'></span>

<h3>Description</h3>

<p>Split interaction data into subsets that are large enough for the chicane model to be fit (see Details), 
based on distance. This step allows the distance term in the model to be fit in a piecewise linear fashion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distance.split(
  interaction.data,
  distance.bins = NULL,
  min.rows.bin = 50,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="distance.split_+3A_interaction.data">interaction.data</code></td>
<td>
<p>Data table of interaction data, typically from <code>prepare.data</code></p>
</td></tr>
<tr><td><code id="distance.split_+3A_distance.bins">distance.bins</code></td>
<td>
<p>Number of distance bins desired. If NULL, a number is chosen to ensure that the negative binomial can be fit in all bins.</p>
</td></tr>
<tr><td><code id="distance.split_+3A_min.rows.bin">min.rows.bin</code></td>
<td>
<p>The minimum number of expected rows in a distance bin. Ignored if distance.bins is set</p>
</td></tr>
<tr><td><code id="distance.split_+3A_verbose">verbose</code></td>
<td>
<p>Logical indicating whether to print progress reports</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Fitting <code>glm.nb</code> fails when there is a lack of overdispersion in the data. The chicane method
contains logic to catch these errors and instead fit a Poisson model. However, to avoid this happening
more than necessary, an attempt is made to avoid distance splits that will clearly result in numerical errors.
This includes bins of data where the count is the same for all rows, 
or a covariate is a perfect predictor of count.
</p>


<h3>Value</h3>

<p>List where each element corresponds to a specified distance bin, and the final one corresponding to trans-interactions (if present)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bre80);
distance.split(bre80);

</code></pre>

<hr>
<h2 id='fill.in.zeros'>fill.in.zeros</h2><span id='topic+fill.in.zeros'></span><span id='topic+fill.in.zeroes'></span>

<h3>Description</h3>

<p>Add zero counts to interaction data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fill.in.zeros(interaction.data, baits, fragments)

fill.in.zeroes(interaction.data, baits, fragments)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fill.in.zeros_+3A_interaction.data">interaction.data</code></td>
<td>
<p>Data table containing interaction data</p>
</td></tr>
<tr><td><code id="fill.in.zeros_+3A_baits">baits</code></td>
<td>
<p>Vector of bait IDs used in the experiment, in format chrN:start-end</p>
</td></tr>
<tr><td><code id="fill.in.zeros_+3A_fragments">fragments</code></td>
<td>
<p>Vector of potential fragments the baits can link up to, in format chrN:start-end</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data table containing origiina
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bre80);
	bait.file &lt;- system.file('extdata', '2q35.bed', package = 'chicane');
	fragment.file &lt;- system.file('extdata', 'GRCh38_HindIII_chr2.bed.gz', package = 'chicane');
	results &lt;- fill.in.zeros(
	bre80,
	baits = read.bed(bait.file), 
	fragments = read.bed(fragment.file)
	);

</code></pre>

<hr>
<h2 id='filter.fragments'>filter.fragments</h2><span id='topic+filter.fragments'></span>

<h3>Description</h3>

<p>Filter low and high-interacting restriction fragments based on the total number of trans counts
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter.fragments(
  interaction.data,
  bait.filters = c(0, 1),
  target.filters = c(0, 1),
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter.fragments_+3A_interaction.data">interaction.data</code></td>
<td>
<p>Data table containing interactions</p>
</td></tr>
<tr><td><code id="filter.fragments_+3A_bait.filters">bait.filters</code></td>
<td>
<p>Vector of length two, where the first element corresponds to the lower-end filter and the second to the upper-end filter.
When global multiple testing correction is performed, altering the bait filtering settings may affect the number of significant results.</p>
</td></tr>
<tr><td><code id="filter.fragments_+3A_target.filters">target.filters</code></td>
<td>
<p>Vector of length two, giving lower and higher filter, respectively. 
Changing this filtering setting may affect multiple testing correction by altering the number of tests performed.</p>
</td></tr>
<tr><td><code id="filter.fragments_+3A_verbose">verbose</code></td>
<td>
<p>Logical indicating whether to print progress reports.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data table containing fragments that passed all filters
</p>


<h3>Author(s)</h3>

<p>Erle Holgersen &lt;Erle.Holgersen@icr.ac.uk&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># filter out lowest 10% of baits
	filter.fragments(bre80, bait.filters = c(0.1, 1))

</code></pre>

<hr>
<h2 id='fit.glm'>fit.glm</h2><span id='topic+fit.glm'></span>

<h3>Description</h3>

<p>Fit GLM according to a specified distribution. This needs to be done separately from <code>glm</code>
in order to include negative binomial and truncated distributions as options.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit.glm(
  formula,
  data,
  distribution = c("negative-binomial", "poisson", "truncated-poisson",
    "truncated-negative-binomial"),
  start = NULL,
  init.theta = NULL,
  maxit = 100,
  epsilon = 1e-08,
  trace = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit.glm_+3A_formula">formula</code></td>
<td>
<p>Formula specifying model of interest</p>
</td></tr>
<tr><td><code id="fit.glm_+3A_data">data</code></td>
<td>
<p>Data frame containing variables specified in formula</p>
</td></tr>
<tr><td><code id="fit.glm_+3A_distribution">distribution</code></td>
<td>
<p>Name of distribution of the counts. Options are 'negative-binomial', 
'poisson', 'truncated-poisson', and 'truncated-negative-binomial'</p>
</td></tr>
<tr><td><code id="fit.glm_+3A_start">start</code></td>
<td>
<p>Starting values for model coefficients</p>
</td></tr>
<tr><td><code id="fit.glm_+3A_init.theta">init.theta</code></td>
<td>
<p>Initial value of theta if fitting the negative binomial distribution</p>
</td></tr>
<tr><td><code id="fit.glm_+3A_maxit">maxit</code></td>
<td>
<p>Maximum number of IWLS iterations for fitting the model (passed to <code>glm.control</code>)</p>
</td></tr>
<tr><td><code id="fit.glm_+3A_epsilon">epsilon</code></td>
<td>
<p>Positive convergence tolerance for Poisson and negative binomial models. Passed to <code>glm.control</code></p>
</td></tr>
<tr><td><code id="fit.glm_+3A_trace">trace</code></td>
<td>
<p>Logical indicating if output should be produced for each of model fitting procedure. Passed to <code>glm.control</code> or <code>gamlss.control</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with elements
</p>
<table>
<tr><td><code>model</code></td>
<td>
<p>model object</p>
</td></tr>
<tr><td><code>expected.values</code></td>
<td>
<p>vector of expected values for each element in original data</p>
</td></tr>
<tr><td><code>p.values</code></td>
<td>
<p>vector of p-values for test of significantly higher response than expected</p>
</td></tr>
</table>

<hr>
<h2 id='fit.model'>fit.model</h2><span id='topic+fit.model'></span>

<h3>Description</h3>

<p>Fit negative binomial model to obtain p-values for interactions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit.model(
  interaction.data,
  distance.bins = NULL,
  distribution = "negative-binomial",
  bait.filters = c(0, 1),
  target.filters = c(0, 1),
  adjustment.terms = NULL,
  maxit = 100,
  epsilon = 1e-08,
  cores = 1,
  trace = FALSE,
  verbose = FALSE,
  interim.data.dir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit.model_+3A_interaction.data">interaction.data</code></td>
<td>
<p>data.table object containing interaction counts. Must contain columns distance, count, and bait_trans_count.</p>
</td></tr>
<tr><td><code id="fit.model_+3A_distance.bins">distance.bins</code></td>
<td>
<p>Number of bins to split distance into. Models are fit separately in each bin.</p>
</td></tr>
<tr><td><code id="fit.model_+3A_distribution">distribution</code></td>
<td>
<p>Name of distribution of the counts. Options are 'negative-binomial', 
'poisson', 'truncated-poisson', and 'truncated-negative-binomial'</p>
</td></tr>
<tr><td><code id="fit.model_+3A_bait.filters">bait.filters</code></td>
<td>
<p>Vector of length two, where the first element corresponds to the lower-end filter and the second to the upper-end filter.
When global multiple testing correction is performed, altering the bait filtering settings may affect the number of significant results.</p>
</td></tr>
<tr><td><code id="fit.model_+3A_target.filters">target.filters</code></td>
<td>
<p>Vector of length two, giving lower and higher filter, respectively. 
Changing this filtering setting may affect multiple testing correction by altering the number of tests performed.</p>
</td></tr>
<tr><td><code id="fit.model_+3A_adjustment.terms">adjustment.terms</code></td>
<td>
<p>Character vector of extra terms to adjust for in the model fit.</p>
</td></tr>
<tr><td><code id="fit.model_+3A_maxit">maxit</code></td>
<td>
<p>Maximum number of IWLS iterations for fitting the model (passed to <code>glm.control</code>)</p>
</td></tr>
<tr><td><code id="fit.model_+3A_epsilon">epsilon</code></td>
<td>
<p>Positive convergence tolerance for Poisson and negative binomial models. Passed to <code>glm.control</code></p>
</td></tr>
<tr><td><code id="fit.model_+3A_cores">cores</code></td>
<td>
<p>Integer value specifying how many cores to use to fit model for cis-interactions.</p>
</td></tr>
<tr><td><code id="fit.model_+3A_trace">trace</code></td>
<td>
<p>Logical indicating if output should be produced for each of model fitting procedure. Passed to <code>glm.control</code> or <code>gamlss.control</code></p>
</td></tr>
<tr><td><code id="fit.model_+3A_verbose">verbose</code></td>
<td>
<p>Logical indicating whether to print progress reports.</p>
</td></tr>
<tr><td><code id="fit.model_+3A_interim.data.dir">interim.data.dir</code></td>
<td>
<p>Path to directory to store intermediate QC data and plots.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Fit a negative binomial model for obtaining p-value for interactions. The data is first sorted by distance, and models
are fit separately in each quantile of the distance-sorted data.
</p>


<h3>Value</h3>

<p>Interactions data with expected number of interactions and p-values added.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
	data(bre80);
fit.model(bre80);


</code></pre>

<hr>
<h2 id='get.combination.count'>get.combination.count</h2><span id='topic+get.combination.count'></span>

<h3>Description</h3>

<p>Calculate the number of possible combinations between baits and fragments, 
excluding self-ligations and only counting bait-to-bait interactions once (e.g. a-b, not b-a)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.combination.count(baits, fragments, cis.only = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get.combination.count_+3A_baits">baits</code></td>
<td>
<p>vector of bait IDs in form chrN:start-end</p>
</td></tr>
<tr><td><code id="get.combination.count_+3A_fragments">fragments</code></td>
<td>
<p>vector of fragment IDs in form chrN:start-end</p>
</td></tr>
<tr><td><code id="get.combination.count_+3A_cis.only">cis.only</code></td>
<td>
<p>logical indicating whether cis-interactions only should be considered</p>
</td></tr>
</table>


<h3>Value</h3>

<p>total number of possible combinations
</p>

<hr>
<h2 id='get.components'>get.components</h2><span id='topic+get.components'></span>

<h3>Description</h3>

<p>Split a fragment in format chr:start-end to a list of corresponding elements
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.components(id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get.components_+3A_id">id</code></td>
<td>
<p>string in format chr:start-end</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with entries 'chr', 'start', 'end'
</p>

<hr>
<h2 id='get.distance'>get.distance</h2><span id='topic+get.distance'></span>

<h3>Description</h3>

<p>Calculate distance between bait and target region
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.distance(interaction.data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get.distance_+3A_interaction.data">interaction.data</code></td>
<td>
<p>data.table with interaction data. Must contain columns bait.chr, bait.start, bait.end, target.chr, target.start, target.end</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of absolute distances (NA for trans-interactions)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	data(bre80);
input.cols &lt;- c('bait.chr', 'bait.start', 'bait.end', 
		'target.chr', 'target.start', 'target.end');
	get.distance( bre80[, input.cols, with = FALSE]);

</code></pre>

<hr>
<h2 id='get.id.components'>get.id.components</h2><span id='topic+get.id.components'></span>

<h3>Description</h3>

<p>Split a segment ID in form <code>chrN:start-end</code> into its different components
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.id.components(id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get.id.components_+3A_id">id</code></td>
<td>
<p>segment ID of form <code>chrN:start-end</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of length three, where the elements are chromosome, start, and end, respectively. 
If <code>id</code> is a vector, a list of the same length is returned
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	get.id.components('chrX:6-30');
get.id.components(c('3:4-10', '22:1000-20000'))

</code></pre>

<hr>
<h2 id='get.interaction.id'>get.interaction.id</h2><span id='topic+get.interaction.id'></span>

<h3>Description</h3>

<p>Generate a unique identifying ID for each interaction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.interaction.id(bait, other.end, bait.to.bait, zero.based = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get.interaction.id_+3A_bait">bait</code></td>
<td>
<p>id of bait in format chr:start-end</p>
</td></tr>
<tr><td><code id="get.interaction.id_+3A_other.end">other.end</code></td>
<td>
<p>id of other end in format chr:start-end</p>
</td></tr>
<tr><td><code id="get.interaction.id_+3A_bait.to.bait">bait.to.bait</code></td>
<td>
<p>logical indicating whether both ends are baits</p>
</td></tr>
<tr><td><code id="get.interaction.id_+3A_zero.based">zero.based</code></td>
<td>
<p>logical indicating if IDs are zero-based</p>
</td></tr>
</table>


<h3>Value</h3>

<p>string identifying interaction
</p>

<hr>
<h2 id='get.trans.counts'>get.trans.counts</h2><span id='topic+get.trans.counts'></span>

<h3>Description</h3>

<p>Calculate the number of trans-interactions per fragment, accounting for the fact that baits can be listed either as bait or target.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.trans.counts(interaction.data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get.trans.counts_+3A_interaction.data">interaction.data</code></td>
<td>
<p>Data table containing interactions</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data table with columns <code>fragment.id</code> and <code>trans.count</code>.
</p>
<table>
<tr><td><code>fragment.id</code></td>
<td>
<p>ID of restriction fragment in chrN:start-end format</p>
</td></tr>
<tr><td><code>trans.count</code></td>
<td>
<p>Number of trans interactions involving the fragment</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>	data(bre80);
	get.trans.counts(bre80[, .(bait.chr, target.chr, bait.id, target.id, count)]);

</code></pre>

<hr>
<h2 id='is.glm.nb.maxiter.warning'>is.glm.nb.maxiter.warning</h2><span id='topic+is.glm.nb.maxiter.warning'></span>

<h3>Description</h3>

<p>Check if a warning object is an iteration limit reached warning from <code>glm.nb</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.glm.nb.maxiter.warning(w)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.glm.nb.maxiter.warning_+3A_w">w</code></td>
<td>
<p>Warning object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Logical indicating if warning matches iteration limit reached warning
</p>

<hr>
<h2 id='is.glm.nb.theta.error'>check.glm.nb.theta.error</h2><span id='topic+is.glm.nb.theta.error'></span>

<h3>Description</h3>

<p>Check if an error matches the error raised by <code>glm.nb</code> due to an inflated theta estimate.
This happens when the variance of the negative binomial does not exceed the mean (i.e. there is no overdispersion).
In such cases, the Poisson distribution may be a suitable alternative.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.glm.nb.theta.error(e)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.glm.nb.theta.error_+3A_e">e</code></td>
<td>
<p>Error object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Boolean indicating if error matches
</p>

<hr>
<h2 id='is.glm.nb.theta.warning'>is.glm.nb.theta.warning</h2><span id='topic+is.glm.nb.theta.warning'></span>

<h3>Description</h3>

<p>Check if a warning matches the square root warning raised by <code>glm.nb</code> due to an inflated theta estimate.
This happens when the variance of the negative binomial does not exceed the mean (i.e. there is no overdispersion).
In such cases, the Poisson distribution may be a suitable alternative.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.glm.nb.theta.warning(w)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.glm.nb.theta.warning_+3A_w">w</code></td>
<td>
<p>Warning object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Boolean indicating if warning matches
</p>

<hr>
<h2 id='model.rows.sanity.check'>model.rows.sanity.check</h2><span id='topic+model.rows.sanity.check'></span>

<h3>Description</h3>

<p>Check that the model fit contains the same number of rows as the data used to fit it, 
and throw an error if not
</p>


<h3>Usage</h3>

<pre><code class='language-R'>model.rows.sanity.check(model.data, model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="model.rows.sanity.check_+3A_model.data">model.data</code></td>
<td>
<p>Data used to fit model</p>
</td></tr>
<tr><td><code id="model.rows.sanity.check_+3A_model">model</code></td>
<td>
<p>Resulting negative binomial model object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>

<hr>
<h2 id='model.try.catch'>model.try.catch</h2><span id='topic+model.try.catch'></span>

<h3>Description</h3>

<p>Internal function for fitting model within a tryCatch loop, handling numerical errors gracefully.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>model.try.catch(
  model.formula,
  data,
  distribution = "negative-binomial",
  maxit = 100,
  epsilon = 1e-08,
  init.theta = NULL,
  start = NULL,
  trace = FALSE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="model.try.catch_+3A_model.formula">model.formula</code></td>
<td>
<p>formula</p>
</td></tr>
<tr><td><code id="model.try.catch_+3A_data">data</code></td>
<td>
<p>model data</p>
</td></tr>
<tr><td><code id="model.try.catch_+3A_distribution">distribution</code></td>
<td>
<p>Name of distribution of the counts. Options are 'negative-binomial', 
'poisson', 'truncated-poisson', and 'truncated-negative-binomial'</p>
</td></tr>
<tr><td><code id="model.try.catch_+3A_maxit">maxit</code></td>
<td>
<p>Maximum number of IWLS iterations for fitting the model (passed to <code>glm.control</code>)</p>
</td></tr>
<tr><td><code id="model.try.catch_+3A_epsilon">epsilon</code></td>
<td>
<p>Positive convergence tolerance for Poisson and negative binomial models. Passed to <code>glm.control</code></p>
</td></tr>
<tr><td><code id="model.try.catch_+3A_init.theta">init.theta</code></td>
<td>
<p>Initial value of theta in negative binomial model</p>
</td></tr>
<tr><td><code id="model.try.catch_+3A_start">start</code></td>
<td>
<p>starting values of coefficients in linear predictor</p>
</td></tr>
<tr><td><code id="model.try.catch_+3A_trace">trace</code></td>
<td>
<p>Logical indicating if output should be produced for each of model fitting procedure. Passed to <code>glm.control</code> or <code>gamlss.control</code></p>
</td></tr>
<tr><td><code id="model.try.catch_+3A_verbose">verbose</code></td>
<td>
<p>Logical indicating whether to print progress reports.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with elements
</p>
<table>
<tr><td><code>model</code></td>
<td>
<p>model object. Set to NULL if no model could be fit.</p>
</td></tr>
<tr><td><code>expected.values</code></td>
<td>
<p>vector of expected values for each element in original data, or vector of NAs if no model could be fit</p>
</td></tr>
<tr><td><code>p.values</code></td>
<td>
<p>vector of p-values for test of significantly higher response than expected, or vector of NAs if no model could be fit</p>
</td></tr>
</table>

<hr>
<h2 id='multiple.testing.correct'>multiple.testing.correct</h2><span id='topic+multiple.testing.correct'></span>

<h3>Description</h3>

<p>Perform multiple testing correction on p-values from interaction test.
By default, multiple testing correction is applied per bait. To change this
to a global multiple testing correction, set <code>bait.level = FALSE</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiple.testing.correct(interaction.data, bait.level = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multiple.testing.correct_+3A_interaction.data">interaction.data</code></td>
<td>
<p>Data table of interaction calls. Must contain columns p.value and bait.id.</p>
</td></tr>
<tr><td><code id="multiple.testing.correct_+3A_bait.level">bait.level</code></td>
<td>
<p>Logical indicating whether multiple testing correction should be performed per bait.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Original data table with new column
</p>
<table>
<tr><td><code>q.value</code></td>
<td>
<p>FDR-corrected p-value</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
	data(bre80);
 results &lt;- fit.model(bre80);	
 adjusted.results &lt;- multiple.testing.correct(results);

## End(Not run)

</code></pre>

<hr>
<h2 id='prepare.data'>prepare.data</h2><span id='topic+prepare.data'></span>

<h3>Description</h3>

<p>Prepare data for running interaction calling. Takes a BAM file and baits and restriction fragments as input, and returns a data table with data ready for analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare.data(
  bam,
  baits,
  fragments,
  replicate.merging.method = "sum",
  include.zeros = c("none", "cis", "all"),
  remove.adjacent = FALSE,
  temp.directory = NULL,
  keep.files = FALSE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare.data_+3A_bam">bam</code></td>
<td>
<p>Path to a BAM file</p>
</td></tr>
<tr><td><code id="prepare.data_+3A_baits">baits</code></td>
<td>
<p>Path to a BED file containing the baits</p>
</td></tr>
<tr><td><code id="prepare.data_+3A_fragments">fragments</code></td>
<td>
<p>Path to a BED file containing all restriction fragments in the genome</p>
</td></tr>
<tr><td><code id="prepare.data_+3A_replicate.merging.method">replicate.merging.method</code></td>
<td>
<p>Method that should be used for merging replicates, if applicable</p>
</td></tr>
<tr><td><code id="prepare.data_+3A_include.zeros">include.zeros</code></td>
<td>
<p>String specifying what zero counts to include. Options are none (default), cis, and all.</p>
</td></tr>
<tr><td><code id="prepare.data_+3A_remove.adjacent">remove.adjacent</code></td>
<td>
<p>Logical indicating whether to remove all reads mapping to adjacent restriction fragments.</p>
</td></tr>
<tr><td><code id="prepare.data_+3A_temp.directory">temp.directory</code></td>
<td>
<p>Directory where temporary files should be stored. Defaults to current directory.</p>
</td></tr>
<tr><td><code id="prepare.data_+3A_keep.files">keep.files</code></td>
<td>
<p>Logical indicating whether to keep temporary files</p>
</td></tr>
<tr><td><code id="prepare.data_+3A_verbose">verbose</code></td>
<td>
<p>Logical indicating whether to print progress reports.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data table object with columns
</p>
<table>
<tr><td><code>target.id</code></td>
<td>
<p>String in chrN:start-end format identifying target fragment</p>
</td></tr>
<tr><td><code>bait.id</code></td>
<td>
<p>String in chrN:start-end format identifying bait fragment</p>
</td></tr>
<tr><td><code>target.chr</code></td>
<td>
<p>Chromosome of target fragment</p>
</td></tr>
<tr><td><code>target.start</code></td>
<td>
<p>Start coordinate of target fragment (zero-based)</p>
</td></tr>
<tr><td><code>target.end</code></td>
<td>
<p>End coordinate of target fragment</p>
</td></tr>
<tr><td><code>bait.chr</code></td>
<td>
<p>Chromosome of bait fragment</p>
</td></tr>
<tr><td><code>bait.start</code></td>
<td>
<p>Start coordinate of bait fragment (zero-based)</p>
</td></tr>
<tr><td><code>bait.end</code></td>
<td>
<p>End coordinate of bait fragment</p>
</td></tr>
<tr><td><code>bait.to.bait</code></td>
<td>
<p>Boolean indicating if the interaction is bait-to-bait (i.e. the fragment listed as target is also a bait)</p>
</td></tr>
<tr><td><code>count</code></td>
<td>
<p>The number of reads linking the two fragments</p>
</td></tr>
<tr><td><code>bait.trans.count</code></td>
<td>
<p>The number of reads linking the bait to fragments in trans (a measure of &quot;interactibility&quot;)</p>
</td></tr>
<tr><td><code>target.trans.count</code></td>
<td>
<p>The number of reads linking the target to fragments in trans (a measure of &quot;interactibility&quot;)</p>
</td></tr>
<tr><td><code>distance</code></td>
<td>
<p>Distance between the midpoints of the bait and target fragments (basepairs). NA for trans interactions</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
if( bedtools.installed() ) {
  bam &lt;- system.file('extdata', 'Bre80_2q35.bam', package = 'chicane');
  baits &lt;- system.file('extdata', '2q35.bed', package = 'chicane');
  fragments &lt;- system.file('extdata', 'GRCh38_HindIII_chr2.bed.gz', package = 'chicane');
  input.data &lt;- prepare.data(
	bam = bam, 
	baits = baits, 
	fragments = fragments
	);
 }


</code></pre>

<hr>
<h2 id='read.bed'>read.bed</h2><span id='topic+read.bed'></span>

<h3>Description</h3>

<p>Read a BED file and return regions in chrN:start-end format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.bed(bed.path, zero.based = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.bed_+3A_bed.path">bed.path</code></td>
<td>
<p>Path to bed file</p>
</td></tr>
<tr><td><code id="read.bed_+3A_zero.based">zero.based</code></td>
<td>
<p>Whether to return ID in zero-based coordinates</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of region IDs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bait.file &lt;- system.file('extdata', '2q35.bed', package = 'chicane');
baits &lt;- read.bed(bait.file);

</code></pre>

<hr>
<h2 id='run.model.fitting'>run.model.fitting</h2><span id='topic+run.model.fitting'></span>

<h3>Description</h3>

<p>Run model fitting procedure for either bait-to-bait or other interactions.
Meant for internal use only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run.model.fitting(
  interaction.data,
  distance.bins = NULL,
  distribution = "negative-binomial",
  bait.to.bait = FALSE,
  adjustment.terms = NULL,
  maxit = 100,
  epsilon = 1e-08,
  cores = 1,
  trace = FALSE,
  verbose = FALSE,
  interim.data.dir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run.model.fitting_+3A_interaction.data">interaction.data</code></td>
<td>
<p>data.table object containing interaction counts. Must contain columns distance, count, and bait_trans_count.</p>
</td></tr>
<tr><td><code id="run.model.fitting_+3A_distance.bins">distance.bins</code></td>
<td>
<p>Number of bins to split distance into. Models are fit separately in each bin.</p>
</td></tr>
<tr><td><code id="run.model.fitting_+3A_distribution">distribution</code></td>
<td>
<p>Name of distribution of the counts. Options are 'negative-binomial', 
'poisson', 'truncated-poisson', and 'truncated-negative-binomial'</p>
</td></tr>
<tr><td><code id="run.model.fitting_+3A_bait.to.bait">bait.to.bait</code></td>
<td>
<p>Logical indicating if model should be fit as bait-to-bait</p>
</td></tr>
<tr><td><code id="run.model.fitting_+3A_adjustment.terms">adjustment.terms</code></td>
<td>
<p>Characted vector of extra terms to adjust for in the model fit</p>
</td></tr>
<tr><td><code id="run.model.fitting_+3A_maxit">maxit</code></td>
<td>
<p>Maximum number of IWLS iterations for fitting the model (passed to <code>glm.control</code>)</p>
</td></tr>
<tr><td><code id="run.model.fitting_+3A_epsilon">epsilon</code></td>
<td>
<p>Positive convergence tolerance for Poisson and negative binomial models. Passed to <code>glm.control</code></p>
</td></tr>
<tr><td><code id="run.model.fitting_+3A_cores">cores</code></td>
<td>
<p>Integer value specifying how many cores to use to fit model for cis-interactions.</p>
</td></tr>
<tr><td><code id="run.model.fitting_+3A_trace">trace</code></td>
<td>
<p>Logical indicating if output should be produced for each of model fitting procedure. Passed to <code>glm.control</code> or <code>gamlss.control</code></p>
</td></tr>
<tr><td><code id="run.model.fitting_+3A_verbose">verbose</code></td>
<td>
<p>Logical indicating whether to print progress reports.</p>
</td></tr>
<tr><td><code id="run.model.fitting_+3A_interim.data.dir">interim.data.dir</code></td>
<td>
<p>Path to directory to store intermediate QC data and plots.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Interactions data with expeceted number of interactions and p-values added.
</p>

<hr>
<h2 id='smart.split'>smart.split</h2><span id='topic+smart.split'></span>

<h3>Description</h3>

<p>Split a data frame into a prespecified number of bins, using 
<code>split</code> and <code>cut</code>. Unlike the default R functions, this does not
fail when asked to split the data into a single bin.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>smart.split(dat, bins)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="smart.split_+3A_dat">dat</code></td>
<td>
<p>Data frame or data table to be split</p>
</td></tr>
<tr><td><code id="smart.split_+3A_bins">bins</code></td>
<td>
<p>Number of bins to split data into</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with <code>bins</code> elements. Each element corresponds to one portion 
of the data
</p>

<hr>
<h2 id='stratified.enrichment.sample'>stratified.enrichment.sample</h2><span id='topic+stratified.enrichment.sample'></span>

<h3>Description</h3>

<p>Generate a stratified sample matching distance distribution of significant interactions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stratified.enrichment.sample(nonsignificant.results, significant.results)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stratified.enrichment.sample_+3A_nonsignificant.results">nonsignificant.results</code></td>
<td>
<p>Data table containing non-significant interactions that should be sampled from</p>
</td></tr>
<tr><td><code id="stratified.enrichment.sample_+3A_significant.results">significant.results</code></td>
<td>
<p>Data table of significant results. Used to determine size of strata in stratified sampling procedure.</p>
</td></tr>
</table>

<hr>
<h2 id='test.enrichment'>test.enrichment</h2><span id='topic+test.enrichment'></span>

<h3>Description</h3>

<p>test.enrichment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test.enrichment(
  interaction.data,
  feature.bed,
  significance.cutoff = 0.05,
  span = 0,
  n = 1000,
  remove.bait.to.bait = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="test.enrichment_+3A_interaction.data">interaction.data</code></td>
<td>
<p>Data table containing details on interactions</p>
</td></tr>
<tr><td><code id="test.enrichment_+3A_feature.bed">feature.bed</code></td>
<td>
<p>BED file with regions of features</p>
</td></tr>
<tr><td><code id="test.enrichment_+3A_significance.cutoff">significance.cutoff</code></td>
<td>
<p>q-value threshold for significant interactions</p>
</td></tr>
<tr><td><code id="test.enrichment_+3A_span">span</code></td>
<td>
<p>Distance around target restriction fragment to consider. If set to zero (default),
only features that overlap with the restriction fragment itself are considered.</p>
</td></tr>
<tr><td><code id="test.enrichment_+3A_n">n</code></td>
<td>
<p>Number of random samples to consider</p>
</td></tr>
<tr><td><code id="test.enrichment_+3A_remove.bait.to.bait">remove.bait.to.bait</code></td>
<td>
<p>Logical specifying whether to exclude bait-to-bait interactions</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with elements
</p>
<table>
<tr><td><code>observed</code></td>
<td>
<p>observed overlap between significant interactions and features</p>
</td></tr> 
<tr><td><code>random</code></td>
<td>
<p>vector of length <code>n</code> giving overlap between random samples and features</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Erle Holgersen &lt;Erle.Holgersen@icr.ac.uk&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if( bedtools.installed() ) {
  data(bre80);
  ctcf.bed &lt;- system.file('extdata', 'T47D_chr2_CTCF.bed.gz', package = 'chicane');

  results &lt;- chicane(interactions = bre80);
  test.enrichment(results, ctcf.bed, significance.cutoff = 0.25);
} 


</code></pre>

<hr>
<h2 id='verify.interaction.data'>verify.interaction.data</h2><span id='topic+verify.interaction.data'></span>

<h3>Description</h3>

<p>Verify that interaction.data object is in expected format. Throws an error if object does not fit requirements.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>verify.interaction.data(interaction.data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="verify.interaction.data_+3A_interaction.data">interaction.data</code></td>
<td>
<p>Object to be verified.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
