<!DOCTYPE html><html><head><title>Help for package ChIPtest</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ChIPtest}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ChIPtest_1.0-package'>
<p>Nonparametric Methods for Identifying Differential Enrichment Regions with ChIP-seq Data</p></a></li>
<li><a href='#data1'>
<p>Input data (matrix) for condition A</p></a></li>
<li><a href='#data4'>
<p>Input data (matrix) for condition B</p></a></li>
<li><a href='#est.c'>
<p>calculate the biologically relevant value c in the null hypothesis H0: TS=c, in assumption-free nonparametric test</p></a></li>
<li><a href='#NormTransformation'>
<p>Variance-stabilizing transformation (VST) procedure</p></a></li>
<li><a href='#TS_kernel'>
<p>Calculate the Test Statistics for kernel-based nonparametric test.</p></a></li>
<li><a href='#TS_twosample'>
<p>Three Nonparametric Test Statistics for two sample ChIP-seq data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Nonparametric Methods for Identifying Differential Enrichment
Regions with ChIP-Seq Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-07-07</td>
</tr>
<tr>
<td>Author:</td>
<td>Vicky Qian Wu ; Kyoung-Jae Won ; Hongzhe Li &lt;hongzhe@upenn.edu&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Vicky Qian Wu &lt;wuqian7@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Nonparametric Tests to identify the differential enrichment region for two conditions or time-course ChIP-seq data. It includes: data preprocessing function, estimation of a small constant used in hypothesis testing, a kernel-based two sample nonparametric test, two assumption-free  two sample nonparametric test.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 2.15.1)</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-07-20 13:27:43 UTC; Lin Chai</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-07-20 17:59:16</td>
</tr>
</table>
<hr>
<h2 id='ChIPtest_1.0-package'>
Nonparametric Methods for Identifying Differential Enrichment Regions with ChIP-seq Data
</h2><span id='topic+ChIPtest_1.0-package'></span><span id='topic+ChIPtest_1.0'></span>

<h3>Description</h3>

<p>Nonparametric Tests to identify the differential enrichment region for two conditions or time-course ChIP-seq data. It includes: data preprocessing function, estimation of a small constant used in hypothesis testing, a kernel-based two sample nonparametric test, two assumption-free two sample nonparametric test.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> ChIPtest_1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-07-07</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL ( &gt;=2 </td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Vicky Qian Wu ; Kyoung-Jae Won ; Hongzhe Li &lt;hongzhe@upenn.edu&gt;
Maintainer: Vicky Qian Wu &lt;wuqian7@gmail.com&gt;
</p>


<h3>References</h3>

<p>Qian Wu, Kyoung-Jae Won and Hongzhe Li. (2015) Nonparametric Methods for Identifying Differential Enrichment Regions with ChIP-seq Data. <code class="reqn">Cancer Informatics</code>,14 (Suppl 1), 11-22
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data1)
data(data4)
Data1=NormTransformation(data1)
Data4=NormTransformation(data4)
tao=est.c(Data1, Data4, max1=5, max4=5)
band=54
TS=TS_twosample(Data1, Data4, tao, band, quant=c(0.9,0.9,0.9))
</code></pre>

<hr>
<h2 id='data1'>
Input data (matrix) for condition A
</h2><span id='topic+data1'></span>

<h3>Description</h3>

<p>In order to identify the genes that show differential histone modification levels between the two conditions, condition A and condition B, ChIPtest consider the upstream 5000 bp region and downstream 2000 bp region around the transcription start site (TSS) for each gene and divide the regions into 280 bins of 25 bps. Since the two ChIP-seq samples are usually sequenced at different depths (total number of reads), the counts were rescaled according to the sequencing depth ratio. In this example, suppose that there are 5 genes and for each gene, there are 280 observations. The input data matrix has 5 rows and 280 columns. Each row represents for one gene, and each column represents for number of short reads covered at one bin after rescaling. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(data1)</code></pre>


<h3>Format</h3>

<p>The format is:
</p>
<p>num [1:5, 1:280] 0 0 0 0 1.43 0 1.43 0 0 1.43 ...
</p>
<p>- attr(*, &quot;dimnames&quot;)=List of 2
</p>
<p>..$ : chr [1:5] &quot;1&quot; &quot;2&quot; &quot;3&quot; &quot;4&quot; ...
</p>
<p>..$ : chr [1:280] &quot;V3&quot; &quot;V4&quot; &quot;V5&quot; &quot;V6&quot; ...
</p>


<h3>Source</h3>

<p>T.S. Mikkelsen, et al. Comparative Epigenomic Analysis of Murine and Human Adipogenesis. Cell, 143 (156-169): 1156-1166 (2010)
</p>


<h3>References</h3>

<p>Qian Wu, Kyoung-Jae Won and Hongzhe Li. (2015) Nonparametric Methods for Identifying Differential Enrichment Regions with ChIP-seq Data. <code class="reqn">Cancer Informatics</code>,14 (Suppl 1), 11-22</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data1)
</code></pre>

<hr>
<h2 id='data4'>
Input data (matrix) for condition B
</h2><span id='topic+data4'></span>

<h3>Description</h3>

<p>In order to identify the genes that show differential histone modification levels between the two conditions, condition A and condition B, ChIPtest consider the upstream 5000 bp region and downstream 2000 bp region around the transcription start site (TSS) for each gene and divide the regions into 280 bins of 25 bps. Since the two ChIP-seq samples are usually sequenced at different depths (total number of reads), the counts were rescaled according to the sequencing depth ratio. In this example, suppose that there are 5 genes and for each gene, there are 280 observations. The input data matrix has 5 rows and 280 columns. Each row represents for one gene, and each column represents for number of short reads covered at one bin after rescaling. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(data4)</code></pre>


<h3>Format</h3>

<p>The format is:
num [1:5, 1:280] 0 0 0 0 0 0 0 0 0 0 ...
- attr(*, &quot;dimnames&quot;)=List of 2
..$ : chr [1:5] &quot;1&quot; &quot;2&quot; &quot;3&quot; &quot;4&quot; ...
..$ : chr [1:280] &quot;V3&quot; &quot;V4&quot; &quot;V5&quot; &quot;V6&quot; ...
</p>


<h3>Source</h3>

<p>T.S. Mikkelsen, et al. Comparative Epigenomic Analysis of Murine and Human Adipogenesis. Cell, 143 (156-169): 1156-1166 (2010)
</p>


<h3>References</h3>

<p>Qian Wu, Kyoung-Jae Won and Hongzhe Li. (2015) Nonparametric Methods for Identifying Differential Enrichment Regions with ChIP-seq Data. <code class="reqn">Cancer Informatics</code>,14 (Suppl 1), 11-22</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data4)
</code></pre>

<hr>
<h2 id='est.c'>
calculate the biologically relevant value c in the null hypothesis H0: TS=c, in assumption-free nonparametric test
</h2><span id='topic+est.c'></span>

<h3>Description</h3>

<p>If there is no INPUT experiment (No control), treat the genes with read counts fewer than 5 as the &quot;null genes&quot;. Test statistics were calculated based on the those &quot;null genes&quot; and take the average to obtain the value c, which is used in the null hypothesis H0: TS=c
</p>


<h3>Usage</h3>

<pre><code class='language-R'>est.c(data1, data4, max1 = 5, max4 = 5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="est.c_+3A_data1">data1</code></td>
<td>

<p>Data Matrix (after VST) for condition A
</p>
</td></tr>
<tr><td><code id="est.c_+3A_data4">data4</code></td>
<td>

<p>Data Matrix (after VST) for condition B
</p>
</td></tr>
<tr><td><code id="est.c_+3A_max1">max1</code></td>
<td>

<p>Threshold used to decide null genes for condition A. Default as 5
</p>
</td></tr>
<tr><td><code id="est.c_+3A_max4">max4</code></td>
<td>

<p>Threshold used to decide null genes for condition B. Default as 5
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Data matrix, the default format is N row by M column. Each row represents for one gene, and each column represents for one bin
</p>


<h3>Value</h3>

<table>
<tr><td><code>tao</code></td>
<td>
<p>value c in the null hypothesis H0: TS=c</p>
</td></tr>
</table>


<h3>References</h3>

<p>Qian Wu, Kyoung-Jae Won and Hongzhe Li. (2015) Nonparametric Methods for Identifying Differential Enrichment Regions with ChIP-seq Data. <code class="reqn">Cancer Informatics</code>,14 (Suppl 1), 11-22</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data1)
data(data4)
Data1=NormTransformation(data1)
Data4=NormTransformation(data4)
tao=est.c(Data1, Data4, max1=5, max4=5)
</code></pre>

<hr>
<h2 id='NormTransformation'>
Variance-stabilizing transformation (VST) procedure
</h2><span id='topic+NormTransformation'></span>

<h3>Description</h3>

<p>Assume observed data approximately follow Poisson Distribution. Apply VST procedure to transform the data as approximate normal with constant variance of 1
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NormTransformation(data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NormTransformation_+3A_data">data</code></td>
<td>

<p>Input ChIP-seq data (counts), which approximately follow Poisson Distribution.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please note the input data can be a single value, a vector or a matrix. If it is a matrix, the default format is N row by M column. Each row represents for one gene, and each column represents for one bin.
</p>


<h3>Value</h3>

<p>After VST transformation, the return data matrix would follow Normal Distribution with a constant variance 1
</p>


<h3>References</h3>

<p>Qian Wu, Kyoung-Jae Won and Hongzhe Li. (2015) Nonparametric Methods for Identifying Differential Enrichment Regions with ChIP-seq Data. <code class="reqn">Cancer Informatics</code>,14 (Suppl 1), 11-22</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data1)
Data1=NormTransformation(data1)
</code></pre>

<hr>
<h2 id='TS_kernel'>
Calculate the Test Statistics for kernel-based nonparametric test.
</h2><span id='topic+TS_kernel'></span>

<h3>Description</h3>

<p>Get the difference between two conditions. Apply Kernel smoothing to fit a smooth curve. Estimate variance for each gene and improve the estimation of variance based on all the genes. Derive test statistics and get the rank list of all the genes. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TS_kernel(data, band, quantile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TS_kernel_+3A_data">data</code></td>
<td>

<p>difference matrix between two conditions
</p>
</td></tr>
<tr><td><code id="TS_kernel_+3A_band">band</code></td>
<td>

<p>bandwidth used in kernel smoothing
</p>
</td></tr>
<tr><td><code id="TS_kernel_+3A_quantile">quantile</code></td>
<td>

<p>threshold used in variance estimation
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note 1: Need to chose a bandwidth. Do not recommend to use cross validation (not gene-specific bandwidth) but chose a fixed biological meaningful bandwidth. A fixed bandwidth which can capture the signal profile and smooth out noise would be recommend. The bandwidth used in reference is 20/280.
</p>
<p>Note 2: quantile value is based on the distribution of variance estimation of each gene. Recommend to use histogram to double check the distribution. Default 0.9 = 90 %
</p>


<h3>Value</h3>

<table>
<tr><td><code>TS</code></td>
<td>
<p> Kernel based test statistics after WH transformation. Please refer the details in the reference</p>
</td></tr>
<tr><td><code>TS_sign</code></td>
<td>
<p>&quot;+&quot; represent for condition B enriched more than condition A; &quot;-&quot; vice versa</p>
</td></tr>
<tr><td><code>Tmean</code></td>
<td>
<p>Original test statistics, which is calculated as integral of square of kernel estimator</p>
</td></tr>
</table>


<h3>References</h3>

<p>Qian Wu, Kyoung-Jae Won and Hongzhe Li. (2015) Nonparametric Methods for Identifying Differential Enrichment Regions with ChIP-seq Data. <code class="reqn">Cancer Informatics</code>,14 (Suppl 1), 11-22</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data1)
data(data4)
Data1=NormTransformation(data1)
Data4=NormTransformation(data4)
data=Data4-Data1
band=54
TS=TS_kernel(data, band, quantile=0.9)

</code></pre>

<hr>
<h2 id='TS_twosample'>
Three Nonparametric Test Statistics for two sample ChIP-seq data
</h2><span id='topic+TS_twosample'></span>

<h3>Description</h3>

<p>It includes three nonparametric test statistics for two sample differential analysis: kernel based nonparametric test, assumption-free nonparametric test with equal variance estimation and unequal variance estimation.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TS_twosample(data1, data4, tao, band, quant)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TS_twosample_+3A_data1">data1</code></td>
<td>

<p>data matrix (after VST) for condition A
</p>
</td></tr>
<tr><td><code id="TS_twosample_+3A_data4">data4</code></td>
<td>

<p>data matrix (after VST) for condition B
</p>
</td></tr>
<tr><td><code id="TS_twosample_+3A_tao">tao</code></td>
<td>

<p>the biologically relevant value c in the null hypothesis H0: TS=c, in assumption-free nonparametric test
</p>
</td></tr>
<tr><td><code id="TS_twosample_+3A_band">band</code></td>
<td>

<p>bandwidth used in kernel smoothing
</p>
</td></tr>
<tr><td><code id="TS_twosample_+3A_quant">quant</code></td>
<td>

<p>threshold used in variance estimation
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>kernel-based test statistics is the same as &quot;TS_kernel&quot;
</p>


<h3>Value</h3>

<table>
<tr><td><code>TS_kn</code></td>
<td>
<p>kernel based test statistics</p>
</td></tr>
<tr><td><code>Deql</code></td>
<td>
<p>assumption-free nonparametric test statistics with equal variance</p>
</td></tr>
<tr><td><code>Dnun</code></td>
<td>
<p>assumption-free nonparametric test statistics with unequal variance</p>
</td></tr>
<tr><td><code>sigma1</code></td>
<td>
<p>variance estimation for conditon A under equal variance assumption</p>
</td></tr>
<tr><td><code>sigma4</code></td>
<td>
<p>variance estimation for condition B under unequal variance assumption</p>
</td></tr>
<tr><td><code>Ts_yvec</code></td>
<td>
<p>Original statistics, which is calculated as integral of square of kernel estimator</p>
</td></tr>
<tr><td><code>Dsum</code></td>
<td>
<p>Original statistics, which is calculated for nonparametric test without smoothing</p>
</td></tr>
<tr><td><code>Sev</code></td>
<td>
<p> variance estimation under equal variance assumption </p>
</td></tr>
<tr><td><code>Suv</code></td>
<td>
<p> variance estimation under unequal variance assumption </p>
</td></tr>
<tr><td><code>Xg</code></td>
<td>
<p> estimation of standard deviation for kernel-based test statistics</p>
</td></tr>
</table>


<h3>References</h3>

<p>Qian Wu, Kyoung-Jae Won and Hongzhe Li. (2015) Nonparametric Methods for Identifying Differential Enrichment Regions with ChIP-seq Data. <code class="reqn">Cancer Informatics</code>,14 (Suppl 1), 11-22</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data1)
data(data4)
Data1=NormTransformation(data1)
Data4=NormTransformation(data4)
tao=est.c(Data1, Data4, max1=5, max4=5)
band=54
TS=TS_twosample(Data1, Data4, tao, band, quant=c(0.9,0.9,0.9))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
